<!DOCTYPE html><html lang="en"><head><title>Help for package RPANDA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RPANDA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RPANDA-package'>
<p>RPANDA</p></a></li>
<li><a href='#add.gts'>
<p>Geological time scale</p></a></li>
<li><a href='#ancestral'>
<p>Estimation of traits ancestral states.</p></a></li>
<li><a href='#Anolis.data'>
<p>Anolis dataset</p></a></li>
<li><a href='#apply_prob_dtt'>
<p>Calculates paleodiversity dynamics with the probabilistic approach.</p></a></li>
<li><a href='#Balaenopteridae'><p>Balaenopteridae phylogeny</p></a></li>
<li><a href='#BGB.examples'>
<p>BioGeoBEARS stochastic maps</p></a></li>
<li><a href='#BICompare'>
<p>Identify modalities in a phylogeny</p></a></li>
<li><a href='#build_network.BipartiteEvol'>
<p>Build the interaction network in BipartiteEvol</p></a></li>
<li><a href='#Calomys'><p>Calomys phylogeny</p></a></li>
<li><a href='#Caprimulgidae'>
<p>The <em>Caprimulgidae</em> phylogeny.</p></a></li>
<li><a href='#Caprimulgidae_ClaDS2'>
<p>An example run of ClaDS2.</p></a></li>
<li><a href='#Cetacea'><p>Cetacean phylogeny</p></a></li>
<li><a href='#Cetacea_clades'><p>Stochastic map of clade membership in Cetacean phylogeny</p></a></li>
<li><a href='#ClaDS0_example'>
<p>An example run of ClaDS0.</p></a></li>
<li><a href='#co2'><p>co2 data since the Jurassic</p></a></li>
<li><a href='#co2_res'><p>co2 data since the beginning of the Cenozoic</p></a></li>
<li><a href='#coccolithophore'><p>Coccolithophore diversity since the Jurassic</p></a></li>
<li><a href='#CreateClassObject'>
<p>Create class object</p></a></li>
<li><a href='#CreateGeobyClassObject'>
<p>Create merged biogeography-by-class object</p></a></li>
<li><a href='#CreateGeoObject'>
<p>Create biogeography object</p></a></li>
<li><a href='#CreateGeoObject_BioGeoBEARS'>
<p>Create biogeography object using a stochastic map from BioGeoBEARS</p></a></li>
<li><a href='#createModel'>
<p>Creation of a PhenotypicModel</p></a></li>
<li><a href='#createModelCoevolution'>
<p>Creation of a PhenotypicGMM</p></a></li>
<li><a href='#d13c'><p>d13c data since the Jurassic</p></a></li>
<li><a href='#define_species.BipartiteEvol'>
<p>Build the phylogenies for BipartiteEvol</p></a></li>
<li><a href='#delineate_phylotypes'>
<p>Automatic phylotypes delineation</p></a></li>
<li><a href='#div.models'>
<p>Model comparison of diversification models</p></a></li>
<li><a href='#div.rates'>
<p>Diversification rates through time</p></a></li>
<li><a href='#fit_bd'>
<p>Maximum likelihood fit of the general birth-death model</p></a></li>
<li><a href='#fit_bd_backbone'>
<p>Maximum likelihood fit of the general birth-death model (backbone)</p></a></li>
<li><a href='#fit_bd_backbone_c'>
<p>Maximum likelihood fit of the general birth-death model (backbone and constraints)</p></a></li>
<li><a href='#fit_bd_in_past'>
<p>Maximum likelihood fit of the general birth-death model excluding the recent past</p></a></li>
<li><a href='#fit_ClaDS'>
<p>Fit ClaDS to a phylogeny</p></a></li>
<li><a href='#fit_ClaDS0'>
<p>Infer ClaDS0's parameter on a phylogeny</p></a></li>
<li><a href='#fit_coal_cst'>
<p>Maximum likelihood fit of the equilibrium model</p></a></li>
<li><a href='#fit_coal_var'>
<p>Fit birth-death model using a coalescent approch</p></a></li>
<li><a href='#fit_env'>
<p>Maximum likelihood fit of the environmental birth-death model</p></a></li>
<li><a href='#fit_env_in_past'>
<p>Maximum likelihood fit of the environmental birth-death model excluding the recent past</p></a></li>
<li><a href='#fit_t_comp'>
<p>Fits models of trait evolution incorporating competitive interactions</p></a></li>
<li><a href='#fit_t_comp_subgroup'>
<p>Fits models of trait evolution incorporating competitive interactions, restricting competition to occur only between members of a subgroup</p></a></li>
<li><a href='#fit_t_env'>
<p>Maximum likelihood fit of the environmental model of trait evolution</p></a></li>
<li><a href='#fit_t_env_ou'>
<p>Maximum likelihood fit of the OU environmental model of trait evolution</p></a></li>
<li><a href='#fit_t_pl'>
<p>High-dimensional phylogenetic models of trait evolution</p></a></li>
<li><a href='#fit_t_standard'>
<p>Fits standard models of trait evolution incorporating known and nuisance measurement error</p></a></li>
<li><a href='#fitTipData'>
<p>Maximum likelihood estimators of a model's parameters</p></a></li>
<li><a href='#fitTipData-methods'><p> ~~ Methods for Function <code>fitTipData</code>  ~~</p></a></li>
<li><a href='#foraminifera'><p>Foraminifera diversity since the Jurassic</p></a></li>
<li><a href='#get.comb.shift'>
<p>Combinations of shifts of diversification.</p></a></li>
<li><a href='#get.sampling.fractions'>
<p>Sampling fractions of subclades</p></a></li>
<li><a href='#getDataLikelihood'>
<p>Likelihood of tip trait values.</p></a></li>
<li><a href='#getDataLikelihood-methods'><p> ~~ Methods for Function <code>getDataLikelihood</code>  ~~</p></a></li>
<li><a href='#getMAPS_ClaDS'>
<p>Gets the Maximum A Posteriori for each ClaDS parameter</p></a></li>
<li><a href='#getMAPS_ClaDS0'>
<p>Gets the Maximum A Posteriori for each ClaDS0 parameter</p></a></li>
<li><a href='#getTipDistribution'>
<p>Distribution of tip trait values.</p></a></li>
<li><a href='#getTipDistribution-methods'><p> Distribution of tip trait values.</p></a></li>
<li><a href='#gic_criterion'>
<p>Generalized Information Criterion (GIC) to compare models fit by Maximum Likelihood (ML) or Penalized Likelihood (PL).</p></a></li>
<li><a href='#GIC.fit_pl.rpanda'>
<p>Generalized Information Criterion (GIC) to compare models fit by Maximum Likelihood (ML) or Penalized Likelihood (PL).</p></a></li>
<li><a href='#greenalgae'><p>Green algae diversity since the Jurassic</p></a></li>
<li><a href='#InfTemp'><p>Paleotemperature data across the Cenozoic</p></a></li>
<li><a href='#JSDt_cluster'>
<p>Clustering on the Jensen-Shannon distance between phylogenetic trait data</p></a></li>
<li><a href='#JSDtree'>
<p>Jensen-Shannon distance between phylogenies</p></a></li>
<li><a href='#JSDtree_cluster'>
<p>Clustering of phylogenies</p></a></li>
<li><a href='#landplant'><p>Land plant diversity since the Jurassic</p></a></li>
<li><a href='#likelihood_bd'>
<p>Likelihood of a phylogeny under the general birth-death model</p></a></li>
<li><a href='#likelihood_bd_backbone'>
<p>Likelihood of a phylogeny under the general birth-death model (backbone)</p></a></li>
<li><a href='#likelihood_coal_cst'>
<p>Likelihood of a phylogeny under the equilibrium diversity model</p></a></li>
<li><a href='#likelihood_coal_var'>
<p>Likelihood of a birth-death model using a coalescent approch</p></a></li>
<li><a href='#likelihood_subgroup_model'>
<p>Likelihood of a dataset under models with biogeography fit to a subgroup.</p></a></li>
<li><a href='#likelihood_t_DD'>
<p>Likelihood of a dataset under diversity-dependent models.</p></a></li>
<li><a href='#likelihood_t_DD_geog'>
<p>Likelihood of a dataset under diversity-dependent models with biogeography.</p></a></li>
<li><a href='#likelihood_t_env'>
<p>Likelihood of a dataset under environmental models of trait evolution.</p></a></li>
<li><a href='#likelihood_t_MC'>
<p>Likelihood of a dataset under the matching competition model.</p></a></li>
<li><a href='#likelihood_t_MC_geog'>
<p>Likelihood of a dataset under the matching competition model with biogeography.</p></a></li>
<li><a href='#lines.fit_t.env'>
<p>Add to a plot line segments joining the phenotypic evolutionary rate through time estimated by the <em>fit_t_env</em> function</p></a></li>
<li><a href='#lines.fit_t.env.ou'>
<p>Add to a plot line segments joining the phenotypic evolutionary optimum through time estimated by the <em>fit_t_env_ou</em> function</p></a></li>
<li><a href='#make_gen.BipartiteEvol'>
<p>Compute the genealogies for BipartiteEvol</p></a></li>
<li><a href='#mantel_test'>
<p>Compute Mantel test</p></a></li>
<li><a href='#mantel_test_nbpartners'>
<p>Compute Mantel test</p></a></li>
<li><a href='#modelSelection'>
<p>Phenotypic model selection from tip trait data.</p></a></li>
<li><a href='#modelSelection-methods'><p> ~~ Methods for Function <code>modelSelection</code>  ~~</p></a></li>
<li><a href='#MPhiFFT'>
<p>A class used internally to compute ClaDS's likelihood</p></a></li>
<li><a href='#mycorrhizal_network'><p>Mycorrhizal network from La Réunion island</p></a></li>
<li><a href='#ostracoda'><p>Ostracod diversity since the Jurassic</p></a></li>
<li><a href='#paleodiv'>
<p>Paleodiversity through time</p></a></li>
<li><a href='#PhenotypicACDC-class'><p>Class <code>"PhenotypicACDC"</code></p></a></li>
<li><a href='#PhenotypicADiag-class'><p>Class <code>"PhenotypicADiag"</code></p></a></li>
<li><a href='#PhenotypicBM-class'><p>Class <code>"PhenotypicBM"</code></p></a></li>
<li><a href='#PhenotypicDD-class'><p>Class <code>"PhenotypicDD"</code></p></a></li>
<li><a href='#PhenotypicGMM-class'><p>Class <code>"PhenotypicGMM"</code></p></a></li>
<li><a href='#PhenotypicModel-class'><p>Class <code>"PhenotypicModel"</code></p></a></li>
<li><a href='#PhenotypicOU-class'><p>Class <code>"PhenotypicOU"</code></p></a></li>
<li><a href='#PhenotypicPM-class'><p>Class <code>"PhenotypicPM"</code></p></a></li>
<li><a href='#Phocoenidae'><p>Phocoenidae phylogeny</p></a></li>
<li><a href='#phyl.pca_pl'>
<p>Regularized Phylogenetic Principal Component Analysis (PCA).</p></a></li>
<li><a href='#Phyllostomidae'><p>Phyllostomidae phylogeny</p></a></li>
<li><a href='#Phyllostomidae_genera'><p>Phylogenies of Phyllostomidae genera</p></a></li>
<li><a href='#phylosignal_network'>
<p>Compute phylogenetic signal in a bipartite interaction network</p></a></li>
<li><a href='#phylosignal_sub_network'>
<p>Compute clade-specific phylogenetic signals in a bipartite interaction network</p></a></li>
<li><a href='#pi_estimator'>
<p>Compute nucleotidic diversity (Pi estimator)</p></a></li>
<li><a href='#plot_BICompare'>
<p>Display modalities on a phylogeny.</p></a></li>
<li><a href='#plot_ClaDS_chains'>
<p>Plot the MCMC chains obtained when infering ClaDS parameters</p></a></li>
<li><a href='#plot_ClaDS_phylo'><p>Plot a phylogeny with branch-specific values</p></a></li>
<li><a href='#plot_ClaDS0_chains'>
<p>Plot the MCMC chains obtained when infering ClaDS0 parameters</p></a></li>
<li><a href='#plot_div.BipartiteEvol'>
<p>Plot the output of BipartiteEvol</p></a></li>
<li><a href='#plot_dtt'>
<p>Plot diversity through time</p></a></li>
<li><a href='#plot_fit_bd'>
<p>Plot speciation, extinction &amp; net diversification rate functions of a fitted model</p></a></li>
<li><a href='#plot_fit_env'>
<p>Plot speciation, extinction &amp; net diversification rate functions of a fitted environmental model</p></a></li>
<li><a href='#plot_net.BipartiteEvol'>
<p>Plot the output of BipartiteEvol</p></a></li>
<li><a href='#plot_phylo_comb'>
<p>Plot shifts of diversifcation on a phylogeny</p></a></li>
<li><a href='#plot_phylosignal_sub_network'>
<p>Plot clade-specific phylogenetic signals in a bipartite interaction network</p></a></li>
<li><a href='#plot_prob_dtt'>
<p>Plot diversity through time with confidence intervals.</p></a></li>
<li><a href='#plot_spectR'>
<p>Spectral density plot of a phylogeny.</p></a></li>
<li><a href='#plot.fit_t.env'>
<p>Plot the phenotypic evolutionary rate through time estimated by the <em>fit_t_env</em> function</p></a></li>
<li><a href='#plot.fit_t.env.ou'>
<p>Plot the phenotypic evolutionary optimum through time estimated by the <em>fit_t_env_ou</em> function</p></a></li>
<li><a href='#Posdef'>
<p>Positive definite symmetric matrices</p></a></li>
<li><a href='#prob_dtt'>
<p>Confidence intervals of diversity through time</p></a></li>
<li><a href='#radiolaria'><p>Radiolaria diversity since the Jurassic</p></a></li>
<li><a href='#redalgae'><p>Red algae diversity since the Jurassic</p></a></li>
<li><a href='#remove.model'>
<p>Removing a model from shift.estimates output</p></a></li>
<li><a href='#sealevel'><p>Sea level data since the Jurassic</p></a></li>
<li><a href='#shift.estimates'>
<p>Estimating clade-shifts of diversification</p></a></li>
<li><a href='#shifts_cetacea'><p>Cetacean shift.estimates results</p></a></li>
<li><a href='#silica'><p>Silica data across the Cenozoic</p></a></li>
<li><a href='#sim_ClaDS'>
<p>Simulation of the ClaDS model</p></a></li>
<li><a href='#sim_env_bd'>
<p>Simulate birth-death tree dependent on an environmental curve</p></a></li>
<li><a href='#sim_MCBD'>
<p>Simulation of macroevolutionary diversification under the integrated model described in Aristide &amp; Morlon 2019</p></a></li>
<li><a href='#sim_sgd'>
<p>Algorithm for simulating a phylogenetic tree under the SGD model</p></a></li>
<li><a href='#sim_t_comp'>
<p>Recursive simulation (root-to-tip) of competition models</p></a></li>
<li><a href='#sim_t_env'>
<p>Recursive simulation (root-to-tip) of the environmental model</p></a></li>
<li><a href='#sim_t_env_ou'>
<p>Recursive simulation (root-to-tip) of the OU environmental model</p></a></li>
<li><a href='#sim_t_tworegime'>
<p>Recursive simulation (root-to-tip) of two-regime models</p></a></li>
<li><a href='#sim.BipartiteEvol'>
<p>Simulation of the BipartiteEvol model</p></a></li>
<li><a href='#sim.convergence.geo'>
<p>Simulation of trait data under the model of convergent character displacement described in Drury et al. 2017</p></a></li>
<li><a href='#sim.divergence.geo'>
<p>Simulation of trait data under the model of divergent character displacement described in Drury et al. 2017</p></a></li>
<li><a href='#simul.comb.shift'>
<p>Simulating trees from shift.estimates() results to test model adequacy</p></a></li>
<li><a href='#simulateTipData'>
<p>Tip trait simulation under a model of phenotypic evolution.</p></a></li>
<li><a href='#simulateTipData-methods'><p> ~~ Methods for Function <code>simulateTipData</code>  ~~</p></a></li>
<li><a href='#spectR'>
<p>Spectral density plot of a phylogeny</p></a></li>
<li><a href='#spectR_t'>
<p>Spectral density plot of phylogenetic trait data</p></a></li>
<li><a href='#taxo_cetacea'><p>Cetacean taxonomy</p></a></li>
<li><a href='#theta_estimator'>
<p>Compute Watterson genetic diversity (Theta estimator)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>2.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-01-14</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Phylogenetic ANalyses of DiversificAtion</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), ape, picante, methods,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>Imports:</td>
<td>BB, bipartite, cluster, coda, deSolve, fields, fpc, geiger,
glassoFast, graphics, grDevices, igraph, Matrix, mvtnorm,
mvMORPH (&ge; 1.1.6), GUniFrac, parallel, ParallelLogger,
phangorn, phytools, pracma, pspline, pvclust, raster,
RColorBrewer, R.utils, Rmpfr, stats, TESS, utils, vegan</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Author:</td>
<td>Hélène Morlon [aut, cre, cph],
  Eric Lewitus [aut, cph],
  Fabien Condamine [aut, cph],
  Marc Manceau [aut, cph],
  Julien Clavel [aut, cph],
  Jonathan Drury [aut, cph],
  Olivier Billaud [aut, cph],
  Odile Maliet [aut, cph],
  Leandro Aristide [aut, cph],
  Benoit Perez-Lamarque [aut, cph],
  Nathan Mazet [aut, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hélène Morlon &lt;helene.morlon@bio.ens.psl.eu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements macroevolutionary analyses on phylogenetic trees. See
    Morlon et al. (2010) &lt;<a href="https://doi.org/10.1371%2Fjournal.pbio.1000493">doi:10.1371/journal.pbio.1000493</a>&gt;, Morlon et al. (2011)
    &lt;<a href="https://doi.org/10.1073%2Fpnas.1102543108">doi:10.1073/pnas.1102543108</a>&gt;, Condamine et al. (2013) &lt;<a href="https://doi.org/10.1111%2Fele.12062">doi:10.1111/ele.12062</a>&gt;,
    Morlon et al. (2014) &lt;<a href="https://doi.org/10.1111%2Fele.12251">doi:10.1111/ele.12251</a>&gt;, Manceau et al. (2015) &lt;<a href="https://doi.org/10.1111%2Fele.12415">doi:10.1111/ele.12415</a>&gt;,
    Lewitus &amp; Morlon (2016) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyv116">doi:10.1093/sysbio/syv116</a>&gt;, Drury et al. (2016) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyw020">doi:10.1093/sysbio/syw020</a>&gt;,
    Manceau et al. (2016) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyw115">doi:10.1093/sysbio/syw115</a>&gt;, Morlon et al. (2016) &lt;<a href="https://doi.org/10.1111%2F2041-210X.12526">doi:10.1111/2041-210X.12526</a>&gt;, Clavel &amp; Morlon (2017) &lt;<a href="https://doi.org/10.1073%2Fpnas.1606868114">doi:10.1073/pnas.1606868114</a>&gt;,
    Drury et al. (2017) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyx079">doi:10.1093/sysbio/syx079</a>&gt;, Lewitus &amp; Morlon (2017) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyx095">doi:10.1093/sysbio/syx095</a>&gt;,
    Drury et al. (2018) &lt;<a href="https://doi.org/10.1371%2Fjournal.pbio.2003563">doi:10.1371/journal.pbio.2003563</a>&gt;, Clavel et al. (2019) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyy045">doi:10.1093/sysbio/syy045</a>&gt;, Maliet et al. (2019) &lt;<a href="https://doi.org/10.1038%2Fs41559-019-0908-0">doi:10.1038/s41559-019-0908-0</a>&gt;,
    Billaud et al. (2019) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyz057">doi:10.1093/sysbio/syz057</a>&gt;, Lewitus et al. (2019) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyz061">doi:10.1093/sysbio/syz061</a>&gt;,
    Aristide &amp; Morlon (2019) &lt;<a href="https://doi.org/10.1111%2Fele.13385">doi:10.1111/ele.13385</a>&gt;, Maliet et al. (2020) &lt;<a href="https://doi.org/10.1111%2Fele.13592">doi:10.1111/ele.13592</a>&gt;, Drury et al. (2021) &lt;<a href="https://doi.org/10.1371%2Fjournal.pbio.3001270">doi:10.1371/journal.pbio.3001270</a>&gt;,
    Perez-Lamarque &amp; Morlon (2022) &lt;<a href="https://doi.org/10.1111%2Fmec.16478">doi:10.1111/mec.16478</a>&gt;, Perez-Lamarque et al. (2022) &lt;<a href="https://doi.org/10.1101%2F2021.08.30.458192">doi:10.1101/2021.08.30.458192</a>&gt;, Mazet et al. (2023) &lt;<a href="https://doi.org/10.1111%2F2041-210X.14195">doi:10.1111/2041-210X.14195</a>&gt;, Drury et al. (2024) &lt;<a href="https://doi.org/10.1016%2Fj.cub.2023.12.055">doi:10.1016/j.cub.2023.12.055</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/hmorlon/PANDA">https://github.com/hmorlon/PANDA</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-14 17:11:06 UTC; morlon</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-14 23:40:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='RPANDA-package'>
RPANDA
</h2><span id='topic+RPANDA-package'></span><span id='topic+RPANDA'></span>

<h3>Description</h3>

<p>Implements macroevolutionary analyses on phylogenetic trees
</p>


<h3>Details</h3>

<p>More information on the RPANDA package and worked examples can be found in <a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12526">Morlon et al. (2016)</a>
</p>


<h3>Author(s)</h3>

<p>Hélène Morlon &lt;helene.morlon@bio.ens.psl.eu&gt;
</p>
<p>Julien Clavel &lt;julien.clavel@univ-lyon1.fr&gt;
</p>
<p>Fabien Condamine &lt;fabien.condamine@gmail.com&gt;
</p>
<p>Jonathan Drury &lt;jonathan.p.drury@durham.ac.uk&gt;
</p>
<p>Eric Lewitus &lt;elewitus@hivresearch.org&gt;
</p>
<p>Marc Manceau &lt;marc.manceau@gmail.com&gt;
</p>
<p>Olivier Billaud &lt;olivier.billaud@agroparistech.fr&gt;
</p>
<p>Odile Maliet &lt;maliet@biologie.ens.fr&gt;
</p>
<p>Leandro Aristide &lt;aristide@biologie.ens.fr&gt;
</p>
<p>Benoît Perez-Lamarque &lt;benoit.perez@ens.psl.eu&gt;
</p>


<h3>References</h3>

<p>Morlon, H., Potts, M.D., Plotkin, J.B. (2010) Inferring the dynamics of diversification: a coalescent approach, PLoS B 8(9): e1000493
</p>
<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record, Proc Nat Acad Sci 108: 16327-16332
</p>
<p>Morlon, H., Kemps, B., Plotkin, J.B., Brisson, D. (2012) Explosive radiation of a bacterial species group, Evolution 66: 2577-2586
</p>
<p>Condamine, F.L., Rolland, J., and Morlon, H. (2013) Macroevolutionary perspectives to environmental change, Eco Lett 16: 72-85
</p>
<p>Morlon, H. (2014) Phylogenetic approaches for studying diversification, Eco Lett 7: 508-525
</p>
<p>Manceau, M., Lambert, A., Morlon, H. (2015) Phylogenies support out-of-equilibrium models of biodiversity, Eco Lett 18: 347-356
</p>
<p>Lewitus, E., Morlon, H. (2016) Characterizing and comparing phylogenies from their Laplacian spectrum, Syst Biol 65: 495-507
</p>
<p>Morlon, H., Lewitus, E., Condamine, F.L., Manceau, M., Clavel, J., Drury, J. (2016) RPANDA: an R package for macroevolutionary analyses on phylogenetic trees, MEE 7: 589-597
</p>
<p>Drury, J., Clavel, J., Manceau, M., Morlon, H. (2016) Estimating the Effect of Competition on Trait Evolution Using Maximum Likelihood Inference, Syst Biol 65: 700-710
</p>
<p>Manceau, M., Lambert, A., Morlon, H. (2017) A Unifying Comparative Phylogenetic Framework Including Traits Coevolving Across Interacting Lineages, Syst Biol 66: 551-568
</p>
<p>Clavel, J., Morlon, H. (2017) Accelerated body size evolution during cold climatic periods in the Cenozoic, Proc Nat Acad Sci 114: 4183-4188
</p>
<p>Drury, J., Tobias, J., Burns, K., Mason, N., Shultz, A., and Morlon, H. (2018) Contrasting impacts of competition on ecological and social trait evolution in songbirds. PLOS Biolog 16: e2003563
</p>
<p>Clavel, J., Aristide, L., Morlon, H. (2019). A Penalized Likelihood framework for high-dimensional phylogenetic comparative methods and an application to new-world monkeys brain evolution. Syst Biol 68: 93-116
</p>
<p>Maliet, O., Hartig, F., Morlon, H. (2019). A model with many small shifts for estimating species-specific diversification rates. Nature Ecol Evol 3: 1086-1092
</p>
<p>Condamine, F.L., Rolland, J., Morlon, H. (2019) Assessing the causes of diversification slowdowns: temperature-dependent and diversity-dependent models receive equivalent support Ecology Letters 22: 1900-1912
</p>
<p>Aristide, L., Morlon, H. (2019) Understanding the effect of competition during evolutionary radiations: an integrated model of phenotypic and species diversification Ecology Letters 22: 2006-2017
</p>
<p>Billaud, O., Moen, D. S., Parsons, T. L., Morlon, H. (2019) Estimating Diversity Through Time using Molecular Phylogenies: Old and Species-Poor Frog Families are the Remnants of a Diverse Past Systematic Biology 69: 363–383
</p>
<p>Lewitus, E., Aristide, L., Morlon, H. (2019) Characterizing and Comparing Phylogenetic Trait Data from Their Normalized Laplacian Spectrum Systematic Biology 69: 234–248
</p>
<p>Maliet, O., Loeuille, N., Morlon, H. (2020) An individual-based model for the eco-evolutionary emergence of bipartite interaction networks Ecology Letters 
</p>
<p>Perez-Lamarque, B., Öpik, M., Maliet, O., Afonso Silva, A.C., Selosse, M-A., Martos, F., Morlon, H. (2022), Analysing diversification dynamics using barcoding data: The case of an obligate mycorrhizal symbiont, Molecular Ecology, 31:3496–512.
</p>
<p>Perez-Lamarque, B., Maliet, O., Pichon, B., Selosse, M-A., Martos, F., Morlon, H. (2022) Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. bioRxiv, 2021.08.30.458192, ver. 6 peer-reviewed and recommended by Peer Community in Evolutionary Biology.
</p>

<hr>
<h2 id='add.gts'>
Geological time scale
</h2><span id='topic+add.gts'></span>

<h3>Description</h3>

<p>Adds geological time scale (GTS) to plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.gts(thickness, quaternary = TRUE, is.phylo = FALSE,
        xpd.x = TRUE, time.interval = 1, names = NULL, fill = TRUE,
        cex = 1, padj = -0.5, direction = "rightwards")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add.gts_+3A_thickness">thickness</code></td>
<td>
  
<p>numeric &lt; 0. Define the thickness of the scale.
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_quaternary">quaternary</code></td>
<td>

<p>bolean. Whether to merge Pleistocene and Holocene into Quaternary. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_is.phylo">is.phylo</code></td>
<td>

<p>bolean. Whether the plot is a phylogeny or not. Default is FALSE.
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_time.interval">time.interval</code></td>
<td>

<p>numeric. Define the minimum time interval (in million years) for the geological time scale. Default is 1 and displays ticks every million year but with numbers at every five million years.
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_xpd.x">xpd.x</code></td>
<td>

<p>bolean. Whether to expand the last period of the geological time scale before root age (mainly for tree). Default is TRUE.
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_names">names</code></td>
<td>

<p>a character vector with the names of geological periods (stages). Can be used to write abbreviations. Default is NULL and display full names (except for Quaternary and Pliocene). 
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_fill">fill</code></td>
<td>

<p>bolean. If TRUE (default), backbground is alternatively filled with grey and white bands to distinguish geological periods. If FALSE, dashed lines are drawn to limit geological periods.
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_cex">cex</code></td>
<td>

<p>numeric. Size of the names of geological periods.
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_padj">padj</code></td>
<td>

<p>padj argument defining space between the axis and the values of the axis (see par() for more details).
</p>
</td></tr>
<tr><td><code id="add.gts_+3A_direction">direction</code></td>
<td>

<p>character. Direct the geological time scale. Can be either &quot;rightwards&quot; (default) of &quot;leftwards&quot; (NOT IMPLEMENTED YET). 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots a geological times scale (GTS). It has been designed for adding GTS to plot of phylogeny, diversification rates and paleodiversity dynamics through time but can be used with any R plot. Time should be negative for other plots than phylogenies.
</p>


<h3>Value</h3>

<p>Draws geological time scale on x axis.
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# with a phylogeny
data("Cetacea")
oldpar &lt;- par(no.readonly = TRUE)
# first plot to get the dimensions of the gts
plot(Cetacea, cex = 0.5, label.offset = 0.2, tip.color = "white")
add.gts(-3, quaternary = TRUE, is.phylo = TRUE, xpd.x = FALSE,
        names = c("Q.", "Pli.", "Miocene", "Oligocene", "Eoc."))
# second plot to display the tree on the gts
par(new = TRUE)
plot(Cetacea, cex = 0.5, label.offset = 0.2)
mtext("Time (Myrs)", side = 1, line = 3, at = 18)
par(oldpar) # restore the old par

# see Appendix S4 from Mazet et al. (2023) for more examples.


</code></pre>

<hr>
<h2 id='ancestral'>
Estimation of traits ancestral states.
</h2><span id='topic+ancestral'></span>

<h3>Description</h3>

<p>Reconstruct the ancestral states at the root (and possibly for each nodes) of a phylogenetic tree from models fit obtained using the <code>fit_t_XX</code> functions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>
ancestral(object, ...)
  
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ancestral_+3A_object">object</code></td>
<td>

<p>A model fit object obtained by the <code>fit_t_XX</code> class of functions.
</p>
</td></tr>
<tr><td><code id="ancestral_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed through (not used yet).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ancestral</code> reconstructs the ancestral states at the root and possibly for each nodes of a phylogenetic tree from the models fit obtained by the <code>fit_t_XX</code> class of functions (e.g., <code>fit_t_pl</code>, <code>fit_t_comp</code> and <code>fit_t_env</code>). Ancestral states are estimated using generalized least squares (GLS; Martins &amp; Hansen 1997, Cunningham et al. 1998 ).</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>root</code></td>
<td>
<p>the reconstructed ancestral states at the root</p>
</td></tr>
<tr><td><code>nodes</code></td>
<td>
<p>the reconstructed ancestral states at each nodes (not yet implemented for all the methods)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The function is used internally in <code>phyl.pca_pl</code> (Clavel et al. 2019).</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J., Aristide, L., Morlon, H., 2019. A Penalized Likelihood framework for high-dimensional phylogenetic comparative methods and an application to new-world monkeys brain evolution. Syst. Biol. 68: 93-116.
</p>
<p>Cunningham C.W., Omland K.E., Oakley T.H. 1998. Reconstructing ancestral character states: a critical reappraisal. Trends Ecol. Evol. 13:361-366.
</p>
<p>Martins E.P., Hansen T.F. 1997. Phylogenies and the comparative method: a general approach to incorporating phylogenetic information into the analysis of interspecific data. Am. Nat. 149:646-667.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_pl">fit_t_pl</a></code>,
<code><a href="#topic+fit_t_env">fit_t_env</a></code>,
<code><a href="#topic+phyl.pca_pl">phyl.pca_pl</a></code>,
<code><a href="mvMORPH.html#topic+GIC">GIC</a></code>, 
<code><a href="#topic+gic_criterion">gic_criterion</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(mvMORPH)){
set.seed(1)
n &lt;- 32 # number of species
p &lt;- 31 # number of traits

tree &lt;- pbtree(n=n) # phylogenetic tree
R &lt;- Posdef(p)      # a random symmetric matrix (covariance)

# simulate a dataset
Y &lt;- mvSIM(tree, model="BM1", nsim=1, param=list(sigma=R))

# fit a multivariate BM with Penalized likelihood
fit &lt;- fit_t_pl(Y, tree, model="BM", method="RidgeAlt")

# Perform the ancestral states reconstruction
anc &lt;- ancestral(fit)

# retrieve the scores
head(anc$nodes)
}

</code></pre>

<hr>
<h2 id='Anolis.data'>
Anolis dataset
</h2><span id='topic+Anolis.data'></span>

<h3>Description</h3>

<p>Phylogeny, trait data, and geography.object for a subclade of Greater Antillean <em>Anolis</em> lizards. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Anolis.data)
</code></pre>


<h3>Details</h3>

<p>Illustrative phylogeny trimmed from the maximum clade credibility tree of Mahler et al. 2013, corresponding phylogenetic principal component data from Mahler et al. 2013, and biogeography data from Mahler &amp; Ingram 2014 (in the form of a geography object, as detailed in the CreateGeoObject help file). 
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Mahler, D.L., Ingram, T., Revell, L., and Losos, J. 2013. Exceptional convergence on the macroevolutionary landscape in island lizard radiations. <em>Science</em>. 341:292-295.
</p>
<p>Mahler, D.L. and Ingram, T. 2014. Phylogenetic comparative methods for studying clade-wide convergence. In <em>Modern Phylogenetic Comparative Methods and Their Application in Evolutionary Biology</em>, ed. L. Garamszegi. pp.425-450.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Anolis.data)
plot(Anolis.data$phylo)
print(Anolis.data$data)
print(Anolis.data$geography.object)
</code></pre>

<hr>
<h2 id='apply_prob_dtt'>
Calculates paleodiversity dynamics with the probabilistic approach.
</h2><span id='topic+apply_prob_dtt'></span>

<h3>Description</h3>

<p>Applies prob_dtt() to outputs from shift.estimates().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_prob_dtt(phylo, data, sampling.fractions, shift.res,
               combi = 1, backbone.option = "crown.shift",
               m = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="apply_prob_dtt_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="apply_prob_dtt_+3A_data">data</code></td>
<td>

<p>a data.frame containing a database of monophyletic groups for which potential shifts can be investigated. This database should be based on taxonomy, ecology or traits and contain a column named &quot;Species&quot; with species names as in phylo.
</p>
</td></tr>
<tr><td><code id="apply_prob_dtt_+3A_sampling.fractions">sampling.fractions</code></td>
<td>

<p>the output resulting from get.sampling.fractions.
</p>
</td></tr>
<tr><td><code id="apply_prob_dtt_+3A_shift.res">shift.res</code></td>
<td>

<p>the output resulting from shift.estimates.
</p>
</td></tr>
<tr><td><code id="apply_prob_dtt_+3A_backbone.option">backbone.option</code></td>
<td>

<p>type of the backbone analysis:
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: for every shift, the probability of the speciation event at the stem age of the subclade is included in the likelihood of the backbone thanks to the argument spec_times.
</p>
</li>
<li><p>&quot;crown.shift&quot;: for every shift, both the probability of the speciation event at the stem age of the subclade and the probability that the stem of the subclade survives to the crown age are included in the likelihood of the backbone thanks to the argument branch_times.
</p>
</li></ul>

</td></tr>
<tr><td><code id="apply_prob_dtt_+3A_combi">combi</code></td>
<td>

<p>numeric. The combination of shifts defined by its rank in the global comparison.
</p>
</td></tr>
<tr><td><code id="apply_prob_dtt_+3A_m">m</code></td>
<td>

<p>NULL or numeric. The set of maximum values for m ranges. Should be as long as the number of parts in the combinaison. Default is NULL (see details).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This funcion calls the function prob_dtt() to calculate paleodiversity dynamics with the probabilistic approach for the different parts of a combination of diversification shifts.
</p>
<p>As explained in Billaud et al. (2020), all the sum of probabilities per million year must be equal to 1. However, it can be difficult to reach 1 for groups showing a paleodiversity decline because the range of paleodiversity over which we need to calculate the probabilities can be very large. To circumvent this issue, apply_prob_dtt() set the range of the paleodiversity to the maximum of the deterministic estimate from the function paleodiv() and successively multiplies this maximum by 2, 3, 5, 7 and 10 until the sums of probabilities for each million year reach a minimum of 95%. In few cases, this value of 95% is not reached for few million years. In this case, it might come from an extremely high range of m and maximum values can be manually set up with the argument m.
</p>


<h3>Value</h3>

<p>A list of results from prob_dtt() for subclades and backbone(s).
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record. Proc. Nat. Acad. Sci. 108: 16327-16332.
</p>
<p>Billaud, O., Moen, D.S., Parsons, T.L., Morlon, H., (2020). Estimating Diversity Through Time Using Molecular Phylogenies: Old and Species-Poor Frog Families are the Remnants of a Diverse Past. Systematic Biology 69, 363–383.
</p>
<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_bd">fit_bd</a>, <a href="#topic+plot_prob_dtt">plot_prob_dtt</a>, <a href="#topic+prob_dtt">prob_dtt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# loading data
data("Cetacea")
data("taxo_cetacea")
data("shifts_cetacea")

taxo_cetacea_no_genus &lt;- taxo_cetacea[names(taxo_cetacea) != "Genus"]

# apply_prob_dtt() needs the sampling fractions
f_df_cetacea &lt;- get.sampling.fractions(phylo = Cetacea,
                                       data = taxo_cetacea_no_genus,
                                       plot = TRUE, cex = 0.3, lad = FALSE)

# use of apply_prob_dtt()
prob_dtt_cetacea &lt;- apply_prob_dtt(phylo = Cetacea,
                                   data = taxo_cetacea_no_genus,
                                   shift.res = shifts_cetacea,
                                   sampling.fractions = f_df_cetacea,
                                   combi = 1)

</code></pre>

<hr>
<h2 id='Balaenopteridae'>Balaenopteridae phylogeny</h2><span id='topic+Balaenopteridae'></span>

<h3>Description</h3>

<p>Ultrametric phylogenetic tree of the 9 extant Balaenopteridae species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Balaenopteridae)</code></pre>


<h3>Details</h3>

<p>This phylogeny was extracted from Steeman et al. Syst Bio 2009 cetacean phylogeny</p>


<h3>References</h3>

<p>Steeman, M.E., et al. (2009) Radiation of extant cetaceans driven by restructuring of the oceans <em>Syst Biol</em> 58:573-585
</p>
<p>Morlon, H., Parsons, T.L., Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Balaenopteridae)
print(Balaenopteridae)
plot(Balaenopteridae)
</code></pre>

<hr>
<h2 id='BGB.examples'>
BioGeoBEARS stochastic maps
</h2><span id='topic+BGB.examples'></span>

<h3>Description</h3>

<p>Phylogenies and example stochastic maps for Canidae (from an unstratified BioGeoBEARS analysis) and Ochotonidae (from a stratified BioGeoBEARS analysis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(BGB.examples)
</code></pre>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Matzke, N. 2014. Model selection in historical biogeography reveals that founder-event speciation is a crucial process in island clades. <em>Systematic Biology</em> 63:951-970.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CreateGeoObject_BioGeoBEARS">CreateGeoObject_BioGeoBEARS</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BGB.examples)
oldpar &lt;- par(no.readonly = TRUE)
par(mfrow=c(1,2))
plot(BGB.examples$Canidae.phylo)
plot(BGB.examples$Ochotonidae.phylo)
par(oldpar) # restore the old par
</code></pre>

<hr>
<h2 id='BICompare'>
Identify modalities in a phylogeny
</h2><span id='topic+BICompare'></span>

<h3>Description</h3>

<p>Computes the BIC values for a specified number of modalities in the distance matrix of a phylogenetic tree and that of randomly bifurcating trees; identifies these modalities using k-means clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BICompare(phylo,t,meth=c("ultrametric"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BICompare_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)</p>
</td></tr>
<tr><td><code id="BICompare_+3A_t">t</code></td>
<td>

<p>the number of modalities to be tested</p>
</td></tr>
<tr><td><code id="BICompare_+3A_meth">meth</code></td>
<td>

<p>whether the randomly bifurcating &quot;control&quot; tree should be ultrametric or non-ultrametric  	
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>BIC_test</code></td>
<td>

<p>BIC values for finding t modalities in the distance matrix of a tree and the lowest five percent of 1000 random (&quot;control&quot;) trees</p>
</td></tr>
<tr><td><code>clusters</code></td>
<td>

<p>a vector specifying which nodes in the tree belong to each of t modalities</p>
</td></tr>
<tr><td><code>BSS/TSS</code></td>
<td>

<p>the ratio of between-cluster sum of squares over total sum of squares</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>E Lewitus</p>


<h3>References</h3>

<p>Lewitus, E., Morlon, H., Characterizing and comparing phylogenies from their Laplacian spectrum, bioRxiv doi: http://dx.doi.org/10.1101/026476</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_BICompare">plot_BICompare</a></code>, <code><a href="#topic+spectR">spectR</a></code>, <code><a href="#topic+JSDtree">JSDtree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)

BICompare(Cetacea,5)

</code></pre>

<hr>
<h2 id='build_network.BipartiteEvol'>
Build the interaction network in BipartiteEvol
</h2><span id='topic+build_network.BipartiteEvol'></span>

<h3>Description</h3>

<p>Build the phylogenies from the output of BipartiteEvol and the corresponding genealogies and phylogenies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_network.BipartiteEvol( gen, spec)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="build_network.BipartiteEvol_+3A_gen">gen</code></td>
<td>

<p>The output of a run of make_gen.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="build_network.BipartiteEvol_+3A_spec">spec</code></td>
<td>

<p>The output of a run of define_species.BipartiteEvol
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix M where M[i,j] is the number of individuals from species i (from guild P) interacting with an individual from species j (from guild H)
</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet, O., Loeuille, N. and Morlon, H. (2020), An individual-based model for the eco-evolutionary emergence of bipartite interaction networks. Ecol Lett. doi:10.1111/ele.13592</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim.BipartiteEvol">sim.BipartiteEvol</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># run the model
set.seed(1)


if(test){

mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 800,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5)

#build the genealogies
gen = make_gen.BipartiteEvol(mod)
plot(gen$H)

#compute the phylogenies
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#plot the result
plot_div.BipartiteEvol(gen,phy1, 1)

#build the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = nx, spatial = FALSE)


## add time steps to a former run
seed=as.integer(10)
set.seed(seed)

mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 200,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5,
                        P=mod$P,H=mod$H)  # former run output

# update the genealogy
gen = make_gen.BipartiteEvol(mod,
                             treeP=gen$P, treeH=gen$H)

# update the phylogenies...
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#... and the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = 10, spatial = FALSE)

}

</code></pre>

<hr>
<h2 id='Calomys'>Calomys phylogeny</h2><span id='topic+Calomys'></span>

<h3>Description</h3>

<p>Ultrametric phylogenetic tree of 11 of the 13 extant Calomys species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Calomys)</code></pre>


<h3>Details</h3>

<p>This phylogeny is from Pigot et al. PloS Biol 2012</p>


<h3>References</h3>

<p>Pigot et al.(2012) Speciation and extinction drive the appearance of directional range size evolution in phylogenies and the fossil record <em>PloS Biol</em> 10:1-9
Manceau, M., Lambert, A., Morlon, H. (2015) Phylogenies support out-of-equilibrium models of biodiversity Ecology Letters 18: 347-356
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Calomys)
print(Calomys)
plot(Calomys)
</code></pre>

<hr>
<h2 id='Caprimulgidae'>
The <em>Caprimulgidae</em> phylogeny.
</h2><span id='topic+Caprimulgidae'></span>

<h3>Description</h3>

<p>The MCC phylogeny for the <em>Caprimulgidae</em>, from Jetz et al. (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Caprimulgidae")</code></pre>


<h3>Source</h3>

<p>Jetz, W., G. Thomas, J. Joy, K. Hartmann, and A. Mooers. 2012. The global diversity of birds in
space and time. <em>Nature 491:444.</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Caprimulgidae")

plot(Caprimulgidae)

</code></pre>

<hr>
<h2 id='Caprimulgidae_ClaDS2'>
An example run of ClaDS2.
</h2><span id='topic+Caprimulgidae_ClaDS2'></span>

<h3>Description</h3>

<p>An example of the run on the inference of ClaDS2 on the <em>Caprimulgidae</em> phylogeny, thinned every 10 iterations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Caprimulgidae_ClaDS2")</code></pre>


<h3>Format</h3>

<p>A list object with fields :
</p>

<dl>
<dt><code>tree</code></dt><dd><p>The <em>Caprimulgidae</em> phylogeny on which we ran the model.</p>
</dd>
<dt><code>sample_fraction</code></dt><dd><p>The sample fraction for the clade.</p>
</dd>
<dt><code>sampler</code></dt><dd><p>The chains obtained by running ClaDS2 on the <em>Caprimulgidae</em> phylogeny.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The <em>Caprimulgidae</em> phylogeny was obtained from Jetz et al. (2012)
</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>Source</h3>

<p>Jetz, W., G. Thomas, J. Joy, K. Hartmann, and A. Mooers. 2012. The global diversity of birds in
space and time. <em>Nature 491:444.</em>
</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_ClaDS">fit_ClaDS</a></code>, <code><a href="#topic+plot_ClaDS_chains">plot_ClaDS_chains</a></code>, <code><a href="#topic+getMAPS_ClaDS0">getMAPS_ClaDS0</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Caprimulgidae_ClaDS2")

# plot the mcmc chains
plot_ClaDS_chains(Caprimulgidae_ClaDS2$sampler)


# extract the Maxima A Posteriori for each parameter
maps = getMAPS_ClaDS(Caprimulgidae_ClaDS2$sampler, thin = 1)
print(paste0("sigma = ", maps[1], " ; alpha = ", 
  maps[2], " ; epsilon = ", maps[3], " ; l_0 = ", maps[4] ))
  
# plot the infered branch specific speciation rates
plot_ClaDS_phylo(Caprimulgidae_ClaDS2$tree, maps[-(1:4)])

</code></pre>

<hr>
<h2 id='Cetacea'>Cetacean phylogeny</h2><span id='topic+Cetacea'></span>

<h3>Description</h3>

<p>Ultrametric phylogenetic tree for 87 of the 89 extant cetacean species</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Cetacea)</code></pre>


<h3>Details</h3>

<p>This phylogeny was constructed by Bayesian phylogenetic inference from six mitochondrial and nine nuclear genes.
It was calibrated using seven paleontological age constraints and a relaxed molecular clock approach. See Steeman et al. (2009) for details.
</p>


<h3>Source</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans, <em>Syst Biol</em> 58:573-585
</p>


<h3>References</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans <em>Syst Biol</em> 58:573-585
</p>
<p>Morlon, H., Parsons, T.L., Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Condamine, F.L., Rolland, J., Morlon, H. (2013) Macroevolutionary perspectives to environmental change <em>Eco Lett</em> 16: 72-85
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)
print(Cetacea)
plot(Cetacea)
</code></pre>

<hr>
<h2 id='Cetacea_clades'>Stochastic map of clade membership in Cetacean phylogeny</h2><span id='topic+Cetacea_clades'></span>

<h3>Description</h3>

<p>simmap object of clade membership in Cetacean phylogeny</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Cetacea_clades)</code></pre>


<h3>Details</h3>

<p>See <code><a href="#topic+Cetacea">Cetacea</a></code>
</p>


<h3>Source</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans, <em>Syst Biol</em> 58:573-585
</p>


<h3>References</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans <em>Syst Biol</em> 58:573-585
</p>
<p>Morlon, H., Parsons, T.L., Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Condamine, F.L., Rolland, J., Morlon, H. (2013) Macroevolutionary perspectives to environmental change <em>Eco Lett</em> 16: 72-85
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea_clades)
print(Cetacea_clades)
plot(Cetacea_clades)
</code></pre>

<hr>
<h2 id='ClaDS0_example'>
An example run of ClaDS0.
</h2><span id='topic+ClaDS0_example'></span>

<h3>Description</h3>

<p>An example of the run on the inference of ClaDS0 on a simulated phylogeny, thinned every 10 iterations.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("ClaDS0_example")</code></pre>


<h3>Format</h3>

<p>A list object with fields :
</p>

<dl>
<dt><code>tree</code></dt><dd><p>The simulated phylogeny on which we ran the model.</p>
</dd>
<dt><code>speciation_rates</code></dt><dd><p>The simulated speciation rates.</p>
</dd>
<dt><code>Cl0_chains</code></dt><dd><p>The output of the <code>run_ClaDS0</code> run.</p>
</dd>
</dl>



<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_ClaDS0">fit_ClaDS0</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ClaDS0_example)

# plot the resulting chains for the first 4 parameters
plot_ClaDS0_chains(ClaDS0_example$Cl0_chains, param = 1:4)

# extract the Maximum A Posteriori for each of the parameters
MAPS = getMAPS_ClaDS0(ClaDS0_example$tree, 
                      ClaDS0_example$Cl0_chains, 
                      thin = 10)

# plot the simulated (on the left) and inferred speciation rates (on the right)
# on the same color scale
plot_ClaDS_phylo(ClaDS0_example$tree, 
          ClaDS0_example$speciation_rates, 
          MAPS[-(1:3)])
</code></pre>

<hr>
<h2 id='co2'>co2 data since the Jurassic</h2><span id='topic+co2'></span>

<h3>Description</h3>

<p>Atmospheric co2 data since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(co2)</code></pre>


<h3>Details</h3>

<p>Atmospheric co2 data since the Jurassic taken from Mayhew et al., (2008, 2012) and derived from the GeoCarb-III model (Berner and Kothavala, 2001). The data are eported as the ratio of the mass of co2 at time t to that at present. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>co2</code></dt><dd><p>a numeric vector corresponding to the estimated co2 at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Mayhew, P.J., Jenkins, G.B., Benton, T.G. (2008) A long-term association between global temperature and biodiversity, origination and extinction in the fossil record <em>Proceedings of the Royal Society B 275:47-53</em>
</p>
<p>Mayhew, P.J., Bell, M.A., Benton, T.G, McGowan, A.J. (2012) Biodiversity tracks temperature over time <em>109:15141-15145</em>
</p>
<p>Berner R.A., Kothavala, Z. (2001) GEOCARB III: A revised model of atmospheric CO2 over
Phanerozoic time <em>Am J Sci 301:182–204</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(co2)
plot(co2)
</code></pre>

<hr>
<h2 id='co2_res'>co2 data since the beginning of the Cenozoic</h2><span id='topic+co2_res'></span>

<h3>Description</h3>

<p>Atmospheric co2 data since the beginning of the Cenozoic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(co2_res)</code></pre>


<h3>Details</h3>

<p>Implied co2 data since the beginning of the Cenozoic taken from Hansen et al., (2013). The data are the amount of co2 in ppm reuquired to yield observed global temperature throughout the Cenozoic:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>co2</code></dt><dd><p>a numeric vector corresponding to the estimated co2 at that age</p>
</dd>
</dl>



<h3>Source</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans, <em>Syst Biol</em> 58:573-585
</p>


<h3>References</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans <em>Syst Biol</em> 58:573-585
</p>
<p>Morlon, H., Parsons, T.L., Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Condamine, F.L., Rolland, J., Morlon, H. (2013) Macroevolutionary perspectives to environmental change <em>Eco Lett</em> 16: 72-85
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)
print(Cetacea)
plot(Cetacea)
</code></pre>

<hr>
<h2 id='coccolithophore'>Coccolithophore diversity since the Jurassic</h2><span id='topic+coccolithophore'></span>

<h3>Description</h3>

<p>Coccolithophore fossil diversity since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(coccolithophore)</code></pre>


<h3>Details</h3>

<p>Coccolithophore fossil diversity since the Jurassic compiled from the Neptune Database (Lazarus, 1994) and Paleobiology Database (https://paleobiodb.org/). Diversity curves are estimated at the genus level using shareholder quorum subsampling (Alroy, 2010) at two-million-year bins. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>coccolithophore</code></dt><dd><p>a numeric vector corresponding to the estimated coccolithophore change at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Lazarus, D. (1994) Neptune: A marine micropaleontology database <em>Mathematical Geology 26:817–832</em>
</p>
<p>Alroy, J. (2010) Geographical, environmental and intrinsic biotic controls on Phanerozoic marine diversification: Controls on phanerozoic marine diversification <em>Palaeontology 53:1211–1235</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(coccolithophore)
plot(coccolithophore)
</code></pre>

<hr>
<h2 id='CreateClassObject'>
Create class object
</h2><span id='topic+CreateClassObject'></span>

<h3>Description</h3>

<p>This function returns names of internode intervals, named descendants of each node,
and a class object formatted in a way that can be
passed to <code><a href="#topic+CreateGeobyClassObject">CreateGeobyClassObject</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
CreateClassObject(map,rnd=5,return.mat=FALSE)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CreateClassObject_+3A_map">map</code></td>
<td>

<p>stochastic map from <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code> in <code>phytools</code>
</p>
</td></tr>
<tr><td><code id="CreateClassObject_+3A_rnd">rnd</code></td>
<td>
 
<p>integer indicating the number of decimal places to which times should be rounded (default value is 5) (see <code><a href="base.html#topic+round">round</a></code>)
</p>
</td></tr>
<tr><td><code id="CreateClassObject_+3A_return.mat">return.mat</code></td>
<td>
 
<p>logical indicating whether to return simmap in a format to be passed to other internal functions (usually FALSE)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function formats the class object so that it can be correctly
passed to the numerical integration performed in <code><a href="#topic+fit_t_comp_subgroup">fit_t_comp_subgroup</a></code>. 
</p>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>class.object</code></td>
<td>
<p>a list of matrices specifying the state of each branch during each internode interval (see Details)</p>
</td></tr>
<tr><td><code>times</code></td>
<td>
<p>a vector containing the time since the root of the tree at which nodes or changes in biogeography occur (used internally in other functions)</p>
</td></tr>
<tr><td><code>spans</code></td>
<td>
<p>a vector specifying the distances between times (used internally in other functions)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a></p>


<h3>References</h3>

<p>Drury, J., Tobias, J., Burns, K., Mason, N., Shultz, A., and Morlon, H. in review. Contrasting impacts of competition on ecological and social trait evolution in songbirds. <em>PLOS Biology</em>.
</p>
<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp_subgroup">fit_t_comp_subgroup</a></code>,<code><a href="#topic+CreateGeobyClassObject">CreateGeobyClassObject</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Anolis.data)

#Create a make.simmap object
require(phytools)
geo&lt;-c(rep("cuba",7),rep("hispaniola",9),"puerto_rico")
names(geo)&lt;-Anolis.data$phylo$tip.label
stochastic.map&lt;-phytools::make.simmap(Anolis.data$phylo, 
									geo, model="ER", nsim=1)
CreateClassObject(stochastic.map)

</code></pre>

<hr>
<h2 id='CreateGeobyClassObject'>
Create merged biogeography-by-class object
</h2><span id='topic+CreateGeobyClassObject'></span>

<h3>Description</h3>

<p>Create a merged biogeography-by-class object to be passed to fit_t_comp_subgroup using a stochastic map created from any model in BioGeoBEARS (see documentation in BioGeoBEARS package) and a simmap object from phytools (see documentation in phytools package).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
CreateGeobyClassObject(phylo,simmap,trim.class,ana.events,clado.events,
	stratified=FALSE,rnd=5)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CreateGeobyClassObject_+3A_phylo">phylo</code></td>
<td>

<p>the object of type 'phylo' (see ape documentation) used to build ancestral range stochastic maps in BioGeoBEARS
</p>
</td></tr>
<tr><td><code id="CreateGeobyClassObject_+3A_simmap">simmap</code></td>
<td>

<p>a phylo object created using <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code> in phytools
</p>
</td></tr>
<tr><td><code id="CreateGeobyClassObject_+3A_trim.class">trim.class</code></td>
<td>

<p>category in the simmap object that represents the subgroup of interest (see Details and Examples)
</p>
</td></tr>
<tr><td><code id="CreateGeobyClassObject_+3A_ana.events">ana.events</code></td>
<td>

<p>the &quot;ana.events&quot; table produced in BioGeoBEARS that lists anagenetic events in the stochastic map
</p>
</td></tr>
<tr><td><code id="CreateGeobyClassObject_+3A_clado.events">clado.events</code></td>
<td>

<p>the &quot;clado.events&quot; table produced in BioGeoBEARS that lists cladogenetic events in the stochastic map
</p>
</td></tr>
<tr><td><code id="CreateGeobyClassObject_+3A_stratified">stratified</code></td>
<td>

<p>logical indicating whether the ancestral biogeography stochastic map was built from a stratified analysis in BioGeoBEARS
</p>
</td></tr>
<tr><td><code id="CreateGeobyClassObject_+3A_rnd">rnd</code></td>
<td>

<p>an integer value indicating the number of decimals to which values should be rounded in order to reconcile class and geo.objects (default is 5)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function merges a class object (which reconstructs group membership through time) and a stochastic map of ancestral biogeography (to reconstruct sympatry through time),
such that lineages can only interact when they belong to the same subgroup AND are sympatric.
</p>
<p>This allows fitting models of competition where only sympatric members of a subgroup can compete (e.g., all lineages that share similar diets or habitats).
</p>
<p>This function should be used to format the geography object so that it can be correctly
passed to the numerical integration performed in <code><a href="#topic+fit_t_comp_subgroup">fit_t_comp_subgroup</a></code>. 
</p>


<h3>Value</h3>

<p>Returns a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>map</code></td>
<td>
<p>a <code>simmap</code> object with phylogeny trimmed to subgroup of interest (including all branches determined to belong to that subgroup)</p>
</td></tr>
<tr><td><code>geography.object</code></td>
<td>
<p>a list with the following components:</p>
</td></tr>
<tr><td><code>geography.matrix</code></td>
<td>
<p>a list of matrices specifying both sympatry &amp; group membership (==1) or allopatry and/or non-membership in the focal subgroup (==0) for each species pair for each internode interval (see Details)</p>
</td></tr>
<tr><td><code>times</code></td>
<td>
<p>a vector containing the time since the root of the tree at which nodes or changes in biogeographyXsubgroup membership occur (used internally in other functions)</p>
</td></tr>
<tr><td><code>spans</code></td>
<td>
<p>a vector specifying the distances between times (used internally in other functions)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a></p>


<h3>References</h3>

<p>Drury, J., Tobias, J., Burns, K., Mason, N., Shultz, A., and Morlon, H. in review. Contrasting impacts of competition on ecological and social trait evolution in songbirds. <em>PLOS Biology</em>.
</p>
<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp_subgroup">fit_t_comp_subgroup</a></code>, <code><a href="#topic+CreateGeoObject_BioGeoBEARS">CreateGeoObject_BioGeoBEARS</a></code> , <code><a href="#topic+CreateClassObject">CreateClassObject</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(BGB.examples)



Canidae.phylo&lt;-BGB.examples$Canidae.phylo
dummy.group&lt;-c(rep("B",3),rep("A",12),rep("B",2),rep("A",6),rep("B",5),rep("A",6))
names(dummy.group)&lt;-Canidae.phylo$tip.label

Canidae.simmap&lt;-phytools::make.simmap(Canidae.phylo,dummy.group)

#build GeobyClass object with "A" as the focal group

Canidae.geobyclass.object&lt;-CreateGeobyClassObject(phylo=Canidae.phylo,simmap=Canidae.simmap, 
trim.class="A",ana.events=BGB.examples$Canidae.ana.events, 
clado.events=BGB.examples$Canidae.clado.events,stratified=FALSE, rnd=5)
	
phytools::plotSimmap(Canidae.geobyclass.object$map)


</code></pre>

<hr>
<h2 id='CreateGeoObject'>
Create biogeography object
</h2><span id='topic+CreateGeoObject'></span>

<h3>Description</h3>

<p>This function returns names of internode intervals, named descendants of each node,
and a geography object formatted in a way that can be
passed to <code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
CreateGeoObject(phylo,map)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CreateGeoObject_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="CreateGeoObject_+3A_map">map</code></td>
<td>

<p>either a matrix modified from <code>phylo$edge</code> or a phylo object created using <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code> (see Details and Examples)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function should be used to format the geography object so that it can be correctly
passed to the numerical integration performed in <code><a href="#topic+fit_t_comp">fit_t_comp</a></code>. 
</p>
<p>The <code>map</code> can either be a matrix formed by specifying the region in which each branch specified by <code>phylo$edge</code> existed, or a stochastic map stored as a phylo object output from <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code> (see Examples).
</p>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>geography.object</code></td>
<td>
<p>a list of matrices specifying sympatry (1) or allopatry (0) for each species pair for each internode interval (see Details)</p>
</td></tr>
<tr><td><code>times</code></td>
<td>
<p>a vector containing the time since the root of the tree at which nodes or changes in biogeography occur (used internally in other functions)</p>
</td></tr>
<tr><td><code>spans</code></td>
<td>
<p>a vector specifying the distances between times (used internally in other functions)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a></p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Anolis.data)
#Create a geography.object with a modified edge matrix
#First, specify which region each branch belonged to:
Anolis.regions&lt;-c(rep("cuba",14),rep("hispaniola",17),"puerto_rico")
Anolis.map&lt;-cbind(Anolis.data$phylo$edge,Anolis.regions)
CreateGeoObject(Anolis.data$phylo,map=Anolis.map)

#Create a geography.object with a make.simmap object
#First, specify which region each branch belonged to:
require(phytools)
geo&lt;-c(rep("cuba",7),rep("hispaniola",9),"puerto_rico")
names(geo)&lt;-Anolis.data$phylo$tip.label
stochastic.map&lt;-phytools::make.simmap(Anolis.data$phylo, 
							geo, model="ER", nsim=1)
CreateGeoObject(Anolis.data$phylo,map=stochastic.map)

</code></pre>

<hr>
<h2 id='CreateGeoObject_BioGeoBEARS'>
Create biogeography object using a stochastic map from BioGeoBEARS
</h2><span id='topic+CreateGeoObject_BioGeoBEARS'></span>

<h3>Description</h3>

<p>Create biogeography object using a stochastic map created from any model in BioGeoBEARS (see documentation in BioGeoBEARS package).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateGeoObject_BioGeoBEARS( full.phylo, trimmed.phylo = NULL, ana.events,
clado.events, stratified=FALSE, simmap.out=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CreateGeoObject_BioGeoBEARS_+3A_full.phylo">full.phylo</code></td>
<td>

<p>the object of type 'phylo' (see ape documentation) that was used to construct the stochastic map in BioGeoBEARS
</p>
</td></tr>
<tr><td><code id="CreateGeoObject_BioGeoBEARS_+3A_trimmed.phylo">trimmed.phylo</code></td>
<td>

<p>if the desired biogeography object excludes some species that were initially included in the stochastic map, this specifies a phylo object for the trimmed set of species
</p>
</td></tr>
<tr><td><code id="CreateGeoObject_BioGeoBEARS_+3A_ana.events">ana.events</code></td>
<td>

<p>the &quot;ana.events&quot; table produced in BioGeoBEARS that lists anagenetic events in the stochastic map
</p>
</td></tr>
<tr><td><code id="CreateGeoObject_BioGeoBEARS_+3A_clado.events">clado.events</code></td>
<td>

<p>the &quot;clado.events&quot; table produced in BioGeoBEARS that lists cladogenetic events in the stochastic map
</p>
</td></tr>
<tr><td><code id="CreateGeoObject_BioGeoBEARS_+3A_stratified">stratified</code></td>
<td>

<p>logical indicating whether the stochastic map was built from a stratified analysis in BioGeoBEARS
</p>
</td></tr>
<tr><td><code id="CreateGeoObject_BioGeoBEARS_+3A_simmap.out">simmap.out</code></td>
<td>

<p>logical indicating whether output should be a stochastic map (simmap) object (see note)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: generating a stochastic map output using <code>simmap.out=TRUE</code> and passing to <a href="#topic+fit_t_comp">fit_t_comp</a> for diversity dependent models with biogeography greatly speeds up model 
fitting compared to output generated when <code>simmap.out=FALSE</code>. This cannot be used for matching competition or any two-regime models with biogeography.
</p>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>geography.object</code></td>
<td>
<p>a list of matrices specifying sympatry (1) or allopatry (0) for each species pair for each internode interval (see Details)</p>
</td></tr>
<tr><td><code>times</code></td>
<td>
<p>a vector containing the time since the root of the tree at which nodes or changes in biogeography occur (used internally in other functions)</p>
</td></tr>
<tr><td><code>spans</code></td>
<td>
<p>a vector specifying the distances between times (used internally in other functions)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Matzke, N. 2014. Model selection in historical biogeography reveals that founder-event speciation is a crucial process in island clades. <em>Systematic Biology</em> 63:951-970.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
<code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(BGB.examples)




##Example with a non-stratified tree

Canidae.geography.object&lt;-CreateGeoObject_BioGeoBEARS(full.phylo=BGB.examples$Canidae.phylo,
ana.events=BGB.examples$Canidae.ana.events, clado.events=BGB.examples$Canidae.clado.events)

#on a subclade
Canidae.trimmed&lt;-drop.tip(BGB.examples$Canidae.phylo 
							,BGB.examples$Canidae.phylo$tip.label[1:9])
							
Canidae.trimmed.geography.object&lt;-CreateGeoObject_BioGeoBEARS(
full.phylo=BGB.examples$Canidae.phylo, trimmed.phylo=Canidae.trimmed, 
ana.events=BGB.examples$Canidae.ana.events, clado.events=BGB.examples$Canidae.clado.events)

##Example with a stratified tree

Ochotonidae.geography.object&lt;-CreateGeoObject_BioGeoBEARS( 
full.phylo = BGB.examples$Ochotonidae.phylo, ana.events = BGB.examples$Ochotonidae.ana.events,
clado.events = BGB.examples$Ochotonidae.clado.events, stratified = TRUE)

#on a subclade
Ochotonidae.trimmed&lt;-drop.tip(BGB.examples$Ochotonidae.phylo, 
BGB.examples$Ochotonidae.phylo$tip.label[1:9])
								
Ochotonidae.trimmed.geography.object&lt;-CreateGeoObject_BioGeoBEARS(
full.phylo=BGB.examples$Ochotonidae.phylo, trimmed.phylo=Ochotonidae.trimmed, 
ana.events=BGB.examples$Ochotonidae.ana.events, 
clado.events=BGB.examples$Ochotonidae.clado.events, stratified=TRUE)



</code></pre>

<hr>
<h2 id='createModel'>
Creation of a PhenotypicModel
</h2><span id='topic+createModel'></span>

<h3>Description</h3>

<p>Creates an object of class PhenotypicModel, intended to represent a model of trait evolution on a specific tree. DIstinct keywords correspond to different models, using one phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createModel(tree, keyword)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createModel_+3A_tree">tree</code></td>
<td>

<p>an object of class 'phylo' as defined in the R package 'ape'.
</p>
</td></tr>
<tr><td><code id="createModel_+3A_keyword">keyword</code></td>
<td>

<p>a string specifying the model. Available models include &quot;BM&quot;, &quot;BM_from0&quot;, &quot;BM_from0_driftless&quot;, &quot;OU&quot;, &quot;OU_from0&quot;, &quot;ACDC&quot;, &quot;DD&quot;, &quot;PM&quot;, &quot;PM_OUless&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the object of class &quot;PhenotypicModel&quot;.
</p>


<h3>Author(s)</h3>

<p>M Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Loading an example tree
newick &lt;- "((((A:1,B:0.5):2,(C:3,D:2.5):1):6,E:10.25):2,(F:6.5,G:8.25):3):1;"
tree &lt;- read.tree(text=newick)

#Creating the models
modelBM &lt;- createModel(tree, 'BM')
modelOU &lt;- createModel(tree, 'OU')

#Printing basic or full informations on the model definitions
show(modelBM)
print(modelOU)
</code></pre>

<hr>
<h2 id='createModelCoevolution'>
Creation of a PhenotypicGMM
</h2><span id='topic+createModelCoevolution'></span>

<h3>Description</h3>

<p>Creates an object of class PhenotypicGMM, a subclass of the class PhenotypicModel intended to represent the Generalist Matching Mutualism model of trait evolution on two specific trees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createModelCoevolution(tree1, tree2, keyword)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createModelCoevolution_+3A_tree1">tree1</code></td>
<td>

<p>an object of class 'phylo' as defined in the R package 'ape'.
</p>
</td></tr>
<tr><td><code id="createModelCoevolution_+3A_tree2">tree2</code></td>
<td>

<p>an object of class 'phylo' as defined in the R package 'ape'.
</p>
</td></tr>
<tr><td><code id="createModelCoevolution_+3A_keyword">keyword</code></td>
<td>

<p>a string object. Defaut value &quot;GMM&quot; returns an object of class PhenotypicGMM, which takes advantage of faster distribution computation. Otherwise, a &quot;PhenotypicModel&quot; is returned, and the computation of the tip distribution will take much longer.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class &quot;PhenotypicModel&quot; or &quot;PhenotypicGMM&quot;.
</p>


<h3>Author(s)</h3>

<p>M Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Loading example trees
newick1 &lt;- "(((A:1,B:1):3,(C:3,D:3):1):2,E:6);"
tree1 &lt;- read.tree(text=newick1)
newick2 &lt;- "((X:1.5,Y:1.5):3,Z:4.5);"
tree2 &lt;- read.tree(text=newick2)

#Creating the model
modelGMM &lt;- createModelCoevolution(tree1, tree2)

#Printing basic or full informations on the model definitions
show(modelGMM)
print(modelGMM)

#Simulates tip trait data
dataGMM &lt;- simulateTipData(modelGMM, c(0,0,5,-5, 1, 1), method=2)
</code></pre>

<hr>
<h2 id='d13c'>d13c data since the Jurassic</h2><span id='topic+d13c'></span>

<h3>Description</h3>

<p>Benthic d13c weathering ratio since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(d13c)</code></pre>


<h3>Details</h3>

<p>Ratio of stable carbon isotopes since the Jurassic calculated by Hannisdal and Peters (2011) and Lazarus et al. (2014) from marine carbonates. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>d13c</code></dt><dd><p>a numeric vector corresponding to the estimated d13c at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Hannisdal, B., Peters, S.E. (2011) hanerozoic Earth system evolution and marine biodiversity <em>Science 334:1121-1124</em>
</p>
<p>Lazarus, D., Barron, J., Renaudie, J., Diver, P., Turke, A. (2014) Cenozoic Planktonic Marine Diatom Diversity and Correlation to Climate Change <em>PLoS ONE 9:e84857</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(d13c)
plot(d13c)
</code></pre>

<hr>
<h2 id='define_species.BipartiteEvol'>
Build the phylogenies for BipartiteEvol</h2><span id='topic+define_species.BipartiteEvol'></span>

<h3>Description</h3>

<p>Build the phylogenies from the output of BipartiteEvol and the corresponding genealogies</p>


<h3>Usage</h3>

<pre><code class='language-R'>define_species.BipartiteEvol(genealogy, threshold = 1, 
      distanceH = NULL, distanceP = NULL, verbose = TRUE,
      monophyly = TRUE, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="define_species.BipartiteEvol_+3A_genealogy">genealogy</code></td>
<td>

<p>The output of a run of make_gen.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="define_species.BipartiteEvol_+3A_threshold">threshold</code></td>
<td>

<p>The species definition ratchet (s)
</p>
</td></tr>
<tr><td><code id="define_species.BipartiteEvol_+3A_distanceh">distanceH</code></td>
<td>

<p>Distance (ie nb of mutations) matrix between the individual of clade H
</p>
</td></tr>
<tr><td><code id="define_species.BipartiteEvol_+3A_distancep">distanceP</code></td>
<td>

<p>Distance (ie nb of mutations) matrix between the individual of clade P</p>
</td></tr>
<tr><td><code id="define_species.BipartiteEvol_+3A_verbose">verbose</code></td>
<td>

<p>Should the progression of the computation be printed?
</p>
</td></tr>
<tr><td><code id="define_species.BipartiteEvol_+3A_monophyly">monophyly</code></td>
<td>
 
<p>Should the species delineations be strictly monophyletic species (TRUE - default) or not (FALSE)? If not, the threshold must be equal to 1.
</p>
</td></tr>
<tr><td><code id="define_species.BipartiteEvol_+3A_seed">seed</code></td>
<td>

<p>If monophyly==FALSE, the seed is used to pick one representative individual per (potentially non-monophyletic) species. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If monophyly==TRUE, species delineation is performed using the model of Speciation by Genetic Differentiation (Manceau et al., 2015) where the 'threshold' (the number of mutations needed to belong to different species) can vary. It results in monophyletic species.
If monophyly==FALSE, we consider that each new mutation (i.e. each new combination of traits) gives rise to a new species (Perez-Lamarque et al., 2021). As a result, species are not necessarily formed by a monophyletic group of individuals. 
</p>


<h3>Value</h3>

<p>a list with
</p>
<table role = "presentation">
<tr><td><code>P</code></td>
<td>
<p>The species identity of each individual in guild P</p>
</td></tr>
<tr><td><code>H</code></td>
<td>
<p>The species identity of each individual in guild H</p>
</td></tr>
<tr><td><code>Pphylo</code></td>
<td>
<p>The phylogeny for guild P</p>
</td></tr>
<tr><td><code>Hphylo</code></td>
<td>
<p>The phylogeny for guild H</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>O. Maliet &amp; B. Perez-Lamarque
</p>


<h3>References</h3>

<p>Manceau, M., A. Lambert, and H. Morlon. (2015). Phylogenies support out-of-equilibrium models of biodiversity. Ecology letters 18:347–356.
</p>
<p>Maliet, O., Loeuille, N. and Morlon, H. (2020). An individual-based model for the eco-evolutionary emergence of bipartite interaction networks. Ecol Lett. doi:10.1111/ele.13592
</p>
<p>Perez‐Lamarque, B., Maliet, O., Pichon B., Selosse, M-A., Martos, F., Morlon H. (2021). Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. bioRxiv. doi: https://doi.org/10.1101/2021.08.30.458192
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim.BipartiteEvol">sim.BipartiteEvol</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># run the model
set.seed(1)


if(test){

mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 800,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5)

#build the genealogies
gen = make_gen.BipartiteEvol(mod)
plot(gen$H)

#compute the phylogenies
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#plot the result
plot_div.BipartiteEvol(gen,phy1, 1)

#build the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = nx, spatial = FALSE)


## add time steps to a former run
seed=as.integer(10)
set.seed(seed)

mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 200,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5,
                        P=mod$P,H=mod$H)  # former run output

# update the genealogy
gen = make_gen.BipartiteEvol(mod,
                             treeP=gen$P, treeH=gen$H)

# update the phylogenies...
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#... and the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = 10, spatial = FALSE)

}

</code></pre>

<hr>
<h2 id='delineate_phylotypes'>
Automatic phylotypes delineation 
</h2><span id='topic+delineate_phylotypes'></span>

<h3>Description</h3>

<p>This function traverses a tree from the root to the tips, at every node computes the average similarity of all sequences descending from the node, and collapses the sequences into a single phylotype if their sequence dissimilarity is lower than a given threshold. The average similarity can be computed using raw measured of the average similarity or using measures of genetic diversity (nucleotidic diversity &quot;pi&quot; (Nei &amp; Li, 1979) or Watterson &quot;theta&quot; (Watterson, 1975)) which correct for gaps in the nucleotidic alignments (Ferretti et al., 2012). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delineate_phylotypes(tree, thresh=97, sequences, 
method="pi", verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="delineate_phylotypes_+3A_tree">tree</code></td>
<td>

<p>a phylogenetic tree of all the sequences. It must be an object of class &quot;phylo&quot; and must be rooted.
</p>
</td></tr>
<tr><td><code id="delineate_phylotypes_+3A_thresh">thresh</code></td>
<td>

<p>a numeric digit between 0 and 100 indicating the minimal average similarity to collapse sequences within the same phylotype. By default, the average similarity is 97.
</p>
</td></tr>
<tr><td><code id="delineate_phylotypes_+3A_sequences">sequences</code></td>
<td>

<p>a matrix representing the nucleotidic alignment of all the sequences present in the phylogenetic tree. 
</p>
</td></tr>
<tr><td><code id="delineate_phylotypes_+3A_method">method</code></td>
<td>

<p>indicates which method to use to compute the average similarity: &quot;mean&quot; computes the average raw distances between pairs of sequences, &quot;pi&quot; (default) measures the nucleotidic diversity (Nei &amp; Li, 1979) while controlling for gaps in the alignment, and &quot;theta&quot; measures the Watterson theta genetic diversity (Watterson, 1975) also controlling for gaps. 
</p>
</td></tr>
<tr><td><code id="delineate_phylotypes_+3A_verbose">verbose</code></td>
<td>

<p>if TRUE, enables printing of messages.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table with its row names corresponding to the sequence names. The first column corresponds to the phylotype assignation and the second columns indicates the name of the representative sequence of each phylotype (longest sequence available). Phylotypes are numbered starting at 1, and all the phylotypes named &quot;0&quot; correspond to singletons.
</p>


<h3>Author(s)</h3>

<p>Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Perez-Lamarque B, Öpik M, Maliet O, Silva A, Selosse M-A, Martos F, and Morlon H. 2022. Analysing diversification dynamics using barcoding data: The case of an obligate mycorrhizal symbiont, Molecular Ecology, 31:3496–512.
</p>
<p>Ferretti L, Raineri E, Ramos-Onsins S. 2012. Neutrality tests for sequences with missing data. Genetics 191: 1397–1401.
</p>
<p>Morlon H, O’Connor TK, Bryant JA, Charkoudian LK, Docherty KM, Jones E, Kembel SW, Green JL, Bohannan BJM. 2015. The biogeography of putative microbial antibiotic production. PLoS ONE 10.
</p>
<p>Nei M &amp; Li WH, Mathematical model for studying genetic variation in terms of restriction endonucleases, 1979, Proc. Natl. Acad. Sci. USA.
</p>
<p>Watterson GA , On the number of segregating sites in genetical models without recombination, 1975, Theor. Popul. Biol.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pi_estimator">pi_estimator</a></code>
<code><a href="#topic+theta_estimator">theta_estimator</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(phytools)

data(woodmouse)

alignment &lt;- as.character(woodmouse) # nucleotidic alignment 

tree &lt;- midpoint.root(nj(dist.dna(woodmouse, pairwise.deletion = TRUE, 
model = "K80"))) # rooted neighbor-joining tree


delineate_phylotypes(tree, thresh = 99, alignment, method = "pi")


</code></pre>

<hr>
<h2 id='div.models'>
Model comparison of diversification models
</h2><span id='topic+div.models'></span>

<h3>Description</h3>

<p>Applies a set of birth-death models to a phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  div.models(phylo, tot_time, f,
             backbone = FALSE, spec_times = NULL, branch_times = NULL,
             models = c("BCST", "BCST_DCST", "BVAR",
                        "BVAR_DCST", "BCST_DVAR", "BVAR_DVAR"),
             cond, verbose = TRUE, n.max = NULL, rate.max = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="div.models_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="div.models_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="div.models_+3A_f">f</code></td>
<td>

<p>numeric. The sampling fraction given as the number of species in the phylogeny over the number of species described in the taxonomy.
</p>
</td></tr>
<tr><td><code id="div.models_+3A_backbone">backbone</code></td>
<td>

<p>character. Allows to analyse a backbone. Default is FALSE and spec_times and branch_times are then ignored.
Otherwise:
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: for every shift, the probability of the speciation event at the stem age of the subclade is included in the likelihood of the backbone thanks to the argument spec_times.
</p>
</li>
<li><p>&quot;crown.shift&quot;: for every shift, both the probability of the speciation event at the stem age of the subclade and the probability that the stem of the subclade survives to the crown age are included in the likelihood of the backbone thanks to the argument branch_times.
</p>
</li></ul>

</td></tr>
<tr><td><code id="div.models_+3A_spec_times">spec_times</code></td>
<td>

<p>a numeric vector of the stem ages of subclades. Used only if backbone = &quot;stem.shift&quot;. Default is NULL.
</p>
</td></tr>
<tr><td><code id="div.models_+3A_branch_times">branch_times</code></td>
<td>

<p>a list of numeric vectors. Each vector contain the stem and crown ages of subclades (in this order). Used only if backbone = &quot;crown.shift&quot;. Default is NULL.
</p>
</td></tr>
<tr><td><code id="div.models_+3A_models">models</code></td>
<td>

<p>a vector of character. Defines the set of birth-death models to applies e.g. BCST means pure-birth constant rate model, BCST_DVAR means birth constant rate and death variable rate model. Default is c(&quot;BCST&quot;, &quot;BCST_DCST&quot;, &quot;BVAR&quot;, &quot;BVAR_DCST&quot;, &quot;BCST_DVAR&quot;, &quot;BVAR_DVAR&quot;) and applies all combinations of constant or variable rates for speciation and extinction. Time dependency is only exponential.
</p>
</td></tr>
<tr><td><code id="div.models_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>

<ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (used when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and on survival of the two daugther lineages (used when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
<tr><td><code id="div.models_+3A_verbose">verbose</code></td>
<td>

<p>bolean. Wether to print model names and AICc values during the calculation.
</p>
</td></tr>
<tr><td><code id="div.models_+3A_rate.max">rate.max</code></td>
<td>

<p>numeric. Set a limit of diversificaton rates in terms of rate values.
</p>
</td></tr>
<tr><td><code id="div.models_+3A_n.max">n.max</code></td>
<td>

<p>numeric. Set a limit of diversificaton rates in terms of diversity estimates with the deterministic approach.
</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>Parameters of birth-death models are defined backward in time such as a positive alpha corresponds to a speciation rate decreasing through time from the past to the present.
</p>


<h3>Value</h3>

<p>A data.frame with number of parameters, likelihood, AICc and parameter values for all models.
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("Cetacea")
res &lt;- div.models(Cetacea, tot_time = max(node.age(Cetacea)$ages),
                  f = 87/89, cond = "crown")

</code></pre>

<hr>
<h2 id='div.rates'>
Diversification rates through time
</h2><span id='topic+div.rates'></span>

<h3>Description</h3>

<p>Calculates diversification rates through time from shift.estimates() output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  div.rates(phylo, shift.res, combi = 1, part = "backbone",
            time.interval = 1, backbone.option = "crown.shift")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="div.rates_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="div.rates_+3A_shift.res">shift.res</code></td>
<td>

<p>the output resulting from shift.estimates.
</p>
</td></tr>
<tr><td><code id="div.rates_+3A_combi">combi</code></td>
<td>

<p>numeric. The combination of shifts defined by its rank in the global comparison.
</p>
</td></tr>
<tr><td><code id="div.rates_+3A_part">part</code></td>
<td>

<p>character. Specifies for which parts of the combination diversification rates has to be calculated. Default is &quot;backbone&quot; and provides only the backbone rate. Can be &quot;all&quot; for all the parts of a combination or &quot;subclades&quot; for subclades only.
</p>
</td></tr>
<tr><td><code id="div.rates_+3A_backbone.option">backbone.option</code></td>
<td>

<p>type of the backbone analysis (see backbone.option in shift.estimates for more details):
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: rates are calculated from the stem age for subclades.
</p>
</li>
<li><p>&quot;crown.shift&quot;: rates are calculated from the crown age for subclades.
</p>
</li></ul>

</td></tr>
<tr><td><code id="div.rates_+3A_time.interval">time.interval</code></td>
<td>

<p>numeric. Define the time interval (in million years) at which diversification rates are calculated. Default is 1 for a value at each million year.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of matrix with two rows (speciation and extinction) and as many columns as million years from the root to the present.
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.estimates">shift.estimates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># loading data
data("Cetacea")
data("shifts_cetacea")

# with shifts_cetacea the output from shift.estimates()
rates &lt;- div.rates(phylo = Cetacea, shift.res = shifts_cetacea,
                   combi = 1, part = "all")
</code></pre>

<hr>
<h2 id='fit_bd'>
Maximum likelihood fit of the general birth-death model
</h2><span id='topic+fit_bd'></span>

<h3>Description</h3>

<p>Fits the birth-death model with potentially time-varying rates and potentially missing extant species to a phylogeny, by maximum likelihood. Notations follow Morlon et al. PNAS 2011.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_bd(phylo, tot_time, f.lamb, f.mu, lamb_par, mu_par, f = 1,
       meth = "Nelder-Mead", cst.lamb = FALSE, cst.mu = FALSE,
       expo.lamb = FALSE, expo.mu = FALSE, fix.mu = FALSE,
       dt=0, cond = "crown")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_bd_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the hypothesized functional form (e.g. constant, linear, exponential, etc.) of the variation of the speciation rate <code class="reqn">\lambda</code> with time. Any functional form may be used.
This function has two arguments: the first argument is time; the second argument is a numeric vector of the parameters of the time-variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the hypothesized functional form (e.g. constant, linear, exponential, etc.) of the variation of the extinction rate <code class="reqn">\mu</code> with time. Any functional form may be used.
This function has two arguments: the first argument is time; the second argument is a numeric vector of the parameters of the time-variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_lamb_par">lamb_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.lamb to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong.</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_mu_par">mu_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.mu to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model without extinction (for example), mu_par should be empty (vector of length 0). Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_f">f</code></td>
<td>

<p>the fraction of extant species included in the phylogeny
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_meth">meth</code></td>
<td>

<p>optimization to use to maximize the likelihood function, see <em>optim</em> for more details.
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_expo.lamb">expo.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_expo.mu">expo.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_fix.mu">fix.mu</code></td>
<td>

<p>logical: if set to TRUE, the extinction rate <code class="reqn">\mu</code> is fixed and will not be optimized.
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_dt">dt</code></td>
<td>

<p>the default value is 0. In this case, integrals in the likelihood are computed using R &quot;integrate&quot; function, which can be quite slow. If a positive dt is given as argument, integrals are computed using a piece-wise contant approximation, and dt represents the length of the intervals on which functions are assumed to be constant. For an exponential dependency of the speciation rate with time, we found that dt=1e-3 gives a good trade-off between precision and computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>
 <ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (use when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and survival of the 2 daugther lineages (use when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>The lengths of lamb_par and mu_par are used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong. In the f.lamb and f.mu functions, the first argument (time) runs from the present to the past. Hence, if the parameter controlling the variation of <code class="reqn">\lambda</code> with time is estimated to be positive (for example), this means that the speciation rate decreases from past to present. Note that abs(f.lamb) and abs(f.mu) are used in the likelihood computation as speciation and extinction rates should always be positive. A consequence of 
this is that negative speciation/extinction rates estimates can be returned. They should be interpreted in aboslute terms. See Morlon et al. 2020 for a more detailed explanation.</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>the name of the fitted model</p>
</td></tr>
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>lamb_par</code></td>
<td>
<p>a numeric vector of estimated f.lamb parameters, in the same order as defined in f.lamb</p>
</td></tr>
<tr><td><code>mu_par</code></td>
<td>
<p>a numeric vector of estimated f.mu parameters, in the same order as defined in f.mu (if fix.mu is FALSE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>H Morlon</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Morlon, H. (2014) Phylogenetic approaches for studying diversification, Eco Lett 17:508-525
</p>
<p>Morlon, H., Rolland, J. and Condamine, F. (2020) Response to Technical Comment ‘A cautionary note for users of linear diversification dependencies’, Eco Lett
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_fit_bd">plot_fit_bd</a></code>, <code><a href="#topic+plot_dtt">plot_dtt</a></code>, <code><a href="#topic+likelihood_bd">likelihood_bd</a>, <a href="#topic+fit_env">fit_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Some examples may take a little bit of time. Be patient!

data(Cetacea)
tot_time&lt;-max(node.age(Cetacea)$ages)

# Fit the pure birth model (no extinction) with a constant speciation rate

f.lamb &lt;-function(t,y){y[1]}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.09)
mu_par&lt;-c()
result_cst &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
                     f=87/89,cst.lamb=TRUE,fix.mu=TRUE,dt=1e-3)
result_cst$model &lt;- "pure birth with constant speciation rate"
 

# Fit the pure birth model (no extinction) with exponential variation
# of the speciation rate with time

f.lamb &lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.05, 0.01)
mu_par&lt;-c()
result_exp &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
                     f=87/89,expo.lamb=TRUE,fix.mu=TRUE,dt=1e-3)
result_exp$model &lt;- "pure birth with exponential variation in speciation rate"



# Fit the pure birth model (no extinction) with linear variation of
# the speciation rate with time
f.lamb &lt;-function(t,y){abs(y[1] + y[2] * t)}
# alternative formulation that can be used depending on the choice made to avoid negative rates: 
# f.lamb &lt;-function(t,y){pmax(0,y[1] + y[2] * t)}, see Morlon et al. (2020)

f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.09, 0.001)
mu_par&lt;-c()
result_lin &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,f=87/89,fix.mu=TRUE,dt=1e-3)
result_lin$model &lt;- "pure birth with linear variation in speciation rate"


# Fit a birth-death model with exponential variation of the speciation
# rate with time and constant extinction

f.lamb&lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu &lt;-function(t,y){y[1]}
lamb_par &lt;- c(0.05, 0.01)
mu_par &lt;-c(0.005)
result_bexp_dcst &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
                           f=87/89,expo.lamb=TRUE,cst.mu=TRUE,dt=1e-3)
result_bexp_dcst$model &lt;- "birth-death with exponential variation in speciation rate
                           and constant extinction"


# Find the best model

index &lt;- which.min(c(result_cst$aicc, result_exp$aicc, result_lin$aicc,result_bexp_dcst$aicc))
rbind(result_cst, result_exp, result_lin, result_bexp_dcst)[index,]

</code></pre>

<hr>
<h2 id='fit_bd_backbone'>
Maximum likelihood fit of the general birth-death model (backbone)
</h2><span id='topic+fit_bd_backbone'></span>

<h3>Description</h3>

<p>Fits the birth-death model with potentially time-varying rates and potentially missing extant species to a phylogeny, by maximum likelihood. Notations follow Morlon et al. PNAS 2011. Modified version of fit_bd for backbones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_bd_backbone(phylo, tot_time, f.lamb, f.mu, lamb_par, mu_par, f = 1,
                backbone, spec_times, branch_times,
                meth = "Nelder-Mead", cst.lamb = FALSE, cst.mu = FALSE,
                expo.lamb = FALSE, expo.mu = FALSE, fix.mu = FALSE,
                dt=1e-3, cond = "crown", model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_bd_backbone_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the hypothesized functional form (e.g. constant, linear, exponential, etc.) of the variation of the speciation rate <code class="reqn">\lambda</code> with time. Any functional form may be used.
This function has two arguments: the first argument is time; the second argument is a numeric vector of the parameters of the time-variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the hypothesized functional form (e.g. constant, linear, exponential, etc.) of the variation of the extinction rate <code class="reqn">\mu</code> with time. Any functional form may be used.
This function has two arguments: the first argument is time; the second argument is a numeric vector of the parameters of the time-variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_lamb_par">lamb_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.lamb to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong.</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_mu_par">mu_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.mu to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model without extinction (for example), mu_par should be empty (vector of length 0). Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_f">f</code></td>
<td>

<p>the fraction of extant species included in the phylogeny
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_backbone">backbone</code></td>
<td>

<p>character. Allows to analyse a backbone. Default is FALSE and spec_times and branch_times are then ignored.
Otherwise
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: for every shift, the probability of the speciation event at the stem age of the subclade is included in the likelihood of the backbone thanks to the argument spec_times.
</p>
</li>
<li><p>&quot;crown.shift&quot;: for every shift, both the probability of the speciation event at the stem age of the subclade and the probability that the stem of the subclade survives to the crown age are included in the likelihood of the backbone thanks to the argument branch_times.
</p>
</li></ul>

</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_spec_times">spec_times</code></td>
<td>

<p>a numeric vector of the stem ages of subclades. Used only if backbone = &quot;stem.shift&quot;. Default is NULL.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_branch_times">branch_times</code></td>
<td>

<p>a list of numeric vectors. Each vector contains the stem and crown ages of subclades (in this order). Used only if backbone = &quot;crown.shift&quot;. Default is NULL.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_meth">meth</code></td>
<td>

<p>optimization to use to maximize the likelihood function, see <em>optim</em> for more details.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_expo.lamb">expo.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_expo.mu">expo.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_fix.mu">fix.mu</code></td>
<td>

<p>logical: if set to TRUE, the extinction rate <code class="reqn">\mu</code> is fixed and will not be optimized.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_dt">dt</code></td>
<td>

<p>the default value is 0. In this case, integrals in the likelihood are computed using R &quot;integrate&quot; function, which can be quite slow. If a positive dt is given as argument, integrals are computed using a piece-wise contant approximation, and dt represents the length of the intervals on which functions are assumed to be constant. For an exponential dependency of the speciation rate with time, we found that dt=1e-3 gives a good trade-off between precision and computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>
 <ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (use when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and survival of the two daugther lineages (use when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
<tr><td><code id="fit_bd_backbone_+3A_model">model</code></td>
<td>

<p>character. The model name as defined in the function div.models.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The lengths of lamb_par and mu_par are used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong. In the f.lamb and f.mu functions, the first argument (time) runs from the present to the past. Hence, if the parameter controlling the variation of <code class="reqn">\lambda</code> with time is estimated to be positive (for example), this means that the speciation rate decreases from past to present. Note that abs(f.lamb) and abs(f.mu) are used in the likelihood computation as speciation and extinction rates should always be positive. A consequence of 
this is that negative speciation/extinction rates estimates can be returned. They should be interpreted in absolute terms. See Morlon et al. 2020 for a more detailed explanation.</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>the name of the fitted model</p>
</td></tr>
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>lamb_par</code></td>
<td>
<p>a numeric vector of estimated f.lamb parameters, in the same order as defined in f.lamb</p>
</td></tr>
<tr><td><code>mu_par</code></td>
<td>
<p>a numeric vector of estimated f.mu parameters, in the same order as defined in f.mu (if fix.mu is FALSE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hélène Morlon, Nathan Mazet</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Morlon, H. (2014) Phylogenetic approaches for studying diversification, Eco Lett 17:508-525
Morlon, H., Rolland, J. and Condamine, F. (2020) Response to Technical Comment ‘A cautionary note for users of linear diversification dependencies’, Eco Lett
Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_fit_bd">plot_fit_bd</a></code>, <code><a href="#topic+plot_dtt">plot_dtt</a></code>, <code><a href="#topic+likelihood_bd">likelihood_bd</a>, <a href="#topic+fit_env">fit_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Some examples may take a little bit of time. Be patient!
data(Cetacea)
tot_time&lt;-max(node.age(Cetacea)$ages)
# Fit the pure birth model (no extinction) with a constant speciation rate
f.lamb &lt;-function(t,y){y[1]}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.09)
mu_par&lt;-c()
#result_cst &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
#                     f=87/89,cst.lamb=TRUE,fix.mu=TRUE,dt=1e-3)
#result_cst$model &lt;- "pure birth with constant speciation rate"
# Fit the pure birth model (no extinction) with exponential variation
# of the speciation rate with time
f.lamb &lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.05, 0.01)
mu_par&lt;-c()
#result_exp &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
#                     f=87/89,expo.lamb=TRUE,fix.mu=TRUE,dt=1e-3)
#result_exp$model &lt;- "pure birth with exponential variation in speciation rate"
# Fit the pure birth model (no extinction) with linear variation of
# the speciation rate with time
f.lamb &lt;-function(t,y){abs(y[1] + y[2] * t)}
# alternative formulation that can be used depending on the choice made to avoid negative rates: 
# f.lamb &lt;-function(t,y){pmax(0,y[1] + y[2] * t)}, see Morlon et al. (2020)
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.09, 0.001)
mu_par&lt;-c()
#result_lin &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,f=87/89,fix.mu=TRUE,dt=1e-3)
#result_lin$model &lt;- "pure birth with linear variation in speciation rate"
# Fit a birth-death model with exponential variation of the speciation
# rate with time and constant extinction
f.lamb&lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu &lt;-function(t,y){y[1]}
lamb_par &lt;- c(0.05, 0.01)
mu_par &lt;-c(0.005)
#result_bexp_dcst &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
#                           f=87/89,expo.lamb=TRUE,cst.mu=TRUE,dt=1e-3)
#result_bexp_dcst$model &lt;- "birth-death with exponential variation in speciation rate
#                           and constant extinction"
# Find the best model
#index &lt;- which.min(c(result_cst$aicc, result_exp$aicc, result_lin$aicc,result_bexp_dcst$aicc))
#rbind(result_cst, result_exp, result_lin, result_bexp_dcst)[index,]
</code></pre>

<hr>
<h2 id='fit_bd_backbone_c'>
Maximum likelihood fit of the general birth-death model (backbone and constraints)
</h2><span id='topic+fit_bd_backbone_c'></span>

<h3>Description</h3>

<p>Fits the birth-death model with potentially time-varying rates and potentially missing extant species to a phylogeny, by maximum likelihood. Notations follow Morlon et al. PNAS 2011. Modified version of fit_bd for backbones and to add constraints on rate estimtes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  fit_bd_backbone_c(phylo, tot_time, f.lamb, f.mu, lamb_par, mu_par, f = 1,
                    backbone, spec_times, branch_times,
                    meth = "Nelder-Mead", cst.lamb = FALSE, cst.mu = FALSE,
                    expo.lamb = FALSE, expo.mu = FALSE, fix.mu = FALSE,
                    dt=1e-3, cond = "crown", model, rate.max, n.max)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_bd_backbone_c_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the hypothesized functional form (e.g. constant, linear, exponential, etc.) of the variation of the speciation rate <code class="reqn">\lambda</code> with time. Any functional form may be used.
This function has two arguments: the first argument is time; the second argument is a numeric vector of the parameters of the time-variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the hypothesized functional form (e.g. constant, linear, exponential, etc.) of the variation of the extinction rate <code class="reqn">\mu</code> with time. Any functional form may be used.
This function has two arguments: the first argument is time; the second argument is a numeric vector of the parameters of the time-variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_lamb_par">lamb_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.lamb to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong.</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_mu_par">mu_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.mu to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model without extinction (for example), mu_par should be empty (vector of length 0). Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_f">f</code></td>
<td>

<p>the fraction of extant species included in the phylogeny
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_backbone">backbone</code></td>
<td>

<p>character. Allows to analyse a backbone. Default is FALSE and spec_times and branch_times are then ignored.
Otherwise
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: the stems of subclades are included in subclade analyses;
</p>
</li>
<li><p>&quot;crown.shift&quot;: the stems of subclades are included in the backbone analysis.
</p>
</li></ul>

</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_spec_times">spec_times</code></td>
<td>

<p>a numeric vector of the stem ages of subclades. Used only if backbone = &quot;stem.shift&quot;. Default is NULL.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_branch_times">branch_times</code></td>
<td>

<p>a list of numeric vectors. Each vector contains the stem and crown ages of subclades (in this order). Used only if backbone = &quot;crown.shift&quot;. Default is NULL.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_meth">meth</code></td>
<td>

<p>optimization to use to maximize the likelihood function, see <em>optim</em> for more details.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_expo.lamb">expo.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_expo.mu">expo.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_fix.mu">fix.mu</code></td>
<td>

<p>logical: if set to TRUE, the extinction rate <code class="reqn">\mu</code> is fixed and will not be optimized.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_dt">dt</code></td>
<td>

<p>the default value is 0. In this case, integrals in the likelihood are computed using R &quot;integrate&quot; function, which can be quite slow. If a positive dt is given as argument, integrals are computed using a piece-wise contant approximation, and dt represents the length of the intervals on which functions are assumed to be constant. For an exponential dependency of the speciation rate with time, we found that dt=1e-3 gives a good trade-off between precision and computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>
 <ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (use when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and survival of the 2 daugther lineages (use when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_model">model</code></td>
<td>

<p>character. The model name as defined in the function div.models.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_rate.max">rate.max</code></td>
<td>

<p>numeric. Set a limit of diversificaton rates in terme of rate values.
</p>
</td></tr>
<tr><td><code id="fit_bd_backbone_c_+3A_n.max">n.max</code></td>
<td>

<p>numeric. Set a limit of diversificaton rates in terms of diversity estimates with the deterministic approach.
</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>The lengths of lamb_par and mu_par are used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong. In the f.lamb and f.mu functions, the first argument (time) runs from the present to the past. Hence, if the parameter controlling the variation of <code class="reqn">\lambda</code> with time is estimated to be positive (for example), this means that the speciation rate decreases from past to present. Note that abs(f.lamb) and abs(f.mu) are used in the likelihood computation as speciation and extinction rates should always be positive. A consequence of 
this is that negative speciation/extinction rates estimates can be returned. They should be interpreted in absolute terms. See Morlon et al. 2020 for a more detailed explanation.</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>the name of the fitted model</p>
</td></tr>
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>lamb_par</code></td>
<td>
<p>a numeric vector of estimated f.lamb parameters, in the same order as defined in f.lamb</p>
</td></tr>
<tr><td><code>mu_par</code></td>
<td>
<p>a numeric vector of estimated f.mu parameters, in the same order as defined in f.mu (if fix.mu is FALSE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hélène Morlon, Nathan Mazet</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Morlon, H. (2014) Phylogenetic approaches for studying diversification, Eco Lett 17:508-525
Morlon, H., Rolland, J. and Condamine, F. (2020) Response to Technical Comment ‘A cautionary note for users of linear diversification dependencies’, Eco Lett
Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_fit_bd">plot_fit_bd</a></code>, <code><a href="#topic+plot_dtt">plot_dtt</a></code>, <code><a href="#topic+likelihood_bd">likelihood_bd</a>, <a href="#topic+fit_env">fit_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Some examples may take a little bit of time. Be patient!
data(Cetacea)
tot_time&lt;-max(node.age(Cetacea)$ages)
# Fit the pure birth model (no extinction) with a constant speciation rate
f.lamb &lt;-function(t,y){y[1]}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.09)
mu_par&lt;-c()
#result_cst &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
#                     f=87/89,cst.lamb=TRUE,fix.mu=TRUE,dt=1e-3)
#result_cst$model &lt;- "pure birth with constant speciation rate"
# Fit the pure birth model (no extinction) with exponential variation
# of the speciation rate with time
f.lamb &lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.05, 0.01)
mu_par&lt;-c()
#result_exp &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
#                     f=87/89,expo.lamb=TRUE,fix.mu=TRUE,dt=1e-3)
#result_exp$model &lt;- "pure birth with exponential variation in speciation rate"
# Fit the pure birth model (no extinction) with linear variation of
# the speciation rate with time
f.lamb &lt;-function(t,y){abs(y[1] + y[2] * t)}
# alternative formulation that can be used depending on the choice made to avoid negative rates: 
# f.lamb &lt;-function(t,y){pmax(0,y[1] + y[2] * t)}, see Morlon et al. (2020)
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.09, 0.001)
mu_par&lt;-c()
#result_lin &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,f=87/89,fix.mu=TRUE,dt=1e-3)
#result_lin$model &lt;- "pure birth with linear variation in speciation rate"
# Fit a birth-death model with exponential variation of the speciation
# rate with time and constant extinction
f.lamb&lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu &lt;-function(t,y){y[1]}
lamb_par &lt;- c(0.05, 0.01)
mu_par &lt;-c(0.005)
#result_bexp_dcst &lt;- fit_bd(Cetacea,tot_time,f.lamb,f.mu,lamb_par,mu_par,
#                           f=87/89,expo.lamb=TRUE,cst.mu=TRUE,dt=1e-3)
#result_bexp_dcst$model &lt;- "birth-death with exponential variation in speciation rate
#                           and constant extinction"
# Find the best model
#index &lt;- which.min(c(result_cst$aicc, result_exp$aicc, result_lin$aicc,result_bexp_dcst$aicc))
#rbind(result_cst, result_exp, result_lin, result_bexp_dcst)[index,]
</code></pre>

<hr>
<h2 id='fit_bd_in_past'>
Maximum likelihood fit of the general birth-death model excluding the recent past
</h2><span id='topic+fit_bd_in_past'></span>

<h3>Description</h3>

<p>Fits the birth-death model with potentially time-varying rates and potentially missing extant species to a phylogeny, by maximum likelihood while excluding the recent past. Notations follow Morlon et al. PNAS 2011.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_bd_in_past(phylo, tot_time, time_stop, f.lamb, f.mu, lamb_par, mu_par, desc, tot_desc, 
       meth = "Nelder-Mead", cst.lamb = FALSE, cst.mu = FALSE,
       expo.lamb = FALSE, expo.mu = FALSE, fix.mu = FALSE,
       dt=0, cond = "crown")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_bd_in_past_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation) that does not include any recent speciation (i.e. no speciation events between time_stop and the present). 
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_time_stop">time_stop</code></td>
<td>

<p>the age of the phylogeny where to stop the birth-death process: it excludes the recent past (between the present and time_stop), while conditioning on the survival of the lineages from time_stop to the present.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the hypothesized functional form (e.g. constant, linear, exponential, etc.) of the variation of the speciation rate <code class="reqn">\lambda</code> with time. Any functional form may be used.
This function has two arguments: the first argument is time; the second argument is a numeric vector of the parameters of the time-variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the hypothesized functional form (e.g. constant, linear, exponential, etc.) of the variation of the extinction rate <code class="reqn">\mu</code> with time. Any functional form may be used.
This function has two arguments: the first argument is time; the second argument is a numeric vector of the parameters of the time-variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_lamb_par">lamb_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.lamb to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong.</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_mu_par">mu_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.mu to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model without extinction (for example), mu_par should be empty (vector of length 0). Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_desc">desc</code></td>
<td>

<p>the number of  lineages present at present in the reconstructed phylogenetic tree.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_tot_desc">tot_desc</code></td>
<td>

<p>the total number of extant species (including in the unsampled ones). 
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_meth">meth</code></td>
<td>

<p>optimization to use to maximize the likelihood function, see <em>optim</em> for more details.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_expo.lamb">expo.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_expo.mu">expo.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_fix.mu">fix.mu</code></td>
<td>

<p>logical: if set to TRUE, the extinction rate <code class="reqn">\mu</code> is fixed and will not be optimized.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_dt">dt</code></td>
<td>

<p>the default value is 0. In this case, integrals in the likelihood are computed using R &quot;integrate&quot; function, which can be quite slow. If a positive dt is given as argument, integrals are computed using a piece-wise contant approximation, and dt represents the length of the intervals on which functions are assumed to be constant. For an exponential dependency of the speciation rate with time, we found that dt=1e-3 gives a good trade-off between precision and computation time.
</p>
</td></tr>
<tr><td><code id="fit_bd_in_past_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>
 <ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (use when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and survival of the 2 daugther lineages (use when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>The lengths of lamb_par and mu_par are used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong. In the f.lamb and f.mu functions, the first argument (time) runs from the present to the past. Hence, if the parameter controlling the variation of <code class="reqn">\lambda</code> with time is estimated to be positive (for example), this means that the speciation rate decreases from past to present. Note that abs(f.lamb) and abs(f.mu) are used in the likelihood computation as speciation and extinction rates should always be positive. A consequence of 
this is that negative speciation/extinction rates estimates can be returned. They should be interpreted in aboslute terms. See Morlon et al. 2020 for a more detailed explanation.</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>the name of the fitted model</p>
</td></tr>
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>lamb_par</code></td>
<td>
<p>a numeric vector of estimated f.lamb parameters, in the same order as defined in f.lamb</p>
</td></tr>
<tr><td><code>mu_par</code></td>
<td>
<p>a numeric vector of estimated f.mu parameters, in the same order as defined in f.mu (if fix.mu is FALSE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>H Morlon, E Lewitus, B Perez-Lamarque</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Lewitus, E., Bittner, L., Malviya, S., Bowler, C., &amp; Morlon, H. (2018) Clade-specific diversification dynamics of marine diatoms since the Jurassic <em>Nature Ecology and Evolution</em>, 2(11), 1715–1723
</p>
<p>Perez-Lamarque, B., Öpik, M., Maliet, O., Afonso Silva, A., Selosse, M-A., Martos, F., Morlon, H. (2022) Analysing diversification dynamics using barcoding data: The case of an obligate mycorrhizal symbiont, <em>Molecular Ecology</em> 31: 3496–3512
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_env_in_past">fit_env_in_past</a></code>,<code><a href="#topic+fit_bd">fit_bd</a></code>,<code><a href="#topic+plot_fit_bd">plot_fit_bd</a></code>, <code><a href="#topic+plot_dtt">plot_dtt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ape)
library(phytools)

data(Cetacea)

plot(Cetacea)
tot_time&lt;-max(node.age(Cetacea)$ages)

# slice the Cetaceae tree 10 Myr ago:
time_stop=10
sliced_tree &lt;- Cetacea
sliced_sub_trees &lt;- treeSlice(sliced_tree,slice = tot_time-time_stop, trivial=TRUE)
for (i in 1:length(sliced_sub_trees)){

  if (Ntip(sliced_sub_trees[[i]])&gt;1){
  sliced_tree &lt;- drop.tip(sliced_tree, 
    tip=sliced_sub_trees[[i]]$tip.label[2:Ntip(sliced_sub_trees[[i]])])
}}

for (i in which(node.depth.edgelength(sliced_tree)&gt;(tot_time-time_stop))){
  temp = sliced_tree$edge.length[which(sliced_tree$edge[,2]==i)]-time_stop
  sliced_tree$edge.length[which(sliced_tree$edge[,2]==i)] &lt;- temp
  }

Ntip(sliced_tree) # 27 lineages present 10 Myr have survived until today

# Now we can fit birth-death models excluding the 10 last Myr

# Fit the pure birth model (no extinction) with a constant speciation rate

f.lamb &lt;-function(t,y){y[1]}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.09)
mu_par&lt;-c()

result_cst &lt;- fit_bd_in_past(sliced_tree, tot_time, time_stop, f.lamb, f.mu, 
                             desc=Ntip(Cetacea), tot_desc=89, lamb_par, mu_par,
                             cst.lamb = TRUE, fix.mu=TRUE, dt=1e-3)

</code></pre>

<hr>
<h2 id='fit_ClaDS'>
Fit ClaDS to a phylogeny
</h2><span id='topic+fit_ClaDS'></span>

<h3>Description</h3>

<p>Performs the inferrence of branch-specific speciation rates and the model's hyper parameters for the model with constant extinction rate (ClaDS1) or constant turnover rate (ClaDS2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_ClaDS(tree,sample_fraction,iterations, thin = 50, file_name = NULL, it_save = 1000,
                     model_id = "ClaDS2", nCPU = 1, mcmcSampler = NULL,
                     verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_ClaDS_+3A_tree">tree</code></td>
<td>

<p>An object of class 'phylo'
</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_sample_fraction">sample_fraction</code></td>
<td>

<p>The sampling fraction for the clade on which the inference is performed.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_iterations">iterations</code></td>
<td>

<p>Number of steps in the MCMC, should be a multiple of <code>thin</code>.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_thin">thin</code></td>
<td>

<p>Number of iterations between two chain state's recordings.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_file_name">file_name</code></td>
<td>

<p>Name of the file in which the result will be saved. Use file_name = NULL (the default) to disable this option.</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_it_save">it_save</code></td>
<td>

<p>Number of iteration between each backup of the result in file_name.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_model_id">model_id</code></td>
<td>

<p>&quot;ClaDS1&quot; for constant extinction rate, &quot;ClaDS2&quot; (the default) for constant turnover rate.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_ncpu">nCPU</code></td>
<td>

<p>The number of CPUs to use. Should be either 1 or 3.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_mcmcsampler">mcmcSampler</code></td>
<td>

<p>Optional output of <code><a href="#topic+fit_ClaDS">fit_ClaDS</a></code> to continue an already started run.</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_verbose">verbose</code></td>
<td>

<p>if TRUE, enables printing of messages.</p>
</td></tr>
<tr><td><code id="fit_ClaDS_+3A_...">...</code></td>
<td>

<p>Optional arguments, see details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses a blocked Differential Evolution (DE) MCMC sampler, with sampling from the
past of the chains (Ter Braak, 2006; ter Braak and Vrugt, 2008). This sampler is self-adaptive
because proposals are generated from the past of the chains. In this sampler, three chains are
run simultaneously. Block updates is implemented by first drawing the number
of parameters to be updated from a truncated geometric distribution with mean 3, drawing
uniformly which parameter to update, and then following the normal DE algorithm.
</p>
<p>The available optional arguments are : 
</p>

<dl>
<dt>Nchain</dt><dd>
<p>Number of MCMC chains (default to 3).
</p>
</dd>
<dt>res_ClaDS0</dt><dd>
<p>The output of ClaDS0 to use as a startpoint. If NULL (the default) a random startpoint is used for the branch-specific speciation rates for each chain.
</p>
</dd>
<dt>l0</dt><dd>
<p>The starting value for lambda_0 (not used if res_ClaDS0 != NULL).
</p>
</dd>
<dt>s0</dt><dd>
<p>The starting value for sigma (not used if res_ClaDS0 != NULL).
</p>
</dd>
<dt>nlambda</dt><dd>
<p>Number of subdivisions for the rate space discretization (use in the likelihood computation). Default to 1000.
</p>
</dd>
<dt>nt</dt><dd>
<p>Number of subdivisions for the time space discretization (use in the likelihood computation). Default to 30.
</p>
</dd></dl>



<h3>Value</h3>

<p>A 'list' object with fields :
</p>
<table role = "presentation">
<tr><td><code>post</code></td>
<td>
<p>The posterior function.</p>
</td></tr>
<tr><td><code>startvalue</code></td>
<td>
<p>The starting value for the MCMC.</p>
</td></tr>
<tr><td><code>numPars</code></td>
<td>
<p>The number of parameter in the model, including the branch-specific speciation rates.</p>
</td></tr>
<tr><td><code>Nchain</code></td>
<td>
<p>The number of MCMC chains ran simultaneously.</p>
</td></tr>
<tr><td><code>currentLPs</code></td>
<td>
<p>The current values of the logposterior for th <code>Nchains</code> chains.</p>
</td></tr>
<tr><td><code>proposalGenerator</code></td>
<td>
<p>The proposal distribution for the MCMC sampler.</p>
</td></tr>
<tr><td><code>former</code></td>
<td>
<p>The last output of <code>post</code> for each of the chains.</p>
</td></tr>
<tr><td><code>thin</code></td>
<td>
<p>Number of iterations between two chain state's recordings.</p>
</td></tr>
<tr><td><code>alpha_effect</code></td>
<td>
<p>A vector of size <code>nrow(tree$edge)</code>, where the ith element is the number of branches on the path from the crown of the tree and branch i (used internally in other functions).</p>
</td></tr>
<tr><td><code>consoleupdates</code></td>
<td>
<p>The frequency at which the sampler state should be printed.</p>
</td></tr>
<tr><td><code>likelihood</code></td>
<td>
<p>The likelihood function, used internally.</p>
</td></tr>
<tr><td><code>relToAbs</code></td>
<td>
<p>A function mapping the relative changes in speciation rates to the absolute speciation rates for the object <code>phylo</code>, used internally.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Ter Braak, C. J. 2006. A markov chain monte carlo version of the genetic algorithm differential
evolution: easy bayesian computing for real parameter spaces. <em>Statistics and Computing 16:239-
249</em>.
</p>
<p>ter Braak, C. J. and J. A. Vrugt. 2008. Differential evolution markov chain with snooker updater
and fewer chains. <em>Statistics and Computing 18:435-446</em>.
</p>
<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_ClaDS0">fit_ClaDS0</a></code>, <code><a href="#topic+plot_ClaDS_chains">plot_ClaDS_chains</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(test){
data("Caprimulgidae")

sample_fraction = 0.61

sampler = fit_ClaDS(Caprimulgidae, sample_fraction, 1000, thin = 50, 
          file_name = NULL, model_id="ClaDS2", nCPU = 1)
plot_ClaDS_chains(sampler)

# continue the same run 
sampler = fit_ClaDS(Caprimulgidae, sample_fraction, 50, mcmcSampler = sampler)




# plot the result of the analysis (saved in "Caprimulgidae_ClaDS2", after thinning)

data("Caprimulgidae_ClaDS2")

# plot the mcmc chains
plot_ClaDS_chains(Caprimulgidae_ClaDS2$sampler)

# extract the Maxima A Posteriori for each parameter
maps = getMAPS_ClaDS(Caprimulgidae_ClaDS2$sampler, thin = 1)
print(paste0("sigma = ", maps[1], " ; alpha = ", 
  maps[2], " ; epsilon = ", maps[3], " ; l_0 = ", maps[4] ))
  
# plot the infered branch specific speciation rates
plot_ClaDS_phylo(Caprimulgidae_ClaDS2$tree, maps[-(1:4)])
}

</code></pre>

<hr>
<h2 id='fit_ClaDS0'>
Infer ClaDS0's parameter on a phylogeny</h2><span id='topic+fit_ClaDS0'></span>

<h3>Description</h3>

<p>Infer branch-specific speciation rates and the model's hyper parameters for the pure-birth model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_ClaDS0(tree, name, pamhLocalName = NULL,  
            iteration = 1e+07, thin = 20000, update = 1000, 
            adaptation = 10, seed = NULL, nCPU = 3,
            verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_ClaDS0_+3A_tree">tree</code></td>
<td>

<p>An object of class 'phylo'.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_name">name</code></td>
<td>

<p>The name of the file in which the results will be saved. Use name = NULL to disable this option.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_pamhlocalname">pamhLocalName</code></td>
<td>

<p>The function is writing in a text file to make the execution quicker, this is the name of this file.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_iteration">iteration</code></td>
<td>

<p>Number of iterations after which the gelman factor is computed and printed. The function stops if it is below 1.05
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_thin">thin</code></td>
<td>

<p>Number of iterations between two chain state's recordings.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_update">update</code></td>
<td>

<p>Number of iterations between two adjustments of the proposal parameters during the adaptation phase of the sampler.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_adaptation">adaptation</code></td>
<td>

<p>Number of times the proposal is adjusted during the adaptation phase of the sampler.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_seed">seed</code></td>
<td>

<p>An optional seed for the MCMC run.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_ncpu">nCPU</code></td>
<td>

<p>The number of CPUs to use. Should be either 1 or 3.
</p>
</td></tr>
<tr><td><code id="fit_ClaDS0_+3A_verbose">verbose</code></td>
<td>

<p>if TRUE, enables printing of messages.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses a Metropolis within Gibbs MCMC sampler with a bactrian proposal (ref) with an initial adaptation phase. During this phase, the proposal is adjusted &quot;adaptation&quot; times every &quot;update&quot; iterations to reach a goal acceptance rate of 0.3.
</p>
<p>To monitor convergence, 3 independant MCMC chains are run simultaneously and the Gelman statistics is computed every &quot;iteration&quot; iterations. The inference is stopped when the maximum of the one dimentional Gelman statistics (computed for each of the parameters) is below 1.05.</p>


<h3>Value</h3>

<p>A mcmc.list object with the three MCMC chains.
</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getMAPS_ClaDS0">getMAPS_ClaDS0</a></code>, <code><a href="#topic+plot_ClaDS0_chains">plot_ClaDS0_chains</a></code>, <code><a href="#topic+fit_ClaDS">fit_ClaDS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)


if(test){

obj= sim_ClaDS( lambda_0=0.1,    
                mu_0=0.5,      
                sigma_lamb=0.7,         
                alpha_lamb=0.90,     
                condition="taxa",    
                taxa_stop = 20,    
                prune_extinct = TRUE)  

tree = obj$tree
speciation_rates = obj$lamb[obj$rates]
extinction_rates = obj$mu[obj$rates]

plot_ClaDS_phylo(tree,speciation_rates)

sampler = fit_ClaDS0(tree=tree,        
              name="ClaDS0_example.Rdata",      
              nCPU=1,               
              pamhLocalName = "local",
              iteration=500000,
              thin=2000,
              update=1000, adaptation=5) 
              
# extract the Maximum A Posteriori for each of the parameters
MAPS = getMAPS_ClaDS0(tree, sampler, thin = 10)

# plot the simulated (on the left) and inferred speciation rates (on the right)
# on the same color scale
plot_ClaDS_phylo(tree, speciation_rates, MAPS[-(1:3)])
     
}         
</code></pre>

<hr>
<h2 id='fit_coal_cst'>
Maximum likelihood fit of the equilibrium model
</h2><span id='topic+fit_coal_cst'></span>

<h3>Description</h3>

<p>Fits the equilibrium diversity model with potentially time-varying turnover rate and potentially missing extant species to a phylogeny, by maximum likelihood. 
The implementation allows only exponential time variation of the turnover rate, although this could be modified using expressions in Morlon et al. PloSB 2010. Notations follow Morlon et al. PLoSB 2010.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_coal_cst(phylo, tau0 = 1e-2, gamma = 1, cst.rate = FALSE,
             meth = "Nelder-Mead", N0 = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_coal_cst_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_coal_cst_+3A_tau0">tau0</code></td>
<td>

<p>initial value of the turnover rate at present (used by the optimization algorithm)
</p>
</td></tr>
<tr><td><code id="fit_coal_cst_+3A_gamma">gamma</code></td>
<td>

<p>initial value of the parameter controlling the exponential variation in turnover rate (used by the optimization algorithm)
</p>
</td></tr>
<tr><td><code id="fit_coal_cst_+3A_cst.rate">cst.rate</code></td>
<td>

<p>logical: should be set to TRUE to fit an equilibrium diversity model with time-constant turnover rate (know as the Hey model, model 1 in Morlon et al. PloSB 2010). By default, a model with expontential time-varying rate exponential is fitted (model 2 in Morlon et al. PloSB 2010).
</p>
</td></tr>
<tr><td><code id="fit_coal_cst_+3A_meth">meth</code></td>
<td>

<p>optimization to use to maximize the likelihood function, see <em>optim</em> for more details.
</p>
</td></tr>
<tr><td><code id="fit_coal_cst_+3A_n0">N0</code></td>
<td>
<p>Number of extant species.
With default value(0), N0 is set to the number of tips in the phylogeny. That is, the phylogeny is assumed to be 100% complete. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits models 1 (when cst.rate=TRUE) and 2 (when cst.rate=FALSE) from the PloSB 2010 paper. Likelihoods arising from these models are directly comparable to likelihoods from the fit_coal_var function, thus allowing to test support for equilibrium versus expanding diversity scenarios. Time runs from the present to the past. Hence, if gamma is estimated to be positive (for example), this means that the speciation rate decreases from past to present.
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>the name of the fitted model</p>
</td></tr>
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>tau0</code></td>
<td>
<p>the estimated turnover rate at present</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>the estimated parameter controlling the exponential variation in turnover rate (if cst.rate is FALSE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>H Morlon</p>


<h3>References</h3>

<p>Hey, J. (1992) Using phylogenetic trees to study speciation and extinction, Evolution, 46: 627-640
</p>
<p>Morlon, H., Potts, M.D., Plotkin, J.B. (2010) Inferring the dynamics of diversification: a coalescent approach, PLoS B, 8(9): e1000493
</p>
<p>Morlon, H., Kemps, B., Plotkin, J.B., Brisson, D. (2012) Explosive radiation of a bacterial species group, Evolution, 66: 2577-2586
</p>
<p>Morlon, H. (2014) Phylogenetic approaches for studying diversification, Eco Lett, 17:508-525
</p>


<h3>See Also</h3>

<p><code><a href="#topic+likelihood_coal_cst">likelihood_coal_cst</a></code>, <code><a href="#topic+fit_coal_var">fit_coal_var</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)


if(test){
result &lt;- fit_coal_cst(Cetacea, tau0=1.e-3, gamma=-1, cst.rate=FALSE, N0=89)
print(result)
}
</code></pre>

<hr>
<h2 id='fit_coal_var'>
Fit birth-death model using a coalescent approch
</h2><span id='topic+fit_coal_var'></span>

<h3>Description</h3>

<p>Fits the expanding diversity model with potentially time-varying rates and potentially missing extant species to a phylogeny, by maximum likelihood. 
The implementation allows only exponential time variation of the speciation and extinction rates, although this could be modified using expressions in Morlon et al. PloSB 2010. Notations follow Morlon et al. PLoSB 2010.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_coal_var(phylo, lamb0 = 0.1, alpha = 1, mu0 = 0.01, beta = 0,
             meth = "Nelder-Mead", N0 = 0, cst.lamb = FALSE, cst.mu = FALSE,
             fix.eps = FALSE, mu.0 = FALSE, pos = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_coal_var_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_lamb0">lamb0</code></td>
<td>

<p>initial value of the speciation rate at present (used by the optimization algorithm)
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_alpha">alpha</code></td>
<td>

<p>initial value of the parameter controlling the exponential variation in speciation rate (used by the optimization algorithm)
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_mu0">mu0</code></td>
<td>

<p>initial value of the extinction rate at present (used by the optimization algorithm)
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_beta">beta</code></td>
<td>

<p>initial value of the parameter controlling the exponential variation in extinction rate.
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_meth">meth</code></td>
<td>

<p>optimization to use to maximize the likelihood function, see <em>optim</em> for more details.
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_n0">N0</code></td>
<td>

<p>Number of extant species.
With default value(0), N0 is set to the number of tips in the phylogeny. That is, the phylogeny is assumed to be 100% complete. 
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time, models 3, 4b &amp; 5 in Morlon et al. PloSB 2010) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time, models 3 &amp; 4a in Morlon et al. PloSB 2010) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_fix.eps">fix.eps</code></td>
<td>

<p>logical: should be set to TRUE only if the extinction fraction is constant (i.e. does not depend on time, model 4c in Morlon et al. PloSB 2010)
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_mu.0">mu.0</code></td>
<td>

<p>logical: should be set to TRUE to force the extinction rate to 0 (models 5 &amp; 6 in Morlon et al. PloSB 2010)
</p>
</td></tr>
<tr><td><code id="fit_coal_var_+3A_pos">pos</code></td>
<td>

<p>logical: should be set to FALSE only to not enforce positive speciation and extinction rates
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function fits models 3 to 6 from the PloSB 2010 paper. Likelihoods arising from these models are computed using the coalescent approximation and are directly comparable to likelihoods from the fit_coal_cst function, thus allowing to test support for equilibrium versus expanding diversity scenarios.
</p>
<p>These models can be fitted using the options specified below:
</p>

<ul>
<li><p> model 3: 
with cst.lamb=TRUE &amp; cst.mu=TRUE
</p>
</li>
<li><p> model 4a: 
with cst.lamb=FALSE &amp; cst.mu=TRUE
</p>
</li>
<li><p> model 4b: 
with cst.lamb=TRUE &amp; cst.mu=FALSE
</p>
</li>
<li><p> model 4c: 
with cst.lamb=FALSE, cst.mu=FALSE &amp; fix.eps=TRUE
</p>
</li>
<li><p> model 4d: 
with cst.lamb=FALSE, cst.mu=FALSE &amp; fix.eps=FALSE
</p>
</li>
<li><p> model 5: 
with cst.lamb=TRUE &amp; mu.0=TRUE
</p>
</li>
<li><p> model 6: 
with cst.lamb=FALSE &amp; mu.0=TRUE
</p>
</li></ul>

<p>Time runs from the present to the past. Hence, if alpha is estimated to be positive (for example), this means that the speciation rate decreases from past to present. 
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>the name of the fitted model</p>
</td></tr>
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>model.parameters</code></td>
<td>
<p>the estimated parameters</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>H Morlon</p>


<h3>References</h3>

<p>Morlon, H., Potts, M.D., Plotkin, J.B. (2010) Inferring the dynamics of diversification: a coalescent approach, PLoS B 8(9): e1000493
</p>
<p>Morlon, H., Kemps, B., Plotkin, J.B., Brisson, D. (2012) Explosive radiation of a bacterial species group, Evolution, 66: 2577-2586
</p>
<p>Morlon, H. (2014) Phylogenetic approaches for studying diversification, Eco Lett, 17:508-525
</p>


<h3>See Also</h3>

<p><code><a href="#topic+likelihood_coal_var">likelihood_coal_var</a></code>, <code><a href="#topic+fit_coal_cst">fit_coal_cst</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)

if(test){
result &lt;- fit_coal_var(Cetacea, lamb0=0.01, alpha=-0.001, mu0=0.0, beta=0, N0=89)
print(result)
}
</code></pre>

<hr>
<h2 id='fit_env'>
Maximum likelihood fit of the environmental birth-death model
</h2><span id='topic+fit_env'></span>

<h3>Description</h3>

<p>Fits the environmental birth-death model with potentially missing extant species to a phylogeny, by maximum likelihood. Notations follow Morlon et al. PNAS 2011 and Condamine et al. ELE 2013.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_env(phylo, env_data, tot_time, f.lamb, f.mu, lamb_par, mu_par, df= NULL, f = 1,
       meth = "Nelder-Mead", cst.lamb = FALSE, cst.mu = FALSE,
       expo.lamb = FALSE, expo.mu = FALSE, fix.mu = FALSE,
       dt=0, cond = "crown")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_env_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_env_data">env_data</code></td>
<td>

<p>environmental data, given as a data frame with two columns. The first column is time, the second column is the environmental data (temperature for instance).
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the hypothesized functional form of the variation of the speciation rate <code class="reqn">\lambda</code> with time and the environmental variable. Any functional form may be used.
This function has three arguments: the first argument is time; the second argument is the environmental variable; the third arguement is a numeric vector of the parameters controlling the time and environmental variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the hypothesized functional form of the variation of the extinction rate <code class="reqn">\mu</code> with time and the environmental variable. Any functional form may be used.
This function has three arguments: the first argument is time; the second argument is the environmental variable; the second argument is a numeric vector of the parameters controlling the time and environmental variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_lamb_par">lamb_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.lamb to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_mu_par">mu_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.mu to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model without extinction (for example), mu_par should be empty (vector of length 0). Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_df">df</code></td>
<td>

<p>the degree of freedom to use to define the spline. As a default, smooth.spline(env_data[,1], env_data[,2])$df is used. See <em>sm.spline</em> for details.  
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_f">f</code></td>
<td>

<p>the fraction of extant species included in the phylogeny
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_meth">meth</code></td>
<td>

<p>optimization to use to maximize the likelihood function, see <em>optim</em> for more details.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time or the environmental variable) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time or the environmental variable) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_expo.lamb">expo.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is an exponential function of time (and does not depend on the environmental variable) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_expo.mu">expo.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is an exponential function of time (and does not depend on the environmental variable) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_fix.mu">fix.mu</code></td>
<td>

<p>logical: if set to TRUE, the extinction rate <code class="reqn">\mu</code> is fixed and will not be optimized.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_dt">dt</code></td>
<td>

<p>the default value is 0. In this case, integrals in the likelihood are computed using R &quot;integrate&quot; function, which can be quite slow. If a positive dt is given as argument, integrals are computed using a piece-wise contant approximation, and dt represents the length of the intervals on which functions are assumed to be constant. We found that 1e-3 generally provides a good trade-off between precision and computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>
 <ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (use when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and survival of the 2 daugther lineages (use when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>The lengths of lamb_par and mu_par are used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong. In the f.lamb and f.mu functions, time runs from the present to the past.
Note that abs(f.lamb) and abs(f.mu) are used in the likelihood computation as speciation and extinction rates should always be positive. A consequence of this is that negative speciation/extinction rates estimates can be returned. They should be interpreted in aboslute terms. See Morlon et al. 2020 for a more detailed explanation.</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>the name of the fitted model</p>
</td></tr>
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>lamb_par</code></td>
<td>
<p>a numeric vector of estimated f.lamb parameters, in the same order as defined in f.lamb</p>
</td></tr>
<tr><td><code>mu_par</code></td>
<td>
<p>a numeric vector of estimated f.mu parameters, in the same order as defined in f.mu (if fix.mu is FALSE)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The speed of convergence of the fit might depend on the degree of freedom chosen to define the spline. 
</p>


<h3>Author(s)</h3>

<p>H Morlon and F Condamine</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Condamine, F.L., Rolland, J., and Morlon, H. (2013) Macroevolutionary perspectives to environmental change, Eco Lett 16: 72-85
</p>
<p>Morlon, H. (2014) Phylogenetic approaches for studying diversification, Eco Lett, 17:508-525
</p>
<p>Morlon, H., Rolland, J. and Condamine, F. (2020) Response to Technical Comment ‘A cautionary note for users of linear diversification dependencies’, Eco Lett
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_fit_env">plot_fit_env</a></code>, <code><a href="#topic+fit_bd">fit_bd</a></code>, <code><a href="#topic+likelihood_bd">likelihood_bd</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)
tot_time&lt;-max(node.age(Cetacea)$ages)
data(InfTemp)
dof&lt;-smooth.spline(InfTemp[,1], InfTemp[,2])$df

# Fits a model with lambda varying as an exponential function of temperature
# and mu fixed to 0 (no extinction).  Here t stands for time and x for temperature.
f.lamb &lt;-function(t,x,y){y[1] * exp(y[2] * x)}
f.mu&lt;-function(t,x,y){0}
lamb_par&lt;-c(0.10, 0.01)
mu_par&lt;-c()
#result_exp &lt;- fit_env(Cetacea,InfTemp,tot_time,f.lamb,f.mu,lamb_par,mu_par,
#                      f=87/89,fix.mu=TRUE,df=dof,dt=1e-3)
</code></pre>

<hr>
<h2 id='fit_env_in_past'>
Maximum likelihood fit of the environmental birth-death model excluding the recent past
</h2><span id='topic+fit_env_in_past'></span>

<h3>Description</h3>

<p>Fits the environmental birth-death model with potentially missing extant species to a phylogeny, by maximum likelihood while excluding the recent past. Notations follow Morlon et al. PNAS 2011 and Condamine et al. ELE 2013.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_env_in_past(phylo, env_data, tot_time, time_stop, f.lamb, f.mu, lamb_par, mu_par, 
       desc, tot_desc, df= NULL, meth = "Nelder-Mead", cst.lamb = FALSE, cst.mu = FALSE,
       expo.lamb = FALSE, expo.mu = FALSE, fix.mu = FALSE,
       dt=0, cond = "crown")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_env_in_past_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation) that does not include any recent speciation (i.e. no speciation events between time_stop and the present). 
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_env_data">env_data</code></td>
<td>

<p>environmental data, given as a data frame with two columns. The first column is time, the second column is the environmental data (temperature for instance).
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_time_stop">time_stop</code></td>
<td>

<p>the age of the phylogeny where to stop the birth-death process: it excludes the recent past (between the present and time_stop), while conditioning on the survival of the lineages from time_stop to the present.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the hypothesized functional form of the variation of the speciation rate <code class="reqn">\lambda</code> with time and the environmental variable. Any functional form may be used.
This function has three arguments: the first argument is time; the second argument is the environmental variable; the third arguement is a numeric vector of the parameters controlling the time and environmental variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the hypothesized functional form of the variation of the extinction rate <code class="reqn">\mu</code> with time and the environmental variable. Any functional form may be used.
This function has three arguments: the first argument is time; the second argument is the environmental variable; the second argument is a numeric vector of the parameters controlling the time and environmental variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_lamb_par">lamb_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.lamb to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_mu_par">mu_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.mu to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model without extinction (for example), mu_par should be empty (vector of length 0). Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_df">df</code></td>
<td>

<p>the degree of freedom to use to define the spline. As a default, smooth.spline(env_data[,1], env_data[,2])$df is used. See <em>sm.spline</em> for details.  
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_desc">desc</code></td>
<td>

<p>the number of  lineages present at present in the reconstructed phylogenetic tree.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_tot_desc">tot_desc</code></td>
<td>

<p>the total number of extant species (including in the unsampled ones). 
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_meth">meth</code></td>
<td>

<p>optimization to use to maximize the likelihood function, see <em>optim</em> for more details.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time or the environmental variable) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time or the environmental variable) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_expo.lamb">expo.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is an exponential function of time (and does not depend on the environmental variable) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_expo.mu">expo.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is an exponential function of time (and does not depend on the environmental variable) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_fix.mu">fix.mu</code></td>
<td>

<p>logical: if set to TRUE, the extinction rate <code class="reqn">\mu</code> is fixed and will not be optimized.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_dt">dt</code></td>
<td>

<p>the default value is 0. In this case, integrals in the likelihood are computed using R &quot;integrate&quot; function, which can be quite slow. If a positive dt is given as argument, integrals are computed using a piece-wise contant approximation, and dt represents the length of the intervals on which functions are assumed to be constant. We found that 1e-3 generally provides a good trade-off between precision and computation time.
</p>
</td></tr>
<tr><td><code id="fit_env_in_past_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>
 <ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (use when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and survival of the 2 daugther lineages (use when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>The lengths of lamb_par and mu_par are used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong. In the f.lamb and f.mu functions, time runs from the present to the past.
Note that abs(f.lamb) and abs(f.mu) are used in the likelihood computation as speciation and extinction rates should always be positive. A consequence of this is that negative speciation/extinction rates estimates can be returned. They should be interpreted in aboslute terms. See Morlon et al. 2020 for a more detailed explanation.</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>the name of the fitted model</p>
</td></tr>
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>lamb_par</code></td>
<td>
<p>a numeric vector of estimated f.lamb parameters, in the same order as defined in f.lamb</p>
</td></tr>
<tr><td><code>mu_par</code></td>
<td>
<p>a numeric vector of estimated f.mu parameters, in the same order as defined in f.mu (if fix.mu is FALSE)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The speed of convergence of the fit might depend on the degree of freedom chosen to define the spline. 
</p>


<h3>Author(s)</h3>

<p>H Morlon, F Condamine, E Lewitus, B Perez-Lamarque</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Condamine, F.L., Rolland, J., and Morlon, H. (2013) Macroevolutionary perspectives to environmental change, Eco Lett 16: 72-85
</p>
<p>Lewitus, E., Bittner, L., Malviya, S., Bowler, C., &amp; Morlon, H. (2018) Clade-specific diversification dynamics of marine diatoms since the Jurassic <em>Nature Ecology and Evolution</em>, 2(11), 1715–1723
</p>
<p>Perez-Lamarque, B., Öpik, M., Maliet, O., Afonso Silva, A., Selosse, M-A., Martos, F., Morlon, H. (2022) Analysing diversification dynamics using barcoding data: The case of an obligate mycorrhizal symbiont, <em>Molecular Ecology</em> 31: 3496–3512
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_fit_env">plot_fit_env</a></code>, <code><a href="#topic+fit_bd_in_past">fit_bd_in_past</a></code>, <code><a href="#topic+fit_env">fit_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ape)
library(phytools)
library(pspline)

data(Cetacea)
tot_time&lt;-max(node.age(Cetacea)$ages)
data(InfTemp)
dof&lt;-smooth.spline(InfTemp[,1], InfTemp[,2])$df

plot(Cetacea)
tot_time&lt;-max(node.age(Cetacea)$ages)

# slice the Cetaceae tree 5 Myr ago:
time_stop=5
sliced_tree &lt;- Cetacea
sliced_sub_trees &lt;- treeSlice(sliced_tree,slice = tot_time-time_stop, trivial=TRUE)
for (i in 1:length(sliced_sub_trees)){
  if (Ntip(sliced_sub_trees[[i]])&gt;1){
    sliced_tree &lt;- drop.tip(sliced_tree,
    tip=sliced_sub_trees[[i]]$tip.label[2:Ntip(sliced_sub_trees[[i]])])
  }
}

for (i in which(node.depth.edgelength(sliced_tree)&gt;(tot_time-time_stop))){
  temp = sliced_tree$edge.length[which(sliced_tree$edge[,2]==i)]-time_stop
  sliced_tree$edge.length[which(sliced_tree$edge[,2]==i)] &lt;- temp
}

Ntip(sliced_tree) # 52 lineages present 5 Myr have survived until today

# Now we can fit environment-dependent birth-death models excluding the 5 last Myr

# Fits a model with lambda varying as an exponential function of temperature
# and mu fixed to 0 (no extinction).  Here t stands for time and x for temperature.
f.lamb &lt;-function(t,x,y){y[1] * exp(y[2] * x)}
f.mu&lt;-function(t,x,y){0}
lamb_par&lt;-c(0.10, 0.01)
mu_par&lt;-c()

#result_env &lt;- fit_env_in_past(sliced_tree, InfTemp, tot_time, time_stop, f.lamb, 
#                             f.mu, lamb_par,mu_par,
#                             desc=Ntip(Cetacea), tot_desc=89, 
#                             fix.mu=TRUE,df=dof,dt=1e-3)

</code></pre>

<hr>
<h2 id='fit_t_comp'>
Fits models of trait evolution incorporating competitive interactions
</h2><span id='topic+fit_t_comp'></span>

<h3>Description</h3>

<p>Fits matching competition (MC), diversity dependent linear (DDlin), or diversity dependent exponential (DDexp) models of trait evolution to a given dataset and phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_t_comp(phylo, data, error=NULL, model=c("MC","DDexp","DDlin"), pars=NULL, 
		geography.object=NULL, regime.map=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_t_comp_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_t_comp_+3A_data">data</code></td>
<td>

<p>a named vector of trait values with names matching <code>phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="fit_t_comp_+3A_error">error</code></td>
<td>

<p>A named vector with standard errors (SE) of trait values for each species (with names matching <code>"phylo$tip.label"</code>). The default is NULL, in this case potential error is ignored in the fit. If set to NA, the SE is estimated from the data (to be used when there are no error measurements, a nuisance parameter is estimated). Note: When standard errors are provided, a nuisance parameter is also estimated.
</p>
</td></tr>
<tr><td><code id="fit_t_comp_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"MC"</code> is the matching competition model of Nuismer &amp; Harmon 2014, <code>"DDlin"</code> is the diversity-dependent linear model, and <code>"DDexp"</code> is the diversity-dependent exponential model of Weir &amp; Mursleen 2013.
</p>
</td></tr>
<tr><td><code id="fit_t_comp_+3A_pars">pars</code></td>
<td>

<p>vector specifying starting parameter values for maximum likelihood optimization. If unspecified, default values are used (see Details)
</p>
</td></tr>
<tr><td><code id="fit_t_comp_+3A_geography.object">geography.object</code></td>
<td>

<p>if incorporating biogeography, a list of sympatry through time created using <code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
</p>
</td></tr>
<tr><td><code id="fit_t_comp_+3A_regime.map">regime.map</code></td>
<td>

<p>if running two-regime versions of models, a stochastic map of the two regimes stored as a simmap object output from <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: if including known measurement error, the model fit incorporates this known error and, in addition, estimates an unknown, nuisance contribution to measurement error. The current implementation does not differentiate between the two, so, for instance, it is not possible to estimate the nuisance measurement error without providing the known, intraspecific error values.
</p>
<p>For single-regime fits without measurement error, <code>par</code> takes the default values of <code>var(data)/max(nodeHeights(phylo))</code> for sig2 and 0 for either <code>S</code> for the matching competition model, 
<code>b</code> for the linear diversity dependence model, or <code>r</code> for the exponential diversity dependence model. Values can be manually entered as a vector with the first element
equal to the desired starting value for sig2 and the second value equal to the desired starting value for either <code>S</code>, <code>b</code>, or <code>r</code>. Note: since likelihood optimization
uses sig rather than sig2, and since the starting value for is exponentiated to stabilize the likelihood search, if you input a <code>par</code> value, the first value specifying sig2 
should be the log(sqrt()) of the desired sig2 starting value.
</p>
<p>For two-regime fits without measurement error, the second and third values for <code>par</code> correspond to the first and second <code>S</code>, <code>b</code>, or <code>r</code> value (run trial fit to see which regime corresponds to each slope).
</p>
<p>For fits including measurement error, the default starting value for sig2 is <code>0.95*var(data)/max(nodeHeights(phylo))</code>, and nuisance values start at <code>0.05*var(data)/max(nodeHeights(phylo))</code>.
In all cases, the nuisance parameter is the last in the <code>par</code> vector, with the order of other variables as described above.
</p>
<p>For two-regime fits, particularly under the matching competition model, we recommend fitting with several different starting values.
</p>


<h3>Value</h3>

<p>a list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>LH</code></td>
<td>
<p>maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aic</code></td>
<td>
<p>Akaike Information Criterion value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>AIC value corrected for small sample size</p>
</td></tr>
<tr><td><code>free.parameters</code></td>
<td>
<p>number of free parameters from the model</p>
</td></tr>
<tr><td><code>sig2</code></td>
<td>
<p>maximum-likelihood estimate of <code>sig2</code> parameter</p>
</td></tr>
<tr><td><code>S</code></td>
<td>
<p>maximum-likelihood estimate of <code>S</code> parameter of matching competition model (see Note)</p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p>maximum-likelihood estimate of <code>b</code> parameter of linear diversity dependence model</p>
</td></tr>
<tr><td><code>r</code></td>
<td>
<p>maximum-likelihood estimate of <code>r</code> parameter of exponential diversity dependence model</p>
</td></tr>
<tr><td><code>z0</code></td>
<td>
<p>maximum-likelihood estimate of <code>z0</code>, the value at the root of the tree</p>
</td></tr>
<tr><td><code>nuisance</code></td>
<td>
<p>maximum-likelihood estimate of <code>nuisance</code>, the unknown, nuisance contribution to measurement error (see details)</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence diagnostics from <code><a href="stats.html#topic+optim">optim</a></code> function (see optim documentation)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>In current version, the <code>S</code> parameter is restricted to take on negative values in MC + geography ML optimization.
</p>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>
<p>Julien Clavel 
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J. Tobias, J., Rolland, J., Sheard, C., and Morlon, H. Tempo and mode of morphological evolution are decoupled from latitude in birds. <em>PLOS Biology</em> doi:10.1371/journal.pbio.3001270
</p>
<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. 
<em>Systematic Biology</em> 65:700-710
</p>
<p>Nuismer, S. &amp; Harmon, L. 2015. Predicting rates of interspecific interaction from phylogenetic trees. <em>Ecology Letters</em> 18:17-27.
</p>
<p>Weir, J. &amp; Mursleen, S. 2012. Diversity-dependent cladogenesis and trait evolution in the adaptive radiation of the auks (Aves: Alcidae). <em>Evolution</em> 67:403-416.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim_t_comp">sim_t_comp</a></code>
<code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
<code><a href="#topic+likelihood_t_MC">likelihood_t_MC</a></code>
<code><a href="#topic+likelihood_t_MC_geog">likelihood_t_MC_geog</a></code>
<code><a href="#topic+likelihood_t_DD">likelihood_t_DD</a></code>
<code><a href="#topic+likelihood_t_DD_geog">likelihood_t_DD_geog</a></code>
<code><a href="#topic+fit_t_comp_subgroup">fit_t_comp_subgroup</a></code>	
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Anolis.data)
geography.object&lt;-Anolis.data$geography.object
pPC1&lt;-Anolis.data$data
phylo&lt;-Anolis.data$phylo
regime.map&lt;-Anolis.data$regime.map


# Fit three models without biogeography to pPC1 data
MC.fit&lt;-fit_t_comp(phylo, pPC1, model="MC")
DDlin.fit&lt;-fit_t_comp(phylo, pPC1, model="DDlin")
DDexp.fit&lt;-fit_t_comp(phylo, pPC1, model="DDexp")

# Now fit models that incorporate biogeography, NOTE these models take longer to fit
MC.geo.fit&lt;-fit_t_comp(phylo, pPC1, model="MC", geography.object=geography.object)
DDlin.geo.fit&lt;-fit_t_comp(phylo, pPC1,model="DDlin", geography.object=geography.object)
DDexp.geo.fit&lt;-fit_t_comp(phylo, pPC1, model="DDexp", geography.object=geography.object)

# Now fit models that estimate parameters separately according to different 'regimes'
MC.two_regime.fit&lt;-fit_t_comp(phylo, pPC1, model="MC", regime.map=regime.map)
DDlin.two_regime.fit&lt;-fit_t_comp(phylo, pPC1,model="DDlin", regime.map=regime.map)
DDexp.two_regime.fit&lt;-fit_t_comp(phylo, pPC1, model="DDexp", regime.map=regime.map)

# Now fit models that estimate parameters separately according to different 'regimes', 
# including biogeography
MC.two_regime.geo.fit&lt;-fit_t_comp(phylo, pPC1, model="MC", 
  geography.object=geography.object, regime.map=regime.map)
DDlin.two_regime.geo.fit&lt;-fit_t_comp(phylo, pPC1,model="DDlin", 
  geography.object=geography.object, regime.map=regime.map)
DDexp.two_regime.geo.fit&lt;-fit_t_comp(phylo, pPC1, model="DDexp", 
  geography.object=geography.object, regime.map=regime.map)


</code></pre>

<hr>
<h2 id='fit_t_comp_subgroup'>
Fits models of trait evolution incorporating competitive interactions, restricting competition to occur only between members of a subgroup
</h2><span id='topic+fit_t_comp_subgroup'></span>

<h3>Description</h3>

<p>Fits matching competition (MC), diversity dependent linear (DDlin), or diversity dependent exponential (DDexp) models of trait evolution to a given dataset, phylogeny, and stochastic maps of both subgroup membership and biogeography.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
fit_t_comp_subgroup(full.phylo, data, subgroup, subgroup.map,
  model=c("MC","DDexp","DDlin"), ana.events=NULL, clado.events=NULL,
  stratified=FALSE, regime.map=NULL,error=NULL, par=NULL, 
  method="Nelder-Mead", bounds=NULL)
	
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_t_comp_subgroup_+3A_full.phylo">full.phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation) containing all of the tips used to estimate ancestral biogeography in BioGeoBEARS
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_data">data</code></td>
<td>

<p>a named vector of trait values for subgroup members with names matching <code>full.phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_subgroup">subgroup</code></td>
<td>

<p>subgroup whose members are competing
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_subgroup.map">subgroup.map</code></td>
<td>

<p>a phylo object created using <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code> in phytools that contains reconstructed subgroup membership
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"MC"</code> is the matching competition model of Nuismer &amp; Harmon 2014, <code>"DDlin"</code> is the diversity-dependent linear model, and <code>"DDexp"</code> is the diversity-dependent exponential model of Weir &amp; Mursleen 2013.
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_ana.events">ana.events</code></td>
<td>

<p>the &quot;ana.events&quot; table produced in BioGeoBEARS that lists anagenetic events in the stochastic map
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_clado.events">clado.events</code></td>
<td>

<p>the &quot;clado.events&quot; table produced in BioGeoBEARS that lists cladogenetic events in the stochastic map
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_stratified">stratified</code></td>
<td>

<p>logical indicating whether the stochastic map was built from a stratified analysis in BioGeoBEARS
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_regime.map">regime.map</code></td>
<td>

<p>a phylo object created using <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code> in phytools that contains reconstructed competitive regime membership (see Details)
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_error">error</code></td>
<td>

<p>A named vector with standard error (SE) for each species (with names matching <code>"phylo$tip.label"</code>). Default is NULL, if NA, then the SE is estimated from the data (a nuisance parameter for unknown errors). Note: When standard error are provided the nuisance parameter is also estimated.
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_par">par</code></td>
<td>

<p>vector specifying starting parameter values for maximum likelihood optimization. If unspecified, default values are used (see Details)
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_method">method</code></td>
<td>

<p>optimization algorithm to use (see <code><a href="stats.html#topic+optim">optim</a></code>; for DD models without biogeography, <code>method="BB"</code> is also supported, which uses <a href="BB.html#topic+spg">spg</a>)
</p>
</td></tr>
<tr><td><code id="fit_t_comp_subgroup_+3A_bounds">bounds</code></td>
<td>

<p>(optional) list of bounds to pass to optimization algorithm (see details at <code><a href="stats.html#topic+optim">optim</a></code>)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If unspecified, <code>par</code> takes the default values of <code>var(data)/max(nodeHeights(phylo))</code> for sig2 and 0 for either <code>S</code> for the matching competition model, 
<code>b</code> for the linear diversity dependence model, or <code>r</code> for the exponential diversity dependence model. Values can be manually entered as a vector with the first element
equal to the desired starting value for sig2 and the second value equal to the desired starting value for either <code>S</code>, <code>b</code>, or <code>r</code>. Note: since likelihood optimization
uses sig rather than sig2, and since the starting value for is exponentiated to stabilize the likelihood search, if you input a <code>par</code> value, the first value specifying sig2 
should be the log(sqrt()) of the desired sig2 starting value. We recommend running ML optimization with several different starting values to ensure convergence.
</p>
<p>Currently, this function can be used to implement the following models:
1. Subgroup pruning with biogeography: matching competition, diversity dependent
2. Subgroup pruning without biogeography: diversity dependent
3. Subgroup pruning without biogeography (two-regimes): diversity dependent (for more details, see <code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
</p>


<h3>Value</h3>

<p>a list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>LH</code></td>
<td>
<p>maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aic</code></td>
<td>
<p>Akaike Information Criterion value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>AIC value corrected for small sample size</p>
</td></tr>
<tr><td><code>free.parameters</code></td>
<td>
<p>number of free parameters from the model</p>
</td></tr>
<tr><td><code>sig2</code></td>
<td>
<p>maximum-likelihood estimate of <code>sig2</code> parameter</p>
</td></tr>
<tr><td><code>S</code></td>
<td>
<p>maximum-likelihood estimate of <code>S</code> parameter of matching competition model (see Note)</p>
</td></tr>
<tr><td><code>b</code></td>
<td>
<p>maximum-likelihood estimate of <code>b</code> parameter of linear diversity dependence model (see Note)</p>
</td></tr>
<tr><td><code>r</code></td>
<td>
<p>maximum-likelihood estimate of <code>r</code> parameter of exponential diversity dependence model (see Note)</p>
</td></tr>
<tr><td><code>z0</code></td>
<td>
<p>maximum-likelihood estimate of <code>z0</code>, the value at the root of the tree</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence diagnostics from <code><a href="stats.html#topic+optim">optim</a></code> function (see optim documentation)</p>
</td></tr>
<tr><td><code>nuisance</code></td>
<td>
<p>maximum-likelihood estimate of <code>nuisance</code>, the unknown, nuisance contribution to measurement error when <code>error</code> argument is used (that is NA or a vector provided by the user)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>In current version, the <code>S</code> parameter is restricted to take on negative values in MC + geography ML optimization.
</p>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J. Tobias, J., Rolland, J., Sheard, C., and Morlon, H. Tempo and mode of morphological evolution are decoupled from latitude in birds. <em>PLOS Biology</em> doi:10.1371/journal.pbio.3001270
</p>
<p>Drury, J., Tobias, J., Burns, K., Mason, N., Shultz, A., and Morlon, H. 2018. Contrasting impacts of competition on ecological and social trait evolution in songbirds. <em>PLOS Biology</em> 16(1): e2003563.
</p>
<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> 65: 700-710
</p>
<p>Nuismer, S. &amp; Harmon, L. 2015. Predicting rates of interspecific interaction from phylogenetic trees. <em>Ecology Letters</em> 18:17-27.
</p>
<p>Weir, J. &amp; Mursleen, S. 2012. Diversity-dependent cladogenesis and trait evolution in the adaptive radiation of the auks (Aves: Alcidae). <em>Evolution</em> 67:403-416.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+likelihood_subgroup_model">likelihood_subgroup_model</a></code>
<code><a href="#topic+CreateGeobyClassObject">CreateGeobyClassObject</a></code>
<code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>



data(BGB.examples)

#Prepare dataset with subgroups and biogeography

Canidae.phylo&lt;-BGB.examples$Canidae.phylo
dummy.group&lt;-c(rep("B",3),rep("A",12),rep("B",2),rep("A",6),rep("B",5),rep("A",6))
names(dummy.group)&lt;-Canidae.phylo$tip.label


Canidae.simmap&lt;-phytools::make.simmap(Canidae.phylo,dummy.group)

set.seed(123)
Canidae.data&lt;-rnorm(length(Canidae.phylo$tip.label))
names(Canidae.data)&lt;-Canidae.phylo$tip.label
Canidae.A&lt;-Canidae.data[which(dummy.group=="A")]


#Fit model with subgroup pruning and biogeography
MC.fit_subgroup_geo&lt;-fit_t_comp_subgroup(full.phylo=Canidae.phylo,
  ana.events=BGB.examples$Canidae.ana.events,
  clado.events=BGB.examples$Canidae.clado.events,
  stratified=FALSE,subgroup.map=Canidae.simmap, 
  data=Canidae.A,subgroup="A",model="MC")

DDexp.fit_subgroup_geo&lt;-fit_t_comp_subgroup(full.phylo=Canidae.phylo,
  ana.events=BGB.examples$Canidae.ana.events, 
  clado.events=BGB.examples$Canidae.clado.events,
  stratified=FALSE,subgroup.map=Canidae.simmap, 
  data=Canidae.A,subgroup="A",model="DDexp")

DDlin.fit_subgroup_geo&lt;-fit_t_comp_subgroup(full.phylo=Canidae.phylo,
  ana.events=BGB.examples$Canidae.ana.events, 
  clado.events=BGB.examples$Canidae.clado.events,
  stratified=FALSE,subgroup.map=Canidae.simmap, 
  data=Canidae.A,subgroup="A",model="DDlin")

#Fit model with subgroup pruning and no biogeography (for DD models only)
DDexp.fit_subgroup_no.geo&lt;-fit_t_comp_subgroup(full.phylo=Canidae.phylo,
  data=Canidae.A, subgroup="A", subgroup.map=Canidae.simmap,model="DDexp")

DDlin.fit_subgroup_no.geo&lt;-fit_t_comp_subgroup(full.phylo=Canidae.phylo,
  data=Canidae.A, subgroup="A", subgroup.map=Canidae.simmap,model="DDlin")


#Prepare regime map for fitting two-regime models with subgroup pruning (for DD models only)
regime&lt;-c(rep("regime1",15),rep("regime2",19))
names(regime)&lt;-Canidae.phylo$tip.label
regime.map&lt;-phytools::make.simmap(Canidae.phylo,regime)

#Fit model with subgroup pruning and two-regimes (for DD models only)
DDexp.fit_subgroup_two.regime&lt;-fit_t_comp_subgroup(full.phylo=Canidae.phylo,
  data=Canidae.A,subgroup="A", subgroup.map=Canidae.simmap,
  model="DDexp", regime.map=regime.map)

DDlin.fit_subgroup_two.regime&lt;-fit_t_comp_subgroup(full.phylo=Canidae.phylo,
  data=Canidae.A, subgroup="A", subgroup.map=Canidae.simmap,
  model="DDlin",regime.map=regime.map)

	


</code></pre>

<hr>
<h2 id='fit_t_env'>
Maximum likelihood fit of the environmental model of trait evolution
</h2><span id='topic+fit_t_env'></span>

<h3>Description</h3>

<p>Fits model of trait evolution for which evolutionary rates depends on an environmental function, or more generally a time varying function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
fit_t_env(phylo, data, env_data, error=NULL, model=c("EnvExp", "EnvLin"), 
          method="Nelder-Mead", control=list(maxit=20000), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_t_env_+3A_phylo">phylo</code></td>
<td>

<p>An object of class 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_t_env_+3A_data">data</code></td>
<td>

<p>A named vector of phenotypic trait values.
</p>
</td></tr>
<tr><td><code id="fit_t_env_+3A_env_data">env_data</code></td>
<td>

<p>Environmental data, given as a time continuous function (see, e.g. splinefun) or a data frame with two columns. The first column is time, the second column is the environmental data (temperature for instance).
</p>
</td></tr>
<tr><td><code id="fit_t_env_+3A_error">error</code></td>
<td>

<p>A named vector with standard errors (SE) of trait values for each species (with names matching <code>"phylo$tip.label"</code>). The default is NULL, in this case potential error is ignored in the fit. If set to NA, the SE is estimated from the data (to be used when there are no error measurements, a nuisance parameter is estimated). Note: When standard errors are provided, a nuisance parameter is also estimated.
</p>
</td></tr>
<tr><td><code id="fit_t_env_+3A_model">model</code></td>
<td>

<p>The model describing the functional form of variation of the evolutionary rate <code class="reqn">\sigma^2</code> with time and the environmental variable. Default models are &quot;EnvExp&quot; and &quot;EnvLin&quot; (see details). An user defined function of any functional form may be used (forward in time).
This function has three arguments: the first argument is time; the second argument is the environmental variable; the third argument is a numeric vector of the parameters controlling the time and environmental variation (to be estimated). See the example below.
</p>
</td></tr>
<tr><td><code id="fit_t_env_+3A_method">method</code></td>
<td>

<p>Methods used by the optimization routine (see ?optim for details).
</p>
</td></tr>
<tr><td><code id="fit_t_env_+3A_control">control</code></td>
<td>

<p>Max. bound for the number of iteration of the optimizer; other options can be fixed on the list (see ?optim). 
</p>
</td></tr>
<tr><td><code id="fit_t_env_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to the function. See details. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>fit_t_env</code> allows fitting environmental models of trait evolution. 
The default models <em>EnvExp</em> and <em>EnvLin</em> represents models for which the 
evolutionary rates are changing as a function of environmental changes though times as 
defined below.
</p>
<p><code>EnvExp</code>:
</p>
<p style="text-align: center;"><code class="reqn">\sigma^2 (t)  =  \sigma_0^2  e^{(\beta T(t))}</code>
</p>

<p><code>EnvLin</code>:
</p>
<p style="text-align: center;"><code class="reqn">\sigma^2 (t)  =  \sigma_0^2 + \beta T(t)</code>
</p>

<p>Users defined models should have the following form (see also examples below):
</p>
<p><code>fun &lt;- function(t, env, param){ param*env(t)}</code>
</p>
<p><em>t</em>: is the time parameter.
</p>
<p><em>env</em>: is a time function of an environmental variable. 
See for instance object created by <code>splinefun</code> when interpolating coordinate of points.
</p>
<p><em>param</em>: is a vector of parameters to estimate.
</p>
<p>For instance, the <code>EnvExp</code> function can be coded as: 
</p>
<p><code>fun &lt;- function(t, env, param){ param[1]*exp(param[2]*env(t))}</code> 
</p>
<p>where <code>param[1]</code> is the <code class="reqn">\sigma^2</code> parameter and <code>param[2]</code> is 
the <code class="reqn">\beta</code> parameter. 
Note that in this later case, two starting values should be provided in the <code>param</code> 
argument.
</p>
<p>e.g.:
</p>
<p><code>sigma=0.1</code>
</p>
<p><code>beta=0</code>
</p>
<p><code>fit_t_env(tree, data, env_data=InfTemp, model=fun, param=c(sigma,beta))</code>
</p>
<p>The various options are passed through &quot;...&quot;. 
</p>
<p>-param: The starting values used for the model. Must match the total number of parameters of the specified models. If &quot;error=NA&quot;, a starting value for the SE to be estimated must be provided with user-defined models.
</p>
<p>-scale: scale the amplitude of the environmental curve between 0 and 1. This may improve the parameters search in some situations.
</p>
<p>-df: the degree of freedom to use for defining the spline. As a default, smooth.spline(env_data[,1], env_data[,2])$df is used. See <em>sm.spline</em> for details.
</p>
<p>-upper: the upper bound for the parameter search when the &quot;L-BFGS-B&quot; method is used. See <em>optim</em> for details.
</p>
<p>-lower: the lower bound for the parameter search when the &quot;L-BFGS-B&quot; method is used. See <em>optim</em> for details.
</p>
<p>-sig2: can be used instead of param to define the starting sigma value only
</p>
<p>-beta: can be used instead of param to define the beta starting value only
</p>
<p>-maxdiff: difference in time between tips and present day for phylogenetic trees with no contemporaneous species (default is 0)
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aic</code></td>
<td>
<p>the Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike’s Information Criterion</p>
</td></tr>
<tr><td><code>free.parameters</code></td>
<td>
<p>the number of estimated parameters</p>
</td></tr>
<tr><td><code>param</code></td>
<td>
<p>a numeric vector of estimated parameters, sigma and beta respectively for the defaults models. In the same order as defined by the user if a     customized model is provided</p>
</td></tr>
<tr><td><code>root</code></td>
<td>
<p>the estimated root value</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence status of the optimizing function; &quot;0&quot; indicates convergence (See ?optim for details)</p>
</td></tr>
<tr><td><code>hess.value</code></td>
<td>
<p>reliability of the likelihood estimates calculated through the eigen-decomposition of the hessian matrix. &quot;0&quot; means that a reliable estimate has been reached</p>
</td></tr>
<tr><td><code>env_func</code></td>
<td>
<p>the environmental function</p>
</td></tr>
<tr><td><code>tot_time</code></td>
<td>
<p>the root age of the tree</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>the fitted model (default models or user specified)</p>
</td></tr>
<tr><td><code>nuisance</code></td>
<td>
<p>maximum-likelihood estimate of <code>nuisance</code>, the unknown, nuisance contribution to measurement error when <code>error</code> argument is used (i.e., NA or a vector provided by the user)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The users defined function is evaluated forward in time i.e.: from the root to the tips (time = 0 at the (present) tips).
The speed of convergence of the fit might depend on the degree of freedom chosen to define the spline. 
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Sciences, 114(16): 4183-4188.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.fit_t.env">plot.fit_t.env</a></code>, 
<code><a href="#topic+likelihood_t_env">likelihood_t_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(test){
data(Cetacea)
data(InfTemp)

# Simulate a trait with temperature dependence on the Cetacean tree
set.seed(123)

trait &lt;- sim_t_env(Cetacea, param=c(0.1,-0.2), env_data=InfTemp, model="EnvExp", 
					root.value=0, step=0.001, plot=TRUE)

## Fit the Environmental-exponential model
  # Fit the environmental model
  result1=fit_t_env(Cetacea, trait, env_data=InfTemp, scale=TRUE)
  plot(result1)

  # Add to the plot the results from different smoothing of the temperature curve
  result2=fit_t_env(Cetacea, trait, env_data=InfTemp, df=10, scale=TRUE)
  lines(result2, col="red")

  result3=fit_t_env(Cetacea, trait, env_data=InfTemp, df=50, scale=TRUE)
  lines(result3, col="blue")

## Fit the environmental linear model

  fit_t_env(Cetacea, trait, env_data=InfTemp, model="EnvLin", df=50, scale=TRUE)

## Fit user defined model (note that several other environmental variables 
## can be simultaneously encapsulated in this function through the env argument)

  # We define the function for the model
  my_fun&lt;-function(t, env_cont, param){ 
      param[1]*exp(param[2]*env_cont(t))
  }
  
  res&lt;-fit_t_env(Cetacea, trait, env_data=InfTemp, model=my_fun, 
                 param=c(0.1,0), scale=TRUE)
  # Retrieve the parameters and compare to 'result1'
  res
  plot(res, col="red")
	

## Fit user defined environmental function

if(require(pspline)){

  	 spline_result &lt;- sm.spline(x=InfTemp[,1],y=InfTemp[,2], df=50)
  	 env_func &lt;- function(t){predict(spline_result,t)}
  	 t&lt;-unique(InfTemp[,1])
  	
  # We build the interpolated smoothing spline function
  	 env_data&lt;-splinefun(t,env_func(t))
  
  # We then fit the model
  	 fit_t_env(Cetacea, trait, env_data=env_data)
 }
 
## Various parameterization (box constraints, df, scaling of the curve...) example
 fit_t_env(Cetacea, trait, env_data=InfTemp, model="EnvLin", method="L-BFGS-B", 
 			scale=TRUE, lower=-30, upper=20, df=10)

## A very general model...

# We define the function for the Early-Burst/AC model:
maxtime = max(branching.times(Cetacea))

# sigma^2*e^(r*t)
my_fun_ebac &lt;- function(t, env_cont, param){
    time = (maxtime - t)
    param[1]*exp(param[2]*time)
}

res&lt;-fit_t_env(Cetacea, trait, env_data=InfTemp, model=my_fun_ebac,
                param=c(0.1,0), scale=TRUE)
res # note that "r" is positive: it's the AC model (~OU model on ultrametric tree)

 }
</code></pre>

<hr>
<h2 id='fit_t_env_ou'>
Maximum likelihood fit of the OU environmental model of trait evolution
</h2><span id='topic+fit_t_env_ou'></span>

<h3>Description</h3>

<p>Fits Ornstein-Uhlenbeck (OU) model of trait evolution for which the optimum depends on an environmental function, or more generally a time varying function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
fit_t_env_ou(phylo, data, env_data, error=NULL, model,
          method="Nelder-Mead", control=list(maxit=20000), ...)
          
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_t_env_ou_+3A_phylo">phylo</code></td>
<td>

<p>An object of class 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_t_env_ou_+3A_data">data</code></td>
<td>

<p>A named vector of phenotypic trait values.
</p>
</td></tr>
<tr><td><code id="fit_t_env_ou_+3A_env_data">env_data</code></td>
<td>

<p>Environmental data, given as a time continuous function (see, e.g. splinefun) or a data frame with two columns. The first column is time, the second column is the environmental data (temperature for instance).
</p>
</td></tr>
<tr><td><code id="fit_t_env_ou_+3A_error">error</code></td>
<td>

<p>A named vector with standard errors (SE) of trait values for each species (with names matching <code>"phylo$tip.label"</code>). The default is NULL, in this case potential error is ignored in the fit. If set to NA, the SE is estimated from the data (to be used when there are no error measurements, a nuisance parameter is estimated). Note: When standard errors are provided, a nuisance parameter is also estimated.
</p>
</td></tr>
<tr><td><code id="fit_t_env_ou_+3A_model">model</code></td>
<td>

<p>A user defined model. If not provided, a default model is used (see details)
</p>
</td></tr>
<tr><td><code id="fit_t_env_ou_+3A_method">method</code></td>
<td>

<p>Methods used by the optimization routine (see ?optim for details).
</p>
</td></tr>
<tr><td><code id="fit_t_env_ou_+3A_control">control</code></td>
<td>

<p>Max. bound for the number of iteration of the optimizer; other options can be fixed on the list (see ?optim). 
</p>
</td></tr>
<tr><td><code id="fit_t_env_ou_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to the function. See details. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>fit_t_env_ou</code> allows fitting OU-environmental models of trait evolution (Troyer et al. 2020, Goswami &amp; Clavel 2024). Compared to model implemented in <code>fit_t_env</code> where the rate of phenotypic evolution evolves as a function of an environmental variable (Clavel &amp; Morlon 2020), here it's the optimum of a generalized Ornstein-Uhlenbeck (also called Hull-White model) that can changes as a function of an environmental variable T(t). More formally, the model is defined by the following process:
</p>
<p style="text-align: center;"><code class="reqn">dX(t)  =  \alpha  (\theta(t) -X(t))dt + \sigma dB(t)</code>
</p>

<p>Note that this model works only on NON-ULTRAMETRIC trees (e.g., with fossils)
</p>
<p>The default model has the optimum changing as a function of environmental changes though times as 
defined below:
</p>
<p style="text-align: center;"><code class="reqn">\theta (t)  =  \theta_0 + \beta T(t)</code>
</p>

<p>Users defined models should have the following form (see also examples below):
</p>
<p><code>fun &lt;- function(t, env, param, theta0){ theta0 + param*env(t)}</code>
</p>
<p><em>t</em>: is the time parameter.
</p>
<p><em>env</em>: is a time function of an environmental variable. 
See for instance object created by <code>splinefun</code> when interpolating coordinate of points.
</p>
<p><em>param</em>: is a vector of parameters to estimate.
</p>
<p><em>theta_0</em>: is the state at the root of the tree.
</p>
<p>For instance, the default model function can be coded as: 
</p>
<p><code>fun &lt;- function(t, env, param, theta0){ theta0 + param[1]*env(t)}</code> 
</p>
<p>where <code>param[1]</code> is the <code class="reqn">\beta</code> parameter. 
Note that in this case, one starting value should be provided in the <code>param</code> 
argument.
</p>
<p>e.g.:
</p>
<p><code>beta=0</code>
</p>
<p><code>fit_t_env(tree, data, env_data=InfTemp, model=fun, param=beta)</code>
</p>
<p>The various options are passed through &quot;...&quot;. 
</p>
<p>-param: The starting values used for the model. Must match the total number of parameters of the specified models. If &quot;error=NA&quot;, a starting value for the SE to be estimated must be provided with user-defined models.
</p>
<p>-scale: scale the amplitude of the environmental curve between 0 and 1. This may improve the parameters search in some situations.
</p>
<p>-df: the degree of freedom to use for defining the spline. As a default, smooth.spline(env_data[,1], env_data[,2])$df is used. See <em>sm.spline</em> for details.
</p>
<p>-upper: the upper bound for the parameter search when the &quot;L-BFGS-B&quot; method is used. See <em>optim</em> for details.
</p>
<p>-lower: the lower bound for the parameter search when the &quot;L-BFGS-B&quot; method is used. See <em>optim</em> for details.
</p>
<p>-maxdiff: difference in time between tips and present day for phylogenetic trees with no contemporaneous species (default is 0)
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>LH</code></td>
<td>
<p>the maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aic</code></td>
<td>
<p>the Akaike's Information Criterion</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>the second order Akaike’s Information Criterion</p>
</td></tr>
<tr><td><code>free.parameters</code></td>
<td>
<p>the number of estimated parameters</p>
</td></tr>
<tr><td><code>param</code></td>
<td>
<p>a numeric vector of estimated parameters, sigma and beta respectively for the defaults models. In the same order as defined by the user if a custom model is provided</p>
</td></tr>
<tr><td><code>root</code></td>
<td>
<p>the estimated root value</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence status of the optimizing function; &quot;0&quot; indicates convergence (See ?optim for details)</p>
</td></tr>
<tr><td><code>hess.value</code></td>
<td>
<p>reliability of the likelihood estimates calculated through the eigen-decomposition of the hessian matrix. &quot;0&quot; means that a reliable estimate has been reached</p>
</td></tr>
<tr><td><code>env_func</code></td>
<td>
<p>the environmental function</p>
</td></tr>
<tr><td><code>tot_time</code></td>
<td>
<p>the root age of the tree</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>the fitted model (default models or user specified)</p>
</td></tr>
<tr><td><code>nuisance</code></td>
<td>
<p>the estimated SE for species mean when &quot;error=NA&quot;</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The users defined function is evaluated forward in time i.e.: from the root to the tips (time = 0 at the (present) tips).
The speed of convergence of the fit might depend on the degree of freedom chosen to define the spline. 
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Science, 114(16): 4183-4188.
</p>
<p>Troyer, E., Betancur-R, R., Hughes, L., Westneat, M., Carnevale, G., White W.T., Pogonoski, J.J., Tyler, J.C., Baldwin, C.C., Orti, G., Brinkworth, A., Clavel, J., Arcila, D., 2022 - The impact of paleoclimatic changes on body size evolution in marine fishes. Proceedings of the National Academy of Sciences, 119 (29), e2122486119.
</p>
<p>Goswami, A. &amp; Clavel, J., 2024. Morphological evolution in a time of Phenomics. EcoEvoRxiv, https://doi.org/10.32942/X22G7Q
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.fit_t.env.ou">plot.fit_t.env.ou</a></code>,<code><a href="#topic+sim_t_env_ou">sim_t_env_ou</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(InfTemp)

# Simulate a trait with temperature dependence of the optimum on a simulated tree


set.seed(9999) # for reproducibility

# Let's start by simulating a trait under a climatic OU
beta = 0.6           # relationship to the climate curve
sim_theta = 4        # value of the optimum if the relationship to the climate curve is 0 
sim_sigma2 = 0.025   # variance of the scatter = sigma^2
sim_alpha = 0.36     # alpha value = strength of the OU; quite high here...
delta = 0.001        # time step used for the forward simulations =&gt; here its 1000y steps
tree &lt;- phytools::pbtree(n=200, d=0.3) # simulate a bd tree with some extinct lineages
root_age = 60        # height of the root (almost all the Cenozoic here)
tree$edge.length &lt;- root_age*tree$edge.length/max(phytools::nodeHeights(tree)) 
# here - for this contrived example - I scale the tree so that the root is at 60 Ma

trait &lt;- sim_t_env_ou(tree, sigma=sqrt(sim_sigma2), alpha=sim_alpha, theta0=sim_theta, 
                      param=beta, env_data=InfTemp, step=0.01, scale=TRUE, plot=TRUE)

## Fit the Environmental model (default)

result1 &lt;- fit_t_env_ou(phylo = tree, data = trait, env_data =InfTemp, 
                        method = "Nelder-Mead", df=50, scale=TRUE)
plot(result1)


## Fit user defined model (note that several other environmental variables 
## can be simultaneously encapsulated in this function through the env argument)

# We re-define the function for the OU model with linear trend to the climatic curve
# NOTE: the env(t) function should return the value at the root for t=0

my_fun&lt;-function(t, env, param, theta0){ 
    theta0 + param[1]*env(t)
}
  
# starting value for param[1]. Here we use an arbitrary value of 0.1
beta_guess = 0.1

# fit the model
result2 &lt;- fit_t_env_ou(phylo = tree, data = trait, env_data =InfTemp, 
                        model = my_fun, param = beta_guess,  
                        method = "Nelder-Mead", df=50, scale=TRUE)
                  
# Retrieve the parameters and compare to 'result1'
result2
lines(result2, col="red", lty=2)


## Fit user defined environmental function

require(pspline)
  	 spline_result &lt;- sm.spline(x=InfTemp[,1],y=InfTemp[,2], df=50)
  	 env_func &lt;- function(t){predict(spline_result,t)}
  	 t&lt;-unique(InfTemp[,1])
  	
  # We build the interpolated smoothing spline function (not scaled here)
  	 env_data&lt;-splinefun(t,env_func(t))
  
  # We then fit the model
  
result3 &lt;- fit_t_env_ou(phylo = tree, data = trait, env_data = env_data, 
                        model = my_fun, param = 0.01, method = "Nelder-Mead")

 
</code></pre>

<hr>
<h2 id='fit_t_pl'>
High-dimensional phylogenetic models of trait evolution
</h2><span id='topic+fit_t_pl'></span>

<h3>Description</h3>

<p>Fits high-dimensional model of trait evolution on trees through penalized likelihood. A phylogenetic Leave-One-Out Cross-Validated log-likelihood (LOOCV) is used to estimate model parameters.</p>


<h3>Usage</h3>

<pre><code class='language-R'>
fit_t_pl(Y, tree, model=c("BM", "OU", "EB", "lambda"),
		 method=c("RidgeAlt", "RidgeArch", "RidgeAltapprox", 
		 "LASSO", "LASSOapprox"), targM=c("null", "Variance", 
		 "unitVariance"), REML=TRUE, up=NULL, low=NULL, 
		 tol=NULL, starting=NULL, SE=NULL,
		 scale.height=TRUE, ...)
  
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_t_pl_+3A_y">Y</code></td>
<td>

<p>A matrix of phenotypic traits values (the variables are represented as columns)
</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_tree">tree</code></td>
<td>

<p>An object of class 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_model">model</code></td>
<td>

<p>The evolutionary model, &quot;BM&quot; is Brownian Motion, &quot;OU&quot; is Ornstein-Uhlenbeck, &quot;EB&quot; is Early Burst, and &quot;lambda&quot; is Pagel's lambda transformation.
</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_method">method</code></td>
<td>

<p>The penalty method. &quot;RidgeArch&quot;: Archetype (linear) Ridge penalty, &quot;RidgeAlt&quot;: Quadratic Ridge penalty, &quot;LASSO&quot;: Least Absolute Selection and Shrinkage Operator. &quot;RidgeAltapprox&quot; and &quot;LASSOapprox&quot; are fast approximations of the LOOCV for the Ridge quadratic and LASSO penalties
</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_targm">targM</code></td>
<td>

<p>The target matrix used for the Ridge regularizations. &quot;null&quot; is a null target, &quot;Variance&quot; for a diagonal unequal variance target, &quot;unitVariance&quot; for an equal diagonal target. Only works with &quot;RidgeArch&quot;,&quot;RidgeAlt&quot;, and &quot;RidgeAltapprox&quot; methods.
</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_reml">REML</code></td>
<td>

<p>Use REML (default) or ML for estimating the parameters.
</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_up">up</code></td>
<td>

<p>Upper bound for the parameter search of the evolutionary model (optional).
</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_low">low</code></td>
<td>

<p>Lower bound for the parameter search of the evolutionary model (optional).
</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_tol">tol</code></td>
<td>

<p>minimum value for the regularization parameter. Singularities can occur with a zero value in high-dimensional cases. (default is NULL)</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_starting">starting</code></td>
<td>

<p>Starting values for the parameter search (optional).</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_se">SE</code></td>
<td>

<p>Standard errors associated with values in Y. If TRUE, SE will be estimated.</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_scale.height">scale.height</code></td>
<td>

<p>Whether the tree should be scaled to unit length or not. (default is TRUE)</p>
</td></tr>
<tr><td><code id="fit_t_pl_+3A_...">...</code></td>
<td>

<p>Options to be passed through. (e.g., echo=FALSE to stop printing messages)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>fit_t_pl</code> allows fitting various multivariate evolutionary models to high-dimensional datasets (where the number of variables <em>p</em> is larger than <em>n</em>). Models estimates are more accurate than maximum likelihood methods. Models fit can be compared using the GIC criterion (see ?GIC). Details about the methods are described in <em>Clavel et al. (2019)</em>.
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>loocv</code></td>
<td>
<p>the (negative) cross-validated penalized likelihood</p>
</td></tr>
<tr><td><code>model.par</code></td>
<td>
<p>the evolutionary model parameter estimates</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>the regularization/tuning parameter of the penalized likelihood</p>
</td></tr>
<tr><td><code>corrstruct</code></td>
<td>
<p>a list with the tansformed variables and the phylogenetic tree with branch length stretched to the model estimated parameters</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>the evolutionary model</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>the penalization method</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>the number of traits</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>the number of species</p>
</td></tr>
<tr><td><code>targM</code></td>
<td>
<p>the target used for Ridge Penalization</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>a list with the estimated evolutionary covariance matrix and it's inverse</p>
</td></tr>
<tr><td><code>REML</code></td>
<td>
<p>logical indicating if the REML (TRUE) or ML (FALSE) method has been used</p>
</td></tr>
<tr><td><code>variables</code></td>
<td>
<p><code>Y</code> is the input dataset and <code>tree</code> is the input phylogenetic tree</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>the estimated standard error</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The LASSO is computationally intensive. Please wait! For highly-dimensional datasets you should favor the &quot;RidgeArch&quot; method to speed up the computations.
The Ridge penalties with &quot;null&quot; or &quot;unitVariance&quot; targets are rotation invariants.</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J., Aristide, L., Morlon, H., 2019. A Penalized Likelihood framework for high-dimensional phylogenetic comparative methods and an application to new-world monkeys brain evolution. Syst. Biol. 68: 93-116.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ancestral">ancestral</a></code>,
<code><a href="#topic+phyl.pca_pl">phyl.pca_pl</a></code>,  
<code><a href="#topic+GIC.fit_pl.rpanda">GIC.fit_pl.rpanda</a></code>, 
<code><a href="#topic+gic_criterion">gic_criterion</a></code>
<code><a href="mvMORPH.html#topic+mvgls">mvgls</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

require(mvMORPH)
set.seed(1)
n &lt;- 32 # number of species
p &lt;- 31 # number of traits

tree &lt;- pbtree(n=n) # phylogenetic tree
R &lt;- Posdef(p)      # a random symmetric matrix (covariance)

# simulate a dataset
Y &lt;- mvSIM(tree, model="BM1", nsim=1, param=list(sigma=R))

# fit the model
fit_t_pl(Y, tree, model="BM", method="RidgeAlt")

# try on rotated axis (using PCA)
trans &lt;- prcomp(Y, center=FALSE)
fit_t_pl(trans$x, tree, model="BM", method="RidgeAlt")

# Estimate the SE (similar to Pagel's lambda for BM). 
# Advised with empirical datasets
fit_t_pl(Y, tree, model="BM", method="RidgeAlt", SE=TRUE)

  
</code></pre>

<hr>
<h2 id='fit_t_standard'>
Fits standard models of trait evolution incorporating known and nuisance measurement error
</h2><span id='topic+fit_t_standard'></span>

<h3>Description</h3>

<p>Fits Brownian motion (BM), Ornstein-Uhlenbeck (OU), or early burst (EB) models of trait evolution to a given dataset and phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_t_standard(phylo, data, model=c("BM","OU","EB"), error=NULL, two.regime=FALSE, 
		method="Nelder-Mead", echo=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_t_standard_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation); if <code>two.regime=TRUE</code>, this must be a simmap object from <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code> with two regimes
</p>
</td></tr>
<tr><td><code id="fit_t_standard_+3A_data">data</code></td>
<td>

<p>a named vector of trait values with names matching <code>phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="fit_t_standard_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"BM"</code> is the Brownian motion model, <code>"OU"</code> is the Ornstein-Uhlenbeck model, and <code>"EB"</code> is the early burst model.
</p>
</td></tr>
<tr><td><code id="fit_t_standard_+3A_error">error</code></td>
<td>

<p>A named vector with standard errors (SE) of trait values for each species (with names matching <code>"phylo$tip.label"</code>). The default is NULL, in this case potential error is ignored in the fit. If set to NA, the SE is estimated from the data (to be used when there are no error measurements, a nuisance parameter is estimated). Note: When standard errors are provided, a nuisance parameter is also estimated.
</p>
</td></tr>
<tr><td><code id="fit_t_standard_+3A_two.regime">two.regime</code></td>
<td>

<p>if <code>TRUE</code>, fits a two-regime model
</p>
</td></tr>
<tr><td><code id="fit_t_standard_+3A_method">method</code></td>
<td>

<p>optimization method from <code>link{optim}</code>
</p>
</td></tr>
<tr><td><code id="fit_t_standard_+3A_echo">echo</code></td>
<td>

<p>prints information to console during fit
</p>
</td></tr>
<tr><td><code id="fit_t_standard_+3A_...">...</code></td>
<td>

<p>Optional arguments. e.g. &quot;upper=xx&quot;, &quot;lower=xx&quot; to specify bounds on the parameter search. &quot;fixedRoot=TRUE&quot; to use an OU model where the root state is assumed fixed (instead of sampled from the stationary distribution)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: if including known measurement error, the model fit incorporates this known error and, in addition, estimates an unknown, nuisance contribution to measurement error. The current implementation does not differentiate between the two, so, for instance, it is not possible to estimate the nuisance measurement error without providing the known, intraspecific error values.
</p>


<h3>Value</h3>

<p>a list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>LH</code></td>
<td>
<p>maximum log-likelihood value</p>
</td></tr>
<tr><td><code>aic</code></td>
<td>
<p>Akaike Information Criterion value</p>
</td></tr>
<tr><td><code>aicc</code></td>
<td>
<p>AIC value corrected for small sample size</p>
</td></tr>
<tr><td><code>free.parameters</code></td>
<td>
<p>number of free parameters from the model</p>
</td></tr>
<tr><td><code>sig2</code></td>
<td>
<p>maximum-likelihood estimate of <code>sig2</code> parameter</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>maximum-likelihood estimate of <code>alpha</code> parameter of OU model (see Note)</p>
</td></tr>
<tr><td><code>r</code></td>
<td>
<p>maximum-likelihood estimate of the slope parameter of early burst model</p>
</td></tr>
<tr><td><code>z0</code></td>
<td>
<p>maximum-likelihood estimate of <code>z0</code>, the value at the root of the tree</p>
</td></tr>
<tr><td><code>nuisance</code></td>
<td>
<p>maximum-likelihood estimate of <code>nuisance</code>, the unknown, nuisance contribution to measurement error (see details)</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence diagnostics from <code><a href="stats.html#topic+optim">optim</a></code> function (see optim documentation)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>
<p>Julien Clavel 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
<code><a href="#topic+sim_t_tworegime">sim_t_tworegime</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(test){
data(Cetacea_clades)
data&lt;-sim_t_tworegime(Cetacea_clades,pars=c(sig2=0.01,r1=-0.01,r2=-0.02),
	root.value=0,Nsegments=1000,model="EB")
error&lt;-rep(0.05,length(Cetacea_clades$tip.label))
names(error)&lt;-Cetacea_clades$tip.label

#Fit single-regime models
BM1.fit&lt;-fit_t_standard(Cetacea_clades,data,model="BM",error,two.regime=FALSE)
OU1.fit&lt;-fit_t_standard(Cetacea_clades,data,model="OU",error,two.regime=FALSE)
EB1.fit&lt;-fit_t_standard(Cetacea_clades,data,model="EB",error,two.regime=FALSE)

#Now fit models that incorporate biogeography, NOTE these models take longer to fit
BM2.fit&lt;-fit_t_standard(Cetacea_clades,data,model="BM",error,two.regime=TRUE)
OU2.fit&lt;-fit_t_standard(Cetacea_clades,data,model="OU",error,two.regime=TRUE)
EB2.fit&lt;-fit_t_standard(Cetacea_clades,data,model="EB",error,two.regime=TRUE)
  }


</code></pre>

<hr>
<h2 id='fitTipData'>
Maximum likelihood estimators of a model's parameters
</h2><span id='topic+fitTipData'></span>

<h3>Description</h3>

<p>Finds the maximum likelihood estimators of the parameters, returns the likelihood and the inferred parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitTipData(object, data, error, params0, GLSstyle, v)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitTipData_+3A_object">object</code></td>
<td>
<p>an object of class 'PhenotypicModel'.</p>
</td></tr>
<tr><td><code id="fitTipData_+3A_data">data</code></td>
<td>
<p>vector of tip trait data.</p>
</td></tr>
<tr><td><code id="fitTipData_+3A_error">error</code></td>
<td>
<p>vector of intraspecific (i.e., tip-level) standard error of the mean. Specify NULL if no error data are available</p>
</td></tr>
<tr><td><code id="fitTipData_+3A_params0">params0</code></td>
<td>
<p>vector of parameters used to initialize the optimization algorithm. Default value is NULL, in which case the optimization procedure starts with the vector 'params0' specified within the 'model' object. </p>
</td></tr>
<tr><td><code id="fitTipData_+3A_glsstyle">GLSstyle</code></td>
<td>
<p>boolean specifying the way the mean trait value at the root is estimated. Default value is FALSE in which case the mean at the root is considered as any other parameter. If TRUE, the mean value at the root is estimated with the GLS method, as explained, e.g. in Hansen 1997.</p>
</td></tr>
<tr><td><code id="fitTipData_+3A_v">v</code></td>
<td>
<p>boolean specifying the verbose mode. Default value : FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Warning : This function uses the standard R optimizer &quot;optim&quot;. It may not always converge well. Please double check the convergence by trying distinct parameter sets for the initialisation.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>value</code></td>
<td>
<p>A numerical value : the lowest -log( likelihood ) value found during the optimization procedure.</p>
</td></tr>
<tr><td><code>inferredParams</code></td>
<td>
<p>The maximum likelihood estimators of the model's parameters.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>An integer code specifying the convergence of the optim function. Please refer to the optim function help files.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>M Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Loading an example tree
newick &lt;- "((((A:1,B:0.5):2,(C:3,D:2.5):1):6,E:10.25):2,(F:6.5,G:8.25):3):1;"
tree &lt;- read.tree(text=newick)

#Creating the models
modelBM &lt;- createModel(tree, 'BM')

#Simulating tip traits under the model :
dataBM &lt;- simulateTipData(modelBM, c(0,0,0,1))

#Fitting the model to the data
fitTipData(modelBM, dataBM, v=TRUE)
</code></pre>

<hr>
<h2 id='fitTipData-methods'> ~~ Methods for Function <code>fitTipData</code>  ~~</h2><span id='topic+fitTipData-methods'></span><span id='topic+fitTipData+2CPhenotypicModel-method'></span>

<h3>Description</h3>

<p>~~ Methods for function <code>fitTipData</code>  ~~
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "PhenotypicModel")</code></dt><dd>
<p>This is the only method available for this function. Same behaviour for any PhenotypicModel.
</p>
</dd>
</dl>

<hr>
<h2 id='foraminifera'>Foraminifera diversity since the Jurassic</h2><span id='topic+foraminifera'></span>

<h3>Description</h3>

<p>Foraminifera fossil diversity since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(foraminifera)</code></pre>


<h3>Details</h3>

<p>Foraminifera fossil diversity since the Jurassic compiled from the Neptune Database (Lazarus, 1994) and Paleobiology Database (https://paleobiodb.org/). Diversity curves are estimated at the genus level using shareholder quorum subsampling (Alroy, 2010) at two-million-year bins. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>foraminifera</code></dt><dd><p>a numeric vector corresponding to the estimated foraminifera change at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Lazarus, D. (1994) Neptune: A marine micropaleontology database <em>Mathematical Geology 26:817–832</em>
</p>
<p>Alroy, J. (2010) Geographical, environmental and intrinsic biotic controls on Phanerozoic marine diversification: Controls on phanerozoic marine diversification <em>Palaeontology 53:1211–1235</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(foraminifera)
plot(foraminifera)
</code></pre>

<hr>
<h2 id='get.comb.shift'>
Combinations of shifts of diversification.
</h2><span id='topic+get.comb.shift'></span>

<h3>Description</h3>

<p>Provides all the combinations of nodes of a phylogeny where shifts of diversification can be tested.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  get.comb.shift(phylo, data, sampling.fractions,
                 clade.size = 5, Ncores = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.comb.shift_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="get.comb.shift_+3A_data">data</code></td>
<td>

<p>a data.frame containing a database of monophyletic groups for which potential shifts can be tested. This database should be based on taxonomy, ecology or traits and must contain a column named &quot;Species&quot; with species names as in phylo.
</p>
</td></tr>
<tr><td><code id="get.comb.shift_+3A_sampling.fractions">sampling.fractions</code></td>
<td>

<p>the output resulting from get.sampling.fractions.
</p>
</td></tr>
<tr><td><code id="get.comb.shift_+3A_clade.size">clade.size</code></td>
<td>

<p>numeric. Define the minimum number of species in a subgroup. Default is 5.
</p>
</td></tr>
<tr><td><code id="get.comb.shift_+3A_ncores">Ncores</code></td>
<td>

<p>numeric. Define the number of CPU cores to use for parallelizing the computation of combinations.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>clade.size argument should be the same value for the whole procedure (same that for get.sampling.fraction and shift.estimates).
</p>


<h3>Value</h3>

<p>a vector of character summaryzing the combination of shifts as a concatenation of node IDs separated by &quot;.&quot; or &quot;/&quot;. Node IDs at the left of &quot;/&quot; correspond to shifts at the origin of subclades (monophyletic and ultrametric subtrees) while node IDs at the right of &quot;/&quot; correspond to shifts at the origin of backbone(s) (pruned trees).
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get.sampling.fractions">get.sampling.fractions</a></code>, <code><a href="#topic+shift.estimates">shift.estimates</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
# loading data
data("Cetacea")
data("taxo_cetacea")

# no shifts tested at genus level
taxo_cetacea_no_genus &lt;- taxo_cetacea[names(taxo_cetacea) != "Genus"]

f_cetacea &lt;- get.sampling.fractions(phylo = Cetacea,
                                    data = taxo_cetacea_no_genus)

comb.shift_cetacea &lt;- get.comb.shift(phylo = Cetacea,
                                     data = taxo_cetacea_no_genus,
                                     sampling.fractions = f_cetacea,
                                     Ncores = 4)
  
</code></pre>

<hr>
<h2 id='get.sampling.fractions'>
Sampling fractions of subclades
</h2><span id='topic+get.sampling.fractions'></span>

<h3>Description</h3>

<p>Provides the sampling fractions of a phylogenetic tree from a complete database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  get.sampling.fractions(phylo, data, clade.size = 5, plot = FALSE,
                         lad = TRUE, text.cex = 1, pch.cex = 0.8, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.sampling.fractions_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="get.sampling.fractions_+3A_data">data</code></td>
<td>

<p>a data.frame containing a database of monophyletic groups for which potential shifts can be tested. This database should be based on taxonomy, ecology or traits and must contain a column named &quot;Species&quot; with species names as in phylo. 
</p>
</td></tr>
<tr><td><code id="get.sampling.fractions_+3A_clade.size">clade.size</code></td>
<td>

<p>numeric. Define the minimum number of species in a subgroup. Default is 5.
</p>
</td></tr>
<tr><td><code id="get.sampling.fractions_+3A_plot">plot</code></td>
<td>

<p>bolean. If TRUE, the tree is plotted and testable nodes are highlighted with red dots. Default is FALSE.
</p>
</td></tr>
<tr><td><code id="get.sampling.fractions_+3A_lad">lad</code></td>
<td>

<p>bolean. Define which way the tree should be represented if plot = T. If TRUE, the smallest clade is at the bottom plot. If FALSE, it is at the top of the plot. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="get.sampling.fractions_+3A_text.cex">text.cex</code></td>
<td>

<p>numeric. Defines the size of the text in legend.
</p>
</td></tr>
<tr><td><code id="get.sampling.fractions_+3A_pch.cex">pch.cex</code></td>
<td>

<p>numeric. Defines the size of the red points at the crown of subclades.
</p>
</td></tr>
<tr><td><code id="get.sampling.fractions_+3A_...">...</code></td>
<td>

<p>further arguments to be passed to plot or to plot.phylo.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All described species should be included to properly calculate sampling fractions. The example of Cetacea uses a taxonomic database but groups can be defined on geography or traits as soon as they are monophyletic. If the taxonomy of the studied group is difficult to establish (e.i. taxonomic uncertainty, etc.), a &quot;fake&quot; taxonomic database can be created with random species names (Gen1_sp1, Gen1_sp2, Gen2_sp1, etc.) to circumvent taxonomic difficulties. Note that sampling fractions of the backbones are calculated in the next step of the pipeline (function get.comb.shift()).
</p>


<h3>Value</h3>

<p>a data.frame with as many rows as nodes in the phylogeny with the following informations in columns:
</p>
<table role = "presentation">
<tr><td><code>nodes</code></td>
<td>
<p>the node IDs</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>the name of the subclade from data</p>
</td></tr>
<tr><td><code>f</code></td>
<td>
<p>the sampling fraction for this subclade</p>
</td></tr>
<tr><td><code>sp_in</code></td>
<td>
<p>the number of species included in the tree</p>
</td></tr>
<tr><td><code>sp_tt</code></td>
<td>
<p>the number of species described in the data</p>
</td></tr>
<tr><td><code>to_test</code></td>
<td>
<p>the node IDs for nodes that are testable according to clade.size</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get.comb.shift">get.comb.shift</a></code>, <code><a href="#topic+shift.estimates">shift.estimates</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># loading data
data("Cetacea")
data("taxo_cetacea")

# no shifts tested at genus level
taxo_cetacea_no_genus &lt;- taxo_cetacea[names(taxo_cetacea) != "Genus"]

# calculating sampling fractions with a plot
f_cetacea &lt;- get.sampling.fractions(phylo = Cetacea, lad = FALSE,
                                    data = taxo_cetacea_no_genus,
                                    plot = TRUE, cex = 0.3)
</code></pre>

<hr>
<h2 id='getDataLikelihood'>
Likelihood of tip trait values.
</h2><span id='topic+getDataLikelihood'></span>

<h3>Description</h3>

<p>Computes -log( likelihood ) of tip trait data under a given set of parameters, and for a specified model of trait evolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDataLikelihood(object, data, error, params, v)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDataLikelihood_+3A_object">object</code></td>
<td>
<p>an object of class 'PhenotypicModel'.</p>
</td></tr>
<tr><td><code id="getDataLikelihood_+3A_data">data</code></td>
<td>
<p>vector of tip trait data.</p>
</td></tr>
<tr><td><code id="getDataLikelihood_+3A_error">error</code></td>
<td>
<p>vector of intraspecific (i.e., tip-level) standard error of the mean. Specify NULL if no error data are available.</p>
</td></tr>
<tr><td><code id="getDataLikelihood_+3A_params">params</code></td>
<td>
<p>vector of parameters, given in the same order as in the 'model' object.</p>
</td></tr>
<tr><td><code id="getDataLikelihood_+3A_v">v</code></td>
<td>
<p>boolean specifying the verbose mode. Default value : FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical value : -log( likelihood ) of the model.
</p>


<h3>Author(s)</h3>

<p>M Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Loading an example tree
newick &lt;- "((((A:1,B:0.5):2,(C:3,D:2.5):1):6,E:10.25):2,(F:6.5,G:8.25):3):1;"
tree &lt;- read.tree(text=newick)

#Creating the models
modelBM &lt;- createModel(tree, 'BM')

#Simulating tip traits under the model :
dataBM &lt;- simulateTipData(modelBM, c(0,0,0,1))

#Likelihood of the data :
getDataLikelihood(modelBM, dataBM, error=NULL, c(0,0,0,1))
</code></pre>

<hr>
<h2 id='getDataLikelihood-methods'> ~~ Methods for Function <code>getDataLikelihood</code>  ~~</h2><span id='topic+getDataLikelihood-methods'></span><span id='topic+getDataLikelihood+2CPhenotypicModel-method'></span>

<h3>Description</h3>

<p>~~ Methods for function <code>getDataLikelihood</code>  ~~
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "PhenotypicModel")</code></dt><dd>
<p>This is the only method available for this function. Same behaviour for any PhenotypicModel.
</p>
</dd>
</dl>

<hr>
<h2 id='getMAPS_ClaDS'>
Gets the Maximum A Posteriori for each ClaDS parameter
</h2><span id='topic+getMAPS_ClaDS'></span>

<h3>Description</h3>

<p>Extract the MAPs (Maximum A Posteriori) for the marginal posterior distributions estimated with fit_ClaDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMAPS_ClaDS(sampler, burn = 1/2, thin = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMAPS_ClaDS_+3A_sampler">sampler</code></td>
<td>

<p>The output of a fit_ClaDS run.
</p>
</td></tr>
<tr><td><code id="getMAPS_ClaDS_+3A_burn">burn</code></td>
<td>

<p>Number of iterations to drop in the beginning of the chains.</p>
</td></tr>
<tr><td><code id="getMAPS_ClaDS_+3A_thin">thin</code></td>
<td>

<p>Thinning parameter, one iteration out of &quot;thin&quot; is kept to compute the MAPs.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector MAPS containing the MAPs for the marginal posterior distribution for each of the model's parameters.
</p>
<p>MAPS[1:4] are the estimated hyperparameters, with MAPS[1] the sigma parameter (new rates stochasticity), MAPS[2] the alpha parameter (new rates trend),  MAPS[3] the turnover rate epsilon, and MAPS[4] the initial speciation rate lambda_0. 
</p>
<p>MAPS[-(1:4)] are the estimated branch-specific speciation rates, given in the same order as the edges of the phylogeny on which the inference was performed.
</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_ClaDS">fit_ClaDS</a></code>, <code><a href="#topic+plot_ClaDS_chains">plot_ClaDS_chains</a></code>, <code><a href="#topic+getMAPS_ClaDS0">getMAPS_ClaDS0</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Caprimulgidae_ClaDS2")


if(test){
MAPS = getMAPS_ClaDS(Caprimulgidae_ClaDS2$sampler, thin = 1)

print(paste0("sigma = ", MAPS[1], " ; alpha = ", 
  MAPS[2], " ; epsilon = ", MAPS[3], " ; l_0 = ", MAPS[4] ))
plot_ClaDS_phylo(Caprimulgidae_ClaDS2$tree, MAPS[-(1:4)])
}
</code></pre>

<hr>
<h2 id='getMAPS_ClaDS0'>
Gets the Maximum A Posteriori for each ClaDS0 parameter
</h2><span id='topic+getMAPS_ClaDS0'></span>

<h3>Description</h3>

<p>Extract the MAPs (Maximum A Posteriori) for the marginal posterior distributions estimated with run_ClaDS0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMAPS_ClaDS0(phylo, sampler, burn=1/2, thin=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMAPS_ClaDS0_+3A_phylo">phylo</code></td>
<td>

<p>An object of class 'phylo'.
</p>
</td></tr>
<tr><td><code id="getMAPS_ClaDS0_+3A_sampler">sampler</code></td>
<td>

<p>The output of a run_ClaDS0 run.
</p>
</td></tr>
<tr><td><code id="getMAPS_ClaDS0_+3A_burn">burn</code></td>
<td>

<p>Number of iterations to drop in the beginning of the chains.</p>
</td></tr>
<tr><td><code id="getMAPS_ClaDS0_+3A_thin">thin</code></td>
<td>

<p>Thinning parameter, one iteration out of &quot;thin&quot; is kept to compute the MAPs.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector MAPS containing the MAPs for the marginal posterior distribution for each of the model's parameters.
</p>
<p>MAPS[1:3] are the estimated hyperparameters, with MAPS[1] the sigma parameter (new rates stochasticity), MAPS[2] the alpha parameter (new rates trend), and MAPS[3] the initial speciation rate lambda_0. 
</p>
<p>MAPS[-(1:3)] are the estimated branch-specific speciation rates, given in the same order as the <code>phylo$edges</code>.
</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_ClaDS0">fit_ClaDS0</a></code>, <code><a href="#topic+plot_ClaDS0_chains">plot_ClaDS0_chains</a></code>, <code><a href="#topic+getMAPS_ClaDS">getMAPS_ClaDS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)


if(test){
obj= sim_ClaDS( lambda_0=0.1,    
                mu_0=0.5,      
                sigma_lamb=0.7,         
                alpha_lamb=0.90,     
                condition="taxa",    
                taxa_stop = 20,    
                prune_extinct = TRUE)  

tree = obj$tree
speciation_rates = obj$lamb[obj$rates]
extinction_rates = obj$mu[obj$rates]
data("ClaDS0_example")

# extract the Maximum A Posteriori for each of the parameters
MAPS = getMAPS_ClaDS0(ClaDS0_example$tree, 
                      ClaDS0_example$Cl0_chains, 
                      thin = 10)

# plot the simulated (on the left) and inferred speciation rates (on the right)
# on the same color scale
plot_ClaDS_phylo(ClaDS0_example$tree, 
          ClaDS0_example$speciation_rates, 
          MAPS[-(1:3)])
}
</code></pre>

<hr>
<h2 id='getTipDistribution'>
Distribution of tip trait values.
</h2><span id='topic+getTipDistribution'></span>

<h3>Description</h3>

<p>Computes the mean and variance of the tip trait distribution under a specified model of trait evolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTipDistribution(object, params, v)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTipDistribution_+3A_object">object</code></td>
<td>

<p>an object of class 'PhenotypicModel'
</p>
</td></tr>
<tr><td><code id="getTipDistribution_+3A_params">params</code></td>
<td>

<p>vector of parameters, given in the same order as in the 'model' object.
</p>
</td></tr>
<tr><td><code id="getTipDistribution_+3A_v">v</code></td>
<td>

<p>boolean specifying the verbose mode. Default value : FALSE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>mean</code></td>
<td>
<p>Expectation vector of the tip trait distribution.</p>
</td></tr>
<tr><td><code>Sigma</code></td>
<td>
<p>Variance-covariance matrix of the tip trait distribution.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>M Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Loading an example tree
newick &lt;- "((((A:1,B:0.5):2,(C:3,D:2.5):1):6,E:10.25):2,(F:6.5,G:8.25):3):1;"
tree &lt;- read.tree(text=newick)

#Creating a BM model
modelBM &lt;- createModel(tree, 'BM')

#Tip trait distribution under the model :
getTipDistribution(modelBM, c(0,0,0,1))
</code></pre>

<hr>
<h2 id='getTipDistribution-methods'> Distribution of tip trait values. </h2><span id='topic+getTipDistribution-methods'></span><span id='topic+getTipDistribution+2CPhenotypicACDC-method'></span><span id='topic+getTipDistribution+2CPhenotypicADiag-method'></span><span id='topic+getTipDistribution+2CPhenotypicBM-method'></span><span id='topic+getTipDistribution+2CPhenotypicDD-method'></span><span id='topic+getTipDistribution+2CPhenotypicGMM-method'></span><span id='topic+getTipDistribution+2CPhenotypicModel-method'></span><span id='topic+getTipDistribution+2CPhenotypicOU-method'></span><span id='topic+getTipDistribution+2CPhenotypicPM-method'></span>

<h3>Description</h3>

<p>Computes the mean and variance of the tip trait distribution under a specified model of trait evolution.
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "PhenotypicModel")</code></dt><dd>
<p>In the most general case, this function computes the expectation vector and the variance-covariance matrix using a numerical integration procedure that may take time.
</p>
</dd>
<dt><code>signature(object = "PhenotypicACDC")</code></dt><dd>
<p>The function has been optimized for this subclass.
</p>
</dd>
<dt><code>signature(object = "PhenotypicADiag")</code></dt><dd>
<p>The function has been optimized for this subclass.
</p>
</dd>
<dt><code>signature(object = "PhenotypicBM")</code></dt><dd>
<p>The function has been optimized for this subclass.
</p>
</dd>
<dt><code>signature(object = "PhenotypicDD")</code></dt><dd>
<p>The function has been optimized for this subclass.
</p>
</dd>
<dt><code>signature(object = "PhenotypicGMM")</code></dt><dd>
<p>The function has been optimized for this subclass.
</p>
</dd>
<dt><code>signature(object = "PhenotypicOU")</code></dt><dd>
<p>The function has been optimized for this subclass.
</p>
</dd>
<dt><code>signature(object = "PhenotypicPM")</code></dt><dd>
<p>The function has been optimized for this subclass.
</p>
</dd>
</dl>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology
</p>

<hr>
<h2 id='gic_criterion'>
Generalized Information Criterion (GIC) to compare models fit by Maximum Likelihood (ML) or Penalized Likelihood (PL).
</h2><span id='topic+gic_criterion'></span>

<h3>Description</h3>

<p>The GIC allows comparing models fit by Maximum Likelihood (ML) or Penalized Likelihood (PL).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>

gic_criterion(Y, tree, model="BM", method=c("RidgeAlt", "RidgeArch", "LASSO", "ML", 
				"RidgeAltapprox", "LASSOapprox"), targM=c("null", 
				"Variance", "unitVariance"), param=NULL, 
				tuning=0, REML=TRUE, ...)
  
  
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gic_criterion_+3A_y">Y</code></td>
<td>

<p>A matrix of phenotypic traits values (the variables are represented as columns)
</p>
</td></tr>
<tr><td><code id="gic_criterion_+3A_tree">tree</code></td>
<td>

<p>An object of class 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="gic_criterion_+3A_model">model</code></td>
<td>

<p>The evolutionary model, &quot;BM&quot; is Brownian Motion, &quot;OU&quot; is Ornstein-Uhlenbeck, &quot;EB&quot; is Early Burst, and &quot;lambda&quot; is Pagel's lambda transformation.
</p>
</td></tr>
<tr><td><code id="gic_criterion_+3A_method">method</code></td>
<td>

<p>The penalty method. &quot;RidgeArch&quot;: Archetype (linear) Ridge penalty, &quot;RidgeAlt&quot;: Quadratic Ridge penalty, &quot;LASSO&quot;: Least Absolute Selection and Shrinkage Operator, &quot;ML&quot;: Maximum Likelihood.
</p>
</td></tr>
<tr><td><code id="gic_criterion_+3A_targm">targM</code></td>
<td>

<p>The target matrix used for the Ridge regularizations. &quot;null&quot; is a null target, &quot;Variance&quot; for a diagonal unequal variance target, &quot;unitVariance&quot; for an equal diagonal target. Only works with &quot;RidgeArch&quot;,&quot;RidgeAlt&quot; methods.
</p>
</td></tr>
<tr><td><code id="gic_criterion_+3A_param">param</code></td>
<td>

<p>Parameter for the evolutionary model (see &quot;model&quot; above).
</p>
</td></tr>
<tr><td><code id="gic_criterion_+3A_tuning">tuning</code></td>
<td>

<p>The tuning/regularization parameter.
</p>
</td></tr>
<tr><td><code id="gic_criterion_+3A_reml">REML</code></td>
<td>

<p>Use REML (default) or ML for estimating the parameters.
</p>
</td></tr>
<tr><td><code id="gic_criterion_+3A_...">...</code></td>
<td>

<p>Additional options. Not used yet.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>gic_criterion</code> allows comparing the fit of various models estimated by Penalized Likelihood (see ?fit_t_pl). Use the wrapper <code>GIC</code> instead for models fit with <code>fit_t_pl</code>.
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>LogLikelihood</code></td>
<td>
<p>the log-likelihood estimated for the model with estimated parameters</p>
</td></tr>
<tr><td><code>GIC</code></td>
<td>
<p>the GIC criterion</p>
</td></tr>
<tr><td><code>bias</code></td>
<td>
<p>the value of the bias term estimated to compute the GIC</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The tuning parameter is assumed to be zero when using the &quot;ML&quot; method.</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Konishi S., Kitagawa G. 1996. Generalised information criteria in model selection. Biometrika. 83:875-890.
</p>
<p>Clavel, J., Aristide, L., Morlon, H., 2019. A Penalized Likelihood framework for high-dimensional phylogenetic comparative methods and an application to new-world monkeys brain evolution. Syst. Biol. 68: 93-116.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GIC.fit_pl.rpanda">GIC.fit_pl.rpanda</a></code>, 
<code><a href="#topic+fit_t_pl">fit_t_pl</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(test){

if(require(mvMORPH)){
set.seed(123)
n &lt;- 32 # number of species
p &lt;- 2 # number of traits

tree &lt;- pbtree(n=n) # phylogenetic tree
R &lt;- Posdef(p)      # a random symmetric matrix (covariance)

# simulate a dataset
Y &lt;- mvSIM(tree, model="BM1", nsim=1, param=list(sigma=R))

# Compute the GIC for ML
gic_criterion(Y, tree, model="BM", method="ML", tuning=0) # ML

# Compare with PL?
#test &lt;- fit_t_pl(Y, tree, model="BM", method="RidgeAlt")
#GIC(test)
}

}
</code></pre>

<hr>
<h2 id='GIC.fit_pl.rpanda'>
Generalized Information Criterion (GIC) to compare models fit by Maximum Likelihood (ML) or Penalized Likelihood (PL).
</h2><span id='topic+GIC.fit_pl.rpanda'></span>

<h3>Description</h3>

<p>The GIC allows comparing models fit by Maximum Likelihood (ML) or Penalized Likelihood (PL).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'fit_pl.rpanda'
GIC(object, ...)
  
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GIC.fit_pl.rpanda_+3A_object">object</code></td>
<td>

<p>An object of class &quot;fit_pl.rpanda&quot;. See ?fit_t_pl</p>
</td></tr>
<tr><td><code id="GIC.fit_pl.rpanda_+3A_...">...</code></td>
<td>

<p>Options to be passed through.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>GIC</code> allows comparing the fit of various models estimated by Penalized Likelihood (see ?fit_t_pl). It's a wrapper to the <code>gic_criterion</code> function.
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>LogLikelihood</code></td>
<td>
<p>the log-likelihood estimated for the model with estimated parameters</p>
</td></tr>
<tr><td><code>GIC</code></td>
<td>
<p>the GIC criterion</p>
</td></tr>
<tr><td><code>bias</code></td>
<td>
<p>the value of the bias term estimated to compute the GIC</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Konishi S., Kitagawa G. 1996. Generalised information criteria in model selection. Biometrika. 83:875-890.
</p>
<p>Clavel, J., Aristide, L., Morlon, H., 2019. A Penalized Likelihood framework for high-dimensional phylogenetic comparative methods and an application to new-world monkeys brain evolution. Syst. Biol. 68: 93-116.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gic_criterion">gic_criterion</a></code>, 
<code><a href="#topic+fit_t_pl">fit_t_pl</a></code>,
<code><a href="mvMORPH.html#topic+mvgls">mvgls</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
      require(mvMORPH)
      set.seed(1)
      n &lt;- 32 # number of species
      p &lt;- 40 # number of traits
      
      tree &lt;- pbtree(n=n) # phylogenetic tree
      R &lt;- Posdef(p)      # a random symmetric matrix (covariance)
      # simulate a dataset
      Y &lt;- mvSIM(tree, model="BM1", nsim=1, param=list(sigma=R))
      
      fit1 &lt;- fit_t_pl(Y, tree, model="BM", method="RidgeAlt")
      fit2 &lt;- fit_t_pl(Y, tree, model="OU", method="RidgeAlt")
      
      GIC(fit1); GIC(fit2)
      
</code></pre>

<hr>
<h2 id='greenalgae'>Green algae diversity since the Jurassic</h2><span id='topic+greenalgae'></span>

<h3>Description</h3>

<p>Green algae fossil diversity since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(greenalgae)</code></pre>


<h3>Details</h3>

<p>Green algae fossil diversity since the Jurassic compiled from the Neptune Database (Lazarus, 1994) and Paleobiology Database (https://paleobiodb.org/). Diversity curves are estimated at the genus level using shareholder quorum subsampling (Alroy, 2010) at two-million-year bins. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>greenalgae</code></dt><dd><p>a numeric vector corresponding to the estimated green algae change at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Lazarus, D. (1994) Neptune: A marine micropaleontology database <em>Mathematical Geology 26:817–832</em>
</p>
<p>Alroy, J. (2010) Geographical, environmental and intrinsic biotic controls on Phanerozoic marine diversification: Controls on phanerozoic marine diversification <em>Palaeontology 53:1211–1235</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(greenalgae)
plot(greenalgae)
</code></pre>

<hr>
<h2 id='InfTemp'>Paleotemperature data across the Cenozoic</h2><span id='topic+InfTemp'></span>

<h3>Description</h3>

<p>Paleotemperature data across the Cenozoic inferred from delta O18 measurements</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(InfTemp)</code></pre>


<h3>Details</h3>

<p>Paleotemperature data inferred from delta 018 measurements using the equation of Epstein et al. (1953). The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>Age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>Temperature</code></dt><dd><p>a numeric vector corresponding to the inferred temperature at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Epstein, S., Buchsbaum, R., Lowenstam, H.A., Urey, H.C. (1953) Revised carbonate-water isotopic temperature scale <em>Geol. Soc. Am. Bull. 64: 1315-1326</em>
</p>
<p>Zachos, J.C., Dickens, G.R., Zeebe, R.E. (2008) An early Cenozoic perspective on greenhouse warming and carbon-cycle dynamics <em>Nature</em> 451: 279-283 
</p>
<p>Condamine, F.L., Rolland, J., Morlon, H. (2013) Macroevolutionary perspectives to environmental change <em>Eco Lett</em> 16: 72-85
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(InfTemp)
plot(InfTemp)
</code></pre>

<hr>
<h2 id='JSDt_cluster'>
Clustering on the Jensen-Shannon distance between phylogenetic trait data
</h2><span id='topic+JSDt_cluster'></span>

<h3>Description</h3>

<p>Computes the Jensen-Shannon distance metric between spectral density profiles of phylogenetic trait data and clusters on those distances. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JSDt_cluster(phylo,mat,plot=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="JSDt_cluster_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)</p>
</td></tr>
<tr><td><code id="JSDt_cluster_+3A_mat">mat</code></td>
<td>

<p>a matrix of trait data with one trait per column and rows aligned to phylo tips</p>
</td></tr>
<tr><td><code id="JSDt_cluster_+3A_plot">plot</code></td>
<td>

<p>plot hierarchical cluster in a new window</p>
</td></tr>	
</table>


<h3>Value</h3>

<p>plots a heatmap and hierarchical cluster with bootstrap support (&gt;0.9) and outputs results of the k-medoids clustering on the optimal number of clusters in the form of a list with the following components
</p>
<table role = "presentation">
<tr><td><code>clusters</code></td>
<td>
<p>a list with the following components: size, max_diss, av_diss, diameter, and separation</p>
</td></tr>
<tr><td><code>J-S matrix</code></td>
<td>
<p>a matrix providing the Jensen-Shannon distance values between pairs of phylogenetic trait data</p>
</td></tr>
<tr><td><code>cluster assignment</code></td>
<td>
<p>a table	that lists for each trait its cluster assignment and silhouete width</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>E Lewitus</p>


<h3>References</h3>

<p>Lewitus, E., Morlon, H. (2019) Characterizing and comparing phylogenetic trait data from their normalized Laplacian spectrum, bioRxiv doi: https://doi.org/10.1101/654087</p>


<h3>See Also</h3>

<p><code><a href="#topic+spectR_t">spectR_t</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Cetacea)
n&lt;-length(Cetacea$tip.label)
mat&lt;-replicate(20, rnorm(n)) 
colnames(mat)&lt;-1:dim(mat)[2]
JSDt_cluster(Cetacea,mat)

</code></pre>

<hr>
<h2 id='JSDtree'>
Jensen-Shannon distance between phylogenies
</h2><span id='topic+JSDtree'></span>

<h3>Description</h3>

<p>Computes the Jensen-Shannon distance metric between spectral density profiles of phylogenies. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JSDtree(phylo,meth=c("standard"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="JSDtree_+3A_phylo">phylo</code></td>
<td>

<p>a list of objects of type 'phylo' (see ape documentation)</p>
</td></tr>
<tr><td><code id="JSDtree_+3A_meth">meth</code></td>
<td>

<p>the method used to compute the spectral density, which can either be &quot;standard&quot;, &quot;normal1&quot;, or &quot;normal2&quot;. if set to &quot;normal1&quot;, computes the spectral density normalized to the degree matrix. if set to &quot;normal2&quot;, computes the spectral density normalized to the number of eigenvalues. if set to &quot;standard&quot;, computes the unnormalized version of the spectral density (see the associated paper for an explanation)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix providing the Jensen-Shannon distance values between phylogeny pairs
</p>


<h3>Author(s)</h3>

<p>E Lewitus</p>


<h3>References</h3>

<p>Lewitus, E., Morlon, H., Characterizing and comparing phylogenies from their Laplacian spectrum, bioRxiv doi: http://dx.doi.org/10.1101/026476</p>


<h3>See Also</h3>

<p><code><a href="#topic+JSDtree_cluster">JSDtree_cluster</a></code>, <code><a href="#topic+spectR">spectR</a></code>, <code><a href="#topic+BICompare">BICompare</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trees&lt;-TESS::tess.sim.age(n=20,age=10,0.15,0.05,MRCA=TRUE)
JSDtree(trees)
</code></pre>

<hr>
<h2 id='JSDtree_cluster'>
Clustering of phylogenies
</h2><span id='topic+JSDtree_cluster'></span>

<h3>Description</h3>

<p>Clusters phylogenies using hierarchical and k-medoids clustering</p>


<h3>Usage</h3>

<pre><code class='language-R'>JSDtree_cluster(JSDtree,alpha=0.9,draw=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="JSDtree_cluster_+3A_jsdtree">JSDtree</code></td>
<td>

<p>a matrix of distances between phylogenie pairs, typically the output of the JSDtree function when the distance is measured as the Jensen-Shannon distance 
</p>
</td></tr>
<tr><td><code id="JSDtree_cluster_+3A_alpha">alpha</code></td>
<td>

<p>the confidence value for demarcating clusters in the hierarchical clustering plot; the default is 0.9
</p>
</td></tr>
<tr><td><code id="JSDtree_cluster_+3A_draw">draw</code></td>
<td>

<p>plot heatmap and hierarchical cluster in new windows
</p>
</td></tr>	
</table>


<h3>Value</h3>

<p>plots a heatmap and a hierarchical cluster with bootstrap support, and outputs results of the k-medoids clustering in the form of a list with the following components
</p>
<table role = "presentation">
<tr><td><code>clusters</code></td>
<td>
<p>the optimal number of clusters around medoids (see pamk documentation)</p>
</td></tr>
<tr><td><code>cluster_assignments</code></td>
<td>
<p>assignments of trees to clusters</p>
</td></tr> 
<tr><td><code>cluster_support</code></td>
<td>
<p>a list with the following components: widths: a table specifying the cluster to which each tree belongs, the neighbor (i.e. most similar) cluster, and the silhouette width of the observation (see silhouette documentation); clus.avg.widths: average silhouette width for each cluster; vg.width: average silhouette width across all clusters</p>
</td></tr> 
</table>


<h3>Note</h3>

<p>The k-medoids clustering may not work with fewer than 10 trees</p>


<h3>Author(s)</h3>

<p>E Lewitus</p>


<h3>References</h3>

<p>Lewitus, E., Morlon, H., Characterizing and comparing phylogenies from their Laplacian spectrum, bioRxiv doi: http://dx.doi.org/10.1101/026476</p>


<h3>See Also</h3>

<p><code><a href="#topic+JSDtree">JSDtree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
trees&lt;-TESS::tess.sim.age(n=20,age=10,0.15,0.05,MRCA=TRUE)
res&lt;-JSDtree(trees)
JSDtree_cluster(res,alpha=0.9,draw=TRUE)

</code></pre>

<hr>
<h2 id='landplant'>Land plant diversity since the Jurassic</h2><span id='topic+landplant'></span>

<h3>Description</h3>

<p>Land plant fossil diversity since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(landplant)</code></pre>


<h3>Details</h3>

<p>Land plant fossil diversity since the Jurassic compiled from the Neptune Database (Lazarus, 1994) and Paleobiology Database (https://paleobiodb.org/). Diversity curves are estimated at the genus level using shareholder quorum subsampling (Alroy, 2010) at two-million-year bins. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>landplant</code></dt><dd><p>a numeric vector corresponding to the estimated land plant change at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Lazarus, D. (1994) Neptune: A marine micropaleontology database <em>Mathematical Geology 26:817–832</em>
</p>
<p>Alroy, J. (2010) Geographical, environmental and intrinsic biotic controls on Phanerozoic marine diversification: Controls on phanerozoic marine diversification <em>Palaeontology 53:1211–1235</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(landplant)
plot(landplant)
</code></pre>

<hr>
<h2 id='likelihood_bd'>
Likelihood of a phylogeny under the general birth-death model
</h2><span id='topic+likelihood_bd'></span>

<h3>Description</h3>

<p>Computes the likelihood of a phylogeny under a birth-death model with potentially time-varying rates and potentially missing extant species. Notations follow Morlon et al. PNAS 2011.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_bd(phylo, tot_time, f.lamb, f.mu, f, cst.lamb = FALSE, cst.mu = FALSE,
              expo.lamb = FALSE, expo.mu = FALSE, dt=0, cond = "crown")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_bd_+3A_phylo">phylo</code></td>
<td>
  
<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the time-variation of the speciation rate <code class="reqn">\lambda</code>. This function as a single argument (time). Any function may be used.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the time-variation of the speciation rate <code class="reqn">\mu</code>. This function as a single argument (time). Any function may be used.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_f">f</code></td>
<td>

<p>the fraction of extant species included in the phylogeny
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_expo.lamb">expo.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_expo.mu">expo.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_dt">dt</code></td>
<td>

<p>the default value is 0. In this case, integrals in the likelihood are computed using R &quot;integrate&quot; function, which can be quite slow. If a positive dt is given as argument, integrals are computed using a piece-wise contant approximation, and dt represents the length of the intervals on which functions are assumed to be constant. For an exponential dependency of the speciation rate with time, we found that dt=1e-3 gives a good trade-off between precision and computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>
 <ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (use when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and survival of the 2 daugther lineages (use when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>When specifying f.lamb and f.mu, time runs from the present to the past (hence if the speciation rate decreases with time, f.lamb must be a positive function of time).
</p>


<h3>Value</h3>

<p>the loglikelihood value of the phylogeny, given f.lamb and f.mu
</p>


<h3>Author(s)</h3>

<p>H Morlon</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)
tot_time &lt;- max(node.age(Cetacea)$ages)
# Compute the likelihood for a pure birth model (no extinction) with
# an exponential variation of speciation rate with time
lamb_par &lt;- c(0.1, 0.01)
f.lamb &lt;- function(t){lamb_par[1] * exp(lamb_par[2] * t)}
f.mu &lt;- function(t){0}
f &lt;- 87/89
lh &lt;- likelihood_bd(Cetacea,tot_time,f.lamb,f.mu,f,cst.mu=TRUE,expo.lamb=TRUE, dt=1e-3)
</code></pre>

<hr>
<h2 id='likelihood_bd_backbone'>
Likelihood of a phylogeny under the general birth-death model (backbone)
</h2><span id='topic+likelihood_bd_backbone'></span>

<h3>Description</h3>

<p>Computes the likelihood of a phylogeny under a birth-death model with potentially time-varying rates and potentially missing extant species. Notations follow Morlon et al. PNAS 2011. Modified version of likelihood_bd for backbones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_bd_backbone(phylo, tot_time, f, f.lamb, f.mu, 
                       backbone, spec_times, branch_times,
                       cst.lamb = FALSE, cst.mu = FALSE,
                       expo.lamb = FALSE, expo.mu = FALSE, dt=0, cond = "crown")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_bd_backbone_+3A_phylo">phylo</code></td>
<td>
  
<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the time-variation of the speciation rate <code class="reqn">\lambda</code>. This function as a single argument (time). Any function may be used.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the time-variation of the speciation rate <code class="reqn">\mu</code>. This function as a single argument (time). Any function may be used.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_f">f</code></td>
<td>

<p>the fraction of extant species included in the phylogeny
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_backbone">backbone</code></td>
<td>

<p>character. Allows to analyse a backbone. Default is NULL and spec_times and branch_times are then ignored.
</p>
<p>Otherwise:
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: for every shift, the probability of the speciation event at the stem age of the subclade is included in the likelihood of the backbone thanks to the argument spec_times.
</p>
</li>
<li><p>&quot;crown.shift&quot;: for every shift, both the probability of the speciation event at the stem age of the subclade and the probability that the stem of the subclade survives to the crown age are included in the likelihood of the backbone thanks to the argument branch_times.
</p>
</li></ul>

</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_spec_times">spec_times</code></td>
<td>

<p>a numeric vector of the stem ages of subclades. Used only if backbone = &quot;stem.shift&quot;. Default is NULL.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_branch_times">branch_times</code></td>
<td>

<p>a list of numeric vectors. Each vector contains the stem and crown ages of subclades (in this order). Used only if backbone = &quot;crown.shift&quot;. Default is NULL.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_cst.lamb">cst.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_cst.mu">cst.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is constant (i.e. does not depend on time) to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_expo.lamb">expo.lamb</code></td>
<td>

<p>logical: should be set to TRUE only if f.lamb is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_expo.mu">expo.mu</code></td>
<td>

<p>logical: should be set to TRUE only if f.mu is exponential to use analytical instead of numerical computation in order to reduce computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_dt">dt</code></td>
<td>

<p>the default value is 0. In this case, integrals in the likelihood are computed using R &quot;integrate&quot; function, which can be quite slow. If a positive dt is given as argument, integrals are computed using a piece-wise contant approximation, and dt represents the length of the intervals on which functions are assumed to be constant. For an exponential dependency of the speciation rate with time, we found that dt=1e-3 gives a good trade-off between precision and computation time.
</p>
</td></tr>
<tr><td><code id="likelihood_bd_backbone_+3A_cond">cond</code></td>
<td>

<p>conditioning to use to fit the model:
</p>
 <ul>
<li><p> FALSE: no conditioning (not recommended);
</p>
</li>
<li><p> &quot;stem&quot;: conditioning on the survival of the stem lineage (use when the stem age is known, in this case tot_time should be the stem age);
</p>
</li>
<li><p> &quot;crown&quot; (default): conditioning on a speciation event at the crown age and survival of the 2 daugther lineages (use when the stem age is not known, in this case tot_time should be the crown age).
</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>When specifying f.lamb and f.mu, time runs from the present to the past (hence if the speciation rate decreases with time, f.lamb must be a positive function of time).
</p>


<h3>Value</h3>

<p>the loglikelihood value of the phylogeny, given f.lamb and f.mu
</p>


<h3>Author(s)</h3>

<p>Hélène Morlon, Nathan Mazet</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)
tot_time &lt;- max(node.age(Cetacea)$ages)
# Compute the likelihood for a pure birth model (no extinction) with
# an exponential variation of speciation rate with time
lamb_par &lt;- c(0.1, 0.01)
f.lamb &lt;- function(t){lamb_par[1] * exp(lamb_par[2] * t)}
f.mu &lt;- function(t){0}
f &lt;- 87/89
# same as likelihood_bd in this case
lh &lt;- likelihood_bd_backbone(Cetacea, tot_time, f, f.lamb, f.mu, 
                             backbone = FALSE, spec_times = NULL, branch_times = NULL,
                             cst.mu = TRUE, expo.lamb = TRUE, dt = 1e-3)
</code></pre>

<hr>
<h2 id='likelihood_coal_cst'>
Likelihood of a phylogeny under the equilibrium diversity model
</h2><span id='topic+likelihood_coal_cst'></span>

<h3>Description</h3>

<p>Computes the likelihood of a phylogeny under the equilibrium diversity model with potentially time-varying rates and potentially missing extant species. Notations follow Morlon et al. PloSB 2010.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_coal_cst(Vtimes, ntips, tau0, gamma, N0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_coal_cst_+3A_vtimes">Vtimes</code></td>
<td>

<p>a vector of branching times (sorted from present to past)
</p>
</td></tr>
<tr><td><code id="likelihood_coal_cst_+3A_ntips">ntips</code></td>
<td>

<p>the number of tips in the phylogeny
</p>
</td></tr>
<tr><td><code id="likelihood_coal_cst_+3A_tau0">tau0</code></td>
<td>

<p>the turnover rate at present
</p>
</td></tr>
<tr><td><code id="likelihood_coal_cst_+3A_gamma">gamma</code></td>
<td>

<p>the parameter controlling the exponential variation in turnover rate. With gamma=0, the turnover rate is constant over time.
</p>
</td></tr>
<tr><td><code id="likelihood_coal_cst_+3A_n0">N0</code></td>
<td>

<p>the number of extant species
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Time runs from the present to the past. Hence, a positive gamma (for example) means that the turnover rate declines from past to present.
</p>


<h3>Value</h3>

<p>a list containing the following components:
</p>
<table role = "presentation">
<tr><td><code>res</code></td>
<td>
<p>the loglikelihood value of the phylogeny, given tau0 and gamma</p>
</td></tr>
<tr><td><code>all</code></td>
<td>
<p>vector of all the individual loglikelihood values corresponding to each branching event</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>H Morlon</p>


<h3>References</h3>

<p>Morlon, H., Potts, M.D., Plotkin, J.B. (2010) Inferring the dynamics of diversification: a coalescent approach, PLoS B 8(9): e1000493
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)
Vtimes &lt;- sort(branching.times(Cetacea))
tau0 &lt;- 0.1
gamma &lt;- 0.001
ntips &lt;- Ntip(Cetacea)
N0 &lt;- 89
likelihood &lt;- likelihood_coal_cst(Vtimes,ntips,tau0,gamma,N0)
</code></pre>

<hr>
<h2 id='likelihood_coal_var'>
Likelihood of a birth-death model using a coalescent approch
</h2><span id='topic+likelihood_coal_var'></span>

<h3>Description</h3>

<p>Computes the likelihood of a phylogeny under the expanding diversity model with potentially time-varying rates and potentially missing extant species to a phylogeny. Notations follow Morlon et al. PloSB 2010.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_coal_var(Vtimes, ntips, lamb0, alpha, mu0, beta, N0, pos = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_coal_var_+3A_vtimes">Vtimes</code></td>
<td>

<p>a vector of branching times (sorted from present to past)
</p>
</td></tr>
<tr><td><code id="likelihood_coal_var_+3A_ntips">ntips</code></td>
<td>

<p>number of species in the phylogeny
</p>
</td></tr>
<tr><td><code id="likelihood_coal_var_+3A_lamb0">lamb0</code></td>
<td>

<p>the speciation rate at present
</p>
</td></tr>
<tr><td><code id="likelihood_coal_var_+3A_alpha">alpha</code></td>
<td>

<p>the parameter controlling the exponential variation in speciation rate.  
</p>
</td></tr>
<tr><td><code id="likelihood_coal_var_+3A_mu0">mu0</code></td>
<td>

<p>the extinction rate at present
</p>
</td></tr>
<tr><td><code id="likelihood_coal_var_+3A_beta">beta</code></td>
<td>

<p>the parameter controlling the exponential variation in extinction rate.  
</p>
</td></tr>
<tr><td><code id="likelihood_coal_var_+3A_n0">N0</code></td>
<td>

<p>the number of extanct species
</p>
</td></tr>
<tr><td><code id="likelihood_coal_var_+3A_pos">pos</code></td>
<td>

<p>logical: should be set to FALSE only to not enforce positive speciation and extinction ratess
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Time runs from the present to the past. Hence, a positive alpha (for example) means that the speciation rate declines from past to present.
</p>


<h3>Value</h3>

<p>a list containing the following components:
</p>
<table role = "presentation">
<tr><td><code>res</code></td>
<td>
<p>the loglikelihood value of the phylogeny, given the parameters</p>
</td></tr>
<tr><td><code>all</code></td>
<td>
<p>vector of all the individual loglikelihood values corresponding to each branching event</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>H Morlon</p>


<h3>References</h3>

<p>Morlon, H., Potts, M.D., Plotkin, J.B. (2010) Inferring the dynamics of diversification: a coalescent approach, PLoS B 8(9): e1000493
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)
Vtimes &lt;- sort(branching.times(Cetacea))
lamb0 &lt;- 0.1
alpha &lt;- 0.001
mu0&lt;-0
beta&lt;-0
ntips &lt;- Ntip(Cetacea)
N0 &lt;- 89
likelihood &lt;- likelihood_coal_var(Vtimes, ntips, lamb0, alpha, mu0, beta, N0)
</code></pre>

<hr>
<h2 id='likelihood_subgroup_model'>
Likelihood of a dataset under models with biogeography fit to a subgroup.
</h2><span id='topic+likelihood_subgroup_model'></span>

<h3>Description</h3>

<p>Computes the likelihood of a dataset under either the linear or exponential diversity dependent model with specified <code>sigma2</code> and slope values and with a <code>geography.object</code> formed using <code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_subgroup_model(data,phylo,geography.object,model=c("MC","DDexp","DDlin"),
	par,return.z0=FALSE,maxN=NULL,error=NULL)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_subgroup_model_+3A_phylo">phylo</code></td>
<td>
  
<p>an object of type 'phylo' (see ape documentation) produced as &quot;map&quot; from CreateGeobyClassObject. NB: the length of this object need not match number of items in data, since map may include tips outside of group with some part of their branch in the group
</p>
</td></tr>
<tr><td><code id="likelihood_subgroup_model_+3A_data">data</code></td>
<td>

<p>a named vector of continuous data for a subgroup of interest with names corresponding to <code>phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="likelihood_subgroup_model_+3A_geography.object">geography.object</code></td>
<td>

<p>a list of sympatry/group membership through time created using <code><a href="#topic+CreateGeobyClassObject">CreateGeobyClassObject</a></code>
</p>
</td></tr>
<tr><td><code id="likelihood_subgroup_model_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"DDlin"</code> is the diversity-dependent linear model, and <code>"DDexp"</code> is the diversity-dependent exponential model of Weir &amp; Mursleen 2013.
</p>
</td></tr>
<tr><td><code id="likelihood_subgroup_model_+3A_par">par</code></td>
<td>

<p>a vector listing a value for <code>log(sig2)</code> (see Note) and either <code>b</code> (for the linear diversity dependent model) or <code>r</code> (for the exponential diversity dependent model), in that order.
</p>
</td></tr>
<tr><td><code id="likelihood_subgroup_model_+3A_return.z0">return.z0</code></td>
<td>

<p>logical indicating whether to return an estimate of the trait value at the root given the parameter values (if <code>TRUE</code>, function returns root value rather than negative log-likelihood)
</p>
</td></tr>
<tr><td><code id="likelihood_subgroup_model_+3A_maxn">maxN</code></td>
<td>

<p>when fitting <code>DDlin</code> model, it is necessary to specify the maximum number of sympatric lineages to ensure that the rate returned does not correspond to negative sig2 values at any point in time (see Details).
</p>
</td></tr>
<tr><td><code id="likelihood_subgroup_model_+3A_error">error</code></td>
<td>

<p>A named vector with standard errors (SE) of trait values for each species (with names matching <code>"phylo$tip.label"</code>). The default is NULL, in this case potential error is ignored in the fit. If set to NA, the SE is estimated from the data (to be used when there are no error measurements, a nuisance parameter is estimated). Note: When standard errors are provided, a nuisance parameter is also estimated.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When specifying <code>par</code>, <code>log(sig2)</code> (see Note) must be listed before the slope parameter (<code>b</code> or <code>r</code>).
</p>
<p>maxN can be calculated using <code>maxN=max(vapply(geo.object$geography.object,function(x)max(rowSums(x)),1))</code>, where geo.object is the output of <code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
</p>


<h3>Value</h3>

<p>The negative log-likelihood value of the dataset (accordingly, the negative of the output should be recorded as the likelihood), given the phylogeny,  <code>sig2</code> and slope values, and <code>geography.object</code>.
</p>
<p>If <code>return.z0=TRUE</code>, the estimated root value for the par values is returned instead of the negative log-likelihood.
</p>


<h3>Note</h3>

<p>To stabilize optimization, this function exponentiates the input <code>sig2</code> value, thus the user must input the log(sig2) value to compute the correct log likelihood (see example).
</p>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>
<p>Julien Clavel
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Weir, J. &amp; Mursleen, S. 2012. Diversity-dependent cladogenesis and trait evolution in the adaptive radiation of the auks (Aves: Alcidae). <em>Evolution</em> 67:403-416.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
<code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
<code><a href="#topic+likelihood_t_DD">likelihood_t_DD</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(BGB.examples)


Canidae.phylo&lt;-BGB.examples$Canidae.phylo
dummy.group&lt;-c(rep("B",3),rep("A",12),rep("B",2),rep("A",6),rep("B",5),rep("A",6))
names(dummy.group)&lt;-Canidae.phylo$tip.label


Canidae.simmap&lt;-phytools::make.simmap(Canidae.phylo, dummy.group)

set.seed(123)
Canidae.data&lt;-rnorm(length(Canidae.phylo$tip.label))
names(Canidae.data)&lt;-Canidae.phylo$tip.label
Canidae.A&lt;-Canidae.data[which(dummy.group=="A")]
Canidae.geobyclass.object&lt;-CreateGeobyClassObject(phylo=Canidae.phylo, 
	simmap=Canidae.simmap, trim.class="A", ana.events=BGB.examples$Canidae.ana.events, 
	clado.events=BGB.examples$Canidae.clado.events,stratified=FALSE, rnd=5)

par &lt;- c(log(0.01),-0.000005)
maxN&lt;-max(vapply(Canidae.geobyclass.object$geo.object$geography.object, 
	function(x)max(rowSums(x)),1))

lh &lt;- -likelihood_subgroup_model(data=Canidae.A, phylo=Canidae.geobyclass.object$map, 
	geography.object=Canidae.geobyclass.object$geo.object, model="DDlin", par=par, 
	return.z0=FALSE, maxN=maxN)
	
	

</code></pre>

<hr>
<h2 id='likelihood_t_DD'>
Likelihood of a dataset under diversity-dependent models.
</h2><span id='topic+likelihood_t_DD'></span>

<h3>Description</h3>

<p>Computes the likelihood of a dataset under either the linear or exponential diversity dependent model with specified <code>sigma2</code> and slope values. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_t_DD(phylo, data, par,model=c("DDlin","DDexp"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_t_DD_+3A_phylo">phylo</code></td>
<td>
  
<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="likelihood_t_DD_+3A_data">data</code></td>
<td>

<p>a named vector of continuous data with names corresponding to <code>phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="likelihood_t_DD_+3A_par">par</code></td>
<td>

<p>a vector listing a value for <code>log(sig2)</code> (see Note) and either <code>b</code> (for the linear diversity dependent model) or <code>r</code> (for the exponential diversity dependent model), in that order.
</p>
</td></tr>
<tr><td><code id="likelihood_t_DD_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"DDlin"</code> is the diversity-dependent linear model, and <code>"DDexp"</code> is the diversity-dependent exponential model of Weir &amp; Mursleen 2013.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When specifying <code>par</code>, <code>log(sig2)</code> must be listed before the slope parameter (<code>b</code> or <code>r</code>).
</p>


<h3>Value</h3>

<p>the negative log-likelihood value of the dataset (accordingly, the negative of the output should be recorded as the likelihood), given the phylogeny and <code>sig2</code> and slope values
</p>


<h3>Note</h3>

<p>To stabilize optimization, this function exponentiates the input <code>sig2</code> value, thus the user must input the log(sig2) value to compute the correct log likelihood (see example).
</p>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>
<p>Julien Clavel
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Weir, J. &amp; Mursleen, S. 2012. Diversity-dependent cladogenesis and trait evolution in the adaptive radiation of the auks (Aves: Alcidae). <em>Evolution</em> 67:403-416.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
<code><a href="#topic+likelihood_t_DD_geog">likelihood_t_DD_geog</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Anolis.data)
phylo &lt;- Anolis.data$phylo
pPC1 &lt;- Anolis.data$data

# Compute the likelihood that the r value is twice the ML estimate for the DDexp model
par &lt;- c(0.08148371, (2*-0.3223835))
lh &lt;- -likelihood_t_DD(phylo,pPC1,par,model="DDexp")
</code></pre>

<hr>
<h2 id='likelihood_t_DD_geog'>
Likelihood of a dataset under diversity-dependent models with biogeography.
</h2><span id='topic+likelihood_t_DD_geog'></span>

<h3>Description</h3>

<p>Computes the likelihood of a dataset under either the linear or exponential diversity dependent model with specified <code>sigma2</code> and slope values and with a <code>geography.object</code> formed using <code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_t_DD_geog(phylo, data, par,geo.object,model=c("DDlin","DDexp"),maxN=NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_t_DD_geog_+3A_phylo">phylo</code></td>
<td>
  
<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="likelihood_t_DD_geog_+3A_data">data</code></td>
<td>

<p>a named vector of continuous data with names corresponding to <code>phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="likelihood_t_DD_geog_+3A_par">par</code></td>
<td>

<p>a vector listing a value for <code>log(sig2)</code> (see Note) and either <code>b</code> (for the linear diversity dependent model) or <code>r</code> (for the exponential diversity dependent model), in that order.
</p>
</td></tr>
<tr><td><code id="likelihood_t_DD_geog_+3A_geo.object">geo.object</code></td>
<td>

<p>a list of sympatry through time created using <code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
</p>
</td></tr>
<tr><td><code id="likelihood_t_DD_geog_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"DDlin"</code> is the diversity-dependent linear model, and <code>"DDexp"</code> is the diversity-dependent exponential model of Weir &amp; Mursleen 2013.
</p>
</td></tr>
<tr><td><code id="likelihood_t_DD_geog_+3A_maxn">maxN</code></td>
<td>

<p>when fitting <code>DDlin</code> model, it is necessary to specify the maximum number of sympatric lineages to ensure that the rate returned does not correspond to negative sig2 values at any point in time (see Details).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When specifying <code>par</code>, <code>log(sig2)</code> (see Note) must be listed before the slope parameter (<code>b</code> or <code>r</code>).
</p>
<p>maxN can be calculated using <code>maxN=max(vapply(geo.object$geography.object,function(x)max(rowSums(x)),1))</code>, where geo.object is the output of <code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
</p>


<h3>Value</h3>

<p>the negative log-likelihood value of the dataset (accordingly, the negative of the output should be recorded as the likelihood), given the phylogeny,  <code>sig2</code> and slope values, and <code>geography.object</code>.
</p>


<h3>Note</h3>

<p>To stabilize optimization, this function exponentiates the input <code>sig2</code> value, thus the user must input the log(sig2) value to compute the correct log likelihood (see example).
</p>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>
<p>Julien Clavel
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Weir, J. &amp; Mursleen, S. 2012. Diversity-dependent cladogenesis and trait evolution in the adaptive radiation of the auks (Aves: Alcidae). <em>Evolution</em> 67:403-416.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
<code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
<code><a href="#topic+likelihood_t_DD">likelihood_t_DD</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Anolis.data)
phylo &lt;- Anolis.data$phylo
pPC1 &lt;- Anolis.data$data
geography.object &lt;- Anolis.data$geography.object

# Compute the likelihood with geography using ML parameters for fit without geography
par &lt;- c(log(0.01153294),-0.0006692378)
maxN&lt;-max(vapply(geography.object$geography.object,function(x)max(rowSums(x)),1))
lh &lt;- -likelihood_t_DD_geog(phylo,pPC1,par,geography.object,model="DDlin",maxN=maxN)
</code></pre>

<hr>
<h2 id='likelihood_t_env'>
Likelihood of a dataset under environmental models of trait evolution.
</h2><span id='topic+likelihood_t_env'></span>

<h3>Description</h3>

<p>Computes the likelihood of a dataset under either the linear or exponential environmental model, or an user defined environmental model. This function is used internally by fit_t_env.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_t_env(phylo, data, model=c("EnvExp", "EnvLin"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_t_env_+3A_phylo">phylo</code></td>
<td>
  
<p>an object of class 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="likelihood_t_env_+3A_data">data</code></td>
<td>

<p>a named vector of continuous data with names corresponding to <code>phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="likelihood_t_env_+3A_...">...</code></td>
<td>

<p>&quot;param&quot;, &quot;fun&quot;, &quot;times&quot;, &quot;mtot&quot; and &quot;error&quot; arguments.
</p>
<p>-param: a vector with the parameters used in the environmental function. The first value is <code>sig2</code> and the second is <code>beta</code>.
</p>
<p>-fun: a time contnuous function of an environmental variable (see e.g. ?fit_t_env)
</p>
<p>-times: a vector of branching times starting at zero 
(e.g. max(branching.times(phylo))-branching.times(phylo))
</p>
<p>-mtot: root age of the tree (e.g. max(branching.times(phylo)))
</p>
<p>-error: a vector of standard error (se) for each species
</p>
<p>If the &quot;times&quot; argument is not provided, the &quot;phylo&quot; object is used to compute it as well as &quot;mtot&quot;.
</p>
<p>Note that the argument &quot;mu&quot; can be used to specify the root state (e.g. when using an mcmc sampler)
</p>
</td></tr>
<tr><td><code id="likelihood_t_env_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"EnvExp"</code> is the exponential-environmental model, and <code>"EnvLin"</code> is the linear-environmental model. Otherwise, an user specified model can be provided.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>the &quot;fun&quot; argument can be filled by an environmental dataframe.
</p>


<h3>Value</h3>

<p>the log-likelihood value of the environmental model
</p>


<h3>Author(s)</h3>

<p>Julien Clavel
</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Science, 114(16): 4183-4188.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_env">fit_t_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(test){
data(Cetacea)
data(InfTemp)

# Simulate a trait with temperature dependence on the Cetacean tree
set.seed(123)

trait &lt;- sim_t_env(Cetacea, param=c(0.1,-0.2), env_data=InfTemp, model="EnvExp", 
					root.value=0, step=0.001, plot=TRUE)
					
# Compute the likelihood 
likelihood_t_env(Cetacea, trait, param=c(0.1, 0), fun=InfTemp, model="EnvExp")

# Provide the times
brtime&lt;-branching.times(Cetacea)
mtot&lt;-max(brtime)
times&lt;-mtot-brtime

likelihood_t_env(Cetacea,trait,param=c(0.1, 0), fun=InfTemp, 
                  times=times, mtot=mtot, model="EnvExp")

# Provide the environmental function rather than the dataset (faster if used recursively)
#require(pspline)
#spline_result &lt;- sm.spline(InfTemp[,1],InfTemp[,2], df=50)
#env_func &lt;- function(t){predict(spline_result,t)}
#t&lt;-unique(InfTemp[,1])
# We build the interpolated smoothing spline function
#env_data&lt;-splinefun(t,env_func(t))
  
#likelihood_t_env(Cetacea, trait, param=c(0.1, 0), fun=env_data, 
#                 times=times, mtot=mtot, model="EnvExp")

	}  
</code></pre>

<hr>
<h2 id='likelihood_t_MC'>
Likelihood of a dataset under the matching competition model.
</h2><span id='topic+likelihood_t_MC'></span>

<h3>Description</h3>

<p>Computes the likelihood of a dataset under the matching competition model with specified <code>sigma2</code> and <code>S</code> values. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_t_MC(phylo, data, par)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_t_MC_+3A_phylo">phylo</code></td>
<td>
  
<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="likelihood_t_MC_+3A_data">data</code></td>
<td>

<p>a named vector of continuous data with names corresponding to <code>phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="likelihood_t_MC_+3A_par">par</code></td>
<td>

<p>a vector listing a value for <code>log(sig2)</code> (see Note) and <code>S</code> (parameters of the matching competition model), in that order
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When specifying <code>par</code>, <code>log(sig2)</code> must be listed before <code>S</code>.
</p>


<h3>Value</h3>

<p>the negative log-likelihood value of the dataset (accordingly, the negative of the output should be recorded as the likelihood), given the phylogeny and <code>sig2</code> and <code>S</code> values
</p>


<h3>Note</h3>

<p>To stabilize optimization, this function exponentiates the input <code>sig2</code> value, thus the user must input the log(sig2) value to compute the correct log likelihood (see example).
</p>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>
<p>Julien Clavel
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Nuismer, S. &amp; Harmon, L. 2015. Predicting rates of interspecific interaction from phylogenetic trees. <em>Ecology Letters</em> 18:17-27.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
<code><a href="#topic+likelihood_t_MC_geog">likelihood_t_MC_geog</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Anolis.data)
phylo &lt;- Anolis.data$phylo
pPC1 &lt;- Anolis.data$data

# Compute the likelihood that the S value is twice the ML estimate
par &lt;- c(0.0003139751, (2*-0.06387258))
lh &lt;- -likelihood_t_MC(phylo,pPC1,par)
</code></pre>

<hr>
<h2 id='likelihood_t_MC_geog'>
Likelihood of a dataset under the matching competition model with biogeography.
</h2><span id='topic+likelihood_t_MC_geog'></span>

<h3>Description</h3>

<p>Computes the likelihood of a dataset under the matching competition model with specified <code>sigma2</code> and <code>S</code> values and with a <code>geography.object</code> formed using <code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood_t_MC_geog(phylo, data, par,geo.object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_t_MC_geog_+3A_phylo">phylo</code></td>
<td>
  
<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="likelihood_t_MC_geog_+3A_data">data</code></td>
<td>

<p>a named vector of continuous data with names corresponding to <code>phylo$tip.label</code>
</p>
</td></tr>
<tr><td><code id="likelihood_t_MC_geog_+3A_par">par</code></td>
<td>

<p>a vector listing a value for <code>log(sig2)</code> (see Note) and <code>S</code> (parameters of the matching competition model), in that order
</p>
</td></tr>
<tr><td><code id="likelihood_t_MC_geog_+3A_geo.object">geo.object</code></td>
<td>

<p>a geography object indicating sympatry through time, created using <code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When specifying <code>par</code>, <code>log(sig2)</code> must be listed before <code>S</code>.
</p>


<h3>Value</h3>

<p>the negative log-likelihood value of the dataset (accordingly, the negative of the output should be recorded as the likelihood), given the phylogeny, <code>sig2</code> and <code>S</code> values, and <code>geography.object</code>.
</p>


<h3>Note</h3>

<p>S must be negative (if it is positive, the likelihood function will multiply input by -1).
</p>
<p>To stabilize optimization, this function exponentiates the input <code>sig2</code> value, thus the user must input the log(sig2) value to compute the correct log likelihood (see example).
</p>


<h3>Author(s)</h3>

<p>Jonathan Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>
<p>Julien Clavel
</p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Nuismer, S. &amp; Harmon, L. 2015. Predicting rates of interspecific interaction from phylogenetic trees. <em>Ecology Letters</em> 18:17-27.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
<code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
<code><a href="#topic+likelihood_t_MC">likelihood_t_MC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Anolis.data)
phylo &lt;- Anolis.data$phylo
pPC1 &lt;- Anolis.data$data
geography.object &lt;-  Anolis.data$geography.object

# Compute the likelihood with geography using ML parameters for fit without geography
par &lt;- c(0.0003139751, -0.06387258)
lh &lt;- -likelihood_t_MC_geog(phylo,pPC1,par,geography.object)
</code></pre>

<hr>
<h2 id='lines.fit_t.env'>
Add to a plot line segments joining the phenotypic evolutionary rate through time estimated by the <em>fit_t_env</em> function
</h2><span id='topic+lines.fit_t.env'></span>

<h3>Description</h3>

<p>Plot estimated evolutionary rate as a function of the environmental data and time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'fit_t.env'
lines(x, steps = 100, ...)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lines.fit_t.env_+3A_x">x</code></td>
<td>

<p>an object of class 'fit_t.env' obtained from a <em>fit_t_env</em> fit.
</p>
</td></tr>
<tr><td><code id="lines.fit_t.env_+3A_steps">steps</code></td>
<td>

<p>the number of steps from the root to the present used to compute the evolutionary rate <code class="reqn">\sigma2</code> through time.
</p>
</td></tr>
<tr><td><code id="lines.fit_t.env_+3A_...">...</code></td>
<td>

<p>further arguments to be passed to <code>plot</code>. See ?<code>plot</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>lines.fit_t.env</code> returns invisibly a list with the following components used to add the line segments to the current plot:
</p>
<table role = "presentation">
<tr><td><code>time_steps</code></td>
<td>
<p>the times steps where the climatic function was evaluated to compute the rate. The number of steps is controlled through the argument <code>steps</code>.</p>
</td></tr>
<tr><td><code>rates</code></td>
<td>
<p>the estimated evolutionary rate through time estimated at each <code>time_steps</code></p>
</td></tr>
</table>


<h3>Note</h3>

<p>All the graphical parameters (see <code><a href="graphics.html#topic+par">par</a></code>) can be passed through (e.g. line type: <code>lty</code>, line width: <code>lwd</code>, color: <code>col</code> ...)
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Science, 114(16): 4183-4188.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.fit_t.env">plot.fit_t.env</a></code>, <code><a href="#topic+likelihood_t_env">likelihood_t_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(test){

data(Cetacea)
data(InfTemp)

# Plot estimated evolutionary rate as a function of the environmental data and time.
set.seed(123)
trait &lt;- sim_t_env(Cetacea, param=c(0.1,-0.2), env_data=InfTemp, model="EnvExp", 
					root.value=0, step=0.01, plot=TRUE)


## Fit the Environmental-exponential model with different smoothing parameters

result1=fit_t_env(Cetacea, trait, env_data=InfTemp, scale=TRUE)
result2=fit_t_env(Cetacea, trait, env_data=InfTemp, scale=TRUE, df=10)

# first plot result1
plot(result1, lwd=3)

# add result2 to the current plot
lines(result2, lty=2, lwd=3, col="red")

}

</code></pre>

<hr>
<h2 id='lines.fit_t.env.ou'>
Add to a plot line segments joining the phenotypic evolutionary optimum through time estimated by the <em>fit_t_env_ou</em> function
</h2><span id='topic+lines.fit_t.env.ou'></span>

<h3>Description</h3>

<p>Plot estimated optimum as a function of the environmental data and time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'fit_t.env.ou'
lines(x, steps = 100, ...)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lines.fit_t.env.ou_+3A_x">x</code></td>
<td>

<p>an object of class 'fit_t.env.ou' obtained from a <em>fit_t_env_ou</em> fit.
</p>
</td></tr>
<tr><td><code id="lines.fit_t.env.ou_+3A_steps">steps</code></td>
<td>

<p>the number of steps from the root to the present used to compute the optimum <code class="reqn">\theta(t)</code> through time.
</p>
</td></tr>
<tr><td><code id="lines.fit_t.env.ou_+3A_...">...</code></td>
<td>

<p>further arguments to be passed to <code>plot</code>. See ?<code>plot</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>lines.fit_t.env.ou</code> returns invisibly a list with the following components used to add the line segments to the current plot:
</p>
<table role = "presentation">
<tr><td><code>time_steps</code></td>
<td>
<p>the times steps where the climatic function was evaluated to compute the rate. The number of steps is controlled through the argument <code>steps</code>.</p>
</td></tr>
<tr><td><code>values</code></td>
<td>
<p>the estimated optimum through time estimated at each <code>time_steps</code></p>
</td></tr>
</table>


<h3>Note</h3>

<p>All the graphical parameters (see <code><a href="graphics.html#topic+par">par</a></code>) can be passed through (e.g. line type: <code>lty</code>, line width: <code>lwd</code>, color: <code>col</code> ...)
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Sciences,  114(16): 4183-4188.
</p>
<p>Troyer, E., Betancur-R, R., Hughes, L., Westneat, M., Carnevale, G., White W.T., Pogonoski, J.J., Tyler, J.C., Baldwin, C.C., Orti, G., Brinkworth, A., Clavel, J., Arcila, D., 2022. The impact of paleoclimatic changes on body size evolution in marine fishes. Proceedings of the National Academy of Sciences, 119 (29), e2122486119.
</p>
<p>Goswami, A. &amp; Clavel, J., 2024. Morphological evolution in a time of Phenomics. EcoEvoRxiv, https://doi.org/10.32942/X22G7Q
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.fit_t.env.ou">plot.fit_t.env.ou</a></code>, <code><a href="#topic+fit_t_env_ou">fit_t_env_ou</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(test){

data(InfTemp)
set.seed(9999) # for reproducibility

# Let's start by simulating a trait under a climatic OU
beta = 0.6           # relationship to the climate curve
sim_theta = 4        # value of the optimum if the relationship to the climate 
# curve is 0 (this corresponds to an 'intercept' in the linear relationship used below)
sim_sigma2 = 0.025   # variance of the scatter = sigma^2
sim_alpha = 0.36     # alpha value = strength of the OU; quite high here...
delta = 0.001        # time step used for the forward simulations =&gt; here its 1000y steps
tree &lt;- pbtree(n=200, d=0.3) # simulate a bd tree with some extinct lineages
root_age = 60        # height of the root (almost all the Cenozoic here)
tree$edge.length &lt;- root_age*tree$edge.length/max(nodeHeights(tree)) 
# here - for this contrived example - I scale the tree so that the root is at 60 Ma

trait &lt;- sim_t_env_ou(tree, sigma=sqrt(sim_sigma2), alpha=sim_alpha, theta0=sim_theta, param=beta, 
              env_data=InfTemp, step=0.01, scale=TRUE, plot=FALSE)

## Fit the Environmental model (default)

result1 &lt;- fit_t_env_ou(phylo = tree, data = trait, env_data =InfTemp,
                        method = "Nelder-Mead", df=50, scale=TRUE)
plot(result1, lty=2)

result2 &lt;- fit_t_env_ou(phylo = tree, data = trait, env_data =InfTemp, 
                        method = "Nelder-Mead", df=10, scale=TRUE)
lines(result2, col="red")

}

</code></pre>

<hr>
<h2 id='make_gen.BipartiteEvol'>
Compute the genealogies for BipartiteEvol</h2><span id='topic+make_gen.BipartiteEvol'></span>

<h3>Description</h3>

<p>Compute the genealogies from a run of BipartiteEvol
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_gen.BipartiteEvol(out, treeP = NULL, treeH = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_gen.BipartiteEvol_+3A_out">out</code></td>
<td>

<p>The output of a run of sim.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="make_gen.BipartiteEvol_+3A_treep">treeP</code></td>
<td>

<p>Optional, a previous genealogy for clade P to which the new tree will be grafted (used if out was the continuation of a former run, see in the example)
</p>
</td></tr>
<tr><td><code id="make_gen.BipartiteEvol_+3A_treeh">treeH</code></td>
<td>

<p>Optional, a previous genealogy for clade H to which the new tree will be grafted (used if out was the continuation of a former run, see in the example)
</p>
</td></tr>
<tr><td><code id="make_gen.BipartiteEvol_+3A_verbose">verbose</code></td>
<td>

<p>Should the progression of the computation be printed?
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list object with
</p>
<table role = "presentation">
<tr><td><code>P</code></td>
<td>
<p>The genealogy of the clade P</p>
</td></tr>
<tr><td><code>H</code></td>
<td>
<p>The genealogy of the clade H</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet, O., Loeuille, N. and Morlon, H. (2020), An individual-based model for the eco-evolutionary emergence of bipartite interaction networks. Ecol Lett. doi:10.1111/ele.13592</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim.BipartiteEvol">sim.BipartiteEvol</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(test){
# run the model
set.seed(1)
mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 800,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5)

#build the genealogies
gen = make_gen.BipartiteEvol(mod)
plot(gen$H)

#compute the phylogenies
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#plot the result
plot_div.BipartiteEvol(gen,phy1, 1)

#build the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = nx, spatial = FALSE)


## add time steps to a former run
seed=as.integer(10)
set.seed(seed)

mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 200,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5,
                        P=mod$P,H=mod$H)  # former run output

# update the genealogy
gen = make_gen.BipartiteEvol(mod,
                             treeP=gen$P, treeH=gen$H)

# update the phylogenies...
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#... and the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = 10, spatial = FALSE)

}
</code></pre>

<hr>
<h2 id='mantel_test'>
Compute Mantel test
</h2><span id='topic+mantel_test'></span>

<h3>Description</h3>

<p>This function computes a Mantel test between two dissimilarity matrices. The available correlations are Pearson, Spearman, and Kendall. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mantel_test(formula = formula(data), data = sys.parent(),
correlation = "Pearson", nperm = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mantel_test_+3A_formula">formula</code></td>
<td>

<p>formula y ~ x describing the test to be conducted where y and x  are distance matrices (as &quot;dist&quot; objects).
</p>
</td></tr>
<tr><td><code id="mantel_test_+3A_data">data</code></td>
<td>

<p>an optional data frame containing the variables in the model as columns of dissimilarities. By default, the variables are taken from the current environment.
</p>
</td></tr>
<tr><td><code id="mantel_test_+3A_correlation">correlation</code></td>
<td>

<p>indicates which correlation (R) must be used among Pearson (default), Spearman, and Kendall correlations.
</p>
</td></tr>
<tr><td><code id="mantel_test_+3A_nperm">nperm</code></td>
<td>

<p>a number of permutations to evaluate the significance of the correlation. By default, it equals 1000, but this can be very long for the Kendall correlation.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is adapted from the function mantel in the R-package ecodist (Goslee &amp; Urban, 2007).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>mantelr</code></td>
<td>
<p>Mantel correlation (R).</p>
</td></tr>
<tr><td><code>pval1</code></td>
<td>
<p>one-tailed p-value (null hypothesis: R &lt;= 0).</p>
</td></tr>
<tr><td><code>pval2</code></td>
<td>
<p>one-tailed p-value (null hypothesis: R &gt;= 0).</p>
</td></tr>
<tr><td><code>pval3</code></td>
<td>
<p>two-tailed p-value (null hypothesis: R = 0).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Perez-Lamarque B, Maliet O, Pichon B, Selosse M-A, Martos F, Morlon H. 2022. Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. Peer Community Journal, Volume 2, article no. e59. doi : 10.24072/pcjournal.179. https://peercommunityjournal.org/articles/10.24072/pcjournal.179/
</p>
<p>Goslee, S.C. &amp; Urban, D.L. (2007). The ecodist package for dissimilarity-based analysis of ecological data. J. Stat. Softw., 22, 1–19.
</p>
<p>Mantel, N. 1967. The detection of disease clustering and a generalized regression approach. Cancer Research 27:209-220.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylosignal_network">phylosignal_network</a></code>
</p>
<p><code><a href="#topic+phylosignal_sub_network">phylosignal_sub_network</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Measuring phylogenetic signal in species interactions using a Mantel test 
# (do closely related species interact with similar partners?)

library(RPANDA)

# Load the data
data(mycorrhizal_network)

network &lt;- mycorrhizal_network[[1]] # bipartite interaction matrix 
tree_orchids &lt;- mycorrhizal_network[[2]] # phylogenetic tree (phylo object)

network &lt;- network[,tree_orchids$tip.label]

ecological_distances &lt;- as.matrix(vegan::vegdist(t(network), "jaccard", binary=FALSE))
    
phylogenetic_distances &lt;- cophenetic.phylo(tree_orchids)

mantel_test(as.dist(ecological_distances) ~ as.dist(phylogenetic_distances), 
correlation="Pearson",  nperm = 10000)
      
</code></pre>

<hr>
<h2 id='mantel_test_nbpartners'>
Compute Mantel test
</h2><span id='topic+mantel_test_nbpartners'></span>

<h3>Description</h3>

<p>This function tests for phylogenetic signal in species interactions in guild  A using a Mantel test that keep constant the number of partners per species.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mantel_test_nbpartners(network, tree_A, tree_B = NULL, method="Jaccard_binary",
nperm = 1000, correlation = "Pearson", verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mantel_test_nbpartners_+3A_network">network</code></td>
<td>

<p>a matrix representing the bipartite interaction network with species from guild A in columns and species from guild B in rows. Row names (resp. columns names) must correspond to the tip labels of tree B (resp. tree A).
</p>
</td></tr>
<tr><td><code id="mantel_test_nbpartners_+3A_tree_a">tree_A</code></td>
<td>

<p>a phylogenetic tree of guild A (the columns of the interaction network). It must be an object of class &quot;phylo&quot;.
</p>
</td></tr>
<tr><td><code id="mantel_test_nbpartners_+3A_tree_b">tree_B</code></td>
<td>

<p>(optional) a phylogenetic tree of guild B (the rows of the interaction network). It must be an object of class &quot;phylo&quot;.
</p>
</td></tr>
<tr><td><code id="mantel_test_nbpartners_+3A_method">method</code></td>
<td>

<p>indicates which method is used to compute the phylogenetic signal in species interactions. If you want to perform a Mantel test between the phylogenetic distances and some ecological distances (do closely related species interact with similar partners?), you can choose &quot;Jaccard_weighted&quot; (default) for computing the ecological distances using Jaccard dissimilarities (or &quot;Jaccard_binary&quot; to not take into account the abundances of the interactions), &quot;Bray-Curtis&quot; for computing the Bray-Curtis dissimilarity, or &quot;GUniFrac&quot; for computing the weighted (or generalized) UniFrac distances (&quot;UniFrac_unweighted&quot; to not take into account the interaction abundances).
</p>
</td></tr>
<tr><td><code id="mantel_test_nbpartners_+3A_nperm">nperm</code></td>
<td>

<p>a number of permutations to evaluate the significance of the correlation. By default, it equals 1000.
</p>
</td></tr>
<tr><td><code id="mantel_test_nbpartners_+3A_correlation">correlation</code></td>
<td>

<p>indicates which correlation (R) must be used among Pearson (default) and Spearman correlations.
</p>
</td></tr>
<tr><td><code id="mantel_test_nbpartners_+3A_verbose">verbose</code></td>
<td>

<p>if TRUE, enables printing of messages.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>mantelr</code></td>
<td>
<p>Mantel correlation (R).</p>
</td></tr>
<tr><td><code>pval1</code></td>
<td>
<p>one-tailed p-value (null hypothesis: R &lt;= 0).</p>
</td></tr>
<tr><td><code>pval2</code></td>
<td>
<p>one-tailed p-value (null hypothesis: R &gt;= 0).</p>
</td></tr>
<tr><td><code>pval3</code></td>
<td>
<p>two-tailed p-value (null hypothesis: R = 0).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Perez-Lamarque B, Maliet O, Pichon B, Selosse M-A, Martos F, Morlon H. 2022. Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. Peer Community Journal, Volume 2, article no. e59. doi : 10.24072/pcjournal.179. https://peercommunityjournal.org/articles/10.24072/pcjournal.179/
</p>
<p>Mantel, N. 1967. The detection of disease clustering and a generalized regression approach. Cancer Research 27:209-220.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylosignal_network">phylosignal_network</a></code>
</p>
<p><code><a href="#topic+phylosignal_sub_network">phylosignal_sub_network</a></code>
</p>
<p><code><a href="#topic+mantel_test">mantel_test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Measuring phylogenetic signal in species interactions using a Mantel test 
# with permutations keeping constant the number of partners per species

library(RPANDA)

# Load the data
data(mycorrhizal_network)

network &lt;- mycorrhizal_network[[1]] # bipartite interaction matrix 
tree_orchids &lt;- mycorrhizal_network[[2]] # phylogenetic tree (phylo object)


mantel_test_nbpartners(network, tree_orchids, method="Jaccard_weighted", 
correlation="Pearson",  nperm = 1000)
   
</code></pre>

<hr>
<h2 id='modelSelection'>
Phenotypic model selection from tip trait data.
</h2><span id='topic+modelSelection'></span>

<h3>Description</h3>

<p>For each model taken as input, fits the model and returns its AIC value in a recap table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modelSelection(object, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="modelSelection_+3A_object">object</code></td>
<td>
<p>a vector of objects of class 'PhenotypicModel'.</p>
</td></tr>
<tr><td><code id="modelSelection_+3A_data">data</code></td>
<td>
<p>vector of tip trait data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Warning : This function relies on the standard R optimizer &quot;optim&quot;. It may not always converge well. Please double check the convergence by trying distinct parameter sets for the initialisation.
</p>


<h3>Value</h3>

<p>A recap table presenting the AIC value of each model.
</p>


<h3>Author(s)</h3>

<p>M Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology
</p>

<hr>
<h2 id='modelSelection-methods'> ~~ Methods for Function <code>modelSelection</code>  ~~</h2><span id='topic+modelSelection-methods'></span><span id='topic+modelSelection+2CPhenotypicModel-method'></span>

<h3>Description</h3>

<p>~~ Methods for function <code>modelSelection</code>  ~~
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "PhenotypicModel")</code></dt><dd>
<p>This is the only method available for this function. Same behaviour for any PhenotypicModel.
</p>
</dd>
</dl>

<hr>
<h2 id='MPhiFFT'>
A class used internally to compute ClaDS's likelihood</h2><span id='topic+MPhiFFT'></span>

<h3>Description</h3>

<p>This class represents a matrix A = (1/rowSums(Toep)) * Toep where Toep is a Toeplitz matrix.</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_ClaDS">fit_ClaDS</a></code>
</p>

<hr>
<h2 id='mycorrhizal_network'>Mycorrhizal network from La Réunion island</h2><span id='topic+mycorrhizal_network'></span>

<h3>Description</h3>

<p>Mycorrhizal intercation network between orchids and mycorrhizal fungi from La Réunion island (Martos et al., 2012) along with the reconstructed phylogenetic trees of the orchids and the fungal OTUs.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mycorrhizal_network)</code></pre>


<h3>Details</h3>

<p>These phylogenies were constructed by maximum likelihood inference from four plastid genes for the orchids and one nuclear gene for the fungi. See Martos et al. (2012) for details.
</p>


<h3>Source</h3>

<p>Martos, F., Munoz, F., Pailler, T., Kottke, I., Gonneau, C. &amp; Selosse, M.-A. (2012). The role of epiphytism in architecture and evolutionary constraint within mycorrhizal networks of tropical orchids. Mol. Ecol., 21, 5098–5109.
</p>


<h3>References</h3>

<p>Martos, F., Munoz, F., Pailler, T., Kottke, I., Gonneau, C. &amp; Selosse, M.-A. (2012). The role of epiphytism in architecture and evolutionary constraint within mycorrhizal networks of tropical orchids. Molecular Ecology, 21, 5098–5109.
</p>
<p>Perez-Lamarque B, Maliet O, Pichon B, Selosse M-A, Martos F, Morlon H. 2022. Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. bioRxiv, 2021.08.30.458192, ver. 6 peer-reviewed and recommended by Peer Community in Evolutionary Biology. https://doi.org/10.1101/2021.08.30.458192
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(mycorrhizal_network)

network &lt;- mycorrhizal_network[[1]] # interaction matrix 
tree_orchids &lt;- mycorrhizal_network[[2]] # phylogenetic tree (phylo object)
tree_fungi &lt;- mycorrhizal_network[[3]] # phylogenetic tree (phylo object)

</code></pre>

<hr>
<h2 id='ostracoda'>Ostracod diversity since the Jurassic</h2><span id='topic+ostracoda'></span>

<h3>Description</h3>

<p>Ostracod fossil diversity since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sealevel)</code></pre>


<h3>Details</h3>

<p>Ostracod fossil diversity since the Jurassic compiled from the Neptune Database (Lazarus, 1994) and Paleobiology Database (https://paleobiodb.org/). Diversity curves are estimated at the genus level using shareholder quorum subsampling (Alroy, 2010) at two-million-year bins. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>ostracoda</code></dt><dd><p>a numeric vector corresponding to the estimated ostracod change at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Lazarus, D. (1994) Neptune: A marine micropaleontology database <em>Mathematical Geology 26:817–832</em>
</p>
<p>Alroy, J. (2010) Geographical, environmental and intrinsic biotic controls on Phanerozoic marine diversification: Controls on phanerozoic marine diversification <em>Palaeontology 53:1211–1235</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ostracoda)
plot(ostracoda)
</code></pre>

<hr>
<h2 id='paleodiv'>
Paleodiversity through time
</h2><span id='topic+paleodiv'></span>

<h3>Description</h3>

<p>Calculates paleodiversity through time from shift.estimates output with the deterministic approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  paleodiv(phylo, data, sampling.fractions, shift.res,
           backbone.option = "crown.shift", combi = 1,
           time.interval = 1, split.div = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="paleodiv_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="paleodiv_+3A_data">data</code></td>
<td>

<p>a data.frame containing a database of monophyletic groups for which potential shifts can be investigated. This database should be based on taxonomy, ecology or traits and contain a column named &quot;Species&quot; with species name as in phylo.
</p>
</td></tr>
<tr><td><code id="paleodiv_+3A_sampling.fractions">sampling.fractions</code></td>
<td>

<p>the output resulting from get.sampling.fractions.
</p>
</td></tr>
<tr><td><code id="paleodiv_+3A_shift.res">shift.res</code></td>
<td>

<p>the output resulting from shift.estimates.
</p>
</td></tr>
<tr><td><code id="paleodiv_+3A_backbone.option">backbone.option</code></td>
<td>

<p>type of the backbone analysis:
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: paleodiversity dynamics are calculated from the stem age for subclades.
</p>
</li>
<li><p>&quot;crown.shift&quot;: paleodiversity dynamics are calculated from the crown age for subclades.
</p>
</li></ul>

</td></tr>
<tr><td><code id="paleodiv_+3A_combi">combi</code></td>
<td>

<p>numeric. The combination of shifts defined by its rank in the global comparison.
</p>
</td></tr>
<tr><td><code id="paleodiv_+3A_time.interval">time.interval</code></td>
<td>

<p>numeric. Define the time interval (in million years) at which paleodiversity values are calculated. Default is 1 for a value at each million year.
</p>
</td></tr>
<tr><td><code id="paleodiv_+3A_split.div">split.div</code></td>
<td>

<p>bolean. Specifies if paleodiversity should be plitted by parts of the selected combination (TRUE) or not. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If split.div = FALSE, paleodiversity dynamics are returned in a matrix with as many rows as parts in the selected combination and as many column as million years from the root to the present. If spit.div = TRUE, global paleodiversity dynamic is returned as a vector with a value per million year.
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.estimates">shift.estimates</a>, <a href="#topic+apply_prob_dtt">apply_prob_dtt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># loading data
data("Cetacea")
data("taxo_cetacea")
data("shifts_cetacea")

# no shifts tested at genus level
taxo_cetacea_no_genus &lt;- taxo_cetacea[names(taxo_cetacea) != "Genus"]
f_cetacea &lt;- get.sampling.fractions(phylo = Cetacea, lad = FALSE,
                                    data = taxo_cetacea_no_genus,
                                    plot = TRUE, cex = 0.3)
# use of paleodiv
paleodiversity &lt;- paleodiv(phylo = Cetacea,
                           data = taxo_cetacea_no_genus,
                           sampling.fractions = f_cetacea,
                           shift.res = shifts_cetacea,
                           combi = 1, split.div = FALSE)
</code></pre>

<hr>
<h2 id='PhenotypicACDC-class'>Class <code>"PhenotypicACDC"</code></h2><span id='topic+PhenotypicACDC-class'></span>

<h3>Description</h3>

<p>Subclass of the PhenotypicModel class intended to represent the model of ACcelerating or DeCelerating phenotypic evolution.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("PhenotypicACDC", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>matrixCoalescenceTimes</code>:</dt><dd><p>Object of class <code>"matrix"</code> ~~ </p>
</dd>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>period</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>aAGamma</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>numbersCopy</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>numbersPaste</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>initialCondition</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>paramsNames</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>constraints</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>params0</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>tipLabels</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>tipLabelsSimu</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>comment</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+PhenotypicModel-class">PhenotypicModel</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>getTipDistribution</dt><dd><p><code>signature(object = "PhenotypicACDC")</code>: ... </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Marc Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology, and the associated Supplementary material.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("PhenotypicACDC")
</code></pre>

<hr>
<h2 id='PhenotypicADiag-class'>Class <code>"PhenotypicADiag"</code></h2><span id='topic+PhenotypicADiag-class'></span>

<h3>Description</h3>

<p>A subclass of the PhenotypicModel class, intended to represent models of phenotypic evolution with a &quot;A&quot; matrix diagonalizable.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("PhenotypicADiag", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>period</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>aAGamma</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>numbersCopy</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>numbersPaste</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>initialCondition</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>paramsNames</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>constraints</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>params0</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>tipLabels</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>tipLabelsSimu</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>comment</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+PhenotypicModel-class">PhenotypicModel</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>getTipDistribution</dt><dd><p><code>signature(object = "PhenotypicADiag")</code>: ... </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Marc Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology, and the associated Supplementary material.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("PhenotypicADiag")
</code></pre>

<hr>
<h2 id='PhenotypicBM-class'>Class <code>"PhenotypicBM"</code></h2><span id='topic+PhenotypicBM-class'></span>

<h3>Description</h3>

<p>A subclass of the PhenotypicModel class, intended to represent the model of Brownian phenotypic evolution.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("PhenotypicBM", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>matrixCoalescenceTimes</code>:</dt><dd><p>Object of class <code>"matrix"</code> ~~ </p>
</dd>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>period</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>aAGamma</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>numbersCopy</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>numbersPaste</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>initialCondition</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>paramsNames</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>constraints</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>params0</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>tipLabels</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>tipLabelsSimu</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>comment</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+PhenotypicModel-class">PhenotypicModel</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>getTipDistribution</dt><dd><p><code>signature(object = "PhenotypicBM")</code>: ... </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Marc Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology, and the associated Supplementary material.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("PhenotypicBM")
</code></pre>

<hr>
<h2 id='PhenotypicDD-class'>Class <code>"PhenotypicDD"</code></h2><span id='topic+PhenotypicDD-class'></span>

<h3>Description</h3>

<p>A subclass of the PhenotypicModel class, intended to represent the model of Density-Dependent phenotypic evolution.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("PhenotypicDD", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>matrixCoalescenceJ</code>:</dt><dd><p>Object of class <code>"matrix"</code> ~~ </p>
</dd>
<dt><code>nLivingLineages</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>period</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>aAGamma</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>numbersCopy</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>numbersPaste</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>initialCondition</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>paramsNames</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>constraints</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>params0</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>tipLabels</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>tipLabelsSimu</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>comment</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+PhenotypicModel-class">PhenotypicModel</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>getTipDistribution</dt><dd><p><code>signature(object = "PhenotypicDD")</code>: ... </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Marc Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology, and the associated Supplementary material.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("PhenotypicDD")
</code></pre>

<hr>
<h2 id='PhenotypicGMM-class'>Class <code>"PhenotypicGMM"</code></h2><span id='topic+PhenotypicGMM-class'></span>

<h3>Description</h3>

<p>A subclass of the PhenotypicModel class, intended to represent the Generalist Matching Mutualism model of phenotypic evolution. This is a model of phenotypic evolution with interactions between two clades, running on two trees.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("PhenotypicGMM", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>n1</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>n2</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>period</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>aAGamma</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>numbersCopy</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>numbersPaste</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>initialCondition</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>paramsNames</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>constraints</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>params0</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>tipLabels</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>tipLabelsSimu</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>comment</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+PhenotypicModel-class">PhenotypicModel</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>getTipDistribution</dt><dd><p><code>signature(object = "PhenotypicGMM")</code>: ... </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Marc Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology, and the associated Supplementary material.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("PhenotypicGMM")
</code></pre>

<hr>
<h2 id='PhenotypicModel-class'>Class <code>"PhenotypicModel"</code></h2><span id='topic+PhenotypicModel-class'></span><span id='topic++5B+3C-+2CPhenotypicModel+2CANY+2CANY+2CANY-method'></span><span id='topic++5B+2CPhenotypicModel+2CANY+2CANY+2CANY-method'></span><span id='topic+print+2CPhenotypicModel-method'></span><span id='topic+show+2CPhenotypicModel-method'></span>

<h3>Description</h3>

<p>This class describes a model of phenotypic evolution running on a phylogenetic tree, with or without interactions between lineages.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("PhenotypicModel", ...)</code>.
Alternatively, you may just want to use the &quot;createModel&quot; function for predefined models.
</p>


<h3>Slots</h3>


<dl>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>period</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>aAGamma</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>numbersCopy</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>numbersPaste</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>initialCondition</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>paramsNames</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>constraints</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>params0</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>tipLabels</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>tipLabelsSimu</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>comment</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>[&lt;-</dt><dd><p><code>signature(x = "PhenotypicModel", i = "ANY", j = "ANY", value = "ANY")</code>: ... </p>
</dd>
<dt>[</dt><dd><p><code>signature(x = "PhenotypicModel", i = "ANY", j = "ANY", drop = "ANY")</code>: ... </p>
</dd>
<dt>fitTipData</dt><dd><p><code>signature(object = "PhenotypicModel")</code>: ... </p>
</dd>
<dt>getDataLikelihood</dt><dd><p><code>signature(object = "PhenotypicModel")</code>: ... </p>
</dd>
<dt>getTipDistribution</dt><dd><p><code>signature(object = "PhenotypicModel")</code>: ... </p>
</dd>
<dt>modelSelection</dt><dd><p><code>signature(object = "PhenotypicModel")</code>: ... </p>
</dd>
<dt>print</dt><dd><p><code>signature(x = "PhenotypicModel")</code>: ... </p>
</dd>
<dt>show</dt><dd><p><code>signature(object = "PhenotypicModel")</code>: ... </p>
</dd>
<dt>simulateTipData</dt><dd><p><code>signature(object = "PhenotypicModel")</code>: ... </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Marc Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology, and the associated Supplementary material.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("PhenotypicModel")
</code></pre>

<hr>
<h2 id='PhenotypicOU-class'>Class <code>"PhenotypicOU"</code></h2><span id='topic+PhenotypicOU-class'></span>

<h3>Description</h3>

<p>A subclass of the PhenotypicModel class, intended to represent the Ornstein-Uhlenbeck model of phenotypic evolution.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("PhenotypicOU", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>matrixCoalescenceTimes</code>:</dt><dd><p>Object of class <code>"matrix"</code> ~~ </p>
</dd>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>period</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>aAGamma</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>numbersCopy</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>numbersPaste</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>initialCondition</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>paramsNames</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>constraints</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>params0</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>tipLabels</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>tipLabelsSimu</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>comment</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+PhenotypicModel-class">PhenotypicModel</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>getTipDistribution</dt><dd><p><code>signature(object = "PhenotypicOU")</code>: ... </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Marc Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology, and the associated Supplementary material.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("PhenotypicOU")
</code></pre>

<hr>
<h2 id='PhenotypicPM-class'>Class <code>"PhenotypicPM"</code></h2><span id='topic+PhenotypicPM-class'></span>

<h3>Description</h3>

<p>A subclass of the PhenotypicModel class, intended to represent the Phenotypic Matching model of phenotypic evolution, by Nuismer and Harmon (Eco Lett, 2014).
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("PhenotypicPM", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>period</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>aAGamma</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>numbersCopy</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>numbersPaste</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>initialCondition</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>paramsNames</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>constraints</code>:</dt><dd><p>Object of class <code>"function"</code> ~~ </p>
</dd>
<dt><code>params0</code>:</dt><dd><p>Object of class <code>"numeric"</code> ~~ </p>
</dd>
<dt><code>tipLabels</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>tipLabelsSimu</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
<dt><code>comment</code>:</dt><dd><p>Object of class <code>"character"</code> ~~ </p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"<a href="#topic+PhenotypicModel-class">PhenotypicModel</a>"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>getTipDistribution</dt><dd><p><code>signature(object = "PhenotypicPM")</code>: ... </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Marc Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology, and the associated Supplementary material.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("PhenotypicPM")
</code></pre>

<hr>
<h2 id='Phocoenidae'>Phocoenidae phylogeny</h2><span id='topic+Phocoenidae'></span>

<h3>Description</h3>

<p>Ultrametric phylogenetic tree of the 6 extant Phocoenidae (porpoise) species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Phocoenidae)</code></pre>


<h3>Details</h3>

<p>This phylogeny was extracted from Steeman et al. Syst Bio 2009 cetacean phylogeny</p>


<h3>References</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans <em>Syst Biol</em> 58:573-585
</p>
<p>Morlon, H., Parsons, T.L., Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Phocoenidae)
print(Phocoenidae)
plot(Phocoenidae)

</code></pre>

<hr>
<h2 id='phyl.pca_pl'>
Regularized Phylogenetic Principal Component Analysis (PCA).
</h2><span id='topic+phyl.pca_pl'></span>

<h3>Description</h3>

<p>Performs a principal component analysis (PCA) on a regularized evolutionary variance-covariance matrix obtained using the <code>fit_t_pl</code> function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>
phyl.pca_pl(object, plot=TRUE, ...)
  
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phyl.pca_pl_+3A_object">object</code></td>
<td>

<p>A penalized likelihood model fit obtained by the <code>fit_t_pl</code> function.
</p>
</td></tr>
<tr><td><code id="phyl.pca_pl_+3A_plot">plot</code></td>
<td>

<p>Plot of the PC's axes. Default is TRUE (see details).'</p>
</td></tr>
<tr><td><code id="phyl.pca_pl_+3A_...">...</code></td>
<td>

<p>Options to be passed through. (e.g., axes=c(1,2), col, pch, cex, mode=&quot;cov&quot; or &quot;corr&quot;, etc.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>phyl.pca_pl</code> allows computing a phylogenetic principal component analysis (following Revell 2009) using a regularized evolutionary variance-covariance matrix from penalized likelihood models fit to high-dimensional datasets (where the number of variables p is potentially larger than n; see details for the models options in <code>fit_t_pl</code>). Models estimates are more accurate than maximum likelihood methods, particularly in the high-dimensional case.
Ploting options, the number of axes to display (<code>axes=c(1,2)</code> is the default), and whether the covariance (<code>mode="cov"</code>) or correlation (<code>mode="corr"</code>) should be used can be specified through the ellipsis &quot;...&quot; argument.
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>values</code></td>
<td>
<p>the eigenvalues of the evolutionary variance-covariance matrix</p>
</td></tr>
<tr><td><code>scores</code></td>
<td>
<p>the PC scores</p>
</td></tr>
<tr><td><code>loadings</code></td>
<td>
<p>the component loadings</p>
</td></tr>
<tr><td><code>nodes_scores</code></td>
<td>
<p>the scores for the ancestral states at the nodes (projected on the space of the tips)</p>
</td></tr>
<tr><td><code>mean</code></td>
<td>
<p>the mean/ancestral value used to center the data</p>
</td></tr>
<tr><td><code>vectors</code></td>
<td>
<p>the eigenvectors of the evolutionary variance-covariance matrix</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Contrary to conventional PCA, the principal axes of the phylogenetic PCA are not orthogonal, they represent the main axes of (independent) evolutionary changes.</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Revell, L.J., 2009. Size-correction and principal components for intraspecific comparative studies. Evolution, 63:3258-3268.
</p>
<p>Clavel, J., Aristide, L., Morlon, H., 2019. A Penalized Likelihood framework for high-dimensional phylogenetic comparative methods and an application to new-world monkeys brain evolution. Syst. Biol. 68: 93-116.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_pl">fit_t_pl</a></code>,
<code><a href="#topic+ancestral">ancestral</a></code>,
<code><a href="#topic+GIC.fit_pl.rpanda">GIC.fit_pl.rpanda</a></code>, 
<code><a href="#topic+gic_criterion">gic_criterion</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(test){
if(require(mvMORPH)){
set.seed(1)
n &lt;- 32 # number of species
p &lt;- 31 # number of traits

tree &lt;- pbtree(n=n) # phylogenetic tree
R &lt;- Posdef(p)      # a random symmetric matrix (covariance)

# simulate a dataset
Y &lt;- mvSIM(tree, model="BM1", nsim=1, param=list(sigma=R))

# fit a multivariate Pagel lambda model with Penalized likelihood
fit &lt;- fit_t_pl(Y, tree, model="lambda", method="RidgeAlt")

# Perform a phylogenetic PCA using the model fit (Pagel lambda model)
pca_results &lt;- phyl.pca_pl(fit, plot=TRUE) 

# retrieve the scores
head(pca_results$scores)
}
}

</code></pre>

<hr>
<h2 id='Phyllostomidae'>Phyllostomidae phylogeny</h2><span id='topic+Phyllostomidae'></span>

<h3>Description</h3>

<p>Ultrametric phylogenetic tree of 150 of the 165 extant known Phyllostomidae species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Phyllostomidae)</code></pre>


<h3>Details</h3>

<p>This phylogeny is the maximum clade credibility tree used in Rolland et al. (2014), which originally comes from the Bininda-Emonds tree (Bininda-Emonds et al. 2007)</p>


<h3>References</h3>

<p>Bininda-Emonds, O. R., et al. (2007) The delayed rise of present-day mammals <em>Nature</em> 446: 507-512
</p>
<p>Rolland, J., Condamine, F. L., Jiguet, F., &amp; Morlon, H. (2014) Faster speciation and reduced extinction in the tropics contribute to the mammalian latitudinal diversity gradient. <em>PLoS Biol</em>, 12(1): e1001775.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Phyllostomidae_genera">Phyllostomidae_genera</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Phyllostomidae)
print(Phyllostomidae)
plot(Phyllostomidae)

</code></pre>

<hr>
<h2 id='Phyllostomidae_genera'>Phylogenies of Phyllostomidae genera</h2><span id='topic+Phyllostomidae_genera'></span>

<h3>Description</h3>

<p>List of 25 ultrametric phylogenetic trees corresponding to 25 Phyllostomidae genera
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Phyllostomidae_genera)</code></pre>


<h3>See Also</h3>

<p><code><a href="#topic+Phyllostomidae">Phyllostomidae</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Phyllostomidae_genera)
print(Phyllostomidae_genera)
</code></pre>

<hr>
<h2 id='phylosignal_network'>
Compute phylogenetic signal in a bipartite interaction network
</h2><span id='topic+phylosignal_network'></span>

<h3>Description</h3>

<p>This function computes the phylogenetic signal in a bipartite interaction network, either the phylogenetic signal in species interactions (do closely related species interact with similar partners?) using Mantel tests, or the phylogenetic signal in the number of partners (i.e. degree; do closely related species interact with the same number of partners?) using Mantel tests or using the Phylogenetic bipartite linear model (PBLM) from Ives and Godfray (2006). 
Mantel tests measuring the phylogenetic signal in species interactions can be computed using quantified or binary networks, with the Jaccard, Bray-Curtis, or UniFrac ecological distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylosignal_network(network, tree_A, tree_B = NULL, 
method = "Jaccard_weighted", nperm = 10000, 
correlation = "Pearson", only_A = FALSE, permutation = "shuffle")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylosignal_network_+3A_network">network</code></td>
<td>

<p>a matrix representing the bipartite interaction network with species from guild A in columns and species from guild B in rows. Row names (resp. columns names) must correspond to the tip labels of tree B (resp. tree A).
</p>
</td></tr>
<tr><td><code id="phylosignal_network_+3A_tree_a">tree_A</code></td>
<td>

<p>a phylogenetic tree of guild A (the columns of the interaction network). It must be an object of class &quot;phylo&quot;.
</p>
</td></tr>
<tr><td><code id="phylosignal_network_+3A_tree_b">tree_B</code></td>
<td>

<p>(optional) a phylogenetic tree of guild B (the rows of the interaction network). It must be an object of class &quot;phylo&quot;.
</p>
</td></tr>
<tr><td><code id="phylosignal_network_+3A_method">method</code></td>
<td>

<p>indicates which method is used to compute the phylogenetic signal in species interactions. If you want to perform a Mantel test between the phylogenetic distances and some ecological distances (do closely related species interact with similar partners?), you can choose &quot;Jaccard_weighted&quot; (default) for computing the ecological distances using Jaccard dissimilarities (or &quot;Jaccard_binary&quot; to not take into account the abundances of the interactions), &quot;Bray-Curtis&quot; for computing the Bray-Curtis dissimilarity, or &quot;GUniFrac&quot; for computing the weighted (or generalized) UniFrac distances (&quot;UniFrac_unweighted&quot; to not take into account the interaction abundances).
</p>
<p>Conversely, if you want to evaluate the phylogenetic signal in the number of partners (do closely related species interact with the same number of partners?), you can choose &quot;degree&quot;. 
</p>
<p>Alternatively (not recommended), you can use the Phylogenetic Bipartite Linear Model &quot;PBLM&quot; (see Ives and Godfray, 2006) or &quot;PBLM_binary&quot; to not consider the abundances of the interactions.
</p>
</td></tr>
<tr><td><code id="phylosignal_network_+3A_correlation">correlation</code></td>
<td>

<p>(optional) indicates which correlation (R) must be used in the Mantel test, among Pearson (default), Spearman, and Kendall correlations. It only applies for the methods &quot;Jaccard_weighted&quot;, &quot;Jaccard_binary&quot;, &quot;Bray-Curtis&quot;, &quot;GUniFrac&quot;, &quot;UniFrac_unweighted&quot;, or &quot;degree&quot;.
</p>
</td></tr>
<tr><td><code id="phylosignal_network_+3A_nperm">nperm</code></td>
<td>

<p>(optional) a number of permutations to evaluate the significance of the Mantel test. By default, it equals 10,000, but this can be very long for the Kendall correlation. It only applies for the methods &quot;Jaccard_weighted&quot;, &quot;Bray-Curtis&quot;, &quot;Jaccard_binary&quot;, &quot;GUniFrac&quot;, &quot;UniFrac_unweighted&quot;, or &quot;degree&quot;.
</p>
</td></tr>
<tr><td><code id="phylosignal_network_+3A_permutation">permutation</code></td>
<td>

<p>(optional) indicates which permutations must be performed to evaluate the significance of the Mantel correlation: either &quot;shuffle&quot; (by default - i.e. random shufflying of the distance matrix) or &quot;nbpartners&quot; (i.e. keeping constant the number of partners per species and shuffling at random their identity).
</p>
</td></tr>
<tr><td><code id="phylosignal_network_+3A_only_a">only_A</code></td>
<td>

<p>(optional) indicates whether the signal should be only computed for guild A (and not for guild B). By default, it is computed for both guilds if &quot;tree_B&quot; is provided. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the tutorial on GitHub (https://github.com/BPerezLamarque/Phylosignal_network). 
</p>


<h3>Value</h3>

<p>For Mantel tests, the function outputs a vector of up to 8 values: the number of species in guild A (&quot;nb_A&quot;), the number of species in guild B (&quot;nb_B&quot;), the correlation for guild A (&quot;mantel_cor_A&quot;), its associated upper p-value (&quot;pvalue_upper_A&quot;, i.e. the fraction of permutations that led to higher correlation values), its associated lower p-value (&quot;pvalue_lower_A&quot;, i.e. the fraction of permutations that led to lower correlation values), and (optional) the correlation for guild B (&quot;mantel_cor_B&quot;), its associated upper p-value (&quot;pvalue_upper_B&quot;), and its associated lower p-value (&quot;pvalue_lower_B&quot;),
</p>
<p>&quot;mantel_cor_A&quot; (or &quot;mantel_cor_B&quot;) indicates the strength of the phylogenetic signal in guild A (or B). The upper p-value &quot;pvalue_upper_A&quot; (or &quot;pvalue_upper_B&quot;) indicates the significance of the phylogenetic signal in guild A (or B). The lower p-value &quot;pvalue_lower_A&quot; (or &quot;pvalue_lower_B&quot;) indicates the significance of the anti-phylogenetic signal in guild A (or B). 
For instance, if &quot;pvalue_upper_A&quot;&lt;0.05, there is a significant phylogenetic signal in guild A. 
</p>
<p>For the PBLM approach (Ives and Godfray, 2006), the function outputs a vector of 8 values: the number of species in guild A (&quot;nb_A&quot;), the number of species in guild B (&quot;nb_B&quot;), the phylogenetic signals in guilds A (&quot;dA&quot;) and B (&quot;dB&quot;), the covariance of interaction matrix (&quot;MSETotal&quot;), the mean square error of the complete model (&quot;MSEFull&quot;), the mean square error of model run on star phylogenies (&quot;MSEStar&quot;), and the mean square error of the model assuming strict Brownian motion evolutions (&quot;MSEBase&quot;). The significance of the phylogenetic signal can be evaluated by comparing &quot;MSEFull&quot; and &quot;MSEStar&quot;.
</p>


<h3>Author(s)</h3>

<p>Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Perez-Lamarque B, Maliet O, Pichon B, Selosse M-A, Martos F, Morlon H. 2022. Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. Peer Community Journal, Volume 2, article no. e59. doi : 10.24072/pcjournal.179. https://peercommunityjournal.org/articles/10.24072/pcjournal.179/
</p>
<p>Goslee, S.C. &amp; Urban, D.L. (2007). The ecodist package for dissimilarity-based analysis of ecological data. J. Stat. Softw., 22, 1–19.
</p>
<p>Ives, A.R. &amp; Godfray, H.C.J. (2006). Phylogenetic analysis of trophic associations. Am. Nat., 168, E1–E14.
</p>
<p>Kembel, S.W., Cowan, P.D., Helmus, M.R., Cornwell, W.K., Morlon, H., Ackerly, D.D., et al. (2010). Picante: R tools for integrating phylogenies and ecology. Bioinformatics, 26, 1463–1464.
</p>
<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., et al. (2012). Associating microbiome composition with environmental covariates using generalized UniFrac distances. Bioinformatics, 28, 2106–2113.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylosignal_sub_network">phylosignal_sub_network</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load the data
data(mycorrhizal_network)

network &lt;- mycorrhizal_network[[1]] # interaction matrix 
tree_orchids &lt;- mycorrhizal_network[[2]] # phylogenetic tree (phylo object)
tree_fungi &lt;- mycorrhizal_network[[3]] # phylogenetic tree (phylo object)



# Using Mantel tests: 

# Step 1: Phylogenetic signal in species interactions 
# (do closely related species interact with similar partners?)

phylosignal_network(network, tree_A = tree_orchids, tree_B = tree_fungi, 
method = "GUniFrac", correlation = "Pearson", nperm = 10000) # measured for both guilds


# Step 2: Phylogenetic signal in species interactions when accouting 
# for the signal in the number of partners 
# Mantel test with permutations that keep constant the number of partners per species

phylosignal_network(network, tree_A = tree_orchids, tree_B = tree_fungi, 
method = "GUniFrac", correlation = "Pearson", nperm = 1000, permutation = "nbpartners")



# Other: Phylogenetic signal in the number of partners 
# (do closely related species interact with the same number of partners?)

phylosignal_network(network, tree_A = tree_orchids, method = "degree", 
correlation = "Pearson", nperm = 10000) # for guild A
phylosignal_network(t(network), tree_A = tree_fungi, method = "degree", 
correlation = "Pearson", nperm = 10000) # for guild B



# Alternative using PBLM (not recommended) - very slow 

phylosignal_network(network, tree_A = tree_orchids, tree_B = tree_fungi, method = "PBLM") 


</code></pre>

<hr>
<h2 id='phylosignal_sub_network'>
Compute clade-specific phylogenetic signals in a bipartite interaction network
</h2><span id='topic+phylosignal_sub_network'></span>

<h3>Description</h3>

<p>This function computes the clade-specific phylogenetic signals in species interactions. For each node of tree A having a certain number of descending species, it computes the phylogenetic signal in the resulting sub-network by performing a Mantel test between the phylogenetic distances and the ecological distances for the given sub-clade of tree A. 
Mantel tests can be computed using quantified or binary networks, with the Jaccard, Bray-Curtis, or UniFrac ecological distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylosignal_sub_network(network, tree_A, tree_B = NULL, 
method = "Jaccard_weighted", nperm = 1000, 
correlation = "Pearson", minimum = 10, degree = FALSE, 
permutation = "shuffle", verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylosignal_sub_network_+3A_network">network</code></td>
<td>

<p>a matrix representing the bipartite interaction network with species from guild A in columns and species from guild B in rows. Row names (resp. columns names) must correspond to the tip labels of tree B (resp. tree A).
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_tree_a">tree_A</code></td>
<td>

<p>a phylogenetic tree of guild A (the columns of the interaction network). It must be an object of class &quot;phylo&quot;.
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_tree_b">tree_B</code></td>
<td>

<p>(optional) a phylogenetic tree of guild B (the rows of the interaction network). It must be an object of class &quot;phylo&quot;.
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_method">method</code></td>
<td>

<p>indicates which method is used to compute the phylogenetic signal in species interactions using Mantel tests. You can choose &quot;Jaccard_weighted&quot; (default) for computing ecological distances using Jaccard dissimilarities (or &quot;Jaccard_binary&quot; to not take into account the abundances of the interactions), &quot;Bray-Curtis&quot; for computing the Bray-Curtis dissimilarity, or &quot;GUniFrac&quot; for computing the weighted (or generalized) UniFrac distances (&quot;UniFrac_unweighted&quot; to not take into account the interaction abundances).
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_nperm">nperm</code></td>
<td>

<p>a number of permutations to evaluate the significance of the Mantel test. By default, it equals 10,000, but this can be very long for the Kendall correlation.
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_correlation">correlation</code></td>
<td>

<p>indicates which correlation (R) must be used in the Mantel test, among Pearson (default), Spearman, and Kendall correlations.
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_minimum">minimum</code></td>
<td>

<p>indicates the minimal number of descending species for a node in tree A to compute its clade-specific phylogenetic signal. 
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_degree">degree</code></td>
<td>

<p>if degree=TRUE, Mantel tests testing for phylogenetic signal in the number of partners are additionally performed in each sub-clade.
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_permutation">permutation</code></td>
<td>

<p>(optional) indicates which permutations must be performed to evaluate the significance of the Mantel correlation: either &quot;shuffle&quot; (by default - i.e. random shufflying of the distance matrix) or &quot;nbpartners&quot; (i.e. keeping constant the number of partners per species and shuffling at random their identity).
</p>
</td></tr>
<tr><td><code id="phylosignal_sub_network_+3A_verbose">verbose</code></td>
<td>

<p>if TRUE, enables printing of messages.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the tutorial on GitHub (https://github.com/BPerezLamarque/Phylosignal_network). 
</p>


<h3>Value</h3>

<p>For Mantel tests, the function outputs a table where each line corresponds to a tested clade and which contains at least 8 columns: the name of the node (&quot;node&quot;), the number of species in the sub-clade A (&quot;nb_A&quot;), the number of species in guild B associated with the sub-clade A (&quot;nb_B&quot;), the Mantel correlation for guild A (&quot;mantel_cor&quot;), its associated upper p-value (&quot;pvalue_upper&quot;), its associated lower p-value (&quot;pvalue_lower&quot;), and the associated Bonferroni corrected p-values (&quot;pvalue_upper_corrected&quot; and &quot;pvalue_lower_corrected&quot;). 
</p>
<p>&quot;mantel_cor&quot; indicates the strength of the phylogenetic signal in the sub-clade A. The upper p-value &quot;pvalue_upper&quot; indicates the significance of the phylogenetic signal in the sub-clade A. The lower p-value &quot;pvalue_lower&quot; indicates the significance of the anti-phylogenetic signal in the sub-clade A. Both Bonferroni p-values are corrected using the number of tested nodes. 
For instance, if &quot;pvalue_upper_corrected&quot;&lt;0.05 for a given node, there is a significant phylogenetic signal in the corresponding sub-clade of A.
</p>
<p>If degree=TRUE, it also indicates in each sub-clade, the phylogenetic signal in the number of partners (&quot;degree_mantel_cor&quot;) and its significance with or without the Bonferroni correction (&quot;degree_pvalue_upper&quot;, &quot;degree_pvalue_lower&quot; and &quot;degree_pvalue_upper_corrected&quot;, &quot;degree_pvalue_lower_corrected&quot;)
</p>


<h3>Author(s)</h3>

<p>Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Perez-Lamarque B, Maliet O, Pichon B, Selosse M-A, Martos F, Morlon H. 2022. Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. Peer Community Journal, Volume 2, article no. e59. doi : 10.24072/pcjournal.179. https://peercommunityjournal.org/articles/10.24072/pcjournal.179/
</p>
<p>Goslee, S.C. &amp; Urban, D.L. (2007). The ecodist package for dissimilarity-based analysis of ecological data. J. Stat. Softw., 22, 1–19.
</p>
<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., et al. (2012). Associating microbiome composition with environmental covariates using generalized UniFrac distances. Bioinformatics, 28, 2106–2113.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylosignal_sub_network">phylosignal_sub_network</a></code>
<code><a href="#topic+plot_phylosignal_sub_network">plot_phylosignal_sub_network</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load the data
data(mycorrhizal_network)

network &lt;- mycorrhizal_network[[1]] # interaction matrix 
tree_orchids &lt;- mycorrhizal_network[[2]] # phylogenetic tree (phylo object)
tree_fungi &lt;- mycorrhizal_network[[3]] # phylogenetic tree (phylo object)


if(test){

# Clade-specific phylogenetic signal in species interactions in guild A 
# (do closely related species interact with similar partners in sub-clades of guild A?)

results_clade_A &lt;- phylosignal_sub_network(network, tree_A = tree_orchids, tree_B = tree_fungi,
method = "GUniFrac", correlation = "Pearson", degree = TRUE)
plot_phylosignal_sub_network(tree_A = tree_orchids, results_clade_A, network)

# Clade-specific phylogenetic signal in species interactions in guild B 
# (do closely related species interact with similar partners in sub-clades of guild B?)

results_clade_B &lt;- phylosignal_sub_network(t(network), tree_A = tree_fungi, tree_B = tree_orchids, 
method = "GUniFrac", correlation = "Pearson", degree = TRUE)
plot_phylosignal_sub_network(tree_A = tree_fungi, results_clade_B, t(network))
}

</code></pre>

<hr>
<h2 id='pi_estimator'>
Compute nucleotidic diversity (Pi estimator)
</h2><span id='topic+pi_estimator'></span>

<h3>Description</h3>

<p>This function computes the Pi estimator of genetic diversity (Nei and Li, 1979) while controlling for the presence of gaps in the alignment (Ferretti et al, 2012), frequent in barcoding datasets. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pi_estimator(sequences)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pi_estimator_+3A_sequences">sequences</code></td>
<td>

<p>a matrix representing the nucleotidic alignment of all the sequences present in the phylogenetic tree. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An estimate of genetic diversity 
</p>


<h3>Author(s)</h3>

<p>Ana C. Afonso Silva &amp; Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Nei M &amp; Li WH, Mathematical model for studying genetic variation in terms of restriction endonucleases, 1979, Proc. Natl. Acad. Sci. USA.
</p>
<p>Ferretti L, Raineri E, Ramos-Onsins S. 2012. Neutrality tests for sequences with missing data. Genetics 191: 1397–1401.
</p>
<p>Perez-Lamarque B, Öpik M, Maliet O, Silva A, Selosse M-A, Martos F, and Morlon H. 2022. Analysing diversification dynamics using barcoding data: The case of an obligate mycorrhizal symbiont, Molecular Ecology, 31:3496–512.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+theta_estimator">theta_estimator</a></code>
<code><a href="#topic+delineate_phylotypes">delineate_phylotypes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(woodmouse)

alignment &lt;- as.character(woodmouse) # nucleotidic alignment 

pi_estimator(alignment)

</code></pre>

<hr>
<h2 id='plot_BICompare'>
Display modalities on a phylogeny.
</h2><span id='topic+plot_BICompare'></span>

<h3>Description</h3>

<p>Plot a phylogeny with branches colored according to modalities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_BICompare(phylo,BICompare)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_BICompare_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)</p>
</td></tr>
<tr><td><code id="plot_BICompare_+3A_bicompare">BICompare</code></td>
<td>

<p>an object of class 'BICompare', output of the 'BICompare' function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot of the phylogeny with branches colored according to which modalities they belong to.
</p>


<h3>Author(s)</h3>

<p>E Lewitus</p>


<h3>References</h3>

<p>Lewitus, E., Morlon, H., Characterizing and comparing phylogenies from their Laplacian spectrum, bioRxiv doi: http://dx.doi.org/10.1101/026476</p>


<h3>See Also</h3>

<p><code><a href="#topic+BICompare">BICompare</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Cetacea)
result &lt;- BICompare(Cetacea,5)
plot_BICompare(Cetacea,result)

</code></pre>

<hr>
<h2 id='plot_ClaDS_chains'>
Plot the MCMC chains obtained when infering ClaDS parameters</h2><span id='topic+plot_ClaDS_chains'></span>

<h3>Description</h3>

<p>Plot the MCMC chains obtained with fit_ClaDS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ClaDS_chains(sampler, burn = 1/2, thin = 1, 
                  param = c("sigma", "alpha", "mu", "LP"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_ClaDS_chains_+3A_sampler">sampler</code></td>
<td>

<p>The output of a fit_ClaDS run.
</p>
</td></tr>
<tr><td><code id="plot_ClaDS_chains_+3A_burn">burn</code></td>
<td>

<p>Number of iterations to drop in the beginning of the chains.
</p>
</td></tr>
<tr><td><code id="plot_ClaDS_chains_+3A_thin">thin</code></td>
<td>

<p>Thinning parameter, one iteration out of &quot;thin&quot; is plotted.
</p>
</td></tr>
<tr><td><code id="plot_ClaDS_chains_+3A_param">param</code></td>
<td>

<p>Either a vector of &quot;character&quot; elements with the name of the parameter to plot, or a vector of integers indicating what parameters to plot. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot representing parameter MCMC chains
</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_ClaDS">fit_ClaDS</a></code>, <code><a href="#topic+getMAPS_ClaDS">getMAPS_ClaDS</a></code>, <code><a href="#topic+plot_ClaDS0_chains">plot_ClaDS0_chains</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Caprimulgidae_ClaDS2")

plot_ClaDS_chains(Caprimulgidae_ClaDS2$sampler)

plot_ClaDS_chains(Caprimulgidae_ClaDS2$sampler, burn = 1/4, 
                  param = c("sigma", "alpha", "l_0", "LP"))

plot_ClaDS_chains(Caprimulgidae_ClaDS2$sampler, burn = 1/5, thin = 5, param = c(1,5,6,15))

</code></pre>

<hr>
<h2 id='plot_ClaDS_phylo'>Plot a phylogeny with branch-specific values
</h2><span id='topic+plot_ClaDS_phylo'></span>

<h3>Description</h3>

<p>Plot a phylogeny with branches colored according to branch-specific rate values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ClaDS_phylo(phylo, rates, rates2 = NULL, 
                same.scale = TRUE, main = NULL, lwd = 2, 
                log = TRUE, show.tip.label = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_ClaDS_phylo_+3A_phylo">phylo</code></td>
<td>
<p>An object of class 'phylo'.</p>
</td></tr>
<tr><td><code id="plot_ClaDS_phylo_+3A_rates">rates</code></td>
<td>
<p>A vector containing the branch-specific rates, in the same order as phylo$edges.</p>
</td></tr>
<tr><td><code id="plot_ClaDS_phylo_+3A_rates2">rates2</code></td>
<td>
<p>An optional second vector containing the branch-specific rates, in the same order as phylo$edges. If NULL (the default), the tree is only plotted once with the rate values from rates. If not, the tree is plotted twice, with the rate values from rates in the left panel and those from rates2 in the right panel. </p>
</td></tr>
<tr><td><code id="plot_ClaDS_phylo_+3A_same.scale">same.scale</code></td>
<td>
<p>A boolean specifying whether the values from rates and rates2 are plotted with the same colorscale. Default to TRUE.</p>
</td></tr>
<tr><td><code id="plot_ClaDS_phylo_+3A_main">main</code></td>
<td>
<p>A title for the plot.</p>
</td></tr>
<tr><td><code id="plot_ClaDS_phylo_+3A_lwd">lwd</code></td>
<td>
<p>Width of the tree branch lengths. Default to 2.</p>
</td></tr>
<tr><td><code id="plot_ClaDS_phylo_+3A_log">log</code></td>
<td>
<p>A boolean specifying whether the rates values are plotted on a log scale. Default to TRUE.</p>
</td></tr>
<tr><td><code id="plot_ClaDS_phylo_+3A_show.tip.label">show.tip.label</code></td>
<td>
<p>A boolean specifying whether the labels of the phylogeny should be displayed. Default to FALSE.</p>
</td></tr>
<tr><td><code id="plot_ClaDS_phylo_+3A_...">...</code></td>
<td>
<p>Optional arguments for <code>plot.phylo</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the phylogeny with branches colored according to branch-specific rate values
</p>


<h3>Author(s)</h3>

<p>O. Maliet</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim_ClaDS">sim_ClaDS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)

obj= sim_ClaDS( lambda_0=0.1,    
                mu_0=0.5,      
                sigma_lamb=0.7,         
                alpha_lamb=0.90,     
                condition="taxa",    
                taxa_stop = 20,    
                prune_extinct = TRUE)  

tree = obj$tree
speciation_rates = obj$lamb[obj$rates]
extinction_rates = obj$mu[obj$rates]

oldpar &lt;- par(no.readonly = TRUE)
par(mar=c(1,1,0,0))
plot_ClaDS_phylo(tree,speciation_rates)

plot_ClaDS_phylo(tree,speciation_rates, lwd = 4, log = FALSE)
par(oldpar) # restore the old par

</code></pre>

<hr>
<h2 id='plot_ClaDS0_chains'>
Plot the MCMC chains obtained when infering ClaDS0 parameters
</h2><span id='topic+plot_ClaDS0_chains'></span>

<h3>Description</h3>

<p>Plot the MCMC chains obtained with run_ClaDS0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ClaDS0_chains(sampler, burn = 1/2, thin = 1, 
                  param = c("sigma", "alpha", "l_0", "LP"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_ClaDS0_chains_+3A_sampler">sampler</code></td>
<td>

<p>The output of a run_ClaDS0 run.
</p>
</td></tr>
<tr><td><code id="plot_ClaDS0_chains_+3A_burn">burn</code></td>
<td>

<p>Number of iterations to drop in the beginning of the chains.
</p>
</td></tr>
<tr><td><code id="plot_ClaDS0_chains_+3A_thin">thin</code></td>
<td>

<p>Thinning parameter, one iteration out of &quot;thin&quot; is plotted.
</p>
</td></tr>
<tr><td><code id="plot_ClaDS0_chains_+3A_param">param</code></td>
<td>

<p>Either a vector of &quot;character&quot; elements with the name of the parameter to plot, or a vector of integers indicating what parameters to plot. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot representing parameter MCMC chains
</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_ClaDS0">fit_ClaDS0</a></code>, <code><a href="#topic+getMAPS_ClaDS0">getMAPS_ClaDS0</a></code>, <code><a href="#topic+plot_ClaDS_chains">plot_ClaDS_chains</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("ClaDS0_example")

plot_ClaDS0_chains(ClaDS0_example$Cl0_chains)
plot_ClaDS0_chains(ClaDS0_example$Cl0_chains, param = paste0("lambda_", c(1,10,5)))

</code></pre>

<hr>
<h2 id='plot_div.BipartiteEvol'>
Plot the output of BipartiteEvol
</h2><span id='topic+plot_div.BipartiteEvol'></span>

<h3>Description</h3>

<p>Plot the genealogies and phylogenies simulated with BipartiteEvol
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_div.BipartiteEvol(gen, spec, trait.id, lwdgen = 1, 
    lwdsp = lwdgen, scale = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_div.BipartiteEvol_+3A_gen">gen</code></td>
<td>

<p>The output of a run of make_gen.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="plot_div.BipartiteEvol_+3A_spec">spec</code></td>
<td>

<p>The output of a run of define_species.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="plot_div.BipartiteEvol_+3A_trait.id">trait.id</code></td>
<td>

<p>The trait dimension used to color the genealogies, phylogenies an network with trait values</p>
</td></tr>
<tr><td><code id="plot_div.BipartiteEvol_+3A_lwdgen">lwdgen</code></td>
<td>

<p>Width of the branches of the genealogies, default to 1
</p>
</td></tr>
<tr><td><code id="plot_div.BipartiteEvol_+3A_lwdsp">lwdsp</code></td>
<td>

<p>Width of the branches of the phylogenies, default to 1
</p>
</td></tr>
<tr><td><code id="plot_div.BipartiteEvol_+3A_scale">scale</code></td>
<td>

<p>Optional, used to force the trait scale
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The upper line shows the genealogies colored with trait values for both guilds (the number above shows the depth of the respective genealogy). 
</p>
<p>The second line shows the phylogenies colored with trait values for both guilds (the number above shows the tip number of the respective phylogeny).
</p>
<p>On the third line there is, from left to right, the trait distribution within individuals in guild P, trait of the individual in H as a function of the trait of the interacting individual in P, and the trait distribution within individuals in guild H (for the dimension trait.id). 
</p>
<p>The lower line shows the quantitative interaction network, with species colored according to their mean trait value (for the dimension trait.id).
</p>


<h3>Value</h3>

<p>Plot simulated genealogies and phylogenies 
</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet, O., Loeuille, N. and Morlon, H. (2020), An individual-based model for the eco-evolutionary emergence of bipartite interaction networks. Ecol Lett. doi:10.1111/ele.13592</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim.BipartiteEvol">sim.BipartiteEvol</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># run the model
set.seed(1)


if(test){
mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 1000,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5)

#build the genealogies
gen = make_gen.BipartiteEvol(mod)
plot(gen$H)

#compute the phylogenies
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#plot the result
plot_div.BipartiteEvol(gen,phy1, 1)
}
</code></pre>

<hr>
<h2 id='plot_dtt'>
Plot diversity through time
</h2><span id='topic+plot_dtt'></span>

<h3>Description</h3>

<p>Plot the estimated number of species through time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_dtt(fit.bd, tot_time, N0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_dtt_+3A_fit.bd">fit.bd</code></td>
<td>

<p>an object of class 'fit.bd', output of the 'fit_bd' function
</p>
</td></tr>
<tr><td><code id="plot_dtt_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the underlying phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="plot_dtt_+3A_n0">N0</code></td>
<td>

<p>number of extant species. If all extant species are represented in the phylogeny, N0 is given by length(phylo$tip.label)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot representing how the estimated number of species vary through time
</p>


<h3>Author(s)</h3>

<p>H Morlon</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Morlon, H. (2014) Phylogenetic approaches for studying diversification, Eco Lett 17:508-525
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_bd">fit_bd</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Balaenopteridae)
tot_time&lt;-max(node.age(Balaenopteridae)$ages)

# Fit the pure birth model (no extinction) with exponential variation of the speciation rate
# with time
f.lamb &lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.08, 0.01)
mu_par&lt;-c()
result &lt;- fit_bd(Balaenopteridae,tot_time,f.lamb,f.mu,lamb_par,mu_par,f=1,
                     expo.lamb = TRUE, fix.mu=TRUE)

# plot estimated number of species through time
plot_dtt(result, tot_time, N0=9)
</code></pre>

<hr>
<h2 id='plot_fit_bd'>
Plot speciation, extinction &amp; net diversification rate functions of a fitted model
</h2><span id='topic+plot_fit_bd'></span>

<h3>Description</h3>

<p>Plot estimated speciation, extinction &amp; net diversification rates through time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_fit_bd(fit.bd, tot_time)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_fit_bd_+3A_fit.bd">fit.bd</code></td>
<td>

<p>an object of class 'fit.bd', output of the 'fit_bd' function
</p>
</td></tr>
<tr><td><code id="plot_fit_bd_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots representing how the estimated speciation, extinction &amp; net diversification rate functions vary through time
</p>


<h3>Author(s)</h3>

<p>H Morlon</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_bd">fit_bd</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Balaenopteridae)
tot_time&lt;-max(node.age(Balaenopteridae)$ages)

# Fit the pure birth model (no extinction) with exponential variation of the speciation rate
# with time
f.lamb &lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.08, 0.01)
mu_par&lt;-c()
result &lt;- fit_bd(Balaenopteridae,tot_time,f.lamb,f.mu,lamb_par,mu_par,
                     expo.lamb = TRUE, fix.mu=TRUE)
# plot fitted rates
plot_fit_bd(result, tot_time)
</code></pre>

<hr>
<h2 id='plot_fit_env'>
Plot speciation, extinction &amp; net diversification rate functions of a fitted environmental model
</h2><span id='topic+plot_fit_env'></span>

<h3>Description</h3>

<p>Plot estimated speciation, extinction &amp; net diversification rates as a function of the environmental data and time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_fit_env(fit.env, env_data, tot_time)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_fit_env_+3A_fit.env">fit.env</code></td>
<td>

<p>an object of class 'fit.env', output of the 'fit_env' function
</p>
</td></tr>
<tr><td><code id="plot_fit_env_+3A_env_data">env_data</code></td>
<td>

<p>environmental data, given as a data frame with two columns. The first column is time, the second column is the environmental data (temperature for instance).
</p>
</td></tr>
<tr><td><code id="plot_fit_env_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots representing how the estimated speciation, extinction &amp; net diversification rate functions vary as a function of the environmental data &amp; time
</p>


<h3>Author(s)</h3>

<p>H Morlon and FL Condamine</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_env">fit_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require(pspline)){
data(Balaenopteridae)
tot_time&lt;-max(node.age(Balaenopteridae)$ages)
data(InfTemp)
dof&lt;-smooth.spline(InfTemp[,1], InfTemp[,2])$df

# Fit the pure birth model (no extinction) with exponential variation of the speciation rate
# with temperature. 
f.lamb &lt;-function(t,x,y){y[1] * exp(y[2] * x)}
f.mu&lt;-function(t,x,y){0}
lamb_par&lt;-c(0.10, 0.01)
mu_par&lt;-c()
result &lt;- fit_env(Balaenopteridae,InfTemp,tot_time,f.lamb,f.mu,
      lamb_par,mu_par,f=1, fix.mu=TRUE, df=dof, dt=1e-3)

# plot fitted rates
plot_fit_env(result, InfTemp, tot_time)
    }
  
</code></pre>

<hr>
<h2 id='plot_net.BipartiteEvol'>
Plot the output of BipartiteEvol
</h2><span id='topic+plot_net.BipartiteEvol'></span>

<h3>Description</h3>

<p>Plot the genealogies, phylogenies and interaction network simulated with BipartiteEvol
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_net.BipartiteEvol(gen, spec, trait.id, link, 
    out, lwdgen = 1, lwdsp = lwdgen, scale = NULL, 
    nx = NULL, cor = FALSE, network.method = "bipartite", 
    spatial = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_net.BipartiteEvol_+3A_gen">gen</code></td>
<td>

<p>The output of a run of make_gen.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_spec">spec</code></td>
<td>

<p>The output of a run of define_species.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_trait.id">trait.id</code></td>
<td>

<p>The trait dimension used to color the genealogies, phylogenies an network with trait values</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_out">out</code></td>
<td>

<p>The output of a run of sim.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_link">link</code></td>
<td>

<p>The output of a run of sim.BipartiteEvol
</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_lwdgen">lwdgen</code></td>
<td>

<p>Width of the branches of the genealogies, default to 1
</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_lwdsp">lwdsp</code></td>
<td>

<p>Width of the branches of the phylogenies, default to 1
</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_scale">scale</code></td>
<td>

<p>Optional, used to force the trait scale
</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_nx">nx</code></td>
<td>

<p>Grid size parameter used in sim.BipartiteEvol. If NULL, squrt(N) is used, where N is the number of individuals in a guild</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_cor">cor</code></td>
<td>

<p>If FALSE (the default), the middle panel displays the interraction network with species positionned in trait space. If TRUE, it shows all the individual in trait space</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_network.method">network.method</code></td>
<td>

<p>How should the network be plotted? Can be &quot;bipartite&quot; (the default) or &quot;matrix&quot;</p>
</td></tr>
<tr><td><code id="plot_net.BipartiteEvol_+3A_spatial">spatial</code></td>
<td>

<p>Should the grid with trait values of the individual of both guilds been shown? Default to FALSE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The upper line shows the genealogies colored with trait values for both guilds (the number above shows the depth of the respective genealogy). 
</p>
<p>The second line shows the phylogenies colored with trait values for both guilds (the number above shows the tip number of the respective phylogeny).
</p>
<p>On the third line there is, from left to right, the trait distribution within individuals in guild P (for the dimension trait.id), the interraction network with species positionned in trait space (if cor = T), and the trait distribution within individuals in guild H (for the dimension trait.id). 
</p>
<p>The lower line shows the quantitative interaction network, with species colored according to their mean trait value (for the dimension trait.id).
</p>


<h3>Value</h3>

<p>Plot outputs of BipartiteEvol</p>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet, O., Loeuille, N. and Morlon, H. (2020), An individual-based model for the eco-evolutionary emergence of bipartite interaction networks. Ecol Lett. doi:10.1111/ele.13592</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim.BipartiteEvol">sim.BipartiteEvol</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># run the model
set.seed(1)


if(test){
mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 1000,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5)

#build the genealogies
gen = make_gen.BipartiteEvol(mod)
plot(gen$H)

#compute the phylogenies
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#build the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = 10, spatial = FALSE)
}
</code></pre>

<hr>
<h2 id='plot_phylo_comb'>
Plot shifts of diversifcation on a phylogeny
</h2><span id='topic+plot_phylo_comb'></span>

<h3>Description</h3>

<p>Plots the phylogeny with colored branches according to shifts of diversification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_phylo_comb(phylo, data, sampling.fractions, shift.res = NULL,
                combi, backbone.option = "crown.shift",
                main = NULL, col.sub = NULL, col.bck = "black",
                lty.bck = 1, tested_nodes = FALSE, lad = TRUE,
                leg = TRUE, text.cex = 1, pch.cex = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_phylo_comb_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_data">data</code></td>
<td>

<p>a data.frame containing a database of monophyletic groups for which potential shifts can be tested. This database should be based on taxonomy, ecology or traits and must contain a column named &quot;Species&quot; with species names as in phylo. 
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_sampling.fractions">sampling.fractions</code></td>
<td>

<p>the output resulting from get.sampling.fractions.
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_shift.res">shift.res</code></td>
<td>

<p>the output resulting from shift.estimates or NULL (default). This latter case allows to represent combinations only from the output of <code><a href="#topic+get.comb.shift">get.comb.shift</a></code> by specifying the combination (see argument combi).
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_combi">combi</code></td>
<td>

<p>character or numeric. If shift.res is provided, this argument is a numeric and corresponds to the rank of the combination in the global comparison (shift.res$total). If shift.res is NULL, this argument should be a character giving a combination of node IDs as in get.comb.shift output. This latter manner to specify combination allows to visualize a combination of shifts before having results.
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_backbone.option">backbone.option</code></td>
<td>

<p>type of the backbone analysis (see backbone.option in shift.estimates for more details):
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: the stems of subclades are included in subclade analyses;
</p>
</li>
<li><p>&quot;crown.shift&quot;: the stems of subclades are included in the backbone analysis (Default).
</p>
</li></ul>

</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_main">main</code></td>
<td>

<p>Character. The name of the plot. Default is NULL and the combination rank with AICc will be printed if shift.res is not NULL.
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_col.sub">col.sub</code></td>
<td>

<p>character. A vector to specify colors of subclade(s). Can be let NULL (see details).
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_col.bck">col.bck</code></td>
<td>

<p>character. A vector to specify colors of backbone(s). Default is &quot;black&quot; for simple backbone (see details).
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_lad">lad</code></td>
<td>

<p>bolean. Allows to ladderize the tree.
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_leg">leg</code></td>
<td>

<p>bolean. If TRUE, legend of the selected combination is added to the plot with names from data and best model names. Default is TRUE. The position is automatically adjusted in function of lad argument.
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_lty.bck">lty.bck</code></td>
<td>

<p>numeric. Define lty for the backbone.
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_tested_nodes">tested_nodes</code></td>
<td>

<p>bolean. If TRUE, all the tested nodes are highlighted by a red point.
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_text.cex">text.cex</code></td>
<td>

<p>numeric. Define the size of legend text.
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_pch.cex">pch.cex</code></td>
<td>

<p>numeric. Define the size of points if tested_nodes = TRUE
</p>
</td></tr>
<tr><td><code id="plot_phylo_comb_+3A_...">...</code></td>
<td>

<p>further arguments to be passed to plot or to plot.phylo.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If col.sub is not specified, color vector for subclades is c(c(brewer.pal(8, &quot;Dark2&quot;),brewer.pal(8, &quot;Set1&quot;),&quot;darkmagenta&quot;,&quot;dodgerblue2&quot;, &quot;orange&quot;, &quot;forestgreen&quot;)). For multiple backbone, default vector is c(&quot;blue4&quot;, &quot;orange4&quot;, &quot;red4&quot;, &quot;grey40&quot;, &quot;coral4&quot;, &quot;deeppink4&quot;, &quot;khaki4&quot;, &quot;darkolivegreen&quot;, &quot;darkslategray&quot;,”black”).
... allows to set different graphical parameters from plot.phylo such as cex for size of tip labels or edge.width for the thickness of the phylogeny edges.
</p>


<h3>Value</h3>

<p>plot the phylogeny and returns the same invisible object as plot.phylo.
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.estimates">shift.estimates</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
# loading data
data("Cetacea")
data("taxo_cetacea")

taxo_cetacea_no_genus &lt;- taxo_cetacea[names(taxo_cetacea) != "Genus"]

# main procedure
f_cetacea &lt;- get.sampling.fractions(phylo = Cetacea, lad = FALSE,
                                    data = taxo_cetacea_no_genus,
                                    plot = TRUE, cex = 0.3)

comb.shift_cetacea &lt;- get.comb.shift(phylo = Cetacea,
                                     data = taxo_cetacea_no_genus,
                                     sampling.fractions = f_cetacea,
                                     Ncores = 4)

# use of plot_phylo_comb
# without shift.estimates results but with comb.shift_cetacea

plot_phylo_comb(phylo = Cetacea,
                data = taxo_cetacea,
                sampling.fractions = f_cetacea,
                combi = comb.shift_cetacea[15],
                label.offset = 0.3,
                main = "", lad = FALSE ,cex = 0.4)

</code></pre>

<hr>
<h2 id='plot_phylosignal_sub_network'>
Plot clade-specific phylogenetic signals in a bipartite interaction network
</h2><span id='topic+plot_phylosignal_sub_network'></span>

<h3>Description</h3>

<p>This function plots the clade-specific phylogenetic signals in species interactions. For each node of tree A having a certain number of descending species, it represents the phylogenetic signal in the resulting sub-network by performing a Mantel test between the phylogenetic distances and the ecological distances for the given sub-clade of tree A. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_phylosignal_sub_network(tree_A, results_sub_clades, network, legend=TRUE, 
show.tip.label=FALSE, where="bottomleft", corrected_pvalue=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_phylosignal_sub_network_+3A_tree_a">tree_A</code></td>
<td>

<p>a phylogenetic tree of guild A (the columns of the interaction network). It must be an object of class &quot;phylo&quot;.
</p>
</td></tr>
<tr><td><code id="plot_phylosignal_sub_network_+3A_results_sub_clades">results_sub_clades</code></td>
<td>

<p>output of the function phylosignal_sub_network.
</p>
</td></tr>
<tr><td><code id="plot_phylosignal_sub_network_+3A_network">network</code></td>
<td>

<p>a matrix representing the bipartite interaction network with species from guild A in columns and species from guild B in rows. Row names (resp. columns names) must correspond to the tip labels of tree B (resp. tree A).
</p>
</td></tr>
<tr><td><code id="plot_phylosignal_sub_network_+3A_legend">legend</code></td>
<td>

<p>indicates whether the legend should be plotted. 
</p>
</td></tr>
<tr><td><code id="plot_phylosignal_sub_network_+3A_show.tip.label">show.tip.label</code></td>
<td>

<p>indicates whether the tip labels should be plotted. 
</p>
</td></tr>
<tr><td><code id="plot_phylosignal_sub_network_+3A_where">where</code></td>
<td>

<p>indicates where to put the legend (default is &quot;bottomleft&quot;).
</p>
</td></tr>
<tr><td><code id="plot_phylosignal_sub_network_+3A_corrected_pvalue">corrected_pvalue</code></td>
<td>

<p>indicates whether the corrected p-values (default is TRUE; using Bonferonni correction) or the original p-values (FALSE) should be used.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the tutorial on GitHub (https://github.com/BPerezLamarque/Phylosignal_network). 
</p>


<h3>Value</h3>

<p>A phylogenetic tree with nodes colored according to the clade-specific phylogenetic signals. Blue nodes are not significant, whereas orange-red nodes present significant phylogenetic signals and their color indicates the strength of the signal (correlation R of the Mantel test).
</p>


<h3>Author(s)</h3>

<p>Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Perez-Lamarque B, Maliet O, Pichon B, Selosse M-A, Martos F, Morlon H. 2022. Do closely related species interact with similar partners? Testing for phylogenetic signal in bipartite interaction networks. Peer Community Journal, Volume 2, article no. e59. doi : 10.24072/pcjournal.179. https://peercommunityjournal.org/articles/10.24072/pcjournal.179/
</p>
<p>Goslee, S.C. &amp; Urban, D.L. (2007). The ecodist package for dissimilarity-based analysis of ecological data. J. Stat. Softw., 22, 1–19.
</p>
<p>Chen, J., Bittinger, K., Charlson, E.S., Hoffmann, C., Lewis, J., Wu, G.D., et al. (2012). Associating microbiome composition with environmental covariates using generalized UniFrac distances. Bioinformatics, 28, 2106–2113.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phylosignal_network">phylosignal_network</a></code>
<code><a href="#topic+phylosignal_sub_network">phylosignal_sub_network</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load the data
data(mycorrhizal_network)

network &lt;- mycorrhizal_network[[1]] # interaction matrix 
tree_orchids &lt;- mycorrhizal_network[[2]] # phylogenetic tree (phylo object)
tree_fungi &lt;- mycorrhizal_network[[3]] # phylogenetic tree (phylo object)


if(test){

# Clade-specific phylogenetic signal in species interactions in guild A 
# (do closely related species interact with similar partners in sub-clades of guild A?)

results_clade_A &lt;- phylosignal_sub_network(network, tree_A = tree_orchids, tree_B = tree_fungi,
method = "GUniFrac", correlation = "Pearson")
plot_phylosignal_sub_network(tree_A = tree_orchids, results_clade_A, network)

# Clade-specific phylogenetic signal in species interactions in guild B 
# (do closely related species interact with similar partners in sub-clades of guild B?)

results_clade_B &lt;- phylosignal_sub_network(t(network), tree_A = tree_fungi, tree_B = tree_orchids,
method = "GUniFrac", correlation = "Pearson")
plot_phylosignal_sub_network(tree_A = tree_fungi, results_clade_B, t(network))
}
</code></pre>

<hr>
<h2 id='plot_prob_dtt'>
Plot diversity through time with confidence intervals.
</h2><span id='topic+plot_prob_dtt'></span>

<h3>Description</h3>

<p>Plots confidence intervals of the estimated number of species through time using a matrix of probabilities given by the function 'prob_dtt'. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  plot_prob_dtt(mat, grain =0.1, plot.prob = TRUE, 
                plot.mean = TRUE, int = TRUE, plot.bound=FALSE,
                conf = 0.95, add = FALSE, col.mean = "red", col.bound = "blue",
                lty="solid", lwd=1, lty.bound=1, add.present=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_prob_dtt_+3A_mat">mat</code></td>
<td>

<p>matrix of probabilities, with species numbers as rows and times as columns with rownames and colnames set to the values of each. 
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_grain">grain</code></td>
<td>

<p>the upper limit of a range of probabilities plotted in a gray scale (lower limit is zero). Higher probabilities are plotted in black. Default value is 0.1.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_plot.prob">plot.prob</code></td>
<td>

<p>logical: set to TRUE (default value) to plot the probabilities.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_plot.mean">plot.mean</code></td>
<td>

<p>logical: set to TRUE (default value) to plot a line for the mean.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_plot.bound">plot.bound</code></td>
<td>

<p>logical: set to TRUE to plot the bounds of the confidence interval, int must be set to TRUE.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_int">int</code></td>
<td>

<p>logical: set to TRUE (default value) to plot a confidence interval.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_conf">conf</code></td>
<td>

<p>confidence level. The default value is 0.95.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_add">add</code></td>
<td>

<p>logical: set to TRUE to add the plot on an existing graph.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_col.mean">col.mean</code></td>
<td>

<p>color of the line for the mean.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_col.bound">col.bound</code></td>
<td>

<p>color of the confidence interval bounds
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_lty">lty</code></td>
<td>

<p>style of the line for the mean (if added on a current plot)
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_lwd">lwd</code></td>
<td>

<p>the line width, a positive number (default to 1)
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_lty.bound">lty.bound</code></td>
<td>

<p>style of the line for the bound (if added on a current plot)
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_add.present">add.present</code></td>
<td>

<p>whether or not to add the present diversity value to the plot. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="plot_prob_dtt_+3A_...">...</code></td>
<td>

<p>further arguments to be passed to plot or to plot.phylo.
</p>
</td></tr>
</table>


<h3>Details</h3>

  
<p>The function assumes that the matrix of probabilities 'mat' has species numbers as rows and times as columns with rownames and colnames set to the values of each.
</p>
<p>'Grain' must be between 0 and 1. If the plot is too pale 'grain' should be diminished (and inversely if the plot is too dark)
</p>


<h3>Value</h3>

<p>Plot representing how the estimated number of species vary through time with confidence intervals. The darker is the plot, the higher is the probability.
</p>


<h3>Author(s)</h3>

<p>O.Billaud, T.L.Parsons, D.S.Moen, H.Morlon</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record. Proc. Nat. Acad. Sci. 108: 16327-16332.
</p>
<p>Billaud, O., Moen, D. S., Parsons, T. L., Morlon, H. (under review) Estimating Diversity Through Time using Molecular Phylogenies: Old and Species-Poor Frog Families are the Remnants of a Diverse Past.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_bd">fit_bd</a></code>, <code><a href="#topic+plot_dtt">plot_dtt</a></code>, <code><a href="#topic+prob_dtt">prob_dtt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Balaenopteridae)
tot_time&lt;-max(node.age(Balaenopteridae)$ages)


if(test){
# Fit the pure birth model (no extinction) with exponential variation of the speciation rate
# with time
f.lamb &lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.08, 0.01)
mu_par&lt;-c()
result &lt;- fit_bd(Balaenopteridae,tot_time,f.lamb,f.mu,lamb_par,mu_par,f=1,
                     expo.lamb = TRUE, fix.mu=TRUE)

# Compute the matrix of probabilities                     
prob &lt;- prob_dtt(result, tot_time, 1:tot_time, N0=9, type="crown")

# Check that the sums of probabilities are equal to 1
colSums(prob)

# Plot Diversity through time
plot_prob_dtt(prob)
}
</code></pre>

<hr>
<h2 id='plot_spectR'>
Spectral density plot of a phylogeny.
</h2><span id='topic+plot_spectR'></span>

<h3>Description</h3>

<p>Plot the spectral density of a phylogeny and all eigenvalues ranked in descending order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_spectR(spectR)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_spectR_+3A_spectr">spectR</code></td>
<td>

<p>an object of class 'spectR', output of the 'spectR' function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 2-panel plot with the spectral density profile on the first panel and the eigenvalues ranked in descending order on the second panel
</p>


<h3>Author(s)</h3>

<p>E Lewitus</p>


<h3>References</h3>

<p>Lewitus, E., Morlon, H., Characterizing and comparing phylogenies from their Laplacian spectrum, bioRxiv doi: http://dx.doi.org/10.1101/026476</p>


<h3>See Also</h3>

<p><code><a href="#topic+spectR">spectR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Cetacea)
result &lt;- spectR(Cetacea)
plot_spectR(result)
</code></pre>

<hr>
<h2 id='plot.fit_t.env'>
Plot the phenotypic evolutionary rate through time estimated by the <em>fit_t_env</em> function
</h2><span id='topic+plot.fit_t.env'></span>

<h3>Description</h3>

<p>Plot estimated evolutionary rate as a function of the environmental data and time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'fit_t.env'
plot(x, steps = 100, ...)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.fit_t.env_+3A_x">x</code></td>
<td>

<p>an object of class 'fit_t.env' obtained from a <em>fit_t_env</em> fit.
</p>
</td></tr>
<tr><td><code id="plot.fit_t.env_+3A_steps">steps</code></td>
<td>

<p>the number of steps from the root to the present used to compute the evolutionary rate <code class="reqn">\sigma2</code> through time.
</p>
</td></tr>
<tr><td><code id="plot.fit_t.env_+3A_...">...</code></td>
<td>

<p>further arguments to be passed to <code>plot</code>. See ?<code>plot</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>plot.fit_t.env</code> returns invisibly a list with the following components used in the current plot:
</p>
<table role = "presentation">
<tr><td><code>time_steps</code></td>
<td>
<p>the times steps where the climatic function was evaluated to compute the rate. The number of steps is controlled through the argument <code>steps</code>.</p>
</td></tr>
<tr><td><code>rates</code></td>
<td>
<p>the estimated evolutionary rate through time estimated at each <code>time_steps</code></p>
</td></tr>
</table>


<h3>Note</h3>

<p>All the graphical parameters (see <code><a href="graphics.html#topic+par">par</a></code>) can be passed through (e.g. line type: <code>lty</code>, line width: <code>lwd</code>, color: <code>col</code> ...)
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Science, 114(16): 4183-4188.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lines.fit_t.env">lines.fit_t.env</a></code>, <code><a href="#topic+likelihood_t_env">likelihood_t_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(test){
data(Cetacea)
data(InfTemp)

# Simulate a trait with temperature dependence on the Cetacean tree
set.seed(123)


trait &lt;- sim_t_env(Cetacea, param=c(0.1,0.2), env_data=InfTemp, model="EnvExp", 
					root.value=0, step=0.01, plot=TRUE)


## Fit the Environmental-exponential model

result1=fit_t_env(Cetacea, trait, env_data=InfTemp, scale=TRUE)
plot(result1)

# further options
plot(result1, lty=2, lwd=2, col="red")

}

</code></pre>

<hr>
<h2 id='plot.fit_t.env.ou'>
Plot the phenotypic evolutionary optimum through time estimated by the <em>fit_t_env_ou</em> function
</h2><span id='topic+plot.fit_t.env.ou'></span>

<h3>Description</h3>

<p>Plot estimated evolutionary optimum as a function of the environmental data and time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'fit_t.env.ou'
plot(x, steps = 100, ...)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.fit_t.env.ou_+3A_x">x</code></td>
<td>

<p>an object of class 'fit_t.env.ou' obtained from a <em>fit_t_env_ou</em> fit.
</p>
</td></tr>
<tr><td><code id="plot.fit_t.env.ou_+3A_steps">steps</code></td>
<td>

<p>the number of steps from the root to the present used to compute the optimum <code class="reqn">\theta(t)</code> through time.
</p>
</td></tr>
<tr><td><code id="plot.fit_t.env.ou_+3A_...">...</code></td>
<td>

<p>further arguments to be passed to <code>plot</code>. See ?<code>plot</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>plot.fit_t.env.ou</code> returns invisibly a list with the following components used in the current plot:
</p>
<table role = "presentation">
<tr><td><code>time_steps</code></td>
<td>
<p>the times steps where the climatic function was evaluated to compute the rate. The number of steps is controlled through the argument <code>steps</code>.</p>
</td></tr>
<tr><td><code>values</code></td>
<td>
<p>the estimated optimum values through time estimated at each <code>time_steps</code></p>
</td></tr>
</table>


<h3>Note</h3>

<p>All the graphical parameters (see <code><a href="graphics.html#topic+par">par</a></code>) can be passed through (e.g. line type: <code>lty</code>, line width: <code>lwd</code>, color: <code>col</code> ...)
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Sciences,  114(16): 4183-4188.
</p>
<p>Troyer, E., Betancur-R, R., Hughes, L., Westneat, M., Carnevale, G., White W.T., Pogonoski, J.J., Tyler, J.C., Baldwin, C.C., Orti, G., Brinkworth, A., Clavel, J., Arcila, D., 2022. The impact of paleoclimatic changes on body size evolution in marine fishes. Proceedings of the National Academy of Sciences, 119 (29), e2122486119.
</p>
<p>Goswami, A. &amp; Clavel, J., 2024. Morphological evolution in a time of Phenomics. EcoEvoRxiv, https://doi.org/10.32942/X22G7Q
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lines.fit_t.env">lines.fit_t.env</a></code>, <code><a href="#topic+fit_t_env_ou">fit_t_env_ou</a></code>, <code><a href="#topic+lines.fit_t.env.ou">lines.fit_t.env.ou</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(test){
data(InfTemp)



set.seed(9999) # for reproducibility

# Let's start by simulating a trait under a climatic OU
beta = 0.6           # relationship to the climate curve
sim_theta = 4        # value of the optimum if the relationship to the climate curve is 0
# (this corresponds to an 'intercept' in the linear relationship used below)
sim_sigma2 = 0.025   # variance of the scatter = sigma^2
sim_alpha = 0.36     # alpha value = strength of the OU; quite high here...
delta = 0.001        # time step used for the forward simulations =&gt; here its 1000y steps
tree &lt;- pbtree(n=200, d=0.3) # simulate a bd tree with some extinct lineages
root_age = 60        # height of the root (almost all the Cenozoic here)
tree$edge.length &lt;- root_age*tree$edge.length/max(nodeHeights(tree)) 
# here - for this contrived example - I scale the tree so that the root is at 60 Ma

trait &lt;- sim_t_env_ou(tree, sigma=sqrt(sim_sigma2), alpha=sim_alpha, 
                      theta0=sim_theta, param=beta, env_data=InfTemp, step=0.01, 
                      scale=TRUE, plot=FALSE)

## Fit the Environmental model (default)

result1 &lt;- fit_t_env_ou(phylo = tree, data = trait, env_data =InfTemp,  
                        method = "Nelder-Mead", df=50, scale=TRUE)
plot(result1, lty=2, col="red")


}

</code></pre>

<hr>
<h2 id='Posdef'>
Positive definite symmetric matrices
</h2><span id='topic+Posdef'></span>

<h3>Description</h3>

<p>Generates a positive definite and symmetric matrix with specified eigen-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>

Posdef(p, ev = rexp(p, 1/100))
  
  
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Posdef_+3A_p">p</code></td>
<td>

<p>The dimension of the matrix</p>
</td></tr>
<tr><td><code id="Posdef_+3A_ev">ev</code></td>
<td>

<p>The eigenvalues. If not specified, eigenvalues are taken from an exponential distribution.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Posdef</code> generates random positive definite covariance matrices with specified eigen-values that can be used to simulate multivariate datasets (see Uyeda et al. 2015 - and supplied R codes). 
</p>


<h3>Value</h3>

<p>Returns a symmetric positive-definite matrix with eigen-values = ev.
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Uyeda J.C., Caetano D.S., Pennell M.W. 2015. Comparative Analysis of Principal Components Can be Misleading. Syst. Biol. 64:677-689.
</p>
<p>Clavel, J., Aristide, L., Morlon, H., 2019. A Penalized Likelihood framework for high-dimensional phylogenetic comparative methods and an application to new-world monkeys brain evolution. Syst. Biol. 68:93-116.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GIC.fit_pl.rpanda">GIC.fit_pl.rpanda</a></code>, 
<code><a href="#topic+fit_t_pl">fit_t_pl</a></code>
<code><a href="#topic+phyl.pca_pl">phyl.pca_pl</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(test){
if(require(mvMORPH)){
set.seed(123)
n &lt;- 32 # number of species
p &lt;- 40 # number of traits

tree &lt;- pbtree(n=n) # phylogenetic tree
R &lt;- Posdef(p) # a random symmetric matrix (covariance)
# simulate a dataset
Y &lt;- mvSIM(tree, model="BM1", nsim=1, param=list(sigma=R))

test &lt;- fit_t_pl(Y, tree, model="BM", method="RidgeAlt")
GIC(test)
}
}

</code></pre>

<hr>
<h2 id='prob_dtt'>
Confidence intervals of diversity through time
</h2><span id='topic+prob_dtt'></span>

<h3>Description</h3>

<p>Returns a matrix of probabilities to have 'm' species at a given time 't' with 'n' observed extant species (complete sampling or not) and 's' species at the root of the phylogeny (s=1 if the tree has a stem, otherwise s=2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  prob_dtt(fit.bd, tot_time, time, N0, l=N0, f = l/N0, 
            m = seq(N0), method="simple", lin = FALSE,
           prec = 1000, type = "stem",logged = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prob_dtt_+3A_fit.bd">fit.bd</code></td>
<td>

<p>an object of class 'fit.bd', output of the 'fit_bd' function.
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_tot_time">tot_time</code></td>
<td>

<p>the age of the underlying phylogeny (crown age, or stem age if known). If working with crown ages, tot_time is given by max(node.age(phylo)$ages).
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_time">time</code></td>
<td>

<p>vector of times on which the function calculates the probabilities of 'm' species. The function goes forward in time, so that <code class="reqn">t = 0</code> is the time of the most recent common ancestor.
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_n0">N0</code></td>
<td>

<p>number of extant species. If all extant species are represented in the phylogeny, N0 is given by length(phylo$tip.label).
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_l">l</code></td>
<td>

<p>number of extant species sampled. Default value is N0 (complete sampling).
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_f">f</code></td>
<td>

<p>the fraction of extant species included in the phylogeny, given by l/N0.
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_m">m</code></td>
<td>

<p>a vector of integers for which we want to know the probability of each value. 
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_method">method</code></td>
<td>

<p>reflects which way of computing is choosen. A 'simple' one (quicker) is used when the number of extant species (N0) is known exactly or when the whole phylogeny is sampled (f==1). A 'hard one', much longer, is used when N0 is not known without doubt and f&lt;1. The default value is &quot;simple&quot;&quot; (the other possibility is &quot;hard&quot;)
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_lin">lin</code></td>
<td>

<p>logical: set to TRUE if <code class="reqn">\lambda</code> &amp; <code class="reqn">\mu</code> are fitted with a linear model.
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_prec">prec</code></td>
<td>

<p>precision (number of bits used) of the computation. The default value is 1000.
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_type">type</code></td>
<td>

<p>reflects whether the clade has a stem or not. Options are the default &quot;stem&quot;&quot; and the alternative &quot;crown&quot;, which means the tree starts with two species at time 0.
</p>
</td></tr>
<tr><td><code id="prob_dtt_+3A_logged">logged</code></td>
<td>

<p>logical: set to TRUE to log probabilities and factorials as much as possible (required, except perhaps for very small, young clades).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the sampling fraction is not equal to 1, the function computes with very high numbers. To be sufficiently accurate, the package 'Rmpfr' is used and &quot;prec&quot; is the precision of the computation. Hence, the calculation may take a lot of time. In case of wrong probabilities (negatives or higher than 1 for instance) you should increase the precision.
</p>
<p>If the sampling fraction is equal to 1, the function doesn't need the package 'Rmpfr' and simply uses the log of probabilities and factorials (argument &quot;logged&quot;). Thus, computation is faster. 
</p>
<p>The matrix columns names go backward in time.
</p>


<h3>Value</h3>

<p>Matrix of probabilities to have 'm' species at a given time 't' with 'n' observed extant species (complete sampling or not).
</p>


<h3>Author(s)</h3>

<p>O.Billaud, T.L.Parsons, D.S.Moen, H.Morlon</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record. Proc. Nat. Acad. Sci. 108: 16327-16332.
</p>
<p>Billaud, O., Moen, D. S., Parsons, T. L., Morlon, H. (under review) Estimating Diversity Through Time using Molecular Phylogenies: Old and Species-Poor Frog Families are the Remnants of a Diverse Past. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_bd">fit_bd</a></code>, <code><a href="#topic+plot_dtt">plot_dtt</a></code>, <code><a href="#topic+plot_prob_dtt">plot_prob_dtt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Balaenopteridae)
tot_time&lt;-max(node.age(Balaenopteridae)$ages)

# Fit the pure birth model (no extinction) with exponential variation of the speciation rate
# with time
f.lamb &lt;-function(t,y){y[1] * exp(y[2] * t)}
f.mu&lt;-function(t,y){0}
lamb_par&lt;-c(0.08, 0.01)
mu_par&lt;-c()


if(test){
result &lt;- fit_bd(Balaenopteridae,tot_time,f.lamb,f.mu,lamb_par,mu_par,f=1,
                     expo.lamb = TRUE, fix.mu=TRUE)
                     
# Compute the matrix of probabilities                     
prob &lt;- prob_dtt(result, tot_time, 1:tot_time, N0=9, type="crown")

# Check that the sums of probabilities are equal to 1
colSums(prob)
}
</code></pre>

<hr>
<h2 id='radiolaria'>Radiolaria diversity since the Jurassic</h2><span id='topic+radiolaria'></span>

<h3>Description</h3>

<p>Radiolaria fossil diversity since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sealevel)</code></pre>


<h3>Details</h3>

<p>Radiolaria fossil diversity since the Jurassic compiled from the Neptune Database (Lazarus, 1994) and Paleobiology Database (https://paleobiodb.org/). Diversity curves are estimated at the genus level using shareholder quorum subsampling (Alroy, 2010) at two-million-year bins. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>radiolaria</code></dt><dd><p>a numeric vector corresponding to the estimated ostracod change at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Lazarus, D. (1994) Neptune: A marine micropaleontology database <em>Mathematical Geology 26:817–832</em>
</p>
<p>Alroy, J. (2010) Geographical, environmental and intrinsic biotic controls on Phanerozoic marine diversification: Controls on phanerozoic marine diversification <em>Palaeontology 53:1211–1235</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(radiolaria)
plot(radiolaria)
</code></pre>

<hr>
<h2 id='redalgae'>Red algae diversity since the Jurassic</h2><span id='topic+redalgae'></span>

<h3>Description</h3>

<p>Red algae fossil diversity since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(redalgae)</code></pre>


<h3>Details</h3>

<p>Red algae fossil diversity since the Jurassic compiled from the Neptune Database (Lazarus, 1994) and Paleobiology Database (https://paleobiodb.org/). Diversity curves are estimated at the genus level using shareholder quorum subsampling (Alroy, 2010) at two-million-year bins. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>redalgae</code></dt><dd><p>a numeric vector corresponding to the estimated Red algae change at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Lazarus, D. (1994) Neptune: A marine micropaleontology database <em>Mathematical Geology 26:817–832</em>
</p>
<p>Alroy, J. (2010) Geographical, environmental and intrinsic biotic controls on Phanerozoic marine diversification: Controls on phanerozoic marine diversification <em>Palaeontology 53:1211–1235</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(redalgae)
plot(redalgae)
</code></pre>

<hr>
<h2 id='remove.model'>
Removing a model from shift.estimates output
</h2><span id='topic+remove.model'></span>

<h3>Description</h3>

<p>Allows to remove a model from the model comparisons of shift.estimates output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  remove.model(shift.res, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="remove.model_+3A_shift.res">shift.res</code></td>
<td>

<p>the output resulting from shift.estimates.
</p>
</td></tr>
<tr><td><code id="remove.model_+3A_model">model</code></td>
<td>

<p>character. Specifies the model to remove from the set of model of diversification applied in shift.res.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allow to remove model one at a time. The idea is to remove a model without having to reanalyse the phylogeny and all the combinations of shifts if a model (e.g. BVAR_DVAR) behaves strangely on the studied phylogeny.
</p>


<h3>Value</h3>

<p>the same output resulting from shift.estimates but without the chosen model in model comparisons.
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.estimates">shift.estimates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># loading data
data("shifts_cetacea")

# Removing "BVAR_DCST" model for the example
shifts_cetacea_noBVAR_DCST &lt;- remove.model(shift.res = shifts_cetacea,
                                           model = "BVAR_DCST")

</code></pre>

<hr>
<h2 id='sealevel'>Sea level data since the Jurassic</h2><span id='topic+sealevel'></span>

<h3>Description</h3>

<p>Global sea level change since the Jurassic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sealevel)</code></pre>


<h3>Details</h3>

<p>Eustatic sea level change since the Jurassic calculated by Miller et al. (2005) from satellite measurements, tide gauges, shoreline markers, reefs, atolls, oxygen isotopes,, the flooding history of continental margins, cratons. The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>sea level</code></dt><dd><p>a numeric vector corresponding to the estimated sea level change at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Miller, K.G., Kominz, M.A., Browning, J.V., Wright, J.D., Mountain, G.S., Katz, M.E., Sugarman, P.J., Cramer, B.S., Christie-Blick, N., Pekar, S.F. (2005) The Phanerozoic Record of Global Sea-Level Change <em>Science 310:1293-1298</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sealevel)
plot(sealevel)
</code></pre>

<hr>
<h2 id='shift.estimates'>
Estimating clade-shifts of diversification
</h2><span id='topic+shift.estimates'></span>

<h3>Description</h3>

<p>Applies models of diversification to each part of all combinations of shifts to detect the best combination of subclades and backbone(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  shift.estimates(phylo, data, sampling.fractions, comb.shift,
                  models = c("BCST", "BCST_DCST", "BVAR",
                  "BVAR_DCST", "BCST_DVAR", "BVAR_DVAR"),
                  backbone.option = "crown.shift",
                  multi.backbone = FALSE, np.sub = 4,
                  rate.max = NULL, n.max = NULL, Ncores = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shift.estimates_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="shift.estimates_+3A_data">data</code></td>
<td>

<p>a data.frame containing a database of monophyletic groups for which potential shifts can be investigated. This database should be based on taxonomy, ecology or traits and contain a column named &quot;Species&quot; with species name as in phylo.
</p>
</td></tr>
<tr><td><code id="shift.estimates_+3A_sampling.fractions">sampling.fractions</code></td>
<td>

<p>the output resulting from get.sampling.fractions.
</p>
</td></tr>
<tr><td><code id="shift.estimates_+3A_comb.shift">comb.shift</code></td>
<td>

<p>the output resulting from get.comb.shift.
</p>
</td></tr>
<tr><td><code id="shift.estimates_+3A_models">models</code></td>
<td>

<p>a vector of character that specifies the set of models of diversification to apply. Default is c(&quot;BCST&quot;, &quot;BCST_DCST&quot;, &quot;BVAR&quot;, &quot;BVAR_DCST&quot;, &quot;BCST_DVAR&quot;, &quot;BVAR_DVAR&quot;).
</p>
</td></tr>
<tr><td><code id="shift.estimates_+3A_backbone.option">backbone.option</code></td>
<td>

<p>type of the backbone analysis:
</p>
 
<ul>
<li><p>&quot;stem.shift&quot;: for every shift, the probability of the speciation event at the stem age of the subclade is included in the likelihood of the backbone thanks to the argument spec_times.
</p>
</li>
<li><p>&quot;crown.shift&quot;: for every shift, both the probability of the speciation event at the stem age of the subclade and the probability that the stem of the subclade survives to the crown age are included in the likelihood of the backbone thanks to the argument branch_times.
</p>
</li></ul>

</td></tr>
<tr><td><code id="shift.estimates_+3A_multi.backbone">multi.backbone</code></td>
<td>

<p>can be either FALSE (default), TRUE or &quot;all&quot;: 
</p>

<ul>
<li><p>FALSE: only combinations with simple backbone will be analyzed.
</p>
</li>
<li><p>TRUE: only combinations with multiple backbones will be analyzed.
</p>
</li>
<li><p>&quot;all&quot;: all combinations are analyzed.
</p>
</li></ul>

</td></tr>
<tr><td><code id="shift.estimates_+3A_np.sub">np.sub</code></td>
<td>

<p>Defines the set of models to apply to subclade based on the number of parameters. By default np.sub = 4 and all models from argument models will be applied. If np.sub = 3, the more complex model &quot;BVAR_DVAR&quot; is excluded. If np.sub = 2, the set of models is reduced to &quot;BCST&quot;, &quot;BCST_DCST&quot; and &quot;BVAR&quot; models. np.sub = &quot;no_extinction&quot; only applies &quot;BCST&quot; and &quot;BVAR&quot; models.
</p>
</td></tr>
<tr><td><code id="shift.estimates_+3A_rate.max">rate.max</code></td>
<td>

<p>numeric. Define a maximum value for diversification rate through time.
</p>
</td></tr>
<tr><td><code id="shift.estimates_+3A_n.max">n.max</code></td>
<td>

<p>numeric. Define a maximum value for diversity through time.
</p>
</td></tr>
<tr><td><code id="shift.estimates_+3A_ncores">Ncores</code></td>
<td>

<p>numeric. Define the number of CPU cores to use for parallelizing the computation of combinations.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output for backbone is a list in which each element corresponds to the backbone model comparisons of a combination. This element contains a list with one table of model comparison per backbone.
</p>
<p>We recommand to remove &quot;BVAR_DVAR&quot; model from the models set and to lead the first analysis with multi.backbone = F to limit the number of combination.
</p>
<p>clade.size argument should be the same value for the whole procedure (same than for get.sampling.fraction and get.comb.shift).
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>whole_tree</code></td>
<td>
<p>a data.frame with the model comparison for the whole tree</p>
</td></tr>
<tr><td><code>subclades</code></td>
<td>
<p>a list of dataframes summaryzing the model comparison for all subclades (same format than div.models outputs)</p>
</td></tr>
<tr><td><code>backbones</code></td>
<td>
<p>a list with the model comparison for all backbones (see details)</p>
</td></tr>
<tr><td><code>total</code></td>
<td>
<p>the global comparison of combinations based on AICc</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get.sampling.fractions">get.sampling.fractions</a></code>, <code><a href="#topic+shift.estimates">shift.estimates</a></code>, <code><a href="#topic+paleodiv">paleodiv</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# loading data
data("Cetacea")
data("taxo_cetacea")

# whole procedure
taxo_cetacea_no_genus &lt;- taxo_cetacea[names(taxo_cetacea) != "Genus"]
f_cetacea &lt;- get.sampling.fractions(phylo = Cetacea, lad = FALSE,
                                    data = taxo_cetacea_no_genus,
                                    plot = TRUE, cex = 0.3)

comb.shift_cetacea &lt;- get.comb.shift(phylo = Cetacea,
                                     data = taxo_cetacea_no_genus,
                                     sampling.fractions = f_cetacea,
                                     Ncores = 4)
                                     
shifts_cetacea &lt;- shift.estimates(phylo = Cetacea,
                                  data = taxo_cetacea_no_genus,
                                  sampling.fractions = f_cetacea,
                                  comb.shift = comb.shift_cetacea,
                                  models = c("BCST","BCST_DCST","BVAR",
                                             "BVAR_DCST","BCST_DVAR"),
                                  backbone.option = "crown.shift",
                                  Ncores = 4)
  
</code></pre>

<hr>
<h2 id='shifts_cetacea'>Cetacean shift.estimates results</h2><span id='topic+shifts_cetacea'></span>

<h3>Description</h3>

<p>Results of shift.estimates applyied to Cetaceans</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(shifts_cetacea)</code></pre>


<h3>Details</h3>

<p>This object is the result of shifts.estimates applied to the Cetacean phylogeny as in the example of shift.estimates function.
</p>


<h3>Source</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>References</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans <em>Syst Biol</em> 58:573-585
</p>
<p>Morlon, H., Parsons, T.L., Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(shifts_cetacea)
print(shifts_cetacea)
</code></pre>

<hr>
<h2 id='silica'>Silica data across the Cenozoic</h2><span id='topic+silica'></span>

<h3>Description</h3>

<p>Silica weathering ratio across the Cenozoic</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(silica)</code></pre>


<h3>Details</h3>

<p>Silica weathering ratio across the Cenozoic calculated by Cermeno et al. (2015) using  the lithium isotope record of seawater from Misra and Froelich (2012). The format is a dataframe with the two following variables:
</p>

<dl>
<dt><code>age</code></dt><dd><p>a numeric vector corresponding to the geological age, in Myrs before the present</p>
</dd>
<dt><code>silica weathering ratio</code></dt><dd><p>a numeric vector corresponding to the estimated CO2 at that age</p>
</dd>
</dl>



<h3>References</h3>

<p>Misra, S., Froelich, P.N. (2012) Lithium isotope history of Cenozoic seawater: Changes in
silicate weathering and reverse weathering. <em>Science 335(6070):818–823</em>
</p>
<p>Cermeno, P., Falkowski, P.G., Romero, O.E., Schaller, M.F., Vallina, S.M. (2015) Continental erosion and the Cenozoic rise of marine diatoms <em>Proceedings of the National Academy of Sciences 112:4239-244</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(silica)
plot(silica)
</code></pre>

<hr>
<h2 id='sim_ClaDS'>
Simulation of the ClaDS model
</h2><span id='topic+sim_ClaDS'></span>

<h3>Description</h3>

<p>Simulate a birth-death phyloh-geny with rate shifts happening at speciation events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_ClaDS(lambda_0, mu_0,
          new_lamb_law="lognormal*shift",new_mu_law="turnover",
          condition="time", time_stop = 0, taxa_stop = Inf,
          sigma_lamb=0.1, alpha_lamb=1, lamb_max=1,lamb_min=0,
          sigma_mu=0, alpha_mu=1, mu_min=mu_0,mu_max=mu_0, 
          theta=1,nShiftMax=Inf,
          return_all_extinct=FALSE,prune_extinct=TRUE,
          maxRate=Inf)
                       </code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_ClaDS_+3A_lambda_0">lambda_0</code></td>
<td>

<p>Initial speciation rate.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_mu_0">mu_0</code></td>
<td>

<p>Initial extinction rate, or turnover rate if new_mu_law == &quot;turnover&quot;.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_new_lamb_law">new_lamb_law</code></td>
<td>

<p>Distribution in which the new speciation rates are drawn at a speciation event. See details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_new_mu_law">new_mu_law</code></td>
<td>

<p>Distribution in which the new extinction rates are drawn at a speciation event. See details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_condition">condition</code></td>
<td>

<p>Stoping condition. Can be &quot;time&quot; (the default) or &quot;taxa&quot;.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_time_stop">time_stop</code></td>
<td>

<p>Stoping time if condition == &quot;time&quot;.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_taxa_stop">taxa_stop</code></td>
<td>

<p>Final number of species if condition == &quot;taxa&quot;. 
</p>
<p>If condition == &quot;time&quot;, the process is stoped if the number of species exceeds taxa_stop. This can be usefull for some parametrizations of the model for which the number of species can reach very large number very quickly, leading to computation time and memory issues. To disable this option, use taxa_stop = Inf (the default).
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_sigma_lamb">sigma_lamb</code></td>
<td>

<p>Parameter of the new speciation rates distribution, see details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_alpha_lamb">alpha_lamb</code></td>
<td>

<p>Parameter of the new speciation rates distribution, see details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_lamb_max">lamb_max</code></td>
<td>

<p>Parameter of the new speciation rates distribution, see details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_lamb_min">lamb_min</code></td>
<td>

<p>Parameter of the new speciation rates distribution, see details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_sigma_mu">sigma_mu</code></td>
<td>

<p>Parameter of the new extinction rates distribution, see details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_alpha_mu">alpha_mu</code></td>
<td>

<p>Parameter of the new extinction rates distribution, see details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_mu_min">mu_min</code></td>
<td>

<p>Parameter of the new extinction rates distribution, see details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_mu_max">mu_max</code></td>
<td>

<p>Parameter of the new extinction rates distribution, see details.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_theta">theta</code></td>
<td>

<p>Probability to have a rate shift at speciation. Default to 1.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_nshiftmax">nShiftMax</code></td>
<td>

<p>Maximum number of rate shifts. If nShiftMax &lt; Inf, theta is set to 0 as soon as there has been nShiftMax rate shifts. Set nShiftMax = Inf (the default) to disable this option.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_return_all_extinct">return_all_extinct</code></td>
<td>

<p>Boolean specifying whether the function should return extinct phylogenies. Default to FALSE.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_prune_extinct">prune_extinct</code></td>
<td>

<p>Boolean specifying whether extinct species should be removed from the resulting phylogeny. Default to TRUE.
</p>
</td></tr>
<tr><td><code id="sim_ClaDS_+3A_maxrate">maxRate</code></td>
<td>

<p>The process is stoped if one of the lineage has a speciation rate that exceeds maxRate. This can be usefull for some parametrizations of the model for which the rates can reach very large values, leading to numerical overflows. To disable this option, use maxRate = Inf (the default).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Available options for new_lamb_law are : 
</p>

<ul>
<li><p> &quot;uniform&quot;, the new speciation rates are drawn uniformly in [lamb_min, lamb_max].
</p>
</li>
<li><p> &quot;normal&quot;, the new speciation rates are drawn in a normal distribution with parameters (sigma_lamb^2, parent_lambda), truncated in 0.
</p>
</li>
<li><p> &quot;lognormal&quot;, the new speciation rates are drawn in a lognormal distribution with parameters (sigma_lamb^2, parent_lambda).
</p>
</li>
<li><p> &quot;lognormal*shift&quot;, the new speciation rates are drawn in a lognormal distribution with parameters (sigma_lamb^2, parent_lambda * alpha_lamb). This is the default option as it corresponds to the ClaDS model.
</p>
</li>
<li><p> &quot;lognormal*t&quot;, the new speciation rates are drawn in a lognormal distribution with parameters (sigma_lamb^2 * t^2, parent_lambda), where t is the age of the mother species.
</p>
</li>
<li><p> &quot;logbrownian&quot;, the new speciation rates are drawn in a lognormal distribution with parameters (sigma_lamb^2 * t, parent_lambda), where t is the age of the mother species. This is used to approximate the case where speciation rates are evolving as the log of a brownian motion, as is done in Beaulieu, J. M. and B. C. O'Meara. (2015).
</p>
</li>
<li><p> &quot;normal+shift&quot;, the new speciation rates are drawn in a normal distribution with parameters (sigma_lamb^2, parent_lambda + alpha_lamb), truncated in 0.
</p>
</li>
<li><p> &quot;normal*shift&quot;, the new speciation rates are drawn in a normal distribution with parameters (sigma_lamb^2, parent_lambda * alpha_lamb), truncated in 0.
</p>
</li></ul>

<p>Available options for new_mu_law are : 
</p>

<ul>
<li><p> &quot;uniform&quot;, the new extinction rates are drawn uniformly in [mu_min, mu_max].
</p>
</li>
<li><p> &quot;normal&quot;, the new extinction rates are drawn in a normal distribution with parameters (sigma_mu^2, parent_mu), truncated in 0.
</p>
</li>
<li><p> &quot;lognormal&quot;, the new extinction rates are drawn in a lognormal distribution with parameters (sigma_mu^2, parent_mu).
</p>
</li>
<li><p> &quot;lognormal*shift&quot;, the new extinction rates are drawn in a lognormal distribution with parameters (sigma_mu^2, parent_mu * alpha_mu).
</p>
</li>
<li><p> &quot;normal*t&quot;, the new speciation rates are drawn in a normal distribution with parameters (sigma_lamb^2 * t^2, parent_lambda), where t is the age of the mother species.
</p>
</li>
<li><p> &quot;turnover&quot;, the turnover rate is constant (in that case mu_0 is the turnover rate), so the new extinction rates are mu_0 times the new speciation rates. This is the default option, corresponding to ClaDS2.
</p>
</li></ul>



<h3>Value</h3>

<p>A list with :
</p>
<table role = "presentation">
<tr><td><code>tree</code></td>
<td>
<p>The resulting phylogeny.</p>
</td></tr>
<tr><td><code>times</code></td>
<td>
<p>A vector with the times of all speciation and extinction events.</p>
</td></tr>
<tr><td><code>nblineages</code></td>
<td>
<p>A vector in which nblineages[i] is the number of species in the clade after the event happening at time times[i].</p>
</td></tr>
<tr><td><code>lamb</code></td>
<td>
<p>A vector with all the different speciation rates resulting from the simulation.</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector with all the different extinction rates resulting from the simulation.</p>
</td></tr>
<tr><td><code>rates</code></td>
<td>
<p>A vector of integer mapping the elements of .$lamb and .$mu to the branches of .$tree.</p>
</td></tr>
<tr><td><code>maxRate</code></td>
<td>
<p>A boolean indicating whether the process was ended before reaching the specified stopping criterion because one of the speciation rates exceeded maxRate (see the &quot;arguments&quot; section).</p>
</td></tr>
<tr><td><code>root_length</code></td>
<td>
<p>The time before the first speciation event.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet O., Hartig F. and Morlon H. 2019, A model with many small shifts for estimating species-specific diversificaton rates, <em>Nature Ecology and Evolution</em>, doi 10.1038/s41559-019-0908-0
</p>
<p>Beaulieu, J. M. and B. C. O'Meara. 2015. Extinction can be estimated from moderately sized
molecular phylogenies. <em>Evolution</em> 69:1036-1043.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_ClaDS_phylo">plot_ClaDS_phylo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulation of a ClaDS2 phylogeny
set.seed(1)

obj= sim_ClaDS( lambda_0=0.1,    
                mu_0=0.5,      
                sigma_lamb=0.7,         
                alpha_lamb=0.90,     
                condition="taxa",    
                taxa_stop = 20,    
                prune_extinct = TRUE)  

tree = obj$tree
speciation_rates = obj$lamb[obj$rates]
extinction_rates = obj$mu[obj$rates]

plot_ClaDS_phylo(tree,speciation_rates)


# Simulation of a phylogeny with constant extinction rate and speciation 
# rates evolving as a logbrownian
set.seed(4321)

obj= sim_ClaDS( lambda_0=0.1,    
                mu_0=0.2,    
                new_mu_law = "uniform",
                new_lamb_law = "logbrownian",
                sigma_lamb=0.4,         
                condition="taxa",    
                taxa_stop = 20,    
                prune_extinct = FALSE)  

tree = obj$tree
speciation_rates = obj$lamb[obj$rates]
extinction_rates = obj$mu[obj$rates]

oldpar &lt;- par(no.readonly = TRUE)
par(mar=c(1,1,0,0))
plot_ClaDS_phylo(tree,speciation_rates)
par(oldpar) # restore the old par


# Simulation of a phylogeny with constant extinction rate and at most one shift
# in speciation rates
set.seed(1221)

obj= sim_ClaDS( lambda_0=0.1,    
                mu_0=0.05,    
                new_mu_law = "uniform",
                new_lamb_law = "uniform",
                lamb_max = 0.5, lamb_min = 0,     
                theta = 0.1, nShiftMax = 1,
                condition="taxa",    
                taxa_stop = 100,    
                prune_extinct = TRUE)  

tree = obj$tree
speciation_rates = obj$lamb[obj$rates]
extinction_rates = obj$mu[obj$rates]

plot_ClaDS_phylo(tree,speciation_rates)
</code></pre>

<hr>
<h2 id='sim_env_bd'>
Simulate birth-death tree dependent on an environmental curve
</h2><span id='topic+sim_env_bd'></span>

<h3>Description</h3>

<p>Simulates a birth-death tree (starting with one lineage) with speciation and/or extinction rate that varies as a function of an input environmental curve. Notations follow Morlon et al. PNAS 2011 and Condamine et al. ELE 2013.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_env_bd(env_data, f.lamb, f.mu, lamb_par, mu_par, df=NULL, time.stop=0, 
			return.all.extinct=TRUE, prune.extinct=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_env_bd_+3A_env_data">env_data</code></td>
<td>

<p>environmental data, given as a data frame with two columns. The first column is time, the second column is the environmental data (temperature for instance).
</p>
</td></tr>
<tr><td><code id="sim_env_bd_+3A_time.stop">time.stop</code></td>
<td>

<p>the age of the phylogeny.
</p>
</td></tr>
<tr><td><code id="sim_env_bd_+3A_f.lamb">f.lamb</code></td>
<td>

<p>a function specifying the hypothesized functional form of the variation of the speciation rate <code class="reqn">\lambda</code> with time and the environmental variable. Any functional form may be used.
This function has three arguments: the first argument is time; the second argument is the environmental variable; the third arguement is a numeric vector of the parameters controlling the time and environmental variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="sim_env_bd_+3A_f.mu">f.mu</code></td>
<td>

<p>a function specifying the hypothesized functional form of the variation of the extinction rate <code class="reqn">\mu</code> with time and the environmental variable. Any functional form may be used.
This function has three arguments: the first argument is time; the second argument is the environmental variable; the second argument is a numeric vector of the parameters controlling the time and environmental variation (to be estimated).
</p>
</td></tr>
<tr><td><code id="sim_env_bd_+3A_lamb_par">lamb_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.lamb to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model with constant speciation rate (for example), lamb_par should be a vector of length 1. Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="sim_env_bd_+3A_mu_par">mu_par</code></td>
<td>

<p>a numeric vector of initial values for the parameters of f.mu to be estimated (these values are used by the optimization algorithm). The length of this vector is used to compute the total number of parameters in the model, so to fit a model without extinction (for example), mu_par should be empty (vector of length 0). Otherwise aic values will be wrong.
</p>
</td></tr>
<tr><td><code id="sim_env_bd_+3A_df">df</code></td>
<td>

<p>the degree of freedom to use to define the spline. As a default, smooth.spline(env_data[,1], env_data[,2])$df is used. See <em>sm.spline</em> for details.  
</p>
</td></tr>
<tr><td><code id="sim_env_bd_+3A_return.all.extinct">return.all.extinct</code></td>
<td>

<p>return all extinction lineages in simulated tree.  
</p>
</td></tr>
<tr><td><code id="sim_env_bd_+3A_prune.extinct">prune.extinct</code></td>
<td>

<p>prune extinct lineages in simulated tree.  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the f.lamb and f.mu functions, time runs from the present to the past.
</p>


<h3>Value</h3>

<p>a list with the following components
</p>
<table role = "presentation">
<tr><td><code>tree</code></td>
<td>
<p>the simulated tree with number tips</p>
</td></tr>
<tr><td><code>times</code></td>
<td>
<p>the times of speciation events starting from the past</p>
</td></tr>
<tr><td><code>nblineages</code></td>
<td>
<p>the labels of surviving lineages and total number of surviving lineages</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The speed of convergence of the fit might depend on the degree of freedom chosen to define the spline. 
</p>


<h3>Author(s)</h3>

<p>E Lewitus and H Morlon</p>


<h3>References</h3>

<p>Morlon, H., Parsons, T.L. and Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Condamine, F.L., Rolland, J., and Morlon, H. (2013) Macroevolutionary perspectives to environmental change, Eco Lett 16: 72-85
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_env">fit_env</a></code>, <code><a href="#topic+fit_bd">fit_bd</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(InfTemp)
dof&lt;-smooth.spline(InfTemp[,1], InfTemp[,2])$df
# Simulates a tree with lambda varying as an exponential function of temperature
# and mu fixed to 0 (no extinction).  Here t stands for time and x for temperature.
f.lamb &lt;-function(t,x,y){y[1] * exp(y[2] * x)}
f.mu&lt;-function(t,x,y){0}
lamb_par&lt;-c(0.10, 0.01)
mu_par&lt;-c()
result_exp &lt;- sim_env_bd(InfTemp,f.lamb,f.mu,lamb_par,mu_par,time.stop=10)

</code></pre>

<hr>
<h2 id='sim_MCBD'>
Simulation of macroevolutionary diversification under the integrated model described in Aristide &amp; Morlon 2019
</h2><span id='topic+sim_MCBD'></span>

<h3>Description</h3>

<p>Simulates the joint diversification of species and a continuous trait, where changes in both dimensions are interlinked through competitive interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_MCBD(pars, root.value = 0, age.max = 50, step.size = 0.01, bounds = c(-Inf,Inf),
         plot = TRUE, ylims=NULL, full.sim = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_MCBD_+3A_pars">pars</code></td>
<td>

<p>Vector of simulation parameters:
</p>
<p><code>pars[1]</code> corresponds to <code class="reqn">lambda1</code>, the speciation intitation rate
</p>
<p><code>pars[2]</code> corresponds to <code class="reqn">tau0</code>, the basal speciation completion rate
</p>
<p><code>pars[3]</code> corresponds to <code class="reqn">beta</code>, the effect of trait differences on the speciation completion rate
</p>
<p><code>pars[4]</code> corresponds to <code class="reqn">mu0</code>, the competitive extinction parameter for good species
</p>
<p><code>pars[5]</code> corresponds to <code class="reqn">mubg</code>, the background good species extinction rate
</p>
<p><code>pars[6]</code> corresponds to <code class="reqn">mui0</code>, the competitive extinction parameter for incipient species
</p>
<p><code>pars[7]</code> corresponds to <code class="reqn">muibg</code>, the background incipient species extinction rate
</p>
<p><code>pars[8]</code> corresponds to <code class="reqn">alpha1</code>, the competition effect on extinction (competition strength)
</p>
<p><code>pars[9]</code> corresponds to <code class="reqn">alpha1</code>, the competition effect on trait evolution (competition strength)
</p>
<p><code>pars[10]</code> corresponds to <code class="reqn">sig2</code>, the variance (rate) of the Brownian motion
</p>
<p><code>pars[11]</code> corresponds to <code class="reqn">m</code>, the relative contribution of character displacement (competition) with respect to stochastic (brownian) evolution
</p>
</td></tr>
<tr><td><code id="sim_MCBD_+3A_root.value">root.value</code></td>
<td>
<p>the starting trait value</p>
</td></tr>
<tr><td><code id="sim_MCBD_+3A_age.max">age.max</code></td>
<td>
<p>maximum time for the simulation (if the process doesn't go extinct)</p>
</td></tr>
<tr><td><code id="sim_MCBD_+3A_step.size">step.size</code></td>
<td>
<p>size of each simulation step</p>
</td></tr>
<tr><td><code id="sim_MCBD_+3A_bounds">bounds</code></td>
<td>
<p>lower and upper value for bounds in trait space</p>
</td></tr>
<tr><td><code id="sim_MCBD_+3A_plot">plot</code></td>
<td>
<p>logical indicating wether to plot the simulation</p>
</td></tr>
<tr><td><code id="sim_MCBD_+3A_ylims">ylims</code></td>
<td>
<p>y axis (trait values) limits for the simulation plot</p>
</td></tr>
<tr><td><code id="sim_MCBD_+3A_full.sim">full.sim</code></td>
<td>
<p>logical indicating wether to return the full simulation (see details)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It might be difficult to find parameter combinations that are sensitive. It is recommended to use the parameter settings of the examples as a staring point and from there modify them to understand the behaviour of the model.
If trees produced are too big, simulation can become too slow to ever finish. 
</p>


<h3>Value</h3>

<p>returns a list with the following elements:
</p>
<p><b>all</b> contains the complete tree of the process (extant and extinct good and incipient lineages) and trait values for each tip in the tree
</p>
<p><b>gsp_fossil</b> contains the extant and extinct good species tree and trait values for each tip in the tree
</p>
<p><b>gsp_extant</b> contains the reconstructed (extant only) good species tree and trait values for each tip in the tree
</p>
<p>If full.sim = TRUE, two additional elements are returned inside <b>all</b>:
</p>
<p>note: both elements are used internally to keep track of the simulation and are dynamically updated, so returned elements only reflect the last state
</p>
<p><b>lin_mat</b> a matrix with information about the diversification process. Each row represents a new lineage in the process with the following elements:
- Parental node, descendent node (0 if a tip), starting time, ending time, status at end (extinct(-2); incipient(-1); good(1)), speciation completion or extinction time; speciation completion time (NA if still incipient).
</p>
<p><b>trait_mat</b> a list with trait values for each lineage at each time step throghout the simulation. Each element is a vector composed of the following:
Lineage number (same as row number in lin_mat), status (as in lin_mat), sister lineage number, trait values (NA if lineage didn't exist yet at that time step)
</p>


<h3>Author(s)</h3>

<p>Leandro Aristide (leandroaristi@gmail.com)
</p>


<h3>References</h3>

<p>Aristide, L., and Morlon, H. 2019. Understanding the effect of competition during evolutionary radiations: an integrated model of phenotypic and species diversification
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lambda1 = 0.25
tau0 = 0.01
beta = 0.6
mu0 = 0.5
mubg = 0.01
mui0 = 0.8
muibg = 0.02
alpha1 = alpha2 = 0.04
sig2 = 0.5
m = 20

pars &lt;- c(lambda1, tau0, beta, mu0, mubg,mui0, muibg, alpha1, alpha2, sig2, m)


if(test){

#1000 steps, unbounded
res &lt;- sim_MCBD(pars, age.max=10, step.size=0.01) 

#asymmetric bounds
res &lt;- sim_MCBD(pars, age.max=10, step.size=0.01, bounds=c(-10,Inf)) 

#only deterministic component
pars &lt;- c(lambda1, tau0, beta, mu0, mubg, mui0, muibg, alpha1, alpha2, sig2=0, m)
res &lt;- sim_MCBD(pars, age.max=10)

plot(res$gsp_extant$tree)

}

</code></pre>

<hr>
<h2 id='sim_sgd'>
Algorithm for simulating a phylogenetic tree under the SGD model
</h2><span id='topic+sim_sgd'></span>

<h3>Description</h3>

<p>Simulates a phylogeny arising from the SGD model with exponentially increasing metapopulation size. Notations follow Manceau et al. (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_sgd(tau, b, d, nu)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_sgd_+3A_tau">tau</code></td>
<td>

<p>the simulation time, which corresponds to the length of the phylogeny
</p>
</td></tr>
<tr><td><code id="sim_sgd_+3A_b">b</code></td>
<td>

<p>the (constant) per-individual birth rate
</p>
</td></tr>
<tr><td><code id="sim_sgd_+3A_d">d</code></td>
<td>

<p>the (constant) per-individual death rate
</p>
</td></tr>
<tr><td><code id="sim_sgd_+3A_nu">nu</code></td>
<td>

<p>the (constant) per-individual mutation rate
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a phylogenetic tree of class &quot;phylo&quot; (see ape documentation)
</p>


<h3>Author(s)</h3>

<p>M Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2015) Phylogenies support out-of-equilibrium models of biodiversity Ecology Letters 18: 347-356
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tau &lt;- 10
b &lt;- 1e6
d &lt;- b-0.5
nu &lt;- 0.6
tree &lt;- sim_sgd(tau,b,d,nu)
plot(tree)
</code></pre>

<hr>
<h2 id='sim_t_comp'>
Recursive simulation (root-to-tip) of competition models 
</h2><span id='topic+sim_t_comp'></span>

<h3>Description</h3>

<p>Simulates datasets for a given phylogeny under matching competition (MC), diversity dependent linear (DDlin), or diversity dependent exponential (DDexp) models of trait evolution. Simulations are carried out from the root to the tip of the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
sim_t_comp(phylo,pars,root.value,Nsegments=1000,model="MC,DDexp,DDlin")

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_t_comp_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="sim_t_comp_+3A_pars">pars</code></td>
<td>

<p>a vector containing the two parameters for the chosen model; all models require <code>sig2</code>, and additionally, the MC model requires <code>S</code>,
specifying the level of competition (larger negative values correspond to higher levels of competition), the DDlin model 
requires <code>b</code> and DDexp require <code>r</code>, the slope parameters (negative in cases of decline in evolutionary 
rates with increasing diversity). <code>sig2</code> must be listed first.
</p>
</td></tr>
<tr><td><code id="sim_t_comp_+3A_root.value">root.value</code></td>
<td>

<p>a number specifying the trait value for the ancestor
</p>
</td></tr>
<tr><td><code id="sim_t_comp_+3A_nsegments">Nsegments</code></td>
<td>

<p>a value specifying the total number of time segments to simulate across for the phylogeny (see Details)
</p>
</td></tr>
<tr><td><code id="sim_t_comp_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"MC"</code> is the matching competition model of Nuismer &amp; Harmon 2014, <code>"DDlin"</code> is the diversity-dependent linear model, and <code>"DDexp"</code> is the diversity-dependent exponential model of Weir &amp; Mursleen 2013.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adjusting <code>Nsegments</code> will impact the length of time the simulations take. 
The length of each segment (<code>max(nodeHeights(phylo))/Nsegments</code>) 
should be much smaller than the smallest branch (<code>min(phylo$edge.length)</code>).
</p>


<h3>Value</h3>

<p>a named vector with simulated trait values for <code class="reqn">n</code> species in the phylogeny
</p>


<h3>Author(s)</h3>

<p>J Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a></p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Nuismer, S. &amp; Harmon, L. 2015. Predicting rates of interspecific interaction from phylogenetic trees. <em>Ecology Letters</em> 18:17-27.
</p>
<p>Weir, J. &amp; Mursleen, S. 2012. Diversity-dependent cladogenesis and trait evolution in the adaptive radiation of the auks (Aves: Alcidae). <em>Evolution</em> 67:403-416.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Cetacea)


# Simulate data under the matching competition model
MC.data&lt;-sim_t_comp(Cetacea,pars=c(sig2=0.01,S=-0.1),root.value=0,Nsegments=1000,model="MC")

# Simulate data under the diversity dependent linear model
DDlin.data&lt;-sim_t_comp(Cetacea,pars=c(sig2=0.01,b=-0.0001),root.value=0,Nsegments=1000,
	model="DDlin")

# Simulate data under the diversity dependent linear model
DDexp.data&lt;-sim_t_comp(Cetacea,pars=c(sig2=0.01,r=-0.01),root.value=0,Nsegments=1000,model="DDexp")


</code></pre>

<hr>
<h2 id='sim_t_env'>
Recursive simulation (root-to-tip) of the environmental model 
</h2><span id='topic+sim_t_env'></span>

<h3>Description</h3>

<p>Simulates datasets for a given phylogeny under the environmental model (see ?fit_t_env)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
sim_t_env(phylo, param, env_data, model, root.value=0, step=0.001, plot=FALSE, ...)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_t_env_+3A_phylo">phylo</code></td>
<td>

<p>An object of class 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="sim_t_env_+3A_param">param</code></td>
<td>

<p>A numeric vector of parameters for the user-defined climatic model. For the <em>EnvExp</em> and <em>EnvLin</em>, there is only two parameters. The first is sigma and the second beta.
</p>
</td></tr>
<tr><td><code id="sim_t_env_+3A_env_data">env_data</code></td>
<td>

<p>Environmental data, given as a time continuous function (see, e.g. splinefun) or a data frame with two columns. The first column is time, the second column is the environmental data (temperature for instance).
</p>
</td></tr>
<tr><td><code id="sim_t_env_+3A_model">model</code></td>
<td>

<p>The model describing the functional form of variation of the evolutionary rate <code class="reqn">\sigma^2</code> with time and the environmental variable. Default models are &quot;EnvExp&quot; and &quot;EnvLin&quot; (see details). An user defined function of any functional form can be used (forward in time).
This function has three arguments: the first argument is time; the second argument is the environmental variable; the third argument is a numeric vector of the parameters controlling the time and environmental variation (to be estimated). See the example below.
</p>
</td></tr>
<tr><td><code id="sim_t_env_+3A_root.value">root.value</code></td>
<td>

<p>A number specifying the trait value for the ancestor
</p>
</td></tr>
<tr><td><code id="sim_t_env_+3A_step">step</code></td>
<td>

<p>This argument describe the length of the segments to simulate across for the phylogeny. The smaller is the segment, the greater is the accuracy of the simulation at the expense of the computation time. 
</p>
</td></tr>
<tr><td><code id="sim_t_env_+3A_plot">plot</code></td>
<td>

<p>If TRUE, the simulated process is plotted.
</p>
</td></tr>
<tr><td><code id="sim_t_env_+3A_...">...</code></td>
<td>

<p>Arguments to be passed through. For instance, &quot;col&quot; for plot=TRUE. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The users defined function is simulated forward in time i.e.: from the root to the tips. The speed of the simulations might depend on the value used for the &quot;step&quot; argument. 
It's possible to estimate the traits with the MLE from another fitted object (see the example below).
</p>


<h3>Value</h3>

<p>A named vector with simulated trait values for <code class="reqn">n</code> species in the phylogeny
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Science,  114(16): 4183-4188.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.fit_t.env">plot.fit_t.env</a></code>, 
<code><a href="#topic+likelihood_t_env">likelihood_t_env</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


if(test){
data(Cetacea)
data(InfTemp)


set.seed(123)
# define the parameters
param &lt;- c(0.1, -0.5)
# define the environmental function
my_fun &lt;- function(t, env, param){ param[1]*exp(param[2]*env(t))}

# simulate the trait
trait &lt;- sim_t_env(Cetacea, param=param, env_data=InfTemp, model=my_fun, root.value=0,
                    step=0.001, plot=TRUE)

# fit the model to the simulated trait.
fit &lt;- fit_t_env(Cetacea, trait, env_data=InfTemp, model=my_fun, param=c(0.1,0))
fit

# Then use the results from the previous fit to simulate a new dataset
trait2 &lt;- sim_t_env(Cetacea, param=fit, step=0.001, plot=TRUE)
fit2 &lt;- fit_t_env(Cetacea, trait2, env_data=InfTemp, model=my_fun, param=c(0.1,0))
fit2

# When providing the environmental function:
if(require(pspline)){
spline_result &lt;- sm.spline(x=InfTemp[,1],y=InfTemp[,2], df=50)
env_func &lt;- function(t){predict(spline_result,t)}
t&lt;-unique(InfTemp[,1])

# We build the interpolated smoothing spline function
env_data&lt;-splinefun(t,env_func(t))

# provide the environmental function to simulate the traits
trait3 &lt;- sim_t_env(Cetacea, param=param, env_data=env_data, model=my_fun,
                     root.value=0, step=0.001, plot=TRUE)
fit3 &lt;- fit_t_env(Cetacea, trait3, env_data=InfTemp, model=my_fun, param=c(0.1,0))
fit3
}
}

  
</code></pre>

<hr>
<h2 id='sim_t_env_ou'>
Recursive simulation (root-to-tip) of the OU environmental model 
</h2><span id='topic+sim_t_env_ou'></span>

<h3>Description</h3>

<p>Simulates datasets for a given phylogeny under the OU environmental model (see ?fit_t_env_ou)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
sim_t_env_ou(phylo, param, env_data, model, step=0.01, 
              plot=FALSE, sigma, alpha, theta0, ...)


</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_t_env_ou_+3A_phylo">phylo</code></td>
<td>

<p>An object of class 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_param">param</code></td>
<td>

<p>A numeric vector of parameters for the user-defined climatic model. For the OU-environmental model, there is only one parameters (beta). If a model fit object of class 'fit_t_env.ou' is provided, the ML parameters are used to generate new datasets.
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_env_data">env_data</code></td>
<td>

<p>Environmental data, given as a time continuous function (see, e.g. splinefun) or a data frame with two columns. The first column is time, the second column is the environmental data (temperature for instance).
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_model">model</code></td>
<td>

<p>The model describing the functional form of variation of the evolutionary trajectory of the optimum &quot;theta(t)&quot; with time and the environmental variable (see details for default model). An user defined function of any functional form can be used (forward in time).
This function has four arguments: the first argument is time; the second argument is the environmental variable; the third argument is a numeric vector of the parameters controlling the time and environmental variation (to be estimated), and the fourth is the theta_0 value. See the example below.
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_step">step</code></td>
<td>

<p>This argument describe the length of the segments to simulate across for the phylogeny. The smaller is the segment, the greater is the accuracy of the simulation at the expense of the computation time. 
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_plot">plot</code></td>
<td>

<p>If TRUE, the simulated process is plotted.
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_sigma">sigma</code></td>
<td>

<p>The &quot;sigma&quot; parameter of the OU process.
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_alpha">alpha</code></td>
<td>

<p>The &quot;alpha&quot; parameter of the OU process.
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_theta0">theta0</code></td>
<td>

<p>The &quot;theta&quot; parameter at the root of the tree (t=0).
</p>
</td></tr>
<tr><td><code id="sim_t_env_ou_+3A_...">...</code></td>
<td>

<p>Arguments to be passed through. For instance, &quot;col&quot; for plot=TRUE. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The users defined function is simulated forward in time i.e.: from the root to the tips. The speed of the simulations might depend on the value used for the &quot;step&quot; argument. 
It's possible to estimate the traits with the MLE from another fitted object (see the example below).
</p>


<h3>Value</h3>

<p>A named vector with simulated trait values for <code class="reqn">n</code> species in the phylogeny
</p>


<h3>Author(s)</h3>

<p>J. Clavel</p>


<h3>References</h3>

<p>Clavel, J. &amp; Morlon, H., 2017. Accelerated body size evolution during cold climatic periods in the Cenozoic. Proceedings of the National Academy of Sciences,  114(16): 4183-4188.
</p>
<p>Troyer, E., Betancur-R, R., Hughes, L., Westneat, M., Carnevale, G., White W.T., Pogonoski, J.J., Tyler, J.C., Baldwin, C.C., Orti, G., Brinkworth, A., Clavel, J., Arcila, D., 2022. The impact of paleoclimatic changes on body size evolution in marine fishes. Proceedings of the National Academy of Sciences, 119 (29), e2122486119.
</p>
<p>Goswami, A. &amp; Clavel, J., 2024. Morphological evolution in a time of Phenomics. EcoEvoRxiv, https://doi.org/10.32942/X22G7Q
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.fit_t.env">plot.fit_t.env</a></code>, 
<code><a href="#topic+fit_t_env">fit_t_env</a></code>,
<code><a href="#topic+fit_t_env_ou">fit_t_env_ou</a></code>,
<code><a href="#topic+plot.fit_t.env.ou">plot.fit_t.env.ou</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


if(test){

data(InfTemp)
set.seed(9999) # for reproducibility

# Let's start by simulating a trait under a climatic OU
beta = 0.6           # relationship to the climate curve
sim_theta = 4        # value of the optimum if the relationship to the climate 
# curve is 0 (this corresponds to an 'intercept' in the linear relationship used below)
sim_sigma2 = 0.025   # variance of the scatter = sigma^2
sim_alpha = 0.36     # alpha value = strength of the OU; quite high here...
delta = 0.001        # time step used for the forward simulations =&gt; here its 1000y steps
tree &lt;- pbtree(n=200, d=0.3) # simulate a bd tree with some extinct lineages
root_age = 60        # height of the root (almost all the Cenozoic here)
tree$edge.length &lt;- root_age*tree$edge.length/max(nodeHeights(tree)) 
# here - for this contrived example - I scale the tree so that the root is at 60 Ma

# define a model - here we replicate the default model used in fit_t_env_ou
my_model &lt;- function(t, env, param, theta0) theta0 + param[1]*env(t)

# simulate the traits
trait &lt;- sim_t_env_ou(tree, sigma=sqrt(sim_sigma2), alpha=sim_alpha,
                      theta0=sim_theta, param=beta, model=my_model,
                      env_data=InfTemp, step=0.01, scale=TRUE, plot=TRUE)

## Fit the Environmental model (default)

result_fit &lt;- fit_t_env_ou(phylo = tree, data = trait, env_data =InfTemp,  
                          method = "Nelder-Mead", df=50, scale=TRUE)
plot(result_fit)


# We can also use the results from the previous fit to simulate a new dataset
trait2 &lt;- sim_t_env_ou(tree, param=result_fit, step=0.001, plot=TRUE)

result_fit2 &lt;- fit_t_env_ou(phylo = tree, data = trait2, env_data =InfTemp, 
                            method = "Nelder-Mead", df=50, scale=TRUE)
result_fit2
}


</code></pre>

<hr>
<h2 id='sim_t_tworegime'>
Recursive simulation (root-to-tip) of two-regime models 
</h2><span id='topic+sim_t_tworegime'></span>

<h3>Description</h3>

<p>Simulates datasets for a given phylogeny under two-regime matching competition (MC), diversity dependent linear (DDlin), diversity dependent exponential (DDexp), or early burst (EB) models of trait evolution. Simulations are carried out from the root to the tip of the tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
sim_t_tworegime(regime.map, pars, root.value, Nsegments=2500, 
                model=c("MC","DDexp","DDlin","EB"),
	            	verbose=TRUE, rnd=6)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_t_tworegime_+3A_regime.map">regime.map</code></td>
<td>

<p>a stochastic map of the two regimes stored as a simmap object output from <code><a href="phytools.html#topic+make.simmap">make.simmap</a></code>
</p>
</td></tr>
<tr><td><code id="sim_t_tworegime_+3A_pars">pars</code></td>
<td>

<p>a vector containing the three parameters for the chosen model; all models require <code>sig2</code>, and additionally, the MC model requires <code>S1</code> and <code>S2</code>,
specifying the level of competition in regime 1 and 2, respectively (larger negative values correspond to higher levels of competition), the DDlin model 
requires <code>b1</code> and <code>b2</code>, the DDexp model requires <code>r1</code>, the slope parameters (negative in cases of decline in evolutionary 
rates with increasing diversity). <code>sig2</code> must be listed first.
</p>
</td></tr>
<tr><td><code id="sim_t_tworegime_+3A_root.value">root.value</code></td>
<td>

<p>a number specifying the trait value for the ancestor
</p>
</td></tr>
<tr><td><code id="sim_t_tworegime_+3A_nsegments">Nsegments</code></td>
<td>

<p>a value specifying the total number of time segments to simulate across for the phylogeny (see Details)
</p>
</td></tr>
<tr><td><code id="sim_t_tworegime_+3A_model">model</code></td>
<td>

<p>model chosen to fit trait data, <code>"MC"</code> is the matching competition model, <code>"DDlin"</code> is the diversity-dependent linear model,  <code>"DDexp"</code> is the diversity-dependent exponential model, and <code>"EB"</code> is the early burst model.
</p>
</td></tr>
<tr><td><code id="sim_t_tworegime_+3A_verbose">verbose</code></td>
<td>

<p>if <code>TRUE</code>, prints the identity of regimes corresponding to each parameter value
</p>
</td></tr>
<tr><td><code id="sim_t_tworegime_+3A_rnd">rnd</code></td>
<td>

<p>number of digits to round timings to (see <code><a href="base.html#topic+round">round</a></code> (see Details)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adjusting <code>Nsegments</code> will impact the length of time the simulations take. 
The length of each segment (<code>max(nodeHeights(phylo))/Nsegments</code>) 
should be much smaller than the smallest branch (<code>min(phylo$edge.length)</code>).
</p>
<p>Adjusting <code>rnd</code> may help if function crashes.
</p>


<h3>Value</h3>

<p>a named vector with simulated trait values for <code class="reqn">n</code> species in the phylogeny
</p>


<h3>Author(s)</h3>

<p>J Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a></p>


<h3>References</h3>

<p>Drury, J., Clavel, J., Manceau, M., and Morlon, H. 2016. Estimating the effect of competition on trait evolution using maximum likelihood inference. <em>Systematic Biology</em> doi 10.1093/sysbio/syw020
</p>
<p>Nuismer, S. &amp; Harmon, L. 2015. Predicting rates of interspecific interaction from phylogenetic trees. <em>Ecology Letters</em> 18:17-27.
</p>
<p>Weir, J. &amp; Mursleen, S. 2012. Diversity-dependent cladogenesis and trait evolution in the adaptive radiation of the auks (Aves: Alcidae). <em>Evolution</em> 67:403-416.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_t_comp">fit_t_comp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Cetacea_clades)



# Simulate data under the matching competition model
MC_tworegime.data&lt;-sim_t_tworegime(Cetacea_clades,pars=c(sig2=0.01,S1=-0.1,S2=-0.01),
	root.value=0,Nsegments=1000,model="MC")

# Simulate data under the diversity dependent linear model
DDlin_tworegime.data&lt;-sim_t_tworegime(Cetacea_clades,pars=c(sig2=0.01,b1=-0.0001,b2=-0.000001),
	root.value=0,Nsegments=1000,model="DDlin")

# Simulate data under the diversity dependent linear model
DDexp_tworegime.data&lt;-sim_t_tworegime(Cetacea_clades,pars=c(sig2=0.01,r1=-0.01,r2=-0.02),
	root.value=0,Nsegments=1000,model="DDexp")

# Simulate data under the diversity dependent linear model
EB.data_tworegime&lt;-sim_t_tworegime(Cetacea_clades,pars=c(sig2=0.01,r1=-0.01,r2=-0.02),
	root.value=0,Nsegments=1000,model="EB")




</code></pre>

<hr>
<h2 id='sim.BipartiteEvol'>
Simulation of the BipartiteEvol model</h2><span id='topic+sim.BipartiteEvol'></span>

<h3>Description</h3>

<p>Simulateof the BipartiteEvol model from Maliet et al. (2020)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.BipartiteEvol(nx, ny = nx, NG, dSpace = Inf, D = 1, muP,
muH, alphaP = 0, alphaH = 0, iniP = 0, iniH = 0, nP = 1, nH = 1, 
rP = 1, rH = 1, effect = 1, verbose = 100, thin = 1, P = NULL, H = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim.BipartiteEvol_+3A_nx">nx</code></td>
<td>

<p>Size of the grid (the grid has size nx * ny)
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_ny">ny</code></td>
<td>

<p>Size of the grid (default to nx, the grid has size nx * ny)
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_ng">NG</code></td>
<td>

<p>Number of time step the model is run
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_dspace">dSpace</code></td>
<td>

<p>Size of the dispersal kernel (default to Inf, meaning there are no restrictions on dispersion)</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_d">D</code></td>
<td>

<p>Dimention of the trait space (default to 3)</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_mup">muP</code></td>
<td>

<p>Mutation probability at reproduction for the individuals of clade P</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_muh">muH</code></td>
<td>

<p>Mutation probability at reproduction for the individuals of clade H</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_alphap">alphaP</code></td>
<td>

<p>alpha parameter for clade P (1/alpha is the niche width)</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_alphah">alphaH</code></td>
<td>

<p>alpha parameter for clade H (1/alpha is the niche width)</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_inip">iniP</code></td>
<td>

<p>Initial trait value for the individuals in clade P</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_inih">iniH</code></td>
<td>

<p>Initial trait value for the individuals in clade P</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_np">nP</code></td>
<td>

<p>Number of individuals of clade P killed at each time step
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_nh">nH</code></td>
<td>

<p>Number of individuals of clade H killed at each time step
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_rp">rP</code></td>
<td>

<p>r parameter for clade P (r is the ratio between the fitness maximum and minimum)</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_rh">rH</code></td>
<td>

<p>r parameter for clade H (r is the ratio between the fitness maximum and minimum)
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_effect">effect</code></td>
<td>

<p>Standard deviation of the trait mutation kernel
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_verbose">verbose</code></td>
<td>

<p>if TRUE, enables printing of messages</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_thin">thin</code></td>
<td>

<p>The number of iterations between two recording of the state of the model (default to 1)
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_p">P</code></td>
<td>

<p>Optional, used to continue one precedent run: traits of the individuals of clade P at the end of the precedent run
</p>
</td></tr>
<tr><td><code id="sim.BipartiteEvol_+3A_h">H</code></td>
<td>

<p>Optional, used to continue one precedent run: traits of the individuals of clade H at the end of the precedent run
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with 
</p>
<table role = "presentation">
<tr><td><code>Pgenealogy</code></td>
<td>
<p>The genalogy of clade P</p>
</td></tr>
<tr><td><code>Hgenealogy</code></td>
<td>
<p>The genalogy of clade H</p>
</td></tr>
<tr><td><code>xP</code></td>
<td>
<p>The trait values at each time step for clade P</p>
</td></tr>
<tr><td><code>xH</code></td>
<td>
<p>The trait values at each time step for cladeH</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>The trait values at present for clade P</p>
</td></tr>
<tr><td><code>H</code></td>
<td>
<p>The trait values at present for clade P</p>
</td></tr>
<tr><td><code>Pmut</code></td>
<td>
<p>The number of new mutations at each time step for clade P</p>
</td></tr>
<tr><td><code>Hmut</code></td>
<td>
<p>The number of new mutations at each time step for clade H</p>
</td></tr>
<tr><td><code>iniP</code></td>
<td>
<p>The initial trait values for the individuals of clade P used in the simulation</p>
</td></tr>
<tr><td><code>iniH</code></td>
<td>
<p>The initial trait values for the individuals of clade H used in the simulation</p>
</td></tr>
<tr><td><code>thin.factor</code></td>
<td>
<p>The thin value used in the simulation</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>O. Maliet
</p>


<h3>References</h3>

<p>Maliet, O., Loeuille, N. and Morlon, H. (2020), An individual-based model for the eco-evolutionary emergence of bipartite interaction networks. Ecol Lett. doi:10.1111/ele.13592</p>


<h3>Examples</h3>

<pre><code class='language-R'># run the model
set.seed(1)


if(test){
mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 500,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5)

#build the genealogies
gen = make_gen.BipartiteEvol(mod)
plot(gen$H)

#compute the phylogenies
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#plot the result
plot_div.BipartiteEvol(gen,phy1, 1)

#build the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = 10, spatial = FALSE)


## add time steps to a former run
seed=as.integer(10)
set.seed(seed)

mod = sim.BipartiteEvol(nx = 8,ny = 4,NG = 500,
                        D = 3, muP = 0.1 , muH = 0.1,
                        alphaP = 0.12,alphaH = 0.12,
                        rP = 10, rH = 10,
                        verbose = 100, thin = 5,
                        P = mod$P, H = mod$H)  # former ru output

# update the genealogy
gen = make_gen.BipartiteEvol(mod,
                             treeP=gen$P, treeH=gen$H)

# update the phylogenies...
phy1 = define_species.BipartiteEvol(gen,threshold=1)

#... and the network
net = build_network.BipartiteEvol(gen, phy1)

trait.id = 1
plot_net.BipartiteEvol(gen,phy1,trait.id, net,mod, nx = 10, spatial = FALSE)
 }
 </code></pre>

<hr>
<h2 id='sim.convergence.geo'>
Simulation of trait data under the model of convergent character displacement described in Drury et al. 2017
</h2><span id='topic+sim.convergence.geo'></span>

<h3>Description</h3>

<p>Simulates the evolution of a continuous character that evolves depending on pairwise similarity in another, OU-evolving trait (e.g., a trait that covaries with resource use). sig2 and z0 are shared between two traits, max and alpha are for focal trait, OU parameters for non-focal trait
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
sim.convergence.geo(phylo,pars, Nsegments=2500, plot=FALSE, geo.object)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim.convergence.geo_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)
</p>
</td></tr>
<tr><td><code id="sim.convergence.geo_+3A_pars">pars</code></td>
<td>
 
<p>A matrix with a number of rows corresponding to the desired number of simulations, columns containing values for <code class="reqn">sig2</code> in <code>[,1]</code>, <code class="reqn">m</code> in <code>[,2]</code>, <code class="reqn">alpha</code> in <code>[,3]</code>, <code>root.value</code> in <code>[,4]</code>, <code class="reqn">psi</code> of the OU model for the non-focal, resource use trait in <code>[,5]</code>, and <code class="reqn">theta</code> in the OU model for the non-focal resource use trait in <code>[,6]</code> 
</p>
</td></tr>
<tr><td><code id="sim.convergence.geo_+3A_nsegments">Nsegments</code></td>
<td>

<p>the minimum number of time steps to simulate
</p>
</td></tr>
<tr><td><code id="sim.convergence.geo_+3A_plot">plot</code></td>
<td>

<p>if <code>TRUE</code>, returns two plots: the top plot is focal trait undergoing convergence, the bottom plot is non-focal trait evolving under BM or OU
</p>
</td></tr>
<tr><td><code id="sim.convergence.geo_+3A_geo.object">geo.object</code></td>
<td>

<p>geography object created using <a href="#topic+CreateGeoObject">CreateGeoObject</a>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adjusting <code>Nsegments</code> will impact the length of time the simulations take. 
The length of each segment (<code>max(nodeHeights(phylo))/Nsegments</code>) 
should be much smaller than the smallest branch (<code>min(phylo$edge.length)</code>).
</p>


<h3>Value</h3>

<p>A list of two matrices with the simulated values for each lineage (one simulation per row; columns correspond to species) for trait1 (focul trait undergoing convergence) and non.focal (resource-use trait that determines strength of convergence in trait1)
</p>


<h3>Author(s)</h3>

<p>J.P. Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
</p>


<h3>References</h3>

<p>Drury, J., Grether, G., Garland Jr., T., and Morlon, H. 2017. A review of phylogenetic methods for assessing the influence of interspecific interactions on phenotypic evolution. <em>Systematic Biology</em> 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(Anolis.data)
phylo&lt;-Anolis.data$phylo
geo.object&lt;-Anolis.data$geography.object

#simulate with the OU process present and absent
pars&lt;-expand.grid(0.05,-0.1,1,0,c(2,0),0)
sim.convergence.geo(phylo,pars,Nsegments=2500, plot=FALSE, geo.object)


</code></pre>

<hr>
<h2 id='sim.divergence.geo'>
Simulation of trait data under the model of divergent character displacement described in Drury et al. 2017
</h2><span id='topic+sim.divergence.geo'></span>

<h3>Description</h3>

<p>Simulates the evolution of a continuous character under a model of evolution where trait values are repelled according to between-species similarity in trait values, taking into account biogeography using a biogeo.object formatted from RPANDA (see CreateGeoObject function in RPANDA package)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
sim.divergence.geo(phylo,pars, Nsegments=2500, plot=FALSE, geo.object)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim.divergence.geo_+3A_phylo">phylo</code></td>
<td>

<p>a phylogenetic tree
</p>
</td></tr>
<tr><td><code id="sim.divergence.geo_+3A_pars">pars</code></td>
<td>

<p>A matrix with a number of rows corresponding to the desired number of simulations, columns containing values for <code class="reqn">sig2</code> in <code>[,1]</code>, <code class="reqn">m</code> in <code>[,2]</code>, <code class="reqn">alpha</code> in <code>[,3]</code>, <code>root.value</code> in <code>[,4]</code>, <code class="reqn">psi</code> of the OU model in <code>[,5]</code>, and <code class="reqn">theta</code> in the OU model in <code>[,6]</code>
</p>
</td></tr>
<tr><td><code id="sim.divergence.geo_+3A_nsegments">Nsegments</code></td>
<td>

<p>the minimum number of time steps to simulate
</p>
</td></tr>
<tr><td><code id="sim.divergence.geo_+3A_plot">plot</code></td>
<td>

<p>logical indicating whether to plot the simulated trait values at each time step
</p>
</td></tr>
<tr><td><code id="sim.divergence.geo_+3A_geo.object">geo.object</code></td>
<td>

<p>geography object created using <a href="#topic+CreateGeoObject">CreateGeoObject</a>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adjusting <code>Nsegments</code> will impact the length of time the simulations take. 
The length of each segment (<code>max(nodeHeights(phylo))/Nsegments</code>) 
should be much smaller than the smallest branch (<code>min(phylo$edge.length)</code>).
</p>


<h3>Value</h3>

<p>A matrix with the simulated values for each lineage (one simulation per row; columns correspond to species) 
</p>


<h3>Author(s)</h3>

<p>J.P. Drury <a href="mailto:jonathan.p.drury@gmail.com">jonathan.p.drury@gmail.com</a>
F. Hartig
</p>


<h3>References</h3>

<p>Drury, J., Grether, G., Garland Jr., T., and Morlon, H. 2017. A review of phylogenetic methods for assessing the influence of interspecific interactions on phenotypic evolution. <em>Systematic Biology</em> 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CreateGeoObject">CreateGeoObject</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(Anolis.data)
phylo&lt;-Anolis.data$phylo
geo.object&lt;-Anolis.data$geography.object

#simulate with the OU process present and absent
pars&lt;-expand.grid(0.05,2,1,0,c(2,0),0)
sim.divergence.geo(phylo,pars,Nsegments=2500, plot=FALSE, geo.object)


</code></pre>

<hr>
<h2 id='simul.comb.shift'>
Simulating trees from shift.estimates() results to test model adequacy
</h2><span id='topic+simul.comb.shift'></span>

<h3>Description</h3>

<p>Simulates trees with combination of shifts from shifts.estimates() output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  simul.comb.shift(n = 10000, phylo, sampling.fractions,
                   shift.res, combi = 1, clade.size = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simul.comb.shift_+3A_n">n</code></td>
<td>

<p>numeric. Defines the number of simulations to generate (see Details).
</p>
</td></tr>
<tr><td><code id="simul.comb.shift_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation).
</p>
</td></tr>
<tr><td><code id="simul.comb.shift_+3A_sampling.fractions">sampling.fractions</code></td>
<td>

<p>the output resulting from get.sampling.fractions.
</p>
</td></tr>
<tr><td><code id="simul.comb.shift_+3A_shift.res">shift.res</code></td>
<td>

<p>the output resulting from shift.estimates.
</p>
</td></tr>
<tr><td><code id="simul.comb.shift_+3A_combi">combi</code></td>
<td>

<p>numeric. Corresponds to the rank of the combination in the global comparison (shift.res$total).
</p>
</td></tr>
<tr><td><code id="simul.comb.shift_+3A_clade.size">clade.size</code></td>
<td>

<p>numeric. Defines the minimum number of species in a subgroup. Default is 5.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Some combinations of shifts might be complex cases to simulate because the backbone needs to be rich enough to graft subclades. Some simulations will not satisfy this condition and will then be discarded. In consequence, the number of simulated phylogenies in the output will not be equal to n for complex simulations. This is why the value of n is high by default (n = 10000), to ensure to have enough simulations (around 500) to test the presence.
</p>
<p>clade.size argument should be the same value for the whole procedure in the empirical case (same than for get.sampling.fraction and get.comb.shift).
</p>


<h3>Value</h3>

<p>a list of simulated phylogenies as object of type phylo. Tips of subclades are named with the letters a, b, c, etc. while tips of backbones are named with letters z, y, etc. The empirical groups are sorted from the more recent to the older one (i.e. group a will be the more recent empirical subclade, etc.).
</p>


<h3>Author(s)</h3>

<p>Nathan Mazet</p>


<h3>References</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>See Also</h3>

<p><code><a href="#topic+shift.estimates">shift.estimates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# loading data
data("Cetacea")
data("taxo_cetacea")
data("shifts_cetacea")

# with the results from shifts.estimates()

# no shifts tested at genus level
taxo_cetacea_no_genus &lt;- taxo_cetacea[names(taxo_cetacea) != "Genus"]

f_cetacea &lt;- get.sampling.fractions(phylo = Cetacea,
                                    data = taxo_cetacea_no_genus)

all_posteriors_cetacea &lt;- simul.comb.shift(phylo = Cetacea,
                                           sampling.fractions = f_cetacea,
                                           shift.res = shifts_cetacea)
  
</code></pre>

<hr>
<h2 id='simulateTipData'>
Tip trait simulation under a model of phenotypic evolution.
</h2><span id='topic+simulateTipData'></span>

<h3>Description</h3>

<p>Simulates tip trait data under a specified model of phenotypic evolution, with three distinct behaviours specified with the 'method' argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateTipData(object, params, method, v)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulateTipData_+3A_object">object</code></td>
<td>
<p>an object of class 'PhenotypicModel'.</p>
</td></tr>
<tr><td><code id="simulateTipData_+3A_params">params</code></td>
<td>
<p>vector of parameters, given in the same order as in the 'model' object.</p>
</td></tr>
<tr><td><code id="simulateTipData_+3A_method">method</code></td>
<td>
<p>an integer specifying the behaviour of the function. If method = 1 (default value), the tip distribution is first computed, before returning a simulated dataset drawn in this distribution. If method = 2, the whole trajectory is simulated step by step, plotted, and returned. Otherwise, the whole trajectory is simulated step by step, and then returned without being plotted.</p>
</td></tr>
<tr><td><code id="simulateTipData_+3A_v">v</code></td>
<td>
<p>boolean specifying the verbose mode. Default value : FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of trait values at the tips of the tree.
</p>


<h3>Author(s)</h3>

<p>M Manceau
</p>


<h3>References</h3>

<p>Manceau M., Lambert A., Morlon H. (2017) A unifying comparative phylogenetic framework including traits coevolving across interacting lineages Systematic Biology
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Loading an example tree
newick &lt;- "((((A:1,B:0.5):2,(C:3,D:2.5):1):6,E:10.25):2,(F:6.5,G:8.25):3):1;"
tree &lt;- read.tree(text=newick)

#Creating the models
modelBM &lt;- createModel(tree, 'BM')
modelOU &lt;- createModel(tree, 'OU')

#Simulating tip traits under both models with distinct behaviours of the functions :
dataBM &lt;- simulateTipData(modelBM, c(0,0,0,1))
dataOU &lt;- simulateTipData(modelOU, c(0,0,1,5,1), method=1)
dataBM2 &lt;- simulateTipData(modelBM, c(0,0,0,1), method=2)
</code></pre>

<hr>
<h2 id='simulateTipData-methods'> ~~ Methods for Function <code>simulateTipData</code>  ~~</h2><span id='topic+simulateTipData-methods'></span><span id='topic+simulateTipData+2CPhenotypicModel-method'></span>

<h3>Description</h3>

<p>~~ Methods for function <code>simulateTipData</code>  ~~
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(object = "PhenotypicModel")</code></dt><dd>
<p>This is the only method available for this function. Same behaviour for any PhenotypicModel.
</p>
</dd>
</dl>

<hr>
<h2 id='spectR'>
Spectral density plot of a phylogeny
</h2><span id='topic+spectR'></span>

<h3>Description</h3>

<p>Computes the spectra of eigenvalues for the modified graph Laplacian of a phylogenetic tree, identifies the spectral gap, then convolves the eigenvalues with a Gaussian kernel, and plots them alongside all eigenvalues ranked in descending order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spectR(phylo, meth=c("standard"),zero_bound=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spectR_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)</p>
</td></tr>
<tr><td><code id="spectR_+3A_meth">meth</code></td>
<td>

<p>the method used to compute the spectral density, which can either be &quot;standard&quot; or &quot;normal&quot;. If set to &quot;standard&quot;, computes the unnormalized version of the spectral density. If set to &quot;normal&quot;, computes the spectral density normalized to the degree matrix (see the associated paper for an explanation)</p>
</td></tr>
<tr><td><code id="spectR_+3A_zero_bound">zero_bound</code></td>
<td>

<p>if false, eigenvalues less than one are discarded</p>
</td></tr>	
</table>


<h3>Details</h3>

<p>Note that the eigengap should in principle be computed with the &quot;standard&quot; option</p>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>eigenvalues</code></td>
<td>

<p>the vector of eigenvalues</p>
</td></tr> 
<tr><td><code>principal_eigenvalue</code></td>
<td>

<p>the largest (or principal) eigenvalue of the spectral density profile</p>
</td></tr>
<tr><td><code>asymmetry</code></td>
<td>

<p>the skewness of the spectral density profile</p>
</td></tr> 
<tr><td><code>peak_height</code></td>
<td>

<p>the largest y-axis value of the spectral density profile</p>
</td></tr>
<tr><td><code>eigengap</code></td>
<td>

<p>the position of the largest difference between eigenvalues, giving the number of modalities in the tree</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>E Lewitus</p>


<h3>References</h3>

<p>Lewitus, E., Morlon, H., Characterizing and comparing phylogenies from their Laplacian spectrum, bioRxiv doi: http://dx.doi.org/10.1101/026476</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_spectR">plot_spectR</a></code>, <code><a href="#topic+JSDtree">JSDtree</a></code>, <code><a href="#topic+BICompare">BICompare</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Cetacea)
spectR(Cetacea,meth="standard",zero_bound=FALSE)
</code></pre>

<hr>
<h2 id='spectR_t'>
Spectral density plot of phylogenetic trait data
</h2><span id='topic+spectR_t'></span>

<h3>Description</h3>

<p>Computes the spectra of eigenvalues for the modified graph Laplacian of a phylogenetic tree with associated tip data, convolves the eigenvalues with a Gaussian kernel and plots the density profile of eigenvalues, and estimates the summary statistics of the profile.</p>


<h3>Usage</h3>

<pre><code class='language-R'>spectR_t(phylo, dat, draw=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spectR_t_+3A_phylo">phylo</code></td>
<td>

<p>an object of type 'phylo' (see ape documentation)</p>
</td></tr>
<tr><td><code id="spectR_t_+3A_dat">dat</code></td>
<td>

<p>a vector of trait data associated with the tips of the phylo object; tips and trait data should be aligned</p>
</td></tr>
<tr><td><code id="spectR_t_+3A_draw">draw</code></td>
<td>

<p>if true, the spectral density profile of the phylogenetic trait data is plotted
</p>
</td></tr>	
</table>


<h3>Value</h3>

<p>a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>eigenvalues</code></td>
<td>

<p>the vector of eigenvalues</p>
</td></tr> 
<tr><td><code>splitter</code></td>
<td>

<p>the largest (or principal) eigenvalue of the spectral density profile</p>
</td></tr>
<tr><td><code>fragmenter</code></td>
<td>

<p>the skewness of the spectral density profile</p>
</td></tr> 
<tr><td><code>tracer</code></td>
<td>

<p>the largest y-axis value of the spectral density profile</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>E Lewitus</p>


<h3>References</h3>

<p>Lewitus, E., Morlon, H. (2019) Characterizing and comparing phylogenetic trait data from their normalized Laplacian spectrum, bioRxiv doi: https://doi.org/10.1101/654087</p>


<h3>Examples</h3>

<pre><code class='language-R'>tr&lt;-rtree(10)
dat&lt;-runif(10,1,2)
spectR_t(tr,dat,draw=TRUE)
</code></pre>

<hr>
<h2 id='taxo_cetacea'>Cetacean taxonomy</h2><span id='topic+taxo_cetacea'></span>

<h3>Description</h3>

<p>Taxonomy of Cetaceans</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(taxo_cetacea)</code></pre>


<h3>Details</h3>

<p>This taxonomy lists all species of Cetaceans to properly calculate sampling fractions by clades. It corresponds to the phylogeny of Steeman et al. (2009).
</p>


<h3>Source</h3>

<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>References</h3>

<p>Steeman ME et al.(2009) Radiation of extant cetaceans driven by restructuring of the oceans <em>Syst Biol</em> 58:573-585
</p>
<p>Morlon, H., Parsons, T.L., Plotkin, J.B. (2011) Reconciling molecular phylogenies with the fossil record <em>Proc Nat Acad Sci</em> 108: 16327-16332
</p>
<p>Mazet, N., Morlon, H., Fabre, P., Condamine, F.L., (2023). Estimating clade‐specific diversification rates and palaeodiversity dynamics from reconstructed phylogenies. <em>Methods in Ecology and in Evolution</em> 14, 2575–2591. https://doi.org/10.1111/2041-210X.14195
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(taxo_cetacea)
print(taxo_cetacea)
</code></pre>

<hr>
<h2 id='theta_estimator'>
Compute Watterson genetic diversity (Theta estimator)
</h2><span id='topic+theta_estimator'></span>

<h3>Description</h3>

<p>This function computes the Theta estimator of genetic diversity (Watterson, 1975) while controlling for the presence of gaps in the alignment (Ferretti et al, 2012), frequent in barcoding datasets. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theta_estimator(sequences)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theta_estimator_+3A_sequences">sequences</code></td>
<td>

<p>a matrix representing the nucleotidic alignment of all the sequences present in the phylogenetic tree. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An estimate of genetic diversity. 
</p>


<h3>Author(s)</h3>

<p>Ana C. Afonso Silva &amp; Benoît Perez-Lamarque</p>


<h3>References</h3>

<p>Watterson GA , On the number of segregating sites in genetical models without recombination, 1975, Theor. Popul. Biol.
</p>
<p>Ferretti L, Raineri E, Ramos-Onsins S. 2012. Neutrality tests for sequences with missing data. Genetics 191: 1397–1401.
</p>
<p>Perez-Lamarque B, Öpik M, Maliet O, Silva A, Selosse M-A, Martos F, and Morlon H. 2022. Analysing diversification dynamics using barcoding data: The case of an obligate mycorrhizal symbiont, Molecular Ecology, 31:3496–512.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pi_estimator">pi_estimator</a></code>
<code><a href="#topic+delineate_phylotypes">delineate_phylotypes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(woodmouse)

alignment &lt;- as.character(woodmouse) # nucleotidic alignment 

theta_estimator(alignment)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
