<!DOCTYPE html><html><head><title>Help for package rPanglaoDB</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rPanglaoDB}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#getMarkers'><p>Get the list of samples from the panglaoDB database with a pattern of expression for a set of molecular markers.</p></a></li>
<li><a href='#getSampleComposition'><p>Get the cell-type content for each sample from the panglaoDB database.</p></a></li>
<li><a href='#getSampleList'><p>Get the sample list from the panglaoDB database.</p></a></li>
<li><a href='#getSamples'><p>Download the expression matrix and annotations from the panglaoDB database.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Download and Merge Single-Cell RNA-Seq Data from the PanglaoDB
Database</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Download and merge labeled single-cell RNA-seq data from the PanglaoDB <a href="https://panglaodb.se/">https://panglaodb.se/</a> into a Seurat object.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/dosorio/rPanglaoDB/">https://github.com/dosorio/rPanglaoDB/</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, XML, xml2, Seurat, pbapply, Matrix</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-12 19:51:08 UTC; danie</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Osorio <a href="https://orcid.org/0000-0003-4424-8422"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Marieke Kuijjer <a href="https://orcid.org/0000-0001-6280-3130"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  James J. Cai <a href="https://orcid.org/0000-0002-8081-6725"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Osorio &lt;daniecos@uio.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-12 22:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='getMarkers'>Get the list of samples from the panglaoDB database with a pattern of expression for a set of molecular markers.</h2><span id='topic+getMarkers'></span>

<h3>Description</h3>

<p>Get the list of samples from the panglaoDB database with a pattern of expression for a set of molecular markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMarkers(include, exclude = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMarkers_+3A_include">include</code></td>
<td>
<p>A set of molecular markers to query the database. This set of genes needs to be expressed in the sample.</p>
</td></tr>
<tr><td><code id="getMarkers_+3A_exclude">exclude</code></td>
<td>
<p>A set of molecular markers to query the database. This set of genes needs to be absent in the sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The rows in the data frame are the samples matching the requested pattern. The returned data frame contain 7 columns as follows:
</p>

<ul>
<li><p>SRA: The SRA identifier of the biological sample in the SRA database
</p>
</li>
<li><p>SRS: The SRS identifier of the biological sample in the SRA database
</p>
</li>
<li><p>Specie:  The specie from which the biological samples originated from
</p>
</li>
<li><p>Tissue:  The tissue from which the biological samples originated from
</p>
</li>
<li><p>Cluster:  The cluster-id assigned by the panglaoDB database to the cells matching the requested pattern
</p>
</li>
<li><p>Cell-Type: The cell-type from which the counts originates from
</p>
</li>
<li><p>Markers: The recovered pattern for the marker genes requested
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Fibrocytes &lt;- getMarkers(include = c('ACTA2', 'CD34', 'FAP'))
Fibrocytes
## End(Not run)

#       SRA        SRS       Specie Tissue Cluster   Cell-Type         Markers
# SRA681285 SRS3121028 Mus musculus Dermis       4 Fibroblasts +ACTA2+CD34+FAP
</code></pre>

<hr>
<h2 id='getSampleComposition'>Get the cell-type content for each sample from the panglaoDB database.</h2><span id='topic+getSampleComposition'></span>

<h3>Description</h3>

<p>Get the cell-type content for each sample from the panglaoDB database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSampleComposition(
  sra = "All",
  srs = "All",
  tissue = "All",
  protocol = "All",
  specie = "All",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSampleComposition_+3A_sra">sra</code></td>
<td>
<p>Filter based on the SRA identifier of the biological sample in the SRA database</p>
</td></tr>
<tr><td><code id="getSampleComposition_+3A_srs">srs</code></td>
<td>
<p>Filter based on the SRS identifier of the biological sample in the SRA database</p>
</td></tr>
<tr><td><code id="getSampleComposition_+3A_tissue">tissue</code></td>
<td>
<p>Filter based on the tissue from which the biological samples originates from</p>
</td></tr>
<tr><td><code id="getSampleComposition_+3A_protocol">protocol</code></td>
<td>
<p>Filter based on the single-cell library preparation protocol used to generate the data</p>
</td></tr>
<tr><td><code id="getSampleComposition_+3A_specie">specie</code></td>
<td>
<p>Filter based on the specie from which the biological samples originates from</p>
</td></tr>
<tr><td><code id="getSampleComposition_+3A_verbose">verbose</code></td>
<td>
<p>A boolean value TRUE or FALSE to activate the verbose mode</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns the cell-type composition of the samples included in the PanglaoDB database in a data frame with 8 columns as follows:
</p>

<ul>
<li><p>SRA: The SRA identifier of the biological sample in the SRA database
</p>
</li>
<li><p>SRS: The SRS identifier of the biological sample in the SRA database
</p>
</li>
<li><p>Tissue: The tissue from which the biological samples originated from
</p>
</li>
<li><p>Protocol: The single-cell library preparation protocol used to generate the data
</p>
</li>
<li><p>Species: The species from which the biological samples originated from
</p>
</li>
<li><p>Cluster:  The cluster-id assigned by the panglaoDB database to the cells in the sample
</p>
</li>
<li><p>Cells: The number of cells included in the cluster
</p>
</li>
<li><p>Cell Type: The cell-type from which the counts originates from
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># From PanglaoDB
# https://panglaodb.se/list_clusters_and_cell_types.html?sra=SRA689041&amp;srs=SRS3166675

SRS3166675 &lt;- getSampleComposition(srs = 'SRS3166675')
head(SRS3166675)

#       SRA        SRS Tissue     Protocol      Species Cluster Cells           Cell Type
# SRA689041 SRS3166675  Colon 10x chromium Mus musculus       0   735         Fibroblasts
# SRA689041 SRS3166675  Colon 10x chromium Mus musculus       1   526 Smooth muscle cells
# SRA689041 SRS3166675  Colon 10x chromium Mus musculus       2   465             Unknown
# SRA689041 SRS3166675  Colon 10x chromium Mus musculus       3   157             Unknown
# SRA689041 SRS3166675  Colon 10x chromium Mus musculus       4   140        Goblet cells
# SRA689041 SRS3166675  Colon 10x chromium Mus musculus       5   100         Fibroblasts
</code></pre>

<hr>
<h2 id='getSampleList'>Get the sample list from the panglaoDB database.</h2><span id='topic+getSampleList'></span>

<h3>Description</h3>

<p>Get the sample list from the panglaoDB database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSampleList()
</code></pre>


<h3>Value</h3>

<p>This function returns a data frame with 6 columns as follows:
</p>

<ul>
<li><p>SRA: The SRA identifier of the biological sample in the SRA database
</p>
</li>
<li><p>SRS: The SRS identifier of the biological sample in the SRA database
</p>
</li>
<li><p>Tissue: The tissue from which the biological samples originated from
</p>
</li>
<li><p>Protocol: The single-cell library preparation protocol used to generate the data
</p>
</li>
<li><p>Species: The specie from which the biological samples originated from
</p>
</li>
<li><p>Cells: The number of cells included in the sample
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># From the PanglaoDB database
# https://panglaodb.se/samples.html

sampleList &lt;- getSampleList()
head(sampleList)

#       SRA        SRS                          Tissue     Protocol      Species Cells
# SRA553822 SRS2119548   Cultured embryonic stem cells 10x chromium Homo sapiens  6501
# SRA570744 SRS2253536                 Lung mesenchyme 10x chromium Mus musculus  4611
# SRA598936 SRS2428405                   Kidney cortex 10x chromium Homo sapiens  3759
# SRA644036 SRS2808714 Cervical and lumbar spinal cord 10x chromium Mus musculus  1025
# SRA670243 SRS3078084                Ventral midbrain 10x chromium Mus musculus  5603
# SRA689041 SRS3166675                           Colon 10x chromium Mus musculus  2878
</code></pre>

<hr>
<h2 id='getSamples'>Download the expression matrix and annotations from the panglaoDB database.</h2><span id='topic+getSamples'></span>

<h3>Description</h3>

<p>Download the expression matrix and annotations from the panglaoDB database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSamples(
  sra = "All",
  srs = "All",
  tissue = "All",
  protocol = "All",
  specie = "All",
  celltype = "All",
  include = NA,
  exclude = NA,
  merge = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSamples_+3A_sra">sra</code></td>
<td>
<p>Filter based on the SRA identifier of the biological sample in the SRA database</p>
</td></tr>
<tr><td><code id="getSamples_+3A_srs">srs</code></td>
<td>
<p>Filter based on the SRS identifier of the biological sample in the SRA database</p>
</td></tr>
<tr><td><code id="getSamples_+3A_tissue">tissue</code></td>
<td>
<p>Filter based on the tissue from which the biological samples originates from</p>
</td></tr>
<tr><td><code id="getSamples_+3A_protocol">protocol</code></td>
<td>
<p>Filter based on the single-cell library preparation protocol used to generate the data</p>
</td></tr>
<tr><td><code id="getSamples_+3A_specie">specie</code></td>
<td>
<p>Filter based on the specie from which the biological samples originates from</p>
</td></tr>
<tr><td><code id="getSamples_+3A_celltype">celltype</code></td>
<td>
<p>Filter based on the cell-type from which the counts originates from</p>
</td></tr>
<tr><td><code id="getSamples_+3A_include">include</code></td>
<td>
<p>A set of molecular markers to filter the dataset. This set of genes needs to be expressed in each cell.</p>
</td></tr>
<tr><td><code id="getSamples_+3A_exclude">exclude</code></td>
<td>
<p>A set of molecular markers to filter the dataset. This set of genes needs to be absent in each cell.</p>
</td></tr>
<tr><td><code id="getSamples_+3A_merge">merge</code></td>
<td>
<p>A boolean value TRUE or FALSE defining if the samples should be returned as a list or as a unique Seurat object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object, as described in <code>?SeuratObject::`Seurat-class`</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># From PanglaoDB SRS3805255
# https://panglaodb.se/view_data.php?sra=SRA705190&amp;srs=SRS4139632

## Not run: 
SRS4139632 &lt;- getSamples(srs = 'SRS4139632')
SRS4139632
## End(Not run)

# An object of class Seurat
# 19859 features across 102 samples within 1 assay
# Active assay: RNA (19859 features, 0 variable features)

# Metadata from the PanglaoDB database can be accessed as follows:
# head(SRS4139632[[]])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
