<!DOCTYPE html><html><head><title>Help for package rPDBapi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rPDBapi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_property'><p>Add or Merge Properties for RCSB PDB Data Fetching</p></a></li>
<li><a href='#autoresolve_sequence_type'><p>Automatically Determine the Sequence Type</p></a></li>
<li><a href='#ChemicalOperator'><p>Chemical Operator for SMILES/InChI Searches</p></a></li>
<li><a href='#ComparisonOperator'><p>Create a Comparison Search Operator</p></a></li>
<li><a href='#ContainsPhraseOperator'><p>Create a Contains Phrase Search Operator</p></a></li>
<li><a href='#ContainsWordsOperator'><p>Create a Contains Words Search Operator</p></a></li>
<li><a href='#data_fetcher'><p>Fetch RCSB PDB Data Based on Specified Criteria</p></a></li>
<li><a href='#DefaultOperator'><p>Create a Default Search Operator</p></a></li>
<li><a href='#describe_chemical'><p>Describe Chemical Compound from RCSB PDB</p></a></li>
<li><a href='#ExactMatchOperator'><p>Create an Exact Match Search Operator</p></a></li>
<li><a href='#ExistsOperator'><p>Create an Existence Search Operator</p></a></li>
<li><a href='#FastaSequence'><p>Create a FASTA Sequence Data Structure</p></a></li>
<li><a href='#fetch_data'><p>Fetch Data from RCSB PDB Using a JSON Query</p></a></li>
<li><a href='#find_papers'><p>Search for and Retrieve Paper Titles from PDB</p></a></li>
<li><a href='#find_results'><p>Retrieve Specific Fields for Search Results from RCSB PDB</p></a></li>
<li><a href='#generate_json_query'><p>Generate a JSON Query for RCSB PDB Data Retrieval</p></a></li>
<li><a href='#get_fasta_from_rcsb_entry'><p>Retrieve FASTA Sequences from PDB Entry</p></a></li>
<li><a href='#get_info'><p>Retrieve Information for a Given PDB ID</p></a></li>
<li><a href='#get_pdb_file'><p>Download PDB Files from RCSB Database</p></a></li>
<li><a href='#infer_search_service'><p>Infer the Appropriate Search Service</p></a></li>
<li><a href='#InOperator'><p>Create an Inclusion Search Operator</p></a></li>
<li><a href='#parse_fasta_text_to_list'><p>Parse Raw FASTA Text into a List of Sequences</p></a></li>
<li><a href='#perform_search'><p>Perform a Search in the RCSB PDB</p></a></li>
<li><a href='#query_search'><p>Search Query Function</p></a></li>
<li><a href='#QueryGroup'><p>Create a Grouped Query Object</p></a></li>
<li><a href='#QueryNode'><p>Create a Query Node Object</p></a></li>
<li><a href='#RangeOperator'><p>Create a Range Search Operator</p></a></li>
<li><a href='#RequestOptions'><p>Define Options for Search Requests</p></a></li>
<li><a href='#return_data_as_dataframe'><p>Transform Response Data into a Dataframe</p></a></li>
<li><a href='#rPDBapi-package'><p>rPDBapi: A Comprehensive Interface for Accessing the Protein Data Bank</p></a></li>
<li><a href='#ScoredResult'><p>Create a Scored Result Object</p></a></li>
<li><a href='#search_graphql'><p>Perform a GraphQL Query to RCSB PDB</p></a></li>
<li><a href='#SeqMotifOperator'><p>Create a Sequence Motif Operator for Searches</p></a></li>
<li><a href='#SequenceOperator'><p>Create a Sequence Operator for Sequence-Based Searches</p></a></li>
<li><a href='#StructureOperator'><p>Create a Structure Operator for Structure-Based Searches</p></a></li>
<li><a href='#walk_nested_dict'><p>Recursively Walk Through a Nested Dictionary</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Comprehensive Interface for Accessing the Protein Data Bank</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Streamlines the interaction with the RCSB Protein Data Bank (PDB) <a href="https://www.rcsb.org/">https://www.rcsb.org/</a>. This interface offers an intuitive and 
            powerful tool for searching and retrieving a diverse range of data types from the PDB. It includes advanced functionalities like 
            BLAST and sequence motif queries. Built upon the existing XML-based API of the PDB, it simplifies the creation of custom requests, 
            thereby enhancing usability and flexibility for researchers.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, purrr, httr, jsonlite, xml2, stringr, bio3d, magrittr,
methods</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-17 09:38:48 UTC; selcukkorkmaz</td>
</tr>
<tr>
<td>Author:</td>
<td>Selcuk Korkmaz <a href="https://orcid.org/0000-0003-4632-6850"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Bilge Eren Yamasan
    <a href="https://orcid.org/0000-0002-6525-2503"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Selcuk Korkmaz &lt;selcukorkmaz@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-17 19:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_property'>Add or Merge Properties for RCSB PDB Data Fetching</h2><span id='topic+add_property'></span>

<h3>Description</h3>

<p>This function is designed for handling properties related to fetching data from the Protein Data Bank (PDB).
It takes a dictionary, where keys represent properties and values are lists of subproperties.
If a property already exists in the input list, the function merges the subproperties,
ensuring each subproperty is unique and maintains character vector format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_property(property)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_property_+3A_property">property</code></td>
<td>
<p>A Python dictionary where keys are the properties (like 'cell', 'exptl') and
values are lists of subproperties (like 'volume', 'angle_beta', 'method').
Each subproperty should be in character vector format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified list with updated properties where subproperties are merged if a property already exists.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>properties &lt;- list(cell = c("volume", "angle_beta"), exptl = c("method"))
add_property(properties)
</code></pre>

<hr>
<h2 id='autoresolve_sequence_type'>Automatically Determine the Sequence Type</h2><span id='topic+autoresolve_sequence_type'></span>

<h3>Description</h3>

<p>This function determines the type of a given sequence (DNA, RNA, or PROTEIN) based on its characters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autoresolve_sequence_type(sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autoresolve_sequence_type_+3A_sequence">sequence</code></td>
<td>
<p>A string representing the sequence to be analyzed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string indicating the resolved sequence type.
</p>

<hr>
<h2 id='ChemicalOperator'>Chemical Operator for SMILES/InChI Searches</h2><span id='topic+ChemicalOperator'></span>

<h3>Description</h3>

<p>Creates an object representing a chemical search operator using SMILES or InChI descriptors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChemicalOperator(descriptor, matching_criterion = "graph-strict")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ChemicalOperator_+3A_descriptor">descriptor</code></td>
<td>
<p>A valid SMILES or InChI string.</p>
</td></tr>
<tr><td><code id="ChemicalOperator_+3A_matching_criterion">matching_criterion</code></td>
<td>
<p>The criterion for matching, one of the values from DescriptorMatchingCriterion.
Defaults to &quot;graph-strict&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list representing a chemical operator.
</p>

<hr>
<h2 id='ComparisonOperator'>Create a Comparison Search Operator</h2><span id='topic+ComparisonOperator'></span>

<h3>Description</h3>

<p>Constructs a ComparisonOperator object for search operations that perform comparison checks on attribute values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComparisonOperator(attribute, value, comparison_type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComparisonOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be compared.</p>
</td></tr>
<tr><td><code id="ComparisonOperator_+3A_value">value</code></td>
<td>
<p>The value to compare against.</p>
</td></tr>
<tr><td><code id="ComparisonOperator_+3A_comparison_type">comparison_type</code></td>
<td>
<p>A string specifying the type of comparison (e.g., 'equal', 'greater_than').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ComparisonOperator'.
</p>

<hr>
<h2 id='ContainsPhraseOperator'>Create a Contains Phrase Search Operator</h2><span id='topic+ContainsPhraseOperator'></span>

<h3>Description</h3>

<p>Constructs a ContainsPhraseOperator object for search operations that look for attributes containing a specific phrase.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContainsPhraseOperator(attribute, value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContainsPhraseOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be evaluated.</p>
</td></tr>
<tr><td><code id="ContainsPhraseOperator_+3A_value">value</code></td>
<td>
<p>The phrase to search for in the attribute.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ContainsPhraseOperator'.
</p>

<hr>
<h2 id='ContainsWordsOperator'>Create a Contains Words Search Operator</h2><span id='topic+ContainsWordsOperator'></span>

<h3>Description</h3>

<p>Constructs a ContainsWordsOperator object for search operations that look for attributes containing certain words.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContainsWordsOperator(attribute, value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContainsWordsOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be evaluated.</p>
</td></tr>
<tr><td><code id="ContainsWordsOperator_+3A_value">value</code></td>
<td>
<p>The words to search for in the attribute.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ContainsWordsOperator'.
</p>

<hr>
<h2 id='data_fetcher'>Fetch RCSB PDB Data Based on Specified Criteria</h2><span id='topic+data_fetcher'></span>

<h3>Description</h3>

<p>This function fetches data based on a given identifier (ID), data type, and a set of properties.
It can return the data either in its original format or as a dataframe.
The function integrates several steps including validating IDs, generating a JSON query,
fetching the data, and formatting the response.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_fetcher(
  id = NULL,
  data_type = "ENTRY",
  properties = NULL,
  return_as_dataframe = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="data_fetcher_+3A_id">id</code></td>
<td>
<p>An identifier or a list of identifiers for the data to be fetched.</p>
</td></tr>
<tr><td><code id="data_fetcher_+3A_data_type">data_type</code></td>
<td>
<p>A string specifying the type of data to fetch.
Default is &quot;ENTRY&quot;.</p>
</td></tr>
<tr><td><code id="data_fetcher_+3A_properties">properties</code></td>
<td>
<p>A list or dictionary of properties to be included in the data fetching process.</p>
</td></tr>
<tr><td><code id="data_fetcher_+3A_return_as_dataframe">return_as_dataframe</code></td>
<td>
<p>A boolean indicating whether to return the response as a dataframe.
Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on the value of 'return_as_dataframe',
this function returns either a dataframe or data in its original format.
</p>

<hr>
<h2 id='DefaultOperator'>Create a Default Search Operator</h2><span id='topic+DefaultOperator'></span>

<h3>Description</h3>

<p>Constructs a DefaultOperator object for use in general search operations within the RCSB PDB.
This operator is a basic search operator with a single value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DefaultOperator(value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DefaultOperator_+3A_value">value</code></td>
<td>
<p>The value to be used in the search operation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'DefaultOperator' representing the default search operator.
</p>

<hr>
<h2 id='describe_chemical'>Describe Chemical Compound from RCSB PDB</h2><span id='topic+describe_chemical'></span>

<h3>Description</h3>

<p>This function retrieves the description of a chemical compound from the RCSB PDB based on its ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>describe_chemical(chem_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="describe_chemical_+3A_chem_id">chem_id</code></td>
<td>
<p>A string representing the 3-character chemical ID.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dictionary containing the chemical description.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>chem_desc &lt;- describe_chemical('NAG')
print(chem_desc$rcsb_chem_comp_descriptor$smiles)
</code></pre>

<hr>
<h2 id='ExactMatchOperator'>Create an Exact Match Search Operator</h2><span id='topic+ExactMatchOperator'></span>

<h3>Description</h3>

<p>Constructs an ExactMatchOperator object for precise search operations in the RCSB PDB.
It matches an exact attribute value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExactMatchOperator(attribute, value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExactMatchOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to match.</p>
</td></tr>
<tr><td><code id="ExactMatchOperator_+3A_value">value</code></td>
<td>
<p>The exact value to search for.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ExactMatchOperator'.
</p>

<hr>
<h2 id='ExistsOperator'>Create an Existence Search Operator</h2><span id='topic+ExistsOperator'></span>

<h3>Description</h3>

<p>Constructs an ExistsOperator object for search operations to check the existence of an attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExistsOperator(attribute)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExistsOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute whose existence is to be checked.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ExistsOperator'.
</p>

<hr>
<h2 id='FastaSequence'>Create a FASTA Sequence Data Structure</h2><span id='topic+FastaSequence'></span>

<h3>Description</h3>

<p>This function constructs a FASTA sequence data structure from given parameters.
It is typically used to format and store FASTA sequence information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FastaSequence(entity_id, chains, sequence, fasta_header)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FastaSequence_+3A_entity_id">entity_id</code></td>
<td>
<p>A string representing the entity ID of the sequence.</p>
</td></tr>
<tr><td><code id="FastaSequence_+3A_chains">chains</code></td>
<td>
<p>A character vector of chain identifiers associated with the sequence.</p>
</td></tr>
<tr><td><code id="FastaSequence_+3A_sequence">sequence</code></td>
<td>
<p>A string representing the nucleotide or amino acid sequence.</p>
</td></tr>
<tr><td><code id="FastaSequence_+3A_fasta_header">fasta_header</code></td>
<td>
<p>A string representing the header of the FASTA sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list representing the FASTA sequence, including entity ID, chains, the sequence itself, and the FASTA header.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fasta_data &lt;- FastaSequence("1XYZ", c("A", "B"), "MVLSPADKT", "header_info")
</code></pre>

<hr>
<h2 id='fetch_data'>Fetch Data from RCSB PDB Using a JSON Query</h2><span id='topic+fetch_data'></span>

<h3>Description</h3>

<p>This function sends a JSON query to the RCSB Protein Data Bank (PDB) and fetches the corresponding data.
It checks for errors in the response and warns if there are discrepancies in the number of IDs found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetch_data(json_query, data_type, ids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fetch_data_+3A_json_query">json_query</code></td>
<td>
<p>A JSON string representing the query to be sent to the PDB.</p>
</td></tr>
<tr><td><code id="fetch_data_+3A_data_type">data_type</code></td>
<td>
<p>A string indicating the type of data to be fetched (not directly used in the function but may be relevant for context).</p>
</td></tr>
<tr><td><code id="fetch_data_+3A_ids">ids</code></td>
<td>
<p>A vector of identifiers to fetch data for.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the data fetched from the PDB, with the names of the list elements set to the corresponding IDs.
If an error is encountered in the data fetching process, the function returns 'NULL'.
</p>

<hr>
<h2 id='find_papers'>Search for and Retrieve Paper Titles from PDB</h2><span id='topic+find_papers'></span>

<h3>Description</h3>

<p>This function searches for papers in the Protein Data Bank (PDB) using a specified search term.
It retrieves the titles of papers up to a specified maximum number of results.
The function assumes the presence of 'query_search' and 'get_info' functions to perform the search and fetch paper details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_papers(search_term, max_results = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_papers_+3A_search_term">search_term</code></td>
<td>
<p>A string specifying the term to search for in the PDB.</p>
</td></tr>
<tr><td><code id="find_papers_+3A_max_results">max_results</code></td>
<td>
<p>An integer indicating the maximum number of paper titles to retrieve.
Defaults to 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list where each element's name is a PDB ID and its value is the title of the corresponding paper.
</p>

<hr>
<h2 id='find_results'>Retrieve Specific Fields for Search Results from RCSB PDB</h2><span id='topic+find_results'></span>

<h3>Description</h3>

<p>This function performs a search in the Protein Data Bank (PDB) using a provided search term
and retrieves information for a specified field (e.g., citation) for each search result.
It relies on 'query_search' and 'get_info' functions for searching and retrieving detailed information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_results(search_term, field = "citation")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_results_+3A_search_term">search_term</code></td>
<td>
<p>A string specifying the term to search for in the PDB.</p>
</td></tr>
<tr><td><code id="find_results_+3A_field">field</code></td>
<td>
<p>A string indicating the specific field to retrieve for each search result.
Default is &quot;citation&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list where each element's name is a PDB ID and its value is the information
for the specified field from the corresponding search result.
</p>

<hr>
<h2 id='generate_json_query'>Generate a JSON Query for RCSB PDB Data Retrieval</h2><span id='topic+generate_json_query'></span>

<h3>Description</h3>

<p>This function constructs a JSON query for retrieving data from the RCSB Protein Data Bank (PDB).
It requires input parameters like IDs, data type, and properties to tailor the query for specific data retrieval needs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_json_query(ids, data_type, properties)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_json_query_+3A_ids">ids</code></td>
<td>
<p>A vector of identifiers for which data needs to be retrieved.</p>
</td></tr>
<tr><td><code id="generate_json_query_+3A_data_type">data_type</code></td>
<td>
<p>A string indicating the type of data to be queried, such as 'ENTRY', 'POLYMER_ENTITY', etc.</p>
</td></tr>
<tr><td><code id="generate_json_query_+3A_properties">properties</code></td>
<td>
<p>A list of properties to be included in the query. Each element of the list should be
a character vector representing properties for the respective data type.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string representing the generated JSON query formatted for PDB data retrieval.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ids &lt;- c("1XYZ", "2XYZ")
properties &lt;- list(cell = c("volume", "angle_beta"), exptl = c("method"))
json_query &lt;- generate_json_query(ids, "ENTRY", properties)
json_query
</code></pre>

<hr>
<h2 id='get_fasta_from_rcsb_entry'>Retrieve FASTA Sequences from PDB Entry</h2><span id='topic+get_fasta_from_rcsb_entry'></span>

<h3>Description</h3>

<p>This function fetches FASTA sequences from the RCSB Protein Data Bank (PDB) for a specified entry ID.
It sends an HTTP request to the PDB and processes the response to extract FASTA sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fasta_from_rcsb_entry(rcsb_id, verbosity = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_fasta_from_rcsb_entry_+3A_rcsb_id">rcsb_id</code></td>
<td>
<p>A string representing the PDB ID for which the FASTA sequence is to be retrieved.</p>
</td></tr>
<tr><td><code id="get_fasta_from_rcsb_entry_+3A_verbosity">verbosity</code></td>
<td>
<p>A boolean flag indicating whether to print status messages during the function execution.
Defaults to TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of FASTA sequences associated with the provided RCSB entry ID.
</p>

<hr>
<h2 id='get_info'>Retrieve Information for a Given PDB ID</h2><span id='topic+get_info'></span>

<h3>Description</h3>

<p>This function looks up all information about a given PDB ID using the REST API.
It handles JSON data and HTTP requests and converts old entry identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_info(pdb_id, url_root = "https://data.rcsb.org/rest/v1/core/entry/")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_info_+3A_pdb_id">pdb_id</code></td>
<td>
<p>A 4-character string specifying a PDB entry of interest.</p>
</td></tr>
<tr><td><code id="get_info_+3A_url_root">url_root</code></td>
<td>
<p>The root URL for the specific request type. Default is 'https://data.rcsb.org/rest/v1/core/entry/'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ordered dictionary (list in R) object corresponding to entry information.
Returns NULL if retrieval fails.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:
# info &lt;- get_info(pdb_id = c"4hhb")
</code></pre>

<hr>
<h2 id='get_pdb_file'>Download PDB Files from RCSB Database</h2><span id='topic+get_pdb_file'></span>

<h3>Description</h3>

<p>This function downloads PDB files from the RCSB database, supporting different file types and optional compression.
It warns to consider compression for CIF files for faster download.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pdb_file(
  pdb_id,
  filetype = "cif",
  rm.insert = FALSE,
  rm.alt = TRUE,
  compression = TRUE,
  save = FALSE,
  path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pdb_file_+3A_pdb_id">pdb_id</code></td>
<td>
<p>A 4-character string specifying a PDB entry of interest.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_filetype">filetype</code></td>
<td>
<p>The file type: 'pdb', 'cif', 'xml', or 'structfact'. Default is 'pdb'.pdb is the older file format and cif is the newer replacement. xlm is also can be obtained and parsed. structfact retrieves structure factors (only available for certain PDB entries).</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_rm.insert">rm.insert</code></td>
<td>
<p>Logical, if TRUE PDB insert records are ignored.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_rm.alt">rm.alt</code></td>
<td>
<p>Logical, if TRUE PDB alternate records are ignored.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_compression">compression</code></td>
<td>
<p>Logical indicating whether to request the data as a compressed version. Default is TRUE</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_save">save</code></td>
<td>
<p>Logical, if TRUE saves PDB file to the desired path.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_path">path</code></td>
<td>
<p>The path where the file should be saved. If NULL, the file is saved in a temporary directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of class <code>"pdb"</code> with the following components:
</p>
<table>
<tr><td><code>atom</code></td>
<td>
<p> a data.frame containing all atomic coordinate ATOM and
HETATM data, with a row per ATOM/HETATM and a column per record type.</p>
</td></tr>
<tr><td><code>xyz</code></td>
<td>
<p> a numeric matrix of class <code>"xyz"</code> containing the
ATOM and HETATM coordinate data. </p>
</td></tr>
<tr><td><code>calpha</code></td>
<td>
<p> logical vector with length equal to <code>nrow(atom)</code> with TRUE
values indicating a C-alpha &ldquo;elety&rdquo;. </p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p> the matched call. </p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>pdb_file &lt;-  get_pdb_file(pdb_id = "4HHB", filetype = "cif")
</code></pre>

<hr>
<h2 id='infer_search_service'>Infer the Appropriate Search Service</h2><span id='topic+infer_search_service'></span>

<h3>Description</h3>

<p>Determines the appropriate search service for a given search operator in RCSB PDB queries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infer_search_service(search_operator)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="infer_search_service_+3A_search_operator">search_operator</code></td>
<td>
<p>A query operator object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The inferred search service.
</p>

<hr>
<h2 id='InOperator'>Create an Inclusion Search Operator</h2><span id='topic+InOperator'></span>

<h3>Description</h3>

<p>Constructs an InOperator object for search operations where the attribute value must be within a specified set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InOperator(attribute, value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be evaluated.</p>
</td></tr>
<tr><td><code id="InOperator_+3A_value">value</code></td>
<td>
<p>The set of values to include in the search.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'InOperator'.
</p>

<hr>
<h2 id='parse_fasta_text_to_list'>Parse Raw FASTA Text into a List of Sequences</h2><span id='topic+parse_fasta_text_to_list'></span>

<h3>Description</h3>

<p>This function parses raw FASTA text into a structured list format.
It splits the FASTA text into individual sequences and extracts relevant information
like entity ID, chains, sequence, and header.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_fasta_text_to_list(raw_fasta_text)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parse_fasta_text_to_list_+3A_raw_fasta_text">raw_fasta_text</code></td>
<td>
<p>A string containing the raw FASTA text to be parsed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of FASTA sequences, each as a separate list element with entity ID, chains, sequence, and header.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>raw_fasta_text &lt;- "&gt;1XYZ|Chains A, B\nMVLSPADKT...\n&gt;2XYZ|Chain C\nGVLSADFT..."
fasta_list &lt;- parse_fasta_text_to_list(raw_fasta_text)
</code></pre>

<hr>
<h2 id='perform_search'>Perform a Search in the RCSB PDB</h2><span id='topic+perform_search'></span>

<h3>Description</h3>

<p>This function facilitates searching the RCSB Protein Data Bank (PDB) using a specified search operator.
It allows various configurations like return type, additional request options, and verbosity control.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_search(
  search_operator,
  return_type = "ENTRY",
  request_options = NULL,
  return_with_scores = FALSE,
  return_raw_json_dict = FALSE,
  verbosity = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perform_search_+3A_search_operator">search_operator</code></td>
<td>
<p>An object specifying the search criteria.</p>
</td></tr>
<tr><td><code id="perform_search_+3A_return_type">return_type</code></td>
<td>
<p>A string specifying the type of data to return, defaulting to 'ENTRY'.</p>
</td></tr>
<tr><td><code id="perform_search_+3A_request_options">request_options</code></td>
<td>
<p>Additional options for the search request, default is NULL.</p>
</td></tr>
<tr><td><code id="perform_search_+3A_return_with_scores">return_with_scores</code></td>
<td>
<p>A boolean indicating whether to return search results with scores, default is FALSE.</p>
</td></tr>
<tr><td><code id="perform_search_+3A_return_raw_json_dict">return_raw_json_dict</code></td>
<td>
<p>A boolean indicating whether to return raw JSON response, default is FALSE.</p>
</td></tr>
<tr><td><code id="perform_search_+3A_verbosity">verbosity</code></td>
<td>
<p>A boolean flag indicating whether to display verbose messages during execution, default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The search results, which can vary based on the return type and options specified.
</p>

<hr>
<h2 id='query_search'>Search Query Function</h2><span id='topic+query_search'></span>

<h3>Description</h3>

<p>This function performs a search query against the RCSB Protein Data Bank
using their REST API. It allows for various types of searches based on the provided parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_search(
  search_term,
  query_type = "full_text",
  return_type = "entry",
  scan_params = NULL,
  num_attempts = 1,
  sleep_time = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="query_search_+3A_search_term">search_term</code></td>
<td>
<p>A string specifying the term to search in the database.</p>
</td></tr>
<tr><td><code id="query_search_+3A_query_type">query_type</code></td>
<td>
<p>A string specifying the type of query to perform.
Supported values include &quot;full_text&quot;, &quot;PubmedIdQuery&quot;,
&quot;TreeEntityQuery&quot;, &quot;ExpTypeQuery&quot;, &quot;AdvancedAuthorQuery&quot;,
&quot;OrganismQuery&quot;, &quot;pfam&quot;, and &quot;uniprot&quot;. Default is &quot;full_text&quot;.</p>
</td></tr>
<tr><td><code id="query_search_+3A_return_type">return_type</code></td>
<td>
<p>A string specifying the type of search result to return.
Possible values are &quot;entry&quot; (default) and &quot;polymer_entity&quot;.</p>
</td></tr>
<tr><td><code id="query_search_+3A_scan_params">scan_params</code></td>
<td>
<p>An optional list containing explicit nested search terms.</p>
</td></tr>
<tr><td><code id="query_search_+3A_num_attempts">num_attempts</code></td>
<td>
<p>An integer specifying the number of attempts to try the query in case of failure.</p>
</td></tr>
<tr><td><code id="query_search_+3A_sleep_time">sleep_time</code></td>
<td>
<p>A numeric value specifying the time in seconds to wait between attempts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on the return_type, it either returns a list of PDB IDs (if &quot;entry&quot;)
or the full response from the API.
</p>

<hr>
<h2 id='QueryGroup'>Create a Grouped Query Object</h2><span id='topic+QueryGroup'></span>

<h3>Description</h3>

<p>Constructs a grouped query object for performing complex searches in RCSB PDB.
It groups multiple query objects using a specified logical operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QueryGroup(queries, logical_operator)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QueryGroup_+3A_queries">queries</code></td>
<td>
<p>A list of query objects to be grouped together.</p>
</td></tr>
<tr><td><code id="QueryGroup_+3A_logical_operator">logical_operator</code></td>
<td>
<p>A string specifying the logical operator (e.g., 'AND', 'OR') to combine the queries.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list representing the grouped query object.
</p>

<hr>
<h2 id='QueryNode'>Create a Query Node Object</h2><span id='topic+QueryNode'></span>

<h3>Description</h3>

<p>Constructs a query node, which can be a terminal node or a grouped node, for complex RCSB PDB searches. This function is used to structure queries for the search system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QueryNode(search_operator, logical_operator = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QueryNode_+3A_search_operator">search_operator</code></td>
<td>
<p>A search operator or group object.</p>
</td></tr>
<tr><td><code id="QueryNode_+3A_logical_operator">logical_operator</code></td>
<td>
<p>A string specifying the logical operator, default is NULL. Used only if the search_operator is a group.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list representing the query node.
</p>

<hr>
<h2 id='RangeOperator'>Create a Range Search Operator</h2><span id='topic+RangeOperator'></span>

<h3>Description</h3>

<p>Constructs a RangeOperator object for search operations that specify a range for attribute values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RangeOperator(
  attribute,
  from_value,
  to_value,
  include_lower = TRUE,
  include_upper = TRUE,
  negation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RangeOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be evaluated within a range.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_from_value">from_value</code></td>
<td>
<p>The starting value of the range.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_to_value">to_value</code></td>
<td>
<p>The ending value of the range.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_include_lower">include_lower</code></td>
<td>
<p>Boolean to include the lower bound in the range.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_include_upper">include_upper</code></td>
<td>
<p>Boolean to include the upper bound in the range.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_negation">negation</code></td>
<td>
<p>Boolean to negate the range condition.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'RangeOperator'.
</p>

<hr>
<h2 id='RequestOptions'>Define Options for Search Requests</h2><span id='topic+RequestOptions'></span>

<h3>Description</h3>

<p>Sets various options for RCSB PDB search requests, such as pagination and sorting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RequestOptions(
  result_start_index = NULL,
  num_results = NULL,
  sort_by = "score",
  desc = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RequestOptions_+3A_result_start_index">result_start_index</code></td>
<td>
<p>An integer specifying the starting index for result pagination, default is NULL.</p>
</td></tr>
<tr><td><code id="RequestOptions_+3A_num_results">num_results</code></td>
<td>
<p>An integer specifying the number of results to return, default is NULL.</p>
</td></tr>
<tr><td><code id="RequestOptions_+3A_sort_by">sort_by</code></td>
<td>
<p>A string indicating the attribute to sort by, default is 'score'.</p>
</td></tr>
<tr><td><code id="RequestOptions_+3A_desc">desc</code></td>
<td>
<p>A boolean indicating whether sorting should be in descending order, default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of request options.
</p>

<hr>
<h2 id='return_data_as_dataframe'>Transform Response Data into a Dataframe</h2><span id='topic+return_data_as_dataframe'></span>

<h3>Description</h3>

<p>This function converts response data from the RCSB Protein Data Bank (PDB) into a structured dataframe.
It handles null or empty responses and flattens the data for dataframe conversion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>return_data_as_dataframe(response, data_type, ids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="return_data_as_dataframe_+3A_response">response</code></td>
<td>
<p>A list containing the response data from a PDB query.</p>
</td></tr>
<tr><td><code id="return_data_as_dataframe_+3A_data_type">data_type</code></td>
<td>
<p>A string indicating the type of data contained in the response.
This parameter is not directly used in the function but might be relevant for context.</p>
</td></tr>
<tr><td><code id="return_data_as_dataframe_+3A_ids">ids</code></td>
<td>
<p>A vector of identifiers corresponding to the response data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe constructed from the response data.
Returns NULL if the response is null or empty.
</p>

<hr>
<h2 id='rPDBapi-package'>rPDBapi: A Comprehensive Interface for Accessing the Protein Data Bank</h2><span id='topic+rPDBapi'></span><span id='topic+rPDBapi-package'></span>

<h3>Description</h3>

<p>Streamlines the interaction with the RCSB Protein Data Bank (PDB) <a href="https://www.rcsb.org/">https://www.rcsb.org/</a>. This interface offers an intuitive and powerful tool for searching and retrieving a diverse range of data types from the PDB. It includes advanced functionalities like BLAST and sequence motif queries. Built upon the existing XML-based API of the PDB, it simplifies the creation of custom requests, thereby enhancing usability and flexibility for researchers.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Selcuk Korkmaz <a href="mailto:selcukorkmaz@gmail.com">selcukorkmaz@gmail.com</a> (<a href="https://orcid.org/0000-0003-4632-6850">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Bilge Eren Yamasan <a href="mailto:berenyamasan@hotmail.com">berenyamasan@hotmail.com</a> (<a href="https://orcid.org/0000-0002-6525-2503">ORCID</a>)
</p>
</li></ul>


<hr>
<h2 id='ScoredResult'>Create a Scored Result Object</h2><span id='topic+ScoredResult'></span>

<h3>Description</h3>

<p>Constructs a scored result object typically used in search results to associate an entity ID with its score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScoredResult(entity_id, score)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ScoredResult_+3A_entity_id">entity_id</code></td>
<td>
<p>A string representing the entity ID.</p>
</td></tr>
<tr><td><code id="ScoredResult_+3A_score">score</code></td>
<td>
<p>A numeric value representing the score associated with the entity.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list representing the scored result.
</p>

<hr>
<h2 id='search_graphql'>Perform a GraphQL Query to RCSB PDB</h2><span id='topic+search_graphql'></span>

<h3>Description</h3>

<p>Executes a GraphQL query against the RCSB Protein Data Bank (PDB).
It sends the query in JSON format and handles the HTTP response, including error checking.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>search_graphql(graphql_json_query)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="search_graphql_+3A_graphql_json_query">graphql_json_query</code></td>
<td>
<p>A list containing the GraphQL query in JSON format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The parsed content of the response from the PDB.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>graphql_json_query &lt;- list(query = "{entries(entry_ids:
                          [\"4LZA\", \"5RU3\"]){cell {volume, angle_beta},
                          exptl {method}}}")
result &lt;- search_graphql(graphql_json_query)
result
</code></pre>

<hr>
<h2 id='SeqMotifOperator'>Create a Sequence Motif Operator for Searches</h2><span id='topic+SeqMotifOperator'></span>

<h3>Description</h3>

<p>Constructs a SeqMotifOperator object for use in sequence motif searches within the RCSB PDB.
This operator allows specifying a pattern, sequence type, and pattern type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeqMotifOperator(pattern, sequence_type, pattern_type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SeqMotifOperator_+3A_pattern">pattern</code></td>
<td>
<p>A string representing the motif pattern to search for.</p>
</td></tr>
<tr><td><code id="SeqMotifOperator_+3A_sequence_type">sequence_type</code></td>
<td>
<p>A string indicating the type of sequence ('DNA', 'RNA', or 'PROTEIN').</p>
</td></tr>
<tr><td><code id="SeqMotifOperator_+3A_pattern_type">pattern_type</code></td>
<td>
<p>A string indicating the type of pattern ('SIMPLE', 'PROSITE', or 'REGEX').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'SeqMotifOperator' representing the sequence motif operator.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seq_motif_operator &lt;- SeqMotifOperator(pattern = "A[TU]G",
                                       sequence_type = "DNA",
                                       pattern_type = "REGEX")
seq_motif_operator
</code></pre>

<hr>
<h2 id='SequenceOperator'>Create a Sequence Operator for Sequence-Based Searches</h2><span id='topic+SequenceOperator'></span>

<h3>Description</h3>

<p>Constructs a SequenceOperator object for use in sequence-based searches within the RCSB PDB.
This operator allows specifying a sequence, its type, and cutoffs for e-value and identity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SequenceOperator(
  sequence,
  sequence_type = NULL,
  evalue_cutoff = 100,
  identity_cutoff = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SequenceOperator_+3A_sequence">sequence</code></td>
<td>
<p>A string representing the sequence to search for.</p>
</td></tr>
<tr><td><code id="SequenceOperator_+3A_sequence_type">sequence_type</code></td>
<td>
<p>Optional: a string indicating the type of sequence ('DNA', 'RNA', or 'PROTEIN').
If not provided, the type is autoresolved.</p>
</td></tr>
<tr><td><code id="SequenceOperator_+3A_evalue_cutoff">evalue_cutoff</code></td>
<td>
<p>A numeric value for the e-value cutoff in the search, default is 100.</p>
</td></tr>
<tr><td><code id="SequenceOperator_+3A_identity_cutoff">identity_cutoff</code></td>
<td>
<p>A numeric value for the identity cutoff in the search, default is 0.95.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'SequenceOperator' representing the sequence operator.
</p>

<hr>
<h2 id='StructureOperator'>Create a Structure Operator for Structure-Based Searches</h2><span id='topic+StructureOperator'></span>

<h3>Description</h3>

<p>Constructs a StructureOperator object for use in structure-based searches within the RCSB PDB.
This operator allows specifying a PDB entry ID, assembly ID, and the search mode.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StructureOperator(
  pdb_entry_id,
  assembly_id = 1,
  search_mode = "STRICT_SHAPE_MATCH"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="StructureOperator_+3A_pdb_entry_id">pdb_entry_id</code></td>
<td>
<p>A string representing the PDB entry ID to search for.</p>
</td></tr>
<tr><td><code id="StructureOperator_+3A_assembly_id">assembly_id</code></td>
<td>
<p>An integer representing the assembly ID, default is 1.</p>
</td></tr>
<tr><td><code id="StructureOperator_+3A_search_mode">search_mode</code></td>
<td>
<p>A string indicating the search mode, such as 'STRICT_SHAPE_MATCH', default is 'STRICT_SHAPE_MATCH'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'StructureOperator' representing the structure search operator.
</p>

<hr>
<h2 id='walk_nested_dict'>Recursively Walk Through a Nested Dictionary</h2><span id='topic+walk_nested_dict'></span>

<h3>Description</h3>

<p>This function performs a recursive search through a nested dictionary-like structure in R,
looking for a specific term and collecting its values. It's useful for extracting specific pieces
of data from complex, deeply nested results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>walk_nested_dict(my_result, term, outputs = list(), depth = 0, maxdepth = 25)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="walk_nested_dict_+3A_my_result">my_result</code></td>
<td>
<p>The nested dictionary-like structure to search through.</p>
</td></tr>
<tr><td><code id="walk_nested_dict_+3A_term">term</code></td>
<td>
<p>The term to search for within the nested dictionary.</p>
</td></tr>
<tr><td><code id="walk_nested_dict_+3A_outputs">outputs</code></td>
<td>
<p>An initially empty list to store the results of the search, default is an empty list.</p>
</td></tr>
<tr><td><code id="walk_nested_dict_+3A_depth">depth</code></td>
<td>
<p>The current depth of the recursion, default is 0.</p>
</td></tr>
<tr><td><code id="walk_nested_dict_+3A_maxdepth">maxdepth</code></td>
<td>
<p>The maximum depth to recurse, default is 25.
If exceeded, the function issues a warning and returns NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of values associated with the term found in the nested dictionary.
Returns NULL if the term is not found or if maximum recursion depth is exceeded.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
