<!DOCTYPE html><html lang="en"><head><title>Help for package rPDBapi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rPDBapi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rPDBapi-package'><p>rPDBapi: A Comprehensive Interface for Accessing the Protein Data Bank</p></a></li>
<li><a href='#add_property'><p>Add or Merge Properties for RCSB PDB Data Fetching</p></a></li>
<li><a href='#autoresolve_sequence_type'><p>Automatically Determine the Sequence Type</p></a></li>
<li><a href='#ChemicalOperator'><p>Create a Chemical Search Operator for SMILES/InChI Descriptors</p></a></li>
<li><a href='#ComparisonOperator'><p>Create a Comparison Search Operator</p></a></li>
<li><a href='#ContainsPhraseOperator'><p>Create a Contains Phrase Search Operator</p></a></li>
<li><a href='#ContainsWordsOperator'><p>Create a Contains Words Search Operator</p></a></li>
<li><a href='#data_fetcher'><p>Fetch RCSB PDB Data Based on Specified Criteria</p></a></li>
<li><a href='#DefaultOperator'><p>Create a Default Search Operator</p></a></li>
<li><a href='#describe_chemical'><p>Describe Chemical Compound from RCSB PDB</p></a></li>
<li><a href='#ExactMatchOperator'><p>Create an Exact Match Search Operator</p></a></li>
<li><a href='#ExistsOperator'><p>Create an Existence Search Operator</p></a></li>
<li><a href='#fetch_data'><p>Fetch Data from RCSB PDB Using a JSON Query</p></a></li>
<li><a href='#find_papers'><p>Search for and Retrieve Paper Titles from PDB</p></a></li>
<li><a href='#find_results'><p>Retrieve Specific Fields for Search Results from RCSB PDB</p></a></li>
<li><a href='#generate_json_query'><p>Generate a JSON Query for RCSB PDB Data Retrieval</p></a></li>
<li><a href='#get_fasta_from_rcsb_entry'><p>Retrieve FASTA Sequence from PDB Entry or Specific Chain</p></a></li>
<li><a href='#get_info'><p>Retrieve Information for a Given PDB ID</p></a></li>
<li><a href='#get_pdb_api_url'><p>Generate a PDB API URL</p></a></li>
<li><a href='#get_pdb_file'><p>Download and Process PDB Files from the RCSB Database</p></a></li>
<li><a href='#handle_api_errors'><p>Handle API Errors</p></a></li>
<li><a href='#infer_search_service'><p>Infer the Appropriate Search Service for RCSB PDB Queries</p></a></li>
<li><a href='#InOperator'><p>Create an Inclusion Search Operator</p></a></li>
<li><a href='#parse_fasta_text_to_list'><p>Helper Function: Parse FASTA Text to List Grouped by Header</p></a></li>
<li><a href='#parse_response'><p>Parse API Response</p></a></li>
<li><a href='#perform_search'><p>Perform a Search in the RCSB PDB</p></a></li>
<li><a href='#query_search'><p>Search Query Function</p></a></li>
<li><a href='#QueryGroup'><p>Create a Grouped Query Object for RCSB PDB Searches</p></a></li>
<li><a href='#QueryNode'><p>Create a Query Node for RCSB PDB Searches</p></a></li>
<li><a href='#RangeOperator'><p>Create a Range Search Operator</p></a></li>
<li><a href='#RequestOptions'><p>Define Request Options for RCSB PDB Search Queries</p></a></li>
<li><a href='#return_data_as_dataframe'><p>Convert RCSB PDB Response Data into a Dataframe</p></a></li>
<li><a href='#ScoredResult'><p>Create a Scored Result Object for PDB Searches</p></a></li>
<li><a href='#search_graphql'><p>Perform a GraphQL Query to RCSB PDB</p></a></li>
<li><a href='#send_api_request'><p>Send API Request to a Specified URL</p></a></li>
<li><a href='#SeqMotifOperator'><p>Create a Sequence Motif Operator for RCSB PDB Searches</p></a></li>
<li><a href='#SequenceOperator'><p>Create a Sequence Operator for Sequence-Based Searches</p></a></li>
<li><a href='#StructureOperator'><p>Create a Structure Operator for Structure-Based Searches</p></a></li>
<li><a href='#walk_nested_dict'><p>Recursively Walk Through a Nested Dictionary</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Comprehensive Interface for Accessing the Protein Data Bank</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Streamlines the interaction with the 'RCSB' Protein Data Bank ('PDB') <a href="https://www.rcsb.org/">https://www.rcsb.org/</a>. This interface offers an intuitive and 
            powerful tool for searching and retrieving a diverse range of data types from the 'PDB'. It includes advanced functionalities like 
            BLAST and sequence motif queries. Built upon the existing XML-based API of the 'PDB', it simplifies the creation of custom requests, 
            thereby enhancing usability and flexibility for researchers.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, purrr, httr, jsonlite, xml2, bio3d, magrittr, methods,
testthat</td>
</tr>
<tr>
<td>Suggests:</td>
<td>r3dmol</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-19 13:07:01 UTC; selcukkorkmaz</td>
</tr>
<tr>
<td>Author:</td>
<td>Selcuk Korkmaz <a href="https://orcid.org/0000-0003-4632-6850"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Bilge Eren Yamasan
    <a href="https://orcid.org/0000-0002-6525-2503"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Selcuk Korkmaz &lt;selcukorkmaz@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-19 13:20:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='rPDBapi-package'>rPDBapi: A Comprehensive Interface for Accessing the Protein Data Bank</h2><span id='topic+rPDBapi'></span><span id='topic+rPDBapi-package'></span>

<h3>Description</h3>

<p>Streamlines the interaction with the 'RCSB' Protein Data Bank ('PDB') <a href="https://www.rcsb.org/">https://www.rcsb.org/</a>. This interface offers an intuitive and powerful tool for searching and retrieving a diverse range of data types from the 'PDB'. It includes advanced functionalities like BLAST and sequence motif queries. Built upon the existing XML-based API of the 'PDB', it simplifies the creation of custom requests, thereby enhancing usability and flexibility for researchers.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Selcuk Korkmaz <a href="mailto:selcukorkmaz@gmail.com">selcukorkmaz@gmail.com</a> (<a href="https://orcid.org/0000-0003-4632-6850">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Bilge Eren Yamasan <a href="mailto:berenyamasan@hotmail.com">berenyamasan@hotmail.com</a> (<a href="https://orcid.org/0000-0002-6525-2503">ORCID</a>)
</p>
</li></ul>


<hr>
<h2 id='add_property'>Add or Merge Properties for RCSB PDB Data Fetching</h2><span id='topic+add_property'></span>

<h3>Description</h3>

<p>This function facilitates the management of properties and subproperties required for data retrieval from the Protein Data Bank (PDB).
It accepts a list of properties where each key represents a property category (e.g., 'cell', 'exptl'), and the corresponding value is a character vector of subproperties (e.g., 'volume', 'method').
The function ensures that if a property already exists, its subproperties are merged without duplication, guaranteeing that each subproperty remains unique.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_property(property)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_property_+3A_property">property</code></td>
<td>
<p>A list where each element corresponds to a property category. The names of the list elements are the properties, and their values are character vectors containing the subproperties.
Each subproperty should be provided as a character vector. The full list of available properties and their descriptions can be found at <a href="https://data.rcsb.org/#data-schema">https://data.rcsb.org/#data-schema</a>.
</p>
<p>For example, a 'property' list might look like:
</p>

<dl>
<dt>cell</dt><dd><p>A character vector of subproperties like <code>"length_a"</code>, <code>"length_b"</code>, <code>"length_c"</code>.</p>
</dd>
<dt>exptl</dt><dd><p>A character vector of subproperties like <code>"method"</code>.</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'add_property' function is particularly useful when users need to dynamically build or update a list of properties required for complex queries in the PDB. By automatically handling duplicate entries, this function streamlines the process of constructing property lists, which can then be used in subsequent data retrieval operations.
</p>
<p>The function operates as follows:
</p>

<ol>
<li><p> Checks if the input 'property' is a list. If not, it throws an error.
</p>
</li>
<li><p> Iterates through each property in the list, ensuring that subproperties are unique and in character vector format.
</p>
</li>
<li><p> If a property already exists in the list, it merges the subproperties while eliminating duplicates.
</p>
</li></ol>



<h3>Value</h3>

<p>A modified list that consolidates the input properties. If a property already exists in the input list, its subproperties are merged, removing any duplicates.
</p>


<h3>Note</h3>

<p>It is important to ensure that the subproperties are correctly formatted as character vectors. The function does not modify the format of the subproperties.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fetch_data">fetch_data</a></code>, <code><a href="#topic+query_search">query_search</a></code> for related functions that utilize properties in querying the PDB.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:
properties &lt;- list(cell = c("length_a", "length_b", "length_c"), exptl = c("method"))
# Add new properties or merge existing ones
updated_properties &lt;- add_property(properties)
print(updated_properties)

</code></pre>

<hr>
<h2 id='autoresolve_sequence_type'>Automatically Determine the Sequence Type</h2><span id='topic+autoresolve_sequence_type'></span>

<h3>Description</h3>

<p>The 'autoresolve_sequence_type' function analyzes the characters in a given sequence to determine whether it is a DNA, RNA, or protein sequence.
The function uses standard IUPAC nucleotide and amino acid codes to classify the sequence based on its composition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autoresolve_sequence_type(sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="autoresolve_sequence_type_+3A_sequence">sequence</code></td>
<td>
<p>A string representing the nucleotide or protein sequence to be analyzed. The sequence should be composed of characters corresponding to standard IUPAC codes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string indicating the resolved sequence type: 'DNA', 'RNA', or 'PROTEIN'.
If the sequence contains ambiguous characters or does not fit clearly into one of these categories, an error is thrown.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of determining the sequence type for a DNA sequence
seq_type_dna &lt;- autoresolve_sequence_type("ATGCGTACGTAGC")
print(seq_type_dna)  # Should return "DNA"

# Example of determining the sequence type for a protein sequence
seq_type_protein &lt;- autoresolve_sequence_type("MVLSPADKTNVKAAW")
print(seq_type_protein)  # Should return "PROTEIN"

# Example of an ambiguous sequence that causes an error
# autoresolve_sequence_type("ATGB")  # Should throw an error due to ambiguity
</code></pre>

<hr>
<h2 id='ChemicalOperator'>Create a Chemical Search Operator for SMILES/InChI Descriptors</h2><span id='topic+ChemicalOperator'></span>

<h3>Description</h3>

<p>The 'ChemicalOperator' function constructs an operator object used for chemical searches within the RCSB Protein Data Bank (PDB). This function is particularly useful for querying the PDB database using chemical structure descriptors, such as SMILES (Simplified Molecular Input Line Entry System) or InChI (International Chemical Identifier) strings. The function supports various matching criteria to tailor the search results according to specific needs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChemicalOperator(descriptor, matching_criterion = "graph-strict")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ChemicalOperator_+3A_descriptor">descriptor</code></td>
<td>
<p>A string representing the chemical structure in either SMILES or InChI format. The function automatically detects the format based on the input string. If the descriptor starts with &quot;InChI=&quot;, it is treated as an InChI string; otherwise, it is assumed to be a SMILES string.</p>
</td></tr>
<tr><td><code id="ChemicalOperator_+3A_matching_criterion">matching_criterion</code></td>
<td>
<p>A string specifying the criterion for matching the chemical structure. The matching criterion determines how closely the input descriptor should match the structures in the PDB database. The possible values are predefined in the 'DescriptorMatchingCriterion' list, with &quot;graph-strict&quot; being the default. Other options may include &quot;graph-exact,&quot; &quot;graph-relaxed,&quot; and &quot;fingerprint-similarity,&quot; among others.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'ChemicalOperator' function is designed for advanced users who need to search for chemical structures in the PDB using specific descriptors. The function allows flexibility in defining the level of matching precision, making it suitable for both exact and fuzzy searches.
</p>
<p>The matching criteria provided by the 'matching_criterion' argument allow users to control the strictness of the search. For example:
</p>

<dl>
<dt>graph-strict</dt><dd><p>Matches chemical structures based on atom type, bond order, and chirality, with strict graph matching.</p>
</dd>
<dt>graph-relaxed</dt><dd><p>Allows for a more relaxed matching by ignoring certain structural details.</p>
</dd>
<dt>fingerprint-similarity</dt><dd><p>Uses molecular fingerprints to find similar structures based on a similarity threshold.</p>
</dd>
</dl>



<h3>Value</h3>

<p>The function returns a list structured as a 'ChemicalOperator' object. This object contains the input descriptor, the type of descriptor (SMILES or InChI), and the specified matching criterion. The resulting 'ChemicalOperator' object can be used in subsequent functions that perform chemical searches in the PDB database.
</p>


<h3>See Also</h3>

<p>'perform_search' for executing a search using the created 'ChemicalOperator'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Search for a chemical using a SMILES string
smiles_operator &lt;- ChemicalOperator(descriptor = "C1=CC=CC=C1", matching_criterion = "graph-strict")
smiles_operator

# Example 2: Search using an InChI string with a relaxed matching criterion
inchi_operator &lt;- ChemicalOperator(descriptor = "InChI=1S/C7H8O/c1-6-2-4-7(9)5-3-6/h2-5,9H,1H3",
                                    matching_criterion = "graph-relaxed")
inchi_operator

</code></pre>

<hr>
<h2 id='ComparisonOperator'>Create a Comparison Search Operator</h2><span id='topic+ComparisonOperator'></span>

<h3>Description</h3>

<p>Constructs a 'ComparisonOperator' object for search operations that perform comparison checks on attribute values.
This operator allows for evaluating attributes using comparison operators such as 'equal', 'greater_than',
or 'less_than', making it suitable for numerical and date-based searches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComparisonOperator(attribute, value, comparison_type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ComparisonOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be compared. This should be the field within the RCSB PDB that you want to evaluate.</p>
</td></tr>
<tr><td><code id="ComparisonOperator_+3A_value">value</code></td>
<td>
<p>The value to compare against. This is the reference value for the comparison.</p>
</td></tr>
<tr><td><code id="ComparisonOperator_+3A_comparison_type">comparison_type</code></td>
<td>
<p>A string specifying the type of comparison (e.g., 'equal', 'greater_than', 'less_than').
Supported comparison types are 'equal', 'not_equal', 'greater_than', 'less_than', etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ComparisonOperator' that can be used in search queries to retrieve entries
where the attribute meets the specified comparison criteria.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Search for entries where an attribute equals a specific value
operator &lt;- ComparisonOperator(attribute = "rcsb_entry_info.resolution_combined",
                                value = 2.0, comparison_type = "EQUAL")
operator
</code></pre>

<hr>
<h2 id='ContainsPhraseOperator'>Create a Contains Phrase Search Operator</h2><span id='topic+ContainsPhraseOperator'></span>

<h3>Description</h3>

<p>Constructs a 'ContainsPhraseOperator' object for search operations that look for attributes containing a specific phrase.
This operator is ideal for scenarios where the search needs to be more precise than just individual words,
such as finding an exact phrase within a text attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContainsPhraseOperator(attribute, value)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ContainsPhraseOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be evaluated. This should be the text field within the RCSB PDB
that you want to search against.</p>
</td></tr>
<tr><td><code id="ContainsPhraseOperator_+3A_value">value</code></td>
<td>
<p>The phrase to search for in the attribute. The search will look for this exact sequence of words
within the specified attribute.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ContainsPhraseOperator' that can be used in search queries to retrieve entries
where the attribute contains the specified phrase.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Search for entries containing a specific phrase in an attribute
operator &lt;- ContainsPhraseOperator(attribute = "rcsb_primary_citation.title",
                                    value = "molecular dynamics")
print(operator)
</code></pre>

<hr>
<h2 id='ContainsWordsOperator'>Create a Contains Words Search Operator</h2><span id='topic+ContainsWordsOperator'></span>

<h3>Description</h3>

<p>Constructs a 'ContainsWordsOperator' object for search operations that look for attributes containing specific words.
This operator is particularly useful for text-based searches where the goal is to find entries that include
particular keywords or phrases within a specified attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContainsWordsOperator(attribute, value)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ContainsWordsOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be evaluated. This should be the text field within the RCSB PDB
that you want to search against.</p>
</td></tr>
<tr><td><code id="ContainsWordsOperator_+3A_value">value</code></td>
<td>
<p>The words to search for in the attribute. This can be a single word or a set of words,
and the search will return entries containing any of these words in the specified attribute.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ContainsWordsOperator' that can be used in search queries to retrieve entries
where the attribute contains the specified words.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Search for entries containing specific words in an attribute
operator &lt;- ContainsWordsOperator(attribute = "rcsb_primary_citation.title",
                                  value = "crystal structure")
print(operator)
</code></pre>

<hr>
<h2 id='data_fetcher'>Fetch RCSB PDB Data Based on Specified Criteria</h2><span id='topic+data_fetcher'></span>

<h3>Description</h3>

<p>The 'data_fetcher' function provides a flexible way to access data from the RCSB Protein Data Bank (PDB). By specifying an identifier, data type, and a set of properties, users can tailor the data retrieval process to meet their specific research needs. The function integrates several steps, including validating IDs, generating a JSON query, fetching the data, and formatting the response.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_fetcher(
  id = NULL,
  data_type = "ENTRY",
  properties = NULL,
  return_as_dataframe = TRUE,
  verbosity = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="data_fetcher_+3A_id">id</code></td>
<td>
<p>A single identifier or a list of identifiers for the data to be fetched. These IDs correspond to the entries, assemblies, polymer entities, or other entities within the RCSB PDB. The ID must match the data type you are querying (e.g., PDB ID for entries, assembly ID for assemblies).</p>
</td></tr>
<tr><td><code id="data_fetcher_+3A_data_type">data_type</code></td>
<td>
<p>A string specifying the type of data to fetch. The available options for <code>data_type</code> include:
</p>

<dl>
<dt><code>"ENTRY"</code></dt><dd><p>Fetches data related to PDB entries. This is the default option and returns basic entry-level information.</p>
</dd>
<dt><code>"ASSEMBLY"</code></dt><dd><p>Fetches data related to biological assemblies. The IDs should be formatted as <code>"PDB_ID-ASSEMBLY_ID"</code>.</p>
</dd>
<dt><code>"POLYMER_ENTITY"</code></dt><dd><p>Fetches data related to polymeric molecular entities, such as protein chains.</p>
</dd>
<dt><code>"BRANCHED_ENTITY"</code></dt><dd><p>Fetches data related to branched entities, such as carbohydrates or other branched molecules.</p>
</dd>
<dt><code>"NONPOLYMER_ENTITY"</code></dt><dd><p>Fetches data related to non-polymeric entities, such as ligands, cofactors, or small molecules.</p>
</dd>
<dt><code>"POLYMER_ENTITY_INSTANCE"</code></dt><dd><p>Fetches data related to specific instances of polymeric entities, also known as chains.</p>
</dd>
<dt><code>"BRANCHED_ENTITY_INSTANCE"</code></dt><dd><p>Fetches data related to specific instances of branched entities.</p>
</dd>
<dt><code>"NONPOLYMER_ENTITY_INSTANCE"</code></dt><dd><p>Fetches data related to specific instances of non-polymeric entities.</p>
</dd>
<dt><code>"CHEMICAL_COMPONENT"</code></dt><dd><p>Fetches data related to chemical components, such as individual atoms or molecular fragments.</p>
</dd>
</dl>

<p>Each <code>data_type</code> corresponds to a specific hierarchy level within the PDB data structure, and selecting the appropriate type ensures that you retrieve relevant and accurate data.</p>
</td></tr>
<tr><td><code id="data_fetcher_+3A_properties">properties</code></td>
<td>
<p>A list or dictionary of properties to be included in the data fetching process. The properties should match the data type you are querying. For example, if you are fetching <code>POLYMER_ENTITY</code> data, you might specify properties such as <code>"rcsb_entity_source_organism"</code> or <code>"rcsb_cluster_membership"</code>.
The <code>properties</code> argument allows users to customize the data they retrieve by specifying exactly which attributes of the data type they are interested in. It is important to match the properties to the correct data type to ensure accurate and meaningful results.
The <a href="https://data.rcsb.org/#data-schema">RCSB PDB Data Schema</a> provides detailed documentation on the properties available for each data type. Users are encouraged to refer to this resource when selecting properties for their queries.</p>
</td></tr>
<tr><td><code id="data_fetcher_+3A_return_as_dataframe">return_as_dataframe</code></td>
<td>
<p>A boolean indicating whether to return the response as a dataframe. If <code>TRUE</code>, the data is returned in a structured dataframe format, which is convenient for analysis and manipulation in R. If <code>FALSE</code>, the data is returned in its original format, which may be a nested list or JSON-like structure. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="data_fetcher_+3A_verbosity">verbosity</code></td>
<td>
<p>A boolean flag indicating whether to print status messages during the function execution. When set to <code>TRUE</code>, the function will output messages detailing the progress and any issues encountered.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'data_fetcher' function is particularly useful for researchers who need to access and analyze specific subsets of PDB data. By providing a list of IDs and the corresponding data type, users can retrieve only the information relevant to their study, reducing the need to manually filter or process large datasets. The function also supports fetching multiple properties simultaneously, allowing for a more comprehensive data retrieval process.
</p>


<h3>Value</h3>

<p>Depending on the value of <code>return_as_dataframe</code>, this function returns either a dataframe or the raw data in its original format. The dataframe format is particularly useful for further data analysis and visualization within R, while the raw format may be preferred for more complex or custom data processing tasks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Fetching basic entry information
properties &lt;- list(cell = c("length_a", "length_b", "length_c"), exptl = c("method"))
data_fetcher(
  id = c("4HHB"),
  data_type = "ENTRY",
  properties = properties,
  return_as_dataframe = TRUE
)

# Example 2: Fetching polymer entity data
properties &lt;- list(
  rcsb_entity_source_organism = c("ncbi_taxonomy_id", "ncbi_scientific_name"),
  rcsb_cluster_membership = c("cluster_id", "identity")
)
data_fetcher(
  id = c("4HHB_1", "12CA_1"),
  data_type = "POLYMER_ENTITY",
  properties = properties,
  return_as_dataframe = TRUE
)

# Example 3: Fetching non-polymer entity data
properties &lt;- list(
  rcsb_nonpolymer_entity = c("details", "formula_weight", "pdbx_description"),
  rcsb_nonpolymer_entity_container_identifiers = c("chem_ref_def_id")
)
data_fetcher(
  id = c("3PQR_5", "3PQR_6"),
  data_type = "NONPOLYMER_ENTITY",
  properties = properties,
  return_as_dataframe = TRUE
)

# Example 4: Fetching chemical component data
properties &lt;- list(
  rcsb_id = list(),
  chem_comp = list("type", "formula_weight", "name", "formula"),
  rcsb_chem_comp_info = list("initial_release_date")
)
data_fetcher(
  id = c("NAG", "EBW"),
  data_type = "CHEMICAL_COMPONENT",
  properties = properties,
  return_as_dataframe = TRUE
)

</code></pre>

<hr>
<h2 id='DefaultOperator'>Create a Default Search Operator</h2><span id='topic+DefaultOperator'></span>

<h3>Description</h3>

<p>Constructs a 'DefaultOperator' object for use in general search operations within the RCSB PDB.
This operator is used when a simple, non-specific search operation is needed, based on a single value.
The 'DefaultOperator' can be employed in scenarios where the search criteria do not require complex conditions
or additional logic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DefaultOperator(value)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DefaultOperator_+3A_value">value</code></td>
<td>
<p>The value to be used in the search operation. This is the key criterion for the search
and should be a valid string or numeric value that the search will use as the matching term.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'DefaultOperator' representing the default search operator.
This object can be used directly in query formulations or further manipulated within complex search logic.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a basic search operator with a specific value
operator &lt;- DefaultOperator("4HHB")
print(operator)
</code></pre>

<hr>
<h2 id='describe_chemical'>Describe Chemical Compound from RCSB PDB</h2><span id='topic+describe_chemical'></span>

<h3>Description</h3>

<p>Retrieves detailed information about a chemical compound from the RCSB Protein Data Bank (PDB) based on its chemical ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>describe_chemical(chem_id, url_root = URL_ROOT)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="describe_chemical_+3A_chem_id">chem_id</code></td>
<td>
<p>A string representing the 3-character chemical ID. This ID is typically an alphanumeric string used to uniquely identify ligands, cofactors, or other small molecules within macromolecular structures. The string must not exceed 3 characters.</p>
</td></tr>
<tr><td><code id="describe_chemical_+3A_url_root">url_root</code></td>
<td>
<p>A string representing the URL for retrieving information about chemical compounds. By default, this is set to the global constant <code>URL_ROOT</code>, but users can specify a different URL if needed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing detailed information about the chemical compound. This list includes various fields such as:
</p>

<dl>
<dt>rcsb_chem_comp_descriptor</dt><dd><p>A sublist containing chemical descriptors like SMILES, InChI strings, molecular weight, and other chemical properties.</p>
</dd>
<dt>rcsb_chem_comp_info</dt><dd><p>Information regarding the compound’s classification, formula, and additional relevant data.</p>
</dd>
<dt>other</dt><dd><p>Other fields may also be included, depending on the specific compound and the data available from the RCSB PDB.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Retrieve chemical information for N-Acetyl-D-Glucosamine (NAG)
chem_desc &lt;- describe_chemical('NAG')
chem_desc

# Access the SMILES string of the compound
smiles_string &lt;- chem_desc$rcsb_chem_comp_descriptor$smiles
smiles_string


## End(Not run)
</code></pre>

<hr>
<h2 id='ExactMatchOperator'>Create an Exact Match Search Operator</h2><span id='topic+ExactMatchOperator'></span>

<h3>Description</h3>

<p>Constructs an 'ExactMatchOperator' object for precise search operations within the RCSB PDB.
This operator is designed to match an exact attribute value, making it ideal for searches where
specificity is required. For example, if you need to find all entries that exactly match a certain
attribute value, this operator will ensure only those precise matches are returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExactMatchOperator(attribute, value)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ExactMatchOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to match. This should be the name of the field within the RCSB PDB
that you want to search against.</p>
</td></tr>
<tr><td><code id="ExactMatchOperator_+3A_value">value</code></td>
<td>
<p>The exact value to search for. This is the specific value of the attribute you are interested in.
The search will return only those records where the attribute exactly matches this value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ExactMatchOperator'. This object can be used in search queries
to retrieve precise matches within the RCSB PDB database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Search for entries with an exact match to a given attribute
operator &lt;- ExactMatchOperator(attribute = "rcsb_entry_info.resolution_combined",
                                value = "2.0")
print(operator)
</code></pre>

<hr>
<h2 id='ExistsOperator'>Create an Existence Search Operator</h2><span id='topic+ExistsOperator'></span>

<h3>Description</h3>

<p>Constructs an 'ExistsOperator' object for search operations to check the existence of an attribute.
This operator is useful in queries where you need to ensure that a certain attribute is present
within the entries being searched, regardless of its value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExistsOperator(attribute)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ExistsOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute whose existence is to be checked. This should be the field within the RCSB PDB
that you want to ensure is present in the search results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'ExistsOperator' that can be used in search queries to retrieve entries
where the specified attribute exists.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Search for entries where a specific attribute exists
operator &lt;- ExistsOperator(attribute = "rcsb_primary_citation.doi")
print(operator)
</code></pre>

<hr>
<h2 id='fetch_data'>Fetch Data from RCSB PDB Using a JSON Query</h2><span id='topic+fetch_data'></span>

<h3>Description</h3>

<p>This function sends a GraphQL JSON query to the RCSB Protein Data Bank (PDB) to fetch data corresponding to a specified set of IDs.
It is designed to handle the complexities of interacting with the PDB's GraphQL API, ensuring that errors in the query process
are handled gracefully and that users are informed of any discrepancies in the expected and returned data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetch_data(json_query, data_type, ids)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fetch_data_+3A_json_query">json_query</code></td>
<td>
<p>A JSON string representing the query to be sent to the PDB. This query must be well-formed and adhere to the GraphQL
query structure required by the PDB's API. It typically includes details such as the fields to be retrieved and the conditions for
data selection.</p>
</td></tr>
<tr><td><code id="fetch_data_+3A_data_type">data_type</code></td>
<td>
<p>A string indicating the type of data to be fetched. While this parameter is not directly used in the function,
it can provide context for the data being retrieved, such as &quot;ENTRY&quot;, &quot;ASSEMBLY&quot;, &quot;POLYMER_ENTITY&quot;, etc.</p>
</td></tr>
<tr><td><code id="fetch_data_+3A_ids">ids</code></td>
<td>
<p>A character vector of identifiers for which data is being requested. These IDs should correspond to valid entries in
the PDB and should match the data structure expected by the PDB's API.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs several checks and operations:
</p>
<p>* It validates the 'json_query' to ensure that it is neither 'NULL' nor empty.
* It attempts to send the query to the PDB's GraphQL endpoint using a helper function (assumed to be 'search_graphql').
* It checks the server's response to determine if the query was successful or if any errors were encountered.
* If the data returned does not match the expected IDs, the function issues warnings and stops execution, providing details on the missing IDs.
* The function ensures that the returned data is correctly named according to the provided IDs.
</p>
<p>The function is particularly useful for developers and researchers who need to programmatically access and manipulate large datasets
from the PDB. It abstracts away some of the complexity of directly interacting with the PDB's API, providing a more user-friendly
interface for data retrieval.
</p>


<h3>Value</h3>

<p>A list containing the data fetched from the PDB, with the names of the list elements set to the corresponding IDs.
If any issues occur during the fetching process (e.g., if some IDs are not found), the function will return 'NULL' and provide
informative error messages to help diagnose the problem.
</p>

<hr>
<h2 id='find_papers'>Search for and Retrieve Paper Titles from PDB</h2><span id='topic+find_papers'></span>

<h3>Description</h3>

<p>This function searches the Protein Data Bank (PDB) for scholarly articles related to a specified search term.
It retrieves the titles of up to a specified maximum number of papers associated with PDB entries.
The function relies on 'query_search' to perform the initial search and 'get_info' to fetch detailed information
for each PDB entry, including the citation titles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_papers(search_term, max_results = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_papers_+3A_search_term">search_term</code></td>
<td>
<p>A string specifying the term to search for in the PDB. This term can relate to any aspect of
the PDB entries, such as keywords, molecular functions, or specific proteins (e.g., &quot;CRISPR&quot;).</p>
</td></tr>
<tr><td><code id="find_papers_+3A_max_results">max_results</code></td>
<td>
<p>An integer indicating the maximum number of paper titles to retrieve. Defaults to 10.
The function will retrieve the titles for the first 'max_results' PDB entries returned by the search.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is useful for researchers who want to quickly find relevant literature associated with specific
PDB entries. The process involves two main steps:
</p>

<ol>
<li><p> **Search Query**: The function uses 'query_search' to find PDB entries matching the search term.
</p>
</li>
<li><p> **Fetching Paper Titles**: For each PDB ID returned by the search, 'get_info' is used to retrieve
detailed information, including the titles of any associated citations.
</p>
</li></ol>

<p>The function includes robust error handling to manage cases where the search term does not return any results,
or where there are issues retrieving details for specific PDB entries. Warnings are provided if no citations are
found for a given PDB ID or if other issues are encountered.
</p>


<h3>Value</h3>

<p>A named list where each element's name is a PDB ID and its value is the title of the corresponding paper.
If no papers are found or an error occurs, the function returns an empty list with warnings or error messages
to help diagnose the issue.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Find papers related to CRISPR and retrieve up to 5 paper titles
crispr_papers &lt;- find_papers("CRISPR", max_results = 5)
print(crispr_papers)


</code></pre>

<hr>
<h2 id='find_results'>Retrieve Specific Fields for Search Results from RCSB PDB</h2><span id='topic+find_results'></span>

<h3>Description</h3>

<p>This function searches the Protein Data Bank (PDB) for entries related to a specified search term and retrieves
specific information from those entries. It is useful for extracting targeted data from search results, such as
citations, experimental methods, or structural details. The function leverages 'query_search' to perform the
initial search and 'get_info' to fetch detailed data for each PDB entry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_results(search_term, field = "citation")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_results_+3A_search_term">search_term</code></td>
<td>
<p>A string specifying the term to search for in the PDB. This term can relate to various aspects
of the PDB entries, such as keywords, molecular functions, protein names, or specific research areas.</p>
</td></tr>
<tr><td><code id="find_results_+3A_field">field</code></td>
<td>
<p>A string indicating the specific field to retrieve for each search result. The default is &quot;citation&quot;.
The field should correspond to one of the following valid options:
</p>

<dl>
<dt>&quot;citation&quot;</dt><dd><p>Information about the primary citation associated with the PDB entry.</p>
</dd>
<dt>&quot;audit_author&quot;</dt><dd><p>Details about the authors who contributed to the PDB entry.</p>
</dd>
<dt>&quot;cell&quot;</dt><dd><p>Cell dimensions and related crystallographic information.</p>
</dd>
<dt>&quot;diffrn&quot;</dt><dd><p>Information about the diffraction experiment.</p>
</dd>
<dt>&quot;diffrn_detector&quot;</dt><dd><p>Details about the detector used in the diffraction experiment.</p>
</dd>
<dt>&quot;diffrn_radiation&quot;</dt><dd><p>Radiation source details used in the diffraction experiment.</p>
</dd>
<dt>&quot;diffrn_source&quot;</dt><dd><p>Source of the radiation used in the experiment.</p>
</dd>
<dt>&quot;entry&quot;</dt><dd><p>Basic information about the PDB entry, including its identifier.</p>
</dd>
<dt>&quot;exptl&quot;</dt><dd><p>Details about the experimental methods used to determine the structure.</p>
</dd>
<dt>&quot;exptl_crystal&quot;</dt><dd><p>Information about the crystals used in the experiment.</p>
</dd>
<dt>&quot;exptl_crystal_grow&quot;</dt><dd><p>Details on the crystal growth conditions.</p>
</dd>
<dt>&quot;pdbx_sgproject&quot;</dt><dd><p>Information on the Structural Genomics Project.</p>
</dd>
<dt>&quot;pdbx_audit_revision_details&quot;</dt><dd><p>Details of any revisions made to the PDB entry.</p>
</dd>
<dt>&quot;pdbx_audit_revision_history&quot;</dt><dd><p>History of the revisions for the PDB entry.</p>
</dd>
<dt>&quot;pdbx_database_related&quot;</dt><dd><p>Related database entries.</p>
</dd>
<dt>&quot;pdbx_database_status&quot;</dt><dd><p>Current status of the PDB entry in the database.</p>
</dd>
<dt>&quot;rcsb_accession_info&quot;</dt><dd><p>Accession information for the PDB entry.</p>
</dd>
<dt>&quot;rcsb_entry_container_identifiers&quot;</dt><dd><p>Identifiers associated with the entry container.</p>
</dd>
<dt>&quot;rcsb_entry_info&quot;</dt><dd><p>General information about the PDB entry.</p>
</dd>
<dt>&quot;rcsb_primary_citation&quot;</dt><dd><p>Details of the primary citation for the PDB entry.</p>
</dd>
<dt>&quot;refine&quot;</dt><dd><p>Information about the refinement of the structure.</p>
</dd>
<dt>&quot;refine_hist&quot;</dt><dd><p>History of the refinement process.</p>
</dd>
<dt>&quot;refine_ls_restr&quot;</dt><dd><p>Details about the least-squares restraints used in refinement.</p>
</dd>
<dt>&quot;reflns&quot;</dt><dd><p>Information about the reflections used in the crystallographic experiment.</p>
</dd>
<dt>&quot;reflns_shell&quot;</dt><dd><p>Details about the shell reflections used in the experiment.</p>
</dd>
<dt>&quot;software&quot;</dt><dd><p>Software used in the structure determination process.</p>
</dd>
<dt>&quot;struct&quot;</dt><dd><p>Structural information about the PDB entry.</p>
</dd>
<dt>&quot;struct_keywords&quot;</dt><dd><p>Keywords associated with the structure.</p>
</dd>
<dt>&quot;symmetry&quot;</dt><dd><p>Symmetry information of the crystal structure.</p>
</dd>
<dt>&quot;rcsb_id&quot;</dt><dd><p>The RCSB ID of the PDB entry.</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is ideal for researchers who need to extract specific data fields from multiple PDB entries efficiently.
The process involves two main steps:
</p>

<ol>
<li><p> **Search Query**: The function uses 'query_search' to find PDB entries that match the provided search term.
</p>
</li>
<li><p> **Field Retrieval**: For each PDB ID returned by the search, 'get_info' is used to retrieve the specified field.
</p>
</li></ol>

<p>Error handling is robust, with informative messages provided when the search term yields no results, when an individual
PDB entry cannot be retrieved, or when the specified field is not found in the retrieved data.
</p>


<h3>Value</h3>

<p>A named list where each element's name is a PDB ID and its value is the information for the specified field
from the corresponding search result. If no results are found, or if an error occurs during data retrieval, the
function returns an empty list with appropriate warnings or error messages.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Retrieve citation information for PDB entries related to CRISPR
crispr_citations &lt;- find_results("CRISPR", field = "citation")
crispr_citations

</code></pre>

<hr>
<h2 id='generate_json_query'>Generate a JSON Query for RCSB PDB Data Retrieval</h2><span id='topic+generate_json_query'></span>

<h3>Description</h3>

<p>This function constructs a JSON query string tailored for retrieving data from the RCSB Protein Data Bank (PDB).
It is designed to accommodate a variety of data types, such as entries, polymer entities, assemblies, and chemical components.
The function is particularly useful when specific properties need to be queried across multiple PDB identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_json_query(ids, data_type, properties)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_json_query_+3A_ids">ids</code></td>
<td>
<p>A character vector of identifiers corresponding to the data you wish to retrieve. These identifiers should
match the type of data specified in the &lsquo;data_type' argument. For example, if 'data_type' is &rsquo;ENTRY', the identifiers
should be PDB entry IDs like &quot;1XYZ&quot; or &quot;2XYZ&quot;.</p>
</td></tr>
<tr><td><code id="generate_json_query_+3A_data_type">data_type</code></td>
<td>
<p>A string indicating the type of data to query. The function supports the following types:
</p>

<dl>
<dt>&quot;ENTRY&quot;</dt><dd><p>Refers to the entire PDB entry, typically associated with the full structural dataset.</p>
</dd>
<dt>&quot;POLYMER_ENTITY&quot;</dt><dd><p>Represents specific polymer entities, such as proteins or nucleic acids, within an entry.</p>
</dd>
<dt>&quot;BRANCHED_ENTITY&quot;</dt><dd><p>Refers to branched entities, often used to describe glycans or other complex molecules.</p>
</dd>
<dt>&quot;NONPOLYMER_ENTITY&quot;</dt><dd><p>Describes non-polymeric entities, such as ligands or cofactors.</p>
</dd>
<dt>&quot;POLYMER_ENTITY_INSTANCE&quot;</dt><dd><p>Specific instances of polymer entities, usually chains within a structure.</p>
</dd>
<dt>&quot;BRANCHED_ENTITY_INSTANCE&quot;</dt><dd><p>Instances of branched entities within a structure.</p>
</dd>
<dt>&quot;NONPOLYMER_ENTITY_INSTANCE&quot;</dt><dd><p>Instances of non-polymeric entities within a structure.</p>
</dd>
<dt>&quot;ASSEMBLY&quot;</dt><dd><p>Refers to macromolecular assemblies, which can include multiple chains and entities.</p>
</dd>
<dt>&quot;CHEMICAL_COMPONENT&quot;</dt><dd><p>Specific chemical components, often small molecules or ligands, within the PDB.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="generate_json_query_+3A_properties">properties</code></td>
<td>
<p>A named list where each element represents a property to be included in the query for the corresponding
'data_type'. Each element of the list should be a character vector containing the specific properties you wish to retrieve.
For example, for 'data_type = &quot;ENTRY&quot;', properties might include 'cell = c(&quot;volume&quot;, &quot;angle_beta&quot;)' to retrieve cell volume
and angle beta of the unit cell.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is designed to be flexible and extensible, allowing users to construct complex queries with multiple properties
and data types. The function internally maps the 'data_type' to the appropriate query format and ensures that the JSON string
is properly constructed. The 'properties' argument should align with the data type; otherwise, the query might not return
the desired results.
</p>
<p>The 'generate_json_query' function is particularly useful for researchers and bioinformaticians who need to retrieve specific
datasets from the PDB, especially when dealing with large-scale structural biology data. The generated JSON query can be used
in subsequent API calls to fetch the required data in a structured and efficient manner.
</p>


<h3>Value</h3>

<p>A string representing the generated JSON query. This string is formatted according to the requirements of the RCSB PDB API
and can be used directly in a GraphQL query to retrieve the specified data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Generate a query for PDB entries with specific properties
ids &lt;- c("1XYZ", "2XYZ")
properties &lt;- list(cell = c("volume", "angle_beta"), exptl = c("method"))
json_query &lt;- generate_json_query(ids, "ENTRY", properties)
print(json_query)

# Example 2: Generate a query for chemical components with specified properties
ids &lt;- c("ATP", "NAD")
properties &lt;- list(chem_comp = c("formula_weight", "type"))
json_query &lt;- generate_json_query(ids, "CHEMICAL_COMPONENT", properties)
print(json_query)

# Example 3: Generate a query for polymer entities within a PDB entry
ids &lt;- c("1XYZ_1", "2XYZ_2")
properties &lt;- list(entity_src_gen = c("organism_scientific", "gene_src_common"))
json_query &lt;- generate_json_query(ids, "POLYMER_ENTITY", properties)
print(json_query)

</code></pre>

<hr>
<h2 id='get_fasta_from_rcsb_entry'>Retrieve FASTA Sequence from PDB Entry or Specific Chain</h2><span id='topic+get_fasta_from_rcsb_entry'></span>

<h3>Description</h3>

<p>This function retrieves FASTA sequences from the RCSB Protein Data Bank (PDB) for a specified entry ID (<code>rcsb_id</code>). It can return either the full set of sequences associated with the entry or, if specified, the sequence corresponding to a particular chain within that entry. This flexibility makes it a useful tool for bioinformaticians and structural biologists needing access to protein or nucleic acid sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fasta_from_rcsb_entry(
  rcsb_id,
  chain_id = NULL,
  verbosity = TRUE,
  fasta_base_url = FASTA_BASE_URL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_fasta_from_rcsb_entry_+3A_rcsb_id">rcsb_id</code></td>
<td>
<p>A string representing the PDB ID for which the FASTA sequence is to be retrieved. This is the primary identifier of the entry in the PDB database.</p>
</td></tr>
<tr><td><code id="get_fasta_from_rcsb_entry_+3A_chain_id">chain_id</code></td>
<td>
<p>A string representing the specific chain ID within the PDB entry for which the FASTA sequence is to be retrieved. If <code>chain_id</code> is NULL (the default), the function will return all sequences associated with the entry. The chain ID should match one of the chain identifiers in the PDB entry (e.g., &quot;A&quot;, &quot;B&quot;).</p>
</td></tr>
<tr><td><code id="get_fasta_from_rcsb_entry_+3A_verbosity">verbosity</code></td>
<td>
<p>A boolean flag indicating whether to print status messages during the function execution. When set to <code>TRUE</code> (the default), the function will output messages detailing the progress and any issues encountered.</p>
</td></tr>
<tr><td><code id="get_fasta_from_rcsb_entry_+3A_fasta_base_url">fasta_base_url</code></td>
<td>
<p>A string representing the base URL for the FASTA retrieval. By default, this is set to the global constant <code>FASTA_BASE_URL</code>, but users can specify a different URL if needed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function queries the RCSB PDB database using the provided entry ID (<code>rcsb_id</code>) and optionally a chain ID (<code>chain_id</code>). It sends an HTTP GET request to retrieve the corresponding FASTA file. The response is then parsed into a list of sequences. If a chain ID is provided, the function will return only the sequence corresponding to that chain. If no chain ID is provided, all sequences are returned.
</p>
<p>If a request fails, the function provides informative error messages. In the case of a network failure, the function will stop execution with a clear error message. Additionally, if the chain ID does not exist within the entry, the function will return an appropriate error message indicating that the chain was not found.
</p>
<p>The function also supports passing a custom base URL for the FASTA file retrieval, providing flexibility for users working with different PDB mirrors or services.
</p>


<h3>Value</h3>

<p>* If <code>chain_id</code> is NULL, the function returns a list of FASTA sequences associated with the provided <code>rcsb_id</code>, where organism names or chain descriptions are used as keys.
* If <code>chain_id</code> is specified, the function returns a character string representing the FASTA sequence for that specific chain.
* If the specified <code>chain_id</code> is not found in the PDB entry, the function will stop execution with an informative error message.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Retrieve all FASTA sequences for the entry 4HHB
all_sequences &lt;- get_fasta_from_rcsb_entry("4HHB", verbosity = TRUE)
print(all_sequences)

# Example 2: Retrieve the FASTA sequence for chain A of entry 4HHB
chain_a_sequence &lt;- get_fasta_from_rcsb_entry("4HHB", chain_id = "A", verbosity = TRUE)
print(chain_a_sequence)

</code></pre>

<hr>
<h2 id='get_info'>Retrieve Information for a Given PDB ID</h2><span id='topic+get_info'></span>

<h3>Description</h3>

<p>This function retrieves comprehensive information for a specified PDB (Protein Data Bank) entry by querying the RCSB PDB RESTful API.
The function handles HTTP requests, processes JSON responses, and can manage legacy PDB identifiers.
It is particularly useful for obtaining all available data related to a specific PDB entry, which can include metadata,
structural details, experimental methods, and more.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_info(pdb_id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_info_+3A_pdb_id">pdb_id</code></td>
<td>
<p>A string specifying the PDB entry of interest. The 'pdb_id' should be a 4-character alphanumeric code,
representing the unique identifier of a PDB entry (e.g., &quot;1XYZ&quot;).
If a legacy PDB identifier is provided in the format 'PDB_ID:CHAIN_ID', it will be automatically converted to the
new format for querying.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'get_info' function is versatile and designed for researchers who need to extract detailed structural and
experimental information from the RCSB PDB. The function is robust, providing error handling for various scenarios,
including network failures, incorrect PDB IDs, and API errors. It automatically manages legacy PDB IDs, ensuring
compatibility with the latest API standards.
</p>
<p>The output is a structured list that can be easily parsed or manipulated for further analysis, making it an essential
tool for bioinformaticians and structural biologists working with PDB data.
</p>
<p>The function also offers flexibility in querying different parts of the RCSB PDB API by adjusting the 'url_root'
parameter, allowing users to target specific datasets within the PDB.
</p>


<h3>Value</h3>

<p>A list object (an ordered dictionary in R) containing detailed information about the specified PDB entry.
The returned list includes various data fields, depending on the content available for the entry. For example,
it may contain information about the structure's authors, resolution, experiment type, macromolecules, ligands, etc.
If the data retrieval fails at any stage (e.g., network issues, invalid PDB ID, API downtime), the function will return 'NULL'
and provide an informative error message.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdb_info &lt;- get_info(pdb_id = "1XYZ")
print(pdb_info)

</code></pre>

<hr>
<h2 id='get_pdb_api_url'>Generate a PDB API URL</h2><span id='topic+get_pdb_api_url'></span>

<h3>Description</h3>

<p>This function constructs a full PDB API URL by concatenating the base URL, an API endpoint, and an identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pdb_api_url(endpoint, id, base_url = BASE_URL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_pdb_api_url_+3A_endpoint">endpoint</code></td>
<td>
<p>A character string representing the specific API endpoint to be accessed (e.g., &quot;/pdb/entry/&quot;).</p>
</td></tr>
<tr><td><code id="get_pdb_api_url_+3A_id">id</code></td>
<td>
<p>A character string representing the identifier for the resource (e.g., a PDB ID or other relevant ID).</p>
</td></tr>
<tr><td><code id="get_pdb_api_url_+3A_base_url">base_url</code></td>
<td>
<p>A string representing the base URL to generate PDB API url. By default, this is set to the global constant <code>BASE_URL</code>, but users can specify a different URL if needed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string containing the full URL for accessing the PDB API resource.
</p>

<hr>
<h2 id='get_pdb_file'>Download and Process PDB Files from the RCSB Database</h2><span id='topic+get_pdb_file'></span>

<h3>Description</h3>

<p>The 'get_pdb_file' function is a versatile tool designed to download Protein Data Bank (PDB) files from the RCSB database.
It supports various file formats such as 'pdb', 'cif', 'xml', and 'structfact', with options for file compression and handling
alternate locations (ALT) and insertion codes (INSERT) in PDB files. This function also provides the flexibility to save
the downloaded files to a specified directory or to a temporary directory for immediate use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pdb_file(
  pdb_id,
  filetype = "cif",
  rm.insert = FALSE,
  rm.alt = TRUE,
  compression = TRUE,
  save = FALSE,
  path = NULL,
  verbosity = TRUE,
  download_base_url = DOWNLOAD_BASE_URL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_pdb_file_+3A_pdb_id">pdb_id</code></td>
<td>
<p>A 4-character string specifying the PDB entry of interest (e.g., &quot;1XYZ&quot;). This identifier uniquely represents
a macromolecular structure within the PDB database.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_filetype">filetype</code></td>
<td>
<p>A string specifying the format of the file to be downloaded. The default is 'cif'. Supported file types include:
</p>

<dl>
<dt>'pdb'</dt><dd><p>The older PDB file format, which provides atomic coordinates and metadata.</p>
</dd>
<dt>'cif'</dt><dd><p>The Crystallographic Information File (CIF) format, which is a newer standard replacing PDB files.</p>
</dd>
<dt>'xml'</dt><dd><p>An XML format file, providing structured data that can be easily parsed for various applications.</p>
</dd>
<dt>'structfact'</dt><dd><p>Structure factor files in CIF format, available for certain PDB entries, containing experimental
data used to determine the structure.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_rm.insert">rm.insert</code></td>
<td>
<p>Logical flag indicating whether to ignore PDB insertion codes. Default is FALSE. If TRUE, records with insertion
codes will be removed from the final data.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_rm.alt">rm.alt</code></td>
<td>
<p>Logical flag indicating whether to ignore alternate location indicators (ALT) in PDB files. Default is TRUE. If
TRUE, only the first alternate location is kept, and others are removed.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_compression">compression</code></td>
<td>
<p>Logical flag indicating whether to download the file in a compressed format (e.g., .gz). Default is TRUE,
which is recommended for faster downloads, especially for CIF files.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_save">save</code></td>
<td>
<p>Logical flag indicating whether to save the downloaded file to a specified directory. Default is FALSE, which means
the file is processed and optionally saved, but not retained after processing unless specified.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_path">path</code></td>
<td>
<p>A string specifying the directory where the downloaded file should be saved. If NULL, the file is saved in a
temporary directory. If 'save' is TRUE, this path is required.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_verbosity">verbosity</code></td>
<td>
<p>A boolean flag indicating whether to print status messages during the function execution.</p>
</td></tr>
<tr><td><code id="get_pdb_file_+3A_download_base_url">download_base_url</code></td>
<td>
<p>A string representing the base URL for the PDB file retrieval. By default, this is set to the global constant <code>DOWNLOAD_BASE_URL</code>, but users can specify a different URL if needed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'get_pdb_file' function is an essential tool for structural biologists and bioinformaticians who need to download and
process PDB files for further analysis. By providing options to handle alternate locations and insertion codes, this function
ensures that the data is clean and ready for downstream applications. Additionally, the ability to save files locally or
work with them in a temporary directory provides flexibility for various workflows. Error handling and informative messages
are included to guide the user in case of issues with file retrieval or processing.
</p>


<h3>Value</h3>

<p>A list of class <code>"pdb"</code> containing the following components:
</p>

<dl>
<dt><code>atom</code></dt><dd><p>A data frame containing atomic coordinate data (ATOM and HETATM records). Each row corresponds to an
atom, and each column to a specific record type (e.g., element, residue, chain).</p>
</dd>
<dt><code>xyz</code></dt><dd><p>A numeric matrix of class <code>"xyz"</code> containing the atomic coordinates from the ATOM and HETATM records.</p>
</dd>
<dt><code>calpha</code></dt><dd><p>A logical vector indicating whether each atom is a C-alpha atom (TRUE) or not (FALSE).</p>
</dd>
<dt><code>call</code></dt><dd><p>The matched call, storing the function call for reference.</p>
</dd>
<dt><code>path</code></dt><dd><p>The file path where the file was saved, if 'save' was TRUE.</p>
</dd>
</dl>

<p>The function handles errors and warnings for various edge cases, such as unsupported file types, failed downloads, or issues
with reading the file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Download a CIF file and process it without saving
  pdb_file &lt;- get_pdb_file(pdb_id = "4HHB", filetype = "cif")

  # Download a PDB file, save it, and remove alternate location records
  pdb_file &lt;- get_pdb_file(pdb_id = "4HHB", filetype = "pdb", save = TRUE, path = tempdir())

  # Understanding the tertiary structure of proteins is
  # crucial for elucidating their functional mechanisms,
  # especially in the context of ligand binding, enzyme catalysis,
  # and protein-protein interactions.
  # The tertiary structure refers to the three-dimensional arrangement
  # of all atoms within a protein,
  # including its secondary structure elements like alpha helices
  # and beta sheets, and how these elements
  # are organized in space. Using the get_pdb_file function
  # to retrieve the PDB file and the r3dmol
  # package for visualization, researchers can gain insights
  # into the overall 3D structure of a protein.
  # The following example demonstrates how to visualize the
  # ltertiary structure of a protein using the
  # PDB entry 1XYZ:

  library(r3dmol)

  # Retrieve and parse a PDB structure
  pdb_path &lt;- get_pdb_file("1XYZ", filetype = "pdb", save = TRUE)

  # Visualize the tertiary structure using r3dmol
  viewer &lt;- r3dmol() %&gt;%
    m_add_model(pdb_path$path, format = "pdb") %&gt;%  # Load the PDB file
    m_set_style(style = m_style_cartoon()) %&gt;%  # Cartoon representation
    m_zoom_to()

  # Display the molecular viewer
  viewer
  # In this example, the protein structure is represented
  # in a cartoon style, which is particularly
  # effective for visualizing the overall fold of the protein,
  # including the orientation and interaction
  # of its secondary structure elements.
  #. To further enhance the analysis,
  # it is often important to
  # highlight specific regions of interest,
  # such as potential ligand-binding sites.
  # These sites can be identified based on prior knowledge,
  # experimental data, or computational predictions.
  # The following code snippet demonstrates
  # how to highlight potential ligand-binding sites in the
  # protein structure:

  # Highlight potential ligand-binding sites
  # Note: Manually define residues of interest based
  # on prior knowledge or external analysis
  binding_sites &lt;- c(45, 60, 85)  # Example residue numbers

  viewer &lt;- viewer %&gt;%
    m_set_style(
      sel = m_sel(resi = binding_sites),
      style = m_style_sphere(color = "red", radius = 1.5)
    )

  # Display the updated viewer with highlighted binding sites
  viewer

  # In this step, specific residues that are
  # hypothesized to participate in ligand binding are
  #highlighted using a spherical representation.
  # The residues are selected manually based on either
  # experimental data or computational predictions.
  # By highlighting these sites, researchers can
  # visually inspect the spatial relationship between
  # these residues and other parts of the protein,
  # which may provide insights into the
  # protein's functional mechanisms.

  # This visualization approach offers a powerful
  # way to explore and communicate the 3D structure
  # of proteins, making it easier to hypothesize about their function and
  # interaction with other molecules.


</code></pre>

<hr>
<h2 id='handle_api_errors'>Handle API Errors</h2><span id='topic+handle_api_errors'></span>

<h3>Description</h3>

<p>This function checks for errors in the HTTP response and stops execution
if the request was not successful.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>handle_api_errors(response, url = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="handle_api_errors_+3A_response">response</code></td>
<td>
<p>An HTTP response object.</p>
</td></tr>
<tr><td><code id="handle_api_errors_+3A_url">url</code></td>
<td>
<p>A string representing the requested URL (for more informative error messages).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None. It stops execution if an error is detected.
</p>

<hr>
<h2 id='infer_search_service'>Infer the Appropriate Search Service for RCSB PDB Queries</h2><span id='topic+infer_search_service'></span>

<h3>Description</h3>

<p>The 'infer_search_service' function determines the appropriate search service for a given search operator.
This function is essential for ensuring that queries are directed to the correct search service, such as basic search, text search, sequence search, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infer_search_service(search_operator)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="infer_search_service_+3A_search_operator">search_operator</code></td>
<td>
<p>A query operator object that specifies the type of search being performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string representing the inferred search service, which is necessary for constructing a valid query.
</p>

<hr>
<h2 id='InOperator'>Create an Inclusion Search Operator</h2><span id='topic+InOperator'></span>

<h3>Description</h3>

<p>Constructs an 'InOperator' object for search operations where the attribute value must be within a specified set.
This operator is useful when the search criteria require the attribute to match one of several possible values.
It can handle multiple potential matches and is ideal for scenarios where multiple values are acceptable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InOperator(attribute, value)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="InOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be evaluated. This should be the field within the RCSB PDB
that you want to search against.</p>
</td></tr>
<tr><td><code id="InOperator_+3A_value">value</code></td>
<td>
<p>The set of values to include in the search. This should be a vector of possible values
that the attribute can match.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'InOperator' that can be used in search queries to retrieve entries
where the attribute matches any of the specified values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Search for entries where the attribute matches one of several values
operator &lt;- InOperator(attribute = "rcsb_entity_source_organism.taxonomy_lineage.name",
                       value = c("Homo sapiens", "Mus musculus"))
print(operator)
</code></pre>

<hr>
<h2 id='parse_fasta_text_to_list'>Helper Function: Parse FASTA Text to List Grouped by Header</h2><span id='topic+parse_fasta_text_to_list'></span>

<h3>Description</h3>

<p>This function parses a FASTA-formatted text into a list of sequences,
where each sequence is keyed by the header (which includes organism name and chain information).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_fasta_text_to_list(fasta_text)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parse_fasta_text_to_list_+3A_fasta_text">fasta_text</code></td>
<td>
<p>A string containing FASTA-formatted text.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list where each element is a FASTA sequence keyed by the header.
</p>

<hr>
<h2 id='parse_response'>Parse API Response</h2><span id='topic+parse_response'></span>

<h3>Description</h3>

<p>This function parses the content of an HTTP response based on the specified format.
It supports JSON and plain text formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_response(response, format = "json")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parse_response_+3A_response">response</code></td>
<td>
<p>An HTTP response object.</p>
</td></tr>
<tr><td><code id="parse_response_+3A_format">format</code></td>
<td>
<p>A string indicating the expected response format (&quot;json&quot; or &quot;text&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Parsed content from the response.
</p>

<hr>
<h2 id='perform_search'>Perform a Search in the RCSB PDB</h2><span id='topic+perform_search'></span>

<h3>Description</h3>

<p>This function allows users to perform highly customizable searches in the RCSB Protein Data Bank (PDB) by specifying detailed search criteria. It interfaces directly with the RCSB PDB's RESTful API, enabling complex queries to retrieve specific data, such as PDB entries, assemblies, polymer entities, non-polymer entities, and more.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_search(
  search_operator,
  return_type = "ENTRY",
  request_options = NULL,
  return_with_scores = FALSE,
  return_raw_json_dict = FALSE,
  verbosity = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="perform_search_+3A_search_operator">search_operator</code></td>
<td>
<p>An object that specifies the search criteria. This object can be constructed using various operator functions:
</p>

<dl>
<dt><code>DefaultOperator</code></dt><dd><p>A basic search operator for general search operations.</p>
</dd>
<dt><code>ExactMatchOperator</code></dt><dd><p>For exact match searches, matching an exact attribute value.</p>
</dd>
<dt><code>InOperator</code></dt><dd><p>For searching where the attribute value must be within a specified set of values.</p>
</dd>
<dt><code>ContainsWordsOperator</code></dt><dd><p>For searching attributes that contain certain words.</p>
</dd>
<dt><code>ContainsPhraseOperator</code></dt><dd><p>For searching attributes that contain a specific phrase.</p>
</dd>
<dt><code>ComparisonOperator</code></dt><dd><p>For comparison-based searches, such as finding values greater than, less than, or equal to a specified value.</p>
</dd>
<dt><code>RangeOperator</code></dt><dd><p>For searching within a range of values for a given attribute.</p>
</dd>
<dt><code>ExistsOperator</code></dt><dd><p>To check the existence of a specific attribute in the database.</p>
</dd>
<dt><code>StructureOperator</code></dt><dd><p>For structure-based searches, using PDB entry IDs, assembly IDs, and search modes.</p>
</dd>
<dt><code>SequenceOperator</code></dt><dd><p>For sequence-based searches, using sequences, sequence types, and cutoffs for e-value and identity.</p>
</dd>
<dt><code>SeqMotifOperator</code></dt><dd><p>For searching sequence motifs, using pattern types like SIMPLE, PROSITE, or REGEX.</p>
</dd>
<dt><code>ChemicalOperator</code></dt><dd><p>For chemical structure searches, using SMILES or InChI descriptors and various matching criteria.</p>
</dd>
</dl>

<p>Please see the Details section.</p>
</td></tr>
<tr><td><code id="perform_search_+3A_return_type">return_type</code></td>
<td>
<p>A string specifying the type of data to return. The available options for <code>return_type</code> include:
</p>

<dl>
<dt><code>"ENTRY"</code></dt><dd><p>Returns a list of PDB IDs corresponding to the entries that match the search criteria. This is the default option and provides entry-level information.</p>
</dd>
<dt><code>"ASSEMBLY"</code></dt><dd><p>Returns a list of PDB IDs appended with assembly IDs (formatted as <code>"PDB_ID-ASSEMBLY_ID"</code>). Useful for accessing specific biological assemblies.</p>
</dd>
<dt><code>"POLYMER_ENTITY"</code></dt><dd><p>Returns a list of PDB IDs appended with entity IDs for polymeric molecular entities. Useful for examining specific polymer chains.</p>
</dd>
<dt><code>"NON_POLYMER_ENTITY"</code></dt><dd><p>Returns a list of PDB IDs appended with entity IDs for non-polymeric entities, such as ligands or small molecules. Useful for detailed chemical analysis.</p>
</dd>
<dt><code>"POLYMER_INSTANCE"</code></dt><dd><p>Returns a list of PDB IDs appended with asym IDs, representing specific instances of polymeric entities (e.g., protein chains).</p>
</dd>
<dt><code>"CHEMICAL_COMPONENT"</code></dt><dd><p>Returns a list of chemical component identifiers, useful for detailed chemical analysis.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="perform_search_+3A_request_options">request_options</code></td>
<td>
<p>A list of additional options to further customize the search request. These options can include:
</p>

<dl>
<dt><code>facets</code></dt><dd><p>Faceted queries allow aggregation of search results into categories (buckets) based on the requested field values. Useful for statistical analysis and data aggregation.</p>
</dd>
<dt><code>sort_by</code></dt><dd><p>Defines the sorting criteria for the search results (e.g., by resolution, release date).</p>
</dd>
<dt><code>pagination</code></dt><dd><p>Controls how many results to return per page and which page of results to return. Useful for handling large datasets.</p>
</dd>
<dt><code>return_all_hits</code></dt><dd><p>If set to <code>TRUE</code>, the search returns all matching results; otherwise, a limited set is returned.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="perform_search_+3A_return_with_scores">return_with_scores</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the search results will include relevance scores. Useful when prioritizing results based on their relevance to the search criteria. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="perform_search_+3A_return_raw_json_dict">return_raw_json_dict</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the function returns the raw JSON response from the PDB API. This option is valuable for advanced users who wish to process the raw data themselves or need access to additional details. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="perform_search_+3A_verbosity">verbosity</code></td>
<td>
<p>Logical; if <code>TRUE</code>, detailed messages will be displayed during execution, providing insights into the query being sent and the response received. Verbose mode is useful for debugging or when you need insights into the function's operation. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The operators allow you to build complex search queries tailored to your specific needs. Detailed documentation for each search operator can be found in the <a href="https://search.rcsb.org/#search-operators">RCSB PDB Search Operators</a>.
The searchable attributes include annotations from the mmCIF dictionary, external resources, and those added by RCSB PDB. Both internal additions to the mmCIF dictionary and external resource annotations are prefixed with 'rcsb_'.
For a complete list of available attributes for text searches, refer to the <a href="https://search.rcsb.org/structure-search-attributes.html">Structure Attributes Search</a> and <a href="https://search.rcsb.org/chemical-search-attributes.html">Chemical Attributes Search</a> pages.
</p>


<h3>Value</h3>

<p>The function returns search results based on the specified <code>return_type</code>:
</p>

<dl>
<dt><code>ENTRY</code></dt><dd><p>A vector of PDB IDs that match the search criteria.</p>
</dd>
<dt><code>ASSEMBLY</code></dt><dd><p>A list of PDB IDs with appended assembly IDs, formatted as <code>"PDB_ID-ASSEMBLY_ID"</code>.</p>
</dd>
<dt><code>POLYMER_ENTITY</code></dt><dd><p>A list of PDB IDs with appended entity IDs for polymeric chains.</p>
</dd>
<dt><code>NON_POLYMER_ENTITY</code></dt><dd><p>A list of PDB IDs with appended entity IDs for non-polymeric components.</p>
</dd>
<dt><code>POLYMER_INSTANCE</code></dt><dd><p>A list of PDB IDs with appended asym IDs for specific polymer instances.</p>
</dd>
<dt><code>CHEMICAL_COMPONENT</code></dt><dd><p>A list of chemical component identifiers.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
# Example 1: Search for Polymer Entities from Mus musculus and Homo sapiens
search_operator &lt;- InOperator(
  attribute = "rcsb_entity_source_organism.taxonomy_lineage.name",
  value = c("Mus musculus", "Homo sapiens")
)
results &lt;- perform_search(
  search_operator = search_operator,
  return_type = "POLYMER_ENTITY"
)
results

# Example 2: Search for Entries Released After a Specific Date
operator_date &lt;- ComparisonOperator(
  attribute = "rcsb_accession_info.initial_release_date",
  value = "2019-08-20",
  comparison_type = "GREATER"
)
request_options &lt;- list(
  facets = list(
    list(
      name = "Methods",
      aggregation_type = "terms",
      attribute = "exptl.method"
    )
  )
)
results &lt;- perform_search(
  search_operator = operator_date,
  return_type = "ENTRY",
  request_options = request_options
)
results

# Example 3: Search for Symmetric Dimers with DNA-Binding Domain
operator_symbol &lt;- ExactMatchOperator(
  attribute = "rcsb_struct_symmetry.symbol",
  value = "C2"
)
operator_kind &lt;- ExactMatchOperator(
  attribute = "rcsb_struct_symmetry.kind",
  value = "Global Symmetry"
)
operator_full_text &lt;- DefaultOperator(
  value = "\"heat-shock transcription factor\""
)
operator_dna_count &lt;- ComparisonOperator(
  attribute = "rcsb_entry_info.polymer_entity_count_DNA",
  value = 1,
  comparison_type = "GREATER_OR_EQUAL"
)
query_group &lt;- list(
  type = "group",
  logical_operator = "and",
  nodes = list(
    list(
      type = "terminal",
      service = "text",
      parameters = operator_symbol
    ),
    list(
      type = "terminal",
      service = "text",
      parameters = operator_kind
    ),
    list(
      type = "terminal",
      service = "full_text",
      parameters = operator_full_text
    ),
    list(
      type = "terminal",
      service = "text",
      parameters = operator_dna_count
    )
  )
)
results &lt;- perform_search(
  search_operator = query_group,
  return_type = "ASSEMBLY"
)
results

</code></pre>

<hr>
<h2 id='query_search'>Search Query Function</h2><span id='topic+query_search'></span>

<h3>Description</h3>

<p>This function performs a search query against the RCSB Protein Data Bank
using their REST API. It allows for various types of searches based on the provided parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_search(
  search_term,
  query_type = "full_text",
  return_type = "entry",
  scan_params = NULL,
  num_attempts = 1,
  sleep_time = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="query_search_+3A_search_term">search_term</code></td>
<td>
<p>A string specifying the term to search in the database.</p>
</td></tr>
<tr><td><code id="query_search_+3A_query_type">query_type</code></td>
<td>
<p>A string specifying the type of query to perform.
Supported values include &quot;full_text&quot;, &quot;PubmedIdQuery&quot;,
&quot;TreeEntityQuery&quot;, &quot;ExpTypeQuery&quot;, &quot;AdvancedAuthorQuery&quot;,
&quot;OrganismQuery&quot;, &quot;pfam&quot;, and &quot;uniprot&quot;. Default is &quot;full_text&quot;.</p>
</td></tr>
<tr><td><code id="query_search_+3A_return_type">return_type</code></td>
<td>
<p>A string specifying the type of search result to return.
Possible values are &quot;entry&quot; (default) and &quot;polymer_entity&quot;.</p>
</td></tr>
<tr><td><code id="query_search_+3A_scan_params">scan_params</code></td>
<td>
<p>Additional parameters for the scan, provided as a list.
This is 'NULL' by default and typically only used for advanced queries.</p>
</td></tr>
<tr><td><code id="query_search_+3A_num_attempts">num_attempts</code></td>
<td>
<p>An integer specifying the number of attempts to try the query in case of failure.</p>
</td></tr>
<tr><td><code id="query_search_+3A_sleep_time">sleep_time</code></td>
<td>
<p>A numeric value specifying the time in seconds to wait between attempts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depending on the return_type, it either returns a list of PDB IDs (if &quot;entry&quot;)
or the full response from the API.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get a list of PDBs for a specific search term

# Search Functions by Specific Terms
pdbs &lt;- query_search("ribosome")
head(pdbs)

# Search by PubMed ID Number
pdbs_by_pubmedid &lt;- query_search(search_term = 27499440, query_type = "PubmedIdQuery")
head(pdbs_by_pubmedid)

# Search by source organism using NCBI TaxId
pdbs_by_ncbi_taxid &lt;- query_search(search_term = "6239", query_type = "TreeEntityQuery")
head(pdbs_by_ncbi_taxid)

# Search by Experimental Method
pdbs = query_search(search_term = 'SOLID-STATE NMR', query_type='ExpTypeQuery')
head(pdbs)

pdbs = query_search(search_term = '4HHB', query_type="structure")
head(pdbs)

## Advanced Searches

# Search by Author
pdbs = query_search(search_term = 'Rzechorzek, N.J.', query_type='AdvancedAuthorQuery')
head(pdbs)

# Search by Organism
pdbs = query_search(search_term = "Escherichia coli", query_type="OrganismQuery")
head(pdbs)

# Search by Uniprot ID (Escherichia coli beta-lactamase)
pdbs = query_search(search_term = "P0A877", query_type="uniprot")
head(pdbs)

# Search by PFAM number (protein kinase domain)
pdbs = query_search(search_term = "PF00069", query_type="pfam")
head(pdbs)


</code></pre>

<hr>
<h2 id='QueryGroup'>Create a Grouped Query Object for RCSB PDB Searches</h2><span id='topic+QueryGroup'></span>

<h3>Description</h3>

<p>The 'QueryGroup' function constructs a grouped query object that allows users to perform complex searches in the RCSB Protein Data Bank (PDB).
This function is particularly useful when multiple query objects need to be combined using logical operators like 'AND' or 'OR'.
The resulting grouped query can be used in advanced search operations to filter or combine results based on multiple criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QueryGroup(queries, logical_operator)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QueryGroup_+3A_queries">queries</code></td>
<td>
<p>A list of query objects to be grouped together. Each query object can be either a simple query or another grouped query.</p>
</td></tr>
<tr><td><code id="QueryGroup_+3A_logical_operator">logical_operator</code></td>
<td>
<p>A string specifying the logical operator to combine the queries. Common values are 'AND' and 'OR', but other logical operators may also be supported.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list representing the grouped query object, which can be passed to search functions for execution.
</p>

<hr>
<h2 id='QueryNode'>Create a Query Node for RCSB PDB Searches</h2><span id='topic+QueryNode'></span>

<h3>Description</h3>

<p>The 'QueryNode' function constructs a query node, which can be either a terminal node (for a simple query) or a grouped node (for complex queries).
This function is crucial for structuring queries that will be sent to the RCSB PDB search system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QueryNode(search_operator, logical_operator = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QueryNode_+3A_search_operator">search_operator</code></td>
<td>
<p>A search operator or group object. This defines the criteria for the search.</p>
</td></tr>
<tr><td><code id="QueryNode_+3A_logical_operator">logical_operator</code></td>
<td>
<p>A string specifying the logical operator to combine multiple queries. Default is 'NULL'. This is used only if the search_operator is a group.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list representing the query node, ready to be included in a larger query structure.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
node &lt;- QueryNode(search_operator = DefaultOperator("some_value"))

</code></pre>

<hr>
<h2 id='RangeOperator'>Create a Range Search Operator</h2><span id='topic+RangeOperator'></span>

<h3>Description</h3>

<p>Constructs a 'RangeOperator' object for search operations that specify a range for attribute values.
This operator is particularly useful for filtering results based on numeric or date ranges,
such as finding entries with resolution between specific values or dates within a certain range.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RangeOperator(
  attribute,
  from_value,
  to_value,
  include_lower = TRUE,
  include_upper = TRUE,
  negation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RangeOperator_+3A_attribute">attribute</code></td>
<td>
<p>The attribute to be evaluated within a range. This should be the numeric or date field
within the RCSB PDB that you want to search against.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_from_value">from_value</code></td>
<td>
<p>The starting value of the range. This is the lower bound of the range.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_to_value">to_value</code></td>
<td>
<p>The ending value of the range. This is the upper bound of the range.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_include_lower">include_lower</code></td>
<td>
<p>Boolean indicating whether to include the lower bound in the range. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_include_upper">include_upper</code></td>
<td>
<p>Boolean indicating whether to include the upper bound in the range. Default is TRUE.</p>
</td></tr>
<tr><td><code id="RangeOperator_+3A_negation">negation</code></td>
<td>
<p>Boolean indicating whether to negate the range condition. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'RangeOperator' that can be used in search queries to retrieve entries
where the attribute falls within the specified range.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Search for entries within a specific range of resolution
operator &lt;- RangeOperator(attribute = "rcsb_entry_info.resolution_combined",
                          from_value = 1.5, to_value = 2.5)
print(operator)
</code></pre>

<hr>
<h2 id='RequestOptions'>Define Request Options for RCSB PDB Search Queries</h2><span id='topic+RequestOptions'></span>

<h3>Description</h3>

<p>The 'RequestOptions' function sets various options for search requests to the RCSB PDB, such as pagination and sorting preferences.
These options help control the volume of search results returned and the order in which they are presented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RequestOptions(
  result_start_index = NULL,
  num_results = NULL,
  sort_by = "score",
  desc = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RequestOptions_+3A_result_start_index">result_start_index</code></td>
<td>
<p>An integer specifying the starting index for result pagination. If 'NULL', pagination is not applied.</p>
</td></tr>
<tr><td><code id="RequestOptions_+3A_num_results">num_results</code></td>
<td>
<p>An integer specifying the number of results to return. If 'NULL', the default number of results is returned.</p>
</td></tr>
<tr><td><code id="RequestOptions_+3A_sort_by">sort_by</code></td>
<td>
<p>A string indicating the attribute to sort the results by. The default value is 'score', which ranks results based on relevance.</p>
</td></tr>
<tr><td><code id="RequestOptions_+3A_desc">desc</code></td>
<td>
<p>A boolean indicating whether the sorting should be in descending order. Default is 'TRUE'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of request options that can be included in a search query to control the results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
options &lt;- RequestOptions(result_start_index = 0, num_results = 100, sort_by = "score", desc = TRUE)

</code></pre>

<hr>
<h2 id='return_data_as_dataframe'>Convert RCSB PDB Response Data into a Dataframe</h2><span id='topic+return_data_as_dataframe'></span>

<h3>Description</h3>

<p>The 'return_data_as_dataframe' function transforms the response data obtained from a query to the RCSB Protein Data Bank (PDB) into a structured dataframe.
This function handles various scenarios, including responses with duplicate names, null or empty responses, and nested data structures.
It ensures that the resulting dataframe is consistently formatted and ready for downstream analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>return_data_as_dataframe(response, data_type, ids)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="return_data_as_dataframe_+3A_response">response</code></td>
<td>
<p>A list containing the response data from a PDB query. This list is expected to be structured according to the RCSB PDB GraphQL or REST API specifications.</p>
</td></tr>
<tr><td><code id="return_data_as_dataframe_+3A_data_type">data_type</code></td>
<td>
<p>A string indicating the type of data contained in the response (e.g., &quot;ENTRY&quot;, &quot;POLYMER_ENTITY&quot;). This parameter is primarily used for contextual information and does not directly influence the function's operations.</p>
</td></tr>
<tr><td><code id="return_data_as_dataframe_+3A_ids">ids</code></td>
<td>
<p>A vector of identifiers corresponding to the response data. These IDs are used to label the resulting dataframe, ensuring that each row corresponds to a specific query identifier.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'return_data_as_dataframe' function is designed to provide a flexible and robust mechanism for converting PDB query responses into dataframes. It addresses several common challenges in handling API responses, such as:
</p>

<dl>
<dt>Null or Empty Responses</dt><dd><p>If the response is null or contains no data, the function immediately returns 'NULL', avoiding unnecessary processing.</p>
</dd>
<dt>Duplicate Names Handling</dt><dd><p>The function detects and manages scenarios where the response contains duplicated names. It simplifies such lists by keeping only the first occurrence of each duplicated element, ensuring that the final dataframe has unique column names.</p>
</dd>
<dt>Recursive Flattening</dt><dd><p>The function flattens nested lists within the response, ensuring that all relevant data is captured in a single-level dataframe structure. This is particularly useful for complex responses that contain deeply nested data elements.</p>
</dd>
<dt>Consistent Column Naming</dt><dd><p>After processing the data, the function ensures that column names are consistent and do not retain unnecessary prefixes. This makes the resulting dataframe easier to interpret and work with in subsequent analyses.</p>
</dd>
</dl>



<h3>Value</h3>

<p>A dataframe constructed from the response data, where each row corresponds to an identifier from the 'ids' vector and each column represents a data field from the response.
If the response is null or empty, the function returns 'NULL'.
</p>


<h3>Note</h3>

<p>The function is equipped to handle responses with varying degrees of complexity. It is recommended to provide valid 'ids' corresponding to the query to ensure that the dataframe rows are correctly labeled.
</p>

<hr>
<h2 id='ScoredResult'>Create a Scored Result Object for PDB Searches</h2><span id='topic+ScoredResult'></span>

<h3>Description</h3>

<p>The 'ScoredResult' function constructs a scored result object, typically used in search results to associate an entity ID with a numerical score.
This is useful in ranking search results or displaying relevance scores alongside the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScoredResult(entity_id, score)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ScoredResult_+3A_entity_id">entity_id</code></td>
<td>
<p>A string representing the entity ID. This could be a PDB ID or any identifier relevant to the search.</p>
</td></tr>
<tr><td><code id="ScoredResult_+3A_score">score</code></td>
<td>
<p>A numeric value representing the score associated with the entity. The score often indicates the relevance or quality of the match.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list representing the scored result, which can be included in the search results or used for further processing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
result &lt;- ScoredResult(entity_id = "1XYZ", score = 95.6)

</code></pre>

<hr>
<h2 id='search_graphql'>Perform a GraphQL Query to RCSB PDB</h2><span id='topic+search_graphql'></span>

<h3>Description</h3>

<p>The 'search_graphql' function sends a GraphQL query to the RCSB Protein Data Bank (PDB) using the provided JSON query format.
This function handles the HTTP request, sends the query, and processes the response, including error handling to ensure that the query executes successfully.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>search_graphql(graphql_json_query, graphql_url = GRAPHQL_URL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="search_graphql_+3A_graphql_json_query">graphql_json_query</code></td>
<td>
<p>A list containing the GraphQL query formatted as JSON. This list should include the 'query' key with a value that represents the GraphQL query string.
The query string can specify various elements to retrieve, such as entry IDs, experimental methods, cell dimensions, etc.</p>
</td></tr>
<tr><td><code id="search_graphql_+3A_graphql_url">graphql_url</code></td>
<td>
<p>A string representing the base URL perform GraphQL query. By default, this is set to the global constant <code>GRAPHQL_URL</code>, but users can specify a different URL if needed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A parsed list containing the content of the response from the RCSB PDB, formatted as an R object. If the request fails, the function stops with an error message.
</p>

<hr>
<h2 id='send_api_request'>Send API Request to a Specified URL</h2><span id='topic+send_api_request'></span>

<h3>Description</h3>

<p>This function sends an HTTP request (GET or POST) to the specified URL. It supports optional request bodies for POST requests, customizable encoding, and content type for API interactions. The function is designed to be a general-purpose API handler for use in querying external APIs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>send_api_request(
  url,
  method = "GET",
  body = NULL,
  encode = "json",
  content_type = "application/json",
  verbosity = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="send_api_request_+3A_url">url</code></td>
<td>
<p>A string representing the target URL for the API request.</p>
</td></tr>
<tr><td><code id="send_api_request_+3A_method">method</code></td>
<td>
<p>A string specifying the HTTP method to use. The default is &quot;GET&quot;, but &quot;POST&quot; can also be used.</p>
</td></tr>
<tr><td><code id="send_api_request_+3A_body">body</code></td>
<td>
<p>Optional: The body of the request, typically required for POST requests. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="send_api_request_+3A_encode">encode</code></td>
<td>
<p>A string representing the encoding type of the body for POST requests. Default is <code>"json"</code>.</p>
</td></tr>
<tr><td><code id="send_api_request_+3A_content_type">content_type</code></td>
<td>
<p>A string specifying the content type for POST requests. Default is <code>"application/json"</code>.</p>
</td></tr>
<tr><td><code id="send_api_request_+3A_verbosity">verbosity</code></td>
<td>
<p>Logical flag indicating whether to print status messages during the function execution. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>send_api_request</code> function is a flexible tool for handling API interactions. It supports both GET and POST methods and provides optional parameters for encoding and content type, making it suitable for a wide range of API requests.
</p>
<p>If a network error occurs during the request, the function will throw an error with a detailed message about the failure.
</p>


<h3>Value</h3>

<p>A response object from the <code>httr</code> package representing the server's response to the API request.
</p>

<hr>
<h2 id='SeqMotifOperator'>Create a Sequence Motif Operator for RCSB PDB Searches</h2><span id='topic+SeqMotifOperator'></span>

<h3>Description</h3>

<p>The 'SeqMotifOperator' function constructs an operator for searching sequence motifs within the RCSB Protein Data Bank (PDB).
This operator is used to specify a search pattern, the type of biological sequence, and the pattern-matching method to be applied in the search.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeqMotifOperator(pattern, sequence_type, pattern_type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SeqMotifOperator_+3A_pattern">pattern</code></td>
<td>
<p>A string representing the motif pattern to search for. This can be a simple string or a more complex pattern, depending on the 'pattern_type'.</p>
</td></tr>
<tr><td><code id="SeqMotifOperator_+3A_sequence_type">sequence_type</code></td>
<td>
<p>A string indicating the type of sequence being searched. Accepted values are 'DNA', 'RNA', or 'PROTEIN'.</p>
</td></tr>
<tr><td><code id="SeqMotifOperator_+3A_pattern_type">pattern_type</code></td>
<td>
<p>A string indicating the pattern matching method to use. Options include 'SIMPLE' for basic patterns, 'PROSITE' for PROSITE-style patterns, and 'REGEX' for regular expressions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'SeqMotifOperator' that encapsulates the specified search criteria. This object can be used as part of a search query within the RCSB PDB system.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of creating a sequence motif operator to search for a DNA motif using a regular expression
seq_motif_operator &lt;- SeqMotifOperator(
  pattern = "A[TU]G",
  sequence_type = "DNA",
  pattern_type = "REGEX"
)
print(seq_motif_operator)
</code></pre>

<hr>
<h2 id='SequenceOperator'>Create a Sequence Operator for Sequence-Based Searches</h2><span id='topic+SequenceOperator'></span>

<h3>Description</h3>

<p>The 'SequenceOperator' function constructs an operator for performing sequence-based searches within the RCSB Protein Data Bank (PDB).
This operator allows users to specify a nucleotide or protein sequence, define the type of sequence, and set thresholds for e-value and identity in the search process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SequenceOperator(
  sequence,
  sequence_type = NULL,
  evalue_cutoff = 100,
  identity_cutoff = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SequenceOperator_+3A_sequence">sequence</code></td>
<td>
<p>A string representing the nucleotide or protein sequence to search for. The sequence should be provided in standard IUPAC format.</p>
</td></tr>
<tr><td><code id="SequenceOperator_+3A_sequence_type">sequence_type</code></td>
<td>
<p>Optional: A string indicating the type of sequence. Accepted values are 'DNA', 'RNA', or 'PROTEIN'.
If not provided, the sequence type is automatically determined based on the characters present in the sequence using the 'autoresolve_sequence_type' function.</p>
</td></tr>
<tr><td><code id="SequenceOperator_+3A_evalue_cutoff">evalue_cutoff</code></td>
<td>
<p>A numeric value for the e-value cutoff in the search. This defines the threshold for statistical significance of the search results. Default is 100.</p>
</td></tr>
<tr><td><code id="SequenceOperator_+3A_identity_cutoff">identity_cutoff</code></td>
<td>
<p>A numeric value for the identity cutoff in the search. This sets the minimum percentage of identity required for a match to be considered. Default is 0.95.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'SequenceOperator' that encapsulates the search criteria for sequence-based queries within the RCSB PDB.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of creating a sequence operator for a protein sequence with specific cutoffs
seq_operator &lt;- SequenceOperator(
  sequence = "MVLSPADKTNVKAAW",
  sequence_type = "PROTEIN",
  evalue_cutoff = 10,
  identity_cutoff = 0.90
)
print(seq_operator)

# Example of creating a sequence operator with automatic sequence type detection
seq_operator_auto &lt;- SequenceOperator(
  sequence = "ATGCGTACGTAGC",
  evalue_cutoff = 50,
  identity_cutoff = 0.85
)
print(seq_operator_auto)
</code></pre>

<hr>
<h2 id='StructureOperator'>Create a Structure Operator for Structure-Based Searches</h2><span id='topic+StructureOperator'></span>

<h3>Description</h3>

<p>The 'StructureOperator' function constructs an operator object for conducting structure-based searches within the RCSB Protein Data Bank (PDB).
This operator allows users to specify a PDB entry ID, an assembly ID, and the mode of search to be used, facilitating precise structural queries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StructureOperator(
  pdb_entry_id,
  assembly_id = 1,
  search_mode = "STRICT_SHAPE_MATCH"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="StructureOperator_+3A_pdb_entry_id">pdb_entry_id</code></td>
<td>
<p>A string representing the PDB entry ID to search for. The PDB entry ID is a unique identifier for each structure in the PDB.</p>
</td></tr>
<tr><td><code id="StructureOperator_+3A_assembly_id">assembly_id</code></td>
<td>
<p>An integer representing the assembly ID within the PDB entry. The assembly ID identifies the specific biological assembly or model within the PDB entry.
By default, this is set to 1, which typically corresponds to the first biological assembly or model.</p>
</td></tr>
<tr><td><code id="StructureOperator_+3A_search_mode">search_mode</code></td>
<td>
<p>A string indicating the search mode to be applied during the structure-based search. Accepted values include 'STRICT_SHAPE_MATCH', 'RELAXED_SHAPE_MATCH', etc.
The default is 'STRICT_SHAPE_MATCH', which ensures a precise comparison based on the structural shape of the molecules.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'StructureOperator' that encapsulates the criteria for performing structure-based searches in the RCSB PDB.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example of creating a structure operator for a specific PDB entry and assembly
struct_operator &lt;- StructureOperator(
  pdb_entry_id = "1XYZ",
  assembly_id = 1,
  search_mode = "STRICT_SHAPE_MATCH"
)
print(struct_operator)

# Example of creating a structure operator with a relaxed search mode
struct_operator_relaxed &lt;- StructureOperator(
  pdb_entry_id = "1ABC",
  assembly_id = 2,
  search_mode = "RELAXED_SHAPE_MATCH"
)
print(struct_operator_relaxed)
</code></pre>

<hr>
<h2 id='walk_nested_dict'>Recursively Walk Through a Nested Dictionary</h2><span id='topic+walk_nested_dict'></span>

<h3>Description</h3>

<p>This function performs a recursive search through a nested dictionary-like structure in R,
looking for a specific term and collecting its values. It's useful for extracting specific pieces
of data from complex, deeply nested results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>walk_nested_dict(my_result, term, outputs = list(), depth = 0, maxdepth = 25)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="walk_nested_dict_+3A_my_result">my_result</code></td>
<td>
<p>The nested dictionary-like structure to search through.</p>
</td></tr>
<tr><td><code id="walk_nested_dict_+3A_term">term</code></td>
<td>
<p>The term to search for within the nested dictionary.</p>
</td></tr>
<tr><td><code id="walk_nested_dict_+3A_outputs">outputs</code></td>
<td>
<p>An initially empty list to store the results of the search, default is an empty list.</p>
</td></tr>
<tr><td><code id="walk_nested_dict_+3A_depth">depth</code></td>
<td>
<p>The current depth of the recursion, default is 0.</p>
</td></tr>
<tr><td><code id="walk_nested_dict_+3A_maxdepth">maxdepth</code></td>
<td>
<p>The maximum depth to recurse, default is 25.
If exceeded, the function issues a warning and returns NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of values associated with the term found in the nested dictionary.
Returns NULL if the term is not found or if maximum recursion depth is exceeded.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
