<!DOCTYPE html><html lang="en"><head><title>Help for package Rphylopars</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Rphylopars}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Rphylopars-package'>
<p>Phylogenetic Comparative Tools for Missing Data and Within-Species Variation</p></a></li>
<li><a href='#anc.recon'>
<p>Ultra-fast maximum likelihood ancestral state reconstruction</p></a></li>
<li><a href='#anova.phylopars.lm'>
<p>Phylopars regression ANOVA</p></a></li>
<li><a href='#fast.SSC'>
<p>Fast Phylogenetic Signal Using Sum of Squared Changes (SSC)</p></a></li>
<li><a href='#logLik.phylopars'>
<p>Extract Log_likelihood</p></a></li>
<li><a href='#logLik.phylopars.lm'>
<p>Extract Log_likelihood</p></a></li>
<li><a href='#phylopars'><p>Estimation of phylogenetic and phenotypic covariance parameters</p></a></li>
<li><a href='#phylopars.lm'><p>Rphylopars regression</p></a></li>
<li><a href='#print.phylopars'>
<p>Print phylopars</p></a></li>
<li><a href='#print.phylopars.lm'>
<p>Print phylopars.lm</p></a></li>
<li><a href='#print.SSC'>
<p>Print SSC</p></a></li>
<li><a href='#simtraits'>
<p>Simulate traits for phylopars estimation</p></a></li>
<li><a href='#summary.phylopars'>
<p>Phylopars summary</p></a></li>
<li><a href='#summary.phylopars.lm'>
<p>phylopars.lm summary</p></a></li>
<li><a href='#write.phylopars'>
<p>Write data and tree files for Python phylopars compatability.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.3.10</td>
</tr>
<tr>
<td>Title:</td>
<td>Phylogenetic Comparative Tools for Missing Data and
Within-Species Variation</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for performing phylogenetic comparative methods for datasets with with multiple observations per species (intraspecific variation or measurement error) and/or missing data (Goolsby et al. 2017). Performs ancestral state reconstruction and missing data imputation on the estimated evolutionary model, which can be specified as Brownian Motion, Ornstein-Uhlenbeck, Early-Burst, Pagel's lambda, kappa, or delta, or a star phylogeny.</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-17</td>
</tr>
<tr>
<td>Depends:</td>
<td>ape</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ericgoolsby/Rphylopars/wiki">https://github.com/ericgoolsby/Rphylopars/wiki</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, doBy, phylolm, phytools, Matrix, MASS, numDeriv</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-18 21:37:10 UTC; Eric</td>
</tr>
<tr>
<td>Author:</td>
<td>Eric W. Goolsby [aut, cre],
  Jorn Bruggeman [aut],
  Cecile Ane [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Eric W. Goolsby &lt;eric.goolsby.evolution@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-18 23:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Rphylopars-package'>
Phylogenetic Comparative Tools for Missing Data and Within-Species Variation
</h2><span id='topic+Rphylopars-package'></span>

<h3>Description</h3>

<p>Tools for performing phylogenetic comparative methods for datasets with with multiple observations per species (intraspecific variation or measurement error) and/or missing data (Goolsby et al. 2017). Performs ancestral state reconstruction and missing data imputation on the estimated evolutionary model, which can be specified as Brownian Motion, Ornstein-Uhlenbeck, Early-Burst, Pagel's lambda, kappa, or delta, or a star phylogeny.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> Rphylopars</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.3.10</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-01-17</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Eric W. Goolsby, Jorn Bruggeman, Cecile Ane
</p>
<p>Maintainer: Eric W. Goolsby <a href="mailto:eric.goolsby.evolution@gmail.com">eric.goolsby.evolution@gmail.com</a>
</p>


<h3>References</h3>

<p>Bruggeman J, Heringa J and Brandt BW. (2009) PhyloPars: estimation of missing parameter values using phylogeny. <em>Nucleic Acids Research</em> 37: W179-W184.
</p>
<p>Goolsby EW, Ane C, Bruggeman J. 2017. &quot;Rphylopars: Fast Multivariate Phylogenetic Comparative Methods for Missing Data and Within-Species Variation.&quot; Methods in Ecology &amp; Evolution. 2017. 8:22-27.
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate data
sim_data &lt;- simtraits(ntaxa = 15,ntraits = 4,nreps = 3,nmissing = 10)

# estimate parameters under Brownian motion
# pheno_error = TRUE assumes intraspecific variation
# pheno_correlated = FALSE assumes intraspecific variation is not correlated
# phylo_correlated = TRUE assumed traits are correlated

PPE &lt;- phylopars(trait_data = sim_data$trait_data,tree = sim_data$tree,
  pheno_error = TRUE,phylo_correlated = TRUE,pheno_correlated = TRUE)

PPE

PPE$anc_recon # Ancestral state reconstruction and species mean prediction
PPE$anc_var # Prediction variance


###NOT RUN
# estimate parameters under multivariate OU
# PPE_OU &lt;- phylopars(trait_data = sim_data$trait_data,tree = sim_data$tree,
#    model="mvOU",pheno_error = TRUE,phylo_correlated = TRUE,
#    pheno_correlated = TRUE)
#
# PPE

</code></pre>

<hr>
<h2 id='anc.recon'>
Ultra-fast maximum likelihood ancestral state reconstruction
</h2><span id='topic+anc.recon'></span>

<h3>Description</h3>

<p>This function performs ancestral state reconstruction using a fast algorithm based on Ho and Ane (2014).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anc.recon(trait_data, tree, vars = FALSE, CI = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="anc.recon_+3A_trait_data">trait_data</code></td>
<td>

<p>A vector or matrix of trait values. Names or row names correspond to species names. Data cannot have any missing data or within-species variation (this type of data can be handled by the <a href="#topic+phylopars">phylopars</a> function).
</p>
</td></tr>
<tr><td><code id="anc.recon_+3A_tree">tree</code></td>
<td>

<p>An object of class <code>phylo</code>.
</p>
</td></tr>
<tr><td><code id="anc.recon_+3A_vars">vars</code></td>
<td>

<p>Whether to return the variances of the restricted maximum likelihood estimates
</p>
</td></tr>
<tr><td><code id="anc.recon_+3A_ci">CI</code></td>
<td>

<p>Whether to return 95% confidence intervals of the restricted maximum likelihood estimates
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named vector of maximum likelihood ancestral states (with names corresponding to node names if available or node numbers from the tree rearranged in postorder, as obtained by the command <code>reorder(tree,"postorder")</code>). If vars or CI is set to TRUE, a list is returned with these values included.
</p>


<h3>Author(s)</h3>

<p>Felsenstein, J. (1985) Phylogenies and the comparative method. American Naturalist, 125, 1-15.
</p>
<p>Ho L.S.T., Ane C. 2014. A linear-time algorithm for Gaussian and non-Gaussian trait evolution models. Syst. Biol. 63:397-408.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol., 3, 217-223.
</p>


<h3>See Also</h3>

<p><a href="phytools.html#topic+fastAnc">fastAnc</a>, <a href="ape.html#topic+ace">ace</a>, <a href="ape.html#topic+pic">pic</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(ape)
tree &lt;- rtree(10000) # random tree with 10,000 taxa
x &lt;- setNames(rnorm(1e4),tree$tip.label) # random trait data
recon &lt;- anc.recon(trait_data=x,tree=tree)
</code></pre>

<hr>
<h2 id='anova.phylopars.lm'>
Phylopars regression ANOVA
</h2><span id='topic+anova.phylopars.lm'></span>

<h3>Description</h3>

<p>Generic S3 method for phylopars
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylopars.lm'
anova(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="anova.phylopars.lm_+3A_object">object</code></td>
<td>

<p>Fitted phylopars.lm object
</p>
</td></tr>
<tr><td><code id="anova.phylopars.lm_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>

<hr>
<h2 id='fast.SSC'>
Fast Phylogenetic Signal Using Sum of Squared Changes (SSC)
</h2><span id='topic+fast.SSC'></span>

<h3>Description</h3>

<p>This function uses a fast ancestral state reconstruction algorithm (<a href="#topic+anc.recon">anc.recon</a>, Goolsby, In review) to calculate the sum of squared changes bewteen ancestral and descendant nodes/tips, as described in Klingenberg and Gidaszewski (2010). Significance is assessed via phylogenetic permutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fast.SSC(trait_data, tree, niter = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fast.SSC_+3A_trait_data">trait_data</code></td>
<td>

<p>A vector or matrix of trait values. Names or row names correspond to species names. Data cannot have any missing data or within-species variation.
</p>
</td></tr>
<tr><td><code id="fast.SSC_+3A_tree">tree</code></td>
<td>

<p>An object of class <code>phylo</code>.
</p>
</td></tr>
<tr><td><code id="fast.SSC_+3A_niter">niter</code></td>
<td>

<p>Number of iterations for hypothesis testing (default=1000).
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pvalue</code></td>
<td>
<p>Description of 'comp1'</p>
</td></tr>
<tr><td><code>scaled.SSC</code></td>
<td>
<p>Scaled sum of squared changes. A value less than 1 indicates less phylogenetic signal as measured by SSC than expected under Brownian motion, and a value greater than 1 indicates greater phylogenetic signal as measured by SSC than expected under Brownian motion.</p>
</td></tr>
<tr><td><code>SSC</code></td>
<td>
<p>Total sum of squared changes (SSC)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Eric W. Goolsby
</p>


<h3>References</h3>

<p>Goolsby E.W. 2016. Likelihood-Based Parameter Estimation for High-Dimensional Phylogenetic Comparative Models: Overcoming the Limitations of 'Distance-Based' Methods. Systematic Biology. Accepted.
</p>
<p>Blomberg SP, Garland T, Ives AR. 2003. Testing for phylogenetic signal in comparative data: behavioral traits are more labile. Evolution, 57:717-745.
</p>
<p>Klingenberg, C. P., and N. A. Gidaszewski. 2010. Testing and quantifying phylogenetic signals and homoplasy in morphometric data. Syst. Biol. 59:245-261.
</p>
<p>Adams, D.C. 2014. A generalized K statistic for estimating phylogenetic signal from shape and other high-dimensional multivariate data. Systematic Biology. 63:685-697.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim_dat &lt;- simtraits(ntaxa = 100,ntraits = 4)
fast.SSC(trait_data = sim_dat$trait_data,tree = sim_dat$tree)
</code></pre>

<hr>
<h2 id='logLik.phylopars'>
Extract Log_likelihood
</h2><span id='topic+logLik.phylopars'></span>

<h3>Description</h3>

<p>Generic S3 method for phylopars
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylopars'
logLik(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="logLik.phylopars_+3A_object">object</code></td>
<td>

<p>Fitted phylopars object
</p>
</td></tr>
<tr><td><code id="logLik.phylopars_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>

<hr>
<h2 id='logLik.phylopars.lm'>
Extract Log_likelihood
</h2><span id='topic+logLik.phylopars.lm'></span>

<h3>Description</h3>

<p>Generic S3 method for phylopars
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylopars.lm'
logLik(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="logLik.phylopars.lm_+3A_object">object</code></td>
<td>

<p>Fitted phylopars.lm object
</p>
</td></tr>
<tr><td><code id="logLik.phylopars.lm_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>

<hr>
<h2 id='phylopars'>Estimation of phylogenetic and phenotypic covariance parameters</h2><span id='topic+phylopars'></span>

<h3>Description</h3>

<p>This function estimates parameters for the phylogenetic and phenotypic variance-covariance matrices for datasets with missing observations and multiple within-species observations. This function can also be used to fit altnerative evolutionary models, including Ornstein-Uhlenbeck, Early-Burst, star phylogeny, or Pagel's lambda, kappa, or delta. Reconstructed ancestral states and predicted species means (i.e., for missing data), along with prediction variances, are also provided.</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylopars(trait_data, tree, model = "BM", pheno_error, phylo_correlated = TRUE,
pheno_correlated = TRUE, REML = TRUE, full_alpha = TRUE, phylocov_start,
phenocov_start, model_par_start, phylocov_fixed, phenocov_fixed, model_par_fixed,
skip_optim = FALSE, skip_EM = FALSE, EM_Fels_limit = 1000, repeat_optim_limit = 1,
EM_missing_limit = 50, repeat_optim_tol = 0.01, model_par_evals = 10, max_delta = 10000,
EM_verbose = FALSE, optim_verbose = FALSE, npd = FALSE,
nested_optim = FALSE, usezscores = TRUE, phenocov_list = list(), ret_args = FALSE,
ret_level = 1, get_cov_CIs = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylopars_+3A_trait_data">trait_data</code></td>
<td>
<p>A data frame with the first column labeled &quot;species&quot; (with species names matching tips on the phylogeny) and one column per trait. Each row corresponds to a single observation, and multiple observations for species are allowed. Missing data should be represented with NA.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_tree">tree</code></td>
<td>
<p>An object of class <code>phylo</code></p>
</td></tr>
<tr><td><code id="phylopars_+3A_model">model</code></td>
<td>
<p>Model of evolution. Default is &quot;BM&quot;. Alternative evolutionary models include &quot;mvOU&quot; (for the multivariate Ornstein-Uhlenbeck), or univariate tree transformations: &quot;OU&quot; &quot;lambda&quot;, &quot;kappa&quot;, &quot;delta&quot;, &quot;EB&quot;, &quot;star&quot;.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_pheno_error">pheno_error</code></td>
<td>
<p>If TRUE (default, unless &lt;=1 observation per species is provided), parameters are estimated assuming within-species variation.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_phylo_correlated">phylo_correlated</code></td>
<td>
<p>If TRUE (default), parameters are estimated assuming traits are correlated.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_pheno_correlated">pheno_correlated</code></td>
<td>
<p>If TRUE (default), parameters are estimated assuming within-species observations traits are correlated.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_reml">REML</code></td>
<td>
<p>If TRUE (default), the algorithm will return REML estimates. If FALSE, maximum likelihood estimates will be returned.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_full_alpha">full_alpha</code></td>
<td>
<p>Only applicable for the multivariate OU (model=&quot;mvOU&quot;). If TRUE (default), a fully parametrized alpha matrix is fit. If FALSE, a diagonal alpha matrix is fit.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_phylocov_start">phylocov_start</code></td>
<td>
<p>Optional starting value for phylogenetic trait variance-covariance matrix. Must be of dimension n_traits by n_traits.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_phenocov_start">phenocov_start</code></td>
<td>
<p>Optional starting value for phenotypic trait variance-covariance matrix. Must be of dimension n_traits by n_traits.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_model_par_start">model_par_start</code></td>
<td>
<p>Optional starting parameters for the evolutionary model. For model=&quot;mvOU&quot;, must be of dimension n_traits by n_traits. Otherwise, must be a single value.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_phylocov_fixed">phylocov_fixed</code></td>
<td>
<p>Optional fixed value for phylogenetic trait variance-covariance matrix. Must be of dimension n_traits by n_traits.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_phenocov_fixed">phenocov_fixed</code></td>
<td>
<p>Optional starting value for phenotypic trait variance-covariance matrix. Must be of dimension n_traits by n_traits.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_model_par_fixed">model_par_fixed</code></td>
<td>
<p>Optional fixed parameter for the evolutionary model. For model=&quot;mvOU&quot;, must be of dimension n_traits by n_traits. Otherwise, must be a single value.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_skip_optim">skip_optim</code></td>
<td>
<p>Whether to skip BFGS optimization (not recommended unless all parameters are fixed).</p>
</td></tr>
<tr><td><code id="phylopars_+3A_skip_em">skip_EM</code></td>
<td>
<p>Whether to skip Expectation-Maximiation prior to generating starting parameters for BFGS optimization (not recommended unless providing fixed parameters).</p>
</td></tr>
<tr><td><code id="phylopars_+3A_em_fels_limit">EM_Fels_limit</code></td>
<td>
<p>Whether to skip Expectation-Maximiation prior to generating starting parameters for BFGS optimization (not recommended unless providing fixed parameters).</p>
</td></tr>
<tr><td><code id="phylopars_+3A_repeat_optim_limit">repeat_optim_limit</code></td>
<td>
<p>The number of times to repeat numerical optimization (default is 1).</p>
</td></tr>
<tr><td><code id="phylopars_+3A_em_missing_limit">EM_missing_limit</code></td>
<td>
<p>Maximum number of iterations for EM.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_repeat_optim_tol">repeat_optim_tol</code></td>
<td>
<p>Maximum tolerance for repeated numerical optimization (only relevant if repeat_optim_limit&gt;1).</p>
</td></tr>
<tr><td><code id="phylopars_+3A_model_par_evals">model_par_evals</code></td>
<td>
<p>Number of times to evaluate univariate tree transformation models along the range of possible parameter values. Used to generate informed starting values for alternative evolutionary models if nested_optim=TRUE.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_max_delta">max_delta</code></td>
<td>
<p>Maximum allowed difference between the log-likelihood for EM-generated starting parameters and new parameters tried under numerical optimization. Extremely large deltas are likely to be numerical artifiacts. Prevents artificial convergence.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_em_verbose">EM_verbose</code></td>
<td>
<p>Whether to print the log-likelihood during Expectation-Maximization.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_optim_verbose">optim_verbose</code></td>
<td>
<p>Whether to print log-likelihooods during numerical optimization.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_npd">npd</code></td>
<td>
<p>Whether to find the nearest positive-definite matrix for all covariance matrices during numerical optimization (slow &ndash; only set to TRUE if converging to singular matrices).</p>
</td></tr>
<tr><td><code id="phylopars_+3A_nested_optim">nested_optim</code></td>
<td>
<p>Only relevant if fitting a univariate alternative evolutionary model. Tries multiple tree transformation parameter values along the range of possible values to make informed starting parameters. Slower than the default (nested_optim=FALSE), in which all parameters are estimated simultaneously.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_usezscores">usezscores</code></td>
<td>
<p>Whether or not ot use centered and standardized data during numerical optimization (recommended).</p>
</td></tr>
<tr><td><code id="phylopars_+3A_phenocov_list">phenocov_list</code></td>
<td>
<p>An optional named list of species-specific within-species covariance matrices to be held fixed, as in Ives et al (2007). This option forces pheno_error and pheno_correlated to be FALSE, and uses mean species values instead of raw data. Raw variance should be divided by the number of observations per species (i.e., squared standard errors). See Ives et al (2007) for more details.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_ret_args">ret_args</code></td>
<td>
<p>For internal use only.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_ret_level">ret_level</code></td>
<td>
<p>For internal use only.</p>
</td></tr>
<tr><td><code id="phylopars_+3A_get_cov_cis">get_cov_CIs</code></td>
<td>
<p>Whether to return 95-percent confidence intervals of covariance parameters (default=FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>phylopars</code>. For models with phenotypic (within-species) covariance, the estimated percentage of variance explained by the phylogeny is provided as 100*(1 - phenotypic_variance/raw_variance), where raw_variance is the variance of all observations for a given trait across species (<code>var(PPE$trait_data[,2:ncol(PPE$trait_data)],na.rm=TRUE)</code>).
</p>
<table role = "presentation">
<tr><td><code>logLik</code></td>
<td>
<p>The log-likelihood of the model</p>
</td></tr>
<tr><td><code>pars</code></td>
<td>
<p>A list composed of phylogenetic trait covariance and phenotypic (within-species) trait covariance, if estimated</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>The model of evolution (e.g., BM, OU, lambda, etc.), and any additional evolutionary model parameters estimated. For OU models, stationary covariance is calculated from both phylogenetic covariance (Sigma) and alpha (see Supplement 1 of Clavel et al. 2015).</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>The estimate ancestral state at the root of the tree.</p>
</td></tr>
<tr><td><code>npars</code></td>
<td>
<p>The total number of parameters estimated by optimization (used for AIC and BIC).</p>
</td></tr>
<tr><td><code>anc_recon</code></td>
<td>
<p>Reconstructed ancestral states and species means. Row names correspond to species names (for the first 1:nspecies rows), and the remaining row names correspond to node numbers on a tree with edges in postorder: <code>reorder(tree,"postorder"). Or, if node labels were included on the original tree, row names correspond to node labels.</code></p>
</td></tr>
<tr><td><code>anc_var</code></td>
<td>
<p>Variance of reconstructed ancestral estimates and imputed species means.</p>
</td></tr>
<tr><td><code>anc_cov</code></td>
<td>
<p>Covariance of estimates among variables.</p>
</td></tr>
<tr><td><code>tree</code></td>
<td>
<p>The phylogenetic tree supplied to <code>phylopars</code></p>
</td></tr>
<tr><td><code>trait_data</code></td>
<td>
<p>The trait data supplied to <code>phylopars</code></p>
</td></tr>
<tr><td><code>REML</code></td>
<td>
<p><code>TRUE</code> if REML, FALSE if ML.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Eric W. Goolsby <a href="mailto:eric.goolsby.evolution@gmail.com">eric.goolsby.evolution@gmail.com</a>, Cecile Ane, Jorn Bruggeman
</p>


<h3>References</h3>

<p>Bruggeman J, Heringa J and Brandt BW. (2009) PhyloPars: estimation of missing parameter values using phylogeny. <em>Nucleic Acids Research</em> 37: W179-W184.
Clavel, J., Escarguel, G. &amp; Merceron, G. (2015) mvmorph: an r package for fitting multivariate 261 evolutionary models to morphometric data. Methods in Ecology and Evolution, 6, 131-1319.
Felsenstein, J. (2008) Comparative methods with sampling error and within-species variation: contrasts revisited and revised. American Naturalist, 171, 713-725.
Ho L.S.T., Ane C. 2014. A linear-time algorithm for Gaussian and non-Gaussian trait evolution models. Syst. Biol. 63:397-408.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # simulate data
  sim_data &lt;- simtraits(ntaxa = 15,ntraits = 4,nreps = 3,nmissing = 10)

  # estimate parameters under Brownian motion
  # pheno_error = TRUE assumes intraspecific variation
  # pheno_correlated = FALSE assumes intraspecific variation is not correlated
  # phylo_correlated = TRUE assumed traits are correlated

  PPE &lt;- phylopars(trait_data = sim_data$trait_data,tree = sim_data$tree,
    pheno_error = TRUE,phylo_correlated = TRUE,pheno_correlated = TRUE)

  PPE

  PPE$anc_recon # Ancestral state reconstruction and species mean prediction
  PPE$anc_var # Prediction variance


  ###NOT RUN
  # estimate parameters under multivariate OU
  # PPE_OU &lt;- phylopars(trait_data = sim_data$trait_data,tree = sim_data$tree,
  #    model="mvOU",pheno_error = TRUE,phylo_correlated = TRUE,
  #    pheno_correlated = TRUE)
  #
  # PPE
</code></pre>

<hr>
<h2 id='phylopars.lm'>Rphylopars regression</h2><span id='topic+phylopars.lm'></span>

<h3>Description</h3>

<p>Performs phylogenetic regression.</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylopars.lm(formula, trait_data, tree, model = "BM", pheno_error,
phylo_correlated = TRUE, pheno_correlated = TRUE, REML = TRUE,
full_alpha = TRUE, phylocov_start, phenocov_start, model_par_start,
phylocov_fixed, phenocov_fixed, model_par_fixed, skip_optim = FALSE,
skip_EM = FALSE, EM_Fels_limit = 1000, repeat_optim_limit = 1,
EM_missing_limit = 50,repeat_optim_tol = 0.01, model_par_evals = 10,
max_delta = 10000, EM_verbose = FALSE,optim_verbose = FALSE, npd = FALSE,
nested_optim = FALSE, usezscores = TRUE, phenocov_list = list(),
ret_args = FALSE, ret_level = 1, get_cov_CIs = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylopars.lm_+3A_formula">formula</code></td>
<td>
<p>Model formula &ndash; e.g. Y~X1+X2</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_trait_data">trait_data</code></td>
<td>
<p>A data frame with the first column labeled &quot;species&quot; (with species names matching tips on the phylogeny) and one column per trait. Each row corresponds to a single observation, and multiple observations for species are allowed. Missing data should be represented with NA.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_tree">tree</code></td>
<td>
<p>An object of class <code>phylo</code></p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_model">model</code></td>
<td>
<p>Model of evolution. Default is &quot;BM&quot;. Alternative evolutionary models include &quot;mvOU&quot; (for the multivariate Ornstein-Uhlenbeck), or univariate tree transformations: &quot;OU&quot; &quot;lambda&quot;, &quot;kappa&quot;, &quot;delta&quot;, &quot;EB&quot;, &quot;star&quot;.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_pheno_error">pheno_error</code></td>
<td>
<p>If TRUE (default, unless &lt;=1 observation per species is provided), parameters are estimated assuming within-species variation.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_phylo_correlated">phylo_correlated</code></td>
<td>
<p>If TRUE (default), parameters are estimated assuming traits are correlated.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_pheno_correlated">pheno_correlated</code></td>
<td>
<p>If TRUE (default), parameters are estimated assuming within-species observations traits are correlated.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_reml">REML</code></td>
<td>
<p>If TRUE (default), the algorithm will return REML estimates. If FALSE, maximum likelihood estimates will be returned.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_full_alpha">full_alpha</code></td>
<td>
<p>Only applicable for the multivariate OU (model=&quot;mvOU&quot;). If TRUE (default), a fully parametrized alpha matrix is fit. If FALSE, a diagonal alpha matrix is fit.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_phylocov_start">phylocov_start</code></td>
<td>
<p>Optional starting value for phylogenetic trait variance-covariance matrix. Must be of dimension n_traits by n_traits.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_phenocov_start">phenocov_start</code></td>
<td>
<p>Optional starting value for phenotypic trait variance-covariance matrix. Must be of dimension n_traits by n_traits.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_model_par_start">model_par_start</code></td>
<td>
<p>Optional starting parameters for the evolutionary model. For model=&quot;mvOU&quot;, must be of dimension n_traits by n_traits. Otherwise, must be a single value.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_phylocov_fixed">phylocov_fixed</code></td>
<td>
<p>Optional fixed value for phylogenetic trait variance-covariance matrix. Must be of dimension n_traits by n_traits.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_phenocov_fixed">phenocov_fixed</code></td>
<td>
<p>Optional starting value for phenotypic trait variance-covariance matrix. Must be of dimension n_traits by n_traits.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_model_par_fixed">model_par_fixed</code></td>
<td>
<p>Optional fixed parameter for the evolutionary model. For model=&quot;mvOU&quot;, must be of dimension n_traits by n_traits. Otherwise, must be a single value.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_skip_optim">skip_optim</code></td>
<td>
<p>Whether to skip BFGS optimization (not recommended unless all parameters are fixed).</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_skip_em">skip_EM</code></td>
<td>
<p>Whether to skip Expectation-Maximiation prior to generating starting parameters for BFGS optimization (not recommended unless providing fixed parameters).</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_em_fels_limit">EM_Fels_limit</code></td>
<td>
<p>Whether to skip Expectation-Maximiation prior to generating starting parameters for BFGS optimization (not recommended unless providing fixed parameters).</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_repeat_optim_limit">repeat_optim_limit</code></td>
<td>
<p>The number of times to repeat numerical optimization (default is 1).</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_em_missing_limit">EM_missing_limit</code></td>
<td>
<p>Maximum number of iterations for EM.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_repeat_optim_tol">repeat_optim_tol</code></td>
<td>
<p>Maximum tolerance for repeated numerical optimization (only relevant if repeat_optim_limit&gt;1).</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_model_par_evals">model_par_evals</code></td>
<td>
<p>Number of times to evaluate univariate tree transformation models along the range of possible parameter values. Used to generate informed starting values for alternative evolutionary models if nested_optim=TRUE.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_max_delta">max_delta</code></td>
<td>
<p>Maximum allowed difference between the log-likelihood for EM-generated starting parameters and new parameters tried under numerical optimization. Extremely large deltas are likely to be numerical artifiacts. Prevents artificial convergence.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_em_verbose">EM_verbose</code></td>
<td>
<p>Whether to print the log-likelihood during Expectation-Maximization.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_optim_verbose">optim_verbose</code></td>
<td>
<p>Whether to print log-likelihooods during numerical optimization.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_npd">npd</code></td>
<td>
<p>Whether to find the nearest positive-definite matrix for all covariance matrices during numerical optimization (slow &ndash; only set to TRUE if converging to singular matrices).</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_nested_optim">nested_optim</code></td>
<td>
<p>Only relevant if fitting a univariate alternative evolutionary model. Tries multiple tree transformation parameter values along the range of possible values to make informed starting parameters. Slower than the default (nested_optim=FALSE), in which all parameters are estimated simultaneously.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_usezscores">usezscores</code></td>
<td>
<p>Whether or not ot use centered and standardized data during numerical optimization (recommended).</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_phenocov_list">phenocov_list</code></td>
<td>
<p>An optional named list of species-specific within-species covariance matrices to be held fixed, as in Ives et al (2007). This option forces pheno_error and pheno_correlated to be FALSE, and uses mean species values instead of raw data. Raw variance should be divided by the number of observations per species (i.e., squared standard errors). See Ives et al (2007) for more details.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_ret_args">ret_args</code></td>
<td>
<p>For internal use only.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_ret_level">ret_level</code></td>
<td>
<p>For internal use only.</p>
</td></tr>
<tr><td><code id="phylopars.lm_+3A_get_cov_cis">get_cov_CIs</code></td>
<td>
<p>Whether to return 95-percent confidence intervals of covariance parameters (default=FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A fitted phylopars.lm object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # simulate data
  sim_data &lt;- simtraits(ntaxa = 15,ntraits = 4)

  phylopars.lm(V4~V1+V2+V3,trait_data=sim_data$trait_data,tree=sim_data$tree)
</code></pre>

<hr>
<h2 id='print.phylopars'>
Print phylopars
</h2><span id='topic+print.phylopars'></span>

<h3>Description</h3>

<p>Generic S3 method for phylopars
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylopars'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.phylopars_+3A_x">x</code></td>
<td>

<p>Fitted phylopars object
</p>
</td></tr>
<tr><td><code id="print.phylopars_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>

<hr>
<h2 id='print.phylopars.lm'>
Print phylopars.lm
</h2><span id='topic+print.phylopars.lm'></span>

<h3>Description</h3>

<p>Generic S3 method for phylopars.lm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylopars.lm'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.phylopars.lm_+3A_x">x</code></td>
<td>

<p>Fitted phylopars.lm object
</p>
</td></tr>
<tr><td><code id="print.phylopars.lm_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>

<hr>
<h2 id='print.SSC'>
Print SSC
</h2><span id='topic+print.SSC'></span>

<h3>Description</h3>

<p>Generic S3 method for objects returned by the function <code>fast.SSC</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SSC'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.SSC_+3A_x">x</code></td>
<td>

<p>Object returned by <code>fast.SSC</code>
</p>
</td></tr>
<tr><td><code id="print.SSC_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>

<hr>
<h2 id='simtraits'>
Simulate traits for phylopars estimation
</h2><span id='topic+simtraits'></span>

<h3>Description</h3>

<p>Simulates traits for <code>phylopars</code> estimation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simtraits(ntaxa = 15, ntraits = 4, nreps = 1, nmissing = 0, tree, v, anc, 
intraspecific, model="BM", parameters, nsim, return.type="data.frame")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simtraits_+3A_ntaxa">ntaxa</code></td>
<td>

<p>Either number of taxa (<code>ntaxa</code>) or a <code>tree</code> can be supplied.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_ntraits">ntraits</code></td>
<td>

<p>Number of traits to be simulated.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_nreps">nreps</code></td>
<td>

<p>Number of replicates per trait per species to simulate.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_nmissing">nmissing</code></td>
<td>

<p>Number of randomly missing trait values.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_tree">tree</code></td>
<td>

<p>Either number of taxa (<code>ntaxa</code>) or a <code>tree</code> can be supplied.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_v">v</code></td>
<td>

<p>Trait covariance (<code>v</code>) can be optionally supplied; otherwise off-diagonal elements are set to <code>0.8</code>.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_anc">anc</code></td>
<td>

<p>Value for ancestral state at root node.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_intraspecific">intraspecific</code></td>
<td>

<p>Optional value for within-species variance.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_model">model</code></td>
<td>

<p>Model of evolution (default=&quot;BM&quot;). Other options include &quot;OUfixedRoot&quot;, &quot;OUrandomRoot&quot;, &quot;lambda&quot;, &quot;kappa&quot;, &quot;delta&quot;, &quot;EB&quot;.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_parameters">parameters</code></td>
<td>

<p>List of parameters for the model. alpha for the selection strength in the OU model, lambda, kappa, delta, or rate for the EB model.
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_nsim">nsim</code></td>
<td>

<p>Number of simulations to perform (default is 1)
</p>
</td></tr>
<tr><td><code id="simtraits_+3A_return.type">return.type</code></td>
<td>

<p>Default is &quot;data.frame&quot;. Can also specify &quot;matrix&quot; if nreps=1.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>trait_data</code></td>
<td>
<p>Data for <code>phylopars()</code></p>
</td></tr>
<tr><td><code>tree</code></td>
<td>
<p>The original phylogenetic tree (either provided to the function or generated internally)</p>
</td></tr>
<tr><td><code>sim_tree</code></td>
<td>
<p>The transformed tree on which trait simulations were performed (identical to tree if model=&quot;BM&quot;)</p>
</td></tr>
<tr><td><code>original_X</code></td>
<td>
<p>If within-species variation is simulated, original_X is the original species mean values before adding within-species variation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Eric W. Goolsby <a href="mailto:eric.goolsby.evolution@gmail.com">eric.goolsby.evolution@gmail.com</a>
</p>


<h3>References</h3>

<p>Bruggeman J, Heringa J and Brandt BW. (2009) PhyloPars: estimation of missing parameter values using phylogeny. <em>Nucleic Acids Research</em> 37: W179-W184.
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008. GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate data
sim_data &lt;- simtraits(ntaxa = 15,ntraits = 4,nreps = 3,nmissing = 10)

# estimate parameters under Brownian motion
# pheno_error = TRUE assumes intraspecific variation
# pheno_correlated = FALSE assumes intraspecific variation is not correlated
# phylo_correlated = TRUE assumed traits are correlated

PPE &lt;- phylopars(trait_data = sim_data$trait_data,tree = sim_data$tree,
  pheno_error = TRUE,phylo_correlated = TRUE,pheno_correlated = FALSE)

PPE
</code></pre>

<hr>
<h2 id='summary.phylopars'>
Phylopars summary
</h2><span id='topic+summary.phylopars'></span>

<h3>Description</h3>

<p>Summarizes phylopars
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylopars'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.phylopars_+3A_object">object</code></td>
<td>

<p>Fitted phylopars object
</p>
</td></tr>
<tr><td><code id="summary.phylopars_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>

<hr>
<h2 id='summary.phylopars.lm'>
phylopars.lm summary
</h2><span id='topic+summary.phylopars.lm'></span>

<h3>Description</h3>

<p>Summarizes phylopars.lm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylopars.lm'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.phylopars.lm_+3A_object">object</code></td>
<td>

<p>Fitted phylopars.lm object
</p>
</td></tr>
<tr><td><code id="summary.phylopars.lm_+3A_...">...</code></td>
<td>

<p>Further arguments passed to or from other methods.
</p>
</td></tr>
</table>

<hr>
<h2 id='write.phylopars'>
Write data and tree files for Python phylopars compatability.
</h2><span id='topic+write.phylopars'></span>

<h3>Description</h3>

<p>Writes data and tree files for Python phylopars compatibility.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.phylopars(trait_data, tree, data_file, tree_file, species_identifier = "species")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write.phylopars_+3A_trait_data">trait_data</code></td>
<td>

<p>A data frame with one column per trait, as well as a column labeled &quot;species&quot; (with species names matching tips on the phylogeny). Each row corresponds to a single observation, and multiple observation for species are allowed. Missing data should be represented with NA.
</p>
</td></tr>
<tr><td><code id="write.phylopars_+3A_tree">tree</code></td>
<td>

<p>An object of class <code>phylo</code>
</p>
</td></tr>
<tr><td><code id="write.phylopars_+3A_data_file">data_file</code></td>
<td>

<p>Desired path to write data file.
</p>
</td></tr>
<tr><td><code id="write.phylopars_+3A_tree_file">tree_file</code></td>
<td>

<p>Desired path to write tree file.
</p>
</td></tr>
<tr><td><code id="write.phylopars_+3A_species_identifier">species_identifier</code></td>
<td>

<p>Title of species column in data file. Defaulted to <code>"species"</code>
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Eric W. Goolsby <a href="mailto:eric.goolsby.evolution@gmail.com">eric.goolsby.evolution@gmail.com</a>
</p>


<h3>References</h3>

<p>Bruggeman J, Heringa J and Brandt BW. (2009) PhyloPars: estimation of missing parameter values using phylogeny. <em>Nucleic Acids Research</em> 37: W179-W184.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# simulate data
sim_data &lt;- simtraits(ntaxa = 15,ntraits = 4,nreps = 3,nmissing = 10)

write.phylopars(trait_data = sim_data$trait_data,tree = sim_data$tree,data_file = "data_path.txt",
tree_file = "tree_path.new")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
