<!DOCTYPE html><html><head><title>Help for package RRate</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RRate}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RRate-package'>
<p>Estimating Replication Rate for Genome-Wide Association Studies</p></a></li>
<li><a href='#HLtest'>
<p>Hosmer-Lemeshow test</p></a></li>
<li><a href='#repSampleSizeRR'>
<p>Sample size determination for the replication study based on RR</p></a></li>
<li><a href='#RRate-functions'>
<p>Estimating Replication Rate for primary associations</p></a></li>
<li><a href='#SEest'>
<p>Estimating standard error of the observed log-odds ratio</p></a></li>
<li><a href='#smryStats'>
<p>Example summary statistics</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimating Replication Rate for Genome-Wide Association Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-08-16</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Jiang, Jing-Hao Xue and Weichuan Yu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Jiang &lt;wjiangaa@connect.ust.hk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Replication Rate (RR) is the probability of replicating a statistically significant association in genome-wide association studies. This R-package provide the estimation method for replication rate which makes use of the summary statistics from the primary study. We can use the estimated RR to determine the sample size of the replication study, and to check the consistency between the results of the primary study and those of the replication study.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), splines</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-08-17 03:41:41 UTC; Brian</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-08-17 10:40:27</td>
</tr>
</table>
<hr>
<h2 id='RRate-package'>
Estimating Replication Rate for Genome-Wide Association Studies
</h2><span id='topic+RRate-package'></span><span id='topic+RRate'></span>

<h3>Description</h3>

<p>Replication Rate (RR) is the probability of replicating a statistically significant association in genome-wide association studies. This R-package provide the estimation method for replication rate which makes use of the summary statistics from the primary study. We can use the estimated RR to determine the sample size of the replication study, and to check the consistency between the results of the primary study and those of the replication study.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> RRate</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-06-13</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The goal of genome-wide association studies (GWAS) is to discover genetic variants associated with diseases/traits. Replication is a common validation method in GWAS. We regard an association as true finding when it shows significance in both the primary and replication studies. A worth pondering question is: what is the probability of a primary association (i.e. statistically significant association in the primary study) being validated in the replication study?
</p>
<p>We refer the Bayesian replication probability as the replication rate (RR). Here we implement the estimation method for RR which makes use of the summary statistics from the primary study. We can use the estimated RR to determine the sample size of the replication study, and to check the consistency between the results of the primary study and those of the replication study.
</p>
<p>The principal component of RRate package is <code>repRateEst</code>. Also we implement sample size determination method (<code>repSampleSizeRR</code> and <code>repSampleSizeRR2</code>) and consistency checking method (Hosmer-Lemeshow test, <code>HLtest</code>).
</p>
<p>1. To estimate the RR, we need obtain the summary statistics of each genotyped SNPs in the primary study. We have put a example summary statistics (smryStats1) in the package. You can use <code>data(smryStats1)</code> to load the example data. You can also obtain the ground-truth parameters (allele frequencies, odds ratios) of the example data using <code>data(param)</code>. We also put the corresponding summary statistics of the replicaition study in the package (smryStats2). 
</p>
<p>2. You can use <code>SEest</code> to estimate the standard error of the observed log-odds ratio.
</p>
<p><code>SEest(n0,n1,fU,fA)</code>
</p>
<p>Details about the function can be seen using <code><a href="#topic+SEest">help(SEest)</a></code>.
</p>
<p>3. You can use <code>repRateEst</code> to estimate the RR for each associations discovered from the primary study (i.e. primary associations). 
</p>
<p><code>repRateEst(MUhat,SE, SE2,zalpha2,zalphaR2,boot=100,output=TRUE,idx=TRUE,dir='output',info=T)</code>
</p>
<p>Details about the function can be seen using <code><a href="#topic+repRateEst">help(repRateEst)</a></code>.
</p>
<p>4. You can use <code>repSampleSizeRR</code> and <code>repSampleSizeRR2</code> to determine the sample size of the replication study. 
</p>
<p><code>repSampleSizeRR(RR, n, MUhat,SE,zalpha2,zalphaR2,idx=TRUE)</code>
</p>
<p><code>repSampleSizeRR2(RR,CCR2, MUhat,SE,fU,fA,zalpha2,zalphaR2, idx=TRUE)</code>
</p>
<p>Details about these functions can be seen using <code><a href="#topic+repSampleSizeRR">help(repSampleSizeRR)</a></code> and <code><a href="#topic+repSampleSizeRR2">help(repSampleSizeRR2)</a></code>.
</p>
<p>5. You can use <code>HLtest</code> to check the consistency between the results of the primary study and those of the replication study.  
</p>
<p><code>HLtest(x,p,g=10,null='all',boot=1000,info=T,dir='.')</code>
</p>
<p>Details about the function can be seen using <code><a href="#topic+HLtest">help(HLtest)</a></code>
</p>


<h3>Author(s)</h3>

<p>Wei Jiang, Jing-Hao Xue and Weichuan Yu
</p>
<p>Maintainer: Wei Jiang &lt;wjiangaa@connect.ust.hk&gt;
</p>


<h3>References</h3>

<p>Jiang, W., Xue, J-H, and Yu, W.  What is the probability of replicating a statistically significant association in genome-wide association studies?. <em>Submitted</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+repRateEst">repRateEst</a></code>, 
<code><a href="#topic+SEest">SEest</a></code>,
<code><a href="#topic+repSampleSizeRR">repSampleSizeRR</a></code>,
<code><a href="#topic+repSampleSizeRR2">repSampleSizeRR2</a></code>,
<code><a href="#topic+HLtest">HLtest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha&lt;-5e-6               #Significance level in the primary study
alphaR&lt;-5e-3              #Significance level in the replication study
zalpha2&lt;-qnorm(1-alpha/2)
zalphaR2&lt;-qnorm(1-alphaR/2)

##Load data
data('smryStats1')        #Example of summary statistics in 1st study
z1&lt;-smryStats1$Z          #Z values in 1st study
n2.0&lt;-2000                #Number of individuals in control group
n2.1&lt;-2000                #Number of individuals in case group

SE2&lt;-SEest(n2.0, n2.1, smryStats1$F_U, smryStats1$F_A) #SE in replication study
######  RR estimation  ######
RRresult&lt;-repRateEst(log(smryStats1$OR),smryStats1$SE, SE2,zalpha2,zalphaR2, output=TRUE,dir='.')
RR&lt;-RRresult$RR           #Estimated RR
</code></pre>

<hr>
<h2 id='HLtest'>
Hosmer-Lemeshow test
</h2><span id='topic+HLtest'></span>

<h3>Description</h3>

<p>Test whether each element of x is sampled with the probability specified by the corrsponding element in p.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HLtest(x, p, g = 10, null = "all", boot = 1000, info = T, dir = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HLtest_+3A_x">x</code></td>
<td>

<p>A boolean vector.
</p>
</td></tr>
<tr><td><code id="HLtest_+3A_p">p</code></td>
<td>

<p>A probability vector having the same length with x.  
</p>
</td></tr>
<tr><td><code id="HLtest_+3A_g">g</code></td>
<td>

<p>The group number used in the test.
</p>
</td></tr>
<tr><td><code id="HLtest_+3A_null">null</code></td>
<td>

<p>a character in c('all', 'chi2','boot'). 
If null=='chi2', then we use (g-1) degree of freedom chi2 distribution to approximately compute p value.
If null=='boot', then we use parametric bootstrap to compute p value.
If null=='all', then both methods are used. This is the default option.
</p>
</td></tr>
<tr><td><code id="HLtest_+3A_boot">boot</code></td>
<td>

<p>The resampling times to compute p value. Only effective when null=='boot' or 'all'
</p>
</td></tr>
<tr><td><code id="HLtest_+3A_info">info</code></td>
<td>

<p>Draw the null distribution of the test statistic.
</p>
</td></tr>
<tr><td><code id="HLtest_+3A_dir">dir</code></td>
<td>

<p>The directory to save the plot of the null distribution. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Null Hypothesis: Each element of x is sampled with a probability which is the corresponding element of p.
We group x to g groups according to p. Then we compare the success proportion with the mean value of p in each group. 
</p>


<h3>Value</h3>

<p>A list is returned:
</p>
<table>
<tr><td><code>H</code></td>
<td>

<p>The test statistic.
</p>
</td></tr>
<tr><td><code>pval_chi2</code></td>
<td>

<p>The p value approximated by using chi2 distribution.
</p>
</td></tr>
<tr><td><code>pval_boot</code></td>
<td>

<p>The p value computed by using parametric bootstrap.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wei Jiang, Jing-Hao Xue and Weichuan Yu
</p>
<p>Maintainer: Wei Jiang &lt;wjiangaa@connect.ust.hk&gt;
</p>


<h3>References</h3>

<p>Hosmer, D. W., &amp; Lemesbow, S. (1980). Goodness of fit tests for the multiple logistic regression model. <em>Communications in statistics-Theory and Methods</em>, 9(10), 1043-1069.
</p>
<p>Jiang, W., Xue, J-H, and Yu, W.  What is the probability of replicating a statistically significant association in genome-wide association studies?. <em>Submitted</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RRate">RRate</a></code>
<code><a href="#topic+repRateEst">repRateEst</a></code>, 
<code><a href="#topic+SEest">SEest</a></code>,
<code><a href="#topic+repSampleSizeRR">repSampleSizeRR</a></code>,
<code><a href="#topic+repSampleSizeRR2">repSampleSizeRR2</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha&lt;-5e-6               #Significance level in the primary study
alphaR&lt;-5e-3              #Significance level in the replication study
zalpha2&lt;-qnorm(1-alpha/2)
zalphaR2&lt;-qnorm(1-alphaR/2)

##Load data
data('smryStats1')        #Example of summary statistics in 1st study
n2.0&lt;-2000                #Number of individuals in control group
n2.1&lt;-2000                #Number of individuals in case group

SE2&lt;-SEest(n2.0, n2.1, smryStats1$F_U, smryStats1$F_A) #SE in replication study
######  RR estimation  ######
RRresult&lt;-repRateEst(log(smryStats1$OR),smryStats1$SE, SE2,zalpha2,zalphaR2, output=TRUE,dir='.')

#### Hosmer-Lemeshow test  ####
data('smryStats2')        #Example of summary statistics in 2nd study
sigIdx&lt;-(smryStats1$P&lt;alpha)
repIdx&lt;-(sign(smryStats1$Z[sigIdx])*smryStats2$Z[sigIdx]&gt;zalphaR2)
groupNum&lt;-10 
HLresult&lt;-HLtest(repIdx,RRresult$RR,g=groupNum,dir='.')
</code></pre>

<hr>
<h2 id='repSampleSizeRR'>
Sample size determination for the replication study based on RR
</h2><span id='topic+repSampleSizeRR'></span><span id='topic+repSampleSizeRR2'></span>

<h3>Description</h3>

<p><code>repSampleSizeRR</code> and <code>repSampleSizeRR2</code> implement the RR-based sample size determination method for the replication study. If the replication study has the same control-to-case ratio with the primary study, then <code>repSampleSizeRR</code> can be used. Otherwise, <code>repSampleSize2</code> is more suitable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>repSampleSizeRR(GRR, n, MUhat, SE, zalpha2, zalphaR2, idx = TRUE)

repSampleSizeRR2(GRR,CCR2, MUhat,SE,fU,fA,zalpha2,zalphaR2, idx=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="repSampleSizeRR_+3A_grr">GRR</code></td>
<td>

<p>The desired global replication rate.
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_n">n</code></td>
<td>

<p>Sample size in the primary study.
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_muhat">MUhat</code></td>
<td>

<p>The observed effect size (log-odds ratio).
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_se">SE</code></td>
<td>

<p>The standard error of MUhat.
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_zalpha2">zalpha2</code></td>
<td>

<p>The critical value of z-values in the primary study, i.e. z_alpha/2.
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_zalphar2">zalphaR2</code></td>
<td>

<p>The critical value of z-values in the replication study, i.e. z_alphaR/2.
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_idx">idx</code></td>
<td>

<p>The indexes of the SNPs having been further inverstigated in the replication study. We only calculate RR for primary associations with indexes in <code>idx</code>.
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_ccr2">CCR2</code></td>
<td>

<p>The control-to-case ratio of the replication study.
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_fu">fU</code></td>
<td>

<p>The allele frequency in the control group.
</p>
</td></tr>
<tr><td><code id="repSampleSizeRR_+3A_fa">fA</code></td>
<td>

<p>The allele frequency in the case group.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The determined sample size of the replication study is returned.
</p>


<h3>Author(s)</h3>

<p>Wei Jiang, Jing-Hao Xue and Weichuan Yu
</p>
<p>Maintainer: Wei Jiang &lt;wjiangaa@connect.ust.hk&gt;
</p>


<h3>References</h3>

<p>Jiang, W., Xue, J-H, and Yu, W.  What is the probability of replicating a statistically significant association in genome-wide association studies?. <em>Submitted</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RRate">RRate</a></code>
<code><a href="#topic+repRateEst">repRateEst</a></code>, 
<code><a href="#topic+SEest">SEest</a></code>,
<code><a href="#topic+HLtest">HLtest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha&lt;-5e-6               #Significance level in the primary study
alphaR&lt;-5e-3              #Significance level in the replication study
zalpha2&lt;-qnorm(1-alpha/2)
zalphaR2&lt;-qnorm(1-alphaR/2)

##Load data
data('smryStats1')        #Example of summary statistics in 1st study
#### Sample size determination ###
n1&lt;-4000                  #Sample size of the primary study
n2_1&lt;-repSampleSizeRR(0.8, n1, log(smryStats1$OR),smryStats1$SE,zalpha2,zalphaR2)

CCR2&lt;-2                   #Control-to-case ration in the replication study
n2_2&lt;-repSampleSizeRR2(0.8, CCR2, log(smryStats1$OR),smryStats1$SE,smryStats1$F_U, 
smryStats1$F_A,zalpha2,zalphaR2)
</code></pre>

<hr>
<h2 id='RRate-functions'>
Estimating Replication Rate for primary associations
</h2><span id='topic+RRate-functions'></span><span id='topic+repRateEst'></span>

<h3>Description</h3>

<p><code>repRateEst</code> implements a replication rate estimation method. Two-component mixture prior is used in the estimation. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>repRateEst(MUhat, SE, SE2, zalpha2, zalphaR2, boot = 100, output = TRUE, 
idx = TRUE, dir = "output", info = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RRate-functions_+3A_muhat">MUhat</code></td>
<td>

<p>The observed effect size (log-odds ratio) in the primary study.
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_se">SE</code></td>
<td>

<p>The standard error of the observed log-odds ratio in the primary study.
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_se2">SE2</code></td>
<td>

<p>The standard error of the observed log-odds ratio in the replication study.
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_zalpha2">zalpha2</code></td>
<td>

<p>The critical value of z-values in the primary study, i.e. z_alpha/2.
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_zalphar2">zalphaR2</code></td>
<td>

<p>The critical value of z-values in the replication study, i.e. z_alphaR/2.
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_boot">boot</code></td>
<td>

<p>The resampling number of bootstrop used for estimating the credible interval of
the RR.
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_output">output</code></td>
<td>

<p>Bool value.  To determine whether to output the estimated results in the
dir or not.
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_idx">idx</code></td>
<td>

<p>The indexes of the SNPs having been further inverstigated in the replication study. We only calculate RR for primary associations with indexes in <code>idx</code>.
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_dir">dir</code></td>
<td>

<p>The directory to save the estimated results. It has effect when <code>output=T</code>
</p>
</td></tr>
<tr><td><code id="RRate-functions_+3A_info">info</code></td>
<td>

<p>Bool value. To determine whether to show the parematers inference results in the terminal or not.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The RR estimation is based on the following two-component mixture model:
mu=pi_0 delta_0+(1-pi_0) N(0, sigma_0^2).
</p>
<p>Details can be seen the following reference paper.
</p>


<h3>Value</h3>

<p><code>repRateEst</code> returns the RR, lfdr, prediction power and infered parameters. The returened value is a LIST:
</p>
<table>
<tr><td><code>idx</code></td>
<td>

<p>The index of the SNPs which RR are estimated.
</p>
</td></tr>
<tr><td><code>pi0</code></td>
<td>

<p>The proportion of nonassociated SNPs.
</p>
</td></tr>
<tr><td><code>sigma02</code></td>
<td>

<p>The variance of the associated SNPs' effect sizes 
</p>
</td></tr>
<tr><td><code>RR</code></td>
<td>

<p>Estimated replication rate.
</p>
</td></tr>
<tr><td><code>RRlow</code></td>
<td>

<p>The lower limit of the 95% CI for RR. 
</p>
</td></tr>
<tr><td><code>RRhigh</code></td>
<td>

<p>The upper limit of the 95% CI for RR.
</p>
</td></tr>
<tr><td><code>lfdr</code></td>
<td>

<p>Estimated local false discovery rate of the primary study
</p>
</td></tr>
<tr><td><code>lfdrLow</code></td>
<td>

<p>The lower limit of the 95% CI for lfdr. 
</p>
</td></tr>
<tr><td><code>lfdrHigh</code></td>
<td>

<p>The upper limit of the 95% CI for lfdr. 
</p>
</td></tr>
<tr><td><code>predPower</code></td>
<td>

<p>The Bayesian predictive power of the replication study.
</p>
</td></tr>
<tr><td><code>predPowerLow</code></td>
<td>

<p>The lower limit of the 95% CI for predPower. 
</p>
</td></tr>
<tr><td><code>perdPowerHigh</code></td>
<td>

<p>The upper limit of the 95% CI for predPower. 
</p>
</td></tr>
<tr><td><code>GRR</code></td>
<td>

<p>The Global Replication Rate (Mean value of RR)
</p>
</td></tr>
<tr><td><code>GRRlow</code></td>
<td>

<p>The lower limit of the 95% CI for GRR. 
</p>
</td></tr>
<tr><td><code>GRRhigh</code></td>
<td>

<p>The upper limit of the 95% CI for GRR. 
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wei Jiang, Jing-Hao Xue and Weichuan Yu
</p>
<p>Maintainer: Wei Jiang &lt;wjiangaa@connect.ust.hk&gt;
</p>


<h3>References</h3>

<p>Jiang, W., Xue, J-H, and Yu, W.  What is the probability of replicating a statistically significant association in genome-wide association studies?. <em>Submitted</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RRate">RRate</a></code>, 
<code><a href="#topic+SEest">SEest</a></code>,
<code><a href="#topic+repSampleSizeRR">repSampleSizeRR</a></code>,
<code><a href="#topic+repSampleSizeRR2">repSampleSizeRR2</a></code>,
<code><a href="#topic+HLtest">HLtest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha&lt;-5e-6               #Significance level in the primary study
alphaR&lt;-5e-3              #Significance level in the replication study
zalpha2&lt;-qnorm(1-alpha/2)
zalphaR2&lt;-qnorm(1-alphaR/2)

##Load data
data('smryStats1')        #Example of summary statistics in 1st study
n2.0&lt;-2000                #Number of individuals in control group
n2.1&lt;-2000                #Number of individuals in case group

SE2&lt;-SEest(n2.0, n2.1, smryStats1$F_U, smryStats1$F_A) #SE in replication study
######  RR estimation  ######
RRresult&lt;-repRateEst(log(smryStats1$OR),smryStats1$SE, SE2,zalpha2,zalphaR2, output=TRUE,dir='.')
RR&lt;-RRresult$RR           #Estimated RR
</code></pre>

<hr>
<h2 id='SEest'>
Estimating standard error of the observed log-odds ratio
</h2><span id='topic+SEest'></span>

<h3>Description</h3>

<p><code>SEest</code> implements the Woolf's method to estimate the standard error of the observed log-odds ratio. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SEest(n0,n1,fU,fA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SEest_+3A_n0">n0</code></td>
<td>

<p>Sample size in the control group.
</p>
</td></tr>
<tr><td><code id="SEest_+3A_n1">n1</code></td>
<td>

<p>Sample size in the case group.
</p>
</td></tr>
<tr><td><code id="SEest_+3A_fu">fU</code></td>
<td>

<p>Allele frequency in the control group.
</p>
</td></tr>
<tr><td><code id="SEest_+3A_fa">fA</code></td>
<td>

<p>Allele frequency in the case group. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Woolf's method to estimate the standard error of log(OR) is based on the following formula:
</p>
<p>se(log(OR))=sqrt(1/(n0 fU(1-fU))+1/(n1 fA(1-fA))).
</p>


<h3>Value</h3>

<p>The estimated standard error is returned.
</p>


<h3>Author(s)</h3>

<p>Wei Jiang, Jing-Hao Xue and Weichuan Yu
</p>
<p>Maintainer: Wei Jiang &lt;wjiangaa@connect.ust.hk&gt;
</p>


<h3>References</h3>

<p>Woolf, B. (1955). On estimating the relation between blood group and disease. <em>Ann Hum Genet</em>, 19(4), 251-253.
</p>
<p>Jiang, W., Xue, J-H, and Yu, W.  What is the probability of replicating a statistically significant association in genome-wide association studies?. <em>Submitted</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RRate">RRate</a></code>,
<code><a href="#topic+repRateEst">repRateEst</a></code>, 
<code><a href="#topic+repSampleSizeRR">repSampleSizeRR</a></code>,
<code><a href="#topic+repSampleSizeRR2">repSampleSizeRR2</a></code>,
<code><a href="#topic+HLtest">HLtest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Load data
data('smryStats1')        #Example of summary statistics in 1st study
n2.0&lt;-2000                #Number of individuals in control group
n2.1&lt;-2000                #Number of individuals in case group

SE2&lt;-SEest(n2.0, n2.1, smryStats1$F_U, smryStats1$F_A) #SE in replication study
</code></pre>

<hr>
<h2 id='smryStats'>
Example summary statistics
</h2><span id='topic+smryStats1'></span><span id='topic+smryStats2'></span><span id='topic+param'></span>

<h3>Description</h3>

<p>Example summary statistics from the primary and replication studies in GWAS. These summary statistics are generated from a simulated study. The primary study (<code>smryStats1</code>) has 2000 cases and 2000 controls. The replication study (<code>smryStats2</code>) has 1000 cases and 1000 controls. The SNP number is 10,000. The disease pravalence is 1%. Minor allele frequencies are from U(0.05, 0.5), and log-odds ratio are from 0.95N(0,1)+0.05N(0, 0.04). The ground-truth parameters for each SNP are in <code>param</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(smryStats1)

data(smryStats2)

data(param)
</code></pre>


<h3>Details</h3>

<p>The data sets are described as 
</p>
<p>SNP: SNP ID
</p>
<p>F_A: Allele frequency in case group
</p>
<p>F_U: Allele frequency in control group
</p>
<p>Z: Z value from log-odds ratio test
</p>
<p>P: P value
</p>
<p>OR: Odds ratio
</p>
<p>SE: Standard error of log(OR)
</p>


<h3>Author(s)</h3>

<p>Wei Jiang, Jing-Hao Xue and Weichuan Yu
</p>
<p>Maintainer: Wei Jiang &lt;wjiangaa@connect.ust.hk&gt;
</p>


<h3>References</h3>

<p>Jiang, W., Xue, J-H, and Yu, W.  What is the probability of replicating a statistically significant association in genome-wide association studies?. <em>Submitted</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RRate">RRate</a></code>
<code><a href="#topic+repRateEst">repRateEst</a></code>, 
<code><a href="#topic+SEest">SEest</a></code>,
<code><a href="#topic+repSampleSizeRR">repSampleSizeRR</a></code>,
<code><a href="#topic+repSampleSizeRR2">repSampleSizeRR2</a></code>,
<code><a href="#topic+HLtest">HLtest</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
