<!DOCTYPE html><html><head><title>Help for package rrBLUP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rrBLUP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rrBLUP-package'><p>Ridge regression and other kernels for genomic selection</p></a></li>
<li><a href='#A.mat'><p>Additive relationship matrix</p></a></li>
<li><a href='#GWAS'>
<p>Genome-wide association analysis</p></a></li>
<li><a href='#kin.blup'>
<p>Genotypic value prediction based on kinship</p></a></li>
<li><a href='#kinship.BLUP'>
<p>Genomic prediction by kinship-BLUP (deprecated)</p></a></li>
<li><a href='#mixed.solve'>
<p>Mixed-model solver</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Ridge Regression and Other Kernels for Genomic Selection</td>
</tr>
<tr>
<td>Version:</td>
<td>4.6.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Jeffrey Endelman </td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jeffrey Endelman &lt;endelman@wisc.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, graphics, grDevices, parallel</td>
</tr>
<tr>
<td>Description:</td>
<td>Software for genomic prediction with the RR-BLUP mixed model (Endelman 2011, &lt;<a href="https://doi.org/10.3835%2Fplantgenome2011.08.0024">doi:10.3835/plantgenome2011.08.0024</a>&gt;). One application is to estimate marker effects by ridge regression; alternatively, BLUPs can be calculated based on an additive relationship matrix or a Gaussian kernel.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;<a href="https://potatobreeding.cals.wisc.edu/software/&amp;gt;">https://potatobreeding.cals.wisc.edu/software/&gt;</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-09 20:44:08 UTC; endelman</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-10 17:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='rrBLUP-package'>Ridge regression and other kernels for genomic selection</h2><span id='topic+rrBLUP-package'></span>

<h3>Description</h3>

<p>This package has been developed primarily for genomic prediction with mixed models (but it can also do genome-wide association mapping with <code><a href="#topic+GWAS">GWAS</a></code>). The heart of the package is the function <code><a href="#topic+mixed.solve">mixed.solve</a></code>, which is a general-purpose solver for mixed models with a single variance component other than the error.  Genomic predictions can be made by estimating marker effects (RR-BLUP) or by estimating line effects (G-BLUP). In Endelman (2011) I made the poor choice of using the letter G to denotype the genotype or marker data.  To be consistent with Endelman (2011) I have retained this notation in <code><a href="#topic+kinship.BLUP">kinship.BLUP</a></code>.  However, that function has now been superseded by <code><a href="#topic+kin.blup">kin.blup</a></code> and <code><a href="#topic+A.mat">A.mat</a></code>, the latter being a utility for estimating the additive relationship matrix (A) from markers.  In these newer functions I adopt the usual convention that G is the genetic covariance (not the marker data), which is also consistent with the notation in Endelman and Jannink (2012).
</p>
<p>Vignettes illustrating some of the features of this package can be found at <a href="https://potatobreeding.cals.wisc.edu/software/">https://potatobreeding.cals.wisc.edu/software/</a>.
</p>


<h3>References</h3>

<p>Endelman, J.B. 2011. Ridge regression and other kernels for genomic selection with R package rrBLUP. Plant Genome 4:250-255. &lt;doi:10.3835/plantgenome2011.08.0024&gt;
</p>
<p>Endelman, J.B., and J.-L. Jannink. 2012. Shrinkage estimation of the realized relationship matrix. G3:Genes, Genomes, Genetics 2:1405-1413. &lt;doi:10.1534/g3.112.004259&gt;
</p>

<hr>
<h2 id='A.mat'>Additive relationship matrix</h2><span id='topic+A.mat'></span>

<h3>Description</h3>

<p>Calculates the realized additive relationship matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A.mat(
  X,
  min.MAF = NULL,
  max.missing = NULL,
  impute.method = "mean",
  tol = 0.02,
  n.core = 1,
  shrink = FALSE,
  return.imputed = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="A.mat_+3A_x">X</code></td>
<td>
<p>matrix (<code class="reqn">n \times m</code>) of unphased genotypes for <code class="reqn">n</code> lines and <code class="reqn">m</code> biallelic markers, coded as {-1,0,1}. Fractional (imputed) and missing values (NA) are allowed.</p>
</td></tr>
<tr><td><code id="A.mat_+3A_min.maf">min.MAF</code></td>
<td>
<p>Minimum minor allele frequency. The A matrix is not sensitive to rare alleles, so by default only monomorphic markers are removed.</p>
</td></tr>
<tr><td><code id="A.mat_+3A_max.missing">max.missing</code></td>
<td>
<p>Maximum proportion of missing data; default removes completely missing markers.</p>
</td></tr>
<tr><td><code id="A.mat_+3A_impute.method">impute.method</code></td>
<td>
<p>There are two options. The default is &quot;mean&quot;, which imputes with the mean for each marker.  The &quot;EM&quot; option imputes with an EM algorithm (see details).</p>
</td></tr>
<tr><td><code id="A.mat_+3A_tol">tol</code></td>
<td>
<p>Specifies the convergence criterion for the EM algorithm (see details).</p>
</td></tr>
<tr><td><code id="A.mat_+3A_n.core">n.core</code></td>
<td>
<p>Specifies the number of cores to use for parallel execution of the EM algorithm</p>
</td></tr>
<tr><td><code id="A.mat_+3A_shrink">shrink</code></td>
<td>
<p>set shrink=FALSE to disable shrinkage estimation.  See Details for how to enable shrinkage estimation.</p>
</td></tr>
<tr><td><code id="A.mat_+3A_return.imputed">return.imputed</code></td>
<td>
<p>When TRUE, the imputed marker matrix is returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>At high marker density, the relationship matrix is estimated as <code class="reqn">A=W W'/c</code>, where <code class="reqn">W_{ik} = X_{ik} + 1 - 2 p_k</code> and <code class="reqn">p_k</code> is the frequency of the 1 allele at marker k.  By using a normalization constant of <code class="reqn">c = 2 \sum_k {p_k (1-p_k)}</code>, the mean of the diagonal elements is <code class="reqn">1 + f</code> (Endelman and Jannink 2012).
The EM imputation algorithm is based on the multivariate normal distribution and was designed for use with GBS (genotyping-by-sequencing) markers, which tend to be high density but with lots of missing data.  Details are given in Poland et al. (2012). The EM algorithm stops at iteration <code class="reqn">t</code> when the RMS error = <code class="reqn">n^{-1} \|A_{t} - A_{t-1}\|_2</code> &lt; tol.  
Shrinkage estimation can improve the accuracy of genome-wide marker-assisted selection, particularly at low marker density (Endelman and Jannink 2012). The shrinkage intensity ranges from 0 (no shrinkage) to 1 (<code class="reqn">A=(1+f)I</code>). Two algorithms for estimating the shrinkage intensity are available.  The first is the method described in Endelman and Jannink (2012) and is specified by <code>shrink=list(method="EJ")</code>.  The second involves designating a random sample of the markers as simulated QTL and then regressing the A matrix based on the QTL against the A matrix based on the remaining markers (Yang et al. 2010; Mueller et al. 2015). The regression method is specified by <code>shrink=list(method="REG",n.qtl=100,n.iter=5)</code>, where the parameters <code>n.qtl</code> and <code>n.iter</code> can be varied to adjust the number of simulated QTL and number of iterations, respectively.  
The shrinkage and EM-imputation options are designed for opposite scenarios (low vs. high density) and cannot be used simultaneously. 
When the EM algorithm is used, the imputed alleles can lie outside the interval [-1,1].  Polymorphic markers that do not meet the min.MAF and max.missing criteria are not imputed.
</p>


<h3>Value</h3>

<p>If return.imputed = FALSE, the <code class="reqn">n \times n</code> additive relationship matrix is returned. If return.imputed = TRUE, the function returns a list containing
</p>

<dl>
<dt>$A</dt><dd><p>the A matrix</p>
</dd>
<dt>$imputed</dt><dd><p>the imputed marker matrix</p>
</dd>
</dl>



<h3>References</h3>

<p>Endelman, J.B., and J.-L. Jannink. 2012. Shrinkage estimation of the realized relationship matrix. G3:Genes, Genomes, Genetics. 2:1405-1413. &lt;doi:10.1534/g3.112.004259&gt;
</p>
<p>Mueller et al. 2015. Shrinkage estimation of the genomic relationship matrix can improve genomic estimated breeding values in the training set. Theor Appl Genet 128:693-703. &lt;doi:10.1007/s00122-015-2464-6&gt;
</p>
<p>Poland, J., J. Endelman et al. 2012. Genomic selection in wheat breeding using genotyping-by-sequencing. Plant Genome 5:103-113. &lt;doi:10.3835/plantgenome2012.06.0006&gt;
</p>
<p>Yang et al. 2010. Common SNPs explain a large proportion of the heritability for human height. Nat. Genetics 42:565-569. &lt;doi:10.1038/ng.608&gt;
</p>

<hr>
<h2 id='GWAS'>
Genome-wide association analysis
</h2><span id='topic+GWAS'></span>

<h3>Description</h3>

<p>Performs genome-wide association analysis based on the mixed model (Yu et al. 2006):
</p>
<p style="text-align: center;"><code class="reqn">y = X \beta + Z g + S \tau + \varepsilon</code>
</p>

<p>where <code class="reqn">\beta</code> is a vector of fixed effects that can model both environmental factors and population structure.  
The variable <code class="reqn">g</code> models the genetic background of each line as a random effect with <code class="reqn">Var[g] = K \sigma^2</code>.  
The variable <code class="reqn">\tau</code> models the additive SNP effect as a fixed effect.  The residual variance is <code class="reqn">Var[\varepsilon] = I \sigma_e^2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWAS(pheno, geno, fixed=NULL, K=NULL, n.PC=0,
     min.MAF=0.05, n.core=1, P3D=TRUE, plot=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GWAS_+3A_pheno">pheno</code></td>
<td>

<p>Data frame where the first column is the line name (gid). The remaining columns can be either a phenotype or the levels of a fixed effect.  Any column not designated as a fixed effect is assumed to be a phenotype.
</p>
</td></tr>
<tr><td><code id="GWAS_+3A_geno">geno</code></td>
<td>

<p>Data frame with the marker names in the first column.  The second and third columns contain the chromosome and map position (either bp or cM), respectively, which are used only when plot=TRUE to make Manhattan plots.  If the markers are unmapped, just use a placeholder for those two columns.  Columns 4 and higher contain the marker scores for each line, coded as {-1,0,1} = {aa,Aa,AA}.  Fractional (imputed) and missing (NA) values are allowed.  The column names must match the line names in the &quot;pheno&quot; data frame.
</p>
</td></tr>
<tr><td><code id="GWAS_+3A_fixed">fixed</code></td>
<td>

<p>An array of strings containing the names of the columns that should be included as (categorical) fixed effects in the mixed model.
</p>
</td></tr>
<tr><td><code id="GWAS_+3A_k">K</code></td>
<td>

<p>Kinship matrix for the covariance between lines due to a polygenic effect.  If not passed, it is calculated from the markers using <code><a href="#topic+A.mat">A.mat</a></code>.
</p>
</td></tr>
<tr><td><code id="GWAS_+3A_n.pc">n.PC</code></td>
<td>

<p>Number of principal components to include as fixed effects.  Default is 0 (equals K model).
</p>
</td></tr>
<tr><td><code id="GWAS_+3A_min.maf">min.MAF</code></td>
<td>

<p>Specifies the minimum minor allele frequency (MAF).  If a marker has a MAF less than min.MAF, it is assigned a zero score.  
</p>
</td></tr>
<tr><td><code id="GWAS_+3A_n.core">n.core</code></td>
<td>

<p>Setting n.core &gt; 1 will enable parallel execution on a machine with multiple cores (use only at UNIX command line).  
</p>
</td></tr>
<tr><td><code id="GWAS_+3A_p3d">P3D</code></td>
<td>

<p>When P3D=TRUE, variance components are estimated by REML only once, without any markers in the model.  When P3D=FALSE, variance components are estimated by REML for each marker separately. 
</p>
</td></tr>
<tr><td><code id="GWAS_+3A_plot">plot</code></td>
<td>

<p>When plot=TRUE, qq and Manhattan plots are generated.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For unbalanced designs where phenotypes come from different environments, the environment mean can be modeled using the fixed option (e.g., fixed=&quot;env&quot; if the column in the pheno data.frame is called &quot;env&quot;).  When principal components are included (P+K model), the loadings are determined from an eigenvalue decomposition of the K matrix.
</p>
<p>The terminology &quot;P3D&quot; (population parameters previously determined) was introduced by Zhang et al. (2010).  When P3D=FALSE, this function is equivalent to EMMA with REML (Kang et al. 2008).  When P3D=TRUE, it is equivalent to EMMAX (Kang et al. 2010).  The P3D=TRUE option is faster but can underestimate significance compared to P3D=FALSE.
</p>
<p>The dashed line in the Manhattan plots corresponds to an FDR rate of 0.05 and is calculated using the qvalue package (Storey and Tibshirani 2003).  The p-value corresponding to a q-value of 0.05 is determined by interpolation.  When there are no q-values less than 0.05, the dashed line is omitted.  
</p>


<h3>Value</h3>

<p>Returns a data frame where the first three columns are the marker name, chromosome, and position, and subsequent columns are the marker scores <code class="reqn">(-log_{10}p)</code> for the traits.
</p>


<h3>References</h3>

<p>Kang et al. 2008. Efficient control of population structure in model organism association mapping. Genetics 178:1709-1723.
</p>
<p>Kang et al. 2010. Variance component model to account for sample structure in genome-wide association studies.
Nat. Genet. 42:348-354.
</p>
<p>Storey and Tibshirani. 2003. Statistical significance for genome-wide studies. PNAS 100:9440-9445.
</p>
<p>Yu et al. 2006. A unified mixed-model method for association mapping that accounts for multiple levels of relatedness. Genetics 38:203-208. 
</p>
<p>Zhang et al. 2010. Mixed linear model approach adapted for genome-wide association studies. Nat. Genet. 42:355-360.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#random population of 200 lines with 1000 markers
M &lt;- matrix(rep(0,200*1000),1000,200)
for (i in 1:200) {
  M[,i] &lt;- ifelse(runif(1000)&lt;0.5,-1,1)
}
colnames(M) &lt;- 1:200
geno &lt;- data.frame(marker=1:1000,chrom=rep(1,1000),pos=1:1000,M,check.names=FALSE)

QTL &lt;- 100*(1:5) #pick 5 QTL
u &lt;- rep(0,1000) #marker effects
u[QTL] &lt;- 1
g &lt;- as.vector(crossprod(M,u))
h2 &lt;- 0.5
y &lt;- g + rnorm(200,mean=0,sd=sqrt((1-h2)/h2*var(g)))

pheno &lt;- data.frame(line=1:200,y=y)
scores &lt;- GWAS(pheno,geno,plot=FALSE)
</code></pre>

<hr>
<h2 id='kin.blup'>
Genotypic value prediction based on kinship
</h2><span id='topic+kin.blup'></span>

<h3>Description</h3>

<p>Genotypic value prediction by G-BLUP, where the genotypic covariance G can be additive or based on a Gaussian kernel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kin.blup(data,geno,pheno,GAUSS=FALSE,K=NULL,fixed=NULL,covariate=NULL,
         PEV=FALSE,n.core=1,theta.seq=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kin.blup_+3A_data">data</code></td>
<td>

<p>Data frame with columns for the phenotype, the genotype identifier, and any environmental variables.
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_geno">geno</code></td>
<td>

<p>Character string for the name of the column in the data frame that contains the genotype identifier.
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_pheno">pheno</code></td>
<td>

<p>Character string for the name of the column in the data frame that contains the phenotype.
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_gauss">GAUSS</code></td>
<td>

<p>To model genetic covariance with a Gaussian kernel, set GAUSS=TRUE and pass the Euclidean distance for K (see below).
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_k">K</code></td>
<td>

<p>There are three options for specifying kinship:  
(1) If K=NULL, genotypes are assumed to be independent <code class="reqn">(G=I \: V_g)</code>. 
(2) For breeding value prediction, set GAUSS=FALSE and use an additive relationship matrix for K to create the model <code class="reqn">(G=K \: V_g)</code>. 
(3) For the Gaussian kernel, set GAUSS=TRUE and pass the Euclidean distance matrix for K to create the model <code class="reqn">G_{ij}=e^{-(K_{ij}/\theta)^2} \: V_g</code>.  
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_fixed">fixed</code></td>
<td>

<p>An array of strings containing the names of columns that should be included as (categorical) fixed effects in the mixed model.
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_covariate">covariate</code></td>
<td>

<p>An array of strings containing the names of columns that should be included as covariates in the mixed model.
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_pev">PEV</code></td>
<td>

<p>When PEV=TRUE, the function returns the prediction error variance for the genotypic values (<code class="reqn">PEV_i = Var[g^*_i-g_i]</code>). 
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_n.core">n.core</code></td>
<td>

<p>Specifies the number of cores to use for parallel execution of the Gaussian kernel method (use only at UNIX command line).
</p>
</td></tr>
<tr><td><code id="kin.blup_+3A_theta.seq">theta.seq</code></td>
<td>

<p>The scale parameter for the Gaussian kernel is set by maximizing the restricted log-likelihood over a grid of values. By default, the grid is constructed by dividing the interval (0,max(K)] into 10 points.  Passing a numeric array to this variable (theta.seq = &quot;theta sequence&quot;) will specify a different set of grid points (e.g., for large problems you might want fewer than 10).  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper for <code><a href="#topic+mixed.solve">mixed.solve</a></code> and thus solves mixed models of the form:
</p>
<p style="text-align: center;"><code class="reqn">y = X \beta + [Z \: 0] g + \varepsilon</code>
</p>

<p>where <code class="reqn">\beta</code> is a vector of fixed effects, <code class="reqn">g</code> is a vector of random genotypic values with covariance 
<code class="reqn">G = Var[g]</code>, and the residuals follow <code class="reqn">Var[\varepsilon_i] = R_i \sigma^2_e</code>, with <code class="reqn">R_i = 1</code> by default.  The design matrix for the genetic values has been partitioned to illustrate that not all lines need phenotypes (i.e., for genomic selection).  Unlike <code><a href="#topic+mixed.solve">mixed.solve</a></code>, this function does not return estimates of the fixed effects, only the BLUP solution for the genotypic values.  It was designed to replace <code><a href="#topic+kinship.BLUP">kinship.BLUP</a></code> and to relieve the user of having to explicitly construct design matrices.  Variance components are estimated by REML and BLUP values are returned for every entry in K, regardless of whether it has been phenotyped.  The rownames of K must match the genotype labels in the data frame for phenotyped lines; missing phenotypes (NA) are simply omitted.  
</p>
<p>Unlike its predecessor, this function does not handle marker data directly.  For breeding value prediction, the user must supply a relationship matrix, which can be calculated from markers with <code><a href="#topic+A.mat">A.mat</a></code>.  For Gaussian kernel predictions, pass the Euclidean distance matrix for K, which can be calculated with <code><a href="stats.html#topic+dist">dist</a></code>.  
</p>
<p>In the terminology of mixed models, both the &quot;fixed&quot; and &quot;covariate&quot; variables are fixed effects (<code class="reqn">\beta</code> in the above equation): the former are treated as factors with distinct levels while the latter are continuous with one coefficient per variable.  The population mean is automatically included as a fixed effect.
</p>
<p>The prediction error variance (PEV) is the square of the SE of the BLUPs (see <code><a href="#topic+mixed.solve">mixed.solve</a></code>) and can be used to estimate the expected accuracy of BLUP predictions according to <code class="reqn">r^2_i = 1 - \frac{PEV_i}{V_g K_{ii}}</code>.
</p>


<h3>Value</h3>

<p>The function always returns 
</p>

<dl>
<dt>$Vg</dt><dd><p>REML estimate of the genetic variance</p>
</dd>
<dt>$Ve</dt><dd><p>REML estimate of the error variance</p>
</dd>
<dt>$g</dt><dd><p>BLUP solution for the genetic values</p>
</dd>
<dt>$resid</dt><dd><p>residuals</p>
</dd>
<dt>$pred</dt><dd><p>predicted genetic values, averaged over the fixed effects</p>
</dd>
</dl>

<p>If PEV = TRUE, the list also includes
</p>

<dl>
<dt>$PEV</dt><dd><p>Prediction error variance for the genetic values</p>
</dd>
</dl>

<p>If GAUSS = TRUE, the list also includes
</p>

<dl>
<dt>$profile</dt><dd><p>the log-likelihood profile for the scale parameter in the Gaussian kernel</p>
</dd>
</dl>



<h3>References</h3>

<p>Endelman, J.B. 2011. Ridge regression and other kernels for genomic selection with R package rrBLUP. Plant Genome 4:250-255. &lt;doi:10.3835/plantgenome2011.08.0024&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#random population of 200 lines with 1000 markers
M &lt;- matrix(rep(0,200*1000),200,1000)
for (i in 1:200) {
  M[i,] &lt;- ifelse(runif(1000)&lt;0.5,-1,1)
}
rownames(M) &lt;- 1:200
A &lt;- A.mat(M)

#random phenotypes
u &lt;- rnorm(1000)
g &lt;- as.vector(crossprod(t(M),u))
h2 &lt;- 0.5  #heritability
y &lt;- g + rnorm(200,mean=0,sd=sqrt((1-h2)/h2*var(g)))

data &lt;- data.frame(y=y,gid=1:200)

#predict breeding values
ans &lt;- kin.blup(data=data,geno="gid",pheno="y",K=A)
accuracy &lt;- cor(g,ans$g)

</code></pre>

<hr>
<h2 id='kinship.BLUP'>
Genomic prediction by kinship-BLUP (deprecated)
</h2><span id='topic+kinship.BLUP'></span>

<h3>Description</h3>

<p>***This function has been superseded by <code><a href="#topic+kin.blup">kin.blup</a></code>; please refer to its help page.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinship.BLUP(y, G.train, G.pred=NULL, X=NULL, Z.train=NULL, 
     K.method="RR", n.profile=10, mixed.method="REML", n.core=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kinship.BLUP_+3A_y">y</code></td>
<td>

<p>Vector (<code class="reqn">n.obs \times 1</code>) of observations.  Missing values (NA) are omitted.
</p>
</td></tr>
<tr><td><code id="kinship.BLUP_+3A_g.train">G.train</code></td>
<td>

<p>Matrix (<code class="reqn">n.train \times m</code>) of unphased genotypes for the training population: <code class="reqn">n.train</code> lines with <code class="reqn">m</code> bi-allelic markers.
Genotypes should be coded as {-1,0,1}; fractional (imputed) and missing (NA) alleles are allowed.  
</p>
</td></tr>
<tr><td><code id="kinship.BLUP_+3A_g.pred">G.pred</code></td>
<td>

<p>Matrix (<code class="reqn">n.pred \times m</code>) of unphased genotypes for the prediction population: <code class="reqn">n.pred</code> lines with <code class="reqn">m</code> bi-allelic markers.
Genotypes should be coded as {-1,0,1}; fractional (imputed) and missing (NA) alleles are allowed.  
</p>
</td></tr>
<tr><td><code id="kinship.BLUP_+3A_x">X</code></td>
<td>

<p>Design matrix (<code class="reqn">n.obs \times p</code>) of fixed effects.  If not passed, a vector of 1's is used 
to model the intercept.
</p>
</td></tr>
<tr><td><code id="kinship.BLUP_+3A_z.train">Z.train</code></td>
<td>

<p>0-1 matrix (<code class="reqn">n.obs \times n.train</code>) relating observations to lines in the training set.  If not passed
the identity matrix is used.
</p>
</td></tr>
<tr><td><code id="kinship.BLUP_+3A_k.method">K.method</code></td>
<td>

<p>&quot;RR&quot; (default) is ridge regression, for which K is the realized additive relationship matrix computed with <code><a href="#topic+A.mat">A.mat</a></code>.  The option &quot;GAUSS&quot; is a Gaussian kernel (<code class="reqn">K = e^{-D^2/\theta^2}</code>) and &quot;EXP&quot; is an exponential kernel (<code class="reqn">K = e^{-D/\theta}</code>), where Euclidean distances <code class="reqn">D</code> are computed with <code><a href="stats.html#topic+dist">dist</a></code>.
</p>
</td></tr>
<tr><td><code id="kinship.BLUP_+3A_n.profile">n.profile</code></td>
<td>

<p>For K.method = &quot;GAUSS&quot; or &quot;EXP&quot;, the number of points to use in the log-likelihood profile for the scale parameter <code class="reqn">\theta</code>.
</p>
</td></tr>
<tr><td><code id="kinship.BLUP_+3A_mixed.method">mixed.method</code></td>
<td>

<p>Either &quot;REML&quot; (default) or &quot;ML&quot;.
</p>
</td></tr>
<tr><td><code id="kinship.BLUP_+3A_n.core">n.core</code></td>
<td>

<p>Setting n.core &gt; 1 will enable parallel execution of the Gaussian kernel computation (use only at UNIX command line).   
</p>
</td></tr>
</table>


<h3>Value</h3>


<dl>
<dt>$g.train</dt><dd><p>BLUP solution for the training set</p>
</dd>
<dt>$g.pred</dt><dd><p>BLUP solution for the prediction set (when G.pred != NULL)</p>
</dd>
<dt>$beta</dt><dd><p>ML estimate of fixed effects</p>
</dd>
</dl>

<p>For GAUSS or EXP, function also returns
</p>

<dl>
<dt>$profile</dt><dd><p>log-likelihood profile for the scale parameter</p>
</dd>
</dl>



<h3>References</h3>

<p>Endelman, J.B. 2011. Ridge regression and other kernels for genomic selection with R package rrBLUP. Plant Genome 4:250-255.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#random population of 200 lines with 1000 markers
G &lt;- matrix(rep(0,200*1000),200,1000)
for (i in 1:200) {
  G[i,] &lt;- ifelse(runif(1000)&lt;0.5,-1,1)
}

#random phenotypes
g &lt;- as.vector(crossprod(t(G),rnorm(1000)))
h2 &lt;- 0.5 
y &lt;- g + rnorm(200,mean=0,sd=sqrt((1-h2)/h2*var(g)))

#split in half for training and prediction
train &lt;- 1:100
pred &lt;- 101:200
ans &lt;- kinship.BLUP(y=y[train],G.train=G[train,],G.pred=G[pred,],K.method="GAUSS")

#correlation accuracy
r.gy &lt;- cor(ans$g.pred,y[pred])
</code></pre>

<hr>
<h2 id='mixed.solve'>
Mixed-model solver
</h2><span id='topic+mixed.solve'></span>

<h3>Description</h3>

<p>Calculates maximum-likelihood (ML/REML) solutions for mixed models of the form
</p>
<p style="text-align: center;"><code class="reqn">y = X \beta + Z u + \varepsilon</code>
</p>

<p>where <code class="reqn">\beta</code> is a vector of fixed effects and <code class="reqn">u</code> is a vector of random effects with 
<code class="reqn">Var[u] = K \sigma^2_u</code>.  The residual variance is <code class="reqn">Var[\varepsilon] = I \sigma^2_e</code>.  This class
of mixed models, in which there is a single variance component other than the residual error,
has a close relationship with ridge regression (ridge parameter <code class="reqn">\lambda = \sigma_e^2 / \sigma^2_u</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixed.solve(y, Z=NULL, K=NULL, X=NULL, method="REML", 
        bounds=c(1e-09, 1e+09), SE=FALSE, return.Hinv=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixed.solve_+3A_y">y</code></td>
<td>

<p>Vector (<code class="reqn">n \times 1</code>) of observations.  Missing values (NA) are omitted, along with the corresponding rows of X and Z. 
</p>
</td></tr>
<tr><td><code id="mixed.solve_+3A_z">Z</code></td>
<td>

<p>Design matrix (<code class="reqn">n \times m</code>) for the random effects.  If not passed, assumed to be the identity matrix.
</p>
</td></tr>
<tr><td><code id="mixed.solve_+3A_k">K</code></td>
<td>

<p>Covariance matrix (<code class="reqn">m \times m</code>) for random effects; must be positive semi-definite.  If not passed, assumed to 
be the identity matrix.
</p>
</td></tr>
<tr><td><code id="mixed.solve_+3A_x">X</code></td>
<td>

<p>Design matrix (<code class="reqn">n \times p</code>) for the fixed effects.  If not passed, a vector of 1's is used
to model the intercept.  X must be full column rank (implies <code class="reqn">\beta</code> is estimable).
</p>
</td></tr>
<tr><td><code id="mixed.solve_+3A_method">method</code></td>
<td>

<p>Specifies whether the full (&quot;ML&quot;) or restricted (&quot;REML&quot;) maximum-likelihood method is used.
</p>
</td></tr>
<tr><td><code id="mixed.solve_+3A_bounds">bounds</code></td>
<td>

<p>Array with two elements specifying the lower and upper bound for the ridge parameter.
</p>
</td></tr>
<tr><td><code id="mixed.solve_+3A_se">SE</code></td>
<td>

<p>If TRUE, standard errors are calculated.
</p>
</td></tr>
<tr><td><code id="mixed.solve_+3A_return.hinv">return.Hinv</code></td>
<td>

<p>If TRUE, the function returns the inverse of <code class="reqn">H = Z K Z' + \lambda I</code>.  This is useful for <code><a href="#topic+GWAS">GWAS</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to predict marker effects or breeding values (see examples).  The numerical method 
is based on the spectral decomposition of <code class="reqn">Z K Z'</code> and <code class="reqn">S Z K Z' S</code>, where <code class="reqn">S = I - X (X' X)^{-1} X'</code> is 
the projection operator for the nullspace of <code class="reqn">X</code> (Kang et al., 2008). This algorithm generates the inverse phenotypic covariance matrix <code class="reqn">V^{-1}</code>, which can then be used to calculate the BLUE and BLUP solutions for the fixed and random effects, respectively, using standard formulas (Searle et al. 1992):
</p>
<p style="text-align: center;"><code class="reqn">BLUE(\beta) = \beta^* = (X'V^{-1}X)^{-1}X'V^{-1}y</code>
</p>

<p style="text-align: center;"><code class="reqn">BLUP(u) = u^* = \sigma^2_u KZ'V^{-1}(y-X\beta^*)</code>
</p>

<p>The standard errors are calculated as the square root of the diagonal elements of the following matrices (Searle et al. 1992):
</p>
<p style="text-align: center;"><code class="reqn">Var[\beta^*] = (X'V^{-1}X)^{-1}</code>
</p>

<p style="text-align: center;"><code class="reqn">Var[u^*-u] = K \sigma^2_u - \sigma^4_u KZ'V^{-1}ZK + \sigma^4_u KZ'V^{-1}XVar[\beta^*]X'V^{-1}ZK</code>
</p>

<p>For marker effects where K = I, the function will run faster if K is not passed than if the user passes the identity matrix.
</p>


<h3>Value</h3>

<p>If SE=FALSE, the function returns a list containing
</p>

<dl>
<dt>$Vu</dt><dd><p>estimator for <code class="reqn">\sigma^2_u</code></p>
</dd>
<dt>$Ve</dt><dd><p>estimator for <code class="reqn">\sigma^2_e</code></p>
</dd>
<dt>$beta</dt><dd><p>BLUE(<code class="reqn">\beta</code>)</p>
</dd>
<dt>$u</dt><dd><p>BLUP(<code class="reqn">u</code>)</p>
</dd>
<dt>$LL</dt><dd><p>maximized log-likelihood (full or restricted, depending on method)</p>
</dd>
</dl>

<p>If SE=TRUE, the list also contains
</p>

<dl>
<dt>$beta.SE</dt><dd><p>standard error for <code class="reqn">\beta</code></p>
</dd>
<dt>$u.SE</dt><dd><p>standard error for <code class="reqn">u^*-u</code></p>
</dd>
</dl>

<p>If return.Hinv=TRUE, the list also contains
</p>

<dl>
<dt>$Hinv</dt><dd><p>the inverse of <code class="reqn">H</code></p>
</dd>
</dl>



<h3>References</h3>

<p>Kang et al. 2008. Efficient control of population structure in model organism association mapping. 
Genetics 178:1709-1723.
</p>
<p>Endelman, J.B. 2011. Ridge regression and other kernels for genomic selection with R package rrBLUP. Plant Genome 4:250-255.
</p>
<p>Searle, S.R., G. Casella and C.E. McCulloch. 1992. Variance Components. John Wiley, Hoboken.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#random population of 200 lines with 1000 markers
M &lt;- matrix(rep(0,200*1000),200,1000)
for (i in 1:200) {
  M[i,] &lt;- ifelse(runif(1000)&lt;0.5,-1,1)
}

#random phenotypes
u &lt;- rnorm(1000)
g &lt;- as.vector(crossprod(t(M),u))
h2 &lt;- 0.5  #heritability
y &lt;- g + rnorm(200,mean=0,sd=sqrt((1-h2)/h2*var(g)))

#predict marker effects
ans &lt;- mixed.solve(y,Z=M)  #By default K = I
accuracy &lt;- cor(u,ans$u)

#predict breeding values
ans &lt;- mixed.solve(y,K=A.mat(M))
accuracy &lt;- cor(g,ans$u)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
