<!DOCTYPE html><html><head><title>Help for package rSEA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rSEA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#plotSEA'><p>topSEA</p></a></li>
<li><a href='#rSEA-package'><p>Simultaneous Enrichment Analysis (SEA) of all possible feature-sets using the unified null hypothesis</p></a></li>
<li><a href='#SEA'><p>SEA</p></a></li>
<li><a href='#setTDP'><p>setTDP</p></a></li>
<li><a href='#setTest'><p>setTest</p></a></li>
<li><a href='#topSEA'><p>topSEA</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simultaneous Enrichment Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Mitra Ebrahimpoor</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mitra Ebrahimpoor&lt;mitra.ebrahimpoor@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>SEA performs simultaneous feature-set testing for (gen)omics data. It tests the unified null hypothesis controls the family-wise error rate for all possible pathways. The unified null hypothesis is defined as: "The proportion of true features in the set is less than or equal to the threshold c", where c is selected by the user. Family-wise error rate control is provided through use of closed testing with Simes test. For more information on closed testing with Simes see Goeman et al. (2019) &lt;<a href="https://doi.org/10.1093%2Fbiomet%2Fasz041">doi:10.1093/biomet/asz041</a>&gt; and for more information about the properties and performance of SEA procedure see Ebrahimpoor et al. (2019) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbz074">doi:10.1093/bib/bbz074</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mitra-ep/rSEA">https://github.com/mitra-ep/rSEA</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), hommel (&ge; 1.4), ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-03-23</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-03-23 11:10:31 UTC; mebrahimpoor</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-03-23 11:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='plotSEA'>topSEA</h2><span id='topic+plotSEA'></span>

<h3>Description</h3>

<p>returns a plotof SEA-chart which illustrates
roportion of discoveries per pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSEA(object, by = "TDP.estimate", threshold = 0.005, n = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSEA_+3A_object">object</code></td>
<td>
<p>A SEA-chart object which is the output of <code>SEA</code> function</p>
</td></tr>
<tr><td><code id="plotSEA_+3A_by">by</code></td>
<td>
<p>the Variable which will we mapped.
It should be either the TDP estimate or TDP bound.The default is TDP bound.</p>
</td></tr>
<tr><td><code id="plotSEA_+3A_threshold">threshold</code></td>
<td>
<p>A real number between 0 and 1. Which will be used as
a visual aid to distinguish significant pathways</p>
</td></tr>
<tr><td><code id="plotSEA_+3A_n">n</code></td>
<td>
<p>Integer. Number of raws from SEA-chart object to be plotted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a plot of SEA_chart according to the selected arguments
</p>


<h3>Author(s)</h3>

<p>Mitra Ebrahimpoor
</p>
<p><a href="mailto:m.ebrahimpoor@lumc.nl">m.ebrahimpoor@lumc.nl</a>
</p>


<h3>References</h3>

<p>Mitra Ebrahimpoor, Pietro Spitali, Kristina Hettne, Roula Tsonaka, Jelle Goeman,
Simultaneous Enrichment Analysis of all Possible Gene-sets: Unifying Self-Contained
and Competitive Methods, Briefings in Bioinformatics,bbz074
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SEA">SEA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#See the examples for \code{\link{SEA}}

</code></pre>

<hr>
<h2 id='rSEA-package'>Simultaneous Enrichment Analysis (SEA) of all possible feature-sets using the unified null hypothesis
</h2><span id='topic+rSEA-package'></span>

<h3>Description</h3>

<p>This package uses raw p-values of genomic features as input and evaluates any given list of feature-sets or pathways. For each set the adjusted p-value and TDP lower-bound are calculated. The type of test can be defined by arguments and can be refined as necessary. The p-values are corrected for every possible set of features, making the method flexible in choice of pathway list and test type.
For more details see: Ebrahimpoor, M (2019) &lt;doi:10.1093/bib/bbz074&gt;
</p>


<h3>Details</h3>

<p>The unified null hypothesis is tested using closed testing procedure and all-resolutions inference. It combines the self-contained and ompetitive approaches in one framework. In short, using p-values of the individual features as input, the package can provide an FWER-adjusted p-value along with a lower bound and a point estimate for the proportion of true discoveries per feature-set. The flexibility in revising the choice of feature-sets without inflating type-I error is the most important property of SEA.
</p>


<h3>Author(s)</h3>

<p>Mitra Ebrahimpoor.
</p>
<p>Maintainer: Mitra Ebrahimpoor&lt;m.ebrahimpoor@lumc.nl&gt;
</p>


<h3>References</h3>

<p>Mitra Ebrahimpoor, Pietro Spitali, Kristina Hettne, Roula Tsonaka, Jelle Goeman,
Simultaneous Enrichment Analysis of all Possible Gene-sets: Unifying Self-Contained
and Competitive Methods, Briefings in Bioinformatics,bbz074
https://doi.org/10.1093/bib/bbz074
</p>

<hr>
<h2 id='SEA'>SEA</h2><span id='topic+SEA'></span>

<h3>Description</h3>

<p>returns SEA chart (a data.frame) including the test results and estimates for the specified
feature-sets from <code>pathlist</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SEA(pvalue, featureIDs, data, pathlist, select, tdphat = TRUE,
  selfcontained = TRUE, competitive = TRUE, thresh = NULL,
  alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SEA_+3A_pvalue">pvalue</code></td>
<td>
<p>Vector of p-values. It can be the name of the covariate representing the Vector of
all raw p-values in the <code>data</code> or a single vector but in the latter case it should match the
<code>featureIDs</code> vector</p>
</td></tr>
<tr><td><code id="SEA_+3A_featureids">featureIDs</code></td>
<td>
<p>Vector of feature IDs. It can be the name of the covariate representing the IDs in the
<code>data</code> or a single vector but in the latter case it should match the <code>pvalue</code> vector</p>
</td></tr>
<tr><td><code id="SEA_+3A_data">data</code></td>
<td>
<p>Optional data frame or matrix containing the variables in <code>pvalue</code> and <code>featureIDs</code></p>
</td></tr>
<tr><td><code id="SEA_+3A_pathlist">pathlist</code></td>
<td>
<p>A list containing pathways defined by <code>featureIDs</code>. Checkout the vignette
for more details and available codes to create your own pathway</p>
</td></tr>
<tr><td><code id="SEA_+3A_select">select</code></td>
<td>
<p>A vector. Number or names of pathways of interest from the <code>pathlist</code> of choice.
If missing, all pathways of the database will be included</p>
</td></tr>
<tr><td><code id="SEA_+3A_tdphat">tdphat</code></td>
<td>
<p>Logical. If <code>TRUE</code> the point estimate of the True Discoveries Proportion
within each pathway will be calculated</p>
</td></tr>
<tr><td><code id="SEA_+3A_selfcontained">selfcontained</code></td>
<td>
<p>Logical. If <code>TRUE</code> the self-contained null hypothesis will be tested
for each pathway and the corresponding adj. p-value is returned</p>
</td></tr>
<tr><td><code id="SEA_+3A_competitive">competitive</code></td>
<td>
<p>Logical. If <code>TRUE</code> the default competitive null hypothesis will be tested
for each pathway and the corresponding adj. p-value is returned, you can define a threshold with
<code>thresh</code> argument</p>
</td></tr>
<tr><td><code id="SEA_+3A_thresh">thresh</code></td>
<td>
<p>A real number between 0 and 1. If specified, the competitive null hypothesis will be tested
against this threshold for each pathway and the corresponding adj. p-value is returned</p>
</td></tr>
<tr><td><code id="SEA_+3A_alpha">alpha</code></td>
<td>
<p>The type I error allowed for TDP bound. The default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame is returned including a list of pathways with corresponding TDP bound estimate,
and if specified, TDP point estimate and adjusted p-values
</p>


<h3>Author(s)</h3>

<p>Mitra Ebrahimpoor
</p>
<p><a href="mailto:m.ebrahimpoor@lumc.nl">m.ebrahimpoor@lumc.nl</a>
</p>


<h3>References</h3>

<p>Mitra Ebrahimpoor, Pietro Spitali, Kristina Hettne, Roula Tsonaka, Jelle Goeman,
Simultaneous Enrichment Analysis of all Possible Gene-sets: Unifying Self-Contained
and Competitive Methods, Briefings in Bioinformatics, , bbz074, https://doi.org/10.1093/bib/bbz074
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setTest">setTest</a></code>, <code><a href="#topic+topSEA">topSEA</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
##Generate a vector of pvalues for a toy example
set.seed(159)

m&lt;- 100
pvalues &lt;- runif(m,0,1)^5
featureIDs &lt;- as.character(1:m)

# perform a self-contained test for all features
setTest(pvalues, featureIDs, testype = "selfcontained")

# create 3 random pathway of size 60, 20 and 45
randpathlist=list(A=as.character(c(sample(1:m, 60))),
             B=as.character(c(sample(1:m, 20))),
             C=as.character(c(sample(1:m, 45))))


# get the seachart for the whole pathlist
S1&lt;-SEA(pvalues, featureIDs, pathlist=randpathlist)
S1

# get the seachart for only first two pathways of the randpathlist
S2&lt;-SEA(pvalues, featureIDs, pathlist=randpathlist, select=1:2)
S2

#sort the list by competitve p-value and select top 2
topSEA(S2, by=Comp.adjP, descending = FALSE, n=2)

#make an enrichment plot based on TDP.estimated of te pathways
#here pathways with more than 20% active signials will be identified
plotSEA(S1,threshold = 0.20,n=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='setTDP'>setTDP</h2><span id='topic+setTDP'></span>

<h3>Description</h3>

<p>Estimates the TDP of the specified set of features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setTDP(pvalue, featureIDs, data, set, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setTDP_+3A_pvalue">pvalue</code></td>
<td>
<p>The vector of p-values. It can be the name of the covariate representing the Vector of
raw p-values in the <code>data</code> or a single vector but in the latter case it should match the
<code>featureIDs</code> vector</p>
</td></tr>
<tr><td><code id="setTDP_+3A_featureids">featureIDs</code></td>
<td>
<p>The vector of feature IDs. It can be the name of the covariate representing the IDs in the
<code>data</code> or a single vector but in the latter case it should match the <code>pvalue</code> vector</p>
</td></tr>
<tr><td><code id="setTDP_+3A_data">data</code></td>
<td>
<p>Optional data frame or matrix containing the variables in <code>pvalue</code> and <code>featureIDs</code></p>
</td></tr>
<tr><td><code id="setTDP_+3A_set">set</code></td>
<td>
<p>The selection of features defining the feature-set based on the the <code>featureIDs</code>.
If missing, the set of all features is evaluated</p>
</td></tr>
<tr><td><code id="setTDP_+3A_alpha">alpha</code></td>
<td>
<p>The type I error allowed. The default is 0.05. NOTE: this shouls be consistent across the study</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named vector including the lower bound and point estimate for the true discovery proportion (TDP)
of the specified test for the feature-set is returned.
</p>


<h3>Author(s)</h3>

<p>Mitra Ebrahimpoor
</p>
<p><a href="mailto:m.ebrahimpoor@lumc.nl">m.ebrahimpoor@lumc.nl</a>
</p>


<h3>References</h3>

<p>Mitra Ebrahimpoor, Pietro Spitali, Kristina Hettne, Roula Tsonaka, Jelle Goeman,
Simultaneous Enrichment Analysis of all Possible Gene-sets: Unifying Self-Contained
and Competitive Methods, Briefings in Bioinformatics, , bbz074, https://doi.org/10.1093/bib/bbz074
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setTest">setTest</a></code>, <code><a href="#topic+SEA">SEA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
set.seed(159)
#generate random p-values with pseudo IDs
m&lt;- 100
pvalues &lt;- runif(m,0,1)^5
featureIDs &lt;- as.character(1:m)

# perform a self-contained test for all features
settest(pvalues, featureIDs, testype = "selfcontained")

# estimate the proportion of true discoveries among all m features
settdp(pvalues, featureIDs)

# create a random pathway of size 60
randset=as.character(c(sample(1:m, 60)))


# estimate the proportion of true discoveries in a random set of size 50
settdp(pvalues, featureIDs, set=randset)


## End(Not run)

</code></pre>

<hr>
<h2 id='setTest'>setTest</h2><span id='topic+setTest'></span>

<h3>Description</h3>

<p>calculates the adjusted p-value for the local hypothesis as defined by <code>testtype</code>
and <code>testvalue</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setTest(pvalue, featureIDs, data, set, testype, testvalue)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setTest_+3A_pvalue">pvalue</code></td>
<td>
<p>The vector of p-values. It can be the name of the covariate representing the Vector of
raw p-values in the <code>data</code> or a single vector but in the latter case it should match the
<code>featureIDs</code> vector</p>
</td></tr>
<tr><td><code id="setTest_+3A_featureids">featureIDs</code></td>
<td>
<p>The vector of feature IDs. It can be the name of the covariate representing the IDs in the
<code>data</code> or a single vector but in the latter case it should match the <code>pvalue</code> vector</p>
</td></tr>
<tr><td><code id="setTest_+3A_data">data</code></td>
<td>
<p>Optional data frame or matrix containing the variables in <code>pvalue</code> and <code>featureIDs</code></p>
</td></tr>
<tr><td><code id="setTest_+3A_set">set</code></td>
<td>
<p>The selection of features defining the feature-set based on the the <code>featureIDs</code>.
If missing, the set of all features is selected</p>
</td></tr>
<tr><td><code id="setTest_+3A_testype">testype</code></td>
<td>
<p>Character, type of the test: &quot;selfcontained&quot; or &quot;competitive&quot;. Choosing the self-contained
option will automatically set the threshold to zero and the <code>testvalue</code> is ignored. Choosing the
competitive option without a <code>testvalue</code> will set the threshold to the overall estimated proportion
of true hypotheses</p>
</td></tr>
<tr><td><code id="setTest_+3A_testvalue">testvalue</code></td>
<td>
<p>Optional value to test against. Setting this value to c along with
<code>testype=="competitive"</code> will lead to testing the null hypothesis against a threshold c.
Note: this value needs to be a proportion</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted p-value of the specified test for the feature-set is returned.
</p>


<h3>Author(s)</h3>

<p>Mitra Ebrahimpoor
</p>
<p><a href="mailto:m.ebrahimpoor@lumc.nl">m.ebrahimpoor@lumc.nl</a>
</p>


<h3>References</h3>

<p>Mitra Ebrahimpoor, Pietro Spitali, Kristina Hettne, Roula Tsonaka, Jelle Goeman,
Simultaneous Enrichment Analysis of all Possible Gene-sets: Unifying Self-Contained
and Competitive Methods, Briefings in Bioinformatics, , bbz074, https://doi.org/10.1093/bib/bbz074
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setTDP">setTDP</a></code> <code><a href="#topic+SEA">SEA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
#Generate a vector of pvalues
set.seed(159)

m&lt;- 100
pvalues &lt;- runif(m,0,1)^5
featureIDs &lt;- as.character(1:m)

# perform a self-contained test for all features
settest(pvalues, featureIDs, testype = "selfcontained")

# create a random pathway of size 60
randset=as.character(c(sample(1:m, 60)))

# perform a competitive test for the random pathway
settest(pvalues, featureIDs, set=randset, testype = "competitive")

# perform a unified null hypothesis test against 0.2 for a set of size 50
settest(pvalues, featureIDs, set=randset, testype = "competitive", testvalue = 0.2 )


## End(Not run)
</code></pre>

<hr>
<h2 id='topSEA'>topSEA</h2><span id='topic+topSEA'></span>

<h3>Description</h3>

<p>returns a permutation  of SEA-chart which rearranges
the feature-sets according to the selected argument into ascending or
descending order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topSEA(object, by, thresh = NULL, descending = TRUE, n = 20, cover)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topSEA_+3A_object">object</code></td>
<td>
<p>A SEA-chart object which is the output of <code>SEA</code> function</p>
</td></tr>
<tr><td><code id="topSEA_+3A_by">by</code></td>
<td>
<p>Variable name by which the ordering should happen. It should be a column of SEA-chart.
The default is TDP_bound.</p>
</td></tr>
<tr><td><code id="topSEA_+3A_thresh">thresh</code></td>
<td>
<p>A real number between 0 and 1. If specified the values of the variable defined in <code>by</code>
will be thresholded accordingly.</p>
</td></tr>
<tr><td><code id="topSEA_+3A_descending">descending</code></td>
<td>
<p>Logical. If <code>TRUE</code> The output chart is organized in a descending order</p>
</td></tr>
<tr><td><code id="topSEA_+3A_n">n</code></td>
<td>
<p>Integer. Number of raws of the output chart</p>
</td></tr>
<tr><td><code id="topSEA_+3A_cover">cover</code></td>
<td>
<p>An optional threshold for coverage, whcih muct be a real number between 0 and 1.
If specified, feature-sets with a coverage lower than or equal to this value are removed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a subset of SEA_chart sorted according to the arguments
</p>


<h3>Author(s)</h3>

<p>Mitra Ebrahimpoor
</p>
<p><a href="mailto:m.ebrahimpoor@lumc.nl">m.ebrahimpoor@lumc.nl</a>
</p>


<h3>References</h3>

<p>Mitra Ebrahimpoor, Pietro Spitali, Kristina Hettne, Roula Tsonaka, Jelle Goeman,
Simultaneous Enrichment Analysis of all Possible Gene-sets: Unifying Self-Contained
and Competitive Methods, Briefings in Bioinformatics,bbz074
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SEA">SEA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#See the examples for \code{\link{SEA}}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
