<!DOCTYPE html><html><head><title>Help for package RTCC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RTCC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#group_information'><p>Genomic data linked to saline lagoons.</p></a></li>
<li><a href='#metadata'><p>Salinity values of saline lagoons.</p></a></li>
<li><a href='#RTCC'><p>RTCC: Detecting trait clustering in environmental gradients</p>
with the Randomized Trait Community Clustering method</a></li>
<li><a href='#rtcc1'><p>Trait selection</p></a></li>
<li><a href='#rtcc2'><p>Clustering signal along an environmental gradient</p></a></li>
<li><a href='#rtcc3'><p>Clustering signal significance.</p></a></li>
<li><a href='#table_presence_absence'><p>Genome presence-absence data of 136 saline lagoons.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Detecting Trait Clustering in Environmental Gradients</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Mateu Menendez-Serra, Vicente J. Ontiveros, Emilio O. Casamayor, David Alonso</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mateu Menendez-Serra &lt;mateu.menendez@ceab.csic.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The Randomized Trait Community Clustering method (Triado-Margarit et al., 2019,
    &lt;<a href="https://doi.org/10.1038%2Fs41396-019-0454-4">doi:10.1038/s41396-019-0454-4</a>&gt;) is a statistical approach which allows to determine whether
    if an observed trait clustering pattern is related to an increasing environmental constrain.
    The method 1) determines whether exists or not a trait clustering on the sampled communities
    and 2) assess if the observed clustering signal is related or not to an increasing environmental
    constrain along an environmental gradient. Also, when the effect of the environmental gradient
    is not linear, allows to determine consistent thresholds on the community assembly based on trait-values.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>matrixStats, vegan, Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>testthat, Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-06-12 17:17:37 UTC; macbookair</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-06-12 17:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='group_information'>Genomic data linked to saline lagoons.</h2><span id='topic+group_information'></span>

<h3>Description</h3>

<p>A dataset containing genomic data of 544 genomes that matched 16s rRNA data from saline lagoons of the Monegros desert area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_information
</code></pre>


<h3>Format</h3>

<p>A data frame with 544 rows and 14 variables:
</p>

<dl>
<dt>genome</dt><dd><p>Genome IMG code</p>
</dd>
<dt>Genome_Size</dt><dd><p>Genome size</p>
</dd>
<dt>GC_perc</dt><dd><p>GC percentage</p>
</dd>
<dt>Coding_base_perc</dt><dd><p>Conding base percentage</p>
</dd>
<dt>CDS_perc</dt><dd><p>CDS percentage</p>
</dd>
<dt>RNA_perc</dt><dd><p>RNA percentage</p>
</dd>
<dt>rRNA_count</dt><dd><p>rRNA count</p>
</dd>
<dt>Transporter_perc</dt><dd><p>Transporter proteins percentage</p>
</dd>
<dt>Signal_peptide_perc</dt><dd><p>Signal peptide percentage</p>
</dd>
<dt>Transmembrane_perc</dt><dd><p>Transmembrane proteins percentage</p>
</dd>
<dt>Gene_Count</dt><dd><p>Gene count</p>
</dd>
<dt>min_env</dt><dd><p>Minimum environmental value where the organism has been observed</p>
</dd>
<dt>max_env</dt><dd><p>Minimum environmental value where the organism has been observed</p>
</dd>
<dt>rel_abundance</dt><dd><p>Relative abundance of the organism on the metacommunity</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Triadó-Margarit, X., Capitán, J.A., Menéndez-Serra, M. et al. A Randomized Trait Community Clustering approach to unveil consistent environmental thresholds in community assembly. ISME J 13, 2681–2689 (2019).
</p>

<hr>
<h2 id='metadata'>Salinity values of saline lagoons.</h2><span id='topic+metadata'></span>

<h3>Description</h3>

<p>A dataset containing salinity values of 136 lagoons on the Monegros desert area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata
</code></pre>


<h3>Format</h3>

<p>A data frame with 136 rows and 2 variables:
</p>

<dl>
<dt>sample_ID</dt><dd><p>Sample internal code</p>
</dd>
<dt>salinity</dt><dd><p>Sample salinity value</p>
</dd>
</dl>



<h3>Source</h3>

<p>Triadó-Margarit, X., Capitán, J.A., Menéndez-Serra, M. et al. A Randomized Trait Community Clustering approach to unveil consistent environmental thresholds in community assembly. ISME J 13, 2681–2689.2019.
</p>

<hr>
<h2 id='RTCC'>RTCC: Detecting trait clustering in environmental gradients
with the Randomized Trait Community Clustering method</h2><span id='topic+RTCC'></span>

<h3>Description</h3>

<p>A set of functions which allows to determine if the observed traits
present clustering/overdispersion patterns on the observed samples,
and if so, to stablish if the observed pattern is linked to the effect
of an environmental gradient.
</p>


<h3>Details</h3>

<p>The study of phenotypic similarities and differences within species along
environmental gradients might be used as a powerful tool complementing
taxon-based approaches when assesing the contribution of stochastic and
deterministic processes in community assembly. For this, this package allows an
easy implementation of a method for detecting clustering/overdispersion patterns
along an environmental gradient (Triado-Margarit et al., 2019). A first function
assesses if the observed traits exhibit a clustering/overdispersion pattern on the
tested samples. If positive, two subsequent functions determine whether the
observed pattern is linked to the effect of an environmental varible and its
statistical significance.
</p>


<h3>Data entry</h3>

<p>The data consists on presence-absence observations
along a measured environmental gradient and trait quantitative
information of the observed organisms.
</p>


<h3>References</h3>

<p>Triado-Margarit, X., Capitan, J.A., Menendez-Serra, M. et al. (2019)
A Randomized Trait Community Clustering approach to unveil consistent environmental
thresholds in community assembly. <em>ISME J</em> <b>13</b>, 2681–2689 .
<a href="https://doi.org/10.1038/s41396-019-0454-4">https://doi.org/10.1038/s41396-019-0454-4</a>
</p>

<hr>
<h2 id='rtcc1'>Trait selection</h2><span id='topic+rtcc1'></span>

<h3>Description</h3>

<p>This function determines whether the selected traits exhibit or not a clustering/overdispersion
signal on the tested samples. For each trait, compares the observed Mean Pairwise Distance (MPD)
of each sample against a distribution of synthetic commmunities MPDs obtained by a randomization
test. Each synthetic community is build maintaining the original sample richness and randomly
selecting organisms form the global pool.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rtcc1(table1, table2, table3, traits_columns, repetitions)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rtcc1_+3A_table1">table1</code></td>
<td>
<p>A data frame containing organisms names on the first column and its trait values on the
consecutive ones. It also has to contain two columns with the maximum and the minimum values of the tested
environmental variable where the organisms have been observed.</p>
</td></tr>
<tr><td><code id="rtcc1_+3A_table2">table2</code></td>
<td>
<p>A presence-absence observations table with the organisms names on the first column and the
sample names as consecutive colnames.</p>
</td></tr>
<tr><td><code id="rtcc1_+3A_table3">table3</code></td>
<td>
<p>A dataframe containing sample names on the first column and environmental parameters on the
consecutive ones.</p>
</td></tr>
<tr><td><code id="rtcc1_+3A_traits_columns">traits_columns</code></td>
<td>
<p>Table 1 column numbers where different trait values appear.</p>
</td></tr>
<tr><td><code id="rtcc1_+3A_repetitions">repetitions</code></td>
<td>
<p>Number of simulated synthetic communities distributions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a dataframe with trait names as colnames and the p-value distribution of the different
traits.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(group_information)
data(table_presence_absence)
data(metadata)
rtcc1(group_information, table_presence_absence, metadata, 2:11, 100)



</code></pre>

<hr>
<h2 id='rtcc2'>Clustering signal along an environmental gradient</h2><span id='topic+rtcc2'></span>

<h3>Description</h3>

<p>For a given trait, this function determines whether the observed trait clustering/overdispersion
on the metacommunity is linked to an environmental gradient. For this, it sequentially remove samples
in decreasing order of the environmental variable and computes at each step the remaining
metacommunity h-index. This index is based on the percentage of samples on a metacommunity
presenting significant trait clustering/overdispersion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rtcc2(
  table1,
  table2,
  table3,
  species_abundances,
  trait_col_number,
  min_env_col,
  max_env_col,
  env_var_col,
  h_iteration,
  repetitions,
  model
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rtcc2_+3A_table1">table1</code></td>
<td>
<p>A data frame containing organisms names on the first column and its trait values on the
consecutive ones. It also has to contain two columns with the maximum and the minimum values of the tested
environmental variable where the organisms have been observed.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_table2">table2</code></td>
<td>
<p>A presence-absence observations table with the organisms names on the first column and the
sample names as consecutive colnames.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_table3">table3</code></td>
<td>
<p>A dataframe containing sample names on the first column and environmental parameters on the
consecutive ones.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_species_abundances">species_abundances</code></td>
<td>
<p>A vector containing the relative abundance of the organisms on the whole data set on
the same order as appear on Table 1.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_trait_col_number">trait_col_number</code></td>
<td>
<p>Table 1 column number of the tested trait.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_min_env_col">min_env_col</code></td>
<td>
<p>Table 1 column number indicating the minimum value of the environmental variable were each
organism has been observed.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_max_env_col">max_env_col</code></td>
<td>
<p>Table 1 column number indicating the maximum value of the environmental variable were each
organism has been observed.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_env_var_col">env_var_col</code></td>
<td>
<p>Table 2 column number indicating the tested environmental variable.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_h_iteration">h_iteration</code></td>
<td>
<p>Number of h-index calculations for computing a confidence interval.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_repetitions">repetitions</code></td>
<td>
<p>Number of simulated synthetic communities distributions.</p>
</td></tr>
<tr><td><code id="rtcc2_+3A_model">model</code></td>
<td>
<p>Model selection. All models build synthetic communities based on the organisms richness of the
observed communities.
</p>
<p>- Model 1: organism are selected randomly from the global pool.
- Model 2: organism are selected randomly with a probability based on its relative abundance on the global pool.
- Model 3: organism are selected randomly, but only those whose environmental range includes the value of
the simulated community are elegible.
- Model 4: organism are selected randomly, but only those whose environmental range includes the value of
the simulated community are elegible and the selection probability is based on its relative abundance on the global pool.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a dataframe with the maximum of the environmental variable on the remaining
metacommunity after the sequential removal, h-index calculation for each environmental value, and its confidence standard deviation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(group_information)
data(table_presence_absence)
data(metadata)
rtcc2(group_information, table_presence_absence, metadata, group_information$sums,
9, 12, 13, 2, 100, 100, model = 1)


</code></pre>

<hr>
<h2 id='rtcc3'>Clustering signal significance.</h2><span id='topic+rtcc3'></span>

<h3>Description</h3>

<p>For a given trait and environmental variable, this function creates a null model of the
clustering/overdispersion pattern in order to test if the observed pattern statistically
differs from the expected by random. For this, it sequentially remove random samples from
the metacommunity and computes at each step the remaining metacommunity h-index. This index
is based on the percentage of samples on a metacoomunity presenting significant trait
clustering/overdispersion. After h iterations, computes a 95
obtained h-index for each point of the environmental gradient.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rtcc3(
  table1,
  table2,
  table3,
  species_abundances,
  trait_col_number,
  min_env_col,
  max_env_col,
  env_var_col,
  h_iteration,
  repetitions,
  model
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rtcc3_+3A_table1">table1</code></td>
<td>
<p>A data frame containing organisms names on the first column and its trait values on the
consecutive ones. It also has to contain two columns with the maximum and the minimum values of the tested
environmental variable where the organisms have been observed.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_table2">table2</code></td>
<td>
<p>A presence-absence observations table with the organisms names on the first column and the
sample names as consecutive colnames.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_table3">table3</code></td>
<td>
<p>A dataframe containing sample names on the first column and environmental parameters on the
consecutive ones.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_species_abundances">species_abundances</code></td>
<td>
<p>A vector containing the relative abundance of the organisms on the whole data set on
the same order as appear on Table 1.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_trait_col_number">trait_col_number</code></td>
<td>
<p>Table 1 column number of the tested trait.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_min_env_col">min_env_col</code></td>
<td>
<p>Table 1 column number indicating the minimum value of the environmental variable were each
organism has been observed.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_max_env_col">max_env_col</code></td>
<td>
<p>Table 1 column number indicating the maximum value of the environmental variable were each
organism has been observed.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_env_var_col">env_var_col</code></td>
<td>
<p>Table 2 column number indicating the tested environmental variable.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_h_iteration">h_iteration</code></td>
<td>
<p>Number of h-index calculations for computing a confidence interval.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_repetitions">repetitions</code></td>
<td>
<p>Number of simulated synthetic communities distributions.</p>
</td></tr>
<tr><td><code id="rtcc3_+3A_model">model</code></td>
<td>
<p>Model selection. All models build synthetic communities based on the organisms richness of the
observed communities.
</p>
<p>- Model 1: organism are selected randomly from the global pool.
- Model 2: organism are selected randomly with a probability based on its relative abundance on the global pool.
- Model 3: organism are selected randomly, but only those whose environmental range includes the value of
the simulated community are elegible.
- Model 4: organism are selected randomly, but only those whose environmental range includes the value of
the simulated community are elegible and the selection probability is based on its relative abundance on the global pool.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a dataframe with the maximum value of environmental variable corresponding to the same
number of samples on the ordered remova, h-index calculation for each environmental value, and the percentiles 0.025,
0.5 and 0.975 of the obtained distribution for each point (mean value and 95
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(group_information)
data(table_presence_absence)
data(metadata)
rtcc3(group_information, table_presence_absence, metadata, group_information$sums,
9, 12, 13, 2, 50, 20, model = 1)


</code></pre>

<hr>
<h2 id='table_presence_absence'>Genome presence-absence data of 136 saline lagoons.</h2><span id='topic+table_presence_absence'></span>

<h3>Description</h3>

<p>A dataset containing presence-absence data of 544 genomes on 136 saline lagoons of the Monegros desert area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table_presence_absence
</code></pre>


<h3>Format</h3>

<p>A data frame with 544 rows and 137 variables:
</p>

<dl>
<dt>genome</dt><dd><p>Genome IMG code</p>
</dd>
<dt>MON_10</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_100</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_101</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_103</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_104</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_106</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_107</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_108</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_109</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_11</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_110</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_111</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_112</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_113</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_114</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_116</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_117</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_118</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_119</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_12</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_120</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_122</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_123</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_124</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_125</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_126</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_127</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_128</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_129</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_13</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_130</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_131</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_133</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_134</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_135</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_136</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_137</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_138</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_139</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_14</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_140</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_141</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_142</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_144</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_145</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_146</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_147</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_148</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_15</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_17</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_18</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_19</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_2</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_20</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_21</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_22</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_23</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_24</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_25</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_26</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_27</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_28</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_29</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_30</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_31</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_32</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_33</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_34</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_35</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_36</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_37</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_38</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_39</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_4</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_40</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_41</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_42</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_43</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_44</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_45</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_46</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_47</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_48</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_49</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_5</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_50</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_51</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_52</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_53</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_54</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_55</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_56</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_57</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_58</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_59</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_60</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_61</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_62</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_63</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_64</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_65</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_66</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_67</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_68</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_69</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_7</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_70</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_71</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_72</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_73</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_74</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_75</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_76</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_77</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_78</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_79</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_8</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_80</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_81</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_82</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_83</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_84</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_85</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_86</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_88</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_9</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_90</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_91</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_92</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_93</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_94</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_95</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_96</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_97</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_98</dt><dd><p>Sample presence-absence observations</p>
</dd>
<dt>MON_99</dt><dd><p>Sample presence-absence observations</p>
</dd>
</dl>
<p>...

</p>


<h3>Source</h3>

<p>Triadó-Margarit, X., Capitán, J.A., Menéndez-Serra, M. et al. A Randomized Trait Community Clustering approach to unveil consistent environmental thresholds in community assembly. ISME J 13, 2681–2689 (2019).
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
