<!DOCTYPE html><html lang="en"><head><title>Help for package RTIGER</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RTIGER}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ATseqlengths'><p>The autosome chromosome lengths for Arabidopsis Thaliana.</p></a></li>
<li><a href='#calcCOnumber'><p>Obtain number of Cross-Over events per sample and chromosome.</p></a></li>
<li><a href='#dev'><p>Function to developers. It runs one EM step</p></a></li>
<li><a href='#fit'><p>Call Julia code to fit the values</p></a></li>
<li><a href='#generateObject'><p>Load data</p></a></li>
<li><a href='#myDat'><p>A fitted example using three own samples of Arabidopsis. More information in publication:</p></a></li>
<li><a href='#optimize_R'><p>Find the otimum R value for a given data set</p></a></li>
<li><a href='#plotCOs'><p>Obtain number of Cross-Over events per sample and chromosome.</p></a></li>
<li><a href='#RTIGER'><p>Load, Fit, and plot</p></a></li>
<li><a href='#RTIGER-class'><p>This class is a generic container for RTIGER analysis</p></a></li>
<li><a href='#setupJulia'><p>Installs the needed packages in JULIA to run the EM algorithm for rHMM.</p></a></li>
<li><a href='#sourceJulia'><p>Function needed before using RTIGER() function. It loads the scripts in Julia that fit the rHMM.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>HMM-Based Model for Genotyping and Cross-Over Identification</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Our method integrates information from all sequenced samples, thus avoiding loss of alleles due to low coverage. Moreover, it increases the statistical power to uncover sequencing or alignment errors &lt;<a href="https://doi.org/10.1093%2Fplphys%2Fkiad191">doi:10.1093/plphys/kiad191</a>&gt;. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6), GenomicRanges, GenomeInfoDb</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>gzip</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, e1071, extraDistr, reshape2, ggplot2, TailRank,
JuliaCall, IRanges, qpdf, grDevices, graphics, stats, utils</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, markdown, Gviz, rtracklayer</td>
</tr>
<tr>
<td>biocViews:</td>
<td>GenomeAnnotation, HiddenMarkovModel, Sequencing</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-29 08:50:35 UTC; campos</td>
</tr>
<tr>
<td>Author:</td>
<td>Rafael Campos-Martin
    <a href="https://orcid.org/0000-0002-1395-8571"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre],
  Sophia Schmickler [aut],
  Manish Goel [ctb],
  Korbinian Schneeberger [aut],
  Achim Tresch [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rafael Campos-Martin &lt;rfael.mpi@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-29 09:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ATseqlengths'>The autosome chromosome lengths for Arabidopsis Thaliana.</h2><span id='topic+ATseqlengths'></span>

<h3>Description</h3>

<p>The autosome chromosome lengths for Arabidopsis Thaliana.
</p>


<h3>Author(s)</h3>

<p>Rafael Campos-Martin
</p>

<hr>
<h2 id='calcCOnumber'>Obtain number of Cross-Over events per sample and chromosome.</h2><span id='topic+calcCOnumber'></span>

<h3>Description</h3>

<p>Obtain number of Cross-Over events per sample and chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcCOnumber(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calcCOnumber_+3A_object">object</code></td>
<td>
<p>a RViterbi object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix m x n. M number of samples and N chromosomes.
</p>
<p>#' @return a matrix with n chromosomes and m samples (n x m) and the number of CO events.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("fittedExample")
co.num = calcCOnumber(myDat)

</code></pre>

<hr>
<h2 id='dev'>Function to developers. It runs one EM step</h2><span id='topic+dev'></span>

<h3>Description</h3>

<p>Function to developers. It runs one EM step
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dev(psi, rigidity = NULL, nstates = 3, transition = NULL, start = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dev_+3A_psi">psi</code></td>
<td>
<p>list of psi probabilities.</p>
</td></tr>
<tr><td><code id="dev_+3A_rigidity">rigidity</code></td>
<td>
<p>Rigidity value.</p>
</td></tr>
<tr><td><code id="dev_+3A_nstates">nstates</code></td>
<td>
<p>Number of states.</p>
</td></tr>
<tr><td><code id="dev_+3A_transition">transition</code></td>
<td>
<p>transition matrix</p>
</td></tr>
<tr><td><code id="dev_+3A_start">start</code></td>
<td>
<p>initial probabilities</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with updates probabilites
</p>

<hr>
<h2 id='fit'>Call Julia code to fit the values</h2><span id='topic+fit'></span>

<h3>Description</h3>

<p>Call Julia code to fit the values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit(rtigerobj, max.iter , eps,
trace, all = TRUE, random = FALSE,
specific = FALSE, nsamples = 20,
post.processing = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_+3A_rtigerobj">rtigerobj</code></td>
<td>
<p>an RTIGER object.</p>
</td></tr>
<tr><td><code id="fit_+3A_max.iter">max.iter</code></td>
<td>
<p>maximum number of iterations to acomplish by the EM.</p>
</td></tr>
<tr><td><code id="fit_+3A_eps">eps</code></td>
<td>
<p>differnece threshold to halt the EM.</p>
</td></tr>
<tr><td><code id="fit_+3A_trace">trace</code></td>
<td>
<p>logical value whether to trace the changes in the parameters along the iterations.</p>
</td></tr>
<tr><td><code id="fit_+3A_all">all</code></td>
<td>
<p>logical value whether to use all data to fit the model.</p>
</td></tr>
<tr><td><code id="fit_+3A_random">random</code></td>
<td>
<p>if all FALSE use random samples.</p>
</td></tr>
<tr><td><code id="fit_+3A_specific">specific</code></td>
<td>
<p>if all FALSE use specific samples.</p>
</td></tr>
<tr><td><code id="fit_+3A_nsamples">nsamples</code></td>
<td>
<p>if random TRUE, how many samples to use.</p>
</td></tr>
<tr><td><code id="fit_+3A_post.processing">post.processing</code></td>
<td>
<p>logical value, whether to run post.processing process.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>RTIGER object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data("fittedExample")
sourceJulia()
myfit = fit(myDat, max.iter = 2, eps=0.01,
            trace = TRUE, all = TRUE,
            random = FALSE, specific = FALSE,
            nsamples = 20, post.processing = TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='generateObject'>Load data</h2><span id='topic+generateObject'></span>

<h3>Description</h3>

<p>Load data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateObject(experimentDesign = NULL,nstates = 3, rigidity=NULL,
seqlengths = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generateObject_+3A_experimentdesign">experimentDesign</code></td>
<td>
<p>a data Frame that contains minimum a column with the files direction (name of the column files) and another with a shorter name to be used inside the function.</p>
</td></tr>
<tr><td><code id="generateObject_+3A_nstates">nstates</code></td>
<td>
<p>the number of states to be fitted in the model. A standard setting would use 3 states (Homozygous1, Heterozygous, and Homozygous2).</p>
</td></tr>
<tr><td><code id="generateObject_+3A_rigidity">rigidity</code></td>
<td>
<p>an integer number specifying the rigidity parameter to be used.</p>
</td></tr>
<tr><td><code id="generateObject_+3A_seqlengths">seqlengths</code></td>
<td>
<p>a named vector with the chromosome lenghts of the organism that the user is working with.</p>
</td></tr>
<tr><td><code id="generateObject_+3A_verbose">verbose</code></td>
<td>
<p>logical value. Whether to print info messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>RTIGER object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("ATseqlengths")
path = system.file("extdata",  package = "RTIGER")
files = list.files(path, full.names = TRUE)
nam = sapply(list.files(path ), function(x) unlist(strsplit(x, split = "[.]"))[1])
expDesign = data.frame(files = files, name = nam)
names(ATseqlengths) = paste0("Chr", 1:5)
myres = generateObject(experimentDesign = expDesign,
              seqlengths = ATseqlengths,
              rigidity = 10
)


</code></pre>

<hr>
<h2 id='myDat'>A fitted example using three own samples of Arabidopsis. More information in publication:</h2><span id='topic+myDat'></span>

<h3>Description</h3>

<p>A fitted example using three own samples of Arabidopsis. More information in publication:
</p>


<h3>Author(s)</h3>

<p>Rafael Campos-Martin
</p>

<hr>
<h2 id='optimize_R'>Find the otimum R value for a given data set</h2><span id='topic+optimize_R'></span>

<h3>Description</h3>

<p>Find the otimum R value for a given data set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimize_R(object,
max_rigidity = 2^9, average_coverage = NULL, crossovers_per_megabase = NULL,
save_it = FALSE, savedir = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="optimize_R_+3A_object">object</code></td>
<td>
<p>an RTIGER object</p>
</td></tr>
<tr><td><code id="optimize_R_+3A_max_rigidity">max_rigidity</code></td>
<td>
<p>R values will be explored up the value given in this parameter. Default = 2^9</p>
</td></tr>
<tr><td><code id="optimize_R_+3A_average_coverage">average_coverage</code></td>
<td>
<p>For conservative results set it to the lowest average coverage of a sample in your experiment, or evne to the lowest average coverage in a (sufficiently large) region in one of your samples. The lower the value, the more conservative (higher) our estimates of the false positive segments rates. If it is not provided it will be computed as the average of all data points.</p>
</td></tr>
<tr><td><code id="optimize_R_+3A_crossovers_per_megabase">crossovers_per_megabase</code></td>
<td>
<p>For conservative results set it to the highest ratio of a sample in your experiment. The higher the value, the more conservative (higher) our estimates of the false positive segments rates. If it is not provided it will be computed as the average of all samples.</p>
</td></tr>
<tr><td><code id="optimize_R_+3A_save_it">save_it</code></td>
<td>
<p>logical values if the results should be saved. Plots might be complicated to interpret. We suggest to read the manuscript to understand them (https://doi.org/10.1093/plphys/kiad191)</p>
</td></tr>
<tr><td><code id="optimize_R_+3A_savedir">savedir</code></td>
<td>
<p>if results are saved, in which directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A value with the optimum rigidity for the data set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("fittedExample")
bestR = optimize_R(myDat)

</code></pre>

<hr>
<h2 id='plotCOs'>Obtain number of Cross-Over events per sample and chromosome.</h2><span id='topic+plotCOs'></span>

<h3>Description</h3>

<p>Obtain number of Cross-Over events per sample and chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCOs(object, file = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotCOs_+3A_object">object</code></td>
<td>
<p>a RViterbi object.</p>
</td></tr>
<tr><td><code id="plotCOs_+3A_file">file</code></td>
<td>
<p>file where to save the plot for CO numbers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("fittedExample")
co.num = calcCOnumber(myDat)

</code></pre>

<hr>
<h2 id='RTIGER'>Load, Fit, and plot</h2><span id='topic+RTIGER'></span>

<h3>Description</h3>

<p>Load, Fit, and plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RTIGER(expDesign, rigidity=NULL, outputdir=NULL, nstates = 3,
seqlengths = NULL, eps=0.01, max.iter=50, autotune = FALSE,
max_rigidity = 2^9, average_coverage = NULL,
crossovers_per_megabase = NULL, trace = FALSE,
tiles = 4e5, all = TRUE, random = FALSE, specific = FALSE,
nsamples = 20, post.processing = TRUE, save.results = TRUE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RTIGER_+3A_expdesign">expDesign</code></td>
<td>
<p>a data Frame that contains minimum a column with the files direction (name of the column files) and another with a shorter name to be used inside the function.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_rigidity">rigidity</code></td>
<td>
<p>an integer number specifying the rigidity parameter to be used.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_outputdir">outputdir</code></td>
<td>
<p>a character string that specifies the directory in which to save the results form the function.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_nstates">nstates</code></td>
<td>
<p>the number of states to be fitted in the model. A standard setting would use 3 states (Homozygous1, Heterozygous, and Homozygous2).</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_seqlengths">seqlengths</code></td>
<td>
<p>a named vector with the chromosome lenghts of the organism that the user is working with.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_eps">eps</code></td>
<td>
<p>the threshold of the difference between the parameters value between the previous and actuay iteration to stope de EM algorithm.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_max.iter">max.iter</code></td>
<td>
<p>maximum number of iterations of the EM algorithm before to stop in case that eps has not been achieved.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_autotune">autotune</code></td>
<td>
<p>Logical value if the R-value should be tuned by our algorithm. This will take longer as it needs a first training with the rigidity value provided by the user and then the optimization step is carried. Finally, a training using the optimum R will be performed and results for the optimum R will be returned.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_max_rigidity">max_rigidity</code></td>
<td>
<p>If autotune true, R values will be explored up the value given in this parameter. Default = 2^9</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_average_coverage">average_coverage</code></td>
<td>
<p>If autotune true, for conservative results set it to the lowest average coverage of a sample in your experiment, or evne to the lowest average coverage in a (sufficiently large) region in one of your samples. The lower the value, the more conservative (higher) our estimates of the false positive segments rates. If it is not provided it will be computed as the average of all data points.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_crossovers_per_megabase">crossovers_per_megabase</code></td>
<td>
<p>If autotune true,  for conservative results set it to the highest ratio of a sample in your experiment. The higher the value, the more conservative (higher) our estimates of the false positive segments rates. If it is not provided it will be computed as the average of all samples.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_trace">trace</code></td>
<td>
<p>logical value. Whether or not to keep track of the parameters for the HMM along the iterations. Deafault FALSE</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_tiles">tiles</code></td>
<td>
<p>length of the tiles by which the genome will be segmented in order to compute the ratio of COs in the complete dataset.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_all">all</code></td>
<td>
<p>logical value. Whether to use the complete data set to fit the rHMM. default TRUE.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_random">random</code></td>
<td>
<p>Logical value. Choose randomly a subset of the complete dataset to fit the rHMM. Default FALSE</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_specific">specific</code></td>
<td>
<p>Logical value to specify which samples to take.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_nsamples">nsamples</code></td>
<td>
<p>if random TRUE, how many samples should be taken randomly.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_post.processing">post.processing</code></td>
<td>
<p>Logical value. Whether to run an extra step that fine maps the segment borthers. Default TRUE</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_save.results">save.results</code></td>
<td>
<p>Logical value, whether to generate and save the plots and igv files.</p>
</td></tr>
<tr><td><code id="RTIGER_+3A_verbose">verbose</code></td>
<td>
<p>Logical, whether to print info to console.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix m x n. M number of samples and N chromosomes.
</p>
<p>RTIGER object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data("ATseqlengths")
sourceJulia()
path = system.file("extdata",  package = "RTIGER")
files = list.files(path, full.names = TRUE)
nam = sapply(list.files(path ), function(x) unlist(strsplit(x, split = "[.]"))[1])
expDesign = data.frame(files = files, name = nam)
names(ATseqlengths) = paste0("Chr", 1:5)
myres = RTIGER(expDesign = expDesign,
               outputdir = "/home/campos/Documents/outputjulia/",
               seqlengths = ATseqlengths,
               rigidity = 4,
               max.iter = 2,
               trace = FALSE,
               save.results = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='RTIGER-class'>This class is a generic container for RTIGER analysis</h2><span id='topic+RTIGER-class'></span><span id='topic+.RTIGER'></span>

<h3>Description</h3>

<p>This class is a generic container for RTIGER analysis
</p>


<h3>Slots</h3>


<dl>
<dt><code>matobs</code></dt><dd><p>Nested lists. the first level is a list of samples. For each sample there are 5 matrices that contains the allele counts for each position.</p>
</dd>
<dt><code>params</code></dt><dd><p>a list with the parameters after training.</p>
</dd>
<dt><code>info</code></dt><dd><p>List with phenotipic data of the samples.</p>
</dd>
<dt><code>Viterbi</code></dt><dd><p>List of chromosomes with the viterbi path per sample.</p>
</dd>
<dt><code>Probabilities</code></dt><dd><p>Computed probabilites for the EM algorithm.</p>
</dd>
<dt><code>num.iter</code></dt><dd><p>Number of iterations needed to stop the EM algorithm.</p>
</dd>
</dl>

<hr>
<h2 id='setupJulia'>Installs the needed packages in JULIA to run the EM algorithm for rHMM.</h2><span id='topic+setupJulia'></span>

<h3>Description</h3>

<p>Installs the needed packages in JULIA to run the EM algorithm for rHMM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setupJulia(JULIA_HOME = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setupJulia_+3A_julia_home">JULIA_HOME</code></td>
<td>
<p>the file folder which contains julia binary, if not set, JuliaCall will look at the global option JULIA_HOME, if the global option is not set, JuliaCall will then look at the environmental variable JULIA_HOME, if still not found, JuliaCall will try to use the julia in path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>empty
</p>

<hr>
<h2 id='sourceJulia'>Function needed before using RTIGER() function. It loads the scripts in Julia that fit the rHMM.</h2><span id='topic+sourceJulia'></span>

<h3>Description</h3>

<p>Function needed before using RTIGER() function. It loads the scripts in Julia that fit the rHMM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sourceJulia()
</code></pre>


<h3>Value</h3>

<p>empty
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
