<!DOCTYPE html><html lang="en"><head><title>Help for package rtpcr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rtpcr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data_1factor'><p>Sample data (one factor three levels)</p></a></li>
<li><a href='#data_2factor'><p>Sample data (two factor)</p></a></li>
<li><a href='#data_2factorBlock'><p>Sample data (two factor with blocking factor)</p></a></li>
<li><a href='#data_3factor'><p>Sample data (three factor)</p></a></li>
<li><a href='#data_efficiency'><p>Sample qPCR data: amplification efficiency</p></a></li>
<li><a href='#data_repeated_measure_1'><p>Repeated measure sample data</p></a></li>
<li><a href='#data_repeated_measure_2'><p>Repeated measure sample data</p></a></li>
<li><a href='#data_ttest'><p>Sample qPCR data from an experiment conducted under two different conditions</p></a></li>
<li><a href='#data_withTechRep'><p>Sample data (with technical replicates)</p></a></li>
<li><a href='#efficiency'><p>Slope, R2 and Efficiency (E) statistics and standard curves</p></a></li>
<li><a href='#Lee_etal2020qPCR'><p>Sample data (with technical replicates)</p></a></li>
<li><a href='#meanTech'><p>Calculating mean of technical replicates</p></a></li>
<li><a href='#multiplot'><p>Multiple plot function</p></a></li>
<li><a href='#oneFACTORplot'><p>Bar plot of the relative gene expression (<code class="reqn">\Delta C_T</code> method) from the <code>qpcrANOVARE</code> output of a single-factor experiment data</p></a></li>
<li><a href='#qpcrANOVAFC'><p>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis using ANOVA and ANCOVA</p></a></li>
<li><a href='#qpcrANOVARE'><p>Relative expression (<code class="reqn">\Delta C_T</code> method) analysis using ANOVA</p></a></li>
<li><a href='#qpcrMeans'><p>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis using a model</p></a></li>
<li><a href='#qpcrREPEATED'><p>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis of repeated measure qPCR data</p></a></li>
<li><a href='#qpcrTTEST'><p>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis of target genes using t-test</p></a></li>
<li><a href='#qpcrTTESTplot'><p>Bar plot of the average fold change (<code class="reqn">\Delta \Delta C_T</code> method) of target genes</p></a></li>
<li><a href='#Taylor_etal2019'><p>Sample qPCR data (one target and two reference genes under two different conditions)</p></a></li>
<li><a href='#threeFACTORplot'><p>Bar plot of the relative gene expression (<code class="reqn">\Delta C_T</code> method) from the <code>qpcrANOVARE</code> output of a a three-factorial experiment data</p></a></li>
<li><a href='#twoFACTORplot'><p>Bar plot of the relative gene expression (<code class="reqn">\Delta C_T</code> method) from the <code>qpcrANOVARE</code> output of a two-factorial experiment data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>qPCR Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Various methods are employed for statistical analysis and graphical presentation of real-time PCR (quantitative PCR  or qPCR) data. 'rtpcr' handles amplification efficiency calculation, statistical analysis and graphical representation of real-time PCR data based on up to two reference genes. By accounting for amplification efficiency values, 'rtpcr' was developed using a general calculation method described by Ganger et al. (2017) &lt;<a href="https://doi.org/10.1186%2Fs12859-017-1949-5">doi:10.1186/s12859-017-1949-5</a>&gt; and Taylor et al. (2019) &lt;<a href="https://doi.org/10.1016%2Fj.tibtech.2018.12.002">doi:10.1016/j.tibtech.2018.12.002</a>&gt;, covering both the Livak and Pfaffl methods. Based on the experimental conditions, the functions of the 'rtpcr' package use t-test (for experiments with a two-level factor), analysis of variance (ANOVA), analysis of covariance (ANCOVA) or analysis of repeated measure data to calculate the fold change (FC, Delta Delta Ct method) or relative expression (RE, Delta Ct method). The functions further provide standard errors and confidence intervals for means, apply statistical mean comparisons and present significance. To facilitate function application, different data sets were used as examples and the outputs were explained. ‘rtpcr’ package also provides bar plots using various controlling arguments. The 'rtpcr' package is user-friendly and easy to work with and provides an applicable resource for analyzing real-time PCR data.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mirzaghaderi/rtpcr">https://github.com/mirzaghaderi/rtpcr</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>multcomp, multcompView, ggplot2, lmerTest, purrr, reshape2,
tidyr, dplyr, grid, emmeans</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-13 18:18:57 UTC; GM</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Author:</td>
<td>Ghader Mirzaghaderi [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ghader Mirzaghaderi &lt;gh.mirzaghaderi@uok.ac.ir&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-13 19:10:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='data_1factor'>Sample data (one factor three levels)</h2><span id='topic+data_1factor'></span>

<h3>Description</h3>

<p>A sample dataset for demonstration purposes. Each line belongs to a separate individual (non-repeated measure experiment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_1factor
</code></pre>


<h3>Format</h3>

<p>A data frame with 9 observations and 6 variables:
</p>

<dl>
<dt>SA</dt><dd><p>First experimental factor</p>
</dd>
<dt>Rep</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>Not applicable
</p>

<hr>
<h2 id='data_2factor'>Sample data (two factor)</h2><span id='topic+data_2factor'></span>

<h3>Description</h3>

<p>A sample dataset for demonstration purposes. Each line belongs to a separate individual (non-repeated measure experiment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_2factor
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations and 7 variables:
</p>

<dl>
<dt>Genotype</dt><dd><p>First experimental factor</p>
</dd>
<dt>Drought</dt><dd><p>Second experimental factor</p>
</dd>
<dt>Rep</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>Not applicable
</p>

<hr>
<h2 id='data_2factorBlock'>Sample data (two factor with blocking factor)</h2><span id='topic+data_2factorBlock'></span>

<h3>Description</h3>

<p>A sample qPCR data set with blocking factor. Each line belongs to a separate individual (non-repeated measure experiment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_2factorBlock
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations and 8 variables:
</p>

<dl>
<dt>factor1</dt><dd><p>First experimental factor</p>
</dd>
<dt>factor2</dt><dd><p>Second experimental factor</p>
</dd>
<dt>block</dt><dd><p>Second experimental factor</p>
</dd>
<dt>Rep</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>Not applicable
</p>

<hr>
<h2 id='data_3factor'>Sample data (three factor)</h2><span id='topic+data_3factor'></span>

<h3>Description</h3>

<p>A sample dataset for demonstration purposes. Each line belongs to a separate individual (non-repeated measure experiment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_3factor
</code></pre>


<h3>Format</h3>

<p>A data frame with 36 observations and 8 variables:
</p>

<dl>
<dt>Type</dt><dd><p>First experimental factor</p>
</dd>
<dt>Conc</dt><dd><p>Second experimental factor</p>
</dd>
<dt>SA</dt><dd><p>Third experimental factor</p>
</dd>
<dt>Replicate</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>Not applicable
</p>

<hr>
<h2 id='data_efficiency'>Sample qPCR data: amplification efficiency</h2><span id='topic+data_efficiency'></span>

<h3>Description</h3>

<p>A sample qPCR dataset for demonstrating efficiency calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_efficiency
</code></pre>


<h3>Format</h3>

<p>A data frame with 21 observations and 4 variables:
</p>

<dl>
<dt>dilutions</dt><dd><p>Dilution factor</p>
</dd>
<dt>C2H2.26</dt><dd><p>Target gene 1</p>
</dd>
<dt>C2H2.01</dt><dd><p>Target gene 2</p>
</dd>
<dt>GAPDH</dt><dd><p>Reference gene</p>
</dd>
</dl>



<h3>Source</h3>

<p>Where the data comes from (if applicable)
</p>

<hr>
<h2 id='data_repeated_measure_1'>Repeated measure sample data</h2><span id='topic+data_repeated_measure_1'></span>

<h3>Description</h3>

<p>A repeated measure sample data in which 3 individuals have been analysed.  In the &quot;id&quot; column, a unique number is assigned to each individual, e.g. all the three number 1 indicate one individual.
samples are taken or measurements are scored over different time points (time column) from each individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_repeated_measure_1
</code></pre>


<h3>Format</h3>

<p>A data frame with 9 observations and 6 variables:
</p>

<dl>
<dt>id</dt><dd><p>experimental factor</p>
</dd>
<dt>time</dt><dd><p>time course levels</p>
</dd>
<dt>Eg</dt><dd><p>Amplification efficiency of target gene</p>
</dd>
<dt>Ctg</dt><dd><p>Ct of target gene</p>
</dd>
<dt>Eref</dt><dd><p>Amplification efficiency of reference gene</p>
</dd>
<dt>Ctref</dt><dd><p>Ct of reference gene</p>
</dd>
</dl>



<h3>Source</h3>

<p>NA
</p>

<hr>
<h2 id='data_repeated_measure_2'>Repeated measure sample data</h2><span id='topic+data_repeated_measure_2'></span>

<h3>Description</h3>

<p>A repeated measure sample data in which 6 individuals have been analysed.  In the &quot;id&quot; column, a unique number is assigned to each individual,  e.g. all the three number 1 indicate one individual.
samples are taken or measurements are scored over different time points (time column) from each individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_repeated_measure_2
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations and 7 variables:
</p>

<dl>
<dt>id</dt><dd><p>experimental factor</p>
</dd>
<dt>treatment</dt><dd><p>treatment</p>
</dd>
<dt>time</dt><dd><p>time course levels</p>
</dd>
<dt>Eg</dt><dd><p>Amplification efficiency of target gene</p>
</dd>
<dt>Ctg</dt><dd><p>Ct of target gene</p>
</dd>
<dt>Eref</dt><dd><p>Amplification efficiency of reference gene</p>
</dd>
<dt>Ctref</dt><dd><p>Ct of reference gene</p>
</dd>
</dl>



<h3>Source</h3>

<p>NA
</p>

<hr>
<h2 id='data_ttest'>Sample qPCR data from an experiment conducted under two different conditions</h2><span id='topic+data_ttest'></span>

<h3>Description</h3>

<p>Sample qPCR data from an experiment conducted under two different conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_ttest
</code></pre>


<h3>Format</h3>

<p>A data frame with 24 observations and 4 variables:
</p>

<dl>
<dt>Condition</dt><dd><p>Experimental conditions</p>
</dd>
<dt>Gene</dt><dd><p>Genes</p>
</dd>
<dt>E</dt><dd><p>Amplification efficiency</p>
</dd>
<dt>Ct</dt><dd><p>Ct values</p>
</dd>
</dl>



<h3>Source</h3>

<p>University of Kurdistan
</p>

<hr>
<h2 id='data_withTechRep'>Sample data (with technical replicates)</h2><span id='topic+data_withTechRep'></span>

<h3>Description</h3>

<p>A sample data for calculating biological replicated. Each line belongs to a separate individual (non-repeated measure experiment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_withTechRep
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations and 9 variables:
</p>

<dl>
<dt>factor1</dt><dd><p>experimental factor</p>
</dd>
<dt>factor2</dt><dd><p>experimental factor</p>
</dd>
<dt>factor3</dt><dd><p>experimental factor</p>
</dd>
<dt>biolrep</dt><dd><p>biological replicate</p>
</dd>
<dt>techrep</dt><dd><p>technical replicates</p>
</dd>
<dt>Etarget</dt><dd><p>Amplification efficiency of target gene</p>
</dd>
<dt>targetCt</dt><dd><p>Ct of target gene</p>
</dd>
<dt>Eref</dt><dd><p>Amplification efficiency of reference gene</p>
</dd>
<dt>refCt</dt><dd><p>Ct of reference gene</p>
</dd>
</dl>



<h3>Source</h3>

<p>Not applicable
</p>

<hr>
<h2 id='efficiency'>Slope, R2 and Efficiency (E) statistics and standard curves</h2><span id='topic+efficiency'></span>

<h3>Description</h3>

<p>The <code>efficiency</code> function calculates amplification efficiency and returns related statistics and standard curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efficiency(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="efficiency_+3A_df">df</code></td>
<td>
<p>a data frame of dilutions and Ct of genes. First column is dilutions and other columns are Ct values for different genes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>efficiency</code> function calculates amplification efficiency of genes, and present the Slope, Efficiency, and R2 statistics.
</p>


<h3>Value</h3>

<p>A list 3 elements.
</p>

<dl>
<dt>efficiency</dt><dd><p>Slope, R2 and Efficiency (E) statistics</p>
</dd>
<dt>Slope_compare</dt><dd><p>slope comparison table</p>
</dd>
<dt>plot</dt><dd><p>standard curves</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# locate and read the sample data
data_efficiency

# Applying the efficiency function
efficiency(data_efficiency)


</code></pre>

<hr>
<h2 id='Lee_etal2020qPCR'>Sample data (with technical replicates)</h2><span id='topic+Lee_etal2020qPCR'></span>

<h3>Description</h3>

<p>A sample data for calculating biological replicated. Each line belongs to a separate individual (non-repeated measure experiment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Lee_etal2020qPCR
</code></pre>


<h3>Format</h3>

<p>A data frame with 72 observations and 8 variables:
</p>

<dl>
<dt>factor1</dt><dd><p>experimental factor</p>
</dd>
<dt>DS</dt><dd><p>DS</p>
</dd>
<dt>biolRep</dt><dd><p>biological replicate</p>
</dd>
<dt>techRep</dt><dd><p>technical replicates</p>
</dd>
<dt>APOE_efficiency</dt><dd><p>Amplification efficiency of APOE gene</p>
</dd>
<dt>APOE_Ct</dt><dd><p>Ct of APOE gene</p>
</dd>
<dt>GAPDH_efficiency</dt><dd><p>Amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDH_Ct</dt><dd><p>Ct of GAPDH gene</p>
</dd>
</dl>



<h3>Source</h3>

<p>Lee et al, (2020) &lt;doi:10.12688/f1000research.23580.2&gt;
</p>

<hr>
<h2 id='meanTech'>Calculating mean of technical replicates</h2><span id='topic+meanTech'></span>

<h3>Description</h3>

<p>Calculating arithmetic mean of technical replicates for subsequent ANOVA analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meanTech(x, groups)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meanTech_+3A_x">x</code></td>
<td>
<p>A raw data frame including technical replicates.</p>
</td></tr>
<tr><td><code id="meanTech_+3A_groups">groups</code></td>
<td>
<p>grouping columns based on which the mean technical replicates are calculated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The meanTech calculates mean of technical replicates. Arithmetic mean of technical replicates can be calculated in order to 
simplify the statistical comparison between sample groups.
</p>


<h3>Value</h3>

<p>A data frame with the mean of technical replicates.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See example input data frame:
data_withTechRep

# Calculating mean of technical replicates
meanTech(data_withTechRep, groups = 1:4)

# Calculating mean of technical replicates
meanTech(Lee_etal2020qPCR, groups = 1:3)


</code></pre>

<hr>
<h2 id='multiplot'>Multiple plot function</h2><span id='topic+multiplot'></span>

<h3>Description</h3>

<p><code>multiplot</code> function combines multiple ggplot objects into a single plate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiplot(..., cols = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiplot_+3A_...">...</code></td>
<td>
<p>ggplot objects can be passed in ... or to plotlist (as a list of ggplot objects)</p>
</td></tr>
<tr><td><code id="multiplot_+3A_cols">cols</code></td>
<td>
<p>Number of columns in the panel</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Combining multiple ggplot objects into a single plate.
</p>


<h3>Value</h3>

<p>A  multiple-plots plate
</p>


<h3>Author(s)</h3>

<p>gist.github.com/pedroj/ffe89c67282f82c1813d
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
p1 &lt;- qpcrTTESTplot(data_ttest, 
                    numberOfrefGenes = 1,
                    ylab = "Average Fold Change (FC)",
                    width = 0.3)


out2 &lt;- qpcrANOVARE(data_1factor, numberOfrefGenes = 1, block = NULL)$Result
p2 &lt;- oneFACTORplot(out2,
                    width = 0.4,
                    fill = "skyblue",
                    y.axis.adjust = 0.5,
                    y.axis.by = 1,
                    errorbar = "ci",
                    show.letters = TRUE,
                    letter.position.adjust = 0.1,
                    ylab = "Relative Expression (RE)",
                    xlab = "Factor Levels",
                    fontsize = 12)
                    
multiplot(p1, p2, cols=2)

multiplot(p1, p2, cols=1)



</code></pre>

<hr>
<h2 id='oneFACTORplot'>Bar plot of the relative gene expression (<code class="reqn">\Delta C_T</code> method) from the <code>qpcrANOVARE</code> output of a single-factor experiment data</h2><span id='topic+oneFACTORplot'></span>

<h3>Description</h3>

<p>Bar plot of the relative expression of a gene along with the standard error (se), 95% confidence interval (ci) and significance. <code>oneFACTORplot</code> is mainly useful for a one-factor experiment with more than two levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oneFACTORplot(
  res,
  width = 0.4,
  fill = "skyblue",
  y.axis.adjust = 0.5,
  y.axis.by = 2,
  errorbar,
  show.letters = TRUE,
  letter.position.adjust = 0.1,
  ylab = "Relative Expression",
  xlab = "none",
  fontsize = 12,
  fontsizePvalue = 5,
  axis.text.x.angle = 0,
  axis.text.x.hjust = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="oneFACTORplot_+3A_res">res</code></td>
<td>
<p>an FC data frame object created by <code>qpcrANOVARE(x)$Result</code> function on a one factor data such as <code>data_1factor</code>.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_fill">fill</code></td>
<td>
<p>specify a fill color.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive number for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_show.letters">show.letters</code></td>
<td>
<p>a logical variable. If TRUE, mean grouping letters are added to the bars.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the grouping letters to the error bars.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_fontsize">fontsize</code></td>
<td>
<p>size of all fonts  of the plot.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_fontsizepvalue">fontsizePvalue</code></td>
<td>
<p>font size of the pvalue labels</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p>angle of x axis text</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_axis.text.x.hjust">axis.text.x.hjust</code></td>
<td>
<p>horizontal justification of x axis text</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>oneFACTORplot</code> function generates the bar plot of the average fold change for target genes along with the significance and the 95% confidence interval as error bars.
</p>


<h3>Value</h3>

<p>Bar plot of the average fold change for target genes along with the significance and the standard error or 95% confidence interval as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Before plotting, the result needs to be extracted as below:
res &lt;- qpcrANOVARE(data_1factor, numberOfrefGenes = 1, block = NULL)$Result

# Bar plot
oneFACTORplot(res,
         width = 0.4,
         fill = "skyblue",
         y.axis.adjust = 1,
         y.axis.by = 0.2,
         errorbar = "se",
         show.letters = TRUE,
         letter.position.adjust = 0.05,
         ylab = "Relative Expression",
         xlab = "Factor Levels",
         fontsize = 12)


</code></pre>

<hr>
<h2 id='qpcrANOVAFC'>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis using ANOVA and ANCOVA</h2><span id='topic+qpcrANOVAFC'></span>

<h3>Description</h3>

<p>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis of qPCR data can be done using 
ANOVA (analysis of variance) and ANCOVA (analysis of covariance) by the 
<code>qpcrANOVAFC</code> function, for uni- or multi-factorial experiment data. The bar plot of the fold change (FC) 
values along with the standard error (se) and confidence interval (ci) are returned by 
the <code>qpcrANOVAFC</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrANOVAFC(
  x,
  numberOfrefGenes,
  mainFactor.column,
  analysisType = "anova",
  mainFactor.level.order = NULL,
  block,
  width = 0.5,
  fill = "skyblue",
  y.axis.adjust = 2,
  y.axis.by = 1,
  letter.position.adjust = 0.1,
  ylab = "Fold Change",
  xlab = "none",
  fontsize = 12,
  fontsizePvalue = 7,
  axis.text.x.angle = 0,
  axis.text.x.hjust = 0.5,
  x.axis.labels.rename = "none",
  p.adj = "none",
  errorbar = "se",
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpcrANOVAFC_+3A_x">x</code></td>
<td>
<p>a data frame of condition(s), biological replicates, efficiency (E) and Ct values of 
target and reference genes. Each Ct value in the data frame is the mean of technical replicates. 
<strong>NOTE:</strong> Each line belongs to a separate individual reflecting a non-repeated measure experiment). 
See <a href="../doc/vignette.html"><code>vignette</code></a>, section &quot;data structure and column arrangement&quot; for details.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes which is 1 or 2 (up to two reference genes can be handled).</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_mainfactor.column">mainFactor.column</code></td>
<td>
<p>the factor for which fold change expression is calculated for its levels. 
If <code>ancova</code> is selected as <code>analysisType</code>, the remaining factors (if any) are considered as covariate(s).</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_analysistype">analysisType</code></td>
<td>
<p>should be one of &quot;ancova&quot; or &quot;anova&quot;. Default is &quot;anova&quot;.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_mainfactor.level.order">mainFactor.level.order</code></td>
<td>
<p>NULL (default) or a vector of main factor level names. If <code>NULL</code>, 
the first level of the <code>mainFactor.column</code> is used 
as calibrator. If a vector of main factor levels (in any order) is specified, the first level in the vector is 
used as calibrator. Calibrator is the reference level or sample that all others are compared to. Examples are untreated 
or time 0. The FC value of the reference or calibrator level is 1 because it is not changed compared to itself.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_block">block</code></td>
<td>
<p>column name of the block if there is a blocking factor (for correct column arrangement see 
example data.). When a qPCR experiment is done in multiple qPCR plates, variation resulting from the 
plates may interfere with the actual amount of gene expression. One solution is to conduct each plate 
as a complete randomized block so that at least one replicate of each treatment and control is present 
on a plate. Block effect is usually considered as random and its interaction with any main effect 
is not considered.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_fill">fill</code></td>
<td>
<p>specify the fill color for the columns in the bar plot. If a vector of two colors is specified, 
the reference level is differentialy colored.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive value for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the signs and the error bars.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_fontsize">fontsize</code></td>
<td>
<p>font size of the plot</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_fontsizepvalue">fontsizePvalue</code></td>
<td>
<p>font size of the pvalue labels</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p>angle of x axis text</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_axis.text.x.hjust">axis.text.x.hjust</code></td>
<td>
<p>horizontal justification of x axis text</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_x.axis.labels.rename">x.axis.labels.rename</code></td>
<td>
<p>a vector replacing the x axis labels in the bar plot</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_p.adj">p.adj</code></td>
<td>
<p>Method for adjusting p values</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p>
</td></tr>
<tr><td><code id="qpcrANOVAFC_+3A_plot">plot</code></td>
<td>
<p>if <code>FALSE</code>, prevents the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis of qPCR data can be done using 
ANOVA (analysis of variance) and ANCOVA (analysis of covariance) by the 
<code>qpcrANOVAFC</code> function, for uni- or multi-factorial experiment data. 
If there are more than one factor, FC value calculations for 
the 'mainFactor.column' and the statistical analysis is performed based on a full model factorial 
experiment by default. However, if 'ancova' is defined for the 'analysisType' argument,
FC values are calculated for the levels of the 'mainFactor.column' and the other factors are 
used as covariate(s) in the analysis of variance, but we should consider ANCOVA table:
if the interaction between the main factor and covariate is significant, ANCOVA is not appropriate in this case. 
ANCOVA is basically used when a factor is affected by uncontrolled quantitative covariate(s). 
For example, suppose that wDCt of a target gene in a plant is affected by temperature. The gene may 
also be affected by drought. Since we already know that temperature affects the target gene, we are 
interested to know if the gene expression is also altered by the drought levels. We can design an 
experiment to understand the gene behavior at both temperature and drought levels at the same time. 
The drought is another factor (the covariate) that may affect the expression of our gene under the 
levels of the first factor i.e. temperature. The data of such an experiment can be analyzed by ANCOVA 
or using ANOVA based on a factorial experiment using <code>qpcrANOVAFC</code>. <code>qpcrANOVAFC</code> function performs FC 
analysis even there is only one factor (without covariate or factor  variable). Bar plot of fold changes 
(FC) values along with the standard errors are also returned by the <code>qpcrANOVAFC</code> function.
</p>


<h3>Value</h3>

<p>A list with 7 elements:
</p>

<dl>
<dt>Final_data</dt><dd><p>Input data frame plus the weighted Delat Ct values (wDCt)</p>
</dd>
<dt>lm_ANOVA</dt><dd><p>lm of factorial analysis-tyle</p>
</dd>
<dt>lm_ANCOVA</dt><dd><p>lm of ANCOVA analysis-type</p>
</dd>
<dt>ANOVA_table</dt><dd><p>ANOVA table</p>
</dd>
<dt>ANCOVA_table</dt><dd><p>ANCOVA table</p>
</dd>
<dt>FC Table</dt><dd><p>Table of FC values, significance and confidence interval and standard error with the lower and upper limits for the main factor levels.</p>
</dd>
<dt>Bar plot of FC values</dt><dd><p>Bar plot of the fold change values for the main factor levels.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method. Methods 25 (4). doi:10.1006/meth.2001.1262.
</p>
<p>Ganger, MT, Dietz GD, and Ewing SJ. 2017. A common base method for analysis of qPCR data
and the application of simple blocking in qPCR experiments. BMC bioinformatics 18, 1-11.
</p>
<p>Yuan, Joshua S, Ann Reed, Feng Chen, and Neal Stewart. 2006.
Statistical Analysis of Real-Time PCR Data. BMC Bioinformatics 7 (85). doi:10.1186/1471-2105-7-85.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

qpcrANOVAFC(data_1factor, numberOfrefGenes = 1, mainFactor.column = 1, block = NULL, 
fill = c("grey", "skyblue"), fontsizePvalue = 5, y.axis.adjust = 0.1)


qpcrANOVAFC(data_2factor, numberOfrefGenes = 1, mainFactor.column = 2, block = NULL, 
analysisType = "ancova", fontsizePvalue = 5, y.axis.adjust = 1)


# Data from Lee et al., 2020, Here, the data set contains technical 
# replicates so we get mean of technical replicates first:
df &lt;- meanTech(Lee_etal2020qPCR, groups = 1:3)
qpcrANOVAFC(df, numberOfrefGenes = 1, analysisType = "ancova", block = NULL, 
mainFactor.column = 2, fill = c("grey", "skyblue"), y.axis.adjust = 0.05)


qpcrANOVAFC(data_2factorBlock,  numberOfrefGenes = 1, mainFactor.column = 1, 
mainFactor.level.order = c("S", "R"), block = "block", 
fill = c("grey", "skyblue"), analysisType = "ancova",
fontsizePvalue = 7, y.axis.adjust = 0.1, width = 0.35)


df &lt;- meanTech(Lee_etal2020qPCR, groups = 1:3) 
df2 &lt;- df[df$factor1 == "DSWHi",][-1]
qpcrANOVAFC(df2, mainFactor.column = 1, fontsizePvalue = 5, y.axis.adjust = 2,
block = NULL, numberOfrefGenes = 1, analysisType = "anova")


addline_format &lt;- function(x,...){gsub('\\s','\n',x)}
qpcrANOVAFC(data_1factor, numberOfrefGenes = 1, mainFactor.column = 1,
block = NULL, fill = c("grey", "skyblue"), y.axis.by = 1, 
letter.position.adjust = 0, y.axis.adjust = 1, ylab = "Fold Change", 
fontsize = 12, x.axis.labels.rename = addline_format(c("Control", 
                                         "Treatment_1 vs Control", 
                                         "Treatment_2 vs Control")))
                                                       
                                                       
</code></pre>

<hr>
<h2 id='qpcrANOVARE'>Relative expression (<code class="reqn">\Delta C_T</code> method) analysis using ANOVA</h2><span id='topic+qpcrANOVARE'></span>

<h3>Description</h3>

<p>Analysis of variance of relative expression (<code class="reqn">\Delta C_T</code> method) values for 
all factor level combinations in the experiment in which the expression level of a 
reference gene is used as normalizer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrANOVARE(x, numberOfrefGenes, block, alpha = 0.05, adjust = "none")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpcrANOVARE_+3A_x">x</code></td>
<td>
<p>a data frame consisting of condition columns, target gene efficiency (E), target Gene Ct, reference 
gene efficiency and reference gene Ct values, respectively. Each Ct in the data frame is the mean of 
technical replicates. Complete amplification efficiencies of 2 was assumed in the example data for 
all wells but the calculated efficienies can be used instead.  <strong>NOTE:</strong> Each line belongs to a separate 
individual reflecting a non-repeated measure experiment). See <a href="../doc/vignette.html"><code>vignette</code></a>, 
section &quot;data structure and column arrangement&quot; for details.</p>
</td></tr>
<tr><td><code id="qpcrANOVARE_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes (1 or 2). Up to two reference genes can be handled.</p>
</td></tr>
<tr><td><code id="qpcrANOVARE_+3A_block">block</code></td>
<td>
<p>column name of the blocking factor (for correct column arrangement see example data.). 
When a qPCR experiment is done in multiple qPCR plates, variation resulting from the plates may 
interfere with the actual amount of gene expression. One solution is to conduct each plate as a 
complete randomized block so that at least one replicate of each treatment and control is present 
on a plate. Block effect is usually considered as random and its interaction with any main effect is 
not considered.</p>
</td></tr>
<tr><td><code id="qpcrANOVARE_+3A_alpha">alpha</code></td>
<td>
<p>significance level</p>
</td></tr>
<tr><td><code id="qpcrANOVARE_+3A_adjust">adjust</code></td>
<td>
<p>method for adjusting p-values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrANOVARE</code> function performs analysis of variance (ANOVA) of relative 
expression (RE) values for all factor level combinations as treatments using the expression 
level of a reference gene is used as normalizer. To get a reliable result, the expression of 
the reference gene needs to be constant across all test samples and it expression should not 
be affected by the experimental treatment under study.
</p>


<h3>Value</h3>

<p>A list with 4 elements:
</p>

<dl>
<dt>Final_data</dt><dd><p>The row data plus weighed delta Ct (wDCt) values.</p>
</dd>
<dt>lm</dt><dd><p>The output of linear model analysis including ANOVA tables</p>
</dd>
<dt>ANOVA</dt><dd><p>ANOVA table based on CRD</p>
</dd>
<dt>Result</dt><dd><p>The result table including treatments and factors, RE (Relative Expression), LCL, UCL, 
letter display for pair-wise comparisons and standard error with the lower and upper limits.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method. Methods 25 (4). doi:10.1006/meth.2001.1262.
</p>
<p>Ganger, MT, Dietz GD, and Ewing SJ. 2017. A common base method for analysis of qPCR data
and the application of simple blocking in qPCR experiments. BMC bioinformatics 18, 1-11.
</p>
<p>Yuan, Joshua S, Ann Reed, Feng Chen, and Neal Stewart. 2006.
Statistical Analysis of Real-Time PCR Data. BMC Bioinformatics 7 (85). doi:10.1186/1471-2105-7-85.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# If the data include technical replicates, means of technical replicates
# should be calculated first using meanTech function.
# Applying ANOVA
qpcrANOVARE(data_3factor, numberOfrefGenes = 1, block = NULL)


qpcrANOVARE(data_2factorBlock, block = "Block", numberOfrefGenes = 1)


</code></pre>

<hr>
<h2 id='qpcrMeans'>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis using a model</h2><span id='topic+qpcrMeans'></span>

<h3>Description</h3>

<p>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis using a model object produced by the
<code>qpcrANOVAFC</code> or <code>qpcrREPEATED</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrMeans(model, specs, p.adj = "none")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpcrMeans_+3A_model">model</code></td>
<td>
<p>an 'lmer' fitted model object created by qpcrANOVAFC or qpcrREPEATED functions</p>
</td></tr>
<tr><td><code id="qpcrMeans_+3A_specs">specs</code></td>
<td>
<p>A character vector specifying the names of the predictors over which FC values are desired</p>
</td></tr>
<tr><td><code id="qpcrMeans_+3A_p.adj">p.adj</code></td>
<td>
<p>Method for adjusting p values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrMeans</code> function performs fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis using a model produced by the
<code>qpcrANOVAFC</code> or <code>qpcrREPEATED</code>. The values can be returned for any effects in the model including simple effects,
interactions and slicing if an ANOVA model is used, but ANCOVA models returned by rtpcr package only include simple effects.
</p>


<h3>Value</h3>

<p>Table of FC values, significance and confidence interval.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Returning fold change values from a fitted model.
# Firstly, result of `qpcrANOVAFC` or `qpcrREPEATED` is 
# acquired which includes a model object:
res &lt;- qpcrANOVAFC(data_3factor, numberOfrefGenes = 1, mainFactor.column = 1, block = NULL)

# Returning fold change values of Type levels from a fitted model:
qpcrMeans(res$lm_ANOVA, specs = "Type")

# Returning fold change values of Conc levels from a fitted model:
qpcrMeans(res$lm_ANOVA, specs = "Conc")

# Returning fold change values of Conc levels sliced by Type:
qpcrMeans(res$lm_ANOVA, specs = "Conc | Type")

# Returning fold change values of Conc levels sliced by Type*SA:
qpcrMeans(res$lm_ANOVA, specs = "Conc | (Type*SA)")


</code></pre>

<hr>
<h2 id='qpcrREPEATED'>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis of repeated measure qPCR data</h2><span id='topic+qpcrREPEATED'></span>

<h3>Description</h3>

<p><code>qpcrREPEATED</code> function performs fold change (<code class="reqn">\Delta \Delta C_T</code> method) 
analysis of observations repeatedly taken over different time courses. 
Data may be obtained over time from a uni- or multi-factorial experiment. The bar plot of the fold changes (FC) 
values along with the standard error (se) or confidence interval (ci) is also returned by the <code>qpcrREPEATED</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrREPEATED(
  x,
  numberOfrefGenes,
  factor,
  block,
  width = 0.5,
  fill = "#BFEFFF",
  y.axis.adjust = 2,
  y.axis.by = 1,
  ylab = "Fold Change",
  xlab = "none",
  fontsize = 12,
  fontsizePvalue = 7,
  axis.text.x.angle = 0,
  axis.text.x.hjust = 0.5,
  x.axis.labels.rename = "none",
  letter.position.adjust = 0,
  p.adj = "none",
  errorbar = "se",
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpcrREPEATED_+3A_x">x</code></td>
<td>
<p>input data frame in which the first column is id, 
followed by the factor column(s) which include at least time. 
The first level of time in data frame is used as calibrator or reference level.
Additional factor(s) may also be present. Other columns are efficiency and Ct values of target and reference genes.
<strong>NOTE:</strong> In the &quot;id&quot; column, a unique number is assigned to each individual from which samples have been taken over time, 
for example see <code>data_repeated_measure_1</code>, 
all the three number 1 indicate one individual which has been sampled over three different time courses.
See <a href="../doc/vignette.html"><code>vignette</code></a>, section &quot;data structure and column arrangement&quot; for details.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes which is 1 or 2 (Up to two reference genes can be handled).
as reference or calibrator which is the reference level or sample that all others are compared to. Examples are untreated 
of time 0. The FC value of the reference or calibrator level is 1 because it is not changed compared to itself.
If NULL, the first level of the main factor column is used as calibrator.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_factor">factor</code></td>
<td>
<p>the factor for which the FC values is analysed. The first level of the specified factor in the input data frame is used as calibrator.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_block">block</code></td>
<td>
<p>column name of the block if there is a blocking factor (for correct column arrangement see example data.). 
Block effect is usually considered as random and its interaction with any main effect is not considered.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width in the output bar plot.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_fill">fill</code></td>
<td>
<p>specify the fill color for the columns in the bar plot. If a vector of two colors is specified, the reference level is differentialy colored.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive value for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_fontsize">fontsize</code></td>
<td>
<p>font size of the plot</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_fontsizepvalue">fontsizePvalue</code></td>
<td>
<p>font size of the pvalue labels</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p>angle of x axis text</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_axis.text.x.hjust">axis.text.x.hjust</code></td>
<td>
<p>horizontal justification of x axis text</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_x.axis.labels.rename">x.axis.labels.rename</code></td>
<td>
<p>a vector replacing the x axis labels in the bar plot</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the signs and the error bars.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_p.adj">p.adj</code></td>
<td>
<p>Method for adjusting p values</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p>
</td></tr>
<tr><td><code id="qpcrREPEATED_+3A_plot">plot</code></td>
<td>
<p>if <code>FALSE</code>, prevents the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrREPEATED</code> function performs fold change (FC) analysis of observations repeatedly taken over time. 
The intended factor (could be time or any other factor) is defined for the analysis by the <code>factor</code> argument, 
then the function performs FC analysis on its levels
so that the first levels (as appears in the input data frame) is used as reference or calibrator. 
The function returns FC values along with confidence interval and standard error for the FC values.
</p>


<h3>Value</h3>

<p>A list with 5 elements:
</p>

<dl>
<dt>Final_data</dt><dd><p>Input data frame plus the weighted Delat Ct values (wDCt)</p>
</dd>
<dt>lm</dt><dd><p>lm of factorial analysis-tyle</p>
</dd>
<dt>ANOVA_table</dt><dd><p>ANOVA table</p>
</dd>
<dt>FC Table</dt><dd><p>Table of FC values, significance, confidence interval and standard error with the lower and upper limits for the selected factor levels.</p>
</dd>
<dt>Bar plot of FC values</dt><dd><p>Bar plot of the fold change values for the main factor levels.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
qpcrREPEATED(data_repeated_measure_1,
            numberOfrefGenes = 1,
            factor = "time", block = NULL)

qpcrREPEATED(data_repeated_measure_2,
             numberOfrefGenes = 1,
             factor = "time", block = NULL)
                                                       
                                                       
</code></pre>

<hr>
<h2 id='qpcrTTEST'>Fold change (<code class="reqn">\Delta \Delta C_T</code> method) analysis of target genes using t-test</h2><span id='topic+qpcrTTEST'></span>

<h3>Description</h3>

<p>t.test based analysis of the fold change expression for any number of target genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrTTEST(x, numberOfrefGenes, paired = FALSE, var.equal = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpcrTTEST_+3A_x">x</code></td>
<td>
<p>a data frame of 4 columns including Conditions, E (efficiency), Gene and Ct values (see examples below). Biological replicates needs to be equal for all Genes. Each Ct value is the mean of technical replicates. Complete amplification efficiencies of 2 is assumed here for all wells but the calculated efficienies can be used instead. See <a href="../doc/vignette.html"><code>vignette</code></a> for details about &quot;data structure and column arrangement&quot;.</p>
</td></tr>
<tr><td><code id="qpcrTTEST_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes. Up to two reference genes can be handled.</p>
</td></tr>
<tr><td><code id="qpcrTTEST_+3A_paired">paired</code></td>
<td>
<p>a logical indicating whether you want a paired t-test.</p>
</td></tr>
<tr><td><code id="qpcrTTEST_+3A_var.equal">var.equal</code></td>
<td>
<p>a logical variable indicating whether to treat the two variances as being equal. If TRUE then the pooled variance is used to estimate the variance otherwise the Welch (or Satterthwaite) approximation to the degrees of freedom is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrTTEST</code> function applies a t.test based analysis to calculate 
fold change (<code class="reqn">\Delta \Delta C_T</code> method) expression and returns related statistics for any number of 
target genes that have been evaluated under control and treatment conditions. Sampling may be paired or 
unpaired. One or two reference genes can be used. Unpaired and paired samples are commonly analyzed 
using unpaired and paired t-test, respectively.  <strong>NOTE:</strong> Paired samples in quantitative PCR refer to two sample 
data that are collected from one set of individuals 
at two different conditions, for example before and after a treatment or at two different time points. While 
for unpaired samples, two sets of individuals are used: one under untreated and the other set under treated 
condition.  Paired samples allow to compare gene expression changes within the same individual, reducing 
inter-individual variability.
</p>


<h3>Value</h3>

<p>A list of two elements:
</p>

<dl>
<dt>Row_data</dt><dd><p>The row data including Genes and weighed delta Ct (wDCt) values.</p>
</dd>
<dt>Result</dt><dd><p>Output table including the Fold Change values, lower and upper confidence interval, pvalue and standard error with the lower and upper limits.</p>
</dd>
</dl>

<p>For more information about the test procedure and its arguments,
refer <code><a href="stats.html#topic+t.test">t.test</a></code>, and <code><a href="stats.html#topic+lm">lm</a></code>.
If the residuals of the model do not follow normal distribution and variances between the two groups are not homoGene, <code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code> procedure may be concidered
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method. Methods 25 (4). doi:10.1006/meth.2001.1262.
</p>
<p>Ganger, MT, Dietz GD, and Ewing SJ. 2017. A common base method for analysis
of qPCR data and the application of simple blocking in qPCR experiments.
BMC bioinformatics 18, 1-11.
</p>
<p>Yuan, Joshua S, Ann Reed, Feng Chen, and Neal Stewart. 2006.
Statistical Analysis of Real-Time PCR Data. BMC Bioinformatics 7 (85).
doi:10.1186/1471-2105-7-85.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See the sample data structure
data_ttest

# Getting t.test results
qpcrTTEST(data_ttest,
   paired = FALSE,
   var.equal = TRUE,
   numberOfrefGenes = 1)



qpcrTTEST(Taylor_etal2019, 
          numberOfrefGenes = 2, 
          var.equal = TRUE)
 


</code></pre>

<hr>
<h2 id='qpcrTTESTplot'>Bar plot of the average fold change (<code class="reqn">\Delta \Delta C_T</code> method) of target genes</h2><span id='topic+qpcrTTESTplot'></span>

<h3>Description</h3>

<p>Bar plot of the fold change (<code class="reqn">\Delta \Delta C_T</code> method) values for for any number of target 
genes under a two-level conditional experimental (e.g. control and treatment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrTTESTplot(
  x,
  order = "none",
  numberOfrefGenes,
  paired = FALSE,
  var.equal = TRUE,
  width = 0.5,
  fill = "skyblue",
  y.axis.adjust = 0,
  y.axis.by = 2,
  letter.position.adjust = 0.3,
  ylab = "Average Fold Change",
  xlab = "none",
  fontsize = 12,
  fontsizePvalue = 7,
  axis.text.x.angle = 0,
  axis.text.x.hjust = 0.5,
  errorbar = "se"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qpcrTTESTplot_+3A_x">x</code></td>
<td>
<p>a data frame. The data frame consists of 4 columns belonging to condition levels, 
E (efficiency), genes and Ct values, respectively. Each Ct in the following data frame is 
the mean of technical replicates. Complete amplification efficiencies of 2 is assumed here 
for all wells but the calculated efficienies can be used we well. We use this data set for 
fold change expression analysis of the target genes in treatment condition compared to 
normal condition. See <a href="../doc/vignette.html"><code>vignette</code></a> for details.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_order">order</code></td>
<td>
<p>a vector determining genes order on the output graph.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes. Up to two reference genes can be handled.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_paired">paired</code></td>
<td>
<p>a logical indicating whether you want a paired t-test.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_var.equal">var.equal</code></td>
<td>
<p>a logical variable indicating whether to treat the two variances as being equal. If TRUE then the pooled variance is used to estimate the variance otherwise the Welch (or Satterthwaite) approximation to the degrees of freedom is used.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_fill">fill</code></td>
<td>
<p>specify the fill color for the columns of the bar plot.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive value for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the signs and the error bars.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_fontsize">fontsize</code></td>
<td>
<p>fonts size of the plot</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_fontsizepvalue">fontsizePvalue</code></td>
<td>
<p>font size of the pvalue labels</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p>angle of x axis text</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_axis.text.x.hjust">axis.text.x.hjust</code></td>
<td>
<p>horizontal justification of x axis text</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrTTESTplot</code> function applies a t.test based analysis to any number of target genes 
along with one or two reference gene(s), that have been evaluated under control and treatment conditions. 
It returns the bar plot of the fold change (FC) values for target genes along with the 95% CI and significance. 
Sampling may be unpaired or paired. Unpaired and paired samples are commonly analyzed using unpaired and 
paired t-test, respectively.Paired samples in quantitative PCR refer to two sample data that are collected 
from one set of individuals 
at two different conditions, for example before and after a treatment or at two different time points. While 
for unpaired samples, two sets of individuals are used: one under untreated and the other set under treated 
condition. Paired samples allow to compare gene expression changes within the same individual, reducing 
inter-individual variability.
</p>


<h3>Value</h3>

<p>Bar  plot of the average fold change for target genes along with the significance and the 95 percent CI as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See a sample data frame
data_ttest


qpcrTTESTplot(data_ttest, 
              numberOfrefGenes = 1,
              errorbar = "ci")


# Producing the plot
qpcrTTESTplot(data_ttest,
              numberOfrefGenes = 1,
              order = c("C2H2-01", "C2H2-12", "C2H2-26"),
              paired = FALSE,
              var.equal = TRUE,
              width = 0.5,
              fill = "skyblue",
              y.axis.adjust = 0,
              y.axis.by = 2,
              letter.position.adjust = 0.3,
              ylab = "Fold Change in Treatment vs Control",
              xlab = "Gene",
              errorbar = "se")


</code></pre>

<hr>
<h2 id='Taylor_etal2019'>Sample qPCR data (one target and two reference genes under two different conditions)</h2><span id='topic+Taylor_etal2019'></span>

<h3>Description</h3>

<p>Sample qPCR data (one target and two reference genes under two different conditions)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Taylor_etal2019
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations and 4 variables:
</p>

<dl>
<dt>Condition</dt><dd><p>Experimental conditions</p>
</dd>
<dt>Gene</dt><dd><p>Genes</p>
</dd>
<dt>E</dt><dd><p>Amplification efficiency</p>
</dd>
<dt>Ct</dt><dd><p>Ct values</p>
</dd>
</dl>



<h3>Source</h3>

<p>Not applicable
</p>

<hr>
<h2 id='threeFACTORplot'>Bar plot of the relative gene expression (<code class="reqn">\Delta C_T</code> method) from the <code>qpcrANOVARE</code> output of a a three-factorial experiment data</h2><span id='topic+threeFACTORplot'></span>

<h3>Description</h3>

<p>Bar plot of the relative expression (<code class="reqn">\Delta C_T</code> method) of a gene along with the confidence interval and significance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threeFACTORplot(
  res,
  arrangement = c(1, 2, 3),
  bar.width = 0.5,
  fill = "Reds",
  xlab = "none",
  ylab = "Relative Expression",
  errorbar,
  y.axis.adjust = 0.5,
  y.axis.by = 2,
  letter.position.adjust = 0.3,
  legend.title = "Legend Title",
  legend.position = c(0.4, 0.8),
  fontsize = 12,
  fontsizePvalue = 5,
  show.letters = TRUE,
  axis.text.x.angle = 0,
  axis.text.x.hjust = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="threeFACTORplot_+3A_res">res</code></td>
<td>
<p>the FC data frame created by <code>qpcrANOVARE(x)$Result</code> function on a three factorial data such as <code>data_3factor</code> example data frame.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_arrangement">arrangement</code></td>
<td>
<p>order based on the columns in the output table (e.g. c(2,3,1) or c(1,3,2)) affecting factor arrangement of the output graph.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_bar.width">bar.width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_fill">fill</code></td>
<td>
<p>a color vector specifying the fill color for the columns of the bar plot. One of the palettes in <code><a href="RColorBrewer.html#topic+display.brewer.all">display.brewer.all</a></code> (e.g. &quot;Reds&quot; or &quot;Blues&quot;, ...) can be applied.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive number for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the grouping letters to the error bars</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_legend.position">legend.position</code></td>
<td>
<p>a two digit vector specifying the legend position.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_fontsize">fontsize</code></td>
<td>
<p>all fonts size of the plot</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_fontsizepvalue">fontsizePvalue</code></td>
<td>
<p>font size of the pvalue labels</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_show.letters">show.letters</code></td>
<td>
<p>a logical variable. If TRUE, mean grouping letters are added to the bars.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p>angle of x axis text</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_axis.text.x.hjust">axis.text.x.hjust</code></td>
<td>
<p>horizontal justification of x axis text</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>threeFACTORplot</code> function generates the bar plot of the average fold change for target genes along with the significance, standard error (se) and the 95% confidence interval (ci).
</p>


<h3>Value</h3>

<p>Bar plot of the average fold change for target genes along with the standard error or 95% confidence interval as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#' # See a sample data frame
data_3factor

# Before plotting, the result needs to be extracted as below:
res &lt;- qpcrANOVARE(data_3factor, numberOfrefGenes = 1, block = NULL)$Result
res

# Arrange the first three colunms of the result table.
# This determines the columns order and shapes the plot output.
threeFACTORplot(res, arrangement = c(3, 1, 2), errorbar = "se",
    xlab = "condition")


threeFACTORplot(res, arrangement = c(1, 2, 3), bar.width = 0.5, fill = "Greys", 
xlab = "Genotype", ylab = "Relative Expression", errorbar = "se")


# Reordering factor levels to a desired order.
res$Conc &lt;- factor(res$Conc, levels = c("L","M","H"))
res$Type &lt;- factor(res$Type, levels = c("S","R"))

# Producing the plot
threeFACTORplot(res, arrangement = c(2, 3, 1), bar.width = 0.5, 
fill = "Reds", xlab = "Drought", ylab = "Relative Expression", 
errorbar = "se", legend.title = "Genotype", legend.position = c(0.2, 0.8))


# When using ci as error, increase the 
# y.axis.adjust value to see the plot correctly!
threeFACTORplot(res, arrangement = c(2, 3, 1), bar.width = 0.8, fill = "Greens", 
xlab = "Drought", ylab = "Relative Expression", errorbar = "ci", 
y.axis.adjust = 1, y.axis.by = 2, letter.position.adjust = 0.6, 
legend.title = "Genotype", fontsize = 12, legend.position = c(0.2, 0.8), 
show.letters = TRUE)



</code></pre>

<hr>
<h2 id='twoFACTORplot'>Bar plot of the relative gene expression (<code class="reqn">\Delta C_T</code> method) from the <code>qpcrANOVARE</code> output of a two-factorial experiment data</h2><span id='topic+twoFACTORplot'></span>

<h3>Description</h3>

<p>Bar plot of the relative expression (<code class="reqn">\Delta C_T</code> method) of a gene along with the standard error (se), 95% confidence interval (ci) and significance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoFACTORplot(
  res,
  x.axis.factor,
  group.factor,
  width = 0.5,
  fill = "Blues",
  y.axis.adjust = 0.5,
  y.axis.by = 2,
  show.errorbars = TRUE,
  errorbar,
  show.letters = TRUE,
  letter.position.adjust = 0.1,
  ylab = "Relative Expression",
  xlab = "none",
  legend.position = c(0.09, 0.8),
  fontsize = 12,
  fontsizePvalue = 5,
  axis.text.x.angle = 0,
  axis.text.x.hjust = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="twoFACTORplot_+3A_res">res</code></td>
<td>
<p>the FC data frame created by <code>qpcrANOVARE(x)$Result</code> function on a two factor data such as <code>data_2factor</code>.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_x.axis.factor">x.axis.factor</code></td>
<td>
<p>x-axis factor.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_group.factor">group.factor</code></td>
<td>
<p>grouping factor.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_fill">fill</code></td>
<td>
<p>specify the fill color vector for the columns of the bar plot. One of the palettes in <code><a href="RColorBrewer.html#topic+display.brewer.all">display.brewer.all</a></code> (e.g. &quot;Reds&quot; or &quot;Blues&quot;, ...) can be applied.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive number for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_show.errorbars">show.errorbars</code></td>
<td>
<p>show errorbars</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>se</code> or <code>ci</code>.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_show.letters">show.letters</code></td>
<td>
<p>a logical variable. If TRUE, mean grouping letters are added to the bars.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the grouping letters to the error bars.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_legend.position">legend.position</code></td>
<td>
<p>a two digit vector specifying the legend position.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_fontsize">fontsize</code></td>
<td>
<p>size of all fonts  of the plot.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_fontsizepvalue">fontsizePvalue</code></td>
<td>
<p>font size of the pvalue labels</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p>angle of x axis text</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_axis.text.x.hjust">axis.text.x.hjust</code></td>
<td>
<p>horizontal justification of x axis text</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>twoFACTORplot</code> function generates the bar plot of the average fold change for target genes along with the significance, standard error (se) and the 95% confidence interval (ci) as error bars.
</p>


<h3>Value</h3>

<p>Bar plot of the average fold change for target genes along with the standard error or 95% confidence interval as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See a sample data frame
data_2factor

# Before generating plot, the result table needs to be extracted as below:
res &lt;- qpcrANOVARE(data_2factor, numberOfrefGenes = 1, block = NULL)$Result

# Plot of the 'res' data with 'Genotype' as grouping factor
twoFACTORplot(res,
   x.axis.factor = Drought,
   group.factor = Genotype,
   width = 0.5,
   fill = "Greens",
   y.axis.adjust = 1,
   y.axis.by = 2,
   ylab = "Relative Expression",
   xlab = "Drought Levels",
   letter.position.adjust = 0.2,
   legend.position = c(0.2, 0.8),
   errorbar = "se")
   

# Plotting the same data with 'Drought' as grouping factor
twoFACTORplot(res,
              x.axis.factor = Genotype,
              group.factor = Drought,
              xlab = "Genotype",
              fill = "Blues",
              fontsizePvalue = 5,
              errorbar = "ci")
              
              
              
# Combining FC results of two different genes:
a &lt;- qpcrREPEATED(data_repeated_measure_1,
                  numberOfrefGenes = 1,
                  factor = "time", block = NULL, plot = FALSE)

b &lt;- qpcrREPEATED(data_repeated_measure_2,
                  factor = "time",
                  numberOfrefGenes = 1, block = NULL, plot = FALSE)

a1 &lt;- a$FC_statistics_of_the_main_factor
b1 &lt;- b$FC_statistics_of_the_main_factor
c &lt;- rbind(a1, b1)
c$gene &lt;- factor(c(1,1,1,2,2,2))
c

twoFACTORplot(c, x.axis.factor = contrast, 
              group.factor = gene, fill = 'Reds',
              ylab = "FC", axis.text.x.angle = 45,
              errorbar = "se", y.axis.adjust = 1,
              axis.text.x.hjust = 1)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
