<!DOCTYPE html><html><head><title>Help for package rtpcr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rtpcr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data_1factor'><p>A sample one factor qPCR dataset</p></a></li>
<li><a href='#data_2factor'><p>A sample two factor qPCR dataset</p></a></li>
<li><a href='#data_2factorBlock'><p>A sample two factor qPCR data set with blocking factor</p></a></li>
<li><a href='#data_3factor_a'><p>A sample multi-factorial qPCR dataset</p></a></li>
<li><a href='#data_3factor_b'><p>A three-factorial sample qPCR data</p></a></li>
<li><a href='#data_efficiency'><p>A sample dataset of amplification efficiency qPCR data.</p></a></li>
<li><a href='#data_ttest'><p>A sample two conditional qPCR data</p></a></li>
<li><a href='#data_ttest2'><p>A sample data with one target and two reference genes</p></a></li>
<li><a href='#data_withTechRep'><p>A sample data with technical replicates</p></a></li>
<li><a href='#efficiency'><p>efficiency, standard curves and related statistics</p></a></li>
<li><a href='#meanTech'><p>Calculating mean of technical replicates</p></a></li>
<li><a href='#multiplot'><p>Multiple plot function</p></a></li>
<li><a href='#oneFACTORplot'><p>Bar plot of the relative gene expression from a one-factor experiment</p></a></li>
<li><a href='#qpcrANCOVA'><p>ANCOVA and ANOVA on a factorial design plus bar plot of FC</p></a></li>
<li><a href='#qpcrANOVA'><p>Analysis of Variance of RE values based on CRD</p></a></li>
<li><a href='#qpcrTTEST'><p>Analysis of the fold change (FC) of target genes assessed under two conditions.</p></a></li>
<li><a href='#qpcrTTESTplot'><p>Bar plot of the average fold change of the target genes</p></a></li>
<li><a href='#threeFACTORplot'><p>Bar plot of the relative gene expression (RE) from a three-factor experiment</p></a></li>
<li><a href='#twoFACTORplot'><p>Bar plot of the relative gene expression (RE) from a two-factor experiment</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>qPCR Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Various methods are employed for statistical analysis and graphical presentation of real-time PCR (quantitative PCR  or qPCR) data. 'rtpcr' handles amplification efficiency calculation, statistical analysis and graphical representation of real-time PCR data based on up to two reference genes. By accounting for amplification efficiency values, 'rtpcr' was developed using a general calculation method described by Ganger et al. (2017) &lt;<a href="https://doi.org/10.1186%2Fs12859-017-1949-5">doi:10.1186/s12859-017-1949-5</a>&gt;, covering both the Livak and Pfaffl methods. Based on the experimental conditions, the functions of the 'rtpcr' package use t-test (for experiments with a two-level factor) or analysis of variance (for cases where more than two levels or factors exist) to calculate the fold change or relative expression. The functions also provide standard deviations and confidence limits for means and apply statistical mean comparisons. To facilitate using 'rtpcr', different datasets have been employed in the examples and the outputs are explained. An outstanding feature of 'rtpcr' package is providing publication-ready bar plots with various controlling arguments for experiments with up to three different factors. The 'rtpcr' package is user-friendly and easy to work with and provides an applicable resource for analyzing real-time PCR data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>agricolae, ggplot2, lme4, reshape2, tidyr, dplyr, grid</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-05 14:18:22 UTC; GM</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Author:</td>
<td>Ghader Mirzaghaderi [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ghader Mirzaghaderi &lt;gh.mirzaghaderi@uok.ac.ir&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-05 17:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='data_1factor'>A sample one factor qPCR dataset</h2><span id='topic+data_1factor'></span>

<h3>Description</h3>

<p>A sample dataset for demonstration purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_1factor)
</code></pre>


<h3>Format</h3>

<p>A data frame with 9 observations and 6 variables:
</p>

<dl>
<dt>SA</dt><dd><p>First experimental factor</p>
</dd>
<dt>Rep</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>Where the data comes from (if applicable)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data_1factor

</code></pre>

<hr>
<h2 id='data_2factor'>A sample two factor qPCR dataset</h2><span id='topic+data_2factor'></span>

<h3>Description</h3>

<p>A sample dataset for demonstration purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_2factor)
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations and 7 variables:
</p>

<dl>
<dt>Genotype</dt><dd><p>First experimental factor</p>
</dd>
<dt>Drought</dt><dd><p>Second experimental factor</p>
</dd>
<dt>Rep</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>Where the data comes from (if applicable)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_2factor)
data_2factor

</code></pre>

<hr>
<h2 id='data_2factorBlock'>A sample two factor qPCR data set with blocking factor</h2><span id='topic+data_2factorBlock'></span>

<h3>Description</h3>

<p>A sample qPCR data set with blocking factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_2factorBlock)
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations and 8 variables:
</p>

<dl>
<dt>factor1</dt><dd><p>First experimental factor</p>
</dd>
<dt>factor2</dt><dd><p>Second experimental factor</p>
</dd>
<dt>block</dt><dd><p>Second experimental factor</p>
</dd>
<dt>Rep</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>Where the data comes from (if applicable)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_2factorBlock)
data_2factorBlock

</code></pre>

<hr>
<h2 id='data_3factor_a'>A sample multi-factorial qPCR dataset</h2><span id='topic+data_3factor_a'></span>

<h3>Description</h3>

<p>A sample dataset for demonstration purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_3factor_a)
</code></pre>


<h3>Format</h3>

<p>A data frame with 36 observations and 8 variables:
</p>

<dl>
<dt>Genotype</dt><dd><p>First experimental factor</p>
</dd>
<dt>Drought</dt><dd><p>Second experimental factor</p>
</dd>
<dt>SA</dt><dd><p>Third experimental factor</p>
</dd>
<dt>Rep</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>Where the data comes from (if applicable)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_3factor_a)
data_3factor_a

</code></pre>

<hr>
<h2 id='data_3factor_b'>A three-factorial sample qPCR data</h2><span id='topic+data_3factor_b'></span>

<h3>Description</h3>

<p>A sample real time PCR data for demonstration purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_3factor_b)
</code></pre>


<h3>Format</h3>

<p>A data frame with 36 observations and 8 variables:
</p>

<dl>
<dt>Type</dt><dd><p>The first experimental factor</p>
</dd>
<dt>Conc</dt><dd><p>The second experimental factor</p>
</dd>
<dt>SA</dt><dd><p>The third experimental factor</p>
</dd>
<dt>Replicate</dt><dd><p>Biological replicates</p>
</dd>
<dt>EPO</dt><dd><p>Mean amplification efficiency of PO gene</p>
</dd>
<dt>POCt</dt><dd><p>Ct values of PO gene. Each is the mean of technical replicates</p>
</dd>
<dt>EGAPDH</dt><dd><p>Mean amplification efficiency of GAPDH gene</p>
</dd>
<dt>GAPDHCt</dt><dd><p>Ct values of GAPDH gene. Each is the mean of technical replicates</p>
</dd>
</dl>



<h3>Source</h3>

<p>University of Kurdistan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_3factor_b)
data_3factor_b


</code></pre>

<hr>
<h2 id='data_efficiency'>A sample dataset of amplification efficiency qPCR data.</h2><span id='topic+data_efficiency'></span>

<h3>Description</h3>

<p>A sample qPCR dataset for demonstrating efficiency calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_efficiency)
</code></pre>


<h3>Format</h3>

<p>A data frame with 21 observations and 3 variables:
</p>

<dl>
<dt>dilutions</dt><dd><p>Dilution factor</p>
</dd>
<dt>C2H2.26</dt><dd><p>Target gene</p>
</dd>
<dt>GAPDH</dt><dd><p>Reference gene</p>
</dd>
</dl>



<h3>Source</h3>

<p>Where the data comes from (if applicable)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_efficiency)
data_efficiency

</code></pre>

<hr>
<h2 id='data_ttest'>A sample two conditional qPCR data</h2><span id='topic+data_ttest'></span>

<h3>Description</h3>

<p>A sample data for demonstrating qPCR data analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_ttest)
</code></pre>


<h3>Format</h3>

<p>A data frame with 24 observations and 4 variables:
</p>

<dl>
<dt>Condition</dt><dd><p>Experimental conditions</p>
</dd>
<dt>E</dt><dd><p>Amplification efficiency</p>
</dd>
<dt>Gene</dt><dd><p>Genes</p>
</dd>
<dt>Ct</dt><dd><p>Ct values</p>
</dd>
</dl>



<h3>Source</h3>

<p>University of Kurdistan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_ttest)
data_ttest

</code></pre>

<hr>
<h2 id='data_ttest2'>A sample data with one target and two reference genes</h2><span id='topic+data_ttest2'></span>

<h3>Description</h3>

<p>One target and two reference gens for demonstrating qPCR data analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_ttest2)
</code></pre>


<h3>Format</h3>

<p>A data frame with 24 observations and 4 variables:
</p>

<dl>
<dt>Condition</dt><dd><p>Experimental conditions</p>
</dd>
<dt>E</dt><dd><p>Amplification efficiency</p>
</dd>
<dt>Gene</dt><dd><p>Genes</p>
</dd>
<dt>Ct</dt><dd><p>Ct values</p>
</dd>
</dl>



<h3>Source</h3>

<p>Na
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_ttest2)
data_ttest2

</code></pre>

<hr>
<h2 id='data_withTechRep'>A sample data with technical replicates</h2><span id='topic+data_withTechRep'></span>

<h3>Description</h3>

<p>A sample data for calculating biological replicated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_withTechRep)
</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations and 9 variables:
</p>

<dl>
<dt>factor1</dt><dd><p>experimental factor</p>
</dd>
<dt>factor2</dt><dd><p>experimental factor</p>
</dd>
<dt>factor3</dt><dd><p>experimental factor</p>
</dd>
<dt>biolrep</dt><dd><p>biological replicate</p>
</dd>
<dt>techrep</dt><dd><p>technical replicates</p>
</dd>
<dt>Eref</dt><dd><p>Amplification efficiency of reference gene</p>
</dd>
<dt>refCt</dt><dd><p>Ct of refence gene</p>
</dd>
<dt>Etarget</dt><dd><p>Amplification efficiency of target gene</p>
</dd>
<dt>targetCt</dt><dd><p>Ct of target gene</p>
</dd>
</dl>



<h3>Source</h3>

<p>Where the data comes from (if applicable)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_withTechRep)
data_withTechRep

</code></pre>

<hr>
<h2 id='efficiency'>efficiency, standard curves and related statistics</h2><span id='topic+efficiency'></span>

<h3>Description</h3>

<p>calculation of amplification efficiency and returns standard curves and related statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efficiency(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="efficiency_+3A_x">x</code></td>
<td>
<p>a data frame</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'efficiency' function calculates amplification efficiency of a target and a reference gene, and present the related standard curves along with the Slope, Efficiency, and R2 statistics. The function also compares the slopes of the two standard curves.
</p>


<h3>Value</h3>

<p>A list including standard curves along with the Slope, Efficiency, and R2 statistics
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Looking at the example data
data_efficiency

# Applying the efficiency function
efficiency(data_efficiency)


</code></pre>

<hr>
<h2 id='meanTech'>Calculating mean of technical replicates</h2><span id='topic+meanTech'></span>

<h3>Description</h3>

<p>Calculating of technical replicates in and output table appropriate for subsequent ANOVA analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meanTech(x, groups = c(1:4))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meanTech_+3A_x">x</code></td>
<td>
<p>A raw data frame including technical replicates.</p>
</td></tr>
<tr><td><code id="meanTech_+3A_groups">groups</code></td>
<td>
<p>grouping columns based on which the mean technical replicates are calculated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The meanTech calculates mean of technical replicates.
</p>


<h3>Value</h3>

<p>A data frame with the mean of technical replicates.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See example input data frame:
data(data_withTechRep)
data_withTechRep

# Calculating mean of technical replicates
meanTech(data_withTechRep, groups = 1:4)


</code></pre>

<hr>
<h2 id='multiplot'>Multiple plot function</h2><span id='topic+multiplot'></span>

<h3>Description</h3>

<p>Multiple plot function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiplot(..., cols = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiplot_+3A_...">...</code></td>
<td>
<p>ggplot objects can be passed in ... or to plotlist (as a list of ggplot objects)</p>
</td></tr>
<tr><td><code id="multiplot_+3A_cols">cols</code></td>
<td>
<p>Number of columns in the panel</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Producing multiple plots plate using ggplot objects
</p>


<h3>Value</h3>

<p>A  multiple-plots plate
</p>


<h3>Author(s)</h3>

<p>gist.github.com/pedroj/ffe89c67282f82c1813d
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
p1 &lt;- qpcrTTESTplot(data_ttest, 
                    numberOfrefGenes = 1,
                    ylab = "Average Fold Change (FC)",
                    width = 0.3)


out2 &lt;- qpcrANOVA(data_1factor, numberOfrefGenes = 1)$Result
p2 &lt;- oneFACTORplot(out2,
                    width = 0.2,
                    fill = "skyblue",
                    y.axis.adjust = 0.5,
                    y.axis.by = 1,
                    errorbar = "ci",
                    show.letters = TRUE,
                    letter.position.adjust = 0.1,
                    ylab = "Relative Expression (RE)",
                    xlab = "Factor Levels",
                    fontsize = 12)
                    
multiplot(p1, p2, cols=2)

multiplot(p1, p2, cols=1)



</code></pre>

<hr>
<h2 id='oneFACTORplot'>Bar plot of the relative gene expression from a one-factor experiment</h2><span id='topic+oneFACTORplot'></span>

<h3>Description</h3>

<p>Bar plot of the relative expression of a gene along with the 95% confidence interval and significance. There is another function, <code>qpcrTTESTplot</code>, that is used when the factor has two levels and represents Fold Change. <code>oneFACTORplot</code> is mainly useful for a one-factor experiment with more than two levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oneFACTORplot(
  x,
  width = 0.2,
  fill = "skyblue",
  y.axis.adjust = 0.5,
  y.axis.by = 2,
  errorbar = "std",
  show.letters = TRUE,
  letter.position.adjust = 0.1,
  ylab = "Relative Expression",
  xlab = "Factor",
  fontsize = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oneFACTORplot_+3A_x">x</code></td>
<td>
<p>a data frame created by <a href="#topic+qpcrANOVA">qpcrANOVA</a> function via running <code>res &lt;- qpcrANOVA(x)$Result</code>.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_fill">fill</code></td>
<td>
<p>specify a fill color.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive number for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>std</code> or <code>ci</code>.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_show.letters">show.letters</code></td>
<td>
<p>a logical variable. If TRUE, mean grouping letters are added to the bars.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the grouping letters to the error bars.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis.</p>
</td></tr>
<tr><td><code id="oneFACTORplot_+3A_fontsize">fontsize</code></td>
<td>
<p>size of all fonts  of the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>oneFACTORplot</code> function generates the bar plot of the average fold change for target genes along with the significance and the 95% confidence interval as error bars.
</p>


<h3>Value</h3>

<p>Bar plot of the average fold change for target genes along with the significance and the 95% confidence interval as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Before plotting, the result needs to be extracted as below:
out &lt;- qpcrANOVA(data_1factor, numberOfrefGenes = 1)$Result

# Bar plot
oneFACTORplot(out,
         width = 0.2,
         fill = "skyblue",
         y.axis.adjust = 0,
         y.axis.by = 0.2,
         errorbar = "std",
         show.letters = TRUE,
         letter.position.adjust = 0.05,
         ylab = "Relative Expression",
         xlab = "Factor Levels",
         fontsize = 12)


</code></pre>

<hr>
<h2 id='qpcrANCOVA'>ANCOVA and ANOVA on a factorial design plus bar plot of FC</h2><span id='topic+qpcrANCOVA'></span>

<h3>Description</h3>

<p>ANCOVA (analysis of covariance) and ANOVA (analysis of variance) can be performed using <code>qpcrANCOVA</code> function, if more than a factor exist. This works even if there is one factor in the experiment. Bar plot of fold changes (FCs) along with the 95% confidence interval is also returned by the <code>qpcrANCOVA</code> function. There is also a function called <code>oneFACTORplot</code> which returns FC values and related plot for a one-factor-experiment with more than two levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrANCOVA(
  x,
  numberOfrefGenes,
  block = NULL,
  analysisType = "ancova",
  main.factor,
  levels,
  level.names = "none",
  width = 0.5,
  fill = "skyblue",
  y.axis.adjust = 1,
  y.axis.by = 1,
  letter.position.adjust = 0.1,
  ylab = "Average Fold Change",
  xlab = "Pairs",
  fontsize = 12,
  p.adj = c("none", "holm", "hommel", "hochberg", "bonferroni", "BH", "BY", "fdr")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qpcrANCOVA_+3A_x">x</code></td>
<td>
<p>a data frame of condition (or conditions) levels, E (efficiency), genes and Ct values. Each Ct value in the data frame is the mean of technical replicates. Please refer to the vignette for preparing your data frame correctly.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes. Up to two reference genes can be handled.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_block">block</code></td>
<td>
<p>column name of the blocking factor (for correct column arrangement see example data.)</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_analysistype">analysisType</code></td>
<td>
<p>should be one of &quot;ancova&quot; or &quot;anova&quot;.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_main.factor">main.factor</code></td>
<td>
<p>main factor (not covariate) for which the levels FC is compared.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_levels">levels</code></td>
<td>
<p>a numeric vector corresponding to the main factor levels. First number indicates Control level to which the other levels are compared.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_level.names">level.names</code></td>
<td>
<p>a vector determining level names in the x axis on the plot.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_fill">fill</code></td>
<td>
<p>specify the fill color for the columns of the bar plot.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive value for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the signs and the error bars.</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_fontsize">fontsize</code></td>
<td>
<p>all fonts size of the plot</p>
</td></tr>
<tr><td><code id="qpcrANCOVA_+3A_p.adj">p.adj</code></td>
<td>
<p>method for adjusting p values (see <code>p.adjust</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrANCOVA</code> function applies both ANCOVA and ANOVA analysis to the data. ANCOVA is a method based on both ANOVA and linear regression. It is basically suitable when the levels of a factor are also affected by an uncontrolled quantitative covariate. 
For example, suppose that wDCt of a target gene in a plant is affected by temperature. The gene may also be affected by drought. since we already know that temperature affects the target gene, we are interesting now if the gene expression is also altered by the drought levels. We can design an experiment to understand the gene behavior at both temperature and drought levels at the same time. The drought is another factor (the covariate) that may affect the expression of our gene under the levels of the first factor i.e. temperature. The data of such an experiment can be analyzed by ANCOVA or even ANOVA based on a factorial experiment using <code>qpcrANCOVA</code> function, if more than a factor exist. Bar plot of fold changes (FCs) along with the 95% confidence interval is also returned by the <code>qpcrANCOVA</code> function. There is also a function called <code>oneFACTORplot</code> which returns FC values and related plot for a one-factor-experiment with more than two levels
</p>


<h3>Value</h3>

<p>A list with 2 elements:
</p>

<dl>
<dt>Final_data</dt><dd></dd>
<dt>lmf</dt><dd><p>lm of factorial analysis-tyle</p>
</dd>
<dt>lmc</dt><dd><p>lm of ANCOVA analysis-type</p>
</dd>
<dt>ANOVA_table</dt><dd><p>ANOVA table</p>
</dd>
<dt>ANCOVA_table</dt><dd><p>ANCOVA table</p>
</dd>
<dt>Table</dt><dd><p>Table of FC values and significance and the 95 percent CI as error bars.</p>
</dd>
<dt>plot</dt><dd><p>Bar plot of the average fold change for the main factor levels.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See sample data
data_2factor


qpcrANCOVA(data_2factor, 
           numberOfrefGenes = 1, 
           analysisType = "ancova", 
           main.factor = 1,
           levels = c(1, 2))
           

qpcrANCOVA(data_2factorBlock, 
           numberOfrefGenes = 1,
           block = "block",  
           analysisType = "ancova", 
           main.factor = 2,
           levels = c(3, 2, 1))
           
           
 qpcrANCOVA(data_1factor, 
           numberOfrefGenes = 1,
           analysisType = "ancova", 
           main.factor = 1,
           levels = 3:1)
           
           

</code></pre>

<hr>
<h2 id='qpcrANOVA'>Analysis of Variance of RE values based on CRD</h2><span id='topic+qpcrANOVA'></span>

<h3>Description</h3>

<p>Analysis of Variance relative efficiency (RE) values based on a completely randomized design (CRD). Even there are more than a factor in the experiment, it is still possible to apply CRD analysis on the factor-level combinations as treatments. Analysis of variance based on factorial design or analysis of covariance can be performed using <code>qpcrANCOVA</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrANOVA(
  x,
  numberOfrefGenes,
  block = NULL,
  p.adj = c("none", "holm", "hommel", "hochberg", "bonferroni", "BH", "BY", "fdr")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qpcrANOVA_+3A_x">x</code></td>
<td>
<p>A data frame consisting of condition columns, target gene efficiency (E), target Gene Ct, reference gene efficiency and reference gene Ct values, respectively. Each Ct in the data frame is the mean of technical replicates. Complete amplification efficiencies of 2 was assumed in the example data for all wells but the calculated efficienies can be used instead.</p>
</td></tr>
<tr><td><code id="qpcrANOVA_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes (1 or 2). Up to two reference genes can be handled.</p>
</td></tr>
<tr><td><code id="qpcrANOVA_+3A_block">block</code></td>
<td>
<p>column name of the blocking factor (for correct column arrangement see example data.)</p>
</td></tr>
<tr><td><code id="qpcrANOVA_+3A_p.adj">p.adj</code></td>
<td>
<p>Method for adjusting p values (see p.adjust)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrANOVA</code> function performs ANOVA (analysis of variance) based on a completely randomized design (CRD). 
It is suitable when relative expression (RE) analysis between different treatment combinations 
(in a Uni- or multi-factorial experiment) is desired. If there are more than a factor in the experiment, 
it is still possible to apply CRD analysis on the factor-level combinations as treatments. 
For this, a column of treatment combinations is made first as a grouping factor Fold change analysis based 
on factorial design or analysis of covariance for the can be performed using <a href="#topic+qpcrANCOVA">qpcrANCOVA</a>.
</p>


<h3>Value</h3>

<p>A list with 5 elements:
</p>

<dl>
<dt>Final_data</dt><dd><p>The row data plus weighed delta Ct (wDCt) values.</p>
</dd>
<dt>lm</dt><dd><p>The output of linear model analysis including ANOVA tables based on factorial experiment and completely randomized design (CRD).</p>
</dd>
<dt>ANOVA_factorial</dt><dd><p>ANOVA table based on factorial arrangement</p>
</dd>
<dt>ANOVA_CRD</dt><dd><p>ANOVA table based on CRD</p>
</dd>
<dt>Result</dt><dd><p>The result table including treatments and factors, RE (Relative Expression), LCL, UCL, letter grouping and standard deviation of relative expression.</p>
</dd>
<dt>Post_hoc_Test</dt><dd><p>Post hoc test of FC (Fold Change), pvalue, significance and confidence interval (LCL, UCL).</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method. Methods 25 (4). doi:10.1006/meth.2001.1262.
</p>
<p>Ganger, MT, Dietz GD, and Ewing SJ. 2017. A common base method for analysis of qPCR data
and the application of simple blocking in qPCR experiments. BMC bioinformatics 18, 1-11.
</p>
<p>Yuan, Joshua S, Ann Reed, Feng Chen, and Neal Stewart. 2006.
Statistical Analysis of Real-Time PCR Data. BMC Bioinformatics 7 (85). doi:10.1186/1471-2105-7-85.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# If the data include technical replicates, means of technical replicates
# should be calculated first using meanTech function.

# Applying ANOVA analysis
qpcrANOVA(
     data_3factor_a,
     numberOfrefGenes = 1,
     p.adj = "none")


qpcrANOVA(
    data_2factorBlock,
    block = "Block",
    numberOfrefGenes = 1)


</code></pre>

<hr>
<h2 id='qpcrTTEST'>Analysis of the fold change (FC) of target genes assessed under two conditions.</h2><span id='topic+qpcrTTEST'></span>

<h3>Description</h3>

<p>t.test based analysis to any number of Genes to calculate the average fold change for target Genes along with the 95 percent confidence interval and significance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrTTEST(x, numberOfrefGenes, paired = FALSE, var.equal = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qpcrTTEST_+3A_x">x</code></td>
<td>
<p>a data frame of 4 columns including Conditions, E (efficiency), Gene and Ct values (see example below). Biological replicates needs to be equal for all Genes. Each Ct value is the mean of technical replicates. Complete amplification efficiencies of 2 is assumed here for all wells but the calculated efficienies can be used instead.</p>
</td></tr>
<tr><td><code id="qpcrTTEST_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes. Up to two reference genes can be handled.</p>
</td></tr>
<tr><td><code id="qpcrTTEST_+3A_paired">paired</code></td>
<td>
<p>a logical indicating whether you want a paired t-test.</p>
</td></tr>
<tr><td><code id="qpcrTTEST_+3A_var.equal">var.equal</code></td>
<td>
<p>a logical variable indicating whether to treat the two variances as being equal. If TRUE then the pooled variance is used to estimate the variance otherwise the Welch (or Satterthwaite) approximation to the degrees of freedom is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrTTEST</code> function applies a t.test based analysis to any number of Genes along with one reference Gene, that have been evaluated under control and treatment conditions. When a series of target Genes is assessed under only two conditions, the average fold change expression can be calculated for each Gene.
</p>


<h3>Value</h3>

<p>A list of two elements:
</p>

<dl>
<dt>Row_data</dt><dd><p>The row data including Genes and weighed delta Ct (wDCt) values.</p>
</dd>
<dt>Result</dt><dd><p>Output table including the Fold Change values, lower and upper confidence interval and the pvalues from compairing fold change between treated and non-treated conditions</p>
</dd>
</dl>

<p>For more information about the test procedure and its arguments,
refer <code><a href="stats.html#topic+t.test">t.test</a></code>, and <code><a href="stats.html#topic+lm">lm</a></code>.
If the residuals of the model do not follow normal distribution and variances between the two groups are not homoGene, <code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code> procedure may be concidered
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method. Methods 25 (4). doi:10.1006/meth.2001.1262.
</p>
<p>Ganger, MT, Dietz GD, and Ewing SJ. 2017. A common base method for analysis
of qPCR data and the application of simple blocking in qPCR experiments.
BMC bioinformatics 18, 1-11.
</p>
<p>Yuan, Joshua S, Ann Reed, Feng Chen, and Neal Stewart. 2006.
Statistical Analysis of Real-Time PCR Data. BMC Bioinformatics 7 (85).
doi:10.1186/1471-2105-7-85.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See the sample data structure
data_ttest

# Getting t.test results
qpcrTTEST(data_ttest,
   paired = FALSE,
   var.equal = TRUE,
   numberOfrefGenes = 1)


</code></pre>

<hr>
<h2 id='qpcrTTESTplot'>Bar plot of the average fold change of the target genes</h2><span id='topic+qpcrTTESTplot'></span>

<h3>Description</h3>

<p>Bar plot of the average fold change (FC) of the target genes along with the 95 percent CI and significance in a two-level experimental condition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qpcrTTESTplot(
  x,
  order = default.order,
  numberOfrefGenes,
  paired = FALSE,
  var.equal = TRUE,
  width = 0.5,
  fill = "skyblue",
  y.axis.adjust = 0,
  y.axis.by = 2,
  letter.position.adjust = 0.3,
  ylab = "Average Fold Change",
  xlab = "Gene",
  fontsize = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qpcrTTESTplot_+3A_x">x</code></td>
<td>
<p>a data frame. The data frame consists of 4 columns belonging to condition levels, E (efficiency), genes and Ct values, respectively. Each Ct in the following data frame is the mean of technical replicates. Complete amplification efficiencies of 2 is assumed here for all wells but the calculated efficienies can be used we well. We use this data set for Fold Change expression analysis of the target genes in treatment condition compared to normal condition.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_order">order</code></td>
<td>
<p>a vector determinig genes order on the output graph.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_numberofrefgenes">numberOfrefGenes</code></td>
<td>
<p>number of reference genes. Up to two reference genes can be handled.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_paired">paired</code></td>
<td>
<p>a logical indicating whether you want a paired t-test.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_var.equal">var.equal</code></td>
<td>
<p>a logical variable indicating whether to treat the two variances as being equal. If TRUE then the pooled variance is used to estimate the variance otherwise the Welch (or Satterthwaite) approximation to the degrees of freedom is used.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_fill">fill</code></td>
<td>
<p>specify the fill color for the columns of the bar plot.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive value for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the signs and the error bars.</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis</p>
</td></tr>
<tr><td><code id="qpcrTTESTplot_+3A_fontsize">fontsize</code></td>
<td>
<p>all fonts size of the plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>qpcrTTESTplot</code> function applies a t.test based analysis to any number of genes along with one reference gene, that have been evaluated under control and treatment conditions, and returns the bar plot of the average fold change for target genes along with the 95% CI and significance.
</p>


<h3>Value</h3>

<p>Bar  plot of the average fold change for target genes along with the significance and the 95 percent CI as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See a sample data frame
data_ttest


qpcrTTESTplot(data_ttest, 
              numberOfrefGenes = 1)


# Producing the plot
qpcrTTESTplot(data_ttest,
              numberOfrefGenes = 1,
              order = c("C2H2-01", "C2H2-12", "C2H2-26"),
              paired = FALSE,
              var.equal = TRUE,
              width = 0.5,
              fill = "skyblue",
              y.axis.adjust = 0,
              y.axis.by = 2,
              letter.position.adjust = 0.3,
              ylab = "Average Fold Change in Treatment vs Control",
              xlab = "Gene")


</code></pre>

<hr>
<h2 id='threeFACTORplot'>Bar plot of the relative gene expression (RE) from a three-factor experiment</h2><span id='topic+threeFACTORplot'></span>

<h3>Description</h3>

<p>Bar plot of the relative expression (RE) of a gene along with the confidence interval and significance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threeFACTORplot(
  x,
  arrangement = c(1, 2, 3),
  bar.width = 0.5,
  fill = "Reds",
  xlab = "Gene",
  ylab = "Relative Expression",
  errorbar = "std",
  y.axis.adjust = 0.5,
  y.axis.by = 2,
  letter.position.adjust = 0.3,
  legend.title = "Legend Title",
  legend.position = c(0.4, 0.8),
  fontsize = 12,
  show.letters = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="threeFACTORplot_+3A_x">x</code></td>
<td>
<p>a data frame created by <a href="#topic+qpcrANOVA">qpcrANOVA</a> function via running <code>res &lt;- qpcrANOVA(x)$Result</code> on a three factorial data such as <code>datathreegr</code> example data.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_arrangement">arrangement</code></td>
<td>
<p>arrangement of the grouping columns in the output graph, for example c(2,3,1) or c(1,3,2). This affects grapg output.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_bar.width">bar.width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_fill">fill</code></td>
<td>
<p>a color vector specifying the fill color for the columns of the bar plot. One of the palettes in <code><a href="RColorBrewer.html#topic+display.brewer.all">display.brewer.all</a></code> (e.g. &quot;Reds&quot; or &quot;Blues&quot;, ...) can be applied.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>std</code> or <code>ci</code>.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive number for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the grouping letters to the error bars</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_legend.position">legend.position</code></td>
<td>
<p>a two digit vector specifying the legend position.</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_fontsize">fontsize</code></td>
<td>
<p>all fonts size of the plot</p>
</td></tr>
<tr><td><code id="threeFACTORplot_+3A_show.letters">show.letters</code></td>
<td>
<p>a logical variable. If TRUE, mean grouping letters are added to the bars.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>threeFACTORplot</code> function generates the bar plot of the average fold change for target genes along with the significance and the 95% confidence interval as error bars.
</p>


<h3>Value</h3>

<p>Bar plot of the average fold change for target genes along with the significance and the 95% confidence interval as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#' # See a sample data frame
data_3factor_b

# Before plotting, the result needs to be extracted as below:
res &lt;- qpcrANOVA(data_3factor_b, numberOfrefGenes = 1)$Result
res

# Arrange the first three colunms of the result table.
# This determines the columns order and shapes the plot output.
threeFACTORplot(res,
    arrangement = c(3, 1, 2),
    xlab = "condition")



threeFACTORplot(res,
   arrangement = c(1, 2, 3),
   bar.width = 0.5,
   fill = "Greys",
   xlab = "Genotype",
   ylab = "Relative Expression")



# Reordering factor levels to a desired order.
res$Conc &lt;- factor(res$Conc, levels = c("L","M","H"))
res$Type &lt;- factor(res$Type, levels = c("S","R"))

# Producing the plot
threeFACTORplot(res,
   arrangement = c(2, 3, 1),
   bar.width = 0.5,
   fill = "Reds",
   xlab = "Drought",
   ylab = "Relative Expression",
   errorbar = "std",
   legend.title = "Genotype",
   legend.position = c(0.2, 0.8))


# When using ci as error, increase the 
# y.axis.adjust value to see the plot correctly!
threeFACTORplot(res,
   arrangement = c(2, 3, 1),
   bar.width = 0.8,
   fill = "Greens",
   xlab = "Drought",
   ylab = "Relative Expression",
   errorbar = "ci",
   y.axis.adjust = 8,
   y.axis.by = 2,
   letter.position.adjust = 0.6,
   legend.title = "Genotype",
   fontsize = 12,
   legend.position = c(0.2, 0.8),
   show.letters = TRUE)



</code></pre>

<hr>
<h2 id='twoFACTORplot'>Bar plot of the relative gene expression (RE) from a two-factor experiment</h2><span id='topic+twoFACTORplot'></span>

<h3>Description</h3>

<p>Bar plot of the relative expression (RE) of a gene along with the 95% confidence interval and significance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twoFACTORplot(
  x,
  x.axis.factor,
  group.factor,
  width = 0.5,
  fill = "Blues",
  y.axis.adjust = 0.5,
  y.axis.by = 2,
  show.errorbars = TRUE,
  errorbar = "std",
  show.letters = TRUE,
  show.points = FALSE,
  letter.position.adjust = 0.1,
  ylab = "Relative Expression",
  xlab = "Gene",
  legend.position = c(0.09, 0.8),
  fontsize = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twoFACTORplot_+3A_x">x</code></td>
<td>
<p>an object created by <a href="#topic+qpcrANOVA">qpcrANOVA</a> function via running <code>res &lt;- qpcrANOVA(x)</code>.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_x.axis.factor">x.axis.factor</code></td>
<td>
<p>x-axis factor.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_group.factor">group.factor</code></td>
<td>
<p>grouping factor.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_width">width</code></td>
<td>
<p>a positive number determining bar width.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_fill">fill</code></td>
<td>
<p>specify the fill color vector for the columns of the bar plot. One of the palettes in <code><a href="RColorBrewer.html#topic+display.brewer.all">display.brewer.all</a></code> (e.g. &quot;Reds&quot; or &quot;Blues&quot;, ...) can be applied.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_y.axis.adjust">y.axis.adjust</code></td>
<td>
<p>a negative or positive number for reducing or increasing the length of the y axis.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_y.axis.by">y.axis.by</code></td>
<td>
<p>determines y axis step length.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_show.errorbars">show.errorbars</code></td>
<td>
<p>show errorbars</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_errorbar">errorbar</code></td>
<td>
<p>Type of error bar, can be <code>std</code> or <code>ci</code>.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_show.letters">show.letters</code></td>
<td>
<p>a logical variable. If TRUE, mean grouping letters are added to the bars.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_show.points">show.points</code></td>
<td>
<p>show points</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_letter.position.adjust">letter.position.adjust</code></td>
<td>
<p>adjust the distance between the grouping letters to the error bars.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_ylab">ylab</code></td>
<td>
<p>the title of the y axis.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_xlab">xlab</code></td>
<td>
<p>the title of the x axis.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_legend.position">legend.position</code></td>
<td>
<p>a two digit vector specifying the legend position.</p>
</td></tr>
<tr><td><code id="twoFACTORplot_+3A_fontsize">fontsize</code></td>
<td>
<p>size of all fonts  of the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>twoFACTORplot</code> function generates the bar plot of the average fold change for target genes along with the significance and the 95% confidence interval as error bars.
</p>


<h3>Value</h3>

<p>Bar plot of the average fold change for target genes along with the significance and the 95% confidence interval as error bars.
</p>


<h3>Author(s)</h3>

<p>Ghader Mirzaghaderi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See a sample data frame
data_2factor

# Before generating plot, the result table needs to be extracted as below:
res &lt;- qpcrANOVA(data_2factor, numberOfrefGenes = 1)
res

# Plot of the 'res' data with 'Genotype' as grouping factor
twoFACTORplot(res,
   x.axis.factor = Drought,
   group.factor = Genotype,
   width = 0.5,
   fill = "Greens",
   y.axis.adjust = 1,
   y.axis.by = 2,
   ylab = "Relative Expression",
   xlab = "Drought Levels",
   show.letters = TRUE, 
   letter.position.adjust = 0.3,
   legend.position = c(0.09, 0.8),
   errorbar = "ci")

# Plotting the same data with 'Drought' as grouping factor
twoFACTORplot(res,
   x.axis.factor = Genotype,
   group.factor = Drought,
   xlab = "Genotype",
   fill = "Blues",
   fontsize = 12,
   show.letters = FALSE,
   show.points = TRUE,
   show.errorbars = FALSE)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
