<!DOCTYPE html><html><head><title>Help for package Rtropical</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Rtropical}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#all_trees'><p>Simulated Tree Data with Different Proximity Parameter Value</p></a></li>
<li><a href='#apicomplexa'><p>Apicomplexa gene trees sample data set.</p></a></li>
<li><a href='#as.matrix.multiPhylo'><p>Vectorize a Set of Phylognetic Trees</p></a></li>
<li><a href='#as.vector.phylo'><p>Vectorize a Phylogenetic Tree</p></a></li>
<li><a href='#assignment_finder'><p>Efficient Finder of Best Assignment</p></a></li>
<li><a href='#coef.cv.tropsvm'><p>Extract Optimal Tropical Hyperplane from a cv.tropsvm object</p></a></li>
<li><a href='#coef.tropsvm'><p>Extract Optimal Tropical Hyperplane from a tropsvm object</p></a></li>
<li><a href='#cv.tropsvm'><p>Cross-Validation for Tropical Support Vector Machines</p></a></li>
<li><a href='#dets'><p>Tropical Determinants for Small Matrices</p></a></li>
<li><a href='#linsp_to_poly'><p>Convert Tropical Linear Space to Convex Hull</p></a></li>
<li><a href='#lungfish'><p>Coelacanths genome and transcriptome data</p></a></li>
<li><a href='#plot.troppca'><p>Plot the Tropical Principal Components with Data Projections</p></a></li>
<li><a href='#polytope_iso'><p>Dimension Reduction by Isometry</p></a></li>
<li><a href='#predict.cv.tropsvm'><p>Predict Method for Tropical Support Vector Machines based on Cross-Validation</p></a></li>
<li><a href='#predict.tropsvm'><p>Predict Method for Tropical Support Vector Machines</p></a></li>
<li><a href='#read.nexus.to.data.matrix'><p>Read NEXUS-formatted trees from two categories into a data matrix</p></a></li>
<li><a href='#read.tree.to.data.matrix'><p>Read Newick-formatted trees in two categories into a data matrix</p></a></li>
<li><a href='#sim_trees'><p>Simulated Tree Data</p></a></li>
<li><a href='#summary.cv.tropsvm'><p>Summarize an Analysis of Cross-Validated Tropical Support Vector Machine</p></a></li>
<li><a href='#tropdet'><p>Tropical Determinant of a Matrix</p></a></li>
<li><a href='#tropFW'><p>Tropical Fermat-Weber Point</p></a></li>
<li><a href='#troppca.linsp'><p>Tropical Principal Component Analysis by Tropical Linear Space</p></a></li>
<li><a href='#troppca.linsp2poly'><p>Tropical Principal Component Analysis by Polytope Converted from Linear Space</p></a></li>
<li><a href='#troppca.obj'><p>Compute Tropical PCA Objective</p></a></li>
<li><a href='#troppca.poly'><p>Tropical Principal Component Analysis by Tropical Polytope</p></a></li>
<li><a href='#tropproj.linsp'><p>Projection on Tropical Linear Space</p></a></li>
<li><a href='#tropproj.poly'><p>Projection on Tropical Polytope</p></a></li>
<li><a href='#tropseg'><p>Compute Tropical Line Segment</p></a></li>
<li><a href='#tropsvm'><p>Tropical Support Vector Machines</p></a></li>
<li><a href='#tropsvm_helper'><p>Helper for <code>cv.tropsvm</code></p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Data Analysis Tools over Space of Phylogenetic Trees Using
Tropical Geometry</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Houjie Wang &lt;wanghoujie6688@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Process phylogenetic trees with tropical support vector machine and principal component analysis defined with tropical geometry. Details about tropical support vector machine are available in : Tang, X., Wang, H. &amp; Yoshida, R. (2020) &lt;<a href="https://arxiv.org/abs/2003.00677">arXiv:2003.00677</a>&gt;. Details about tropical principle component analysis are available in : Page, R., Yoshida, R. &amp; Zhang L. (2020) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtaa564">doi:10.1093/bioinformatics/btaa564</a>&gt; and Yoshida, R., Zhang, L. &amp; Zhang, X. (2019) &lt;<a href="https://doi.org/10.1007%2Fs11538-018-0493-4">doi:10.1007/s11538-018-0493-4</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, lpSolve, lpSolveAPI, parallel, Rfast, RcppAlgos, caret</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, e1071, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/HoujieWang/Rtropical">https://github.com/HoujieWang/Rtropical</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-09 12:08:20 UTC; wanghoujie</td>
</tr>
<tr>
<td>Author:</td>
<td>Houjie Wang [aut, cre],
  Kaizhang Wang [aut],
  Grady Weyenberg [aut],
  Xiaoxian Tang [aut],
  Ruriko Yoshida [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-09 18:50:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='all_trees'>Simulated Tree Data with Different Proximity Parameter Value</h2><span id='topic+all_trees'></span>

<h3>Description</h3>

<p>Simulated Tree Data with Different Proximity Parameter Value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(all_trees)
</code></pre>


<h3>Format</h3>

<p>A list of length 12 with each element a sublist containing an
ape <code>multiPhylo</code> object with 300 rooted trees on 5 tips and the tree categories.
</p>

<hr>
<h2 id='apicomplexa'>Apicomplexa gene trees sample data set.</h2><span id='topic+apicomplexa'></span>

<h3>Description</h3>

<p>Apicomplexa gene trees sample data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apicomplexa
</code></pre>


<h3>Format</h3>

<p>An ape <code>multiPhylo</code> object with 268 rooted trees on 8 tips.
</p>


<h3>Source</h3>

<p>Chih-Horng Kuo, John P. Wares, Jessica C. Kissinger.
<em>The Apicomplexan Whole-Genome Phylogeny: An Analysis of Incongruence among Gene Trees
Molecular Biology and Evolution, Volume 25, Issue 12, December 2008, Pages 2689–2698.</em>
</p>

<hr>
<h2 id='as.matrix.multiPhylo'>Vectorize a Set of Phylognetic Trees</h2><span id='topic+as.matrix.multiPhylo'></span>

<h3>Description</h3>

<p>Unifies tip labels of all phylogenetic trees in <code>multiPhylo</code>
object the same as the first tree and returns the
cophenetic distance of their corresponding chronogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'multiPhylo'
as.matrix(x, tipOrder = x[[1]]$tip.label, parallel = FALSE, ncores = 2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.matrix.multiPhylo_+3A_x">x</code></td>
<td>
<p>an object of class <code>multiPhylo</code></p>
</td></tr>
<tr><td><code id="as.matrix.multiPhylo_+3A_tiporder">tipOrder</code></td>
<td>
<p>a numeric vector of order of leaf names to which all trees
in the <code>multiPhylo</code> object will unified. If not specified on purpose,
the tip order of the first tree will be used.</p>
</td></tr>
<tr><td><code id="as.matrix.multiPhylo_+3A_parallel">parallel</code></td>
<td>
<p>a logical value indicating if parallel computing should be used. (default: FALSE)</p>
</td></tr>
<tr><td><code id="as.matrix.multiPhylo_+3A_ncores">ncores</code></td>
<td>
<p>a numeric value indicating the number of threads
utilized for multi-cored CPUs. (default: 2)</p>
</td></tr>
<tr><td><code id="as.matrix.multiPhylo_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to as.vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data matrix with each row a vector representation of a chronogram. Each element of the vector is the distance between two leaves.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(apicomplexa)
data &lt;- as.matrix(apicomplexa[1: 10]) # matrixize first ten trees

</code></pre>

<hr>
<h2 id='as.vector.phylo'>Vectorize a Phylogenetic Tree</h2><span id='topic+as.vector.phylo'></span>

<h3>Description</h3>

<p>Computes the cophenetic distance and outputs them in a vector
of a phylogenetic tree in <code>phylo</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylo'
as.vector(x, mode = "any")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.vector.phylo_+3A_x">x</code></td>
<td>
<p>A object of class phylo</p>
</td></tr>
<tr><td><code id="as.vector.phylo_+3A_mode">mode</code></td>
<td>
<p>The same as <code>base::as.vector</code>. But only numeric
output in vector form is accepted for other functions in <code>Rtropical</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with its elements the distance between two leaves of the tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ape)
tree &lt;- rcoal(5)
tree_vec &lt;- as.vector(tree)

</code></pre>

<hr>
<h2 id='assignment_finder'>Efficient Finder of Best Assignment</h2><span id='topic+assignment_finder'></span>

<h3>Description</h3>

<p>Efficient Finder of Best Assignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assignment_finder(P, Q, t = 25)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assignment_finder_+3A_p">P</code></td>
<td>
<p>a data matrix of the first category.</p>
</td></tr>
<tr><td><code id="assignment_finder_+3A_q">Q</code></td>
<td>
<p>a data matrix of the second category.</p>
</td></tr>
<tr><td><code id="assignment_finder_+3A_t">t</code></td>
<td>
<p>a numeric value controlling the number of selected best assignments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with each row a unique assignment, starting from the best assignment
and ending with the worst.
</p>

<hr>
<h2 id='coef.cv.tropsvm'>Extract Optimal Tropical Hyperplane from a cv.tropsvm object</h2><span id='topic+coef.cv.tropsvm'></span>

<h3>Description</h3>

<p>Obtain the optimal tropical hyperplane in the form of vectors from a cv.tropsvm object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cv.tropsvm'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.cv.tropsvm_+3A_object">object</code></td>
<td>
<p>a fitted <code>"cv.tropsvm"</code> object.</p>
</td></tr>
<tr><td><code id="coef.cv.tropsvm_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An output of the apex of the fitted optimal tropical hyperplane.
</p>

<hr>
<h2 id='coef.tropsvm'>Extract Optimal Tropical Hyperplane from a tropsvm object</h2><span id='topic+coef.tropsvm'></span>

<h3>Description</h3>

<p>Obtain the optimal tropical hyperplane in the form of vectors from a tropsvm object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tropsvm'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.tropsvm_+3A_object">object</code></td>
<td>
<p>a fitted <code>"tropsvm"</code> object.</p>
</td></tr>
<tr><td><code id="coef.tropsvm_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An output of the apex of the fitted optimal tropical hyperplane.
</p>

<hr>
<h2 id='cv.tropsvm'>Cross-Validation for Tropical Support Vector Machines</h2><span id='topic+cv.tropsvm'></span>

<h3>Description</h3>

<p>Conduct k-fold cross validation for tropsvm and return an object <code>"cv.tropsvm"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv.tropsvm(x, y, parallel = FALSE, nfold = 10, nassignment = 10, ncores = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv.tropsvm_+3A_x">x</code></td>
<td>
<p>a data matrix, of dimension nobs x nvars; each row is an observation vector.</p>
</td></tr>
<tr><td><code id="cv.tropsvm_+3A_y">y</code></td>
<td>
<p>a response vector with one label for each row/component of x.</p>
</td></tr>
<tr><td><code id="cv.tropsvm_+3A_parallel">parallel</code></td>
<td>
<p>a logical value indicating if parallel computing should be used. (default: FALSE)</p>
</td></tr>
<tr><td><code id="cv.tropsvm_+3A_nfold">nfold</code></td>
<td>
<p>a numeric value of the number of data folds for cross-validation. (default: 10)</p>
</td></tr>
<tr><td><code id="cv.tropsvm_+3A_nassignment">nassignment</code></td>
<td>
<p>a numeric value indicating the size of the parameter grid of assignments. (default: 10)</p>
</td></tr>
<tr><td><code id="cv.tropsvm_+3A_ncores">ncores</code></td>
<td>
<p>a numeric value indicating the number of threads utilized for multi-cored CPUs. (default: 2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object with S3 class <code>cv.tropsvm</code> containing the fitted model, including:
</p>
<table>
<tr><td><code>apex</code></td>
<td>
<p>The negative apex of the fitted optimal tropical hyperplane.</p>
</td></tr>
<tr><td><code>assignment</code></td>
<td>
<p>The best <code>assignment</code> tuned by cross-validation.</p>
</td></tr>
<tr><td><code>index</code></td>
<td>
<p>The best classification method tuned by cross-validation.</p>
</td></tr>
<tr><td><code>levels</code></td>
<td>
<p>The name of each category, consistent with categories in <code>y</code>.</p>
</td></tr>
<tr><td><code>accuracy</code></td>
<td>
<p>The validation accuracy for each fold.</p>
</td></tr>
<tr><td><code>nfold</code></td>
<td>
<p>The number of folds used in cross-validation.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code>summary</code>, <code>predict</code>, <code>coef</code> and the <code>tropsvm</code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# data generation
library(Rfast)
set.seed(101)
e &lt;- 20
n &lt;- 10
N &lt;- 10
s &lt;- 5
x &lt;- rbind(
  rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(n, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
y &lt;- as.factor(c(rep(1, n), rep(2, n)))
newx &lt;- rbind(
  rmvnorm(N, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(N, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
newy &lt;- as.factor(rep(c(1, 2), each = N))

# train the tropical svm
cv_tropsvm_fit &lt;- cv.tropsvm(x, y, parallel = FALSE)

summary(cv_tropsvm_fit)
coef(cv_tropsvm_fit)

# test with new data
pred &lt;- predict(cv_tropsvm_fit, newx)

# check with accuracy
table(pred, newy)

# compute testing accuracy
sum(pred == newy) / length(newy)
</code></pre>

<hr>
<h2 id='dets'>Tropical Determinants for Small Matrices</h2><span id='topic+dets'></span><span id='topic+trop_det2d'></span><span id='topic+trop_det3d'></span><span id='topic+trop_det4d'></span><span id='topic+trop_det5d'></span>

<h3>Description</h3>

<p>Tropical Determinants for Small Matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop_det2d(X)

trop_det3d(X)

trop_det4d(X)

trop_det5d(X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dets_+3A_x">X</code></td>
<td>
<p>a square matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Tropical determinant of a given matrix
</p>

<hr>
<h2 id='linsp_to_poly'>Convert Tropical Linear Space to Convex Hull</h2><span id='topic+linsp_to_poly'></span>

<h3>Description</h3>

<p>Convert Tropical Linear Space to Convex Hull
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linsp_to_poly(V)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linsp_to_poly_+3A_v">V</code></td>
<td>
<p>a data matrix, of dimension s x e, with each row a basis of tropical linear space.
e is the dimension of the tropical space and s is the dimension of the linear space.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the return is a e choose s-1 x e matrix.
</p>

<hr>
<h2 id='lungfish'>Coelacanths genome and transcriptome data</h2><span id='topic+lungfish'></span>

<h3>Description</h3>

<p>Coelacanths genome and transcriptome data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lungfish
</code></pre>


<h3>Format</h3>

<p>An ape <code>multiPhylo</code> object with 1193 rooted trees on 10 tips.
</p>


<h3>Source</h3>

<p>Tom M. W. Nye, Xiaoxian Tang, Grady Weyenberg and Ruriko Yoshida.
<em>Principal component analysis and the locus of the Fréchet mean in the space of phylogenetic trees,
Biometrika, Volume 104, Issue 4, December 2017, Pages 901–922.</em>
</p>

<hr>
<h2 id='plot.troppca'>Plot the Tropical Principal Components with Data Projections</h2><span id='topic+plot.troppca'></span>

<h3>Description</h3>

<p>Visualize the second order tropical principle components in <code>troppca</code>
as a tropical triangle with projections on a two-dimensional plot via tropical isometry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'troppca'
plot(x, plab = NULL, fw = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.troppca_+3A_x">x</code></td>
<td>
<p>a fitted <code>troppca</code> object.</p>
</td></tr>
<tr><td><code id="plot.troppca_+3A_plab">plab</code></td>
<td>
<p>a vector of labels of all points in the given data matrix.
Not needed for unlabeled data. (default: NULL)</p>
</td></tr>
<tr><td><code id="plot.troppca_+3A_fw">fw</code></td>
<td>
<p>a logical variable to determine if to add
Fermat-Weber point of the data projection. (default: FALSE)</p>
</td></tr>
<tr><td><code id="plot.troppca_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>plot.troppca</code> does not return anything other than the plot.
</p>

<hr>
<h2 id='polytope_iso'>Dimension Reduction by Isometry</h2><span id='topic+polytope_iso'></span>

<h3>Description</h3>

<p>Dimension Reduction by Isometry
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polytope_iso(D, P)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="polytope_iso_+3A_d">D</code></td>
<td>
<p>a data matrix of size s x e whose rows are vertices of a tropical polytope.</p>
</td></tr>
<tr><td><code id="polytope_iso_+3A_p">P</code></td>
<td>
<p>a vector of a point in row spaces of <code>D</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of a point corresponding to <code>P</code> in column span of <code>D</code>.
</p>

<hr>
<h2 id='predict.cv.tropsvm'>Predict Method for Tropical Support Vector Machines based on Cross-Validation</h2><span id='topic+predict.cv.tropsvm'></span>

<h3>Description</h3>

<p>Predicts values based upon a model trained by <code>cv.tropsvm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cv.tropsvm'
predict(object, newx, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.cv.tropsvm_+3A_object">object</code></td>
<td>
<p>a fitted <code>"cv.tropsvm"</code> object.</p>
</td></tr>
<tr><td><code id="predict.cv.tropsvm_+3A_newx">newx</code></td>
<td>
<p>a data matrix, of dimension nobs x nvars used as testing data.</p>
</td></tr>
<tr><td><code id="predict.cv.tropsvm_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of predicted values of a vector of labels.
</p>


<h3>See Also</h3>

<p><code>summary</code>, <code>coef</code> and the <code>cv.tropsvm</code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# data generation
library(Rfast)
e &lt;- 20
n &lt;- 10
N &lt;- 10
s &lt;- 5
x &lt;- rbind(
  rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(n, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
y &lt;- as.factor(c(rep(1, n), rep(2, n)))
newx &lt;- rbind(
  rmvnorm(N, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(N, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
newy &lt;- as.factor(rep(c(1, 2), each = N))

# train the tropical svm
cv_tropsvm_fit &lt;- cv.tropsvm(x, y, parallel = FALSE)

# test with new data
pred &lt;- predict(cv_tropsvm_fit, newx)

# check with accuracy
table(pred, newy)

# compute testing accuracy
sum(pred == newy) / length(newy)
</code></pre>

<hr>
<h2 id='predict.tropsvm'>Predict Method for Tropical Support Vector Machines</h2><span id='topic+predict.tropsvm'></span>

<h3>Description</h3>

<p>Predicts values based upon a model trained by <code>tropsvm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tropsvm'
predict(object, newx, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.tropsvm_+3A_object">object</code></td>
<td>
<p>a fitted <code>tropsvm</code> object.</p>
</td></tr>
<tr><td><code id="predict.tropsvm_+3A_newx">newx</code></td>
<td>
<p>a data matrix, of dimension nobs x nvars used as testing data.</p>
</td></tr>
<tr><td><code id="predict.tropsvm_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of predicted values of a vector of labels.
</p>


<h3>See Also</h3>

<p><code>summary</code>, <code>coef</code> and the <code>tropsvm</code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# data generation
library(Rfast)
e &lt;- 100
n &lt;- 10
N &lt;- 10
s &lt;- 5
x &lt;- rbind(
  rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(n, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
y &lt;- as.factor(c(rep(1, n), rep(2, n)))
newx &lt;- rbind(
  rmvnorm(N, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(N, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
newy &lt;- as.factor(rep(c(1, 2), each = N))

# train the tropical svm
tropsvm_fit &lt;- tropsvm(x, y, auto.assignment = TRUE, ind = 1)

# test with new data
pred &lt;- predict(tropsvm_fit, newx)

# check with accuracy
table(pred, newy)

# compute testing accuracy
sum(pred == newy) / length(newy)
</code></pre>

<hr>
<h2 id='read.nexus.to.data.matrix'>Read NEXUS-formatted trees from two categories into a data matrix</h2><span id='topic+read.nexus.to.data.matrix'></span>

<h3>Description</h3>

<p>Read NEXUS-formatted trees from two categories into a data matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.nexus.to.data.matrix(data.file1, data.file2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.nexus.to.data.matrix_+3A_data.file1">data.file1</code></td>
<td>
<p>A data set with trees from one category.</p>
</td></tr>
<tr><td><code id="read.nexus.to.data.matrix_+3A_data.file2">data.file2</code></td>
<td>
<p>A data set with trees from the other category.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data matrix with the first x rows corresponding the x trees in the first file and the last y rows are the trees from the second file.
</p>

<hr>
<h2 id='read.tree.to.data.matrix'>Read Newick-formatted trees in two categories into a data matrix</h2><span id='topic+read.tree.to.data.matrix'></span>

<h3>Description</h3>

<p>Read Newick-formatted trees in two categories into a data matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.tree.to.data.matrix(data.file1, data.file2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.tree.to.data.matrix_+3A_data.file1">data.file1</code></td>
<td>
<p>A file containing trees in Newick form in a category.</p>
</td></tr>
<tr><td><code id="read.tree.to.data.matrix_+3A_data.file2">data.file2</code></td>
<td>
<p>A file containing trees in Newick form in an assumed different category.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>read.tree.to.data.matrix</code> has the same return as <code>read.nexus.to.data.matrix</code>.
</p>

<hr>
<h2 id='sim_trees'>Simulated Tree Data</h2><span id='topic+sim_trees'></span>

<h3>Description</h3>

<p>Simulated Tree Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sim_trees)
</code></pre>


<h3>Format</h3>

<p>An ape <code>multiPhylo</code> object with 300 rooted trees on 5 tips.
</p>

<hr>
<h2 id='summary.cv.tropsvm'>Summarize an Analysis of Cross-Validated Tropical Support Vector Machine</h2><span id='topic+summary.cv.tropsvm'></span>

<h3>Description</h3>

<p>Return a summary with a more detailed explanation of the object <code>"cv.tropsvm"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cv.tropsvm'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.cv.tropsvm_+3A_object">object</code></td>
<td>
<p>a fitted <code>"cv.tropsvm"</code> object.</p>
</td></tr>
<tr><td><code id="summary.cv.tropsvm_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to summary.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A summary of the crucial information of a tropical support vector machine is printed, including
the selected best assignment and classification methods and the validation accuracy of each data fold. The
summary section of classification methods specifies the sectors and their intersections used to classify
points of two different categories.
</p>


<h3>See Also</h3>

<p><code>predict</code>, <code>coef</code> and the <code>cv.tropsvm</code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# data generation
library(Rfast)
e &lt;- 20
n &lt;- 10
N &lt;- 10
s &lt;- 5
x &lt;- rbind(
  rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(n, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
y &lt;- as.factor(c(rep(1, n), rep(2, n)))
newx &lt;- rbind(
  rmvnorm(N, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(N, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
newy &lt;- as.factor(rep(c(1, 2), each = N))

# train the tropical svm with cross-validation
cv_tropsvm_fit &lt;- cv.tropsvm(x, y, parallel = FALSE)

summary(cv_tropsvm_fit)
</code></pre>

<hr>
<h2 id='tropdet'>Tropical Determinant of a Matrix</h2><span id='topic+tropdet'></span>

<h3>Description</h3>

<p>Compute the tropical determinant for a given matrix. This is equivalent to
solving an assignment problem.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropdet(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropdet_+3A_x">x</code></td>
<td>
<p>a square matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The determinant of the given matrix,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>R &lt;- matrix(sample(1:9, 9), nrow = 3)
tropdet(R)
</code></pre>

<hr>
<h2 id='tropFW'>Tropical Fermat-Weber Point</h2><span id='topic+tropFW'></span>

<h3>Description</h3>

<p>Compute the tropical Fermat-Weber (FW) point for a given data matrix.
The FW point minimizes the summed tropical distance to the trees described
in the data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropFW(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropFW_+3A_x">x</code></td>
<td>
<p>a data matrix, of dimension nobs x nvars; each row is an observation vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>fw</code></td>
<td>
<p>The fermat-weber point.</p>
</td></tr>
<tr><td><code>distsum</code></td>
<td>
<p>The sum of distance from each observation to the fermat-weber point.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- matrix(rnorm(100), ncol = 10)
tropFW(x)
</code></pre>

<hr>
<h2 id='troppca.linsp'>Tropical Principal Component Analysis by Tropical Linear Space</h2><span id='topic+troppca.linsp'></span>

<h3>Description</h3>

<p>Approximate the principal component as a tropical linear space
for a given data matrix and returns the results as an object of class <code>troppca</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>troppca.linsp(x, pcs = 2, iteration = list(), ncores = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="troppca.linsp_+3A_x">x</code></td>
<td>
<p>a data matrix, of size n x e, with each row an observation vector.
e is the dimension of the tropical space</p>
</td></tr>
<tr><td><code id="troppca.linsp_+3A_pcs">pcs</code></td>
<td>
<p>a numeric value indicating the order of principal component. (default: 2)</p>
</td></tr>
<tr><td><code id="troppca.linsp_+3A_iteration">iteration</code></td>
<td>
<p>a list with arguments controlling the iteration of the algorithm.
</p>

<dl>
<dt>exhaust</dt><dd><p>a logical variable indicating if to iterate over all possible combinations of the linear space
based on the given data matrix <code>x</code>. If FALSE, please input a number of iteration for <code>niter</code>.
If TRUE, please enter 0 for <code>niter</code> and this function will iterate over all possible combinations of linear space.
This could be time consuming when <code>x</code> is large. (default: FALSE)</p>
</dd>
<dt>niter</dt><dd><p>a numeric variable indicating the number of iterations. (default: 100)</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="troppca.linsp_+3A_ncores">ncores</code></td>
<td>
<p>a numeric value indicating the number of threads utilized for multi-cored CPUs. (default: 2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of S3 class <code>"troppca"</code>, including:
</p>
<table>
<tr><td><code>pc</code></td>
<td>
<p>The principal component as a tropical linear space</p>
</td></tr>
<tr><td><code>obj</code></td>
<td>
<p>The tropical PCA objective, the sum of tropical distance from each point to the projection.</p>
</td></tr>
<tr><td><code>projection</code></td>
<td>
<p>The projections of all data points.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>The geometry of principal component.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Rfast)
n &lt;- 100
e &lt;- 10
sig2 &lt;- 1
x &lt;- rbind(rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(sig2, e)))
troppca_fit &lt;- troppca.linsp(x)


</code></pre>

<hr>
<h2 id='troppca.linsp2poly'>Tropical Principal Component Analysis by Polytope Converted from Linear Space</h2><span id='topic+troppca.linsp2poly'></span>

<h3>Description</h3>

<p>Approximate the principal component as a tropical polytope converted from tropical linear space for a given data matrix
via MCMC and return the results as an object of class <code>troppca</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>troppca.linsp2poly(x, pcs = 2, nsample = 1000, ncores = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="troppca.linsp2poly_+3A_x">x</code></td>
<td>
<p>a data matrix, of size n x e, with each row an observation vector.
e is the dimension of the tropical space#'</p>
</td></tr>
<tr><td><code id="troppca.linsp2poly_+3A_pcs">pcs</code></td>
<td>
<p>a numeric value indicating the order of principal component. (default: 2)</p>
</td></tr>
<tr><td><code id="troppca.linsp2poly_+3A_nsample">nsample</code></td>
<td>
<p>a numeric value indicating the number of samples of MCMC. (default: 1000)</p>
</td></tr>
<tr><td><code id="troppca.linsp2poly_+3A_ncores">ncores</code></td>
<td>
<p>a numeric value indicating the number of threads utilized for multi-cored CPUs. (default: 2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of S3 class <code>"troppca"</code>, including:
</p>
<table>
<tr><td><code>pc</code></td>
<td>
<p>The principal component as a tropical linear space</p>
</td></tr>
<tr><td><code>obj</code></td>
<td>
<p>The tropical PCA objective, the sum of tropical distance from each point to the projection.</p>
</td></tr>
<tr><td><code>projection</code></td>
<td>
<p>The projections of all data points.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>The geometry of principal component.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Rfast)
n &lt;- 50
e &lt;- 50
s &lt;- 5
x &lt;- rbind(
  rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(n, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
troppca_fit &lt;- troppca.linsp2poly(x)


</code></pre>

<hr>
<h2 id='troppca.obj'>Compute Tropical PCA Objective</h2><span id='topic+troppca.obj'></span><span id='topic+troppca.obj2'></span>

<h3>Description</h3>

<p>Compute Tropical PCA Objective
</p>


<h3>Usage</h3>

<pre><code class='language-R'>troppca.obj(pc, x_list, cl)

troppca.obj2(pc, x_list, cl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="troppca.obj_+3A_pc">pc</code></td>
<td>
<p>a matrix of principal components</p>
</td></tr>
<tr><td><code id="troppca.obj_+3A_x_list">x_list</code></td>
<td>
<p>a list of vectors</p>
</td></tr>
<tr><td><code id="troppca.obj_+3A_cl">cl</code></td>
<td>
<p>cluster for parallel computing</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The numeric value of the objective function of tropical principle component analysis.
This is the sum of all tropical distance from each point to its projection on the tropical polytope.
</p>

<hr>
<h2 id='troppca.poly'>Tropical Principal Component Analysis by Tropical Polytope</h2><span id='topic+troppca.poly'></span>

<h3>Description</h3>

<p>Approximates the principal component as a tropical polytope for a given data matrix
via MCMC and return the results as an object of class <code>troppca</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>troppca.poly(x, pcs = 2, nsample = 1000, ncores = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="troppca.poly_+3A_x">x</code></td>
<td>
<p>a data matrix, of size n x e, with each row an observation vector.
e is the dimension of the tropical space#'</p>
</td></tr>
<tr><td><code id="troppca.poly_+3A_pcs">pcs</code></td>
<td>
<p>a numeric value indicating the order of principal component. (default: 2)</p>
</td></tr>
<tr><td><code id="troppca.poly_+3A_nsample">nsample</code></td>
<td>
<p>a numeric value indicating the number of samples of MCMC. (default: 1000)</p>
</td></tr>
<tr><td><code id="troppca.poly_+3A_ncores">ncores</code></td>
<td>
<p>a numeric value indicating the number of threads utilized for multi-cored CPUs. (default: 2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of S3 class <code>"troppca"</code>, including:
</p>
<table>
<tr><td><code>pc</code></td>
<td>
<p>The principal component as a tropical linear space</p>
</td></tr>
<tr><td><code>obj</code></td>
<td>
<p>The tropical PCA objective, the sum of tropical distance from each point to the projection.</p>
</td></tr>
<tr><td><code>projection</code></td>
<td>
<p>The projections of all data points.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>The geometry of principal component.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Rfast)
n &lt;- 50
e &lt;- 50
s &lt;- 5
x &lt;- rbind(
  rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(n, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
troppca_fit &lt;- troppca.poly(x)
plot(troppca_fit)


</code></pre>

<hr>
<h2 id='tropproj.linsp'>Projection on Tropical Linear Space</h2><span id='topic+tropproj.linsp'></span>

<h3>Description</h3>

<p>Compute projection of data points on a given tropical linear space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropproj.linsp(x, V)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropproj.linsp_+3A_x">x</code></td>
<td>
<p>a data matrix, of size n x e, with each row an observation.</p>
</td></tr>
<tr><td><code id="tropproj.linsp_+3A_v">V</code></td>
<td>
<p>a data matrix, of dimension s x e, with each row a basis of tropical linear space.
e is the dimension of the tropical space and s is the dimension of the linear space.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of projections of all data points.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Rfast)
n &lt;- 100
e &lt;- 10
sig2 &lt;- 1
s &lt;- 3
x &lt;- rbind(rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(sig2, e)))
V &lt;- matrix(runif(s * e, -10, 10), nrow = s, ncol = e)
x_proj &lt;- tropproj.linsp(x, V)
head(x_proj)
</code></pre>

<hr>
<h2 id='tropproj.poly'>Projection on Tropical Polytope</h2><span id='topic+tropproj.poly'></span>

<h3>Description</h3>

<p>Project a point onto a given tropical polytope.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropproj.poly(x, tconv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropproj.poly_+3A_x">x</code></td>
<td>
<p>a data vector, of length e.</p>
</td></tr>
<tr><td><code id="tropproj.poly_+3A_tconv">tconv</code></td>
<td>
<p>a data matrix, of size e x s, with each column a vertex of the tropical polytope.
e is the dimension of the tropical space and s is the number of vertices of the polytope</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A projected vector on the given tropical polytope.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate a tropical polytope consisting of three trees each with 5 leaves
library(ape)
pltp &lt;- sapply(1:3, function(i) {
  as.vector(rcoal(5))
})
# Generate an observation and vectorize it
tree &lt;- rcoal(5)
tree_vec &lt;- as.vector(tree)
tropproj.poly(tree_vec, pltp)
</code></pre>

<hr>
<h2 id='tropseg'>Compute Tropical Line Segment</h2><span id='topic+tropseg'></span>

<h3>Description</h3>

<p>Compute Tropical Line Segment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropseg(D1, D2, flag = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropseg_+3A_d1">D1</code></td>
<td>
<p>a vector of length 2.</p>
</td></tr>
<tr><td><code id="tropseg_+3A_d2">D2</code></td>
<td>
<p>a vector of length 2.</p>
</td></tr>
<tr><td><code id="tropseg_+3A_flag">flag</code></td>
<td>
<p>a logical value indicating if to swap <code>D1</code> and <code>D2</code>. (default: FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following:
</p>
<table>
<tr><td><code>1</code></td>
<td>
<p>End points as columns of each line component.</p>
</td></tr>
<tr><td><code>2</code></td>
<td>
<p>Overall length of the tropical line segment</p>
</td></tr>
</table>

<hr>
<h2 id='tropsvm'>Tropical Support Vector Machines</h2><span id='topic+tropsvm'></span>

<h3>Description</h3>

<p>Fit a discriminative two-class classifier via linear programming defined by the tropical
hyperplane which maximizes the minimum tropical distance from data points
to itself in order to separate the data points into sectors (half-spaces)
in the tropical projective torus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropsvm(x, y, auto.assignment = FALSE, assignment = NULL, ind = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropsvm_+3A_x">x</code></td>
<td>
<p>a data matrix, of dimension nobs x nvars; each row is an observation vector.</p>
</td></tr>
<tr><td><code id="tropsvm_+3A_y">y</code></td>
<td>
<p>a response vector with one label for each row/component of x.</p>
</td></tr>
<tr><td><code id="tropsvm_+3A_auto.assignment">auto.assignment</code></td>
<td>
<p>a logical value indicating if to provide an <code>assignment</code> manually.
If <code>FALSE</code>, an input is required, otherwise the function automatically
finds a good assignment.(default: FALSE)</p>
</td></tr>
<tr><td><code id="tropsvm_+3A_assignment">assignment</code></td>
<td>
<p>a numeric vector of length 4 indicating the sectors of tropical hyperplane that the
data will be assigned to. The first and third elements in the <code>assignment</code> are the coordinates of
an observed point in data matrix <code>x</code> believed from the first category where the maximum and second maximum
of the vector addition between the fitted optimal tropical hyperplane and the point itself are achieved.
The meanings for the second and the fourth element in the <code>assignment</code> are the same
but for the points in the second category. Namely, the first and second values in the <code>assignment</code>
are the indices of sectors where the two point clouds are assigned. Not needed when <code>auto.assignment = TRUE</code>. (default: NULL)</p>
</td></tr>
<tr><td><code id="tropsvm_+3A_ind">ind</code></td>
<td>
<p>a numeric value or a numeric vector ranging from 1 to 70 indicating which classification method
to be used. There are 70 different classification methods. Details of a given method can be retrieved by <code>summary</code>.
The different classification methods are proposed to resolve the issue when points fall on the intersections of sectors.
Users can have personal choices if better knowledge is assumed. (default: 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with S3 class <code>tropsvm</code> containing the fitted model, including:
</p>
<table>
<tr><td><code>apex</code></td>
<td>
<p>The negative apex of the fitted optimal tropical hyperplane.</p>
</td></tr>
<tr><td><code>assignment</code></td>
<td>
<p>The user-input or auto-found <code>assignment</code>.</p>
</td></tr>
<tr><td><code>index</code></td>
<td>
<p>The user-input classification method.</p>
</td></tr>
<tr><td><code>levels</code></td>
<td>
<p>The name of each category, consistent with categories in <code>y</code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code>predict</code>, <code>coef</code> and the <code>cv.tropsvm</code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# data generation
library(Rfast)
e &lt;- 100
n &lt;- 10
N &lt;- 100
s &lt;- 10
x &lt;- rbind(
  rmvnorm(n, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(n, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
y &lt;- as.factor(c(rep(1, n), rep(2, n)))
newx &lt;- rbind(
  rmvnorm(N, mu = c(5, -5, rep(0, e - 2)), sigma = diag(s, e)),
  rmvnorm(N, mu = c(-5, 5, rep(0, e - 2)), sigma = diag(s, e))
)
newy &lt;- as.factor(rep(c(1, 2), each = N))

# train the tropical svm
tropsvm_fit &lt;- tropsvm(x, y, auto.assignment = TRUE, ind = 1)

coef(tropsvm_fit)

# test with new data
pred &lt;- predict(tropsvm_fit, newx)

# check with accuracy
table(pred, newy)

# compute testing accuracy
sum(pred == newy) / length(newy)
</code></pre>

<hr>
<h2 id='tropsvm_helper'>Helper for <code>cv.tropsvm</code></h2><span id='topic+tropsvm_helper'></span>

<h3>Description</h3>

<p>Helper for <code>cv.tropsvm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropsvm_helper(x, y, assignment = NULL, ind = 1, newx = NULL, newy = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropsvm_helper_+3A_x">x</code></td>
<td>
<p>a data matrix, of dimension nobs x nvars; each row is an observation vector.</p>
</td></tr>
<tr><td><code id="tropsvm_helper_+3A_y">y</code></td>
<td>
<p>a response vector with one label for each row/component of x.</p>
</td></tr>
<tr><td><code id="tropsvm_helper_+3A_assignment">assignment</code></td>
<td>
<p>a numeric vector of length 4 indicating the sectors of tropical hyperplane that the
data will be assigned to. The first and third elements in the <code>assignment</code> are the coordinates of
an observed point in data matrix <code>x</code> believed from the first category where the maximum and second maximum
of the vector addition between the fitted optimal tropical hyperplane and the point itself are achieved.
The meanings for the second and the fourth element in the <code>assignment</code> are the same
but for the points in the second category. Namely, the first and second values in the <code>assignment</code>
are the indices of sectors where the two point cloud will be assigned. Not needed when <code>auto.assignment = TRUE</code>. (default: NULL)</p>
</td></tr>
<tr><td><code id="tropsvm_helper_+3A_ind">ind</code></td>
<td>
<p>a numeric value or a numeric vector ranging from 1 to 70 indicating which classification method
to be used. There are 70 different classification methods. Details of a given method can be retrieved by <code>summary</code>.
The different classification methods are proposed to resolve the issue when points fall on the intersections of sectors.
Users can have personal choices if better knowledge is assumed. (default: 1)</p>
</td></tr>
<tr><td><code id="tropsvm_helper_+3A_newx">newx</code></td>
<td>
<p>the same as &quot;x&quot; but only needed in <code>cv.tropsvm</code>, which is used as validation data. (default: <code>NULL</code>)</p>
</td></tr>
<tr><td><code id="tropsvm_helper_+3A_newy">newy</code></td>
<td>
<p>the same as &quot;y&quot; but only needed in <code>cv.tropsvm</code>, which is used as validation labels (default: <code>NULL</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Classification accuracy
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
