<!DOCTYPE html><html lang="en"><head><title>Help for package ruijter</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ruijter}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ruijter-package'><p>ruijter: Technical Data Sets by Ruijter et al. (2013)</p></a></li>
<li><a href='#ds_380'><p>380-replicates set</p></a></li>
<li><a href='#ds_94_4'><p>94-replicates-4-dilutions set</p></a></li>
<li><a href='#ds_competimer'><p>Competimer set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Technical Data Sets by Ruijter et al. (2013)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Description:</td>
<td>The real-time quantitative polymerase chain reaction
    (qPCR) technical data sets by Ruijter et al. (2013)
    &lt;<a href="https://doi.org/10.1016%2Fj.ymeth.2012.08.011">doi:10.1016/j.ymeth.2012.08.011</a>&gt;: (i) the four-point 10-fold dilution
    series; (ii) 380 replicates; and (iii) the competimer data set. These
    three data sets can be used to benchmark qPCR methods. Original data set
    is available at
    <a href="https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip">https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip</a>.
    This package fixes incorrect annotations in the original data sets.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by/4.0">CC BY 4.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ramiromagno/ruijter">https://github.com/ramiromagno/ruijter</a>, <a href="https://rmagno.eu/ruijter/">https://rmagno.eu/ruijter/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ramiromagno/ruijter/issues">https://github.com/ramiromagno/ruijter/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>tibble</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-22 22:48:21 UTC; rmagno</td>
</tr>
<tr>
<td>Author:</td>
<td>Ramiro Magno <a href="https://orcid.org/0000-0001-5226-3441"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Pattern Institute [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ramiro Magno &lt;rmagno@pattern.institute&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-22 23:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='ruijter-package'>ruijter: Technical Data Sets by Ruijter et al. (2013)</h2><span id='topic+ruijter'></span><span id='topic+ruijter-package'></span>

<h3>Description</h3>

<p>The real-time quantitative polymerase chain reaction (qPCR) technical data sets by Ruijter et al. (2013) <a href="https://doi.org/10.1016/j.ymeth.2012.08.011">doi:10.1016/j.ymeth.2012.08.011</a>: (i) the four-point 10-fold dilution series; (ii) 380 replicates; and (iii) the competimer data set. These three data sets can be used to benchmark qPCR methods. Original data set is available at <a href="https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip">https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip</a>. This package fixes incorrect annotations in the original data sets.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Ramiro Magno <a href="mailto:rmagno@pattern.institute">rmagno@pattern.institute</a> (<a href="https://orcid.org/0000-0001-5226-3441">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Pattern Institute [copyright holder, funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/ramiromagno/ruijter">https://github.com/ramiromagno/ruijter</a>
</p>
</li>
<li> <p><a href="https://rmagno.eu/ruijter/">https://rmagno.eu/ruijter/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ramiromagno/ruijter/issues">https://github.com/ramiromagno/ruijter/issues</a>
</p>
</li></ul>


<hr>
<h2 id='ds_380'>380-replicates set</h2><span id='topic+ds_380'></span>

<h3>Description</h3>

<p>The 380-replicates set is a dilution consisting of 150,000 molecules (see the
Admonition section below) of the MYCN oligonucleotide was created in 10 ng/µl
yeast tRNA carrier. qPCR amplifications were performed in 380 replicated 8 µl
reactions and quadruplicated reactions of the NTC sample were performed on
the same 384-well plate.
</p>
<p>qPCR was done on a CFX 384 instrument (Bio-Rad). A 384-well qPCR plate was prepared using a
96-well head pipetting robot (Tecan Freedom Evo 150). qPCR amplifications
were performed in 8 µl containing:
</p>

<ul>
<li><p> 4 µl iQ SYBR Green Supermix (Bio-Rad);
</p>
</li>
<li><p> 0.4 µl forward and 0.4 µl reverse primer (5 µM each);
</p>
</li>
<li><p> 0.2 µl nuclease-free water;
</p>
</li>
<li><p> 3 µl of standard oligonucleotide.
</p>
</li></ul>

<p>All reactions were performed in 384-well plates (Hard-Shell 384-well
microplates and Microseal B clear using adhesive seals (Bio-Rad).
</p>
<p>The cycling conditions were comprised of 3 min polymerase activation at 95 °C
and 45 cycles of 15 s at 95 °C and 30 s at 60 °C followed by a dissociation
curve analysis from 60 to 95 °C.
</p>


<h4>Admonition</h4>

<p>The concentration information for the curves in this data set was seemingly
incorrectly indicated in the original publication (confirmed by email with
the senior author Prof. Jo Vandesompele). Instead of the 15,000 copies
reported in the original paper, the number of copies was most likely 150,000
copies (10x the value indicated).
</p>



<h3>Usage</h3>

<pre><code class='language-R'>ds_380
</code></pre>


<h3>Format</h3>



<h4><code>ds_380</code></h4>

<p>A data frame with 17,280 rows and 9 columns:
</p>

<dl>
<dt><code>well</code></dt><dd><p>Well identifier.</p>
</dd>
<dt><code>replicate</code></dt><dd><p>Replicate identifier.</p>
</dd>
<dt><code>dye</code></dt><dd><p>In all reactions the SYBR Green I master mix (Roche) was used,
so the value is always <code>"SYBR"</code>.</p>
</dd>
<dt><code>target</code></dt><dd><p>Target identifier: MYCN.</p>
</dd>
<dt><code>sample_type</code></dt><dd><p>Sample type: <code>"ntc"</code> (no template control) or <code>"std"</code> (standard).</p>
</dd>
<dt><code>copies</code></dt><dd><p>Standard copy number: either <code>0</code> (for NTCs), or <code>150000</code> for standards.</p>
</dd>
<dt><code>dilution</code></dt><dd><p>Dilution factor (the reciprocal of <code>copies</code>). Higher number
means greater dilution.</p>
</dd>
<dt><code>cycle</code></dt><dd><p>PCR cycle.</p>
</dd>
<dt><code>fluor</code></dt><dd><p>Raw fluorescence values.</p>
</dd>
</dl>




<h3>Source</h3>

<p><a href="https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip">https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip</a>
</p>

<hr>
<h2 id='ds_94_4'>94-replicates-4-dilutions set</h2><span id='topic+ds_94_4'></span>

<h3>Description</h3>

<p>The 94-replicates-4-dilutions set is a dilution series consisting of four
10-fold serial dilution points, starting from 15,000 molecules down to 15
molecules, created using 10 ng/µl yeast tRNA as carrier (Roche). The same
dilution of the carrier was used to create the NTC sample.
</p>
<p>qPCR was done on a CFX 384 instrument (Bio-Rad). A 384-well qPCR plate was prepared using a
96-well head pipetting robot (Tecan Freedom Evo 150). qPCR amplifications
were performed in 8 µl containing:
</p>

<ul>
<li><p> 4 µl iQ SYBR Green Supermix (Bio-Rad);
</p>
</li>
<li><p> 0.4 µl forward and 0.4 µl reverse primer (5 µM each);
</p>
</li>
<li><p> 0.2 µl nuclease-free water;
</p>
</li>
<li><p> 3 µl of standard oligonucleotide.
</p>
</li></ul>

<p>A total of 94 replicated reactions were dispensed for each of 4 dilution
points. In addition, the NTC reaction was analyzed in 8 replicates.
</p>
<p>All reactions were performed in 384-well plates (Hard-Shell 384-well
microplates and Microseal B clear using adhesive seals (Bio-Rad).
</p>
<p>The cycling conditions were comprised of 3 min polymerase activation at 95 °C
and 45 cycles of 15 s at 95 °C and 30 s at 60 °C followed by a dissociation
curve analysis from 60 to 95 °C.
</p>


<h4>Admonition</h4>

<p>The concentration information for the curves in this data set was seemingly
incorrectly indicated (confirmed by email with the senior author Prof. Jo
Vandesompele). The annotation provided in the supplementary material that
accompanies the original publication was reversed. So, to correct for this,
the annotation of the number of copies in <code>ds_94_4</code> has been changed:
</p>

<ul>
<li><p> 15 became 15,000
</p>
</li>
<li><p> 150 became 1,500
</p>
</li>
<li><p> 1,500 became 150
</p>
</li>
<li><p> 15,000 became 15
</p>
</li></ul>




<h3>Usage</h3>

<pre><code class='language-R'>ds_94_4
</code></pre>


<h3>Format</h3>



<h4><code>ds_94_4</code></h4>

<p>A data frame with 17,280 rows and 9 columns:
</p>

<dl>
<dt><code>well</code></dt><dd><p>Well identifier.</p>
</dd>
<dt><code>replicate</code></dt><dd><p>Replicate identifier.</p>
</dd>
<dt><code>dye</code></dt><dd><p>In all reactions the SYBR Green I master mix (Roche) was used,
so the value is always <code>"SYBR"</code>.</p>
</dd>
<dt><code>target</code></dt><dd><p>Target identifier: MYCN.</p>
</dd>
<dt><code>sample_type</code></dt><dd><p>Sample type: <code>"ntc"</code> (no template control) or <code>"std"</code> (standard).</p>
</dd>
<dt><code>copies</code></dt><dd><p>Standard copy number: either <code>0</code> (for NTCs), or <code>15</code>, <code>150</code>,
<code>1500</code> or <code>15000</code> for standards.</p>
</dd>
<dt><code>dilution</code></dt><dd><p>Dilution factor (the reciprocal of <code>copies</code>). Higher number
means greater dilution.</p>
</dd>
<dt><code>cycle</code></dt><dd><p>PCR cycle.</p>
</dd>
<dt><code>fluor</code></dt><dd><p>Raw fluorescence values.</p>
</dd>
</dl>




<h3>Source</h3>

<p><a href="https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip">https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip</a>
</p>

<hr>
<h2 id='ds_competimer'>Competimer set</h2><span id='topic+ds_competimer'></span>

<h3>Description</h3>

<p>Competitive primers were synthesized on basis of identical sequence and
blocked by amination at the 3' end to allow annealing, but avoid elongation
during the PCR process.
</p>
<p>A six-point 4-fold serial dilution series made from reference human genomic
DNA (Roche), starting from 64 ng/µl down to 0.0625 ng/µl, was created in 10
ng/µl yeast tRNA as carrier. The same dilution of the carrier was used to
create a NTC sample.
</p>
<p>qPCR amplifications were performed in 7.5 µl total reaction volume containing:
</p>

<ul>
<li><p> 3.75 µl 2x custom made qPCR SYBR green I Mastermix (Eurogentec)
</p>
</li>
<li><p> 0.375 µl forward primer (5 µM) and 0.375 µl reverse primer (5 µM)
</p>
</li>
<li><p> 1 µl of a mixture of nuclease-free water and equal amounts of both forward and reverse competitive (aminated) primers
</p>
</li>
<li><p> 2 µl diluted standard
</p>
</li></ul>

<p>A total of 7 competitive mixes were prepared for each dilution point,
containing 0%, 5%, 10%, 20%, 30%, 40%, and 50% (of the total amount of
primer) competitive (aminated) forward and reverse primers. Each reaction was
run in triplicate.
</p>
<p>The qPCR cycling was performed on the LightCycler480 (Roche) using white
LightCycler480 384-multiwell plates with Light-Cycler480 sealing foils
(Roche).
</p>
<p>The cycling conditions were com  prised of 10 min polymerase activation at 95
°C, and 45 cycles of 15 s at 95 °C, 30 s at 60 °C, followed by a dissociation
curve analysis from 60 to 95 °C.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ds_competimer
</code></pre>


<h3>Format</h3>



<h4><code>ds_competimer</code></h4>

<p>A data frame with 6,615 rows and 10 columns:
</p>

<dl>
<dt><code>well</code></dt><dd><p>Well identifier.</p>
</dd>
<dt><code>replicate</code></dt><dd><p>Replicate identifier.</p>
</dd>
<dt><code>dye</code></dt><dd><p>In all reactions the SYBR Green I master mix (Roche) was used,
so the value is always <code>"SYBR"</code>.</p>
</dd>
<dt><code>pct</code></dt><dd><p>Percentage of competitive (aminated) primers in the mix.</p>
</dd>
<dt><code>conc</code></dt><dd><p>Concentration of reference human genomic DNA (Roche): from 64
ng/µl down to 0.0625 ng/µl.</p>
</dd>
<dt><code>target</code></dt><dd><p>Target identifier: AluSx.</p>
</dd>
<dt><code>sample_type</code></dt><dd><p>Sample type: <code>"ntc"</code> (no template control) or <code>"std"</code> (standard).</p>
</dd>
<dt><code>dilution</code></dt><dd><p>Dilution factor (the reciprocal of <code>conc</code>, i.e. <code>64 / conc</code>). Higher number
means greater dilution.</p>
</dd>
<dt><code>cycle</code></dt><dd><p>PCR cycle.</p>
</dd>
<dt><code>fluor</code></dt><dd><p>Raw fluorescence values.</p>
</dd>
</dl>




<h3>Source</h3>

<p><a href="https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip">https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
