<!DOCTYPE html><html lang="en"><head><title>Help for package RVA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RVA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#c2BroadSets'><p>This is data to be included in package</p></a></li>
<li><a href='#cal.pathway.scores'><p>calculate pathway scores</p></a></li>
<li><a href='#calc.cfb'><p>Calculate CFB</p></a></li>
<li><a href='#count_table'><p>This is data to be included in package</p></a></li>
<li><a href='#dlPathwaysDB'><p>DL Pathways DB</p></a></li>
<li><a href='#get.cpm.colors'><p>Get CPM Colors</p></a></li>
<li><a href='#get.cutoff.df'><p>Create ggplot object for number of differntially expressed genes with</p>
different FDR and fold change cutoff.</a></li>
<li><a href='#get.cutoff.ggplot'><p>Create ggplot object for number of differntially expressed genes with</p>
different FDR and fold change cutoff.</a></li>
<li><a href='#make.cutoff.plotly'><p>Create plotly object for number of DE genes at different cutoff combinations</p></a></li>
<li><a href='#multiPlot'><p>Multi Plot</p></a></li>
<li><a href='#nullreturn'><p>Null Return</p></a></li>
<li><a href='#plot_cutoff'><p>Check number of DE genes at different cutoff combinations</p></a></li>
<li><a href='#plot_cutoff_single'><p>Create plotly object for number of DE genes at different cutoff combinations</p></a></li>
<li><a href='#plot_gene'><p>Plot gene expression</p></a></li>
<li><a href='#plot_heatmap.cfb'><p>Plot a CFB Heatmap</p></a></li>
<li><a href='#plot_heatmap.cpm'><p>Plot a CPM Heatmap</p></a></li>
<li><a href='#plot_heatmap.expr'><p>Plot Heatmap From Raw CPM</p></a></li>
<li><a href='#plot_pathway'><p>Pathway analysis and visualization</p></a></li>
<li><a href='#plot_qq'><p>Plot qqplot</p></a></li>
<li><a href='#plot_volcano'><p>Plot volcanoplot</p></a></li>
<li><a href='#prettyGraphs'><p>Pretty Graphs</p></a></li>
<li><a href='#produce.cutoff.message'><p>Create a message about fold change and pvalues used to produce the plot.</p></a></li>
<li><a href='#produce.cutoff.warning'><p>Create a warning about pvalue or FDR minimum value</p></a></li>
<li><a href='#reformat.ensembl'><p>Reformat Ensembl GeneIDs</p></a></li>
<li><a href='#sample_annotation'><p>This is data to be included in package</p></a></li>
<li><a href='#sample_count_cpm'><p>This is data to be included in package</p></a></li>
<li><a href='#Sample_disease_gene_set'><p>This is data to be included in package</p></a></li>
<li><a href='#Sample_summary_statistics_table'><p>This is data to be included in package</p></a></li>
<li><a href='#Sample_summary_statistics_table1'><p>This is data to be included in package</p></a></li>
<li><a href='#secondCutoffErr'><p>Second Cutoff Error</p></a></li>
<li><a href='#transform.geneid'><p>Transform GeneIDs</p></a></li>
<li><a href='#validate.annot'><p>Validate Annotation Table</p></a></li>
<li><a href='#validate.baseline'><p>Validate Baseline Values</p></a></li>
<li><a href='#validate.col.types'><p>Check Summary Statistics Required Column Types</p></a></li>
<li><a href='#validate.comp.names'><p>Validate Comp Names</p></a></li>
<li><a href='#validate.data'><p>Validate Data Input</p></a></li>
<li><a href='#validate.data.annot'><p>Validate Data in the Context of Annotation</p></a></li>
<li><a href='#validate.FC'><p>Validate Foldchange</p></a></li>
<li><a href='#validate.flag'><p>Validate Flag Value Is Valid</p></a></li>
<li><a href='#validate.genes.present'><p>Validate genes present</p></a></li>
<li><a href='#validate.geneset'><p>Validate Geneset</p></a></li>
<li><a href='#validate.numeric'><p>Validate Numeric Column</p></a></li>
<li><a href='#validate.pathways.db'><p>Validate Pathways DB</p></a></li>
<li><a href='#validate.pval.range'><p>Validate P-value Range</p></a></li>
<li><a href='#validate.pvalflag'><p>Validate pval flag</p></a></li>
<li><a href='#validate.pvals'><p>Validate Pvalues</p></a></li>
<li><a href='#validate.single.table.isnotlist'><p>Validate Single Table is not list</p></a></li>
<li><a href='#validate.stats'><p>Validate Summary Statistics File</p></a></li>
<li><a href='#validate.stats.cols'><p>Check Summary Statistics Required Columns</p></a></li>
<li><a href='#wpA2020'><p>This is data to be included in package</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>RNAseq Visualization Automation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.5</td>
</tr>
<tr>
<td>Description:</td>
<td>Automate downstream visualization &amp; pathway analysis in RNAseq analysis. 'RVA' is a collection of functions that efficiently visualize RNAseq differential expression analysis result from summary statistics tables. It also utilize the Fisher's exact test to evaluate gene set or pathway enrichment in a convenient and efficient manner.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xingpeng Li &lt;xingpeng.li@pfizer.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/THERMOSTATS/RVA">https://github.com/THERMOSTATS/RVA</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>GSVAdata (&ge; 1.22.0), clusterProfiler (&ge; 3.15.1), data.table
(&ge; 1.12.8), edgeR (&ge; 3.28.1), org.Hs.eg.db (&ge; 3.10.0),
ComplexHeatmap (&ge; 2.2.0), GSEABase (&ge; 1.48.0), circlize (&ge;
0.4.10), dplyr (&ge; 1.0.0), ggplot2 (&ge; 3.3.2), ggpubr (&ge;
0.4.0), grid (&ge; 3.6.1), gridExtra (&ge; 2.3), haven (&ge; 2.3.1),
msigdbr (&ge; 7.1.1), plotly (&ge; 4.9.2.1), purrr (&ge; 0.3.4),
rWikiPathways (&ge; 1.6.1), stringr (&ge; 1.4.0), tibble, tidyr (&ge;
1.1.0), XML, rlang</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-01 20:48:52 UTC; lix410</td>
</tr>
<tr>
<td>Author:</td>
<td>Xingpeng Li <a href="https://orcid.org/0000-0002-1331-1225"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-01 21:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='c2BroadSets'>This is data to be included in package</h2><span id='topic+c2BroadSets'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c2BroadSets
</code></pre>


<h3>Format</h3>

<p>GeneSetCollection
</p>

<dl>
<dt>Genesetcollection</dt><dd><p>GeneSetCollection from BroadCollection</p>
</dd>
</dl>


<hr>
<h2 id='cal.pathway.scores'>calculate pathway scores</h2><span id='topic+cal.pathway.scores'></span>

<h3>Description</h3>

<p>Calculate pathway scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cal.pathway.scores(
  data,
  pathway.db,
  gene.id.type,
  FCflag,
  FDRflag,
  FC.cutoff,
  FDR.cutoff,
  OUT.Directional = NULL,
  IS.list = FALSE,
  customized.pathways,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cal.pathway.scores_+3A_data">data</code></td>
<td>
<p>A summary statistics table (data.frame) or <code>data.list</code> generated by DE analysis software like limma or DEseq2</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_pathway.db">pathway.db</code></td>
<td>
<p>pathway database used</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_gene.id.type">gene.id.type</code></td>
<td>
<p>gene.id.type</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_fcflag">FCflag</code></td>
<td>
<p>The column name (character) of fold change information, assuming the FC is log2 transformed. Default = &quot;logFC&quot;.</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_fdrflag">FDRflag</code></td>
<td>
<p>The column name (character) of adjusted p value or FDR. Default = &quot;adj.P.Val&quot;.</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_fc.cutoff">FC.cutoff</code></td>
<td>
<p>The fold change cutoff (numeric) selected to subset summary statistics table. Default = 1.5.</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_fdr.cutoff">FDR.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) to subset summary statistics table. Default = 0.05.</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_out.directional">OUT.Directional</code></td>
<td>
<p>logical, whether output directional or non-directional pathway analysis result, default: NULL.</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_is.list">IS.list</code></td>
<td>
<p>logical, whether the input is a list, default: NULL</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_customized.pathways">customized.pathways</code></td>
<td>
<p>the customized pathways in the format of two column dataframe to be used in analysis</p>
</td></tr>
<tr><td><code id="cal.pathway.scores_+3A_...">...</code></td>
<td>
<p>pass over parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a dataframe.
</p>


<h3>References</h3>

<p>Xingpeng Li &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='calc.cfb'>Calculate CFB</h2><span id='topic+calc.cfb'></span>

<h3>Description</h3>

<p>This function calculates the change from baseline.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.cfb(data, annot, baseline.flag, baseline.val)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.cfb_+3A_data">data</code></td>
<td>
<p>Dataframe with subject id, annotation flag, gene id and cpm value
(from count tables) columns.</p>
</td></tr>
<tr><td><code id="calc.cfb_+3A_annot">annot</code></td>
<td>
<p>A long-format dataframe with any pertinent treatment data about
the samples. The only required column is one titled the <code>sample.id</code>
value with values matching the column names of sample IDs in <code>data</code>.
Additional columns can contain information such as treament
compounds, dates of sample collection, or dosage quantities.</p>
</td></tr>
<tr><td><code id="calc.cfb_+3A_baseline.flag">baseline.flag</code></td>
<td>
<p>A character vector of column names. These columns in <code>annot</code>
contain the values to compare across.</p>
</td></tr>
<tr><td><code id="calc.cfb_+3A_baseline.val">baseline.val</code></td>
<td>
<p>A character vector of values. This vector must be the
same length as <code>baseline.flag</code>, and the value at each index must
represent a value from the column given by the corresponding index
in <code>baseline.flag</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='count_table'>This is data to be included in package</h2><span id='topic+count_table'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_table
</code></pre>


<h3>Format</h3>

<p>An example count table where row names are gene ID, each column is a sample
</p>

<dl>
<dt>counttable</dt><dd><p>count table</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='dlPathwaysDB'>DL Pathways DB</h2><span id='topic+dlPathwaysDB'></span>

<h3>Description</h3>

<p>Download gene database for enrichment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dlPathwaysDB(pathway.db, customized.pathways = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dlPathwaysDB_+3A_pathway.db">pathway.db</code></td>
<td>
<p>The databse to be used for encrichment analysis. Can be one of the following, &quot;rWikiPathways&quot;, &quot;KEGG&quot;, &quot;REACTOME&quot;, &quot;Hallmark&quot;,&quot;rWikiPathways_aug_2020&quot;</p>
</td></tr>
<tr><td><code id="dlPathwaysDB_+3A_customized.pathways">customized.pathways</code></td>
<td>
<p>the user provided pathway added for analysis.</p>
</td></tr>
<tr><td><code id="dlPathwaysDB_+3A_...">...</code></td>
<td>
<p>pass over parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a dataframe.
</p>


<h3>References</h3>

<p>Xingpeng Li &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='get.cpm.colors'>Get CPM Colors</h2><span id='topic+get.cpm.colors'></span>

<h3>Description</h3>

<p>This function creates the color gradient for the cpm data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.cpm.colors(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.cpm.colors_+3A_data">data</code></td>
<td>
<p>The CPM dataset.</p>
</td></tr>
</table>

<hr>
<h2 id='get.cutoff.df'>Create ggplot object for number of differntially expressed genes with
different FDR and fold change cutoff.</h2><span id='topic+get.cutoff.df'></span>

<h3>Description</h3>

<p>This function processes dataframe from plot_cutoff_single function
and produces a ggplot object which depicts the number of differntially expressed
genes with different FDR and fold change cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.cutoff.df(datin, pvalues, FCs, FCflag = "logFC", FDRflag = "adj.P.Val")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.cutoff.df_+3A_datin">datin</code></td>
<td>
<p>Dataframe from plot_cutoff_single.</p>
</td></tr>
<tr><td><code id="get.cutoff.df_+3A_pvalues">pvalues</code></td>
<td>
<p>A set of p-values for FDR cutoff to be checked.</p>
</td></tr>
<tr><td><code id="get.cutoff.df_+3A_fcs">FCs</code></td>
<td>
<p>A set of fold change cutoff to be checked.</p>
</td></tr>
<tr><td><code id="get.cutoff.df_+3A_fcflag">FCflag</code></td>
<td>
<p>The column name of the log2FC in the summary statistics table.</p>
</td></tr>
<tr><td><code id="get.cutoff.df_+3A_fdrflag">FDRflag</code></td>
<td>
<p>The column name of the False Discovery Rate (FDR) in the summary statistics table.</p>
</td></tr>
</table>

<hr>
<h2 id='get.cutoff.ggplot'>Create ggplot object for number of differntially expressed genes with
different FDR and fold change cutoff.</h2><span id='topic+get.cutoff.ggplot'></span>

<h3>Description</h3>

<p>This function processes dataframe from plot_cutoff_single function
and produces a ggplot object which depicts the number of differntially expressed
genes with different FDR and fold change cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.cutoff.ggplot(df, FCflag, FDRflag)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.cutoff.ggplot_+3A_df">df</code></td>
<td>
<p>Dataframe from plot_cutoff_single.</p>
</td></tr>
<tr><td><code id="get.cutoff.ggplot_+3A_fcflag">FCflag</code></td>
<td>
<p>The column name of the log2FC in the summary statistics table.</p>
</td></tr>
<tr><td><code id="get.cutoff.ggplot_+3A_fdrflag">FDRflag</code></td>
<td>
<p>The column name of the False Discovery Rate (FDR) in the summary statistics table.</p>
</td></tr>
</table>

<hr>
<h2 id='make.cutoff.plotly'>Create plotly object for number of DE genes at different cutoff combinations</h2><span id='topic+make.cutoff.plotly'></span>

<h3>Description</h3>

<p>This function processes summary statistics table generated by differential expression analysis
like <code>limma</code> or <code>DESeq2</code> to produce an interactibe visual object
which depicts the number of differntially expressed genes with different FDR and
fold change cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.cutoff.plotly(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make.cutoff.plotly_+3A_df">df</code></td>
<td>
<p>Summary statistics table from limma or DEseq2, where each row is a gene.</p>
</td></tr>
</table>

<hr>
<h2 id='multiPlot'>Multi Plot</h2><span id='topic+multiPlot'></span>

<h3>Description</h3>

<p>Multi plot is for directional and non-directional plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiPlot(allID, backup.d.sig, nd.res, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiPlot_+3A_allid">allID</code></td>
<td>
<p>A vector of all pathway ID's from directional and non directional enriched datasets.</p>
</td></tr>
<tr><td><code id="multiPlot_+3A_backup.d.sig">backup.d.sig</code></td>
<td>
<p>A dataframe type of object with directional pathways data prior to any cutoff's being applied</p>
</td></tr>
<tr><td><code id="multiPlot_+3A_nd.res">nd.res</code></td>
<td>
<p>A dataframe type of object with non directional pathways data prior to any cutoff's being applied</p>
</td></tr>
<tr><td><code id="multiPlot_+3A_...">...</code></td>
<td>
<p>pass on variables</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multi plot is for directional and non-directional plots, when one of the plots doesn't contain observations.
</p>


<h3>Value</h3>

<p>Returns ggplot.
</p>


<h3>References</h3>

<p>Xingpeng Li &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='nullreturn'>Null Return</h2><span id='topic+nullreturn'></span>

<h3>Description</h3>

<p>The function takes in a boolean value and a numeric value, which it uses to decide what to output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nullreturn(IS.list, type = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nullreturn_+3A_is.list">IS.list</code></td>
<td>
<p>Indicator of whether the data frame being input is list or not.</p>
</td></tr>
<tr><td><code id="nullreturn_+3A_type">type</code></td>
<td>
<p>If type = 1(default) return directional null plot. If type = 2 return non directional null plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>nullreturn is a function that returns NULL for single df inputs that don't hold true for threshold values. It returns an empty dataframe for
list inputs which don't satisfy the cutoff's
</p>


<h3>Value</h3>

<p>The function returns either returns a data frame or the value NULL.
</p>


<h3>References</h3>

<p>Xingpeng Li &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='plot_cutoff'>Check number of DE genes at different cutoff combinations</h2><span id='topic+plot_cutoff'></span>

<h3>Description</h3>

<p>This function processes summary statistics table generated by differential expression analysis
like <code>limma</code> or <code>DESeq2</code> to evaluate the number of differntially expressed genes with different FDR and
fold change cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_cutoff(
  data = data,
  comp.names = NULL,
  FCflag = "logFC",
  FDRflag = "adj.P.Val",
  FCmin = 1.2,
  FCmax = 2,
  FCstep = 0.1,
  p.min = 0,
  p.max = 0.2,
  p.step = 0.01,
  plot.save.to = NULL,
  gen.3d.plot = TRUE,
  gen.plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_cutoff_+3A_data">data</code></td>
<td>
<p>Summary statistics table or a list of summary statistics tables from limma or DEseq2, where each row is a gene.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_comp.names">comp.names</code></td>
<td>
<p>A character vector that contains the comparison names which correspond to the same order as <code>data</code>.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_fcflag">FCflag</code></td>
<td>
<p>The column name of the log2FC in the summary statistics table. Default = &quot;logFC&quot;.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_fdrflag">FDRflag</code></td>
<td>
<p>The column name of the False Discovery Rate (FDR) in the summary statistics table. Default = &quot;adj.P.Val&quot;.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_fcmin">FCmin</code></td>
<td>
<p>The minimum starting fold change cutoff to be checked, so the minimum fold change cutoff to be evaluated
will be FCmin + FCstep, FCmin default = 1.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_fcmax">FCmax</code></td>
<td>
<p>The maximum fold change cutoff to be checked, default = 2.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_fcstep">FCstep</code></td>
<td>
<p>The step from the minimum to maximum fold change cutoff, one step increase at a time, default = 0.01.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_p.min">p.min</code></td>
<td>
<p>The minimum starting FDR cutoff to be checked, so the minimum fold change cutoff to be evaluated
will be p.min + p.step, p.min default = 0.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_p.max">p.max</code></td>
<td>
<p>The maximum FDR cutoff to be checked, default = 0.2.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_p.step">p.step</code></td>
<td>
<p>The step from the minimum to maximum fold change cutoff, one step increase at a time, default = 0.005.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_plot.save.to">plot.save.to</code></td>
<td>
<p>The address where to save the plot from simplified cutoff combination with FDR of 0.01, 0.05, 0.1, and 0.2.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_gen.3d.plot">gen.3d.plot</code></td>
<td>
<p>Whether generate a 3d plotly object to visualize the result, only applys to single dataframe input, default = F.</p>
</td></tr>
<tr><td><code id="plot_cutoff_+3A_gen.plot">gen.plot</code></td>
<td>
<p>Whether generate a plot to visualize the result, default = T.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes the summary statistics and returns a list which contains 3 objects: a table which describes the number
of DE genes with different cutoff combinations of FDR and fold change, a ggplot object which depicts a simplified version of
cutoff selection combination, and a plotly 3d visulization object which depicts a high resolution of cutoff combinations.
The default range of the fold change is from 1 to 2, and p value is from 0 to 0.2, with the step of 0.01
for FC and 0.005 for FDR.
</p>


<h3>Value</h3>

<p>If the input <code>data</code> is a data list, then a multi-facet ggplot plot object which contains each
of the summary statistics table will be returned; otherwise, if the input <code>data</code> is a data frame, then the function will return a list which contains 3 elements:
</p>
<table role = "presentation">
<tr><td><code>df.sub</code></td>
<td>
<p>A dataframe, which contains the number of genes(3rd column) with FDR (1st column), Fold Change (2nd column)</p>
</td></tr>
<tr><td><code>plot3d</code></td>
<td>
<p>A plotly object to show the 3d illustration of all possible cutoff selectiosn and
the number of DE genes in the 3d surface</p>
</td></tr>
<tr><td><code>gp</code></td>
<td>
<p>A ggplot object to show the simplified cutoff combination result</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li &amp; Olya Besedina, RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_cutoff(Sample_summary_statistics_table)

plot_cutoff(data = list(Sample_summary_statistics_table, Sample_summary_statistics_table1),
           comp.names = c("A", "B"))



</code></pre>

<hr>
<h2 id='plot_cutoff_single'>Create plotly object for number of DE genes at different cutoff combinations</h2><span id='topic+plot_cutoff_single'></span>

<h3>Description</h3>

<p>This function processes summary statistics table generated by differential expression analysis
like <code>limma</code> or <code>DESeq2</code> and produces a table which contains gene counts
for each of the pvalue and FC combination
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_cutoff_single(datin, FCflag, FDRflag, FCs, pvalues)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_cutoff_single_+3A_datin">datin</code></td>
<td>
<p>Summary statistics table from limma or DEseq2, where each row is a gene.</p>
</td></tr>
<tr><td><code id="plot_cutoff_single_+3A_fcflag">FCflag</code></td>
<td>
<p>The column name of the log2FC in the summary statistics table.</p>
</td></tr>
<tr><td><code id="plot_cutoff_single_+3A_fdrflag">FDRflag</code></td>
<td>
<p>The column name of the False Discovery Rate (FDR) in the summary statistics table.</p>
</td></tr>
<tr><td><code id="plot_cutoff_single_+3A_fcs">FCs</code></td>
<td>
<p>A set of fold change cutoff to be checked.</p>
</td></tr>
<tr><td><code id="plot_cutoff_single_+3A_pvalues">pvalues</code></td>
<td>
<p>A set of p-values for FDR cutoff to be checked.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_gene'>Plot gene expression</h2><span id='topic+plot_gene'></span>

<h3>Description</h3>

<p>This is the function to process the gene count table to show gene expression variations over time or across groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gene(
  data = ~dat,
  anno = ~meta,
  gene.names = c("AAAS", "A2ML1", "AADACL3"),
  ct.table.id.type = "ENSEMBL",
  gene.id.type = "SYMBOL",
  treatment = "Treatment",
  sample.id = "sample_id",
  time = "day",
  log.option = TRUE,
  plot.save.to = NULL,
  input.type = "count"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_gene_+3A_data">data</code></td>
<td>
<p>Count table in the format of dataframe with gene id as row.names.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_anno">anno</code></td>
<td>
<p>Annotation table that provides design information.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_gene.names">gene.names</code></td>
<td>
<p>Genes to be visualized, in the format of character vector.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_ct.table.id.type">ct.table.id.type</code></td>
<td>
<p>The gene id format in <code>data</code> should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_gene.id.type">gene.id.type</code></td>
<td>
<p>The gene id format of <code>gene.names</code>, should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_treatment">treatment</code></td>
<td>
<p>The column name to specify treatment groups.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_sample.id">sample.id</code></td>
<td>
<p>The column name to specify sample IDs.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_time">time</code></td>
<td>
<p>The column name to specify different time points.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_log.option">log.option</code></td>
<td>
<p>Logical option, whether to log2 transform the CPM as y-axis. Default = True.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_plot.save.to">plot.save.to</code></td>
<td>
<p>The address to save the plot from simplified cutoff combination with FDR of 0.01, 0.05, 0.1, and 0.2.</p>
</td></tr>
<tr><td><code id="plot_gene_+3A_input.type">input.type</code></td>
<td>
<p>One of <code>count</code> or <code>cpm</code> indicating what the input data type is. If <code>count</code>, the CPM of the input data will be
calculated using <code><a href="edgeR.html#topic+cpm">edgeR::cpm()</a></code>. Default = <code>count</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes the gene counts and returns a ggplot that shows gene expression variation over time or group.
</p>


<h3>Value</h3>

<p>The function returns a ggplot object.
</p>


<h3>References</h3>

<p>Xingpeng Li,Tatiana Gelaf Romer &amp; Aliyah Olaniyan, RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_gene(data = count_table,
anno = sample_annotation)

</code></pre>

<hr>
<h2 id='plot_heatmap.cfb'>Plot a CFB Heatmap</h2><span id='topic+plot_heatmap.cfb'></span>

<h3>Description</h3>

<p>An alias for <code>plot_heatmap.expr(annot, cpm, fill = "CFB", ...)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap.cfb(cpm, annot, title = "RVA CFB Heatmap", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heatmap.cfb_+3A_cpm">cpm</code></td>
<td>
<p>cpm data</p>
</td></tr>
<tr><td><code id="plot_heatmap.cfb_+3A_annot">annot</code></td>
<td>
<p>A long-format dataframe with any pertinent treatment data about
the samples. The only required column is one titled the <code>sample.id</code>
value with values matching the column names of sample IDs in <code>data</code>.
Additional columns can contain information such as treament
compounds, dates of sample collection, or dosage quantities.</p>
</td></tr>
<tr><td><code id="plot_heatmap.cfb_+3A_title">title</code></td>
<td>
<p>A title for the heatmap. Default = &quot;RVA Heatmap&quot;.</p>
</td></tr>
<tr><td><code id="plot_heatmap.cfb_+3A_...">...</code></td>
<td>
<p>pass over parameters</p>
</td></tr>
</table>

<hr>
<h2 id='plot_heatmap.cpm'>Plot a CPM Heatmap</h2><span id='topic+plot_heatmap.cpm'></span>

<h3>Description</h3>

<p>An alias for <code>plot_heatmap.expr(annot, cpm, fill = "CPM", ...)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap.cpm(cpm, annot, title = "RVA CPM Heatmap", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heatmap.cpm_+3A_cpm">cpm</code></td>
<td>
<p>cpm data</p>
</td></tr>
<tr><td><code id="plot_heatmap.cpm_+3A_annot">annot</code></td>
<td>
<p>A long-format dataframe with any pertinent treatment data about
the samples. The only required column is one titled the <code>sample.id</code>
value with values matching the column names of sample IDs in <code>data</code>.
Additional columns can contain information such as treament
compounds, dates of sample collection, or dosage quantities.</p>
</td></tr>
<tr><td><code id="plot_heatmap.cpm_+3A_title">title</code></td>
<td>
<p>A title for the heatmap. Default = &quot;RVA Heatmap&quot;.</p>
</td></tr>
<tr><td><code id="plot_heatmap.cpm_+3A_...">...</code></td>
<td>
<p>pass over parameters</p>
</td></tr>
</table>

<hr>
<h2 id='plot_heatmap.expr'>Plot Heatmap From Raw CPM</h2><span id='topic+plot_heatmap.expr'></span>

<h3>Description</h3>

<p>Create a heatmap with either CFB or CPM averaged across
individual samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap.expr(
  data = ~count,
  annot = ~meta,
  sample.id = "sample_id",
  annot.flags = c("day", "Treatment", "tissue"),
  ct.table.id.type = "ENSEMBL",
  gene.id.type = "SYMBOL",
  gene.names = NULL,
  gene.count = 10,
  title = "RVA Heatmap",
  fill = "CFB",
  baseline.flag = "day",
  baseline.val = "0",
  plot.save.to = NULL,
  input.type = "count"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heatmap.expr_+3A_data">data</code></td>
<td>
<p>A wide-format dataframe with geneid rownames, sample column
names, and fill data matching <code>input.type</code>.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_annot">annot</code></td>
<td>
<p>A long-format dataframe with any pertinent treatment data about
the samples. The only required column is one titled the <code>sample.id</code>
value with values matching the column names of sample IDs in <code>data</code>.
Additional columns can contain information such as treament
compounds, dates of sample collection, or dosage quantities.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_sample.id">sample.id</code></td>
<td>
<p>The column name to specify sample ID.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_annot.flags">annot.flags</code></td>
<td>
<p>A vector of column names corresponding to column names
in <code>annot</code> which will be used to define the x-axis for the heatmap.
Default = <code>c("day", "dose")</code>.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_ct.table.id.type">ct.table.id.type</code></td>
<td>
<p>The gene id format in <code>data</code> should be one of:
ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT, ENSEMBLTRANS,
ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI,
MAP, OMIM, ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ,
SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_gene.id.type">gene.id.type</code></td>
<td>
<p>The gene id format of <code>gene.names</code>, should be one of:
ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT, ENSEMBLTRANS,
ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI,
MAP, OMIM, ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ,
SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_gene.names">gene.names</code></td>
<td>
<p>A character vector or list of ensembl IDs for which to
display gene information. If <code>NULL</code>, all genes will be included.
Default = <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_gene.count">gene.count</code></td>
<td>
<p>The number of genes to include, where genes are selected
based on ranking by values in <code>fill</code>. Default = 10.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_title">title</code></td>
<td>
<p>A title for the heatmap. Default = &quot;RVA Heatmap&quot;.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_fill">fill</code></td>
<td>
<p>One of <code>c("CPM", "CFB")</code> to fill the heatmap cells with.
Default = &quot;CFB&quot;.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_baseline.flag">baseline.flag</code></td>
<td>
<p>A character vector of column names. If <code>fill = "CFB"</code>,
these columns in <code>annot</code> contain the values to compare
across. Ignored if <code>fill = "CPM"</code>. Default = &quot;timepoint&quot;.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_baseline.val">baseline.val</code></td>
<td>
<p>A character vector of values. This vector must be the
same length as <code>baseline.flag</code>, and the value at each index must
represent a value from the column given by the corresponding index
in <code>baseline.flag</code>. The samples corresponding to these values will
be used as a baseline when calculating CFB. Ignored if
<code>fill = "CPM"</code>. Default = &quot;Week 0&quot;.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_plot.save.to">plot.save.to</code></td>
<td>
<p>The address to save the heatmap plot.</p>
</td></tr>
<tr><td><code id="plot_heatmap.expr_+3A_input.type">input.type</code></td>
<td>
<p>One of <code>count</code> or <code>cpm</code> indicating what the input data type
is. If <code>count</code>, the CPM of the input data will be calculated using
<code><a href="edgeR.html#topic+cpm">edgeR::cpm()</a></code>. Default = <code>count</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes raw CPM data and returns both a list containing a data frame
with values based on the fill parameter and a heatmap plot.
</p>


<h3>Value</h3>

<p>The function returns a list with 2 items:
</p>
<table role = "presentation">
<tr><td><code>df.sub</code></td>
<td>
<p>&quot;A data frame of change from baselines values (fill = CFB in this example) for each gene id that is divided by a combination of treatment group and time point</p>
</td></tr>
<tr><td><code>gp</code></td>
<td>
<p>A Heatmap object from ComplexHeatmap which can be plotted</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li,Tatiana Gelaf Romer &amp; Aliyah Olaniyan, RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot &lt;- plot_heatmap.expr(data = count_table[,1:20],annot = sample_annotation[1:20,])
</code></pre>

<hr>
<h2 id='plot_pathway'>Pathway analysis and visualization</h2><span id='topic+plot_pathway'></span>

<h3>Description</h3>

<p>This is the function to do pathway enrichment analysis (and visualization) with rWikipathways (also KEGG, REACTOME &amp; Hallmark) from a summary statistics table generated by
differential expression analysis like <code>limma</code> or <code>DESeq2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_pathway(
  data = ~df,
  comp.names = NULL,
  gene.id.type = "ENSEMBL",
  FC.cutoff = 1.2,
  FDR.cutoff = 0.05,
  FCflag = "logFC",
  FDRflag = "adj.P.Val",
  Fisher.cutoff = 0.1,
  Fisher.up.cutoff = 0.1,
  Fisher.down.cutoff = 0.1,
  plot.save.to = NULL,
  pathway.db = "rWikiPathways",
  customized.pathways = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_pathway_+3A_data">data</code></td>
<td>
<p>A summary statistics table (data.frame) or <code>data.list</code> generated by DE analysis software like limma or DEseq2, where rownames are gene id.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_comp.names">comp.names</code></td>
<td>
<p>A character vector containing the comparison names corresponding to the same order of the <code>data.list</code>. Default = NULL.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_gene.id.type">gene.id.type</code></td>
<td>
<p>The gene id format in <code>data</code> should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_fc.cutoff">FC.cutoff</code></td>
<td>
<p>The fold change cutoff (numeric) selected to subset summary statistics table. Default = 1.5.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_fdr.cutoff">FDR.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) to subset summary statistics table. Default = 0.05.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_fcflag">FCflag</code></td>
<td>
<p>The column name (character) of fold change information, assuming the FC is log2 transformed. Default = &quot;logFC&quot;.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_fdrflag">FDRflag</code></td>
<td>
<p>The column name (character) of adjusted p value or FDR. Default = &quot;adj.P.Val&quot;.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_fisher.cutoff">Fisher.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) for the pathway enrichment analysis' Fisher's exact test with all determined
Differentially Expressed (DE) genes by <code>FC.cutoff</code> and <code>FDR.cutoff</code>.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_fisher.up.cutoff">Fisher.up.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) for the pathway enrichment analysis' Fisher's exact test with the upregulated gene set.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_fisher.down.cutoff">Fisher.down.cutoff</code></td>
<td>
<p>The FDR cutoff selected (numeric) for the pathway enrichment analysis' Fisher's exact test with the downregulated gene set.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_plot.save.to">plot.save.to</code></td>
<td>
<p>The address to save the plot from simplified cutoff combination with FDR of 0.01, 0.05, 0.1, and 0.2.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_pathway.db">pathway.db</code></td>
<td>
<p>The databse to be used for encrichment analysis. Can be one of the following, &quot;rWikiPathways&quot;, &quot;KEGG&quot;, &quot;REACTOME&quot;, &quot;Hallmark&quot;,&quot;rWikiPathways_aug_2020&quot;.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_customized.pathways">customized.pathways</code></td>
<td>
<p>the customized pathways in the format of two column dataframe (column name as &quot;gs_name&quot; and &quot;entrez_gene&quot;) to be used in analysis.</p>
</td></tr>
<tr><td><code id="plot_pathway_+3A_...">...</code></td>
<td>
<p>pass on variables</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes the summary statistics table and use user selected parameter based on check.cutoff to do pathway enrichment analysis
</p>


<h3>Value</h3>

<p>The function returns a list of 5 objects:
</p>
<table role = "presentation">
<tr><td><code>1</code></td>
<td>
<p>result table from directional pathway enrichment analysis</p>
</td></tr>
<tr><td><code>2</code></td>
<td>
<p>result table from non-directional pathway enrichment analysis</p>
</td></tr>
<tr><td><code>3</code></td>
<td>
<p>plot from directional pathway enrichment analysis</p>
</td></tr>
<tr><td><code>4</code></td>
<td>
<p>plot from non-directional pathway enrichment analysis</p>
</td></tr>
<tr><td><code>5</code></td>
<td>
<p>plot combining both directional and non-directional plot</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- plot_pathway(data = Sample_summary_statistics_table,
gene.id.type = "ENSEMBL",
FC.cutoff = 1.5,
p.cutoff = 0.05,
pathway.db = "rWikiPathways_aug_2020"
)
</code></pre>

<hr>
<h2 id='plot_qq'>Plot qqplot</h2><span id='topic+plot_qq'></span>

<h3>Description</h3>

<p>This function generates a QQ-plot object with confidence interval from summary statistics table generated by differential expression analysis
like <code>limma</code> or <code>DESeq2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_qq(
  data = data,
  comp.names = NULL,
  p.value.flag = "P.Value",
  ci = 0.95,
  plot.save.to = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_qq_+3A_data">data</code></td>
<td>
<p>Summary statistics table or a list that contains multiple summary statistics tables from limma or DEseq2, where each row is a gene.</p>
</td></tr>
<tr><td><code id="plot_qq_+3A_comp.names">comp.names</code></td>
<td>
<p>A character vector that contains the comparison names which correspond to the same order as <code>data</code>. No default.</p>
</td></tr>
<tr><td><code id="plot_qq_+3A_p.value.flag">p.value.flag</code></td>
<td>
<p>The column name of <code>P-VALUE</code> (NOT FDR, NO multiplicity adjusted p-value) in the summary statistics table. Default = &quot;P.Value&quot;.</p>
</td></tr>
<tr><td><code id="plot_qq_+3A_ci">ci</code></td>
<td>
<p>Confidence interval. Default = 0.95</p>
</td></tr>
<tr><td><code id="plot_qq_+3A_plot.save.to">plot.save.to</code></td>
<td>
<p>The file name and the address where to save the qq-plot &quot;~/address_to_folder/qqplot.png&quot;. Default = NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function produces the qqplot to evaluate the result from differential expression analysis. The output is a ggplot object.
</p>


<h3>Value</h3>

<p>The function return a ggplot object of qqplot
</p>


<h3>References</h3>

<p>Xingpeng Li &amp; Tatiana Gelaf Romer &amp; Olya Besedina, RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_qq(data = Sample_summary_statistics_table)
plot_qq(data = list(Sample_summary_statistics_table, Sample_summary_statistics_table1),
        comp.names = c("A","B"))

</code></pre>

<hr>
<h2 id='plot_volcano'>Plot volcanoplot</h2><span id='topic+plot_volcano'></span>

<h3>Description</h3>

<p>This function processes the summary statistics table generated by differential expression analysis
like <code>limma</code> or <code>DESeq2</code> to show on the volcano plot with the highlight gene set option (like disease
related genes from Disease vs Healthy comparison).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_volcano(
  data = data,
  comp.names = NULL,
  geneset = NULL,
  geneset.FCflag = "logFC",
  highlight.1 = NULL,
  highlight.2 = NULL,
  upcolor = "#FF0000",
  downcolor = "#0000FF",
  plot.save.to = NULL,
  xlim = c(-4, 4),
  ylim = c(0, 12),
  FCflag = "logFC",
  FDRflag = "adj.P.Val",
  highlight.FC.cutoff = 1.5,
  highlight.FDR.cutoff = 0.05,
  title = "Volcano plot",
  xlab = "log2 Fold Change",
  ylab = "log10(FDR)"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_volcano_+3A_data">data</code></td>
<td>
<p>Summary statistics table or a list contain multiple summary statistics tables from limma or DEseq2, where each row is a gene.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_comp.names">comp.names</code></td>
<td>
<p>A character vector that contains the comparison names which correspond to the same order as <code>data</code>. Required if <code>data</code> is list. No default.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_geneset">geneset</code></td>
<td>
<p>Summary statistic table that contains the genes which needed to be highlighted, the gene name format (in row names)
needs to be consistent with the main summary statistics table). For example, this summary statistics
table could be the output summary statistics table from the Disease vs Healthy comparison (Only contains
the subsetted significant genes to be highlighted).</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_geneset.fcflag">geneset.FCflag</code></td>
<td>
<p>The column name of fold change in <code>geneset</code>, Default = &quot;logFC&quot;.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_highlight.1">highlight.1</code></td>
<td>
<p>Genes to be highlighted, in the format of a vector consists of gene names. The gene name format
needs to be consistent to the main summary statistics table.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_highlight.2">highlight.2</code></td>
<td>
<p>Genes to be highlighted, in the format of a vector consists of gene names. The gene name format
needs to be consistent to the main summary statistics table.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_upcolor">upcolor</code></td>
<td>
<p>The color of the gene names in <code>highlight.1</code> or the positive fold change gene
in <code>geneset</code>, default = &quot;#FDE725FF&quot; (viridis color palette).</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_downcolor">downcolor</code></td>
<td>
<p>The color of the gene names in <code>highlight.2</code> or the negative fold change gene
in <code>geneset</code>, default = &quot;#440154FF&quot; (viridis color palette).</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_plot.save.to">plot.save.to</code></td>
<td>
<p>The file name and address where to save the volcano plot, e.g. &quot;~/address_to_folder/volcano_plot.png&quot;.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_xlim">xlim</code></td>
<td>
<p>Range of x axis. Default = <code>c(-3,3)</code>.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_ylim">ylim</code></td>
<td>
<p>Range of x axis. Default = <code>c(0,6)</code>.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_fcflag">FCflag</code></td>
<td>
<p>Column name of log2FC in the summary statistics table. Default = &quot;logFC&quot;.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_fdrflag">FDRflag</code></td>
<td>
<p>Column name of FDR in the summary statistics table. Default = &quot;adj.P.Val&quot;.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_highlight.fc.cutoff">highlight.FC.cutoff</code></td>
<td>
<p>Fold change cutoff line want to be shown on the plot. Default = 1.5.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_highlight.fdr.cutoff">highlight.FDR.cutoff</code></td>
<td>
<p>FDR cutoff shades want to be shown on the plot. Default = 0.05.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_title">title</code></td>
<td>
<p>The plot title. Default &quot;Volcano plot&quot;.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_xlab">xlab</code></td>
<td>
<p>The label for x-axis. Default &quot;log2 Fold Change&quot;.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_ylab">ylab</code></td>
<td>
<p>The label for y-axis. Default &quot;log10(FDR)&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes the summary statistics table and returns a ggplot, with the option to highlight
genes, e.g. disease signature genes, the genes which are up-regulated and down-regulated in diseased subjects.
</p>


<h3>Value</h3>

<p>The function return a volcano plot as a ggplot object.
</p>


<h3>References</h3>

<p>Xingpeng Li &amp; Tatiana Gelaf Romer &amp; Olya Besedina, RVA - RNAseq Visualization Automation tool.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_volcano(data = Sample_summary_statistics_table,
             geneset = Sample_disease_gene_set)

plot_volcano(data = list(Sample_summary_statistics_table, Sample_summary_statistics_table1),
            comp.names = c("A", "B"),
            geneset = Sample_disease_gene_set)
</code></pre>

<hr>
<h2 id='prettyGraphs'>Pretty Graphs</h2><span id='topic+prettyGraphs'></span>

<h3>Description</h3>

<p>Special cases where list input and at least one treatment has signal but others don't.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prettyGraphs(vizdf, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prettyGraphs_+3A_vizdf">vizdf</code></td>
<td>
<p>A dataframes of enriched pathways.</p>
</td></tr>
<tr><td><code id="prettyGraphs_+3A_...">...</code></td>
<td>
<p>pass on variables</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pretty Graphs is a function specifically meant to be in cases where one of the input treatments meet cutoff, but
one or more of the other treatments don't meet the cutoff values. This is important so that ggplot doesn't throw any errors.
</p>


<h3>Value</h3>

<p>Returns a dataframe.
</p>


<h3>References</h3>

<p>Xingpeng Li &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='produce.cutoff.message'>Create a message about fold change and pvalues used to produce the plot.</h2><span id='topic+produce.cutoff.message'></span>

<h3>Description</h3>

<p>This function processes summary statistics table generated by differential expression analysis
like <code>limma</code> or <code>DESeq2</code> and produces a message about pvalues and fold change used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>produce.cutoff.message(
  data,
  FCmin,
  FCmax,
  FCstep,
  FDRflag,
  p.min,
  p.max,
  p.step
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="produce.cutoff.message_+3A_data">data</code></td>
<td>
<p>Summary statistics table from limma or DEseq2, where each row is a gene.</p>
</td></tr>
<tr><td><code id="produce.cutoff.message_+3A_fcmin">FCmin</code></td>
<td>
<p>The minimum starting fold change cutoff to be checked, so the minimum fold change cutoff to be evaluated
will be FCmin + FCstep, FCmin default = 1.</p>
</td></tr>
<tr><td><code id="produce.cutoff.message_+3A_fcmax">FCmax</code></td>
<td>
<p>The maximum fold change cutoff to be checked, default = 2.</p>
</td></tr>
<tr><td><code id="produce.cutoff.message_+3A_fcstep">FCstep</code></td>
<td>
<p>The step from the minimum to maximum fold change cutoff, one step increase at a time, default = 0.01.</p>
</td></tr>
<tr><td><code id="produce.cutoff.message_+3A_fdrflag">FDRflag</code></td>
<td>
<p>The column name of the False Discovery Rate (FDR) in the summary statistics table.</p>
</td></tr>
<tr><td><code id="produce.cutoff.message_+3A_p.min">p.min</code></td>
<td>
<p>The minimum starting FDR cutoff to be checked, so the minimum fold change cutoff to be evaluated
will be p.min + p.step, p.min default = 0.</p>
</td></tr>
<tr><td><code id="produce.cutoff.message_+3A_p.max">p.max</code></td>
<td>
<p>The maximum FDR cutoff to be checked, default = 0.2.</p>
</td></tr>
<tr><td><code id="produce.cutoff.message_+3A_p.step">p.step</code></td>
<td>
<p>The step from the minimum to maximum fold change cutoff, one step increase at a time, default = 0.005.</p>
</td></tr>
</table>

<hr>
<h2 id='produce.cutoff.warning'>Create a warning about pvalue or FDR minimum value</h2><span id='topic+produce.cutoff.warning'></span>

<h3>Description</h3>

<p>This function processes summary statistics table generated by differential expression analysis
like <code>limma</code> or <code>DESeq2</code> and produces a warning about pvalue or FDR minimum value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>produce.cutoff.warning(data, FDRflag)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="produce.cutoff.warning_+3A_data">data</code></td>
<td>
<p>Summary statistics table from limma or DEseq2, where each row is a gene.</p>
</td></tr>
<tr><td><code id="produce.cutoff.warning_+3A_fdrflag">FDRflag</code></td>
<td>
<p>The column name of the False Discovery Rate (FDR) in the summary statistics table.</p>
</td></tr>
</table>

<hr>
<h2 id='reformat.ensembl'>Reformat Ensembl GeneIDs</h2><span id='topic+reformat.ensembl'></span>

<h3>Description</h3>

<p>This is the function to exclude the version number from the input ensembl type gene ids.
</p>
<p>This is the function to exclude the version number from the input ensembl type gene ids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reformat.ensembl(logcpm, ct.table.id.type)

reformat.ensembl(logcpm, ct.table.id.type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reformat.ensembl_+3A_logcpm">logcpm</code></td>
<td>
<p>The input count table transformed into log counts per million.</p>
</td></tr>
<tr><td><code id="reformat.ensembl_+3A_ct.table.id.type">ct.table.id.type</code></td>
<td>
<p>The gene id format in <code>logcpm</code> should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td></tr>
</table>

<hr>
<h2 id='sample_annotation'>This is data to be included in package</h2><span id='topic+sample_annotation'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_annotation
</code></pre>


<h3>Format</h3>

<p>Sample annotation document
</p>

<dl>
<dt>sample_id</dt><dd><p>sample name</p>
</dd>
<dt>tissue</dt><dd><p>tissue for comparison</p>
</dd>
<dt>subject_id</dt><dd><p>subject id</p>
</dd>
<dt>day</dt><dd><p>time points</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='sample_count_cpm'>This is data to be included in package</h2><span id='topic+sample_count_cpm'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_count_cpm
</code></pre>


<h3>Format</h3>

<p>An example cpm table where row names are gene ID, each column is a sample
</p>

<dl>
<dt>counttable</dt><dd><p>count cpm table</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='Sample_disease_gene_set'>This is data to be included in package</h2><span id='topic+Sample_disease_gene_set'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sample_disease_gene_set
</code></pre>


<h3>Format</h3>

<p>An example disease gene set from summary statistics table as dataframe, row names are gene ID
the summary statistics can be calculated from disease vs healthy, which is this example.
</p>

<dl>
<dt>logFC</dt><dd><p>log2 fold change from comparison</p>
</dd>
<dt>AveExpr</dt><dd><p>Average expression for this gene</p>
</dd>
<dt>P.Value</dt><dd><p>p value</p>
</dd>
<dt>adj.P.Val</dt><dd><p>adjusted p value or FDR</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='Sample_summary_statistics_table'>This is data to be included in package</h2><span id='topic+Sample_summary_statistics_table'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sample_summary_statistics_table
</code></pre>


<h3>Format</h3>

<p>An example summary statistics table as dataframe, row names are gene ID
</p>

<dl>
<dt>logFC</dt><dd><p>log2 fold change from comparison</p>
</dd>
<dt>AveExpr</dt><dd><p>Average expression for this gene</p>
</dd>
<dt>P.Value</dt><dd><p>p value</p>
</dd>
<dt>adj.P.Val</dt><dd><p>adjusted p value or FDR</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='Sample_summary_statistics_table1'>This is data to be included in package</h2><span id='topic+Sample_summary_statistics_table1'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sample_summary_statistics_table1
</code></pre>


<h3>Format</h3>

<p>Second example summary statistics table as dataframe, row names are gene ID
</p>

<dl>
<dt>logFC</dt><dd><p>log2 fold change from comparison</p>
</dd>
<dt>AveExpr</dt><dd><p>Average expression for this gene</p>
</dd>
<dt>P.Value</dt><dd><p>p value</p>
</dd>
<dt>adj.P.Val</dt><dd><p>adjusted p value or FDR</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='secondCutoffErr'>Second Cutoff Error</h2><span id='topic+secondCutoffErr'></span>

<h3>Description</h3>

<p>The function takes in a list of dataframe, comp names and a specified type, to output a dataframe styled for ggplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>secondCutoffErr(df, comp.names, TypeQ = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="secondCutoffErr_+3A_df">df</code></td>
<td>
<p>A list of dataframes.</p>
</td></tr>
<tr><td><code id="secondCutoffErr_+3A_comp.names">comp.names</code></td>
<td>
<p>a character vector contain the comparison names corresponding to the same order to the <code>dat.list</code>. default = NULL.</p>
</td></tr>
<tr><td><code id="secondCutoffErr_+3A_typeq">TypeQ</code></td>
<td>
<p>If type = 1(default) return directional null plot. If type = 2 return non directional null plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>secondCutoffErr is a function specifically meant to be used for list inputs. It is used for cases where after
applying filter to the data, one of the comparison ID gets left out, this adversely effects the ggplot
</p>


<h3>Value</h3>

<p>Returns a dataframe.
</p>


<h3>References</h3>

<p>Xingpeng Li &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='transform.geneid'>Transform GeneIDs</h2><span id='topic+transform.geneid'></span>

<h3>Description</h3>

<p>This is the function to transform the input gene id type to another gene id type.
</p>
<p>This is the function to transform the input gene id type to another gene id type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'geneid'
transform(gene.names, from = ~gene.id.type, to = ~ct.table.id.type)

## S3 method for class 'geneid'
transform(gene.names, from = ~gene.id.type, to = ~ct.table.id.type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="transform.geneid_+3A_gene.names">gene.names</code></td>
<td>
<p>Genes,in the format of character vector, to be transformed.</p>
</td></tr>
<tr><td><code id="transform.geneid_+3A_from">from</code></td>
<td>
<p>The gene id format of <code>gene.names</code>, should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td></tr>
<tr><td><code id="transform.geneid_+3A_to">to</code></td>
<td>
<p>The new gene id format should be one of: ACCNUM, ALIAS, ENSEMBL, ENSEMBLPROT,
ENSEMBLTRANS, ENTREZID, ENZYME, EVIDENCE, EVIDENCEALL, GENENAME, GO, GOALL, IPI, MAP, OMIM,
ONTOLOGY, ONTOLOGYALL, PATH, PFAM, PMID, PROSITE, REFSEQ, SYMBOL, UCSCKG, UNIGENE, UNIPROT.</p>
</td></tr>
</table>

<hr>
<h2 id='validate.annot'>Validate Annotation Table</h2><span id='topic+validate.annot'></span>

<h3>Description</h3>

<p>Ensure that an annotation has all of the required columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.annot(
  data,
  annot,
  annot.flags,
  sample.id,
  fill = "CPM",
  baseline.flag = NULL,
  baseline.val = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.annot_+3A_data">data</code></td>
<td>
<p>The input count data.</p>
</td></tr>
<tr><td><code id="validate.annot_+3A_annot">annot</code></td>
<td>
<p>The annotation dataframe.</p>
</td></tr>
<tr><td><code id="validate.annot_+3A_annot.flags">annot.flags</code></td>
<td>
<p>The vector of annotation flags passed by the user.</p>
</td></tr>
<tr><td><code id="validate.annot_+3A_sample.id">sample.id</code></td>
<td>
<p>Sample id label to check if in annot.</p>
</td></tr>
<tr><td><code id="validate.annot_+3A_fill">fill</code></td>
<td>
<p>The fill value indicated by the user,&quot;count&quot; or &quot;CPM&quot;.</p>
</td></tr>
<tr><td><code id="validate.annot_+3A_baseline.flag">baseline.flag</code></td>
<td>
<p>The baseline.flag passed by the user.</p>
</td></tr>
<tr><td><code id="validate.annot_+3A_baseline.val">baseline.val</code></td>
<td>
<p>The baseline value passed by the user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will check the following:
</p>

<ul>
<li><p> The <code>annot.flags</code> values are columns in <code>annot</code>
</p>
</li>
<li><p> If <code>fill</code> = &quot;cfb&quot;: validate the <code>baseline.flag</code> and
<code>baseline.val</code> parameters.
</p>
</li>
<li> <p><code>sample.id</code> is a column in <code>annot</code>.
</p>
</li></ul>



<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.baseline'>Validate Baseline Values</h2><span id='topic+validate.baseline'></span>

<h3>Description</h3>

<p>Ensures that user-input <code>baseline.val</code> and <code>baseline.flag</code>
parameters are valid with respect to the <code>annot</code> dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.baseline(annot, baseline.val, baseline.flag)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.baseline_+3A_annot">annot</code></td>
<td>
<p>The annotation dataframe.</p>
</td></tr>
<tr><td><code id="validate.baseline_+3A_baseline.val">baseline.val</code></td>
<td>
<p>The baseline value passed by the user.</p>
</td></tr>
<tr><td><code id="validate.baseline_+3A_baseline.flag">baseline.flag</code></td>
<td>
<p>The baseline.flag passed by the user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Specifically, validates that <code>baseline.flag</code> value(s) are columns
in <code>annot</code>, and that <code>baseline.val</code> value(s) occur at least once in
their respective <code>baseline.flag</code> columns.
</p>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.col.types'>Check Summary Statistics Required Column Types</h2><span id='topic+validate.col.types'></span>

<h3>Description</h3>

<p><code>FCflag</code> and <code>FDRflag</code> must be numeric.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.col.types(datin, name = 1, flags)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.col.types_+3A_datin">datin</code></td>
<td>
<p>the summary statistics file.</p>
</td></tr>
<tr><td><code id="validate.col.types_+3A_name">name</code></td>
<td>
<p>summary statistics file position indicator</p>
</td></tr>
<tr><td><code id="validate.col.types_+3A_flags">flags</code></td>
<td>
<p><code>FCflag</code> or <code>FDRflag</code> to be checked</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.comp.names'>Validate Comp Names</h2><span id='topic+validate.comp.names'></span>

<h3>Description</h3>

<p>This function ensures that when a list of data frames are used as input the
the number of comp names are the same as the number of data frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.comp.names(comp.names, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.comp.names_+3A_comp.names">comp.names</code></td>
<td>
<p>a character vector contain the comparison names corresponding to the same order to the <code>dat.list</code>. default = NULL.</p>
</td></tr>
<tr><td><code id="validate.comp.names_+3A_data">data</code></td>
<td>
<p>summary statistics table (data.frame) from limma or DEseq2, where rownames are gene id.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.data'>Validate Data Input</h2><span id='topic+validate.data'></span>

<h3>Description</h3>

<p>Ensures that the data input has the required formatting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.data(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.data_+3A_data">data</code></td>
<td>
<p>The wide-format dataframe with input data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Specifically, checks if <code>data</code> has rownaems and that all other
columns can be coerced to numeric.
</p>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.data.annot'>Validate Data in the Context of Annotation</h2><span id='topic+validate.data.annot'></span>

<h3>Description</h3>

<p>Ensures that the annotation file matches the data file with
respect to sample IDs. Throws warnings if there are discrepencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.data.annot(data, annot, sample.id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.data.annot_+3A_data">data</code></td>
<td>
<p>input data</p>
</td></tr>
<tr><td><code id="validate.data.annot_+3A_annot">annot</code></td>
<td>
<p>annotation file</p>
</td></tr>
<tr><td><code id="validate.data.annot_+3A_sample.id">sample.id</code></td>
<td>
<p>sample id in the input</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.FC'>Validate Foldchange</h2><span id='topic+validate.FC'></span>

<h3>Description</h3>

<p>This function ensures the fold change minimum, maximum, and step
are valid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.FC(FCmin, FCmax, FCstep)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.FC_+3A_fcmin">FCmin</code></td>
<td>
<p>The minimum starting fold change cutoff to be checked, so the minimum fold change cutoff to be evaluated
will be FCmin + FCstep, FCmin default = 1.</p>
</td></tr>
<tr><td><code id="validate.FC_+3A_fcmax">FCmax</code></td>
<td>
<p>The maximum fold change cutoff to be checked, default = 2.</p>
</td></tr>
<tr><td><code id="validate.FC_+3A_fcstep">FCstep</code></td>
<td>
<p>The step from the minimum to maximum fold change cutoff, one step increase at a time, default = 0.01.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Specifically it checks that the FCmax is greater than the FCmin,
that at least 1 FCstep can fit within the FCmax and FCmin, that
FCmax and FCmin values are non-negative, and that FCstep is positive.
</p>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.flag'>Validate Flag Value Is Valid</h2><span id='topic+validate.flag'></span>

<h3>Description</h3>

<p>Enures that the <code>value</code> is one of <code>Options</code> and throws an error
otherwise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.flag(value, name, Options)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.flag_+3A_value">value</code></td>
<td>
<p>The user-input value for the parameter</p>
</td></tr>
<tr><td><code id="validate.flag_+3A_name">name</code></td>
<td>
<p>The name of the parameter to be displayed in the error</p>
</td></tr>
<tr><td><code id="validate.flag_+3A_options">Options</code></td>
<td>
<p>A vector of valid values for <code>value</code></p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.genes.present'>Validate genes present</h2><span id='topic+validate.genes.present'></span>

<h3>Description</h3>

<p>Checks how many of the gene id's in the dataset are there in the geneset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.genes.present(data.genes, geneset)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.genes.present_+3A_data.genes">data.genes</code></td>
<td>
<p>The gene id's.</p>
</td></tr>
<tr><td><code id="validate.genes.present_+3A_geneset">geneset</code></td>
<td>
<p>a summary statistic table contain the genes want to be highlighted, the gene name format (in row names)
needs to be consistent to the main summary statistics table). For example, this summary statistics
table coulb be the output summary statistics table from Disease vs Healthy comparison (Only contain
the subsetted significant genes want to be highlighted).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.geneset'>Validate Geneset</h2><span id='topic+validate.geneset'></span>

<h3>Description</h3>

<p>This function ensures that the input geneset to check.cutoff
is formatted properly and in a usable form.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.geneset(data, geneset, highlight.1, highlight.2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.geneset_+3A_data">data</code></td>
<td>
<p>summary statistics table or a list contain multiple summary statistics tables from limma or DEseq2, where each row is a gene.</p>
</td></tr>
<tr><td><code id="validate.geneset_+3A_geneset">geneset</code></td>
<td>
<p>a summary statistic table contain the genes want to be highlighted, the gene name format (in row names)
needs to be consistent to the main summary statistics table). For example, this summary statistics
table coulb be the output summary statistics table from Disease vs Healthy comparison (Only contain
the subsetted significant genes want to be highlighted).</p>
</td></tr>
<tr><td><code id="validate.geneset_+3A_highlight.1">highlight.1</code></td>
<td>
<p>genes want to be highlighted, in the format of a vector consists of gene names. The gene name format
needs to be consistent to the main summary statistics table.</p>
</td></tr>
<tr><td><code id="validate.geneset_+3A_highlight.2">highlight.2</code></td>
<td>
<p>genes want to be highlighted, in the format of a vector consists of gene names. The gene name format
needs to be consistent to the main summary statistics table.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function ensures that only a dataframe or vectors are supplied,
that at least one or the other is supplied, and that their formatting
is correct if supplied. It also checks if any of the genes overlap
with the genes in the datanames.
</p>


<h3>Value</h3>

<p>A character value indicating if the geneset was passed as a
dataframe (<code>df</code>) or two vectors (<code>vec</code>), if a list is input
the number of returned values equal the length of the list
</p>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.numeric'>Validate Numeric Column</h2><span id='topic+validate.numeric'></span>

<h3>Description</h3>

<p>Ensures that a column in a dataframe which must be numeric is
numeric and throws an error otherwise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.numeric(datin, col, name = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.numeric_+3A_datin">datin</code></td>
<td>
<p>The data in question.</p>
</td></tr>
<tr><td><code id="validate.numeric_+3A_col">col</code></td>
<td>
<p>The column to validate as numeric.</p>
</td></tr>
<tr><td><code id="validate.numeric_+3A_name">name</code></td>
<td>
<p>the position of dataset</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This specifically checks if any of the values in the column can
be coerced as numeric.
</p>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.pathways.db'>Validate Pathways DB</h2><span id='topic+validate.pathways.db'></span>

<h3>Description</h3>

<p>To ensure selected db name is correct.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.pathways.db(pathway.db, customized.pathways)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.pathways.db_+3A_pathway.db">pathway.db</code></td>
<td>
<p>The databse to be used for encrichment analysis. Can be one of the following, &quot;rWikiPathways&quot;, &quot;KEGG&quot;, &quot;REACTOME&quot;, &quot;Hallmark&quot;,&quot;rWikiPathways_aug_2020&quot;</p>
</td></tr>
<tr><td><code id="validate.pathways.db_+3A_customized.pathways">customized.pathways</code></td>
<td>
<p>the customized pathways in the format of two column dataframe (column name as &quot;gs_name&quot; and &quot;entrez_gene&quot;) to be used in analysis</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.pval.range'>Validate P-value Range</h2><span id='topic+validate.pval.range'></span>

<h3>Description</h3>

<p>Error-handling for invalid p-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.pval.range(pval, name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.pval.range_+3A_pval">pval</code></td>
<td>
<p>The pvalue</p>
</td></tr>
<tr><td><code id="validate.pval.range_+3A_name">name</code></td>
<td>
<p>The name of the value to include in the error.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.pvalflag'>Validate pval flag</h2><span id='topic+validate.pvalflag'></span>

<h3>Description</h3>

<p>To ensure p value flags are the same accross datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.pvalflag(data, value)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.pvalflag_+3A_data">data</code></td>
<td>
<p>A list of summary statistics table (data.frame) from limma or DEseq2, where rownames are gene id.</p>
</td></tr>
<tr><td><code id="validate.pvalflag_+3A_value">value</code></td>
<td>
<p>P value flag.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.pvals'>Validate Pvalues</h2><span id='topic+validate.pvals'></span>

<h3>Description</h3>

<p>This function ensures the fold change minimum, maximum, and step
are valid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.pvals(p.min, p.max, p.step)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.pvals_+3A_p.min">p.min</code></td>
<td>
<p>The minimum starting FDR cutoff to be checked, so the minimum fold change cutoff to be evaluated
will be p.min + p.step, p.min default = 0.</p>
</td></tr>
<tr><td><code id="validate.pvals_+3A_p.max">p.max</code></td>
<td>
<p>The maximum FDR cutoff to be checked, default = 0.2.</p>
</td></tr>
<tr><td><code id="validate.pvals_+3A_p.step">p.step</code></td>
<td>
<p>The step from the minimum to maximum fold change cutoff, one step increase at a time, default = 0.005.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Specifically it checks that the pvalues are between 0-1, and that
at least 1 <code>p.step</code> fits within the <code>p.min</code> and <code>p.max</code> bounds and
is positive.
</p>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.single.table.isnotlist'>Validate Single Table is not list</h2><span id='topic+validate.single.table.isnotlist'></span>

<h3>Description</h3>

<p>Makes sure the summary table being input is of the right class and format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.single.table.isnotlist(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.single.table.isnotlist_+3A_data">data</code></td>
<td>
<p>summary statistics table (data.frame) from limma or DEseq2, where rownames are gene id.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.stats'>Validate Summary Statistics File</h2><span id='topic+validate.stats'></span>

<h3>Description</h3>

<p>Check for required column names and types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.stats(datin, name = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.stats_+3A_datin">datin</code></td>
<td>
<p>the summary statistics file.</p>
</td></tr>
<tr><td><code id="validate.stats_+3A_name">name</code></td>
<td>
<p>summary statistics file position indicator</p>
</td></tr>
<tr><td><code id="validate.stats_+3A_...">...</code></td>
<td>
<p>pass on variables</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='validate.stats.cols'>Check Summary Statistics Required Columns</h2><span id='topic+validate.stats.cols'></span>

<h3>Description</h3>

<p>Required columns are <code>FCflag</code> and <code>FDRflag</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate.stats.cols(datin, name = 1, req.cols)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate.stats.cols_+3A_datin">datin</code></td>
<td>
<p>the summary statistics file.</p>
</td></tr>
<tr><td><code id="validate.stats.cols_+3A_name">name</code></td>
<td>
<p>summary statistics file position indicator</p>
</td></tr>
<tr><td><code id="validate.stats.cols_+3A_req.cols">req.cols</code></td>
<td>
<p>required column names of <code>FCflag</code> and <code>FDRflag</code> pass on from validate.stats</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xingpeng Li, Tatiana Gelaf Romer &amp; Siddhartha Pachhai RVA - RNAseq Visualization Automation tool.
</p>

<hr>
<h2 id='wpA2020'>This is data to be included in package</h2><span id='topic+wpA2020'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wpA2020
</code></pre>


<h3>Format</h3>

<p>Rwikipathway data downloaded version 2020
</p>

<dl>
<dt>name</dt><dd><p>pathway name</p>
</dd>
<dt>version</dt><dd><p>version</p>
</dd>
<dt>wpid</dt><dd><p>pathway id</p>
</dd>
<dt>org</dt><dd><p>host name</p>
</dd>
</dl>
<p>...

</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
