<!DOCTYPE html><html><head><title>Help for package rvHPDT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rvHPDT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#convert.factors.to.strings.in.dataframe'><p>Convert data.frame columns from factors to characters</p></a></li>
<li><a href='#fullPedigree'><p>Internal function.</p></a></li>
<li><a href='#HapMendlianCheck'><p>Internal function.</p></a></li>
<li><a href='#mydaoshu'><p>Internal function.</p></a></li>
<li><a href='#mypaste'><p>Internal function.</p></a></li>
<li><a href='#randomPDT'><p>Internal function.</p></a></li>
<li><a href='#rchild'><p>Internal function.</p></a></li>
<li><a href='#rhapPDT'><p>Calling haplotype-based and variant-based pedigree disequilibrium test for rare variants in pedigrees.</p></a></li>
<li><a href='#rvPDT.test'><p>Variants-based pedigree disequilibrium test for rare variants in pedigrees.</p></a></li>
<li><a href='#rvPDT.test.permu'><p>Variants-based pedigree disequilibrium test for rare variants in pedigrees.</p></a></li>
<li><a href='#rvPDT.test.sub'><p>Internal function.</p></a></li>
<li><a href='#whap.prehap'><p>Prepare haplotype pairs for hPDT tests in pedigree data.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calling Haplotype-Based and Variant-Based Pedigree
Disequilibrium Test for Rare Variants in Pedigrees</td>
</tr>
<tr>
<td>Version:</td>
<td>4.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-5-9</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Guo &lt;wei.guo3@nih.gov&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Guo &lt;wei.guo3@nih.gov&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>gtools, R(&ge; 2.15.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils</td>
</tr>
<tr>
<td>Description:</td>
<td>To detecting rare variants for binary traits using general pedigrees, the pedigree disequilibrium tests are proposed by collapsing rare haplotypes/variants with/without weights. To run the test, MERLIN is needed in Linux for haplotyping.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-12 14:03:51 UTC; guow4</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-12 21:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='convert.factors.to.strings.in.dataframe'>Convert data.frame columns from factors to characters</h2><span id='topic+convert.factors.to.strings.in.dataframe'></span>

<h3>Description</h3>

<p>Convert data.frame columns from factors to characters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  
convert.factors.to.strings.in.dataframe(dataframe)
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr><td><code id="convert.factors.to.strings.in.dataframe_+3A_dataframe">dataframe</code></td>
<td>
<p> dataframe with columns of factors</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>dataframe</code></td>
<td>
<p> dataframe with columns of characters</p>
</td></tr>
</table>

<hr>
<h2 id='fullPedigree'>Internal function.</h2><span id='topic+fullPedigree'></span>

<h3>Description</h3>

<p>Complete family members as requested by Merlin software.
</p>

<hr>
<h2 id='HapMendlianCheck'>Internal function.</h2><span id='topic+HapMendlianCheck'></span>

<h3>Description</h3>

<p>Check Mendlian error in families.
</p>

<hr>
<h2 id='mydaoshu'>Internal function.</h2><span id='topic+mydaoshu'></span>

<h3>Description</h3>

<p>function of 1/x
</p>

<hr>
<h2 id='mypaste'>Internal function.</h2><span id='topic+mypaste'></span>

<h3>Description</h3>

<p>Modified paste function.
</p>

<hr>
<h2 id='randomPDT'>Internal function.</h2><span id='topic+randomPDT'></span>

<h3>Description</h3>

<p>Calculate PDT statistics by permuting the transmission and non-transmission status for each child based on parents' genotype.
</p>

<hr>
<h2 id='rchild'>Internal function.</h2><span id='topic+rchild'></span>

<h3>Description</h3>

<p>Generate child's genotype by permuting the transmission and non-transmission status based on parents' genotype.
</p>

<hr>
<h2 id='rhapPDT'>Calling haplotype-based and variant-based pedigree disequilibrium test for rare variants in pedigrees.</h2><span id='topic+rhapPDT'></span>

<h3>Description</h3>

<p>To detecting rare variants for binary traits using general pedigrees, the pedigree disequilibrium tests are proposed by collapsing rare haplotypes/variants with/without weights.</p>


<h3>Usage</h3>

<pre><code class='language-R'> rhapPDT(ped, map, aff=2, unaff=1, mu=1.04,
merlinFN.prefix="merlin", nperm=1000, trace=TRUE) 
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr><td><code id="rhapPDT_+3A_ped">ped</code></td>
<td>
<p>input data, has same format with PLINK but having column names. The PED file is a white-space (space or tab) delimited file, and the first six columns are mandatory: 
FID: Family ID;
IID: Individual ID;
FA: Paternal ID;
MO: Maternal ID;
SEX: Sex (1=male; 2=female; other=unknown);
PHENO: Phenotype; 
Genotypes (column 7 onwards) should also be white-space delimited; they are coded as 0, 1 and 2, indicating the number of coding allele, and NA is for missing genotype. </p>
</td></tr> 
<tr><td><code id="rhapPDT_+3A_map">map</code></td>
<td>
<p>input data, has same format with MAP file required by MERLIN. The MAP file is a white-space (space or tab) delimited file with 3 columns as follows,  
CHROMOSOME: chromosome (1-22, X, Y or 0 if unplaced)  
MARKER: marker name in PED file that is usually rs# or snp identifier         
POSITION: Genetic distance (morgans)
The data file and map file can include different sets of markers, but markers that are absent from the map file will be ignored by MERLIN. 
</p>
</td></tr> 
<tr><td><code id="rhapPDT_+3A_aff">aff</code></td>
<td>
<p>indicates the values that represents affected status in &quot;PHENO&quot; column of PED data; default is 2.</p>
</td></tr>
<tr><td><code id="rhapPDT_+3A_unaff">unaff</code></td>
<td>
<p>indicates the values that represents unaffected status in &quot;PHENO&quot; column of PED data; default is 1.</p>
</td></tr>  
<tr><td><code id="rhapPDT_+3A_mu">mu</code></td>
<td>
<p>indicates mu value that defines causal in the training data; default is 1.04.</p>
</td></tr>  
<tr><td><code id="rhapPDT_+3A_merlinfn.prefix">merlinFN.prefix</code></td>
<td>
<p> Requests that output file of MERLIN names should be derived from outFN.prefix. For example, when it is set to be &quot;merlin&quot; as default, estimated haplotypes should be stored in a file called merlin.chr.</p>
</td></tr>   
<tr><td><code id="rhapPDT_+3A_nperm">nperm</code></td>
<td>
<p>indicates the times of permutation; default is 1000.</p>
</td></tr>     
<tr><td><code id="rhapPDT_+3A_trace">trace</code></td>
<td>
<p> Indicates whether or not the intermediate outcomes should be printed; default is FALSE.</p>
</td></tr>  
</table>


<h3>Value</h3>

 
<table>
<tr><td><code>hPDT_v0</code></td>
<td>
<p> P value of unweighted haplotype PDT test statistic.</p>
</td></tr>    
<tr><td><code>hPDT_v1</code></td>
<td>
<p> P value of weighted haplotype PDT test statistic.</p>
</td></tr> 
<tr><td><code>rvPDT_v0</code></td>
<td>
<p> P value of unweighted rvPDT test statistic.</p>
</td></tr>   
<tr><td><code>rvPDT_v1</code></td>
<td>
<p> P value of weighted rvPDT test statistic.</p>
</td></tr>  
</table>


<h3>References</h3>

    
<p>Guo W , Shugart YY, Does Haplotype-based Collapsing Tests Gain More Power than Variant-based Collapsing Tests for Detecting Rare Variants in Pedigrees  
(manuscript). 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#ped&lt;-read.table("MLIP.ped",head=1,stringsAsFactors=FALSE)
#map&lt;-read.table("MLIP.map",head=1,stringsAsFactors=FALSE)  
#test&lt;-rhapPDT(ped, map, trace=TRUE) 
#test 
#$hPDT_v0
#[1] 0.4231359

#$hPDT_v1
#[1] 0.1481145

#$rvPDT_v0
#[1] 0.03237073

#$rvPDT_v1
#[1] 0.162997
</code></pre>

<hr>
<h2 id='rvPDT.test'>Variants-based pedigree disequilibrium test for rare variants in pedigrees.</h2><span id='topic+rvPDT.test'></span>

<h3>Description</h3>

<p>To detecting rare variants for binary traits using general pedigrees, the pedigree disequilibrium tests are extended by collapsing rare variants with/without weights.</p>


<h3>Usage</h3>

<pre><code class='language-R'> rvPDT.test(seed=NULL,ped, aff=2,unaff=1, snpCol, hfreq=NULL,
 training=0.3, mu=1.28,useFamWeight=TRUE,trace=FALSE)   
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr><td><code id="rvPDT.test_+3A_seed">seed</code></td>
<td>
<p>indicates the seed for randomly selectiong training data.</p>
</td></tr> 
<tr><td><code id="rvPDT.test_+3A_ped">ped</code></td>
<td>
<p>input data, has same format with PLINK but having column names. The PED file is a white-space (space or tab) delimited file: the first six columns are mandatory: 
FID: Family ID;
IID: Individual ID;
FA: Paternal ID;
MO: Maternal ID;
SEX: Sex (1=male; 2=female; other=unknown);
PHENO: Phenotype; 
Genotypes (column 7 onwards) should also be white-space delimited; they are coded as 0, 1 and 2, indicating the number of coding allele, and NA is for missing genotype. </p>
</td></tr> 
<tr><td><code id="rvPDT.test_+3A_aff">aff</code></td>
<td>
<p>indicates the values that represents affected status in ped data; default is 2.</p>
</td></tr>
<tr><td><code id="rvPDT.test_+3A_unaff">unaff</code></td>
<td>
<p>indicates the values that represents unaffected status in ped data; default is 1.</p>
</td></tr> 
<tr><td><code id="rvPDT.test_+3A_snpcol">snpCol</code></td>
<td>
<p>indicates the columns of variants in ped data.</p>
</td></tr>
<tr><td><code id="rvPDT.test_+3A_hfreq">hfreq</code></td>
<td>
<p>indicates the frequencies of variants that used in calculating weights; when it is NULL, the frequencies are estimated by ped data.</p>
</td></tr> 
<tr><td><code id="rvPDT.test_+3A_training">training</code></td>
<td>
<p>indicates the proportion of training data; default is 0.3.</p>
</td></tr>  
<tr><td><code id="rvPDT.test_+3A_mu">mu</code></td>
<td>
<p>indicates mu value that defines causal in the training data; default is 1.04.</p>
</td></tr>    
<tr><td><code id="rvPDT.test_+3A_usefamweight">useFamWeight</code></td>
<td>
<p>indicates whether the family weights need to be used in the test.</p>
</td></tr>     
<tr><td><code id="rvPDT.test_+3A_trace">trace</code></td>
<td>
<p>indicates whether or not the intermediate outcomes should be printed; default is FALSE.</p>
</td></tr>  
</table>


<h3>Value</h3>

<table>
<tr><td><code>TDT</code></td>
<td>
<p> Transmission/disequilibrium matrix for each pedigrees.</p>
</td></tr>   
<tr><td><code>Sib</code></td>
<td>
<p> Discordant sib pairs matrix for each pedigrees.</p>
</td></tr>   
<tr><td><code>PDT</code></td>
<td>
<p> Pedigree disequilibrium matrix for each pedigrees, which is the sum of TDT and Sib.</p>
</td></tr>    
<tr><td><code>W</code></td>
<td>
<p> Weights used in Weighted rvPDT test.</p>
</td></tr>              
<tr><td><code>test.v1</code></td>
<td>
<p> Weighted rvPDT test statistic with weights W.</p>
</td></tr>    
<tr><td><code>test.v0</code></td>
<td>
<p> Unweighted rvPDT test statistic with weights=1.</p>
</td></tr>    
<tr><td><code>pvalue.v1</code></td>
<td>
<p> P value of weighted rvPDT test statistic (test.v1).</p>
</td></tr>    
<tr><td><code>pvalue.v0</code></td>
<td>
<p> P value of unweighted rvPDT test statistic (test.v0).</p>
</td></tr>  
</table>


<h3>References</h3>

    
<p>Guo W , Shugart YY, Does Haplotype-based Collapsing Tests Gain More Power than Variant-based Collapsing Tests for Detecting Rare Variants in Pedigrees  
(manuscript). 
</p>

<hr>
<h2 id='rvPDT.test.permu'>Variants-based pedigree disequilibrium test for rare variants in pedigrees.</h2><span id='topic+rvPDT.test.permu'></span>

<h3>Description</h3>

<p>To detecting rare variants for binary traits using general pedigrees, the pedigree disequilibrium tests are extended by collapsing rare variants with/without weights.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvPDT.test.permu(ped, aff=2,unaff=1, snpCol, hfreq=NULL,
useFamWeight=TRUE, nperm=1000,trace=FALSE)    
</code></pre>


<h3>Arguments</h3>

   
<table>
<tr><td><code id="rvPDT.test.permu_+3A_ped">ped</code></td>
<td>
<p>input data, has same format with PLINK but having column names. The PED file is a white-space (space or tab) delimited file: the first six columns are mandatory: 
FID: Family ID;
IID: Individual ID;
FA: Paternal ID;
MO: Maternal ID;
SEX: Sex (1=male; 2=female; other=unknown);
PHENO: Phenotype; 
Genotypes (column 7 onwards) should also be white-space delimited; they are coded as 0, 1 and 2, indicating the number of coding allele, and NA is for missing genotype. </p>
</td></tr> 
<tr><td><code id="rvPDT.test.permu_+3A_aff">aff</code></td>
<td>
<p>indicates the values that represents affected status in ped data; default is 2.</p>
</td></tr>
<tr><td><code id="rvPDT.test.permu_+3A_unaff">unaff</code></td>
<td>
<p>indicates the values that represents unaffected status in ped data; default is 1.</p>
</td></tr> 
<tr><td><code id="rvPDT.test.permu_+3A_snpcol">snpCol</code></td>
<td>
<p>indicates the columns of variants in ped data.</p>
</td></tr>
<tr><td><code id="rvPDT.test.permu_+3A_hfreq">hfreq</code></td>
<td>
<p>indicates the frequencies of variants that used in calculating weights; when it is NULL, the frequencies are estimated by ped data.</p>
</td></tr> 
<tr><td><code id="rvPDT.test.permu_+3A_usefamweight">useFamWeight</code></td>
<td>
<p>indicates whether the family weights need to be used in the test.</p>
</td></tr>    
<tr><td><code id="rvPDT.test.permu_+3A_nperm">nperm</code></td>
<td>
<p>indicates the times of permutation; default is 1000.</p>
</td></tr>    
<tr><td><code id="rvPDT.test.permu_+3A_trace">trace</code></td>
<td>
<p>indicates wether or not the intermediate outcomes should be printed; default is FALSE.</p>
</td></tr>  
</table>


<h3>Value</h3>

<table>
<tr><td><code>TDT</code></td>
<td>
<p> Transmission/disequilibrium matrix for each pedigrees.</p>
</td></tr>   
<tr><td><code>Sib</code></td>
<td>
<p> Discordant sib pairs matrix for each pedigrees.</p>
</td></tr>   
<tr><td><code>PDT</code></td>
<td>
<p> Pedigree disequilibrium matrix for each pedigrees, which is the sum of TDT and Sib.</p>
</td></tr>    
<tr><td><code>W</code></td>
<td>
<p> Weights used in Weighted rvPDT test.</p>
</td></tr>              
<tr><td><code>test.v1</code></td>
<td>
<p> Weighted rvPDT test statistic with weights W.</p>
</td></tr>    
<tr><td><code>test.v0</code></td>
<td>
<p> Unweighted rvPDT test statistic with weights=1.</p>
</td></tr>    
<tr><td><code>pvalue.v1</code></td>
<td>
<p> P value of weighted rvPDT test statistic (test.v1).</p>
</td></tr>    
<tr><td><code>pvalue.v0</code></td>
<td>
<p> P value of unweighted rvPDT test statistic (test.v0).</p>
</td></tr>  
</table>


<h3>References</h3>

    
<p>Guo W , Shugart YY, Does Haplotype-based Collapsing Tests Gain More Power than Variant-based Collapsing Tests for Detecting Rare Variants in Pedigrees  
(manuscript). 
</p>

<hr>
<h2 id='rvPDT.test.sub'>Internal function.</h2><span id='topic+rvPDT.test.sub'></span>

<h3>Description</h3>

<p>Internal function of testing rare variants for binary traits using general pedigrees.
</p>

<hr>
<h2 id='whap.prehap'>Prepare haplotype pairs for hPDT tests in pedigree data.</h2><span id='topic+whap.prehap'></span>

<h3>Description</h3>

<p>Before running hPDT test, haplotype pairs are inferred by calling MERLIN in linux for all pedigree members, and then perpare some internal statistics. 
Require the R package of &quot;gregmisc&quot; and MERLIN software.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> 
whap.prehap(ped,map, merlinDir="", outFN.prefix="merlin",aff=2,trace=FALSE)  
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr><td><code id="whap.prehap_+3A_ped">ped</code></td>
<td>
<p>input data, has same format with PLINK but having column names. The PED file is a white-space (space or tab) delimited file, and the first six columns are mandatory: 
FID: Family ID;
IID: Individual ID;
FA: Paternal ID;
MO: Maternal ID;
SEX: Sex (1=male; 2=female; other=unknown);
PHENO: Phenotype; 
Genotypes (column 7 onwards) should also be white-space delimited; they are coded as 0, 1 and 2, indicating the number of coding allele, and NA is for missing genotype. </p>
</td></tr> 
<tr><td><code id="whap.prehap_+3A_map">map</code></td>
<td>
<p>input data, has same format with MAP file required by MERLIN. The MAP file is a white-space (space or tab) delimited file with 3 columns as follows,  
CHROMOSOME: chromosome (1-22, X, Y or 0 if unplaced)  
MARKER: marker name in PED file that is usually rs# or snp identifier         
POSITION: Genetic distance (morgans)
The data file and map file can include different sets of markers, but markers that are absent from the map file will be ignored by MERLIN. 
</p>
</td></tr>
<tr><td><code id="whap.prehap_+3A_merlindir">merlinDir</code></td>
<td>
<p>indicates the directory of Merlin, for example, merlinDir=&quot;./Merlin/&quot;; use the default=&quot;&quot; when Merlin is in current directory or your bin directory.</p>
</td></tr>  
<tr><td><code id="whap.prehap_+3A_outfn.prefix">outFN.prefix</code></td>
<td>
<p>Requests that output file of MERLIN names should be derived from outFN.prefix. For example, when it is set to be &quot;merlin&quot; as default, estimated haplotypes should be stored in a file called merlin.chr.</p>
</td></tr>  
<tr><td><code id="whap.prehap_+3A_aff">aff</code></td>
<td>
<p>indicates the values that represents affected status in ped data; default is 2.</p>
</td></tr> 
<tr><td><code id="whap.prehap_+3A_trace">trace</code></td>
<td>
<p>indicates whether or not the intermediate outcomes should be printed; default is FALSE.</p>
</td></tr>  
</table>


<h3>Value</h3>

<table>
<tr><td><code>SNPname</code></td>
<td>
<p> SNP names of testing.</p>
</td></tr>   
<tr><td><code>hapData</code></td>
<td>
<p> Haplotype data for each individuals.</p>
</td></tr>   
<tr><td><code>freq</code></td>
<td>
<p> Estimated frequencies of haplotypes.</p>
</td></tr>    
<tr><td><code>trans</code></td>
<td>
<p> Transmission matrix of haplotypes.</p>
</td></tr>              
<tr><td><code>hapScore</code></td>
<td>
<p> Score matrix of haplotypes.</p>
</td></tr>    
</table>


<h3>References</h3>

    
<p>Guo W , Shugart YY, Does Haplotype-based Collapsing Tests Gain More Power than Variant-based Collapsing Tests for Detecting Rare Variants in Pedigrees  
(manuscript). 
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
