<!DOCTYPE html><html lang="en-GB"><head><title>Help for package RWsearch</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RWsearch}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RWsearch-package'><p>Package RWsearch</p></a></li>
<li><a href='#archivedb'><p>CRAN archive (CRAN-archive.html + archivedb)</p></a></li>
<li><a href='#binarydb'><p>CRAN Matrix Of Available Binary Packages (archivedb.rda)</p></a></li>
<li><a href='#checkdb'><p>CRAN checks file (check_results.rds)</p></a></li>
<li><a href='#cnsc'><p>Combine Standard and Non-Standard Content</p></a></li>
<li><a href='#crandb'><p>CRAN Packages (crandb.rda)</p></a></li>
<li><a href='#cranmirrors_down'><p>CRAN archive (CRAN-archive.html + archivedb)</p></a></li>
<li><a href='#e_check'><p>Check Results of Packages Identified by their Email Address</p></a></li>
<li><a href='#f_args'><p>Names and Arguments of Functions</p></a></li>
<li><a href='#f_pdf'><p>PDF Pages of Functions</p></a></li>
<li><a href='#funmaintext'><p>Modify the Main Text and the Markdown Header in p_text Function</p></a></li>
<li><a href='#h_direct'><p>Open a Web Page in the your browser</p></a></li>
<li><a href='#h_engine'><p>Explore the Web with Various Search Engines</p></a></li>
<li><a href='#h_R'><p>Open a Web Page in the Browser</p></a></li>
<li><a href='#h_ttp'><p>Open a Web Page in the Browser</p></a></li>
<li><a href='#p_archive'><p>Read Packages in CRAN archive</p></a></li>
<li><a href='#p_check'><p>Return CRAN Package Check Results</p></a></li>
<li><a href='#p_deadline'><p>Packages Flagged by CRAN With a Deadline</p></a></li>
<li><a href='#p_deps'><p>Dependencies and Reverse Dependencies of Packages</p></a></li>
<li><a href='#p_display'><p>Display Package Information in HTML Pages</p></a></li>
<li><a href='#p_down'><p>Download the Package Documentation in One Directory or in Several Subdirectories</p></a></li>
<li><a href='#p_graph'><p>Network and Graphs of Package Dependencies</p></a></li>
<li><a href='#p_html'><p>HTML Help Page, PDF Manual and Vignettes</p></a></li>
<li><a href='#p_inst'><p>Immediate or delayed package installation</p></a></li>
<li><a href='#p_inun'><p>List of Installed, Uninstalled and Non-Existing Packages</p></a></li>
<li><a href='#p_table2pdf'><p>Package Information in Console and PDF Files</p></a></li>
<li><a href='#p_text2pdf'><p>Download Package Documentation in Text Files</p></a></li>
<li><a href='#p_unload_all'><p>Unload all non-base and non-recommended packages from the namespace</p></a></li>
<li><a href='#p_vers'><p>Package Version and Number of Dependencies</p></a></li>
<li><a href='#s_crandb'><p>Search Packages by Keywords in data.frame crandb</p></a></li>
<li><a href='#s_crandb_tvdb'><p>Search For Recent Packages In crandb And In Task View</p></a></li>
<li><a href='#s_hs'><p>Search Packages and Functions in Installed Packages</p></a></li>
<li><a href='#s_sos'><p>Search Packages and Functions on R-Project Help pages and RDocumentation</p></a></li>
<li><a href='#s_tvdb'><p>Search Packages in Task Views</p></a></li>
<li><a href='#tvdb'><p>Task Views (tvdb.rda)</p></a></li>
<li><a href='#zcrandb'><p>File zcrandb.rda: A Subset of crandb Dataset</p></a></li>
<li><a href='#ztvdb'><p>File ztvdb.rda: A Subset of tvdb Dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Lazy Search in R Packages, Task Views, CRAN, the Web. All-in-One
Download</td>
</tr>
<tr>
<td>Description:</td>
<td>Search by keywords in R packages, task views, CRAN, the web and display the results in the console or in txt, html or pdf files. Download the package documentation (html index, README, NEWS, pdf manual, vignettes, source code, binaries) with a single instruction. Visualize the package dependencies and CRAN checks. Compare the package versions, unload and install the packages and their dependencies in a safe order. Explore CRAN archives. Use the above functions for task view maintenance. Access web search engines from the console thanks to 80+ bookmarks. All functions accept standard and non-standard evaluation.</td>
</tr>
<tr>
<td>Version:</td>
<td>5.2.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-03-13</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>brew, latexpdf, networkD3, sig, sos, XML</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ctv, findR, foghorn, pacman, litedown, xfun</td>
</tr>
<tr>
<td>Author:</td>
<td>Patrice Kiener <a href="https://orcid.org/0000-0002-0505-9920"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Patrice Kiener &lt;rpackages@inmodelia.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>litedown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-13 15:21:50 UTC; Patrice</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-13 15:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='RWsearch-package'>Package RWsearch</h2><span id='topic+RWsearch-package'></span><span id='topic+RWsearch'></span>

<h3>Description</h3>

<p>Search by keywords in R packages, task views, CRAN, the web and display the
results in the console or in txt, html or pdf files. Download the package
documentation (html index, README, NEWS, pdf manual, vignettes, source code,
binaries) with a single instruction. Visualize the package dependencies and
CRAN checks. Compare the package versions, unload and install the packages
and their dependencies in a safe order. Explore CRAN archives. Use the above
functions for task view maintenance. Access web search engines from the
console thanks to 80+ bookmarks. All functions accept standard and
non-standard evaluation. Inspired by the packages ctv, foghorn, latexpdf,
pacman and sos.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Patrice Kiener <a href="mailto:rpackages@inmodelia.com">rpackages@inmodelia.com</a> (<a href="https://orcid.org/0000-0002-0505-9920">ORCID</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### THE W IN RWsearch: LAUNCH WEBSITES AND SEARCH ENGINES
if (interactive()) {
h_cranbydate(repos = "https://cloud.r-project.org")
h_yt("Serge Gainsbourg Ne dis rien")
h_osm("La Ferriere sous Jougne")
h_mw(recension)
h_lexilogos()
}

### A CONVENIENT FUNCTION FOR NON-STANDARD EVALUATION
## Non-standard content (nsc1, nsc2), standard content ("stc3", "double word4")
## and regular object (obj) stored in .GlobalEnv can be merged with cnsc()
obj &lt;- c("obj5", "obj6")
cnsc(nsc1, nsc2, "stc3", "double word4", obj)

### DOWNLOAD CRANDB, BINARYDB AND CHECKDB
## In real life, download crandb, binarydb and checkdb from CRAN or load them
## with functions crandb_down(), binarydb_down(), checkdb_down() or 
## crandb_load(), binarydb_load() and checkdb_load().
## checkdb can be ignored if less than npkgs are explored.
## In this example, we use three small files.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
binarydb_load(system.file("data", "zbinarydb.rda", package = "RWsearch"))
checkdb_load(system.file("aabb", "zcheck_results.rds", package = "RWsearch"))

### PRINT THE LATEST PACKAGES UPLOADED ON CRAN (LAST DATE = "2024-09-20")
crandb_fromto(from = -30, to = max(crandb$Published))
crandb_fromto(from = "2024-01-01", to = Sys.Date())

### SEARCH IN CRANDB
## Search in crandb. Use standard or non-standard content.
## Display the results in a vector or in a list.
s_crandb(search, find, cran, web)
s_crandb(search, find, cran, web, select = "PD", mode = "and")
s_crandb("^f", select = "P")
s_crandb(c("thermodynamic", "chemical reaction"))
(lst &lt;- s_crandb_list(thermodynamic, "chemical reaction"))
p_vers(lst)

### DISPLAY THE RESULTS
## in the console, in (txt, md, pdf) files or in the browser.
if (interactive()) {
p_table2(lst)
p_table7pdf(lst, dir = file.path(tempdir(), "ptable"), openpdf = TRUE)
p_text(lst, dir = file.path(tempdir(), "ptext1"), pager = TRUE,
           repos = "https://cloud.r-project.org")
p_text2pdf(lst, dir = file.path(tempdir(), "ptext1"), openpdf = TRUE, 
           repos = "https://cloud.r-project.org")
p_display(lst, dir = file.path(tempdir(), "pdisp1"))
}

### VISUALIZE THE DOCUMENTATION (IN A BROWSER)
## from the installed packages on your computer
## or from the packages listed in https://search.r-project.org 
if (interactive()) {
p_html(brew, sig)
p_man(foghorn)
p_pdfweb(sos, repos = "https://cloud.r-project.org")
}

### VISUALIZE THE PACKAGE DEPENDENCIES AND THE PACKAGE STATUS
if (interactive()) {
p_graphF(actuar, fitdistrplus, reverse = TRUE) # Children
p_graphF(RWsearch)    # Parents
p_vers(RWsearch)      # Installed versions
p_vers_deps(RWsearch) # Installed versions
}

### VISUALIZE THE PACKAGE CHECKS (USE checkdb FOR FASTER RESULTS)
if (interactive()) {
p_check(RWsearch, repos = "https://cloud.r-project.org")
res &lt;- p_checkdeps_lst(RWsearch, repos = "https://cloud.r-project.org")
head(res, 3)
}

### DOWNLOAD THE DOCUMENTATION
## Vector =&gt; download in the "docpkgs" directory ("." is the current directory)
## List   =&gt; download in subdirectories named after the keywords
## (non-standard content is accepted)

p_down(pacman, pdfsearch, sos, dir = file.path(tempdir(), "pdown"),
       repos = "https://cloud.r-project.org")
p_down(lst, dir = file.path(tempdir(), "pdown"), repos = "https://cloud.r-project.org")


### SEARCH WITH sos (ON R-PROJECT HELP PAGES), rdrr AND rdoc
if (interactive()) {
(res &lt;- s_sos(aluminium))
head(data.frame(res), 3)
tail(data.frame(res), 3)
h_rdrr(aluminium)
h_rdoc(distillation)
}

### TASK VIEW MAINTENANCE
## In real life, download crandb and tvdb from CRAN or load them from your directory
## with functions crandb_down(), tvdb_down(), crandb_load(), tvdb_load().
## In this example, we use small files.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
tvdb_load(system.file("data", "ztvdb.rda", package = "RWsearch"))

## List the task views
tvdb_vec()
tvdb_pkgs(ChemPhys, Robust)

## Search for some packages in the task views
s_tvdb(actuar, FatTailsR, MASS, zoo, nopackage)

## Search for recent packages in crandb that contain the keyword
## and verify if the packages are already referred in the task view.
## from = "2024-01-01" and "2018-01-01" are selected for this small example.
s_crandb_tvdb("distribution", tv = "Distributions", from = "2024-01-01")
s_crandb_tvdb("distribution", tv = "Distributions", from = "2018-01-01")

### EXPLORE CRAN ARCHIVE AND DOWNLOAD OLD tar.gz FILES
## In real life, download archivedb and crandb from CRAN
## with the functions archivedb_down() and crandb_down().
## In this example, we load two small files (94 and 85 packages).
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
archivedb_load(system.file("aabb", "zCRAN-archive.html", package = "RWsearch"))
archivedb_npkgs()
lapply(archivedb_list(), tail)

## Download the latest tar.gz version from CRAN archive
## (this works for both both existing and removed packages).

p_vers(fitur, zmatrix)   # zmatrix is an archived package
p_downarch(fitur, zmatrix, dir = file.path(tempdir(), "pdownarch"))



</code></pre>

<hr>
<h2 id='archivedb'>CRAN archive (CRAN-archive.html + archivedb)</h2><span id='topic+archivedb'></span><span id='topic+archivedb_down'></span><span id='topic+archivedb_load'></span><span id='topic+archivedb_npkgs'></span><span id='topic+archivedb_pkgs'></span><span id='topic+archivedb_rempkgs'></span><span id='topic+archivedb_list'></span>

<h3>Description</h3>

<p>The following functions deal with the packages archived in CRAN. The html file
downloaded from CRAN contains the regular packages that have been updated once and
the packages that have been removed from CRAN by CRAN administrators. It does not contain
the first version of the packages uploaded to CRAN and never updated. These files and
the files removed from CRAN index can be guessed through a comparison with <code>crandb</code>.
</p>
<p><code>archivedb_down</code> downloads from CRAN the html file of the archived packages, saves
it on the disk under the name <code>filename</code>, extracts from it and loads in .GlobalEnv
a data.frame named <code>archivedb</code>.
</p>
<p><code>archivedb_load</code> reads the html file <code>filename</code> saved on the disk, extracts
from it and loads in .GlobalEnv a data.frame named <code>archivedb</code>.
</p>
<p><code>archivedb_npkgs</code> returns the number of packages listed each category: number of
packages in <code>crandb</code>, in <code>archivedb</code>, at first version, at subsequent version
and removed from <code>crandb</code> (CRAN index).
</p>
<p><code>archivedb_pkgs</code> returns the packages listed in CRAN archive (= <code>archivedb</code>).
</p>
<p><code>archivedb_rempkgs</code> returns the packages removed from CRAN but available in CRAN
archive. The result can be combined with <code><a href="#topic+p_check">p_check</a></code> to display the last
CRAN check performed (if available). See the example.
</p>
<p><code>archivedb_list</code> compares the data.frame <code>archivedb</code> and <code>crandb</code> and
returns a list with the following items:
</p>

<ul>
<li><p>pkgs_crandb: the packages listed in <code>crandb</code>.
</p>
</li>
<li><p>pkgs_archivedb: the packages listed in <code>archivedb</code>.
</p>
</li>
<li><p>pkgs_first: the packages in first version in <code>crandb</code>.
</p>
</li>
<li><p>pkgs_updated: the packages with more than one version in <code>crandb</code>.
</p>
</li>
<li><p>pkgs_removed: the archived packages removed from CRAN regular index,
i.e. not listed in <code>crandb</code>.
</p>
</li>
<li><p>dfr_crandb: data.frame pkgs_crandb + Published date.
</p>
</li>
<li><p>dfr_archivedb: data.frame pkgs_archivedb + Archived date.
</p>
</li>
<li><p>dfr_first: data.frame pkgs_first + Published date.
</p>
</li>
<li><p>dfr_updated: data.frame pkgs_updated + Published date.
</p>
</li>
<li><p>dfr_removed: data.frame pkgs_removed+ Archived date.
</p>
</li>
<li><p>npkgs: the number of packages in each category.
</p>
</li></ul>

<p>Use <code><a href="#topic+p_archive_lst">p_archive_lst</a></code> to list the package versions stored in CRAN archive.
</p>
<p>Use <code><a href="#topic+p_downarch">p_downarch</a></code> to download packages from CRAN archive, either the
latest version or a specific version number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>archivedb_down(filename = "CRAN-archive.html", dir = ".",
  url = "https://cran.r-project.org/src/contrib/Archive")

archivedb_load(filename = "CRAN-archive.html")

archivedb_npkgs(archivedb = get("archivedb", envir = .GlobalEnv),
  crandb = get("crandb", envir = .GlobalEnv))

archivedb_pkgs(archivedb = get("archivedb", envir = .GlobalEnv))

archivedb_rempkgs(archivedb = get("archivedb", envir = .GlobalEnv),
  crandb = get("crandb", envir = .GlobalEnv))

archivedb_list(archivedb = get("archivedb", envir = .GlobalEnv),
  crandb = get("crandb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="archivedb_+3A_filename">filename</code></td>
<td>
<p>character. The path to file &quot;CRAN-archive.html&quot; (or equivalent).</p>
</td></tr>
<tr><td><code id="archivedb_+3A_dir">dir</code></td>
<td>
<p>character. The directory where <code>filename</code> or tar.gz files
are saved. Default value <code>"."</code> is the current directory.</p>
</td></tr>
<tr><td><code id="archivedb_+3A_url">url</code></td>
<td>
<p>character. The url address of CRAN archive html file.</p>
</td></tr>
<tr><td><code id="archivedb_+3A_archivedb">archivedb</code></td>
<td>
<p>data.frame <code>archivedb</code>. The archivedb data.frame format loaded
in memory by <code>archivedb_down</code> or <code>archivedb_load</code>.</p>
</td></tr>
<tr><td><code id="archivedb_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. The data.frame of CRAN packages.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>### DOWNLOAD archivedb AND COMPARE IT WITH crandb.
## In real life, download archivedb and crandb from CRAN
## with the functions archivedb_down() and crandb_down().
## In this example, we load two small files made with 94 and 85 packages.

crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
archivedb_load(system.file("aabb", "zCRAN-archive.html", package = "RWsearch"))
archivedb_npkgs()
archivedb_pkgs()
archivedb_rempkgs()

lst &lt;- archivedb_list()
lapply(lst, head)
lapply(lst, tail)

xlim &lt;- as.Date(range(lst$dfr_archivedb$Archived)) ; xlim
op &lt;- par(mfrow = c(2,1))
hist(as.Date(lst$dfr_first$Published),
     breaks = 12, freq = TRUE, las = 1, xlim = xlim)
hist(as.Date(lst$dfr_archivedb$Archived),
     breaks = 12, freq = TRUE, las = 1, xlim = xlim)
par(op)

</code></pre>

<hr>
<h2 id='binarydb'>CRAN Matrix Of Available Binary Packages (archivedb.rda)</h2><span id='topic+binarydb'></span><span id='topic+binarydb_down'></span><span id='topic+binarydb_load'></span>

<h3>Description</h3>

<p><code>binarydb_down</code> downloads the matrix of the packages available in binary
format available in CRAN (Windows or macOS depending the computer), saves it
it in a file named <code>binarydb</code> and finally loads it in .GlobalEnv under
the name <code>binarydb</code>. It is a wrapper around the function
<code>available.packages(type = "binary")</code>.
</p>
<p><code>binarydb_load</code> loads the file <code>filename</code> in .GlobalEnv under the
name <code>binarydb</code>.
</p>
<p>When loaded in .GlobalEnv, <code>binarydb</code> is recognized by the functions
<code><a href="#topic+p_vers">p_vers</a></code> and <code><a href="#topic+p_vers_deps">p_vers_deps</a></code> to compare the package
version numbers of the installed packages with the most recent versions of
the binary and the source versions available in CRAN.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binarydb_down(dir = ".", repos = getOption("repos")[1])

binarydb_load(filename = "binarydb.rda")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="binarydb_+3A_dir">dir</code></td>
<td>
<p>character. The directory where &quot;binarydb.rda&quot; is saved
Default value <code>"."</code> is the current directory.</p>
</td></tr>
<tr><td><code id="binarydb_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="binarydb_+3A_filename">filename</code></td>
<td>
<p>character. The (path to a) file &quot;binarydb.rda&quot; or an equivalent.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, to get the latest version, use
## binarydb_down()
## Here, we use a small file
binarydb_load(system.file("data", "zbinarydb.rda", package = "RWsearch"))
class(binarydb)
colnames(binarydb)
dim(binarydb)
head(binarydb)

</code></pre>

<hr>
<h2 id='checkdb'>CRAN checks file (check_results.rds)</h2><span id='topic+checkdb'></span><span id='topic+checkdb_down'></span><span id='topic+checkdb_load'></span>

<h3>Description</h3>

<p><code>checkdb_down</code> downloads from CRAN the file <em>check_results.rds</em>,
saves it unchanged in the designated directory and loads it in .GlobalEnv
under the name <code>checkdb</code>.
</p>
<p><code>checkdb_load</code> loads the file <code>check_results.rds</code> in .GlobalEnv
under the name <code>checkdb</code>.
</p>
<p><code>checkdb</code> is a data.frame of dimension 200000 x 10 (approximatively). It is
used by the functions <code><a href="#topic+p_check_lst">p_check_lst</a></code> and <code><a href="#topic+p_checkdeps_lst">p_checkdeps_lst</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkdb_down(dir = ".", repos = getOption("repos")[1])

checkdb_load(filename = "check_results.rds")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkdb_+3A_dir">dir</code></td>
<td>
<p>character. The directory where <code>filename</code> or tar.gz files
are saved. Default value <code>"."</code> is the current directory.</p>
</td></tr>
<tr><td><code id="checkdb_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="checkdb_+3A_filename">filename</code></td>
<td>
<p>character. The path to file &quot;check_results.rds&quot; (or equivalent).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>### zcheck_results.rds is a subset of 94 packages synchronized with zcrandb.

checkdb_load(system.file("aabb", "zcheck_results.rds", package = "RWsearch"))
unique(checkdb$Package)
dim(checkdb)
head(checkdb, 15)

</code></pre>

<hr>
<h2 id='cnsc'>Combine Standard and Non-Standard Content</h2><span id='topic+cnsc'></span><span id='topic+cnscinfun'></span><span id='topic+cnscinfun2'></span>

<h3>Description</h3>

<p><code>cnsc</code> reads standard content (object in <code>.GlobalEnv</code> or quoted
characters) and non-standard content. Standard content, including <em>call</em>s,
is evaluated. Non-standard content and non-existing objects (in <code>.GlobalEnv</code>)
are converted into character chains. Regular lists are kept unchanged.
</p>
<p><code>cnscinfun</code> is used inside <code>cnsc</code>. It should not be called directly.
If you find <code>cnsc</code> and <code>cnscinfun</code> appealing, copy the code of
<code>cnscinfun</code> in your package and use it as an hidden function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cnsc(...)

cnscinfun()

cnscinfun2(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cnsc_+3A_...">...</code></td>
<td>
<p>Character vectors, standard or non-standard, existing or non-existing
R objects, regular call. Examples : <code>"word1"</code>; <code>c("word1 word2")</code>;
<code>c("word1", "word2")</code>;  <code>"word1", "word2", "word3"</code>;
<code>word1, word2, c("word3", "word4")</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>### cnsc
## Non-standard content (nsc1, nsc2), standard content ("stc3", "double word4")
## and regular object (vec) stored in .GlobalEnv are merged.
vec &lt;- c("obj5", "obj6")
cnsc(nsc1, nsc2, "stc3", "double word4", vec)

## Lists, either name in .GlobalEnv or call, are evaluated.
lst &lt;- list(A = c("txt1","txt2","txt3"), B = c("txt4", "txt5"))
cnsc(lst)
cnsc(list(C = c("pkg1","pkg2","pkg3"), D = c("pkg4","pkg5")))

### cnscinfun
fun &lt;- function(...) cnscinfun()
fun(nsc1, nsc2, "stc3", "double word4", vec)
fun(lst)

### cnscinfun used in RWsearch: one line at the begining of each function.
### An easy-to-use Non Standard Evaluation, mainly for characters.
funsort &lt;- function(..., char = NULL) {
  words &lt;- if (is.null(char)) cnscinfun() else char
  sort(words)
  # or more complex code
}
funsort(nsc1, nsc2, "stc3", "double word4", vec)
funsort(char = sample(vec, 5, replace = TRUE))

</code></pre>

<hr>
<h2 id='crandb'>CRAN Packages (crandb.rda)</h2><span id='topic+crandb'></span><span id='topic+crandb_down'></span><span id='topic+crandb_load'></span><span id='topic+crandb_comp'></span><span id='topic+crandb_pkgs'></span><span id='topic+crandb_fromto'></span>

<h3>Description</h3>

<p><code>crandb_down</code> downloads from CRAN the file <em>packages.rds</em>, a file
refreshed everyday that describes the packages available in CRAN for this day,
opens it as a data.frame and cleans this data.frame with the following oprations:
rename (with <code>make.names</code>) the column names that are syntactically invalid,
remove the duplicated lines located at the end of the file, clean some bad
characters in the Description column. The resulting clean data.frame is then
loaded in .GlobalEnv under the name <code>crandb</code> and saved in the current
directory with the filename <code>crandb.rda</code>. If <code>oldfile</code> is defined,
the vector of packages between the two files is compared and a short message
is printed about the differences (removed packages, new packages, updated packages).
</p>
<p><code>crandb_load</code> loads the file <code>filename</code> in .GlobalEnv under the name
<code>crandb</code>. It embeds the function <code>load("crandb.rda")</code> and add a short
message about the data.frame properties.
</p>
<p><code>crandb_pkgs</code> displays all packages listed in <code>crandb</code>. The number of
packages is larger than the number obtained with <code>nrow(available.packages())</code>
since packages for all OSes are counted.
</p>
<p><code>crandb_fromto</code> displays the packages published in CRAN between two dates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crandb_down(dir = ".", oldfile = "crandb.rda", verbose = TRUE,
  repos = getOption("repos")[1])

crandb_load(filename = "crandb.rda")

crandb_comp(filename = "crandb.rda", oldfile = "crandb-old.rda",
  addtxt = "")

crandb_pkgs(bydate = FALSE, rev = FALSE, crandb = get("crandb", envir =
  .GlobalEnv))

crandb_fromto(from = -10, to = Sys.Date(), crandb = get("crandb", envir =
  .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="crandb_+3A_dir">dir</code></td>
<td>
<p>character. The directory where &quot;crandb.rda&quot; is saved and the old
&quot;crandb.rda&quot; is read. Default value <code>"."</code> is the current
directory.</p>
</td></tr>
<tr><td><code id="crandb_+3A_oldfile">oldfile</code></td>
<td>
<p>character or NULL. The (path to an) old file that will be compared
to a freshly downloaded version of &quot;crandb.rda&quot; or to <code>filename</code>.
Set to <code>NULL</code> if no comparison is required.</p>
</td></tr>
<tr><td><code id="crandb_+3A_verbose">verbose</code></td>
<td>
<p>logical. <code>TRUE</code> prints the result. <code>FALSE</code> keeps it invisible.</p>
</td></tr>
<tr><td><code id="crandb_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="crandb_+3A_filename">filename</code></td>
<td>
<p>character. The (path to a) file &quot;crandb.rda&quot; or an equivalent.</p>
</td></tr>
<tr><td><code id="crandb_+3A_addtxt">addtxt</code></td>
<td>
<p>character. Internal use.</p>
</td></tr>
<tr><td><code id="crandb_+3A_bydate">bydate</code></td>
<td>
<p>logical. List the package by date of publication rather than by
alphabetical order.</p>
</td></tr>
<tr><td><code id="crandb_+3A_rev">rev</code></td>
<td>
<p>logical. Print in reverse order.</p>
</td></tr>
<tr><td><code id="crandb_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. The data.frame of CRAN packages.</p>
</td></tr>
<tr><td><code id="crandb_+3A_from">from</code></td>
<td>
<p>Negative integer or character representing a date. The number of
days preceeding <code>to</code> or a date before <code>to</code>.</p>
</td></tr>
<tr><td><code id="crandb_+3A_to">to</code></td>
<td>
<p>date. The upper date in the search.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>### In this example, we use the small file zcrandb.rda made of 94 packages
## extracted from CRAN on 2024-09-27. In real life, use crandb_down().
# crandb_down(dir = tempdir(), repos = "https://cloud.r-project.org")

crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
crandb_pkgs()
dim(crandb)
colnames(crandb)
crandb$Published
crandb_fromto(from = "2024-01-01", to = Sys.Date())
pkgs &lt;- crandb_fromto(from = -60, to = max(crandb$Published)) ; pkgs

## Print the table in the console (better if full width)
p_table2(pkgs)

## Display in the browser
if (interactive()) {
p_display7(pkgs, dir = file.path(tempdir(), "crandbdown"))
}


### In the real life, we use a fresh file downloaded from CRAN (6.4 Mo / 20").
## Here, we retrieve the packages uploaded during the last 2 days.

</code></pre>

<hr>
<h2 id='cranmirrors_down'>CRAN archive (CRAN-archive.html + archivedb)</h2><span id='topic+cranmirrors_down'></span>

<h3>Description</h3>

<p><code>cranmirrors_down</code> downloads the csv file of CRAN mirrors, modifies the
&quot;Maintainer&quot; and &quot;Host&quot; columns, eventually saves the modified data.frame on
the disk, loads this data.frame in .GlobalEnv and prints in the console a
subset of the data.frame made of the selected columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cranmirrors_down(filename = "CRAN-mirrors1.csv", dir = ".",
  columns = c(1, 2, 3, 7), save = FALSE,
  url = "https://cran.r-project.org/CRAN_mirrors.csv")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cranmirrors_down_+3A_filename">filename</code></td>
<td>
<p>character. The path to file &quot;CRAN-mirrors1.csv&quot; (or equivalent).</p>
</td></tr>
<tr><td><code id="cranmirrors_down_+3A_dir">dir</code></td>
<td>
<p>character. The directory where <code>filename</code> is saved.
Default value <code>"."</code> is the current directory.</p>
</td></tr>
<tr><td><code id="cranmirrors_down_+3A_columns">columns</code></td>
<td>
<p>a vector of integers or a vector of names.
The column numbers or the column names. Allowed numbers
are within 1:9. Allowed names are: &quot;Name&quot;, &quot;Country&quot;,
&quot;City&quot;, &quot;URL&quot;, &quot;Host&quot;, &quot;Maintainer&quot;, &quot;OK&quot;, &quot;CountryCode&quot;
and &quot;Comment&quot;.</p>
</td></tr>
<tr><td><code id="cranmirrors_down_+3A_save">save</code></td>
<td>
<p>logical. Save the file. If <code>FALSE</code>, the default, the
file is just loaded in .GlobalEnv and a subset is printed
in the console.</p>
</td></tr>
<tr><td><code id="cranmirrors_down_+3A_url">url</code></td>
<td>
<p>character. The url address of the CRAN csv file.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>cranmirrors_down(dir = file.path(tempdir(), "cranmirrors"), save = TRUE)

</code></pre>

<hr>
<h2 id='e_check'>Check Results of Packages Identified by their Email Address</h2><span id='topic+e_check'></span>

<h3>Description</h3>

<p><code>e_check</code> opens the browser and returns the &quot;CRAN Check Results&quot; page(s)
of the packages maintained by one or several maintainers identified by their
regular email addresses (but not the orphaned ones). An internet connection
is required. This function is a simplified version of the functions proposed
in the package <em>foghorn</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>e_check(..., char = NULL, repos = getOption("repos")[1])
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="e_check_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A vector of quoted &quot;e-mail adresses&quot;.</p>
</td></tr>
<tr><td><code id="e_check_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="e_check_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
e_check(c("rpackages@inmodelia.com", "christophe.dutang@ensimag.fr"),
        repos = "https://cloud.r-project.org")
}
</code></pre>

<hr>
<h2 id='f_args'>Names and Arguments of Functions</h2><span id='topic+f_args'></span><span id='topic+f_sig'></span><span id='topic+p_sig'></span>

<h3>Description</h3>

<p><code>f_args</code> is a wrapper of the base function <code><a href="base.html#topic+args">args</a></code>.
</p>
<p><code>f_sig</code> prints the name and arguments of one or several functions in a readable
style. It wraps the function <code>sig::sig</code>.
</p>
<p><code>p_sig</code> prints the name and arguments of the functions exported by one or
several packages. It wraps the function <code>sig::list_sigs</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f_args(..., char = NULL)

f_sig(..., char = NULL)

p_sig(..., char = NULL, pattern = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="f_args_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A vector of unquoted &quot;functions&quot; or characters.</p>
</td></tr>
<tr><td><code id="f_args_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="f_args_+3A_pattern">pattern</code></td>
<td>
<p>a regular expression. See the example.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>f_args(mean, p_display)
f_sig( mean, p_display)

library(brew)
library(sos)
p_sig(brew)
p_sig(RWsearch, sos, pattern = "^f")

</code></pre>

<hr>
<h2 id='f_pdf'>PDF Pages of Functions</h2><span id='topic+f_pdf'></span>

<h3>Description</h3>

<p><code>f_pdf</code> generates in the current directory the pdf pages of one or several
functions. The pdf pages are printed but not opened. Miktex or Texlive is required.
This function wraps <code>{utils::help}</code> with the pdf option activated.
Similar functions are available in the packages <em>document</em> and <em>sinew</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f_pdf(..., char = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="f_pdf_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A vector of quoted &quot;package::function&quot;.</p>
</td></tr>
<tr><td><code id="f_pdf_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## FALSE is here to avoid a NOTE in CRAN checks. Ignore this line.
if (FALSE) {
   f_pdf(c("RWsearch::cnsc", "RWsearch::p_inun"))
}
</code></pre>

<hr>
<h2 id='funmaintext'>Modify the Main Text and the Markdown Header in p_text Function</h2><span id='topic+funmaintext'></span><span id='topic+funmaintex'></span><span id='topic+funheadermd'></span><span id='topic+funheadertex'></span><span id='topic+funfootertex'></span>

<h3>Description</h3>

<p>Use <code>funmaintext</code> or <code>funmaintext2</code> to select the function that displays
the main text. Usage is <code>f\_maintext = funmaintext</code> (without curly braces).
</p>
<p>Use <code>funheadermd</code> to insert markdown header in function <code>sep1 = funheadermd()</code>
(with curly braces).
</p>
<p>See the example in <code>p\_text</code>. To create you own functions, use these functions
as a pattern. The five parameters in <code>f\_maintext, funmaintext</code> are mandatory.
<code>funheadermd</code> can be freely modified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>funmaintext(pkg, sep1, sep2, eol, crandb, repos)

funmaintex(pkg, sep1, sep2, eol, crandb, repos)

funheadermd(title = "TITLE", author = "AUTHOR", date = Sys.Date(),
  keep_tex = "false", toc = "false", number_sections = "true",
  fontsize = "10pt", papersize = "a4paper", margin = "1in")

funheadertex(fontsize = "10pt", papersize = "a4paper", margin = "1in")

funfootertex()
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="funmaintext_+3A_pkg">pkg</code></td>
<td>
<p>character. The package name.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_sep1">sep1</code></td>
<td>
<p>character. The symbols written just before each package name.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_sep2">sep2</code></td>
<td>
<p>character. The symbols written just after each package name.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_eol">eol</code></td>
<td>
<p>character. The end of line for the main text (but not for the
header and the footer). <code>"\n"</code> for text, <code>"  \n"</code>
for rmarkdown, <code>" \\ \n"</code> for latex.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. The data.frame of CRAN packages.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_title">title</code></td>
<td>
<p>character. The title of the .md document (and then in the .pdf file).</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_author">author</code></td>
<td>
<p>character. The author of the .md document.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_date">date</code></td>
<td>
<p>character. The date of the document. Any text format is accepted.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_keep_tex">keep_tex</code></td>
<td>
<p>character. &quot;true&quot; or &quot;false&quot;.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_toc">toc</code></td>
<td>
<p>character. &quot;true&quot; or &quot;false&quot;.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_number_sections">number_sections</code></td>
<td>
<p>character. &quot;true&quot; or &quot;false&quot;.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_fontsize">fontsize</code></td>
<td>
<p>character. Usually &quot;10pt&quot;, &quot;11pt&quot;, &quot;12pt.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_papersize">papersize</code></td>
<td>
<p>character. The usual tex format. Example: &quot;a4paper&quot;.</p>
</td></tr>
<tr><td><code id="funmaintext_+3A_margin">margin</code></td>
<td>
<p>character. In inches, cm or mm. Example: &quot;0.5in&quot;, &quot;1.5cm&quot;, &quot;25mm&quot;.</p>
</td></tr>
</table>

<hr>
<h2 id='h_direct'>Open a Web Page in the your browser</h2><span id='topic+h_direct'></span><span id='topic+h_academie'></span><span id='topic+h_deepl'></span><span id='topic+h_etz'></span><span id='topic+h_framasoft'></span><span id='topic+h_framasoft0'></span><span id='topic+h_googletranslate'></span><span id='topic+h_interglot'></span><span id='topic+h_lexilogos'></span><span id='topic+h_linguee'></span><span id='topic+h_meteoblue'></span><span id='topic+h_promt'></span><span id='topic+h_reverso'></span><span id='topic+h_systran'></span><span id='topic+h_tad'></span><span id='topic+h_tadsm'></span><span id='topic+h_windy'></span><span id='topic+h_yacy'></span>

<h3>Description</h3>

<p><code>h_framasoft</code> gives access to several free web services (as in speech and in
half pint of beer) that are good alternatives to GAFA services.
</p>
<p><code>h_academie</code> is a bookmark to the Academie francaise's dictionary.
</p>
<p><code>h_lexilogos</code> gives access to hundreds of dictionaries in many languages.
</p>
<p><code>h_deepl</code>, <code>h_googletranslate</code>, <code>h_interglot</code>, <code>h_reverso</code>,
<code>h_linguee</code>, <code>h_promt</code>, <code>h_reverso</code>, <code>h_systran</code> provide
translation engines. <code>h_linguee</code> returns examples with long sentences.
</p>
<p><code>h_yacy</code> is a decentralized peer-to-peer web search software.
</p>
<p><code>h_etz</code> is a bookmark to the EveryTimeZone website.
</p>
<p><code>h_tad</code> and <code>h_tadsm</code> are bookmarks to timeanddate.com, a website
dedicated to date and time conversion plus timezone management.
</p>
<p><code>h_meteoblue</code> and <code>h_windy</code> are bookmarks to the Meteoblue and Windy
meteo and weather websites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>h_academie()

h_deepl()

h_etz()

h_framasoft()

h_framasoft0()

h_googletranslate()

h_interglot()

h_lexilogos()

h_linguee()

h_meteoblue()

h_promt()

h_reverso()

h_systran()

h_tad()

h_tadsm()

h_windy()

h_yacy()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
h_linguee()
h_lexilogos()
}
</code></pre>

<hr>
<h2 id='h_engine'>Explore the Web with Various Search Engines</h2><span id='topic+h_engine'></span><span id='topic+h_abcbourse'></span><span id='topic+h_ask'></span><span id='topic+h_arxiv'></span><span id='topic+h_arxivpdf'></span><span id='topic+h_baidu'></span><span id='topic+h_blackle'></span><span id='topic+h_bing'></span><span id='topic+h_biorxiv'></span><span id='topic+h_biorxivpdf'></span><span id='topic+h_bmap'></span><span id='topic+h_boursorama'></span><span id='topic+h_cnrtl'></span><span id='topic+h_collins'></span><span id='topic+h_cpan'></span><span id='topic+h_crossref'></span><span id='topic+h_ctan'></span><span id='topic+h_daum'></span><span id='topic+h_ddg'></span><span id='topic+h_dm'></span><span id='topic+h_doi'></span><span id='topic+h_ecosia'></span><span id='topic+h_egerin'></span><span id='topic+h_estrep'></span><span id='topic+h_evene'></span><span id='topic+h_exalead'></span><span id='topic+h_excite'></span><span id='topic+h_framabee'></span><span id='topic+h_gigablast'></span><span id='topic+h_github'></span><span id='topic+h_gitlab'></span><span id='topic+h_gmap'></span><span id='topic+h_google'></span><span id='topic+h_gscholar'></span><span id='topic+h_ianaTLD'></span><span id='topic+h_ianaWHOIS'></span><span id='topic+h_info'></span><span id='topic+h_ixquick'></span><span id='topic+h_khoj'></span><span id='topic+h_lesechos'></span><span id='topic+h_lilo'></span><span id='topic+h_lt'></span><span id='topic+h_lycos'></span><span id='topic+h_mappy'></span><span id='topic+h_mw'></span><span id='topic+h_nate'></span><span id='topic+h_naver'></span><span id='topic+h_orcid'></span><span id='topic+h_osm'></span><span id='topic+h_osmn'></span><span id='topic+h_parsijoo'></span><span id='topic+h_peertube'></span><span id='topic+h_peru'></span><span id='topic+h_pipilika'></span><span id='topic+h_qwant'></span><span id='topic+h_qwfr'></span><span id='topic+h_reverso_d'></span><span id='topic+h_sapo'></span><span id='topic+h_searx'></span><span id='topic+h_so'></span><span id='topic+h_sogou'></span><span id='topic+h_ssrn'></span><span id='topic+h_ssrnauth'></span><span id='topic+h_startpage'></span><span id='topic+h_twfr'></span><span id='topic+h_twitter'></span><span id='topic+h_un'></span><span id='topic+h_verbes'></span><span id='topic+h_via'></span><span id='topic+h_vimeo'></span><span id='topic+h_wego'></span><span id='topic+h_wp'></span><span id='topic+h_wpfr'></span><span id='topic+h_yahoo'></span><span id='topic+h_yahoofin'></span><span id='topic+h_yandex'></span><span id='topic+h_yooz'></span><span id='topic+h_yt'></span><span id='topic+h_zbib'></span>

<h3>Description</h3>

<p>Launch the default browser and search in: ABC Bourse (short
stock names), arXiv (vectorized), Ask, Baidu, Blackle, Bing, Bing Map (bmap),
Boursorama (short stocknames), CNRTL (French dictionary),
Collins English Dictionary, CPAN and metaCPAN (Perl),
Crossref (DOI and bibliographic metadata), CTAN (Latex), Daum, DailyMotion (dm),
DOI, DuckDuckGo (ddg), Ecosia, Egerin, Evene (citations), Exalead, Excite,
Gigablast, GitHub, GitLab, Google Map (gmap), Google, Google Scholar (gscholar),
IANA TLD root domain database, IANA WHOIS service, Info, Khoj, Les Echos,
La Tribune (lt), Lilo, Lycos, Mappy Map, Merriam-Webster (mw, English dictionary),
Nabble, Nate, Naver (see N2H4 package), Orcid, Open Street Map, OSM Nominatim,
Parsijoo, PeerTube, Peru, Pipilika,
Qwant (qw + qwfr), R-bloggers, Rdocumentation (rdoc), Rdocumentation task views (rdoctv),
Rdrr, Reverso dictionnary, Rseek, Sapo, Searx, Sogou, SSRN and SSRN Author (vectorized),
Stackoverflow (so), Startpage (ex-Ixquick), Twitter (+ twfr),
L'Usine Nouvelle (un), ViaMichelin Map and Routes (via), Les Verbes, Vimeo, Wego (Here maps),
Wikipedia (wp + wpfr), Yahoo, Yahoo Finance, Yandex, Yooz, Youtube (yt).
</p>
<p><code>h_zbib</code> is a bookmark to ZoteroBib, a service that returns the complete
bibliographic reference from a fragment of information: URL, ISBN, DOI, PMID,
arXiv id or title and generates a MD5 number to retrieve it later.t
</p>
<p>Using the regular R format &quot;w1 w2 w3&quot; rather than w1, w2, w3 makes sense as most
functions collapse the words into character chains &quot;w1 w2 w3&quot;, &quot;w1+w2+w3&quot; or &quot;w1-w2-w3&quot;.
</p>
<p>Visit <a href="https://en.wikipedia.org/wiki/Web_search_engine">https://en.wikipedia.org/wiki/Web_search_engine</a> for a list of web search
engines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>h_abcbourse(..., char = NULL)

h_ask(..., char = NULL)

h_arxiv(..., char = NULL)

h_arxivpdf(..., char = NULL)

h_baidu(..., char = NULL)

h_blackle(..., char = NULL)

h_bing(..., char = NULL)

h_biorxiv(..., char = NULL)

h_biorxivpdf(..., char = NULL)

h_bmap(..., char = NULL)

h_boursorama(..., char = NULL)

h_cnrtl(..., char = NULL)

h_collins(..., char = NULL)

h_cpan(..., char = NULL)

h_crossref(..., char = NULL)

h_ctan(..., char = NULL)

h_daum(..., char = NULL)

h_ddg(..., char = NULL)

h_dm(..., char = NULL)

h_doi(..., char = NULL)

h_ecosia(..., char = NULL)

h_egerin(..., char = NULL)

h_estrep(..., char = NULL)

h_evene(..., char = NULL)

h_exalead(..., char = NULL)

h_excite(..., char = NULL)

h_framabee(..., char = NULL)

h_gigablast(..., char = NULL)

h_github(..., char = NULL)

h_gitlab(..., char = NULL)

h_gmap(..., char = NULL)

h_google(..., char = NULL)

h_gscholar(..., char = NULL)

h_ianaTLD(..., char = NULL)

h_ianaWHOIS(..., char = NULL)

h_info(..., char = NULL)

h_ixquick(..., char = NULL)

h_khoj(..., char = NULL)

h_lesechos(..., char = NULL)

h_lilo(..., char = NULL)

h_lt(..., char = NULL)

h_lycos(..., char = NULL)

h_mappy(..., char = NULL)

h_mw(..., char = NULL)

h_nate(..., char = NULL)

h_naver(..., char = NULL)

h_orcid(..., char = NULL)

h_osm(..., char = NULL)

h_osmn(..., char = NULL)

h_parsijoo(..., char = NULL)

h_peertube(..., char = NULL)

h_peru(..., char = NULL)

h_pipilika(..., char = NULL)

h_qwant(..., char = NULL, lang = "en")

h_qwfr(..., char = NULL)

h_reverso_d(..., char = NULL)

h_sapo(..., char = NULL)

h_searx(..., char = NULL)

h_so(..., char = NULL)

h_sogou(..., char = NULL)

h_ssrn(..., char = NULL)

h_ssrnauth(..., char = NULL)

h_startpage(..., char = NULL)

h_twfr(..., char = NULL)

h_twitter(..., char = NULL, lang = "en")

h_un(..., char = NULL)

h_verbes(..., char = NULL)

h_via(..., char = NULL)

h_vimeo(..., char = NULL)

h_wego(..., char = NULL)

h_wp(..., char = NULL, lang = "en")

h_wpfr(..., char = NULL)

h_yahoo(..., char = NULL, lang = "en")

h_yahoofin(..., char = NULL, lang = "en")

h_yandex(..., char = NULL)

h_yooz(..., char = NULL)

h_yt(..., char = NULL)

h_zbib(..., char = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="h_engine_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A vector of packages.</p>
</td></tr>
<tr><td><code id="h_engine_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.</p>
</td></tr>
<tr><td><code id="h_engine_+3A_lang">lang</code></td>
<td>
<p>character. The language accepted by the search engine, usually
&quot;en&quot;, &quot;de&quot;, &quot;es&quot;, &quot;fr&quot;, &quot;jp&quot;, etc.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
h_yt("Serge Gainsbourg Ne dis rien")
h_so(R, deep, neural, network)
h_osm("Le Chateau d'Oleron")
h_mw(recension)
h_arxiv(c(1212.4320, 1605.08732))
h_doi("10.1016/j.ejor.2013.06.029")
}
</code></pre>

<hr>
<h2 id='h_R'>Open a Web Page in the Browser</h2><span id='topic+h_R'></span><span id='topic+h_Rblog'></span><span id='topic+h_Rman'></span><span id='topic+h_Rml'></span><span id='topic+h_Rnews'></span><span id='topic+h_Rpkgs'></span><span id='topic+h_Rversions'></span><span id='topic+h_cran'></span><span id='topic+h_cranbydate'></span><span id='topic+h_cranbyname'></span><span id='topic+h_cranchecks'></span><span id='topic+h_crancheckwindows'></span><span id='topic+h_crantv'></span><span id='topic+h_cranstatus'></span><span id='topic+h_cranberries'></span><span id='topic+h_gepuro'></span><span id='topic+h_nabble'></span><span id='topic+h_rbloggers'></span><span id='topic+h_rdoc'></span><span id='topic+h_rdoctv'></span><span id='topic+h_rdrr'></span><span id='topic+h_rseek'></span><span id='topic+h_biocstats'></span>

<h3>Description</h3>

<p><code>h_R</code> opens the page <a href="https://www.r-project.org">https://www.r-project.org</a>.
</p>
<p><code>h_Rml</code> opens the page dedicated to the mailing lists
<a href="https://www.r-project.org/mail.html">https://www.r-project.org/mail.html</a>.
</p>
<p><code>h_Rnews</code> opens the page <a href="https://cran.r-project.org/doc/manuals/r-devel/NEWS.html">https://cran.r-project.org/doc/manuals/r-devel/NEWS.html</a>.
</p>
<p><code>h_Rpkgs</code> opens the page of the HTML documentation of base and CRAN packages (+ full list) <a href="https://search.r-project.org/R/doc/html/packages.html">https://search.r-project.org/R/doc/html/packages.html</a>.
</p>
<p><code>h_Rversions</code> opens a page (from rversions package) that keeps a record
of all R versions and their release dates.
</p>
<p><code>h_cran</code> opens the page of you local CRAN.
</p>
<p><code>h_cranbydate</code> and <code>h_cranbyname</code> open the page of CRAN packages
sorted by date of publication and in alphabetical order.
</p>
<p><code>h_cranchecks</code> and <code>h_crancheckwindows</code> open the pages related to
the checks of all packages listed by name, maintainers, dates, os. A special
page is dedicated to Windows packages with the results for the previous, the
current and the devel R versions.
</p>
<p><code>h_crantv</code> opens the page of CRAN task views.
</p>
<p><code>h_cranberries</code>, <code>h_nabble</code>, <code>h_rbloggers</code>, <code>h_rdoc</code>, <code>h_rdoctv</code>
(RDocumentation), <code>h_rdrr</code>, <code>h_rseek</code> open the pages of web sites related
to R.
</p>
<p><code>h_gepuro</code> lists all (most) R packages available on GitHub. A huge file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>h_R()

h_Rblog()

h_Rman()

h_Rml()

h_Rnews(repos = getOption("repos")[1])

h_Rpkgs()

h_Rversions(repos = getOption("repos")[1])

h_cran(repos = getOption("repos")[1])

h_cranbydate(repos = getOption("repos")[1])

h_cranbyname(repos = getOption("repos")[1])

h_cranchecks(repos = getOption("repos")[1])

h_crancheckwindows(repos = getOption("repos")[1])

h_crantv(repos = getOption("repos")[1])

h_cranstatus()

h_cranberries()

h_gepuro()

h_nabble(..., char = NULL)

h_rbloggers(..., char = NULL)

h_rdoc(..., char = NULL)

h_rdoctv(..., char = NULL)

h_rdrr(..., char = NULL)

h_rseek(..., char = NULL)

h_biocstats()
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="h_R_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="h_R_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A regular web address.</p>
</td></tr>
<tr><td><code id="h_R_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
h_crantv(repos = "https://cloud.r-project.org")
h_cranberries()
}
</code></pre>

<hr>
<h2 id='h_ttp'>Open a Web Page in the Browser</h2><span id='topic+h_ttp'></span>

<h3>Description</h3>

<p><code>h_ttp</code> opens the page coresponding to the mentionned address in the default browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>h_ttp(..., char = NULL, https = TRUE, www = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="h_ttp_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A regular web address.</p>
</td></tr>
<tr><td><code id="h_ttp_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.</p>
</td></tr>
<tr><td><code id="h_ttp_+3A_https">https</code></td>
<td>
<p>logical. Use https or http.</p>
</td></tr>
<tr><td><code id="h_ttp_+3A_www">www</code></td>
<td>
<p>logical. Add www. to the address.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
h_ttp("www.r-project.org")
}
</code></pre>

<hr>
<h2 id='p_archive'>Read Packages in CRAN archive</h2><span id='topic+p_archive'></span><span id='topic+p_archive_lst'></span><span id='topic+l_targz'></span>

<h3>Description</h3>

<p><code>p_archive</code> opens in the browser one page per package and displays
the package versions stored in CRAN archive.
</p>
<p><code>p_archive_lst</code> prints in the console a list of the package versions
stored in CRAN archive.
</p>
<p><code><a href="#topic+l_targz">l_targz</a></code> takes as input the list issued by <code>p_archive_lst</code>
and lists the last package versions archived before a certain date.
</p>
<p>Use <code><a href="#topic+p_downarch">p_downarch</a></code> to download packages from the CRAN archives,
either the latest version stored or a specific version number.
</p>
<p>Use <code><a href="#topic+archivedb_list">archivedb_list</a></code> to list all packages stored in CRAN archive
(does not include the valid packages having a single version which are stored
in regular CRAN).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_archive(..., char = NULL)

p_archive_lst(..., char = NULL)

l_targz(lst, before = Sys.Date())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_archive_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A vector of packages.</p>
</td></tr>
<tr><td><code id="p_archive_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if <code>...</code>
fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_archive_+3A_lst">lst</code></td>
<td>
<p>list. A list produced by <code>p_archive_lst</code>.</p>
</td></tr>
<tr><td><code id="p_archive_+3A_before">before</code></td>
<td>
<p>character which can be converted to a Date, for instance
&quot;2017-05-14&quot;. Extract from CRAN archive the package(s)
available before this date. Can be synchronized with
the release dates of base-R versions listed at:
<a href="https://CRAN.R-project.org/src/contrib/">https://CRAN.R-project.org/src/contrib/</a> and
<a href="https://CRAN.R-project.org/package=rversions/readme/README.html">https://CRAN.R-project.org/package=rversions/readme/README.html</a></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) p_archive(brew, RWsearch, NotAPkg)

lst &lt;- p_archive_lst(RWsearch, zmatrix, NotAPkg) ; lst
l_targz(lst, before = "2020-01-01")

</code></pre>

<hr>
<h2 id='p_check'>Return CRAN Package Check Results</h2><span id='topic+p_check'></span><span id='topic+p_check_lst'></span><span id='topic+p_checkdeps'></span><span id='topic+p_checkdeps_lst'></span>

<h3>Description</h3>

<p><code>p_check</code> opens the default browser, connects to your local CRAN and displays
for each package the CRAN Package Check Results or the last Check Results recorded in
CRAN archive (with the date of the archive). An internet connexion is required.
</p>
<p><code>p_check_lst</code> reads the check results from the repository and print the
results as a list in the console, with a message for the archived package(s).
An internet connexion is required. If a large number of packages is to be analyzed,
a preload of <code>checkdb</code> is required before launching the instruction (via
<code><a href="#topic+checkdb_down">checkdb_down</a></code> or <code><a href="#topic+checkdb_load">checkdb_load</a></code>). This preload speeds
up significantly the analysis.
</p>
<p><code>p_checkdeps</code> and <code>p_checkdeps_lst</code> extend the analysis to the
package dependencies.
</p>
<p>Comprehensive tables of the check results for package sources and Windows binaries
can be displayed with <code><a href="#topic+h_cranchecks">h_cranchecks</a></code> and <code><a href="#topic+h_crancheckwindows">h_crancheckwindows</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_check(..., char = NULL, repos = getOption("repos")[1])

p_check_lst(..., char = NULL, npkgs = 10, repos = getOption("repos")[1])

p_checkdeps(..., char = NULL, which = "DIL", recursive = TRUE,
  reverse = FALSE, crandb = get("crandb", envir = .GlobalEnv),
  repos = getOption("repos")[1])

p_checkdeps_lst(..., char = NULL, which = "DIL", recursive = TRUE,
  reverse = FALSE, npkgs = 10, crandb = get("crandb", envir =
  .GlobalEnv), repos = getOption("repos")[1])
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_check_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A vector of packages.</p>
</td></tr>
<tr><td><code id="p_check_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_check_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="p_check_+3A_npkgs">npkgs</code></td>
<td>
<p>integer. The number of packages from which a preload of
<code>checkdb</code> is required (via <code><a href="#topic+checkdb_down">checkdb_down</a></code>
or <code><a href="#topic+checkdb_load">checkdb_load</a></code>).</p>
</td></tr>
<tr><td><code id="p_check_+3A_which">which</code></td>
<td>
<p>character vector. A sub-vector of
<code>c("Depends", "Imports", "LinkingTo", "Suggests", "Enhances")</code>.
The short forms &quot;D&quot;, &quot;I&quot;, &quot;L&quot;, &quot;S&quot;, &quot;N&quot;, &quot;DL&quot;, &quot;DI&quot;, &quot;DIL&quot;, &quot;DILS&quot;,
&quot;DILN&quot;, &quot;DILSN&quot;, &quot;SN&quot; are accepted. &quot;N&quot; is for &quot;Enhances&quot; as the
single letter &quot;E&quot; is used by R as a shortcut to EXPR, a reserved word.</p>
</td></tr>
<tr><td><code id="p_check_+3A_recursive">recursive</code></td>
<td>
<p>logical. Search for (reverse) dependencies of (reverse) dependencies.</p>
</td></tr>
<tr><td><code id="p_check_+3A_reverse">reverse</code></td>
<td>
<p>logical. Search for reverse dependencies.</p>
</td></tr>
<tr><td><code id="p_check_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. Also accepted is <code>NULL</code> which will
search in the local <code>installed.packages()</code>. This later form
allows (private) packages that are not listed in <code>crandb</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb and checkdb from CRAN or load them
## with functions crandb_down(), crandb_load(), checkdb_down(), checkdb_load().
## checkdb can be ignored if less than npkgs are explored.
## In these examples, we use two small files of 110 and 107 packages.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
checkdb_load(system.file("aabb", "zcheck_results.rds", package = "RWsearch"))

if (interactive()) {
p_check(RWsearch, igraph, zmatrix, NotApkg, repos = "https://cloud.r-project.org")
p_check_lst(RWsearch, igraph, zmatrix, NotApkg, repos = "https://cloud.r-project.org")
p_checkdeps_lst(igraph, zmatrix, NotApkg, repos = "https://cloud.r-project.org")
}
</code></pre>

<hr>
<h2 id='p_deadline'>Packages Flagged by CRAN With a Deadline</h2><span id='topic+p_deadline'></span>

<h3>Description</h3>

<p><code>p_deadline</code> returns a table with the packages that failed to R CMD check  
and are flagged by CRAN to make some changes before a certain deadline. This
feature has been introduced in the new version R_4.4.1 released mid-June 2024.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_deadline(crandb = get("crandb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_deadline_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb with function crandb_down(), crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
dfr &lt;- p_deadline() ; dfr
if (interactive() &amp; nrow(dfr) &gt; 0) {
p_check(dfr$Package, repos = "https://cloud.r-project.org")
}
</code></pre>

<hr>
<h2 id='p_deps'>Dependencies and Reverse Dependencies of Packages</h2><span id='topic+p_deps'></span><span id='topic+p_depsrec'></span><span id='topic+p_depsrev'></span><span id='topic+p_deps_deps'></span><span id='topic+p_deps_ndeps'></span><span id='topic+p_deps_count'></span><span id='topic+p_deps_inpkgs'></span><span id='topic+p_deps_unpkgs'></span><span id='topic+p_deps_inun'></span>

<h3>Description</h3>

<p><code>p_deps</code> returns the (reverse) dependencies of a (vector of) package(s).
It is a wrapper of the <code>tools::package_dependencies</code> function. A warning
is issued for packages that are not in <code>crandb + .libPaths()</code>
(for instance in CRAN archive, Bioconductor, Github or your own directories).
</p>
<p><code>p_depsrec</code> is a shortcut to <code>p_deps(recursive = TRUE)</code>. It returns
the recursive dependencies (e.g. the list of all ancestors).
</p>
<p><code>p_depsrev</code> is a shortcut to <code>p_deps(reverse = TRUE)</code>. It returns
the reverse dependencies (e.g. the children packages).
</p>
<p><code>p_deps_deps</code> returns a list with the recursive dependencies for the packages
packages and every first level dependencies (including or excluding the ones
in <code>.Library</code>.
</p>
<p><code>p_deps_ndeps</code> returns a vector of the number of dependencies for each
package and their parent dependencies. With the argument <code>sort = TRUE</code>,
the packages are listed from no dependency to the largest number of dependencies.
</p>
<p><code>p_deps_count</code> counts the number of (recursive/reverse) dependencies for
each package and returns a data.frame with 4 columns:
<code>Parents1</code>, <code>ParentsN</code>, <code>Children1</code>, <code>ChildrenN</code>.
</p>
<p><code>p_deps_inpkgs</code> returns the package dependencies that are installed
in the computer.
</p>
<p><code>p_deps_unpkgs</code> returns the package dependencies that are not installed
in the computer.
</p>
<p><code>p_deps_inun</code> combines <code>p_deps</code> and <code>p_inun_crandb</code>, then returns
the status of all dependencies: installed or not installed in the computer,
available or not available in the current crandb (see CRAN archives, Bioconductor,
Github, your own packages).
</p>
<p>The missing packages available on CRAN can be downloaded with <code><a href="#topic+p_down0">p_down0</a></code>,
downloaded and checked (by R CMD check) with <code>xfun::rev_check</code> or installed
with <code>install.packages</code>. The packages removed from CRAN but available in
CRAN archive can be downloaded with <code><a href="#topic+p_downarch">p_downarch</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_deps(..., char = NULL, which = "DIL", recursive = FALSE,
  reverse = FALSE, verbose = TRUE, crandb = get("crandb", envir =
  .GlobalEnv))

p_depsrec(..., char = NULL, which = "DIL", reverse = FALSE,
  verbose = TRUE, crandb = get("crandb", envir = .GlobalEnv))

p_depsrev(..., char = NULL, which = "DIL", recursive = FALSE,
  verbose = TRUE, crandb = get("crandb", envir = .GlobalEnv))

p_deps_deps(..., char = NULL, which = "DIL", Library = FALSE,
  verbose = TRUE, crandb = get("crandb", envir = .GlobalEnv))

p_deps_ndeps(..., char = NULL, which = "DIL", Library = FALSE,
  sort = TRUE, verbose = TRUE, crandb = get("crandb", envir =
  .GlobalEnv))

p_deps_count(..., char = NULL, which = "DIL", verbose = TRUE,
  crandb = get("crandb", envir = .GlobalEnv))

p_deps_inpkgs(..., char = NULL, which = "DIL", recursive = TRUE,
  reverse = FALSE, crandb = get("crandb", envir = .GlobalEnv))

p_deps_unpkgs(..., char = NULL, which = "DIL", recursive = TRUE,
  reverse = FALSE, crandb = get("crandb", envir = .GlobalEnv))

p_deps_inun(..., char = NULL, which = "DIL", recursive = TRUE,
  reverse = FALSE, crandb = get("crandb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_deps_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, excluding list.
A package or a vector of packages listed in <code>crandb</code> or in
<code>installed.packages()</code>.</p>
</td></tr>
<tr><td><code id="p_deps_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_deps_+3A_which">which</code></td>
<td>
<p>character vector. A sub-vector of
<code>c("Depends", "Imports", "LinkingTo", "Suggests", "Enhances")</code>.
The short forms &quot;D&quot;, &quot;I&quot;, &quot;L&quot;, &quot;S&quot;, &quot;N&quot;, &quot;DL&quot;, &quot;DI&quot;, &quot;DIL&quot;, &quot;DILS&quot;,
&quot;DILN&quot;, &quot;DILSN&quot;, &quot;SN&quot; are accepted. &quot;N&quot; is for &quot;Enhances&quot; as the
single letter &quot;E&quot; is used by R as a shortcut to EXPR, a reserved word.</p>
</td></tr>
<tr><td><code id="p_deps_+3A_recursive">recursive</code></td>
<td>
<p>logical. Search for (reverse) dependencies of (reverse) dependencies.</p>
</td></tr>
<tr><td><code id="p_deps_+3A_reverse">reverse</code></td>
<td>
<p>logical. Search for reverse dependencies.</p>
</td></tr>
<tr><td><code id="p_deps_+3A_verbose">verbose</code></td>
<td>
<p>logical. Print the message.</p>
</td></tr>
<tr><td><code id="p_deps_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. Also accepted is <code>NULL</code> which will
search in the local <code>installed.packages()</code>. This later form
allows (private) packages that are not listed in <code>crandb</code>.</p>
</td></tr>
<tr><td><code id="p_deps_+3A_library">Library</code></td>
<td>
<p>logical. The default <code>FALSE</code> excludes the base and recommended
packages stored in <code>.Library</code>. <code>TRUE</code> includes them.</p>
</td></tr>
<tr><td><code id="p_deps_+3A_sort">sort</code></td>
<td>
<p>logical. The default <code>TRUE</code> sorts the package by the number
of dependencies.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))

p_deps(networkD3, visNetwork)
p_deps(networkD3, visNetwork, recursive = TRUE)
p_deps(actuar, fitdistrplus, reverse = TRUE, which = "DILSN")

p_deps_inpkgs(RWsearch, canprot)
p_deps_unpkgs(RWsearch, canprot)
p_deps_inun(RWsearch, canprot, NotApkg)

p_deps_deps(actuar, networkD3, FatTailsR, RWsearch, NotApkg)
p_deps_ndeps(actuar, networkD3, FatTailsR, RWsearch, NotApkg)
p_deps_count(actuar, networkD3, FatTailsR, RWsearch, NotApkg)

</code></pre>

<hr>
<h2 id='p_display'>Display Package Information in HTML Pages</h2><span id='topic+p_display'></span><span id='topic+p_display5'></span><span id='topic+p_display7'></span>

<h3>Description</h3>

<p><code>p_display</code>, <code>p_display5</code> and <code>p_display7</code> open the default browser and
display the results of <code>p_table</code>, <code>p_table5</code> and <code>p_table7</code> in one or
several html pages. If <code>...</code> (or <code>char</code>) is a list, several pages are opened.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_display(..., char = NULL, columns = c("Package", "Title", "Description"),
  dir = tempdir(), verbose = FALSE, crandb = get("crandb", envir =
  .GlobalEnv))

p_display5(..., char = NULL, dir = tempdir(), verbose = FALSE,
  crandb = get("crandb", envir = .GlobalEnv))

p_display7(..., char = NULL, dir = tempdir(), verbose = FALSE,
  crandb = get("crandb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_display_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, including list.
A vector or a list of packages. Or a vector or a list of data.frame
produced by <code>p_table</code>.</p>
</td></tr>
<tr><td><code id="p_display_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if <code>...</code> fails
or if you call the function from another function. If used,
argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_display_+3A_columns">columns</code></td>
<td>
<p>character vector. A sub-vector of <code>colnames(crandb)</code>. The
short form &quot;P&quot;, &quot;T&quot;, &quot;D&quot;, &quot;PT&quot;, &quot;PD&quot;, &quot;TD&quot;, &quot;PTD&quot;, &quot;A&quot;, &quot;M&quot;, &quot;AM&quot;
describing the Package name, Title, Description, Author, Maintainer
or a combination of them is accepted.</p>
</td></tr>
<tr><td><code id="p_display_+3A_dir">dir</code></td>
<td>
<p>character. The directory in which the html file(s) is (are) saved.
<code>tempdir()</code> or <code>getwd()</code> are common paths.</p>
</td></tr>
<tr><td><code id="p_display_+3A_verbose">verbose</code></td>
<td>
<p>logical. List the generated html file(s).</p>
</td></tr>
<tr><td><code id="p_display_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. The data.frame of CRAN packages.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
## No package has the 'distillation' keyword. An empty table is returned.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
(lst &lt;- s_crandb_list("thermodynamic", "chemical reaction", "distillation"))

if (interactive()) {
dir &lt;- file.path(tempdir(), "pdisplay")

## Vector =&gt; 1 page
p_display(RWsearch, pacman, pdfsearch, sos, brew, dir = dir)

## List with 3 items =&gt; 3 pages with one empty
p_display5(lst, dir = dir)
}
</code></pre>

<hr>
<h2 id='p_down'>Download the Package Documentation in One Directory or in Several Subdirectories</h2><span id='topic+p_down'></span><span id='topic+p_down0'></span><span id='topic+p_downarch'></span><span id='topic+targz_down'></span>

<h3>Description</h3>

<p>If <code>pkgs</code> is a vector of packages obtained from <code><a href="#topic+s_crandb">s_crandb</a></code>,
<code>p_down</code> downloads from CRAN and saves in the <code>dir</code> directory (by default
the current directory) the index page, the manual, the vignettes, the README, NEWS,
ChangeLog, CRAN checks files, the source code in <em>pkg_ver.tar.gz</em> format,
the binary code in <em>pkg_ver.tgz</em> (Mac OSX) or <em>pkg_ver.zip</em> (Windows)
format and a minimal R-script of each package. The files that do not exist are
ignored, with no warning.
</p>
<p>If <code>pkgs</code> is a list of packages obtained from <code><a href="#topic+s_crandb_list">s_crandb_list</a></code>,
<code>p_down</code> saves the downloaded files in subdirectories named after the names
of the list, e.g. the keywords used at the search step. The names are
eventually modified with <code>gsub(".", "_", make.names(pkg), fixed = TRUE)</code>
to cope with Unix and Windows directory names.
</p>
<p><code>p_down0</code> calls <code>p_down</code> with different values for each argument.
With the default configuration, this function downloads nothing. It is mostly
used to download one specific item which has not been previously downloaded.
</p>
<p><code>p_downarch</code> downloads from CRAN archive the tar.gz file of one or several
packages, either the last version(s) with <code>Sys.Date()</code> or the version(s)
before a given date. It combines 3 functions: <code><a href="#topic+p_archive_lst">p_archive_lst</a></code>
lists the packages stored in CRAN archive and their version numbers,
<code><a href="#topic+l_targz">l_targz</a></code> extracts the appropriate version numbers available
before a given date, <code>targz_down</code> downloads the tar.gz files in the selected
directory (default is the current directory) and eventually decompresses it.
</p>
<p><code>targz_down</code> downloads the tar.gz files from CRAN archive to the selected
directory (default is the current directory) and eventually decompresses it.
If <code>url = "https://cran.r-project.org/src/contrib"</code>, <code>targz_down</code>
will take the latest version of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_down(..., char = NULL, index = TRUE, manual = TRUE, vignettes = TRUE,
  RqmdRmd = FALSE, README = TRUE, NEWS = FALSE, ChangeLog = FALSE,
  checks = FALSE, targz = FALSE, untar = FALSE, binary = FALSE,
  type = "binary", script = FALSE, dir = ".", crandb = get("crandb",
  envir = .GlobalEnv), repos = getOption("repos")[1])

p_down0(..., char = NULL, index = FALSE, manual = FALSE,
  vignettes = FALSE, RqmdRmd = FALSE, README = FALSE, NEWS = FALSE,
  ChangeLog = FALSE, checks = FALSE, targz = FALSE, untar = FALSE,
  binary = FALSE, type = "binary", script = FALSE, dir = ".",
  crandb = get("crandb", envir = .GlobalEnv),
  repos = getOption("repos")[1])

p_downarch(..., char = NULL, before = Sys.Date(), dir = ".",
  untar = FALSE, url = "https://cran.r-project.org/src/contrib/Archive")

targz_down(ptargz, dir = ".", untar = FALSE,
  url = "https://cran.r-project.org/src/contrib/Archive")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_down_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, including list.
A vector or packages or a named list of packages (with names
being the keywords).</p>
</td></tr>
<tr><td><code id="p_down_+3A_char">char</code></td>
<td>
<p>(name to) a character vector or a list. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_down_+3A_index">index</code></td>
<td>
<p>logical. Download the html index page of each package.</p>
</td></tr>
<tr><td><code id="p_down_+3A_manual">manual</code></td>
<td>
<p>logical. Download the pdf manual.</p>
</td></tr>
<tr><td><code id="p_down_+3A_vignettes">vignettes</code></td>
<td>
<p>logical. Download the html and pdf vignettes, if they exist.</p>
</td></tr>
<tr><td><code id="p_down_+3A_rqmdrmd">RqmdRmd</code></td>
<td>
<p>logical. Download the R, Rmd and qmd source files of the vignettes, if they exist.</p>
</td></tr>
<tr><td><code id="p_down_+3A_readme">README</code></td>
<td>
<p>logical. Download the README file, if it exists.</p>
</td></tr>
<tr><td><code id="p_down_+3A_news">NEWS</code></td>
<td>
<p>logical. Download the NEWS file, if it exists.</p>
</td></tr>
<tr><td><code id="p_down_+3A_changelog">ChangeLog</code></td>
<td>
<p>logical. Download the ChangeLog file, if it exists.</p>
</td></tr>
<tr><td><code id="p_down_+3A_checks">checks</code></td>
<td>
<p>logical. Download the CRAN checks file.</p>
</td></tr>
<tr><td><code id="p_down_+3A_targz">targz</code></td>
<td>
<p>logical. Download the *.tar.gz source file.</p>
</td></tr>
<tr><td><code id="p_down_+3A_untar">untar</code></td>
<td>
<p>logical. Decompress the downloaded tar.gz file.</p>
</td></tr>
<tr><td><code id="p_down_+3A_binary">binary</code></td>
<td>
<p>logical. Download the *.tgz (Mac OSX) or *.zip (Windows)
binary file, depending the <code>type</code> value.</p>
</td></tr>
<tr><td><code id="p_down_+3A_type">type</code></td>
<td>
<p>character. Either <code>"mac.binary"</code>, <code>"mac.binary.el-capitan"</code>,
or <code>"win.binary"</code>. The default, <code>"binary"</code>,
automatically detects the local OS and the variants
between R-3.6.3, R-4.0.0 or (for Windows) gcc8. See the
<code>type</code> section of <code>utils::install.packages</code>.</p>
</td></tr>
<tr><td><code id="p_down_+3A_script">script</code></td>
<td>
<p>logical. Create a mini-script and save it in a *.R file.</p>
</td></tr>
<tr><td><code id="p_down_+3A_dir">dir</code></td>
<td>
<p>character. The directory in which the files are saved.
Default value <code>"."</code> is the current directory.</p>
</td></tr>
<tr><td><code id="p_down_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. The data.frame of CRAN packages.</p>
</td></tr>
<tr><td><code id="p_down_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="p_down_+3A_before">before</code></td>
<td>
<p>character which can be converted to a Date, for instance
&quot;2017-05-14&quot;. Extract from CRAN archive the package(s)
available before this date. Can be synchronized with
the release dates of base-R versions listed at:
<a href="https://CRAN.R-project.org/src/contrib/">https://CRAN.R-project.org/src/contrib/</a> and
<a href="https://CRAN.R-project.org/package=rversions/readme/README.html">https://CRAN.R-project.org/package=rversions/readme/README.html</a></p>
</td></tr>
<tr><td><code id="p_down_+3A_url">url</code></td>
<td>
<p>character. The url address of CRAN archive html file.</p>
</td></tr>
<tr><td><code id="p_down_+3A_ptargz">ptargz</code></td>
<td>
<p>character. A vector of package(s) with their version number
and tar.gz extension stored in CRAN archive. These packages
can be identified with <code><a href="#topic+l_targz">l_targz</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))

## Download the documentation in the "dirpkgs" directory. Flat representation.
dir &lt;- file.path(tempdir(), "dirpkgs")
p_down(RWsearch, pdfsearch, sos, dir = dir, repos = "https://cloud.r-project.org")
list.files(dir, recursive = TRUE, full.names = TRUE)

## Download the documentation in subdirectories named after the keywords.
dir   &lt;- file.path(tempdir(), "dirpkgslist")
(lst  &lt;- s_crandb_list(thermodynamic, "chemical reaction"))
(lst2 &lt;- lapply(lst, function(x) x[1:2]))
system.time(
  p_down(lst2, dir = dir, repos = "https://cloud.r-project.org")
)
list.files(dir, recursive = TRUE, full.names = TRUE)

## Download tar.gz files stored in CRAN archive.
dir &lt;- file.path(tempdir(), "targzip")
p_downarch(fitur, zmatrix, NotAPkg, before = "2017-05-14", dir = dir)
targz_down("SVN_1.0.tar.gz", dir = dir, untar = TRUE)
list.files(dir, recursive = TRUE, full.names = TRUE)

</code></pre>

<hr>
<h2 id='p_graph'>Network and Graphs of Package Dependencies</h2><span id='topic+p_graph'></span><span id='topic+p_graphF'></span><span id='topic+p_graphS'></span><span id='topic+p_network'></span><span id='topic+n_graphF'></span><span id='topic+n_graphS'></span>

<h3>Description</h3>

<p><code>p_graphF</code> calculates the (recursive/reverse) dependencies of a (vector
of) package(s) and displays in the default browser a standard graph (F/Force
in the networkD3 terminology) of the package dependencies. It combines the
<code>p_network</code> and <code>n_graphF</code> functions.
</p>
<p><code>p_graphF</code> calculates the (recursive/reverse) dependencies of a (vector
of) package(s) and displays in the default browser a Sankey graph
(in the networkD3 terminology) of the package dependencies. It combines the
<code>p_network</code> and <code>n_graphS</code> functions.
</p>
<p><code>p_network</code> returns the (recursive/reverse) dependencies of a (vector
of) package(s) as a network of nodes and links.
</p>
<p><code>n_graphF</code> taks as input a network of package nodes and links and displays
them in the default browser as a standard graph (F/Force in the networkD3
terminology) representing the package dependencies.
</p>
<p><code>n_graphF</code> taks as input a network of package nodes and links and displays
them in the default browser as a Sankey graph (in the networkD3 terminology)
representing the package dependencies.
</p>
<p>Remember that the option <code>exclpkgs = ...</code> whose default value <code>TRUE</code>
is equivalent to
<code>exclpkgs = c("graphics", "grDevices", "methods", "stats", "tools", "utils")</code>,
can substantially modify the aspect of the graph, especially for <code>reverse = FALSE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_graphF(..., char = NULL, which = "DIL", recursive = TRUE,
  reverse = FALSE, exclpkgs = TRUE, group = 2, fontFamily = "serif",
  fontSize = 11, linkDistance = 50, charge = -100,
  crandb = get("crandb", envir = .GlobalEnv))

p_graphS(..., char = NULL, which = "DIL", recursive = TRUE,
  reverse = FALSE, exclpkgs = TRUE, group = 2, fontFamily = "serif",
  fontSize = 14, nodeWidth = 30, nodePadding = 10,
  crandb = get("crandb", envir = .GlobalEnv))

p_network(..., char = NULL, which = "DIL", recursive = TRUE,
  reverse = FALSE, exclpkgs = TRUE, crandb = get("crandb", envir =
  .GlobalEnv))

n_graphF(netw, group = 2, fontFamily = "serif", fontSize = 11,
  linkDistance = 50, charge = -100)

n_graphS(netw, group = 2, fontFamily = "serif", fontSize = 14,
  nodeWidth = 30, nodePadding = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_graph_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>. Lists are accepted
for <code>p_graphF</code> and <code>p_graphS</code> (and will result in
multiple html pages) but not in <code>p_network</code>.
A vector or a list of package(s) listed in <code>crandb</code> or in
<code>installed.packages()</code>.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_which">which</code></td>
<td>
<p>character vector. A sub-vector of
<code>c("Depends", "Imports", "LinkingTo", "Suggests", "Enhances")</code>.
The short forms &quot;D&quot;, &quot;I&quot;, &quot;L&quot;, &quot;S&quot;, &quot;N&quot;, &quot;DL&quot;, &quot;DI&quot;, &quot;DIL&quot;, &quot;DILS&quot;,
&quot;DILN&quot;, &quot;DILSN&quot;, &quot;SN&quot; are accepted. &quot;N&quot; is for &quot;Enhances&quot; as the
single letter &quot;E&quot; is used by R as a shortcut to EXPR, a reserved word.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_recursive">recursive</code></td>
<td>
<p>logical. Search for (reverse) dependencies of (reverse) dependencies.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_reverse">reverse</code></td>
<td>
<p>logical. Search for reverse dependencies.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_exclpkgs">exclpkgs</code></td>
<td>
<p>logical or character vector. <code>TRUE</code> excludes
from the network of nodes and links the dependencies
<code>c("graphics", "grDevices", "methods", "stats", "tools", "utils")</code>.
<code>FALSE</code> includes them.
You can provide your own vector of packages to exclude them
from the network of nodes and links, for instance:
<code>exclpkgs = c("ggplot2", list.files(.Library))</code>.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_group">group</code></td>
<td>
<p>integer, currently 1, 2 or 3. The suffix of the &quot;NGroup&quot; column
in <em>netw</em>. Define a scheme for colouring the nodes.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_fontfamily">fontFamily</code></td>
<td>
<p>character. Either &quot;serif&quot; or &quot;sans-serif&quot;.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_fontsize">fontSize</code></td>
<td>
<p>integer. The size of the font.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_linkdistance">linkDistance</code></td>
<td>
<p>integer. The minimal distance of a link between two nodes.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_charge">charge</code></td>
<td>
<p>integer. A repulsive value between two nodes.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. Also accepted is <code>NULL</code> which will
search in the local <code>installed.packages()</code>. This later form
allows (private) packages that are not listed in <code>crandb</code>.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_nodewidth">nodeWidth</code></td>
<td>
<p>integer. The width of the rectangular nodes in the Sankey graph.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_nodepadding">nodePadding</code></td>
<td>
<p>integer. The vertical space between two nodes in the same column
of a Sankey graph.</p>
</td></tr>
<tr><td><code id="p_graph_+3A_netw">netw</code></td>
<td>
<p>a list of class &quot;pkgsnetwork&quot; produced by <code><a href="#topic+p_network">p_network</a></code>
that describes the dependencies of one or several packages
with nodes and links (a network).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))

lst  &lt;- list("RWsearch", "visNetwork") ; lst
netw &lt;- p_network(actuar, fitdistrplus, reverse = TRUE) ; netw

if (interactive()) {
p_graphF(lst)
p_graphS(RWsearch, visNetwork)

n_graphF(netw)
n_graphS(netw)
}
</code></pre>

<hr>
<h2 id='p_html'>HTML Help Page, PDF Manual and Vignettes</h2><span id='topic+p_html'></span><span id='topic+p_page'></span><span id='topic+p_html2'></span><span id='topic+p_htmlweb'></span><span id='topic+p_man'></span><span id='topic+p_pdf'></span><span id='topic+p_pdfweb'></span><span id='topic+p_vig'></span><span id='topic+p_vig_all'></span>

<h3>Description</h3>

<p><code>p_page</code> opens the default browser, connects to your local CRAN and displays the
home page of the package(s). An internet connexion is required.
</p>
<p><code>p_html</code> and <code>p_html2</code> open the default browser and display the html help
page of the package, if it is installed. On Windows, <code>p_html</code> returns a local
server address <em>http://127.0.0.1:*.html</em> and subfunctions listed in the page can
be explored whereas <code>p_html2</code> returns a file address <em>file:///C:/*.html</em>
with no links to the subfunctions.
</p>
<p><code>p_man</code> and its alias <code>p_htmlweb</code> open the default browser and display 
the html help pages stored by the R-project at <em>https://search.r-project.org</em>. 
An internet connexion is required. A message is returned if the package does not 
exist in CRAN. In such case, use the function <code><a href="#topic+s_man">s_man</a></code> for a deeper 
exploration.
</p>
<p><code>p_pdf</code> displays in a pdf reader the pdf manual of the package, or generates
it on the fly
in the current directory if the package is installed. Miktex or Texlive is required.
This is a very fast function if the files already exist (and <code>overwrite= FALSE</code>) and
a (relatively) slow function if the files needs to be generated, usually much slower than:
</p>
<p><code>p_pdfweb</code> downloads from you local CRAN the pdf manual of the package, saves it
in the current directory and opens it in the pdf application of your browser.
An internet connexion is required.
</p>
<p><code>p_vig</code> is a wrapper of <code>utils::browseVignettes</code>. It opens the default browser
and displays a list of the vignettes related to a package, if they exist.
</p>
<p><code>p_vig_all</code> wraps <code>utils::browseVignettes(NULL)</code>. It opens the default browser
and displays all vignettes available in the computer. This can be a very large html file.
</p>
<p>Use <code><a href="#topic+p_archive">p_archive</a></code> or <code><a href="#topic+p_archive">p_archive</a></code> to display in the browser
or in the console the package archives. An internet connexion is required.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_page(..., char = NULL, repos = getOption("repos")[1])

p_html(..., char = NULL)

p_html2(..., char = NULL)

p_htmlweb(..., char = NULL)

p_man(..., char = NULL)

p_pdf(..., char = NULL, overwrite = FALSE, dir = ".")

p_pdfweb(..., char = NULL, repos = getOption("repos")[1])

p_vig(..., char = NULL)

p_vig_all()
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_html_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
A vector of packages.</p>
</td></tr>
<tr><td><code id="p_html_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_html_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="p_html_+3A_overwrite">overwrite</code></td>
<td>
<p>logical. Overwrite already existing file (and use LaTeX intensively).</p>
</td></tr>
<tr><td><code id="p_html_+3A_dir">dir</code></td>
<td>
<p>character. The directory in which the files are read or written.
Default value <code>"."</code> is the current directory.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
p_page(RWsearch, sos, repos = "https://cloud.r-project.org")
p_html(RWsearch, sos)
p_man(RWsearch)
p_pdfweb(sos, repos = "https://cloud.r-project.org")

## Try
p_pdf(sos, dir = file.path(tempdir(), "ppdf"))
p_vig(RWsearch)
}
</code></pre>

<hr>
<h2 id='p_inst'>Immediate or delayed package installation</h2><span id='topic+p_inst'></span><span id='topic+p_oldpkgs'></span><span id='topic+p_inst_batsh'></span>

<h3>Description</h3>

<p><code>p_inst</code> is a wrapper around <code><a href="utils.html#topic+install.packages">install.packages</a></code>. It tries
hard to select the most appropriate <code>lib</code> and stops in case of conflict.
Use <code>install.packages</code> if any additional argument is needed.
</p>
<p><code>p_inst_batsh</code> writes one executable file <code>.sh</code> (Linux, macOS) or 
two files <code>.bat, .R</code> (Windows) that allow a delayed installation of old 
packages in a R &ndash;vanilla environment. This is useful if the default configuration
of R defined in <code>R/etc/Rprofile.site</code> launchs additional packages with 
compiled code, since these packages cannot be updated without modifying the 
<code>R/etc/Rprofile.site</code> file. 
</p>
<p><code>p_oldpkgs</code> is a wrapper around <code><a href="utils.html#topic+old.packages">old.packages</a></code>.
</p>
<p>For the three above functions, if <code><a href="#topic+crandb">crandb</a></code> loaded in <code>.GlobalEnv</code>,
then the packages are sorted by their increasing number of dependencies. 
</p>
<p><code>...</code> allows a non-standard evaluation of unquoted packages separated by commas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_inst(..., char = NULL, lib = NULL, repos = getOption("repos")[1],
  contriburl = NULL, dependencies = NA, type = getOption("pkgType"))

p_oldpkgs(type = "both", lib.loc = .libPaths()[1],
  repos = getOption("repos")[1])

p_inst_batsh(..., char = NULL, type = "binary", verbose = TRUE,
  dir = ".", lib = .libPaths()[1], repos = getOption("repos")[1])
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_inst_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, excluding list.
A vector of packages.</p>
</td></tr>
<tr><td><code id="p_inst_+3A_char">char</code></td>
<td>
<p>(name to) a character vector or a list. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_inst_+3A_lib">lib</code></td>
<td>
<p>character. The directory where to install the packages, usually
one of the directories listed by <code><a href="base.html#topic+.libPaths">.libPaths</a></code>. If NULL,
select automatically the most relevant directory.</p>
</td></tr>
<tr><td><code id="p_inst_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="p_inst_+3A_contriburl">contriburl</code></td>
<td>
<p>character. The address of your private repository.</p>
</td></tr>
<tr><td><code id="p_inst_+3A_dependencies">dependencies</code></td>
<td>
<p>logical. <code>FALSE</code> skips the installation of dependencies. 
<code>NA</code> installs <code>c("Depends", "Imports", "LinkingTo")</code>
dependencies.</p>
</td></tr>
<tr><td><code id="p_inst_+3A_type">type</code></td>
<td>
<p>character. Either <code>"source"</code>, <code>"both"</code>, <code>"binary"</code>
(or its variants <code>"mac.binary"</code>, <code>"mac.binary.el-capitan"</code>,
<code>"win.binary"</code>).</p>
</td></tr>
<tr><td><code id="p_inst_+3A_lib.loc">lib.loc</code></td>
<td>
<p>character. The directory where to search for old packages.</p>
</td></tr>
<tr><td><code id="p_inst_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Print information of the download process.</p>
</td></tr>
<tr><td><code id="p_inst_+3A_dir">dir</code></td>
<td>
<p>character. The directory where to write the files <code>.sh, .bat, .R</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>tmpdir &lt;- tempdir()
pkgs &lt;- p_oldpkgs(type = "source", repos = "https://cloud.r-project.org")
pkgs &lt;- c("brew","fs","XML","RWsearch")
p_inst_batsh(pkgs, type = "source", dir = tmpdir, repos = "https://cloud.r-project.org")
if (.Platform$OS.type == "windows") {
    unlink(file.path(tmpdir, "installrpkgs.R"))
    unlink(file.path(tmpdir, "installrpkgs.bat")) 
} else unlink(file.path(tmpdir, "installrpkgs.sh"))

</code></pre>

<hr>
<h2 id='p_inun'>List of Installed, Uninstalled and Non-Existing Packages</h2><span id='topic+p_inun'></span><span id='topic+p_incrandb'></span><span id='topic+p_inun_crandb'></span>

<h3>Description</h3>

<p><code>p_incrandb</code> returns TRUE if all packages are listed in <code>crandb</code> and a vector
of FALSE with the names of the packages not listed in <code>crandb</code>.
</p>
<p><code>p_inun</code> returns a list of packages installed or not installed in the computer.
</p>
<p><code>p_inun_crandb</code> checks if the packages exist or do not exist
in <code>crandb</code> (see CRAN archives, Bioconductor, Github, your own packages).
</p>
<p>The missing packages available on CRAN can be downloaded with <code><a href="#topic+p_down0">p_down0</a></code>,
downloaded and checked (by R CMD check) with <code>xfun::rev_check</code> or installed
with <code>install.packages</code>. The packages removed from CRAN but available in
CRAN archive can be downloaded with <code><a href="#topic+p_downarch">p_downarch</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_incrandb(..., char = NULL, crandb = get("crandb", envir = .GlobalEnv))

p_inun(..., char = NULL)

p_inun_crandb(..., char = NULL, crandb = get("crandb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_inun_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, including list.
A vector or a list of packages.</p>
</td></tr>
<tr><td><code id="p_inun_+3A_char">char</code></td>
<td>
<p>(name to) a character vector or a list. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_inun_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))

## Check if packages are installed or not, and exist or not in crandb
p_incrandb(RWsearch, NotAPkg1, pacman, NotAPkg2, sos)
p_inun(RWsearch, NotAPkg1, pacman, NotAPkg2, sos)
p_inun_crandb(RWsearch, NotAPkg1, pacman, NotAPkg2, sos)

</code></pre>

<hr>
<h2 id='p_table2pdf'>Package Information in Console and PDF Files</h2><span id='topic+p_table2pdf'></span><span id='topic+p_table'></span><span id='topic+p_table2'></span><span id='topic+p_table5'></span><span id='topic+p_table7'></span><span id='topic+table_pdf'></span><span id='topic+p_table3pdf'></span><span id='topic+p_table5pdf'></span><span id='topic+p_table7pdf'></span>

<h3>Description</h3>

<p><code>p_table</code> returns a subset of <code>crandb</code> for the given packages and the selected
columns, by default the Package name, the Title and the Description.
</p>
<p><code>p_table2</code> has a preset value to 2 columns: &quot;Package&quot;, &quot;Title&quot; and prints the
results in the console with a left alignment.
</p>
<p><code>p_table5</code> has a preset value to 5 columns: &quot;Package&quot;, &quot;Title&quot;, &quot;Description&quot;,
&quot;Author&quot;, &quot;Maintainer&quot;.
</p>
<p><code>p_table7</code> has a preset value to 7 columns: &quot;Package&quot;, &quot;Version&quot;, &quot;Published&quot;,
&quot;Title&quot;, &quot;Description&quot;, &quot;Author&quot;, &quot;Maintainer&quot;.
</p>
<p><code>table_pdf</code> prints the results of <code>p_table</code>, <code>p_table5</code> or <code>p_table7</code>
in pdf file(s). Miktex or Texlive is required.
</p>
<p><code>p_table2pdf</code>, <code>p_table3pdf</code>, <code>p_table5pdf</code>, <code>p_table7pdf</code> combine the
above functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_table(..., char = NULL, columns = c("Package", "Title", "Description"),
  crandb = get("crandb", envir = .GlobalEnv))

p_table2(..., char = NULL, crandb = get("crandb", envir = .GlobalEnv))

p_table5(..., char = NULL, crandb = get("crandb", envir = .GlobalEnv))

p_table7(..., char = NULL, crandb = get("crandb", envir = .GlobalEnv))

table_pdf(x, filename = "SelectedPkgs.tex", dir = ".",
  texops = "a4paper,landscape,10pt", pdf = TRUE, cleantex = TRUE,
  openpdf = TRUE, verbose = TRUE)

p_table2pdf(..., char = NULL, filename = "Selectedpkgs", dir = ".",
  texops = "a4paper,landscape,10pt", pdf = TRUE, cleantex = TRUE,
  openpdf = TRUE, verbose = TRUE, crandb = get("crandb", envir =
  .GlobalEnv))

p_table3pdf(..., char = NULL, filename = "Selectedpkgs", dir = ".",
  texops = "a4paper,landscape,10pt", pdf = TRUE, cleantex = TRUE,
  openpdf = TRUE, verbose = TRUE, crandb = get("crandb", envir =
  .GlobalEnv))

p_table5pdf(..., char = NULL, filename = "Selectedpkgs", dir = ".",
  texops = "a4paper,landscape,10pt", pdf = TRUE, cleantex = TRUE,
  openpdf = TRUE, verbose = TRUE, crandb = get("crandb", envir =
  .GlobalEnv))

p_table7pdf(..., char = NULL, filename = "Selectedpkgs", dir = ".",
  texops = "a4paper,landscape,10pt", pdf = TRUE, cleantex = TRUE,
  openpdf = TRUE, verbose = TRUE, crandb = get("crandb", envir =
  .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_table2pdf_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, including list.
A vector or a list of packages.</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if <code>...</code> fails
or if you call the function from another function. If used,
argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_columns">columns</code></td>
<td>
<p>character vector. A sub-vector of <code>colnames(crandb)</code>. The
short form &quot;P&quot;, &quot;T&quot;, &quot;D&quot;, &quot;PT&quot;, &quot;PD&quot;, &quot;TD&quot;, &quot;PTD&quot;, &quot;A&quot;, &quot;M&quot;, &quot;AM&quot;
describing the Package name, Title, Description, Author, Maintainer
or a combination of them is accepted.</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. The data.frame of CRAN packages.</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_x">x</code></td>
<td>
<p>(list of) data.frame produced by <code>p_table</code> (with 3 columns),
<code>p_table5</code> (5 columns) or <code>p_table7</code> (7 columns). If <code>x</code>
is a list, the names of the list will be appended to <code>filename</code>.</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_filename">filename</code></td>
<td>
<p>character. The file name (with or without extension).</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_dir">dir</code></td>
<td>
<p>character. The directory in which the files are read or written.
Default value <code>"."</code> is the current directory.</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_texops">texops</code></td>
<td>
<p>character vector. Options passed to instruction <code>documentclass</code>
in *.tex file.</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_pdf">pdf</code></td>
<td>
<p>logical. <code>FALSE</code> generates the *.tex file.
<code>TRUE</code> generates both the *.tex and *.pdf files.</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_cleantex">cleantex</code></td>
<td>
<p>logical. Remove the <code>.tex</code> file(s) (only if <code>pdf = TRUE</code>).</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_openpdf">openpdf</code></td>
<td>
<p>logical. Open the generated *.pdf file(s) in a pdf viewer (only if
<code>pdf = TRUE</code>).</p>
</td></tr>
<tr><td><code id="p_table2pdf_+3A_verbose">verbose</code></td>
<td>
<p>logical. Print the path(s) to the generated file(s).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))

## Use a large console (useful for p_table2())
p_table2(pacman, pdfsearch, sos)
(lst &lt;- s_crandb_list("thermodynamic", "chemical reaction", "distillation"))
p_table2(lst)

## Print the tables as pdf files and open them in a pdf viewer.
if (interactive()) {
dir &lt;- file.path(tempdir(), "ptablepdf")
p_table5pdf(pacman, pdfsearch, sos, filename = "table5", dir = dir)
p_table7pdf(lst, filename = "table7", dir = dir, cleantex = FALSE)
}

</code></pre>

<hr>
<h2 id='p_text2pdf'>Download Package Documentation in Text Files</h2><span id='topic+p_text2pdf'></span><span id='topic+p_text'></span><span id='topic+p_text2md'></span><span id='topic+p_text2tex'></span>

<h3>Description</h3>

<p><code>p_text</code> extracts from CRAN the most relevant information related to one or
several packages and print them in a text file which can be tailored to various
formats: *.txt, *.md, *.tex for further treatment.
</p>
<p><code>p_text2md</code> has preset values for markdown files.
</p>
<p><code>p_text2tex</code> has preset values for latex files.
</p>
<p><code>p_text2pdf</code> has preset values for pdf files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_text(..., char = NULL, filename = "txtpkgs.txt", dir = ".",
  beforetext = "", f_maintext = funmaintext, sep1 = "== ",
  sep2 = " ==", eol = "\n", README = TRUE, NEWS = TRUE,
  ChangeLog = TRUE, vignettes = TRUE, RqmdRmd = FALSE, DOI = FALSE,
  aftertext = "", editor = FALSE, pager = FALSE, verbose = TRUE,
  crandb = get("crandb", envir = .GlobalEnv),
  repos = getOption("repos")[1])

p_text2md(..., char = NULL, filename = "mdpkgs.md", dir = ".",
  beforetext = funheadermd(), f_maintext = funmaintext, sep1 = "# ",
  sep2 = "  ", eol = "  \n", README = TRUE, NEWS = TRUE,
  ChangeLog = TRUE, vignettes = TRUE, RqmdRmd = FALSE, DOI = FALSE,
  aftertext = "", editor = FALSE, pager = FALSE, verbose = TRUE,
  crandb = get("crandb", envir = .GlobalEnv),
  repos = getOption("repos")[1])

p_text2tex(..., char = NULL, filename = "texpkgs.tex", dir = ".",
  beforetext = funheadertex(), f_maintext = funmaintex,
  sep1 = "\\section{", sep2 = "}", eol = " \\\\\n",
  README = TRUE, NEWS = TRUE, ChangeLog = TRUE, vignettes = TRUE,
  RqmdRmd = FALSE, DOI = FALSE, aftertext = funfootertex(),
  editor = FALSE, pager = FALSE, verbose = TRUE, crandb = get("crandb",
  envir = .GlobalEnv), repos = getOption("repos")[1])

p_text2pdf(..., char = NULL, filename = "pdfpkgs.pdf", dir = ".",
  beforetext = funheadertex(), f_maintext = funmaintex,
  sep1 = "\\section{", sep2 = "}", eol = " \\\\\n",
  README = TRUE, NEWS = TRUE, ChangeLog = TRUE, vignettes = TRUE,
  RqmdRmd = FALSE, DOI = FALSE, aftertext = funfootertex(),
  cleantex = TRUE, openpdf = FALSE, verbose = TRUE,
  crandb = get("crandb", envir = .GlobalEnv),
  repos = getOption("repos")[1])
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_text2pdf_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, including list.
A vector or packages or a named list of packages (with names
being the keywords).</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_char">char</code></td>
<td>
<p>(name to) a character vector or a list. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_filename">filename</code></td>
<td>
<p>character. The file name with extension. If <code>...</code> (or <code>...</code>)
is a list, the names of the list will be appended to <code>filename</code>.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_dir">dir</code></td>
<td>
<p>character. The directory in which the files are read or written.
Default value <code>"."</code> is the current directory.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_beforetext">beforetext</code></td>
<td>
<p>character. The text written at the beginning of the file.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_f_maintext">f_maintext</code></td>
<td>
<p>function name. The function used to extract the main text from
<code>crandb</code> (supplied with no parenthesis).</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_sep1">sep1</code></td>
<td>
<p>character. The symbols written just before each package name.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_sep2">sep2</code></td>
<td>
<p>character. The symbols written just after each package name.
If used with with mardkdown, add two blank characters at the end
to force a new line.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_eol">eol</code></td>
<td>
<p>character. The end of line for the main text (but not for the
header and the footer). &quot;\n&quot; for text, &quot;  \n&quot; for rmarkdown,
&quot; \\ \n&quot; for latex.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_readme">README</code></td>
<td>
<p>logical. Write the line related to the README page, if it exists.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_news">NEWS</code></td>
<td>
<p>logical. Write the line related to the NEWS page, if it exists.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_changelog">ChangeLog</code></td>
<td>
<p>logical. Write the line related to the ChangeLog page, if it exists.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_vignettes">vignettes</code></td>
<td>
<p>logical. Write the lines related to the vignette(s), if they exist.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_rqmdrmd">RqmdRmd</code></td>
<td>
<p>logical. Add to the vignettes the source files in R, Rmd or qmd format, if they exist.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_doi">DOI</code></td>
<td>
<p>logical. Write the line related to the DOI link, if it exists.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_aftertext">aftertext</code></td>
<td>
<p>character. The text written at the end of the file.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_editor">editor</code></td>
<td>
<p>logical. Open the text file with <code>editor</code>.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_pager">pager</code></td>
<td>
<p>logical. Open the text file with <code>pager</code>.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_verbose">verbose</code></td>
<td>
<p>logical. List the generated file(s).</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. The data.frame of CRAN packages.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_cleantex">cleantex</code></td>
<td>
<p>logical. Remove the <code>.tex</code> file(s).</p>
</td></tr>
<tr><td><code id="p_text2pdf_+3A_openpdf">openpdf</code></td>
<td>
<p>logical. Open the pdf files in the default pdf viewer.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))

## Search in crandb
vec &lt;- s_crandb(search, find, select = "PT") ; vec
lst &lt;- s_crandb_list(thermodynamic, "chemical reaction") ; lst
lst2 &lt;- lapply(lst, function(x) x[1:2]) ; lst2
dir &lt;- file.path(tempdir(), "ptext")

## Generate a txt file

p_text(vec[1:5], filename = "SearchFind.txt", dir = dir, 
       RqmdRmd = TRUE, DOI = TRUE, repos = "https://cloud.r-project.org")

## Generate 2 tex + 2 pdf files (10-20 seconds)
## Try the options cleantex = FALSE and openpdf = TRUE on lst
if (interactive()) {
p_text2pdf(lst2, dir = dir, cleantex = TRUE, openpdf = FALSE,
           RqmdRmd = TRUE, DOI = TRUE, repos = "https://cloud.r-project.org")
}

</code></pre>

<hr>
<h2 id='p_unload_all'>Unload all non-base and non-recommended packages from the namespace</h2><span id='topic+p_unload_all'></span>

<h3>Description</h3>

<p><code>p_unload_all</code> unloads in a safe order all packages, except the base and
recommended packages, that are attached or loaded in the namespace, plus their
respective DLLs.
</p>
<p>It must be used before installing new versions of packages that are currently
loaded in the namespace and require some forced unloading, as revealed by the
first column of the data.frame produced by <code><a href="#topic+p_vers_deps">p_vers_deps</a></code>. Warning:
this function also removes RWsearch and its dependencies. It is therefore
recommended to perform the installation with <code><a href="utils.html#topic+install.packages">install.packages</a></code>
rather than <code><a href="#topic+p_inst">p_inst</a></code>.
</p>
<p>This function is safe enough to reinstall locked packages but not safe enough
for a further use as it does not unload the S3 and S4 classes nor the documentation
which is detected as corrupted. It is therefore recommend to stop and restart R
after the installation of the fresh packages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_unload_all(unload = FALSE, crandb = get("crandb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_unload_all_+3A_unload">unload</code></td>
<td>
<p>logical. <code>FALSE</code> prints a list with 3 items: the loaded
packages, the packages in the namespace, the loaded DLLs.
<code>TRUE</code> removes the packages and their DLLs, then prints
the refreshed list.</p>
</td></tr>
<tr><td><code id="p_unload_all_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))

p_unload_all()
## Then run p_unload_all(TRUE), but only for maintenance!

</code></pre>

<hr>
<h2 id='p_vers'>Package Version and Number of Dependencies</h2><span id='topic+p_vers'></span><span id='topic+p_vers_deps'></span>

<h3>Description</h3>

<p>The information displayed by <code>p_vers</code> depends on the availability of
<code>crandb</code> and <code>binarydb</code> in <code>.GlobalEnv</code>.
</p>
<p>If <code>crandb</code> is not loaded in <code>.GlobalEnv</code>, <code>p_vers</code> returns a
data.frame with two columns: first column <code>nsloaded</code> (TRUE/FALSE) detects
(with <code>base::isNamespaceLoaded</code>) if the package namespaces are loaded.
Second column <code>version</code> is the version number of the installed packages.
</p>
<p>If <code>crandb</code> is loaded in <code>.GlobalEnv</code>, three columns are added.
Column <code>crandb</code> displays the version number of the source packages recorded
in the <code>crandb</code> file. Column <code>gcc</code> displays the nedd for a compilation.
Column <code>compare</code> compares this version number with the version installed
on the computer. Possible values are:
-2 for a package not installed on the computer (NA) but available in crandb ;
-1 for an installed package older than the source package available in CRAN ;
0 for an installed package with the same version number than CRAN ;
+1 for a more recent package than the one available in CRAN ;
+2 for a package installed on the computer and not available in CRAN (NA) ;
+3 for a package not installed on the computer (NA) and not available in CRAN (NA).
</p>
<p>If <code>binarydb</code> is loaded in <code>.GlobalEnv</code>, two or three columns are added.
Column <code>binary</code> displays the version number of the binary packages recorded
in the <code>binarydb = available.packages, type = "binary")</code> matrix.
Column <code>difvb</code> compares the installed version on the computer with this
binary version and column <code>difbc</code> compares (if crandb is in <code>.GlobalEnv</code>)
the binary version with the source package available in CRAN (which can differ
for recently updated packages, a matter of 1 to 3 days).
The numbering is identical to the one used for <code>crandb</code>.
</p>
<p>If <code>ndeps = TRUE</code>, two more columns are added with the number
of recursive dependencies per package. Column <code>tdeps</code> includes the base
and recommended packages. Column <code>ndeps</code> does not count them. This option
can take some time. Set it to <code>FALSE</code> if you need speed.
</p>
<p><code>p_vers_deps</code> calculates the same information but includes the recursive
dependencies. Subsetting by <code>"compare &lt; 0"</code> returns a shorter data.frame
with the uninstalled (-2) and the outdated (-1) packages. Packages marked with
<code>nsloaded = TRUE</code> must be detached and unloaded before any reinstallation.
Using this instruction before running <code>install.packages</code> or <code>p_inst</code>
is very useful as it detects packages that are locked and cannot be reinstalled.
The order provided by <code>p_vers_deps</code> is the best one for the reinstallation
of outdated packages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_vers(..., char = NULL, ndeps = TRUE, split = FALSE, reserved = "")

p_vers_deps(..., char = NULL, ndeps = TRUE, subset = "compare &lt; 4",
  crandb = get("crandb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_vers_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>.
A (list of) vector of packages for <code>p_vers</code>.
A vector of packages for <code>p_vers_deps</code></p>
</td></tr>
<tr><td><code id="p_vers_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if <code>...</code>
fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="p_vers_+3A_ndeps">ndeps</code></td>
<td>
<p>logical. Calculate the number of recursive dependencies.
<code>crandb</code> in <code>.GlobalEnv</code> is required for this option.</p>
</td></tr>
<tr><td><code id="p_vers_+3A_split">split</code></td>
<td>
<p>logical. Split the output in a list of data.frame with 
packages allocated in at most 6 sections : 
library, reserved, nsloaded, cran, source, binary.</p>
</td></tr>
<tr><td><code id="p_vers_+3A_reserved">reserved</code></td>
<td>
<p>character. The packages in the reserved section.</p>
</td></tr>
<tr><td><code id="p_vers_+3A_subset">subset</code></td>
<td>
<p>character. Subset the output data.frame on some columns.
The default <code>"compare &lt; 4"</code> does not subset. Usual values
are <code>"compare &lt; 0"</code> or <code>"compare &lt; 0 &amp;&amp; nsloaded == TRUE"</code></p>
</td></tr>
<tr><td><code id="p_vers_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>. The data.frame of CRAN packages.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>pkgs &lt;- cnsc(RWsearch, MASS, Matrix, NotAPkg, R)
p_vers(pkgs)

## Now with crandb and binarydb loaded in .GlobalEnv. In real life, use:
## crandb_down() ; binarydb_down()
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
binarydb_load(system.file("data", "zbinarydb.rda", package = "RWsearch"))

p_vers(pkgs)
p_vers(p_deps(pkgs))
p_vers_deps(pkgs)    # Dependencies can be visualized with p_graphF(pkgs)
p_vers(char=c(p_depsrec(RWsearch)$RWsearch, "RWsearch"), split = TRUE)

</code></pre>

<hr>
<h2 id='s_crandb'>Search Packages by Keywords in data.frame crandb</h2><span id='topic+s_crandb'></span><span id='topic+s_crandb_list'></span><span id='topic+s_crandb_PTD'></span><span id='topic+s_crandb_AM'></span>

<h3>Description</h3>

<p>The most important functions in this package along with <code><a href="#topic+p_down">p_down</a></code>.
</p>
<p>Search packages in data.frame <code>crandb</code> that contain one or several keywords
in the columns &quot;Package&quot;, &quot;Title&quot;, &quot;Description&quot;, &quot;Author&quot; or &quot;Maintainer&quot;.
</p>
<p><code>s_crandb</code> returns a vector of the packages that contain the keywords.
</p>
<p><code>s_crandb_list</code> returns a list where each element of the list is one of the
keywords.
</p>
<p><code>s_crandb_PTD</code> returns a list split by results in columns &quot;Package&quot;, &quot;Title&quot;
and &quot;Description&quot;. Option <code>mode = "and", "relax"</code> is ignored.
</p>
<p><code>s_crandb_AM</code> returns a list split by results in columns &quot;Author&quot; and
&quot;Maintainer&quot;. Option <code>mode = "and", "relax"</code> is ignored.
</p>
<p>Use <code><a href="#topic+p_table2">p_table2</a></code> to print the results of <code>s_crandb</code> and
<code>s_crandb_list</code> in the console.
Use <code><a href="#topic+p_text">p_text</a></code> to send the results in txt, md or pdf files.
Use <code><a href="#topic+p_display">p_display</a></code> to visualize the results in html pages in the browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s_crandb(..., char = NULL, select = "PTD", mode = "or",
  sensitive = FALSE, perl = FALSE, fixed = FALSE, agrep = FALSE,
  max.distance = 0.1, costs = NULL, crandb = get("crandb", envir =
  .GlobalEnv))

s_crandb_list(..., char = NULL, select = "PTD", mode = "or",
  sensitive = FALSE, perl = FALSE, fixed = FALSE, agrep = FALSE,
  max.distance = 0.1, costs = NULL, crandb = get("crandb", envir =
  .GlobalEnv))

s_crandb_PTD(..., char = NULL, mode = "or", sensitive = FALSE,
  perl = FALSE, fixed = FALSE, agrep = FALSE, max.distance = 0.1,
  costs = NULL, crandb = get("crandb", envir = .GlobalEnv))

s_crandb_AM(..., char = NULL, mode = "or", sensitive = FALSE,
  perl = FALSE, fixed = FALSE, agrep = FALSE, max.distance = 0.1,
  costs = NULL, crandb = get("crandb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="s_crandb_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
One or several keywords.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if <code>...</code> fails
or if you call the function from another function. If used,
argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_select">select</code></td>
<td>
<p>character vector. A sub-vector of <code>colnames(crandb)</code>. The
short form &quot;P&quot;, &quot;T&quot;, &quot;D&quot;, &quot;PT&quot;, &quot;PD&quot;, &quot;TD&quot;, &quot;PTD&quot;, &quot;A&quot;, &quot;M&quot;, &quot;AM&quot;
describing the Package name, Title, Description, Author, Maintainer
or a combination of them is accepted.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_mode">mode</code></td>
<td>
<p>character among &quot;or&quot;, &quot;and&quot;, &quot;relax&quot;. The search mode.
&quot;relax&quot; is for 3 words and more. It is an intermediate between
&quot;or&quot; and &quot;and&quot; as it requires just 2 matching words:
<code>("word1" AND "word2")</code> <code>OR</code> <code>("word1" AND "word3")</code>
<code>OR</code> <code>("word1" AND "word3")</code>.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_sensitive">sensitive</code></td>
<td>
<p>logical. <code>TRUE</code> forces the search to be case sensitive.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_perl">perl</code></td>
<td>
<p>logical. Used only if <code>fixed = FALSE</code>. <code>TRUE</code> uses
Perl-compatible regex. <code>FALSE</code> uses default regexps.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_fixed">fixed</code></td>
<td>
<p>logical. <code>TRUE</code> matchs the keywords as is (and <code>sensitive</code>
is forced to <code>TRUE</code>). <code>FALSE</code> allows grep or Perl regexps.
See <code><a href="base.html#topic+grep">grep</a></code>. Not used by <code>agrep</code>.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_agrep">agrep</code></td>
<td>
<p>logical. For approximate matching, use <code><a href="base.html#topic+agrep">agrep</a></code>
function rather than grep.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_max.distance">max.distance</code></td>
<td>
<p>integer or numeric. See <code><a href="base.html#topic+agrep">agrep</a></code>.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_costs">costs</code></td>
<td>
<p>NULL or list. See <code><a href="base.html#topic+agrep">agrep</a></code>.</p>
</td></tr>
<tr><td><code id="s_crandb_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download crandb from CRAN or load it from your directory
## with functions crandb_down() or crandb_load().
## In this example, we use a small file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))

## Search using standard or non-standard content
s_crandb(c("thermodynamic", "chemical reaction", "distillation"))
s_crandb_list(thermodynamic, "chemical reaction", distillation)

## Search using the various options
s_crandb("^f", select = "P")
s_crandb(pH, sensitive = TRUE)
s_crandb_PTD(pH, sensitive = TRUE)
s_crandb_PTD("C++", fixed = TRUE)
s_crandb(search, find, cran, web, select = "PD", mode = "and")
s_crandb(search, find, cran, web, select = "PD", mode = "relax")
s_crandb(search, find, cran, web, select = "PD", mode = "or")

## Search for some authors using the various options
s_crandb_AM(Kiener, Dutang, ORPHANED)

## Non-standard content can be unquoted words or objects in .GlobalEnv
## They are transformed into character or are evaluated
## Here, the searched keywords are "find" and "search".
OTHER &lt;- "search"
(lst &lt;- s_crandb_list(find, OTHER, select = "P", sensitive = TRUE))

## Display in the browser this list of packages
if (interactive()) {
p_display5(lst, dir = file.path(tempdir(), "scrandb"))
}

</code></pre>

<hr>
<h2 id='s_crandb_tvdb'>Search For Recent Packages In crandb And In Task View</h2><span id='topic+s_crandb_tvdb'></span>

<h3>Description</h3>

<p>This is a function for task view maintenance.
</p>
<p>Search packages in a subset of <code>crandb</code> within dates <code>from</code> and <code>to</code>
that contain one or several keywords in the columns &quot;Package&quot;, &quot;Title&quot;, &quot;Description&quot;,
'Author&quot; or &quot;Maintainer&quot;, then verify if these packages are already refereed in one
of the task views stored in <code>tvdb</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s_crandb_tvdb(..., char = NULL, tv = "Distributions", from = -10,
  to = Sys.Date(), select = "PTD", mode = "or", sensitive = FALSE,
  perl = FALSE, fixed = FALSE, agrep = FALSE, max.distance = 0.1,
  costs = NULL, crandb = get("crandb", envir = .GlobalEnv),
  tvdb = get("tvdb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="s_crandb_tvdb_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
One or several keywords.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if <code>...</code> fails
or if you call the function from another function. If used,
argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_tv">tv</code></td>
<td>
<p>character. One task view among those listed in tvdb.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_from">from</code></td>
<td>
<p>character representing a date earlier than date <code>to</code>.
Or a negative integer representing the number of days preceeding
the date <code>to</code>.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_to">to</code></td>
<td>
<p>date. The upper date in the search.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_select">select</code></td>
<td>
<p>character vector. A sub-vector of <code>colnames(crandb)</code>. The
short form &quot;P&quot;, &quot;T&quot;, &quot;D&quot;, &quot;PT&quot;, &quot;PD&quot;, &quot;TD&quot;, &quot;PTD&quot;, &quot;A&quot;, &quot;M&quot;, &quot;AM&quot;
describing the Package name, Title, Description, Author, Maintainer
or a combination of them is accepted.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_mode">mode</code></td>
<td>
<p>character among &quot;or&quot;, &quot;and&quot;, &quot;relax&quot;. The search mode.
&quot;relax&quot; is for 3 words and more. It is an intermediate between
&quot;or&quot; and &quot;and&quot; as it requires just 2 matching words:
<code>("word1" AND "word2")</code> <code>OR</code> <code>("word1" AND "word3")</code>
<code>OR</code> <code>("word1" AND "word3")</code>.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_sensitive">sensitive</code></td>
<td>
<p>logical. <code>TRUE</code> forces the search to be case sensitive.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_perl">perl</code></td>
<td>
<p>logical. Used only if <code>fixed = FALSE</code>. <code>TRUE</code> uses
Perl-compatible regex. <code>FALSE</code> uses default regexps.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_fixed">fixed</code></td>
<td>
<p>logical. <code>TRUE</code> matchs the keywords as is (and <code>sensitive</code>
is forced to <code>TRUE</code>). <code>FALSE</code> allows grep or Perl regexps.
See <code><a href="base.html#topic+grep">grep</a></code>. Not used by <code>agrep</code>.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_agrep">agrep</code></td>
<td>
<p>logical. For approximate matching, use <code><a href="base.html#topic+agrep">agrep</a></code>
function rather than grep.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_max.distance">max.distance</code></td>
<td>
<p>integer or numeric. See <code><a href="base.html#topic+agrep">agrep</a></code>.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_costs">costs</code></td>
<td>
<p>NULL or list. See <code><a href="base.html#topic+agrep">agrep</a></code>.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_crandb">crandb</code></td>
<td>
<p>data.frame <code>crandb</code>.</p>
</td></tr>
<tr><td><code id="s_crandb_tvdb_+3A_tvdb">tvdb</code></td>
<td>
<p>list. The list of the task views.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following vectors:
</p>

<ul>
<li><p> spkgs: The selected packages that contain the keyword(s).
</p>
</li>
<li><p> inTV: The packages that contain the keyword(s) already refereed in the task view.
</p>
</li>
<li><p> notinTV: The packages that contain the keyword(s) not (yet) refereed in the task view.
</p>
</li>
<li><p> inTV_in: Among the packages available in the task view, those installed in the computer.
</p>
</li>
<li><p> inTV_un: Among the packages available in the task view, those not installed in the computer.
</p>
</li>
<li><p> notinTV_in: Among the packages not refereed in the task view, those installed in the computer.
</p>
</li>
<li><p> notinTV_un: Among the packages not refereed in the task view, those not installed in the computer.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>### TASK VIEW MAINTENANCE (tvdb + crandb)
## In real life, download crandb and tvdb from CRAN or load them from your directory
## with functions crandb_down(), crandb_load(), tvdb_down(), tvdb_load().
## In this example, we use small files.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
tvdb_load(system.file("data", "ztvdb.rda", package = "RWsearch"))

## List the task views
tvdb_vec()

## Search for the recent packages in crandb that contain the keyword
## and verify that the packages are already refereed in the task view.
(lst &lt;- s_crandb_tvdb("Gumbel", tv = "Distributions", from = "2018-01-01"))

if (interactive()) {
p_display7(lst[c("inTV", "notinTV")], dir = file.path(tempdir(), "scrandbtvdb"))
}
</code></pre>

<hr>
<h2 id='s_hs'>Search Packages and Functions in Installed Packages</h2><span id='topic+s_hs'></span>

<h3>Description</h3>

<p><code>s_hs</code> is a wrapper of the well known function <code>??</code> and its parent function
<code>help.search</code>. Visit the help page <code><a href="utils.html#topic+help.search">help.search</a></code> for details
on the various arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s_hs(..., char = NULL, fields = c("alias", "concept", "title"), apropos,
  keyword, whatis, ignore.case = TRUE, package = NULL, agrep = NULL,
  use_UTF8 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="s_hs_+3A_...">...</code></td>
<td>
<p>one single character string recognized by <code><a href="#topic+cnsc">cnsc</a></code>.
One and only one pattern</p>
</td></tr>
<tr><td><code id="s_hs_+3A_char">char</code></td>
<td>
<p>(name to) a single character string. Use this argument if <code>...</code>
fails or if you call the function from another function. If used,
argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="s_hs_+3A_fields">fields</code></td>
<td>
<p>See <code>help.search</code>.</p>
</td></tr>
<tr><td><code id="s_hs_+3A_apropos">apropos</code></td>
<td>
<p>See <code>help.search</code>.</p>
</td></tr>
<tr><td><code id="s_hs_+3A_keyword">keyword</code></td>
<td>
<p>See <code>help.search</code>.</p>
</td></tr>
<tr><td><code id="s_hs_+3A_whatis">whatis</code></td>
<td>
<p>See <code>help.search</code>.</p>
</td></tr>
<tr><td><code id="s_hs_+3A_ignore.case">ignore.case</code></td>
<td>
<p>See <code>help.search</code>.</p>
</td></tr>
<tr><td><code id="s_hs_+3A_package">package</code></td>
<td>
<p>See <code>help.search</code>.</p>
</td></tr>
<tr><td><code id="s_hs_+3A_agrep">agrep</code></td>
<td>
<p>See <code>help.search</code>.</p>
</td></tr>
<tr><td><code id="s_hs_+3A_use_utf8">use_UTF8</code></td>
<td>
<p>See <code>help.search</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='s_sos'>Search Packages and Functions on R-Project Help pages and RDocumentation</h2><span id='topic+s_sos'></span><span id='topic+s_man'></span>

<h3>Description</h3>

<p><code>s_sos</code> searchs in all R documentation packages and functions that contain
one or several keywords, open the default browser and display the results in two 
html pages, one for the package::functions and one for the package::vignettes.
<code>s_sos</code> usually find more results than <code><a href="#topic+s_crandb">s_crandb</a></code> as it goes 
deeper in the documentation, down to the function level. An internet connection 
is required to reach the website <a href="https://search.r-project.org">https://search.r-project.org</a>. 
<code>s_sos</code> is a minimal wrapper of the function <em>sos::findFn</em>.
Use this function if you need advanced search options.
</p>
<p><code>s_man</code> sends a query to the same search engine but the results is returned 
in a less edited format than <code>s_sos</code>. Use the function <code><a href="#topic+p_man">p_man</a></code> 
if your search is about a package and not a function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s_sos(..., char = NULL)

s_man(..., char = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="s_sos_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
One or several keywords.</p>
</td></tr>
<tr><td><code id="s_sos_+3A_char">char</code></td>
<td>
<p>(name to) a character vector. Use this argument if <code>...</code> fails
or if you call the function from another function. If used,
argument <code>...</code> is ignored.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="https://CRAN.R-project.org/package=sos">https://CRAN.R-project.org/package=sos</a> (index and vignette).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Search using standard or non-standard content
## and display the results in a browser.
if (interactive()) {
s_sos("chemical reaction")
(res &lt;- s_sos(distillation))
tail(data.frame(res))
s_man("cran_incoming")
}
</code></pre>

<hr>
<h2 id='s_tvdb'>Search Packages in Task Views</h2><span id='topic+s_tvdb'></span>

<h3>Description</h3>

<p><code>s_tvdb</code> searchs if one or several package(s) are referred in some task views
and lists these task views.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s_tvdb(..., char = NULL, tvdb = get("tvdb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="s_tvdb_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
The names of one or several task views.</p>
</td></tr>
<tr><td><code id="s_tvdb_+3A_char">char</code></td>
<td>
<p>(name to) a character vector or a list. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
<tr><td><code id="s_tvdb_+3A_tvdb">tvdb</code></td>
<td>
<p>list. The list of the task views.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## In real life, download tvdb from CRAN or load it from your directory
## with functions tvdb_down() or tvdb_load().
## In this example, we use a small file.
tvdb_load(system.file("data", "ztvdb.rda", package = "RWsearch"))
tvdb_dfr()
s_tvdb(actuar, FatTailsR, MASS, zoo, NotAPkg)

</code></pre>

<hr>
<h2 id='tvdb'>Task Views (tvdb.rda)</h2><span id='topic+tvdb'></span><span id='topic+tvdb_down'></span><span id='topic+tvdb_load'></span><span id='topic+tvdb_vec'></span><span id='topic+tvdb_dfr'></span><span id='topic+tvdb_list'></span><span id='topic+tvdb_pkgs'></span>

<h3>Description</h3>

<p><code>tvdb_down</code> downloads from CRAN the file &quot;Views.rds&quot;, a file refreshed every day that
describes the task views available in CRAN for this day, rearranges the list in an
alphabetical order and gives names to the list names, then loads in .GlobalEnv
this list (of class ctvlist) under the name <code>tvdb</code> and saves it with the filename
<code>tvdb.rda</code>.
</p>
<p><code>tvdb_load</code> loads the file <code>filename</code> in .GlobalEnv under the name <code>tvdb</code>.
Equivalent to <code>load("tvdb.rda")</code>.
</p>
<p><code>tvdb_vec</code> displays the list of the task views. There are 36 task views in August 2018.
</p>
<p><code>tvdb_dfr</code> extracts from <code>tvdb</code> a data.frame <em>version, name, topic</em>
of the task views.
</p>
<p><code>tvdb_list</code> extracts from <code>tvdb</code> the list of the task views and the referenced
packages.
</p>
<p><code>tvdb_pkgs</code> displays the packages referenced by one or several task views.
</p>
<p>Visit <code><a href="#topic+s_crandb_tvdb">s_crandb_tvdb</a></code> to conduct task view maintenance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tvdb_down(dir = ".", repos = getOption("repos")[1])

tvdb_load(filename = "tvdb.rda")

tvdb_vec(tvdb = get("tvdb", envir = .GlobalEnv))

tvdb_dfr(tvdb = get("tvdb", envir = .GlobalEnv))

tvdb_list(tvdb = get("tvdb", envir = .GlobalEnv))

tvdb_pkgs(..., char = NULL, tvdb = get("tvdb", envir = .GlobalEnv))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tvdb_+3A_dir">dir</code></td>
<td>
<p>character. The directory where &quot;tvdb.rda&quot; is saved.
Default value <code>"."</code> is the current directory.</p>
</td></tr>
<tr><td><code id="tvdb_+3A_repos">repos</code></td>
<td>
<p>character. The address of your local CRAN.</p>
</td></tr>
<tr><td><code id="tvdb_+3A_filename">filename</code></td>
<td>
<p>character. The path to file &quot;tvdb.rda&quot;.
The default is to read in the current directory.</p>
</td></tr>
<tr><td><code id="tvdb_+3A_tvdb">tvdb</code></td>
<td>
<p>list. The list of the task views.</p>
</td></tr>
<tr><td><code id="tvdb_+3A_...">...</code></td>
<td>
<p>any format recognized by <code><a href="#topic+cnsc">cnsc</a></code>, except list.
The names of one or several task views.</p>
</td></tr>
<tr><td><code id="tvdb_+3A_char">char</code></td>
<td>
<p>(name to) a character vector or a list. Use this argument if
<code>...</code> fails or if you call the function from another function.
If used, argument <code>...</code> is ignored.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>### DOWNLOAD AND VISUALIZE THE TASK VIEWS (tvdb)
## In real life, download tvdb from CRAN or load it from your directory
## with functions tvdb_down() or tvdb_load().
## In this example, we use a small file.
tvdb_load(system.file("data", "ztvdb.rda", package = "RWsearch"))
length(tvdb)

## List the task views
tvdb_vec()
tvdb_dfr()
tvdb_pkgs("Robust")
lengths(tvdb_list())

## Here, 'lst' is subsetted from the small crandb file.
crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
funIN &lt;- function (x, y) x[match(x, y, nomatch = 0)  &gt; 0]
lst   &lt;- lapply(tvdb_list()[1:2], funIN, crandb$Package) ; lst

if (interactive()) p_display7(lst[[1]], dir = file.path(tempdir(), "ptvdbdown"))

</code></pre>

<hr>
<h2 id='zcrandb'>File zcrandb.rda: A Subset of crandb Dataset</h2><span id='topic+zcrandb'></span>

<h3>Description</h3>

<p>The file <em>zcrandb.rda</em> contains a data.frame named <code>crandb</code> of
dimension 94 x 69. It is is a subset of 94 packages of the large <code>crandb</code>
data.frame usually downloaded from CRAN by the function <code><a href="#topic+crandb_down">crandb_down</a></code>.
The use of <code>zcrandb.rda</code> avoids inappropriate connections to CRAN and
increases the speed of the examples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>crandb_load(system.file("data", "zcrandb.rda", package = "RWsearch"))
</code></pre>

<hr>
<h2 id='ztvdb'>File ztvdb.rda: A Subset of tvdb Dataset</h2><span id='topic+ztvdb'></span>

<h3>Description</h3>

<p>The file <em>ztvdb.rda</em> contains a list of 4 task views named <code>tvdb</code>.
It is a subset of the large file <em>tvdb.rda</em> that contain 45 task views
usually downloaded from CRAN by the function <code><a href="#topic+tvdb_down">tvdb_down</a></code>.
The use of <em>ztvdb.rda</em> avoids inappropriate connections to CRAN and
increases the speed of the examples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tvdb_load(system.file("data", "ztvdb.rda", package = "RWsearch"))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
