<!DOCTYPE html><html lang="en"><head><title>Help for package RxnSim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {RxnSim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#RxnSim-package'><p>Functions to compute chemical reaction and molecular similarity</p></a></li>
<li><a href='#ms.compute'><p>Computes Similarity of Molecules</p></a></li>
<li><a href='#rs.clearCache'><p>Clears Fingerprint Cache</p></a></li>
<li><a href='#rs.compute'><p>Computes Similarity of Reactions</p></a></li>
<li><a href='#rs.makeDB'><p>Converts Text File to Reaction Database</p></a></li>
<li><a href='#rs.mask'><p>Masks a Sub-structure in Input Molecule or Reaction</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Functions to Compute Chemical and Chemical Reaction Similarity</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-07-15</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Varun Giri &lt;varungiri@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Methods to compute chemical similarity between two or more reactions and molecules. Allows masking of chemical substructures for weighted similarity computations. Uses packages 'rCDK' and 'fingerprint' for cheminformatics functionality. Methods for reaction similarity and sub-structure masking are as described in: Giri et al. (2015) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtv416">doi:10.1093/bioinformatics/btv416</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, rJava, fingerprint, data.table, rcdk (&ge; 3.8.1)</td>
</tr>
<tr>
<td>Author:</td>
<td>Varun Giri [aut, cre]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-19 15:49:15 UTC; varun</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-19 22:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='RxnSim-package'>Functions to compute chemical reaction and molecular similarity</h2><span id='topic+RxnSim-package'></span><span id='topic+RxnSim'></span><span id='topic+rxnsim'></span>

<h3>Description</h3>

<p>RxnSim provides methods to compute molecular and reaction similarity. It uses rCDK package (R Interface to the CDK Libraries) and fingerprints package for chemoinformatic routines.
</p>


<h3>Details</h3>

<p>RxnSim provides methods to compute chemical similarity between two or more reactions and molecules. Molecular similarity is computed based on structural features. Reaction similarity is a function of similarities of participating molecules. The package provides multiple methods to extract structural features as fingerprints (or feature vectors) and similarity metrics. It additionally provides functionality to mask chemical substructures for weighted similarity computations. It uses rCDK and fingerprint packages for cheminformatics functionality. 
</p>
<p><strong>User functions:</strong>
</p>

<dl>
<dt><code><a href="#topic+rs.compute">rs.compute</a></code></dt><dd><p>computes similarity between two reactions.</p>
</dd>
<dt><code><a href="#topic+rs.compute">rs.compute.list</a></code></dt><dd><p>computes similarity between all pairs of reactions from two lists.</p>
</dd>
<dt><code><a href="#topic+rs.compute">rs.compute.sim.matrix</a></code></dt><dd><p>computes pairwise similarity between all reactions in a list.</p>
</dd>
<dt><code><a href="#topic+rs.compute">rs.compute.DB</a></code></dt><dd><p>computes similarity of a reaction to those in a reaction database (DB) object read from a text file.</p>
</dd>
<dt><code><a href="#topic+rs.makeDB">rs.makeDB</a></code></dt><dd><p>reads a text file containing EC Numbers, Reaction Names and Reaction SMILES and converts it into a reaction DB object.</p>
</dd>
<dt><code><a href="#topic+ms.compute">ms.compute</a></code></dt><dd><p>computes similarity between two molecules.</p>
</dd>
<dt><code><a href="#topic+ms.compute.sim.matrix">ms.compute.sim.matrix</a></code></dt><dd><p>computes pairwise similarity between all molecules in a list.</p>
</dd>
<dt><code><a href="#topic+rs.clearCache">rs.clearCache</a></code></dt><dd><p>clears fingerprint cache.</p>
</dd>
<dt><code><a href="#topic+rs.mask">rs.mask</a></code></dt><dd><p>substitutes given sub-structure in the molecules of a reaction by a user defined mask.</p>
</dd>
<dt><code><a href="#topic+ms.mask">ms.mask</a></code></dt><dd><p>substitutes given sub-structure in a molecule by a user defined mask.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Varun Giri <a href="mailto:varungiri@gmail.com">varungiri@gmail.com</a>
</p>
<p>Maintainer: Varun Giri
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rs.compute">rs.compute</a></code>, <code><a href="#topic+ms.compute">ms.compute</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Reaction similarity
rs.compute('CCCO&gt;&gt;CCC=O', 'CC(O)C&gt;&gt;CC(=O)C')

# Metabolite similarity
ms.compute('CCC=O', 'CC(O)C')
</code></pre>

<hr>
<h2 id='ms.compute'>Computes Similarity of Molecules</h2><span id='topic+ms.compute'></span><span id='topic+ms.compute.sim.matrix'></span><span id='topic+ms.compute.PCA'></span>

<h3>Description</h3>

<p>Computes chemical similarity between two (or more) input molecules.<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ms.compute (molA, molB, format = 'smiles', standardize = TRUE, explicitH = FALSE,
            sim.method = 'tanimoto', fp.type = 'extended', fp.mode = 'bit', fp.depth = 6,
            fp.size = 1024, fpCached = FALSE)
ms.compute.sim.matrix (molA, format = 'smiles', standardize = TRUE, explicitH = FALSE,
            sim.method = 'tanimoto', fp.type = 'extended', fp.mode = 'bit', fp.depth = 6,
            fp.size = 1024, clearCache = TRUE)
ms.compute.PCA(molA, format = 'smiles', standardize = TRUE, explicitH = FALSE, 
            fp.type = 'extended', fp.mode = 'bit', fp.depth = 6, fp.size = 1024,
            clearCache = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ms.compute_+3A_mola">molA</code></td>
<td>
<p>input molecule in SMILES format or name (with path) of MDL MOL file. <code>ms.compute.sim.matrix</code> accepts list of molecules as input.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_molb">molB</code></td>
<td>
<p>input molecule in SMILES format or name (with path) of MDL MOL file.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_format">format</code></td>
<td>
<p>specifies format of input molecule(s). Molecule(s) can be provided in one of following formats: <code>'SMILES'</code> (default) or <code>'MOL'</code>.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_standardize">standardize</code></td>
<td>
<p>suppresses all explicit hydrogen if set as <code>TRUE</code> (default).</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_explicith">explicitH</code></td>
<td>
<p>converts all implicit hydrogen to explicit if set as <code>TRUE</code>. It is set as <code>FALSE</code> by default.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_sim.method">sim.method</code></td>
<td>
<p>similarity metric to be used to evaluate molecule similarity. Allowed types include:<br /> <code>'simple'</code>, <code>'jaccard'</code>, <code>'tanimoto'</code> (default), <code>'russelrao'</code>, <code>'dice'</code>, <code>'rodgerstanimoto'</code>, <code>'achiai'</code>, <code>'cosine'</code>, <code>'kulczynski2'</code>, <code>'mt'</code>, <code>'baroniurbanibuser'</code>, <code>'tversky'</code>, <code>'robust'</code>, <code>'hamann'</code>, <code>'pearson'</code>, <code>'yule'</code>, <code>'mcconnaughey'</code>, <code>'simpson'</code>, <code>'jaccard-count'</code> and <code>'tanimoto-count'</code>.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_fp.type">fp.type</code></td>
<td>
<p>fingerprint type to use. Allowed types include:<br /> <code>'standard'</code>, <code>'extended'</code> (default), <code>'graph'</code>, <code>'estate'</code>, <code>'hybridization'</code>, <code>'maccs'</code>, <code>'pubchem'</code>, <code>'kr'</code>, <code>'shortestpath'</code>, <code>'signature'</code> and <code>'circular'</code>.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_fp.mode">fp.mode</code></td>
<td>
<p>fingerprint mode to be used. It can either be set to <code>'bit'</code> (default) or <code>'count'</code>.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_fp.depth">fp.depth</code></td>
<td>
<p>search depth for fingerprint construction. This argument is ignored for <code>'pubchem'</code>, <code>'maccs'</code>, <code>'kr'</code> and <code>'estate'</code> fingerprints.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_fp.size">fp.size</code></td>
<td>
<p>length of the fingerprint bit string. This argument is ignored for <code>'pubchem'</code>, <code>'maccs'</code>, <code>'kr'</code>, <code>'estate'</code>, <code>'circular'</code> (count mode) and <code>'signature'</code> fingerprints.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_fpcached">fpCached</code></td>
<td>
<p>boolean that enables fingerprint caching. It is set to <code>FALSE</code> by default.</p>
</td></tr>
<tr><td><code id="ms.compute_+3A_clearcache">clearCache</code></td>
<td>
<p>boolean that resets the cache before (and after) processing molecule lists.  It is set to <code>TRUE</code> by default. Cache can also be explicitly cleared by using <code><a href="#topic+rs.clearCache">rs.clearCache</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <code><a href="#topic+rs.compute">rs.compute</a></code> functions, for details for fingerprints and similarity matrices.
<code>ms.compute</code> can use fingerprint caching by enabling <code>fpCached</code> option. <code>ms.compute</code> and <code>ms.compute.sim.matrix</code> use same cache as <code>rs.compute</code> and other functions in the package.
<code>ms.compute.PCA</code> computes PCA based on the fingerprints using prcomp funtion.
</p>


<h3>Value</h3>

<p>Returns similarity value(s).
</p>
<table role = "presentation">
<tr><td><code>ms.compute</code></td>
<td>
<p>returns a similarity value.</p>
</td></tr>
<tr><td><code>ms.compute.sim.matrix</code></td>
<td>
<p>returns a <code class="reqn">m \times m</code> symmetric matrix of similarity values. <code class="reqn">m</code> is the length of the input list.</p>
</td></tr>
<tr><td><code>ms.compute.PCA</code></td>
<td>
<p>returns prcomp object.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Fingerprint cache stores fingerprints generated for a molecule index based on its SMILES. When caching is enabled, the fingerprint for a molecule, if present, is retrieved from the cache. The parameters pertaining to fingerprint generation are thus ignored. If the fingerprint for the molecule is not already cached, fingerprint based on the input parameters is generated and stored in the cache.
</p>


<h3>Author(s)</h3>

<p>Varun Giri <a href="mailto:varungiri@gmail.com">varungiri@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rs.compute">rs.compute</a></code>, <code><a href="#topic+rs.clearCache">rs.clearCache</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ms.compute('N', '[H]N([H])[H]', standardize = FALSE)
</code></pre>

<hr>
<h2 id='rs.clearCache'>Clears Fingerprint Cache</h2><span id='topic+rs.clearCache'></span>

<h3>Description</h3>

<p>Clears fingerprint cache. <code>rs.clearCache</code> should be called before fingerprint
method (or related properties) are modified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rs.clearCache()
</code></pre>


<h3>Value</h3>

<p>No return value
</p>


<h3>Author(s)</h3>

<p>Varun Giri <a href="mailto:varungiri@gmail.com">varungiri@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rs.compute">rs.compute</a></code>, <code><a href="#topic+ms.compute">ms.compute</a></code>
</p>

<hr>
<h2 id='rs.compute'>Computes Similarity of Reactions</h2><span id='topic+rs.compute'></span><span id='topic+rs.compute.list'></span><span id='topic+rs.compute.sim.matrix'></span><span id='topic+rs.compute.DB'></span>

<h3>Description</h3>

<p>Computes similarity between two (or more) input reactions.<br />
<code>rs.compute</code> computes similarity of two reactions.<br />
<code>rs.compute.list</code> computes similarity of two lists of reactions.<br />
<code>rs.compute.sim.matrix</code> computes similarity of reactions in a list.<br />
<code>rs.compute.DB</code> computes similarity of a reaction against a database (parsed from text file).<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rs.compute (rxnA, rxnB, format = 'rsmi', standardize = TRUE, explicitH = FALSE, 
            reversible = TRUE, algo = 'msim', sim.method = 'tanimoto', 
            fp.type = 'extended', fp.mode = 'bit', fp.depth = 6, fp.size = 1024,
            verbose = FALSE, fpCached = FALSE)
rs.compute.list (rxnA, rxnB, format = 'rsmi', standardize = TRUE, explicitH = FALSE,
              reversible = TRUE, algo = 'msim', sim.method = 'tanimoto',
              fp.type = 'extended', fp.mode = 'bit',fp.depth = 6, fp.size = 1024,
              clearCache = TRUE)
rs.compute.sim.matrix (rxnA, format = 'rsmi', standardize = TRUE, explicitH = FALSE,
                    reversible = TRUE, algo = 'msim', sim.method = 'tanimoto', 
                    fp.type = 'extended', fp.mode = 'bit', fp.depth = 6, fp.size = 1024,
                    clearCache = TRUE)
rs.compute.DB (rxnA, DB, format = 'rsmi', ecrange = '*', reversible = TRUE,
              algo = 'msim', sim.method = 'tanimoto', sort = TRUE, fpCached = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rs.compute_+3A_rxna">rxnA</code></td>
<td>
<p>input reaction in RSMI format or name (with path) of MDL RXN file. <code>rs.compute.list</code> and <code>rs.compute.sim.matrix</code> accept list of reactions as input.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_rxnb">rxnB</code></td>
<td>
<p>input reaction in RSMI format or name (with path) of MDL RXN file. <code>rs.compute.list</code> accepts list of reactions as input.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_db">DB</code></td>
<td>
<p>parsed database object as returned by <code><a href="#topic+rs.makeDB">rs.makeDB</a></code>.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_format">format</code></td>
<td>
<p>specifies format of input reaction(s). Reaction(s) can be provided in one of following formats: 'RSMI' (default) or 'RXN'.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_ecrange">ecrange</code></td>
<td>
<p>EC number(s) search pattern while comparing against reaction DB. * is used as wildcard. E.g., 1.2.1.* will restricted search to all reactions with EC numbers starting with 1.2.1.- .</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_standardize">standardize</code></td>
<td>
<p>suppresses all explicit hydrogen if set as <code>TRUE</code> (default).</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_explicith">explicitH</code></td>
<td>
<p>converts all implicit hydrogen to explicit if set as <code>TRUE</code>. It is set as <code>FALSE</code> by default.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_reversible">reversible</code></td>
<td>
<p>boolean that indicates reversibility of input reaction(s). If set as <code>TRUE</code> (default), reaction(s) are aligned by comparing them in forward direction and by reversing one of them to compute maximum similarity value.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_algo">algo</code></td>
<td>
<p>reaction similarity algorithm to be used. One of following algorithms can be used: <code>'msim'</code> (default), <code>'msim_max'</code>, <code>'rsim'</code> and <code>'rsim2'</code>. See description for the details of the algorithms.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_sim.method">sim.method</code></td>
<td>
<p>similarity metric to be used to evaluate reaction similarity. Allowed types include:<br /> <code>'simple'</code>, <code>'jaccard'</code>, <code>'tanimoto'</code> (default), <code>'russelrao'</code>, <code>'dice'</code>, <code>'rodgerstanimoto'</code>, <code>'achiai'</code>, <code>'cosine'</code>, <code>'kulczynski2'</code>, <code>'mt'</code>, <code>'baroniurbanibuser'</code>, <code>'tversky'</code>, <code>'robust'</code>, <code>'hamann'</code>, <code>'pearson'</code>, <code>'yule'</code>, <code>'mcconnaughey'</code>, <code>'simpson'</code>, <code>'jaccard-count'</code> and <code>'tanimoto-count'</code>.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_fp.type">fp.type</code></td>
<td>
<p>fingerprint type to use. Allowed types include:<br /> <code>'standard'</code>, <code>'extended'</code> (default), <code>'graph'</code>, <code>'estate'</code>, <code>'hybridization'</code>, <code>'maccs'</code>, <code>'pubchem'</code>, <code>'kr'</code>, <code>'shortestpath'</code>, <code>'signature'</code> and <code>'circular'</code>.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_fp.mode">fp.mode</code></td>
<td>
<p>fingerprint mode to be used. It can either be set to <code>'bit'</code> (default) or <code>'count'</code>.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_fp.depth">fp.depth</code></td>
<td>
<p>search depth for fingerprint construction. This argument is ignored for <code>'pubchem'</code>, <code>'maccs'</code>, <code>'kr'</code> and <code>'estate'</code> fingerprints.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_fp.size">fp.size</code></td>
<td>
<p>length of the fingerprint bit string. This argument is ignored for the <code>'pubchem'</code>, <code>'maccs'</code>, <code>'kr'</code>, <code>'estate'</code>, <code>'circular'</code> (count mode) and <code>'signature'</code> fingerprints.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_verbose">verbose</code></td>
<td>
<p>boolean that enables display of detailed molecule pairing and reaction alignment (and respective similarity values). The argument is ignored for <code>'rsim2'</code> algorithm.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_sort">sort</code></td>
<td>
<p>boolean than enables <code>rs.compute.DB</code> to return data frame sorted based upon decreasing value of similarities.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_fpcached">fpCached</code></td>
<td>
<p>boolean that enables fingerprint caching. It is set to <code>FALSE</code> by default.</p>
</td></tr>
<tr><td><code id="rs.compute_+3A_clearcache">clearCache</code></td>
<td>
<p>boolean that resets the cache before (and after) processing reaction lists. It is set to <code>TRUE</code> by default. Cache can also be explicitly cleared using <code>rs.clearCache</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><span class="pkg">RxnSim</span> implements four algorithms to compute reaction similarity, namely <code>msim</code>, <code>msim_max</code>, <code>rsim</code> and <code>rsim2</code>.
</p>

<dl>
<dt><code>msim</code></dt><dd><p>is based on individual similarities of molecules in two reactions. First, each reactant (product) of a reaction is paired with an equivalent (similar) reactant (product) of the other reaction based on pairwise similarity values using hierarchical grouping. A <code>0</code> similarity value is assigned to each unpaired molecule. Reaction similarity is then computed by averaging the similarity values for each pair of equivalent molecule(s) and unpaired molecule(s). Molecule equivalences computed can be reviewed using <code>verbose</code> mode in <code>rs.compute</code>.</p>
</dd>
<dt><code>msim_max</code></dt><dd><p>reaction similarity is computed in the same way as described for <code>msim</code> except that the unpaired molecules are not used for computing average.</p>
</dd>
<dt><code>rsim</code></dt><dd><p>is based on cumulative features of reactant(s) and product(s) of two reactions. Each reaction is represented by two fingerprints, one each for the reactants and another for products. Reaction similarity is computed by averaging similarity values obtained by comparing reactants fingerprint and products fingerprints.</p>
</dd>
<dt><code>rsim2</code></dt><dd><p>is based on cumulative features of all molecules in a reaction forming a reaction fingerprint. Reaction similarity is computed based on the reaction fingerprints of two reactions.</p>
</dd>
</dl>

<p>For reversible reactions (<code>reversible = TRUE</code>), apart from comparing reactions in the forward direction they are also compared by reversing one of the reactions. The greater of the two similarity values is reported.
</p>
<p><strong>Fingerprint Caching</strong><br />
<code>rs.compute</code> and <code>rs.compute.DB</code> functions can use fingerprint caching. If <code>fpCached</code> is set as <code>TRUE</code>, cache is queried first before generating fingerprints. Any new fingerprint generated is stored in the cache. Setting <code>fpCached = FALSE</code> makes no change to cache. Cache can be cleared by calling <code><a href="#topic+rs.clearCache">rs.clearCache</a></code>.
</p>
<p><code>rs.compute.list</code> and <code>rs.compute.sim.matrix</code> functions internally use caching. To ensure consistency of fingerprints, <code><a href="#topic+rs.clearCache">rs.clearCache</a></code> is called internally. Use <code>clearCache = FALSE</code> to override this behaviour; it will use current state of cache and add new fingerprints to it.
</p>
<p>Same cache is used for all functions.
</p>
<p><strong>Similarity metric included in <span class="pkg">RxnSim</span>.</strong> These metric (except <code>jaccard-count</code> and <code>tanimoto-count</code>) are derived from <span class="pkg"><a href="fingerprint.html#topic+distance">fingerprint pacakge</a></span>.
</p>

<table>
<tr>
 <td style="text-align: left;">
  <b>ID</b> </td><td style="text-align: left;"> <b>Name</b> </td><td style="text-align: left;"> <b>Remarks</b></td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>simple</code> </td><td style="text-align: left;"> Sokal &amp; Michener </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>jaccard</code> </td><td style="text-align: left;"> Jaccard </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>tanimoto</code> </td><td style="text-align: left;"> Tanimoto (bit) </td><td style="text-align: left;"> bit and count </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>jaccard-count</code> </td><td style="text-align: left;"> Jaccard (count) </td><td style="text-align: left;"> count </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>tanimoto-count</code> </td><td style="text-align: left;"> Tanimoto (count) </td><td style="text-align: left;"> count ^</td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>dice</code> </td><td style="text-align: left;"> Dice (bit)</td><td style="text-align: left;"> bit and count </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>russelrao</code> </td><td style="text-align: left;"> Russel And Rao </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>rodgerstanimoto</code> </td><td style="text-align: left;"> Roger And Tanimoto </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>achiai</code> </td><td style="text-align: left;"> Ochiai </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>cosine</code> </td><td style="text-align: left;"> Cosine </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>kulczynski2</code> </td><td style="text-align: left;"> Kulczynski 2 </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>mt</code> </td><td style="text-align: left;"> Modified Tanimoto </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>baroniurbanibuser</code> </td><td style="text-align: left;"> Baroni-Urbani/Buser </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>robust</code> </td><td style="text-align: left;"> Robust (bit)</td><td style="text-align: left;"> bit and count </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>tversky</code> </td><td style="text-align: left;"> Tversky* </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>hamann</code> </td><td style="text-align: left;"> Hamann </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>pearson</code> </td><td style="text-align: left;"> Pearson </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>yule</code> </td><td style="text-align: left;"> Yule </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>mcconnaughey</code> </td><td style="text-align: left;"> McConnaughey </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>simpson</code> </td><td style="text-align: left;"> Simpson </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>*Tversky coefficients can be specified by combining them into a vector, e.g., <code>c('tversky', a, b)</code>.
</p>
<p><code>tanimoto</code> (bit), <code>dice</code> (bit) and <code>robust</code> (bit) compute similarity of feature vectors (count mode) by translating them to equivalent fingerprint vectors. Default similarity metric used is <code>tanimoto</code>.<br /><br />
<strong>List of fingerprints included in <span class="pkg">RxnSim</span></strong>. These are derived from <a href="rcdk.html#topic+get.fingerprint">rCDK package</a>.
</p>

<table>
<tr>
 <td style="text-align: left;">
  <b>ID</b> </td><td style="text-align: left;"> <b>Name of the Fingerprint</b> </td><td style="text-align: left;"> <b>Mode</b></td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>standard</code> </td><td style="text-align: left;"> Standard </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>extended</code> </td><td style="text-align: left;"> Extended </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>estate</code> </td><td style="text-align: left;"> EState </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>graph</code> </td><td style="text-align: left;"> Graphonly </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>hybridization</code> </td><td style="text-align: left;"> Hybridization </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>maccs</code> </td><td style="text-align: left;"> MACCS </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>pubchem</code> </td><td style="text-align: left;"> Pubchem </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>kr</code> </td><td style="text-align: left;"> Klekota-Roth </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>shortestpath</code> </td><td style="text-align: left;"> Shortestpath </td><td style="text-align: left;"> bit </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>signature</code> </td><td style="text-align: left;"> Signature </td><td style="text-align: left;"> count </td>
</tr>
<tr>
 <td style="text-align: left;">
  <code>circular</code> </td><td style="text-align: left;"> Circular </td><td style="text-align: left;"> bit and count </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<table role = "presentation">
<tr><td><code>rs.compute</code></td>
<td>
<p>returns a similarity value.</p>
</td></tr>
<tr><td><code>rs.compute.list</code></td>
<td>
<p>returns a <code class="reqn">m \times n</code> matrix of similarity values. <code class="reqn">m</code> and <code class="reqn">n</code> are the length of two input lists respectively.</p>
</td></tr>
<tr><td><code>rs.compute.sim.matrix</code></td>
<td>
<p>returns a <code class="reqn">m \times m</code> symmetric matrix of similarity values. <code class="reqn">m</code> is the length of the input list.</p>
</td></tr>
<tr><td><code>rs.compute.DB</code></td>
<td>
<p>returns a data frame.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>While using fingerprint caching (by setting <code>fpCached = TRUE</code> in <code>rs.compute</code> and <code>rs.compute.DB</code> or <code>clearCache = FALSE</code> in <code>rs.compute.list</code> and <code>rs.compute.sim.matrix</code>), ensure that the fingerprints are generated using same parameters values (<code>fp.type</code>, <code>fp.mode</code>, <code>fp.depth</code> and <code>fp.size</code>). To reset cache, call <code><a href="#topic+rs.clearCache">rs.clearCache</a></code>.
</p>
<p><code>rs.compute.DB</code> uses same parameter values for creating fingerprint as used for (and stored with) DB object (created using <code><a href="#topic+rs.makeDB">rs.makeDB</a></code>) passed as argument.
</p>


<h3>Author(s)</h3>

<p>Varun Giri <a href="mailto:varungiri@gmail.com">varungiri@gmail.com</a>
</p>


<h3>References</h3>

<p>^ Carbonell, P., Planson, A-G., Fichera, D., &amp; Faulon J-L. (2011) A retrosynthetic biology approach to metabolic pathway design for therapeutic production. <em>BMC Systems Biology</em>, <b>5</b>:122.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rs.makeDB">rs.makeDB</a></code>, <code><a href="#topic+rs.clearCache">rs.clearCache</a></code>, <code><a href="#topic+ms.compute">ms.compute</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Reaction similarity using msim algorithm
rs.compute(rct1, rct2, verbose = TRUE)
</code></pre>

<hr>
<h2 id='rs.makeDB'>Converts Text File to Reaction Database</h2><span id='topic+rs.makeDB'></span>

<h3>Description</h3>

<p>Reads and parses input text file containing reaction smiles into reaction database object. The reaction database is used for querying reaction similarity of candidate reactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rs.makeDB (txtFile, header = FALSE, sep = '\t', standardize = TRUE, explicitH = FALSE,
          fp.type = 'extended', fp.mode = 'bit', fp.depth = 6, fp.size = 1024,
          useMask = FALSE, maskStructure, mask, recursive = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rs.makeDB_+3A_txtfile">txtFile</code></td>
<td>
<p>input file containing EC numbers, reaction name and RSMI. See description for format of input file.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_header">header</code></td>
<td>
<p>boolean to indicate if the input file contains a header. It is set to <code>FALSE</code> by default.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_sep">sep</code></td>
<td>
<p>the field separator character to be used while reading the input file.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_standardize">standardize</code></td>
<td>
<p>suppresses all explicit hydrogen if set as <code>TRUE</code> (default).</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_explicith">explicitH</code></td>
<td>
<p>converts all implicit hydrogen to explicit if set as <code>TRUE</code>. It is set as <code>FALSE</code> by default.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_fp.type">fp.type</code></td>
<td>
<p>Fingerprint type to use. Allowed types include:<br /> <code>'standard'</code>, <code>'extended'</code> (default), <code>'graph'</code>, <code>'estate'</code>, <code>'hybridization'</code>, <code>'maccs'</code>, <code>'pubchem'</code>, <code>'kr'</code>, <code>'shortestpath'</code>, <code>'signature'</code> and <code>'circular'</code>.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_fp.mode">fp.mode</code></td>
<td>
<p>fingerprint mode to be used. It can either be set to <code>'bit'</code> (default) or <code>'count'</code>.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_fp.depth">fp.depth</code></td>
<td>
<p>search depth for fingerprint construction. This argument is ignored for <code>'pubchem'</code>, <code>'maccs'</code>, <code>'kr'</code> and <code>'estate'</code> fingerprints.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_fp.size">fp.size</code></td>
<td>
<p>length of the fingerprint bit string. This argument is ignored for <code>'pubchem'</code>, <code>'maccs'</code>, <code>'kr'</code>, <code>'estate'</code>, <code>'circular'</code> (count mode) and <code>'signature'</code> fingerprints.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_usemask">useMask</code></td>
<td>
<p>boolean to indicate use of masking. If <code>TRUE</code>, each reaction is processed to mask given substructure. See <code><a href="#topic+rs.mask">rs.mask</a></code> for details.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_maskstructure">maskStructure</code></td>
<td>
<p>SMILES or SMARTS of the structure to be searched and masked.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_mask">mask</code></td>
<td>
<p>SMILES of structure to be used as mask.</p>
</td></tr>
<tr><td><code id="rs.makeDB_+3A_recursive">recursive</code></td>
<td>
<p>if <code>TRUE</code>, all the occurrences of input substructure are replaced recursively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parameters used to generate fingerprints are stored in the database object and returned with the parsed data. Same parameter values are used while parsing input reaction in <code><a href="#topic+rs.compute.DB">rs.compute.DB</a></code>.
</p>
<p>The input text file should contain following three fields, separated with  <code>TAB</code> (or any appropriate field separator). A field can be left blank.
</p>

<table>
<tr>
 <td style="text-align: left;">
[EC Number] </td><td style="text-align: left;"> [Reaction Name] </td><td style="text-align: left;"> [Reaction SMILES (RSMI)]
</td>
</tr>

</table>

<p>The package comes with a sample reaction database file extracted from Rhea database (Morgat et al., 2015). If no <code>textfile</code> is provided, default sample database file is used:
</p>
<pre>
rs.makeDB()
</pre>
<p>A larger dataset containing all reactions from Rhea database (v.83) is also provided with the package.
</p>


<h3>Value</h3>

<p>Returns a list, containing parsed input data, reaction fingerprints.
</p>
<table role = "presentation">
<tr><td><code>Data</code></td>
<td>
<p>data frame containing EC Numbers, Reaction Names and RSMI as read from the input file. MaskedRSMI are also included if masking is used.</p>
</td></tr>
<tr><td><code>FP</code></td>
<td>
<p>list of molecular fingerprints for each reaction in the input file. These fingerprints are further processed based on the reaction similarity algorithm.</p>
</td></tr>
</table>
<p>It also contains the parameter values used for generating fingerprints, viz., <code>standardize</code>, <code>explicitH</code>, <code>fp.type</code>,  <code>fp.mode</code>, <code>fp.depth</code> and <code>fp.size</code>.
</p>


<h3>Author(s)</h3>

<p>Varun Giri <a href="mailto:varungiri@gmail.com">varungiri@gmail.com</a>
</p>


<h3>References</h3>

<p>Morgat, A., Lombardot, T., Axelsen, K., Aimo, L., Niknejad, A., Hyka-Nouspikel, N., Coudert, E., Pozzato, M., Pagni, M., Moretti, S., Rosanoff, S., Onwubiko, J., Bougueleret, L., Xenarios, I., Redaschi, N., Bridge, A. (2017) Updates in Rhea - an expert curated resource of biochemical reactions. <em>Nucleic Acids Research</em>, <b>45</b>:D415-D418; doi: 10.1093/nar/gkw990
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rs.compute.DB">rs.compute.DB</a></code>, <code><a href="#topic+rs.mask">rs.mask</a></code>
</p>

<hr>
<h2 id='rs.mask'>Masks a Sub-structure in Input Molecule or Reaction</h2><span id='topic+rs.mask'></span><span id='topic+ms.mask'></span>

<h3>Description</h3>

<p>Replaces a sub-structure, provided as SMILES or SMARTS, with the given mask.<br />
<code>rs.mask</code> masks input sub-structure in reaction.<br />
<code>ms.mask</code> masks input sub-structure in molecule.<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rs.mask (substructure, mask, reaction, format = 'rsmi', standardize = TRUE,
         explicitH = FALSE, recursive = FALSE)
ms.mask (substructure, mask, molecule, format = 'smiles', standardize = TRUE,
         explicitH = FALSE, recursive = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rs.mask_+3A_substructure">substructure</code></td>
<td>
<p>SMILES or SMARTS of the structure to be searched and masked.</p>
</td></tr>
<tr><td><code id="rs.mask_+3A_mask">mask</code></td>
<td>
<p>SMILES of structure to be used as mask.</p>
</td></tr>
<tr><td><code id="rs.mask_+3A_reaction">reaction</code></td>
<td>
<p>Input reaction to be processed.</p>
</td></tr>
<tr><td><code id="rs.mask_+3A_molecule">molecule</code></td>
<td>
<p>Input molecule to be processed.</p>
</td></tr>
<tr><td><code id="rs.mask_+3A_format">format</code></td>
<td>
<p>specifies format of input reaction/molecule. It can be one of following for a reaction: 'RSMI' or 'RXN'; for a molecule: 'SMILES' or 'MOL'.</p>
</td></tr>
<tr><td><code id="rs.mask_+3A_standardize">standardize</code></td>
<td>
<p>suppresses all explicit hydrogen if set as <code>TRUE</code> (default).</p>
</td></tr>
<tr><td><code id="rs.mask_+3A_explicith">explicitH</code></td>
<td>
<p>converts all implicit hydrogen to explicit if set as <code>TRUE</code>. It is set as <code>FALSE</code> by default.</p>
</td></tr>
<tr><td><code id="rs.mask_+3A_recursive">recursive</code></td>
<td>
<p>if TRUE, all the occurrences of input sub-structure are replaced.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The sub-structure is searched in input reaction/molecule and replaced with the mask. All the bonds between identified sub-structure and the remaining atoms are mapped to the mask. If mask contains more than one atom, all the bonds are connected to the last atom in mask. By default, the first identified sub-structure is replaced. To replace all occurrences, recursive should be set to <code>TRUE</code>. Valence is not checked for the mask atom and the final structure.
</p>


<h3>Value</h3>

<p>Returns SMILES with mask.
</p>


<h3>Note</h3>

<p>Aromatic form of SMILES of the query sub-structure should be used for masking aromatic structures. Automatic aromaticity perception is not done on query structures.
</p>


<h3>Author(s)</h3>

<p>Varun Giri <a href="mailto:varungiri@gmail.com">varungiri@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rs.makeDB">rs.makeDB</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ms.mask('OP(=O)O', '[Cs]', 'O=P(O)(O)OP(=O)(O)OP(=O)(O)OCC3OC(n2cnc1c(ncnc12)N)C(O)C3O')
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
