<!DOCTYPE html><html lang="en"><head><title>Help for package rxSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {rxSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data.A'><p>Sample data example for autosomal genes</p></a></li>
<li><a href='#data.X'><p>Sample data example for X chromosome genes</p></a></li>
<li><a href='#get.tausB'><p>Produce <em>Xce</em> estimates for mice with allele specific reads</p></a></li>
<li><a href='#nLogLik'><p>Negative log likelihood for coefficients provided in results of the fit using process function</p></a></li>
<li><a href='#process'><p>Optimization wrapper, maximizing either the joint model of total (TReC) and allele specific (ASE) counts or just TReC</p></a></li>
<li><a href='#rcA'><p>Reformatted data for autosomal set to be used as input to process function</p></a></li>
<li><a href='#rcX'><p>Reformatted data for chromosome X set to be used as input to process function</p></a></li>
<li><a href='#readCounts'><p>A list object that should be used as input to optimization process function.</p></a></li>
<li><a href='#simRX'><p>Produce simulated counts</p></a></li>
<li><a href='#trecase.A.out'><p>Example of results produced by optimizing step using process function on autosomal genes. Structured as a list.</p></a></li>
<li><a href='#trecase.X.out'><p>Example of results produced by optimizing step using process function on X chromosome genes. Structured as a list.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Combined Total and Allele Specific Reads Sequencing Study</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-11-30</td>
</tr>
<tr>
<td>Author:</td>
<td>Vasyl Zhabotynsky [aut, cre],
  Wei Sun [aut],
  Fei Zou [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Vasyl Zhabotynsky &lt;vasyl@unc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of combined total and allele specific reads from the reciprocal cross study with RNA-seq data.  </td>
</tr>
<tr>
<td>Depends:</td>
<td>MASS,VGAM,numDeriv</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-12-01 09:33:06 UTC; zhabotyn</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-12-01 12:45:14</td>
</tr>
</table>
<hr>
<h2 id='data.A'>Sample data example for autosomal genes</h2><span id='topic+data.A'></span>

<h3>Description</h3>

<p>This data set provides with example of experimental data for a subset of autosomal genes.
The full model requires a combination of total read counts (<b>y</b>) - all the reads belonging for a gene, 
and finding out which of these reads we can specifically attribute to allele A or allele B - allele specific counts (<b>n</b>),
separately the reads attributed specifically to allele B (<b>n0B</b>).
Also, it includes the other data pieces to fit the model: <b>kappas</b> - total number of counts for each mouse, on log scale, 
index - specifying which cross each mouse belongs to, and geneids - Ensembl ids of genes.
They, as well as the datasets simulated with <b>simRX</b> can be fitted using <b>proc.trecase.A</b> or <b>proc.trec.A</b>.
</p>


<h3>Value</h3>

  
<table role = "presentation">
<tr><td><code>index</code></td>
<td>
<p>vector defining the cross of the mouse, female - AB=1, BA=2, AA=3, BB=4, and male - AB=5, BA=6, AA=7, BB=8.
If mice are of only one sex, AB=1, BA=2, AA=3, BB=4.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>matrix of TReC counts. Note, the expected input assumes that inbred mice will be in the last columns of the table, after the last F1 mouse.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>matrix of ASE counts for corresponding F1 mouse (classes 1, 2, 5, 6) for corresponding genes.</p>
</td></tr>  
<tr><td><code>n0B</code></td>
<td>
<p>matrix of ASE counts belonging for allele B, for correponding genes and mice as in <b>n</b>.</p>
</td></tr>  
<tr><td><code>kappas</code></td>
<td>
<p>A parameter, specifying log(overall TReC) for each mouse.</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>ids of genes, if not provided, rownames of the matrix <b>y</b> will be used</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>,<code><a href="#topic+readCounts">readCounts</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'># see total read counts (TReC) for first 2  autosomal genes of a data example:
data.A$y[1:2,]
</code></pre>

<hr>
<h2 id='data.X'>Sample data example for X chromosome genes</h2><span id='topic+data.X'></span>

<h3>Description</h3>

<p>This data set provides with example of experimental data for a subset of autosomal genes.
The full model requires a combination of total read counts (<b>y</b>) - all the reads belonging for a gene, 
and finding out which of these reads we can specifically attribute to allele A or allele B - allele specific counts (<b>n</b>),
separately the reads attributed specifically to allele B (<b>n0B</b>).
Also, <b>tausB</b> - is the <em>Xce</em> effect for each F1 mouse, which specifies the proportion of allele specific reads belonging to allele B.
Also, it includes the other data pieces to fit the model: <b>kappas</b> - total number of counts for each mouse, on log scale, 
index - specifying which cross each mouse belongs to, and geneids - Ensembl ids of genes.
They, as well as the datasets simulated with <b>simRX</b> can be fitted using <b>proc.trecase.X</b> or <b>proc.trec.X</b>.  
</p>


<h3>Value</h3>

  
<table role = "presentation">
<tr><td><code>index</code></td>
<td>
<p>vector defining the cross of the mouse, female - AB=1, BA=2, AA=3, BB=4, and male - AB=5, BA=6, AA=7, BB=8.
If mice are of only one sex, AB=1, BA=2, AA=3, BB=4.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>matrix of TReC counts. Note, the expected input assumes that inbred mice will be in the last columns of the table, after the last F1 mouse.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>matrix of ASE counts for corresponding F1 mouse (classes 1,2,5,6) for corresponding genes.</p>
</td></tr>  
<tr><td><code>n0B</code></td>
<td>
<p>matrix of ASE counts belonging for allele B, for correponding genes and mice as in <b>n</b>.</p>
</td></tr>  
<tr><td><code>kappas</code></td>
<td>
<p>A parameter, specifying as overall TReC for the mouse, on log scale</p>
</td></tr>
<tr><td><code>tausB</code></td>
<td>
<p><em>Xce</em> effect: expression of allele B relative to the overall allele specific count for each mouse. Use some allele specific counts to estimate the effect.</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>ids of genes, if not provided, rownames of the matrix <b>y</b> will be used</p>
</td></tr>
<tr><td><code>index</code></td>
<td>
<p>index defining which mouse belongs to which cross</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>modified total read counts</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>modified allele specific counts</p>
</td></tr>
<tr><td><code>n0B</code></td>
<td>
<p>modified allele specific coutns, belonging to allele B</p>
</td></tr>
<tr><td><code>kappas</code></td>
<td>
<p>offset, defining a library size for each mouse</p>
</td></tr>
<tr><td><code>tausB</code></td>
<td>
<p><em>Xce</em> effect for each mouse, for a given cross</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>Ensembl gene ids</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>,<code><a href="#topic+readCounts">readCounts</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'># see total read counts (TReC) for first 2 X chromosome genes of a data example:
data.X$y[1:2,]
</code></pre>

<hr>
<h2 id='get.tausB'>Produce <em>Xce</em> estimates for mice with allele specific reads</h2><span id='topic+get.tausB'></span>

<h3>Description</h3>

<p><em>Xce</em> estimation for mice with allele specific reads.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.tausB(n, n0B, geneids, min.cnt=50, exclude.prop=.05, Xist.ID="ENSMUSG00000086503")</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.tausB_+3A_n">n</code></td>
<td>
<p>vector of allele specific counts for each mouse</p>
</td></tr>  
<tr><td><code id="get.tausB_+3A_n0b">n0B</code></td>
<td>
<p>vector of allele specific counts for allele B</p>
</td></tr>
<tr><td><code id="get.tausB_+3A_geneids">geneids</code></td>
<td>
<p>gene IDs</p>
</td></tr>
<tr><td><code id="get.tausB_+3A_min.cnt">min.cnt</code></td>
<td>
<p>minimum number of allele specific counts</p>
</td></tr>
<tr><td><code id="get.tausB_+3A_exclude.prop">exclude.prop</code></td>
<td>
<p>minimum proportion of allele specific counts for each allele</p>
</td></tr>
<tr><td><code id="get.tausB_+3A_xist.id">Xist.ID</code></td>
<td>
<p>and ID of <em>Xist</em>, to exclude it from estimating <em>Xce</em>, since <em>Xce</em> would 1-tausB</p>
</td></tr>
</table>


<h3>Value</h3>

<p>output - matrix of 4 rows: 
</p>
<table role = "presentation">
<tr><td><code>med.tauB</code></td>
<td>
<p>taus estimated via median</p>
</td></tr>
<tr><td><code>ave.tauB</code></td>
<td>
<p>taus estimated via percent of allele B counts</p>
</td></tr>
<tr><td><code>all.genes</code></td>
<td>
<p>number of genes that had passed minimum count</p>
</td></tr>
<tr><td><code>used.genes</code></td>
<td>
<p>number of genes that had required percent of each allele</p>
</td></tr>
</table>
<p>each column represent respective mouse.
</p>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>,<code><a href="#topic+nLogLik">nLogLik</a></code>, <code><a href="#topic+data.X">data.X</a></code>, <code><a href="#topic+rcX">rcX</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'># Estimating XCE effect for each mouse for X chromosome
get.tausB(n=data.X$n, n0B=data.X$n0B, geneids=data.X$geneids)
</code></pre>

<hr>
<h2 id='nLogLik'>Negative log likelihood for coefficients provided in results of the fit using process function</h2><span id='topic+nLogLik'></span>

<h3>Description</h3>

<p>Calculates negative log(likelihood) of an X chromosome joint TReC and ASE counts model at a given set of parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nLogLik(res, rc, genei, hessian=FALSE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nLogLik_+3A_res">res</code></td>
<td>
<p>result object from process function</p>
</td></tr>
<tr><td><code id="nLogLik_+3A_rc">rc</code></td>
<td>
<p>Read count data object created by readCounts function</p>
</td></tr>
<tr><td><code id="nLogLik_+3A_genei">genei</code></td>
<td>
<p>get results for i'th gene</p>
</td></tr>
<tr><td><code id="nLogLik_+3A_hessian">hessian</code></td>
<td>
<p>a logical option whether to calculate a Hessian matrix, the default values is set to FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>output - list(nll=-log.likelihood,hessian=hessian matrix)
</p>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>, <code><a href="#topic+rcA">rcA</a></code>, <code><a href="#topic+readCounts">readCounts</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# get negative-log likelihood at the given point:
nLogLik(res=trecase.X.out, rc=rcX, genei=1, hessian=TRUE)

## End(Not run)</code></pre>

<hr>
<h2 id='process'>Optimization wrapper, maximizing either the joint model of total (TReC) and allele specific (ASE) counts or just TReC</h2><span id='topic+process'></span>

<h3>Description</h3>

<p>Performs optimization of one of four combinations: joint TReC and ASE or just TReC for autosome or X chromosome and tests with lrt test several hypotheses:
additive, parent of origin, dominance, consistency of TreC and ASE additive effect, ASE only additive effect, 
sex, sex specific additive, dominance deviation for males.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process(rc, hessian=FALSE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="process_+3A_rc">rc</code></td>
<td>
<p>an object of class readCounts.</p>
</td></tr>
<tr><td><code id="process_+3A_hessian">hessian</code></td>
<td>
<p>a flag whether Hessian matrix for these genes should be calculated, by default set to FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of following matrices (if there is only one sex, only the relevant tests and matrices are outputed) :
</p>
<table role = "presentation">
<tr><td><code>pvals</code></td>
<td>
<p>matrix of p-values from description for each gene corresponding row</p>
</td></tr>
<tr><td><code>coef.full</code></td>
<td>
<p>matrix of full model fit coefficients, -log(likelihood at these coefficients),phi, theta (2 overdispersion parameters used)</p>
</td></tr>  
<tr><td><code>coef.add</code></td>
<td>
<p>matrix of additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.poo</code></td>
<td>
<p>matrix of parent of origin restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.dom</code></td>
<td>
<p>matrix of dominance restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.same</code></td>
<td>
<p>matrix of TReC=ASE additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.ase.add</code></td>
<td>
<p>matrix of ASE additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.sex</code></td>
<td>
<p>matrix of sex restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.sex.add</code></td>
<td>
<p>matrix of sex specific additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.dev.dom</code></td>
<td>
<p>matrix of dominance deviation for male restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>errorlist</code></td>
<td>
<p>a list of errors</p>
</td></tr>
<tr><td><code>hess.lst</code></td>
<td>
<p>a list of heassian matrices, if parameter <b>hessian</b> is set to TRUE</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+get.tausB">get.tausB</a></code>,<code><a href="#topic+nLogLik">nLogLik</a></code>, <code><a href="#topic+data.X">data.X</a></code>, <code><a href="#topic+data.A">data.A</a></code>, <code><a href="#topic+rcA">rcA</a></code>, <code><a href="#topic+readCounts">readCounts</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# fitting X chromosome data example, for a full model, i.e. assuming we have allele specific reads:
trecase.A.out = process(rc=rcA)
names(trecase.A.out)
trecase.A.out$pval

#alternatively for X chromosome:
trecase.X.out = process(rc=rcX)
names(trecase.X.out)
trecase.X.out$pval

## End(Not run)</code></pre>

<hr>
<h2 id='rcA'>Reformatted data for autosomal set to be used as input to process function</h2><span id='topic+rcA'></span>

<h3>Description</h3>

<p>This is an object of type readCounts provides with example of experimental data for a subset of autosomal genes.
The full model requires a combination of total read counts (<b>y</b>) - all the reads belonging for a gene, 
and finding out which of these reads we can specifically attribute to allele A or allele B - allele specific counts (<b>n</b>),
separately the reads attributed specifically to allele B (<b>n0B</b>).
In autosomes <em>Xce</em> effect is absent, so it would be set to NULL for this dataset.
Also, it includes the other data pieces to fit the model: <b>kappas</b> - total number of counts for each mouse, on log scale, 
index - specifying which cross each mouse belongs to, and geneids - Ensembl ids of genes.
Such values also can be simulated with <b>simRX</b> can be fitted using <b>process</b> with appropriate options chrom=&quot;X&quot; and field model to be either &quot;full&quot; or &quot;short&quot;.  
</p>


<h3>Value</h3>

  
<table role = "presentation">
<tr><td><code>index</code></td>
<td>
<p>vector defining the cross of the mouse, female - AB=1, BA=2, AA=3, BB=4, and male - AB=5, BA=6, AA=7, BB=8.
If mice are of only one sex, AB=1, BA=2, AA=3, BB=4.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>matrix of TReC counts. Note, the expected input assumes that inbred mice will be in the last columns of the table, after the last F1 mouse.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>matrix of ASE counts for corresponding F1 mouse (classes 1,2,5,6) for corresponding genes.</p>
</td></tr>  
<tr><td><code>n0B</code></td>
<td>
<p>matrix of ASE counts belonging for allele B, for correponding genes and mice as in <b>n</b>.</p>
</td></tr>  
<tr><td><code>kappas</code></td>
<td>
<p>A parameter, specifying as overall TReC for the mouse, on log scale</p>
</td></tr>
<tr><td><code>tausB</code></td>
<td>
<p><em>Xce</em> effect: expression of allele B relative to the overall allele specific count for each mouse. Set to NULL in autosomes.</p>
</td></tr>
<tr><td><code>gene.switch</code></td>
<td>
<p>For which genes <em>Xce</em> effect should be switched. Null for autosomes.</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>ids of genes, if not provided, rownames of the matrix <b>y</b> will be used</p>
</td></tr>
<tr><td><code>chrom</code></td>
<td>
<p>this field would be set to be &quot;X&quot; since this is dataset for chromosome X</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>set to be &quot;full&quot;, can be modified to &quot;short&quot; to run a TReC oply model</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>Ensembl gene ids</p>
</td></tr>  
<tr><td><code>tech.ctrl</code></td>
<td>
<p>a list of overdispersion boundaries and log(2)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>,<code><a href="#topic+readCounts">readCounts</a></code>, <code><a href="#topic+simRX">simRX</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'># see total read counts (TReC) for first 2 X chromosome genes of a data example:
rcX
</code></pre>

<hr>
<h2 id='rcX'>Reformatted data for chromosome X set to be used as input to process function</h2><span id='topic+rcX'></span>

<h3>Description</h3>

<p>This is an object of type readCounts provides with example of experimental data for a subset of X chromosome genes.
The full model requires a combination of total read counts (<b>y</b>) - all the reads belonging for a gene, 
and finding out which of these reads we can specifically attribute to allele A or allele B - allele specific counts (<b>n</b>),
separately the reads attributed specifically to allele B (<b>n0B</b>).
Also, <b>tausB</b> - is the <em>Xce</em> effect for each F1 mouse, which specifies the proportion of allele specific reads belonging to allele B.
Also, it includes the other data pieces to fit the model: <b>kappas</b> - total number of counts for each mouse, on log scale, 
index - specifying which cross each mouse belongs to, and geneids - Ensembl ids of genes.
They, as well as the datasets simulated with <b>simRX</b> can be fitted using <b>process</b> with appropriate options chrom=&quot;X&quot; and field model to be either &quot;full&quot; or &quot;short&quot;.  
</p>


<h3>Value</h3>

  
<p>genes.switch=genes.switch,
geneids=geneids,
</p>
<table role = "presentation">
<tr><td><code>index</code></td>
<td>
<p>vector defining the cross of the mouse, female - AB=1, BA=2, AA=3, BB=4, and male - AB=5, BA=6, AA=7, BB=8.
If mice are of only one sex, AB=1, BA=2, AA=3, BB=4.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>matrix of TReC counts. Note, the expected input assumes that inbred mice will be in the last columns of the table, after the last F1 mouse.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>matrix of ASE counts for corresponding F1 mouse (classes 1,2,5,6) for corresponding genes.</p>
</td></tr>  
<tr><td><code>n0B</code></td>
<td>
<p>matrix of ASE counts belonging for allele B, for correponding genes and mice as in <b>n</b>.</p>
</td></tr>  
<tr><td><code>kappas</code></td>
<td>
<p>A parameter, specifying as overall TReC for the mouse, on log scale</p>
</td></tr>
<tr><td><code>tausB</code></td>
<td>
<p><em>Xce</em> effect: expression of allele B relative to the overall allele specific count for each mouse. Use some allele specific counts to estimate the effect.</p>
</td></tr>
<tr><td><code>gene.switch</code></td>
<td>
<p>For which genes <em>Xce</em> effect should be switched. Xist gene set to be switched in this set.</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>ids of genes, if not provided, rownames of the matrix <b>y</b> will be used</p>
</td></tr>
<tr><td><code>chrom</code></td>
<td>
<p>this field would be set to be &quot;X&quot; since this is dataset for chromosome X</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>set to be &quot;full&quot;, can be modified to &quot;short&quot; to run a TReC oply model</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>Ensembl gene ids</p>
</td></tr>  
<tr><td><code>tech.ctrl</code></td>
<td>
<p>a list of overdispersion boundaries and log(2)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>,<code><a href="#topic+readCounts">readCounts</a></code>,<code><a href="#topic+simRX">simRX</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'># see total read counts (TReC) for first 2 X chromosome genes of a data example:
rcX
</code></pre>

<hr>
<h2 id='readCounts'>A list object that should be used as input to optimization process function.</h2><span id='topic+readCounts'></span>

<h3>Description</h3>

<p>It should contain at least total read counts (TReC) and classification of crosses 1 to 8. To fit the full model should also have appropriate allele specific counts n and n0B. Also is used along with results of optimization as input to nLogLik function if one needs to calculate Hessian matrix.
</p>


<h3>Value</h3>

  
<table role = "presentation">
<tr><td><code>index</code></td>
<td>
<p>vector defining the cross of the mouse, female - AB=1, BA=2, AA=3, BB=4, and male - AB=5, BA=6, AA=7, BB=8.
If mice are of only one sex, AB=1, BA=2, AA=3, BB=4.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>matrix of TReC counts. Note, the expected input assumes that inbred mice will be in the last columns of the table, after the last F1 mouse.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>matrix of ASE counts for corresponding F1 mouse (classes 1,2,5,6) for corresponding genes.</p>
</td></tr>  
<tr><td><code>n0B</code></td>
<td>
<p>matrix of ASE counts belonging for allele B, for correponding genes and mice as in <b>n</b>.</p>
</td></tr>  
<tr><td><code>kappas</code></td>
<td>
<p>A parameter, specifying as overall TReC for the mouse, on log scale</p>
</td></tr>
<tr><td><code>tausB</code></td>
<td>
<p><em>Xce</em> effect: expression of allele B relative to the overall allele specific count for each mouse. Set to NULL in autosomes.</p>
</td></tr>
<tr><td><code>gene.switch</code></td>
<td>
<p>For which genes <em>Xce</em> effect should be switched. Null for autosomes.</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>ids of genes, if not provided, rownames of the matrix <b>y</b> will be used</p>
</td></tr>
<tr><td><code>chrom</code></td>
<td>
<p>this field would be set to be &quot;X&quot; since this is dataset for chromosome X</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>set to be &quot;full&quot;, can be modified to &quot;short&quot; to run a TReC oply model</p>
</td></tr>
<tr><td><code>geneids</code></td>
<td>
<p>Ensembl gene ids</p>
</td></tr>  
<tr><td><code>tech.ctrl</code></td>
<td>
<p>a list of overdispersion boundaries and log(2)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>,<code><a href="#topic+nLogLik">nLogLik</a></code>, <code><a href="#topic+simRX">simRX</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'># see total read counts (TReC) for first 2 X chromosome genes of a data example:
rcA = readCounts(index=data.A$index, y=data.A$y[1:2,], n=data.A$n[1:2,], n0B=data.A$n0B[1:2,], 
                 kappas=data.A$kappas, geneids=data.A$geneids[1:2])
</code></pre>

<hr>
<h2 id='simRX'>Produce simulated counts</h2><span id='topic+simRX'></span>

<h3>Description</h3>

<p>This function is producing simulated counts for the joint model with Negative-Binomial distribution for TReC and Beta-Binomial for ASE counts. The simulated dataset should be reformatted to readCounts format to be used for optimization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simRX(b0f, b0m, b1f, b1m, beta_sex, beta_dom, beta_k=1, phi=1, theta=1, n=6, 
      mean.base.cnt=50, range.base.cnt=60, perc.ase=.35, n.simu=1E4, 
      is.X=FALSE, tauB=NULL, seed=NULL)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simRX_+3A_b0f">b0f</code></td>
<td>
<p>a female additive strain effect</p>
</td></tr>  
<tr><td><code id="simRX_+3A_b0m">b0m</code></td>
<td>
<p>a male additive strain effect</p>
</td></tr>  
<tr><td><code id="simRX_+3A_b1f">b1f</code></td>
<td>
<p>a female parent of origin effect</p>
</td></tr>  
<tr><td><code id="simRX_+3A_b1m">b1m</code></td>
<td>
<p>a male parent of origin effect</p>
</td></tr>  
<tr><td><code id="simRX_+3A_beta_sex">beta_sex</code></td>
<td>
<p>a sex effect</p>
</td></tr>
<tr><td><code id="simRX_+3A_beta_dom">beta_dom</code></td>
<td>
<p>a dominance effect</p>
</td></tr>
<tr><td><code id="simRX_+3A_beta_k">beta_k</code></td>
<td>
<p>an effect associated with the library size kappas</p>
</td></tr>
<tr><td><code id="simRX_+3A_phi">phi</code></td>
<td>
<p>a Negative-Binomial overdispersion, default value is 1</p>
</td></tr>
<tr><td><code id="simRX_+3A_theta">theta</code></td>
<td>
<p>a Beta-Binomial overdispersion, default value is 1</p>
</td></tr>
<tr><td><code id="simRX_+3A_n">n</code></td>
<td>
<p>a vector defining number of mice in each cross, default value is 6</p>
</td></tr>
<tr><td><code id="simRX_+3A_mean.base.cnt">mean.base.cnt</code></td>
<td>
<p>a target expected number of counts for the base group (with no effects), default value is 50</p>
</td></tr>
<tr><td><code id="simRX_+3A_range.base.cnt">range.base.cnt</code></td>
<td>
<p>a range in which the expected number of counts  for the base group will vary, default value is 60</p>
</td></tr>
<tr><td><code id="simRX_+3A_perc.ase">perc.ase</code></td>
<td>
<p>a percent reads that are allele-specific, default value is 35%</p>
</td></tr>
<tr><td><code id="simRX_+3A_n.simu">n.simu</code></td>
<td>
<p>a number of simulations, default value is 1E4</p>
</td></tr>
<tr><td><code id="simRX_+3A_is.x">is.X</code></td>
<td>
<p>a flag if the value to be simulated is X for chromosome (otherwise autosome), default value is FALSE</p>
</td></tr>
<tr><td><code id="simRX_+3A_taub">tauB</code></td>
<td>
<p>a value describing allelic imbalance - <em>Xce</em> effect for the cross, default value is NULL, in which case 50% will be simulated</p>
</td></tr>
<tr><td><code id="simRX_+3A_seed">seed</code></td>
<td>
<p>a random seed to be set, no set by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>output - 3 matrices with one row - one gene, one column - one mouse: 
</p>
<table role = "presentation">
<tr><td><code>index</code></td>
<td>
<p>vector defining the cross of the mouse, female - AB=1, BA=2, AA=3, BB=4, and male - AB=5, BA=6, AA=7, BB=8.
If mice are of only one sex, AB=1, BA=2, AA=3, BB=4.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>A matrix of total read counts</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>A matrix of allele specific counts</p>
</td></tr>
<tr><td><code>n0B</code></td>
<td>
<p>A matrix of allele specific counts associated with allele B</p>
</td></tr>
<tr><td><code>kappas</code></td>
<td>
<p>Offset parameter, given as overall TReC for the mouse.</p>
</td></tr>
<tr><td><code>tausB</code></td>
<td>
<p>In case of the simulating X chromosome the provided <em>Xce</em> effect is returned: expression of allele B relative to the overall allele specific count for each mouse. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>,<code><a href="#topic+readCounts">readCounts</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulating autosomal data:
dat.A = simRX(b0f=.5, b0m=.6, b1f=.3, b1m=.4, beta_sex=.1, beta_dom=.1, n.simu=1E1)
names(dat.A)
# simulating autosomal data:
dat.X = simRX(b0f=.5, b0m=.6, b1f=.3, b1m=.4, beta_sex=.1, beta_dom=.1, n.simu=1E1, 
              is.X=TRUE, tauB=.3)
names(dat.X)
</code></pre>

<hr>
<h2 id='trecase.A.out'>Example of results produced by optimizing step using process function on autosomal genes. Structured as a list.</h2><span id='topic+trecase.A.out'></span>

<h3>Description</h3>

<p>A list containing test results as well as parameter estimates for joint model evaluated by process function for autosomal genes.
</p>


<h3>Value</h3>

<p>a list of following matrices (if there is only one sex, only the relevant tests and matrices are outputed) :
</p>
<table role = "presentation">
<tr><td><code>pvals</code></td>
<td>
<p>matrix of p-values from description for each gene corresponding row</p>
</td></tr>
<tr><td><code>coef.full</code></td>
<td>
<p>matrix of full model fit coefficients, -log(likelihood at these coefficients),phi, theta (2 overdispersion parameters used)</p>
</td></tr>  
<tr><td><code>coef.add</code></td>
<td>
<p>matrix of additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.poo</code></td>
<td>
<p>matrix of parent of origin restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.dom</code></td>
<td>
<p>matrix of dominance restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.same</code></td>
<td>
<p>matrix of TReC=ASE additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.ase.add</code></td>
<td>
<p>matrix of ASE additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.sex</code></td>
<td>
<p>matrix of sex restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.sex.add</code></td>
<td>
<p>matrix of sex specific additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.dev.dom</code></td>
<td>
<p>matrix of dominance deviation for male restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>errorlist</code></td>
<td>
<p>a list of errors</p>
</td></tr>
<tr><td><code>hess.lst</code></td>
<td>
<p>a list of heassian matrices, if parameter <b>hessian</b> is set to TRUE</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>, <code><a href="#topic+rcA">rcA</a></code>, <code><a href="#topic+data.A">data.A</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>names(trecase.A.out)
</code></pre>

<hr>
<h2 id='trecase.X.out'>Example of results produced by optimizing step using process function on X chromosome genes. Structured as a list.</h2><span id='topic+trecase.X.out'></span>

<h3>Description</h3>

<p>A list containing test results as well as parameter estimates for joint model evaluated by process function for autosomal genes.
</p>


<h3>Value</h3>

<p>a list of following matrices (if there is only one sex, only the relevant tests and matrices are outputed) :
</p>
<table role = "presentation">
<tr><td><code>pvals</code></td>
<td>
<p>matrix of p-values from description for each gene corresponding row</p>
</td></tr>
<tr><td><code>coef.full</code></td>
<td>
<p>matrix of full model fit coefficients, -log(likelihood at these coefficients),phi, theta (2 overdispersion parameters used)</p>
</td></tr>  
<tr><td><code>coef.add</code></td>
<td>
<p>matrix of additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.poo</code></td>
<td>
<p>matrix of parent of origin restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.dom</code></td>
<td>
<p>matrix of dominance restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.same</code></td>
<td>
<p>matrix of TReC=ASE additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.ase.add</code></td>
<td>
<p>matrix of ASE additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.sex</code></td>
<td>
<p>matrix of sex restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.sex.add</code></td>
<td>
<p>matrix of sex specific additive restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>coef.dev.dom</code></td>
<td>
<p>matrix of dominance deviation for male restricted fit coefficients, -log(likelihood at these coefficients),phi, theta</p>
</td></tr>
<tr><td><code>errorlist</code></td>
<td>
<p>a list of errors</p>
</td></tr>
<tr><td><code>hess.lst</code></td>
<td>
<p>a list of heassian matrices, if parameter <b>hessian</b> is set to TRUE</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p> Vasyl Zhabotynsky vasyl@unc.edu </p>


<h3>See Also</h3>

<p><code><a href="#topic+process">process</a></code>, <code><a href="#topic+rcX">rcX</a></code>, <code><a href="#topic+data.X">data.X</a></code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>names(trecase.X.out)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
