<!DOCTYPE html><html><head><title>Help for package s4vd</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {s4vd}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BCheatmap'>
<p>Overlap Heatmap for the visualization of overlapping biclusters</p></a></li>
<li><a href='#BCs4vd'>
<p>Robust biclustering by sparse singular value decomposition incorporating stability selection</p></a></li>
<li><a href='#BCssvd'>
<p>Biclustering via sparse singular value decomposition</p></a></li>
<li><a href='#jaccardmat'>
<p>jaccard matrix</p></a></li>
<li><a href='#lung200'><p>lung</p></a></li>
<li><a href='#stabpath'>
<p>Stability paths plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.1-1</td>
</tr>
<tr>
<td>Title:</td>
<td>Biclustering via Sparse Singular Value Decomposition
Incorporating Stability Selection</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-11-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Martin Sill, Sebastian Kaiser</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Martin Sill &lt;m.sill@dkfz.de&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>biclust,methods,irlba,foreach</td>
</tr>
<tr>
<td>Description:</td>
<td>The main function s4vd() performs a biclustering via sparse
        singular value decomposition with a nested stability selection.
        The results is an biclust object and thus all methods of the
        biclust package can be applied.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Repository/R-Forge/Project:</td>
<td>s4vd</td>
</tr>
<tr>
<td>Repository/R-Forge/Revision:</td>
<td>109</td>
</tr>
<tr>
<td>Repository/R-Forge/DateTimeStamp:</td>
<td>2015-11-18 10:48:14</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-11-19 17:15:53</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-11-18 11:10:38 UTC; rforge</td>
</tr>
</table>
<hr>
<h2 id='BCheatmap'>
Overlap Heatmap for the visualization of overlapping biclusters
</h2><span id='topic+BCheatmap'></span><span id='topic+OverlapHeat'></span>

<h3>Description</h3>

<p> Heatmap function to plot biclustering results. Overlapping biclusters are indicated by colored rectangles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BCheatmap(X, res, cexR = 1.5, cexC = 1.25, axisR = FALSE, axisC= TRUE,
heatcols = maPalette(low="blue",mid="white",high="red", k=50),
clustercols = c(1:5), allrows = FALSE, allcolumns = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BCheatmap_+3A_x">X</code></td>
<td>

<p>the data matrix  
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_res">res</code></td>
<td>

<p>the biclustering result
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_cexr">cexR</code></td>
<td>

<p>relativ font size of the row labels 
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_cexc">cexC</code></td>
<td>

<p>relativ font size of the column labels
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_axisr">axisR</code></td>
<td>

<p>if TRUE the row labels will be plotted
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_axisc">axisC</code></td>
<td>

<p>if TRUE the column labels will be plotted
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_heatcols">heatcols</code></td>
<td>

<p>a character vector specifing the heatcolors 
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_clustercols">clustercols</code></td>
<td>

<p>a character vector specifing the colors of the rectangles that indicate the rows and columns that belong to a bicluster
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_allrows">allrows</code></td>
<td>

<p>if FALSE only the rows assigned to any bicluster will be plotted
</p>
</td></tr>
<tr><td><code id="BCheatmap_+3A_allcolumns">allcolumns</code></td>
<td>

<p>if FALSE only the columns assigned to any bicluster will be plotted
</p>
</td></tr>


</table>


<h3>Author(s)</h3>

<p>Martin Sill \
<a href="mailto:m.sill@dkfz.de">m.sill@dkfz.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#lung cancer data set   Bhattacharjee et al. 2001
data(lung200)
set.seed(12)
res1 &lt;- biclust(lung200,method=BCs4vd(),pcerv=.5,pceru=0.01,ss.thr=c(0.6,0.65)
,start.iter=3,size=0.632,cols.nc=TRUE,steps=100,pointwise=TRUE
,merr=0.0001,iter=100,nbiclust=10,col.overlap=FALSE)
BCheatmap(lung200,res1)

</code></pre>

<hr>
<h2 id='BCs4vd'>
Robust biclustering by sparse singular value decomposition incorporating stability selection
</h2><span id='topic+BCs4vd'></span><span id='topic+s4vd'></span><span id='topic+BCs4vd-class'></span><span id='topic++5CS4method+7Bbiclust+7D+7Bmatrix+2CBCs4vd+7D'></span>

<h3>Description</h3>

<p>The function performs biclustering of the data matrix by sparse singular value decomposition with nested stability selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'matrix,BCs4vd'
biclust(x, method=BCs4vd(),
		steps = 100,
		pcerv = 0.05,
		pceru = 0.05,
		ss.thr = c(0.6,0.65),
		size = 0.632,
		gamm = 0,
		iter = 100,
		nbiclust = 10,
		merr = 10^(-4),
		cols.nc=FALSE,
		rows.nc=TRUE,
		row.overlap=TRUE,
		col.overlap=TRUE,
		row.min=4,
		col.min=4,
		pointwise=TRUE,
		start.iter=0,
		savepath=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BCs4vd_+3A_x">x</code></td>
<td>

<p>The matrix to be clustered.
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_method">method</code></td>
<td>

<p>calls the BCs4vd() method
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_steps">steps</code></td>
<td>

<p>Number of subsamples used to perform the stability selection.
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_pcerv">pcerv</code></td>
<td>

<p>Per comparsion wise error rate to control the number of falsely selected right singular vector coefficients (columns/samples).
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_pceru">pceru</code></td>
<td>

<p>Per comparsion wise error rate to control the number of falsely selected left singular vector coefficients (rows/genes).
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_ss.thr">ss.thr</code></td>
<td>

<p>Range of the cutoff threshold (relative selection frequency) for the stability selection.
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_size">size</code></td>
<td>

<p>Size of the subsamples used to perform the stability selection.  
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_gamm">gamm</code></td>
<td>

<p>Weight parameter for the adaptive LASSO, nonnegative constant (default = 0, LASSO).
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_iter">iter</code></td>
<td>

<p>Maximal number of iterations to fit a single bicluster.
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_nbiclust">nbiclust</code></td>
<td>

<p>Maximal number of biclusters. 
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_merr">merr</code></td>
<td>

<p>Threshold to decide convergence. 
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_cols.nc">cols.nc</code></td>
<td>

<p>Allow for negative correlation of columns (samples) over rows (genes).
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_rows.nc">rows.nc</code></td>
<td>

<p>Allow for negative correlation of rows (genes) over columns (samples).
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_row.overlap">row.overlap</code></td>
<td>

<p>Allow rows to overlap between biclusters. 
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_col.overlap">col.overlap</code></td>
<td>

<p>Allow columns to overlap between biclusters. 
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_row.min">row.min</code></td>
<td>

<p>Minimal number of rows.
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_col.min">col.min</code></td>
<td>

<p>Minimal number of columns.
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_pointwise">pointwise</code></td>
<td>

<p>If TRUE performs a fast pointwise stability selection instead of calculating the complete stability path.  
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_start.iter">start.iter</code></td>
<td>

<p>Number of starting iterations in which the algorithm is not allowed to converge. 
</p>
</td></tr>
<tr><td><code id="BCs4vd_+3A_savepath">savepath</code></td>
<td>

<p>Saves the stability path in order plot the path with the stabpathplot function.
Note that pointwise needs to be TRUE to save the path. For extreme high dimensional data sets (e.g. the lung cancer example) the resulting
biclust object may exceed the available memory.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>Biclust</code>.
</p>


<h3>Author(s)</h3>

<p>Martin Sill \
<a href="mailto:m.sill@dkfz.de">m.sill@dkfz.de</a>
</p>


<h3>References</h3>

<p>Martin Sill, Sebastian Kaiser, Axel Benner and Annette Kopp-Schneider
&quot;Robust biclustering by sparse singular value decomposition incorporating stability selection&quot;, Bioinformatics, 2011 
</p>


<h3>See Also</h3>

 <p><code><a href="biclust.html#topic+biclust">biclust</a></code>, <code><a href="biclust.html#topic+Biclust">Biclust</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
# example data set according to the simulation study in Lee et al. 2010
# generate artifical data set and a correspondig biclust object
u &lt;- c(10,9,8,7,6,5,4,3,rep(2,17),rep(0,75))
v &lt;- c(10,-10,8,-8,5,-5,rep(3,5),rep(-3,5),rep(0,34))
u &lt;- u/sqrt(sum(u^2)) 
v &lt;- v/sqrt(sum(v^2))
d &lt;- 50
set.seed(1)
X &lt;- (d*u%*%t(v)) + matrix(rnorm(100*50),100,50)
params &lt;- info &lt;- list()
RowxNumber &lt;- matrix(rep(FALSE,100),ncol=1)
NumberxCol &lt;- matrix(rep(FALSE,50),nrow=1)
RowxNumber[u!=0,1] &lt;- TRUE 
NumberxCol[1,v!=0] &lt;- TRUE
Number &lt;- 1
ressim &lt;- BiclustResult(params,RowxNumber,NumberxCol,Number,info)

#perform s4vd biclustering 
ress4vd &lt;- biclust(X,method=BCs4vd,pcerv=0.5,pceru=0.5,pointwise=FALSE,nbiclust=1,savepath=TRUE)
#perform s4vd biclustering with fast pointwise stability selection
ress4vdpw &lt;- biclust(X,method=BCs4vd,pcerv=0.5,pceru=0.5,pointwise=TRUE,nbiclust=1)
#perform ssvd biclustering
resssvd &lt;- biclust(X,BCssvd,K=1)
#agreement of the results with the simulated bicluster
jaccardind(ressim,ress4vd)
jaccardind(ressim,ress4vdpw)
jaccardind(ressim,resssvd)

</code></pre>

<hr>
<h2 id='BCssvd'>
Biclustering via sparse singular value decomposition 
</h2><span id='topic+BCssvd'></span><span id='topic+BCssvd-class'></span><span id='topic+ssvd'></span><span id='topic++5CS4method+7Bbiclust+7D+7Bmatrix+2CBCssvd+7D'></span>

<h3>Description</h3>

<p>The function performs a biclustering of the data matrix by sparse singular value decomposition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'matrix,BCssvd'
biclust(x,method=BCssvd(),
	 K=10,
	 threu = 1,
	 threv = 1,
	 gamu = 0,
	 gamv =0,
	 u0 = svd(X)$u[,1],
	 v0 = svd(X)$v[,1],
	 merr = 10^(-4),
	 niter = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BCssvd_+3A_x">x</code></td>
<td>

<p>the matrix to be clustered
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_method">method</code></td>
<td>

<p>calls the BCssvd() method
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_k">K</code></td>
<td>

<p>number of SSVD-layers 
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_threu">threu</code></td>
<td>

<p>type of penalty (thresholding rule) for the left singular vector,
1 = (Adaptive) LASSO (default)
2 = hard thresholding
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_threv">threv</code></td>
<td>

<p>type of penalty (thresholding rule) for the right singular vector,
1 = (Adaptive) LASSO (default)
2 = hard thresholding	
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_gamu">gamu</code></td>
<td>

<p>weight parameter in Adaptive LASSO for the left singular vector, nonnegative constant (default = 0, LASSO)
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_gamv">gamv</code></td>
<td>

<p>weight parameter in Adaptive LASSO for the right singular vector, nonnegative constant (default = 0, LASSO)
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_u0">u0</code></td>
<td>

<p>initial left singular vector                
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_v0">v0</code></td>
<td>

<p>initial right singular vector   
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_merr">merr</code></td>
<td>

<p>threshold to decide convergence
</p>
</td></tr>
<tr><td><code id="BCssvd_+3A_niter">niter</code></td>
<td>

<p>maximum number of iterations
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an Biclust object.
</p>


<h3>Author(s)</h3>

<p>Adaptation of original code from Mihee Lee by Martin Sill \
<a href="mailto:m.sill@dkfz.de">m.sill@dkfz.de</a>
</p>


<h3>References</h3>

<p>Mihee Lee, Haipeng Shen, Jianhua Z. Huang and J. S. Marron1
&quot;Biclustering via Sparse Singular Value Decomposition&quot;, Biometrics, 2010 
</p>


<h3>See Also</h3>

 <p><code><a href="biclust.html#topic+biclust">biclust</a></code>, <code><a href="biclust.html#topic+Biclust">Biclust</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
# example data set according to the simulation study in Lee et al. 2010
# generate artifical data set and a correspondig biclust object
u &lt;- c(10,9,8,7,6,5,4,3,rep(2,17),rep(0,75))
v &lt;- c(10,-10,8,-8,5,-5,rep(3,5),rep(-3,5),rep(0,34))
u &lt;- u/sqrt(sum(u^2)) 
v &lt;- v/sqrt(sum(v^2))
d &lt;- 50
set.seed(1)
X &lt;- (d*u%*%t(v)) + matrix(rnorm(100*50),100,50)
params &lt;- info &lt;- list()
RowxNumber &lt;- matrix(rep(FALSE,100),ncol=1)
NumberxCol &lt;- matrix(rep(FALSE,50),nrow=1)
RowxNumber[u!=0,1] &lt;- TRUE 
NumberxCol[1,v!=0] &lt;- TRUE
Number &lt;- 1
ressim &lt;- BiclustResult(params,RowxNumber,NumberxCol,Number,info)

#perform s4vd biclustering 
ress4vd &lt;- biclust(X,method=BCs4vd,pcerv=0.5,pceru=0.5,pointwise=FALSE,nbiclust=1,savepath=TRUE)
#perform s4vd biclustering with fast pointwise stability selection
ress4vdpw &lt;- biclust(X,method=BCs4vd,pcerv=0.5,pceru=0.5,pointwise=TRUE,nbiclust=1)
#perform ssvd biclustering
resssvd &lt;- biclust(X,BCssvd,K=1)
#agreement of the results with the simulated bicluster
jaccardind(ressim,ress4vd)
jaccardind(ressim,ress4vdpw)
jaccardind(ressim,resssvd)

</code></pre>

<hr>
<h2 id='jaccardmat'>
jaccard matrix
</h2><span id='topic+jaccardmat'></span>

<h3>Description</h3>

<p>The function calculates the pairwise jaccard coefficients between the biclusters of two biclustering results 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jaccardmat(res1,res2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jaccardmat_+3A_res1">res1</code></td>
<td>

<p>A biclustering result as an object of class Biclust  
</p>
</td></tr>
<tr><td><code id="jaccardmat_+3A_res2">res2</code></td>
<td>

<p>A biclustering result as an object of class Biclust
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The result is matrix of pairwise jaccard coefficents between the biclusters of res1 and res2. 
</p>


<h3>Author(s)</h3>

<p>Martin Sill \
<a href="mailto:m.sill@dkfz.de">m.sill@dkfz.de</a>
</p>


<h3>See Also</h3>

<p>jaccardind
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#lung cancer data set   Bhattacharjee et al. 2001
data(lung200)
set.seed(12)
res1 &lt;- biclust(lung200,method=BCs4vd(),pcerv=.5,pceru=0.01,ss.thr=c(0.6,0.65)
,start.iter=3,size=0.632,cols.nc=TRUE,steps=100,pointwise=TRUE
,merr=0.0001,iter=100,nbiclust=10,col.overlap=FALSE)
res2 &lt;- biclust(lung200,method=BCPlaid())
jaccardmat(res1,res2)

</code></pre>

<hr>
<h2 id='lung200'>lung</h2><span id='topic+lung200'></span><span id='topic+lung+20cancer+20microarray+20data+20set200'></span>

<h3>Description</h3>

<p>Lung cancer gene expression data set</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(lung200)
  </code></pre>


<h3>Format</h3>

<p>This data set contain 56 samples and gene expression values of a subset of 200 genes 
showing the highest variance of the 12 625 genes measured
using the Affymetrix 95av2 GeneChip. The samples comprise 20 pulmonary carcinoid samples (Carcinoid), 13 colon cancer
metastasis samples (Colon), 17 normal lung samples (Normal)
and 6 small cell lung carcinoma samples (SmallCell).
The rownames are affymetrix gene ids. 
</p>


<h3>Source</h3>

<p>http://www.pnas.org/content/98/24/13790/suppl/DC1
</p>


<h3>References</h3>

<p>Bhattacharjee, A., Richards, W. G., Staunton, J., Li, C., Monti, S., Vasa, P., Ladd, C.,&lt;br&gt;
Beheshti, J., Bueno, R., Gillette, M., Loda, M., Weber, G., Mark, E. J., Lander,&lt;br&gt;
E. S., Wong, W., Johnson, B. E., Golub, T. R., Sugarbaker, D. J., and Meyerson,&lt;br&gt;
M. (2001). Classification of human lung carcinomas by mRNA expression profiling&lt;br&gt;
reveals distinct adenocarcinoma subclasses. Proceedings of the National Academy&lt;br&gt;
of Sciences of the United States of America.</p>

<hr>
<h2 id='stabpath'>
Stability paths plot
</h2><span id='topic+stabpath'></span><span id='topic+stabilitypath'></span>

<h3>Description</h3>

<p>The function plots the stability path of a S4VD result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stabpath(res,number)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stabpath_+3A_res">res</code></td>
<td>

<p>the S4VD result  
</p>
</td></tr>
<tr><td><code id="stabpath_+3A_number">number</code></td>
<td>

<p>the bicluster for which the stability path shall be plotted
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the stability path for the rows and the columns regarding the last iteration of the s4vd algorithm.
Note that if the pointwise error control was used or if savepath=FALSE the final selection probabilities for the rows and the columns will be plotted.  
</p>


<h3>Author(s)</h3>

<p>Martin Sill \
<a href="mailto:m.sill@dkfz.de">m.sill@dkfz.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# example data set according to the simulation study in Lee et al. 2010
# generate artifical data set and a correspondig biclust object
u &lt;- c(10,9,8,7,6,5,4,3,rep(2,17),rep(0,75))
v &lt;- c(10,-10,8,-8,5,-5,rep(3,5),rep(-3,5),rep(0,34))
u &lt;- u/sqrt(sum(u^2)) 
v &lt;- v/sqrt(sum(v^2))
d &lt;- 50
set.seed(1)
X &lt;- (d*u%*%t(v)) + matrix(rnorm(100*50),100,50)
params &lt;- info &lt;- list()
RowxNumber &lt;- matrix(rep(FALSE,100),ncol=1)
NumberxCol &lt;- matrix(rep(FALSE,50),nrow=1)
RowxNumber[u!=0,1] &lt;- TRUE 
NumberxCol[1,v!=0] &lt;- TRUE
Number &lt;- 1
ressim &lt;- BiclustResult(params,RowxNumber,NumberxCol,Number,info)
#perform s4vd biclustering 
ress4vd &lt;- biclust(X,method=BCs4vd,pcerv=0.5,
                        pceru=0.5,ss.thr=c(0.6,0.65),steps=500,
                        pointwise=FALSE,nbiclust=1,savepath=TRUE)
#perform s4vd biclustering with fast pointwise stability selection
ress4vdpw &lt;- biclust(X,method=BCs4vd,pcerv=0.5,
                          pceru=0.5,ss.thr=c(0.6,0.65),steps=500,
                          pointwise=TRUE,nbiclust=1)
#stability paths
stabpath(ress4vd,1)
#selection probabilitys for the pointwise stability selection
stabpath(ress4vdpw,1)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
