<!DOCTYPE html><html><head><title>Help for package saasCNV</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {saasCNV}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#saasCNV-package'>
<p>Somatic Copy Number Alteration Analysis Using Sequencing and SNP Array Data</p></a></li>
<li><a href='#cnv.call'>
<p>CNV Calling from Sequencing Data</p></a></li>
<li><a href='#cnv.data'>
<p>Construct Data Frame for CNV Inference with NGS Data</p></a></li>
<li><a href='#diagnosis.cluster.plot'>
<p>Visualize Genome-Wide SCNA Profile in 2D Cluster Plot</p></a></li>
<li><a href='#diagnosis.seg.plot.chr'>
<p>Visualize Segmentation Results for Diagnosis</p></a></li>
<li><a href='#GC.adjust'>
<p>GC Content Adjustment</p></a></li>
<li><a href='#genome.wide.plot'>
<p>Visualize Genome-Wide SCNA Profile</p></a></li>
<li><a href='#internals'>
<p>Internal Functions and Data</p></a></li>
<li><a href='#joint.segmentation'>
<p>Joint Segmentation on log2ratio and log2mBAF Dimensions</p></a></li>
<li><a href='#merging.segments'>
<p>Merge Adjacent Segments</p></a></li>
<li><a href='#NGS.CNV'>
<p>CNV Analysis Pipeline for WGS and WES Data</p></a></li>
<li><a href='#reannotate.CNV.res'>
<p>Gene Annotation</p></a></li>
<li><a href='#SNP.CNV'>
<p>CNV Analysis Pipeline for SNP array Data</p></a></li>
<li><a href='#snp.cnv.data'>
<p>Construct Data Frame for CNV Inference with SNP Array Data</p></a></li>
<li><a href='#snp.refine.boundary'>
<p>Refine Segment Boundaries</p></a></li>
<li><a href='#vcf2txt'>
<p>Covert VCF File to A Data Frame</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-05-10</td>
</tr>
<tr>
<td>Title:</td>
<td>Somatic Copy Number Alteration Analysis Using Sequencing and SNP
Array Data</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhongyang Zhang [aut, cre], 
    Ke Hao [aut], 
    Nancy R. Zhang [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), RANN, DNAcopy</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform joint segmentation on two signal dimensions derived from 
    total read depth (intensity) and allele specific read depth (intensity) for 
    whole genome sequencing (WGS), whole exome sequencing (WES) and SNP array data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://zhangz05.u.hpc.mssm.edu/saasCNV/">https://zhangz05.u.hpc.mssm.edu/saasCNV/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-05-17 14:58:22 UTC; zhangz05</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-05-18 02:04:56</td>
</tr>
</table>
<hr>
<h2 id='saasCNV-package'>
Somatic Copy Number Alteration Analysis Using Sequencing and SNP Array Data
</h2><span id='topic+saasCNV-package'></span><span id='topic+saasCNV'></span>

<h3>Description</h3>

<p>Perform joint segmentation on two signal dimensions derived from 
total read depth (intensity) and allele specific read depth (intensity) for 
whole genome sequencing (WGS), whole exome sequencing (WES) and SNP array data.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> saasCNV</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.3.4</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-05-10</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>See the vignettes of the package for more details.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang [aut, cre], Ke Hao [aut], Nancy R. Zhang [ctb]
</p>
<p>Maintainer: Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Zhang, Z. and Hao, K. (2015) SAAS-CNV: A joint segmentation approach on aggregated and allele specific signals 
for the identification of somatic copy number alterations with next-generation sequencing Data. 
<em>PLoS Computational Biology</em>, <b>11(11)</b>:e1004618.
</p>
<p>Zhang, N. R., Siegmund, D. O., Ji, H., Li, J. Z. (2010) 
Detecting simultaneous changepoints in multiple sequences. <em>Biometrika</em>, <b>97(3)</b>:631&ndash;645.
</p>


<h3>See Also</h3>

<p><code>DNAcopy</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## See the vignettes of the package for examples.
</code></pre>

<hr>
<h2 id='cnv.call'>
CNV Calling from Sequencing Data
</h2><span id='topic+cnv.call'></span>

<h3>Description</h3>

<p>Assign SCNA state to each segment directly from joint segmentation or from 
the results after segments merging step.</p>


<h3>Usage</h3>

<pre><code class='language-R'>cnv.call(data, sample.id, segs.stat, maxL = NULL, N = 1000, 
         pvalue.cutoff = 0.05, seed = NULL,
         do.manual.baseline=FALSE,
         log2mBAF.left=NULL, log2mBAF.right=NULL, 
         log2ratio.bottom=NULL, log2ratio.up=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cnv.call_+3A_data">data</code></td>
<td>
<p>a data frame containing log2ratio and log2mBAF data generated 
by <code><a href="#topic+cnv.data">cnv.data</a></code>.
</p>
</td></tr>
<tr><td><code id="cnv.call_+3A_sample.id">sample.id</code></td>
<td>
<p>sample ID to be displayed.
</p>
</td></tr>
<tr><td><code id="cnv.call_+3A_segs.stat">segs.stat</code></td>
<td>
<p>a data frame containing segment locations and summary statistics 
resulting from <code><a href="#topic+joint.segmentation">joint.segmentation</a></code> or <code><a href="#topic+merging.segments">merging.segments</a></code>.
</p>
</td></tr>
<tr><td><code id="cnv.call_+3A_maxl">maxL</code></td>
<td>
<p>integer. The maximum length in terms of number of probes a bootstrapped segment
may span. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically specified 
as 1/100 of the number of data points.
</p>
</td></tr>
<tr><td><code id="cnv.call_+3A_n">N</code></td>
<td>
<p>the number of replicates drawn by bootstrap.
</p>
</td></tr>
<tr><td><code id="cnv.call_+3A_pvalue.cutoff">pvalue.cutoff</code></td>
<td>
<p>a p-value cut-off for CNV calling.
</p>
</td></tr>
<tr><td><code id="cnv.call_+3A_seed">seed</code></td>
<td>
<p>integer. Random seed can be set for reproducibility of results.
</p>
</td></tr>
<tr><td><code id="cnv.call_+3A_do.manual.baseline">do.manual.baseline</code></td>
<td>
<p>logical. If baseline adjustment to be done manually. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="cnv.call_+3A_log2mbaf.left">log2mBAF.left</code>, <code id="cnv.call_+3A_log2mbaf.right">log2mBAF.right</code>, <code id="cnv.call_+3A_log2ratio.bottom">log2ratio.bottom</code>, <code id="cnv.call_+3A_log2ratio.up">log2ratio.up</code></td>
<td>

<p>left, right, bottom and up boundaries to be specified manually by a visual inspectio of 
2-D diagnosis plot generated by <code><a href="#topic+diagnosis.cluster.plot">diagnosis.cluster.plot</a></code>. 
These parameters are active when <code>do.manual.baseline=TRUE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The baseline adjustment step is incorporated implicitly in the function.
</p>


<h3>Value</h3>

<p>A few more columns have been add to the data frame resulting from 
<code><a href="#topic+joint.segmentation">joint.segmentation</a></code> or <code><a href="#topic+merging.segments">merging.segments</a></code>, 
which summarize the baseline adjusted median log2ratio, log2mBAF, 
p-values and CNV state for each segment.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+joint.segmentation">joint.segmentation</a></code>, <code><a href="#topic+merging.segments">merging.segments</a></code>, <code><a href="#topic+cnv.data">cnv.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(seq.data)
data(seq.segs.merge)

## Not run: 
seq.cnv &lt;- cnv.call(data=seq.data, sample.id="PT116", 
                    segs.stat=seq.segs.merge, maxL=2000, N=1000,
                    pvalue.cutoff=0.05)

## End(Not run)

## how the results look like
data(seq.cnv)
head(seq.cnv)
</code></pre>

<hr>
<h2 id='cnv.data'>
Construct Data Frame for CNV Inference with NGS Data
</h2><span id='topic+cnv.data'></span>

<h3>Description</h3>

<p>Transform read depth information into log2ratio and log2mBAF
that we use for joint segmentation and CNV calling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cnv.data(vcf, min.chr.probe = 100, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cnv.data_+3A_vcf">vcf</code></td>
<td>
<p>a data frame constructed from a vcf file. See <code><a href="#topic+vcf2txt">vcf2txt</a></code>.
</p>
</td></tr>
<tr><td><code id="cnv.data_+3A_min.chr.probe">min.chr.probe</code></td>
<td>
<p>the minimum number of probes tagging a chromosome for it 
to be passed to the subsequent analysis.
</p>
</td></tr>
<tr><td><code id="cnv.data_+3A_verbose">verbose</code></td>
<td>
<p>logical. If more details to be output. Default is <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the log2raio and log2mBAF values for each probe site.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Staaf, J., Vallon-Christersson, J., Lindgren, D., Juliusson, G., Rosenquist, R., 
Hoglund, M., Borg, A., Ringner, M. (2008) Normalization of Illumina Infinium 
whole-genome SNP data improves copy number estimates and allelic intensity ratios. 
<em>BMC bioinformatics</em>, <b>9</b>:409.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vcf2txt">vcf2txt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load a data frame constructed from a vcf file with vcf2txt

## Not run: 
## download vcf_table.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/vcf_table.txt.gz"
tryCatch({download.file(url=url, destfile="vcf_table.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="vcf_table.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

vcf_table &lt;- read.delim(file="vcf_table.txt.gz", as.is=TRUE)
seq.data &lt;- cnv.data(vcf=vcf_table, min.chr.probe=100, verbose=TRUE)

## End(Not run)

## see how seq.data looks like
data(seq.data)
head(seq.data)
</code></pre>

<hr>
<h2 id='diagnosis.cluster.plot'>
Visualize Genome-Wide SCNA Profile in 2D Cluster Plot
</h2><span id='topic+diagnosis.cluster.plot'></span>

<h3>Description</h3>

<p>An optional function to visualize genome-wide SCNA Profile in 
log2mBAF-log2ratio 2D cluster plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagnosis.cluster.plot(segs, chrs, min.snps, max.cex = 3, ref.num.probe = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diagnosis.cluster.plot_+3A_segs">segs</code></td>
<td>
<p>a data frame containing segment location, summary statistics 
and SCNA status resulting from <code><a href="#topic+cnv.call">cnv.call</a></code>.</p>
</td></tr>
<tr><td><code id="diagnosis.cluster.plot_+3A_chrs">chrs</code></td>
<td>
<p>the chromosomes to be visualized. For example, 1:22.
</p>
</td></tr>
<tr><td><code id="diagnosis.cluster.plot_+3A_min.snps">min.snps</code></td>
<td>
<p>the minimum number of probes a segment span.
</p>
</td></tr>
<tr><td><code id="diagnosis.cluster.plot_+3A_max.cex">max.cex</code></td>
<td>
<p>the maximum of cex a circle is associated with. See details.
</p>
</td></tr>
<tr><td><code id="diagnosis.cluster.plot_+3A_ref.num.probe">ref.num.probe</code></td>
<td>
<p>integer. The reference number of probes against which 
a segment is compared in order to determine the cex of the segment 
to be displayed. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically 
specified as 1/100 of the number of data points.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>on the main log2mBAF-log2ratio panel, each circle corresponds to a segment, 
with the size reflecting the length of the segment; the color code is specified 
in legend; the dashed gray lines indicate the adjusted baselines. 
The side panels, corresponding to log2ratio and log2mBAF dimension respectively, 
show the distribution of median values of each segment weighted by its length.
</p>


<h3>Value</h3>

<p>An R plot will be generated.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+joint.segmentation">joint.segmentation</a></code>, <code><a href="#topic+cnv.call">cnv.call</a></code>, 
<code><a href="#topic+diagnosis.seg.plot.chr">diagnosis.seg.plot.chr</a></code>, <code><a href="#topic+genome.wide.plot">genome.wide.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(seq.data)
data(seq.cnv)

diagnosis.cluster.plot(segs=seq.cnv, 
                       chrs=sub("^chr","",unique(seq.cnv$chr)), 
                       min.snps=10, max.cex=3, ref.num.probe=1000)
</code></pre>

<hr>
<h2 id='diagnosis.seg.plot.chr'>
Visualize Segmentation Results for Diagnosis
</h2><span id='topic+diagnosis.seg.plot.chr'></span>

<h3>Description</h3>

<p>The results from joint segmentation and segments merging are visualized 
for the specified choromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagnosis.seg.plot.chr(data, segs, sample.id = "Sample", chr = 1, cex = 0.3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diagnosis.seg.plot.chr_+3A_data">data</code></td>
<td>
<p>a data frame containing log2ratio and log2mBAF data generated 
by <code><a href="#topic+cnv.data">cnv.data</a></code>.
</p>
</td></tr>
<tr><td><code id="diagnosis.seg.plot.chr_+3A_segs">segs</code></td>
<td>
<p>a data frame containing segment locations and summary statistics 
resulting from <code><a href="#topic+joint.segmentation">joint.segmentation</a></code> or <code><a href="#topic+merging.segments">merging.segments</a></code>.
</p>
</td></tr>
<tr><td><code id="diagnosis.seg.plot.chr_+3A_sample.id">sample.id</code></td>
<td>
<p>sample ID to be displayed in the title of the plot.
</p>
</td></tr>
<tr><td><code id="diagnosis.seg.plot.chr_+3A_chr">chr</code></td>
<td>
<p>the chromosome number (e.g. 1) to be visualized.
</p>
</td></tr>
<tr><td><code id="diagnosis.seg.plot.chr_+3A_cex">cex</code></td>
<td>
<p>a numerical value giving the amount by which plotting text and 
symbols should be magnified relative to the default. It can be adjusted 
in order to make the plot legible.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An R plot will be generated.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+joint.segmentation">joint.segmentation</a></code>, <code><a href="#topic+merging.segments">merging.segments</a></code>, <code><a href="#topic+cnv.data">cnv.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## visual diagnosis of joint segmentation results
data(seq.data)
data(seq.segs)
diagnosis.seg.plot.chr(data=seq.data, segs=seq.segs, 
                       sample.id="Joint Segmentation", 
                       chr=1, cex=0.3)

## visual diagnosis of results from merging step
data(seq.segs.merge)
diagnosis.seg.plot.chr(data=seq.data, segs=seq.segs.merge, 
                       sample.id="After Segments Merging Step", 
                       chr=1, cex=0.3)
</code></pre>

<hr>
<h2 id='GC.adjust'>
GC Content Adjustment
</h2><span id='topic+GC.adjust'></span>

<h3>Description</h3>

<p>This function adjusts log2ratio by GC content using LOESS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC.adjust(data, gc, maxNumDataPoints = 10000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GC.adjust_+3A_data">data</code></td>
<td>
<p>A data frame generated by <code><a href="#topic+cnv.data">cnv.data</a></code> 
or <code><a href="#topic+snp.cnv.data">snp.cnv.data</a></code>.
</p>
</td></tr>
<tr><td><code id="GC.adjust_+3A_gc">gc</code></td>
<td>
<p>A data frame containing three columns: <code>chr</code>, <code>position</code> 
and <code>GC</code>. See the example data below for details.
</p>
</td></tr>
<tr><td><code id="GC.adjust_+3A_maxnumdatapoints">maxNumDataPoints</code></td>
<td>
<p>The maximum number of data points used for loess fit.
Default is 10000.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The method for GC content adjustment was adopted from CNAnorm (Gusnato et al. 2012).
</p>


<h3>Value</h3>

<p>A data frame containing the log2ratio (GC adjusted) and log2mBAF values 
for each probe site in the same format as generated by <code><a href="#topic+cnv.data">cnv.data</a></code> 
or <code><a href="#topic+snp.cnv.data">snp.cnv.data</a></code>. The original log2ratio is renamed as 
<code>log2ratio.woGCAdj</code>. The GC-adjusted log2ratio is nameed as <code>log2ratio</code>.
</p>


<h3>Note</h3>

<p>This function is optional in the analysis pipeline and is now in beta version.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Gusnanto, A, Wood HM, Pawitan Y, Rabbitts P, Berri S (2012) Correcting for 
cancer genome size and tumour cell content enables better estimation of 
copy number alterations from next-generation sequence data. 
<em>Bioinformatics</em>, <b>28</b>:40-47.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cnv.data">cnv.data</a></code>, <code><a href="#topic+snp.cnv.data">snp.cnv.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## CNV data generated by cnv.data
data(seq.data)
head(seq.data)

## Not run: 
## an example GC content file
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/GC_1kb_hg19.txt.gz"
tryCatch({download.file(url=url, destfile="GC_1kb_hg19.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="GC_1kb_hg19.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

gc &lt;- read.delim(file = "GC_1kb_hg19.txt.gz", as.is=TRUE)
head(gc)

## GC content adjustment
seq.data &lt;- GC.adjust(data = seq.data, gc = gc, maxNumDataPoints = 10000)
head(seq.data)

## End(Not run)

</code></pre>

<hr>
<h2 id='genome.wide.plot'>
Visualize Genome-Wide SCNA Profile
</h2><span id='topic+genome.wide.plot'></span>

<h3>Description</h3>

<p>An optional function to visualize genome-wide SCNA Profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome.wide.plot(data, segs, sample.id, chrs, cex = 0.3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genome.wide.plot_+3A_data">data</code></td>
<td>
<p>a data frame containing log2ratio and log2mBAF data generated 
by <code><a href="#topic+cnv.data">cnv.data</a></code>.
</p>
</td></tr>
<tr><td><code id="genome.wide.plot_+3A_segs">segs</code></td>
<td>
<p>a data frame containing segment location, summary statistics 
and SCNA status resulting from <code><a href="#topic+cnv.call">cnv.call</a></code>.
</p>
</td></tr>
<tr><td><code id="genome.wide.plot_+3A_sample.id">sample.id</code></td>
<td>
<p>sample ID to be displayed in the title of the plot.
</p>
</td></tr>
<tr><td><code id="genome.wide.plot_+3A_chrs">chrs</code></td>
<td>
<p>the chromosomes to be visualized. For example, 1:22.
</p>
</td></tr>
<tr><td><code id="genome.wide.plot_+3A_cex">cex</code></td>
<td>
<p>a numerical value giving the amount by which plotting text and 
symbols should be magnified relative to the default. It can be adjusted 
in order to make the plot legible.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>On the top panel, the log2ratio signal is plotted against 
chromosomal position and on the panels blow, the log2mBAF, 
tumor mBAF, and normal mBAF signals. The dots, each representing 
a probe data point, are colored alternately to distinguish chromosomes. 
The segments, each representing a DNA segment resulting from the 
joint segmentation, are colored based on inferred copy number status. 
</p>


<h3>Value</h3>

<p>An R plot will be generated.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+joint.segmentation">joint.segmentation</a></code>, <code><a href="#topic+cnv.call">cnv.call</a></code>, 
<code><a href="#topic+diagnosis.seg.plot.chr">diagnosis.seg.plot.chr</a></code>, <code><a href="#topic+diagnosis.cluster.plot">diagnosis.cluster.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(seq.data)
data(seq.cnv)

genome.wide.plot(data=seq.data, segs=seq.cnv, 
                 sample.id="PT116", 
                 chrs=sub("^chr","",unique(seq.cnv$chr)), 
                 cex=0.3)
</code></pre>

<hr>
<h2 id='internals'>
Internal Functions and Data
</h2><span id='topic+compute.var'></span><span id='topic+computeBeta'></span><span id='topic+computeMoments'></span><span id='topic+computeTiltDirect'></span><span id='topic+ComputeZ.fromS.R'></span><span id='topic+ComputeZ.fromS.R.partial'></span><span id='topic+computeZ.onechange'></span><span id='topic+computeZ.onechange.sample'></span><span id='topic+computeZ.squarewave.sample'></span><span id='topic+dchi'></span><span id='topic+fcompute.max.Z'></span><span id='topic+fmscbs'></span><span id='topic+fscan.max'></span><span id='topic+getCutoffMultisampleWeightedChisq'></span><span id='topic+matrix.max'></span><span id='topic+mscbs.classify'></span><span id='topic+pmarg.sumweightedchisq'></span><span id='topic+pvalueMultisampleWeightedChisq'></span><span id='topic+vu'></span><span id='topic+BAF2mBAF'></span><span id='topic+Mode'></span><span id='topic+check.overlap'></span><span id='topic+cnv.data.chr'></span><span id='topic+compute.baseline'></span><span id='topic+compute.baseline.manual'></span><span id='topic+delta.sd'></span><span id='topic+diagnosis.QQ.plot'></span><span id='topic+impute.missing.data'></span><span id='topic+merging.segments.chr'></span><span id='topic+seg.summary'></span><span id='topic+seq.cnv'></span><span id='topic+seq.data'></span><span id='topic+seq.segs'></span><span id='topic+seq.segs.merge'></span><span id='topic+snp.cnv'></span><span id='topic+snp.cnv.anno'></span><span id='topic+snp.cnv.refine'></span><span id='topic+snp.cnv.refine.anno'></span><span id='topic+snp.data'></span><span id='topic+snp.segs'></span><span id='topic+snp.segs.merge'></span><span id='topic+snp_table'></span><span id='topic+vcf_table'></span>

<h3>Description</h3>

<p>These are the functions and data to which the users do not need to directly get access.
</p>

<hr>
<h2 id='joint.segmentation'>
Joint Segmentation on log2ratio and log2mBAF Dimensions
</h2><span id='topic+joint.segmentation'></span>

<h3>Description</h3>

<p>We employ the algorithm developed by (Zhang et al., 2010) to perform joint segmentation on log2ratio 
and log2mBAF dimensions. The function outputs the starting and ending points of each 
CNV segment as well as some summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>joint.segmentation(data, min.snps = 10, global.pval.cutoff = 1e-04, 
    max.chpts = 30, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="joint.segmentation_+3A_data">data</code></td>
<td>
<p>a data frame containing log2ratio and log2mBAF data generated 
by <code><a href="#topic+cnv.data">cnv.data</a></code>.
</p>
</td></tr>
<tr><td><code id="joint.segmentation_+3A_min.snps">min.snps</code></td>
<td>
<p>the minimum number of probes a segment needs to span.
</p>
</td></tr>
<tr><td><code id="joint.segmentation_+3A_global.pval.cutoff">global.pval.cutoff</code></td>
<td>
<p>the p-value cut-off a (or a pair) of change points to
be determined as significant in each cycle of joint segmentation.
</p>
</td></tr>
<tr><td><code id="joint.segmentation_+3A_max.chpts">max.chpts</code></td>
<td>
<p>the maximum number of change points to be detected for each 
chromosome.
</p>
</td></tr>
<tr><td><code id="joint.segmentation_+3A_verbose">verbose</code></td>
<td>
<p>logical. If more details to be output. Default is <code>TRUE</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the starting and ending points of each 
CNV segment as well as some summary statistics.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Zhang, N. R., Siegmund, D. O., Ji, H., Li, J. Z. (2010) 
Detecting simultaneous changepoints in multiple sequences. 
<em>Biometrika</em>, <b>97</b>:631&ndash;645.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cnv.data">cnv.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(seq.data)

## Not run: 
seq.segs &lt;- joint.segmentation(data=seq.data, min.snps=10, 
                               global.pval.cutoff=1e-4, max.chpts=30, 
                               verbose=TRUE)

## End(Not run)

## how the joint segmentation results look like
data(seq.segs)
head(seq.segs)
</code></pre>

<hr>
<h2 id='merging.segments'>
Merge Adjacent Segments
</h2><span id='topic+merging.segments'></span>

<h3>Description</h3>

<p>It is an option to merge adjacent segments, for which the median values 
in either or both log2ratio and log2mBAF dimensions are not substantially 
different. For WGS and SNP array, it is recommended to do so.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merging.segments(data, segs.stat, use.null.data = TRUE, 
               N = 1000, maxL = NULL, merge.pvalue.cutoff = 0.05, 
               do.manual.baseline=FALSE,
               log2mBAF.left=NULL, log2mBAF.right=NULL, 
               log2ratio.bottom=NULL, log2ratio.up=NULL,
               seed = NULL, 
               verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merging.segments_+3A_data">data</code></td>
<td>
<p>a data frame containing log2ratio and log2mBAF data generated 
by <code><a href="#topic+cnv.data">cnv.data</a></code>.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_segs.stat">segs.stat</code></td>
<td>
<p>a data frame containing segment locations and summary statistics 
resulting from <code><a href="#topic+joint.segmentation">joint.segmentation</a></code>.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_use.null.data">use.null.data</code></td>
<td>
<p>logical. If only data for probes located in normal copy 
segments to be used for bootstrapping. Default is <code>TRUE</code>. If a more 
aggressive merging is needed, it can be switched to <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_n">N</code></td>
<td>
<p>the number of replicates drawn by bootstrap.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_maxl">maxL</code></td>
<td>
<p>integer. The maximum length in terms of number of probes a bootstrapped segment
may span. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically specified 
as 1/100 of the number of data points.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_merge.pvalue.cutoff">merge.pvalue.cutoff</code></td>
<td>
<p>a p-value cut-off for merging. If the empirical p-value is 
greater than the cut-off value, the two adjacent segments under consideration will 
be merged.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_do.manual.baseline">do.manual.baseline</code></td>
<td>
<p>logical. If baseline adjustment to be done manually. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_log2mbaf.left">log2mBAF.left</code>, <code id="merging.segments_+3A_log2mbaf.right">log2mBAF.right</code>, <code id="merging.segments_+3A_log2ratio.bottom">log2ratio.bottom</code>, <code id="merging.segments_+3A_log2ratio.up">log2ratio.up</code></td>
<td>

<p>left, right, bottom and up boundaries to be specified manually by a visual inspectio of 
2-D diagnosis plot generated by <code><a href="#topic+diagnosis.cluster.plot">diagnosis.cluster.plot</a></code>. 
These parameters are active when <code>do.manual.baseline=TRUE</code>.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_seed">seed</code></td>
<td>
<p>integer. Random seed can be set for reproducibility of results.
</p>
</td></tr>
<tr><td><code id="merging.segments_+3A_verbose">verbose</code></td>
<td>
<p>logical. If more details to be output. Default is <code>TRUE</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the same columns as the one generated by 
<code><a href="#topic+joint.segmentation">joint.segmentation</a></code>.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cnv.data">cnv.data</a></code>, <code><a href="#topic+joint.segmentation">joint.segmentation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(seq.data)
data(seq.segs)

## Not run: 
seq.segs.merge &lt;- merging.segments(data=seq.data, segs.stat=seq.segs, 
                                 use.null.data=TRUE,
                                 N=1000, maxL=2000, 
                                 merge.pvalue.cutoff=0.05, verbose=TRUE)

## End(Not run)

## how the results look like
data(seq.segs.merge)
head(seq.segs.merge) 
</code></pre>

<hr>
<h2 id='NGS.CNV'>
CNV Analysis Pipeline for WGS and WES Data
</h2><span id='topic+NGS.CNV'></span>

<h3>Description</h3>

<p>All analysis steps are integrate into a pipeline. The results, 
including visualization plots are placed in a directory as specified by user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NGS.CNV(vcf, output.dir, sample.id, 
    do.GC.adjust = FALSE, 
    gc.file = system.file("extdata","GC_1kb_hg19.txt.gz",package="saasCNV"), 
    min.chr.probe = 100, min.snps = 10, 
    joint.segmentation.pvalue.cutoff = 1e-04, max.chpts = 30, 
    do.merge = TRUE, use.null.data = TRUE, 
    num.perm = 1000, maxL = NULL, 
    merge.pvalue.cutoff = 0.05, 
    do.cnvcall.on.merge = TRUE, 
    cnvcall.pvalue.cutoff = 0.05, 
    do.plot = TRUE, cex = 0.3, ref.num.probe = NULL, 
    do.gene.anno = FALSE, 
    gene.anno.file = NULL,
    seed = NULL, 
    verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NGS.CNV_+3A_vcf">vcf</code></td>
<td>
<p>a data frame constructed from a vcf file. See <code><a href="#topic+vcf2txt">vcf2txt</a></code>.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_output.dir">output.dir</code></td>
<td>
<p>the directory to which all the results will be located.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_sample.id">sample.id</code></td>
<td>
<p>sample ID to be displayed in the data frame of the results and 
the title of some diagnosis plots.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_do.gc.adjust">do.GC.adjust</code></td>
<td>
<p>logical. If GC content adjustment on <code>log2ratio</code> to be carried out. 
Default is <code>FALSE</code>. See <code><a href="#topic+GC.adjust">GC.adjust</a></code> for details.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_gc.file">gc.file</code></td>
<td>
<p>the location of tab-delimit file with GC content (averaged per 1kb window) 
information. See <code><a href="#topic+GC.adjust">GC.adjust</a></code> for details.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_min.chr.probe">min.chr.probe</code></td>
<td>
<p>the minimum number of probes tagging a chromosome for it 
to be passed to the subsequent analysis.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_min.snps">min.snps</code></td>
<td>
<p>the minimum number of probes a segment needs to span.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_joint.segmentation.pvalue.cutoff">joint.segmentation.pvalue.cutoff</code></td>
<td>
<p>the p-value cut-off one (or a pair) of change points to
be determined as significant in each cycle of joint segmentation.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_max.chpts">max.chpts</code></td>
<td>
<p>the maximum number of change points to be detected for each 
chromosome.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_do.merge">do.merge</code></td>
<td>
<p>logical. If segments merging step to be carried out. 
Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_use.null.data">use.null.data</code></td>
<td>
<p>logical. If only data for probes located in normal copy 
segments to be used for bootstrapping. Default is <code>TRUE</code>. If a more 
aggressive merging is needed, it can be switched to <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_num.perm">num.perm</code></td>
<td>
<p>the number of replicates drawn by bootstrap.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_maxl">maxL</code></td>
<td>
<p>integer. The maximum length in terms of number of probes a bootstrapped segment
may span. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically specified 
as 1/100 of the number of data points.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_merge.pvalue.cutoff">merge.pvalue.cutoff</code></td>
<td>
<p>a p-value cut-off for merging. If the empirical p-value is 
greater than the cut-off value, the two adjacent segments under consideration will 
be merged.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_do.cnvcall.on.merge">do.cnvcall.on.merge</code></td>
<td>
<p>logical. If CNV call to be done for the segments after 
merging step. Default is <code>TRUE</code>. If <code>TRUE</code>, CNV call will be done on the 
segments resulting directly from joint segmentation without merging step.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_cnvcall.pvalue.cutoff">cnvcall.pvalue.cutoff</code></td>
<td>
<p>a p-value cut-off for CNV calling.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_do.plot">do.plot</code></td>
<td>
<p>logical. If diagnosis plots to be output. Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_cex">cex</code></td>
<td>
<p>a numerical value giving the amount by which plotting text and 
symbols should be magnified relative to the default. It can be adjusted 
in order to make the plot legible.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_ref.num.probe">ref.num.probe</code></td>
<td>
<p>integer. The reference number of probes against which 
a segment is compared in order to determine the cex of the segment 
to be displayed. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically 
specified as 1/100 of the number of data points.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_do.gene.anno">do.gene.anno</code></td>
<td>
<p>logical. If gene annotation step to be performed. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_gene.anno.file">gene.anno.file</code></td>
<td>
<p>a tab-delimited file containing gene annotation information. 
For example, RefSeq annotation file which can be found at UCSC genome browser.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_seed">seed</code></td>
<td>
<p>integer. Random seed can be set for reproducibility of results.
</p>
</td></tr>
<tr><td><code id="NGS.CNV_+3A_verbose">verbose</code></td>
<td>
<p>logical. If more details to be output. Default is <code>TRUE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the vignettes of the package for more details.
</p>


<h3>Value</h3>

<p>The results, including visualization plots are placed in 
subdirectories of the output directory <code>output.dir</code>
as specified by user.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Zhongyang Zhang and Ke Hao. (2015) SAAS-CNV: A Joint Segmentation Approach on Aggregated and Allele Specific Signals 
for the Identification of Somatic Copy Number Alterations with Next-Generation Sequencing Data. 
PLoS Computational Biology, 11(11):e1004618.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vcf2txt">vcf2txt</a></code>, <code><a href="#topic+cnv.data">cnv.data</a></code>,
<code><a href="#topic+joint.segmentation">joint.segmentation</a></code>, <code><a href="#topic+merging.segments">merging.segments</a></code>
<code><a href="#topic+cnv.call">cnv.call</a></code>, <code><a href="#topic+diagnosis.seg.plot.chr">diagnosis.seg.plot.chr</a></code>,
<code><a href="#topic+genome.wide.plot">genome.wide.plot</a></code>, <code><a href="#topic+diagnosis.cluster.plot">diagnosis.cluster.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## NGS pipeline analysis
## download vcf_table.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/vcf_table.txt.gz"
tryCatch({download.file(url=url, destfile="vcf_table.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="vcf_table.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

vcf_table &lt;- read.delim(file="vcf_table.txt.gz", as.is=TRUE)

## download refGene_hg19.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/refGene_hg19.txt.gz"
tryCatch({download.file(url=url, destfile="refGene_hg19.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="refGene_hg19.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

sample.id &lt;- "WES_0116"
output.dir &lt;- file.path(getwd(), "test_saasCNV")

NGS.CNV(vcf=vcf_table, output.dir=output.dir, sample.id=sample.id,
        min.chr.probe=100,
        min.snps=10,
        joint.segmentation.pvalue.cutoff=1e-4,
        max.chpts=30,
        do.merge=TRUE, use.null.data=TRUE, num.perm=1000, maxL=2000, 
        merge.pvalue.cutoff=0.05,
        do.cnvcall.on.merge=TRUE, 
        cnvcall.pvalue.cutoff=0.05,
        do.plot=TRUE, cex=0.3, ref.num.probe=1000,
        do.gene.anno=TRUE,
        gene.anno.file="refGene_hg19.txt.gz",
        seed=123456789,
        verbose=TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='reannotate.CNV.res'>
Gene Annotation
</h2><span id='topic+reannotate.CNV.res'></span>

<h3>Description</h3>

<p>An optional function to add gene annotation to each CNV segment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reannotate.CNV.res(res, gene, only.CNV = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reannotate.CNV.res_+3A_res">res</code></td>
<td>
<p>a data frame resultingfrom <code><a href="#topic+cnv.call">cnv.call</a></code>.
</p>
</td></tr>
<tr><td><code id="reannotate.CNV.res_+3A_gene">gene</code></td>
<td>
<p>a data frame containing gene annotation information.
</p>
</td></tr>
<tr><td><code id="reannotate.CNV.res_+3A_only.cnv">only.CNV</code></td>
<td>
<p>logical. If only segment assigned to gain/loss/LOH 
to be annotated and output. Default is <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The RefSeq gene annotation file can be downloaded from UCSC Genome Browser.
</p>


<h3>Value</h3>

<p>A gene annotation column have been add to the data frame resulting from 
<code><a href="#topic+cnv.call">cnv.call</a></code>.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+joint.segmentation">joint.segmentation</a></code>, <code><a href="#topic+cnv.call">cnv.call</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
## An example of RefSeq gene annotation file,
## the original version of which can be downloaded from UCSC Genome Browser
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/refGene_hg19.txt.gz"
tryCatch({download.file(url=url, destfile="refGene_hg19.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="refGene_hg19.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

gene.anno &lt;- read.delim(file="refGene_hg19.txt.gz", as.is=TRUE, comment.char="")
data(seq.cnv)
seq.cnv.anno &lt;- reannotate.CNV.res(res=seq.cnv, gene=gene.anno, only.CNV=TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='SNP.CNV'>
CNV Analysis Pipeline for SNP array Data
</h2><span id='topic+SNP.CNV'></span>

<h3>Description</h3>

<p>All analysis steps are integrate into a pipeline. The results, 
including visualization plots are placed in a directory as specified by user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNP.CNV(snp, output.dir, sample.id, 
    do.GC.adjust = FALSE,
    gc.file = system.file("extdata","GC_1kb_hg19.txt.gz",package="saasCNV"),
    min.chr.probe = 100, min.snps = 10, 
    joint.segmentation.pvalue.cutoff = 1e-04, max.chpts = 30, 
    do.merge = TRUE, use.null.data = TRUE, 
    num.perm = 1000, maxL = NULL, 
    merge.pvalue.cutoff = 0.05, 
    do.cnvcall.on.merge = TRUE, 
    cnvcall.pvalue.cutoff = 0.05, 
    do.boundary.refine = FALSE, 
    do.plot = TRUE, cex = 0.3, 
    ref.num.probe = NULL, 
    do.gene.anno = FALSE, 
    gene.anno.file = NULL, 
    seed = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNP.CNV_+3A_snp">snp</code></td>
<td>
<p>a data frame constructed from a text file with LRR and BAF information. 

</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_output.dir">output.dir</code></td>
<td>
<p>the directory to which all the results will be located.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_sample.id">sample.id</code></td>
<td>
<p>sample ID to be displayed in the data frame of the results and 
the title of some diagnosis plots.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_do.gc.adjust">do.GC.adjust</code></td>
<td>
<p>logical. If GC content adjustment on <code>log2ratio</code> to be carried out. 
Default is <code>FALSE</code>. See <code><a href="#topic+GC.adjust">GC.adjust</a></code> for details.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_gc.file">gc.file</code></td>
<td>
<p>the location of tab-delimit file with GC content (averaged per 1kb window) 
information. See <code><a href="#topic+GC.adjust">GC.adjust</a></code> for details.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_min.chr.probe">min.chr.probe</code></td>
<td>
<p>the minimum number of probes tagging a chromosome for it 
to be passed to the subsequent analysis.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_min.snps">min.snps</code></td>
<td>
<p>the minimum number of probes a segment needs to span.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_joint.segmentation.pvalue.cutoff">joint.segmentation.pvalue.cutoff</code></td>
<td>
<p>the p-value cut-off one (or a pair) of change points to
be determined as significant in each cycle of joint segmentation.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_max.chpts">max.chpts</code></td>
<td>
<p>the maximum number of change points to be detected for each 
chromosome.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_do.merge">do.merge</code></td>
<td>
<p>logical. If segments merging step to be carried out. 
Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_use.null.data">use.null.data</code></td>
<td>
<p>logical. If only data for probes located in normal copy 
segments to be used for bootstrapping. Default is <code>TRUE</code>. If a more 
aggressive merging is needed, it can be switched to <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_num.perm">num.perm</code></td>
<td>
<p>the number of replicates drawn by bootstrap.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_maxl">maxL</code></td>
<td>
<p>integer. The maximum length in terms of number of probes a bootstrapped segment
may span. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically specified 
as 1/100 of the number of data points.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_merge.pvalue.cutoff">merge.pvalue.cutoff</code></td>
<td>
<p>a p-value cut-off for merging. If the empirical p-value is 
greater than the cut-off value, the two adjacent segments under consideration will 
be merged.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_do.cnvcall.on.merge">do.cnvcall.on.merge</code></td>
<td>
<p>logical. If CNV call to be done for the segments after 
merging step. Default is <code>TRUE</code>. If <code>TRUE</code>, CNV call will be done on the 
segments resulting directly from joint segmentation without merging step.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_cnvcall.pvalue.cutoff">cnvcall.pvalue.cutoff</code></td>
<td>
<p>a p-value cut-off for CNV calling.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_do.boundary.refine">do.boundary.refine</code></td>
<td>
<p>logical. If the segment boundaries based on the grid of 
heterozygous probes to be refined by all probes with LRR data. Default is <code>FALSE</code>. 
We do not recommend to perform this step except in the case that the segment 
boundaries need to be aligned well on the same grid of probes for downstream analysis.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_do.plot">do.plot</code></td>
<td>
<p>logical. If diagnosis plots to be output. Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_cex">cex</code></td>
<td>
<p>a numerical value giving the amount by which plotting text and 
symbols should be magnified relative to the default. It can be adjusted 
in order to make the plot legible.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_ref.num.probe">ref.num.probe</code></td>
<td>
<p>integer. The reference number of probes against which 
a segment is compared in order to determine the cex of the segment 
to be displayed. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically 
specified as 1/100 of the number of data points.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_do.gene.anno">do.gene.anno</code></td>
<td>
<p>logical. If gene annotation step to be performed. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_gene.anno.file">gene.anno.file</code></td>
<td>
<p>a tab-delimited file containing gene annotation information. 
For example, RefSeq annotation file which can be found at UCSC genome browser.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_seed">seed</code></td>
<td>
<p>integer. Random seed can be set for reproducibility of results.
</p>
</td></tr>
<tr><td><code id="SNP.CNV_+3A_verbose">verbose</code></td>
<td>
<p>logical. If more details to be output. Default is <code>TRUE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the vignettes of the package for more details.
</p>


<h3>Value</h3>

<p>The results, including visualization plots are placed in 
subdirectories of the output directory <code>output.dir</code>
as specified by user.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Zhongyang Zhang and Ke Hao. (2015) SAAS-CNV: A Joint Segmentation Approach on Aggregated and Allele Specific Signals 
for the Identification of Somatic Copy Number Alterations with Next-Generation Sequencing Data. 
PLoS Computational Biology, 11(11):e1004618.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+NGS.CNV">NGS.CNV</a></code>, <code><a href="#topic+snp.cnv.data">snp.cnv.data</a></code>,
<code><a href="#topic+joint.segmentation">joint.segmentation</a></code>, <code><a href="#topic+merging.segments">merging.segments</a></code>
<code><a href="#topic+cnv.call">cnv.call</a></code>, <code><a href="#topic+diagnosis.seg.plot.chr">diagnosis.seg.plot.chr</a></code>,
<code><a href="#topic+genome.wide.plot">genome.wide.plot</a></code>, <code><a href="#topic+diagnosis.cluster.plot">diagnosis.cluster.plot</a></code>, 
<code><a href="#topic+snp.refine.boundary">snp.refine.boundary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## the pipeline for SNP array analysis
## download snp_table.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/snp_table.txt.gz"
tryCatch({download.file(url=url, destfile="snp_table.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="snp_table.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

snp_table &lt;- read.delim(file="snp_table.txt.gz", as.is=TRUE)

## download refGene_hg19.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/refGene_hg19.txt.gz"
tryCatch({download.file(url=url, destfile="refGene_hg19.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="refGene_hg19.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

sample.id &lt;- "SNP_0116"
output.dir &lt;- file.path(getwd(), "test_saasCNV")
	
SNP.CNV(snp=snp_table, output.dir=output.dir, sample.id=sample.id,
        min.chr.probe=100,
        min.snps=10,
        joint.segmentation.pvalue.cutoff=1e-4,
        max.chpts=30,
        do.merge=TRUE, use.null.data=TRUE, num.perm=1000, maxL=5000, 
        merge.pvalue.cutoff=0.05,
        do.cnvcall.on.merge=TRUE, 
        cnvcall.pvalue.cutoff=0.05,
        do.boundary.refine=TRUE,
        do.plot=TRUE, cex=0.3, ref.num.probe=5000,
        do.gene.anno=TRUE,
        gene.anno.file="refGene_hg19.txt.gz",
        seed=123456789,
        verbose=TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='snp.cnv.data'>
Construct Data Frame for CNV Inference with SNP Array Data
</h2><span id='topic+snp.cnv.data'></span>

<h3>Description</h3>

<p>Transform LRR and BAF information into log2ratio and log2mBAF
that we use for joint segmentation and CNV calling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snp.cnv.data(snp, min.chr.probe = 100, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snp.cnv.data_+3A_snp">snp</code></td>
<td>
<p>a data frame with LRR and BAF information from SNP array. See the 
example below for details.
</p>
</td></tr>
<tr><td><code id="snp.cnv.data_+3A_min.chr.probe">min.chr.probe</code></td>
<td>
<p>the minimum number of probes tagging a chromosome for it 
to be passed to the subsequent analysis.
</p>
</td></tr>
<tr><td><code id="snp.cnv.data_+3A_verbose">verbose</code></td>
<td>
<p>logical. If more details to be output. Default is <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the log2raio and log2mBAF values for each probe site.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Staaf, J., Vallon-Christersson, J., Lindgren, D., Juliusson, G., Rosenquist, R., 
Hoglund, M., Borg, A., Ringner, M. (2008) Normalization of Illumina Infinium 
whole-genome SNP data improves copy number estimates and allelic intensity ratios. 
<em>BMC bioinformatics</em>, <b>9</b>:409.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cnv.data">cnv.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## an example data with LRR and BAF information
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/snp_table.txt.gz"
tryCatch({download.file(url=url, destfile="snp_table.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="snp_table.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

snp_table &lt;- read.delim(file="snp_table.txt.gz", as.is=TRUE)
snp.data &lt;- snp.cnv.data(snp=snp_table, min.chr.probe=100, verbose=TRUE) 

## see how seq.data looks like
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/snp.data.RData"
tryCatch({download.file(url=url, destfile="snp.data.RData")
         }, error = function(e) {
          download.file(url=url, destfile="snp.data.RData", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

load("snp.data.RData")
head(snp.data)

## End(Not run)

</code></pre>

<hr>
<h2 id='snp.refine.boundary'>
Refine Segment Boundaries
</h2><span id='topic+snp.refine.boundary'></span>

<h3>Description</h3>

<p>Refine the segment boundaries based on the grid of heterozygous probes 
by all probes with LRR data. We do not recommend to perform this step 
except in the case that the segment boundaries need to be aligned well 
on the same grid of probes for downstream analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snp.refine.boundary(data, segs.stat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snp.refine.boundary_+3A_data">data</code></td>
<td>
<p>a data frame containing log2ratio and log2mBAF data generated 
by <code><a href="#topic+snp.cnv.data">snp.cnv.data</a></code>.
</p>
</td></tr>
<tr><td><code id="snp.refine.boundary_+3A_segs.stat">segs.stat</code></td>
<td>
<p>a data frame containing segment locations and summary statistics 
resulting from <code><a href="#topic+cnv.call">cnv.call</a></code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the same columns as the one generated by 
<code><a href="#topic+cnv.call">cnv.call</a></code> with the columns <code>posStart</code>, 
<code>posEnd</code>, <code>length</code>, <code>chrIdxStart</code>, 
<code>chrIdxEnd</code> and <code>numProbe</code> updated accordingly.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+snp.cnv.data">snp.cnv.data</a></code>, <code><a href="#topic+cnv.call">cnv.call</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## download snp.data.RData
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/snp.data.RData"
tryCatch({download.file(url=url, destfile="snp.data.RData")
         }, error = function(e) {
          download.file(url=url, destfile="snp.data.RData", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

load("snp.data.RData")
data(snp.cnv)
snp.cnv.refine &lt;- snp.refine.boundary(data=snp.data, segs.stat=snp.cnv)

## End(Not run)

## how the results look like
data(snp.cnv.refine)
head(snp.cnv.refine)
</code></pre>

<hr>
<h2 id='vcf2txt'>
Covert VCF File to A Data Frame
</h2><span id='topic+vcf2txt'></span>

<h3>Description</h3>

<p>It parses a VCF file and extract necessary information for CNV analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf2txt(vcf.file, normal.col = 10, tumor.col = 11, MQ.cutoff = 30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcf2txt_+3A_vcf.file">vcf.file</code></td>
<td>
<p>vcf file name.
</p>
</td></tr>
<tr><td><code id="vcf2txt_+3A_normal.col">normal.col</code></td>
<td>
<p>the number of the column in which the genotype
and read depth information of normal tissue are located in the vcf file.
</p>
</td></tr>
<tr><td><code id="vcf2txt_+3A_tumor.col">tumor.col</code></td>
<td>
<p>the number of the column in which the genotype
and read depth information of tumor tissue are located in the vcf file.
</p>
</td></tr>
<tr><td><code id="vcf2txt_+3A_mq.cutoff">MQ.cutoff</code></td>
<td>
<p>the minimum criterion of mapping quality.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the first 9 columns in vcf file are mandatory, followed by the
information for called variant starting from the 10th column.
</p>


<h3>Value</h3>

<p>A data frame of detailed information about each variant, including 
chrosome position, reference and alternative alleles, genotype and read depth 
carrying reference and alternative alleles for normal and tumor respectively.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M. A., 
Handsaker, R. E., Lunter, G., Marth, G. T., Sherry, S. T., et al. (2011) 
The variant call format and VCFtools. <em>Bioinformatics</em>, <b>27</b>:2156&ndash;2158.
</p>
<p>http://www.1000genomes.org/node/101
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## an example VCF file from WES
## download WES_example.vcf.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/WES_example.vcf.gz"
tryCatch({download.file(url=url, destfile="WES_example.vcf.gz")
         }, error = function(e) {
          download.file(url=url, destfile="WES_example.vcf.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

## convert VCf file to a data frame
vcf_table &lt;- vcf2txt(vcf.file="WES_example.vcf.gz", normal.col=9+1, tumor.col=9+2)

## see how vcf_table looks like
## download vcf_table.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/vcf_table.txt.gz"
tryCatch({download.file(url=url, destfile="vcf_table.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="vcf_table.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

vcf_table &lt;- read.delim(file="vcf_table.txt.gz", as.is=TRUE)
head(vcf_table)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
