<!DOCTYPE html><html lang="en"><head><title>Help for package SADEG</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SADEG}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SADEG.ACGT'><p>SADEG.ACGT</p></a></li>
<li><a href='#SADEG.CAI'><p>SADEG.CAI</p></a></li>
<li><a href='#SADEG.CodonUsage'><p>SADEG.CodonUsage</p></a></li>
<li><a href='#SADEG.Dimer'><p>SADEG.Dimer</p></a></li>
<li><a href='#SADEG.DimerStackingEnergy'><p>SADEG.DimerStackingEnergy</p></a></li>
<li><a href='#SADEG.EENC'><p>SADEG.EENC</p></a></li>
<li><a href='#SADEG.GC'><p>SADEG.GC</p></a></li>
<li><a href='#SADEG.GC1'><p>SADEG.GC1</p></a></li>
<li><a href='#SADEG.GC12'><p>SADEG.GC12</p></a></li>
<li><a href='#SADEG.GC2'><p>SADEG.GC2</p></a></li>
<li><a href='#SADEG.GC3'><p>SADEG.GC3</p></a></li>
<li><a href='#SADEG.ReverseComplement'><p>SADEG.ReverseComplement</p></a></li>
<li><a href='#SADEG.RSCU'><p>SADEG.RSCU</p></a></li>
<li><a href='#SADEG.StackingEnergy'><p>SADEG.StackingEnergy</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Stability Analysis in Differentially Expressed Genes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Babak Khorsand &lt;khorsand@mail.um.ac.ir&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We analyzed the nucleotide composition of genes with a special emphasis on stability of DNA sequences. Besides, in a variety of different organisms unequal use of synonymous codons, or codon usage bias, occurs which also show variation among genes in the same genome. Seemingly, codon usage bias is affected by both selective constraints and mutation bias which allows and enables us to examine and detect changes in these two evolutionary forces between genomes or along one genome. Therefore, we determined the codon adaptation index (CAI), effective number of codons (ENC) and codon usage analysis with calculation of the relative synonymous codon usage (RSCU), and subsequently predicted the translation efficiency and accuracy through GC-rich codon usages. Furthermore, we estimated the relative stability of the DNA sequence following calculation of the average free energy (Delta G) and Dimer base-stacking energy level.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-01-12 21:45:26 UTC; Babak</td>
</tr>
<tr>
<td>Author:</td>
<td>Babak Khorsand [aut, cre],
  Ehsan Sadeghnezhad [aut],
  Javad Zahiri [ths],
  Mohsen Sharifi [ths],
  Hassan Zare-mayvan [ths]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-01-13 00:43:16</td>
</tr>
</table>
<hr>
<h2 id='SADEG.ACGT'>SADEG.ACGT</h2><span id='topic+SADEG.ACGT'></span>

<h3>Description</h3>

<p>Number and frequencies of occurrence of each nucleotide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.ACGT(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.ACGT_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Number of occurrence of each nucleotide.(A, T, C, and G). Furtheremore, The frequencies of occurrence of each nucleotide. (A 
</p>


<h3>Value</h3>

<p>Table which first row represent number of each nucleotide and row 2 represent percentage of each nucleotide.
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.ACGT("actagtcacgatcag")
</code></pre>

<hr>
<h2 id='SADEG.CAI'>SADEG.CAI</h2><span id='topic+SADEG.CAI'></span>

<h3>Description</h3>

<p>Codon Adaption Index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.CAI(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.CAI_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Geometric mean of the RSCU values presented as relative adaptiveness value (w), The CAI for a gene is then defined as the geometric mean of w values for codons in that gene excluding methionine, tryptophan, and stop codons.
</p>


<h3>Value</h3>

<p>CAI
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.CAI(Nucleotide_Sequence="atggctgctgcagcggccagtcacgatcagaggtaagttgtcgcagcatgt")
</code></pre>

<hr>
<h2 id='SADEG.CodonUsage'>SADEG.CodonUsage</h2><span id='topic+SADEG.CodonUsage'></span>

<h3>Description</h3>

<p>Frequency of occurrence of each Amino acids codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.CodonUsage(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.CodonUsage_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Frequency of occurrence of each Amino acids codons.
</p>


<h3>Value</h3>

<p>CodonUsage
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.CodonUsage(Nucleotide_Sequence="atggctgcagcggccagtcacgatcagaggtaagttgtc")
</code></pre>

<hr>
<h2 id='SADEG.Dimer'>SADEG.Dimer</h2><span id='topic+SADEG.Dimer'></span>

<h3>Description</h3>

<p>Number and frequency of each Dimers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.Dimer(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.Dimer_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Count the number of each Dimers and calculate frequencies of them.
</p>


<h3>Value</h3>

<p>Table which first row represent number of each Dimers and row 2 represent percentage of each Dimers.
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.Dimer("actagtcacgatcag")
</code></pre>

<hr>
<h2 id='SADEG.DimerStackingEnergy'>SADEG.DimerStackingEnergy</h2><span id='topic+SADEG.DimerStackingEnergy'></span>

<h3>Description</h3>

<p>Sum of Stacking energy of each Dimers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.DimerStackingEnergy(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.DimerStackingEnergy_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Count the number of each Dimers and multiply by related stacking energy.
</p>


<h3>Value</h3>

<p>Sum of Stacking energy of each Dimers.
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.DimerStackingEnergy("actagtcacgatcag")
</code></pre>

<hr>
<h2 id='SADEG.EENC'>SADEG.EENC</h2><span id='topic+SADEG.EENC'></span>

<h3>Description</h3>

<p>Expected Effective Number Of Codon
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.EENC(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.EENC_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ENC analysis was used to quantify the absolute codon usage bias by determining the degree of codon usage bias exhibited by the coding sequences, regardless of gene length and the number of amino acids. ENC values ranged between 20 and 61. ENC value of 20, indicates extreme codon usage bias using only one of the possible synonymous codons for the corresponding amino acid and value 61 denotes no bias using all possible synonymous codons equally for the corresponding amino acid. A smaller ENC value correlates with the larger extension of codon preference in a gene. It is also generally accepted that when the ENC value is less than or equal to 35, genes tend to have a significant codon bias.
</p>


<h3>Value</h3>

<p>EENC
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.EENC(Nucleotide_Sequence="atggctgctgcagcggccagtcacgatcagaggtaagttgtcgcagcatgt")
</code></pre>

<hr>
<h2 id='SADEG.GC'>SADEG.GC</h2><span id='topic+SADEG.GC'></span>

<h3>Description</h3>

<p>GC content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.GC(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.GC_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GC content : Sum of frequencies of G and C.
</p>


<h3>Value</h3>

<p>GC content
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.GC("actagtcacgatcag")
</code></pre>

<hr>
<h2 id='SADEG.GC1'>SADEG.GC1</h2><span id='topic+SADEG.GC1'></span>

<h3>Description</h3>

<p>GC1 content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.GC1(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.GC1_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GC1 content : Sum of frequencies of G and C at the first position of each codon.
</p>


<h3>Value</h3>

<p>GC1
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.GC1(Nucleotide_Sequence="atggctgctgcagcggccagtcacgatcagaggtaagttgtcgcagcatgt")
</code></pre>

<hr>
<h2 id='SADEG.GC12'>SADEG.GC12</h2><span id='topic+SADEG.GC12'></span>

<h3>Description</h3>

<p>GC12 content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.GC12(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.GC12_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GC12 content : Sum of frequencies of G and C at first and second position of each codon.
</p>


<h3>Value</h3>

<p>GC12
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.GC12(Nucleotide_Sequence="atggctgctgcagcggccagtcacgatcagaggtaagttgtcgcagcatgt")
</code></pre>

<hr>
<h2 id='SADEG.GC2'>SADEG.GC2</h2><span id='topic+SADEG.GC2'></span>

<h3>Description</h3>

<p>GC2 content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.GC2(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.GC2_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GC2 content : Sum of frequencies of G and C at the second position of each codon.
</p>


<h3>Value</h3>

<p>GC2
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.GC2(Nucleotide_Sequence="atggctgctgcagcggccagtcacgatcagaggtaagttgtcgcagcatgt")
</code></pre>

<hr>
<h2 id='SADEG.GC3'>SADEG.GC3</h2><span id='topic+SADEG.GC3'></span>

<h3>Description</h3>

<p>GC3 content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.GC3(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.GC3_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GC3 content : Sum of frequencies of G and C at the third position of each codon.
</p>


<h3>Value</h3>

<p>GC3
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.GC3(Nucleotide_Sequence="atggctgctgcagcggccagtcacgatcagaggtaagttgtcgcagcatgt")
</code></pre>

<hr>
<h2 id='SADEG.ReverseComplement'>SADEG.ReverseComplement</h2><span id='topic+SADEG.ReverseComplement'></span>

<h3>Description</h3>

<p>Complement of the reversed sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.ReverseComplement(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.ReverseComplement_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Complement of the reversed sequence. (atcagt -&gt; Reverse: tgacta -&gt; Reverse-Complement: actgat)
</p>


<h3>Value</h3>

<p>Reverse-Complement
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.ReverseComplement("actagtcacgatcag")
</code></pre>

<hr>
<h2 id='SADEG.RSCU'>SADEG.RSCU</h2><span id='topic+SADEG.RSCU'></span>

<h3>Description</h3>

<p>Relative Synonymous Codon Usage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.RSCU(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.RSCU_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Relative Synonymous Codon Usage (RSCU) is the number of times a codon appears in a gene divided by the number of expected occurrences under equal codon usage.
</p>


<h3>Value</h3>

<p>RSCU
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.RSCU(Nucleotide_Sequence="atggctgcagcggccagtcacgatcagaggtaagttgtc")
</code></pre>

<hr>
<h2 id='SADEG.StackingEnergy'>SADEG.StackingEnergy</h2><span id='topic+SADEG.StackingEnergy'></span>

<h3>Description</h3>

<p>Average free energy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SADEG.StackingEnergy(Nucleotide_Sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SADEG.StackingEnergy_+3A_nucleotide_sequence">Nucleotide_Sequence</code></td>
<td>
<p>Nucleotide Sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Average free energy value was obtained via multiplying average energy of each dimer from each sequence in number of stacked bases. In order to designate and identify a product-favored process, minus sign was denoted before the average free energy (AFE) value. Lastly, all relevant stacked bases in each gene cumulatively defined the total stacking energy for respective sequences.
</p>


<h3>Value</h3>

<p>StackingEnergy
</p>


<h3>Author(s)</h3>

<p>Babak Khorsand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SADEG.StackingEnergy("actagtcacgatcag")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
