<!DOCTYPE html><html lang="en"><head><title>Help for package sand</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sand}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aidsblog'><p>AIDS blog citation network</p></a></li>
<li><a href='#calldata'><p>Austrian phone call network data</p></a></li>
<li><a href='#Ecoli'><p>E. coli gene expression levels</p></a></li>
<li><a href='#fblog'><p>Network of French political blogs</p></a></li>
<li><a href='#g.bip'><p>A toy bipartite network</p></a></li>
<li><a href='#hc'><p>Hospital encounter network data</p></a></li>
<li><a href='#install_sand_packages'><p>Install all packages used in the book</p></a></li>
<li><a href='#lazega'><p>Lazega lawyers network data</p></a></li>
<li><a href='#ppi.CC'><p>Yeast protein interaction network</p></a></li>
<li><a href='#sand'><p>The sand package</p></a></li>
<li><a href='#sandwichprobe'><p>Internet packet probes data</p></a></li>
<li><a href='#strike'><p>Michael's strike network</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>2.0.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Analysis of Network Data with R, 2nd Edition</td>
</tr>
<tr>
<td>Author:</td>
<td>Eric Kolaczyk [aut, cre],
  Gábor Csárdi [aut],
  Carolyn Kolaczyk [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Eric Kolaczyk &lt;eric.kolaczyk@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), igraph, igraphdata</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>GO.db, GOstats, ROCR, ape, blockmodels, car, eigenmodel,
ergm, fdrtool, ggplot2, huge, kernlab, lattice, network,
networkDynamic, networkTomography, ngspatial, org.Sc.sgd.db,
sna, vioplot</td>
</tr>
<tr>
<td>Description:</td>
<td>Data sets and code blocks for the book 'Statistical Analysis of 
  Network Data with R, 2nd Edition'.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kolaczyk/sand">https://github.com/kolaczyk/sand</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kolaczyk/sand/issues">https://github.com/kolaczyk/sand/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-07-01 14:31:46 UTC; erick</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-07-02 07:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='aidsblog'>AIDS blog citation network</h2><span id='topic+aidsblog'></span>

<h3>Description</h3>

<p>A snapshot of the pattern of citation among 146 unique blogs 
related to AIDS, patients, and their support networks,
collected by Suchi Gopal (see reference below) over a randomly
selected three-day period in August 2005. A directed edge from one
blog to another indicates that the former has a link to the latter in
their web page (more specifically, the former refers to the latter in
their so-called &lsquo;blogroll&rsquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aidsblog
</code></pre>


<h3>Format</h3>

<p>A directed igraph graph object with 146 vertices and 187 edges.
</p>


<h3>Source</h3>

<p>This dataset was provided to us by Suchi Gopal.
Please cite the reference below if you use this dataset in your work.
</p>


<h3>References</h3>

<p>S. Gopal, The evolving social geography of blogs. In Societies and
Cities in the Age of Instant Access, ed. by H. Miller (Springer,
Berlin, 2007), 139 pp. 275-294.
</p>

<hr>
<h2 id='calldata'>Austrian phone call network data</h2><span id='topic+calldata'></span>

<h3>Description</h3>

<p>A set of data for phone traffic 60 between 32 telecommunication
districts in Austria throughout a period during the 61 year 1991.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calldata
</code></pre>


<h3>Format</h3>

<p>A data frame with 32 x 31 flow measurements, 992 rows, and seven
columns:
</p>

<ul>
<li> <p><code>Orig</code>: factor, the origin district.   
</p>
</li>
<li> <p><code>Dest</code>: factor, the destination district.
</p>
</li>
<li> <p><code>DistEuc</code>: numeric, Euclidean distance between the
districts.
</p>
</li>
<li> <p><code>DistRd</code>: numeric, road distance between districts.
</p>
</li>
<li> <p><code>O.GRP</code>: numeric, gross regional product of the origin
district, in Austrian schillings.
</p>
</li>
<li> <p><code>D.GRP</code>: numeric, gross regional product of the
destination district, in Austrian schillings.
</p>
</li>
<li> <p><code>Flow</code>: the &ldquo;amount&rdquo; of phone calls from the
origin district to the destination district, in erlang units
(number of phone calls, including faxes, times the average length
of the call divided by the duration of the measurement period).
</p>
</li></ul>



<h3>Source</h3>

<p>This dataset was provided to us by Suchi Gopal.
Please cite the reference below if you use this dataset in your work.
</p>


<h3>References</h3>

<p>M. Fischer, S. Gopal: Artificial neural networks: a new approach to
modeling interregional telecommunication flows. J. Reg. Sci. 34(4),
503-527 (1994).
</p>

<hr>
<h2 id='Ecoli'>E. coli gene expression levels</h2><span id='topic+Ecoli'></span><span id='topic+Ecoli.expr'></span><span id='topic+Ecoli.data'></span><span id='topic+regDB.adj'></span>

<h3>Description</h3>

<p>Gene expression levels in the bacteria Escherichia coli (E. coli),
measured for 153 genes under each of 40 different experimental
conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Ecoli.data)
Ecoli.expr
regDB.adj
</code></pre>


<h3>Format</h3>

<p><code>Ecoli.expr</code> is a 40 by 153 matrix of (log) gene expression
levels in the bacteria Escherichia coli (E. coli), measured for
153 transcription factors under each of 40 different experimental
conditions,  averaged over three replicates of each experiment. The data 
are a subset of those published in the reference below. The
experiments were genetic perturbation experiments,in which a given
gene was &lsquo;turned off&rsquo;, for each of 40 different genes.
</p>
<p><code>regDB.adj</code> is an adjacency matrix of regulatory relationships in
E. coli, extracted from the RegulonDB
(<a href="http://regulondb.ccg.unam.mx/">http://regulondb.ccg.unam.mx/</a>) database at the same time the
experimental data were collected. 
</p>


<h3>Source</h3>

<p>See the reference below. Please cite it if you use this dataset in
your work.
</p>


<h3>References</h3>

<p>J. Faith, B. Hayete, J. Thaden, I. Mogno, J. Wierzbowski, G. Cottarel,
S. Kasif, J. Collins, T. Gardner: Large-scale mapping and
validation of Escherichia coli transcriptional regulation from a
compendium of expression profiles. PLoS Biol. 5(1), e8 (2007).
</p>

<hr>
<h2 id='fblog'>Network of French political blogs</h2><span id='topic+fblog'></span>

<h3>Description</h3>

<p>Subnetwork of French political blogs, extracted from a snapshot of
over 1,100 such blogs on a single day in October of 2006 and
classified by the &ldquo;Observatoire Presidentielle&rdquo; project as to
political affiliation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fblog
</code></pre>


<h3>Format</h3>

<p>An undirected igraph graph with 192 vertices and 1431 edges. Note that
the graph is undirected. The graph has two vertex attributes,
&lsquo;name&rsquo; is the URL of the blog, and &lsquo;PolParty&rsquo; is the
assigned political affiliation, a political party.
</p>


<h3>Source</h3>

<p>The <code>mixer</code> R package.
</p>

<hr>
<h2 id='g.bip'>A toy bipartite network</h2><span id='topic+g.bip'></span>

<h3>Description</h3>

<p>A toy bipartite network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g.bip
</code></pre>


<h3>Format</h3>

<p>An undirected bipartite igraph graph object, with vertex attributes
&lsquo;name&rsquo; and &lsquo;type&rsquo;.
</p>

<hr>
<h2 id='hc'>Hospital encounter network data</h2><span id='topic+hc'></span>

<h3>Description</h3>

<p>Records of contacts among patients and various types of health care
workers in the geriatric unit of a hospital in Lyon, France, in
2010, from 1pm on Monday, December 6 to 2pm on Friday, December
10. Each of the 75 people in this study consented to wear RFID 
sensors on small identification badges during this period, which made
it possible to record when any two of them were in face-to-face
contact with each other (i.e., within 1-1.5 m of each other) during
a 20-second interval of time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hc
</code></pre>


<h3>Format</h3>

<p>A data frame, where each row is an interaction. It has five columns:
</p>

<ul>
<li> <p><code>Time</code>: integer, time in seconds when the 20 second
encounter terminated.
</p>
</li>
<li> <p><code>ID1</code>: integer, numeric ID of the first person.
</p>
</li>
<li> <p><code>ID2</code>: integer, numeric ID of the second person.
</p>
</li>
<li> <p><code>S1</code>: factor, the status of the first person, see below.
</p>
</li>
<li> <p><code>S2</code>: factor, the status of the second person, see below.
</p>
</li></ul>

<p>Status codes: administrative staff (ADM), medical doctor (MED),
paramedical staff, such as nurses or nurses' aides (NUR), and
patients (PAT).
</p>


<h3>Source</h3>

<p>See the reference below.
Please cite the it if you use this dataset in your work.
</p>


<h3>References</h3>

<p>P. Vanhems, A. Barrat, C. Cattuto, J.-F. Pinton, N. Khanafer,
C. Regis, B.-a. Kim, B. Comte, N. Voirin: Estimating potential
infection transmission routes in hospital wards using wearable
proximity sensors. PloS One 8(9), e73970 306 (2013).
</p>

<hr>
<h2 id='install_sand_packages'>Install all packages used in the book</h2><span id='topic+install_sand_packages'></span>

<h3>Description</h3>

<p>This function makes it easy to download and install
all R packages that are used in the book &lsquo;Statistical Analysis
of Network Data with R, 2nd Edition&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_sand_packages()
</code></pre>


<h3>Details</h3>

<p>The function uses the BioConductor installer, as this can
install both all required BioConductor and CRAN packages.
</p>


<h3>Value</h3>

<p>Returns the names of the installed packages, invisibly.
</p>


<h3>Author(s)</h3>

<p>Gabor Csardi <a href="mailto:csardi.gabor@gmail.com">csardi.gabor@gmail.com</a>
</p>

<hr>
<h2 id='lazega'>Lazega lawyers network data</h2><span id='topic+lazega'></span><span id='topic+elist.lazega'></span><span id='topic+v.attr.lazega'></span>

<h3>Description</h3>

<p>This data set comes from a network study of corporate law partnership
that was carried out in a Northeastern US corporate law firm, referred
to as SG&amp;R, 1988-1991 in New England. It includes (among others)
measurements of networks among the 71 attorneys (partners and
associates) of this firm, i.e. their strong-coworker network, advice
network, friendship network, and indirect control networks. Various
members' attributes are also part of the dataset, including seniority,
formal status, office in which they work, gender, lawschool attended,
individual performance measurements (hours worked, fees brought in),
attitudes concerning various management policy options, etc.
</p>
<p>Note that this is only a subset of the originally collected data,
including the 36 partners of the firm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lazega
elist.lazega
v.attr.lazega
</code></pre>


<h3>Format</h3>

<p><code>lazega</code> is an igraph graph object, undirected. It has the
following vertex attributes: &lsquo;name&rsquo;, &lsquo;Seniority&rsquo;,
&lsquo;Status&rsquo; (all 1, meaning partner), &lsquo;Gender&rsquo; (1 is man, 2
is woman), &lsquo;Office&rsquo; (1 is Boston, 2 is Hartford, 3 is
Providence), &lsquo;Years&rsquo; (years with the firm), &lsquo;Age&rsquo;,
&lsquo;Practice&rsquo; (1 is litigation, 2 is corporate),
and &lsquo;School&rsquo; (1 is Harvard or Yale, 2 is University of
Connecticut, 3 is other). See the reference below for more.
</p>
<p><code>elist.lazega</code> is a data frame containing an edge list of the
network.
</p>
<p><code>v.attr.lazega</code> is a data frame containing the vertex attributes
only.
</p>


<h3>Source</h3>

<p>Provided to us by Emmanuel Lazega. Please cite the
reference below if you use this dataset in your work.
</p>


<h3>References</h3>

<p>E. Lazega, The Collegial Phenomenon: The Social Mechanisms of
Cooperation Among Peers in a Corporate Law Partnership. Oxford
University Press, Oxford (2001).
</p>

<hr>
<h2 id='ppi.CC'>Yeast protein interaction network</h2><span id='topic+ppi.CC'></span>

<h3>Description</h3>

<p>A network of 241 interactions among 134 proteins.  They were assembled
by Jiang et al. (see below), from various sources, and pertain to only
those proteins annotated, as of January 2007, with the term &ldquo;cell 
communication&rdquo; in the gene ontology (GO) database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppi.CC
</code></pre>


<h3>Format</h3>

<p>An undirected igraph graph object, with vertex attributes:
</p>

<ul>
<li> <p>&lsquo;name&rsquo;: the name of the protein.
</p>
</li>
<li> <p>&lsquo;ICSC&rsquo;: whether the protein is annotated with the
&ldquo;intracellular signaling cascade&rdquo; GO term, zero or one.
</p>
</li>
<li> <p>&lsquo;IPR000198&rsquo;: whether the protein contains the
&lsquo;rho GTPase-activating protein domain&rsquo; (IPR000198) motif.
</p>
</li>
<li> <p>&lsquo;IPR000403&rsquo;: whether the protein contains the
&lsquo;phosphatidylinositol 3-/4-kinase, catalytic domain&rsquo;
(IPR000403) motif.
</p>
</li>
<li> <p>&lsquo;IPR001806&rsquo;: whether the protein contains the
&lsquo;small GTPase superfamily&rsquo; (IPR001806) motif.
</p>
</li>
<li> <p>&lsquo;IPR001849&rsquo;: whether the protein contains the
&lsquo;pleckstrin homology domain&rsquo; (IPR001849) motif.
</p>
</li>
<li> <p>&lsquo;IPR002041&rsquo;: whether the protein contains the
&lsquo;ran GTPase&rsquo; (IPR002041) motif.
</p>
</li>
<li> <p>&lsquo;IPR003527&rsquo;: whether the protein contains the
&lsquo;mitogen-activated protein (MAP) kinase, conserved site&rsquo;
(IPR003527) motif.
</p>
</li></ul>



<h3>Source</h3>

<p>See the reference below. Please cite it if you use this dataset in
your work.
</p>


<h3>References</h3>

<p>X. Jiang, N. Nariai, M. Steffen, S. Kasif, E. Kolaczyk: Integration of
relational and hierarchical network information for protein
function prediction.  BMC Bioinform. 9, 350 (2008).
</p>

<hr>
<h2 id='sand'>The sand package</h2><span id='topic+sand'></span><span id='topic+C2'></span><span id='topic+C3'></span><span id='topic+C4'></span><span id='topic+C5'></span><span id='topic+C6'></span><span id='topic+C7'></span><span id='topic+C8'></span><span id='topic+C9'></span><span id='topic+C10'></span><span id='topic+C11'></span><span id='topic+N'></span><span id='topic+P'></span>

<h3>Description</h3>

<p>This R package accompanies the book &lsquo;Statistical Analysis of
Network Data with R, 2nd Edition&rsquo;. It contains some of the data sets used in the book (the others are in the <code>igraphdata</code> package). It also
contains the code from the book, and some simple functions to run the
code without the need for typing it in.
</p>


<h3>In brief</h3>

<p>Type in <code>N&lt;enter&gt;</code> to run the next chunk of code,
and <code>C&lt;x&gt;</code> to jump to Chapter <code>x</code>, where <code>x</code> is between
2 and 11. E.g. <code>C6&lt;enter&gt;</code> resets R and &ldquo;loads&rdquo; Chapter
6. <code>P&lt;enter&gt;</code> prints the next code chunk to be run (without
actually running it).
</p>


<h3>The data sets</h3>

<p>The various data sets are loaded from the code chunks in the book. The
sand package contains the following data sets, each is documented in
its on manual page: <code><a href="#topic+Ecoli">Ecoli</a></code>, <code><a href="#topic+aidsblog">aidsblog</a></code>,
<code><a href="#topic+calldata">calldata</a></code>, <code><a href="#topic+elist.lazega">elist.lazega</a></code>,
<code><a href="#topic+fblog">fblog</a></code>, <code><a href="#topic+g.bip">g.bip</a></code>, <code><a href="#topic+hc">hc</a></code>,
<code><a href="#topic+lazega">lazega</a></code>, <code><a href="#topic+ppi.CC">ppi.CC</a></code>,
<code><a href="#topic+sandwichprobe">sandwichprobe</a></code>, <code><a href="#topic+strike">strike</a></code>, <code><a href="#topic+v.attr.lazega">v.attr.lazega</a></code>.
</p>


<h3>Code chunks</h3>

<p>Code chunks of the book are numbered by chapter and each chunk is
identified the chapter number and the chunk number connected by a
dot.
</p>
<p>The reader is supposed to run the code chapter by chapter, ideally,
starting from a clean, new R session. This might not be critical, but
it is not always possible to unload packages in R, so it is the only
way to make sure that the code works correctly.
</p>
<p>To make it easy to step through the code, the sand package define some
&ldquo;commands&rdquo;. Note that these are are not functions, and also
q  that they are meant to be used interactively, and not
programatically.
</p>


<h3>The cursor</h3>

<p>The cursor marks the point the reader is at in the book, and commands
discussed below move the cursor and run the code the cursor is at.
</p>
<p>The &lsquo;C&rsquo; commands clear R, i.e. unload all loaded packages
except for <code>sand</code> and its dependencies, and delete all objects
from the global workspace. They also set the cursor to the first
chunk of the given chapter: there are nine &lsquo;C&rsquo; commands, from
&lsquo;C2&rsquo; to &lsquo;C11&rsquo;, one for each Chapter of the
book. (Chapter 1 has no code to run.)
</p>
<p>The command &lsquo;N&rsquo; runs the chunk at the cursor, and steps the
cursor to the next chunk. It is possible to run multiple chunks at
once, with the form &lsquo;N + x&rsquo; (with or without the spaces), where
&lsquo;x&rsquo; is the number of additional chunks to run. (I.e. &lsquo;N
+ 2&rsquo; runs three chunks.)
</p>
<p>The command &lsquo;P&rsquo; prints the chunk at the cursor, without running
it. It is possible to print other chunks as well: &lsquo;P - 1&rsquo;
prints the previous chunk, &lsquo;P - 2&rsquo; the one before that, etc.,
&lsquo;P + 1&rsquo; prints the next chunk, etc.
</p>
<p>The reader is welcome to inspect R objects, or run arbitrary R code
between the &lsquo;N&rsquo; and &lsquo;P&rsquo; commands.
</p>


<h3>Author(s)</h3>

<p>Gabor Csardi &lt;csardi.gabor@gmail.com&gt;</p>


<h3>See Also</h3>

<p><code><a href="#topic+install_sand_packages">install_sand_packages</a></code> to install all R packages
needed for the book.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Start with Chapter 2
C2

## Run first code chunk
N

## Run next code chunk
N

## Jump to Chapter 5
C5

## Run first code chunk in Chapter 5
## It will create a plot
N
</code></pre>

<hr>
<h2 id='sandwichprobe'>Internet packet probes data</h2><span id='topic+sandwichprobe'></span><span id='topic+delaydata'></span><span id='topic+host.locs'></span>

<h3>Description</h3>

<p>These data correspond to an experiment conducted by Coates et al. to 
measure the difference in delay experienced by packet probes sent over
the Internet during a short period in 2001, from a desktop computer in
the ECE department at Rice University to similar machines at ten other
university locations. The data were intended for use with a newly
proposed method of Internet topology inference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delaydata
host.locs
</code></pre>


<h3>Format</h3>

<p>The data is provided in two files. <code>delaydata</code> is a three-column
data frame. The first columnf is the difference in delay of the small
packets (in milliseconds). The second column is the numeric code of
the destination of small packets. The third column is the numeric code
of the destination of large packet.
</p>
<p><code>host.locs</code> contains the character code of the destinations:
</p>

<ol>
<li> <p>&lsquo;IST&rsquo; Instituto Superior Tecnico (Portugal)
</p>
</li>
<li> <p>&lsquo;IT&rsquo; Instituto de Telecomunicacoes (Portugal)
</p>
</li>
<li> <p>&lsquo;Bkly&rsquo; University of California, Berkeley
</p>
</li>
<li> <p>&lsquo;MSU1&rsquo; Michigan State University (Host 1)
</p>
</li>
<li> <p>&lsquo;MSU2&rsquo; Michigan State University (Host 2)
</p>
</li>
<li> <p>&lsquo;UIUC&rsquo; University of Illinois, Urbana-Champaign
</p>
</li>
<li> <p>&lsquo;UWisc1&rsquo; University of Wisconsin, Madison (Host 1)
</p>
</li>
<li> <p>&lsquo;UWisc2&rsquo; University of Wisconsin, Madison (Host 2)
</p>
</li>
<li> <p>&lsquo;RiceU1&rsquo; Rice University (Host 1)
</p>
</li>
<li> <p>&lsquo;RiceU2&rsquo; Rice University (Host 2)
</p>
</li></ol>



<h3>Source</h3>

<p>Provided by Mark Coates, see reference below.
Please cite the reference below if you use this dataset in your work.
</p>


<h3>References</h3>

<p>M. Coates, R. Castro, R. Nowak, M. Gadhiok, R. King, Y. Tsang, Maximum
likelihood network topology identification from edge-based unicast
measurements. Proceedings of the 2002 ACM SIGMETRICS International
Conference on Measurement and Modeling of Computer Systems, 2002,
pp. 11-20.
</p>

<hr>
<h2 id='strike'>Michael's strike network</h2><span id='topic+strike'></span>

<h3>Description</h3>

<p>A representation of who consults with whom in the context of a strike at a forest products manufacturing facility, following proposed changes to the workers' compensation package.  An edge between two workers indicates that at least one of them said they consult with the other with moderate frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strike
</code></pre>


<h3>Format</h3>

<p>An undirected igraph graph object with 24 vertices and 38 edges.  The graph has two vertex attributes, &lsquo;name&rsquo; is the first name of the individual and &lsquo;race&rsquo; is a compilation of age (young or old) and language spoken (English or Spanish).
</p>


<h3>Source</h3>

<p>This dataset was constructed from the corresponding version in Chapter 7 of W. De Nooy, A. Mrvar, and V. Batagelj, Exploratory Social Network Analysis with Pajek. Cambridge University Press, 2011, vol. 27.
</p>
<p>Please cite the original reference below if you use this dataset in your work.
</p>


<h3>References</h3>

<p>J.H. Michael, &quot;Labor dispute reconciliation in a forest productsmanufacturing   facility,&quot; Forest Products Journal, vol. 47, no. 11/12, p. 41, 1997.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
