<!DOCTYPE html><html lang="en"><head><title>Help for package SAVER</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SAVER}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SAVER-package'><p>SAVER: Single-cell Analysis Via Expression Recovery</p></a></li>
<li><a href='#calc.a'><p>Optimizes variance</p></a></li>
<li><a href='#calc.estimate'><p>Calculate estimate</p></a></li>
<li><a href='#calc.loglik.a'><p>Calculates marginal likelihood</p></a></li>
<li><a href='#calc.maxcor'><p>Calculate maximum correlation</p></a></li>
<li><a href='#calc.post'><p>Calculates SAVER posterior</p></a></li>
<li><a href='#combine.saver'><p>Combines SAVER</p></a></li>
<li><a href='#cor.genes'><p>Calculates gene-to-gene and cell-to-cell SAVER correlation</p></a></li>
<li><a href='#expr.predict'><p>Calculates SAVER prediction.</p></a></li>
<li><a href='#get.mu'><p>Output prior predictions</p></a></li>
<li><a href='#linnarsson'><p>Mouse brain single-cell RNA-seq dataset</p></a></li>
<li><a href='#linnarsson_saver'><p>SAVER recovered mouse brain single-cell RNA-seq dataset</p></a></li>
<li><a href='#sample.saver'><p>Samples from SAVER</p></a></li>
<li><a href='#saver'><p>Runs SAVER</p></a></li>
<li><a href='#saver.fit'><p>Fits SAVER</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Title:</td>
<td>Single-Cell RNA-Seq Gene Expression Recovery</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of a regularized regression prediction and 
    empirical Bayes method to recover the true gene expression profile in 
    noisy and sparse single-cell RNA-seq data. See Huang M, et al (2018) 
    &lt;<a href="https://doi.org/10.1038%2Fs41592-018-0033-z">doi:10.1038/s41592-018-0033-z</a>&gt; for more details.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mo Huang &lt;mohuangx@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet, foreach, methods, iterators, doParallel, Matrix</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.0</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mohuangx/SAVER">https://github.com/mohuangx/SAVER</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mohuangx/SAVER/issues">https://github.com/mohuangx/SAVER/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-11-12 22:32:28 UTC; mohuang</td>
</tr>
<tr>
<td>Author:</td>
<td>Mo Huang [aut, cre],
  Nancy Zhang [aut],
  Mingyao Li [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-13 19:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='SAVER-package'>SAVER: Single-cell Analysis Via Expression Recovery</h2><span id='topic+SAVER-package'></span>

<h3>Description</h3>

<p>The SAVER package implements SAVER, a gene expression recovery method for
single-cell RNA sequencing (scRNA-seq). Borrowing information across all
genes and cells, SAVER provides estimates for true expression levels as well
as posterior distributions to account for estimation uncertainty. See
<a href="https://doi.org/10.1038/s41592-018-0033-z">Huang et al (2018)</a> for
more details.
</p>


<h3>Author(s)</h3>

<p>Mo Huang (Maintainer), <a href="mailto:mohuangx@gmail.com">mohuangx@gmail.com</a>
</p>
<p>Nancy Zhang, <a href="mailto:nzh@wharton.upenn.edu">nzh@wharton.upenn.edu</a>
</p>
<p>Mingyao Li, <a href="mailto:mingyao@pennmedicine.upenn.edu">mingyao@pennmedicine.upenn.edu</a>
</p>


<h3>Source</h3>

<p><a href="https://github.com/mohuangx/SAVER">https://github.com/mohuangx/SAVER</a>
</p>

<hr>
<h2 id='calc.a'>Optimizes variance</h2><span id='topic+calc.a'></span><span id='topic+calc.b'></span><span id='topic+calc.k'></span>

<h3>Description</h3>

<p>Finds the prior parameter that maximizes the marginal likelihood given
the prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.a(y, mu, sf)

calc.b(y, mu, sf)

calc.k(y, mu, sf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.a_+3A_y">y</code></td>
<td>
<p>A vector of observed gene counts.</p>
</td></tr>
<tr><td><code id="calc.a_+3A_mu">mu</code></td>
<td>
<p>A vector of predictions from <code><a href="#topic+expr.predict">expr.predict</a></code>.</p>
</td></tr>
<tr><td><code id="calc.a_+3A_sf">sf</code></td>
<td>
<p>Vector of normalized size factors.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>calc.a</code> returns a prior alpha parameter assuming constant
coefficient of variation. <code>calc.b</code> returns a prior beta parameter
assuming constant Fano factor. <code>calc.k</code> returns a prior variance
parameter assuming constant variance.
</p>


<h3>Value</h3>

<p>A vector with the optimized parameter and the negative
log-likelihood.
</p>

<hr>
<h2 id='calc.estimate'>Calculate estimate</h2><span id='topic+calc.estimate'></span><span id='topic+calc.estimate.mean'></span><span id='topic+calc.estimate.null'></span>

<h3>Description</h3>

<p>Calculates SAVER estimate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.estimate(
  x,
  x.est,
  cutoff = 0,
  coefs = NULL,
  sf,
  scale.sf,
  pred.gene.names,
  pred.cells,
  null.model,
  nworkers,
  calc.maxcor,
  estimates.only
)

calc.estimate.mean(x, sf, scale.sf, mu, nworkers, estimates.only)

calc.estimate.null(x, sf, scale.sf, nworkers, estimates.only)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.estimate_+3A_x">x</code></td>
<td>
<p>An expression count matrix. The rows correspond to genes and
the columns correspond to cells.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_x.est">x.est</code></td>
<td>
<p>The log-normalized predictor matrix. The rows correspond to
cells and the columns correspond to genes.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_cutoff">cutoff</code></td>
<td>
<p>Maximum absolute correlation to determine whether a gene
should be predicted.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_coefs">coefs</code></td>
<td>
<p>Coefficients of a linear fit of log-squared ratio of
largest lambda to lambda of lowest cross-validation error. Used to estimate
model with lowest cross-validation error.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_sf">sf</code></td>
<td>
<p>Normalized size factor.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_scale.sf">scale.sf</code></td>
<td>
<p>Scale of size factor.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_pred.gene.names">pred.gene.names</code></td>
<td>
<p>Names of genes to perform regression prediction.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_pred.cells">pred.cells</code></td>
<td>
<p>Index of cells to perform regression prediction.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_null.model">null.model</code></td>
<td>
<p>Whether to use mean gene expression as prediction.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_nworkers">nworkers</code></td>
<td>
<p>Number of cores registered to parallel backend.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_calc.maxcor">calc.maxcor</code></td>
<td>
<p>Whether to calculate maximum absolute correlation.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_estimates.only">estimates.only</code></td>
<td>
<p>Only return SAVER estimates. Default is FALSE.</p>
</td></tr>
<tr><td><code id="calc.estimate_+3A_mu">mu</code></td>
<td>
<p>Matrix of prior means</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SAVER method starts by estimating the prior mean and variance for the
true expression level for each gene and cell. The prior mean is obtained
through predictions from a LASSO Poisson regression for each gene
implemented using the <code>glmnet</code> package. Then, the variance is estimated
through maximum likelihood assuming constant variance, Fano factor, or
coefficient of variation variance structure for each gene. The posterior
distribution is calculated and the posterior mean is reported as the SAVER
estimate.
</p>


<h3>Value</h3>

<p>A list with the following components
</p>
<table role = "presentation">
<tr><td><code>est</code></td>
<td>
<p>Recovered (normalized) expression</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>Standard error of estimates</p>
</td></tr>
<tr><td><code>maxcor</code></td>
<td>
<p>Maximum absolute correlation for each gene. 2 if not
calculated</p>
</td></tr>
<tr><td><code>lambda.max</code></td>
<td>
<p>Smallest value of lambda which gives the null
model.</p>
</td></tr>
<tr><td><code>lambda.min</code></td>
<td>
<p>Value of lambda from which the prediction model is
used</p>
</td></tr>
<tr><td><code>sd.cv</code></td>
<td>
<p>Difference in the number of standard deviations in
deviance between the model with lowest cross-validation error and the null
model</p>
</td></tr>
<tr><td><code>ct</code></td>
<td>
<p>Time taken to generate predictions.</p>
</td></tr>
<tr><td><code>vt</code></td>
<td>
<p>Time taken to estimate variance.</p>
</td></tr>
</table>

<hr>
<h2 id='calc.loglik.a'>Calculates marginal likelihood</h2><span id='topic+calc.loglik.a'></span><span id='topic+calc.loglik.b'></span><span id='topic+calc.loglik.k'></span>

<h3>Description</h3>

<p>Calculates the marginal likelihood given the prediction under constant
coefficient of variation (a), Fano factor (b), and variance (k).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.loglik.a(a, y, mu, sf)

calc.loglik.b(b, y, mu, sf)

calc.loglik.k(k, y, mu, sf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.loglik.a_+3A_a">a</code>, <code id="calc.loglik.a_+3A_b">b</code>, <code id="calc.loglik.a_+3A_k">k</code></td>
<td>
<p>Prior parameter.</p>
</td></tr>
<tr><td><code id="calc.loglik.a_+3A_y">y</code></td>
<td>
<p>A vector of observed gene counts.</p>
</td></tr>
<tr><td><code id="calc.loglik.a_+3A_mu">mu</code></td>
<td>
<p>A vector of predictions from <code><a href="#topic+expr.predict">expr.predict</a></code>.</p>
</td></tr>
<tr><td><code id="calc.loglik.a_+3A_sf">sf</code></td>
<td>
<p>Vector of normalized size factors.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>calc.loglik.a</code> returns the shifted negative log-likelihood under
constant coefficient of variation.
<code>calc.loglik.b</code> returns the shifted negative log-likelihood under
constant Fano factor.
<code>calc.loglik.k</code> returns the shifted negative log-likelihood under
constant variance.
</p>


<h3>Value</h3>

<p>A shifted negative marginal log-likelihood.
</p>

<hr>
<h2 id='calc.maxcor'>Calculate maximum correlation</h2><span id='topic+calc.maxcor'></span>

<h3>Description</h3>

<p>Calculates the maximum absolute correlation between two matrices along the
columns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.maxcor(x1, x2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.maxcor_+3A_x1">x1</code></td>
<td>
<p>Named matrix 1</p>
</td></tr>
<tr><td><code id="calc.maxcor_+3A_x2">x2</code></td>
<td>
<p>Named matrix 2</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the maximum absolute correlation for each column of
<code>x2</code> against each column of <code>x1</code>. The matrices are named and if
the names overlap between <code>x1</code> and <code>x2</code>, the correlation between
the same named entries is set to zero.
</p>


<h3>Value</h3>

<p>A vector of maximum absolute correlations
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x1 &lt;- matrix(rnorm(500), 100, 5)
x2 &lt;- x1 + matrix(rnorm(500), 100, 5)
colnames(x1) &lt;- c("A", "B", "C", "D", "E")
colnames(x2) &lt;- c("A", "B", "C", "D", "E")
cor(x1, x2)
calc.maxcor(x1, x2)

</code></pre>

<hr>
<h2 id='calc.post'>Calculates SAVER posterior</h2><span id='topic+calc.post'></span>

<h3>Description</h3>

<p>Given prediction and prior variance, calculates the Gamma posterior
distribution parameters for a single gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.post(y, mu, sf, scale.sf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.post_+3A_y">y</code></td>
<td>
<p>A vector of observed gene counts.</p>
</td></tr>
<tr><td><code id="calc.post_+3A_mu">mu</code></td>
<td>
<p>A vector of prior means.</p>
</td></tr>
<tr><td><code id="calc.post_+3A_sf">sf</code></td>
<td>
<p>Vector of normalized size factors.</p>
</td></tr>
<tr><td><code id="calc.post_+3A_scale.sf">scale.sf</code></td>
<td>
<p>Mean of the original size factors.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">\alpha</code> be the shape parameter and <code class="reqn">\beta</code> be the rate
parameter of the prior Gamma distribution. Then, the posterior Gamma
distribution will be
</p>
<p style="text-align: center;"><code class="reqn">Gamma(y + \alpha, sf + \beta),</code>
</p>

<p>where y is the observed gene count and sf is the size factor.
</p>


<h3>Value</h3>

<p>A list with the following components
</p>
<table role = "presentation">
<tr><td><code>estimate</code></td>
<td>
<p>Recovered (normalized) expression</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>Standard error of expression estimate</p>
</td></tr>
</table>

<hr>
<h2 id='combine.saver'>Combines SAVER</h2><span id='topic+combine.saver'></span>

<h3>Description</h3>

<p>Combines SAVER objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine.saver(saver.list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combine.saver_+3A_saver.list">saver.list</code></td>
<td>
<p>List of SAVER objects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If SAVER was applied to a dataset for chunks of genes (by specifying
<code>pred.genes</code> and <code>pred.genes.only = TRUE</code>), this function combines
the individual SAVER objects into one SAVER object.
</p>


<h3>Value</h3>

<p>A combined SAVER object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("linnarsson")

## Not run: 
a &lt;- saver(linnarsson, pred.genes = 1:5, pred.genes.only = TRUE)
b &lt;- saver(linnarsson, pred.genes = 6:10, pred.genes.only = TRUE)
ab &lt;- combine.saver(list(a, b))

## End(Not run)

</code></pre>

<hr>
<h2 id='cor.genes'>Calculates gene-to-gene and cell-to-cell SAVER correlation</h2><span id='topic+cor.genes'></span><span id='topic+cor.cells'></span>

<h3>Description</h3>

<p>Adjusts for SAVER estimation uncertainty by calculating and adjusting
gene-to-gene and cell-to-cell correlation matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cor.genes(x, cor.mat = NULL)

cor.cells(x, cor.mat = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cor.genes_+3A_x">x</code></td>
<td>
<p>A <code>saver</code> object.</p>
</td></tr>
<tr><td><code id="cor.genes_+3A_cor.mat">cor.mat</code></td>
<td>
<p>If a correlation matrix of the SAVER estimates was already
obtained, then it can be provided as an input to avoid recomputation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SAVER estimates that are produced have varying levels of uncertainty
depending on the gene and the cell. These functions adjust the gene-to-gene
and cell-to-cell correlations of the SAVER estimates to reflect the
estimation uncertainty.
</p>


<h3>Value</h3>

<p>An adjusted correlation matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("linnarsson_saver")

gene.cor &lt;- cor.genes(linnarsson_saver)

</code></pre>

<hr>
<h2 id='expr.predict'>Calculates SAVER prediction.</h2><span id='topic+expr.predict'></span>

<h3>Description</h3>

<p>Uses <code>cv.glmnet</code> from the <code>glmnet</code> package to return the SAVER
prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expr.predict(
  x,
  y,
  pred.cells = 1:length(y),
  seed = NULL,
  lambda.max = NULL,
  lambda.min = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expr.predict_+3A_x">x</code></td>
<td>
<p>A log-normalized expression count matrix of genes to be used in the
prediction.</p>
</td></tr>
<tr><td><code id="expr.predict_+3A_y">y</code></td>
<td>
<p>A normalized expression count vector of the gene to be predicted.</p>
</td></tr>
<tr><td><code id="expr.predict_+3A_pred.cells">pred.cells</code></td>
<td>
<p>Index of cells to use for prediction. Default is to use
all cells.</p>
</td></tr>
<tr><td><code id="expr.predict_+3A_seed">seed</code></td>
<td>
<p>Sets the seed for reproducible results.</p>
</td></tr>
<tr><td><code id="expr.predict_+3A_lambda.max">lambda.max</code></td>
<td>
<p>Maximum value of lambda which gives null model.</p>
</td></tr>
<tr><td><code id="expr.predict_+3A_lambda.min">lambda.min</code></td>
<td>
<p>Value of lambda from which the prediction model is
used</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SAVER method starts with predicting the prior mean for each cell for a
specific gene. The prediction is performed using the observed normalized
gene count as the response and the normalized gene counts of other genes as
predictors. <code>cv.glmnet</code> from the <code>glmnet</code> package is used to fit
the LASSO Poisson regression. The model with the lowest cross-validation
error is chosen and the fitted response values are returned and used as the
SAVER prediction.
</p>


<h3>Value</h3>

<p>A vector of predicted gene expression.
</p>

<hr>
<h2 id='get.mu'>Output prior predictions</h2><span id='topic+get.mu'></span>

<h3>Description</h3>

<p>Outputs prior predictions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.mu(x, saver.obj)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.mu_+3A_x">x</code></td>
<td>
<p>Original count matrix.</p>
</td></tr>
<tr><td><code id="get.mu_+3A_saver.obj">saver.obj</code></td>
<td>
<p>SAVER output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function outputs prior mean predictions <code class="reqn">\mu</code> used in fitting the 
SAVER model.
</p>


<h3>Value</h3>

<p>A matrix of prior mean predictions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("linnarsson")
data("linnarsson_saver")

mu &lt;- get.mu(linnarsson, linnarsson_saver)

</code></pre>

<hr>
<h2 id='linnarsson'>Mouse brain single-cell RNA-seq dataset</h2><span id='topic+linnarsson'></span>

<h3>Description</h3>

<p>3,529 genes and 200 cells from a mouse brain scRNA-seq dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linnarsson
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> with 3529 rows and 200 columns.</p>


<h3>References</h3>

<p>Zeisel, A., Munoz-Manchado, A. B., Codeluppi, S.,
Lonnerberg, P., La Manno, G., Jureus, A., ... Linnarsson, S. (2015). Cell
types in the mouse cortex and hippocampus revealed by single-cell RNA-seq.
<em>Science, 347</em>(6226), 1138-1142.
</p>

<hr>
<h2 id='linnarsson_saver'>SAVER recovered mouse brain single-cell RNA-seq dataset</h2><span id='topic+linnarsson_saver'></span>

<h3>Description</h3>

<p>Output of running 'saver' on the 'linnarsson' dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linnarsson_saver
</code></pre>


<h3>Format</h3>

<p>An object of class <code>saver</code> of length 3.</p>


<h3>References</h3>

<p>Zeisel, A., Munoz-Manchado, A. B., Codeluppi, S.,
Lonnerberg, P., La Manno, G., Jureus, A., ... Linnarsson, S. (2015). Cell
types in the mouse cortex and hippocampus revealed by single-cell RNA-seq.
<em>Science, 347</em>(6226), 1138-1142.
</p>

<hr>
<h2 id='sample.saver'>Samples from SAVER</h2><span id='topic+sample.saver'></span>

<h3>Description</h3>

<p>Samples from the posterior distribution output by SAVER.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample.saver(x, rep = 1, efficiency.known = FALSE, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample.saver_+3A_x">x</code></td>
<td>
<p>A <code>saver</code> object.</p>
</td></tr>
<tr><td><code id="sample.saver_+3A_rep">rep</code></td>
<td>
<p>Number of sampled datasets. Default is 1.</p>
</td></tr>
<tr><td><code id="sample.saver_+3A_efficiency.known">efficiency.known</code></td>
<td>
<p>Whether the efficiency is known. Default is
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="sample.saver_+3A_seed">seed</code></td>
<td>
<p>seed used in <code>set.seed</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SAVER method outputs a posterior distribution, which we can sample from
for downstream analysis. The posterior distribution accounts for uncertainty
in the SAVER estimation procedure. If the efficiency is known,
negative binomial sampling is performed; otherwise, gamma sampling is
performed.
</p>


<h3>Value</h3>

<p>A matrix of expression values sampled from SAVER posterior. If
<code>rep</code> &gt; 1, a list of matrices is returned
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("linnarsson_saver")

samp1 &lt;- sample.saver(linnarsson_saver, seed = 50)


</code></pre>

<hr>
<h2 id='saver'>Runs SAVER</h2><span id='topic+saver'></span>

<h3>Description</h3>

<p>Recovers expression using the SAVER method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saver(
  x,
  do.fast = TRUE,
  ncores = 1,
  size.factor = NULL,
  npred = NULL,
  pred.cells = NULL,
  pred.genes = NULL,
  pred.genes.only = FALSE,
  null.model = FALSE,
  mu = NULL,
  estimates.only = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="saver_+3A_x">x</code></td>
<td>
<p>An expression count matrix. The rows correspond to genes and
the columns correspond to cells. Can be sparse.</p>
</td></tr>
<tr><td><code id="saver_+3A_do.fast">do.fast</code></td>
<td>
<p>Approximates the prediction step. Default is TRUE.</p>
</td></tr>
<tr><td><code id="saver_+3A_ncores">ncores</code></td>
<td>
<p>Number of cores to use. Default is 1.</p>
</td></tr>
<tr><td><code id="saver_+3A_size.factor">size.factor</code></td>
<td>
<p>Vector of cell size normalization factors.
If <code>x</code> is already normalized or normalization is not desired, use
<code>size.factor = 1</code>. Default uses mean library size normalization.</p>
</td></tr>
<tr><td><code id="saver_+3A_npred">npred</code></td>
<td>
<p>Number of genes for regression prediction. Selects the top
<code>npred</code> genes in terms of mean expression for regression prediction.
Default is all genes.</p>
</td></tr>
<tr><td><code id="saver_+3A_pred.cells">pred.cells</code></td>
<td>
<p>Indices of cells to perform regression prediction.
Default is all cells.</p>
</td></tr>
<tr><td><code id="saver_+3A_pred.genes">pred.genes</code></td>
<td>
<p>Indices of specific genes to perform regression
prediction. Overrides <code>npred</code>. Default is all genes.</p>
</td></tr>
<tr><td><code id="saver_+3A_pred.genes.only">pred.genes.only</code></td>
<td>
<p>Return expression levels of only <code>pred.genes</code>.
Default is FALSE (returns expression levels of all genes).</p>
</td></tr>
<tr><td><code id="saver_+3A_null.model">null.model</code></td>
<td>
<p>Whether to use mean gene expression as prediction.</p>
</td></tr>
<tr><td><code id="saver_+3A_mu">mu</code></td>
<td>
<p>Matrix of prior means.</p>
</td></tr>
<tr><td><code id="saver_+3A_estimates.only">estimates.only</code></td>
<td>
<p>Only return SAVER estimates. Default is FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SAVER method starts by estimating the prior mean and variance for the
true expression level for each gene and cell. The prior mean is obtained
through predictions from a LASSO Poisson regression for each gene
implemented using the <code>glmnet</code> package. Then, the variance is estimated
through maximum likelihood assuming constant variance, Fano factor, or
coefficient of variation variance structure for each gene. The posterior
distribution is calculated and the posterior mean is reported as the SAVER
estimate.
</p>


<h3>Value</h3>

<p>If 'estimates.only = TRUE', then a matrix of SAVER estimates.
</p>
<p>If 'estimates.only = FALSE', a list with the following components
</p>
<table role = "presentation">
<tr><td><code>estimate</code></td>
<td>
<p>Recovered (normalized) expression.</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>Standard error of estimates.</p>
</td></tr>
<tr><td><code>info</code></td>
<td>
<p>Information about dataset.</p>
</td></tr>
</table>
<p>The <code>info</code> element is a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>size.factor</code></td>
<td>
<p>Size factor used for normalization.</p>
</td></tr>
<tr><td><code>maxcor</code></td>
<td>
<p>Maximum absolute correlation for each gene. 2 if not
calculated</p>
</td></tr>
<tr><td><code>lambda.max</code></td>
<td>
<p>Smallest value of lambda which gives the null
model.</p>
</td></tr>
<tr><td><code>lambda.min</code></td>
<td>
<p>Value of lambda from which the prediction model is
used</p>
</td></tr>
<tr><td><code>sd.cv</code></td>
<td>
<p>Difference in the number of standard deviations in
deviance between the model with lowest cross-validation error and the null
model</p>
</td></tr>
<tr><td><code>pred.time</code></td>
<td>
<p>Time taken to generate predictions.</p>
</td></tr>
<tr><td><code>var.time</code></td>
<td>
<p>Time taken to estimate variance.</p>
</td></tr>
<tr><td><code>maxcor</code></td>
<td>
<p>Maximum absolute correlation cutoff used to determine
if a gene should be predicted.</p>
</td></tr>
<tr><td><code>lambda.coefs</code></td>
<td>
<p>Coefficients for estimating lambda with lowest
cross-validation error.</p>
</td></tr>
<tr><td><code>total.time</code></td>
<td>
<p>Total time for SAVER estimation.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data("linnarsson")

## Not run: 
system.time(linnarsson_saver &lt;- saver(linnarsson, ncores = 12))

## End(Not run)

# predictions for top 5 highly expressed genes
## Not run: 
saver2 &lt;- saver(linnarsson, npred = 5)

## End(Not run)

# predictions for certain genes
## Not run: 
genes &lt;- c("Thy1", "Mbp", "Stim2", "Psmc6", "Rps19")
genes.ind &lt;- which(rownames(linnarsson) 
saver3 &lt;- saver(linnarsson, pred.genes = genes.ind)

## End(Not run)

# return only certain genes
## Not run: 
saver4 &lt;- saver(linnarsson, pred.genes = genes.ind, pred.genes.only = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='saver.fit'>Fits SAVER</h2><span id='topic+saver.fit'></span><span id='topic+saver.fit.mean'></span><span id='topic+saver.fit.null'></span>

<h3>Description</h3>

<p>Fits SAVER object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saver.fit(
  x,
  x.est,
  do.fast,
  ncores,
  sf,
  scale.sf,
  pred.genes,
  pred.cells,
  null.model,
  ngenes = nrow(x),
  ncells = ncol(x),
  gene.names = rownames(x),
  cell.names = colnames(x),
  estimates.only
)

saver.fit.mean(
  x,
  ncores,
  sf,
  scale.sf,
  mu,
  ngenes = nrow(x),
  ncells = ncol(x),
  gene.names = rownames(x),
  cell.names = colnames(x),
  estimates.only
)

saver.fit.null(
  x,
  ncores,
  sf,
  scale.sf,
  ngenes = nrow(x),
  ncells = ncol(x),
  gene.names = rownames(x),
  cell.names = colnames(x),
  estimates.only
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="saver.fit_+3A_x">x</code></td>
<td>
<p>An expression count matrix. The rows correspond to genes and
the columns correspond to cells.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_x.est">x.est</code></td>
<td>
<p>The log-normalized predictor matrix. The rows correspond to
cells and the columns correspond to genes.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_do.fast">do.fast</code></td>
<td>
<p>Approximates the prediction step. Default is TRUE.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_ncores">ncores</code></td>
<td>
<p>Number of cores to use. Default is 1.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_sf">sf</code></td>
<td>
<p>Normalized size factor.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_scale.sf">scale.sf</code></td>
<td>
<p>Scale of size factor.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_pred.genes">pred.genes</code></td>
<td>
<p>Index of genes to perform regression prediction.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_pred.cells">pred.cells</code></td>
<td>
<p>Index of cells to perform regression prediction.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_null.model">null.model</code></td>
<td>
<p>Whether to use mean gene expression as prediction.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_ngenes">ngenes</code></td>
<td>
<p>Number of genes.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_ncells">ncells</code></td>
<td>
<p>Number of cells.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_gene.names">gene.names</code></td>
<td>
<p>Name of genes.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_cell.names">cell.names</code></td>
<td>
<p>Name of cells.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_estimates.only">estimates.only</code></td>
<td>
<p>Only return SAVER estimates. Default is FALSE.</p>
</td></tr>
<tr><td><code id="saver.fit_+3A_mu">mu</code></td>
<td>
<p>Matrix of prior means.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SAVER method starts by estimating the prior mean and variance for the
true expression level for each gene and cell. The prior mean is obtained
through predictions from a Lasso Poisson regression for each gene
implemented using the <code>glmnet</code> package. Then, the variance is estimated
through maximum likelihood assuming constant variance, Fano factor, or
coefficient of variation variance structure for each gene. The posterior
distribution is calculated and the posterior mean is reported as the SAVER
estimate.
</p>


<h3>Value</h3>

<p>A list with the following components
</p>
<table role = "presentation">
<tr><td><code>estimate</code></td>
<td>
<p>Recovered (normalized) expression</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>Standard error of estimates</p>
</td></tr>
<tr><td><code>info</code></td>
<td>
<p>Information about fit</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
