<!DOCTYPE html><html><head><title>Help for package sbl</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sbl}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gen'><p>Genotype</p></a></li>
<li><a href='#intercept'><p>Intercept</p></a></li>
<li><a href='#map'><p>Genetic map</p></a></li>
<li><a href='#phe'><p>Phenotype</p></a></li>
<li><a href='#sblgwas'><p>Sparse Bayesian Learning for QTL Mapping and Genome-Wide Association Studies</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Sparse Bayesian Learning for QTL Mapping and Genome-Wide
Association Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Meiyue Wang and Shizhong Xu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Meiyue Wang &lt;mwang024@ucr.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements sparse Bayesian learning method for QTL mapping and genome-wide association studies.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>stats, R (&ge; 2.10)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-10-17</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-10-17 19:29:53 UTC; Dell</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-10-26 15:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='gen'>Genotype</h2><span id='topic+gen'></span>

<h3>Description</h3>

<p>A  matrix with dimension of 30*50 containing the numeric genotype
indicator for 30 simulated individuals of an F2 family generated from the cross
of two inbred lines according to <code>map</code> provided in the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen
</code></pre>


<h3>Format</h3>

<p>A matrix with 30 rows and 50 columns:
</p>

<dl>
<dt>ID</dt><dd><p>individual ID</p>
</dd>
<dt>marker</dt><dd><p>name of marker</p>
</dd>
</dl>

<hr>
<h2 id='intercept'>Intercept</h2><span id='topic+intercept'></span>

<h3>Description</h3>

<p>Design vector of the intercept for the 30 simulated individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intercept
</code></pre>


<h3>Format</h3>

<p>A sequence of number containing 30 replicates of number 1.</p>

<hr>
<h2 id='map'>Genetic map</h2><span id='topic+map'></span>

<h3>Description</h3>

<p>The genetic map used in the generation of numeric genotype indicators
for 30 simulated individuals of an F2 family generated from the cross of two
inbred lines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map
</code></pre>


<h3>Format</h3>

<p>A data frame with 50 rows and 4 variables:
</p>

<dl>
<dt>chr</dt><dd><p>chromosome</p>
</dd>
<dt>marker</dt><dd><p>name of marker</p>
</dd>
<dt>cm</dt><dd><p>genetic distance measured in centi-Morgan</p>
</dd>
<dt>effect</dt><dd><p>simulated effect assigned to marker in the simulation experiment</p>
</dd>
</dl>

<hr>
<h2 id='phe'>Phenotype</h2><span id='topic+phe'></span>

<h3>Description</h3>

<p>Observations of the phenotype for 30 simulated individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phe
</code></pre>


<h3>Format</h3>

<p>A sequence of number with length equal to 30.</p>

<hr>
<h2 id='sblgwas'>Sparse Bayesian Learning for QTL Mapping and Genome-Wide Association Studies</h2><span id='topic+sblgwas'></span>

<h3>Description</h3>

<p>The sparse Bayesian learning (SBL) method for quantitative trait
locus (QTL) mapping and genome-wide association studies (GWAS)
deals with a linear mixed model. This is also
a multiple locus model that includes all markers (random effects) in a single
model and detect significant markers simultaneously. SBL method adopts
coordinate descent algorithm to update parameters by estimating one parameter
at a time conditional on the posterior modes of all other parameters. The parameter
estimation process requires multiple iterations and the final estimated
parameters take the values when the entire program converges.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sblgwas(x, y, z, t = -1, max.iter = 200, min.err = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sblgwas_+3A_x">x</code></td>
<td>
<p>a design matrix for fixed effects</p>
</td></tr>
<tr><td><code id="sblgwas_+3A_y">y</code></td>
<td>
<p>a vector of response variables</p>
</td></tr>
<tr><td><code id="sblgwas_+3A_z">z</code></td>
<td>
<p>a design matrix for random effects</p>
</td></tr>
<tr><td><code id="sblgwas_+3A_t">t</code></td>
<td>
<p>a number between [-2,0] to control sparseness of the model,
default is -1.</p>
</td></tr>
<tr><td><code id="sblgwas_+3A_max.iter">max.iter</code></td>
<td>
<p>maximum number of iterations set by user to stop the program,
default is 200.</p>
</td></tr>
<tr><td><code id="sblgwas_+3A_min.err">min.err</code></td>
<td>
<p>minimum threshold of mean squared error of random effects estimated
from the current and the previous iteration to stop the program, default is 1e-6.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The multiple locus hierarchical linear mixed model of SBL is
</p>
<p style="text-align: center;"><code class="reqn">y=X\beta+Z\gamma+\epsilon</code>
</p>

<p>where y is an <code class="reqn">n*1</code> vector of response variables (observations of the trait);
X is an <code class="reqn">n*p</code> design matrix for fixed effects; <code class="reqn">\beta</code> is a <code class="reqn">p*1</code>
vector of fixed effect; Z is an <code class="reqn">n*m</code> genotype indicator matrix;
<code class="reqn">\gamma</code> is an <code class="reqn">m*1</code> vector of marker effects and <code class="reqn">\epsilon</code> is
an <code class="reqn">n*1</code> vector of residual errors with an aassumed
<code class="reqn">\epsilon~N(0,\Sigma)</code> distribution. Each marker effect, <code class="reqn">\gamma[k]</code>
for marker <em>k</em>, is treated as a random variable following <code class="reqn">N(0,\Phi[k])</code>
distribution, where <code class="reqn">\Phi[k]</code> is the prior variance. The estimate of
<code class="reqn">\gamma[k]</code> is best linear unbiased prediciton (BLUP). The estimate of
<code class="reqn">\Phi[k]</code> is miximum likelihood estimate (MLE).
</p>


<h3>Value</h3>

<table>
<tr><td><code>iteration</code></td>
<td>
<p>a matrix storing intermediate results of each iteration before
the entire program converges, including <br />
</p>
<p>&quot;<code>iter</code>&quot; iteration indicator <br />
</p>
<p>&quot;<code>error</code>&quot; mean squared error of random effects estimated from the current and
the previous iteration <br />
</p>
<p>&quot;<code>s2</code>&quot; estimated variance of residual error <br />
</p>
<p>&quot;<code class="reqn">beta[1]\dots beta[p]</code>&quot; estimates of fixed effects <br />
</p>
<p>&quot;<code class="reqn">gamma[1]\dots gamma[m]</code>&quot; estimates of random effects</p>
</td></tr>
<tr><td><code>parm</code></td>
<td>
<p>a vector containing 5 elements: &quot;<code>iter</code>&quot;, &quot;<code>error</code>&quot;,
&quot;<code>s2</code>&quot;, &quot;<code>beta</code>&quot; and &quot;<code>df</code>&quot; <br />
</p>
<p>&quot;<code>iter</code>&quot; the number of iterations required by program to stop <br />
</p>
<p>&quot;<code>error</code>&quot; mean square error of random effect estimated from the last iteration
before the program stops <br />
</p>
<p>&quot;<code>s2</code>&quot; estimated variance of residual error <br />
</p>
<p>&quot;<code>beta</code>&quot; estimate of fixed effect <br />
</p>
<p>&quot;<code>df</code>&quot; the effective degree of freedom from total random effects</p>
</td></tr>
<tr><td><code>blup</code></td>
<td>
<p>a matrix containing 4 columns: &quot;<code>gamma</code>&quot;, &quot;<code>vg</code>&quot;,
&quot;<code>wald</code>&quot; and &quot;<code>p_wald</code>&quot; <br />
</p>
<p>&quot;<code>gamma</code>&quot; estimate of random effect <br />
</p>
<p>&quot;<code>vg</code>&quot; estimated variance of random effect <br />
</p>
<p>&quot;<code>wald</code>&quot; Wald statistic calculated as <code class="reqn">\gamma^2</code>/<code class="reqn">\Phi</code>
</p>
<p>&quot;<code>p_wald</code>&quot; the <em>p</em>-value of Wald statistic following Chi-squared
distribution with 1 degree of freedom</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Meiyue Wang and Shizhong Xu <br />
Maintainer: Meiyue Wang <a href="mailto:mwang024@ucr.edu">mwang024@ucr.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data from sbl package
data(gen)
data(phe)
data(intercept)

# Run sblgwas() to perform association study of example data
# setting t = 0 leads to the most sparse model
fit&lt;-sblgwas(x=intercept, y=phe, z=gen, t=0)
my.blup&lt;-fit$blup

# setting t = -2 leads to the least sparse model
fit&lt;-sblgwas(x=intercept, y=phe, z=gen, t=-2)
my.blup&lt;-fit$blup
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
