<!DOCTYPE html><html><head><title>Help for package ScaleSpikeSlab</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ScaleSpikeSlab}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#riboflavin'><p>Riboflavin GWAS dataset</p></a></li>
<li><a href='#spike_slab_linear'><p>spike_slab_linear</p></a></li>
<li><a href='#spike_slab_logistic'><p>spike_slab_logistic</p></a></li>
<li><a href='#spike_slab_params'><p>spike_slab_params</p></a></li>
<li><a href='#spike_slab_probit'><p>spike_slab_probit</p></a></li>
<li><a href='#synthetic_data'><p>synthetic_data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Scalable Spike-and-Slab</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-13</td>
</tr>
<tr>
<td>Description:</td>
<td>A scalable Gibbs sampling implementation for high dimensional Bayesian regression with the continuous spike-and-slab prior. Niloy Biswas, Lester Mackey and Xiao-Li Meng, "Scalable Spike-and-Slab" (2022) &lt;<a href="https://doi.org/10.48550/arXiv.2204.01668">doi:10.48550/arXiv.2204.01668</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, stats, TruncatedNormal</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-17 20:40:23 UTC; niloybiswas</td>
</tr>
<tr>
<td>Author:</td>
<td>Niloy Biswas <a href="https://orcid.org/0000-0001-9081-5702"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Lester Mackey [aut],
  Xiao-Li Meng [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Niloy Biswas &lt;niloy_biswas@g.harvard.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-18 17:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='riboflavin'>Riboflavin GWAS dataset</h2><span id='topic+riboflavin'></span>

<h3>Description</h3>

<p>Dataset of riboflavin production by Bacillus subtilis containing n = 71 observations 
of a one-dimensional response (riboflavin production) and p = 4088 predictors (gene expressions).
The one-dimensional response corresponds to riboflavin production.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(riboflavin)
</code></pre>


<h3>Format</h3>

<p>A data frame containing a vector y of length 71 (responses) and 
a matrix X of dimension 71 by 4088 (gene expressions)
</p>


<h3>Details</h3>

<p>The processed dataset is the same as in the R packages qut and hdi.
</p>


<h3>References</h3>

<p>Buhlmann, P., Kalisch, M. and Meier, L. (2014) 
<em>High-dimensional statistics with a view towards
applications in biology</em>. Annual Review of Statistics and
its Applications <b>1</b>, 255&ndash;278
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(riboflavin)
y &lt;- as.vector(riboflavin$y)
X &lt;- as.matrix(riboflavin$x)
</code></pre>

<hr>
<h2 id='spike_slab_linear'>spike_slab_linear</h2><span id='topic+spike_slab_linear'></span>

<h3>Description</h3>

<p>Generates Markov chain targeting the posterior corresponding to
Bayesian linear regression with spike and slab priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spike_slab_linear(
  chain_length,
  X,
  y,
  tau0,
  tau1,
  q,
  a0 = 1,
  b0 = 1,
  rinit = NULL,
  verbose = FALSE,
  burnin = 0,
  store = TRUE,
  Xt = NULL,
  XXt = NULL,
  tau0_inverse = NULL,
  tau1_inverse = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spike_slab_linear_+3A_chain_length">chain_length</code></td>
<td>
<p>Markov chain length</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_x">X</code></td>
<td>
<p>matrix of length n by p</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_y">y</code></td>
<td>
<p>Response</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_tau0">tau0</code></td>
<td>
<p>prior hyperparameter (non-negative real)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_tau1">tau1</code></td>
<td>
<p>prior hyperparameter (non-negative real)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_q">q</code></td>
<td>
<p>prior hyperparameter (strictly between 0 and 1)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_a0">a0</code></td>
<td>
<p>prior hyperparameter (non-negative real)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_b0">b0</code></td>
<td>
<p>prior hyperparameter (non-negative real)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_rinit">rinit</code></td>
<td>
<p>initial distribution of Markov chain (default samples from the prior)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_verbose">verbose</code></td>
<td>
<p>print iteration of the Markov chain (boolean)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_burnin">burnin</code></td>
<td>
<p>chain burnin (non-negative integer)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_store">store</code></td>
<td>
<p>store chain trajectory (boolean)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_xt">Xt</code></td>
<td>
<p>Pre-calculated transpose of X</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_xxt">XXt</code></td>
<td>
<p>Pre-calculated matrix X*transpose(X) (n by n matrix)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_tau0_inverse">tau0_inverse</code></td>
<td>
<p>Pre-calculated matrix inverse(I + tau0^2*XXt) (n by n matrix)</p>
</td></tr>
<tr><td><code id="spike_slab_linear_+3A_tau1_inverse">tau1_inverse</code></td>
<td>
<p>Pre-calculated matrix inverse(I + tau1^2*XXt) (n by n matrix)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output from Markov chain targeting the posterior corresponding to
Bayesian linear regression with spike and slab priors
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Synthetic dataset
syn_data &lt;- synthetic_data(n=100,p=200,s0=5,error_std=2,type='linear')
X &lt;- syn_data$X
y &lt;- syn_data$y

# Hyperparamters
params &lt;- spike_slab_params(n=nrow(X),p=ncol(X))

# Run S^3
sss_chain &lt;- spike_slab_linear(chain_length=4e3,burnin=1e3,X=X,y=y,
tau0=params$tau0,tau1=params$tau1,q=params$q,a0=params$a0,b0=params$b0,
verbose=FALSE,store=FALSE)

# Use posterior probabilities for variable selection
sss_chain$z_ergodic_avg[1:10]
</code></pre>

<hr>
<h2 id='spike_slab_logistic'>spike_slab_logistic</h2><span id='topic+spike_slab_logistic'></span>

<h3>Description</h3>

<p>Generates Markov chain targeting the posterior corresponding to
Bayesian logistic regression with spike and slab priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spike_slab_logistic(
  chain_length,
  X,
  y,
  tau0,
  tau1,
  q,
  rinit = NULL,
  verbose = FALSE,
  burnin = 0,
  store = TRUE,
  Xt = NULL,
  XXt = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spike_slab_logistic_+3A_chain_length">chain_length</code></td>
<td>
<p>Markov chain length</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_x">X</code></td>
<td>
<p>matrix of length n by p</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_y">y</code></td>
<td>
<p>Response</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_tau0">tau0</code></td>
<td>
<p>prior hyperparameter (non-negative real)</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_tau1">tau1</code></td>
<td>
<p>prior hyperparameter (non-negative real)</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_q">q</code></td>
<td>
<p>prior hyperparameter (strictly between 0 and 1)</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_rinit">rinit</code></td>
<td>
<p>initial distribution of Markov chain (default samples from the prior)</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_verbose">verbose</code></td>
<td>
<p>print iteration of the Markov chain (boolean)</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_burnin">burnin</code></td>
<td>
<p>chain burnin (non-negative integer)</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_store">store</code></td>
<td>
<p>store chain trajectory (boolean)</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_xt">Xt</code></td>
<td>
<p>Pre-calculated transpose of X</p>
</td></tr>
<tr><td><code id="spike_slab_logistic_+3A_xxt">XXt</code></td>
<td>
<p>Pre-calculated matrix X*transpose(X) (n by n matrix)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output from Markov chain targeting the posterior corresponding to
Bayesian logistic regression with spike and slab priors
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Synthetic dataset
syn_data &lt;- synthetic_data(n=100,p=200,s0=5,error_std=2,type='logistic')
X &lt;- syn_data$X
y &lt;- syn_data$y

# Hyperparamters
params &lt;- spike_slab_params(n=nrow(X),p=ncol(X))

# Run S^3
sss_chain &lt;- spike_slab_logistic(chain_length=4e3,burnin=1e3,X=X,y=y,
tau0=params$tau0,tau1=params$tau1,q=params$q,verbose=FALSE,store=FALSE)

# Use posterior probabilities for variable selection
sss_chain$z_ergodic_avg[1:10]
</code></pre>

<hr>
<h2 id='spike_slab_params'>spike_slab_params</h2><span id='topic+spike_slab_params'></span>

<h3>Description</h3>

<p>Generates hyperparameters for spike-and-slab
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spike_slab_params(n, p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spike_slab_params_+3A_n">n</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code id="spike_slab_params_+3A_p">p</code></td>
<td>
<p>number of covariates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>spike-and-slab hyperparameters q, tau0, tau1, a0, b0
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hyper_params &lt;- spike_slab_params(n=100,p=200)
print(hyper_params)
</code></pre>

<hr>
<h2 id='spike_slab_probit'>spike_slab_probit</h2><span id='topic+spike_slab_probit'></span>

<h3>Description</h3>

<p>Generates Markov chain targeting the posterior corresponding to
Bayesian probit regression with spike and slab priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spike_slab_probit(
  chain_length,
  X,
  y,
  tau0,
  tau1,
  q,
  rinit = NULL,
  verbose = FALSE,
  burnin = 0,
  store = TRUE,
  Xt = NULL,
  XXt = NULL,
  tau0_inverse = NULL,
  tau1_inverse = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spike_slab_probit_+3A_chain_length">chain_length</code></td>
<td>
<p>Markov chain length</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_x">X</code></td>
<td>
<p>matrix of length n by p</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_y">y</code></td>
<td>
<p>Response</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_tau0">tau0</code></td>
<td>
<p>prior hyperparameter (non-negative real)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_tau1">tau1</code></td>
<td>
<p>prior hyperparameter (non-negative real)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_q">q</code></td>
<td>
<p>prior hyperparameter (strictly between 0 and 1)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_rinit">rinit</code></td>
<td>
<p>initial distribution of Markov chain (default samples from the prior)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_verbose">verbose</code></td>
<td>
<p>print iteration of the Markov chain (boolean)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_burnin">burnin</code></td>
<td>
<p>chain burnin (non-negative integer)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_store">store</code></td>
<td>
<p>store chain trajectory (boolean)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_xt">Xt</code></td>
<td>
<p>Pre-calculated transpose of X</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_xxt">XXt</code></td>
<td>
<p>Pre-calculated matrix X*transpose(X) (n by n matrix)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_tau0_inverse">tau0_inverse</code></td>
<td>
<p>Pre-calculated matrix inverse(I + tau0^2*XXt) (n by n matrix)</p>
</td></tr>
<tr><td><code id="spike_slab_probit_+3A_tau1_inverse">tau1_inverse</code></td>
<td>
<p>Pre-calculated matrix inverse(I + tau1^2*XXt) (n by n matrix)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output from Markov chain targeting the posterior corresponding to
Bayesian logistic regression with spike and slab priors
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Synthetic dataset
syn_data &lt;- synthetic_data(n=100,p=200,s0=5,error_std=2,type='probit')
X &lt;- syn_data$X
Xt &lt;- t(X)
y &lt;- syn_data$y

# Hyperparamters
params &lt;- spike_slab_params(n=nrow(X),p=ncol(X))

# Run S^3
sss_chain &lt;- spike_slab_probit(chain_length=4e3,burnin=1e3,X=X,y=y,
tau0=params$tau0,tau1=params$tau1,q=params$q,verbose=FALSE,store=FALSE)

# Use posterior probabilities for variable selection
sss_chain$z_ergodic_avg[1:10]
</code></pre>

<hr>
<h2 id='synthetic_data'>synthetic_data</h2><span id='topic+synthetic_data'></span>

<h3>Description</h3>

<p>Generates synthetic linear and logistic regression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>synthetic_data(
  n,
  p,
  s0,
  error_std,
  type = "linear",
  scale = TRUE,
  signal = "constant"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="synthetic_data_+3A_n">n</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code id="synthetic_data_+3A_p">p</code></td>
<td>
<p>number of covariates</p>
</td></tr>
<tr><td><code id="synthetic_data_+3A_s0">s0</code></td>
<td>
<p>sparsity (number of non-zero components of the true signal)</p>
</td></tr>
<tr><td><code id="synthetic_data_+3A_error_std">error_std</code></td>
<td>
<p>Standard deviation of the Gaussian noise (linear regression only)</p>
</td></tr>
<tr><td><code id="synthetic_data_+3A_type">type</code></td>
<td>
<p>dataset type ('linear' or 'logistic')</p>
</td></tr>
<tr><td><code id="synthetic_data_+3A_scale">scale</code></td>
<td>
<p>design matrix X has columns mean zero and standard deviation 1 (TRUE or FALSE)</p>
</td></tr>
<tr><td><code id="synthetic_data_+3A_signal">signal</code></td>
<td>
<p>non-zero components of the true signal ('constant' or 'deacy')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Design matrix, response and true signal vector for linear and logistic regression
</p>


<h3>Examples</h3>

<pre><code class='language-R'>syn_data &lt;- synthetic_data(n=100,p=200,s0=5,error_std=2)

# syn_data$X is an n by p design matrix
dim(syn_data$X)

# syn_data$y is a length n response vector
length(syn_data$y) 

# syn_data$true_beta is a length n response vector with only the first s0 entries non-zero
all(syn_data$true_beta[1:5]!=0)
all(syn_data$true_beta[-c(1:5)]==0)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
