<!DOCTYPE html><html><head><title>Help for package scAnnotate</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scAnnotate}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#eva_cal'><p>eva_cal</p></a></li>
<li><a href='#pbmc1'><p>pbmc1</p></a></li>
<li><a href='#pbmc2'><p>pbmc2</p></a></li>
<li><a href='#predict_label'><p>predict_label</p></a></li>
<li><a href='#scAnnotate'><p>scAnnotate</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>An Automated Cell Type Annotation Tool for Single-Cell
RNA-Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3</td>
</tr>
<tr>
<td>Description:</td>
<td>An entirely data-driven cell type annotation tools, which requires training data to learn the classifier, but not biological knowledge to make subjective decisions. It consists of three steps: preprocessing training and test data, model fitting on training data, and cell classification on test data. See Xiangling Ji,Danielle Tsao, Kailun Bai, Min Tsao, Li Xing, Xuekui Zhang.(2022)&lt;<a href="https://doi.org/10.1101%2F2022.02.19.481159">doi:10.1101/2022.02.19.481159</a>&gt; for more details. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 4.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://doi.org/10.1101/2022.02.19.481159">https://doi.org/10.1101/2022.02.19.481159</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat (&ge; 3.0.0), rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet, stats, Seurat (&ge; 5.0.1), harmony, SeuratObject</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-13 22:29:52 UTC; xzhang</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiangling Ji [aut],
  Danielle Tsao [aut],
  Kailun Bai [ctb],
  Min Tsao [aut],
  Li Xing [aut],
  Xuekui Zhang [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xuekui Zhang &lt;xuekui@uvic.ca&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-14 00:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='eva_cal'>eva_cal</h2><span id='topic+eva_cal'></span>

<h3>Description</h3>

<p>calculate the F1 score of each cell population, mean of F1 score and overall accuracy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eva_cal(prediction, cell_label)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eva_cal_+3A_prediction">prediction</code></td>
<td>
<p>A vector of annotate cell type labels</p>
</td></tr>
<tr><td><code id="eva_cal_+3A_cell_label">cell_label</code></td>
<td>
<p>A vector of original cell type labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix contain the F1 score of each cell population, mean of F1 score and overall accuracy
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(predict_label)
data(pbmc2)
eva_cal(prediction = predict_label,cell_label = pbmc2[,1])

</code></pre>

<hr>
<h2 id='pbmc1'>pbmc1</h2><span id='topic+pbmc1'></span>

<h3>Description</h3>

<p>A subset of human Peripheral Blood Mononuclear Cells (PBMC) scRNA-seq data that
was sequenced using Drop-seq platform. The Seurat(version 4.0.5) package was used for
normalized using the NormalizeData function with the &quot;LogNormalize&quot; method and
a scale factor of 10,000. After modeling the mean-variance relationship with the
FindVariableFeautre function within &quot;vst&quot; methods, we selected the top 2,000 highly
variable genes and only used this selection going forward. The dataframe of the
cell type label and a gene expression matrix of 598 cells in the row and 2,000
genes in the column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pbmc1, package="scAnnotate")
</code></pre>


<h3>Format</h3>

<p>a data frame
</p>


<h3>References</h3>

<p>Ding,  J.et  al.(2019).   Systematic  comparative  analysis  of  single  cellrna-sequencing methods.bioRxiv
</p>

<hr>
<h2 id='pbmc2'>pbmc2</h2><span id='topic+pbmc2'></span>

<h3>Description</h3>

<p>A subset of human PBMC scRNA-seq data that was sequenced using inDrops platform.
The Seurat(version 4.0.5) package was used for normalized using the NormalizeData
function with the &quot;LogNormalize&quot; method and a scale factor of 10,000. After modeling
the mean-variance relationship with the  FindVariableFeautre function within &quot;vst&quot;
methods, we selected the top 2,000 highly variable genes and only used this selection
going forward. The dataframe of the cell type label and a gene expression matrix of
644 cells in the row and 2,000 genes in the column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pbmc2, package="scAnnotate")
</code></pre>


<h3>Format</h3>

<p>a data frame
</p>


<h3>References</h3>

<p>Ding,  J.et  al.(2019).   Systematic  comparative  analysis  of  single  cellrna-sequencing methods.bioRxiv
</p>

<hr>
<h2 id='predict_label'>predict_label</h2><span id='topic+predict_label'></span>

<h3>Description</h3>

<p>Cell type annotation of pbmc2 data that training from pbmc1 data by 'scAnnotate'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(predict_label, package="scAnnotate")
</code></pre>


<h3>Format</h3>

<p>a data frame
</p>

<hr>
<h2 id='scAnnotate'>scAnnotate</h2><span id='topic+scAnnotate'></span>

<h3>Description</h3>

<p>Annotate cell type labels of test data using a trained mixture model from training data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scAnnotate(
  train,
  test,
  distribution = c("normal", "dep"),
  correction = c("auto", "harmony", "seurat"),
  screening = c("wilcox", "t.test"),
  threshold = 0,
  lognormalized = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scAnnotate_+3A_train">train</code></td>
<td>
<p>A data frame of cell type label in the first column and a gene expression matrix where each row is a cell and each column is a gene from training data</p>
</td></tr>
<tr><td><code id="scAnnotate_+3A_test">test</code></td>
<td>
<p>A data matrix where each row is a cell and each column is a gene from test data</p>
</td></tr>
<tr><td><code id="scAnnotate_+3A_distribution">distribution</code></td>
<td>
<p>A character string indicates the distribution assumption on positive gene expression, which should be one of &quot;normal&quot;(default) or &quot;dep&quot;. &quot;dep&quot; refers to depth measure, which is a non-parametric distribution estimation approach.</p>
</td></tr>
<tr><td><code id="scAnnotate_+3A_correction">correction</code></td>
<td>
<p>A character string indicates the batch effect removal, which should be one of &quot;auto&quot;(default), &quot;seurat&quot;, or &quot;harmony&quot;. &quot;auto&quot; will automatically select the batch effect removal to follow our suggestion. That uses Seurat for dataset with at most one rare cell population (at most one cell population less than 100 cells) and Harmony for dataset with at least two rare cell populations (at least two cell populations less than 100 cells).</p>
</td></tr>
<tr><td><code id="scAnnotate_+3A_screening">screening</code></td>
<td>
<p>A character string indicates the gene screening methods, which should be one of &quot;wilcox&quot;(default) or &quot;t.test&quot;.</p>
</td></tr>
<tr><td><code id="scAnnotate_+3A_threshold">threshold</code></td>
<td>
<p>A numeric number indicates the threshold used for probabilities to classify cells, which should be a number from &quot;0&quot;(default) to &quot;1&quot;. If there's no probability higher than the threshold associated with a cell type, the cell will be labeled as &quot;unassigned.&quot;</p>
</td></tr>
<tr><td><code id="scAnnotate_+3A_lognormalized">lognormalized</code></td>
<td>
<p>A logical string indicates if both input data are log-normalized or raw matrix. TRUE (default) indicates input data are log-normalized, and FALSE indicates input data are raw data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector contain annotate cell type labels for test data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pbmc1)
data(pbmc2)
## Not run: 
  predict_label=scAnnotate(train=pbmc1,
                           test=pbmc2[,-1],
                           distribution="normal",
                           correction ="harmony",
                           screening ="wilcox",
                           threshold=0,
                           lognormalized=TRUE)
 
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
