<!DOCTYPE html><html><head><title>Help for package scaper</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scaper}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#scaper-package'><p>scaper: Single Cell Transcriptomics-Level Cytokine Activity Prediction and Estimation</p></a></li>
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#genesetCytoSig'><p>CytoSig gene set construction.</p></a></li>
<li><a href='#genesetReactome'><p>Reactome gene set construction.</p></a></li>
<li><a href='#scape'><p>Cytokine activity scores for a normalized matrix.</p></a></li>
<li><a href='#scapeForSeurat'><p>Cytokine activity scores for a Seurat matrix.</p></a></li>
<li><a href='#supportedCytokines'><p>Gene set to score for the CytoSig or the Reactome databases.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Single Cell Transcriptomics-Level Cytokine Activity Prediction
and Estimation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Azka Javaid &lt;azka.javaid.gr@dartmouth.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Generates cell-level cytokine activity estimates using relevant information from gene sets constructed with the 'CytoSig' and the 'Reactome' databases and scored using the modified 'Variance-adjusted Mahalanobis (VAM)' framework for single-cell RNA-sequencing (scRNA-seq) data. 'CytoSig' database is described in: Jiang at al., (2021) &lt;<a href="https://doi.org/10.1038%2Fs41592-021-01274-5">doi:10.1038/s41592-021-01274-5</a>&gt;. 'Reactome' database is described in: Gillespie et al., (2021) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkab1028">doi:10.1093/nar/gkab1028</a>&gt;. The 'VAM' method is outlined in: Frost (2020) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkaa582">doi:10.1093/nar/gkaa582</a>&gt;.</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, pheatmap, rmarkdown, usethis</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr, xml2, stringr, dplyr, Seurat, SeuratObject, VAM,
utils</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-18 20:08:25 UTC; azkajavaid</td>
</tr>
<tr>
<td>Author:</td>
<td>H. Robert Frost [aut],
  Azka Javaid [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-19 07:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='scaper-package'>scaper: Single Cell Transcriptomics-Level Cytokine Activity Prediction and Estimation</h2><span id='topic+scaper-package'></span><span id='topic+_PACKAGE'></span>

<h3>Description</h3>

<p>Generates cell-level cytokine activity estimates using relevant information from gene sets constructed with the 'CytoSig' and the 'Reactome' databases and scored using the modified 'Variance-adjusted Mahalanobis (VAM)' framework for single-cell RNA-sequencing (scRNA-seq) data. 'CytoSig' database is described in: Jiang at al., (2021) <a href="https://doi.org/10.1038/s41592-021-01274-5">doi:10.1038/s41592-021-01274-5</a>. 'Reactome' database is described in: Gillespie et al., (2021) <a href="https://doi.org/10.1093/nar/gkab1028">doi:10.1093/nar/gkab1028</a>. The 'VAM' method is outlined in: Frost (2020) <a href="https://doi.org/10.1093/nar/gkaa582">doi:10.1093/nar/gkaa582</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Azka Javaid <a href="mailto:azka.javaid.gr@dartmouth.edu">azka.javaid.gr@dartmouth.edu</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> H. Robert Frost <a href="mailto:hildreth.r.frost@dartmouth.edu">hildreth.r.frost@dartmouth.edu</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='genesetCytoSig'>CytoSig gene set construction.</h2><span id='topic+genesetCytoSig'></span>

<h3>Description</h3>

<p>Returns the gene set associated with input cytokine(s) from the CytoSig database
given manually specified cytokine-specific output csv file(s) under
the extdata directory with file name beginning with
the specified cytokine (i.e., 'IL6_output.csv') as currently provided for the IL6 cytokine.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genesetCytoSig(cytokine.eval, file.name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genesetCytoSig_+3A_cytokine.eval">cytokine.eval</code></td>
<td>
<p>Cytokine(s) associated with the query.</p>
</td></tr>
<tr><td><code id="genesetCytoSig_+3A_file.name">file.name</code></td>
<td>
<p>List of XML file(s) associated with the cytokine(s) beginning with the specific cytokine name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe consisting of genes and the associated log2 fold change values associated with the specific cytokine(s).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file.name.cytosig1 &lt;- system.file("extdata", "IL6_output.csv", package = "scaper")
genesetCytoSig(cytokine.eval = "IL6", file.name = file.name.cytosig1) %&gt;% head(10)
</code></pre>

<hr>
<h2 id='genesetReactome'>Reactome gene set construction.</h2><span id='topic+genesetReactome'></span>

<h3>Description</h3>

<p>Returns the gene set consisting of genes on the Reactome pathway
hierarchy given input cytokine(s) and specified list of pathway
hierarchy xml file(s) with file name beginning with the specified cytokine
(i.e., 'IL6_Interleukin6_signaling.xml') as currently provided for the IL6 cytokine.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genesetReactome(cytokine.eval, file.name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genesetReactome_+3A_cytokine.eval">cytokine.eval</code></td>
<td>
<p>Cytokine(s) associated with the query.</p>
</td></tr>
<tr><td><code id="genesetReactome_+3A_file.name">file.name</code></td>
<td>
<p>List of XML file(s) associated with the cytokine(s) beginning with the specific cytokine name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe consisting of list of genes on the molecular pathway associated with the specified cytokine(s).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file.name.reactome1 &lt;- system.file("extdata", "IL6_Interleukin6_signaling.xml", 
package = "scaper")
genesetReactome(cytokine.eval = "IL6", file.name = file.name.reactome1) %&gt;% head(10)
</code></pre>

<hr>
<h2 id='scape'>Cytokine activity scores for a normalized matrix.</h2><span id='topic+scape'></span>

<h3>Description</h3>

<p>Computes cell-level estimates of cytokine activity for a normalized scRNA-seq
count matrix using the SCAPE method. SCAPE activity estimates are computed by scoring weighted genes
sets from the CytoSig or Reactome databases using the <code><a href="VAM.html#topic+vamForCollection">VAM::vamForCollection()</a></code>
function. Individual gene sets for subsequent scoring can be reconstructed using
the <code><a href="#topic+genesetCytoSig">genesetCytoSig</a></code> and the <code><a href="#topic+genesetReactome">genesetReactome</a></code> functions
for the CytoSig and the Reactome database, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scape(counts.matrix, database = "cytosig", cytokine = "all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scape_+3A_counts.matrix">counts.matrix</code></td>
<td>
<p>A <code class="reqn">m x n</code> normalized counts matrix with <code class="reqn">m</code> samples and <code class="reqn">n</code> genes.</p>
</td></tr>
<tr><td><code id="scape_+3A_database">database</code></td>
<td>
<p>Database used for gene set construction and set scoring.
</p>

<ul>
<li><p> &quot;cytosig&quot; performs scoring for up to 41 cytokines using the CytoSig database.
</p>
</li>
<li><p> &quot;reactome&quot; performs scoring for up to 30 cytokines using the Reactome database.
</p>
</li></ul>
</td></tr>
<tr><td><code id="scape_+3A_cytokine">cytokine</code></td>
<td>
<p>Vector of cytokine names to score for activity. The default value
of &quot;all&quot; will score all 41 cytokines supported by CytoSig or 31 supported by Reactome. Please see
function <code><a href="#topic+supportedCytokines">supportedCytokines</a></code> to view all the CytoSig or the Reactome
specific scored cytokines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code class="reqn">m x p</code> matrix consisting of the cell-level cytokine activity scores for <code class="reqn">p</code> cytokines.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+genesetCytoSig">genesetCytoSig</a></code>, <code><a href="#topic+genesetReactome">genesetReactome</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
library(SeuratObject)
pbmc_small &lt;- NormalizeData(pbmc_small)
counts.matrix &lt;- as.data.frame(t(as.matrix(pbmc_small@assays$RNA@data)))
CytoSig.score.output &lt;- scape(counts.matrix = counts.matrix, 
database = "cytosig")
head(CytoSig.score.output)[,1:3]
CytoSig.score.output.specific &lt;- scape(counts.matrix = counts.matrix, 
database = "cytosig", cytokine = c("IL4", "IL13"))
head(CytoSig.score.output.specific)

</code></pre>

<hr>
<h2 id='scapeForSeurat'>Cytokine activity scores for a Seurat matrix.</h2><span id='topic+scapeForSeurat'></span>

<h3>Description</h3>

<p>Computes cell-level estimates of cytokine activity for a scRNA-seq Seurat
count matrix using the scapeForSeurat method. SCAPE activity estimates are computed by scoring weighted gene
sets from the CytoSig or Reactome databases using the Variance-adjusted Mahalanobis (VAM) method
as implemented in the <code><a href="VAM.html#topic+vamForSeurat">VAM::vamForSeurat()</a></code> function. Individual gene sets for subsequent scoring can be reconstructed using
the <code><a href="#topic+genesetCytoSig">genesetCytoSig</a></code> and the <code><a href="#topic+genesetReactome">genesetReactome</a></code> functions
for the CytoSig and the Reactome database, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scapeForSeurat(
  seurat.object,
  database = "cytosig",
  cytokine = "all",
  normalize = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scapeForSeurat_+3A_seurat.object">seurat.object</code></td>
<td>
<p>Seurat counts matrix.</p>
</td></tr>
<tr><td><code id="scapeForSeurat_+3A_database">database</code></td>
<td>
<p>Database used for gene set construction and set scoring.
</p>

<ul>
<li><p> &quot;cytosig&quot; (default) performs scoring for up to 41 cytokines using the CytoSig database.
</p>
</li>
<li><p> &quot;reactome&quot; performs scoring for up to 30 cytokines using the Reactome database.
</p>
</li></ul>
</td></tr>
<tr><td><code id="scapeForSeurat_+3A_cytokine">cytokine</code></td>
<td>
<p>Vector of cytokine names to score for activity. The default value of &quot;all&quot;
will score all 41 cytokines supported by CytoSig or 31 supported by Reactome. Please see
function <code><a href="#topic+supportedCytokines">supportedCytokines</a></code> to view all the CytoSig or the Reactome
specific scored cytokines.</p>
</td></tr>
<tr><td><code id="scapeForSeurat_+3A_normalize">normalize</code></td>
<td>
<p>Boolean indicator for whether normalization should be performed before performing gene set scoring.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat object consisting of cell-level cytokine activity scores returned as a separate assay (scape for scoring via the CytoSig database and VAMcdf for scoring via the Reactome database).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+genesetCytoSig">genesetCytoSig</a></code>, <code><a href="#topic+genesetReactome">genesetReactome</a></code>, <code><a href="#topic+scape">scape</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SeuratObject)
CytoSig.score.output.all &lt;- scapeForSeurat(seurat.object = pbmc_small, 
database = "cytosig", cytokine = "all", normalize=TRUE)
(as.data.frame(CytoSig.score.output.all@assays$scape@data))[1:6,1:3]
CytoSig.score.output.specific &lt;- scapeForSeurat(seurat.object = pbmc_small, 
database = "cytosig", cytokine = c("IL4", "IL13"), normalize=TRUE)
(as.data.frame(CytoSig.score.output.specific@assays$scape@data))[,1:3]

</code></pre>

<hr>
<h2 id='supportedCytokines'>Gene set to score for the CytoSig or the Reactome databases.</h2><span id='topic+supportedCytokines'></span>

<h3>Description</h3>

<p>Returns the names of the cytokines supported by either the CytoSig or the Reactome databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>supportedCytokines(database = "cytosig")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="supportedCytokines_+3A_database">database</code></td>
<td>
<p>Database used for gene set construction and set scoring.
</p>

<ul>
<li><p> &quot;cytosig&quot; returns the 41 cytokines scored using the CytoSig database.
</p>
</li>
<li><p> &quot;reactome&quot; returns the 30 cytokines scored using the Reactome database.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>List of cytokines associated with the CytoSig or the Reactome databases.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>supportedCytokines(database = "cytosig")
supportedCytokines(database = "reactome")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
