<!DOCTYPE html><html><head><title>Help for package scBio</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scBio}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BulkFlu'><p>Gene expression profiles of flu and pbs sample.</p></a></li>
<li><a href='#CPM'><p>The Cellular Population Mapping (CPM) algorithm.</p></a></li>
<li><a href='#SCCellSpace'><p>Single-cell cell space.</p></a></li>
<li><a href='#SCFlu'><p>Gene expression profiles of lung cells after influenza infection.</p></a></li>
<li><a href='#SCLabels'><p>Single-cell classification into cell types.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single Cell Genomics for Enhancing Cell Composition Inference
from Bulk Genomics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.6</td>
</tr>
<tr>
<td>Author:</td>
<td>Amit Frishberg [aut, cre], Naama Peshes-Yaloz [aut], Irit Gat-Viks [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Amit Frishberg &lt;amfrishberg@gmail.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/amitfrish/scBio">https://github.com/amitfrish/scBio</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/amitfrish/scBio/issues">https://github.com/amitfrish/scBio/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Cellular population mapping (CPM) a deconvolution algorithm in which single-cell genomics is required in only one or a few samples, where in other samples of the same tissue, only bulk genomics is measured and the underlying fine resolution cellular heterogeneity is inferred.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>biocViews:</td>
<td>limma</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>sp, foreach, parallel, doSNOW, raster, fields, LiblineaR,
limma</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-05-20 08:49:30 UTC; IritGNB5</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-05-20 10:10:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='BulkFlu'>Gene expression profiles of flu and pbs sample.</h2><span id='topic+BulkFlu'></span>

<h3>Description</h3>

<p>A dataset containing the RNA-seq profiles of colaborative-cross (CC) mice 2 days after the infection with either the flu virus or PBS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BulkFlu
</code></pre>


<h3>Format</h3>

<p>A matrix with 1858 rows (genes) and 74 columns (samples).</p>

<hr>
<h2 id='CPM'>The Cellular Population Mapping (CPM) algorithm.</h2><span id='topic+CPM'></span>

<h3>Description</h3>

<p>This function initiate the Cellular Population Mapping (CPM) algorithm - a deconvolution algorithm in which single-cell genomics is required in only one or a few samples, where in other samples of the same tissue, only bulk genomics is measured and the underlying fine resolution cellular heterogeneity is inferred.
CPM predicts the abundance of cells (and cell types) ranging monotonically from negative to positive levels. Using a relative framework these values correspond to decrease and increase in cell abundance levels, respectively. On the other hand, in an absolute framework lower values (including negatives) correspond to lower abundances and vise versa. These values are comparable between samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CPM(
  SCData,
  SCLabels,
  BulkData,
  cellSpace,
  no_cores = NULL,
  neighborhoodSize = 10,
  modelSize = 50,
  minSelection = 5,
  quantifyTypes = F,
  typeTransformation = F,
  calculateCI = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CPM_+3A_scdata">SCData</code></td>
<td>
<p>A matrix containing the single-cell RNA-seq data. Each row corresponds to a certain gene and each column to a certain cell. Importantly, CPM relies on many iterative processes and therefore might take a long running time. For extremely large single cell datasets, we suggest to use only a portion of the data, using random sampling of the cells.</p>
</td></tr>
<tr><td><code id="CPM_+3A_sclabels">SCLabels</code></td>
<td>
<p>A vector containing the labels of each of the cells.</p>
</td></tr>
<tr><td><code id="CPM_+3A_bulkdata">BulkData</code></td>
<td>
<p>A matrix containing heterogenous RNA-seq data for one or more samples. Each row corresponds to a certain gene and each column to a certain sample.</p>
</td></tr>
<tr><td><code id="CPM_+3A_cellspace">cellSpace</code></td>
<td>
<p>The cell state space corresponding to the single-cell RNA-seq data. It can be a vector for a 1-dim space or a 2D matrix for a two space where each column represents a different dimension. The cell space should incorporate the similarities of cells within cell types. Similarities between cells from different cell types, based on the cell space, are not taken into account in CPM.</p>
</td></tr>
<tr><td><code id="CPM_+3A_no_cores">no_cores</code></td>
<td>
<p>A number for the amount of cores which will be used for the analysis. The defalt (NULL) is total number of cores minus 1.</p>
</td></tr>
<tr><td><code id="CPM_+3A_neighborhoodsize">neighborhoodSize</code></td>
<td>
<p>Cell neighborhood size which will be used for the analysis. This should be lower than the number of cells in the smallest cell type. The defalt is 10.</p>
</td></tr>
<tr><td><code id="CPM_+3A_modelsize">modelSize</code></td>
<td>
<p>The reference subset size in each iteration of CPM. This should be lower than the total number of cells. The defalt is 50.</p>
</td></tr>
<tr><td><code id="CPM_+3A_minselection">minSelection</code></td>
<td>
<p>The minimum number of times in which each reference cell is selected. Increasing this value might have a large effect on the algorithm's running time. The defalt is 5.</p>
</td></tr>
<tr><td><code id="CPM_+3A_quantifytypes">quantifyTypes</code></td>
<td>
<p>A boolean parameter indicating whether the prediction of cell type quantities is needed. This is recommended only in the case of homogenicity within cell types. Cell types with high inner cellular variability will recieve less reliabe values. The default is FALSE.</p>
</td></tr>
<tr><td><code id="CPM_+3A_typetransformation">typeTransformation</code></td>
<td>
<p>This parameter will have an effect only if quantifyTypes = TRUE. A boolean parameter indicating whether cell type deconvolution should be provided in fractions. This is done by substracting all cell types by values of the minimal cell type and dividing in their sum. This is not recommended, since it reduces the comparability between sample. The default is FALSE.</p>
</td></tr>
<tr><td><code id="CPM_+3A_calculateci">calculateCI</code></td>
<td>
<p>A boolean parameter indicating whether the calculation of confidence itervals is needed. The default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list including:
</p>
<table>
<tr><td><code>predicted</code></td>
<td>
<p>CPM predicted cell abundance matrix. Each row represents a sample and each column a single cell.</p>
</td></tr>
<tr><td><code>cellTypePredictions</code></td>
<td>
<p>CPM predicted cell-type abundance matrix. Each row represnts a sample and each column a single cell-type. This is calculated if quantifyTypes = TRUE. </p>
</td></tr>
<tr><td><code>confIntervals</code></td>
<td>
<p>A matrix containing the confidence iterval for each cell and sample. Each row represnts a sample and each column a single cell. This is calculated if calculateCI = TRUE.</p>
</td></tr>
<tr><td><code>numOfRuns</code></td>
<td>
<p>The number of deconvolution repeats preformed by CPM. </p>
</td></tr>
</table>


<h3>References</h3>

<p>Frishberg, A., Peshes-Yaloz, N., Cohn, O., Rosentul, D., Steuerman, Y., Valadarsky, L., Yankovitz, G., Mandelboim, M., Iraqi, F.A., Amit, I. et al. (2019) Cell composition analysis of bulk genomics using single-cell data. Nature Methods, 16, 327-332.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SCLabels)
data(SCFlu)
data(BulkFlu)
data(SCCellSpace)

# Creating relative bulk data (Infleunza infection compared to PBS):
BulkFluReduced = BulkFlu - rowMeans(BulkFlu[,grep("pbs",colnames(BulkFlu))])
BulkFluReduced = BulkFluReduced[,grep("flu",colnames(BulkFluReduced))]

# Running CPM using only a single cell-type:
oneCellTypeIndexes = which(SCLabels == "MPS")
res = CPM(SCData = SCFlu[,oneCellTypeIndexes], SCLabels = SCLabels[oneCellTypeIndexes],
          BulkData = BulkFluReduced, cellSpace = SCCellSpace[oneCellTypeIndexes,], no_cores = 2)

## Not run: 

# Running CPM using a variety of cell-types:
res = CPM(SCFlu, SCLabels, BulkFluReduced, SCCellSpace, no_cores = 2)
### Full multi-threading : CPM(SCFlu, SCLabels, BulkFluReduced, SCCellSpace)

## End(Not run)
</code></pre>

<hr>
<h2 id='SCCellSpace'>Single-cell cell space.</h2><span id='topic+SCCellSpace'></span>

<h3>Description</h3>

<p>A dataset containing a 2-dim cell space of all single-cells in the SCFlu dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCCellSpace
</code></pre>


<h3>Format</h3>

<p>A matrix with 349 rows (cells) and 2 columns (dimensions).</p>

<hr>
<h2 id='SCFlu'>Gene expression profiles of lung cells after influenza infection.</h2><span id='topic+SCFlu'></span>

<h3>Description</h3>

<p>A dataset containing the RNA-seq profiles of lung cells from multiple cell types, taken from two mice 2 days after the infection with either the flu virus or PBS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCFlu
</code></pre>


<h3>Format</h3>

<p>A matrix with 1858 rows (genes) and 349 columns (cells).</p>

<hr>
<h2 id='SCLabels'>Single-cell classification into cell types.</h2><span id='topic+SCLabels'></span>

<h3>Description</h3>

<p>A dataset containing the classification of each cell in the SCFlu dataset to a specific cell type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCLabels
</code></pre>


<h3>Format</h3>

<p>A vector with 349 values.</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
