<!DOCTYPE html><html lang="en"><head><title>Help for package scCATCH</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scCATCH}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cellmatch'><p>cellmatch</p></a></li>
<li><a href='#createscCATCH'><p>scCATCH object</p></a></li>
<li><a href='#demo_data'><p>Demo data of single-cell RNA-seq data</p></a></li>
<li><a href='#demo_geneinfo'><p>Demo data of geneinfo</p></a></li>
<li><a href='#demo_marker'><p>Demo data of markers</p></a></li>
<li><a href='#findcelltype'><p>Evidence-based score and annotation for each cluster</p></a></li>
<li><a href='#findmarkergene'><p>Find potential marker genes for each cluster</p></a></li>
<li><a href='#geneinfo'><p>geneinfo</p></a></li>
<li><a href='#rev_gene'><p>Pre-processing step: revising gene symbols</p></a></li>
<li><a href='#scCATCH'><p>Definition of 'scCATCH' class</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single Cell Cluster-Based Annotation Toolkit for Cellular
Heterogeneity</td>
</tr>
<tr>
<td>Version:</td>
<td>3.2.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Author:</td>
<td>Xin Shao</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xin Shao&lt;xin_shao@zju.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An automatic cluster-based annotation pipeline based on evidence-based score by matching the marker genes with known cell markers in tissue-specific cell taxonomy reference database for single-cell RNA-seq data. See Shao X, et al (2020) &lt;<a href="https://doi.org/10.1016%2Fj.isci.2020.100882">doi:10.1016/j.isci.2020.100882</a>&gt; for more details.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ZJUFanLab/scCATCH">https://github.com/ZJUFanLab/scCATCH</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, testthat, prettydoc</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix, methods, progress, stats, reshape2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-23 13:07:07 UTC; 61046</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-23 14:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cellmatch'>cellmatch</h2><span id='topic+cellmatch'></span>

<h3>Description</h3>

<p>Marker genes of <code>'Human'</code> and <code>'Mouse'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cellmatch
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 49560 rows and 11 columns.
</p>


<h3>Source</h3>

<p><a href="https://github.com/ZJUFanLab/scCATCH/tree/master/data">https://github.com/ZJUFanLab/scCATCH/tree/master/data</a>
</p>

<hr>
<h2 id='createscCATCH'>scCATCH object</h2><span id='topic+createscCATCH'></span>

<h3>Description</h3>

<p>create scCATCH object using single-cell count data and cluster information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createscCATCH(data, cluster)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createscCATCH_+3A_data">data</code></td>
<td>
<p>A matrix or dgCMatrix containing normalized single-cell RNA-seq data, each column representing a cell, each row representing a gene. See <code><a href="#topic+demo_data">demo_data</a></code>.</p>
</td></tr>
<tr><td><code id="createscCATCH_+3A_cluster">cluster</code></td>
<td>
<p>A character containing the cluster information for each cell. The length of it must be equal to the ncol of the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scCATCH object
</p>

<hr>
<h2 id='demo_data'>Demo data of single-cell RNA-seq data</h2><span id='topic+demo_data'></span>

<h3>Description</h3>

<p>Demo data of single-cell RNA-seq data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_data()
</code></pre>


<h3>Details</h3>

<p><code>data</code> used in <code><a href="#topic+createscCATCH">createscCATCH</a></code> must be a <code>matrix</code> object, each column representing a cell, each row representing a gene.
</p>


<h3>Value</h3>

<p>A demo data matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data_demo &lt;- demo_data()
</code></pre>

<hr>
<h2 id='demo_geneinfo'>Demo data of geneinfo</h2><span id='topic+demo_geneinfo'></span>

<h3>Description</h3>

<p>Demo data of geneinfo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_geneinfo()
</code></pre>


<h3>Details</h3>

<p><code>geneinfo</code> used in <code><a href="#topic+rev_gene">rev_gene</a></code> must be a <code>data.frame</code> object with three columns, namely <code>'symbol'</code>, <code>'synonyms'</code>, <code>'species'</code>.
</p>


<h3>Value</h3>

<p>A demo geneinfo data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geneinfo_demo &lt;- demo_geneinfo()
</code></pre>

<hr>
<h2 id='demo_marker'>Demo data of markers</h2><span id='topic+demo_marker'></span>

<h3>Description</h3>

<p>Demo data of markers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_marker()
</code></pre>


<h3>Details</h3>

<p><code>markers</code> used in <code><a href="#topic+findmarkergene">findmarkergene</a></code> must be a <code>data.frame</code> object with eleven columns.
</p>


<h3>Value</h3>

<p>A demo marker data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>markers_demo &lt;- demo_marker()
</code></pre>

<hr>
<h2 id='findcelltype'>Evidence-based score and annotation for each cluster</h2><span id='topic+findcelltype'></span>

<h3>Description</h3>

<p>Evidence-based score and annotation for each cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findcelltype(object, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findcelltype_+3A_object">object</code></td>
<td>
<p>scCATCH object generated from <code><a href="#topic+findmarkergene">findmarkergene</a></code>.</p>
</td></tr>
<tr><td><code id="findcelltype_+3A_verbose">verbose</code></td>
<td>
<p>Show progress messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scCATCH object containing the results of predicted cell types for each cluster.
</p>

<hr>
<h2 id='findmarkergene'>Find potential marker genes for each cluster</h2><span id='topic+findmarkergene'></span>

<h3>Description</h3>

<p>Identify potential marker genes for each cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findmarkergene(
  object,
  species = NULL,
  cluster = "All",
  if_use_custom_marker = FALSE,
  marker = NULL,
  cancer = "Normal",
  tissue = NULL,
  use_method = "1",
  comp_cluster = NULL,
  cell_min_pct = 0.25,
  logfc = 0.25,
  pvalue = 0.05,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findmarkergene_+3A_object">object</code></td>
<td>
<p>scCATCH object generated from <code><a href="#topic+createscCATCH">createscCATCH</a></code>.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_species">species</code></td>
<td>
<p>The specie of cells. The species must be defined. 'Human' or 'Mouse'. When if_use_custom_marker is set TRUE, no need to define the species.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_cluster">cluster</code></td>
<td>
<p>Select which clusters for potential marker genes identification. e.g. '1', '2', etc. The default is 'All' to find potential makrer genes for each cluster.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_if_use_custom_marker">if_use_custom_marker</code></td>
<td>
<p>Whether to use custom markers data.frame.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_marker">marker</code></td>
<td>
<p>A data.frame containing marker genes. See <code><a href="#topic+demo_marker">demo_marker</a></code>. Default is to use the system <code><a href="#topic+cellmatch">cellmatch</a></code> data.frame.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_cancer">cancer</code></td>
<td>
<p>If the sample is from cancer tissue, then the cancer type may be defined. When if_use_custom_marker is set TRUE, no need to define the cancer.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_tissue">tissue</code></td>
<td>
<p>Tissue origin of cells must be defined. Select one or more related tissue types. When if_use_custom_marker is set TRUE, no need to define the tissue.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_use_method">use_method</code></td>
<td>
<p>'1' is to compare with other every cluster. '2' is to compare with other clusters together.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_comp_cluster">comp_cluster</code></td>
<td>
<p>Number of clusters to compare. Default is to compare all other cluster for each cluster. Set it between 1 and length of unique clusters. More marker genes will be obtained for smaller comp_cluster.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_cell_min_pct">cell_min_pct</code></td>
<td>
<p>Include the gene detected in at least this many cells in each cluster.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_logfc">logfc</code></td>
<td>
<p>Include the gene with at least this fold change of average gene expression compared to every other clusters.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_pvalue">pvalue</code></td>
<td>
<p>Include the significantly highly expressed gene with this cutoff of p value from wilcox test compared to every other clusters.</p>
</td></tr>
<tr><td><code id="findmarkergene_+3A_verbose">verbose</code></td>
<td>
<p>Show progress messages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Details of available tissues see <a href="https://github.com/ZJUFanLab/scCATCH/wiki">https://github.com/ZJUFanLab/scCATCH/wiki</a>
</p>


<h3>Value</h3>

<p>scCATCH object
</p>

<hr>
<h2 id='geneinfo'>geneinfo</h2><span id='topic+geneinfo'></span>

<h3>Description</h3>

<p>Gene symbols of <code>'Human'</code> and <code>'Mouse'</code> updated on Jan. 2, 2022 for revising genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneinfo
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 240502 rows and 3 columns.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/gene">https://www.ncbi.nlm.nih.gov/gene</a>
</p>

<hr>
<h2 id='rev_gene'>Pre-processing step: revising gene symbols</h2><span id='topic+rev_gene'></span>

<h3>Description</h3>

<p>Revise genes according to NCBI Gene symbols updated in June 19, 2022 for count matrix, user-custom cell marker data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rev_gene(data = NULL, data_type = NULL, species = NULL, geneinfo = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rev_gene_+3A_data">data</code></td>
<td>
<p>A matrix or dgCMatrix containing count or normalized data, each column representing a spot or a cell, each row representing a gene; Or a data.frame containing cell markers, use <code><a href="#topic+demo_marker">demo_marker</a></code>.</p>
</td></tr>
<tr><td><code id="rev_gene_+3A_data_type">data_type</code></td>
<td>
<p>A character to define the type of <code>data</code>, select <code>'data'</code> for the data matrix, <code>'marker'</code> for the data.frame containing cell markers.</p>
</td></tr>
<tr><td><code id="rev_gene_+3A_species">species</code></td>
<td>
<p>Species of the data.<code>'Human'</code> or <code>'Mouse'</code>.</p>
</td></tr>
<tr><td><code id="rev_gene_+3A_geneinfo">geneinfo</code></td>
<td>
<p>A data.frame of the system data containing gene symbols of <code>'Human'</code> and <code>'Mouse'</code> updated on Jan. 1, 2022.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new matrix or data.frame.
</p>

<hr>
<h2 id='scCATCH'>Definition of 'scCATCH' class</h2><span id='topic+scCATCH'></span><span id='topic+scCATCH-class'></span>

<h3>Description</h3>

<p>An S4 class containing the data, meta, and results of inferred cell types.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>A list containing normalized data. See <code><a href="#topic+demo_data">demo_data</a></code>.</p>
</dd>
<dt><code>meta</code></dt><dd><p>A data frame containing the meta data.</p>
</dd>
<dt><code>para</code></dt><dd><p>A list containing the parameters.</p>
</dd>
<dt><code>markergene</code></dt><dd><p>A data frame containing the identified markers for each cluster.</p>
</dd>
<dt><code>celltype</code></dt><dd><p>A data frame containing the cell types for each cluster.</p>
</dd>
<dt><code>marker</code></dt><dd><p>A data frame containing the known markers. See <code><a href="#topic+demo_marker">demo_marker</a></code>.</p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
