<!DOCTYPE html><html><head><title>Help for package scCAN</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scCAN}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjustedRandIndex'><p>adjustedRandIndex</p></a></li>
<li><a href='#calculate_celltype_prob'><p>calculate_celltype_prob</p></a></li>
<li><a href='#curate_markers'><p>curate_markers</p></a></li>
<li><a href='#find_markers'><p>find_markers</p></a></li>
<li><a href='#find_specific_marker'><p>find_specific_marker</p></a></li>
<li><a href='#get_cluster_markers'><p>get_cluster_markers</p></a></li>
<li><a href='#scCAN'><p>scCAN</p></a></li>
<li><a href='#SCE'><p>SCE</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single-Cell Clustering using Autoencoder and Network Fusion</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.5</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bang Tran &lt;s.tran@csus.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A single-cell Clustering method using 'Autoencoder' and Network fusion ('scCAN') Bang Tran (2022) &lt;<a href="https://doi.org/10.1038%2Fs41598-022-14218-6">doi:10.1038/s41598-022-14218-6</a>&gt; for segregating the cells from the high-dimensional 'scRNA-Seq' data. The software automatically determines the optimal number of clusters and then partitions the cells in a way such that the results are robust to noise and dropouts. 'scCAN' is fast and it supports Windows, Linux, and Mac OS.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2">LGPL-2</a> | <a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0), scDHA, FNN, purrr</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-13 08:04:34 UTC; bangt</td>
</tr>
<tr>
<td>Author:</td>
<td>Bang Tran [aut, cre],
  Duc Tran [aut],
  Hung Nguyen [aut],
  Tin Nguyen [fnd]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-13 18:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='adjustedRandIndex'>adjustedRandIndex</h2><span id='topic+adjustedRandIndex'></span>

<h3>Description</h3>

<p>The function to calculate adjusted Rand index value with the
inputs of true clusters and predicted clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjustedRandIndex(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjustedRandIndex_+3A_x">x</code></td>
<td>
<p>A vector that contain predicted cluster assignment.</p>
</td></tr>
<tr><td><code id="adjustedRandIndex_+3A_y">y</code></td>
<td>
<p>A vector that contain true cluster assignment.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An value number ranging from 0 to 1 where 1 indicates a perfect
clustering result and 0 means random partition.
</p>

<hr>
<h2 id='calculate_celltype_prob'>calculate_celltype_prob</h2><span id='topic+calculate_celltype_prob'></span>

<h3>Description</h3>

<p>Calculate clusters and cell types similarity based on the markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_celltype_prob(clt_marker_list, marker_database_list, type = "jacc")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_celltype_prob_+3A_clt_marker_list">clt_marker_list</code></td>
<td>
<p>A list of markers for all cluster.</p>
</td></tr>
<tr><td><code id="calculate_celltype_prob_+3A_marker_database_list">marker_database_list</code></td>
<td>
<p>A list of markers of all reference cell types.</p>
</td></tr>
<tr><td><code id="calculate_celltype_prob_+3A_type">type</code></td>
<td>
<p>A parameter to select the method to measure cluster and cell type
similarity
</p>

<ul>
<li><p> jacc - Jaccard index.
</p>
</li>
<li><p> ac - Accuracy.
</p>
</li>
<li><p> f1 - F1 score.
</p>
</li></ul>
<p>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A confusion matrix between clusters and cell types. Each cell represents
a probabilty of a cluster belongs to a cell type.
</p>

<hr>
<h2 id='curate_markers'>curate_markers</h2><span id='topic+curate_markers'></span>

<h3>Description</h3>

<p>Filter genes that have low p-value and fold-change.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>curate_markers(
  whole_list,
  gene_names,
  wilcox_threshold = 0.001,
  logfc_threshold = 1.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="curate_markers_+3A_whole_list">whole_list</code></td>
<td>
<p>A list of markers for all clusters.</p>
</td></tr>
<tr><td><code id="curate_markers_+3A_gene_names">gene_names</code></td>
<td>
<p>All the gene names of the expression matrix.</p>
</td></tr>
<tr><td><code id="curate_markers_+3A_wilcox_threshold">wilcox_threshold</code></td>
<td>
<p>A threshold for p-value <code>wilcox_threshold = 0.001</code> by default.</p>
</td></tr>
<tr><td><code id="curate_markers_+3A_logfc_threshold">logfc_threshold</code></td>
<td>
<p>A threshold for fold-change <code>logfc_threshold = 1.5</code> by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of markers that are strong expressed for discovered clusters.
</p>

<hr>
<h2 id='find_markers'>find_markers</h2><span id='topic+find_markers'></span>

<h3>Description</h3>

<p>Perform cluster-wise Wilcox test and fold-change for each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_markers(input_data_matrix, cluster_labels, identity = 1, threads = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_markers_+3A_input_data_matrix">input_data_matrix</code></td>
<td>
<p>An expression matrix in which rows are genes and
columns are cells.</p>
</td></tr>
<tr><td><code id="find_markers_+3A_cluster_labels">cluster_labels</code></td>
<td>
<p>A vector of cluster labels obtained from clustering methods.</p>
</td></tr>
<tr><td><code id="find_markers_+3A_identity">identity</code></td>
<td>
<p>A parameter to select specific cluster <code>identity = 1</code>
by default.</p>
</td></tr>
<tr><td><code id="find_markers_+3A_threads">threads</code></td>
<td>
<p>A parameter to control number of cores used for analysis
<code>threads = 1</code> by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains p-value and fold-change ratio for all genes of each cluster.
</p>

<hr>
<h2 id='find_specific_marker'>find_specific_marker</h2><span id='topic+find_specific_marker'></span>

<h3>Description</h3>

<p>Calculate cluster and cell type similarity based on the markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_specific_marker(gene_name, f_list, type = "jacc")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_specific_marker_+3A_gene_name">gene_name</code></td>
<td>
<p>A list of markers belong to the cluster.</p>
</td></tr>
<tr><td><code id="find_specific_marker_+3A_f_list">f_list</code></td>
<td>
<p>A list of markers belongs to a reference cell type.</p>
</td></tr>
<tr><td><code id="find_specific_marker_+3A_type">type</code></td>
<td>
<p>A parameter to select the method to measure cluster and cell type
similarity
</p>

<ul>
<li><p> jacc - Jaccard index.
</p>
</li>
<li><p> ac - Accuracy.
</p>
</li>
<li><p> f1 - F1 score.
</p>
</li></ul>
<p>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of probabilties of a cluster belongs to cell types.
</p>

<hr>
<h2 id='get_cluster_markers'>get_cluster_markers</h2><span id='topic+get_cluster_markers'></span>

<h3>Description</h3>

<p>Find markers for each cluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cluster_markers(input_data_matrix, labels_vector, threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cluster_markers_+3A_input_data_matrix">input_data_matrix</code></td>
<td>
<p>An expression matrix in which rows are genes and
columns are cells.</p>
</td></tr>
<tr><td><code id="get_cluster_markers_+3A_labels_vector">labels_vector</code></td>
<td>
<p>A vector of cluster labels ontained from clustering methods.</p>
</td></tr>
<tr><td><code id="get_cluster_markers_+3A_threads">threads</code></td>
<td>
<p>A parameter to control number of cores used for analysis
<code>threads = 1</code> by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains markers for each cluster.
</p>

<hr>
<h2 id='scCAN'>scCAN</h2><span id='topic+scCAN'></span>

<h3>Description</h3>

<p>This is the main function to perform sc-RNA seq data clustering clustering. scCAN is fully unsupervised scRNA-seq
clustering framework that uses deep neural network and network fusion-based clustering algorithm.
First, scCAN applies a non-negative autoencoder to filter scRNA-seq data.
Second, the filtered data is passed to stacked Bayesian autoencoder to get multiple low-dimensional representations of input data.
Subsequently, scCAN converts these compressed data into networks and unify those networks to a single graph.
Then, scCAN uses a spectral clustering algorithm to obtain final clusters assignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scCAN(
  data,
  sparse = FALSE,
  n.neighbors = 30,
  alpha = 0.5,
  n.iters = 10,
  ncores = 10,
  r.seed = 1,
  subsamp = TRUE,
  k = 2:15,
  samp.size = 5000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scCAN_+3A_data">data</code></td>
<td>
<p>Gene expression matrix, with rows represent samples and columns represent genes.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_sparse">sparse</code></td>
<td>
<p>Boolen variable indicating whether data is a sparse matrix. The input must be a non negative sparse matrix.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_n.neighbors">n.neighbors</code></td>
<td>
<p>Number of neighboring cells that are used to caculate the edge's weight. The number of neighbors are set <code>n.neighbors = 30</code> by default.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_alpha">alpha</code></td>
<td>
<p>A hyper parameter that control the weight of graph. This values is set to <code>alpha = 0.5</code> by default.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_n.iters">n.iters</code></td>
<td>
<p>A hyper-parameter to set the number of network fusion iterations. It is set to <code>n.iters = 10</code> by default.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_ncores">ncores</code></td>
<td>
<p>Number of processor cores to use.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_r.seed">r.seed</code></td>
<td>
<p>A parameter to set a seed for reproducibility. This values is set to <code>r.seed = 1</code> by default.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_subsamp">subsamp</code></td>
<td>
<p>Enable subsampling process for big data. This values is set to <code>subsamp = T</code> by default.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_k">k</code></td>
<td>
<p>A vector to search for optimal number of cluster.</p>
</td></tr>
<tr><td><code id="scCAN_+3A_samp.size">samp.size</code></td>
<td>
<p>A parameter to control number of sub-sampled cells.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with the following keys:
</p>

<ul>
<li><p> cluster - A numeric vector containing cluster assignment for each sample.
</p>
</li>
<li><p> k  - The optimal number of cluster.
</p>
</li>
<li><p> latent - The latent data generated from autoencoders.
</p>
</li></ul>



<h3>References</h3>

<p>1. Duc Tran, Hung Nguyen, Bang Tran, Carlo La Vecchia, Hung N. Luu, Tin Nguyen (2021). Fast and precise single-cell data analysis using a hierarchical autoencoder. Nature Communications, 12, 1029. doi: 10.1038/s41467-021-21312-2
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Not run if scDHA has not installed yet.
# Load the package and the example data (SCE dataset)
library(scCAN)
#Load example data
data("SCE")

#Get data matrix and label
data &lt;- t(SCE$data); label &lt;- as.character(SCE$cell_type1)

#Generate clustering result, the input matrix has rows as samples and columns as genes
result &lt;- scCAN(data, r.seed = 1)

#Get the clustering result
cluster &lt;- result$cluster

#Calculate adjusted Rand Index
ari &lt;- round(scCAN::adjustedRandIndex(cluster,label), 2)
message(paste0("ARI = ", ari))


## End(Not run)
</code></pre>

<hr>
<h2 id='SCE'>SCE</h2><span id='topic+SCE'></span>

<h3>Description</h3>

<p>SCE dataset includes scRNA-seq data and cell type information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCE
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
