<!DOCTYPE html><html><head><title>Help for package sccca</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sccca}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculate_cor_mat'><p>Performs aggregation based on cell clusters and condition. Then, it calculates correlation matrix of genes</p></a></li>
<li><a href='#calculate_normalized_ratio'><p>Calculate cell scores based on number of genes</p></a></li>
<li><a href='#correct_gene_symbols'><p>Process the cell markers names</p></a></li>
<li><a href='#enrich_genes'><p>Performs parallel function on two lists</p></a></li>
<li><a href='#filter_correlation'><p>Filter the genes based on specific correlation threshold</p></a></li>
<li><a href='#filter_list'><p>Process the cell markers that overlap between the cell markers and scRNA matrix</p></a></li>
<li><a href='#fisher_test'><p>Performs fisher exact test to get the significant overlap between genes for cell type assignment</p></a></li>
<li><a href='#match_characters'><p>Process the cell markers that pass specific threshold in the gene correlation matrix</p></a></li>
<li><a href='#phyper_test'><p>Performs phyper test to get the significant overlap between genes for cell type assignment</p></a></li>
<li><a href='#process_clus'><p>Gets the associated cell types using correlation-based approach</p></a></li>
<li><a href='#process_database'><p>Process the database for the sctype approach</p></a></li>
<li><a href='#process_markers'><p>Process the cell markers database and return the processed list</p></a></li>
<li><a href='#sccca'><p>Run the pipeline for the cell type assignment</p></a></li>
<li><a href='#sctype'><p>Run the sctype approach as it's implemented by Ianevski, A., Giri, A.K. and Aittokallio, T.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single-Cell Correlation Based Cell Type Annotation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mohamed Soudy &lt;Mohmedsoudy2009@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performing cell type annotation based on cell markers from a unified database. The approach utilizes correlation-based approach combined with association analysis using Fisher-exact and phyper statistical tests (Upton, Graham JG. (1992) &lt;<a href="https://doi.org/10.2307%2F2982890">doi:10.2307/2982890</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>Seurat, dplyr, plyr, scales, HGNChelper, openxlsx</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-13 10:48:54 UTC; mohamed.soudy</td>
</tr>
<tr>
<td>Author:</td>
<td>Mohamed Soudy [aut, cre],
  Sophie LE BARS [aut],
  Enrico Glaab [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-13 11:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculate_cor_mat'>Performs aggregation based on cell clusters and condition. Then, it calculates correlation matrix of genes</h2><span id='topic+calculate_cor_mat'></span>

<h3>Description</h3>

<p>This Function is used to perform cell aggregation by averaging the expression of scRNA-seq matrix and then perform correlation matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_cor_mat(expression_mat, condition  = NULL, clusters, assay = "RNA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_cor_mat_+3A_expression_mat">expression_mat</code></td>
<td>
<p>Seurat object that contains the expression matrix.</p>
</td></tr>
<tr><td><code id="calculate_cor_mat_+3A_condition">condition</code></td>
<td>
<p>column name of the condition in th meta data of the Seurat object.</p>
</td></tr>
<tr><td><code id="calculate_cor_mat_+3A_clusters">clusters</code></td>
<td>
<p>column name of the cluster numbers in the meta data of the Seurat object.</p>
</td></tr>
<tr><td><code id="calculate_cor_mat_+3A_assay">assay</code></td>
<td>
<p>the assay to be used default is set to RNA</p>
</td></tr>
</table>


<h3>Value</h3>

<p>correlation matrix of genes.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='calculate_normalized_ratio'>Calculate cell scores based on number of genes</h2><span id='topic+calculate_normalized_ratio'></span>

<h3>Description</h3>

<p>This Function is used to calculate cell scores based on number of genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_normalized_ratio(vec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_normalized_ratio_+3A_vec">vec</code></td>
<td>
<p>list of genes of cell types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of cell scores based on the number of overlapped genes with the input matrix.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='correct_gene_symbols'>Process the cell markers names</h2><span id='topic+correct_gene_symbols'></span>

<h3>Description</h3>

<p>This Function is used to return the cell markers names processed for the sctype approach
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correct_gene_symbols(markers)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correct_gene_symbols_+3A_markers">markers</code></td>
<td>
<p>list of unique cell markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of genes names which overlap with the correlation matrix.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='enrich_genes'>Performs parallel function on two lists</h2><span id='topic+enrich_genes'></span>

<h3>Description</h3>

<p>This Function is used to perform parallel function on two lists
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrich_genes(ref_list, overlap_list, func)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enrich_genes_+3A_ref_list">ref_list</code></td>
<td>
<p>reference list.</p>
</td></tr>
<tr><td><code id="enrich_genes_+3A_overlap_list">overlap_list</code></td>
<td>
<p>overlap list.</p>
</td></tr>
<tr><td><code id="enrich_genes_+3A_func">func</code></td>
<td>
<p>function to be applied.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list where each element is the result of applying the function 'func' to the corresponding elements of 'ref_list' and 'overlap_list'.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='filter_correlation'>Filter the genes based on specific correlation threshold</h2><span id='topic+filter_correlation'></span>

<h3>Description</h3>

<p>This Function is used to filter the gene correlation matrix based on user-defined threshold
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_correlation(cor_mat, gene_list, threshold = 0.7)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_correlation_+3A_cor_mat">cor_mat</code></td>
<td>
<p>correlation matrix generated from calculate_cor_mat function.</p>
</td></tr>
<tr><td><code id="filter_correlation_+3A_gene_list">gene_list</code></td>
<td>
<p>cell markers that passed threshold.</p>
</td></tr>
<tr><td><code id="filter_correlation_+3A_threshold">threshold</code></td>
<td>
<p>absolute correlation threshold.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of gene names that pass user-defined correlation threshold.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='filter_list'>Process the cell markers that overlap between the cell markers and scRNA matrix</h2><span id='topic+filter_list'></span>

<h3>Description</h3>

<p>This Function is used to return the cell markers that overlap between the cell markers and scRNA matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_list(gene_list, passed_cells)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_list_+3A_gene_list">gene_list</code></td>
<td>
<p>list of unique genes of cell types.</p>
</td></tr>
<tr><td><code id="filter_list_+3A_passed_cells">passed_cells</code></td>
<td>
<p>cells types that pass the specified threshold.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of cell types which genes are found in the input matrix.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='fisher_test'>Performs fisher exact test to get the significant overlap between genes for cell type assignment</h2><span id='topic+fisher_test'></span>

<h3>Description</h3>

<p>This Function is used to perform fisher exact test to get cell types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisher_test(ref, gene_overlap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fisher_test_+3A_ref">ref</code></td>
<td>
<p>reference gene set.</p>
</td></tr>
<tr><td><code id="fisher_test_+3A_gene_overlap">gene_overlap</code></td>
<td>
<p>genes that pass the correlation threshold.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of p-value and overlap.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fisher_test(c("PAX8","PAX6","TP53","AOC3","LIPF"), c("LIPF","PAX8","PAX6","TP53","TSHB","AOC3"))

</code></pre>

<hr>
<h2 id='match_characters'>Process the cell markers that pass specific threshold in the gene correlation matrix</h2><span id='topic+match_characters'></span>

<h3>Description</h3>

<p>This Function is used to return the cell markers that pass specific threshold in the gene correlation matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_characters(genes, gene_mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_characters_+3A_genes">genes</code></td>
<td>
<p>list of unique genes of cell types.</p>
</td></tr>
<tr><td><code id="match_characters_+3A_gene_mat">gene_mat</code></td>
<td>
<p>correlation matrix of genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of genes names which overlap with the correlation matrix.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='phyper_test'>Performs phyper test to get the significant overlap between genes for cell type assignment</h2><span id='topic+phyper_test'></span>

<h3>Description</h3>

<p>This Function is used to perform phyper test to get cell types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyper_test(ref, overlap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyper_test_+3A_ref">ref</code></td>
<td>
<p>reference gene set.</p>
</td></tr>
<tr><td><code id="phyper_test_+3A_overlap">overlap</code></td>
<td>
<p>genes that pass the correlation threshold.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of p-value and overlap.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phyper_test(c("PAX8","PAX6","TP53","AOC3","LIPF"), c("LIPF","PAX8","PAX6","TP53","TSHB","AOC3"))

</code></pre>

<hr>
<h2 id='process_clus'>Gets the associated cell types using correlation-based approach</h2><span id='topic+process_clus'></span>

<h3>Description</h3>

<p>This Function is used to get the associated cell clusters using correlation-based approach
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_clus(cluster,sobj,assay="RNA",clus,markers,cor_m,m_t=0.9,c_t=0.7,test="p")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_clus_+3A_cluster">cluster</code></td>
<td>
<p>associated cluster name.</p>
</td></tr>
<tr><td><code id="process_clus_+3A_sobj">sobj</code></td>
<td>
<p>Seurat object.</p>
</td></tr>
<tr><td><code id="process_clus_+3A_assay">assay</code></td>
<td>
<p>assay to be used default is set to RNA.</p>
</td></tr>
<tr><td><code id="process_clus_+3A_clus">clus</code></td>
<td>
<p>cell clusters.</p>
</td></tr>
<tr><td><code id="process_clus_+3A_markers">markers</code></td>
<td>
<p>cell markers database.</p>
</td></tr>
<tr><td><code id="process_clus_+3A_cor_m">cor_m</code></td>
<td>
<p>gene correlation matrix.</p>
</td></tr>
<tr><td><code id="process_clus_+3A_m_t">m_t</code></td>
<td>
<p>overlap threshold between cell markers and expression matrix.</p>
</td></tr>
<tr><td><code id="process_clus_+3A_c_t">c_t</code></td>
<td>
<p>correlation threshold between genes.</p>
</td></tr>
<tr><td><code id="process_clus_+3A_test">test</code></td>
<td>
<p>statistical test that check if overlap is significant could be &quot;p&quot; for phyper or &quot;f&quot; for fisher.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame of proposed cell types.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='process_database'>Process the database for the sctype approach</h2><span id='topic+process_database'></span>

<h3>Description</h3>

<p>This Function is used to process the database that will be used for sctype approach
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_database(database_name = "sctype", org = 'a', tissue, tissue_type = 'n')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_database_+3A_database_name">database_name</code></td>
<td>
<p>name of the database to be used that can be 'sctype' or 'UMD'.</p>
</td></tr>
<tr><td><code id="process_database_+3A_org">org</code></td>
<td>
<p>name of organism to be used that can be 'h' for human, 'm' for mouse, and 'a' for all markers.</p>
</td></tr>
<tr><td><code id="process_database_+3A_tissue">tissue</code></td>
<td>
<p>specified tissue from which the data comes.</p>
</td></tr>
<tr><td><code id="process_database_+3A_tissue_type">tissue_type</code></td>
<td>
<p>tissue type whether 'a' for all types 'n' for normal tissues only or &quot;c&quot; for cancer tissues.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of genes names which overlap with the correlation matrix.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='process_markers'>Process the cell markers database and return the processed list</h2><span id='topic+process_markers'></span>

<h3>Description</h3>

<p>This Function is used to process the cell markers database and return the processed list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_markers(markers_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_markers_+3A_markers_df">markers_df</code></td>
<td>
<p>data frame with markers named as gene_original and cell names as cell type.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of lists of the processed markers
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='sccca'>Run the pipeline for the cell type assignment</h2><span id='topic+sccca'></span>

<h3>Description</h3>

<p>This Function is used to run the main pipeline that does the cell type assignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sccca(sobj,assay="RNA",cluster,marker,tissue,tt="a",cond,m_t=0.9,c_t=0.7,test="p",org="a")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sccca_+3A_sobj">sobj</code></td>
<td>
<p>Seurat object.</p>
</td></tr>
<tr><td><code id="sccca_+3A_assay">assay</code></td>
<td>
<p>assay to be used default is set to RNA.</p>
</td></tr>
<tr><td><code id="sccca_+3A_cluster">cluster</code></td>
<td>
<p>colname in the mata.data that have the cell cluster numbers.</p>
</td></tr>
<tr><td><code id="sccca_+3A_marker">marker</code></td>
<td>
<p>cell markers database path.</p>
</td></tr>
<tr><td><code id="sccca_+3A_tissue">tissue</code></td>
<td>
<p>specified tissue from which the data comes.</p>
</td></tr>
<tr><td><code id="sccca_+3A_tt">tt</code></td>
<td>
<p>tissue type whether 'a' for all types 'n' for normal tissues only or &quot;c&quot; for cancer tissues.</p>
</td></tr>
<tr><td><code id="sccca_+3A_cond">cond</code></td>
<td>
<p>colname in the meta.data that have the condition names.</p>
</td></tr>
<tr><td><code id="sccca_+3A_m_t">m_t</code></td>
<td>
<p>overlap threshold between cell markers and expression matrix.</p>
</td></tr>
<tr><td><code id="sccca_+3A_c_t">c_t</code></td>
<td>
<p>correlation threshold between genes.</p>
</td></tr>
<tr><td><code id="sccca_+3A_test">test</code></td>
<td>
<p>statistical test that check if overlap is significant could be &quot;p&quot; for phyper or &quot;f&quot; for fisher.</p>
</td></tr>
<tr><td><code id="sccca_+3A_org">org</code></td>
<td>
<p>organism to be used that can be 'h' for human, 'm' for mouse, and 'a' for all markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of Seurat object that have the assigned clusters, and top 3 proposed cell types.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

<hr>
<h2 id='sctype'>Run the sctype approach as it's implemented by Ianevski, A., Giri, A.K. and Aittokallio, T.</h2><span id='topic+sctype'></span>

<h3>Description</h3>

<p>This Function is used to run the sctype approach with faster implementation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sctype(sobj,assay="RNA",tissue,tt="a",clus,org="a",scaled=T,database="sctype")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sctype_+3A_sobj">sobj</code></td>
<td>
<p>Seurat object.</p>
</td></tr>
<tr><td><code id="sctype_+3A_assay">assay</code></td>
<td>
<p>assay to be used default is set to RNA.</p>
</td></tr>
<tr><td><code id="sctype_+3A_tissue">tissue</code></td>
<td>
<p>specified tissue from which the data comes.</p>
</td></tr>
<tr><td><code id="sctype_+3A_tt">tt</code></td>
<td>
<p>tissue type whether 'a' for all types 'n' for normal tissues only or &quot;c&quot; for cancer tissues.</p>
</td></tr>
<tr><td><code id="sctype_+3A_clus">clus</code></td>
<td>
<p>colname in the mata.data that have the cell cluster numbers.</p>
</td></tr>
<tr><td><code id="sctype_+3A_org">org</code></td>
<td>
<p>organism to be used that can be 'h' for human, 'm' for mouse, and 'a' for all markers.</p>
</td></tr>
<tr><td><code id="sctype_+3A_scaled">scaled</code></td>
<td>
<p>indicates whether the matrix is scaled (TRUE by default)</p>
</td></tr>
<tr><td><code id="sctype_+3A_database">database</code></td>
<td>
<p>name of the database to be used that can be 'sctype' or 'UMD'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of genes names which overlap with the correlation matrix.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@uni.lu">Mohamed.soudy@uni.lu</a> and Sohpie LE BARS <a href="mailto:sophie.lebars@uni.lu">sophie.lebars@uni.lu</a> and Enrico Glaab <a href="mailto:enrico.glaab@uni.lu">enrico.glaab@uni.lu</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
