<!DOCTYPE html><html><head><title>Help for package scCustomize</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scCustomize}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Add_Alt_Feature_ID'><p>Add Alternative Feature IDs</p></a></li>
<li><a href='#Add_Cell_Complexity'><p>Add Cell Complexity</p></a></li>
<li><a href='#Add_Cell_QC_Metrics'><p>Add Multiple Cell Quality Control Values with Single Function</p></a></li>
<li><a href='#Add_CellBender_Diff'><p>Calculate and add differences post-cell bender analysis</p></a></li>
<li><a href='#Add_Mito_Ribo'><p>Add Mito and Ribo percentages</p></a></li>
<li><a href='#Add_Pct_Diff'><p>Add percentage difference to DE results</p></a></li>
<li><a href='#Add_Sample_Meta'><p>Add Sample Level Meta Data</p></a></li>
<li><a href='#Add_Top_Gene_Pct_Seurat'><p>Add Percent of High Abundance Genes</p></a></li>
<li><a href='#as.anndata'><p>Convert objects to anndata objects</p></a></li>
<li><a href='#as.LIGER'><p>Convert objects to LIGER objects</p></a></li>
<li><a href='#as.Seurat.liger'><p>Convert objects to <code>Seurat</code> objects</p></a></li>
<li><a href='#Barcode_Plot'><p>Create Barcode Rank Plot</p></a></li>
<li><a href='#Blank_Theme'><p>Blank Theme</p></a></li>
<li><a href='#Case_Check'><p>Check for alternate case features</p>
Checks Seurat object for the presence of features with the same spelling but alternate case.</a></li>
<li><a href='#Cell_Highlight_Plot'><p>Meta Highlight Plot</p></a></li>
<li><a href='#CellBender_Diff_Plot'><p>Plot Number of Cells/Nuclei per Sample</p></a></li>
<li><a href='#CellBender_Feature_Diff'><p>CellBender Feature Differences</p></a></li>
<li><a href='#Change_Delim_All'><p>Change all delimiters in cell name</p></a></li>
<li><a href='#Change_Delim_Prefix'><p>Change barcode prefix delimiter</p></a></li>
<li><a href='#Change_Delim_Suffix'><p>Change barcode suffix delimiter</p></a></li>
<li><a href='#CheckMatrix_scCustom'><p>Check Matrix Validity</p></a></li>
<li><a href='#Cluster_Highlight_Plot'><p>Cluster Highlight Plot</p></a></li>
<li><a href='#Cluster_Stats_All_Samples'><p>Calculate Cluster Stats</p></a></li>
<li><a href='#Clustered_DotPlot'><p>Clustered DotPlot</p></a></li>
<li><a href='#ColorBlind_Pal'><p>Color Universal Design Short Palette</p></a></li>
<li><a href='#Convert_Assay'><p>Convert between Seurat Assay types</p></a></li>
<li><a href='#Copy_From_GCP'><p>Copy folder from GCP bucket from R Console</p></a></li>
<li><a href='#Copy_To_GCP'><p>Copy folder to GCP bucket from R Console</p></a></li>
<li><a href='#Create_10X_H5'><p>Create H5 from 10X Outputs</p></a></li>
<li><a href='#Create_CellBender_Merged_Seurat'><p>Create Seurat Object with Cell Bender and Raw data</p></a></li>
<li><a href='#Create_Cluster_Annotation_File'><p>Create cluster annotation csv file</p></a></li>
<li><a href='#Dark2_Pal'><p>Dark2 Palette</p></a></li>
<li><a href='#DimPlot_All_Samples'><p>DimPlot by Meta Data Column</p></a></li>
<li><a href='#DimPlot_LIGER'><p>DimPlot LIGER Version</p></a></li>
<li><a href='#DimPlot_scCustom'><p>DimPlot with modified default settings</p></a></li>
<li><a href='#DiscretePalette_scCustomize'><p>Discrete color palettes</p></a></li>
<li><a href='#DotPlot_scCustom'><p>Customized DotPlot</p></a></li>
<li><a href='#ensembl_mito_id'><p>Ensembl Mito IDs</p></a></li>
<li><a href='#ensembl_ribo_id'><p>Ensembl Ribo IDs</p></a></li>
<li><a href='#Extract_Modality'><p>Extract multi-modal data into list by modality</p></a></li>
<li><a href='#Extract_Sample_Meta'><p>Extract sample level meta.data</p></a></li>
<li><a href='#Extract_Top_Markers'><p>Extract Top N Marker Genes</p></a></li>
<li><a href='#Feature_Present'><p>Check if genes/features are present</p></a></li>
<li><a href='#FeaturePlot_DualAssay'><p>Customize FeaturePlot of two assays</p></a></li>
<li><a href='#FeaturePlot_scCustom'><p>Customize FeaturePlot</p></a></li>
<li><a href='#FeatureScatter_scCustom'><p>Modified version of FeatureScatter</p></a></li>
<li><a href='#Fetch_Meta'><p>Get meta data from object</p></a></li>
<li><a href='#Gene_Present'><p>Check if genes/features are present  <a href="https://lifecycle.r-lib.org/articles/stages.html#soft-deprecated"><img src="../help/figures/lifecycle-soft-deprecated.svg" alt='[Soft-deprecated]' /></a></p></a></li>
<li><a href='#Hue_Pal'><p>Hue_Pal</p></a></li>
<li><a href='#ieg_gene_list'><p>Immediate Early Gene (IEG) gene lists</p></a></li>
<li><a href='#Iterate_Barcode_Rank_Plot'><p>Iterative Barcode Rank Plots</p></a></li>
<li><a href='#Iterate_Cluster_Highlight_Plot'><p>Iterate Cluster Highlight Plot</p></a></li>
<li><a href='#Iterate_DimPlot_bySample'><p>Iterate DimPlot By Sample</p></a></li>
<li><a href='#Iterate_FeaturePlot_scCustom'><p>Iterative Plotting of Gene Lists using Custom FeaturePlots</p></a></li>
<li><a href='#Iterate_Meta_Highlight_Plot'><p>Iterate Meta Highlight Plot</p></a></li>
<li><a href='#Iterate_PC_Loading_Plots'><p>Iterate PC Loading Plots</p></a></li>
<li><a href='#Iterate_Plot_Density_Custom'><p>Iterative Plotting of Gene Lists using Custom Density Plots</p></a></li>
<li><a href='#Iterate_Plot_Density_Joint'><p>Iterative Plotting of Gene Lists using Custom Joint Density Plots</p></a></li>
<li><a href='#Iterate_VlnPlot_scCustom'><p>Iterative Plotting of Gene Lists using VlnPlot_scCustom</p></a></li>
<li><a href='#JCO_Four'><p>Four Color Palette (JCO)</p></a></li>
<li><a href='#LIGER_Features'><p>Extract Features from LIGER Object</p></a></li>
<li><a href='#Liger_to_Seurat'><p>Create a Seurat object containing the data from a liger object <a href="https://lifecycle.r-lib.org/articles/stages.html#soft-deprecated"><img src="../help/figures/lifecycle-soft-deprecated.svg" alt='[Soft-deprecated]' /></a></p></a></li>
<li><a href='#MAD_Stats'><p>Median Absolute Deviation Statistics</p></a></li>
<li><a href='#Median_Stats'><p>Median Statistics</p></a></li>
<li><a href='#Merge_Seurat_List'><p>Merge a list of Seurat Objects</p></a></li>
<li><a href='#Merge_Sparse_Data_All'><p>Merge a list of Sparse Matrices</p></a></li>
<li><a href='#Merge_Sparse_Multimodal_All'><p>Merge a list of Sparse Matrices contain multi-modal data.</p></a></li>
<li><a href='#Meta_Highlight_Plot'><p>Meta Highlight Plot</p></a></li>
<li><a href='#Meta_Numeric'><p>Check if meta data columns are numeric</p></a></li>
<li><a href='#Meta_Present'><p>Check if meta data are present</p></a></li>
<li><a href='#Meta_Remove_Seurat'><p>Remove meta data columns containing Seurat Defaults</p></a></li>
<li><a href='#Move_Legend'><p>Move Legend Position</p></a></li>
<li><a href='#msigdb_qc_gene_list'><p>QC Gene Lists</p></a></li>
<li><a href='#NavyAndOrange'><p>Navy and Orange Dual Color Palette</p></a></li>
<li><a href='#PalettePlot'><p>Plot color palette in viewer</p></a></li>
<li><a href='#PC_Plotting'><p>PC Plots</p></a></li>
<li><a href='#Percent_Expressing'><p>Calculate percent of expressing cells</p></a></li>
<li><a href='#Plot_Cells_per_Sample'><p>Plot Number of Cells/Nuclei per Sample</p></a></li>
<li><a href='#Plot_Density_Custom'><p>Nebulosa Density Plot</p></a></li>
<li><a href='#Plot_Density_Joint_Only'><p>Nebulosa Joint Density Plot</p></a></li>
<li><a href='#Plot_Median_Genes'><p>Plot Median Genes per Cell per Sample</p></a></li>
<li><a href='#Plot_Median_Mito'><p>Plot Median Percent Mito per Cell per Sample</p></a></li>
<li><a href='#Plot_Median_Other'><p>Plot Median other variable per Cell per Sample</p></a></li>
<li><a href='#Plot_Median_UMIs'><p>Plot Median UMIs per Cell per Sample</p></a></li>
<li><a href='#plotFactors_scCustom'><p>Customized version of plotFactors</p></a></li>
<li><a href='#Pull_Cluster_Annotation'><p>Pull cluster information from annotation csv file.</p></a></li>
<li><a href='#Pull_Directory_List'><p>Pull Directory List</p></a></li>
<li><a href='#QC_Histogram'><p>QC Histogram Plots</p></a></li>
<li><a href='#QC_Plot_GenevsFeature'><p>QC Plots Genes vs Misc</p></a></li>
<li><a href='#QC_Plot_UMIvsFeature'><p>QC Plots UMI vs Misc</p></a></li>
<li><a href='#QC_Plot_UMIvsGene'><p>QC Plots Genes vs UMIs</p></a></li>
<li><a href='#QC_Plots_Combined_Vln'><p>QC Plots Genes, UMIs, &amp; % Mito</p></a></li>
<li><a href='#QC_Plots_Complexity'><p>QC Plots Cell &quot;Complexity&quot;</p></a></li>
<li><a href='#QC_Plots_Feature'><p>QC Plots Feature</p></a></li>
<li><a href='#QC_Plots_Genes'><p>QC Plots Genes</p></a></li>
<li><a href='#QC_Plots_Mito'><p>QC Plots Mito</p></a></li>
<li><a href='#QC_Plots_UMIs'><p>QC Plots UMIs</p></a></li>
<li><a href='#Read_CellBender_h5_Mat'><p>Load CellBender h5 matrices (corrected)</p></a></li>
<li><a href='#Read_CellBender_h5_Multi_Directory'><p>Load CellBender h5 matrices (corrected) from multiple directories</p></a></li>
<li><a href='#Read_CellBender_h5_Multi_File'><p>Load CellBender h5 matrices (corrected) from multiple files</p></a></li>
<li><a href='#Read_GEO_Delim'><p>Load in NCBI GEO data formatted as single file per sample</p></a></li>
<li><a href='#Read_Metrics_10X'><p>Read Overall Statistics from 10X Cell Ranger Count</p></a></li>
<li><a href='#Read10X_GEO'><p>Load in NCBI GEO data from 10X</p></a></li>
<li><a href='#Read10X_h5_GEO'><p>Load in NCBI GEO data from 10X in HDF5 file format</p></a></li>
<li><a href='#Read10X_h5_Multi_Directory'><p>Load 10X h5 count matrices from multiple directories</p></a></li>
<li><a href='#Read10X_Multi_Directory'><p>Load 10X count matrices from multiple directories</p></a></li>
<li><a href='#Reduction_Loading_Present'><p>Check if reduction loadings are present</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#Rename_Clusters'><p>Rename Cluster Seurat</p></a></li>
<li><a href='#Replace_Suffix'><p>Replace barcode suffixes</p></a></li>
<li><a href='#scCustomize_Palette'><p>Color Palette Selection for scCustomize</p></a></li>
<li><a href='#scCustomize-package'><p>scCustomize: Custom Visualizations &amp; Functions for Streamlined Analyses of Single Cell Sequencing</p></a></li>
<li><a href='#Seq_QC_Plot_Alignment_Combined'><p>QC Plots Sequencing metrics (Alignment) (Layout)</p></a></li>
<li><a href='#Seq_QC_Plot_Antisense'><p>QC Plots Sequencing metrics (Alignment)</p></a></li>
<li><a href='#Seq_QC_Plot_Basic_Combined'><p>QC Plots Sequencing metrics (Layout)</p></a></li>
<li><a href='#Seq_QC_Plot_Exonic'><p>QC Plots Sequencing metrics (Alignment)</p></a></li>
<li><a href='#Seq_QC_Plot_Genes'><p>QC Plots Sequencing metrics</p></a></li>
<li><a href='#Seq_QC_Plot_Genome'><p>QC Plots Sequencing metrics (Alignment)</p></a></li>
<li><a href='#Seq_QC_Plot_Intergenic'><p>QC Plots Sequencing metrics (Alignment)</p></a></li>
<li><a href='#Seq_QC_Plot_Intronic'><p>QC Plots Sequencing metrics (Alignment)</p></a></li>
<li><a href='#Seq_QC_Plot_Number_Cells'><p>QC Plots Sequencing metrics</p></a></li>
<li><a href='#Seq_QC_Plot_Reads_in_Cells'><p>QC Plots Sequencing metrics</p></a></li>
<li><a href='#Seq_QC_Plot_Reads_per_Cell'><p>QC Plots Sequencing metrics</p></a></li>
<li><a href='#Seq_QC_Plot_Saturation'><p>QC Plots Sequencing metrics</p></a></li>
<li><a href='#Seq_QC_Plot_Total_Genes'><p>QC Plots Sequencing metrics</p></a></li>
<li><a href='#Seq_QC_Plot_Transcriptome'><p>QC Plots Sequencing metrics (Alignment)</p></a></li>
<li><a href='#Seq_QC_Plot_UMIs'><p>QC Plots Sequencing metrics</p></a></li>
<li><a href='#Setup_scRNAseq_Project'><p>Setup project directory structure</p></a></li>
<li><a href='#Single_Color_Palette'><p>Single Color Palettes for Plotting</p></a></li>
<li><a href='#Split_FeatureScatter'><p>Deprecated functions <a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a></p></a></li>
<li><a href='#Split_Layers'><p>Split Seurat object into layers</p></a></li>
<li><a href='#Split_Vector'><p>Split vector into list</p></a></li>
<li><a href='#Stacked_VlnPlot'><p>Stacked Violin Plot</p></a></li>
<li><a href='#Store_Misc_Info_Seurat'><p>Store misc data in Seurat object</p></a></li>
<li><a href='#Store_Palette_Seurat'><p>Store color palette in Seurat object</p></a></li>
<li><a href='#theme_ggprism_mod'><p>Modified ggprism theme</p></a></li>
<li><a href='#Top_Genes_Factor'><p>Extract top loading genes for LIGER factor</p></a></li>
<li><a href='#UnRotate_X'><p>Unrotate x axis on VlnPlot</p></a></li>
<li><a href='#Updated_HGNC_Symbols'><p>Update HGNC Gene Symbols</p></a></li>
<li><a href='#Variable_Features_ALL_LIGER'><p>Perform variable gene selection over whole dataset</p></a></li>
<li><a href='#VariableFeaturePlot_scCustom'><p>Custom Labeled Variable Features Plot</p></a></li>
<li><a href='#viridis_plasma_dark_high'><p>Viridis Shortcuts</p></a></li>
<li><a href='#VlnPlot_scCustom'><p>VlnPlot with modified default settings</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Custom Visualizations &amp; Functions for Streamlined Analyses of
Single Cell Sequencing</td>
</tr>
<tr>
<td>Description:</td>
<td>Collection of functions created and/or curated to aid in the visualization and analysis of single-cell data using 'R'.  'scCustomize' aims to provide 1) Customized visualizations for aid in ease of use and to create more aesthetic and functional visuals. 2) Improve speed/reproducibility of common tasks/pieces of code in scRNA-seq analysis with a single or group of functions.  For citation please use: Marsh SE (2021) "Custom Visualizations &amp; Functions for Streamlined Analyses of Single Cell Sequencing" &lt;<a href="https://doi.org/10.5281%2Fzenodo.5706430">doi:10.5281/zenodo.5706430</a>&gt; RRID:SCR_024675.</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-27</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/samuel-marsh/scCustomize">https://github.com/samuel-marsh/scCustomize</a>,
<a href="https://samuel-marsh.github.io/scCustomize/">https://samuel-marsh.github.io/scCustomize/</a>,
<a href="https://doi.org/10.5281/zenodo.5706431">https://doi.org/10.5281/zenodo.5706431</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/samuel-marsh/scCustomize/issues">https://github.com/samuel-marsh/scCustomize/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), Seurat (&ge; 4.3.0.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>circlize, cli (&ge; 3.2.0), cowplot, data.table, dplyr, forcats,
ggbeeswarm, ggplot2, ggprism, ggrastr, ggrepel, glue,
grDevices, grid, janitor, lifecycle, magrittr, Matrix (&ge;
1.5.0), methods, paletteer, patchwork, pbapply, purrr, rlang
(&ge; 1.1.3), scales, scattermore (&ge; 1.2), SeuratObject (&ge;
5.0.0), stats, stringi, stringr, tibble, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocFileCache, ComplexHeatmap, dittoSeq, DropletUtils,
ggpubr, hdf5r, knitr, Nebulosa, remotes, reticulate, rliger,
rmarkdown, scuttle, tidyselect, qs, viridis</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-27 15:16:08 UTC; marsh_mbp</td>
</tr>
<tr>
<td>Author:</td>
<td>Samuel Marsh <a href="https://orcid.org/0000-0002-3012-6945"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ming Tang [ctb],
  Velina Kozareva [ctb],
  Lucas Graybuck [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samuel Marsh &lt;samuel.marsh@childrens.harvard.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-28 19:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Add_Alt_Feature_ID'>Add Alternative Feature IDs</h2><span id='topic+Add_Alt_Feature_ID'></span>

<h3>Description</h3>

<p>Add alternative feature ids to the assay level meta.data slot in Assay5 compatible object (Seurat V5.0.0 or greater)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Add_Alt_Feature_ID(
  seurat_object,
  features_tsv_file = NULL,
  hdf5_file = NULL,
  assay = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Add_Alt_Feature_ID_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Add_Alt_Feature_ID_+3A_features_tsv_file">features_tsv_file</code></td>
<td>
<p>output file from Cell Ranger used for creation of Seurat object.
(Either provide this of <code>hdf5_file</code>)</p>
</td></tr>
<tr><td><code id="Add_Alt_Feature_ID_+3A_hdf5_file">hdf5_file</code></td>
<td>
<p>output file from Cell Ranger used for creation of Seurat object.
(Either provide this of <code>features_tsv_file</code>)</p>
</td></tr>
<tr><td><code id="Add_Alt_Feature_ID_+3A_assay">assay</code></td>
<td>
<p>name of assay(s) to add the alternative features to.  Can specify &quot;all&quot;
to add to all assays.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat Object with new entries in the <code>obj@assays$ASSAY@meta.data</code> slot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Using features.tsv.gz file
   # Either file from filtered or raw outputs can be used as they are identical.
obj &lt;- Add_Alt_Feature_ID(seurat_object = obj,
features_tsv = "sample01/outs/filtered_feature_bc_matrix/features.tsv.gz", assay = "RNA")

#' # Using hdf5 file
   # Either filtered_feature_bc or raw_feature_bc can be used as the features slot is identical
   # Though it is faster to load filtered_feature_bc file due to droplet filtering
obj &lt;- Add_Alt_Feature_ID(seurat_object = obj,
hdf5_file = "sample01/outs/outs/filtered_feature_bc_matrix.h5", assay = "RNA")

## End(Not run)

</code></pre>

<hr>
<h2 id='Add_Cell_Complexity'>Add Cell Complexity</h2><span id='topic+Add_Cell_Complexity'></span><span id='topic+Add_Cell_Complexity.liger'></span><span id='topic+Add_Cell_Complexity.Seurat'></span>

<h3>Description</h3>

<p>Add measure of cell complexity/novelty (log10GenesPerUMI) for data QC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Add_Cell_Complexity(object, ...)

## S3 method for class 'liger'
Add_Cell_Complexity(
  object,
  meta_col_name = "log10GenesPerUMI",
  overwrite = FALSE,
  ...
)

## S3 method for class 'Seurat'
Add_Cell_Complexity(
  object,
  meta_col_name = "log10GenesPerUMI",
  assay = "RNA",
  overwrite = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Add_Cell_Complexity_+3A_object">object</code></td>
<td>
<p>Seurat or LIGER object</p>
</td></tr>
<tr><td><code id="Add_Cell_Complexity_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Add_Cell_Complexity_+3A_meta_col_name">meta_col_name</code></td>
<td>
<p>name to use for new meta data column.  Default is &quot;log10GenesPerUMI&quot;.</p>
</td></tr>
<tr><td><code id="Add_Cell_Complexity_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical.  Whether to overwrite existing an meta.data column.  Default is FALSE meaning that
function will abort if column with name provided to <code>meta_col_name</code> is present in meta.data slot.</p>
</td></tr>
<tr><td><code id="Add_Cell_Complexity_+3A_assay">assay</code></td>
<td>
<p>assay to use in calculation.  Default is &quot;RNA&quot;.  <em>Note</em> This should only be changed if
storing corrected and uncorrected assays in same object (e.g. outputs of both Cell Ranger and Cell Bender).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>object</code> with columns added to object meta data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Liger
liger_object &lt;- Add_Cell_Complexity(object = liger_object)

## End(Not run)

# Seurat
library(Seurat)
pbmc_small &lt;- Add_Cell_Complexity(object = pbmc_small)

</code></pre>

<hr>
<h2 id='Add_Cell_QC_Metrics'>Add Multiple Cell Quality Control Values with Single Function</h2><span id='topic+Add_Cell_QC_Metrics'></span>

<h3>Description</h3>

<p>Add Mito/Ribo %, Cell Complexity (log10GenesPerUMI), Top Gene Percent with single function call
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Add_Cell_QC_Metrics(
  seurat_object,
  add_mito_ribo = TRUE,
  add_complexity = TRUE,
  add_top_pct = TRUE,
  add_MSigDB = TRUE,
  add_IEG = TRUE,
  add_cell_cycle = TRUE,
  species,
  mito_name = "percent_mito",
  ribo_name = "percent_ribo",
  mito_ribo_name = "percent_mito_ribo",
  complexity_name = "log10GenesPerUMI",
  top_pct_name = NULL,
  oxphos_name = "percent_oxphos",
  apop_name = "percent_apop",
  dna_repair_name = "percent_dna_repair",
  ieg_name = "percent_ieg",
  mito_pattern = NULL,
  ribo_pattern = NULL,
  mito_features = NULL,
  ribo_features = NULL,
  ensembl_ids = FALSE,
  num_top_genes = 50,
  assay = NULL,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_add_mito_ribo">add_mito_ribo</code></td>
<td>
<p>logical, whether to add percentage of counts belonging to mitochondrial/ribosomal
genes to object (Default is TRUE).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_add_complexity">add_complexity</code></td>
<td>
<p>logical, whether to add Cell Complexity to object (Default is TRUE).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_add_top_pct">add_top_pct</code></td>
<td>
<p>logical, whether to add Top Gene Percentages to object (Default is TRUE).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_add_msigdb">add_MSigDB</code></td>
<td>
<p>logical, whether to add percentages of counts belonging to genes from of mSigDB hallmark
gene lists: &quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;, &quot;HALLMARK_APOPTOSIS&quot;, and &quot;HALLMARK_DNA_REPAIR&quot; to
object (Default is TRUE).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_add_ieg">add_IEG</code></td>
<td>
<p>logical, whether to add percentage of counts belonging to IEG genes to object (Default is TRUE).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_add_cell_cycle">add_cell_cycle</code></td>
<td>
<p>logical, whether to addcell cycle scores and phase based on
<code><a href="Seurat.html#topic+CellCycleScoring">CellCycleScoring</a></code>.  Only applicable if <code>species = "human"</code>.  (Default is TRUE).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_species">species</code></td>
<td>
<p>Species of origin for given Seurat Object.  If mouse, human, marmoset, zebrafish, rat,
drosophila, or rhesus macaque (name or abbreviation) are provided the function will automatically
generate mito_pattern and ribo_pattern values.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_mito_name">mito_name</code></td>
<td>
<p>name to use for the new meta.data column containing percent mitochondrial counts.
Default is &quot;percent_mito&quot;.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_ribo_name">ribo_name</code></td>
<td>
<p>name to use for the new meta.data column containing percent ribosomal counts.
Default is &quot;percent_ribo&quot;.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_mito_ribo_name">mito_ribo_name</code></td>
<td>
<p>name to use for the new meta.data column containing percent
mitochondrial+ribosomal counts.  Default is &quot;percent_mito_ribo&quot;.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_complexity_name">complexity_name</code></td>
<td>
<p>name to use for new meta data column for <code>Add_Cell_Complexity_Seurat</code>.
Default is &quot;log10GenesPerUMI&quot;.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_top_pct_name">top_pct_name</code></td>
<td>
<p>name to use for new meta data column for <code>Add_Top_Gene_Pct_Seurat</code>.
Default is &quot;percent_topXX&quot;, where XX is equal to the value provided to <code>num_top_genes</code>.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_oxphos_name">oxphos_name</code></td>
<td>
<p>name to use for new meta data column for percentage of MSigDB oxidative phosphorylation
counts.  Default is &quot;percent_oxphos&quot;.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_apop_name">apop_name</code></td>
<td>
<p>name to use for new meta data column for percentage of MSigDB apoptosis counts.
Default is &quot;percent_apop&quot;.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_dna_repair_name">dna_repair_name</code></td>
<td>
<p>name to use for new meta data column for percentage of MSigDB DNA repair
counts.  Default is &quot;percent_dna_repair&quot;..</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_ieg_name">ieg_name</code></td>
<td>
<p>name to use for new meta data column for percentage of IEG counts.  Default is &quot;percent_ieg&quot;.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_mito_pattern">mito_pattern</code></td>
<td>
<p>A regex pattern to match features against for mitochondrial genes (will set automatically if
species is mouse or human; marmoset features list saved separately).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_ribo_pattern">ribo_pattern</code></td>
<td>
<p>A regex pattern to match features against for ribosomal genes
(will set automatically if species is mouse, human, or marmoset).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_mito_features">mito_features</code></td>
<td>
<p>A list of mitochondrial gene names to be used instead of using regex pattern.
Will override regex pattern if both are present (including default saved regex patterns).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_ribo_features">ribo_features</code></td>
<td>
<p>A list of ribosomal gene names to be used instead of using regex pattern.
Will override regex pattern if both are present (including default saved regex patterns).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_ensembl_ids">ensembl_ids</code></td>
<td>
<p>logical, whether feature names in the object are gene names or
ensembl IDs (default is FALSE; set TRUE if feature names are ensembl IDs).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_num_top_genes">num_top_genes</code></td>
<td>
<p>An integer vector specifying the size(s) of the top set of high-abundance genes.
Used to compute the percentage of library size occupied by the most highly expressed genes in each cell.</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_assay">assay</code></td>
<td>
<p>assay to use in calculation.  Default is &quot;RNA&quot;.  <em>Note</em> This should only be changed if
storing corrected and uncorrected assays in same object (e.g. outputs of both Cell Ranger and Cell Bender).</p>
</td></tr>
<tr><td><code id="Add_Cell_QC_Metrics_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical.  Whether to overwrite existing an meta.data column.  Default is FALSE meaning that
function will abort if column with name provided to <code>meta_col_name</code> is present in meta.data slot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat Object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
obj &lt;- Add_Cell_QC_Metrics(seurat_object = obj, species = "Human")

## End(Not run)

</code></pre>

<hr>
<h2 id='Add_CellBender_Diff'>Calculate and add differences post-cell bender analysis</h2><span id='topic+Add_CellBender_Diff'></span>

<h3>Description</h3>

<p>Calculate the difference in features and UMIs per cell when both cell bender and raw assays are present.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Add_CellBender_Diff(seurat_object, raw_assay_name, cell_bender_assay_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Add_CellBender_Diff_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Add_CellBender_Diff_+3A_raw_assay_name">raw_assay_name</code></td>
<td>
<p>name of the assay containing the raw data.</p>
</td></tr>
<tr><td><code id="Add_CellBender_Diff_+3A_cell_bender_assay_name">cell_bender_assay_name</code></td>
<td>
<p>name of the assay containing the Cell Bender'ed data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat object with 2 new columns in the meta.data slot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
object &lt;- Add_CellBender_Diff(seurat_object = obj, raw_assay_name = "RAW",
cell_bender_assay_name = "RNA")

## End(Not run)

</code></pre>

<hr>
<h2 id='Add_Mito_Ribo'>Add Mito and Ribo percentages</h2><span id='topic+Add_Mito_Ribo'></span><span id='topic+Add_Mito_Ribo.liger'></span><span id='topic+Add_Mito_Ribo.Seurat'></span>

<h3>Description</h3>

<p>Add Mito, Ribo, &amp; Mito+Ribo percentages to meta.data slot of Seurat Object or
cell.data slot of Liger object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Add_Mito_Ribo(object, ...)

## S3 method for class 'liger'
Add_Mito_Ribo(
  object,
  species,
  mito_name = "percent_mito",
  ribo_name = "percent_ribo",
  mito_ribo_name = "percent_mito_ribo",
  mito_pattern = NULL,
  ribo_pattern = NULL,
  mito_features = NULL,
  ribo_features = NULL,
  ensembl_ids = FALSE,
  overwrite = FALSE,
  list_species_names = FALSE,
  ...
)

## S3 method for class 'Seurat'
Add_Mito_Ribo(
  object,
  species,
  mito_name = "percent_mito",
  ribo_name = "percent_ribo",
  mito_ribo_name = "percent_mito_ribo",
  mito_pattern = NULL,
  ribo_pattern = NULL,
  mito_features = NULL,
  ribo_features = NULL,
  ensembl_ids = FALSE,
  assay = NULL,
  overwrite = FALSE,
  list_species_names = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Add_Mito_Ribo_+3A_object">object</code></td>
<td>
<p>Seurat or LIGER object</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_species">species</code></td>
<td>
<p>Species of origin for given Seurat Object.  If mouse, human, marmoset, zebrafish, rat,
drosophila, or rhesus macaque (name or abbreviation) are provided the function will automatically
generate mito_pattern and ribo_pattern values.</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_mito_name">mito_name</code></td>
<td>
<p>name to use for the new meta.data column containing percent mitochondrial counts.
Default is &quot;percent_mito&quot;.</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_ribo_name">ribo_name</code></td>
<td>
<p>name to use for the new meta.data column containing percent ribosomal counts.
Default is &quot;percent_ribo&quot;.</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_mito_ribo_name">mito_ribo_name</code></td>
<td>
<p>name to use for the new meta.data column containing percent
mitochondrial+ribosomal counts.  Default is &quot;percent_mito_ribo&quot;.</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_mito_pattern">mito_pattern</code></td>
<td>
<p>A regex pattern to match features against for mitochondrial genes (will set automatically if
species is mouse or human; marmoset features list saved separately).</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_ribo_pattern">ribo_pattern</code></td>
<td>
<p>A regex pattern to match features against for ribosomal genes
(will set automatically if species is mouse, human, or marmoset).</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_mito_features">mito_features</code></td>
<td>
<p>A list of mitochondrial gene names to be used instead of using regex pattern.
Will override regex pattern if both are present (including default saved regex patterns).</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_ribo_features">ribo_features</code></td>
<td>
<p>A list of ribosomal gene names to be used instead of using regex pattern.
Will override regex pattern if both are present (including default saved regex patterns).</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_ensembl_ids">ensembl_ids</code></td>
<td>
<p>logical, whether feature names in the object are gene names or
ensembl IDs (default is FALSE; set TRUE if feature names are ensembl IDs).</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical.  Whether to overwrite existing meta.data columns.  Default is FALSE meaning that
function will abort if columns with any one of the names provided to <code>mito_name</code> <code>ribo_name</code> or
<code>mito_ribo_name</code> is present in meta.data slot.</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_list_species_names">list_species_names</code></td>
<td>
<p>returns list of all accepted values to use for default species names which
contain internal regex/feature lists (human, mouse, marmoset, zebrafish, rat, drosophila, and
rhesus macaque).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Add_Mito_Ribo_+3A_assay">assay</code></td>
<td>
<p>Assay to use (default is the current object default assay).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>object</code> with columns added to object meta data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Liger
liger_object &lt;- Add_Mito_Ribo(object = liger_object, species = "human")

## End(Not run)

## Not run: 
# Seurat
seurat_object &lt;- Add_Mito_Ribo(object = seurat_object, species = "human")

## End(Not run)

</code></pre>

<hr>
<h2 id='Add_Pct_Diff'>Add percentage difference to DE results</h2><span id='topic+Add_Pct_Diff'></span>

<h3>Description</h3>

<p>Adds new column labeled &quot;pct_diff&quot; to the data.frame output of <code><a href="Seurat.html#topic+FindMarkers">FindMarkers</a></code>,  <code><a href="Seurat.html#topic+FindAllMarkers">FindAllMarkers</a></code>, or other DE test data.frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Add_Pct_Diff(
  marker_dataframe,
  pct.1_name = "pct.1",
  pct.2_name = "pct.2",
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Add_Pct_Diff_+3A_marker_dataframe">marker_dataframe</code></td>
<td>
<p>data.frame containing the results of <code><a href="Seurat.html#topic+FindMarkers">FindMarkers</a></code>,  <code><a href="Seurat.html#topic+FindAllMarkers">FindAllMarkers</a></code>, or other DE test data.frame.</p>
</td></tr>
<tr><td><code id="Add_Pct_Diff_+3A_pct.1_name">pct.1_name</code></td>
<td>
<p>the name of data.frame column corresponding to percent expressed in group 1.
Default is Seurat default &quot;pct.1&quot;.</p>
</td></tr>
<tr><td><code id="Add_Pct_Diff_+3A_pct.2_name">pct.2_name</code></td>
<td>
<p>the name of data.frame column corresponding to percent expressed in group 2.
Default is Seurat default &quot;pct.2&quot;.</p>
</td></tr>
<tr><td><code id="Add_Pct_Diff_+3A_overwrite">overwrite</code></td>
<td>
<p>logical.  If the <code>marker_dataframe</code> already contains column named &quot;pct_diff&quot; whether to
overwrite or return error message.  Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns input <code>marker_dataframe</code> with additional &quot;pct_diff&quot; column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
marker_df &lt;- FindAllMarkers(object = obj_name)
marker_df &lt;- Add_Pct_Diff(marker_dataframe = marker_df)
# or piped with function
marker_df &lt;- FindAllMarkers(object = obj_name) %&gt;%
  Add_Pct_Diff()

## End(Not run)

</code></pre>

<hr>
<h2 id='Add_Sample_Meta'>Add Sample Level Meta Data</h2><span id='topic+Add_Sample_Meta'></span>

<h3>Description</h3>

<p>Add meta data from ample level data.frame/tibble to cell level seurat <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> slot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Add_Sample_Meta(
  seurat_object,
  meta_data,
  join_by_seurat,
  join_by_meta,
  na_ok = FALSE,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Add_Sample_Meta_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Add_Sample_Meta_+3A_meta_data">meta_data</code></td>
<td>
<p>data.frame/tibble containing meta data or path to file to read.  Must be formatted as
either data.frame or tibble.</p>
</td></tr>
<tr><td><code id="Add_Sample_Meta_+3A_join_by_seurat">join_by_seurat</code></td>
<td>
<p>name of the column in <code>seurat_object@meta.data</code> that contains matching
variables to <code>join_by_meta</code> in <code>meta_data.</code></p>
</td></tr>
<tr><td><code id="Add_Sample_Meta_+3A_join_by_meta">join_by_meta</code></td>
<td>
<p>name of the column in <code>meta_data</code> that contains matching
variables to <code>join_by_seurat</code> in <code>seurat_object@meta.data</code>.</p>
</td></tr>
<tr><td><code id="Add_Sample_Meta_+3A_na_ok">na_ok</code></td>
<td>
<p>logical, is it ok to add NA values to <code>seurat_object@meta.data</code>.  Default is FALSE.
Be very careful if setting TRUE because if there is error in join operation it may result in all
<code style="white-space: pre;">&#8288;@meta.data&#8288;</code> values being replaced with NA.</p>
</td></tr>
<tr><td><code id="Add_Sample_Meta_+3A_overwrite">overwrite</code></td>
<td>
<p>logical, if there are shared columns between <code>seurat_object@meta.data</code> and <code>meta_data</code>
should the current <code>seurat_object@meta.data</code> columns be overwritten.  Default is FALSE.  This parameter
excludes values provided to <code>join_by_seurat</code> and <code>join_by_meta</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat object with new <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# meta_data present in environment
sample_level_meta &lt;- data.frame(...)
obj &lt;- Add_Sample_Meta(seurat_object = obj, meta_data = sample_level_meta,
join_by_seurat = "orig.ident", join_by_meta = "sample_ID")

# from meta data file
obj &lt;- Add_Sample_Meta(seurat_object = obj, meta_data = "meta_data/sample_level_meta.csv",
join_by_seurat = "orig.ident", join_by_meta = "sample_ID")

## End(Not run)

</code></pre>

<hr>
<h2 id='Add_Top_Gene_Pct_Seurat'>Add Percent of High Abundance Genes</h2><span id='topic+Add_Top_Gene_Pct_Seurat'></span>

<h3>Description</h3>

<p>Add the percentage of counts occupied by the top XX most highly expressed genes in each cell.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Add_Top_Gene_Pct_Seurat(
  seurat_object,
  num_top_genes = 50,
  meta_col_name = NULL,
  assay = "RNA",
  overwrite = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Add_Top_Gene_Pct_Seurat_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Add_Top_Gene_Pct_Seurat_+3A_num_top_genes">num_top_genes</code></td>
<td>
<p>An integer vector specifying the size(s) of the top set of high-abundance genes.
Used to compute the percentage of library size occupied by the most highly expressed genes in each cell.</p>
</td></tr>
<tr><td><code id="Add_Top_Gene_Pct_Seurat_+3A_meta_col_name">meta_col_name</code></td>
<td>
<p>name to use for new meta data column.  Default is &quot;percent_topXX&quot;, where XX is
equal to the value provided to <code>num_top_genes</code>.</p>
</td></tr>
<tr><td><code id="Add_Top_Gene_Pct_Seurat_+3A_assay">assay</code></td>
<td>
<p>assay to use in calculation.  Default is &quot;RNA&quot;.  <em>Note</em> This should only be changed if
storing corrected and uncorrected assays in same object (e.g. outputs of both Cell Ranger and Cell Bender).</p>
</td></tr>
<tr><td><code id="Add_Top_Gene_Pct_Seurat_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical.  Whether to overwrite existing an meta.data column.  Default is FALSE meaning that
function will abort if column with name provided to <code>meta_col_name</code> is present in meta.data slot.</p>
</td></tr>
<tr><td><code id="Add_Top_Gene_Pct_Seurat_+3A_verbose">verbose</code></td>
<td>
<p>logical, whether to print messages with status updates, default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat Object
</p>


<h3>References</h3>

<p>This function uses scuttle package (license: GPL-3) to calculate the percent of expression
coming from top XX genes in each cell.  Parameter description for <code>num_top_genes</code> also from scuttle.
If using this function in analysis, in addition to citing scCustomize, please cite scuttle:
McCarthy DJ, Campbell KR, Lun ATL, Willis QF (2017). “Scater: pre-processing, quality control,
normalisation and visualisation of single-cell RNA-seq data in R.” Bioinformatics, 33, 1179-1186.
<a href="doi:10.1093/bioinformatics/btw777">doi:10.1093/bioinformatics/btw777</a>.
</p>


<h3>See Also</h3>

<p><a href="https://bioconductor.org/packages/release/bioc/html/scuttle.html">https://bioconductor.org/packages/release/bioc/html/scuttle.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(Seurat)
pbmc_small &lt;- Add_Top_Gene_Pct_Seurat(seurat_object = pbmc_small, num_top_genes = 50)

## End(Not run)

</code></pre>

<hr>
<h2 id='as.anndata'>Convert objects to anndata objects</h2><span id='topic+as.anndata'></span><span id='topic+as.anndata.Seurat'></span><span id='topic+as.anndata.liger'></span>

<h3>Description</h3>

<p>Convert objects (Seurat &amp; LIGER) to anndata objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.anndata(x, ...)

## S3 method for class 'Seurat'
as.anndata(
  x,
  file_path,
  file_name,
  assay = "RNA",
  main_layer = "data",
  other_layers = "counts",
  transer_dimreduc = TRUE,
  verbose = TRUE,
  ...
)

## S3 method for class 'liger'
as.anndata(
  x,
  file_path,
  file_name,
  transfer_norm.data = FALSE,
  reduction_label = NULL,
  add_barcode_names = FALSE,
  barcode_prefix = TRUE,
  barcode_cell_id_delimiter = "_",
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.anndata_+3A_x">x</code></td>
<td>
<p>Seurat or LIGER object</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_file_name">file_name</code></td>
<td>
<p>file name.</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_assay">assay</code></td>
<td>
<p>Assay containing data to use, (default is &quot;RNA&quot;).</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_main_layer">main_layer</code></td>
<td>
<p>the layer of data to become default layer in anndata object (default is &quot;data&quot;).</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_other_layers">other_layers</code></td>
<td>
<p>other data layers to transfer to anndata object (default is &quot;counts&quot;).</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_transer_dimreduc">transer_dimreduc</code></td>
<td>
<p>logical, whether to transfer dimensionality reduction coordinates from
Seurat to anndata object (default is TRUE).</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_verbose">verbose</code></td>
<td>
<p>logical, whether to print status messages during object conversion (default is TRUE).</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_transfer_norm.data">transfer_norm.data</code></td>
<td>
<p>logical, whether to transfer the norm.data in addition to
raw.data, default is FALSE.</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_reduction_label">reduction_label</code></td>
<td>
<p>What to label the visualization dimensionality reduction.
LIGER does not store name of technique and therefore needs to be set manually.</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_add_barcode_names">add_barcode_names</code></td>
<td>
<p>logical, whether to add dataset names to the cell barcodes when
merging object data, default is FALSE.</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_barcode_prefix">barcode_prefix</code></td>
<td>
<p>logical, if <code>add_barcode_names = TRUE</code> should the names be added as
prefix to current cell barcodes/names or a suffix (default is TRUE; prefix).</p>
</td></tr>
<tr><td><code id="as.anndata_+3A_barcode_cell_id_delimiter">barcode_cell_id_delimiter</code></td>
<td>
<p>The delimiter to use when adding dataset id to barcode
prefix/suffix.  Default is &quot;_&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an anndata object generated from <code>x</code>, saved at path provided.
</p>


<h3>References</h3>

<p>Seurat version modified and enhanced version of <code>sceasy::seurat2anndata</code> (sceasy package: <a href="https://github.com/cellgeni/sceasy">https://github.com/cellgeni/sceasy</a>; License: GPL-3.  Function has additional checks and supports Seurat V3 and V5 object structure.
</p>
<p>LIGER version inspired by <code>sceasy::seurat2anndata</code> modified and updated to apply to LIGER objects (sceasy package: <a href="https://github.com/cellgeni/sceasy">https://github.com/cellgeni/sceasy</a>; License: GPL-3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
as.anndata(x = seurat_object, file_path = "/folder_name", file_name = "anndata_converted.h5ad")

## End(Not run)

## Not run: 
as.anndata(x = liger_object, file_path = "/folder_name", file_name = "anndata_converted.h5ad")

## End(Not run)

</code></pre>

<hr>
<h2 id='as.LIGER'>Convert objects to LIGER objects</h2><span id='topic+as.LIGER'></span><span id='topic+as.LIGER.Seurat'></span><span id='topic+as.LIGER.list'></span>

<h3>Description</h3>

<p>Convert objects (Seurat &amp; lists of Seurat Objects) to anndata objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.LIGER(x, ...)

## S3 method for class 'Seurat'
as.LIGER(
  x,
  group.by = "orig.ident",
  layers_name = NULL,
  assay = "RNA",
  remove_missing = FALSE,
  renormalize = TRUE,
  use_seurat_var_genes = FALSE,
  use_seurat_dimreduc = FALSE,
  reduction = NULL,
  keep_meta = TRUE,
  verbose = TRUE,
  ...
)

## S3 method for class 'list'
as.LIGER(
  x,
  group.by = "orig.ident",
  dataset_names = NULL,
  assay = "RNA",
  remove_missing = FALSE,
  renormalize = TRUE,
  use_seurat_var_genes = FALSE,
  var_genes_method = "intersect",
  keep_meta = TRUE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.LIGER_+3A_x">x</code></td>
<td>
<p>An object to convert to class <code>liger</code></p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_group.by">group.by</code></td>
<td>
<p>Variable in meta data which contains variable to split data by, (default is &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_layers_name">layers_name</code></td>
<td>
<p>name of meta.data column used to split layers if setting <code>group.by = "layers"</code>.</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_assay">assay</code></td>
<td>
<p>Assay containing raw data to use, (default is &quot;RNA&quot;).</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_remove_missing">remove_missing</code></td>
<td>
<p>logical, whether to remove missing genes with no counts when converting to
LIGER object (default is FALSE).</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_renormalize">renormalize</code></td>
<td>
<p>logical, whether to perform normalization after LIGER object creation (default is TRUE).</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_use_seurat_var_genes">use_seurat_var_genes</code></td>
<td>
<p>logical, whether to transfer variable features from Seurat object to
new LIGER object (default is FALSE).</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_use_seurat_dimreduc">use_seurat_dimreduc</code></td>
<td>
<p>logical, whether to transfer dimensionality reduction coordinates from
Seurat to new LIGER object (default is FALSE).</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_reduction">reduction</code></td>
<td>
<p>Name of Seurat reduction to transfer if <code>use_seurat_dimreduc = TRUE</code>.</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_keep_meta">keep_meta</code></td>
<td>
<p>logical, whether to transfer columns in Seurat meta.data slot to LIGER cell.data
slot (default is TRUE).</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_verbose">verbose</code></td>
<td>
<p>logical, whether to print status messages during object conversion (default is TRUE).</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_dataset_names">dataset_names</code></td>
<td>
<p>optional, vector of names to use for naming datasets.</p>
</td></tr>
<tr><td><code id="as.LIGER_+3A_var_genes_method">var_genes_method</code></td>
<td>
<p>how variable genes should be selected from Seurat objects if <code>use_seurat_var_genes = TRUE</code>.  Can be either &quot;intersect&quot; or &quot;union&quot;, (default is &quot;intersect&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a liger object generated from <code>x</code>
</p>


<h3>References</h3>

<p>modified and enhanced version of <code>rliger::seuratToLiger</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
liger_object &lt;- as.LIGER(x = seurat_object)

## End(Not run)

## Not run: 
liger_object &lt;- as.LIGER(x = seurat_object_list)

## End(Not run)

</code></pre>

<hr>
<h2 id='as.Seurat.liger'>Convert objects to <code>Seurat</code> objects</h2><span id='topic+as.Seurat.liger'></span>

<h3>Description</h3>

<p>Merges raw.data and scale.data of object, and creates Seurat object with these values along with slots
containing dimensionality reduction coordinates, iNMF factorization, and cluster assignments.
Supports Seurat V3/4 and V4.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'liger'
as.Seurat(
  x,
  nms = names(x@H),
  renormalize = TRUE,
  use.liger.genes = TRUE,
  by.dataset = FALSE,
  keep_meta = TRUE,
  reduction_label = "UMAP",
  seurat_assay = "RNA",
  assay_type = NULL,
  add_barcode_names = FALSE,
  barcode_prefix = TRUE,
  barcode_cell_id_delimiter = "_",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.Seurat.liger_+3A_x">x</code></td>
<td>
<p><code>liger</code> object.</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_nms">nms</code></td>
<td>
<p>By default, labels cell names with dataset of origin (this is to account for cells in
different datasets which may have same name). Other names can be passed here as vector, must have
same length as the number of datasets. (default names(H)).</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_renormalize">renormalize</code></td>
<td>
<p>Whether to log-normalize raw data using Seurat defaults (default TRUE).</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_use.liger.genes">use.liger.genes</code></td>
<td>
<p>Whether to carry over variable genes (default TRUE).</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_by.dataset">by.dataset</code></td>
<td>
<p>Include dataset of origin in cluster identity in Seurat object (default FALSE).</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_keep_meta">keep_meta</code></td>
<td>
<p>logical. Whether to transfer additional metadata (nGene/nUMI/dataset already transferred)
to new Seurat Object.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_reduction_label">reduction_label</code></td>
<td>
<p>Name of dimensionality reduction technique used.  Enables accurate transfer
or name to Seurat object instead of defaulting to &quot;tSNE&quot;.</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_seurat_assay">seurat_assay</code></td>
<td>
<p>Name to set for assay in Seurat Object.  Default is &quot;RNA&quot;.</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_assay_type">assay_type</code></td>
<td>
<p>what type of Seurat assay to create in new object (Assay vs Assay5).
Default is NULL which will default to the current user settings.
See <code><a href="#topic+Convert_Assay">Convert_Assay</a></code> parameter <code>convert_to</code> for acceptable values.</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_add_barcode_names">add_barcode_names</code></td>
<td>
<p>logical, whether to add dataset names to the cell barcodes when
creating Seurat object, default is FALSE.</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_barcode_prefix">barcode_prefix</code></td>
<td>
<p>logical, if <code>add_barcode_names = TRUE</code> should the names be added as
prefix to current cell barcodes/names or a suffix (default is TRUE; prefix).</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_barcode_cell_id_delimiter">barcode_cell_id_delimiter</code></td>
<td>
<p>The delimiter to use when adding dataset id to barcode
prefix/suffix.  Default is &quot;_&quot;.</p>
</td></tr>
<tr><td><code id="as.Seurat.liger_+3A_...">...</code></td>
<td>
<p>unused.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Stores original dataset identity by default in new object metadata if dataset names are passed
in nms. iNMF factorization is stored in dim.reduction object with key &quot;iNMF&quot;.
</p>


<h3>Value</h3>

<p>Seurat object with raw.data, scale.data, reduction_label, iNMF, and ident slots set.
</p>
<p>Seurat object.
</p>


<h3>References</h3>

<p>Original function is part of LIGER package <a href="https://github.com/welch-lab/liger">https://github.com/welch-lab/liger</a> (Licence: GPL-3).
Function was modified for use in scCustomize with additional parameters/functionality.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
seurat_object &lt;- as.Seurat(x = liger_object)

## End(Not run)

</code></pre>

<hr>
<h2 id='Barcode_Plot'>Create Barcode Rank Plot</h2><span id='topic+Barcode_Plot'></span>

<h3>Description</h3>

<p>Plot UMI vs. Barcode Rank with inflection and knee.  Requires input from DropletUtils package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Barcode_Plot(
  br_out,
  pt.size = 6,
  plot_title = "Barcode Ranks",
  raster_dpi = c(1024, 1024),
  plateau = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Barcode_Plot_+3A_br_out">br_out</code></td>
<td>
<p>DFrame output from <code><a href="DropletUtils.html#topic+barcodeRanks">barcodeRanks</a></code>.</p>
</td></tr>
<tr><td><code id="Barcode_Plot_+3A_pt.size">pt.size</code></td>
<td>
<p>point size for plotting, default is 6.</p>
</td></tr>
<tr><td><code id="Barcode_Plot_+3A_plot_title">plot_title</code></td>
<td>
<p>Title for plot, default is &quot;Barcode Ranks&quot;.</p>
</td></tr>
<tr><td><code id="Barcode_Plot_+3A_raster_dpi">raster_dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(1024, 1024).</p>
</td></tr>
<tr><td><code id="Barcode_Plot_+3A_plateau">plateau</code></td>
<td>
<p>numerical value at which to add vertical line designating estimated
empty droplet plateau (default is NULL).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mat &lt;- Read10X_h5(filename = "raw_feature_bc_matrix.h5")

br_results &lt;- DropletUtils::barcodeRanks(mat)

Barcode_Plot(br_out = br_results)

## End(Not run)

</code></pre>

<hr>
<h2 id='Blank_Theme'>Blank Theme</h2><span id='topic+Blank_Theme'></span>

<h3>Description</h3>

<p>Shortcut for thematic modification to remove all axis labels and grid lines
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Blank_Theme(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Blank_Theme_+3A_...">...</code></td>
<td>
<p>extra arguments passed to <code>ggplot2::theme()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list-like object of class <em>theme</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a plot and customize theme
library(ggplot2)
df &lt;- data.frame(x = rnorm(n = 100, mean = 20, sd = 2), y = rbinom(n = 100, size = 100, prob = 0.2))
p &lt;- ggplot(data = df, mapping = aes(x = x, y = y)) + geom_point(mapping = aes(color = 'red'))
p + Blank_Theme()

</code></pre>

<hr>
<h2 id='Case_Check'>Check for alternate case features
Checks Seurat object for the presence of features with the same spelling but alternate case.</h2><span id='topic+Case_Check'></span>

<h3>Description</h3>

<p>Check for alternate case features
Checks Seurat object for the presence of features with the same spelling but alternate case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Case_Check(
  seurat_object,
  gene_list,
  case_check_msg = TRUE,
  return_features = TRUE,
  assay = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Case_Check_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Case_Check_+3A_gene_list">gene_list</code></td>
<td>
<p>vector of genes to check.</p>
</td></tr>
<tr><td><code id="Case_Check_+3A_case_check_msg">case_check_msg</code></td>
<td>
<p>logical. Whether to print message to console if alternate case features are
found in addition to inclusion in returned list.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Case_Check_+3A_return_features">return_features</code></td>
<td>
<p>logical. Whether to return vector of alternate case features.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Case_Check_+3A_assay">assay</code></td>
<td>
<p>Name of assay to pull feature names from. If NULL will use the result of <code>DefaultAssay(seurat_object)</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If features found returns vector of found alternate case features and prints message depending on
parameters specified.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
alt_features &lt;- Case_Check(seurat_object = obj_name, gene_list = DEG_list)

## End(Not run)

</code></pre>

<hr>
<h2 id='Cell_Highlight_Plot'>Meta Highlight Plot</h2><span id='topic+Cell_Highlight_Plot'></span>

<h3>Description</h3>

<p>Create Plot with meta data variable of interest highlighted
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cell_Highlight_Plot(
  seurat_object,
  cells_highlight,
  highlight_color = NULL,
  background_color = "lightgray",
  pt.size = NULL,
  aspect_ratio = NULL,
  figure_plot = FALSE,
  raster = NULL,
  raster.dpi = c(512, 512),
  label = FALSE,
  split.by = NULL,
  split_seurat = FALSE,
  ggplot_default_colors = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Cell_Highlight_Plot_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_cells_highlight">cells_highlight</code></td>
<td>
<p>Cell names to highlight in named list.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>Color to highlight cells.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_background_color">background_color</code></td>
<td>
<p>non-highlighted cell colors (default is &quot;lightgray&quot;)..</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_pt.size">pt.size</code></td>
<td>
<p>point size for both highlighted cluster and background.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_figure_plot">figure_plot</code></td>
<td>
<p>logical.  Whether to remove the axes and plot with legend on left of plot denoting
axes labels.  (Default is FALSE).  Requires <code>split_seurat = TRUE</code>.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_label">label</code></td>
<td>
<p>Whether to label the highlighted meta data variable(s).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_split.by">split.by</code></td>
<td>
<p>Variable in <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> to split the plot by.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_split_seurat">split_seurat</code></td>
<td>
<p>logical.  Whether or not to display split plots like Seurat (shared y axis) or as
individual plots in layout.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>highlight_color = NULL</code>, Whether or not to return plot
using default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="Cell_Highlight_Plot_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to<code><a href="Seurat.html#topic+DimPlot">DimPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)

# Creating example non-overlapping vectors of cells
MS4A1 &lt;- WhichCells(object = pbmc_small, expression = MS4A1 &gt; 4)
GZMB &lt;- WhichCells(object = pbmc_small, expression = GZMB &gt; 4)

# Format as named list
cells &lt;- list("MS4A1" = MS4A1,
              "GZMB" = GZMB)

Cell_Highlight_Plot(seurat_object = pbmc_small, cells_highlight = cells)

</code></pre>

<hr>
<h2 id='CellBender_Diff_Plot'>Plot Number of Cells/Nuclei per Sample</h2><span id='topic+CellBender_Diff_Plot'></span>

<h3>Description</h3>

<p>Plot of total cell or nuclei number per sample grouped by another meta data variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CellBender_Diff_Plot(
  feature_diff_df,
  pct_diff_threshold = 25,
  num_features = NULL,
  label = TRUE,
  num_labels = 20,
  min_count_label = 1,
  repel = TRUE,
  custom_labels = NULL,
  plot_line = TRUE,
  plot_title = "Raw Counts vs. Cell Bender Counts",
  x_axis_label = "Raw Data Counts",
  y_axis_label = "Cell Bender Counts",
  xnudge = 0,
  ynudge = 0,
  max.overlaps = 100,
  label_color = "dodgerblue",
  fontface = "bold",
  label_size = 3.88,
  bg.color = "white",
  bg.r = 0.15,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CellBender_Diff_Plot_+3A_feature_diff_df">feature_diff_df</code></td>
<td>
<p>name of data.frame created using <code><a href="#topic+CellBender_Feature_Diff">CellBender_Feature_Diff</a></code>.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_pct_diff_threshold">pct_diff_threshold</code></td>
<td>
<p>threshold to use for feature plotting.  Resulting plot will only contain
features which exhibit percent change &gt;= value.  Default is 25.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_num_features">num_features</code></td>
<td>
<p>Number of features to plot.  Will ignore <code>pct_diff_threshold</code> and return
plot with specified number of features.  Default is NULL.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_label">label</code></td>
<td>
<p>logical, whether or not to label the features that have largest percent difference
between raw and CellBender counts (Default is TRUE).</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_num_labels">num_labels</code></td>
<td>
<p>Number of features to label if <code>label = TRUE</code>, (default is 20).</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_min_count_label">min_count_label</code></td>
<td>
<p>Minimum number of raw counts per feature necessary to be included in
plot labels (default is 1)</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_repel">repel</code></td>
<td>
<p>logical, whether to use geom_text_repel to create a nicely-repelled labels; this is
slow when a lot of points are being plotted. If using repel, set xnudge and ynudge to 0, (Default is TRUE).</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_custom_labels">custom_labels</code></td>
<td>
<p>A custom set of features to label instead of the features most different between
raw and CellBender counts.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_plot_line">plot_line</code></td>
<td>
<p>logical, whether to plot diagonal line with slope = 1 (Default is TRUE).</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot title.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_xnudge">xnudge</code></td>
<td>
<p>Amount to nudge X and Y coordinates of labels by.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_ynudge">ynudge</code></td>
<td>
<p>Amount to nudge X and Y coordinates of labels by.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_max.overlaps">max.overlaps</code></td>
<td>
<p>passed to <code><a href="ggrepel.html#topic+geom_text_repel">geom_text_repel</a></code>, exclude text labels that
overlap too many things. Defaults to 100.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_label_color">label_color</code></td>
<td>
<p>Color to use for text labels.</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_fontface">fontface</code></td>
<td>
<p>font face to use for text labels (“plain”, “bold”, “italic”, “bold.italic”) (Default is &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_label_size">label_size</code></td>
<td>
<p>text size for feature labels (passed to <code><a href="ggrepel.html#topic+geom_text_repel">geom_text_repel</a></code>).</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_bg.color">bg.color</code></td>
<td>
<p>color to use for shadow/outline of text labels (passed to <code><a href="ggrepel.html#topic+geom_text_repel">geom_text_repel</a></code>) (Default is white).</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_bg.r">bg.r</code></td>
<td>
<p>radius to use for shadow/outline of text labels (passed to <code><a href="ggrepel.html#topic+geom_text_repel">geom_text_repel</a></code>) (Default is 0.15).</p>
</td></tr>
<tr><td><code id="CellBender_Diff_Plot_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="ggrepel.html#topic+geom_text_repel">geom_text_repel</a></code> through
<code><a href="Seurat.html#topic+LabelPoints">LabelPoints</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# get cell bender differences data.frame
cb_stats &lt;- CellBender_Feature_Diff(seurat_object - obj, raw_assay = "RAW",
cell_bender_assay = "RNA")

# plot
CellBender_Diff_Plot(feature_diff_df = cb_stats, pct_diff_threshold = 25)

## End(Not run)

</code></pre>

<hr>
<h2 id='CellBender_Feature_Diff'>CellBender Feature Differences</h2><span id='topic+CellBender_Feature_Diff'></span>

<h3>Description</h3>

<p>Get quick values for raw counts, CellBender counts, count differences, and percent count differences
per feature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CellBender_Feature_Diff(
  seurat_object = NULL,
  raw_assay = NULL,
  cell_bender_assay = NULL,
  raw_mat = NULL,
  cell_bender_mat = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CellBender_Feature_Diff_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="CellBender_Feature_Diff_+3A_raw_assay">raw_assay</code></td>
<td>
<p>Name of the assay containing the raw count data.</p>
</td></tr>
<tr><td><code id="CellBender_Feature_Diff_+3A_cell_bender_assay">cell_bender_assay</code></td>
<td>
<p>Name of the assay containing the CellBender count data.</p>
</td></tr>
<tr><td><code id="CellBender_Feature_Diff_+3A_raw_mat">raw_mat</code></td>
<td>
<p>Name of raw count matrix in environment if not using Seurat object.</p>
</td></tr>
<tr><td><code id="CellBender_Feature_Diff_+3A_cell_bender_mat">cell_bender_mat</code></td>
<td>
<p>Name of CellBender count matrix in environment if not using Seurat object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing summed raw counts, CellBender counts, count difference, and
percent difference in counts.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cb_stats &lt;- CellBender_Feature_Diff(seurat_object - obj, raw_assay = "RAW",
cell_bender_assay = "RNA")

## End(Not run)

</code></pre>

<hr>
<h2 id='Change_Delim_All'>Change all delimiters in cell name</h2><span id='topic+Change_Delim_All'></span>

<h3>Description</h3>

<p>Change all instances of delimiter in cell names from list of data.frames/matrices or single data.frame/matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Change_Delim_All(data, current_delim, new_delim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Change_Delim_All_+3A_data">data</code></td>
<td>
<p>Either matrix/data.frame or list of matrices/data.frames with the cell barcodes in the column names.</p>
</td></tr>
<tr><td><code id="Change_Delim_All_+3A_current_delim">current_delim</code></td>
<td>
<p>a single value of current delimiter.</p>
</td></tr>
<tr><td><code id="Change_Delim_All_+3A_new_delim">new_delim</code></td>
<td>
<p>a single value of new delimiter desired.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix or data.frame with new column names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dge_matrix &lt;- Change_Delim_All(data = dge_matrix, current_delim = ".", new_delim = "-")

## End(Not run)

</code></pre>

<hr>
<h2 id='Change_Delim_Prefix'>Change barcode prefix delimiter</h2><span id='topic+Change_Delim_Prefix'></span>

<h3>Description</h3>

<p>Change barcode prefix delimiter from list of data.frames/matrices or single data.frame/matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Change_Delim_Prefix(data, current_delim, new_delim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Change_Delim_Prefix_+3A_data">data</code></td>
<td>
<p>Either matrix/data.frame or list of matrices/data.frames with the cell barcodes in the column names.</p>
</td></tr>
<tr><td><code id="Change_Delim_Prefix_+3A_current_delim">current_delim</code></td>
<td>
<p>a single value of current delimiter.</p>
</td></tr>
<tr><td><code id="Change_Delim_Prefix_+3A_new_delim">new_delim</code></td>
<td>
<p>a single value of new delimiter desired.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix or data.frame with new column names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dge_matrix &lt;- Change_Delim_Prefix(data = dge_matrix, current_delim = ".", new_delim = "-")

## End(Not run)

</code></pre>

<hr>
<h2 id='Change_Delim_Suffix'>Change barcode suffix delimiter</h2><span id='topic+Change_Delim_Suffix'></span>

<h3>Description</h3>

<p>Change barcode suffix delimiter from list of data.frames/matrices or single data.frame/matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Change_Delim_Suffix(data, current_delim, new_delim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Change_Delim_Suffix_+3A_data">data</code></td>
<td>
<p>Either matrix/data.frame or list of matrices/data.frames with the cell barcodes in the column names.</p>
</td></tr>
<tr><td><code id="Change_Delim_Suffix_+3A_current_delim">current_delim</code></td>
<td>
<p>a single value of current delimiter.</p>
</td></tr>
<tr><td><code id="Change_Delim_Suffix_+3A_new_delim">new_delim</code></td>
<td>
<p>a single value of new delimiter desired.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix or data.frame with new column names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dge_matrix &lt;- Change_Delim_Suffix(data = dge_matrix, current_delim = ".", new_delim = "-")

## End(Not run)

</code></pre>

<hr>
<h2 id='CheckMatrix_scCustom'>Check Matrix Validity</h2><span id='topic+CheckMatrix_scCustom'></span>

<h3>Description</h3>

<p>Native implementation of SeuratObjects CheckMatrix but with modified warning messages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckMatrix_scCustom(
  object,
  checks = c("infinite", "logical", "integer", "na")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckMatrix_scCustom_+3A_object">object</code></td>
<td>
<p>A matrix</p>
</td></tr>
<tr><td><code id="CheckMatrix_scCustom_+3A_checks">checks</code></td>
<td>
<p>Type of checks to perform, choose one or more from:
</p>

<ul>
<li> <p>&ldquo;<code>infinite</code>&rdquo;: Emit a warning if any value is infinite
</p>
</li>
<li> <p>&ldquo;<code>logical</code>&rdquo;: Emit a warning if any value is a logical
</p>
</li>
<li> <p>&ldquo;<code>integer</code>&rdquo;: Emit a warning if any value is <em>not</em>
an integer
</p>
</li>
<li> <p>&ldquo;<code>na</code>&rdquo;: Emit a warning if any value is an <code>NA</code>
or <code>NaN</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>Emits warnings for each test and invisibly returns <code>NULL</code>
</p>


<h3>References</h3>

<p>Re-implementing <code>CheckMatrix</code> only for sparse matrices with modified warning messages.  Original function from SeuratObject <a href="https://github.com/satijalab/seurat-object/blob/9c0eda946e162d8595696e5280a6ecda6284db39/R/utils.R#L625-L650">https://github.com/satijalab/seurat-object/blob/9c0eda946e162d8595696e5280a6ecda6284db39/R/utils.R#L625-L650</a> (License: MIT).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mat &lt;- Read10X(...)
CheckMatrix_scCustom(object = mat)

## End(Not run)

</code></pre>

<hr>
<h2 id='Cluster_Highlight_Plot'>Cluster Highlight Plot</h2><span id='topic+Cluster_Highlight_Plot'></span>

<h3>Description</h3>

<p>Create Plot with cluster of interest highlighted
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cluster_Highlight_Plot(
  seurat_object,
  cluster_name,
  highlight_color = NULL,
  background_color = "lightgray",
  pt.size = NULL,
  aspect_ratio = NULL,
  figure_plot = FALSE,
  raster = NULL,
  raster.dpi = c(512, 512),
  label = FALSE,
  split.by = NULL,
  split_seurat = FALSE,
  ggplot_default_colors = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Cluster_Highlight_Plot_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_cluster_name">cluster_name</code></td>
<td>
<p>Name(s) (or number(s)) identity of cluster to be highlighted.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>Color(s) to highlight cells.  The default is NULL and plot will use
<code>scCustomize_Palette()</code>.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_background_color">background_color</code></td>
<td>
<p>non-highlighted cell colors.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_pt.size">pt.size</code></td>
<td>
<p>point size for both highlighted cluster and background.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_figure_plot">figure_plot</code></td>
<td>
<p>logical.  Whether to remove the axes and plot with legend on left of plot denoting
axes labels.  (Default is FALSE).  Requires <code>split_seurat = TRUE</code>.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_label">label</code></td>
<td>
<p>Whether to label the highlighted cluster(s).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_split.by">split.by</code></td>
<td>
<p>Feature to split plots by (i.e. &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_split_seurat">split_seurat</code></td>
<td>
<p>logical.  Whether or not to display split plots like Seurat (shared y axis) or as
individual plots in layout.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="Cluster_Highlight_Plot_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+DimPlot">DimPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Cluster_Highlight_Plot(seurat_object = pbmc_small, cluster_name = "1", highlight_color = "gold",
background_color = "lightgray",  pt.size = 2)

</code></pre>

<hr>
<h2 id='Cluster_Stats_All_Samples'>Calculate Cluster Stats</h2><span id='topic+Cluster_Stats_All_Samples'></span>

<h3>Description</h3>

<p>Calculates both overall and per sample cell number and percentages per cluster based on orig.ident
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cluster_Stats_All_Samples(seurat_object, group_by_var = "orig.ident")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Cluster_Stats_All_Samples_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Cluster_Stats_All_Samples_+3A_group_by_var">group_by_var</code></td>
<td>
<p>meta data column to classify samples (default = &quot;orig.ident&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
stats &lt;- Cluster_Stats_All_Samples(seurat_object = object, group_by_var = "orig.ident")

## End(Not run)

</code></pre>

<hr>
<h2 id='Clustered_DotPlot'>Clustered DotPlot</h2><span id='topic+Clustered_DotPlot'></span>

<h3>Description</h3>

<p>Clustered DotPlots using ComplexHeatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Clustered_DotPlot(
  seurat_object,
  features,
  split.by = NULL,
  colors_use_exp = viridis_plasma_dark_high,
  exp_color_min = -2,
  exp_color_middle = NULL,
  exp_color_max = 2,
  exp_value_type = "scaled",
  print_exp_quantiles = FALSE,
  colors_use_idents = NULL,
  x_lab_rotate = TRUE,
  plot_padding = NULL,
  flip = FALSE,
  k = 1,
  feature_km_repeats = 1000,
  ident_km_repeats = 1000,
  row_label_size = 8,
  row_label_fontface = "plain",
  grid_color = NULL,
  cluster_feature = TRUE,
  cluster_ident = TRUE,
  column_label_size = 8,
  legend_label_size = 10,
  legend_title_size = 10,
  raster = FALSE,
  plot_km_elbow = TRUE,
  elbow_kmax = NULL,
  assay = NULL,
  group.by = NULL,
  idents = NULL,
  show_parent_dend_line = TRUE,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Clustered_DotPlot_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_features">features</code></td>
<td>
<p>Features to plot.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_split.by">split.by</code></td>
<td>
<p>Variable in <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> to split the identities plotted by.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_colors_use_exp">colors_use_exp</code></td>
<td>
<p>Color palette to use for plotting expression scale.  Default is <code>viridis::plasma(n = 20, direction = -1)</code>.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_exp_color_min">exp_color_min</code></td>
<td>
<p>Minimum scaled average expression threshold (everything smaller will be set to this).
Default is -2.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_exp_color_middle">exp_color_middle</code></td>
<td>
<p>What scaled expression value to use for the middle of the provided <code>colors_use_exp</code>.
By default will be set to value in middle of <code>exp_color_min</code> and <code>exp_color_max</code>.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_exp_color_max">exp_color_max</code></td>
<td>
<p>Minimum scaled average expression threshold (everything smaller will be set to this).
Default is 2.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_exp_value_type">exp_value_type</code></td>
<td>
<p>Whether to plot average normalized expression or
scaled average normalized expression.  Only valid when <code>split.by</code> is provided.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_print_exp_quantiles">print_exp_quantiles</code></td>
<td>
<p>Whether to print the quantiles of expression data in addition to plots.
Default is FALSE.  NOTE: These values will be altered by choices of <code>exp_color_min</code> and <code>exp_color_min</code>
if there are values below or above those cutoffs, respectively.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_colors_use_idents">colors_use_idents</code></td>
<td>
<p>specify color palette to used for identity labels.  By default if
number of levels plotted is less than or equal to 36 it will use &quot;polychrome&quot; and if greater than 36
will use &quot;varibow&quot; with shuffle = TRUE both from <code>DiscretePalette_scCustomize</code>.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>How to rotate column labels.  By default set to <code>TRUE</code> which rotates labels 45 degrees.
If set <code>FALSE</code> rotation is set to 0 degrees.  Users can also supply custom angle for text rotation.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_plot_padding">plot_padding</code></td>
<td>
<p>if plot needs extra white space padding so no plot or labels are cutoff.
The parameter accepts TRUE or numeric vector of length 4.  If TRUE padding will be set to
c(2, 10, 0 0) (bottom, left, top, right).  Can also be customized further with numeric
vector of length 4 specifying the amount of padding in millimeters.
Default is NULL, no padding.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_flip">flip</code></td>
<td>
<p>logical, whether to flip the axes of final plot.  Default is FALSE; rows = features and
columns = idents.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_k">k</code></td>
<td>
<p>Value to use for k-means clustering on features  Sets (km) parameter in <code>ComplexHeatmap::Heatmap()</code>.
From <code>ComplexHeatmap::Heatmap()</code>: Apply k-means clustering on rows. If the value is larger than 1, the
heatmap will be split by rows according to the k-means clustering. For each row slice, hierarchical
clustering is still applied with parameters above.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_feature_km_repeats">feature_km_repeats</code></td>
<td>
<p>Number of k-means runs to get a consensus k-means clustering for features.
Note if <code>feature_km_repeats</code> is set to value greater than one, the final number of groups might be
smaller than row_km, but this might mean the original row_km is not a good choice.  Default is 1000.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_ident_km_repeats">ident_km_repeats</code></td>
<td>
<p>Number of k-means runs to get a consensus k-means clustering. Similar to
<code>feature_km_repeats</code>.  Default is 1000.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_row_label_size">row_label_size</code></td>
<td>
<p>Size of the feature labels.  Provided to <code>row_names_gp</code> in Heatmap call.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_row_label_fontface">row_label_fontface</code></td>
<td>
<p>Fontface to use for row labels.  Provided to <code>row_names_gp</code> in Heatmap call.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_grid_color">grid_color</code></td>
<td>
<p>color to use for heatmap grid.  Default is NULL which &quot;removes&quot; grid by using NA color.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_cluster_feature">cluster_feature</code></td>
<td>
<p>logical, whether to cluster and reorder feature axis.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_cluster_ident">cluster_ident</code></td>
<td>
<p>logical, whether to cluster and reorder identity axis.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_column_label_size">column_label_size</code></td>
<td>
<p>Size of the feature labels.  Provided to <code>column_names_gp</code> in Heatmap call.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_legend_label_size">legend_label_size</code></td>
<td>
<p>Size of the legend text labels.  Provided to <code>labels_gp</code> in Heatmap legend call.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>Sise of the legend title text labels.  Provided to <code>title_gp</code> in Heatmap legend call.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_raster">raster</code></td>
<td>
<p>Logical, whether to render in raster format (faster plotting, smaller files).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_plot_km_elbow">plot_km_elbow</code></td>
<td>
<p>Logical, whether or not to return the Sum Squared Error Elbow Plot for k-means clustering.
Estimating elbow of this plot is one way to determine &quot;optimal&quot; value for <code>k</code>.
Based on: <a href="https://stackoverflow.com/a/15376462/15568251">https://stackoverflow.com/a/15376462/15568251</a>.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_elbow_kmax">elbow_kmax</code></td>
<td>
<p>The maximum value of k to use for <code>plot_km_elbow</code>.  Suggest setting larger value so the
true shape of plot can be observed.  Value must be 1 less than number of features provided.  If NULL parameter
will be set dependent on length of feature list up to <code>elbow_kmax = 20</code>.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_assay">assay</code></td>
<td>
<p>Name of assay to use, defaults to the active assay.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_group.by">group.by</code></td>
<td>
<p>Group (color) cells in different ways (for example, orig.ident).</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_idents">idents</code></td>
<td>
<p>Which classes to include in the plot (default is all).</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_show_parent_dend_line">show_parent_dend_line</code></td>
<td>
<p>Logical, Sets parameter of same name in <code>ComplexHeatmap::Heatmap()</code>.
From <code>ComplexHeatmap::Heatmap()</code>: When heatmap is split, whether to add a dashed line to mark parent
dendrogram and children dendrograms.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="Clustered_DotPlot_+3A_seed">seed</code></td>
<td>
<p>Sets seed for reproducible plotting (ComplexHeatmap plot).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ComplexHeatmap or if plot_km_elbow = TRUE a list containing ggplot2 object and ComplexHeatmap.
</p>


<h3>Author(s)</h3>

<p>Ming Tang (Original Code), Sam Marsh (Wrap single function, added/modified functionality)
</p>


<h3>References</h3>

<p><a href="https://divingintogeneticsandgenomics.rbind.io/post/clustered-dotplot-for-single-cell-rnaseq/">https://divingintogeneticsandgenomics.rbind.io/post/clustered-dotplot-for-single-cell-rnaseq/</a>
</p>


<h3>See Also</h3>

<p><a href="https://twitter.com/tangming2005">https://twitter.com/tangming2005</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Seurat)
Clustered_DotPlot(seurat_object = pbmc_small, features = c("CD3E", "CD8", "GZMB", "MS4A1"))


</code></pre>

<hr>
<h2 id='ColorBlind_Pal'>Color Universal Design Short Palette</h2><span id='topic+ColorBlind_Pal'></span>

<h3>Description</h3>

<p>Shortcut ta a modified 8 color palette based on Color Universal Design (CUD) colorblindness friendly palette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ColorBlind_Pal()
</code></pre>


<h3>Value</h3>

<p>modified/reordered color palette (8 colors) based on ditto-seq
</p>


<h3>References</h3>

<p>palette is slightly modified version of the Color Universal Design (CUD) colorblindness
friendly palette <a href="https://jfly.uni-koeln.de/color/">https://jfly.uni-koeln.de/color/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cols &lt;- ColorBlind_Pal()
PalettePlot(pal = cols)

</code></pre>

<hr>
<h2 id='Convert_Assay'>Convert between Seurat Assay types</h2><span id='topic+Convert_Assay'></span>

<h3>Description</h3>

<p>Will convert assays within a Seurat object between &quot;Assay&quot; and &quot;Assay5&quot; types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Convert_Assay(seurat_object, assay = NULL, convert_to)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Convert_Assay_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Convert_Assay_+3A_assay">assay</code></td>
<td>
<p>name(s) of assays to convert.  Default is NULL and will check with users
which assays they want to convert.</p>
</td></tr>
<tr><td><code id="Convert_Assay_+3A_convert_to">convert_to</code></td>
<td>
<p>value of what assay type to convert current assays to.
#' </p>

<ul>
<li><p> Accepted values for V3/4 are: &quot;Assay&quot;, &quot;assay&quot;, &quot;V3&quot;, or &quot;v3&quot;.
</p>
</li>
<li><p> Accepted values for V5 are: &quot;Assay5&quot;, &quot;assay5&quot;, &quot;V5&quot;, or &quot;v5&quot;.
</p>
</li></ul>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Convert to V3/4 assay
obj &lt;- Convert_Assay(seurat_object = obj, convert_to = "V3")

# Convert to 5 assay
obj &lt;- Convert_Assay(seurat_object = obj, convert_to = "V5")

## End(Not run)

</code></pre>

<hr>
<h2 id='Copy_From_GCP'>Copy folder from GCP bucket from R Console</h2><span id='topic+Copy_From_GCP'></span>

<h3>Description</h3>

<p>Run command from R console without moving to terminal to copy folder from GCP bucket to local storage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Copy_From_GCP(folder_file_path, gcp_bucket_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Copy_From_GCP_+3A_folder_file_path">folder_file_path</code></td>
<td>
<p>folder to be copied to GCP bucket.</p>
</td></tr>
<tr><td><code id="Copy_From_GCP_+3A_gcp_bucket_path">gcp_bucket_path</code></td>
<td>
<p>GCP bucket path to copy to files.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.  Performs system copy from GCP bucket.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Copy_From_GCP(folder_file_path = "plots/", gcp_bucket_path = "gs://bucket_name_and_folder_path")

## End(Not run)

</code></pre>

<hr>
<h2 id='Copy_To_GCP'>Copy folder to GCP bucket from R Console</h2><span id='topic+Copy_To_GCP'></span>

<h3>Description</h3>

<p>Run command from R console without moving to terminal to copy folder to GCP bucket
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Copy_To_GCP(folder_file_path, gcp_bucket_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Copy_To_GCP_+3A_folder_file_path">folder_file_path</code></td>
<td>
<p>folder to be copied to GCP bucket.</p>
</td></tr>
<tr><td><code id="Copy_To_GCP_+3A_gcp_bucket_path">gcp_bucket_path</code></td>
<td>
<p>GCP bucket path to copy to files.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.  Performs system copy to GCP bucket.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Copy_To_GCP(folder_file_path = "plots/", gcp_bucket_path = "gs://bucket_name_and_folder_path")

## End(Not run)

</code></pre>

<hr>
<h2 id='Create_10X_H5'>Create H5 from 10X Outputs</h2><span id='topic+Create_10X_H5'></span>

<h3>Description</h3>

<p>Creates HDF5 formatted output analogous to the outputs created by Cell Ranger and can be read into
Seurat, LIGER, or SCE class object.  Requires DropletUtils package from Bioconductor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Create_10X_H5(
  raw_data_file_path,
  source_type = "10X",
  save_file_path,
  save_name
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Create_10X_H5_+3A_raw_data_file_path">raw_data_file_path</code></td>
<td>
<p>file path to raw data file(s).</p>
</td></tr>
<tr><td><code id="Create_10X_H5_+3A_source_type">source_type</code></td>
<td>
<p>type of source data (Default is &quot;10X&quot;).  Alternatively can provide &quot;Matrix&quot; or &quot;data.frame&quot;.</p>
</td></tr>
<tr><td><code id="Create_10X_H5_+3A_save_file_path">save_file_path</code></td>
<td>
<p>file path to directory to save file.</p>
</td></tr>
<tr><td><code id="Create_10X_H5_+3A_save_name">save_name</code></td>
<td>
<p>name prefix for output H5 file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A HDF5 format file that will be recognized as 10X Cell Ranger formatted file by Seurat or LIGER.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Create_10X_H5(raw_data_file_path = "file_path", save_file_path = "file_path2", save_name = "NAME")

## End(Not run)

</code></pre>

<hr>
<h2 id='Create_CellBender_Merged_Seurat'>Create Seurat Object with Cell Bender and Raw data</h2><span id='topic+Create_CellBender_Merged_Seurat'></span>

<h3>Description</h3>

<p>Enables easy creation of Seurat object which contains both cell bender data and raw count data as
separate assays within the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Create_CellBender_Merged_Seurat(
  raw_cell_bender_matrix = NULL,
  raw_counts_matrix = NULL,
  raw_assay_name = "RAW",
  min_cells = 5,
  min_features = 200,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Create_CellBender_Merged_Seurat_+3A_raw_cell_bender_matrix">raw_cell_bender_matrix</code></td>
<td>
<p>matrix file containing the cell bender correct counts.</p>
</td></tr>
<tr><td><code id="Create_CellBender_Merged_Seurat_+3A_raw_counts_matrix">raw_counts_matrix</code></td>
<td>
<p>matrix file contain the uncorrected Cell Ranger (or other) counts.</p>
</td></tr>
<tr><td><code id="Create_CellBender_Merged_Seurat_+3A_raw_assay_name">raw_assay_name</code></td>
<td>
<p>a key value to use specifying the name of assay.  Default is &quot;RAW&quot;.</p>
</td></tr>
<tr><td><code id="Create_CellBender_Merged_Seurat_+3A_min_cells">min_cells</code></td>
<td>
<p>value to supply to min.cells parameter of <code><a href="SeuratObject.html#topic+CreateSeuratObject">CreateSeuratObject</a></code>.
Default is 5.</p>
</td></tr>
<tr><td><code id="Create_CellBender_Merged_Seurat_+3A_min_features">min_features</code></td>
<td>
<p>value to supply to min.features parameter of <code><a href="SeuratObject.html#topic+CreateSeuratObject">CreateSeuratObject</a></code>.
Default is 200.</p>
</td></tr>
<tr><td><code id="Create_CellBender_Merged_Seurat_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="SeuratObject.html#topic+CreateSeuratObject">CreateSeuratObject</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat Object contain both the Cell Bender corrected counts (&quot;RNA&quot; assay) and uncorrected
counts (&quot;RAW&quot; assay; or other name specified to <code>raw_assay_name</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
seurat_obj &lt;- Create_CellBender_Merged_Seurat(raw_cell_bender_matrix = cb_matrix,
raw_counts_matrix = cr_matrix)

## End(Not run)

</code></pre>

<hr>
<h2 id='Create_Cluster_Annotation_File'>Create cluster annotation csv file</h2><span id='topic+Create_Cluster_Annotation_File'></span>

<h3>Description</h3>

<p>create annotation file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Create_Cluster_Annotation_File(
  file_path = NULL,
  file_name = "cluster_annotation"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Create_Cluster_Annotation_File_+3A_file_path">file_path</code></td>
<td>
<p>path to directory to save file.  Default is current working directory.</p>
</td></tr>
<tr><td><code id="Create_Cluster_Annotation_File_+3A_file_name">file_name</code></td>
<td>
<p>name to use for annotation file.  Function automatically adds file type &quot;.csv&quot; suffix.
Default is &quot;cluster_annotation&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value returned.  Creates .csv file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Create_Cluster_Annotation_File(file_path = "cluster_annotation_folder_name")

## End(Not run)

</code></pre>

<hr>
<h2 id='Dark2_Pal'>Dark2 Palette</h2><span id='topic+Dark2_Pal'></span>

<h3>Description</h3>

<p>Shortcut to Dark2 color palette from RColorBrewer (8 Colors)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dark2_Pal()
</code></pre>


<h3>Value</h3>

<p>&quot;Dark2&quot; color palette (8 colors)
</p>


<h3>References</h3>

<p>Dark2 palette from RColorBrewer being called through paletteer.
See RColorBrewer for more info on palettes
<a href="https://CRAN.R-project.org/package=RColorBrewer">https://CRAN.R-project.org/package=RColorBrewer</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cols &lt;- Dark2_Pal()
PalettePlot(pal= cols)

</code></pre>

<hr>
<h2 id='DimPlot_All_Samples'>DimPlot by Meta Data Column</h2><span id='topic+DimPlot_All_Samples'></span>

<h3>Description</h3>

<p>Creates DimPlot layout containing all samples within Seurat Object from orig.ident column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DimPlot_All_Samples(
  seurat_object,
  meta_data_column = "orig.ident",
  colors_use = "black",
  pt.size = NULL,
  aspect_ratio = NULL,
  title_size = 15,
  num_columns = NULL,
  reduction = NULL,
  dims = c(1, 2),
  raster = NULL,
  raster.dpi = c(512, 512),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DimPlot_All_Samples_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_meta_data_column">meta_data_column</code></td>
<td>
<p>Meta data column to split plots by.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_colors_use">colors_use</code></td>
<td>
<p>single color to use for all plots or a vector of colors equal to the number of plots.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_title_size">title_size</code></td>
<td>
<p>size for plot title labels.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_num_columns">num_columns</code></td>
<td>
<p>number of columns in final layout plot.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_dims">dims</code></td>
<td>
<p>Which dimensions to plot.  Defaults to c(1,2) if not specified.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="DimPlot_All_Samples_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+DimPlot">DimPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)

pbmc_small$sample_id &lt;- sample(c("sample1", "sample2"), size = ncol(pbmc_small), replace = TRUE)

DimPlot_All_Samples(seurat_object = pbmc_small, meta_data_column = "sample_id", color = "black",
num_columns = 2)

</code></pre>

<hr>
<h2 id='DimPlot_LIGER'>DimPlot LIGER Version</h2><span id='topic+DimPlot_LIGER'></span>

<h3>Description</h3>

<p>Standard and modified version of LIGER's plotByDatasetAndCluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DimPlot_LIGER(
  liger_object,
  group_by = NULL,
  split_by = NULL,
  colors_use_cluster = NULL,
  colors_use_meta = NULL,
  pt_size = NULL,
  shuffle = TRUE,
  shuffle_seed = 1,
  reduction_label = "UMAP",
  aspect_ratio = NULL,
  label = TRUE,
  label_size = NA,
  label_repel = FALSE,
  label_box = FALSE,
  label_color = "black",
  combination = FALSE,
  raster = NULL,
  raster.dpi = c(512, 512),
  num_columns = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DimPlot_LIGER_+3A_liger_object">liger_object</code></td>
<td>
<p><code>liger</code> liger_object.  Need to perform clustering before calling this function</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_group_by">group_by</code></td>
<td>
<p>Variable to be plotted.  If <code>NULL</code> will plot clusters from <code>liger@clusters</code> slot.
If <code>combination = TRUE</code> will plot both clusters and meta data variable.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_split_by">split_by</code></td>
<td>
<p>Variable to split plots by.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_colors_use_cluster">colors_use_cluster</code></td>
<td>
<p>colors to use for plotting by clusters.  By default if number of levels plotted is
less than or equal to 36 will use &quot;polychrome&quot; and if greater than 36 will use &quot;varibow&quot; with shuffle = TRUE
both from <code><a href="#topic+DiscretePalette_scCustomize">DiscretePalette_scCustomize</a></code>.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_colors_use_meta">colors_use_meta</code></td>
<td>
<p>colors to use for plotting by meta data (cell.data) variable.  By default if number
of levels plotted is less than or equal to 36 it will use &quot;polychrome&quot; and if greater than 36 will use
&quot;varibow&quot; with shuffle = TRUE both from DiscretePalette_scCustomize.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_pt_size">pt_size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_shuffle">shuffle</code></td>
<td>
<p>logical. Whether to randomly shuffle the order of points. This can be useful for crowded plots
if points of interest are being buried. (Default is TRUE).</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_shuffle_seed">shuffle_seed</code></td>
<td>
<p>Sets the seed if randomly shuffling the order of points.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_reduction_label">reduction_label</code></td>
<td>
<p>What to label the x and y axes of resulting plots.  LIGER does not store name of
technique and therefore needs to be set manually.  Default is &quot;UMAP&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_label">label</code></td>
<td>
<p>logical.  Whether or not to label the clusters.  ONLY applies to plotting by cluster.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_label_size">label_size</code></td>
<td>
<p>size of cluster labels.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_label_repel">label_repel</code></td>
<td>
<p>logical.  Whether to repel cluster labels from each other if plotting by
cluster (if <code>group_by = NULL</code> or <code style="white-space: pre;">&#8288;group_by = "cluster&#8288;</code>).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_label_box">label_box</code></td>
<td>
<p>logical.  Whether to put a box around the label text (uses <code>geom_text</code> vs <code>geom_label</code>).
Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_label_color">label_color</code></td>
<td>
<p>Color to use for cluster labels.  Default is &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_combination">combination</code></td>
<td>
<p>logical, whether to return patchwork displaying both plots side by side.  (Default is FALSE).</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_num_columns">num_columns</code></td>
<td>
<p>Number of columns in plot layout.  Only valid if <code>split.by != NULL</code>.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="DimPlot_LIGER_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot/patchwork object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
DimPlot_LIGER(liger_object = obj_name, reduction_label = "UMAP")

## End(Not run)

</code></pre>

<hr>
<h2 id='DimPlot_scCustom'>DimPlot with modified default settings</h2><span id='topic+DimPlot_scCustom'></span>

<h3>Description</h3>

<p>Creates DimPlot with some of the settings modified from their Seurat defaults (colors_use, shuffle, label).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DimPlot_scCustom(
  seurat_object,
  colors_use = NULL,
  pt.size = NULL,
  reduction = NULL,
  group.by = NULL,
  split.by = NULL,
  split_seurat = FALSE,
  figure_plot = FALSE,
  aspect_ratio = NULL,
  shuffle = TRUE,
  seed = 1,
  label = NULL,
  label.size = 4,
  label.color = "black",
  label.box = FALSE,
  dims = c(1, 2),
  repel = FALSE,
  raster = NULL,
  raster.dpi = c(512, 512),
  num_columns = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DimPlot_scCustom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_colors_use">colors_use</code></td>
<td>
<p>color palette to use for plotting.  By default if number of levels plotted is less than
or equal to 36 it will use &quot;polychrome&quot; and if greater than 36 will use &quot;varibow&quot; with shuffle = TRUE
both from <code>DiscretePalette_scCustomize</code>.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_split.by">split.by</code></td>
<td>
<p>Feature to split plots by (i.e. &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_split_seurat">split_seurat</code></td>
<td>
<p>logical.  Whether or not to display split plots like Seurat (shared y axis) or as
individual plots in layout.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_figure_plot">figure_plot</code></td>
<td>
<p>logical.  Whether to remove the axes and plot with legend on left of plot denoting
axes labels.  (Default is FALSE).  Requires <code>split_seurat = TRUE</code>.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_shuffle">shuffle</code></td>
<td>
<p>logical. Whether to randomly shuffle the order of points. This can be useful for crowded
plots if points of interest are being buried. (Default is TRUE).</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_seed">seed</code></td>
<td>
<p>Sets the seed if randomly shuffling the order of points.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_label">label</code></td>
<td>
<p>Whether to label the clusters.  By default if <code>group.by = NULL</code> label = TRUE, and
otherwise it is FALSE.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_label.size">label.size</code></td>
<td>
<p>Sets size of labels.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_label.color">label.color</code></td>
<td>
<p>Sets the color of the label text.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_label.box">label.box</code></td>
<td>
<p>Whether to put a box around the label text (geom_text vs geom_label).</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_dims">dims</code></td>
<td>
<p>Which dimensions to plot.  Defaults to c(1,2) if not specified.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_repel">repel</code></td>
<td>
<p>Repel labels.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_num_columns">num_columns</code></td>
<td>
<p>Number of columns in plot layout.  Only valid if <code>split.by != NULL</code>.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="DimPlot_scCustom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+DimPlot">DimPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>References</h3>

<p>Many of the param names and descriptions are from Seurat to facilitate ease of use as
this is simply a wrapper to alter some of the default parameters <a href="https://github.com/satijalab/seurat/blob/master/R/visualization.R">https://github.com/satijalab/seurat/blob/master/R/visualization.R</a> (License: GPL-3).
<code>figure_plot</code> parameter/code modified from code by Tim Stuart via twitter: <a href="https://twitter.com/timoast/status/1526237116035891200?s=20&amp;t=foJOF81aPSjr1t7pk1cUPg">https://twitter.com/timoast/status/1526237116035891200?s=20&amp;t=foJOF81aPSjr1t7pk1cUPg</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
DimPlot_scCustom(seurat_object = pbmc_small)

</code></pre>

<hr>
<h2 id='DiscretePalette_scCustomize'>Discrete color palettes</h2><span id='topic+DiscretePalette_scCustomize'></span>

<h3>Description</h3>

<p>Helper function to return a number of discrete color palettes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DiscretePalette_scCustomize(
  num_colors,
  palette = NULL,
  shuffle_pal = FALSE,
  seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiscretePalette_scCustomize_+3A_num_colors">num_colors</code></td>
<td>
<p>Number of colors to be generated.</p>
</td></tr>
<tr><td><code id="DiscretePalette_scCustomize_+3A_palette">palette</code></td>
<td>
<p>Options are
&quot;alphabet&quot;, &quot;alphabet2&quot;, &quot;glasbey&quot;, &quot;polychrome&quot;, &quot;stepped&quot;, &quot;ditto_seq&quot;, &quot;varibow&quot;.</p>
</td></tr>
<tr><td><code id="DiscretePalette_scCustomize_+3A_shuffle_pal">shuffle_pal</code></td>
<td>
<p>randomly shuffle the outputted palette.  Most useful for <code>varibow</code> palette as
that is normally an ordered palette.</p>
</td></tr>
<tr><td><code id="DiscretePalette_scCustomize_+3A_seed">seed</code></td>
<td>
<p>random seed for the palette shuffle.  Default = 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of colors
</p>


<h3>References</h3>

<p>This function uses the paletteer package <a href="https://github.com/EmilHvitfeldt/paletteer">https://github.com/EmilHvitfeldt/paletteer</a> to
provide simplified access to color palettes from many different R package sources while
minimizing scCustomize current and future dependencies.
</p>
<p>The following packages &amp; palettes are called by this function (see individual packages for
palette references/citations):
</p>

<ol>
<li><p> pals (via paletteer) <a href="https://CRAN.R-project.org/package=pals">https://CRAN.R-project.org/package=pals</a>
</p>

<ul>
<li><p> alphabet, alphabet2, glasbey, polychrome, and stepped.
</p>
</li></ul>

</li>
<li><p> dittoSeq <a href="https://bioconductor.org/packages/release/bioc/html/dittoSeq.html">https://bioconductor.org/packages/release/bioc/html/dittoSeq.html</a>
</p>

<ul>
<li><p> dittoColors.
</p>
</li></ul>

</li>
<li><p> colorway <a href="https://github.com/hypercompetent/colorway">https://github.com/hypercompetent/colorway</a>
</p>

<ul>
<li><p> varibow
</p>
</li></ul>

</li></ol>

<p>Function name and implementation modified from Seurat (License: GPL-3).
<a href="https://github.com/satijalab/seurat">https://github.com/satijalab/seurat</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pal &lt;- DiscretePalette_scCustomize(num_colors = 36, palette = "varibow")
PalettePlot(pal= pal)

</code></pre>

<hr>
<h2 id='DotPlot_scCustom'>Customized DotPlot</h2><span id='topic+DotPlot_scCustom'></span>

<h3>Description</h3>

<p>Code for creating customized DotPlot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DotPlot_scCustom(
  seurat_object,
  features,
  group.by = NULL,
  colors_use = viridis_plasma_dark_high,
  remove_axis_titles = TRUE,
  x_lab_rotate = FALSE,
  y_lab_rotate = FALSE,
  facet_label_rotate = FALSE,
  flip_axes = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DotPlot_scCustom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_features">features</code></td>
<td>
<p>Features to plot.</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_colors_use">colors_use</code></td>
<td>
<p>specify color palette to used.  Default is viridis_plasma_dark_high.</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_remove_axis_titles">remove_axis_titles</code></td>
<td>
<p>logical. Whether to remove the x and y axis titles.  Default = TRUE.</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>Rotate x-axis labels 45 degrees (Default is FALSE).</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_y_lab_rotate">y_lab_rotate</code></td>
<td>
<p>Rotate x-axis labels 45 degrees (Default is FALSE).</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_facet_label_rotate">facet_label_rotate</code></td>
<td>
<p>Rotate facet labels on grouped <code>DotPlots</code> by 45 degrees (Default is FALSE).</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_flip_axes">flip_axes</code></td>
<td>
<p>whether or not to flip and X and Y axes (Default is FALSE).</p>
</td></tr>
<tr><td><code id="DotPlot_scCustom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+DotPlot">DotPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Seurat)
DotPlot_scCustom(seurat_object = pbmc_small, features = c("CD3E", "CD8", "GZMB", "MS4A1"))


</code></pre>

<hr>
<h2 id='ensembl_mito_id'>Ensembl Mito IDs</h2><span id='topic+ensembl_mito_id'></span>

<h3>Description</h3>

<p>A list of ensembl ids for mitochondrial genes (Ensembl version 105)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensembl_mito_id
</code></pre>


<h3>Format</h3>

<p>A list of six vectors
</p>

<dl>
<dt>Mus_musculus_mito_ensembl</dt><dd><p>Ensembl IDs for mouse mitochondrial genes</p>
</dd>
<dt>Homo_sapiens_mito_ensembl</dt><dd><p>Ensembl IDs for human mitochondrial genes</p>
</dd>
<dt>Danio_rerio_mito_ensembl</dt><dd><p>Ensembl IDs for zebrafish mitochondrial genes</p>
</dd>
<dt>Rattus_norvegicus_mito_ensembl</dt><dd><p>Ensembl IDs for rat mitochondrial genes</p>
</dd>
<dt>Drosophila_melanogaster_mito_ensembl</dt><dd><p>Ensembl IDs for fly mitochondrial genes</p>
</dd>
<dt>Macaca_mulatta_mito_ensembl</dt><dd><p>Ensembl IDs for macaque mitochondrial genes</p>
</dd>
</dl>


<hr>
<h2 id='ensembl_ribo_id'>Ensembl Ribo IDs</h2><span id='topic+ensembl_ribo_id'></span>

<h3>Description</h3>

<p>A list of ensembl ids for ribosomal genes (Ensembl version 105)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensembl_ribo_id
</code></pre>


<h3>Format</h3>

<p>A list of seven vectors
</p>

<dl>
<dt>Mus_musculus_ribo_ensembl</dt><dd><p>Ensembl IDs for mouse ribosomal genes</p>
</dd>
<dt>Homo_sapiens_ribo_ensembl</dt><dd><p>Ensembl IDs for human ribosomal genes</p>
</dd>
<dt>Callithrix_jacchus_ribo_ensembl</dt><dd><p>Ensembl IDs for marmoset ribosomal genes</p>
</dd>
<dt>Danio_rerio_ribo_ensembl</dt><dd><p>Ensembl IDs for zebrafish ribosomal genes</p>
</dd>
<dt>Rattus_norvegicus_ribo_ensembl</dt><dd><p>Ensembl IDs for rat ribosomal genes</p>
</dd>
<dt>Drosophila_melanogaster_ribo_ensembl</dt><dd><p>Ensembl IDs for fly ribosomal genes</p>
</dd>
<dt>Macaca_mulatta_ribo_ensembl</dt><dd><p>Ensembl IDs for macaque ribosomal genes</p>
</dd>
</dl>


<hr>
<h2 id='Extract_Modality'>Extract multi-modal data into list by modality</h2><span id='topic+Extract_Modality'></span>

<h3>Description</h3>

<p>Reorganize multi-modal data after import with <code>Read10X()</code> or scCustomize read functions.
Organizes sub-lists by data modality instead of by sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Extract_Modality(matrix_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Extract_Modality_+3A_matrix_list">matrix_list</code></td>
<td>
<p>list of matrices to split by modality</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of lists, with one sublist per data modality.  Sub-list contain 1 matrix entry per sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
multi_mat &lt;- Read10X(...)
new_multi_mat &lt;- Extract_Modality(matrix_list = multi_mat)

## End(Not run)

</code></pre>

<hr>
<h2 id='Extract_Sample_Meta'>Extract sample level meta.data</h2><span id='topic+Extract_Sample_Meta'></span>

<h3>Description</h3>

<p>Returns a by identity meta.data data.frame with one row per sample.  Useful for downstream
quick view of sample breakdown, meta data table creation, and/or use in pseudobulk analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Extract_Sample_Meta(
  object,
  sample_name = "orig.ident",
  variables_include = NULL,
  variables_exclude = NULL,
  include_all = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Extract_Sample_Meta_+3A_object">object</code></td>
<td>
<p>Seurat object</p>
</td></tr>
<tr><td><code id="Extract_Sample_Meta_+3A_sample_name">sample_name</code></td>
<td>
<p>meta.data column to use as sample.  Output data.frame will contain one row per
level or unique value in this variable.</p>
</td></tr>
<tr><td><code id="Extract_Sample_Meta_+3A_variables_include">variables_include</code></td>
<td>
<p><code style="white-space: pre;">&#8288;@meta.data&#8288;</code> columns to keep in final data.frame.  All other columns will
be discarded.  Default is NULL.</p>
</td></tr>
<tr><td><code id="Extract_Sample_Meta_+3A_variables_exclude">variables_exclude</code></td>
<td>
<p>columns to discard in final data.frame.  Many cell level columns are
irrelevant at the sample level (e.g., nFeature_RNA, percent_mito).
</p>

<ul>
<li><p> Default parameter value is <code>NULL</code> but internally will set to discard nFeature_ASSAY(s),
nCount_ASSAY(s), percent_mito, percent_ribo, percent_mito_ribo, and log10GenesPerUMI.
</p>
</li>
<li><p> If sample level median values are desired for these type of variables the output of this
function can be joined with output of <code><a href="#topic+Median_Stats">Median_Stats</a></code>.
</p>
</li>
<li><p> Set parameter to <code>include_all = TRUE</code> to prevent any columns from being excluded.
</p>
</li></ul>
</td></tr>
<tr><td><code id="Extract_Sample_Meta_+3A_include_all">include_all</code></td>
<td>
<p>logical, whether or not to include all object meta data columns in output data.frame.
Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with one row per <code>sample_name</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
pbmc_small[["batch"]] &lt;- sample(c("batch1", "batch2"), size = ncol(pbmc_small), replace = TRUE)

sample_meta &lt;- Extract_Sample_Meta(object = pbmc_small, sample_name = "orig.ident")

# Only return specific columns from meta data (orig.ident and batch)
sample_meta2 &lt;- Extract_Sample_Meta(object = pbmc_small, sample_name = "orig.ident",
variables_include = "batch")

# Return all columns from meta data
sample_meta3 &lt;- Extract_Sample_Meta(object = pbmc_small, sample_name = "orig.ident",
include_all = TRUE)

</code></pre>

<hr>
<h2 id='Extract_Top_Markers'>Extract Top N Marker Genes</h2><span id='topic+Extract_Top_Markers'></span>

<h3>Description</h3>

<p>Extract vector gene list (or named gene vector) from data.frame results of <code><a href="Seurat.html#topic+FindAllMarkers">FindAllMarkers</a></code>
or similar analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Extract_Top_Markers(
  marker_dataframe,
  num_genes = 10,
  group_by = "cluster",
  rank_by = "avg_log2FC",
  gene_column = "gene",
  gene_rownames_to_column = FALSE,
  data_frame = FALSE,
  named_vector = TRUE,
  make_unique = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Extract_Top_Markers_+3A_marker_dataframe">marker_dataframe</code></td>
<td>
<p>data.frame output from <code><a href="Seurat.html#topic+FindAllMarkers">FindAllMarkers</a></code> or similar analysis.</p>
</td></tr>
<tr><td><code id="Extract_Top_Markers_+3A_num_genes">num_genes</code></td>
<td>
<p>number of genes per group (e.g., cluster) to include in output list.</p>
</td></tr>
<tr><td><code id="Extract_Top_Markers_+3A_group_by">group_by</code></td>
<td>
<p>column name of <code>marker_dataframe</code> to group data by.  Default is &quot;cluster&quot; based on
<code><a href="Seurat.html#topic+FindAllMarkers">FindAllMarkers</a></code>.</p>
</td></tr>
<tr><td><code id="Extract_Top_Markers_+3A_rank_by">rank_by</code></td>
<td>
<p>column name of <code>marker_dataframe</code> to rank data by when selecting <code>num_genes</code> per <code>group_by</code>.
Default is &quot;avg_log2FC&quot; based on <code><a href="Seurat.html#topic+FindAllMarkers">FindAllMarkers</a></code>.</p>
</td></tr>
<tr><td><code id="Extract_Top_Markers_+3A_gene_column">gene_column</code></td>
<td>
<p>column name of <code>marker_dataframe</code> that contains the gene IDs.  Default is &quot;gene&quot;
based on <code><a href="Seurat.html#topic+FindAllMarkers">FindAllMarkers</a></code>.</p>
</td></tr>
<tr><td><code id="Extract_Top_Markers_+3A_gene_rownames_to_column">gene_rownames_to_column</code></td>
<td>
<p>logical. Whether gene IDs are stored in rownames and should be moved to
column.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Extract_Top_Markers_+3A_data_frame">data_frame</code></td>
<td>
<p>Logical, whether or not to return filtered data.frame of the original <code>markers_dataframe</code> or
to return a vector of gene IDs.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Extract_Top_Markers_+3A_named_vector">named_vector</code></td>
<td>
<p>Logical, whether or not to name the vector of gene names that is returned by the function.
If <code>TRUE</code> will name the vector using the column provided to <code>group_by</code>.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Extract_Top_Markers_+3A_make_unique">make_unique</code></td>
<td>
<p>Logical, whether an unnamed vector should return only unique values.  Default is FALSE.
Not applicable when <code>data_frame = TRUE</code> or <code>named_vector = TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>filtered data.frame, vector, or named vector containing gene IDs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
top10_genes &lt;- Extract_Top_Markers(marker_dataframe = markers_results, num_genes = 10,
group_by = "cluster", rank_by = "avg_log2FC")

## End(Not run)

</code></pre>

<hr>
<h2 id='Feature_Present'>Check if genes/features are present</h2><span id='topic+Feature_Present'></span>

<h3>Description</h3>

<p>Check if genes are present in object and return vector of found genes.  Return warning messages for
genes not found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Feature_Present(
  data,
  features,
  case_check = TRUE,
  case_check_msg = TRUE,
  print_msg = TRUE,
  omit_warn = TRUE,
  return_none = FALSE,
  seurat_assay = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Feature_Present_+3A_data">data</code></td>
<td>
<p>Name of input data.  Currently only data of classes: Seurat, liger, data.frame,
dgCMatrix, dgTMatrix, tibble are accepted.  Gene_IDs must be present in rownames of the data.</p>
</td></tr>
<tr><td><code id="Feature_Present_+3A_features">features</code></td>
<td>
<p>vector of features to check.</p>
</td></tr>
<tr><td><code id="Feature_Present_+3A_case_check">case_check</code></td>
<td>
<p>logical. Whether or not to check if features are found if the case is changed from the
input list (Sentence case to Upper and vice versa).  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Feature_Present_+3A_case_check_msg">case_check_msg</code></td>
<td>
<p>logical. Whether to print message to console if alternate case features are found
in addition to inclusion in returned list.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Feature_Present_+3A_print_msg">print_msg</code></td>
<td>
<p>logical. Whether message should be printed if all features are found.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Feature_Present_+3A_omit_warn">omit_warn</code></td>
<td>
<p>logical. Whether to print message about features that are not found in current object.
Default is TRUE.</p>
</td></tr>
<tr><td><code id="Feature_Present_+3A_return_none">return_none</code></td>
<td>
<p>logical. Whether list of found vs. bad features should still be returned if no
features are found.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Feature_Present_+3A_seurat_assay">seurat_assay</code></td>
<td>
<p>Name of assay to pull feature names from if <code>data</code> is Seurat Object.
Default is NULL which will check against features from all assays present.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length 3 containing 1) found features, 2) not found features, 3) features found if
case was modified.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
features &lt;- Feature_Present(data = obj_name, features = DEG_list, print_msg = TRUE,
case_check = TRUE)
found_features &lt;- features[[1]]

## End(Not run)

</code></pre>

<hr>
<h2 id='FeaturePlot_DualAssay'>Customize FeaturePlot of two assays</h2><span id='topic+FeaturePlot_DualAssay'></span>

<h3>Description</h3>

<p>Create Custom FeaturePlots and preserve scale (no binning) from same features in two assays
simultaneously.  Intended for plotting same modality present in two assays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FeaturePlot_DualAssay(
  seurat_object,
  features,
  assay1 = "RAW",
  assay2 = "RNA",
  colors_use = viridis_plasma_dark_high,
  na_color = "lightgray",
  order = TRUE,
  pt.size = NULL,
  aspect_ratio = NULL,
  reduction = NULL,
  na_cutoff = 1e-09,
  raster = NULL,
  raster.dpi = c(512, 512),
  slot = deprecated(),
  layer = "data",
  num_columns = NULL,
  alpha_exp = NULL,
  alpha_na_exp = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FeaturePlot_DualAssay_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_features">features</code></td>
<td>
<p>Feature(s) to plot.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_assay1">assay1</code></td>
<td>
<p>name of assay one.  Default is &quot;RAW&quot; as featured in <code><a href="#topic+Create_CellBender_Merged_Seurat">Create_CellBender_Merged_Seurat</a></code></p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_assay2">assay2</code></td>
<td>
<p>name of assay two  Default is &quot;RNA&quot; as featured in <code><a href="#topic+Create_CellBender_Merged_Seurat">Create_CellBender_Merged_Seurat</a></code></p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_colors_use">colors_use</code></td>
<td>
<p>list of colors or color palette to use.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_na_color">na_color</code></td>
<td>
<p>color to use for points below lower limit.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_order">order</code></td>
<td>
<p>whether to move positive cells to the top (default = TRUE).</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_na_cutoff">na_cutoff</code></td>
<td>
<p>Value to use as minimum expression cutoff.  To set no cutoff set to <code>NA</code>.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_slot">slot</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> soft-deprecated.  See <code>layer</code></p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_layer">layer</code></td>
<td>
<p>Which layer to pull expression data from?  Default is &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_num_columns">num_columns</code></td>
<td>
<p>Number of columns in plot layout.  If number of features &gt; 1 then <code>num_columns</code>
dictates the number of columns in overall layout (<code>num_columns = 1</code> means stacked layout &amp; <code>num_columns = 2</code>
means adjacent layout).</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_alpha_exp">alpha_exp</code></td>
<td>
<p>new alpha level to apply to expressing cell color palette (<code>colors_use</code>).  Must be
value between 0-1.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_alpha_na_exp">alpha_na_exp</code></td>
<td>
<p>new alpha level to apply to non-expressing cell color palette (<code>na_color</code>).  Must be
value between 0-1.</p>
</td></tr>
<tr><td><code id="FeaturePlot_DualAssay_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+FeaturePlot">FeaturePlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
FeaturePlot_DualAssay(seurat_object = object, features = "Cx3cr1", assay1 = "RAW", assay2 = "RNA",
colors_use = viridis_plasma_dark_high, na_color = "lightgray")

## End(Not run)

</code></pre>

<hr>
<h2 id='FeaturePlot_scCustom'>Customize FeaturePlot</h2><span id='topic+FeaturePlot_scCustom'></span>

<h3>Description</h3>

<p>Create Custom FeaturePlots and preserve scale (no binning)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FeaturePlot_scCustom(
  seurat_object,
  features,
  colors_use = viridis_plasma_dark_high,
  na_color = "lightgray",
  order = TRUE,
  pt.size = NULL,
  reduction = NULL,
  na_cutoff = 1e-09,
  raster = NULL,
  raster.dpi = c(512, 512),
  split.by = NULL,
  split_collect = NULL,
  aspect_ratio = NULL,
  figure_plot = FALSE,
  num_columns = NULL,
  slot = deprecated(),
  layer = "data",
  alpha_exp = NULL,
  alpha_na_exp = NULL,
  label = FALSE,
  label_feature_yaxis = FALSE,
  combine = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FeaturePlot_scCustom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_features">features</code></td>
<td>
<p>Feature(s) to plot.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_colors_use">colors_use</code></td>
<td>
<p>list of colors or color palette to use.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_na_color">na_color</code></td>
<td>
<p>color to use for points below lower limit.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_order">order</code></td>
<td>
<p>whether to move positive cells to the top (default = TRUE).</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_na_cutoff">na_cutoff</code></td>
<td>
<p>Value to use as minimum expression cutoff.  This will be lowest value plotted use
palette provided to <code>colors_use</code>.  Leave as default value to plot only positive non-zero values using
color scale and zero/negative values as NA.  To plot all values using color palette set to <code>NA</code>.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_split.by">split.by</code></td>
<td>
<p>Variable in <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> to split the plot by.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_split_collect">split_collect</code></td>
<td>
<p>logical, whether to collect the legends/guides when plotting with <code>split.by</code>.
Default is TRUE if one value is provided to <code>features</code> otherwise is set to FALSE.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_figure_plot">figure_plot</code></td>
<td>
<p>logical.  Whether to remove the axes and plot with legend on left of plot denoting
axes labels.  (Default is FALSE).  Requires <code>split_seurat = TRUE</code>.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_num_columns">num_columns</code></td>
<td>
<p>Number of columns in plot layout.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_slot">slot</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> soft-deprecated.  See <code>layer</code></p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_layer">layer</code></td>
<td>
<p>Which layer to pull expression data from?  Default is &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_alpha_exp">alpha_exp</code></td>
<td>
<p>new alpha level to apply to expressing cell color palette (<code>colors_use</code>).  Must be
value between 0-1.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_alpha_na_exp">alpha_na_exp</code></td>
<td>
<p>new alpha level to apply to non-expressing cell color palette (<code>na_color</code>).  Must be
value between 0-1.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_label">label</code></td>
<td>
<p>logical, whether to label the clusters.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_label_feature_yaxis">label_feature_yaxis</code></td>
<td>
<p>logical, whether to place feature labels on secondary y-axis as opposed to
above legend key.  Default is FALSE.  When setting <code>label_feature_yaxis = TRUE</code> the number of columns
in plot output will automatically be set to the number of levels in <code style="white-space: pre;">&#8288;split.by'&#8288;</code></p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_combine">combine</code></td>
<td>
<p>Combine plots into a single <code><a href="patchwork.html#topic+patchwork">patchwork</a>ed</code> ggplot object.
If FALSE, return a list of ggplot objects.</p>
</td></tr>
<tr><td><code id="FeaturePlot_scCustom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+FeaturePlot">FeaturePlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
FeaturePlot_scCustom(seurat_object = pbmc_small, features = "CD3E",
colors_use = viridis_plasma_dark_high, na_color = "lightgray")

</code></pre>

<hr>
<h2 id='FeatureScatter_scCustom'>Modified version of FeatureScatter</h2><span id='topic+FeatureScatter_scCustom'></span>

<h3>Description</h3>

<p>Create customized FeatureScatter plots with scCustomize defaults.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FeatureScatter_scCustom(
  seurat_object,
  feature1 = NULL,
  feature2 = NULL,
  colors_use = NULL,
  pt.size = NULL,
  group.by = NULL,
  split.by = NULL,
  split_seurat = FALSE,
  shuffle = TRUE,
  aspect_ratio = NULL,
  title_size = 15,
  plot.cor = TRUE,
  num_columns = NULL,
  raster = NULL,
  raster.dpi = c(512, 512),
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FeatureScatter_scCustom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_feature1">feature1</code></td>
<td>
<p>First feature to plot.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_feature2">feature2</code></td>
<td>
<p>Second feature to plot.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_colors_use">colors_use</code></td>
<td>
<p>color for the points on plot.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident).
Default is active ident.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_split.by">split.by</code></td>
<td>
<p>Feature to split plots by (i.e. &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_split_seurat">split_seurat</code></td>
<td>
<p>logical.  Whether or not to display split plots like Seurat (shared y axis) or as
individual plots in layout.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_shuffle">shuffle</code></td>
<td>
<p>logical, whether to randomly shuffle the order of points. This can be useful for crowded plots if points of interest are being buried. Default is TRUE.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_title_size">title_size</code></td>
<td>
<p>size for plot title labels. Does NOT apply if <code>split_seurat = TRUE</code>.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_plot.cor">plot.cor</code></td>
<td>
<p>Display correlation in plot subtitle (or title if <code>split_seurat = TRUE</code>).</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_num_columns">num_columns</code></td>
<td>
<p>number of columns in final layout plot.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="FeatureScatter_scCustom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+FeatureScatter">FeatureScatter</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Seurat)

pbmc_small$sample_id &lt;- sample(c("sample1", "sample2"), size = ncol(pbmc_small), replace = TRUE)

FeatureScatter_scCustom(seurat_object = pbmc_small, feature1 = "nCount_RNA",
feature2 = "nFeature_RNA", split.by = "sample_id")


</code></pre>

<hr>
<h2 id='Fetch_Meta'>Get meta data from object</h2><span id='topic+Fetch_Meta'></span><span id='topic+Fetch_Meta.Seurat'></span><span id='topic+Fetch_Meta.liger'></span>

<h3>Description</h3>

<p>Quick function to properly pull meta.data from objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Fetch_Meta(object)

## S3 method for class 'Seurat'
Fetch_Meta(object)

## S3 method for class 'liger'
Fetch_Meta(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Fetch_Meta_+3A_object">object</code></td>
<td>
<p>Object of class Seurat or liger.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing cell-level meta data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
meta_data &lt;- Fetch_Meta(object = pbmc_small)
head(meta_data, 5)

</code></pre>

<hr>
<h2 id='Gene_Present'>Check if genes/features are present  <a href="https://lifecycle.r-lib.org/articles/stages.html#soft-deprecated"><img src="../help/figures/lifecycle-soft-deprecated.svg" alt='[Soft-deprecated]' /></a></h2><span id='topic+Gene_Present'></span>

<h3>Description</h3>

<p>Check if genes are present in object and return vector of found genes.  Return warning messages for
genes not found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gene_Present(
  data,
  gene_list,
  case_check = TRUE,
  case_check_msg = TRUE,
  print_msg = TRUE,
  omit_warn = TRUE,
  return_none = FALSE,
  seurat_assay = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Gene_Present_+3A_data">data</code></td>
<td>
<p>Name of input data.  Currently only data of classes: Seurat, liger, data.frame,
dgCMatrix, dgTMatrix, tibble are accepted.  Gene_IDs must be present in rownames of the data.</p>
</td></tr>
<tr><td><code id="Gene_Present_+3A_gene_list">gene_list</code></td>
<td>
<p>vector of genes to check.</p>
</td></tr>
<tr><td><code id="Gene_Present_+3A_case_check">case_check</code></td>
<td>
<p>logical. Whether or not to check if features are found if the case is changed from the
input list (Sentence case to Upper and vice versa).  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Gene_Present_+3A_case_check_msg">case_check_msg</code></td>
<td>
<p>logical. Whether to print message to console if alternate case features are found
in addition to inclusion in returned list.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Gene_Present_+3A_print_msg">print_msg</code></td>
<td>
<p>logical. Whether message should be printed if all features are found.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Gene_Present_+3A_omit_warn">omit_warn</code></td>
<td>
<p>logical. Whether to print message about features that are not found in current object.
Default is TRUE.</p>
</td></tr>
<tr><td><code id="Gene_Present_+3A_return_none">return_none</code></td>
<td>
<p>logical. Whether list of found vs. bad features should still be returned if no
features are found.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Gene_Present_+3A_seurat_assay">seurat_assay</code></td>
<td>
<p>Name of assay to pull feature names from if <code>data</code> is Seurat Object.
Default is NULL which will check against features from all assays present.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length 3 containing 1) found features, 2) not found features, 3) features found if
case was modified.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
features &lt;- Gene_Present(data = obj_name, gene_list = DEG_list, print_msg = TRUE, case_check = TRUE)
found_features &lt;- features[[1]]

## End(Not run)

</code></pre>

<hr>
<h2 id='Hue_Pal'>Hue_Pal</h2><span id='topic+Hue_Pal'></span>

<h3>Description</h3>

<p>Shortcut to hue_pal to return to ggplot2 defaults if user desires, from scales package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hue_Pal(num_colors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Hue_Pal_+3A_num_colors">num_colors</code></td>
<td>
<p>number of colors to return in palette.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>hue color palette (as many colors as desired)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cols &lt;- Hue_Pal(num_colors = 8)
PalettePlot(pal= cols)

</code></pre>

<hr>
<h2 id='ieg_gene_list'>Immediate Early Gene (IEG) gene lists</h2><span id='topic+ieg_gene_list'></span>

<h3>Description</h3>

<p>Gene symbols for immediate early genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ieg_gene_list
</code></pre>


<h3>Format</h3>

<p>A list of seven vectors
</p>

<dl>
<dt>Mus_musculus_IEGs</dt><dd><p>Gene symbols for IEGs from source publication (see below)</p>
</dd>
<dt>Homo_sapiens_IEGs</dt><dd><p>Human gene symbols for homologous genes from mouse gene list</p>
</dd>
</dl>



<h3>Source</h3>

<p>Mouse gene list is from: SI Table 4 from <a href="https://doi.org/10.1016/j.neuron.2017.09.026">doi:10.1016/j.neuron.2017.09.026</a>.  Human
gene list was compiled by first creating homologous gene list using biomaRt and then adding some manually curated
homologs according to HGNC.
</p>

<hr>
<h2 id='Iterate_Barcode_Rank_Plot'>Iterative Barcode Rank Plots</h2><span id='topic+Iterate_Barcode_Rank_Plot'></span>

<h3>Description</h3>

<p>Read data, calculate <code>DropletUtils::barcodeRanks</code>, create barcode rank plots, and outout single PDF output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_Barcode_Rank_Plot(
  dir_path_h5,
  multi_directory = TRUE,
  h5_filename = "raw_feature_bc_matrix.h5",
  cellranger_multi = FALSE,
  parallel = FALSE,
  num_cores = NULL,
  file_path = NULL,
  file_name = NULL,
  pt.size = 6,
  raster_dpi = c(1024, 1024),
  plateau = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_dir_path_h5">dir_path_h5</code></td>
<td>
<p>path to parent directory (if <code>multi_directory = TRUE</code>) or directory containing
all h5 files (if <code>multi_directory = FALSE</code>).</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_multi_directory">multi_directory</code></td>
<td>
<p>logical, whether or not all h5 files are in their own subdirectories or in a
single directory (default is TRUE; each in own subdirectory (e.g. output from Cell Ranger)).</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_h5_filename">h5_filename</code></td>
<td>
<p>Either the file name of h5 file (if <code>multi_directory = TRUE</code>) or the shared
suffix (if <code>multi_directory = FALSE</code>)</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_cellranger_multi">cellranger_multi</code></td>
<td>
<p>logical, whether the outputs to be read are from Cell Ranger <code>multi</code> as opposed
to Cell Ranger <code>count</code> (default is FALSE).  Only valid if <code>multi_directory = FALSE</code>.</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_parallel">parallel</code></td>
<td>
<p>logical, should files be read in parallel (default is FALSE).</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_num_cores">num_cores</code></td>
<td>
<p>Number of cores to use in parallel if <code>parallel = TRUE</code>.</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_file_path">file_path</code></td>
<td>
<p>file path to use for saving PDF output.</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_file_name">file_name</code></td>
<td>
<p>Name of PDF output file.</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_pt.size">pt.size</code></td>
<td>
<p>point size for plotting, default is 6.</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_raster_dpi">raster_dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(1024, 1024).</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_plateau">plateau</code></td>
<td>
<p>numerical values at which to add vertical line designating estimated
empty droplet plateau (default is NULL).  Must be vector equal in length to number of samples.</p>
</td></tr>
<tr><td><code id="Iterate_Barcode_Rank_Plot_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to <code>Read10X_h5_Multi_Directory</code> or <code>Read10X_h5_GEO</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pdf document
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_Barcode_Rank_Plot(dir_path_h5 = "H5_PATH/", multi_directory = TRUE,
h5_filename = "raw_feature_bc_matrix", parallel = TRUE, num_cores = 12, file_path = "OUTPUT_PATH",
file_name = "Barcode_Rank_Plots")

## End(Not run)

</code></pre>

<hr>
<h2 id='Iterate_Cluster_Highlight_Plot'>Iterate Cluster Highlight Plot</h2><span id='topic+Iterate_Cluster_Highlight_Plot'></span>

<h3>Description</h3>

<p>Iterate the create plots with cluster of interest highlighted across all cluster (active.idents)
in given Seurat Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_Cluster_Highlight_Plot(
  seurat_object,
  highlight_color = "dodgerblue",
  background_color = "lightgray",
  pt.size = NULL,
  reduction = NULL,
  file_path = NULL,
  file_name = NULL,
  file_type = NULL,
  single_pdf = FALSE,
  dpi = 600,
  raster = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>Color to highlight cells (default &quot;navy&quot;).  Can provide either single color to use for
all clusters/plots or a vector of colors equal to the number of clusters to use (in order) for the clusters/plots.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_background_color">background_color</code></td>
<td>
<p>non-highlighted cell colors.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_pt.size">pt.size</code></td>
<td>
<p>point size for both highlighted cluster and background.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_file_name">file_name</code></td>
<td>
<p>name suffix to append after sample name.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save output as.  Must be one of following: &quot;.pdf&quot;, &quot;.png&quot;, &quot;.tiff&quot;, &quot;.jpeg&quot;, or &quot;.svg&quot;.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_single_pdf">single_pdf</code></td>
<td>
<p>saves all plots to single PDF file (default = FALSE).  'file_type&ldquo; must be .pdf</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_dpi">dpi</code></td>
<td>
<p>dpi for image saving.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="Iterate_Cluster_Highlight_Plot_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to<code><a href="Seurat.html#topic+DimPlot">DimPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saved plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_Cluster_Highlight_Plot(seurat_object = object, highlight_color = "navy",
background_color = "lightgray", file_path = "path/", file_name = "name", file_type = "pdf",
single_pdf = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='Iterate_DimPlot_bySample'>Iterate DimPlot By Sample</h2><span id='topic+Iterate_DimPlot_bySample'></span>

<h3>Description</h3>

<p>Iterate DimPlot by orig.ident column from Seurat object metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_DimPlot_bySample(
  seurat_object,
  sample_column = "orig.ident",
  file_path = NULL,
  file_name = NULL,
  file_type = NULL,
  single_pdf = FALSE,
  dpi = 600,
  color = "black",
  no_legend = TRUE,
  title_prefix = NULL,
  reduction = NULL,
  dims = c(1, 2),
  pt.size = NULL,
  raster = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_sample_column">sample_column</code></td>
<td>
<p>name of meta.data column containing sample names/ids (default is &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_file_name">file_name</code></td>
<td>
<p>name suffix to append after sample name.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_file_type">file_type</code></td>
<td>
<p>File type to save output as.  Must be one of following: &quot;.pdf&quot;, &quot;.png&quot;, &quot;.tiff&quot;, &quot;.jpeg&quot;, or &quot;.svg&quot;.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_single_pdf">single_pdf</code></td>
<td>
<p>saves all plots to single PDF file (default = FALSE).  'file_type&ldquo; must be .pdf</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_dpi">dpi</code></td>
<td>
<p>dpi for image saving.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_color">color</code></td>
<td>
<p>color scheme to use.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_no_legend">no_legend</code></td>
<td>
<p>logical, whether or not to include plot legend, default is TRUE.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_title_prefix">title_prefix</code></td>
<td>
<p>Value that should be used for plot title prefix if <code>no_legend = TRUE</code>.
If NULL the value of <code>meta_data_column</code> will be used.  Default is NULL.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (default is object default).</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_dims">dims</code></td>
<td>
<p>Dimensions to plot.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="Iterate_DimPlot_bySample_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+DimPlot">DimPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_DimPlot_bySample(seurat_object = object, file_path = "plots/", file_name = "tsne",
file_type = ".jpg", dpi = 600, color = "black")

## End(Not run)

</code></pre>

<hr>
<h2 id='Iterate_FeaturePlot_scCustom'>Iterative Plotting of Gene Lists using Custom FeaturePlots</h2><span id='topic+Iterate_FeaturePlot_scCustom'></span>

<h3>Description</h3>

<p>Create and Save plots for Gene list with Single Command
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_FeaturePlot_scCustom(
  seurat_object,
  features,
  gene_list = deprecated(),
  colors_use = viridis_plasma_dark_high,
  na_color = "lightgray",
  na_cutoff = 1e-09,
  split.by = NULL,
  order = TRUE,
  return_plots = FALSE,
  file_path = NULL,
  file_name = NULL,
  file_type = NULL,
  single_pdf = FALSE,
  features_per_page = 1,
  num_columns = NULL,
  landscape = TRUE,
  dpi = 600,
  pt.size = NULL,
  reduction = NULL,
  raster = NULL,
  alpha_exp = NULL,
  alpha_na_exp = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_features">features</code></td>
<td>
<p>vector of features to plot.  If a named vector is provided then the names for each gene
will be incorporated into plot title if <code>single_pdf = TRUE</code> or into file name if <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_gene_list">gene_list</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> soft-deprecated.  See <code>features</code>.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_colors_use">colors_use</code></td>
<td>
<p>color scheme to use.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_na_color">na_color</code></td>
<td>
<p>color for non-expressed cells.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_na_cutoff">na_cutoff</code></td>
<td>
<p>Value to use as minimum expression cutoff.  To set no cutoff set to <code>NA</code>.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_split.by">split.by</code></td>
<td>
<p>Variable in <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> to split the plot by.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_order">order</code></td>
<td>
<p>whether to move positive cells to the top (default = TRUE).</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_return_plots">return_plots</code></td>
<td>
<p>logical. Whether to return plots to list instead of saving them to file(s).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_file_name">file_name</code></td>
<td>
<p>name suffix and file extension.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_file_type">file_type</code></td>
<td>
<p>File type to save output as.  Must be one of following: &quot;.pdf&quot;, &quot;.png&quot;, &quot;.tiff&quot;, &quot;.jpeg&quot;, or &quot;.svg&quot;.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_single_pdf">single_pdf</code></td>
<td>
<p>saves all plots to single PDF file (default = FALSE).</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_features_per_page">features_per_page</code></td>
<td>
<p>numeric, number of features to plot on single page if <code>single_pdf = TRUE</code>.  Default is 1.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_num_columns">num_columns</code></td>
<td>
<p>Number of columns in plot layout (only applicable if <code>single_pdf = TRUE</code> AND
<code>features_per_page</code> &gt; 1).</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_landscape">landscape</code></td>
<td>
<p>logical, when plotting multiple features per page in single PDF whether to use landscape or portrait
page dimensions (default is TRUE).</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_dpi">dpi</code></td>
<td>
<p>dpi for image saving.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_alpha_exp">alpha_exp</code></td>
<td>
<p>new alpha level to apply to expressing cell color palette (<code>colors_use</code>).  Must be
value between 0-1.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_alpha_na_exp">alpha_na_exp</code></td>
<td>
<p>new alpha level to apply to non-expressing cell color palette (<code>na_color</code>).  Must be
value between 0-1.</p>
</td></tr>
<tr><td><code id="Iterate_FeaturePlot_scCustom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+FeaturePlot">FeaturePlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saved plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_FeaturePlot_scCustom(seurat_object = object, gene_list = DEG_list,
colors_use = viridis_plasma_dark_high, na_color = "lightgray", file_path = "plots/",
file_name = "tsne", file_type = ".jpg", dpi = 600)

## End(Not run)

</code></pre>

<hr>
<h2 id='Iterate_Meta_Highlight_Plot'>Iterate Meta Highlight Plot</h2><span id='topic+Iterate_Meta_Highlight_Plot'></span>

<h3>Description</h3>

<p>Iterate the create plots with meta data variable of interest highlighted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_Meta_Highlight_Plot(
  seurat_object,
  meta_data_column,
  new_meta_order = NULL,
  meta_data_sort = TRUE,
  highlight_color = "navy",
  background_color = "lightgray",
  pt.size = NULL,
  no_legend = FALSE,
  title_prefix = NULL,
  reduction = NULL,
  file_path = NULL,
  file_name = NULL,
  file_type = NULL,
  single_pdf = FALSE,
  dpi = 600,
  raster = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_meta_data_column">meta_data_column</code></td>
<td>
<p>Name of the column in <code>seurat_object@meta.data</code> slot to pull value from for highlighting.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_new_meta_order">new_meta_order</code></td>
<td>
<p>The order in which to plot each level within <code>meta_data_column</code> if <code>single_PDF</code> is TRUE.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_meta_data_sort">meta_data_sort</code></td>
<td>
<p>logical.  Whether or not to sort and relevel the levels in <code>meta_data_column</code> if
<code>single_PDF</code> is TRUE.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>Color to highlight cells (default &quot;navy&quot;).  Can provide either single color to use for
all clusters/plots or a vector of colors equal to the number of clusters to use (in order) for the clusters/plots.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_background_color">background_color</code></td>
<td>
<p>non-highlighted cell colors.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_pt.size">pt.size</code></td>
<td>
<p>point size for both highlighted cluster and background.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_no_legend">no_legend</code></td>
<td>
<p>logical, whether or not to remove plot legend and move to plot title.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_title_prefix">title_prefix</code></td>
<td>
<p>Value that should be used for plot title prefix if <code>no_legend = TRUE</code>.
If NULL the value of <code>meta_data_column</code> will be used.  Default is NULL.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_file_name">file_name</code></td>
<td>
<p>name suffix to append after sample name.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_file_type">file_type</code></td>
<td>
<p>File type to save output as.  Must be one of following: &quot;.pdf&quot;, &quot;.png&quot;, &quot;.tiff&quot;, &quot;.jpeg&quot;, or &quot;.svg&quot;.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_single_pdf">single_pdf</code></td>
<td>
<p>saves all plots to single PDF file (default = FALSE).  'file_type&ldquo; must be .pdf</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_dpi">dpi</code></td>
<td>
<p>dpi for image saving.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="Iterate_Meta_Highlight_Plot_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to<code><a href="Seurat.html#topic+DimPlot">DimPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saved plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_Meta_Highlight_Plot(seurat_object = object, meta_data_column = "sample_id",
highlight_color = "navy", background_color = "lightgray", file_path = "path/",
file_name = "name", file_type = "pdf", single_pdf = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='Iterate_PC_Loading_Plots'>Iterate PC Loading Plots</h2><span id='topic+Iterate_PC_Loading_Plots'></span>

<h3>Description</h3>

<p>Plot PC Heatmaps and Dim Loadings for exploratory analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_PC_Loading_Plots(
  seurat_object,
  dims_plot = NULL,
  file_path = NULL,
  name_prefix = NULL,
  file_name = "PC_Loading_Plots",
  return_plots = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_PC_Loading_Plots_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Iterate_PC_Loading_Plots_+3A_dims_plot">dims_plot</code></td>
<td>
<p>number of PCs to plot (integer).  Default is all dims present in PCA.</p>
</td></tr>
<tr><td><code id="Iterate_PC_Loading_Plots_+3A_file_path">file_path</code></td>
<td>
<p>directory file path to save file.</p>
</td></tr>
<tr><td><code id="Iterate_PC_Loading_Plots_+3A_name_prefix">name_prefix</code></td>
<td>
<p>prefix for file name (optional).</p>
</td></tr>
<tr><td><code id="Iterate_PC_Loading_Plots_+3A_file_name">file_name</code></td>
<td>
<p>suffix for file name.  Default is &quot;PC_Loading_Plots&quot;.</p>
</td></tr>
<tr><td><code id="Iterate_PC_Loading_Plots_+3A_return_plots">return_plots</code></td>
<td>
<p>Whether to return the plot list (Default is FALSE).  Must assign to environment
to save plot list.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of plots outputted as pdf
</p>


<h3>See Also</h3>

<p><code><a href="Seurat.html#topic+PCHeatmap">PCHeatmap</a></code> and <code><a href="Seurat.html#topic+VizDimLoadings">VizDimLoadings</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_PC_Loading_Plots(seurat_object = seurat, dims_plot = 25, file_path = "plots/")

## End(Not run)

</code></pre>

<hr>
<h2 id='Iterate_Plot_Density_Custom'>Iterative Plotting of Gene Lists using Custom Density Plots</h2><span id='topic+Iterate_Plot_Density_Custom'></span>

<h3>Description</h3>

<p>Create and save plots for gene list with single command.  Requires Nebulosa package from Bioconductor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_Plot_Density_Custom(
  seurat_object,
  gene_list,
  viridis_palette = "magma",
  custom_palette = NULL,
  pt.size = 1,
  file_path = NULL,
  file_name = NULL,
  file_type = NULL,
  single_pdf = FALSE,
  dpi = 600,
  reduction = NULL,
  combine = TRUE,
  joint = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_gene_list">gene_list</code></td>
<td>
<p>vector of genes to plot.  If a named vector is provided then the names for each gene
will be incorporated into plot title if <code>single_pdf = TRUE</code> or into file name if <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_viridis_palette">viridis_palette</code></td>
<td>
<p>color scheme to use.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_custom_palette">custom_palette</code></td>
<td>
<p>color for non-expressed cells.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_file_name">file_name</code></td>
<td>
<p>name suffix and file extension.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_file_type">file_type</code></td>
<td>
<p>File type to save output as.  Must be one of following: &quot;.pdf&quot;, &quot;.png&quot;, &quot;.tiff&quot;, &quot;.jpeg&quot;, or &quot;.svg&quot;.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_single_pdf">single_pdf</code></td>
<td>
<p>saves all plots to single PDF file (default = FALSE).  'file_type&ldquo; must be .pdf.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_dpi">dpi</code></td>
<td>
<p>dpi for image saving.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default)</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_combine">combine</code></td>
<td>
<p>Create a single plot? If FALSE, a list with ggplot objects is returned.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_joint">joint</code></td>
<td>
<p>NULL.  This function only supports <code>joint = FALSE</code>.  Leave as NULL to generate plots.  To iterate joint plots see function: <code>Iterate_Plot_Density_Joint</code>.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Custom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Nebulosa.html#topic+plot_density">plot_density</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saved plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_Plot_Density_Custom(seurat_object = object, gene_list = DEG_list, viridis_palette = "magma",
file_path = "plots/", file_name = "_density_plots", file_type = ".jpg", dpi = 600)

## End(Not run)

</code></pre>

<hr>
<h2 id='Iterate_Plot_Density_Joint'>Iterative Plotting of Gene Lists using Custom Joint Density Plots</h2><span id='topic+Iterate_Plot_Density_Joint'></span>

<h3>Description</h3>

<p>Create and save plots for gene list with single command.  Requires Nebulosa package from Bioconductor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_Plot_Density_Joint(
  seurat_object,
  gene_list,
  viridis_palette = "magma",
  custom_palette = NULL,
  pt.size = 1,
  file_path = NULL,
  file_name = NULL,
  file_type = NULL,
  single_pdf = FALSE,
  dpi = 600,
  reduction = NULL,
  combine = TRUE,
  joint = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_gene_list">gene_list</code></td>
<td>
<p>a list of vectors of genes to plot jointly.  Each entry in the list will be plotted
for the joint density.  All entries in list must be greater than 2 features.  If a named list is provided
then the names for each list entry will be incorporated into plot title if <code>single_pdf = TRUE</code> or
into file name if <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_viridis_palette">viridis_palette</code></td>
<td>
<p>color scheme to use.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_custom_palette">custom_palette</code></td>
<td>
<p>color for non-expressed cells.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_file_name">file_name</code></td>
<td>
<p>name suffix and file extension.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_file_type">file_type</code></td>
<td>
<p>File type to save output as.  Must be one of following: &quot;.pdf&quot;, &quot;.png&quot;, &quot;.tiff&quot;, &quot;.jpeg&quot;, or &quot;.svg&quot;.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_single_pdf">single_pdf</code></td>
<td>
<p>saves all plots to single PDF file (default = FALSE).  'file_type&ldquo; must be .pdf.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_dpi">dpi</code></td>
<td>
<p>dpi for image saving.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default)</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_combine">combine</code></td>
<td>
<p>Create a single plot? If FALSE, a list with ggplot objects is returned.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_joint">joint</code></td>
<td>
<p>NULL.  This function only supports <code>joint = FALSE</code>.  Leave as NULL to generate plots.  To iterate joint plots see function: <code>Iterate_Plot_Density_Joint</code>.</p>
</td></tr>
<tr><td><code id="Iterate_Plot_Density_Joint_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Nebulosa.html#topic+plot_density">plot_density</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saved plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_Plot_Density_Joint(seurat_object = object, gene_list = DEG_list, viridis_palette = "magma",
file_path = "plots/", file_name = "joint_plots", file_type = ".jpg", dpi = 600)

## End(Not run)

</code></pre>

<hr>
<h2 id='Iterate_VlnPlot_scCustom'>Iterative Plotting of Gene Lists using VlnPlot_scCustom</h2><span id='topic+Iterate_VlnPlot_scCustom'></span>

<h3>Description</h3>

<p>Create and Save plots for Gene list with Single Command
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Iterate_VlnPlot_scCustom(
  seurat_object,
  features,
  gene_list = deprecated(),
  colors_use = NULL,
  pt.size = NULL,
  group.by = NULL,
  split.by = NULL,
  file_path = NULL,
  file_name = NULL,
  file_type = NULL,
  single_pdf = FALSE,
  raster = NULL,
  dpi = 600,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_features">features</code></td>
<td>
<p>vector of features to plot.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_gene_list">gene_list</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> soft-deprecated.  See <code>features</code>.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_colors_use">colors_use</code></td>
<td>
<p>color palette to use for plotting.  By default if number of levels plotted is less than
or equal to 36 it will use &quot;polychrome&quot; and if greater than 36 will use &quot;varibow&quot; with shuffle = TRUE
both from <code>DiscretePalette_scCustomize</code>.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_pt.size">pt.size</code></td>
<td>
<p>point size for plotting.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) plot by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_split.by">split.by</code></td>
<td>
<p>Feature to split plots by (i.e. &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_file_name">file_name</code></td>
<td>
<p>name suffix and file extension.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_file_type">file_type</code></td>
<td>
<p>File type to save output as.  Must be one of following: &quot;.pdf&quot;, &quot;.png&quot;, &quot;.tiff&quot;, &quot;.jpeg&quot;, or &quot;.svg&quot;.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_single_pdf">single_pdf</code></td>
<td>
<p>saves all plots to single PDF file (default = FALSE).  'file_type&ldquo; must be .pdf.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_dpi">dpi</code></td>
<td>
<p>dpi for image saving.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="Iterate_VlnPlot_scCustom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saved plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Iterate_VlnPlot_scCustom(seurat_object = object, gene_list = DEG_list, colors = color_list,
file_path = "plots/", file_name = "_vln", file_type = ".jpg", dpi = 600)

## End(Not run)

</code></pre>

<hr>
<h2 id='JCO_Four'>Four Color Palette (JCO)</h2><span id='topic+JCO_Four'></span>

<h3>Description</h3>

<p>Shortcut to a specific JCO 4 color palette from ggsci package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JCO_Four()
</code></pre>


<h3>Value</h3>

<p>4 color palette from the JCO ggsci palette
</p>


<h3>References</h3>

<p>Selection of colors from the JCO palette from ggsci being called through paletteer.
See ggsci for more info on palettes
<a href="https://CRAN.R-project.org/package=ggsci">https://CRAN.R-project.org/package=ggsci</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cols &lt;- JCO_Four()
PalettePlot(pal= cols)

</code></pre>

<hr>
<h2 id='LIGER_Features'>Extract Features from LIGER Object</h2><span id='topic+LIGER_Features'></span>

<h3>Description</h3>

<p>Extract all unique features from LIGER object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LIGER_Features(liger_object, by_dataset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LIGER_Features_+3A_liger_object">liger_object</code></td>
<td>
<p>LIGER object name.</p>
</td></tr>
<tr><td><code id="LIGER_Features_+3A_by_dataset">by_dataset</code></td>
<td>
<p>logical, whether to return list with vector of features for each dataset in
LIGER object or to return single vector of unique features across all datasets in object
(default is FALSE; return vector of unique features)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector or list depending on <code>by_dataset</code> parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# return single vector of all unique features
all_features &lt;- LIGER_Features(liger_object = object, by_dataset = FALSE)

# return list of vectors containing features from each individual dataset in object
dataset_features &lt;- LIGER_Features(liger_object = object, by_dataset = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='Liger_to_Seurat'>Create a Seurat object containing the data from a liger object <a href="https://lifecycle.r-lib.org/articles/stages.html#soft-deprecated"><img src="../help/figures/lifecycle-soft-deprecated.svg" alt='[Soft-deprecated]' /></a></h2><span id='topic+Liger_to_Seurat'></span>

<h3>Description</h3>

<p>Merges raw.data and scale.data of object, and creates Seurat object with these values along with
tsne.coords, iNMF factorization, and cluster assignments. Supports Seurat V2 and V3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Liger_to_Seurat(
  liger_object,
  nms = names(liger_object@H),
  renormalize = TRUE,
  use.liger.genes = TRUE,
  by.dataset = FALSE,
  keep_meta = TRUE,
  reduction_label = "UMAP",
  seurat_assay = "RNA",
  assay_type = NULL,
  add_barcode_names = FALSE,
  barcode_prefix = TRUE,
  barcode_cell_id_delimiter = "_"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Liger_to_Seurat_+3A_liger_object">liger_object</code></td>
<td>
<p><code>liger</code> object.</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_nms">nms</code></td>
<td>
<p>By default, labels cell names with dataset of origin (this is to account for cells in
different datasets which may have same name). Other names can be passed here as vector, must have
same length as the number of datasets. (default names(H)).</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_renormalize">renormalize</code></td>
<td>
<p>Whether to log-normalize raw data using Seurat defaults (default TRUE).</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_use.liger.genes">use.liger.genes</code></td>
<td>
<p>Whether to carry over variable genes (default TRUE).</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_by.dataset">by.dataset</code></td>
<td>
<p>Include dataset of origin in cluster identity in Seurat object (default FALSE).</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_keep_meta">keep_meta</code></td>
<td>
<p>logical. Whether to transfer additional metadata (nGene/nUMI/dataset already transferred)
to new Seurat Object.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_reduction_label">reduction_label</code></td>
<td>
<p>Name of dimensionality reduction technique used.  Enables accurate transfer
or name to Seurat object instead of defaulting to &quot;tSNE&quot;.</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_seurat_assay">seurat_assay</code></td>
<td>
<p>Name to set for assay in Seurat Object.  Default is &quot;RNA&quot;.</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_assay_type">assay_type</code></td>
<td>
<p>what type of Seurat assay to create in new object (Assay vs Assay5).
Default is NULL which will default to the current user settings.
See <code><a href="#topic+Convert_Assay">Convert_Assay</a></code> parameter <code>convert_to</code> for acceptable values.</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_add_barcode_names">add_barcode_names</code></td>
<td>
<p>logical, whether to add dataset names to the cell barcodes when
creating Seurat object, default is FALSE.</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_barcode_prefix">barcode_prefix</code></td>
<td>
<p>logical, if <code>add_barcode_names = TRUE</code> should the names be added as
prefix to current cell barcodes/names or a suffix (default is TRUE; prefix).</p>
</td></tr>
<tr><td><code id="Liger_to_Seurat_+3A_barcode_cell_id_delimiter">barcode_cell_id_delimiter</code></td>
<td>
<p>The delimiter to use when adding dataset id to barcode
prefix/suffix.  Default is &quot;_&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Stores original dataset identity by default in new object metadata if dataset names are passed
in nms. iNMF factorization is stored in dim.reduction object with key &quot;iNMF&quot;.
</p>


<h3>Value</h3>

<p>Seurat object with raw.data, scale.data, reduction_label, iNMF, and ident slots set.
</p>


<h3>References</h3>

<p>Original function is part of LIGER package <a href="https://github.com/welch-lab/liger">https://github.com/welch-lab/liger</a> (Licence: GPL-3).
Function was slightly modified for use in scCustomize with keep.meta parameter.  Also posted as
PR to liger GitHub.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
seurat_object &lt;- Liger_to_Seurat(liger_object = LIGER_OBJ, reduction_label = "UMAP")

## End(Not run)
</code></pre>

<hr>
<h2 id='MAD_Stats'>Median Absolute Deviation Statistics</h2><span id='topic+MAD_Stats'></span>

<h3>Description</h3>

<p>Get quick values for X x median absolute deviation for Genes, UMIs, %mito per cell grouped by meta.data variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAD_Stats(
  seurat_object,
  group_by_var = "orig.ident",
  default_var = TRUE,
  mad_var = NULL,
  mad_num = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAD_Stats_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="MAD_Stats_+3A_group_by_var">group_by_var</code></td>
<td>
<p>Column in meta.data slot to group results by (default = &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="MAD_Stats_+3A_default_var">default_var</code></td>
<td>
<p>logical.  Whether to include the default meta.data variables of: &quot;nCount_RNA&quot;,
&quot;nFeature_RNA&quot;, &quot;percent_mito&quot;, &quot;percent_ribo&quot;, &quot;percent_mito_ribo&quot;, and &quot;log10GenesPerUMI&quot;
in addition to variables supplied to <code>mad_var</code>.</p>
</td></tr>
<tr><td><code id="MAD_Stats_+3A_mad_var">mad_var</code></td>
<td>
<p>Column(s) in <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> to calculate medians for in addition to defaults.
Must be of <code>class()</code> integer or numeric.</p>
</td></tr>
<tr><td><code id="MAD_Stats_+3A_mad_num">mad_num</code></td>
<td>
<p>integer value to multiply the MAD in returned data.frame (default is 2).
Often helpful when calculating a outlier range to base of of median + (X*MAD).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mad_stats &lt;- MAD_Stats(seurat_object = obj, group_by_var = "orig.ident")

## End(Not run)

</code></pre>

<hr>
<h2 id='Median_Stats'>Median Statistics</h2><span id='topic+Median_Stats'></span>

<h3>Description</h3>

<p>Get quick values for median Genes, UMIs, %mito per cell grouped by meta.data variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Median_Stats(
  seurat_object,
  group_by_var = "orig.ident",
  default_var = TRUE,
  median_var = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Median_Stats_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Median_Stats_+3A_group_by_var">group_by_var</code></td>
<td>
<p>Column in meta.data slot to group results by (default = &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="Median_Stats_+3A_default_var">default_var</code></td>
<td>
<p>logical.  Whether to include the default meta.data variables of: &quot;nCount_RNA&quot;,
&quot;nFeature_RNA&quot;, &quot;percent_mito&quot;, &quot;percent_ribo&quot;, &quot;percent_mito_ribo&quot;, and &quot;log10GenesPerUMI&quot;
in addition to variables supplied to <code>median_var</code>.</p>
</td></tr>
<tr><td><code id="Median_Stats_+3A_median_var">median_var</code></td>
<td>
<p>Column(s) in <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> to calculate medians for in addition to defaults.
Must be of <code>class()</code> integer or numeric.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
med_stats &lt;- Median_Stats(seurat_object - obj, group_by_var = "orig.ident")

## End(Not run)

</code></pre>

<hr>
<h2 id='Merge_Seurat_List'>Merge a list of Seurat Objects</h2><span id='topic+Merge_Seurat_List'></span>

<h3>Description</h3>

<p>Enables easy merge of a list of Seurat Objects.  See  See <code><a href="SeuratObject.html#topic+merge">merge</a></code> for more information,
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Merge_Seurat_List(
  list_seurat,
  add.cell.ids = NULL,
  merge.data = TRUE,
  project = "SeuratProject"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Merge_Seurat_List_+3A_list_seurat">list_seurat</code></td>
<td>
<p>list composed of multiple Seurat Objects.</p>
</td></tr>
<tr><td><code id="Merge_Seurat_List_+3A_add.cell.ids">add.cell.ids</code></td>
<td>
<p>A character vector of equal length to the number of objects in <code>list_seurat</code>.
Appends the corresponding values to the start of each objects' cell names.  See <code><a href="SeuratObject.html#topic+merge">merge</a></code>.</p>
</td></tr>
<tr><td><code id="Merge_Seurat_List_+3A_merge.data">merge.data</code></td>
<td>
<p>Merge the data slots instead of just merging the counts (which requires renormalization).
This is recommended if the same normalization approach was applied to all objects.
See <code><a href="SeuratObject.html#topic+merge">merge</a></code>.</p>
</td></tr>
<tr><td><code id="Merge_Seurat_List_+3A_project">project</code></td>
<td>
<p>Project name for the Seurat object. See <code><a href="SeuratObject.html#topic+merge">merge</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat Object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
object_list &lt;- list(obj1, obj2, obj3, ...)
merged_object &lt;- Merge_Seurat_List(list_seurat = object_list)

## End(Not run)

</code></pre>

<hr>
<h2 id='Merge_Sparse_Data_All'>Merge a list of Sparse Matrices</h2><span id='topic+Merge_Sparse_Data_All'></span>

<h3>Description</h3>

<p>Enables easy merge of a list of sparse matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Merge_Sparse_Data_All(
  matrix_list,
  add_cell_ids = NULL,
  prefix = TRUE,
  cell_id_delimiter = "_"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Merge_Sparse_Data_All_+3A_matrix_list">matrix_list</code></td>
<td>
<p>list of matrices to merge.</p>
</td></tr>
<tr><td><code id="Merge_Sparse_Data_All_+3A_add_cell_ids">add_cell_ids</code></td>
<td>
<p>a vector of sample ids to add as prefix to cell barcode during merge.</p>
</td></tr>
<tr><td><code id="Merge_Sparse_Data_All_+3A_prefix">prefix</code></td>
<td>
<p>logical.  Whether <code>add_cell_ids</code> should be added as prefix to current cell barcodes/names
or as suffix to current cell barcodes/names.  Default is TRUE, add as prefix.</p>
</td></tr>
<tr><td><code id="Merge_Sparse_Data_All_+3A_cell_id_delimiter">cell_id_delimiter</code></td>
<td>
<p>The delimiter to use when adding cell id prefix/suffix.  Default is &quot;_&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sparse Matrix
</p>


<h3>References</h3>

<p>Original function is part of LIGER package as non-exported function
<a href="https://github.com/welch-lab/liger/blob/master/R/utilities.R">https://github.com/welch-lab/liger/blob/master/R/utilities.R</a> (License: GPL-3).
Function was modified for use in scCustomize (add progress bar, prefix vs. suffix, and delimiter options).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data_list &lt;- Read10X_GEO(...)
merged &lt;- Merge_Sparse_Data_All(matrix_list = data_list, add_cell_ids = names(data_list),
prefix = TRUE, cell_id_delimiter = "_")

## End(Not run)

</code></pre>

<hr>
<h2 id='Merge_Sparse_Multimodal_All'>Merge a list of Sparse Matrices contain multi-modal data.</h2><span id='topic+Merge_Sparse_Multimodal_All'></span>

<h3>Description</h3>

<p>Enables easy merge of a list of sparse matrices for multi-modal data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Merge_Sparse_Multimodal_All(
  matrix_list,
  add_cell_ids = NULL,
  prefix = TRUE,
  cell_id_delimiter = "_"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Merge_Sparse_Multimodal_All_+3A_matrix_list">matrix_list</code></td>
<td>
<p>list of matrices to merge.</p>
</td></tr>
<tr><td><code id="Merge_Sparse_Multimodal_All_+3A_add_cell_ids">add_cell_ids</code></td>
<td>
<p>a vector of sample ids to add as prefix to cell barcode during merge.</p>
</td></tr>
<tr><td><code id="Merge_Sparse_Multimodal_All_+3A_prefix">prefix</code></td>
<td>
<p>logical.  Whether <code>add_cell_ids</code> should be added as prefix to current cell barcodes/names
or as suffix to current cell barcodes/names.  Default is TRUE, add as prefix.</p>
</td></tr>
<tr><td><code id="Merge_Sparse_Multimodal_All_+3A_cell_id_delimiter">cell_id_delimiter</code></td>
<td>
<p>The delimiter to use when adding cell id prefix/suffix.  Default is &quot;_&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing one sparse matrix for each modality
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data_list &lt;- Read10X_GEO(...)
merged_list &lt;- Merge_Sparse_Multimodal_All(matrix_list = data_list, add_cell_ids = names(data_list),
prefix = TRUE, cell_id_delimiter = "_")

## End(Not run)

</code></pre>

<hr>
<h2 id='Meta_Highlight_Plot'>Meta Highlight Plot</h2><span id='topic+Meta_Highlight_Plot'></span>

<h3>Description</h3>

<p>Create Plot with meta data variable of interest highlighted
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Meta_Highlight_Plot(
  seurat_object,
  meta_data_column,
  meta_data_highlight,
  highlight_color = NULL,
  background_color = "lightgray",
  pt.size = NULL,
  aspect_ratio = NULL,
  figure_plot = FALSE,
  raster = NULL,
  raster.dpi = c(512, 512),
  label = FALSE,
  split.by = NULL,
  split_seurat = FALSE,
  ggplot_default_colors = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Meta_Highlight_Plot_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_meta_data_column">meta_data_column</code></td>
<td>
<p>Name of the column in <code>seurat_object@meta.data</code> slot to pull value from for highlighting.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_meta_data_highlight">meta_data_highlight</code></td>
<td>
<p>Name of variable(s) within <code>meta_data_name</code> to highlight in the plot.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_highlight_color">highlight_color</code></td>
<td>
<p>Color to highlight cells (default &quot;navy&quot;).</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_background_color">background_color</code></td>
<td>
<p>non-highlighted cell colors.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_pt.size">pt.size</code></td>
<td>
<p>point size for both highlighted cluster and background.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_figure_plot">figure_plot</code></td>
<td>
<p>logical.  Whether to remove the axes and plot with legend on left of plot denoting
axes labels.  (Default is FALSE).  Requires <code>split_seurat = TRUE</code>.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_label">label</code></td>
<td>
<p>Whether to label the highlighted meta data variable(s).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_split.by">split.by</code></td>
<td>
<p>Variable in <code style="white-space: pre;">&#8288;@meta.data&#8288;</code> to split the plot by.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_split_seurat">split_seurat</code></td>
<td>
<p>logical.  Whether or not to display split plots like Seurat (shared y axis) or as
individual plots in layout.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="Meta_Highlight_Plot_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to<code><a href="Seurat.html#topic+DimPlot">DimPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
pbmc_small$sample_id &lt;- sample(c("sample1", "sample2"), size = ncol(pbmc_small), replace = TRUE)

Meta_Highlight_Plot(seurat_object = pbmc_small, meta_data_column = "sample_id",
meta_data_highlight = "sample1", highlight_color = "gold", background_color = "lightgray",
pt.size = 2)

</code></pre>

<hr>
<h2 id='Meta_Numeric'>Check if meta data columns are numeric</h2><span id='topic+Meta_Numeric'></span>

<h3>Description</h3>

<p>Check if any present meta data columns are numeric and returns vector of valid numeric columns.
Issues warning message if any columns not in numeric form.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Meta_Numeric(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Meta_Numeric_+3A_data">data</code></td>
<td>
<p>a data.frame contain meta.data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of meta data columns that are numeric.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
numeric_meta_columns &lt;- Meta_Numeric(data = meta_data)

## End(Not run)

</code></pre>

<hr>
<h2 id='Meta_Present'>Check if meta data are present</h2><span id='topic+Meta_Present'></span>

<h3>Description</h3>

<p>Check if meta data columns are present in object and return vector of found columns
Return warning messages for meta data columns not found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Meta_Present(
  object,
  seurat_object = deprecated(),
  meta_col_names,
  print_msg = TRUE,
  omit_warn = TRUE,
  return_none = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Meta_Present_+3A_object">object</code></td>
<td>
<p>Seurat or Liger object name.</p>
</td></tr>
<tr><td><code id="Meta_Present_+3A_seurat_object">seurat_object</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> deprecated.  Please use <code>object</code> instead.</p>
</td></tr>
<tr><td><code id="Meta_Present_+3A_meta_col_names">meta_col_names</code></td>
<td>
<p>vector of column names to check.</p>
</td></tr>
<tr><td><code id="Meta_Present_+3A_print_msg">print_msg</code></td>
<td>
<p>logical. Whether message should be printed if all features are found.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Meta_Present_+3A_omit_warn">omit_warn</code></td>
<td>
<p>logical. Whether to print message about features that are not found in current object. Default is TRUE.</p>
</td></tr>
<tr><td><code id="Meta_Present_+3A_return_none">return_none</code></td>
<td>
<p>logical. Whether list of found vs. bad features should still be returned if no
<code>meta_col_names</code> are found.  Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of meta data columns that are present
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
meta_variables &lt;- Meta_Present(object = obj_name, meta_col_names = "percent_mito", print_msg = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='Meta_Remove_Seurat'>Remove meta data columns containing Seurat Defaults</h2><span id='topic+Meta_Remove_Seurat'></span>

<h3>Description</h3>

<p>Remove any columns from new meta_data data.frame in preparation for adding back to Seurat Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Meta_Remove_Seurat(
  meta_data,
  seurat_object,
  barcodes_to_rownames = FALSE,
  barcodes_colname = "barcodes"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Meta_Remove_Seurat_+3A_meta_data">meta_data</code></td>
<td>
<p>data.frame containing meta data.</p>
</td></tr>
<tr><td><code id="Meta_Remove_Seurat_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Meta_Remove_Seurat_+3A_barcodes_to_rownames">barcodes_to_rownames</code></td>
<td>
<p>logical, are barcodes present as column and should they be moved to
rownames (to be compatible with <code>Seurat::AddMetaData</code>).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Meta_Remove_Seurat_+3A_barcodes_colname">barcodes_colname</code></td>
<td>
<p>name of barcodes column in meta_data.  Required if <code>barcodes_to_rownames = TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with only new columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
new_meta &lt;- Meta_Remove_Seurat(meta_data = meta_data_df, seurat_object = object)
object &lt;- AddMetaData(object = object, metadata = new_meta)

## End(Not run)

</code></pre>

<hr>
<h2 id='Move_Legend'>Move Legend Position</h2><span id='topic+Move_Legend'></span>

<h3>Description</h3>

<p>Shortcut for thematic modification to move legend position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Move_Legend(position = "right", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Move_Legend_+3A_position">position</code></td>
<td>
<p>valid position to move legend.  Default is &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="Move_Legend_+3A_...">...</code></td>
<td>
<p>extra arguments passed to <code>ggplot2::theme()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list-like object of class <em>theme</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a plot and customize theme
library(ggplot2)
df &lt;- data.frame(x = rnorm(n = 100, mean = 20, sd = 2), y = rbinom(n = 100, size = 100, prob = 0.2))
p &lt;- ggplot(data = df, mapping = aes(x = x, y = y)) + geom_point(mapping = aes(color = 'red'))
p + Move_Legend("left")

</code></pre>

<hr>
<h2 id='msigdb_qc_gene_list'>QC Gene Lists</h2><span id='topic+msigdb_qc_gene_list'></span>

<h3>Description</h3>

<p>Gene symbols for qc percentages from MSigDB database.  The gene sets are from 3 MSigDB lists:
&quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;, &quot;HALLMARK_APOPTOSIS&quot;, and &quot;HALLMARK_DNA_REPAIR&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msigdb_qc_gene_list
</code></pre>


<h3>Format</h3>

<p>A list of 18 vectors
</p>

<dl>
<dt>Homo_sapiens_msigdb_oxphos</dt><dd><p>Genes in msigdb &quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot; list for human</p>
</dd>
<dt>Homo_sapiens_msigdb_apop</dt><dd><p>Genes in msigdb &quot;HALLMARK_APOPTOSIS&quot; list for human</p>
</dd>
<dt>Homo_sapiens_msigdb_dna_repair</dt><dd><p>Genes in msigdb &quot;HALLMARK_DNA_REPAIR&quot; list for human</p>
</dd>
<dt>Mus_musculus_msigdb_oxphos</dt><dd><p>Genes in msigdb &quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot; list for mouse</p>
</dd>
<dt>Mus_musculus_msigdb_apop</dt><dd><p>Genes in msigdb &quot;HALLMARK_APOPTOSIS&quot; list for mouse</p>
</dd>
<dt>Mus_musculus_msigdb_dna_repair</dt><dd><p>Genes in msigdb &quot;HALLMARK_DNA_REPAIR&quot; list for mouse</p>
</dd>
<dt>Rattus_norvegicus_msigdb_oxphos</dt><dd><p>Genes in msigdb &quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot; list for rat</p>
</dd>
<dt>Rattus_norvegicus_msigdb_apop</dt><dd><p>Genes in msigdb &quot;HALLMARK_APOPTOSIS&quot; list for rat</p>
</dd>
<dt>Rattus_norvegicus_msigdb_dna_repair</dt><dd><p>Genes in msigdb &quot;HALLMARK_DNA_REPAIR&quot; list for rat</p>
</dd>
<dt>Drosophila_melanogaster_msigdb_oxphos</dt><dd><p>Genes in msigdb &quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot; list for fly</p>
</dd>
<dt>Drosophila_melanogaster_msigdb_apop</dt><dd><p>Genes in msigdb &quot;HALLMARK_APOPTOSIS&quot; list for fly</p>
</dd>
<dt>Drosophila_melanogaster_msigdb_dna_repair</dt><dd><p>Genes in msigdb &quot;HALLMARK_DNA_REPAIR&quot; list for fly</p>
</dd>
<dt>Dario_rerio_msigdb_oxphos</dt><dd><p>Genes in msigdb &quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot; list for zebrafish</p>
</dd>
<dt>Dario_rerio_msigdb_apop</dt><dd><p>Genes in msigdb &quot;HALLMARK_APOPTOSIS&quot; list for zebrafish</p>
</dd>
<dt>Dario_rerio_msigdb_dna_repair</dt><dd><p>Genes in msigdb &quot;HALLMARK_DNA_REPAIR&quot; list for zebrafish</p>
</dd>
<dt>Macaca_mulatta_msigdb_oxphos</dt><dd><p>Genes in msigdb &quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot; list for macaque</p>
</dd>
<dt>Macaca_mulatta_msigdb_apop</dt><dd><p>Genes in msigdb &quot;HALLMARK_APOPTOSIS&quot; list for macaque</p>
</dd>
<dt>Macaca_mulatta_msigdb_dna_repair</dt><dd><p>Genes in msigdb &quot;HALLMARK_DNA_REPAIR&quot; list for macaque</p>
</dd>
</dl>



<h3>Source</h3>

<p>MSigDB gene sets via msigdbr package <a href="https://cran.r-project.org/package=msigdbr">https://cran.r-project.org/package=msigdbr</a>
</p>

<hr>
<h2 id='NavyAndOrange'>Navy and Orange Dual Color Palette</h2><span id='topic+NavyAndOrange'></span>

<h3>Description</h3>

<p>Shortcut to navy orange color plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NavyAndOrange(flip_order = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NavyAndOrange_+3A_flip_order">flip_order</code></td>
<td>
<p>Whether to flip the order of colors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Navy orange palette
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cols &lt;- NavyAndOrange()
PalettePlot(pal= cols)

</code></pre>

<hr>
<h2 id='PalettePlot'>Plot color palette in viewer</h2><span id='topic+PalettePlot'></span>

<h3>Description</h3>

<p>Plots given color vector/palette in viewer to evaluate palette before plotting on data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PalettePlot(pal = NULL, label_color_num = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PalettePlot_+3A_pal">pal</code></td>
<td>
<p>a vector of colors (either named colors of hex codes).</p>
</td></tr>
<tr><td><code id="PalettePlot_+3A_label_color_num">label_color_num</code></td>
<td>
<p>logical, whether or not to numerically label the colors in output plot.
Default is TRUE is number of colors in <code>pal</code> is less than 75 and FALSE is greater than 75.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot of all colors in supplied palette/vector
</p>


<h3>References</h3>

<p>Adapted from colorway package <code>build_palette</code> internals (License: GPL-3).
<a href="https://github.com/hypercompetent/colorway">https://github.com/hypercompetent/colorway</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pal &lt;- DiscretePalette_scCustomize(num_colors = 36, palette = "varibow")
PalettePlot(pal = pal)

</code></pre>

<hr>
<h2 id='PC_Plotting'>PC Plots</h2><span id='topic+PC_Plotting'></span>

<h3>Description</h3>

<p>Plot PC Heatmaps and Dim Loadings for exploratory analysis.  Plots a single Heatmap and Gene Loading Plot.
Used for PC_Loading_Plots function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PC_Plotting(seurat_object, dim_number)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PC_Plotting_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat Object.</p>
</td></tr>
<tr><td><code id="PC_Plotting_+3A_dim_number">dim_number</code></td>
<td>
<p>A single dim to plot (integer).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of PC heatmap and gene loadings for single
</p>


<h3>See Also</h3>

<p><code><a href="Seurat.html#topic+PCHeatmap">PCHeatmap</a></code> and <code><a href="Seurat.html#topic+VizDimLoadings">VizDimLoadings</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
PC_Plotting(seurat_object = pbmc_small, dim_number = 1)

</code></pre>

<hr>
<h2 id='Percent_Expressing'>Calculate percent of expressing cells</h2><span id='topic+Percent_Expressing'></span>

<h3>Description</h3>

<p>Calculates the percent of cells that express a given set of features by various grouping factors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Percent_Expressing(
  seurat_object,
  features,
  threshold = 0,
  group_by = NULL,
  split_by = NULL,
  entire_object = FALSE,
  slot = deprecated(),
  layer = "data",
  assay = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Percent_Expressing_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Percent_Expressing_+3A_features">features</code></td>
<td>
<p>Feature(s) to plot.</p>
</td></tr>
<tr><td><code id="Percent_Expressing_+3A_threshold">threshold</code></td>
<td>
<p>Expression threshold to use for calculation of percent expressing (default is 0).</p>
</td></tr>
<tr><td><code id="Percent_Expressing_+3A_group_by">group_by</code></td>
<td>
<p>Factor to group the cells by.</p>
</td></tr>
<tr><td><code id="Percent_Expressing_+3A_split_by">split_by</code></td>
<td>
<p>Factor to split the groups by.</p>
</td></tr>
<tr><td><code id="Percent_Expressing_+3A_entire_object">entire_object</code></td>
<td>
<p>logical (default = FALSE).  Whether to calculate percent of expressing cells
across the entire object as opposed to by cluster or by <code>group_by</code> variable.</p>
</td></tr>
<tr><td><code id="Percent_Expressing_+3A_slot">slot</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> soft-deprecated.  See <code>layer</code></p>
</td></tr>
<tr><td><code id="Percent_Expressing_+3A_layer">layer</code></td>
<td>
<p>Which layer to pull expression data from?  Default is &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="Percent_Expressing_+3A_assay">assay</code></td>
<td>
<p>Assay to pull feature data from.  Default is active assay.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>References</h3>

<p>Part of code is modified from Seurat package as used by <code><a href="Seurat.html#topic+DotPlot">DotPlot</a></code>
to generate values to use for plotting.  Source code can be found here:
<a href="https://github.com/satijalab/seurat/blob/4e868fcde49dc0a3df47f94f5fb54a421bfdf7bc/R/visualization.R#L3391">https://github.com/satijalab/seurat/blob/4e868fcde49dc0a3df47f94f5fb54a421bfdf7bc/R/visualization.R#L3391</a> (License: GPL-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
percent_stats &lt;- Percent_Expressing(seurat_object = object, features = "Cx3cr1", threshold = 0)

## End(Not run)

</code></pre>

<hr>
<h2 id='Plot_Cells_per_Sample'>Plot Number of Cells/Nuclei per Sample</h2><span id='topic+Plot_Cells_per_Sample'></span>

<h3>Description</h3>

<p>Plot of total cell or nuclei number per sample grouped by another meta data variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_Cells_per_Sample(
  seurat_object,
  sample_col = "orig.ident",
  group_by = NULL,
  colors_use = NULL,
  dot_size = 1,
  plot_title = "Cells/Nuclei per Sample",
  y_axis_label = "Number of Cells",
  x_axis_label = NULL,
  legend_title = NULL,
  x_lab_rotate = TRUE,
  color_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_Cells_per_Sample_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_sample_col">sample_col</code></td>
<td>
<p>Specify which column in meta.data specifies sample ID (i.e. orig.ident).</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_group_by">group_by</code></td>
<td>
<p>Column in meta.data slot to group results by (i.e. &quot;Treatment&quot;).</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_colors_use">colors_use</code></td>
<td>
<p>List of colors or color palette to use.</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>group_by</code> is not NULL.  Default is 1.</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot title.</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_legend_title">legend_title</code></td>
<td>
<p>Label for plot legend.</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Plot_Cells_per_Sample_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Plot_Cells_per_Sample(seurat_object = obj, sample_col = "orig.ident", group_by = "Treatment")

## End(Not run)

</code></pre>

<hr>
<h2 id='Plot_Density_Custom'>Nebulosa Density Plot</h2><span id='topic+Plot_Density_Custom'></span>

<h3>Description</h3>

<p>Allow for customization of Nebulosa plot_density.  Requires Nebulosa package from Bioconductor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_Density_Custom(
  seurat_object,
  features,
  joint = FALSE,
  viridis_palette = "magma",
  custom_palette = NULL,
  pt.size = 1,
  aspect_ratio = NULL,
  reduction = NULL,
  combine = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_Density_Custom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_features">features</code></td>
<td>
<p>Features to plot.</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_joint">joint</code></td>
<td>
<p>logical. Whether to return joint density plot. Default is FALSE.</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_viridis_palette">viridis_palette</code></td>
<td>
<p>default viridis palette to use (must be one of: &quot;viridis&quot;, &quot;magma&quot;, &quot;cividis&quot;,
&quot;inferno&quot;, &quot;plasma&quot;).  Default is &quot;magma&quot;.</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_custom_palette">custom_palette</code></td>
<td>
<p>non-default color palette to be used in place of default viridis options.</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_combine">combine</code></td>
<td>
<p>Create a single plot? If FALSE, a list with ggplot objects is returned.</p>
</td></tr>
<tr><td><code id="Plot_Density_Custom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Nebulosa.html#topic+plot_density">plot_density</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(Seurat)
Plot_Density_Custom(seurat_object = pbmc_small, features = "CD3E")

## End(Not run)

</code></pre>

<hr>
<h2 id='Plot_Density_Joint_Only'>Nebulosa Joint Density Plot</h2><span id='topic+Plot_Density_Joint_Only'></span>

<h3>Description</h3>

<p>Return only the joint density plot from Nebulosa plot_density function.  Requires Nebulosa package from Bioconductor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_Density_Joint_Only(
  seurat_object,
  features,
  viridis_palette = "magma",
  custom_palette = NULL,
  pt.size = 1,
  aspect_ratio = NULL,
  reduction = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_Density_Joint_Only_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Plot_Density_Joint_Only_+3A_features">features</code></td>
<td>
<p>Features to plot.</p>
</td></tr>
<tr><td><code id="Plot_Density_Joint_Only_+3A_viridis_palette">viridis_palette</code></td>
<td>
<p>default viridis palette to use (must be one of: &quot;viridis&quot;, &quot;magma&quot;, &quot;cividis&quot;,
&quot;inferno&quot;, &quot;plasma&quot;).  Default is &quot;magma&quot;.</p>
</td></tr>
<tr><td><code id="Plot_Density_Joint_Only_+3A_custom_palette">custom_palette</code></td>
<td>
<p>non-default color palette to be used in place of default viridis options.</p>
</td></tr>
<tr><td><code id="Plot_Density_Joint_Only_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="Plot_Density_Joint_Only_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Control the aspect ratio (y:x axes ratio length).  Must be numeric value;
Default is NULL.</p>
</td></tr>
<tr><td><code id="Plot_Density_Joint_Only_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality Reduction to use (if NULL then defaults to Object default).</p>
</td></tr>
<tr><td><code id="Plot_Density_Joint_Only_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Nebulosa.html#topic+plot_density">plot_density</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(Seurat)
Plot_Density_Joint_Only(seurat_object = pbmc_small, features = c("CD8A", "CD3E"))

## End(Not run)

</code></pre>

<hr>
<h2 id='Plot_Median_Genes'>Plot Median Genes per Cell per Sample</h2><span id='topic+Plot_Median_Genes'></span>

<h3>Description</h3>

<p>Plot of median genes per cell per sample grouped by desired meta data variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_Median_Genes(
  seurat_object,
  sample_col = "orig.ident",
  group_by = NULL,
  colors_use = NULL,
  dot_size = 1,
  plot_title = "Median Genes/Cell per Sample",
  y_axis_label = "Median Genes",
  x_axis_label = NULL,
  legend_title = NULL,
  x_lab_rotate = TRUE,
  color_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_Median_Genes_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_sample_col">sample_col</code></td>
<td>
<p>Specify which column in meta.data specifies sample ID (i.e. orig.ident).</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_group_by">group_by</code></td>
<td>
<p>Column in meta.data slot to group results by (i.e. &quot;Treatment&quot;).</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_colors_use">colors_use</code></td>
<td>
<p>List of colors or color palette to use.  Only applicable if <code>group_by</code> is not NULL.</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>group_by</code> is not NULL.  Default is 1.</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot title.</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_legend_title">legend_title</code></td>
<td>
<p>Label for plot legend.</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Plot_Median_Genes_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
# Create example groups
pbmc_small$sample_id &lt;- sample(c("sample1", "sample2"), size = ncol(pbmc_small), replace = TRUE)

# Plot
Plot_Median_Genes(seurat_object = pbmc_small, sample_col = "orig.ident",  group_by = "sample_id")

</code></pre>

<hr>
<h2 id='Plot_Median_Mito'>Plot Median Percent Mito per Cell per Sample</h2><span id='topic+Plot_Median_Mito'></span>

<h3>Description</h3>

<p>Plot of median percent mito per cell per sample grouped by desired meta data variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_Median_Mito(
  seurat_object,
  sample_col = "orig.ident",
  group_by = NULL,
  colors_use = NULL,
  dot_size = 1,
  plot_title = "Median % Mito per Sample",
  y_axis_label = "Percent Mitochondrial Reads",
  x_axis_label = NULL,
  legend_title = NULL,
  x_lab_rotate = TRUE,
  color_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_Median_Mito_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_sample_col">sample_col</code></td>
<td>
<p>Specify which column in meta.data specifies sample ID (i.e. orig.ident).</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_group_by">group_by</code></td>
<td>
<p>Column in meta.data slot to group results by (i.e. &quot;Treatment&quot;).</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_colors_use">colors_use</code></td>
<td>
<p>List of colors or color palette to use.  Only applicable if <code>group_by</code> is not NULL.</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>group_by</code> is not NULL.  Default is 1.</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot title.</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_legend_title">legend_title</code></td>
<td>
<p>Label for plot legend.</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Plot_Median_Mito_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Add mito
obj &lt;- Add_Mito_Ribo_Seurat(seurat_object = obj, species = "human")

# Plot
Plot_Median_Mito(seurat_object = obj, sample_col = "orig.ident",  group_by = "sample_id")

## End(Not run)

</code></pre>

<hr>
<h2 id='Plot_Median_Other'>Plot Median other variable per Cell per Sample</h2><span id='topic+Plot_Median_Other'></span>

<h3>Description</h3>

<p>Plot of median other variable per cell per sample grouped by desired meta data variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_Median_Other(
  seurat_object,
  median_var,
  sample_col = "orig.ident",
  group_by = NULL,
  colors_use = NULL,
  dot_size = 1,
  plot_title = NULL,
  y_axis_label = NULL,
  x_axis_label = NULL,
  legend_title = NULL,
  x_lab_rotate = TRUE,
  color_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_Median_Other_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_median_var">median_var</code></td>
<td>
<p>Variable in meta.data slot to calculate and plot median values for.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_sample_col">sample_col</code></td>
<td>
<p>Specify which column in meta.data specifies sample ID (i.e. orig.ident).</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_group_by">group_by</code></td>
<td>
<p>Column in meta.data slot to group results by (i.e. &quot;Treatment&quot;).</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_colors_use">colors_use</code></td>
<td>
<p>List of colors or color palette to use.  Only applicable if <code>group_by</code> is not NULL.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>group_by</code> is not NULL.  Default is 1.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot title.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_legend_title">legend_title</code></td>
<td>
<p>Label for plot legend.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Plot_Median_Other_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(Seurat)
cd_features &lt;- list(c('CD79B', 'CD79A', 'CD19', 'CD180', 'CD200', 'CD3D', 'CD2','CD3E',
'CD7','CD8A', 'CD14', 'CD1C', 'CD68', 'CD9', 'CD247'))

pbmc_small &lt;- AddModuleScore(object = pbmc_small, features = cd_features, ctrl = 5,
name = 'CD_Features')

Plot_Median_Other(seurat_object = pbmc_small, median_var = "CD_Features1",
sample_col = "orig.ident", group_by = "Treatment")

## End(Not run)

</code></pre>

<hr>
<h2 id='Plot_Median_UMIs'>Plot Median UMIs per Cell per Sample</h2><span id='topic+Plot_Median_UMIs'></span>

<h3>Description</h3>

<p>Plot of median UMIs per cell per sample grouped by desired meta data variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot_Median_UMIs(
  seurat_object,
  sample_col = "orig.ident",
  group_by = NULL,
  colors_use = NULL,
  dot_size = 1,
  plot_title = "Median UMIs/Cell per Sample",
  y_axis_label = "Median UMIs",
  x_axis_label = NULL,
  legend_title = NULL,
  x_lab_rotate = TRUE,
  color_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot_Median_UMIs_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_sample_col">sample_col</code></td>
<td>
<p>Specify which column in meta.data specifies sample ID (i.e. orig.ident).</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_group_by">group_by</code></td>
<td>
<p>Column in meta.data slot to group results by (i.e. &quot;Treatment&quot;).</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_colors_use">colors_use</code></td>
<td>
<p>List of colors or color palette to use.  Only applicable if <code>group_by</code> is not NULL.</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>group_by</code> is not NULL.  Default is 1.</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot title.</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_legend_title">legend_title</code></td>
<td>
<p>Label for plot legend.</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Plot_Median_UMIs_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
# Create example groups
pbmc_small$sample_id &lt;- sample(c("sample1", "sample2"), size = ncol(pbmc_small), replace = TRUE)

# Plot
Plot_Median_UMIs(seurat_object = pbmc_small, sample_col = "orig.ident",  group_by = "sample_id")

</code></pre>

<hr>
<h2 id='plotFactors_scCustom'>Customized version of plotFactors</h2><span id='topic+plotFactors_scCustom'></span>

<h3>Description</h3>

<p>Modified and optimized version of <code>plotFactors</code> function from LIGER package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotFactors_scCustom(
  liger_object,
  num_genes = 8,
  colors_use_factors = NULL,
  colors_use_dimreduc = c("lemonchiffon", "red"),
  pt.size_factors = 1,
  pt.size_dimreduc = 1,
  reduction_label = "UMAP",
  plot_legend = TRUE,
  raster = TRUE,
  raster.dpi = c(512, 512),
  order = FALSE,
  plot_dimreduc = TRUE,
  save_plots = TRUE,
  file_path = NULL,
  file_name = NULL,
  return_plots = FALSE,
  cells.highlight = NULL,
  reorder_datasets = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotFactors_scCustom_+3A_liger_object">liger_object</code></td>
<td>
<p><code>liger</code> liger_object.  Need to perform clustering and factorization before calling this function</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_num_genes">num_genes</code></td>
<td>
<p>Number of genes to display for each factor (Default 8).</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_colors_use_factors">colors_use_factors</code></td>
<td>
<p>colors to use for plotting factor loadings  By default datasets will be
plotted using &quot;varibow&quot; with shuffle = TRUE from both from <code><a href="#topic+DiscretePalette_scCustomize">DiscretePalette_scCustomize</a></code>.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_colors_use_dimreduc">colors_use_dimreduc</code></td>
<td>
<p>colors to use for plotting factor loadings on dimensionality reduction
coordinates (tSNE/UMAP).  Default is c('lemonchiffon', 'red'),</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_pt.size_factors">pt.size_factors</code></td>
<td>
<p>Adjust point size for plotting in the factor plots.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_pt.size_dimreduc">pt.size_dimreduc</code></td>
<td>
<p>Adjust point size for plotting in dimensionality reduction plots.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_reduction_label">reduction_label</code></td>
<td>
<p>What to label the x and y axes of resulting plots.  LIGER does not store name of
technique and therefore needs to be set manually.  Default is &quot;UMAP&quot;.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_plot_legend">plot_legend</code></td>
<td>
<p>logical, whether to plot the legend on factor loading plots, default is TRUE.
Helpful if number of datasets is large to avoid crowding the plot with legend.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 200,000 cells.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_order">order</code></td>
<td>
<p>logical. Whether to plot higher loading cells on top of cells with lower loading values in the
dimensionality reduction plots (Default = FALSE).</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_plot_dimreduc">plot_dimreduc</code></td>
<td>
<p>logical.  Whether to plot factor loadings on dimensionality reduction coordinates.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_save_plots">save_plots</code></td>
<td>
<p>logical.  Whether to save plots.  Default is TRUE</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_file_path">file_path</code></td>
<td>
<p>directory file path and/or file name prefix.  Defaults to current wd.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_file_name">file_name</code></td>
<td>
<p>name suffix to append after sample name.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_return_plots">return_plots</code></td>
<td>
<p>logical. Whether or not to return plots to the environment.  (Default is FALSE)</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_cells.highlight">cells.highlight</code></td>
<td>
<p>Names of specific cells to highlight in plot (black) (default NULL).</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_reorder_datasets">reorder_datasets</code></td>
<td>
<p>New order to plot datasets in for the factor plots if different from current
factor level order in cell.data slot.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use_factors = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;varibow&quot; palette.</p>
</td></tr>
<tr><td><code id="plotFactors_scCustom_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the palette shuffle if <code>colors_use_factors = NULL</code>.  Default = 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of ggplot/patchwork objects and/or PDF file.
</p>


<h3>Author(s)</h3>

<p>Velina Kozareva (Original code for modified function), Sam Marsh (Added/modified functionality)
</p>


<h3>References</h3>

<p>Based on <code>plotFactors</code> functionality from original LIGER package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plotFactors_scCustom(liger_object = liger_obj, return_plots = FALSE, plot_dimreduc = TRUE,
raster = FALSE, save_plots = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='Pull_Cluster_Annotation'>Pull cluster information from annotation csv file.</h2><span id='topic+Pull_Cluster_Annotation'></span>

<h3>Description</h3>

<p>shortcut filter and pull function compatible with annotation files created by <code>Create_Cluster_Annotation_File</code>
by default but also any other csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pull_Cluster_Annotation(
  annotation = NULL,
  cluster_name_col = "cluster",
  cell_type_col = "cell_type"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Pull_Cluster_Annotation_+3A_annotation">annotation</code></td>
<td>
<p>name of the data.frame/tibble or path to CSV file containing cluster annotation.</p>
</td></tr>
<tr><td><code id="Pull_Cluster_Annotation_+3A_cluster_name_col">cluster_name_col</code></td>
<td>
<p>name of column containing cluster names/numbers (default is &quot;cluster&quot;).</p>
</td></tr>
<tr><td><code id="Pull_Cluster_Annotation_+3A_cell_type_col">cell_type_col</code></td>
<td>
<p>name of column contain the cell type annotation (default is &quot;cell_type&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of named vectors for every cell type in the <code>cell_type_col</code> column of the annotation table and
vectors new cluster names (for use with <code>Rename_Clusters</code> function or manual identity renaming).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# If pulling from a data.frame/tibble
cluster_annotation &lt;- Pull_Cluster_Annotation(annotation = annotation_df,
cluster_name_col = "cluster", cell_type_col = "cell_type")

# If pulling from csv file
cluster_annotation &lt;- Pull_Cluster_Annotation(annotation = "file_path/file_name.csv",
cluster_name_col = "cluster", cell_type_col = "cell_type")

## End(Not run)

</code></pre>

<hr>
<h2 id='Pull_Directory_List'>Pull Directory List</h2><span id='topic+Pull_Directory_List'></span>

<h3>Description</h3>

<p>Enables easy listing of all sub-directories for use as input library lists in Read10X multi functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pull_Directory_List(base_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Pull_Directory_List_+3A_base_path">base_path</code></td>
<td>
<p>path to the parent directory which contains all of the subdirectories of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of sub-directories within <code>base_path</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data_dir &lt;- 'path/to/data/directory'
library_list &lt;- Pull_Directory_List(base_path = data_dir)

## End(Not run)

</code></pre>

<hr>
<h2 id='QC_Histogram'>QC Histogram Plots</h2><span id='topic+QC_Histogram'></span>

<h3>Description</h3>

<p>Custom histogram for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Histogram(
  seurat_object,
  features,
  low_cutoff = NULL,
  high_cutoff = NULL,
  split.by = NULL,
  bins = 250,
  colors_use = "dodgerblue",
  num_columns = NULL,
  plot_title = NULL,
  assay = NULL,
  print_defaults = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Histogram_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_features">features</code></td>
<td>
<p>Feature from meta.data, assay features, or feature name shortcut to plot.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_low_cutoff">low_cutoff</code></td>
<td>
<p>Plot line a potential low threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_high_cutoff">high_cutoff</code></td>
<td>
<p>Plot line a potential high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_split.by">split.by</code></td>
<td>
<p>Feature to split plots by (i.e. &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_bins">bins</code></td>
<td>
<p>number of bins to plot default is 250.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_colors_use">colors_use</code></td>
<td>
<p>color to fill histogram bars, default is &quot;dodgerblue&quot;.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_num_columns">num_columns</code></td>
<td>
<p>Number of columns in plot layout.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_plot_title">plot_title</code></td>
<td>
<p>optional, vector to use for plot title.  Default is the name of the
variable being plotted.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_assay">assay</code></td>
<td>
<p>assay to pull features from, default is active assay.</p>
</td></tr>
<tr><td><code id="QC_Histogram_+3A_print_defaults">print_defaults</code></td>
<td>
<p>return list of accepted default shortcuts to provide to <code>features</code> instead
of full name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A patchwork object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
QC_Histogram(seurat_object = object, features = "nFeature_RNA")

## End(Not run)

</code></pre>

<hr>
<h2 id='QC_Plot_GenevsFeature'>QC Plots Genes vs Misc</h2><span id='topic+QC_Plot_GenevsFeature'></span>

<h3>Description</h3>

<p>Custom FeatureScatter for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plot_GenevsFeature(
  seurat_object,
  feature1,
  x_axis_label = NULL,
  y_axis_label = "Genes per Cell/Nucleus",
  low_cutoff_gene = NULL,
  high_cutoff_gene = NULL,
  low_cutoff_feature = NULL,
  high_cutoff_feature = NULL,
  colors_use = NULL,
  pt.size = 1,
  group.by = NULL,
  raster = NULL,
  raster.dpi = c(512, 512),
  ggplot_default_colors = FALSE,
  color_seed = 123,
  shuffle_seed = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_feature1">feature1</code></td>
<td>
<p>First feature to plot.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_low_cutoff_gene">low_cutoff_gene</code></td>
<td>
<p>Plot line a potential low threshold for filtering genes per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_high_cutoff_gene">high_cutoff_gene</code></td>
<td>
<p>Plot line a potential high threshold for filtering genes per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_low_cutoff_feature">low_cutoff_feature</code></td>
<td>
<p>Plot line a potential low threshold for filtering feature1 per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_high_cutoff_feature">high_cutoff_feature</code></td>
<td>
<p>Plot line a potential high threshold for filtering feature1 per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plotting by identity.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident).
Default is <code style="white-space: pre;">&#8288;@active.ident&#8288;</code>.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if greater
than 100,000 cells.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using default
ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_shuffle_seed">shuffle_seed</code></td>
<td>
<p>Sets the seed if randomly shuffling the order of points (Default is 1).</p>
</td></tr>
<tr><td><code id="QC_Plot_GenevsFeature_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+FeatureScatter">FeatureScatter</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
QC_Plot_GenevsFeature(seurat_object = obj, y_axis_label = "Feature per Cell")

## End(Not run)

</code></pre>

<hr>
<h2 id='QC_Plot_UMIvsFeature'>QC Plots UMI vs Misc</h2><span id='topic+QC_Plot_UMIvsFeature'></span>

<h3>Description</h3>

<p>Custom FeatureScatter for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plot_UMIvsFeature(
  seurat_object,
  feature1,
  x_axis_label = NULL,
  y_axis_label = "UMIs per Cell/Nucleus",
  low_cutoff_UMI = NULL,
  high_cutoff_UMI = NULL,
  low_cutoff_feature = NULL,
  high_cutoff_feature = NULL,
  colors_use = NULL,
  pt.size = 1,
  group.by = NULL,
  raster = NULL,
  raster.dpi = c(512, 512),
  ggplot_default_colors = FALSE,
  color_seed = 123,
  shuffle_seed = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_feature1">feature1</code></td>
<td>
<p>First feature to plot.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_low_cutoff_umi">low_cutoff_UMI</code></td>
<td>
<p>Plot line a potential low threshold for filtering UMI per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_high_cutoff_umi">high_cutoff_UMI</code></td>
<td>
<p>Plot line a potential high threshold for filtering UMI per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_low_cutoff_feature">low_cutoff_feature</code></td>
<td>
<p>Plot line a potential low threshold for filtering feature1 per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_high_cutoff_feature">high_cutoff_feature</code></td>
<td>
<p>Plot line a potential high threshold for filtering feature1 per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plotting by identity.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident).
Default is <code style="white-space: pre;">&#8288;@active.ident&#8288;</code>.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if greater
than 100,000 cells.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_shuffle_seed">shuffle_seed</code></td>
<td>
<p>Sets the seed if randomly shuffling the order of points (Default is 1).</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsFeature_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+FeatureScatter">FeatureScatter</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
QC_Plot_UMIvsFeature(seurat_object = obj, y_axis_label = "Feature per Cell")

## End(Not run)

</code></pre>

<hr>
<h2 id='QC_Plot_UMIvsGene'>QC Plots Genes vs UMIs</h2><span id='topic+QC_Plot_UMIvsGene'></span>

<h3>Description</h3>

<p>Custom FeatureScatter for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plot_UMIvsGene(
  seurat_object,
  x_axis_label = "UMIs per Cell/Nucleus",
  y_axis_label = "Genes per Cell/Nucleus",
  low_cutoff_gene = -Inf,
  high_cutoff_gene = Inf,
  low_cutoff_UMI = -Inf,
  high_cutoff_UMI = Inf,
  colors_use = NULL,
  meta_gradient_name = NULL,
  meta_gradient_color = viridis_plasma_dark_high,
  meta_gradient_na_color = "lightgray",
  meta_gradient_low_cutoff = NULL,
  cells = NULL,
  combination = FALSE,
  ident_legend = TRUE,
  pt.size = 1,
  group.by = NULL,
  raster = NULL,
  raster.dpi = c(512, 512),
  ggplot_default_colors = FALSE,
  color_seed = 123,
  shuffle_seed = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_low_cutoff_gene">low_cutoff_gene</code></td>
<td>
<p>Plot line a potential low threshold for filtering genes per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_high_cutoff_gene">high_cutoff_gene</code></td>
<td>
<p>Plot line a potential high threshold for filtering genes per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_low_cutoff_umi">low_cutoff_UMI</code></td>
<td>
<p>Plot line a potential low threshold for filtering UMIs per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_high_cutoff_umi">high_cutoff_UMI</code></td>
<td>
<p>Plot line a potential high threshold for filtering UMIs per cell.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plotting by identity.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_meta_gradient_name">meta_gradient_name</code></td>
<td>
<p>Name of continuous meta data variable to color points in plot by.
(MUST be continuous variable i.e. &quot;percent_mito&quot;).</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_meta_gradient_color">meta_gradient_color</code></td>
<td>
<p>The gradient color palette to use for plotting of meta variable (default is
viridis &quot;Plasma&quot; palette with dark colors high).</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_meta_gradient_na_color">meta_gradient_na_color</code></td>
<td>
<p>Color to use for plotting values when a <code>meta_gradient_low_cutoff</code> is
set (default is &quot;lightgray&quot;).</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_meta_gradient_low_cutoff">meta_gradient_low_cutoff</code></td>
<td>
<p>Value to use as threshold for plotting.  <code>meta_gradient_name</code> values
below this value will be plotted using <code>meta_gradient_na_color</code>.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_cells">cells</code></td>
<td>
<p>Cells to include on the scatter plot (default is all cells).</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_combination">combination</code></td>
<td>
<p>logical (default FALSE).  Whether or not to return a plot layout with both the
plot colored by identity and the meta data gradient plot.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_ident_legend">ident_legend</code></td>
<td>
<p>logical, whether to plot the legend containing identities (left plot) when
<code>combination = TRUE</code>.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_pt.size">pt.size</code></td>
<td>
<p>Passes size of points to both <code><a href="Seurat.html#topic+FeatureScatter">FeatureScatter</a></code> and <code>geom_point</code>.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident).
Default is <code style="white-space: pre;">&#8288;@active.ident&#8288;</code>.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 cells.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_color_seed">color_seed</code></td>
<td>
<p>Random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_shuffle_seed">shuffle_seed</code></td>
<td>
<p>Sets the seed if randomly shuffling the order of points (Default is 1).</p>
</td></tr>
<tr><td><code id="QC_Plot_UMIvsGene_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+FeatureScatter">FeatureScatter</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
QC_Plot_UMIvsGene(seurat_object = pbmc_small, x_axis_label = "UMIs per Cell/Nucleus",
y_axis_label = "Genes per Cell/Nucleus")

</code></pre>

<hr>
<h2 id='QC_Plots_Combined_Vln'>QC Plots Genes, UMIs, &amp; % Mito</h2><span id='topic+QC_Plots_Combined_Vln'></span>

<h3>Description</h3>

<p>Custom VlnPlot for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plots_Combined_Vln(
  seurat_object,
  group.by = NULL,
  feature_cutoffs = NULL,
  UMI_cutoffs = NULL,
  mito_cutoffs = NULL,
  mito_name = "percent_mito",
  pt.size = NULL,
  plot_median = FALSE,
  median_size = 15,
  plot_boxplot = FALSE,
  colors_use = NULL,
  x_lab_rotate = TRUE,
  y_axis_log = FALSE,
  raster = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_feature_cutoffs">feature_cutoffs</code></td>
<td>
<p>Numeric vector of length 1 or 2 to plot lines for  potential low/high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_umi_cutoffs">UMI_cutoffs</code></td>
<td>
<p>Numeric vector of length 1 or 2 to plot lines for  potential low/high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_mito_cutoffs">mito_cutoffs</code></td>
<td>
<p>Numeric vector of length 1 or 2 to plot lines for  potential low/high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_mito_name">mito_name</code></td>
<td>
<p>The column name containing percent mitochondrial counts information.  Default value is
&quot;percent_mito&quot; which is default value created when using <code>Add_Mito_Ribo_Seurat()</code>.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_pt.size">pt.size</code></td>
<td>
<p>Point size for plotting</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_plot_median">plot_median</code></td>
<td>
<p>logical, whether to plot median for each ident on the plot (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_median_size">median_size</code></td>
<td>
<p>Shape size for the median is plotted.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_plot_boxplot">plot_boxplot</code></td>
<td>
<p>logical, whether to plot boxplot inside of violin (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plot.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>Rotate x-axis labels 45 degrees (Default is TRUE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_y_axis_log">y_axis_log</code></td>
<td>
<p>logical. Whether to change y axis to log10 scale (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plots_Combined_Vln_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
QC_Plots_Combined_Vln(seurat_object = object)

## End(Not run)

</code></pre>

<hr>
<h2 id='QC_Plots_Complexity'>QC Plots Cell &quot;Complexity&quot;</h2><span id='topic+QC_Plots_Complexity'></span>

<h3>Description</h3>

<p>Custom VlnPlot for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plots_Complexity(
  seurat_object,
  feature = "log10GenesPerUMI",
  group.by = NULL,
  x_axis_label = NULL,
  y_axis_label = "log10(Genes) / log10(UMIs)",
  plot_title = "Cell Complexity",
  low_cutoff = NULL,
  high_cutoff = NULL,
  pt.size = NULL,
  plot_median = FALSE,
  plot_boxplot = FALSE,
  median_size = 15,
  colors_use = NULL,
  x_lab_rotate = TRUE,
  y_axis_log = FALSE,
  raster = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plots_Complexity_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_feature">feature</code></td>
<td>
<p>Feature from Meta Data to plot.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot Title.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_low_cutoff">low_cutoff</code></td>
<td>
<p>Plot line a potential low threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_high_cutoff">high_cutoff</code></td>
<td>
<p>Plot line a potential high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_pt.size">pt.size</code></td>
<td>
<p>Point size for plotting</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_plot_median">plot_median</code></td>
<td>
<p>logical, whether to plot median for each ident on the plot (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_plot_boxplot">plot_boxplot</code></td>
<td>
<p>logical, whether to plot boxplot inside of violin (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_median_size">median_size</code></td>
<td>
<p>Shape size for the median is plotted.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plot.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>Rotate x-axis labels 45 degrees (Default is TRUE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_y_axis_log">y_axis_log</code></td>
<td>
<p>logical. Whether to change y axis to log10 scale (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plots_Complexity_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
pbmc_small &lt;- Add_Cell_Complexity(pbmc_small)

QC_Plots_Complexity(seurat_object = pbmc_small)

</code></pre>

<hr>
<h2 id='QC_Plots_Feature'>QC Plots Feature</h2><span id='topic+QC_Plots_Feature'></span>

<h3>Description</h3>

<p>Custom VlnPlot for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plots_Feature(
  seurat_object,
  feature,
  group.by = NULL,
  x_axis_label = NULL,
  y_axis_label = NULL,
  plot_title = NULL,
  low_cutoff = NULL,
  high_cutoff = NULL,
  pt.size = NULL,
  plot_median = FALSE,
  median_size = 15,
  plot_boxplot = FALSE,
  colors_use = NULL,
  x_lab_rotate = TRUE,
  y_axis_log = FALSE,
  raster = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plots_Feature_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_feature">feature</code></td>
<td>
<p>Feature from Meta Data to plot.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot Title.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_low_cutoff">low_cutoff</code></td>
<td>
<p>Plot line a potential low threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_high_cutoff">high_cutoff</code></td>
<td>
<p>Plot line a potential high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_pt.size">pt.size</code></td>
<td>
<p>Point size for plotting.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_plot_median">plot_median</code></td>
<td>
<p>logical, whether to plot median for each ident on the plot (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_median_size">median_size</code></td>
<td>
<p>Shape size for the median is plotted.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_plot_boxplot">plot_boxplot</code></td>
<td>
<p>logical, whether to plot boxplot inside of violin (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plot.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>Rotate x-axis labels 45 degrees (Default is TRUE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_y_axis_log">y_axis_log</code></td>
<td>
<p>logical. Whether to change y axis to log10 scale (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plots_Feature_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
QC_Plots_Feature(seurat_object = object, feature = "FEATURE_NAME",
y_axis_label = "FEATURE per Cell", plot_title = "FEATURE per Cell", high_cutoff = 10,
low_cutoff = 2)

## End(Not run)

</code></pre>

<hr>
<h2 id='QC_Plots_Genes'>QC Plots Genes</h2><span id='topic+QC_Plots_Genes'></span>

<h3>Description</h3>

<p>Custom VlnPlot for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plots_Genes(
  seurat_object,
  plot_title = "Genes Per Cell/Nucleus",
  group.by = NULL,
  x_axis_label = NULL,
  y_axis_label = "Features",
  low_cutoff = NULL,
  high_cutoff = NULL,
  pt.size = NULL,
  plot_median = FALSE,
  plot_boxplot = FALSE,
  median_size = 15,
  colors_use = NULL,
  x_lab_rotate = TRUE,
  y_axis_log = FALSE,
  raster = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plots_Genes_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot Title.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_low_cutoff">low_cutoff</code></td>
<td>
<p>Plot line a potential low threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_high_cutoff">high_cutoff</code></td>
<td>
<p>Plot line a potential high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_pt.size">pt.size</code></td>
<td>
<p>Point size for plotting.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_plot_median">plot_median</code></td>
<td>
<p>logical, whether to plot median for each ident on the plot (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_plot_boxplot">plot_boxplot</code></td>
<td>
<p>logical, whether to plot boxplot inside of violin (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_median_size">median_size</code></td>
<td>
<p>Shape size for the median is plotted.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plot.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>Rotate x-axis labels 45 degrees (Default is TRUE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_y_axis_log">y_axis_log</code></td>
<td>
<p>logical. Whether to change y axis to log10 scale (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plots_Genes_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
QC_Plots_Genes(seurat_object = pbmc_small, plot_title = "Genes per Cell", low_cutoff = 40,
high_cutoff = 85)

</code></pre>

<hr>
<h2 id='QC_Plots_Mito'>QC Plots Mito</h2><span id='topic+QC_Plots_Mito'></span>

<h3>Description</h3>

<p>#' Custom VlnPlot for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plots_Mito(
  seurat_object,
  mito_name = "percent_mito",
  plot_title = "Mito Gene % per Cell/Nucleus",
  group.by = NULL,
  x_axis_label = NULL,
  y_axis_label = "% Mitochondrial Gene Counts",
  low_cutoff = NULL,
  high_cutoff = NULL,
  pt.size = NULL,
  plot_median = FALSE,
  median_size = 15,
  plot_boxplot = FALSE,
  colors_use = NULL,
  x_lab_rotate = TRUE,
  y_axis_log = FALSE,
  raster = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plots_Mito_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_mito_name">mito_name</code></td>
<td>
<p>The column name containing percent mitochondrial counts information.  Default value is
&quot;percent_mito&quot; which is default value created when using <code>Add_Mito_Ribo_Seurat()</code>.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot Title.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_low_cutoff">low_cutoff</code></td>
<td>
<p>Plot line a potential low threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_high_cutoff">high_cutoff</code></td>
<td>
<p>Plot line a potential high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_pt.size">pt.size</code></td>
<td>
<p>Point size for plotting.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_plot_median">plot_median</code></td>
<td>
<p>logical, whether to plot median for each ident on the plot (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_median_size">median_size</code></td>
<td>
<p>Shape size for the median is plotted.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_plot_boxplot">plot_boxplot</code></td>
<td>
<p>logical, whether to plot boxplot inside of violin (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plot.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>Rotate x-axis labels 45 degrees (Default is TRUE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_y_axis_log">y_axis_log</code></td>
<td>
<p>logical. Whether to change y axis to log10 scale (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plots_Mito_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
QC_Plots_Mito(seurat_object = object, plot_title = "Percent Mito per Cell", high_cutoff = 10)

## End(Not run)

</code></pre>

<hr>
<h2 id='QC_Plots_UMIs'>QC Plots UMIs</h2><span id='topic+QC_Plots_UMIs'></span>

<h3>Description</h3>

<p>#' Custom VlnPlot for initial QC checks including lines for thresholding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC_Plots_UMIs(
  seurat_object,
  plot_title = "UMIs per Cell/Nucleus",
  group.by = NULL,
  x_axis_label = NULL,
  y_axis_label = "UMIs",
  low_cutoff = NULL,
  high_cutoff = NULL,
  pt.size = NULL,
  plot_median = FALSE,
  median_size = 15,
  plot_boxplot = FALSE,
  colors_use = NULL,
  x_lab_rotate = TRUE,
  y_axis_log = FALSE,
  raster = NULL,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_Plots_UMIs_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_plot_title">plot_title</code></td>
<td>
<p>Plot Title.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_x_axis_label">x_axis_label</code></td>
<td>
<p>Label for x axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_y_axis_label">y_axis_label</code></td>
<td>
<p>Label for y axis.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_low_cutoff">low_cutoff</code></td>
<td>
<p>Plot line a potential low threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_high_cutoff">high_cutoff</code></td>
<td>
<p>Plot line a potential high threshold for filtering.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_pt.size">pt.size</code></td>
<td>
<p>Point size for plotting.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_plot_median">plot_median</code></td>
<td>
<p>logical, whether to plot median for each ident on the plot (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_median_size">median_size</code></td>
<td>
<p>Shape size for the median is plotted.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_plot_boxplot">plot_boxplot</code></td>
<td>
<p>logical, whether to plot boxplot inside of violin (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_colors_use">colors_use</code></td>
<td>
<p>vector of colors to use for plot.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>Rotate x-axis labels 45 degrees (Default is TRUE).</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_y_axis_log">y_axis_log</code></td>
<td>
<p>logical. Whether to change y axis to log10 scale (Default is FALSE).</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="QC_Plots_UMIs_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
QC_Plots_UMIs(seurat_object = pbmc_small, plot_title = "UMIs per Cell", low_cutoff = 75,
high_cutoff = 600)

</code></pre>

<hr>
<h2 id='Read_CellBender_h5_Mat'>Load CellBender h5 matrices (corrected)</h2><span id='topic+Read_CellBender_h5_Mat'></span>

<h3>Description</h3>

<p>Extract sparse matrix with corrected counts from CellBender h5 output file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read_CellBender_h5_Mat(
  file_name,
  use.names = TRUE,
  unique.features = TRUE,
  h5_group_name = NULL,
  feature_slot_name = "features"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read_CellBender_h5_Mat_+3A_file_name">file_name</code></td>
<td>
<p>Path to h5 file.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Mat_+3A_use.names">use.names</code></td>
<td>
<p>Label row names with feature names rather than ID numbers (default TRUE).</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Mat_+3A_unique.features">unique.features</code></td>
<td>
<p>Make feature names unique (default TRUE).</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Mat_+3A_h5_group_name">h5_group_name</code></td>
<td>
<p>Name of the group within H5 file that contains count data.  This is only
required if H5 file contains multiple subgroups and non-default names.  Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Mat_+3A_feature_slot_name">feature_slot_name</code></td>
<td>
<p>Name of the slot contain feature names/ids.  Must be one of:
&quot;features&quot;(Cell Ranger v3+) or &quot;genes&quot; (Cell Ranger v1/v2 or STARsolo).  Default is &quot;features&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sparse matrix
</p>


<h3>References</h3>

<p>Code used in function has been modified from <code>Seurat::Read10X_h5</code> function of
Seurat package <a href="https://github.com/satijalab/seurat">https://github.com/satijalab/seurat</a> (License: GPL-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mat &lt;- Read_CellBender_h5_Mat(file_name = "/SampleA_out_filtered.h5")

## End(Not run)

</code></pre>

<hr>
<h2 id='Read_CellBender_h5_Multi_Directory'>Load CellBender h5 matrices (corrected) from multiple directories</h2><span id='topic+Read_CellBender_h5_Multi_Directory'></span>

<h3>Description</h3>

<p>Extract sparse matrix with corrected counts from CellBender h5 output file across multiple sample
subdirectories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read_CellBender_h5_Multi_Directory(
  base_path,
  secondary_path = NULL,
  filtered_h5 = TRUE,
  custom_name = NULL,
  sample_list = NULL,
  sample_names = NULL,
  h5_group_name = NULL,
  feature_slot_name = "features",
  replace_suffix = FALSE,
  new_suffix_list = NULL,
  parallel = FALSE,
  num_cores = NULL,
  merge = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_base_path">base_path</code></td>
<td>
<p>path to the parent directory which contains all of the subdirectories of interest.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_secondary_path">secondary_path</code></td>
<td>
<p>path from the parent directory to count matrix files for each sample.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_filtered_h5">filtered_h5</code></td>
<td>
<p>logical (default TRUE).  Will set the shared file name suffix <code>custom_name</code> is NULL.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_custom_name">custom_name</code></td>
<td>
<p>if file name was customized in CellBender then this parameter should contain the portion
of file name that is shared across all samples.  Must included the &quot;.h5&quot; extension as well.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_sample_list">sample_list</code></td>
<td>
<p>a vector of sample directory names if only specific samples are desired.  If <code>NULL</code> will
read in subdirectories in parent directory.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_sample_names">sample_names</code></td>
<td>
<p>a set of sample names to use for each sample entry in returned list.  If <code>NULL</code> will
set names to the subdirectory name of each sample.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_h5_group_name">h5_group_name</code></td>
<td>
<p>Name of the group within H5 file that contains count data.  This is only
required if H5 file contains multiple subgroups and non-default names.  Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_feature_slot_name">feature_slot_name</code></td>
<td>
<p>Name of the slot contain feature names/ids.  Must be one of:
&quot;features&quot;(Cell Ranger v3+) or &quot;genes&quot; (Cell Ranger v1/v2 or STARsolo).  Default is &quot;features&quot;.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_replace_suffix">replace_suffix</code></td>
<td>
<p>logical (default FALSE).  Whether or not to replace the barcode suffixes of matrices
using <code><a href="#topic+Replace_Suffix">Replace_Suffix</a></code>.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_new_suffix_list">new_suffix_list</code></td>
<td>
<p>a vector of new suffixes to replace existing suffixes if <code>replace_suffix = TRUE</code>.
See <code><a href="#topic+Replace_Suffix">Replace_Suffix</a></code> for more information.  To remove all suffixes set <code>new_suffix_list = ""</code>.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_parallel">parallel</code></td>
<td>
<p>logical (default FALSE) whether or not to use multi core processing to read in matrices.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_num_cores">num_cores</code></td>
<td>
<p>how many cores to use for parallel processing.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_merge">merge</code></td>
<td>
<p>logical (default FALSE) whether or not to merge samples into a single matrix or return
list of matrices.  If TRUE each sample entry in list will have cell barcode prefix added.  The prefix
will be taken from <code>sample_names</code>.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_Directory_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="#topic+Read_CellBender_h5_Mat">Read_CellBender_h5_Mat</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of sparse matrices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
base_path &lt;- 'path/to/data/directory'
mat_list &lt;- Read_CellBender_h5_Multi_Directory(base_path = base_path)

## End(Not run)

</code></pre>

<hr>
<h2 id='Read_CellBender_h5_Multi_File'>Load CellBender h5 matrices (corrected) from multiple files</h2><span id='topic+Read_CellBender_h5_Multi_File'></span>

<h3>Description</h3>

<p>Extract sparse matrix with corrected counts from CellBender h5 output file across multiple samples
within the same directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read_CellBender_h5_Multi_File(
  data_dir = NULL,
  filtered_h5 = TRUE,
  custom_name = NULL,
  sample_list = NULL,
  sample_names = NULL,
  h5_group_name = NULL,
  feature_slot_name = "features",
  parallel = FALSE,
  num_cores = NULL,
  merge = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_data_dir">data_dir</code></td>
<td>
<p>Directory containing the .h5 files output by CellBender.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_filtered_h5">filtered_h5</code></td>
<td>
<p>logical (default TRUE).  Will set the shared file name suffix if <code>custom_name</code> is NULL.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_custom_name">custom_name</code></td>
<td>
<p>if file name was customized in CellBender then this parameter should contain the portion
of file name that is shared across all samples.  Must included the &quot;.h5&quot; extension as well.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_sample_list">sample_list</code></td>
<td>
<p>a vector of sample names if only specific samples are desired.  If <code>NULL</code> will
read in all files within <code>data_dir</code> directory.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_sample_names">sample_names</code></td>
<td>
<p>a set of sample names to use for each sample entry in returned list.  If <code>NULL</code> will
set names to the subdirectory name of each sample.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_h5_group_name">h5_group_name</code></td>
<td>
<p>Name of the group within H5 file that contains count data.  This is only
required if H5 file contains multiple subgroups and non-default names.  Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_feature_slot_name">feature_slot_name</code></td>
<td>
<p>Name of the slot contain feature names/ids.  Must be one of:
&quot;features&quot;(Cell Ranger v3+) or &quot;genes&quot; (Cell Ranger v1/v2 or STARsolo).  Default is &quot;features&quot;.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_parallel">parallel</code></td>
<td>
<p>logical (default FALSE) whether or not to use multi core processing to read in matrices</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_num_cores">num_cores</code></td>
<td>
<p>how many cores to use for parallel processing.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_merge">merge</code></td>
<td>
<p>logical (default FALSE) whether or not to merge samples into a single matrix or return
list of matrices.  If TRUE each sample entry in list will have cell barcode prefix added.  The prefix
will be taken from <code>sample_names</code>.</p>
</td></tr>
<tr><td><code id="Read_CellBender_h5_Multi_File_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="#topic+Read_CellBender_h5_Mat">Read_CellBender_h5_Mat</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of sparse matrices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
base_path &lt;- 'path/to/data/directory'
mat_list &lt;- Read_CellBender_h5_Multi_File(data_dir = base_path)

## End(Not run)

</code></pre>

<hr>
<h2 id='Read_GEO_Delim'>Load in NCBI GEO data formatted as single file per sample</h2><span id='topic+Read_GEO_Delim'></span>

<h3>Description</h3>

<p>Can read delimited file types (i.e. csv, tsv, txt)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read_GEO_Delim(
  data_dir,
  file_suffix,
  move_genes_rownames = TRUE,
  sample_list = NULL,
  full_names = FALSE,
  sample_names = NULL,
  barcode_suffix_period = FALSE,
  parallel = FALSE,
  num_cores = NULL,
  merge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read_GEO_Delim_+3A_data_dir">data_dir</code></td>
<td>
<p>Directory containing the files.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_file_suffix">file_suffix</code></td>
<td>
<p>The file suffix of the individual files.  Must be the same across all files being
imported.  This is used to detect files to import and their GEO IDs.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_move_genes_rownames">move_genes_rownames</code></td>
<td>
<p>logical.  Whether gene IDs are present in first column or in row names of
delimited file.  If TRUE will move the first column to row names before creating final matrix.
Default is TRUE.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_sample_list">sample_list</code></td>
<td>
<p>a vector of samples within directory to read in (can be either with or
without <code>file_suffix</code> see <code>full_names</code>).  If NULL will read in all subdirectories.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_full_names">full_names</code></td>
<td>
<p>logical (default FALSE).  Whether or not the <code>sample_list</code> vector includes the file suffix.
If <code>FALSE</code> the function will add suffix based on <code>file_suffix</code> parameter.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_sample_names">sample_names</code></td>
<td>
<p>a set of sample names to use for each sample entry in returned list.
If <code>NULL</code> will set names to the directory name of each sample.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_barcode_suffix_period">barcode_suffix_period</code></td>
<td>
<p>Is the barcode suffix a period and should it be changed to &quot;-&quot;.  Default (FALSE;
barcodes will be left identical to their format in input files.).  If TRUE &quot;.&quot; in barcode suffix will
be changed to &quot;-&quot;.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_parallel">parallel</code></td>
<td>
<p>logical (default FALSE).  Whether to use multiple cores when reading in data.
Only possible on Linux based systems.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_num_cores">num_cores</code></td>
<td>
<p>if <code>parallel = TRUE</code> indicates the number of cores to use for multicore processing.</p>
</td></tr>
<tr><td><code id="Read_GEO_Delim_+3A_merge">merge</code></td>
<td>
<p>logical (default FALSE) whether or not to merge samples into a single matrix or return
list of matrices.  If TRUE each sample entry in list will have cell barcode prefix added.  The prefix
will be taken from <code>sample_names</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of gene x cell matrices in list format named by sample name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data_dir &lt;- 'path/to/data/directory'
expression_matrices &lt;- Read_GEO_Delim(data_dir = data_dir)

## End(Not run)

</code></pre>

<hr>
<h2 id='Read_Metrics_10X'>Read Overall Statistics from 10X Cell Ranger Count</h2><span id='topic+Read_Metrics_10X'></span>

<h3>Description</h3>

<p>Get data.frame with all metrics from the Cell Ranger count analysis (present in web_summary.html)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read_Metrics_10X(
  base_path,
  secondary_path = NULL,
  default_10X = TRUE,
  cellranger_multi = FALSE,
  lib_list = NULL,
  lib_names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read_Metrics_10X_+3A_base_path">base_path</code></td>
<td>
<p>path to the parent directory which contains all of the subdirectories of interest.</p>
</td></tr>
<tr><td><code id="Read_Metrics_10X_+3A_secondary_path">secondary_path</code></td>
<td>
<p>path from the parent directory to count &quot;outs/&quot; folder which contains the
&quot;metrics_summary.csv&quot; file.</p>
</td></tr>
<tr><td><code id="Read_Metrics_10X_+3A_default_10x">default_10X</code></td>
<td>
<p>logical (default TRUE) sets the secondary path variable to the default 10X directory structure.</p>
</td></tr>
<tr><td><code id="Read_Metrics_10X_+3A_cellranger_multi">cellranger_multi</code></td>
<td>
<p>logical, whether or not metrics come from Cell Ranger <code>count</code> or from Cell Ranger <code>multi</code>.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Read_Metrics_10X_+3A_lib_list">lib_list</code></td>
<td>
<p>a list of sample names (matching directory names) to import.  If <code>NULL</code> will read
in all samples in parent directory.</p>
</td></tr>
<tr><td><code id="Read_Metrics_10X_+3A_lib_names">lib_names</code></td>
<td>
<p>a set of sample names to use for each sample.  If <code>NULL</code> will set names to the
directory name of each sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with sample metrics from cell ranger.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
metrics &lt;- Read_Metrics_10X(base_path = "/path/to/directories", default_10X = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='Read10X_GEO'>Load in NCBI GEO data from 10X</h2><span id='topic+Read10X_GEO'></span>

<h3>Description</h3>

<p>Enables easy loading of sparse data matrices provided by 10X genomics. That have file prefixes
added to them by NCBI GEO or other repos.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read10X_GEO(
  data_dir = NULL,
  sample_list = NULL,
  sample_names = NULL,
  gene.column = 2,
  cell.column = 1,
  unique.features = TRUE,
  strip.suffix = FALSE,
  parallel = FALSE,
  num_cores = NULL,
  merge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read10X_GEO_+3A_data_dir">data_dir</code></td>
<td>
<p>Directory containing the matrix.mtx, genes.tsv (or features.tsv), and barcodes.tsv
files provided by 10X.</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_sample_list">sample_list</code></td>
<td>
<p>A vector of file prefixes/names if specific samples are desired.  Default is <code>NULL</code> and
will load all samples in given directory.</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_sample_names">sample_names</code></td>
<td>
<p>a set of sample names to use for each sample entry in returned list.  If <code>NULL</code> will
set names to the file name of each sample.</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_gene.column">gene.column</code></td>
<td>
<p>Specify which column of genes.tsv or features.tsv to use for gene names; default is 2.</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_cell.column">cell.column</code></td>
<td>
<p>Specify which column of barcodes.tsv to use for cell names; default is 1.</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_unique.features">unique.features</code></td>
<td>
<p>Make feature names unique (default TRUE).</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_strip.suffix">strip.suffix</code></td>
<td>
<p>Remove trailing &quot;-1&quot; if present in all cell barcodes.</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_parallel">parallel</code></td>
<td>
<p>logical (default FALSE).  Whether to use multiple cores when reading in data.
Only possible on Linux based systems.</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_num_cores">num_cores</code></td>
<td>
<p>if <code>parallel = TRUE</code> indicates the number of cores to use for multicore processing.</p>
</td></tr>
<tr><td><code id="Read10X_GEO_+3A_merge">merge</code></td>
<td>
<p>logical (default FALSE) whether or not to merge samples into a single matrix or return
list of matrices.  If TRUE each sample entry in list will have cell barcode prefix added.  The prefix
will be taken from <code>sample_names</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If features.csv indicates the data has multiple data types, a list
containing a sparse matrix of the data from each type will be returned.
Otherwise a sparse matrix containing the expression data will be returned.
</p>


<h3>References</h3>

<p>Code used in function has been slightly modified from <code>Seurat::Read10X</code> function of
Seurat package <a href="https://github.com/satijalab/seurat">https://github.com/satijalab/seurat</a> (License: GPL-3).  Function was modified to
support file prefixes and altered loop by Samuel Marsh for scCustomize (also previously posted as
potential PR to Seurat GitHub).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data_dir &lt;- 'path/to/data/directory'
expression_matrices &lt;- Read10X_GEO(data_dir = data_dir)
# To create object from single file
seurat_object = CreateSeuratObject(counts = expression_matrices[[1]])

## End(Not run)

</code></pre>

<hr>
<h2 id='Read10X_h5_GEO'>Load in NCBI GEO data from 10X in HDF5 file format</h2><span id='topic+Read10X_h5_GEO'></span>

<h3>Description</h3>

<p>Enables easy loading of HDF5 data matrices provided by 10X genomics. That have file prefixes added to
them by NCBI GEO or other repos or programs (i.e. Cell Bender)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read10X_h5_GEO(
  data_dir = NULL,
  sample_list = NULL,
  sample_names = NULL,
  shared_suffix = NULL,
  parallel = FALSE,
  num_cores = NULL,
  merge = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read10X_h5_GEO_+3A_data_dir">data_dir</code></td>
<td>
<p>Directory containing the .h5 files provided by 10X.</p>
</td></tr>
<tr><td><code id="Read10X_h5_GEO_+3A_sample_list">sample_list</code></td>
<td>
<p>A vector of file prefixes/names if specific samples are desired.  Default is <code>NULL</code> and
will load all samples in given directory.</p>
</td></tr>
<tr><td><code id="Read10X_h5_GEO_+3A_sample_names">sample_names</code></td>
<td>
<p>a set of sample names to use for each sample entry in returned list.  If <code>NULL</code>
will set names to the file name of each sample.</p>
</td></tr>
<tr><td><code id="Read10X_h5_GEO_+3A_shared_suffix">shared_suffix</code></td>
<td>
<p>a suffix and file extension shared by all samples.</p>
</td></tr>
<tr><td><code id="Read10X_h5_GEO_+3A_parallel">parallel</code></td>
<td>
<p>logical (default FALSE).  Whether to use multiple cores when reading in data.
Only possible on Linux based systems.</p>
</td></tr>
<tr><td><code id="Read10X_h5_GEO_+3A_num_cores">num_cores</code></td>
<td>
<p>if <code>parallel = TRUE</code> indicates the number of cores to use for multicore processing.</p>
</td></tr>
<tr><td><code id="Read10X_h5_GEO_+3A_merge">merge</code></td>
<td>
<p>logical (default FALSE) whether or not to merge samples into a single matrix or return
list of matrices.  If TRUE each sample entry in list will have cell barcode prefix added.  The prefix
will be taken from <code>sample_names</code>.</p>
</td></tr>
<tr><td><code id="Read10X_h5_GEO_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="Seurat.html#topic+Read10X_h5">Read10X_h5</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>If the data has multiple data types, a list
containing a sparse matrix of the data from each type will be returned.
Otherwise a sparse matrix containing the expression data will be returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data_dir &lt;- 'path/to/data/directory'
expression_matrices &lt;- Read10X_h5_GEO(data_dir = data_dir)
# To create object from single file
seurat_object = CreateSeuratObject(counts = expression_matrices[[1]])

## End(Not run)

</code></pre>

<hr>
<h2 id='Read10X_h5_Multi_Directory'>Load 10X h5 count matrices from multiple directories</h2><span id='topic+Read10X_h5_Multi_Directory'></span>

<h3>Description</h3>

<p>Enables easy loading of sparse data matrices provided by 10X genomics that are present in multiple
subdirectories.  Can function with either default output directory structure of Cell Ranger or
custom directory structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read10X_h5_Multi_Directory(
  base_path,
  secondary_path = NULL,
  default_10X_path = TRUE,
  cellranger_multi = FALSE,
  h5_filename = "filtered_feature_bc_matrix.h5",
  cell_bender = deprecated(),
  sample_list = NULL,
  sample_names = NULL,
  replace_suffix = FALSE,
  new_suffix_list = NULL,
  parallel = FALSE,
  num_cores = NULL,
  merge = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_base_path">base_path</code></td>
<td>
<p>path to the parent directory which contains all of the subdirectories of interest.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_secondary_path">secondary_path</code></td>
<td>
<p>path from the parent directory to count matrix files for each sample.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_default_10x_path">default_10X_path</code></td>
<td>
<p>logical (default TRUE) sets the secondary path variable to the default 10X
directory structure.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_cellranger_multi">cellranger_multi</code></td>
<td>
<p>logical, whether samples were processed with Cell Ranger <code>multi</code>, default is FALSE.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_h5_filename">h5_filename</code></td>
<td>
<p>name of h5 file (including .h5 suffix).  If all h5 files have same name (i.e. Cell Ranger output)
then use full file name.  By default function uses Cell Ranger name: &quot;filtered_feature_bc_matrix.h5&quot;.
If h5 files have sample specific prefixes (i.e. from Cell Bender) then use only the shared part of file
name (e.g., &quot;_filtered_out.h5&quot;).</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_cell_bender">cell_bender</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> CellBender read functions are now independent family of functions.
See <code style="white-space: pre;">&#8288;Read_CellBender_*&#8288;</code> functions.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_sample_list">sample_list</code></td>
<td>
<p>a vector of sample directory names if only specific samples are desired.  If <code>NULL</code> will
read in subdirectories in parent directory.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_sample_names">sample_names</code></td>
<td>
<p>a set of sample names to use for each sample entry in returned list.  If <code>NULL</code> will
set names to the subdirectory name of each sample.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_replace_suffix">replace_suffix</code></td>
<td>
<p>logical (default FALSE).  Whether or not to replace the barcode suffixes of matrices
using <code><a href="#topic+Replace_Suffix">Replace_Suffix</a></code>.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_new_suffix_list">new_suffix_list</code></td>
<td>
<p>a vector of new suffixes to replace existing suffixes if <code>replace_suffix = TRUE</code>.
See <code><a href="#topic+Replace_Suffix">Replace_Suffix</a></code> for more information.  To remove all suffixes set <code>new_suffix_list = ""</code>.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_parallel">parallel</code></td>
<td>
<p>logical (default FALSE) whether or not to use multi core processing to read in matrices.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_num_cores">num_cores</code></td>
<td>
<p>how many cores to use for parallel processing.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_merge">merge</code></td>
<td>
<p>logical (default FALSE) whether or not to merge samples into a single matrix or return
list of matrices.  If TRUE each sample entry in list will have cell barcode prefix added.  The prefix
will be taken from <code>sample_names</code>.</p>
</td></tr>
<tr><td><code id="Read10X_h5_Multi_Directory_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+Read10X_h5">Read10X_h5</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of sparse matrices (merge = FALSE) or a single sparse matrix (merge = TRUE).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
base_path &lt;- 'path/to/data/directory'
expression_matrices &lt;- Read10X_h5_Multi_Directory(base_path = base_path)

## End(Not run)

</code></pre>

<hr>
<h2 id='Read10X_Multi_Directory'>Load 10X count matrices from multiple directories</h2><span id='topic+Read10X_Multi_Directory'></span>

<h3>Description</h3>

<p>Enables easy loading of sparse data matrices provided by 10X genomics that are present in multiple
subdirectories.  Can function with either default output directory structure of Cell Ranger or
custom directory structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Read10X_Multi_Directory(
  base_path,
  secondary_path = NULL,
  default_10X_path = TRUE,
  cellranger_multi = FALSE,
  sample_list = NULL,
  sample_names = NULL,
  parallel = FALSE,
  num_cores = NULL,
  merge = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Read10X_Multi_Directory_+3A_base_path">base_path</code></td>
<td>
<p>path to the parent directory which contains all of the subdirectories of interest.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_secondary_path">secondary_path</code></td>
<td>
<p>path from the parent directory to count matrix files for each sample.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_default_10x_path">default_10X_path</code></td>
<td>
<p>logical (default TRUE) sets the secondary path variable to the default 10X
directory structure.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_cellranger_multi">cellranger_multi</code></td>
<td>
<p>logical, whether samples were processed with Cell Ranger <code>multi</code>, default is FALSE.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_sample_list">sample_list</code></td>
<td>
<p>a vector of sample directory names if only specific samples are desired.  If <code>NULL</code> will
read in subdirectories in parent directory.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_sample_names">sample_names</code></td>
<td>
<p>a set of sample names to use for each sample entry in returned list.  If <code>NULL</code> will
set names to the subdirectory name of each sample.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_parallel">parallel</code></td>
<td>
<p>logical (default FALSE) whether or not to use multi core processing to read in matrices.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_num_cores">num_cores</code></td>
<td>
<p>how many cores to use for parallel processing.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_merge">merge</code></td>
<td>
<p>logical (default FALSE) whether or not to merge samples into a single matrix or return
list of matrices.  If TRUE each sample entry in list will have cell barcode prefix added.  The prefix
will be taken from <code>sample_names</code>.</p>
</td></tr>
<tr><td><code id="Read10X_Multi_Directory_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+Read10X">Read10X</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of sparse matrices (merge = FALSE) or a single sparse matrix (merge = TRUE).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
base_path &lt;- 'path/to/data/directory'
expression_matrices &lt;- Read10X_Multi_Directory(base_path = base_path)

## End(Not run)

</code></pre>

<hr>
<h2 id='Reduction_Loading_Present'>Check if reduction loadings are present</h2><span id='topic+Reduction_Loading_Present'></span>

<h3>Description</h3>

<p>Check if reduction loadings are present in object and return vector of found loading names.  Return
warning messages for genes not found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Reduction_Loading_Present(
  seurat_object,
  reduction_names,
  print_msg = TRUE,
  omit_warn = TRUE,
  return_none = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Reduction_Loading_Present_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Reduction_Loading_Present_+3A_reduction_names">reduction_names</code></td>
<td>
<p>vector of genes to check.</p>
</td></tr>
<tr><td><code id="Reduction_Loading_Present_+3A_print_msg">print_msg</code></td>
<td>
<p>logical. Whether message should be printed if all features are found.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="Reduction_Loading_Present_+3A_omit_warn">omit_warn</code></td>
<td>
<p>logical. Whether to print message about features that are not found in current object.
Default is TRUE.</p>
</td></tr>
<tr><td><code id="Reduction_Loading_Present_+3A_return_none">return_none</code></td>
<td>
<p>logical. Whether list of found vs. bad features should still be returned if no
features are found.  Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length 3 containing 1) found features, 2) not found features.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
reductions &lt;- Reduction_Loading_Present(seurat_object = obj_name, reduction_name = "PC_1")
found_features &lt;- features[[1]]

## End(Not run)

</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+as.Seurat'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>SeuratObject</dt><dd><p><code><a href="SeuratObject.html#topic+as.Seurat">as.Seurat</a></code></p>
</dd>
</dl>


<h3>Note</h3>

<p>See <code><a href="#topic+as.Seurat.liger">as.Seurat.liger</a></code> for scCustomize extension of this generic to converting Liger objects.
</p>

<hr>
<h2 id='Rename_Clusters'>Rename Cluster Seurat</h2><span id='topic+Rename_Clusters'></span>

<h3>Description</h3>

<p>Wrapper function to rename active identities in Seurat Object with new idents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Rename_Clusters(seurat_object, new_idents, meta_col_name = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Rename_Clusters_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Rename_Clusters_+3A_new_idents">new_idents</code></td>
<td>
<p>vector of new cluster names.  Must be equal to the length of current active.ident
in Seurat Object.  Will accept named vector (with old idents as names) or will name the new_idents vector internally.</p>
</td></tr>
<tr><td><code id="Rename_Clusters_+3A_meta_col_name">meta_col_name</code></td>
<td>
<p>(Optional).  Whether or not to create new named column in <code>Object@meta.data</code>
to store the old identities.</p>
</td></tr>
<tr><td><code id="Rename_Clusters_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="SeuratObject.html#topic+RenameIdents">RenameIdents</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat Object with new identities placed in active.ident slot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
obj &lt;- Rename_Clusters(seurat_object = obj_name, new_idents = new_idents_vec,
meta_col_name = "Round01_Res0.6_Idents")

## End(Not run)

</code></pre>

<hr>
<h2 id='Replace_Suffix'>Replace barcode suffixes</h2><span id='topic+Replace_Suffix'></span>

<h3>Description</h3>

<p>Replace barcode suffixes in matrix, data.frame, or list of matrices/data.frames
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Replace_Suffix(data, current_suffix, new_suffix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Replace_Suffix_+3A_data">data</code></td>
<td>
<p>Either matrix/data.frame or list of matrices/data.frames with the cell barcodes in the column names.</p>
</td></tr>
<tr><td><code id="Replace_Suffix_+3A_current_suffix">current_suffix</code></td>
<td>
<p>a single value or vector of values representing current barcode suffix.
If suffix is the same for all matrices/data.frames in list only single value is required.</p>
</td></tr>
<tr><td><code id="Replace_Suffix_+3A_new_suffix">new_suffix</code></td>
<td>
<p>a single value or vector of values representing new barcode suffix to be added.
If desired suffix is the same for all matrices/data.frames in list only single value is required.
If no suffix is desired set <code>new_suffix = ""</code>.'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix or data.frame with new column names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dge_matrix &lt;- Replace_Suffix(data = dge_matrix, current_suffix = "-1", new_suffix = "-2")

## End(Not run)

</code></pre>

<hr>
<h2 id='scCustomize_Palette'>Color Palette Selection for scCustomize</h2><span id='topic+scCustomize_Palette'></span>

<h3>Description</h3>

<p>Function to return package default discrete palettes depending on number of groups plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scCustomize_Palette(
  num_groups,
  ggplot_default_colors = FALSE,
  color_seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scCustomize_Palette_+3A_num_groups">num_groups</code></td>
<td>
<p>number of groups to be plotted. If <code>ggplot_default_colors = FALSE</code> then by default:
</p>

<ul>
<li><p> If number of levels plotted equal to 2 then colors will be <code>NavyAndOrange()</code>.
</p>
</li>
<li><p> If number of levels plotted greater than 2 but less than or equal to 36 it will use &quot;polychrome&quot; from <code>DiscretePalette_scCustomize()</code>.
</p>
</li>
<li><p> If greater than 36 will use &quot;varibow&quot; with shuffle = TRUE from <code>DiscretePalette_scCustomize</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="scCustomize_Palette_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  Whether to use default ggplot hue palette or not.</p>
</td></tr>
<tr><td><code id="scCustomize_Palette_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed to use for shuffling the &quot;varibow&quot; palette.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of colors to use for plotting.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cols &lt;- scCustomize_Palette(num_groups = 24, ggplot_default_colors = FALSE)
PalettePlot(pal= cols)

</code></pre>

<hr>
<h2 id='scCustomize-package'>scCustomize: Custom Visualizations &amp; Functions for Streamlined Analyses of Single Cell Sequencing</h2><span id='topic+scCustomize'></span><span id='topic+scCustomize-package'></span>

<h3>Description</h3>

<p>Collection of functions created and/or curated to aid in the visualization and analysis of single-cell data using 'R'. 'scCustomize' aims to provide 1) Customized visualizations for aid in ease of use and to create more aesthetic and functional visuals. 2) Improve speed/reproducibility of common tasks/pieces of code in scRNA-seq analysis with a single or group of functions. For citation please use: Marsh SE (2021) &quot;Custom Visualizations &amp; Functions for Streamlined Analyses of Single Cell Sequencing&quot; <a href="https://doi.org/10.5281/zenodo.5706430">doi:10.5281/zenodo.5706430</a> RRID:SCR_024675.
</p>


<h3>Package options</h3>

<p>scCustomize uses the following <code><a href="base.html#topic+options">options()</a></code> to configure behavior:
</p>

<dl>
<dt><code>scCustomize_warn_raster_iterative</code></dt><dd><p>Show message about setting <code>raster</code> parameter
in <code><a href="#topic+Iterate_FeaturePlot_scCustom">Iterate_FeaturePlot_scCustom</a></code> if <code>raster = FALSE</code> and <code>single_pdf = TRUE</code>
due to large file sizes.</p>
</dd>
<dt><code>scCustomize_warn_raster_LIGER</code></dt><dd><p>Show warning about rasterization of points in
<code><a href="#topic+DimPlot_LIGER">DimPlot_LIGER</a></code> due to new functionality compared to LIGER.</p>
</dd>
<dt><code>scCustomize_warn_na_cutoff</code></dt><dd><p>Show message about properly setting <code>na_cutoff</code> parameter in <code><a href="#topic+FeaturePlot_scCustom">FeaturePlot_scCustom</a></code>.</p>
</dd>
</dl>
<p>#'   </p>
<dl>
<dt><code>scCustomize_warn_zero_na_cutoff</code></dt><dd><p>Show message about properly setting <code>na_cutoff</code> parameter in <code><a href="#topic+FeaturePlot_scCustom">FeaturePlot_scCustom</a></code> if <code>na_cutoff</code> is set to exactly zero.</p>
</dd>
<dt><code>scCustomize_warn_vln_raster_iterative</code></dt><dd><p>Show message about <code><a href="#topic+Iterate_VlnPlot_scCustom">Iterate_VlnPlot_scCustom</a></code>
when <code>pt.size &gt; 0</code> due to current lack of raster support in <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code></p>
</dd>
<dt><code>scCustomize_warn_LIGER_dim_labels</code></dt><dd><p>Show message about <code><a href="#topic+DimPlot_LIGER">DimPlot_LIGER</a></code>
parameter <code>reduction_label</code> as LIGER objects do not store dimensionality reduction name and
and therefore needs to be set manually.</p>
</dd>
<dt><code>scCustomize_warn_DimPlot_split_type</code></dt><dd><p>Show message about <code><a href="#topic+DimPlot_scCustom">DimPlot_scCustom</a></code>
parameter <code>split.by</code> and <code>split_seurat</code> to alert user to difference in returned plots between
scCustomize and Seurat.</p>
</dd>
<dt><code>scCustomize_warn_FeatureScatter_split_type</code></dt><dd><p>Show message about <code><a href="#topic+FeatureScatter_scCustom">FeatureScatter_scCustom</a></code>
parameter <code>split.by</code> and <code>split_seurat</code> to alert user to difference in returned plots between
scCustomize and Seurat.</p>
</dd>
<dt><code>scCustomize_warn_LIGER_dim_labels_plotFactors</code></dt><dd><p>Show message about <code><a href="#topic+plotFactors_scCustom">plotFactors_scCustom</a></code>
parameter <code>reduction_label</code> as LIGER objects do not store dimensionality reduction name and
and therefore needs to be set manually.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Samuel Marsh <a href="mailto:samuel.marsh@childrens.harvard.edu">samuel.marsh@childrens.harvard.edu</a> (<a href="https://orcid.org/0000-0002-3012-6945">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Ming Tang <a href="mailto:tangming2005@gmail.com">tangming2005@gmail.com</a> [contributor]
</p>
</li>
<li><p> Velina Kozareva [contributor]
</p>
</li>
<li><p> Lucas Graybuck <a href="mailto:lucasg@alleninstitute.org">lucasg@alleninstitute.org</a> [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/samuel-marsh/scCustomize">https://github.com/samuel-marsh/scCustomize</a>
</p>
</li>
<li> <p><a href="https://samuel-marsh.github.io/scCustomize/">https://samuel-marsh.github.io/scCustomize/</a>
</p>
</li>
<li> <p><a href="https://doi.org/10.5281/zenodo.5706431">doi:10.5281/zenodo.5706431</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/samuel-marsh/scCustomize/issues">https://github.com/samuel-marsh/scCustomize/issues</a>
</p>
</li></ul>


<hr>
<h2 id='Seq_QC_Plot_Alignment_Combined'>QC Plots Sequencing metrics (Alignment) (Layout)</h2><span id='topic+Seq_QC_Plot_Alignment_Combined'></span>

<h3>Description</h3>

<p>Plot a combined plot of the Alignment QC metrics from sequencing output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Alignment_Combined(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  patchwork_title = "Sequencing QC Plots: Read Alignment Metrics",
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Alignment_Combined_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Alignment_Combined_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Alignment_Combined_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Alignment_Combined_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Alignment_Combined_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Alignment_Combined_+3A_patchwork_title">patchwork_title</code></td>
<td>
<p>Title to use for the patchworked plot output.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Alignment_Combined_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Alignment_Combined_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Alignment_Combined(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Antisense'>QC Plots Sequencing metrics (Alignment)</h2><span id='topic+Seq_QC_Plot_Antisense'></span>

<h3>Description</h3>

<p>Plot the fraction of reads mapped Antisense to Gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Antisense(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Antisense_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Antisense_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Antisense_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Antisense_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Antisense_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Antisense_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Antisense_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Antisense(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Basic_Combined'>QC Plots Sequencing metrics (Layout)</h2><span id='topic+Seq_QC_Plot_Basic_Combined'></span>

<h3>Description</h3>

<p>Plot a combined plot of the basic QC metrics from sequencing output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Basic_Combined(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  patchwork_title = "Sequencing QC Plots: Basic Cell Metrics",
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Basic_Combined_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Basic_Combined_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Basic_Combined_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Basic_Combined_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Basic_Combined_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Basic_Combined_+3A_patchwork_title">patchwork_title</code></td>
<td>
<p>Title to use for the patchworked plot output.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Basic_Combined_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Basic_Combined_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Basic_Combined(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Exonic'>QC Plots Sequencing metrics (Alignment)</h2><span id='topic+Seq_QC_Plot_Exonic'></span>

<h3>Description</h3>

<p>Plot the fraction of reads confidently mapped to Exonic regions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Exonic(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Exonic_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Exonic_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Exonic_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Exonic_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Exonic_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Exonic_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Exonic_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Exonic(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Genes'>QC Plots Sequencing metrics</h2><span id='topic+Seq_QC_Plot_Genes'></span>

<h3>Description</h3>

<p>Plot the median genes per cell per sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Genes(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Genes_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genes_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genes_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genes_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genes_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genes_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genes_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Genes(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Genome'>QC Plots Sequencing metrics (Alignment)</h2><span id='topic+Seq_QC_Plot_Genome'></span>

<h3>Description</h3>

<p>Plot the fraction of reads confidently mapped to genome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Genome(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Genome_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genome_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genome_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genome_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genome_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genome_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Genome_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Genome(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Intergenic'>QC Plots Sequencing metrics (Alignment)</h2><span id='topic+Seq_QC_Plot_Intergenic'></span>

<h3>Description</h3>

<p>Plot the fraction of reads confidently mapped to intergenic regions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Intergenic(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Intergenic_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intergenic_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intergenic_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intergenic_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intergenic_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intergenic_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intergenic_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Intergeneic(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Intronic'>QC Plots Sequencing metrics (Alignment)</h2><span id='topic+Seq_QC_Plot_Intronic'></span>

<h3>Description</h3>

<p>Plot the fraction of reads confidently mapped to intronic regions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Intronic(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Intronic_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intronic_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intronic_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intronic_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intronic_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intronic_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Intronic_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Intronic(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Number_Cells'>QC Plots Sequencing metrics</h2><span id='topic+Seq_QC_Plot_Number_Cells'></span>

<h3>Description</h3>

<p>Plot the number of cells per sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Number_Cells(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Number_Cells_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Number_Cells_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Number_Cells_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Number_Cells_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Number_Cells_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Number_Cells_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Number_Cells_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Number_Cells(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Reads_in_Cells'>QC Plots Sequencing metrics</h2><span id='topic+Seq_QC_Plot_Reads_in_Cells'></span>

<h3>Description</h3>

<p>Plot the fraction of reads in cells per sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Reads_in_Cells(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Reads_in_Cells_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_in_Cells_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_in_Cells_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_in_Cells_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_in_Cells_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_in_Cells_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_in_Cells_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Reads_in_Cells(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Reads_per_Cell'>QC Plots Sequencing metrics</h2><span id='topic+Seq_QC_Plot_Reads_per_Cell'></span>

<h3>Description</h3>

<p>Plot the mean number of reads per cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Reads_per_Cell(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Reads_per_Cell_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_per_Cell_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_per_Cell_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_per_Cell_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_per_Cell_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_per_Cell_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Reads_per_Cell_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Reads_per_Cell(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Saturation'>QC Plots Sequencing metrics</h2><span id='topic+Seq_QC_Plot_Saturation'></span>

<h3>Description</h3>

<p>Plot the sequencing saturation percentage per sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Saturation(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Saturation_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Saturation_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Saturation_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Saturation_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Saturation_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Saturation_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Saturation_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Saturation(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Total_Genes'>QC Plots Sequencing metrics</h2><span id='topic+Seq_QC_Plot_Total_Genes'></span>

<h3>Description</h3>

<p>Plot the total genes detected per sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Total_Genes(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Total_Genes_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Total_Genes_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Total_Genes_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Total_Genes_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Total_Genes_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Total_Genes_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Total_Genes_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Total_Genes(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_Transcriptome'>QC Plots Sequencing metrics (Alignment)</h2><span id='topic+Seq_QC_Plot_Transcriptome'></span>

<h3>Description</h3>

<p>Plot the fraction of reads confidently mapped to transcriptome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_Transcriptome(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_Transcriptome_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame  contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Transcriptome_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Transcriptome_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Transcriptome_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Transcriptome_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Transcriptome_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_Transcriptome_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_Transcriptome(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Seq_QC_Plot_UMIs'>QC Plots Sequencing metrics</h2><span id='topic+Seq_QC_Plot_UMIs'></span>

<h3>Description</h3>

<p>Plot the median UMIs per cell per sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seq_QC_Plot_UMIs(
  metrics_dataframe,
  plot_by = "sample_id",
  colors_use = NULL,
  dot_size = 1,
  x_lab_rotate = FALSE,
  significance = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Seq_QC_Plot_UMIs_+3A_metrics_dataframe">metrics_dataframe</code></td>
<td>
<p>data.frame contain Cell Ranger QC Metrics (see <code><a href="#topic+Read_Metrics_10X">Read_Metrics_10X</a></code>).</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_UMIs_+3A_plot_by">plot_by</code></td>
<td>
<p>Grouping factor for the plot.  Default is to plot as single group with single point per sample.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_UMIs_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plot if plotting by group.  Defaults to RColorBrewer Dark2 palette if
less than 8 groups and <code>DiscretePalette_scCustomize(palette = "polychrome")</code> if more than 8.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_UMIs_+3A_dot_size">dot_size</code></td>
<td>
<p>size of the dots plotted if <code>plot_by</code> is not <code>sample_id</code>  Default is 1.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_UMIs_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical.  Whether to rotate the axes labels on the x-axis.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_UMIs_+3A_significance">significance</code></td>
<td>
<p>logical.  Whether to calculate and plot p-value comparisons when plotting by
grouping factor.  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Seq_QC_Plot_UMIs_+3A_...">...</code></td>
<td>
<p>Other variables to pass to <code>ggpubr::stat_compare_means</code> when doing significance testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Seq_QC_Plot_UMIs(metrics_dataframe = metrics)

## End(Not run)

</code></pre>

<hr>
<h2 id='Setup_scRNAseq_Project'>Setup project directory structure</h2><span id='topic+Setup_scRNAseq_Project'></span>

<h3>Description</h3>

<p>Create reproducible project directory organization when initiating a new analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Setup_scRNAseq_Project(
  custom_dir_file = NULL,
  cluster_annotation_path = NULL,
  cluster_annotation_file_name = "cluster_annotation.csv"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Setup_scRNAseq_Project_+3A_custom_dir_file">custom_dir_file</code></td>
<td>
<p>file to file containing desired directory structure.  Default is NULL and
will provide generic built-in directory structure.</p>
</td></tr>
<tr><td><code id="Setup_scRNAseq_Project_+3A_cluster_annotation_path">cluster_annotation_path</code></td>
<td>
<p>path to place cluster annotation file using <code><a href="#topic+Create_Cluster_Annotation_File">Create_Cluster_Annotation_File</a></code>.</p>
</td></tr>
<tr><td><code id="Setup_scRNAseq_Project_+3A_cluster_annotation_file_name">cluster_annotation_file_name</code></td>
<td>
<p>name to use for annotation file if created (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return value.  Creates system folders.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# If using built-in directory structure.
Setup_scRNAseq_Project()

## End(Not run)

</code></pre>

<hr>
<h2 id='Single_Color_Palette'>Single Color Palettes for Plotting</h2><span id='topic+Single_Color_Palette'></span>

<h3>Description</h3>

<p>Selects colors from modified versions of RColorBrewer single colors palettes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Single_Color_Palette(pal_color, num_colors = NULL, seed_use = 123)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Single_Color_Palette_+3A_pal_color">pal_color</code></td>
<td>
<p>color palette to select (Options are: 'reds', 'blues',
'greens', 'purples', 'oranges', 'grays').</p>
</td></tr>
<tr><td><code id="Single_Color_Palette_+3A_num_colors">num_colors</code></td>
<td>
<p>set number of colors (max = 7).</p>
</td></tr>
<tr><td><code id="Single_Color_Palette_+3A_seed_use">seed_use</code></td>
<td>
<p>set seed for reproducibility (default: 123).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of colors
</p>


<h3>References</h3>

<p>See RColorBrewer for more info on palettes
<a href="https://CRAN.R-project.org/package=RColorBrewer">https://CRAN.R-project.org/package=RColorBrewer</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pal &lt;- Single_Color_Palette(pal_color = "reds", num_colors = 7)
PalettePlot(pal= pal)

</code></pre>

<hr>
<h2 id='Split_FeatureScatter'>Deprecated functions <a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a></h2><span id='topic+Split_FeatureScatter'></span><span id='topic+deprecated'></span><span id='topic+Add_Mito_Ribo_Seurat'></span><span id='topic+Add_Mito_Ribo_LIGER'></span><span id='topic+Add_Cell_Complexity_Seurat'></span><span id='topic+Add_Cell_Complexity_LIGER'></span><span id='topic+Meta_Present_LIGER'></span>

<h3>Description</h3>

<p>Use <code><a href="#topic+FeatureScatter_scCustom">FeatureScatter_scCustom()</a></code> instead of <code>Split_FeatureScatter()</code>.
</p>
<p>Use <code><a href="#topic+Add_Mito_Ribo">Add_Mito_Ribo()</a></code> instead of <code>Add_Mito_Ribo_Seurat()</code>.
</p>
<p>Use <code><a href="#topic+Add_Mito_Ribo">Add_Mito_Ribo()</a></code> instead of <code>Add_Mito_Ribo_LIGER()</code>.
</p>
<p>Use <code><a href="#topic+Add_Cell_Complexity">Add_Cell_Complexity()</a></code> instead of <code>Add_Cell_Complexity_Seurat()</code>.
</p>
<p>Use <code><a href="#topic+Add_Cell_Complexity">Add_Cell_Complexity()</a></code> instead of <code>Add_Cell_Complexity_LIGER()</code>.
</p>
<p>Use <code><a href="#topic+Meta_Present">Meta_Present()</a></code> instead of <code>Meta_Present_LIGER()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Split_FeatureScatter(...)

Add_Mito_Ribo_Seurat(...)

Add_Mito_Ribo_LIGER(...)

Add_Cell_Complexity_Seurat(...)

Add_Cell_Complexity_LIGER(...)

Meta_Present_LIGER(...)
</code></pre>

<hr>
<h2 id='Split_Layers'>Split Seurat object into layers</h2><span id='topic+Split_Layers'></span>

<h3>Description</h3>

<p>Split Assay5 of Seurat object into layers by variable in meta.data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Split_Layers(seurat_object, assay = "RNA", split.by)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Split_Layers_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Split_Layers_+3A_assay">assay</code></td>
<td>
<p>name(s) of assays to convert.  Defaults to current active assay.</p>
</td></tr>
<tr><td><code id="Split_Layers_+3A_split.by">split.by</code></td>
<td>
<p>Variable in meta.data to use for splitting layers.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Split object by "treatment"
obj &lt;- Split_Layers(object = obj, assay = "RNA", split.by = "treatment")

## End(Not run)

</code></pre>

<hr>
<h2 id='Split_Vector'>Split vector into list</h2><span id='topic+Split_Vector'></span>

<h3>Description</h3>

<p>Splits vector into chunks of x sizes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Split_Vector(x, chunk_size = 100, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Split_Vector_+3A_x">x</code></td>
<td>
<p>vector to split</p>
</td></tr>
<tr><td><code id="Split_Vector_+3A_chunk_size">chunk_size</code></td>
<td>
<p>size of chunks for vector to be split into, default is 100.</p>
</td></tr>
<tr><td><code id="Split_Vector_+3A_verbose">verbose</code></td>
<td>
<p>logical, print details of vector and split, default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with vector of X length
</p>


<h3>References</h3>

<p>Base code from stackoverflow post:
<a href="https://stackoverflow.com/a/3321659/15568251">https://stackoverflow.com/a/3321659/15568251</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vector &lt;- c("gene1", "gene2", "gene3", "gene4", "gene5", "gene6")

vector_list &lt;- Split_Vector(x = vector, chunk_size = 3)

</code></pre>

<hr>
<h2 id='Stacked_VlnPlot'>Stacked Violin Plot</h2><span id='topic+Stacked_VlnPlot'></span>

<h3>Description</h3>

<p>Code for creating stacked violin plot gene expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Stacked_VlnPlot(
  seurat_object,
  features,
  group.by = NULL,
  split.by = NULL,
  idents = NULL,
  x_lab_rotate = FALSE,
  plot_legend = FALSE,
  colors_use = NULL,
  color_seed = 123,
  ggplot_default_colors = FALSE,
  plot_spacing = 0.15,
  spacing_unit = "cm",
  vln_linewidth = NULL,
  pt.size = NULL,
  raster = NULL,
  add.noise = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Stacked_VlnPlot_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_features">features</code></td>
<td>
<p>Features to plot.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_group.by">group.by</code></td>
<td>
<p>Group (color) cells in different ways (for example, orig.ident).</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_split.by">split.by</code></td>
<td>
<p>A variable to split the violin plots by,</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_idents">idents</code></td>
<td>
<p>Which classes to include in the plot (default is all).</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_x_lab_rotate">x_lab_rotate</code></td>
<td>
<p>logical or numeric.  If logical whether to rotate x-axis labels 45 degrees
(Default is FALSE).  If numeric must be either 45 or 90.  Setting 45 is equivalent to setting TRUE.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_plot_legend">plot_legend</code></td>
<td>
<p>logical.  Adds plot legend containing <code>idents</code> to the returned plot.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_colors_use">colors_use</code></td>
<td>
<p>specify color palette to used in <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.  By default if
number of levels plotted is less than or equal to 36 it will use &quot;polychrome&quot; and if greater than 36
will use &quot;varibow&quot; with shuffle = TRUE both from <code>DiscretePalette_scCustomize</code>.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_plot_spacing">plot_spacing</code></td>
<td>
<p>Numerical value specifying the vertical spacing between each plot in the stack.
Default is 0.15 (&quot;cm&quot;).  Spacing dependent on unit provided to <code>spacing_unit</code>.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_spacing_unit">spacing_unit</code></td>
<td>
<p>Unit to use in specifying vertical spacing between plots.  Default is &quot;cm&quot;.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_vln_linewidth">vln_linewidth</code></td>
<td>
<p>Adjust the linewidth of violin outline.  Must be numeric.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.  Default for <code>Stacked_VlnPlot</code> is 0 to avoid issues with
rendering so many points in vector form.  Alternatively, see <code>raster</code> parameter.</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_add.noise">add.noise</code></td>
<td>
<p>logical, determine if adding a small noise for plotting (Default is TRUE).</p>
</td></tr>
<tr><td><code id="Stacked_VlnPlot_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Author(s)</h3>

<p>Ming Tang (Original Code), Sam Marsh (Wrap single function, added/modified functionality)
</p>


<h3>References</h3>

<p><a href="https://divingintogeneticsandgenomics.rbind.io/post/stacked-violin-plot-for-visualizing-single-cell-data-in-seurat/">https://divingintogeneticsandgenomics.rbind.io/post/stacked-violin-plot-for-visualizing-single-cell-data-in-seurat/</a>
</p>


<h3>See Also</h3>

<p><a href="https://twitter.com/tangming2005">https://twitter.com/tangming2005</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
Stacked_VlnPlot(seurat_object = pbmc_small, features = c("CD3E", "CD8", "GZMB", "MS4A1"),
x_lab_rotate = TRUE)

</code></pre>

<hr>
<h2 id='Store_Misc_Info_Seurat'>Store misc data in Seurat object</h2><span id='topic+Store_Misc_Info_Seurat'></span>

<h3>Description</h3>

<p>Wrapper function save variety of data types to the <code>object@misc</code> slot of Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Store_Misc_Info_Seurat(
  seurat_object,
  data_to_store,
  data_name,
  list_as_list = FALSE,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Store_Misc_Info_Seurat_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Store_Misc_Info_Seurat_+3A_data_to_store">data_to_store</code></td>
<td>
<p>data to be stored in <code style="white-space: pre;">&#8288;@misc&#8288;</code> slot.  Can be single piece of data or list.
If list of data see <code>list_as_list</code> parameter for control over data storage.</p>
</td></tr>
<tr><td><code id="Store_Misc_Info_Seurat_+3A_data_name">data_name</code></td>
<td>
<p>name to give the entry in <code style="white-space: pre;">&#8288;@misc&#8288;</code> slot.  Must be of equal length of the number
of data items being stored.</p>
</td></tr>
<tr><td><code id="Store_Misc_Info_Seurat_+3A_list_as_list">list_as_list</code></td>
<td>
<p>logical.  If <code>data_to_store</code> is a list, this dictates whether to store in <code style="white-space: pre;">&#8288;@misc&#8288;</code> slot
as list (TRUE) or whether to store each entry in the list separately (FALSE).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Store_Misc_Info_Seurat_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical.  Whether to overwrite existing items with the same name.  Default is FALSE, meaning
that function will abort if item with <code>data_name</code> is present in misc slot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat Object with new entries in the <code style="white-space: pre;">&#8288;@misc&#8288;</code> slot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
clu_pal &lt;- c("red", "green", "blue")

pbmc_small &lt;- Store_Misc_Info_Seurat(seurat_object = pbmc_small, data_to_store = clu_pal,
data_name = "rd1_colors")

</code></pre>

<hr>
<h2 id='Store_Palette_Seurat'>Store color palette in Seurat object</h2><span id='topic+Store_Palette_Seurat'></span>

<h3>Description</h3>

<p>Wrapper function around <code>Store_Misc_Info_Seurat</code> to store color palettes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Store_Palette_Seurat(
  seurat_object,
  palette,
  palette_name,
  list_as_list = FALSE,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Store_Palette_Seurat_+3A_seurat_object">seurat_object</code></td>
<td>
<p>object name.</p>
</td></tr>
<tr><td><code id="Store_Palette_Seurat_+3A_palette">palette</code></td>
<td>
<p>vector or list of vectors containing color palettes to store.  If list of palettes
see <code>list_as_list</code> parameter for control over data storage.</p>
</td></tr>
<tr><td><code id="Store_Palette_Seurat_+3A_palette_name">palette_name</code></td>
<td>
<p>name to give the palette(s) in <code style="white-space: pre;">&#8288;@misc&#8288;</code> slot.  Must be of equal length to the number
of data items being stored.</p>
</td></tr>
<tr><td><code id="Store_Palette_Seurat_+3A_list_as_list">list_as_list</code></td>
<td>
<p>logical.  If <code>data_to_store</code> is a list, this dictates whether to store in <code style="white-space: pre;">&#8288;@misc&#8288;</code> slot
as list (TRUE) or whether to store each entry in the list separately (FALSE).  Default is FALSE.</p>
</td></tr>
<tr><td><code id="Store_Palette_Seurat_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical.  Whether to overwrite existing items with the same name.  Default is FALSE, meaning
that function will abort if item with <code>data_name</code> is present in misc slot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat Object with new entries in the <code style="white-space: pre;">&#8288;@misc&#8288;</code> slot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
clu_pal &lt;- c("red", "green", "blue")

pbmc_small &lt;- Store_Misc_Info_Seurat(seurat_object = pbmc_small, data_to_store = clu_pal,
data_name = "rd1_colors")

</code></pre>

<hr>
<h2 id='theme_ggprism_mod'>Modified ggprism theme</h2><span id='topic+theme_ggprism_mod'></span>

<h3>Description</h3>

<p>Modified ggprism theme which restores the legend title.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_ggprism_mod(
  palette = "black_and_white",
  base_size = 14,
  base_family = "sans",
  base_fontface = "bold",
  base_line_size = base_size/20,
  base_rect_size = base_size/20,
  axis_text_angle = 0,
  border = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theme_ggprism_mod_+3A_palette">palette</code></td>
<td>
<p><code>string</code>. Palette name, use
<code>names(ggprism_data$themes)</code> to show all valid palette names.</p>
</td></tr>
<tr><td><code id="theme_ggprism_mod_+3A_base_size">base_size</code></td>
<td>
<p><code>numeric</code>. Base font size, given in <code>"pt"</code>.</p>
</td></tr>
<tr><td><code id="theme_ggprism_mod_+3A_base_family">base_family</code></td>
<td>
<p><code>string</code>. Base font family, default is <code>"sans"</code>.</p>
</td></tr>
<tr><td><code id="theme_ggprism_mod_+3A_base_fontface">base_fontface</code></td>
<td>
<p><code>string</code>. Base font face, default is <code>"bold"</code>.</p>
</td></tr>
<tr><td><code id="theme_ggprism_mod_+3A_base_line_size">base_line_size</code></td>
<td>
<p><code>numeric</code>. Base linewidth for line elements</p>
</td></tr>
<tr><td><code id="theme_ggprism_mod_+3A_base_rect_size">base_rect_size</code></td>
<td>
<p><code>numeric</code>. Base linewidth for rect elements</p>
</td></tr>
<tr><td><code id="theme_ggprism_mod_+3A_axis_text_angle">axis_text_angle</code></td>
<td>
<p><code>integer</code>. Angle of axis text in degrees.
One of: <code style="white-space: pre;">&#8288;0, 45, 90, 270&#8288;</code>.</p>
</td></tr>
<tr><td><code id="theme_ggprism_mod_+3A_border">border</code></td>
<td>
<p><code>logical</code>. Should a border be drawn around the plot?
Clipping will occur unless e.g. <code>coord_cartesian(clip = "off")</code> is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list-like object of class <em>theme</em>.
</p>


<h3>References</h3>

<p>theme is a modified version of <code>theme_prism</code> from ggprism package <a href="https://github.com/csdaw/ggprism">https://github.com/csdaw/ggprism</a>
(License: GPL-3).  Param text is from <code>ggprism:theme_prism()</code> documentation <code><a href="ggprism.html#topic+theme_prism">theme_prism</a></code>.
Theme adaptation based on ggprism vignette
<a href="https://csdaw.github.io/ggprism/articles/themes.html#make-your-own-ggprism-theme-1">https://csdaw.github.io/ggprism/articles/themes.html#make-your-own-ggprism-theme-1</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a plot and customize theme
library(ggplot2)
df &lt;- data.frame(x = rnorm(n = 100, mean = 20, sd = 2), y = rbinom(n = 100, size = 100, prob = 0.2))
p &lt;- ggplot(data = df, mapping = aes(x = x, y = y)) + geom_point(mapping = aes(color = 'red'))
p + theme_ggprism_mod()

</code></pre>

<hr>
<h2 id='Top_Genes_Factor'>Extract top loading genes for LIGER factor</h2><span id='topic+Top_Genes_Factor'></span>

<h3>Description</h3>

<p>Extract vector to the top loading genes for specified LIGER iNMF factor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Top_Genes_Factor(liger_object, liger_factor, num_genes = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Top_Genes_Factor_+3A_liger_object">liger_object</code></td>
<td>
<p>LIGER object name.</p>
</td></tr>
<tr><td><code id="Top_Genes_Factor_+3A_liger_factor">liger_factor</code></td>
<td>
<p>LIGER factor number to pull genes from.</p>
</td></tr>
<tr><td><code id="Top_Genes_Factor_+3A_num_genes">num_genes</code></td>
<td>
<p>number of top loading genes to return as vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A LIGER Object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
top_genes_factor10 &lt;- Top_Genes_Factor(liger_object = object, num_genes = 10)

## End(Not run)

</code></pre>

<hr>
<h2 id='UnRotate_X'>Unrotate x axis on VlnPlot</h2><span id='topic+UnRotate_X'></span>

<h3>Description</h3>

<p>Shortcut for thematic modification to unrotate the x axis (e.g., for Seurat VlnPlot is rotated by default).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UnRotate_X(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UnRotate_X_+3A_...">...</code></td>
<td>
<p>extra arguments passed to <code>ggplot2::theme()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list-like object of class <em>theme</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
p &lt;- VlnPlot(object = pbmc_small, features = "CD3E")
p + UnRotate_X()

</code></pre>

<hr>
<h2 id='Updated_HGNC_Symbols'>Update HGNC Gene Symbols</h2><span id='topic+Updated_HGNC_Symbols'></span>

<h3>Description</h3>

<p>Update human gene symbols using data from HGNC.  This function will store cached data in package directory using (BiocFileCache).  Use of this function requires internet connection on first use (or if setting <code>update_symbol_data = TRUE</code>).  Subsequent use does not require connection and will pull from cached data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Updated_HGNC_Symbols(
  input_data,
  update_symbol_data = NULL,
  case_check_as_warn = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Updated_HGNC_Symbols_+3A_input_data">input_data</code></td>
<td>
<p>Data source containing gene names.  Accepted formats are:
</p>

<ul>
<li> <p><code>charcter vector</code>
</p>
</li>
<li> <p><code>Seurat Objects</code>
</p>
</li>
<li> <p><code>data.frame</code>: genes as rownames
</p>
</li>
<li> <p><code>dgCMatrix/dgTMatrix</code>: genes as rownames
</p>
</li>
<li> <p><code>tibble</code>: genes in first column
</p>
</li></ul>
</td></tr>
<tr><td><code id="Updated_HGNC_Symbols_+3A_update_symbol_data">update_symbol_data</code></td>
<td>
<p>logical, whether to update cached HGNC data, default is NULL.
If <code>NULL</code> BiocFileCache will check and prompt for update if cache is stale.
If <code>FALSE</code> the BiocFileCache stale check will be skipped and current cache will be used.
If <code>TRUE</code> the BiocFileCache stale check will be skipped and HGNC data will be downloaded.</p>
</td></tr>
<tr><td><code id="Updated_HGNC_Symbols_+3A_case_check_as_warn">case_check_as_warn</code></td>
<td>
<p>logical, whether case checking of features should cause abort or
only warn, default is FALSE (abort).  Set to TRUE if atypical names (i.e. old LOC naming) are
present in input_data.</p>
</td></tr>
<tr><td><code id="Updated_HGNC_Symbols_+3A_verbose">verbose</code></td>
<td>
<p>logical, whether to print results detailing numbers of symbols, found, updated,
and not found; default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame containing columns: input_features, Approved_Symbol (already approved; output unchanged), Not_Found_Symbol (symbol not in HGNC; output unchanged), Updated_Symbol (new symbol from HGNC; output updated).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
new_names &lt;- Updated_HGNC_Symbols(input_data = Seurat_Object)

## End(Not run)

</code></pre>

<hr>
<h2 id='Variable_Features_ALL_LIGER'>Perform variable gene selection over whole dataset</h2><span id='topic+Variable_Features_ALL_LIGER'></span>

<h3>Description</h3>

<p>Performs variable gene selection for LIGER object across the entire object instead of by
dataset and then taking union.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Variable_Features_ALL_LIGER(
  liger_object,
  num_genes = NULL,
  var.thresh = 0.3,
  alpha.thresh = 0.99,
  tol = 1e-04,
  do.plot = FALSE,
  pt.size = 0.3,
  chunk = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Variable_Features_ALL_LIGER_+3A_liger_object">liger_object</code></td>
<td>
<p>LIGER object name.</p>
</td></tr>
<tr><td><code id="Variable_Features_ALL_LIGER_+3A_num_genes">num_genes</code></td>
<td>
<p>Number of genes to find. Optimizes the value of <code>var.thresh</code>  to get
this number of genes, (Default is NULL).</p>
</td></tr>
<tr><td><code id="Variable_Features_ALL_LIGER_+3A_var.thresh">var.thresh</code></td>
<td>
<p>Variance threshold. Main threshold used to identify variable genes.
Genes with expression variance greater than threshold (relative to mean) are selected.
(higher threshold -&gt; fewer selected genes).</p>
</td></tr>
<tr><td><code id="Variable_Features_ALL_LIGER_+3A_alpha.thresh">alpha.thresh</code></td>
<td>
<p>Alpha threshold. Controls upper bound for expected mean gene
expression (lower threshold -&gt; higher upper bound). (default 0.99)</p>
</td></tr>
<tr><td><code id="Variable_Features_ALL_LIGER_+3A_tol">tol</code></td>
<td>
<p>Tolerance to use for optimization if num.genes values passed in (default 0.0001).</p>
</td></tr>
<tr><td><code id="Variable_Features_ALL_LIGER_+3A_do.plot">do.plot</code></td>
<td>
<p>Display log plot of gene variance vs. gene expression. Selected genes are
plotted in green. (Default FALSE)</p>
</td></tr>
<tr><td><code id="Variable_Features_ALL_LIGER_+3A_pt.size">pt.size</code></td>
<td>
<p>Point size for plot.</p>
</td></tr>
<tr><td><code id="Variable_Features_ALL_LIGER_+3A_chunk">chunk</code></td>
<td>
<p>size of chunks in hdf5 file. (Default 1000)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A LIGER Object with variable genes in correct slot.
</p>


<h3>References</h3>

<p>Matching function parameter text descriptions are taken from <code>rliger::selectGenes</code>
which is called by this function after creating new temporary object/dataset.
<a href="https://github.com/welch-lab/liger">https://github.com/welch-lab/liger</a>. (License: GPL-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
liger_obj &lt;- Variable_Features_ALL_LIGER(liger_object = liger_obj, num_genes = 2000)

## End(Not run)

</code></pre>

<hr>
<h2 id='VariableFeaturePlot_scCustom'>Custom Labeled Variable Features Plot</h2><span id='topic+VariableFeaturePlot_scCustom'></span>

<h3>Description</h3>

<p>Creates variable features plot with N number of features already labeled by default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VariableFeaturePlot_scCustom(
  seurat_object,
  num_features = 10,
  custom_features = NULL,
  label = TRUE,
  pt.size = 1,
  colors_use = c("black", "red"),
  repel = TRUE,
  y_axis_log = FALSE,
  assay = NULL,
  selection.method = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_num_features">num_features</code></td>
<td>
<p>Number of top variable features to highlight by color/label.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_custom_features">custom_features</code></td>
<td>
<p>A vector of custom feature names to label on plot instead of labeling top
variable genes.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_label">label</code></td>
<td>
<p>logical. Whether to label the top features.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_colors_use">colors_use</code></td>
<td>
<p>colors to use for plotting.  Default is &quot;black&quot; and &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_repel">repel</code></td>
<td>
<p>logical (default TRUE).  Whether or not to repel the feature labels on plot.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_y_axis_log">y_axis_log</code></td>
<td>
<p>logical. Whether to change y axis to log10 scale (Default is FALSE).</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_assay">assay</code></td>
<td>
<p>Assay to pull variable features from.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_selection.method">selection.method</code></td>
<td>
<p>If more then one method use to calculate variable features specify which
method to use for plotting.  See <code>selection.method</code> parameter in <code><a href="Seurat.html#topic+VariableFeaturePlot">VariableFeaturePlot</a></code>
for list of options.</p>
</td></tr>
<tr><td><code id="VariableFeaturePlot_scCustom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VariableFeaturePlot">VariableFeaturePlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
VariableFeaturePlot_scCustom(seurat_object = pbmc_small, num_features = 10)

</code></pre>

<hr>
<h2 id='viridis_plasma_dark_high'>Viridis Shortcuts</h2><span id='topic+viridis_plasma_dark_high'></span><span id='topic+viridis_plasma_light_high'></span><span id='topic+viridis_inferno_dark_high'></span><span id='topic+viridis_inferno_light_high'></span><span id='topic+viridis_magma_dark_high'></span><span id='topic+viridis_magma_light_high'></span><span id='topic+viridis_dark_high'></span><span id='topic+viridis_light_high'></span>

<h3>Description</h3>

<p>Quick shortcuts to access viridis palettes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>viridis_plasma_dark_high

viridis_plasma_light_high

viridis_inferno_dark_high

viridis_inferno_light_high

viridis_magma_dark_high

viridis_magma_light_high

viridis_dark_high

viridis_light_high
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 250.
</p>
<p>An object of class <code>character</code> of length 250.
</p>
<p>An object of class <code>character</code> of length 250.
</p>
<p>An object of class <code>character</code> of length 250.
</p>
<p>An object of class <code>character</code> of length 250.
</p>
<p>An object of class <code>character</code> of length 250.
</p>
<p>An object of class <code>character</code> of length 250.
</p>
<p>An object of class <code>character</code> of length 250.
</p>


<h3>Value</h3>

<p>A color palette for plotting
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
FeaturePlot_scCustom(object = seurat_object, features = "Cx3cr1",
colors_use = viridis_plasma_dark_high, na_color = "lightgray")

## End(Not run)

</code></pre>

<hr>
<h2 id='VlnPlot_scCustom'>VlnPlot with modified default settings</h2><span id='topic+VlnPlot_scCustom'></span>

<h3>Description</h3>

<p>Creates DimPlot with some of the settings modified from their Seurat defaults (colors_use, shuffle, label).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VlnPlot_scCustom(
  seurat_object,
  features,
  colors_use = NULL,
  pt.size = NULL,
  group.by = NULL,
  split.by = NULL,
  plot_median = FALSE,
  plot_boxplot = FALSE,
  median_size = 15,
  idents = NULL,
  num_columns = NULL,
  raster = NULL,
  add.noise = TRUE,
  ggplot_default_colors = FALSE,
  color_seed = 123,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VlnPlot_scCustom_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_features">features</code></td>
<td>
<p>Feature(s) to plot.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_colors_use">colors_use</code></td>
<td>
<p>color palette to use for plotting.  By default if number of levels plotted is less than
or equal to 36 it will use &quot;polychrome&quot; and if greater than 36 will use &quot;varibow&quot; with shuffle = TRUE
both from <code>DiscretePalette_scCustomize</code>.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_pt.size">pt.size</code></td>
<td>
<p>Adjust point size for plotting.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_group.by">group.by</code></td>
<td>
<p>Name of one or more metadata columns to group (color) cells by (for example, orig.ident);
default is the current active.ident of the object.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_split.by">split.by</code></td>
<td>
<p>Feature to split plots by (i.e. &quot;orig.ident&quot;).</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_plot_median">plot_median</code></td>
<td>
<p>logical, whether to plot median for each ident on the plot (Default is FALSE).</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_plot_boxplot">plot_boxplot</code></td>
<td>
<p>logical, whether to plot boxplot inside of violin (Default is FALSE).</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_median_size">median_size</code></td>
<td>
<p>Shape size for the median is plotted.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_idents">idents</code></td>
<td>
<p>Which classes to include in the plot (default is all).</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_num_columns">num_columns</code></td>
<td>
<p>Number of columns in plot layout.  Only valid if <code>split.by != NULL</code>.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_raster">raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_add.noise">add.noise</code></td>
<td>
<p>logical, determine if adding a small noise for plotting (Default is TRUE).</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_ggplot_default_colors">ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 &quot;hue&quot; palette instead of default &quot;polychrome&quot; or &quot;varibow&quot; palettes.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_color_seed">color_seed</code></td>
<td>
<p>random seed for the &quot;varibow&quot; palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td></tr>
<tr><td><code id="VlnPlot_scCustom_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="Seurat.html#topic+VlnPlot">VlnPlot</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>References</h3>

<p>Many of the param names and descriptions are from Seurat to facilitate ease of use as
this is simply a wrapper to alter some of the default parameters <a href="https://github.com/satijalab/seurat/blob/master/R/visualization.R">https://github.com/satijalab/seurat/blob/master/R/visualization.R</a> (License: GPL-3).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Seurat)
VlnPlot_scCustom(seurat_object = pbmc_small, features = "CD3E")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
