<!DOCTYPE html><html><head><title>Help for package SCdeconR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SCdeconR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SCdeconR-package'><p>SCdeconR: Deconvolution of Bulk RNA-Seq Data using Single-Cell RNA-Seq Data as Reference</p></a></li>
<li><a href='#bulk_generator'><p>Generate artificial bulk RNA-seq samples based on simulation</p></a></li>
<li><a href='#celltype_expression'><p>Compute cell type specific gene expression</p></a></li>
<li><a href='#comparedeg_scatter'><p>Generate a scatter plot comparing two differential expression results</p></a></li>
<li><a href='#comparegsea_scatter'><p>Generate a scatter plot comparing two gene set enrichment analysis results</p></a></li>
<li><a href='#compute_metrics'><p>Statistical evaluations of predicted cell proportions</p></a></li>
<li><a href='#construct_ref'><p>Integration of single-cell/nuclei RNA-seq data as reference</p></a></li>
<li><a href='#gsea_heatmap'><p>Heatmap to demonstrate enrichment of selected gene-sets</p></a></li>
<li><a href='#gsea_rwplot'><p>GSEA random-walk plot</p></a></li>
<li><a href='#gsea_sumplot'><p>Summary plot of gene set enrichment analysis</p></a></li>
<li><a href='#load_scdata'><p>Load, filter and normalize scRNA-seq/snRNA-seq data</p></a></li>
<li><a href='#prepare_rnk'><p>Prepare .rnk file for GSEA preranked analysis</p></a></li>
<li><a href='#prop_barplot'><p>Bar plot of cell type proportions across samples</p></a></li>
<li><a href='#reformat_gmt'><p>Methods to manipulate .gmt files</p></a></li>
<li><a href='#run_de'><p>Differential expression analysis</p></a></li>
<li><a href='#scaling'><p>Normalization of gene expression data</p></a></li>
<li><a href='#scdecon'><p>Deconvolution of bulk RNA-seq data</p></a></li>
<li><a href='#transformation'><p>Transformation of gene expression data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Deconvolution of Bulk RNA-Seq Data using Single-Cell RNA-Seq
Data as Reference</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Streamlined workflow from deconvolution of bulk RNA-seq data to downstream differential expression and gene-set enrichment analysis. Provide various visualization functions. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Liuy12/SCdeconR/">https://github.com/Liuy12/SCdeconR/</a>,
<a href="https://liuy12.github.io/SCdeconR/">https://liuy12.github.io/SCdeconR/</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, gtools, harmony, MASS, Matrix, NMF, reshape2,
Biobase, mgcv, grDevices, glmGamPoi</td>
</tr>
<tr>
<td>Suggests:</td>
<td>FARDEEP, reticulate, plotly, DESeq2, knitr, limma, nnls,
preprocessCore, rmarkdown, scater, scran, SingleCellExperiment,
doFuture, future, sctransform, Linnorm</td>
</tr>
<tr>
<td>Depends:</td>
<td>dplyr, edgeR, foreach, ggplot2, R (&ge; 2.10), Seurat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-21 15:14:09 UTC; m182980</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuanhang Liu [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuanhang Liu &lt;liu.yuanhang@mayo.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-22 19:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='SCdeconR-package'>SCdeconR: Deconvolution of Bulk RNA-Seq Data using Single-Cell RNA-Seq Data as Reference</h2><span id='topic+SCdeconR'></span><span id='topic+SCdeconR-package'></span>

<h3>Description</h3>

<p>Streamlined workflow from deconvolution of bulk RNA-seq data to downstream differential expression and gene-set enrichment analysis. Provide various visualization functions.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yuanhang Liu <a href="mailto:liu.yuanhang@mayo.edu">liu.yuanhang@mayo.edu</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/Liuy12/SCdeconR/">https://github.com/Liuy12/SCdeconR/</a>
</p>
</li>
<li> <p><a href="https://liuy12.github.io/SCdeconR/">https://liuy12.github.io/SCdeconR/</a>
</p>
</li></ul>


<hr>
<h2 id='bulk_generator'>Generate artificial bulk RNA-seq samples based on simulation</h2><span id='topic+bulk_generator'></span>

<h3>Description</h3>

<p>Generate artificial bulk RNA-seq samples with random or pre-defined cell-type proportions for benchmarking deconvolution algorithms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bulk_generator(
  ref,
  phenodata,
  num_mixtures = 500,
  num_mixtures_sprop = 10,
  pool_size = 100,
  seed = 1234,
  prop = NULL,
  replace = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bulk_generator_+3A_ref">ref</code></td>
<td>
<p>a matrix-like object of gene expression values with rows representing genes, columns representing cells.</p>
</td></tr>
<tr><td><code id="bulk_generator_+3A_phenodata">phenodata</code></td>
<td>
<p>a data.frame with rows representing cells, columns representing cell attributes. It should at least contain the first two columns as:
</p>

<ol>
<li><p> cell barcodes
</p>
</li>
<li><p> cell types
</p>
</li></ol>
</td></tr>
<tr><td><code id="bulk_generator_+3A_num_mixtures">num_mixtures</code></td>
<td>
<p>total number of simulated bulk samples. Have to be multiple of <code>num_mixtures_sprop</code>. Default to 500.</p>
</td></tr>
<tr><td><code id="bulk_generator_+3A_num_mixtures_sprop">num_mixtures_sprop</code></td>
<td>
<p>number of simulated bulk samples with the same simulated cell type proportions. Only applicable when <code>prop</code> is not specified.
Those samples will be used to estimate bias &amp; variance. Default to 10.</p>
</td></tr>
<tr><td><code id="bulk_generator_+3A_pool_size">pool_size</code></td>
<td>
<p>number of cells to use to construct each artificial bulk sample. Default to 100.</p>
</td></tr>
<tr><td><code id="bulk_generator_+3A_seed">seed</code></td>
<td>
<p>seed to use for simulation. Default to 1234.</p>
</td></tr>
<tr><td><code id="bulk_generator_+3A_prop">prop</code></td>
<td>
<p>a data.frame with two columns. The first column includes unique cell types in phenodata; the second column includes cell type proportions.
If specified, bulk samples will be simulated based on the specified cell proportions.</p>
</td></tr>
<tr><td><code id="bulk_generator_+3A_replace">replace</code></td>
<td>
<p>logical value indicating whether to sample cells with replacement. Default to FALSE, to sample cells without replacement.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>prop</code> is not specified, cell type proportions will be firstly randomly generated with at least two cell types present. Then, for each cell proportion
vector, <code>num_mixtures_sprop</code> number of samples is simulated. Eventually, a total of <code>num_mixtures</code> number of samples is simulated. If prop is
specified, then a total of <code>num_mixtures</code> number of samples will be simulated based on the same cell proportion vector specified.
</p>


<h3>Value</h3>

<p>a list of two objects:
</p>

<ol>
<li><p> simulated bulk RNA-seq data, with rows representing genes, columns representing samples
</p>
</li>
<li><p> cell type proportions used to simulate the bulk RNA-seq data, with rows representing cell types, columns representing samples
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>
ref_list &lt;- c(paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample1"),
              paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample2"))
phenopath1 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample1.txt")
phenopath2 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample2.txt")
phenodata_list &lt;- c(phenopath1,phenopath2)

# construct integrated reference using harmony algorithm
refdata &lt;- construct_ref(ref_list = ref_list,
                      phenodata_list = phenodata_list,
                      data_type = "cellranger",
                      method = "harmony",
                      group_var = "subjectid",
                      nfeature_rna = 50,
                      vars_to_regress = "percent_mt", verbose = FALSE)
phenodata &lt;- data.frame(cellid = colnames(refdata),
                        celltypes = refdata$celltype,
                        subjectid = refdata$subjectid)
prop &lt;- data.frame(celltypes = unique(refdata$celltype), 
proportion = rep(1/length(unique(refdata$celltype)), length(unique(refdata$celltype))))
bulk_sim &lt;- bulk_generator(ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                           phenodata = phenodata,
                           num_mixtures = 20,
                           prop = prop,
                           num_mixtures_sprop = 1)

</code></pre>

<hr>
<h2 id='celltype_expression'>Compute cell type specific gene expression</h2><span id='topic+celltype_expression'></span>

<h3>Description</h3>

<p>Compute cell type specific gene expression based on predicted cell proportions and reference data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>celltype_expression(bulk, ref, phenodata, prop, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="celltype_expression_+3A_bulk">bulk</code></td>
<td>
<p>a matrix-like object of bulk RNA-seq data with rows representing genes, columns representing samples</p>
</td></tr>
<tr><td><code id="celltype_expression_+3A_ref">ref</code></td>
<td>
<p>a matrix-like object of scRNA-seq data with rows representing genes, columns representing cells.</p>
</td></tr>
<tr><td><code id="celltype_expression_+3A_phenodata">phenodata</code></td>
<td>
<p>a data.frame with rows representing cells, columns representing cell attributes. It should at least contain the first two
columns as:
</p>

<ol>
<li><p> cell barcodes
</p>
</li>
<li><p> cell types
</p>
</li></ol>
</td></tr>
<tr><td><code id="celltype_expression_+3A_prop">prop</code></td>
<td>
<p>a matrix-like object of cell proportion values with rows representing cell types, columns representing samples.</p>
</td></tr>
<tr><td><code id="celltype_expression_+3A_...">...</code></td>
<td>
<p>additional parameters passed to <code>create.RCTD</code> from <code>spacexr</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>this function is inspired by cell-type specific gene expression estimation for doublet mode in <code>spacexr</code>. See examples from <code><a href="#topic+run_de">run_de</a></code>.
</p>


<h3>Value</h3>

<p>a list with length equal to number of unique cell types in phenodata. Each element in the list represents gene expression matrix for each unique cell type.
</p>

<hr>
<h2 id='comparedeg_scatter'>Generate a scatter plot comparing two differential expression results</h2><span id='topic+comparedeg_scatter'></span>

<h3>Description</h3>

<p>Generate a scatter plot of fold changes comparing two differential expression results, e.g. w/wo adjusting for cell proportion differences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comparedeg_scatter(
  results1,
  results2,
  result_names = NULL,
  fc_cutoff,
  pval_cutoff,
  pvalflag = TRUE,
  interactive = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comparedeg_scatter_+3A_results1">results1</code></td>
<td>
<p>a data.frame containing differential expression results with five columns: &quot;Gene name&quot;, &quot;log2 fold change&quot;, &quot;log2 average expression&quot;,
&quot;p value&quot;, &quot;adjusted p value&quot;. The second element of the output from function <code><a href="#topic+run_de">run_de</a></code>.</p>
</td></tr>
<tr><td><code id="comparedeg_scatter_+3A_results2">results2</code></td>
<td>
<p>similar to <code>results1</code>.</p>
</td></tr>
<tr><td><code id="comparedeg_scatter_+3A_result_names">result_names</code></td>
<td>
<p>a vector of length 2 indicating the names of the two differential results. If NULL, names will be set to c(&quot;results1&quot;, &quot;results2&quot;)</p>
</td></tr>
<tr><td><code id="comparedeg_scatter_+3A_fc_cutoff">fc_cutoff</code></td>
<td>
<p>fold change cutoff to identify differential expressed genes.</p>
</td></tr>
<tr><td><code id="comparedeg_scatter_+3A_pval_cutoff">pval_cutoff</code></td>
<td>
<p>p value cutoff to identify differential expressed genes.</p>
</td></tr>
<tr><td><code id="comparedeg_scatter_+3A_pvalflag">pvalflag</code></td>
<td>
<p>a logical value indicating whether to use adjusted p value in selecting differential expressed genes.</p>
</td></tr>
<tr><td><code id="comparedeg_scatter_+3A_interactive">interactive</code></td>
<td>
<p>a logical value indicating whether to generate an interactive plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See examples from <code><a href="#topic+run_de">run_de</a></code>.
</p>


<h3>Value</h3>

<p>a <code>ggplot</code> object or <code>plotly</code> object if interactive is set to TRUE
</p>

<hr>
<h2 id='comparegsea_scatter'>Generate a scatter plot comparing two gene set enrichment analysis results</h2><span id='topic+comparegsea_scatter'></span>

<h3>Description</h3>

<p>Generate a scatter plot of normalized enrichment scores comparing two results for gene set enrichment analysis, e.g. w/wo adjusting for cell proportion differences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comparegsea_scatter(
  gseares_path1,
  gseares_path2,
  result_names = NULL,
  nes_cutoff = 2,
  pval_cutoff = 0.1,
  pvalflag = TRUE,
  interactive = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comparegsea_scatter_+3A_gseares_path1">gseares_path1</code></td>
<td>
<p>path to GSEA output.</p>
</td></tr>
<tr><td><code id="comparegsea_scatter_+3A_gseares_path2">gseares_path2</code></td>
<td>
<p>path to a second GSEA output.</p>
</td></tr>
<tr><td><code id="comparegsea_scatter_+3A_result_names">result_names</code></td>
<td>
<p>a vector of length 2 indicating the names of the two GSEA results. If NULL, names will be set to c(&quot;results1&quot;, &quot;results2&quot;)</p>
</td></tr>
<tr><td><code id="comparegsea_scatter_+3A_nes_cutoff">nes_cutoff</code></td>
<td>
<p>normalized enrichment score cutoff to identify enriched gene-sets.</p>
</td></tr>
<tr><td><code id="comparegsea_scatter_+3A_pval_cutoff">pval_cutoff</code></td>
<td>
<p>p value cutoff to identify enriched gene-sets.</p>
</td></tr>
<tr><td><code id="comparegsea_scatter_+3A_pvalflag">pvalflag</code></td>
<td>
<p>a logical value indicating whether to use adjusted p value in selecting enriched gene-sets. Default to TRUE.</p>
</td></tr>
<tr><td><code id="comparegsea_scatter_+3A_interactive">interactive</code></td>
<td>
<p>a logical value indicating whether to generate an interactive plot. Default to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>this function does not support output from GSEA <a href="https://github.com/GSEA-MSigDB/GSEA_R">R implementation</a>
</p>


<h3>Value</h3>

<p>a <code>ggplot</code> object or <code>plotly</code> object if interactive is set to TRUE
</p>

<hr>
<h2 id='compute_metrics'>Statistical evaluations of predicted cell proportions</h2><span id='topic+compute_metrics'></span>

<h3>Description</h3>

<p>Compute RMSE, bias &amp; variance metrics for predicted cell proportions by comparing with expected cell proportions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_metrics(prop_pred, prop_sim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_metrics_+3A_prop_pred">prop_pred</code></td>
<td>
<p>a matrix-like object of predicted cell proportion values with rows representing cell types, columns representing samples.</p>
</td></tr>
<tr><td><code id="compute_metrics_+3A_prop_sim">prop_sim</code></td>
<td>
<p>a matrix-like object of simulated/expected cell proportion values with rows representing cell types, columns representing samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of two objects:
</p>

<ol>
<li><p> a data.fame of summary metrics containing RMSE, bias &amp; variance grouped by cell types and mixture ids (simulated samples with the same expected cell proportions).
</p>
</li>
<li><p> a data.frame of aggregated RMSE values across all cell types within each sample.
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>
## generate artificial bulk samples
ref_list &lt;- c(paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample1"),
              paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample2"))
phenopath1 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample1.txt")
phenopath2 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample2.txt")
phenodata_list &lt;- c(phenopath1,phenopath2)

# construct integrated reference using harmony algorithm
refdata &lt;- construct_ref(ref_list = ref_list,
                      phenodata_list = phenodata_list,
                      data_type = "cellranger",
                      method = "harmony",
                      group_var = "subjectid",
                      nfeature_rna = 50,
                      vars_to_regress = "percent_mt", verbose = FALSE)
phenodata &lt;- data.frame(cellid = colnames(refdata),
                        celltypes = refdata$celltype,
                        subjectid = refdata$subjectid)
prop &lt;- data.frame(celltypes = unique(refdata$celltype), 
proportion = rep(1/length(unique(refdata$celltype)), length(unique(refdata$celltype))))                        
bulk_sim &lt;- bulk_generator(ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                           phenodata = phenodata,
                           num_mixtures = 20,
                           prop = prop,
                           num_mixtures_sprop = 1)

## perform deconvolution based on "OLS" algorithm
decon_res &lt;- scdecon(bulk = bulk_sim[[1]],
                     ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                     phenodata = phenodata,
                     filter_ref = TRUE,
                     decon_method = "OLS",
                     norm_method_sc = "LogNormalize",
                     norm_method_bulk = "TMM",
                     trans_method_sc = "none",
                     trans_method_bulk = "log2",
                     marker_strategy = "all")

## compute metrics
metrics_res &lt;- compute_metrics(decon_res[[1]], bulk_sim[[2]])

</code></pre>

<hr>
<h2 id='construct_ref'>Integration of single-cell/nuclei RNA-seq data as reference</h2><span id='topic+construct_ref'></span>

<h3>Description</h3>

<p>Integration of scRNA-seq or snRNA-seq data using either harmony or seurat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>construct_ref(
  ref_list,
  phenodata_list,
  data_type = c("cellranger", "h5", "matrix"),
  method = c("harmony", "seurat"),
  group_var,
  nfeature_rna = 200,
  percent_mt = 40,
  vars_to_regress = c("percent_mt", "phase"),
  ex_features = NULL,
  cluster = TRUE,
  resolution = 0.8,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="construct_ref_+3A_ref_list">ref_list</code></td>
<td>
<p>a character vector of data paths to scRNA-seq/snRNA-seq. See <code>data_type</code> for accepted data types.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_phenodata_list">phenodata_list</code></td>
<td>
<p>a character vector of data paths to metadata for elements in ref_list. All metadata within phenodata_list should have consistent column names. Columns represent
cell attributes, such as cell type, rows represent cells. Each element in <code>phenodata_list</code> should at least contain the first two columns as:
</p>

<ol>
<li><p> cell barcodes
</p>
</li>
<li><p> cell types
</p>
</li></ol>
</td></tr>
<tr><td><code id="construct_ref_+3A_data_type">data_type</code></td>
<td>
<p>data type of the input scRNA-seq/snRNA-seq data. Could be either a single character value from &quot;cellranger&quot;, &quot;h5&quot;, &quot;matrix&quot;, or
a vector/list of values with the same length as ref_list indicating the data type for each element.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_method">method</code></td>
<td>
<p>character value specifying the method to use. Has to be one of &quot;harmony&quot; or &quot;seurat&quot;. See details for more information.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_group_var">group_var</code></td>
<td>
<p>a vector of character values indicating which variables within phenodata_list metadata to use for integration. Only applicable when method
is set to &quot;harmony&quot;.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_nfeature_rna">nfeature_rna</code></td>
<td>
<p>minimum # of features with non-zero UMIs. Cells with # of features lower than nfeature_rna will be removed. Default to 200.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_percent_mt">percent_mt</code></td>
<td>
<p>maximum percentage of mitochondria (MT) mapped UMIs. Cells with MT percentage higher than percent_mt will be removed. Default to 40.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_vars_to_regress">vars_to_regress</code></td>
<td>
<p>a list of character values indicating the variables to regress for SCTransform normalization step. Default is to regress
out MT percentage (&quot;percent_mt&quot;) &amp; cell cycle effects (&quot;phase&quot;)</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_ex_features">ex_features</code></td>
<td>
<p>a vector of character values indicating genes to exclude from anchor features. Those genes will not be considered as
anchor features for integration, but will still be present in the integrated data.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_cluster">cluster</code></td>
<td>
<p>logical value indicating whether to perform clustering on the integrated data. If TRUE, unsupervised clustering will be performed, and
the results will be saved in &quot;seurat_clusters&quot; metadata in the output Seurat object.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_resolution">resolution</code></td>
<td>
<p>numeric value specifying resolution to use when cluster is set to TRUE.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_verbose">verbose</code></td>
<td>
<p>logical value indicating whether to print messages.</p>
</td></tr>
<tr><td><code id="construct_ref_+3A_...">...</code></td>
<td>
<p>additional parameters passed to <code><a href="Seurat.html#topic+SCTransform">SCTransform</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>data_type can be chosen from:
</p>

<dl>
<dt>cellranger</dt><dd><p>path to a directory containing the matrix.mtx, genes.tsv (or features.tsv), and barcodes.tsv files outputted by 10x's cell-ranger</p>
</dd>
<dt>h5</dt><dd><p>path to .h5 file outputted by 10x's cell-ranger</p>
</dd>
<dt>matrix</dt><dd><p>path to a matrix-like file, with rows representing genes, columns representing cells.</p>
</dd>
</dl>

<p>SCTransform with <code>vst.flavor = "v2"</code> is used for normalization of individual data. Integration methods can be chosen from either &quot;harmony&quot;
or &quot;seurat&quot;. Harmony typically is more memory efficient and, recommended if you have large # of cells for integration.
</p>


<h3>Value</h3>

<p>a <code><a href="Seurat.html#topic+Seurat-class">Seurat-class</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## random subset of two scRNA-seq datasets for breast tissue
ref_list &lt;- c(paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample1"),
              paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample2"))
phenopath1 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample1.txt")
phenopath2 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample2.txt")
phenodata_list &lt;- c(phenopath1,phenopath2)

## Register backend for parallel processing
#doFuture::registerDoFuture()
#future::plan("multisession", workers = 4)

## construct integrated reference data
refdata &lt;- construct_ref(ref_list = ref_list,
                         phenodata_list = phenodata_list,
                         data_type = "cellranger",
                         method = "harmony",
                         group_var = "subjectid",
                         nfeature_rna = 50,
                         vars_to_regress = "percent_mt")

</code></pre>

<hr>
<h2 id='gsea_heatmap'>Heatmap to demonstrate enrichment of selected gene-sets</h2><span id='topic+gsea_heatmap'></span>

<h3>Description</h3>

<p>Heatmap to demonstrate enrichment of selected gene-sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsea_heatmap(
  normdata,
  teststats,
  gmtfile,
  numgenes,
  gsname_up,
  gsname_down,
  anncol,
  color,
  anncolors = NULL,
  rankcol = TRUE,
  zscore_range = c(-3, 3)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsea_heatmap_+3A_normdata">normdata</code></td>
<td>
<p>a matrix-like object of normalized &amp; untransformed bulk RNA-seq data, with rows representing genes and columns representing samples.
The first element of the output from function <code><a href="#topic+run_de">run_de</a></code>.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_teststats">teststats</code></td>
<td>
<p>a data.frame containing differential expression results with five columns: &quot;Gene name&quot;, &quot;log2 fold change&quot;, &quot;log2 average expression&quot;,
&quot;p value&quot;, &quot;adjusted p value&quot;. The second element of the output from function <code><a href="#topic+run_de">run_de</a></code>.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_gmtfile">gmtfile</code></td>
<td>
<p>path to gmt file used for GSEA analysis.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_numgenes">numgenes</code></td>
<td>
<p>Number of genes to include in the heatmap. Will choose <code>numgenes</code> # of top up-regulated genes, as well as <code>numgenes</code> # of down-regulated genes</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_gsname_up">gsname_up</code></td>
<td>
<p>a character value indicating selected up-regulated gene-set.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_gsname_down">gsname_down</code></td>
<td>
<p>a character value indicating selected down-regulated gene-set.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_anncol">anncol</code></td>
<td>
<p>a data.frame of sample meta information to include as column annotation bars. See option annCol from <code><a href="NMF.html#topic+aheatmap">aheatmap</a></code> for more details.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_color">color</code></td>
<td>
<p>color used for heatmap. See option color option from <code><a href="NMF.html#topic+aheatmap">aheatmap</a></code> for details.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_anncolors">anncolors</code></td>
<td>
<p>optional data.frame to define colors for column annotations in <code>anncol</code>.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_rankcol">rankcol</code></td>
<td>
<p>a logical value indicating whether to sort samples based on correlation between fold change &amp; gene expression for better visualization. Default to TRUE.</p>
</td></tr>
<tr><td><code id="gsea_heatmap_+3A_zscore_range">zscore_range</code></td>
<td>
<p>a vector of length two indicating the desired range of z-score transformed data. Default to c(-3, 3).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>this function does not support output from GSEA <a href="https://github.com/GSEA-MSigDB/GSEA_R">R implementation</a>
</p>


<h3>Value</h3>

<p>a heatmap outputted from <code>aheatmap</code> function from <code>NMF</code> package
</p>

<hr>
<h2 id='gsea_rwplot'>GSEA random-walk plot</h2><span id='topic+gsea_rwplot'></span>

<h3>Description</h3>

<p>Generate high-quality GSEA random-walk figures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsea_rwplot(gseares_path, gsname, class_name, metric_range = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsea_rwplot_+3A_gseares_path">gseares_path</code></td>
<td>
<p>path to GSEA output.</p>
</td></tr>
<tr><td><code id="gsea_rwplot_+3A_gsname">gsname</code></td>
<td>
<p>a character value indicating the gene-set name to generate random-walk plot.</p>
</td></tr>
<tr><td><code id="gsea_rwplot_+3A_class_name">class_name</code></td>
<td>
<p>a character value indicating the class of the gene-set, e.g. &quot;GO&quot;.</p>
</td></tr>
<tr><td><code id="gsea_rwplot_+3A_metric_range">metric_range</code></td>
<td>
<p>optional range of the ranking metric.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>this function does not support output from GSEA <a href="https://github.com/GSEA-MSigDB/GSEA_R">R implementation</a>. Scripts initially implemented by Thomas Kuilman.
</p>


<h3>Value</h3>

<p>a random-walk plot for selected gene-set
</p>

<hr>
<h2 id='gsea_sumplot'>Summary plot of gene set enrichment analysis</h2><span id='topic+gsea_sumplot'></span>

<h3>Description</h3>

<p>Summary plot of selected up/down regulated gene-sets for gene set enrichment analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsea_sumplot(
  gseares_path,
  pos_sel,
  neg_sel,
  pvalflag = TRUE,
  interactive = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsea_sumplot_+3A_gseares_path">gseares_path</code></td>
<td>
<p>path to GSEA output.</p>
</td></tr>
<tr><td><code id="gsea_sumplot_+3A_pos_sel">pos_sel</code></td>
<td>
<p>a character vector of upregulated gene-set names.</p>
</td></tr>
<tr><td><code id="gsea_sumplot_+3A_neg_sel">neg_sel</code></td>
<td>
<p>a character vector of downregulated gene-set names.</p>
</td></tr>
<tr><td><code id="gsea_sumplot_+3A_pvalflag">pvalflag</code></td>
<td>
<p>a logical value indicating whether to use adjusted p value in the plot. Default to TRUE.</p>
</td></tr>
<tr><td><code id="gsea_sumplot_+3A_interactive">interactive</code></td>
<td>
<p>a logical value indicating whether to generate an interactive plot. Default to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>this function does not support output from GSEA <a href="https://github.com/GSEA-MSigDB/GSEA_R">R implementation</a>
</p>


<h3>Value</h3>

<p>a <code>ggplot</code> object or <code>plotly</code> object if interactive is set to TRUE
</p>

<hr>
<h2 id='load_scdata'>Load, filter and normalize scRNA-seq/snRNA-seq data</h2><span id='topic+load_scdata'></span>

<h3>Description</h3>

<p>Load and preprocess scRNA-seq/snRNA-seq data using seurat SCTransform workflow.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_scdata(
  ref,
  data_type = c("cellranger", "h5", "matrix"),
  meta_info,
  nfeature_rna = 200,
  percent_mt = 40,
  cc.genes = NULL,
  vars_to_regress = c("percent_mt", "phase"),
  id,
  verbose,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_scdata_+3A_ref">ref</code></td>
<td>
<p>path to scRNA-seq/snRNA-seq data.</p>
</td></tr>
<tr><td><code id="load_scdata_+3A_data_type">data_type</code></td>
<td>
<p>a character value specifying data type of the input scRNA-seq/snRNA-seq data, should be one of &quot;cellranger&quot;, &quot;h5&quot;, &quot;matrix&quot;.</p>
</td></tr>
<tr><td><code id="load_scdata_+3A_meta_info">meta_info</code></td>
<td>
<p>a data.frame with rows representing cells, columns representing cell attributes.</p>
</td></tr>
<tr><td><code id="load_scdata_+3A_nfeature_rna">nfeature_rna</code></td>
<td>
<p>minimum # of features with non-zero UMIs. Cells with # of features lower than nfeature_rna will be removed. Default to 200.</p>
</td></tr>
<tr><td><code id="load_scdata_+3A_percent_mt">percent_mt</code></td>
<td>
<p>maximum percentage of mitochondria (MT) mapped UMIs. Cells with MT percentage higher than percent_mt will be removed. Default to 40.</p>
</td></tr>
<tr><td><code id="load_scdata_+3A_cc.genes">cc.genes</code></td>
<td>
<p>cell-cycle genes curated by Seurat. Can be loaded via <code>data(cc.genes)</code></p>
</td></tr>
<tr><td><code id="load_scdata_+3A_vars_to_regress">vars_to_regress</code></td>
<td>
<p>a list of character values indicating the variables to regress for SCTransform normalization step. Default is to regress
out MT percentage (&quot;percent_mt&quot;) &amp; cell cycle effects (&quot;phase&quot;)</p>
</td></tr>
<tr><td><code id="load_scdata_+3A_id">id</code></td>
<td>
<p>a character value specifying project or sample id. Only used for printing purposes.</p>
</td></tr>
<tr><td><code id="load_scdata_+3A_verbose">verbose</code></td>
<td>
<p>logical value indicating whether to print messages.</p>
</td></tr>
<tr><td><code id="load_scdata_+3A_...">...</code></td>
<td>
<p>additional parameters passed to <code><a href="Seurat.html#topic+SCTransform">SCTransform</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more details, refer to <code><a href="#topic+construct_ref">construct_ref</a></code>
</p>


<h3>Value</h3>

<p>a <code><a href="Seurat.html#topic+Seurat-class">Seurat-class</a></code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
samplepath1 &lt;- paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample1")
samplepath2 &lt;- paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample2")
ref_list &lt;- c(samplepath1, samplepath2)
phenopath1 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample1.txt")
phenopath2 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample2.txt")
phenodata_list &lt;- c(phenopath1,phenopath2)
tmp &lt;- load_scdata(
  ref = ref_list[[1]],
  data_type = c("cellranger"),
  meta_info = data.table::fread(file = phenodata_list[[1]], check.names = FALSE, header = TRUE),
  nfeature_rna = 50,
  vars_to_regress = c("percent_mt"),
  id = 1,
  verbose = TRUE)

</code></pre>

<hr>
<h2 id='prepare_rnk'>Prepare .rnk file for GSEA preranked analysis</h2><span id='topic+prepare_rnk'></span>

<h3>Description</h3>

<p>Prepare .rnk file for GSEA preranked analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_rnk(teststats, outputfile, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_rnk_+3A_teststats">teststats</code></td>
<td>
<p>a data.frame containing differential expression results with five columns: &quot;Gene name&quot;, &quot;log2 fold change&quot;, &quot;log2 average expression&quot;,
&quot;p value&quot;, &quot;adjusted p value&quot;. The second element of the output from function <code><a href="#topic+run_de">run_de</a></code>.</p>
</td></tr>
<tr><td><code id="prepare_rnk_+3A_outputfile">outputfile</code></td>
<td>
<p>full path including file name to .rnk file.</p>
</td></tr>
<tr><td><code id="prepare_rnk_+3A_replace">replace</code></td>
<td>
<p>a logical value indicating whether to replace the output file if it already exists. Default to FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>

<hr>
<h2 id='prop_barplot'>Bar plot of cell type proportions across samples</h2><span id='topic+prop_barplot'></span>

<h3>Description</h3>

<p>Bar plot of cell type proportions across samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prop_barplot(prop, sort = TRUE, interactive = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prop_barplot_+3A_prop">prop</code></td>
<td>
<p>a matrix or data.frame of cell proportion values with rows representing cell types, columns representing samples.</p>
</td></tr>
<tr><td><code id="prop_barplot_+3A_sort">sort</code></td>
<td>
<p>a logical value indicating whether to sort the samples based on cell type with highest median cell proportion across samples. Default to TRUE.</p>
</td></tr>
<tr><td><code id="prop_barplot_+3A_interactive">interactive</code></td>
<td>
<p>a logical value indicating whether to generate interactive plot. Default to FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object or <code>plotly</code> object if interactive is set to TRUE
</p>

<hr>
<h2 id='reformat_gmt'>Methods to manipulate .gmt files</h2><span id='topic+reformat_gmt'></span><span id='topic+read_gmt'></span><span id='topic+write_gmt'></span>

<h3>Description</h3>

<p>Reformat, read &amp; write .gmt file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reformat_gmt(gmtfile, outputfile, replace = FALSE)

read_gmt(gmtfile)

write_gmt(gmt, outputfile, replace = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reformat_gmt_+3A_gmtfile">gmtfile</code></td>
<td>
<p>path to a gene set definition file in .gmt format.</p>
</td></tr>
<tr><td><code id="reformat_gmt_+3A_outputfile">outputfile</code></td>
<td>
<p>full path including file name to export reformatted .gmt file.</p>
</td></tr>
<tr><td><code id="reformat_gmt_+3A_replace">replace</code></td>
<td>
<p>a logical value indicating whether to replace the output file if it already exists. Default to FALSE.</p>
</td></tr>
<tr><td><code id="reformat_gmt_+3A_gmt">gmt</code></td>
<td>
<p>a gmt object returned by <code>read_gmt</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>reformat_gmt</code> replaces blank spaces within the gene-set names to help string-matching methods in downstream plot functions <code><a href="#topic+gsea_sumplot">gsea_sumplot</a></code>
,<code><a href="#topic+gsea_rwplot">gsea_rwplot</a></code>, <code><a href="#topic+gsea_heatmap">gsea_heatmap</a></code>.
</p>


<h3>Value</h3>

<p>for read_gmt, returns a list object with length equal to the total number of gene sets within the .gmt file. Each list contains three elements:
&quot;id&quot;, &quot;name&quot;, &quot;genes&quot;. read_gmt &amp; write_gmt are reimplemented based on functions from <code>ActivePathways</code> package.
</p>

<hr>
<h2 id='run_de'>Differential expression analysis</h2><span id='topic+run_de'></span>

<h3>Description</h3>

<p>Performing differential expression analysis adjusting for cell proportion differences, and other covariates using a additive model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_de(
  bulk,
  prop = NULL,
  sampleinfo,
  control = NULL,
  case = NULL,
  de_method = c("edgeR", "DESeq2", "limma_voom", "limma"),
  padj_method = "BH",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_de_+3A_bulk">bulk</code></td>
<td>
<p>a matrix-like object of gene expression values with rows representing genes, columns representing samples</p>
</td></tr>
<tr><td><code id="run_de_+3A_prop">prop</code></td>
<td>
<p>a matrix-like object of cell proportion values with rows representing cell types, columns representing samples. Default to NULL,
not adjust for cell proportion.</p>
</td></tr>
<tr><td><code id="run_de_+3A_sampleinfo">sampleinfo</code></td>
<td>
<p>a data.frame of metadata for the samples. Rows represents samples; columns represents covariates to adjust for. The first column of
sampleinfo should contains group information for differential analysis.</p>
</td></tr>
<tr><td><code id="run_de_+3A_control">control</code></td>
<td>
<p>a character value indicating the control group in sampleinfo. Set to NULL to perform ANOVA-like analysis.</p>
</td></tr>
<tr><td><code id="run_de_+3A_case">case</code></td>
<td>
<p>a character value indicating the case group in sampleinfo. Set to NULL to perform ANOVA-like analysis.</p>
</td></tr>
<tr><td><code id="run_de_+3A_de_method">de_method</code></td>
<td>
<p>a character value indicating the method to use for testing differential expression. Should be one of &quot;edgeR&quot;, &quot;DESeq2&quot;, &quot;limma_voom&quot;, &quot;limma&quot;</p>
</td></tr>
<tr><td><code id="run_de_+3A_padj_method">padj_method</code></td>
<td>
<p>method for adjusting multiple hypothesis testing. Default to &quot;BH&quot;. See <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for more details.</p>
</td></tr>
<tr><td><code id="run_de_+3A_...">...</code></td>
<td>
<p>parameters pass to DE methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To perform ANOVA like analysis (differences between any groups), set <code>control</code> &amp; <code>case</code> options to <code>NULL</code> and choose one of the following methods:
edgeR, limma_voom or limma. DESeq2 does not provide direct support for this type of comparison.
</p>


<h3>Value</h3>

<p>a list of two elements:
</p>

<ol>
<li><p> a data.frame containing normalized gene expression data.
</p>
</li>
<li><p> a data.frame containing detailed differential expression statistics. Columns represent &quot;Gene name&quot;, &quot;log2 fold change&quot;, &quot;log2 average expression&quot;,
&quot;p value&quot;, &quot;adjusted p value&quot; respectively
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>
ref_list &lt;- c(paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample1"),
              paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample2"))
phenopath1 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample1.txt")
phenopath2 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample2.txt")
phenodata_list &lt;- c(phenopath1,phenopath2)

# construct integrated reference using harmony algorithm
refdata &lt;- construct_ref(ref_list = ref_list,
                      phenodata_list = phenodata_list,
                      data_type = "cellranger",
                      method = "harmony",
                      group_var = "subjectid",
                      nfeature_rna = 50,
                      vars_to_regress = "percent_mt", verbose = FALSE)
phenodata &lt;- data.frame(cellid = colnames(refdata),
                        celltypes = refdata$celltype,
                        subjectid = refdata$subjectid)
## construct a vector with same proportions across cell types
prop1 &lt;- data.frame(celltypes = unique(refdata$celltype),
                   proportion = rep(0.125, 8))
## simulate 20 bulk samples based on specified cell type proportion
bulk_sim1 &lt;- bulk_generator(ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                            phenodata = phenodata,
                            num_mixtures = 20,
                            prop = prop1, replace = TRUE)
## generate another vector with high proportion for a certian cell type
prop2 &lt;- data.frame(celltypes = unique(refdata$celltype),
                    proportion = c(0.8, 0.1, 0.1, rep(0, 5)))
bulk_sim2 &lt;- bulk_generator(ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                            phenodata = phenodata,
                            num_mixtures = 20,
                            prop = prop2, replace = TRUE)
## compare data for differential analysis
bulk_sim &lt;- list(cbind(bulk_sim1[[1]], bulk_sim2[[1]]),
                 cbind(bulk_sim1[[2]], bulk_sim2[[2]]))
## force to be integer for DE purposes
bulk &lt;- round(bulk_sim[[1]], digits=0)
colnames(bulk) &lt;- paste0("sample", 1:ncol(bulk))
## predict cell type proportions using "OLS" algorithm
decon_res &lt;- scdecon(bulk = bulk,
                     ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                     phenodata = phenodata,
                     filter_ref = TRUE,
                     decon_method = "OLS",
                     norm_method_sc = "LogNormalize",
                     norm_method_bulk = "TMM",
                     trans_method_sc = "none",
                     trans_method_bulk = "log2",
                     marker_strategy = "all")
## create sampleinfo
sampleinfo &lt;- data.frame(condition = rep(c("group1", "group2"), each =20))
row.names(sampleinfo) &lt;- colnames(bulk)
deres &lt;- run_de(bulk = bulk,
               prop = decon_res[[1]],
               sampleinfo = sampleinfo,
               control = "group1",
               case = "group2",
               de_method = "edgeR")

 ## run differential analysis without adjusting for cell proportion differences
 deres_notadjust &lt;- run_de(bulk = bulk,
                           prop = NULL,
                           sampleinfo = sampleinfo,
                           control = "group1",
                           case = "group2",
                           de_method = "edgeR")

## scatter plot to compare the effect of adjusting cell proportion differences
comparedeg_scatter(results1 = deres[[2]],
                   results2 = deres_notadjust[[2]],
                   result_names = c("adjust for cell proportion", "not adjust for cell proportion"),
                   fc_cutoff = 1.5,
                   pval_cutoff = 0.05,
                   pvalflag = TRUE,
                   interactive = FALSE)

## generate cell-type specific gene expression
ct_exprs_list &lt;- celltype_expression(bulk = bulk,
                                     ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                                     phenodata = phenodata,
                                     prop = decon_res[[1]],
                                     UMI_min = 0,
                                     CELL_MIN_INSTANCE = 1)



</code></pre>

<hr>
<h2 id='scaling'>Normalization of gene expression data</h2><span id='topic+scaling'></span>

<h3>Description</h3>

<p>Methods to use for data normalization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scaling(
  matrix,
  option,
  gene_length = NULL,
  seed = 1234,
  ffpe_artifacts = FALSE,
  model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scaling_+3A_matrix">matrix</code></td>
<td>
<p>a matrix-like objector of gene expression values with rows representing genes, columns representing samples or cells</p>
</td></tr>
<tr><td><code id="scaling_+3A_option">option</code></td>
<td>
<p>character value specifying the normalization method to use. Has to be one of &quot;none&quot;, &quot;LogNormalize&quot;, &quot;TMM&quot;, &quot;median_ratios&quot;, &quot;TPM&quot;,
&quot;SCTransform&quot;, &quot;scran&quot;, &quot;scater&quot;, &quot;Linnorm&quot;.</p>
</td></tr>
<tr><td><code id="scaling_+3A_gene_length">gene_length</code></td>
<td>
<p>a data.frame with two columns. The first column represents gene names that match with provided bulk data. The second column
represents length of each gene. Only applicable when norm_method is selected as &quot;TPM&quot;</p>
</td></tr>
<tr><td><code id="scaling_+3A_seed">seed</code></td>
<td>
<p>random seed used for simulating FFPE artifacts. Only applicable when ffpe_artifacts is set to TRUE.</p>
</td></tr>
<tr><td><code id="scaling_+3A_ffpe_artifacts">ffpe_artifacts</code></td>
<td>
<p>logical value indicating whether to add simulated ffpe artifacts in the bulk data. Only applicable to simulation experiments in
evaluating the effect of FFPE artifacts.</p>
</td></tr>
<tr><td><code id="scaling_+3A_model">model</code></td>
<td>
<p>pre-constructed ffpe model data. Can be downloaded from github: https://github.com/Liuy12/SCdeconR/blob/master/data/ffpemodel.rda</p>
</td></tr>
</table>


<h3>Details</h3>

<p>refer to <code><a href="#topic+scdecon">scdecon</a></code> for more details.
</p>


<h3>Value</h3>

<p>a matrix-like object with the same dimension of input object after data normalization.
</p>

<hr>
<h2 id='scdecon'>Deconvolution of bulk RNA-seq data</h2><span id='topic+scdecon'></span>

<h3>Description</h3>

<p>Deconvolution of bulk RNA-seq data based on single-cell reference data. Eight bulk deconvolution methods, along with eight normalization
methods and four transformation methods are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scdecon(
  bulk,
  ref,
  phenodata,
  filter_ref = TRUE,
  marker_genes = NULL,
  genes_to_remove = NULL,
  min_pct_ct = 0.05,
  decon_method = c("scaden", "CIBERSORT", "OLS", "nnls", "FARDEEP", "RLR", "MuSiC",
    "SCDC", "scTAPE"),
  norm_method_sc = c("none", "LogNormalize", "SCTransform", "scran", "scater", "Linnorm"),
  norm_method_bulk = c("none", "TMM", "median_ratios", "TPM"),
  trans_method_sc = c("none", "log2", "sqrt", "vst"),
  trans_method_bulk = c("none", "log2", "sqrt", "vst"),
  gene_length = NULL,
  lfc_markers = log2(1.5),
  marker_strategy = c("all", "pos_fc", "top_50p_logFC", "top_50p_AveExpr"),
  to_remove = NULL,
  ffpe_artifacts = FALSE,
  model = NULL,
  prop = NULL,
  cibersortpath = NULL,
  pythonpath = NULL,
  tmpdir = NULL,
  remove_tmpdir = TRUE,
  seed = 1234,
  nsamples = 1000,
  return_value_only = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scdecon_+3A_bulk">bulk</code></td>
<td>
<p>a matrix or data.frame of unnormalizaed &amp; untransformed bulk RNA-seq gene expression values with rows representing genes, columns
representing samples</p>
</td></tr>
<tr><td><code id="scdecon_+3A_ref">ref</code></td>
<td>
<p>a matrix or data.frame of untransformed scRNA-seq gene expression counts with rows representing genes, columns representing cells.
This data will be used to deconvolute provided bulk RNA-seq data.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_phenodata">phenodata</code></td>
<td>
<p>a data.frame with rows representing cells, columns representing cell attributes. It should at least contain the first three
columns as:
</p>

<ol>
<li><p> cell barcodes
</p>
</li>
<li><p> cell types
</p>
</li>
<li><p> subject ids
</p>
</li></ol>
</td></tr>
<tr><td><code id="scdecon_+3A_filter_ref">filter_ref</code></td>
<td>
<p>logical value indicating whether outlier genes &amp; cells should be removed from the provided reference data. Defaults to TRUE</p>
</td></tr>
<tr><td><code id="scdecon_+3A_marker_genes">marker_genes</code></td>
<td>
<p>a data.frame of two columns. First column represents cell types in ref; second column represents gene names of marker genes. If specified,
those genes will be used to construct signature matrix for mark-gene based deconvolution methods, such as CIBERSORT, OLS, nnls, FARDEEP and RLR. Default to NULL,
carry out differential analysis to identify marker genes for each cell type in ref.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_genes_to_remove">genes_to_remove</code></td>
<td>
<p>a vector of gene names to remove from the reference scRNA-seq data. Default to NULL.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_min_pct_ct">min_pct_ct</code></td>
<td>
<p>a numeric value indicating the minimum required proportion of expressing cells per cell type for marker gene identification. Only applicable when marker_genes
is NULL. Default to 0.05.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_decon_method">decon_method</code></td>
<td>
<p>character value specifying the deconvolution method to use. Has to be one of &quot;scaden&quot;, &quot;CIBERSORT&quot;, &quot;OLS&quot;, &quot;nnls&quot;, &quot;FARDEEP&quot;, &quot;RLR&quot;,
&quot;MuSiC&quot;, &quot;SCDC&quot;, &quot;scTAPE&quot;. See details for more information.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_norm_method_sc">norm_method_sc</code></td>
<td>
<p>character value specifying the normalization method to use for reference data. Has to be one of &quot;none&quot;,&quot;LogNormalize&quot;,
&quot;SCTransform&quot;, &quot;scran&quot;, &quot;scater&quot;, &quot;Linnorm&quot;. See details for more information.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_norm_method_bulk">norm_method_bulk</code></td>
<td>
<p>character value specifying the normalization method to use for bulk data. Has to be one of &quot;none&quot;, &quot;TMM&quot;,
&quot;median_ratios&quot;, &quot;TPM&quot;. See details for more information.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_trans_method_sc">trans_method_sc</code></td>
<td>
<p>character value specifying the transformation method to use for both bulk &amp; reference data. Has to be one of &quot;none&quot;, &quot;log2&quot;, &quot;sqrt&quot;,
&quot;vst&quot;. See details for more information.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_trans_method_bulk">trans_method_bulk</code></td>
<td>
<p>character value specifying the transformation method to use for both bulk &amp; reference data. Has to be one of &quot;none&quot;, &quot;log2&quot;, &quot;sqrt&quot;,
&quot;vst&quot;. See details for more information.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_gene_length">gene_length</code></td>
<td>
<p>a data.frame with two columns. The first column represents gene names that match with provided bulk data. The second column
represents length of each gene. Only applicable when norm_method is selected as &quot;TPM&quot;.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_lfc_markers">lfc_markers</code></td>
<td>
<p>log2 fold change cutoff used to identify marker genes for deconvolution. The option only applicable to marker-gene based
approaches, such as CIBERSORT, OLS, nnls, FARDEEP and RLR. Only applicable when marker_genes is NULL.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_marker_strategy">marker_strategy</code></td>
<td>
<p>further strategy in selecting marker genes besides applying the log2 fold change cutoff. Can be chosen from: &quot;all&quot;, &quot;pos_fc&quot;,
&quot;top_50p_logFC&quot; or &quot;top_50p_AveExpr&quot;. See details for more information. Only applicable when marker_genes is NULL.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_to_remove">to_remove</code></td>
<td>
<p>character value representing the cell type to remove from reference data. Only applicable to simulation experiments in evaluating
effect of cell type removal from reference.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_ffpe_artifacts">ffpe_artifacts</code></td>
<td>
<p>logical value indicating whether to add simulated ffpe artifacts in the bulk data. Only applicable to simulation experiments in
evaluating the effect of FFPE artifacts.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_model">model</code></td>
<td>
<p>pre-constructed ffpe model data. Can be downloaded from github: https://github.com/Liuy12/SCdeconR/blob/master/data/ffpemodel.rda.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_prop">prop</code></td>
<td>
<p>a matrix or data.frame of simulated cell proportion values with rows representing cell types, columns representing samples. Only applicable to simulation
experiments in evaluating the effect of cell type removal from reference.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_cibersortpath">cibersortpath</code></td>
<td>
<p>full path to CIBERSORT.R script.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_pythonpath">pythonpath</code></td>
<td>
<p>full path to python binary where scaden was installed with.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_tmpdir">tmpdir</code></td>
<td>
<p>temporary processing directory for scaden or scTAPE.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_remove_tmpdir">remove_tmpdir</code></td>
<td>
<p>a logical value indicating whether to remove tmpdir once scaden is completed. Default to TRUE.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_seed">seed</code></td>
<td>
<p>random seed used for simulating FFPE artifacts. Only applicable when ffpe_artifacts is set to TRUE.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_nsamples">nsamples</code></td>
<td>
<p>number of artificial bulk samples to simulate for scaden. Default to 1000.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_return_value_only">return_value_only</code></td>
<td>
<p>return a list of values only without performing deconvolution. This could be helpful in
cases where the user want to apply their own deconvolution algorithms. Default to FALSE.</p>
</td></tr>
<tr><td><code id="scdecon_+3A_verbose">verbose</code></td>
<td>
<p>a logical value indicating whether to print messages. Default to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>decon_method should be one of the following:
</p>

<dl>
<dt><a href="https://github.com/KevinMenden/scaden">scaden</a></dt><dd><p>a deep learning based method using three multi-layer deep neural nets. To use scaden,
you need to firstly install scaden via <code>pip</code> or <code>conda</code>, the provide the python path to <code>pythonpath</code> option.</p>
</dd>
<dt>CIBERSORT</dt><dd><p>a marker gene based support vectors regression approach. CIBERSOR does not allow redistribution. To use CIBERSORT,
you need to request the source code from the authors &amp; provide the path of CIBERSORT.R script to <code>cibersortpath</code> option.</p>
</dd>
<dt>OLS</dt><dd><p>ordinary least squares.</p>
</dd>
<dt><a href="https://CRAN.R-project.org/package=nnls">nnls</a></dt><dd><p>non-negative least squares.</p>
</dd>
<dt><a href="https://CRAN.R-project.org/package=FARDEEP">FARDEEP</a></dt><dd><p>robust regression using least trimmed squares</p>
</dd>
<dt><a href="https://CRAN.R-project.org/package=MASS">RLR</a></dt><dd><p>robust regression using an M estimator</p>
</dd>
<dt><a href="https://github.com/xuranw/MuSiC">MuSiC</a></dt><dd><p>multi-subject single-cell deconvolution</p>
</dd>
<dt><a href="https://github.com/meichendong/SCDC">SCDC</a></dt><dd><p>an ENSEMBLE method to integrate deconvolution results from different scRNA-seq datasets</p>
</dd>
<dt><a href="https://github.com/poseidonchan/TAPE">scTAPE</a></dt><dd><p>Deep autoencoder based deconvolution</p>
</dd>
</dl>

<p>norm_method should be one of the following:
</p>

<dl>
<dt>none</dt><dd><p>no normalization is performed.</p>
</dd>
<dt>LogNormalize</dt><dd><p><code><a href="Seurat.html#topic+LogNormalize">LogNormalize</a></code> method from seurat.</p>
</dd>
<dt>TMM</dt><dd><p>TMM method from <code><a href="edgeR.html#topic+calcNormFactors">calcNormFactors</a></code> function from edgeR.</p>
</dd>
<dt>median_ratios</dt><dd><p>median ratio method from <code><a href="DESeq2.html#topic+estimateSizeFactors+2CDESeqDataSet-method">estimateSizeFactors,DESeqDataSet-method</a></code> function from DESeq2.</p>
</dd>
<dt>TPM</dt><dd><p>Transcript per million. TPM has to be chosen if ffpe_artifacts is set to TRUE.</p>
</dd>
<dt>SCTransform</dt><dd><p><code><a href="Seurat.html#topic+SCTransform">SCTransform</a></code> method from Seurat.</p>
</dd>
<dt>scran</dt><dd><p><code><a href="scran.html#topic+computeSumFactors">computeSumFactors</a></code> method from scran.</p>
</dd>
<dt>scater</dt><dd><p><code><a href="scater.html#topic+librarySizeFactors">librarySizeFactors</a></code> method from scater.</p>
</dd>
<dt>Linnorm</dt><dd><p><code>Linnorm</code> method from Linnorm.</p>
</dd>
</dl>

<p>trans_method should be one of the following:
</p>

<dl>
<dt>none</dt><dd><p>no transformation is performed.</p>
</dd>
<dt>log2</dt><dd><p>log2 transformation. 0.1 is added to the data to avoid logarithm of 0s.</p>
</dd>
<dt>sqrt</dt><dd><p>square root transformation.</p>
</dd>
<dt>vst</dt><dd><p><code><a href="DESeq2.html#topic+varianceStabilizingTransformation">varianceStabilizingTransformation</a></code> method from DESeq2.</p>
</dd>
</dl>

<p>marker_strategy should be one of the following:
</p>

<dl>
<dt>all</dt><dd><p>all genes passed fold change threshold will be used.</p>
</dd>
<dt>pos_fc</dt><dd><p>only genes with positive fold changes will be used.</p>
</dd>
<dt>top_50p_logFC</dt><dd><p>only genes with top 50 percent positive fold changes will be used.</p>
</dd>
<dt>top_50p_AveExpr</dt><dd><p>only genes with top 50 percent average expression will be used.</p>
</dd>
</dl>



<h3>Value</h3>

<p>a list containing two or four elements.
</p>

<dl>
<dt>first element</dt><dd><p>a data.frame of predicted cell-type proportions, with rows representing cell types, columns representing samples.</p>
</dd>
<dt>second element</dt><dd><p>a data.frame of fitting errors of the algorithm; first column represents sample names, second column represents RMSEs.</p>
</dd>
<dt>optional third element</dt><dd><p>a data.frame of simulated cell proportion after removing the specified cell_type. Only applicable to simulation experiments.</p>
</dd>
<dt>optional fourth element</dt><dd><p>a data.frame of marker genes used for deconvolution. Only applicable to marker-gene based deconvolution methods.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
ref_list &lt;- c(paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample1"),
              paste0(system.file("extdata", package = "SCdeconR"), "/refdata/sample2"))
phenopath1 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample1.txt")
phenopath2 &lt;- paste0(system.file("extdata", package = "SCdeconR"),
"/refdata/phenodata_sample2.txt")
phenodata_list &lt;- c(phenopath1,phenopath2)

# construct integrated reference using harmony algorithm
refdata &lt;- construct_ref(ref_list = ref_list,
                      phenodata_list = phenodata_list,
                      data_type = "cellranger",
                      method = "harmony",
                      group_var = "subjectid",
                      nfeature_rna = 50,
                      vars_to_regress = "percent_mt", verbose = FALSE)

phenodata &lt;- data.frame(cellid = colnames(refdata),
                        celltypes = refdata$celltype,
                        subjectid = refdata$subjectid)
                        
prop &lt;- data.frame(celltypes = unique(refdata$celltype), 
proportion = rep(1/length(unique(refdata$celltype)), length(unique(refdata$celltype))))
bulk_sim &lt;- bulk_generator(ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                           phenodata = phenodata,
                           num_mixtures = 20,
                           prop = prop,
                           num_mixtures_sprop = 1)

## perform deconvolution based on "OLS" algorithm
decon_res &lt;- scdecon(bulk = bulk_sim[[1]],
                     ref = GetAssayData(refdata, slot = "data", assay = "SCT"),
                     phenodata = phenodata,
                     filter_ref = TRUE,
                     decon_method = "OLS",
                     norm_method_sc = "LogNormalize",
                     norm_method_bulk = "TMM",
                     trans_method_sc = "none",
                     trans_method_bulk = "log2",
                     marker_strategy = "all")

</code></pre>

<hr>
<h2 id='transformation'>Transformation of gene expression data</h2><span id='topic+transformation'></span>

<h3>Description</h3>

<p>Methods to use for data transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transformation(matrix, option)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transformation_+3A_matrix">matrix</code></td>
<td>
<p>a matrix-like objector of gene expression values with rows representing genes, columns representing samples or cells</p>
</td></tr>
<tr><td><code id="transformation_+3A_option">option</code></td>
<td>
<p>character value specifying the transformation method to use. Has to be one of &quot;none&quot;, &quot;log&quot;, &quot;sqrt&quot;, &quot;vst&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>refer to <code><a href="#topic+scdecon">scdecon</a></code> for more details.
</p>


<h3>Value</h3>

<p>a matrix-like object with the same dimension of input object after data transformation.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
