<!DOCTYPE html><html><head><title>Help for package scDHA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scDHA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Goolam'><p>Goolam</p></a></li>
<li><a href='#Goolam_result'><p>Goolam_result</p></a></li>
<li><a href='#scDHA'><p>scDHA</p></a></li>
<li><a href='#scDHA.class'><p>scDHA classification</p></a></li>
<li><a href='#scDHA.pt'><p>scDHA pseudo time inference</p></a></li>
<li><a href='#scDHA.vis'><p>scDHA visulization</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single-Cell Decomposition using Hierarchical Autoencoder</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ha Nguyen &lt;hvn0006@auburn.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a fast and accurate pipeline for single-cell analyses. 
    The 'scDHA' software package can perform clustering, dimension reduction and visualization, classification, and time-trajectory inference on single-cell data (Tran et.al. (2021) &lt;<a href="https://doi.org/10.1038%2Fs41467-021-21312-2">doi:10.1038/s41467-021-21312-2</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>matrixStats, foreach, doParallel, igraph, Matrix, uwot,
cluster, Rcpp, RcppParallel, RcppAnnoy, methods, torch (&ge;
0.3.0), RhpcBLASctl, coro</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppParallel, RcppAnnoy</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, mclust</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/duct317/scDHA">https://github.com/duct317/scDHA</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/duct317/scDHA/issues">https://github.com/duct317/scDHA/issues</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-02 21:03:58 UTC; hanguyen</td>
</tr>
<tr>
<td>Author:</td>
<td>Ha Nguyen [cre],
  Duc Tran [aut],
  Tin Nguyen [fnd]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-02 22:42:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='Goolam'>Goolam</h2><span id='topic+Goolam'></span>

<h3>Description</h3>

<p>Goolam dataset in list format, include scRNA-seq data and cell type information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Goolam
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='Goolam_result'>Goolam_result</h2><span id='topic+Goolam_result'></span>

<h3>Description</h3>

<p>Result of processing Goolam dataset using 'scDHA' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Goolam_result
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 4.
</p>

<hr>
<h2 id='scDHA'>scDHA</h2><span id='topic+scDHA'></span>

<h3>Description</h3>

<p>The main function to perform dimension deduction and clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scDHA(
  data = data,
  k = NULL,
  method = "scDHA",
  sparse = FALSE,
  n = 5000,
  ncores = 10L,
  gen_fil = TRUE,
  do.clus = TRUE,
  sample.prob = NULL,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scDHA_+3A_data">data</code></td>
<td>
<p>Gene expression matrix, with rows represent samples and columns represent genes.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_k">k</code></td>
<td>
<p>Number of clusters, leave as default for auto detection. Has no effect when <code>do.clus = False</code>.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_method">method</code></td>
<td>
<p>Method used for clustering. It can be &quot;scDHA&quot; or &quot;louvain&quot;. The default setting is &quot;scDHA&quot;.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_sparse">sparse</code></td>
<td>
<p>Boolen variable indicating whether data is a sparse matrix. The input must be a non negative sparse matrix.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_n">n</code></td>
<td>
<p>Number of genes to keep after feature selection step.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_ncores">ncores</code></td>
<td>
<p>Number of processor cores to use.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_gen_fil">gen_fil</code></td>
<td>
<p>Boolean variable indicating whether to perform scDHA gene filtering before performing dimension deduction and clustering.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_do.clus">do.clus</code></td>
<td>
<p>Boolean variable indicating whether to perform scDHA clustering. If <code>do.clus = False</code>, only dimension deduction is performed.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_sample.prob">sample.prob</code></td>
<td>
<p>Probability used for classification application only. Leave this parameter as default, no user input is required.</p>
</td></tr>
<tr><td><code id="scDHA_+3A_seed">seed</code></td>
<td>
<p>Seed for reproducibility.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with the following keys:
</p>

<ul>
<li><p> cluster - A numeric vector containing cluster assignment for each sample. If <code>do.clus = False</code>, this values is always <code>NULL</code>.
</p>
</li>
<li><p> latent - A matrix representing compressed data from the input data, with rows represent samples and columns represent latent variables.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library(scDHA)
#Load example data (Goolam dataset)
data('Goolam'); data &lt;- t(Goolam$data); label &lt;- as.character(Goolam$label)
#Log transform the data 
data &lt;- log2(data + 1)
if(torch::torch_is_installed()) #scDHA need libtorch installed
{
  #Generate clustering result, the input matrix has rows as samples and columns as genes
  result &lt;- scDHA(data, ncores = 2, seed = 1)
  #The clustering result can be found here 
  cluster &lt;- result$cluster
}

</code></pre>

<hr>
<h2 id='scDHA.class'>scDHA classification</h2><span id='topic+scDHA.class'></span>

<h3>Description</h3>

<p>Perform classification of new data based on available data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scDHA.class(
  train = train,
  train.label = train.label,
  test = test,
  ncores = 10L,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scDHA.class_+3A_train">train</code></td>
<td>
<p>Expression matrix of available data, with rows represent samples and columns represent genes.</p>
</td></tr>
<tr><td><code id="scDHA.class_+3A_train.label">train.label</code></td>
<td>
<p>A vector containing label for each sample in training data.</p>
</td></tr>
<tr><td><code id="scDHA.class_+3A_test">test</code></td>
<td>
<p>Expression matrix new data for classification, with rows represent samples and columns represent genes.</p>
</td></tr>
<tr><td><code id="scDHA.class_+3A_ncores">ncores</code></td>
<td>
<p>Number of processor cores to use.</p>
</td></tr>
<tr><td><code id="scDHA.class_+3A_seed">seed</code></td>
<td>
<p>Seed for reproducibility.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector contain classified labels for new data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scDHA)
#Load example data (Goolam dataset)
data('Goolam'); data &lt;- t(Goolam$data); label &lt;- as.character(Goolam$label)
#Log transform the data 
data &lt;- log2(data + 1)
#Split data into training and testing sets
set.seed(1)
idx &lt;- sample.int(nrow(data), size = round(nrow(data)*0.75))
train.x &lt;- data[idx, ]; train.y &lt;- label[idx]
test.x &lt;- data[-idx, ]; test.y &lt;- label[-idx]
if(torch::torch_is_installed()) #scDHA need libtorch installed
{
  #Predict the labels of cells in testing set
  prediction &lt;- scDHA.class(train = train.x, train.label = train.y, test = test.x, 
                            ncores = 2, seed = 1)
  #Calculate accuracy of the predictions
  acc &lt;- round(sum(test.y == prediction)/length(test.y), 2)
  print(paste0("Accuracy = ", acc))
}

</code></pre>

<hr>
<h2 id='scDHA.pt'>scDHA pseudo time inference</h2><span id='topic+scDHA.pt'></span>

<h3>Description</h3>

<p>Inferring pseudo-time data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scDHA.pt(sc = sc, start.point = 1, ncores = 10L, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scDHA.pt_+3A_sc">sc</code></td>
<td>
<p>Embedding object, produced by <code>scDHA</code> function.</p>
</td></tr>
<tr><td><code id="scDHA.pt_+3A_start.point">start.point</code></td>
<td>
<p>Starting point of the trajectory.</p>
</td></tr>
<tr><td><code id="scDHA.pt_+3A_ncores">ncores</code></td>
<td>
<p>Number of processor cores to use.</p>
</td></tr>
<tr><td><code id="scDHA.pt_+3A_seed">seed</code></td>
<td>
<p>Seed for reproducibility.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with the following keys:
</p>

<ul>
<li><p> pt - Pseudo-time values for each sample.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library(scDHA)
#Load example data (Goolam dataset)
data('Goolam'); data &lt;- t(Goolam$data); label &lt;- as.character(Goolam$label)
#Log transform the data 
data &lt;- log2(data + 1)
if(torch::torch_is_installed()) #scDHA need libtorch installed
{
  #Generate clustering result, the input matrix has rows as samples and columns as genes
  result &lt;- scDHA(data, ncores = 2, seed = 1)
  #Cell stage order in Goolam dataset
  cell.stages &lt;- c("2cell", "4cell", "8cell", "16cell", "blast")
  #Generate pseudo-time for each cell, the input is the output from scDHA function
  result &lt;- scDHA.pt(result, start.point = 1, ncores = 2, seed = 1)
  #Calculate R-squared value 
  r2 &lt;- round(cor(result$pt, as.numeric(factor(label, levels = cell.stages)))^2, digits = 2)
}

</code></pre>

<hr>
<h2 id='scDHA.vis'>scDHA visulization</h2><span id='topic+scDHA.vis'></span>

<h3>Description</h3>

<p>Generating 2D embeded data for visulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scDHA.vis(sc = sc, method = "UMAP", ncores = 10L, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scDHA.vis_+3A_sc">sc</code></td>
<td>
<p>Embedding object produced by the <code>scDHA</code> function.</p>
</td></tr>
<tr><td><code id="scDHA.vis_+3A_method">method</code></td>
<td>
<p>Visualization method to use. It can be &quot;UMAP&quot; or &quot;scDHA&quot;. The default setting is &quot;UMAP&quot;.</p>
</td></tr>
<tr><td><code id="scDHA.vis_+3A_ncores">ncores</code></td>
<td>
<p>Number of processor cores to use.</p>
</td></tr>
<tr><td><code id="scDHA.vis_+3A_seed">seed</code></td>
<td>
<p>Seed for reproducibility.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following keys:
</p>

<ul>
<li><p> pred - A matrix representing the 2D projection of single-cell data, where rows represent samples and columns represent latent components.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library(scDHA)
#Load example data (Goolam dataset)
data('Goolam'); data &lt;- t(Goolam$data); label &lt;- as.character(Goolam$label)
#Log transform the data 
data &lt;- log2(data + 1)
if(torch::torch_is_installed()) #scDHA need libtorch installed
{
  #Generate clustering result, the input matrix has rows as samples and columns as genes
  result &lt;- scDHA(data, ncores = 2, seed = 1)
  #Generate 2D representation, the input is the output from scDHA function
  result &lt;- scDHA.vis(result, ncores = 2, seed = 1)
  #Plot the representation of the dataset, different colors represent different cell types
  plot(result$pred, col=factor(label), xlab = "scDHA1", ylab = "scDHA2")
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
