<!DOCTYPE html><html><head><title>Help for package scDiffCom</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scDiffCom}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BuildNetwork'><p>Display cell-type to cell-type interactive networks</p></a></li>
<li><a href='#BuildShiny'><p>A shiny app to display scDiffCom results</p></a></li>
<li><a href='#EraseRawCCI'><p>Create a copy of a scDiffCom object without <code>cci_table_raw</code></p></a></li>
<li><a href='#FilterCCI'><p>Filter a scDiffCom object with new filtering parameters</p></a></li>
<li><a href='#gene_ontology_level'><p>All gene ontology terms annotated with their levels</p></a></li>
<li><a href='#GetDistributions'><p>Return the slot <code>distributions</code> from a scDiffCom object</p></a></li>
<li><a href='#GetParameters'><p>Return the slot <code>parameters</code> from a scDiffCom object</p></a></li>
<li><a href='#GetTableCCI'><p>Return (a subset) of the slot <code>cci_table_raw</code> or</p>
<code>cci_table_detected</code> from a scDiffCom object</a></li>
<li><a href='#GetTableORA'><p>Return some or all ORA tables from the slot <code>ora_table</code></p>
from a scDiffCom object</a></li>
<li><a href='#LRI_human'><p>A collection of human ligand-receptor interactions.</p></a></li>
<li><a href='#LRI_mouse'><p>A collection of mouse ligand-receptor interactions.</p></a></li>
<li><a href='#LRI_rat'><p>A collection of rat ligand-receptor interactions.</p></a></li>
<li><a href='#PlotORA'><p>Display top over-represented keywords from a category of interest</p></a></li>
<li><a href='#ReduceGO'><p>Reduce scDiffCom GO Terms</p></a></li>
<li><a href='#run_interaction_analysis'><p>Run (differential) intercellular communication analysis</p></a></li>
<li><a href='#RunORA'><p>Run over-representation analysis</p></a></li>
<li><a href='#scDiffCom-class'><p>The scDiffCom Class</p></a></li>
<li><a href='#seurat_sample_tms_liver'><p>A down-sampled Seurat object to use for testing and benchmarking</p></a></li>
<li><a href='#show,scDiffCom-method'><p>Display a scDiffCom object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Differential Analysis of Intercellular Communication from
scRNA-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis tools to investigate changes in intercellular
    communication from scRNA-seq data. Using a Seurat object as input,
    the package infers which cell-cell interactions are present in the dataset
    and how these interactions change between two conditions of interest
    (e.g. young vs old). It relies on an internal database of ligand-receptor
    interactions (available for human, mouse and rat) that have been gathered
    from several published studies. Detection and differential analyses
    rely on permutation tests. The package also contains several tools
    to perform over-representation analysis and visualize the results. See
    Lagger, C. et al. (2023) &lt;<a href="https://doi.org/10.1038%2Fs43587-023-00514-x">doi:10.1038/s43587-023-00514-x</a>&gt; for a full 
    description of the methodology.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>biomaRt, covr, DT, ggplot2, GOSemSim, igraph, kableExtra,
KEGGREST, knitr, ontologyIndex, ontoProc, pkgdown, plotly,
RColorBrewer, rmarkdown, rrvgo, spelling, shiny, shinythemes,
shinyWidgets, SingleCellSignalR, testthat (&ge; 3.0.0),
visNetwork</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, DelayedArray, future, future.apply, magrittr,
methods, Seurat (&ge; 4.0.0), stats, utils</td>
</tr>
<tr>
<td>Collate:</td>
<td>'data.R' 'interaction_analysis.R' 'objects.R' 'utils_LRI.R'
'utils_cci.R' 'utils_filtering.R' 'utils_interactiveNetworks.R'
'utils_ora.R' 'utils_permutation.R' 'utils_preprocessing.R'
'utils_validation.R' 'utils_shiny.R'</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://cyrillagger.github.io/scDiffCom/">https://cyrillagger.github.io/scDiffCom/</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-03 17:51:09 UTC; clagger</td>
</tr>
<tr>
<td>Author:</td>
<td>Cyril Lagger <a href="https://orcid.org/0000-0003-1701-6896"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Eugen Ursu [aut],
  Anais Equey [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Cyril Lagger &lt;lagger.cyril@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-03 18:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BuildNetwork'>Display cell-type to cell-type interactive networks</h2><span id='topic+BuildNetwork'></span><span id='topic+BuildNetwork+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Create and plot an interactive network that summarize how
cell-types and their interactions are over-represented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BuildNetwork(
  object,
  network_type = c("ORA_network"),
  layout_type = c("bipartite", "conventional"),
  abbreviation_table = NULL
)

## S4 method for signature 'scDiffCom'
BuildNetwork(
  object,
  network_type = c("ORA_network"),
  layout_type = c("bipartite", "conventional"),
  abbreviation_table = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BuildNetwork_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
<tr><td><code id="BuildNetwork_+3A_network_type">network_type</code></td>
<td>
<p>Type of network to display. Currently, only
<code>ORA_network</code> (default) is supported.</p>
</td></tr>
<tr><td><code id="BuildNetwork_+3A_layout_type">layout_type</code></td>
<td>
<p>Layout of the network to display. Can either be
<code>"bipartite"</code> (default) or <code>"conventional"</code>.</p>
</td></tr>
<tr><td><code id="BuildNetwork_+3A_abbreviation_table">abbreviation_table</code></td>
<td>
<p>Table with abbreviations
for the cell types present in the <code>object</code>. If <code>NULL</code> (default),
full names of the cell-types are displayed. Otherwise, it must be a
data.frame or data.table with exactly two columns with names
<code>ORIGINAL_CELLTYPE</code> and <code>ABBR_CELLTYPE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A visNetwork object.
</p>

<hr>
<h2 id='BuildShiny'>A shiny app to display scDiffCom results</h2><span id='topic+BuildShiny'></span><span id='topic+BuildShiny+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Launch a shiny app to explore scDiffCom results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BuildShiny(object, reduced_go_table = NULL, ...)

## S4 method for signature 'scDiffCom'
BuildShiny(object, reduced_go_table = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BuildShiny_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
<tr><td><code id="BuildShiny_+3A_reduced_go_table">reduced_go_table</code></td>
<td>
<p>If <code>NULL</code> (default), over-represented GO terms
are displayed as dot plots. If the output of
<code>scDiffCom::ReduceGO(object)</code>, GO terms are displayed on a on treemap
based on their semantic similarity and over-representation score</p>
</td></tr>
<tr><td><code id="BuildShiny_+3A_...">...</code></td>
<td>
<p>Additional parameters to <code>shiny::runApp</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Launch a shiny app
</p>

<hr>
<h2 id='EraseRawCCI'>Create a copy of a scDiffCom object without <code>cci_table_raw</code></h2><span id='topic+EraseRawCCI'></span><span id='topic+EraseRawCCI+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>This function will replace <code>cci_table_raw</code> by an empty list. Useful to
save space for large datasets. However, after this operation,
no filtering can be re-run on the new object, meaning that obtaining
results for different filtering parameters will require the perform the full
analysis from scratch.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EraseRawCCI(object)

## S4 method for signature 'scDiffCom'
EraseRawCCI(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EraseRawCCI_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scDiffCom object with an empty list for <code>cci_table_raw</code>.
</p>

<hr>
<h2 id='FilterCCI'>Filter a scDiffCom object with new filtering parameters</h2><span id='topic+FilterCCI'></span><span id='topic+FilterCCI+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Filtering (and ORA) is performed with new parameter on an existing
<code>scDiffCom</code> object. The slots <code>cci_table_detected</code> and
<code>ora_table</code> are updated accordingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FilterCCI(
  object,
  new_threshold_quantile_score = NULL,
  new_threshold_p_value_specificity = NULL,
  new_threshold_p_value_de = NULL,
  new_threshold_logfc = NULL,
  skip_ora = FALSE,
  extra_annotations = NULL,
  verbose = TRUE
)

## S4 method for signature 'scDiffCom'
FilterCCI(
  object,
  new_threshold_quantile_score = NULL,
  new_threshold_p_value_specificity = NULL,
  new_threshold_p_value_de = NULL,
  new_threshold_logfc = NULL,
  skip_ora = FALSE,
  extra_annotations = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FilterCCI_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
<tr><td><code id="FilterCCI_+3A_new_threshold_quantile_score">new_threshold_quantile_score</code></td>
<td>
<p>New threshold value to update
<code>threshold_quantile_score</code>. If <code>NULL</code> (default),
the value is not updated.</p>
</td></tr>
<tr><td><code id="FilterCCI_+3A_new_threshold_p_value_specificity">new_threshold_p_value_specificity</code></td>
<td>
<p>New threshold value to update
<code>threshold_p_value_specificity</code>. If <code>NULL</code> (default),
the value is not updated.</p>
</td></tr>
<tr><td><code id="FilterCCI_+3A_new_threshold_p_value_de">new_threshold_p_value_de</code></td>
<td>
<p>New threshold value to update
<code>threshold_p_value_de</code>. If <code>NULL</code> (default),
the value is not updated.</p>
</td></tr>
<tr><td><code id="FilterCCI_+3A_new_threshold_logfc">new_threshold_logfc</code></td>
<td>
<p>New threshold value to update
<code>threshold_logfc</code>. If <code>NULL</code> (default),
the value is not updated.</p>
</td></tr>
<tr><td><code id="FilterCCI_+3A_skip_ora">skip_ora</code></td>
<td>
<p>Default is <code>FALSE</code>. If <code>TRUE</code>, ORA is not
performed with the new parameters and <code>ora_table</code> is set to an
empty list. May be useful if one wants to quickly test (loop-over) several
values of parameters and by-pass the ORA computing time.</p>
</td></tr>
<tr><td><code id="FilterCCI_+3A_extra_annotations">extra_annotations</code></td>
<td>
<p>Convenience parameter to perform ORA on user-defined
non-standard categories. If <code>NULL</code> (default), ORA is
performed on standard categories. Otherwise it must be a list of data.tables
or data.frames (see Details).</p>
</td></tr>
<tr><td><code id="FilterCCI_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code> (default) progress messages are printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When <code>FilterCCI</code> is called with new parameters, both
<code>cci_table_detected</code> and <code>ora_table</code> are updated. For
ORA, a call to <code>RunORA</code> is automatically performed on all standard
categories. Additional user-defined ORA categories can be added via the
parameter <code>extra_annotations</code>. The data.frames or data.tables in this
list must have exactly two columns that indicates a relationship between
values from a standard category (first column) to values of the new
category (second column). As a typical example, this
<a href="https://cyrillagger.github.io/scDiffCom/articles/scDiffCom-vignette.html">vignette</a>
shows how to perform ORA on cell type families attached to each cell type.
</p>


<h3>Value</h3>

<p>A scDiffCom object with updated results in <code>cci_table_detected</code>
and <code>ora_table</code>.
</p>

<hr>
<h2 id='gene_ontology_level'>All gene ontology terms annotated with their levels</h2><span id='topic+gene_ontology_level'></span>

<h3>Description</h3>

<p>This data.table contains all GO terms retrieved from the package
ontoProc. Each term is annotated with its number of ancestors,
parents and children, as well as with its level (i.e. depth) in
the gene ontology graph. Levels are computed by scDiffCom according to
<code>scDiffCom:::get_GO_LEVELS()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_ontology_level)
</code></pre>


<h3>Format</h3>

<p>A data.table
</p>


<h3>References</h3>

<p><a href="https://doi.org/doi:10.18129/B9.bioc.ontoProc">ontoProc</a>
</p>

<hr>
<h2 id='GetDistributions'>Return the slot <code>distributions</code> from a scDiffCom object</h2><span id='topic+GetDistributions'></span><span id='topic+GetDistributions+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Return the slot <code>distributions</code> from a scDiffCom object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetDistributions(object)

## S4 method for signature 'scDiffCom'
GetDistributions(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetDistributions_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of matrices with the null distributions of each
CCI.
</p>

<hr>
<h2 id='GetParameters'>Return the slot <code>parameters</code> from a scDiffCom object</h2><span id='topic+GetParameters'></span><span id='topic+GetParameters+2CscDiffComBase-method'></span>

<h3>Description</h3>

<p>Return the parameters that have been passed to
<code><a href="#topic+run_interaction_analysis">run_interaction_analysis</a></code> as well as a few other
parameters computed alongside the analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetParameters(object)

## S4 method for signature 'scDiffComBase'
GetParameters(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetParameters_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of parameters.
</p>

<hr>
<h2 id='GetTableCCI'>Return (a subset) of the slot <code>cci_table_raw</code> or
<code>cci_table_detected</code> from a scDiffCom object</h2><span id='topic+GetTableCCI'></span><span id='topic+GetTableCCI+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Return (a subset) of the slot <code>cci_table_raw</code> or
<code>cci_table_detected</code> from a scDiffCom object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetTableCCI(object, type, simplified)

## S4 method for signature 'scDiffCom'
GetTableCCI(object, type = c("detected", "raw"), simplified = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetTableCCI_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
<tr><td><code id="GetTableCCI_+3A_type">type</code></td>
<td>
<p>Table to extract information from. Can be either
<code>"detected"</code> (default) or <code>"raw"</code>.</p>
</td></tr>
<tr><td><code id="GetTableCCI_+3A_simplified">simplified</code></td>
<td>
<p>If <code>TRUE</code> (default) only the most
informative columns of the data.table are returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table.
</p>

<hr>
<h2 id='GetTableORA'>Return some or all ORA tables from the slot <code>ora_table</code>
from a scDiffCom object</h2><span id='topic+GetTableORA'></span><span id='topic+GetTableORA+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Return some or all ORA tables from the slot <code>ora_table</code>
from a scDiffCom object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetTableORA(object, categories, simplified)

## S4 method for signature 'scDiffCom'
GetTableORA(object, categories = "all", simplified = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetTableORA_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
<tr><td><code id="GetTableORA_+3A_categories">categories</code></td>
<td>
<p>Names of the ORA categories to return. If <code>"all"</code>
(default), returns all of them.</p>
</td></tr>
<tr><td><code id="GetTableORA_+3A_simplified">simplified</code></td>
<td>
<p>If <code>TRUE</code> (default) only the most
informative columns of the data.table are returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data.tables.
</p>

<hr>
<h2 id='LRI_human'>A collection of human ligand-receptor interactions.</h2><span id='topic+LRI_human'></span>

<h3>Description</h3>

<p>This dataset contains a data.table of curated human ligand-receptor
interactions as well as related annotations (GO Terms, KEGG Pathways) and
metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LRI_human)
</code></pre>


<h3>Format</h3>

<p>A list with the following items:
</p>

<ol>
<li> <p><code>LRI_curated</code>: a data.table of curated LRIs
</p>
</li>
<li> <p><code>LRI_curated_GO</code>: a data.table with GO terms attached to
curated LRIs
</p>
</li>
<li> <p><code>LRI_curated_KEGG</code>: a data.table with KEGG pathways attached to
curated LRIs
</p>
</li>
<li> <p><code>LRI_retrieved_dates</code>: dates at which data have been retrieved
from the seven external databases
</p>
</li>
<li> <p><code>LRI_retrieved_from</code>: paths or packages from where data have
been retrieved
</p>
</li>
<li> <p><code>LRI_biomart_ensembl_version</code>: version of ensembl used for
GO annotation
</p>
</li></ol>



<h3>Details</h3>

<p>The dataset has been built internally in scDiffCom according to
<code>scDiffCom:::build_LRI(species = "human")</code>. The LRIs have been retrieved
from seven databases (see References). Note that only curated LRIs
have been kept.
</p>


<h3>References</h3>

<p>CellChat (<a href="https://pubmed.ncbi.nlm.nih.gov/33597522/">PMID: 33597522</a>),
CellPhoneDB (<a href="https://pubmed.ncbi.nlm.nih.gov/32103204/">PMID: 32103204</a>),
CellTalkDB (<a href="https://pubmed.ncbi.nlm.nih.gov/33147626/">PMID: 33147626</a>),
connectomeDB2020 (<a href="https://pubmed.ncbi.nlm.nih.gov/33024107/">PMID: 33024107</a>),
ICELLNET (<a href="https://pubmed.ncbi.nlm.nih.gov/33597528/">PMID: 33597528</a>),
NicheNet (<a href="https://pubmed.ncbi.nlm.nih.gov/31819264/">PMID: 31819264</a>),
SingleCellSignalR (<a href="https://pubmed.ncbi.nlm.nih.gov/32196115/">PMID: 32196115</a>)
</p>

<hr>
<h2 id='LRI_mouse'>A collection of mouse ligand-receptor interactions.</h2><span id='topic+LRI_mouse'></span>

<h3>Description</h3>

<p>This dataset contains a data.table of curated mouse ligand-receptor
interactions as well as related annotations (GO Terms, KEGG Pathways) and
metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LRI_mouse)
</code></pre>


<h3>Format</h3>

<p>A list with the following items:
</p>

<ol>
<li> <p><code>LRI_curated</code>: a data.table of curated LRIs
</p>
</li>
<li> <p><code>LRI_curated_GO</code>: a data.table with GO terms attached to
curated LRI
</p>
</li>
<li> <p><code>LRI_curated_KEGG</code>: a data.table with KEGG pathways attached to
curated LRIs
</p>
</li>
<li> <p><code>LRI_retrieved_dates</code>: dates at which data have been retrieved
from the seven external databases
</p>
</li>
<li> <p><code>LRI_retrieved_from</code>: paths or packages from where data have
been retrieved
</p>
</li>
<li> <p><code>LRI_biomart_ensembl_version</code>: version of ensembl used for
GO annotation and orthology conversion
</p>
</li></ol>



<h3>Details</h3>

<p>The dataset has been built internally in scDiffCom according to
<code>scDiffCom:::build_LRI(species = "mouse")</code>. The LRIs have been retrieved
from seven databases (see References). Note that only curated LRIs
have been kept.
</p>


<h3>References</h3>

<p>CellChat (<a href="https://pubmed.ncbi.nlm.nih.gov/33597522/">PMID: 33597522</a>),
CellPhoneDB (<a href="https://pubmed.ncbi.nlm.nih.gov/32103204/">PMID: 32103204</a>),
CellTalkDB (<a href="https://pubmed.ncbi.nlm.nih.gov/33147626/">PMID: 33147626</a>),
connectomeDB2020 (<a href="https://pubmed.ncbi.nlm.nih.gov/33024107/">PMID: 33024107</a>),
ICELLNET (<a href="https://pubmed.ncbi.nlm.nih.gov/33597528/">PMID: 33597528</a>),
NicheNet (<a href="https://pubmed.ncbi.nlm.nih.gov/31819264/">PMID: 31819264</a>),
SingleCellSignalR (<a href="https://pubmed.ncbi.nlm.nih.gov/32196115/">PMID: 32196115</a>)
</p>

<hr>
<h2 id='LRI_rat'>A collection of rat ligand-receptor interactions.</h2><span id='topic+LRI_rat'></span>

<h3>Description</h3>

<p>This dataset contains a data.table of curated rat ligand-receptor
interactions as well as related annotations (GO Terms, KEGG Pathways) and
metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LRI_rat)
</code></pre>


<h3>Format</h3>

<p>A list with the following items:
</p>

<ol>
<li> <p><code>LRI_curated</code>: a data.table of curated LRIs
</p>
</li>
<li> <p><code>LRI_curated_GO</code>: a data.table with GO terms attached to
curated LRI
</p>
</li>
<li> <p><code>LRI_curated_KEGG</code>: a data.table with KEGG pathways attached to
curated LRIs
</p>
</li>
<li> <p><code>LRI_retrieved_dates</code>: dates at which data have been retrieved
from the seven external databases
</p>
</li>
<li> <p><code>LRI_retrieved_from</code>: paths or packages from where data have
been retrieved
</p>
</li>
<li> <p><code>LRI_biomart_ensembl_version</code>: version of ensembl used for
GO annotation and orthology conversion
</p>
</li></ol>



<h3>Details</h3>

<p>The dataset has been built internally in scDiffCom according to
<code>scDiffCom:::build_LRI(species = "rat")</code>. The LRIs have been retrieved
from seven databases (see References). Note that only curated LRIs
have been kept.
</p>


<h3>References</h3>

<p>CellChat (<a href="https://pubmed.ncbi.nlm.nih.gov/33597522/">PMID: 33597522</a>),
CellPhoneDB (<a href="https://pubmed.ncbi.nlm.nih.gov/32103204/">PMID: 32103204</a>),
CellTalkDB (<a href="https://pubmed.ncbi.nlm.nih.gov/33147626/">PMID: 33147626</a>),
connectomeDB2020 (<a href="https://pubmed.ncbi.nlm.nih.gov/33024107/">PMID: 33024107</a>),
ICELLNET (<a href="https://pubmed.ncbi.nlm.nih.gov/33597528/">PMID: 33597528</a>),
NicheNet (<a href="https://pubmed.ncbi.nlm.nih.gov/31819264/">PMID: 31819264</a>),
SingleCellSignalR (<a href="https://pubmed.ncbi.nlm.nih.gov/32196115/">PMID: 32196115</a>)
</p>

<hr>
<h2 id='PlotORA'>Display top over-represented keywords from a category of interest</h2><span id='topic+PlotORA'></span><span id='topic+PlotORA+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Plot a graph that shows the top over-represented terms of a given
category for a given regulation. Terms are ordered by their ORA scores,
computed from their odds ratios and adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotORA(
  object,
  category,
  regulation = c("UP", "DOWN", "FLAT"),
  max_terms_show = 20,
  GO_aspect = c("biological_process", "molecular_function", "cellular_component"),
  OR_threshold = 1,
  bh_p_value_threshold = 0.05
)

## S4 method for signature 'scDiffCom'
PlotORA(
  object,
  category,
  regulation = c("UP", "DOWN", "FLAT"),
  max_terms_show = 20,
  GO_aspect = c("biological_process", "molecular_function", "cellular_component"),
  OR_threshold = 1,
  bh_p_value_threshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotORA_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
<tr><td><code id="PlotORA_+3A_category">category</code></td>
<td>
<p>ORA category to display. Must be the name of one of the
category present in <code>ora_table</code>.</p>
</td></tr>
<tr><td><code id="PlotORA_+3A_regulation">regulation</code></td>
<td>
<p>ORA regulation to display. Can be either <code>UP</code>
(default), <code>DOWN</code> or <code>FLAT</code>.</p>
</td></tr>
<tr><td><code id="PlotORA_+3A_max_terms_show">max_terms_show</code></td>
<td>
<p>Maximum number of terms to display. Default is
<code>20</code>.</p>
</td></tr>
<tr><td><code id="PlotORA_+3A_go_aspect">GO_aspect</code></td>
<td>
<p>Name of the GO aspect to display when
<code>category == "GO_TERMS"</code>. Can be either <code>biological_process</code> (
default), <code>molecular_function</code> or <code>cellular_component</code>.</p>
</td></tr>
<tr><td><code id="PlotORA_+3A_or_threshold">OR_threshold</code></td>
<td>
<p>Only the terms with an odds ratio above this threshold
will be displayed. Default is <code>1</code>, meaning no filtering is performed.</p>
</td></tr>
<tr><td><code id="PlotORA_+3A_bh_p_value_threshold">bh_p_value_threshold</code></td>
<td>
<p>Only the terms with an adjusted p-value below
this threshold (and always below 0.05) will be displayed. Default is
<code>0.05</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The ORA score is computed as the product between <code>log2(odds ratio)</code> and
<code>-log10(adj. p-value)</code>.
</p>


<h3>Value</h3>

<p>A ggplot object.
</p>

<hr>
<h2 id='ReduceGO'>Reduce scDiffCom GO Terms</h2><span id='topic+ReduceGO'></span><span id='topic+ReduceGO+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Perform semantic similarity analysis and reduction of the
overrepresented GO terms of an scDiffCom object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReduceGO(
  object,
  method = c("Rel", "Resnik", "Lin", "Jiang", "Wang"),
  threshold = 0.7
)

## S4 method for signature 'scDiffCom'
ReduceGO(
  object,
  method = c("Rel", "Resnik", "Lin", "Jiang", "Wang"),
  threshold = 0.7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReduceGO_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
<tr><td><code id="ReduceGO_+3A_method">method</code></td>
<td>
<p>A distance method supported by rrvgo and GOSemSim:
c(&quot;Rel&quot;, &quot;Resnik&quot;, &quot;Lin&quot;, &quot;Jiang&quot;, &quot;Wang&quot;)</p>
</td></tr>
<tr><td><code id="ReduceGO_+3A_threshold">threshold</code></td>
<td>
<p>Similarity threshold used by <code>rrvgo::reduceSimMatrix</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is basically a wrapper around <code>rrvgo::calculateSimMatrix</code>
and <code>rrvgo::reduceSimMatrix</code>.
</p>


<h3>Value</h3>

<p>A data.table of GO terms with their reduction
</p>

<hr>
<h2 id='run_interaction_analysis'>Run (differential) intercellular communication analysis</h2><span id='topic+run_interaction_analysis'></span>

<h3>Description</h3>

<p>Perform (differential) cell type to cell type communication analysis from a
<a href="https://satijalab.org/seurat/index.html">Seurat object</a>,
using an internal database of ligand-receptor interactions (LRIs). It
infers biologically relevant cell-cell interactions (CCIs) and how they
change between two conditions of interest. Over-representation analysis
is automatically performed to determine dominant differential signals at the
level of the genes, cell types, GO Terms and KEGG Pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_interaction_analysis(
  seurat_object,
  LRI_species,
  seurat_celltype_id,
  seurat_condition_id,
  iterations = 1000,
  scdiffcom_object_name = "scDiffCom_object",
  seurat_assay = "RNA",
  seurat_slot = "data",
  log_scale = FALSE,
  score_type = "geometric_mean",
  threshold_min_cells = 5,
  threshold_pct = 0.1,
  threshold_quantile_score = 0.2,
  threshold_p_value_specificity = 0.05,
  threshold_p_value_de = 0.05,
  threshold_logfc = log(1.5),
  return_distributions = FALSE,
  seed = 42,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_interaction_analysis_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object that must contain normalized
data and relevant <code>meta.data</code> columns (see below). Gene names must be
MGI (mouse) or HGNC (human) approved symbols.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_lri_species">LRI_species</code></td>
<td>
<p>Either <code>"mouse"</code>, <code>"human"</code> or <code>"rat"</code>. Indicates which
LRI database to use and corresponds to the species of the <code>seurat_object</code>.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_seurat_celltype_id">seurat_celltype_id</code></td>
<td>
<p>Name of the <code>meta.data</code> column in
<code>seurat_object</code> that contains cell-type annotations
(e.g.: <code>"CELL_TYPE"</code>).</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_seurat_condition_id">seurat_condition_id</code></td>
<td>
<p>List that contains information regarding the
two conditions on which to perform differential analysis. Must contain
the following three named items:
</p>

<ol>
<li> <p><code>column_name</code>: name of the <code>meta.data</code> column in
<code>seurat_object</code> that indicates the condition on each cell (e.g. <code>"AGE"</code>)
</p>
</li>
<li> <p><code>cond1_name</code>: name of the first condition (e.g. <code>"YOUNG"</code>)
</p>
</li>
<li> <p><code>cond2_name</code>: name of the second condition (e.g. <code>"OLD"</code>)
</p>
</li></ol>

<p>Can also be set to <code>NULL</code> to only perform a detection analysis
(see Details).</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_iterations">iterations</code></td>
<td>
<p>Number of permutations to perform the statistical
analysis. The default (<code>1000</code>) is a good compromise for an exploratory
analysis and to obtain reasonably accurate p-values in a short time.
Otherwise, we recommend using <code>10000</code> iterations and to run the
analysis in parallel (see Details). Can also be set to <code>0</code> for
debugging and quickly returning partial results without
statistical significance.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_scdiffcom_object_name">scdiffcom_object_name</code></td>
<td>
<p>Name of the <code>scDiffCom</code> S4 object that will
be returned (<code>"scDiffCom_object"</code> by default).</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_seurat_assay">seurat_assay</code></td>
<td>
<p>Assay of <code>seurat_object</code> from which to extract data.
See Details for an explanation on how data are extracted based on the three
parameters <code>seurat_assay</code>, <code>seurat_slot</code> and <code>log_scale</code>.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_seurat_slot">seurat_slot</code></td>
<td>
<p>Slot of <code>seurat_object</code> from which to extract data.
See Details for an explanation on how data are extracted based on the three
parameters <code>seurat_assay</code>, <code>seurat_slot</code> and <code>log_scale</code>.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_log_scale">log_scale</code></td>
<td>
<p>When <code>FALSE</code> (the default, recommended), data are
treated as normalized but not log1p-transformed. See Details for an
explanation on how data are extracted based on the three
parameters <code>seurat_assay</code>, <code>seurat_slot</code> and <code>log_scale</code>.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_score_type">score_type</code></td>
<td>
<p>Metric used to compute cell-cell interaction (CCI) scores.
Can either be <code>"geometric_mean"</code> (default) or <code>"arithmetic_mean"</code>.
It is strongly recommended to use the geometric mean, especially when
performing differential analysis. The arithmetic mean might be used when
uniquely doing a detection analysis or if the results want to be compared
with those of another package.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_threshold_min_cells">threshold_min_cells</code></td>
<td>
<p>Minimal number of cells - of a given cell type
and condition - required to express a gene for this gene to be considered
expressed in the corresponding cell type. Incidentally, cell types with
less cells than this threshold are removed from the analysis.
Set to <code>5</code> by default.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_threshold_pct">threshold_pct</code></td>
<td>
<p>Minimal fraction of cells - of a given cell type
and condition - required to express a gene for this gene to be considered
expressed in the corresponding cell type. Set to  <code>0.1</code> by default.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_threshold_quantile_score">threshold_quantile_score</code></td>
<td>
<p>Threshold value used in conjunction with
<code>threshold_p_value_specificity</code> to establish if a CCI is considered
&quot;detected&quot;. The default (<code>0.2</code>) indicates that CCIs with a score
in the 20% lowest-scores are not considered detected. Can be modified
without the need to re-perform the permutation analysis (see Details).</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_threshold_p_value_specificity">threshold_p_value_specificity</code></td>
<td>
<p>Threshold value used in conjunction
with <code>threshold_quantile_score</code> to establish if a CCI is considered
&quot;detected&quot;. CCIs with a (BH-adjusted) specificity p-value above the
threshold (<code>0.05</code> by default) are not considered detected. Can be
modified without the need to re-perform the permutation analysis
(see Details).</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_threshold_p_value_de">threshold_p_value_de</code></td>
<td>
<p>Threshold value used in conjunction
with <code>threshold_logfc</code> to establish how CCIs are differentially
expressed between <code>cond1_name</code> and <code>cond2_name</code>. CCIs with a
(BH-adjusted) differential p-value above the threshold (<code>0.05</code> by
default) are not considered to change significantly. Can be modified
without the need to re-perform the permutation analysis (see Details).</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_threshold_logfc">threshold_logfc</code></td>
<td>
<p>Threshold value used in conjunction with
<code>threshold_p_value_de</code> to establish how CCIs are differentially
expressed between <code>cond1_name</code> and <code>cond2_name</code>. CCIs with an
absolute logFC below the threshold (<code>log(1.5)</code> by default) are
considered &quot;FLAT&quot;. Can be modified without the need to
re-perform the permutation analysis (see Details).</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_return_distributions">return_distributions</code></td>
<td>
<p><code>FALSE</code> by default. If <code>TRUE</code>, the
distributions obtained from the permutation test are returned alongside
the other results. May be used for testing or benchmarking purposes. Can
only be enabled when <code>iterations</code> is less than <code>1000</code> in order
to avoid out of memory issues.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_seed">seed</code></td>
<td>
<p>Set a random seed (<code>42</code> by default) to obtain reproducible
results.</p>
</td></tr>
<tr><td><code id="run_interaction_analysis_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code> (default), print progress messages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The primary use of this function (and of the package) is to perform
differential intercellular communication analysis. However, it is also
possible to only perform a detection analysis (by setting
<code>seurat_condition_id</code> to <code>NULL</code>), e.g. if one wants to
infer cell-cell interactions from a dataset without having conditions on the cells.
</p>
<p>By convention, when performing differential analysis, LOGFC are computed as
<code>log(score(cond2_name)/score(cond1_name))</code>. In other words,
&quot;UP&quot;-regulated CCIs have a larger score in <code>cond2_name</code>.
</p>
<p>Parallel computing. If possible, it is recommended to
run this function in parallel in order to speed up the analysis for large
dataset and/or to obtain better accuracy on the p-values by setting a higher
number of <code>iterations</code>. This is as simple as loading the
<a href="https://CRAN.R-project.org/package=future">future</a>
package and setting an appropriate <code>plan</code> (see also our
<a href="https://cyrillagger.github.io/scDiffCom/articles/scDiffCom-vignette.html">vignette</a>).
</p>
<p>Data extraction. The UMI or read counts matrix is extracted from
the assay <code>seurat_assay</code> and the slot <code>seurat_slot</code>. By default,
it is assumed that <code>seurat_object</code> contains log1p-transformed
normalized data in the slot &quot;data&quot; of its assay &quot;RNA&quot;. If <code>log_scale</code>
is <code>FALSE</code> (as recommended), the data are <code>expm1()</code> transformed
in order to recover normalized values not in log scale.
</p>
<p>Modifying filtering parameters (differential analysis only). As long as
the slot <code>cci_table_raw</code> of
the returned scDiffCom object is not erased, filtering parameters can be
modified to recompute the slots <code>cci_table_detected</code> and
<code>ora_table</code>, without re-performing the time consuming permutation
analysis. This may be useful if one wants a fast way to analyze how the
results behave in function of, say, different LOGFC thresholds. In practice,
this can be done by calling the functions <code><a href="#topic+FilterCCI">FilterCCI</a></code> or
<code><a href="#topic+RunORA">RunORA</a></code> (see also our
<a href="https://cyrillagger.github.io/scDiffCom/articles/scDiffCom-vignette.html">vignette</a>).
</p>


<h3>Value</h3>

<p>An S4 object of class <code><a href="#topic+scDiffCom-class">scDiffCom-class</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
run_interaction_analysis(
  seurat_object = seurat_sample_tms_liver,
  LRI_species = "mouse",
  seurat_celltype_id = "cell_type",
  seurat_condition_id = list(
    column_name = "age_group",
    cond1_name = "YOUNG",
    cond2_name = "OLD"
  )
)

## End(Not run)
</code></pre>

<hr>
<h2 id='RunORA'>Run over-representation analysis</h2><span id='topic+RunORA'></span><span id='topic+RunORA+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Perform over-representation analysis (ORA) on a scDiffCom object, with
the possibility to define new categories in addition to the standard
ones supported by default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunORA(
  object,
  categories = c("LRI", "LIGAND_COMPLEX", "RECEPTOR_COMPLEX", "ER_CELLTYPES",
    "EMITTER_CELLTYPE", "RECEIVER_CELLTYPE", "GO_TERMS", "KEGG_PWS"),
  extra_annotations = NULL,
  overwrite = TRUE,
  verbose = TRUE
)

## S4 method for signature 'scDiffCom'
RunORA(
  object,
  categories = c("LRI", "LIGAND_COMPLEX", "RECEPTOR_COMPLEX", "ER_CELLTYPES",
    "EMITTER_CELLTYPE", "RECEIVER_CELLTYPE", "GO_TERMS", "KEGG_PWS"),
  extra_annotations = NULL,
  overwrite = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RunORA_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
<tr><td><code id="RunORA_+3A_categories">categories</code></td>
<td>
<p>Names of the standard categories on which to perform ORA.
Default is all standard categories, namely
<code>c("LRI", "LIGAND_COMPLEX", "RECEPTOR_COMPLEX", "ER_CELLTYPES",
"EMITTER_CELLTYPE", "RECEIVER_CELLTYPE", "GO_TERMS", "KEGG_PWS")</code></p>
</td></tr>
<tr><td><code id="RunORA_+3A_extra_annotations">extra_annotations</code></td>
<td>
<p>Convenience parameter to perform ORA on user-defined
non-standard categories. If <code>NULL</code> (default), ORA is
performed only on standard categories from <code>categories</code>. Otherwise it
must be a list of data.tables
or data.frames (see Details).</p>
</td></tr>
<tr><td><code id="RunORA_+3A_overwrite">overwrite</code></td>
<td>
<p>If <code>TRUE</code> (default), previous results are overwritten
in case they correspond to a category passed in <code>categories</code>.</p>
</td></tr>
<tr><td><code id="RunORA_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code> (default), progress messages are printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Additional user-defined ORA categories can be added via the
parameter <code>extra_annotations</code>. The data.frames or data.tables in this
list must have exactly two columns that indicates a relationship between
values from a standard category (first column) to values of the new
category (second column). As a typical example, this
<a href="https://cyrillagger.github.io/scDiffCom/articles/scDiffCom-vignette.html">vignette</a>
shows how to perform ORA on cell type families attached to each cell type.
</p>


<h3>Value</h3>

<p>A scDiffCom object with updated slot <code>ora_table</code>.
</p>

<hr>
<h2 id='scDiffCom-class'>The scDiffCom Class</h2><span id='topic+scDiffCom-class'></span>

<h3>Description</h3>

<p>An object of this class stores the intercellular communication results
obtained when calling <code><a href="#topic+run_interaction_analysis">run_interaction_analysis</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>parameters</code></dt><dd><p>List of parameters passed to
<code><a href="#topic+run_interaction_analysis">run_interaction_analysis</a></code> and used to build the object.</p>
</dd>
<dt><code>cci_table_raw</code></dt><dd><p>Data.table with all hypothetic CCIs induced from
the original Seurat object and the internal LRI database. Can be erased
with <code><a href="#topic+EraseRawCCI">EraseRawCCI</a></code> to obtain a lighter object, but might be
worth keeping if one intends to modify the filtering parameters
(see also our
<a href="https://cyrillagger.github.io/scDiffCom/articles/scDiffCom-vignette.html">vignette</a>).</p>
</dd>
<dt><code>cci_table_detected</code></dt><dd><p>Data.table with only the detected CCIs. If
<code>cci_table_raw</code> is not <code>NULL</code>, can be updated with new filtering
parameters without running the full permutation analysis (see
<code><a href="#topic+FilterCCI">FilterCCI</a></code>).</p>
</dd>
<dt><code>ora_table</code></dt><dd><p>List of data.tables with the results of the
over-representation analysis for each category. Results for additional
categories can be added with <code><a href="#topic+RunORA">RunORA</a></code>.</p>
</dd>
<dt><code>distributions</code></dt><dd><p>List of matrices with the null distributions of each
CCI. <code>NULL</code> by default.</p>
</dd>
</dl>

<hr>
<h2 id='seurat_sample_tms_liver'>A down-sampled Seurat object to use for testing and benchmarking</h2><span id='topic+seurat_sample_tms_liver'></span>

<h3>Description</h3>

<p>This Seurat object has been down-sampled from the original
Tabula Muris Senis liver object. Pre-processing and normalization has
been performed before down-sampling. It contains 726 features (genes) and
468 samples (cells). It is only intended to be used for testing and
benchmarking and does not contain meaningful biological information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(seurat_sample_tms_liver)
</code></pre>


<h3>Format</h3>

<p>An object of class Seurat.
</p>


<h3>References</h3>

<p><em>A single-cell transcriptomic atlas characterizes
ageing tissues in the mouse</em>, Tabula Muris Consortium (2020)
(<a href="https://pubmed.ncbi.nlm.nih.gov/32669714/">PMID: 32669714</a>)
</p>

<hr>
<h2 id='show+2CscDiffCom-method'>Display a scDiffCom object</h2><span id='topic+show+2CscDiffCom-method'></span>

<h3>Description</h3>

<p>Display a scDiffCom object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'scDiffCom'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CscDiffCom-method_+3A_object">object</code></td>
<td>
<p><code>scDiffCom</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Print summary to the console, no return value.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
