<!DOCTYPE html><html><head><title>Help for package SCEnt</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SCEnt}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gene_het'><p>Find the Heterogeneity of a Gene Within a Population</p></a></li>
<li><a href='#gene_hom'><p>Find the Homogeneity of a Gene Within a Population</p></a></li>
<li><a href='#normalise'><p>Normalise Counts into a Distribution</p></a></li>
<li><a href='#scent_select'><p>Feature Selection by Gene Heterogeneity</p></a></li>
<li><a href='#scent_select_tidy'><p>A Tidy Wrapper for Feature Selection by Heterogeneity</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Single Cell Entropy Analysis of Gene Heterogeneity in Cell
Populations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Analyse single cell RNA sequencing data using entropy to
    calculate heterogeneity and homogeneity of genes amongst the cell
    population. From the work of Michael J. Casey, Ruben J. Sanchez-Garcia 
    and Ben D. MacArthur.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>entropy, tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-29 12:47:02 UTC; hwarden</td>
</tr>
<tr>
<td>Author:</td>
<td>Hugh Warden <a href="https://orcid.org/0000-0002-4308-7316"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hugh Warden &lt;hugh.warden@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-30 08:30:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='gene_het'>Find the Heterogeneity of a Gene Within a Population</h2><span id='topic+gene_het'></span>

<h3>Description</h3>

<p>Find the Heterogeneity of a Gene Within a Population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_het(expr, unit = "log2", normalise = TRUE, transpose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_het_+3A_expr">expr</code></td>
<td>
<p>A vector or matrix of gene expressions. For the matrix, genes
should be represented as rows and cells as columns.</p>
</td></tr>
<tr><td><code id="gene_het_+3A_unit">unit</code></td>
<td>
<p>The units to be parsed to the entropy function.</p>
</td></tr>
<tr><td><code id="gene_het_+3A_normalise">normalise</code></td>
<td>
<p>A logical value representing whether the gene frequencies
should be normalised into a distribution.</p>
</td></tr>
<tr><td><code id="gene_het_+3A_transpose">transpose</code></td>
<td>
<p>A logical value representing whether the matrix should be
transposed before any calculations are performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of the information gained from the gene distribution
compared to the uniform distribution. The higher the value more
heterogeneous the cell is within the population.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Creating Data
gene1 &lt;- c(0,0,0,0,1,2,3)
gene2 &lt;- c(5,5,3,2,0,0,0)
gene3 &lt;- c(2,0,2,1,3,0,1)
gene4 &lt;- c(3,3,3,3,3,3,3)
gene5 &lt;- c(0,0,0,0,5,0,0)
gene_counts &lt;- matrix(c(gene1,gene2,gene3,gene4,gene5), ncol = 5)
rownames(gene_counts) &lt;- paste0("cell",1:7)
colnames(gene_counts) &lt;- paste0("gene",1:5)

#Calculating Heterogeneity For Each Gene
gene_het(gene1)
gene_het(gene2)
gene_het(gene3)
gene_het(gene4)
gene_het(gene5)

#Calculating Heterogeneity For a Matrix
gene_het(gene_counts)
</code></pre>

<hr>
<h2 id='gene_hom'>Find the Homogeneity of a Gene Within a Population</h2><span id='topic+gene_hom'></span>

<h3>Description</h3>

<p>Find the Homogeneity of a Gene Within a Population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_hom(expr, unit = "log2", normalise = TRUE, transpose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_hom_+3A_expr">expr</code></td>
<td>
<p>A vector or matrix of gene expressions. For the matrix, genes
should be represented as rows and cells as columns.</p>
</td></tr>
<tr><td><code id="gene_hom_+3A_unit">unit</code></td>
<td>
<p>The units to be parsed to the entropy function.</p>
</td></tr>
<tr><td><code id="gene_hom_+3A_normalise">normalise</code></td>
<td>
<p>A logical value representing whether the gene frequencies
should be normalised into a distribution.</p>
</td></tr>
<tr><td><code id="gene_hom_+3A_transpose">transpose</code></td>
<td>
<p>A legical value representing whether the matrix should be
transposed before any calculations are performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of the information contained in the distribution of each
gene. The higher this is, the more homogeneous the gene is within the cell
population.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Creating Data
gene1 &lt;- c(0,0,0,0,1,2,3)
gene2 &lt;- c(5,5,3,2,0,0,0)
gene3 &lt;- c(2,0,2,1,3,0,1)
gene4 &lt;- c(3,3,3,3,3,3,3)
gene5 &lt;- c(0,0,0,0,5,0,0)
gene_counts &lt;- matrix(c(gene1,gene2,gene3,gene4,gene5), ncol = 5)
rownames(gene_counts) &lt;- paste0("cell",1:7)
colnames(gene_counts) &lt;- paste0("gene",1:5)

#Calculating Homogeneity For Each Gene
gene_hom(gene1)
gene_hom(gene2)
gene_hom(gene3)
gene_hom(gene4)
gene_hom(gene5)

#Calculating Homogeneity For a Matrix
gene_hom(gene_counts)
</code></pre>

<hr>
<h2 id='normalise'>Normalise Counts into a Distribution</h2><span id='topic+normalise'></span>

<h3>Description</h3>

<p>A function that takes frequency count data and normalises it into a
probability distribution. Only available internally within SCEnt.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalise(dist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalise_+3A_dist">dist</code></td>
<td>
<p>A vector of a frequency distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of a probability distribution relative to the
frequencies.
</p>

<hr>
<h2 id='scent_select'>Feature Selection by Gene Heterogeneity</h2><span id='topic+scent_select'></span>

<h3>Description</h3>

<p>Feature Selection by Gene Heterogeneity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scent_select(
  expr,
  bit_threshold = NULL,
  count_threshold = NULL,
  perc_threshold = NULL,
  unit = "log2",
  normalise = TRUE,
  transpose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scent_select_+3A_expr">expr</code></td>
<td>
<p>A matrix of gene expression data. Cells should be represented
as rows and genes should be represented as columns.</p>
</td></tr>
<tr><td><code id="scent_select_+3A_bit_threshold">bit_threshold</code></td>
<td>
<p>The threshold for the amount of bits of information a
gene must add to be selected as a feature. Only one threshold can be used
at a time.</p>
</td></tr>
<tr><td><code id="scent_select_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A number represented how many of the most
heterogeneous cells should be selected. Only one threshold can be used at
a time.</p>
</td></tr>
<tr><td><code id="scent_select_+3A_perc_threshold">perc_threshold</code></td>
<td>
<p>The percentile of the hetergeneity distribution above
which a gene should be to be selected as a feature.</p>
</td></tr>
<tr><td><code id="scent_select_+3A_unit">unit</code></td>
<td>
<p>The units to be used when calculating entropy.</p>
</td></tr>
<tr><td><code id="scent_select_+3A_normalise">normalise</code></td>
<td>
<p>A logical value representing whether the gene counts
should be normalised into a probability distribution.</p>
</td></tr>
<tr><td><code id="scent_select_+3A_transpose">transpose</code></td>
<td>
<p>A logical value representing whether the matrix should
be transposed before having any operations computed on it.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of gene expression values where genes with low
heterogeneity have been removed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Creating Data
gene1 &lt;- c(0,0,0,0,1,2,3)
gene2 &lt;- c(5,5,3,2,0,0,0)
gene3 &lt;- c(2,0,2,1,3,0,1)
gene4 &lt;- c(3,3,3,3,3,3,3)
gene5 &lt;- c(0,0,0,0,5,0,0)
gene_counts &lt;- matrix(c(gene1,gene2,gene3,gene4,gene5), ncol = 5)
rownames(gene_counts) &lt;- paste0("cell",1:7)
colnames(gene_counts) &lt;- paste0("gene",1:5)

#Performing Feature Selection
scent_select(gene_counts, bit_threshold = 0.85)
scent_select(gene_counts, count_threshold = 2)
scent_select(gene_counts, perc_threshold = 0.25)
</code></pre>

<hr>
<h2 id='scent_select_tidy'>A Tidy Wrapper for Feature Selection by Heterogeneity</h2><span id='topic+scent_select_tidy'></span>

<h3>Description</h3>

<p>A Tidy Wrapper for Feature Selection by Heterogeneity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scent_select_tidy(
  expr,
  bit_threshold = NULL,
  count_threshold = NULL,
  perc_threshold = NULL,
  unit = "log2",
  normalise = TRUE,
  transpose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scent_select_tidy_+3A_expr">expr</code></td>
<td>
<p>A tibble of gene expression data. Cells should be represented
as rows and genes should be represented as columns.</p>
</td></tr>
<tr><td><code id="scent_select_tidy_+3A_bit_threshold">bit_threshold</code></td>
<td>
<p>The threshold for the amount of bits of information a
gene must add to be selected as a feature. Only one threshold can be used
at a time.</p>
</td></tr>
<tr><td><code id="scent_select_tidy_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A number represented how many of the most
heterogeneous cells should be selected. Only one threshold can be used at
a time.</p>
</td></tr>
<tr><td><code id="scent_select_tidy_+3A_perc_threshold">perc_threshold</code></td>
<td>
<p>The percentile of the hetergeneity distribution above
which a gene should be to be selected as a feature.</p>
</td></tr>
<tr><td><code id="scent_select_tidy_+3A_unit">unit</code></td>
<td>
<p>The units to be used when calculating entropy.</p>
</td></tr>
<tr><td><code id="scent_select_tidy_+3A_normalise">normalise</code></td>
<td>
<p>A logical value representing whether the gene counts
should be normalised into a probability distribution.</p>
</td></tr>
<tr><td><code id="scent_select_tidy_+3A_transpose">transpose</code></td>
<td>
<p>A logical value representing whether the matrix should
be transposed before having any operations computed on it.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble of gene expression values where genes with low
heterogeneity have been removed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Creating Data
library(tibble)
gene1 &lt;- c(0,0,0,0,1,2,3)
gene2 &lt;- c(5,5,3,2,0,0,0)
gene3 &lt;- c(2,0,2,1,3,0,1)
gene4 &lt;- c(3,3,3,3,3,3,3)
gene5 &lt;- c(0,0,0,0,5,0,0)
gene_counts &lt;- matrix(c(gene1,gene2,gene3,gene4,gene5), ncol = 5)
rownames(gene_counts) &lt;- paste0("cell",1:7)
colnames(gene_counts) &lt;- paste0("gene",1:5)
gene_counts &lt;- as_tibble(gene_counts)

#Performing Feature Selection
scent_select_tidy(gene_counts, bit_threshold = 0.85)
scent_select_tidy(gene_counts, count_threshold = 2)
scent_select_tidy(gene_counts, perc_threshold = 0.25)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
