<!DOCTYPE html><html><head><title>Help for package scfetch</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scfetch}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Bam2Fastq'><p>Convert bam files to fastq files.</p></a></li>
<li><a href='#DownloadBam'><p>Download bam.</p></a></li>
<li><a href='#DownloadSRA'><p>Download SRA.</p></a></li>
<li><a href='#ExportSeurat'><p>Export SeuratObject to Other Formats.</p></a></li>
<li><a href='#ExtractCBComposition'><p>Extract Cell Type Composition of UCSC Cell Browser Datasets.</p></a></li>
<li><a href='#ExtractCBDatasets'><p>Extract UCSC Cell Browser Datasets with Attributes.</p></a></li>
<li><a href='#ExtractCELLxGENEMeta'><p>Extract Metadata of CELLxGENE Datasets with Attributes.</p></a></li>
<li><a href='#ExtractGEOExp'><p>Extract Raw Count Matrix or Fortmat Supplementary Files to 10x.</p></a></li>
<li><a href='#ExtractGEOExpSupp'><p>Extract Raw Count Matrix from Supplementary Files.</p></a></li>
<li><a href='#ExtractGEOExpSupp10x'><p>Fortmat Supplementary Files to 10x.</p></a></li>
<li><a href='#ExtractGEOExpSuppAll'><p>Extract Raw Count Matrix from Supplementary Files or Fortmat Supplementary Files to 10x.</p></a></li>
<li><a href='#ExtractGEOInfo'><p>Extract GEO Study Information.</p></a></li>
<li><a href='#ExtractGEOMeta'><p>Extract Sample Metadata from GEO.</p></a></li>
<li><a href='#ExtractGEOSubMeta'><p>Extract Sample Metadata.</p></a></li>
<li><a href='#ExtractHCAMeta'><p>Extract Metadata of Human Cell Atlas Projects with Attributes.</p></a></li>
<li><a href='#ExtractPanglaoDBComposition'><p>Extract Cell Type Composition of PanglaoDB Datasets.</p></a></li>
<li><a href='#ExtractPanglaoDBMeta'><p>Extract Metadata of scRNA-seq Datasets in PanglaoDB.</p></a></li>
<li><a href='#ExtractRun'><p>Extract Runs with GEO Accession Number or GSM Number.</p></a></li>
<li><a href='#ExtractZenodoMeta'><p>Prepare Dataframe with Zenodo DOIs.</p></a></li>
<li><a href='#ImportSeurat'><p>Convert Other Formats to SeuratObject.</p></a></li>
<li><a href='#PanglaoDBComposition'><p>All Sample Composition of PanglaoDB Datasets</p></a></li>
<li><a href='#PanglaoDBMeta'><p>All Sample Metadata of PanglaoDB Datasets</p></a></li>
<li><a href='#ParseCBDatasets'><p>Download UCSC Cell Browser Datasets.</p></a></li>
<li><a href='#ParseCELLxGENE'><p>Download CELLxGENE Datasets.</p></a></li>
<li><a href='#ParseGEO'><p>Download Matrix from GEO and Load to Seurat.</p></a></li>
<li><a href='#ParseHCA'><p>Download Human Cell Atlas Datasets.</p></a></li>
<li><a href='#ParsePanglaoDB'><p>Parse PanglaoDB Data.</p></a></li>
<li><a href='#ParseZenodo'><p>Download Data with Zenodo DOI.</p></a></li>
<li><a href='#SCEAnnData'><p>Data Format Conversion between SingleCellExperiment and AnnData.</p></a></li>
<li><a href='#SCELoom'><p>Data Format Conversion between SingleCellExperiment and loom.</p></a></li>
<li><a href='#ShowCBDatasets'><p>Show All Available Datasets in UCSC Cell Browser.</p></a></li>
<li><a href='#ShowCELLxGENEDatasets'><p>Show All Available Datasets in CELLxGENE.</p></a></li>
<li><a href='#ShowHCAProjects'><p>Show All Available Projects in Human Cell Atlas.</p></a></li>
<li><a href='#SplitSRA'><p>Split SRA to fastq Files and Format to 10x Standard Style.</p></a></li>
<li><a href='#StatDBAttribute'><p>Stat Database Attributes.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Access and Format Single-Cell RNA-Seq Datasets from Public
Resources</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yabing Song &lt;songyb0519@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The goal of 'scfetch' is to access and format single-cell RNA-seq datasets. It can be used to download single-cell RNA-seq
    datasets from widely used public resources, including GEO <a href="https://www.ncbi.nlm.nih.gov/geo/">https://www.ncbi.nlm.nih.gov/geo/</a>, Zenodo <a href="https://zenodo.org/">https://zenodo.org/</a>, 
    CELLxGENE <a href="https://cellxgene.cziscience.com/">https://cellxgene.cziscience.com/</a>, Human Cell Atlas <a href="https://www.humancellatlas.org/">https://www.humancellatlas.org/</a>, PanglaoDB <a href="https://panglaodb.se/index.html">https://panglaodb.se/index.html</a> 
    and UCSC Cell Browser <a href="https://cells.ucsc.edu/">https://cells.ucsc.edu/</a>. And, it can also be used to perform object conversion between SeuratObject <a href="https://satijalab.org/seurat/">https://satijalab.org/seurat/</a>, 
    loom <a href="http://loompy.org/">http://loompy.org/</a>, h5ad <a href="https://scanpy.readthedocs.io/en/stable/">https://scanpy.readthedocs.io/en/stable/</a>, SingleCellExperiment <a href="https://bioconductor.org/packages/release/bioc/html/scran.html">https://bioconductor.org/packages/release/bioc/html/scran.html</a>, 
    CellDataSet <a href="http://cole-trapnell-lab.github.io/monocle-release/">http://cole-trapnell-lab.github.io/monocle-release/</a> and cell_data_set <a href="https://cole-trapnell-lab.github.io/monocle3/">https://cole-trapnell-lab.github.io/monocle3/</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biobase, curl, data.table, dplyr, GEOquery, jsonlite,
magrittr, Matrix, openxlsx, parallel, pbapply, purrr,
rPanglaoDB, reticulate, Seurat, tibble, tools, utils,
SingleCellExperiment, SummarizedExperiment, scater,
LoomExperiment, httr, rlang, tidyr, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, scRNAseq, BiocStyle, htmltools, sceasy,
SeuratDisk, SeuratWrappers, zellkonverter, GEOfastq</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Additional_repositories:</td>
<td><a href="https://showteeth.github.io/drat/">https://showteeth.github.io/drat/</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/showteeth/scfetch">https://github.com/showteeth/scfetch</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/showteeth/scfetch/issues">https://github.com/showteeth/scfetch/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-21 09:18:20 UTC; soyabean</td>
</tr>
<tr>
<td>Author:</td>
<td>Yabing Song [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-21 18:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='Bam2Fastq'>Convert bam files to fastq files.</h2><span id='topic+Bam2Fastq'></span>

<h3>Description</h3>

<p>Convert bam files to fastq files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bam2Fastq(
  bam.folder = NULL,
  bam.path = NULL,
  bam.type = c("10x", "other"),
  pair.end = NULL,
  bamtofastq.path = NULL,
  bamtofastq.paras = "--nthreads 4",
  sort.name = FALSE,
  sort.thread = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Bam2Fastq_+3A_bam.folder">bam.folder</code></td>
<td>
<p>Folder contains bam files, obtained from <code>DownloadSRA</code>. Default: NULL.</p>
</td></tr>
<tr><td><code id="Bam2Fastq_+3A_bam.path">bam.path</code></td>
<td>
<p>Paths of bams. <code>bam.folder</code> and <code>bam.path</code> cannot be both NULL. Default: NULL.</p>
</td></tr>
<tr><td><code id="Bam2Fastq_+3A_bam.type">bam.type</code></td>
<td>
<p>The source of bam files, choose from 10x (e.g. CellRanger) or other. Default: 10x.</p>
</td></tr>
<tr><td><code id="Bam2Fastq_+3A_pair.end">pair.end</code></td>
<td>
<p>The bam files are pair-end or single-end, used when <code>bam.type</code> is other. Default: NULL (identify with flag).</p>
</td></tr>
<tr><td><code id="Bam2Fastq_+3A_bamtofastq.path">bamtofastq.path</code></td>
<td>
<p>Path to 10x bamtofastq (<code>bam.type</code> is 10x) or samtools (<code>bam.type</code> is other).
Default: NULL (conduct automatic detection with bamtofastq_linux/samtools).</p>
</td></tr>
<tr><td><code id="Bam2Fastq_+3A_bamtofastq.paras">bamtofastq.paras</code></td>
<td>
<p>Parameters for <code>bamtofastq.path</code>. Default: &quot;&ndash;nthreads 4&quot;.</p>
</td></tr>
<tr><td><code id="Bam2Fastq_+3A_sort.name">sort.name</code></td>
<td>
<p>Logical value, whether the bam files are sorted by name, required when <code>bam.type</code> is other. Default: FALSE.</p>
</td></tr>
<tr><td><code id="Bam2Fastq_+3A_sort.thread">sort.thread</code></td>
<td>
<p>The number of threads for bam sorting, used when <code>bam.type</code> is other. Default: 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL or paths of failed bams.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# need users to provide prefetch.path and bamtofastq.path
GSE138266.runs &lt;- ExtractRun(acce = "GSE138266", platform = "GPL18573")
GSE138266.down &lt;- DownloadBam(
  gsm.df = GSE138266.runs, prefetch.path = "/path/to/prefetch",
  out.folder = "/path/to/output"
)
GSE138266.convert &lt;- Bam2Fastq(
  bam.folder = "/path/to/output",
  bamtofastq.path = "/path/to/bamtofastq_linux or samtools",
  bamtofastq.paras = "--nthreads 4"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='DownloadBam'>Download bam.</h2><span id='topic+DownloadBam'></span>

<h3>Description</h3>

<p>Download bam.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DownloadBam(
  gsm.df,
  bam.type = c("10x", "other"),
  prefetch.path = NULL,
  samdump.path = NULL,
  out.folder = NULL,
  prefetch.paras = "-X 100G",
  samdump.paras = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DownloadBam_+3A_gsm.df">gsm.df</code></td>
<td>
<p>Dataframe contains GSM and Run numbers, obtained from <code>ExtractRun</code>.</p>
</td></tr>
<tr><td><code id="DownloadBam_+3A_bam.type">bam.type</code></td>
<td>
<p>The source of bam files to download, choose from 10x (e.g. CellRanger) or other. Default: 10x.</p>
</td></tr>
<tr><td><code id="DownloadBam_+3A_prefetch.path">prefetch.path</code></td>
<td>
<p>Path to prefetch. Default: NULL (conduct automatic detection).</p>
</td></tr>
<tr><td><code id="DownloadBam_+3A_samdump.path">samdump.path</code></td>
<td>
<p>Path to sam-dump, used when <code>bam.type</code> is other. Default: NULL (conduct automatic detection).</p>
</td></tr>
<tr><td><code id="DownloadBam_+3A_out.folder">out.folder</code></td>
<td>
<p>Output folder. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="DownloadBam_+3A_prefetch.paras">prefetch.paras</code></td>
<td>
<p>Parameters for <code>prefetch</code>. This should not contain &ndash;type or -T values. Default: &quot;-X 100G&quot;.</p>
</td></tr>
<tr><td><code id="DownloadBam_+3A_samdump.paras">samdump.paras</code></td>
<td>
<p>Parameters for <code>sam-dump</code>. Default: &quot;&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains failed runs or NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# need users to provide prefetch.path
GSE138266.runs &lt;- ExtractRun(acce = "GSE138266", platform = "GPL18573")
GSE138266.down &lt;- DownloadBam(
  gsm.df = GSE138266.runs, bam.type = "10x",
  prefetch.path = "/path/to/prefetch",
  out.folder = "/path/to/output"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='DownloadSRA'>Download SRA.</h2><span id='topic+DownloadSRA'></span>

<h3>Description</h3>

<p>Download SRA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DownloadSRA(
  gsm.df,
  prefetch.path = NULL,
  out.folder = NULL,
  prefetch.paras = "-X 100G"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DownloadSRA_+3A_gsm.df">gsm.df</code></td>
<td>
<p>Dataframe contains GSM and Run numbers, obtained from <code>ExtractRun</code>.</p>
</td></tr>
<tr><td><code id="DownloadSRA_+3A_prefetch.path">prefetch.path</code></td>
<td>
<p>Path to prefetch. Default: NULL (conduct automatic detection).</p>
</td></tr>
<tr><td><code id="DownloadSRA_+3A_out.folder">out.folder</code></td>
<td>
<p>Output folder. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="DownloadSRA_+3A_prefetch.paras">prefetch.paras</code></td>
<td>
<p>Parameters for <code>prefetch</code>. Default: &quot;-X 100G&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains failed runs or NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# need users to provide the prefetch.path and out.folder
GSE186003.runs &lt;- ExtractRun(acce = "GSE186003", platform = "GPL24247")
GSE186003.down &lt;- DownloadSRA(
  gsm.df = GSE186003.runs, prefetch.path = "/path/to/prefetch",
  out.folder = "/path/to/output"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='ExportSeurat'>Export SeuratObject to Other Formats.</h2><span id='topic+ExportSeurat'></span>

<h3>Description</h3>

<p>Export SeuratObject to Other Formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExportSeurat(
  seu.obj,
  assay = NULL,
  reduction = NULL,
  to = c("SCE", "AnnData", "CellDataSet", "cell_data_set", "loom"),
  anndata.file = NULL,
  loom.file = NULL,
  conda.path = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExportSeurat_+3A_seu.obj">seu.obj</code></td>
<td>
<p>A seurat object.</p>
</td></tr>
<tr><td><code id="ExportSeurat_+3A_assay">assay</code></td>
<td>
<p>Which assay to use. Default: NULL (get with <code>DefaultAssay</code>).</p>
</td></tr>
<tr><td><code id="ExportSeurat_+3A_reduction">reduction</code></td>
<td>
<p>Name of DimReduc to set to main reducedDim in cds.</p>
</td></tr>
<tr><td><code id="ExportSeurat_+3A_to">to</code></td>
<td>
<p>The target format, chosen from &quot;SCE&quot; (SingleCellExperiment), &quot;AnnData&quot;, &quot;CellDataSet&quot;, &quot;cell_data_set&quot;, &quot;loom&quot;.
Default: &quot;SCE&quot;.</p>
</td></tr>
<tr><td><code id="ExportSeurat_+3A_anndata.file">anndata.file</code></td>
<td>
<p>File used to save AnnData results. Default: NULL.</p>
</td></tr>
<tr><td><code id="ExportSeurat_+3A_loom.file">loom.file</code></td>
<td>
<p>File used to save loom results. Default: NULL.</p>
</td></tr>
<tr><td><code id="ExportSeurat_+3A_conda.path">conda.path</code></td>
<td>
<p>Conda environment path, used when <code>to</code> is &quot;AnnData&quot;. Default: NULL.</p>
</td></tr>
<tr><td><code id="ExportSeurat_+3A_...">...</code></td>
<td>
<p>Parameter for <code>as.SingleCellExperiment</code>, <code>sceasy::convertFormat</code>, <code>as.CellDataSet</code>,
<code>as.cell_data_set</code>, <code>SaveLoom</code>, corresponding to <code>to</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object corresponding to <code>to</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(Seurat)
# export to SingleCellExperiment
sce.obj &lt;- ExportSeurat(seu.obj = pbmc_small, assay = "RNA", to = "SCE")
# export to CellDataSet
cds.obj &lt;- ExportSeurat(seu.obj = pbmc_small, assay = "RNA", reduction = "tsne", to = "CellDataSet")
# export to cell_data_set
cds3.obj &lt;- ExportSeurat(seu.obj = pbmc_small, assay = "RNA", to = "cell_data_set")
# export to AnnData, need users to provide the conda path and the output file
ExportSeurat(
  seu.obj = pbmc_small, assay = "RNA", to = "AnnData", conda.path = "/path/to/anaconda3",
  anndata.file = "/path/to/pbmc_small.h5ad"
)
# export to loom, need users to provide the output file
ExportSeurat(
  seu.obj = pbmc_small, assay = "RNA", to = "loom",
  loom.file = "/path/to/pbmc_small.loom"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='ExtractCBComposition'>Extract Cell Type Composition of UCSC Cell Browser Datasets.</h2><span id='topic+ExtractCBComposition'></span>

<h3>Description</h3>

<p>Extract Cell Type Composition of UCSC Cell Browser Datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractCBComposition(
  json.folder = NULL,
  sample.df = NULL,
  all.samples.df = NULL,
  collection = NULL,
  sub.collection = NULL,
  organ = NULL,
  disease = NULL,
  organism = NULL,
  project = NULL,
  fuzzy.match = TRUE,
  cell.num = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractCBComposition_+3A_json.folder">json.folder</code></td>
<td>
<p>Folder contains datasets json files, same as <code>json.folder</code> of <code>ShowCBDatasets</code>.
Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_sample.df">sample.df</code></td>
<td>
<p>Dataframe contains used datasets. Default: NULL.</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_all.samples.df">all.samples.df</code></td>
<td>
<p>Dataframe contains all samples metadata, obtained with <code>ShowCBDatasets</code>. Default: NULL.
<code>sample.df</code> and <code>all.samples.df</code> cannot be both NULL.</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_collection">collection</code></td>
<td>
<p>The collection of the datasets, corresponds to <code>shortLabel</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_sub.collection">sub.collection</code></td>
<td>
<p>The sub-collection of the datasets, corresponds to <code>subLabel</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_organ">organ</code></td>
<td>
<p>The organ of the datasets, corresponds to <code>body_parts</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_disease">disease</code></td>
<td>
<p>The disease of the datasets, corresponds to <code>diseases</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_organism">organism</code></td>
<td>
<p>The specie of the datasets, corresponds to <code>organisms</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_project">project</code></td>
<td>
<p>The project of the datasets, corresponds to <code>projects</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_fuzzy.match">fuzzy.match</code></td>
<td>
<p>Logical value, whether to perform fuzzy match with provided attribute values. Default: TRUE.</p>
</td></tr>
<tr><td><code id="ExtractCBComposition_+3A_cell.num">cell.num</code></td>
<td>
<p>Cell number filter. If NULL, no filter; if one value, lower filter; if two values, low and high filter. Deault: NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains sample information and cell type composition.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# lazy mode, load datasets json files locally, need users to provide json folder
ucsc.cb.samples &lt;- ShowCBDatasets(lazy = TRUE, json.folder = NULL, update = FALSE)
# cell number is between 1000 and 2000
hbb.sample.df &lt;- ExtractCBDatasets(
  all.samples.df = ucsc.cb.samples, organ = c("brain", "blood"),
  organism = "Human (H. sapiens)", cell.num = c(1000, 2000)
)
hbb.sample.ct &lt;- ExtractCBComposition(json.folder = NULL, sample.df = hbb.sample.df)

## End(Not run)
</code></pre>

<hr>
<h2 id='ExtractCBDatasets'>Extract UCSC Cell Browser Datasets with Attributes.</h2><span id='topic+ExtractCBDatasets'></span>

<h3>Description</h3>

<p>Extract UCSC Cell Browser Datasets with Attributes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractCBDatasets(
  all.samples.df,
  collection = NULL,
  sub.collection = NULL,
  organ = NULL,
  disease = NULL,
  organism = NULL,
  project = NULL,
  fuzzy.match = TRUE,
  cell.num = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractCBDatasets_+3A_all.samples.df">all.samples.df</code></td>
<td>
<p>Dataframe contains all samples metadata, obtained with <code>ShowCBDatasets</code>.</p>
</td></tr>
<tr><td><code id="ExtractCBDatasets_+3A_collection">collection</code></td>
<td>
<p>The collection of the datasets, corresponds to <code>shortLabel</code> column
of <code>ShowCBDatasets</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBDatasets_+3A_sub.collection">sub.collection</code></td>
<td>
<p>The sub-collection of the datasets, corresponds to <code>subLabel</code> column
of <code>ShowCBDatasets</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBDatasets_+3A_organ">organ</code></td>
<td>
<p>The organ of the datasets, corresponds to <code>body_parts</code> column
of <code>ShowCBDatasets</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBDatasets_+3A_disease">disease</code></td>
<td>
<p>The disease of the datasets, corresponds to <code>diseases</code> column
of <code>ShowCBDatasets</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBDatasets_+3A_organism">organism</code></td>
<td>
<p>The specie of the datasets, corresponds to <code>organisms</code> column
of <code>ShowCBDatasets</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBDatasets_+3A_project">project</code></td>
<td>
<p>The project of the datasets, corresponds to <code>projects</code> column
of <code>ShowCBDatasets</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ExtractCBDatasets_+3A_fuzzy.match">fuzzy.match</code></td>
<td>
<p>Logical value, whether to perform fuzzy match with provided attribute values. Default: TRUE.</p>
</td></tr>
<tr><td><code id="ExtractCBDatasets_+3A_cell.num">cell.num</code></td>
<td>
<p>Cell number filter. If NULL, no filter; if one value, lower filter; if two values, low and high filter.
Deault: NULL(without filtering).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains filtered datasets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# lazy mode, load datasets json files locally, need users to provide json folder
ucsc.cb.samples &lt;- ShowCBDatasets(lazy = TRUE, json.folder = NULL, update = FALSE)
# cell number is between 1000 and 2000
hbb.sample.df &lt;- ExtractCBDatasets(
  all.samples.df = ucsc.cb.samples, organ = c("brain", "blood"),
  organism = "Human (H. sapiens)", cell.num = c(1000, 2000)
)

## End(Not run)
</code></pre>

<hr>
<h2 id='ExtractCELLxGENEMeta'>Extract Metadata of CELLxGENE Datasets with Attributes.</h2><span id='topic+ExtractCELLxGENEMeta'></span>

<h3>Description</h3>

<p>Extract Metadata of CELLxGENE Datasets with Attributes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractCELLxGENEMeta(
  all.samples.df,
  organism = NULL,
  ethnicity = NULL,
  sex = NULL,
  tissue = NULL,
  disease = NULL,
  assay = NULL,
  suspension.type = NULL,
  cell.type = NULL,
  cell.num = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_all.samples.df">all.samples.df</code></td>
<td>
<p>All detail information of CELLxGENE datasets, obtained with <code>ShowCELLxGENEDatasets</code>.</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_organism">organism</code></td>
<td>
<p>The organism of the datasets, choose from &quot;Homo sapiens&quot;, &quot;Mus musculus&quot;, &quot;Callithrix jacchus&quot;,
&quot;Macaca mulatta&quot;, &quot;Sus scrofa domesticus&quot;, one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_ethnicity">ethnicity</code></td>
<td>
<p>The ethnicity of the datasets, choose from &quot;Asian&quot;, &quot;European&quot;, &quot;unknown&quot;, &quot;na&quot;, &quot;African&quot;, &quot;Bangladeshi&quot;,
&quot;British&quot;, &quot;Irish&quot;, &quot;East Asian&quot;, &quot;African American&quot;, &quot;Hispanic or Latin American&quot;, &quot;African American or Afro-Caribbean&quot;,
&quot;European American&quot;, &quot;Finnish&quot;, &quot;Indian&quot;, &quot;Japanese&quot;, &quot;Korean&quot;, &quot;Malaysian&quot;, &quot;Singaporean Chinese&quot;, &quot;American&quot;, &quot;Pacific Islander&quot;,
&quot;admixed ancestry&quot;, &quot;Eskimo&quot;, &quot;Han Chinese&quot;, &quot;Greater Middle Eastern  (Middle Eastern, North African or Persian)&quot;, &quot;multiethnic&quot;,
&quot;Jewish Israeli&quot;, &quot;South Asian&quot;, &quot;Oceanian&quot;, &quot;Chinese&quot;, one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_sex">sex</code></td>
<td>
<p>The sex of the datasets, choose from &quot;female&quot;, &quot;male&quot;, &quot;unknown&quot;, one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_tissue">tissue</code></td>
<td>
<p>The tissue of the datasets, obtain available values with <code>StatDBAttribute</code>. One or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_disease">disease</code></td>
<td>
<p>The disease of the datasets, obtain available values with <code>StatDBAttribute</code>. One or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_assay">assay</code></td>
<td>
<p>The assay of the datasets, choose from &quot;10x 3' v1&quot;, &quot;10x 3' v2&quot;, &quot;10x 3' v3&quot;, &quot;10x 3' transcription profiling&quot;,
&quot;10x 5' v1&quot;, &quot;10x 5' v2&quot;, &quot;10x 5' transcription profiling&quot;, &quot;10x multiome&quot;, &quot;10x scATAC-seq&quot;, &quot;sci-RNA-seq&quot;, &quot;Drop-seq&quot;,
&quot;Smart-seq&quot;, &quot;Smart-seq2&quot;, &quot;Smart-seq v4&quot;, &quot;snmC-Seq2&quot;, &quot;Visium Spatial Gene Expression&quot;, &quot;Seq-Well&quot;, &quot;Seq-Well S3&quot;, &quot;Patch-seq&quot;,
&quot;sci-Plex&quot;, &quot;BD Rhapsody Targeted mRNA&quot;, &quot;BD Rhapsody Whole Transcriptome Analysis&quot;, &quot;Slide-seqV2&quot;, &quot;GEXSCOPE technology&quot;, &quot;inDrop&quot;,
&quot;microwell-seq&quot;, &quot;CEL-seq2&quot;, &quot;STRT-seq&quot;, &quot;DroNc-seq&quot;, &quot;MERFISH&quot;, &quot;scATAC-seq&quot;, &quot;MARS-seq&quot;, &quot;TruDrop&quot;, one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_suspension.type">suspension.type</code></td>
<td>
<p>The suspension type of the datasets, choose from &quot;nucleus&quot;, &quot;cell&quot;, &quot;na&quot;, one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_cell.type">cell.type</code></td>
<td>
<p>The cell type of the datasets, obtain available values with <code>StatDBAttribute</code>. One or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractCELLxGENEMeta_+3A_cell.num">cell.num</code></td>
<td>
<p>Cell number filter. If NULL, no filter; if one value, lower filter; if two values, low and high filter.
Deault: NULL(without filtering).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains filtered datasets.
</p>


<h3>References</h3>

<p>https://gist.github.com/ivirshup/f1a1603db69de3888eacb4bdb6a9317a
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# all available datasets
all.cellxgene.datasets &lt;- ShowCELLxGENEDatasets()
# human 10x v2 and v3 datasets
human.10x.cellxgene.meta &lt;- ExtractCELLxGENEMeta(
  all.samples.df = all.cellxgene.datasets,
  assay = c("10x 3' v2", "10x 3' v3"),
  organism = "Homo sapiens"
)

</code></pre>

<hr>
<h2 id='ExtractGEOExp'>Extract Raw Count Matrix or Fortmat Supplementary Files to 10x.</h2><span id='topic+ExtractGEOExp'></span>

<h3>Description</h3>

<p>Extract Raw Count Matrix or Fortmat Supplementary Files to 10x.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractGEOExp(
  pf.obj,
  acce,
  supp.idx = 1,
  down.supp = FALSE,
  timeout = 3600,
  supp.type = c("count", "10x"),
  out.folder = NULL,
  gene2feature = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractGEOExp_+3A_pf.obj">pf.obj</code></td>
<td>
<p>GEO object of platform.</p>
</td></tr>
<tr><td><code id="ExtractGEOExp_+3A_acce">acce</code></td>
<td>
<p>GEO accession number.</p>
</td></tr>
<tr><td><code id="ExtractGEOExp_+3A_supp.idx">supp.idx</code></td>
<td>
<p>The index of supplementary files to download. Default: 1.</p>
</td></tr>
<tr><td><code id="ExtractGEOExp_+3A_down.supp">down.supp</code></td>
<td>
<p>Logical value, whether to download supplementary files to create count matrix. If TRUE, always
download supplementary files. If FALSE, use <code>ExpressionSet</code> (If contains non-integer or emoty,
download supplementary files automatically). Default: FALSE.</p>
</td></tr>
<tr><td><code id="ExtractGEOExp_+3A_timeout">timeout</code></td>
<td>
<p>Timeout for <code><a href="utils.html#topic+download.file">download.file</a></code>. Default: 3600.</p>
</td></tr>
<tr><td><code id="ExtractGEOExp_+3A_supp.type">supp.type</code></td>
<td>
<p>The type of downloaded supplementary files, choose from count (count matrix file or single count matrix file)
and 10x (cellranger output files, contains barcodes, genes/features and matrix). Default: count.</p>
</td></tr>
<tr><td><code id="ExtractGEOExp_+3A_out.folder">out.folder</code></td>
<td>
<p>Output folder to save 10x files. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ExtractGEOExp_+3A_gene2feature">gene2feature</code></td>
<td>
<p>Logical value, whether to rename <code>genes.tsv.gz</code> to <code>features.tsv.gz</code>. Default: TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Count matrix (<code>supp.type</code> is count) or NULL (<code>supp.type</code> is 10x).
</p>

<hr>
<h2 id='ExtractGEOExpSupp'>Extract Raw Count Matrix from Supplementary Files.</h2><span id='topic+ExtractGEOExpSupp'></span>

<h3>Description</h3>

<p>Extract Raw Count Matrix from Supplementary Files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractGEOExpSupp(acce, timeout = 3600, supp.idx = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractGEOExpSupp_+3A_acce">acce</code></td>
<td>
<p>GEO accession number.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSupp_+3A_timeout">timeout</code></td>
<td>
<p>Timeout for <code><a href="utils.html#topic+download.file">download.file</a></code>. Default: 3600.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSupp_+3A_supp.idx">supp.idx</code></td>
<td>
<p>The index of supplementary files to download. Default: 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe.
</p>

<hr>
<h2 id='ExtractGEOExpSupp10x'>Fortmat Supplementary Files to 10x.</h2><span id='topic+ExtractGEOExpSupp10x'></span>

<h3>Description</h3>

<p>Fortmat Supplementary Files to 10x.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractGEOExpSupp10x(
  acce,
  supp.idx = 1,
  timeout = 3600,
  out.folder = NULL,
  gene2feature = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractGEOExpSupp10x_+3A_acce">acce</code></td>
<td>
<p>GEO accession number.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSupp10x_+3A_supp.idx">supp.idx</code></td>
<td>
<p>The index of supplementary files to download. Default: 1.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSupp10x_+3A_timeout">timeout</code></td>
<td>
<p>Timeout for <code><a href="utils.html#topic+download.file">download.file</a></code>. Default: 3600.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSupp10x_+3A_out.folder">out.folder</code></td>
<td>
<p>Output folder to save 10x files. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSupp10x_+3A_gene2feature">gene2feature</code></td>
<td>
<p>Logical value, whether to rename <code>genes.tsv.gz</code> to <code>features.tsv.gz</code>.
Default: TURE.</p>
</td></tr>
</table>

<hr>
<h2 id='ExtractGEOExpSuppAll'>Extract Raw Count Matrix from Supplementary Files or Fortmat Supplementary Files to 10x.</h2><span id='topic+ExtractGEOExpSuppAll'></span>

<h3>Description</h3>

<p>Extract Raw Count Matrix from Supplementary Files or Fortmat Supplementary Files to 10x.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractGEOExpSuppAll(
  acce,
  supp.idx = 1,
  timeout = 3600,
  supp.type = c("count", "10x"),
  out.folder = NULL,
  gene2feature = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractGEOExpSuppAll_+3A_acce">acce</code></td>
<td>
<p>GEO accession number.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSuppAll_+3A_supp.idx">supp.idx</code></td>
<td>
<p>The index of supplementary files to download. Default: 1.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSuppAll_+3A_timeout">timeout</code></td>
<td>
<p>Timeout for <code><a href="utils.html#topic+download.file">download.file</a></code>. Default: 3600.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSuppAll_+3A_supp.type">supp.type</code></td>
<td>
<p>The type of downloaded supplementary files, choose from count (count matrix file or single count matrix file)
and 10x (cellranger output files, contains barcodes, genes/features and matrix). Default: count.</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSuppAll_+3A_out.folder">out.folder</code></td>
<td>
<p>Output folder to save 10x files. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ExtractGEOExpSuppAll_+3A_gene2feature">gene2feature</code></td>
<td>
<p>Logical value, whether to rename <code>genes.tsv.gz</code> to <code>features.tsv.gz</code>. Default: TRUE.
Default: TURE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Count matrix (<code>supp.type</code> is count) or NULL (<code>supp.type</code> is 10x).
</p>

<hr>
<h2 id='ExtractGEOInfo'>Extract GEO Study Information.</h2><span id='topic+ExtractGEOInfo'></span>

<h3>Description</h3>

<p>Extract GEO Study Information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractGEOInfo(pf.obj, sample.wise = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractGEOInfo_+3A_pf.obj">pf.obj</code></td>
<td>
<p>GEO object of platform.</p>
</td></tr>
<tr><td><code id="ExtractGEOInfo_+3A_sample.wise">sample.wise</code></td>
<td>
<p>Logical value, whether to extract sample-wise information. Default: FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe.
</p>

<hr>
<h2 id='ExtractGEOMeta'>Extract Sample Metadata from GEO.</h2><span id='topic+ExtractGEOMeta'></span>

<h3>Description</h3>

<p>Extract Sample Metadata from GEO.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractGEOMeta(acce, platform = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractGEOMeta_+3A_acce">acce</code></td>
<td>
<p>GEO accession number.</p>
</td></tr>
<tr><td><code id="ExtractGEOMeta_+3A_platform">platform</code></td>
<td>
<p>Platform information/field. Default: NULL (all platforms).</p>
</td></tr>
<tr><td><code id="ExtractGEOMeta_+3A_...">...</code></td>
<td>
<p>Parameters for <code>getGEO</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains all metadata of provided GEO accession number.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# users may need to set the size of the connection buffer
# Sys.setenv("VROOM_CONNECTION_SIZE" = 131072 * 60)
# extract metadata of specified platform
GSE200257.meta &lt;- ExtractGEOMeta(acce = "GSE200257", platform = "GPL24676")

</code></pre>

<hr>
<h2 id='ExtractGEOSubMeta'>Extract Sample Metadata.</h2><span id='topic+ExtractGEOSubMeta'></span>

<h3>Description</h3>

<p>Extract Sample Metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractGEOSubMeta(pf.obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractGEOSubMeta_+3A_pf.obj">pf.obj</code></td>
<td>
<p>GEO object of platform.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe.
</p>

<hr>
<h2 id='ExtractHCAMeta'>Extract Metadata of Human Cell Atlas Projects with Attributes.</h2><span id='topic+ExtractHCAMeta'></span>

<h3>Description</h3>

<p>Extract Metadata of Human Cell Atlas Projects with Attributes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractHCAMeta(
  all.projects.df,
  organism = NULL,
  sex = NULL,
  organ = NULL,
  organ.part = NULL,
  disease = NULL,
  sample.type = NULL,
  preservation.method = NULL,
  protocol = NULL,
  suspension.type = NULL,
  cell.type = NULL,
  cell.num = NULL,
  sequencing.type = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractHCAMeta_+3A_all.projects.df">all.projects.df</code></td>
<td>
<p>All detail information of HCA projects, obtained with <code>ShowHCAProjects</code>.</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_organism">organism</code></td>
<td>
<p>The organism of the projects, choose from &quot;Homo sapiens&quot;, &quot;Mus musculus&quot;,
&quot;Macaca mulatta&quot;, &quot;canis lupus familiaris&quot;, one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_sex">sex</code></td>
<td>
<p>The sex of the projects, choose from &quot;female&quot;, &quot;male&quot;, &quot;mixed&quot;, &quot;unknown&quot;,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_organ">organ</code></td>
<td>
<p>The organ of the projects (e.g. brain), obtain available values with <code>StatDBAttribute</code>,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_organ.part">organ.part</code></td>
<td>
<p>The organ part of the projects (e.g. cortex), obtain available values with <code>StatDBAttribute</code>,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_disease">disease</code></td>
<td>
<p>The disease of the projects (e.g. normal), obtain available values with <code>StatDBAttribute</code>,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_sample.type">sample.type</code></td>
<td>
<p>The sex of the projects, choose from &quot;specimens&quot;, &quot;organoids&quot;, &quot;cellLines&quot;,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_preservation.method">preservation.method</code></td>
<td>
<p>The preservation method of the projects (e.g. fresh), obtain available values with <code>StatDBAttribute</code>,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_protocol">protocol</code></td>
<td>
<p>The protocol of the projects (e.g. 10x 3' v2), obtain available values with <code>StatDBAttribute</code>,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_suspension.type">suspension.type</code></td>
<td>
<p>The suspension type of the projects, choose from &quot;single cell&quot;, &quot;single nucleus&quot;, &quot;bulk cell&quot;, &quot;bulk nuclei&quot;,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_cell.type">cell.type</code></td>
<td>
<p>The cell type of the projects (e.g. neuron), obtain available values with <code>StatDBAttribute</code>,
one or multiple values. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_cell.num">cell.num</code></td>
<td>
<p>Cell number filter. If NULL, no filter; if one value, lower filter; if two values, low and high filter.
Deault: NULL(without filtering).</p>
</td></tr>
<tr><td><code id="ExtractHCAMeta_+3A_sequencing.type">sequencing.type</code></td>
<td>
<p>The sequencing instrument type of the projects (e.g. illumina hiseq 2500),
obtain available values with <code>StatDBAttribute</code>, one or multiple values. Default: NULL (All).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains filtered projects.
</p>


<h3>References</h3>

<p>https://bioconductor.org/packages/release/bioc/html/hca.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# all available projects
all.hca.projects &lt;- ShowHCAProjects()
# all human projects
all.human.projects &lt;- ExtractHCAMeta(all.projects.df = all.hca.projects, organism = "Homo sapiens")
# all human and 10x 3' v2
all.human.10x.projects &lt;- ExtractHCAMeta(
  all.projects.df = all.hca.projects,
  organism = "Homo sapiens",
  protocol = c("10x 3' v2", "10x 3' v3")
)

</code></pre>

<hr>
<h2 id='ExtractPanglaoDBComposition'>Extract Cell Type Composition of PanglaoDB Datasets.</h2><span id='topic+ExtractPanglaoDBComposition'></span>

<h3>Description</h3>

<p>Extract Cell Type Composition of PanglaoDB Datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractPanglaoDBComposition(
  sra = NULL,
  srs = NULL,
  species = NULL,
  protocol = NULL,
  tissue = NULL,
  local.data = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractPanglaoDBComposition_+3A_sra">sra</code></td>
<td>
<p>The SRA identifier of the datasets, obtain available values with <code>StatDBAttribute</code>,
one or multiple value. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBComposition_+3A_srs">srs</code></td>
<td>
<p>The SRS identifier of the datasets, obtain available values with <code>StatDBAttribute</code>,
one or multiple value. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBComposition_+3A_species">species</code></td>
<td>
<p>The species of the datasets, choose from &quot;Homo sapiens&quot;, &quot;Mus musculus&quot;, one or multiple value. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBComposition_+3A_protocol">protocol</code></td>
<td>
<p>Protocol used to generate the datasets, choose from &quot;10x chromium&quot;, &quot;drop-seq&quot;, &quot;microwell-seq&quot;,
&quot;C1 Fluidigm&quot;, &quot;inDrops&quot;, &quot;Smart-seq2&quot;, &quot;CEL-seq&quot;, one or multiple value. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBComposition_+3A_tissue">tissue</code></td>
<td>
<p>The tissue of the datasets, obtain available values with <code>StatDBAttribute</code>,
one or multiple value. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBComposition_+3A_local.data">local.data</code></td>
<td>
<p>Logical value, whether to use local data (PanglaoDB is no longer maintained). Default: TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains sample metadata, cluster, cell number and cell type information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
human.composition &lt;- ExtractPanglaoDBComposition(
  species = "Homo sapiens",
  protocol = c("Smart-seq2", "10x chromium")
)

</code></pre>

<hr>
<h2 id='ExtractPanglaoDBMeta'>Extract Metadata of scRNA-seq Datasets in PanglaoDB.</h2><span id='topic+ExtractPanglaoDBMeta'></span>

<h3>Description</h3>

<p>Extract Metadata of scRNA-seq Datasets in PanglaoDB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractPanglaoDBMeta(
  species = NULL,
  protocol = NULL,
  tissue = NULL,
  cell.num = NULL,
  show.cell.type = TRUE,
  local.data = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractPanglaoDBMeta_+3A_species">species</code></td>
<td>
<p>The species of the datasets, choose from &quot;Homo sapiens&quot;, &quot;Mus musculus&quot;, one or multiple value. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBMeta_+3A_protocol">protocol</code></td>
<td>
<p>Protocol used to generate the datasets, choose from &quot;10x chromium&quot;, &quot;drop-seq&quot;, &quot;microwell-seq&quot;,
&quot;C1 Fluidigm&quot;, &quot;inDrops&quot;, &quot;Smart-seq2&quot;, &quot;CEL-seq&quot;, one or multiple value. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBMeta_+3A_tissue">tissue</code></td>
<td>
<p>The tissue of the datasets, obtain available values with <code>StatDBAttribute</code>. Default: NULL (All).</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBMeta_+3A_cell.num">cell.num</code></td>
<td>
<p>Cell number filter. If NULL, no filter; if one value, lower filter; if two values, low and high filter. Deault: NULL.</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBMeta_+3A_show.cell.type">show.cell.type</code></td>
<td>
<p>Logical value, whether to show inferred cell type. Default: TRUE.</p>
</td></tr>
<tr><td><code id="ExtractPanglaoDBMeta_+3A_local.data">local.data</code></td>
<td>
<p>Logical value, whether to use local data (PanglaoDB is no longer maintained). Default: TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains SRA, SRS, Tissue, Protocol, Species, Cells, CellType (inferred).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
human.meta &lt;- ExtractPanglaoDBMeta(
  species = "Homo sapiens",
  protocol = c("Smart-seq2", "10x chromium"),
  cell.num = c(1000, 2000)
)

</code></pre>

<hr>
<h2 id='ExtractRun'>Extract Runs with GEO Accession Number or GSM Number.</h2><span id='topic+ExtractRun'></span>

<h3>Description</h3>

<p>Extract Runs with GEO Accession Number or GSM Number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractRun(gsm = NULL, acce = NULL, platform = NULL, parallel = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractRun_+3A_gsm">gsm</code></td>
<td>
<p>GSM number. Default: NULL (use <code>acce</code>).</p>
</td></tr>
<tr><td><code id="ExtractRun_+3A_acce">acce</code></td>
<td>
<p>GEO accession number. Default: NULL (use <code>gsm</code>).
<code>acce</code> and <code>gsm</code> cannot be both NULL.</p>
</td></tr>
<tr><td><code id="ExtractRun_+3A_platform">platform</code></td>
<td>
<p>Platform information/field. Default: NULL (all platforms).</p>
</td></tr>
<tr><td><code id="ExtractRun_+3A_parallel">parallel</code></td>
<td>
<p>Logical value, whether to process GSM parallelly. Default: TRUE.</p>
</td></tr>
<tr><td><code id="ExtractRun_+3A_...">...</code></td>
<td>
<p>Parameters for <code>getGEO</code>. Used when <code>acce</code> is not NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains GSM and Runs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
GSE186003.runs &lt;- ExtractRun(acce = "GSE186003", platform = "GPL24247", parallel = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ExtractZenodoMeta'>Prepare Dataframe with Zenodo DOIs.</h2><span id='topic+ExtractZenodoMeta'></span>

<h3>Description</h3>

<p>Prepare Dataframe with Zenodo DOIs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractZenodoMeta(doi, file.ext = c("rdata", "h5ad"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractZenodoMeta_+3A_doi">doi</code></td>
<td>
<p>A vector of Zenodo DOIs, should start with &quot;10.5281/zenodo.&quot;.</p>
</td></tr>
<tr><td><code id="ExtractZenodoMeta_+3A_file.ext">file.ext</code></td>
<td>
<p>The valid file extension for download. When NULL, use all files. Default: c(&quot;rdata&quot;, &quot;h5ad&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains files with valid extension in given Zenodo DOI.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
zebrafish.df &lt;- ExtractZenodoMeta(doi = "10.5281/zenodo.7243603")
ExtractZenodoMeta(doi = "10.5281/zenodo.48065") # Restricted Access
# vector of dois
multi.dois &lt;- ExtractZenodoMeta(doi = c(
  "1111", "10.5281/zenodo.7243603",
  "10.5281/zenodo.7244441"
))

</code></pre>

<hr>
<h2 id='ImportSeurat'>Convert Other Formats to SeuratObject.</h2><span id='topic+ImportSeurat'></span>

<h3>Description</h3>

<p>Convert Other Formats to SeuratObject.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ImportSeurat(
  obj = NULL,
  assay = "RNA",
  from = c("SCE", "AnnData", "CellDataSet", "cell_data_set", "loom"),
  count.assay = "counts",
  data.assay = "logcounts",
  slot = "counts",
  anndata.file = NULL,
  loom.file = NULL,
  conda.path = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ImportSeurat_+3A_obj">obj</code></td>
<td>
<p>Other formats object (eg: SingleCellExperiment, CellDataSet).
Default: NULL (used when <code>from</code> is &quot;AnnData&quot;).</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_assay">assay</code></td>
<td>
<p>Assay name to store expression matrices in SeuratObject. Default: RNA.</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_from">from</code></td>
<td>
<p>The source formats, chosen from &quot;SCE&quot; (SingleCellExperiment), &quot;AnnData&quot;, &quot;CellDataSet&quot;, &quot;cell_data_set&quot;. Default: &quot;SCE&quot;.</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_count.assay">count.assay</code></td>
<td>
<p>The assay of source formats to save raw counts,
used when <code>from</code> is &quot;SCE&quot; or cell_data_set. Default: counts.</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_data.assay">data.assay</code></td>
<td>
<p>The assay of source formats to save log transformed counts,
used when <code>from</code> is &quot;SCE&quot; or cell_data_set. Default: logcounts.</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_slot">slot</code></td>
<td>
<p>Slot to store expression data as, used when <code>from</code> is &quot;CellDataSet&quot;. Default: counts.</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_anndata.file">anndata.file</code></td>
<td>
<p>The file contains AnnData. Default: NULL.</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_loom.file">loom.file</code></td>
<td>
<p>The file contains loom. Default: NULL.</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_conda.path">conda.path</code></td>
<td>
<p>Conda environment path, used when <code>from</code> is &quot;AnnData&quot;. Default: NULL.</p>
</td></tr>
<tr><td><code id="ImportSeurat_+3A_...">...</code></td>
<td>
<p>Parameter for <code>as.Seurat</code>, <code>sceasy::convertFormat</code>, <code>as.Seurat</code>, <code>as.Seurat</code>,
<code>as.Seurat</code>, corresponding to <code>from</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# import data from SingleCellExperiment
seu.obj &lt;- ImportSeurat(
  obj = sce.obj, from = "SCE", count.assay = "counts",
  data.assay = "logcounts", assay = "RNA"
)
# import data from CellDataSet
seu.obj &lt;- ImportSeurat(obj = cds.obj, from = "CellDataSet", count.assay = "counts", assay = "RNA")
# import data from cell_data_set
seu.obj &lt;- ImportSeurat(
  obj = sce.obj, from = "cell_data_set", count.assay = "counts",
  data.assay = "logcounts", assay = "RNA"
)
# import data from AnnData, need users to provide the file for conversion
seu.obj &lt;- ImportSeurat(anndata.file = "path/to/h5ad", from = "AnnData", assay = "RNA")
# import data from loom, need users to provide the file for conversion
seu.obj &lt;- ImportSeurat(loom.file = "path/to/loom", from = "loom")

## End(Not run)
</code></pre>

<hr>
<h2 id='PanglaoDBComposition'>All Sample Composition of PanglaoDB Datasets</h2><span id='topic+PanglaoDBComposition'></span>

<h3>Description</h3>

<p>A dataset contains all sample composition of PanglaoDB datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PanglaoDBComposition
</code></pre>


<h3>Format</h3>

<p>A data frame with 19,449 rows and 8 variables:
</p>

<dl>
<dt>SRA</dt><dd><p>The SRA identifier</p>
</dd>
<dt>SRS</dt><dd><p>The SRS identifier</p>
</dd>
<dt>Tissue</dt><dd><p>The tissue of the sample</p>
</dd>
<dt>Protocol</dt><dd><p>The protocol used to generate this sample</p>
</dd>
<dt>Species</dt><dd><p>The species of this sample</p>
</dd>
<dt>Cluster</dt><dd><p>Seurat cluster</p>
</dd>
<dt>Cells</dt><dd><p>Cluster cell number</p>
</dd>
<dt>Cell Type</dt><dd><p>Predicted cluster cell types</p>
</dd>
</dl>


<hr>
<h2 id='PanglaoDBMeta'>All Sample Metadata of PanglaoDB Datasets</h2><span id='topic+PanglaoDBMeta'></span>

<h3>Description</h3>

<p>A dataset contains all sample metadata and cell types of PanglaoDB datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PanglaoDBMeta
</code></pre>


<h3>Format</h3>

<p>A data frame with 1,368 rows and 7 variables:
</p>

<dl>
<dt>SRA</dt><dd><p>The SRA identifier</p>
</dd>
<dt>SRS</dt><dd><p>The SRS identifier</p>
</dd>
<dt>Tissue</dt><dd><p>The tissue of the sample</p>
</dd>
<dt>Protocol</dt><dd><p>The protocol used to generate this sample</p>
</dd>
<dt>Species</dt><dd><p>The species of this sample</p>
</dd>
<dt>Cells</dt><dd><p>Total cell number of this sample</p>
</dd>
<dt>CellType</dt><dd><p>Predicted cell types, separated by comma</p>
</dd>
</dl>


<hr>
<h2 id='ParseCBDatasets'>Download UCSC Cell Browser Datasets.</h2><span id='topic+ParseCBDatasets'></span>

<h3>Description</h3>

<p>Download UCSC Cell Browser Datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParseCBDatasets(
  sample.df = NULL,
  all.samples.df = NULL,
  collection = NULL,
  sub.collection = NULL,
  organ = NULL,
  disease = NULL,
  organism = NULL,
  project = NULL,
  fuzzy.match = TRUE,
  cell.num = NULL,
  timeout = 1000,
  merge = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParseCBDatasets_+3A_sample.df">sample.df</code></td>
<td>
<p>Dataframe contains used datasets. Default: NULL.</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_all.samples.df">all.samples.df</code></td>
<td>
<p>Dataframe contains all samples metadata, obtained with <code>ShowCBDatasets</code>. Default: NULL.
<code>sample.df</code> and <code>all.samples.df</code> cannot be both NULL.</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_collection">collection</code></td>
<td>
<p>The collection of the datasets, corresponds to <code>shortLabel</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_sub.collection">sub.collection</code></td>
<td>
<p>The sub-collection of the datasets, corresponds to <code>subLabel</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_organ">organ</code></td>
<td>
<p>The organ of the datasets, corresponds to <code>body_parts</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_disease">disease</code></td>
<td>
<p>The disease of the datasets, corresponds to <code>diseases</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_organism">organism</code></td>
<td>
<p>The specie of the datasets, corresponds to <code>organisms</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_project">project</code></td>
<td>
<p>The project of the datasets, corresponds to <code>projects</code> column
of <code>all.samples.df</code>, obtain available values with <code>StatDBAttribute</code>. Default: NULL (without filtering).</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_fuzzy.match">fuzzy.match</code></td>
<td>
<p>Logical value, whether to perform fuzzy match with provided attribute values. Default: TRUE.</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_cell.num">cell.num</code></td>
<td>
<p>Cell number filter. If NULL, no filter; if one value, lower filter; if two values, low and high filter. Deault: NULL.</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_timeout">timeout</code></td>
<td>
<p>Maximum request time when loading data online. Default: 1000.</p>
</td></tr>
<tr><td><code id="ParseCBDatasets_+3A_merge">merge</code></td>
<td>
<p>Logical value, whether to merge Seurat list. Default: FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat object (if <code>merge</code> is TRUE) or list of Seurat objects (if <code>merge</code> is FALSE).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# lazy mode, load datasets json files locally, need users to provide json folder
ucsc.cb.samples &lt;- ShowCBDatasets(lazy = TRUE, json.folder = NULL, update = FALSE)
# cell number is between 1000 and 2000
hbb.sample.df &lt;- ExtractCBDatasets(
  all.samples.df = ucsc.cb.samples, organ = c("brain", "blood"),
  organism = "Human (H. sapiens)", cell.num = c(1000, 2000)
)
hbb.sample.seu &lt;- ParseCBDatasets(sample.df = hbb.sample.df)
# test 10x and matrix load
complex.df &lt;- ucsc.cb.samples[c(1, 927, 379), ] # two 10x and one matrix
complex.seu.list &lt;- ParseCBDatasets(sample.df = test.df, merge = F)

## End(Not run)
</code></pre>

<hr>
<h2 id='ParseCELLxGENE'>Download CELLxGENE Datasets.</h2><span id='topic+ParseCELLxGENE'></span>

<h3>Description</h3>

<p>Download CELLxGENE Datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParseCELLxGENE(
  meta,
  file.ext = c("rds", "h5ad"),
  out.folder = NULL,
  timeout = 3600,
  quiet = FALSE,
  parallel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParseCELLxGENE_+3A_meta">meta</code></td>
<td>
<p>Metadata used to download, can be from <code>ExtractCELLxGENEMeta</code>,
should contain dataset_id, rds_id/h5ad_id (depend on <code>file.ext</code>) and name columns.</p>
</td></tr>
<tr><td><code id="ParseCELLxGENE_+3A_file.ext">file.ext</code></td>
<td>
<p>The valid file extension for download. When NULL, use &quot;rds&quot; and &quot;h5ad&quot;. Default: c(&quot;rds&quot;, &quot;h5ad&quot;).</p>
</td></tr>
<tr><td><code id="ParseCELLxGENE_+3A_out.folder">out.folder</code></td>
<td>
<p>The output folder. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ParseCELLxGENE_+3A_timeout">timeout</code></td>
<td>
<p>Maximum request time. Default: 3600.</p>
</td></tr>
<tr><td><code id="ParseCELLxGENE_+3A_quiet">quiet</code></td>
<td>
<p>Logical value, whether to show downloading progress. Default: FALSE (show).</p>
</td></tr>
<tr><td><code id="ParseCELLxGENE_+3A_parallel">parallel</code></td>
<td>
<p>Logical value, whether to download parallelly. Default: TRUE. When &quot;libcurl&quot; is available for <code>download.file</code>,
the parallel is done by default (<code>parallel</code> can be FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains failed datasets or NULL.
</p>


<h3>References</h3>

<p>https://gist.github.com/ivirshup/f1a1603db69de3888eacb4bdb6a9317a
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# all available datasets
all.cellxgene.datasets &lt;- ShowCELLxGENEDatasets()
# human 10x v2 and v3 datasets
human.10x.cellxgene.meta &lt;- ExtractCELLxGENEMeta(
  all.samples.df = all.cellxgene.datasets,
  assay = c("10x 3' v2", "10x 3' v3"),
  organism = "Homo sapiens"
)
# download, need to provide the output folder
ParseCELLxGENE(meta = human.10x.cellxgene.meta, out.folder = "/path/to/output")

## End(Not run)
</code></pre>

<hr>
<h2 id='ParseGEO'>Download Matrix from GEO and Load to Seurat.</h2><span id='topic+ParseGEO'></span>

<h3>Description</h3>

<p>Download Matrix from GEO and Load to Seurat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParseGEO(
  acce,
  platform = NULL,
  down.supp = FALSE,
  supp.idx = 1,
  timeout = 3600,
  data.type = c("sc", "bulk"),
  supp.type = c("count", "10x"),
  out.folder = NULL,
  gene2feature = TRUE,
  merge = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParseGEO_+3A_acce">acce</code></td>
<td>
<p>GEO accession number.</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_platform">platform</code></td>
<td>
<p>Platform information/field. Disable when <code>down.supp</code> is TRUE. Default: NULL (disable).</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_down.supp">down.supp</code></td>
<td>
<p>Logical value, whether to download supplementary files to create count matrix. If TRUE, always
download supplementary files. If FALSE, use <code>ExpressionSet</code> (If contains non-integer or empty,
download supplementary files automatically). Default: FALSE.</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_supp.idx">supp.idx</code></td>
<td>
<p>The index of supplementary files to download. This should be consistent with <code>platform</code>. Default: 1.</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_timeout">timeout</code></td>
<td>
<p>Timeout for <code><a href="utils.html#topic+download.file">download.file</a></code>. Default: 3600.</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_data.type">data.type</code></td>
<td>
<p>The data type of the dataset, choose from &quot;sc&quot; (single-cell) and &quot;bulk&quot; (bulk). Default: &quot;sc&quot;.</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_supp.type">supp.type</code></td>
<td>
<p>The type of downloaded supplementary files, choose from count (count matrix file or single count matrix file)
and 10x (cellranger output files, contains barcodes, genes/features and matrix). Default: count.</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_out.folder">out.folder</code></td>
<td>
<p>Output folder to save 10x files. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_gene2feature">gene2feature</code></td>
<td>
<p>Logical value, whether to rename <code>genes.tsv.gz</code> to <code>features.tsv.gz</code>. Default: TRUE.</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_merge">merge</code></td>
<td>
<p>Logical value, whether to merge Seurat list when there are multiple 10x files (<code>supp.type</code> is 10x). Default: FALSE.</p>
</td></tr>
<tr><td><code id="ParseGEO_+3A_...">...</code></td>
<td>
<p>Parameters for <code>getGEO</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>data.type</code> is &quot;sc&quot;, return Seurat object (if <code>merge</code> is TRUE) or Seurat object list (if <code>merge</code> is FALSE).
If <code>data.type</code> is &quot;bulk&quot;, return count matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# the supp files are count matrix
GSE94820.seu &lt;- ParseGEO(acce = "GSE94820", down.supp = TRUE, supp.idx = 1, supp.type = "count")
# the supp files are cellranger output files: barcodes, genes/features and matrix
# need users to provide the output folder
GSE200257.seu &lt;- ParseGEO(
  acce = "GSE200257", down.supp = TRUE, supp.idx = 1, supp.type = "10x",
  out.folder = "/path/to/output/folder"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='ParseHCA'>Download Human Cell Atlas Datasets.</h2><span id='topic+ParseHCA'></span>

<h3>Description</h3>

<p>Download Human Cell Atlas Datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParseHCA(
  meta,
  file.ext = c("rds", "rdata", "h5", "h5ad", "loom"),
  out.folder = NULL,
  timeout = 3600,
  quiet = FALSE,
  parallel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParseHCA_+3A_meta">meta</code></td>
<td>
<p>Metadata used to download, can be from <code>ExtractHCAMeta</code>,
should contain entryId and name catalog.</p>
</td></tr>
<tr><td><code id="ParseHCA_+3A_file.ext">file.ext</code></td>
<td>
<p>The valid file extension for download. When NULL, use &quot;rds&quot;, &quot;rdata&quot;, &quot;h5&quot;, &quot;h5ad&quot;, &quot;loom&quot;.
Default: c(&quot;rds&quot;, &quot;rdata&quot;, &quot;h5&quot;, &quot;h5ad&quot;, &quot;loom&quot;).</p>
</td></tr>
<tr><td><code id="ParseHCA_+3A_out.folder">out.folder</code></td>
<td>
<p>The output folder. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ParseHCA_+3A_timeout">timeout</code></td>
<td>
<p>Maximum request time. Default: 3600.</p>
</td></tr>
<tr><td><code id="ParseHCA_+3A_quiet">quiet</code></td>
<td>
<p>Logical value, whether to show downloading progress. Default: FALSE (show).</p>
</td></tr>
<tr><td><code id="ParseHCA_+3A_parallel">parallel</code></td>
<td>
<p>Logical value, whether to download parallelly. Default: TRUE. When &quot;libcurl&quot; is available for <code>download.file</code>,
the parallel is done by default (<code>parallel</code> can be FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains failed projects or NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# all available projects
all.hca.projects &lt;- ShowHCAProjects()
# all human and 10x 3' v2
all.human.10x.projects &lt;- ExtractHCAMeta(
  all.projects.df = all.hca.projects,
  organism = "Homo sapiens",
  protocol = c("10x 3' v2", "10x 3' v3")
)
# download, need users to provide the output folder
ParseHCA(meta = all.human.10x.projects, out.folder = "/path/to/output")

## End(Not run)
</code></pre>

<hr>
<h2 id='ParsePanglaoDB'>Parse PanglaoDB Data.</h2><span id='topic+ParsePanglaoDB'></span>

<h3>Description</h3>

<p>Parse PanglaoDB Data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParsePanglaoDB(
  meta,
  cell.type = "All",
  include.gene = NA,
  exclude.gene = NA,
  merge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParsePanglaoDB_+3A_meta">meta</code></td>
<td>
<p>Metadata contains &quot;SRA&quot;, &quot;SRS&quot;, &quot;Tissue&quot;, &quot;Protocol&quot;, &quot;Species&quot;, can be obtained with <code>ExtractPanglaoDBMeta</code>.</p>
</td></tr>
<tr><td><code id="ParsePanglaoDB_+3A_cell.type">cell.type</code></td>
<td>
<p>Extract samples with specified cell types. For samples without SRS (notused), this value can only be &quot;All&quot; or &quot;None&quot;, or
these samples will be filtered. Obtain available values with <code>StatDBAttribute</code>, one or multiple value. Default: &quot;All&quot;.</p>
</td></tr>
<tr><td><code id="ParsePanglaoDB_+3A_include.gene">include.gene</code></td>
<td>
<p>Include cells expressing the genes. Default: NA.</p>
</td></tr>
<tr><td><code id="ParsePanglaoDB_+3A_exclude.gene">exclude.gene</code></td>
<td>
<p>Exclude cells expressing the genes. Default: NA.</p>
</td></tr>
<tr><td><code id="ParsePanglaoDB_+3A_merge">merge</code></td>
<td>
<p>Logical value, whether to merge Seurat list. Default: FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seurat object (if <code>merge</code> is TRUE) or list of Seurat objects (if <code>merge</code> is FALSE).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
hsa.meta &lt;- ExtractPanglaoDBMeta(
  species = "Homo sapiens",
  protocol = c("Smart-seq2", "10x chromium"),
  show.cell.type = TRUE, cell.num = c(1000, 2000)
)
hsa.seu &lt;- ParsePanglaoDB(hsa.meta, merge = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ParseZenodo'>Download Data with Zenodo DOI.</h2><span id='topic+ParseZenodo'></span>

<h3>Description</h3>

<p>Download Data with Zenodo DOI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParseZenodo(
  doi = NULL,
  file.ext = c("rdata", "rds", "h5ad"),
  doi.df = NULL,
  out.folder = NULL,
  timeout = 1000,
  quiet = FALSE,
  parallel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParseZenodo_+3A_doi">doi</code></td>
<td>
<p>A vector of Zenodo DOIs to download. Default: NULL.</p>
</td></tr>
<tr><td><code id="ParseZenodo_+3A_file.ext">file.ext</code></td>
<td>
<p>The valid file extension for download. When NULL, use all files. Default: c(&quot;rdata&quot;, &quot;rds&quot;, &quot;h5ad&quot;).</p>
</td></tr>
<tr><td><code id="ParseZenodo_+3A_doi.df">doi.df</code></td>
<td>
<p>DOI dataframe for download. This is useful when something wrong happens in downloading
(e.g. MD5 verification failure, <code>DownloadZenodo</code> will return a dataframe contains failure terms.). Default: NULL.
It is required to provide either <code>doi</code> or <code>doi.df</code>.</p>
</td></tr>
<tr><td><code id="ParseZenodo_+3A_out.folder">out.folder</code></td>
<td>
<p>The output folder. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ParseZenodo_+3A_timeout">timeout</code></td>
<td>
<p>Maximum request time. Default: 1000.</p>
</td></tr>
<tr><td><code id="ParseZenodo_+3A_quiet">quiet</code></td>
<td>
<p>Logical value, whether to show downloading progress. Default: FALSE (show).</p>
</td></tr>
<tr><td><code id="ParseZenodo_+3A_parallel">parallel</code></td>
<td>
<p>Logical value, whether to download parallelly. Default: TRUE. When &quot;libcurl&quot; is available for <code>download.file</code>,
the parallel is done by default (<code>parallel</code> can be FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>When successful, NULL. When MD5 verification failure, a dataframe contains failure terms.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# need users to provide the output folder
multi.dois.parse &lt;- ParseZenodo(
  doi = c(
    "1111", "10.5281/zenodo.7243603",
    "10.5281/zenodo.7244441"
  ),
  file.ext = c("rdata", "rds"),
  out.folder = "/path/to/outfoder"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='SCEAnnData'>Data Format Conversion between SingleCellExperiment and AnnData.</h2><span id='topic+SCEAnnData'></span>

<h3>Description</h3>

<p>Data Format Conversion between SingleCellExperiment and AnnData.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCEAnnData(
  from = c("SingleCellExperiment", "AnnData"),
  to = c("AnnData", "SingleCellExperiment"),
  sce = NULL,
  anndata.file = NULL,
  slot.name = "counts",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCEAnnData_+3A_from">from</code></td>
<td>
<p>The source data format to convert, chosen from SingleCellExperiment and AnnData.
Default: SingleCellExperiment.</p>
</td></tr>
<tr><td><code id="SCEAnnData_+3A_to">to</code></td>
<td>
<p>The target data format to convert, chosen from AnnData and SingleCellExperiment.
Default: AnnData.</p>
</td></tr>
<tr><td><code id="SCEAnnData_+3A_sce">sce</code></td>
<td>
<p>The SingleCellExperiment object to convert. Default: NULL.</p>
</td></tr>
<tr><td><code id="SCEAnnData_+3A_anndata.file">anndata.file</code></td>
<td>
<p>File used to save or contains AnnData results. Default: NULL.</p>
</td></tr>
<tr><td><code id="SCEAnnData_+3A_slot.name">slot.name</code></td>
<td>
<p>Slot name used to save count matrix, used when converting from AnnData to SingleCellExperiment.
Default: counts.</p>
</td></tr>
<tr><td><code id="SCEAnnData_+3A_...">...</code></td>
<td>
<p>Parameters for <code>writeH5AD</code> and <code>readH5AD</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL or SingleCellExperiment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(scRNAseq)
seger &lt;- SegerstolpePancreasData()
SCEAnnData(from = "SingleCellExperiment", to = "AnnData", sce = seger, X_name = "counts")
# need users to provide the output file
sce &lt;- SCEAnnData(
  from = "AnnData", to = "SingleCellExperiment",
  anndata.file = "path/to/seger.h5ad"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='SCELoom'>Data Format Conversion between SingleCellExperiment and loom.</h2><span id='topic+SCELoom'></span>

<h3>Description</h3>

<p>Data Format Conversion between SingleCellExperiment and loom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCELoom(
  from = c("SingleCellExperiment", "loom"),
  to = c("loom", "SingleCellExperiment"),
  sce = NULL,
  loom.file = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCELoom_+3A_from">from</code></td>
<td>
<p>The source data format to convert, chosen from SingleCellExperiment and AnnData.
Default: SingleCellExperiment.</p>
</td></tr>
<tr><td><code id="SCELoom_+3A_to">to</code></td>
<td>
<p>The target data format to convert, chosen from AnnData and SingleCellExperiment.
Default: loom.</p>
</td></tr>
<tr><td><code id="SCELoom_+3A_sce">sce</code></td>
<td>
<p>The SingleCellExperiment object to convert. Default: NULL.</p>
</td></tr>
<tr><td><code id="SCELoom_+3A_loom.file">loom.file</code></td>
<td>
<p>File used to save or contains loom results. Default: NULL.</p>
</td></tr>
<tr><td><code id="SCELoom_+3A_...">...</code></td>
<td>
<p>Parameters for <code>sceasy::convertFormat</code> and <code>sceasy::convertFormat</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL or SingleCellExperiment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# convert from loom to SingleCellExperiment, need users to provide the loom file
sce.obj &lt;- SCELoom(
  from = "loom", to = "SingleCellExperiment",
  loom.file = "path/to/loom"
)
# convert from SingleCellExperiment to loom, need users to provide the loom file
SCELoom(
  from = "SingleCellExperiment", to = "loom", sce = sce.obj,
  loom.file = "path/to/loom"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='ShowCBDatasets'>Show All Available Datasets in UCSC Cell Browser.</h2><span id='topic+ShowCBDatasets'></span>

<h3>Description</h3>

<p>Show All Available Datasets in UCSC Cell Browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ShowCBDatasets(lazy = TRUE, json.folder = NULL, update = FALSE, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ShowCBDatasets_+3A_lazy">lazy</code></td>
<td>
<p>Logical value, whether to always load datasets online or locally. Default: TRUE (load locally).</p>
</td></tr>
<tr><td><code id="ShowCBDatasets_+3A_json.folder">json.folder</code></td>
<td>
<p>Folder used to save or load datasets json files. Default: NULL (current working directory).</p>
</td></tr>
<tr><td><code id="ShowCBDatasets_+3A_update">update</code></td>
<td>
<p>Logical value, whther to update local datasets json. Default: FALSE. For the first time, you should set <code>lazy</code> TRUE and
<code>update</code> TRUE to save json files.</p>
</td></tr>
<tr><td><code id="ShowCBDatasets_+3A_quiet">quiet</code></td>
<td>
<p>Logical value, whether to show downloading progress. Default: FALSE (show).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains all available datasets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# first time run (lazy mode), need users to provide json folder
ucsc.cb.samples &lt;- ShowCBDatasets(lazy = TRUE, update = TRUE)
# second time run (lazy mode), need users to provide json folder
ucsc.cb.samples &lt;- ShowCBDatasets(lazy = TRUE, update = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ShowCELLxGENEDatasets'>Show All Available Datasets in CELLxGENE.</h2><span id='topic+ShowCELLxGENEDatasets'></span>

<h3>Description</h3>

<p>Show All Available Datasets in CELLxGENE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ShowCELLxGENEDatasets()
</code></pre>


<h3>Value</h3>

<p>Dataframe contains all available datasets.
</p>


<h3>References</h3>

<p>https://gist.github.com/ivirshup/f1a1603db69de3888eacb4bdb6a9317a
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# all available datasets
all.cellxgene.datasets &lt;- ShowCELLxGENEDatasets()

</code></pre>

<hr>
<h2 id='ShowHCAProjects'>Show All Available Projects in Human Cell Atlas.</h2><span id='topic+ShowHCAProjects'></span>

<h3>Description</h3>

<p>Show All Available Projects in Human Cell Atlas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ShowHCAProjects(catalog = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ShowHCAProjects_+3A_catalog">catalog</code></td>
<td>
<p>The catalog of the projects. Different catalogs may share some projects. Choose from &quot;dcp29&quot;,
&quot;dcp30&quot;, &quot;dcp1&quot;, &quot;lm2&quot;, &quot;lm3&quot;, one or multiple values. Default: NULL (all catalogs, remove duplicated projects).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe contains all available projects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# all available projects
all.hca.projects &lt;- ShowHCAProjects()

</code></pre>

<hr>
<h2 id='SplitSRA'>Split SRA to fastq Files and Format to 10x Standard Style.</h2><span id='topic+SplitSRA'></span>

<h3>Description</h3>

<p>Split SRA to fastq Files and Format to 10x Standard Style.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SplitSRA(
  sra.folder = NULL,
  sra.path = NULL,
  fastq.type = c("10x", "other"),
  split.cmd.path = NULL,
  sratools.path = NULL,
  split.cmd.paras = NULL,
  split.cmd.threads = NULL,
  format.10x = TRUE,
  remove.raw = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SplitSRA_+3A_sra.folder">sra.folder</code></td>
<td>
<p>Folder contains all sras, obtained from <code>DownloadSRA</code>. Default: NULL.</p>
</td></tr>
<tr><td><code id="SplitSRA_+3A_sra.path">sra.path</code></td>
<td>
<p>Paths of sras. <code>sra.folder</code> and <code>sra.path</code> cannot be both NULL. Default: NULL.</p>
</td></tr>
<tr><td><code id="SplitSRA_+3A_fastq.type">fastq.type</code></td>
<td>
<p>The source of fastq files, choose from 10x (use <code>--split-files</code> to split sra) or
other (use <code>--split-3</code> to split sra). Default: 10x.</p>
</td></tr>
<tr><td><code id="SplitSRA_+3A_split.cmd.path">split.cmd.path</code></td>
<td>
<p>The full command path used to split, can be path to parallel-fastq-dump,
fasterq-dump and fastq-dump. Default: NULL (conduct automatic detection).</p>
</td></tr>
<tr><td><code id="SplitSRA_+3A_sratools.path">sratools.path</code></td>
<td>
<p>Path to sratoolkit bin. When <code>split.cmd.path</code> is path to parallel-fastq-dump,
it requires sratoolkit. Default: NULL (conduct automatic detection).</p>
</td></tr>
<tr><td><code id="SplitSRA_+3A_split.cmd.paras">split.cmd.paras</code></td>
<td>
<p>Parameters for <code>split.cmd.path</code>. Default: NULL.</p>
</td></tr>
<tr><td><code id="SplitSRA_+3A_split.cmd.threads">split.cmd.threads</code></td>
<td>
<p>Threads, used when <code>split.cmd.path</code> is path to parallel-fastq-dump or fasterq-dump.
Default: NULL (1).</p>
</td></tr>
<tr><td><code id="SplitSRA_+3A_format.10x">format.10x</code></td>
<td>
<p>Logical value, whether to format split fastqs to 10x standard format. Default: TRUE.</p>
</td></tr>
<tr><td><code id="SplitSRA_+3A_remove.raw">remove.raw</code></td>
<td>
<p>Logical value, whether to remove old split fastqs (unformatted), used when <code>format.10x</code> is TRUE. Default: FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL or paths of failed sras.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# need users to provide the prefetch.path, sra.folder, split.cmd.path, sratools.path and out.folder
GSE186003.runs &lt;- ExtractRun(acce = "GSE186003", platform = "GPL24247")
GSE186003.down &lt;- DownloadSRA(
  gsm.df = GSE186003.runs, prefetch.path = "/path/to/prefetch",
  out.folder = "/path/to/output"
)
GSE186003.split &lt;- SplitSRA(
  sra.folder = "/path/to/output",
  split.cmd.path = "/path/to/parallel-fastq-dump",
  sratools.path = "/path/to/sra/bin", fastq.type = "10x",
  split.cmd.threads = 4
)

## End(Not run)
</code></pre>

<hr>
<h2 id='StatDBAttribute'>Stat Database Attributes.</h2><span id='topic+StatDBAttribute'></span>

<h3>Description</h3>

<p>Stat Database Attributes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StatDBAttribute(
  df,
  filter,
  database = c("PanglaoDB", "UCSC", "CELLxGENE", "HCA")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="StatDBAttribute_+3A_df">df</code></td>
<td>
<p>All metadata, can be <code>PanglaoDBMeta</code> and obtained with <code>ShowCBDatasets</code>, <code>ShowCELLxGENEDatasets</code>,
and <code>ShowHCAProjects</code>.</p>
</td></tr>
<tr><td><code id="StatDBAttribute_+3A_filter">filter</code></td>
<td>
<p>Vector of attributes.</p>
</td></tr>
<tr><td><code id="StatDBAttribute_+3A_database">database</code></td>
<td>
<p>Database name, choose from &quot;PanglaoDB&quot;, &quot;UCSC&quot;, &quot;CELLxGENE&quot;, &quot;HCA&quot;. Default: &quot;PanglaoDB&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of attributes information, including attribute, value and number.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# PanglaoDB
StatDBAttribute(df = PanglaoDBMeta, filter = c("species", "protocol"), database = "PanglaoDB")
# UCSC Cell Browser, need users to provide the json folder
ucsc.cb.samples &lt;- ShowCBDatasets(lazy = TRUE, json.folder = NULL, update = FALSE)
StatDBAttribute(df = ucsc.cb.samples, filter = c("organism", "organ"), database = "UCSC")
# CELLxGENE
all.cellxgene.datasets &lt;- ShowCELLxGENEDatasets()
StatDBAttribute(
  df = all.cellxgene.datasets, filter = c("organism", "sex"),
  database = "CELLxGENE"
)
# HCA
all.hca.projects &lt;- ShowHCAProjects()
StatDBAttribute(df = all.hca.projects, filter = c("organism", "sex"), database = "HCA")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
