<!DOCTYPE html><html><head><title>Help for package scGate</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scGate}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#combine_scGate_multiclass'><p>Combine scGate annotations</p></a></li>
<li><a href='#gating_model'><p>Model creation and editing</p></a></li>
<li><a href='#genes.blacklist.default'><p>Blocklist of genes for dimensionality reduction</p></a></li>
<li><a href='#get_scGateDB'><p>Load scGate model database</p></a></li>
<li><a href='#get_testing_data'><p>Download sample data</p></a></li>
<li><a href='#load_scGate_model'><p>Load a single scGate model</p></a></li>
<li><a href='#performance.metrics'><p>Performance metrics</p></a></li>
<li><a href='#plot_levels'><p>Plot scGate filtering results by level</p></a></li>
<li><a href='#plot_tree'><p>Plot model tree</p></a></li>
<li><a href='#plot_UCell_scores'><p>Plot UCell scores by level</p></a></li>
<li><a href='#query.seurat'><p>Toy dataset to test the package</p></a></li>
<li><a href='#scGate'><p>Filter single-cell data by cell type</p></a></li>
<li><a href='#test_my_model'><p>Test your model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Marker-Based Cell Type Purification for Single-Cell Sequencing
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A common bioinformatics task in single-cell data analysis is to purify a cell type or cell population of interest from heterogeneous datasets. 'scGate' automatizes marker-based purification of specific cell populations, without requiring training data or reference gene expression profiles. Briefly, 'scGate' takes as input: i) a gene expression matrix stored in a 'Seurat' object and ii) a “gating model” (GM), consisting of a set of marker genes that define the cell population of interest. The GM can be as simple as a single marker gene, or a combination of positive and negative markers. More complex GMs can be constructed in a hierarchical fashion, akin to gating strategies employed in flow cytometry. 'scGate' evaluates the strength of signature marker expression in each cell using the rank-based method 'UCell', and then performs k-nearest neighbor (kNN) smoothing by calculating the mean 'UCell' score across neighboring cells. kNN-smoothing aims at compensating for the large degree of sparsity in scRNA-seq data. Finally, a universal threshold over kNN-smoothed signature scores is applied in binary decision trees generated from the user-provided gating model, to annotate cells as either “pure” or “impure”, with respect to the cell population of interest. See the related publication Andreatta et al. (2022) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtac141">doi:10.1093/bioinformatics/btac141</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Seurat (&ge; 4.0.0), UCell (&ge; 2.6.0), dplyr, stats, utils,
methods, patchwork, ggridges, reshape2, ggplot2, BiocParallel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggparty, partykit, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/carmonalab/scGate">https://github.com/carmonalab/scGate</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/carmonalab/scGate/issues">https://github.com/carmonalab/scGate/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-06 11:55:11 UTC; mass</td>
</tr>
<tr>
<td>Author:</td>
<td>Massimo Andreatta <a href="https://orcid.org/0000-0002-8036-2647"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ariel Berenstein <a href="https://orcid.org/0000-0001-8540-5389"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Changsheng Li [aut],
  Santiago Carmona <a href="https://orcid.org/0000-0002-2495-0671"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Massimo Andreatta &lt;massimo.andreatta@unil.ch&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-06 12:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='combine_scGate_multiclass'>Combine scGate annotations</h2><span id='topic+combine_scGate_multiclass'></span>

<h3>Description</h3>

<p>If a single-cell dataset has precomputed results for multiple scGate models, combined them in multi-class annotation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_scGate_multiclass(
  obj,
  prefix = "is.pure_",
  scGate_classes = NULL,
  min_cells = 20,
  multi.asNA = FALSE,
  out_column = "scGate_multi"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_scGate_multiclass_+3A_obj">obj</code></td>
<td>
<p>Seurat object with scGate results for multiple models stored as metadata</p>
</td></tr>
<tr><td><code id="combine_scGate_multiclass_+3A_prefix">prefix</code></td>
<td>
<p>Prefix in metadata column names for scGate result models</p>
</td></tr>
<tr><td><code id="combine_scGate_multiclass_+3A_scgate_classes">scGate_classes</code></td>
<td>
<p>Vector of scGate model names. If NULL, use all columns that start with &quot;prefix&quot; above.</p>
</td></tr>
<tr><td><code id="combine_scGate_multiclass_+3A_min_cells">min_cells</code></td>
<td>
<p>Minimum number of cells for a cell label to be considered</p>
</td></tr>
<tr><td><code id="combine_scGate_multiclass_+3A_multi.asna">multi.asNA</code></td>
<td>
<p>How to label cells that are &quot;Pure&quot; for multiple annotations: &quot;Multi&quot; (FALSE) or NA (TRUE)</p>
</td></tr>
<tr><td><code id="combine_scGate_multiclass_+3A_out_column">out_column</code></td>
<td>
<p>The name of the metadata column where to store the multi-class cell labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object with multi-class annotations based on the combination of multiple models. A new
column (by default &quot;scGate_multi&quot;) is added to the metadata of the Seurat object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Define gating models
model.B &lt;- gating_model(name = "Bcell", signature = c("MS4A1")) 
model.T &lt;- gating_model(name = "Tcell", signature = c("CD2","CD3D","CD3E"))
# Apply scGate with these models
data(query.seurat)
query.seurat &lt;- scGate(query.seurat, model=model.T,
    reduction="pca", output.col.name = "is.pure_Tcell")
query.seurat &lt;- scGate(query.seurat, model=model.B,
    reduction="pca", output.col.name = "is.pure_Bcell")
query.seurat &lt;- combine_scGate_multiclass(query.seurat, scGate_class=c("Tcell","Bcell"))      
table(query.seurat$scGate_multi)

</code></pre>

<hr>
<h2 id='gating_model'>Model creation and editing</h2><span id='topic+gating_model'></span>

<h3>Description</h3>

<p>Generate an scGate model from scratch or edit an existing one
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gating_model(
  model = NULL,
  level = 1,
  name,
  signature,
  positive = TRUE,
  negative = FALSE,
  remove = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gating_model_+3A_model">model</code></td>
<td>
<p>scGate model to be modified. When is NULL (default) a new model will be initialized.</p>
</td></tr>
<tr><td><code id="gating_model_+3A_level">level</code></td>
<td>
<p>integer. It refers to the hierarchical level of the model tree in which the signature will be added (level=1 by default)</p>
</td></tr>
<tr><td><code id="gating_model_+3A_name">name</code></td>
<td>
<p>Arbitrary signature name (i.e. Immune, Tcell, NK etc).</p>
</td></tr>
<tr><td><code id="gating_model_+3A_signature">signature</code></td>
<td>
<p>character vector indicating gene symbols to be included in the signature (e.g. CD3D). If a minus sign is placed to the end of a gene name (e.g. &quot;CD3D-&quot;), this gene will be used as negative in UCell computing. See UCell documentation for details</p>
</td></tr>
<tr><td><code id="gating_model_+3A_positive">positive</code></td>
<td>
<p>Logical indicating if the signature must be used as a positive signature in those model level. Default is TRUE.</p>
</td></tr>
<tr><td><code id="gating_model_+3A_negative">negative</code></td>
<td>
<p>Same as 'positive' but negated (negative=TRUE equals to positive=FALSE)</p>
</td></tr>
<tr><td><code id="gating_model_+3A_remove">remove</code></td>
<td>
<p>Whether to remove the given signature from the model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scGate model that can be used by <code><a href="#topic+scGate">scGate</a></code> to filter target cell types.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create a simple gating model
my_model &lt;- gating_model(level = 1, name = "immune", signature = c("PTPRC"))
my_model &lt;- gating_model(model = my_model, level = 1, positive = FALSE,
    name = "Epithelial", signature = c("CDH1","FLT1") )
# Remove an existing signature
dropped_model &lt;- gating_model(model = my_model, remove =TRUE, level = 1, name = "Epithelial")
</code></pre>

<hr>
<h2 id='genes.blacklist.default'>Blocklist of genes for dimensionality reduction</h2><span id='topic+genes.blacklist.default'></span>

<h3>Description</h3>

<p>A list of signatures, for mouse and human. These include cell cycling,
heat-shock genes, mitochondrial genes, and other genes classes, that may
confound the identification of cell types. These are used internally by scGate
and excluded from the calculation of dimensional reductions (PCA).
</p>


<h3>Format</h3>

<p>A list of signatures
</p>

<hr>
<h2 id='get_scGateDB'>Load scGate model database</h2><span id='topic+get_scGateDB'></span>

<h3>Description</h3>

<p>Download, update or load local version of the scGate model database. These are stored in a GitHub repository, from where you can download specific 
versions of the database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_scGateDB(
  destination = tempdir(),
  force_update = FALSE,
  version = "latest",
  branch = c("master", "dev"),
  verbose = FALSE,
  repo_url = "https://github.com/carmonalab/scGate_models"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_scGateDB_+3A_destination">destination</code></td>
<td>
<p>Destination path for storing the DB. The default is tempdir(); if you wish to edit locally the models and
link them to the current project, set this parameter to a new directory name, e.g. scGateDB</p>
</td></tr>
<tr><td><code id="get_scGateDB_+3A_force_update">force_update</code></td>
<td>
<p>Whether to update an existing database.</p>
</td></tr>
<tr><td><code id="get_scGateDB_+3A_version">version</code></td>
<td>
<p>Specify the version of the scGate_models database (e.g. 'v0.1'). By default downloads the latest available version.</p>
</td></tr>
<tr><td><code id="get_scGateDB_+3A_branch">branch</code></td>
<td>
<p>branch of the scGate model repository, either 'master' (default) or 'dev' for the latest models</p>
</td></tr>
<tr><td><code id="get_scGateDB_+3A_verbose">verbose</code></td>
<td>
<p>display progress messages</p>
</td></tr>
<tr><td><code id="get_scGateDB_+3A_repo_url">repo_url</code></td>
<td>
<p>URL path to scGate model repository database</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Models for scGate are dataframes where each line is a signature for a given filtering level. A database of models can be downloaded using the function
<code>get_scGateDB</code>. You may directly use the models from the database, or edit one of these models to generate your own custom gating model.
</p>


<h3>Value</h3>

<p>A list of models, organized according to the folder structure of the database. See the examples below.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+scGate">scGate</a></code> <code><a href="#topic+load_scGate_model">load_scGate_model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scGate.model.db &lt;- get_scGateDB()
# To see a specific model, browse the list of models:
scGate.model.db$human$generic$Myeloid
</code></pre>

<hr>
<h2 id='get_testing_data'>Download sample data</h2><span id='topic+get_testing_data'></span>

<h3>Description</h3>

<p>Helper function to obtain some sample data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_testing_data(version = "hsa.latest", destination = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_testing_data_+3A_version">version</code></td>
<td>
<p>Which sample dataset</p>
</td></tr>
<tr><td><code id="get_testing_data_+3A_destination">destination</code></td>
<td>
<p>Save to this directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of datasets that can be used to test scGate
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
testing.datasets &lt;- get_testing_data(version = 'hsa.latest')

</code></pre>

<hr>
<h2 id='load_scGate_model'>Load a single scGate model</h2><span id='topic+load_scGate_model'></span>

<h3>Description</h3>

<p>Loads a custom scGate model into R. For the format of these models, have a
look or edit one of the default models obtained with <code><a href="#topic+get_scGateDB">get_scGateDB</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_scGate_model(model_file, master.table = "master_table.tsv")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_scGate_model_+3A_model_file">model_file</code></td>
<td>
<p>scGate model file, in .tsv format.</p>
</td></tr>
<tr><td><code id="load_scGate_model_+3A_master.table">master.table</code></td>
<td>
<p>File name of the master table (in repo_path folder) that contains cell type signatures.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scGate model in dataframe format, which can given as input to the <code><a href="#topic+scGate">scGate</a></code> function.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+scGate">scGate</a></code> <code><a href="#topic+get_scGateDB">get_scGateDB</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dir &lt;- tempdir() # this may also be set to your working directory
models &lt;- get_scGateDB(destination=dir)
# Original or edited model
model.path &lt;- paste0(dir,"/scGate_models-master/human/generic/Bcell_scGate_Model.tsv")
master.path &lt;- paste0(dir,"/scGate_models-master/human/generic/master_table.tsv")
my.model &lt;- load_scGate_model(model.path, master.path)
my.model
</code></pre>

<hr>
<h2 id='performance.metrics'>Performance metrics</h2><span id='topic+performance.metrics'></span>

<h3>Description</h3>

<p>Evaluate model performance for binary tasks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>performance.metrics(actual, pred, return_contingency = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="performance.metrics_+3A_actual">actual</code></td>
<td>
<p>Logical or numeric binary vector giving the actual cell labels.</p>
</td></tr>
<tr><td><code id="performance.metrics_+3A_pred">pred</code></td>
<td>
<p>Logical or numeric binary vector giving the predicted cell labels.</p>
</td></tr>
<tr><td><code id="performance.metrics_+3A_return_contingency">return_contingency</code></td>
<td>
<p>Logical indicating if contingency table must be returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prediction performance metrics (Precision, Recall, MCC) between actual and
predicted cell type labels.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>results &lt;- performance.metrics(actual= sample(c(1,0),20,replace=TRUE),
    pred =  sample(c(1,0),20,replace=TRUE,prob = c(0.65,0.35) ) )
</code></pre>

<hr>
<h2 id='plot_levels'>Plot scGate filtering results by level</h2><span id='topic+plot_levels'></span>

<h3>Description</h3>

<p>Fast plotting of gating results over each model level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_levels(obj, pure.col = "green", impure.col = "gray")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_levels_+3A_obj">obj</code></td>
<td>
<p>Gated Seurat object output of scGate filtering function</p>
</td></tr>
<tr><td><code id="plot_levels_+3A_pure.col">pure.col</code></td>
<td>
<p>Color code for pure category</p>
</td></tr>
<tr><td><code id="plot_levels_+3A_impure.col">impure.col</code></td>
<td>
<p>Color code for impure category</p>
</td></tr>
</table>


<h3>Value</h3>

<p>UMAP plots with 'Pure'/'Impure' labels for each level of the scGate model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
scGate.model.db &lt;- get_scGateDB()
model &lt;- scGate.model.db$human$generic$Myeloid
# Apply scGate with this model
data(query.seurat)
query.seurat &lt;- scGate(query.seurat, model=model,
    reduction="pca", save.levels=TRUE)
library(patchwork)     
pll &lt;- plot_levels(query.seurat)
wrap_plots(pll)

</code></pre>

<hr>
<h2 id='plot_tree'>Plot model tree</h2><span id='topic+plot_tree'></span>

<h3>Description</h3>

<p>View scGate model as a decision tree (require ggparty package)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_tree(model, box.size = 8, edge.text.size = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_tree_+3A_model">model</code></td>
<td>
<p>A scGate model to be visualized</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_box.size">box.size</code></td>
<td>
<p>Box size</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_edge.text.size">edge.text.size</code></td>
<td>
<p>Edge text size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of the model as a decision tree. At each level, green boxes
indicate the 'positive' (accepted) cell types, red boxed indicate the
'negative' cell types (filtered out). The final Pure population is the
bottom right subset in the tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggparty)
models &lt;- get_scGateDB()
plot_tree(models$human$generic$Tcell)
</code></pre>

<hr>
<h2 id='plot_UCell_scores'>Plot UCell scores by level</h2><span id='topic+plot_UCell_scores'></span>

<h3>Description</h3>

<p>Show distribution of UCell scores for each level of a given scGate model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_UCell_scores(
  obj,
  model,
  overlay = 5,
  pos.thr = 0.2,
  neg.thr = 0.2,
  ncol = NULL,
  combine = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_UCell_scores_+3A_obj">obj</code></td>
<td>
<p>Gated Seurat object (output of scGate)</p>
</td></tr>
<tr><td><code id="plot_UCell_scores_+3A_model">model</code></td>
<td>
<p>scGate model used to identify a target population in obj</p>
</td></tr>
<tr><td><code id="plot_UCell_scores_+3A_overlay">overlay</code></td>
<td>
<p>Degree of overlay for ggridges</p>
</td></tr>
<tr><td><code id="plot_UCell_scores_+3A_pos.thr">pos.thr</code></td>
<td>
<p>Threshold for positive signatures used in scGate model (set to NULL to disable)</p>
</td></tr>
<tr><td><code id="plot_UCell_scores_+3A_neg.thr">neg.thr</code></td>
<td>
<p>Threshold for negative signatures used in scGate model (set to NULL to disable)</p>
</td></tr>
<tr><td><code id="plot_UCell_scores_+3A_ncol">ncol</code></td>
<td>
<p>Number of columns in output object (passed to wrap_plots)</p>
</td></tr>
<tr><td><code id="plot_UCell_scores_+3A_combine">combine</code></td>
<td>
<p>Whether to combine plots into a single object, or to return a list of plots</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a density plot of UCell scores for the signatures in the scGate model,
for each level of the model
</p>
<p>Either a plot combined by patchwork (combine=T) or a list of plots (combine=F)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
scGate.model.db &lt;- get_scGateDB()
model &lt;- scGate.model.db$human$generic$Tcell
# Apply scGate with this model
data(query.seurat)
query.seurat &lt;- scGate(query.seurat, model=model,
    reduction="pca", save.levels=TRUE)
# View UCell score distribution
plot_UCell_scores(query.seurat, model)

</code></pre>

<hr>
<h2 id='query.seurat'>Toy dataset to test the package</h2><span id='topic+query.seurat'></span>

<h3>Description</h3>

<p>A downsampled version (300 cells) of the single-cell dataset by Zilionis et
al. (2019) &lt;doi:10.1016/j.immuni.2019.03.009&gt;, with precalculated PCA and
UMAP reductions.
</p>


<h3>Format</h3>

<p>A Seurat object
</p>

<hr>
<h2 id='scGate'>Filter single-cell data by cell type</h2><span id='topic+scGate'></span>

<h3>Description</h3>

<p>Apply scGate to filter specific cell types in a query dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scGate(
  data,
  model,
  pos.thr = 0.2,
  neg.thr = 0.2,
  assay = NULL,
  slot = "data",
  ncores = 1,
  BPPARAM = NULL,
  seed = 123,
  keep.ranks = FALSE,
  reduction = c("calculate", "pca", "umap", "harmony"),
  min.cells = 30,
  nfeatures = 2000,
  pca.dim = 30,
  param_decay = 0.25,
  maxRank = 1500,
  output.col.name = "is.pure",
  k.param = 30,
  smooth.decay = 0.1,
  smooth.up.only = FALSE,
  genes.blacklist = "default",
  multi.asNA = FALSE,
  additional.signatures = NULL,
  save.levels = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scGate_+3A_data">data</code></td>
<td>
<p>Seurat object containing a query data set - filtering will be applied to this object</p>
</td></tr>
<tr><td><code id="scGate_+3A_model">model</code></td>
<td>
<p>A single scGate model, or a list of scGate models. See Details for this format</p>
</td></tr>
<tr><td><code id="scGate_+3A_pos.thr">pos.thr</code></td>
<td>
<p>Minimum UCell score value for positive signatures</p>
</td></tr>
<tr><td><code id="scGate_+3A_neg.thr">neg.thr</code></td>
<td>
<p>Maximum UCell score value for negative signatures</p>
</td></tr>
<tr><td><code id="scGate_+3A_assay">assay</code></td>
<td>
<p>Seurat assay to use</p>
</td></tr>
<tr><td><code id="scGate_+3A_slot">slot</code></td>
<td>
<p>Data slot in Seurat object to calculate UCell scores</p>
</td></tr>
<tr><td><code id="scGate_+3A_ncores">ncores</code></td>
<td>
<p>Number of processors for parallel processing</p>
</td></tr>
<tr><td><code id="scGate_+3A_bpparam">BPPARAM</code></td>
<td>
<p>A [BiocParallel::bpparam()] object that tells scGate
how to parallelize. If provided, it overrides the 'ncores' parameter.</p>
</td></tr>
<tr><td><code id="scGate_+3A_seed">seed</code></td>
<td>
<p>Integer seed for random number generator</p>
</td></tr>
<tr><td><code id="scGate_+3A_keep.ranks">keep.ranks</code></td>
<td>
<p>Store UCell rankings in Seurat object. This will speed up calculations if the same object is applied again with new signatures.</p>
</td></tr>
<tr><td><code id="scGate_+3A_reduction">reduction</code></td>
<td>
<p>Dimensionality reduction to use for knn smoothing. By default, calculates a new reduction
based on the given <code>assay</code>; otherwise you may specify a precalculated dimensionality reduction (e.g.
in the case of an integrated dataset after batch-effect correction)</p>
</td></tr>
<tr><td><code id="scGate_+3A_min.cells">min.cells</code></td>
<td>
<p>Minimum number of cells to cluster or define cell types</p>
</td></tr>
<tr><td><code id="scGate_+3A_nfeatures">nfeatures</code></td>
<td>
<p>Number of variable genes for dimensionality reduction</p>
</td></tr>
<tr><td><code id="scGate_+3A_pca.dim">pca.dim</code></td>
<td>
<p>Number of principal components for dimensionality reduction</p>
</td></tr>
<tr><td><code id="scGate_+3A_param_decay">param_decay</code></td>
<td>
<p>Controls decrease in parameter complexity at each iteration, between 0 and 1.
<code>param_decay == 0</code> gives no decay, increasingly higher <code>param_decay</code> gives increasingly stronger decay</p>
</td></tr>
<tr><td><code id="scGate_+3A_maxrank">maxRank</code></td>
<td>
<p>Maximum number of genes that UCell will rank per cell</p>
</td></tr>
<tr><td><code id="scGate_+3A_output.col.name">output.col.name</code></td>
<td>
<p>Column name with 'pure/impure' annotation</p>
</td></tr>
<tr><td><code id="scGate_+3A_k.param">k.param</code></td>
<td>
<p>Number of nearest neighbors for knn smoothing</p>
</td></tr>
<tr><td><code id="scGate_+3A_smooth.decay">smooth.decay</code></td>
<td>
<p>Decay parameter for knn weights: (1-decay)^n</p>
</td></tr>
<tr><td><code id="scGate_+3A_smooth.up.only">smooth.up.only</code></td>
<td>
<p>If TRUE, only let smoothing increase signature scores</p>
</td></tr>
<tr><td><code id="scGate_+3A_genes.blacklist">genes.blacklist</code></td>
<td>
<p>Genes blacklisted from variable features. The default loads the list of genes in <code>scGate::genes.blacklist.default</code>;
you may deactivate blacklisting by setting <code>genes.blacklist=NULL</code></p>
</td></tr>
<tr><td><code id="scGate_+3A_multi.asna">multi.asNA</code></td>
<td>
<p>How to label cells that are &quot;Pure&quot; for multiple annotations: &quot;Multi&quot; (FALSE) or NA (TRUE)</p>
</td></tr>
<tr><td><code id="scGate_+3A_additional.signatures">additional.signatures</code></td>
<td>
<p>A list of additional signatures, not included in the model, to be evaluated (e.g. a cycling signature). The scores for this
list of signatures will be returned but not used for filtering.</p>
</td></tr>
<tr><td><code id="scGate_+3A_save.levels">save.levels</code></td>
<td>
<p>Whether to save in metadata the filtering output for each gating model level</p>
</td></tr>
<tr><td><code id="scGate_+3A_verbose">verbose</code></td>
<td>
<p>Verbose output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Models for scGate are data frames where each line is a signature for a given filtering level.
A database of models can be downloaded using the function <code>get_scGateDB</code>.
You may directly use the models from the database, or edit one of these models to generate your own custom gating model.
</p>
<p>Multiple models can also be evaluated at once, by running scGate with a list of models. Gating for each individual model is
returned as metadata, with a consensus annotation stored in <code>scGate_multi</code> metadata field. This allows using scGate as a
multi-class classifier, where only cells that are &quot;Pure&quot; for a single model are assigned a label, cells that are &quot;Pure&quot; for
more than one gating model are labeled as &quot;Multi&quot;, all others cells are annotated as NA.
</p>


<h3>Value</h3>

<p>A new metadata column <code>is.pure</code> is added to the query Seurat object, indicating which cells passed the scGate filter.
The <code>active.ident</code> is also set to this variable.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_scGate_model">load_scGate_model</a></code> <code><a href="#topic+get_scGateDB">get_scGateDB</a></code> <code><a href="#topic+plot_tree">plot_tree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
### Test using a small toy set
data(query.seurat)
# Define basic gating model for B cells
my_scGate_model &lt;- gating_model(name = "Bcell", signature = c("MS4A1")) 
query.seurat &lt;- scGate(query.seurat, model = my_scGate_model, reduction="pca")
table(query.seurat$is.pure)
### Test with larger datasets
library(Seurat)
testing.datasets &lt;- get_testing_data(version = 'hsa.latest')
seurat_object &lt;- testing.datasets[["JerbyArnon"]]
# Download pre-defined models
models &lt;- get_scGateDB()
seurat_object &lt;- scGate(seurat_object, model=models$human$generic$PanBcell)
DimPlot(seurat_object)
seurat_object_filtered &lt;- subset(seurat_object, subset=is.pure=="Pure")

### Run multiple models at once
models &lt;- get_scGateDB()
model.list &lt;- list("Bcell" = models$human$generic$Bcell,
                   "Tcell" = models$human$generic$Tcell)
seurat_object &lt;- scGate(seurat_object, model=model.list)
DimPlot(seurat_object, group.by = "scGate_multi")

</code></pre>

<hr>
<h2 id='test_my_model'>Test your model</h2><span id='topic+test_my_model'></span>

<h3>Description</h3>

<p>Wrapper for fast model testing on 3 sampled datasets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_my_model(
  model,
  testing.version = "hsa.latest",
  custom.dataset = NULL,
  target = NULL,
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test_my_model_+3A_model">model</code></td>
<td>
<p>scGate model in data.frame format</p>
</td></tr>
<tr><td><code id="test_my_model_+3A_testing.version">testing.version</code></td>
<td>
<p>Character indicating the version of testing tatasets
to be used. By default &quot;hsa-latest&quot; will be used. It will be ignored if
a custom dataset is provided (in Seurat format).</p>
</td></tr>
<tr><td><code id="test_my_model_+3A_custom.dataset">custom.dataset</code></td>
<td>
<p>Seurat object to be used as a testing dataset. For
testing purposes, metadata seurat object must contain a column named
'cell_type' to be used as a gold standard. Also a set of positive
targets must be provided in the target variable.</p>
</td></tr>
<tr><td><code id="test_my_model_+3A_target">target</code></td>
<td>
<p>Positive target cell types. If default testing version is used
this variable must be a character indicating one of the available target
models ('immune','Lymphoid','Myeloid','Tcell','Bcell','CD8T','CD4T',
'NK','MoMacDC','Plasma_cell','PanBcell'). 
If a custom dataset is provided in Seurat format, this variable must be
a vector of positive cell types in your data. The last case also require
that such labels were named as in your cell_type meta.data column.</p>
</td></tr>
<tr><td><code id="test_my_model_+3A_plot">plot</code></td>
<td>
<p>Whether to return plots to device</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns performance metrics for the benchmarking datasets, and optionally
plots of the predicted cell type labels in reduced dimensionality space.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
scGate.model.db &lt;- get_scGateDB()
# Browse the list of models and select one:
model.panBcell &lt;-  scGate.model.db$human$generic$PanBcell
# Test the model with available testing datasets
panBcell.performance &lt;- test_my_model(model.panBcell, target = "PanBcell")
model.Myeloid &lt;-  scGate.model.db$human$generic$Myeloid
myeloid.performance &lt;- test_my_model(model.Myeloid, target = "Myeloid")
     
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
