<!DOCTYPE html><html lang="en"><head><title>Help for package scGOclust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scGOclust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#analyzeGOSeurat'><p>standard seurat analysis on GO_seurat object</p></a></li>
<li><a href='#cellTypeGOCorr'><p>calculate correlation between cell types represented by scaled GO, per-species</p></a></li>
<li><a href='#crossSpeciesCellTypeGOCorr'><p>calculate cross-species correlation between cell types represented by scaled GO</p></a></li>
<li><a href='#dme_subset'><p>Drosophila gut scRNA-seq data, 10X Chromium</p>
Subset to 45 cells per cell type as an example data</a></li>
<li><a href='#dme_tbl'><p>Drosophila EMSEMBL gene and GO annotation, subset to genes present in 'dme_subset'</p></a></li>
<li><a href='#ensemblToGo'><p>get requested ensembl ID to GO mapping table</p></a></li>
<li><a href='#getCellTypeGO'><p>get per cell type average scaled vector of GO terms</p></a></li>
<li><a href='#getCellTypeSharedGO'><p>get shared up and down regulated GO terms for all pairs of cell types</p></a></li>
<li><a href='#getCellTypeSharedTerms'><p>query co-up and co-down regulated GO terms from certain cell type pairs</p></a></li>
<li><a href='#globalvariables'><p>record some global variables: pre-defined column name in biomaRt query and markers</p></a></li>
<li><a href='#makeGOSeurat'><p>create a seurat object with GO terms</p></a></li>
<li><a href='#mmu_subset'><p>Mouse stomach and intestine scRNA-seq data, microwell-seq</p>
Subset to 50 cells per cell type as an example data</a></li>
<li><a href='#mmu_tbl'><p>Mouse EMSEMBL gene and GO annotation, subset to genes present in 'mmu_subset'</p></a></li>
<li><a href='#plotCellTypeCorrHeatmap'><p>plot clustered heatmap for cell type corr</p></a></li>
<li><a href='#plotCellTypeSankey'><p>plot Sankey diagram for cell type links above a certain threshould</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Measuring Cell Type Similarity with Gene Ontology in Single-Cell
RNA-Seq</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Traditional methods for analyzing single cell RNA-seq datasets focus solely on gene expression, but this package introduces a novel approach that goes beyond this limitation. Using Gene Ontology terms as features, the package allows for the functional profile of cell populations, and comparison within and between datasets from the same or different species. Our approach enables the discovery of previously unrecognized functional similarities and differences between cell types and has demonstrated success in identifying cell types' functional correspondence even between evolutionarily distant species.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Papatheodorou-Group/scGOclust">https://github.com/Papatheodorou-Group/scGOclust</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Papatheodorou-Group/scGOclust/issues">https://github.com/Papatheodorou-Group/scGOclust/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>bzip2</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>limma, Seurat(&ge; 5.0.0), biomaRt, dplyr, magrittr, stats,
tibble, tidyr, Matrix, utils, networkD3, slanter</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, devtools, pheatmap, rmarkdown, httr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-23 17:05:34 UTC; ysong</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuyao Song [aut, cre, ctb],
  Irene Papatheodorou [aut, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuyao Song &lt;ysong@ebi.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-24 14:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='analyzeGOSeurat'>standard seurat analysis on GO_seurat object</h2><span id='topic+analyzeGOSeurat'></span>

<h3>Description</h3>

<p>standard seurat analysis on GO_seurat object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyzeGOSeurat(
  go_seurat_obj,
  cell_type_col,
  norm_log1p = TRUE,
  scale.factor = 10000,
  nfeatures = 2000,
  cluster_res = 1,
  min.dist = 0.3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyzeGOSeurat_+3A_go_seurat_obj">go_seurat_obj</code></td>
<td>
<p>go seurat object created by makeGOSeurat</p>
</td></tr>
<tr><td><code id="analyzeGOSeurat_+3A_cell_type_col">cell_type_col</code></td>
<td>
<p>column name in mera.data storing cell type classes</p>
</td></tr>
<tr><td><code id="analyzeGOSeurat_+3A_norm_log1p">norm_log1p</code></td>
<td>
<p>whether or not to perform data normalisation and log1p transformation, default TRUE</p>
</td></tr>
<tr><td><code id="analyzeGOSeurat_+3A_scale.factor">scale.factor</code></td>
<td>
<p>param for Seurat NormalizeData</p>
</td></tr>
<tr><td><code id="analyzeGOSeurat_+3A_nfeatures">nfeatures</code></td>
<td>
<p>param for Seurat FindVariableFeatures</p>
</td></tr>
<tr><td><code id="analyzeGOSeurat_+3A_cluster_res">cluster_res</code></td>
<td>
<p>resolution for Seurat FindClusters</p>
</td></tr>
<tr><td><code id="analyzeGOSeurat_+3A_min.dist">min.dist</code></td>
<td>
<p>param for Seurat RunUMAP</p>
</td></tr>
<tr><td><code id="analyzeGOSeurat_+3A_...">...</code></td>
<td>
<p>additional params for all Seurat functions involved in this function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>standard analyzed GO seurat object until UMAP
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)
go_seurat_obj = makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")

analyzeGOSeurat(go_seurat_obj = go_seurat_obj, cell_type_col = "cell_type_annotation")

</code></pre>

<hr>
<h2 id='cellTypeGOCorr'>calculate correlation between cell types represented by scaled GO, per-species</h2><span id='topic+cellTypeGOCorr'></span>

<h3>Description</h3>

<p>calculate correlation between cell types represented by scaled GO, per-species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cellTypeGOCorr(cell_type_go, corr_method = "pearson")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cellTypeGOCorr_+3A_cell_type_go">cell_type_go</code></td>
<td>
<p>cell type GO table calculated via getCellTypeGO</p>
</td></tr>
<tr><td><code id="cellTypeGOCorr_+3A_corr_method">corr_method</code></td>
<td>
<p>correlation method, choose among &quot;pearson&quot;, &quot;kendall&quot;, &quot;spearman&quot;, default 'pearson'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe with correlation between cell types
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)
go_seurat_obj = makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")

cell_type_go = getCellTypeGO(go_seurat_obj = go_seurat_obj, cell_type_co = "cell_type_annotation")

cellTypeGOCorr(cell_type_go = cell_type_go, corr_method = "pearson")


</code></pre>

<hr>
<h2 id='crossSpeciesCellTypeGOCorr'>calculate cross-species correlation between cell types represented by scaled GO</h2><span id='topic+crossSpeciesCellTypeGOCorr'></span>

<h3>Description</h3>

<p>calculate cross-species correlation between cell types represented by scaled GO
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crossSpeciesCellTypeGOCorr(
  species_1,
  species_2,
  cell_type_go_sp1,
  cell_type_go_sp2,
  corr_method = "pearson"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="crossSpeciesCellTypeGOCorr_+3A_species_1">species_1</code></td>
<td>
<p>name of species one</p>
</td></tr>
<tr><td><code id="crossSpeciesCellTypeGOCorr_+3A_species_2">species_2</code></td>
<td>
<p>name of species two</p>
</td></tr>
<tr><td><code id="crossSpeciesCellTypeGOCorr_+3A_cell_type_go_sp1">cell_type_go_sp1</code></td>
<td>
<p>cell type GO table of species one calculated via getCellTypeGO</p>
</td></tr>
<tr><td><code id="crossSpeciesCellTypeGOCorr_+3A_cell_type_go_sp2">cell_type_go_sp2</code></td>
<td>
<p>cell type GO table of species two calculated via getCellTypeGO</p>
</td></tr>
<tr><td><code id="crossSpeciesCellTypeGOCorr_+3A_corr_method">corr_method</code></td>
<td>
<p>correlation method, choose among &quot;pearson&quot;, &quot;kendall&quot;, &quot;spearman&quot;, default 'pearson'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>correlation between cell types
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)
data(dme_tbl)
data(dme_subset)
mmu_go_obj = makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")
dme_go_obj = makeGOSeurat(ensembl_to_GO = dme_tbl,
 seurat_obj = dme_subset,
 feature_type = "external_gene_name")

mmu_cell_type_go = getCellTypeGO(go_seurat_obj = mmu_go_obj, cell_type_co = "cell_type_annotation")
dme_cell_type_go = getCellTypeGO(go_seurat_obj = dme_go_obj, cell_type_co = "annotation")

crossSpeciesCellTypeGOCorr(species_1 = 'mmusculus',
 species_2 = 'dmelanogaster',
 cell_type_go_sp1 = mmu_cell_type_go,
 cell_type_go_sp2 = dme_cell_type_go)

</code></pre>

<hr>
<h2 id='dme_subset'>Drosophila gut scRNA-seq data, 10X Chromium
Subset to 45 cells per cell type as an example data</h2><span id='topic+dme_subset'></span>

<h3>Description</h3>

<p>Drosophila gut scRNA-seq data, 10X Chromium
Subset to 45 cells per cell type as an example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dme_subset
</code></pre>


<h3>Format</h3>

<p>a 'Seurat' object
</p>


<h3>Source</h3>

<p>&lt;https://flycellatlas.org/&gt;
</p>

<hr>
<h2 id='dme_tbl'>Drosophila EMSEMBL gene and GO annotation, subset to genes present in 'dme_subset'</h2><span id='topic+dme_tbl'></span>

<h3>Description</h3>

<p>Drosophila EMSEMBL gene and GO annotation, subset to genes present in 'dme_subset'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dme_tbl
</code></pre>


<h3>Format</h3>

<p>a 'data.frame' object
</p>


<h3>Source</h3>

<p>&lt;http://www.ensembl.org/&gt;
</p>

<hr>
<h2 id='ensemblToGo'>get requested ensembl ID to GO mapping table</h2><span id='topic+ensemblToGo'></span>

<h3>Description</h3>

<p>get requested ensembl ID to GO mapping table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensemblToGo(
  species,
  GO_type = "biological_process",
  GO_linkage_type = c("standard"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ensemblToGo_+3A_species">species</code></td>
<td>
<p>species name matching ensembl biomaRt naming, such as hsapiens, mmusculus</p>
</td></tr>
<tr><td><code id="ensemblToGo_+3A_go_type">GO_type</code></td>
<td>
<p>GO term type, choose among 'biological_process', 'molecular_function', 'cellular_component', default 'biological_process'</p>
</td></tr>
<tr><td><code id="ensemblToGo_+3A_go_linkage_type">GO_linkage_type</code></td>
<td>
<p>GO annotation evidence codes to include. Default is 'standard', which means only including manually checked records (excluding IEA) and excluding those inferred from gene expression experiments to maximally suffice the species expression independence assumption. 'Stringent' means only including those with experimental evidence, also not from gene expression experiment, or from manual curation with evidence (excluding those from mass-annotation pipelines). Choose among 'experimental', 'phylogenetic', 'computational', 'author', 'curator', 'electronic', 'standard', stringent'</p>
</td></tr>
<tr><td><code id="ensemblToGo_+3A_...">...</code></td>
<td>
<p>additional params for useEnsembl function called in this function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table with ensembl to GO terms mapping including requested linkage type
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
ensemblToGo("mmusculus", GO_type = "biological_process", GO_linkage_type = 'stringent')

</code></pre>

<hr>
<h2 id='getCellTypeGO'>get per cell type average scaled vector of GO terms</h2><span id='topic+getCellTypeGO'></span>

<h3>Description</h3>

<p>get per cell type average scaled vector of GO terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCellTypeGO(go_seurat_obj, cell_type_col, norm_log1p = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCellTypeGO_+3A_go_seurat_obj">go_seurat_obj</code></td>
<td>
<p>go seurat object created by makeGOSeurat</p>
</td></tr>
<tr><td><code id="getCellTypeGO_+3A_cell_type_col">cell_type_col</code></td>
<td>
<p>column name in mera.data storing cell type classes</p>
</td></tr>
<tr><td><code id="getCellTypeGO_+3A_norm_log1p">norm_log1p</code></td>
<td>
<p>whether or not to perform data normalisation and log1p transformation, default TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table of scaled GO representation per cell type (averaged)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)
go_seurat_obj = makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")
getCellTypeGO(go_seurat_obj = go_seurat_obj, cell_type_co = "cell_type_annotation")

</code></pre>

<hr>
<h2 id='getCellTypeSharedGO'>get shared up and down regulated GO terms for all pairs of cell types</h2><span id='topic+getCellTypeSharedGO'></span>

<h3>Description</h3>

<p>get shared up and down regulated GO terms for all pairs of cell types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCellTypeSharedGO(
  species_1,
  species_2,
  analyzed_go_seurat_sp1,
  analyzed_go_seurat_sp2,
  cell_type_col_sp1,
  cell_type_col_sp2,
  layer_use = "data",
  p_val_threshould = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCellTypeSharedGO_+3A_species_1">species_1</code></td>
<td>
<p>name of species one</p>
</td></tr>
<tr><td><code id="getCellTypeSharedGO_+3A_species_2">species_2</code></td>
<td>
<p>name of species two</p>
</td></tr>
<tr><td><code id="getCellTypeSharedGO_+3A_analyzed_go_seurat_sp1">analyzed_go_seurat_sp1</code></td>
<td>
<p>analyzed GO seurat object of species one</p>
</td></tr>
<tr><td><code id="getCellTypeSharedGO_+3A_analyzed_go_seurat_sp2">analyzed_go_seurat_sp2</code></td>
<td>
<p>analyzed GO seurat object of species two</p>
</td></tr>
<tr><td><code id="getCellTypeSharedGO_+3A_cell_type_col_sp1">cell_type_col_sp1</code></td>
<td>
<p>cell type column name for species 1 data</p>
</td></tr>
<tr><td><code id="getCellTypeSharedGO_+3A_cell_type_col_sp2">cell_type_col_sp2</code></td>
<td>
<p>cell type column name for species 2 data</p>
</td></tr>
<tr><td><code id="getCellTypeSharedGO_+3A_layer_use">layer_use</code></td>
<td>
<p>layer to use for marker computation, default 'data' which after NormalizeData will be log1p normalized data.</p>
</td></tr>
<tr><td><code id="getCellTypeSharedGO_+3A_p_val_threshould">p_val_threshould</code></td>
<td>
<p>p value threshold for selecting DEG (p_adjust)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with sp1 raw, sp2 raw and shared, significant up and down regulated GO terms per cell type (pair)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)
data(dme_tbl)
data(dme_subset)

mmu_go_obj = makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")
dme_go_obj = makeGOSeurat(ensembl_to_GO = dme_tbl,
 seurat_obj = dme_subset,
 feature_type = "external_gene_name")


mmu_go_obj_analyzed = analyzeGOSeurat(mmu_go_obj, "cell_type_annotation")
dme_go_obj_analyzed = analyzeGOSeurat(dme_go_obj, "annotation")

getCellTypeSharedGO(species_1 = 'mmusculus',
species_2 = 'dmelanogaster',
analyzed_go_seurat_sp1 =  mmu_go_obj_analyzed,
analyzed_go_seurat_sp2 =  dme_go_obj_analyzed,
cell_type_col_sp1 = 'cell_type_annotation',
cell_type_col_sp2 = 'annotation',
layer_use = "data",
p_val_threshould = 0.01)

</code></pre>

<hr>
<h2 id='getCellTypeSharedTerms'>query co-up and co-down regulated GO terms from certain cell type pairs</h2><span id='topic+getCellTypeSharedTerms'></span>

<h3>Description</h3>

<p>query co-up and co-down regulated GO terms from certain cell type pairs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCellTypeSharedTerms(
  shared_go,
  cell_type_sp1,
  cell_type_sp2,
  return_full = FALSE,
  arrange_avg_log2FC = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCellTypeSharedTerms_+3A_shared_go">shared_go</code></td>
<td>
<p>cell type shared GO table from getCellTypeSharedGO</p>
</td></tr>
<tr><td><code id="getCellTypeSharedTerms_+3A_cell_type_sp1">cell_type_sp1</code></td>
<td>
<p>cell type from sp1 to query</p>
</td></tr>
<tr><td><code id="getCellTypeSharedTerms_+3A_cell_type_sp2">cell_type_sp2</code></td>
<td>
<p>cell type from sp2 to query</p>
</td></tr>
<tr><td><code id="getCellTypeSharedTerms_+3A_return_full">return_full</code></td>
<td>
<p>if return also pvals and logfc info, default FALSE</p>
</td></tr>
<tr><td><code id="getCellTypeSharedTerms_+3A_arrange_avg_log2fc">arrange_avg_log2FC</code></td>
<td>
<p>arrange result by decreasing mean avg_log2FC, default TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe displaying co-up or co-down regulated GO terms for the queried cell type pair
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)
data(dme_tbl)
data(dme_subset)

mmu_go_obj = makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")
dme_go_obj = makeGOSeurat(ensembl_to_GO = dme_tbl,
 seurat_obj = dme_subset,
 feature_type = "external_gene_name")


mmu_go_obj_analyzed = analyzeGOSeurat(mmu_go_obj, "cell_type_annotation")
dme_go_obj_analyzed = analyzeGOSeurat(dme_go_obj, "annotation")

shared_go = getCellTypeSharedGO(species_1 = 'mmusculus',
species_2 = 'dmelanogaster',
analyzed_go_seurat_sp1 = mmu_go_obj_analyzed,
analyzed_go_seurat_sp2 = dme_go_obj_analyzed,
cell_type_col_sp1 = 'cell_type_annotation',
cell_type_col_sp2 = 'annotation',
layer_use = "data",
p_val_threshould = 0.01)


getCellTypeSharedTerms(shared_go = shared_go,
cell_type_sp1 = 'intestine_Enteroendocrine cell',
cell_type_sp2 = 'enteroendocrine cell',
return_full = FALSE)

</code></pre>

<hr>
<h2 id='globalvariables'>record some global variables: pre-defined column name in biomaRt query and markers</h2><span id='topic+globalvariables'></span>

<h3>Description</h3>

<p>record some global variables: pre-defined column name in biomaRt query and markers
</p>

<hr>
<h2 id='makeGOSeurat'>create a seurat object with GO terms</h2><span id='topic+makeGOSeurat'></span>

<h3>Description</h3>

<p>create a seurat object with GO terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeGOSeurat(ensembl_to_GO, seurat_obj, feature_type = "ensembl_gene_id")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makeGOSeurat_+3A_ensembl_to_go">ensembl_to_GO</code></td>
<td>
<p>ensembl_to_go mapping table from function ensemblToGo</p>
</td></tr>
<tr><td><code id="makeGOSeurat_+3A_seurat_obj">seurat_obj</code></td>
<td>
<p>count matrix with genes to cells</p>
</td></tr>
<tr><td><code id="makeGOSeurat_+3A_feature_type">feature_type</code></td>
<td>
<p>feature type of count matrix, choose from ensembl_gene_id, external_gene_name, default ensembl_gene_id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a seurat object with GO terms as features
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)
makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")

</code></pre>

<hr>
<h2 id='mmu_subset'>Mouse stomach and intestine scRNA-seq data, microwell-seq
Subset to 50 cells per cell type as an example data</h2><span id='topic+mmu_subset'></span>

<h3>Description</h3>

<p>Mouse stomach and intestine scRNA-seq data, microwell-seq
Subset to 50 cells per cell type as an example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmu_subset
</code></pre>


<h3>Format</h3>

<p>a 'Seurat' object
</p>


<h3>Source</h3>

<p>&lt;https://bis.zju.edu.cn/MCA/&gt;
</p>

<hr>
<h2 id='mmu_tbl'>Mouse EMSEMBL gene and GO annotation, subset to genes present in 'mmu_subset'</h2><span id='topic+mmu_tbl'></span>

<h3>Description</h3>

<p>Mouse EMSEMBL gene and GO annotation, subset to genes present in 'mmu_subset'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmu_tbl
</code></pre>


<h3>Format</h3>

<p>a 'data.frame' object
</p>


<h3>Source</h3>

<p>&lt;http://www.ensembl.org/&gt;
</p>

<hr>
<h2 id='plotCellTypeCorrHeatmap'>plot clustered heatmap for cell type corr</h2><span id='topic+plotCellTypeCorrHeatmap'></span>

<h3>Description</h3>

<p>plot clustered heatmap for cell type corr
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCellTypeCorrHeatmap(corr_matrix, scale = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotCellTypeCorrHeatmap_+3A_corr_matrix">corr_matrix</code></td>
<td>
<p>correlation matrix from cellTypeGOCorr or crossSpeciesCellTypeGOCorr</p>
</td></tr>
<tr><td><code id="plotCellTypeCorrHeatmap_+3A_scale">scale</code></td>
<td>
<p>scale value by column, row, or default no scaling (NA)</p>
</td></tr>
<tr><td><code id="plotCellTypeCorrHeatmap_+3A_...">...</code></td>
<td>
<p>params to pass to slanter::sheatmap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a sheatmap heatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)

go_seurat_obj = makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")

cell_type_go = getCellTypeGO(go_seurat_obj = go_seurat_obj, cell_type_co = "cell_type_annotation")

corr_matrix = cellTypeGOCorr(cell_type_go = cell_type_go, corr_method = "pearson")

plotCellTypeCorrHeatmap(corr_matrix = corr_matrix, scale = "column")


</code></pre>

<hr>
<h2 id='plotCellTypeSankey'>plot Sankey diagram for cell type links above a certain threshould</h2><span id='topic+plotCellTypeSankey'></span>

<h3>Description</h3>

<p>plot Sankey diagram for cell type links above a certain threshould
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCellTypeSankey(corr_matrix, corr_threshould = 0.1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotCellTypeSankey_+3A_corr_matrix">corr_matrix</code></td>
<td>
<p>cell type corr matrix from crossSpeciesCellTypeGOCorr</p>
</td></tr>
<tr><td><code id="plotCellTypeSankey_+3A_corr_threshould">corr_threshould</code></td>
<td>
<p>minimum corr value for positively related cell types, default 0.6</p>
</td></tr>
<tr><td><code id="plotCellTypeSankey_+3A_...">...</code></td>
<td>
<p>additional params for sankeyNetwork</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Sankey plot showing related cell types
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(scGOclust)
library(httr)
httr::set_config(httr::config(ssl_verifypeer = FALSE))
data(mmu_tbl)
data(mmu_subset)
go_seurat_obj = makeGOSeurat(ensembl_to_GO = mmu_tbl,
 seurat_obj = mmu_subset,
 feature_type = "external_gene_name")

cell_type_go = getCellTypeGO(go_seurat_obj = go_seurat_obj, cell_type_co = "cell_type_annotation")
corr_matrix = cellTypeGOCorr(cell_type_go = cell_type_go, corr_method = "pearson")

plotCellTypeSankey(corr_matrix = corr_matrix, 0.1)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
