<!DOCTYPE html><html><head><title>Help for package scimo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scimo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#scimo-package'><p>scimo: Extra Recipes Steps for Dealing with Omics Data</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#cheese_abundance'><p>Abundance of Fungal Communities in Cheese</p></a></li>
<li><a href='#cv'><p>Coefficient of variation</p></a></li>
<li><a href='#pedcan_expression'><p>Gene Expression of Pediatric Cancer</p></a></li>
<li><a href='#step_aggregate_hclust'><p>Feature aggregation step based on a hierarchical clustering</p></a></li>
<li><a href='#step_aggregate_list'><p>Feature aggregation step based on a defined list</p></a></li>
<li><a href='#step_rownormalize_tss'><p>Feature normalization step using total sum scaling</p></a></li>
<li><a href='#step_select_background'><p>Feature selection step using background level</p></a></li>
<li><a href='#step_select_cv'><p>Feature selection step using the coefficient of variation</p></a></li>
<li><a href='#step_select_kruskal'><p>Feature selection step using Kruskal test</p></a></li>
<li><a href='#step_select_wilcoxon'><p>Feature selection step using Wilcoxon test</p></a></li>
<li><a href='#var_to_keep'><p>Decide which variable to keep</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Extra Recipes Steps for Dealing with Omics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Omics data (e.g. transcriptomics, proteomics,
    metagenomics...)  offer a detailed and multi-dimensional perspective
    on the molecular components and interactions within complex biological
    (eco)systems.  Analyzing these data requires adapted procedures, which
    are implemented as steps according to the 'recipes' package.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/abichat/scimo">https://github.com/abichat/scimo</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/abichat/scimo/issues">https://github.com/abichat/scimo/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), recipes</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, generics, magrittr, rlang, stats, tibble, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-02 17:20:13 UTC; antoinebichat</td>
</tr>
<tr>
<td>Author:</td>
<td>Antoine BICHAT <a href="https://orcid.org/0000-0001-6599-7081"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Julie AUBERT <a href="https://orcid.org/0000-0001-5203-5748"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Antoine BICHAT &lt;antoine.bichat@proton.me&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-03 19:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='scimo-package'>scimo: Extra Recipes Steps for Dealing with Omics Data</h2><span id='topic+scimo'></span><span id='topic+scimo-package'></span>

<h3>Description</h3>

<p>Omics data (e.g. transcriptomics, proteomics, metagenomics...) offer a detailed and multi-dimensional perspective on the molecular components and interactions within complex biological (eco)systems. Analyzing these data requires adapted procedures, which are implemented as steps according to the 'recipes' package.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Antoine BICHAT <a href="mailto:antoine.bichat@proton.me">antoine.bichat@proton.me</a> (<a href="https://orcid.org/0000-0001-6599-7081">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Julie AUBERT <a href="mailto:julie.aubert@inrae.fr">julie.aubert@inrae.fr</a> (<a href="https://orcid.org/0000-0001-5203-5748">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/abichat/scimo">https://github.com/abichat/scimo</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/abichat/scimo/issues">https://github.com/abichat/scimo/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='cheese_abundance'>Abundance of Fungal Communities in Cheese</h2><span id='topic+cheese_abundance'></span><span id='topic+cheese_taxonomy'></span>

<h3>Description</h3>

<p>Fungal community abundance of 74 ASVs sampled from the surface of three
different French cheeses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("cheese_abundance", package = "scimo")

data("cheese_taxonomy", package = "scimo")
</code></pre>


<h3>Format</h3>

<p>For <code>cheese_abundance</code>, a <a href="tibble.html#topic+tibble-package">tibble</a>
with columns:
</p>

<dl>
<dt>sample</dt><dd><p>Sample ID.</p>
</dd>
<dt>cheese</dt><dd><p>Appellation of the cheese. One of <code>Saint-Nectaire</code>,
<code>Livarot</code> or <code>Epoisses</code>.</p>
</dd>
<dt>rind_type</dt><dd><p>One of <code>Natural</code> or <code>Washed</code>.</p>
</dd>
<dt>other columns</dt><dd><p>Count of the ASV.</p>
</dd>
</dl>

<p>For <code>cheese_taxonomy</code>, a <a href="tibble.html#topic+tibble-package">tibble</a>
with columns:
</p>

<dl>
<dt>asv</dt><dd><p>Amplicon Sequence Variant (ASV) ID.</p>
</dd>
<dt>lineage</dt><dd><p>Character corresponding to a standard concatenation of
taxonomic clades.</p>
</dd>
<dt>other columns</dt><dd><p>Clade to which the ASV belongs.</p>
</dd>
</dl>



<h3>Source</h3>

<p>This dataset came from <a href="https://doi.org/10.24072/pcjournal.321">doi:10.24072/pcjournal.321</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("cheese_abundance", package = "scimo")
cheese_abundance
data("cheese_taxonomy", package = "scimo")
cheese_taxonomy
</code></pre>

<hr>
<h2 id='cv'>Coefficient of variation</h2><span id='topic+cv'></span>

<h3>Description</h3>

<p>Coefficient of variation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv(x, na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv_+3A_x">x</code></td>
<td>
<p>A numeric vector.</p>
</td></tr>
<tr><td><code id="cv_+3A_na.rm">na.rm</code></td>
<td>
<p>Logical indicating whether NA values should be stripped
before the computation proceeds. Default to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The coefficient of variation of <code>x</code>.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cv(1:10)
</code></pre>

<hr>
<h2 id='pedcan_expression'>Gene Expression of Pediatric Cancer</h2><span id='topic+pedcan_expression'></span>

<h3>Description</h3>

<p>Gene expression of 108 CCLE cell lines from 5 different pediatric cancers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pedcan_expression", package = "scimo")
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> with columns:
</p>

<dl>
<dt>cell_line</dt><dd><p>Cell line name.</p>
</dd>
<dt>sex</dt><dd><p>One of <code>Male</code>, <code>Female</code> or <code>Unknown</code>.</p>
</dd>
<dt>event</dt><dd><p>One of <code>Primary</code>, <code>Metastasis</code> or <code>Unknown</code>.</p>
</dd>
<dt>disease</dt><dd><p>One of <code>Neuroblastoma</code>, <code style="white-space: pre;">&#8288;Ewing Sarcoma&#8288;</code>,
<code>Rhabdomyosarcoma</code>, <code style="white-space: pre;">&#8288;Embryonal Tumor&#8288;</code> or <code>Osteosarcoma</code>.</p>
</dd>
<dt>other columns</dt><dd><p>Expression of the gene, given in log2(TPM + 1).</p>
</dd>
</dl>



<h3>Source</h3>

<p>This dataset is generated from DepMap Public 23Q4 primary files.
<a href="https://depmap.org/portal/download/all/">https://depmap.org/portal/download/all/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pedcan_expression", package = "scimo")
pedcan_expression
</code></pre>

<hr>
<h2 id='step_aggregate_hclust'>Feature aggregation step based on a hierarchical clustering</h2><span id='topic+step_aggregate_hclust'></span><span id='topic+tidy.step_aggregate_hclust'></span>

<h3>Description</h3>

<p>Aggregate variables according to hierarchical clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>step_aggregate_hclust(
  recipe,
  ...,
  role = "predictor",
  trained = FALSE,
  n_clusters,
  fun_agg,
  dist_metric = "euclidean",
  linkage_method = "complete",
  res = NULL,
  prefix = "cl_",
  keep_original_cols = FALSE,
  skip = FALSE,
  id = rand_id("aggregate_hclust")
)

## S3 method for class 'step_aggregate_hclust'
tidy(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="step_aggregate_hclust_+3A_recipe">recipe</code></td>
<td>
<p>A recipe object. The step will be added to the sequence of
operations for this recipe.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_...">...</code></td>
<td>
<p>One or more selector functions to choose variables
for this step. See <code><a href="recipes.html#topic+selections">selections()</a></code> for more details.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_role">role</code></td>
<td>
<p>For model terms created by this step, what analysis role should
they be assigned? By default, the new columns created by this step from
the original variables will be used as <code>predictors</code> in a model.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_trained">trained</code></td>
<td>
<p>A logical to indicate if the quantities for preprocessing
have been estimated.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_n_clusters">n_clusters</code></td>
<td>
<p>Number of cluster to create.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_fun_agg">fun_agg</code></td>
<td>
<p>Aggregation function like <code>sum</code> or <code>mean</code>.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_dist_metric">dist_metric</code></td>
<td>
<p>Default to <code>euclidean</code>. See <code><a href="stats.html#topic+dist">stats::dist()</a></code> for more
details.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_linkage_method">linkage_method</code></td>
<td>
<p>Default to <code>complete</code>. See <code><a href="stats.html#topic+hclust">stats::hclust()</a></code> for more
details.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_res">res</code></td>
<td>
<p>This parameter is only produced after the recipe has been trained.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_prefix">prefix</code></td>
<td>
<p>A character string for the prefix of the resulting new
variables.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_keep_original_cols">keep_original_cols</code></td>
<td>
<p>A logical to keep the original variables in
the output. Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_skip">skip</code></td>
<td>
<p>A logical. Should the step be skipped when the
recipe is baked by <code><a href="recipes.html#topic+bake">bake()</a></code>? While all operations are baked
when <code><a href="recipes.html#topic+prep">prep()</a></code> is run, some operations may not be able to be
conducted on new data (e.g. processing the outcome variable(s)).
Care should be taken when using <code>skip = TRUE</code> as it may affect
the computations for subsequent operations.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_id">id</code></td>
<td>
<p>A character string that is unique to this step to identify it.</p>
</td></tr>
<tr><td><code id="step_aggregate_hclust_+3A_x">x</code></td>
<td>
<p>A <code>step_aggregate_hclust</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated version of recipe with the new step added to the
sequence of any existing operations.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rec &lt;-
  iris %&gt;%
  recipe(formula = Species ~ .) %&gt;%
  step_aggregate_hclust(all_numeric_predictors(),
                        n_clusters = 2, fun_agg = sum) %&gt;%
  prep()
rec
tidy(rec, 1)
juice(rec)
</code></pre>

<hr>
<h2 id='step_aggregate_list'>Feature aggregation step based on a defined list</h2><span id='topic+step_aggregate_list'></span><span id='topic+tidy.step_aggregate_list'></span>

<h3>Description</h3>

<p>Aggregate variables according to prior knowledge.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>step_aggregate_list(
  recipe,
  ...,
  role = "predictor",
  trained = FALSE,
  list_agg = NULL,
  fun_agg = NULL,
  others = "discard",
  name_others = "others",
  res = NULL,
  prefix = "agg_",
  keep_original_cols = FALSE,
  skip = FALSE,
  id = rand_id("aggregate_list")
)

## S3 method for class 'step_aggregate_list'
tidy(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="step_aggregate_list_+3A_recipe">recipe</code></td>
<td>
<p>A recipe object. The step will be added to the sequence of
operations for this recipe.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_...">...</code></td>
<td>
<p>One or more selector functions to choose variables
for this step. See <code><a href="recipes.html#topic+selections">selections()</a></code> for more details.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_role">role</code></td>
<td>
<p>For model terms created by this step, what analysis role should
they be assigned? By default, the new columns created by this step from
the original variables will be used as <code>predictors</code> in a model.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_trained">trained</code></td>
<td>
<p>A logical to indicate if the quantities for preprocessing
have been estimated.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_list_agg">list_agg</code></td>
<td>
<p>Named list of aggregated variables.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_fun_agg">fun_agg</code></td>
<td>
<p>Aggregation function like <code>sum</code> or <code>mean</code>.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_others">others</code></td>
<td>
<p>Behavior for the selected variables in <code>...</code> that are not present in
<code>list_agg</code>. If <code>discard</code> (the default), they are not kept. If <code>asis</code>, they
are kept without modification. If <code>aggregate</code>, they are aggregated in a
new variable.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_name_others">name_others</code></td>
<td>
<p>If <code>others</code> is set to <code>aggregate</code>, name of the
aggregated variable. Not used otherwise.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_res">res</code></td>
<td>
<p>This parameter is only produced after the recipe has been trained.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_prefix">prefix</code></td>
<td>
<p>A character string for the prefix of the resulting new
variables that are not named in <code>list_agg</code>.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_keep_original_cols">keep_original_cols</code></td>
<td>
<p>A logical to keep the original variables in
the output. Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_skip">skip</code></td>
<td>
<p>A logical. Should the step be skipped when the
recipe is baked by <code><a href="recipes.html#topic+bake">bake()</a></code>? While all operations are baked
when <code><a href="recipes.html#topic+prep">prep()</a></code> is run, some operations may not be able to be
conducted on new data (e.g. processing the outcome variable(s)).
Care should be taken when using <code>skip = TRUE</code> as it may affect
the computations for subsequent operations.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_id">id</code></td>
<td>
<p>A character string that is unique to this step to identify it.</p>
</td></tr>
<tr><td><code id="step_aggregate_list_+3A_x">x</code></td>
<td>
<p>A <code>step_aggregate_list</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated version of recipe with the new step added to the
sequence of any existing operations.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>list_iris &lt;- list(sepal.size = c("Sepal.Length", "Sepal.Width"),
                  petal.size = c("Petal.Length", "Petal.Width"))
rec &lt;-
  iris %&gt;%
  recipe(formula = Species ~ .) %&gt;%
  step_aggregate_list(all_numeric_predictors(),
                      list_agg = list_iris, fun_agg = prod) %&gt;%
  prep()
rec
tidy(rec, 1)
juice(rec)
</code></pre>

<hr>
<h2 id='step_rownormalize_tss'>Feature normalization step using total sum scaling</h2><span id='topic+step_rownormalize_tss'></span><span id='topic+tidy.step_rownormalize_tss'></span>

<h3>Description</h3>

<p>Normalize a set of variables by converting them to proportion, making
them sum to 1. Also known as simplex projection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>step_rownormalize_tss(
  recipe,
  ...,
  role = NA,
  trained = FALSE,
  res = NULL,
  skip = FALSE,
  id = rand_id("rownormalize_tss")
)

## S3 method for class 'step_rownormalize_tss'
tidy(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="step_rownormalize_tss_+3A_recipe">recipe</code></td>
<td>
<p>A recipe object. The step will be added to the sequence of
operations for this recipe.</p>
</td></tr>
<tr><td><code id="step_rownormalize_tss_+3A_...">...</code></td>
<td>
<p>One or more selector functions to choose variables
for this step. See <code><a href="recipes.html#topic+selections">selections()</a></code> for more details.</p>
</td></tr>
<tr><td><code id="step_rownormalize_tss_+3A_role">role</code></td>
<td>
<p>Not used by this step since no new variables are created.</p>
</td></tr>
<tr><td><code id="step_rownormalize_tss_+3A_trained">trained</code></td>
<td>
<p>A logical to indicate if the quantities for preprocessing
have been estimated.</p>
</td></tr>
<tr><td><code id="step_rownormalize_tss_+3A_res">res</code></td>
<td>
<p>This parameter is only produced after the recipe has been trained.</p>
</td></tr>
<tr><td><code id="step_rownormalize_tss_+3A_skip">skip</code></td>
<td>
<p>A logical. Should the step be skipped when the
recipe is baked by <code><a href="recipes.html#topic+bake">bake()</a></code>? While all operations are baked
when <code><a href="recipes.html#topic+prep">prep()</a></code> is run, some operations may not be able to be
conducted on new data (e.g. processing the outcome variable(s)).
Care should be taken when using <code>skip = TRUE</code> as it may affect
the computations for subsequent operations.</p>
</td></tr>
<tr><td><code id="step_rownormalize_tss_+3A_id">id</code></td>
<td>
<p>A character string that is unique to this step to identify it.</p>
</td></tr>
<tr><td><code id="step_rownormalize_tss_+3A_x">x</code></td>
<td>
<p>A <code>step_rownormalize_tss</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated version of recipe with the new step added to the
sequence of any existing operations.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rec &lt;-
  recipe(Species ~ ., data = iris) %&gt;%
  step_rownormalize_tss(all_numeric_predictors()) %&gt;%
  prep()
rec
tidy(rec, 1)
juice(rec)
</code></pre>

<hr>
<h2 id='step_select_background'>Feature selection step using background level</h2><span id='topic+step_select_background'></span><span id='topic+tidy.step_select_background'></span>

<h3>Description</h3>

<p>Select features that exceed a background level in at least a defined
number of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>step_select_background(
  recipe,
  ...,
  role = NA,
  trained = FALSE,
  background_level = NULL,
  n_samples = NULL,
  prop_samples = NULL,
  res = NULL,
  skip = FALSE,
  id = rand_id("select_background")
)

## S3 method for class 'step_select_background'
tidy(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="step_select_background_+3A_recipe">recipe</code></td>
<td>
<p>A recipe object. The step will be added to the sequence of
operations for this recipe.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_...">...</code></td>
<td>
<p>One or more selector functions to choose variables
for this step. See <code><a href="recipes.html#topic+selections">selections()</a></code> for more details.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_role">role</code></td>
<td>
<p>Not used by this step since no new variables are created.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_trained">trained</code></td>
<td>
<p>A logical to indicate if the quantities for preprocessing
have been estimated.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_background_level">background_level</code></td>
<td>
<p>Background level to exceed.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_n_samples">n_samples</code>, <code id="step_select_background_+3A_prop_samples">prop_samples</code></td>
<td>
<p>Count or proportion of samples in which a
feature exceeds <code>background_level</code> to be retained.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_res">res</code></td>
<td>
<p>This parameter is only produced after the recipe has been trained.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_skip">skip</code></td>
<td>
<p>A logical. Should the step be skipped when the
recipe is baked by <code><a href="recipes.html#topic+bake">bake()</a></code>? While all operations are baked
when <code><a href="recipes.html#topic+prep">prep()</a></code> is run, some operations may not be able to be
conducted on new data (e.g. processing the outcome variable(s)).
Care should be taken when using <code>skip = TRUE</code> as it may affect
the computations for subsequent operations.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_id">id</code></td>
<td>
<p>A character string that is unique to this step to identify it.</p>
</td></tr>
<tr><td><code id="step_select_background_+3A_x">x</code></td>
<td>
<p>A <code>step_select_background</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated version of recipe with the new step added to the
sequence of any existing operations.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rec &lt;-
  iris %&gt;%
  recipe(formula = Species ~ .) %&gt;%
  step_select_background(all_numeric_predictors(),
                         background_level = 4, prop_samples = 0.5) %&gt;%
  prep()
rec
tidy(rec, 1)
juice(rec)
</code></pre>

<hr>
<h2 id='step_select_cv'>Feature selection step using the coefficient of variation</h2><span id='topic+step_select_cv'></span><span id='topic+tidy.step_select_cv'></span>

<h3>Description</h3>

<p>Select variables with highest coefficient of variation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>step_select_cv(
  recipe,
  ...,
  role = NA,
  trained = FALSE,
  n_kept = NULL,
  prop_kept = NULL,
  cutoff = NULL,
  res = NULL,
  skip = FALSE,
  id = rand_id("select_cv")
)

## S3 method for class 'step_select_cv'
tidy(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="step_select_cv_+3A_recipe">recipe</code></td>
<td>
<p>A recipe object. The step will be added to the sequence of
operations for this recipe.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_...">...</code></td>
<td>
<p>One or more selector functions to choose variables
for this step. See <code><a href="recipes.html#topic+selections">selections()</a></code> for more details.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_role">role</code></td>
<td>
<p>Not used by this step since no new variables are created.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_trained">trained</code></td>
<td>
<p>A logical to indicate if the quantities for preprocessing
have been estimated.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_n_kept">n_kept</code></td>
<td>
<p>Number of variables to keep.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_prop_kept">prop_kept</code></td>
<td>
<p>A numeric value between 0 and 1 representing the proportion
of variables to keep. <code>n_kept</code> and <code>prop_kept</code> are mutually exclusive.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_cutoff">cutoff</code></td>
<td>
<p>Threshold beyond which (below or above) the variables are
discarded.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_res">res</code></td>
<td>
<p>This parameter is only produced after the recipe has been trained.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_skip">skip</code></td>
<td>
<p>A logical. Should the step be skipped when the
recipe is baked by <code><a href="recipes.html#topic+bake">bake()</a></code>? While all operations are baked
when <code><a href="recipes.html#topic+prep">prep()</a></code> is run, some operations may not be able to be
conducted on new data (e.g. processing the outcome variable(s)).
Care should be taken when using <code>skip = TRUE</code> as it may affect
the computations for subsequent operations.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_id">id</code></td>
<td>
<p>A character string that is unique to this step to identify it.</p>
</td></tr>
<tr><td><code id="step_select_cv_+3A_x">x</code></td>
<td>
<p>A <code>step_select_cv</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated version of recipe with the new step added to the
sequence of any existing operations.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rec &lt;-
  recipe(Species ~ ., data = iris) %&gt;%
  step_select_cv(all_numeric_predictors(), n_kept = 2) %&gt;%
  prep()
rec
tidy(rec, 1)
juice(rec)
</code></pre>

<hr>
<h2 id='step_select_kruskal'>Feature selection step using Kruskal test</h2><span id='topic+step_select_kruskal'></span><span id='topic+tidy.step_select_kruskal'></span>

<h3>Description</h3>

<p>Select variables with the lowest (adjusted) p-value of a
Kruskal-Wallis test against an outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>step_select_kruskal(
  recipe,
  ...,
  role = NA,
  trained = FALSE,
  outcome = NULL,
  n_kept = NULL,
  prop_kept = NULL,
  cutoff = NULL,
  correction = "none",
  res = NULL,
  skip = FALSE,
  id = rand_id("select_kruskal")
)

## S3 method for class 'step_select_kruskal'
tidy(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="step_select_kruskal_+3A_recipe">recipe</code></td>
<td>
<p>A recipe object. The step will be added to the sequence of
operations for this recipe.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_...">...</code></td>
<td>
<p>One or more selector functions to choose variables
for this step. See <code><a href="recipes.html#topic+selections">selections()</a></code> for more details.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_role">role</code></td>
<td>
<p>Not used by this step since no new variables are created.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_trained">trained</code></td>
<td>
<p>A logical to indicate if the quantities for preprocessing
have been estimated.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_outcome">outcome</code></td>
<td>
<p>Name of the variable to perform the test against.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_n_kept">n_kept</code></td>
<td>
<p>Number of variables to keep.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_prop_kept">prop_kept</code></td>
<td>
<p>A numeric value between 0 and 1 representing the proportion
of variables to keep. <code>n_kept</code> and <code>prop_kept</code> are mutually exclusive.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_cutoff">cutoff</code></td>
<td>
<p>Threshold beyond which (below or above) the variables are
discarded.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_correction">correction</code></td>
<td>
<p>Multiple testing correction method. One of
<code>p.adjust.methods</code>. Default to <code>"none"</code>.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_res">res</code></td>
<td>
<p>This parameter is only produced after the recipe has been trained.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_skip">skip</code></td>
<td>
<p>A logical. Should the step be skipped when the
recipe is baked by <code><a href="recipes.html#topic+bake">bake()</a></code>? While all operations are baked
when <code><a href="recipes.html#topic+prep">prep()</a></code> is run, some operations may not be able to be
conducted on new data (e.g. processing the outcome variable(s)).
Care should be taken when using <code>skip = TRUE</code> as it may affect
the computations for subsequent operations.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_id">id</code></td>
<td>
<p>A character string that is unique to this step to identify it.</p>
</td></tr>
<tr><td><code id="step_select_kruskal_+3A_x">x</code></td>
<td>
<p>A <code>step_select_kruskal</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated version of recipe with the new step added to the
sequence of any existing operations.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rec &lt;-
  iris %&gt;%
  recipe(formula = Species ~ .) %&gt;%
  step_select_kruskal(all_numeric_predictors(), outcome = "Species",
                      correction = "fdr", prop_kept = 0.5) %&gt;%
  prep()
rec
tidy(rec, 1)
juice(rec)
</code></pre>

<hr>
<h2 id='step_select_wilcoxon'>Feature selection step using Wilcoxon test</h2><span id='topic+step_select_wilcoxon'></span><span id='topic+tidy.step_select_wilcoxon'></span>

<h3>Description</h3>

<p>Select variables with the lowest (adjusted) p-value of a
Wilcoxon-Mann-Whitney test against an outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>step_select_wilcoxon(
  recipe,
  ...,
  role = NA,
  trained = FALSE,
  outcome = NULL,
  n_kept = NULL,
  prop_kept = NULL,
  cutoff = NULL,
  correction = "none",
  res = NULL,
  skip = FALSE,
  id = rand_id("select_wilcoxon")
)

## S3 method for class 'step_select_wilcoxon'
tidy(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="step_select_wilcoxon_+3A_recipe">recipe</code></td>
<td>
<p>A recipe object. The step will be added to the sequence of
operations for this recipe.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_...">...</code></td>
<td>
<p>One or more selector functions to choose variables
for this step. See <code><a href="recipes.html#topic+selections">selections()</a></code> for more details.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_role">role</code></td>
<td>
<p>Not used by this step since no new variables are created.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_trained">trained</code></td>
<td>
<p>A logical to indicate if the quantities for preprocessing
have been estimated.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_outcome">outcome</code></td>
<td>
<p>Name of the variable to perform the test against.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_n_kept">n_kept</code></td>
<td>
<p>Number of variables to keep.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_prop_kept">prop_kept</code></td>
<td>
<p>A numeric value between 0 and 1 representing the proportion
of variables to keep. <code>n_kept</code> and <code>prop_kept</code> are mutually exclusive.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_cutoff">cutoff</code></td>
<td>
<p>Threshold beyond which (below or above) the variables are
discarded.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_correction">correction</code></td>
<td>
<p>Multiple testing correction method. One of
<code>p.adjust.methods</code>. Default to <code>"none"</code>.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_res">res</code></td>
<td>
<p>This parameter is only produced after the recipe has been trained.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_skip">skip</code></td>
<td>
<p>A logical. Should the step be skipped when the
recipe is baked by <code><a href="recipes.html#topic+bake">bake()</a></code>? While all operations are baked
when <code><a href="recipes.html#topic+prep">prep()</a></code> is run, some operations may not be able to be
conducted on new data (e.g. processing the outcome variable(s)).
Care should be taken when using <code>skip = TRUE</code> as it may affect
the computations for subsequent operations.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_id">id</code></td>
<td>
<p>A character string that is unique to this step to identify it.</p>
</td></tr>
<tr><td><code id="step_select_wilcoxon_+3A_x">x</code></td>
<td>
<p>A <code>step_select_wilcoxon</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated version of recipe with the new step added to the
sequence of any existing operations.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rec &lt;-
  iris %&gt;%
  dplyr::filter(Species != "virginica") %&gt;%
  recipe(formula = Species ~ .) %&gt;%
  step_select_wilcoxon(all_numeric_predictors(), outcome = "Species",
                       correction = "fdr", prop_kept = 0.5) %&gt;%
  prep()
rec
tidy(rec, 1)
juice(rec)
</code></pre>

<hr>
<h2 id='var_to_keep'>Decide which variable to keep</h2><span id='topic+var_to_keep'></span>

<h3>Description</h3>

<p>Decide which variable to keep
</p>


<h3>Usage</h3>

<pre><code class='language-R'>var_to_keep(
  values,
  n_kept = NULL,
  prop_kept = NULL,
  cutoff = NULL,
  maximize = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="var_to_keep_+3A_values">values</code></td>
<td>
<p>A numeric vector, with one value per variable to keep or
discard.</p>
</td></tr>
<tr><td><code id="var_to_keep_+3A_n_kept">n_kept</code></td>
<td>
<p>Number of variables to keep.</p>
</td></tr>
<tr><td><code id="var_to_keep_+3A_prop_kept">prop_kept</code></td>
<td>
<p>A numeric value between 0 and 1 representing the proportion
of variables to keep. <code>n_kept</code> and <code>prop_kept</code> are mutually exclusive.</p>
</td></tr>
<tr><td><code id="var_to_keep_+3A_cutoff">cutoff</code></td>
<td>
<p>Threshold beyond which (below or above) the variables are
discarded.</p>
</td></tr>
<tr><td><code id="var_to_keep_+3A_maximize">maximize</code></td>
<td>
<p>Whether to minimize (<code>FALSE</code>) or maximize (<code>TRUE</code>, the
default) the quantity given by <code>values</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector indicating if variables are kept or discarded.
</p>


<h3>Author(s)</h3>

<p>Antoine Bichat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>var_to_keep(1:5, n_kept = 3, maximize = TRUE)
var_to_keep(1:10, cutoff = 8, maximize = FALSE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
