<!DOCTYPE html><html><head><title>Help for package SCIntRuler</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SCIntRuler}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SCIntRuler-package'><p>SCIntRuler: Integration of Single-Cell RNA-seq Datasets</p></a></li>
<li><a href='#CalcuSCIR'><p>Calculate SCIntRuler</p></a></li>
<li><a href='#crossdist'><p>Cross-Distance Matrix Calculation</p></a></li>
<li><a href='#FindCell'><p>Find cells indicating shared biological features across conditions</p></a></li>
<li><a href='#FindNNDist'><p>Find the nearest neighbors</p></a></li>
<li><a href='#FindNNDistC'><p>Find the nearest neighbors</p></a></li>
<li><a href='#GetCluster'><p>Get broad and fine clusters</p></a></li>
<li><a href='#NormData'><p>Normalized RNA data matrix</p></a></li>
<li><a href='#PermTest'><p>Permutation Test</p></a></li>
<li><a href='#PlotSCIR'><p>Plot SCIntRuler</p></a></li>
<li><a href='#SCEtoSeurat'><p>Input and Split SingleCellExperiment Data</p></a></li>
<li><a href='#sim_data_sce'><p>My Example Dataset</p></a></li>
<li><a href='#sim_result'><p>My Example Dataset</p></a></li>
<li><a href='#SummCluster'><p>Get maximum number of broad clusters</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Guiding the Integration of Multiple Single-Cell RNA-Seq Datasets</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.6</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yue Lyu &lt;yuelyu0521@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The accumulation of single-cell RNA-seq ('scRNA-seq') studies highlights the potential benefits of integrating multiple datasets. By augmenting sample sizes and enhancing analytical robustness, integration can lead to more insightful biological conclusions. However, challenges arise due to the inherent diversity and batch discrepancies within and across studies. 'SCIntRuler', a novel R package, addresses these challenges by guiding the integration of multiple 'scRNA-seq' datasets.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, Matrix, batchelor, base, Seurat, SeuratObject,
MatrixGenerics, SingleCellExperiment, SummarizedExperiment,
dplyr, coin, harmony, ggplot2, gridExtra, cowplot, magrittr,
stats</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/yuelyu21/SCIntRuler">https://github.com/yuelyu21/SCIntRuler</a>,
<a href="https://yuelyu21.github.io/SCIntRuler/">https://yuelyu21.github.io/SCIntRuler/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/yuelyu21/SCIntRuler/issues">https://github.com/yuelyu21/SCIntRuler/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocStyle, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-11 06:55:10 UTC; a98455</td>
</tr>
<tr>
<td>Author:</td>
<td>Yue Lyu <a href="https://orcid.org/0000-0002-8912-6624"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-12 15:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='SCIntRuler-package'>SCIntRuler: Integration of Single-Cell RNA-seq Datasets</h2><span id='topic+SCIntRuler-package'></span><span id='topic+SCIntRuler'></span>

<h3>Description</h3>

<p>The SCIntRuler package addresses the challenges of integrating multiple single-cell RNA-seq (scRNA-seq) datasets. It provides tools to enhance analytical robustness by augmenting sample sizes and reducing batch discrepancies. Developed using the Seurat framework, SCIntRuler includes both existing and novel workflows for single-cell analysis.
</p>


<h3>Value</h3>

<p>This is the main page for SCIntRuler package.
</p>


<h3>Why SCIntRuler? Integrating scRNA-seq datasets can be complex due to various factors such as batch effects and sample diversity. SCIntRuler provides a statistical metric to aid in crucial decisions regarding dataset integration, ensuring more robust and accurate analyses.</h3>

<p>NA
</p>


<h3>Features</h3>


<ul>
<li><p>Informed Decision Making: Helps researchers decide on the necessity
of data integration and the most suitable method.
</p>
</li>
<li><p>Flexibility: Suitable for various scenarios, accommodating different
levels of data heterogeneity.
</p>
</li>
<li><p>Robustness: Enhances analytical robustness in joint analyses of merged
or integrated scRNA-seq datasets.
</p>
</li>
<li><p>User-Friendly: Streamlines decision-making processes, simplifying the
complexities involved in scRNA-seq data integration.
</p>
</li></ul>



<h3>Getting Started</h3>

<p>Refer to the &quot;Getting Started with SCIntRuler&quot; article in the package vignettes for detailed user instructions.
</p>


<h3>Author(s)</h3>

<p>Yue Lyu
</p>

<hr>
<h2 id='CalcuSCIR'>Calculate SCIntRuler</h2><span id='topic+CalcuSCIR'></span>

<h3>Description</h3>

<p>Calculate SCIntRuler
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalcuSCIR(fullcluster, seuratlist, testres, p = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CalcuSCIR_+3A_fullcluster">fullcluster</code></td>
<td>
<p>A list of clusters that generated by the function GetCluster()</p>
</td></tr>
<tr><td><code id="CalcuSCIR_+3A_seuratlist">seuratlist</code></td>
<td>
<p>A list of Seurat objects, usually can be got by SplitObject().</p>
</td></tr>
<tr><td><code id="CalcuSCIR_+3A_testres">testres</code></td>
<td>
<p>Result from function PermTest()</p>
</td></tr>
<tr><td><code id="CalcuSCIR_+3A_p">p</code></td>
<td>
<p>P-value that will be used as the cut-off, default value is 0.1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SCIntRuler
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(sim_result)
data(sim_data_sce)
sim_data &lt;- SCEtoSeurat(sim_data_sce)
seuratlist &lt;- Seurat::SplitObject(sim_data, split.by = "Study")
CalcuSCIR(sim_result[[1]], seuratlist, sim_result[[4]])

</code></pre>

<hr>
<h2 id='crossdist'>Cross-Distance Matrix Calculation</h2><span id='topic+crossdist'></span>

<h3>Description</h3>

<p>Computes the pairwise Euclidean distance between rows of two matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crossdist(m1, m2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="crossdist_+3A_m1">m1</code></td>
<td>
<p>Numeric matrix.</p>
</td></tr>
<tr><td><code id="crossdist_+3A_m2">m2</code></td>
<td>
<p>Numeric matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric matrix of distances.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mat1 &lt;- matrix(1:4, ncol = 2)
mat2 &lt;- matrix(5:8, ncol = 2)
dist_matrix &lt;- crossdist(mat1, mat2)
</code></pre>

<hr>
<h2 id='FindCell'>Find cells indicating shared biological features across conditions</h2><span id='topic+FindCell'></span>

<h3>Description</h3>

<p>Find cells indicating shared biological features across conditions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindCell(seuratobj, seuratlist, fullcluster, distmat, firstn = 15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindCell_+3A_seuratobj">seuratobj</code></td>
<td>
<p>The Seurat object that all samples/subjects were merged together.</p>
</td></tr>
<tr><td><code id="FindCell_+3A_seuratlist">seuratlist</code></td>
<td>
<p>A list of Seurat objects, usually can be got by SplitObject().</p>
</td></tr>
<tr><td><code id="FindCell_+3A_fullcluster">fullcluster</code></td>
<td>
<p>A list of clusters that generated by the function GetCluster().</p>
</td></tr>
<tr><td><code id="FindCell_+3A_distmat">distmat</code></td>
<td>
<p>A list of distance vectors generated by the function FindNNDist().</p>
</td></tr>
<tr><td><code id="FindCell_+3A_firstn">firstn</code></td>
<td>
<p>The number of nearest cells were detected that you want to include in the permutation test. Default to be 15.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two vectors: one is for which cluster of which sample will be highlighted and the second one is which cells will be selected.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(sim_data_sce)
data(sim_result)
sim_data &lt;- SCEtoSeurat(sim_data_sce)
seuratlist &lt;- Seurat::SplitObject(sim_data, split.by = "Study")
FindCell(sim_data, seuratlist, sim_result[[1]], sim_result[[3]], 15)

</code></pre>

<hr>
<h2 id='FindNNDist'>Find the nearest neighbors</h2><span id='topic+FindNNDist'></span>

<h3>Description</h3>

<p>Find the nearest neighbors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindNNDist(fullcluster, normCount, meaningn = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindNNDist_+3A_fullcluster">fullcluster</code></td>
<td>
<p>A list of clusters that generated by the function GetCluster().</p>
</td></tr>
<tr><td><code id="FindNNDist_+3A_normcount">normCount</code></td>
<td>
<p>A list of normalized gene count matrix generated by the function NormData().</p>
</td></tr>
<tr><td><code id="FindNNDist_+3A_meaningn">meaningn</code></td>
<td>
<p>default to be 20</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of distance vectors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim_result)
meaningn &lt;- 20
FindNNDist(sim_result[[1]], sim_result[[2]], meaningn = meaningn)
</code></pre>

<hr>
<h2 id='FindNNDistC'>Find the nearest neighbors</h2><span id='topic+FindNNDistC'></span>

<h3>Description</h3>

<p>Find the nearest neighbors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindNNDistC(fullcluster, normCount, meaningn = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindNNDistC_+3A_fullcluster">fullcluster</code></td>
<td>
<p>A list of clusters that generated by the function GetCluster().</p>
</td></tr>
<tr><td><code id="FindNNDistC_+3A_normcount">normCount</code></td>
<td>
<p>A list of normalized gene count matrix generated by the function NormData().</p>
</td></tr>
<tr><td><code id="FindNNDistC_+3A_meaningn">meaningn</code></td>
<td>
<p>default to be 20</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of distance vectors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim_result)
meaningn &lt;- 20
FindNNDistC(sim_result[[1]], sim_result[[2]], meaningn = meaningn)
</code></pre>

<hr>
<h2 id='GetCluster'>Get broad and fine clusters</h2><span id='topic+GetCluster'></span>

<h3>Description</h3>

<p>Get broad and fine clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetCluster(seuratlist, n1 = 50, n2 = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetCluster_+3A_seuratlist">seuratlist</code></td>
<td>
<p>A list of Seurat objects, usually can be got by SplitObject(). We also accept the SingleCellExperiment object input.</p>
</td></tr>
<tr><td><code id="GetCluster_+3A_n1">n1</code></td>
<td>
<p>If the number of cells was smaller than n1, then the cluster will remain unchanged called rare cluster. The default value of n1 is 50.</p>
</td></tr>
<tr><td><code id="GetCluster_+3A_n2">n2</code></td>
<td>
<p>If the count of cells within a broad cluster is more than n2, the cluster is subdivided randomly into three fine clusters. If the cell count falls within the range of n1 to n2, two fine clusters are generated randomly. Default value is 200.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data frames.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(sim_data_sce)
sim_data &lt;- SCEtoSeurat(sim_data_sce)
seuratlist &lt;- Seurat::SplitObject(sim_data, split.by = "Study")
fullcluster &lt;- GetCluster(seuratlist)

</code></pre>

<hr>
<h2 id='NormData'>Normalized RNA data matrix</h2><span id='topic+NormData'></span>

<h3>Description</h3>

<p>Normalized RNA data matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NormData(seuratlist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NormData_+3A_seuratlist">seuratlist</code></td>
<td>
<p>A list of Seurat objects, usually can be got by SplitObject().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim_data_sce)
sim_data &lt;- SCEtoSeurat(sim_data_sce)
seuratlist &lt;- Seurat::SplitObject(sim_data, split.by = "Study")
normCount &lt;- NormData(seuratlist)
</code></pre>

<hr>
<h2 id='PermTest'>Permutation Test</h2><span id='topic+PermTest'></span>

<h3>Description</h3>

<p>Permutation Test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PermTest(fullcluster, distmat, firstn)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PermTest_+3A_fullcluster">fullcluster</code></td>
<td>
<p>A list of clusters that generated by the function GetCluster()</p>
</td></tr>
<tr><td><code id="PermTest_+3A_distmat">distmat</code></td>
<td>
<p>A list of distance vectors generated by the function FindNNDist().</p>
</td></tr>
<tr><td><code id="PermTest_+3A_firstn">firstn</code></td>
<td>
<p>The number of nearest cells were detected that you want to include in the permutation test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two lists, one is the relative within-between distance and another is p-value of permutation test. Default to be 15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim_result)
testres &lt;- PermTest(sim_result[[1]], sim_result[[3]],15)
</code></pre>

<hr>
<h2 id='PlotSCIR'>Plot SCIntRuler</h2><span id='topic+PlotSCIR'></span>

<h3>Description</h3>

<p>Plot SCIntRuler
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotSCIR(fullcluster, seuratlist, testres, legendtitle = NULL, title = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotSCIR_+3A_fullcluster">fullcluster</code></td>
<td>
<p>A list of clusters that generated by the function GetCluster.</p>
</td></tr>
<tr><td><code id="PlotSCIR_+3A_seuratlist">seuratlist</code></td>
<td>
<p>A list of Seurat objects, usually can be got by SplitObject().</p>
</td></tr>
<tr><td><code id="PlotSCIR_+3A_testres">testres</code></td>
<td>
<p>Result from function PermTest()</p>
</td></tr>
<tr><td><code id="PlotSCIR_+3A_legendtitle">legendtitle</code></td>
<td>
<p>Title of legend, default to be NULL</p>
</td></tr>
<tr><td><code id="PlotSCIR_+3A_title">title</code></td>
<td>
<p>Title of figure, default to be NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(sim_data_sce)
data(sim_result)
sim_data &lt;- SCEtoSeurat(sim_data_sce)
seuratlist &lt;- Seurat::SplitObject(sim_data, split.by = "Study")
PlotSCIR(sim_result[[1]], seuratlist, sim_result[[4]])

</code></pre>

<hr>
<h2 id='SCEtoSeurat'>Input and Split SingleCellExperiment Data</h2><span id='topic+SCEtoSeurat'></span>

<h3>Description</h3>

<p>This function takes a SingleCellExperiment object and a variable by which to split it,
converts it to a Seurat object, and then splits it according to the specified variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCEtoSeurat(sce)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCEtoSeurat_+3A_sce">sce</code></td>
<td>
<p>A SingleCellExperiment object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim_data_sce)
# seuratlist &lt;- InputData(sim_data_sce,"Study")
seuratobj &lt;- SCEtoSeurat(sim_data_sce)
</code></pre>

<hr>
<h2 id='sim_data_sce'>My Example Dataset</h2><span id='topic+sim_data_sce'></span>

<h3>Description</h3>

<p>An example PBMC data with SingleCellExperiment format, including 3000 cells and 800 genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_data_sce
</code></pre>


<h3>Format</h3>

<p>An example PBMC data with SingleCellExperiment format
</p>

<dl>
<dt>int_elementMetadata</dt><dd><p>A DataFrame with 3000 rows and 1 column, storing simulated gene information.</p>
</dd>
<dt>int_colData</dt><dd><p>A DataFrame with 800 rows and 3 columns, representing metadata for each cell.</p>
</dd>
<dt>int_metadata</dt><dd><p>A list containing two elements that provide additional global metadata about the experiment.</p>
</dd>
<dt>rowRanges</dt><dd><p>A CompressedGRangesList object providing genomic range data associated with each row/gene.</p>
</dd>
<dt>colData</dt><dd><p>A DataFrame with 800 rows and 8 columns, detailing cell-level metadata.</p>
</dd>
<dt>assays</dt><dd><p>A SimpleAssay object with matrix dimensions 3000x800, representing the gene expression matrix.</p>
</dd>
<dt>elementMetadata</dt><dd><p>A DataFrame linked with assays, providing gene-level metadata.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The &quot;sim_data_sce&quot; object is designed to serve as a teaching and development aid for methods that require complex
single-cell expression data. It includes several typical features found in single-cell datasets, such as varied levels of
gene expression and metadata describing both cells and genes.
</p>
<p>The data within this object are entirely synthetic and should not be used for real analysis. The main use case is for
testing and development of single-cell analysis methodologies.
</p>


<h3>Value</h3>

<p>Simulation data to exemplify the usage of the method.
</p>


<h3>References</h3>

<p>The data were generated using a combination of random number generation for expression values and curated sources for
metadata to simulate realistic experimental scenarios.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("sim_data_sce")
</code></pre>

<hr>
<h2 id='sim_result'>My Example Dataset</h2><span id='topic+sim_result'></span>

<h3>Description</h3>

<p>An result example data with results from different functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_result
</code></pre>


<h3>Format</h3>

<p>An result example data
</p>

<dl>
<dt>fullcluster</dt><dd><p>A runable example of GetCluster, which is a list of clusters for each study.</p>
</dd>
<dt>normCount</dt><dd><p>A runable example of NormData, which is a list of normalized RNA expression matrixs for each study.</p>
</dd>
<dt>distmat</dt><dd><p>A runable example of FindNNDist, which is a list of distance matrixs for each study.</p>
</dd>
<dt>testres</dt><dd><p>A runable example of CalcuSCIR, which is a list of test results for each study.</p>
</dd>
</dl>



<h3>Value</h3>

<p>Simulation data to examplify the usage of the method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data
data("sim_result")
</code></pre>

<hr>
<h2 id='SummCluster'>Get maximum number of broad clusters</h2><span id='topic+SummCluster'></span>

<h3>Description</h3>

<p>Get maximum number of broad clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SummCluster(fullcluster)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SummCluster_+3A_fullcluster">fullcluster</code></td>
<td>
<p>A list of clusters that generated by the function GetCluster()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim_result)
SCout &lt;- SummCluster(sim_result[[1]])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
