<!DOCTYPE html><html><head><title>Help for package scINSIGHT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scINSIGHT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#create_scINSIGHT'><p>Create an scINSIGHT object.</p></a></li>
<li><a href='#run_scINSIGHT'><p>Perform scINSIGHT on normalized datasets</p></a></li>
<li><a href='#scINSIGHT-class'><p>The scINSIGHT Class</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interpretation of Heterogeneous Single-Cell Gene Expression Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-29</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kun Qian &lt;Kun_Qian@foxmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We develop a novel matrix factorization tool named 'scINSIGHT' to jointly analyze multiple single-cell gene expression samples from biologically heterogeneous sources, such as different disease phases, treatment groups, or developmental stages. Given multiple gene expression samples from different biological conditions, 'scINSIGHT' simultaneously identifies common and condition-specific gene modules and quantify their expression levels in each sample in a lower-dimensional space. With the factorized results, the inferred expression levels and memberships of common gene modules can be used to cluster cells and detect cell identities, and the condition-specific gene modules can help compare functional differences in transcriptomes from distinct conditions. Please also see Qian K, Fu SW, Li HW, Li WV (2022) &lt;<a href="https://doi.org/10.1186%2Fs13059-022-02649-3">doi:10.1186/s13059-022-02649-3</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, RANN, igraph, parallel, stats, stringr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Depends:</td>
<td>methods</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Vivianstats/scINSIGHT">https://github.com/Vivianstats/scINSIGHT</a>,
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-022-02649-3">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-022-02649-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-29 06:09:02 UTC; tea_flh01</td>
</tr>
<tr>
<td>Author:</td>
<td>Kun Qian <a href="https://orcid.org/0000-0002-2354-2238"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    ctb, cre],
  Wei Vivian Li <a href="https://orcid.org/0000-0002-2087-2709"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-29 22:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='create_scINSIGHT'>Create an scINSIGHT object.</h2><span id='topic+create_scINSIGHT'></span>

<h3>Description</h3>

<p>This function initializes an scINSIGHT object with normalized data passed in.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_scINSIGHT(norm.data, condition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_scINSIGHT_+3A_norm.data">norm.data</code></td>
<td>
<p>List of normalized expression matrices (genes by cells). Gene names should be the same in all matrices.</p>
</td></tr>
<tr><td><code id="create_scINSIGHT_+3A_condition">condition</code></td>
<td>
<p>Vector specifying sample conditions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>scINSIGHT</code> object with norm.data slot set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Demonstration using matrices with randomly generated numbers
S1 &lt;- matrix(runif(50000,0,2), 500,100)
S2 &lt;- matrix(runif(60000,0,2), 500,120)
S3 &lt;- matrix(runif(80000,0,2), 500,160)
S4 &lt;- matrix(runif(75000,0,2), 500,150)
data = list(S1, S2, S3, S4)
sample = c("sample1", "sample2", "sample3", "sample4")
condition = c("control", "activation", "control", "activation")
names(data) = sample
names(condition) = sample
scINSIGHTx &lt;- create_scINSIGHT(data, condition)
</code></pre>

<hr>
<h2 id='run_scINSIGHT'>Perform scINSIGHT on normalized datasets</h2><span id='topic+run_scINSIGHT'></span>

<h3>Description</h3>

<p>Perform INterpreting single cell gene expresSIon bioloGically Heterogeneous daTa (scINSIGHT) to return factorized <code class="reqn">W_{\ell1}</code>, <code class="reqn">W_{\ell2}</code>, <code class="reqn">H</code> and <code class="reqn">V</code> matrices.
</p>
<p>This factorization produces a <code class="reqn">W_{\ell1}</code> matrix (cells by <code class="reqn">K_j</code>), a <code class="reqn">W_{\ell2}</code> matrix (cells by <code class="reqn">K</code>), a shared <code class="reqn">V</code> matrix (<code class="reqn">K</code> by genes)
for each sample, and a <code class="reqn">H</code> (<code class="reqn">K_j</code> by genes) matrix for each condition. <code class="reqn">W_{\ell2}</code> are the expression matrices of <code class="reqn">K</code> common gene modules for all samples,
<code class="reqn">V</code> is the membership matrix of <code class="reqn">K</code> common gene modules, and it's shared by all samples.
<code class="reqn">W_{\ell1}</code> are the expression matrices of <code class="reqn">K_j</code> condition-specific gene modules for all samples,
and <code class="reqn">H</code> are the membership matrices of <code class="reqn">K_j</code> condition-specific gene modules for all conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_scINSIGHT(
  object,
  K = seq(5, 15, 2),
  K_j = 2,
  LDA = c(0.001, 0.01, 0.1, 1, 10),
  thre.niter = 500,
  thre.delta = 0.01,
  num.cores = 1,
  B = 5,
  out.dir = NULL,
  method = "increase"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_scINSIGHT_+3A_object">object</code></td>
<td>
<p><code>scINSIGHT</code> object.</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_k">K</code></td>
<td>
<p>Number of common gene modules. (default <code>c(5, 7, 9, 11, 13, 15)</code>)</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_k_j">K_j</code></td>
<td>
<p>Number of dataset-specific gene modules. (default 2)</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_lda">LDA</code></td>
<td>
<p>Regularization parameters. (default <code>c(0.001, 0.01, 0.1, 1, 10)</code>)</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_thre.niter">thre.niter</code></td>
<td>
<p>Maximum number of block coordinate descent iterations to perform. (default 500)</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_thre.delta">thre.delta</code></td>
<td>
<p>Stop iteration when the reduction of objective function is less than the threshold. (default 0.01)</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_num.cores">num.cores</code></td>
<td>
<p>Number of cores used for optimizing factorizations in parallel (default 1).</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_b">B</code></td>
<td>
<p>Number of repeats with random seed from 1 to B. (default 5)</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_out.dir">out.dir</code></td>
<td>
<p>Output directory of scINSIGHT results. (default NULL)</p>
</td></tr>
<tr><td><code id="run_scINSIGHT_+3A_method">method</code></td>
<td>
<p>Method of updating the factorization (default &quot;increase&quot;). If provide multiple <code class="reqn">K</code>, user can choose method between &quot;increase&quot; and &quot;decrease&quot;.
</p>
<p>For &quot;increase&quot;, the algorithm will first perform factorization with the least <code class="reqn">K=K_1</code>. Then initialize <code class="reqn">K_2-K_1</code> facotrs,
where <code class="reqn">K_2</code> is the <code class="reqn">K</code> sightly larger than <code class="reqn">K_1</code>, and perform facotrization with these new facotrs. Continue this process until the largest <code class="reqn">K</code>.
</p>
<p>For &quot;increase&quot;, the algorithm will first perform factorization with the largest <code class="reqn">K=K_1</code>. Then choose <code class="reqn">K_2</code> facotrs,
where <code class="reqn">K_2</code> is the <code class="reqn">K</code> sightly less than <code class="reqn">K_1</code>, and perform facotrization with these new facotrs. Continue this process until the least <code class="reqn">K</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>scINSIGHT</code> object with <code class="reqn">W_1</code>, <code class="reqn">W_2</code>, <code class="reqn">H</code>, <code class="reqn">V</code> and parameters slots set.
</p>

<hr>
<h2 id='scINSIGHT-class'>The scINSIGHT Class</h2><span id='topic+scINSIGHT-class'></span><span id='topic+scINSIGHT'></span>

<h3>Description</h3>

<p>The scINSIGHT object is created from two or more single cell datasets. To construct a
scINSIGHT object, the user needs to provide at least two normalized expression (or another
single-cell modality) matrices and the condition vector.
</p>


<h3>Details</h3>

<p>The key slots used in the scINSIGHT object are described below.
</p>


<h3>Slots</h3>


<dl>
<dt><code>norm.data</code></dt><dd><p>List of normalized expression matrices (genes by cells). Each matrix should have the same number and name of genes.</p>
</dd>
<dt><code>condition</code></dt><dd><p>Vector specifying each sample's condition name.</p>
</dd>
<dt><code>W_1</code></dt><dd><p>List of <code class="reqn">W_{\ell1}</code> estimated by scINSIGHT, names correspond to sample names.</p>
</dd>
<dt><code>W_2</code></dt><dd><p>List of <code class="reqn">W_{\ell2}</code> estimated by scINSIGHT, names correspond to sample names.</p>
</dd>
<dt><code>H</code></dt><dd><p>List of <code class="reqn">H</code> estimated by scINSIGHT, names correspond to condition names.</p>
</dd>
<dt><code>V</code></dt><dd><p>Matrix <code class="reqn">V</code> estimated by scINSIGHT.</p>
</dd>
<dt><code>norm.W_2</code></dt><dd><p>List of <code class="reqn">W_{\ell2}</code> after normalization. Recommended for downstream analysis.</p>
</dd>
<dt><code>clusters</code></dt><dd><p>List of cluster results.</p>
</dd>
<dt><code>parameters</code></dt><dd><p>List of selected parameters, including <code class="reqn">K</code> and <code class="reqn">\lambda</code>.</p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
