<!DOCTYPE html><html lang="en"><head><title>Help for package SCINA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SCINA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#plotheat.SCINA'><p>A function to plot SCINA results in a heatmap.</p></a></li>
<li><a href='#preprocess.signatures'><p>A function to convert signatures uploaded via .csv files to lists used by SCINA.</p></a></li>
<li><a href='#SCINA'><p>A semi-supervised cell type identification and assignment tool.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Semi-Supervised Category Identification and Assignment Tool</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-7-16</td>
</tr>
<tr>
<td>Author:</td>
<td>Ze Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ze Zhang &lt;Ze.Zhang@utsouthwestern.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.0), MASS, gplots</td>
</tr>
<tr>
<td>Description:</td>
<td>An automatic cell type detection and assignment algorithm for single cell RNA-Seq and Cytof/FACS data. 'SCINA' is capable of assigning cell type identities to a pool of cells profiled by scRNA-Seq or Cytof/FACS data with prior knowledge of markers, such as genes and protein symbols that are highly or lowly expressed in each category. See Zhang Z, et al (2019) &lt;<a href="https://doi.org/10.3390%2Fgenes10070531">doi:10.3390/genes10070531</a>&gt; for more details.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://lce.biohpc.swmed.edu/scina/">http://lce.biohpc.swmed.edu/scina/</a>
<a href="https://github.com/jcao89757/SCINA">https://github.com/jcao89757/SCINA</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-07-17 18:39:19 UTC; s421955</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-07-18 06:38:25 UTC</td>
</tr>
</table>
<hr>
<h2 id='plotheat.SCINA'>A function to plot SCINA results in a heatmap.</h2><span id='topic+plotheat.SCINA'></span>

<h3>Description</h3>

<p>A function to plot SCINA results in a heatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotheat.SCINA(exp, results, signatures)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotheat.SCINA_+3A_exp">exp</code></td>
<td>
<p>See more details in <code><a href="#topic+SCINA">SCINA</a></code></p>
</td></tr>
<tr><td><code id="plotheat.SCINA_+3A_results">results</code></td>
<td>
<p>An output object returned from SCINA.</p>
</td></tr>
<tr><td><code id="plotheat.SCINA_+3A_signatures">signatures</code></td>
<td>
<p>See more details in <code><a href="#topic+SCINA">SCINA</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot a heatmap showing signature genes' expression level and SCINA predicted cell types.
</p>

<hr>
<h2 id='preprocess.signatures'>A function to convert signatures uploaded via .csv files to lists used by SCINA.</h2><span id='topic+preprocess.signatures'></span>

<h3>Description</h3>

<p>A function to convert signatures uploaded via .csv files to lists used by SCINA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocess.signatures(file_path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="preprocess.signatures_+3A_file_path">file_path</code></td>
<td>
<p>The path of the .csv file. The first row of the file should be cell type names. Each column is occupied by the signature genes/protein markers for the cell type in the first row. Please find more details in <code><a href="#topic+SCINA">SCINA</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of signature gene lists as an input for SCINA.
</p>

<hr>
<h2 id='SCINA'>A semi-supervised cell type identification and assignment tool.</h2><span id='topic+SCINA'></span>

<h3>Description</h3>

<p>An automatic cell type detection and assignment algorithm for single cell RNA-Seq (scRNA-seq) and Cytof/FACS data. See Zhang Z, et al (2019) &lt;doi:10.3390/genes10070531&gt; for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCINA(exp, signatures, max_iter = 100, convergence_n = 10, convergence_rate = 0.99, 
    sensitivity_cutoff = 1, rm_overlap = 1, allow_unknown = 1, log_file = "SCINA.log")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SCINA_+3A_exp">exp</code></td>
<td>
<p>A normalized matrix representing the gene expression levels. The log-transformation is suggested to avoid heavy-tailed datasets. Columns correpond to cells, rows correspond to genes or protein symbols.</p>
</td></tr>
<tr><td><code id="SCINA_+3A_signatures">signatures</code></td>
<td>
<p>A list contains multiple signature vectors. Each signature vector contains genes or protein symbols, representing the prior knowledge for one cell type.</p>
</td></tr>
<tr><td><code id="SCINA_+3A_max_iter">max_iter</code></td>
<td>
<p>An integer &gt; 0. Default is 100. Max iterations allowed for the EM algorithm.</p>
</td></tr>
<tr><td><code id="SCINA_+3A_convergence_rate">convergence_rate</code></td>
<td>
<p>A float between 0 and 1. Default is 0.99. Percentage of cells for which the type assignment remains stable for the last n rounds.</p>
</td></tr>
<tr><td><code id="SCINA_+3A_sensitivity_cutoff">sensitivity_cutoff</code></td>
<td>
<p>A float between 0 and 1. Default is 1. The cutoff to remove signatures whose cells types are deemed as non-existent at all in the data by the SCINA algorithm.</p>
</td></tr>
<tr><td><code id="SCINA_+3A_rm_overlap">rm_overlap</code></td>
<td>
<p>A binary value, default 1 (TRUE), denotes that shared symbols between signature lists will be removed. If 0 (FALSE) then allows different cell types to share the same identifiers.</p>
</td></tr>
<tr><td><code id="SCINA_+3A_allow_unknown">allow_unknown</code></td>
<td>
<p>A binary value, default 1 (TRUE). If 0 (FALSE) then no cell will be assigned to the 'unknown' category.</p>
</td></tr>
<tr><td><code id="SCINA_+3A_convergence_n">convergence_n</code></td>
<td>
<p>An integer &gt; 0. Default is 10. Stop the SCINA algorithm if during the last n rounds of iterations, cell type assignment keeps steady above the convergence_rate.</p>
</td></tr>
<tr><td><code id="SCINA_+3A_log_file">log_file</code></td>
<td>
<p>A string names the record of the running status of the SCINA algorithem, default 'SCINA.log'.</p>
</td></tr>
</table>


<h3>Details</h3>

 
<p>More detailed information can be found from our web server: <a href="http://lce.biohpc.swmed.edu/scina/.">http://lce.biohpc.swmed.edu/scina/.</a>
</p>
<p>For any symbols in signature list, if the cell type is identified with symbol X's 
low detection level, please specify the symbol as 'low_X'. The name for the list is the cell type.
</p>
<p>Details for 'low_X' (take scRNA-Seqs as an example):<br />
(a) There are 4 cell types, the first one highly express one gene A, and the other three lowly express the same gene. 
Then it is better to specify A as the high marker for cell type 1, but it is not a good idea to specify A as the low 
expression marker for cell type 2,3,4.<br />
(b) There are 4 cell types, the first one lowly express one gene A, and the other three highly express the same gene. 
Then is it better to specify A as the low marker for cell type 1, but it is not a good idea to specify A as the 
high expression marker for cell type 2,3,4.<br />
(c) There are 4 cell types, the first one lowly express one gene A, the second and third one moderately express gene A, 
and the last one highly express gene A. Then is it better to specify A as the low marker for cell type 1, and as the high 
expression marker for cell type 4.<br />
(d) The same specification can be applied to protein markers in CyTOF anlysis.
</p>
<p>Small sensitivity_cutoff leads to more signatures to be removed, and 1 denotes that no signature is removed.
</p>


<h3>Value</h3>

<p>cell_labels return a vector contains cell type mapping results for each cell.
</p>
<p>probabilities return a probability matrix indicating the predicted probability for each cell belongs to each cell type respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file('extdata','example_expmat.RData', package = "SCINA"))
load(system.file('extdata','example_signatures.RData', package = "SCINA"))
exp = exp_test$exp_data
results = SCINA(exp, signatures, max_iter = 120, convergence_n = 12, 
    convergence_rate = 0.999, sensitivity_cutoff = 0.9)
table(exp_test$true_label, results$cell_labels)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
