<!DOCTYPE html><html><head><title>Help for package scistreer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scistreer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annotate_tree'><p>Find maximum lilkelihood assignment of mutations on a tree</p></a></li>
<li><a href='#get_mut_graph'><p>Convert a single-cell phylogeny with mutation placements into a mutation graph</p></a></li>
<li><a href='#gtree_small'><p>Smaller example annotated tree built from P_small</p></a></li>
<li><a href='#label_edges'><p>Annotate the direct upstream or downstream mutations on the edges</p></a></li>
<li><a href='#ladderize'><p>From ape; will remove once new ape version is released</p>
https://github.com/emmanuelparadis/ape/issues/54</a></li>
<li><a href='#mut_nodes_small'><p>Mutation placements calculated from tree_small and P_small</p></a></li>
<li><a href='#mut_to_tree'><p>Transfer mutation assignment onto a single-cell phylogeny</p></a></li>
<li><a href='#P_example'><p>Example genotype probability matrix</p></a></li>
<li><a href='#P_small'><p>Smaller example genotype probability matrix</p></a></li>
<li><a href='#perform_nni'><p>Maximum likelihood tree search via NNI</p></a></li>
<li><a href='#plot_phylo_heatmap'><p>Plot phylogeny and mutation heatmap</p></a></li>
<li><a href='#run_scistree'><p>Run the scistree workflow</p></a></li>
<li><a href='#score_tree'><p>Score a tree based on maximum likelihood</p></a></li>
<li><a href='#to_phylo'><p>Convert the phylogeny from tidygraph to phylo object</p>
modified from R package alakazam, converts a tbl_graph to a phylo object</a></li>
<li><a href='#transfer_links'><p>Annotate the direct upstream or downstream node on the edges</p></a></li>
<li><a href='#tree_small'><p>Smaller example tree built from P_small</p></a></li>
<li><a href='#tree_upgma'><p>Example tree built using UPGMA from P_small</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Maximum-Likelihood Perfect Phylogeny Inference at Scale</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kharchenkolab/scistreer">https://github.com/kharchenkolab/scistreer</a>,
<a href="https://kharchenkolab.github.io/scistreer/">https://kharchenkolab.github.io/scistreer/</a></td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Fast maximum-likelihood phylogeny inference from noisy single-cell data using the 'ScisTree' algorithm by Yufeng Wu (2019) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtz676">doi:10.1093/bioinformatics/btz676</a>&gt;. 'scistreer' provides an 'R' interface and improves speed via 'Rcpp' and 'RcppParallel', making the method applicable to massive single-cell datasets (&gt;10,000 cells).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, dplyr, ggplot2, ggtree, igraph, parallelDist, patchwork,
phangorn, Rcpp, reshape2, RcppParallel, RhpcBLASctl, stringr,
tidygraph</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppParallel</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>Author:</td>
<td>Teng Gao [cre, aut],
  Evan Biederstedt [aut],
  Peter Kharchenko [aut],
  Yufeng Wu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Teng Gao &lt;tgaoteng@gmail.com&gt;</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-15 19:49:42 UTC; teng</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-15 20:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='annotate_tree'>Find maximum lilkelihood assignment of mutations on a tree</h2><span id='topic+annotate_tree'></span>

<h3>Description</h3>

<p>Find maximum lilkelihood assignment of mutations on a tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotate_tree(tree, P)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotate_tree_+3A_tree">tree</code></td>
<td>
<p>phylo Single-cell phylogenetic tree</p>
</td></tr>
<tr><td><code id="annotate_tree_+3A_p">P</code></td>
<td>
<p>matrix Genotype probability matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl_graph A single-cell phylogeny with mutation placements
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gtree_small = annotate_tree(tree_small, P_small)
</code></pre>

<hr>
<h2 id='get_mut_graph'>Convert a single-cell phylogeny with mutation placements into a mutation graph</h2><span id='topic+get_mut_graph'></span>

<h3>Description</h3>

<p>Convert a single-cell phylogeny with mutation placements into a mutation graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mut_graph(gtree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_mut_graph_+3A_gtree">gtree</code></td>
<td>
<p>tbl_graph The single-cell phylogeny</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph Mutation graph
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mut_graph = get_mut_graph(gtree_small)
</code></pre>

<hr>
<h2 id='gtree_small'>Smaller example annotated tree built from P_small</h2><span id='topic+gtree_small'></span>

<h3>Description</h3>

<p>Smaller example annotated tree built from P_small
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtree_small
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_graph</code> (inherits from <code>igraph</code>) of length 199.
</p>

<hr>
<h2 id='label_edges'>Annotate the direct upstream or downstream mutations on the edges</h2><span id='topic+label_edges'></span>

<h3>Description</h3>

<p>Annotate the direct upstream or downstream mutations on the edges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>label_edges(G)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="label_edges_+3A_g">G</code></td>
<td>
<p>igraph Mutation graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph Mutation graph
</p>

<hr>
<h2 id='ladderize'>From ape; will remove once new ape version is released
https://github.com/emmanuelparadis/ape/issues/54</h2><span id='topic+ladderize'></span>

<h3>Description</h3>

<p>From ape; will remove once new ape version is released
https://github.com/emmanuelparadis/ape/issues/54
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ladderize(phy, right = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ladderize_+3A_phy">phy</code></td>
<td>
<p>phylo The phylogeny</p>
</td></tr>
<tr><td><code id="ladderize_+3A_right">right</code></td>
<td>
<p>logical Whether ladderize to the right</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>tree_small = ladderize(tree_small)
</code></pre>

<hr>
<h2 id='mut_nodes_small'>Mutation placements calculated from tree_small and P_small</h2><span id='topic+mut_nodes_small'></span>

<h3>Description</h3>

<p>Mutation placements calculated from tree_small and P_small
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_nodes_small
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 9 rows and 2 columns.
</p>

<hr>
<h2 id='mut_to_tree'>Transfer mutation assignment onto a single-cell phylogeny</h2><span id='topic+mut_to_tree'></span>

<h3>Description</h3>

<p>Transfer mutation assignment onto a single-cell phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_to_tree(gtree, mut_nodes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mut_to_tree_+3A_gtree">gtree</code></td>
<td>
<p>tbl_graph The single-cell phylogeny</p>
</td></tr>
<tr><td><code id="mut_to_tree_+3A_mut_nodes">mut_nodes</code></td>
<td>
<p>dataframe Mutation placements</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl_graph A single-cell phylogeny with mutation placements
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gtree_small = mut_to_tree(gtree_small, mut_nodes_small)
</code></pre>

<hr>
<h2 id='P_example'>Example genotype probability matrix</h2><span id='topic+P_example'></span>

<h3>Description</h3>

<p>Example genotype probability matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>P_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 25 columns.
</p>

<hr>
<h2 id='P_small'>Smaller example genotype probability matrix</h2><span id='topic+P_small'></span>

<h3>Description</h3>

<p>Smaller example genotype probability matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>P_small
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 100 rows and 25 columns.
</p>

<hr>
<h2 id='perform_nni'>Maximum likelihood tree search via NNI</h2><span id='topic+perform_nni'></span>

<h3>Description</h3>

<p>Maximum likelihood tree search via NNI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_nni(
  tree_init,
  P,
  max_iter = 100,
  eps = 0.01,
  ncores = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perform_nni_+3A_tree_init">tree_init</code></td>
<td>
<p>phylo Intial tree</p>
</td></tr>
<tr><td><code id="perform_nni_+3A_p">P</code></td>
<td>
<p>matrix Genotype probability matrix</p>
</td></tr>
<tr><td><code id="perform_nni_+3A_max_iter">max_iter</code></td>
<td>
<p>integer Maximum number of iterations</p>
</td></tr>
<tr><td><code id="perform_nni_+3A_eps">eps</code></td>
<td>
<p>numeric Tolerance threshold in likelihood difference for stopping</p>
</td></tr>
<tr><td><code id="perform_nni_+3A_ncores">ncores</code></td>
<td>
<p>integer Number of cores to use</p>
</td></tr>
<tr><td><code id="perform_nni_+3A_verbose">verbose</code></td>
<td>
<p>logical Verbosity</p>
</td></tr>
</table>


<h3>Value</h3>

<p>multiPhylo List of trees corresponding to the rearrangement steps
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree_list = perform_nni(tree_upgma, P_small)
</code></pre>

<hr>
<h2 id='plot_phylo_heatmap'>Plot phylogeny and mutation heatmap</h2><span id='topic+plot_phylo_heatmap'></span>

<h3>Description</h3>

<p>Plot phylogeny and mutation heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_phylo_heatmap(tree, P, branch_width = 0.5, root_edge = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_phylo_heatmap_+3A_tree">tree</code></td>
<td>
<p>phylo or tbl_graph Phylogeny</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_p">P</code></td>
<td>
<p>matrix Genotype probability matrix</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_branch_width">branch_width</code></td>
<td>
<p>numeric Branch width</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_root_edge">root_edge</code></td>
<td>
<p>logical Whether to plot root edge</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot Plot visualizing the single-cell phylogeny and mutation probability heatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_phylo_heatmap(tree_small, P_small)
</code></pre>

<hr>
<h2 id='run_scistree'>Run the scistree workflow</h2><span id='topic+run_scistree'></span>

<h3>Description</h3>

<p>Run the scistree workflow
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_scistree(
  P,
  init = "UPGMA",
  ncores = 1,
  max_iter = 100,
  eps = 0.01,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_scistree_+3A_p">P</code></td>
<td>
<p>matrix Genotype probability matrix (cell x mutation). Each entry is a probability (0-1) that the cell harbors the mutation</p>
</td></tr>
<tr><td><code id="run_scistree_+3A_init">init</code></td>
<td>
<p>character Initialization strategy; UPGMA or NJ</p>
</td></tr>
<tr><td><code id="run_scistree_+3A_ncores">ncores</code></td>
<td>
<p>integer Number of cores to use</p>
</td></tr>
<tr><td><code id="run_scistree_+3A_max_iter">max_iter</code></td>
<td>
<p>integer Maximum number of iterations</p>
</td></tr>
<tr><td><code id="run_scistree_+3A_eps">eps</code></td>
<td>
<p>numeric Tolerance threshold in likelihood difference for stopping</p>
</td></tr>
<tr><td><code id="run_scistree_+3A_verbose">verbose</code></td>
<td>
<p>logical Verbosity</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phylo A maximum-likelihood phylogeny
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree_small = run_scistree(P_small)
</code></pre>

<hr>
<h2 id='score_tree'>Score a tree based on maximum likelihood</h2><span id='topic+score_tree'></span>

<h3>Description</h3>

<p>Score a tree based on maximum likelihood
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_tree(tree, P, get_l_matrix = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="score_tree_+3A_tree">tree</code></td>
<td>
<p>phylo object</p>
</td></tr>
<tr><td><code id="score_tree_+3A_p">P</code></td>
<td>
<p>genotype probability matrix</p>
</td></tr>
<tr><td><code id="score_tree_+3A_get_l_matrix">get_l_matrix</code></td>
<td>
<p>whether to compute the whole likelihood matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list Likelihood scores of a tree
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree_likelihood = score_tree(tree_upgma, P_small)$l_tree
</code></pre>

<hr>
<h2 id='to_phylo'>Convert the phylogeny from tidygraph to phylo object
modified from R package alakazam, converts a tbl_graph to a phylo object</h2><span id='topic+to_phylo'></span>

<h3>Description</h3>

<p>Convert the phylogeny from tidygraph to phylo object
modified from R package alakazam, converts a tbl_graph to a phylo object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to_phylo(graph)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="to_phylo_+3A_graph">graph</code></td>
<td>
<p>tbl_graph The single-cell phylogeny</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phylo The single-cell phylogeny
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree_small = to_phylo(annotate_tree(tree_small, P_small))
</code></pre>

<hr>
<h2 id='transfer_links'>Annotate the direct upstream or downstream node on the edges</h2><span id='topic+transfer_links'></span>

<h3>Description</h3>

<p>Annotate the direct upstream or downstream node on the edges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transfer_links(G)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transfer_links_+3A_g">G</code></td>
<td>
<p>igraph Mutation graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph Mutation graph
</p>

<hr>
<h2 id='tree_small'>Smaller example tree built from P_small</h2><span id='topic+tree_small'></span>

<h3>Description</h3>

<p>Smaller example tree built from P_small
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_small
</code></pre>


<h3>Format</h3>

<p>An object of class <code>phylo</code> of length 5.
</p>

<hr>
<h2 id='tree_upgma'>Example tree built using UPGMA from P_small</h2><span id='topic+tree_upgma'></span>

<h3>Description</h3>

<p>Example tree built using UPGMA from P_small
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_upgma
</code></pre>


<h3>Format</h3>

<p>An object of class <code>phylo</code> of length 4.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
