<!DOCTYPE html><html><head><title>Help for package scISR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scISR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Goolam'><p>Goolam</p></a></li>
<li><a href='#scISR'><p>scISR: Single-cell Imputation using Subspace Regression</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single-Cell Imputation using Subspace Regression</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Duc Tran &lt;duct@nevada.unr.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides an imputation pipeline for single-cell RNA sequencing data. 
  The 'scISR' method uses a hypothesis-testing technique to identify zero-valued entries that are most likely affected by dropout events and estimates the dropout values using a subspace regression model (Tran et.al. (2022) &lt;<a href="https://doi.org/10.1038%2Fs41598-022-06500-4">doi:10.1038/s41598-022-06500-4</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2">LGPL-2</a> | <a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cluster, entropy, stats, utils, parallel, irlba, PINSPlus,
matrixStats, markdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, mclust</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/duct317/scISR">https://github.com/duct317/scISR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/duct317/scISR/issues">https://github.com/duct317/scISR/issues</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-28 19:32:08 UTC; dtran</td>
</tr>
<tr>
<td>Author:</td>
<td>Duc Tran [aut, cre],
  Bang Tran [aut],
  Hung Nguyen [aut],
  Tin Nguyen [fnd]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-30 06:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='Goolam'>Goolam</h2><span id='topic+Goolam'></span>

<h3>Description</h3>

<p>Goolam dataset with data and cell types information.The number of genes is reduced to 10,000.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Goolam
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='scISR'>scISR: Single-cell Imputation using Subspace Regression</h2><span id='topic+scISR'></span>

<h3>Description</h3>

<p>Perform single-cell Imputation using Subspace Regression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scISR(
  data,
  ncores = 1,
  force_impute = FALSE,
  do_fast = TRUE,
  preprocessing = TRUE,
  batch_impute = FALSE,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scISR_+3A_data">data</code></td>
<td>
<p>Input matrix or data frame. Rows represent genes while columns represent samples</p>
</td></tr>
<tr><td><code id="scISR_+3A_ncores">ncores</code></td>
<td>
<p>Number of cores that the algorithm should use. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="scISR_+3A_force_impute">force_impute</code></td>
<td>
<p>Always perform imputation.</p>
</td></tr>
<tr><td><code id="scISR_+3A_do_fast">do_fast</code></td>
<td>
<p>Use fast imputation implementation.</p>
</td></tr>
<tr><td><code id="scISR_+3A_preprocessing">preprocessing</code></td>
<td>
<p>Perform preprocessing on original data to filter out low quality features.</p>
</td></tr>
<tr><td><code id="scISR_+3A_batch_impute">batch_impute</code></td>
<td>
<p>Perform imputation in batches to reduce memory consumption.</p>
</td></tr>
<tr><td><code id="scISR_+3A_seed">seed</code></td>
<td>
<p>Seed for reproducibility. Default value is <code>1</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>scISR performs imputation for single-cell sequencing data. scISR identifies the true dropout values in the scRNA-seq dataset using
hyper-geomtric testing approach. Based on the result obtained from hyper-geometric testing, the original dataset is segregated into two
subsets including training data and imputable data. Next, training data is used for constructing a generalize linear regression model that
is used for imputation on the imputable data.
</p>


<h3>Value</h3>

<p><code>scISR</code> returns an imputed single-cell expression matrix where rows represent genes while columns represent samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# Load the package
library(scISR)
# Load Goolam dataset
data('Goolam');
# Use only 500 random genes for example
set.seed(1)
raw &lt;- Goolam$data[sample(seq_len(nrow(Goolam$data)), 500), ]
label &lt;- Goolam$label

# Perform the imputation
imputed &lt;- scISR(data = raw)

if(requireNamespace('mclust'))
{
  library(mclust)
  # Perform PCA and k-means clustering on raw data
  set.seed(1)
  # Filter genes that have only zeros from raw data
  raw_filer &lt;- raw[rowSums(raw != 0) &gt; 0, ]
  pca_raw &lt;- irlba::prcomp_irlba(t(raw_filer), n = 50)$x
  cluster_raw &lt;- kmeans(pca_raw, length(unique(label)),
                        nstart = 2000, iter.max = 2000)$cluster
  print(paste('ARI of clusters using raw data:',
              round(adjustedRandIndex(cluster_raw, label),3)))

  # Perform PCA and k-means clustering on imputed data
  set.seed(1)
  pca_imputed &lt;- irlba::prcomp_irlba(t(imputed), n = 50)$x
  cluster_imputed &lt;- kmeans(pca_imputed, length(unique(label)),
                            nstart = 2000, iter.max = 2000)$cluster
  print(paste('ARI of clusters using imputed data:',
              round(adjustedRandIndex(cluster_imputed, label),3)))
}
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
