<!DOCTYPE html><html><head><title>Help for package scLink</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scLink}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#sclink_cor'><p>Calculate scLink's correlation matrix</p></a></li>
<li><a href='#sclink_net'><p>Infer gene co-expression networks</p></a></li>
<li><a href='#sclink_norm'><p>Pre-process data for scLink</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Inferring Functional Gene Co-Expression Networks from Single
Cell Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Uses statistical network modeling to understand the co-expression relationships among genes and to construct sparse gene co-expression networks from single-cell gene expression data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), parallel, glasso</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-15 00:29:18 UTC; wei</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Vivian Li <a href="https://orcid.org/0000-0002-2087-2709"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Vivian Li &lt;vivian.li@rutgers.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-08-26 14:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='sclink_cor'>Calculate scLink's correlation matrix</h2><span id='topic+sclink_cor'></span>

<h3>Description</h3>

<p>Calculate scLink's correlation matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sclink_cor(expr, ncores, nthre = 20, dthre = 0.9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sclink_cor_+3A_expr">expr</code></td>
<td>
<p>A gene expression matrix with rows representing cells and columns representing genes.
Gene names are given as column names. Can be the output of <code>sclink_norm</code> or user constructed gene
expression matrices.</p>
</td></tr>
<tr><td><code id="sclink_cor_+3A_ncores">ncores</code></td>
<td>
<p>Number of cores if using parallel computation.</p>
</td></tr>
<tr><td><code id="sclink_cor_+3A_nthre">nthre</code></td>
<td>
<p>An integer specifying a threshold on the number of complete observations.
Defaults to 20.</p>
</td></tr>
<tr><td><code id="sclink_cor_+3A_dthre">dthre</code></td>
<td>
<p>A number specifying the threshold on dropout probabilities. Defaults to 0.9.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A correlation matrix for gene co-expression relationships.
</p>


<h3>Author(s)</h3>

<p>Wei Vivian Li, <a href="mailto:vivian.li@rutgers.edu">vivian.li@rutgers.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
count = readRDS(system.file("extdata", "example.rds", package = "scLink"))
count.norm = sclink_norm(count, scale.factor = 1e6, filter.genes = TRUE, n = 500)
corr = sclink_cor(expr = count.norm, ncores = 1)

</code></pre>

<hr>
<h2 id='sclink_net'>Infer gene co-expression networks</h2><span id='topic+sclink_net'></span>

<h3>Description</h3>

<p>Infer gene co-expression networks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sclink_net(expr, ncores, lda = seq(1, 0.1, -0.05), nthre = 20, dthre = 0.9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sclink_net_+3A_expr">expr</code></td>
<td>
<p>A gene expression matrix with rows representing cells and columns representing genes.
Gene names are given as column names. Can be the output of <code>sclink_norm</code> or user constructed gene
expression matrices.</p>
</td></tr>
<tr><td><code id="sclink_net_+3A_ncores">ncores</code></td>
<td>
<p>Number of cores if using parallel computation.</p>
</td></tr>
<tr><td><code id="sclink_net_+3A_lda">lda</code></td>
<td>
<p>A vector specifying a sequence of lambda values to be used in the penalized likelihood.</p>
</td></tr>
<tr><td><code id="sclink_net_+3A_nthre">nthre</code></td>
<td>
<p>An integer specifying a threshold on the number of complete observations.
Defaults to 20.</p>
</td></tr>
<tr><td><code id="sclink_net_+3A_dthre">dthre</code></td>
<td>
<p>A number specifying the threshold on dropout probabilities. Defaults to 0.9.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list for gene co-expression relationships. The list contains a <code>cor</code> element for
scLink's correlation matrix and a <code>summary</code> element for the gene networks. <code>summary</code> is a list
with each element corresponding to the result of one lambda value. Each element of <code>summary</code>
contains the following information:
</p>

<dl>
<dt>adj:</dt><dd><p>the adjacency matrix specifying the gene-gene edges;</p>
</dd>
<dt>Sigma:</dt><dd><p>the estimated concentration matrix;</p>
</dd>
<dt>nedge:</dt><dd><p>number of edges in the gene network;</p>
</dd>
<dt>bic:</dt><dd><p>BIC score;</p>
</dd>
<dt>lambda:</dt><dd><p>value of lambda in the penalty.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Wei Vivian Li, <a href="mailto:vivian.li@rutgers.edu">vivian.li@rutgers.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
count = readRDS(system.file("extdata", "example.rds", package = "scLink"))
count.norm = sclink_norm(count, scale.factor = 1e6, filter.genes = TRUE, n = 500)
networks = sclink_net(expr = count.norm, ncores = 1, lda = seq(0.5, 0.1, -0.05))

</code></pre>

<hr>
<h2 id='sclink_norm'>Pre-process data for scLink</h2><span id='topic+sclink_norm'></span>

<h3>Description</h3>

<p>Pre-process data for scLink
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sclink_norm(
  count,
  scale.factor = 1e+06,
  filter.genes = FALSE,
  gene.names = NULL,
  n = 500
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sclink_norm_+3A_count">count</code></td>
<td>
<p>A full gene count matrix with rows representing cells and columns representing genes.
Gene names are given as column names.</p>
</td></tr>
<tr><td><code id="sclink_norm_+3A_scale.factor">scale.factor</code></td>
<td>
<p>A number specifying the sclae factor used for library size normalization.
Defaults to 1e6.</p>
</td></tr>
<tr><td><code id="sclink_norm_+3A_filter.genes">filter.genes</code></td>
<td>
<p>A Boolean specifying whether scLink should select genes based on mean expression.
When set to <code>FALSE</code>, users need to speicfy a set of genes to be used for network construction with
<code>gene.names</code>. When set to <code>TRUE</code>, scLink will select genes based on their mean expression,
and users need to specify the number of genes to be selected with <code>n</code>.</p>
</td></tr>
<tr><td><code id="sclink_norm_+3A_gene.names">gene.names</code></td>
<td>
<p>A character vector specifying the genes used for network construction.
Only needed when <code>filter.genes = FALSE</code>.</p>
</td></tr>
<tr><td><code id="sclink_norm_+3A_n">n</code></td>
<td>
<p>An integer specifying the number of genes to be selected by scLink (defaults to 500).
Only needed when <code>filter.genes = TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A transformed and normalized gene expression matrix that can be used for correlation calculation
and network construction.
</p>


<h3>Author(s)</h3>

<p>Wei Vivian Li, <a href="mailto:vivian.li@rutgers.edu">vivian.li@rutgers.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
count = readRDS(system.file("extdata", "example.rds", package = "scLink"))
count.norm = sclink_norm(count, scale.factor = 1e6, filter.genes = TRUE, n = 500)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
