<!DOCTYPE html><html><head><title>Help for package scMappR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scMappR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cellmarker_enrich'><p>Fisher's Exact Cell-Type Identification.</p></a></li>
<li><a href='#coEnrich'><p>Identify co-expressed cell-types</p></a></li>
<li><a href='#compare_deconvolution_methods'><p>compare_deconvolution_methods</p></a></li>
<li><a href='#cwFoldChange_evaluate'><p>Measure cell-type specificity of cell-weighted Fold-changes</p></a></li>
<li><a href='#DeconRNAseq_CRAN'><p>DeconRNASeq CRAN compatible</p></a></li>
<li><a href='#deconvolute_and_contextualize'><p>Generate cell weighted Fold-Changes (cwFold-changes)</p></a></li>
<li><a href='#extract_genes_cell'><p>Extract Markers</p></a></li>
<li><a href='#generes_to_heatmap'><p>Generate signature matrix</p></a></li>
<li><a href='#get_gene_symbol'><p>Internal &ndash; get gene symbol from Panglao.db assigned gene-names (symbol-ensembl).</p></a></li>
<li><a href='#get_signature_matrices'><p>Get signature matrices.</p></a></li>
<li><a href='#gmt'><p>gmt_example</p></a></li>
<li><a href='#gProfiler_cellWeighted_Foldchange'><p>Pathway enrichment for cwFold-changes</p></a></li>
<li><a href='#gsva_cellIdentify'><p>Cell-type naming with GSVA</p></a></li>
<li><a href='#heatmap_generation'><p>Generate Heatmap</p></a></li>
<li><a href='#human_mouse_ct_marker_enrich'><p>Consensus cell-type naming (Fisher's Exact)</p></a></li>
<li><a href='#make_TF_barplot'><p>Plot g:profileR Barplot (TF)</p></a></li>
<li><a href='#pathway_enrich_internal'><p>Internal - Pathway enrichment for cellWeighted_Foldchanges and bulk gene list</p></a></li>
<li><a href='#PBMC_example'><p>PBMC_scMappR</p></a></li>
<li><a href='#plotBP'><p>Plot gProfileR Barplot</p></a></li>
<li><a href='#POA_example'><p>Preoptic_Area</p></a></li>
<li><a href='#process_dgTMatrix_lists'><p>Count Matrix To Signature Matrix</p></a></li>
<li><a href='#process_from_count'><p>Count Matrix To Seurat Object</p></a></li>
<li><a href='#scMappR_and_pathway_analysis'><p>Generate cellWeighted_Foldchanges, visualize, and enrich.</p></a></li>
<li><a href='#scMappR_tissues'><p>scMappR_tissues</p></a></li>
<li><a href='#seurat_to_generes'><p>Identify all cell-type markers</p></a></li>
<li><a href='#single_gene_preferences'><p>Single cell-type gene preferences</p></a></li>
<li><a href='#sm'><p>single_cell_process</p></a></li>
<li><a href='#tissue_by_celltype_enrichment'><p>tissue_by_celltype_enrichment</p></a></li>
<li><a href='#tissue_scMappR_custom'><p>Gene List Visualization and Enrichment with Custom Signature Matrix</p></a></li>
<li><a href='#tissue_scMappR_internal'><p>Gene List Visualization and Enrichment (Internal)</p></a></li>
<li><a href='#tochr'><p>To Character.</p></a></li>
<li><a href='#toNum'><p>To Numeric.</p></a></li>
<li><a href='#topgenes_extract'><p>Extract Top Markers</p></a></li>
<li><a href='#two_method_pathway_enrichment'><p>two_method_pathway_enrichment</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Single Cell Mapper</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.11</td>
</tr>
<tr>
<td>Description:</td>
<td>The single cell mapper (scMappR) R package contains a suite of bioinformatic tools that provide experimentally relevant cell-type specific information to a list of differentially expressed genes (DEG). The function "scMappR_and_pathway_analysis" reranks DEGs to generate cell-type specificity scores called cell-weighted fold-changes. Users input a list of DEGs, normalized counts, and a signature matrix into this function. scMappR then re-weights bulk DEGs by cell-type specific expression from the signature matrix, cell-type proportions from RNA-seq deconvolution and the ratio of cell-type proportions between the two conditions to account for changes in cell-type proportion. With cwFold-changes calculated, scMappR uses two approaches to utilize cwFold-changes to complete cell-type specific pathway analysis. The "process_dgTMatrix_lists" function in the scMappR package contains an automated scRNA-seq processing pipeline where users input scRNA-seq count data, which is made compatible for scMappR and other R packages that analyze scRNA-seq data. We further used this to store hundreds up regularly updating signature matrices. The functions "tissue_by_celltype_enrichment", "tissue_scMappR_internal", and "tissue_scMappR_custom" combine these consistently processed scRNAseq count data with gene-set enrichment tools to allow for cell-type marker enrichment of a generic gene list (e.g. GWAS hits). Reference: Sokolowski,D.J., Faykoo-Martinez,M., Erdman,L., Hou,H., Chan,C., Zhu,H., Holmes,M.M., Goldenberg,A. and Wilson,M.D. (2021) Single-cell mapper (scMappR): using scRNA-seq to infer cell-type specificities of differentially expressed genes. NAR Genomics and Bioinformatics. 3(1). Iqab011. &lt;<a href="https://doi.org/10.1093%2Fnargab%2Flqab011">doi:10.1093/nargab/lqab011</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, pheatmap, graphics, Seurat, GSVA, stats, utils,
downloader, pcaMethods, grDevices, gProfileR, limSolve,
gprofiler2, pbapply, ADAPTS, reshape,</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-27 19:13:38 UTC; djsok</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Dustin Sokolowski [aut, cre],
  Mariela Faykoo-Martinez [aut],
  Lauren Erdman [aut],
  Houyun Hou [aut],
  Cadia Chan [aut],
  Helen Zhu [aut],
  Melissa Holmes [aut],
  Anna Goldenberg [aut],
  Michael Wilson [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dustin Sokolowski &lt;djsokolowski95@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-30 08:40:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='cellmarker_enrich'>Fisher's Exact Cell-Type Identification.</h2><span id='topic+cellmarker_enrich'></span>

<h3>Description</h3>

<p>This function uses the CellMarker and Panglao datasets to identify cell-type differentially expressed genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cellmarker_enrich(
  gene_list,
  p_thresh,
  gmt = "cellmarker_list.Rdata",
  fixed_length = 13000,
  min_genes = 5,
  max_genes = 3000,
  isect_size = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cellmarker_enrich_+3A_gene_list">gene_list</code></td>
<td>
<p>A character vector of gene symbols with the same designation (e.g. mouse symbol - mouse, human symbol - human) as the gene set database.</p>
</td></tr>
<tr><td><code id="cellmarker_enrich_+3A_p_thresh">p_thresh</code></td>
<td>
<p>The Fisher's test cutoff for a cell-marker to be enriched.</p>
</td></tr>
<tr><td><code id="cellmarker_enrich_+3A_gmt">gmt</code></td>
<td>
<p>Either a path to an rda file containing an object called &quot;gmt&quot;, which is a named list where each element of the list is a vector of gene symbols website for more detail on the file type (https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats). The gmt list may also be inputted.</p>
</td></tr>
<tr><td><code id="cellmarker_enrich_+3A_fixed_length">fixed_length</code></td>
<td>
<p>Estimated number of genes in your background.</p>
</td></tr>
<tr><td><code id="cellmarker_enrich_+3A_min_genes">min_genes</code></td>
<td>
<p>Minimum number of genes in the cell-type markers.</p>
</td></tr>
<tr><td><code id="cellmarker_enrich_+3A_max_genes">max_genes</code></td>
<td>
<p>Maximum number of genes in the cell-type markers.</p>
</td></tr>
<tr><td><code id="cellmarker_enrich_+3A_isect_size">isect_size</code></td>
<td>
<p>Number of genes in your list and the cell-type.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Complete a Fisher's exact test of an input list of genes against a gene set saved in an *.RData object.
The RData object is storing a named list of genes called &quot;gmt&quot;.
</p>


<h3>Value</h3>

<p><code>cellmarker_enrich</code> Gene set enrichment of cell-types on your inputted gene list. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(POA_example)
POA_generes &lt;- POA_example$POA_generes
POA_OR_signature &lt;- POA_example$POA_OR_signature
POA_Rank_signature &lt;- POA_example$POA_Rank_signature
Signature &lt;- POA_Rank_signature
rowname &lt;- get_gene_symbol(Signature)
rownames(Signature) &lt;- rowname$rowname
genes &lt;- rownames(Signature)[1:100]
data(gmt)
enriched &lt;- cellmarker_enrich(gene_list = genes, 
                              p_thresh = 0.05, gmt = gmt)

 
</code></pre>

<hr>
<h2 id='coEnrich'>Identify co-expressed cell-types</h2><span id='topic+coEnrich'></span>

<h3>Description</h3>

<p>This function identifies genes with similar cell-type markers and if those markers are driving enrichment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coEnrich(
  sig,
  gene_list_heatmap,
  background_heatmap,
  study_name,
  outDir,
  toSave = FALSE,
  path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coEnrich_+3A_sig">sig</code></td>
<td>
<p>A The number of combinations of significant cell-types to enrich.</p>
</td></tr>
<tr><td><code id="coEnrich_+3A_gene_list_heatmap">gene_list_heatmap</code></td>
<td>
<p>Signature matrix of inputted genes in heatmap and the cell-type preferences &ndash; output of heatmap generation.</p>
</td></tr>
<tr><td><code id="coEnrich_+3A_background_heatmap">background_heatmap</code></td>
<td>
<p>Signature matrix of background matrix in heatmap and cell-type preferences &ndash; output of heatmap generation.</p>
</td></tr>
<tr><td><code id="coEnrich_+3A_study_name">study_name</code></td>
<td>
<p>Name of the outputted table.</p>
</td></tr>
<tr><td><code id="coEnrich_+3A_outdir">outDir</code></td>
<td>
<p>Name of the directory this table will be printed in.</p>
</td></tr>
<tr><td><code id="coEnrich_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F).</p>
</td></tr>
<tr><td><code id="coEnrich_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes significantly enriched cell-types from the single CT_enrich before testing to see if the genes driving their enrichment are overlapping to a significant proportion using Fisher's exact test.
To save computational time and to not complete this with an incredible number of permutations, scMappR stops at overlapping 5 cell-types.
</p>


<h3>Value</h3>

<p><code>coEnrich</code> Enrichment of cell-types that are expressed by the same genes, up to 4 sets of cell-types. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load in signature matrices
data(POA_example)
POA_generes &lt;- POA_example$POA_generes
POA_OR_signature &lt;- POA_example$POA_OR_signature
POA_Rank_signature &lt;- POA_example$POA_Rank_signature
sig &lt;- get_gene_symbol(POA_Rank_signature)
Signature &lt;- POA_Rank_signature
rownames(Signature) &lt;- sig$rowname
genes &lt;- rownames(Signature)[1:60]
heatmap_test &lt;- tissue_scMappR_custom(gene_list = genes, signature_matrix = Signature,
output_directory =  "scMappR_test", toSave = FALSE)
group_preferences &lt;- heatmap_test$group_celltype_preferences

</code></pre>

<hr>
<h2 id='compare_deconvolution_methods'>compare_deconvolution_methods</h2><span id='topic+compare_deconvolution_methods'></span>

<h3>Description</h3>

<p>This function calculates cell-type proportions of an inputted bulk sample using DeconRNA-seq, WGCNA, and DCQ methods. Outputted cell-type proportions are then compared.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_deconvolution_methods(
  count_file,
  signature_matrix,
  print_plot = FALSE,
  order_celltype = NULL,
  useWGCNA = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_deconvolution_methods_+3A_count_file">count_file</code></td>
<td>
<p>Normalized (CPM, TPM, RPKM) RNA-seq count matrix where rows are gene symbols and columns are individuals. Either the object itself of the path of a .tsv file.</p>
</td></tr>
<tr><td><code id="compare_deconvolution_methods_+3A_signature_matrix">signature_matrix</code></td>
<td>
<p>Signature matrix (odds ratios) of cell-type specificity of genes. Either the object itself or a pathway to an .RData file containing an object named &quot;wilcoxon_rank_mat_or&quot; - generally internal.</p>
</td></tr>
<tr><td><code id="compare_deconvolution_methods_+3A_print_plot">print_plot</code></td>
<td>
<p>print the barplot of estimated cell-type proportions from each method into the R console (logical: TRUE/FALSE)</p>
</td></tr>
<tr><td><code id="compare_deconvolution_methods_+3A_order_celltype">order_celltype</code></td>
<td>
<p>Specify the order that cell-type are placed on the barplot. NULL = alphabetical, otherwise a character vector of cell-type labels (i.e. column names of the signature matrix).</p>
</td></tr>
<tr><td><code id="compare_deconvolution_methods_+3A_usewgcna">useWGCNA</code></td>
<td>
<p>specify if WGCNA is installed = TRUE/FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>cellWeighted_Foldchange</code></td>
<td>
<p>data frame of cellweightedFold-changes for each gene.</p>
</td></tr>
<tr><td><code>cellType_Proportions</code></td>
<td>
<p>data frame of cell-type proportions from DeconRNA-seq.</p>
</td></tr>
<tr><td><code>leave_one_out_proportions</code></td>
<td>
<p>data frame of average cell-type proportions for case and control when gene is removed.</p>
</td></tr>
<tr><td><code>processed_signature_matrix</code></td>
<td>
<p>signature matrix used in final analysis.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
 
data(PBMC_example)
norm_counts &lt;- PBMC_example$bulk_normalized
signature &lt;- PBMC_example$odds_ratio_in
tst &lt;- compare_deconvolution_methods(count_file = norm_counts, 
 signature_matrix = signature, print_plot = FALSE,
 order_celltype = c("I_mono", "C_mono", "CD8_CM", "CD8_TE", 
 "B_SM", "B_NSM", "B_naive"), useWGCNA = FALSE)

 
</code></pre>

<hr>
<h2 id='cwFoldChange_evaluate'>Measure cell-type specificity of cell-weighted Fold-changes</h2><span id='topic+cwFoldChange_evaluate'></span>

<h3>Description</h3>

<p>This function normalizes cwFold-changes by each gene to help visualize the cell-type specificity of DEGs. It then tests if a cell-type has a large change in correlation from bulk DEGs. Finally, it identifies genes that may be specific to each cell-type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cwFoldChange_evaluate(
  cwFC,
  celltype_prop,
  DEG_list,
  gene_cutoff = NULL,
  sd_cutoff = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cwFoldChange_evaluate_+3A_cwfc">cwFC</code></td>
<td>
<p>A matrix or data frame of cell-weighted fold-changes of DEGs. Rows are DEGs and columns are cell-types.</p>
</td></tr>
<tr><td><code id="cwFoldChange_evaluate_+3A_celltype_prop">celltype_prop</code></td>
<td>
<p>A matrix or data frame of cell-type proportions. Rows are different cell-types and columns are different samples. These cell-type proportions can come from any source (not just scMappR).</p>
</td></tr>
<tr><td><code id="cwFoldChange_evaluate_+3A_deg_list">DEG_list</code></td>
<td>
<p>An object with the first column as gene symbols within the bulk dataset (doesn't have to be in signature matrix), second column is the adjusted p-value, and the third the log2FC path to a .tsv file containing this info is also acceptable.</p>
</td></tr>
<tr><td><code id="cwFoldChange_evaluate_+3A_gene_cutoff">gene_cutoff</code></td>
<td>
<p>Additional cut-off of normalized cwFold-change to see if a gene is cut-off.</p>
</td></tr>
<tr><td><code id="cwFoldChange_evaluate_+3A_sd_cutoff">sd_cutoff</code></td>
<td>
<p>Number of standard deviations or median absolute deviations to calculate outliers.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>cwFold-changes and re-normalized and re-processed to interrogate cell-type specificity at the level of the cell-type and at the level of the gene.
At the level of the cell-type, cwFold-changes are correlated to bulk DEGs. The difference in rank between bulk DEGs and cwFold-changes are also compared.
At the level of the gene, cwFold-changes are re-normalized so that each gene sums to 1. Normalization of their distributions are tested with a Shapiro test. 
Then, outlier cell-types for each gene are measured by testing for &lsquo;sd_cutoff'&rsquo;s mad or sd's greater than the median or mean depending on if the cwFold-change is non-normally or normally distributed respectively.
Cell-types considered outliers are then further filtered so their normalized cwFold-changes are greater than the cell-type proportions of that gene and 'gene_cutoff' if the user sets it.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>gene_level_investigation</code></td>
<td>
<p>data frame of genes showing the Euclidian distances between cwFold-change and null vector as well as if cwFold-changes are distributed.</p>
</td></tr>
<tr><td><code>celltype_level_investigation</code></td>
<td>
<p>data frame of Spearman's and Pearson's correlation between bulk DEGs and cwFold-changes.</p>
</td></tr>
<tr><td><code>cwFoldchange_vs_bulk_rank_change</code></td>
<td>
<p>data frame of the change in rank of DEG between the bulk fold-change and cwFold-change.</p>
</td></tr>
<tr><td><code>cwFoldChange_normalized</code></td>
<td>
<p>cwFold-change normalized such that each gene sums to 1.</p>
</td></tr>
<tr><td><code>cwFoldchange_gene_assigned</code></td>
<td>
<p>List of cell-types where genes are designated to cell-type specific differential expression.</p>
</td></tr>
<tr><td><code>cwFoldchange_gene_flagged_FP</code></td>
<td>
<p>Mapped cwFoldchanges that are flagged as false-positives. These are genes that are driven by the reciprical ratio of cell-type proportions between case and control. These genes may be DE in a non-cell-type specific manner but are falsely assigned to cell-types with very large differences in proportion between condition.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PBMC_example)
bulk_DE_cors &lt;- PBMC_example$bulk_DE_cors
bulk_normalized &lt;- PBMC_example$bulk_normalized
odds_ratio_in &lt;- PBMC_example$odds_ratio_in
case_grep &lt;- "_female"
control_grep &lt;- "_male"
max_proportion_change &lt;- 10
print_plots &lt;- FALSE
theSpecies &lt;- "human"
toOut &lt;- scMappR_and_pathway_analysis(count_file = bulk_normalized, 
                                      signature_matrix = odds_ratio_in, 
                                      DEG_list = bulk_DE_cors, case_grep = case_grep,
                                      control_grep = control_grep, rda_path = "", 
                                      max_proportion_change = 10, print_plots = TRUE, 
                                       plot_names = "tst1", theSpecies = "human", 
                                       output_directory = "tester",
                                       sig_matrix_size = 3000, up_and_downregulated = FALSE, 
                                       internet = FALSE)
                                       
cwFC1 &lt;- toOut$cellWeighted_Foldchange
prop1 &lt;- toOut$cellType_Proportions
DE &lt;- bulk_DE_cors
eval_test &lt;- cwFoldChange_evaluate(cwFC = cwFC1, celltype_prop = prop1,
                                    DEG_list = DE)

</code></pre>

<hr>
<h2 id='DeconRNAseq_CRAN'>DeconRNASeq CRAN compatible</h2><span id='topic+DeconRNAseq_CRAN'></span>

<h3>Description</h3>

<p>This function runs DeconRNAseq with default parameters such that it is compatible with CRAN and scMappR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DeconRNAseq_CRAN(
  datasets,
  signatures,
  proportions = NULL,
  checksig = FALSE,
  known.prop = FALSE,
  use.scale = TRUE,
  fig = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DeconRNAseq_CRAN_+3A_datasets">datasets</code></td>
<td>
<p>Normalized RNA-seq dataset</p>
</td></tr>
<tr><td><code id="DeconRNAseq_CRAN_+3A_signatures">signatures</code></td>
<td>
<p>Signature matrix of odds ratios</p>
</td></tr>
<tr><td><code id="DeconRNAseq_CRAN_+3A_proportions">proportions</code></td>
<td>
<p>If cell-type proportion is already inputted - always NULL for scMappR</p>
</td></tr>
<tr><td><code id="DeconRNAseq_CRAN_+3A_checksig">checksig</code></td>
<td>
<p>Check to see if plotting is significant - always false for scMappR</p>
</td></tr>
<tr><td><code id="DeconRNAseq_CRAN_+3A_known.prop">known.prop</code></td>
<td>
<p>If proportions were known - always false for scMappR</p>
</td></tr>
<tr><td><code id="DeconRNAseq_CRAN_+3A_use.scale">use.scale</code></td>
<td>
<p>Scale and center value - always TRUE for scMappR</p>
</td></tr>
<tr><td><code id="DeconRNAseq_CRAN_+3A_fig">fig</code></td>
<td>
<p>Make figures - always FALSE for scMappR</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the exact same function as the primary function in the Bioconductor package, DeconRNAseq (PMID: 23428642)
except it is now compatible with CRAN packages.
</p>


<h3>Value</h3>

<p><code>DeconRNAseq_CRAN</code> Estimated cell-type proportions with DeconRNAseq. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PBMC_example)
bulk_DE_cors &lt;- PBMC_example$bulk_DE_cors
bulk_normalized &lt;- PBMC_example$bulk_normalized
odds_ratio_in &lt;- PBMC_example$odds_ratio_in
out &lt;- DeconRNAseq_CRAN(datasets = as.data.frame(bulk_normalized),
                       signatures = as.data.frame(odds_ratio_in))

                                      
</code></pre>

<hr>
<h2 id='deconvolute_and_contextualize'>Generate cell weighted Fold-Changes (cwFold-changes)</h2><span id='topic+deconvolute_and_contextualize'></span>

<h3>Description</h3>

<p>This function takes a count matrix, signature matrix, and differentially expressed genes (DEGs) before generating cwFold-changes for each cell-type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deconvolute_and_contextualize(
  count_file,
  signature_matrix,
  DEG_list,
  case_grep,
  control_grep,
  max_proportion_change = -9,
  print_plots = TRUE,
  plot_names = "scMappR",
  theSpecies = "human",
  FC_coef = TRUE,
  sig_matrix_size = 3000,
  drop_unknown_celltype = TRUE,
  toSave = FALSE,
  path = NULL,
  deconMethod = "DeconRNASeq",
  rareCT_filter = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deconvolute_and_contextualize_+3A_count_file">count_file</code></td>
<td>
<p>Normalized (e.g. CPM, TPM, RPKM) RNA-seq count matrix where rows are gene symbols and columns are individuals. Either the matrix itself of class &quot;matrix&quot; or data.frame&quot; or a path to a tsv file containing these DEGs. The gene symbols in the count file, signature matrix, and DEG list must match.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_signature_matrix">signature_matrix</code></td>
<td>
<p>Signature matrix (fold-change ratios) of cell-type specificity of genes. Either the object itself or a pathway to an .RData file containing an object named &quot;wilcoxon_rank_mat_or&quot;. We strongly recommend inputting the signature matrix directly.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_deg_list">DEG_list</code></td>
<td>
<p>An object with the first column as gene symbols within the bulk dataset (doesn't have to be in signature matrix), second column is the adjusted P-value, and the third the log2FC. Path to a tsv file containing this info is also acceptable.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_case_grep">case_grep</code></td>
<td>
<p>Tag in the column name for cases (i.e. samples representing upregulated) OR an index of cases.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_control_grep">control_grep</code></td>
<td>
<p>Tag in the column name for control (i.e. samples representing downregulated) OR an index of cases.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_max_proportion_change">max_proportion_change</code></td>
<td>
<p>Maximum cell-type proportion change. May be useful if a cell-type does not exist in one condition, thus preventing infinite values.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_print_plots">print_plots</code></td>
<td>
<p>Whether boxplots of the estimated CT proportion for the leave-one-out method of CT deconvolution should be printed (T/F).</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_plot_names">plot_names</code></td>
<td>
<p>If plots are being printed, the pre-fix of their .pdf files.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_thespecies">theSpecies</code></td>
<td>
<p>internal species designation to be passed from 'scMappR_and_pathway_analysis'. It only impacts this function if data are taken directly from the PanglaoDB database (i.e. not reprocessed by scMappR or the user).</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_fc_coef">FC_coef</code></td>
<td>
<p>Making cwFold-changes based on fold-change (TRUE) or rank := (-log10(Pval)) (FALSE) rank. After testing, we strongly recommend to keep true (T/F).</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_sig_matrix_size">sig_matrix_size</code></td>
<td>
<p>Number of genes in signature matrix for cell-type deconvolution.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_drop_unknown_celltype">drop_unknown_celltype</code></td>
<td>
<p>Whether or not to remove &quot;unknown&quot; cell-types from the signature matrix (T/F).</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F).</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_deconmethod">deconMethod</code></td>
<td>
<p>Which RNA-seq deconvolution method to use to estimate cell-type proporitons. Options are &quot;WGCNA&quot;, &quot;DCQ&quot;, or &quot;DeconRNAseq&quot;</p>
</td></tr>
<tr><td><code id="deconvolute_and_contextualize_+3A_rarect_filter">rareCT_filter</code></td>
<td>
<p>option to keep cell-types rarer than 0.1 percent of the population (T/F). Setting to FALSE may lead to false-positives.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function completes the pre-processing, normalization, and scaling steps in the scMappR algorithm before calculating cwFold-changes. cwFold-changes scales bulk fold-changes by the cell-type specificity of the gene, cell-type gene-normalized cell-type proportions, and the reciprocal ratio of cell-type proportions between the two conditions.
cwFold-changes are generated for genes that are in both the count matrix and in the list of DEGs. It does not have to also be in the signature matrix.
First, this function will estimate cell-type proportions with all genes included before estimating changes in cell-type proportion between case/control using a t-test.
Then, it takes a leave-one-out approach to cell-type deconvolution such that estimated cell-type proportions are computed for every inputted DEG.
Optionally, the differences between cell-type proportions before and after a gene is removed is plotted in boxplots.
Then, for every gene, cwFold-changes are computed with the following formula (the example for upreguatled genes)
val &lt;- cell-preferences * cell-type_proportion * cell-type_proportion_fold-change * sign*2^abs(gene_DE$log2fc).
A matrix of cwFold-changes for all DEGs are returned.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>cellWeighted_Foldchange</code></td>
<td>
<p>data frame of cellweightedFold changes for each gene.</p>
</td></tr>
<tr><td><code>cellType_Proportions</code></td>
<td>
<p>data frame of cell-type proportions from DeconRNA-seq.</p>
</td></tr>
<tr><td><code>leave_one_out_proportions</code></td>
<td>
<p>data frame of average cell-type proportions for case and control when gene is removed.</p>
</td></tr>
<tr><td><code>processed_signature_matrix</code></td>
<td>
<p>signature matrix used in final analysis.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PBMC_example)
bulk_DE_cors &lt;- PBMC_example$bulk_DE_cors
bulk_normalized &lt;- PBMC_example$bulk_normalized
odds_ratio_in &lt;- PBMC_example$odds_ratio_in
case_grep &lt;- "_female"
control_grep &lt;- "_male"
max_proportion_change &lt;- 10
print_plots &lt;- FALSE
theSpecies &lt;- "human"
cwFC &lt;- deconvolute_and_contextualize(count_file = bulk_normalized,
                                    signature_matrix = odds_ratio_in, DEG_list = bulk_DE_cors,
                                    case_grep = case_grep, control_grep = control_grep,
                                     max_proportion_change = max_proportion_change,
                                      print_plots = print_plots, 
                                     theSpecies = theSpecies, toSave = FALSE)

                                      
</code></pre>

<hr>
<h2 id='extract_genes_cell'>Extract Markers</h2><span id='topic+extract_genes_cell'></span>

<h3>Description</h3>

<p>Extracting cell-type markers from a signature matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_genes_cell(
  geneHeat,
  cellTypes = "ALL",
  val = 1,
  isMax = FALSE,
  isPvalue = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_genes_cell_+3A_geneheat">geneHeat</code></td>
<td>
<p>The heatmap of ranks from your scRNA-seq dataset with your genes subsetted.</p>
</td></tr>
<tr><td><code id="extract_genes_cell_+3A_celltypes">cellTypes</code></td>
<td>
<p>The cell-types that you're interested in extracting. They need to be colnames (not case sensitive).</p>
</td></tr>
<tr><td><code id="extract_genes_cell_+3A_val">val</code></td>
<td>
<p>How associated a gene is with a particular cell type to include in your list - default is slightly associated.</p>
</td></tr>
<tr><td><code id="extract_genes_cell_+3A_ismax">isMax</code></td>
<td>
<p>If you are taking the single best CT marker (T/F) &ndash; TRUE not recommended.</p>
</td></tr>
<tr><td><code id="extract_genes_cell_+3A_ispvalue">isPvalue</code></td>
<td>
<p>If the signature matrix is raw p-value (T/F) &ndash; TRUE not recommended.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a signature matrix and 
extracts cell-type markers above a p-value or fold-change threshold.
</p>


<h3>Value</h3>

<p><code>extract_genes_cell</code> A vector of genes above the threshold for each sample. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(POA_example)
Signature &lt;- POA_example$POA_Rank_signature
RowName &lt;- get_gene_symbol(Signature)
rownames(Signature) &lt;-RowName$rowname
# extract genes with a -log10(Padj &gt; 1)
Signat &lt;- extract_genes_cell(Signature)



</code></pre>

<hr>
<h2 id='generes_to_heatmap'>Generate signature matrix</h2><span id='topic+generes_to_heatmap'></span>

<h3>Description</h3>

<p>Convert a list of cell-type markers from FindMarkers in Seurat to a signature matrix defined by odds ratio and rank.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generes_to_heatmap(
  generes = generes,
  species = "human",
  naming_preference = -9,
  rda_path = "",
  make_names = TRUE,
  internal = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generes_to_heatmap_+3A_generes">generes</code></td>
<td>
<p>A list of cell-type markers with fold-changes and p-values (FindMarkers output in Seurat).</p>
</td></tr>
<tr><td><code id="generes_to_heatmap_+3A_species">species</code></td>
<td>
<p>The species of gene symbols, if not internal, &quot;human&quot; or &quot;mouse&quot;.</p>
</td></tr>
<tr><td><code id="generes_to_heatmap_+3A_naming_preference">naming_preference</code></td>
<td>
<p>Likely cell-types given tissues (to be passed into human_mouse_ct_marker_enrich).</p>
</td></tr>
<tr><td><code id="generes_to_heatmap_+3A_rda_path">rda_path</code></td>
<td>
<p>Path to output direcotry, if toSave is true.</p>
</td></tr>
<tr><td><code id="generes_to_heatmap_+3A_make_names">make_names</code></td>
<td>
<p>Identify names of cell-type markers using the Fisher's exact test method (T/F).</p>
</td></tr>
<tr><td><code id="generes_to_heatmap_+3A_internal">internal</code></td>
<td>
<p>If this function is pre-processing from Panglao (T/F).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Take a list of compiled differentially expressed genes from different cell-types, identify what the cell-types are using the Fisher's exact test, and then convert into a signature matrix for both the adjusted p-value and odds ratio.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>pVal</code></td>
<td>
<p>A dataframe containing the signature matrix of ranks (-log10(Padj) * sign(fold-change)).</p>
</td></tr>
<tr><td><code>OR</code></td>
<td>
<p>A dataframe containing the signature matrix of odds ratios.</p>
</td></tr>
<tr><td><code>cellname</code></td>
<td>
<p>A vector of the cell-labels returned from the GSVA method.</p>
</td></tr>
<tr><td><code>topGenes</code></td>
<td>
<p>the top 30 mos expressed genes in each cell-type.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(POA_example)
 POA_generes &lt;- POA_example$POA_generes
signature &lt;- generes_to_heatmap(POA_generes,species = -9, make_names = FALSE)



</code></pre>

<hr>
<h2 id='get_gene_symbol'>Internal &ndash; get gene symbol from Panglao.db assigned gene-names (symbol-ensembl).</h2><span id='topic+get_gene_symbol'></span>

<h3>Description</h3>

<p>Internal &ndash; removes Ensembl signature appended to signature matrix from Panglao and figure out species by pre-fix Ensembl of the Ensembl ID that is appended to gene names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_symbol(wilcoxon_rank_mat_t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gene_symbol_+3A_wilcoxon_rank_mat_t">wilcoxon_rank_mat_t</code></td>
<td>
<p>Matrix where row names are &quot;GeneSymbol-Ensembl&quot; (human or mouse).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internal: This function removes the ENGMUS/ENGS tag from Panglao created gene names (symbol-ENGS).
From the ENSG/ENSMUS, this function determines if the species is mouse/human and returns the gene symbols.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>rowname</code></td>
<td>
<p>Genes in the signature matrix excluding the ensemble name.</p>
</td></tr>
<tr><td><code>species</code></td>
<td>
<p>&quot;mouse&quot; or &quot;human&quot; depending on appended ensembl symbols.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
 # load signature
data(POA_example) 
POA_OR_signature &lt;- POA_example$POA_OR_signature
symbols &lt;- get_gene_symbol(POA_OR_signature)


</code></pre>

<hr>
<h2 id='get_signature_matrices'>Get signature matrices.</h2><span id='topic+get_signature_matrices'></span>

<h3>Description</h3>

<p>This function downloads and returns signature matrices and associated cell-type labels from the scMappR_data repo.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_signature_matrices(type = "all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_signature_matrices_+3A_type">type</code></td>
<td>
<p>a character vector that can be 'all', 'pVal', or 'OR'</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>get_signature_matrices</code> Returns the signature matrices currently stored in scMappR_Data. Associated cell-type labels from different methods for each signature matrix is also provided.<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 
signatures &lt;- get_signature_matrices(type = "all")
 

 
</code></pre>

<hr>
<h2 id='gmt'>gmt_example</h2><span id='topic+gmt'></span>

<h3>Description</h3>

<p>Markers of 5 glial cell-types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gmt)
</code></pre>


<h3>Format</h3>

<p>A list with 5 character vectors, each containing genes.
</p>

<dl>
<dt>Astrocytes_panglao</dt><dd><p> astrocyte markers identified by panglao</p>
</dd>
<dt>Schwann_panglao</dt><dd><p> Schwann markers identified by panglao</p>
</dd>
<dt>Bergmann glia_panglao</dt><dd><p> Bergmann glia markers identified by panglao</p>
</dd>
<dt>Kupffer_panglao</dt><dd><p> Kupffer markers identified by panglao</p>
</dd>
<dt>Oligodendrocyte progenitor_panglao</dt><dd><p> Oligodendrocyte progenitor markers identified by panglao</p>
</dd>
</dl>



<h3>Details</h3>

<p>A named list containing the cell-type markers of 5 glial cell types. Used for testing cell-type naming functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gmt)
</code></pre>

<hr>
<h2 id='gProfiler_cellWeighted_Foldchange'>Pathway enrichment for cwFold-changes</h2><span id='topic+gProfiler_cellWeighted_Foldchange'></span>

<h3>Description</h3>

<p>This function runs through each list of cell weighted Fold changes (cwFold-changes) and completes both pathway and transcription factor (TF) enrichment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gProfiler_cellWeighted_Foldchange(
  cellWeighted_Foldchange_matrix,
  species,
  background,
  gene_cut,
  newGprofiler
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gProfiler_cellWeighted_Foldchange_+3A_cellweighted_foldchange_matrix">cellWeighted_Foldchange_matrix</code></td>
<td>
<p>Matrix of cell weighted Fold changes from the deconvolute_and_contextualize functions.</p>
</td></tr>
<tr><td><code id="gProfiler_cellWeighted_Foldchange_+3A_species">species</code></td>
<td>
<p>Human, mouse, or a name that is compatible with gProfileR (e.g. &quot;mmusculus&quot;).</p>
</td></tr>
<tr><td><code id="gProfiler_cellWeighted_Foldchange_+3A_background">background</code></td>
<td>
<p>A list of background genes to test against.</p>
</td></tr>
<tr><td><code id="gProfiler_cellWeighted_Foldchange_+3A_gene_cut">gene_cut</code></td>
<td>
<p>The top number of genes in pathway analysis.</p>
</td></tr>
<tr><td><code id="gProfiler_cellWeighted_Foldchange_+3A_newgprofiler">newGprofiler</code></td>
<td>
<p>Using gProfileR or gprofiler2, (T/F).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a matrix of cellWeighted_Foldchange and a species (human, mouse, or a character directly compatible with g:ProfileR).
Before completing pathway analysis with g:ProfileR. Enriched pathways are stored in a list and returned.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>BP</code></td>
<td>
<p>gprofiler enrichment of biological pathways for each cell-type</p>
</td></tr>
<tr><td><code>TF</code></td>
<td>
<p>gprofiler enrichment of transcription factors for eachc cell-type.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

data(PBMC_example)

bulk_DE_cors &lt;- PBMC_example$bulk_DE_cors
bulk_normalized &lt;- PBMC_example$bulk_normalized
odds_ratio_in &lt;- PBMC_example$odds_ratio_in

case_grep &lt;- "_female"
control_grep &lt;- "_male"
max_proportion_change &lt;- 10
print_plots &lt;- FALSE
theSpecies &lt;- "human"
norm &lt;- deconvolute_and_contextualize(count_file = bulk_normalized,
                                     signature_matrix = odds_ratio_in,
                                      DEG_list = bulk_DE_cors, case_grep = case_grep, 
                                      control_grep = control_grep,
                                       max_proportion_change = max_proportion_change,
                                       print_plots = print_plots,
                                       theSpecies = theSpecies)
background = rownames(bulk_normalized)
STVs &lt;- gProfiler_cellWeighted_Foldchange(
 cellWeighted_Foldchange_matrix = norm$cellWeighted_Foldchange,
 species = theSpecies, background = background, gene_cut = -9,
 newGprofiler = TRUE)

 
 
</code></pre>

<hr>
<h2 id='gsva_cellIdentify'>Cell-type naming with GSVA</h2><span id='topic+gsva_cellIdentify'></span>

<h3>Description</h3>

<p>This function computes the mean expression of every cell-type before predicting the most likely cell-type using the GSVA method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsva_cellIdentify(
  pbmc,
  theSpecies,
  naming_preference = -9,
  rda_path = "",
  toSave = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsva_cellIdentify_+3A_pbmc">pbmc</code></td>
<td>
<p>Processed Seurat object without named cells.</p>
</td></tr>
<tr><td><code id="gsva_cellIdentify_+3A_thespecies">theSpecies</code></td>
<td>
<p>&quot;human&quot; or &quot;mouse&quot; &ndash; it will determine which species cell-type markers will originate from.</p>
</td></tr>
<tr><td><code id="gsva_cellIdentify_+3A_naming_preference">naming_preference</code></td>
<td>
<p>Once top cell-type markers are identified, naming_preferences will then extract CT markers within a more appropriate tissue type.</p>
</td></tr>
<tr><td><code id="gsva_cellIdentify_+3A_rda_path">rda_path</code></td>
<td>
<p>Path to pre-computed cell-type .gmt files (rda objects).</p>
</td></tr>
<tr><td><code id="gsva_cellIdentify_+3A_tosave">toSave</code></td>
<td>
<p>If scMappR is allowed to write files and directories.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function inputs a Seurat object and uses the average normalized expression of each gene in each cluster to identify cell-types using the GSVA method.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>cellMarker</code></td>
<td>
<p>Most likely cell-types predicted from CellMarker database.</p>
</td></tr>
<tr><td><code>panglao</code></td>
<td>
<p>Most likely cell-types predicted from Panglao database.</p>
</td></tr>
<tr><td><code>avg_expression</code></td>
<td>
<p>Average expression of each gene in each cell-type.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

data(sm)
toProcess &lt;- list(example = sm)
tst1 &lt;- process_from_count(countmat_list = toProcess,name = "testProcess",
                            theSpecies = "mouse")
cellnames &lt;- gsva_cellIdentify(pbmc = tst1, theSpecies = "mouse",
 naming_preference = "brain", rda_path = "")


 
</code></pre>

<hr>
<h2 id='heatmap_generation'>Generate Heatmap</h2><span id='topic+heatmap_generation'></span>

<h3>Description</h3>

<p>This function takes an inputted signature matrix as well as a list of genes and overlaps them.
Then, if there is overlap, it prints a heatmap or barplot (depending on the number of overlapping genes).
Then, for every cell-type, genes considered over-represented are saved in a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_generation(
  genesIn,
  comp,
  reference,
  cex = 0.8,
  rd_path = "",
  cellTypes = "ALL",
  pVal = 0.01,
  isPval = TRUE,
  isMax = FALSE,
  isBackground = FALSE,
  which_species = "human",
  toSave = FALSE,
  path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_generation_+3A_genesin">genesIn</code></td>
<td>
<p>A list of gene symbols (all caps) to have their cell type enrichment.</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_comp">comp</code></td>
<td>
<p>The name of the comparison.</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_reference">reference</code></td>
<td>
<p>Path to signature matrix or the signature matrix itself.</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_cex">cex</code></td>
<td>
<p>The size of the genes in the column label for the heatmap.</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_rd_path">rd_path</code></td>
<td>
<p>The directory to RData files &ndash; if they are not in this directory, then the files will be downloaded.</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_celltypes">cellTypes</code></td>
<td>
<p>Colnames of the cell-types you will extract (passed to extract_genes_cell).</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_pval">pVal</code></td>
<td>
<p>The level of association a gene has within a cell type (passed to extract_genes_cell).</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_ispval">isPval</code></td>
<td>
<p>If the signature matrix is raw p-value (T/F) &ndash; TRUE not recommended</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_ismax">isMax</code></td>
<td>
<p>If you are taking the single best CT marker (T/F) &ndash; TRUE not recommended</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_isbackground">isBackground</code></td>
<td>
<p>If the heatmap is from the entire signature matrix or just the inputted gene list (T/F). isBackground == TRUE is used for internal.</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_which_species">which_species</code></td>
<td>
<p>Species of gene symbols &ndash; &quot;human&quot; or &quot;mouse&quot; .</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the path directory (T/F).</p>
</td></tr>
<tr><td><code id="heatmap_generation_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>genesIn</code></td>
<td>
<p>Vector of genes intersecting gene list and signature matrix.</p>
</td></tr>
<tr><td><code>genesNoIn</code></td>
<td>
<p>Vector of inputted genes not in signature matrix.</p>
</td></tr>
<tr><td><code>geneHeat</code></td>
<td>
<p>Signature matrix subsetted by inputted gene list</p>
</td></tr>
<tr><td><code>preferences</code></td>
<td>
<p>Cell-markers mapping to cell-types.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

# load in signature matrices
data(POA_example)
POA_generes &lt;- POA_example$POA_generes
POA_OR_signature &lt;- POA_example$POA_OR_signature
POA_Rank_signature &lt;- POA_example$POA_Rank_signature
Signature &lt;- POA_Rank_signature
rowname &lt;- get_gene_symbol(Signature)
rownames(Signature) &lt;- rowname$rowname
genes &lt;- rownames(Signature)[1:100]
heatmap_test &lt;- heatmap_generation(genesIn = genes, "scMappR_test",
                                   reference = Signature, which_species = "mouse")

 

</code></pre>

<hr>
<h2 id='human_mouse_ct_marker_enrich'>Consensus cell-type naming (Fisher's Exact)</h2><span id='topic+human_mouse_ct_marker_enrich'></span>

<h3>Description</h3>

<p>This function completes the Fisher's exact test cell-type naming for all cell-types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>human_mouse_ct_marker_enrich(
  gene_lists,
  theSpecies = "human",
  cell_marker_path = "",
  naming_preference = -9
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="human_mouse_ct_marker_enrich_+3A_gene_lists">gene_lists</code></td>
<td>
<p>A named list of vectors containing cell-type markers (mouse or human gene-symbols) which will be named as a cell-type via the Fisher's exact test method.</p>
</td></tr>
<tr><td><code id="human_mouse_ct_marker_enrich_+3A_thespecies">theSpecies</code></td>
<td>
<p>The species of the gene symbols: &quot;human&quot; or &quot;mouse&quot;.</p>
</td></tr>
<tr><td><code id="human_mouse_ct_marker_enrich_+3A_cell_marker_path">cell_marker_path</code></td>
<td>
<p>If local, path to cell-type marker rda files, otherwise, we will try to download data files.</p>
</td></tr>
<tr><td><code id="human_mouse_ct_marker_enrich_+3A_naming_preference">naming_preference</code></td>
<td>
<p>Either -9 if there is no expected cell-type or one of the categories from get_naming_preference_options(). This is useful if you previously have an idea of which cell-type you were going to enrich.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fisher's exact test method of cell-type identification using the Panglao and CellMarker databases. It extracts significant pathways (pFDR &lt; 0.05).
Then, if naming_preference != -9, it will extract the enriched cell-types within the cell-types identified with the naming preferences option.
Generally, this method seems to be biased to cell-types with a greater number of markers.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>cellTypes</code></td>
<td>
<p>most likely marker for each cell-type from each database.</p>
</td></tr>
<tr><td><code>marker_sets</code></td>
<td>
<p>all enriched cell-types for each cluster from each dataset.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

data(POA_example)
POA_generes &lt;- POA_example$POA_generes
POA_OR_signature &lt;- POA_example$POA_OR_signature
POA_Rank_signature &lt;- POA_example$POA_Rank_signature
Signature &lt;- POA_Rank_signature
rowname &lt;- get_gene_symbol(Signature)
rownames(Signature) &lt;- rowname$rowname
genes &lt;- rownames(Signature)[1:100]
enriched &lt;- human_mouse_ct_marker_enrich(gene_lists = genes, theSpecies = "mouse", 
                                         cell_marker_path = "", naming_preference = "brain")
 
</code></pre>

<hr>
<h2 id='make_TF_barplot'>Plot g:profileR Barplot (TF)</h2><span id='topic+make_TF_barplot'></span>

<h3>Description</h3>

<p>Make a barplot of the top transcription factors enriched by gprofileR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_TF_barplot(ordered_back_all_tf, top_tf = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_TF_barplot_+3A_ordered_back_all_tf">ordered_back_all_tf</code></td>
<td>
<p>Output of the g:profileR function.</p>
</td></tr>
<tr><td><code id="make_TF_barplot_+3A_top_tf">top_tf</code></td>
<td>
<p>The number of transcription factors to be plotted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a gprofileR output and prints the top &quot;top_tfs&quot; most significantly
enriched fdr adjusted p-values before plotting the rank of their p-values.
</p>


<h3>Value</h3>

<p><code>make_TF_barplot</code> A barplot of the number of &quot;top_tf&quot; tf names (not motifs), ranked by -log10(Pfdr). <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(POA_example)
 POA_generes &lt;- POA_example$POA_generes
 POA_OR_signature &lt;- POA_example$POA_OR_signature
 POA_Rank_signature &lt;- POA_example$POA_Rank_signature
Signature &lt;- as.data.frame(POA_Rank_signature)
rowname &lt;- get_gene_symbol(Signature)
rownames(Signature) &lt;- rowname$rowname
ordered_back_all &lt;- gprofiler2::gost(query = rowname$rowname[1:100], organism = "mmusculus", 
ordered_query = TRUE, significant = TRUE, exclude_iea = FALSE, multi_query = FALSE, 
measure_underrepresentation = FALSE, evcodes = FALSE, user_threshold = 0.05, 
correction_method = "fdr",   numeric_ns = "", sources = c("GO:BP", "KEGG", "REAC"))  
ordered_back_all &lt;- ordered_back_all$result
ordered_back_all &lt;- ordered_back_all[ordered_back_all$term_size &gt; 15 &amp;
 ordered_back_all$term_size &lt; 2000 &amp; ordered_back_all$intersection_size &gt; 2,]
ordered_back_all_tf &lt;- gprofiler2::gost(query = rowname$rowname[1:150], organism = "mmusculus",
 ordered_query = TRUE, significant = TRUE, exclude_iea = FALSE, multi_query = FALSE,
 measure_underrepresentation = FALSE, evcodes = FALSE, user_threshold = 0.05,
  correction_method = "fdr",  numeric_ns = "", sources = c("TF"))  
ordered_back_all_tf &lt;- ordered_back_all_tf$result
ordered_back_all_tf &lt;- ordered_back_all_tf[ordered_back_all_tf$term_size &gt; 15
 &amp; ordered_back_all_tf$term_size &lt; 5000 &amp; ordered_back_all_tf$intersection_size &gt; 2,]
TF = ordered_back_all_tf
BP &lt;- ordered_back_all
bp &lt;- plotBP(BP)
tf &lt;- make_TF_barplot(TF)
 
</code></pre>

<hr>
<h2 id='pathway_enrich_internal'>Internal - Pathway enrichment for cellWeighted_Foldchanges and bulk gene list</h2><span id='topic+pathway_enrich_internal'></span>

<h3>Description</h3>

<p>This function completes pathway enrichment of cellWeighted_Foldchanges and bulk gene list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway_enrich_internal(
  DEGs,
  theSpecies,
  scMappR_vals,
  background_genes,
  output_directory,
  plot_names,
  number_genes = -9,
  newGprofiler = FALSE,
  toSave = FALSE,
  path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_enrich_internal_+3A_degs">DEGs</code></td>
<td>
<p>Differentially expressed genes (gene_name, padj, log2fc).</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_thespecies">theSpecies</code></td>
<td>
<p>Human, mouse, or a character that is compatible with g:ProfileR.</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_scmappr_vals">scMappR_vals</code></td>
<td>
<p>cell weighted Fold-changes of differentially expressed genes.</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_background_genes">background_genes</code></td>
<td>
<p>A list of background genes to test against.</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_output_directory">output_directory</code></td>
<td>
<p>Path to the directory where files will be saved.</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_plot_names">plot_names</code></td>
<td>
<p>Names of output.</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_number_genes">number_genes</code></td>
<td>
<p>Number of genes to if there are many, many DEGs.</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_newgprofiler">newGprofiler</code></td>
<td>
<p>Whether to use g:ProfileR or gprofiler2 (T/F).</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F).</p>
</td></tr>
<tr><td><code id="pathway_enrich_internal_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internal: Pathway analysis of differentially expressed genes (DEGs) and cell weighted Fold-changes (cellWeighted_Foldchanges) for each cell-type. Returns .RData objects of differential analysis as well as plots of the top bulk pathways.
It is a wrapper for making barplots, bulk pathway analysis, and gProfiler_cellWeighted_Foldchange.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>BPs</code></td>
<td>
<p>Enriched biological pathways for each cell-type.</p>
</td></tr>
<tr><td><code>TFs</code></td>
<td>
<p>Enriched transcription factors for each cell-type.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

data(PBMC_example)
bulk_DE_cors &lt;- PBMC_example$bulk_DE_cors
bulk_normalized &lt;- PBMC_example$bulk_normalized
odds_ratio_in &lt;- PBMC_example$odds_ratio_in
case_grep &lt;- "_female"
control_grep &lt;- "_male"
max_proportion_change &lt;- 10
print_plots &lt;- FALSE
theSpecies &lt;- "human"
toOut &lt;- scMappR_and_pathway_analysis(bulk_normalized, odds_ratio_in, 
                                      bulk_DE_cors, case_grep = case_grep,
                                      control_grep = control_grep, rda_path = "", 
                                      max_proportion_change = 10, print_plots = TRUE, 
                                       plot_names = "tst1", theSpecies = "human", 
                                       output_directory = "tester",
                                       sig_matrix_size = 3000, up_and_downregulated = FALSE, 
                                       internet = FALSE)


</code></pre>

<hr>
<h2 id='PBMC_example'>PBMC_scMappR</h2><span id='topic+PBMC_example'></span>

<h3>Description</h3>

<p>Toy example of data where cell-weighted fold-changes and related downsteam analyses can be completed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(PBMC_example)
</code></pre>


<h3>Format</h3>

<p>A list containing three data frames, normalized count data, a signature matrix, and a list of differentially expressed genes.
</p>

<dl>
<dt>bulk_normalized</dt><dd><p> A 3231 x 9 matrix where rows are genes, columns are samples, and the matrix is filled with CPM normalized counts.</p>
</dd>
<dt>odds_ratio_in</dt><dd><p> A 2336 x 7 matrix where rows are genes, columns are cell-types and matrix is filled with the odds-ratio that a gene is in each cell-type.</p>
</dd>
<dt>bulk_DE_cors</dt><dd><p>A 59 x 3 matrix of sex-specific genes found between male and female PBMC samples (female biased = upregulated). row and rownames are genes, columns are gene name, FDR adjusted p-value, and log2 fold-change. DEGs were computed with DESeq2 and genes with a log2FC &gt; 1 were kept. </p>
</dd>
</dl>



<h3>Details</h3>

<p>A named list called &quot;PBMC_example&quot; containing the count data, signature matrix, and DEGs. The count data and signature matrix are shortened to fit the size of the package and do not reflect biologically relevant data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PBMC_example)
</code></pre>

<hr>
<h2 id='plotBP'>Plot gProfileR Barplot</h2><span id='topic+plotBP'></span>

<h3>Description</h3>

<p>Make a barplot of the top biological factors enriched by g:ProfileR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBP(ordered_back_all, top_bp = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBP_+3A_ordered_back_all">ordered_back_all</code></td>
<td>
<p>Output of the g:ProfileR function.</p>
</td></tr>
<tr><td><code id="plotBP_+3A_top_bp">top_bp</code></td>
<td>
<p>The number of pathways you want to plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a gProfileR output and prints the top &quot;top_bp&quot; most significantly
enriched FDR adjusted p-values before plotting the rank of their p-values.
</p>


<h3>Value</h3>

<p><code>plotBP</code> A barplot of the number of &quot;top_bp&quot; pathways, ranked by -log10(Pfdr). <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(POA_example)
 POA_generes &lt;- POA_example$POA_generes
 POA_OR_signature &lt;- POA_example$POA_OR_signature
 POA_Rank_signature &lt;- POA_example$POA_Rank_signature
Signature &lt;- as.data.frame(POA_Rank_signature)
rowname &lt;- get_gene_symbol(Signature)
rownames(Signature) &lt;- rowname$rowname
ordered_back_all &lt;- gprofiler2::gost(query = rowname$rowname[1:100], organism = "mmusculus",
 ordered_query = TRUE, significant = TRUE, exclude_iea = FALSE, multi_query = FALSE,
  measure_underrepresentation = FALSE, evcodes = FALSE, user_threshold = 0.05,
   correction_method = "fdr",   numeric_ns = "", sources = c("GO:BP", "KEGG", "REAC"))  
ordered_back_all &lt;- ordered_back_all$result
ordered_back_all &lt;- ordered_back_all[ordered_back_all$term_size &gt; 15
 &amp; ordered_back_all$term_size &lt; 2000 &amp; ordered_back_all$intersection_size &gt; 2,]
ordered_back_all_tf &lt;- gprofiler2::gost(query = rowname$rowname[1:150], organism = "mmusculus",
 ordered_query = TRUE, significant = TRUE, exclude_iea = FALSE, multi_query = FALSE,
  measure_underrepresentation = FALSE, evcodes = FALSE, user_threshold = 0.05,
   correction_method = "fdr",  numeric_ns = "", sources = c("TF"))  
ordered_back_all_tf &lt;- ordered_back_all_tf$result
ordered_back_all_tf &lt;- ordered_back_all_tf[ordered_back_all_tf$term_size &gt; 15 
&amp; ordered_back_all_tf$term_size &lt; 5000 &amp; ordered_back_all_tf$intersection_size &gt; 2,]
TF = ordered_back_all_tf
BP &lt;- ordered_back_all
bp &lt;- plotBP(ordered_back_all = BP)
tf &lt;- make_TF_barplot(ordered_back_all_tf = TF)

 
</code></pre>

<hr>
<h2 id='POA_example'>Preoptic_Area</h2><span id='topic+POA_example'></span>

<h3>Description</h3>

<p>Toy data for tissue_scMappR_custom, tissue_scMappR_internal, generes_to_heatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(POA_example)
</code></pre>


<h3>Format</h3>

<p>A list containing three objects: summary statistics of cell-type markers, a signature matrix of odds ratios, and a signature matrix of ranks.
</p>

<dl>
<dt>POA_generes</dt><dd><p>A list of 27 data frames containing (up to 30) cell-type markers. Each element of the list is a dataframe where rows are genes, and columns are p-value, log2FC, percentage of cells expressing gene in cell-type, percentage of cells expressing gene in other cell-types, and FDR adjusted p-value. </p>
</dd>
<dt>POA_OR_signature</dt><dd><p>A 266 x 27 matrix where rows are genes, columns are cell-types and matrix is filled with the odds-ratio that a gene is in each cell-type.</p>
</dd>
<dt>POA_Rank_signature</dt><dd><p>A 266 x 27 matrix of  matrix where rows are genes, columns are cell-types and matrix is filled with the rank := -log10(P_fdr) that a gene is in each cell-type. </p>
</dd>
</dl>



<h3>Details</h3>

<p>A named list called POA_example (pre-optic area example) containing three objects, POA_generes: a list of truncated dataframes containing summary statistics for each cell-type marker, POA_OR_signature a truncated signature matrix of odds ratio's for cell-types in the POA, and POA_Rank_signature a truncated signature matrix of -log10(Padj) for cell-type markers in the POA.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(POA_example)
</code></pre>

<hr>
<h2 id='process_dgTMatrix_lists'>Count Matrix To Signature Matrix</h2><span id='topic+process_dgTMatrix_lists'></span>

<h3>Description</h3>

<p>This function takes a list of count matrices, processes them, calls cell-types, and generates signature matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_dgTMatrix_lists(
  dgTMatrix_list,
  name,
  species_name,
  naming_preference = -9,
  rda_path = "",
  panglao_set = FALSE,
  haveUMAP = FALSE,
  saveSCObject = FALSE,
  internal = FALSE,
  toSave = FALSE,
  path = NULL,
  use_sctransform = FALSE,
  test_ctname = "wilcox",
  genes_integrate = 2000,
  genes_include = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_dgTMatrix_lists_+3A_dgtmatrix_list">dgTMatrix_list</code></td>
<td>
<p>A list of matrices in the class of dgTMatrix object &ndash; sparce object &ndash; compatible with Seurat rownames should be of the same species for each.</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_name">name</code></td>
<td>
<p>The name of the outputted signature matrices, cell-type preferences, and Seurat objects if you choose to save them.</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_species_name">species_name</code></td>
<td>
<p>Mouse or human symbols, -9 if internal as Panglao objects have gene symbol and ensembl combined.</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_naming_preference">naming_preference</code></td>
<td>
<p>For cell-type naming, see if cell-types given the inputted tissues are more likely to be named within one of the categories. These categories are: &quot;brain&quot;, &quot;epithelial&quot;, &quot;endothelial&quot;, &quot;blood&quot;, &quot;connective&quot;,&quot;eye&quot;, &quot;epidermis&quot;, &quot;Digestive&quot;, &quot;Immune&quot;, &quot;pancreas&quot;, &quot;liver&quot;, &quot;reproductive&quot;, &quot;kidney&quot;, &quot;respiratory&quot;.</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_rda_path">rda_path</code></td>
<td>
<p>If saved, directory to where data from scMappR_data is downloaded.</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_panglao_set">panglao_set</code></td>
<td>
<p>If the inputted matrices are from Panglao (i.e. if they're internal).</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_haveumap">haveUMAP</code></td>
<td>
<p>Save the UMAPs - requires additional packages (see Seurat for details).</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_savescobject">saveSCObject</code></td>
<td>
<p>Save the Seurat object as an RData object (T/F).</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_internal">internal</code></td>
<td>
<p>Was this used as part of the internal processing of Panglao datasets (T/F).</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F)</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_use_sctransform">use_sctransform</code></td>
<td>
<p>If you should use sctransform or the Normalize/VariableFeatures/ScaleData pipeline (T/F).</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_test_ctname">test_ctname</code></td>
<td>
<p>statistical test for calling CT markers &ndash; must be in Seurat</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_genes_integrate">genes_integrate</code></td>
<td>
<p>The number of genes to include in the integration anchors feature when combining datasets.</p>
</td></tr>
<tr><td><code id="process_dgTMatrix_lists_+3A_genes_include">genes_include</code></td>
<td>
<p>TRUE or FALSE &ndash; include 2000 genes in signature matrix or all matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a one line wrapper to process count matrices into a signature matrix.
It combines process_from_count, two methods of identifying cell-type identities (GSVA and Fisher's test).
Then, it takes the output of cell-type markers and converts it into a signature matrix of p-value ranks and odds ratios.
It saves the Seurat object (if chosen with saveSCObject), cell-type identities from GSVA (its own object), and the signature matrices.
Cell-type marker outputs are also saved in the generes .RData list. This is a list of cell-types containing all of the cell-type markers found with the FindMarkers function. Names of the generes lists and the signature matrices are kept.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>wilcoxon_rank_mat_t</code></td>
<td>
<p>A dataframe containing the signature matrix of ranks (-log10(Padj) * sign(fold-change)).</p>
</td></tr>
<tr><td><code>wilcoxon_rank_mat_or</code></td>
<td>
<p>A dataframe containing the signature matrix of odds-ratios.</p>
</td></tr>
<tr><td><code>generes</code></td>
<td>
<p>All cell-type markers for each cell-type with p-value and fold changes.</p>
</td></tr> 
<tr><td><code>cellLabel</code></td>
<td>
<p>matrix where each row is a cluster and each column provides information on the cell-type. Columns provide info on the cluster from seurat, the cell-type label from CellMarker and Panglao using the fisher's exact test and GSVA, and the top 30 markers per cluser.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(sm)
toProcess &lt;- list(example = sm)
tst1 &lt;- process_dgTMatrix_lists(dgTMatrix_list = toProcess, name = "testProcess",
                               species_name = "mouse", naming_preference = "eye", rda_path = "")


</code></pre>

<hr>
<h2 id='process_from_count'>Count Matrix To Seurat Object</h2><span id='topic+process_from_count'></span>

<h3>Description</h3>

<p>This function processes a list of count matrices (same species/gene symbols in each list) and converts them to a Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_from_count(
  countmat_list,
  name,
  theSpecies = -9,
  haveUmap = FALSE,
  saveALL = FALSE,
  panglao_set = FALSE,
  toSave = FALSE,
  path = NULL,
  use_sctransform = FALSE,
  genes_integrate = 2000,
  genes_include = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_from_count_+3A_countmat_list">countmat_list</code></td>
<td>
<p>A list of count matrices that will be integrated using the IntegrationAnchors features they should have the same rownames. A dgCMatrix or matrix object is also acceptable, and no samples will be integrated.</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_name">name</code></td>
<td>
<p>The output of the normalized and fused Seurat object if you choose to keep it.</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_thespecies">theSpecies</code></td>
<td>
<p>Gene symbols for human, mouse, or -9 if internal. If your species is not human or mouse gene symbols, make sure that you have &quot;MT-&quot; before your mitochondrial gene names then pick &quot;human&quot;.</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_haveumap">haveUmap</code></td>
<td>
<p>Write a UMAP (T/F).</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_saveall">saveALL</code></td>
<td>
<p>Save the Seurat object generated (T/F).</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_panglao_set">panglao_set</code></td>
<td>
<p>If the function is being used from internal (T/F).</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_tosave">toSave</code></td>
<td>
<p>Allows scMappR to print files and make directories locally (T/F).</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_use_sctransform">use_sctransform</code></td>
<td>
<p>If you should use sctransform or the Normalize/VariableFeatures/ScaleData pipeline (T/F).</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_genes_integrate">genes_integrate</code></td>
<td>
<p>The number of genes to include in the integration anchors feature when combining datasets</p>
</td></tr>
<tr><td><code id="process_from_count_+3A_genes_include">genes_include</code></td>
<td>
<p>TRUE or FALSE &ndash; include 2000 genes in signature matrix or all matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a list of count matrices and returns a Seurat object of the count matrices integrated using Seurat v4 (and IntegrationAnchors feature). Different normalization features such as the SCTransform pipeline are also available in this function.
Different options are used when the function is being ran internally (i.e. reprocessing count matrices from PanglaoDB) or if it is running from custom scRNA-seq data.
Larger scRNA-seq datasets can take considerable amounts of memory and run-time. See Seurat for details.
</p>


<h3>Value</h3>

<p><code>process_from_count</code> A processed and integrated Seurat object that has been scaled and clustered. It can be returned as an internal object or also stored as an RData object if necessary. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(sm)
toProcess &lt;- list(example = sm)
tst1 &lt;- process_from_count(countmat_list = toProcess, name = "testProcess", 
                           theSpecies = "mouse")


</code></pre>

<hr>
<h2 id='scMappR_and_pathway_analysis'>Generate cellWeighted_Foldchanges, visualize, and enrich.</h2><span id='topic+scMappR_and_pathway_analysis'></span>

<h3>Description</h3>

<p>This function generates cell weighted Fold-changes (cellWeighted_Foldchange), visualizes them in a heatmap, and completes pathway enrichment of cellWeighted_Foldchanges and the bulk gene list using g:ProfileR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scMappR_and_pathway_analysis(
  count_file,
  signature_matrix,
  DEG_list,
  case_grep,
  control_grep,
  rda_path = "",
  max_proportion_change = -9,
  print_plots = T,
  plot_names = "scMappR",
  theSpecies = "human",
  output_directory = "scMappR_analysis",
  sig_matrix_size = 3000,
  drop_unknown_celltype = TRUE,
  internet = TRUE,
  up_and_downregulated = FALSE,
  gene_label_size = 0.4,
  number_genes = -9,
  toSave = FALSE,
  newGprofiler = TRUE,
  path = NULL,
  deconMethod = "DeconRNASeq",
  rareCT_filter = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_count_file">count_file</code></td>
<td>
<p>Normalized (i.e. TPM, RPKM, CPM) RNA-seq count matrix where rows are gene symbols and columns are individuals. Inputted data should be a data.frame or matrix. A character vector to a tsv file where this data can be loaded is also acceptable. Gene symbols from the count file, signature matrix, and DEG list should all match (case sensitive, gene symbol or ensembl, etc.)</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_signature_matrix">signature_matrix</code></td>
<td>
<p>Signature matrix: a gene by cell-type matrix populated with the fold-change of gene expression in cell-type marker &quot;i&quot; vs all other cell-types. Object should be a data.frame or matrix.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_deg_list">DEG_list</code></td>
<td>
<p>An object with the first column as gene symbols within the bulk dataset (doesn't have to be in signature matrix), second column is the adjusted p-value, and the third the log2FC path to a .tsv file containing this info is also acceptable.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_case_grep">case_grep</code></td>
<td>
<p>A character representing what designates the &quot;cases&quot; (i.e. upregulated is 'case' biased) in the columns of the count file. A numeric vector of the index of &quot;cases&quot; is also acceptable.  Tag in the column name for cases (i.e. samples representing upregulated) OR an index of cases.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_control_grep">control_grep</code></td>
<td>
<p>A character representing what designates the &quot;control&quot; (i.e. downregulated is 'control biased) in the columns of the count file. A numeric vector of the index of &quot;control&quot; is also acceptable.  Tag in the column name for cases (i.e. samples representing upregulated) OR an index of cases.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_rda_path">rda_path</code></td>
<td>
<p>If downloaded, path to where data from scMappR_data is stored.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_max_proportion_change">max_proportion_change</code></td>
<td>
<p>Maximum cell-type proportion change &ndash; may be useful if there are many rare cell-type. Alternatively, if a cell-type is only present in one condition but not the other, it will prevent possible infinite or 0 cwFold-changes.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_print_plots">print_plots</code></td>
<td>
<p>Whether boxplots of the estimated CT proportion for the leave-one-out method of CT deconvolution should be printed. The same name of the plots will be completed for top pathways.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_plot_names">plot_names</code></td>
<td>
<p>The prefix of plot pdf files.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_thespecies">theSpecies</code></td>
<td>
<p>human, mouse, or a species directly compatible with gProfileR (i.e. g:ProfileR).</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_output_directory">output_directory</code></td>
<td>
<p>The name of the directory that will contain output of the analysis.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_sig_matrix_size">sig_matrix_size</code></td>
<td>
<p>Maximum number of genes in signature matrix for cell-type deconvolution.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_drop_unknown_celltype">drop_unknown_celltype</code></td>
<td>
<p>Whether or not to remove &quot;unknown&quot; cell-types from the signature matrix.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_internet">internet</code></td>
<td>
<p>Whether you have stable Wifi (T/F).</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_up_and_downregulated">up_and_downregulated</code></td>
<td>
<p>Whether you are additionally splitting up/downregulated genes (T/F).</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_gene_label_size">gene_label_size</code></td>
<td>
<p>The size of the gene label on the plot.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_number_genes">number_genes</code></td>
<td>
<p>The number of genes to cut-off for pathway analysis (good with many DEGs).</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F).</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_newgprofiler">newGprofiler</code></td>
<td>
<p>Whether to use gProfileR or gprofiler2 (T/F).</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_deconmethod">deconMethod</code></td>
<td>
<p>Which RNA-seq deconvolution method to use to estimate cell-type proporitons. Options are &quot;WGCNA&quot;, &quot;DCQ&quot;, or &quot;DeconRNAseq&quot;</p>
</td></tr>
<tr><td><code id="scMappR_and_pathway_analysis_+3A_rarect_filter">rareCT_filter</code></td>
<td>
<p>option to keep cell-types rarer than 0.1 percent of the population (T/F). Setting to FALSE may lead to false-positives.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates cellWeighted_Foldchanges for every cell-type (see deconvolute_and_contextualize), as well as accompanying data such as cell-type proportions with the DeconRNA-seq, WGCNA, or DCQ methods.
Then, it generates heatmaps of all cellWeighted_Foldchanges, cellWeighted_Foldchanges overlapping with the signature matrix, the entire signature matrix, the cell-type preference values from the signature matrix that overlap with inputted differentially expressed genes.
Then, assuming there is available internet, it will complete gProfileR of the reordered cellWeighted_Foldchanges as well as a the ordered list of genes.
This function is a wrapper for deconvolute_and_contextualize and pathway_enrich_internal and the primary function within the package.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>cellWeighted_Foldchanges</code></td>
<td>
<p>Cellweighted Fold-changes for all differentially expressed genes.</p>
</td></tr>
<tr><td><code>paths</code></td>
<td>
<p>Enriched biological pathways for each cell-type.</p>
</td></tr>
<tr><td><code>TFs</code></td>
<td>
<p>Enriched TFs for each cell-type.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PBMC_example)
bulk_DE_cors &lt;- PBMC_example$bulk_DE_cors
bulk_normalized &lt;- PBMC_example$bulk_normalized
odds_ratio_in &lt;- PBMC_example$odds_ratio_in
case_grep &lt;- "_female"
control_grep &lt;- "_male"
max_proportion_change &lt;- 10
print_plots &lt;- FALSE
theSpecies &lt;- "human"
toOut &lt;- scMappR_and_pathway_analysis(count_file = bulk_normalized, 
                                      signature_matrix = odds_ratio_in, 
                                      DEG_list = bulk_DE_cors, case_grep = case_grep,
                                      control_grep = control_grep, rda_path = "", 
                                      max_proportion_change = 10, print_plots = TRUE, 
                                      plot_names = "tst1", theSpecies = "human", 
                                      output_directory = "tester",
                                      sig_matrix_size = 3000,
                                      up_and_downregulated = FALSE, 
                                      internet = FALSE)



</code></pre>

<hr>
<h2 id='scMappR_tissues'>scMappR_tissues</h2><span id='topic+scMappR_tissues'></span>

<h3>Description</h3>

<p>Tissues available in scMappR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(scMappR_tissues)
</code></pre>


<h3>Format</h3>

<p>A vector of tissue names available for tissue_scMappR_internal or to download and use in scMappR_and_pathway_analysis.
</p>

<dl>
<dt>scMappR_tissues</dt><dd><p> A list of 174 tissue names from PanglaoDB. </p>
</dd>
</dl>



<h3>Details</h3>

<p>A vector of tissues available in scMappR.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(scMappR_tissues)
</code></pre>

<hr>
<h2 id='seurat_to_generes'>Identify all cell-type markers</h2><span id='topic+seurat_to_generes'></span>

<h3>Description</h3>

<p>Takes processed Seurat matrix and identifies cell-type markers with FindMarkers in Seurat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seurat_to_generes(pbmc, test = "wilcox")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seurat_to_generes_+3A_pbmc">pbmc</code></td>
<td>
<p>Processed Seurat object.</p>
</td></tr>
<tr><td><code id="seurat_to_generes_+3A_test">test</code></td>
<td>
<p>statistical test for calling CT markers &ndash; must be in Seurat.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internal: This function runs the FindMarkers function from Seurat in a loop, will use the Seurat v2 or Seurat v3 object after identifying which Seurat object is inputted. 
It then takes the output of the FindMarkers and puts it in a list, returning it.
</p>


<h3>Value</h3>

<p><code>seurat_to_generes</code> A list of genes where their over-representation in the i'th cell-type is computed. Each element contains the gene name, adjusted p-value, and the log2Fold-Change of each gene being present in that cell-type. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
 
data(sm)
toProcess &lt;- list(example = sm)
tst1 &lt;- process_from_count(countmat_list = toProcess,name = "testProcess",
                          theSpecies = "mouse")
generes &lt;- seurat_to_generes(pbmc = tst1)


</code></pre>

<hr>
<h2 id='single_gene_preferences'>Single cell-type gene preferences</h2><span id='topic+single_gene_preferences'></span>

<h3>Description</h3>

<p>Measure enrichment of individual cell-types in a signature matrix.
</p>
<p>Internal function as part of tissue_scMappR_internal().
This function takes genes preferentially expressed within a gene list, each cell-type
and the background (i.e. all genes within the signature matrix) before completing 
the cell-type specific enrichment of the inputted gene list on each cell type.
This function then returns a table describing the cell-type enrichments (p-value and odds ratio)
of each cell-type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>single_gene_preferences(
  hg_short,
  hg_full,
  study_name,
  outDir,
  toSave = FALSE,
  path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="single_gene_preferences_+3A_hg_short">hg_short</code></td>
<td>
<p>A list with two objects: a &quot;preferences&quot; and a &quot;genesIn&quot;. Preferences is a list of gene symbols over-represented in each cell-type and genesIn were all the inputted genes.</p>
</td></tr>
<tr><td><code id="single_gene_preferences_+3A_hg_full">hg_full</code></td>
<td>
<p>The same as hg_short but for every gene in the signature matrix.</p>
</td></tr>
<tr><td><code id="single_gene_preferences_+3A_study_name">study_name</code></td>
<td>
<p>Name of output table.</p>
</td></tr>
<tr><td><code id="single_gene_preferences_+3A_outdir">outDir</code></td>
<td>
<p>Directory where table is outputted.</p>
</td></tr>
<tr><td><code id="single_gene_preferences_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F).</p>
</td></tr>
<tr><td><code id="single_gene_preferences_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>single_gene_preferences</code> A gene-set enrichment table of individual cell-type enrichment. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# load in signature matrices
data(POA_example)
POA_generes &lt;- POA_example$POA_generes
POA_OR_signature &lt;- POA_example$POA_OR_signature
POA_Rank_signature &lt;- POA_example$POA_Rank_signature
sig &lt;- get_gene_symbol(POA_Rank_signature)
Signature &lt;- POA_Rank_signature
rownames(Signature) &lt;- sig$rowname
genes &lt;- rownames(Signature)[1:60]
heatmap_test &lt;- tissue_scMappR_custom(gene_list = genes, signature_matrix = Signature,
                                      output_directory =  "scMappR_test", toSave = FALSE)
single_preferences &lt;- heatmap_test$single_celltype_preferences
 
</code></pre>

<hr>
<h2 id='sm'>single_cell_process</h2><span id='topic+sm'></span>

<h3>Description</h3>

<p>Example data for processing scRNA-seq count data with Seurat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sm)
</code></pre>


<h3>Format</h3>

<p>A 752 x 236 matrix of class dgCMatrix where rows are genes and columns are cells. Data matrix is filled with counts detected from scRNAseq.
</p>

<dl>
<dt>TCTCTAACACAGGCCT</dt><dd><p> Barcode of one of the sequenced cells present. Each column is the count from a scRNA-seq dataset reprocessed by PanglaoDB.</p>
</dd>
</dl>



<h3>Details</h3>

<p>A dgCMatrix object containing count data for scRNA-seq processing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sm)
</code></pre>

<hr>
<h2 id='tissue_by_celltype_enrichment'>tissue_by_celltype_enrichment</h2><span id='topic+tissue_by_celltype_enrichment'></span>

<h3>Description</h3>

<p>This function uses a Fisher's-exact-test to rank gene-set enrichment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tissue_by_celltype_enrichment(
  gene_list,
  species,
  name = "CT_Tissue_example",
  p_thresh = 0.05,
  rda_path = "",
  isect_size = 3,
  return_gmt = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tissue_by_celltype_enrichment_+3A_gene_list">gene_list</code></td>
<td>
<p>A character vector of gene symbols with the same designation (e.g. mouse symbol - mouse, human symbol - human) as the gene set database.</p>
</td></tr>
<tr><td><code id="tissue_by_celltype_enrichment_+3A_species">species</code></td>
<td>
<p>Species of cell-type marker to use ('human' or 'mouse').</p>
</td></tr>
<tr><td><code id="tissue_by_celltype_enrichment_+3A_name">name</code></td>
<td>
<p>Name of the pdf to be printed.</p>
</td></tr>
<tr><td><code id="tissue_by_celltype_enrichment_+3A_p_thresh">p_thresh</code></td>
<td>
<p>The Fisher's test cut-off for a cell-marker to be enriched.</p>
</td></tr>
<tr><td><code id="tissue_by_celltype_enrichment_+3A_rda_path">rda_path</code></td>
<td>
<p>Path to a .rda file containing an object called &quot;gmt&quot;. Either human or mouse cell-type markers split by experiment. If the correct file isn't present they will be downloaded from https://github.com/wilsonlabgroup/scMappR_Data.</p>
</td></tr>
<tr><td><code id="tissue_by_celltype_enrichment_+3A_isect_size">isect_size</code></td>
<td>
<p>Number of genes in your list and the cell-type.</p>
</td></tr>
<tr><td><code id="tissue_by_celltype_enrichment_+3A_return_gmt">return_gmt</code></td>
<td>
<p>Return .gmt file &ndash; recommended if downloading from online as it may have updated (T/F).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Complete a Fisher's-exact test of an input list of genes against one of the two curated tissue by cell-type marker datasets from scMappR.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>enriched</code></td>
<td>
<p>Data frame of enriched cell-types from tissues.</p>
</td></tr>
<tr><td><code>gmt</code></td>
<td>
<p>Cell-markers in enriched cell-types from tissues.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

data(POA_example)
POA_generes &lt;- POA_example$POA_generes
POA_OR_signature &lt;- POA_example$POA_OR_signature
POA_Rank_signature &lt;- POA_example$POA_Rank_signature
Signature &lt;- POA_Rank_signature
rowname &lt;- get_gene_symbol(Signature)
rownames(Signature) &lt;- rowname$rowname
genes &lt;- rownames(Signature)[1:100]

enriched &lt;- tissue_by_celltype_enrichment(gene_list = genes, 
species = "mouse",p_thresh = 0.05, isect_size = 3)


 
</code></pre>

<hr>
<h2 id='tissue_scMappR_custom'>Gene List Visualization and Enrichment with Custom Signature Matrix</h2><span id='topic+tissue_scMappR_custom'></span>

<h3>Description</h3>

<p>This function visualizes signature matrix, clusters subsetted genes, completes enrichment of individual cell-types and co-enrichment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tissue_scMappR_custom(
  gene_list,
  signature_matrix,
  output_directory = "custom_test",
  toSave = FALSE,
  path = NULL,
  gene_cutoff = 1,
  is_pvalue = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tissue_scMappR_custom_+3A_gene_list">gene_list</code></td>
<td>
<p>A list of gene symbols matching that of the signature_matrix. Any gene symbol is acceptable.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_custom_+3A_signature_matrix">signature_matrix</code></td>
<td>
<p>Pre-computed signature matrix with matching gene names.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_custom_+3A_output_directory">output_directory</code></td>
<td>
<p>Directory made containing output of functions.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_custom_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F).</p>
</td></tr>
<tr><td><code id="tissue_scMappR_custom_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_custom_+3A_gene_cutoff">gene_cutoff</code></td>
<td>
<p>Value cut-off (generally rank := log10(Padj)) for a gene to be considered a marker.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_custom_+3A_is_pvalue">is_pvalue</code></td>
<td>
<p>If signature matrix is p-value before rank is applied (not recommended) (T/F).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is roughly the same as tissue_scMappR_internal, however now there is a custom signature matrix.
It generates a heatmap of the signature matrix and your inputted gene list, as well as single cell-type and 
co-celltype enrichment.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>background_heatmap</code></td>
<td>
<p>Data frame of the entire gene by cell-type signature matrix inputted.</p>
</td></tr>
<tr><td><code>gene_list_heatmap</code></td>
<td>
<p>Data frame of inputted signature matrix subsetted by input genes.</p>
</td></tr>
<tr><td><code>single_celltype_preferences</code></td>
<td>
<p>Data frame of enriched cell-types.</p>
</td></tr>
<tr><td><code>group_celtype_preference</code></td>
<td>
<p>Data frame of groups of cell-types enriched by the same genes.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>


# load in signature matrices
data(POA_example)
POA_generes &lt;- POA_example$POA_generes
POA_OR_signature &lt;- POA_example$POA_OR_signature
POA_Rank_signature &lt;- POA_example$POA_Rank_signature
sig &lt;- get_gene_symbol(POA_Rank_signature)
Signature &lt;- POA_Rank_signature
rownames(Signature) &lt;- sig$rowname
genes &lt;- rownames(Signature)[1:60]
heatmap_test &lt;- tissue_scMappR_custom(gene_list =  genes, signature_matrix = Signature,
                                      output_directory =  "scMappR_test", toSave = FALSE)



</code></pre>

<hr>
<h2 id='tissue_scMappR_internal'>Gene List Visualization and Enrichment (Internal)</h2><span id='topic+tissue_scMappR_internal'></span>

<h3>Description</h3>

<p>This function loops through every signature matrix in a particular tissue and generates heatmaps, cell-type preferences, and co-enrichment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tissue_scMappR_internal(
  gene_list,
  species,
  output_directory,
  tissue,
  rda_path = "",
  cluster = "Pval",
  genecex = 0.01,
  raw_pval = FALSE,
  path = NULL,
  toSave = FALSE,
  drop_unkown_celltype = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tissue_scMappR_internal_+3A_gene_list">gene_list</code></td>
<td>
<p>A list of gene symbols, mouse or human.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_species">species</code></td>
<td>
<p>&quot;mouse&quot;, &quot;human&quot; or &quot;-9&quot; if using a precomputed signature matrix.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_output_directory">output_directory</code></td>
<td>
<p>If toSave = TRUE, the name of the output directory that would be built.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_tissue">tissue</code></td>
<td>
<p>Name of the tissue in &quot;get_tissues&quot;.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_rda_path">rda_path</code></td>
<td>
<p>Path to the .rda file containing all of the signature matrices.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_cluster">cluster</code></td>
<td>
<p>'Pval' or 'OR' depending on if you want to cluster odds ratios or p-values of cell-type preferences.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_genecex">genecex</code></td>
<td>
<p>The size of the gene names of the rows in the heatmap.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_raw_pval">raw_pval</code></td>
<td>
<p>If the inputted signature matrix are raw (untransformed) p-values &ndash; recommended to generate rank first (T/F).</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F).</p>
</td></tr>
<tr><td><code id="tissue_scMappR_internal_+3A_drop_unkown_celltype">drop_unkown_celltype</code></td>
<td>
<p>Whether or not to remove &quot;unknown&quot; cell-types from the signature matrix (T/F).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a list of genes and a tissue that is contained in current signature matrices before and generating heatmaps of cell-type preferences.
It then completes cell-type enrichment of each individual cell-type, then, if more than two cell-types are significantly enriched, co-enrichment.
of those enriched cell-types is then computed.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>background_heatmap</code></td>
<td>
<p>Data frame of the entire gene by cell-type signature matrix inputted.</p>
</td></tr>
<tr><td><code>gene_list_heatmap</code></td>
<td>
<p>Data frame of inputted signature matrix subsetted by input genes.</p>
</td></tr>
<tr><td><code>single_celltype_preferences</code></td>
<td>
<p>Data frame of enriched cell-types.</p>
</td></tr>
<tr><td><code>group_celtype_preference</code></td>
<td>
<p>Data frame of groups of cell-types enriched by the same genes.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

data(POA_example) # region to preoptic area
Signature &lt;- POA_example$POA_Rank_signature # signature matrix
rowname &lt;- get_gene_symbol(Signature) # get signature
rownames(Signature) &lt;- rowname$rowname
genes &lt;- rownames(Signature)[1:60]
rda_path1 = "" # data directory (if it exists)

# set toSave = TRUE and path = output directory of your choice
internal &lt;- tissue_scMappR_internal(gene_list = genes, species = "mouse",
                                   output_directory = "scMappR_TesInternal",
                                   tissue = "hypothalamus", toSave = FALSE) 

 
</code></pre>

<hr>
<h2 id='tochr'>To Character.</h2><span id='topic+tochr'></span>

<h3>Description</h3>

<p>This function checks if your vector is not a character and if not, will convert it to a character.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tochr(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tochr_+3A_x">x</code></td>
<td>
<p>A character, factor or numeric vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>tochr</code> Returns a character vector. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 
 # vector of factors
 fact &lt;- factor(c("a", "b", "c", "d"))
 # convert to character
 char &lt;- tochr(x = fact)
 
</code></pre>

<hr>
<h2 id='toNum'>To Numeric.</h2><span id='topic+toNum'></span>

<h3>Description</h3>

<p>This function checks if your vector is not a character and if it is, then converts it to a numeric.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toNum(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="toNum_+3A_x">x</code></td>
<td>
<p>A character, factor, or numeric vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>toNum</code> Returns a numeric vector. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 
 # vector of factors
fact &lt;- factor(c("1", "2", "3", "4"))
# convert to numeric
num &lt;- toNum(x = fact)
 
</code></pre>

<hr>
<h2 id='topgenes_extract'>Extract Top Markers</h2><span id='topic+topgenes_extract'></span>

<h3>Description</h3>

<p>Internal &ndash; Extracts strongest cell-type markers from a Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topgenes_extract(generes, padj = 0.05, FC = 1.5, topNum = 30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topgenes_extract_+3A_generes">generes</code></td>
<td>
<p>A list of cell-type markers with fold-changes and p-values (FindMarkers output in Seurat).</p>
</td></tr>
<tr><td><code id="topgenes_extract_+3A_padj">padj</code></td>
<td>
<p>The p-value (FDR) cutoff.</p>
</td></tr>
<tr><td><code id="topgenes_extract_+3A_fc">FC</code></td>
<td>
<p>The fold-change cutoff.</p>
</td></tr>
<tr><td><code id="topgenes_extract_+3A_topnum">topNum</code></td>
<td>
<p>The number of genes to extract.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internal, this function runs through a list of outputs from FindMarkers objects in Seurat
and will extract genes past a padj and fold-change threshold. Then it extracts the topNum number of genes.
if you have not used the FindMarkers function, then a list of summary statistics with 
fold-change designated by avg_logFC and p-val by p_val_adj.
</p>


<h3>Value</h3>

<p><code>topgenes_extract</code> Returns a list of character vectors with the top (topNum) of gene markers for each cell-type. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
 
 # load generes object
 data(POA_example)
 topGenes &lt;- topgenes_extract(generes = POA_example$POA_generes)


</code></pre>

<hr>
<h2 id='two_method_pathway_enrichment'>two_method_pathway_enrichment</h2><span id='topic+two_method_pathway_enrichment'></span>

<h3>Description</h3>

<p>Pathway analysis of each cell-type based on cell-type specificity and rank improvement by scMappR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>two_method_pathway_enrichment(
  DEG_list,
  theSpecies,
  scMappR_vals,
  background_genes = NULL,
  output_directory = "output",
  plot_names = "reweighted",
  number_genes = -9,
  newGprofiler = TRUE,
  toSave = FALSE,
  path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="two_method_pathway_enrichment_+3A_deg_list">DEG_list</code></td>
<td>
<p>Differentially expressed genes (gene_name, padj, log2fc).</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_thespecies">theSpecies</code></td>
<td>
<p>Human, mouse, or a character that is compatible with g:ProfileR.</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_scmappr_vals">scMappR_vals</code></td>
<td>
<p>cell weighted Fold-changes of differentially expressed genes.</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_background_genes">background_genes</code></td>
<td>
<p>A list of background genes to test against. NULL assumes all genes in g:profileR gene set databases.</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_output_directory">output_directory</code></td>
<td>
<p>Path to the directory where files will be saved.</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_plot_names">plot_names</code></td>
<td>
<p>Names of output.</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_number_genes">number_genes</code></td>
<td>
<p>Number of genes to if there are many, many DEGs.</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_newgprofiler">newGprofiler</code></td>
<td>
<p>Whether to use g:ProfileR or gprofiler2 (T/F).</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_tosave">toSave</code></td>
<td>
<p>Allow scMappR to write files in the current directory (T/F).</p>
</td></tr>
<tr><td><code id="two_method_pathway_enrichment_+3A_path">path</code></td>
<td>
<p>If toSave == TRUE, path to the directory where files will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function re-ranks cwFoldChanges based on their absolute cell-type specificity scores (per-celltype) as well as their rank increase in cell-type specificity before completing an ordered pathway analysis. In the second method, only genes with a rank increase in cell-type specificity were included.
</p>


<h3>Value</h3>

<p>List with the following elements:
</p>
<table>
<tr><td><code>rank_increase</code></td>
<td>
<p>A list containing the degree of rank change between bulk DE genes and cwFold-changes. Pathway enrichment and TF enrichment of these reranked genes.</p>
</td></tr>
<tr><td><code>non_rank_increase</code></td>
<td>
<p>list of DFs containing the pathway and TF enrichment of cwFold-changes.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

# load data for scMappR
data(PBMC_example)
bulk_DE_cors &lt;- PBMC_example$bulk_DE_cors
bulk_normalized &lt;- PBMC_example$bulk_normalized
odds_ratio_in &lt;- PBMC_example$odds_ratio_in
case_grep &lt;- "_female"
control_grep &lt;- "_male"
max_proportion_change &lt;- 10
print_plots &lt;- FALSE
theSpecies &lt;- "human"

# calculate cwFold-changes
toOut &lt;- scMappR_and_pathway_analysis(count_file = bulk_normalized, 
                                      signature_matrix = odds_ratio_in, 
                                      DEG_list = bulk_DE_cors, case_grep = case_grep,
                                      control_grep = control_grep, rda_path = "", 
                                      max_proportion_change = 10, print_plots = TRUE, 
                                      plot_names = "tst1", theSpecies = "human", 
                                      output_directory = "tester",
                                      sig_matrix_size = 3000,
                                      up_and_downregulated = FALSE, 
                                      internet = FALSE)

# complete pathway enrichment using both methods
twoOutFiles &lt;- two_method_pathway_enrichment(DEG_list = bulk_DE_cors,theSpecies = "human",
scMappR_vals = toOut$cellWeighted_Foldchange, background_genes = rownames(bulk_normalized), 
output_directory = "newfun_test",plot_names = "nonreranked_", toSave = FALSE)


 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
