<!DOCTYPE html><html lang="en"><head><title>Help for package scOntoMatch</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scOntoMatch}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#check_ontology_translate'><p>make sure ontology names are all translated to ontology ids</p>
while warning, consider manual reannotation</a></li>
<li><a href='#fill_query'><p>Get a names list of ontology and id by id</p></a></li>
<li><a href='#getOntologyId'><p>Get a names list of ontology and id by name</p></a></li>
<li><a href='#getOntologyName'><p>Get a names list of ontology and id by id</p></a></li>
<li><a href='#getOntoMapping'><p>match ontology terms by direct mapping and mapping descendants to ancestors</p></a></li>
<li><a href='#getOntoMinimal'><p>match descendant terms to ancestor terms within a dataset</p></a></li>
<li><a href='#getOntoMultiMapping'><p>Match descendants to ancestors in multiple ontology id lists</p></a></li>
<li><a href='#getSeuratRds'><p>read in seurat object .rds files as a named list of seurat object</p></a></li>
<li><a href='#ontoMinimal'><p>get the minimal ontology tree of a dataset by reducing descendant terms to ancestor terms</p>
return  obj meta.data[[&quot;cell_ontology_base&quot;]] storing the reduced ontology annotation</a></li>
<li><a href='#ontoMultiMatch'><p>Core function of scOntoMatch</p>
Match the ontology annotation of multiple seurat objects</a></li>
<li><a href='#ontoMultiMinimal'><p>get the minimal ontology tree of a list of seurat objects by reducing descendant terms to ancestor terms</p>
return a named list of seurat objects with meta.data[[&quot;cell_ontology_base&quot;]] storing the reduced ontology annotation</a></li>
<li><a href='#ontoTranslate'><p>translate named list of obj_list to named list of cell ontology ids per obj</p></a></li>
<li><a href='#plotMatchedOntoTree'><p>Plot a ontology tree with matched ontology from ontoMatch</p></a></li>
<li><a href='#plotOntoTree'><p>Plot a tree representation of ontology terms</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Aligning Ontology Annotation Across Single Cell Datasets with
'scOntoMatch'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuyao Song  &lt;ysong@ebi.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Unequal granularity of cell type annotation makes it difficult to compare scRNA-seq datasets at scale. Leveraging the ontology system for defining cell type hierarchy, 'scOntoMatch' aims to align cell type annotations to make them comparable across studies. The alignment involves two core steps: first is to trim the cell type tree within each dataset so each cell type does not have descendants, and then map cell type labels cross-studies by direct matching and mapping descendants to ancestors. Various functions for plotting cell type trees and manipulating ontology terms are also provided. In the Single Cell Expression Atlas hosted at EBI, a compendium of datasets with curated ontology labels are great inputs to this package.</td>
</tr>
<tr>
<td>Imports:</td>
<td>ontologyIndex, ontologyPlot, purrr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Papatheodorou-Group/scOntoMatch">https://github.com/Papatheodorou-Group/scOntoMatch</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Papatheodorou-Group/scOntoMatch/issues">https://github.com/Papatheodorou-Group/scOntoMatch/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.5)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, devtools, SeuratObject</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-27 12:33:09 UTC; ysong</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuyao Song [aut, cre, ctb],
  Irene Papatheodorou [aut, ths]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-27 12:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='check_ontology_translate'>make sure ontology names are all translated to ontology ids
while warning, consider manual reannotation</h2><span id='topic+check_ontology_translate'></span>

<h3>Description</h3>

<p>make sure ontology names are all translated to ontology ids
while warning, consider manual reannotation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_ontology_translate(obj, onts, ont, anno_col)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_ontology_translate_+3A_obj">obj</code></td>
<td>
<p>seurat rds object</p>
</td></tr>
<tr><td><code id="check_ontology_translate_+3A_onts">onts</code></td>
<td>
<p>ontology ids from translate</p>
</td></tr>
<tr><td><code id="check_ontology_translate_+3A_ont">ont</code></td>
<td>
<p>ontologyIndex object</p>
</td></tr>
<tr><td><code id="check_ontology_translate_+3A_anno_col">anno_col</code></td>
<td>
<p>annotation column in obj@meta.data that is translated to onts ids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>do not return a value but output messages
</p>

<hr>
<h2 id='fill_query'>Get a names list of ontology and id by id</h2><span id='topic+fill_query'></span>

<h3>Description</h3>

<p>Get a names list of ontology and id by id
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fill_query(all, query)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fill_query_+3A_all">all</code></td>
<td>
<p>all ontology id to plot tree</p>
</td></tr>
<tr><td><code id="fill_query_+3A_query">query</code></td>
<td>
<p>query ontology id to fill</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a color object to fill query in onto_plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fill_query(all = c("CL0000548", "CL:0000066", "CL:0000082"), query = c("CL:0000082"))

## End(Not run)
</code></pre>

<hr>
<h2 id='getOntologyId'>Get a names list of ontology and id by name</h2><span id='topic+getOntologyId'></span>

<h3>Description</h3>

<p>Get a names list of ontology and id by name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOntologyId(ont, onto_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getOntologyId_+3A_ont">ont</code></td>
<td>
<p>ontology object</p>
</td></tr>
<tr><td><code id="getOntologyId_+3A_onto_name">onto_name</code></td>
<td>
<p>character vector of ontology names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list mapping ontology id and ontology name
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getOntologyId(onto_name = "epithelial cell of lung", ont = ont)

## End(Not run)
</code></pre>

<hr>
<h2 id='getOntologyName'>Get a names list of ontology and id by id</h2><span id='topic+getOntologyName'></span>

<h3>Description</h3>

<p>Get a names list of ontology and id by id
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOntologyName(ont, onto_id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getOntologyName_+3A_ont">ont</code></td>
<td>
<p>ontology object</p>
</td></tr>
<tr><td><code id="getOntologyName_+3A_onto_id">onto_id</code></td>
<td>
<p>character vector of ontology ids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list mapping ontology id and ontology name
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getOntologyName(onto_id = "epithelial cell of lung", ont = ont)

## End(Not run)
</code></pre>

<hr>
<h2 id='getOntoMapping'>match ontology terms by direct mapping and mapping descendants to ancestors</h2><span id='topic+getOntoMapping'></span>

<h3>Description</h3>

<p>match ontology terms by direct mapping and mapping descendants to ancestors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOntoMapping(ont, onts1, onts2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getOntoMapping_+3A_ont">ont</code></td>
<td>
<p>the ontology object from get_OBO</p>
</td></tr>
<tr><td><code id="getOntoMapping_+3A_onts1">onts1</code></td>
<td>
<p>a character vector of ontology id</p>
</td></tr>
<tr><td><code id="getOntoMapping_+3A_onts2">onts2</code></td>
<td>
<p>a character vector of ontology id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list for ontology id mapping looks like ontology_id:ontology_id
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getOntoMapping(ont = ont, onts1 = "CL:0000548", onts2 = c("CL0000548", "CL:0000066"))

## End(Not run)
</code></pre>

<hr>
<h2 id='getOntoMinimal'>match descendant terms to ancestor terms within a dataset</h2><span id='topic+getOntoMinimal'></span>

<h3>Description</h3>

<p>match descendant terms to ancestor terms within a dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOntoMinimal(ont, onts)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getOntoMinimal_+3A_ont">ont</code></td>
<td>
<p>the ontology object from get_OBO</p>
</td></tr>
<tr><td><code id="getOntoMinimal_+3A_onts">onts</code></td>
<td>
<p>a character vector of ontology id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list for ontology id mapping looks like ontology_id:ontology_id
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getOntoMinimal(ont = ont, onts = c("CL0000548", "CL:0000066", "CL:0000082"))

## End(Not run)
</code></pre>

<hr>
<h2 id='getOntoMultiMapping'>Match descendants to ancestors in multiple ontology id lists</h2><span id='topic+getOntoMultiMapping'></span>

<h3>Description</h3>

<p>Match descendants to ancestors in multiple ontology id lists
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOntoMultiMapping(ont, onts)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getOntoMultiMapping_+3A_ont">ont</code></td>
<td>
<p>ontologyIndex object</p>
</td></tr>
<tr><td><code id="getOntoMultiMapping_+3A_onts">onts</code></td>
<td>
<p>named list of ontology ids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named character of mapping from:mapping to
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getOntoMultiMapping(ont = ont, onts = c("CL0000548", "CL:0000066", "CL:0000082"))

## End(Not run)
</code></pre>

<hr>
<h2 id='getSeuratRds'>read in seurat object .rds files as a named list of seurat object</h2><span id='topic+getSeuratRds'></span>

<h3>Description</h3>

<p>read in seurat object .rds files as a named list of seurat object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSeuratRds(metadata, sep)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getSeuratRds_+3A_metadata">metadata</code></td>
<td>
<p>a metadata file indicating name, path to 'seurat' rds file</p>
</td></tr>
<tr><td><code id="getSeuratRds_+3A_sep">sep</code></td>
<td>
<p>sep of the metadata file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list contains data name and the corresponding 'seurat' object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getSeuratRds(metadata = 'metadata.tsv', sep = '\t')

## End(Not run)
</code></pre>

<hr>
<h2 id='ontoMinimal'>get the minimal ontology tree of a dataset by reducing descendant terms to ancestor terms
return  obj meta.data[[&quot;cell_ontology_base&quot;]] storing the reduced ontology annotation</h2><span id='topic+ontoMinimal'></span>

<h3>Description</h3>

<p>get the minimal ontology tree of a dataset by reducing descendant terms to ancestor terms
return  obj meta.data[[&quot;cell_ontology_base&quot;]] storing the reduced ontology annotation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ontoMinimal(obj, ont, anno_col, onto_id_col)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ontoMinimal_+3A_obj">obj</code></td>
<td>
<p>the seurat object</p>
</td></tr>
<tr><td><code id="ontoMinimal_+3A_ont">ont</code></td>
<td>
<p>ontologyIndex object</p>
</td></tr>
<tr><td><code id="ontoMinimal_+3A_anno_col">anno_col</code></td>
<td>
<p>the cell ontology text annotation column name</p>
</td></tr>
<tr><td><code id="ontoMinimal_+3A_onto_id_col">onto_id_col</code></td>
<td>
<p>if also have ontology id column for direct mapping</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an seurat object with  meta.data[[&quot;cell_ontology_base&quot;]]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ontoMinimal(obj = seurat_obj, ont = ont, anno_col = 'ontology_name', onto_id_col = 'ontology_id')

## End(Not run)
</code></pre>

<hr>
<h2 id='ontoMultiMatch'>Core function of scOntoMatch
Match the ontology annotation of multiple seurat objects</h2><span id='topic+ontoMultiMatch'></span>

<h3>Description</h3>

<p>Core function of scOntoMatch
Match the ontology annotation of multiple seurat objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ontoMultiMatch(obj_list, anno_col, onto_id_col, ont)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ontoMultiMatch_+3A_obj_list">obj_list</code></td>
<td>
<p>a namesd list of seurat objects to match</p>
</td></tr>
<tr><td><code id="ontoMultiMatch_+3A_anno_col">anno_col</code></td>
<td>
<p>the cell ontology text annotation column name</p>
</td></tr>
<tr><td><code id="ontoMultiMatch_+3A_onto_id_col">onto_id_col</code></td>
<td>
<p>if also have ontology id column for direct mapping</p>
</td></tr>
<tr><td><code id="ontoMultiMatch_+3A_ont">ont</code></td>
<td>
<p>ontologyIndex object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of seurat objects with annotation ontology mapped to each-other in obs[['cell_ontology_mapped']]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ontoMultiMatch(seurat_obj_list, ont, "ontology_name", 'ontology_id')

## End(Not run)
</code></pre>

<hr>
<h2 id='ontoMultiMinimal'>get the minimal ontology tree of a list of seurat objects by reducing descendant terms to ancestor terms
return a named list of seurat objects with meta.data[[&quot;cell_ontology_base&quot;]] storing the reduced ontology annotation</h2><span id='topic+ontoMultiMinimal'></span>

<h3>Description</h3>

<p>get the minimal ontology tree of a list of seurat objects by reducing descendant terms to ancestor terms
return a named list of seurat objects with meta.data[[&quot;cell_ontology_base&quot;]] storing the reduced ontology annotation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ontoMultiMinimal(obj_list, ont, anno_col = "cell_ontology_base", onto_id_col)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ontoMultiMinimal_+3A_obj_list">obj_list</code></td>
<td>
<p>a named list of seurat objects</p>
</td></tr>
<tr><td><code id="ontoMultiMinimal_+3A_ont">ont</code></td>
<td>
<p>ontologyIndex object</p>
</td></tr>
<tr><td><code id="ontoMultiMinimal_+3A_anno_col">anno_col</code></td>
<td>
<p>the cell ontology text annotation column name</p>
</td></tr>
<tr><td><code id="ontoMultiMinimal_+3A_onto_id_col">onto_id_col</code></td>
<td>
<p>if also have ontology id column for direct mapping</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list of seurat objects with meta.data[[&quot;cell_ontology_base&quot;]]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ontoMultiMinimal(seurat_obj_list, ont, "cell_ontology_base", 'ontology_id')

## End(Not run)
</code></pre>

<hr>
<h2 id='ontoTranslate'>translate named list of obj_list to named list of cell ontology ids per obj</h2><span id='topic+ontoTranslate'></span>

<h3>Description</h3>

<p>translate named list of obj_list to named list of cell ontology ids per obj
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ontoTranslate(obj_list, ont, onto_id_col, anno_col)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ontoTranslate_+3A_obj_list">obj_list</code></td>
<td>
<p>a named list of seurat object</p>
</td></tr>
<tr><td><code id="ontoTranslate_+3A_ont">ont</code></td>
<td>
<p>ontologyIndex object</p>
</td></tr>
<tr><td><code id="ontoTranslate_+3A_onto_id_col">onto_id_col</code></td>
<td>
<p>if also have ontology id column for direct mapping</p>
</td></tr>
<tr><td><code id="ontoTranslate_+3A_anno_col">anno_col</code></td>
<td>
<p>the cell ontology text annotation column name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list of cell ontology ids
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ontoTranslate(seurat_obj_list, ont, 'ontology_name', 'ontology_id')

## End(Not run)
</code></pre>

<hr>
<h2 id='plotMatchedOntoTree'>Plot a ontology tree with matched ontology from ontoMatch</h2><span id='topic+plotMatchedOntoTree'></span>

<h3>Description</h3>

<p>Plot a ontology tree with matched ontology from ontoMatch
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMatchedOntoTree(
  obj_list,
  ont,
  anno_col = "cell_ontology_mapped",
  onto_id_col,
  roots = c("CL:0000548"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotMatchedOntoTree_+3A_obj_list">obj_list</code></td>
<td>
<p>a list of seurat obj files as the output of ontoMatch</p>
</td></tr>
<tr><td><code id="plotMatchedOntoTree_+3A_ont">ont</code></td>
<td>
<p>ontology object</p>
</td></tr>
<tr><td><code id="plotMatchedOntoTree_+3A_anno_col">anno_col</code></td>
<td>
<p>the cell ontology text annotation column name</p>
</td></tr>
<tr><td><code id="plotMatchedOntoTree_+3A_onto_id_col">onto_id_col</code></td>
<td>
<p>if also have ontology id column for direct mapping</p>
</td></tr>
<tr><td><code id="plotMatchedOntoTree_+3A_roots">roots</code></td>
<td>
<p>root ontology in tree to plot, default &quot;animal cells&quot; in cell ontology</p>
</td></tr>
<tr><td><code id="plotMatchedOntoTree_+3A_...">...</code></td>
<td>
<p>additional parameters for ontologyPlot::onto_plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a lit of matched ontology tree plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plotMatchedOntoTree(seurat_obj_list, ont, 'cell_ontology_mapped', 'ontology_id')

## End(Not run)
</code></pre>

<hr>
<h2 id='plotOntoTree'>Plot a tree representation of ontology terms</h2><span id='topic+plotOntoTree'></span>

<h3>Description</h3>

<p>Plot a tree representation of ontology terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotOntoTree(
  ont,
  onts,
  plot_ancestors = TRUE,
  ont_query = NULL,
  roots = c("CL:0000548"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotOntoTree_+3A_ont">ont</code></td>
<td>
<p>ontology object</p>
</td></tr>
<tr><td><code id="plotOntoTree_+3A_onts">onts</code></td>
<td>
<p>ontology ids to plot</p>
</td></tr>
<tr><td><code id="plotOntoTree_+3A_plot_ancestors">plot_ancestors</code></td>
<td>
<p>if plot ancestors or not</p>
</td></tr>
<tr><td><code id="plotOntoTree_+3A_ont_query">ont_query</code></td>
<td>
<p>query ontology to highlight in the tree</p>
</td></tr>
<tr><td><code id="plotOntoTree_+3A_roots">roots</code></td>
<td>
<p>root ontology in tree, default &quot;animal cells&quot; in cell ontology</p>
</td></tr>
<tr><td><code id="plotOntoTree_+3A_...">...</code></td>
<td>
<p>additional parameters for ontologyPlot::onto_plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an ontology tree plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plotOntoTree(ont = ont, onts = c("CL:0000066", "CL:0000082"), ont_query = c("CL:0000082"))

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
