<!DOCTYPE html><html><head><title>Help for package scoper</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scoper}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#scoper-package'><p>scoper: Spectral Clustering-Based Method for Identifying B Cell Clones</p></a></li>
<li><a href='#ExampleDb'><p>Example database</p></a></li>
<li><a href='#hierarchicalClones'><p>Hierarchical clustering method for clonal partitioning</p></a></li>
<li><a href='#identicalClones'><p>Sequence identity method for clonal partitioning</p></a></li>
<li><a href='#plotCloneSummary'><p>Plot clonal clustering summary</p></a></li>
<li><a href='#scoper'><p>The SCOPer package</p></a></li>
<li><a href='#ScoperClones-class'><p>S4 class containing clonal assignments and summary data</p></a></li>
<li><a href='#spectralClones'><p>Spectral clustering method for clonal partitioning</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-05</td>
</tr>
<tr>
<td>Title:</td>
<td>Spectral Clustering-Based Method for Identifying B Cell Clones</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a computational framework for identification of B cell clones from 
             Adaptive Immune Receptor Repertoire sequencing (AIRR-Seq) data. Three main 
             functions are included (identicalClones, hierarchicalClones, and spectralClones) 
             that perform clustering among sequences of BCRs/IGs (B cell receptors/immunoglobulins) 
             which share the same V gene, J gene and junction length.
             Nouri N and Kleinstein SH (2018) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty235">doi:10.1093/bioinformatics/bty235</a>&gt;.
             Nouri N and Kleinstein SH (2019) &lt;<a href="https://doi.org/10.1101%2F788620">doi:10.1101/788620</a>&gt;.
             Gupta NT, et al. (2017) &lt;<a href="https://doi.org/10.4049%2Fjimmunol.1601850">doi:10.4049/jimmunol.1601850</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://scoper.readthedocs.io">https://scoper.readthedocs.io</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://bitbucket.org/kleinstein/scoper/issues">https://bitbucket.org/kleinstein/scoper/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>BuildVignettes:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), ggplot2 (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>alakazam (&ge; 1.3.0), shazam (&ge; 1.2.0), data.table,
doParallel, dplyr (&ge; 1.0), foreach, methods, Rcpp (&ge;
0.12.12), rlang, scales, stats, stringi, tidyr (&ge; 1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Collate:</td>
<td>'Data.R' 'Scoper.R' 'Functions.R' 'RcppExports.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-06 08:53:08 UTC; susanna</td>
</tr>
<tr>
<td>Author:</td>
<td>Nima Nouri [aut],
  Edel Aron [ctb],
  Gisela Gabernet [ctb],
  Susanna Marquez [ctb, cre],
  Jason Vander Heiden [aut],
  Steven Kleinstein [aut, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Susanna Marquez &lt;susanna.marquez@yale.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-06 09:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='scoper-package'>scoper: Spectral Clustering-Based Method for Identifying B Cell Clones</h2><span id='topic+scoper-package'></span><span id='topic+_PACKAGE'></span>

<h3>Description</h3>

<p>Provides a computational framework for identification of B cell clones from Adaptive Immune Receptor Repertoire sequencing (AIRR-Seq) data. Three main functions are included (identicalClones, hierarchicalClones, and spectralClones) that perform clustering among sequences of BCRs/IGs (B cell receptors/immunoglobulins) which share the same V gene, J gene and junction length. Nouri N and Kleinstein SH (2018) <a href="https://doi.org/%2010.1093/bioinformatics/bty235">doi: 10.1093/bioinformatics/bty235</a>. Nouri N and Kleinstein SH (2019) <a href="https://doi.org/%2010.1101/788620">doi: 10.1101/788620</a>. Gupta NT, et al. (2017) <a href="https://doi.org/%2010.4049/jimmunol.1601850">doi: 10.4049/jimmunol.1601850</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Susanna Marquez <a href="mailto:susanna.marquez@yale.edu">susanna.marquez@yale.edu</a> [contributor]
</p>
<p>Authors:
</p>

<ul>
<li><p> Nima Nouri <a href="mailto:nima.nouri@yale.edu">nima.nouri@yale.edu</a>
</p>
</li>
<li><p> Jason Vander Heiden <a href="mailto:jason.vanderheiden@gmail.com">jason.vanderheiden@gmail.com</a>
</p>
</li>
<li><p> Steven Kleinstein <a href="mailto:steven.kleinstein@yale.edu">steven.kleinstein@yale.edu</a> [copyright holder]
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Edel Aron <a href="mailto:edel.aron@yale.edu">edel.aron@yale.edu</a> [contributor]
</p>
</li>
<li><p> Gisela Gabernet <a href="mailto:gisela.gabernet@yale.edu">gisela.gabernet@yale.edu</a> [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://scoper.readthedocs.io">https://scoper.readthedocs.io</a>
</p>
</li>
<li><p> Report bugs at <a href="https://bitbucket.org/kleinstein/scoper/issues">https://bitbucket.org/kleinstein/scoper/issues</a>
</p>
</li></ul>


<hr>
<h2 id='ExampleDb'>Example database</h2><span id='topic+ExampleDb'></span>

<h3>Description</h3>

<p>A small example database subset from Laserson and Vigneault et al, 2014.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExampleDb
</code></pre>


<h3>Format</h3>

<p>A data.frame with the following columns:
</p>

<ul>
<li>  <p><code>sequence_id</code>:                Sequence identifier
</p>
</li>
<li>  <p><code>sequence_alignment</code>:         IMGT-gapped observed sequence.
</p>
</li>
<li>  <p><code>germline_alignment</code>:         IMGT-gapped germline sequence.
</p>
</li>
<li>  <p><code>germline_alignment_d_mask</code>:  IMGT-gapped germline sequence with N, P and
D regions masked.
</p>
</li>
<li>  <p><code>v_call</code>:                     V region allele assignments.
</p>
</li>
<li>  <p><code>v_call_genotyped</code>:           TIgGER corrected V region allele assignment.
</p>
</li>
<li>  <p><code>d_call</code>:                     D region allele assignments.
</p>
</li>
<li>  <p><code>j_call</code>:                     J region allele assignments.
</p>
</li>
<li>  <p><code>junction</code>:                   Junction region sequence.
</p>
</li>
<li>  <p><code>junction_length</code>:            Length of the junction region in nucleotides.
</p>
</li>
<li>  <p><code>np1_length</code>:                 Number of nucleotides between V and D segments
</p>
</li>
<li>  <p><code>np2_length</code>:                 Number of nucleotides between D and J segments
</p>
</li>
<li>  <p><code>sample_id</code>:                  Sample identifier
</p>
</li>
<li>  <p><code>c_call</code>:                     C region assignment.
</p>
</li>
<li>  <p><code>duplicate_count</code>:            Copy number of the sequence
</p>
</li>
<li>  <p><code>locus</code>:                      Locus of the receptor
</p>
</li></ul>



<h3>References</h3>


<ol>
<li><p>  Laserson U and Vigneault F, et al. High-resolution antibody dynamics of
vaccine-induced immune responses.
Proc Natl Acad Sci USA. 2014 111:4928-33.
</p>
</li></ol>


<hr>
<h2 id='hierarchicalClones'>Hierarchical clustering method for clonal partitioning</h2><span id='topic+hierarchicalClones'></span>

<h3>Description</h3>

<p><code>hierarchicalClones</code> provides a hierarchical agglomerative clustering 
approach to infer clonal relationships in high-throughput Adaptive Immune Receptor 
Repertoire sequencing (AIRR-seq) data. This approach clusters B or T cell receptor 
sequences based on junction region sequence similarity within partitions that share the 
same V gene, J gene, and junction length, allowing for ambiguous V or J gene annotations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hierarchicalClones(
  db,
  threshold,
  method = c("nt", "aa"),
  linkage = c("single", "average", "complete"),
  normalize = c("len", "none"),
  junction = "junction",
  v_call = "v_call",
  j_call = "j_call",
  clone = "clone_id",
  fields = NULL,
  cell_id = NULL,
  locus = "locus",
  only_heavy = TRUE,
  split_light = TRUE,
  first = FALSE,
  cdr3 = FALSE,
  mod3 = FALSE,
  max_n = 0,
  nproc = 1,
  verbose = FALSE,
  log = NULL,
  summarize_clones = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hierarchicalClones_+3A_db">db</code></td>
<td>
<p>data.frame containing sequence data.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_threshold">threshold</code></td>
<td>
<p>numeric scalar where the tree should be cut (the distance threshold for clonal grouping).</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_method">method</code></td>
<td>
<p>one of the <code>"nt"</code> for nucleotide based clustering or 
<code>"aa"</code> for amino acid based clustering.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_linkage">linkage</code></td>
<td>
<p>available linkage are <code>"single"</code>, <code>"average"</code>, and <code>"complete"</code>.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_normalize">normalize</code></td>
<td>
<p>method of normalization. The default is <code>"len"</code>, which divides the distance by the length 
of the sequence group. If <code>"none"</code> then no normalization if performed.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_junction">junction</code></td>
<td>
<p>character name of the column containing junction sequences.
Also used to determine sequence length for grouping.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_v_call">v_call</code></td>
<td>
<p>name of the column containing the V-segment allele calls.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_j_call">j_call</code></td>
<td>
<p>name of the column containing the J-segment allele calls.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_clone">clone</code></td>
<td>
<p>output column name containing the clonal cluster identifiers.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_fields">fields</code></td>
<td>
<p>character vector of additional columns to use for grouping. 
Sequences with disjoint values in the specified fields will be classified 
as separate clones.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_cell_id">cell_id</code></td>
<td>
<p>name of the column containing cell identifiers or barcodes. 
If specified, grouping will be performed in single-cell mode
with the behavior governed by the <code>locus</code> and 
<code>only_heavy</code> arguments. If set to <code>NULL</code> then the 
bulk sequencing data is assumed.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_locus">locus</code></td>
<td>
<p>name of the column containing locus information. 
Only applicable to single-cell data.
Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_only_heavy">only_heavy</code></td>
<td>
<p>use only the IGH (BCR) or TRB/TRD (TCR) sequences 
for grouping. Only applicable to single-cell data.
Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_split_light">split_light</code></td>
<td>
<p>split clones by light chains. Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_first">first</code></td>
<td>
<p>specifies how to handle multiple V(D)J assignments for initial grouping. 
If <code>TRUE</code> only the first call of the gene assignments is used. 
If <code>FALSE</code> the union of ambiguous gene assignments is used to 
group all sequences with any overlapping gene calls.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_cdr3">cdr3</code></td>
<td>
<p>if <code>TRUE</code> removes 3 nucleotides from both ends of <code>"junction"</code> 
prior to clustering (converts IMGT junction to CDR3 region). 
If <code>TRUE</code> this will also remove records with a junction length 
less than 7 nucleotides.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_mod3">mod3</code></td>
<td>
<p>if <code>TRUE</code> removes records with a <code>junction</code> length that is not divisible by 
3 in nucleotide space.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_max_n">max_n</code></td>
<td>
<p>The maximum number of degenerate characters to permit in the junction sequence 
before excluding the record from clonal assignment. Note, with 
<code>linkage="single"</code> non-informative positions can create artifactual 
links between unrelated sequences. Use with caution. 
Default is set to be zero. Set it as <code>"NULL"</code> for no action.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_nproc">nproc</code></td>
<td>
<p>number of cores to distribute the function over.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_verbose">verbose</code></td>
<td>
<p>if <code>TRUE</code> prints out a summary of each step cloning process.
if <code>FALSE</code> (default) process cloning silently.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_log">log</code></td>
<td>
<p>output path and filename to save the <code>verbose</code> log. 
The input file directory is used if path is not specified.
The default is <code>NULL</code> for no action.</p>
</td></tr>
<tr><td><code id="hierarchicalClones_+3A_summarize_clones">summarize_clones</code></td>
<td>
<p>if <code>TRUE</code> performs a series of analysis to assess the clonal landscape
and returns a <a href="#topic+ScoperClones">ScoperClones</a> object. If <code>FALSE</code> then
a modified input <code>db</code> is returned. When grouping by <code>fields</code>, 
<code>summarize_clones</code> should be <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>summarize_clones=TRUE</code> (default) a <a href="#topic+ScoperClones">ScoperClones</a> object is returned that includes the 
clonal assignment summary information and a modified input <code>db</code> in the <code>db</code> slot that 
contains clonal identifiers in the specified <code>clone</code> column.
If <code>summarize_clones=FALSE</code> modified <code>data.frame</code> is returned with clone identifiers in the 
specified <code>clone</code> column.
</p>


<h3>Single-cell data</h3>

<p>To invoke single-cell mode the <code>cell_id</code> argument must be specified and the <code>locus</code> 
column must be correct. Otherwise, clustering will be performed with bulk sequencing assumptions, 
using all input sequences regardless of the values in the <code>locus</code> column.
</p>
<p>Values in the <code>locus</code> column must be one of <code>c("IGH", "IGI", "IGK", "IGL")</code> for BCR 
or <code>c("TRA", "TRB", "TRD", "TRG")</code> for TCR sequences. Otherwise, the operation will exit and 
return an error message.
</p>
<p>Under single-cell mode with paired-chain sequences, there is a choice of whether 
grouping should be done by (a) using IGH (BCR) or TRB/TRD (TCR) sequences only or
(b) using IGH plus IGK/IGL (BCR) or TRB/TRD plus TRA/TRG (TCR) sequences. 
This is governed by the <code>only_heavy</code> argument. There is also choice as to whether 
inferred clones should be split by the light/short chain (IGK, IGL, TRA, TRG) following 
heavy/long chain clustering, which is governed by the <code>split_light</code> argument.
</p>
<p>In single-cell mode, clonal clustering will not be performed on data where cells are 
assigned multiple heavy/long chain sequences (IGH, TRB, TRD). If observed, the operation 
will exit and return an error message. Cells that lack a heavy/long chain sequence (i.e., cells with 
light/short chains only) will be assigned a <code>clone_id</code> of <code>NA</code>.
</p>


<h3>See Also</h3>

<p>See <a href="#topic+plotCloneSummary">plotCloneSummary</a> for plotting summary results. See <a href="alakazam.html#topic+groupGenes">groupGenes</a> for 
more details about grouping requirements.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Find clonal groups
results &lt;- hierarchicalClones(ExampleDb, threshold=0.15)

# Retrieve modified input data with clonal clustering identifiers
df &lt;- as.data.frame(results)

# Plot clonal summaries
plot(results, binwidth=0.02)

</code></pre>

<hr>
<h2 id='identicalClones'>Sequence identity method for clonal partitioning</h2><span id='topic+identicalClones'></span>

<h3>Description</h3>

<p><code>identicalClones</code> provides a simple sequence identity based partitioning 
approach for inferring clonal relationships in high-throughput Adaptive Immune Receptor 
Repertoire sequencing (AIRR-seq) data. This approach partitions B or T cell receptor 
sequences into clonal groups based on junction region sequence identity within 
partitions that share the same V gene, J gene, and junction length, allowing for 
ambiguous V or J gene annotations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>identicalClones(
  db,
  method = c("nt", "aa"),
  junction = "junction",
  v_call = "v_call",
  j_call = "j_call",
  clone = "clone_id",
  fields = NULL,
  cell_id = NULL,
  locus = "locus",
  only_heavy = TRUE,
  split_light = TRUE,
  first = FALSE,
  cdr3 = FALSE,
  mod3 = FALSE,
  max_n = 0,
  nproc = 1,
  verbose = FALSE,
  log = NULL,
  summarize_clones = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="identicalClones_+3A_db">db</code></td>
<td>
<p>data.frame containing sequence data.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_method">method</code></td>
<td>
<p>one of the <code>"nt"</code> for nucleotide based clustering or 
<code>"aa"</code> for amino acid based clustering.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_junction">junction</code></td>
<td>
<p>character name of the column containing junction sequences.
Also used to determine sequence length for grouping.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_v_call">v_call</code></td>
<td>
<p>name of the column containing the V-segment allele calls.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_j_call">j_call</code></td>
<td>
<p>name of the column containing the J-segment allele calls.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_clone">clone</code></td>
<td>
<p>output column name containing the clonal cluster identifiers.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_fields">fields</code></td>
<td>
<p>character vector of additional columns to use for grouping. 
Sequences with disjoint values in the specified fields will be classified 
as separate clones.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_cell_id">cell_id</code></td>
<td>
<p>name of the column containing cell identifiers or barcodes. 
If specified, grouping will be performed in single-cell mode
with the behavior governed by the <code>locus</code> and 
<code>only_heavy</code> arguments. If set to <code>NULL</code> then the 
bulk sequencing data is assumed.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_locus">locus</code></td>
<td>
<p>name of the column containing locus information. 
Only applicable to single-cell data.
Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_only_heavy">only_heavy</code></td>
<td>
<p>use only the IGH (BCR) or TRB/TRD (TCR) sequences 
for grouping. Only applicable to single-cell data.
Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_split_light">split_light</code></td>
<td>
<p>split clones by light chains. Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_first">first</code></td>
<td>
<p>specifies how to handle multiple V(D)J assignments for initial grouping. 
If <code>TRUE</code> only the first call of the gene assignments is used. 
If <code>FALSE</code> the union of ambiguous gene assignments is used to 
group all sequences with any overlapping gene calls.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_cdr3">cdr3</code></td>
<td>
<p>if <code>TRUE</code> removes 3 nucleotides from both ends of <code>"junction"</code> 
prior to clustering (converts IMGT junction to CDR3 region). 
If <code>TRUE</code> this will also remove records with a junction length 
less than 7 nucleotides.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_mod3">mod3</code></td>
<td>
<p>if <code>TRUE</code> removes records with a <code>junction</code> length that is not divisible by 
3 in nucleotide space.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_max_n">max_n</code></td>
<td>
<p>The maximum number of degenerate characters to permit in the junction sequence before excluding the 
record from clonal assignment. Default is set to be zero. Set it as <code>"NULL"</code> for no 
action.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_nproc">nproc</code></td>
<td>
<p>number of cores to distribute the function over.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_verbose">verbose</code></td>
<td>
<p>if <code>TRUE</code> prints out a summary of each step cloning process.
if <code>FALSE</code> (default) process cloning silently.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_log">log</code></td>
<td>
<p>output path and filename to save the <code>verbose</code> log. 
The input file directory is used if path is not specified.
The default is <code>NULL</code> for no action.</p>
</td></tr>
<tr><td><code id="identicalClones_+3A_summarize_clones">summarize_clones</code></td>
<td>
<p>if <code>TRUE</code> performs a series of analysis to assess the clonal landscape
and returns a <a href="#topic+ScoperClones">ScoperClones</a> object. If <code>FALSE</code> then
a modified input <code>db</code> is returned. When grouping by <code>fields</code>, 
<code>summarize_clones</code> should be <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>summarize_clones=TRUE</code> (default) a <a href="#topic+ScoperClones">ScoperClones</a> object is returned that includes the 
clonal assignment summary information and a modified input <code>db</code> in the <code>db</code> slot that 
contains clonal identifiers in the specified <code>clone</code> column.
If <code>summarize_clones=FALSE</code> modified <code>data.frame</code> is returned with clone identifiers in the 
specified <code>clone</code> column.
</p>


<h3>Single-cell data</h3>

<p>To invoke single-cell mode the <code>cell_id</code> argument must be specified and the <code>locus</code> 
column must be correct. Otherwise, clustering will be performed with bulk sequencing assumptions, 
using all input sequences regardless of the values in the <code>locus</code> column.
</p>
<p>Values in the <code>locus</code> column must be one of <code>c("IGH", "IGI", "IGK", "IGL")</code> for BCR 
or <code>c("TRA", "TRB", "TRD", "TRG")</code> for TCR sequences. Otherwise, the operation will exit and 
return an error message.
</p>
<p>Under single-cell mode with paired-chain sequences, there is a choice of whether 
grouping should be done by (a) using IGH (BCR) or TRB/TRD (TCR) sequences only or
(b) using IGH plus IGK/IGL (BCR) or TRB/TRD plus TRA/TRG (TCR) sequences. 
This is governed by the <code>only_heavy</code> argument. There is also choice as to whether 
inferred clones should be split by the light/short chain (IGK, IGL, TRA, TRG) following 
heavy/long chain clustering, which is governed by the <code>split_light</code> argument.
</p>
<p>In single-cell mode, clonal clustering will not be performed on data where cells are 
assigned multiple heavy/long chain sequences (IGH, TRB, TRD). If observed, the operation 
will exit and return an error message. Cells that lack a heavy/long chain sequence (i.e., cells with 
light/short chains only) will be assigned a <code>clone_id</code> of <code>NA</code>.
</p>


<h3>See Also</h3>

<p>See <a href="#topic+plotCloneSummary">plotCloneSummary</a> for plotting summary results. See <a href="alakazam.html#topic+groupGenes">groupGenes</a> for 
more details about grouping requirements.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Find clonal groups
results &lt;- identicalClones(ExampleDb)

# Retrieve modified input data with clonal clustering identifiers
df &lt;- as.data.frame(results)

# Plot clonal summaries
plot(results, binwidth=0.02)

</code></pre>

<hr>
<h2 id='plotCloneSummary'>Plot clonal clustering summary</h2><span id='topic+plotCloneSummary'></span>

<h3>Description</h3>

<p><code>plotCloneSummary</code> plots the results in a <code>ScoperClones</code> object returned 
by <code>spectralClones</code>, <code>identicalClones</code> or <code>hierarchicalClones</code>.  
Includes the minimum inter (between) and maximum intra (within) clonal distances 
and the calculated efective threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCloneSummary(
  data,
  xmin = NULL,
  xmax = NULL,
  breaks = NULL,
  binwidth = NULL,
  title = NULL,
  size = 0.75,
  silent = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotCloneSummary_+3A_data">data</code></td>
<td>
<p><a href="#topic+ScoperClones">ScoperClones</a> object output by the <a href="#topic+spectralClones">spectralClones</a>, 
<a href="#topic+identicalClones">identicalClones</a> or <a href="#topic+hierarchicalClones">hierarchicalClones</a>.</p>
</td></tr>
<tr><td><code id="plotCloneSummary_+3A_xmin">xmin</code></td>
<td>
<p>minimum limit for plotting the x-axis. If <code>NULL</code> the limit will 
be set automatically.</p>
</td></tr>
<tr><td><code id="plotCloneSummary_+3A_xmax">xmax</code></td>
<td>
<p>maximum limit for plotting the x-axis. If <code>NULL</code> the limit will 
be set automatically.</p>
</td></tr>
<tr><td><code id="plotCloneSummary_+3A_breaks">breaks</code></td>
<td>
<p>number of breaks to show on the x-axis. If <code>NULL</code> the breaks will 
be set automatically.</p>
</td></tr>
<tr><td><code id="plotCloneSummary_+3A_binwidth">binwidth</code></td>
<td>
<p>binwidth for the histogram. If <code>NULL</code> the binwidth 
will be set automatically.</p>
</td></tr>
<tr><td><code id="plotCloneSummary_+3A_title">title</code></td>
<td>
<p>string defining the plot title.</p>
</td></tr>
<tr><td><code id="plotCloneSummary_+3A_size">size</code></td>
<td>
<p>numeric value for lines in the plot.</p>
</td></tr>
<tr><td><code id="plotCloneSummary_+3A_silent">silent</code></td>
<td>
<p>if <code>TRUE</code> do not draw the plot and just return the ggplot2 
object; if <code>FALSE</code> draw the plot.</p>
</td></tr>
<tr><td><code id="plotCloneSummary_+3A_...">...</code></td>
<td>
<p>additional arguments to pass to ggplot2::theme.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object defining the plot.
</p>


<h3>See Also</h3>

<p>See <a href="#topic+ScoperClones">ScoperClones</a> for the the input object definition.  
See <a href="#topic+spectralClones">spectralClones</a>, <a href="#topic+identicalClones">identicalClones</a> and <a href="#topic+hierarchicalClones">hierarchicalClones</a> 
for generating the input object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Find clones
results &lt;- hierarchicalClones(ExampleDb, threshold=0.15)

# Plot clonal summaries
plot(results, binwidth=0.02)

</code></pre>

<hr>
<h2 id='scoper'>The SCOPer package</h2><span id='topic+scoper'></span>

<h3>Description</h3>

<p><code>scoper</code> is a member of the Immcantation framework and provides computational approaches 
for the identification of B cell clones from adaptive immune receptor repertoire sequencing 
(AIRR-Seq) datasets. It includes methods for assigning clonal identifiers using
sequence identity, hierarchical clustering, and spectral clustering.
</p>


<h3>Clonal clustering</h3>


<ul>
<li>  <p><a href="#topic+identicalClones">identicalClones</a>:  Clonal assignment using sequence identity partitioning.
</p>
</li>
<li>  <p><a href="#topic+hierarchicalClones">hierarchicalClones</a>:  Hierarchical clustering approach to clonal assignment.
</p>
</li>
<li>  <p><a href="#topic+spectralClones">spectralClones</a>:  Spectral clustering approach to clonal assignment.
</p>
</li></ul>



<h3>Visualization</h3>


<ul>
<li>  <p><a href="#topic+plotCloneSummary">plotCloneSummary</a>:  Visualize inter- and intra-clone distances.
</p>
</li></ul>



<h3>References</h3>


<ol>
<li><p>  Nouri N and Kleinstein SH (2018). A spectral clustering-based method for identifying clones
from high-throughput B cell repertoire sequencing data. Bioinformatics, 34(13):i341-i349.
</p>
</li>
<li><p>  Nouri N and Kleinstein SH (2019). Somatic hypermutation analysis for improved identification 
of B cell clonal families from next-generation sequencing data. bioRxiv, 10.1101/788620.
</p>
</li>
<li><p> Gupta NT, et al. (2017). Hierarchical clustering can identify B cell clones with high 
confidence in Ig repertoire sequencing data. The Journal of Immunology, 198(6):2489-2499.
</p>
</li></ol>


<hr>
<h2 id='ScoperClones-class'>S4 class containing clonal assignments and summary data</h2><span id='topic+ScoperClones-class'></span><span id='topic+ScoperClones'></span><span id='topic+print+2CScoperClones-method'></span><span id='topic+ScoperClones-method'></span><span id='topic+summary+2CScoperClones-method'></span><span id='topic+plot+2CScoperClones+2Cmissing-method'></span><span id='topic+as.data.frame+2CScoperClones-method'></span>

<h3>Description</h3>

<p><code>ScoperClones</code> stores output from <a href="#topic+identicalClones">identicalClones</a>, <a href="#topic+hierarchicalClones">hierarchicalClones</a> and
<a href="#topic+spectralClones">spectralClones</a> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ScoperClones'
print(x)

## S4 method for signature 'ScoperClones'
summary(object)

## S4 method for signature 'ScoperClones,missing'
plot(x, y, ...)

## S4 method for signature 'ScoperClones'
as.data.frame(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ScoperClones-class_+3A_x">x</code></td>
<td>
<p>ScoperClones object</p>
</td></tr>
<tr><td><code id="ScoperClones-class_+3A_object">object</code></td>
<td>
<p>ScoperClones object</p>
</td></tr>
<tr><td><code id="ScoperClones-class_+3A_y">y</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="ScoperClones-class_+3A_...">...</code></td>
<td>
<p>arguments to pass to <a href="#topic+plotCloneSummary">plotCloneSummary</a>.</p>
</td></tr>
</table>


<h3>Slots</h3>


<dl>
<dt><code>db</code></dt><dd><p><code>data.frame</code> of repertoire data including with clonal identifiers in 
the column specified during processing.</p>
</dd>
<dt><code>vjl_groups</code></dt><dd><p><code>data.frame</code> of clonal summary, including sequence count, V gene, 
J gene, junction length, and clone counts.</p>
</dd>
<dt><code>inter_intra</code></dt><dd><p><code>data.frame</code> containing minimum inter (between) and maximum intra 
(within) clonal distances.</p>
</dd>
<dt><code>eff_threshold</code></dt><dd><p>effective cut-off separating the inter (between) and intra (within) clonal 
distances.</p>
</dd>
</dl>


<h3>See Also</h3>

<p><a href="#topic+identicalClones">identicalClones</a>, <a href="#topic+hierarchicalClones">hierarchicalClones</a> and <a href="#topic+spectralClones">spectralClones</a>
</p>

<hr>
<h2 id='spectralClones'>Spectral clustering method for clonal partitioning</h2><span id='topic+spectralClones'></span>

<h3>Description</h3>

<p><code>spectralClones</code> provides an unsupervised spectral clustering 
approach to infer clonal relationships in high-throughput Adaptive Immune Receptor 
Repertoire sequencing (AIRR-seq) data. This approach clusters B or T cell receptor 
sequences based on junction region sequence similarity and shared mutations within 
partitions that share the same V gene, J gene, and junction length, allowing for 
ambiguous V or J gene annotations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spectralClones(
  db,
  method = c("novj", "vj"),
  germline = "germline_alignment",
  sequence = "sequence_alignment",
  junction = "junction",
  v_call = "v_call",
  j_call = "j_call",
  clone = "clone_id",
  fields = NULL,
  cell_id = NULL,
  locus = "locus",
  only_heavy = TRUE,
  split_light = TRUE,
  targeting_model = NULL,
  len_limit = NULL,
  first = FALSE,
  cdr3 = FALSE,
  mod3 = FALSE,
  max_n = 0,
  threshold = NULL,
  base_sim = 0.95,
  iter_max = 1000,
  nstart = 1000,
  nproc = 1,
  verbose = FALSE,
  log = NULL,
  summarize_clones = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spectralClones_+3A_db">db</code></td>
<td>
<p>data.frame containing sequence data.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_method">method</code></td>
<td>
<p>one of the <code>"novj"</code> or <code>"vj"</code>. See Details for description.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_germline">germline</code></td>
<td>
<p>character name of the column containing the germline or reference sequence.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_sequence">sequence</code></td>
<td>
<p>character name of the column containing input sequences.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_junction">junction</code></td>
<td>
<p>character name of the column containing junction sequences.
Also used to determine sequence length for grouping.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_v_call">v_call</code></td>
<td>
<p>name of the column containing the V-segment allele calls.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_j_call">j_call</code></td>
<td>
<p>name of the column containing the J-segment allele calls.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_clone">clone</code></td>
<td>
<p>output column name containing the clonal cluster identifiers.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_fields">fields</code></td>
<td>
<p>character vector of additional columns to use for grouping. 
Sequences with disjoint values in the specified fields will be classified 
as separate clones.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_cell_id">cell_id</code></td>
<td>
<p>name of the column containing cell identifiers or barcodes. 
If specified, grouping will be performed in single-cell mode
with the behavior governed by the <code>locus</code> and 
<code>only_heavy</code> arguments. If set to <code>NULL</code> then the 
bulk sequencing data is assumed.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_locus">locus</code></td>
<td>
<p>name of the column containing locus information. 
Only applicable to single-cell data.
Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_only_heavy">only_heavy</code></td>
<td>
<p>use only the IGH (BCR) or TRB/TRD (TCR) sequences 
for grouping. Only applicable to single-cell data.
Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_split_light">split_light</code></td>
<td>
<p>split clones by light chains. Ignored if <code>cell_id=NULL</code>.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_targeting_model">targeting_model</code></td>
<td>
<p><a href="shazam.html#topic+TargetingModel">TargetingModel</a> object. Only applicable if 
<code>method="vj"</code>. See Details for description.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_len_limit">len_limit</code></td>
<td>
<p><a href="shazam.html#topic+IMGT_V">IMGT_V</a> object defining the regions and boundaries of the Ig 
sequences. If NULL, mutations are counted for entire sequence. Only 
applicable if <code>method</code> = <code>"vj"</code>.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_first">first</code></td>
<td>
<p>specifies how to handle multiple V(D)J assignments for initial grouping. 
If <code>TRUE</code> only the first call of the gene assignments is used. 
If <code>FALSE</code> the union of ambiguous gene assignments is used to 
group all sequences with any overlapping gene calls.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_cdr3">cdr3</code></td>
<td>
<p>if <code>TRUE</code> removes 3 nucleotides from both ends of <code>"junction"</code> 
prior to clustering (converts IMGT junction to CDR3 region). 
If <code>TRUE</code> this will also remove records with a junction length 
less than 7 nucleotides.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_mod3">mod3</code></td>
<td>
<p>if <code>TRUE</code> removes records with a <code>junction</code> length that is not divisible by 
3 in nucleotide space.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_max_n">max_n</code></td>
<td>
<p>the maximum number of degenerate characters to permit in the junction sequence before excluding the 
record from clonal assignment. Default is set to be zero. Set it as <code>"NULL"</code> 
for no action.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_threshold">threshold</code></td>
<td>
<p>the supervising cut-off to enforce an upper-limit distance for clonal grouping.
A numeric value between (0,1).</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_base_sim">base_sim</code></td>
<td>
<p>required similarity cut-off for sequences in equal distances from each other.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_iter_max">iter_max</code></td>
<td>
<p>the maximum number of iterations allowed for kmean clustering step.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_nstart">nstart</code></td>
<td>
<p>the number of random sets chosen for kmean clustering initialization.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_nproc">nproc</code></td>
<td>
<p>number of cores to distribute the function over.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_verbose">verbose</code></td>
<td>
<p>if <code>TRUE</code> prints out a summary of each step cloning process.
if <code>FALSE</code> (default) process cloning silently.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_log">log</code></td>
<td>
<p>output path and filename to save the <code>verbose</code> log. 
The input file directory is used if path is not specified.
The default is <code>NULL</code> for no action.</p>
</td></tr>
<tr><td><code id="spectralClones_+3A_summarize_clones">summarize_clones</code></td>
<td>
<p>if <code>TRUE</code> performs a series of analysis to assess the clonal landscape
and returns a <a href="#topic+ScoperClones">ScoperClones</a> object. If <code>FALSE</code> then
a modified input <code>db</code> is returned. When grouping by <code>fields</code>, 
<code>summarize_clones</code> should be <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>method="novj"</code>, then clonal relationships are inferred using an adaptive 
threshold that indicates the level of similarity among junction sequences in a local neighborhood. 
</p>
<p>If <code>method="vj"</code>, then clonal relationships are inferred not only on 
junction region homology, but also taking into account the mutation profiles in the V 
and J segments. Mutation counts are determined by comparing the input sequences (in the 
column specified by <code>sequence</code>) to the effective germline sequence (IUPAC representation 
of sequences in the column specified by <code>germline</code>). 
</p>
<p>While not mandatory, the influence of SHM hot-/cold-spot biases in the clonal inference 
process will be noted if a SHM targeting model is provided through the <code>targeting_model</code> 
argument. See <a href="shazam.html#topic+TargetingModel">TargetingModel</a> for more technical details.
</p>
<p>If the <code>threshold</code> argument is specified, then an upper limit for clonal grouping will 
be imposed to prevent sequences with dissimilarity above the threshold from grouping together. 
Any sequence with a distance greater than the <code>threshold</code> value from the other sequences, 
will be assigned to a singleton group.
</p>


<h3>Value</h3>

<p>If <code>summarize_clones=TRUE</code> (default) a <a href="#topic+ScoperClones">ScoperClones</a> object is returned that includes the 
clonal assignment summary information and a modified input <code>db</code> in the <code>db</code> slot that 
contains clonal identifiers in the specified <code>clone</code> column.
If <code>summarize_clones=FALSE</code> modified <code>data.frame</code> is returned with clone identifiers in the 
specified <code>clone</code> column.
</p>


<h3>Single-cell data</h3>

<p>To invoke single-cell mode the <code>cell_id</code> argument must be specified and the <code>locus</code> 
column must be correct. Otherwise, clustering will be performed with bulk sequencing assumptions, 
using all input sequences regardless of the values in the <code>locus</code> column.
</p>
<p>Values in the <code>locus</code> column must be one of <code>c("IGH", "IGI", "IGK", "IGL")</code> for BCR 
or <code>c("TRA", "TRB", "TRD", "TRG")</code> for TCR sequences. Otherwise, the operation will exit and 
return an error message.
</p>
<p>Under single-cell mode with paired-chain sequences, there is a choice of whether 
grouping should be done by (a) using IGH (BCR) or TRB/TRD (TCR) sequences only or
(b) using IGH plus IGK/IGL (BCR) or TRB/TRD plus TRA/TRG (TCR) sequences. 
This is governed by the <code>only_heavy</code> argument. There is also choice as to whether 
inferred clones should be split by the light/short chain (IGK, IGL, TRA, TRG) following 
heavy/long chain clustering, which is governed by the <code>split_light</code> argument.
</p>
<p>In single-cell mode, clonal clustering will not be performed on data were cells are 
assigned multiple heavy/long chain sequences (IGH, TRB, TRD). If observed, the operation 
will exit and return an error message. Cells that lack a heavy/long chain sequence (i.e., cells with 
light/short chains only) will be assigned a <code>clone_id</code> of <code>NA</code>.
</p>


<h3>See Also</h3>

<p>See <a href="#topic+plotCloneSummary">plotCloneSummary</a> for plotting summary results. See <a href="alakazam.html#topic+groupGenes">groupGenes</a> for 
more details about grouping requirements.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Subset example data
db &lt;- subset(ExampleDb, c_call == "IGHG")

# Find clonal groups
results &lt;- spectralClones(db, method="novj", germline="germline_alignment_d_mask")

# Retrieve modified input data with clonal clustering identifiers
df &lt;- as.data.frame(results)
  
# Plot clonal summaries
plot(results, binwidth=0.02)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
