<!DOCTYPE html><html lang="en"><head><title>Help for package ScoreEB</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ScoreEB}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ebayes_EM'>
<p>Empirical Bayes for multi-locus selection</p></a></li>
<li><a href='#geno'>
<p>Genotype of example data</p></a></li>
<li><a href='#likelihood'>
<p>Carry out likelihood ratio test</p></a></li>
<li><a href='#multinormal'>
<p>Multivariate normal distribution</p></a></li>
<li><a href='#PCG'>
<p>Preconditioned Conjugate Gradient</p></a></li>
<li><a href='#pheno'>
<p>Phenotype of example data</p></a></li>
<li><a href='#ScoreEB'>
<p>Score Test Integrated with Empirical Bayes for Association Study</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Score Test Integrated with Empirical Bayes for Association Study</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Wenlong Ren</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wenlong Ren &lt;wenlongren@ntu.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform association test within linear mixed model framework using score test integrated with Empirical Bayes for genome-wide association study. Firstly, score test was conducted for each marker under linear mixed model framework, taking into account the genetic relatedness and population structure. And then all the potentially associated markers were selected with a less stringent criterion. Finally, all the selected markers were placed into a multi-locus model to identify the true quantitative trait nucleotide.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-15 15:40:45 UTC; ThinkPad</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-15 21:10:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='ebayes_EM'>
Empirical Bayes for multi-locus selection
</h2><span id='topic+ebayes_EM'></span>

<h3>Description</h3>

<p>Empirical Bayes using expectationâ€“maximization algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ebayes_EM(x,z,y,EMB.tau,EMB.omega)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ebayes_EM_+3A_x">x</code></td>
<td>
<p>fixed effect vector or matrix.</p>
</td></tr>
<tr><td><code id="ebayes_EM_+3A_z">z</code></td>
<td>
<p>genotype data.</p>
</td></tr>
<tr><td><code id="ebayes_EM_+3A_y">y</code></td>
<td>
<p>phenotype data.</p>
</td></tr>
<tr><td><code id="ebayes_EM_+3A_emb.tau">EMB.tau</code></td>
<td>
<p>one of hyperparameters in inverse chi-square distribution.</p>
</td></tr>
<tr><td><code id="ebayes_EM_+3A_emb.omega">EMB.omega</code></td>
<td>
<p>one of hyperparameters in inverse chi-square distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>u</code></td>
<td>
<p>The effect values of markers, and their absolute values are used as the basis
for further screening.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
data(pheno)
EMB.tau &lt;- 0
EMB.omega &lt;- 0
z &lt;- t(geno[,-c(1:4)])
y &lt;- as.matrix(pheno)
nsample &lt;- dim(z)[1]
x &lt;- as.matrix(rep(1,nsample))
ebayes_EM(x,z,y,EMB.tau,EMB.omega)
</code></pre>

<hr>
<h2 id='geno'>
Genotype of example data
</h2><span id='topic+geno'></span>

<h3>Description</h3>

<p>Genotype dataset with SNP chromosome, position and etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(geno)
</code></pre>


<h3>Details</h3>

<p>Dataset input of genotype in ScoreEB function.
</p>

<hr>
<h2 id='likelihood'>
Carry out likelihood ratio test
</h2><span id='topic+likelihood'></span>

<h3>Description</h3>

<p>Snps selected via EM-Bayes to further identified by likelihood ratio test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood(xxn,xxx,yn,bbo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="likelihood_+3A_xxn">xxn</code></td>
<td>
<p>fixed effect vector or matrix.</p>
</td></tr>
<tr><td><code id="likelihood_+3A_xxx">xxx</code></td>
<td>
<p>snp matrix which are selected by EM-Bayes.</p>
</td></tr>
<tr><td><code id="likelihood_+3A_yn">yn</code></td>
<td>
<p>phenotype data.</p>
</td></tr>
<tr><td><code id="likelihood_+3A_bbo">bbo</code></td>
<td>
<p>effect value of snp estimated by EM-Bayes.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>lod</code></td>
<td>
<p>Odds of logarithm vector of markers.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
data(pheno)
z &lt;- t(geno[,-c(1:4)])
y &lt;- as.matrix(pheno)
n.sample &lt;- dim(z)[1]
m.marker &lt;- dim(z)[2]
x &lt;- as.matrix(rep(1,n.sample))
beta &lt;- rnorm(m.marker)
likelihood(x,z,y,beta)
</code></pre>

<hr>
<h2 id='multinormal'>
Multivariate normal distribution
</h2><span id='topic+multinormal'></span>

<h3>Description</h3>

<p>Obtain P value with multivariate normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multinormal(y,mean,sigma)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multinormal_+3A_y">y</code></td>
<td>
<p>column vector.</p>
</td></tr>
<tr><td><code id="multinormal_+3A_mean">mean</code></td>
<td>
<p>arithmetic mean.</p>
</td></tr>
<tr><td><code id="multinormal_+3A_sigma">sigma</code></td>
<td>
<p>standard deviation.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>pdf_value</code></td>
<td>
<p>A vector of multivariate normal distribution
density function.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(pheno)
y &lt;- pheno
mean &lt;- 2.0
sigma &lt;- 1.5
multinormal(y,mean,sigma)
</code></pre>

<hr>
<h2 id='PCG'>
Preconditioned Conjugate Gradient
</h2><span id='topic+PCG'></span>

<h3>Description</h3>

<p>Conduct preconditioned conjugate gradient method to accelerate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCG(G,b,m.marker,sigma.k2,sigma.e2,tol,miter)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PCG_+3A_g">G</code></td>
<td>
<p>genotype data.</p>
</td></tr>
<tr><td><code id="PCG_+3A_b">b</code></td>
<td>
<p>column vector.</p>
</td></tr>
<tr><td><code id="PCG_+3A_m.marker">m.marker</code></td>
<td>
<p>the number of markers.</p>
</td></tr>
<tr><td><code id="PCG_+3A_sigma.k2">sigma.k2</code></td>
<td>
<p>variance of polygenic.</p>
</td></tr>
<tr><td><code id="PCG_+3A_sigma.e2">sigma.e2</code></td>
<td>
<p>variance of residual error.</p>
</td></tr>
<tr><td><code id="PCG_+3A_tol">tol</code></td>
<td>
<p>convergence threshold.</p>
</td></tr>
<tr><td><code id="PCG_+3A_miter">miter</code></td>
<td>
<p>the maximum number of iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p>x is approximate solution of linear equations.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
G &lt;- t(geno[,-c(1:4)])
n.sample &lt;- dim(G)[1]
m.marker &lt;- dim(G)[2]
b &lt;- rnorm(n.sample)
sigma.k2 &lt;- 6.0
sigma.e2 &lt;- 10.0
tol &lt;- 5e-4
miter &lt;- 20
PCG(G,b,m.marker,sigma.k2,sigma.e2,tol,miter)
</code></pre>

<hr>
<h2 id='pheno'>
Phenotype of example data
</h2><span id='topic+pheno'></span>

<h3>Description</h3>

<p>Phenotype dataset of multiple traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pheno)
</code></pre>


<h3>Details</h3>

<p>Dataset input of phenotype in ScoreEB function.
</p>

<hr>
<h2 id='ScoreEB'>
Score Test Integrated with Empirical Bayes for Association Study
</h2><span id='topic+ScoreEB'></span>

<h3>Description</h3>

<p>Perform association test within linear mixed model framework using score test integrated with Empirical Bayes for genome-wide association study. Firstly, score test was conducted for each marker under linear mixed model framework, taking into account the genetic relatedness and population structure. And then all the potentially associated markers were selected with a less stringent criterion. Finally, all the selected markers were placed into a multi-locus model to identify the true quantitative trait nucleotide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScoreEB(genofile, phenofile, popfile = NULL, trait.num = 1, EMB.tau = 0,
EMB.omega = 0, B.Moment = 20, tol.pcg = 1e-4, iter.pcg = 100, bin = 100,
lod.cutoff = 3.0, seed.num = 10000, dir_out)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ScoreEB_+3A_genofile">genofile</code></td>
<td>
<p>Genotype file name, change the file path where it is located,
i.e.,&quot;D:/Genotype_Example.csv&quot;.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_phenofile">phenofile</code></td>
<td>
<p>Phenotype file name, change the file path where it is located,
i.e.,&quot;D:/Phenotype_Example.csv&quot;.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_popfile">popfile</code></td>
<td>
<p>Population structure file name, change the file path
where it is located,i.e.,&quot;D:/Population.csv&quot;.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_trait.num">trait.num</code></td>
<td>
<p>trait.num stands for computing trait from the 1st to the &quot;trait.num&quot;</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_emb.tau">EMB.tau</code></td>
<td>
<p>EMB.tau and EMB.omega are two values of hyperparameters in
empirical Bayes step, which are set to 0 by default.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_emb.omega">EMB.omega</code></td>
<td>
<p>As describe in EMB.tau</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_b.moment">B.Moment</code></td>
<td>
<p>B.Moment is a parameter to obtain trace of NxN matrix approximately
using method of moment. B.Moment is set to 20 by default.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_tol.pcg">tol.pcg</code></td>
<td>
<p>tol.pcg and iter.pcg are tolerance and maximum iteration number in
preconditioned conjugate gradient algorithm.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_iter.pcg">iter.pcg</code></td>
<td>
<p>As describe in tol.pcg</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_bin">bin</code></td>
<td>
<p>bin is to choose the maximum score within a certain range.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_lod.cutoff">lod.cutoff</code></td>
<td>
<p>lod.cutoff is the threshold to determine identified QTNs.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_seed.num">seed.num</code></td>
<td>
<p>Set a random number.</p>
</td></tr>
<tr><td><code id="ScoreEB_+3A_dir_out">dir_out</code></td>
<td>
<p>Give the path where it will be saved,i.e.,&quot;D:/Result&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>result.total</code></td>
<td>
<p>A data frame of identified markers, including &quot;Trait&quot;, &quot;Id&quot;, &quot;Chr&quot;,
&quot;Pos&quot;, &quot;Score&quot;, &quot;Beta&quot;, &quot;Lod&quot; and &quot;Pvalue&quot; of markers.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>1. genofile and phenofile are the required input file, while popfile is the optional input file.<br />
2. In the &quot;tempdir()&quot; folder, there are two results files &quot;ScoreEB.Result.csv&quot; and &quot;ScoreEB.time.csv&quot; generated and saved after the run.<br />
3. The results file &quot;ScoreEB.Result.csv&quot; has 8 columns, including &quot;Trait&quot;, &quot;Id&quot;, &quot;Chr&quot;, &quot;Pos&quot;, &quot;Score&quot;, &quot;Beta&quot;, &quot;Lod&quot; and &quot;Pvalue&quot;.<br />
4. The time file &quot;ScoreEB.time.csv&quot; includes 3 rows, which are &quot;User&quot;, &quot;System&quot;, &quot;Elapse&quot; time, respectively.<br />
</p>


<h3>Author(s)</h3>

<p>Wenlong Ren<br />
Wenlong Ren &lt;wenlongren@ntu.edu.cn&gt;<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genofile &lt;- system.file("extdata", "Genotype_Example.csv", package="ScoreEB")
phenofile &lt;-system.file("extdata", "Phenotype_Example.csv", package="ScoreEB")
dir_out &lt;- tempdir()
ScoreEB(genofile, phenofile, popfile = NULL, trait.num = 1, EMB.tau = 0,
EMB.omega = 0, B.Moment = 20, tol.pcg = 1e-4, iter.pcg = 100, bin = 100,
lod.cutoff = 3.0, seed.num = 10000, dir_out)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
