<!DOCTYPE html><html><head><title>Help for package SCORPION</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SCORPION}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#scorpion'><p>Constructs PANDA gene regulatory networks from single-cell gene expression data</p></a></li>
<li><a href='#scorpionTest'><p>Example single-cell gene expression, motif, and ppi data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single Cell Oriented Reconstruction of PANDA Individual
Optimized Networks</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Constructs gene regulatory networks from single-cell gene expression data using the PANDA (Passing Attributes between Networks for Data Assimilation) algorithm.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>cli, methods, irlba, igraph, RANN, Matrix, pbapply,
RhpcBLASctl</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-21 18:32:38 UTC; danie</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Osorio <a href="https://orcid.org/0000-0003-4424-8422"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Marieke L. Kuijjer
    <a href="https://orcid.org/0000-0001-6280-3130"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Osorio &lt;daniecos@uio.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-22 09:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='scorpion'>Constructs PANDA gene regulatory networks from single-cell gene expression data</h2><span id='topic+scorpion'></span>

<h3>Description</h3>

<p>Constructs gene regulatory networks from single-cell gene expression data using the PANDA (Passing Attributes between Networks for Data Assimilation) algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scorpion(
  tfMotifs = NULL,
  gexMatrix,
  ppiNet = NULL,
  nCores = 1,
  gammaValue = 10,
  nPC = 25,
  assocMethod = "pearson",
  alphaValue = 0.1,
  hammingValue = 0.001,
  nIter = Inf,
  outNet = c("regulatory", "coregulatory", "cooperative"),
  zScaling = TRUE,
  showProgress = TRUE,
  randomizationMethod = "None",
  scaleByPresent = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scorpion_+3A_tfmotifs">tfMotifs</code></td>
<td>
<p>A motif dataset, a data.frame or a matrix containing 3 columns. Each row describes an motif associated with a transcription factor (column 1) a gene (column 2) and a score (column 3) for the motif.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_gexmatrix">gexMatrix</code></td>
<td>
<p>An expression dataset, with genes in the rows and barcodes (cells) in the columns.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_ppinet">ppiNet</code></td>
<td>
<p>A Protein-Protein-Interaction dataset, a data.frame or matrix containing 3 columns. Each row describes a protein-protein interaction between transcription factor 1(column 1), transcription factor 2 (column 2) and a score (column 3) for the interaction.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_ncores">nCores</code></td>
<td>
<p>Number of processors to be used if BLAS or MPI is active.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_gammavalue">gammaValue</code></td>
<td>
<p>Graining level of data (proportion of number of single cells in the initial dataset to the number of super-cells in the final dataset)</p>
</td></tr>
<tr><td><code id="scorpion_+3A_npc">nPC</code></td>
<td>
<p>Number of principal components to use for construction of single-cell kNN network.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_assocmethod">assocMethod</code></td>
<td>
<p>Association method. Must be one of 'pearson', 'spearman' or 'pcNet'.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_alphavalue">alphaValue</code></td>
<td>
<p>Value to be used for update variable.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_hammingvalue">hammingValue</code></td>
<td>
<p>Value at which to terminate the process based on Hamming distance.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_niter">nIter</code></td>
<td>
<p>Sets the maximum number of iterations PANDA can run before exiting.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_outnet">outNet</code></td>
<td>
<p>A vector containing which networks to return. Options include &quot;regulatory&quot;, &quot;coregulatory&quot;, &quot;cooperative&quot;.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_zscaling">zScaling</code></td>
<td>
<p>Boolean to indicate use of Z-Scores in output. False will use [0,1] scale.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_showprogress">showProgress</code></td>
<td>
<p>Boolean to indicate printing of output for algorithm progress.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_randomizationmethod">randomizationMethod</code></td>
<td>
<p>Method by which to randomize gene expression matrix. Default &quot;None&quot;. Must be one of &quot;None&quot;, &quot;within.gene&quot;, &quot;by.genes&quot;. &quot;within.gene&quot; randomization scrambles each row of the gene expression matrix, &quot;by.gene&quot; scrambles gene labels.</p>
</td></tr>
<tr><td><code id="scorpion_+3A_scalebypresent">scaleByPresent</code></td>
<td>
<p>Boolean to indicate scaling of correlations by percentage of positive samples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of matrices describing networks achieved by convergence with PANDA algorithm.
</p>


<h3>Author(s)</h3>

<p>Daniel Osorio &lt;daniecos@uio.no&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Loading example data
data(scorpionTest)

# The structure of the data
str(scorpionTest)

# List of 3
# $ gex:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
# .. ..@ i       : int [1:4456] 1 5 8 11 22 30 33 34 36 38 ...
# .. ..@ p       : int [1:81] 0 47 99 149 205 258 306 342 387 423 ...
# .. ..@ Dim     : int [1:2] 230 80
# .. ..@ Dimnames:List of 2
# .. .. ..$ : chr [1:230] "MS4A1" "CD79B" "CD79A" "HLA-DRA" ...
# .. .. ..$ : chr [1:80] "ATGCCAGAACGACT" "CATGGCCTGTGCAT" "GAACCTGATGAACC" "TGACTGGATTCTCA" ...
# .. ..@ x       : num [1:4456] 1 1 3 1 1 4 1 5 1 1 ...
# .. ..@ factors : list()
# $ tf :'data.frame':	4485 obs. of  3 variables:
#   ..$ tf    : chr [1:4485] "ADNP" "ADNP" "ADNP" "AEBP2" ...
# ..$ target: chr [1:4485] "PRF1" "TMEM40" "TNFRSF1B" "CFP" ...
# ..$ mor   : num [1:4485] 1 1 1 1 1 1 1 1 1 1 ...
# $ ppi:'data.frame':	12754 obs. of  3 variables:
#   ..$ X.node1       : chr [1:12754] "ADNP" "ADNP" "ADNP" "AEBP2" ...
# ..$ node2         : chr [1:12754] "ZBTB14" "NFIA" "CDC5L" "YY1" ...
# ..$ combined_score: num [1:12754] 0.769 0.64 0.581 0.597 0.54 0.753 0.659 0.548 0.59 0.654 ...

# Running SCORPION with large alphaValue for testing purposes.
scorpionOutput &lt;- scorpion(tfMotifs = scorpionTest$tf,
                           gexMatrix = scorpionTest$gex,
                           ppiNet = scorpionTest$ppi,
                           alphaValue = 0.8)

# -- SCORPION --------------------------------------------------------------------------------------
# + Initializing and validating
# + Verified sufficient samples
# i Normalizing networks
# i Learning Network
# i Using tanimoto similarity
# + Successfully ran SCORPION on 214 Genes and 783 TFs

# Structure of the output.
str(scorpionOutput)

# List of 6
# $ regNet  :Formal class 'dgeMatrix' [package "Matrix"] with 4 slots
# .. ..@ x       : num [1:167562] -0.413 1.517 -1.311 0.364 -1.041 ...
# .. ..@ Dim     : int [1:2] 783 214
# .. ..@ Dimnames:List of 2
# .. .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...
# .. .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...
# .. ..@ factors : list()
# $ coregNet:Formal class 'dgeMatrix' [package "Matrix"] with 4 slots
# .. ..@ x       : num [1:45796] 7.07e+06 -4.06 1.76e+01 -1.16e+01 -1.62e+01 ...
# .. ..@ Dim     : int [1:2] 214 214
# .. ..@ Dimnames:List of 2
# .. .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...
# .. .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...
# .. ..@ factors : list()
# $ coopNet :Formal class 'dgeMatrix' [package "Matrix"] with 4 slots
# .. ..@ x       : num [1:613089] 5.65e+06 -5.16 -3.79 -3.63 2.94 ...
# .. ..@ Dim     : int [1:2] 783 783
# .. ..@ Dimnames:List of 2
# .. .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...
# .. .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...
# .. ..@ factors : list()
# $ numGenes: int 214
# $ numTFs  : int 783
# $ numEdges: int 167562
</code></pre>

<hr>
<h2 id='scorpionTest'>Example single-cell gene expression, motif, and ppi data</h2><span id='topic+scorpionTest'></span>

<h3>Description</h3>

<p>This data is a list containing three objects. The motif <code>data.frame</code> describes a set of pairwise connections where a specific known sequence motif of a transcription factor was found upstream of the corresponding gene. The expression <code>dgCMatrix</code> is a set of 230 gene expression levels measured across 80 cells. Finally, the ppi <code>data.frame</code> describes a set of known pairwise protein-protein interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(scorpionTest)
</code></pre>


<h3>Format</h3>

<p>A list containing three datasets.
</p>

<dl>
<dt><code>gex</code></dt><dd><p>A subsetted version of 10X Genomics' 3k PBMC dataset provided by the <code>Seurat</code> package.</p>
</dd>
<dt><code>tf</code></dt><dd><p>Subset of the transcription-factor and target gene list provided by the <code>dorothea</code> package for Homo sapiens.</p>
</dd>
<dt><code>ppi</code></dt><dd><p>The known protein-protein interactions and the combined score downloaded from the STRING database</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'># Loading example data
data(scorpionTest)

# The structure of the data
str(scorpionTest)

# List of 3
# $ gex:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
# .. ..@ i       : int [1:4456] 1 5 8 11 22 30 33 34 36 38 ...
# .. ..@ p       : int [1:81] 0 47 99 149 205 258 306 342 387 423 ...
# .. ..@ Dim     : int [1:2] 230 80
# .. ..@ Dimnames:List of 2
# .. .. ..$ : chr [1:230] "MS4A1" "CD79B" "CD79A" "HLA-DRA" ...
# .. .. ..$ : chr [1:80] "ATGCCAGAACGACT" "CATGGCCTGTGCAT" "GAACCTGATGAACC" "TGACTGGATTCTCA" ...
# .. ..@ x       : num [1:4456] 1 1 3 1 1 4 1 5 1 1 ...
# .. ..@ factors : list()
# $ tf :'data.frame':	4485 obs. of  3 variables:
#   ..$ tf    : chr [1:4485] "ADNP" "ADNP" "ADNP" "AEBP2" ...
# ..$ target: chr [1:4485] "PRF1" "TMEM40" "TNFRSF1B" "CFP" ...
# ..$ mor   : num [1:4485] 1 1 1 1 1 1 1 1 1 1 ...
# $ ppi:'data.frame':	12754 obs. of  3 variables:
#   ..$ X.node1       : chr [1:12754] "ADNP" "ADNP" "ADNP" "AEBP2" ...
# ..$ node2         : chr [1:12754] "ZBTB14" "NFIA" "CDC5L" "YY1" ...
# ..$ combined_score: num [1:12754] 0.769 0.64 0.581 0.597 0.54 0.753 0.659 0.548 0.59 0.654 ...
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
