<!DOCTYPE html><html lang="en"><head><title>Help for package scPipeline</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scPipeline}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AnnotateCellsWithSingleR'><p>Annotate cells in a Seurat object using SingleR with Celldex</p></a></li>
<li><a href='#ConvertGeneIdentifiers'><p>Convert Gene Identifiers in a Seurat Object</p></a></li>
<li><a href='#ReactomeData'><p>Reactome Data Analysis for Seurat Object</p></a></li>
<li><a href='#SeuratLowDim'><p>Create a Low dimensional Seurat object from scaled seurat object</p></a></li>
<li><a href='#SeuratMarkers'><p>A thresholded markers list for better calculation of DE genes</p></a></li>
<li><a href='#SeuratPreprocess'><p>Preprocess count data and create a Seurat object</p></a></li>
<li><a href='#TransferAnnotations'><p>Transfer annotations to Seurat clusters</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>A Wrapper for 'Seurat' and Related R Packages for End-to-End
Single Cell Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Viswanadham Sridhara
    <a href="https://orcid.org/0000-0003-0688-6140"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Viswanadham Sridhara &lt;Sridhara.Omics@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Reports markers list, differentially expressed genes, associated pathways, cell-type annotations, does batch correction and other related single cell analyses all wrapped within 'Seurat'.</td>
</tr>
<tr>
<td>Imports:</td>
<td>Seurat, batchelor, dplyr, ReactomeGSA, celldex, SingleR,
SummarizedExperiment, biomaRt, magrittr, rlang</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-05 20:54:01 UTC; vishs</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-07 11:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='AnnotateCellsWithSingleR'>Annotate cells in a Seurat object using SingleR with Celldex</h2><span id='topic+AnnotateCellsWithSingleR'></span>

<h3>Description</h3>

<p>This function annotates the cells in a Seurat object using the SingleR package
with reference data obtained from the Celldex package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AnnotateCellsWithSingleR(seurat_object, reference_data = NULL, assay = "RNA")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AnnotateCellsWithSingleR_+3A_seurat_object">seurat_object</code></td>
<td>
<p>A Seurat object to be annotated.</p>
</td></tr>
<tr><td><code id="AnnotateCellsWithSingleR_+3A_reference_data">reference_data</code></td>
<td>
<p>A reference dataset to use for annotation (e.g., HumanPrimaryCellAtlasData from Celldex).
If NULL, HumanPrimaryCellAtlasData is used by default.</p>
</td></tr>
<tr><td><code id="AnnotateCellsWithSingleR_+3A_assay">assay</code></td>
<td>
<p>The assay in the Seurat object to use for annotation. Default is &quot;RNA&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Seurat object with cell annotations added to the metadata.
</p>

<hr>
<h2 id='ConvertGeneIdentifiers'>Convert Gene Identifiers in a Seurat Object</h2><span id='topic+ConvertGeneIdentifiers'></span>

<h3>Description</h3>

<p>This function takes a Seurat object with gene identifiers as row names (e.g., RefSeq, Ensembl, Entrez)
and converts those identifiers to gene symbols (or Ensembl Gene IDs) using the biomaRt package.
The function can handle various types of gene identifiers and returns a Seurat object with updated row names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConvertGeneIdentifiers(
  seurat_object,
  id_type = "refseq",
  to_id_type = "symbol"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ConvertGeneIdentifiers_+3A_seurat_object">seurat_object</code></td>
<td>
<p>A Seurat object. The row names of the Seurat object's data or assay slot should represent gene identifiers (e.g., RefSeq, Ensembl, or Entrez IDs).</p>
</td></tr>
<tr><td><code id="ConvertGeneIdentifiers_+3A_id_type">id_type</code></td>
<td>
<p>A string specifying the type of the input gene identifiers. Options are: &quot;refseq&quot;, &quot;ensembl&quot;, &quot;entrez&quot;. Default is &quot;refseq&quot;.</p>
</td></tr>
<tr><td><code id="ConvertGeneIdentifiers_+3A_to_id_type">to_id_type</code></td>
<td>
<p>A string specifying the type of output gene identifiers. Options are: &quot;symbol&quot;, &quot;ensembl&quot;. Default is &quot;symbol&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object with updated gene names (row names) based on the specified conversion.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Read 10X counts data from matrix.mtx, barcodes.tsv and genes.tsv
library(Seurat)
counts &lt;- Read10X(data.dir = "../inst/extdata", gene.column = 1)

# Create Seurat object without batch correction
seurat_obj &lt;- SeuratPreprocess(counts)
seurat_obj &lt;- SeuratLowDim(seurat_obj)
# Convert RefSeq IDs to gene symbols
seurat_obj_converted &lt;- ConvertGeneIdentifiers(
  seurat_obj,
  id_type = "refseq",
  to_id_type = "symbol"
)

</code></pre>

<hr>
<h2 id='ReactomeData'>Reactome Data Analysis for Seurat Object</h2><span id='topic+ReactomeData'></span>

<h3>Description</h3>

<p>This function performs pathway analysis using ReactomeGSA on a Seurat object
with cluster information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReactomeData(lowdim_seurat_object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ReactomeData_+3A_lowdim_seurat_object">lowdim_seurat_object</code></td>
<td>
<p>Seurat object that has clusters information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
- GSVA result (<code>gsva_result</code>)
- Pathway expression data (<code>pathway_expression</code>)
- Max difference between pathway expression values (<code>max_difference</code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Seurat)
# Read 10X counts data from matrix.mtx, barcodes.tsv and genes.tsv
counts &lt;- Read10X(data.dir = "../inst/extdata", gene.column = 1)

# Create Seurat object without batch correction
seurat_obj &lt;- SeuratPreprocess(counts)
seurat_obj &lt;- SeuratLowDim(seurat_obj)
# Reactome Analysis
seurat_reactome &lt;- ReactomeData(seurat_obj)

</code></pre>

<hr>
<h2 id='SeuratLowDim'>Create a Low dimensional Seurat object from scaled seurat object</h2><span id='topic+SeuratLowDim'></span>

<h3>Description</h3>

<p>This function converts the transformed data to low-dimensional data for downstream analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeuratLowDim(scaled_seurat_object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SeuratLowDim_+3A_scaled_seurat_object">scaled_seurat_object</code></td>
<td>
<p>A scaled Seurat object.</p>
</td></tr>
<tr><td><code id="SeuratLowDim_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed for downstream analyses.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Seurat)
# Read 10X counts data from matrix.mtx, barcodes.tsv and genes.tsv
counts &lt;- Read10X(data.dir = "../inst/extdata", gene.column = 1)

# Create Seurat object without batch correction
seurat_obj &lt;- SeuratPreprocess(counts)
seurat_obj &lt;- SeuratLowDim(seurat_obj)

</code></pre>

<hr>
<h2 id='SeuratMarkers'>A thresholded markers list for better calculation of DE genes</h2><span id='topic+SeuratMarkers'></span>

<h3>Description</h3>

<p>This function calculates differentially expressed genes using Seurat::FindAllMarkers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeuratMarkers(lowdim_seurat_object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SeuratMarkers_+3A_lowdim_seurat_object">lowdim_seurat_object</code></td>
<td>
<p>Seurat object with cluster information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two marker lists:
- Full markers list
- Thresholded markers list with <code>min.pct = 0.1</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Seurat)
# Read 10X counts data from matrix.mtx, barcodes.tsv and genes.tsv
counts &lt;- Read10X(data.dir = "../inst/extdata", gene.column = 1)

# Create Seurat object without batch correction
seurat_obj &lt;- SeuratPreprocess(counts)
seurat_obj &lt;- SeuratLowDim(seurat_obj)
# Create Markers list
seurat_markers &lt;- SeuratMarkers(seurat_obj)

</code></pre>

<hr>
<h2 id='SeuratPreprocess'>Preprocess count data and create a Seurat object</h2><span id='topic+SeuratPreprocess'></span>

<h3>Description</h3>

<p>This function preprocesses count data, optionally applying batch correction using batchelor::fastMNN,
and creates a Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeuratPreprocess(
  counts_data,
  meta.data = NULL,
  batch_column = NULL,
  use_fastMNN = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SeuratPreprocess_+3A_counts_data">counts_data</code></td>
<td>
<p>A matrix or data frame of count data.</p>
</td></tr>
<tr><td><code id="SeuratPreprocess_+3A_meta.data">meta.data</code></td>
<td>
<p>A data frame containing metadata to include in the Seurat object. Default is NULL.</p>
</td></tr>
<tr><td><code id="SeuratPreprocess_+3A_batch_column">batch_column</code></td>
<td>
<p>A vector or factor specifying batch assignments for each cell. Default is NULL.</p>
</td></tr>
<tr><td><code id="SeuratPreprocess_+3A_use_fastmnn">use_fastMNN</code></td>
<td>
<p>Logical. Whether to apply batch correction using fastMNN. Default is FALSE.</p>
</td></tr>
<tr><td><code id="SeuratPreprocess_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to Seurat::CreateSeuratObject.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(Seurat)
# Read 10X counts data from matrix.mtx, barcodes.tsv and genes.tsv
counts &lt;- Read10X(data.dir = "../inst/extdata", gene.column = 1)

# Create Seurat object without batch correction
seurat_obj &lt;- SeuratPreprocess(counts)

</code></pre>

<hr>
<h2 id='TransferAnnotations'>Transfer annotations to Seurat clusters</h2><span id='topic+TransferAnnotations'></span>

<h3>Description</h3>

<p>This function assigns cluster-level annotations in a Seurat object based on the majority
annotation of cells within each cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TransferAnnotations(seurat_object, annotation_col, cluster_col, output_col)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TransferAnnotations_+3A_seurat_object">seurat_object</code></td>
<td>
<p>Seurat object containing cluster and annotation information.</p>
</td></tr>
<tr><td><code id="TransferAnnotations_+3A_annotation_col">annotation_col</code></td>
<td>
<p>The name of the metadata column with annotations (character string).</p>
</td></tr>
<tr><td><code id="TransferAnnotations_+3A_cluster_col">cluster_col</code></td>
<td>
<p>The name of the metadata column with cluster information (character string).</p>
</td></tr>
<tr><td><code id="TransferAnnotations_+3A_output_col">output_col</code></td>
<td>
<p>The name of the output column to store cluster annotations (character string).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Seurat object with an additional column in its metadata, specified by <code>output_col</code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
