<!DOCTYPE html><html><head><title>Help for package scPloidy</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scPloidy}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#fragmentoverlapcount'><p>Count Overlap of ATAC-seq Fragments</p></a></li>
<li><a href='#ploidy'><p>Infer Ploidy from ATAC-seq Fragment Overlap</p></a></li>
<li><a href='#SHR_m154211'><p>Liver Cells from a Rat</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Infer Ploidy of Single Cells</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Compute ploidy of single cells (or nuclei)
    based on single-cell (or single-nucleus)
    ATAC-seq (Assay for Transposase-Accessible Chromatin using sequencing)
    data <a href="https://github.com/fumi-github/scPloidy">https://github.com/fumi-github/scPloidy</a>.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/fumi-github/scPloidy/issues">https://github.com/fumi-github/scPloidy/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, GenomicRanges, magrittr, matrixStats, mixtools, rlang,
Rsamtools, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>IRanges, knitr, readr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-09 07:45:55 UTC; fumi</td>
</tr>
<tr>
<td>Author:</td>
<td>Fumihiko Takeuchi <a href="https://orcid.org/0000-0003-3185-5661"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fumihiko Takeuchi &lt;fumihiko@takeuchi.name&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-12 07:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='fragmentoverlapcount'>Count Overlap of ATAC-seq Fragments</h2><span id='topic+fragmentoverlapcount'></span>

<h3>Description</h3>

<p>Count Overlap of ATAC-seq Fragments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fragmentoverlapcount(
  file,
  targetregions,
  excluderegions = NULL,
  targetbarcodes = NULL,
  Tn5offset = c(1, 0)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fragmentoverlapcount_+3A_file">file</code></td>
<td>
<p>Filename of the file for ATAC-seq fragments.
The file must be block gzipped (using the <code>bgzip</code> command)
and accompanied with the index file (made using the <code>tabix</code> command).
The uncompressed file must be a tab delimited file,
where each row represents one fragment.
The first four columns are chromosome name, start position, end position,
and barcode (i.e., name) of the cell including the fragment.
The remaining columns are ignored.
See vignette for details.</p>
</td></tr>
<tr><td><code id="fragmentoverlapcount_+3A_targetregions">targetregions</code></td>
<td>
<p>GRanges object for the regions where overlaps are counted.
Usually all of the autosomes.
If there is memory problem, split a chromosome into smaller chunks,
for example by 10 Mb.
The function loads each element of <code>targetregions</code> sequentially,
and smaller elements require less memory.</p>
</td></tr>
<tr><td><code id="fragmentoverlapcount_+3A_excluderegions">excluderegions</code></td>
<td>
<p>GRanges object for the regions to be excluded.
Simple repeats in the genome should be listed here,
because repeats can cause false overlaps.
A fragment is discarded if its 5' or 3' end is located in <code>excluderegions</code>.
If <code>NULL</code>, fragments are not excluded by this criterion.</p>
</td></tr>
<tr><td><code id="fragmentoverlapcount_+3A_targetbarcodes">targetbarcodes</code></td>
<td>
<p>Character vector for the barcodes of cells to be analyzed,
such as those passing quality control.
If <code>NULL</code>, all barcodes in the input file are analyzed.</p>
</td></tr>
<tr><td><code id="fragmentoverlapcount_+3A_tn5offset">Tn5offset</code></td>
<td>
<p>Numeric vector of length two.
The enzyme for ATAC-seq is a homodimer of Tn5.
The transposition sites of two Tn5 proteins are 9 bp apart,
and the (representative) site of accessibility is in between.
If the start and end position of your input file is taken from BAM file,
set the paramater to <code>c(4, -5)</code> to adjust the offset.
Alternatively, values such as <code>c(0, -9)</code> could generate similar results;
what matters the most is the difference between the two numbers.
The fragments.tsv.gz file generated by 10x Cell Ranger already adjusts the shift
but is recorded as a BED file. In this case, use <code>c(1, 0)</code> (default value).
If unsure, set to <code>"guess"</code>,
in which case the program returns a guess.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with each row corresponding to a cell.
For each cell, its barcode, the total count of the fragments <code>nfrag</code>,
and the count distinguished by overlap depth are given.
</p>

<hr>
<h2 id='ploidy'>Infer Ploidy from ATAC-seq Fragment Overlap</h2><span id='topic+ploidy'></span>

<h3>Description</h3>

<p>Infer Ploidy from ATAC-seq Fragment Overlap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ploidy(fragmentoverlap, levels, s = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ploidy_+3A_fragmentoverlap">fragmentoverlap</code></td>
<td>
<p>Frequency of fragment overlap in each cell
computed by the function <code>fragmentoverlapcount</code>.</p>
</td></tr>
<tr><td><code id="ploidy_+3A_levels">levels</code></td>
<td>
<p>Possible values of ploidy. For example,
<code>c(2, 4)</code> if the cells can be diploids or tetraploids.
The values must be larger than one.</p>
</td></tr>
<tr><td><code id="ploidy_+3A_s">s</code></td>
<td>
<p>Seed for random numbers used in EM algorithm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with each row corresponding to a cell.
For each cell, its barcode, ploidy inferred by moment method,
the same with additional K-means clustering,
and ploidy inferred by EM algorithm of mixture are given.
I recommend using <code>ploidy.moment</code>.
</p>

<hr>
<h2 id='SHR_m154211'>Liver Cells from a Rat</h2><span id='topic+SHR_m154211'></span>

<h3>Description</h3>

<p>The dataset includes 3572 nuclei obtained from the liver of
a 16 weeks old male rat, which was fed normal diet.
Overlapping of single-nucleus ATAC-seq fragments was computed with the
<code>fragmentoverlapcount</code> function and saved as <code>fragmentoverlap</code>.
The cell type of the nuclei are saved in the data.frame <code>cells</code>.
The data for rat SHR_m154211 was taken from the publication cited below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SHR_m154211)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>


<h3>Source</h3>

<p>Takeuchi et al. (2022) bioRxiv <a href="https://doi.org/10.1101/2022.07.12.499681">doi:10.1101/2022.07.12.499681</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SHR_m154211)
fragmentoverlap = SHR_m154211$fragmentoverlap
p = ploidy(fragmentoverlap, c(2, 4, 8))
head(p)
cells = SHR_m154211$cells
table(cells$celltype, p$ploidy.moment[match(cells$barcode, p$barcode)])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
