<!DOCTYPE html><html lang="en"><head><title>Help for package SCpubr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SCpubr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#do_AlluvialPlot'><p>Generate Alluvial plots.</p></a></li>
<li><a href='#do_BarPlot'><p>Create Bar Plots.</p></a></li>
<li><a href='#do_BeeSwarmPlot'><p>BeeSwarm plot.</p></a></li>
<li><a href='#do_BoxPlot'><p>Generate Box Plots.</p></a></li>
<li><a href='#do_CellularStatesPlot'><p>Cellular States plot.</p></a></li>
<li><a href='#do_ChordDiagramPlot'><p>Generate a Chord diagram.</p></a></li>
<li><a href='#do_ColorPalette'><p>Generate color scales based on a value.</p></a></li>
<li><a href='#do_CopyNumberVariantPlot'><p>Display CNV scores from inferCNV as Feature Plots.</p></a></li>
<li><a href='#do_CorrelationPlot'><p>Create correlation matrix heatmaps.</p></a></li>
<li><a href='#do_DimPlot'><p>Wrapper for DimPlot.</p></a></li>
<li><a href='#do_DotPlot'><p>This function is a wrapper for DotPlot. It provides most of its functionalities while adding extra.</p>
You can</a></li>
<li><a href='#do_EnrichmentHeatmap'><p>Create enrichment scores heatmaps.</p></a></li>
<li><a href='#do_ExpressionHeatmap'><p>Create heatmaps of averaged expression by groups.</p></a></li>
<li><a href='#do_FeaturePlot'><p>Wrapper for FeaturePlot.</p></a></li>
<li><a href='#do_FunctionalAnnotationPlot'><p>Compute functional annotation plots using GO or KEGG ontologies</p></a></li>
<li><a href='#do_GeyserPlot'><p>Generate a Geyser plot.</p></a></li>
<li><a href='#do_GroupedGOTermPlot'><p>Compute an overview of the GO terms associated with the input list of genes.</p></a></li>
<li><a href='#do_GroupwiseDEPlot'><p>Compute a heatmap with the results of a group-wise DE analysis.</p></a></li>
<li><a href='#do_NebulosaPlot'><p>Wrapper for Nebulosa::plot_density in Seurat.</p></a></li>
<li><a href='#do_PathwayActivityPlot'><p>Plot Pathway Activities from decoupleR using Progeny prior knowledge.</p></a></li>
<li><a href='#do_RidgePlot'><p>Create ridge plots.</p></a></li>
<li><a href='#do_TermEnrichmentPlot'><p>Display the enriched terms for a given list of genes.</p></a></li>
<li><a href='#do_TFActivityPlot'><p>Plot TF Activities from decoupleR using Dorothea prior knowledge.</p></a></li>
<li><a href='#do_ViolinPlot'><p>Wrapper for VlnPlot.</p></a></li>
<li><a href='#do_VolcanoPlot'><p>Compute a Volcano plot out of DE genes.</p></a></li>
<li><a href='#doc_function'><p>Mock function used to document all main function.</p></a></li>
<li><a href='#human_chr_locations'><p>Chromosome arm locations for human genome GRCh38.</p></a></li>
<li><a href='#named_list'><p>Named list.</p></a></li>
<li><a href='#named_vector'><p>Named vector.</p></a></li>
<li><a href='#package_report'><p>Generate a status report of SCpubr and its dependencies.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Generate Publication Ready Visualizations of Single Cell
Transcriptomics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>A system that provides a streamlined way of generating
    publication ready plots for known Single-Cell transcriptomics data in
    a “publication ready” format. This is, the goal is to automatically
    generate plots with the highest quality possible, that can be used
    right away or with minimal modifications for a research article.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/enblacar/SCpubr/">https://github.com/enblacar/SCpubr/</a>,
<a href="https://enblacar.github.io/SCpubr-book/">https://enblacar.github.io/SCpubr-book/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/enblacar/SCpubr/issues/">https://github.com/enblacar/SCpubr/issues/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>AnnotationDbi, assertthat, AUCell, circlize, cli, cluster,
clusterProfiler, colorspace, ComplexHeatmap, covr, decoupleR,
dplyr (&ge; 1.1.0), enrichplot, forcats, ggalluvial, ggbeeswarm,
ggdist, ggExtra, ggh4x, ggnewscale, ggplot2 (&ge; 3.4.0),
ggplotify, ggrastr, ggrepel, ggridges, ggsignif, graphics,
infercnv, knitr, labeling, magrittr, MASS, Matrix, methods,
Nebulosa, org.Hs.eg.db, patchwork, pheatmap, plyr, purrr, qpdf,
RColorBrewer, rjags, rlang, rmarkdown, scales, scattermore,
Seurat, SeuratObject, sf, stringr, svglite, testthat (&ge;
3.0.0), tibble, tidyr, UCell, viridis, withr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Software, SingleCell, Visualization</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-11 09:31:17 UTC; e420a</td>
</tr>
<tr>
<td>Author:</td>
<td>Enrique Blanco-Carmona
    <a href="https://orcid.org/0000-0002-1208-1691"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Enrique Blanco-Carmona &lt;scpubr@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-11 09:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='do_AlluvialPlot'>Generate Alluvial plots.</h2><span id='topic+do_AlluvialPlot'></span>

<h3>Description</h3>

<p>This function is based on the <span class="pkg">ggalluvial</span> package. It allows you to generate alluvial plots from a given Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_AlluvialPlot(
  sample,
  first_group,
  last_group,
  middle_groups = NULL,
  colors.use = NULL,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  font.size = 14,
  font.type = "sans",
  xlab = NULL,
  ylab = "Number of cells",
  repel = FALSE,
  fill.by = last_group,
  use_labels = FALSE,
  stratum.color = "black",
  stratum.fill = "white",
  stratum.width = 1/3,
  stratum.fill.conditional = FALSE,
  use_geom_flow = FALSE,
  alluvium.color = "white",
  flow.color = "white",
  flip = FALSE,
  label.color = "black",
  curve_type = "sigmoid",
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  plot.grid = FALSE,
  grid.color = "grey75",
  grid.type = "dashed",
  na.value = "white",
  legend.position = "right",
  legend.title = NULL,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_AlluvialPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_first_group">first_group</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Categorical metadata variable. First group of nodes of the alluvial plot.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_last_group">last_group</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Categorical metadata variable. Last group of nodes of the alluvial plot.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_middle_groups">middle_groups</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Categorical metadata variable. Vector of groups of nodes of the alluvial plot.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Named list of colors corresponding to the unique values in fill.by (which defaults to last_group).</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_plot.title">plot.title</code>, <code id="do_AlluvialPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_AlluvialPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_xlab">xlab</code>, <code id="do_AlluvialPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_repel">repel</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to repel the text labels.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_fill.by">fill.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of first_group, middle_groups (one of the values, if multiple mid_groups) or last_group. These values will be used to color the alluvium/flow.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_use_labels">use_labels</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use labels instead of text for the stratum.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_stratum.color">stratum.color</code>, <code id="do_AlluvialPlot_+3A_alluvium.color">alluvium.color</code>, <code id="do_AlluvialPlot_+3A_flow.color">flow.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the alluvium (and flow) and stratum.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_stratum.fill">stratum.fill</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color to fill the stratum.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_stratum.width">stratum.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Width of the stratum.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_stratum.fill.conditional">stratum.fill.conditional</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to fill the stratum with the same colors as the alluvium/flow.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_use_geom_flow">use_geom_flow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use <code><a href="ggalluvial.html#topic+geom_flow">geom_flow</a></code> instead of <code><a href="ggalluvial.html#topic+geom_alluvium">geom_alluvium</a></code>. Visual results might differ.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_label.color">label.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the text labels.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_curve_type">curve_type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of curve used in <code><a href="ggalluvial.html#topic+geom_alluvium">geom_alluvium</a></code>. One of:
</p>

<ul>
<li> <p><em><code>linear</code></em>.
</p>
</li>
<li> <p><em><code>cubic</code></em>.
</p>
</li>
<li> <p><em><code>quintic</code></em>.
</p>
</li>
<li> <p><em><code>sine</code></em>.
</p>
</li>
<li> <p><em><code>arctangent</code></em>.
</p>
</li>
<li> <p><em><code>sigmoid</code></em>.
</p>
</li>
<li> <p><em><code>xspline</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_plot.grid">plot.grid</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot grid lines.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_grid.type">grid.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of the possible linetype options:
</p>

<ul>
<li> <p><em><code>blank</code></em>.
</p>
</li>
<li> <p><em><code>solid</code></em>.
</p>
</li>
<li> <p><em><code>dashed</code></em>.
</p>
</li>
<li> <p><em><code>dotted</code></em>.
</p>
</li>
<li> <p><em><code>dotdash</code></em>.
</p>
</li>
<li> <p><em><code>longdash</code></em>.
</p>
</li>
<li> <p><em><code>twodash</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_AlluvialPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_AlluvialPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_AlluvialPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_AlluvialPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_AlluvialPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_AlluvialPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_AlluvialPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_AlluvialPlot", passive = TRUE)
  message(value)
  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Compute basic sankey plot.
    p &lt;- SCpubr::do_AlluvialPlot(sample = sample,
                                 first_group = "orig.ident",
                                 last_group = "seurat_clusters")

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_BarPlot'>Create Bar Plots.</h2><span id='topic+do_BarPlot'></span>

<h3>Description</h3>

<p>Create Bar Plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_BarPlot(
  sample,
  group.by,
  order = FALSE,
  add.n = FALSE,
  add.n.face = "bold",
  add.n.expand = c(0, 1.15),
  add.n.size = 4,
  order.by = NULL,
  split.by = NULL,
  facet.by = NULL,
  position = "stack",
  font.size = 14,
  font.type = "sans",
  legend.position = "bottom",
  legend.title = NULL,
  legend.ncol = NULL,
  legend.nrow = NULL,
  legend.byrow = FALSE,
  axis.text.x.angle = 45,
  xlab = NULL,
  ylab = NULL,
  colors.use = NULL,
  flip = FALSE,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  plot.grid = FALSE,
  grid.color = "grey75",
  grid.type = "dashed",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain",
  strip.text.face = "bold",
  return_data = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_BarPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata column to compute the counts of. Has to be either a character or factor column.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_order">order</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to order the results in descending order of counts.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_add.n">add.n</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to add the total counts on top of each bar.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_add.n.face">add.n.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Font face of the labels added by <strong><code>add.n</code></strong>.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_add.n.expand">add.n.expand</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Vector of two numerics representing the start and end of the scale. Minimum should be 0 and max should be above 1. This basically expands the Y axis so that the labels fit when <strong><code>flip = TRUE</code></strong>.
</p>

<ul>
<li> <p><em><code>stack</code></em>: Set the bars side by side, displaying the total number of counts. Uses <a href="ggplot2.html#topic+position_stack">position_stack</a>.
</p>
</li>
<li> <p><em><code>fill</code></em>: Set the bars on top of each other, displaying the proportion of counts from the total that each group represents. Uses <a href="ggplot2.html#topic+position_fill">position_fill</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_add.n.size">add.n.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the labels</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_order.by">order.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | When <strong><code>split.by</code></strong> is used, value of <strong><code>group.by</code></strong> to reorder the columns based on its value.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata column to split the values of group.by by. If not used, defaults to the active idents.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_facet.by">facet.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata column to gather the columns by. This is useful if you have other overarching metadata.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_position">position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position function from <span class="pkg">ggplot2</span>. Either stack or fill.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_legend.nrow">legend.nrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of rows in the legend.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the legend is filled by row or not.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_xlab">xlab</code>, <code id="do_BarPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of valid color representations (either name of HEX codes) with as many named colors as unique values of group.by. If group.by is not provided, defaults to the unique values of <a href="Seurat.html#topic+Idents">Idents</a>. If not provided, a color scale will be set by default.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_plot.title">plot.title</code>, <code id="do_BarPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_BarPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_plot.grid">plot.grid</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot grid lines.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_grid.type">grid.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of the possible linetype options:
</p>

<ul>
<li> <p><em><code>blank</code></em>.
</p>
</li>
<li> <p><em><code>solid</code></em>.
</p>
</li>
<li> <p><em><code>dashed</code></em>.
</p>
</li>
<li> <p><em><code>dotted</code></em>.
</p>
</li>
<li> <p><em><code>dotdash</code></em>.
</p>
</li>
<li> <p><em><code>longdash</code></em>.
</p>
</li>
<li> <p><em><code>twodash</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_BarPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_BarPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_BarPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_BarPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_BarPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_BarPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_strip.text.face">strip.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the strip text.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BarPlot_+3A_return_data">return_data</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Returns a data.frame with the count and proportions displayed in the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object containing a Bar plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_BarPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Basic bar plot, horizontal.
    p1 &lt;- SCpubr::do_BarPlot(sample = sample,
                             group.by = "seurat_clusters",
                             legend.position = "none",
                             plot.title = "Number of cells per cluster")

    # Split by a second variable.
    sample$modified_orig.ident &lt;- sample(x = c("Sample_A", "Sample_B", "Sample_C"),
                                         size = ncol(sample),
                                         replace = TRUE,
                                         prob = c(0.2, 0.7, 0.1))

    p &lt;- SCpubr::do_BarPlot(sample,
                            group.by = "seurat_clusters",
                            split.by = "modified_orig.ident",
                            plot.title = "Number of cells per cluster in each sample",
                            position = "stack")


  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_BeeSwarmPlot'>BeeSwarm plot.</h2><span id='topic+do_BeeSwarmPlot'></span>

<h3>Description</h3>

<p>BeeSwarm plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_BeeSwarmPlot(
  sample,
  feature_to_rank,
  group.by = NULL,
  assay = NULL,
  reduction = NULL,
  slot = NULL,
  continuous_feature = FALSE,
  order = FALSE,
  colors.use = NULL,
  legend.title = NULL,
  legend.type = "colorbar",
  legend.position = "bottom",
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.length = 20,
  legend.width = 1,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.ncol = NULL,
  legend.icon.size = 4,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  xlab = NULL,
  ylab = NULL,
  font.size = 14,
  font.type = "sans",
  remove_x_axis = FALSE,
  remove_y_axis = FALSE,
  flip = FALSE,
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = 1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  verbose = TRUE,
  raster = FALSE,
  raster.dpi = 300,
  plot_cell_borders = TRUE,
  border.size = 1.5,
  border.color = "black",
  pt.size = 2,
  min.cutoff = NA,
  max.cutoff = NA,
  na.value = "grey75",
  number.breaks = 5,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_BeeSwarmPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_feature_to_rank">feature_to_rank</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Feature for which the cells are going to be ranked. Ideal case is that this feature is stored as a metadata column.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_reduction">reduction</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Reduction to use. Can be the canonical ones such as &quot;umap&quot;, &quot;pca&quot;, or any custom ones, such as &quot;diffusion&quot;. If you are unsure about which reductions you have, use <code>Seurat::Reductions(sample)</code>. Defaults to &quot;umap&quot; if present or to the last computed reduction if the argument is not provided.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_continuous_feature">continuous_feature</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Is the feature to rank and color for continuous? I.e: an enrichment score.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_order">order</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to reorder the groups based on the median of the ranking.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of valid color representations (either name of HEX codes) with as many named colors as unique values of group.by. If group.by is not provided, defaults to the unique values of <a href="Seurat.html#topic+Idents">Idents</a>. If not provided, a color scale will be set by default.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_BeeSwarmPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.length">legend.length</code>, <code id="do_BeeSwarmPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_legend.icon.size">legend.icon.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the icons in legend.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_plot.title">plot.title</code>, <code id="do_BeeSwarmPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_BeeSwarmPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_xlab">xlab</code>, <code id="do_BeeSwarmPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_remove_x_axis">remove_x_axis</code>, <code id="do_BeeSwarmPlot_+3A_remove_y_axis">remove_y_axis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Remove X axis labels and ticks from the plot.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_verbose">verbose</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show extra comments, warnings,etc.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_raster">raster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to raster the resulting plot. This is recommendable if plotting a lot of cells.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Pixel resolution for rasterized plots. Defaults to 1024. Only activates on Seurat versions higher or equal than 4.1.0.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_plot_cell_borders">plot_cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot border around cells.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_min.cutoff">min.cutoff</code>, <code id="do_BeeSwarmPlot_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_BeeSwarmPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_BeeSwarmPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_BeeSwarmPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_BeeSwarmPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_BeeSwarmPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_BeeSwarmPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_BeeSwarmPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object containing a Bee Swarm plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_BeeSwarmPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Basic Bee Swarm plot - categorical coloring.
    # This will color based on the unique values of seurat_clusters.
    p &lt;- SCpubr::do_BeeSwarmPlot(sample = sample,
                                 feature_to_rank = "PC_1",
                                 group.by = "seurat_clusters",
                                 continuous_feature = FALSE)

    # Basic Bee Swarm plot - continuous coloring.
    # This will color based on the PC_1 values.
    p &lt;- SCpubr::do_BeeSwarmPlot(sample = sample,
                                 feature_to_rank = "PC_1",
                                 group.by = "seurat_clusters",
                                 continuous_feature = TRUE)
  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }


</code></pre>

<hr>
<h2 id='do_BoxPlot'>Generate Box Plots.</h2><span id='topic+do_BoxPlot'></span>

<h3>Description</h3>

<p>Generate Box Plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_BoxPlot(
  sample,
  feature,
  group.by = NULL,
  split.by = NULL,
  assay = NULL,
  slot = "data",
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  colors.use = NULL,
  na.value = "grey75",
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  xlab = NULL,
  ylab = NULL,
  legend.title = NULL,
  legend.title.position = "top",
  legend.position = "bottom",
  boxplot.line.color = "black",
  outlier.color = "black",
  outlier.alpha = 0.5,
  boxplot.linewidth = 0.5,
  boxplot.width = NULL,
  plot.grid = TRUE,
  grid.color = "grey75",
  grid.type = "dashed",
  flip = FALSE,
  order = FALSE,
  use_silhouette = FALSE,
  use_test = FALSE,
  comparisons = NULL,
  test = "wilcox.test",
  map_signif_level = TRUE,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_BoxPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_feature">feature</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Feature to represent.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of valid color representations (either name of HEX codes) with as many named colors as unique values of group.by. If group.by is not provided, defaults to the unique values of <a href="Seurat.html#topic+Idents">Idents</a>. If not provided, a color scale will be set by default.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_plot.title">plot.title</code>, <code id="do_BoxPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_BoxPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_xlab">xlab</code>, <code id="do_BoxPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_legend.title.position">legend.title.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position for the title of the legend. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the legend.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the legend.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the legend.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_boxplot.line.color">boxplot.line.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the borders of the boxplots if use_silhouette is FALSE.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_outlier.color">outlier.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the outlier dots.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_outlier.alpha">outlier.alpha</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Alpha applied to the outliers.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_boxplot.linewidth">boxplot.linewidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines in the boxplots. Also controls the lines of the tests applied if use_test is set to true.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_boxplot.width">boxplot.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the boxplots.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_plot.grid">plot.grid</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot grid lines.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_grid.type">grid.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of the possible linetype options:
</p>

<ul>
<li> <p><em><code>blank</code></em>.
</p>
</li>
<li> <p><em><code>solid</code></em>.
</p>
</li>
<li> <p><em><code>dashed</code></em>.
</p>
</li>
<li> <p><em><code>dotted</code></em>.
</p>
</li>
<li> <p><em><code>dotdash</code></em>.
</p>
</li>
<li> <p><em><code>longdash</code></em>.
</p>
</li>
<li> <p><em><code>twodash</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_order">order</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to order the boxplots by average values. Can not be used alongside split.by.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_use_silhouette">use_silhouette</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to color the borders of the boxplots instead of the inside area.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_use_test">use_test</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to apply a statistical test to a given pair of elements. Can not be used alongside split.by.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_comparisons">comparisons</code></td>
<td>
<p>A list of length-2 vectors. The entries in the vector are
either the names of 2 values on the x-axis or the 2 integers that
correspond to the index of the columns of interest.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_test">test</code></td>
<td>
<p>the name of the statistical test that is applied to the values of
the 2 columns (e.g. <code>t.test</code>, <code>wilcox.test</code> etc.). If you implement a
custom test make sure that it returns a list that has an entry called
<code>p.value</code>.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_map_signif_level">map_signif_level</code></td>
<td>
<p>Boolean value, if the p-value are directly written as
annotation or asterisks are used instead. Alternatively one can provide a
named numeric vector to create custom mappings from p-values to annotation:
For example: <code>c("***"=0.001, "**"=0.01, "*"=0.05)</code>.
Alternatively, one can provide a function that takes a numeric argument
(the p-value) and returns a string.</p>
</td></tr>
<tr><td><code id="do_BoxPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_BoxPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_BoxPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_BoxPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_BoxPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_BoxPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_BoxPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_BoxPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Basic box plot.
    p &lt;- SCpubr::do_BoxPlot(sample = sample,
                            feature = "nCount_RNA")
    p

    # Use silhouette style.
    p &lt;- SCpubr::do_BoxPlot(sample = sample,
                            feature = "nCount_RNA",
                            use_silhouette = TRUE)
    p

    # Order by mean values.
    p &lt;- SCpubr::do_BoxPlot(sample = sample,
                            feature = "nCount_RNA",
                            order = TRUE)
    p

    # Apply second grouping.
    sample$orig.ident &lt;- ifelse(sample$seurat_clusters %in% c("0", "1", "2", "3"), "A", "B")
    p &lt;- SCpubr::do_BoxPlot(sample = sample,
                            feature = "nCount_RNA",
                            split.by = "orig.ident")
    p

    # Apply statistical tests.
    p &lt;- SCpubr::do_BoxPlot(sample = sample,
                            feature = "nCount_RNA",
                            group.by = "orig.ident",
                            use_test = TRUE,
                            comparisons = list(c("A", "B")))
    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_CellularStatesPlot'>Cellular States plot.</h2><span id='topic+do_CellularStatesPlot'></span>

<h3>Description</h3>

<p>This plot aims to show the relationships between distinct enrichment scores. If 3 variables are provided, the relationship is between the Y axis and the dual X axis.
If 4 variables are provided, each corner of the plot represents how enriched the cells are in that given list. How to interpret this? In a 3-variable plot, the Y axis
just means one variable. The higher the cells are in the Y axis the more enriched they are in that given variable. The X axis is a dual parameter one. Cells falling
into each extreme of the axis are highly enriched for either x1 or x2, while cells falling in between are not enriched for any of the two. In a 4-variable plot, each corner
shows the enrichment for one of the 4 given features. Cells will tend to locate in either of the four corners, but there will be cases of cells locating mid-way between two
given corners (enriched in both features) or in the middle of the plot (not enriched for any).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_CellularStatesPlot(
  sample,
  input_gene_list,
  x1,
  y1,
  x2 = NULL,
  y2 = NULL,
  group.by = NULL,
  colors.use = NULL,
  legend.position = "bottom",
  legend.icon.size = 4,
  legend.ncol = NULL,
  legend.nrow = NULL,
  legend.byrow = FALSE,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  font.size = 14,
  font.type = "sans",
  xlab = NULL,
  ylab = NULL,
  axis.ticks = TRUE,
  axis.text = TRUE,
  verbose = FALSE,
  enforce_symmetry = FALSE,
  plot_marginal_distributions = FALSE,
  marginal.type = "density",
  marginal.size = 5,
  marginal.group = TRUE,
  plot_cell_borders = TRUE,
  plot_enrichment_scores = FALSE,
  border.size = 2,
  border.color = "black",
  pt.size = 2,
  raster = FALSE,
  raster.dpi = 1024,
  plot_features = FALSE,
  features = NULL,
  use_viridis = TRUE,
  viridis.palette = "G",
  viridis.direction = 1,
  sequential.palette = "YlGnBu",
  sequential.direction = -1,
  nbin = 24,
  ctrl = 100,
  number.breaks = 5,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_CellularStatesPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_input_gene_list">input_gene_list</code></td>
<td>
<p><strong><code><a href="#topic+named_list">named_list</a></code></strong> | Named list of lists of genes to be used as input.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_x1">x1</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A name of a list from input_gene_list. First feature in the X axis. Will go on the right side of the X axis if y2 is not provided and top-right quadrant if provided.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_y1">y1</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A name of a list from input_gene_list. First feature on the Y axis. Will become the Y axis if y2 is not provided and bottom-right quadrant if provided.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_x2">x2</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A name of a list from input_gene_list. Second feature on the X axis. Will go on the left side of the X axis if y2 is not provided and top-left quadrant if provided.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_y2">y2</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A name of a list from input_gene_list. Second feature on the Y axis. Will become the bottom-left quadrant if provided.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of valid color representations (either name of HEX codes) with as many named colors as unique values of group.by. If group.by is not provided, defaults to the unique values of <a href="Seurat.html#topic+Idents">Idents</a>. If not provided, a color scale will be set by default.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_legend.icon.size">legend.icon.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the icons in legend.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_legend.nrow">legend.nrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of rows in the legend.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the legend is filled by row or not.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_plot.title">plot.title</code>, <code id="do_CellularStatesPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_CellularStatesPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_xlab">xlab</code>, <code id="do_CellularStatesPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_axis.ticks">axis.ticks</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show axis ticks.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_axis.text">axis.text</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show axis text.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_verbose">verbose</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show extra comments, warnings,etc.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to enforce the plot to follow a symmetry (3 variables, the X axis has 0 as center, 4 variables, all axis have the same range and the plot is squared).</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_plot_marginal_distributions">plot_marginal_distributions</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> |  Whether to plot marginal distributions on the figure or not.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_marginal.type">marginal.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of:
</p>

<ul>
<li> <p><em><code>density</code></em>: Compute density plots on the margins.
</p>
</li>
<li> <p><em><code>histogram</code></em>: Compute histograms on the margins.
</p>
</li>
<li> <p><em><code>boxplot</code></em>: Compute boxplot on the margins.
</p>
</li>
<li> <p><em><code>violin</code></em>: Compute violin plots on the margins.
</p>
</li>
<li> <p><em><code>densigram</code></em>: Compute densigram plots on the margins.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_marginal.size">marginal.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size ratio between the main and marginal plots. A value of 5 means that the main plot is 5 times bigger than the marginal plots.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_marginal.group">marginal.group</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to group the marginal distribution by group.by or current identities.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_plot_cell_borders">plot_cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot border around cells.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_plot_enrichment_scores">plot_enrichment_scores</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to report enrichment scores for the input lists as plots.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_raster">raster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to raster the resulting plot. This is recommendable if plotting a lot of cells.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Pixel resolution for rasterized plots. Defaults to 1024. Only activates on Seurat versions higher or equal than 4.1.0.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_plot_features">plot_features</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to also report any other feature onto the primary plot.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_features">features</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Additional features to plot.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_nbin">nbin</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of bins to use in <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a>.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_ctrl">ctrl</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of genes in the control set to use in <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a>.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_CellularStatesPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_CellularStatesPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_CellularStatesPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_CellularStatesPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_CellularStatesPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_CellularStatesPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_CellularStatesPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This plots are based on the following publications:
</p>

<ul>
<li><p> Neftel, C. <em>et al</em>. An Integrative Model of Cellular States, Plasticity, and Genetics for Glioblastoma. Cell 178, 835-849.e21 (2019). <a href="https://doi.org/10.1016/j.cell.2019.06.024">doi:10.1016/j.cell.2019.06.024</a>
</p>
</li>
<li><p> Tirosh, I., Venteicher, A., Hebert, C. <em>et al</em>. Single-cell RNA-seq supports a developmental hierarchy in human oligodendroglioma. Nature 539, 309–313 (2016). <a href="https://doi.org/10.1038/nature20123">doi:10.1038/nature20123</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A ggplot2 object containing a butterfly plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_CellularStatesPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Define some gene sets to query. It has to be a named list.
    gene_set &lt;- list("A" = rownames(sample)[1:10],
                     "B" = rownames(sample)[11:20],
                     "C" = rownames(sample)[21:30],
                     "D" = rownames(sample)[31:40])

    # Using two variables: A scatter plot X vs Y.
    p &lt;- SCpubr::do_CellularStatesPlot(sample = sample,
                                       input_gene_list = gene_set,
                                       x1 = "A",
                                       y1 = "B",
                                       nbin = 1,
                                       ctrl = 10)
    p


    # Using three variables. Figure from: https://www.nature.com/articles/nature20123.
    p &lt;- SCpubr::do_CellularStatesPlot(sample = sample,
                                       input_gene_list = gene_set,
                                       x1 = "A",
                                       y1 = "B",
                                       x2 = "C",
                                       nbin = 1,
                                       ctrl = 10)
    p


    # Using four variables. Figure from: https://pubmed.ncbi.nlm.nih.gov/31327527/
    p &lt;- SCpubr::do_CellularStatesPlot(sample = sample,
                                       input_gene_list = gene_set,
                                       x1 = "A",
                                       y1 = "C",
                                       x2 = "B",
                                       y2 = "D",
                                       nbin = 1,
                                       ctrl = 10)
    p
  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }


</code></pre>

<hr>
<h2 id='do_ChordDiagramPlot'>Generate a Chord diagram.</h2><span id='topic+do_ChordDiagramPlot'></span>

<h3>Description</h3>

<p>Generate a Chord diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_ChordDiagramPlot(
  sample = NULL,
  from = NULL,
  to = NULL,
  colors.from = NULL,
  colors.to = NULL,
  big.gap = 10,
  small.gap = 1,
  link.border.color = NA,
  link.border.width = 1,
  highlight_group = NULL,
  alpha.highlight = 25,
  link.sort = NULL,
  link.decreasing = TRUE,
  z_index = FALSE,
  self.link = 1,
  symmetric = FALSE,
  directional = 1,
  direction.type = c("diffHeight", "arrows"),
  link.arr.type = "big.arrow",
  scale = FALSE,
  alignment = "default",
  annotationTrack = c("grid", "axis"),
  padding_labels = 4,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_ChordDiagramPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_from">from</code>, <code id="do_ChordDiagramPlot_+3A_to">to</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Categorical metadata variable to be used as origin and end points of the interactions.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_colors.from">colors.from</code>, <code id="do_ChordDiagramPlot_+3A_colors.to">colors.to</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of colors corresponding to the unique values of &quot;from&quot; and &quot;to&quot;.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_big.gap">big.gap</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Space between the groups in &quot;from&quot; and &quot;to&quot;.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_small.gap">small.gap</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Space within the groups.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_link.border.color">link.border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the links. NA = no color.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_link.border.width">link.border.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border line of the links.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_highlight_group">highlight_group</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A value from from that will be used to highlight only the links coming from it.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_alpha.highlight">alpha.highlight</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | A value between 00 (double digits) and 99 to depict the alpha of the highlighted links. No transparency needs &quot;FF&quot;</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_link.sort">link.sort</code></td>
<td>
<p>pass to <code><a href="circlize.html#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="circlize.html#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_link.decreasing">link.decreasing</code></td>
<td>
<p>pass to <code><a href="circlize.html#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="circlize.html#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_z_index">z_index</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to bring the bigger links to the top.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_self.link">self.link</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Behavior of the links. One of:
</p>

<ul>
<li> <p><em><code>1</code></em>: Prevents self linking.
</p>
</li>
<li> <p><em><code>2</code></em>: Allows self linking.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_symmetric">symmetric</code></td>
<td>
<p>pass to <code><a href="circlize.html#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code>
</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_directional">directional</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the direction of the links. One of:
</p>

<ul>
<li> <p><em><code>0</code></em>: Non-directional data.
</p>
</li>
<li> <p><em><code>1</code></em>: Links go from &quot;from&quot; to &quot;to&quot;.
</p>
</li>
<li> <p><em><code>-1</code></em>: Links go from &quot;to&quot; to &quot;from&quot;.
</p>
</li>
<li> <p><em><code>2</code></em>: Links go in both directions.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_direction.type">direction.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | How to display the directions. One of:
</p>

<ul>
<li> <p><em><code>diffHeight</code></em>: Sets a line at the origin of the group showing to how many groups and in which proportion this group is linked to.
</p>
</li>
<li> <p><em><code>arrows</code></em>: Sets the connection as arrows.
</p>
</li>
<li> <p><em><code>both</code></em>: Sets up both behaviors. Use as: <code>c("diffHeight", "arrows")</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_link.arr.type">link.arr.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Sets the appearance of the arrows. One of:
</p>

<ul>
<li> <p><em><code>triangle</code></em>: Arrow with a triangle tip at the end displayed on top of the link.
</p>
</li>
<li> <p><em><code>big.arrow</code></em>: The link itself ends in a triangle shape.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_scale">scale</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to put all nodes the same width.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_alignment">alignment</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | How to align the diagram.  One of:
</p>

<ul>
<li> <p><em><code>default</code></em>: Allows <span class="pkg">circlize</span> to set up the plot as it sees fit.
</p>
</li>
<li> <p><em><code>horizontal</code></em>: Sets the break between &quot;from&quot; and &quot;to&quot; groups on the horizontal axis.
</p>
</li>
<li> <p><em><code>vertical</code></em>: Sets the break between &quot;from&quot; and &quot;to&quot; groups on the vertical axis.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_annotationtrack">annotationTrack</code></td>
<td>
<p>pass to <code><a href="circlize.html#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="circlize.html#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_padding_labels">padding_labels</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of extra padding (white spaces) of the labels so that they do not overlap with the scales.</p>
</td></tr>
<tr><td><code id="do_ChordDiagramPlot_+3A_...">...</code></td>
<td>
<p>For internal use only.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A circlize plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_ChordDiagramPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Basic chord diagram.
    sample$assignment &lt;- ifelse(sample$seurat_clusters %in% c("0", "4", "7"), "A", "B")
    sample$assignment[sample$seurat_clusters %in% c("1", "2")] &lt;- "C"
    sample$assignment[sample$seurat_clusters %in% c("10", "5")] &lt;- "D"
    sample$assignment[sample$seurat_clusters %in% c("8", "9")] &lt;- "E"

    p &lt;- SCpubr::do_ChordDiagramPlot(sample = sample,
                                     from = "seurat_clusters",
                                     to = "assignment")

    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }


</code></pre>

<hr>
<h2 id='do_ColorPalette'>Generate color scales based on a value.</h2><span id='topic+do_ColorPalette'></span>

<h3>Description</h3>

<p>This function is an adaptation of colortools package. As the package was removed from CRAN on 23-06-2022, this utility function came to existence in order to cover the gap. It is, on its basis,
an adaptation of the package into a single function. Original code, developed by Gaston Sanchez, can be found in: <a href="https://github.com/gastonstat/colortools">https://github.com/gastonstat/colortools</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_ColorPalette(
  colors.use,
  n = 12,
  opposite = FALSE,
  adjacent = FALSE,
  triadic = FALSE,
  split_complementary = FALSE,
  tetradic = FALSE,
  square = FALSE,
  complete_output = FALSE,
  plot = FALSE,
  font.size = 14,
  font.type = "sans"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_ColorPalette_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One color upon which generate the color scale. Can be a name or a HEX code.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_n">n</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of colors to include in the color wheel. Use it when all other options are FALSE, otherwise, it becomes 12.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_opposite">opposite</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return the opposing color to the one provided.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_adjacent">adjacent</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return the adjacent colors to the one provided.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_triadic">triadic</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return the triadic combination of colors to the one provided.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_split_complementary">split_complementary</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return the split complementary combination of colors to the one provided.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_tetradic">tetradic</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return the tetradic combination of colors to the one provided.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_square">square</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return the square combination of colors to the one provided.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_complete_output">complete_output</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Runs all the previous options and returns all the outputs as a list that contains all color vectors, all plots and a combined plot with everything.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_plot">plot</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to also return a plot displaying the values instead of a vector with the color.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_ColorPalette_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector with the desired color scale.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_ColorPalette", passive = TRUE)
  
  if (isTRUE(value)){
    # Generate a color wheel based on a single value.
    colors &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue")
    p &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                 plot = TRUE)
  
    # Generate a pair of opposite colors based on a given one.
    colors &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                      opposite = TRUE)
    p &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                 opposite = TRUE,
                                 plot = TRUE)
  
    # Generate a trio of adjacent colors based on a given one.
    colors &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                      adjacent = TRUE)
    p &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                 adjacent = TRUE,
                                 plot = TRUE)
  
    # Generate a trio of triadic colors based on a given one.
    colors &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                      triadic = TRUE)
    p &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                 triadic = TRUE,
                                 plot = TRUE)
  
    # Generate a trio of split complementary colors based on a given one.
    colors &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                      split_complementary = TRUE)
    p &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                 split_complementary = TRUE,
                                 plot = TRUE)
  
    # Generate a group of tetradic colors based on a given one.
    colors &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                      tetradic = TRUE)
    p &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                 tetradic = TRUE,
                                 plot = TRUE)
  
    # Generate a group of square colors based on a given one.
    colors &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                      square = TRUE)
    p &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                 square = TRUE,
                                 plot = TRUE)
  
    # Retrieve the output of all options.
    out &lt;- SCpubr::do_ColorPalette(colors.use = "steelblue",
                                   complete_output = TRUE)
    ## Retrieve the colors.
    colors &lt;- out$colors
    ## Retrieve the plots.
    plots &lt;- out$plots
    ## Retrieve a combined plot with all the options.
    p &lt;- out$combined_plot
  
  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_CopyNumberVariantPlot'>Display CNV scores from inferCNV as Feature Plots.</h2><span id='topic+do_CopyNumberVariantPlot'></span>

<h3>Description</h3>

<p>Display CNV scores from inferCNV as Feature Plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_CopyNumberVariantPlot(
  sample,
  infercnv_object,
  chromosome_locations,
  group.by = NULL,
  using_metacells = FALSE,
  metacell_mapping = NULL,
  legend.type = "colorbar",
  legend.position = "bottom",
  legend.length = 20,
  legend.width = 1,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  font.size = 14,
  pt.size = 1,
  font.type = "sans",
  axis.text.x.angle = 45,
  enforce_symmetry = TRUE,
  legend.title = NULL,
  na.value = "grey75",
  viridis.palette = "G",
  viridis.direction = 1,
  verbose = FALSE,
  min.cutoff = NA,
  max.cutoff = NA,
  number.breaks = 5,
  diverging.palette = "RdBu",
  diverging.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = -1,
  use_viridis = TRUE,
  return_object = FALSE,
  grid.color = "white",
  border.color = "black",
  flip = FALSE,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_CopyNumberVariantPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_infercnv_object">infercnv_object</code></td>
<td>
<p><strong><code><a href="infercnv.html#topic+infercnv">infercnv</a></code></strong> | Output inferCNV object run on the same Seurat object.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_chromosome_locations">chromosome_locations</code></td>
<td>
<p><strong><code><a href="tibble.html#topic+tibble">tibble</a></code></strong> | Tibble containing the chromosome regions to use. Can be obtained using <strong><code>utils::data("human_chr_locations", package = "SCpubr")</code></strong>.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_using_metacells">using_metacells</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether inferCNV was run using metacells or not.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_metacell_mapping">metacell_mapping</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Vector or cell - metacell mapping.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_legend.length">legend.length</code>, <code id="do_CopyNumberVariantPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_CopyNumberVariantPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return a symmetrical plot axes-wise or continuous color scale-wise, when applicable.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_verbose">verbose</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show extra comments, warnings,etc.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_min.cutoff">min.cutoff</code>, <code id="do_CopyNumberVariantPlot_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_return_object">return_object</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Returns the Seurat object with the modifications performed in the function. Nomally, this contains a new assay with the data that can then be used for any other visualization desired.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_CopyNumberVariantPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_CopyNumberVariantPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_CopyNumberVariantPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_CopyNumberVariantPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_CopyNumberVariantPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_CopyNumberVariantPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_CopyNumberVariantPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing Feature Plots for different chromosome regions and corresponding dot plots by groups..
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_CopyNumberVariantPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # This function expects that you have run inferCNV on your
    # own and you have access to the output object.

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds",
                                  package = "SCpubr"))

    # Define your inferCNV object.
    infercnv_object &lt;- readRDS(system.file("extdata/infercnv_object_example.rds",
                                           package = "SCpubr"))

    # Get human chromosome locations.
    chromosome_locations = SCpubr::human_chr_locations

    # Compute for a all chromosomes.
    p &lt;- SCpubr::do_CopyNumberVariantPlot(sample = sample,
                                          infercnv_object = infercnv_object,
                                          using_metacells = FALSE,
                                          chromosome_locations = chromosome_locations)

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_CorrelationPlot'>Create correlation matrix heatmaps.</h2><span id='topic+do_CorrelationPlot'></span>

<h3>Description</h3>

<p>Create correlation matrix heatmaps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_CorrelationPlot(
  sample = NULL,
  input_gene_list = NULL,
  cluster = TRUE,
  remove.diagonal = TRUE,
  mode = "hvg",
  assay = NULL,
  group.by = NULL,
  legend.title = "Pearson coef.",
  enforce_symmetry = ifelse(mode == "hvg", TRUE, FALSE),
  font.size = 14,
  font.type = "sans",
  na.value = "grey75",
  legend.width = 1,
  legend.length = 20,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.type = "colorbar",
  legend.position = "bottom",
  min.cutoff = NA,
  max.cutoff = NA,
  number.breaks = 5,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  diverging.palette = "RdBu",
  diverging.direction = -1,
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  axis.text.x.angle = 45,
  grid.color = "white",
  border.color = "black",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_CorrelationPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_input_gene_list">input_gene_list</code></td>
<td>
<p><strong><code><a href="#topic+named_list">named_list</a></code></strong> | Named list of lists of genes to be used as input.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_cluster">cluster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to cluster the elements in the heatmap or not.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_remove.diagonal">remove.diagonal</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to convert diagnoal to NA. Normally this value would be 1, heavily shifting the color scale.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_mode">mode</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Different types of correlation matrices can be computed. Right now, the only possible value is &quot;hvg&quot;, standing for Highly Variable Genes. The sample is subset for the HVG and the data is re-scaled. Scale data is used for the correlation.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return a symmetrical plot axes-wise or continuous color scale-wise, when applicable.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_legend.length">legend.length</code>, <code id="do_CorrelationPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_CorrelationPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_min.cutoff">min.cutoff</code>, <code id="do_CorrelationPlot_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_plot.title">plot.title</code>, <code id="do_CorrelationPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_CorrelationPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_CorrelationPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_CorrelationPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_CorrelationPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_CorrelationPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_CorrelationPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_CorrelationPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_CorrelationPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_CorrelationPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Default values.
    p &lt;- SCpubr::do_CorrelationPlot(sample = sample)
    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }


</code></pre>

<hr>
<h2 id='do_DimPlot'>Wrapper for <a href="Seurat.html#topic+DimPlot">DimPlot</a>.</h2><span id='topic+do_DimPlot'></span>

<h3>Description</h3>

<p>Wrapper for <a href="Seurat.html#topic+DimPlot">DimPlot</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_DimPlot(
  sample,
  reduction = NULL,
  group.by = NULL,
  split.by = NULL,
  colors.use = NULL,
  shuffle = TRUE,
  order = NULL,
  raster = FALSE,
  pt.size = 1,
  label = FALSE,
  label.color = "black",
  label.fill = "white",
  label.size = 4,
  label.box = TRUE,
  repel = FALSE,
  cells.highlight = NULL,
  idents.highlight = NULL,
  idents.keep = NULL,
  sizes.highlight = 1,
  ncol = NULL,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  legend.title = NULL,
  legend.position = "bottom",
  legend.title.position = "top",
  legend.ncol = NULL,
  legend.nrow = NULL,
  legend.icon.size = 4,
  legend.byrow = FALSE,
  raster.dpi = 2048,
  dims = c(1, 2),
  font.size = 14,
  font.type = "sans",
  na.value = "grey75",
  plot_cell_borders = TRUE,
  border.size = 2,
  border.color = "black",
  border.density = 1,
  plot_marginal_distributions = FALSE,
  marginal.type = "density",
  marginal.size = 5,
  marginal.group = TRUE,
  plot.axes = FALSE,
  plot_density_contour = FALSE,
  contour.position = "bottom",
  contour.color = "grey90",
  contour.lineend = "butt",
  contour.linejoin = "round",
  contour_expand_axes = 0.25,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_DimPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_reduction">reduction</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Reduction to use. Can be the canonical ones such as &quot;umap&quot;, &quot;pca&quot;, or any custom ones, such as &quot;diffusion&quot;. If you are unsure about which reductions you have, use <code>Seurat::Reductions(sample)</code>. Defaults to &quot;umap&quot; if present or to the last computed reduction if the argument is not provided.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of valid color representations (either name of HEX codes) with as many named colors as unique values of group.by. If group.by is not provided, defaults to the unique values of <a href="Seurat.html#topic+Idents">Idents</a>. If not provided, a color scale will be set by default.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_shuffle">shuffle</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to shuffle the cells or not, so that they are not plotted cluster-wise. Recommended.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_order">order</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of identities to be plotted. Either one with all identities or just some, which will be plotted last.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_raster">raster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to raster the resulting plot. This is recommendable if plotting a lot of cells.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_label">label</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot the cluster labels in the UMAP. The cluster labels will have the same color as the cluster colors.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_label.color">label.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the labels in the plot.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_label.fill">label.fill</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color to fill the labels. Has to be a single color, that will be used for all labels. If <strong><code>NULL</code></strong>, the colors of the clusters will be used instead.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_label.size">label.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the labels in the plot.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_label.box">label.box</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot the plot labels as <strong><code><a href="ggplot2.html#topic+geom_text">geom_text</a></code></strong> (FALSE) or <strong><code><a href="ggplot2.html#topic+geom_label">geom_label</a></code></strong> (TRUE).</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_repel">repel</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to repel the text labels.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_cells.highlight">cells.highlight</code>, <code id="do_DimPlot_+3A_idents.highlight">idents.highlight</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of cells/identities to focus into. The identities have to much those in <code>Seurat::Idents(sample)</code> The rest of the cells will be grayed out. Both parameters can be used at the same time.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_idents.keep">idents.keep</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of identities to keep. This will effectively set the rest of the cells that do not match the identities provided to NA, therefore coloring them according to na.value parameter.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_sizes.highlight">sizes.highlight</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Point size of highlighted cells using cells.highlight parameter.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_ncol">ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns used in the arrangement of the output plot using &quot;split.by&quot; parameter.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_plot.title">plot.title</code>, <code id="do_DimPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_DimPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_legend.title.position">legend.title.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position for the title of the legend. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the legend.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the legend.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the legend.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_legend.nrow">legend.nrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of rows in the legend.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_legend.icon.size">legend.icon.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the icons in legend.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the legend is filled by row or not.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Pixel resolution for rasterized plots. Defaults to 1024. Only activates on Seurat versions higher or equal than 4.1.0.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_dims">dims</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Vector of 2 numerics indicating the dimensions to plot out of the selected reduction. Defaults to c(1, 2) if not specified.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_plot_cell_borders">plot_cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot border around cells.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_border.density">border.density</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of cells used when <strong><code>plot_cell_borders = TRUE</code></strong>. Value between 0 and 1. It computes a 2D kernel density and based on this cells that have a density below the specified quantile will be used to generate the cluster contour. The lower this number, the less cells will be selected, thus reducing the overall size of the plot but also potentially preventing all the contours to be properly drawn.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_plot_marginal_distributions">plot_marginal_distributions</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> |  Whether to plot marginal distributions on the figure or not.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_marginal.type">marginal.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of:
</p>

<ul>
<li> <p><em><code>density</code></em>: Compute density plots on the margins.
</p>
</li>
<li> <p><em><code>histogram</code></em>: Compute histograms on the margins.
</p>
</li>
<li> <p><em><code>boxplot</code></em>: Compute boxplot on the margins.
</p>
</li>
<li> <p><em><code>violin</code></em>: Compute violin plots on the margins.
</p>
</li>
<li> <p><em><code>densigram</code></em>: Compute densigram plots on the margins.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_marginal.size">marginal.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size ratio between the main and marginal plots. A value of 5 means that the main plot is 5 times bigger than the marginal plots.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_marginal.group">marginal.group</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to group the marginal distribution by group.by or current identities.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_plot.axes">plot.axes</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot axes or not.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_plot_density_contour">plot_density_contour</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot density contours in the UMAP.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_contour.position">contour.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Whether to plot density contours on top or at the bottom of the visualization layers, thus overlapping the clusters/cells or not.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_contour.color">contour.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the density lines.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_contour.lineend">contour.lineend</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Line end style (round, butt, square).</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_contour.linejoin">contour.linejoin</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Line join style (round, mitre, bevel).</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_contour_expand_axes">contour_expand_axes</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | To make the contours fit the plot, the limits of the X and Y axis are expanding a given percentage from the min and max values for each axis. This controls such percentage.</p>
</td></tr>
<tr><td><code id="do_DimPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_DimPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_DimPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_DimPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_DimPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_DimPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_DimPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object containing a DimPlot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_DimPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Basic DimPlot.
    p &lt;- SCpubr::do_DimPlot(sample = sample)


    # Restrict the amount of identities displayed.
    p &lt;- SCpubr::do_DimPlot(sample = sample,
                            idents.keep = c("1", "3", "5"))

    # Group by another variable rather than `Seurat::Idents(sample)`
    p &lt;- SCpubr::do_DimPlot(sample = sample,
                            group.by = "seurat_clusters")

    # Split the output in as many plots as unique identities.
    p &lt;- SCpubr::do_DimPlot(sample = sample,
                            split.by = "seurat_clusters")

    # Highlight given identities
    p &lt;- SCpubr::do_DimPlot(sample,
                            idents.highlight = c("1", "3"))

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }




</code></pre>

<hr>
<h2 id='do_DotPlot'>This function is a wrapper for <a href="Seurat.html#topic+DotPlot">DotPlot</a>. It provides most of its functionalities while adding extra.
You can</h2><span id='topic+do_DotPlot'></span>

<h3>Description</h3>

<p>This function is a wrapper for <a href="Seurat.html#topic+DotPlot">DotPlot</a>. It provides most of its functionalities while adding extra.
You can
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_DotPlot(
  sample,
  features,
  assay = NULL,
  group.by = NULL,
  scale = FALSE,
  legend.title = NULL,
  legend.type = "colorbar",
  legend.position = "bottom",
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.length = 20,
  legend.width = 1,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  colors.use = NULL,
  dot.scale = 6,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  xlab = NULL,
  ylab = NULL,
  font.size = 14,
  font.type = "sans",
  cluster = FALSE,
  flip = FALSE,
  axis.text.x.angle = 45,
  scale.by = "size",
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  na.value = "grey75",
  dot_border = TRUE,
  plot.grid = TRUE,
  grid.color = "grey75",
  grid.type = "dashed",
  number.breaks = 5,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_DotPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_features">features</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Features to represent.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_scale">scale</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the data should be scaled or not. Non-scaled data allows for comparison across genes. Scaled data allows for an easier comparison along the same gene.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_DotPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_legend.length">legend.length</code>, <code id="do_DotPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of valid color representations (either name of HEX codes) with as many named colors as unique values of group.by. If group.by is not provided, defaults to the unique values of <a href="Seurat.html#topic+Idents">Idents</a>. If not provided, a color scale will be set by default.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_dot.scale">dot.scale</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Scale the size of the dots.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_plot.title">plot.title</code>, <code id="do_DotPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_DotPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_xlab">xlab</code>, <code id="do_DotPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_cluster">cluster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to cluster the identities based on the expression of the features.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_scale.by">scale.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | How to scale the size of the dots. One of:
</p>

<ul>
<li> <p><em><code>radius</code></em>: use radius aesthetic.
</p>
</li>
<li> <p><em><code>size</code></em>: use size aesthetic.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_dot_border">dot_border</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot a border around dots.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_plot.grid">plot.grid</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot grid lines.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_grid.type">grid.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of the possible linetype options:
</p>

<ul>
<li> <p><em><code>blank</code></em>.
</p>
</li>
<li> <p><em><code>solid</code></em>.
</p>
</li>
<li> <p><em><code>dashed</code></em>.
</p>
</li>
<li> <p><em><code>dotted</code></em>.
</p>
</li>
<li> <p><em><code>dotdash</code></em>.
</p>
</li>
<li> <p><em><code>longdash</code></em>.
</p>
</li>
<li> <p><em><code>twodash</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_DotPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_DotPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_DotPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_DotPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_DotPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_DotPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_DotPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object containing a Dot Plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_DotPlot", passive = TRUE)

  if (isTRUE(value)){
    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Basic Dot plot.
    p &lt;- SCpubr::do_DotPlot(sample = sample,
                            features = "EPC1")

    # Querying multiple features.
    genes &lt;- rownames(sample)[1:14]
    p &lt;- SCpubr::do_DotPlot(sample = sample,
                            features = genes)

    # Inverting the axes.
    p &lt;- SCpubr::do_DotPlot(sample = sample,
                            features = genes,
                            cluster = TRUE,
                            plot.title = "Clustered",
                            flip = TRUE)

    # Modifying default colors.
    # Two colors to generate a gradient.
    p &lt;- SCpubr::do_DotPlot(sample = sample,
                            features = genes,
                            colors.use = c("#001219", "#e9d8a6"))

    # Querying multiple features as a named list - splitting by each item in list.
    # Genes have to be unique.
    genes &lt;- list("Naive CD4+ T" = rownames(sample)[1:2],
                  "EPC1+ Mono" = rownames(sample)[3:4],
                  "Memory CD4+" = rownames(sample)[5],
                  "B" = rownames(sample)[6],
                  "CD8+ T" = rownames(sample)[7],
                  "FCGR3A+ Mono" = rownames(sample)[8:9],
                  "NK" = rownames(sample)[10:11],
                  "DC" = rownames(sample)[12:13],
                  "Platelet" = rownames(sample)[14])

    p &lt;- SCpubr::do_DotPlot(sample = sample,
                            features = genes)

    # Clustering the identities.
    p &lt;- SCpubr::do_DotPlot(sample = sample,
                            features = genes,
                            cluster = TRUE,
                            plot.title = "Clustered")
  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_EnrichmentHeatmap'>Create enrichment scores heatmaps.</h2><span id='topic+do_EnrichmentHeatmap'></span>

<h3>Description</h3>

<p>This function computes the enrichment scores for the cells using <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a> and then aggregates the scores by the metadata variables provided by the user and displays it as a heatmap, computed by <a href="ComplexHeatmap.html#topic+Heatmap">Heatmap</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_EnrichmentHeatmap(
  sample,
  input_gene_list,
  features.order = NULL,
  groups.order = NULL,
  cluster = TRUE,
  scale_scores = TRUE,
  assay = NULL,
  slot = NULL,
  reduction = NULL,
  group.by = NULL,
  verbose = FALSE,
  na.value = "grey75",
  legend.position = "bottom",
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = 1,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.length = 20,
  legend.width = 1,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.type = "colorbar",
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  enforce_symmetry = FALSE,
  nbin = 24,
  ctrl = 100,
  flavor = "Seurat",
  legend.title = NULL,
  ncores = 1,
  storeRanks = TRUE,
  min.cutoff = NA,
  max.cutoff = NA,
  pt.size = 1,
  plot_cell_borders = TRUE,
  border.size = 2,
  return_object = FALSE,
  number.breaks = 5,
  sequential.palette = "YlGnBu",
  diverging.palette = "RdBu",
  diverging.direction = -1,
  sequential.direction = 1,
  flip = FALSE,
  grid.color = "white",
  border.color = "black",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_EnrichmentHeatmap_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_input_gene_list">input_gene_list</code></td>
<td>
<p><strong><code><a href="#topic+named_list">named_list</a></code></strong> | Named list of lists of genes to be used as input.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_features.order">features.order</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Should the gene sets be ordered in a specific way? Provide it as a vector of characters with the same names as the names of the gene sets.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_groups.order">groups.order</code></td>
<td>
<p><strong><code><a href="#topic+named_list">named_list</a></code></strong> | Should the groups in theheatmaps be ordered in a specific way? Provide it as a named list (as many lists as values in <strong><code>group.by</code></strong>) with the order for each of the elements in the groups.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_cluster">cluster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to perform clustering of rows and columns.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_scale_scores">scale_scores</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to transform the scores to a range of 0-1 for plotting.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_reduction">reduction</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Reduction to use. Can be the canonical ones such as &quot;umap&quot;, &quot;pca&quot;, or any custom ones, such as &quot;diffusion&quot;. If you are unsure about which reductions you have, use <code>Seurat::Reductions(sample)</code>. Defaults to &quot;umap&quot; if present or to the last computed reduction if the argument is not provided.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_verbose">verbose</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show extra comments, warnings,etc.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_EnrichmentHeatmap_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_legend.length">legend.length</code>, <code id="do_EnrichmentHeatmap_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the geyser and feature plot has a symmetrical color scale.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_nbin">nbin</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of bins to use in <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a>.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_ctrl">ctrl</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of genes in the control set to use in <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a>.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_flavor">flavor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of: Seurat, UCell. Compute the enrichment scores using <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a> or <a href="UCell.html#topic+AddModuleScore_UCell">AddModuleScore_UCell</a>.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_ncores">ncores</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of cores used to run UCell scoring.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_storeranks">storeRanks</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to store the ranks for faster UCell scoring computations. Might require large amounts of RAM.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_min.cutoff">min.cutoff</code>, <code id="do_EnrichmentHeatmap_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_plot_cell_borders">plot_cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot border around cells.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_return_object">return_object</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return the Seurat object with the enrichment scores stored.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_EnrichmentHeatmap_+3A_plot.title.face">plot.title.face</code>, <code id="do_EnrichmentHeatmap_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_EnrichmentHeatmap_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_EnrichmentHeatmap_+3A_axis.title.face">axis.title.face</code>, <code id="do_EnrichmentHeatmap_+3A_axis.text.face">axis.text.face</code>, <code id="do_EnrichmentHeatmap_+3A_legend.title.face">legend.title.face</code>, <code id="do_EnrichmentHeatmap_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_EnrichmentHeatmap", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Genes have to be unique.
    genes &lt;- list("A" = rownames(sample)[1:5],
                  "B" = rownames(sample)[6:10],
                  "C" = rownames(sample)[11:15])

    # Default parameters.
    p &lt;- SCpubr::do_EnrichmentHeatmap(sample = sample,
                                      input_gene_list = genes,
                                      nbin = 1,
                                      ctrl = 10)
    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_ExpressionHeatmap'>Create heatmaps of averaged expression by groups.</h2><span id='topic+do_ExpressionHeatmap'></span>

<h3>Description</h3>

<p>This function generates a heatmap with averaged expression values by the unique groups of the metadata variables provided by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_ExpressionHeatmap(
  sample,
  features,
  group.by = NULL,
  assay = NULL,
  cluster = TRUE,
  features.order = NULL,
  groups.order = NULL,
  slot = "data",
  legend.title = "Avg. Expression",
  na.value = "grey75",
  legend.position = "bottom",
  legend.width = 1,
  legend.length = 20,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.type = "colorbar",
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  enforce_symmetry = FALSE,
  min.cutoff = NA,
  max.cutoff = NA,
  diverging.palette = "RdBu",
  diverging.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  number.breaks = 5,
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = -1,
  flip = FALSE,
  grid.color = "white",
  border.color = "black",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_ExpressionHeatmap_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_features">features</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Features to represent.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_cluster">cluster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to perform clustering of rows and columns.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_features.order">features.order</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Should the gene sets be ordered in a specific way? Provide it as a vector of characters with the same names as the names of the gene sets.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_groups.order">groups.order</code></td>
<td>
<p><strong><code><a href="#topic+named_list">named_list</a></code></strong> | Should the groups in theheatmaps be ordered in a specific way? Provide it as a named list (as many lists as values in <strong><code>group.by</code></strong>) with the order for each of the elements in the groups.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_legend.length">legend.length</code>, <code id="do_ExpressionHeatmap_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_ExpressionHeatmap_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return a symmetrical plot axes-wise or continuous color scale-wise, when applicable.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_min.cutoff">min.cutoff</code>, <code id="do_ExpressionHeatmap_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_ExpressionHeatmap_+3A_plot.title.face">plot.title.face</code>, <code id="do_ExpressionHeatmap_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_ExpressionHeatmap_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_ExpressionHeatmap_+3A_axis.title.face">axis.title.face</code>, <code id="do_ExpressionHeatmap_+3A_axis.text.face">axis.text.face</code>, <code id="do_ExpressionHeatmap_+3A_legend.title.face">legend.title.face</code>, <code id="do_ExpressionHeatmap_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_ExpressionHeatmap", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Define list of genes.
    genes &lt;- rownames(sample)[1:10]

    # Default parameters.
    p &lt;- SCpubr::do_ExpressionHeatmap(sample = sample,
                                      features = genes,
                                      viridis.direction = -1)
    
    p
  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_FeaturePlot'>Wrapper for <a href="Seurat.html#topic+FeaturePlot">FeaturePlot</a>.</h2><span id='topic+do_FeaturePlot'></span>

<h3>Description</h3>

<p>Wrapper for <a href="Seurat.html#topic+FeaturePlot">FeaturePlot</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_FeaturePlot(
  sample,
  features,
  assay = NULL,
  reduction = NULL,
  slot = NULL,
  order = FALSE,
  group.by = NULL,
  group.by.colors.use = NULL,
  group.by.legend = NULL,
  group.by.show.dots = TRUE,
  group.by.dot.size = 8,
  group.by.cell_borders = FALSE,
  group.by.cell_borders.alpha = 0.1,
  split.by = NULL,
  idents.keep = NULL,
  cells.highlight = NULL,
  idents.highlight = NULL,
  dims = c(1, 2),
  enforce_symmetry = FALSE,
  symmetry.type = "absolute",
  symmetry.center = NA,
  pt.size = 1,
  font.size = 14,
  font.type = "sans",
  legend.title = NULL,
  legend.type = "colorbar",
  legend.position = "bottom",
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.length = 20,
  legend.width = 1,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.ncol = NULL,
  legend.nrow = NULL,
  legend.byrow = FALSE,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  individual.titles = NULL,
  individual.subtitles = NULL,
  individual.captions = NULL,
  ncol = NULL,
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = 1,
  raster = FALSE,
  raster.dpi = 1024,
  plot_cell_borders = TRUE,
  border.size = 2,
  border.color = "black",
  border.density = 1,
  na.value = "grey75",
  verbose = TRUE,
  plot.axes = FALSE,
  min.cutoff = rep(NA, length(features)),
  max.cutoff = rep(NA, length(features)),
  plot_density_contour = FALSE,
  contour.position = "bottom",
  contour.color = "grey90",
  contour.lineend = "butt",
  contour.linejoin = "round",
  contour_expand_axes = 0.25,
  label = FALSE,
  label.color = "black",
  label.size = 4,
  number.breaks = 5,
  diverging.palette = "RdBu",
  diverging.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_FeaturePlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_features">features</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Features to represent.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_reduction">reduction</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Reduction to use. Can be the canonical ones such as &quot;umap&quot;, &quot;pca&quot;, or any custom ones, such as &quot;diffusion&quot;. If you are unsure about which reductions you have, use <code>Seurat::Reductions(sample)</code>. Defaults to &quot;umap&quot; if present or to the last computed reduction if the argument is not provided.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_order">order</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to order the cells based on expression.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable based on which cells are grouped. This will effectively introduce a big dot in the center of each cluster, colored using a categorical color scale or with the values provided by the user in <strong><code>group.by.colors.use</code></strong>. It will also displays a legend.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_group.by.colors.use">group.by.colors.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Colors to use for the group dots.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_group.by.legend">group.by.legend</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend when <strong><code>group.by</code></strong> is used. Use <strong><code>NA</code></strong> to disable it and <strong><code>NULL</code></strong> to use the default column title provided in <strong><code>group.by</code></strong>.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_group.by.show.dots">group.by.show.dots</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Controls whether to place in the middle of the groups.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_group.by.dot.size">group.by.dot.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots placed in the middle of the groups.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_group.by.cell_borders">group.by.cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Plots another border around the cells displaying the same color code of the dots displayed with <strong><code>group.by</code></strong>. Legend is shown always with alpha = 1 regardless of the alpha settings.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_group.by.cell_borders.alpha">group.by.cell_borders.alpha</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the transparency of the new borders drawn by <strong><code>group.by.cell_borders</code></strong>.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_idents.keep">idents.keep</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of identities to plot. The gradient scale will also be subset to only the values of such identities.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_cells.highlight">cells.highlight</code>, <code id="do_FeaturePlot_+3A_idents.highlight">idents.highlight</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of cells/identities to focus into. The identities have to much those in <code>Seurat::Idents(sample)</code> The rest of the cells will be grayed out. Both parameters can be used at the same time.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_dims">dims</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Vector of 2 numerics indicating the dimensions to plot out of the selected reduction. Defaults to c(1, 2) if not specified.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return a symmetrical plot axes-wise or continuous color scale-wise, when applicable.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_symmetry.type">symmetry.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetry to be enforced. One of:
</p>

<ul>
<li> <p><em><code>absolute</code></em>: The highest absolute value will be taken into a account to generate the color scale. Works after <strong><code>min.cutoff</code></strong> and <strong><code>max.cutoff</code></strong>.
</p>
</li>
<li> <p><em><code>centered</code></em>: Centers the scale around the provided value in <strong><code>symmetry.center</code></strong>. Works after <strong><code>min.cutoff</code></strong> and <strong><code>max.cutoff</code></strong>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_symmetry.center">symmetry.center</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Value upon which the scale will be centered.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_FeaturePlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.length">legend.length</code>, <code id="do_FeaturePlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.nrow">legend.nrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of rows in the legend.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the legend is filled by row or not.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_plot.title">plot.title</code>, <code id="do_FeaturePlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_FeaturePlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_individual.titles">individual.titles</code>, <code id="do_FeaturePlot_+3A_individual.subtitles">individual.subtitles</code>, <code id="do_FeaturePlot_+3A_individual.captions">individual.captions</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles or subtitles. for each feature if needed. Either NULL or a vector of equal length of features.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_ncol">ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns used in the arrangement of the output plot using &quot;split.by&quot; parameter.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_raster">raster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to raster the resulting plot. This is recommendable if plotting a lot of cells.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Pixel resolution for rasterized plots. Defaults to 1024. Only activates on Seurat versions higher or equal than 4.1.0.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_plot_cell_borders">plot_cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot border around cells.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_border.density">border.density</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of cells used when <strong><code>plot_cell_borders = TRUE</code></strong>. Value between 0 and 1. It computes a 2D kernel density and based on this cells that have a density below the specified quantile will be used to generate the cluster contour. The lower this number, the less cells will be selected, thus reducing the overall size of the plot but also potentially preventing all the contours to be properly drawn.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_verbose">verbose</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show extra comments, warnings,etc.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_plot.axes">plot.axes</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot axes or not.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_min.cutoff">min.cutoff</code>, <code id="do_FeaturePlot_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_plot_density_contour">plot_density_contour</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot density contours in the UMAP.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_contour.position">contour.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Whether to plot density contours on top or at the bottom of the visualization layers, thus overlapping the clusters/cells or not.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_contour.color">contour.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the density lines.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_contour.lineend">contour.lineend</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Line end style (round, butt, square).</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_contour.linejoin">contour.linejoin</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Line join style (round, mitre, bevel).</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_contour_expand_axes">contour_expand_axes</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | To make the contours fit the plot, the limits of the X and Y axis are expanding a given percentage from the min and max values for each axis. This controls such percentage.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_label">label</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot the cluster labels in the UMAP. The cluster labels will have the same color as the cluster colors.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_label.color">label.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the labels in the plot.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_label.size">label.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the labels in the plot.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_FeaturePlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_FeaturePlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_FeaturePlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_FeaturePlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_FeaturePlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_FeaturePlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_FeaturePlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object containing a Feature Plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_FeaturePlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Regular FeaturePlot.
    p &lt;- SCpubr::do_FeaturePlot(sample = sample,
                                features = "nCount_RNA")

    # FeaturePlot with a subset of identities
    # (in Seurat::Idents(sample)) maintaining the original UMAP shape.
    idents.use &lt;- levels(sample)[!(levels(sample) %in% c("2", "5", "8"))]
    p &lt;- SCpubr::do_FeaturePlot(sample = sample,
                                idents.highlight = idents.use,
                                features = c("EPC1"))

    # Splitting the FeaturePlot by a variable and
    # maintaining the color scale and the UMAP shape.
    p &lt;- SCpubr::do_FeaturePlot(sample = sample,
                                features = "EPC1",
                                split.by = "seurat_clusters")

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_FunctionalAnnotationPlot'>Compute functional annotation plots using GO or KEGG ontologies</h2><span id='topic+do_FunctionalAnnotationPlot'></span>

<h3>Description</h3>

<p>Compute functional annotation plots using GO or KEGG ontologies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_FunctionalAnnotationPlot(
  genes,
  org.db,
  organism = "hsa",
  database = "GO",
  GO_ontology = "BP",
  min.overlap = NULL,
  p.adjust.cutoff = 0.05,
  pAdjustMethod = "BH",
  minGSSize = 10,
  maxGSSize = 500,
  font.size = 10,
  font.type = "sans",
  axis.text.x.angle = 45,
  xlab = NULL,
  ylab = NULL,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  legend.type = "colorbar",
  legend.position = "bottom",
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.length = 10,
  legend.width = 1,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  number.breaks = 5,
  return_matrix = FALSE,
  grid.color = "white",
  border.color = "black",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_genes">genes</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of gene symbols to query for functional annotation.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_org.db">org.db</code></td>
<td>
<p><strong><code>OrgDB</code></strong> | Database object to use for the query.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_organism">organism</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Supported KEGG organism.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_database">database</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Database to run the analysis on. One of:
</p>

<ul>
<li> <p><em><code>GO</code></em>.
</p>
</li>
<li> <p><em><code>KEGG</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_go_ontology">GO_ontology</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | GO ontology to use. One of:
</p>

<ul>
<li> <p><em><code>BP</code></em>: For <strong>B</strong>iological <strong>P</strong>rocess.
</p>
</li>
<li> <p><em><code>MF</code></em>: For <strong>M</strong>olecular <strong>F</strong>unction.
</p>
</li>
<li> <p><em><code>CC</code></em>: For <strong>C</strong>ellular <strong>C</strong>omponent.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_min.overlap">min.overlap</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Filter the output result to the terms which are supported by this many genes.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_p.adjust.cutoff">p.adjust.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Significance cutoff used to filter non-significant terms.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_padjustmethod">pAdjustMethod</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Method to adjust for multiple testing.  One of:
</p>

<ul>
<li> <p><em><code>holm</code></em>.
</p>
</li>
<li> <p><em><code>hochberg</code></em>.
</p>
</li>
<li> <p><em><code>hommel</code></em>.
</p>
</li>
<li> <p><em><code>bonferroni</code></em>.
</p>
</li>
<li> <p><em><code>BH</code></em>.
</p>
</li>
<li> <p><em><code>BY</code></em>.
</p>
</li>
<li> <p><em><code>fdr</code></em>.
</p>
</li>
<li> <p><em><code>none</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_mingssize">minGSSize</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Minimal size of genes annotated by Ontology term for testing.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_maxgssize">maxGSSize</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Maximal size of genes annotated for testing.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_xlab">xlab</code>, <code id="do_FunctionalAnnotationPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_plot.title">plot.title</code>, <code id="do_FunctionalAnnotationPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_FunctionalAnnotationPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_FunctionalAnnotationPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_legend.length">legend.length</code>, <code id="do_FunctionalAnnotationPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_return_matrix">return_matrix</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Returns the matrices with the enriched Terms for further use.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_FunctionalAnnotationPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_FunctionalAnnotationPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_FunctionalAnnotationPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_FunctionalAnnotationPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_FunctionalAnnotationPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_FunctionalAnnotationPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_FunctionalAnnotationPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing a heatmap of the presence/absence of the genes in the enriched term, as well as a bar plot, dot plot and tree plot of the enriched terms.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_FunctionalAnnotationPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Need to load this library or equivalent.
    suppressMessages(library("org.Hs.eg.db"))

    # Define list of genes to query.
    genes.use &lt;- c("CCR7", "CD14", "LYZ",
                   "S100A4", "MS4A1",
                   "MS4A7", "GNLY", "NKG7", "FCER1A",
                   "CST3", "PPBP")

    # Compute the grouped GO terms.
    out &lt;- SCpubr::do_FunctionalAnnotationPlot(genes = genes.use,
                                               org.db = org.Hs.eg.db)

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_GeyserPlot'>Generate a Geyser plot.</h2><span id='topic+do_GeyserPlot'></span>

<h3>Description</h3>

<p>A Geyser plot is a custom plot in which we plot continuous values on the Y axis grouped by a categorical value in the X. This is plotted as a dot plot, jittered so that the dots span
all the way to the other groups. On top of this, the mean and .66 and .95 of the data is plotted, depicting the overall distribution of the dots. The cells can, then, be colored by
a continuous variable (same as Y axis or different) or a categorical one (same as X axis or different).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_GeyserPlot(
  sample,
  features,
  assay = NULL,
  slot = "data",
  group.by = NULL,
  split.by = NULL,
  enforce_symmetry = FALSE,
  scale_type = "continuous",
  order = TRUE,
  plot_cell_borders = TRUE,
  jitter = 0.45,
  pt.size = 1,
  border.size = 2,
  border.color = "black",
  legend.position = "bottom",
  legend.width = 1,
  legend.length = 20,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.type = "colorbar",
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  viridis.palette = "G",
  viridis.direction = 1,
  colors.use = NULL,
  na.value = "grey75",
  legend.ncol = NULL,
  legend.nrow = NULL,
  legend.icon.size = 4,
  legend.byrow = FALSE,
  legend.title = NULL,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  xlab = "Groups",
  ylab = feature,
  flip = FALSE,
  min.cutoff = rep(NA, length(features)),
  max.cutoff = rep(NA, length(features)),
  number.breaks = 5,
  diverging.palette = "RdBu",
  diverging.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = -1,
  use_viridis = TRUE,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_GeyserPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_features">features</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Features to represent.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return a symmetrical plot axes-wise or continuous color scale-wise, when applicable.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_scale_type">scale_type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of color scale to use.  One of:
</p>

<ul>
<li> <p><em><code>categorical</code></em>: Use a categorical color scale based on the values of &quot;group.by&quot;.
</p>
</li>
<li> <p><em><code>continuous</code></em>: Use a continuous color scale based on the values of &quot;feature&quot;.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_order">order</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to order the groups by the median of the data (highest to lowest).</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_plot_cell_borders">plot_cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot border around cells.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_jitter">jitter</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Amount of jitter in the plot along the X axis. The lower the value, the more compacted the dots are.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.length">legend.length</code>, <code id="do_GeyserPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_GeyserPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Named vector of colors to use. Has to match the unique values of group.by when scale_type is set to categorical.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.nrow">legend.nrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of rows in the legend.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.icon.size">legend.icon.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the icons in legend.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the legend is filled by row or not.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_plot.title">plot.title</code>, <code id="do_GeyserPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_GeyserPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_xlab">xlab</code>, <code id="do_GeyserPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_min.cutoff">min.cutoff</code>, <code id="do_GeyserPlot_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_GeyserPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_GeyserPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_GeyserPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_GeyserPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_GeyserPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_GeyserPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_GeyserPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>Special thanks to Christina Blume for coming up with the name of the plot.
</p>


<h3>Value</h3>

<p>Either a plot of a list of plots, depending on the number of features provided.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_GeyserPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Geyser plot with categorical color scale.
    p &lt;- SCpubr::do_GeyserPlot(sample = sample,
                              features = "nCount_RNA",
                              scale_type = "categorical")
    p

    # Geyser plot with continuous color scale.
    p &lt;- SCpubr::do_GeyserPlot(sample = sample,
                              features = "nCount_RNA",
                              scale_type = "continuous")


    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_GroupedGOTermPlot'>Compute an overview of the GO terms associated with the input list of genes.</h2><span id='topic+do_GroupedGOTermPlot'></span>

<h3>Description</h3>

<p>Compute an overview of the GO terms associated with the input list of genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_GroupedGOTermPlot(
  genes,
  org.db,
  levels.use = NULL,
  GO_ontology = "BP",
  min.overlap = 3,
  flip = TRUE,
  colors.use = c(Present = "#1e3d59", Absent = "#bccbcd"),
  legend.position = "bottom",
  reverse.levels = TRUE,
  axis.text.x.angle = 45,
  font.size = 10,
  font.type = "sans",
  plot.title = paste0("GO | ", GO_ontology),
  plot.subtitle = NULL,
  plot.caption = NULL,
  verbose = FALSE,
  return_matrices = FALSE,
  grid.color = "white",
  border.color = "black",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_GroupedGOTermPlot_+3A_genes">genes</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of gene symbols to query for functional annotation.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_org.db">org.db</code></td>
<td>
<p><strong><code>OrgDB</code></strong> | Database object to use for the query.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_levels.use">levels.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Vector of numerics corresponding to the GO ontology levels to plot. If NULL will compute all recursively until there are no results.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_go_ontology">GO_ontology</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | GO ontology to use. One of:
</p>

<ul>
<li> <p><em><code>BP</code></em>: For <strong>B</strong>iological <strong>P</strong>rocess.
</p>
</li>
<li> <p><em><code>MF</code></em>: For <strong>M</strong>olecular <strong>F</strong>unction.
</p>
</li>
<li> <p><em><code>CC</code></em>: For <strong>C</strong>ellular <strong>C</strong>omponent.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_min.overlap">min.overlap</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Filter the output result to the terms which are supported by this many genes.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Named vector with two colors assigned to the names <strong><code>Present</code></strong> and <strong><code>Absent</code></strong>.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_reverse.levels">reverse.levels</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to place the higher levels first when computing the joint heatmap.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_plot.title">plot.title</code>, <code id="do_GroupedGOTermPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_GroupedGOTermPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_verbose">verbose</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show extra comments, warnings,etc.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_return_matrices">return_matrices</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Returns the matrices of grouped GO terms.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_GroupedGOTermPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_GroupedGOTermPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_GroupedGOTermPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_GroupedGOTermPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_GroupedGOTermPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_GroupedGOTermPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_GroupedGOTermPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing all the matrices for the respective GO levels and all the individual and combined heatmaps.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_GroupedGOTermPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Need to load this library or equivalent.
    suppressMessages(library("org.Hs.eg.db"))

    # Define list of genes to query.
    genes.use &lt;- c("CCR7", "CD14", "LYZ",
                   "S100A4", "MS4A1",
                   "MS4A7", "GNLY", "NKG7", "FCER1A",
                   "CST3", "PPBP")

    # Compute the grouped GO terms.
    out &lt;- SCpubr::do_GroupedGOTermPlot(genes = genes.use,
                                        org.db = org.Hs.eg.db)
  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_GroupwiseDEPlot'>Compute a heatmap with the results of a group-wise DE analysis.</h2><span id='topic+do_GroupwiseDEPlot'></span>

<h3>Description</h3>

<p>Compute a heatmap with the results of a group-wise DE analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_GroupwiseDEPlot(
  sample,
  de_genes,
  group.by = NULL,
  number.breaks = 5,
  top_genes = 5,
  use_viridis = FALSE,
  viridis.direction = -1,
  viridis.palette.pvalue = "C",
  viridis.palette.logfc = "E",
  viridis.palette.expression = "G",
  sequential.direction = 1,
  sequential.palette.pvalue = "YlGn",
  sequential.palette.logfc = "YlOrRd",
  sequential.palette.expression = "YlGnBu",
  assay = NULL,
  slot = "data",
  legend.position = "bottom",
  legend.width = 1,
  legend.length = 20,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.type = "colorbar",
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  min.cutoff = NA,
  max.cutoff = NA,
  na.value = "grey75",
  grid.color = "white",
  border.color = "black",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_GroupwiseDEPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_de_genes">de_genes</code></td>
<td>
<p><strong><code><a href="tibble.html#topic+tibble">tibble</a></code></strong> | DE genes matrix resulting of running <code>Seurat::FindAllMarkers()</code>.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_top_genes">top_genes</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Top N differentially expressed (DE) genes by group to retrieve.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_viridis.palette.pvalue">viridis.palette.pvalue</code>, <code id="do_GroupwiseDEPlot_+3A_viridis.palette.logfc">viridis.palette.logfc</code>, <code id="do_GroupwiseDEPlot_+3A_viridis.palette.expression">viridis.palette.expression</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Viridis color palettes for the p-value, logfc and expression heatmaps. A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_sequential.palette.pvalue">sequential.palette.pvalue</code>, <code id="do_GroupwiseDEPlot_+3A_sequential.palette.expression">sequential.palette.expression</code>, <code id="do_GroupwiseDEPlot_+3A_sequential.palette.logfc">sequential.palette.logfc</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Sequential palettes for p-value, logfc and expression heatmaps. Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_legend.length">legend.length</code>, <code id="do_GroupwiseDEPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_GroupwiseDEPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_min.cutoff">min.cutoff</code>, <code id="do_GroupwiseDEPlot_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_GroupwiseDEPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_GroupwiseDEPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_GroupwiseDEPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_GroupwiseDEPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_GroupwiseDEPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_GroupwiseDEPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_GroupwiseDEPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap composed of 3 main panels: -log10(adjusted p-value), log2(FC) and mean expression by cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_GroupwiseDEPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Compute DE genes and transform to a tibble.
    de_genes &lt;- readRDS(system.file("extdata/de_genes_example.rds", package = "SCpubr"))

    # Default output.
    p &lt;- SCpubr::do_GroupwiseDEPlot(sample = sample,
                                    de_genes = de_genes)

    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_NebulosaPlot'>Wrapper for Nebulosa::plot_density in Seurat.</h2><span id='topic+do_NebulosaPlot'></span>

<h3>Description</h3>

<p>Wrapper for Nebulosa::plot_density in Seurat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_NebulosaPlot(
  sample,
  features,
  slot = NULL,
  dims = c(1, 2),
  pt.size = 1,
  reduction = NULL,
  combine = TRUE,
  method = c("ks", "wkde"),
  joint = FALSE,
  return_only_joint = FALSE,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  legend.type = "colorbar",
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.length = 20,
  legend.width = 1,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  font.size = 14,
  font.type = "sans",
  legend.position = "bottom",
  plot_cell_borders = TRUE,
  border.size = 2,
  border.color = "black",
  viridis.palette = "G",
  viridis.direction = 1,
  verbose = TRUE,
  na.value = "grey75",
  plot.axes = FALSE,
  number.breaks = 5,
  use_viridis = FALSE,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_NebulosaPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_features">features</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Features to represent.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_dims">dims</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Vector of 2 numerics indicating the dimensions to plot out of the selected reduction. Defaults to c(1, 2) if not specified.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_reduction">reduction</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Reduction to use. Can be the canonical ones such as &quot;umap&quot;, &quot;pca&quot;, or any custom ones, such as &quot;diffusion&quot;. If you are unsure about which reductions you have, use <code>Seurat::Reductions(sample)</code>. Defaults to &quot;umap&quot; if present or to the last computed reduction if the argument is not provided.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_combine">combine</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to create a single plot out of multiple features.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_method">method</code></td>
<td>
<p>Kernel density estimation method:
</p>

<ul>
<li> <p><code>ks</code>: Computes density using the <code>kde</code> function from the
<code>ks</code> package.
</p>
</li>
<li> <p><code>wkde</code>: Computes density using a modified version of the
<code>kde2d</code> function from the <code>MASS</code>
package to allow weights. Bandwidth selection from the <code>ks</code> package
is used instead.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_joint">joint</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot different features as joint density.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_return_only_joint">return_only_joint</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to only return the joint density panel.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_plot.title">plot.title</code>, <code id="do_NebulosaPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_NebulosaPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_NebulosaPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_legend.length">legend.length</code>, <code id="do_NebulosaPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_plot_cell_borders">plot_cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot border around cells.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_verbose">verbose</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show extra comments, warnings,etc.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_plot.axes">plot.axes</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot axes or not.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_NebulosaPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_NebulosaPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_NebulosaPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_NebulosaPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_NebulosaPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_NebulosaPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_NebulosaPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object containing a Nebulosa plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_NebulosaPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Basic Nebulosa plot.
    p &lt;- SCpubr::do_NebulosaPlot(sample = sample,
                                 features = "EPC1")

    # Compute joint density.
    p &lt;- SCpubr::do_NebulosaPlot(sample = sample,
                                 features = c("EPC1", "TOX2"),
                                 joint = TRUE)

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_PathwayActivityPlot'>Plot Pathway Activities from decoupleR using Progeny prior knowledge.</h2><span id='topic+do_PathwayActivityPlot'></span>

<h3>Description</h3>

<p>Plot Pathway Activities from decoupleR using Progeny prior knowledge.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_PathwayActivityPlot(
  sample,
  activities,
  group.by = NULL,
  split.by = NULL,
  slot = "scale.data",
  statistic = "norm_wmean",
  pt.size = 1,
  border.size = 2,
  na.value = "grey75",
  legend.position = "bottom",
  legend.width = 1,
  legend.length = 20,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.type = "colorbar",
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  enforce_symmetry = TRUE,
  min.cutoff = NA,
  max.cutoff = NA,
  number.breaks = 5,
  diverging.palette = "RdBu",
  diverging.direction = -1,
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  flip = FALSE,
  return_object = FALSE,
  grid.color = "white",
  border.color = "black",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_PathwayActivityPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_activities">activities</code></td>
<td>
<p><strong><code><a href="tibble.html#topic+tibble">tibble</a></code></strong> | Result of running decoupleR method with progeny regulon prior knowledge.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_statistic">statistic</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | DecoupleR statistic to use. One of:
</p>

<ul>
<li> <p><em><code>wmean</code></em>: For weighted mean.
</p>
</li>
<li> <p><em><code>norm_wmean</code></em>: For normalized weighted mean.
</p>
</li>
<li> <p><em><code>corr_wmean</code></em>: For corrected weighted mean.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_legend.length">legend.length</code>, <code id="do_PathwayActivityPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_PathwayActivityPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return a symmetrical plot axes-wise or continuous color scale-wise, when applicable.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_min.cutoff">min.cutoff</code>, <code id="do_PathwayActivityPlot_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_return_object">return_object</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Returns the Seurat object with the modifications performed in the function. Nomally, this contains a new assay with the data that can then be used for any other visualization desired.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_PathwayActivityPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_PathwayActivityPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_PathwayActivityPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_PathwayActivityPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_PathwayActivityPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_PathwayActivityPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_PathwayActivityPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_PathwayActivityPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds",
                                  package = "SCpubr"))

    # Define your activities object.
    progeny_activities &lt;- readRDS(system.file("extdata/progeny_activities_example.rds",
                                              package = "SCpubr"))

    # General heatmap.
    out &lt;- SCpubr::do_PathwayActivityPlot(sample = sample,
                                          activities = progeny_activities)
    p &lt;- out$heatmaps$average_scores
    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_RidgePlot'>Create ridge plots.</h2><span id='topic+do_RidgePlot'></span>

<h3>Description</h3>

<p>This function computes ridge plots based on the <span class="pkg">ggridges</span> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_RidgePlot(
  sample,
  feature,
  group.by = NULL,
  split.by = NULL,
  assay = "SCT",
  slot = "data",
  continuous_scale = FALSE,
  legend.title = NULL,
  legend.ncol = NULL,
  legend.nrow = NULL,
  legend.byrow = FALSE,
  legend.position = NULL,
  legend.width = 1,
  legend.length = 20,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.type = "colorbar",
  colors.use = NULL,
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  xlab = NULL,
  ylab = NULL,
  compute_quantiles = FALSE,
  compute_custom_quantiles = FALSE,
  quantiles = c(0.25, 0.5, 0.75),
  compute_distribution_tails = FALSE,
  prob_tails = 0.025,
  color_by_probabilities = FALSE,
  use_viridis = TRUE,
  viridis.palette = "G",
  viridis.direction = 1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  plot.grid = TRUE,
  grid.color = "grey75",
  grid.type = "dashed",
  flip = FALSE,
  number.breaks = 5,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_RidgePlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_feature">feature</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Feature to represent.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_continuous_scale">continuous_scale</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to color the ridges depending on a categorical or continuous scale.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.nrow">legend.nrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of rows in the legend.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the legend is filled by row or not.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.length">legend.length</code>, <code id="do_RidgePlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_RidgePlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Named vector of colors to use. Has to match the unique values of group.by or color.by (if used) when scale_type is set to categorical.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_plot.title">plot.title</code>, <code id="do_RidgePlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_RidgePlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_xlab">xlab</code>, <code id="do_RidgePlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_compute_quantiles">compute_quantiles</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to compute quantiles of the distribution and color the ridge plots by them.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_compute_custom_quantiles">compute_custom_quantiles</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to compute custom quantiles.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_quantiles">quantiles</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Numeric vector of quantiles.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_compute_distribution_tails">compute_distribution_tails</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to compute distribution tails and color them.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_prob_tails">prob_tails</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | The accumulated probability that the tails should contain.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_color_by_probabilities">color_by_probabilities</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to color the ridges depending on the probability.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_plot.grid">plot.grid</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot grid lines.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_grid.type">grid.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of the possible linetype options:
</p>

<ul>
<li> <p><em><code>blank</code></em>.
</p>
</li>
<li> <p><em><code>solid</code></em>.
</p>
</li>
<li> <p><em><code>dashed</code></em>.
</p>
</li>
<li> <p><em><code>dotted</code></em>.
</p>
</li>
<li> <p><em><code>dotdash</code></em>.
</p>
</li>
<li> <p><em><code>longdash</code></em>.
</p>
</li>
<li> <p><em><code>twodash</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_RidgePlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_RidgePlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_RidgePlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_RidgePlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_RidgePlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_RidgePlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_RidgePlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_RidgePlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Compute the most basic ridge plot.
    p &lt;- SCpubr::do_RidgePlot(sample = sample,
                              feature = "nFeature_RNA")
    p

    # Use continuous color scale.
    p &lt;- SCpubr::do_RidgePlot(sample = sample,
                              feature = "nFeature_RNA",
                              continuous_scale = TRUE,
                              viridis.direction = 1)
    p

    # Draw quantiles of the distribution.
    p &lt;- SCpubr::do_RidgePlot(sample = sample,
                              feature = "nFeature_RNA",
                              continuous_scale = TRUE,
                              compute_quantiles = TRUE,
                              compute_custom_quantiles = TRUE)
    p

    # Draw probability tails.
    p &lt;- SCpubr::do_RidgePlot(sample = sample,
                              feature = "nFeature_RNA",
                              continuous_scale = TRUE,
                              compute_quantiles = TRUE,
                              compute_distribution_tails = TRUE)
    p

    # Draw probability tails.
    p &lt;- SCpubr::do_RidgePlot(sample = sample,
                              feature = "nFeature_RNA",
                              continuous_scale = TRUE,
                              compute_quantiles = TRUE,
                              color_by_probabilities = TRUE)
    p
  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_TermEnrichmentPlot'>Display the enriched terms for a given list of genes.</h2><span id='topic+do_TermEnrichmentPlot'></span>

<h3>Description</h3>

<p>Display the enriched terms for a given list of genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_TermEnrichmentPlot(
  enriched_terms,
  nchar_wrap = 20,
  nterms = 10,
  font.size = 14,
  font.type = "sans",
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  legend.position = "bottom",
  legend.type = "colorbar",
  colors.use = NULL,
  text_labels_size = 4,
  legend.length = 30,
  legend.width = 1,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_TermEnrichmentPlot_+3A_enriched_terms">enriched_terms</code></td>
<td>
<p><strong><code><a href="base.html#topic+list">list</a></code></strong> | List containing the output(s) of running Enrichr.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_nchar_wrap">nchar_wrap</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of characters to use as a limit to wrap the term names. The higher this value, the longer the lines would be for each term in the plots. Defaults to 60.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_nterms">nterms</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of terms to report for each database. Terms are arranged by adjusted p-value and selected from lowest to highest. Defaults to 5.
</p>

<ul>
<li> <p><em><code>Enrichr</code></em>.
</p>
</li>
<li> <p><em><code>FlyEnrichr</code></em>.
</p>
</li>
<li> <p><em><code>WormEnrichr</code></em>.
</p>
</li>
<li> <p><em><code>YeastEnrichr</code></em>.
</p>
</li>
<li> <p><em><code>FishEnrichr</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_plot.title">plot.title</code>, <code id="do_TermEnrichmentPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_TermEnrichmentPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Character vector of 2 colors (low and high ends of the color scale) to generate the gradient.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_text_labels_size">text_labels_size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls how big or small labels are in the plot.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_legend.length">legend.length</code>, <code id="do_TermEnrichmentPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_TermEnrichmentPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_TermEnrichmentPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_TermEnrichmentPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_TermEnrichmentPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_TermEnrichmentPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_TermEnrichmentPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_TermEnrichmentPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_TermEnrichmentPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object with enriched terms.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_TermEnrichmentPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your enriched terms.
    enriched_terms &lt;- readRDS(system.file("extdata/enriched_terms_example.rds", package = "SCpubr"))
    enriched_terms$GO_Cellular_Component_2021 &lt;- NULL
    enriched_terms$Azimuth_Cell_Types_2021 &lt;- NULL

    # Default plot.
    p &lt;- SCpubr::do_TermEnrichmentPlot(enriched_terms = enriched_terms)
    p
  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_TFActivityPlot'>Plot TF Activities from decoupleR using Dorothea prior knowledge.</h2><span id='topic+do_TFActivityPlot'></span>

<h3>Description</h3>

<p>Plot TF Activities from decoupleR using Dorothea prior knowledge.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_TFActivityPlot(
  sample,
  activities,
  n_tfs = 25,
  slot = "scale.data",
  statistic = "norm_wmean",
  tfs.use = NULL,
  group.by = NULL,
  split.by = NULL,
  na.value = "grey75",
  legend.position = "bottom",
  legend.width = 1,
  legend.length = 20,
  legend.framewidth = 0.5,
  legend.tickwidth = 0.5,
  legend.framecolor = "grey50",
  legend.tickcolor = "white",
  legend.type = "colorbar",
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  enforce_symmetry = TRUE,
  diverging.palette = "RdBu",
  diverging.direction = -1,
  use_viridis = FALSE,
  viridis.palette = "G",
  viridis.direction = -1,
  sequential.palette = "YlGnBu",
  sequential.direction = 1,
  min.cutoff = NA,
  max.cutoff = NA,
  number.breaks = 5,
  flip = FALSE,
  return_object = FALSE,
  grid.color = "white",
  border.color = "black",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_TFActivityPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_activities">activities</code></td>
<td>
<p><strong><code><a href="tibble.html#topic+tibble">tibble</a></code></strong> | Result of running decoupleR method with dorothea regulon prior knowledge.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_n_tfs">n_tfs</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of top regulons to consider for downstream analysis.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_statistic">statistic</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | DecoupleR statistic to use. One of:
</p>

<ul>
<li> <p><em><code>wmean</code></em>: For weighted mean.
</p>
</li>
<li> <p><em><code>norm_wmean</code></em>: For normalized weighted mean.
</p>
</li>
<li> <p><em><code>corr_wmean</code></em>: For corrected weighted mean.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_tfs.use">tfs.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Restrict the analysis to given regulons.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_legend.length">legend.length</code>, <code id="do_TFActivityPlot_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_legend.framewidth">legend.framewidth</code>, <code id="do_TFActivityPlot_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the geyser and feature plot has a symmetrical color scale.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_min.cutoff">min.cutoff</code>, <code id="do_TFActivityPlot_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_return_object">return_object</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Returns the Seurat object with the modifications performed in the function. Nomally, this contains a new assay with the data that can then be used for any other visualization desired.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_TFActivityPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_TFActivityPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_TFActivityPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_TFActivityPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_TFActivityPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_TFActivityPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_TFActivityPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_TFActivityPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds",
                                  package = "SCpubr"))

    # Define your activities object.
    dorothea_activities &lt;- readRDS(system.file("extdata/dorothea_activities_example.rds",
                                               package = "SCpubr"))

    # General heatmap.
    out &lt;- SCpubr::do_TFActivityPlot(sample = sample,
                                     activities = dorothea_activities)
    p &lt;- out$heatmaps$average_scores
    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_ViolinPlot'>Wrapper for <a href="Seurat.html#topic+VlnPlot">VlnPlot</a>.</h2><span id='topic+do_ViolinPlot'></span>

<h3>Description</h3>

<p>Wrapper for <a href="Seurat.html#topic+VlnPlot">VlnPlot</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_ViolinPlot(
  sample,
  features,
  assay = NULL,
  slot = NULL,
  group.by = NULL,
  split.by = NULL,
  colors.use = NULL,
  pt.size = 0,
  line_width = 0.5,
  y_cut = rep(NA, length(features)),
  plot_boxplot = TRUE,
  boxplot_width = 0.2,
  legend.position = "bottom",
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  xlab = rep(NA, length(features)),
  ylab = rep(NA, length(features)),
  font.size = 14,
  font.type = "sans",
  axis.text.x.angle = 45,
  plot.grid = TRUE,
  grid.color = "grey75",
  grid.type = "dashed",
  flip = FALSE,
  ncol = NULL,
  share.y.lims = FALSE,
  legend.title = NULL,
  legend.title.position = "top",
  legend.ncol = NULL,
  legend.nrow = NULL,
  legend.byrow = FALSE,
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_ViolinPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_features">features</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Features to represent.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of valid color representations (either name of HEX codes) with as many named colors as unique values of group.by. If group.by is not provided, defaults to the unique values of <a href="Seurat.html#topic+Idents">Idents</a>. If not provided, a color scale will be set by default.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of points in the Violin plot.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_line_width">line_width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines drawn in the plot. Defaults to 1.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_y_cut">y_cut</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Vector with the values in which the Violins should be cut. Only works for one feature.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_plot_boxplot">plot_boxplot</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot a Box plot inside the violin or not.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_boxplot_width">boxplot_width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the boxplots. Defaults to 0.2.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_plot.title">plot.title</code>, <code id="do_ViolinPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_ViolinPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_xlab">xlab</code>, <code id="do_ViolinPlot_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_plot.grid">plot.grid</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot grid lines.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_grid.type">grid.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of the possible linetype options:
</p>

<ul>
<li> <p><em><code>blank</code></em>.
</p>
</li>
<li> <p><em><code>solid</code></em>.
</p>
</li>
<li> <p><em><code>dashed</code></em>.
</p>
</li>
<li> <p><em><code>dotted</code></em>.
</p>
</li>
<li> <p><em><code>dotdash</code></em>.
</p>
</li>
<li> <p><em><code>longdash</code></em>.
</p>
</li>
<li> <p><em><code>twodash</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_ncol">ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns used in the arrangement of the output plot using &quot;split.by&quot; parameter.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_share.y.lims">share.y.lims</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | When querying multiple features, force the Y axis of all of them to be on the same range of values (this being the max and min of all features combined).</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_legend.title.position">legend.title.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position for the title of the legend. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the legend.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the legend.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the legend.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_legend.nrow">legend.nrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of rows in the legend.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the legend is filled by row or not.</p>
</td></tr>
<tr><td><code id="do_ViolinPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_ViolinPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_ViolinPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_ViolinPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_ViolinPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_ViolinPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_ViolinPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object containing a Violin Plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_ViolinPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Basic violin plot.
    p &lt;- SCpubr::do_ViolinPlot(sample = sample,
                               feature = "nCount_RNA")
    p

    # Remove the box plots.
    p &lt;- SCpubr::do_ViolinPlot(sample = sample,
                               feature = "nCount_RNA",
                               plot_boxplot = FALSE)
    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='do_VolcanoPlot'>Compute a Volcano plot out of DE genes.</h2><span id='topic+do_VolcanoPlot'></span>

<h3>Description</h3>

<p>Compute a Volcano plot out of DE genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_VolcanoPlot(
  sample,
  de_genes,
  pval_cutoff = 0.05,
  FC_cutoff = 2,
  pt.size = 2,
  border.size = 1.5,
  border.color = "black",
  font.size = 14,
  font.type = "sans",
  plot.title = NULL,
  plot.subtitle = NULL,
  plot.caption = NULL,
  plot_lines = TRUE,
  line_color = "grey75",
  line_size = 0.5,
  add_gene_tags = TRUE,
  order_tags_by = "both",
  n_genes = 5,
  use_labels = FALSE,
  colors.use = "steelblue",
  plot.title.face = "bold",
  plot.subtitle.face = "plain",
  plot.caption.face = "italic",
  axis.title.face = "bold",
  axis.text.face = "plain",
  legend.title.face = "bold",
  legend.text.face = "plain"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="do_VolcanoPlot_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_de_genes">de_genes</code></td>
<td>
<p><strong><code><a href="tibble.html#topic+tibble">tibble</a></code></strong> | Output of <code>Seurat::FindMarkers()</code>.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_pval_cutoff">pval_cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Cutoff for the p-value.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_fc_cutoff">FC_cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Cutoff for the avg_log2FC.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_plot.title">plot.title</code>, <code id="do_VolcanoPlot_+3A_plot.subtitle">plot.subtitle</code>, <code id="do_VolcanoPlot_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_plot_lines">plot_lines</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot the division lines.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_line_color">line_color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the lines.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_line_size">line_size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the lines in the plot.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_add_gene_tags">add_gene_tags</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot the top genes.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_order_tags_by">order_tags_by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Either &quot;both&quot;, &quot;pvalue&quot; or &quot;logfc&quot;.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_n_genes">n_genes</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of top genes in each side to plot.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_use_labels">use_labels</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use labels instead of text for the tags.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color to generate a tetradic color scale with.</p>
</td></tr>
<tr><td><code id="do_VolcanoPlot_+3A_plot.title.face">plot.title.face</code>, <code id="do_VolcanoPlot_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="do_VolcanoPlot_+3A_plot.caption.face">plot.caption.face</code>, <code id="do_VolcanoPlot_+3A_axis.title.face">axis.title.face</code>, <code id="do_VolcanoPlot_+3A_axis.text.face">axis.text.face</code>, <code id="do_VolcanoPlot_+3A_legend.title.face">legend.title.face</code>, <code id="do_VolcanoPlot_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A volcano plot as a ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Check Suggests.
  value &lt;- SCpubr:::check_suggests(function_name = "do_VolcanoPlot", passive = TRUE)

  if (isTRUE(value)){
    # Consult the full documentation in https://enblacar.github.io/SCpubr-book/

    # Define your Seurat object.
    sample &lt;- readRDS(system.file("extdata/seurat_dataset_example.rds", package = "SCpubr"))

    # Retrieve DE genes.
    de_genes &lt;- readRDS(system.file("extdata/de_genes_example.rds", package = "SCpubr"))

    # Generate a volcano plot.
    p &lt;- SCpubr::do_VolcanoPlot(sample = sample,
                                de_genes = de_genes)
    p

  } else if (base::isFALSE(value)){
    message("This function can not be used without its suggested packages.")
    message("Check out which ones are needed using `SCpubr::state_dependencies()`.")
  }

</code></pre>

<hr>
<h2 id='doc_function'>Mock function used to document all main function.</h2><span id='topic+doc_function'></span>

<h3>Description</h3>

<p>Mock function used to document all main function.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="doc_function_+3A_sample">sample</code></td>
<td>
<p><strong><code><a href="SeuratObject.html#topic+Seurat">Seurat</a></code></strong> | A Seurat object, generated by <a href="Seurat.html#topic+CreateSeuratObject">CreateSeuratObject</a>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_font.size">font.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Overall font size of the plot. All plot elements will have a size relationship with this font size.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_font.type">font.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Base font family for the plot. One of:
</p>

<ul>
<li> <p><em><code>mono</code></em>: Mono spaced font.
</p>
</li>
<li> <p><em><code>serif</code></em>: Serif font family.
</p>
</li>
<li> <p><em><code>sans</code></em>: Default font family.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.type">legend.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of legend to display. One of:
</p>

<ul>
<li> <p><em><code>normal</code></em>: Default legend displayed by <span class="pkg">ggplot2</span>.
</p>
</li>
<li> <p><em><code>colorbar</code></em>: Redefined colorbar legend, using <a href="ggplot2.html#topic+guide_colorbar">guide_colorbar</a>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.position">legend.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position of the legend in the plot. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the figure.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the figure.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the figure.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the figure.
</p>
</li>
<li> <p><em><code>none</code></em>: No legend is displayed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.framewidth">legend.framewidth</code>, <code id="doc_function_+3A_legend.tickwidth">legend.tickwidth</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.framecolor">legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the lines of the box in the legend.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.tickcolor">legend.tickcolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the ticks of the box in the legend.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.length">legend.length</code>, <code id="doc_function_+3A_legend.width">legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Length and width of the legend. Will adjust automatically depending on legend side.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_plot.title">plot.title</code>, <code id="doc_function_+3A_plot.subtitle">plot.subtitle</code>, <code id="doc_function_+3A_plot.caption">plot.caption</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title, subtitle or caption to use in the plot.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_assay">assay</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Assay to use. Defaults to the current assay.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_slot">slot</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Data slot to use. Only one of: counts, data, scale.data. Defaults to &quot;data&quot;.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_reduction">reduction</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Reduction to use. Can be the canonical ones such as &quot;umap&quot;, &quot;pca&quot;, or any custom ones, such as &quot;diffusion&quot;. If you are unsure about which reductions you have, use <code>Seurat::Reductions(sample)</code>. Defaults to &quot;umap&quot; if present or to the last computed reduction if the argument is not provided.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_viridis.palette">viridis.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_raster">raster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to raster the resulting plot. This is recommendable if plotting a lot of cells.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_raster.dpi">raster.dpi</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Pixel resolution for rasterized plots. Defaults to 1024. Only activates on Seurat versions higher or equal than 4.1.0.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_plot_cell_borders">plot_cell_borders</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot border around cells.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_border.size">border.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width of the border of the cells.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_border.color">border.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color for the border of the heatmap body.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_na.value">na.value</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color value for NA.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_axis.text.x.angle">axis.text.x.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree to rotate the X labels. One of: 0, 45, 90.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_xlab">xlab</code>, <code id="doc_function_+3A_ylab">ylab</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Titles for the X and Y axis.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_pt.size">pt.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the dots.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_verbose">verbose</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to show extra comments, warnings,etc.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_flip">flip</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to invert the axis of the displayed plot.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_group.by">group.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Metadata variable to group the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_split.by">split.by</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Secondary metadata variable to further group (split) the output by. Has to be a character of factor column.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_colors.use">colors.use</code></td>
<td>
<p><strong><code><a href="#topic+named_vector">named_vector</a></code></strong> | Named vector of valid color representations (either name of HEX codes) with as many named colors as unique values of group.by. If group.by is not provided, defaults to the unique values of <a href="Seurat.html#topic+Idents">Idents</a>. If not provided, a color scale will be set by default.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.title">legend.title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the legend.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.icon.size">legend.icon.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the icons in legend.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.byrow">legend.byrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the legend is filled by row or not.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.nrow">legend.nrow</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of rows in the legend.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_plot_marginal_distributions">plot_marginal_distributions</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> |  Whether to plot marginal distributions on the figure or not.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_marginal.type">marginal.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of:
</p>

<ul>
<li> <p><em><code>density</code></em>: Compute density plots on the margins.
</p>
</li>
<li> <p><em><code>histogram</code></em>: Compute histograms on the margins.
</p>
</li>
<li> <p><em><code>boxplot</code></em>: Compute boxplot on the margins.
</p>
</li>
<li> <p><em><code>violin</code></em>: Compute violin plots on the margins.
</p>
</li>
<li> <p><em><code>densigram</code></em>: Compute densigram plots on the margins.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_marginal.size">marginal.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size ratio between the main and marginal plots. A value of 5 means that the main plot is 5 times bigger than the marginal plots.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_marginal.group">marginal.group</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to group the marginal distribution by group.by or current identities.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_enforce_symmetry">enforce_symmetry</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Return a symmetrical plot axes-wise or continuous color scale-wise, when applicable.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_column_title">column_title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the columns of the heatmaps. Only works with single heatmaps.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_row_title">row_title</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Title for the rows of the heatmaps. Only works with single heatmaps.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Cluster the columns or rows of the heatmaps.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Cluster the rows or rows of the heatmaps.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_column_names_rot">column_names_rot</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree in which to rotate the column labels.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_row_names_rot">row_names_rot</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree in which to rotate the row labels.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_cell_size">cell_size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of each cell in the heatmap.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_input_gene_list">input_gene_list</code></td>
<td>
<p><strong><code><a href="#topic+named_list">named_list</a></code></strong> | Named list of lists of genes to be used as input.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_column_title_rot">column_title_rot</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree in which to rotate the column titles.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_row_title_rot">row_title_rot</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Degree in which to rotate the row titles.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_column_names_side">column_names_side</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Side to put the column names. Either left or right.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_row_names_side">row_names_side</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Side to put the row names. Either left or right.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_column_title_side">column_title_side</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Side to put the column titles Either left or right.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_row_title_side">row_title_side</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Side to put the row titles Either left or right.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_heatmap.legend.length">heatmap.legend.length</code>, <code id="doc_function_+3A_heatmap.legend.width">heatmap.legend.width</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Width and length of the legend in the heatmap.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_heatmap.legend.framecolor">heatmap.legend.framecolor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the edges and ticks of the legend in the heatmap.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_scale_direction">scale_direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the viridis scales. Either -1 or 1.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_heatmap_gap">heatmap_gap</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Gap in cm between heatmaps.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend_gap">legend_gap</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Gap in cm between legends.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_cells.highlight">cells.highlight</code>, <code id="doc_function_+3A_idents.highlight">idents.highlight</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of cells/identities to focus into. The identities have to much those in <code>Seurat::Idents(sample)</code> The rest of the cells will be grayed out. Both parameters can be used at the same time.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_ncol">ncol</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of columns used in the arrangement of the output plot using &quot;split.by&quot; parameter.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_dims">dims</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Vector of 2 numerics indicating the dimensions to plot out of the selected reduction. Defaults to c(1, 2) if not specified.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_feature">feature</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Feature to represent.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_features">features</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Features to represent.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_use_viridis">use_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to use viridis color scales.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_viridis.direction">viridis.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Controls how the gradient of viridis scale is formed.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_plot.grid">plot.grid</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot grid lines.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_grid.color">grid.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the grid in the plot. In heatmaps, color of the border of the cells.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_grid.type">grid.type</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of the possible linetype options:
</p>

<ul>
<li> <p><em><code>blank</code></em>.
</p>
</li>
<li> <p><em><code>solid</code></em>.
</p>
</li>
<li> <p><em><code>dashed</code></em>.
</p>
</li>
<li> <p><em><code>dotted</code></em>.
</p>
</li>
<li> <p><em><code>dotdash</code></em>.
</p>
</li>
<li> <p><em><code>longdash</code></em>.
</p>
</li>
<li> <p><em><code>twodash</code></em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_plot.axes">plot.axes</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot axes or not.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_individual.titles">individual.titles</code>, <code id="doc_function_+3A_individual.subtitles">individual.subtitles</code>, <code id="doc_function_+3A_individual.captions">individual.captions</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector. Title, subtitle or caption to use in the plot when multiple features are passed on. Use NA to keep the original title.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_legend.title.position">legend.title.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Position for the title of the legend. One of:
</p>

<ul>
<li> <p><em><code>top</code></em>: Top of the legend.
</p>
</li>
<li> <p><em><code>bottom</code></em>: Bottom of the legend.
</p>
</li>
<li> <p><em><code>left</code></em>: Left of the legend.
</p>
</li>
<li> <p><em><code>right</code></em>: Right of the legend.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_repel">repel</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to repel the text labels.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_plot_density_contour">plot_density_contour</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot density contours in the UMAP.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_contour.position">contour.position</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Whether to plot density contours on top or at the bottom of the visualization layers, thus overlapping the clusters/cells or not.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_contour.color">contour.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the density lines.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_contour.lineend">contour.lineend</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Line end style (round, butt, square).</p>
</td></tr>
<tr><td><code id="doc_function_+3A_contour.linejoin">contour.linejoin</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Line join style (round, mitre, bevel).</p>
</td></tr>
<tr><td><code id="doc_function_+3A_contour_expand_axes">contour_expand_axes</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | To make the contours fit the plot, the limits of the X and Y axis are expanding a given percentage from the min and max values for each axis. This controls such percentage.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_label">label</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot the cluster labels in the UMAP. The cluster labels will have the same color as the cluster colors.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_label.color">label.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the labels in the plot.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_label.fill">label.fill</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color to fill the labels. Has to be a single color, that will be used for all labels. If <strong><code>NULL</code></strong>, the colors of the clusters will be used instead.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_label.size">label.size</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Size of the labels in the plot.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_label.box">label.box</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to plot the plot labels as <strong><code><a href="ggplot2.html#topic+geom_text">geom_text</a></code></strong> (FALSE) or <strong><code><a href="ggplot2.html#topic+geom_label">geom_label</a></code></strong> (TRUE).</p>
</td></tr>
<tr><td><code id="doc_function_+3A_min.overlap">min.overlap</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Filter the output result to the terms which are supported by this many genes.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_go_ontology">GO_ontology</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | GO ontology to use. One of:
</p>

<ul>
<li> <p><em><code>BP</code></em>: For <strong>B</strong>iological <strong>P</strong>rocess.
</p>
</li>
<li> <p><em><code>MF</code></em>: For <strong>M</strong>olecular <strong>F</strong>unction.
</p>
</li>
<li> <p><em><code>CC</code></em>: For <strong>C</strong>ellular <strong>C</strong>omponent.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_genes">genes</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Vector of gene symbols to query for functional annotation.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_org.db">org.db</code></td>
<td>
<p><strong><code>OrgDB</code></strong> | Database object to use for the query.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_disable_white_in_viridis">disable_white_in_viridis</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Remove the white in viridis color scale when <strong><code>viridis.direction</code></strong> is set to -1.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_number.breaks">number.breaks</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of breaks in continuous color scales of ggplot2-based plots.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_strip.spacing">strip.spacing</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the size between the different facets.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_strip.text.color">strip.text.color</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Color of the strip text.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_strip.text.angle">strip.text.angle</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Rotation of the strip text (angles).</p>
</td></tr>
<tr><td><code id="doc_function_+3A_diverging.palette">diverging.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of symmetrical color palette to use. Out of the diverging palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_diverging.direction">diverging.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Either 1 or -1. Direction of the divering palette. This basically flips the two ends.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_subsample">subsample</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of cells to subset for the analysis. NA will use all. Cells are selected at random.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_plot.title.face">plot.title.face</code>, <code id="doc_function_+3A_plot.subtitle.face">plot.subtitle.face</code>, <code id="doc_function_+3A_plot.caption.face">plot.caption.face</code>, <code id="doc_function_+3A_axis.title.face">axis.title.face</code>, <code id="doc_function_+3A_axis.text.face">axis.text.face</code>, <code id="doc_function_+3A_legend.title.face">legend.title.face</code>, <code id="doc_function_+3A_legend.text.face">legend.text.face</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Controls the style of the font for the corresponding theme element.  One of:
</p>

<ul>
<li> <p><em><code>plain</code></em>: For normal text.
</p>
</li>
<li> <p><em><code>italic</code></em>: For text in itallic.
</p>
</li>
<li> <p><em><code>bold</code></em>: For text in bold.
</p>
</li>
<li> <p><em><code>bold.italic</code></em>: For text both in itallic and bold.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_flavor">flavor</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | One of: Seurat, UCell. Compute the enrichment scores using <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a> or <a href="UCell.html#topic+AddModuleScore_UCell">AddModuleScore_UCell</a>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_features.order">features.order</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Should the gene sets be ordered in a specific way? Provide it as a vector of characters with the same names as the names of the gene sets.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_groups.order">groups.order</code></td>
<td>
<p><strong><code><a href="#topic+named_list">named_list</a></code></strong> | Should the groups in theheatmaps be ordered in a specific way? Provide it as a named list (as many lists as values in <strong><code>group.by</code></strong>) with the order for each of the elements in the groups.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_interpolate">interpolate</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Smoothes the output heatmap, saving space on disk when saving the image. However, the image is not as crisp.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_viridis.palette.pvalue">viridis.palette.pvalue</code>, <code id="doc_function_+3A_viridis.palette.logfc">viridis.palette.logfc</code>, <code id="doc_function_+3A_viridis.palette.expression">viridis.palette.expression</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Viridis color palettes for the p-value, logfc and expression heatmaps. A capital letter from A to H or the scale name as in <a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_nbin">nbin</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of bins to use in <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_ctrl">ctrl</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Number of genes in the control set to use in <a href="Seurat.html#topic+AddModuleScore">AddModuleScore</a>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_min.cutoff">min.cutoff</code>, <code id="doc_function_+3A_max.cutoff">max.cutoff</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Set the min/max ends of the color scale. Any cell/group with a value lower than min.cutoff will turn into min.cutoff and any cell with a value higher than max.cutoff will turn into max.cutoff. In FeaturePlots, provide as many values as features. Use NAs to skip a feature.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_border.density">border.density</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Controls the number of cells used when <strong><code>plot_cell_borders = TRUE</code></strong>. Value between 0 and 1. It computes a 2D kernel density and based on this cells that have a density below the specified quantile will be used to generate the cluster contour. The lower this number, the less cells will be selected, thus reducing the overall size of the plot but also potentially preventing all the contours to be properly drawn.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_sequential.palette">sequential.palette</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_sequential.palette.pvalue">sequential.palette.pvalue</code>, <code id="doc_function_+3A_sequential.palette.expression">sequential.palette.expression</code>, <code id="doc_function_+3A_sequential.palette.logfc">sequential.palette.logfc</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | Sequential palettes for p-value, logfc and expression heatmaps. Type of sequential color palette to use. Out of the sequential palettes defined in <strong><code><a href="RColorBrewer.html#topic+brewer.pal">brewer.pal</a></code></strong>.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_sequential.direction">sequential.direction</code></td>
<td>
<p><strong><code><a href="base.html#topic+numeric">numeric</a></code></strong> | Direction of the sequential color scale. Either 1 or -1.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_return_object">return_object</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Returns the Seurat object with the modifications performed in the function. Nomally, this contains a new assay with the data that can then be used for any other visualization desired.</p>
</td></tr>
<tr><td><code id="doc_function_+3A_statistic">statistic</code></td>
<td>
<p><strong><code><a href="base.html#topic+character">character</a></code></strong> | DecoupleR statistic to use. One of:
</p>

<ul>
<li> <p><em><code>wmean</code></em>: For weighted mean.
</p>
</li>
<li> <p><em><code>norm_wmean</code></em>: For normalized weighted mean.
</p>
</li>
<li> <p><em><code>corr_wmean</code></em>: For corrected weighted mean.
</p>
</li></ul>
</td></tr>
<tr><td><code id="doc_function_+3A_cluster">cluster</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether to perform clustering of rows and columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. This is a mock function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# This a mock function that stores the documentation for many other functions.
# It is not intended for user usage.
</code></pre>

<hr>
<h2 id='human_chr_locations'>Chromosome arm locations for human genome GRCh38.</h2><span id='topic+human_chr_locations'></span>

<h3>Description</h3>

<p>A tibble containing the chromosome, arm and start and end coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(human_chr_locations)
</code></pre>


<h3>Format</h3>

<p>A tibble with 48 rows and 4 columns:
</p>

<dl>
<dt>chr</dt><dd><p>Chromosome.</p>
</dd>
<dt>arm</dt><dd><p>Chromosome arm.</p>
</dd>
<dt>start</dt><dd><p>Start coordinates.</p>
</dd>
<dt>end</dt><dd><p>End coordinates.</p>
</dd>
</dl>


<hr>
<h2 id='named_list'>Named list.</h2><span id='topic+named_list'></span>

<h3>Description</h3>

<p>Named list.
</p>


<h3>Value</h3>

<p>Nothing. This is a mock function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This is a named vector.
x &lt;- list("first_element" = c("GENE A", "GENE B"),
          "second_element" = c("GENE C", "GENE D"))
print(x)

</code></pre>

<hr>
<h2 id='named_vector'>Named vector.</h2><span id='topic+named_vector'></span>

<h3>Description</h3>

<p>Named vector.
</p>


<h3>Value</h3>

<p>Nothing. This is a mock function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This is a named vector.
x &lt;- c("first_element" = 3,
       "second_element" = TRUE)
print(x)

</code></pre>

<hr>
<h2 id='package_report'>Generate a status report of SCpubr and its dependencies.</h2><span id='topic+package_report'></span>

<h3>Description</h3>

<p>This function generates a summary report of the installation status of SCpubr, which packages are still missing and which functions can or can not currently be used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>package_report(startup = FALSE, extended = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="package_report_+3A_startup">startup</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the message should be displayed at startup, therefore, also containing welcoming messages and tips. If <strong><code>FALSE</code></strong>, only the report itself will be printed.</p>
</td></tr>
<tr><td><code id="package_report_+3A_extended">extended</code></td>
<td>
<p><strong><code><a href="base.html#topic+logical">logical</a></code></strong> | Whether the message should also include installed packages, current and available version, and which <strong><code>SCpubr</code></strong> functions can be used with the currently installed packages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Print a package report.
SCpubr::package_report(startup = FALSE, extended = FALSE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
