<!DOCTYPE html><html><head><title>Help for package scquantum</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scquantum}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#l2e.normal.sd'><p>Functions for processing raw profiles</p></a></li>
<li><a href='#ploidy.inference'><p>Ploidy inference with quantogram</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Estimate Ploidy and Absolute Copy Number from Single Cell
Sequencing</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Given bincount data from single-cell copy number profiling (segmented or unsegmented), estimates ploidy, and uses the ploidy estimate to scale the data to absolute copy numbers. Uses the modular quantogram proposed by Kendall (1986) &lt;<a href="https://doi.org/10.1002%2F0471667196.ess2129.pub2">doi:10.1002/0471667196.ess2129.pub2</a>&gt;, modified by weighting segments according to confidence, and quantifying confidence in the estimate using a theoretical quantogram. Includes optional fused-lasso segmentation with the algorithm in Johnson (2013) &lt;<a href="https://doi.org/10.1080%2F10618600.2012.681238">doi:10.1080/10618600.2012.681238</a>&gt;, using the implementation from glmgen by Arnold, Sadhanala, and Tibshirani.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-03-16 17:28:46 UTC; alexdavis</td>
</tr>
<tr>
<td>Author:</td>
<td>Alexander Davis [aut, cre],
  Taylor Arnold [cph] (Author of segmentation function),
  Veeranjaneyulu Sadhanala [cph] (Author of segmentation function),
  Ryan Tibshirani [cph] (Author of segmentation function)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alexander Davis &lt;alexlovesmolecules@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-03-23 09:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='l2e.normal.sd'>Functions for processing raw profiles</h2><span id='topic+l2e.normal.sd'></span><span id='topic+gat'></span><span id='topic+tf.dp'></span><span id='topic+segment.summarize'></span><span id='topic+prof2invals'></span><span id='topic+seg2invals'></span><span id='topic+weighted.ecf'></span><span id='topic+ecf.global.max'></span><span id='topic+expected.peak.heights'></span><span id='topic+quantum.sd'></span><span id='topic+irises.pluspurple'></span><span id='topic+timeseries.iod'></span>

<h3>Description</h3>

<p>Functions for processing raw profiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>l2e.normal.sd(xs)

gat(x, iod)

tf.dp(y, lam)

segment.summarize(inprof, penalty, trans, seg, loc, se)

prof2invals(
  inprof,
  penalty,
  annotations,
  chrom.colname,
  bin.start.colname,
  bin.end.colname
)

seg2invals(seg_mean, seg_length, iod, annotations)

weighted.ecf(y, sds, s)

ecf.global.max(y, sds, smin = 1, smax = 8)

expected.peak.heights(cn, ratio.se, scale, svals)

quantum.sd(x, mu)

irises.pluspurple

timeseries.iod(v)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 8.
</p>

<hr>
<h2 id='ploidy.inference'>Ploidy inference with quantogram</h2><span id='topic+ploidy.inference'></span>

<h3>Description</h3>

<p>Infer ploidy of a cell, given a copy number profile.
Constructs a quantogram (either modular or cosine, depending on parameters).
The maximum of the quantogram is the estimated ploidy.
If unsegmented bincounts are given, segmentation will be performed using the
fused lasso.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ploidy.inference(
  x,
  chrom = NULL,
  start = NULL,
  end = NULL,
  penalty = 25,
  do_segmentation = TRUE,
  seg_length = NULL,
  iod = NULL,
  mean_bincount = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ploidy.inference_+3A_x">x</code></td>
<td>
<p>Bincounts or segment means</p>
</td></tr>
<tr><td><code id="ploidy.inference_+3A_chrom">chrom</code></td>
<td>
<p>Optional chromosome numbers</p>
</td></tr>
<tr><td><code id="ploidy.inference_+3A_start">start</code></td>
<td>
<p>Optional bin/segment start positions</p>
</td></tr>
<tr><td><code id="ploidy.inference_+3A_end">end</code></td>
<td>
<p>Optional bin/segment end positions</p>
</td></tr>
<tr><td><code id="ploidy.inference_+3A_penalty">penalty</code></td>
<td>
<p>If segmenting, penalty parameter for the fused lasso (higher penalty, fewer segments)</p>
</td></tr>
<tr><td><code id="ploidy.inference_+3A_do_segmentation">do_segmentation</code></td>
<td>
<p>Boolean, whether to do segmentation (set this to TRUE if giving unsegmented bincounts)</p>
</td></tr>
<tr><td><code id="ploidy.inference_+3A_seg_length">seg_length</code></td>
<td>
<p>If giving already segmented data, length of each segment</p>
</td></tr>
<tr><td><code id="ploidy.inference_+3A_iod">iod</code></td>
<td>
<p>If giving already segmented data, the index of dispersion of the bincount distribution (that is, within segments, not including between-segment variance)</p>
</td></tr>
<tr><td><code id="ploidy.inference_+3A_mean_bincount">mean_bincount</code></td>
<td>
<p>If giving already segmented ratio values, the original mean bincount</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ploidy inference object
</p>

<dl>
<dt>penalty</dt><dd><p>The segmentation penalty given as an argument, if any</p>
</dd>
<dt>multiply_ratios_by</dt><dd><p>To convert ratios to (unrounded) copy number estimates, multiply by this number</p>
</dd>
<dt>subtract_from_scaled_ratios</dt><dd><p>To convert ratios to (unrounded) copy numbers, after multiplying, subtract this number. Only required if the count data have some extra reads even at copy number 0, generally due to mapping problems</p>
</dd>
<dt>ploidy</dt><dd><p>The estimated ploidy</p>
</dd>
<dt>peak_height</dt><dd><p>The height of the quantogram peak at the estimated ploidy. Between 0 and 1. Higher values indicate a stronger signal</p>
</dd>
<dt>segmentation</dt><dd><p>The segmented values (either given as an argument, or produced interally by segmentation)</p>
</dd>
<dt>polar_quantogram</dt><dd><p>The complex-valued quantogram, whose absolute values measure consistency with each possible ploidy</p>
</dd>
<dt>bincounts</dt><dd><p>The raw bincounts given as an argument (if a segmentation was not given directly)</p>
</dd>
<dt>theoretical_quantogram</dt><dd><p>Based on the inferred copy numbers and index of dispersion, what the absolute value of the quantogram should look like. Deviation of this theoretical quantogram from the real one indicate that the ploidy estimate may be wrong</p>
</dd>
<dt>theoretical_peak_height</dt><dd><p>Height of the peak in the theoretical quantogram, measuring the expected strength of signal for the ploidy value</p>
</dd>
<dt>confidence_ratio</dt><dd><p>Ratio of actual to theoretical peak height. Values near (or above) 1 indicate the signal was as strong as would be expected gievn this data quality and ploidy; low values indicate that the ploidy inference may be wrong or that there are unexpected quality issues with the data</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'># Generating a random copy number profile
set.seed(705)
cns &lt;- rpois(30, 3) + 1
x &lt;- unlist(lapply(cns, function(cn) rpois(100, 25 * cn)))
annotations &lt;- data.frame(chrom = 1, start = 1:length(x), end = 1:length(x))

# Inferring ploidy
# Annotations and penalty are optional
estimate.from.bincounts &lt;- ploidy.inference(x, annotations$chrom, annotations$start, penalty = 25)

# Using scquantum internal functions to segment the data and estimate index
# of git@github.com:navinlabcode/scquantum.git
# dispersion
mu.est &lt;- mean(x)
iod.est &lt;- timeseries.iod(x)
seg &lt;- prof2invals(x, 25, annotations, "chrom", "start", "end")
mean.est &lt;- mean(x)
iod.est &lt;- timeseries.iod(x)
estimate.from.segmentation &lt;-
  ploidy.inference(
    seg$mean,
    seg$chrom,
    seg$start,
    seg$end,
    iod = iod.est,
    mean_bincount = mean.est,
    do_segmentation = FALSE
  )
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
