<!DOCTYPE html><html><head><title>Help for package SCRIP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SCRIP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#acinar.data'><p>parameter files estimated from acinar.data using splatEstimate</p></a></li>
<li><a href='#bridge'><p>Brownian bridge</p></a></li>
<li><a href='#bringItemsForward'><p>Bring items forward</p></a></li>
<li><a href='#getLNormFactors'><p>Get log-normal factors</p></a></li>
<li><a href='#getPathOrder'><p>Get path order</p></a></li>
<li><a href='#logistic'><p>Logistic function</p></a></li>
<li><a href='#params_acinar'><p>A data frame with 1000 genes and 80 cells</p></a></li>
<li><a href='#SCRIPsimBatchCellMeans'><p>Simulate batch means</p></a></li>
<li><a href='#SCRIPsimBatchEffects'><p>Simulate batch effects</p></a></li>
<li><a href='#SCRIPsimBCVMeans'><p>Simulate BCV means</p></a></li>
<li><a href='#SCRIPsimDropout'><p>Simulate dropout</p></a></li>
<li><a href='#SCRIPsimGeneMeans'><p>Simulate gene means</p></a></li>
<li><a href='#SCRIPsimGroupCellMeans'><p>Simulate Group CellMeans</p></a></li>
<li><a href='#SCRIPsimGroupDE'><p>Simulate group differential expression</p></a></li>
<li><a href='#SCRIPsimLibSizes'><p>Simulate library sizes</p></a></li>
<li><a href='#SCRIPsimPathCellMeans'><p>sim PathCellMeans</p></a></li>
<li><a href='#SCRIPsimPathDE'><p>Sim PathDE</p></a></li>
<li><a href='#SCRIPsimSingleCellMeans'><p>Simulate cell means</p></a></li>
<li><a href='#SCRIPsimTrueCounts'><p>Simulate true counts</p></a></li>
<li><a href='#SCRIPsimu'><p>SCRIP simulation</p></a></li>
<li><a href='#simu_cluster'><p>SCRIP simulation for clustering analysis with multiple cell types</p></a></li>
<li><a href='#simu_DE'><p>SCRIP simulation for differential expression</p></a></li>
<li><a href='#simu.VEGs'><p>SCRIP simulation for clustering analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>An Accurate Simulator for Single-Cell RNA Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-15</td>
</tr>
<tr>
<td>Description:</td>
<td>We provide a comprehensive scheme that is capable of simulating Single Cell RNA Sequencing data for various parameters of Biological Coefficient of Variation,
    busting kinetics, differential expression (DE), cell or sample groups, cell trajectory, batch effect and other experimental designs.
    'SCRIP' proposed and compared two frameworks with Gamma-Poisson and Beta-Gamma-Poisson models for simulating Single Cell RNA Sequencing data.
	Other reference is available in Zappia et al. (2017) <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1305-0">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1305-0</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>splatter(&ge; 1.16.1), S4Vectors(&ge; 0.30.0),
SummarizedExperiment(&ge; 1.22.0), SingleCellExperiment(&ge;
1.14.1), edgeR(&ge; 3.34.0), methods, stats, mgcv, knitr,
BiocManager, BiocGenerics, Seurat, crayon, fitdistrplus,
checkmate (&ge; 2.0.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/thecailab/SCRIP">https://github.com/thecailab/SCRIP</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-18 16:16:16 UTC; fqin</td>
</tr>
<tr>
<td>Author:</td>
<td>Fei Qin [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fei Qin &lt;fqin@email.sc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-19 07:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='acinar.data'>parameter files estimated from acinar.data using splatEstimate</h2><span id='topic+acinar.data'></span>

<h3>Description</h3>

<p>parameter files estimated from acinar.data using splatEstimate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acinar.data
</code></pre>


<h3>Format</h3>

<p>parameters estimated using splatEstimate
</p>

<hr>
<h2 id='bridge'>Brownian bridge</h2><span id='topic+bridge'></span>

<h3>Description</h3>

<p>Calculate a smoothed Brownian bridge between two points. A Brownian bridge is
a random walk with fixed end points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bridge(x = 0, y = 0, N = 5, n = 100, sigma.fac = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bridge_+3A_x">x</code></td>
<td>
<p>starting value.</p>
</td></tr>
<tr><td><code id="bridge_+3A_y">y</code></td>
<td>
<p>end value.</p>
</td></tr>
<tr><td><code id="bridge_+3A_n">N</code></td>
<td>
<p>number of steps in random walk.</p>
</td></tr>
<tr><td><code id="bridge_+3A_n">n</code></td>
<td>
<p>number of points in smoothed bridge.</p>
</td></tr>
<tr><td><code id="bridge_+3A_sigma.fac">sigma.fac</code></td>
<td>
<p>multiplier specifying how extreme each step can be.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of length n following a path from x to y.
</p>

<hr>
<h2 id='bringItemsForward'>Bring items forward</h2><span id='topic+bringItemsForward'></span>

<h3>Description</h3>

<p>Move selected items to the start of a list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bringItemsForward(ll, items)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bringItemsForward_+3A_ll">ll</code></td>
<td>
<p>list to adjust item order.</p>
</td></tr>
<tr><td><code id="bringItemsForward_+3A_items">items</code></td>
<td>
<p>vector of items to bring to the front. Any not in the list will
be ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with selected items first
</p>

<hr>
<h2 id='getLNormFactors'>Get log-normal factors</h2><span id='topic+getLNormFactors'></span>

<h3>Description</h3>

<p>Randomly generate multiplication factors from a log-normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLNormFactors(n.facs, sel.prob, neg.prob, fac.loc, fac.scale)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLNormFactors_+3A_n.facs">n.facs</code></td>
<td>
<p>Number of factors to generate.</p>
</td></tr>
<tr><td><code id="getLNormFactors_+3A_sel.prob">sel.prob</code></td>
<td>
<p>Probability that a factor will be selected to be different
from 1.</p>
</td></tr>
<tr><td><code id="getLNormFactors_+3A_neg.prob">neg.prob</code></td>
<td>
<p>Probability that a selected factor is less than one.</p>
</td></tr>
<tr><td><code id="getLNormFactors_+3A_fac.loc">fac.loc</code></td>
<td>
<p>Location parameter for the log-normal distribution.</p>
</td></tr>
<tr><td><code id="getLNormFactors_+3A_fac.scale">fac.scale</code></td>
<td>
<p>Scale factor for the log-normal distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector containing generated factors.
</p>

<hr>
<h2 id='getPathOrder'>Get path order</h2><span id='topic+getPathOrder'></span>

<h3>Description</h3>

<p>Identify the correct order to process paths so that preceding paths have
already been simulated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPathOrder(path.from)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPathOrder_+3A_path.from">path.from</code></td>
<td>
<p>vector giving the path endpoints that each path originates
from.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector giving the order to process paths in.
</p>

<hr>
<h2 id='logistic'>Logistic function</h2><span id='topic+logistic'></span>

<h3>Description</h3>

<p>Implementation of the logistic function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logistic(x, x0, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logistic_+3A_x">x</code></td>
<td>
<p>value to apply the function to.</p>
</td></tr>
<tr><td><code id="logistic_+3A_x0">x0</code></td>
<td>
<p>midpoint parameter. Gives the centre of the function.</p>
</td></tr>
<tr><td><code id="logistic_+3A_k">k</code></td>
<td>
<p>shape parameter. Gives the slope of the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Value of logistic function with given parameters
</p>

<hr>
<h2 id='params_acinar'>A data frame with 1000 genes and 80 cells</h2><span id='topic+params_acinar'></span>

<h3>Description</h3>

<p>A data frame with 1000 genes and 80 cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>params_acinar
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 genes and 80 cells
</p>

<hr>
<h2 id='SCRIPsimBatchCellMeans'>Simulate batch means</h2><span id='topic+SCRIPsimBatchCellMeans'></span>

<h3>Description</h3>

<p>Simulate a mean for each gene in each cell incorporating batch effect factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimBatchCellMeans(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimBatchCellMeans_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add batch means to.</p>
</td></tr>
<tr><td><code id="SCRIPsimBatchCellMeans_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with simulated batch means.
</p>

<hr>
<h2 id='SCRIPsimBatchEffects'>Simulate batch effects</h2><span id='topic+SCRIPsimBatchEffects'></span>

<h3>Description</h3>

<p>Simulate batch effects. Batch effect factors for each batch are produced
using <code><a href="#topic+getLNormFactors">getLNormFactors</a></code> and these are added along with updated
means for each batch.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimBatchEffects(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimBatchEffects_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add batch effects to.</p>
</td></tr>
<tr><td><code id="SCRIPsimBatchEffects_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with simulated batch effects.
</p>

<hr>
<h2 id='SCRIPsimBCVMeans'>Simulate BCV means</h2><span id='topic+SCRIPsimBCVMeans'></span>

<h3>Description</h3>

<p>Simulate means for each gene in each cell that are adjusted to follow a
mean-variance trend using Biological Coefficient of Variation taken from
and inverse gamma distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimBCVMeans(data, sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimBCVMeans_+3A_data">data</code></td>
<td>
<p>data are used to fit the mean-BCV trend for simulation</p>
</td></tr>
<tr><td><code id="SCRIPsimBCVMeans_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add BCV means to.</p>
</td></tr>
<tr><td><code id="SCRIPsimBCVMeans_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with simulated BCV means.
</p>

<hr>
<h2 id='SCRIPsimDropout'>Simulate dropout</h2><span id='topic+SCRIPsimDropout'></span>

<h3>Description</h3>

<p>A logistic function is used to form a relationship between the expression
level of a gene and the probability of dropout, giving a probability for each
gene in each cell. These probabilities are used in a Bernoulli distribution
to decide which counts should be dropped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimDropout(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimDropout_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add dropout to.</p>
</td></tr>
<tr><td><code id="SCRIPsimDropout_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with simulated dropout and observed counts.
</p>

<hr>
<h2 id='SCRIPsimGeneMeans'>Simulate gene means</h2><span id='topic+SCRIPsimGeneMeans'></span>

<h3>Description</h3>

<p>Simulate gene means from a gamma distribution. Also simulates outlier
expression factors. Genes with an outlier factor not equal to 1 are replaced
with the median mean expression multiplied by the outlier factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimGeneMeans(data, sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimGeneMeans_+3A_data">data</code></td>
<td>
<p>raw dataset.</p>
</td></tr>
<tr><td><code id="SCRIPsimGeneMeans_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add gene means to.</p>
</td></tr>
<tr><td><code id="SCRIPsimGeneMeans_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with simulated gene means.
</p>

<hr>
<h2 id='SCRIPsimGroupCellMeans'>Simulate Group CellMeans</h2><span id='topic+SCRIPsimGroupCellMeans'></span>

<h3>Description</h3>

<p>Simulate group cell means
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimGroupCellMeans(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimGroupCellMeans_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add cell means to.</p>
</td></tr>
<tr><td><code id="SCRIPsimGroupCellMeans_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with added cell means.
</p>

<hr>
<h2 id='SCRIPsimGroupDE'>Simulate group differential expression</h2><span id='topic+SCRIPsimGroupDE'></span>

<h3>Description</h3>

<p>Simulate differential expression. Differential expression factors for each
group are produced using <code><a href="#topic+getLNormFactors">getLNormFactors</a></code> and these are added
along with updated means for each group. For paths care is taken to make sure
they are simulated in the correct order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimGroupDE(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimGroupDE_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add differential expression to.</p>
</td></tr>
<tr><td><code id="SCRIPsimGroupDE_+3A_params">params</code></td>
<td>
<p>splatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with simulated differential expression.
</p>

<hr>
<h2 id='SCRIPsimLibSizes'>Simulate library sizes</h2><span id='topic+SCRIPsimLibSizes'></span>

<h3>Description</h3>

<p>Simulate expected library sizes. Typically a log-normal distribution is used but there is also the option to use a normal distribution. In this case any negative values are set to half the minimum non-zero value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimLibSizes(sim, params, libsize)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimLibSizes_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add library size to.</p>
</td></tr>
<tr><td><code id="SCRIPsimLibSizes_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
<tr><td><code id="SCRIPsimLibSizes_+3A_libsize">libsize</code></td>
<td>
<p>Provide the library size directly instread of using parameters to estimate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with simulated library sizes.
</p>

<hr>
<h2 id='SCRIPsimPathCellMeans'>sim PathCellMeans</h2><span id='topic+SCRIPsimPathCellMeans'></span>

<h3>Description</h3>

<p>simulate cell means for path
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimPathCellMeans(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimPathCellMeans_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add dropout to.</p>
</td></tr>
<tr><td><code id="SCRIPsimPathCellMeans_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with cell means for path simulation.
</p>

<hr>
<h2 id='SCRIPsimPathDE'>Sim PathDE</h2><span id='topic+SCRIPsimPathDE'></span>

<h3>Description</h3>

<p>simulate DE factors for path
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimPathDE(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimPathDE_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add dropout to.</p>
</td></tr>
<tr><td><code id="SCRIPsimPathDE_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with DE for path simulation.
</p>

<hr>
<h2 id='SCRIPsimSingleCellMeans'>Simulate cell means</h2><span id='topic+SCRIPsimSingleCellMeans'></span>

<h3>Description</h3>

<p>Simulate a gene by cell matrix giving the mean expression for each gene in
each cell. Cells start with the mean expression for the group they belong to
(when simulating groups) or cells are assigned the mean expression from a
random position on the appropriate path (when simulating paths). The selected
means are adjusted for each cell's expected library size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimSingleCellMeans(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimSingleCellMeans_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add cell means to.</p>
</td></tr>
<tr><td><code id="SCRIPsimSingleCellMeans_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with added cell means.
</p>

<hr>
<h2 id='SCRIPsimTrueCounts'>Simulate true counts</h2><span id='topic+SCRIPsimTrueCounts'></span>

<h3>Description</h3>

<p>Simulate a true counts matrix. Counts are simulated from a poisson
distribution where Each gene in each cell has it's own mean based on the
group (or path position), expected library size and BCV.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimTrueCounts(sim, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimTrueCounts_+3A_sim">sim</code></td>
<td>
<p>SingleCellExperiment to add true counts to.</p>
</td></tr>
<tr><td><code id="SCRIPsimTrueCounts_+3A_params">params</code></td>
<td>
<p>SplatParams object with simulation parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment with simulated true counts.
</p>

<hr>
<h2 id='SCRIPsimu'>SCRIP simulation</h2><span id='topic+SCRIPsimu'></span>

<h3>Description</h3>

<p>Simulate count data for single cell RNA-sequencing using SCIRP method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCRIPsimu(
  data,
  params,
  method = "single",
  base_allcellmeans_SC = NULL,
  pre.bcv.df = NULL,
  libsize = NULL,
  bcv.shrink = 1,
  Dropout_rate = NULL,
  mode = "GP-trendedBCV",
  de.prob = NULL,
  de.downProb = NULL,
  de.facLoc = NULL,
  de.facScale = NULL,
  path.skew = NULL,
  batch.facLoc = NULL,
  batch.facScale = NULL,
  path.nSteps = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCRIPsimu_+3A_data">data</code></td>
<td>
<p>data matrix required to fit the mean-BCV trend for simulation</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_params">params</code></td>
<td>
<p>SplatParams object containing parameters for the simulation</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_method">method</code></td>
<td>
<p>&quot;single&quot;, &quot;groups&quot; or &quot;paths&quot;</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_base_allcellmeans_sc">base_allcellmeans_SC</code></td>
<td>
<p>base mean vector provided to help setting DE analysis</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_pre.bcv.df">pre.bcv.df</code></td>
<td>
<p>BCV.df enables us to change the variation of BCV values</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_libsize">libsize</code></td>
<td>
<p>library size can be provided directly</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_bcv.shrink">bcv.shrink</code></td>
<td>
<p>factor to control the BCV levels</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_dropout_rate">Dropout_rate</code></td>
<td>
<p>factor to control the dropout rate directly</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_mode">mode</code></td>
<td>
<p>&quot;GP-commonBCV&quot;, &quot;BP-commonBCV&quot;, &quot;BP&quot;, &quot;BGP-commonBCV&quot; and &quot;BGP-trendedBCV&quot;</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_de.prob">de.prob</code></td>
<td>
<p>the proportion of DE genes</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_de.downprob">de.downProb</code></td>
<td>
<p>the proportion of down-regulated DE genes</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_de.facloc">de.facLoc</code></td>
<td>
<p>DE location factor</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_de.facscale">de.facScale</code></td>
<td>
<p>DE scale factor</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_path.skew">path.skew</code></td>
<td>
<p>Controls how likely cells are from the start or end point of the path</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_batch.facloc">batch.facLoc</code></td>
<td>
<p>DE location factor in batch</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_batch.facscale">batch.facScale</code></td>
<td>
<p>DE scale factor in batch</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_path.nsteps">path.nSteps</code></td>
<td>
<p>number of steps between the start point and end point for each path</p>
</td></tr>
<tr><td><code id="SCRIPsimu_+3A_...">...</code></td>
<td>
<p>Other parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SingleCellExperiment file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(params_acinar)
data(acinar.data)
sim_trend = SCRIPsimu(data=acinar.data, params=params_acinar, mode="GP-trendedBCV")

</code></pre>

<hr>
<h2 id='simu_cluster'>SCRIP simulation for clustering analysis with multiple cell types</h2><span id='topic+simu_cluster'></span>

<h3>Description</h3>

<p>Simulate count data for clustering analysis by preserving variably expressed genes with multiple cell types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simu_cluster(expre_data, pheno_data, CTlist, mode, nfeatures, seed = 2021)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simu_cluster_+3A_expre_data">expre_data</code></td>
<td>
<p>data matrix required for simulation</p>
</td></tr>
<tr><td><code id="simu_cluster_+3A_pheno_data">pheno_data</code></td>
<td>
<p>phenotype data information</p>
</td></tr>
<tr><td><code id="simu_cluster_+3A_ctlist">CTlist</code></td>
<td>
<p>cell types used for simulation</p>
</td></tr>
<tr><td><code id="simu_cluster_+3A_mode">mode</code></td>
<td>
<p>&quot;GP-commonBCV&quot;, &quot;BP-commonBCV&quot;, &quot;BP&quot;, &quot;BGP-commonBCV&quot; and &quot;BGP-trendedBCV&quot;</p>
</td></tr>
<tr><td><code id="simu_cluster_+3A_nfeatures">nfeatures</code></td>
<td>
<p>parameter required for FinalVariable function in Seurat package</p>
</td></tr>
<tr><td><code id="simu_cluster_+3A_seed">seed</code></td>
<td>
<p>seed used for simulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>simulated read counts data with cell type information
</p>

<hr>
<h2 id='simu_DE'>SCRIP simulation for differential expression</h2><span id='topic+simu_DE'></span>

<h3>Description</h3>

<p>Simulate count data for differential expression analysis using SCRIP
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simu_DE(
  expre_data,
  params,
  nGenes = NULL,
  nDE,
  ncells = NULL,
  FC,
  Dropout_rate = NULL,
  libsize = NULL,
  pre.bcv.df = NULL,
  bcv.shrink = 1,
  seed = 2021
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simu_DE_+3A_expre_data">expre_data</code></td>
<td>
<p>data matrix required for simulation</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_params">params</code></td>
<td>
<p>SplatParams object containing parameters for the simulation</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_ngenes">nGenes</code></td>
<td>
<p>number of genes simulated</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_nde">nDE</code></td>
<td>
<p>number of differentially expressed genes simulated</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_ncells">ncells</code></td>
<td>
<p>number of cells simulated</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_fc">FC</code></td>
<td>
<p>fold change rate simulated between two groups</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_dropout_rate">Dropout_rate</code></td>
<td>
<p>factor to control the dropout rate directly</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_libsize">libsize</code></td>
<td>
<p>library size used for simulation</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_pre.bcv.df">pre.bcv.df</code></td>
<td>
<p>BCV.df enables us to change the variation of BCV values</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_bcv.shrink">bcv.shrink</code></td>
<td>
<p>factor to control the BCV levels</p>
</td></tr>
<tr><td><code id="simu_DE_+3A_seed">seed</code></td>
<td>
<p>seed for simulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SummarizedExperiment files from both groups for DE analysis and DE genes index
</p>

<hr>
<h2 id='simu.VEGs'>SCRIP simulation for clustering analysis</h2><span id='topic+simu.VEGs'></span>

<h3>Description</h3>

<p>Simulate count data for clustering analysis by preserving variably expressed genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simu.VEGs(
  counts.matrix,
  params = params,
  base_allcellmeans,
  mode = "GP-trendedBCV",
  nCells,
  nfeatures = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simu.VEGs_+3A_counts.matrix">counts.matrix</code></td>
<td>
<p>data matrix required for simulation</p>
</td></tr>
<tr><td><code id="simu.VEGs_+3A_params">params</code></td>
<td>
<p>SplatParams object containing parameters for the simulation</p>
</td></tr>
<tr><td><code id="simu.VEGs_+3A_base_allcellmeans">base_allcellmeans</code></td>
<td>
<p>base cell means specified directly for simulating counts</p>
</td></tr>
<tr><td><code id="simu.VEGs_+3A_mode">mode</code></td>
<td>
<p>&quot;GP-commonBCV&quot;, &quot;BP-commonBCV&quot;, &quot;BP&quot;, &quot;BGP-commonBCV&quot; and &quot;BGP-trendedBCV&quot;</p>
</td></tr>
<tr><td><code id="simu.VEGs_+3A_ncells">nCells</code></td>
<td>
<p>number of cells simulated</p>
</td></tr>
<tr><td><code id="simu.VEGs_+3A_nfeatures">nfeatures</code></td>
<td>
<p>parameter required for FinalVariable function in Seurat package</p>
</td></tr>
</table>


<h3>Value</h3>

<p>simulated read counts data
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
