<!DOCTYPE html><html lang="en"><head><title>Help for package scrm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scrm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#scrm-package'><p>Simulating the evolution of biological sequences</p></a></li>
<li><a href='#scrm'><p>Simulate the evolution of biological sequences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulating the Evolution of Biological Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.7.5</td>
</tr>
<tr>
<td>Author:</td>
<td>Paul Staab [aut, cph],
  Zhu Sha [aut, cph],
  Dirk Metzler [aut, cre, cph, ths],
  Gerton Lunter [aut, cph, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dirk Metzler &lt;metzler@bio.lmu.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A coalescent simulator that allows the rapid simulation of
    biological sequences under neutral models of evolution, see
    Staab et al. (2015) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtu861">doi:10.1093/bioinformatics/btu861</a>&gt;.
    Different to other coalescent based simulations,
    it has an optional approximation parameter that
    allows for high accuracy while maintaining a linear run time cost for long
    sequences. It is optimized for simulating massive data sets as produced by Next-
    Generation Sequencing technologies for up to several thousand sequences.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/scrm/scrm-r">https://github.com/scrm/scrm-r</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/scrm/scrm-r/issues">https://github.com/scrm/scrm-r/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.2)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ape, knitr, rmarkdown, testthat (&ge; 0.9.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-08 09:15:53 UTC; dirk</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-08 13:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='scrm-package'>Simulating the evolution of biological sequences</h2><span id='topic+scrm-package'></span>

<h3>Description</h3>

<p>The Sequential Coalescent with Recombination Model (SCRM) is an
approximation of the Ancestral Recombination Graph. It can be used to
simulate the neutral evolution of chromosomes/biological sequences
subject to possibly complicated population structure.
The program <em>scrm</em> is an implementation of this model that is designed to
act as an drop-in replacement for the widely adopted coalescent simulator
<em>ms</em>. This package contains <em>scrm</em> along with an R interface.
</p>


<h3>Author(s)</h3>

<p>Paul Staab,
Zhu Sha,
Dirk Metzler &amp;
Gerton Lunter
</p>
<p>Maintainer: Paul Staab <a href="mailto:develop@paulstaab.de">develop@paulstaab.de</a>
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+scrm">scrm</a></code> for details on how to use <em>scrm</em>,
</p>
</li>
<li> <p><code>vignette('scrm-Arguments')</code> for an overview of command line arguments and
</p>
</li>
<li> <p><code>vignette('scrm-TreesForApe')</code> for an example on using
genealogies simulated with <em>scrm</em> with package 'ape'.
</p>
</li></ul>


<hr>
<h2 id='scrm'>Simulate the evolution of biological sequences</h2><span id='topic+scrm'></span>

<h3>Description</h3>

<p>This function provides an interface for calling <em>scrm</em> from R.
The command line options are passed via the <code>args</code> argument.
The vignette 'scrm-Arguments' contains details about the available options.
Summary statistics are converted into an R format. Additionally, there
is an option to write the original command line output into a file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scrm(args, file = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scrm_+3A_args">args</code></td>
<td>
<p>A string containing the command-line arguments for scrm. Look at
scrms vignette for a description of available arguments.</p>
</td></tr>
<tr><td><code id="scrm_+3A_file">file</code></td>
<td>
<p>If provided, scrm will additionally write it is output into
a file with the given file, using an ms-like text output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of summary statistics. Most summary statistics are
again a list, where each entry contains the value for one locus.
For the site frequency spectrum, the summary statistic is a matrix,
where each row contains the spectrum for one locus.
</p>


<h3>Seeding</h3>

<p>The R version of scrm uses random number from R's random generator.
Therefore, the '-seed' argument of the command-line version will be ignored,
and no seed is given in the output.
Use the R function <code><a href="base.html#topic+set.seed">set.seed</a></code> prior to calling this
function to ensure reproducibility of results.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>vignette('scrm-Arguments')</code> for an overview of command line arguments and
</p>
</li>
<li> <p><code>vignette('scrm-TreesForApe')</code> for an example on using
genealogies simulated with <em>scrm</em> with package 'ape'.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(789)
# 5 Chromosomes with 100 bases each with recombination and mutation
sum_stats &lt;- scrm('5 1 -r 3.1 100 -t 1.5 -T -L')
str(sum_stats)

# Simulate the site frequency spectrum at 3 loci. For each locus
# 10 Chromosomes of 1Mb length are sampled from two populations with
# migration inbetween.
scrm('10 3 -r 400 1000000 -l 100000 -I 2 4 6 0.5 -t 300 -oSFS')$sfs

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
