<!DOCTYPE html><html><head><title>Help for package scRNAstat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scRNAstat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#AJ064_small_last_sce'><p>Small 'AJ064' Seurat Data After Processed</p></a></li>
<li><a href='#AJ064_small_sce'><p>Small 'AJ064' Seurat Data Set</p></a></li>
<li><a href='#basic_filter'><p>basic_filter</p></a></li>
<li><a href='#basic_find_markers'><p>Basic Find Markers</p></a></li>
<li><a href='#basic_markers'><p>Basic Markers</p></a></li>
<li><a href='#basic_qc'><p>Basic Quality Control</p></a></li>
<li><a href='#basic_workflow'><p>Basic Workflow</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Pipeline to Process Single Cell RNAseq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yonghe Xia &lt;xiayh17@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A pipeline that can process single or multiple Single Cell RNAseq 
    samples primarily specializes in Clustering and Dimensionality Reduction. 
    Meanwhile we use common cell type marker genes for T cells, B cells, Myeloid cells, 
    Epithelial cells, and stromal cells (Fiboblast, Endothelial cells, Pericyte,
    Smooth muscle cells) to visualize the Seurat clusters, to facilitate labeling 
    them by biological names. Once users named each cluster, they can evaluate the 
    quality of them again and find the de novo marker genes also.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Seurat, ggplot2, stringr, clustree, magrittr, Matrix, dplyr,
patchwork</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-21 02:39:07 UTC; yonghe</td>
</tr>
<tr>
<td>Author:</td>
<td>Jianming Zeng [aut],
  Yonghe Xia [ctb, cre],
  Biotrainee group [cph, fnd]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-22 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='AJ064_small_last_sce'>Small 'AJ064' Seurat Data After Processed</h2><span id='topic+AJ064_small_last_sce'></span>

<h3>Description</h3>

<p>An object of class Seurat
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AJ064_small_last_sce
</code></pre>


<h3>Format</h3>

<p>An object of class <code>Seurat</code> with 627 rows and 800 columns.
</p>

<hr>
<h2 id='AJ064_small_sce'>Small 'AJ064' Seurat Data Set</h2><span id='topic+AJ064_small_sce'></span>

<h3>Description</h3>

<p>An object of class Seurat
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AJ064_small_sce
</code></pre>


<h3>Format</h3>

<p>An object of class <code>Seurat</code> with 713 rows and 1000 columns.
</p>

<hr>
<h2 id='basic_filter'>basic_filter</h2><span id='topic+basic_filter'></span>

<h3>Description</h3>

<p>filter the genes which show expression less than 3 cells.
filter the cells which percent_mito &lt; 25 &amp; percent_ribo &gt; 3 &amp; percent_hb &lt; 10
filter the cells which  nFeature_RNA &gt; 300 &amp; nFeature_RNA &lt; 8000
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basic_filter(sce)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basic_filter_+3A_sce">sce</code></td>
<td>
<p>An object of class Seurat</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sce.all.filt  An object of class Seurat
</p>


<h3>Examples</h3>

<pre><code class='language-R'>basic_filter(AJ064_small_sce)

</code></pre>

<hr>
<h2 id='basic_find_markers'>Basic Find Markers</h2><span id='topic+basic_find_markers'></span>

<h3>Description</h3>

<p>To find de 'novo' markers by 'FindAllMarkers' from Seurat with default setting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basic_find_markers(sce, group = "seurat_clusters", dir = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basic_find_markers_+3A_sce">sce</code></td>
<td>
<p>An object of class Seurat</p>
</td></tr>
<tr><td><code id="basic_find_markers_+3A_group">group</code></td>
<td>
<p>default:seurat_clusters, you can change it to  celltype</p>
</td></tr>
<tr><td><code id="basic_find_markers_+3A_dir">dir</code></td>
<td>
<p>path for saving results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sce.markers a data.frame of markers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
basic_find_markers(AJ064_small_last_sce,dir=tempdir())

</code></pre>

<hr>
<h2 id='basic_markers'>Basic Markers</h2><span id='topic+basic_markers'></span>

<h3>Description</h3>

<p>Basic Markers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basic_markers(sce, org = "human", group = "orig.ident", dir = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basic_markers_+3A_sce">sce</code></td>
<td>
<p>An object of class Seurat</p>
</td></tr>
<tr><td><code id="basic_markers_+3A_org">org</code></td>
<td>
<p>human or mouse, default: human</p>
</td></tr>
<tr><td><code id="basic_markers_+3A_group">group</code></td>
<td>
<p>default:'orig.ident', you can change it to 'seurat_clusters' or 'celltype'</p>
</td></tr>
<tr><td><code id="basic_markers_+3A_dir">dir</code></td>
<td>
<p>the path for saving the figures by 'DotPlot' with known famous markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of figures by 'DotPlot'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
basic_markers(AJ064_small_last_sce,dir=tempdir())

</code></pre>

<hr>
<h2 id='basic_qc'>Basic Quality Control</h2><span id='topic+basic_qc'></span>

<h3>Description</h3>

<p>add 'percent_mito','percent_ribo','percent_hb' to the Seurat class.
And draw 'VlnPlot' for these 'qc' values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basic_qc(sce, org = "human", group = "orig.ident", dir = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basic_qc_+3A_sce">sce</code></td>
<td>
<p>An object of class Seurat</p>
</td></tr>
<tr><td><code id="basic_qc_+3A_org">org</code></td>
<td>
<p>human or mouse, default: human</p>
</td></tr>
<tr><td><code id="basic_qc_+3A_group">group</code></td>
<td>
<p>default:'orig.ident',you can change it to 'seurat_clusters' or 'celltype'</p>
</td></tr>
<tr><td><code id="basic_qc_+3A_dir">dir</code></td>
<td>
<p>the path for saving the figures by 'DotPlot' with known famous markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list(p1,p2,p3,sce), the last one in the new 'sce'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>basic_qc(AJ064_small_sce,dir= tempdir())

</code></pre>

<hr>
<h2 id='basic_workflow'>Basic Workflow</h2><span id='topic+basic_workflow'></span>

<h3>Description</h3>

<p>the workflow from Seurat, including: 'NormalizeData','FindVariableFeatures','ScaleData',
'RunPCA','RunTSNE','RunUMAP','FindNeighbors','FindClusters(sce, resolution = seq(0.1,1,by=0.1))'
we use 'clustree' to check the different resolution for 'FindClusters'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basic_workflow(sce, dir = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basic_workflow_+3A_sce">sce</code></td>
<td>
<p>An object of class Seurat</p>
</td></tr>
<tr><td><code id="basic_workflow_+3A_dir">dir</code></td>
<td>
<p>the path for saving the figures by 'DotPlot' with known famous markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list(p1,p2,p3,sce), the last one in the new sce with PCA,tSNE,UMAP information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
basic_workflow(AJ064_small_sce,dir=tempdir())

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
