<!DOCTYPE html><html lang="en"><head><title>Help for package scRNAtools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scRNAtools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#scRNAtools-package'><p>Single Cell RNA Sequencing Data Analysis Tools</p></a></li>
<li><a href='#corr_re'><p>correlation index</p></a></li>
<li><a href='#DEGs'><p>Users interested genes or differentially expressed genes</p></a></li>
<li><a href='#exam'><p>exam</p></a></li>
<li><a href='#exam1'><p>exam1</p></a></li>
<li><a href='#example'><p>example</p></a></li>
<li><a href='#example1'><p>example1</p></a></li>
<li><a href='#scRNAtools_cluster'><p>Cluster section</p></a></li>
<li><a href='#scRNAtools_cor_map'><p>Correlation analysis</p></a></li>
<li><a href='#scRNAtools_cor_map_r'><p>Present correlation index in figure</p></a></li>
<li><a href='#scRNAtools_DEGsA'><p>Identification of differentially expressed genes</p></a></li>
<li><a href='#scRNAtools_Gene2exp_1'><p>Present the expression of two genes</p></a></li>
<li><a href='#scRNAtools_Gene3exp_1'><p>Present gene expression</p></a></li>
<li><a href='#scRNAtools_Geneexp'><p>Expressed genes in scRNA sequencing data</p></a></li>
<li><a href='#scRNAtools_Geneexp_1'><p>Present gene expression</p></a></li>
<li><a href='#scRNAtools_inter_net'><p>Construction of interactive network in scRNA sequencing data</p></a></li>
<li><a href='#scRNAtools_pca'><p>PCA analysis</p></a></li>
<li><a href='#scRNAtools_pca_3D'><p>3D PCA analysis</p></a></li>
<li><a href='#scRNAtools_PEA'><p>Pathway enrichment analysis</p></a></li>
<li><a href='#scRNAtools_tsne'><p>t-SNE analysis</p></a></li>
<li><a href='#types'><p>types</p></a></li>
<li><a href='#types_all'><p>types_all</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single Cell RNA Sequencing Data Analysis Tools</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-06-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Qian Yang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qian Yang &lt;bioqianyang@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We integrated the common analysis methods utilized in single cell RNA sequencing data, which included cluster method, principal components analysis (PCA), the filter of differentially expressed genes, pathway enrichment analysis and correlated analysis methods.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), foreach, base</td>
</tr>
<tr>
<td>Imports:</td>
<td>ALL, ConsensusClusterPlus, scatterplot3d, ggplot2, Rtsne,
limma, edgeR, TPEA, Rmisc, lattice, plyr, ggthemes, reshape2,
PerformanceAnalytics, corrplot, Hmisc, igraph, survival</td>
</tr>
<tr>
<td>Suggests:</td>
<td>grid</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-07-04 00:49:45 UTC; dell</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-05 14:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='scRNAtools-package'>Single Cell RNA Sequencing Data Analysis Tools</h2><span id='topic+scRNAtools-package'></span>

<h3>Description</h3>

<p>We integrated the common analysis methods utilized in single cell RNA sequencing data, which included cluster method, PCA, the filter of differentially expressed genes, pathway enrichment analysis and correlated analysis methods.
</p>


<h3>Author(s)</h3>

<p>Qian Yang
Maintainer: Qian Yang &lt;bioqianyang@163.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>####Here list three main function, cluster, PCA and t-SNE####
####cluster####
data(example1);##Example data in this package.
k&lt;-6;##set K based on your own requirement.
scRNAtools_cluster(example1,k)
####PCA####
data(example1)
data(types)
pdf(file=file.path(tempdir(), "PCA_result-R.pdf"))##Save the figures of PCA results.
scRNAtools_pca(example1,types)
dev.off()
####t-SNE#####
data(exam)
scRNAtools_tsne(exam)

####Gene expression###
data(example)
types&lt;-"1"
num&lt;-0.8
scRNAtools_Geneexp(example,types,num)
</code></pre>

<hr>
<h2 id='corr_re'>correlation index</h2><span id='topic+corr_re'></span>

<h3>Description</h3>

<p>The correlation index of the genes in the section of correlated analysis.</p>

<hr>
<h2 id='DEGs'>Users interested genes or differentially expressed genes</h2><span id='topic+DEGs'></span>

<h3>Description</h3>

<p>Gene list with two columns. The first column is Entrez ID of genes and the second column is gene symbol</p>

<hr>
<h2 id='exam'>exam</h2><span id='topic+exam'></span>

<h3>Description</h3>

<p>Example data in t-SNE method</p>

<hr>
<h2 id='exam1'>exam1</h2><span id='topic+exam1'></span>

<h3>Description</h3>

<p>Example data in correlated analysis</p>

<hr>
<h2 id='example'>example</h2><span id='topic+example'></span>

<h3>Description</h3>

<p>scRNA sequencing data in 50 cells and 1000 genes</p>

<hr>
<h2 id='example1'>example1</h2><span id='topic+example1'></span>

<h3>Description</h3>

<p>scRNA sequencing data</p>

<hr>
<h2 id='scRNAtools_cluster'>Cluster section</h2><span id='topic+scRNAtools_cluster'></span>

<h3>Description</h3>

<p>Do consistent clustering analysis use clusterProfiler method</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_cluster(example1, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_cluster_+3A_example1">example1</code></td>
<td>

<p>scRNA sequencing data with header.
</p>
</td></tr>
<tr><td><code id="scRNAtools_cluster_+3A_k">k</code></td>
<td>

<p>The number of class. If you set k is 6, you will obtain 6 results of cluster.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The results are presented in your occurrent path.
</p>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>References</h3>

<p>Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology 2012, 16(5):284-287.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##setwd("")###Set the path your data in.
data(example1)##Example data in this package.
k&lt;-6##set K based on your own requirement.
scRNAtools_cluster(example1,k)
</code></pre>

<hr>
<h2 id='scRNAtools_cor_map'>Correlation analysis</h2><span id='topic+scRNAtools_cor_map'></span>

<h3>Description</h3>

<p>Correlation analysis of interested gene set or differentially expressed gene set.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_cor_map(exam1, types_all, type, methods)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_cor_map_+3A_exam1">exam1</code></td>
<td>

<p>scRNA sequencing data of several genes and cells.
</p>
</td></tr>
<tr><td><code id="scRNAtools_cor_map_+3A_types_all">types_all</code></td>
<td>

<p>Cell names of each type.
</p>
</td></tr>
<tr><td><code id="scRNAtools_cor_map_+3A_type">type</code></td>
<td>

<p>Cell type.
</p>
</td></tr>
<tr><td><code id="scRNAtools_cor_map_+3A_methods">methods</code></td>
<td>

<p>correlation methods including &quot;pearson&quot;, &quot;kendall&quot; and &quot;spearman&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Return the correlation index of each two genes.
</p>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exam1)
data(types_all)
type&lt;-"Malignant";
methods&lt;-"pearson";##methods = c("pearson", "kendall", "spearman").
pdf(file=file.path(tempdir(), "correlation_color.pdf"))
scRNAtools_cor_map(exam1,types_all,type,methods)
dev.off()
</code></pre>

<hr>
<h2 id='scRNAtools_cor_map_r'>Present correlation index in figure</h2><span id='topic+scRNAtools_cor_map_r'></span>

<h3>Description</h3>

<p>Correlation analysis with correlation index of interested gene set or differentially expressed gene set.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_cor_map_r(exam1, types_all, type, methods)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_cor_map_r_+3A_exam1">exam1</code></td>
<td>

<p>scRNA sequencing data of several genes and cells.
</p>
</td></tr>
<tr><td><code id="scRNAtools_cor_map_r_+3A_types_all">types_all</code></td>
<td>

<p>Cell names of each type.
</p>
</td></tr>
<tr><td><code id="scRNAtools_cor_map_r_+3A_type">type</code></td>
<td>

<p>Cell type.
</p>
</td></tr>
<tr><td><code id="scRNAtools_cor_map_r_+3A_methods">methods</code></td>
<td>

<p>correlation methods including &quot;pearson&quot;, &quot;kendall&quot; and &quot;spearman&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Return the correlation index of each two genes.
</p>


<h3>Author(s)</h3>

<p>Qian Yang</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exam1)
data(types_all)
type&lt;-"Malignant";
methods&lt;-"pearson";##methods = c("pearson", "kendall", "spearman").
pdf(file=file.path(tempdir(), "correlation_num.pdf"))
scRNAtools_cor_map_r(exam1,types_all,type,methods)
dev.off()
</code></pre>

<hr>
<h2 id='scRNAtools_DEGsA'>Identification of differentially expressed genes</h2><span id='topic+scRNAtools_DEGsA'></span>

<h3>Description</h3>

<p>Users can identify differentially expressed genes between two type of cells based on fold change value.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_DEGsA(example, types_all, type1, type2, num)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_DEGsA_+3A_example">example</code></td>
<td>

<p>scRNA sequencing data with header.
</p>
</td></tr>
<tr><td><code id="scRNAtools_DEGsA_+3A_types_all">types_all</code></td>
<td>

<p>Cell types in the example data.
</p>
</td></tr>
<tr><td><code id="scRNAtools_DEGsA_+3A_type1">type1</code></td>
<td>

<p>Cell type one.
</p>
</td></tr>
<tr><td><code id="scRNAtools_DEGsA_+3A_type2">type2</code></td>
<td>

<p>Cell type two.
</p>
</td></tr>
<tr><td><code id="scRNAtools_DEGsA_+3A_num">num</code></td>
<td>

<p>Threshold value of expressed genes in appointed cell types. For example, we set 0.8 in example section.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output data is the fold change value of differentially expressed genes.
</p>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example)
data(types)
type1&lt;-"No malignant"
type2&lt;-"Malignant"
num&lt;-0.8;###type1 Vs type2
pdf(file=file.path(tempdir(), "DEGs.pdf"))
scRNAtools_DEGsA(example,types_all,type1,type2,num)
dev.off()
</code></pre>

<hr>
<h2 id='scRNAtools_Gene2exp_1'>Present the expression of two genes</h2><span id='topic+scRNAtools_Gene2exp_1'></span>

<h3>Description</h3>

<p>This function can present the expression of two gene in appointed cell type.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_Gene2exp_1(example, types_all, gene1, gene2, n, col_1, col_2, pch, lwd)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_example">example</code></td>
<td>

<p>scRNA sequencing data without header.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_types_all">types_all</code></td>
<td>

<p>Cell names of each type.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_gene1">gene1</code></td>
<td>

<p>Gene one you are interested in.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_gene2">gene2</code></td>
<td>

<p>Gene two you are interested in.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_n">n</code></td>
<td>

<p>Number of cell names in scRNA sequening data.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_col_1">col_1</code></td>
<td>

<p>The color of line of gene one in the figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_col_2">col_2</code></td>
<td>

<p>The color of line of gene two in the figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_pch">pch</code></td>
<td>

<p>The shape of nodes in figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene2exp_1_+3A_lwd">lwd</code></td>
<td>

<p>The width of lines in figure.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example)
data(types_all)
gene1&lt;-"CHD1"
gene2&lt;-"CD82"
col_1="red"
col_2="blue"
pch=19
lwd=1
n&lt;-2
scRNAtools_Gene2exp_1(example,types_all,gene1,gene2,n,col_1,col_2,pch,lwd)
</code></pre>

<hr>
<h2 id='scRNAtools_Gene3exp_1'>Present gene expression</h2><span id='topic+scRNAtools_Gene3exp_1'></span>

<h3>Description</h3>

<p>This function can present the expression of two gene in appointed cell type.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_Gene3exp_1(example,types_all,gene1,gene2,gene3,n,col_1,col_2,col_3,pch,lwd)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_example">example</code></td>
<td>

<p>scRNA sequencing data without header.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_types_all">types_all</code></td>
<td>

<p>Cell names of each type.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_gene1">gene1</code></td>
<td>

<p>Gene one you are interested in.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_gene2">gene2</code></td>
<td>

<p>Gene two you are interested in.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_gene3">gene3</code></td>
<td>

<p>Gene three you are interested in.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_n">n</code></td>
<td>

<p>Number of cell names in scRNA sequening data.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_col_1">col_1</code></td>
<td>

<p>The color of line of gene one in the figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_col_2">col_2</code></td>
<td>

<p>The color of line of gene two in the figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_col_3">col_3</code></td>
<td>

<p>The color of line of gene three in the figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_pch">pch</code></td>
<td>

<p>The shape of nodes in figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Gene3exp_1_+3A_lwd">lwd</code></td>
<td>

<p>The width of lines in figure.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example)
data(types_all)
gene1&lt;-"CHD1"
gene2&lt;-"CD82"
gene3&lt;-"ASS1"
col_1="red"
col_2="blue"
col_3="green"
pch=19
lwd=2
n&lt;-3
scRNAtools_Gene3exp_1(example,types_all,gene1,gene2,gene3,n,col_1,col_2,col_3,pch,lwd)
</code></pre>

<hr>
<h2 id='scRNAtools_Geneexp'>Expressed genes in scRNA sequencing data</h2><span id='topic+scRNAtools_Geneexp'></span>

<h3>Description</h3>

<p>Extracted the genes expressed in cells. Users can set the threshold value.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_Geneexp(example, types, num)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_Geneexp_+3A_example">example</code></td>
<td>

<p>scRNA sequencing data without header.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Geneexp_+3A_types">types</code></td>
<td>

<p>Cell types in the example data.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Geneexp_+3A_num">num</code></td>
<td>

<p>Threshold value of expressed genes in appointed cell types. For example, we set 0.8 in example section.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>zset</code></td>
<td>
<p>Gene expression data required the threshold value.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example)
types&lt;-"1"
num&lt;-0.8
scRNAtools_Geneexp(example,types,num)
</code></pre>

<hr>
<h2 id='scRNAtools_Geneexp_1'>Present gene expression</h2><span id='topic+scRNAtools_Geneexp_1'></span>

<h3>Description</h3>

<p>This function can present the expression of one gene in appointed cell type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_Geneexp_1(example, gene, types_all, n, col, pch, lwd)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_Geneexp_1_+3A_example">example</code></td>
<td>

<p>scRNA sequencing data without header.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Geneexp_1_+3A_gene">gene</code></td>
<td>

<p>One gene you are interested in.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Geneexp_1_+3A_types_all">types_all</code></td>
<td>

<p>Cell names of each type.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Geneexp_1_+3A_n">n</code></td>
<td>

<p>Number of cell names in scRNA sequening data.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Geneexp_1_+3A_col">col</code></td>
<td>

<p>The color of line in the figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Geneexp_1_+3A_pch">pch</code></td>
<td>

<p>The shape of nodes in figure.
</p>
</td></tr>
<tr><td><code id="scRNAtools_Geneexp_1_+3A_lwd">lwd</code></td>
<td>

<p>The width of lines in figure.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example)
data(types_all)
gene&lt;-"CHD1";###Set the gene you are interested in.
n&lt;-3;###Set the type of cells you are interested in.
col&lt;-"red";###Set the color of line in the figure.
pch&lt;-19;###Set the shape of nodes in figure.
lwd&lt;-2;###Set the width of lines in figure.
scRNAtools_Geneexp_1(example,gene,types_all,n,col,pch,lwd)
</code></pre>

<hr>
<h2 id='scRNAtools_inter_net'>Construction of interactive network in scRNA sequencing data</h2><span id='topic+scRNAtools_inter_net'></span>

<h3>Description</h3>

<p>Construction of interactive network based on scRNa sequencing data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_inter_net(corr_re, p, r, size, color)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_inter_net_+3A_corr_re">corr_re</code></td>
<td>

<p>The results of correlation analysis, which including four columns, the first two columns are genes and the last two columns are correlation index and p-value,respectively.
</p>
</td></tr>
<tr><td><code id="scRNAtools_inter_net_+3A_p">p</code></td>
<td>

<p>The p-value of correlation index.
</p>
</td></tr>
<tr><td><code id="scRNAtools_inter_net_+3A_r">r</code></td>
<td>

<p>Correlation index
</p>
</td></tr>
<tr><td><code id="scRNAtools_inter_net_+3A_size">size</code></td>
<td>

<p>The size of nodes in the network.
</p>
</td></tr>
<tr><td><code id="scRNAtools_inter_net_+3A_color">color</code></td>
<td>

<p>The color of nodes in the network.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(corr_re)
p&lt;-0.05
r&lt;-0.9
size&lt;-5 #nodes size
color&lt;-"#00B2EE" ##Color of nodes.
pdf(file=file.path(tempdir(), "interact_net.pdf"))
scRNAtools_inter_net(corr_re,p,r,size,color)
dev.off()
</code></pre>

<hr>
<h2 id='scRNAtools_pca'>PCA analysis</h2><span id='topic+scRNAtools_pca'></span>

<h3>Description</h3>

<p>PCA analysis for scRNA sequencing data</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_pca(example1, types)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_pca_+3A_example1">example1</code></td>
<td>

<p>scRNA sequencing data with header.
</p>
</td></tr>
<tr><td><code id="scRNAtools_pca_+3A_types">types</code></td>
<td>

<p>Cell types in the example data.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example1)
data(types)
pdf(file=file.path(tempdir(), "PCA_result-R.pdf"))##Save the figures of PCA results.
scRNAtools_pca(example1,types)
dev.off()
</code></pre>

<hr>
<h2 id='scRNAtools_pca_3D'>3D PCA analysis</h2><span id='topic+scRNAtools_pca_3D'></span>

<h3>Description</h3>

<p>PCA analysis for scRNA sequencing data and present 3D figure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_pca_3D(example1, types)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_pca_3D_+3A_example1">example1</code></td>
<td>

<p>scRNA sequencing data with header.
</p>
</td></tr>
<tr><td><code id="scRNAtools_pca_3D_+3A_types">types</code></td>
<td>

<p>Cell types in the example data.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##3D PCA analysis
data(example1)
data(types)
scRNAtools_pca_3D(example1,types)##3D figure of PCA results.
</code></pre>

<hr>
<h2 id='scRNAtools_PEA'>Pathway enrichment analysis</h2><span id='topic+scRNAtools_PEA'></span>

<h3>Description</h3>

<p>Pathway enrichment analysis using the interested gene set or differentially expressed gene set provided by users. This data contains two column (Enterz ID and gene sybmols)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_PEA(DEGs,number)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_PEA_+3A_degs">DEGs</code></td>
<td>

<p>Interested gene set of differentially expressed gene set.
</p>
</td></tr>
<tr><td><code id="scRNAtools_PEA_+3A_number">number</code></td>
<td>

<p>The number of random, for example, users can set 1000, 5000 or more.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function integrated method to do the pathway enrichment analysis, TPEA.</p>


<h3>Value</h3>

<p>The significant pathways are wrote in the occurrent path.</p>


<h3>Author(s)</h3>

<p>Qian Yang</p>


<h3>References</h3>

<p>Wei Jiang (2017). TPEA: A Novel Topology-Based Pathway Enrichment Analysis Approach.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DEGs)
number&lt;-10
pdf(file=file.path(tempdir(), "enrichment analysis.pdf"))
scRNAtools_PEA(DEGs,number)
dev.off()
</code></pre>

<hr>
<h2 id='scRNAtools_tsne'>t-SNE analysis</h2><span id='topic+scRNAtools_tsne'></span>

<h3>Description</h3>

<p>t-SNE analysis for scRNA sequencing data</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNAtools_tsne(exam)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scRNAtools_tsne_+3A_exam">exam</code></td>
<td>

<p>scRNA sequencing data with four genes. Users can reference the format and input their own data.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>References</h3>

<p>L.J.P. van der Maaten and G.E. Hinton. Visualizing High-Dimensional Data Using t-SNE. Journal of Machine Learning Research 9(Nov):2579-2605, 2008.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exam)
scRNAtools_tsne(exam)
</code></pre>

<hr>
<h2 id='types'>types</h2><span id='topic+types'></span>

<h3>Description</h3>

<p>Cell types in the example data</p>

<hr>
<h2 id='types_all'>types_all</h2><span id='topic+types_all'></span>

<h3>Description</h3>

<p>Cell names of each type</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
