<!DOCTYPE html><html><head><title>Help for package ScRNAIMM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ScRNAIMM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cluster_cells'><p>Perform cell clustering based on scDHA method</p></a></li>
<li><a href='#evaluate_clustering'><p>Evaluate the clustering if you have the original labels</p></a></li>
<li><a href='#filter_ScRNA'><p>Remove genes which are not expressed in at least one cell</p></a></li>
<li><a href='#prepare_dataset'><p>Prepare the data set for the imputation</p></a></li>
<li><a href='#run_pipeline'><p>Run the main pipeline for ScRNAIMM</p></a></li>
<li><a href='#ScRNA_imp_MM'><p>Perform ScRNA-seq imputation using mean/Median</p></a></li>
<li><a href='#scRNA_MMI'><p>Performs a distribution check for the data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Performing Single-Cell RNA-Seq Imputation by Using Mean/Median
Imputation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mohamed Soudy &lt;Mohmedsoudy2009@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performing single-cell imputation in a way that preserves the biological variations in the data. The package clusters the input data to do imputation for each cluster, and do a distribution check using the Anderson-Darling normality test to impute dropouts using mean or median (Yazici, B., &amp; Yolacan, S. (2007) &lt;<a href="https://doi.org/10.1080%2F10629360600678310">doi:10.1080/10629360600678310</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr, dplyr, nortest, mclust, scDHA</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-17 14:25:50 UTC; Soudy</td>
</tr>
<tr>
<td>Author:</td>
<td>Mohamed Soudy [aut, cre],
  Sascha Jung [aut],
  Antonio DEL SOL [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-17 20:10:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='cluster_cells'>Perform cell clustering based on scDHA method</h2><span id='topic+cluster_cells'></span>

<h3>Description</h3>

<p>Cluster cells based on scDHA methods to get cluster labels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_cells(ScRNA_filtered, Normalize = TRUE, k=NULL, n=5000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster_cells_+3A_scrna_filtered">ScRNA_filtered</code></td>
<td>
<p>ScRNA-seq data set generated by filter_ScRNA function</p>
</td></tr>
<tr><td><code id="cluster_cells_+3A_normalize">Normalize</code></td>
<td>
<p>Boolean parameter whether to apply log10 normalization for the data or not</p>
</td></tr>
<tr><td><code id="cluster_cells_+3A_k">k</code></td>
<td>
<p>Number of clusters if there is a prior knowledge about that</p>
</td></tr>
<tr><td><code id="cluster_cells_+3A_n">n</code></td>
<td>
<p>Number of genes to keep after feature selection step</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector that contains the cell labels
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='evaluate_clustering'>Evaluate the clustering if you have the original labels</h2><span id='topic+evaluate_clustering'></span>

<h3>Description</h3>

<p>Evaluate the clustering if you have the original labels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluate_clustering(cluster_labels, original_labels)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evaluate_clustering_+3A_cluster_labels">cluster_labels</code></td>
<td>
<p>Cluster labels generated by cluster_cells functions or user-defined</p>
</td></tr>
<tr><td><code id="evaluate_clustering_+3A_original_labels">original_labels</code></td>
<td>
<p>Original labels of the ScRNA-seq data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ARI of clustering 'a value between 0 and 1' 1 indicates best clustering
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>evaluate_clustering(c(1,1,1,1,2,2,3,3), c(1,1,1,1,3,3,3,2))

</code></pre>

<hr>
<h2 id='filter_ScRNA'>Remove genes which are not expressed in at least one cell</h2><span id='topic+filter_ScRNA'></span>

<h3>Description</h3>

<p>Filter out the genes that are not expressed in at least one cell type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_ScRNA(ScRNA_mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_ScRNA_+3A_scrna_mat">ScRNA_mat</code></td>
<td>
<p>ScRNA-Seq matrix where genes are in rows and cells are in columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Filtered ScRNA-seq matrix that contains genes that are at least expressed in one cell type
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='prepare_dataset'>Prepare the data set for the imputation</h2><span id='topic+prepare_dataset'></span>

<h3>Description</h3>

<p>This function aims to get the indices where all genes are zeros or genes that is expressed on only one sample per cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_dataset(filtered_data, cluster_labels)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_dataset_+3A_filtered_data">filtered_data</code></td>
<td>
<p>ScRNA-seq data set generate by filter_ScRNA function</p>
</td></tr>
<tr><td><code id="prepare_dataset_+3A_cluster_labels">cluster_labels</code></td>
<td>
<p>cell labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame that contains the processed ScRNA-seq data
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='run_pipeline'>Run the main pipeline for ScRNAIMM</h2><span id='topic+run_pipeline'></span>

<h3>Description</h3>

<p>Run the main pipeline for ScRNAIMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_pipeline(ScRNA,label=NULL,k=NULL,cells=TRUE,genes=TRUE,outdir=NULL,dataset=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_pipeline_+3A_scrna">ScRNA</code></td>
<td>
<p>ScRNA-seq data set generated by prepare_dataset function</p>
</td></tr>
<tr><td><code id="run_pipeline_+3A_label">label</code></td>
<td>
<p>Prior knowledge about cluster labels if NULL, will use our clustering function</p>
</td></tr>
<tr><td><code id="run_pipeline_+3A_k">k</code></td>
<td>
<p>Prior knowledge about number of clusters if NULL, will use our clustering function</p>
</td></tr>
<tr><td><code id="run_pipeline_+3A_cells">cells</code></td>
<td>
<p>Boolean whether to do the imputation based on cell clustering or not</p>
</td></tr>
<tr><td><code id="run_pipeline_+3A_genes">genes</code></td>
<td>
<p>Boolean whether to do the imputation based on genes or not</p>
</td></tr>
<tr><td><code id="run_pipeline_+3A_outdir">outdir</code></td>
<td>
<p>Path to output directory to write the imputed data</p>
</td></tr>
<tr><td><code id="run_pipeline_+3A_dataset">dataset</code></td>
<td>
<p>Name of the data set to be the name of the output directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with the imputed values
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='ScRNA_imp_MM'>Perform ScRNA-seq imputation using mean/Median</h2><span id='topic+ScRNA_imp_MM'></span>

<h3>Description</h3>

<p>Perform ScRNA-seq imputation using mean/Median
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScRNA_imp_MM(ScRNA_filtered, cluster_labels = NULL, cells = TRUE, genes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ScRNA_imp_MM_+3A_scrna_filtered">ScRNA_filtered</code></td>
<td>
<p>ScRNA-seq data set generated by prepare_dataset function</p>
</td></tr>
<tr><td><code id="ScRNA_imp_MM_+3A_cluster_labels">cluster_labels</code></td>
<td>
<p>Cluster labels generated by cluster_cells function or user-defined</p>
</td></tr>
<tr><td><code id="ScRNA_imp_MM_+3A_cells">cells</code></td>
<td>
<p>Boolean whether to do the imputation based on cell clustering or not</p>
</td></tr>
<tr><td><code id="ScRNA_imp_MM_+3A_genes">genes</code></td>
<td>
<p>Boolean whether to do the imputation based on genes or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with the imputed values
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='scRNA_MMI'>Performs a distribution check for the data</h2><span id='topic+scRNA_MMI'></span>

<h3>Description</h3>

<p>Performs a distribution check for the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scRNA_MMI(scRNA_dataset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scRNA_MMI_+3A_scrna_dataset">scRNA_dataset</code></td>
<td>
<p>ScRNA-seq data set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ScRNA-Seq Imputed data set
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
