<!DOCTYPE html><html lang="en"><head><title>Help for package scTenifoldKnk</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scTenifoldKnk}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#scTenifoldKnk'><p>scTenifoldKNK</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>In-Silico Knockout Experiments from Single-Cell Gene Regulatory
Networks</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A workflow based on 'scTenifoldNet' to perform in-silico knockout experiments using single-cell RNA sequencing (scRNA-seq) data from wild-type (WT) control samples as input.  First, the package constructs a single-cell gene regulatory network (scGRN) and knocks out a target gene from the adjacency matrix of the WT scGRN by setting the gene’s outdegree edges to zero. Then, it compares the knocked out scGRN with the WT scGRN to identify differentially regulated genes, called virtual-knockout perturbed genes, which are used to assess the impact of the gene knockout and reveal the gene’s function in the analyzed cells.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/cailab-tamu/scTenifoldKnk">https://github.com/cailab-tamu/scTenifoldKnk</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cailab-tamu/scTenifoldKnk/issues">https://github.com/cailab-tamu/scTenifoldKnk/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>pbapply, RSpectra, Matrix, methods, stats, utils, MASS,
scTenifoldNet</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.1.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-22 19:28:57 UTC; dcosorioh</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Osorio <a href="https://orcid.org/0000-0003-4424-8422"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Yan Zhong [aut, ctb],
  Guanxun Li [aut, ctb],
  Qian Xu [aut, ctb],
  Andrew Hillhouse [aut, ctb],
  Jingshu Chen [aut, ctb],
  Laurie Davidson [aut, ctb],
  Yanan Tian [aut, ctb],
  Robert Chapkin [aut, ctb],
  Jianhua Huang [aut, ctb],
  James Cai <a href="https://orcid.org/0000-0002-8081-6725"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    ctb, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Osorio &lt;dcosorioh@tamu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-22 20:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='scTenifoldKnk'>scTenifoldKNK</h2><span id='topic+scTenifoldKnk'></span>

<h3>Description</h3>

<p>Predict gene perturbations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scTenifoldKnk(
  countMatrix,
  gKO = NULL,
  qc_mtThreshold = 0.1,
  qc_minLSize = 1000,
  nc_lambda = 0,
  nc_nNet = 10,
  nc_nCells = 500,
  nc_nComp = 3,
  nc_scaleScores = TRUE,
  nc_symmetric = FALSE,
  nc_q = 0.9,
  td_K = 3,
  td_maxIter = 1000,
  td_maxError = 1e-05,
  td_nDecimal = 3,
  ma_nDim = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scTenifoldKnk_+3A_countmatrix">countMatrix</code></td>
<td>
<p>countMatrix</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_gko">gKO</code></td>
<td>
<p>gKO</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_qc_mtthreshold">qc_mtThreshold</code></td>
<td>
<p>A decimal value between 0 and 1. Defines the maximum ratio of mitochondrial reads (mithocondrial reads / library size) present in a cell to be included in the analysis. It's computed using the symbol genes starting with 'MT-' non-case sensitive.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_qc_minlsize">qc_minLSize</code></td>
<td>
<p>An integer value. Defines the minimum library size required for a cell to be included in the analysis.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_nc_lambda">nc_lambda</code></td>
<td>
<p>A continuous value between 0 and 1. Defines the multiplicative value (1-lambda) to be applied over the weaker edge connecting two genes to maximize the adjacency matrix directionality.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_nc_nnet">nc_nNet</code></td>
<td>
<p>An integer value. The number of networks based on principal components regression to generate.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_nc_ncells">nc_nCells</code></td>
<td>
<p>An integer value. The number of cells to subsample each time to generate a network.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_nc_ncomp">nc_nComp</code></td>
<td>
<p>An integer value. The number of principal components in PCA to generate the networks. Should be greater than 2 and lower than the total number of genes.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_nc_scalescores">nc_scaleScores</code></td>
<td>
<p>A boolean value (TRUE/FALSE), if TRUE, the weights will be normalized such that the maximum absolute value is 1.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_nc_symmetric">nc_symmetric</code></td>
<td>
<p>A boolean value (TRUE/FALSE), if TRUE, the weights matrix returned will be symmetric.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_nc_q">nc_q</code></td>
<td>
<p>A decimal value between 0 and 1. Defines the cut-off threshold of top q% relationships to be returned.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_td_k">td_K</code></td>
<td>
<p>An integer value. Defines the number of rank-one tensors used to approximate the data using CANDECOMP/PARAFAC (CP) Tensor Decomposition.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_td_maxiter">td_maxIter</code></td>
<td>
<p>An integer value. Defines the maximum number of iterations if error stay above <code>td_maxError</code>.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_td_maxerror">td_maxError</code></td>
<td>
<p>A decimal value between 0 and 1. Defines the relative Frobenius norm error tolerance.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_td_ndecimal">td_nDecimal</code></td>
<td>
<p>An integer value indicating the number of decimal places to be used.</p>
</td></tr>
<tr><td><code id="scTenifoldKnk_+3A_ma_ndim">ma_nDim</code></td>
<td>
<p>An integer value. Defines the number of dimensions of the low-dimensional feature space to be returned from the non-linear manifold alignment.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniel Osorio &lt;dcosorioh@tamu.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Loading single-cell data
scRNAseq &lt;- system.file("single-cell/example.csv",package="scTenifoldKnk")
scRNAseq &lt;- read.csv(scRNAseq, row.names = 1)

# Running scTenifoldKnk
scTenifoldKnk(countMatrix = scRNAseq, gKO = 'G100', qc_minLSize = 0)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
