<!DOCTYPE html><html><head><title>Help for package scUtils</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {scUtils}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#closed_breaks_log2'><p>Closed breaks for log scale</p></a></li>
<li><a href='#closed_labels'><p>Human-readable labels for closed breaks</p></a></li>
<li><a href='#colVars_spm'><p>Variance computation for sparse matrices</p></a></li>
<li><a href='#feat'><p>Feature Plot</p></a></li>
<li><a href='#is_wholenumber'><p>Check if number(s) is/are integers. In contrast to is.integer, is_wholenumber</p>
does not check the class but accepts all numbers that are integers with reasonable
precision.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Utility Functions for Single-Cell RNA Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of single-cell RNA sequencing data can be simple and
    clear with the right utility functions. This package collects such functions,
    aiming to fulfill the following criteria:
    code clarity over performance (i.e. plain R code instead of C code),
    most important analysis steps over completeness
    (analysis 'by hand', not automated integration etc.),
    emphasis on quantitative visualization (intensity-coded color scale, etc.).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, Matrix, scales, assertthat, dplyr, viridis,
viridisLite, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, tibble</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-06-23 10:03:01 UTC; felix</td>
</tr>
<tr>
<td>Author:</td>
<td>Felix Frauhammer [aut, cre],
  Simon Anders [ctb] (Simon Anders wrote the colVars_spm function.)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Felix Frauhammer &lt;felixwertek@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-06-25 16:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='closed_breaks_log2'>Closed breaks for log scale</h2><span id='topic+closed_breaks_log2'></span>

<h3>Description</h3>

<p>Finds breaks that are powers of 2,
and forces inclusion of upper and lower limits
(displaying the closed interval).
Including limits specifically is particularly useful for ggplot2's color/fill,
as it
emphasizes the meaning of maximal/minimal color intensities (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>closed_breaks_log2(lims)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="closed_breaks_log2_+3A_lims">lims</code></td>
<td>
<p>Vector with lower and upper limits (in that order) of the data
that you want breaks for.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>feat</code> function uses <code>closed_breaks_log2</code> to color by
gene expression,
where the maximal expression gives valuable
intuition for a gene's overall expression strength.
For x- or y-axis (<code>scale_*_log10</code>),
I still recommend <code>breaks_log</code> from the scales package.
</p>


<h3>Value</h3>

<p>Numeric vector with breaks.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+closed_labels">closed_labels</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># closed breaks include maximum, breaks_log do not:
closed_breaks_log2(lims = c(.01, 977.1))
scales::breaks_log()(c(.01, 977.1))
</code></pre>

<hr>
<h2 id='closed_labels'>Human-readable labels for closed breaks</h2><span id='topic+closed_labels'></span>

<h3>Description</h3>

<p>Complements the closed_breaks_log2 function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>closed_labels(x, min_is_zero = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="closed_labels_+3A_x">x</code></td>
<td>
<p>Vector of breaks for which to produce labels.
Typically, this is the output of <code>closed_breaks_log2</code>.</p>
</td></tr>
<tr><td><code id="closed_labels_+3A_min_is_zero">min_is_zero</code></td>
<td>
<p>Should the smallest break be
displayed as zero (TRUE) or as the actual value (FALSE). Default: FALSE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a helper for the <code>feat</code> function.
<code>feat</code> replaces numeric zeros with the next-smallest expression value
to avoid taking the logarithm of zero. <code>min_is_zero</code> can be used to
display the lowest break of the color scale as zero in these cases.
</p>


<h3>Value</h3>

<p>Character vector with labels, used by <code>feat</code> function.
</p>


<h3>See Also</h3>

<p><code><a href="scales.html#topic+label_scientific">label_scientific</a></code>
<code><a href="scales.html#topic+label_number_auto">label_number_auto</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # human readable output:
 closed_labels(c(.001111,.122, 0.5, 10, 100, 1800))
</code></pre>

<hr>
<h2 id='colVars_spm'>Variance computation for sparse matrices</h2><span id='topic+colVars_spm'></span><span id='topic+rowVars_spm'></span>

<h3>Description</h3>

<p>Compute variance for each column / each row of
a dgCMatrix (from Matrix package).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colVars_spm(spm)

rowVars_spm(spm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colVars_spm_+3A_spm">spm</code></td>
<td>
<p>A sparse matrix of class dgCMatrix from the Matrix package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The only supported format currently is dgCMatrix. While the Matrix
package has other formats, this one is used for scRNAseq raw count data.
Function code written by Simon Anders.
</p>


<h3>Value</h3>

<p>Vector with variances.
</p>


<h3>See Also</h3>

<p><code>vignette("Intro2Matrix", package="Matrix")</code>
<a href="Matrix.html#topic+CsparseMatrix-class">CsparseMatrix-class</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Matrix)
 mat &lt;- as(matrix(rpois(900,1), ncol=3), "dgCMatrix")
 colVars_spm(mat)
</code></pre>

<hr>
<h2 id='feat'>Feature Plot</h2><span id='topic+feat'></span>

<h3>Description</h3>

<p>Highlight gene expression data in a 2D-embedding (UMAP, tSNE, etc.).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>feat(embedding, expression, legend_name = "Expression")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="feat_+3A_embedding">embedding</code></td>
<td>
<p>A matrix/data.frame/tibble/... with exactly two columns.
If colnames are missing, the axis will be named &quot;Dim1&quot; and &quot;Dim2&quot;.
Other classes than matrix/data.frame/tibble are possible, as long as
<code>data.frame(embedding)</code>) produces a numeric data.frame.</p>
</td></tr>
<tr><td><code id="feat_+3A_expression">expression</code></td>
<td>
<p>Numeric vector with expression values of the gene of
interest. Order has to correspond to the row order in <code>embedding</code>.
Typically, <code>expression</code> is normalized gene expression and we recommend
<code>k/s/mean(1/s)</code>, where <code>k</code> are UMI counts for the gene of interest
and <code>s</code> are totalUMI of the cell (aka library size).</p>
</td></tr>
<tr><td><code id="feat_+3A_legend_name">legend_name</code></td>
<td>
<p>Text displayed above the legend. Most commonly the name
of the displayed gene.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function discourages customization on purpose, because it
bundles geoms, themes and settings that I found important for
visualizing gene expression in scRNAseq data:
</p>

<ul>
<li><p> coord_fixed, to avoid distortion of embeddings
</p>
</li>
<li><p> geom_point with size=.4, to ameliorate overplotting
</p>
</li>
<li><p> No background grid, because distances and axis units
in embeddings do not
carry meaning for most dimensionality reduction techniques.
</p>
</li>
<li><p> Intensity-coded color scales (viridis) displayed with
log2-transformation. Makes visualization independent of colorblindness
and appropriate for gene expression data (which is usually Log Normal
distributed).
</p>
</li>
<li><p> Color scale breaks are displayed as 'closed interval', i.e.
<code>max(expression)</code> and <code>min(expression)</code> are the most extreme
breaks. Rounding makes them human-readable. This functionality is provided
by <a href="#topic+closed_breaks_log2">closed_breaks_log2</a> and <a href="#topic+closed_labels">closed_labels</a>.
</p>
</li></ul>

<p>If you insist on customizing, think of this function as a great starting point, you can simply
copy-paste the code after typing <code>feat</code> into your
console.
</p>


<h3>Value</h3>

<p>A <code>ggplot2</code> object storing a colored scatter plot.
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>,
<code><a href="#topic+closed_labels">closed_labels</a></code>,
<code><a href="#topic+closed_breaks_log2">closed_breaks_log2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # expression goes from 0 to 22:
 set.seed(100)
 feat(matrix(rnorm(2000, c(.1, 3)), ncol=2), rpois(1000, c(.1, 11)))
 # expression goes from 2 to 52:
 set.seed(100)
 feat(matrix(rnorm(2000, c(.1, 3)), ncol=2), rpois(1000, c(10, 31)))
</code></pre>

<hr>
<h2 id='is_wholenumber'>Check if number(s) is/are integers. In contrast to is.integer, is_wholenumber
does not check the class but accepts all numbers that are integers with reasonable
precision.</h2><span id='topic+is_wholenumber'></span>

<h3>Description</h3>

<p>Check if number(s) is/are integers. In contrast to is.integer, is_wholenumber
does not check the class but accepts all numbers that are integers with reasonable
precision.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_wholenumber(x, tol = .Machine$double.eps^0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_wholenumber_+3A_x">x</code></td>
<td>
<p>Number to test</p>
</td></tr>
<tr><td><code id="is_wholenumber_+3A_tol">tol</code></td>
<td>
<p>tolerance for testing</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
