<!DOCTYPE html><html><head><title>Help for package SEAGLE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SEAGLE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#applyAt'><p>Function for applying t(A) on the left for REML EM</p></a></li>
<li><a href='#cosihap'><p>Synthetic haplotype data generated from COSI software</p></a></li>
<li><a href='#estimate.vc'><p>REML EM Algorithm</p></a></li>
<li><a href='#makeSimData'><p>Generate synthetic data according to a fixed effects model</p></a></li>
<li><a href='#prep.SEAGLE'><p>Prepare data for input into SEAGLE function</p></a></li>
<li><a href='#Rinv.AtG'><p>Function for applying R inverse to AtG in REML EM algorithm</p></a></li>
<li><a href='#Rinv.u'><p>Function for applying R inverse to u in REML EM algorithm</p></a></li>
<li><a href='#SEAGLE'><p>Compute score-like test statistic and p-value for GxE test with SEAGLE algorithm</p></a></li>
<li><a href='#Vinv'><p>Function for applying V inverse in Algorithm 1</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Scalable Exact Algorithm for Large-Scale Set-Based
Gene-Environment Interaction Tests</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>
    The explosion of biobank data offers immediate opportunities for 
    gene-environment (GxE) interaction studies of complex diseases because of the 
    large sample sizes and rich collection in genetic and non-genetic information. 
    However, the extremely large sample size also introduces new computational 
    challenges in GxE assessment, especially for set-based GxE variance component (VC) 
    tests, a widely used strategy to boost overall GxE signals and to evaluate the 
    joint GxE effect of multiple variants from a biologically meaningful unit 
    (e.g., gene). 
    We present 'SEAGLE', a Scalable Exact AlGorithm for Large-scale Set-based 
    GxE tests, to permit GxE VC test scalable to biobank data. 'SEAGLE' employs modern 
    matrix computations to achieve the same “exact” results as the original GxE VC 
    tests, and does not impose additional assumptions nor relies on approximations. 
    'SEAGLE' can easily accommodate sample sizes in the order of 10^5, is implementable 
    on standard laptops, and does not require specialized equipment. 
    The accompanying manuscript for this package can be found at Chi, Ipsen, Hsiao, 
    Lin, Wang, Lee, Lu, and Tzeng. (2021+) &lt;<a href="https://arxiv.org/abs/2105.03228">arXiv:2105.03228</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jocelynchi/SEAGLE">https://github.com/jocelynchi/SEAGLE</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), Matrix, CompQuadForm</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-05 21:19:27 UTC; jtc</td>
</tr>
<tr>
<td>Author:</td>
<td>Jocelyn Chi [aut, cre],
  Ilse Ipsen [aut],
  Jung-Ying Tzeng [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jocelyn Chi &lt;jocetchi@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-05 21:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='applyAt'>Function for applying t(A) on the left for REML EM</h2><span id='topic+applyAt'></span>

<h3>Description</h3>

<p>Function for applying t(A) on the left for REML EM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>applyAt(qrXtilde, RHS)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyAt_+3A_qrxtilde">qrXtilde</code></td>
<td>
<p>Object from QR decomposition of Xtilde</p>
</td></tr>
<tr><td><code id="applyAt_+3A_rhs">RHS</code></td>
<td>
<p>Object on right hand side of null of Xtilde^T</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix or vector resulting from left multiplication of At with matrix or vector input RHS
</p>

<hr>
<h2 id='cosihap'>Synthetic haplotype data generated from COSI software</h2><span id='topic+cosihap'></span>

<h3>Description</h3>

<p>A dataset containing 10,000 haplotypes of SNP sequences
mimicking the European population generated from the COSI software
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cosihap)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>dgCMatrix</code> with 10000 rows and 604 columns.
</p>


<h3>Source</h3>

<p><a href="https://genome.cshlp.org/content/15/11/1576.abstract">https://genome.cshlp.org/content/15/11/1576.abstract</a>
</p>

<hr>
<h2 id='estimate.vc'>REML EM Algorithm</h2><span id='topic+estimate.vc'></span>

<h3>Description</h3>

<p>REML EM algorithm for estimating variance components
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate.vc(
  y,
  Xtilde,
  qrXtilde,
  beta,
  G,
  init.sigma = 0.5,
  init.tau = 0.5,
  tol = 0.001,
  maxiters = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate.vc_+3A_y">y</code></td>
<td>
<p>Vector of observed phenotypes</p>
</td></tr>
<tr><td><code id="estimate.vc_+3A_xtilde">Xtilde</code></td>
<td>
<p>Matrix of covariates (first column contains the intercept, last column contains the E factor for studying the GxE effect)</p>
</td></tr>
<tr><td><code id="estimate.vc_+3A_qrxtilde">qrXtilde</code></td>
<td>
<p>Object containing QR decomposition of Xtilde</p>
</td></tr>
<tr><td><code id="estimate.vc_+3A_beta">beta</code></td>
<td>
<p>Coefficient vector for covariate matrix Xtilde</p>
</td></tr>
<tr><td><code id="estimate.vc_+3A_g">G</code></td>
<td>
<p>Matrix of genotype markers</p>
</td></tr>
<tr><td><code id="estimate.vc_+3A_init.sigma">init.sigma</code></td>
<td>
<p>Initial sigma input (Default is 0.5)</p>
</td></tr>
<tr><td><code id="estimate.vc_+3A_init.tau">init.tau</code></td>
<td>
<p>Initial tau input (Default is 0.5)</p>
</td></tr>
<tr><td><code id="estimate.vc_+3A_tol">tol</code></td>
<td>
<p>Tolerance for convergence (Default is 1e-3)</p>
</td></tr>
<tr><td><code id="estimate.vc_+3A_maxiters">maxiters</code></td>
<td>
<p>Maximum number of iterations (Default is 1000)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimates for tau and sigma
</p>

<hr>
<h2 id='makeSimData'>Generate synthetic data according to a fixed effects model</h2><span id='topic+makeSimData'></span>

<h3>Description</h3>

<p>This function generates synthetic from the fixed effects model described in the
experimental studies portion of the paper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeSimData(
  H,
  n,
  L = 100,
  maf = 0.01,
  gamma0 = 1,
  gammaX = 1,
  gammaE = 1,
  gammaG,
  gammaGE,
  causal = 40,
  seed = 12345
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeSimData_+3A_h">H</code></td>
<td>
<p>Matrix of haplotype data (e.g. cosihap)</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_n">n</code></td>
<td>
<p>Number of individuals</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_l">L</code></td>
<td>
<p>Number of SNPs in the G matrix (Default is 100), should be a value between 1 and 604</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_maf">maf</code></td>
<td>
<p>Minor allele frequency (Default is 0.01)</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_gamma0">gamma0</code></td>
<td>
<p>gamma0 Fixed effect coefficient for intercept (Default is 1)</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_gammax">gammaX</code></td>
<td>
<p>gammaX Fixed effect coefficient for confounding covariates (Default is 1)</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_gammae">gammaE</code></td>
<td>
<p>gammaE Fixed effect coefficient for E effect (Default is 1)</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_gammag">gammaG</code></td>
<td>
<p>gammaG Fixed effect coefficient for G main effect</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_gammage">gammaGE</code></td>
<td>
<p>gammaGE Fixed effect coefficient for GxE interaction effect</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_causal">causal</code></td>
<td>
<p>Number of causal SNPs (default is 40)</p>
</td></tr>
<tr><td><code id="makeSimData_+3A_seed">seed</code></td>
<td>
<p>Seed (Default is 12345)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Synthetic dataset containing y, X, E, G, epsilon, and number of causal SNPs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- makeSimData(H=cosihap, n=500, L=10, gammaG=1, gammaGE=0, causal=4, seed=1)

</code></pre>

<hr>
<h2 id='prep.SEAGLE'>Prepare data for input into SEAGLE function</h2><span id='topic+prep.SEAGLE'></span>

<h3>Description</h3>

<p>This function checks and formats data for input into SEAGLE function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep.SEAGLE(y, X, intercept, E, G)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prep.SEAGLE_+3A_y">y</code></td>
<td>
<p>Vector of observed phenotypes</p>
</td></tr>
<tr><td><code id="prep.SEAGLE_+3A_x">X</code></td>
<td>
<p>Matrix of covariates without genetic marker interactions</p>
</td></tr>
<tr><td><code id="prep.SEAGLE_+3A_intercept">intercept</code></td>
<td>
<p>1 if the first column of X is the all ones vector, 0 otherwise</p>
</td></tr>
<tr><td><code id="prep.SEAGLE_+3A_e">E</code></td>
<td>
<p>E Vector of environmental covariates</p>
</td></tr>
<tr><td><code id="prep.SEAGLE_+3A_g">G</code></td>
<td>
<p>G Matrix of genotype data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List object containing prepared data for input into SEAGLE function
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- makeSimData(H=cosihap, n=500, L=10, gammaG=1, gammaGE=0, causal=4, seed=1)
objSEAGLE &lt;- prep.SEAGLE(y=dat$y, X=dat$X, intercept=1, E=dat$E, G=dat$G)

</code></pre>

<hr>
<h2 id='Rinv.AtG'>Function for applying R inverse to AtG in REML EM algorithm</h2><span id='topic+Rinv.AtG'></span>

<h3>Description</h3>

<p>Function for applying R inverse to AtG in REML EM algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Rinv.AtG(G, AtG, GtAAtG, tau, sigma)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Rinv.AtG_+3A_g">G</code></td>
<td>
<p>Matrix of genotype markers (size n x L)</p>
</td></tr>
<tr><td><code id="Rinv.AtG_+3A_atg">AtG</code></td>
<td>
<p>AtG from pre-computation</p>
</td></tr>
<tr><td><code id="Rinv.AtG_+3A_gtaatg">GtAAtG</code></td>
<td>
<p>GtAAtG from pre-computation</p>
</td></tr>
<tr><td><code id="Rinv.AtG_+3A_tau">tau</code></td>
<td>
<p>Variance component from G main effect</p>
</td></tr>
<tr><td><code id="Rinv.AtG_+3A_sigma">sigma</code></td>
<td>
<p>Variance component from model noise epsilon</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix resulting from left multiplication of Rinv with input matrix AtG
</p>

<hr>
<h2 id='Rinv.u'>Function for applying R inverse to u in REML EM algorithm</h2><span id='topic+Rinv.u'></span>

<h3>Description</h3>

<p>Function for applying R inverse to u in REML EM algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Rinv.u(G, AtG, GtAAtG, GtAu, u, tau, sigma)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Rinv.u_+3A_g">G</code></td>
<td>
<p>Matrix of genotype markers (size n x L)</p>
</td></tr>
<tr><td><code id="Rinv.u_+3A_atg">AtG</code></td>
<td>
<p>AtG from precomputation</p>
</td></tr>
<tr><td><code id="Rinv.u_+3A_gtaatg">GtAAtG</code></td>
<td>
<p>GtAAtG from precomputation</p>
</td></tr>
<tr><td><code id="Rinv.u_+3A_gtau">GtAu</code></td>
<td>
<p>GtAu from precomputation</p>
</td></tr>
<tr><td><code id="Rinv.u_+3A_u">u</code></td>
<td>
<p>u=Aty from REML EM</p>
</td></tr>
<tr><td><code id="Rinv.u_+3A_tau">tau</code></td>
<td>
<p>Variance component from G main effect</p>
</td></tr>
<tr><td><code id="Rinv.u_+3A_sigma">sigma</code></td>
<td>
<p>Variance component from model noise epsilon</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector resulting from left multiplication of Rinv with input vector u
</p>

<hr>
<h2 id='SEAGLE'>Compute score-like test statistic and p-value for GxE test with SEAGLE algorithm</h2><span id='topic+SEAGLE'></span>

<h3>Description</h3>

<p>This function computes the score test statistic and corresponding
p-value for the GxE test with the SEAGLE algorithm with input data
that have been prepared with the prep.SEAGLE function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SEAGLE(obj.SEAGLE, init.tau = 0.5, init.sigma = 0.5, pv = "liu")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SEAGLE_+3A_obj.seagle">obj.SEAGLE</code></td>
<td>
<p>Input data prepared with prep.SEAGLE function</p>
</td></tr>
<tr><td><code id="SEAGLE_+3A_init.tau">init.tau</code></td>
<td>
<p>Initial estimate for tau (Default is 0.5)</p>
</td></tr>
<tr><td><code id="SEAGLE_+3A_init.sigma">init.sigma</code></td>
<td>
<p>Initial estimate for sigma (Default is 0.5)</p>
</td></tr>
<tr><td><code id="SEAGLE_+3A_pv">pv</code></td>
<td>
<p>Method of obtaining p-value (Either &quot;liu&quot; or &quot;davies&quot;, Default is liu)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Score-like test statistic T for the GxE effect and corresponding p-value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- makeSimData(H=cosihap, n=500, L=10, gammaG=1, gammaGE=0, causal=4, seed=1)
objSEAGLE &lt;- prep.SEAGLE(y=dat$y, X=dat$X, intercept=1, E=dat$E, G=dat$G)
res &lt;- SEAGLE(objSEAGLE, init.tau=0.5, init.sigma=0.5)

</code></pre>

<hr>
<h2 id='Vinv'>Function for applying V inverse in Algorithm 1</h2><span id='topic+Vinv'></span>

<h3>Description</h3>

<p>This function applies V inverse via the Woodbury matrix identity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Vinv(G, qrM, tau_over_sigma, sigma, RHS)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Vinv_+3A_g">G</code></td>
<td>
<p>Matrix of genotype markers (size n x L)</p>
</td></tr>
<tr><td><code id="Vinv_+3A_qrm">qrM</code></td>
<td>
<p>Pre-computation for LxL linear system solve</p>
</td></tr>
<tr><td><code id="Vinv_+3A_tau_over_sigma">tau_over_sigma</code></td>
<td>
<p>Tau over sigma from precomputation</p>
</td></tr>
<tr><td><code id="Vinv_+3A_sigma">sigma</code></td>
<td>
<p>Variance component from model noise epsilon</p>
</td></tr>
<tr><td><code id="Vinv_+3A_rhs">RHS</code></td>
<td>
<p>Matrix or vector on right-hand side of V inverse</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix or vector resulting from left multiplication of Vinv with input RHS
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
