<!DOCTYPE html><html lang="en"><head><title>Help for package seAMLess</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {seAMLess}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cli'><p>Command Line Interface for seAMLess</p></a></li>
<li><a href='#exampleTCGA'><p>TCGA-LAML bulk RNA-seq data downloaded from GDC</p></a></li>
<li><a href='#exampleTCGAmeta'><p>TCGA-LAML example data meta file downloaded from GDC</p></a></li>
<li><a href='#grch38'><p>Grch38</p></a></li>
<li><a href='#minRes'><p>A minimal seAMLess result list object</p></a></li>
<li><a href='#seAMLess'><p>Given the count matrices of bulk-RNA samples, this function deconvolutes</p>
each sample into its cell types using a healthy BM reference, and
calculates the sample's in vitro resistance to Venetoclax.</a></li>
<li><a href='#ternaryPlot'><p>Given the immune compositions (ICs) of bulk-RNA samples, this function creates</p>
a ternary plot similar to ALOT tube from EuroFlow analysis and Figure 1E of our paper.</a></li>
<li><a href='#venoModel'><p>Trained RF model on Venetoclax Resistance</p></a></li>
<li><a href='#verboseFn'><p>verboseFn</p></a></li>
<li><a href='#wrangleMat'><p>removes ERCC peaks and duplicated genes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Single Cell Transcriptomics Based Deconvolution Pipeline for
Leukemia</td>
</tr>
<tr>
<td>Description:</td>
<td>Given a bulk transcriptomic (RNA-seq) sample of an Myeloid Leukemia patient calculates immune composition and drug resistance for different small-molecule inhibitors. Published in <a href="https://www.nature.com/articles/s41698-024-00596-9">https://www.nature.com/articles/s41698-024-00596-9</a>.</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biobase, ggplot2, optparse, data.table,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>MuSiC, ggtern, seAMLessData, randomForest</td>
</tr>
<tr>
<td>Additional_repositories:</td>
<td><a href="https://eonurk.github.io/drat/">https://eonurk.github.io/drat/</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/eonurk/seAMLess">https://github.com/eonurk/seAMLess</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-11 10:18:00 UTC; onur-lumc</td>
</tr>
<tr>
<td>Author:</td>
<td>E Onur Karakaslar [aut, cre],
  Redmar van den Berg [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>E Onur Karakaslar &lt;eonurkara@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-11 12:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cli'>Command Line Interface for seAMLess</h2><span id='topic+cli'></span>

<h3>Description</h3>

<p>Provides a command line interface to run seAMLess deconvolution analysis
on bulk RNA-seq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cli()
</code></pre>


<h3>Value</h3>

<p>List of validated command line options
</p>

<hr>
<h2 id='exampleTCGA'>TCGA-LAML bulk RNA-seq data downloaded from GDC</h2><span id='topic+exampleTCGA'></span>

<h3>Description</h3>

<p>TCGA-LAML bulk RNA-seq data downloaded from GDC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exampleTCGA)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 60483 rows and 21 columns.
</p>

<hr>
<h2 id='exampleTCGAmeta'>TCGA-LAML example data meta file downloaded from GDC</h2><span id='topic+exampleTCGAmeta'></span>

<h3>Description</h3>

<p>TCGA-LAML example data meta file downloaded from GDC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exampleTCGAmeta)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 20 rows and 34 columns.
</p>

<hr>
<h2 id='grch38'>Grch38</h2><span id='topic+grch38'></span>

<h3>Description</h3>

<p>Grch38
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(grch38)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 67495 rows and 3 columns.
</p>

<hr>
<h2 id='minRes'>A minimal seAMLess result list object</h2><span id='topic+minRes'></span>

<h3>Description</h3>

<p>A minimal seAMLess result list object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(minRes)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='seAMLess'>Given the count matrices of bulk-RNA samples, this function deconvolutes
each sample into its cell types using a healthy BM reference, and
calculates the sample's in vitro resistance to Venetoclax.</h2><span id='topic+seAMLess'></span>

<h3>Description</h3>

<p>Given the count matrices of bulk-RNA samples, this function deconvolutes
each sample into its cell types using a healthy BM reference, and
calculates the sample's in vitro resistance to Venetoclax.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seAMLess(
  mat,
  scRef = seAMLessData::scRef,
  scRef.sample = "Sample",
  scRef.label = "label.new",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="seAMLess_+3A_mat">mat</code></td>
<td>
<p>count matrix (genes by 1+samples).</p>
</td></tr>
<tr><td><code id="seAMLess_+3A_scref">scRef</code></td>
<td>
<p>reference matrix for single cell data</p>
</td></tr>
<tr><td><code id="seAMLess_+3A_scref.sample">scRef.sample</code></td>
<td>
<p>column name for the samples in single cell reference</p>
</td></tr>
<tr><td><code id="seAMLess_+3A_scref.label">scRef.label</code></td>
<td>
<p>column name for the cell names in single cell reference</p>
</td></tr>
<tr><td><code id="seAMLess_+3A_verbose">verbose</code></td>
<td>
<p>prints detailed messages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of deconvoluted cell type percentages and predicted drug resistances
</p>

<hr>
<h2 id='ternaryPlot'>Given the immune compositions (ICs) of bulk-RNA samples, this function creates
a ternary plot similar to ALOT tube from EuroFlow analysis and Figure 1E of our paper.</h2><span id='topic+ternaryPlot'></span>

<h3>Description</h3>

<p>Given the immune compositions (ICs) of bulk-RNA samples, this function creates
a ternary plot similar to ALOT tube from EuroFlow analysis and Figure 1E of our paper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ternaryPlot(res)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ternaryPlot_+3A_res">res</code></td>
<td>
<p>seAMLess object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seAMLess)

data(minRes)
ternaryPlot(minRes)

</code></pre>

<hr>
<h2 id='venoModel'>Trained RF model on Venetoclax Resistance</h2><span id='topic+venoModel'></span>

<h3>Description</h3>

<p>Trained RF model on Venetoclax Resistance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(venoModel)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>randomForest</code> of length 17.
</p>

<hr>
<h2 id='verboseFn'>verboseFn</h2><span id='topic+verboseFn'></span>

<h3>Description</h3>

<p>returns a printing function to be used with in the script
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verboseFn(verbose)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verboseFn_+3A_verbose">verbose</code></td>
<td>
<p>boolean, determines whether the output going be printed or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>print function
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Prints output
verbosePrint &lt;- verboseFn(TRUE)
verbosePrint("Hello World!")
# &gt; "Hello World!"

# Does not print
verbosePrint &lt;- verboseFn(FALSE)
verbosePrint("Hello World!")
</code></pre>

<hr>
<h2 id='wrangleMat'>removes ERCC peaks and duplicated genes</h2><span id='topic+wrangleMat'></span>

<h3>Description</h3>

<p>removes ERCC peaks and duplicated genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wrangleMat(mat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wrangleMat_+3A_mat">mat</code></td>
<td>
<p>pre-filters and orders bulk rna-seq data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>filtered and ordered count-matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seAMLess)

data("exampleTCGA")

exampleTCGA &lt;- wrangleMat(exampleTCGA)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
