<!DOCTYPE html><html><head><title>Help for package SeedMatchR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SeedMatchR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SeedMatchR-package'><p>SeedMatchR: Find Matches to Canonical SiRNA Seeds in Genomic Features</p></a></li>
<li><a href='#check_gene_list_overlap'><p>Check if input gene lists overlap</p></a></li>
<li><a href='#deseq_fc_ecdf'><p>Plot the ECDF for DESeq2 log2(Fold Changes)</p></a></li>
<li><a href='#download_parse_file'><p>Download and parse DESeq2 output from GSE184929</p></a></li>
<li><a href='#ecdf_stat_test'><p>Test for differences in log2(Fold Change) ECDFs between two gene lists using</p>
the <code>stats</code> package</a></li>
<li><a href='#filter_deseq'><p>Filter DESEQ2 Results for SeedMatchR</p></a></li>
<li><a href='#get_example_data'><p>Download example DESeq2 data from GEO</p></a></li>
<li><a href='#get_feature_seqs'><p>Get transcripts features and feature sequences</p></a></li>
<li><a href='#get_seed'><p>Get the target seed sequence given a canonical seed name and input sequence</p></a></li>
<li><a href='#load_example_data'><p>Load example DESeq2 data into the environment</p></a></li>
<li><a href='#load_species_anno_db'><p>Load species specific <code>AnnotationDb</code></p></a></li>
<li><a href='#plot_seeds'><p>Plot the Guide Strand with different optional seeds</p></a></li>
<li><a href='#SeedMatchR'><p>Find seed matches in genomic features</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Find Matches to Canonical SiRNA Seeds in Genomic Features</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>On-target gene knockdown using siRNA ideally results from binding 
    fully complementary regions in mRNA transcripts to induce cleavage. 
    Off-target siRNA gene knockdown can occur through several modes, one being 
    a seed-mediated mechanism mimicking miRNA gene regulation. Seed-mediated 
    off-target effects occur when the ~8 nucleotides at the 5’ end of the guide 
    strand, called a seed region, bind the 3’ untranslated regions of mRNA, 
    causing reduced translation. Experiments using siRNA knockdown paired with 
    RNA-seq can be used to detect siRNA sequences with potential off-target 
    effects driven by the seed region. 'SeedMatchR' provides tools for exploring 
    and detecting potential seed-mediated off-target effects of siRNA in RNA-seq 
    experiments. 'SeedMatchR' is designed to extend current differential 
    expression analysis tools, such as 'DESeq2', by annotating results with 
    predicted seed matches. Using publicly available data, we demonstrate the 
    ability of 'SeedMatchR' to detect cumulative changes in differential gene 
    expression attributed to siRNA seed regions. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0), org.Rn.eg.db</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, Biostrings, ggmsa, msa, ggplot2, utils, AnnotationHub,
GenomeInfoDb, GenomicFeatures, cowplot, testit, lifecycle,
stringr, stats</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://tacazares.github.io/SeedMatchR/">https://tacazares.github.io/SeedMatchR/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-24 14:16:38 UTC; l199328</td>
</tr>
<tr>
<td>Author:</td>
<td>Tareian Cazares <a href="https://orcid.org/0000-0003-4624-6156"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Gulcin Ozer <a href="https://orcid.org/0000-0002-8315-8051"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Jibo Wang [aut],
  Rick Higgs [aut],
  Eli Lilly and Company [cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tareian Cazares &lt;tareian.cazares@lilly.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-24 20:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='SeedMatchR-package'>SeedMatchR: Find Matches to Canonical SiRNA Seeds in Genomic Features</h2><span id='topic+SeedMatchR-package'></span><span id='topic+_PACKAGE'></span>

<h3>Description</h3>

<p>On-target gene knockdown using siRNA ideally results from binding fully complementary regions in mRNA transcripts to induce cleavage. Off-target siRNA gene knockdown can occur through several modes, one being a seed-mediated mechanism mimicking miRNA gene regulation. Seed-mediated off-target effects occur when the ~8 nucleotides at the 5’ end of the guide strand, called a seed region, bind the 3’ untranslated regions of mRNA, causing reduced translation. Experiments using siRNA knockdown paired with RNA-seq can be used to detect siRNA sequences with potential off-target effects driven by the seed region. 'SeedMatchR' provides tools for exploring and detecting potential seed-mediated off-target effects of siRNA in RNA-seq experiments. 'SeedMatchR' is designed to extend current differential expression analysis tools, such as 'DESeq2', by annotating results with predicted seed matches. Using publicly available data, we demonstrate the ability of 'SeedMatchR' to detect cumulative changes in differential gene expression attributed to siRNA seed regions.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Tareian Cazares <a href="mailto:tareian.cazares@lilly.com">tareian.cazares@lilly.com</a> (<a href="https://orcid.org/0000-0003-4624-6156">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Gulcin Ozer <a href="mailto:ozer_gulcin@lilly.com">ozer_gulcin@lilly.com</a> (<a href="https://orcid.org/0000-0002-8315-8051">ORCID</a>)
</p>
</li>
<li><p> Jibo Wang
</p>
</li>
<li><p> Rick Higgs
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Eli Lilly and Company [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://tacazares.github.io/SeedMatchR/">https://tacazares.github.io/SeedMatchR/</a>
</p>
</li></ul>


<hr>
<h2 id='check_gene_list_overlap'>Check if input gene lists overlap</h2><span id='topic+check_gene_list_overlap'></span>

<h3>Description</h3>

<p>Check if input gene lists overlap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gene_list_overlap(gene.lists)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_gene_list_overlap_+3A_gene.lists">gene.lists</code></td>
<td>
<p>A list of gene lists.
example: list(c(&quot;gene1&quot;, &quot;gene2&quot;), c(&quot;gene1&quot;))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Warning if gene sets overlap
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Overlap
check_gene_list_overlap(list(c("gene1", "gene2"), c("gene1")))

#No overlap
check_gene_list_overlap(list(c("gene1", "gene2"), c("gene3")))
</code></pre>

<hr>
<h2 id='deseq_fc_ecdf'>Plot the ECDF for DESeq2 log2(Fold Changes)</h2><span id='topic+deseq_fc_ecdf'></span>

<h3>Description</h3>

<p>This functions will take DESeq2 results as a <code>data.frame</code> and
plot the ecdf for the input <code>gene.lists</code>.
</p>
<p>The gene sets to plot should be provided as a list of lists.
</p>
<p>Example:
</p>
<p><code>gene.lists = list("Background" = c("gene1", "gene2"), "Target" = c("gene2", "gene3"), "Overlap" = c("gene2"))</code>
</p>
<p>This function will also perform statistical testing if <code>plot.hist</code> is TRUE.
The output will be saved to a PDF if an <code>output.filename</code> is provided.
</p>
<p>Users can define the groups that are to be compared in the statistical test
using the <code>null.name</code> and <code>target.name</code> arguments. The names must be found
in <code>gene.lists</code>. The <code>factor.order</code> is used to order the groups in the
analysis.
</p>
<p>This functions returns:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;$plot&#8288;</code>: The ECDF plot
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;$stats&#8288;</code>: The stats results object
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>deseq_fc_ecdf(
  res,
  gene.lists,
  title = "ECDF",
  output.filename = NULL,
  palette = SeedMatchR.palette,
  factor.order = NULL,
  x.lims = c(-1, 1),
  stats.test = NULL,
  alternative = "greater",
  null.name = 1,
  target.name = 2,
  height = 5,
  width = 5,
  dpi = 320
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deseq_fc_ecdf_+3A_res">res</code></td>
<td>
<p>The DESeq2 results dataframe</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_gene.lists">gene.lists</code></td>
<td>
<p>A nest list of gene names. Example:
gene.lists = list(&quot;Background&quot; = gene.list2, &quot;Target&quot; = gene.list1,
&quot;Overlap&quot; = gene.list3)</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_title">title</code></td>
<td>
<p>The tile of the plot</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_output.filename">output.filename</code></td>
<td>
<p>If the output filename is provided, then the plot is
saved.</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_palette">palette</code></td>
<td>
<p>The color palette to use for your curves</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_factor.order">factor.order</code></td>
<td>
<p>The order to use for the legends</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_x.lims">x.lims</code></td>
<td>
<p>The xlimits range</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_stats.test">stats.test</code></td>
<td>
<p>The statistic test to use. Options: KS, Kuiper, DTS, CVM,
AD, Wass</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_alternative">alternative</code></td>
<td>
<p>The alternative hypothesis to test. Options: greater, less, two.sided</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_null.name">null.name</code></td>
<td>
<p>The name in the gene.list to use as the null for ecdf plots</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_target.name">target.name</code></td>
<td>
<p>The name in the gene.list to use as the target for ecdf plots</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_height">height</code></td>
<td>
<p>Plot height in inches</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_width">width</code></td>
<td>
<p>Plot width in inches</p>
</td></tr>
<tr><td><code id="deseq_fc_ecdf_+3A_dpi">dpi</code></td>
<td>
<p>The dpi resolution for the figure</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object for the ECDF plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

guide.seq = "UUAUAGAGCAAGAACACUGUUUU"

anno.db = load_species_anno_db("human")

features = get_feature_seqs(anno.db$tx.db, anno.db$dna)

# Load test data
get_example_data("sirna")

sirna.data = load_example_data("sirna")

res &lt;- sirna.data$Schlegel_2022_Ttr_D1_30mkg

# Filter DESeq2 results for SeedMatchR
res = filter_deseq(res, fdr.cutoff=1, fc.cutoff=0, rm.na.log2fc = TRUE)

res = SeedMatchR(res, anno.db$gtf, features$seqs, guide.seq, "mer7m8")

# Gene set 1
mer7m8.list = res$gene_id[res$mer7m8 &gt;= 1]

# Gene set 2
background.list = res$gene_id[!(res$mer7m8 %in% mer7m8.list)]

ecdf.results = deseq_fc_ecdf(res,
list("Background" = background.list, "mer7m8" = mer7m8.list),
stats.test = "KS",
factor.order = c("Background", "mer7m8"),
null.name = "Background",
target.name = "mer7m8")

</code></pre>

<hr>
<h2 id='download_parse_file'>Download and parse DESeq2 output from GSE184929</h2><span id='topic+download_parse_file'></span>

<h3>Description</h3>

<p>Download and parse DESeq2 output from GSE184929
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_parse_file(download.path, output.path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download_parse_file_+3A_download.path">download.path</code></td>
<td>
<p>File url to be downloaded</p>
</td></tr>
<tr><td><code id="download_parse_file_+3A_output.path">output.path</code></td>
<td>
<p>Filename used for saving downloaded file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DESeq2 results as a data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
download_parse_file()

</code></pre>

<hr>
<h2 id='ecdf_stat_test'>Test for differences in log2(Fold Change) ECDFs between two gene lists using
the <code>stats</code> package</h2><span id='topic+ecdf_stat_test'></span>

<h3>Description</h3>

<p>This function uses the <code>stats</code> package to test the ECDF
of log2(Fold Changes) between two groups based on DESeq2 analysis.
</p>
<p>The inputs of this function are a DESeq2 results <code>data.frame</code> and two sets of
gene IDs called <code>gene.list1</code> and <code>gene.list2</code>. The functions will look for a
column called <code>log2FoldChange</code> in the dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ecdf_stat_test(
  res,
  gene.list1,
  gene.list2,
  stats.test = "KS",
  alternative = "greater"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ecdf_stat_test_+3A_res">res</code></td>
<td>
<p>Input results file data frame</p>
</td></tr>
<tr><td><code id="ecdf_stat_test_+3A_gene.list1">gene.list1</code></td>
<td>
<p>Gene list 1: Usually null distribution</p>
</td></tr>
<tr><td><code id="ecdf_stat_test_+3A_gene.list2">gene.list2</code></td>
<td>
<p>Gene list 2: Target set of genes</p>
</td></tr>
<tr><td><code id="ecdf_stat_test_+3A_stats.test">stats.test</code></td>
<td>
<p>Stats test to use. Options: KS or Wilcoxen</p>
</td></tr>
<tr><td><code id="ecdf_stat_test_+3A_alternative">alternative</code></td>
<td>
<p>The alternative hypothesis to test. Options: greater, less, two.sided</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing the dstat and pvalue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

guide.seq = "UUAUAGAGCAAGAACACUGUUUU"

anno.db = load_species_anno_db("human")

features = get_feature_seqs(anno.db$tx.db, anno.db$dna)

# Load test data
get_example_data("sirna")

sirna.data = load_example_data("sirna")

res &lt;- sirna.data$Schlegel_2022_Ttr_D1_30mkg

# Filter DESeq2 results for SeedMatchR
res = filter_deseq(res, fdr.cutoff=1, fc.cutoff=0, rm.na.log2fc = TRUE)

res = SeedMatchR(res, anno.db$gtf, features$seqs, guide.seq, "mer7m8")

# Gene set 1
mer7m8.list = res$gene_id[res$mer7m8 &gt;= 1]

# Gene set 2
background.list = res$gene_id[!(res$mer7m8 %in% mer7m8.list)]

ecdf.res = ecdf_stat_test(res, mer7m8.list, background.list)

</code></pre>

<hr>
<h2 id='filter_deseq'>Filter DESEQ2 Results for SeedMatchR</h2><span id='topic+filter_deseq'></span>

<h3>Description</h3>

<p>Filter <code>DESeqDataSet</code> results for use with seed matching
and counting functions.
</p>
<p>The filtering criteria are:
</p>
<p>Filter out genes that are not expressed or counted at all:
baseMean = 0 &amp; pvalue = NA &amp; log2FoldChange = NA
</p>
<p>Filter out genes that are expressed, but there is not difference across
groups: log2FoldChange = 0
</p>
<p>Filter out genes with extreme outliers: pvalue = NA and padj = NA
</p>
<p>Filter out genes that have been excluded by independent filtering. padj = NA
</p>
<p>Filter results by the <code>fdr.cutoff</code>
</p>
<p>Filter the results by the <code>log2FoldChange</code>
</p>
<p>Filter the results by the <code>baseMean</code>
</p>
<p>Remove NA gene_ids and log2FoldChange values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_deseq(
  res,
  fdr.cutoff = 1,
  fc.cutoff = 0,
  rm.na.name = FALSE,
  rm.na.log2fc = FALSE,
  baseMean.cutoff = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_deseq_+3A_res">res</code></td>
<td>
<p>The DESEQ2 results as a data frame</p>
</td></tr>
<tr><td><code id="filter_deseq_+3A_fdr.cutoff">fdr.cutoff</code></td>
<td>
<p>The false discovery rate cutoff to use.</p>
</td></tr>
<tr><td><code id="filter_deseq_+3A_fc.cutoff">fc.cutoff</code></td>
<td>
<p>The fold change cutoff to use. The absolute
value will be used as the cutoff and values greater-than-or-equal-to will
be kept.</p>
</td></tr>
<tr><td><code id="filter_deseq_+3A_rm.na.name">rm.na.name</code></td>
<td>
<p>Remove na values from the gene_name column</p>
</td></tr>
<tr><td><code id="filter_deseq_+3A_rm.na.log2fc">rm.na.log2fc</code></td>
<td>
<p>Remove na values from the log2FoldChange column</p>
</td></tr>
<tr><td><code id="filter_deseq_+3A_basemean.cutoff">baseMean.cutoff</code></td>
<td>
<p>The minimum baseMean expression cutoff</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified DESEQ2 results table that has been filtered
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load test data
get_example_data("sirna")

sirna.data = load_example_data("sirna")

res &lt;- sirna.data$Schlegel_2022_Ttr_D1_30mkg

# Filter DESeq2 results for SeedMatchR
res = filter_deseq(res, fdr.cutoff=1, fc.cutoff=0, rm.na.log2fc = TRUE)

</code></pre>

<hr>
<h2 id='get_example_data'>Download example DESeq2 data from GEO</h2><span id='topic+get_example_data'></span>

<h3>Description</h3>

<p>This function will download data that can be used  for SeedMatchR.
Choosing 'sirna' will download 3 DESeq2 results files from GSE184929. Choosing
'mirna' will download the miRDB database as a tsv.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_example_data(example.type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_example_data_+3A_example.type">example.type</code></td>
<td>
<p>Name of the example to load. Options: sirna, mirna</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None?
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
get_example_data()

</code></pre>

<hr>
<h2 id='get_feature_seqs'>Get transcripts features and feature sequences</h2><span id='topic+get_feature_seqs'></span>

<h3>Description</h3>

<p>This function is used to get the genomic features of interest
and the DNA sequences associated with them. This function takes advantage of
the <code>GenomicFeatures</code> package functions <code>threeUTRsByTranscript</code>,
<code>fiveUTRsByTranscript</code>, <code>exonsBy</code>, <code>intronsByTranscript</code>, and <code>cdsBy</code>. These
functions are used to generate the features given an input <code>tx.db</code> object. A
2bit <code>dna</code> input is also required for extracting features sequences.
</p>
<p>The output of the this function is:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;$db&#8288;</code>: the feature <code>GRanges</code> object
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;$seqs&#8288;</code>: <code>DNAStringSet</code> of sequences associated to those features
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>get_feature_seqs(tx.db, dna, feature.type = "3UTR")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_feature_seqs_+3A_tx.db">tx.db</code></td>
<td>
<p>A tx.db object</p>
</td></tr>
<tr><td><code id="get_feature_seqs_+3A_dna">dna</code></td>
<td>
<p>A 2bit dna sequence</p>
</td></tr>
<tr><td><code id="get_feature_seqs_+3A_feature.type">feature.type</code></td>
<td>
<p>The type of feature to return. Options: 3UTR, 5UTR,
exons, introns, cds</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing the feature db object and the feature sequences
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
anno.db = load_species_anno_db("human")

features = get_feature_seqs(anno.db$tx.db, anno.db$dna)

</code></pre>

<hr>
<h2 id='get_seed'>Get the target seed sequence given a canonical seed name and input sequence</h2><span id='topic+get_seed'></span>

<h3>Description</h3>

<p>Given a sequence greater than 8 bp oriented 5' -&gt; 3' and a seed
definition, this function will return an object containing seed-specific sequence
information. Users can input a custom seed name, but must provide the start
position (<code>start.pos</code>) and stop position (<code>stop.pos</code>) that define the
range of the seed sequence.
</p>
<p>Built-in options: <code>mer8</code>,  <code>mer7A1</code>, <code>mer7m8</code>, <code>mer6</code>
</p>
<p>Note: The seed definitions <code>mer8</code> and <code>mer7A1</code> force a U at position g1.
This results in an A in the target sequence being searched.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_seed(guide.seq, seed.name = "mer7m8", start.pos = 1, stop.pos = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_seed_+3A_guide.seq">guide.seq</code></td>
<td>
<p>A character string greater than 8 bp and oriented 5'-&gt; 3'.</p>
</td></tr>
<tr><td><code id="get_seed_+3A_seed.name">seed.name</code></td>
<td>
<p>The seed name of interest. Options: mer8,  mer7A1, mer7m8,
mer6. If not in the default list, the start.pos and stop.pos arguments
will be used to define the seed.</p>
</td></tr>
<tr><td><code id="get_seed_+3A_start.pos">start.pos</code></td>
<td>
<p>The start position for a custom seed definition</p>
</td></tr>
<tr><td><code id="get_seed_+3A_stop.pos">stop.pos</code></td>
<td>
<p>The stop position for a custom seed definition</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with the entries:
</p>

<ul>
<li> <p><code>Guide</code>: Input guide sequence. Input is expected to be RNA.
</p>
</li>
<li> <p><code>Seed.Name</code>: The seed name.
</p>
</li>
<li> <p><code>Seed.Seq.RNA</code>: The seed sequence as a <code>RNAString</code>
</p>
</li>
<li> <p><code>Seed.Seq.DNA</code>: The seed sequence as a <code>DNAString</code>
</p>
</li>
<li> <p><code>Target.Seq</code>: The target DNA sequence based on the reverse complement of
the seed as a <code>DNAString</code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Example Ttr from Schlegel et al. 2022
guide.seq = "UUAUAGAGCAAGAACACUGUUUU"

# Get seed match
seed.seq = get_seed(guide.seq, "mer7m8")
</code></pre>

<hr>
<h2 id='load_example_data'>Load example DESeq2 data into the environment</h2><span id='topic+load_example_data'></span>

<h3>Description</h3>

<p>Load example DESeq2 data into the environment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_example_data(example.type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_example_data_+3A_example.type">example.type</code></td>
<td>
<p>Name of the example to load. Options: sirna, mirna</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Loads either the Schlegel 2022 RNAseq data or miRDB into the environment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
load_example_data()

</code></pre>

<hr>
<h2 id='load_species_anno_db'>Load species specific <code>AnnotationDb</code></h2><span id='topic+load_species_anno_db'></span>

<h3>Description</h3>

<p>Use <code>AnnotationHub</code> to load species-specific GTF and 2bit DNA
sequences. This function currently works for human, rat, and mouse.
</p>
<p>The function will return:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;$gtf&#8288;</code>: A <code>GRanges</code> object containing the GTF information
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;$tx.db&#8288;</code>: A <code>tx.db</code> object made from the GTF
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;$dna&#8288;</code>: The 2bit DNA sequence as a <code>DNAStringSet</code>
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>load_species_anno_db(species.name, remove.na.rows = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_species_anno_db_+3A_species.name">species.name</code></td>
<td>
<p>Species name. Options: human, rat, mouse</p>
</td></tr>
<tr><td><code id="load_species_anno_db_+3A_remove.na.rows">remove.na.rows</code></td>
<td>
<p>Remove rows with NA in the gene_id column</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Species specific <code>AnnotationDb</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
anno.db = load_species_anno_db("human")

</code></pre>

<hr>
<h2 id='plot_seeds'>Plot the Guide Strand with different optional seeds</h2><span id='topic+plot_seeds'></span>

<h3>Description</h3>

<p>Plot the Guide Strand with different optional seeds
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_seeds(guide.seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_seeds_+3A_guide.seq">guide.seq</code></td>
<td>
<p>Guide a.k.a anti-sense sequence oriented 5' &gt; 3'. Sequence must
be greater than 8 bp.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A msaggplot of the guide sequence in addition to the available
seed sequences
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(msa)

# Ttr siRNA sequence
guide.seq = "UUAUAGAGCAAGAACACUGUUUU"

# generate seed plot
plotted.seeds = plot_seeds(guide.seq)
</code></pre>

<hr>
<h2 id='SeedMatchR'>Find seed matches in genomic features</h2><span id='topic+SeedMatchR'></span>

<h3>Description</h3>

<p>Find seed matches in a <code>DNAStringSet</code> object of
sequences. This function will use <code>get.seed</code> extract the seed sequence
from the guide sequence. The seed is then searched across all rows of the
<code>DNAStringSet</code> object using <code>vpatterncount</code>.
</p>
<p>This function returns the input DESeq2 results <code>data.frame</code> with an
additional column that contains the counts for the input <code>seed.name</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeedMatchR(
  res,
  gtf,
  seqs,
  sequence,
  seed.name = "mer7m8",
  col.name = NULL,
  mismatches = 0,
  indels = FALSE,
  tx.id.col = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SeedMatchR_+3A_res">res</code></td>
<td>
<p>A DESeq2 results <code>data.frame</code></p>
</td></tr>
<tr><td><code id="SeedMatchR_+3A_gtf">gtf</code></td>
<td>
<p>GTF file used to map features to genes. The object must have
columns transcript_id and gene_id</p>
</td></tr>
<tr><td><code id="SeedMatchR_+3A_seqs">seqs</code></td>
<td>
<p>The <code>DNAStringSet</code> object with sequence information
for features. The names of the sequences should be the transcript names.</p>
</td></tr>
<tr><td><code id="SeedMatchR_+3A_sequence">sequence</code></td>
<td>
<p>The <code>DNAString</code> guide sequence oriented 5' &gt; 3'.</p>
</td></tr>
<tr><td><code id="SeedMatchR_+3A_seed.name">seed.name</code></td>
<td>
<p>The name of specific seed to extract. Options are:  mer8,
mer7A1, mer7m8, mer6</p>
</td></tr>
<tr><td><code id="SeedMatchR_+3A_col.name">col.name</code></td>
<td>
<p>The string to use for the column name. Defaults to seed name</p>
</td></tr>
<tr><td><code id="SeedMatchR_+3A_mismatches">mismatches</code></td>
<td>
<p>The number of mismatches to allow in search</p>
</td></tr>
<tr><td><code id="SeedMatchR_+3A_indels">indels</code></td>
<td>
<p>Whether to allow indels in search</p>
</td></tr>
<tr><td><code id="SeedMatchR_+3A_tx.id.col">tx.id.col</code></td>
<td>
<p>Use the transcript_id column instead of gene_id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified DESeq2 results dataframe that has column named after the
seed of choice representing the number of match counts.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)

seq = "UUAUAGAGCAAGAACACUGUUUU"

anno.db = load_species_anno_db("human")

features = get_feature_seqs(anno.db$tx.db, anno.db$dna)

# Load test data
res &lt;- Schlegel_2022_Ttr_D1_30mkg

# Filter DESeq2 results for SeedMatchR
res = filter_deseq(res, fdr.cutoff=1, fc.cutoff=0, rm.na.log2fc = TRUE)

res = SeedMatchR(res, anno.db$gtf, features$seqs, seq, "mer7m8")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
