<!DOCTYPE html><html><head><title>Help for package seedCCA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {seedCCA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#coef.seedCCA'><p>Coefficients of ordinary and partial least squares through iterative projections</p></a></li>
<li><a href='#cookie'><p>cookie dataset</p></a></li>
<li><a href='#covplot'><p>scree-ploting cov(X, Y)</p></a></li>
<li><a href='#finalCCA'><p>finalized CCA in seeded CCA</p></a></li>
<li><a href='#fitted.seedCCA'><p>Fitted values of ordinary and partial least squares</p></a></li>
<li><a href='#iniCCA'><p>Initialized CCA in seeded CCA</p></a></li>
<li><a href='#nutrimouse'><p>Nutrimouse dataset</p></a></li>
<li><a href='#plot.seedCCA'><p>Plotting class &quot;seedCCA&quot; depending on the value of <code>type</code></p></a></li>
<li><a href='#Pm'><p>Projection of a seed matrix on to the column subspace of M with respect to <code>Sx</code> inner-product</p></a></li>
<li><a href='#print.seedCCA'><p>basic function for printing class &quot;seedCCA&quot;</p></a></li>
<li><a href='#seedCCA'><p>Seeded Canonical correlation analysis</p></a></li>
<li><a href='#seeding'><p>increments of iterative projections</p></a></li>
<li><a href='#seeding.auto.stop'><p>increments of iterative projections with automatic stopping</p></a></li>
<li><a href='#seedols'><p>Ordinary least squares</p></a></li>
<li><a href='#seedpls'><p>Partial least squares through iterative projections</p></a></li>
<li><a href='#selectu'><p>Function that guides a selection of the terminating index when using seedCCA function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Seeded Canonical Correlation Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Jae Keun Yoo, Bo-Young Kim</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jae Keun Yoo &lt;peter.yoo@ewha.ac.kr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for dimension reduction through the seeded canonical correlation analysis are provided. A classical canonical correlation analysis (CCA) is one of useful statistical methods in multivariate data analysis, but it is limited in use due to the matrix inversion for large p small n data. To overcome this, a seeded CCA has been proposed in Im, Gang and Yoo (2015) \doi{10.1002/cem.2691}. The seeded CCA is a two-step procedure. The sets of variables are initially reduced by successively projecting cov(X,Y) or cov(Y,X) onto cov(X) and cov(Y), respectively, without loss of information on canonical correlation analysis, following Cook, Li and Chiaromonte (2007) \doi{10.1093/biomet/asm038} and Lee and Yoo (2014) \doi{10.1111/anzs.12057}. Then, the canonical correlation is finalized with the initially-reduced two sets of variables.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2.0)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>CCA, corpcor, stats, graphics</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-09 00:48:57 UTC; user</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-09 04:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='coef.seedCCA'>Coefficients of ordinary and partial least squares through iterative projections</h2><span id='topic+coef.seedCCA'></span>

<h3>Description</h3>

<p>Returns coefficients of partial least squares through iterative projections. It works only for subclasses &quot;seedols&quot; and seedpls&quot;.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'seedCCA'
coef(object, u=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.seedCCA_+3A_object">object</code></td>
<td>
<p>The name of an object of class &quot;seedCCA&quot;</p>
</td></tr>
<tr><td><code id="coef.seedCCA_+3A_u">u</code></td>
<td>
<p>numeric, the number of projections. The default is NULL. This option is valid for PLS alone. The option returns the coefficient estimates for u projections. For example, if it is specified at k, then the coefficient estimates with k projections are returned.</p>
</td></tr>
<tr><td><code id="coef.seedCCA_+3A_...">...</code></td>
<td>
<p>arguments passed to the coef.method</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>########  data(cookie) ########
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])

fit.ols1 &lt;- seedCCA(X[,1:4], Y[,1], type="cca")
fit.pls1 &lt;- seedCCA(X,Y[,1],type="pls")

coef(fit.ols1)
coef(fit.pls1)
coef(fit.pls1, u=4)
</code></pre>

<hr>
<h2 id='cookie'>cookie dataset</h2><span id='topic+cookie'></span>

<h3>Description</h3>

<p>This data set contains measurements from quantitative NIR spectroscopy. The example studied arises from an experiment done to test the feasibility of NIR spectroscopy to measure the composition of biscuit dough pieces (formed but unbaked biscuits). Two similar sample sets were made up, with the standard recipe varied to provide a large range for each of the four constituents under investigation: fat, sucrose, dry flour, and water. The calculated percentages of these four ingredients represent the 4 responses. There are 40 samples in the calibration or training set (with sample 23 being an outlier) and a further 32 samples in the separate prediction or validation set (with example 21 considered as an outlier). 
</p>
<p>An NIR reflectance spectrum is available for each dough piece. The spectral data consist of 700 points measured from 1100 to 2498 nanometers (nm) in steps of 2 nm. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cookie)
</code></pre>


<h3>Format</h3>

<p>A data frame of dimension 72 x 704. The first 700 columns correspond to the NIR reflectance spectrum, the last four columns correspond to the four constituents fat, sucrose, dry flour, and water. The first 40 rows correspond to the calibration data, the last 32 rows correspond to the prediction data.
</p>


<h3>References</h3>

<p>Please cite the following papers if you use this data set. 
</p>
<p>P.J. Brown, T. Fearn, and M. Vannucci (2001) Bayesian Wavelet Regression on Curves with Applications to a Spectroscopic Calibration Problem. Journal of the American Statistical Association, 96, pp. 398-408. 
</p>
<p>B.G. Osborne, T. Fearn, A.R. Miller, and S. Douglas (1984) Application of Near-Infrared Reflectance Spectroscopy to Compositional Analysis of Biscuits and Biscuit Dough. Journal of the Science of Food and Agriculture, 35, pp. 99 - 105. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(cookie) # load data
    X&lt;-as.matrix(cookie[,1:700]) # extract NIR spectra
    Y&lt;-as.matrix(cookie[,701:704]) # extract constituents
    Xtrain&lt;-X[1:40,] # extract training data
    Ytrain&lt;-Y[1:40,] # extract training data
    Xtest&lt;-X[41:72,] # extract test data
    Ytest&lt;-Y[41:72,] # extract test data
</code></pre>

<hr>
<h2 id='covplot'>scree-ploting cov(X, Y)</h2><span id='topic+covplot'></span>

<h3>Description</h3>

<p>Returns a scree-plot of the eigenvalues of cov(first.set, second.set) to select its first d largest eigenvectors.</p>


<h3>Usage</h3>

<pre><code class='language-R'>covplot(X, Y, mind=NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covplot_+3A_x">X</code></td>
<td>
<p>numeric matrix (n * p), X</p>
</td></tr>
<tr><td><code id="covplot_+3A_y">Y</code></td>
<td>
<p>numeric matrix (n * r), Y</p>
</td></tr>
<tr><td><code id="covplot_+3A_mind">mind</code></td>
<td>
<p>numeric, the number of the eigenvalues to show their cumulative percentages. The default is NULL, and then it is equal to min(p,r)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>eigenvalues</code></td>
<td>
<p>the ordiered eigenvalues of cov(X,Y) </p>
</td></tr>
<tr><td><code>cum.percent</code></td>
<td>
<p>the cumulative percentages of the eigenvalues</p>
</td></tr>
<tr><td><code>num.evecs</code></td>
<td>
<p>a vector of the numbers of the eigenvectors which forces the cumulative percentages bigger than 0.6, 0.7, 0.8, 0.9</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])
covplot(X, Y)
covplot(X, Y, mind=4)
</code></pre>

<hr>
<h2 id='finalCCA'>finalized CCA in seeded CCA</h2><span id='topic+finalCCA'></span>

<h3>Description</h3>

<p>Returns the results of the finalized step in seeded CCA.</p>


<h3>Usage</h3>

<pre><code class='language-R'>finalCCA(X, Y)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="finalCCA_+3A_x">X</code></td>
<td>
<p>numeric matrix (n * d), the initially-CCAed first set of variables</p>
</td></tr>
<tr><td><code id="finalCCA_+3A_y">Y</code></td>
<td>
<p>numeric matrix (n * d), the initially-CCAed second set of variables</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>cor</code></td>
<td>
<p>canonical correlations in finalized step</p>
</td></tr>
<tr><td><code>xcoef</code></td>
<td>
<p>the estimated canonical coefficient matrix of the initially-CCAed first set of variables</p>
</td></tr>
<tr><td><code>ycoef</code></td>
<td>
<p>the estimated canonical coefficient matrix of the initially-CCAed second set of variables</p>
</td></tr>
<tr><td><code>Xscores</code></td>
<td>
<p>the finalized canonical variates of the first set of variables</p>
</td></tr>
<tr><td><code>Yscores</code></td>
<td>
<p>the finalized canonical variates of the second set of variables</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>########  data(cookie) ########
data(cookie)
myseq &lt;- seq(141, 651, by=2)
X &lt;- as.matrix(cookie[-c(23,61), myseq])
Y &lt;- as.matrix(cookie[-c(23,61), 701:704])
min.pr &lt;- min( dim(X)[2], dim(Y)[2])
MX0 &lt;- iniCCA(X, Y, u=4, num.d=min.pr)
ini.X &lt;- X %*% MX0
finalCCA(ini.X, Y)

########  data(nutrimouse) ########
data(nutrimouse)
Y&lt;-as.matrix(nutrimouse$lipid)
X&lt;-as.matrix(nutrimouse$gene)
MX0 &lt;- iniCCA(X, Y, u=4, num.d=4)
MY0 &lt;- iniCCA(Y, X, u=5, num.d=4)
ini.X &lt;- X %*% MX0
ini.Y &lt;- Y %*% MY0
finalCCA(ini.X, ini.Y)
</code></pre>

<hr>
<h2 id='fitted.seedCCA'>Fitted values of ordinary and partial least squares</h2><span id='topic+fitted.seedCCA'></span>

<h3>Description</h3>

<p>Returns fitted values of ordinary and partial least squares through iterative projections. It works only for subclasses &quot;seedols&quot; and &quot;seedpls&quot;.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'seedCCA'
fitted(object, u=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitted.seedCCA_+3A_object">object</code></td>
<td>
<p>The name of an object of class &quot;seedCCA&quot;</p>
</td></tr>
<tr><td><code id="fitted.seedCCA_+3A_u">u</code></td>
<td>
<p>numeric, the number of projections. The default is NULL. This option is valid for PLS alone. The option returns the fitted values for u projections. For example, if it is specified at k, then the fitted values with k projections are returned.</p>
</td></tr>
<tr><td><code id="fitted.seedCCA_+3A_...">...</code></td>
<td>
<p>arguments passed to the fitted.method</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>########  data(cookie) ########
########  data(cookie) ########
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])

fit.ols1 &lt;- seedCCA(X[,1:4], Y[,1], type="cca")
fit.pls1 &lt;- seedCCA(X, Y[,1], type="pls")
fit.pls2 &lt;- seedCCA(X, Y[,1], type="pls", scale=TRUE)

fitted(fit.ols1)
fitted(fit.pls1)
fitted(fit.pls1, u=4)
fitted(fit.pls2, u=4)
</code></pre>

<hr>
<h2 id='iniCCA'>Initialized CCA in seeded CCA</h2><span id='topic+iniCCA'></span>

<h3>Description</h3>

<p>Returns the canonical coefficient matrices from the initialized step in seeded CCA. The initialzied CCA is done only for the first set in its first argument. The <code>num.d</code> must be less than or equal to the dimension of the second set.</p>


<h3>Usage</h3>

<pre><code class='language-R'>iniCCA(X, Y, u, num.d)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iniCCA_+3A_x">X</code></td>
<td>
<p>numeric matrix (n * p), the first set of variables: this set of variables alone is reduced.</p>
</td></tr>
<tr><td><code id="iniCCA_+3A_y">Y</code></td>
<td>
<p>numeric matrix (n * r), the second set of variables</p>
</td></tr>
<tr><td><code id="iniCCA_+3A_u">u</code></td>
<td>
<p>numeric, the terminiating index of the projection</p>
</td></tr>
<tr><td><code id="iniCCA_+3A_num.d">num.d</code></td>
<td>
<p>numeric, the first &quot;num.d&quot; eigenvectors of cov(X,Y) to replace cov(X,Y), if min(p,r) relatively bigger than n. The <code>num.d</code> must be less than or equal to min(p,r).</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>B</code></td>
<td>
<p>the initialized CCAed coefficient matrix projected by the value of <code>u</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>########  data(cookie) ########
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])
min.pr &lt;- min( dim(X)[2], dim(Y)[2])
MX0 &lt;- iniCCA(X, Y, u=4, num.d=min.pr)
ini.X &lt;- X%*%MX0

########  data(nutrimouse) ########
data(nutrimouse)
Y&lt;-as.matrix(nutrimouse$lipid)
X&lt;-as.matrix(nutrimouse$gene)
MX0 &lt;- iniCCA(X, Y, u=4, num.d=4)
MY0 &lt;- iniCCA(Y, X, u=5, num.d=4)
ini.X &lt;- X %*% MX0
ini.Y &lt;- Y %*% MY0
</code></pre>

<hr>
<h2 id='nutrimouse'>Nutrimouse dataset</h2><span id='topic+nutrimouse'></span>

<h3>Description</h3>

<p>The nutrimouse dataset comes from a nutrition study in the mouse. It was provided by Pascal Martin from the Toxicology and Pharmacology Laboratory (French National Institute for Agronomic Research). </p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nutrimouse)
</code></pre>


<h3>Format</h3>

<p>gene: data frame (40 * 120) with numerical variables
</p>
<p>lipid: data frame (40 * 21) with numerical variables
</p>
<p>diet: factor vector (40)
</p>
<p>genotype: factor vector (40)
</p>


<h3>Details</h3>

<p>Two sets of variables were measured on 40 mice:
</p>
<p>expressions of 120 genes potentially involved in nutritional problems.
</p>
<p>concentrations of 21 hepatic fatty acids:
The 40 mice were distributed in a 2-factors experimental design (4 replicates).
</p>
<p>Genotype (2-levels factor): wild-type and PPARalpha -/-
</p>
<p>Diet (5-levels factor): Oils used for experimental diets preparation were corn and colza oils (50/50) for a reference diet (REF), hydrogenated coconut oil for a saturated fatty acid diet (COC), sunflower oil for an Omega6 fatty acid-rich diet (SUN), linseed oil for an Omega3-rich diet (LIN) and corn/colza/enriched fish oils for the FISH diet (43/43/14).
</p>


<h3>References</h3>

<p>P. Martin, H. Guillou, F. Lasserre, S. D??jean, A. Lan, J-M. Pascussi, M. San Cristobal, P. Legrand, P. Besse, T. Pineau (2007)
Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. Hepatology, 45, 767???777
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nutrimouse)
boxplot(nutrimouse$lipid)
</code></pre>

<hr>
<h2 id='plot.seedCCA'>Plotting class &quot;seedCCA&quot; depending on the value of <code>type</code></h2><span id='topic+plot.seedCCA'></span>

<h3>Description</h3>

<p>The function is for plotting class &quot;seedCCA&quot;. Depending on subclass defined by the value of <code>type</code>, its resulting plot is different.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'seedCCA'
plot(x, ref=90, eps=0.01, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.seedCCA_+3A_x">x</code></td>
<td>
<p>The name of an object of class &quot;seedCCA&quot;</p>
</td></tr>
<tr><td><code id="plot.seedCCA_+3A_ref">ref</code></td>
<td>
<p>numeric, the value for reference line. It must be chosen between 0 and 100. It works only for subclass &quot;finalCCA&quot;.</p>
</td></tr>
<tr><td><code id="plot.seedCCA_+3A_eps">eps</code></td>
<td>
<p>numeric, a value to terminate projections. It must be chosen between 0 and 1. The default is equal to 0.01. It works only for subclass &quot;seedpls&quot;.</p>
</td></tr>
<tr><td><code id="plot.seedCCA_+3A_...">...</code></td>
<td>
<p>arguments passed to the plot.method</p>
</td></tr>
</table>


<h3>Details</h3>

<p>subclass &quot;finalCCA&quot;: the function makes a plot for percents of cumulative canonical correlations.
</p>
<p>subclass &quot;seedpls&quot;: the function returns a proper number of projections and plot of the projection increment against the number of projections. A proper number of projections is indicated with a blue-color vertical bar in the plot. Only for subclass &quot;seedpls&quot;, the output is retured. See Value part.
</p>
<p>subclass &quot;seedols&quot;: No plotting
</p>
<p>subclass &quot;selectu&quot;: the function makes a plot for increment of iterative projections by the output of subclass &quot;selectu&quot;.
</p>


<h3>Value</h3>

<table>
<tr><td><code>proper.u</code></td>
<td>
<p>proper value of the number of projections</p>
</td></tr>
<tr><td><code>nFu</code></td>
<td>
<p>incrments (n*Fu) of the iterative projection.</p>
</td></tr>
<tr><td><code>u</code></td>
<td>
<p>the maximum number of projections from &quot;seedpls&quot; object</p>
</td></tr>
<tr><td><code>eps</code></td>
<td>
<p>a value for terminating the projection.  The default value is equal to 0.01.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>########  data(cookie) ########
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])

fit.cca &lt;- seedCCA(X[,1:4],Y[,1:4],type="cca")
fit.seed1 &lt;- seedCCA(X,Y, type="seed1")
fit.pls1 &lt;- seedCCA(X,Y[,1],type="pls")
fit.selu &lt;- selectu(X,Y, type="seed2")

plot(fit.cca)
plot(fit.seed1, ref=95)
plot(fit.pls1)
plot(fit.pls1, eps=0.00001)
plot(fit.selu)
</code></pre>

<hr>
<h2 id='Pm'>Projection of a seed matrix on to the column subspace of M with respect to <code>Sx</code> inner-product</h2><span id='topic+Pm'></span>

<h3>Description</h3>

<p>The function reuturns a projection of a seed matrix on to the column subspace of M with respect to <code>Sx</code> inner-product.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pm(M, Sx, seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Pm_+3A_m">M</code></td>
<td>
<p>numeric matrix (p * k), a basis matrix of the column space of M</p>
</td></tr>
<tr><td><code id="Pm_+3A_sx">Sx</code></td>
<td>
<p>a inner-product matrix</p>
</td></tr>
<tr><td><code id="Pm_+3A_seed">seed</code></td>
<td>
<p>seed matrix</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])

## using cov(X,Y) as a seed matrix
seed &lt;- cov(X,Y)
col.num &lt;- dim(seed)[2]
M &lt;- iniCCA(X, Y, u=4, num.d=col.num)
Sx &lt;- cov(X)
Pm(M, Sx, seed)

## using the first 2 largest eigenvectors of cov(X,Y) as a seed matrix
seed2 &lt;- svd(cov(X,Y))$u[,1:2]
M2 &lt;- iniCCA(X, Y, u=4, num.d=2)
Pm(M, Sx, seed2)
</code></pre>

<hr>
<h2 id='print.seedCCA'>basic function for printing class &quot;seedCCA&quot;</h2><span id='topic+print.seedCCA'></span>

<h3>Description</h3>

<p>The function controls to print class &quot;seedCCA&quot;.
The function prints the estimated coefficents, if they exist.
For subclass &quot;finalCCA&quot;, canonical correlations are additionally reported.
For subsclass &quot;selectu&quot;, increments, suggested number of projections and the values of <code>type</code> and <code>eps</code> are reported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'seedCCA'
print(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.seedCCA_+3A_x">x</code></td>
<td>
<p>The name of an object of class &quot;seedCCA&quot;</p>
</td></tr>
<tr><td><code id="print.seedCCA_+3A_...">...</code></td>
<td>
<p>arguments passed to the print.method</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>########  data(cookie) ########
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])

fit.seed2 &lt;- seedCCA(X,Y)
fit.seed2
print(fit.seed2)
</code></pre>

<hr>
<h2 id='seedCCA'>Seeded Canonical correlation analysis</h2><span id='topic+seedCCA'></span>

<h3>Description</h3>

<p>The function <code>seedCCA</code> is mainly for implementing seeded canonical correlation analysis proposed by Im et al. (2015). The function conducts the following four methods, depending on the value of <code>type</code>. The option <code>type</code> has one of <code>c("cca", "seed1", "seed2", "pls")</code>. </p>


<h3>Usage</h3>

<pre><code class='language-R'>seedCCA(X,Y,type="seed2",ux=NULL,uy=NULL,u=10,eps=0.01,cut=0.9,d=NULL,AS=TRUE,scale=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seedCCA_+3A_x">X</code></td>
<td>
<p>numeric vector or matrix (n * p), the first set of variables</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_y">Y</code></td>
<td>
<p>numeric vector or matrix (n * r), the second set of variables</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_type">type</code></td>
<td>
<p>character, a choice of methods among <code>c("cca", "seed1", "seed2", "pls")</code>. The default is <code>"seed2"</code>.</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_ux">ux</code></td>
<td>
<p>numeric, maximum number of projections for X. The default is NULL. If this is not NULL, it surpasses the option <code>u</code> with <code>type="seed1"</code> and p&gt;r. For <code>type="seed2"</code>, if <code>ux</code> and <code>uy</code> are not NULL, they surpass <code>u</code>.</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_uy">uy</code></td>
<td>
<p>numeric, maximum number of projections for Y. The default is NULL. If this is not NULL, it surpasses the option <code>u</code> with <code>type="seed1"</code> and p&lt;r. For <code>type="seed2"</code>, if <code>ux</code> and <code>uy</code> are not NULL, they surpass <code>u</code>.</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_u">u</code></td>
<td>
<p>numeric, maximum number of projections. The default is 10. This is used for <code>type="seed1"</code>, <code>type="seed2"</code> and tyepe=&quot;pls&quot;.</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_eps">eps</code></td>
<td>
<p>numeric, the criteria to terminate iterative projections. The default is 0.01. If increment of projections is less than <code>eps</code>, then the iterative projection is terminated.</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_cut">cut</code></td>
<td>
<p>numeric, between 0 and 1. The default is 0.9.
If <code>d</code> is NULL, <code>cut</code> is used for automatic replacements of cov(X,Y) and cov(Y,X) with their eigenvectors, depending on the value of <code>cut</code>. So, if any value of <code>d</code> is given, <code>cut</code> is not effective. cov(X,Y) and cov(Y,X) are replaced with their largest eigenvectors, whose cumulative eigenvalue proportion is bigger than the value of <code>cut</code>. This only works for <code>type="seed2"</code>.</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_d">d</code></td>
<td>
<p>numeric, the user-selected number of largest eigenvectors of cov(X, Y) and cov(Y, X). The default is NULL. This only works for <code>type="seed2"</code>. If any value of <code>d</code> is given, <code>cut</code> does not work.</p>
</td></tr>
<tr><td><code id="seedCCA_+3A_as">AS</code></td>
<td>
<p>logical, status of automatic stop of projections. The default is <code>TRUE</code>. If TRUE, the iterative projection is automatically stopped, when the terminaion condition <code>eps</code> is satisfied. If<code>AS=FALSE</code>, the iterative projections are stopped at the value of <code>u</code>. </p>
</td></tr>
<tr><td><code id="seedCCA_+3A_scale">scale</code></td>
<td>
<p>logical. scaling predictors to have zero mean and one standard deviation. The default is <code>FALSE</code>. If <code>scale=TRUE</code>, each predictor is scaled with mean 0 and variance 1 for partial least squares. This option works only for <code>type="pls"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let p and r stand for the numbers of variables in the two sets and n stands for the sample size. The option of <code>type="cca"</code> can work only when max(p,r) &lt; n, and <code>seedCCA</code> conducts standard canonical correlation analysis (Johnson and Wichern, 2007). If <code>type="cca"</code> is given and either p or r is equal to one, ordinary least squares (OLS) is done instead of canonical correlation analysis. If max(p,r) &gt;= n, either <code>type="seed1"</code> or <code>type="seed2"</code> has to be chosen. This is the main purpose of <code>seedCCA</code>. If <code>type="seed1"</code>, only one set of variables, saying X with p for convenience, to have more variables than the other, saying Y with r, is initially reduced by the iterative projection approach (Cook et al. 2007). And then, the canonical correlation analysis of the initially-reduced X and the original Y is finalized. If <code>type="seed2"</code>, both X and Y are initially reduced. And then, the canonical correlation analysis of the two initially-reduced X and Y are finalzed. If <code>type="pls"</code>, partial least squares (PLS) is done. If <code>type="pls"</code> is given, the first set of variables in <code>seedCCA</code> is predictors and the second set is response. This matters The response can be multivariate. Depeding on the value of <code>type</code>, the resulted subclass by <code>seedCCA</code> are different.:
</p>
<p><code>type="cca"</code>: subclass &quot;finalCCA&quot; (p &gt;2; r &gt;2; p,r&lt;n)
</p>
<p><code>type="cca"</code>: subclass &quot;seedols&quot;  (either p or r is equal to 1.)
</p>
<p><code>type="seed1"</code> and <code>type="seed2"</code>: subclass &quot;finalCCA&quot; (max(p,r)&gt;n)
</p>
<p><code>type="pls"</code>: subclass &quot;seedpls&quot; (p&gt;n and r &lt;n)
</p>
<p>So, <code>plot(object)</code> will result in different figure depending on the object.
</p>
<p>The order of the values depending on type is follows.:
</p>
<p><code>type="cca"</code>: standard CCA (max(p,r)&lt;n, min(p,r)&gt;1) / &quot;finalCCA&quot; subclass
</p>
<p><code>type="cca"</code>: ordinary least squares (max(p,r)&lt;n, min(p,r)=1) / &quot;seedols&quot; subclass
</p>
<p><code>type="seed1"</code>: seeded CCA with case1 (max(p,r)&gt;n and p&gt;r) / &quot;finalCCA&quot; subclass
</p>
<p><code>type="seed1"</code>: seeded CCA with case1 (max(p,r)&gt;n and p&lt;=r) / &quot;finalCCA&quot; subclass
</p>
<p><code>type="seed2"</code>: seeded CCA with case2 (max(p,r)&gt;n) / &quot;finalCCA&quot; subclass
</p>
<p><code>type="pls"</code>: partial least squares (p&gt;n and r&lt;n) / &quot;seedpls&quot; subclass
</p>


<h3>Value</h3>

<table>
<tr><td><code>type="cca"</code></td>
<td>
<p>Values with selecting <code>type="cca"</code>: standard CCA (max(p,r)&lt;n, min(p,r)&gt;1) / &quot;finalCCA&quot; subclass</p>
</td></tr>
<tr><td><code>cor</code></td>
<td>
<p>canonical correlations</p>
</td></tr>
<tr><td><code>xcoef</code></td>
<td>
<p>the estimated canonical coefficients for X</p>
</td></tr>
<tr><td><code>ycoef</code></td>
<td>
<p>the estimated canonical coefficients for Y</p>
</td></tr>
<tr><td><code>Xscores</code></td>
<td>
<p>the estimated canonical variates for X</p>
</td></tr>
<tr><td><code>Yscores</code></td>
<td>
<p>the estimated canonical variates for Y</p>
</td></tr>
<tr><td><code>type="cca"</code></td>
<td>
<p>Values with selecting <code>type="cca"</code>: ordinary least squares (max(p,r)&lt;n, min(p,r)=1) / &quot;seedols&quot; subclass</p>
</td></tr>
<tr><td><code>coef</code></td>
<td>
<p>the estimated ordinary least squares coefficients</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>X, the first set</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Y, the second set</p>
</td></tr>
<tr><td><code>type="seed1"</code></td>
<td>
<p>Values with selecting <code>type="seed1"</code>: seeded CCA with case1 (max(p,r)&gt;n and p&gt;r) / &quot;finalCCA&quot; subclass</p>
</td></tr>
<tr><td><code>cor</code></td>
<td>
<p>canonical correlations</p>
</td></tr>
<tr><td><code>xcoef</code></td>
<td>
<p>the estimated canonical coefficients for X</p>
</td></tr>
<tr><td><code>ycoef</code></td>
<td>
<p>the estimated canonical coefficients for Y</p>
</td></tr>
<tr><td><code>proper.u</code></td>
<td>
<p>a suggested proper number of projections for X</p>
</td></tr>
<tr><td><code>initialMX0</code></td>
<td>
<p>the initialized canonical coefficient matrices of X</p>
</td></tr>
<tr><td><code>newX</code></td>
<td>
<p>initially-reduced X</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>the original Y</p>
</td></tr>
<tr><td><code>Xscores</code></td>
<td>
<p>the estimated canonical variates for X</p>
</td></tr>
<tr><td><code>Yscores</code></td>
<td>
<p>the estimated canonical variates for Y</p>
</td></tr>
<tr><td><code>type="seed1"</code></td>
<td>
<p>Values with selecting <code>type="seed1"</code>: seeded CCA with case1 (max(p,r)&gt;n and p&lt;=r) / &quot;finalCCA&quot; subclass)</p>
</td></tr>
<tr><td><code>cor</code></td>
<td>
<p>canonical correlations</p>
</td></tr>
<tr><td><code>xcoef</code></td>
<td>
<p>the estimated canonical coefficients for X</p>
</td></tr>
<tr><td><code>ycoef</code></td>
<td>
<p>the estimated canonical coefficients for Y</p>
</td></tr>
<tr><td><code>proper.u</code></td>
<td>
<p>a suggested proper number of projections for Y</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>the original X</p>
</td></tr>
<tr><td><code>initialMY0</code></td>
<td>
<p>the initialized canonical coefficient matrices of Y</p>
</td></tr>
<tr><td><code>newY</code></td>
<td>
<p>initially-reduced Y</p>
</td></tr>
<tr><td><code>Xscores</code></td>
<td>
<p>the estimated canonical variates for X</p>
</td></tr>
<tr><td><code>Yscores</code></td>
<td>
<p>the estimated canonical variates for Y</p>
</td></tr>
<tr><td><code>type="seed2"</code></td>
<td>
<p>Values with selecting <code>type="seed2"</code>: seeded CCA with case2 (max(p,r)&gt;n) / &quot;finalCCA&quot; subclass</p>
</td></tr>
<tr><td><code>cor</code></td>
<td>
<p>canonical correlations</p>
</td></tr>
<tr><td><code>xcoef</code></td>
<td>
<p>the estimated canonical coefficients for X</p>
</td></tr>
<tr><td><code>ycoef</code></td>
<td>
<p>the estimated canonical coefficients for Y</p>
</td></tr>
<tr><td><code>proper.ux</code></td>
<td>
<p>a suggested proper number of projections for X</p>
</td></tr>
<tr><td><code>proper.uy</code></td>
<td>
<p>a suggested proper number of projections for Y</p>
</td></tr>
<tr><td><code>d</code></td>
<td>
<p>suggested number of eigenvectors of cov(X,Y)</p>
</td></tr>
<tr><td><code>initialMX0</code></td>
<td>
<p>the initialized canonical coefficient matrices of X</p>
</td></tr>
<tr><td><code>initialMY0</code></td>
<td>
<p>the initialized canonical coefficient matrices of Y</p>
</td></tr>
<tr><td><code>newX</code></td>
<td>
<p>initially-reduced X</p>
</td></tr>
<tr><td><code>newY</code></td>
<td>
<p>initially-reduced Y</p>
</td></tr>
<tr><td><code>Xscores</code></td>
<td>
<p>the estimated canonical variates for X</p>
</td></tr>
<tr><td><code>Yscores</code></td>
<td>
<p>the estimated canonical variates for Y</p>
</td></tr>
<tr><td><code>type="pls"</code></td>
<td>
<p>Values with selecting <code>type="pls"</code>:: partial least squares (p&gt;n and r&lt;n) / &quot;seedpls&quot; subclass</p>
</td></tr>
<tr><td><code>coef</code></td>
<td>
<p>the estimated coefficients for each iterative projection upto u</p>
</td></tr>
<tr><td><code>u</code></td>
<td>
<p>the maximum number of projections</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>predictors</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>response</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>status of scaling predictors</p>
</td></tr>
<tr><td><code>cases</code></td>
<td>
<p>the number of observations</p>
</td></tr>
</table>


<h3>References</h3>

<p>R. D. Cook, B. Li and F. Chiaromonte. Dimension reduction in regression without matrix inversion. Biometrika 2007; 94: 569-584.
</p>
<p>Y. Im, H. Gang and JK. Yoo. High-throughput data dimension reduction via seeded canonical correlation analysis, J. Chemometrics 2015; 29: 193-199.
</p>
<p>R. A. Johnson and D. W. Wichern. Applied Multivariate Statistical Analysis. Pearson Prentice Hall: New Jersey, USA; 6 edition. 2007; 539-574.
</p>
<p>K. Lee and JK. Yoo. Canonical correlation analysis through linear modeling, AUST. NZ. J. STAT. 2014;  56: 59-72.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>######  data(cookie) ######
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])
dim(X);dim(Y)

## standard CCA
fit.cca &lt;-seedCCA(X[,1:4], Y, type="cca")  ## standard canonical correlation analysis is done.
plot(fit.cca)

## ordinary least squares
fit.ols1 &lt;-seedCCA(X[,1:4], Y[,1], type="cca")  ## ordinary least squares is done, because r=1.
fit.ols2 &lt;-seedCCA(Y[,1], X[,1:4], type="cca")  ## ordinary least squares is done, because p=1.

## seeded CCA with case 1
fit.seed1 &lt;- seedCCA(X, Y, type="seed1") ## suggested proper value of u is equal to 3.
fit.seed1.ux &lt;- seedCCA(X, Y, ux=6, type="seed1") ## iterative projections done 6 times.
fit.seed1.uy &lt;- seedCCA(Y, X, uy=6, type="seed1", AS=FALSE)  ## projections not done until uy=6.
plot(fit.seed1)

## partial least squares
fit.pls1 &lt;- seedCCA(X, Y[,1], type="pls")
fit.pls.m &lt;- seedCCA(X, Y, type="pls") ## multi-dimensional response
par(mfrow=c(1,2))
plot(fit.pls1); plot(fit.pls.m)


########  data(nutrimouse) ########
data(nutrimouse)
X&lt;-as.matrix(nutrimouse$gene)
Y&lt;-as.matrix(nutrimouse$lipid)
dim(X);dim(Y)

## seeded CCA with case 2
fit.seed2 &lt;- seedCCA(X, Y, type="seed2")  ## d not specified, so cut=0.9 (default) used.
fit.seed2.99 &lt;- seedCCA(X, Y, type="seed2", cut=0.99)  ## cut=0.99 used.
fit.seed2.d3 &lt;- seedCCA(X, Y, type="seed2", d=3)  ## d is specified with 3.

## ux and uy specified, so proper values not suggested.
fit.seed2.uxuy &lt;- seedCCA(X, Y, type="seed2", ux=10, uy=10)
plot(fit.seed2)
</code></pre>

<hr>
<h2 id='seeding'>increments of iterative projections</h2><span id='topic+seeding'></span>

<h3>Description</h3>

<p>Returns increments (nFu) of iterative projections of a seed matrix onto a covariance matrix <code>u</code> times.)</p>


<h3>Usage</h3>

<pre><code class='language-R'>seeding(seed, covx, n,  u=10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seeding_+3A_seed">seed</code></td>
<td>
<p>numeric matrix (p * d), a seed matrix</p>
</td></tr>
<tr><td><code id="seeding_+3A_covx">covx</code></td>
<td>
<p>numeric matrix (p * p), covariance matrix of X</p>
</td></tr>
<tr><td><code id="seeding_+3A_n">n</code></td>
<td>
<p>numeric, sample sizes</p>
</td></tr>
<tr><td><code id="seeding_+3A_u">u</code></td>
<td>
<p>numeric, maximum number of projections</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>nFu</code></td>
<td>
<p>n*Fu values</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])
seed &lt;- cov(X,Y)
covx &lt;- cov(X)
seeding(seed, covx, n=dim(X)[1], u=4)
</code></pre>

<hr>
<h2 id='seeding.auto.stop'>increments of iterative projections with automatic stopping</h2><span id='topic+seeding.auto.stop'></span>

<h3>Description</h3>

<p>Returns increments (nFu) of iterative projections of a seed matrix onto a covariance matrix upto k, which properly chosen by satisfying the terminating condition <code>eps</code> (<code>eps</code> can be selected by users).</p>


<h3>Usage</h3>

<pre><code class='language-R'>seeding.auto.stop(seed, covx, n, u.max=30, eps=0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seeding.auto.stop_+3A_seed">seed</code></td>
<td>
<p>numeric matrix (p * d), a seed matrix</p>
</td></tr>
<tr><td><code id="seeding.auto.stop_+3A_covx">covx</code></td>
<td>
<p>numeric matrix (p * p), covariance matrix of X</p>
</td></tr>
<tr><td><code id="seeding.auto.stop_+3A_n">n</code></td>
<td>
<p>numeric, sample sizes</p>
</td></tr>
<tr><td><code id="seeding.auto.stop_+3A_u.max">u.max</code></td>
<td>
<p>numeric, maximum number of projection.  The default value is equal to 30.</p>
</td></tr>
<tr><td><code id="seeding.auto.stop_+3A_eps">eps</code></td>
<td>
<p>numeric, a value of a condition for terminating the projection.  The default value is equal to 0.01.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>nFu</code></td>
<td>
<p>n*Fu values</p>
</td></tr>
<tr><td><code>u</code></td>
<td>
<p>the number of projection properly chosen by satisfying the terminating condition <code>eps</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])
seed &lt;- cov(X,Y)
covx &lt;- cov(X)
seeding.auto.stop(seed, covx, n=dim(X)[1])
seeding.auto.stop(seed, covx, n=dim(X)[1], u.max=20, eps=0.001)
</code></pre>

<hr>
<h2 id='seedols'>Ordinary least squares</h2><span id='topic+seedols'></span>

<h3>Description</h3>

<p>Returns ordinary least squares estimates. And, the function results in subclass &quot;seedols&quot;. For this function to work, either X or Y has to be one-dimensional. It is not necessary that X and Y should be predictors and response, respectively. Regardless of the position in the arguments, the one-dimensional and multi-dimensional variables become response and predictors, respectively.</p>


<h3>Usage</h3>

<pre><code class='language-R'>seedols(X, Y)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seedols_+3A_x">X</code></td>
<td>
<p>numeric vector or matrix, a first set of variables</p>
</td></tr>
<tr><td><code id="seedols_+3A_y">Y</code></td>
<td>
<p>numeric vector or matrix, a second set of variables</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>coef</code></td>
<td>
<p>the estimated coefficients for each iterative projection upto u</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>X, the first set</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Y, the second set</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>########  data(cookie) ########
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])

ols1 &lt;- seedols(X[,1:4],Y[,1])
ols2 &lt;- seedols(Y[,1],X[,1:4])
## ols1 and ols2 are the same results.
</code></pre>

<hr>
<h2 id='seedpls'>Partial least squares through iterative projections</h2><span id='topic+seedpls'></span>

<h3>Description</h3>

<p>Returns partial least squares estimates through iterative projections. And, the function results in subclass &quot;seedpls&quot;.</p>


<h3>Usage</h3>

<pre><code class='language-R'>seedpls(X, Y, u=5, scale=FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seedpls_+3A_x">X</code></td>
<td>
<p>numeric matrix (n * p), a set of predictors</p>
</td></tr>
<tr><td><code id="seedpls_+3A_y">Y</code></td>
<td>
<p>numeric vector or matrix (n * r), responses (it can be multi-dimensional)</p>
</td></tr>
<tr><td><code id="seedpls_+3A_u">u</code></td>
<td>
<p>numeric, the number of projections. The default is 5.</p>
</td></tr>
<tr><td><code id="seedpls_+3A_scale">scale</code></td>
<td>
<p>logical, FALSE is default. If TRUE, each predictor is standardized with mean 0 and variance 1</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>coef</code></td>
<td>
<p>the estimated coefficients for each iterative projection upto u</p>
</td></tr>
<tr><td><code>u</code></td>
<td>
<p>the maximum number of projections</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>Predictors</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Response</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>status of scaling predictors</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>########  data(cookie) ########
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])

fit.pls1 &lt;- seedpls(X,Y[,1]) ## one-dimensional response
fit.pls2 &lt;- seedpls(X,Y, u=6, scale=TRUE) ## four-dimensional response

</code></pre>

<hr>
<h2 id='selectu'>Function that guides a selection of the terminating index when using seedCCA function
</h2><span id='topic+selectu'></span>

<h3>Description</h3>

<p>The usage of <code>selectu</code> depends on one of its arguments, <code>type</code>. If <code>tyep="seed1"</code>, the n*F_u is computed for a higher dimension one of X and Y and a proper number of prjections is reported. For example, suppose that the dimension of X is higher than Y. Then <code>selectu(X,Y, type="case1")</code> and <code>selectu(Y, X, u=5, type="case1")</code> gives the same results, and it is for X. If <code>type="seed2"</code>, n*F_u is computed for X and Y and proper numbers of projections for X and Y are reported. And, For <code>type="seed2"</code>, <code>num.d</code> must be specified. Its defualt value is 2. The argument <code>eps</code> is a terminating condition for stopping projections. The projection is stopped, when the increment is less than the value of <code>eps</code>. The argument <code>auto.stop=TRUE</code> has the function automatically stopped as soon as the increment is less than the value of <code>eps</code>. If not, the increments are computed until the value of <code>u.max</code> is reached. The function <code>selectu</code> results in subclass &quot;selectu&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selectu(X, Y, type="seed2", u.max=30, auto.stop=TRUE, num.d=2, eps=0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selectu_+3A_x">X</code></td>
<td>
<p>numeric matrix (n * p), the first set of variables</p>
</td></tr>
<tr><td><code id="selectu_+3A_y">Y</code></td>
<td>
<p>numeric matrix (n * r), the second set of variables</p>
</td></tr>
<tr><td><code id="selectu_+3A_type">type</code></td>
<td>
<p>character, the default is &quot;seed2&quot;. &quot;seed1&quot; is for the first case of Seeded CCA (One set of variable is initially-reduced). &quot;seed2&quot; is for the second case of Seeded CCA (Two sets of variables are initially reduced).</p>
</td></tr>
<tr><td><code id="selectu_+3A_u.max">u.max</code></td>
<td>
<p>numeric, t he maximum number of u. The default is equal to 30.</p>
</td></tr>
<tr><td><code id="selectu_+3A_auto.stop">auto.stop</code></td>
<td>
<p>logical, The default value is TRUE. If TRUE, the iterative projection is automatically stopped, when the terminaion condition eps is satisfied. If FALSE, the iterative projections are stopped at the value of u.max.</p>
</td></tr>
<tr><td><code id="selectu_+3A_num.d">num.d</code></td>
<td>
<p>numeric, the number of the &quot;num.d&quot; largest eigenvectors of cov(first.set, second.set), if case1=FALSE. The default value is equal to 2. This option works only for type=&quot;seed2&quot;.</p>
</td></tr>
<tr><td><code id="selectu_+3A_eps">eps</code></td>
<td>
<p>numeric, the default value is equal to 0.01. A value for terminating the projection.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The order of the values depending on type is follows:
</p>
<p><code>type="seed1"</code>
</p>
<p><code>type="seed2"</code>
</p>


<h3>Value</h3>

<table>
<tr><td><code>type="seed1"</code></td>
<td>
<p>Values with selecting <code>type="seed1"</code>:</p>
</td></tr>
<tr><td><code>nFu</code></td>
<td>
<p>incrments (n*Fu) of the iterative projection for initally reduction one set of variable.</p>
</td></tr>
<tr><td><code>proper.u</code></td>
<td>
<p>proper value of the number of projections for X</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>types of seeded CCA</p>
</td></tr>
<tr><td><code>eps</code></td>
<td>
<p>a value for terminating the projection.  The default value is equal to 0.01.</p>
</td></tr>
<tr><td><code>type="seed2"</code></td>
<td>
<p>Values with selecting <code>type="seed2"</code>:</p>
</td></tr>
<tr><td><code>nFu.x</code></td>
<td>
<p>incrments (n*Fu) of the iterative projection for initially reducing X.</p>
</td></tr>
<tr><td><code>nFu.y</code></td>
<td>
<p>incrments (n*Fu) of the iterative projection for initially reducing Y.</p>
</td></tr>
<tr><td><code>proper.ux</code></td>
<td>
<p>proper value of the number of projections for X</p>
</td></tr>
<tr><td><code>proper.uy</code></td>
<td>
<p>proper value of the number of projections for Y</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>types of seeded CCA</p>
</td></tr>
<tr><td><code>eps</code></td>
<td>
<p>a value for terminating the projection.  The default value is equal to 0.01.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>######  data(cookie) ######
data(cookie)
myseq&lt;-seq(141,651,by=2)
X&lt;-as.matrix(cookie[-c(23,61),myseq])
Y&lt;-as.matrix(cookie[-c(23,61),701:704])

selectu(X, Y, type="seed1")
selectu(X, Y, type="seed1", auto.stop=FALSE)
selectu(X, Y, type="seed2", eps=0.001, num.d=3)
selectu(X, Y, type="seed2", auto.stop=FALSE)



########  data(nutrimouse) ########
data(nutrimouse)
Y&lt;-as.matrix(nutrimouse$lipid)
X&lt;-as.matrix(nutrimouse$gene)
selectu(X, Y, type="seed2", num.d=4)
selectu(X, Y, type="seed2", num.d=4, eps=0.001)
selectu(X, Y, type="seed2", auto.stop=FALSE, num.d=4, eps=0.001)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
