<!DOCTYPE html><html><head><title>Help for package seeker</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {seeker}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#checkDefaultCommands'><p>Check for presence of command-line interfaces</p></a></li>
<li><a href='#fastqc'><p>Run FastQC</p></a></li>
<li><a href='#fastqscreen'><p>Run FastQ Screen</p></a></li>
<li><a href='#fetch'><p>Fetch files</p></a></li>
<li><a href='#fetchMetadata'><p>Fetch metadata for a genomic study</p></a></li>
<li><a href='#getPlatforms'><p>Get supported microarray platforms</p></a></li>
<li><a href='#getSalmonMetadata'><p>Aggregrate metadata from salmon quantifications</p></a></li>
<li><a href='#getTx2gene'><p>Get mapping between transcripts and genes</p></a></li>
<li><a href='#installCustomCdfPackages'><p>Install custom CDF packages</p></a></li>
<li><a href='#installSysDeps'><p>Install seeker's system dependencies</p></a></li>
<li><a href='#multiqc'><p>Run MultiQC</p></a></li>
<li><a href='#salmon'><p>Run Salmon</p></a></li>
<li><a href='#seeker'><p>Process RNA-seq data end to end</p></a></li>
<li><a href='#seekerArray'><p>Process microarray data end to end</p></a></li>
<li><a href='#trimgalore'><p>Run Trim Galore!</p></a></li>
<li><a href='#tximport'><p>Run tximport on RNA-seq quantifications</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simplified Fetching and Processing of Microarray and RNA-Seq
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.5</td>
</tr>
<tr>
<td>Description:</td>
<td>Wrapper around various existing tools and command-line interfaces,
  providing a standard interface, simple parallelization, and detailed logging.
  For microarray data, maps probe sets to standard gene IDs, building on
  'GEOquery' Davis and Meltzer (2007) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtm254">doi:10.1093/bioinformatics/btm254</a>&gt;,
  'ArrayExpress' Kauffmann et al. (2009) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtp354">doi:10.1093/bioinformatics/btp354</a>&gt;,
  Robust multi-array average 'RMA' Irizarry et al. (2003) &lt;<a href="https://doi.org/10.1093%2Fbiostatistics%2F4.2.249">doi:10.1093/biostatistics/4.2.249</a>&gt;,
  and 'BrainArray' Dai et al. (2005) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgni179">doi:10.1093/nar/gni179</a>&gt;.
  For RNA-seq data, fetches metadata and raw reads from National Center for Biotechnology
  Information (NCBI) Sequence Read Archive (SRA), performs standard adapter and
  quality trimming using 'TrimGalore' Krueger <a href="https://github.com/FelixKrueger/TrimGalore">https://github.com/FelixKrueger/TrimGalore</a>,
  performs quality control checks using 'FastQC' Andrews <a href="https://github.com/s-andrews/FastQC">https://github.com/s-andrews/FastQC</a>,
  quantifies transcript abundances using 'salmon' Patro et al. (2017) &lt;<a href="https://doi.org/10.1038%2Fnmeth.4197">doi:10.1038/nmeth.4197</a>&gt; and potentially
  'refgenie' Stolarczyk et al. (2020) &lt;<a href="https://doi.org/10.1093%2Fgigascience%2Fgiz149">doi:10.1093/gigascience/giz149</a>&gt;,
  aggregates the results using 'MultiQC' Ewels et al. (2016) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw354">doi:10.1093/bioinformatics/btw354</a>&gt;,
  maps transcripts to genes using 'biomaRt' Durinkck et al. (2009) &lt;<a href="https://doi.org/10.1038%2Fnprot.2009.97">doi:10.1038/nprot.2009.97</a>&gt;,
  and summarizes transcript-level quantifications for gene-level analyses using
  'tximport' Soneson et al. (2015) &lt;<a href="https://doi.org/10.12688%2Ff1000research.7563.2">doi:10.12688/f1000research.7563.2</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://seeker.hugheylab.org">https://seeker.hugheylab.org</a>, <a href="https://github.com/hugheylab/seeker">https://github.com/hugheylab/seeker</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>affy (&ge; 1.68.0), AnnotationDbi (&ge; 1.52.0), BiocManager (&ge;
1.30.0), biomaRt (&ge; 2.36.1), checkmate (&ge; 2.1.0), curl (&ge;
3.2), data.table (&ge; 1.11.8), foreach (&ge; 1.4.4), GEOquery (&ge;
2.58.0), glue (&ge; 1.5.0), jsonlite (&ge; 1.7.2), methods, qs (&ge;
0.21.2), R.utils (&ge; 2.11.0), RCurl (&ge; 1.98), readr (&ge;
1.4.0), sessioninfo (&ge; 1.2.0), tximport (&ge; 1.8.0), withr (&ge;
2.4.2), yaml (&ge; 2.2.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ArrayExpress (&ge; 1.62.0), Biobase, doParallel (&ge; 1.0.17),
knitr, org.Mm.eg.db, rmarkdown, testthat (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-22 18:37:12 UTC; jakehughey</td>
</tr>
<tr>
<td>Author:</td>
<td>Jake Hughey [aut, cre],
  Josh Schoenbachler [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jake Hughey &lt;jakejhughey@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-22 18:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='checkDefaultCommands'>Check for presence of command-line interfaces</h2><span id='topic+checkDefaultCommands'></span>

<h3>Description</h3>

<p>This function checks whether the command-line tools used by seeker are
accessible in the expected places.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkDefaultCommands(keepIdx = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkDefaultCommands_+3A_keepidx">keepIdx</code></td>
<td>
<p>Logical indicating whether to keep the <code>idx</code> column of the
resulting data.table. For internal use only.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with columns for command, path, and version.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+installSysDeps">installSysDeps()</a></code>
</p>

<hr>
<h2 id='fastqc'>Run FastQC</h2><span id='topic+fastqc'></span>

<h3>Description</h3>

<p>This function calls
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a> using
<code><a href="base.html#topic+system2">system2()</a></code>. To run in parallel, register a parallel backend, e.g., using
<code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastqc(filepaths, outputDir = "fastqc_output", cmd = "fastqc", args = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fastqc_+3A_filepaths">filepaths</code></td>
<td>
<p>Paths to fastq files. For single-end reads, each element
should be a single filepath. For paired-end reads, each element can be two
filepaths separated by &quot;;&quot;.</p>
</td></tr>
<tr><td><code id="fastqc_+3A_outputdir">outputDir</code></td>
<td>
<p>Directory in which to store output. Will be created if it
doesn't exist.</p>
</td></tr>
<tr><td><code id="fastqc_+3A_cmd">cmd</code></td>
<td>
<p>Name or path of the command-line interface.</p>
</td></tr>
<tr><td><code id="fastqc_+3A_args">args</code></td>
<td>
<p>Additional arguments to pass to the command-line interface.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of exit codes, invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>
</p>

<hr>
<h2 id='fastqscreen'>Run FastQ Screen</h2><span id='topic+fastqscreen'></span>

<h3>Description</h3>

<p>This function calls
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/">fastq_screen</a>
using <code><a href="base.html#topic+system2">system2()</a></code>. To run in parallel, register a parallel backend, e.g.,
using <code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastqscreen(
  filepaths,
  outputDir = "fastqscreen_output",
  cmd = "fastq_screen",
  args = c("--threads", foreach::getDoParWorkers(), "--conf",
    "~/FastQ_Screen_Genomes/fastq_screen.conf")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fastqscreen_+3A_filepaths">filepaths</code></td>
<td>
<p>Paths to fastq files. For single-end reads, each element
should be a single filepath. For paired-end reads, each element can be two
filepaths separated by &quot;;&quot;.</p>
</td></tr>
<tr><td><code id="fastqscreen_+3A_outputdir">outputDir</code></td>
<td>
<p>Directory in which to store output. Will be created if it
doesn't exist.</p>
</td></tr>
<tr><td><code id="fastqscreen_+3A_cmd">cmd</code></td>
<td>
<p>Name or path of the command-line interface.</p>
</td></tr>
<tr><td><code id="fastqscreen_+3A_args">args</code></td>
<td>
<p>Additional arguments to pass to the command-line interface.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of exit codes, invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>
</p>

<hr>
<h2 id='fetch'>Fetch files</h2><span id='topic+fetch'></span>

<h3>Description</h3>

<p>This function uses the NCBI SRA Toolkit via <code><a href="base.html#topic+system2">system2()</a></code> to download files
from SRA and convert them to fastq.gz. To process files in parallel, register
a parallel backend, e.g., using <code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>. Beware
that intermediate files created by fasterq-dump are uncompressed and could
require hundreds of gigabytes if files are processed in parallel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetch(
  accessions,
  outputDir,
  overwrite = FALSE,
  keepSra = FALSE,
  prefetchCmd = "prefetch",
  prefetchArgs = NULL,
  fasterqdumpCmd = "fasterq-dump",
  fasterqdumpArgs = NULL,
  pigzCmd = "pigz",
  pigzArgs = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fetch_+3A_accessions">accessions</code></td>
<td>
<p>Character vector of SRA run accessions.</p>
</td></tr>
<tr><td><code id="fetch_+3A_outputdir">outputDir</code></td>
<td>
<p>String indicating the local directory in which to save the
files. Will be created if it doesn't exist.</p>
</td></tr>
<tr><td><code id="fetch_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical indicating whether to overwrite files that already
exist in <code>outputDir</code>.</p>
</td></tr>
<tr><td><code id="fetch_+3A_keepsra">keepSra</code></td>
<td>
<p>Logical indicating whether to keep the &quot;.sra&quot; files.</p>
</td></tr>
<tr><td><code id="fetch_+3A_prefetchcmd">prefetchCmd</code></td>
<td>
<p>String indicating command for prefetch, which downloads
&quot;.sra&quot; files.</p>
</td></tr>
<tr><td><code id="fetch_+3A_prefetchargs">prefetchArgs</code></td>
<td>
<p>Character vector indicating arguments to pass to
prefetch.</p>
</td></tr>
<tr><td><code id="fetch_+3A_fasterqdumpcmd">fasterqdumpCmd</code></td>
<td>
<p>String indicating command for fasterq-dump, which
uses &quot;.sra&quot; files to create &quot;.fastq&quot; files.</p>
</td></tr>
<tr><td><code id="fetch_+3A_fasterqdumpargs">fasterqdumpArgs</code></td>
<td>
<p>Character vector indicating arguments to pass to
fasterq-dump.</p>
</td></tr>
<tr><td><code id="fetch_+3A_pigzcmd">pigzCmd</code></td>
<td>
<p>String indicating command for pigz, which converts &quot;.fastq&quot;
files to &quot;.fastq.gz&quot; files.</p>
</td></tr>
<tr><td><code id="fetch_+3A_pigzargs">pigzArgs</code></td>
<td>
<p>Character vector indicating arguments to pass to pigz.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list. As the function runs, it updates a tab-delimited log file in
<code>outputDir</code> called &quot;progress.tsv&quot;.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>, <code><a href="#topic+fetchMetadata">fetchMetadata()</a></code>
</p>

<hr>
<h2 id='fetchMetadata'>Fetch metadata for a genomic study</h2><span id='topic+fetchMetadata'></span>

<h3>Description</h3>

<p>This function can use the API of the European Nucleotide Archive
(recommended) or the Sequence Read Archive.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetchMetadata(
  bioproject,
  host = c("ena", "sra"),
  fields = c("study_accession", "sample_accession", "secondary_sample_accession",
    "sample_alias", "sample_title", "experiment_accession", "run_accession", "fastq_md5",
    "fastq_ftp", "fastq_aspera"),
  file = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fetchMetadata_+3A_bioproject">bioproject</code></td>
<td>
<p>String indicating bioproject accession.</p>
</td></tr>
<tr><td><code id="fetchMetadata_+3A_host">host</code></td>
<td>
<p>String indicating from where to fetch the metadata.</p>
</td></tr>
<tr><td><code id="fetchMetadata_+3A_fields">fields</code></td>
<td>
<p>Character vector indicating which fields to fetch, if <code>host</code>
is &quot;ena&quot;.</p>
</td></tr>
<tr><td><code id="fetchMetadata_+3A_file">file</code></td>
<td>
<p>String indicating output file path, if not <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>, <code><a href="#topic+fetch">fetch()</a></code>
</p>

<hr>
<h2 id='getPlatforms'>Get supported microarray platforms</h2><span id='topic+getPlatforms'></span>

<h3>Description</h3>

<p>Get supported microarray platforms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPlatforms(type = c("cdf", "mapping"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPlatforms_+3A_type">type</code></td>
<td>
<p>String indicating whether to get supported platforms for
processing raw Affymetrix data using custom CDF or for mapping already
processed data from probes to genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code>.
</p>

<hr>
<h2 id='getSalmonMetadata'>Aggregrate metadata from salmon quantifications</h2><span id='topic+getSalmonMetadata'></span>

<h3>Description</h3>

<p>Aggregrate metadata from salmon quantifications
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSalmonMetadata(inputDir, outputDir = "data")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSalmonMetadata_+3A_inputdir">inputDir</code></td>
<td>
<p>Directory that contains output from salmon.</p>
</td></tr>
<tr><td><code id="getSalmonMetadata_+3A_outputdir">outputDir</code></td>
<td>
<p>Directory in which to save the result, a file named
&quot;salmon_meta_info.csv&quot;. If <code>NULL</code>, no file is saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table, invisibly.
</p>
<p>#' @seealso <code><a href="#topic+seeker">seeker()</a></code>, <code><a href="#topic+salmon">salmon()</a></code>
</p>

<hr>
<h2 id='getTx2gene'>Get mapping between transcripts and genes</h2><span id='topic+getTx2gene'></span>

<h3>Description</h3>

<p>This function uses the
<a href="https://doi.org/doi:10.18129/B9.bioc.biomaRt">biomaRt package</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTx2gene(
  organism = "mmusculus",
  version = NULL,
  outputDir = "data",
  checkArgsOnly = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTx2gene_+3A_organism">organism</code></td>
<td>
<p>String used to pass <code>paste0(organism, "_gene_ensembl")</code> as the
<code>dataset</code> argument to <code><a href="biomaRt.html#topic+useEnsembl">biomaRt::useEnsembl()</a></code>. To see available datasets,
do <code style="white-space: pre;">&#8288;mart = biomaRt::useEnsembl("genes"); biomaRt::listDatasets(mart)&#8288;</code>.</p>
</td></tr>
<tr><td><code id="getTx2gene_+3A_version">version</code></td>
<td>
<p>Passed to <code><a href="biomaRt.html#topic+useEnsembl">biomaRt::useEnsembl()</a></code>. <code>NULL</code> indicates the latest
version. To see available versions, do <code>biomaRt::listEnsemblArchives()</code>.</p>
</td></tr>
<tr><td><code id="getTx2gene_+3A_outputdir">outputDir</code></td>
<td>
<p>Directory in which to save the result, a file named
&quot;tx2gene.csv.gz&quot;. If <code>NULL</code>, no file is saved.</p>
</td></tr>
<tr><td><code id="getTx2gene_+3A_checkargsonly">checkArgsOnly</code></td>
<td>
<p>Logical indicating whether to only check function
arguments. Used for testing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>checkArgsOnly</code> is <code>FALSE</code>, a data.table based on the result from
<code><a href="biomaRt.html#topic+getBM">biomaRt::getBM()</a></code>, with an attribute &quot;version&quot;. Otherwise <code>0</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>, <code><a href="#topic+tximport">tximport()</a></code>
</p>

<hr>
<h2 id='installCustomCdfPackages'>Install custom CDF packages</h2><span id='topic+installCustomCdfPackages'></span>

<h3>Description</h3>

<p>Install Brainarray custom CDFs for processing raw Affymetrix data. See
<a href="http://brainarray.mbni.med.umich.edu/Brainarray/Database/CustomCDF/CDF_download.asp">http://brainarray.mbni.med.umich.edu/Brainarray/Database/CustomCDF/CDF_download.asp</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>installCustomCdfPackages(pkgs, ver = 25, dryRun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="installCustomCdfPackages_+3A_pkgs">pkgs</code></td>
<td>
<p>Character vector of package names, e.g., &quot;hgu133ahsentrezgcdf&quot;.</p>
</td></tr>
<tr><td><code id="installCustomCdfPackages_+3A_ver">ver</code></td>
<td>
<p>Integer version number (25 as of 5 Jan 2021).</p>
</td></tr>
<tr><td><code id="installCustomCdfPackages_+3A_dryrun">dryRun</code></td>
<td>
<p>Logical indicating whether to actually install the packages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of URLs, invisibly.
</p>

<hr>
<h2 id='installSysDeps'>Install seeker's system dependencies</h2><span id='topic+installSysDeps'></span>

<h3>Description</h3>

<p>This function installs and configures the various programs required for
seeker to fetch and process RNA-seq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>installSysDeps(
  sraToolkitDir,
  minicondaDir,
  refgenieDir,
  rprofileDir,
  minicondaEnv = "seeker",
  refgenieGenomes = NULL,
  fastqscreenDir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="installSysDeps_+3A_sratoolkitdir">sraToolkitDir</code></td>
<td>
<p>String indicating directory in which to install the
<a href="https://github.com/ncbi/sra-tools">SRA Toolkit</a>. Recommended to use &quot;~&quot;,
the home directory. If <code>NULL</code>, the Toolkit will not be installed.</p>
</td></tr>
<tr><td><code id="installSysDeps_+3A_minicondadir">minicondaDir</code></td>
<td>
<p>String indicating directory in which to install
<a href="https://docs.conda.io/en/latest/miniconda.html">Miniconda</a>. Recommended
to use &quot;~&quot;, the home directory. If <code>NULL</code>, Miniconda will not be installed.</p>
</td></tr>
<tr><td><code id="installSysDeps_+3A_refgeniedir">refgenieDir</code></td>
<td>
<p>String indicating directory in which to store the
directory of genome assets from refgenie, which will be named
&quot;refgenie_genomes&quot;. Recommended to use &quot;~&quot;, the home directory. Only used
if <code>minicondaDir</code> is not <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="installSysDeps_+3A_rprofiledir">rprofileDir</code></td>
<td>
<p>String indicating directory in which to create or modify
.Rprofile, which is run by R on startup. Common options are &quot;~&quot; or &quot;.&quot;.</p>
</td></tr>
<tr><td><code id="installSysDeps_+3A_minicondaenv">minicondaEnv</code></td>
<td>
<p>String indicating name of the Miniconda environment in
which to install various conda packages (fastq-screen, fastqc, multiqc,
pigz, refgenie, salmon, and trim-galore).</p>
</td></tr>
<tr><td><code id="installSysDeps_+3A_refgeniegenomes">refgenieGenomes</code></td>
<td>
<p>Character vector indicating genome assets, such as
transcriptome indexes for <code><a href="#topic+salmon">salmon()</a></code>, to pull from
<a href="http://refgenomes.databio.org/index">refgenomes</a> using refgenie. If
<code>NULL</code>, no assets are fetched.</p>
</td></tr>
<tr><td><code id="installSysDeps_+3A_fastqscreendir">fastqscreenDir</code></td>
<td>
<p>String indicating directory in which to download the
genomes for <code><a href="#topic+fastqscreen">fastqscreen()</a></code>. This takes a long time. If <code>NULL</code>, genomes are
not downloaded.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>NULL</code>, invisibly
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>
</p>

<hr>
<h2 id='multiqc'>Run MultiQC</h2><span id='topic+multiqc'></span>

<h3>Description</h3>

<p>This function calls <a href="https://multiqc.info/">multiqc</a> using <code><a href="base.html#topic+system2">system2()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiqc(
  parentDir = ".",
  outputDir = "multiqc_output",
  cmd = "multiqc",
  args = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiqc_+3A_parentdir">parentDir</code></td>
<td>
<p>Directory that contains output to be aggregated.</p>
</td></tr>
<tr><td><code id="multiqc_+3A_outputdir">outputDir</code></td>
<td>
<p>Directory in which to store output. Will be created if it
doesn't exist.</p>
</td></tr>
<tr><td><code id="multiqc_+3A_cmd">cmd</code></td>
<td>
<p>Name or path of the command-line interface.</p>
</td></tr>
<tr><td><code id="multiqc_+3A_args">args</code></td>
<td>
<p>Additional arguments to pass to the command-line interface.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An exit code, invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>
</p>

<hr>
<h2 id='salmon'>Run Salmon</h2><span id='topic+salmon'></span>

<h3>Description</h3>

<p>This function calls
<a href="https://combine-lab.github.io/salmon/">salmon</a>
using <code><a href="base.html#topic+system2">system2()</a></code>. To run in parallel, register a parallel backend, e.g.,
using <code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>salmon(
  filepaths,
  samples,
  indexDir,
  outputDir = "salmon_output",
  cmd = "salmon",
  args = c("-l A -q --seqBias --gcBias --no-version-check -p",
    foreach::getDoParWorkers()),
  compress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="salmon_+3A_filepaths">filepaths</code></td>
<td>
<p>Paths to fastq files. For single-end reads, each element
should be a single filepath. For paired-end reads, each element should be
two filepaths separated by &quot;;&quot;.</p>
</td></tr>
<tr><td><code id="salmon_+3A_samples">samples</code></td>
<td>
<p>Corresponding sample names for fastq files.</p>
</td></tr>
<tr><td><code id="salmon_+3A_indexdir">indexDir</code></td>
<td>
<p>Directory that contains salmon index.</p>
</td></tr>
<tr><td><code id="salmon_+3A_outputdir">outputDir</code></td>
<td>
<p>Directory in which to store output. Will be created if it
doesn't exist.</p>
</td></tr>
<tr><td><code id="salmon_+3A_cmd">cmd</code></td>
<td>
<p>Name or path of the command-line interface.</p>
</td></tr>
<tr><td><code id="salmon_+3A_args">args</code></td>
<td>
<p>Additional arguments to pass to the command-line interface.</p>
</td></tr>
<tr><td><code id="salmon_+3A_compress">compress</code></td>
<td>
<p>Logical indicating whether to gzip the quantification file
(quant.sf) from salmon. Does not affect downstream analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of exit codes, invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>, <code><a href="#topic+getSalmonMetadata">getSalmonMetadata()</a></code>
</p>

<hr>
<h2 id='seeker'>Process RNA-seq data end to end</h2><span id='topic+seeker'></span>

<h3>Description</h3>

<p>This function selectively performs various steps to process RNA-seq data.
See also the vignettes: <code>browseVignettes('seeker')</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seeker(params, parentDir = ".", dryRun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seeker_+3A_params">params</code></td>
<td>
<p>Named list of parameters with components:
</p>

<ul>
<li> <p><code>study</code>: String used to name the output directory within <code>parentDir</code>.
</p>
</li>
<li> <p><code>metadata</code>: Named list with components:
</p>

<ul>
<li> <p><code>run</code>: Logical indicating whether to fetch metadata. See
<code><a href="#topic+fetchMetadata">fetchMetadata()</a></code>. If <code>TRUE</code>, saves a file
<code>parentDir</code>/<code>study</code>/metadata.csv. If <code>FALSE</code>, expects that file to
already exist. The unmodified fetched or found metadata is saved to a
file <code>parentDir</code>/<code>study</code>/metadata_original.csv. Following components
are only checked if <code>run</code> is <code>TRUE</code>.
</p>
</li>
<li> <p><code>bioproject</code>: String indicating the study's bioproject accession.
</p>
</li>
<li> <p><code>include</code>: Optional named list for specifying which rows of metadata to
include for further processing, with components:
</p>

<ul>
<li> <p><code>colname</code>: String indicating column in metadata
</p>
</li>
<li> <p><code>values</code>: Vector indicating values within <code>colname</code>
</p>
</li></ul>

</li>
<li> <p><code>exclude</code>: Optional named list for specifying which rows of metadata to
exclude from further processing (superseding <code>include</code>), with components:
</p>

<ul>
<li> <p><code>colname</code>: String indicating column in metadata
</p>
</li>
<li> <p><code>values</code>: Vector indicating values within <code>colname</code>
</p>
</li></ul>

</li></ul>

</li>
<li> <p><code>fetch</code>: Named list with components:
</p>

<ul>
<li> <p><code>run</code>: Logical indicating whether to fetch files from SRA. See <code><a href="#topic+fetch">fetch()</a></code>.
If <code>TRUE</code>, saves files to <code>parentDir</code>/<code>study</code>/fetch_output. Whether
<code>TRUE</code> or <code>FALSE</code>, expects metadata to have a column &quot;run_accession&quot;, and
updates metadata with column &quot;fastq_fetched&quot; containing paths to files in
<code>parentDir</code>/<code>study</code>/fetch_output. Following components are only checked
if <code>run</code> is <code>TRUE</code>.
</p>
</li>
<li> <p><code>keep</code>: Logical indicating whether to keep fastq.gz files when all
processing steps have completed. <code>NULL</code> indicates <code>TRUE</code>.
</p>
</li>
<li> <p><code>overwrite</code>: Logical indicating whether to overwrite files that already
exist. <code>NULL</code> indicates to use the default in <code><a href="#topic+fetch">fetch()</a></code>.
</p>
</li>
<li> <p><code>keepSra</code>: Logical indicating whether to keep the &quot;.sra&quot; files. <code>NULL</code>
indicates to use the default in <code><a href="#topic+fetch">fetch()</a></code>.
</p>
</li>
<li> <p><code>prefetchCmd</code>: String indicating command for prefetch, which downloads
&quot;.sra&quot; files. <code>NULL</code> indicates to use the default in <code><a href="#topic+fetch">fetch()</a></code>.
</p>
</li>
<li> <p><code>prefetchArgs</code>: Character vector indicating arguments to pass to
prefetch. <code>NULL</code> indicates to use the default in <code><a href="#topic+fetch">fetch()</a></code>.
</p>
</li>
<li> <p><code>fasterqdumpCmd</code>: String indicating command for fasterq-dump, which
uses &quot;.sra&quot; files to create &quot;.fastq&quot; files. <code>NULL</code> indicates to use the
default in <code><a href="#topic+fetch">fetch()</a></code>.
</p>
</li>
<li> <p><code>prefetchArgs</code>: Character vector indicating arguments to pass to
fasterq-dump. <code>NULL</code> indicates to use the default in <code><a href="#topic+fetch">fetch()</a></code>.
</p>
</li>
<li> <p><code>pigzCmd</code>: String indicating command for pigz, which converts &quot;.fastq&quot;
files to &quot;.fastq.gz&quot; files. <code>NULL</code> indicates to use the default in
<code><a href="#topic+fetch">fetch()</a></code>.
</p>
</li>
<li> <p><code>pigzArgs</code>: Character vector indicating arguments to pass to pigz. <code>NULL</code>
indicates to use the default in <code><a href="#topic+fetch">fetch()</a></code>.
</p>
</li></ul>

</li>
<li> <p><code>trimgalore</code>: Named list with components:
</p>

<ul>
<li> <p><code>run</code>: Logical indicating whether to perform quality/adapter trimming of
reads. See <code><a href="#topic+trimgalore">trimgalore()</a></code>. If <code>TRUE</code>, expects metadata to have a column
&quot;fastq_fetched&quot; containing paths to fastq files in
<code>parentDir</code>/<code>study</code>/fetch_output, saves trimmed files to
<code>parentDir</code>/<code>study</code>/trimgalore_output, and updates metadata with column
&quot;fastq_trimmed&quot;. If <code>FALSE</code>, expects and does nothing. Following
components are only checked if <code>run</code> is <code>TRUE</code>.
</p>
</li>
<li> <p><code>keep</code>: Logical indicating whether to keep trimmed fastq files when all
processing steps have completed. <code>NULL</code> indicates <code>TRUE</code>.
</p>
</li>
<li> <p><code>cmd</code>: Name or path of the command-line interface. <code>NULL</code> indicates to
use the default in <code><a href="#topic+trimgalore">trimgalore()</a></code>.
</p>
</li>
<li> <p><code>args</code>: Additional arguments to pass to the command-line interface.
<code>NULL</code> indicates to use the default in <code><a href="#topic+trimgalore">trimgalore()</a></code>.
</p>
</li>
<li> <p><code>pigzCmd</code>: String indicating command for pigz, which converts &quot;.fastq&quot;
files to &quot;.fastq.gz&quot; files. <code>NULL</code> indicates to use the default in
<code><a href="#topic+trimgalore">trimgalore()</a></code>.
</p>
</li></ul>

</li>
<li> <p><code>fastqc</code>: Named list with components:
</p>

<ul>
<li> <p><code>run</code>: Logical indicating whether to perform QC on reads. See <code><a href="#topic+fastqc">fastqc()</a></code>.
If <code>TRUE</code> and <code>trimgalore$run</code> is <code>TRUE</code>, expects metadata to have a
column &quot;fastq_trimmed&quot; containing paths to fastq files in
<code>parentDir</code>/<code>study</code>/trimgalore_output. If <code>TRUE</code> and <code>trimgalore$run</code> is
<code>FALSE</code>, expects metadata to have a column &quot;fastq_fetched&quot; containing
paths to fastq files in <code>parentDir</code>/<code>study</code>/fetch_output. If <code>TRUE</code>,
saves results to <code>parentDir</code>/<code>study</code>/fastqc_output. If <code>FALSE</code>, expects
and does nothing. Following components are only checked if <code>run</code> is
<code>TRUE</code>.
</p>
</li>
<li> <p><code>keep</code>: Logical indicating whether to keep fastqc files when all
processing steps have completed. <code>NULL</code> indicates <code>TRUE</code>.
</p>
</li>
<li> <p><code>cmd</code>: Name or path of the command-line interface. <code>NULL</code> indicates to
use the default in <code><a href="#topic+fastqc">fastqc()</a></code>.
</p>
</li>
<li> <p><code>args</code>: Additional arguments to pass to the command-line interface.
<code>NULL</code> indicates to use the default in <code><a href="#topic+fastqc">fastqc()</a></code>.
</p>
</li></ul>

</li>
<li> <p><code>salmon</code>: Named list with components:
</p>

<ul>
<li> <p><code>run</code>: Logical indicating whether to quantify transcript abundances. See
<code><a href="#topic+salmon">salmon()</a></code>. If <code>TRUE</code> and <code>trimgalore$run</code> is <code>TRUE</code>, expects metadata to
have a column &quot;fastq_trimmed&quot; containing paths to fastq files in
<code>parentDir</code>/<code>study</code>/trimgalore_output. If <code>TRUE</code> and <code>trimgalore$run</code> is
<code>FALSE</code>, expects metadata to have a column &quot;fastq_fetched&quot; containing
paths to fastq files in <code>parentDir</code>/<code>study</code>/fetch_output. If <code>TRUE</code>,
saves results to <code>parentDir</code>/<code>study</code>/salmon_output and
<code>parentDir</code>/<code>study</code>/salmon_meta_info.csv. If <code>FALSE</code>, expects and does
nothing. Following components are only checked if <code>run</code> is <code>TRUE</code>.
</p>
</li>
<li> <p><code>indexDir</code>: Directory that contains salmon index.
</p>
</li>
<li> <p><code>sampleColname</code>: String indicating column in metadata containing sample
ids. <code>NULL</code> indicates &quot;sample_accession&quot;, which should work for data
from SRA and ENA.
</p>
</li>
<li> <p><code>keep</code>: Logical indicating whether to keep quantification results when
all processing steps have completed. <code>NULL</code> indicates <code>TRUE</code>.
</p>
</li>
<li> <p><code>cmd</code>: Name or path of the command-line interface. <code>NULL</code> indicates to
use the default in <code><a href="#topic+salmon">salmon()</a></code>.
</p>
</li>
<li> <p><code>args</code>: Additional arguments to pass to the command-line interface.
<code>NULL</code> indicates to use the default in <code><a href="#topic+salmon">salmon()</a></code>.
</p>
</li></ul>

</li>
<li> <p><code>multiqc</code>: Named list with components:
</p>

<ul>
<li> <p><code>run</code>: Logical indicating whether to aggregrate results of various
processing steps. See <code><a href="#topic+multiqc">multiqc()</a></code>. If <code>TRUE</code>, saves results to
<code>parentDir</code>/<code>study</code>/multiqc_output. If <code>FALSE</code>, expects and does nothing.
Following components are only checked if <code>run</code> is <code>TRUE</code>.
</p>
</li>
<li> <p><code>cmd</code>: Name or path of the command-line interface. <code>NULL</code> indicates to
use the default in <code><a href="#topic+multiqc">multiqc()</a></code>.
</p>
</li>
<li> <p><code>args</code>: Additional arguments to pass to the command-line interface.
<code>NULL</code> indicates to use the default in <code><a href="#topic+multiqc">multiqc()</a></code>.
</p>
</li></ul>

</li>
<li> <p><code>tximport</code>: Named list with components:
</p>

<ul>
<li> <p><code>run</code>: Logical indicating whether to summarize transcript- or gene-level
estimates for downstream analysis. See <code><a href="#topic+tximport">tximport()</a></code>. If <code>TRUE</code>, expects
metadata to have a column <code>sampleColname</code> of sample ids, and expects a
directory <code>parentDir</code>/<code>study</code>/salmon_output containing directories of
quantification results, and saves results to
<code>parentDir</code>/<code>study</code>/tximport_output.qs. If <code>FALSE</code>, expects and does
nothing. Following components are only checked if <code>run</code> is <code>TRUE</code>.
</p>
</li>
<li> <p><code>tx2gene</code>: Optional named list with components:
</p>

<ul>
<li> <p><code>organism</code>: String indicating organism and thereby ensembl gene dataset.
See <code><a href="#topic+getTx2gene">getTx2gene()</a></code>.
</p>
</li>
<li> <p><code>version</code>: Optional number indicating ensembl version. <code>NULL</code> indicates
the latest version. See <code><a href="#topic+getTx2gene">getTx2gene()</a></code>.
</p>
</li>
<li> <p><code>filename</code>: Optional string indicating name of pre-existing text file
in <code>parentDir</code>/<code>params$study</code> containing mapping between transcripts
(first column) and genes (second column), with column names in the
first row. If <code>filename</code> is specified, <code>organism</code> and <code>version</code> must not
be specified.
</p>
</li></ul>

<p>If not <code>NULL</code>, saves a file <code>parentDir</code>/<code>study</code>/tx2gene.csv.gz.
</p>
</li>
<li> <p><code>countsFromAbundance</code>: String indicating whether or how to estimate
counts using estimated abundances. See <code><a href="tximport.html#topic+tximport">tximport::tximport()</a></code>.
</p>
</li>
<li> <p><code>ignoreTxVersion</code>: Logical indicating whether to the version suffix on
transcript ids. <code>NULL</code> indicates to use <code>TRUE</code>. See
<code><a href="tximport.html#topic+tximport">tximport::tximport()</a></code>.
</p>
</li></ul>

</li></ul>

<p><code>params</code> can be derived from a yaml file, see
<code>vignette("introduction", package = "seeker")</code>. The yaml representation
of <code>params</code> will be saved to <code>parentDir</code>/<code>params$study</code>/params.yml.</p>
</td></tr>
<tr><td><code id="seeker_+3A_parentdir">parentDir</code></td>
<td>
<p>Directory in which to store the output, which will be a
directory named according to <code>params$study</code>.</p>
</td></tr>
<tr><td><code id="seeker_+3A_dryrun">dryRun</code></td>
<td>
<p>Logical indicating whether to check the validity of inputs
without actually fetching or processing any data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path to the output directory <code>parentDir</code>/<code>params$study</code>, invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fetchMetadata">fetchMetadata()</a></code>, <code><a href="#topic+fetch">fetch()</a></code>, <code><a href="#topic+trimgalore">trimgalore()</a></code>, <code><a href="#topic+fastqc">fastqc()</a></code>,
<code><a href="#topic+salmon">salmon()</a></code>, <code><a href="#topic+multiqc">multiqc()</a></code>, <code><a href="#topic+tximport">tximport()</a></code>, <code><a href="#topic+installSysDeps">installSysDeps()</a></code>, <code><a href="#topic+seekerArray">seekerArray()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
doParallel::registerDoParallel()
params = yaml::read_yaml('my_params.yaml')
seeker(params)

## End(Not run)

</code></pre>

<hr>
<h2 id='seekerArray'>Process microarray data end to end</h2><span id='topic+seekerArray'></span>

<h3>Description</h3>

<p>This function fetches data and metadata from NCBI GEO and ArrayExpress,
processes raw Affymetrix data using RMA and custom CDFs from Brainarray, and
maps probes to genes. See also the vignettes:
<code>browseVignettes('seeker')</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seekerArray(
  study,
  geneIdType,
  platform = NULL,
  parentDir = ".",
  metadataOnly = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seekerArray_+3A_study">study</code></td>
<td>
<p>String indicating the study accession and used to name the output
directory within <code>parentDir</code>. Must start with &quot;GSE&quot;, &quot;E-&quot;, or &quot;LOCAL&quot;. If
starts with &quot;GSE&quot;, data are fetched using <code><a href="GEOquery.html#topic+getGEO">GEOquery::getGEO()</a></code>. If starts
with &quot;E-&quot;, data are fetched using <code><a href="ArrayExpress.html#topic+getAE">ArrayExpress::getAE()</a></code>. If starts with
&quot;LOCAL&quot;, data in the form of cel(.gz) files must in the directory
<code>parentDir</code>/<code>study</code>/raw, and <code>parentDir</code>/<code>study</code> must contain
a file &quot;sample_metadata.csv&quot; that has a column <code>sample_id</code> containing the
names of the cel(.gz) files without the file extension.</p>
</td></tr>
<tr><td><code id="seekerArray_+3A_geneidtype">geneIdType</code></td>
<td>
<p>String indicating whether to map probes to gene IDs from
Ensembl (&quot;ensembl&quot;) or Entrez (&quot;entrez&quot;).</p>
</td></tr>
<tr><td><code id="seekerArray_+3A_platform">platform</code></td>
<td>
<p>String indicating the GEO-based platform accession for the raw
data. See <a href="https://www.ncbi.nlm.nih.gov/geo/browse/?view=platforms">https://www.ncbi.nlm.nih.gov/geo/browse/?view=platforms</a>.
Only necessary if <code>study</code> starts with &quot;LOCAL&quot;, or starts with &quot;GSE&quot;
and the study uses multiple platforms.</p>
</td></tr>
<tr><td><code id="seekerArray_+3A_parentdir">parentDir</code></td>
<td>
<p>Directory in which to store the output, which will be a
directory named according to <code>study</code>.</p>
</td></tr>
<tr><td><code id="seekerArray_+3A_metadataonly">metadataOnly</code></td>
<td>
<p>Logical indicating whether to only process the sample
metadata, and skip processing the expression data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The standard output:
</p>

<ul>
<li><p> naive_expression_set.qs: Initial <code>ExpresssionSet</code> generated by
<a href="GEOquery.html#topic+getGEO">GEOquery::getGEO</a> or <code><a href="ArrayExpress.html#topic+ae2bioc">ArrayExpress::ae2bioc()</a></code>. Should generally <em>not</em> be
used if sample_metadata.csv and gene_expression_matrix.qs are available.
</p>
</li>
<li><p> sample_metadata.csv: Table of sample metadata. Column <code>sample_id</code> matches
colnames of the gene expression matrix.
</p>
</li>
<li><p> gene_expression_matrix.qs: Rows correspond to genes, columns to samples.
Expression values are log2-transformed.
</p>
</li>
<li><p> custom_cdf_name.txt: Name of custom CDF package used by <code><a href="affy.html#topic+justrma">affy::justRMA()</a></code>
to process and normalize raw Affymetrix data and map probes to genes.
</p>
</li>
<li><p> feature_metadata.qs: <code>GPL</code> object, if gene expression matrix was generated
from processed data.
</p>
</li>
<li><p> probe_gene_mapping.csv.gz: Table of probes and genes, if gene expression
matrix was generated from processed data.
</p>
</li>
<li><p> &quot;raw&quot; directory: Contains raw Affymetrix files.
</p>
</li>
<li><p> params.yml: Parameters used to process the dataset.
</p>
</li>
<li><p> session.log: R session information.
</p>
</li></ul>

<p>The output may include other files from NCBI GEO or ArrayExpress. Files with
extension &quot;qs&quot; can be read into R using <code><a href="qs.html#topic+qread">qs::qread()</a></code>.
</p>


<h3>Value</h3>

<p>Path to the output directory <code>parentDir</code>/<code>study</code>, invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
seekerArray('GSE25585', 'entrez')

## End(Not run)

</code></pre>

<hr>
<h2 id='trimgalore'>Run Trim Galore!</h2><span id='topic+trimgalore'></span>

<h3>Description</h3>

<p>This function calls
<a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">trim_galore</a>
using <code><a href="base.html#topic+system2">system2()</a></code>, and is only designed to handle standard adapter/quality
trimming. To run in parallel, register a parallel backend, e.g., using
<code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trimgalore(
  filepaths,
  outputDir = "trimgalore_output",
  cmd = "trim_galore",
  args = NULL,
  pigzCmd = "pigz"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trimgalore_+3A_filepaths">filepaths</code></td>
<td>
<p>Paths to fastq files. For single-end reads, each element
should be a single filepath. For paired-end reads, each element should be
two filepaths separated by &quot;;&quot;.</p>
</td></tr>
<tr><td><code id="trimgalore_+3A_outputdir">outputDir</code></td>
<td>
<p>Directory in which to store output. Will be created if it
doesn't exist.</p>
</td></tr>
<tr><td><code id="trimgalore_+3A_cmd">cmd</code></td>
<td>
<p>Name or path of the command-line interface.</p>
</td></tr>
<tr><td><code id="trimgalore_+3A_args">args</code></td>
<td>
<p>Additional arguments to pass to the command-line interface.
Output files will always be compressed. Arguments &quot;&ndash;gzip&quot;, &quot;&ndash;cores&quot;,
&quot;-j&quot;, and &quot;&ndash;basename&quot; are not allowed. Arguments &quot;-o&quot; and &quot;&ndash;paired&quot;
should not be specified here.</p>
</td></tr>
<tr><td><code id="trimgalore_+3A_pigzcmd">pigzCmd</code></td>
<td>
<p>String for pigz command, which will gzip the output files.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of exit codes, invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>
</p>

<hr>
<h2 id='tximport'>Run tximport on RNA-seq quantifications</h2><span id='topic+tximport'></span>

<h3>Description</h3>

<p>This function uses the
<a href="https://doi.org/doi:10.18129/B9.bioc.tximport">tximport package</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tximport(
  inputDir,
  tx2gene,
  samples = NULL,
  outputDir = "data",
  type = c("salmon", "kallisto"),
  countsFromAbundance = "lengthScaledTPM",
  ignoreTxVersion = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tximport_+3A_inputdir">inputDir</code></td>
<td>
<p>Directory that contains the quantification directories.</p>
</td></tr>
<tr><td><code id="tximport_+3A_tx2gene">tx2gene</code></td>
<td>
<p><code>NULL</code> or data.frame of mapping between transcripts and
genes, as returned by <code><a href="#topic+getTx2gene">getTx2gene()</a></code>, passed to <code><a href="tximport.html#topic+tximport">tximport::tximport()</a></code>.</p>
</td></tr>
<tr><td><code id="tximport_+3A_samples">samples</code></td>
<td>
<p>Names of quantification directories to include. <code>NULL</code>
indicates all.</p>
</td></tr>
<tr><td><code id="tximport_+3A_outputdir">outputDir</code></td>
<td>
<p>Directory in which to save the result, a file named
&quot;tximport_output.qs&quot;, using <code><a href="qs.html#topic+qsave">qs::qsave()</a></code>. If <code>NULL</code>, no file is saved.</p>
</td></tr>
<tr><td><code id="tximport_+3A_type">type</code></td>
<td>
<p>Passed to <code><a href="tximport.html#topic+tximport">tximport::tximport()</a></code>.</p>
</td></tr>
<tr><td><code id="tximport_+3A_countsfromabundance">countsFromAbundance</code></td>
<td>
<p>Passed to <code><a href="tximport.html#topic+tximport">tximport::tximport()</a></code>.</p>
</td></tr>
<tr><td><code id="tximport_+3A_ignoretxversion">ignoreTxVersion</code></td>
<td>
<p>Passed to <code><a href="tximport.html#topic+tximport">tximport::tximport()</a></code>.</p>
</td></tr>
<tr><td><code id="tximport_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="tximport.html#topic+tximport">tximport::tximport()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, as returned by <code><a href="tximport.html#topic+tximport">tximport::tximport()</a></code>, invisibly.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seeker">seeker()</a></code>, <code><a href="#topic+getTx2gene">getTx2gene()</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
