<!DOCTYPE html><html lang="en"><head><title>Help for package SegCorr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SegCorr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SegCorr-package'>
<p>Detecting Correlated Genomic Regions</p></a></li>
<li><a href='#CNV_correction'>
<p>Corrects Gene Expression for CNV</p></a></li>
<li><a href='#EXP_raw'>
<p>Simulated Gene Expression</p></a></li>
<li><a href='#SegCorr'>
<p>Performs CNV Correction and Correlation Matrix Segmentation</p></a></li>
<li><a href='#segmentation'>
<p>Correlation Matrix Segmentation</p></a></li>
<li><a href='#segmented_signal'>
<p>Mean Segmentation</p></a></li>
<li><a href='#SNP'>
<p>Simulated SNP signal</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Detecting Correlated Genomic Regions</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-12-12</td>
</tr>
<tr>
<td>Author:</td>
<td>Eleni Ioanna Delatola, Emilie Lebarbier, Tristan Mary-Huard, Francois
    Radvanyi, Stephane Robin, Jennifer Wong</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Eleni Ioanna Delatola &lt;eldelatola@yahoo.gr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs correlation matrix segmentation and applies a test
    procedure to detect highly correlated regions in gene expression.</td>
</tr>
<tr>
<td>Depends:</td>
<td>jointseg, methods</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-12-17 12:40:07 UTC; Eleanna</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-12-17 12:49:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='SegCorr-package'>
Detecting Correlated Genomic Regions
</h2><span id='topic+SegCorr-package'></span>

<h3>Description</h3>

<p>Performs correlation matrix segmentation and applies a test procedure to detect highly correlated regions in gene expression. The segmentation procedure detects changes in the patterns of the gene expression correlation matrix. The test procedure asseses which regions exhibit a significantly high
level of correlation. Additionally, a preprocessing procedure is provided to correct gene expression for copy number variation.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> SegCorr</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-01-19</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>
<p>Maintainer:  Eleni Ioanna Delatola &lt;eldelatola@yahoo.gr&gt;
</p>


<h3>References</h3>

<p>Delatola E. I., Lebarbier E., Mary-Huard T., Radvanyi F., Robin S., Wong J.(2017). SegCorr: a statistical procedure for the detection of genomic regions of correlated expression. BMC Bioinformatics, 18:333.</p>


<h3>See Also</h3>

<p><code><a href="jointseg.html#topic+Fpsn">Fpsn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data.sets = c('SNP','EXP_raw')
## Each gene corresponds to one SNP probe ##
#Position_EXP = matrix(1:1000,nrow=500,byrow=TRUE)
#Position_SNP = seq(2,1000,by=2)
#data(list=data.sets)
#CHR = rep(1,dim(EXP_raw)[1])
#SNP.CHR = rep(1,dim(SNP)[1])

#results = SegCorr(CHR = CHR, EXP = EXP_raw, CNV = TRUE, SNPSMOOTH=TRUE,
#Position.EXP = Position_EXP, SNP.CHR = SNP.CHR, SNP=SNP , Position.SNP = Position_SNP)

################drawing the heatmap for one region ###########################
#tau = results$Region.List[1,2]: results$Region.List[1,3]
#EXP.CNV =  results$EXP.corrected
#heatmap(EXP.CNV[tau,])

</code></pre>

<hr>
<h2 id='CNV_correction'>
Corrects Gene Expression for CNV
</h2><span id='topic+CNV_correction'></span>

<h3>Description</h3>

<p>Correcting gene expression signal for CNV. 

</p>


<h3>Usage</h3>

<pre><code class='language-R'>CNV_correction(s.Position.EXP, e.Position.EXP, Position.SNP, mu.SNP, EXP)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CNV_correction_+3A_s.position.exp">s.Position.EXP</code></td>
<td>

<p>vector with gene start position
</p>
</td></tr>
<tr><td><code id="CNV_correction_+3A_e.position.exp">e.Position.EXP</code></td>
<td>
<p>vector with gene end position</p>
</td></tr>
<tr><td><code id="CNV_correction_+3A_position.snp">Position.SNP</code></td>
<td>
<p>vector with SNP/CGH positions</p>
</td></tr>
<tr><td><code id="CNV_correction_+3A_mu.snp">mu.SNP</code></td>
<td>
<p>Smoothed genomic signal matrix not containing NA values. Rows correspond to probes, while columns to patients. The ordering of the patients must be the same as in the EXP matrix.</p>
</td></tr>
<tr><td><code id="CNV_correction_+3A_exp">EXP</code></td>
<td>
<p>Gene expression matrix must not contain NA's and genes with same expression value (i.e. null gene). Rows correspond to probes, while columns to patients.  Again, ordering of patients must be the same between EXP and mu.SNP matrices.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Overlapping genes may correspond to the same SNP/CGH probes.
</p>


<h3>Value</h3>

<p>CNV corrected signal matrix.
</p>


<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>


<h3>References</h3>

<p>Delatola E. I., Lebarbier E., Mary-Huard T., Radvanyi F., Robin S., Wong J.(2017). SegCorr: a statistical procedure for the detection of genomic regions of correlated expression. BMC Bioinformatics, 18:333.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+segmented_signal">segmented_signal</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data.sets = c('SNP','EXP_raw')
## Each gene corresponds to one SNP probe ##
#Position_EXP = matrix(1:1000,nrow=500,byrow=TRUE)
#Position_SNP = seq(2,1000,by=2)
#data(list=data.sets)
#mu.SNP = segmented_signal(SNP ,100) ## smoothed SNP signal
#EXP.CNV = CNV_correction(Position_EXP[,1], Position_EXP[,2], Position_SNP,
#mu.SNP, EXP_raw)## corrected signal

</code></pre>

<hr>
<h2 id='EXP_raw'>
Simulated Gene Expression
</h2><span id='topic+EXP_raw'></span>

<h3>Description</h3>

<p>Gene expression profiles have been generated for 30 patients and 500 genes. Background correlation is set to 0.08 and the correlation for H1 regions to 0.5. The location of the H1 regions is as suggested in the work of Lai et al.(2005), i.e region 1 [101, 105], region 2 [201, 210], region 3 [301, 320] and region 4 [401, 440].  </p>


<h3>Usage</h3>

<pre><code class='language-R'>data("EXP_raw")</code></pre>


<h3>Format</h3>

<p>Data frame containing the gene expression signal for 500 genes (rows) on 30 patients (columns).
</p>


<h3>References</h3>

<p>Lai, W. R., Johnson, M. D., Kucherlapati, R., &amp; Park, P. J. (2005). Comparative analysis of algorithms for identifying amplifications and deletions in array CGH data. Bioinformatics, 21(19), 3763-3770.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(EXP_raw)
#G = cor(t(EXP_raw))## calculating the gene x gene correlation matrix
#image(G)## plotting the correlation matrix
</code></pre>

<hr>
<h2 id='SegCorr'>
Performs CNV Correction and Correlation Matrix Segmentation
</h2><span id='topic+SegCorr'></span>

<h3>Description</h3>

<p>Gene expression is corrected for CNV events  must not contain NA's and genes with same expression value (i.e. null gene expression). Segmentation is used to detect changes in the correlation pattern. Regions with high
correlation are identified using an exact test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SegCorr(CHR, EXP, genes,S, CNV, SNPSMOOTH, Position.EXP, SNP.CHR, SNP, Position.SNP, Kmax)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SegCorr_+3A_chr">CHR</code></td>
<td>

<p>Chromosome allocation vector for the genes.
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_exp">EXP</code></td>
<td>

<p>Gene expression matrix (raw/corrected for CNV). Columns correspond to patients and rows to genes. The expression matrix must not contain either NA's or genes with same expression value (i.e. null gene expression)</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_genes">genes</code></td>
<td>

<p>Gene ID(name) vector.
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_s">S</code></td>
<td>

<p>Threshold for model selection. Default S=0.7.
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_cnv">CNV</code></td>
<td>

<p>Logical variable indicating whether to perform CNV correction. When CNV=T, the correction is performed. Default value CNV=F.
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_snpsmooth">SNPSMOOTH</code></td>
<td>

<p>(Optional Argument when CNV=T) Logical variable indicating whether to perform SNPSMOOTH. When SNPSMOOTH=T, the smoothing is performed. Default value SNPSMOOTH=F.
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_position.exp">Position.EXP</code></td>
<td>
<p>(Optional Argument when CNV=T) Expression position matrix.
First column is the start position and the second is the end position.
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_snp.chr">SNP.CHR</code></td>
<td>
<p>(Optional Argument when CNV=T)
Chromosome allocation vector for genomic probes.
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_snp">SNP</code></td>
<td>
<p>(Optional Argument when CNV=T)
SNP profile matrix not containing NA's. Columns correspond to patients and rows to probes.
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_position.snp">Position.SNP</code></td>
<td>
<p>(Optional Argument when CNV=T)
vector with SNP positions
</p>
</td></tr>
<tr><td><code id="SegCorr_+3A_kmax">Kmax</code></td>
<td>
<p>(Optional Argument when CNV=T and SNPSMOOTH=T)
Maximum number of segments. (mean profile segmentation)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Overlapping genes may correspond to the same genomic probes.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Results</code></td>
<td>
<p>Matrix containing information about the genomic regions. Each region corresponds to a row of the matrix, the one  with the smallest p-value is on the top of the list.</p>
</td></tr>
<tr><td><code>Results$CHR</code></td>
<td>
<p>Chromosome</p>
</td></tr>
<tr><td><code>Results$Start/End</code></td>
<td>
<p>the region boundaries  with repsect to the physical location of the gene in the chromosome</p>
</td></tr>
<tr><td><code>Results$Rho</code></td>
<td>
<p><code class="reqn">\rho</code> correlation</p>
</td></tr>
<tr><td><code>Results$length</code></td>
<td>
<p>number of genes in the region</p>
</td></tr>
<tr><td><code>Results$first/last gene</code></td>
<td>
<p>name of the first/last gene in the region</p>
</td></tr>
<tr><td><code>Results$p-value</code></td>
<td>
<p>p-value as obtained from the test</p>
</td></tr>
<tr><td><code>Results$genes</code></td>
<td>
<p>names of the genes belonging to the region</p>
</td></tr>
<tr><td><code>Results$p-valueadj</code></td>
<td>
<p>p-value of the region corrected for multiple testing</p>
</td></tr>
<tr><td><code>Chromosome.Inf</code></td>
<td>
<p>Matrix containing the estimated background correlation (rho0.hat) per chromsome, the number of segments and the log-loglikehood.</p>
</td></tr>
<tr><td><code>EXP.corrected</code></td>
<td>
<p>If the CNV option is chosen, the corrected signal is given.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>


<h3>References</h3>

<p>Delatola E. I., Lebarbier E., Mary-Huard T., Radvanyi F., Robin S., Wong J.(2017). SegCorr: a statistical procedure for the detection of genomic regions of correlated expression. BMC Bioinformatics, 18:333.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CNV_correction">CNV_correction</a></code>,<code><a href="#topic+segmentation">segmentation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data('EXP_raw')
#CHR = rep(1,dim(EXP_raw)[1])

#results = SegCorr(CHR = CHR, EXP = EXP_raw, CNV = FALSE,S=0.7)

################drawing the heatmap for one region ###########################
#tau = results$Region.List[1,2]: results$Region.List[1,3]
#heatmap(as.matrix(EXP_raw[tau,]))

</code></pre>

<hr>
<h2 id='segmentation'>
Correlation Matrix Segmentation
</h2><span id='topic+segmentation'></span>

<h3>Description</h3>

<p>For a given chromosome, gene correlation matrix segmentation is performed. Regions with high
correlation are identified using an exact test. The expression matrix must not contain NA's and genes with same expression value (i.e. null gene expression).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>segmentation(CHR, EXP, genes, S)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="segmentation_+3A_chr">CHR</code></td>
<td>

<p>chromosome name
</p>
</td></tr>
<tr><td><code id="segmentation_+3A_exp">EXP</code></td>
<td>

<p>Gene expression matrix (raw/corrected for CNV). Columns correspond to patients and rows to genes. The expression matrix must not contain either NA's or genes with same expression value (i.e. null gene expression).
</p>
</td></tr>
<tr><td><code id="segmentation_+3A_genes">genes</code></td>
<td>

<p>Gene ID(name) vector.
</p>
</td></tr>
<tr><td><code id="segmentation_+3A_s">S</code></td>
<td>

<p>Threshold for model selection. Default S=0.7.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Results</code></td>
<td>
<p>Matrix containing information about the genomic regions. Each region corresponds to a row of the matrix, the one  with the smallest p-value is on the top of the list.</p>
</td></tr>
<tr><td><code>Results$CHR</code></td>
<td>
<p>Chromosome</p>
</td></tr>
<tr><td><code>Results$Start/End</code></td>
<td>
<p>region boundaries  with respect to the physical location of the gene in the chromosome</p>
</td></tr>
<tr><td><code>Results$Rho</code></td>
<td>
<p><code class="reqn">\rho</code> correlation</p>
</td></tr>
<tr><td><code>Results$length</code></td>
<td>
<p>number of genes in the region</p>
</td></tr>
<tr><td><code>Results$first/last gene</code></td>
<td>
<p>name of the first/last gene in the region</p>
</td></tr>
<tr><td><code>Results$p-value</code></td>
<td>
<p>p-value as obtained from the test</p>
</td></tr>
<tr><td><code>Results$genes</code></td>
<td>
<p>names of genes belonging to the region</p>
</td></tr>
<tr><td><code>rho0</code></td>
<td>
<p>estimate of the background correlation</p>
</td></tr>
<tr><td><code>likelihood</code></td>
<td>
<p>log-likelihood</p>
</td></tr>
<tr><td><code>K</code></td>
<td>
<p>number of segments</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>


<h3>References</h3>

<p>Delatola E. I., Lebarbier E., Mary-Huard T., Radvanyi F., Robin S., Wong J.(2017). SegCorr: a statistical procedure for the detection of genomic regions of correlated expression. BMC Bioinformatics, 18:333.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(EXP_raw)
#G = cor(t(EXP_raw))## calculating the gene x gene correlation matrix
#image(G)## plotting the correlation matrix
#results = segmentation(EXP = EXP_raw)
</code></pre>

<hr>
<h2 id='segmented_signal'>
Mean Segmentation
</h2><span id='topic+segmented_signal'></span>

<h3>Description</h3>

<p>Mean segmentation on the genomic signal is performed using the Fpsn function of the jointseg package. </p>


<h3>Usage</h3>

<pre><code class='language-R'>segmented_signal(SNP.Chr, Kmax)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="segmented_signal_+3A_snp.chr">SNP.Chr</code></td>
<td>

<p>SNP/CGH profile matrix for a given chromosome (NA's not allowed). Columns correspond to patients and rows to probes.
</p>
</td></tr>
<tr><td><code id="segmented_signal_+3A_kmax">Kmax</code></td>
<td>

<p>Maximum number of segments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Smoothed genomic signal matrix. Rows correspond to probes and columns to patients.
</p>


<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>


<h3>References</h3>

<p>Morgane Pierre-Jean, Guillem Rigaill and Pierre Neuvial. Performance evaluation of DNA copy number segmentation methods. Briefings in Bioinformatics (2015) 16 (4): 600-615.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CNV_correction">CNV_correction</a></code>, <code><a href="jointseg.html#topic+Fpsn">Fpsn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(SNP)
#mu.SNP = segmented_signal(SNP ,100)
</code></pre>

<hr>
<h2 id='SNP'>
Simulated SNP signal
</h2><span id='topic+SNP'></span>

<h3>Description</h3>

<p>SNP profiles for 30 patients and 500 probes have been simulated. Each gene corresponds to one SNP value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SNP")</code></pre>


<h3>Format</h3>

<p>Data frame with 500 probes (rows) on 30 patients (patients).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(SNP)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
