<!DOCTYPE html><html><head><title>Help for package segregatr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {segregatr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#segregatr-package'><p>segregatr: Segregation Analysis for Variant Interpretation</p></a></li>
<li><a href='#FLB'><p>Full-likelihood Bayes factor</p></a></li>
<li><a href='#plotSegregation'><p>Pedigree plot for segregation analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Segregation Analysis for Variant Interpretation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.0</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of the full-likelihood Bayes factor (FLB)
    for evaluating segregation evidence in clinical medical genetics. The
    method was introduced by Thompson et al. (2003) &lt;<a href="https://doi.org/10.1086%2F378100">doi:10.1086/378100</a>&gt;.
    This implementation supports custom penetrance values and liability
    classes, and allows visualisations and robustness analysis as
    presented in Ratajska et al. (2023) &lt;<a href="https://doi.org/10.1002%2Fmgg3.2107">doi:10.1002/mgg3.2107</a>&gt;. See also the 
    online app 'shinyseg', <a href="https://chrcarrizosa.shinyapps.io/shinyseg">https://chrcarrizosa.shinyapps.io/shinyseg</a>, which 
    offers interactive segregation analysis with many additional features 
    (Carrizosa et al. (2024) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtae201">doi:10.1093/bioinformatics/btae201</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/magnusdv/segregatr">https://github.com/magnusdv/segregatr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/magnusdv/segregatr/issues">https://github.com/magnusdv/segregatr/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>pedtools (&ge; 2.2.0), R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>pedprobr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-28 14:01:38 UTC; magnu</td>
</tr>
<tr>
<td>Author:</td>
<td>Magnus Dehli Vigeland
    <a href="https://orcid.org/0000-0002-9134-4962"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Christian Carrizosa
    <a href="https://orcid.org/0000-0002-3919-0011"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Magnus Dehli Vigeland &lt;m.d.vigeland@medisin.uio.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-28 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='segregatr-package'>segregatr: Segregation Analysis for Variant Interpretation</h2><span id='topic+segregatr'></span><span id='topic+segregatr-package'></span>

<h3>Description</h3>

<p>An implementation of the full-likelihood Bayes factor (FLB) for evaluating segregation evidence in clinical medical genetics. The method was introduced by Thompson et al. (2003) <a href="https://doi.org/10.1086/378100">doi:10.1086/378100</a>. This implementation supports custom penetrance values and liability classes, and allows visualisations and robustness analysis as presented in Ratajska et al. (2023) <a href="https://doi.org/10.1002/mgg3.2107">doi:10.1002/mgg3.2107</a>. See also the online app 'shinyseg', <a href="https://chrcarrizosa.shinyapps.io/shinyseg">https://chrcarrizosa.shinyapps.io/shinyseg</a>, which offers interactive segregation analysis with many additional features (Carrizosa et al. (2024) <a href="https://doi.org/10.1093/bioinformatics/btae201">doi:10.1093/bioinformatics/btae201</a>).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Magnus Dehli Vigeland <a href="mailto:m.d.vigeland@medisin.uio.no">m.d.vigeland@medisin.uio.no</a> (<a href="https://orcid.org/0000-0002-9134-4962">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Christian Carrizosa (<a href="https://orcid.org/0000-0002-3919-0011">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/magnusdv/segregatr">https://github.com/magnusdv/segregatr</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/magnusdv/segregatr/issues">https://github.com/magnusdv/segregatr/issues</a>
</p>
</li></ul>


<hr>
<h2 id='FLB'>Full-likelihood Bayes factor</h2><span id='topic+FLB'></span>

<h3>Description</h3>

<p>Computes the Bayes factor for co-segregation, as originally described by
Thompson et al. (2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FLB(
  x,
  carriers = NULL,
  homozygous = NULL,
  noncarriers = NULL,
  freq = NULL,
  affected = NULL,
  unknown = NULL,
  proband = NULL,
  penetrances = NULL,
  liability = NULL,
  loopBreakers = NULL,
  Xchrom = FALSE,
  details = FALSE,
  plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FLB_+3A_x">x</code></td>
<td>
<p>A <code><a href="pedtools.html#topic+ped">pedtools::ped()</a></code> object.</p>
</td></tr>
<tr><td><code id="FLB_+3A_carriers">carriers</code></td>
<td>
<p>A character vector (or coercible to such), containing the ID
labels of pedigree members known to carry one copy of the variant in
question.</p>
</td></tr>
<tr><td><code id="FLB_+3A_homozygous">homozygous</code></td>
<td>
<p>A character vector (or coercible to such), containing the
ID labels of pedigree members known to carry two copies of the variant in
question.</p>
</td></tr>
<tr><td><code id="FLB_+3A_noncarriers">noncarriers</code></td>
<td>
<p>A character vector (or coercible to such), containing the
ID labels of pedigree members known <em>not</em> to carry the variant in question.</p>
</td></tr>
<tr><td><code id="FLB_+3A_freq">freq</code></td>
<td>
<p>A single number strictly between 0 and 1: the population
frequency of the observed allele.</p>
</td></tr>
<tr><td><code id="FLB_+3A_affected">affected</code></td>
<td>
<p>The affected pedigree members.</p>
</td></tr>
<tr><td><code id="FLB_+3A_unknown">unknown</code></td>
<td>
<p>Pedigree members with unknown affection status.</p>
</td></tr>
<tr><td><code id="FLB_+3A_proband">proband</code></td>
<td>
<p>The ID label of the proband. This person must also be in both
<code>carriers</code> and <code>affected</code>.</p>
</td></tr>
<tr><td><code id="FLB_+3A_penetrances">penetrances</code></td>
<td>
<p>For autosomal models, a numeric vector of length 3 <code style="white-space: pre;">&#8288;(f0, f1, f2)&#8288;</code>, or a matrix-like with 3 columns, where row <code>i</code> contains the
penetrances of liability class <code>i</code>. For X-linked models, a list of two
vectors named &quot;male&quot; and &quot;female&quot;, of lengths 2 <code style="white-space: pre;">&#8288;(f0, f1)&#8288;</code> and 3 <code style="white-space: pre;">&#8288;(f0, f1, f2)&#8288;</code> respectively. Alternatively, each list entry may be matrix-like (with
the same number of columns) where each row represents a liability class.</p>
</td></tr>
<tr><td><code id="FLB_+3A_liability">liability</code></td>
<td>
<p>A vector of length <code>pedsize(x)</code>, containing for each
pedigree member the row number of <code>penetrances</code> which should be used for
that individual. If unnamed, it is assumed that the individuals are taken
in order. (If <code>penetrances</code> is just a vector (or one for each sex in
X-linked models), it will be used for all classes.) If <code>liability</code> is NULL
(the default), it is set to <code>1</code> for all individuals.</p>
</td></tr>
<tr><td><code id="FLB_+3A_loopbreakers">loopBreakers</code></td>
<td>
<p>(Relevant only if <code>x</code> has loops.) A vector of ID labels
indicating loop breakers. The default value (NULL) initiates automatic loop
breaking, which is recommended in most cases.</p>
</td></tr>
<tr><td><code id="FLB_+3A_xchrom">Xchrom</code></td>
<td>
<p>A logical, indicating if a model of X-linked inheritance should
be applied.</p>
</td></tr>
<tr><td><code id="FLB_+3A_details">details</code></td>
<td>
<p>A logical, indicating if detailed output should be returned
(for debugging purposes).</p>
</td></tr>
<tr><td><code id="FLB_+3A_plot">plot</code></td>
<td>
<p>A logical.</p>
</td></tr>
<tr><td><code id="FLB_+3A_...">...</code></td>
<td>
<p>Optional plot parameters passed on to <code><a href="pedtools.html#topic+plot.ped">pedtools::plot.ped()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A positive number, the FLB score. If <code>details = TRUE</code>, a list
including intermediate results.
</p>


<h3>References</h3>

<p>Thompson D, Easton DF, Goldgar DE. <em>A full-likelihood method for
the evaluation of causality of sequence variants from family data.</em> Am J
Hum Genet, 2003. <a href="https://doi.org/10.1086/378100">doi:10.1086/378100</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
### Autosomal dominant

x = nuclearPed(2)

FLB(x, carriers = 3:4, aff = 3:4, unknown = 1:2,
    freq = 0.0001, penetrances = c(0, 1, 1), proband = 3)


### Autosomal recessive with phenocopies and reduced penetrance

y = nuclearPed(4)

FLB(y, carriers = 4:5, homozygous = 3, noncarriers = 6,
    aff = 3, unknown = 1:2, freq = 0.0001, proband = 3,
    penetrances = c(0.01, 0.01, 0.99), plot = TRUE)


### X-linked recessive

z = nuclearPed(3, sex = c(1, 1, 2)) |&gt;
  addChildren(mother = 5, nch = 2, sex = 1:2)

FLB(z, carriers = c(3, 7), nonc = 4, aff = c(3, 7), unknown = 1:2,
    freq = 0.0001, penetrances = list(male = c(0, 1), female = c(0, 0, 1)),
    proband = 7, Xchrom = TRUE, plot = TRUE)

</code></pre>

<hr>
<h2 id='plotSegregation'>Pedigree plot for segregation analysis</h2><span id='topic+plotSegregation'></span>

<h3>Description</h3>

<p>Plots a pedigree showing the segregation of a variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSegregation(
  x,
  affected = NULL,
  unknown = NULL,
  proband = NULL,
  carriers = NULL,
  homozygous = NULL,
  noncarriers = NULL,
  cex = 1,
  margins = 1,
  pos.geno = "bottom",
  pos.arrow = "bottomleft",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSegregation_+3A_x">x</code></td>
<td>
<p>A <code><a href="pedtools.html#topic+ped">pedtools::ped()</a></code> object.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_affected">affected</code></td>
<td>
<p>The affected pedigree members.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_unknown">unknown</code></td>
<td>
<p>Pedigree members with unknown affection status.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_proband">proband</code></td>
<td>
<p>The ID label of the proband. This person must also be in both
<code>carriers</code> and <code>affected</code>.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_carriers">carriers</code></td>
<td>
<p>A character vector (or coercible to such), containing the ID
labels of pedigree members known to carry one copy of the variant in
question.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_homozygous">homozygous</code></td>
<td>
<p>A character vector (or coercible to such), containing the
ID labels of pedigree members known to carry two copies of the variant in
question.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_noncarriers">noncarriers</code></td>
<td>
<p>A character vector (or coercible to such), containing the
ID labels of pedigree members known <em>not</em> to carry the variant in question.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_cex">cex</code>, <code id="plotSegregation_+3A_margins">margins</code></td>
<td>
<p>Arguments passed on to <code><a href="pedtools.html#topic+plot.ped">pedtools::plot.ped()</a></code>.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_pos.geno">pos.geno</code></td>
<td>
<p>Position of genotype labels relative to pedigree symbols;
either &quot;bottom&quot; (default), &quot;topleft&quot; or &quot;topright&quot;.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_pos.arrow">pos.arrow</code></td>
<td>
<p>Position of the proband arrow; either &quot;bottomleft&quot;,
&quot;bottomright&quot;, &quot;topleft&quot; or &quot;topright&quot;.</p>
</td></tr>
<tr><td><code id="plotSegregation_+3A_...">...</code></td>
<td>
<p>Optional plot parameters passed on to <code><a href="pedtools.html#topic+plot.ped">pedtools::plot.ped()</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
x = nuclearPed(2)
plotSegregation(x, proband = 3, carriers = 3:4, noncarriers = 1,
                aff = 3:4, unknown = 1:2)

# Same with various options
plotSegregation(x, proband = 3, carriers = 3:4, noncarriers = 1,
                aff = 3:4, unknown = 1:2,
                pos.geno = "topright", pos.arrow = "topleft",
                labs = NULL, title = "Family 1", cex.main = 1.5)

# Recessive example
y = cousinPed(1, child = TRUE)
plotSegregation(y, affected = 9, unknown = 1:6, carrier = 7:8,
                homozygous = 9, noncarriers = c(4,6), proband = 9)

# Different symbol placements
plotSegregation(y, affected = 9, unknown = 1:6, carrier = 7:8,
                homozygous = 9, noncarriers = c(4,6), proband = 9,
                pos.geno = "topleft", pos.arrow = "bottomright")

# Incest case
y = nuclearPed() |&gt; addChildren(father = 3, mother = 2, nch = 3)

plotSegregation(y, proband = 4, aff = 4:6, unknown = 2, carrier = 4:6, deceased = 1,
                pos.geno = "topleft", pos.arrow = "bottomright")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
