<!DOCTYPE html><html lang="en"><head><title>Help for package segtest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {segtest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#segtest-package'><p>segtest: Tests for Segregation Distortion in Polyploids</p></a></li>
<li><a href='#chisq_g4'><p>Chi Square test when genotypes are known</p></a></li>
<li><a href='#chisq_gl4'><p>Chi-Sq for GL</p></a></li>
<li><a href='#em_li'><p>EM algorithm from Li (2011)</p></a></li>
<li><a href='#gcount_to_gvec'><p>Converts genotype counts to genotype vectors.</p></a></li>
<li><a href='#gvec_to_gcount'><p>Inverse function of <code>gcount_to_gvec()</code>.</p></a></li>
<li><a href='#is_valid_2'><p>Tests if the two parameter model is valid</p></a></li>
<li><a href='#iter.array'><p>Iterator over array</p></a></li>
<li><a href='#like_gknown_2'><p>Likelihood under three parameter model when genotypes are known</p></a></li>
<li><a href='#like_gknown_3'><p>Likelihood under three parameter model when genotypes are known</p></a></li>
<li><a href='#like_glpknown_2'><p>Likelihood under three parameter model when using offspring genotypes</p>
likelihoods but parent genotypes are known.</a></li>
<li><a href='#like_glpknown_3'><p>Likelihood under three parameter model when using offspring genotypes</p>
likelihoods but parent genotypes are known.</a></li>
<li><a href='#llike_li'><p>Objective function for <code>em_li()</code></p></a></li>
<li><a href='#lrt_men_g4'><p>Likelihood ratio test for segregation distortion with known genotypes</p></a></li>
<li><a href='#lrt_men_gl4'><p>Likelihood ratio test using genotype likelihoods.</p></a></li>
<li><a href='#multi_lrt'><p>Parallelized likelihood ratio test for segregation distortion.</p></a></li>
<li><a href='#multidog_to_g'><p>Converts multidog output to a format usable for multi_lrt()</p></a></li>
<li><a href='#nextElem.arrayiter'><p>Next element in an array</p></a></li>
<li><a href='#offspring_geno'><p>Simulates genotypes given genotype frequencies.</p></a></li>
<li><a href='#offspring_gf_2'><p>Calculates offspring genotype frequencies under the two-parameter model.</p></a></li>
<li><a href='#offspring_gf_3'><p>Calculates offspring genotype frequencies under the three-parameter model.</p></a></li>
<li><a href='#otest_g'><p>Jointly tests for segregation distortion and number of incompatible genotypes</p></a></li>
<li><a href='#po_gl'><p>Generate genotype likelihoods from offspring genotypes.</p></a></li>
<li><a href='#polymapr_test'><p>Run segregation distortion tests as implemented in the polymapR package.</p></a></li>
<li><a href='#pvec_tet_2'><p>Tetraploid gamete frequencies of gametes when one parent's genotype is known</p></a></li>
<li><a href='#pvec_tet_3'><p>Tetraploid gamete frequencies of gametes when one parent's genotype is known</p></a></li>
<li><a href='#simf1g'><p>Simulate genotype counts from F1 individuals</p></a></li>
<li><a href='#simf1gl'><p>Simulate genotype likelihoods of F1 individuals.</p></a></li>
<li><a href='#simgl'><p>Simulate genotype likelihoods from genotype counts</p></a></li>
<li><a href='#three_to_two'><p>Convert from three parameters to two parameters</p></a></li>
<li><a href='#ufit'><p>Genotype data from Cappai et al. (2020)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Tests for Segregation Distortion in Polyploids</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a suite of tests for
    segregation distortion in F1 polyploid populations (for
    now, just tetraploids). This is under different assumptions of
    meiosis. Details of these methods are described in
    Gerard et al. (2025) &lt;<a href="https://doi.org/10.1007%2Fs00122-025-04816-z">doi:10.1007/s00122-025-04816-z</a>&gt;.
    This material is based upon work supported by the
    National Science Foundation under Grant No. 2132247.  The
    opinions, findings, and conclusions or recommendations expressed
    are those of the author and do not necessarily reflect the views
    of the National Science Foundation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/dcgerard/segtest/issues">https://github.com/dcgerard/segtest/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/dcgerard/segtest">https://github.com/dcgerard/segtest</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Biarch:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>doFuture, doRNG, foreach, future, iterators, Rcpp, updog</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, polymapR, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-16 16:29:58 UTC; dgerard</td>
</tr>
<tr>
<td>Author:</td>
<td>David Gerard <a href="https://orcid.org/0000-0001-9450-5023"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Mira Thakkar [aut],
  NSF DBI 2132247 [fnd]
    (https://www.nsf.gov/awardsearch/showAward?AWD_ID=2132247)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Gerard &lt;gerard.1787@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-16 17:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='segtest-package'>segtest: Tests for Segregation Distortion in Polyploids</h2><span id='topic+segtest'></span><span id='topic+segtest-package'></span>

<h3>Description</h3>

<p>Provides a suite of tests for segregation distortion in F1 polyploid populations (for now, just tetraploids). This is under different assumptions of meiosis. Details of these methods are described in Gerard et al. (2025) <a href="https://doi.org/10.1007/s00122-025-04816-z">doi:10.1007/s00122-025-04816-z</a>. This material is based upon work supported by the National Science Foundation under Grant No. 2132247. The opinions, findings, and conclusions or recommendations expressed are those of the author and do not necessarily reflect the views of the National Science Foundation.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: David Gerard <a href="mailto:gerard.1787@gmail.com">gerard.1787@gmail.com</a> (<a href="https://orcid.org/0000-0001-9450-5023">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Mira Thakkar <a href="mailto:chrtmira@gmail.com">chrtmira@gmail.com</a>
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> NSF DBI 2132247 (https://www.nsf.gov/awardsearch/showAward?AWD_ID=2132247) [funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/dcgerard/segtest">https://github.com/dcgerard/segtest</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/dcgerard/segtest/issues">https://github.com/dcgerard/segtest/issues</a>
</p>
</li></ul>


<hr>
<h2 id='chisq_g4'>Chi Square test when genotypes are known</h2><span id='topic+chisq_g4'></span>

<h3>Description</h3>

<p>This chi-squared test is run under the assumption of no double reduction
and no preferential pairing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chisq_g4(x, g1, g2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chisq_g4_+3A_x">x</code></td>
<td>
<p>Vector of observed genotype counts</p>
</td></tr>
<tr><td><code id="chisq_g4_+3A_g1">g1</code></td>
<td>
<p>Parent 1's genotype</p>
</td></tr>
<tr><td><code id="chisq_g4_+3A_g2">g2</code></td>
<td>
<p>Parent 2's genotype</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the chi-squared statistic, degrees of
freedom, and p-value.
</p>


<h3>Author(s)</h3>

<p>Mira Thakkar and David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(1, 2, 4, 3, 0)
g1 &lt;- 2
g2 &lt;- 2
chisq_g4(x, g1, g2)

x &lt;- c(10, 25, 10, 0, 0)
g1 &lt;- 1
g2 &lt;- 1
chisq_g4(x, g1, g2)

</code></pre>

<hr>
<h2 id='chisq_gl4'>Chi-Sq for GL</h2><span id='topic+chisq_gl4'></span>

<h3>Description</h3>

<p>Calculates the MLE genotype and runs a chi-squared test assuming
no double reduction and no preferential pairing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chisq_gl4(gl, g1, g2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chisq_gl4_+3A_gl">gl</code></td>
<td>
<p>A matrix of offspring genotype log-likelihoods. The rows index the
individuals and the columns index the possible genotypes. So
<code>gl[i, k]</code> is the offspring genotype log-likelihood for individual i and
genotype k-1.</p>
</td></tr>
<tr><td><code id="chisq_gl4_+3A_g1">g1</code></td>
<td>
<p>The first parent's genotype.</p>
</td></tr>
<tr><td><code id="chisq_gl4_+3A_g2">g2</code></td>
<td>
<p>The second parent's genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the chi-squared statistic, degrees of
freedom, and p-value.
</p>


<h3>Author(s)</h3>

<p>Mira Thakkar and David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## null sim
set.seed(1)
g1 &lt;- 2
g2 &lt;- 2
gl &lt;- simf1gl(n = 25, g1 = g1, g2 = g2, alpha = 0, xi2 = 1/3)
chisq_gl4(gl = gl, g1 = g1, g2 = g2)

</code></pre>

<hr>
<h2 id='em_li'>EM algorithm from Li (2011)</h2><span id='topic+em_li'></span>

<h3>Description</h3>

<p>EM algorithm to estimate prior genotype probabilities from genotype
likelihoods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>em_li(B, itermax = 100L, eps = 1e-05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="em_li_+3A_b">B</code></td>
<td>
<p>Matrix of genotype log-likelihoods. The rows index the individuals
and the columns index the genotypes.</p>
</td></tr>
<tr><td><code id="em_li_+3A_itermax">itermax</code></td>
<td>
<p>The maximum number of iterations.</p>
</td></tr>
<tr><td><code id="em_li_+3A_eps">eps</code></td>
<td>
<p>The stopping criteria.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of log prior probabilities for each genotype.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>References</h3>


<ul>
<li><p>Li, H. (2011). A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. <em>Bioinformatics</em>, 27(21), 2987-2993. <a href="https://doi.org/10.1093/bioinformatics/btr509">doi:10.1093/bioinformatics/btr509</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Simulate some data
set.seed(1)
gl &lt;- simgl(nvec = c(3, 2, 4, 1, 2))
# Run em
lprob &lt;- em_li(B = gl)
# Exponentiate to get probabilities
prob &lt;- exp(c(lprob))
prob

</code></pre>

<hr>
<h2 id='gcount_to_gvec'>Converts genotype counts to genotype vectors.</h2><span id='topic+gcount_to_gvec'></span>

<h3>Description</h3>

<p>Converts genotype counts to genotype vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gcount_to_gvec(gcount)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gcount_to_gvec_+3A_gcount">gcount</code></td>
<td>
<p>The vector of genotype counts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of length <code>sum(gcount)</code>, containing <code>gcount[1]</code>
copies of <code>0</code>, <code>gcount[2]</code> copies of <code>1</code>, <code>gcount[3]</code>
copies of <code>2</code>, etc.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gvec_to_gcount">gvec_to_gcount()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gcount &lt;- c(1, 2, 3, 0, 5)
gcount_to_gvec(gcount = gcount)
</code></pre>

<hr>
<h2 id='gvec_to_gcount'>Inverse function of <code><a href="#topic+gcount_to_gvec">gcount_to_gvec</a>()</code>.</h2><span id='topic+gvec_to_gcount'></span>

<h3>Description</h3>

<p>Inverse function of <code><a href="#topic+gcount_to_gvec">gcount_to_gvec</a>()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gvec_to_gcount(gvec, ploidy = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gvec_to_gcount_+3A_gvec">gvec</code></td>
<td>
<p>The vector of genotypes. <code>gvec[i]</code> is the genotype
for individual i.</p>
</td></tr>
<tr><td><code id="gvec_to_gcount_+3A_ploidy">ploidy</code></td>
<td>
<p>The ploidy of the species.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of counts. Element <code>k</code> is the number of individuals
with genotype <code>k-1</code>.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gcount_to_gvec">gcount_to_gvec()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gvec &lt;- c(1, 2, 3, 2, 3, 1, 4, 0, 1, 0, 0, 1, 0, 0)
gvec_to_gcount(gvec = gvec)

</code></pre>

<hr>
<h2 id='is_valid_2'>Tests if the two parameter model is valid</h2><span id='topic+is_valid_2'></span>

<h3>Description</h3>

<p>There is a dependence on the bounds of two-parameter model. This function
returns TRUE if those bounds are satisfied and FALSE otherwise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_valid_2(dr, pp, drbound = 1/6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_valid_2_+3A_dr">dr</code></td>
<td>
<p>The double reduction rate.</p>
</td></tr>
<tr><td><code id="is_valid_2_+3A_pp">pp</code></td>
<td>
<p>The preferential pairing parameter.</p>
</td></tr>
<tr><td><code id="is_valid_2_+3A_drbound">drbound</code></td>
<td>
<p>The maximum double reduction rate possible.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>TRUE</code> if the model is valid, <code>FALSE</code> otherwise.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>TOL &lt;- 1e-6
is_valid_2(dr = 1/6, pp = 1/3, drbound = 1/6) # Valid
is_valid_2(dr = 1/6, pp = 1/3 - TOL, drbound = 1/6) # Not valid
is_valid_2(dr = 1/6, pp = 1/3 + TOL, drbound = 1/6) # Not valid


</code></pre>

<hr>
<h2 id='iter.array'>Iterator over array</h2><span id='topic+iter.array'></span>

<h3>Description</h3>

<p>Iterator over array
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'array'
iter(obj, by = 1, recycle = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iter.array_+3A_obj">obj</code></td>
<td>
<p>An array.</p>
</td></tr>
<tr><td><code id="iter.array_+3A_by">by</code></td>
<td>
<p>The dimension to iterate over.</p>
</td></tr>
<tr><td><code id="iter.array_+3A_recycle">recycle</code></td>
<td>
<p>Should the iterator reset?</p>
</td></tr>
<tr><td><code id="iter.array_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An iterator. This is an S3 arrayiter object, used in conjunction
with nextElem to iterate over one index of an array.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nextElem.arrayiter">nextElem.arrayiter()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>glist &lt;- multidog_to_g(mout = ufit, type = "all_gl", p1 = "indigocrisp", p2 = "sweetcrisp")
g &lt;- iterators::iter(glist$g, by = 3)
head(iterators::nextElem(g))
head(iterators::nextElem(g))
head(iterators::nextElem(g))

</code></pre>

<hr>
<h2 id='like_gknown_2'>Likelihood under three parameter model when genotypes are known</h2><span id='topic+like_gknown_2'></span>

<h3>Description</h3>

<p>This is under the two parameter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>like_gknown_2(x, alpha, xi1, xi2, g1, g2, log_p = TRUE, pen = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="like_gknown_2_+3A_x">x</code></td>
<td>
<p>A vector of length 5. <code>x[i]</code> is the count of individuals with
genotype <code>i-1</code>.</p>
</td></tr>
<tr><td><code id="like_gknown_2_+3A_alpha">alpha</code></td>
<td>
<p>The double reduction rate.</p>
</td></tr>
<tr><td><code id="like_gknown_2_+3A_xi1">xi1</code></td>
<td>
<p>The preferential pairing parameter of parent 1.</p>
</td></tr>
<tr><td><code id="like_gknown_2_+3A_xi2">xi2</code></td>
<td>
<p>The preferential pairing parameter of parent 2.</p>
</td></tr>
<tr><td><code id="like_gknown_2_+3A_g1">g1</code></td>
<td>
<p>Parent 1's genotype.</p>
</td></tr>
<tr><td><code id="like_gknown_2_+3A_g2">g2</code></td>
<td>
<p>Parent 2's genotype.</p>
</td></tr>
<tr><td><code id="like_gknown_2_+3A_log_p">log_p</code></td>
<td>
<p>A logical. Should we return the log likelihood or not?</p>
</td></tr>
<tr><td><code id="like_gknown_2_+3A_pen">pen</code></td>
<td>
<p>A tiny penalty to help with numerical stability</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The (log) likelihood.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(1, 4, 5, 3, 1)
alpha &lt;- 0.01
xi1 &lt;- 0.5
xi2 &lt;- 0.3
g1 &lt;- 1
g2 &lt;- 2
like_gknown_2(
  x = x,
  alpha = alpha,
  xi1 = xi1,
  xi2 = xi2,
  g1 = g1,
  g2 = g2)

</code></pre>

<hr>
<h2 id='like_gknown_3'>Likelihood under three parameter model when genotypes are known</h2><span id='topic+like_gknown_3'></span>

<h3>Description</h3>

<p>This is under the three parameter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>like_gknown_3(x, tau, beta, gamma1, gamma2, g1, g2, log_p = TRUE, pen = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="like_gknown_3_+3A_x">x</code></td>
<td>
<p>A vector of length 5. <code>x[i]</code> is the count of individuals with
genotype <code>i-1</code>.</p>
</td></tr>
<tr><td><code id="like_gknown_3_+3A_tau">tau</code></td>
<td>
<p>The probability of quadrivalent formation.</p>
</td></tr>
<tr><td><code id="like_gknown_3_+3A_beta">beta</code></td>
<td>
<p>The probability of double reduction given quadrivalent formation.</p>
</td></tr>
<tr><td><code id="like_gknown_3_+3A_gamma1">gamma1</code></td>
<td>
<p>The probability of AA_aa pairing for parent 1.</p>
</td></tr>
<tr><td><code id="like_gknown_3_+3A_gamma2">gamma2</code></td>
<td>
<p>The probability of AA_aa pairing for parent 2.</p>
</td></tr>
<tr><td><code id="like_gknown_3_+3A_g1">g1</code></td>
<td>
<p>Parent 1's genotype.</p>
</td></tr>
<tr><td><code id="like_gknown_3_+3A_g2">g2</code></td>
<td>
<p>Parent 2's genotype.</p>
</td></tr>
<tr><td><code id="like_gknown_3_+3A_log_p">log_p</code></td>
<td>
<p>A logical. Should we return the log likelihood or not?</p>
</td></tr>
<tr><td><code id="like_gknown_3_+3A_pen">pen</code></td>
<td>
<p>A tiny penalty to help with numerical stability</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The (log) likelihood.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(1, 4, 5, 3, 1)
tau &lt;- 0.5
beta &lt;- 0.1
gamma1 &lt;- 0.5
gamma2 &lt;- 0.3
g1 &lt;- 1
g2 &lt;- 2
like_gknown_3(
  x = x,
  tau = tau,
  beta = beta,
  gamma1 = gamma1,
  gamma2 = gamma2,
  g1 = g1,
  g2 = g2)

</code></pre>

<hr>
<h2 id='like_glpknown_2'>Likelihood under three parameter model when using offspring genotypes
likelihoods but parent genotypes are known.</h2><span id='topic+like_glpknown_2'></span>

<h3>Description</h3>

<p>This is under the two parameter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>like_glpknown_2(gl, alpha, xi1, xi2, g1, g2, log_p = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="like_glpknown_2_+3A_gl">gl</code></td>
<td>
<p>The matrix of genotype likelihoods of the offspring. Rows index
The individuals, columns index the genotypes.</p>
</td></tr>
<tr><td><code id="like_glpknown_2_+3A_alpha">alpha</code></td>
<td>
<p>The double reduction rate.</p>
</td></tr>
<tr><td><code id="like_glpknown_2_+3A_xi1">xi1</code></td>
<td>
<p>The preferential pairing parameter of parent 1.</p>
</td></tr>
<tr><td><code id="like_glpknown_2_+3A_xi2">xi2</code></td>
<td>
<p>The preferential pairing parameter of parent 2.</p>
</td></tr>
<tr><td><code id="like_glpknown_2_+3A_g1">g1</code></td>
<td>
<p>Parent 1's genotype.</p>
</td></tr>
<tr><td><code id="like_glpknown_2_+3A_g2">g2</code></td>
<td>
<p>Parent 2's genotype.</p>
</td></tr>
<tr><td><code id="like_glpknown_2_+3A_log_p">log_p</code></td>
<td>
<p>A logical. Should we return the log likelihood or not?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The (log) likelihood of the two parameter model when using
genotype likelihoods.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g1 &lt;- 1
g2 &lt;- 0
gl &lt;- simf1gl(
  n = 25,
  g1 = g1,
  g2 = g2,
  rd = 10,
  alpha = 0,
  xi1 = 1/3,
  xi2 = 1/3)
like_glpknown_2(
  gl = gl,
  alpha = 0.01,
  xi1 = 0.5,
  xi2 = 0.3,
  g1 = g1,
  g2 = g2,
  log_p = TRUE)

</code></pre>

<hr>
<h2 id='like_glpknown_3'>Likelihood under three parameter model when using offspring genotypes
likelihoods but parent genotypes are known.</h2><span id='topic+like_glpknown_3'></span>

<h3>Description</h3>

<p>This is under the three parameter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>like_glpknown_3(gl, tau, beta, gamma1, gamma2, g1, g2, log_p = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="like_glpknown_3_+3A_gl">gl</code></td>
<td>
<p>The matrix of genotype likelihoods of the offspring. Rows index
The individuals, columns index the genotypes.</p>
</td></tr>
<tr><td><code id="like_glpknown_3_+3A_tau">tau</code></td>
<td>
<p>The probability of quadrivalent formation.</p>
</td></tr>
<tr><td><code id="like_glpknown_3_+3A_beta">beta</code></td>
<td>
<p>The probability of double reduction given quadrivalent formation.</p>
</td></tr>
<tr><td><code id="like_glpknown_3_+3A_gamma1">gamma1</code></td>
<td>
<p>The probability of AA_aa pairing for parent 1.</p>
</td></tr>
<tr><td><code id="like_glpknown_3_+3A_gamma2">gamma2</code></td>
<td>
<p>The probability of AA_aa pairing for parent 2.</p>
</td></tr>
<tr><td><code id="like_glpknown_3_+3A_g1">g1</code></td>
<td>
<p>Parent 1's genotype.</p>
</td></tr>
<tr><td><code id="like_glpknown_3_+3A_g2">g2</code></td>
<td>
<p>Parent 2's genotype.</p>
</td></tr>
<tr><td><code id="like_glpknown_3_+3A_log_p">log_p</code></td>
<td>
<p>A logical. Should we return the log likelihood or not?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The (log) likelihood of the three parameter model when using
genotype likelihoods.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g1 &lt;- 1
g2 &lt;- 0
gl &lt;- simf1gl(
  n = 25,
  g1 = g1,
  g2 = g2,
  rd = 10,
  alpha = 0,
  xi1 = 1/3,
  xi2 = 1/3)
like_glpknown_3(
  gl = gl,
  tau = 1/2,
  beta = 1/12,
  gamma1 = 1/3,
  gamma2 = 1/3,
  g1 = g1,
  g2 = g2,
  log_p = TRUE)

</code></pre>

<hr>
<h2 id='llike_li'>Objective function for <code><a href="#topic+em_li">em_li</a>()</code></h2><span id='topic+llike_li'></span>

<h3>Description</h3>

<p>Objective function for <code><a href="#topic+em_li">em_li</a>()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>llike_li(B, lpivec)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="llike_li_+3A_b">B</code></td>
<td>
<p>The log-likelihood matrix. Rows are individuals columns are
genotypes.</p>
</td></tr>
<tr><td><code id="llike_li_+3A_lpivec">lpivec</code></td>
<td>
<p>The log prior vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log-likelihood of a vector of genotype frequencies when
using genotype likelihoods. This is from Li (2011).
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>References</h3>


<ul>
<li><p>Li, H. (2011). A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. <em>Bioinformatics</em>, 27(21), 2987-2993. <a href="https://doi.org/10.1093/bioinformatics/btr509">doi:10.1093/bioinformatics/btr509</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Simulate some data
set.seed(1)
gl &lt;- simgl(nvec = c(3, 2, 4, 1, 2))
# Log-likelihood at given log-priors
prob &lt;- c(0.1, 0.2, 0.4, 0.2, 0.1)
lprob &lt;- log(prob)
llike_li(B = gl, lpivec = lprob)

</code></pre>

<hr>
<h2 id='lrt_men_g4'>Likelihood ratio test for segregation distortion with known genotypes</h2><span id='topic+lrt_men_g4'></span>

<h3>Description</h3>

<p>This will run a likelihood ratio test using the genotypes of an F1 population
of tetraploids for the null of Mendelian segregation (accounting for double
reduction and preferential pairing) against the alternative of
segregation distortion. This is when the genotypes are assumed known.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lrt_men_g4(
  x,
  g1,
  g2,
  drbound = 1/6,
  pp = TRUE,
  dr = TRUE,
  alpha = 0,
  xi1 = 1/3,
  xi2 = 1/3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lrt_men_g4_+3A_x">x</code></td>
<td>
<p>A vector of genotype counts. <code>x[i]</code> is the number of
offspring with genotype <code>i-1</code>.</p>
</td></tr>
<tr><td><code id="lrt_men_g4_+3A_g1">g1</code></td>
<td>
<p>The genotype of parent 1.</p>
</td></tr>
<tr><td><code id="lrt_men_g4_+3A_g2">g2</code></td>
<td>
<p>The genotype of parent 2.</p>
</td></tr>
<tr><td><code id="lrt_men_g4_+3A_drbound">drbound</code></td>
<td>
<p>The maximum rate of double reduction. A default of 1/6
is provided, which is the rate under the complete equational
segregation model of meiosis.</p>
</td></tr>
<tr><td><code id="lrt_men_g4_+3A_pp">pp</code></td>
<td>
<p>A logical. Should we account for preferential pairing
(<code>TRUE</code>) or not (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="lrt_men_g4_+3A_dr">dr</code></td>
<td>
<p>A logical. Should we account for double reduction
(<code>TRUE</code>) or not (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="lrt_men_g4_+3A_alpha">alpha</code></td>
<td>
<p>If <code>dr = FALSE</code>, this is the known rate of double
reduction.</p>
</td></tr>
<tr><td><code id="lrt_men_g4_+3A_xi1">xi1</code></td>
<td>
<p>If <code>pp = FALSE</code>, this is the known preferential pairing
parameter of parent 1.</p>
</td></tr>
<tr><td><code id="lrt_men_g4_+3A_xi2">xi2</code></td>
<td>
<p>If <code>pp = FALSE</code>, this is the known preferential pairing
parameter of parent 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following elements
</p>

<dl>
<dt><code>statistic</code></dt><dd><p>The log-likelihood ratio test statistic.</p>
</dd>
<dt><code>df</code></dt><dd><p>The degrees of freedom.</p>
</dd>
<dt><code>p_value</code></dt><dd><p>The p-value.</p>
</dd>
<dt><code>alpha</code></dt><dd><p>The estimated double reduction rate.</p>
</dd>
<dt><code>xi1</code></dt><dd><p>The estimated preferential pairing parameter of parent 1.</p>
</dd>
<dt><code>xi2</code></dt><dd><p>The estimated preferential pairing parameter of parent 2.</p>
</dd>
</dl>



<h3>Impossible genotypes</h3>

<p>Some offspring genotype combinations are impossible given the parental
genotypes. If these impossible genotypes combinations show up, we return a
p-value of 0, a log-likelihood ratio statistic of Infinity, and missing
values for all other return items. The impossible genotypes are:
</p>

<dl>
<dt><code>g1 = 0 &amp;&amp; g2 = 0</code></dt><dd><p>Only offspring genotypes of 0 are possible.</p>
</dd>
<dt><code>g1 = 4 &amp;&amp; g2 = 4</code></dt><dd><p>Only offspring genotypes of 4 are possible.</p>
</dd>
<dt><code>g1 = 0 &amp;&amp; g2 = 4 || g1 == 4 &amp;&amp; g2 == 0</code></dt><dd><p>Only offspring genotypes of 2 are possible.</p>
</dd>
<dt><code>g1 = 0 &amp;&amp; g2 %in% c(1, 2, 3) || g1 = %in% c(1, 2, 3) &amp;&amp; g2 == 0</code></dt><dd><p>Only offspring genotypes of 0, 1, and 2 are possible.</p>
</dd>
<dt><code>g1 = 4 &amp;&amp; g2 %in% c(1, 2, 3) || g1 = %in% c(1, 2, 3) &amp;&amp; g2 == 4</code></dt><dd><p>Only offspring genotypes of 2, 3, and 4 are possible.</p>
</dd>
</dl>



<h3>Unidentified parameters</h3>

<p>When <code>g1 = 2</code> or <code>g2 = 2</code> (or both), the model is not identified
and those estimates (<code>alpha</code>, <code>xi1</code>, and <code>xi2</code>) are
meaningless. Do NOT interpret them.
</p>
<p>The estimate of <code>alpha</code> (double reduction rate) IS identified as
long as at least one parent is simplex, and no parent is duplex.
However, the estimates of the double reduction rate have extremely high
variance.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(100)
gf &lt;- offspring_gf_2(alpha = 1/12, xi1 = 0.2, xi2 = 0.6, p1 = 1, p2 = 0)
x &lt;- offspring_geno(gf = gf, n = 100)
lrt_men_g4(x = x, g1 = 1, g2 = 0)

</code></pre>

<hr>
<h2 id='lrt_men_gl4'>Likelihood ratio test using genotype likelihoods.</h2><span id='topic+lrt_men_gl4'></span>

<h3>Description</h3>

<p>This will run a likelihood ratio test using the genotypes of an F1 population
of tetraploids for the null of Mendelian segregation (accounting for double
reduction and preferential pairing) against the alternative of
segregation distortion. This is when genotype uncertainty is accounted
for through genotype likelihoods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lrt_men_gl4(
  gl,
  g1 = NULL,
  g2 = NULL,
  drbound = 1/6,
  pp = TRUE,
  dr = TRUE,
  alpha = 0,
  xi1 = 1/3,
  xi2 = 1/3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lrt_men_gl4_+3A_gl">gl</code></td>
<td>
<p>The genotype log-likelihoods. The rows index the individuals
and the columns index the genotypes.</p>
</td></tr>
<tr><td><code id="lrt_men_gl4_+3A_g1">g1</code></td>
<td>
<p>Either parent 1's genotype, or parent 1's genotype log-likelihoods.</p>
</td></tr>
<tr><td><code id="lrt_men_gl4_+3A_g2">g2</code></td>
<td>
<p>Either parent 2's genotype, or parent 2's genotype log-likelihoods.</p>
</td></tr>
<tr><td><code id="lrt_men_gl4_+3A_drbound">drbound</code></td>
<td>
<p>The upper bound on the double reduction rate.</p>
</td></tr>
<tr><td><code id="lrt_men_gl4_+3A_pp">pp</code></td>
<td>
<p>Is (partial) preferential pairing possible (<code>TRUE</code>) or
not (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="lrt_men_gl4_+3A_dr">dr</code></td>
<td>
<p>Is double reduction possible (<code>TRUE</code>) or
not (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="lrt_men_gl4_+3A_alpha">alpha</code></td>
<td>
<p>If <code>dr = FALSE</code>, this is the known rate of double
reduction.</p>
</td></tr>
<tr><td><code id="lrt_men_gl4_+3A_xi1">xi1</code></td>
<td>
<p>If <code>pp = FALSE</code>, this is the known preferential pairing
parameter of parent 1.</p>
</td></tr>
<tr><td><code id="lrt_men_gl4_+3A_xi2">xi2</code></td>
<td>
<p>If <code>pp = FALSE</code>, this is the known preferential pairing
parameter of parent 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following elements
</p>

<dl>
<dt><code>statistic</code></dt><dd><p>The log-likelihood ratio test statistic.</p>
</dd>
<dt><code>df</code></dt><dd><p>The degrees of freedom.</p>
</dd>
<dt><code>p_value</code></dt><dd><p>The p-value.</p>
</dd>
<dt><code>alpha</code></dt><dd><p>The estimated double reduction rate.</p>
</dd>
<dt><code>xi1</code></dt><dd><p>The estimated preferential pairing parameter of parent 1.</p>
</dd>
<dt><code>xi2</code></dt><dd><p>The estimated preferential pairing parameter of parent 2.</p>
</dd>
</dl>



<h3>Unidentified parameters</h3>

<p>When <code>g1 = 2</code> or <code>g2 = 2</code> (or both), the model is not identified
and those estimates (<code>alpha</code>, <code>xi1</code>, and <code>xi2</code>) are
meaningless. Do NOT interpret them.
</p>
<p>The estimate of <code>alpha</code> (double reduction rate) IS identified as
long as at least one parent is simplex, and no parent is duplex.
However, the estimates of the double reduction rate have extremely high
variance.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## null simulation
set.seed(1)
g1 &lt;- 2
g2 &lt;- 2
gl &lt;- simf1gl(n = 25, g1 = g1, g2 = g2, alpha = 1/12, xi2 = 1/2)

## LRT when parent genotypes are known.
lrt_men_gl4(gl = gl, g1 = g1, g2 = g2)

## LRT when parent genotypes are not known
lrt_men_gl4(gl = gl)

## Alternative simulation
gl &lt;- simgl(nvec = rep(5, 5))
lrt_men_gl4(gl = gl, g1 = g1, g2 = g2)

</code></pre>

<hr>
<h2 id='multi_lrt'>Parallelized likelihood ratio test for segregation distortion.</h2><span id='topic+multi_lrt'></span>

<h3>Description</h3>

<p>Uses the <code>future</code> package to implement parallelization support for
the likelihood ratio tests for segregation distortion. Right now, this is
only supported for tetraploids (allo, auto, or segmental). This function
is only somewhat tested (the single-locus LRT functions in the &quot;See Also&quot;
section are very well tested). So please send any bugs you notice to
<a href="https://github.com/dcgerard/segtest/issues">https://github.com/dcgerard/segtest/issues</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_lrt(
  g,
  p1,
  p2,
  drbound = 1/6,
  pp = TRUE,
  dr = TRUE,
  alpha = 0,
  xi1 = 1/3,
  xi2 = 1/3,
  nullprop = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multi_lrt_+3A_g">g</code></td>
<td>
<p>One of two inputs
</p>

<ul>
<li><p>A matrix of genotype counts. The rows index the loci and the columns index the genotypes.
</p>
</li>
<li><p>An array of genotype log-likelihoods. The rows index the loci, the columns index the individuals, and the slices index the genotypes. Log-likelihoods are base e (natural log).
</p>
</li></ul>
</td></tr>
<tr><td><code id="multi_lrt_+3A_p1">p1</code></td>
<td>
<p>One of three inputs
</p>

<ul>
<li><p>A vector of parent 1's genotypes.
</p>
</li>
<li><p>A matrix of parent 1's genotype log-likelihoods. The rows index the loci and the columns index the genotypes. Logs are in base e (natural log).
</p>
</li>
<li><p><code>NULL</code> (only supported when using genotype likelihoods for the offspring)
</p>
</li></ul>
</td></tr>
<tr><td><code id="multi_lrt_+3A_p2">p2</code></td>
<td>
<p>One of three inputs
</p>

<ul>
<li><p>A vector of parent 1's genotypes.
</p>
</li>
<li><p>A matrix of parent 1's genotype log-likelihoods. The rows index the loci and the columns index the genotypes. Logs are in base e (natural log).
</p>
</li>
<li><p><code>NULL</code> (only supported when using genotype likelihoods for the offspring)
</p>
</li></ul>
</td></tr>
<tr><td><code id="multi_lrt_+3A_drbound">drbound</code></td>
<td>
<p>The upper bound on the double reduction rate.</p>
</td></tr>
<tr><td><code id="multi_lrt_+3A_pp">pp</code></td>
<td>
<p>Is (partial) preferential pairing possible (<code>TRUE</code>) or
not (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="multi_lrt_+3A_dr">dr</code></td>
<td>
<p>Is double reduction possible (<code>TRUE</code>) or
not (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="multi_lrt_+3A_alpha">alpha</code></td>
<td>
<p>If <code>dr = FALSE</code>, this is the known rate of double
reduction.</p>
</td></tr>
<tr><td><code id="multi_lrt_+3A_xi1">xi1</code></td>
<td>
<p>If <code>pp = FALSE</code>, this is the known preferential pairing
parameter of parent 1.</p>
</td></tr>
<tr><td><code id="multi_lrt_+3A_xi2">xi2</code></td>
<td>
<p>If <code>pp = FALSE</code>, this is the known preferential pairing
parameter of parent 2.</p>
</td></tr>
<tr><td><code id="multi_lrt_+3A_nullprop">nullprop</code></td>
<td>
<p>Should we return the null proportions (<code>TRUE</code>) or not (<code>FALSE</code>)?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the following elements:
</p>

<dl>
<dt><code>statistic</code></dt><dd><p>The likelihood ratio test statistic</p>
</dd>
<dt><code>p_value</code></dt><dd><p>The p-value of the likelihood ratio test.</p>
</dd>
<dt><code>df</code></dt><dd><p>The degrees of freedom of the test.</p>
</dd>
<dt><code>alpha</code></dt><dd><p>The MLE of the double reduction rate. Do not use for real work.</p>
</dd>
<dt><code>xi1</code></dt><dd><p>The MLE of the first parent's partial preferential pairing parameter. Do not use for real work.</p>
</dd>
<dt><code>xi2</code></dt><dd><p>The MLE of the second parent's partial preferential pairing parameter. Do not use for real work.</p>
</dd>
<dt><code>p1</code></dt><dd><p>(Estimate of) the first parent's genotype.</p>
</dd>
<dt><code>p2</code></dt><dd><p>(Estimate of) the second parent's genotype.</p>
</dd>
<dt><code>snp</code></dt><dd><p>The name of the SNP.</p>
</dd>
</dl>



<h3>Parallel Computation</h3>

<p>The <code>multi_lrt()</code> function supports parallel computing. It does
so through the <a href="https://cran.r-project.org/package=future">future</a>
package.
</p>
<p>You first specify the evaluation plan with <code><a href="future.html#topic+plan">plan</a>()</code>
from the <code>future</code> package. On a local machine, this is typically
just <code>future::plan(future::multisession, workers = nc)</code> where
<code>nc</code> is the number of workers you want. You can find the maximum
number of possible workers with <code><a href="future.html#topic+availableCores">availableCores</a>()</code>.
You then run <code>multi_lrt()</code>, then shut down the workers with
<code>future::plan(future::sequential)</code>.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+lrt_men_g4">lrt_men_g4()</a></code> Single locus LRT for segregation distortion when genotypes are known.
</p>
</li>
<li> <p><code><a href="#topic+lrt_men_gl4">lrt_men_gl4()</a></code> Single locus LRT for segregation distortion when using genotype likelihoods.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Assuming genotypes are known (typically a bad idea)
glist &lt;- multidog_to_g(mout = ufit, type = "all_g", p1 = "indigocrisp", p2 = "sweetcrisp")
p1_1 &lt;- glist$p1
p2_1 &lt;- glist$p2
g_1 &lt;- glist$g
multi_lrt(g = g_1, p1 = p1_1, p2 = p2_1)

## Using genotype likelihoods (typically a good idea)
glist &lt;- multidog_to_g(mout = ufit, type = "all_gl", p1 = "indigocrisp", p2 = "sweetcrisp")
p1_2 &lt;- glist$p1
p2_2 &lt;- glist$p2
g_2 &lt;- glist$g
multi_lrt(g = g_2, p1 = p1_2, p2 = p2_2)

## Offspring genotype likelihoods and parent genotypes known
multi_lrt(g = g_2, p1 = p1_1, p2 = p2_1)

## Offspring genotype likelihoods and no information on parent genotypes
multi_lrt(g = g_2, p1 = NULL, p2 = NULL)

## Parallel computing is supported through the future package
future::plan(future::multisession, workers = 2)
multi_lrt(g = g_2, p1 = p1_2, p2 = p2_2)
future::plan(future::sequential)


</code></pre>

<hr>
<h2 id='multidog_to_g'>Converts multidog output to a format usable for multi_lrt()</h2><span id='topic+multidog_to_g'></span>

<h3>Description</h3>

<p>Converts multidog output to a format usable for multi_lrt()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multidog_to_g(
  mout,
  type = c("off_gl", "all_gl", "all_g", "off_g"),
  p1 = NULL,
  p2 = NULL,
  ploidy = 4
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multidog_to_g_+3A_mout">mout</code></td>
<td>
<p>The output of <code><a href="updog.html#topic+multidog">multidog</a>()</code>.</p>
</td></tr>
<tr><td><code id="multidog_to_g_+3A_type">type</code></td>
<td>

<dl>
<dt><code>"off_gl"</code></dt><dd><p>Genotype likelihoods of offspring but not parents. This is the typical choice if you used the &quot;f1&quot; or &quot;f1pp&quot; options when genotyping.</p>
</dd>
<dt><code>"all_gl"</code></dt><dd><p>Genotype likelihoods of offspring and parents. This is only done if you did <em>not</em> use the &quot;f1&quot; or &quot;f1pp&quot; options when genotyping. If this is the case, then you need to specify which individuals are the parents.</p>
</dd>
<dt><code>"off_g"</code></dt><dd><p>Genotypes, assuming that they are known. You used the &quot;f1&quot; or &quot;f1pp&quot; option when genotyping.</p>
</dd>
<dt><code>"all_g"</code></dt><dd><p>Genotypes, assuming that they are known. You did <em>not</em> use the &quot;f1&quot; or &quot;f1pp&quot; option when genotyping. If this is the case, then you need to specify which individuals are the parents</p>
</dd></dl>
<p>.
</p>
</td></tr>
<tr><td><code id="multidog_to_g_+3A_p1">p1</code></td>
<td>
<p>The first parent name if using <code>type = "all_gl"</code> or <code>type = "all_g"</code>.</p>
</td></tr>
<tr><td><code id="multidog_to_g_+3A_p2">p2</code></td>
<td>
<p>The second parent name if using <code>type = "all_gl"</code> or <code>type = "all_g"</code>.</p>
</td></tr>
<tr><td><code id="multidog_to_g_+3A_ploidy">ploidy</code></td>
<td>
<p>The ploidy. Note that most methods in this package
(including those in <code><a href="#topic+multi_lrt">multi_lrt</a>()</code>) assume that the ploidy
is 4. But we allow for arbitrary ploidy in this function since it
might be useful in the future.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following elements
</p>

<dl>
<dt><code>g</code></dt><dd><p>Either a matrix of counts, where the columns index the genotype
and the rows index the loci (<code>type = "all_g"</code> or
<code>type = "off_g"</code>). Or an array of genotype (natural) log-likelihoods
where the rows index the loci, the columns index the
individuals, and the slices index the genotypes
(<code>type = "all_gl"</code> or <code>type = "off_gl"</code>).</p>
</dd>
<dt><code>p1</code></dt><dd><p>Either a vector of known parental genotypes
(<code>type = "off_gl"</code>, <code>type = "all_g"</code> or <code>type = "off_g"</code>).
Or a matrix of genotype (natural) log-likelihoods where the
rows index the loci and the columns index the genotypes
(<code>type = "all_gl"</code>).</p>
</dd>
<dt><code>p2</code></dt><dd><p>Either a vector of known parental genotypes
(<code>type = "off_gl"</code>, <code>type = "all_g"</code> or <code>type = "off_g"</code>).
Or a matrix of genotype (natural) log-likelihoods where the
rows index the loci and the columns index the genotypes
(<code>type = "all_gl"</code>).</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>multidog_to_g(mout = ufit, type = "all_g", p1 = "indigocrisp", p2 = "sweetcrisp")
multidog_to_g(mout = ufit, type = "all_gl", p1 = "indigocrisp", p2 = "sweetcrisp")
multidog_to_g(mout = ufit2, type = "off_g")
multidog_to_g(mout = ufit2, type = "off_gl")
multidog_to_g(mout = ufit3, type = "off_g")
multidog_to_g(mout = ufit3, type = "off_gl")


</code></pre>

<hr>
<h2 id='nextElem.arrayiter'>Next element in an array</h2><span id='topic+nextElem.arrayiter'></span>

<h3>Description</h3>

<p>This is applied to an <code>arrayiter</code> object to obtain the next sub-array
along one of the dimensions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'arrayiter'
nextElem(obj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nextElem.arrayiter_+3A_obj">obj</code></td>
<td>
<p>An arrayiter object</p>
</td></tr>
<tr><td><code id="nextElem.arrayiter_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The next sub-array.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+iter.array">iter.array()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>glist &lt;- multidog_to_g(mout = ufit, type = "all_gl", p1 = "indigocrisp", p2 = "sweetcrisp")
g &lt;- iterators::iter(glist$g, by = 3)
head(iterators::nextElem(g))
head(iterators::nextElem(g))
head(iterators::nextElem(g))

</code></pre>

<hr>
<h2 id='offspring_geno'>Simulates genotypes given genotype frequencies.</h2><span id='topic+offspring_geno'></span>

<h3>Description</h3>

<p>Takes as input the offspring
genotype frequencies and a sample size and returns simulated genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>offspring_geno(gf, n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="offspring_geno_+3A_gf">gf</code></td>
<td>
<p>Vector of offspring genotype frequencies</p>
</td></tr>
<tr><td><code id="offspring_geno_+3A_n">n</code></td>
<td>
<p>Sample size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Simulated genotypes
</p>


<h3>Author(s)</h3>

<p>Mira Thakkar
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
gf &lt;- offspring_gf_2(alpha = 1/6, xi1 = 1/3, xi2 = 1/3, p1 = 2, p2 = 3)
offspring_geno(gf = gf, n = 10)

</code></pre>

<hr>
<h2 id='offspring_gf_2'>Calculates offspring genotype frequencies under the two-parameter model.</h2><span id='topic+offspring_gf_2'></span>

<h3>Description</h3>

<p>Calculates offspring genotype frequencies under the two-parameter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>offspring_gf_2(alpha, xi1, xi2 = xi1, p1, p2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="offspring_gf_2_+3A_alpha">alpha</code></td>
<td>
<p>The double reduction rate</p>
</td></tr>
<tr><td><code id="offspring_gf_2_+3A_xi1">xi1</code></td>
<td>
<p>The preferential pairing parameter of the first parent.</p>
</td></tr>
<tr><td><code id="offspring_gf_2_+3A_xi2">xi2</code></td>
<td>
<p>The preferential pairing parameter of the second parent.</p>
</td></tr>
<tr><td><code id="offspring_gf_2_+3A_p1">p1</code></td>
<td>
<p>The first parent's genotype</p>
</td></tr>
<tr><td><code id="offspring_gf_2_+3A_p2">p2</code></td>
<td>
<p>The second parent's genotype</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Offspring genotype frequencies
</p>


<h3>Author(s)</h3>

<p>Mira Thakkar
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha &lt;- 1/6
xi1 &lt;- 1/3
xi2 &lt;- 1/3
p1 &lt;- 2
p2 &lt;- 3
offspring_gf_2(alpha = alpha, xi1 = xi1, xi2 = xi2, p1 = p1, p2 = p2)

</code></pre>

<hr>
<h2 id='offspring_gf_3'>Calculates offspring genotype frequencies under the three-parameter model.</h2><span id='topic+offspring_gf_3'></span>

<h3>Description</h3>

<p>Calculates offspring genotype frequencies under the three-parameter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>offspring_gf_3(tau, beta, gamma1, gamma2 = gamma1, p1, p2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="offspring_gf_3_+3A_tau">tau</code></td>
<td>
<p>Probability of quadrivalent formation</p>
</td></tr>
<tr><td><code id="offspring_gf_3_+3A_beta">beta</code></td>
<td>
<p>Probability of double reduction given quadrivalent formation</p>
</td></tr>
<tr><td><code id="offspring_gf_3_+3A_gamma1">gamma1</code></td>
<td>
<p>Probability of AA_aa pairing in parent 1</p>
</td></tr>
<tr><td><code id="offspring_gf_3_+3A_gamma2">gamma2</code></td>
<td>
<p>Probability of AA_aa pairing in parent 2</p>
</td></tr>
<tr><td><code id="offspring_gf_3_+3A_p1">p1</code></td>
<td>
<p>The first parent's genotype</p>
</td></tr>
<tr><td><code id="offspring_gf_3_+3A_p2">p2</code></td>
<td>
<p>The second parent's genotype</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Offspring genotype frequencies
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>offspring_gf_3(
  tau = 1/2,
  beta = 1/6,
  gamma1 = 1/3,
  gamma2 = 1/3,
  p1 = 1,
  p2 = 2)

</code></pre>

<hr>
<h2 id='otest_g'>Jointly tests for segregation distortion and number of incompatible genotypes</h2><span id='topic+otest_g'></span>

<h3>Description</h3>

<p>This is experimental. I haven't tested it out in lots of scenarios yet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>otest_g(
  x,
  g1,
  g2,
  pbad = 0.03,
  drbound = 1/6,
  pp = TRUE,
  dr = TRUE,
  alpha = 0,
  xi1 = 1/3,
  xi2 = 1/3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="otest_g_+3A_x">x</code></td>
<td>
<p>A vector of genotype counts. <code>x[i]</code> is the number of
offspring with genotype <code>i-1</code>.</p>
</td></tr>
<tr><td><code id="otest_g_+3A_g1">g1</code></td>
<td>
<p>The genotype of parent 1.</p>
</td></tr>
<tr><td><code id="otest_g_+3A_g2">g2</code></td>
<td>
<p>The genotype of parent 2.</p>
</td></tr>
<tr><td><code id="otest_g_+3A_pbad">pbad</code></td>
<td>
<p>The upper bound on the number of bad genotypes</p>
</td></tr>
<tr><td><code id="otest_g_+3A_drbound">drbound</code></td>
<td>
<p>The maximum rate of double reduction. A default of 1/6
is provided, which is the rate under the complete equational
segregation model of meiosis.</p>
</td></tr>
<tr><td><code id="otest_g_+3A_pp">pp</code></td>
<td>
<p>A logical. Should we account for preferential pairing
(<code>TRUE</code>) or not (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="otest_g_+3A_dr">dr</code></td>
<td>
<p>A logical. Should we account for double reduction
(<code>TRUE</code>) or not (<code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="otest_g_+3A_alpha">alpha</code></td>
<td>
<p>If <code>dr = FALSE</code>, this is the known rate of double
reduction.</p>
</td></tr>
<tr><td><code id="otest_g_+3A_xi1">xi1</code></td>
<td>
<p>If <code>pp = FALSE</code>, this is the known preferential pairing
parameter of parent 1.</p>
</td></tr>
<tr><td><code id="otest_g_+3A_xi2">xi2</code></td>
<td>
<p>If <code>pp = FALSE</code>, this is the known preferential pairing
parameter of parent 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Here, we test if the compatible genotypes are consistent with F1 populations
and separately test that the number of incompatible genotypes isn't too
large (less than 3 percent by default). This is the strategy the
polymapR software uses. But we use a Bonferroni correction to combine
these tests (minimum of two times the p-values), while they just multiply
the p-values together. So our approach accounts for double reduction and
preferential pairing, while also controlling the family-wise error rate.
</p>


<h3>Value</h3>

<p>A list with the following elements
</p>

<dl>
<dt><code>statistic</code></dt><dd><p>The log-likelihood ratio test statistic.</p>
</dd>
<dt><code>df</code></dt><dd><p>The degrees of freedom.</p>
</dd>
<dt><code>p_value</code></dt><dd><p>The Bonferroni corrected p-value.</p>
</dd>
<dt><code>p_lrt</code></dt><dd><p>The p-value of the LRT.</p>
</dd>
<dt><code>p_binom</code></dt><dd><p>The p-value of the one-sided binomial test.</p>
</dd>
<dt><code>alpha</code></dt><dd><p>The estimated double reduction rate.</p>
</dd>
<dt><code>xi1</code></dt><dd><p>The estimated preferential pairing parameter of parent 1.</p>
</dd>
<dt><code>xi2</code></dt><dd><p>The estimated preferential pairing parameter of parent 2.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Run a test where genotypes 0, 1, and 2 are possible
x &lt;- c(10, 10, 4, 0, 5)
otest_g(x = x, g1 = 1, g2 = 0)

# polymapR's multiplication and the Bonferroni differ
df &lt;- expand.grid(p1 = seq(0, 1, length.out = 20), p2 = seq(0, 1, length.out = 20))
df$polymapr &lt;- NA
df$bonferroni &lt;- NA
for (i in seq_len(nrow(df))) {
  df$polymapr[[i]] &lt;- df$p1[[i]] * df$p2[[i]]
  df$bonferroni[[i]] &lt;- 2 * min(c(df$p1[[i]], df$p2[[i]], 0.5))
}
graphics::plot(df$polymapr, df$bonferroni)


</code></pre>

<hr>
<h2 id='po_gl'>Generate genotype likelihoods from offspring genotypes.</h2><span id='topic+po_gl'></span>

<h3>Description</h3>

<p>Takes as input (i) the parent genotypes,
(ii) the offspring genotype frequency, (iii) sequencing error rate, (iv) read
depth, (v) bias, and (vi) overdispersion. It returns genotype likelihoods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>po_gl(
  genovec,
  ploidy,
  p1_geno = NULL,
  p2_geno = NULL,
  rd = 10,
  seq = 0.01,
  bias = 1,
  od = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="po_gl_+3A_genovec">genovec</code></td>
<td>
<p>Offspring genotypes. <code>genovec[i]</code> is the dosage for individual i.</p>
</td></tr>
<tr><td><code id="po_gl_+3A_ploidy">ploidy</code></td>
<td>
<p>Ploidy</p>
</td></tr>
<tr><td><code id="po_gl_+3A_p1_geno">p1_geno</code></td>
<td>
<p>Parent 1 genotype</p>
</td></tr>
<tr><td><code id="po_gl_+3A_p2_geno">p2_geno</code></td>
<td>
<p>Parent 2 genotype</p>
</td></tr>
<tr><td><code id="po_gl_+3A_rd">rd</code></td>
<td>
<p>Read depth. Lower is more uncertain.</p>
</td></tr>
<tr><td><code id="po_gl_+3A_seq">seq</code></td>
<td>
<p>Sequencing error rate. Higher means more uncertain.</p>
</td></tr>
<tr><td><code id="po_gl_+3A_bias">bias</code></td>
<td>
<p>Bias. 1 means no bias.</p>
</td></tr>
<tr><td><code id="po_gl_+3A_od">od</code></td>
<td>
<p>Overdispersion. Typical value is like 0.01. Higher means more uncertain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype likelihoods
</p>


<h3>Author(s)</h3>

<p>Mira Thakkar
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
po_gl(genovec = c(1, 2, 1, 1, 3), p1_geno = 2, p2_geno = 2, ploidy = 4)

</code></pre>

<hr>
<h2 id='polymapr_test'>Run segregation distortion tests as implemented in the polymapR package.</h2><span id='topic+polymapr_test'></span>

<h3>Description</h3>

<p>The polymapR package tests for segregation distortion by iterating through all
possible forms of disomic or polysomic inheritance from either parent,
tests for concordance of the offspring genotypes using a chi-squared
test, and returns the largest p-value. It sometimes chooses a different
p-value based on other heuristics. They also sometimes return NA.
When <code>type = "segtest"</code>, we only look at patterns of the
given parent genotypes, choosing the largest p-value. When
<code>type = "polymapR"</code>, we return what they use via their heuristics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polymapr_test(x, g1 = NULL, g2 = NULL, type = c("segtest", "polymapR"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="polymapr_test_+3A_x">x</code></td>
<td>
<p>Either a vector of genotype counts, or a matrix of genotype
posteriors where the rows index the individuals and the columns
index the genotypes.</p>
</td></tr>
<tr><td><code id="polymapr_test_+3A_g1">g1</code></td>
<td>
<p>Parent 1's genotype.</p>
</td></tr>
<tr><td><code id="polymapr_test_+3A_g2">g2</code></td>
<td>
<p>Parent 2's genotype.</p>
</td></tr>
<tr><td><code id="polymapr_test_+3A_type">type</code></td>
<td>
<p>Either my implementation which approximates that of
polymapR (<code>"segtest"</code>) or the implementation
through polymapR (<code>"polymapR"</code>). Note that
polymapR needs to be installed for <code>type = "polymapR"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<dl>
<dt>p_value</dt><dd><p>The p-value of the test.</p>
</dd>
<dt>bestfit</dt><dd><p>The best fit model, using the same notation as in
<code><a href="polymapR.html#topic+checkF1">checkF1</a>()</code>.</p>
</dd>
<dt>frq_invalid</dt><dd><p>The frequency of invalid genotypes.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>See Also</h3>

<p><code><a href="polymapR.html#topic+checkF1">checkF1</a>()</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g1 &lt;- 0
g2 &lt;- 1
x &lt;- c(4, 16, 0, 0, 0)
polymapr_test(x = x, g1 = g1, g2 = g2, type = "segtest")



</code></pre>

<hr>
<h2 id='pvec_tet_2'>Tetraploid gamete frequencies of gametes when one parent's genotype is known</h2><span id='topic+pvec_tet_2'></span>

<h3>Description</h3>

<p>This is under the two parameter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvec_tet_2(alpha, xi, ell)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pvec_tet_2_+3A_alpha">alpha</code></td>
<td>
<p>The double reduction rate</p>
</td></tr>
<tr><td><code id="pvec_tet_2_+3A_xi">xi</code></td>
<td>
<p>The preferential pairing parameter</p>
</td></tr>
<tr><td><code id="pvec_tet_2_+3A_ell">ell</code></td>
<td>
<p>The parental genotype</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The gamete genotype frequencies
</p>


<h3>Author(s)</h3>

<p>Mira Thakkar and David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>alpha &lt;- 1/6
xi &lt;- 1/3
pvec_tet_2(alpha = alpha, xi = xi, ell = 0)
pvec_tet_2(alpha = alpha, xi = xi, ell = 1)
pvec_tet_2(alpha = alpha, xi = xi, ell = 2)
pvec_tet_2(alpha = alpha, xi = xi, ell = 3)
pvec_tet_2(alpha = alpha, xi = xi, ell = 4)

</code></pre>

<hr>
<h2 id='pvec_tet_3'>Tetraploid gamete frequencies of gametes when one parent's genotype is known</h2><span id='topic+pvec_tet_3'></span>

<h3>Description</h3>

<p>This is under the three parameter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvec_tet_3(tau, beta, gamma, ell)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pvec_tet_3_+3A_tau">tau</code></td>
<td>
<p>Probability of quadrivalent formation</p>
</td></tr>
<tr><td><code id="pvec_tet_3_+3A_beta">beta</code></td>
<td>
<p>Probability of double reduction given quadrivalent formation</p>
</td></tr>
<tr><td><code id="pvec_tet_3_+3A_gamma">gamma</code></td>
<td>
<p>Probability of AA/aa pairing given bivalent formation</p>
</td></tr>
<tr><td><code id="pvec_tet_3_+3A_ell">ell</code></td>
<td>
<p>The parent genotype</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The gamete genotype frequencies
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pvec_tet_3(tau = 0.5, beta = 0.1, gamma = 0.5, ell = 2)

</code></pre>

<hr>
<h2 id='simf1g'>Simulate genotype counts from F1 individuals</h2><span id='topic+simf1g'></span>

<h3>Description</h3>

<p>Simulate genotype counts from F1 individuals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simf1g(n, g1, g2, alpha = 0, xi1 = 1/3, xi2 = 1/3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simf1g_+3A_n">n</code></td>
<td>
<p>Sample size.</p>
</td></tr>
<tr><td><code id="simf1g_+3A_g1">g1</code></td>
<td>
<p>The first parent's genotype.</p>
</td></tr>
<tr><td><code id="simf1g_+3A_g2">g2</code></td>
<td>
<p>The second parent's genotype.</p>
</td></tr>
<tr><td><code id="simf1g_+3A_alpha">alpha</code></td>
<td>
<p>The double reduction rate.</p>
</td></tr>
<tr><td><code id="simf1g_+3A_xi1">xi1</code></td>
<td>
<p>The first parent's preferential pairing parameter.</p>
</td></tr>
<tr><td><code id="simf1g_+3A_xi2">xi2</code></td>
<td>
<p>The second parent's preferential pairing parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of counts, where element <code>i</code> is the number of
simulated individuals with genotype <code>i-1</code>.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
simf1g(n = 10, g1 = 1, g2 = 2)

</code></pre>

<hr>
<h2 id='simf1gl'>Simulate genotype likelihoods of F1 individuals.</h2><span id='topic+simf1gl'></span>

<h3>Description</h3>

<p>Simulate genotype likelihoods of F1 individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simf1gl(n, g1, g2, rd = 10, alpha = 0, xi1 = 1/3, xi2 = 1/3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simf1gl_+3A_n">n</code></td>
<td>
<p>Sample size.</p>
</td></tr>
<tr><td><code id="simf1gl_+3A_g1">g1</code></td>
<td>
<p>The first parent's genotype.</p>
</td></tr>
<tr><td><code id="simf1gl_+3A_g2">g2</code></td>
<td>
<p>The second parent's genotype.</p>
</td></tr>
<tr><td><code id="simf1gl_+3A_rd">rd</code></td>
<td>
<p>The read depth.</p>
</td></tr>
<tr><td><code id="simf1gl_+3A_alpha">alpha</code></td>
<td>
<p>The double reduction rate.</p>
</td></tr>
<tr><td><code id="simf1gl_+3A_xi1">xi1</code></td>
<td>
<p>The first parent's preferential pairing parameter.</p>
</td></tr>
<tr><td><code id="simf1gl_+3A_xi2">xi2</code></td>
<td>
<p>The second parent's preferential pairing parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The matrix of offspring genotype log-likelihoods.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
simf1gl(n = 10, g1 = 1, g2 = 2)

</code></pre>

<hr>
<h2 id='simgl'>Simulate genotype likelihoods from genotype counts</h2><span id='topic+simgl'></span>

<h3>Description</h3>

<p>Provide a vector of genotype counts and this will return a matrix of
genotype log-likelihoods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simgl(nvec, rd = 10, seq = 0.01, bias = 1, od = 0.01)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simgl_+3A_nvec">nvec</code></td>
<td>
<p>A vector of counts. <code>nvec[k]</code> is the number of folks with a genotype of k-1.</p>
</td></tr>
<tr><td><code id="simgl_+3A_rd">rd</code></td>
<td>
<p>Read depth. Lower is more uncertain.</p>
</td></tr>
<tr><td><code id="simgl_+3A_seq">seq</code></td>
<td>
<p>Sequencing error rate. Higher means more uncertain.</p>
</td></tr>
<tr><td><code id="simgl_+3A_bias">bias</code></td>
<td>
<p>Bias. 1 means no bias.</p>
</td></tr>
<tr><td><code id="simgl_+3A_od">od</code></td>
<td>
<p>Overdispersion. Typical value is like 0.01. Higher means more uncertain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of genotype log-likelihoods. The rows index the
individuals and the columns index the genotypes. This is natural
log (base e).
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
simgl(nvec = c(1, 2, 1, 1, 3))


</code></pre>

<hr>
<h2 id='three_to_two'>Convert from three parameters to two parameters</h2><span id='topic+three_to_two'></span>

<h3>Description</h3>

<p>Convert from three parameters to two parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>three_to_two(tau, beta, gamma)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="three_to_two_+3A_tau">tau</code></td>
<td>
<p>Probability of quadrivalent formation</p>
</td></tr>
<tr><td><code id="three_to_two_+3A_beta">beta</code></td>
<td>
<p>Probability of double reduction given quadrivalent formation</p>
</td></tr>
<tr><td><code id="three_to_two_+3A_gamma">gamma</code></td>
<td>
<p>Probability of AA/aa pairing given bivalent formation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of length two. The first is the double reduction rate
(<code>alpha</code>), and the second is the preferential pairing parameter
(<code>xi</code>).
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>three_to_two(tau = 0.1, beta = 1/6, gamma = 1/4)

</code></pre>

<hr>
<h2 id='ufit'>Genotype data from Cappai et al. (2020)</h2><span id='topic+ufit'></span><span id='topic+ufit2'></span><span id='topic+ufit3'></span>

<h3>Description</h3>

<p>A subset of data from Cappai et al. (2020), fit using
<code><a href="updog.html#topic+multidog">multidog</a>()</code>. This just contains a random set of 10 loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ufit

ufit2

ufit3
</code></pre>


<h3>Format</h3>

<p>An object of type <code>multidog</code> output from <code><a href="updog.html#topic+multidog">multidog</a>()</code>.
</p>

<dl>
<dt><code>ufit</code></dt><dd><p>Uses the <code>model = "norm"</code> option.</p>
</dd>
<dt><code>ufit2</code></dt><dd><p>Uses the <code>model = "f1pp"</code> option.</p>
</dd>
<dt><code>ufit3</code></dt><dd><p>Uses the <code>model = "f1"</code> option.</p>
</dd>
</dl>

<p>An object of class <code>multidog</code> of length 2.
</p>
<p>An object of class <code>multidog</code> of length 2.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.5281/zenodo.13715703">doi:10.5281/zenodo.13715703</a>
</p>


<h3>References</h3>


<ul>
<li><p>Cappai, F., Amadeu, R. R., Benevenuto, J., Cullen, R., Garcia, A., Grossman, A., Ferro, L., &amp; Munoz, P. (2020). High-resolution linkage map and QTL analyses of fruit firmness in autotetraploid blueberry. <em>Frontiers in plant science</em>, 11, 562171. <a href="https://doi.org/10.3389/fpls.2020.562171">doi:10.3389/fpls.2020.562171</a>.
</p>
</li></ul>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
