<!DOCTYPE html><html><head><title>Help for package selfingTree</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {selfingTree}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#buildSelfingTree'><p>Builds the probability tree</p></a></li>
<li><a href='#evalProb'><p>Evaluates the genotype probability</p></a></li>
<li><a href='#extractProbs'><p>Extracts conditional genotype probabilities</p></a></li>
<li><a href='#F4'><p>Genotypes of one million simulated F4 lines.</p></a></li>
<li><a href='#genSubtree'><p>Generates a sub-tree</p></a></li>
<li><a href='#getTargets'><p>Obtain all possible genotypes that match a certain target</p>
configuration</a></li>
<li><a href='#haploProb'><p>Probability of observing the target haplotype given the parental genotype.</p></a></li>
<li><a href='#map'><p>Genetic map of the three markers in the F4 data set.</p></a></li>
<li><a href='#nodeProbabilities'><p>Multiplies conditional probabilities along all branches of the tree</p></a></li>
<li><a href='#selfingTree-package'>
<p>Genotype Probabilities in Intermediate Generations of Inbreeding Through Selfing</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genotype Probabilities in Intermediate Generations of Inbreeding
Through Selfing</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-3-Clause">BSD_3_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>Copyright:</td>
<td>(c) 2014, Pioneer Hi-Bred International, Inc.</td>
</tr>
<tr>
<td>Date:</td>
<td>2014-12-18</td>
</tr>
<tr>
<td>Author:</td>
<td>Frank Technow [aut, cre] (Pioneer Hi-Bred International, Inc.,
    Johnston, Iowa)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Frank Technow &lt;Frank.Technow@pioneer.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.1),foreach</td>
</tr>
<tr>
<td>Description:</td>
<td>A probability tree allows to compute probabilities of
	     complex events, such as genotype probabilities in intermediate generations of inbreeding
	     through recurrent self-fertilization (selfing). This package implements functionality to compute
	     probability trees for two- and three-marker genotypes in the F2 to F7 selfing
	     generations. The conditional probabilities are derived automatically
	     and in symbolic form. The package also provides functionality to
	     extract and evaluate the relevant probabilities.</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2014-12-18 15:43:52 UTC; frank</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2014-12-18 17:31:18</td>
</tr>
</table>
<hr>
<h2 id='buildSelfingTree'>Builds the probability tree</h2><span id='topic+buildSelfingTree'></span>

<h3>Description</h3>

<p>This function builds the probability tree for recurrent
selfing.</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildSelfingTree(genF,generation,gam1,gam2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="buildSelfingTree_+3A_genf">genF</code></td>
<td>
<p>A function that generates a sub-tree of all possible
genotypes given a parental genotype, either <code>genSubtree.2M</code> or
<code>genSubtree.3M</code>.</p>
</td></tr>
<tr><td><code id="buildSelfingTree_+3A_generation">generation</code></td>
<td>
<p>Integer giving the selfing generation to which the
tree will be build. Values can range from 2 to 7, e.g., the F2
generation build by default and the highest possible generation is
currently the F7.</p>
</td></tr>
<tr><td><code id="buildSelfingTree_+3A_gam1">gam1</code>, <code id="buildSelfingTree_+3A_gam2">gam2</code></td>
<td>
<p>Three (three marker genotypes) or two (two marker
genotypes) character string with the configuration of gametes one and
two of the parental F1 genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A recursive data type in the form of a nested <code>list</code>. Each
element is a <code>list</code> with three elements. Element <code>[[1]]</code>
holds the genotype configuration as &quot;gam1-gam2&quot; (e.g.,
<code>"ABA-BAB"</code>), element <code>[[2]]</code> the symbolic formula
representing the probability of observing this genotype given the
parental genotype and element <code>[[3]]</code> is again a list containing
the sub-tree rooted at this genotype.</p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## F2 and F3 genotypes
F.2M &lt;- buildSelfingTree(genSubtree.2M,3,"AA","BB")

## F2 and F3 genotypes
F.3M &lt;- buildSelfingTree(genSubtree.3M,3,"AAA","BBB")

</code></pre>

<hr>
<h2 id='evalProb'>Evaluates the genotype probability</h2><span id='topic+evalProb'></span>

<h3>Description</h3>

<p>This function symbolically sums the marginal probabilities
of relevant nodes and evaluates them with user specified values for the
recombination frequencies.</p>


<h3>Usage</h3>

<pre><code class='language-R'>evalProb(node.prob, x = 0, y = 0, z = 0, chunk.size = min(length(node.prob),75))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evalProb_+3A_node.prob">node.prob</code></td>
<td>
<p>Character vector with symbolic marginal node
probabilities, i.e., a (subset of) an element of the list returned by
function <code>nodeProbabilities</code>.</p>
</td></tr>
<tr><td><code id="evalProb_+3A_x">x</code>, <code id="evalProb_+3A_y">y</code>, <code id="evalProb_+3A_z">z</code></td>
<td>
<p>Recombination frequencies. For three-marker genotypes,
<code>x</code> is the recombination frequency between markers 1 and 2 and
<code>y</code> that between markers 2 and 3. For two-marker genotypes,
<code>z</code> is recombination frequency between markers 1 and 2.</p>
</td></tr>
<tr><td><code id="evalProb_+3A_chunk.size">chunk.size</code></td>
<td>
<p><code>node.prob</code> is split into several parts of size
equal to <code>chunk.size</code> and summation done within each chunk first
and then across chunks.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The genotype probability (<code>numeric</code>).</p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
evalProb(extractProbs(genSubtree.3M("BAA","AAB")),x = 0.123,y = 0.344)

</code></pre>

<hr>
<h2 id='extractProbs'>Extracts conditional genotype probabilities</h2><span id='topic+extractProbs'></span>

<h3>Description</h3>

<p>This function extracts the symbolic formulas for the
conditional genotype probabilities from the uppermost level of the
(sub)tree.</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractProbs(F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractProbs_+3A_f">F</code></td>
<td>
<p>A sub-tree in the format generated by function
<code>genSubtree.2M</code> or <code>genSubtree.3M</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector with the symbolic formulas. For three-marker
genotypes, symbol <code>x</code> is the recombination frequency between
markers 1 and 2 and <code>y</code> that between markers 2 and 3. For
two-marker genotypes, symbol <code>z</code> is the recombination frequency
between markers 1 and 2.  The names of the elements indicate the
allelic configuration of the two gametes comprising the genotype as
<code>gamete1-gamete2</code> (e.g., <code>"AAB-AAA"</code>). The elements sum to
1.</p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
probs.2M &lt;- extractProbs(genSubtree.2M("BA","AA"))
probs.3M &lt;- extractProbs(genSubtree.3M("BAA","AAB"))

## must sum to 1
stopifnot(all.equal(evalProb(probs.2M, z = 0.044),1))
stopifnot(all.equal(evalProb(probs.3M, x = 0.123, y = 0.344),1))

</code></pre>

<hr>
<h2 id='F4'>Genotypes of one million simulated F4 lines.</h2><span id='topic+F4'></span>

<h3>Description</h3>

<p>These are three-marker genotypes of one million F4 lines
from a cross between parent A and B, simulated using <span class="rlang"><b>R</b></span> package
<span class="pkg">hypred</span>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>F4</code></pre>


<h3>Format</h3>

<p>A character matrix with one million rows and three columns.
Homozygousity for parents A or B is coded as &quot;A&quot; and &quot;B&quot;,
respectively. Heterozygousity as &quot;H&quot;.</p>


<h3>References</h3>

<p>Frank Technow(2013). hypred: Simulation of Genomic Data in Applied
Genetics. R package version 0.4.
</p>

<hr>
<h2 id='genSubtree'>Generates a sub-tree</h2><span id='topic+genSubtree.2M'></span><span id='topic+genSubtree.3M'></span>

<h3>Description</h3>

<p>These functions generate sub-trees consisting of all
genotypes (and their conditional probabilities) that can result after
selfing the parental genotype.</p>


<h3>Usage</h3>

<pre><code class='language-R'>genSubtree.2M(gam1,gam2) ## two-marker genotypes
genSubtree.3M(gam1,gam2) ## three-marker genotypes
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genSubtree_+3A_gam1">gam1</code>, <code id="genSubtree_+3A_gam2">gam2</code></td>
<td>
<p>Three (three-marker genotypes) or two (two-marker
genotypes) character string with the configuration of gamete one and two of
the parental genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>list</code> with one element per possible genotype. Each
element is itself a <code>list</code> with two elements. Element
<code>[[1]]</code> holds the genotype configuration as &quot;gam1-gam2&quot; (e.g.,
<code>"ABA-BAB"</code>), element <code>[[1]]</code> the symbolic formula
representing the probability of observing this genotype given the
parental genotype.</p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genSubtree.2M("AB","AA")
genSubtree.3M("ABA","AAA")
</code></pre>

<hr>
<h2 id='getTargets'>Obtain all possible genotypes that match a certain target
configuration</h2><span id='topic+getTargets'></span>

<h3>Description</h3>

<p>This convenience function finds all genotypes that match a
certain target configuration. It is used only if the target
configuration contains heterozygous states, but order (e.g., A/B or
B/A) does not matter.</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTargets(target.geno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTargets_+3A_target.geno">target.geno</code></td>
<td>
<p>Three (three-marker genotypes) or two (two-marker
genotypes) character string specifying the target
configuration. Homozygousity for parent A allele is indicated as
<code>A</code>, homozygousity for parent B allele as <code>B</code>,
heterozygousity (A/B or B/A) as <code>H</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector with all genotypes matching the target
configuration. The format complies with the output format of
<code>branchProbabilities</code> (<code>gamete1-gamete2</code>, e.g., <code>"AAB-AAA"</code>)</p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
getTargets("AHB")

</code></pre>

<hr>
<h2 id='haploProb'>Probability of observing the target haplotype given the parental genotype.</h2><span id='topic+haploProb.2M'></span><span id='topic+haploProb.3M'></span>

<h3>Description</h3>

<p>These functions derive the symbolic formula for the
probability of observing the target haplotype given the parental
genotype.</p>


<h3>Usage</h3>

<pre><code class='language-R'>haploProb.2M(gam1,gam2,target) ## two-marker genotypes
haploProb.3M(gam1,gam2,target) ## three-marker genotypes
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haploProb_+3A_gam1">gam1</code>, <code id="haploProb_+3A_gam2">gam2</code></td>
<td>
<p>Three (three-marker genotypes) or two (two-marker
genotypes) character string with the configuration of gamete one and
two of the parental genotype.</p>
</td></tr>
<tr><td><code id="haploProb_+3A_target">target</code></td>
<td>
<p>Three (three-marker genotypes) or two (two-marker
genotypes) character string with the configuration of the target
haplotype.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The idea behind the algorithm is to conceptually &quot;recode&quot; the
alleles of the parental genotype into &quot;target&quot; and &quot;non-target&quot;,
where &quot;target&quot; is relative to the target haplotype. Then the rules are
determined that would rearrange the gametes of the parental genotype into a
&quot;target-target-target&quot; haplotype. These rearrangement rules are then
translated into the symbolic formula.</p>


<h3>Value</h3>

<p>A character string with the symbolic formula. For three-marker
genotypes, <code>x</code> is the recombination frequency between markers 1
and 2 and <code>y</code> that between markers 2 and 3. For two-marker
genotypes, <code>z</code> is the recombination frequency between markers 1 and
2.</p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>haploProb.2M("AA","BB","AB")
haploProb.3M("AAA","BBB","ABA")
</code></pre>

<hr>
<h2 id='map'>Genetic map of the three markers in the F4 data set.</h2><span id='topic+map'></span>

<h3>Description</h3>

<p>Genetic map of the three markers in the F4 data set. The
unit is Morgan. This map can be used to compute the recombination
frequencies between the markers using the inverse of the Haldane
mapping function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>map</code></pre>


<h3>Format</h3>

<p>A numeric vector with three elements (<code>c(0.00, 0.05, 0.20)</code>).</p>


<h3>References</h3>

<p>Frank Technow(2013). hypred: Simulation of Genomic Data in Applied
Genetics. R package version 0.4.
</p>

<hr>
<h2 id='nodeProbabilities'>Multiplies conditional probabilities along all branches of the tree</h2><span id='topic+nodeProbabilities'></span>

<h3>Description</h3>

<p>This function generates the symbolic formulas representing
the marginal node probabilities.</p>


<h3>Usage</h3>

<pre><code class='language-R'>nodeProbabilities(F,generation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nodeProbabilities_+3A_f">F</code></td>
<td>
<p>A recurrent selfing tree, as generated by function
<code>buildSelfingTree</code>.</p>
</td></tr>
<tr><td><code id="nodeProbabilities_+3A_generation">generation</code></td>
<td>
<p>Integer giving the highest selfing generation contained
in <code>F</code>. Values can range from 2 to 7, e.g., the F2 generation
build by default and the highest possible generation is currently the
F7.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each formula represents the marginal probability of a
particular node. Summing over all nodes for a particular genotype
gives the probability of observing this genotype in this
generation. The sum over all marginal node probabilities within a
generation is 1.</p>


<h3>Value</h3>

<p>A <code>list</code> with as many elements as there were generations in
<code>F</code>. The list elements are named &quot;F2&quot;, &quot;F3&quot;, etc. Each element is a
vector with the symbolic formulas for the marginal probabilities of all
possible nodes. The vector elements are named
and the names indicate the allelic configuration of the two gametes
comprising the genotype as <code>gamete1-gamete2</code> (e.g.,
<code>"AAB-AAA"</code>).  </p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## F2 and F3 genotypes
node.probs &lt;- nodeProbabilities(buildSelfingTree(genSubtree.2M,3,"AA","BB"),3)

## must sum to 1
stopifnot(all.equal(evalProb(node.probs[["F2"]],z = 0.045),1))
stopifnot(all.equal(evalProb(node.probs[["F3"]],z = 0.045),1))

</code></pre>

<hr>
<h2 id='selfingTree-package'>
Genotype Probabilities in Intermediate Generations of Inbreeding Through Selfing
</h2><span id='topic+selfingTree-package'></span><span id='topic+selfingTree'></span>

<h3>Description</h3>

<p> A probability tree allows to compute probabilities of
complex events, such as genotype probabilities in intermediate
generations of inbreeding through recurrent self-fertilization
(selfing). This package implements functionality to compute probability
trees for two- and three-marker genotypes in the F2 to F7 selfing
generations. The conditional probabilities are derived automatically and
in symbolic form. The package also provides functionality to extract and
evaluate the relevant probabilities.
</p>
<p>Copyright (c) 2014, Pioneer Hi-Bred International, Inc.
</p>
<p>Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are
met:
Redistributions of source code must retain the above copyright
notice, this list of conditions and the following disclaimer.
</p>
<p>Redistributions in binary form must reproduce the above copyright
notice, this list of conditions and the following disclaimer in
the documentation and/or other materials provided with the
distribution.
</p>
<p>Neither the name of Pioneer Hi-Bred International, Inc. nor the
names of its contributors may be used to endorse or promote
products derived from this software without specific prior written
permission.
</p>
<p>THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
&quot;AS IS&quot; AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> selfingTree</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2014-12-18</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyData: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> foreach</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Function <code>buildSelfingTree</code> generates the probability trees for
two- and three-marker genotypes. This is done by recursively calling
functions <code>genSubtree.2M</code> (two-marker genotypes) or
<code>genSubtree.3M</code> (three-marker genotypes). The core functionality of
deriving the symbolic conditional haplotype probabilities is implemented
in functions <code>haploProb.2M</code> and <code>haploProb.3M</code>. The function
<code>nodeProbabilities</code> is used to symbolically multiply the
conditional probabilities along all branches and uses function
<code>extractProbs</code> to extract the conditional probabilities from the
trees. Finally function <code>evalProb</code> symbolically sums the marginal
probabilities of relevant nodes and evaluates them with user specified
values for the recombination frequencies. The function <code>getTargets</code>
can be used to identify relevant events given a target genotype.</p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>
<p>at Pioneer Hi-Bred International, Inc., Breeding Technologies, Johnson/IA, USA.
</p>
<p>Maintainer: Frank Technow <a href="mailto:Frank.Technow@pioneer.com">Frank.Technow@pioneer.com</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
