<!DOCTYPE html><html lang="en"><head><title>Help for package sensiPhy</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sensiPhy}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alien'><p>Alien Mammals dataset: Example dataset for the package sensiPhy</p></a></li>
<li><a href='#alien.data'><p>Alien Mammals dataset: Example dataset for the package sensiPhy</p></a></li>
<li><a href='#alien.phy'><p>Alien Mammals dataset: Example dataset for the package sensiPhy</p></a></li>
<li><a href='#clade_continuous'><p>Influential Clade Detection - Trait Evolution Continuous Characters</p></a></li>
<li><a href='#clade_discrete'><p>Influential Clade Detection - Trait Evolution Discrete Characters</p></a></li>
<li><a href='#clade_phyglm'><p>Estimate the impact on model estimates of phylogenetic logistic regression after</p>
removing clades from the analysis.</a></li>
<li><a href='#clade_phylm'><p>Influential clade detection - Phylogenetic Linear Regression</p></a></li>
<li><a href='#clade_physig'><p>Influential clade detection - Phylogenetic signal</p></a></li>
<li><a href='#influ_continuous'><p>Influential Species Detection - Trait Evolution Continuous Characters</p></a></li>
<li><a href='#influ_discrete'><p>Influential Species Detection - Trait Evolution Discrete Characters</p></a></li>
<li><a href='#influ_phyglm'><p>Influential species detection - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#influ_phylm'><p>Influential species detection - Phylogenetic Linear Regression</p></a></li>
<li><a href='#influ_physig'><p>Influential species detection - Phylogenetic signal</p></a></li>
<li><a href='#intra_clade_phyglm'><p>Interaction between intraspecific variability and influential clade detection - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#intra_clade_phylm'><p>Interaction between intraspecific variability and influential clade detection - Phylogenetic Linear Regression</p></a></li>
<li><a href='#intra_influ_phyglm'><p>Interaction between intraspecific variability and influential species - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#intra_influ_phylm'><p>Interaction between intraspecific variability and influential species - Phylogenetic Linear Regression</p></a></li>
<li><a href='#intra_phyglm'><p>Intraspecific variability - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#intra_phylm'><p>Intraspecific variability - Phylogenetic Linear Regression</p></a></li>
<li><a href='#intra_physig'><p>Intraspecific variability - Phylogenetic signal</p></a></li>
<li><a href='#intra_samp_phyglm'><p>Interaction between intraspecific variability and species sampling - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#intra_samp_phylm'><p>Interaction between intraspecific variability and species sampling - Phylogenetic Linear Regression</p></a></li>
<li><a href='#match_dataphy'><p>Match data and phylogeny based on model formula</p></a></li>
<li><a href='#miss.phylo.d'><p>Phylogenetic signal in missing data</p></a></li>
<li><a href='#primates'><p>Primates dataset: Example dataset for the package sensiPhy</p></a></li>
<li><a href='#primates.data'><p>Primates dataset: Example dataset for the package sensiPhy</p></a></li>
<li><a href='#primates.phy'><p>Primates dataset: Example dataset for the package sensiPhy</p></a></li>
<li><a href='#samp_continuous'><p>Species Sampling uncertainty - Trait Evolution Continuous Characters</p></a></li>
<li><a href='#samp_discrete'><p>Species Sampling uncertainty - Trait Evolution Discrete Characters</p></a></li>
<li><a href='#samp_phyglm'><p>Sensitivity Analysis Species Sampling  - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#samp_phylm'><p>Sensitivity Analysis Species Sampling  - Phylogenetic Linear Regression</p></a></li>
<li><a href='#samp_physig'><p>Sensitivity Analysis Species Sampling  - Phylogenetic signal</p></a></li>
<li><a href='#sensi_plot'><p>Graphical sensitivity analysis for comparative methods</p></a></li>
<li><a href='#sensi_plot.clade.physig'><p>Graphical diagnostics for class 'clade.physig'</p></a></li>
<li><a href='#sensi_plot.influ.physig'><p>Graphical diagnostics for class 'influ.physig'</p></a></li>
<li><a href='#sensi_plot.intra.physig'><p>Graphical diagnostics for class 'intra.physig'</p></a></li>
<li><a href='#sensi_plot.samp.physig'><p>Graphical diagnostics for class 'samp.physig'</p></a></li>
<li><a href='#sensi_plot.sensiClade'><p>Graphical diagnostics for class 'sensiClade'</p></a></li>
<li><a href='#sensi_plot.sensiClade.TraitEvol'><p>Graphical diagnostics for class 'sensiClade.TraitEvol'</p></a></li>
<li><a href='#sensi_plot.sensiInflu'><p>Graphical diagnostics for class 'sensiInflu'</p></a></li>
<li><a href='#sensi_plot.sensiInflu.TraitEvol'><p>Graphical diagnostics for class 'sensiInflu.TraitEvol'</p></a></li>
<li><a href='#sensi_plot.sensiIntra'><p>Graphical diagnostics for class 'sensiIntra'</p></a></li>
<li><a href='#sensi_plot.sensiIntra_Clade'><p>Graphical diagnostics for class 'sensiIntra_Clade'</p></a></li>
<li><a href='#sensi_plot.sensiIntra_Influ'><p>Graphical diagnostics for class 'sensiIntra_Influ'</p></a></li>
<li><a href='#sensi_plot.sensiIntra_Samp'><p>Graphical diagnostics for class 'sensiIntra_Samp' and 'sensiTree_samp'</p></a></li>
<li><a href='#sensi_plot.sensiSamp'><p>Graphical diagnostics for class 'sensiSamp'</p></a></li>
<li><a href='#sensi_plot.sensiSamp.TraitEvol'><p>Graphical diagnostics for class 'sensiSamp.TraitEvol'</p></a></li>
<li><a href='#sensi_plot.sensiTree'><p>Graphical diagnostics for class 'sensiTree'</p></a></li>
<li><a href='#sensi_plot.sensiTree_Clade'><p>Graphical diagnostics for class 'sensiTree_Clade'</p></a></li>
<li><a href='#sensi_plot.sensiTree_Influ'><p>Graphical diagnostics for class 'sensiTree_Influ'</p></a></li>
<li><a href='#sensi_plot.sensiTree_Intra'><p>Graphical diagnostics for class 'sensiIntra_Tree'</p></a></li>
<li><a href='#sensi_plot.sensiTree_Samp'><p>Graphical diagnostics for class 'sensiTree_samp'</p></a></li>
<li><a href='#sensi_plot.sensiTree.TraitEvol'><p>Graphical diagnostics for class 'sensiTree.TraitEvol'</p></a></li>
<li><a href='#sensi_plot.tree.bd'><p>Graphical diagnostics for class 'tree.physig'</p></a></li>
<li><a href='#sensi_plot.tree.physig'><p>Graphical diagnostics for class 'tree.physig'</p></a></li>
<li><a href='#sensiPhy'><p>Sensitivity Analysis for Comparative Methods</p></a></li>
<li><a href='#tree_bd'><p>Phylogenetic uncertainty - net diversification rate</p></a></li>
<li><a href='#tree_clade_phyglm'><p>Interaction between phylogenetic uncertainty and influential clade detection - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#tree_clade_phylm'><p>Interaction between phylogenetic uncertainty and influential clade detection - Phylogenetic Linear Regression</p></a></li>
<li><a href='#tree_continuous'><p>Phylogenetic uncertainty - Trait Evolution Continuous Characters</p></a></li>
<li><a href='#tree_discrete'><p>Phylogenetic uncertainty - Trait Evolution Discrete Characters</p></a></li>
<li><a href='#tree_influ_phyglm'><p>Interaction between phylogenetic uncertainty and influential species detection - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#tree_influ_phylm'><p>Interaction between phylogenetic uncertainty and influential species detection - Phylogenetic Linear Regression</p></a></li>
<li><a href='#tree_intra_phyglm'><p>Interaction between phylogenetic uncertainty and intraspecific variability - Phylogenetic logistic Regression</p></a></li>
<li><a href='#tree_intra_phylm'><p>Interaction between phylogenetic uncertainty and intraspecific variability  - Phylogenetic Linear Regression</p></a></li>
<li><a href='#tree_phyglm'><p>Phylogenetic uncertainty - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#tree_phylm'><p>Phylogenetic uncertainty - Phylogenetic Linear Regression</p></a></li>
<li><a href='#tree_physig'><p>Phylogenetic uncertainty - Phylogenetic signal</p></a></li>
<li><a href='#tree_samp_phyglm'><p>Interaction between phylogenetic uncertainty and sensitivity to species sampling  - Phylogenetic Logistic Regression</p></a></li>
<li><a href='#tree_samp_phylm'><p>Interaction between phylogenetic uncertainty and sensitivity to species sampling - Phylogenetic Linear Regression</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Sensitivity Analysis for Comparative Methods</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-03-31</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gustavo Paterno &lt;paternogbc@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of sensitivity analysis for phylogenetic comparative
 methods. The package is an umbrella of statistical and graphical methods that 
 estimate and report different types of uncertainty in PCM:
 (i) Species Sampling uncertainty (sample size; influential species and clades).
 (ii) Phylogenetic uncertainty (different topologies and/or branch lengths).
 (iii) Data uncertainty (intraspecific variation and measurement error).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0), ape (&ge; 3.3), phylolm (&ge; 2.4), ggplot2 (&ge;
2.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>caper (&ge; 0.5.2), phytools (&ge; 0.6), geiger (&ge; 2.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/paternogbc/sensiPhy">https://github.com/paternogbc/sensiPhy</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/paternogbc/sensiPhy/issues">https://github.com/paternogbc/sensiPhy/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-02 13:37:55 UTC; guga</td>
</tr>
<tr>
<td>Author:</td>
<td>Gustavo Paterno [cre, aut],
  Gijsbert Werner [aut],
  Caterina Penone [aut],
  Pablo Martinez [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-02 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='alien'>Alien Mammals dataset: Example dataset for the package sensiPhy</h2><span id='topic+alien'></span>

<h3>Description</h3>

<p>A comparative dataset containing traits for 94 alien mammal species 
(alien.data) and a multiphylo object with 101 phylogenies matching the 
data (alien.phy). Tip labels are the binomial species names and match 
with data rownames. Data was taken from (Gonzalez-Suarez et al. 2015) and phylogenies from
(Fritz et al 2009) and (Kuhn et al 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(alien)
</code></pre>


<h3>Format</h3>

<p>A data frame with 94 rows and 7 variables:
</p>

<ul>
<li><p> family: Taxonomic family
</p>
</li>
<li><p> adultMass: Mean adult body mass (g)
</p>
</li>
<li><p> gestaLen: Mean gestation length (days)
</p>
</li>
<li><p> homeRange: Mean home range (km)
</p>
</li>
<li><p> SE_mass: Standard deviation (intraspecific) for mean adult body mass (g)
</p>
</li>
<li><p> SE_gesta: Standard deviation (intraspecific) for mean gestation length (days)
</p>
</li>
<li><p> SE_range: Standard deviation (intraspecific) for mean home range (km)
</p>
</li></ul>


<h3>References</h3>

<p>Alien mammal data: Gonzalez-Suarez, Manuela, Sven Bacher, and Jonathan M. Jeschke. 
&quot;Intraspecific trait variation is correlated with establishment success of alien mammals.&quot; 
The American Naturalist 185.6 (2015): 737-746  DOI: 10.1086/681105
</p>
<p>Downloaded from: Gonzalez-Surez M, Bacher S, Jeschke J (2015) Data from: Intraspecific trait 
variation is correlated with establishment success of alien mammals.
Dryad Digital Repository. http://dx.doi.org/10.5061/dryad.sp963
</p>
<p>Phylogeny: Kuhn, Tyler S., Arne O. Mooers, and Gavin H. Thomas. &quot;A simple polytomy resolver for 
dated phylogenies.&quot; Methods in Ecology and Evolution 2.5 (2011): 427-436.
</p>
<p>Fritz, Susanne A., Olaf RP Bininda-Emonds, and Andy Purvis. &quot;Geographical variation in predictors 
of mammalian extinction risk: big is bad, but only in the tropics.&quot; Ecology letters 12.6 (2009): 538-549.
</p>

<hr>
<h2 id='alien.data'>Alien Mammals dataset: Example dataset for the package sensiPhy</h2><span id='topic+alien.data'></span>

<h3>Description</h3>

<p>A comparative dataset containing traits for 94 alien mammal species 
(alien.data) and a multiphylo object with 101 phylogenies matching the 
data (alien.phy). Tip labels are the binomial species names and match 
with data rownames. Data was taken from (Gonzalez-Suarez et al. 2015) and phylogenies from
(Fritz et al 2009) and (Kuhn et al 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(alien)
</code></pre>


<h3>Format</h3>

<p>A data frame with 94 rows and 7 variables:
</p>

<ul>
<li><p> family: Taxonomic family
</p>
</li>
<li><p> adultMass: Mean adult body mass (g)
</p>
</li>
<li><p> gestaLen: Mean gestation length (days)
</p>
</li>
<li><p> homeRange: Mean home range (km)
</p>
</li>
<li><p> SE_mass: Standard deviation (intraspecific) for mean adult body mass (g)
</p>
</li>
<li><p> SE_gesta: Standard deviation (intraspecific) for mean gestation length (days)
</p>
</li>
<li><p> SE_range: Standard deviation (intraspecific) for mean home range (km)
</p>
</li></ul>


<h3>References</h3>

<p>Alien mammal data: Gonzalez-Suarez, Manuela, Sven Bacher, and Jonathan M. Jeschke. 
&quot;Intraspecific trait variation is correlated with establishment success of alien mammals.&quot; 
The American Naturalist 185.6 (2015): 737-746  DOI: 10.1086/681105
</p>
<p>Downloaded from: Gonzalez-Surez M, Bacher S, Jeschke J (2015) Data from: Intraspecific trait 
variation is correlated with establishment success of alien mammals.
Dryad Digital Repository. http://dx.doi.org/10.5061/dryad.sp963
</p>
<p>Phylogeny: Kuhn, Tyler S., Arne O. Mooers, and Gavin H. Thomas. &quot;A simple polytomy resolver for 
dated phylogenies.&quot; Methods in Ecology and Evolution 2.5 (2011): 427-436.
</p>
<p>Fritz, Susanne A., Olaf RP Bininda-Emonds, and Andy Purvis. &quot;Geographical variation in predictors 
of mammalian extinction risk: big is bad, but only in the tropics.&quot; Ecology letters 12.6 (2009): 538-549.
</p>

<hr>
<h2 id='alien.phy'>Alien Mammals dataset: Example dataset for the package sensiPhy</h2><span id='topic+alien.phy'></span>

<h3>Description</h3>

<p>A comparative dataset containing traits for 94 alien mammal species 
(alien.data) and a multiphylo object with 101 phylogenies matching the 
data (alien.phy). Tip labels are the binomial species names and match 
with data rownames. Data was taken from (Gonzalez-Suarez et al. 2015) and phylogenies from
(Fritz et al 2009) and (Kuhn et al 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(alien)
</code></pre>


<h3>Format</h3>

<p>A data frame with 94 rows and 7 variables:
</p>

<ul>
<li><p> family: Taxonomic family
</p>
</li>
<li><p> adultMass: Mean adult body mass (g)
</p>
</li>
<li><p> gestaLen: Mean gestation length (days)
</p>
</li>
<li><p> homeRange: Mean home range (km)
</p>
</li>
<li><p> SE_mass: Standard deviation (intraspecific) for mean adult body mass (g)
</p>
</li>
<li><p> SE_gesta: Standard deviation (intraspecific) for mean gestation length (days)
</p>
</li>
<li><p> SE_range: Standard deviation (intraspecific) for mean home range (km)
</p>
</li></ul>


<h3>References</h3>

<p>Alien mammal data: Gonzalez-Suarez, Manuela, Sven Bacher, and Jonathan M. Jeschke. 
&quot;Intraspecific trait variation is correlated with establishment success of alien mammals.&quot; 
The American Naturalist 185.6 (2015): 737-746  DOI: 10.1086/681105
</p>
<p>Downloaded from: Gonzalez-Surez M, Bacher S, Jeschke J (2015) Data from: Intraspecific trait 
variation is correlated with establishment success of alien mammals.
Dryad Digital Repository. http://dx.doi.org/10.5061/dryad.sp963
</p>
<p>Phylogeny: Kuhn, Tyler S., Arne O. Mooers, and Gavin H. Thomas. &quot;A simple polytomy resolver for 
dated phylogenies.&quot; Methods in Ecology and Evolution 2.5 (2011): 427-436.
</p>
<p>Fritz, Susanne A., Olaf RP Bininda-Emonds, and Andy Purvis. &quot;Geographical variation in predictors 
of mammalian extinction risk: big is bad, but only in the tropics.&quot; Ecology letters 12.6 (2009): 538-549.
</p>

<hr>
<h2 id='clade_continuous'>Influential Clade Detection - Trait Evolution Continuous Characters</h2><span id='topic+clade_continuous'></span>

<h3>Description</h3>

<p>Fits models for trait evolution of continuous characters, 
detecting influential clades
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clade_continuous(
  data,
  phy,
  model,
  trait.col,
  clade.col,
  n.species = 5,
  n.sim = 20,
  bounds = list(),
  n.cores = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clade_continuous_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_model">model</code></td>
<td>
<p>The evolutionary model (see Details).</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_trait.col">trait.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
trait to analyse (which should be a factor with two level)</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_bounds">bounds</code></td>
<td>
<p>settings to constrain parameter estimates. See <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code></p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use. If 'NULL', number of cores is detected.</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="clade_continuous_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one clade at a time,
fits a model of continuous character evolution using <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code>, 
repeats this many times (controlled by <code>n.sim</code>), stores the results and calculates 
the effects on model parameters Currently, only binary continuous traits are supported. 
</p>
<p>Additionally, to account for the influence of the number of species on each 
clade (clade sample size), this function also estimates a null distribution
expected for the number of species in a given clade. This is done by fitting
models without the same number of species as in the given clade.The number of 
simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>Different evolutionary models from <code>fitContinuous</code> can be used, i.e. <code>BM</code>,<code>OU</code>,
<code>EB</code>, <code>trend</code>, <code>lambda</code>, <code>kappa</code>, <code>delta</code> and <code>drift</code>.
</p>
<p>See <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code> for more details on evolutionary models. 
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_continuous</code> returns a list with the following
components:
</p>
<p><code>call</code>: The function call
</p>
<p><code>data</code>: The original full data frame.
</p>
<p><code>full.model.estimates</code>: Parameter estimates (rate of evolution <code>sigsq</code>
and where applicable <code>optpar</code>), root state <code>z0</code>, 
AICc for the full model without deleted clades.
</p>
<p><code>sensi.estimates</code>: Parameter estimates (sigsq and optpar), (percentual) difference 
in parameter estimate compared to the full model (DIFsigsq, sigsq.perc, DIFoptpar, optpar.perc),
AICc and z0 for each repeat with a clade removed.
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distributions
for all clades analysed.
</p>
<p><code>errors</code>: Clades where deletion resulted in errors.
</p>
<p><code>clade.col</code>: Which column was used to specify the clades?
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>


<h3>Author(s)</h3>

<p>Gijsbert Werner &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Yang Z. 2006. Computational Molecular Evolution. Oxford University Press: Oxford. 
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008.
GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data("primates")
#Model trait evolution accounting for phylogenetic uncertainty
clade_cont&lt;-clade_continuous(data=primates$data,phy = primates$phy[[1]], model="OU",
trait.col = "adultMass",clade.col="family",n.sim=30,n.species=10,n.cores = 2,track=TRUE)
#Print summary statistics
summary(clade_cont)
sensi_plot(clade_cont,graph="all")
sensi_plot(clade_cont,clade="Cercopithecidae",graph = "sigsq")
sensi_plot(clade_cont,clade="Cercopithecidae",graph = "optpar")
#Change the evolutionary model, tree transformation or minimum number of species per clade
clade_cont2&lt;-clade_continuous(data=primates$data,phy = primates$phy[[1]],model="delta",
trait.col = "adultMass",clade.col="family",n.sim=30,n.species=5,n.cores = 2,track=TRUE)
summary(clade_cont2)
sensi_plot(clade_cont2)
clade_cont3&lt;-clade_continuous(data=primates$data,phy = primates$phy[[1]],model="BM",
trait.col = "adultMass",clade.col="family",n.sim=30,n.species=5,n.cores = 2,track=TRUE)
summary(clade_cont3)
sensi_plot(clade_cont3,graph="sigsq")

## End(Not run)
</code></pre>

<hr>
<h2 id='clade_discrete'>Influential Clade Detection - Trait Evolution Discrete Characters</h2><span id='topic+clade_discrete'></span>

<h3>Description</h3>

<p>Fits models for trait evolution of discrete (binary) characters, 
detecting influential clades
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clade_discrete(
  data,
  phy,
  model,
  transform = "none",
  trait.col,
  clade.col,
  n.species = 5,
  n.sim = 20,
  bounds = list(),
  n.cores = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clade_discrete_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_model">model</code></td>
<td>
<p>The Mkn model to use (see Details).</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_transform">transform</code></td>
<td>
<p>The evolutionary model to transform the tree (see Details). Default is <code>none</code>.</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_trait.col">trait.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
trait to analyse (which should be a factor with two level)</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_bounds">bounds</code></td>
<td>
<p>settings to constrain parameter estimates. See <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code></p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use. If 'NULL', number of cores is detected.</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="clade_discrete_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one clade at a time,
fits a model of discrete character evolution using <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code>, 
repeats this this many times (controlled by <code>n.sim</code>), stores the results and calculates 
the effects on model parameters. Currently, only binary discrete traits are supported. 
</p>
<p>Additionally, to account for the influence of the number of species on each 
clade (clade sample size), this function also estimates a null distribution
expected for the number of species in a given clade. This is done by fitting
models without the same number of species as in the given clade.The number of 
simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>Different character model from <code>fitDiscrete</code> can be used, including <code>ER</code> (equal-rates), 
<code>SYM</code> (symmetric), <code>ARD</code> (all-rates-different) and <code>meristic</code> (stepwise fashion). 
</p>
<p>All transformations to the phylogenetic tree from <code>fitDiscrete</code> can be used, i.e. <code>none</code>,
<code>EB</code>, <code>lambda</code>, <code>kappa</code> and<code>delta</code>.
</p>
<p>See <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code> for more details on character models and tree transformations. 
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_discrete</code> returns a list with the following
components:
</p>
<p><code>call</code>: The function call
</p>
<p><code>data</code>: The original full data frame.
</p>
<p><code>full.model.estimates</code>: Parameter estimates (transition rates q12 and q21), 
AICc and the optimised value of the phylogenetic transformation parameter (e.g. <code>lambda</code>) 
for the full model without deleted clades.
</p>
<p><code>sensi.estimates</code>: Parameter estimates (transition rates q12 and q21),(percentual) difference 
in parameter estimate compared to the full model (DIFq12, sigsq.q12, DIFq21, optpar.q21), 
AICc and the optimised value of the phylogenetic transformation parameter (e.g. <code>lambda</code>) 
for each repeat with a clade removed.
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distributions
for all clades analysed.
</p>
<p><code>errors</code>: Clades where deletion resulted in errors.
</p>
<p><code>clade.col</code>: Which column was used to specify the clades?
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>


<h3>Author(s)</h3>

<p>Gijsbert Werner &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Yang Z. 2006. Computational Molecular Evolution. Oxford University Press: Oxford. 
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008.
GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load data:
data("primates")
#Create a binary trait factor 
primates$data$adultMass_binary&lt;-ifelse(primates$data$adultMass &gt; 7350, "big", "small")
clade_disc&lt;-clade_discrete(data=primates$data,phy = primates$phy[[1]],model="SYM",
trait.col = "adultMass_binary",clade.col="family",n.sim=30,n.species=10,n.cores = 2)
summary(clade_disc)
sensi_plot(clade_disc)
sensi_plot(clade_disc, clade = "Cebidae", graph = "q12")
#Change the evolutionary model, tree transformation or minimum number of species per clade
clade_disc_2&lt;-clade_discrete(data=primates$data,phy = primates$phy[[1]],
model="ARD",transform="kappa",
trait.col = "adultMass_binary",clade.col="family",n.sim=30,
n.species=8,n.cores = 2)
summary(clade_disc_2)
sensi_plot(clade_disc_2)
sensi_plot(clade_disc_2, graph = "q12")
sensi_plot(clade_disc_2, graph = "q21")

## End(Not run)
</code></pre>

<hr>
<h2 id='clade_phyglm'>Estimate the impact on model estimates of phylogenetic logistic regression after 
removing clades from the analysis.</h2><span id='topic+clade_phyglm'></span>

<h3>Description</h3>

<p>Estimate the impact on model estimates of phylogenetic logistic regression after 
removing clades from the analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clade_phyglm(
  formula,
  data,
  phy,
  btol = 50,
  track = TRUE,
  clade.col,
  n.species = 5,
  n.sim = 100,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clade_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="clade_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="clade_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="clade_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="clade_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="clade_phyglm_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="clade_phyglm_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="clade_phyglm_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="clade_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one clade at a time, fits a phylogenetic
logistic regression model using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code> and stores the
results. The impact of of a specific clade on model estimates is calculated by a
comparison between the full model (with all species) and the model without 
the species belonging to a clade.
</p>
<p>To account for the influence of the number of species on each 
clade (clade sample size), this function also estimates a null distribution
expected for the number of species in a given clade. This is done by fitting
models without the same number of species as in the given clade. 
The number of simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>Currently, only logistic regression using the &quot;logistic_MPLE&quot;-method from
<code>phyloglm</code> is implemented.
</p>
<p><code>clade_phyglm</code> detects influential clades based on
difference in intercept and/or estimate when removing a given clade compared
to the full model including all species.
</p>
<p>Additionally, to account for the influence of the number of species on each 
clade (clade sample size), this function also estimates a null distribution 
expected for the number of species in a given clade. This is done by fitting
models without the same number of species in the given clade. 
The number of simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>clade_phyglm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>alpha</code>) for the full model
without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the percentage of change
in intercept compared to the full model (<code>intercept.perc</code>) and intercept
p-value (<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter are
reported.
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distributions
for all clades analysed.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Clades where deletion resulted in errors.
</p>
<p><code>clade.col</code>: Which column was used to specify the clades?
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno &amp; Gijsbert Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+clade_phylm">clade_phylm</a></code>,
<code><a href="#topic+influ_phyglm">influ_phyglm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Simulate Data:
set.seed(6987)
phy = rtree(150)
x = rTrait(n=1,phy=phy)
X = cbind(rep(1,150),x)
y = rbinTrait(n=1,phy=phy, beta=c(-1,0.5), alpha=.7 ,X=X)
cla &lt;- rep(c("A","B","C","D","E"), each = 30)
dat = data.frame(y, x, cla)
# Run sensitivity analysis:
clade &lt;- clade_phyglm(y ~ x, phy = phy, data = dat, n.sim = 30, clade.col = "cla")
# To check summary results and most influential clades:
summary(clade)
# Visual diagnostics for clade removal:
sensi_plot(clade)
# Specify which clade removal to plot:
sensi_plot(clade, "B")
sensi_plot(clade, "C")
sensi_plot(clade, "D") #The clade with the largest effect on slope and intercept

## End(Not run)
</code></pre>

<hr>
<h2 id='clade_phylm'>Influential clade detection - Phylogenetic Linear Regression</h2><span id='topic+clade_phylm'></span>

<h3>Description</h3>

<p>Estimate the impact on model estimates of phylogenetic linear regression after 
removing clades from the analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clade_phylm(
  formula,
  data,
  phy,
  model = "lambda",
  track = TRUE,
  clade.col,
  n.species = 5,
  n.sim = 100,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clade_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="clade_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="clade_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="clade_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="clade_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="clade_phylm_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="clade_phylm_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="clade_phylm_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="clade_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one clade at a time, fits a phylogenetic
linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code> and stores the
results. The impact of of a specific clade on model estimates is calculated by a
comparison between the full model (with all species) and the model without 
the species belonging to a clade.
</p>
<p>Additionally, to account for the influence of the number of species on each 
clade (clade sample size), this function also estimates a null distribution
expected for the number of species in a given clade. This is done by fitting
models without the same number of species as in the given clade. 
The number of simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p><code>clade_phylm</code> detects influential clades based on
difference in intercept and/or estimate when removing a given clade compared
to the full model including all species.
</p>
<p>Currently, this function can only implement simple linear models (i.e. 
<code class="reqn">y = a + bx</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>clade_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the percentage of change
in intercept compared to the full model (<code>intercept.perc</code>) and intercept
p-value (<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter 
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) 
are reported.
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distributions
for all clades analysed.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Clades where deletion resulted in errors.
</p>
<p><code>clade.col</code>: Which column was used to specify the clades?
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+samp_phylm">samp_phylm</a></code>,
<code><a href="#topic+influ_phylm">influ_phylm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>,
<code><a href="#topic+sensi_plot">sensi_plot</a></code>,<code><a href="#topic+clade_phyglm">clade_phyglm</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load data:
data(primates)
# run analysis:
clade &lt;- clade_phylm(log(sexMaturity) ~ log(adultMass), 
phy = primates$phy[[1]], data = primates$data, n.sim = 30, clade.col = "family")
# To check summary results and most influential clades:
summary(clade)
# Visual diagnostics for clade removal:
sensi_plot(clade)
# Specify which clade removal to plot:
sensi_plot(clade, "Cercopithecidae")
sensi_plot(clade, "Cebidae")

## End(Not run)
</code></pre>

<hr>
<h2 id='clade_physig'>Influential clade detection - Phylogenetic signal</h2><span id='topic+clade_physig'></span>

<h3>Description</h3>

<p>Estimate the influence of clade removal on phylogenetic signal estimates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clade_physig(
  trait.col,
  data,
  phy,
  clade.col,
  n.species = 5,
  n.sim = 100,
  method = "K",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clade_physig_+3A_trait.col">trait.col</code></td>
<td>
<p>The name of a column in the provided data frame with trait 
to be analyzed  (e.g. &quot;Body_mass&quot;).</p>
</td></tr>
<tr><td><code id="clade_physig_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="clade_physig_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="clade_physig_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="clade_physig_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="clade_physig_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="clade_physig_+3A_method">method</code></td>
<td>
<p>Method to compute signal: can be &quot;K&quot; or &quot;lambda&quot;.</p>
</td></tr>
<tr><td><code id="clade_physig_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="clade_physig_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="phytools.html#topic+phylosig">phylosig</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one clade at a time, estimates phylogenetic
signal (K or lambda) using <code><a href="phytools.html#topic+phylosig">phylosig</a></code> and stores the
results. The impact of a specific clade on signal estimates is calculated by the
comparison between the full data (with all species) and reduced data estimates
(without the species belonging to a clade).
</p>
<p>To account for the influence of the number of species on each 
clade (clade sample size), this function also estimate a null distribution of signal estimates
expected by the removal of the same number of species as in a given clade. This is done by estimating
phylogenetic signal without the same number of species in the given clade. 
The number of simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>clade_physig</code> returns a list with the following
components:
</p>
<p><code>trait.col</code>: Column name of the trait analysed
</p>
<p><code>full.data.estimates</code>: Phylogenetic signal estimate (K or lambda)
and the P value (for the full data).
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade. Columns report the calculated
phylogenetic signal (K or lambda) (<code>estimate</code>), difference between simulation
signal and full data signal (<code>DF</code>), the percentage of change
in signal compared to the full data estimate (<code>perc</code>) and 
p-value of the phylogenetic signal with the reduced data (<code>pval</code>).
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distribution
of phylogenetic signal for all clades analysed.
</p>
<p><code>data</code>: Original full dataset.
</p>


<h3>Note</h3>

<p>The argument &quot;se&quot; from <code><a href="phytools.html#topic+phylosig">phylosig</a></code> is not available in this function. Use the 
argument &quot;V&quot; instead with <code><a href="#topic+intra_physig">intra_physig</a></code> to indicate the name of the column containing the standard 
deviation or the standard error of the trait variable instead.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Blomberg, S. P., T. Garland Jr., A. R. Ives (2003) 
Testing for phylogenetic signal in comparative data: 
Behavioral traits are more labile. Evolution, 57, 717-745.
</p>
<p>Pagel, M. (1999) Inferring the historical patterns of biological evolution. 
Nature, 401, 877-884.
</p>
<p>Kamilar, J. M., &amp; Cooper, N. (2013). Phylogenetic signal in primate behaviour,
ecology and life history. Philosophical Transactions of the Royal Society 
B: Biological Sciences, 368: 20120341.
</p>


<h3>See Also</h3>

<p><code><a href="phytools.html#topic+phylosig">phylosig</a></code>, 
<code><a href="#topic+clade_phylm">clade_phylm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alien)
alien.data&lt;-alien$data
alien.phy&lt;-alien$phy
# Logtransform data
alien.data$logMass &lt;- log(alien.data$adultMass) 
# Run sensitivity analysis:
clade &lt;- clade_physig(trait.col = "logMass", data = alien.data, n.sim = 20,
                 phy = alien.phy[[1]], clade.col = "family", method = "K")
summary(clade)
sensi_plot(clade, "Bovidae")
sensi_plot(clade, "Sciuridae")
</code></pre>

<hr>
<h2 id='influ_continuous'>Influential Species Detection - Trait Evolution Continuous Characters</h2><span id='topic+influ_continuous'></span>

<h3>Description</h3>

<p>Fits models for trait evolution of continuous characters, 
detecting influential species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>influ_continuous(
  data,
  phy,
  model,
  bounds = list(),
  cutoff = 2,
  n.cores = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="influ_continuous_+3A_data">data</code></td>
<td>
<p>Data vector for a single continuous trait, with names matching tips in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="influ_continuous_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="influ_continuous_+3A_model">model</code></td>
<td>
<p>The evolutionary model (see Details).</p>
</td></tr>
<tr><td><code id="influ_continuous_+3A_bounds">bounds</code></td>
<td>
<p>settings to constrain parameter estimates. See <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code></p>
</td></tr>
<tr><td><code id="influ_continuous_+3A_cutoff">cutoff</code></td>
<td>
<p>The cut-off parameter for influential species (see Details).</p>
</td></tr>
<tr><td><code id="influ_continuous_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use. If 'NULL', number of cores is detected.</p>
</td></tr>
<tr><td><code id="influ_continuous_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="influ_continuous_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one species at a time,  
fits different models of continuous character evolution using <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code>, 
stores the results and calculates the effects on model parameters.
</p>
<p><code>influ_continuous</code> detects influential species based on the standardised
difference in the rate parameter <code>sigsq</code> and the optimisation parameter <code>optpar</code> 
(e.g. lamda, kappa, alpha, depending on which <code>model</code> is set), when removing 
a given species compared to the full model including all species. 
Species with a standardised difference above the value of 
<code>cutoff</code> are identified as influential. 
</p>
<p>Different evolutionary models from <code>fitContinuous</code> can be used, i.e. <code>BM</code>,<code>OU</code>,
<code>EB</code>, <code>trend</code>, <code>lambda</code>, <code>kappa</code>, <code>delta</code> and <code>drift</code>.
</p>
<p>See <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code> for more details on evolutionary models.
</p>


<h3>Value</h3>

<p>The function <code>tree_discrete</code> returns a list with the following
components:
</p>
<p><code>call</code>: The function call
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>data</code>: The original full data vector
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>
<p><code>full.model.estimates</code>: Parameter estimates (rate of evolution <code>sigsq</code>
and where applicable <code>optpar</code>), root state <code>z0</code>, 
AICc for the full model without deleted species.
</p>
<p><code>influential_species</code>: List of influential species, based on standardised 
difference in estimates for sigsq and optpar. Species are ordered from most influential to 
less influential and only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>sensi.estimates</code>: Parameter estimates (sigsq and optpar),(percentual) difference 
in parameter estimate compared to the full model (DIFsigsq, sigsq.perc,sDIFsigsq, 
DIFoptpar, optpar.perc,sDIFoptpar), 
AICc and z0 for each repeat with a species removed.
</p>


<h3>Author(s)</h3>

<p>Gijsbert Werner &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Yang Z. 2006. Computational Molecular Evolution. Oxford University Press: Oxford. 
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008.
GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load data:
data("primates")
#Model trait evolution accounting for influential species
adultMass&lt;-primates$data$adultMass
names(adultMass)&lt;-rownames(primates$data)
influ_cont&lt;-influ_continuous(data = adultMass,phy = primates$phy[[1]],
model = "OU",cutoff = 2,n.cores = 2,track = TRUE)
#Print summary statistics
summary(influ_cont)
sensi_plot(influ_cont)
sensi_plot(influ_cont,graphs="sigsq")
#' sensi_plot(influ_cont,graphs="optpar")
#Use a different evolutionary model or cutoff 
influ_cont2&lt;-influ_continuous(data = adultMass,phy = primates$phy[[1]],
model = "lambda",cutoff = 1.2,n.cores = 2,track = TRUE)
summary(influ_cont2)
sensi_plot(influ_cont2)
influ_cont3&lt;-influ_continuous(data = adultMass,phy = primates$phy[[1]],
model = "BM",cutoff = 2,n.cores = 2,track = TRUE)
summary(influ_cont3)

## End(Not run)
</code></pre>

<hr>
<h2 id='influ_discrete'>Influential Species Detection - Trait Evolution Discrete Characters</h2><span id='topic+influ_discrete'></span>

<h3>Description</h3>

<p>Fits models for trait evolution of discrete (binary) characters, 
detecting influential species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>influ_discrete(
  data,
  phy,
  model,
  transform = "none",
  bounds = list(),
  cutoff = 2,
  n.cores = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="influ_discrete_+3A_data">data</code></td>
<td>
<p>Data vector for a single binary trait, with names matching tips in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="influ_discrete_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="influ_discrete_+3A_model">model</code></td>
<td>
<p>The Mkn model to use (see Details).</p>
</td></tr>
<tr><td><code id="influ_discrete_+3A_transform">transform</code></td>
<td>
<p>The evolutionary model to transform the tree (see Details). Default is <code>none</code>.</p>
</td></tr>
<tr><td><code id="influ_discrete_+3A_bounds">bounds</code></td>
<td>
<p>settings to constrain parameter estimates. See <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code></p>
</td></tr>
<tr><td><code id="influ_discrete_+3A_cutoff">cutoff</code></td>
<td>
<p>The cut-off parameter for influential species (see Details).</p>
</td></tr>
<tr><td><code id="influ_discrete_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use. If 'NULL', number of cores is detected.</p>
</td></tr>
<tr><td><code id="influ_discrete_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="influ_discrete_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one species at a time,  
fits a model of discrete character evolution using <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code>, 
stores the results and calculates the effects on model parameters. Currently, only 
binary discrete traits are supported.
</p>
<p><code>influ_discrete</code> detects influential species based on the standardised
difference in q12 or q21 when removing a given species compared
to the full model including all species. Species with a standardised difference
above the value of <code>cutoff</code> are identified as influential. 
</p>
<p>Different character model from <code>fitDiscrete</code> can be used, including <code>ER</code> (equal-rates), 
<code>SYM</code> (symmetric), <code>ARD</code> (all-rates-different) and <code>meristic</code> (stepwise fashion). 
</p>
<p>Different transformations to the phylogenetic tree from <code>fitDiscrete</code> can be used, i.e. <code>none</code>,
<code>EB</code>, <code>lambda</code>, <code>kappa</code> and<code>delta</code>.
</p>
<p>See <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code> for more details on character models and tree transformations.
</p>


<h3>Value</h3>

<p>The function <code>tree_discrete</code> returns a list with the following
components:
</p>
<p><code>call</code>: The function call
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>data</code>: The original full data vector
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>
<p><code>full.model.estimates</code>: Parameter estimates (transition rates q12 and q21), 
AICc and the optimised value of the phylogenetic transformation parameter (e.g. <code>lambda</code>) 
for the full model.
</p>
<p><code>influential_species</code>: List of influential species, based on standardised 
difference in estimates for q12 and q21. Species are ordered from most influential to 
less influential and only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>sensi.estimates</code>: Parameter estimates (transition rates q12 and q21),,(percentual) difference 
in parameter estimate compared to the full model (DIFq12, sigsq.q12,sDIFq12, DIFq21, optpar.q21,sDIFq21),  
AICc and the optimised value of the phylogenetic transformation parameter (e.g. <code>lambda</code>) 
for each analysis with a species deleted.
</p>


<h3>Author(s)</h3>

<p>Gijsbert Werner &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Yang Z. 2006. Computational Molecular Evolution. Oxford University Press: Oxford. 
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008.
GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load data:
data("primates")
#Create a binary trait factor 
adultMass_binary&lt;-ifelse(primates$data$adultMass &gt; 7350, "big", "small")
adultMass_binary&lt;-as.factor(as.factor(adultMass_binary))
names(adultMass_binary)&lt;-rownames(primates$data)
#Model trait evolution accounting for influential species
influ_binary&lt;-influ_discrete(data = adultMass_binary,phy = primates$phy[[1]],
model = "SYM",transform = "none",cutoff = 2,n.cores = 2,track = TRUE)
#Print summary statistics
summary(influ_binary)
sensi_plot(influ_binary) #q12 and q21 are, as expected, exactly the same in symmetrical model. 
#Use a different evolutionary model. 
influ_binary2&lt;-influ_discrete(data = adultMass_binary,phy = primates$phy[[1]],
model = "SYM",transform = "delta",n.cores = 2,track = TRUE)
summary(influ_binary2)
sensi_plot(influ_binary2)
#Or change the cutoff and transformation
influ_binary3&lt;-influ_discrete(data = adultMass_binary,phy = primates$phy[[1]],
model = "ARD",transform = "none",cutoff = 1.2,n.cores = 2,track = TRUE)
summary(influ_binary3)
sensi_plot(influ_binary3) 

## End(Not run)
</code></pre>

<hr>
<h2 id='influ_phyglm'>Influential species detection - Phylogenetic Logistic Regression</h2><span id='topic+influ_phyglm'></span>

<h3>Description</h3>

<p>Performs leave-one-out deletion analysis for phylogenetic logistic regression,
and detects influential species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>influ_phyglm(formula, data, phy, btol = 50, cutoff = 2, track = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="influ_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="influ_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="influ_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="influ_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="influ_phyglm_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff value used to identify for influential species
(see Details)</p>
</td></tr>
<tr><td><code id="influ_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="influ_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one species at a time, fits a phylogenetic
logistic regression model using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, stores the
results and detects influential species.
</p>
<p>Currently only logistic regression using the &quot;logistic_MPLE&quot;-method from
<code>phyloglm</code> is implemented.
</p>
<p><code>influ_phyglm</code> detects influential species based on the standardised
difference in intercept and/or slope when removing a given species compared
to the full model including all species. Species with a standardised difference
above the value of <code>cutoff</code> are identified as influential. The default
value for the cutoff is 2 standardised differences change.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>influ_phyglm</code> returns a list with the following
components:
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (i.e. <code>alpha</code>) for the full model
without deleted species.
</p>
<p><code>influential_species</code>: List of influential species, both
based on standardised difference in intercept and in the slope of the
regression. Species are ordered from most influential to less influential and
only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the standardised
difference (<code>sDIFintercept</code>), the percentage of change in intercept compared
to the full model (<code>intercept.perc</code>) and intercept p-value
(<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter
(i.e. <code>alpha</code>) are reported.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Species where deletion resulted in errors.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+samp_phyglm">samp_phyglm</a></code>,
<code><a href="#topic+influ_phylm">influ_phylm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate Data:
set.seed(6987)
phy = rtree(100)
x = rTrait(n=1,phy=phy)
X = cbind(rep(1,100),x)
y = rbinTrait(n=1,phy=phy, beta=c(-1,0.5), alpha=.7 ,X=X)
dat = data.frame(y, x)
# Run sensitivity analysis:
influ &lt;- influ_phyglm(y ~ x, data = dat, phy = phy) 
# To check summary results and most influential species:
summary(influ)
# Visual diagnostics for clade removal:
sensi_plot(influ)
</code></pre>

<hr>
<h2 id='influ_phylm'>Influential species detection - Phylogenetic Linear Regression</h2><span id='topic+influ_phylm'></span>

<h3>Description</h3>

<p>Performs leave-one-out deletion analysis for phylogenetic linear regression,
and detects influential species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>influ_phylm(
  formula,
  data,
  phy,
  model = "lambda",
  cutoff = 2,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="influ_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="influ_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="influ_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="influ_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="influ_phylm_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff value used to identify for influential species
(see Details)</p>
</td></tr>
<tr><td><code id="influ_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="influ_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one species at a time, fits a phylogenetic
linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, stores the
results and detects influential species.
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p><code>influ_phylm</code> detects influential species based on the standardised
difference in intercept and/or slope when removing a given species compared
to the full model including all species. Species with a standardised difference
above the value of <code>cutoff</code> are identified as influential. The default
value for the cutoff is 2 standardised differences change.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>influ_phylm</code> returns a list with the following
components:
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>influential_species</code>: List of influential species, both
based on standardised difference in intercept and in the slope of the
regression. Species are ordered from most influential to less influential and
only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the standardised
difference (<code>sDIFintercept</code>), the percentage of change in intercept compared
to the full model (<code>intercept.perc</code>) and intercept p-value
(<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) are
reported.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Species where deletion resulted in errors.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+samp_phylm">samp_phylm</a></code>,
<code><a href="#topic+influ_phyglm">influ_phyglm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data:
data(alien)
# run analysis:
influ &lt;- influ_phylm(log(gestaLen) ~ log(adultMass), phy = alien$phy[[1]], 
data = alien$data)
# To check summary results:
summary(influ)
# Most influential speciesL
influ$influential.species
# Visual diagnostics
sensi_plot(influ)
# You can specify which graph and parameter ("estimate" or "intercept") to print: 
sensi_plot(influ, param = "estimate", graphs = 2)
</code></pre>

<hr>
<h2 id='influ_physig'>Influential species detection - Phylogenetic signal</h2><span id='topic+influ_physig'></span>

<h3>Description</h3>

<p>Performs leave-one-out deletion analysis for phylogenetic signal estimates,
and detects influential species for K or lambda.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>influ_physig(trait.col, data, phy, method = "K", cutoff = 2, track = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="influ_physig_+3A_trait.col">trait.col</code></td>
<td>
<p>The name of a column in the provided data frame with trait 
to be analyzed  (e.g. &quot;Body_mass&quot;).</p>
</td></tr>
<tr><td><code id="influ_physig_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="influ_physig_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="influ_physig_+3A_method">method</code></td>
<td>
<p>Method to compute signal: can be &quot;K&quot; or &quot;lambda&quot;.</p>
</td></tr>
<tr><td><code id="influ_physig_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff value used to identify for influential species
(see Details)</p>
</td></tr>
<tr><td><code id="influ_physig_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="influ_physig_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="phytools.html#topic+phylosig">phylosig</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one species at a time, ans estimates phylogenetic
signal (K or lambda) using <code><a href="phytools.html#topic+phylosig">phylosig</a></code>, stores the
results and detects the most influential species.
</p>
<p><code>influ_physig</code> detects influential species based on the standardised
difference in signal estimate (K or lambda) when removing a given species compared
to the full data estimate (with all species). Species with a standardised difference
above the value of <code>cutoff</code> are identified as influential. The default
value for the cutoff is 2 standardised differences in signal estimate.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>influ_physig</code> returns a list with the following
components:
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>trait.col</code>: Column name of the trait analysed
</p>
<p><code>full.data.estimates</code>: Phylogenetic signal estimate (K or lambda)
and the P value (for the full data).
</p>
<p><code>influential_species</code>: List of influential species,
based on standardised difference in K or lambda. 
Species are ordered from most influential to less influential and
only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>influ.physig.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted species 
Columns report the calculated signal estimate (<code>k</code>) or (<code>lambda</code>),
difference between signal estimation of the reduced and full data 
(<code>DF</code>), the percentage of change in signal compared
to the full data signal (<code>perc</code>) and p-value for the phylogenetic signal
test (<code>pval</code>)
</p>
<p><code>data</code>: Original full dataset.
</p>


<h3>Note</h3>

<p>The argument &quot;se&quot; from <code><a href="phytools.html#topic+phylosig">phylosig</a></code> is not available in this function. Use the 
argument &quot;V&quot; instead with <code><a href="#topic+intra_physig">intra_physig</a></code> to indicate the name of the column containing the standard 
deviation or the standard error of the trait variable instead.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Blomberg, S. P., T. Garland Jr., A. R. Ives (2003) 
Testing for phylogenetic signal in comparative data: 
Behavioral traits are more labile. Evolution, 57, 717-745.
</p>
<p>Pagel, M. (1999) Inferring the historical patterns of biological evolution. 
Nature, 401, 877-884.
</p>
<p>Kamilar, J. M., &amp; Cooper, N. (2013). Phylogenetic signal in primate behaviour,
ecology and life history. Philosophical Transactions of the Royal Society 
B: Biological Sciences, 368: 20120341.
</p>


<h3>See Also</h3>

<p><code><a href="phytools.html#topic+phylosig">phylosig</a></code>,
<code><a href="#topic+influ_phylm">influ_phylm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# Load data:
data(alien)
# Logtransform data
alien.data$logMass &lt;- log(alien.data$adultMass) 
# Run sensitivity analysis:
influ &lt;- influ_physig("logMass", data = alien.data, phy = alien.phy[[1]])
# To check summary results:
summary(influ)
# Most influential speciesL
influ$influential.species
# Visual diagnostics
sensi_plot(influ)
# You can specify which graph to print: 
sensi_plot(influ, graphs = 1)
sensi_plot(influ, graphs = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='intra_clade_phyglm'>Interaction between intraspecific variability and influential clade detection - Phylogenetic Logistic Regression</h2><span id='topic+intra_clade_phyglm'></span>

<h3>Description</h3>

<p>Estimate the impact on model estimates of phylogenetic logistic regression after 
removing clades from the analysis, while taking into account potential
interactions with intraspecific variability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_clade_phyglm(
  formula,
  data,
  phy,
  clade.col,
  n.species = 5,
  n.sim = 100,
  n.intra = 2,
  Vx = NULL,
  distrib = "normal",
  x.transf = NULL,
  btol = 50,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_clade_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of datasets resimulated taking into account intraspecific variation (see: <code>"intra_phyglm"</code>)</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx or Vy = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="intra_clade_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one clade at a time, fits a phylogenetic
logistic regression model using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code> and stores the
results. The impact of of a specific clade on model estimates is calculated by the
comparison between the full model (with all species) and the model without 
the species belonging to a clade. This operation is repeated <code>n.intra</code> times for
simulated values of the dataset, taking into account intraspecific variation. At each iteration, the function 
generates a random value for each row in the dataset using the standard deviation or errors supplied, and 
detect the influential species within that iteration. 
</p>
<p>Additionally, to account for the influence of the number of species on each 
clade (clade sample size), this function also estimate a null distribution 
expected for the number of species in a given clade. This is done by fitting
models without the same number of species in the given clade. 
The number of simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>All phylogenetic models from <code>phyloglm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phyloglm</code> for details.
</p>
<p><code>clade_phyglm</code> detects influential clades based on
difference in intercept and/or slope when removing a given clade compared
to the full model including all species.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. 
<code class="reqn">y = a + bx</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>intra_clade_phyglm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the percentage of change
in intercept compared to the full model (<code>intercept.perc</code>) and intercept
p-value (<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter 
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) 
are reported.
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distributions
for all clades analysed.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Clades and/or iterations where deletion resulted in errors.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+intra_phyglm">intra_phyglm</a></code>,
<code><a href="#topic+clade_phyglm">clade_phyglm</a></code>,  <code><a href="#topic+intra_clade_phylm">intra_clade_phylm</a></code>, 
<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed(6987)
phy = rtree(100)
x = rTrait(n=1,phy=phy,parameters=list(ancestral.state=2,optimal.value=2,sigma2=1,alpha=1))
X = cbind(rep(1,100),x)
y = rbinTrait(n=1,phy=phy, beta=c(-1,0.5), alpha=.7 ,X=X)
z = rnorm(n = length(x),mean = mean(x),sd = 0.1*mean(x))
cla &lt;- rep(c("A","B","C","D"), each = 25)
dat = data.frame(y, x, z, cla)
intra_clade &lt;- intra_clade_phyglm(formula=y ~ x, data = dat, phy = phy,
                                  clade.col = "cla", n.sim = 30, n.intra = 3,
                                  x.transf = log, Vx = "z", distrib="normal")
sensi_plot(intra_clade)
sensi_plot(intra_clade, clade = "B", graphs = 2)

## End(Not run)

</code></pre>

<hr>
<h2 id='intra_clade_phylm'>Interaction between intraspecific variability and influential clade detection - Phylogenetic Linear Regression</h2><span id='topic+intra_clade_phylm'></span>

<h3>Description</h3>

<p>Estimate the impact on model estimates of phylogenetic linear regression after 
removing clades from the analysis, while taking into account potential
interactions with intraspecific variability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_clade_phylm(
  formula,
  data,
  phy,
  clade.col,
  n.species = 5,
  n.sim = 100,
  n.intra = 2,
  Vy = NULL,
  Vx = NULL,
  distrib = "normal",
  y.transf = NULL,
  x.transf = NULL,
  model = "lambda",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_clade_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of datasets resimulated taking into account intraspecific variation (see: <code>"intra_phylm"</code>)</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_vy">Vy</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the response 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code>.</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx or Vy = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_y.transf">y.transf</code></td>
<td>
<p>Transformation for the response variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="intra_clade_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one clade at a time, fits a phylogenetic
linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code> and stores the
results. The impact of of a specific clade on model estimates is calculated by the
comparison between the full model (with all species) and the model without 
the species belonging to a clade. This operation is repeated <code>n.intra</code> times for
simulated values of the dataset, taking into account intraspecific variation. At each iteration, the function 
generates a random value for each row in the dataset using the standard deviation or errors supplied, and 
detect the influential species within that iteration. 
</p>
<p>Additionally, to account for the influence of the number of species on each 
clade (clade sample size), this function also estimates a null distribution 
expected for the number of species in a given clade. This is done by fitting
models without the same number of species in the given clade. 
The number of simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p><code>clade_phylm</code> detects influential clades based on
difference in intercept and/or slope when removing a given clade compared
to the full model including all species.
</p>
<p>Currently, this function can only implement simple linear models (i.e. 
<code class="reqn">y = a + bx</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>intra_clade_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the percentage of change
in intercept compared to the full model (<code>intercept.perc</code>) and intercept
p-value (<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter 
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) 
are reported.
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distributions
for all clades analysed.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Clades and/or iterations where deletion resulted in errors.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+intra_phylm">intra_phylm</a></code>,
<code><a href="#topic+clade_phylm">clade_phylm</a></code>, <code><a href="#topic+intra_clade_phyglm">intra_clade_phyglm</a></code>, 
<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#load data
data(alien)
intra_clade &lt;- intra_clade_phylm(gestaLen ~ adultMass, phy = alien$phy[[1]],
 data = alien$data, clade.col = "family", n.sim = 30, n.intra = 5, 
 y.transf = log, x.transf = log, Vy="SD_gesta")
summary(intra_clade)
sensi_plot(intra_clade)
sensi_plot(intra_clade, clade = "Bovidae", graphs = 2)
sensi_plot(intra_clade, clade = "Mustelidae", graphs = 2)

## End(Not run)

</code></pre>

<hr>
<h2 id='intra_influ_phyglm'>Interaction between intraspecific variability and influential species - Phylogenetic Logistic Regression</h2><span id='topic+intra_influ_phyglm'></span>

<h3>Description</h3>

<p>Performs leave-one-out deletion analysis for phylogenetic logistic regression,
and detects influential species, while taking into account potential
interactions with intraspecific variability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_influ_phyglm(
  formula,
  data,
  phy,
  Vx = NULL,
  n.intra = 30,
  x.transf = NULL,
  distrib = "normal",
  cutoff = 2,
  btol = 50,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_influ_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula:  <code>response~predictor</code>.</p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of datasets resimulated taking into account intraspecific variation (see: <code>"intra_phylgm"</code>)</p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff value used to identify for influential species
(see Details)</p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="intra_influ_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a phylogenetic linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, and detects
influential species by sequentially deleting one at a time. The regression is repeated <code>n.intra</code> times for 
simulated values of the dataset, taking into account intraspecific variation. At each iteration, the function 
generates a random value for each row in the dataset using the standard deviation or errors supplied, and 
detect the influential species within that iteration. 
</p>
<p><code>influ_phylm</code> detects influential species based on the standardised
difference in intercept and/or slope when removing a given species compared
to the full model including all species. Species with a standardised difference
above the value of <code>cutoff</code> are identified as influential. The default
value for the cutoff is 2 standardised differences change.
</p>
<p>Currently, this function can only implement simple models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>intra_influ_phylm</code> returns a list with the following
components:
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>influential_species</code>: List of influential species, both
based on standardised difference in intercept and in the slope of the
regression. Species are ordered from most influential to less influential and
only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade for an iteration of resimulated
data. Columns report the calculated regression intercept (<code>intercept</code>), 
difference between simulation intercept and full model intercept (<code>DIFintercept</code>), 
the standardised difference (<code>sDIFintercept</code>), the percentage of change in intercept compared
to the full model (<code>intercept.perc</code>) and intercept p-value
(<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) are
reported.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Species where deletion resulted in errors.
</p>


<h3>Warning</h3>

<p>When Vx exceeds X negative (or null) values can be generated, this might cause problems
for data transformation (e.g. log-transformation). In these cases, the function will skip the simulation. This problem can
be solved by increasing <code>n.intra</code>, changing the transformation type and/or checking the target species in output$sp.pb.
</p>
<p>Setting <code>n.intra</code> at high values can take a long time to execute, since the total number of iterations equals <code>n.intra * nrow(data)</code>.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+intra_phyglm">intra_phyglm</a></code>,
<code><a href="#topic+influ_phyglm">influ_phyglm</a></code>,<code><a href="#topic+intra_influ_phylm">intra_influ_phylm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Generate data
set.seed(6987)
phy = rtree(100)
x = rTrait(n=1,phy=phy,parameters=list(ancestral.state=2,optimal.value=2,sigma2=1,alpha=1))
X = cbind(rep(1,100),x)
y = rbinTrait(n=1,phy=phy, beta=c(-1,0.5), alpha=.7 ,X=X)
z = rnorm(n = length(x),mean = mean(x),sd = 0.1*mean(x))
dat = data.frame(y, x, z)
# Run sensitivity analysis:
intra_influ &lt;- intra_influ_phyglm(formula = y ~ x, data = dat, phy = phy, 
                       Vx = "z", n.intra = 5,track = TRUE,distrib="normal",x.transf=NULL) 
# To check summary results and most influential species:
summary(intra_influ)
# Visual diagnostics for clade removal:
sensi_plot(intra_influ)

## End(Not run)

</code></pre>

<hr>
<h2 id='intra_influ_phylm'>Interaction between intraspecific variability and influential species - Phylogenetic Linear Regression</h2><span id='topic+intra_influ_phylm'></span>

<h3>Description</h3>

<p>Performs leave-one-out deletion analysis for phylogenetic linear regression,
and detects influential species, while taking into account potential
interactions with intraspecific variability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_influ_phylm(
  formula,
  data,
  phy,
  Vy = NULL,
  Vx = NULL,
  y.transf = NULL,
  x.transf = NULL,
  n.intra = 10,
  distrib = "normal",
  model = "lambda",
  cutoff = 2,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_influ_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula:  <code>response~predictor</code>.</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_vy">Vy</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the response 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code>.</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_y.transf">y.transf</code></td>
<td>
<p>Transformation for the response variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of datasets resimulated taking into account intraspecific variation (see: <code>"intra_phylm"</code>)</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx or Vy = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff value used to identify for influential species
(see Details)</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="intra_influ_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a phylogenetic linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, and detects
influential species by sequentially deleting one at a time. The regression is repeated <code>n.intra</code> times for 
simulated values of the dataset, taking into account intraspecific variation. At each iteration, the function 
generates a random value for each row in the dataset using the standard deviation or errors supplied, and 
detect the influential species within that iteration. 
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p><code>influ_phylm</code> detects influential species based on the standardised
difference in intercept and/or slope when removing a given species compared
to the full model including all species. Species with a standardised difference
above the value of <code>cutoff</code> are identified as influential. The default
value for the cutoff is 2 standardised differences change.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>intra_influ_phylm</code> returns a list with the following
components:
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>influential_species</code>: List of influential species, both
based on standardised difference in intercept and in the slope of the
regression. Species are ordered from most influential to less influential and
only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade for an iteration of resimulated
data. Columns report the calculated regression intercept (<code>intercept</code>), 
difference between simulation intercept and full model intercept (<code>DIFintercept</code>), 
the standardised difference (<code>sDIFintercept</code>), the percentage of change in intercept compared
to the full model (<code>intercept.perc</code>) and intercept p-value
(<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) are
reported.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Species where deletion resulted in errors.
</p>


<h3>Warning</h3>

<p>When Vy or Vx exceed Y or X, respectively, negative (or null) values can be generated, this might cause problems
for data transformation (e.g. log-transformation). In these cases, the function will skip the simulation. 
</p>
<p>Setting <code>n.intra</code> at high values can take a long time to execute, since the total number of iterations 
equals <code>n.intra * nrow(data)</code>.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+intra_phylm">intra_phylm</a></code>,
<code><a href="#topic+influ_phylm">influ_phylm</a></code>,<code><a href="#topic+intra_influ_phyglm">intra_influ_phyglm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load data:
data(alien)
# run analysis:
intra_influ &lt;- intra_influ_phylm(formula = gestaLen ~ adultMass, phy = alien$phy[[1]],
data=alien$data,model="lambda",y.transf = log,x.transf = NULL,Vy="SD_gesta",Vx=NULL,
n.intra=30,distrib = "normal")
summary(intra_influ)
sensi_plot(intra_influ)

## End(Not run)

</code></pre>

<hr>
<h2 id='intra_phyglm'>Intraspecific variability - Phylogenetic Logistic Regression</h2><span id='topic+intra_phyglm'></span>

<h3>Description</h3>

<p>Performs Phylogenetic logistic regression evaluating
intraspecific variability in predictor variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_phyglm(
  formula,
  data,
  phy,
  Vx = NULL,
  n.intra = 30,
  x.transf = NULL,
  distrib = "normal",
  btol = 50,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula: <code>response~predictor</code>.</p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with species as row names.</p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of times to repeat the analysis generating a random value for the predictor variable.
If NULL, <code>n.intra</code> = 2</p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>log</code> or <code>sqrt</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the
predictor variable. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="intra_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a phylogenetic logistic regression model using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>.
The regression is repeated <code>n.intra</code> times. At each iteration the function generates a random value
for each row in the dataset using the standard deviation or error supplied and assuming a normal or uniform distribution.
To calculate means and se for your raw data, you can use the <code>summarySE</code> function from the 
package <code>Rmisc</code>.
</p>
<p>All phylogenetic models from <code>phyloglm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phyloglm</code> for details.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>intra_phyglm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>data</code>: Original full dataset
</p>
<p><code>sensi.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for each regression.
</p>
<p><code>N.obs</code>: Size of the dataset after matching it with tree tips and removing NA's.
</p>
<p><code>stats</code>: Main statistics for model parameters.<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>
<p><code>all.stats</code>: Complete statistics for model parameters. <code>sd_intra</code> is the standard deviation 
due to intraspecific variation. <code>CI_low</code> and <code>CI_high</code> are the lower and upper limits 
of the 95
</p>
<p><code>sp.pb</code>: Species that caused problems with data transformation (see details above).
</p>


<h3>Warning</h3>

<p>When Vx exceeds X negative (or null) values can be generated, this might cause problems
for data transformation (e.g. log-transformation). In these cases, the function will skip the simulation. This problem can
be solved by increasing <code>n.intra</code>, changing the transformation type and/or checking the target species in output$sp.pb.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Pablo Ariel Martinez
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Martinez, P. a., Zurano, J.P., Amado, T.F., Penone, C., Betancur-R, R., 
Bidau, C.J. &amp; Jacobina, U.P. (2015). Chromosomal diversity in tropical reef 
fishes is related to body size and depth range. Molecular Phylogenetics and 
Evolution, 93, 1-4
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate Data:
set.seed(6987)
phy = rtree(150)
x = rTrait(n=1,phy=phy)
x_sd = rnorm(150,mean = 0.8,sd=0.2)
X = cbind(rep(1,150),x)
y = rbinTrait(n=1,phy=phy, beta=c(-1,0.5), alpha=.7 ,X=X)
dat = data.frame(y, x, x_sd)

# Run phylogenetic logistic regression accounting for intraspecific variation:
intra_glm &lt;- intra_phyglm(y~x,Vx = "x_sd",data = dat,phy=phy,distrib = "normal")

#Print summary of sensitivity analysis
summary(intra_glm)
head(intra_glm$sensi.estimates)
#Visual output
sensi_plot(intra_glm)
</code></pre>

<hr>
<h2 id='intra_phylm'>Intraspecific variability - Phylogenetic Linear Regression</h2><span id='topic+intra_phylm'></span>

<h3>Description</h3>

<p>Performs Phylogenetic linear regression evaluating
intraspecific variability in response and/or predictor variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_phylm(
  formula,
  data,
  phy,
  Vy = NULL,
  Vx = NULL,
  y.transf = NULL,
  x.transf = NULL,
  n.intra = 30,
  distrib = "normal",
  model = "lambda",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula: <code>response~predictor</code>.</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits and species names as row names.</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_vy">Vy</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the response 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code>.</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_y.transf">y.transf</code></td>
<td>
<p>Transformation for the response variable (e.g. <code>log</code> or <code>sqrt</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>log</code> or <code>sqrt</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of times to repeat the analysis generating a random value for response and/or predictor variables.
If NULL, <code>n.intra</code> = 30</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx or Vy = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="intra_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a phylogenetic linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>.
The regression is repeated <code>n.intra</code> times. At each iteration the function generates a random value
for each row in the dataset using the standard deviation or errors supplied and assuming a normal or uniform distribution.
To calculate means and se for your raw data, you can use the <code>summarySE</code> function from the 
package <code>Rmisc</code>.
</p>
<p>#' All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>intra_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>data</code>: Original full dataset
</p>
<p><code>sensi.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for each regression.
</p>
<p><code>N.obs</code>: Size of the dataset after matching it with tree tips and removing NA's.
</p>
<p><code>stats</code>: Main statistics for model parameters.<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>
<p><code>all.stats</code>: Complete statistics for model parameters. <code>sd_intra</code> is the standard deviation 
due to intraspecific variation. <code>CI_low</code> and <code>CI_high</code> are the lower and upper limits 
of the 95
</p>
<p><code>sp.pb</code>: Species that caused problems with data transformation (see details above).
</p>


<h3>Warning</h3>

<p>When Vy or Vx exceed Y or X, respectively, negative (or null) values can be generated, this might cause problems
for data transformation (e.g. log-transformation). In these cases, the function will skip the simulation. This problem can
be solved by increasing <code>n.intra</code>, changing the transformation type and/or checking the target species in output$sp.pb.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Pablo Ariel Martinez
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Martinez, P. a., Zurano, J.P., Amado, T.F., Penone, C., Betancur-R, R., 
Bidau, C.J. &amp; Jacobina, U.P. (2015). Chromosomal diversity in tropical reef 
fishes is related to body size and depth range. Molecular Phylogenetics and 
Evolution, 93, 1-4
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data:
data(alien)
# run PGLS accounting for intraspecific variation:
intra &lt;- intra_phylm(gestaLen ~ adultMass, y.transf = log, x.transf = log, 
phy = alien$phy[[1]], data = alien$data, Vy = "SD_gesta", n.intra = 30)
# To check summary results:
summary(intra)
# Visual diagnostics
sensi_plot(intra)

</code></pre>

<hr>
<h2 id='intra_physig'>Intraspecific variability - Phylogenetic signal</h2><span id='topic+intra_physig'></span>

<h3>Description</h3>

<p>Performs Phylogenetic signal estimates evaluating
trait intraspecific variability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_physig(
  trait.col,
  data,
  phy,
  V = NULL,
  n.intra = 100,
  distrib = "normal",
  method = "K",
  track = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_physig_+3A_trait.col">trait.col</code></td>
<td>
<p>The name of a column in the provided data frame with trait 
to be analyzed  (e.g. &quot;Body_mass&quot;).</p>
</td></tr>
<tr><td><code id="intra_physig_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="intra_physig_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="intra_physig_+3A_v">V</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the trait 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code>.</p>
</td></tr>
<tr><td><code id="intra_physig_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of times to repeat the analysis generating a random trait value.
If NULL, <code>n.intra</code> = 30</p>
</td></tr>
<tr><td><code id="intra_physig_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx or Vy = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_physig_+3A_method">method</code></td>
<td>
<p>Method to compute signal: can be &quot;K&quot; or &quot;lambda&quot;.</p>
</td></tr>
<tr><td><code id="intra_physig_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function estimates phylogenetic signal using <code><a href="phytools.html#topic+phylosig">phylosig</a></code>.
The analysis is repeated <code>n.intra</code> times. At each iteration the function generates a random value
for each row in the dataset using the standard deviation or errors supplied and assuming a normal or uniform distribution.
To calculate means and se for your raw data, you can use the <code>summarySE</code> function from the 
package <code>Rmisc</code>. 
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>intra_physig</code> returns a list with the following
components:
</p>
<p><code>Trait</code>: Column name of the trait analysed
</p>
<p><code>data</code>: Original full dataset
</p>
<p><code>intra.physig.estimates</code>: Run number, phylogenetic signal estimate 
(lambda or K) and the p-value for each run with a different simulated datset.
</p>
<p><code>N.obs</code>: Size of the dataset after matching it with tree tips and removing NA's.
</p>
<p><code>stats</code>: Main statistics for signal estimate<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>


<h3>Note</h3>

<p>The argument &quot;se&quot; from <code><a href="phytools.html#topic+phylosig">phylosig</a></code> is not available in this function. Use the 
argument &quot;V&quot; instead with <code><a href="#topic+intra_physig">intra_physig</a></code> to indicate the name of the column containing the standard 
deviation or the standard error of the trait variable instead.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Pablo Ariel Martinez &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Martinez, P. a., Zurano, J.P., Amado, T.F., Penone, C., Betancur-R, R., 
Bidau, C.J. &amp; Jacobina, U.P. (2015). Chromosomal diversity in tropical reef 
fishes is related to body size and depth range. Molecular Phylogenetics and 
Evolution, 93, 1-4
</p>
<p>Blomberg, S. P., T. Garland Jr., A. R. Ives (2003) 
Testing for phylogenetic signal in comparative data: 
Behavioral traits are more labile. Evolution, 57, 717-745.
</p>
<p>Pagel, M. (1999) Inferring the historical patterns of biological evolution. 
Nature, 401, 877-884.
</p>
<p>Kamilar, J. M., &amp; Cooper, N. (2013). Phylogenetic signal in primate behaviour,
ecology and life history. Philosophical Transactions of the Royal Society 
B: Biological Sciences, 368: 20120341.
</p>


<h3>See Also</h3>

<p><code><a href="phytools.html#topic+phylosig">phylosig</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(alien)
alien.data&lt;-alien$data
alien.phy&lt;-alien$phy
# Run sensitivity analysis:
intra &lt;- intra_physig(trait.col = "gestaLen", V = "SD_gesta" ,
                     data = alien.data, phy = alien.phy[[1]])
summary(intra)
sensi_plot(intra)
sensi_plot(intra, graphs = 1)
sensi_plot(intra, graphs = 2)

## End(Not run)


</code></pre>

<hr>
<h2 id='intra_samp_phyglm'>Interaction between intraspecific variability and species sampling - Phylogenetic Logistic Regression</h2><span id='topic+intra_samp_phyglm'></span>

<h3>Description</h3>

<p>Performs analyses of sensitivity to species sampling by randomly removing
species and detecting the effects on parameter estimates in a phylogenetic
logistic regression, while taking into account potential
interactions with intraspecific variability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_samp_phyglm(
  formula,
  data,
  phy,
  n.sim = 10,
  n.intra = 3,
  breaks = seq(0.1, 0.5, 0.1),
  btol = 50,
  Vx = NULL,
  distrib = "normal",
  x.transf = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_samp_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times species are randomly deleted for each
<code>break</code>.</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of datasets resimulated taking into account intraspecific variation (see: <code>"intra_phyglm"</code>)</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="intra_samp_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by
<code>breaks</code>) from the full phylogenetic logistic regression, fits a phylogenetic
logistic regression model without these species using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>,
repeats this many times (controlled by <code>n.sim</code>), stores the results and
calculates the effects on model parameters. 
This operation is repeated <code>n.intra</code> times for simulated values of the dataset, 
taking into account intraspecific variation. At each iteration, the function generates a 
random value for each row in the dataset using the standard deviation or errors supplied, and 
evaluates the effects of sampling within that iteration.
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>samp_phyglm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code> or <code>kappa</code>) for
the full model without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a model rerun with a given number of species
<code>n.remov</code> removed, representing <code>n.percent</code> of the full dataset.
Columns report the calculated regression intercept (<code>intercept</code>),
difference between simulation intercept and full model intercept (<code>DIFintercept</code>),
the percentage of change in intercept compared to the full model (<code>intercept.perc</code>)
and intercept p-value (<code>pval.intercept</code>). All these parameters are also reported
for the regression slope (<code>DIFestimate</code> etc.). Additionally, model aic value
(<code>AIC</code>) and the optimised value (<code>optpar</code>) of the phylogenetic
parameter (e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model
used) are reported. Lastly we reported the standardised difference in intercept 
(<code>sDIFintercept</code>) and slope (<code>sDIFestimate</code>).
</p>
<p><code>sign.analysis</code> For each break (i.e. each percentage of species
removed) this reports the percentage of statistically significant (at p&lt;0.05)
intercepts (<code>perc.sign.intercept</code>) over all repetitions as well as the
percentage of statisticaly significant (at p&lt;0.05) slopes (<code>perc.sign.estimate</code>).
</p>
<p><code>data</code>: Original full dataset.
</p>


<h3>Note</h3>

<p>Please be aware that dropping species may reduce power to detect 
significant slopes/intercepts and may partially be responsible for a potential 
effect of species removal on p-values. Please also consult standardised differences
in the (summary) output.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Gijsbert D.A. Werner &amp; Caterina Penone
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014).
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation
in angiosperms. Nature Communications, 5, 4087.
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+samp_phyglm">samp_phyglm</a></code>,
<code><a href="#topic+intra_phyglm">intra_phyglm</a></code>,<code><a href="#topic+intra_samp_phylm">intra_samp_phylm</a></code>,
<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed(6987)
phy = rtree(100)
x = rTrait(n=1,phy=phy,parameters=list(ancestral.state=2,optimal.value=2,sigma2=1,alpha=1))
X = cbind(rep(1,100),x)
y = rbinTrait(n=1,phy=phy, beta=c(-1,0.5), alpha=.7 ,X=X)
z = rnorm(n = length(x),mean = mean(x),sd = 0.1*mean(x))
dat = data.frame(y, x, z)
#Run sensitivity analysis:
intra_samp &lt;- intra_samp_phyglm(formula = y ~ x, data = dat, phy = phy, 
                               n.sim=10, n.intra = 3,
                               breaks=seq(.1,.5,.1),
                               Vx = "z", distrib="normal", x.transf=NULL)
summary(intra_samp)
sensi_plot(intra_samp)

## End(Not run)
</code></pre>

<hr>
<h2 id='intra_samp_phylm'>Interaction between intraspecific variability and species sampling - Phylogenetic Linear Regression</h2><span id='topic+intra_samp_phylm'></span>

<h3>Description</h3>

<p>Performs analyses of sensitivity to species sampling by randomly removing
species and detecting the effects on parameter estimates in a phylogenetic
linear regression, while taking into account potential
interactions with intraspecific variability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intra_samp_phylm(
  formula,
  data,
  phy,
  n.sim = 10,
  n.intra = 3,
  breaks = seq(0.1, 0.5, 0.1),
  model = "lambda",
  Vy = NULL,
  Vx = NULL,
  distrib = "normal",
  y.transf = NULL,
  x.transf = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intra_samp_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times species are randomly deleted for each
<code>break</code>.</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of datasets resimulated taking into account intraspecific variation (see: <code>"intra_phylm"</code>)</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.
#' @param Vy Name of the column containing the standard deviation or the standard error of the response 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code>.</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_vy">Vy</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the response 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code>.</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx or Vy = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_y.transf">y.transf</code></td>
<td>
<p>Transformation for the response variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>"log"</code> or <code>"sqrt"</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="intra_samp_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by
<code>breaks</code>) from the full phylogenetic linear regression, fits a phylogenetic
linear regression model without these species using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>,
repeats this many times (controlled by <code>n.sim</code>), stores the results and
calculates the effects on model parameters. 
This operation is repeated <code>n.intra</code> times for simulated values of the dataset, 
taking into account intraspecific variation. At each iteration, the function generates a 
random value for each row in the dataset using the standard deviation or errors supplied, and 
evaluates the effects of sampling within that iteration.
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>samp_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code> or <code>kappa</code>) for
the full model without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a model rerun with a given number of species
<code>n.remov</code> removed, representing <code>n.percent</code> of the full dataset.
Columns report the calculated regression intercept (<code>intercept</code>),
difference between simulation intercept and full model intercept (<code>DIFintercept</code>),
the percentage of change in intercept compared to the full model (<code>intercept.perc</code>)
and intercept p-value (<code>pval.intercept</code>). All these parameters are also reported
for the regression slope (<code>DIFestimate</code> etc.). Additionally, model aic value
(<code>AIC</code>) and the optimised value (<code>optpar</code>) of the phylogenetic
parameter (e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model
used) are reported. Lastly we reported the standardised difference in intercept 
(<code>sDIFintercept</code>) and slope (<code>sDIFestimate</code>).
</p>
<p><code>sign.analysis</code> For each break (i.e. each percentage of species
removed) this reports the percentage of statistically significant (at p&lt;0.05)
intercepts (<code>perc.sign.intercept</code>) over all repetitions as well as the
percentage of statisticaly significant (at p&lt;0.05) slopes (<code>perc.sign.estimate</code>).
</p>
<p><code>data</code>: Original full dataset.
</p>


<h3>Note</h3>

<p>Please be aware that dropping species may reduce power to detect 
significant slopes/intercepts and may partially be responsible for a potential 
effect of species removal on p-values. Please also consult standardised differences
in the (summary) output.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Gijsbert D.A. Werner &amp; Caterina Penone
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014).
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation
in angiosperms. Nature Communications, 5, 4087.
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>,<code><a href="#topic+samp_phylm">samp_phylm</a></code>, 
<code><a href="#topic+intra_phylm">intra_phylm</a></code>, <code><a href="#topic+intra_samp_phyglm">intra_samp_phyglm</a></code>,
<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load data:
data(alien)
# Run analysis:
samp &lt;- intra_samp_phylm(gestaLen ~ adultMass, phy = alien$phy[[1]],
                         y.transf = log,x.transf = NULL,Vy="SD_gesta",Vx=NULL,
                         data = alien$data, n.intra = 5, n.sim=10)
summary(samp)
head(samp$sensi.estimates)
# Visual diagnostics
sensi_plot(samp)
# You can specify which graph and parameter ("estimate" or "intercept") to print: 
sensi_plot(samp, graphs = 1)
sensi_plot(samp, graphs = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='match_dataphy'>Match data and phylogeny based on model formula</h2><span id='topic+match_dataphy'></span>

<h3>Description</h3>

<p>Combines phylogeny and data to ensure that tips in phylogeny match data and that observations
with missing values are removed. This function uses variables provided in the 
'formula' argument to:
</p>

<ul>
<li><p>Remove NA's:    Check if there is any row with NA in the variables included in 
the formula. All rows containing NA will be removed from the data
</p>
</li>
<li><p>Match data and phy:    Check if tips from phylogeny match rownames in
data. Tips not present in data and phy will be removed from the phylogeny and
data
</p>
</li>
<li><p>Return matched data and phy:    The returned data
has no NA in the variables included in 'formula' and only rows that match phylogeny
tips. Returned phy has only tips that match data
</p>
</li></ul>

<p>Used internally in <code><a href="#topic+samp_phylm">samp_phylm</a></code>, <code><a href="#topic+samp_phyglm">samp_phyglm</a></code>, <code><a href="#topic+clade_phylm">clade_phylm</a></code>,
<code><a href="#topic+clade_phyglm">clade_phyglm</a></code>, <code><a href="#topic+intra_phylm">intra_phylm</a></code>, <code><a href="#topic+intra_phyglm">intra_phyglm</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>,
<code><a href="#topic+tree_phyglm">tree_phyglm</a></code> and all function analysing interactions.
Users can also directly use this function to combine a phylogeny and a dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_dataphy(formula, data, phy, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="match_dataphy_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="match_dataphy_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="match_dataphy_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo' or 'multiphylo')</p>
</td></tr>
<tr><td><code id="match_dataphy_+3A_verbose">verbose</code></td>
<td>
<p>Print the number of species that match data and phylogeny and warnings. We highly recommend to use the 
default (verbose = T), but warning and information can be silenced for advanced use.</p>
</td></tr>
<tr><td><code id="match_dataphy_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>match_dataphy</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses all variables provided in the 'formula' to match
data and phylogeny. To avoid cropping the full dataset, 'match_dataphy' searches
for NA values only on variables provided by formula. Missing values on 
other variables, not included in 'formula', will not be removed from data.
If no species names are provided as row names in the dataset but the number of rows in the dataset
is the same as the number of tips in the phylogeny, the function assumes that the dataset and the 
phylogeny are in the same order.
</p>
<p>This ensures consistency between data and phylogeny only for the variables 
that are being used in the model (set by 'formula').
</p>
<p>If phy is a 'multiphylo' object, all phylogenies will be cropped
to match data. But the dataset order will only match the first tree provided.
The returned phylogeny will be a 'multiphylo' object.
</p>


<h3>Value</h3>

<p>The function <code>match_dataphy</code> returns a list with the following
components:
</p>
<p><code>data</code>: Cropped dataset matching phylogeny
</p>
<p><code>phy</code>: Cropped phylogeny matching data
</p>
<p><code>dropped</code>: Species dropped from phylogeny and removed from data.
</p>


<h3>Note</h3>

<p>If tips are removed from the phylogeny and data or if rows containing
missing values are removed from data, a message will be printed with the 
details. Further, the final number of species that match data and phy will
always be reported by a message.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>This function is largely inspired by the function <code>comparative.data</code> in caper package
David Orme, Rob Freckleton, Gavin Thomas, Thomas Petzoldt, Susanne Fritz, Nick Isaac and Will Pearse
(2013). caper: Comparative Analyses of Phylogenetics and Evolution in R. R package version 0.5.2.
http://CRAN.R-project.org/package=caper
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data:
data(alien)
head(alien$data)
# Match data and phy based on model formula:
comp.data &lt;- match_dataphy(gestaLen ~ homeRange, data = alien$data, alien$phy[[1]])
# Check data:
head(comp.data$data)
# Check phy:
comp.data$phy
# See species dropped from phy or data:
comp.data$dropped
# Example2:
# Match data and phy based on model formula:
comp.data2 &lt;- match_dataphy(gestaLen ~ adultMass, data = alien$data, alien$phy)
# Check data (missing data on variables not included in the formula are preserved)
head(comp.data2$data)
# Check phy:
comp.data2$phy
# See species dropped from phy or data:
comp.data2$dropped
</code></pre>

<hr>
<h2 id='miss.phylo.d'>Phylogenetic signal in missing data</h2><span id='topic+miss.phylo.d'></span>

<h3>Description</h3>

<p>Calculates D statistic (Fritz &amp; Purvis 2010), a measure of phylogenetic
signal, for missing data. Missingness is recoded into a binary variable 
(1=missing, 0=non missing). This function is an adaptation of
<code><a href="caper.html#topic+phylo.d">phylo.d</a></code> for missing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>miss.phylo.d(data, phy, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="miss.phylo.d_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with species as row names.</p>
</td></tr>
<tr><td><code id="miss.phylo.d_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="miss.phylo.d_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylo.d</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function builds on <code><a href="caper.html#topic+phylo.d">phylo.d</a></code> to calculate a phylogenetic signal
in missing data. The variable of interest, usually a trait, is recoded into a binary variable
(1=missing data, 0=non missing data). Then the <code><a href="caper.html#topic+phylo.d">phylo.d</a></code> function tests the estimated 
D value for significant departure from both random association and the clumping expected under a Brownian 
evolution threshold model (Fritz &amp; Purvis, 2010).
</p>
<p>Output can be visualised using <code>print()</code> and <code>plot()</code>
</p>


<h3>Value</h3>

<p>The function <code>miss.phylo.d</code> returns an object of class &quot;phylo.d&quot; with the following
components, for complete list of arguments see <code><a href="caper.html#topic+phylo.d">phylo.d</a></code> :
</p>
<p><code>DEstimate</code>: The estimated D value
</p>
<p><code>Pval1</code>: A p value, giving the result of testing whether D is significantly different from one
</p>
<p><code>Pval0</code>: A p value, giving the result of testing whether D is significantly different from zero
The function also prints the percentage of missing data per variable in the dataset.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Fritz, S. A. and Purvis, A. (2010). Selectivity in mammalian extinction risk and threat types: a new measure of 
phylogenetic signal strength in binary traits. Conservation Biology, 24(4):1042-1051.
</p>
<p>David Orme, Rob Freckleton, Gavin Thomas, Thomas Petzoldt, Susanne Fritz, Nick Isaac and Will Pearse (2013).
caper: Comparative Analyses of Phylogenetics and Evolution in R. R package version 0.5.2.
https://CRAN.R-project.org/package=caper
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load caper:
library(caper)
# Load data
data(primates)
data&lt;-alien$data
phy=alien$phy[[1]]

# Test phylogenetic signal for missing data:
sexNAsig &lt;- miss.phylo.d(data,phy,binvar=homeRange)
print(sexNAsig)
plot(sexNAsig)

massNAsig &lt;- miss.phylo.d(data,phy,binvar=adultMass)
print(massNAsig)
plot(massNAsig)
</code></pre>

<hr>
<h2 id='primates'>Primates dataset: Example dataset for the package sensiPhy</h2><span id='topic+primates'></span>

<h3>Description</h3>

<p>A comparative dataset containing traits for 95 Primates species 
(primates.data) and a multiphylo object with 101 phylogenies matching the 
data (primates.phy). Tip labels are the binomial species names and match 
with data rownames. Data was taken from (Jones et al. 2009) and phylogenies from
(Fritz et al 2009) and (Kuhn et al 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(primates)
</code></pre>


<h3>Format</h3>

<p>A data frame with 95 rows and 3 variables:
</p>

<ul>
<li><p> family: Taxonomic family
</p>
</li>
<li><p> adultMass: Mean adult body mass (g)
</p>
</li>
<li><p> sexMaturity: Age when individuals are first physically capable of reproducing (days)
</p>
</li>
<li><p> homeRange: Mean home range (km)
</p>
</li></ul>


<h3>References</h3>

<p>Data downloaded from: http://esapubs.org/archive/ecol/E090/184/
</p>
<p>Jones, K. E., Bielby, J., Cardillo, M., Fritz, S. A., O'Dell, J., Orme, C. D. L., Safi, K., Sechrest,
W., Boakes, E. H., Carbone, C., Connolly, C., Cutts, M. J., Foster, J. K., Grenyer, R., Habib, M., Plaster, 
C. A., Price, S. A., Rigby, E. A., Rist, J., Teacher, A., Bininda-Emonds, O. R. P., Gittleman, J. L., 
Mace, G. M., Purvis, A. (2009), PanTHERIA: a species-level database of life history, ecology, 
and geography of extant and recently extinct mammals. Ecology, 90: 2648. doi: 10.1890/08-1494.1
</p>
<p>Phylogeny: Kuhn, Tyler S., Arne O. Mooers, and Gavin H. Thomas. &quot;A simple polytomy resolver for 
dated phylogenies.&quot; Methods in Ecology and Evolution 2.5 (2011): 427-436.
</p>
<p>Fritz, Susanne A., Olaf RP Bininda-Emonds, and Andy Purvis. &quot;Geographical variation in predictors 
of mammalian extinction risk: big is bad, but only in the tropics.&quot; Ecology letters 12.6 (2009): 538-549.
</p>

<hr>
<h2 id='primates.data'>Primates dataset: Example dataset for the package sensiPhy</h2><span id='topic+primates.data'></span>

<h3>Description</h3>

<p>A comparative dataset containing traits for 95 Primates species 
(primates.data) and a multiphylo object with 101 phylogenies matching the 
data (primates.phy). Tip labels are the binomial species names and match 
with data rownames. Data was taken from (Jones et al. 2009) and phylogenies from
(Fritz et al 2009) and (Kuhn et al 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(primates)
</code></pre>


<h3>Format</h3>

<p>A data frame with 95 rows and 3 variables:
</p>

<ul>
<li><p> family: Taxonomic family
</p>
</li>
<li><p> adultMass: Mean adult body mass (g)
</p>
</li>
<li><p> sexMaturity: Age when individuals are first physically capable of reproducing (days)
</p>
</li>
<li><p> homeRange: Mean home range (km)
</p>
</li></ul>


<h3>References</h3>

<p>Data downloaded from: http://esapubs.org/archive/ecol/E090/184/
</p>
<p>Jones, K. E., Bielby, J., Cardillo, M., Fritz, S. A., O'Dell, J., Orme, C. D. L., Safi, K., Sechrest,
W., Boakes, E. H., Carbone, C., Connolly, C., Cutts, M. J., Foster, J. K., Grenyer, R., Habib, M., Plaster, 
C. A., Price, S. A., Rigby, E. A., Rist, J., Teacher, A., Bininda-Emonds, O. R. P., Gittleman, J. L., 
Mace, G. M., Purvis, A. (2009), PanTHERIA: a species-level database of life history, ecology, 
and geography of extant and recently extinct mammals. Ecology, 90: 2648. doi: 10.1890/08-1494.1
</p>
<p>Phylogeny: Kuhn, Tyler S., Arne O. Mooers, and Gavin H. Thomas. &quot;A simple polytomy resolver for 
dated phylogenies.&quot; Methods in Ecology and Evolution 2.5 (2011): 427-436.
</p>
<p>Fritz, Susanne A., Olaf RP Bininda-Emonds, and Andy Purvis. &quot;Geographical variation in predictors 
of mammalian extinction risk: big is bad, but only in the tropics.&quot; Ecology letters 12.6 (2009): 538-549.
</p>

<hr>
<h2 id='primates.phy'>Primates dataset: Example dataset for the package sensiPhy</h2><span id='topic+primates.phy'></span>

<h3>Description</h3>

<p>A comparative dataset containing traits for 95 Primates species 
(primates.data) and a multiphylo object with 101 phylogenies matching the 
data (primates.phy). Tip labels are the binomial species names and match 
with data rownames. Data was taken from (Jones et al. 2009) and phylogenies from
(Fritz et al 2009) and (Kuhn et al 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(primates)
</code></pre>


<h3>Format</h3>

<p>A data frame with 95 rows and 3 variables:
</p>

<ul>
<li><p> family: Taxonomic family
</p>
</li>
<li><p> adultMass: Mean adult body mass (g)
</p>
</li>
<li><p> sexMaturity: Age when individuals are first physically capable of reproducing (days)
</p>
</li>
<li><p> homeRange: Mean home range (km)
</p>
</li></ul>


<h3>References</h3>

<p>Data downloaded from: http://esapubs.org/archive/ecol/E090/184/
</p>
<p>Jones, K. E., Bielby, J., Cardillo, M., Fritz, S. A., O'Dell, J., Orme, C. D. L., Safi, K., Sechrest,
W., Boakes, E. H., Carbone, C., Connolly, C., Cutts, M. J., Foster, J. K., Grenyer, R., Habib, M., Plaster, 
C. A., Price, S. A., Rigby, E. A., Rist, J., Teacher, A., Bininda-Emonds, O. R. P., Gittleman, J. L., 
Mace, G. M., Purvis, A. (2009), PanTHERIA: a species-level database of life history, ecology, 
and geography of extant and recently extinct mammals. Ecology, 90: 2648. doi: 10.1890/08-1494.1
</p>
<p>Phylogeny: Kuhn, Tyler S., Arne O. Mooers, and Gavin H. Thomas. &quot;A simple polytomy resolver for 
dated phylogenies.&quot; Methods in Ecology and Evolution 2.5 (2011): 427-436.
</p>
<p>Fritz, Susanne A., Olaf RP Bininda-Emonds, and Andy Purvis. &quot;Geographical variation in predictors 
of mammalian extinction risk: big is bad, but only in the tropics.&quot; Ecology letters 12.6 (2009): 538-549.
</p>

<hr>
<h2 id='samp_continuous'>Species Sampling uncertainty - Trait Evolution Continuous Characters</h2><span id='topic+samp_continuous'></span>

<h3>Description</h3>

<p>Fits models for trait evolution of continuous characters, 
evaluating sampling uncertainty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samp_continuous(
  data,
  phy,
  n.sim = 30,
  breaks = seq(0.1, 0.5, 0.1),
  model,
  n.cores = NULL,
  bounds = list(),
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="samp_continuous_+3A_data">data</code></td>
<td>
<p>Data vector for a single binary trait, with names matching tips in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="samp_continuous_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="samp_continuous_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times species are randomly deleted for each <code>break</code>.</p>
</td></tr>
<tr><td><code id="samp_continuous_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="samp_continuous_+3A_model">model</code></td>
<td>
<p>The evolutionary model (see Details).</p>
</td></tr>
<tr><td><code id="samp_continuous_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use. If 'NULL', number of cores is detected.</p>
</td></tr>
<tr><td><code id="samp_continuous_+3A_bounds">bounds</code></td>
<td>
<p>settings to constrain parameter estimates. See <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code></p>
</td></tr>
<tr><td><code id="samp_continuous_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="samp_continuous_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by <code>breaks</code>),  
fits different models of continuous character evolution using <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code>, 
repeats this this many times (controlled by <code>n.sim</code>), stores the results and calculates 
the effects on model parameters.
</p>
<p>Different evolutionary models from <code>fitContinuous</code> can be used, i.e. <code>BM</code>,<code>OU</code>,
<code>EB</code>, <code>trend</code>, <code>lambda</code>, <code>kappa</code>, <code>delta</code> and <code>drift</code>.
</p>
<p>See <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code> for more details on character models and tree transformations. 
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_continuous</code> returns a list with the following
components:
</p>
<p><code>call</code>: The function call
</p>
<p><code>data</code>: The original full data vector
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>
<p><code>full.model.estimates</code>: Parameter estimates (rate of evolution <code>sigsq</code>
and where applicable <code>optpar</code>), root state <code>z0</code>, 
AICc for the full model without deleted species.
</p>
<p><code>break.summary.tab</code>: Summary per <code>break</code> of the mean and median effects 
of species removal on percentage and absolute change parameter estimates.
</p>
<p><code>sensi.estimates</code>: Parameter estimates (sigsq and optpar),(percentual) difference 
in parameter estimate compared to the full model (DIFsigsq, sigsq.perc,sDIFsigsq, 
DIFoptpar, optpar.perc,sDIFoptpar), 
AICc and z0 for each repeat with random species removed.
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>


<h3>Author(s)</h3>

<p>Gijsbert Werner &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Yang Z. 2006. Computational Molecular Evolution. Oxford University Press: Oxford. 
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008.
GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014). 
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation in angiosperms. Nature Communications, 5, 4087.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
#Load data:
data("primates")
#Model trait evolution accounting for sampling size 
adultMass&lt;-primates$data$adultMass
names(adultMass)&lt;-rownames(primates$data)
samp_cont&lt;-samp_continuous(data = adultMass,phy = primates$phy[[1]],
model = "OU",n.sim=25,breaks=seq(.05,.2,.05),n.cores = 2, track = TRUE)
#Print summary statistics
summary(samp_cont)
sensi_plot(samp_cont)
sensi_plot(samp_cont, graphs = 1)
#Use a different evolutionary model 
samp_cont2&lt;-samp_continuous(data = adultMass,phy = primates$phy[[1]],
model = "kappa",n.sim=25,breaks=seq(.05,.2,.05),n.cores = 2,track = TRUE)
summary(samp_cont2)
sensi_plot(samp_cont2)
sensi_plot(samp_cont2, graphs = 2)
samp_cont3&lt;-samp_continuous(data = adultMass,phy = primates$phy[[1]],
model = "BM",n.sim=25,breaks=seq(.05,.2,.05),n.cores = 2,track = TRUE)
summary(samp_cont3)

## End(Not run)
</code></pre>

<hr>
<h2 id='samp_discrete'>Species Sampling uncertainty - Trait Evolution Discrete Characters</h2><span id='topic+samp_discrete'></span>

<h3>Description</h3>

<p>Fits models for trait evolution of discrete (binary) characters, 
evaluating sampling uncertainty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samp_discrete(
  data,
  phy,
  n.sim = 30,
  breaks = seq(0.1, 0.5, 0.1),
  model,
  transform = "none",
  bounds = list(),
  n.cores = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="samp_discrete_+3A_data">data</code></td>
<td>
<p>Data vector for a single binary trait, with names matching tips in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times species are randomly deleted for each <code>break</code>.</p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_model">model</code></td>
<td>
<p>The Mkn model to use (see Details).</p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_transform">transform</code></td>
<td>
<p>The evolutionary model to transform the tree (see Details). Default is <code>none</code>.</p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_bounds">bounds</code></td>
<td>
<p>settings to constrain parameter estimates. See <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code></p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use. If 'NULL', number of cores is detected.</p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="samp_discrete_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by <code>breaks</code>),  
fits different models of discrete character evolution using <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code>, 
repeats this this many times (controlled by <code>n.sim</code>), stores the results and calculates 
the effects on model parameters Currently, only binary discrete traits are supported.
</p>
<p>Different character model from <code>fitDiscrete</code> can be used, including <code>ER</code> (equal-rates), 
<code>SYM</code> (symmetric), <code>ARD</code> (all-rates-different) and <code>meristic</code> (stepwise fashion). 
</p>
<p>Transformations to the phylogenetic tree from <code>fitDiscrete</code> can be used, i.e. <code>none</code>,
<code>EB</code>, <code>lambda</code>, <code>kappa</code> and<code>delta</code>.
</p>
<p>See <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code> for more details on character models and tree transformations. 
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_discrete</code> returns a list with the following
components:
</p>
<p><code>call</code>: The function call
</p>
<p><code>data</code>: The original full data vector
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>
<p><code>full.model.estimates</code>: Parameter estimates (transition rates q12 and q21), 
AICc and the optimised value of the phylogenetic transformation parameter (e.g. <code>lambda</code>) 
for the full model without deleted species.
</p>
<p><code>break.summary.tab</code>: Summary per <code>break</code> of the mean and median effects 
of species removal on percentage and absolute change in parameters q12 and q21.
</p>
<p><code>sensi.estimates</code>: Parameter estimates (transition rates q12 and q21),(percentual) difference 
in parameter estimate compared to the full model (DIFq12, sigsq.q12,sDIFq12, DIFq21, optpar.q21,sDIFq21),  
AICc and the optimised value of the phylogenetic transformation parameter (e.g. <code>lambda</code>) 
for each analysis with a species deleted.
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>


<h3>Author(s)</h3>

<p>Gijsbert Werner &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Yang Z. 2006. Computational Molecular Evolution. Oxford University Press: Oxford. 
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008.
GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014). 
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation in angiosperms. Nature Communications, 5, 4087.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load data:
data("primates")
#Create a binary trait factor 
adultMass_binary&lt;-ifelse(primates$data$adultMass &gt; 7350, "big", "small")
adultMass_binary&lt;-as.factor(as.factor(adultMass_binary))
names(adultMass_binary)&lt;-rownames(primates$data)
#Model trait evolution accounting for sampling size 
samp_binary&lt;-samp_discrete(data = adultMass_binary,phy = primates$phy[[1]],
n.sim=25,breaks=seq(.1,.3,.1),model = "SYM",transform = "none",n.cores = 2,track = TRUE)
#Print summary statistics
summary(samp_binary)
sensi_plot(samp_binary)
sensi_plot(samp_binary,graphs=1)
sensi_plot(samp_binary,graphs=2)
#Use a different evolutionary model or transformation 
samp_binary2&lt;-samp_discrete(data = adultMass_binary,phy = primates$phy[[1]],
n.sim=25,breaks=seq(.1,.3,.1),model = "ARD",transform = "lambda",n.cores = 2,track = TRUE)
summary(samp_binary2)
sensi_plot(samp_binary2)
sensi_plot(samp_binary2,graphs=1)
sensi_plot(samp_binary2,graphs=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='samp_phyglm'>Sensitivity Analysis Species Sampling  - Phylogenetic Logistic Regression</h2><span id='topic+samp_phyglm'></span>

<h3>Description</h3>

<p>Performs analyses of sensitivity to species sampling by randomly removing
species and detecting the effects on parameter estimates in phylogenetic
logistic regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samp_phyglm(
  formula,
  data,
  phy,
  n.sim = 30,
  breaks = seq(0.1, 0.5, 0.1),
  btol = 50,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="samp_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="samp_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="samp_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="samp_phyglm_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times species are randomly deleted for each
<code>break</code>.</p>
</td></tr>
<tr><td><code id="samp_phyglm_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="samp_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="samp_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="samp_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by
<code>breaks</code>) from the full phylogenetic logistic regression, fits a phylogenetic
logistic regression model without these species using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>,
repeats this many times (controlled by <code>n.sim</code>), stores the results and
calculates the effects on model parameters.
</p>
<p>Only logistic regression using the &quot;logistic_MPLE&quot;-method from
<code>phyloglm</code> is implemented.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>samp_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code> or <code>kappa</code>) for
the full model without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a model rerun with a given number of species
<code>n.remov</code> removed, representing <code>n.percent</code> of the full dataset.
Columns report the calculated regression intercept (<code>intercept</code>),
difference between simulation intercept and full model intercept (<code>DIFintercept</code>),
the percentage of change in intercept compared to the full model (<code>intercept.perc</code>)
and intercept p-value (<code>pval.intercept</code>). All these parameters are also reported
for the regression slope (<code>DIFestimate</code> etc.). Additionally, model aic value
(<code>AIC</code>) and the optimised value (<code>optpar</code>) of the phylogenetic
parameter (e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model
used) are reported. Lastly we reported the standardised difference in intercept 
(<code>sDIFintercept</code>) and slope (<code>sDIFestimate</code>).
</p>
<p><code>sign.analysis</code> For each break (i.e. each percentage of species
removed) this reports the percentage of statistically significant (at p&lt;0.05)
intercepts (<code>perc.sign.intercept</code>) over all repetitions as well as the
percentage of statisticaly significant (at p&lt;0.05) slopes (<code>perc.sign.estimate</code>).
</p>
<p><code>data</code>: Original full dataset.
</p>


<h3>Note</h3>

<p>Please be aware that dropping species may reduce power to detect 
significant slopes/intercepts and may partially be responsible for a potential 
effect of species removal on p-values. Please also consult standardised differences
in the (summary) output.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014).
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation
in angiosperms. Nature Communications, 5, 4087.
</p>
<p>#' Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+samp_phylm">samp_phylm</a></code>,
<code><a href="#topic+influ_phyglm">influ_phyglm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate Data:
set.seed(6987)
phy = rtree(100)
x = rTrait(n=1,phy=phy)
X = cbind(rep(1,100),x)
y = rbinTrait(n=1,phy=phy, beta=c(-1,0.5), alpha=.7 ,X=X)
dat = data.frame(y, x)
# Run sensitivity analysis:
samp &lt;- samp_phyglm(y ~ x, data = dat, phy = phy, n.sim = 10) 
# To check summary results and most influential species:
summary(samp)
## Not run: 
# Visual diagnostics for clade removal:
sensi_plot(samp)

## End(Not run)
</code></pre>

<hr>
<h2 id='samp_phylm'>Sensitivity Analysis Species Sampling  - Phylogenetic Linear Regression</h2><span id='topic+samp_phylm'></span>

<h3>Description</h3>

<p>Performs analyses of sensitivity to species sampling by randomly removing
species and detecting the effects on parameter estimates in a phylogenetic
linear regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samp_phylm(
  formula,
  data,
  phy,
  n.sim = 30,
  breaks = seq(0.1, 0.5, 0.1),
  model = "lambda",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="samp_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="samp_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="samp_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="samp_phylm_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times species are randomly deleted for each
<code>break</code>.</p>
</td></tr>
<tr><td><code id="samp_phylm_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="samp_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="samp_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="samp_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by
<code>breaks</code>) from the full phylogenetic linear regression, fits a phylogenetic
linear regression model without these species using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>,
repeats this many times (controlled by <code>n.sim</code>), stores the results and
calculates the effects on model parameters.
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>samp_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code> or <code>kappa</code>) for
the full model without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a model rerun with a given number of species
<code>n.remov</code> removed, representing <code>n.percent</code> of the full dataset.
Columns report the calculated regression intercept (<code>intercept</code>),
difference between simulation intercept and full model intercept (<code>DIFintercept</code>),
the percentage of change in intercept compared to the full model (<code>intercept.perc</code>)
and intercept p-value (<code>pval.intercept</code>). All these parameters are also reported
for the regression slope (<code>DIFestimate</code> etc.). Additionally, model aic value
(<code>AIC</code>) and the optimised value (<code>optpar</code>) of the phylogenetic
parameter (e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model
used) are reported. Lastly we reported the standardised difference in intercept 
(<code>sDIFintercept</code>) and slope (<code>sDIFestimate</code>).
</p>
<p><code>sign.analysis</code> For each break (i.e. each percentage of species
removed) this reports the percentage of statistically significant (at p&lt;0.05)
intercepts (<code>perc.sign.intercept</code>) over all repetitions as well as the
percentage of statisticaly significant (at p&lt;0.05) slopes (<code>perc.sign.estimate</code>).
</p>
<p><code>data</code>: Original full dataset.
#' @note Please be aware that dropping species may reduce power to detect 
significant slopes/intercepts and may partially be responsible for a potential 
effect of species removal on p-values. Please also consult standardised differences
in the (summary) output.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014).
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation
in angiosperms. Nature Communications, 5, 4087.
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+samp_phyglm">samp_phyglm</a></code>,
<code><a href="#topic+influ_phylm">influ_phylm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data:
data(alien)
# Run analysis:
samp &lt;- samp_phylm(log(gestaLen) ~ log(adultMass), phy = alien$phy[[1]], 
data = alien$data, n.sim = 10)
summary(samp)
head(samp$sensi.estimates)
# Visual diagnostics
## Not run: 
sensi_plot(samp)
# You can specify which graph and parameter ("estimate" or "intercept") to print: 
sensi_plot(samp, graphs = 1, param = "estimate")
sensi_plot(samp, graphs = 2, param = "intercept")

## End(Not run)
</code></pre>

<hr>
<h2 id='samp_physig'>Sensitivity Analysis Species Sampling  - Phylogenetic signal</h2><span id='topic+samp_physig'></span>

<h3>Description</h3>

<p>Performs analyses of sensitivity to species sampling by randomly removing
species and detecting the effects on phylogenetic 
signal estimates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samp_physig(
  trait.col,
  data,
  phy,
  n.sim = 30,
  breaks = seq(0.1, 0.5, 0.1),
  method = "K",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="samp_physig_+3A_trait.col">trait.col</code></td>
<td>
<p>The name of a column in the provided data frame with trait 
to be analyzed  (e.g. &quot;Body_mass&quot;).</p>
</td></tr>
<tr><td><code id="samp_physig_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="samp_physig_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="samp_physig_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times to repeat species random removal for each
<code>break</code> interval.</p>
</td></tr>
<tr><td><code id="samp_physig_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="samp_physig_+3A_method">method</code></td>
<td>
<p>Method to compute signal: can be &quot;K&quot; or &quot;lambda&quot;.</p>
</td></tr>
<tr><td><code id="samp_physig_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="samp_physig_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="phytools.html#topic+phylosig">phylosig</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by
<code>breaks</code>) from the full data, estimates phylogenetic
signal for a given trait (K or lambda) without these species using 
<code><a href="phytools.html#topic+phylosig">phylosig</a></code>, then
repeats the analysis many times (controlled by <code>n.sim</code>), stores the results and
calculates the effect of random species removal on phylogenetic signal estimates.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>samp_phylosig</code> returns a list with the following
components:
</p>
<p><code>Trait</code>: Column name of the trait analysed
</p>
<p><code>full.model.estimates</code>: Phylogenetic signal (K or lambda) and 
p-value  using the full dataset (without deleted species). See 
<code><a href="phytools.html#topic+phylosig">phylosig</a></code> for details.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a rerun with a given number of species
<code>n.remov</code> removed, representing <code>n.percent</code> of the full dataset.
Columns report the calculated signal estimate (<code>estimate</code>),
difference between reduced data signal estimate and full data signal (<code>DF</code>),
the percentage of change in signal compared to the full data estimate (<code>perc</code>)
and signal p-value for the reduced data estimate(<code>pval</code>).
</p>
<p><code>sign.analysis</code> For each break (i.e. each percentage of species
removed) this reports the percentage of statistically significant (at p&lt;0.05)
phylogenetic signal over all repetitions with reduced data sets.
</p>
<p><code>data</code>: Original full dataset used in the analysis.
#' @note Please be aware that dropping species may reduce power to detect 
significant signal and may partially be responsible for a potential 
effect of species removal on p-values. Please also consult standardised differences
in the (summary) output.
</p>


<h3>Note</h3>

<p>The argument &quot;se&quot; from <code><a href="phytools.html#topic+phylosig">phylosig</a></code> is not available in this function. Use the 
argument &quot;V&quot; instead with <code><a href="#topic+intra_physig">intra_physig</a></code> to indicate the name of the column containing the standard 
deviation or the standard error of the trait variable instead.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.  
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014).
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation
in angiosperms. Nature Communications, 5, 4087.
</p>
<p>Blomberg, S. P., T. Garland Jr., A. R. Ives (2003) 
Testing for phylogenetic signal in comparative data: 
Behavioral traits are more labile. Evolution, 57, 717-745.
</p>
<p>Pagel, M. (1999) Inferring the historical patterns of biological evolution. 
Nature, 401, 877-884.
</p>
<p>Kamilar, J. M., &amp; Cooper, N. (2013). Phylogenetic signal in primate behaviour,
ecology and life history. Philosophical Transactions of the Royal Society 
B: Biological Sciences, 368: 20120341.
</p>


<h3>See Also</h3>

<p><code><a href="phytools.html#topic+phylosig">phylosig</a></code>, 
<code><a href="#topic+samp_phylm">samp_phylm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(alien)
alien.data&lt;-alien$data
alien.phy&lt;-alien$phy
# Logtransform data
alien.data$logMass &lt;- log(alien.data$adultMass) 
# Run sensitivity analysis:
samp &lt;- samp_physig(trait.col = "logMass", data = alien.data, n.sim = 30,
phy = alien.phy[[1]])
summary(samp)
sensi_plot(samp)
sensi_plot(samp, graphs = 1)
sensi_plot(samp, graphs = 2)

## End(Not run)

</code></pre>

<hr>
<h2 id='sensi_plot'>Graphical sensitivity analysis for comparative methods</h2><span id='topic+sensi_plot'></span>

<h3>Description</h3>

<p>Generic function for plotting results from
any sensitivity analysis performed with 'sensiPhy'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sensi_plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot_+3A_x">x</code></td>
<td>
<p>any output from the sensiPhy package.</p>
</td></tr>
<tr><td><code id="sensi_plot_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>sensi_plot recognize and print different sets of graphs depending 
on the function that generated 'x'. See the links below for details about
the graphs generated for each sensiPhy function:
</p>

<p><strong>PGLS regressions (single uncertainty):</strong>
</p>
<ul>
<li><p>clade_phylm: <code><a href="#topic+sensi_plot.sensiClade">sensi_plot.sensiClade</a></code>
</p>
</li>
<li><p>influ_phylm: <code><a href="#topic+sensi_plot.sensiInflu">sensi_plot.sensiInflu</a></code>
</p>
</li>
<li><p>samp_phylm: <code><a href="#topic+sensi_plot.sensiSamp">sensi_plot.sensiSamp</a></code>
</p>
</li>
<li><p>intra_phylm: <code><a href="#topic+sensi_plot.sensiIntra">sensi_plot.sensiIntra</a></code>
</p>
</li>
<li><p>tree_phylm: <code><a href="#topic+sensi_plot.sensiTree">sensi_plot.sensiTree</a></code>
</p>
<p><strong>PGLS regressions (interacting uncertainties):</strong>
</p>
</li>
<li><p>tree_intra_phylm: <code><a href="#topic+sensi_plot.sensiTree_Intra">sensi_plot.sensiTree_Intra</a></code>
</p>
</li>
<li><p>intra_clade_phylm: <code><a href="#topic+sensi_plot.sensiIntra_Clade">sensi_plot.sensiIntra_Clade</a></code>
</p>
</li>
<li><p>intra_influ_phylm: <code><a href="#topic+sensi_plot.sensiIntra_Influ">sensi_plot.sensiIntra_Influ</a></code>
</p>
</li>
<li><p>intra_samp_phylm: <code><a href="#topic+sensi_plot.sensiIntra_Samp">sensi_plot.sensiIntra_Samp</a></code>
</p>
</li>
<li><p>tree_clade_phylm: <code><a href="#topic+sensi_plot.sensiTree_Clade">sensi_plot.sensiTree_Clade</a></code>
</p>
</li>
<li><p>tree_influ_phylm: <code><a href="#topic+sensi_plot.sensiTree_Influ">sensi_plot.sensiTree_Influ</a></code>
</p>
</li>
<li><p>tree_samp_phylm: <code><a href="#topic+sensi_plot.sensiTree_Samp">sensi_plot.sensiTree_Samp</a></code>
</p>
<p><strong>Phylogenetic signal:</strong>
</p>
</li>
<li><p>clade_physig: <code><a href="#topic+sensi_plot.clade.physig">sensi_plot.clade.physig</a></code>
</p>
</li>
<li><p>influ_physig: <code><a href="#topic+sensi_plot.influ.physig">sensi_plot.influ.physig</a></code>
</p>
</li>
<li><p>samp_physig:  <code><a href="#topic+sensi_plot.samp.physig">sensi_plot.samp.physig</a></code>
</p>
</li>
<li><p>tree_physig:  <code><a href="#topic+sensi_plot.tree.physig">sensi_plot.tree.physig</a></code>
</p>
</li>
<li><p>intra_physig:  <code><a href="#topic+sensi_plot.intra.physig">sensi_plot.intra.physig</a></code>
</p>
<p><strong>trait evolution (continuous &amp; discrete characters):</strong>
</p>
</li>
<li><p>clade_continuous &amp; _discrete <code><a href="#topic+sensi_plot.sensiClade.TraitEvol">sensi_plot.sensiClade.TraitEvol</a></code>
</p>
</li>
<li><p>influ_continuous &amp; _discrete <code><a href="#topic+sensi_plot.sensiInflu.TraitEvol">sensi_plot.sensiInflu.TraitEvol</a></code>
</p>
</li>
<li><p>tree_continuous &amp; _discrete <code><a href="#topic+sensi_plot.sensiTree.TraitEvol">sensi_plot.sensiTree.TraitEvol</a></code>
</p>
</li>
<li><p>samp_continuous &amp; _discrete <code><a href="#topic+sensi_plot.sensiSamp.TraitEvol">sensi_plot.sensiSamp.TraitEvol</a></code>
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>References</h3>

<p>The function 'multiplot', developed by Winston Chang, is used inside sensi_plot
to print multiple graphs in one frame. 
The source code is available here:
<a href="http://www.cookbook-r.com/Graphs/Multiple_graphs_on_one_page_(ggplot2)/">http://www.cookbook-r.com/Graphs/Multiple_graphs_on_one_page_(ggplot2)/</a>
</p>

<hr>
<h2 id='sensi_plot.clade.physig'>Graphical diagnostics for class 'clade.physig'</h2><span id='topic+sensi_plot.clade.physig'></span>

<h3>Description</h3>

<p><code>plot_clade_physig</code> Plot results from <code>clade_physig</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clade.physig'
sensi_plot(x, clade = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.clade.physig_+3A_x">x</code></td>
<td>
<p>output from <code>influ_physig</code></p>
</td></tr>
<tr><td><code id="sensi_plot.clade.physig_+3A_clade">clade</code></td>
<td>
<p>The name of the clade to be evaluated (see details)</p>
</td></tr>
<tr><td><code id="sensi_plot.clade.physig_+3A_...">...</code></td>
<td>
<p>further arguments to methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from clade_physig:
</p>
<p><strong>Graph 1:</strong> Distribution of the simulated signal estimates (Null distribution
for a given clade sample size).
The red dashed line represents the estimated signal for the reduced data 
(without the focal clade) and the black line represents the signal estimate
for the full data.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="phytools.html#topic+phylosig">phylosig</a></code>
</p>

<hr>
<h2 id='sensi_plot.influ.physig'>Graphical diagnostics for class 'influ.physig'</h2><span id='topic+sensi_plot.influ.physig'></span>

<h3>Description</h3>

<p><code>sensi_plot.influ_physig</code> Plot results from <code>influ_physig</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'influ.physig'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.influ.physig_+3A_x">x</code></td>
<td>
<p>output from <code>influ_physig</code></p>
</td></tr>
<tr><td><code id="sensi_plot.influ.physig_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1, 2)</p>
</td></tr>
<tr><td><code id="sensi_plot.influ.physig_+3A_...">...</code></td>
<td>
<p>further arguments to methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from influ_physig:
</p>
<p><strong>Graph 1:</strong> Distribution of estimated phylogenetic signal (K or lambda) for each 
simulation (leave-one-out deletion). Dashed red vertical line represents the original
estimate of phylogenetic signal with the full data (with all species). 
</p>
<p><strong>Graph 2:</strong> Distribution of P values for the phylogenetic signal (K or lambda) 
for each simulation (leave-one-out deletion). Red vertical line represents 
the alpha significance level = 0.05.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="phytools.html#topic+phylosig">phylosig</a></code>
</p>

<hr>
<h2 id='sensi_plot.intra.physig'>Graphical diagnostics for class 'intra.physig'</h2><span id='topic+sensi_plot.intra.physig'></span>

<h3>Description</h3>

<p><code>sensi_plot_intra.physig</code> Plot results from <code>intra_physig</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'intra.physig'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.intra.physig_+3A_x">x</code></td>
<td>
<p>output from <code>intra_physig</code></p>
</td></tr>
<tr><td><code id="sensi_plot.intra.physig_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed in the output (&quot;all&quot;, 1 or 2)</p>
</td></tr>
<tr><td><code id="sensi_plot.intra.physig_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from <code>intra_physig</code>
</p>
<p>Graphs 1: Distribution of estimated phylogenetic signal (lambda or K) for each simulation
Red vertical line represents the average signal among all estimates. 
</p>
<p>Graph 2: Distribution of p-values for the phylogenetic signal (K or lambda) 
for each simulation. Red vertical line represents the alpha significance level = 0.05.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone and Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+intra_phylm">intra_phylm</a></code>
<code><a href="#topic+intra_phylm">intra_phylm</a></code>
</p>

<hr>
<h2 id='sensi_plot.samp.physig'>Graphical diagnostics for class 'samp.physig'</h2><span id='topic+sensi_plot.samp.physig'></span>

<h3>Description</h3>

<p><code>plot_samp_phylm</code> Plot results from <code>samp_physig</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'samp.physig'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.samp.physig_+3A_x">x</code></td>
<td>
<p>output from <code>samp_physig</code></p>
</td></tr>
<tr><td><code id="sensi_plot.samp.physig_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1,2 and 3 )</p>
</td></tr>
<tr><td><code id="sensi_plot.samp.physig_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from samp_physig:
</p>
<p><strong>Graph 1:</strong> Estimated phylogenetic signal for each simulation across
percentages of species removed. Colours represent percentage
of change in comparison with the full data estimate (blue = lower than 5, orange =
between 5 and 10 and red = higher than 10).
The red horizontal line represents the original phylogenetic signal
estimated from the full model (with all species).
</p>
<p><strong>Graph 2:</strong> The proportion of estimated signal in each category
across the percentage of species removed.
</p>
<p><strong>Graph 3:</strong> The percentage of significant signal estimates across the
percentage of species removed.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+samp_phylm">samp_phylm</a></code>
<code><a href="#topic+samp_physig">samp_physig</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiClade'>Graphical diagnostics for class 'sensiClade'</h2><span id='topic+sensi_plot.sensiClade'></span>

<h3>Description</h3>

<p>Plot results from <code>clade_phylm</code> and <code>clade_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiClade'
sensi_plot(x, clade = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiClade_+3A_x">x</code></td>
<td>
<p>output from <code>clade_phylm</code> or <code>clade_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiClade_+3A_clade">clade</code></td>
<td>
<p>The name of the clade to be evaluated (see details)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiClade_+3A_...">...</code></td>
<td>
<p>further arguments to methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from clade_phylm or clade_phyglm:
</p>
<p><strong>Graph 1:</strong> The original scatterplot <code class="reqn">y = a + bx</code> (with the 
full dataset) and a comparison between the regression lines of the full dataset
and the rerun without the selected clade (set by <code>clade</code>). For further
details about this method, see <code><a href="#topic+clade_phylm">clade_phylm</a></code>.
</p>
<p>Species from the selected clade are represented in red (removed species), black
solid line represents the regression with the full model and red dashed line represents
the regression of the model without the species from the selected clade.
To check the available clades to plot, see <code>x$sensi.estimates$clade</code> 
in the object returned from <code>clade_phylm</code> or <code>clade_phyglm</code>. 
</p>
<p><strong>Graph 2:</strong> Distribution of the simulated slopes (Null distribution
for a given clade sample size).
The red dashed line represents the estimated slope for the reduced model 
(without the focal clade) and the black line represents the slope for the 
full model.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clade_phylm">clade_phylm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiClade.TraitEvol'>Graphical diagnostics for class 'sensiClade.TraitEvol'</h2><span id='topic+sensi_plot.sensiClade.TraitEvol'></span>

<h3>Description</h3>

<p>Plot results from <code>clade_discrete</code> and <code>clade_continuous</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiClade.TraitEvol'
sensi_plot(x, clade = NULL, graph = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiClade.TraitEvol_+3A_x">x</code></td>
<td>
<p>output from <code>clade_discrete</code> or <code>clade_continuous</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiClade.TraitEvol_+3A_clade">clade</code></td>
<td>
<p>The name of the clade to be evaluated (see details)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiClade.TraitEvol_+3A_graph">graph</code></td>
<td>
<p>The graph to be printed. Default <code>all</code>, or 
for <code>clade_continuous</code> set <code>sigsq</code> or <code>optpar</code> 
and for <code>clade_discrete</code> set<code>q12</code> or <code>q21</code>.</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiClade.TraitEvol_+3A_...">...</code></td>
<td>
<p>further arguments to methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from clade_discrete or clade_continuous:
</p>
<p><strong>Graph 1:</strong> Distribution of the simulated parameter estimates (Null distribution
for a given clade sample size).
The red dashed line represents the estimated signal for the reduced data 
(without the focal clade) and the black line represents the signal estimate
for the full data.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno &amp; Gijsbert Werner
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clade_continuous">clade_continuous</a></code> or <code><a href="#topic+clade_discrete">clade_discrete</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiInflu'>Graphical diagnostics for class 'sensiInflu'</h2><span id='topic+sensi_plot.sensiInflu'></span>

<h3>Description</h3>

<p><code>plot_influ_phylm</code> Plot results from <code>influ_phylm</code> and 
<code>influ_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiInflu'
sensi_plot(x, graphs = "all", param = "estimate", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiInflu_+3A_x">x</code></td>
<td>
<p>output from <code>influ_phylm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiInflu_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1,2,3 or 4)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiInflu_+3A_param">param</code></td>
<td>
<p>choose which parameter (&quot;intercept&quot; or &quot;estimate&quot; should be printed)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiInflu_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from influ_phylm or influ_phyglm:
</p>
<p><strong>Graph 1:</strong> Distribution of estimated slopes (estimates) or intercepts for each 
simulation (leave-one-out deletion). Red vertical line represents the original
slope or intercept from the full model (with all species). 
</p>
<p><strong>Graph 2:</strong> Original regression plot (<code class="reqn">trait~predictor</code>). Standardized 
difference in slope or intercept is represented by a continuous size scale. 
The names of the most influential species (sDF &gt; cutoff) are ploted in the
graph. 
</p>
<p><strong>Graph 3:</strong> Distribution of standardized difference in slope or intercept. Red 
colour indicates influential species (with a standardised difference above 
the value of <code>cutoff</code>).
</p>
<p><strong>Graph 4:</strong> Distribution of the percentage of change in slope or intercept.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiInflu.TraitEvol'>Graphical diagnostics for class 'sensiInflu.TraitEvol'</h2><span id='topic+sensi_plot.sensiInflu.TraitEvol'></span>

<h3>Description</h3>

<p><code>sensi_plot.sensiTree.TraitEvol</code> Plot results from <code>influ_discrete</code> and <code>influ_continuous</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiInflu.TraitEvol'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiInflu.TraitEvol_+3A_x">x</code></td>
<td>
<p>output from <code>influ_discrete</code> or <code>influ_continuous</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiInflu.TraitEvol_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed in the output (&quot;all&quot;, &quot;q12&quot;, &quot;q21&quot;, &quot;aic&quot; or&quot; &quot;optpar&quot;)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiInflu.TraitEvol_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gijsbert Werner
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+influ_discrete">influ_discrete</a></code>
<code><a href="#topic+influ_continuous">influ_continuous</a></code>
</p>
<p>The following graphs are printed. 
</p>
<p><strong>Graph aicc:</strong> Distribution of estimated AICc-values across all single-species deletions. 
Red vertical line represents the mean signal among all estimates. 
Blue vertical line represents the median signal among all estimates. 
</p>
<p><strong>Graph optpar:</strong> Distribution of estimated values for optimisation parameter specified using 'transform' (if applicable)
Red vertical line represents the mean signal among all estimates. 
Blue vertical line represents the median signal among all estimates. 
</p>
<p>Additionally, only for <code>tree_discrete</code> the function creates the following graphs. 
</p>
<p><strong>Graph q12:</strong> Distribution of estimated parameter values for transition rates q12 across all single-species deletions. 
Red vertical line represents the mean signal among all estimates. 
Blue vertical line represents the median signal among all estimates. 
</p>
<p><strong>Graph q21:</strong> Distribution of estimated parameter values for transition rates q21. 
Red vertical line represents the mean signal among all estimates. 
Blue vertical line represents the median signal among all estimates. 
</p>
<p>While only for <code>tree_continuous</code> the function creates the following graphs. 
</p>
<p><strong>Graph sigsq:</strong> Distribution of estimated parameter values for rate of evolution sigsq across all single-species deletions. .
Red vertical line represents the mean signal among all estimates. 
Blue vertical line represents the median signal among all estimates. 
</p>
<p><strong>Graph z0:</strong> Distribution of estimated parameter values for z0.
Red vertical line represents the mean signal among all estimates. 
Blue vertical line represents the median signal among all estimates.
</p>

<hr>
<h2 id='sensi_plot.sensiIntra'>Graphical diagnostics for class 'sensiIntra'</h2><span id='topic+sensi_plot.sensiIntra'></span>

<h3>Description</h3>

<p><code>plot_tree.intra_phylm</code> Plot results from <code>tree_phylm</code>,
<code>intra_phylm</code> and <code>intra_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiIntra'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiIntra_+3A_x">x</code></td>
<td>
<p>output from <code>tree_phylm</code>, <code>tree_phyglm</code>,
<code>intra_phylm</code> or <code>intra_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed in the output (&quot;all&quot;, 1, 2 or 3)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from <code>tree_phylm</code>, <code>tree_phyglm</code>,
<code>intra_phylm</code> or <code>intra_phyglm</code>:
</p>
<p><strong>Graphs 1 and 2:</strong> Distribution of estimated slopes (estimates) and intercepts for each tree (for <code>tree_phylm</code>) or 
value generated within a given interval (<code>intra_phylm</code>)
Red vertical line represents the mean estimate or intercept for all models. 
</p>
<p><strong>Graph 3:</strong> Scatterplot with mean regression (black line) and standard deviation of the regression (red dotted lines).
</p>


<h3>Author(s)</h3>

<p>Caterina Penone and Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>
<code><a href="#topic+intra_phylm">intra_phylm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiIntra_Clade'>Graphical diagnostics for class 'sensiIntra_Clade'</h2><span id='topic+sensi_plot.sensiIntra_Clade'></span>

<h3>Description</h3>

<p>Plot results from <code>intra_clade_phylm</code> and <code>intra_clade_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiIntra_Clade'
sensi_plot(x, clade = NULL, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiIntra_Clade_+3A_x">x</code></td>
<td>
<p>output from <code>intra_clade_phylm</code> or <code>intra_clade_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_Clade_+3A_clade">clade</code></td>
<td>
<p>The name of the clade to be evaluated (see details)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_Clade_+3A_graphs">graphs</code></td>
<td>
<p>Choose which graph should be printed on the output (&quot;all&quot;, 1 or 2). Defaults to &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_Clade_+3A_...">...</code></td>
<td>
<p>further arguments to methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from intra_clade_phylm or intra_clade_phyglm:
</p>
<p><strong>Graph 1:</strong> Estimated slopes after clade removal (reduced data) across multiple simulations.
Small dots represent estimates reruns between simulations while larger 
dots represents the average estimate between all simulations for each clade. 
The solid black line represents the average slope estimate among trees
using the full dataset.
</p>
<p><strong>Graph 2:</strong> The effect of clade removal on slope estimate across all individual 
simulations for each clade analyzed. The black line indicates estimates 
with the full dataset while the red line represent estimates without the focal
clade (reduced data) across different simulation The blue dots represent null expectation
estimates after removing the same number of species of the focal clade,
with dots falling outside the red line area indicating a larger than expected 
absolute effect.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone
</p>


<h3>See Also</h3>

<p><code><a href="#topic+intra_clade_phylm">intra_clade_phylm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiIntra_Influ'>Graphical diagnostics for class 'sensiIntra_Influ'</h2><span id='topic+sensi_plot.sensiIntra_Influ'></span>

<h3>Description</h3>

<p><code>sensi_plot.intra_influ</code> Plot results from <code>intra_influ_phylm</code> and 
<code>intra_influ_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiIntra_Influ'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiIntra_Influ_+3A_x">x</code></td>
<td>
<p>output from <code>influ_phylm</code> or <code>intra_influ_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_Influ_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1,2,3 or 4)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_Influ_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from intra_influ_phylm or intra_influ_phyglm:
</p>
<p><strong>Graph 1:</strong> Most common influential species on model estimates.
Percentage of iterations (n.tree or n.intra) where the removal of individual
species caused significant change in model estimate (sDIFestimate &gt; cutoff). 
</p>
<p><strong>Graph 2:</strong> Shift on model estimate after species removal for most influential
species. Small red dots represent individual reruns between different trees or 
simulations while large red dots represent the average DIFestimate among all 
iterations.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone
</p>


<h3>See Also</h3>

<p><code><a href="#topic+intra_influ_phylm">intra_influ_phylm</a></code>; 
<code><a href="#topic+intra_influ_phyglm">intra_influ_phyglm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiIntra_Samp'>Graphical diagnostics for class 'sensiIntra_Samp' and 'sensiTree_samp'</h2><span id='topic+sensi_plot.sensiIntra_Samp'></span>

<h3>Description</h3>

<p><code>sensi_plot.sensiIntra_Samp</code> Plot results from <code>sensiIntra_Samp_phylm</code> and
<code>sensiIntra_Samp_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiIntra_Samp'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiIntra_Samp_+3A_x">x</code></td>
<td>
<p>output from <code>sensiIntra_Samp_phylm</code> or <code>sensiIntra_Samp_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_Samp_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1,2,3 or 4)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiIntra_Samp_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from sensiIntra_Samp_phylm or sensiIntra_Samp_phyglm:
</p>
<p><strong>Graph 1:</strong> : Estimated estimates for each simulation across percentages of 
species removed. Small red dots represent individual estimates for each iteration 
(tree or intra) across percentages of removed species. Large dots represent the
average among these estimates within each iteration and percentage of species removal.
Different estimates within the same percentage interval represent different 
iterations (tree or intra). 
</p>
<p><strong>Graph 2:</strong> : The percentage of significant estimates (p &lt; 0.05) across the 
percentage of species removed. Small red dots represent individual 
iterations (tree or intra) while large dots represent the average among these iterations.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+samp_phylm">samp_phylm</a></code>
<code><a href="#topic+samp_phyglm">samp_phyglm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiSamp'>Graphical diagnostics for class 'sensiSamp'</h2><span id='topic+sensi_plot.sensiSamp'></span>

<h3>Description</h3>

<p><code>plot_samp_phylm</code> Plot results from <code>samp_phylm</code> and
<code>influ_phyloglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiSamp'
sensi_plot(x, graphs = "all", param = "estimate", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiSamp_+3A_x">x</code></td>
<td>
<p>output from <code>samp_phylm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiSamp_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1,2,3 or 4)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiSamp_+3A_param">param</code></td>
<td>
<p>choose which model parameter should be ploted  (&quot;intercept&quot; or &quot;estimate&quot;)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiSamp_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from samp_phylm or samp_phyglm:
</p>
<p><strong>Graph 1:</strong> Estimated slopes or intercepts for each simulation across  
percentages of species removed. Colours represent percentage 
of change in comparison with the full model (blue = lower than 5, orange = 
between 5 and 10 and red = higher than 10).
The red horizontal line represents the original slope or 
intercept from the full model (with all species). 
</p>
<p><strong>Graph 2:</strong> The proportion of estimated slopes and intercepts in each category 
across the percentage of species removed.
</p>
<p><strong>Graph 3:</strong> Estimated phylogenetic model parameter for each simulation across
the percentage of species removed.
</p>
<p><strong>Graph 4:</strong> The percentage of significant slopes or intercepts across the 
percentage of species removed.
</p>


<h3>Note</h3>

<p>If model = &quot;BM&quot;, only plots 1, 2 and 4 are printed. Plot 3, phylogenetic
model parameter is not available for model = &quot;BM&quot;
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+samp_phylm">samp_phylm</a></code>
<code><a href="#topic+samp_phyglm">samp_phyglm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiSamp.TraitEvol'>Graphical diagnostics for class 'sensiSamp.TraitEvol'</h2><span id='topic+sensi_plot.sensiSamp.TraitEvol'></span>

<h3>Description</h3>

<p><code>sensi_plot.sensiSamp.TraitEvol</code> Plot results from <code>samp_continuous</code> and
<code>samp_discrete</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiSamp.TraitEvol'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiSamp.TraitEvol_+3A_x">x</code></td>
<td>
<p>output from <code>samp_continuous</code> or <code>samp_continuous</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiSamp.TraitEvol_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1,2,3 or 4)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiSamp.TraitEvol_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from samp_continuous or samp_discrete:
</p>
<p><strong>Graph 1:</strong> Estimated q12 (discrete) or sigsq (discrete) for each simulation across  
percentages of species removed. Colours represent percentage 
of change in comparison with the full model (blue = lower than 5, orange = 
between 5 and 10 and red = higher than 10).The red horizontal line represents 
the original value from the full model (with all species). 
</p>
<p><strong>Graph 2:</strong> The proportion of estimated  q12 (discrete) or sigsq (discrete) in each category 
across the percentage of species removed.
</p>
<p><strong>Graph 3:</strong> Estimated q21 for each simulation across the percentage of species removed  
(only for <code>samp_discrete</code>).
</p>
<p><strong>Graph 4:</strong> The percentage of significant q21 across the 
percentage of species removed (only for <code>samp_discrete</code>).
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno &amp; Gijsbert Werner
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+samp_continuous">samp_continuous</a></code>
<code><a href="#topic+samp_discrete">samp_discrete</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiTree'>Graphical diagnostics for class 'sensiTree'</h2><span id='topic+sensi_plot.sensiTree'></span>

<h3>Description</h3>

<p><code>plot_tree.intra_phylm</code> Plot results from <code>tree_phylm</code>,
<code>intra_phylm</code> and <code>intra_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiTree'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiTree_+3A_x">x</code></td>
<td>
<p>output from <code>tree_phylm</code>, <code>tree_phyglm</code>,
<code>intra_phylm</code> or <code>intra_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed in the output (&quot;all&quot;, 1, 2 or 3)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from <code>tree_phylm</code>, <code>tree_phyglm</code>,
<code>intra_phylm</code> or <code>intra_phyglm</code>:
</p>
<p><strong>Graphs 1 and 2:</strong> Distribution of estimated estimates and intercepts for each tree (for <code>tree_phylm</code>) or 
value generated within a given interval (<code>intra_phylm</code>)
Red vertical line represents the mean estimate or intercept for all models. 
</p>
<p><strong>Graph 3:</strong> Scatterplot with mean regression (black line) and standard deviation of the regression (blue dotted lines).
</p>


<h3>Author(s)</h3>

<p>Caterina Penone and Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>
<code><a href="#topic+intra_phylm">intra_phylm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiTree_Clade'>Graphical diagnostics for class 'sensiTree_Clade'</h2><span id='topic+sensi_plot.sensiTree_Clade'></span>

<h3>Description</h3>

<p>Plot results from <code>tree_clade_phylm</code> and <code>tree_clade_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiTree_Clade'
sensi_plot(x, clade = NULL, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiTree_Clade_+3A_x">x</code></td>
<td>
<p>output from <code>tree_clade_phylm</code> or <code>tree_clade_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Clade_+3A_clade">clade</code></td>
<td>
<p>The name of the clade to be evaluated (see details)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Clade_+3A_graphs">graphs</code></td>
<td>
<p>Choose which graph should be printed on the output (&quot;all&quot;, 1 or 2). Defaults to &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Clade_+3A_...">...</code></td>
<td>
<p>further arguments to methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from tree_clade_phylm or tree_clade_phyglm:
</p>
<p><strong>Graph 1:</strong> Estimated slopes after clade removal (reduced data) across multiple trees.
Small dots represent estimates reruns between phylogenetic trees while larger 
dots represents the average estimate between all trees for each clade. 
The solid black line represents the average slope estimate among trees
using the full dataset.
</p>
<p><strong>Graph 2:</strong> The effect of clade removal on slope estimate across all individual 
phylogenetic trees for each clade analyzed. The black line indicates estimates 
with the full dataset while the red line represent estimates without the focal
clade (reduced data) across different trees. The blue dots represent null expectation
estimates after removing the same number of species of the focal clade,
with dots falling outside the red line area indicating a larger than expected 
absolute effect.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tree_clade_phylm">tree_clade_phylm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiTree_Influ'>Graphical diagnostics for class 'sensiTree_Influ'</h2><span id='topic+sensi_plot.sensiTree_Influ'></span>

<h3>Description</h3>

<p><code>sensi_plot.sensiTree_Influ</code> Plot results from <code>tree_influ_phylm</code> and 
<code>tree_influ_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiTree_Influ'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiTree_Influ_+3A_x">x</code></td>
<td>
<p>output from <code>tree_influ_phylm</code> or <code>tree_influ_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Influ_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1,2,3 or 4)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Influ_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from sensiTree_Influ_phylm or sensiTree_Influ_phyglm:
</p>
<p><strong>Graph 1:</strong> Most common influential species on model estimates.
Percentage of iterations (n.tree or n.intra) where the removal of individual
species caused significant change in model estimate (sDIFestimate &gt; cutoff). 
</p>
<p><strong>Graph 2:</strong> Shift on model estimate after species removal for most influential
species. Small red dots represent individual reruns between different trees or 
simulations while large red dots represent the average DIFestimate among all 
iterations.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiTree_Intra'>Graphical diagnostics for class 'sensiIntra_Tree'</h2><span id='topic+sensi_plot.sensiTree_Intra'></span>

<h3>Description</h3>

<p><code>plot_tree.intra_phylm</code> Plot results from <code>tree_intra_phylm</code>,
<code>tree_intra_phylm</code> and <code>tree_intra_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiTree_Intra'
sensi_plot(x, graphs = "all", uncer.type = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiTree_Intra_+3A_x">x</code></td>
<td>
<p>output from <code>tree_intra_phylm</code>, <code>tree_intra_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Intra_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed in the output (&quot;all&quot;, 1, 2 or 3)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Intra_+3A_uncer.type">uncer.type</code></td>
<td>
<p>choose which uncertainty type should be printed (&quot;all&quot;, &quot;intra&quot;, &quot;tree&quot;)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Intra_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from <code>tree_intra_phylm</code> or <code>tree_intra_phyglm</code>:
</p>
<p><strong>Graphs 1 and 2:</strong> Distribution of estimated estimates and intercepts for each tree (for <code>tree_phylm</code>) or 
value generated within a given interval (<code>interaction_intra_tree_phylm</code>)
Red vertical line represents the mean estimate or intercept for all models. 
</p>
<p><strong>Graph 3:</strong> Scatterplot with mean regression (black line) and standard deviation of the regression (blue dotted lines).
</p>


<h3>Author(s)</h3>

<p>Caterina Penone and Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>
<code><a href="#topic+intra_phylm">intra_phylm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiTree_Samp'>Graphical diagnostics for class 'sensiTree_samp'</h2><span id='topic+sensi_plot.sensiTree_Samp'></span>

<h3>Description</h3>

<p><code>sensi_plot.sensiTree_Samp</code> Plot results from <code>sensiTree_Samp_phylm</code> and
<code>sensiTree_Samp_phyglm</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiTree_Samp'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiTree_Samp_+3A_x">x</code></td>
<td>
<p>output from <code>sensiTree_Samp_phylm</code> or <code>sensiTree_Samp_phyglm</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Samp_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed on the output (&quot;all&quot;, 1,2,3 or 4)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree_Samp_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from sensiTree_Samp_phylm or sensiTree_Samp_phyglm:
</p>
<p><strong>Graph 1:</strong> : Estimated estimates for each simulation across percentages of 
species removed. Small red dots represent individual estimates for each iteration 
(tree or intra) across percentages of removed species. Large dots represent the
average among these estimates within each iteration and percentage of species removal.
Different estimates within the same percentage interval represent different 
iterations (tree or intra). 
</p>
<p><strong>Graph 2:</strong> : The percentage of significant estimates (p &lt; 0.05) across the 
percentage of species removed. Small red dots represent individual 
iterations (tree or intra) while large dots represent the average among these iterations.
</p>


<h3>Note</h3>

<p>If model = &quot;BM&quot;, only plots 1, 2 and 4 are printed. Plot 3, phylogenetic
model parameter is not available for model = &quot;BM&quot;
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+samp_phylm">samp_phylm</a></code>
<code><a href="#topic+samp_phyglm">samp_phyglm</a></code>
</p>

<hr>
<h2 id='sensi_plot.sensiTree.TraitEvol'>Graphical diagnostics for class 'sensiTree.TraitEvol'</h2><span id='topic+sensi_plot.sensiTree.TraitEvol'></span>

<h3>Description</h3>

<p><code>sensi_plot.sensiTree.TraitEvol</code> Plot results from <code>tree_discrete</code> and <code>tree_continuous</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensiTree.TraitEvol'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.sensiTree.TraitEvol_+3A_x">x</code></td>
<td>
<p>output from <code>tree_fitDiscrete</code> or <code>tree_fitContinuous</code></p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree.TraitEvol_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed in the output (&quot;all&quot;, &quot;q12&quot;, &quot;q21&quot;, &quot;aic&quot; or&quot; &quot;optpar&quot;)</p>
</td></tr>
<tr><td><code id="sensi_plot.sensiTree.TraitEvol_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gijsbert Werner
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+tree_discrete">tree_discrete</a></code>
<code><a href="#topic+tree_continuous">tree_continuous</a></code>
</p>
<p>The following graphs are printed. 
</p>
<p><strong>Graph aicc:</strong> Distribution of estimated AICc-values across each tree.
Red vertical line represents the mean value among all estimates. 
Blue vertical line represents the median value among all estimates. 
</p>
<p><strong>Graph optpar:</strong> Distribution of estimated values for optimisation parameter specified using 'transform' (if applicable)
Red vertical line represents the mean value among all estimates. 
Blue vertical line represents the median value among all estimates. 
</p>
<p>Additionally, only for <code>tree_discrete</code> the function creates the following graphs. 
</p>
<p><strong>Graph q12:</strong> Distribution of estimated parameter values for transition rates q12 for each tree.
Red vertical line represents the mean value among all estimates. 
Blue vertical line represents the median value among all estimates. 
</p>
<p><strong>Graph q21:</strong> Distribution of estimated parameter values for transition rates q21 for each tree.
Red vertical line represents the mean value among all estimates. 
Blue vertical line represents the median value among all estimates. 
</p>
<p>While only for <code>tree_continuous</code> the function creates the following graphs. 
</p>
<p><strong>Graph sigsq:</strong> Distribution of estimated parameter values for rate of evolution sigsq for each tree.
Red vertical line represents the mean value among all estimates. 
Blue vertical line represents the median value among all estimates. 
</p>
<p><strong>Graph z0:</strong> Distribution of estimated parameter values for z0 for each tree.
Red vertical line represents the mean value among all estimates. 
Blue vertical line represents the median value among all estimates.
</p>

<hr>
<h2 id='sensi_plot.tree.bd'>Graphical diagnostics for class 'tree.physig'</h2><span id='topic+sensi_plot.tree.bd'></span>

<h3>Description</h3>

<p><code>sensi_plot_tree.bd</code> Plot results from <code>tree_bd</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tree.bd'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.tree.bd_+3A_x">x</code></td>
<td>
<p>output from <code>tree_bd</code></p>
</td></tr>
<tr><td><code id="sensi_plot.tree.bd_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed in the output (&quot;all&quot;, 1 or 2)</p>
</td></tr>
<tr><td><code id="sensi_plot.tree.bd_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from <code>tree_bd</code>
</p>
<p><strong>Graphs 1:</strong> Distribution of estimated diversification or speciation rate
for each tree. Red vertical line represents the average signal among all estimates. 
</p>
<p><strong>Graphs 2:</strong> Estimates across each phylogenetic tree.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone and Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>
<code><a href="#topic+intra_phylm">intra_phylm</a></code>
</p>

<hr>
<h2 id='sensi_plot.tree.physig'>Graphical diagnostics for class 'tree.physig'</h2><span id='topic+sensi_plot.tree.physig'></span>

<h3>Description</h3>

<p><code>sensi_plot_tree.physig</code> Plot results from <code>tree_physig</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tree.physig'
sensi_plot(x, graphs = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensi_plot.tree.physig_+3A_x">x</code></td>
<td>
<p>output from <code>tree_physig</code></p>
</td></tr>
<tr><td><code id="sensi_plot.tree.physig_+3A_graphs">graphs</code></td>
<td>
<p>choose which graph should be printed in the output (&quot;all&quot;, 1 or 2)</p>
</td></tr>
<tr><td><code id="sensi_plot.tree.physig_+3A_...">...</code></td>
<td>
<p>further arguments to methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For 'x' from <code>tree_physig</code>
</p>
<p><strong>Graphs 1:</strong> Distribution of estimated phylogenetic signal (lambda or K) for each tree
Red vertical line represents the average signal among all estimates. 
</p>
<p><strong>Graph 2:</strong> Distribution of p-values for the phylogenetic signal (K or lambda) 
for each tree. Red vertical line represents the alpha significance level = 0.05.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone and Gustavo Paterno
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>
<code><a href="#topic+intra_phylm">intra_phylm</a></code>
</p>

<hr>
<h2 id='sensiPhy'>Sensitivity Analysis for Comparative Methods</h2><span id='topic+sensiPhy'></span>

<h3>Description</h3>

<p>An implementation of sensitivity analysis for phylogenetic comparative
methods. The package is an umbrella of statistical and graphical methods that 
estimate and report different types of uncertainty in PCM:  
(i) Species Sampling uncertainty (sample size; influential species and clades).  
(ii) Phylogenetic uncertainty (different topologies and/or branch lengths).  
(iii) Data uncertainty (intraspecific variation and measurement error).
</p>


<h3>Bug reports</h3>

<p><a href="https://github.com/paternogbc/sensiPhy/issues">https://github.com/paternogbc/sensiPhy/issues</a>.
</p>


<h3>How to contribute?</h3>

<p>You can find instructions on how to contribute to <b>sensiPhy</b> at
this link: <a href="https://github.com/paternogbc/sensiPhy/wiki/How-to-support-sensiPhy">https://github.com/paternogbc/sensiPhy/wiki/How-to-support-sensiPhy</a>
</p>
<p>A quick tutorial is available at this link: 
<a href="https://github.com/paternogbc/sensiPhy/wiki">https://github.com/paternogbc/sensiPhy/wiki</a>
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno; Caterina Penone &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467.
</p>

<hr>
<h2 id='tree_bd'>Phylogenetic uncertainty - net diversification rate</h2><span id='topic+tree_bd'></span>

<h3>Description</h3>

<p>Performs estimates of diversification rate evaluating
uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_bd(phy, n.tree = "all", method = "ms", track = F, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_bd_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'multiPhylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_bd_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file. (If <code>n.tree</code> = &quot;all&quot;, diversification will be estimated
among the set of trees provided in <code>phy</code>)</p>
</td></tr>
<tr><td><code id="tree_bd_+3A_method">method</code></td>
<td>
<p>the method for estimating diversification rate (&quot;ms&quot; or &quot;km&quot;) (see Details).</p>
</td></tr>
<tr><td><code id="tree_bd_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_bd_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylosig</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function estimates net diversification rate using <code><a href="geiger.html#topic+bd.ms">bd.ms</a></code> 
(Magallon and Sanderson (2000) method) or speciation rate using <code><a href="geiger.html#topic+bd.km">bd.km</a></code>
(Kendall-Moran method) for n trees, randomly picked from a multiPhylo file.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_bd</code> returns a list with the following
components:
</p>
<p><code>tree.bd.estimates</code>: Three number, diversification/speciation rate estimate 
(&quot;Magallon and Sanderson&quot; or &quot;Kendall-Moran&quot;) for each run with a different phylogenetic tree.
</p>
<p><code>stats</code>: Main statistics for estimates across trees.<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Magallon S and MJ Sanderson. 2000. Absolute diversification rates in 
angiosperm clades. Evolution 55:1762-1780.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+bd.ms">bd.ms</a></code>,
<code><a href="#topic+tree_phylm">tree_phylm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("primates")
# To estimate diversification rate with Magallon and Sanderson method:
fit &lt;- tree_bd(phy = primates.phy, n.tree = 30, method = "ms")
summary(fit)
sensi_plot(fit)
# To estimate speciation rate Kendall-Moran method
fit &lt;- tree_bd(phy = primates.phy, n.tree = 30, method = "km")
summary(fit)
sensi_plot(fit)
</code></pre>

<hr>
<h2 id='tree_clade_phyglm'>Interaction between phylogenetic uncertainty and influential clade detection - Phylogenetic Logistic Regression</h2><span id='topic+tree_clade_phyglm'></span>

<h3>Description</h3>

<p>Estimate the impact on model estimates of phylogenetic logistic regression after 
removing clades from the analysis and evaluating uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_clade_phyglm(
  formula,
  data,
  phy,
  clade.col,
  n.species = 5,
  n.sim = 100,
  n.tree = 2,
  btol = 50,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_clade_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'multiPhylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.
If NULL, <code>n.tree</code> = 2</p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_clade_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently only logistic regression using the &quot;logistic_MPLE&quot;-method from
<code>phyloglm</code> is implemented.
</p>
<p>This function sequentially removes one clade at a time, fits a phylogenetic
logistic regression model using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code> and stores the
results. The impact of of a specific clade on model estimates is calculated by the
comparison between the full model (with all species) and the model without 
the species belonging to a clade. It repeats this operation using n trees, 
randomly picked in a multiPhylo file.
</p>
<p>Additionally, to account for the influence of the number of species on each 
clade (clade sample size), this function also estimates a null distribution of slopes
expected for the number of species in a given clade. This is done by fitting
models without the same number of species in the given clade. 
The number of simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p><code>clade_phyglm</code> detects influential clades based on
difference in intercept and/or slope when removing a given clade compared
to the full model including all species. This is done for n trees in the multiphylo file.
</p>
<p>Currently, this function can only implements simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>clade_phyglm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade for a tree iteration. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the percentage of change
in intercept compared to the full model (<code>intercept.perc</code>) and intercept
p-value (<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter 
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) 
are reported.
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distributions
for all clades analysed.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Clades and/or trees where deletion resulted in errors.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+tree_phyglm">tree_phyglm</a></code>,
<code><a href="#topic+clade_phyglm">clade_phyglm</a></code>, <code><a href="#topic+tree_clade_phylm">tree_clade_phylm</a></code>,
<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Simulate Data:
set.seed(6987)
mphy = rmtree(150, N = 30)
x = rTrait(n=1,phy=mphy[[1]])
X = cbind(rep(1,150),x)
y = rbinTrait(n=1,phy=mphy[[1]], beta=c(-1,0.5), alpha=.7 ,X=X)
cla &lt;- rep(c("A","B","C","D","E"), each = 30)
dat = data.frame(y, x, cla)
# Run sensitivity analysis:
tree_clade &lt;- tree_clade_phyglm(y ~ x, phy = mphy, data = dat, 
n.tree = 10, n.sim = 10, clade.col = "cla")
# To check summary results and most influential clades:
summary(tree_clade)
# Visual diagnostics for clade removal:
sensi_plot(tree_clade)
# Specify which clade removal to plot:
sensi_plot(tree_clade, "B")
sensi_plot(tree_clade, "C", graphs = 2)
sensi_plot(tree_clade, "D", graphs = 2) 

## End(Not run)

</code></pre>

<hr>
<h2 id='tree_clade_phylm'>Interaction between phylogenetic uncertainty and influential clade detection - Phylogenetic Linear Regression</h2><span id='topic+tree_clade_phylm'></span>

<h3>Description</h3>

<p>Estimate the impact on model estimates of phylogenetic linear regression after 
removing clades from the analysis and evaluating uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_clade_phylm(
  formula,
  data,
  phy,
  clade.col,
  n.species = 5,
  n.sim = 100,
  n.tree = 2,
  model = "lambda",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_clade_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'multiPhylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_clade.col">clade.col</code></td>
<td>
<p>The column in the provided data frame which specifies the
clades (a character vector with clade names).</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_n.species">n.species</code></td>
<td>
<p>Minimum number of species in a clade for the clade to be
included in the leave-one-out deletion analysis. Default is <code>5</code>.</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations for the randomization test.</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.
If NULL, <code>n.tree</code> = 2</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_clade_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one clade at a time, fits a phylogenetic
linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code> and stores the
results. The impact of of a specific clade on model estimates is calculated by the
comparison between the full model (with all species) and the model without 
the species belonging to a clade. It repeats this operation using n trees, 
randomly picked in a multiPhylo file.
</p>
<p>Additionally, to account for the influence of the number of species on each 
clade (clade sample size), this function also estimate a null distribution of slopes
expected for the number of species in a given clade. This is done by fitting
models without the same number of species in the given clade. 
The number of simulations to be performed is set by 'n.sim'. To test if the 
clade influence differs from the null expectation for a clade of that size, 
a randomization test can be performed using 'summary(x)'. 
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p><code>clade_phylm</code> detects influential clades based on
difference in intercept and/or slope when removing a given clade compared
to the full model including all species. This is done for n trees in the multiphylo file.
</p>
<p>Currently, this function can only implement simple linear models (i.e. 
<code class="reqn">y = a + bx</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>clade_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade for a tree iteration. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the percentage of change
in intercept compared to the full model (<code>intercept.perc</code>) and intercept
p-value (<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter 
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) 
are reported.
</p>
<p><code>null.dist</code>: A data frame with estimates for the null distributions
for all clades analysed.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Clades and/or trees where deletion resulted in errors.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>,
<code><a href="#topic+clade_phylm">clade_phylm</a></code>, <code><a href="#topic+tree_clade_phyglm">tree_clade_phyglm</a></code>,
<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load data:
data(primates)
# run analysis:
clade_tree &lt;- tree_clade_phylm(log(sexMaturity) ~ log(adultMass), 
phy = primates$phy, data = primates$data, clade.col = "family", n.sim = 50, n.tree = 5)
# To check summary results and most influential clades:
summary(clade_tree)
# Visual diagnostics for clade removal:
sensi_plot(clade_tree)
# Specify which clade removal to plot:
sensi_plot(clade_tree)
sensi_plot(clade_tree, "Cercopithecidae")
sensi_plot(clade_tree, clade = "Cebidae", graphs = 2)

## End(Not run)

</code></pre>

<hr>
<h2 id='tree_continuous'>Phylogenetic uncertainty - Trait Evolution Continuous Characters</h2><span id='topic+tree_continuous'></span>

<h3>Description</h3>

<p>Fits models for trait evolution of continuous characters, 
evaluating phylogenetic uncertainty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_continuous(
  data,
  phy,
  n.tree = 10,
  model,
  bounds = list(),
  n.cores = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_continuous_+3A_data">data</code></td>
<td>
<p>Data vector for a single continuous trait, with names matching tips in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_continuous_+3A_phy">phy</code></td>
<td>
<p>Phylogenies (class 'multiPhylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_continuous_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file. If NULL, <code>n.tree</code> = 10</p>
</td></tr>
<tr><td><code id="tree_continuous_+3A_model">model</code></td>
<td>
<p>The evolutionary model (see Details).</p>
</td></tr>
<tr><td><code id="tree_continuous_+3A_bounds">bounds</code></td>
<td>
<p>settings to constrain parameter estimates. See <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code></p>
</td></tr>
<tr><td><code id="tree_continuous_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use. If 'NULL', number of cores is detected.</p>
</td></tr>
<tr><td><code id="tree_continuous_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_continuous_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits different models of continuous character evolution using <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code>
to n trees, randomly picked in a multiPhylo file. 
</p>
<p>Different evolutionary models from <code>fitContinuous</code> can be used, i.e. <code>BM</code>,<code>OU</code>,
<code>EB</code>, <code>trend</code>, <code>lambda</code>, <code>kappa</code>, <code>delta</code> and <code>drift</code>.
</p>
<p>See <code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code> for more details on character models and tree transformations. 
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_continuous</code> returns a list with the following
components:
</p>
<p><code>call</code>: The function call
</p>
<p><code>data</code>: The original full data vector
</p>
<p><code>sensi.estimates</code>:  (rate of evolution <code>sigsq</code>, 
root state <code>z0</code> and where applicable <code>optpar</code>),
AICc and the optimised value of the phylogenetic transformation parameter (e.g. <code>lambda</code>) 
for each analysis with a different phylogenetic tree.
</p>
<p><code>N.tree</code>: Number of trees <code>n.tree</code> analysed
</p>
<p><code>stats</code>: Main statistics for model parameters, i.e. minimum, maximum, mean, median and sd-values
</p>
<p><code>optpar</code>: Evolutionary model used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>


<h3>Author(s)</h3>

<p>Gijsbert Werner &amp; Caterina Penone
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Yang Z. 2006. Computational Molecular Evolution. Oxford University Press: Oxford. 
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008.
GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+fitContinuous">fitContinuous</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
#Load data:
data("primates")
#Model trait evolution accounting for phylogenetic uncertainty
adultMass&lt;-primates$data$adultMass
names(adultMass)&lt;-rownames(primates$data)
tree_cont&lt;-tree_continuous(data = adultMass,phy = primates$phy,
model = "OU",n.tree=30,n.cores = 2,track = TRUE)
#Print summary statistics
summary(tree_cont)
sensi_plot(tree_cont)
sensi_plot(tree_cont,graphs="sigsq")
sensi_plot(tree_cont,graphs="optpar")
#Use a different evolutionary model 
tree_cont2&lt;-tree_continuous(data = adultMass,phy = primates$phy,
model = "delta",n.tree=30,n.cores = 2,track = TRUE)
summary(tree_cont2)
sensi_plot(tree_cont2)

## End(Not run)
</code></pre>

<hr>
<h2 id='tree_discrete'>Phylogenetic uncertainty - Trait Evolution Discrete Characters</h2><span id='topic+tree_discrete'></span>

<h3>Description</h3>

<p>Fits models for trait evolution of discrete (binary) characters, 
evaluating phylogenetic uncertainty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_discrete(
  data,
  phy,
  n.tree = 10,
  model,
  transform = "none",
  bounds = list(),
  n.cores = NULL,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_discrete_+3A_data">data</code></td>
<td>
<p>Data vector for a single binary trait, with names matching tips in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_discrete_+3A_phy">phy</code></td>
<td>
<p>Phylogenies (class 'multiPhylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_discrete_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file. If NULL, <code>n.tree</code> = 10</p>
</td></tr>
<tr><td><code id="tree_discrete_+3A_model">model</code></td>
<td>
<p>The Mkn model to use (see Details).</p>
</td></tr>
<tr><td><code id="tree_discrete_+3A_transform">transform</code></td>
<td>
<p>The evolutionary model to transform the tree (see Details). Default is <code>none</code>.</p>
</td></tr>
<tr><td><code id="tree_discrete_+3A_bounds">bounds</code></td>
<td>
<p>settings to constrain parameter estimates. See <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code></p>
</td></tr>
<tr><td><code id="tree_discrete_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use. If 'NULL', number of cores is detected.</p>
</td></tr>
<tr><td><code id="tree_discrete_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_discrete_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits different models of discrete character evolution using <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code>
to n trees, randomly picked in a multiPhylo file. Currently, only binary discrete traits are supported
</p>
<p>Different character model from <code>fitDiscrete</code> can be used, including <code>ER</code> (equal-rates), 
<code>SYM</code> (symmetric), <code>ARD</code> (all-rates-different) and <code>meristic</code> (stepwise fashion). 
</p>
<p>All transformations to the phylogenetic tree from <code>fitDiscrete</code> can be used, i.e. <code>none</code>,
<code>EB</code>, <code>lambda</code>, <code>kappa</code> and<code>delta</code>.
</p>
<p>See <code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code> for more details on character models and tree transformations. 
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_discrete</code> returns a list with the following
components:
</p>
<p><code>call</code>: The function call
</p>
<p><code>data</code>: The original full data vector
</p>
<p><code>sensi.estimates</code>: Parameter estimates (transition rates q12 and q21), 
AICc and the optimised value of the phylogenetic transformation parameter (e.g. <code>lambda</code>) 
for each analysis with a different phylogenetic tree.
</p>
<p><code>N.tree</code>: Number of trees <code>n.tree</code> analysed
</p>
<p><code>stats</code>: Main statistics for model parameters, i.e. minimum, maximum, mean, median and sd-values
</p>
<p><code>optpar</code>: Transformation parameter used (e.g. <code>lambda</code>, <code>kappa</code> etc.)
</p>


<h3>Author(s)</h3>

<p>Gijsbert Werner &amp; Caterina Penone
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Yang Z. 2006. Computational Molecular Evolution. Oxford University Press: Oxford. 
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008.
GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>


<h3>See Also</h3>

<p><code><a href="geiger.html#topic+fitDiscrete">fitDiscrete</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Load data:
data("primates")
#Create a binary trait factor 
adultMass_binary&lt;-ifelse(primates$data$adultMass &gt; 7350, "big", "small")
adultMass_binary&lt;-as.factor(as.factor(adultMass_binary))
names(adultMass_binary)&lt;-rownames(primates$data)
#Model trait evolution accounting for phylogenetic uncertainty
tree_binary&lt;-tree_discrete(data = adultMass_binary,phy = primates$phy,
model = "ARD",transform = "none",n.tree = 30,n.cores = 2,track = TRUE)
#Print summary statistics
summary(tree_binary)
sensi_plot(tree_binary)
sensi_plot(tree_binary,graphs="q12")
sensi_plot(tree_binary,graphs="q21")
#Use a different evolutionary model or transformation.
tree_binary_lambda&lt;-tree_discrete(data = adultMass_binary,phy = primates$phy,
model = "SYM",transform = "lambda",n.tree = 30,n.cores = 2,track = TRUE)
summary(tree_binary_lambda) #Using Pagel's Lambda
sensi_plot(tree_binary_lambda)  
#Symmetrical rates, with an Early Burst (EB) model of trait evolution
tree_binary_SYM_EB&lt;-tree_discrete(data = adultMass_binary,phy = primates$phy,
model = "SYM",transform = "EB",n.tree = 30,n.cores = 2,track = TRUE)
summary(tree_binary_SYM_EB)
sensi_plot(tree_binary_SYM_EB) 
sensi_plot(tree_binary_SYM_EB,graphs="optpar") 

## End(Not run)
</code></pre>

<hr>
<h2 id='tree_influ_phyglm'>Interaction between phylogenetic uncertainty and influential species detection - Phylogenetic Logistic Regression</h2><span id='topic+tree_influ_phyglm'></span>

<h3>Description</h3>

<p>Performs leave-one-out deletion analysis for phylogenetic logistic regression,
and detects influential species while evaluating uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_influ_phyglm(
  formula,
  data,
  phy,
  n.tree = 2,
  cutoff = 2,
  btol = 50,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_influ_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="tree_influ_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_influ_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="tree_influ_phyglm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.</p>
</td></tr>
<tr><td><code id="tree_influ_phyglm_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff value used to identify for influential species
(see Details)</p>
</td></tr>
<tr><td><code id="tree_influ_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="tree_influ_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_influ_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one species at a time, fits a phylogenetic
logistic regression model using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, stores the
results and detects influential species. It repeats this operation using n trees, 
randomly picked in a multiPhylo file.
</p>
<p>Currently only logistic regression using the &quot;logistic_MPLE&quot;-method from
<code>phyloglm</code> is implemented.
</p>
<p><code>influ_phyglm</code> detects influential species based on the standardised
difference in intercept and/or slope when removing a given species compared
to the full model including all species. Species with a standardised difference
above the value of <code>cutoff</code> are identified as influential. The default
value for the cutoff is 2 standardised differences change.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>influ_phyglm</code> returns a list with the following
components:
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (i.e. <code>alpha</code>) for the full model
without deleted species.
</p>
<p><code>influential_species</code>: List of influential species, both
based on standardised difference in interecept and in the slope of the
regression. Species are ordered from most influential to less influential and
only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade for a given random tree. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the standardised
difference (<code>sDIFintercept</code>), the percentage of change in intercept compared
to the full model (<code>intercept.perc</code>) and intercept p-value
(<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter
(i.e. <code>alpha</code>) are reported.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Species where deletion resulted in errors.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+tree_phyglm">tree_phyglm</a></code>,
<code><a href="#topic+influ_phyglm">influ_phyglm</a></code>, <code><a href="#topic+tree_influ_phyglm">tree_influ_phyglm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Simulate Data:
set.seed(6987)
mphy = rmtree(100, N = 30)
x = rTrait(n=1,phy=mphy[[1]])
X = cbind(rep(1,100),x)
y = rbinTrait(n=1,phy=mphy[[1]], beta=c(-1,0.5), alpha=.7 ,X=X)
dat = data.frame(y, x)
# Run sensitivity analysis:
tree_influ &lt;- tree_influ_phyglm(y ~ x, data = dat, phy = mphy, n.tree = 5)
summary(tree_influ)
sensi_plot(tree_influ)
sensi_plot(tree_influ, graphs = 1)
sensi_plot(tree_influ, graphs = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='tree_influ_phylm'>Interaction between phylogenetic uncertainty and influential species detection - Phylogenetic Linear Regression</h2><span id='topic+tree_influ_phylm'></span>

<h3>Description</h3>

<p>Performs leave-one-out deletion analysis for phylogenetic linear regression,
and detects influential species while evaluating uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_influ_phylm(
  formula,
  data,
  phy,
  n.tree = 2,
  cutoff = 2,
  model = "lambda",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_influ_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="tree_influ_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_influ_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="tree_influ_phylm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.</p>
</td></tr>
<tr><td><code id="tree_influ_phylm_+3A_cutoff">cutoff</code></td>
<td>
<p>The cutoff value used to identify for influential species
(see Details)</p>
</td></tr>
<tr><td><code id="tree_influ_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="tree_influ_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_influ_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function sequentially removes one species at a time, fits a phylogenetic
linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, stores the
results and detects influential species. It repeats this operation using n trees, 
randomly picked in a multiPhylo file.
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p><code>influ_phylm</code> detects influential species based on the standardised
difference in intercept and/or slope when removing a given species compared
to the full model including all species. Species with a standardised difference
above the value of <code>cutoff</code> are identified as influential. The default
value for the cutoff is 2 standardised differences change.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>influ_phylm</code> returns a list with the following
components:
</p>
<p><code>cutoff</code>: The value selected for <code>cutoff</code>
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for the full model
without deleted species.
</p>
<p><code>influential_species</code>: List of influential species, both
based on standardised difference in intercept and in the slope of the
regression. Species are ordered from most influential to less influential and
only include species with a standardised difference &gt; <code>cutoff</code>.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a deleted clade for a given random tree. Columns report the calculated
regression intercept (<code>intercept</code>), difference between simulation
intercept and full model intercept (<code>DIFintercept</code>), the standardised
difference (<code>sDIFintercept</code>), the percentage of change in intercept compared
to the full model (<code>intercept.perc</code>) and intercept p-value
(<code>pval.intercept</code>). All these parameters are also reported for the regression
slope (<code>DIFestimate</code> etc.). Additionally, model aic value (<code>AIC</code>) and
the optimised value (<code>optpar</code>) of the phylogenetic parameter
(e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model used) are
reported.
</p>
<p><code>data</code>: Original full dataset.
</p>
<p><code>errors</code>: Species where deletion resulted in errors.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Caterina Penone &amp; Gijsbert D.A. Werner
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>, 
<code><a href="#topic+influ_phylm">influ_phylm</a></code>, <code><a href="#topic+tree_influ_phyglm">tree_influ_phyglm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load data:
data(alien)
# run analysis:
tree_influ &lt;- tree_influ_phylm(log(gestaLen) ~ log(adultMass), phy = alien$phy, 
data = alien$data, n.tree = 5)
# To check summary results:
summary(tree_influ)
# Visual diagnostics
sensi_plot(tree_influ)
sensi_plot(tree_influ, graphs = 1)
sensi_plot(tree_influ, graphs = 2)

data(alien)
tree_influ &lt;- tree_influ_phylm(log(gestaLen) ~ log(adultMass), phy = alien$phy, 
data = alien$data[1:25, ], n.tree = 2)
summary(tree_influ)

## End(Not run)
</code></pre>

<hr>
<h2 id='tree_intra_phyglm'>Interaction between phylogenetic uncertainty and intraspecific variability - Phylogenetic logistic Regression</h2><span id='topic+tree_intra_phyglm'></span>

<h3>Description</h3>

<p>Performs Phylogenetic logistic regression evaluating
intraspecific variability in response and/or predictor variables
and uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_intra_phyglm(
  formula,
  data,
  phy,
  Vx = NULL,
  x.transf = NULL,
  n.intra = 10,
  n.tree = 2,
  distrib = "normal",
  track = TRUE,
  btol = 50,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_intra_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula: <code>response~predictor</code>.</p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits and species names as row names.</p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>log</code> or <code>sqrt</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of times to repeat the analysis generating a random value for response and/or predictor variables.
If NULL, <code>n.intra</code> = 30</p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.
If NULL, <code>n.tree</code> = 2</p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx or Vy = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="tree_intra_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a phylogenetic logistic regression model using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code> to n trees (<code>n.tree</code>), 
randomly picked in a multiPhylo file. The regression is also repeated <code>n.intra</code> times.
At each iteration the function generates a random value for each row in the dataset using the standard deviation 
or errors supplied and assuming a normal or uniform distribution. To calculate means and se for your raw data, 
you can use the <code>summarySE</code> function from the package <code>Rmisc</code>.
</p>
<p>#' All phylogenetic models from <code>phyloglm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phyloglm</code> for details.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_intra_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>data</code>: Original full dataset
</p>
<p><code>sensi.estimates</code>: Coefficients, aic and the optimised value of the phylogenetic 
parameter (e.g. <code>lambda</code>) for each regression using a value in the interval of variation and 
a different phylogenetic tree.
</p>
<p><code>N.obs</code>: Size of the dataset after matching it with tree tips and removing NA's.
</p>
<p><code>stats</code>: Main statistics for model parameters.<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>
<p><code>all.stats</code>: Complete statistics for model parameters.
Fields coded using <code>all</code> describe statistics due to both intraspecific variation and phylogenetic uncertainty.
Fields coded using <code>intra</code> describe statistics due to intraspecific variation only.
Fields coded using <code>tree</code> describe statistics due to phylogenetic uncertainty only.
<code>sd</code> is the standard deviation. <code>CI_low</code> and <code>CI_high</code> are the lower and upper limits 
of the 95
</p>
<p><code>sp.pb</code>: Species that caused problems with data transformation (see details above).
</p>


<h3>Warning</h3>

<p>When Vy or Vx exceed Y or X, respectively, negative (or null) values can be generated, this might cause problems
for data transformation (e.g. log-transformation). In these cases, the function will skip the simulation. This problem can
be solved by increasing <code>times</code>, changing the transformation type and/or checking the target species in output$sp.pb.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Pablo Ariel Martinez
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Martinez, P. a., Zurano, J.P., Amado, T.F., Penone, C., Betancur-R, R., 
Bidau, C.J. &amp; Jacobina, U.P. (2015). Chromosomal diversity in tropical reef 
fishes is related to body size and depth range. Molecular Phylogenetics and 
Evolution, 93, 1-4
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+tree_phyglm">tree_phyglm</a></code>, <code><a href="#topic+intra_phyglm">intra_phyglm</a></code>,
<code><a href="#topic+tree_intra_phylm">tree_intra_phylm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate data
set.seed(6987)
mphy = ape::rmtree(150, N = 30)
x = phylolm::rTrait(n=1,phy=mphy[[1]])
x_sd = rnorm(150,mean = 0.8,sd=0.2)
X = cbind(rep(1,150),x)
y = rbinTrait(n=1,phy=mphy[[1]], beta=c(-1,0.5), alpha=.7 ,X=X)
dat = data.frame(y, x, x_sd)
intra.tree &lt;- tree_intra_phyglm(y ~ x, data = dat, phy = mphy, n.intra = 3, 
                                           n.tree = 3, Vx = "x_sd")
# summary results:
summary(intra.tree)
# Visual diagnostics for phylogenetic uncertainty:
sensi_plot(intra.tree, uncer.type = "all") #or uncer.type = "tree", uncer.type = "intra"
</code></pre>

<hr>
<h2 id='tree_intra_phylm'>Interaction between phylogenetic uncertainty and intraspecific variability  - Phylogenetic Linear Regression</h2><span id='topic+tree_intra_phylm'></span>

<h3>Description</h3>

<p>Performs Phylogenetic linear regression evaluating
intraspecific variability in response and/or predictor variables
and uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_intra_phylm(
  formula,
  data,
  phy,
  Vy = NULL,
  Vx = NULL,
  y.transf = NULL,
  x.transf = NULL,
  n.intra = 10,
  n.tree = 2,
  distrib = "normal",
  model = "lambda",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_intra_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula: <code>response~predictor</code>.</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits and species names as row names.</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_vy">Vy</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the response 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code>.</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_vx">Vx</code></td>
<td>
<p>Name of the column containing the standard deviation or the standard error of the predictor 
variable. When information is not available for one taxon, the value can be 0 or <code>NA</code></p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_y.transf">y.transf</code></td>
<td>
<p>Transformation for the response variable (e.g. <code>log</code> or <code>sqrt</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_x.transf">x.transf</code></td>
<td>
<p>Transformation for the predictor variable (e.g. <code>log</code> or <code>sqrt</code>). Please use this 
argument instead of transforming data in the formula directly (see also details below).</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_n.intra">n.intra</code></td>
<td>
<p>Number of times to repeat the analysis generating a random value for response and/or predictor variables.
If NULL, <code>n.intra</code> = 30</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.
If NULL, <code>n.tree</code> = 2</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_distrib">distrib</code></td>
<td>
<p>A character string indicating which distribution to use to generate a random value for the response 
and/or predictor variables. Default is normal distribution: &quot;normal&quot; (function <code><a href="stats.html#topic+rnorm">rnorm</a></code>).
Uniform distribution: &quot;uniform&quot; (<code><a href="stats.html#topic+runif">runif</a></code>)
Warning: we recommend to use normal distribution with Vx or Vy = standard deviation of the mean.</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_intra_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a phylogenetic linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code> to n trees (<code>n.tree</code>), 
randomly picked in a multiPhylo file. The regression is also repeated <code>n.intra</code> times.
At each iteration the function generates a random value for each row in the dataset using the standard deviation 
or errors supplied and assuming a normal or uniform distribution. To calculate means and se for your raw data, 
you can use the <code>summarySE</code> function from the package <code>Rmisc</code>.
</p>
<p>#' All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_intra_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>data</code>: Original full dataset
</p>
<p><code>sensi.estimates</code>: Coefficients, aic and the optimised value of the phylogenetic 
parameter (e.g. <code>lambda</code>) for each regression using a value in the interval of variation and 
a different phylogenetic tree.
</p>
<p><code>N.obs</code>: Size of the dataset after matching it with tree tips and removing NA's.
</p>
<p><code>stats</code>: Main statistics for model parameters.<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>
<p><code>all.stats</code>: Complete statistics for model parameters.
Fields coded using <code>all</code> describe statistics due to both intraspecific variation and phylogenetic uncertainty.
Fields coded using <code>intra</code> describe statistics due to intraspecific variation only.
Fields coded using <code>tree</code> describe statistics due to phylogenetic uncertainty only.
<code>sd</code> is the standard deviation. <code>CI_low</code> and <code>CI_high</code> are the lower and upper limits 
of the 95
</p>
<p><code>sp.pb</code>: Species that caused problems with data transformation (see details above).
</p>


<h3>Warning</h3>

<p>When Vy or Vx exceed Y or X, respectively, negative (or null) values can be generated, this might cause problems
for data transformation (e.g. log-transformation). In these cases, the function will skip the simulation. This problem can
be solved by increasing <code>times</code>, changing the transformation type and/or checking the target species in output$sp.pb.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Pablo Ariel Martinez
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Martinez, P. a., Zurano, J.P., Amado, T.F., Penone, C., Betancur-R, R., 
Bidau, C.J. &amp; Jacobina, U.P. (2015). Chromosomal diversity in tropical reef 
fishes is related to body size and depth range. Molecular Phylogenetics and 
Evolution, 93, 1-4
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+tree_phylm">tree_phylm</a></code>, <code><a href="#topic+intra_phylm">intra_phylm</a></code>,
<code><a href="#topic+tree_intra_phyglm">tree_intra_phyglm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data:
data(alien)
# run PGLS accounting for intraspecific and phylogenetic variation:
intra.tree &lt;- tree_intra_phylm(gestaLen ~ adultMass, data = alien$data, phy = alien$phy,
Vy = "SD_gesta", n.intra = 3, n.tree = 3, y.transf = log, x.transf = log)
# To check summary results:
summary(intra.tree)
# Visual diagnostics
sensi_plot(intra.tree, uncer.type = "all") #or uncer.type = "tree", uncer.type = "intra"
</code></pre>

<hr>
<h2 id='tree_phyglm'>Phylogenetic uncertainty - Phylogenetic Logistic Regression</h2><span id='topic+tree_phyglm'></span>

<h3>Description</h3>

<p>Performs Phylogenetic logistic regression evaluating
uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_phyglm(formula, data, phy, n.tree = 2, btol = 50, track = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="tree_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with species as row names.</p>
</td></tr>
<tr><td><code id="tree_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'multiPhylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_phyglm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.
If NULL, <code>n.tree</code> = 2</p>
</td></tr>
<tr><td><code id="tree_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code>.</p>
</td></tr>
<tr><td><code id="tree_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a phylogenetic logistic regression model using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>
to n trees, randomly picked in a multiPhylo file.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_phyglm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>data</code>: Original full dataset
</p>
<p><code>sensi.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for each regression with a 
different phylogenetic tree.
</p>
<p><code>N.obs</code>: Size of the dataset after matching it with tree tips and removing NA's.
</p>
<p><code>stats</code>: Main statistics for model parameters.<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>
<p><code>all.stats</code>: Complete statistics for model parameters. <code>sd_intra</code> is the standard deviation 
due to intraspecific variation. <code>CI_low</code> and <code>CI_high</code> are the lower and upper limits 
of the 95
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Pablo Ariel Martinez
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Donoghue, M.J. &amp; Ackerly, D.D. (1996). Phylogenetic Uncertainties and 
Sensitivity Analyses in Comparative Biology. Philosophical Transactions:
Biological Sciences, pp. 1241-1249.
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>,<code><a href="#topic+tree_phylm">tree_phylm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Simulating Data:
set.seed(6987)
mphy = rmtree(150, N = 30)
x = rTrait(n=1,phy=mphy[[1]])
X = cbind(rep(1,150),x)
y = rbinTrait(n=1,phy=mphy[[1]], beta=c(-1,0.5), alpha=.7 ,X=X)
dat = data.frame(y, x)
# Run sensitivity analysis:
tree &lt;- tree_phyglm(y ~ x, data = dat, phy = mphy, n.tree = 30)
# summary results:
summary(tree)
# Visual diagnostics for phylogenetic uncertainty:
sensi_plot(tree)
</code></pre>

<hr>
<h2 id='tree_phylm'>Phylogenetic uncertainty - Phylogenetic Linear Regression</h2><span id='topic+tree_phylm'></span>

<h3>Description</h3>

<p>Performs Phylogenetic linear regression evaluating
uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_phylm(formula, data, phy, n.tree = 2, model = "lambda", track = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="tree_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with species as row names.</p>
</td></tr>
<tr><td><code id="tree_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'multiPhylo', see ?<code>ape</code>).</p>
</td></tr>
<tr><td><code id="tree_phylm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.
If NULL, <code>n.tree</code> = 2</p>
</td></tr>
<tr><td><code id="tree_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="tree_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a phylogenetic linear regression model using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>
to n trees, randomly picked in a multiPhylo file.
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>data</code>: Original full dataset
</p>
<p><code>sensi.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code>) for each regression with a 
different phylogenetic tree.
</p>
<p><code>N.obs</code>: Size of the dataset after matching it with tree tips and removing NA's.
</p>
<p><code>stats</code>: Main statistics for model parameters.<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>
<p><code>all.stats</code>: Complete statistics for model parameters. <code>sd_intra</code> is the standard deviation 
due to intraspecific variation. <code>CI_low</code> and <code>CI_high</code> are the lower and upper limits 
of the 95
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Pablo Ariel Martinez
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Donoghue, M.J. &amp; Ackerly, D.D. (1996). Phylogenetic Uncertainties and 
Sensitivity Analyses in Comparative Biology. Philosophical Transactions:
Biological Sciences, pp. 1241-1249.
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+sensi_plot">sensi_plot</a></code>, <code><a href="#topic+tree_phyglm">tree_phyglm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data:
data(alien)
# This analysis needs a multiphylo file:
class(alien$phy)
alien$phy
# run PGLS accounting for phylogenetic uncertain:
tree &lt;- tree_phylm(log(gestaLen) ~ log(adultMass), phy = alien$phy, 
data = alien$data, n.tree = 30)
# To check summary results:
summary(tree)
# Visual diagnostics
sensi_plot(tree)
# You can specify which graph to print: 
sensi_plot(tree, graphs = 3)
</code></pre>

<hr>
<h2 id='tree_physig'>Phylogenetic uncertainty - Phylogenetic signal</h2><span id='topic+tree_physig'></span>

<h3>Description</h3>

<p>Performs phylogenetic signal estimates evaluating
uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_physig(
  trait.col,
  data,
  phy,
  n.tree = "all",
  method = "K",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_physig_+3A_trait.col">trait.col</code></td>
<td>
<p>The name of a column in the provided data frame with trait 
to be analyzed  (e.g. &quot;Body_mass&quot;).</p>
</td></tr>
<tr><td><code id="tree_physig_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_physig_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="tree_physig_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file. (If <code>n.tree</code> = &quot;all&quot;, phylogenetic signal will be estimated
among the all set of trees provided in <code>phy</code>)</p>
</td></tr>
<tr><td><code id="tree_physig_+3A_method">method</code></td>
<td>
<p>Method to compute signal: can be &quot;K&quot; or &quot;lambda&quot;.</p>
</td></tr>
<tr><td><code id="tree_physig_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_physig_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylosig</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function estimates phylogenetic signal using <code><a href="phytools.html#topic+phylosig">phylosig</a></code>
to n trees, randomly picked in a multiPhylo file.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>tree_physig</code> returns a list with the following
components:
</p>
<p><code>Trait</code>: Column name of the trait analysed
</p>
<p><code>data</code>: Original full dataset
</p>
<p><code>tree.physig.estimates</code>: Three number, phylogenetic signal estimate 
(lambda or K) and the p-value for each run with a different phylogenetic tree.
</p>
<p><code>N.obs</code>: Size of the dataset after matching it with tree tips and removing NA's.
</p>
<p><code>stats</code>: Main statistics for phylogenetic estimates.<code>CI_low</code> and <code>CI_high</code> are the lower 
and upper limits of the 95
</p>


<h3>Note</h3>

<p>The argument &quot;se&quot; from <code><a href="phytools.html#topic+phylosig">phylosig</a></code> is not available in this function. Use the 
argument &quot;V&quot; instead with <code><a href="#topic+intra_physig">intra_physig</a></code> to indicate the name of the column containing the standard 
deviation or the standard error of the trait variable instead.
</p>


<h3>Author(s)</h3>

<p>Caterina Penone &amp; Gustavo Paterno
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Donoghue, M.J. &amp; Ackerly, D.D. (1996). Phylogenetic Uncertainties and 
Sensitivity Analyses in Comparative Biology. Philosophical Transactions:
Biological Sciences, pp. 1241-1249.
</p>
<p>Blomberg, S. P., T. Garland Jr., A. R. Ives (2003) 
Testing for phylogenetic signal in comparative data: 
Behavioral traits are more labile. Evolution, 57, 717-745.
</p>
<p>Pagel, M. (1999) Inferring the historical patterns of biological evolution. 
Nature, 401, 877-884.
</p>
<p>Kamilar, J. M., &amp; Cooper, N. (2013). Phylogenetic signal in primate behaviour,
ecology and life history. Philosophical Transactions of the Royal Society 
B: Biological Sciences, 368: 20120341.
</p>


<h3>See Also</h3>

<p><code><a href="phytools.html#topic+phylosig">phylosig</a></code>, 
<code><a href="#topic+tree_phylm">tree_phylm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data:
data(alien)
alien.data&lt;-alien$data
alien.phy&lt;-alien$phy
# Logtransform data
alien.data$logMass &lt;- log(alien.data$adultMass) 
# Run sensitivity analysis:
tree &lt;- tree_physig(trait.col = "logMass", data = alien.data, 
phy = alien.phy, n.tree = 10)
summary(tree)
sensi_plot(tree)
sensi_plot(tree, graphs = 1)
sensi_plot(tree, graphs = 2)
</code></pre>

<hr>
<h2 id='tree_samp_phyglm'>Interaction between phylogenetic uncertainty and sensitivity to species sampling  - Phylogenetic Logistic Regression</h2><span id='topic+tree_samp_phyglm'></span>

<h3>Description</h3>

<p>Performs analyses of sensitivity to species sampling by randomly removing
species and detecting the effects on parameter estimates in phylogenetic
logistic regression, while evaluating uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_samp_phyglm(
  formula,
  data,
  phy,
  n.sim = 30,
  n.tree = 2,
  breaks = seq(0.1, 0.5, 0.1),
  btol = 50,
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_samp_phyglm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="tree_samp_phyglm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_samp_phyglm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="tree_samp_phyglm_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times species are randomly deleted for each
<code>break</code>.</p>
</td></tr>
<tr><td><code id="tree_samp_phyglm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.</p>
</td></tr>
<tr><td><code id="tree_samp_phyglm_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="tree_samp_phyglm_+3A_btol">btol</code></td>
<td>
<p>Bound on searching space. For details see <code>phyloglm</code></p>
</td></tr>
<tr><td><code id="tree_samp_phyglm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_samp_phyglm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phyloglm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by
<code>breaks</code>) from the full phylogenetic logistic regression, fits a phylogenetic
logistic regression model without these species using <code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>,
repeats this many times (controlled by <code>times</code>), stores the results and
calculates the effects on model parameters. It repeats this operation using n trees, 
randomly picked in a multiPhylo file.
</p>
<p>Only logistic regression using the &quot;logistic_MPLE&quot;-method from
<code>phyloglm</code> is implemented.
</p>
<p>Currently, this function can only implement simple logistic models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>samp_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code> or <code>kappa</code>) for
the full model without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a model rerun with a given number of species
<code>n.remov</code> removed, representing <code>n.percent</code> of the full dataset.
Columns report the calculated regression intercept (<code>intercept</code>),
difference between simulation intercept and full model intercept (<code>DIFintercept</code>),
the percentage of change in intercept compared to the full model (<code>intercept.perc</code>)
and intercept p-value (<code>pval.intercept</code>). All these parameters are also reported
for the regression slope (<code>DIFestimate</code> etc.). Additionally, model aic value
(<code>AIC</code>) and the optimised value (<code>optpar</code>) of the phylogenetic
parameter (e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model
used) are reported. Lastly we reported the standardised difference in intercept 
(<code>sDIFintercept</code>) and slope (<code>sDIFestimate</code>).
</p>
<p><code>sign.analysis</code> For each break (i.e. each percentage of species
removed) this reports the percentage of statistically significant (at p&lt;0.05)
intercepts (<code>perc.sign.intercept</code>) over all repetitions as well as the
percentage of statisticaly significant (at p&lt;0.05) slopes (<code>perc.sign.estimate</code>).
</p>
<p><code>data</code>: Original full dataset.
</p>


<h3>Note</h3>

<p>Please be aware that dropping species may reduce power to detect 
significant slopes/intercepts and may partially be responsible for a potential 
effect of species removal on p-values. Please also consult standardised differences
in the (summary) output.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Gijsbert D.A. Werner &amp; Caterina Penone
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014).
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation
in angiosperms. Nature Communications, 5, 4087.
</p>
<p>#' Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phyloglm">phyloglm</a></code>, <code><a href="#topic+samp_phyglm">samp_phyglm</a></code>,
<code><a href="#topic+tree_phyglm">tree_phyglm</a></code>, <code><a href="#topic+tree_samp_phylm">tree_samp_phylm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Simulate Data:
set.seed(6987)
mphy = rmtree(100, N = 30)
x = rTrait(n=1,phy=mphy[[1]])
X = cbind(rep(1,100),x)
y = rbinTrait(n=1,phy=mphy[[1]], beta=c(-1,0.5), alpha=.7 ,X=X)
dat = data.frame(y, x)
# Run sensitivity analysis:
tree_samp &lt;- tree_samp_phyglm(y ~ x, data = dat, phy = mphy, n.tree = 3, n.sim=10) 
summary(tree_samp)
sensi_plot(tree_samp)
sensi_plot(tree_samp, graphs = 1)
sensi_plot(tree_samp, graphs = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='tree_samp_phylm'>Interaction between phylogenetic uncertainty and sensitivity to species sampling - Phylogenetic Linear Regression</h2><span id='topic+tree_samp_phylm'></span>

<h3>Description</h3>

<p>Performs analyses of sensitivity to species sampling by randomly removing
species and detecting the effects on parameter estimates in a phylogenetic
linear regression, while evaluating uncertainty in trees topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_samp_phylm(
  formula,
  data,
  phy,
  n.sim = 30,
  n.tree = 2,
  breaks = seq(0.1, 0.5, 0.1),
  model = "lambda",
  track = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree_samp_phylm_+3A_formula">formula</code></td>
<td>
<p>The model formula</p>
</td></tr>
<tr><td><code id="tree_samp_phylm_+3A_data">data</code></td>
<td>
<p>Data frame containing species traits with row names matching tips
in <code>phy</code>.</p>
</td></tr>
<tr><td><code id="tree_samp_phylm_+3A_phy">phy</code></td>
<td>
<p>A phylogeny (class 'phylo') matching <code>data</code>.</p>
</td></tr>
<tr><td><code id="tree_samp_phylm_+3A_n.sim">n.sim</code></td>
<td>
<p>The number of times species are randomly deleted for each
<code>break</code>.</p>
</td></tr>
<tr><td><code id="tree_samp_phylm_+3A_n.tree">n.tree</code></td>
<td>
<p>Number of times to repeat the analysis with n different trees picked 
randomly in the multiPhylo file.</p>
</td></tr>
<tr><td><code id="tree_samp_phylm_+3A_breaks">breaks</code></td>
<td>
<p>A vector containing the percentages of species to remove.</p>
</td></tr>
<tr><td><code id="tree_samp_phylm_+3A_model">model</code></td>
<td>
<p>The phylogenetic model to use (see Details). Default is <code>lambda</code>.</p>
</td></tr>
<tr><td><code id="tree_samp_phylm_+3A_track">track</code></td>
<td>
<p>Print a report tracking function progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="tree_samp_phylm_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>phylolm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly removes a given percentage of species (controlled by
<code>breaks</code>) from the full phylogenetic linear regression, fits a phylogenetic
linear regression model without these species using <code><a href="phylolm.html#topic+phylolm">phylolm</a></code>,
repeats this many times (controlled by <code>n.sim</code>), stores the results and
calculates the effects on model parameters. It repeats this operation using n trees, 
randomly picked in a multiPhylo file.
</p>
<p>All phylogenetic models from <code>phylolm</code> can be used, i.e. <code>BM</code>,
<code>OUfixedRoot</code>, <code>OUrandomRoot</code>, <code>lambda</code>, <code>kappa</code>,
<code>delta</code>, <code>EB</code> and <code>trend</code>. See ?<code>phylolm</code> for details.
</p>
<p>Currently, this function can only implement simple linear models (i.e. <code class="reqn">trait~
predictor</code>). In the future we will implement more complex models.
</p>
<p>Output can be visualised using <code>sensi_plot</code>.
</p>


<h3>Value</h3>

<p>The function <code>samp_phylm</code> returns a list with the following
components:
</p>
<p><code>formula</code>: The formula
</p>
<p><code>full.model.estimates</code>: Coefficients, aic and the optimised
value of the phylogenetic parameter (e.g. <code>lambda</code> or <code>kappa</code>) for
the full model without deleted species.
</p>
<p><code>sensi.estimates</code>: A data frame with all simulation
estimates. Each row represents a model rerun with a given number of species
<code>n.remov</code> removed, representing <code>n.percent</code> of the full dataset.
Columns report the calculated regression intercept (<code>intercept</code>),
difference between simulation intercept and full model intercept (<code>DIFintercept</code>),
the percentage of change in intercept compared to the full model (<code>intercept.perc</code>)
and intercept p-value (<code>pval.intercept</code>). All these parameters are also reported
for the regression slope (<code>DIFestimate</code> etc.). Additionally, model aic value
(<code>AIC</code>) and the optimised value (<code>optpar</code>) of the phylogenetic
parameter (e.g. <code>kappa</code> or <code>lambda</code>, depending on the phylogenetic model
used) are reported. Lastly we reported the standardised difference in intercept 
(<code>sDIFintercept</code>) and slope (<code>sDIFestimate</code>).
</p>
<p><code>sign.analysis</code> For each break (i.e. each percentage of species
removed) this reports the percentage of statistically significant (at p&lt;0.05)
intercepts (<code>perc.sign.intercept</code>) over all repetitions as well as the
percentage of statisticaly significant (at p&lt;0.05) slopes (<code>perc.sign.estimate</code>).
</p>
<p><code>data</code>: Original full dataset.
#' @note Please be aware that dropping species may reduce power to detect 
significant slopes/intercepts and may partially be responsible for a potential 
effect of species removal on p-values. Please also consult standardised differences
in the (summary) output.
</p>


<h3>Author(s)</h3>

<p>Gustavo Paterno, Gijsbert D.A. Werner &amp; Caterina Penone
</p>


<h3>References</h3>

<p>Paterno, G. B., Penone, C. Werner, G. D. A. 
<a href="http://doi.wiley.com/10.1111/2041-210X.12990">sensiPhy: 
An r-package for sensitivity analysis in phylogenetic 
comparative methods.</a> Methods in Ecology and Evolution 
2018, 9(6):1461-1467
</p>
<p>Werner, G.D.A., Cornwell, W.K., Sprent, J.I., Kattge, J. &amp; Kiers, E.T. (2014).
A single evolutionary innovation drives the deep evolution of symbiotic N2-fixation
in angiosperms. Nature Communications, 5, 4087.
</p>
<p>Ho, L. S. T. and Ane, C. 2014. &quot;A linear-time algorithm for 
Gaussian and non-Gaussian trait evolution models&quot;. Systematic Biology 63(3):397-408.
</p>


<h3>See Also</h3>

<p><code><a href="phylolm.html#topic+phylolm">phylolm</a></code>, <code><a href="#topic+samp_phylm">samp_phylm</a></code>,
<code><a href="#topic+tree_phylm">tree_phylm</a></code>,<code><a href="#topic+tree_samp_phyglm">tree_samp_phyglm</a></code>,<code><a href="#topic+sensi_plot">sensi_plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load data:
data(alien)
# Run analysis:
samp &lt;- tree_samp_phylm(log(gestaLen) ~ log(adultMass), phy = alien$phy,
                                    data = alien$data, n.tree = 5, n.sim=10)
summary(samp)
head(samp$sensi.estimates)
# Visual diagnostics
sensi_plot(samp)
sensi_plot(samp, graphs = 1)
sensi_plot(samp, graphs = 2)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
