<!DOCTYPE html><html lang="en"><head><title>Help for package seq2R</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {seq2R}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#seq2R-package'>
<p>Simple method to detect compositional changes in genomic sequences</p></a></li>
<li><a href='#critical'>
<p>Critical points (maxima and minima)</p></a></li>
<li><a href='#find.points'>
<p>Simple method to detect compositional changes in genomic sequences</p></a></li>
<li><a href='#mtDNAhum'>
<p>Human Mitochondrial DNA</p></a></li>
<li><a href='#plot.change.points'><p>Visualization of change.points objects</p></a></li>
<li><a href='#print.change.points'>
<p>Short find.points summary</p></a></li>
<li><a href='#read.all'>
<p>Read FASTA and GBK formatted files</p></a></li>
<li><a href='#read.genbank'>
<p>Read DNA sequences from GenBank via internet</p></a></li>
<li><a href='#transform'>
<p>Convert biological sequences into binary code</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simple Method to Detect Compositional Changes in Genomic
Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-09-27</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nora M. Villanueva &lt;nmvillanueva@uvigo.gal&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.1)</td>
</tr>
<tr>
<td>Description:</td>
<td>This software is useful for loading '.fasta' or '.gbk' files, and for retrieving sequences from 'GenBank' dataset <a href="https://www.ncbi.nlm.nih.gov/genbank/">https://www.ncbi.nlm.nih.gov/genbank/</a>. This package allows  to detect differences or asymmetries based on nucleotide composition by using local linear kernel smoothers. Also, it is possible to draw inference about critical points (i. e. maximum or minimum points) related with the derivative curves. Additionally, bootstrap methods have been used  for estimating confidence intervals and speed computational techniques (binning techniques) have been implemented in 'seq2R'.</td>
</tr>
<tr>
<td>Imports:</td>
<td>seqinr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-30 13:01:08 UTC; sestelo</td>
</tr>
<tr>
<td>Author:</td>
<td>Nora M. Villanueva
    <a href="https://orcid.org/0000-0001-8085-2745"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Marta Sestelo <a href="https://orcid.org/0000-0003-4284-6509"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Alan Miller [ctb] (FORTRAN code lsq.f90: weighted least-squares module)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-30 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='seq2R-package'>
Simple method to detect compositional changes in genomic sequences
</h2><span id='topic+seq2R-package'></span><span id='topic+seq2R'></span>

<h3>Description</h3>

<p><code>seq2R</code> is just a shortcut for &quot;sequence to <code>R</code>&quot;. This software is useful for loading .fasta or .gbk files, and for recovering sequences from GenBank database.
This package allows  to detect differences or asymmetries based on nucleotide composition by using local linear kernel smoothers. Also, it is possible to draw inference about critical points (i. e. maximum or minimum points) related with the derivative curves. Additionally, bootstrap methods have been used  for estimating confidence intervals and speed computational techniques (binning techniques) have been implemented in &quot;seq2R&quot;.
</p>


<h3>Author(s)</h3>

<p>Nora M. Villanueva and Marta Sestelo.
</p>
<p>Maintainer: Nora M. Villanueva &lt;nmvillanueva@uvigo.es&gt;
</p>


<h3>References</h3>

<p>Efron, B. (1979). Bootstrap methods: another look at the jackknife. Annals of Statistics, 7:1-26.
</p>
<p>Efron, E. and Tibshirani, R. J. (1993). An introduction to the Bootstrap. Chapman and Hall, London.
</p>
<p>Fan, J. and Marron, J.S. (1994). Fast implementation of nonparametric curve estimators. Journal of Computational and Graphical Statistics, 3:35-56.
</p>
<p>Hastie, T. and  Tibshirani, R. (1990). Generalized additive models. Chapman and Hall, London.
</p>
<p>Stone, C. J. (1977). Consistent nonparametric regression. Annals of Statistics, 5: 595-620.
</p>
<p>Villanueva, N. M., Sestelo, M.,  Fonseca, M. M. and Roca-Pardinas, J.  (2023).
seq2R: An R package to detect change points in DNA sequences. Mathematics, 11 (10), 2299.
</p>
<p>Wand, M. P. and Jones, M. C. (1995). Kernel Smoothing. Chapman and Hall, London.
</p>

<hr>
<h2 id='critical'>
Critical points (maxima and minima)
</h2><span id='topic+critical'></span>

<h3>Description</h3>

<p>Function that maximizes or minimizes the first derivative of the model obtained with <code><a href="#topic+find.points">find.points</a></code> function. Also, it is included their 95% confidence intervals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>critical(model, base.pairs = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="critical_+3A_model">model</code></td>
<td>
<p><code>change.points</code> object.</p>
</td></tr>
<tr><td><code id="critical_+3A_base.pairs">base.pairs</code></td>
<td>
<p>Character string for A vs. T or C vs G.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned list has two component (<code>$AT</code>, <code>$CG</code>). Both of them containing  a matrix  with values about their critical points (maxima and minima), lower  and upper 95% confidence intervals.
</p>
<table role = "presentation">
<tr><td><code>AT</code></td>
<td>
<p>Critical points for AT.</p>
</td></tr>
<tr><td><code>CG</code></td>
<td>
<p>Critical points for CG.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nora M. Villanueva and Marta Sestelo.
</p>


<h3>References</h3>

<p>N. M. Villanueva, M. Sestelo, M. M. Fonseca and J. Roca-Pardinas (2023).
seq2R: An R package to detect change points in DNA sequences. Mathematics, 11 (10), 2299.</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seq2R)

#mtDNAhum &lt;- read.genbank("NC_012920")
data(mtDNAhum)
DNA &lt;- transform(mtDNAhum)
seq1 &lt;- find.points(DNA, nboot = 10)


critical(seq1,base.pairs="CG")


critical(seq1,base.pairs="AT")
</code></pre>

<hr>
<h2 id='find.points'>
Simple method to detect compositional changes in genomic sequences
</h2><span id='topic+find.points'></span>

<h3>Description</h3>

<p><code>find</code> is used to detect changes at genomic sequences composition. The method is based on fitting nonparametric models by using local linear kernel smoothers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.points(x,kbin= 300, p= 3, bandwidth=-1, weights= 1, nboot=100, kernel="gaussian",
n.bandwidths= 20, seed = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find.points_+3A_x">x</code></td>
<td>

<p>Sequences in binary system (by using <code>change.binary</code> function previously) are to be analyzed from.
</p>
</td></tr>
<tr><td><code id="find.points_+3A_kbin">kbin</code></td>
<td>

<p>The number of binning nodes over which the function is to be estimated.
</p>
</td></tr>
<tr><td><code id="find.points_+3A_p">p</code></td>
<td>

<p>Degree of the polynomial. By default <code>p=3</code>.
</p>
</td></tr>
<tr><td><code id="find.points_+3A_bandwidth">bandwidth</code></td>
<td>

<p>The kernel bandwidth or smoothing parameter. Large values of bandwidth make smoother estimates, smaller values of bandwidth make less smooth estimates. The default <code>h=-1</code> is a bandwidth compute by cross validation.
</p>
</td></tr>
<tr><td><code id="find.points_+3A_weights">weights</code></td>
<td>

<p>Weights.
</p>
</td></tr>
<tr><td><code id="find.points_+3A_nboot">nboot</code></td>
<td>

<p>Number of bootstrap repeats.
</p>
</td></tr>
<tr><td><code id="find.points_+3A_kernel">kernel</code></td>
<td>

<p>Character which denotes the kernel function (a symmetric density). By default <code>kernel = "gaussian"</code>, this is, the Gaussian density function. Also, other types of kernel functions can be used: Epanechnikov and triangle, <code>kernel="Epanech"</code> and <code>kernel="triang"</code>, respectively.
</p>
</td></tr>
<tr><td><code id="find.points_+3A_n.bandwidths">n.bandwidths</code></td>
<td>
<p>Number that it will be used to calculate the grid of bandwidths in a range between 0 and 1. In this grid, it will be selected the optimum bandwidth by cross-validation.If the optimum bandwidth value is close to 0, we will obtain rough estimates; when it is close to 1, we will obtain smooth estimates.
</p>
</td></tr>
<tr><td><code id="find.points_+3A_seed">seed</code></td>
<td>
<p>Seed to be used in the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="find.points_+3A_...">...</code></td>
<td>
<p>Other options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each genomic sequence the AT and CG skews profiles were calculated as <code class="reqn">A vs. T = (A-T)/(A+T)</code> and <code class="reqn">C vs. G = (C-G)/(C+G)</code>.
</p>


<h3>Value</h3>

<p>The function computes and returns a list of short information for a fitted <code>change.points</code> object.
</p>
<table role = "presentation">
<tr><td><code>Number of A-T base pairs</code></td>
<td>
<p>The returned value is the total nucleotide  (adenine and thymine) contained at the sequence analyzed.</p>
</td></tr>
<tr><td><code>Number of C-G base pairs</code></td>
<td>
<p>In this case, the returned value is the sum of cytosine and guanine contained at the sequence.</p>
</td></tr>
<tr><td><code>Number of binning nodes</code></td>
<td>
<p>The number of binning nodes over which the function is to be estimated.</p>
</td></tr>
<tr><td><code>Number of bootstrap repeats</code></td>
<td>
<p>Number of bootstrap repeats.</p>
</td></tr>
<tr><td><code>Bandwidth</code></td>
<td>
<p>Value of the kernel bandwidth or smoothing parameter used in the fitting for A vs. T and C vs. G.</p>
</td></tr>
<tr><td><code>Exists any critical point</code></td>
<td>
<p>Emphasize if there is or not any critical.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nora M. Villanueva and Marta Sestelo.
</p>


<h3>References</h3>

<p>N. M. Villanueva, M. Sestelo, M. M. Fonseca and J. Roca-Pardinas (2023).
seq2R: An R package to detect change points in DNA sequences. Mathematics, 11 (10), 2299.</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seq2R)


#mtDNAhum &lt;- read.genbank("NC_012920")
data(mtDNAhum)
DNA &lt;- transform(mtDNAhum)
seq1&lt;-find.points(DNA)
seq1
</code></pre>

<hr>
<h2 id='mtDNAhum'>
Human Mitochondrial DNA
</h2><span id='topic+mtDNAhum'></span>

<h3>Description</h3>

<p>The complete sequence of the human mitochondrial genome contains 16569 base pair. The sequence presents extreme economy in that the genes have none or only a few noncoding bases between them, and in many cases the termination codons are not coded in the DNA but are created post-transcriptionally by polyadenylation of the mRNAs. The genes for the 12S and 16S rRNAs, 22tRNAs, cytochrome c oxidase subunits I, II, and III, ATPase subunit 6, cytochrome b and eight other predicted protein coding genes have been located.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mtDNAhum)</code></pre>


<h3>References</h3>

<p>Anderson, S. and Bankier, A. T. and Barrell, B. G. and de Bruijn, M. H. L. and Coulson, A. R. and Drouin, J. and Eperon, I. C. and Nierlich, D. P. and Roe, B. A. and Sanger, F. and Schreier, P. H. and Smith, A. J. H. and Staden, R. and Young, I. G.(1981) Sequence and organization of the human mitochondrial genome. Nature, 5806(290):457:465
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mtDNAhum)
</code></pre>

<hr>
<h2 id='plot.change.points'>Visualization of change.points objects
</h2><span id='topic+plot.change.points'></span>

<h3>Description</h3>

<p>Useful for drawing the estimation and first derivative of the skew profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'change.points'
plot(x = model, y = NULL, base.pairs = NULL, der = NULL,
xlab = "x", ylab = "y", col = "black", CIcol = "black", main = NULL, type = "l",
CItype = "l", critical = FALSE, CIcritical = FALSE,ylim=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.change.points_+3A_x">x</code></td>
<td>

<p><code>change.points</code> object.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_y">y</code></td>
<td>
<p><code>NULL</code>
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_base.pairs">base.pairs</code></td>
<td>
<p>Character string about the skew profile for A vs. T or C vs. G.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_der">der</code></td>
<td>
<p> Number which determines inference process to be drawing into the plot. By default <code>der</code> is <code>NULL</code>. If it is <code>0</code>, the plot represents the initial estimate. If <code>der</code> is <code>1</code>, the first derivative is plotted.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_xlab">xlab</code></td>
<td>

<p>Title for x axis.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_ylab">ylab</code></td>
<td>

<p>Title for y axis.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_col">col</code></td>
<td>

<p>A specification for the default plotting color.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_cicol">CIcol</code></td>
<td>

<p>A specification for the default confidence intervals plotting color.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_main">main</code></td>
<td>

<p>An overall title for the plot.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_type">type</code></td>
<td>

<p>Type of plot should be drawn. Possible types are, <code>p</code> for points, <code>l</code> for lines, <code>o</code> for overplotted, etc. See details in <code>?par</code>.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_citype">CItype</code></td>
<td>
<p> Type of plot should be drawn for confidence intervals. Possible types are, <code>p</code> for points, <code>l</code> for lines, <code>o</code> for overplotted, etc. See details in <code>?par</code>.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_critical">critical</code></td>
<td>
<p> A logical value. If <code>TRUE</code> (not by default), the critical points are drawn into the plot.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_cicritical">CIcritical</code></td>
<td>
<p> A logical value. If <code>TRUE</code> (not by default), the 95% confidence intervals for the critical points are drawn into the plot.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_ylim">ylim</code></td>
<td>
<p>The y limits of the plot.
</p>
</td></tr>
<tr><td><code id="plot.change.points_+3A_...">...</code></td>
<td>

<p>Other options.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Simply produce a plot.
</p>


<h3>Author(s)</h3>

<p>Nora M. Villanueva and Marta Sestelo.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seq2R)


#mtDNAhum &lt;- read.genbank("NC_012920")
data(mtDNAhum)
DNA &lt;- transform(mtDNAhum)
seq1 &lt;- find.points(DNA)



plot(seq1,der=0,base.pairs="CG",CIcritical=TRUE,ylim=c(0.08,0.67))
plot(seq1,der=1,base.pairs="CG",CIcritical=TRUE,ylim=c(-0.0005,0.00045))
abline(h=0)



plot(seq1,critical=TRUE, CIcritical=TRUE)
</code></pre>

<hr>
<h2 id='print.change.points'>
Short find.points summary
</h2><span id='topic+print.change.points'></span>

<h3>Description</h3>

<p><code><a href="#topic+find.points">find.points</a></code> summary.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'change.points'
print(x=model,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.change.points_+3A_x">x</code></td>
<td>
<p><code>change.points</code> object.</p>
</td></tr>
<tr><td><code id="print.change.points_+3A_...">...</code></td>
<td>
<p>Other options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function computes and returns a list of short information for a fitted <code>change.points</code> object.
</p>
<table role = "presentation">
<tr><td><code>Number of A-T base pairs</code></td>
<td>
<p>The returned value is the total nucleotide  (adenine and thymine) contained in the sequence analyzed.</p>
</td></tr>
<tr><td><code>Number of C-G base pairs</code></td>
<td>
<p>In this case, the returned value is the sum of cytosine and guanine contained at the sequence.</p>
</td></tr>
<tr><td><code>Number of binning nodes</code></td>
<td>
<p>The number of binning nodes over which the function is to be estimated.</p>
</td></tr>
<tr><td><code>Number of bootstrap repeats</code></td>
<td>
<p>Number of bootstrap repeats.</p>
</td></tr>
<tr><td><code>Bandwidth</code></td>
<td>
<p>Value of the Kernel bandwidth or smoothing parameter used in the fiiting for A vs. T and C vs. G.</p>
</td></tr>
<tr><td><code>Exists any critical point</code></td>
<td>
<p>Emphasize if there is or not any critical.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>See details in <code><a href="#topic+find.points">find.points</a></code>.
</p>


<h3>Author(s)</h3>

<p>Nora M. Villanueva and Marta Sestelo.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seq2R)
#mtDNAhum &lt;- read.genbank("NC_012920")
data(mtDNAhum)
DNA &lt;- transform(mtDNAhum)
seq1 &lt;- find.points(DNA)
seq1
</code></pre>

<hr>
<h2 id='read.all'>
Read FASTA and GBK formatted files
</h2><span id='topic+read.all'></span>

<h3>Description</h3>

<p>Read nucleic acid sequences from a file in FASTA or GBK format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.all(file = system.file(""), seqtype = "DNA")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.all_+3A_file">file</code></td>
<td>
<p> The name of the file which the sequences in FASTA or GBK format are to be
read from.
</p>
</td></tr>
<tr><td><code id="read.all_+3A_seqtype">seqtype</code></td>
<td>
<p>The nature of the sequence. Nowadays only <code>DNA</code>, in further updates it will be possible to use for  different type of sequences.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fasta is a widely used format in molecular biology. Sequence in FASTA format starts with a single-line description, distinguished by a greater-than &lsquo;&gt;&rsquo; symbol,   followed by sequence data on the next lines.
</p>
<p>'GenBank' format files have the extension GBK, by convention. Files contain fields with different types of information well-labeled. The header of the file has information describing the sequence, such as its  type, shape, length  and source. The features of the genome sequence follow the header, and include protein translations. The DNA sequence is the last element of the file, which ends with (and must include) a soluble slash. Complete genomes in this format are available at the <a href="https://ftp.ncbi.nlm.nih.gov/genbank/">https://ftp.ncbi.nlm.nih.gov/genbank/</a>.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Sequence</code></td>
<td>
<p>The returned list has a component  <code>Sequence</code> containing the DNA sequence taken from the field &ldquo;ORIGIN&rdquo; in GenBank. The sequence is a vector of single characters.</p>
</td></tr>
<tr><td><code>Locus or accession</code></td>
<td>
<p>the returned list has a component <code>Locus/Accession</code> containing the names of the locus or accession number taken from the field &ldquo;LOCUS&rdquo; or &ldquo;ACCESSION&rdquo; in 'GenBank'. Also, return sequence length.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nora M. Villanueva and Marta Sestelo.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seq2R)
data(mtDNAhum)
## Not run: 
data&lt;-read.all("file.fasta")
data&lt;-read.all("file.gbk")

## End(Not run)
</code></pre>

<hr>
<h2 id='read.genbank'>
Read DNA sequences from GenBank via internet
</h2><span id='topic+read.genbank'></span>

<h3>Description</h3>

<p>This function connects to the GenBank database, and reads nucleotide sequences using locus code given as arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.genbank(locus)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.genbank_+3A_locus">locus</code></td>
<td>
<p> Vector of mode character giving the  locus code or accession number.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses the site <a href="https://pubmed.ncbi.nlm.nih.gov/">https://pubmed.ncbi.nlm.nih.gov/</a> (E-utilities) from where the sequences are downloaded. E-utilities are a set of eight server-side programs that provide a stable interface into the Entrez query and database system at the National Center for Biotechnology Informatio (NCBI). The E-utilities use a fixed URL syntax that translates a standard set of input parameters into the values necessary for various NCBI software components to search for and retrieve the requested data. The E-utilities are therefore the structured interface to the Entrez system, which currently includes 38 databases covering a variety of biomedical data, including nucleotide and protein sequences, gene records, three-dimensional molecular structures, and the biomedical literature.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Sequence</code></td>
<td>
<p>The returned list has a component  <code>Sequence</code> containing the DNA sequence taken from the field &ldquo;ORIGIN&rdquo; in GenBank. The sequence is a vector of single characters.</p>
</td></tr>
<tr><td><code>Locus or accession</code></td>
<td>
<p>The returned list has a component <code>Locus/Accession</code> containing the names of the locus or accession number taken from the field &ldquo;LOCUS&rdquo; or &ldquo;ACCESSION&rdquo; in GenBank.</p>
</td></tr>
<tr><td><code>Species</code></td>
<td>
<p>The returned list has an attribute <code>Species</code> containing the names of the species taken from the field &ldquo;ORGANISM&rdquo; in GenBank.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>If the computer is not connected to the internet, this function will not work.
</p>


<h3>Author(s)</h3>

<p>Nora M. Villanueva and Marta Sestelo.
</p>


<h3>References</h3>

<p>Bethesda M. D. (2010) Entrez Programming Utilities Help. NCBI Help Manual. NCBI, USA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seq2R)
#mthumanDNA &lt;- read.genbank("NC_012920")
#mthumanDNA
</code></pre>

<hr>
<h2 id='transform'>
Convert biological sequences into binary code
</h2><span id='topic+transform'></span>

<h3>Description</h3>

<p>Biological sequences are categorical variables. With this function the four nucleotides are coded with two bits, 0 and 1 (binary numeral system) for being used by almost all modern computers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transform(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="transform_+3A_x">x</code></td>
<td>

<p>The object obtained with <code><a href="#topic+read.all">read.all</a></code> or <code><a href="#topic+read.genbank">read.genbank</a></code> functions is the argument required for <code>transform</code>. The nature of the sequence is DNA. Sequences are returned as a vector of single characters.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned list has two component (<code>$AT</code>, <code>$CG</code>). Both of them containing  a matrix  with values about their critical points (maximum and minimum), and their lower  and upper  95% confidence intervals.
</p>
<table role = "presentation">
<tr><td><code>AT</code></td>
<td>
<p>Variable A and T with binary system.</p>
</td></tr>
<tr><td><code>CG</code></td>
<td>
<p>Variable C and G with binary system. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nora M. Villanueva and Marta Sestelo.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(seq2R)

#mtDNAhum &lt;- read.genbank("NC_012920")
data(mtDNAhum)
DNA &lt;- transform(mtDNAhum)
DNA
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
