<!DOCTYPE html><html><head><title>Help for package SeqFeatR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SeqFeatR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#assocpair'><p>Finding pairs of alignment positions that jointly mutate</p></a></li>
<li><a href='#assocpairfeat'><p>Finding pairs of sequence alignment positions associated with sequence feature.</p></a></li>
<li><a href='#assocpoint'><p>Searches for associations of single alignment positions with feature(s)</p></a></li>
<li><a href='#assocpointhierarchical'><p>Searches for associations of single alignment positions with feature(s) with a simple hierarchical model</p></a></li>
<li><a href='#assocpointpair'><p>Compare results of SeqFeatRs assocpoint with SeqFeatRs assocpair,</p>
assocpairfeat</a></li>
<li><a href='#assoctuple'><p>Searches for associations of a tuple of alignment positions with feature(s)</p></a></li>
<li><a href='#comparewithancestral'><p>Compares ancestral sequence with sequence alignment</p></a></li>
<li><a href='#foundereffectfinder'><p>Tries to identify a founder effect in results from SeqFeatRs assocpair</p></a></li>
<li><a href='#getfreqs'><p>Get frequencies for amino acids/nucleotides in alignment window (epitope)</p></a></li>
<li><a href='#orPlot'><p>Visualize odds ratios and p-values from SeqFeatRs assocpoint</p></a></li>
<li><a href='#q-values'><p>Calculate q-values from p-values</p></a></li>
<li><a href='#rewritetuple'><p>Rewrite result from SeqFeatRs assoctuple</p></a></li>
<li><a href='#SeqFeatRGUI'><p>GUI for SeqFeatR</p></a></li>
<li><a href='#smallmanhattan'><p>A small version of manhattan plot</p></a></li>
<li><a href='#tartan'><p>Visualize association of mutation pairs and features from two (comparable) analyses.</p></a></li>
<li><a href='#visualizepair'><p>Visualize pairs of alignment positions that jointly mutate</p></a></li>
<li><a href='#visualizepairfeat'><p>Visualize pairs of sequence alignment positions associated with sequence feature.</p></a></li>
<li><a href='#volcanoplot'><p>Create volcano plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Tool to Associate FASTA Sequences and Features</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-10-04</td>
</tr>
<tr>
<td>Author:</td>
<td>Bettina Budeus</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bettina Budeus &lt;bettina.budeus@stud.uni-due.de&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.2.2),tcltk,tcltk2,Biostrings, plotrix</td>
</tr>
<tr>
<td>Imports:</td>
<td>plyr, phangorn, qvalue, widgetTools, calibrate, ggplot2,
R2jags, coda, scales, ape</td>
</tr>
<tr>
<td>Suggests:</td>
<td>xtable</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides user friendly methods for the identification of sequence patterns that are statistically significantly associated with a property of the sequence. For instance, SeqFeatR allows to identify viral immune escape mutations for hosts of given HLA types. The underlying statistical method is Fisher's exact test, with appropriate corrections for multiple testing, or Bayes. Patterns may be point mutations or n-tuple of mutations. SeqFeatR offers several ways to visualize the results of the statistical analyses, see Budeus (2016) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0146409">doi:10.1371/journal.pone.0146409</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-04-12 07:20:10 UTC; bettina</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-04-12 12:02:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='assocpair'>Finding pairs of alignment positions that jointly mutate</h2><span id='topic+assocpair'></span>

<h3>Description</h3>

<p>Determines pairs of sequence alignment positions that
mutate in a correlated fashion with respect to a consensus sequence.</p>


<h3>Usage</h3>

<pre><code class='language-R'>assocpair(path_to_file_sequence_alignment = NULL,
    path_to_file_consensus = NULL, save_name_csv, dna = FALSE,
    significance_level = 0.05, multiple_testing_correction = "bonferroni")</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assocpair_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>FASTA file with sequence alignment. See example file.</p>
</td></tr>
<tr><td><code id="assocpair_+3A_path_to_file_consensus">path_to_file_consensus</code></td>
<td>
<p>FASTA file with consensus sequence. See example file.</p>
</td></tr>
<tr><td><code id="assocpair_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are written in csv format.</p>
</td></tr>
<tr><td><code id="assocpair_+3A_dna">dna</code></td>
<td>
<p>indicates whether sequences are DNA or amino acids.</p>
</td></tr>
<tr><td><code id="assocpair_+3A_significance_level">significance_level</code></td>
<td>
<p>significance level for Fisher's exact test.</p>
</td></tr>
<tr><td><code id="assocpair_+3A_multiple_testing_correction">multiple_testing_correction</code></td>
<td>
<p>multiple testing correction applied to p-values. Input can be: &quot;holm&quot;,<br />
&quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For every position in the sequence alignment from the FASTA
file a Fisher's exact test is applied with every other position in the
sequence to check whether at both positions we have correlated
mutations with respect to a given consensus sequence. Significant p-values are
collected in one big table. p.adjust from stats package is used for multiple testing
correction; corrected values are given as extra column in csv output.
</p>
<p>In contrast to assocpairfeat, assocpair does not use features, but uses a consensus
approach. Please be sure, that this is really what you want to use. Otherwise, use
assocpairfeat or assoctuple instead.
</p>


<h3>Value</h3>

<p>A csv file with every possible co-mutation below the given p-value.</p>


<h3>Note</h3>

<p>For graphical output use:<br />
<code><a href="#topic+visualizepair">visualizepair</a></code>.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+visualizepairfeat">visualizepairfeat</a></code>, <code><a href="#topic+assocpairfeat">assocpairfeat</a></code>, <code><a href="#topic+assoctuple">assoctuple</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
## Not run: 
fasta_input &lt;- system.file("extdata", "Example_aa.fasta", package="SeqFeatR")
consensus_input &lt;- system.file("extdata", "Example_Consensus_aa.fasta", package="SeqFeatR")
	
#Usage
assocpair(
	path_to_file_sequence_alignment=fasta_input,
	path_to_file_consensus=consensus_input,
	save_name_csv="assocpair_results.csv",
	dna=FALSE,
	significance_level=0.05,
	multiple_testing_correction="bonferroni")

## End(Not run)
</code></pre>

<hr>
<h2 id='assocpairfeat'>Finding pairs of sequence alignment positions associated with sequence feature.</h2><span id='topic+assocpairfeat'></span>

<h3>Description</h3>

<p>Determines pairs of alignment positions that jointly mutate 
depending on sequence feature.</p>


<h3>Usage</h3>

<pre><code class='language-R'>assocpairfeat(path_to_file_sequence_alignment = NULL, save_name_csv, dna = FALSE, 
    patnum_threshold = 1, significance_level = 0.05, 
    A11a, A12a, A21a, A22a, B11a, B12a, B21a, B22a,
    multiple_testing_correction = "bonferroni")</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assocpairfeat_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>FASTA file with sequence alignment. See example file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are written in csv format.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_dna">dna</code></td>
<td>
<p>DNA or amino acid sequences.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_patnum_threshold">patnum_threshold</code></td>
<td>
<p>minimum number of patients per HLA type to consider in calculation.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_significance_level">significance_level</code></td>
<td>
<p>significance level in Fisher's exact test.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_a11a">A11a</code></td>
<td>
<p>position of start of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_a12a">A12a</code></td>
<td>
<p>position of end of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_a21a">A21a</code></td>
<td>
<p>position of start of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_a22a">A22a</code></td>
<td>
<p>position of end of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_b11a">B11a</code></td>
<td>
<p>position of start of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_b12a">B12a</code></td>
<td>
<p>position of end of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_b21a">B21a</code></td>
<td>
<p>position of start of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_b22a">B22a</code></td>
<td>
<p>position of end of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpairfeat_+3A_multiple_testing_correction">multiple_testing_correction</code></td>
<td>
<p>multiple testing correction applied to p-values. Input can be: &quot;holm&quot;,<br />
&quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Features may be HLA types, indicated by four blocks in the FASTA
comment lines. The positions of these blocks in the comment lines are
defined by parameters A11, ..., B22. For patients with a homozygous
HLA allele the second allele has to be &quot;00&quot; (without the double
quotes). 
</p>
<p>For every position in the sequence alignment from the FASTA
file a Fisher's exact test is applied with every other position in the
sequence and every HLA-type. Significant p-values are collected in one
big table. p.adjust from stats package is used for multiple testing
correction; corrected values are given as extra column in csv output.
</p>
<p>In contrast to assocpair, assocpairfeat uses an analysis with 'features'
without a consensus sequence.
</p>


<h3>Value</h3>

<p>Table with all alignment position pairs with
significant association with sequence feature.</p>


<h3>Note</h3>

<p>Use <code><a href="#topic+visualizepairfeat">visualizepairfeat</a></code> for graphical output.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+visualizepairfeat">visualizepairfeat</a></code>, <code><a href="#topic+assocpair">assocpair</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input file
## Not run: 
fasta_input &lt;- system.file("extdata", "Example_aa.fasta", package="SeqFeatR")

#Usage
assocpairfeat(
	path_to_file_sequence_alignment=fasta_input,
	save_name_csv="assocpairfeat_results.csv",
	dna=FALSE,
	patnum_threshold=1,
	significance_level=0.05,
	A11=10,
	A12=11,
	A21=13,
	A22=14,
	B11=17,
	B12=18,
	B21=20,
	B22=21,
	multiple_testing_correction="bonferroni")

## End(Not run)
</code></pre>

<hr>
<h2 id='assocpoint'>Searches for associations of single alignment positions with feature(s)</h2><span id='topic+assocpoint'></span>

<h3>Description</h3>

<p>Searches for associations of single nucleotide or amino
acid sequence alignment positions with feature(s).</p>


<h3>Usage</h3>

<pre><code class='language-R'>assocpoint(path_to_file_sequence_alignment = NULL,
    path_to_file_known_epitopes = NULL, path_to_file_binding_motifs = NULL, save_name_pdf,
    save_name_csv, dna = FALSE, 
    patnum_threshold = 1, optical_significance_level = 0.05, 
    star_significance_level = 0.001, A11, A12, A21, A22, B11, B12, B21, 
    B22, C11, C12, C21, C22, has_C=FALSE, multiple_testing_correction = "bonferroni",
    bayes_factor = FALSE, constant_dirichlet_precision_parameter=FALSE,
    dirichlet_precision_parameter=20, phylo_bias_check = FALSE,
    path_to_file_reference_sequence=NULL, one_feature=FALSE,
    window_size=9, epi_plot=FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assocpoint_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>FASTA file with sequence alignment. For reference
see example file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_path_to_file_known_epitopes">path_to_file_known_epitopes</code></td>
<td>
<p>csv file with known epitopes. See example file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_path_to_file_binding_motifs">path_to_file_binding_motifs</code></td>
<td>
<p>csv file with HLA binding motifs if
available. See example file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_save_name_pdf">save_name_pdf</code></td>
<td>
<p>name of file to which results are saved in pdf format.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are saved in csv format.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_dna">dna</code></td>
<td>
<p>DNA or amino acid sequences.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_patnum_threshold">patnum_threshold</code></td>
<td>
<p>minimum number of patients per HLA type to consider in calculation.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_optical_significance_level">optical_significance_level</code></td>
<td>
<p>height of horizontal line in
graphical output indicating significance level, e.g.\ 0.05.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_star_significance_level">star_significance_level</code></td>
<td>
<p>height of invisible horizontal line
above which all points are marked as stars, e.g.\ 0.001 as level for high
significance.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_a11">A11</code></td>
<td>
<p>position of start of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_a12">A12</code></td>
<td>
<p>position of end of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_a21">A21</code></td>
<td>
<p>position of start of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_a22">A22</code></td>
<td>
<p>position of end of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_b11">B11</code></td>
<td>
<p>position of start of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_b12">B12</code></td>
<td>
<p>position of end of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_b21">B21</code></td>
<td>
<p>position of start of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_b22">B22</code></td>
<td>
<p>position of end of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_c11">C11</code></td>
<td>
<p>position of start of first HLA C allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_c12">C12</code></td>
<td>
<p>position of end of first HLA C allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_c21">C21</code></td>
<td>
<p>position of start of second HLA C allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_c22">C22</code></td>
<td>
<p>position of end of second HLA C allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_has_c">has_C</code></td>
<td>
<p>set to TRUE if there is a C allel information in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_multiple_testing_correction">multiple_testing_correction</code></td>
<td>
<p>multiple testing correction applied to p-values. Input can be: &quot;holm&quot;,<br />
&quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_bayes_factor">bayes_factor</code></td>
<td>
<p>if bayes factor should be applied instead of Fisher's exact test for association. See details.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_constant_dirichlet_precision_parameter">constant_dirichlet_precision_parameter</code></td>
<td>
<p>if Dirichlet precision parameter K is a fixed value. See details.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_dirichlet_precision_parameter">dirichlet_precision_parameter</code></td>
<td>
<p>the Dirichlet precision parameter for evaluation with Bayes factor. See details.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_phylo_bias_check">phylo_bias_check</code></td>
<td>
<p>simple mechanism to detect phylogenetic bias in data. See details.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_path_to_file_reference_sequence">path_to_file_reference_sequence</code></td>
<td>
<p>Reference sequence for orientation. Will be
added as extra column in the results. Not used in calculation itself.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_one_feature">one_feature</code></td>
<td>
<p>if there is only one feature. See details.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_window_size">window_size</code></td>
<td>
<p>size of sliding window in graphical output.</p>
</td></tr>
<tr><td><code id="assocpoint_+3A_epi_plot">epi_plot</code></td>
<td>
<p>add epitope plot to results pdf file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each sequence alignment position (each column of sequence
alignment), Fisher's exact test is evaluated for a 2-by-2 contingency
table of amino acid (or nucleic acid) vs. feature. The resulting
p-values are returned in a table. p-values can be corrected for
multiple testing using various methods (option
multiple_testing_correction).
</p>
<p>If bayes_factor is TRUE, a simple calculation of a bayes_factor is applied instead of the
Fisher's exact test. An independent model is compared with a nearly
independent model (see. Mixtures of Dirichlet Distributions and Estimation in Contingency Tables).
The user can choose if a fixed dirichlet precision parameter or a calculated one should be used.
The calculated dirichlet precision parameter is dependend from the number of entries in
the 2-by-2 contingency table.
For more information see the mentioned paper and &quot;Bayesian Computation with R&quot;. With Bayes Factor
a usage of multiple testing is not possible (and in most of all cases not needed). 
If constant_dirichlet_precision_parameter is FALSE, the Dirichlet precision parameter K
for the calculation of the Bayes Factors is calculated by the system.
</p>
<p>The input sequence alignment may be consist either of DNA sequences
(switch dna = TRUE) or amino acid sequences (dna =
FALSE). Undetermined nucleotides or amino acids have to be indicated
by the letter &quot;X&quot;.
</p>
<p>A phylogenetic bias check can be made if bayes_factor = FALSE and the result added as an extra column
to the csv result file. For each alignment position with a p-value below a
user given threshold (optical_significance_level), the mean distance of all
sequences with the mutation, which gave rise to this p-value, is calculated
and compared with the mean distance of all sequences.
</p>
<p>Features may be either a single feature, or HLA types, the latter indicated by four blocks in the FASTA
comment lines. The positions of these blocks in the comment lines are
defined by parameters A11, ..., B22 (..C22 if there is information about the C allel).
For patients with a homozygous HLA allele the second allele has to be &quot;00&quot; (without the double
quotes). For single features (no HLA types), set option one_feature=TRUE. The
value of the feature (e.g. 'yes / no', or '1 / 2 / 3') should then be
given at the end of each FASTA comment, separated from the part before
that by a semicolon.
</p>


<h3>Value</h3>

<p>The function generates various types of output: a table of
p-values, p-values corrected for multiple testing, z-scores, amino acid with
the lowest p-value at this position, result of the phylogenetic bias test,
and several plots. A Manhattan plot is generated
for each feature in a pdf file with alignment positions on the x-axis and
p-values on the y-axis.
Two different significance levels can be indicated by a line
(optical_significance_level) and the change from the normal dot symbol to
a star (star_significance_level).
</p>
<p>In case of bayes_factor = TRUE, the Bayes Factor and not a p-value and
the estimate of the simulation standard error of the computed value of the
Bayes factor is given.
</p>
<p>Optionally (epi_plot = TRUE), a second plot is provided with the number
of 'significant' p-values or Bayes Factors (value of significance chosen by
optical_significance_level) in a sliding window of x amino acids
( x can be any number from 1 up to the length of the alignment chosen
by 'window_size'. The typical length of an MHC I binding motif is nine).
Each point in this plot is the number of 'significant' p-values or Bayes Factors
in the next x alignment positions. Additionally, with given HLA motifs
(path_to_file_binding_motifs), a second line is added.
This line shows potential epitopes. A potential epitope is assigned if 
a 'significant' p-value or Bayes Factor
(value of significance chosen by optical_significance_level) is inside one
of the motifs for this HLA type.
</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>References</h3>

<p>Albert, James H.; Gupta, Arjun K.
Mixtures of Dirichlet Distributions and Estimation in Contingency Tables. 
Ann. Statist. 10 (1982), no. 4, 1261&ndash;1268. 
doi:10.1214/aos/1176345991. 
http://projecteuclid.org/euclid.aos/1176345991.
</p>
<p>Albert, J.
Bayesian Computation with R.
Springer; 2nd ed. 2009 edition (May 15, 2009)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpointhierarchical">assocpointhierarchical</a></code>, <code><a href="#topic+assocpointpair">assocpointpair</a></code>, <code><a href="#topic+orPlot">orPlot</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
fasta_input &lt;- system.file("extdata", "Example_aa.fasta", package="SeqFeatR")
epitopes_input &lt;- system.file("extdata", "Example_epitopes_aa.csv", package="SeqFeatR")
motifs_input &lt;- system.file("extdata", "Example_HLA_binding_motifs_aa.csv", package="SeqFeatR")
reference_input &lt;- system.file("extdata", "Example_reference_aa.fasta", package="SeqFeatR")

#Usage
## Not run: 
assocpoint(
	path_to_file_sequence_alignment=fasta_input,
	path_to_file_known_epitopes=epitopes_input,
	path_to_file_binding_motifs=motifs_input,
	save_name_pdf="assocpoint_results.pdf",
	save_name_csv="assocpoint_results.csv",
	dna=FALSE,
	patnum_threshold=1,
	optical_significance_level=0.01, 
	star_significance_level=0.5,
	A11=10,
	A12=11,
	A21=13,
	A22=14,
	B11=17,
	B12=18,
	B21=20,
	B22=21,
	C11=24,
	C12=25,
	C21=27,
	C22=28,
	has_C=FALSE,
	multiple_testing_correction="bonferroni",
	bayes_factor=FALSE,
	constant_dirichlet_precision_parameter=TRUE,
	dirichlet_precision_parameter=20,
	phylo_bias_check=FALSE,
	path_to_file_reference_sequence=reference_input,
	one_feature=FALSE,
	window_size=9,
	epi_plot=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='assocpointhierarchical'>Searches for associations of single alignment positions with feature(s) with a simple hierarchical model</h2><span id='topic+assocpointhierarchical'></span>

<h3>Description</h3>

<p>Searches for associations of single nucleotide or amino
acid sequence alignment positions with feature(s).</p>


<h3>Usage</h3>

<pre><code class='language-R'>assocpointhierarchical(path_to_file_sequence_alignment = NULL,
    path_to_file_known_epitopes = NULL, path_to_file_binding_motifs = NULL, save_name_pdf,
    save_name_csv, dna = FALSE, 
    patnum_threshold = 1, optical_significance_level = 0.05, 
    star_significance_level = 0.001, A11, A12, A21, A22, B11, B12, B21, 
    B22, path_to_file_reference_sequence=NULL, one_feature=FALSE,
    window_size=9, epi_plot=FALSE, own_model, number_of_simulations,
    number_of_burnin, number_of_chains, response_inits,	further_response_data,
    response_parameters)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assocpointhierarchical_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>FASTA file with sequence alignment. For reference
see example file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_path_to_file_known_epitopes">path_to_file_known_epitopes</code></td>
<td>
<p>csv file with known epitopes. See example file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_path_to_file_binding_motifs">path_to_file_binding_motifs</code></td>
<td>
<p>csv file with HLA binding motifs if
available. See example file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_save_name_pdf">save_name_pdf</code></td>
<td>
<p>name of file to which results are saved in pdf format.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are saved in csv format.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_dna">dna</code></td>
<td>
<p>DNA or amino acid sequences.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_patnum_threshold">patnum_threshold</code></td>
<td>
<p>minimum number of patients per HLA type to consider in calculation.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_optical_significance_level">optical_significance_level</code></td>
<td>
<p>height of horizontal line in
graphical output indicating significance level, e.g.\ 0.05.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_star_significance_level">star_significance_level</code></td>
<td>
<p>height of invisible horizontal line
above which all points are marked as stars, e.g.\ 0.001 as level for high
significance.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_a11">A11</code></td>
<td>
<p>position of start of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_a12">A12</code></td>
<td>
<p>position of end of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_a21">A21</code></td>
<td>
<p>position of start of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_a22">A22</code></td>
<td>
<p>position of end of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_b11">B11</code></td>
<td>
<p>position of start of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_b12">B12</code></td>
<td>
<p>position of end of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_b21">B21</code></td>
<td>
<p>position of start of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_b22">B22</code></td>
<td>
<p>position of end of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_path_to_file_reference_sequence">path_to_file_reference_sequence</code></td>
<td>
<p>Reference sequence for orientation. Will be
added as extra column in the results. Not used in calculation itself.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_one_feature">one_feature</code></td>
<td>
<p>if there is only one feature. See details.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_window_size">window_size</code></td>
<td>
<p>size of sliding window in graphical output.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_epi_plot">epi_plot</code></td>
<td>
<p>add epitope plot to results pdf file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_own_model">own_model</code></td>
<td>
<p>txt file with jags model - optional. See example file.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_number_of_simulations">number_of_simulations</code></td>
<td>
<p>number of total iterations per chain (including burn in) - see R2jags package.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_number_of_burnin">number_of_burnin</code></td>
<td>
<p>length of burn in, i.e. number of iterations to discard at the beginning - see R2jags package.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_number_of_chains">number_of_chains</code></td>
<td>
<p>number of Markov chains - see R2jags package.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_response_inits">response_inits</code></td>
<td>
<p>specification of initial value - see details and R2jags package.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_further_response_data">further_response_data</code></td>
<td>
<p>additional response data - see details and rjags/R2jags package.</p>
</td></tr>
<tr><td><code id="assocpointhierarchical_+3A_response_parameters">response_parameters</code></td>
<td>
<p>name of response parameters - see details and rjags/R2jags package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given alignment a simple hierarchical model (amino acids/ nucleotides at each position
are multi-nominally distributed, features are gamma(0.1,0.1) distributed - see example file).
</p>
<p>The input sequence alignment may be consist either of DNA sequences
(switch dna = TRUE) or amino acid sequences (dna =
FALSE). Undetermined nucleotides or amino acids have to be indicated
by the letter &quot;X&quot;.
</p>
<p>Features may be either a single feature, or HLA types, the latter indicated by four blocks in the FASTA
comment lines. The positions of these blocks in the comment lines are
defined by parameters A11, ..., B22. For patients with a homozygous
HLA allele the second allele has to be &quot;00&quot; (without the double
quotes). For single features (no HLA types), set option one_feature=TRUE. The
value of the feature (e.g. 'yes / no', or '1 / 2 / 3') should then be
given at the end of each FASTA comment, separated from the part before
that by a semicolon.
</p>
<p>However a user can add his own model and parameter (only for EXPERTS!):
The jags model file has to be given as 'own_model'. This 'activates' the following parameters:
1. Further, the user can add other response data ('further_response_data') besides the internally already given
matrix with all amino acids/nucleotides at every position and every feature and the colSums of this matrix.
2. The user has to insert initial values ('response_inits') and 
3. response parameters ('response_parameters').
For the definition of initial values the following variables can be used:
- all_of_letters = all amino acids/nucleotides plus gap symbol and unknown (X)
- min_FASTA_length = length of alignment
- allels = all HLA-allels/features
A response parameter named theta will be evaluated by the part of the program after the model,
therefore the user should keep in mind to name this in his own model:
The result theta has to be of the following form: 3 dimensional, with first dimension: feature vs no feature,
second dimension: amino acid/nucleotide, third dimension: position (see examples file).
</p>


<h3>Value</h3>

<p>The function generates various types of output: a table of
probabilities, z-scores, amino acid/nucleotide with the highest probability at this position and several plots.
A Manhattan plot is generated for each feature in a pdf file with alignment positions
on the x-axis and probabilities on the y-axis.
Two different significance levels can be indicated by a line
(optical_significance_level) and the change from the normal dot symbol to
a star (star_significance_level).
</p>
<p>Optionally (epi_plot = TRUE), a second plot is provided with the number
of 'significant' p-values (value of significance chosen by optical_significance_level)
in a sliding window of x amino acids ( x can be any number from 1 up to the
length of the alignment chosen by 'window_size'. The typical length of an MHC I binding motif is nine).
Each point in this plot is the number of 'significant' probabilities in the next x
alignment positions. Additionally, with given HLA motifs
(path_to_file_binding_motifs), a second line is added.
This line shows potential epitopes. A potential epitope is assigned if 
a 'significant' probabilities (value of significance chosen by optical_significance_level)
is inside one of the motifs for this HLA type. 
</p>


<h3>Note</h3>

<p>To uses this function, it is essential to install Jags. This seems to be easy for Windows
and Linux systems, but complicated for MacOS. Therefore this function is not tested in
installation process.
</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>References</h3>

<p>Albert, J.
Bayesian Computation with R.
Springer; 2nd ed. 2009 edition (May 15, 2009)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpoint">assocpoint</a></code>, <code><a href="#topic+assocpointpair">assocpointpair</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
fasta_input &lt;- system.file("extdata", "Example_aa.fasta", package="SeqFeatR")
epitopes_input &lt;- system.file("extdata", "Example_epitopes_aa.csv", package="SeqFeatR")
motifs_input &lt;- system.file("extdata", "Example_HLA_binding_motifs_aa.csv", package="SeqFeatR")
own_model_input &lt;- system.file("extdata", "Example_model_for_bayes.txt", package="SeqFeatR")
reference_input &lt;- system.file("extdata", "Example_reference_aa.fasta", package="SeqFeatR")

#Usage
## Not run: 
assocpointhierarchical(
	path_to_file_sequence_alignment=fasta_input,
	path_to_file_known_epitopes=epitopes_input,
	path_to_file_binding_motifs=motifs_input,
	save_name_pdf = "assocpointhierarchical_results.pdf",
	save_name_csv = "assocpointhierarchical_results.csv",
	dna = FALSE,
	patnum_threshold = 1,
	optical_significance_level = 97, 
	star_significance_level = 99,
	A11 = 10,
	A12 = 11,
	A21 = 13,
	A22 = 14,
	B11 = 17,
	B12 = 18,
	B21 = 20,
	B22 = 21,
	path_to_file_reference_sequence = reference_input,
	one_feature=FALSE,
	window_size=9,
	epi_plot=TRUE,
	own_model=own_model_input,
	number_of_simulations=20,
	number_of_burnin=floor(20/2),
	number_of_chains=2,
	response_inits="'vcounts' = rep(1, length(all_of_letters)), 
       'theta' = array(1/length(all_of_letters),
        dim=c(2, length(all_of_letters), min_FASTA_length))",
	further_response_data=c(),
	response_parameters=c("theta", "vcounts"))

## End(Not run)
</code></pre>

<hr>
<h2 id='assocpointpair'>Compare results of SeqFeatRs assocpoint with SeqFeatRs assocpair,
assocpairfeat</h2><span id='topic+assocpointpair'></span>

<h3>Description</h3>

<p>Calculates which significant positions in result of assocpoint
are also in results of assocpair/assocpairfeat.</p>


<h3>Usage</h3>

<pre><code class='language-R'>assocpointpair(path_to_file_sequence_alignment = NULL, 
    path_to_file_assocpoint_csv_result = NULL, 
    path_to_file_assocpairfeat_csv_result = NULL,
    significance_level = 0.05, save_name_csv, save_name_pos)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assocpointpair_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>FASTA file with sequence alignment. For reference
see example file.</p>
</td></tr>
<tr><td><code id="assocpointpair_+3A_path_to_file_assocpoint_csv_result">path_to_file_assocpoint_csv_result</code></td>
<td>
<p>file with result from SeqFeatRs assocpoint.
For reference see example file.</p>
</td></tr>
<tr><td><code id="assocpointpair_+3A_path_to_file_assocpairfeat_csv_result">path_to_file_assocpairfeat_csv_result</code></td>
<td>
<p>file with results from SeqFeatRs assocpair,
assocpairfeat. For reference see example file.</p>
</td></tr>
<tr><td><code id="assocpointpair_+3A_significance_level">significance_level</code></td>
<td>
<p>p-value to be defined as significant.</p>
</td></tr>
<tr><td><code id="assocpointpair_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are saved in
csv format.</p>
</td></tr>
<tr><td><code id="assocpointpair_+3A_save_name_pos">save_name_pos</code></td>
<td>
<p>name of file to which results are saved in csv format
(contains possible compensatory mutations).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes the results from assocpoint and assocpair/assocpairfeat and
tries to combine them into one result: we are looking for pairs of
sequence alignment positions where (a) each of the positions is significantly associated
with the feature (for significance level see parameter
significance_level), and (b) the sequence states of both positions are
associated. Pairs of positions fulfilling these two criteria may e.g.
carry compensatory mutations.
If at least one such mutation is found, a detailed output is additionally
created with possible compensatory mutations</p>


<h3>Value</h3>

<p>A csv file with p-value for alignment position pairs and if existing
a csv file with possible compensatory mutations.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpoint">assocpoint</a></code>
</p>
<p><code><a href="#topic+assocpairfeat">assocpairfeat</a></code>
</p>
<p><code><a href="#topic+assocpair">assocpair</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
fasta_input &lt;- system.file("extdata", "Example_aa.fasta", package="SeqFeatR")
assocpoint_result &lt;- system.file("extdata", "assocpoint_results.csv", package="SeqFeatR")
assocpairfeat_result &lt;- system.file("extdata", "assocpairfeat_results.csv", package="SeqFeatR")

#Usage
assocpointpair(
	path_to_file_sequence_alignment=fasta_input, 
	path_to_file_assocpoint_csv_result=assocpoint_result, 
	path_to_file_assocpairfeat_csv_result=assocpairfeat_result, 
	significance_level=0.05,
	save_name_csv="assocpointpair_result.csv",
	save_name_pos="possible_compensatory_mutation.csv")
</code></pre>

<hr>
<h2 id='assoctuple'>Searches for associations of a tuple of alignment positions with feature(s)</h2><span id='topic+assoctuple'></span>

<h3>Description</h3>

<p>Searches for associations of tuple of nucleotide or amino
acid sequence alignment positions with feature(s).</p>


<h3>Usage</h3>

<pre><code class='language-R'>assoctuple(path_to_file_sequence_alignment, path_to_file_assocpoint_csv_result, threshold,
	min_number_of_elements_in_tuple, max_number_of_elements_in_tuple,
	save_name_csv, column_of_feature, column_of_position, column_of_p_values, 
        column_of_aa, A11, A12, A21, A22, B11, B12, B21, B22, one_feature, feature)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assoctuple_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>FASTA file with sequence alignment. For reference
see example file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_path_to_file_assocpoint_csv_result">path_to_file_assocpoint_csv_result</code></td>
<td>
<p>results from SeqFeatRs assocpoint</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_threshold">threshold</code></td>
<td>
<p>p-value threshold for sequence alignment positions to
be considered.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_min_number_of_elements_in_tuple">min_number_of_elements_in_tuple</code></td>
<td>
<p>minimal number of members in tuple.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_max_number_of_elements_in_tuple">max_number_of_elements_in_tuple</code></td>
<td>
<p>maximal number of members in tuple.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are saved in csv format.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_column_of_feature">column_of_feature</code></td>
<td>
<p>column number in which feature is located for which analysis should be done.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_column_of_position">column_of_position</code></td>
<td>
<p>column number in which sequence position is located.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_column_of_p_values">column_of_p_values</code></td>
<td>
<p>column number from which p-values should be taken. See details.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_column_of_aa">column_of_aa</code></td>
<td>
<p>column number from which amino acids should be taken. See details.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_a11">A11</code></td>
<td>
<p>position of start of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_a12">A12</code></td>
<td>
<p>position of end of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_a21">A21</code></td>
<td>
<p>position of start of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_a22">A22</code></td>
<td>
<p>position of end of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_b11">B11</code></td>
<td>
<p>position of start of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_b12">B12</code></td>
<td>
<p>position of end of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_b21">B21</code></td>
<td>
<p>position of start of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_b22">B22</code></td>
<td>
<p>position of end of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_one_feature">one_feature</code></td>
<td>
<p>if there is only one feature.</p>
</td></tr>
<tr><td><code id="assoctuple_+3A_feature">feature</code></td>
<td>
<p>feature identifier which should be analyzed. See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each tuple of sequence alignment positions, Fisher's exact test is
evaluated for a 2-by-2 contingency table of amino acid tuple
(or nucleic acid tuple) vs. feature. The resulting p-values are returned in a table.
</p>
<p>For this to work properly the result from SeqFeatRs assocpoint can be used, 
but also a user generated csv file in which at least one column describes the
sequence position and another the p-value at this position, as well as a
FASTA file from which those file originated.
</p>
<p>assoctuple takes only those sequence positions which have a p-value lower than
the given threshold ('threshold'). Be aware that for big datasets the 
calculation time can be high and a calculation of every position with
every other position will most definitely result in low quality data.
</p>
<p>Please be also aware that it just uses the position in your csv file.
If this is NOT the correct position, because of removal of empty or near empty
alignment positions, correct the csv file before starting. 
</p>
<p>Use the same FASTA file you had used for assocpoint!
</p>
<p>The sequence positions to be included in this analysis are normally chosen
from the corrected p-values (but can be anything else as long as it
is between 0 and 1, even an own added column). 
The size of the tuples can be anything from 2 to number of rows (= number
of alignment positions) in the csv input file.
The input sequence alignment may be consist either of DNA sequences
(switch dna = TRUE) or amino acid sequences (dna =
FALSE). Undetermined nucleotides or amino acids have to be indicated
by the letter &quot;X&quot;. 
</p>
<p>Features may be HLA types, indicated by four blocks in the FASTA
comment lines. The positions of these blocks in the comment lines are
defined by parameters A11, ..., B22. For patients with a homozygous
HLA allele the second allele has to be &quot;00&quot; (without the double
quotes). For non-HLA-type features, set option one_feature=TRUE. The
value of the feature (e.g. 'yes / no', or '1 / 2 / 3') should then be
given at the end of each FASTA comment, separated from the part before
that by a semicolon. 
</p>
<p>The analysis is done only for one single feature. This is chosen by either
'feature' if there is only 'one_feature', or column_of_feature if
there are HLA types.
</p>


<h3>Value</h3>

<p>A csv list of tuple positions and the p-value of there association.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpoint">assocpoint</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
## Not run: 
fasta_input &lt;- system.file("extdata", "Example_aa.fasta", package="SeqFeatR")
assocpoint_result &lt;- system.file("extdata", "assocpoint_results.csv", package="SeqFeatR")

#Usage
assoctuple(
	path_to_file_sequence_alignment=fasta_input,
	path_to_file_assocpoint_csv_result=assocpoint_result,
	threshold=0.2,
	min_number_of_elements_in_tuple=2,
	max_number_of_elements_in_tuple=2,
	save_name_csv="assoctuple_result.csv",
	column_of_feature=9,
	column_of_position=1,
	column_of_p_values=9,
	column_of_aa=12,
	A11=10,
	A12=11,
	A21=13,
	A22=14,
	B11=17,
	B12=18,
	B21=20,
	B22=21,
	one_feature=FALSE,
	feature="")

## End(Not run)
</code></pre>

<hr>
<h2 id='comparewithancestral'>Compares ancestral sequence with sequence alignment</h2><span id='topic+comparewithancestral'></span>

<h3>Description</h3>

<p>Generates from a nexus tree file the possible ancestral sequence and compares this with the sequence alignment.</p>


<h3>Usage</h3>

<pre><code class='language-R'>comparewithancestral(path_to_file_sequence_alignment = NULL,
    path_to_file_nexus_tree = NULL, save_name_csv, 
    patnum_threshold = 1, significance_level = 0.05,
    A11, A12, A21, A22, B11, B12, B21, B22)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comparewithancestral_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>FASTA file with sequence alignment. For reference
see example file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_path_to_file_nexus_tree">path_to_file_nexus_tree</code></td>
<td>
<p>nexus tree file of sequences alignment. For reference see example file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are saved in csv format.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_patnum_threshold">patnum_threshold</code></td>
<td>
<p>minimum number of patients per HLA type to consider in calculation.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_significance_level">significance_level</code></td>
<td>
<p>significance level in Fisher's exact test.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_a11">A11</code></td>
<td>
<p>position of start of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_a12">A12</code></td>
<td>
<p>position of end of first HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_a21">A21</code></td>
<td>
<p>position of start of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_a22">A22</code></td>
<td>
<p>position of end of second HLA A allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_b11">B11</code></td>
<td>
<p>position of start of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_b12">B12</code></td>
<td>
<p>position of end of first HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_b21">B21</code></td>
<td>
<p>position of start of second HLA B allele in header line of FASTA file.</p>
</td></tr>
<tr><td><code id="comparewithancestral_+3A_b22">B22</code></td>
<td>
<p>position of end of second HLA B allele in header line of FASTA file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Features may be HLA types, indicated by four blocks in the FASTA
comment lines. The positions of these blocks in the comment lines are
defined by parameters A11, ..., B22. For patients with a homozygous
HLA allele the second allele has to be &quot;00&quot; (without the double
quotes). 
</p>
<p>Uses the R-packages ape and phangorn to create from a given nexus tree
file the ancestral sequence. For every sequence position the ancestral amino acid
is compared with the amino acid of every HLA type and with a Fisher's exact test
the significant differences are calculated.
</p>


<h3>Value</h3>

<p>Generates a csv file with p-values and q-values (generated by the
R-package qvalue) for every HLA type and sequence position with a p-value
below given threshold (significance_level).
</p>


<h3>Note</h3>

<p>BEWARE! You have to name the FASTA sequences (= comment line in the file)
so that the R-packages ape and phangorn can read and write them correctly. 
It will replace any space character with &quot;_&quot; (without the double quotes)
and ignore &quot;:&quot; (without the double quotes)!
</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
## Not run: 
fasta_input &lt;- system.file("extdata", "Example_aa_for_cwa.fasta", package="SeqFeatR")
tree_input &lt;- system.file("extdata", "Example_tree.nh", package="SeqFeatR")

#Usage
comparewithancestral(
	path_to_file_sequence_alignment=fasta_input,
	path_to_file_nexus_tree=tree_input,
	save_name_csv="hlaTree_results.csv",
	patnum_threshold=1,
	significance_level=0.05,
	A11=10,
	A12=11,
	A21=13,
	A22=14,
	B11=17,
	B12=18,
	B21=20,
	B22=21)

## End(Not run)
</code></pre>

<hr>
<h2 id='foundereffectfinder'>Tries to identify a founder effect in results from SeqFeatRs assocpair</h2><span id='topic+foundereffectfinder'></span>

<h3>Description</h3>

<p>Takes the result of co-mutation analysis like SeqFeatRs assocpair function and adds a
column with a note if the row is in the same branch.</p>


<h3>Usage</h3>

<pre><code class='language-R'>foundereffectfinder(path_to_file_sequence_alignment = NULL,
    path_to_file_assocpair_csv_result = NULL, path_to_file_nexus_tree = NULL,
    save_name_csv, threshold = 7)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="foundereffectfinder_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>FASTA file with sequence alignment. For reference
see example file.</p>
</td></tr>
<tr><td><code id="foundereffectfinder_+3A_path_to_file_assocpair_csv_result">path_to_file_assocpair_csv_result</code></td>
<td>
<p>csv file with results from SeqFeatRs assocpair function.
For reference see example file.</p>
</td></tr>
<tr><td><code id="foundereffectfinder_+3A_path_to_file_nexus_tree">path_to_file_nexus_tree</code></td>
<td>
<p>nexus file with tree data for the given sequence
alignment. For reference see example file.</p>
</td></tr>
<tr><td><code id="foundereffectfinder_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are saved in csv format.</p>
</td></tr>
<tr><td><code id="foundereffectfinder_+3A_threshold">threshold</code></td>
<td>
<p>threshold for the number of allowed occurrences of tips
in one branch.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses the results from SeqFeatRs assocpair analysis and a generated tree
for the same sequences in nexus format
(for tree generation, e.g http://mafft.cbrc.jp/alignment/server/ can be used).
For every pair of amino acids the number of sequences with both amino acids
inside one branch is calculated. 
If this number exceeds 'threshold' times the number of found associations, this
association is considered to be due to a founder effect.</p>


<h3>Value</h3>

<p>Input file with an added column with founder effect information.</p>


<h3>Note</h3>

<p>Only use files generated with SeqFeatRs assocpair or assoctuple.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>References</h3>

<p>Mayr, Ernst (1954). &quot;Change of genetic environment and evolution&quot;. In Julian Huxley. 
Evolution as a Process. London: George Allen &amp; Unwin. OCLC 974739</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpair">assocpair</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
fasta_input &lt;- system.file("extdata", "Example_aa.fasta", package="SeqFeatR")
assocpair_result &lt;- system.file("extdata", "assocpair_results.csv", package="SeqFeatR")
tree_input &lt;- system.file("extdata", "Example_tree.nh", package="SeqFeatR")

#Usage
foundereffectfinder(
	path_to_file_sequence_alignment=fasta_input,
	path_to_file_assocpair_csv_result=assocpair_result,
	path_to_file_nexus_tree=tree_input,
	save_name_csv="foundereffectfinder_result.csv",
	threshold=7)
</code></pre>

<hr>
<h2 id='getfreqs'>Get frequencies for amino acids/nucleotides in alignment window (epitope)</h2><span id='topic+getfreqs'></span>

<h3>Description</h3>

<p>Gets frequencies of amino acids/nucleotides inside an epitope out of an alignment. Epitope start and end positions must be in the range of the length of the alignment, starting with the first position of the alignment as one. You can add the whole sequence consensus, but also a part of it.</p>


<h3>Usage</h3>

<pre><code class='language-R'>getfreqs(path_to_file_sequence_alignment = NULL, save_csv, save_png, 
 epitope_position_start, epitope_position_end,
 path_to_file_consensus, patnum_threshold = 1, A11, A12, A21, A22,
 B11, B12, B21, B22, one_feature=FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getfreqs_+3A_path_to_file_sequence_alignment">path_to_file_sequence_alignment</code></td>
<td>
<p>a FASTA file with sequence data. For reference please look in example file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_save_csv">save_csv</code></td>
<td>
<p>name for the csv output.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_save_png">save_png</code></td>
<td>
<p>name for the png/svg output.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_epitope_position_start">epitope_position_start</code></td>
<td>
<p>starting position of the epitope in the alignment.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_epitope_position_end">epitope_position_end</code></td>
<td>
<p>ending position of the epitope in the alignment.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_path_to_file_consensus">path_to_file_consensus</code></td>
<td>
<p>a FASTA file with the consensus sequence from epi_pos_start till epi_pos_end.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_patnum_threshold">patnum_threshold</code></td>
<td>
<p>the minimum number of patients of one HLA type to consider in the calculation.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_a11">A11</code></td>
<td>
<p>the position of the start of the first HLA A allele in the description block of the FASTA file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_a12">A12</code></td>
<td>
<p>the position of the end of the first HLA A allele in the description block of the FASTA file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_a21">A21</code></td>
<td>
<p>the position of the start of the second HLA A allele in the description block of the FASTA file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_a22">A22</code></td>
<td>
<p>the position of the end of the second HLA A allele in the description block of the FASTA file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_b11">B11</code></td>
<td>
<p>the position of the start of the first HLA B allele in the description block of the FASTA file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_b12">B12</code></td>
<td>
<p>the position of the end of the first HLA B allele in the description block of the FASTA file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_b21">B21</code></td>
<td>
<p>the position of the start of the second HLA B allele in the description block of the FASTA file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_b22">B22</code></td>
<td>
<p>the position of the end of the second HLA B allele in the description block of the FASTA file.</p>
</td></tr>
<tr><td><code id="getfreqs_+3A_one_feature">one_feature</code></td>
<td>
<p>if there is only one feature.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function counts the number of amino acids NOT being the same as the consensus
and generates an output graphic as well as a result csv file.
</p>
<p>The features may be HLA types, indicated by four blocks in the FASTA
comment lines. The positions of these blocks in the comment lines are
defined by parameters A11, ..., B22. For patients with a homozygous
HLA allele the second allele has to be &quot;00&quot; (without the double
quotes). For non-HLA-type features, set option one_feature=TRUE. The
value of the feature (e.g. 'yes / no', or '1 / 2 / 3') should then be
given at the end of each FASTA comment, separated from the part before
that by a semicolon.
</p>


<h3>Value</h3>

<p>A csv file with a row for each feature and two columns (absolute and relative) for the frequencies of the amino acids in the epitope.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpoint">assocpoint</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
	fasta_input &lt;- system.file("extdata", "Example_aa.fasta", package="SeqFeatR")
	consensus_input &lt;- system.file("extdata", "Example_Consensus_aa.fasta", package="SeqFeatR")
	
#Usage
getfreqs(
	path_to_file_sequence_alignment=fasta_input,
	save_csv="getfreqs_result.csv",
	save_png="getfreqs_result.png",
	epitope_position_start=1,
	epitope_position_end=40,
	path_to_file_consensus=consensus_input,
	patnum_threshold=1,
	A11=10,
	A12=11,
	A21=13,
	A22=14,
	B11=17,
	B12=18,
	B21=20,
	B22=21,
	one_feature=FALSE)
</code></pre>

<hr>
<h2 id='orPlot'>Visualize odds ratios and p-values from SeqFeatRs assocpoint</h2><span id='topic+orPlot'></span>

<h3>Description</h3>

<p>Creates odds ratio/p-value plots for every feature and sequence alignment position.</p>


<h3>Usage</h3>

<pre><code class='language-R'>orPlot(path_to_file_assocpoint_results = NULL,
	save_name_pdf, separator = ";",
	number_of_cases = 2, odds_column_position = c(6),
	p_value_column_position = c(2), name_column_position = c(2), freq = 7,
	sequence_column_position = NULL, max_y_axis = 50, interval = 10,
        has_color = 0, bias = 5, p_or_od = "P")</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="orPlot_+3A_path_to_file_assocpoint_results">path_to_file_assocpoint_results</code></td>
<td>
<p>csv file with results from SeqFeatRs assocpoint function. For reference
see example file.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_save_name_pdf">save_name_pdf</code></td>
<td>
<p>name of file to which results are saved in csv format.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_separator">separator</code></td>
<td>
<p>letter with which csv is separated (usually &quot;;&quot; or &quot;,&quot;, or &quot;\t&quot;).</p>
</td></tr>
<tr><td><code id="orPlot_+3A_number_of_cases">number_of_cases</code></td>
<td>
<p>number of different features in input file.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_odds_column_position">odds_column_position</code></td>
<td>
<p>column number of first odds ratios.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_p_value_column_position">p_value_column_position</code></td>
<td>
<p>column number of first p-value.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_name_column_position">name_column_position</code></td>
<td>
<p>column number of feature name in first row.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_freq">freq</code></td>
<td>
<p>frequency of repeat in columns with more than two cases.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_sequence_column_position">sequence_column_position</code></td>
<td>
<p>column number of first amino acid/nucleotide sequence.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_max_y_axis">max_y_axis</code></td>
<td>
<p>estimated guess of the best p-value/ highest Odds ratio.
Usuall the number behind the e as positive number.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_interval">interval</code></td>
<td>
<p>interval on the y axis.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_has_color">has_color</code></td>
<td>
<p>if there should be presented another kind of information as
a color feature. See details.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_bias">bias</code></td>
<td>
<p>bias of colors. Select with integer values if you want the first
color further down. See details.</p>
</td></tr>
<tr><td><code id="orPlot_+3A_p_or_od">p_or_od</code></td>
<td>
<p>if p-values or OR should be plotted as height. &quot;P&quot; for p-value,
&quot;OR&quot; for Odds ratios.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates from SeqFeatRs assocpoint results a graphical output where
each position in the sequence alignment is shown as bar. The height of this
bar can be the log10 of p-value at this position or the odds ratio, the
direction information (up or down) based upon odds ratio (below one is down,
above one is up). If the user adds a column number for an amino acid or DNA
base, this sequence is added above each bar.
</p>
<p>The bar plot can be additionally colored by another type of information
(e.g. display p-values as height, odds ratio as direction and sequence
entropy as color). The source of this color information has to be added
in 'has_color' as a column number. This column can be added manually before
creating this plot. The column should contain only numbers between zero and
one. A bias can be given to change the color distribution
(see R-function colorRampPalette). </p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpoint">assocpoint</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input file
assocpoint_result &lt;- system.file("extdata", "assocpoint_results.csv", package="SeqFeatR")
	
#Usage
orPlot(
	path_to_file_assocpoint_results=assocpoint_result, 
	save_name_pdf="orPlot.pdf",
	separator=";",
	number_of_cases=8,
	odds_column_position=6,
	p_value_column_position=2,
	name_column_position=2,
	freq=7,
	sequence_column_position=NULL,
	max_y_axis=4,
	interval=0.2,
	has_color=0,
	bias=5,
	p_or_od="P")
</code></pre>

<hr>
<h2 id='q-values'>Calculate q-values from p-values</h2><span id='topic+qvalues'></span>

<h3>Description</h3>

<p>Calculates and adds a column of q-values to a list of given p-values.</p>


<h3>Usage</h3>

<pre><code class='language-R'>qvalues(path_to_file_csv = NULL, save_name_csv)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="q-values_+3A_path_to_file_csv">path_to_file_csv</code></td>
<td>
<p>file with a column with p-values. Has to contain a column called &quot;p_value&quot; (without the double
quotes). For reference see example file.</p>
</td></tr>
<tr><td><code id="q-values_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are saved in csv format.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes a csv file with a column which is called 'p_values' and uses the qvalues
package to calculate from this column the corresponding q-values. 
</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input file
file_input &lt;- system.file("extdata", "assocpair_results.csv", package="SeqFeatR")

#Usage
qvalues(
	path_to_file_csv=file_input, 
	save_name_csv="q-values_result.csv")
</code></pre>

<hr>
<h2 id='rewritetuple'>Rewrite result from SeqFeatRs assoctuple</h2><span id='topic+rewritetuple'></span>

<h3>Description</h3>

<p>Function to reqrite output from SeqFeatRs assoctuple into an usable form for SeqFeatRs tartan.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rewritetuple(path_to_file_assoctuple_csv_result = NULL, save_name_csv, 
first_position, second_position, value_position, separator, threshold)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rewritetuple_+3A_path_to_file_assoctuple_csv_result">path_to_file_assoctuple_csv_result</code></td>
<td>
<p>csv file with results from SeqFeatRs assoctuple function. For reference
see example file.</p>
</td></tr>
<tr><td><code id="rewritetuple_+3A_save_name_csv">save_name_csv</code></td>
<td>
<p>name of file to which results are saved in csv format.</p>
</td></tr>
<tr><td><code id="rewritetuple_+3A_first_position">first_position</code></td>
<td>
<p>column position of first sequence position.</p>
</td></tr>
<tr><td><code id="rewritetuple_+3A_second_position">second_position</code></td>
<td>
<p>column position of second sequence position.</p>
</td></tr>
<tr><td><code id="rewritetuple_+3A_value_position">value_position</code></td>
<td>
<p>column position of p-value.</p>
</td></tr>
<tr><td><code id="rewritetuple_+3A_separator">separator</code></td>
<td>
<p>separator of csv input file (usually &quot;;&quot;, &quot;,&quot;, &quot;\t&quot;).</p>
</td></tr>
<tr><td><code id="rewritetuple_+3A_threshold">threshold</code></td>
<td>
<p>p-value threshold, below which data is to be included in result.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Extracts the two sequence positions and p-value if it is below the
threshold and generates output csv file for SeqFeatRs tartan with these values.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assoctuple">assoctuple</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input file
assoctuple_result &lt;- system.file("extdata", "assoctuple_result.csv", package="SeqFeatR")

#Usage
rewritetuple(
	path_to_file_assoctuple_csv_result=assoctuple_result, 
	save_name_csv="rewritetuple_result.csv", 
	first_position=3, 
	second_position=4, 
	value_position=10, 
	separator="\t", 
	threshold=0.1)
</code></pre>

<hr>
<h2 id='SeqFeatRGUI'>GUI for SeqFeatR</h2><span id='topic+SeqFeatR_GUI'></span>

<h3>Description</h3>

<p>A GUI for all program features of SeqFeatR</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeqFeatR_GUI()</code></pre>


<h3>Details</h3>

<p>This is a GUI for the program SeqFeatR. The first tab of the
GUI deals with the association of alignment positions with features
(e.g. HLA types); a subset of the functionality is provided by a web-server
at http://seqfeatr.zmb.uni-due. See also video tutorial at 
https://www.youtube.com/watch?v=-CYidGPE6dw and vignette in this package. </p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: SeqFeatR_GUI()

</code></pre>

<hr>
<h2 id='smallmanhattan'>A small version of manhattan plot</h2><span id='topic+smallmanhattan'></span>

<h3>Description</h3>

<p>Creates a small version of the manhattan plot from assocpoint with annotated
alignment positions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>smallmanhattan(path_to_file_csv, save_name_png, save_name_svg, feature,
       corrected=FALSE, x_axis_breaks)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smallmanhattan_+3A_path_to_file_csv">path_to_file_csv</code></td>
<td>
<p>csv file with results from assocpoint. For reference
see example file.</p>
</td></tr>
<tr><td><code id="smallmanhattan_+3A_save_name_png">save_name_png</code></td>
<td>
<p>name of file to which graphic is saved in png format.</p>
</td></tr>
<tr><td><code id="smallmanhattan_+3A_save_name_svg">save_name_svg</code></td>
<td>
<p>name of file to which graphic is saved in svg format.</p>
</td></tr>
<tr><td><code id="smallmanhattan_+3A_feature">feature</code></td>
<td>
<p>column number in input csv file with the values. See details.</p>
</td></tr>
<tr><td><code id="smallmanhattan_+3A_corrected">corrected</code></td>
<td>
<p>if the corrected p-values should be used.</p>
</td></tr>
<tr><td><code id="smallmanhattan_+3A_x_axis_breaks">x_axis_breaks</code></td>
<td>
<p>x axis breaks to be displayed. See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input file should be a result file from SeqFeatRs assocpoint.
</p>
<p>The feature, that should be displayed as an extra manhattan plot, must be given
just like it is displayed in the csv file, e.g for HLA-A2, insert &quot;A2&quot; as
value for feature. If you want to show the corrected p-values, change
corrected to TRUE.
</p>
<p>The x_axis_breaks will mark ticks on the x axis, so that a special
alignment position can be displayed. Those axis breaks have to be given
in the following format only:<br />
- starting an ending with quotes<br />
- between those quotes comma separated ascending alignemnt position numbers
(pleas keep in mind, that the plot will only display a little more than 
the highest alignment number, whatever big number you have given here)<br />
- no blanks<br />
- can be left empty (&quot;&quot;), then breaks with 50 positions will be displayed
</p>


<h3>Value</h3>

<p>A small version of the manhattan plot as png/svg picture file. svg files
can be changed more easily than png files. 
</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpoint">assocpoint</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
## Not run: 
assocpoint_result &lt;- system.file("extdata", "assocpoint_results.csv", package="SeqFeatR")

#Usage
smallmanhattan(
	assocpoint_result,
        save_name_png = "small_manhattan.png", 
        save_name_svg = "small_manhattan.svg", 
        feature = "A2",
        corrected = FALSE, 
        x_axis_breaks = "1,10,16,20,30,40")

## End(Not run)
</code></pre>

<hr>
<h2 id='tartan'>Visualize association of mutation pairs and features from two (comparable) analyses.</h2><span id='topic+tartan'></span>

<h3>Description</h3>

<p>Combines the results of two of SeqFeatRs assocpair or assoctuple analyses in one plot.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tartan(path_to_file_assocpair_csv_result=NULL,
 path_to_file_assocpair_csv_result2=NULL, save_name_pdf, space,
 colors, name_positions, names, ticks, first_position_1,
 second_position_1, value_1, first_position_2, second_position_2,
 value_2, with_distance_matrix, path_to_distance_matrix)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tartan_+3A_path_to_file_assocpair_csv_result">path_to_file_assocpair_csv_result</code></td>
<td>
<p>csv file with results from SeqFeatRs assocpair or assoctuple function. For reference
see example file.</p>
</td></tr>
<tr><td><code id="tartan_+3A_path_to_file_assocpair_csv_result2">path_to_file_assocpair_csv_result2</code></td>
<td>
<p>csv file with results from SeqFeatRs assocpair or assoctuple function. For reference
see example file.</p>
</td></tr>
<tr><td><code id="tartan_+3A_save_name_pdf">save_name_pdf</code></td>
<td>
<p>name of file to which results are saved in pdf format.</p>
</td></tr>
<tr><td><code id="tartan_+3A_space">space</code></td>
<td>
<p>space between blocks. See details.</p>
</td></tr>
<tr><td><code id="tartan_+3A_colors">colors</code></td>
<td>
<p>the colors of the tartan plot. See details.</p>
</td></tr>
<tr><td><code id="tartan_+3A_name_positions">name_positions</code></td>
<td>
<p>Positions on x and y axis where the labels should be inserted.</p>
</td></tr>
<tr><td><code id="tartan_+3A_names">names</code></td>
<td>
<p>labels to be inserted on x and y axis. 
Beware! name_position and names must have the same length!</p>
</td></tr>
<tr><td><code id="tartan_+3A_ticks">ticks</code></td>
<td>
<p>ticks which should mark interesting spots and are also the
breakpoints on which the blocks are separated. See details.</p>
</td></tr>
<tr><td><code id="tartan_+3A_first_position_1">first_position_1</code></td>
<td>
<p>column in which first alignment position is
located in the first csv file.</p>
</td></tr>
<tr><td><code id="tartan_+3A_second_position_1">second_position_1</code></td>
<td>
<p>column in which second alignment position is
located in the first csv file.</p>
</td></tr>
<tr><td><code id="tartan_+3A_value_1">value_1</code></td>
<td>
<p>column in which value to be shown is located in first csv file.</p>
</td></tr>
<tr><td><code id="tartan_+3A_first_position_2">first_position_2</code></td>
<td>
<p>column in which first alignment position is
located in the second csv file.</p>
</td></tr>
<tr><td><code id="tartan_+3A_second_position_2">second_position_2</code></td>
<td>
<p>column in which second alignment position is
located in the second csv file.</p>
</td></tr>
<tr><td><code id="tartan_+3A_value_2">value_2</code></td>
<td>
<p>column in which value to be shown is located in csv second file.</p>
</td></tr>
<tr><td><code id="tartan_+3A_with_distance_matrix">with_distance_matrix</code></td>
<td>
<p>if there is a distance matrix available and should be used.</p>
</td></tr>
<tr><td><code id="tartan_+3A_path_to_distance_matrix">path_to_distance_matrix</code></td>
<td>
<p>path to the distance matrix. Can be zero or empty.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each sequence x sequence position for one feature the corresponding
(p-)values are added as a colored dot.
</p>
<p>The input files may be the results from SeqFeatRs assocpair or assoctuple
analyses, but any csv file with two different columns of sequence positions
and one column with values is sufficient. The values should be in a
comparable range, but they can be other information as p-values
(e.g. mutual or direct information or distance matrix given as an extra
input file).
</p>
<p>The displayed colors are chosen by the user. The format for 'colors' must
be color names in quotes (e.g. &quot;red&quot;, &quot;blue&quot;, etc.) separated by &quot;,&quot;
(without the double quotes). More colors will give a better distinction.
</p>
<p>To distinguish different parts of the sequence (e.g proteins, markers, etc.)
the user may add sequence positions as ticks ('ticks') at which the graphical
display is split and a little space ('space') is inserted. The user can also
add names to be displayed on the x-axis/y-axis ('name_positions', 'names').
</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpair">assocpair</a>, <a href="#topic+assoctuple">assoctuple</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input files
assocpair_result &lt;- system.file(
                    "extdata", "shared_mutations_result_for_tartan.csv", package="SeqFeatR")
assocpair_result2 &lt;- system.file(
                    "extdata", "assocpair_results.csv", package="SeqFeatR")

#Usage
tartan(
	path_to_file_assocpair_csv_result=assocpair_result,
	path_to_file_assocpair_csv_result2=assocpair_result2,
	save_name_pdf="tartan_plot.pdf",
	space=5,
	colors=c("wheat", "darkblue", "black", "green"),
	name_positions=c(1,30),
	names=c("S","F"),
	ticks=c(10,30),
	first_position_1=2,
	second_position_1=3,
	value_1=4,
	first_position_2=2,
	second_position_2=3,
	value_2=11,
	with_distance_matrix=FALSE,
	path_to_distance_matrix=NULL)
</code></pre>

<hr>
<h2 id='visualizepair'>Visualize pairs of alignment positions that jointly mutate</h2><span id='topic+visualizepair'></span>

<h3>Description</h3>

<p>Creates a pdf output with a visualization of the results from the analysis from SeqFeatRs assocpair function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizepair(path_to_file_assocpair_csv_result = NULL, 
    save_name_pdf, significance_level = 0.01)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizepair_+3A_path_to_file_assocpair_csv_result">path_to_file_assocpair_csv_result</code></td>
<td>
<p>csv file with results from SeqFeatRs assocpair function. For reference see example file.</p>
</td></tr>
<tr><td><code id="visualizepair_+3A_save_name_pdf">save_name_pdf</code></td>
<td>
<p>name of file to which results are saved in pdf format.</p>
</td></tr>
<tr><td><code id="visualizepair_+3A_significance_level">significance_level</code></td>
<td>
<p>significance value below which the results of the analysis are considered to be relevant enough to be plotted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An output is created with a sequence position versus sequence position
graphic. Every dot color(one position with the other position)
is based on its p-value. The color code is on the right side. In contrast to
visualizepairfeat, no feature information will be used.</p>


<h3>Note</h3>

<p>Only use files generated from SeqFeatRs assocpair function.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpair">assocpair</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input file
assocpair_result &lt;- system.file("extdata", "assocpair_results.csv", package="SeqFeatR")

#Usage
visualizepair(
	path_to_file_assocpair_csv_result=assocpair_result,
	save_name_pdf="vispair_plot.pdf",
	significance_level=0.05)
</code></pre>

<hr>
<h2 id='visualizepairfeat'>Visualize pairs of sequence alignment positions associated with sequence feature.</h2><span id='topic+visualizepairfeat'></span>

<h3>Description</h3>

<p>Creates a pdf output with a visualization of the results from the analysis from SeqFeatRs assocpairfeat function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizepairfeat(path_to_file_assocpairfeat_csv_result = NULL, save_name_pdf, 
    significance_level = 0.01)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizepairfeat_+3A_path_to_file_assocpairfeat_csv_result">path_to_file_assocpairfeat_csv_result</code></td>
<td>
<p>csv file with results from SeqFeatRs assocpairfeat function. For reference see example file.</p>
</td></tr>
<tr><td><code id="visualizepairfeat_+3A_save_name_pdf">save_name_pdf</code></td>
<td>
<p>name of file to which results are saved in pdf format.</p>
</td></tr>
<tr><td><code id="visualizepairfeat_+3A_significance_level">significance_level</code></td>
<td>
<p>significance value below which the results of the analysis are considered to be relevant enough to be plotted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For every feature (if only one p-value is below given threshold ('significance_level'))
an output page (pdf page) is created with a sequence position versus
sequence position graphic. 
Every dot color(one position with the other position) is based on its p-value. 
The color code is on the right side.</p>


<h3>Note</h3>

<p>Only use files generated from SeqFeatRs assocpairfeat function.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>See Also</h3>

<p><code><a href="#topic+assocpairfeat">assocpairfeat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input file
assocpairfeat_result &lt;- system.file("extdata", "assocpairfeat_results.csv", package="SeqFeatR")

#Usage
visualizepairfeat(
	path_to_file_assocpairfeat_csv_result=assocpairfeat_result,
	save_name_pdf="vispairfeat_plot.pdf", 
	significance_level=0.05)
</code></pre>

<hr>
<h2 id='volcanoplot'>Create volcano plot</h2><span id='topic+volcanoplot'></span>

<h3>Description</h3>

<p>Creates a pdf output file with a volcano plot out of the results from SeqFeatRs assocpoint.</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcanoplot(path_to_file_assocpoint_csv_result,
 save_name_pdf, p_values_pos, odds_pos, pos_pos,
 level_of_sig_p, level_of_sig_odds)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="volcanoplot_+3A_path_to_file_assocpoint_csv_result">path_to_file_assocpoint_csv_result</code></td>
<td>
<p>csv file with results from SeqFeatRs assocpoint. For reference see example file.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_save_name_pdf">save_name_pdf</code></td>
<td>
<p>name of file to which results are saved in pdf format.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_p_values_pos">p_values_pos</code></td>
<td>
<p>column number with p-values.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_odds_pos">odds_pos</code></td>
<td>
<p>column number with odds ratios.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_pos_pos">pos_pos</code></td>
<td>
<p>column number with positions.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_level_of_sig_p">level_of_sig_p</code></td>
<td>
<p>level of significance p values.</p>
</td></tr>
<tr><td><code id="volcanoplot_+3A_level_of_sig_odds">level_of_sig_odds</code></td>
<td>
<p>level of significance odds ratios.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots log$_10$ Odds ratios versus -log$_10$ p-values and annotates the dots with sequence positions.</p>


<h3>Author(s)</h3>

<p>Bettina Budeus</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Input file
assocpoint_result &lt;- system.file("extdata", "assocpoint_results.csv", package="SeqFeatR")

#Usage
volcanoplot(
	path_to_file_assocpoint_csv_result=assocpoint_result,
	save_name_pdf="volcano_plot.pdf",
	p_values_pos = 3,
	odds_pos = 6,
	pos_pos = 1,
	level_of_sig_p = 0.05,
	level_of_sig_odds = 1)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
