<!DOCTYPE html><html lang="en"><head><title>Help for package SeqKat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SeqKat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#combine.table'><p>Combine Table</p></a></li>
<li><a href='#final.score'><p>Final Score</p></a></li>
<li><a href='#get.context'><p>Get Context</p></a></li>
<li><a href='#get.exprobntcx'><p>get.exprobntcx</p></a></li>
<li><a href='#get.nucleotide.chunk.counts'><p>Get Nucleotide Chunk Counts</p></a></li>
<li><a href='#get.pair'><p>Get Pair</p></a></li>
<li><a href='#get.tn'><p>Get Trinucleotides</p></a></li>
<li><a href='#get.toptn'><p>Get Top Trinucleotides</p></a></li>
<li><a href='#get.trinucleotide.counts'><p>Get Trinucleotide Counts</p></a></li>
<li><a href='#seqkat'><p>SeqKat</p></a></li>
<li><a href='#test.kataegis'><p>Test Kataegis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Detection of Kataegis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-03-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Fouad Yousif, Xihui Lin, Fan Fan, Christopher Lalansingh, John Macdonald</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Paul C. Boutros &lt;pboutros@mednet.ucla.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Kataegis is a localized hypermutation occurring when a region is enriched in somatic SNVs. Kataegis can result from multiple cytosine deaminations catalyzed by the AID/APOBEC family of proteins. This package contains functions to detect kataegis from SNVs in BED format. This package reports two scores per kataegic event, a hypermutation score and an APOBEC mediated kataegic score. Yousif, F. et al.; The Origins and Consequences of Localized and Global Somatic Hypermutation; Biorxiv 2018 &lt;<a href="https://doi.org/10.1101%2F287839">doi:10.1101/287839</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.1), foreach, doParallel</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp(&ge; 0.11.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, doMC, rmarkdown, knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-03-09 21:07:12 UTC; root</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-03-11 00:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='combine.table'>Combine Table</h2><span id='topic+combine.table'></span>

<h3>Description</h3>

<p>Merges overlapped windows to identify genomic boundaries of kataegic events. This function also assigns hypermuation and kataegic score for combined windows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine.table(test.table, somatic, mutdistance, segnum, output.name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combine.table_+3A_test.table">test.table</code></td>
<td>
<p>Data frame of kataegis test scores</p>
</td></tr>
<tr><td><code id="combine.table_+3A_somatic">somatic</code></td>
<td>
<p>Data frame of somatic variants</p>
</td></tr>
<tr><td><code id="combine.table_+3A_mutdistance">mutdistance</code></td>
<td>
<p>The maximum intermutational distance allowed for SNVs to be grouped in the same kataegic event. Recommended value: 3.2</p>
</td></tr>
<tr><td><code id="combine.table_+3A_segnum">segnum</code></td>
<td>
<p>Minimum mutation count. The minimum number of mutations required within a cluster to be identified as kataegic. Recommended value: 4</p>
</td></tr>
<tr><td><code id="combine.table_+3A_output.name">output.name</code></td>
<td>
<p>Name of the generated output directory.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fouad Yousif
</p>
<p>Fan Fan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/somatic.rda"
		)
	);

load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/final.score.rda"
		)
	);

combine.table(
	final.score,
	somatic,
	3.2,
	4,
	tempdir()
	);
</code></pre>

<hr>
<h2 id='final.score'>Final Score</h2><span id='topic+final.score'></span>

<h3>Description</h3>

<p>Assigns hypermutation score (hm.score) and kataegic score (k.score)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>final.score(test.table, cutoff, somatic, output.name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="final.score_+3A_test.table">test.table</code></td>
<td>
<p>Data frame of kataegis test scores</p>
</td></tr>
<tr><td><code id="final.score_+3A_cutoff">cutoff</code></td>
<td>
<p>The minimum hypermutation score used to classify the windows in the sliding binomial test as significant windows. The score is calculated per window as follows: -log10(binomial test p-value). Recommended value: 5</p>
</td></tr>
<tr><td><code id="final.score_+3A_somatic">somatic</code></td>
<td>
<p>Data frame of somatic variants</p>
</td></tr>
<tr><td><code id="final.score_+3A_output.name">output.name</code></td>
<td>
<p>Name of the generated output directory.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Fan
</p>
<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/somatic.rda"
		)
	);

load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/test.table.rda"
		)
	);

final.score(
	test.table,
	5,
	somatic,
	tempdir()
	);
</code></pre>

<hr>
<h2 id='get.context'>Get Context</h2><span id='topic+get.context'></span>

<h3>Description</h3>

<p>Gets the 5' and 3' neighboring bases to the mutated base
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.context(file, start)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.context_+3A_file">file</code></td>
<td>
<p>Reference files directory</p>
</td></tr>
<tr><td><code id="get.context_+3A_start">start</code></td>
<td>
<p>The position of the mutation gene</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The trinucleotide context.
</p>


<h3>Author(s)</h3>

<p>Fouad Yousif
</p>
<p>Fan Fan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);
get.context(file.path(example.ref.dir, 'chr4.fa'), c(1582933, 1611781))
</code></pre>

<hr>
<h2 id='get.exprobntcx'>get.exprobntcx</h2><span id='topic+get.exprobntcx'></span>

<h3>Description</h3>

<p>Gets the expected probability for each trinucleotide and total number of tcx
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.exprobntcx(somatic, ref.dir, trinucleotide.count.file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.exprobntcx_+3A_somatic">somatic</code></td>
<td>
<p>Data frame of somatic variants</p>
</td></tr>
<tr><td><code id="get.exprobntcx_+3A_ref.dir">ref.dir</code></td>
<td>
<p>Path to a directory containing the reference genome.</p>
</td></tr>
<tr><td><code id="get.exprobntcx_+3A_trinucleotide.count.file">trinucleotide.count.file</code></td>
<td>
<p>A tab seprarated file containing a count of all trinucleotides present in the reference genome. This can be generated with the get.trinucleotide.counts() function in this package.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Fan
</p>
<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/somatic.rda"
		)
	);

trinucleotide.count.file &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/tn_count.txt"
	);

example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);

get.exprobntcx(somatic, example.ref.dir, trinucleotide.count.file)
</code></pre>

<hr>
<h2 id='get.nucleotide.chunk.counts'>Get Nucleotide Chunk Counts</h2><span id='topic+get.nucleotide.chunk.counts'></span>

<h3>Description</h3>

<p>Obtain counts for all possible trinucleotides within a specified genomic region
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.nucleotide.chunk.counts(key, chr, upstream = 1, downstream = 1,
  start = 1, end = -1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.nucleotide.chunk.counts_+3A_key">key</code></td>
<td>
<p>List of specify trinucleotides to count</p>
</td></tr>
<tr><td><code id="get.nucleotide.chunk.counts_+3A_chr">chr</code></td>
<td>
<p>Chromosome</p>
</td></tr>
<tr><td><code id="get.nucleotide.chunk.counts_+3A_upstream">upstream</code></td>
<td>
<p>Length upstream to read</p>
</td></tr>
<tr><td><code id="get.nucleotide.chunk.counts_+3A_downstream">downstream</code></td>
<td>
<p>Length downstream to read</p>
</td></tr>
<tr><td><code id="get.nucleotide.chunk.counts_+3A_start">start</code></td>
<td>
<p>Starting position</p>
</td></tr>
<tr><td><code id="get.nucleotide.chunk.counts_+3A_end">end</code></td>
<td>
<p>Ending position</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);

bases.raw &lt;- c('A','C','G','T','N');
tri.types.raw &lt;- c(
	outer(
		c(outer(bases.raw, bases.raw, function(x, y) paste0(x,y))),
		bases.raw, function(x, y) paste0(x,y))
	);
tri.types.raw &lt;- sort(tri.types.raw);
get.nucleotide.chunk.counts(
	tri.types.raw,
	file.path(example.ref.dir, 'chr4.fa'),
	upstream = 1,
	downstream = 1,
	start = 1,
	end = -1
	);
</code></pre>

<hr>
<h2 id='get.pair'>Get Pair</h2><span id='topic+get.pair'></span>

<h3>Description</h3>

<p>Generates the reverse compliment of a nucleotide sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.pair(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.pair_+3A_x">x</code></td>
<td>
<p>asdf</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reverses and compliments the bases of the input string. Bases must be (A, C, G, T, or N).
</p>


<h3>Author(s)</h3>

<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get.pair("GATTACA")
</code></pre>

<hr>
<h2 id='get.tn'>Get Trinucleotides</h2><span id='topic+get.tn'></span>

<h3>Description</h3>

<p>Count the frequencies of 32 trinucleotide in a region respectively
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.tn(chr, start.bp, end.bp, ref.dir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.tn_+3A_chr">chr</code></td>
<td>
<p>Chromosome</p>
</td></tr>
<tr><td><code id="get.tn_+3A_start.bp">start.bp</code></td>
<td>
<p>Starting position</p>
</td></tr>
<tr><td><code id="get.tn_+3A_end.bp">end.bp</code></td>
<td>
<p>Ending position</p>
</td></tr>
<tr><td><code id="get.tn_+3A_ref.dir">ref.dir</code></td>
<td>
<p>Path to a directory containing the reference genome.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Fan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);
get.tn(chr=4, start.bp=1, end.bp=-1, example.ref.dir)
</code></pre>

<hr>
<h2 id='get.toptn'>Get Top Trinucleotides</h2><span id='topic+get.toptn'></span>

<h3>Description</h3>

<p>Generate a tri-nucleotide summary for each sliding window
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.toptn(somatic.subset, chr, start.bp, end.bp, ref.dir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.toptn_+3A_somatic.subset">somatic.subset</code></td>
<td>
<p>Data frame of somatic variants subset for a specific chromosome</p>
</td></tr>
<tr><td><code id="get.toptn_+3A_chr">chr</code></td>
<td>
<p>Chromosome</p>
</td></tr>
<tr><td><code id="get.toptn_+3A_start.bp">start.bp</code></td>
<td>
<p>Starting position</p>
</td></tr>
<tr><td><code id="get.toptn_+3A_end.bp">end.bp</code></td>
<td>
<p>Ending position</p>
</td></tr>
<tr><td><code id="get.toptn_+3A_ref.dir">ref.dir</code></td>
<td>
<p>Path to a directory containing the reference genome.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Fan
</p>
<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get.toptn(somatic.subset, chr, start.bp, end.bp, ref.dir)

## End(Not run)

</code></pre>

<hr>
<h2 id='get.trinucleotide.counts'>Get Trinucleotide Counts</h2><span id='topic+get.trinucleotide.counts'></span>

<h3>Description</h3>

<p>Aggregates the total counts of each possible trinucleotide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.trinucleotide.counts(ref.dir, ref.name, output.dir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.trinucleotide.counts_+3A_ref.dir">ref.dir</code></td>
<td>
<p>Path to a directory containing the reference genome.</p>
</td></tr>
<tr><td><code id="get.trinucleotide.counts_+3A_ref.name">ref.name</code></td>
<td>
<p>Name of the reference genome being used (i.e. hg19, GRCh38, etc)</p>
</td></tr>
<tr><td><code id="get.trinucleotide.counts_+3A_output.dir">output.dir</code></td>
<td>
<p>Path to a directory where output will be created.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Fan
</p>
<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get.trinucleotide.counts(ref.dir, "hg19", tempdir());

## End(Not run)
</code></pre>

<hr>
<h2 id='seqkat'>SeqKat</h2><span id='topic+seqkat'></span>

<h3>Description</h3>

<p>Kataegis detection from SNV BED files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seqkat(sigcutoff = 5, mutdistance = 3.2, segnum = 4, ref.dir = NULL,
  bed.file = "./", output.dir = "./", chromosome = "all",
  chromosome.length.file = NULL, trinucleotide.count.file = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="seqkat_+3A_sigcutoff">sigcutoff</code></td>
<td>
<p>The minimum hypermutation score used to classify the windows in the sliding binomial test as significant windows. The score is calculated per window as follows: -log10(binomial test p-value). Recommended value: 5</p>
</td></tr>
<tr><td><code id="seqkat_+3A_mutdistance">mutdistance</code></td>
<td>
<p>The maximum intermutational distance allowed for SNVs to be grouped in the same kataegic event. Recommended value: 3.2</p>
</td></tr>
<tr><td><code id="seqkat_+3A_segnum">segnum</code></td>
<td>
<p>Minimum mutation count. The minimum number of mutations required within a cluster to be identified as kataegic. Recommended value: 4</p>
</td></tr>
<tr><td><code id="seqkat_+3A_ref.dir">ref.dir</code></td>
<td>
<p>Path to a directory containing the reference genome. Each chromosome should have its own .fa file and chromosomes X and Y are named as chr23 and chr24. The fasta files should contain no header</p>
</td></tr>
<tr><td><code id="seqkat_+3A_bed.file">bed.file</code></td>
<td>
<p>Path to the SNV BED file. The BED file should contain the following information: Chromosome, Position, Reference allele, Alternate allele</p>
</td></tr>
<tr><td><code id="seqkat_+3A_output.dir">output.dir</code></td>
<td>
<p>Path to a directory where output will be created.</p>
</td></tr>
<tr><td><code id="seqkat_+3A_chromosome">chromosome</code></td>
<td>
<p>The chromosome to be analysed. This can be (1, 2, ..., 23, 24) or &quot;all&quot; to run sequentially on all chromosomes.</p>
</td></tr>
<tr><td><code id="seqkat_+3A_chromosome.length.file">chromosome.length.file</code></td>
<td>
<p>A tab separated file containing the lengths of all chromosomes in the reference genome.</p>
</td></tr>
<tr><td><code id="seqkat_+3A_trinucleotide.count.file">trinucleotide.count.file</code></td>
<td>
<p>A tab seprarated file containing a count of all trinucleotides present in the reference genome. This can be generated with the get.trinucleotide.counts() function in this package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default paramters in SeqKat have been optimized using Alexanrov's &quot;Signatures of mutational processes in human cancer&quot; dataset. SeqKat accepts a BED file and outputs the results in TXT format. A file per chromosome is generated if a kataegic event is detected, otherwise no file is generated. SeqKat reports two scores per kataegic event, a hypermutation score and an APOBEC mediated kataegic score.
</p>


<h3>Author(s)</h3>

<p>Fouad Yousif
</p>
<p>Fan Fan
</p>
<p>Christopher Lalansingh
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example.bed.file &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/PD4120a-chr4-1-2000000_test_snvs.bed"
	);
example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);
example.chromosome.length.file &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/length_hg19_chr_test.txt"
	);
seqkat(
	5,
	3.2,
	2,
	bed.file = example.bed.file,
	output.dir = tempdir(),
	chromosome = "4",
	ref.dir = example.ref.dir,
	chromosome.length.file = example.chromosome.length.file
	);
</code></pre>

<hr>
<h2 id='test.kataegis'>Test Kataegis</h2><span id='topic+test.kataegis'></span>

<h3>Description</h3>

<p>Performs exact binomial test to test the deviation of the 32 tri-nucleotides counts from expected
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test.kataegis(chromosome.num, somatic, units, exprobntcx, output.name, ref.dir,
  chromosome.length.file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test.kataegis_+3A_chromosome.num">chromosome.num</code></td>
<td>
<p>Chromosome</p>
</td></tr>
<tr><td><code id="test.kataegis_+3A_somatic">somatic</code></td>
<td>
<p>Data frame of somatic variants</p>
</td></tr>
<tr><td><code id="test.kataegis_+3A_units">units</code></td>
<td>
<p>Base window size</p>
</td></tr>
<tr><td><code id="test.kataegis_+3A_exprobntcx">exprobntcx</code></td>
<td>
<p>Expected probability for each trinucleotide and total number of tcx</p>
</td></tr>
<tr><td><code id="test.kataegis_+3A_output.name">output.name</code></td>
<td>
<p>Name of the generated output directory.</p>
</td></tr>
<tr><td><code id="test.kataegis_+3A_ref.dir">ref.dir</code></td>
<td>
<p>Path to a directory containing the reference genome.</p>
</td></tr>
<tr><td><code id="test.kataegis_+3A_chromosome.length.file">chromosome.length.file</code></td>
<td>
<p>A tab separated file containing the lengths of all chromosomes in the reference genome.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/somatic.rda"
		)
	);

load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/exprobntcx.rda"
		)
	);

example.chromosome.length.file &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/length_hg19_chr_test.txt"
	);

example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);

test.kataegis(
	4,
	somatic,
	2,
	exprobntcx,
	tempdir(),
	example.ref.dir,
	example.chromosome.length.file
	);
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
