<!DOCTYPE html><html lang="en"><head><title>Help for package seqminer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {seqminer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SEQMINER'><p>Efficiently Read Sequencing Data (VCF format, METAL format) into R</p></a></li>
<li><a href='#+5B.PlinkFile'><p>Read from binary PLINK file and return a genotype matrix</p></a></li>
<li><a href='#addJob'><p>Add a job to a workflow</p></a></li>
<li><a href='#annotateGene'><p>Annotate a test variant</p></a></li>
<li><a href='#annotatePlain'><p>Annotate a plain text file</p></a></li>
<li><a href='#annotateVcf'><p>Annotate a VCF file</p></a></li>
<li><a href='#createSingleChromosomeBCFIndex'><p>Create a single chromosome index</p></a></li>
<li><a href='#createSingleChromosomeVCFIndex'><p>Create a single chromosome index</p></a></li>
<li><a href='#download.annotation.resource'><p>Download annotation resources to a directory</p></a></li>
<li><a href='#getCovPair'><p>Extract pair of positions by ranges</p></a></li>
<li><a href='#getRefBase'><p>Annotate a test variant</p></a></li>
<li><a href='#hasIndex'><p>Check input file has tabix index</p></a></li>
<li><a href='#isDirWritable'><p>Test whether directory is writable</p></a></li>
<li><a href='#isInRange'><p>Test whether a vector of positions are inside given ranges</p></a></li>
<li><a href='#isTabixRange'><p>Check if the inputs are valid tabix range such as chr1:2-300</p></a></li>
<li><a href='#isURL'><p>Check if the input is url e.g. http:// or ftp://</p></a></li>
<li><a href='#makeAnnotationParameter'><p>Construct a usable set of annotation parameters</p></a></li>
<li><a href='#newJob'><p>Create a new job</p></a></li>
<li><a href='#newWorkflow'><p>Create a new workflow</p></a></li>
<li><a href='#openPlink'><p>Open binary PLINK files</p></a></li>
<li><a href='#readBGENToListByGene'><p>Read information from BGEN file in a given range and return a list</p></a></li>
<li><a href='#readBGENToListByRange'><p>Read information from BGEN file in a given range and return a list</p></a></li>
<li><a href='#readBGENToMatrixByGene'><p>Read a gene from BGEN file and return a genotype matrix</p></a></li>
<li><a href='#readBGENToMatrixByRange'><p>Read a gene from BGEN file and return a genotype matrix</p></a></li>
<li><a href='#readPlinkToMatrixByIndex'><p>Read from binary PLINK file and return a genotype matrix</p></a></li>
<li><a href='#readSingleChromosomeBCFToMatrixByRange'><p>Read a range from BCF file and return a genotype matrix</p></a></li>
<li><a href='#readSingleChromosomeVCFToMatrixByRange'><p>Read a range from VCF file and return a genotype matrix</p></a></li>
<li><a href='#readVCFToListByGene'><p>Read information from VCF file in a given range and return a list</p></a></li>
<li><a href='#readVCFToListByRange'><p>Read information from VCF file in a given range and return a list</p></a></li>
<li><a href='#readVCFToMatrixByGene'><p>Read a gene from VCF file and return a genotype matrix</p></a></li>
<li><a href='#readVCFToMatrixByRange'><p>Read a gene from VCF file and return a genotype matrix</p></a></li>
<li><a href='#rvmeta.readCovByRange'><p>Read covariance by range from METAL-format files.</p></a></li>
<li><a href='#rvmeta.readDataByGene'><p>Read association statistics by gene from METAL-format files. Both score statistics and covariance statistics will be extracted.</p></a></li>
<li><a href='#rvmeta.readDataByRange'><p>Read association statistics by range from METAL-format files. Both score statistics and covariance statistics will be extracted.</p></a></li>
<li><a href='#rvmeta.readNullModel'><p>Read null model statistics</p></a></li>
<li><a href='#rvmeta.readScoreByRange'><p>Read score test statistics by range from METAL-format files.</p></a></li>
<li><a href='#rvmeta.readSkewByRange'><p>Read skew by range from METAL-format files.</p></a></li>
<li><a href='#rvmeta.writeCovData'><p>Write covariance association statistics files.</p></a></li>
<li><a href='#rvmeta.writeScoreData'><p>Write score-based association statistics files.</p></a></li>
<li><a href='#tabix.createIndex'><p>Create tabix index file, similar to running tabix in command line.</p></a></li>
<li><a href='#tabix.createIndex.meta'><p>Create tabix index for bgzipped MetaScore/MetaCov file</p></a></li>
<li><a href='#tabix.createIndex.vcf'><p>Create tabix index for bgzipped VCF file</p></a></li>
<li><a href='#tabix.read'><p>Read tabix file, similar to running tabix in command line.</p></a></li>
<li><a href='#tabix.read.header'><p>Read tabix file, similar to running tabix in command line.</p></a></li>
<li><a href='#tabix.read.table'><p>Read tabix file, similar to running tabix in command line.</p></a></li>
<li><a href='#validateAnnotationParameter'><p>Validate annotate parameter is valid</p></a></li>
<li><a href='#verifyFilename'><p>validate the inVcf can be created, and outVcf can be write to.</p>
will stop if any error occurs</a></li>
<li><a href='#writeWorkflow'><p>Export workflow to Makefile</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Efficiently Read Sequence Data (VCF Format, BCF Format, METAL
Format and BGEN Format) into R</td>
</tr>
<tr>
<td>Version:</td>
<td>9.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-09-30</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaowei Zhan &lt;zhanxw@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Integrate sequencing data (Variant call format, e.g. VCF or BCF) or meta-analysis results in R. This package can help you (1) read VCF/BCF/BGEN files by chromosomal ranges (e.g. 1:100-200); (2) read RareMETAL summary statistics files; (3) read tables from a tabix-indexed files; (4) annotate VCF/BCF files; (5) create customized workflow based on Makefile.</td>
</tr>
<tr>
<td>Copyright:</td>
<td>We have used the following software and made minimal
necessary changes: tabix, Heng Li &lt;lh3@live.co.uk&gt; (MIT
license), SQLite (Public Domain), Zstandard (BSD license). For
tabix, we removed standard IO related functions, e.g. printf,
fprintf ; also changed its un-safe pointer arithmetics. For
zstandard, we removed compiler (clang, MSVC) specific
preprocessing flags.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE [expanded from: GPL | file LICENSE]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://zhanxw.github.io/seqminer/">http://zhanxw.github.io/seqminer/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/zhanxw/seqminer/issues">https://github.com/zhanxw/seqminer/issues</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-02 03:13:09 UTC; zhanxw</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, SKAT</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++17, zlib headers and libraries, GNU make,
optionally also bzip2 and POSIX-compliant regex functions.</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiaowei Zhan [aut, cre],
  Dajiang Liu [aut],
  Attractive Chaos [cph] (We have used the following software and made
    minimal necessary changes: Tabix, Heng Li &lt;lh3@live.co.uk&gt; (MIT
    license). We removed standard IO related functions, e.g. printf,
    fprintf ; also changed its un-safe pointer arithmetics.),
  Broad Institute / Massachusetts Institute of Technology [cph],
  Genome Research Ltd (GRL) [cph],
  Facebook, Inc [cph],
  D. Richard Hipp [cph]</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-02 06:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='SEQMINER'>Efficiently Read Sequencing Data (VCF format, METAL format) into R</h2><span id='topic+seqminer'></span><span id='topic+seqminer-package'></span>

<h3>Description</h3>

<p>SeqMiner provides functions to easily load Variant Call Format (VCF) or METAL format into R
</p>


<h3>Details</h3>

<p>The aim of this package is to save your time parsing large text file.
That means data processing time can be saved for other researches.
This packages requires Bgzip compressed and Tabix indexed files as input.
If input files contaians annotation by TabAnno (),
it is possible to extract information at the unit of genes.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Xiaowei Zhan <a href="mailto:zhanxw@gmail.com">zhanxw@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Dajiang Liu <a href="mailto:dajiang.liu@gmail.com">dajiang.liu@gmail.com</a>
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Attractive Chaos <a href="mailto:attractor@live.co.uk">attractor@live.co.uk</a> (We have used the following software and made minimal necessary changes: Tabix, Heng Li &lt;lh3@live.co.uk&gt; (MIT license). We removed standard IO related functions, e.g. printf, fprintf ; also changed its un-safe pointer arithmetics.) [copyright holder]
</p>
</li>
<li><p> Broad Institute / Massachusetts Institute of Technology [copyright holder]
</p>
</li>
<li><p> Genome Research Ltd (GRL) [copyright holder]
</p>
</li>
<li><p> Facebook, Inc [copyright holder]
</p>
</li>
<li><p> D. Richard Hipp [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="http://zhanxw.github.io/seqminer/">http://zhanxw.github.io/seqminer/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/zhanxw/seqminer/issues">https://github.com/zhanxw/seqminer/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+5B.PlinkFile'>Read from binary PLINK file and return a genotype matrix</h2><span id='topic++5B.PlinkFile'></span>

<h3>Description</h3>

<p>Read from binary PLINK file and return a genotype matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PlinkFile'
plinkFileObject[sampleIndex, markerIndex]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B5B.PlinkFile_+3A_plinkfileobject">plinkFileObject</code></td>
<td>
<p>a PlinkFileObject obtained by openPlink()</p>
</td></tr>
<tr><td><code id="+2B5B.PlinkFile_+3A_sampleindex">sampleIndex</code></td>
<td>
<p>integer, 1-basd, index of samples to be extracted</p>
</td></tr>
<tr><td><code id="+2B5B.PlinkFile_+3A_markerindex">markerIndex</code></td>
<td>
<p>integer, 1-basd, index of markers to be extracted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype matrix, marker by sample
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## these indice are nonsynonymous markers for 1:196621007-196716634",
## refer to the readVCFToMatrixByRange()
fileName = system.file("plink/all.anno.filtered.extract.bed", package = "seqminer")
filePrefix = sub(fileName, pattern = ".bed", replacement = "")
plinkObj = openPlink(filePrefix)
sampleIndex = seq(3)
markerIndex =c(14, 36)
cfh &lt;- plinkObj[sampleIndex, markerIndex]
</code></pre>

<hr>
<h2 id='addJob'>Add a job to a workflow</h2><span id='topic+addJob'></span>

<h3>Description</h3>

<p>Add a job to a workflow
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addJob(wf, job)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addJob_+3A_wf">wf</code></td>
<td>
<p>a variable of workflow class</p>
</td></tr>
<tr><td><code id="addJob_+3A_job">job</code></td>
<td>
<p>a variable of job class</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>j1 &lt;- newJob('id1', 'cmd out1', 'out1')
j2 &lt;- newJob('id2', 'cmd out2', 'out2', depend = 'out1')
w &lt;- newWorkflow("wf")
w &lt;- addJob(w, j1)
w &lt;- addJob(w, j2)

outFile &lt;- file.path(tempdir(), "Makefile")
writeWorkflow(w, outFile)
cat('Outputted Makefile file are in the temp directory:', outFile, '\n')
</code></pre>

<hr>
<h2 id='annotateGene'>Annotate a test variant</h2><span id='topic+annotateGene'></span>

<h3>Description</h3>

<p>Annotate a test variant
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotateGene(param, chrom, position, ref, alt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="annotateGene_+3A_param">param</code></td>
<td>
<p>a list of annotation configuaration (e.g. reference file, gene definition)</p>
</td></tr>
<tr><td><code id="annotateGene_+3A_chrom">chrom</code></td>
<td>
<p>a vector of chromosome names</p>
</td></tr>
<tr><td><code id="annotateGene_+3A_position">position</code></td>
<td>
<p>a vector of chromosome positions</p>
</td></tr>
<tr><td><code id="annotateGene_+3A_ref">ref</code></td>
<td>
<p>a vector of reference alleles</p>
</td></tr>
<tr><td><code id="annotateGene_+3A_alt">alt</code></td>
<td>
<p>a vector of alternative alleles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>annotated results in a data frame structure
</p>


<h3>See Also</h3>

<p>makeAnnotationParameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (.Platform$endian == "little") {
  param &lt;- list(reference = system.file("tabanno/test.fa", package = "seqminer"),
                geneFile = system.file("tabanno/test.gene.txt", package = "seqminer"))
  param &lt;- makeAnnotationParameter(param)
  print(param)
  annotateGene(param, c("1", "1"), c(3, 5) , c("A", "C"), c("G", "C"))
} else {
  message("Tabix does not work well for big endian for now")
}
</code></pre>

<hr>
<h2 id='annotatePlain'>Annotate a plain text file</h2><span id='topic+annotatePlain'></span>

<h3>Description</h3>

<p>Annotate a plain text file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotatePlain(inFile, outFile, params)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="annotatePlain_+3A_infile">inFile</code></td>
<td>
<p>input file name</p>
</td></tr>
<tr><td><code id="annotatePlain_+3A_outfile">outFile</code></td>
<td>
<p>output file name</p>
</td></tr>
<tr><td><code id="annotatePlain_+3A_params">params</code></td>
<td>
<p>parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>0 if succeed
</p>


<h3>Examples</h3>

<pre><code class='language-R'>param &lt;- list(reference = system.file("tabanno/test.fa", package = "seqminer"),
              geneFile = system.file("tabanno/test.gene.txt", package = "seqminer"),
              inputFormat = "plain")
param &lt;- makeAnnotationParameter(param)
inFile &lt;- system.file("tabanno/input.test.plain.txt", package = "seqminer")
outFile &lt;- file.path(tempdir(), "out.annotated.txt")
annotatePlain(inFile, outFile, param)
cat('Outputted annotation results are in the temp directory:', outFile, '\n')
</code></pre>

<hr>
<h2 id='annotateVcf'>Annotate a VCF file</h2><span id='topic+annotateVcf'></span>

<h3>Description</h3>

<p>Annotate a VCF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotateVcf(inVcf, outVcf, params)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="annotateVcf_+3A_invcf">inVcf</code></td>
<td>
<p>input VCF file name</p>
</td></tr>
<tr><td><code id="annotateVcf_+3A_outvcf">outVcf</code></td>
<td>
<p>output VCF file name</p>
</td></tr>
<tr><td><code id="annotateVcf_+3A_params">params</code></td>
<td>
<p>parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>0 if succeed
</p>


<h3>Examples</h3>

<pre><code class='language-R'>param &lt;- list(reference = system.file("tabanno/test.fa", package = "seqminer"),
              geneFile = system.file("tabanno/test.gene.txt", package = "seqminer"))
param &lt;- makeAnnotationParameter(param)
inVcf &lt;- system.file("tabanno/input.test.vcf", package = "seqminer")
outVcf &lt;- file.path(tempdir(), "/", "out.vcf")
annotateVcf (inVcf, outVcf, param)
cat('Annotated VCF files are in the temp directory:', outVcf, '\n')
</code></pre>

<hr>
<h2 id='createSingleChromosomeBCFIndex'>Create a single chromosome index</h2><span id='topic+createSingleChromosomeBCFIndex'></span>

<h3>Description</h3>

<p>Create a single chromosome index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createSingleChromosomeBCFIndex(fileName, indexFileName = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createSingleChromosomeBCFIndex_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input BCF file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="createSingleChromosomeBCFIndex_+3A_indexfilename">indexFileName</code></td>
<td>
<p>character, by default, create 'fileName'.scIdx</p>
</td></tr>
</table>


<h3>Value</h3>

<p>indexFileName if success, or NULL is failed
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.headerFixed.bcf.gz", package = "seqminer")
cfh &lt;- createSingleChromosomeBCFIndex(fileName)
</code></pre>

<hr>
<h2 id='createSingleChromosomeVCFIndex'>Create a single chromosome index</h2><span id='topic+createSingleChromosomeVCFIndex'></span>

<h3>Description</h3>

<p>Create a single chromosome index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createSingleChromosomeVCFIndex(fileName, indexFileName = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createSingleChromosomeVCFIndex_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input VCF file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="createSingleChromosomeVCFIndex_+3A_indexfilename">indexFileName</code></td>
<td>
<p>character, by default, create 'fileName'.scIdx</p>
</td></tr>
</table>


<h3>Value</h3>

<p>indexFileName if success, or NULL is failed
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
cfh &lt;- createSingleChromosomeVCFIndex(fileName)
</code></pre>

<hr>
<h2 id='download.annotation.resource'>Download annotation resources to a directory</h2><span id='topic+download.annotation.resource'></span>

<h3>Description</h3>

<p>Download annotation resources to a directory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download.annotation.resource(outputDirectory)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="download.annotation.resource_+3A_outputdirectory">outputDirectory</code></td>
<td>
<p>the directory to store annotation resources</p>
</td></tr>
</table>


<h3>Value</h3>

<p>will not return anything
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
download.annotation.resource("/tmp")

## End(Not run)
</code></pre>

<hr>
<h2 id='getCovPair'>Extract pair of positions by ranges</h2><span id='topic+getCovPair'></span>

<h3>Description</h3>

<p>Extract pair of positions by ranges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCovPair(covData, rangeList1, rangeList2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCovPair_+3A_covdata">covData</code></td>
<td>
<p>a covariance matrix with positions as dimnames</p>
</td></tr>
<tr><td><code id="getCovPair_+3A_rangelist1">rangeList1</code></td>
<td>
<p>character specify a range, 1-based index</p>
</td></tr>
<tr><td><code id="getCovPair_+3A_rangelist2">rangeList2</code></td>
<td>
<p>character specify a range, 1-based index</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a covariance matrix
covFileName = system.file(&quot;rvtests/rvtest.MetaCov.assoc.gz&quot;, package = &quot;seqminer&quot;)
cfh &lt;- rvmeta.readCovByRange(covFileName, &quot;1:196621007-196716634&quot;)
rangeList1 &lt;- &quot;1:196621007-196700000&quot;
rangeList2 &lt;- &quot;1:196700000-196716634&quot;
getCovPair(cfh, rangeList1, rangeList2)
</p>

<hr>
<h2 id='getRefBase'>Annotate a test variant</h2><span id='topic+getRefBase'></span>

<h3>Description</h3>

<p>Annotate a test variant
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRefBase(reference, chrom, position, len = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getRefBase_+3A_reference">reference</code></td>
<td>
<p>path to the reference genome file (.fa file)</p>
</td></tr>
<tr><td><code id="getRefBase_+3A_chrom">chrom</code></td>
<td>
<p>a vector of chromosome names</p>
</td></tr>
<tr><td><code id="getRefBase_+3A_position">position</code></td>
<td>
<p>a vector of chromosome positions</p>
</td></tr>
<tr><td><code id="getRefBase_+3A_len">len</code></td>
<td>
<p>a vector of length</p>
</td></tr>
</table>


<h3>Value</h3>

<p>based extracted from the reference genome
</p>

<hr>
<h2 id='hasIndex'>Check input file has tabix index</h2><span id='topic+hasIndex'></span>

<h3>Description</h3>

<p>Check input file has tabix index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hasIndex(fileName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hasIndex_+3A_filename">fileName</code></td>
<td>
<p>character vector, representing file names an input file name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if an index file with &quot;.tbi&quot; or &quot;bci&quot; exists, and created later than VCF/BCF file
</p>

<hr>
<h2 id='isDirWritable'>Test whether directory is writable</h2><span id='topic+isDirWritable'></span>

<h3>Description</h3>

<p>Test whether directory is writable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isDirWritable(outDir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isDirWritable_+3A_outdir">outDir</code></td>
<td>
<p>the name of the directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the file is writable
isDirWritable(&quot;~&quot;)
</p>

<hr>
<h2 id='isInRange'>Test whether a vector of positions are inside given ranges</h2><span id='topic+isInRange'></span>

<h3>Description</h3>

<p>Test whether a vector of positions are inside given ranges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isInRange(positions, rangeList)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isInRange_+3A_positions">positions</code></td>
<td>
<p>characters, positions. e.g. c(&quot;1:2-3&quot;, &quot;1:4&quot;)</p>
</td></tr>
<tr><td><code id="isInRange_+3A_rangelist">rangeList</code></td>
<td>
<p>character, ranges, e.g. &quot;1:1-3,1:2-4&quot;, 1-based index</p>
</td></tr>
</table>


<h3>Value</h3>

<p>logical vector, TRUE/FALSE/NA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>positions &lt;- c("1:2-3", "1:4", "XX")
ranges &lt;- "1:1-3,1:2-4,1:5-10"
isInRange(positions, ranges)
</code></pre>

<hr>
<h2 id='isTabixRange'>Check if the inputs are valid tabix range such as chr1:2-300</h2><span id='topic+isTabixRange'></span>

<h3>Description</h3>

<p>Check if the inputs are valid tabix range such as chr1:2-300
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isTabixRange(range)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isTabixRange_+3A_range">range</code></td>
<td>
<p>characer vector</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>valid &lt;- isTabixRange(c("chr1:1-200", "X:1", "1:100-100", "chr1", "1:1-20,1:30-40"))
stopifnot(all(valid))
invalid &lt;- isTabixRange(c(":1", "chr1::", ":-"))
stopifnot(all(!invalid))
</code></pre>

<hr>
<h2 id='isURL'>Check if the input is url e.g. http:// or ftp://</h2><span id='topic+isURL'></span>

<h3>Description</h3>

<p>Check if the input is url e.g. http:// or ftp://
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isURL(fileName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isURL_+3A_filename">fileName</code></td>
<td>
<p>character vector</p>
</td></tr>
</table>

<hr>
<h2 id='makeAnnotationParameter'>Construct a usable set of annotation parameters</h2><span id='topic+makeAnnotationParameter'></span>

<h3>Description</h3>

<p>Construct a usable set of annotation parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeAnnotationParameter(param = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makeAnnotationParameter_+3A_param">param</code></td>
<td>
<p>a list of annotation elements</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list, a complete list of supported parameters
</p>

<hr>
<h2 id='newJob'>Create a new job</h2><span id='topic+newJob'></span>

<h3>Description</h3>

<p>Create a new job
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newJob(id, cmd, outFile, depend = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="newJob_+3A_id">id</code></td>
<td>
<p>character, job ids.</p>
</td></tr>
<tr><td><code id="newJob_+3A_cmd">cmd</code></td>
<td>
<p>character, commands to run</p>
</td></tr>
<tr><td><code id="newJob_+3A_outfile">outFile</code></td>
<td>
<p>character, the output file names after command are run successfully</p>
</td></tr>
<tr><td><code id="newJob_+3A_depend">depend</code></td>
<td>
<p>character vector, specify the prerequisite files (e.g. outFile from other jobs)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>j1 &lt;- newJob('id1', 'cmd out1', 'out1')
j2 &lt;- newJob('id2', 'cmd out2', 'out2', depend = 'out1')
</code></pre>

<hr>
<h2 id='newWorkflow'>Create a new workflow</h2><span id='topic+newWorkflow'></span>

<h3>Description</h3>

<p>Create a new workflow
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newWorkflow(name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="newWorkflow_+3A_name">name</code></td>
<td>
<p>character, specify the name of the workflow</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>w &lt;- newWorkflow("wf")
</code></pre>

<hr>
<h2 id='openPlink'>Open binary PLINK files</h2><span id='topic+openPlink'></span>

<h3>Description</h3>

<p>Open binary PLINK files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>openPlink(fileName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="openPlink_+3A_filename">fileName</code></td>
<td>
<p>character, represents the prefix of PLINK input file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an PLINK file object with class name (&quot;PlinkFile&quot;)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("plink/all.anno.filtered.extract.bed", package = "seqminer")
fileName = sub(fileName, pattern = ".bed", replacement = "")
plinkObj &lt;- openPlink(fileName)
str(plinkObj)
</code></pre>

<hr>
<h2 id='readBGENToListByGene'>Read information from BGEN file in a given range and return a list</h2><span id='topic+readBGENToListByGene'></span>

<h3>Description</h3>

<p>Read information from BGEN file in a given range and return a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBGENToListByGene(fileName, geneFile, geneName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readBGENToListByGene_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input BGEN file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readBGENToListByGene_+3A_genefile">geneFile</code></td>
<td>
<p>character, a text file listing all genes in refFlat format</p>
</td></tr>
<tr><td><code id="readBGENToListByGene_+3A_genename">geneName</code></td>
<td>
<p>character vector, which gene(s) to be extracted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of chrom, pos, varid, rsid, alleles, isPhased, probability, sampleId
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("bgen/all.anno.filtered.extract.bgen", package = "seqminer")
geneFile = system.file("vcf/refFlat_hg19_6col.txt.gz", package = "seqminer")
cfh &lt;- readBGENToListByGene(fileName, geneFile, "CFH")
</code></pre>

<hr>
<h2 id='readBGENToListByRange'>Read information from BGEN file in a given range and return a list</h2><span id='topic+readBGENToListByRange'></span>

<h3>Description</h3>

<p>Read information from BGEN file in a given range and return a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBGENToListByRange(fileName, range)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readBGENToListByRange_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input BGEN file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readBGENToListByRange_+3A_range">range</code></td>
<td>
<p>character, a text indicating which range in the BGEN file to extract. e.g. 1:100-200, 1-based index</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of chrom, pos, varid, rsid, alleles, isPhased, probability, sampleId
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("bgen/all.anno.filtered.extract.bgen", package = "seqminer")
cfh &lt;- readBGENToListByRange(fileName, "1:196621007-196716634")
</code></pre>

<hr>
<h2 id='readBGENToMatrixByGene'>Read a gene from BGEN file and return a genotype matrix</h2><span id='topic+readBGENToMatrixByGene'></span>

<h3>Description</h3>

<p>Read a gene from BGEN file and return a genotype matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBGENToMatrixByGene(fileName, geneFile, geneName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readBGENToMatrixByGene_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input BGEN file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readBGENToMatrixByGene_+3A_genefile">geneFile</code></td>
<td>
<p>character, a text file listing all genes in refFlat format</p>
</td></tr>
<tr><td><code id="readBGENToMatrixByGene_+3A_genename">geneName</code></td>
<td>
<p>character vector, which gene(s) to be extracted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype matrix
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("bgen/all.anno.filtered.extract.bgen", package = "seqminer")
geneFile = system.file("vcf/refFlat_hg19_6col.txt.gz", package = "seqminer")
cfh &lt;- readBGENToMatrixByGene(fileName, geneFile, "CFH")
</code></pre>

<hr>
<h2 id='readBGENToMatrixByRange'>Read a gene from BGEN file and return a genotype matrix</h2><span id='topic+readBGENToMatrixByRange'></span>

<h3>Description</h3>

<p>Read a gene from BGEN file and return a genotype matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBGENToMatrixByRange(fileName, range)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readBGENToMatrixByRange_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input BGEN file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readBGENToMatrixByRange_+3A_range">range</code></td>
<td>
<p>character, a text indicating which range in the BGEN file to extract. e.g. 1:100-200, 1-based index</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype matrix
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("bgen/all.anno.filtered.extract.bgen", package = "seqminer")
cfh &lt;- readBGENToMatrixByRange(fileName, "1:196621007-196716634")
</code></pre>

<hr>
<h2 id='readPlinkToMatrixByIndex'>Read from binary PLINK file and return a genotype matrix</h2><span id='topic+readPlinkToMatrixByIndex'></span>

<h3>Description</h3>

<p>Read from binary PLINK file and return a genotype matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readPlinkToMatrixByIndex(plinkFilePrefix, sampleIndex, markerIndex)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readPlinkToMatrixByIndex_+3A_plinkfileprefix">plinkFilePrefix</code></td>
<td>
<p>a PlinkFileObject obtained by openPlink()</p>
</td></tr>
<tr><td><code id="readPlinkToMatrixByIndex_+3A_sampleindex">sampleIndex</code></td>
<td>
<p>integer, 1-basd, index of samples to be extracted</p>
</td></tr>
<tr><td><code id="readPlinkToMatrixByIndex_+3A_markerindex">markerIndex</code></td>
<td>
<p>integer, 1-basd, index of markers to be extracted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype matrix, marker by sample
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## these indice are nonsynonymous markers for 1:196621007-196716634",
## refer to the readVCFToMatrixByRange()
fileName = system.file("plink/all.anno.filtered.extract.bed", package = "seqminer")
fileName = sub(fileName, pattern = ".bed", replacement = "")
sampleIndex = seq(3)
markerIndex =c(14, 36)
cfh &lt;- readPlinkToMatrixByIndex(fileName, sampleIndex, markerIndex)
</code></pre>

<hr>
<h2 id='readSingleChromosomeBCFToMatrixByRange'>Read a range from BCF file and return a genotype matrix</h2><span id='topic+readSingleChromosomeBCFToMatrixByRange'></span>

<h3>Description</h3>

<p>Read a range from BCF file and return a genotype matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readSingleChromosomeBCFToMatrixByRange(fileName, range, indexFileName = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readSingleChromosomeBCFToMatrixByRange_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input BCF file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readSingleChromosomeBCFToMatrixByRange_+3A_range">range</code></td>
<td>
<p>character, a text indicating which range in the BCF file to extract. e.g. 1:100-200, 1-based index</p>
</td></tr>
<tr><td><code id="readSingleChromosomeBCFToMatrixByRange_+3A_indexfilename">indexFileName</code></td>
<td>
<p>character, index file, by default, it s 'fileName'.scIdx</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype matrix
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.headerFixed.bcf.gz", package = "seqminer")
cfh &lt;- readSingleChromosomeBCFToMatrixByRange(fileName, "1:196621007-196716634")
</code></pre>

<hr>
<h2 id='readSingleChromosomeVCFToMatrixByRange'>Read a range from VCF file and return a genotype matrix</h2><span id='topic+readSingleChromosomeVCFToMatrixByRange'></span>

<h3>Description</h3>

<p>Read a range from VCF file and return a genotype matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readSingleChromosomeVCFToMatrixByRange(fileName, range, indexFileName = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readSingleChromosomeVCFToMatrixByRange_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input VCF file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readSingleChromosomeVCFToMatrixByRange_+3A_range">range</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200, 1-based index</p>
</td></tr>
<tr><td><code id="readSingleChromosomeVCFToMatrixByRange_+3A_indexfilename">indexFileName</code></td>
<td>
<p>character, index file, by default, it s 'fileName'.scIdx</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype matrix
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
cfh &lt;- readSingleChromosomeVCFToMatrixByRange(fileName, "1:196621007-196716634")
</code></pre>

<hr>
<h2 id='readVCFToListByGene'>Read information from VCF file in a given range and return a list</h2><span id='topic+readVCFToListByGene'></span>

<h3>Description</h3>

<p>Read information from VCF file in a given range and return a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVCFToListByGene(
  fileName,
  geneFile,
  geneName,
  annoType,
  vcfColumn,
  vcfInfo,
  vcfIndv
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readVCFToListByGene_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input VCF file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readVCFToListByGene_+3A_genefile">geneFile</code></td>
<td>
<p>character, a text file listing all genes in refFlat format</p>
</td></tr>
<tr><td><code id="readVCFToListByGene_+3A_genename">geneName</code></td>
<td>
<p>character vector, which gene(s) to be extracted</p>
</td></tr>
<tr><td><code id="readVCFToListByGene_+3A_annotype">annoType</code></td>
<td>
<p>character, annotated types you would like to extract, such as &quot;Nonsynonymous&quot;, &quot;Synonymous&quot;. This can be left empty.</p>
</td></tr>
<tr><td><code id="readVCFToListByGene_+3A_vcfcolumn">vcfColumn</code></td>
<td>
<p>character vector, which vcf columns to extract. It can be chosen from CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO, FORMAT and etc.</p>
</td></tr>
<tr><td><code id="readVCFToListByGene_+3A_vcfinfo">vcfInfo</code></td>
<td>
<p>character vector, which should be tags in the INFO columns to extarct. Common choices include: DP, AC, AF, NS</p>
</td></tr>
<tr><td><code id="readVCFToListByGene_+3A_vcfindv">vcfIndv</code></td>
<td>
<p>character vector, which values to extract at individual level. Common choices are: GT, GQ, GD</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of genes, and each elements has specified vcfColumn, vcfinfo, vcfIndv
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
geneFile = system.file("vcf/refFlat_hg19_6col.txt.gz", package = "seqminer")
cfh &lt;- readVCFToListByGene(fileName, geneFile, "CFH", "Synonymous",
                           c("CHROM", "POS"), c("AF", "AC"), c("GT") )
</code></pre>

<hr>
<h2 id='readVCFToListByRange'>Read information from VCF file in a given range and return a list</h2><span id='topic+readVCFToListByRange'></span>

<h3>Description</h3>

<p>Read information from VCF file in a given range and return a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVCFToListByRange(fileName, range, annoType, vcfColumn, vcfInfo, vcfIndv)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readVCFToListByRange_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input VCF file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readVCFToListByRange_+3A_range">range</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200, 1-based index</p>
</td></tr>
<tr><td><code id="readVCFToListByRange_+3A_annotype">annoType</code></td>
<td>
<p>character, annotated types you would like to extract, such as &quot;Nonsynonymous&quot;, &quot;Synonymous&quot;. This can be left empty.</p>
</td></tr>
<tr><td><code id="readVCFToListByRange_+3A_vcfcolumn">vcfColumn</code></td>
<td>
<p>character vector, which vcf columns to extract. It can be chosen from CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO, FORMAT and etc.</p>
</td></tr>
<tr><td><code id="readVCFToListByRange_+3A_vcfinfo">vcfInfo</code></td>
<td>
<p>character vector, which should be tags in the INFO columns to extarct. Common choices include: DP, AC, AF, NS</p>
</td></tr>
<tr><td><code id="readVCFToListByRange_+3A_vcfindv">vcfIndv</code></td>
<td>
<p>character vector, which values to extract at individual level. Common choices are: GT, GQ, GD</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of genes, and each elements has specified vcfColumn, vcfinfo, vcfIndv
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
cfh &lt;- readVCFToListByRange(fileName, "1:196621007-196716634", "Nonsynonymous",
                            c("CHROM", "POS"), c("AF", "AC"), c("GT") )
</code></pre>

<hr>
<h2 id='readVCFToMatrixByGene'>Read a gene from VCF file and return a genotype matrix</h2><span id='topic+readVCFToMatrixByGene'></span>

<h3>Description</h3>

<p>Read a gene from VCF file and return a genotype matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVCFToMatrixByGene(fileName, geneFile, geneName, annoType)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readVCFToMatrixByGene_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input VCF file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readVCFToMatrixByGene_+3A_genefile">geneFile</code></td>
<td>
<p>character, a text file listing all genes in refFlat format</p>
</td></tr>
<tr><td><code id="readVCFToMatrixByGene_+3A_genename">geneName</code></td>
<td>
<p>character vector, which gene(s) to be extracted</p>
</td></tr>
<tr><td><code id="readVCFToMatrixByGene_+3A_annotype">annoType</code></td>
<td>
<p>character, annotated types you would like to extract, such as &quot;Nonsynonymous&quot;, &quot;Synonymous&quot;. This can be left empty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype matrix
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
geneFile = system.file("vcf/refFlat_hg19_6col.txt.gz", package = "seqminer")
cfh &lt;- readVCFToMatrixByGene(fileName, geneFile, "CFH", "Synonymous")
</code></pre>

<hr>
<h2 id='readVCFToMatrixByRange'>Read a gene from VCF file and return a genotype matrix</h2><span id='topic+readVCFToMatrixByRange'></span>

<h3>Description</h3>

<p>Read a gene from VCF file and return a genotype matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVCFToMatrixByRange(fileName, range, annoType)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readVCFToMatrixByRange_+3A_filename">fileName</code></td>
<td>
<p>character, represents an input VCF file (Bgzipped, with Tabix index)</p>
</td></tr>
<tr><td><code id="readVCFToMatrixByRange_+3A_range">range</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200, 1-based index</p>
</td></tr>
<tr><td><code id="readVCFToMatrixByRange_+3A_annotype">annoType</code></td>
<td>
<p>character, annotated types you would like to extract, such as &quot;Nonsynonymous&quot;, &quot;Synonymous&quot;. This can be left empty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype matrix
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
cfh &lt;- readVCFToMatrixByRange(fileName, "1:196621007-196716634", "Nonsynonymous")
</code></pre>

<hr>
<h2 id='rvmeta.readCovByRange'>Read covariance by range from METAL-format files.</h2><span id='topic+rvmeta.readCovByRange'></span>

<h3>Description</h3>

<p>Read covariance by range from METAL-format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvmeta.readCovByRange(covFile, tabixRange)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rvmeta.readCovByRange_+3A_covfile">covFile</code></td>
<td>
<p>character, a covariance file (rvtests outputs using &ndash;meta cov)</p>
</td></tr>
<tr><td><code id="rvmeta.readCovByRange_+3A_tabixrange">tabixRange</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of covariance within given range
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>covFileName = system.file("rvtests/rvtest.MetaCov.assoc.gz", package = "seqminer")
cfh &lt;- rvmeta.readCovByRange(covFileName, "1:196621007-196716634")
</code></pre>

<hr>
<h2 id='rvmeta.readDataByGene'>Read association statistics by gene from METAL-format files. Both score statistics and covariance statistics will be extracted.</h2><span id='topic+rvmeta.readDataByGene'></span>

<h3>Description</h3>

<p>Read association statistics by gene from METAL-format files. Both score statistics and covariance statistics will be extracted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvmeta.readDataByGene(
  scoreTestFiles,
  covFiles,
  geneFile,
  geneName,
  multiAllelic = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rvmeta.readDataByGene_+3A_scoretestfiles">scoreTestFiles</code></td>
<td>
<p>character vector, score test output files (rvtests outputs using &ndash;meta score)</p>
</td></tr>
<tr><td><code id="rvmeta.readDataByGene_+3A_covfiles">covFiles</code></td>
<td>
<p>character vector, covaraite files (rvtests outputs using &ndash;meta cov)</p>
</td></tr>
<tr><td><code id="rvmeta.readDataByGene_+3A_genefile">geneFile</code></td>
<td>
<p>character, a text file listing all genes in refFlat format</p>
</td></tr>
<tr><td><code id="rvmeta.readDataByGene_+3A_genename">geneName</code></td>
<td>
<p>character vector, which gene(s) to be extracted</p>
</td></tr>
<tr><td><code id="rvmeta.readDataByGene_+3A_multiallelic">multiAllelic</code></td>
<td>
<p>boolean, whether to read multi-allelic sites as multiple variants or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of statistics including chromosome, position, allele frequency, score statistics, covariance and annotation(if input files are annotated).
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scoreFileName = system.file("rvtests/rvtest.MetaScore.assoc.anno.gz", package = "seqminer")
covFileName = system.file("rvtests/rvtest.MetaCov.assoc.gz", package = "seqminer")
geneFile = system.file("vcf/refFlat_hg19_6col.txt.gz", package = "seqminer")
cfh &lt;- rvmeta.readDataByGene(scoreFileName, covFileName, geneFile, "CFH")
</code></pre>

<hr>
<h2 id='rvmeta.readDataByRange'>Read association statistics by range from METAL-format files. Both score statistics and covariance statistics will be extracted.</h2><span id='topic+rvmeta.readDataByRange'></span>

<h3>Description</h3>

<p>Read association statistics by range from METAL-format files. Both score statistics and covariance statistics will be extracted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvmeta.readDataByRange(scoreTestFiles, covFiles, ranges, multiAllelic = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rvmeta.readDataByRange_+3A_scoretestfiles">scoreTestFiles</code></td>
<td>
<p>character vector, score test output files (rvtests outputs using &ndash;meta score)</p>
</td></tr>
<tr><td><code id="rvmeta.readDataByRange_+3A_covfiles">covFiles</code></td>
<td>
<p>character vector, covaraite files (rvtests outputs using &ndash;meta cov)</p>
</td></tr>
<tr><td><code id="rvmeta.readDataByRange_+3A_ranges">ranges</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200, 1-based index</p>
</td></tr>
<tr><td><code id="rvmeta.readDataByRange_+3A_multiallelic">multiAllelic</code></td>
<td>
<p>boolean, whether to read multi-allelic sites as multiple variants or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of statistics including chromosome, position, allele frequency, score statistics, covariance and annotation(if input files are annotated).
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scoreFileName = system.file("rvtests/rvtest.MetaScore.assoc.anno.gz", package = "seqminer")
covFileName = system.file("rvtests/rvtest.MetaCov.assoc.gz", package = "seqminer")
geneFile = system.file("vcf/refFlat_hg19_6col.txt.gz", package = "seqminer")
cfh &lt;- rvmeta.readDataByRange(scoreFileName, covFileName, "1:196621007-196716634")
</code></pre>

<hr>
<h2 id='rvmeta.readNullModel'>Read null model statistics</h2><span id='topic+rvmeta.readNullModel'></span>

<h3>Description</h3>

<p>Read null model statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvmeta.readNullModel(scoreTestFiles)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rvmeta.readNullModel_+3A_scoretestfiles">scoreTestFiles</code></td>
<td>
<p>character vector, score test output files (rvtests outputs using &ndash;meta score)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of statistics fitted under the null mode (without genetic effects)
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scoreFileName = system.file("rvtests/rvtest.MetaScore.assoc.anno.gz", package = "seqminer")
</code></pre>

<hr>
<h2 id='rvmeta.readScoreByRange'>Read score test statistics by range from METAL-format files.</h2><span id='topic+rvmeta.readScoreByRange'></span>

<h3>Description</h3>

<p>Read score test statistics by range from METAL-format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvmeta.readScoreByRange(scoreTestFiles, tabixRange)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rvmeta.readScoreByRange_+3A_scoretestfiles">scoreTestFiles</code></td>
<td>
<p>character vector, score test output files (rvtests outputs using &ndash;meta score)</p>
</td></tr>
<tr><td><code id="rvmeta.readScoreByRange_+3A_tabixrange">tabixRange</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200</p>
</td></tr>
</table>


<h3>Value</h3>

<p>score test statistics within given range
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scoreFileName = system.file("rvtests/rvtest.MetaScore.assoc.anno.gz", package = "seqminer")
cfh &lt;- rvmeta.readScoreByRange(scoreFileName, "1:196621007-196716634")
</code></pre>

<hr>
<h2 id='rvmeta.readSkewByRange'>Read skew by range from METAL-format files.</h2><span id='topic+rvmeta.readSkewByRange'></span>

<h3>Description</h3>

<p>Read skew by range from METAL-format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvmeta.readSkewByRange(skewFile, tabixRange)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rvmeta.readSkewByRange_+3A_skewfile">skewFile</code></td>
<td>
<p>character, a skew file (rvtests outputs using &ndash;meta skew)</p>
</td></tr>
<tr><td><code id="rvmeta.readSkewByRange_+3A_tabixrange">tabixRange</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an 3-dimensional array of skewness within given range
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>skewFileName = system.file("rvtests/rvtest.MetaSkew.assoc.gz", package = "seqminer")
cfh &lt;- rvmeta.readSkewByRange(skewFileName, "1:196621007-196716634")
</code></pre>

<hr>
<h2 id='rvmeta.writeCovData'>Write covariance association statistics files.</h2><span id='topic+rvmeta.writeCovData'></span>

<h3>Description</h3>

<p>Write covariance association statistics files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvmeta.writeCovData(rvmetaData, outName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rvmeta.writeCovData_+3A_rvmetadata">rvmetaData</code></td>
<td>
<p>a list vector. It's usually read by rvmeta.readDataByRange or rvmeta.readDataByGene function</p>
</td></tr>
<tr><td><code id="rvmeta.writeCovData_+3A_outname">outName</code></td>
<td>
<p>character, a text indicating output file prefix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE only if succeed
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scoreFileName = system.file("rvtests/rvtest.MetaScore.assoc.anno.gz", package = "seqminer")
covFileName = system.file("rvtests/rvtest.MetaCov.assoc.gz", package = "seqminer")
geneFile = system.file("vcf/refFlat_hg19_6col.txt.gz", package = "seqminer")
cfh &lt;- rvmeta.readDataByRange(scoreFileName, covFileName, "1:196621007-196716634")

outFile &lt;- file.path(tempdir(), "cfh.MetaCov.assoc.gz")
rvmeta.writeCovData(cfh, outFile)
cat('Outputted MetaCov file are in the temp directory:', outFile, '\n')
</code></pre>

<hr>
<h2 id='rvmeta.writeScoreData'>Write score-based association statistics files.</h2><span id='topic+rvmeta.writeScoreData'></span>

<h3>Description</h3>

<p>Write score-based association statistics files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvmeta.writeScoreData(rvmetaData, outName, createIndex = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rvmeta.writeScoreData_+3A_rvmetadata">rvmetaData</code></td>
<td>
<p>a list vector. It's usually read by rvmeta.readDataByRange or rvmeta.readDataByGene function</p>
</td></tr>
<tr><td><code id="rvmeta.writeScoreData_+3A_outname">outName</code></td>
<td>
<p>character, a text indicating output file prefix</p>
</td></tr>
<tr><td><code id="rvmeta.writeScoreData_+3A_createindex">createIndex</code></td>
<td>
<p>boolean, (default FALSE), whether or not to create the index</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE only if succeed
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scoreFileName = system.file("rvtests/rvtest.MetaScore.assoc.anno.gz", package = "seqminer")
covFileName = system.file("rvtests/rvtest.MetaCov.assoc.gz", package = "seqminer")
geneFile = system.file("vcf/refFlat_hg19_6col.txt.gz", package = "seqminer")
cfh &lt;- rvmeta.readDataByRange(scoreFileName, covFileName, "1:196621007-196716634")

outFile &lt;- file.path(tempdir(), "cfh.MetaScore.assoc")
rvmeta.writeScoreData(cfh, outFile)
cat('Outputted MetaScore file are in the temp directory:', outFile, '\n')
</code></pre>

<hr>
<h2 id='tabix.createIndex'>Create tabix index file, similar to running tabix in command line.</h2><span id='topic+tabix.createIndex'></span>

<h3>Description</h3>

<p>Create tabix index file, similar to running tabix in command line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tabix.createIndex(
  bgzipFile,
  sequenceColumn = 1,
  startColumn = 4,
  endColumn = 5,
  metaChar = "#",
  skipLines = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tabix.createIndex_+3A_bgzipfile">bgzipFile</code></td>
<td>
<p>character, an tabix indexed file</p>
</td></tr>
<tr><td><code id="tabix.createIndex_+3A_sequencecolumn">sequenceColumn</code></td>
<td>
<p>integer, sequence name column</p>
</td></tr>
<tr><td><code id="tabix.createIndex_+3A_startcolumn">startColumn</code></td>
<td>
<p>integer, start column</p>
</td></tr>
<tr><td><code id="tabix.createIndex_+3A_endcolumn">endColumn</code></td>
<td>
<p>integer, end column</p>
</td></tr>
<tr><td><code id="tabix.createIndex_+3A_metachar">metaChar</code></td>
<td>
<p>character, symbol for comment/meta lines</p>
</td></tr>
<tr><td><code id="tabix.createIndex_+3A_skiplines">skipLines</code></td>
<td>
<p>integer, first this number of lines will be skipped</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
tabix.createIndex(fileName, 1, 2, 0, '#', 0)
</code></pre>

<hr>
<h2 id='tabix.createIndex.meta'>Create tabix index for bgzipped MetaScore/MetaCov file</h2><span id='topic+tabix.createIndex.meta'></span>

<h3>Description</h3>

<p>Create tabix index for bgzipped MetaScore/MetaCov file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tabix.createIndex.meta(bgzipFile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tabix.createIndex.meta_+3A_bgzipfile">bgzipFile</code></td>
<td>
<p>character, input vcf file</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>
<p>http://zhanxw.github.io/rvtests/ for rvtests
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("rvtests/rvtest.MetaScore.assoc.anno.gz", package = "seqminer")
tabix.createIndex.meta(fileName)
</code></pre>

<hr>
<h2 id='tabix.createIndex.vcf'>Create tabix index for bgzipped VCF file</h2><span id='topic+tabix.createIndex.vcf'></span>

<h3>Description</h3>

<p>Create tabix index for bgzipped VCF file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tabix.createIndex.vcf(bgzipVcfFile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tabix.createIndex.vcf_+3A_bgzipvcffile">bgzipVcfFile</code></td>
<td>
<p>character, input vcf file</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
tabix.createIndex.vcf(fileName)
</code></pre>

<hr>
<h2 id='tabix.read'>Read tabix file, similar to running tabix in command line.</h2><span id='topic+tabix.read'></span>

<h3>Description</h3>

<p>Read tabix file, similar to running tabix in command line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tabix.read(tabixFile, tabixRange)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tabix.read_+3A_tabixfile">tabixFile</code></td>
<td>
<p>character, an tabix indexed file</p>
</td></tr>
<tr><td><code id="tabix.read_+3A_tabixrange">tabixRange</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector, each elements is an individual line
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (.Platform$endian == "little") {
  fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
  snp &lt;- tabix.read(fileName, "1:196623337-196632470")
} else {
  message("Tabix does not work well for big endian for now")
}
</code></pre>

<hr>
<h2 id='tabix.read.header'>Read tabix file, similar to running tabix in command line.</h2><span id='topic+tabix.read.header'></span>

<h3>Description</h3>

<p>Read tabix file, similar to running tabix in command line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tabix.read.header(tabixFile, skippedLine = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tabix.read.header_+3A_tabixfile">tabixFile</code></td>
<td>
<p>character, an tabix indexed file</p>
</td></tr>
<tr><td><code id="tabix.read.header_+3A_skippedline">skippedLine</code></td>
<td>
<p>logical, whether to read tabix skipped lines (when used 'tabix -S NUM')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
snp &lt;- tabix.read.header(fileName)
</code></pre>

<hr>
<h2 id='tabix.read.table'>Read tabix file, similar to running tabix in command line.</h2><span id='topic+tabix.read.table'></span>

<h3>Description</h3>

<p>Read tabix file, similar to running tabix in command line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tabix.read.table(
  tabixFile,
  tabixRange,
  col.names = TRUE,
  stringsAsFactors = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tabix.read.table_+3A_tabixfile">tabixFile</code></td>
<td>
<p>character, an tabix indexed file</p>
</td></tr>
<tr><td><code id="tabix.read.table_+3A_tabixrange">tabixRange</code></td>
<td>
<p>character, a text indicating which range in the VCF file to extract. e.g. 1:100-200</p>
</td></tr>
<tr><td><code id="tabix.read.table_+3A_col.names">col.names</code></td>
<td>
<p>logical, use tabix file header as result headers (default: TRUE)</p>
</td></tr>
<tr><td><code id="tabix.read.table_+3A_stringsasfactors">stringsAsFactors</code></td>
<td>
<p>logical, store loaded data as factors (default: FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame, each elements is an individual line
</p>


<h3>See Also</h3>

<p>http://zhanxw.com/seqminer/ for online manual and examples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fileName = system.file("vcf/all.anno.filtered.extract.vcf.gz", package = "seqminer")
snp &lt;- tabix.read.table(fileName, "1:196623337-196632470")
</code></pre>

<hr>
<h2 id='validateAnnotationParameter'>Validate annotate parameter is valid</h2><span id='topic+validateAnnotationParameter'></span>

<h3>Description</h3>

<p>Validate annotate parameter is valid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validateAnnotationParameter(param, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validateAnnotationParameter_+3A_param">param</code></td>
<td>
<p>a list of annotation elements</p>
</td></tr>
<tr><td><code id="validateAnnotationParameter_+3A_debug">debug</code></td>
<td>
<p>show extra debug information or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list, first element is TRUE/FALSE if parameter is valid/invalid;
</p>

<hr>
<h2 id='verifyFilename'>validate the inVcf can be created, and outVcf can be write to.
will stop if any error occurs</h2><span id='topic+verifyFilename'></span>

<h3>Description</h3>

<p>validate the inVcf can be created, and outVcf can be write to.
will stop if any error occurs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verifyFilename(inVcf, outVcf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verifyFilename_+3A_invcf">inVcf</code></td>
<td>
<p>input file</p>
</td></tr>
<tr><td><code id="verifyFilename_+3A_outvcf">outVcf</code></td>
<td>
<p>output file</p>
</td></tr>
</table>

<hr>
<h2 id='writeWorkflow'>Export workflow to Makefile</h2><span id='topic+writeWorkflow'></span>

<h3>Description</h3>

<p>Export workflow to Makefile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeWorkflow(wf, outFile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeWorkflow_+3A_wf">wf</code></td>
<td>
<p>a variable workflow class</p>
</td></tr>
<tr><td><code id="writeWorkflow_+3A_outfile">outFile</code></td>
<td>
<p>character, typically named &quot;Makefile&quot;</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>j1 &lt;- newJob('id1', 'cmd out1', 'out1')
j2 &lt;- newJob('id2', 'cmd out2', 'out2', depend = 'out1')
w &lt;- newWorkflow("wf")
w &lt;- addJob(w, j1)
w &lt;- addJob(w, j2)

outFile &lt;-  file.path(tempdir(), "Makefile")
writeWorkflow(w, outFile)
cat('Outputted Makefile file are in the temp directory:', outFile, '\n')
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
