<!DOCTYPE html><html><head><title>Help for package SeqMADE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SeqMADE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SeqMADE-package'><p>Network Module-Based Model in the Differential Expression Analysis for RNA-Seq</p></a></li>
<li><a href='#exprs'><p>Gene Expression Dataset</p></a></li>
<li><a href='#Factor'>
<p>Construction of Variable Factors</p></a></li>
<li><a href='#moduleMatrix'><p>Modulematrix Construction</p></a></li>
<li><a href='#nbGLM'><p>Identify Differential Expression Modules Based on the Generalized Linear Model</p></a></li>
<li><a href='#nbGLMdir'><p>Identify Differential Expression Modules Based on the GLM Model with Up or Down-regulated Change</p></a></li>
<li><a href='#nbGLMdirperm'><p>Identify Differential Expression Modules Based on the GLM Method by Shuffling the Phenotypic Variables</p></a></li>
<li><a href='#networkModule'><p>NetworkModule</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Network Module-Based Model in the Differential Expression
Analysis for RNA-Seq</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-06-27</td>
</tr>
<tr>
<td>Author:</td>
<td>Mingli Lei, Jia Xu, Li-Ching Huang, Lily Wang, Jing Li</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mingli Lei&lt;leimingli2013@sjtu.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A network module-based generalized linear model for differential expression analysis with the count-based sequence data from RNA-Seq.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.1.0),MASS</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-06-27 03:06:53 UTC; Mingli Lei</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-06-27 06:59:54</td>
</tr>
</table>
<hr>
<h2 id='SeqMADE-package'>Network Module-Based Model in the Differential Expression Analysis for RNA-Seq</h2><span id='topic+SeqMADE-package'></span><span id='topic+SeqMADE'></span>

<h3>Description</h3>

<p>A network module-based generalized linear model for differential expression analysis with the count-based sequence data from RNA-Seq.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> SeqMADE</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-06-27</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;2)</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The main functions in this package are
<code><a href="#topic+Factor">Factor</a></code>   A function of constructing the Group variables, Direction variables, and the Count variables,
<code><a href="#topic+moduleMatrix">moduleMatrix</a></code> a function of constructing the modulematrix for all the modules,
<code><a href="#topic+nbGLM">nbGLM</a></code> Identify differential expression modules based on the GLM method using Group and Module variables,
<code><a href="#topic+nbGLMdir">nbGLMdir</a></code> Identify differential expression modules based on the Generalized Linear Model(GLM) using Group, Module and Direction variables, and
<code><a href="#topic+nbGLMdirperm">nbGLMdirperm</a></code> Identify differential expression modules based on the GLM method by shuffling the phenotypic variables.
</p>


<h3>Author(s)</h3>

<p>Mingli Lei, Jia Xu, Li-Ching Huang, Lily Wang, Jing Li
Maintainer: Mingli Lei&lt;leimingli2013@sjtu.edu.cn&gt;
</p>


<h3>References</h3>

<p>Xu, J., Wang, L. and Li, J. (2014) Biological network module-based model for the analysis of differential expression in shotgun proteomics,
J Proteome Res, 13, 5743-5750.</p>


<h3>See Also</h3>

<p>glm(),lm()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exprs)
data(networkModule)
case &lt;- c("A1","A2","A3","A4","A5","A6","A7")
control &lt;- c("B1","B2","B3","B4","B5","B6","B7")
factors &lt;- Factor(exprs,case,control)
modulematrix &lt;- moduleMatrix(exprs,networkModule)
Result1&lt;- nbGLM(factors,14,networkModule,modulematrix,distribution="NB")
Result2&lt;- nbGLMdir(factors,14,networkModule,modulematrix,distribution="NB")
Result3&lt;- nbGLMdirperm(exprs,case,control,factors,networkModule,
                       modulematrix,10,distribution="NB")
</code></pre>

<hr>
<h2 id='exprs'>Gene Expression Dataset</h2><span id='topic+exprs'></span>

<h3>Description</h3>

<p>Gene expression dataset, containing 100 genes and 14 samples(7 case and 7 control respectively).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exprs)</code></pre>


<h3>Details</h3>

<p>In this dataset, there are 100 genes and 14 samples which consist of the expression dataset, in which 7 samples are in case groups and other 7 samples are in control groups.
</p>


<h3>Author(s)</h3>

<p>Mingli Lei </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exprs)
</code></pre>

<hr>
<h2 id='Factor'>
Construction of Variable Factors
</h2><span id='topic+Factor'></span>

<h3>Description</h3>

<p>A function of constructing the Group variables, Direction variables, and the Count variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Factor(exprs, case, control)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Factor_+3A_exprs">exprs</code></td>
<td>

<p>exprs is a data frame or matrix for two groups or conditions, with rows as variables (genes) and columns as samples.
</p>
</td></tr>
<tr><td><code id="Factor_+3A_case">case</code></td>
<td>

<p>case is the sample names in case groups.
</p>
</td></tr>
<tr><td><code id="Factor_+3A_control">control</code></td>
<td>

<p>control is the sample names in control groups.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two indicator variables Group and Direction corresponding to the different groups and the direction of the gene expression changes in the context of an RNA-Seq experiment, respectively.
And in this part, 1 represents that a gene belongs to case group or up-regulated and 0 represents a gene belongs to control group or down-regulated. Besides, Count variables are the 
expression value in different samples for genes.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Count</code></td>
<td>
<p>The gene expression count values.</p>
</td></tr>
<tr><td><code>Group</code></td>
<td>
<p>The indicator variables represent that whether a gene belongs to case group or not.</p>
</td></tr>
<tr><td><code>Direction</code></td>
<td>
<p>The indicator variables represent that a gene is up-regulated or down-regulated.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mingli Lei, Jia Xu, Li-Ching Huang, Lily Wang, Jing Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exprs)
case &lt;- c("A1","A2","A3","A4","A5","A6","A7")
control &lt;- c("B1","B2","B3","B4","B5","B6","B7")
factors &lt;- Factor(exprs, case, control) 
</code></pre>

<hr>
<h2 id='moduleMatrix'>Modulematrix Construction</h2><span id='topic+moduleMatrix'></span>

<h3>Description</h3>

<p>A function of constructing the modulematrix for the modules was used to indicate whether genes belong to a given module or not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moduleMatrix(exprs, networkModule)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="moduleMatrix_+3A_exprs">exprs</code></td>
<td>

<p>exprs is a data frame or matrix for two groups or conditions, with rows as variables (genes) and columns as samples
</p>
</td></tr>
<tr><td><code id="moduleMatrix_+3A_networkmodule">networkModule</code></td>
<td>

<p>NetworkModule is the gene sets or modules in the biological network or metabolic pathway, with the 1th column as the module names and the 2th columnn as the gene symbols constituting the module</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Modulematrix is a matrix, in which the indicator variables 1 or 0 represent whether a gene belong to a given module or not.
</p>


<h3>Author(s)</h3>

<p>Mingli Lei, Jia Xu, Li-Ching Huang, Lily Wang, Jing Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exprs)
data(networkModule)
modulematrix &lt;- moduleMatrix(exprs,networkModule)
</code></pre>

<hr>
<h2 id='nbGLM'>Identify Differential Expression Modules Based on the Generalized Linear Model</h2><span id='topic+nbGLM'></span>

<h3>Description</h3>

<p>The algorithm identify differential expression modules using Generalized Linear Model (GLM)
for differential expression analysis in RNA-Seq data, and in the model two indicator variables Group and 
Module are adopted to fit the GLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbGLM(factors, N, networkModule, modulematrix, distribution = c("poisson", "NB")[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nbGLM_+3A_factors">factors</code></td>
<td>
<p>Factors with three variables including Count, Group, Direction.</p>
</td></tr>
<tr><td><code id="nbGLM_+3A_n">N</code></td>
<td>
<p>The total sample sizes.</p>
</td></tr>
<tr><td><code id="nbGLM_+3A_networkmodule">networkModule</code></td>
<td>
<p>NetworkModule is the gene sets or modules in the biological network or metabolic pathway, with the 1th column as the module names and the 2th columnn as the gene symbol constituting the module.</p>
</td></tr>
<tr><td><code id="nbGLM_+3A_modulematrix">modulematrix</code></td>
<td>
<p>Modulematrix is a matrix, in which the indicator variables 1 or 0 represent whether a gene belong to a given module or not.</p>
</td></tr>
<tr><td><code id="nbGLM_+3A_distribution">distribution</code></td>
<td>
<p>a character string indicating the distribution of RNA-Seq count value, default is 'NB'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The GLM method was determined by the distribution of RNA-Seq count value including Poisson and Negative Binomial distribution, and there are two indicator
variables Group and Module, Module=1 when a gene belongs to the module and Module= 0 otherwise; Group=1 for case values and Group=0 for control values.
Group * Module represents the interaction effects between Group and Module, and the significance of a module is decided by the interaction and adjusted
p-values are calculated to correct for multiple testing.
</p>


<h3>Value</h3>

<p>The nominal pvalue and FDR for the significance of each gene set or module.
</p>


<h3>Author(s)</h3>

<p>Mingli Lei, Jia Xu, Li-Ching Huang, Lily Wang, Jing Li
</p>


<h3>See Also</h3>

<p>glm()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exprs)
data(networkModule)
case &lt;- c("A1","A2","A3","A4","A5","A6","A7")
control &lt;- c("B1","B2","B3","B4","B5","B6","B7")
factors &lt;- Factor(exprs, case, control) 
modulematrix &lt;- moduleMatrix(exprs,networkModule)
Result &lt;- nbGLM(factors, 14, networkModule, modulematrix, distribution = "NB")
</code></pre>

<hr>
<h2 id='nbGLMdir'>Identify Differential Expression Modules Based on the GLM Model with Up or Down-regulated Change</h2><span id='topic+nbGLMdir'></span>

<h3>Description</h3>

<p>The algorithm identify differential expression modules using Generalized Linear Model (GLM)
for differential expression analysis in RNA-Seq data, and in the model three indicator variables Group, Module and Direction
are adopted to fit the GLM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbGLMdir(factors, N, networkModule, modulematrix, distribution = c("poisson", "NB")[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nbGLMdir_+3A_factors">factors</code></td>
<td>
<p>Factors with three variables including Count, Group, Direction.</p>
</td></tr>
<tr><td><code id="nbGLMdir_+3A_n">N</code></td>
<td>
<p>The total sample size.</p>
</td></tr>
<tr><td><code id="nbGLMdir_+3A_networkmodule">networkModule</code></td>
<td>
<p>NetworkModule is the gene sets or modules in the biological network or metabolic pathway, with the 1th column as the module names and the 2th columnn as the gene symbol constituting the module.</p>
</td></tr>
<tr><td><code id="nbGLMdir_+3A_modulematrix">modulematrix</code></td>
<td>
<p>Modulematrix is a matrix, in which the indicator variables 1 or 0 represent whether a gene belong to a given module or not.</p>
</td></tr>
<tr><td><code id="nbGLMdir_+3A_distribution">distribution</code></td>
<td>
<p>a character string indicating the distribution of RNA-Seq count value, default is 'NB'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The GLM method was determined by the distribution of RNA-Seq count value, such as poisson or negative binomial, and there are three indicator
variables Group, Module and Direction. Module=1 when a gene belongs to the module and Module= 0 otherwise; Group=1 for case values and Group=0 for control values;
Direction=1 for up-regulated and Direction=-1 for down-regualted. Group * Module * Direction represents the interaction effects between Group, Module and Direction.</p>


<h3>Value</h3>

<p>The nominal pvalue  and FDR for the significance of each gene set or module.
</p>


<h3>Author(s)</h3>

<p>Mingli Lei, Li-Ching Huang</p>


<h3>See Also</h3>

<p>glm()</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exprs)
data(networkModule)
case &lt;- c("A1","A2","A3","A4","A5","A6","A7")
control &lt;- c("B1","B2","B3","B4","B5","B6","B7")
factors &lt;- Factor(exprs, case, control) 
modulematrix &lt;- moduleMatrix(exprs,networkModule)
Result &lt;- nbGLMdir(factors, 14, networkModule, modulematrix,distribution="NB")
</code></pre>

<hr>
<h2 id='nbGLMdirperm'>Identify Differential Expression Modules Based on the GLM Method by Shuffling the Phenotypic Variables</h2><span id='topic+nbGLMdirperm'></span>

<h3>Description</h3>

<p>Identify differential expression modules based on the Generalized Linear Model(GLM), including Group, Module and Direction
variables, then generate the empirical null distribution for the statistic z-values and calculate a empirical estimate of p-value 
of each module in the permutation null distribution by shuffling the phenotypic variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbGLMdirperm(exprs, case, control, factors, 
             networkModule, modulematrix, N,
			 distribution = c("poisson", "NB")[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nbGLMdirperm_+3A_exprs">exprs</code></td>
<td>
<p>exprs is a data frame or matrix for two groups or conditions, with rows as variables (genes) and columns as samples.</p>
</td></tr>
<tr><td><code id="nbGLMdirperm_+3A_case">case</code></td>
<td>
<p>case is the sample names in case groups.</p>
</td></tr>
<tr><td><code id="nbGLMdirperm_+3A_control">control</code></td>
<td>
<p>control is the sample names in control groups.</p>
</td></tr>
<tr><td><code id="nbGLMdirperm_+3A_factors">factors</code></td>
<td>
<p>Factors with three variables including Count, Group, Direction.</p>
</td></tr>
<tr><td><code id="nbGLMdirperm_+3A_networkmodule">networkModule</code></td>
<td>
<p>NetworkModule is the gene sets or modules in the biological network or metabolic pathway, with the 1th column as the module names and the 2th columnn as the gene symbol constituting the module.</p>
</td></tr>
<tr><td><code id="nbGLMdirperm_+3A_modulematrix">modulematrix</code></td>
<td>
<p>Modulematrix is a matrix, in which the indicator variables 1 or 0 represent whether a gene belong to a given module or not.</p>
</td></tr>
<tr><td><code id="nbGLMdirperm_+3A_n">N</code></td>
<td>
<p>permutation times. If N&gt;0, the permutation step will be implemented. The default value for N is 0.</p>
</td></tr>
<tr><td><code id="nbGLMdirperm_+3A_distribution">distribution</code></td>
<td>
<p>a character string indicating the distribution of RNA-Seq count value, default is 'NB'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The GLM method was determined by the distribution of RNA-Seq count value including poisson and Negative Binomial distribution, and there are 
three indicator variables Group, Module and Direction, in which Module=1 when a gene belongs to the module and Module= 0 otherwise; Group=1 for case values and 
Group=0 for control values;Direction=1 for up-regulated and Direction=-1 for down-regualted. We therefore construct the contrast vector to
test the null hypothesis by fitting the GLM and then focus on the interaction term Group*Module*Direction. Then the samples between the two
conditions will be disturbed and by shuffling the phenotypic variables, we can generate the empirical null distribution for each module. 
Repeat the above process for N times. Pool all the z score together to form a null distribution of z-value. The corresponding statistical 
significance (p-value) is estimated against null statistics.
</p>


<h3>Value</h3>

<p>The matrix for the sigificance of each module in differential expression analysis.
</p>


<h3>Author(s)</h3>

<p>Mingli Lei, Jia Xu, Li-Ching Huang, Lily Wang, Jing Li
</p>


<h3>See Also</h3>

<p>glm()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exprs)
data(networkModule)
case &lt;- c("A1","A2","A3","A4","A5","A6","A7")
control &lt;- c("B1","B2","B3","B4","B5","B6","B7")
factors &lt;- Factor(exprs, case, control) 
modulematrix &lt;- moduleMatrix(exprs,networkModule)
result &lt;- nbGLMdirperm(exprs,case,control,factors,
                       networkModule, modulematrix,
					   5, distribution="NB")
</code></pre>

<hr>
<h2 id='networkModule'>NetworkModule</h2><span id='topic+networkModule'></span>

<h3>Description</h3>

<p>Different gene sets or modules in the biological network or metabolic pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(networkModule)</code></pre>


<h3>Details</h3>

<p>In this networkModule, there are five modules consist of different genes.
</p>


<h3>Author(s)</h3>

<p>Mingli Lei, Jia Xu, Li-Ching Huang, Lily Wang, Jing Li </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(networkModule)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
