<!DOCTYPE html><html><head><title>Help for package seqmagick</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {seqmagick}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bam2DNAStringSet'><p>bam2DNAStringSet</p></a></li>
<li><a href='#bam2DNAStringSet2'><p>bam2DNAStringSet2</p></a></li>
<li><a href='#bs_aln'><p>bs_aln</p></a></li>
<li><a href='#bs_filter'><p>bs_filter</p></a></li>
<li><a href='#bs_hamming'><p>bs_hamming</p></a></li>
<li><a href='#bs_rename'><p>bs_rename</p></a></li>
<li><a href='#consensus'><p>consensus</p></a></li>
<li><a href='#download_genbank'><p>download_genbank</p></a></li>
<li><a href='#fa_combine'><p>fa_combine</p></a></li>
<li><a href='#fa_filter'><p>fa_filter</p></a></li>
<li><a href='#fa_rename'><p>fa_rename</p></a></li>
<li><a href='#fa_to_interleaved'><p>fa_to_interleaved</p></a></li>
<li><a href='#fa_write'><p>fa_write</p></a></li>
<li><a href='#fas2phy'><p>fas2phy</p></a></li>
<li><a href='#gb_read'><p>gb_read</p></a></li>
<li><a href='#get_id'><p>get_id</p></a></li>
<li><a href='#mega_read'><p>read sequence alignment file</p></a></li>
<li><a href='#phy_read'><p>phy_read</p></a></li>
<li><a href='#phy_write'><p>phy_write</p></a></li>
<li><a href='#phy2fas'><p>phy2fas</p></a></li>
<li><a href='#renameTXT'><p>renameTXT</p></a></li>
<li><a href='#replaceInside'><p>replaceInside</p></a></li>
<li><a href='#seqlen'><p>seqlen</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Sequence Manipulation Utilities</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.7</td>
</tr>
<tr>
<td>Description:</td>
<td>Supports reading and writing sequences for different formats (currently interleaved and sequential formats for 'FASTA' and 'PHYLIP'), file conversion, and manipulation (e.g. filter sequences that contain specify pattern, export consensus sequence from an alignment).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, magrittr, stats, utils, yulab.utils (&gt; 0.1.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>downloader, knitr, rmarkdown, GenomicAlignments,
GenomicRanges, IRanges, muscle, Rsamtools, prettydoc</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/YuLab-SMU/seqmagick">https://github.com/YuLab-SMU/seqmagick</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YuLab-SMU/seqmagick/issues">https://github.com/YuLab-SMU/seqmagick/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-09 02:10:27 UTC; HUAWEI</td>
</tr>
<tr>
<td>Author:</td>
<td>Guangchuang Yu <a href="https://orcid.org/0000-0002-6485-8781"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Guangchuang Yu &lt;guangchuangyu@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-09 03:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bam2DNAStringSet'>bam2DNAStringSet</h2><span id='topic+bam2DNAStringSet'></span>

<h3>Description</h3>

<p>convert bam file to aligned fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bam2DNAStringSet(bamfile, refseq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bam2DNAStringSet_+3A_bamfile">bamfile</code></td>
<td>
<p>bam file</p>
</td></tr>
<tr><td><code id="bam2DNAStringSet_+3A_refseq">refseq</code></td>
<td>
<p>refseq, DNAStringSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DNAStringSet object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

<hr>
<h2 id='bam2DNAStringSet2'>bam2DNAStringSet2</h2><span id='topic+bam2DNAStringSet2'></span>

<h3>Description</h3>

<p>convert bam file to aligned fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bam2DNAStringSet2(bamfile, refseq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bam2DNAStringSet2_+3A_bamfile">bamfile</code></td>
<td>
<p>bam file</p>
</td></tr>
<tr><td><code id="bam2DNAStringSet2_+3A_refseq">refseq</code></td>
<td>
<p>refseq, DNAStringSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DNAStringSet object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

<hr>
<h2 id='bs_aln'>bs_aln</h2><span id='topic+bs_aln'></span>

<h3>Description</h3>

<p>sequence alignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bs_aln(x, method = "muscle", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bs_aln_+3A_x">x</code></td>
<td>
<p>XStringSet object</p>
</td></tr>
<tr><td><code id="bs_aln_+3A_method">method</code></td>
<td>
<p>alignment method</p>
</td></tr>
<tr><td><code id="bs_aln_+3A_...">...</code></td>
<td>
<p>additional parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>aligned sequences, XStringSet object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
x &lt;- fa_read(fa_file)
bs_aln(x)

## End(Not run)
</code></pre>

<hr>
<h2 id='bs_filter'>bs_filter</h2><span id='topic+bs_filter'></span>

<h3>Description</h3>

<p>biological sequence filter by searching pattern
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bs_filter(x, pattern, by = "description", ignore.case = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bs_filter_+3A_x">x</code></td>
<td>
<p>BStringSet object</p>
</td></tr>
<tr><td><code id="bs_filter_+3A_pattern">pattern</code></td>
<td>
<p>keyword for filter</p>
</td></tr>
<tr><td><code id="bs_filter_+3A_by">by</code></td>
<td>
<p>one of 'description' and 'sequence'</p>
</td></tr>
<tr><td><code id="bs_filter_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BStringSet object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
x &lt;- fa_read(fa_file)
bs_filter(x, 'ATGAAAGTAAAA', by='sequence')
</code></pre>

<hr>
<h2 id='bs_hamming'>bs_hamming</h2><span id='topic+bs_hamming'></span>

<h3>Description</h3>

<p>hamming distances of sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bs_hamming(x, count_indel = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bs_hamming_+3A_x">x</code></td>
<td>
<p>BStringSet object</p>
</td></tr>
<tr><td><code id="bs_hamming_+3A_count_indel">count_indel</code></td>
<td>
<p>whether count indel or not</p>
</td></tr>
<tr><td><code id="bs_hamming_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>hamming distance
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
x &lt;- fa_read(fa_file)
## align first 5 sequences, use `bs_aln(x)` to align all sequences
aln &lt;- bs_aln(x[1:5])
bs_hamming(aln)

## End(Not run)
</code></pre>

<hr>
<h2 id='bs_rename'>bs_rename</h2><span id='topic+bs_rename'></span>

<h3>Description</h3>

<p>rename sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bs_rename(x, mapping, position, sep, mode)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bs_rename_+3A_x">x</code></td>
<td>
<p>BStringSet object</p>
</td></tr>
<tr><td><code id="bs_rename_+3A_mapping">mapping</code></td>
<td>
<p>two column data.frame</p>
</td></tr>
<tr><td><code id="bs_rename_+3A_position">position</code></td>
<td>
<p>rename token at specific position</p>
</td></tr>
<tr><td><code id="bs_rename_+3A_sep">sep</code></td>
<td>
<p>sepator to divide token</p>
</td></tr>
<tr><td><code id="bs_rename_+3A_mode">mode</code></td>
<td>
<p>one of 'replace', 'prefix' or 'suffix'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BStringSet
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

<hr>
<h2 id='consensus'>consensus</h2><span id='topic+consensus'></span><span id='topic+bs_consensus'></span>

<h3>Description</h3>

<p>consensus of aligned sequences
</p>
<p>consensus of aligned sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consensus(x, type = "DNA")

bs_consensus(x, type = "DNA", r = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="consensus_+3A_x">x</code></td>
<td>
<p>BStringSet object</p>
</td></tr>
<tr><td><code id="consensus_+3A_type">type</code></td>
<td>
<p>currently, only DNA supported</p>
</td></tr>
<tr><td><code id="consensus_+3A_r">r</code></td>
<td>
<p>if any NT &gt; r, it will be selected as representative base</p>
</td></tr>
</table>


<h3>Value</h3>

<p>consensus sequence string
</p>
<p>consensus sequence string
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
x &lt;- fa_read(fa_file)
## align first 5 sequences, use `bs_aln(x)` to align all sequences
aln &lt;- bs_aln(x[1:5])
## or bs_consensus(aln)
consensus(aln)

## End(Not run)
</code></pre>

<hr>
<h2 id='download_genbank'>download_genbank</h2><span id='topic+download_genbank'></span>

<h3>Description</h3>

<p>download genbank or fasta file by accession number
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_genbank(acc, db = "nuccore", format = "genbank", outfile = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download_genbank_+3A_acc">acc</code></td>
<td>
<p>accession number(s)</p>
</td></tr>
<tr><td><code id="download_genbank_+3A_db">db</code></td>
<td>
<p>supported db, currently 'nuccore'</p>
</td></tr>
<tr><td><code id="download_genbank_+3A_format">format</code></td>
<td>
<p>one of 'genbank' or 'fasta'</p>
</td></tr>
<tr><td><code id="download_genbank_+3A_outfile">outfile</code></td>
<td>
<p>output file, by default, acc.gb or acc.fa</p>
</td></tr>
<tr><td><code id="download_genbank_+3A_...">...</code></td>
<td>
<p>additional parameters for download.file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>output file vector
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tmpgb &lt;- tempfile(fileext = '.gb')
tmpfa &lt;- tempfile(fileext = '.fa')
download_genbank(acc='AB115403', format='genbank', outfile=tmpgb)
download_genbank(acc='AB115403', format='fasta', outfile=tmpfa)

## End(Not run)
</code></pre>

<hr>
<h2 id='fa_combine'>fa_combine</h2><span id='topic+fa_combine'></span>

<h3>Description</h3>

<p>combine 2 fasta files into 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fa_combine(file1, file2, outfile = NULL, type = "interleaved")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fa_combine_+3A_file1">file1</code></td>
<td>
<p>fasta file 1</p>
</td></tr>
<tr><td><code id="fa_combine_+3A_file2">file2</code></td>
<td>
<p>fasta file 2</p>
</td></tr>
<tr><td><code id="fa_combine_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
<tr><td><code id="fa_combine_+3A_type">type</code></td>
<td>
<p>one of interleaved and sequential</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BStringSet
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
x &lt;- fa_read(fa_file)
fa1 &lt;- tempfile(fileext=".fa")
fa2 &lt;- tempfile(fileext=".fa")
fa_write(x[1:5], fa1)
fa_write(x[6:10], fa2)
fa_combine(fa1, fa2)
</code></pre>

<hr>
<h2 id='fa_filter'>fa_filter</h2><span id='topic+fa_filter'></span>

<h3>Description</h3>

<p>fasta filter by searching pattern
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fa_filter(
  fasfile,
  pattern,
  by = "description",
  ignore.case = FALSE,
  outfile = NULL,
  type = "interleaved"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fa_filter_+3A_fasfile">fasfile</code></td>
<td>
<p>input fasta file</p>
</td></tr>
<tr><td><code id="fa_filter_+3A_pattern">pattern</code></td>
<td>
<p>keyword for filter</p>
</td></tr>
<tr><td><code id="fa_filter_+3A_by">by</code></td>
<td>
<p>one of 'description' and 'sequence'</p>
</td></tr>
<tr><td><code id="fa_filter_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="fa_filter_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
<tr><td><code id="fa_filter_+3A_type">type</code></td>
<td>
<p>one of 'interleaved' and 'sequential'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BStringSet object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

<hr>
<h2 id='fa_rename'>fa_rename</h2><span id='topic+fa_rename'></span>

<h3>Description</h3>

<p>rename fasta sequence name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fa_rename(fasfile, mapping_file, position, sep, mode, outfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fa_rename_+3A_fasfile">fasfile</code></td>
<td>
<p>fasta file</p>
</td></tr>
<tr><td><code id="fa_rename_+3A_mapping_file">mapping_file</code></td>
<td>
<p>mapping file</p>
</td></tr>
<tr><td><code id="fa_rename_+3A_position">position</code></td>
<td>
<p>rename token at specific position</p>
</td></tr>
<tr><td><code id="fa_rename_+3A_sep">sep</code></td>
<td>
<p>sepator to divide token</p>
</td></tr>
<tr><td><code id="fa_rename_+3A_mode">mode</code></td>
<td>
<p>one of 'replace', 'prefix' or 'suffix'</p>
</td></tr>
<tr><td><code id="fa_rename_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BStringSet object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

<hr>
<h2 id='fa_to_interleaved'>fa_to_interleaved</h2><span id='topic+fa_to_interleaved'></span><span id='topic+fa_to_sequential'></span>

<h3>Description</h3>

<p>convert fasta file to interleaved format
</p>
<p>convert fasta file to sequential format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fa_to_interleaved(file, outfile)

fa_to_sequential(file, outfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fa_to_interleaved_+3A_file">file</code></td>
<td>
<p>fasta file</p>
</td></tr>
<tr><td><code id="fa_to_interleaved_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>
<p>None
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
fa1 &lt;- tempfile(fileext = '.fa')
fa2 &lt;- tempfile(fileext = '.fa')
fa_to_interleaved(fa_file, fa1)
fa_to_sequential(fa_file, fa2)
</code></pre>

<hr>
<h2 id='fa_write'>fa_write</h2><span id='topic+fa_write'></span>

<h3>Description</h3>

<p>write fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fa_write(x, outfile, type = "interleaved")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fa_write_+3A_x">x</code></td>
<td>
<p>XStringSet object</p>
</td></tr>
<tr><td><code id="fa_write_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
<tr><td><code id="fa_write_+3A_type">type</code></td>
<td>
<p>one of interleaved and sequential</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>References</h3>

<p><a href="https://phylipweb.github.io/phylip/">https://phylipweb.github.io/phylip/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phy_file &lt;- system.file("extdata/HA.phy", package="seqmagick")
x &lt;- phy_read(phy_file)
fa_file &lt;- tempfile(fileext = '.fas')
fa_write(x, fa_file)
</code></pre>

<hr>
<h2 id='fas2phy'>fas2phy</h2><span id='topic+fas2phy'></span>

<h3>Description</h3>

<p>convert fasta (aligned sequences) to phylip format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fas2phy(fasfile, outfile = "out.phy", type = "sequential")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fas2phy_+3A_fasfile">fasfile</code></td>
<td>
<p>aligned sequences in fasta format</p>
</td></tr>
<tr><td><code id="fas2phy_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
<tr><td><code id="fas2phy_+3A_type">type</code></td>
<td>
<p>one of interleaved and sequential</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
fa_file &lt;- system.file(&quot;extdata/HA.fas&quot;, package=&quot;seqmagick&quot;)
phy_file &lt;- tempfile(fileext = &quot;.phy&quot;)
fas2phy(fa_file, phy_file)
</p>

<hr>
<h2 id='gb_read'>gb_read</h2><span id='topic+gb_read'></span>

<h3>Description</h3>

<p>extract accession number and sequence from genbank file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gb_read(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gb_read_+3A_file">file</code></td>
<td>
<p>input genbank file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sequence object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

<hr>
<h2 id='get_id'>get_id</h2><span id='topic+get_id'></span>

<h3>Description</h3>

<p>get id at specific position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_id(x, sep = " ", position)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_id_+3A_x">x</code></td>
<td>
<p>sequence description line</p>
</td></tr>
<tr><td><code id="get_id_+3A_sep">sep</code></td>
<td>
<p>separator to split x</p>
</td></tr>
<tr><td><code id="get_id_+3A_position">position</code></td>
<td>
<p>id position</p>
</td></tr>
</table>


<h3>Value</h3>

<p>id
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
x &lt;- fa_read(fa_file)
get_id(names(x)[1:5], sep = " ", position=1)
</code></pre>

<hr>
<h2 id='mega_read'>read sequence alignment file</h2><span id='topic+mega_read'></span><span id='topic+fa_read'></span><span id='topic+clw_read'></span><span id='topic+sth_read'></span>

<h3>Description</h3>

<p>read sequence alignment file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mega_read(file)

fa_read(file, type = "auto")

clw_read(file, type = "auto")

sth_read(file, type = "auto")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mega_read_+3A_file">file</code></td>
<td>
<p>multiple sequence file</p>
</td></tr>
<tr><td><code id="mega_read_+3A_type">type</code></td>
<td>
<p>one of 'DNA', 'RNA', 'AA', 'Protein', 'unknown' or 'auto'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BStringSet object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
fa_read(fa_file)

mega_file &lt;- system.file("extdata/mega/Crab_rRNA.meg", package="seqmagick")
mega_read(mega_file)
</code></pre>

<hr>
<h2 id='phy_read'>phy_read</h2><span id='topic+phy_read'></span>

<h3>Description</h3>

<p>read aligned sequences in phylip format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phy_read(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phy_read_+3A_file">file</code></td>
<td>
<p>phylip file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>BStringSet object
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phy_file &lt;- system.file("extdata/HA.phy", package="seqmagick")
phy_read(phy_file)
</code></pre>

<hr>
<h2 id='phy_write'>phy_write</h2><span id='topic+phy_write'></span>

<h3>Description</h3>

<p>write phylip file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phy_write(x, outfile, type = "sequential")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phy_write_+3A_x">x</code></td>
<td>
<p>XStringSet object</p>
</td></tr>
<tr><td><code id="phy_write_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
<tr><td><code id="phy_write_+3A_type">type</code></td>
<td>
<p>one of interleaved and sequential</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fa_file &lt;- system.file("extdata/HA.fas", package="seqmagick")
x &lt;- fa_read(fa_file)
aln &lt;- bs_aln(x[1:5])
phy_file &lt;- tempfile(fileext = '.phy')
phy_write(aln, phy_file)

## End(Not run)
</code></pre>

<hr>
<h2 id='phy2fas'>phy2fas</h2><span id='topic+phy2fas'></span>

<h3>Description</h3>

<p>convert phylip file to fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phy2fas(phyfile, outfile = "out.fas", type = "interleaved")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phy2fas_+3A_phyfile">phyfile</code></td>
<td>
<p>phylip file</p>
</td></tr>
<tr><td><code id="phy2fas_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
<tr><td><code id="phy2fas_+3A_type">type</code></td>
<td>
<p>one of interleaved and sequential</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phy_file &lt;- system.file("extdata/HA.phy", package="seqmagick")
fa_file &lt;- tempfile(fileext = '.fas')
phy2fas(phy_file, fa_file)
</code></pre>

<hr>
<h2 id='renameTXT'>renameTXT</h2><span id='topic+renameTXT'></span>

<h3>Description</h3>

<p>rename txt file (eg Description line of fasta file)
according to first token (eg accession number)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>renameTXT(txt_file, name_file, sep = "_", split = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="renameTXT_+3A_txt_file">txt_file</code></td>
<td>
<p>txt file</p>
</td></tr>
<tr><td><code id="renameTXT_+3A_name_file">name_file</code></td>
<td>
<p>name file</p>
</td></tr>
<tr><td><code id="renameTXT_+3A_sep">sep</code></td>
<td>
<p>separator</p>
</td></tr>
<tr><td><code id="renameTXT_+3A_split">split</code></td>
<td>
<p>logical, split result or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

<hr>
<h2 id='replaceInside'>replaceInside</h2><span id='topic+replaceInside'></span>

<h3>Description</h3>

<p>replace character for example from '-' to 'N' of fasta sequence that only applied inside sequence
any '-' character at start/end of the sequence (aligned seqs may contains '-' at prefix/suffix) will not be replaced
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replaceInside(fasfile, from = "-", to = "N", outfile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="replaceInside_+3A_fasfile">fasfile</code></td>
<td>
<p>fasta file</p>
</td></tr>
<tr><td><code id="replaceInside_+3A_from">from</code></td>
<td>
<p>character to be replaced, '-' by default</p>
</td></tr>
<tr><td><code id="replaceInside_+3A_to">to</code></td>
<td>
<p>replace 'from' to 'to', 'N' by default</p>
</td></tr>
<tr><td><code id="replaceInside_+3A_outfile">outfile</code></td>
<td>
<p>output file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DNAStringSet
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

<hr>
<h2 id='seqlen'>seqlen</h2><span id='topic+seqlen'></span>

<h3>Description</h3>

<p>sequence length
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seqlen(fasfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seqlen_+3A_fasfile">fasfile</code></td>
<td>
<p>fasta file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector
</p>


<h3>Author(s)</h3>

<p>Guangchuang Yu
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
