<!DOCTYPE html><html lang="en"><head><title>Help for package SeuratExplorer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SeuratExplorer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cellRatioPlot'><p>plot cell percentage barplot</p></a></li>
<li><a href='#explorer_body_ui'><p>generate the body UI for each menu item specified in <code>explorer_sidebar_ui</code></p></a></li>
<li><a href='#explorer_server'><p>server side functions related to <code>explorer_sidebar_ui</code></p></a></li>
<li><a href='#explorer_sidebar_ui'><p>some menu items of the dashboard sidebar</p></a></li>
<li><a href='#getColors'><p>getColors</p></a></li>
<li><a href='#launchSeuratExplorer'><p>Launch shiny app</p></a></li>
<li><a href='#server'><p>Server</p></a></li>
<li><a href='#top_genes'><p>Find Top Genes by Cell</p></a></li>
<li><a href='#ui'><p>UI</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>An 'Shiny' App for Exploring scRNA-seq Data Processed in
'Seurat'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A simple, one-command package which runs an interactive dashboard capable of common visualizations for single cell RNA-seq. 'SeuratExplorer' requires a processed 'Seurat' object, which is saved as 'rds' or 'qs2' file.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, utils, Seurat, shiny, shinydashboard, DT,
shinycssloaders, patchwork, tools, colourpicker, shinyWidgets,
scales, SeuratObject, shinyjqui, shinyBS, ggalluvial, dplyr,
ComplexHeatmap, qs2, circlize, reshape2, stats, htmltools</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocManager, DESeq2, MAST, roxygen2, devtools</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-12 00:05:21 UTC; Xi_Lab</td>
</tr>
<tr>
<td>Author:</td>
<td>Yongchao Zhang [aut, cre] (https://github.com/fentouxungui)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yongchao Zhang &lt;zhangyongchao@nibs.ac.cn&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-13 13:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='cellRatioPlot'>plot cell percentage barplot</h2><span id='topic+cellRatioPlot'></span>

<h3>Description</h3>

<p>support facet, codes refer to: https://github.com/junjunlab/scRNAtoolVis/blob/master/R/cellRatioPlot.R, with modification
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cellRatioPlot(
  object = NULL,
  sample.name = NULL,
  sample.order = NULL,
  celltype.name = NULL,
  celltype.order = NULL,
  facet.name = NULL,
  facet.order = NULL,
  col.width = 0.7,
  flow.alpha = 0.25,
  flow.curve = 0,
  color.choice = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cellRatioPlot_+3A_object">object</code></td>
<td>
<p>an Seurat object</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_sample.name">sample.name</code></td>
<td>
<p>x axis</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_sample.order">sample.order</code></td>
<td>
<p>order for x axis</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_celltype.name">celltype.name</code></td>
<td>
<p>column fill by</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_celltype.order">celltype.order</code></td>
<td>
<p>order for fill by</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_facet.name">facet.name</code></td>
<td>
<p>column name for facet</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_facet.order">facet.order</code></td>
<td>
<p>the order for facet</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_col.width">col.width</code></td>
<td>
<p>column width, from 0-1</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_flow.alpha">flow.alpha</code></td>
<td>
<p>transparency for flow</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_flow.curve">flow.curve</code></td>
<td>
<p>curve for flow</p>
</td></tr>
<tr><td><code id="cellRatioPlot_+3A_color.choice">color.choice</code></td>
<td>
<p>color choice for fill</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#NULL
</code></pre>

<hr>
<h2 id='explorer_body_ui'>generate the body UI for each menu item specified in <code>explorer_sidebar_ui</code></h2><span id='topic+explorer_body_ui'></span>

<h3>Description</h3>

<p>generate the body UI for each menu item specified in <code>explorer_sidebar_ui</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>explorer_body_ui(tab_list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="explorer_body_ui_+3A_tab_list">tab_list</code></td>
<td>
<p>a tag list for the body UI of shiny dashboard</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a filled tag list for body UI
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tab_list &lt;- list()
tab_list &lt;- explorer_body_ui(tab_list = tab_list)

</code></pre>

<hr>
<h2 id='explorer_server'>server side functions related to <code>explorer_sidebar_ui</code></h2><span id='topic+explorer_server'></span>

<h3>Description</h3>

<p>server side functions related to <code>explorer_sidebar_ui</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>explorer_server(input, output, session, data, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="explorer_server_+3A_input">input</code></td>
<td>
<p>server input</p>
</td></tr>
<tr><td><code id="explorer_server_+3A_output">output</code></td>
<td>
<p>server output</p>
</td></tr>
<tr><td><code id="explorer_server_+3A_session">session</code></td>
<td>
<p>server session</p>
</td></tr>
<tr><td><code id="explorer_server_+3A_data">data</code></td>
<td>
<p>the Seurat object and related parameters</p>
</td></tr>
<tr><td><code id="explorer_server_+3A_verbose">verbose</code></td>
<td>
<p>for debug use</p>
</td></tr>
</table>


<h3>Value</h3>

<p>server side functions related to <code>explorer_sidebar_ui</code>
</p>

<hr>
<h2 id='explorer_sidebar_ui'>some menu items of the dashboard sidebar</h2><span id='topic+explorer_sidebar_ui'></span>

<h3>Description</h3>

<p>to generate some menu items for the dashboard, which can be integrated to other
packages, such as 'fentouxungui/SeuratExplorerServer' from github.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>explorer_sidebar_ui()
</code></pre>


<h3>Value</h3>

<p>return some menu items for the dashboard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>explorer_sidebar_ui()

</code></pre>

<hr>
<h2 id='getColors'>getColors</h2><span id='topic+getColors'></span>

<h3>Description</h3>

<p>getColors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getColors(color.platte = NULL, choice = NULL, n = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getColors_+3A_color.platte">color.platte</code></td>
<td>
<p>predefined color list</p>
</td></tr>
<tr><td><code id="getColors_+3A_choice">choice</code></td>
<td>
<p>color name</p>
</td></tr>
<tr><td><code id="getColors_+3A_n">n</code></td>
<td>
<p>how many colors to return</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a color list
</p>


<h3>Examples</h3>

<pre><code class='language-R'># null
</code></pre>

<hr>
<h2 id='launchSeuratExplorer'>Launch shiny app</h2><span id='topic+launchSeuratExplorer'></span>

<h3>Description</h3>

<p>Launch shiny app
</p>


<h3>Usage</h3>

<pre><code class='language-R'>launchSeuratExplorer(verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="launchSeuratExplorer_+3A_verbose">verbose</code></td>
<td>
<p>for debug use</p>
</td></tr>
</table>


<h3>Value</h3>

<p>In-browser Shiny Application launch
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()){launchSeuratExplorer()}
</code></pre>

<hr>
<h2 id='server'>Server</h2><span id='topic+server'></span>

<h3>Description</h3>

<p>Server
</p>


<h3>Usage</h3>

<pre><code class='language-R'>server(input, output, session)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="server_+3A_input">input</code></td>
<td>
<p>Input from the UI</p>
</td></tr>
<tr><td><code id="server_+3A_output">output</code></td>
<td>
<p>Output to send back to UI</p>
</td></tr>
<tr><td><code id="server_+3A_session">session</code></td>
<td>
<p>from shiny server function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the server functions of shiny app
</p>

<hr>
<h2 id='top_genes'>Find Top Genes by Cell</h2><span id='topic+top_genes'></span>

<h3>Description</h3>

<p>for each cell, find genes that has high UMI percentage, for example, if a cell has 10000 UMIs, and the UMI percentage cutoff is set to 0.01,
then all genes that has more than 10000 * 0.01 = 100 UMIs is thought to be the highly expressed genes for this cell.summary those genes for each cluster,
firstly get all highly expressed genes in a cluster, some genes may has less cells, then for each gene, count cells in which this genes is highly expressed,
and also calculate the mean and median UMI percentage in those highly expressed cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>top_genes(SeuratObj, expr.cut = 0.01, group.by)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="top_genes_+3A_seuratobj">SeuratObj</code></td>
<td>
<p>Seurat object</p>
</td></tr>
<tr><td><code id="top_genes_+3A_expr.cut">expr.cut</code></td>
<td>
<p>UMI percentage cutoff, in a cell, if a gene with UMIs ratio more than this cutoff, this gene will be assigned to highly expressed gene for this cell</p>
</td></tr>
<tr><td><code id="top_genes_+3A_group.by">group.by</code></td>
<td>
<p>how to group cells</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame
</p>

<hr>
<h2 id='ui'>UI</h2><span id='topic+ui'></span>

<h3>Description</h3>

<p>UI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ui()
</code></pre>


<h3>Value</h3>

<p>the UI part of the shiny app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ui()

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
