<!DOCTYPE html><html><head><title>Help for package SeuratObject</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SeuratObject}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SeuratObject-package'><p>SeuratObject: Data Structures for Single Cell Data</p></a></li>
<li><a href='#.AssayClass'><p>Assay Class Label</p></a></li>
<li><a href='#.BPMatrixMode'><p><span class="pkg">BPCells</span> Matrix Mode</p></a></li>
<li><a href='#.CalcN'><p>Calculate nCount and nFeature</p></a></li>
<li><a href='#.CheckFmargin'><p>Check Feature Margin</p></a></li>
<li><a href='#.ClassPkg'><p>Get the Package that Defines a Class</p></a></li>
<li><a href='#.Collections'><p>Identify Object Collections</p></a></li>
<li><a href='#.Contains'><p>Get Parent S4 Classes</p></a></li>
<li><a href='#.CreateStdAssay'><p>Generic Assay Creation</p></a></li>
<li><a href='#.DefaultFOV'><p>Find the Default FOV</p></a></li>
<li><a href='#.Deprecate'><p>Deprecate Functions and Arguments</p></a></li>
<li><a href='#.DiskLoad'><p>Disk Loading Function</p></a></li>
<li><a href='#.DollarNames.Assay'><p>Dollar-sign Autocompletion</p></a></li>
<li><a href='#.DollarNames.Assay5'><p>Dollar-sign Autocompletion</p></a></li>
<li><a href='#.DollarNames.Seurat'><p>Dollar-sign Autocompletion</p></a></li>
<li><a href='#.DollarNames.SeuratCommand'><p>Dollar-sign Autocompletion</p></a></li>
<li><a href='#.DollarNames.StdAssay'><p>Dollar-sign Autocompletion</p></a></li>
<li><a href='#.FileMove'><p>Move Files and Directories</p></a></li>
<li><a href='#.FilePath'><p>Find a File Path</p></a></li>
<li><a href='#.FilterObjects'><p>Find Subobjects Of A Certain Class</p></a></li>
<li><a href='#.FindObject'><p>Find A Subobject</p></a></li>
<li><a href='#.GetMethod'><p>Get a Method</p></a></li>
<li><a href='#.IsFutureSeurat'><p>Test Future Compatibility with <span class="pkg">Seurat</span></p></a></li>
<li><a href='#.KeyPattern'><p>Regex Pattern for Keys</p></a></li>
<li><a href='#.MARGIN'><p>Get the Margin of an Object</p></a></li>
<li><a href='#.PropagateList'><p>Propagate a List</p></a></li>
<li><a href='#.RandomKey'><p>Generate a Random Key</p></a></li>
<li><a href='#.SelectFeatures'><p>Combine and Select Features</p></a></li>
<li><a href='#.SparseSlots'><p>Identify Sparse Slots</p></a></li>
<li><a href='#.Subobjects'><p>Get the Subobject Names</p></a></li>
<li><a href='#[,LogMap'><p>Matrix-like Subsetting for Logical Maps</p></a></li>
<li><a href='#[.Assay'><p>Layer Data</p></a></li>
<li><a href='#[.Assay5'><p>Layer Data</p></a></li>
<li><a href='#[.DimReduc'><p>Get Feature Loadings</p></a></li>
<li><a href='#[.SeuratCommand'><p>Command Log Data Access</p></a></li>
<li><a href='#[.StdAssay'><p>Layer Data</p></a></li>
<li><a href='#[[,LogMap,integer,missing-method'><p><code>LogMap</code> Interaction Methods</p></a></li>
<li><a href='#[[.Assay'><p>Feature-Level Meta Data</p></a></li>
<li><a href='#[[.Assay5'><p>Feature-Level Meta Data</p></a></li>
<li><a href='#[[.DimReduc'><p>Get Cell Embeddings</p></a></li>
<li><a href='#[[.Seurat'><p>Subobjects and Cell-Level Meta Data</p></a></li>
<li><a href='#[[.StdAssay'><p>Feature-Level Meta Data</p></a></li>
<li><a href='#[[&lt;-,Seurat'><p>Add Subobjects</p></a></li>
<li><a href='#[[&lt;-,Seurat,character,missing,StdAssay-method'><p>Add Subobjects</p></a></li>
<li><a href='#[[&lt;-,Seurat,NULL'><p>Remove Subobjects and Cell-Level Meta Data</p></a></li>
<li><a href='#$.Assay'><p>Layer Data</p></a></li>
<li><a href='#$.Assay5'><p>Layer Data</p></a></li>
<li><a href='#$.Seurat'><p>Cell-Level Meta Data</p></a></li>
<li><a href='#$.SeuratCommand'><p>Command Log Parameter Access</p></a></li>
<li><a href='#$.StdAssay'><p>Layer Data</p></a></li>
<li><a href='#AddMetaData'><p>Add in metadata associated with either cells or features.</p></a></li>
<li><a href='#AddMetaData-StdAssay'><p>Add in metadata associated with either cells or features.</p></a></li>
<li><a href='#aggregate'><p>Aggregate Molecules into an Expression Matrix</p></a></li>
<li><a href='#Angles'><p>Radian/Degree Conversions</p></a></li>
<li><a href='#as.Centroids'><p>Convert Segmentation Layers</p></a></li>
<li><a href='#as.Graph'><p>Coerce to a <code>Graph</code> Object</p></a></li>
<li><a href='#as.list.SeuratCommand'><p>Coerce a <code>SeuratCommand</code> to a list</p></a></li>
<li><a href='#as.matrix.LogMap'><p>Coerce Logical Maps to Matrices</p></a></li>
<li><a href='#as.Neighbor'><p>Coerce to a <code>Neighbor</code> Object</p></a></li>
<li><a href='#as.Seurat'><p>Coerce to a <code>Seurat</code> Object</p></a></li>
<li><a href='#as.sparse'><p>Cast to Sparse</p></a></li>
<li><a href='#Assay-class'><p>The Assay Class</p></a></li>
<li><a href='#Assay-validity'><p>V3 Assay Validity</p></a></li>
<li><a href='#Assay5-class'><p>The v5 <code>Assay</code> Object</p></a></li>
<li><a href='#Assay5-validity'><p>V5 Assay Validity</p></a></li>
<li><a href='#Assay5T-class'><p>The Transposed v5 <code>Assay</code> Object</p></a></li>
<li><a href='#AssayData'><p>Get and Set Assay Data</p></a></li>
<li><a href='#AssayData-StdAssay'><p>Get and Set Assay Data</p></a></li>
<li><a href='#Assays'><p>Query Specific Object Types</p></a></li>
<li><a href='#AttachDeps'><p>Attach Required Packages</p></a></li>
<li><a href='#Boundaries'><p>Get, Set, and Query Segmentation Boundaries</p></a></li>
<li><a href='#CastAssay'><p>Cast Assay Layers</p></a></li>
<li><a href='#CastAssay-StdAssay'><p>Cast Assay Layers</p></a></li>
<li><a href='#Cells'><p>Cell and Feature Names</p></a></li>
<li><a href='#Cells-StdAssay'><p>Cell and Feature Names</p></a></li>
<li><a href='#CellsByIdentities'><p>Get cell names grouped by identity class</p></a></li>
<li><a href='#CellsByImage'><p>Get a vector of cell names associated with an image (or set of images)</p></a></li>
<li><a href='#Centroids-class'><p>The <code>Centroids</code> Class</p></a></li>
<li><a href='#Centroids-methods'><p><code>Centroids</code> Methods</p></a></li>
<li><a href='#CheckDots'><p>Check the Use of Dots</p></a></li>
<li><a href='#CheckFeaturesNames'><p>Check features names format</p></a></li>
<li><a href='#CheckGC'><p>Conditional Garbage Collection</p></a></li>
<li><a href='#CheckLayersName'><p>Check layers names for the input list</p></a></li>
<li><a href='#CheckMatrix'><p>Check Matrix Validity</p></a></li>
<li><a href='#ClassKey'><p>Generate a Class Key</p></a></li>
<li><a href='#colMeans,Assay-method'><p>Row and Column Sums and Means</p></a></li>
<li><a href='#colMeans,Seurat-method'><p>Row and Column Sums and Means</p></a></li>
<li><a href='#Command'><p>Get SeuratCommands</p></a></li>
<li><a href='#CreateAssay5Object'><p>Create a v5 Assay object</p></a></li>
<li><a href='#CreateAssayObject'><p>Create an Assay object</p></a></li>
<li><a href='#CreateCentroids'><p>Create a <code>Centroids</code> Objects</p></a></li>
<li><a href='#CreateDimReducObject'><p>Create a DimReduc object</p></a></li>
<li><a href='#CreateFOV'><p>Create Spatial Coordinates</p></a></li>
<li><a href='#CreateMolecules'><p>Create a <code>Molecules</code> Object</p></a></li>
<li><a href='#CreateSegmentation'><p>Create a <code>Segmentation</code> Objects</p></a></li>
<li><a href='#CreateSeuratObject'><p>Create a <code>Seurat</code> object</p></a></li>
<li><a href='#Crop'><p>Crop Coordinates</p></a></li>
<li><a href='#DefaultAssay'><p>Default Assay</p></a></li>
<li><a href='#DefaultAssay-StdAssay'><p>Default Assay</p></a></li>
<li><a href='#DefaultDimReduc'><p>Find the default <code>DimReduc</code></p></a></li>
<li><a href='#DefaultFOV'><p>Get and Set the Default FOV</p></a></li>
<li><a href='#DefaultLayer'><p>Default Layer</p></a></li>
<li><a href='#DefaultLayer-StdAssay'><p>Default Layer</p></a></li>
<li><a href='#dim.Assay'><p>Feature and Cell Numbers</p></a></li>
<li><a href='#dim.Assay5'><p>Feature and Cell Numbers</p></a></li>
<li><a href='#dim.DimReduc'><p>Dimensional Reduction Meta-Information</p></a></li>
<li><a href='#dim.Seurat'><p>Feature and Cell Numbers</p></a></li>
<li><a href='#dim.StdAssay'><p>Feature and Cell Numbers</p></a></li>
<li><a href='#dimnames.Assay'><p>Assay-Level Feature and Cell Names</p></a></li>
<li><a href='#dimnames.Assay5'><p>Assay-Level Feature and Cell Names</p></a></li>
<li><a href='#dimnames.Seurat'><p>Feature and Cell Names</p></a></li>
<li><a href='#dimnames.StdAssay'><p>Assay-Level Feature and Cell Names</p></a></li>
<li><a href='#DimReduc-class'><p>The Dimensional Reduction Class</p></a></li>
<li><a href='#DimReduc-validity'><p>Dimensional Reduction Validity</p></a></li>
<li><a href='#Distances'><p>Get the Neighbor nearest neighbors distance matrix</p></a></li>
<li><a href='#droplevels.LogMap'><p>Drop Unused Logical Map Values</p></a></li>
<li><a href='#Embeddings'><p>Get Cell Embeddings</p></a></li>
<li><a href='#EmptyDF'><p>Empty Data Frames</p></a></li>
<li><a href='#ExtractField'><p>Extract delimiter information from a string.</p></a></li>
<li><a href='#FetchData'><p>Access cellular data</p></a></li>
<li><a href='#FilterObjects'><p>Find Sub-objects of a Certain Class</p></a></li>
<li><a href='#FOV-class'><p>The Field of View Object</p></a></li>
<li><a href='#FOV-methods'><p><code>FOV</code> Methods</p></a></li>
<li><a href='#FOV-validity'><p>FOV Validity</p></a></li>
<li><a href='#GetImage'><p>Get image data</p></a></li>
<li><a href='#GetTissueCoordinates'><p>Get tissue coordinates</p></a></li>
<li><a href='#Graph-class'><p>The Graph Class</p></a></li>
<li><a href='#HVFInfo'><p>Highly Variable Features</p></a></li>
<li><a href='#Idents'><p>Get, set, and manipulate an object's identity classes</p></a></li>
<li><a href='#Images'><p>Pull spatial image names</p></a></li>
<li><a href='#Index'><p>Get Neighbor algorithm index</p></a></li>
<li><a href='#Indices'><p>Get Neighbor nearest neighbor index matrices</p></a></li>
<li><a href='#intersect.LogMap'><p>Find Common Logical Map Values</p></a></li>
<li><a href='#IsGlobal'><p>Is an object global/persistent?</p></a></li>
<li><a href='#IsMatrixEmpty'><p>Check if a matrix is empty</p></a></li>
<li><a href='#IsNamedList'><p>Check List Names</p></a></li>
<li><a href='#IsSparse'><p>Is a Matrix Sparse</p></a></li>
<li><a href='#JackStrawData-class'><p>The JackStrawData Class</p></a></li>
<li><a href='#JackStrawData-methods'><p><code>JackStrawData</code> Methods</p></a></li>
<li><a href='#JoinLayers'><p>Split and Join Layers Together</p></a></li>
<li><a href='#JS'><p>Get and set JackStraw information</p></a></li>
<li><a href='#Key'><p>Get and set object keys</p></a></li>
<li><a href='#Key-validity'><p>Key Validity</p></a></li>
<li><a href='#KeyMixin-class'><p>A Mixin for Keyed objects</p></a></li>
<li><a href='#labels.LogMap'><p>Find Observations by Value</p></a></li>
<li><a href='#LayerData'><p>Query and Manipulate Assay Layers</p></a></li>
<li><a href='#Layers-StdAssay'><p>Query and Manipulate Assay Layers</p></a></li>
<li><a href='#Loadings'><p>Get and set feature loadings</p></a></li>
<li><a href='#LogMap'><p>A Logical Map</p></a></li>
<li><a href='#LogMap-validity'><p>Logical Map Validity</p></a></li>
<li><a href='#LogSeuratCommand'><p>Log a command</p></a></li>
<li><a href='#MatchCells'><p>Match Cells</p></a></li>
<li><a href='#merge.Assay'><p>Merge Assays</p></a></li>
<li><a href='#merge.Assay5'><p>Merge Assays</p></a></li>
<li><a href='#merge.DimReduc'><p>Merge Dimensional Reductions</p></a></li>
<li><a href='#merge.Seurat'><p>Merge Seurat Objects</p></a></li>
<li><a href='#merge.StdAssay'><p>Merge Assays</p></a></li>
<li><a href='#Misc'><p>Get and set miscellaneous data</p></a></li>
<li><a href='#Misc-StdAssay'><p>Get and set miscellaneous data</p></a></li>
<li><a href='#Molecules-class'><p>The Spatial Molecules Class</p></a></li>
<li><a href='#Molecules-methods'><p><code>Molecules</code> Methods</p></a></li>
<li><a href='#names.Seurat'><p>Subobject Names</p></a></li>
<li><a href='#Neighbor-class'><p>The Neighbor class</p></a></li>
<li><a href='#Neighbor-methods'><p><code>Neighbor</code> Methods</p></a></li>
<li><a href='#old-assign'><p>Original double-bracket assign</p></a></li>
<li><a href='#Overlay'><p>Overlay <code>Spatial</code> Objects Over One Another</p></a></li>
<li><a href='#PackageCheck'><p>Check the existence of a package</p></a></li>
<li><a href='#pbmc_small'><p>A small example version of the PBMC dataset</p></a></li>
<li><a href='#PolyVtx'><p>Polygon Vertices</p></a></li>
<li><a href='#print.DimReduc'><p>Print Top Feature Loadings</p></a></li>
<li><a href='#Project'><p>Get and set project information</p></a></li>
<li><a href='#Radius'><p>Get the spot radius from an image</p></a></li>
<li><a href='#RandomName'><p>Generate a random name</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#RegisterSparseMatrix'><p>Register Sparse Matrix Classes</p></a></li>
<li><a href='#RenameAssays'><p>Rename assays in a <code>Seurat</code> object</p></a></li>
<li><a href='#RenameCells'><p>Rename cells</p></a></li>
<li><a href='#RenameCells-StdAssay'><p>Rename cells</p></a></li>
<li><a href='#RowMergeSparseMatrices'><p>Merge Sparse Matrices by Row</p></a></li>
<li><a href='#s4list'><p>S4/List Conversion</p></a></li>
<li><a href='#SaveSeuratRds'><p>Save and Load <code>Seurat</code> Objects from Rds files</p></a></li>
<li><a href='#Segmentation-class'><p>The <code>Segmentation</code> Class</p></a></li>
<li><a href='#Segmentation-methods'><p><code>Segmentation</code> Methods</p></a></li>
<li><a href='#set-if-na'><p>Set If or If Not <code>NA</code></p></a></li>
<li><a href='#set-if-null'><p>Set If or If Not <code>NULL</code></p></a></li>
<li><a href='#seurat-class'><p>The Seurat Class</p></a></li>
<li><a href='#Seurat-class'><p>The Seurat Class</p></a></li>
<li><a href='#Seurat-validity'><p>Seurat Object Validity</p></a></li>
<li><a href='#SeuratCommand-class'><p>The <code>SeuratCommand</code> Class</p></a></li>
<li><a href='#SeuratObject-options'><p><span class="pkg">Seurat</span> Options</p></a></li>
<li><a href='#show,Assay-method'><p>V3 Assay Overview</p></a></li>
<li><a href='#show,DimReduc-method'><p>Dimensional Reduction Overview</p></a></li>
<li><a href='#show,Graph-method'><p>Graph Object Overview</p></a></li>
<li><a href='#show,LogMap-method'><p><code>LogMap</code> Object Overview</p></a></li>
<li><a href='#show,seurat-method'><p>Old Seurat Object Overview</p></a></li>
<li><a href='#show,Seurat-method'><p>Seurat Object Overview</p></a></li>
<li><a href='#show,SeuratCommand-method'><p>Command Log Overview</p></a></li>
<li><a href='#show,StdAssay-method'><p>V5 Assay Overview</p></a></li>
<li><a href='#Simplify'><p>Simplify Geometry</p></a></li>
<li><a href='#SparseEmptyMatrix'><p>Generate empty dgC sparse matrix</p></a></li>
<li><a href='#SpatialImage-class'><p>The SpatialImage class</p></a></li>
<li><a href='#SpatialImage-methods'><p><code>SpatialImage</code> methods</p></a></li>
<li><a href='#split.Assay'><p>Split an Assay</p></a></li>
<li><a href='#split.Assay5'><p>Split an Assay</p></a></li>
<li><a href='#split.Seurat'><p>Split an Assay</p></a></li>
<li><a href='#split.StdAssay'><p>Split an Assay</p></a></li>
<li><a href='#StdAssay-class'><p>Core Assay Infrastructure</p></a></li>
<li><a href='#StdAssay-validity'><p>V5 Assay Validity</p></a></li>
<li><a href='#Stdev'><p>Get the standard deviations for an object</p></a></li>
<li><a href='#StitchMatrix'><p>Stitch Matrices Together</p></a></li>
<li><a href='#subset.Assay'><p>Subset an Assay</p></a></li>
<li><a href='#subset.Assay5'><p>Subset an Assay</p></a></li>
<li><a href='#subset.DimReduc'><p>Subset a Dimensional Reduction</p></a></li>
<li><a href='#subset.Seurat'><p>Subset <code>Seurat</code> Objects</p></a></li>
<li><a href='#subset.StdAssay'><p>Subset an Assay</p></a></li>
<li><a href='#Theta'><p>Get the offset angle</p></a></li>
<li><a href='#Tool'><p>Get and Set Additional Tool Data</p></a></li>
<li><a href='#UpdateSeuratObject'><p>Update old Seurat object to accommodate new features</p></a></li>
<li><a href='#UpdateSlots'><p>Update slots in an object</p></a></li>
<li><a href='#v5-assay-summaries'><p>V5 Assay Summaries</p></a></li>
<li><a href='#VariableFeatures-StdAssay'><p>Highly Variable Features</p></a></li>
<li><a href='#Version'><p>Get Version Information</p></a></li>
<li><a href='#WhichCells'><p>Identify cells matching certain criteria</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Structures for Single Cell Data</td>
</tr>
<tr>
<td>Version:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Defines S4 classes for single-cell genomic data and associated
  information, such as dimensionality reduction embeddings, nearest-neighbor
  graphs, and spatially-resolved coordinates. Provides data access methods and
  R-native hooks to ensure the Seurat object is familiar to other R users. See
  Satija R, Farrell J, Gennert D, et al (2015) &lt;<a href="https://doi.org/10.1038%2Fnbt.3192">doi:10.1038/nbt.3192</a>&gt;,
  Macosko E, Basu A, Satija R, et al (2015) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2015.05.002">doi:10.1016/j.cell.2015.05.002</a>&gt;,
  and Stuart T, Butler A, et al (2019) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2019.05.031">doi:10.1016/j.cell.2019.05.031</a>&gt;,
  Hao Y, Hao S, et al (2021) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2021.04.048">doi:10.1016/j.cell.2021.04.048</a>&gt; and
  Hao Y, et al (2023) &lt;<a href="https://doi.org/10.1101%2F2022.02.24.481684">doi:10.1101/2022.02.24.481684</a>&gt; for
  more details.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://mojaveazure.github.io/seurat-object/">https://mojaveazure.github.io/seurat-object/</a>,
<a href="https://github.com/mojaveazure/seurat-object">https://github.com/mojaveazure/seurat-object</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mojaveazure/seurat-object/issues">https://github.com/mojaveazure/seurat-object/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Additional_repositories:</td>
<td><a href="https://bnprks.r-universe.dev">https://bnprks.r-universe.dev</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), sp (&ge; 1.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>future, future.apply, grDevices, grid, Matrix (&ge; 1.6.3),
methods, progressr, Rcpp (&ge; 1.0.5), rlang (&ge; 0.4.7), stats,
tools, utils, spam, lifecycle, generics</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DelayedArray, fs (&ge; 1.5.2), ggplot2, HDF5Array, rmarkdown,
testthat, BPCells, sf (&ge; 1.0.0)</td>
</tr>
<tr>
<td>Collate:</td>
<td>'RcppExports.R' 'zzz.R' 'generics.R' 'keymixin.R' 'graph.R'
'default.R' 'assay.R' 'logmap.R' 'layers.R' 'assay5.R'
'centroids.R' 'command.R' 'compliance.R' 'data.R' 'jackstraw.R'
'dimreduc.R' 'segmentation.R' 'molecules.R' 'spatial.R' 'fov.R'
'neighbor.R' 'seurat.R' 'sparse.R' 'utils.R'</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>Enhances:</td>
<td>Seurat</td>
</tr>
<tr>
<td>Config/Needs/website:</td>
<td>pkgdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-16 21:36:12 UTC; paul</td>
</tr>
<tr>
<td>Author:</td>
<td>Rahul Satija <a href="https://orcid.org/0000-0001-9448-8833"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Paul Hoffman <a href="https://orcid.org/0000-0002-7693-8957"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Yuhan Hao <a href="https://orcid.org/0000-0002-1810-0822"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Austin Hartman <a href="https://orcid.org/0000-0001-7278-1852"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Gesmira Molla <a href="https://orcid.org/0000-0002-8628-5056"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Andrew Butler <a href="https://orcid.org/0000-0003-3608-0463"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Tim Stuart <a href="https://orcid.org/0000-0002-3044-0897"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Madeline Kowalski <a href="https://orcid.org/0000-0002-5655-7620"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Saket Choudhary <a href="https://orcid.org/0000-0001-5202-7633"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Skylar Li [ctb],
  Longda Jiang <a href="https://orcid.org/0000-0003-4964-6497"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Jeff Farrell [ctb],
  Shiwei Zheng <a href="https://orcid.org/0000-0001-6682-6743"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Christoph Hafemeister
    <a href="https://orcid.org/0000-0001-6365-8254"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Patrick Roelli [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Paul Hoffman &lt;seurat@nygenome.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-17 22:40:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='SeuratObject-package'>SeuratObject: Data Structures for Single Cell Data</h2><span id='topic+SeuratObject'></span><span id='topic+SeuratObject-package'></span>

<h3>Description</h3>

<p>Defines S4 classes for single-cell genomic data and associated information, such as dimensionality reduction embeddings, nearest-neighbor graphs, and spatially-resolved coordinates. Provides data access methods and R-native hooks to ensure the Seurat object is familiar to other R users. See Satija R, Farrell J, Gennert D, et al (2015) <a href="https://doi.org/10.1038/nbt.3192">doi:10.1038/nbt.3192</a>, Macosko E, Basu A, Satija R, et al (2015) <a href="https://doi.org/10.1016/j.cell.2015.05.002">doi:10.1016/j.cell.2015.05.002</a>, and Stuart T, Butler A, et al (2019) <a href="https://doi.org/10.1016/j.cell.2019.05.031">doi:10.1016/j.cell.2019.05.031</a>, Hao Y, Hao S, et al (2021) <a href="https://doi.org/10.1016/j.cell.2021.04.048">doi:10.1016/j.cell.2021.04.048</a> and Hao Y, et al (2023) <a href="https://doi.org/10.1101/2022.02.24.481684">doi:10.1101/2022.02.24.481684</a> for more details.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Paul Hoffman <a href="mailto:seurat@nygenome.org">seurat@nygenome.org</a> (<a href="https://orcid.org/0000-0002-7693-8957">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Rahul Satija <a href="mailto:rsatija@nygenome.org">rsatija@nygenome.org</a> (<a href="https://orcid.org/0000-0001-9448-8833">ORCID</a>)
</p>
</li>
<li><p> Yuhan Hao <a href="mailto:yhao@nygenome.org">yhao@nygenome.org</a> (<a href="https://orcid.org/0000-0002-1810-0822">ORCID</a>)
</p>
</li>
<li><p> Austin Hartman <a href="mailto:ahartman@nygenome.org">ahartman@nygenome.org</a> (<a href="https://orcid.org/0000-0001-7278-1852">ORCID</a>)
</p>
</li>
<li><p> Gesmira Molla <a href="mailto:gmolla@nygenome.org">gmolla@nygenome.org</a> (<a href="https://orcid.org/0000-0002-8628-5056">ORCID</a>)
</p>
</li>
<li><p> Andrew Butler <a href="mailto:abutler@nygenome.org">abutler@nygenome.org</a> (<a href="https://orcid.org/0000-0003-3608-0463">ORCID</a>)
</p>
</li>
<li><p> Tim Stuart <a href="mailto:tstuart@nygenome.org">tstuart@nygenome.org</a> (<a href="https://orcid.org/0000-0002-3044-0897">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Madeline Kowalski <a href="mailto:mkowalski@nygenome.org">mkowalski@nygenome.org</a> (<a href="https://orcid.org/0000-0002-5655-7620">ORCID</a>) [contributor]
</p>
</li>
<li><p> Saket Choudhary <a href="mailto:schoudhary@nygenome.org">schoudhary@nygenome.org</a> (<a href="https://orcid.org/0000-0001-5202-7633">ORCID</a>) [contributor]
</p>
</li>
<li><p> Skylar Li <a href="mailto:sli@nygenome.org">sli@nygenome.org</a> [contributor]
</p>
</li>
<li><p> Longda Jiang <a href="mailto:ljiang@nygenome.org">ljiang@nygenome.org</a> (<a href="https://orcid.org/0000-0003-4964-6497">ORCID</a>) [contributor]
</p>
</li>
<li><p> Jeff Farrell <a href="mailto:jfarrell@g.harvard.edu">jfarrell@g.harvard.edu</a> [contributor]
</p>
</li>
<li><p> Shiwei Zheng <a href="mailto:szheng@nygenome.org">szheng@nygenome.org</a> (<a href="https://orcid.org/0000-0001-6682-6743">ORCID</a>) [contributor]
</p>
</li>
<li><p> Christoph Hafemeister <a href="mailto:chafemeister@nygenome.org">chafemeister@nygenome.org</a> (<a href="https://orcid.org/0000-0001-6365-8254">ORCID</a>) [contributor]
</p>
</li>
<li><p> Patrick Roelli <a href="mailto:proelli@nygenome.org">proelli@nygenome.org</a> [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://mojaveazure.github.io/seurat-object/">https://mojaveazure.github.io/seurat-object/</a>
</p>
</li>
<li> <p><a href="https://github.com/mojaveazure/seurat-object">https://github.com/mojaveazure/seurat-object</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/mojaveazure/seurat-object/issues">https://github.com/mojaveazure/seurat-object/issues</a>
</p>
</li></ul>


<hr>
<h2 id='.AssayClass'>Assay Class Label</h2><span id='topic+.AssayClass'></span><span id='topic+.AssayClass.StdAssay'></span><span id='topic+.AssayClass.default'></span>

<h3>Description</h3>

<p>Assay Class Label
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.AssayClass(object)

## S3 method for class 'StdAssay'
.AssayClass(object)

## Default S3 method:
.AssayClass(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".AssayClass_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+StdAssay">StdAssay</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The assay class label for <code>object</code>
</p>

<hr>
<h2 id='.BPMatrixMode'><span class="pkg">BPCells</span> Matrix Mode</h2><span id='topic+.BPMatrixMode'></span>

<h3>Description</h3>

<p>Get the mode (on-disk, in-memory) of an <code>IterableMatrix</code> object
from <span class="pkg">BPCells</span>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.BPMatrixMode(object, simplify = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".BPMatrixMode_+3A_object">object</code></td>
<td>
<p>An <code>IterableMatrix</code></p>
</td></tr>
<tr><td><code id=".BPMatrixMode_+3A_simplify">simplify</code></td>
<td>
<p>Return &ldquo;<code>disk</code>&rdquo; for on-disk matrices</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One of the following, depending on the mode of <code>object</code>:
</p>

<ul>
<li> <p>&ldquo;<code>memory</code>&rdquo;
</p>
</li>
<li> <p>&ldquo;<code>file</code>&rdquo;
</p>
</li>
<li> <p>&ldquo;<code>directory</code>&rdquo;
</p>
</li></ul>

<p>If <code>simplify</code> is <code>TRUE</code>, returns &ldquo;<code>disk</code>&rdquo; instead of
&ldquo;<code>file</code>&rdquo; or &ldquo;<code>directory</code>&rdquo;
</p>

<hr>
<h2 id='.CalcN'>Calculate nCount and nFeature</h2><span id='topic+.CalcN'></span>

<h3>Description</h3>

<p>Calculate nCount and nFeature
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.CalcN(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".CalcN_+3A_object">object</code></td>
<td>
<p>An assay-like object</p>
</td></tr>
<tr><td><code id=".CalcN_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list with ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>calcn &lt;- .CalcN(pbmc_small[["RNA"]])
head(as.data.frame(calcn))

</code></pre>

<hr>
<h2 id='.CheckFmargin'>Check Feature Margin</h2><span id='topic+.CheckFmargin'></span>

<h3>Description</h3>

<p>Check Feature Margin
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.CheckFmargin(fmargin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".CheckFmargin_+3A_fmargin">fmargin</code></td>
<td>
<p>Either <code>1</code> or <code>2</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>fmargin</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.CheckFmargin(1L)
.CheckFmargin(2.3)

# Error if `fmargin` is outside of [1:2]
if (FALSE) {
  .CheckFmargin(3L)
}

</code></pre>

<hr>
<h2 id='.ClassPkg'>Get the Package that Defines a Class</h2><span id='topic+.ClassPkg'></span><span id='topic+.ClassPkg.default'></span><span id='topic+.ClassPkg.DelayedArray'></span><span id='topic+.ClassPkg.R6'></span><span id='topic+.ClassPkg.R6ClassGenerator'></span>

<h3>Description</h3>

<p>Get the Package that Defines a Class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.ClassPkg(object)

## Default S3 method:
.ClassPkg(object)

## S3 method for class 'DelayedArray'
.ClassPkg(object)

## S3 method for class 'R6'
.ClassPkg(object)

## S3 method for class 'R6ClassGenerator'
.ClassPkg(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".ClassPkg_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The package that defines the class of <code>object</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.ClassPkg(pbmc_small)

</code></pre>

<hr>
<h2 id='.Collections'>Identify Object Collections</h2><span id='topic+.Collections'></span>

<h3>Description</h3>

<p>Find all collection (named lists) slots in an S4 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.Collections(object, exclude = character(length = 0L), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".Collections_+3A_object">object</code></td>
<td>
<p>An <a href="methods.html#topic+Classes_Details">S4</a> object</p>
</td></tr>
<tr><td><code id=".Collections_+3A_exclude">exclude</code></td>
<td>
<p>A character vector of slot names to exclude</p>
</td></tr>
<tr><td><code id=".Collections_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+IsNamedList">IsNamedList</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of names of collection slots
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.FilterObjects">.FilterObjects</a>()</code>,
<code><a href="#topic+.FindObject">.FindObject</a>()</code>,
<code><a href="#topic+.Subobjects">.Subobjects</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.Collections(pbmc_small)

</code></pre>

<hr>
<h2 id='.Contains'>Get Parent S4 Classes</h2><span id='topic+.Contains'></span>

<h3>Description</h3>

<p>Get Parent S4 Classes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.Contains(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".Contains_+3A_object">object</code></td>
<td>
<p>An <a href="methods.html#topic+Classes_Details">S4</a> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of class names that <code>object</code> inherits from
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.Contains(pbmc_small)

</code></pre>

<hr>
<h2 id='.CreateStdAssay'>Generic Assay Creation</h2><span id='topic+.CreateStdAssay'></span><span id='topic+.CreateStdAssay.default'></span><span id='topic+.CreateStdAssay.list'></span><span id='topic+.CreateStdAssay.Matrix'></span><span id='topic+.CreateStdAssay.matrix'></span>

<h3>Description</h3>

<p>Create an assay object; runs a standardized filtering scheme that
works regardless of the direction of the data (eg. cells as columns
and features as rows or vice versa) and creates an assay object based
on the  initialization scheme defined for <code><a href="#topic+StdAssay">StdAssay</a></code>-derived
class <code>type</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.CreateStdAssay(
  counts,
  min.cells = 0,
  min.features = 0,
  cells = NULL,
  features = NULL,
  transpose = FALSE,
  type = "Assay5",
  ...
)

## Default S3 method:
.CreateStdAssay(
  counts,
  min.cells = 0,
  min.features = 0,
  cells = NULL,
  features = NULL,
  transpose = FALSE,
  type = "Assay5",
  layer = "counts",
  ...
)

## S3 method for class 'list'
.CreateStdAssay(
  counts,
  min.cells = 0,
  min.features = 0,
  cells = NULL,
  features = NULL,
  transpose = FALSE,
  type = "Assay5",
  csum = Matrix::colSums,
  fsum = Matrix::rowSums,
  ...
)

## S3 method for class 'Matrix'
.CreateStdAssay(
  counts,
  min.cells = 0,
  min.features = 0,
  cells = NULL,
  features = NULL,
  transpose = FALSE,
  type = "Assay5",
  layer = "counts",
  ...
)

## S3 method for class 'matrix'
.CreateStdAssay(
  counts,
  min.cells = 0,
  min.features = 0,
  cells = NULL,
  features = NULL,
  transpose = FALSE,
  type = "Assay5",
  layer = "counts",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".CreateStdAssay_+3A_counts">counts</code></td>
<td>
<p>A two-dimensional expression matrix</p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_min.cells">min.cells</code></td>
<td>
<p>Include features detected in at least this many cells;
will subset the counts matrix as well. To reintroduce excluded features,
create a new object with a lower cutoff</p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_min.features">min.features</code></td>
<td>
<p>Include cells where at least this many features
are detected</p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_cells">cells</code></td>
<td>
<p>Vector of cell names</p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_features">features</code></td>
<td>
<p>Vector of feature names</p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_type">type</code></td>
<td>
<p>Type of assay object to create; must be the name of a class
that's derived from <code><a href="#topic+StdAssay">StdAssay</a></code></p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="methods.html#topic+new">new</a></code> for
assay creation; used to set slots not defined by <code><a href="#topic+StdAssay">StdAssay</a></code></p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_layer">layer</code></td>
<td>
<p>Name of layer to store <code>counts</code> as</p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_csum">csum</code></td>
<td>
<p>Function for calculating cell sums</p>
</td></tr>
<tr><td><code id=".CreateStdAssay_+3A_fsum">fsum</code></td>
<td>
<p>Function for calculating feature sums</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>type</code> with a layer named <code>layer</code>
containing the data found in <code>counts</code>
</p>

<hr>
<h2 id='.DefaultFOV'>Find the Default FOV</h2><span id='topic+.DefaultFOV'></span>

<h3>Description</h3>

<p>Attempts to find the &ldquo;default&rdquo; FOV using the revamped
spatial framework
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.DefaultFOV(object, assay = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".DefaultFOV_+3A_object">object</code></td>
<td>
<p>A <code>{Seurat}</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>...
</p>

<hr>
<h2 id='.Deprecate'>Deprecate Functions and Arguments</h2><span id='topic+.Deprecate'></span>

<h3>Description</h3>

<p>Provides automatic deprecation and defunctation of functions and arguments;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.Deprecate(
  when,
  what,
  with = NULL,
  ...,
  pkg = NULL,
  env = missing_arg(),
  user_env = missing_arg()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".Deprecate_+3A_when">when</code></td>
<td>
<p>A string giving the version when the behaviour was deprecated.</p>
</td></tr>
<tr><td><code id=".Deprecate_+3A_what">what</code></td>
<td>
<p>A string describing what is deprecated:
</p>

<ul>
<li><p> Deprecate a whole function with <code>"foo()"</code>.
</p>
</li>
<li><p> Deprecate an argument with <code>"foo(arg)"</code>.
</p>
</li>
<li><p> Partially deprecate an argument with
<code>"foo(arg = 'must be a scalar integer')"</code>.
</p>
</li>
<li><p> Deprecate anything else with a custom message by wrapping it in <code>I()</code>.
</p>
</li></ul>

<p>You can optionally supply the namespace: <code>"ns::foo()"</code>, but this is
usually not needed as it will be inferred from the caller environment.</p>
</td></tr>
<tr><td><code id=".Deprecate_+3A_with">with</code></td>
<td>
<p>An optional string giving a recommended replacement for the
deprecated behaviour. This takes the same form as <code>what</code>.</p>
</td></tr>
<tr><td><code id=".Deprecate_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="lifecycle.html#topic+deprecate_soft">lifecycle::deprecate_soft</a></code>
</p>

<dl>
<dt><code>details</code></dt><dd><p>In most cases the deprecation message can be
automatically generated from <code>with</code>. When it can't, use <code>details</code>
to provide a hand-written message.
</p>
<p><code>details</code> can either be a single string or a character vector,
which will be converted to a <a href="cli.html#topic+cli_bullets">bulleted list</a>.
By default, info bullets are used. Provide a named vectors to
override.</p>
</dd>
<dt><code>id</code></dt><dd><p>The id of the deprecation. A warning is issued only once
for each <code>id</code>. Defaults to the generated message, but you should
give a unique ID when the message in <code>details</code> is built
programmatically and depends on inputs, or when you'd like to
deprecate multiple functions but warn only once for all of them.</p>
</dd>
<dt><code>env,user_env</code></dt><dd><p>Pair of environments that define where <code style="white-space: pre;">&#8288;deprecate_*()&#8288;</code>
was called (used to determine the package name) and where the function
called the deprecating function was called (used to determine if
<code>deprecate_soft()</code> should message).
</p>
<p>These are only needed if you're calling <code style="white-space: pre;">&#8288;deprecate_*()&#8288;</code> from an internal
helper, in which case you should forward <code>env = caller_env()</code> and
<code>user_env = caller_env(2)</code>.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id=".Deprecate_+3A_pkg">pkg</code></td>
<td>
<p>Name of package to use for comparison</p>
</td></tr>
<tr><td><code id=".Deprecate_+3A_env">env</code>, <code id=".Deprecate_+3A_user_env">user_env</code></td>
<td>
<p>Managed internally by <code>.Deprecate()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Run for its side effect and invisibly returns <code>NULL</code>
</p>


<h3>See Also</h3>

<p><code><a href="lifecycle.html#topic+deprecate_soft">lifecycle::deprecate_soft</a>()</code>
<code><a href="lifecycle.html#topic+deprecate_warn">lifecycle::deprecate_warn</a>()</code>
<code><a href="lifecycle.html#topic+deprecate_stop">lifecycle::deprecate_stop</a>()</code>
</p>

<hr>
<h2 id='.DiskLoad'>Disk Loading Function</h2><span id='topic+.DiskLoad'></span><span id='topic+.DiskLoad.default'></span><span id='topic+.DiskLoad.10xMatrixH5'></span><span id='topic+.DiskLoad.AnnDataMatrixH5'></span><span id='topic+.DiskLoad.DelayedMatrix'></span><span id='topic+.DiskLoad.H5ADMatrix'></span><span id='topic+.DiskLoad.HDF5Matrix'></span><span id='topic+.DiskLoad.IterableMatrix'></span><span id='topic+.DiskLoad.MatrixDir'></span><span id='topic+.DiskLoad.MatrixH5'></span><span id='topic+.DiskLoad.TileDBMatrix'></span>

<h3>Description</h3>

<p>Generate a function to load a matrix from an on-disk file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.DiskLoad(x)

## Default S3 method:
.DiskLoad(x)

## S3 method for class ''10xMatrixH5''
.DiskLoad(x)

## S3 method for class 'AnnDataMatrixH5'
.DiskLoad(x)

## S3 method for class 'DelayedMatrix'
.DiskLoad(x)

## S3 method for class 'H5ADMatrix'
.DiskLoad(x)

## S3 method for class 'HDF5Matrix'
.DiskLoad(x)

## S3 method for class 'IterableMatrix'
.DiskLoad(x)

## S3 method for class 'MatrixDir'
.DiskLoad(x)

## S3 method for class 'MatrixH5'
.DiskLoad(x)

## S3 method for class 'TileDBMatrix'
.DiskLoad(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".DiskLoad_+3A_x">x</code></td>
<td>
<p>A file-backed object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A one-length character that defines a function to load a matrix from
a file
</p>

<hr>
<h2 id='.DollarNames.Assay'>Dollar-sign Autocompletion</h2><span id='topic+.DollarNames.Assay'></span>

<h3>Description</h3>

<p>Autocompletion for <code>$</code> access on an
<code><a href="#topic+Assay">Assay</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
.DollarNames(x, pattern = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".DollarNames.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id=".DollarNames.Assay_+3A_pattern">pattern</code></td>
<td>
<p> A regular expression.  Only matching names are
returned.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The layer name matches for <code>pattern</code>
</p>


<h3>See Also</h3>

<p><code><a href="utils.html#topic+.DollarNames">utils::.DollarNames</a></code>
</p>

<hr>
<h2 id='.DollarNames.Assay5'>Dollar-sign Autocompletion</h2><span id='topic+.DollarNames.Assay5'></span>

<h3>Description</h3>

<p>Autocompletion for <code>$</code> access on an <code><a href="#topic+Assay5">Assay5</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
.DollarNames(x, pattern = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".DollarNames.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id=".DollarNames.Assay5_+3A_pattern">pattern</code></td>
<td>
<p> A regular expression.  Only matching names are
returned.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The layer name matches for <code>pattern</code>
</p>


<h3>See Also</h3>

<p><code><a href="utils.html#topic+.DollarNames">utils::.DollarNames</a></code>
</p>

<hr>
<h2 id='.DollarNames.Seurat'>Dollar-sign Autocompletion</h2><span id='topic+.DollarNames.Seurat'></span>

<h3>Description</h3>

<p>Autocompletion for <code>$</code> access on a <code><a href="#topic+Seurat">Seurat</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
.DollarNames(x, pattern = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".DollarNames.Seurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id=".DollarNames.Seurat_+3A_pattern">pattern</code></td>
<td>
<p> A regular expression.  Only matching names are
returned.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The meta data matches for <code>pattern</code>
</p>


<h3>See Also</h3>

<p><code><a href="utils.html#topic+.DollarNames">utils::.DollarNames</a></code>
</p>

<hr>
<h2 id='.DollarNames.SeuratCommand'>Dollar-sign Autocompletion</h2><span id='topic+.DollarNames.SeuratCommand'></span>

<h3>Description</h3>

<p>Autocompletion for <code>$</code> access on a
<code><a href="#topic+SeuratCommand">SeuratCommand</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SeuratCommand'
.DollarNames(x, pattern = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".DollarNames.SeuratCommand_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+SeuratCommand">SeuratCommand</a></code> object</p>
</td></tr>
<tr><td><code id=".DollarNames.SeuratCommand_+3A_pattern">pattern</code></td>
<td>
<p> A regular expression.  Only matching names are
returned.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The parameter name matches for <code>pattern</code>
</p>


<h3>See Also</h3>

<p>Command log object and interaction methods
<code><a href="#topic++24.SeuratCommand">$.SeuratCommand</a>()</code>,
<code><a href="#topic+LogSeuratCommand">LogSeuratCommand</a>()</code>,
<code><a href="#topic+SeuratCommand-class">SeuratCommand-class</a></code>,
<code><a href="#topic++5B.SeuratCommand">[.SeuratCommand</a>()</code>,
<code><a href="#topic+as.list.SeuratCommand">as.list.SeuratCommand</a>()</code>
</p>

<hr>
<h2 id='.DollarNames.StdAssay'>Dollar-sign Autocompletion</h2><span id='topic+.DollarNames.StdAssay'></span>

<h3>Description</h3>

<p>Autocompletion for <code>$</code> access on an <code><a href="#topic+Assay5">Assay5</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
.DollarNames(x, pattern = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".DollarNames.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id=".DollarNames.StdAssay_+3A_pattern">pattern</code></td>
<td>
<p> A regular expression.  Only matching names are
returned.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The layer name matches for <code>pattern</code>
</p>


<h3>See Also</h3>

<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='.FileMove'>Move Files and Directories</h2><span id='topic+.FileMove'></span>

<h3>Description</h3>

<p>Move files and directories with <span class="pkg">fs</span>; includes a handler for when
<code>path</code> is a directory on a different filesystem than <code>new_path</code>
by explicitly copying and deleting <code>path</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.FileMove(path, new_path, overwrite = FALSE, n = 1L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".FileMove_+3A_path">path</code></td>
<td>
<p>A character vector of one or more paths.</p>
</td></tr>
<tr><td><code id=".FileMove_+3A_new_path">new_path</code></td>
<td>
<p>New file path. If <code>new_path</code> is existing directory, the file
will be moved into that directory; otherwise it will be moved/renamed to
the full path.
</p>
<p>Should either be the same length as <code>path</code>, or a single directory.</p>
</td></tr>
<tr><td><code id=".FileMove_+3A_n">n</code></td>
<td>
<p>The number of callers to go back.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The new path (invisibly).
</p>


<h3>Note</h3>

<p>This function requires the
<a href="https://cran.r-project.org/package=fs"><span class="pkg">fs</span></a> package
to be installed
</p>


<h3>See Also</h3>

<p><code><a href="fs.html#topic+file_move">fs::file_move</a>()</code>
</p>

<hr>
<h2 id='.FilePath'>Find a File Path</h2><span id='topic+.FilePath'></span><span id='topic+.FilePath.default'></span><span id='topic+.FilePath.DelayedMatrix'></span><span id='topic+.FilePath.IterableMatrix'></span>

<h3>Description</h3>

<p>Find a File Path
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.FilePath(x)

## Default S3 method:
.FilePath(x)

## S3 method for class 'DelayedMatrix'
.FilePath(x)

## S3 method for class 'IterableMatrix'
.FilePath(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".FilePath_+3A_x">x</code></td>
<td>
<p>A file-backed object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path to the file that backs <code>x</code>; if <code>x</code> is not a
file-backed object, returns <code>NULL</code>
</p>

<hr>
<h2 id='.FilterObjects'>Find Subobjects Of A Certain Class</h2><span id='topic+.FilterObjects'></span>

<h3>Description</h3>

<p>Find Subobjects Of A Certain Class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.FilterObjects(object, classes.keep = c("Assay", "StdAssay", "DimReduc"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".FilterObjects_+3A_object">object</code></td>
<td>
<p>An <a href="methods.html#topic+Classes_Details">S4</a> object</p>
</td></tr>
<tr><td><code id=".FilterObjects_+3A_classes.keep">classes.keep</code></td>
<td>
<p>A vector of classes to keep</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of object names that are of class <code>classes.keep</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.Collections">.Collections</a>()</code>,
<code><a href="#topic+.FindObject">.FindObject</a>()</code>,
<code><a href="#topic+.Subobjects">.Subobjects</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.FilterObjects(pbmc_small)
.FilterObjects(pbmc_small, "Graph")

</code></pre>

<hr>
<h2 id='.FindObject'>Find A Subobject</h2><span id='topic+.FindObject'></span>

<h3>Description</h3>

<p>Determine the slot that a subobject is contained in
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.FindObject(object, name, exclude = c("misc", "tools"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".FindObject_+3A_object">object</code></td>
<td>
<p>An <a href="methods.html#topic+Classes_Details">S4</a> object</p>
</td></tr>
<tr><td><code id=".FindObject_+3A_name">name</code></td>
<td>
<p>Name of subobject to find</p>
</td></tr>
<tr><td><code id=".FindObject_+3A_exclude">exclude</code></td>
<td>
<p>A character vector of slot names to exclude</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The name of the slot that contains <code>name</code>; returns <code>NULL</code>
if a subobject named <code>name</code> cannot be found
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.Collections">.Collections</a>()</code>,
<code><a href="#topic+.FilterObjects">.FilterObjects</a>()</code>,
<code><a href="#topic+.Subobjects">.Subobjects</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.FindObject(pbmc_small, "tsne")

</code></pre>

<hr>
<h2 id='.GetMethod'>Get a Method</h2><span id='topic+.GetMethod'></span>

<h3>Description</h3>

<p>Get a Method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.GetMethod(fxn, cls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".GetMethod_+3A_fxn">fxn</code></td>
<td>
<p>Name of a function as a character</p>
</td></tr>
<tr><td><code id=".GetMethod_+3A_cls">cls</code></td>
<td>
<p>The class to find a method of <code>fxn</code> for</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The method of <code>fxn</code> for class <code>cls</code>; if no method found,
returns the default method. If no default method found; returns <code>NULL</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.GetMethod('t', 'Matrix')
.GetMethod('t', 'data.frame')

</code></pre>

<hr>
<h2 id='.IsFutureSeurat'>Test Future Compatibility with <span class="pkg">Seurat</span></h2><span id='topic+.IsFutureSeurat'></span><span id='topic+IsFutureSeurat'></span>

<h3>Description</h3>

<p>Check to see if <span class="pkg">SeuratObject</span> and/or <span class="pkg">Seurat</span> are at least a
specific version or if they're configured to act as if they're a
specific version (see details below). This allows testing compatibility with
future requirements for both <span class="pkg">SeuratObject</span> and <span class="pkg">Seurat</span>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.IsFutureSeurat(version, lib.loc = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".IsFutureSeurat_+3A_version">version</code></td>
<td>
<p>A version string or object of class
<code><a href="base.html#topic+package_version">package_version</a></code></p>
</td></tr>
<tr><td><code id=".IsFutureSeurat_+3A_lib.loc">lib.loc</code></td>
<td>
<p>a character vector of directory names of <span class="rlang"><b>R</b></span> libraries,
or <code>NULL</code>.  The default value of <code>NULL</code> corresponds to all
libraries currently known.  If the default is used, the loaded
packages and namespaces are searched before the libraries.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Blah blah blah
</p>


<h3>Value</h3>

<p><code>TRUE</code> if <span class="pkg">SeuratObject</span> and/or <span class="pkg">Seurat</span>
</p>

<hr>
<h2 id='.KeyPattern'>Regex Pattern for Keys</h2><span id='topic+.KeyPattern'></span>

<h3>Description</h3>

<p>Regex Pattern for Keys
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.KeyPattern()
</code></pre>


<h3>Value</h3>

<p>Returns the regex pattern for keys
(&ldquo;^[a-zA-Z][a-zA-Z0-9]*_$&rdquo;)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.RandomKey">.RandomKey</a>()</code>,
<code><a href="#topic+Key-validity">Key-validity</a></code>,
<code><a href="#topic+KeyMixin-class">KeyMixin-class</a></code>
</p>

<hr>
<h2 id='.MARGIN'>Get the Margin of an Object</h2><span id='topic+.MARGIN'></span>

<h3>Description</h3>

<p>Get the Margin of an Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.MARGIN(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".MARGIN_+3A_x">x</code></td>
<td>
<p>An object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The margin, eg. <code>1</code> for rows or <code>2</code> for columns
</p>

<hr>
<h2 id='.PropagateList'>Propagate a List</h2><span id='topic+.PropagateList'></span>

<h3>Description</h3>

<p>Propagate a List
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.PropagateList(x, names, default = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".PropagateList_+3A_x">x</code></td>
<td>
<p>A list or character vector</p>
</td></tr>
<tr><td><code id=".PropagateList_+3A_names">names</code></td>
<td>
<p>A vector of names to keep from <code>x</code></p>
</td></tr>
<tr><td><code id=".PropagateList_+3A_default">default</code></td>
<td>
<p>A default value for unassigned values of <code>x</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list where the names are present in both <code>x</code> and
<code>names</code> and the values are either the values from <code>x</code> or
<code>default</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.PropagateList("counts", c("RNA", "ADT", "SCT"))
.PropagateList(c("counts", "data"), c("RNA", "ADT", "SCT"))
.PropagateList("ADT", c("RNA", "ADT", "SCT"))
.PropagateList(c("RNA", "SCT"), c("RNA", "ADT", "SCT"))
.PropagateList(c("RNA", ADT = "counts"), c("RNA", "ADT", "SCT"))
.PropagateList(list(SCT = c("counts", "data"), ADT = "counts"), c("RNA", "ADT", "SCT"))
.PropagateList(list(SCT = c("counts", "data"), "ADT"), c("RNA", "ADT", "SCT"))

</code></pre>

<hr>
<h2 id='.RandomKey'>Generate a Random Key</h2><span id='topic+.RandomKey'></span>

<h3>Description</h3>

<p>Generate a Random Key
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.RandomKey(length = 7L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".RandomKey_+3A_length">length</code></td>
<td>
<p>How long should the name be</p>
</td></tr>
<tr><td><code id=".RandomKey_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="base.html#topic+sample">sample</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a valid key
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.KeyPattern">.KeyPattern</a>()</code>,
<code><a href="#topic+Key-validity">Key-validity</a></code>,
<code><a href="#topic+KeyMixin-class">KeyMixin-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(42L)
.RandomKey()

</code></pre>

<hr>
<h2 id='.SelectFeatures'>Combine and Select Features</h2><span id='topic+.SelectFeatures'></span><span id='topic+.SelectFeatures.list'></span>

<h3>Description</h3>

<p>Combine and Select Features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.SelectFeatures(object, ...)

## S3 method for class 'list'
.SelectFeatures(object, all.features = NULL, nfeatures = Inf, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".SelectFeatures_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id=".SelectFeatures_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of features selected
</p>

<hr>
<h2 id='.SparseSlots'>Identify Sparse Slots</h2><span id='topic+.SparseSlots'></span><span id='topic+.SparseSlots.CsparseMatrix'></span><span id='topic+.SparseSlots.RsparseMatrix'></span><span id='topic+.SparseSlots.spam'></span>

<h3>Description</h3>

<p>Identify Sparse Slots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.SparseSlots(x, type = c("pointers", "indices", "entries"))

## S3 method for class 'CsparseMatrix'
.SparseSlots(x, type = c("pointers", "entries", "indices"))

## S3 method for class 'RsparseMatrix'
.SparseSlots(x, type = c("pointers", "indices", "entries"))

## S3 method for class 'spam'
.SparseSlots(x, type = c("pointers", "indices", "entries"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".SparseSlots_+3A_x">x</code></td>
<td>
<p>A sparse matrix</p>
</td></tr>
<tr><td><code id=".SparseSlots_+3A_type">type</code></td>
<td>
<p>...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>...
</p>


<h3>See Also</h3>

<p><code><a href="#topic+IsSparse">IsSparse</a>()</code>,
<code><a href="#topic+RegisterSparseMatrix">RegisterSparseMatrix</a>()</code>
</p>

<hr>
<h2 id='.Subobjects'>Get the Subobject Names</h2><span id='topic+.Subobjects'></span>

<h3>Description</h3>

<p>Get the Subobject Names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.Subobjects(object, exclude = c("misc", "tools"), collapse = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".Subobjects_+3A_object">object</code></td>
<td>
<p>An <a href="methods.html#topic+Classes_Details">S4</a> object</p>
</td></tr>
<tr><td><code id=".Subobjects_+3A_exclude">exclude</code></td>
<td>
<p>A character vector of slot names to exclude</p>
</td></tr>
<tr><td><code id=".Subobjects_+3A_collapse">collapse</code></td>
<td>
<p>Collapse the list into a vector</p>
</td></tr>
<tr><td><code id=".Subobjects_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+IsNamedList">IsNamedList</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>collapse = TRUE</code>, then a vector with the names of all
subobjects; otherwise, a named list where the names are the names of the
collections and the values are the names of subobjects within the collection
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.Collections">.Collections</a>()</code>,
<code><a href="#topic+.FilterObjects">.FilterObjects</a>()</code>,
<code><a href="#topic+.FindObject">.FindObject</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.Subobjects(pbmc_small)

</code></pre>

<hr>
<h2 id='+5B+2CLogMap'>Matrix-like Subsetting for <a href="#topic+LogMap">Logical Maps</a></h2><span id='topic++5B+2CLogMap'></span><span id='topic++5B+2CLogMap+2Cmissing+2Cmissing+2CANY-method'></span><span id='topic++5B+2CLogMap+2Ccharacter+2Ccharacter+2CANY-method'></span><span id='topic++5B+2CLogMap+2Ccharacter+2Cmissing+2CANY-method'></span><span id='topic++5B+2CLogMap+2Cmissing+2Ccharacter+2CANY-method'></span><span id='topic++5B+2CLogMap+2Cnumeric+2Cmissing+2CANY-method'></span><span id='topic++5B+2CLogMap+2Cmissing+2Cnumeric+2CANY-method'></span><span id='topic++5B+2CLogMap+2Cnumeric+2Cnumeric+2CANY-method'></span>

<h3>Description</h3>

<p>Matrix-like Subsetting for <a href="#topic+LogMap">Logical Maps</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'LogMap,missing,missing,ANY'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'LogMap,character,character,ANY'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'LogMap,character,missing,ANY'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'LogMap,missing,character,ANY'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'LogMap,numeric,missing,ANY'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'LogMap,missing,numeric,ANY'
x[i, j, ..., drop = FALSE]

## S4 method for signature 'LogMap,numeric,numeric,ANY'
x[i, j, ..., drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2CLogMap_+3A_x">x</code></td>
<td>
<p>A <code>LogMap</code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CLogMap_+3A_i">i</code>, <code id="+2B5B+2B2CLogMap_+3A_j">j</code></td>
<td>
<p>Vectors of values (<code>i</code>) and observations (<code>j</code>) to pull
from <code>x</code></p>
</td></tr>
<tr><td><code id="+2B5B+2B2CLogMap_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CLogMap_+3A_drop">drop</code></td>
<td>
<p>For matrices and arrays.  If <code>TRUE</code> the result is
coerced to the lowest possible dimension (see the examples).  This
only works for extracting elements, not for the replacement.  See
<code><a href="base.html#topic+drop">drop</a></code> for further details.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p><code>i</code> is not reordable; passing a different order for <code>i</code>
will return a subset with rows in the same order as <code>x</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map &lt;- LogMap(letters[1:10])
map[['obs']] &lt;- c(1, 3, 7)
map[['entry']] &lt;- c(2, 7, 10)

map[]
map[1:5, 2L]
map[c("b", "c", "f"), "obs"]

# Pass `drop = TRUE` to cast to `matrix`
map[1:3, , drop = TRUE]

# Note that `i` is non-reordable
rownames(map)[1:3]
map[c("b", "c", "a"), , drop = TRUE]

</code></pre>

<hr>
<h2 id='+5B.Assay'>Layer Data</h2><span id='topic++5B.Assay'></span><span id='topic++5B+3C-+2CAssay+2Ccharacter+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>Get and set layer data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
x[i = missing_arg(), j = missing_arg(), ...]

## S4 replacement method for signature 'Assay,character,ANY,ANY'
x[i, j, ...] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B.Assay_+3A_i">i</code></td>
<td>
<p>Name of layer data to get or set</p>
</td></tr>
<tr><td><code id="+2B5B.Assay_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B.Assay_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+LayerData">LayerData</a></code></p>
</td></tr>
<tr><td><code id="+2B5B.Assay_+3A_value">value</code></td>
<td>
<p>A matrix-like object to add as a new layer</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>[</code>: The layer data for layer <code>i</code>
</p>
<p><code>[&lt;-</code>: <code>x</code> with layer data <code>value</code> saved as <code>i</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+LayerData">LayerData</a></code>
</p>
<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]

# Get a vector of layer names in this assay
rna[]

# Fetch layer data
rna["data"][1:10, 1:4]

# Set layer data
rna["data"] &lt;- rna["counts"]
rna["data"][1:10, 1:4]

</code></pre>

<hr>
<h2 id='+5B.Assay5'>Layer Data</h2><span id='topic++5B.Assay5'></span><span id='topic++5B+3C-+2CAssay5+2Ccharacter+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>Get and set layer data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
x[i = missing_arg(), j = missing_arg(), ...]

## S4 replacement method for signature 'Assay5,character,ANY,ANY'
x[i, j, ...] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B.Assay5_+3A_i">i</code></td>
<td>
<p>Name of layer data to get or set</p>
</td></tr>
<tr><td><code id="+2B5B.Assay5_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B.Assay5_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+LayerData">LayerData</a></code></p>
</td></tr>
<tr><td><code id="+2B5B.Assay5_+3A_value">value</code></td>
<td>
<p>A matrix-like object to add as a new layer</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>[</code>: The layer data for layer <code>i</code>
</p>
<p><code>[&lt;-</code>: <code>x</code> with layer data <code>value</code> saved as <code>i</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+LayerData">LayerData</a></code>
</p>
<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>

<hr>
<h2 id='+5B.DimReduc'>Get Feature Loadings</h2><span id='topic++5B.DimReduc'></span>

<h3>Description</h3>

<p>Pull feature loadings from a <a href="#topic+DimReduc">dimensional reduction</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DimReduc'
x[i, j, drop = FALSE, ...]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.DimReduc_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+DimReduc">DimReduc</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B.DimReduc_+3A_i">i</code></td>
<td>
<p>Feature identifiers or indices</p>
</td></tr>
<tr><td><code id="+2B5B.DimReduc_+3A_j">j</code></td>
<td>
<p>Dimension identifiers or indices</p>
</td></tr>
<tr><td><code id="+2B5B.DimReduc_+3A_drop">drop</code></td>
<td>
<p>Coerce the result to the lowest possible dimension; see
<code><a href="base.html#topic+drop">drop</a></code> for further details</p>
</td></tr>
<tr><td><code id="+2B5B.DimReduc_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>[</code> does not distinguish between projected and unprojected feature
loadings; to select whether projected or unprojected loadings should be
pulled, please use <code><a href="#topic+Loadings">Loadings</a></code>
</p>


<h3>Value</h3>

<p>Feature loadings for features <code>i</code> and dimensions <code>j</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Loadings">Loadings</a></code>
</p>
<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+CreateDimReducObject">CreateDimReducObject</a>()</code>,
<code><a href="#topic+DimReduc-class">DimReduc-class</a></code>,
<code><a href="#topic+DimReduc-validity">DimReduc-validity</a></code>,
<code><a href="#topic++5B+5B.DimReduc">[[.DimReduc</a>()</code>,
<code><a href="#topic+dim.DimReduc">dim.DimReduc</a>()</code>,
<code><a href="#topic+merge.DimReduc">merge.DimReduc</a>()</code>,
<code><a href="#topic+print.DimReduc">print.DimReduc</a>()</code>,
<code><a href="#topic+subset.DimReduc">subset.DimReduc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- pbmc_small[["pca"]]
pca[1:10, 1:5]

</code></pre>

<hr>
<h2 id='+5B.SeuratCommand'>Command Log Data Access</h2><span id='topic++5B.SeuratCommand'></span>

<h3>Description</h3>

<p>Access data from a <code>SeuratCommand</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SeuratCommand'
x[i, ...]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.SeuratCommand_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+SeuratCommand">SeuratCommand</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B.SeuratCommand_+3A_i">i</code></td>
<td>
<p>The name of a command log slot</p>
</td></tr>
<tr><td><code id="+2B5B.SeuratCommand_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>[</code>: Slot <code>i</code> from <code>x</code>
</p>


<h3>See Also</h3>

<p>Command log object and interaction methods
<code><a href="#topic++24.SeuratCommand">$.SeuratCommand</a>()</code>,
<code><a href="#topic+.DollarNames.SeuratCommand">.DollarNames.SeuratCommand</a>()</code>,
<code><a href="#topic+LogSeuratCommand">LogSeuratCommand</a>()</code>,
<code><a href="#topic+SeuratCommand-class">SeuratCommand-class</a></code>,
<code><a href="#topic+as.list.SeuratCommand">as.list.SeuratCommand</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cmd &lt;- pbmc_small[["NormalizeData.RNA"]]
cmd["call.string"]

</code></pre>

<hr>
<h2 id='+5B.StdAssay'>Layer Data</h2><span id='topic++5B.StdAssay'></span><span id='topic++5B+3C-+2CStdAssay+2Ccharacter+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>Get and set layer data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
x[i = missing_arg(), j = missing_arg(), ...]

## S4 replacement method for signature 'StdAssay,character,ANY,ANY'
x[i, j, ...] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B.StdAssay_+3A_i">i</code></td>
<td>
<p>Name of layer data to get or set</p>
</td></tr>
<tr><td><code id="+2B5B.StdAssay_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B.StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+LayerData">LayerData</a></code></p>
</td></tr>
<tr><td><code id="+2B5B.StdAssay_+3A_value">value</code></td>
<td>
<p>A matrix-like object to add as a new layer</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>[</code>: The layer data for layer <code>i</code>
</p>
<p><code>[&lt;-</code>: <code>x</code> with layer data <code>value</code> saved as <code>i</code>
</p>


<h3>See Also</h3>

<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+2CLogMap+2Cinteger+2Cmissing-method'><code><a href="#topic+LogMap">LogMap</a></code> Interaction Methods</h2><span id='topic++5B+5B+2CLogMap+2Cinteger+2Cmissing-method'></span><span id='topic++5B+5B+2CLogMap+2Cnumeric+2Cmissing-method'></span>

<h3>Description</h3>

<p>Additional methods for using <code>[[</code> with <code><a href="#topic+LogMap">LogMap</a></code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'LogMap,integer,missing'
x[[i, j, ...]]

## S4 method for signature 'LogMap,numeric,missing'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2CLogMap+2B2Cinteger+2B2Cmissing-method_+3A_x">x</code></td>
<td>
<p>A <code>LogMap</code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2CLogMap+2B2Cinteger+2B2Cmissing-method_+3A_i">i</code></td>
<td>
<p>An integer or numeric vector of length 1</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2CLogMap+2B2Cinteger+2B2Cmissing-method_+3A_j">j</code></td>
<td>
<p>Not used</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2CLogMap+2B2Cinteger+2B2Cmissing-method_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The rownames that are mapped to <code>i</code>
</p>

<hr>
<h2 id='+5B+5B.Assay'>Feature-Level Meta Data</h2><span id='topic++5B+5B.Assay'></span><span id='topic++5B+5B+3C-+2CAssay+2CANY+2CANY+2CANY-method'></span><span id='topic+head.Assay'></span><span id='topic+tail.Assay'></span><span id='topic++5B+5B+3C-+2CAssay+2Cmissing+2Cmissing+2Cdata.frame-method'></span>

<h3>Description</h3>

<p>Get and set feature-level meta data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
x[[i, ..., drop = FALSE]]

## S4 replacement method for signature 'Assay,ANY,ANY,ANY'
x[[i, j, ...]] &lt;- value

## S3 method for class 'Assay'
head(x, n = 10L, ...)

## S3 method for class 'Assay'
tail(x, n = 10L, ...)

## S4 replacement method for signature 'Assay,missing,missing,data.frame'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay_+3A_i">i</code></td>
<td>
<p>Name of feature-level meta data to fetch or add</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay_+3A_drop">drop</code></td>
<td>
<p>See <code><a href="base.html#topic+drop">drop</a></code></p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay_+3A_value">value</code></td>
<td>
<p>Feature-level meta data to add</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay_+3A_n">n</code></td>
<td>
<p>Number of meta data rows to show</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>[[</code>: The feature-level meta data for <code>i</code>
</p>
<p><code>[[&lt;-</code>: <code>x</code> with <code>value</code> added as <code>i</code>
in feature-level meta data
</p>
<p><code>head</code>: The first <code>n</code> rows of feature-level meta data
</p>
<p><code>tail</code>: the last <code>n</code> rows of feature-level meta data
</p>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]

# Pull the entire feature-level meta data data frame
head(rna[[]])

# Pull a specific column of feature-level meta data
head(rna[["vst.mean"]])
head(rna[["vst.mean", drop = TRUE]])

# `head` and `tail` can be used to quickly view feature-level meta data
head(rna)

tail(rna)

</code></pre>

<hr>
<h2 id='+5B+5B.Assay5'>Feature-Level Meta Data</h2><span id='topic++5B+5B.Assay5'></span><span id='topic++5B+5B+3C-+2CAssay5+2CANY+2CANY+2CANY-method'></span><span id='topic+head.Assay5'></span><span id='topic+tail.Assay5'></span>

<h3>Description</h3>

<p>Get and set feature-level meta data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
x[[i, j, ..., drop = FALSE]]

## S4 replacement method for signature 'Assay5,ANY,ANY,ANY'
x[[i, j, ...]] &lt;- value

## S3 method for class 'Assay5'
head(x, n = 10L, ...)

## S3 method for class 'Assay5'
tail(x, n = 10L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay5_+3A_i">i</code></td>
<td>
<p>Name of feature-level meta data to fetch or add</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay5_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay5_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay5_+3A_drop">drop</code></td>
<td>
<p>See <code><a href="base.html#topic+drop">drop</a></code></p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay5_+3A_value">value</code></td>
<td>
<p>Feature-level meta data to add</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Assay5_+3A_n">n</code></td>
<td>
<p>Number of meta data rows to show</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>[[</code>: The feature-level meta data for <code>i</code>
</p>
<p><code>[[&lt;-</code>: <code>x</code> with <code>value</code> added as <code>i</code>
in feature-level meta data
</p>
<p><code>head</code>: The first <code>n</code> rows of feature-level meta data
</p>
<p><code>tail</code>: the last <code>n</code> rows of feature-level meta data
</p>


<h3>See Also</h3>

<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>

<hr>
<h2 id='+5B+5B.DimReduc'>Get Cell Embeddings</h2><span id='topic++5B+5B.DimReduc'></span>

<h3>Description</h3>

<p>Pull cell embeddings from a <a href="#topic+DimReduc">dimensional reduction</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DimReduc'
x[[i, j, drop = FALSE, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B.DimReduc_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+DimReduc">DimReduc</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.DimReduc_+3A_i">i</code></td>
<td>
<p>Cell names or indices</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.DimReduc_+3A_j">j</code></td>
<td>
<p>Dimension identifiers or indices</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.DimReduc_+3A_drop">drop</code></td>
<td>
<p>Coerce the result to the lowest possible dimension; see
<code><a href="base.html#topic+drop">drop</a></code> for further details</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.DimReduc_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cell embeddings for cells <code>i</code> and dimensions <code>j</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Embeddings">Embeddings</a></code>
</p>
<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+CreateDimReducObject">CreateDimReducObject</a>()</code>,
<code><a href="#topic+DimReduc-class">DimReduc-class</a></code>,
<code><a href="#topic+DimReduc-validity">DimReduc-validity</a></code>,
<code><a href="#topic++5B.DimReduc">[.DimReduc</a>()</code>,
<code><a href="#topic+dim.DimReduc">dim.DimReduc</a>()</code>,
<code><a href="#topic+merge.DimReduc">merge.DimReduc</a>()</code>,
<code><a href="#topic+print.DimReduc">print.DimReduc</a>()</code>,
<code><a href="#topic+subset.DimReduc">subset.DimReduc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- pbmc_small[["pca"]]
pca[[1:10, 1:5]]

</code></pre>

<hr>
<h2 id='+5B+5B.Seurat'>Subobjects and Cell-Level Meta Data</h2><span id='topic++5B+5B.Seurat'></span><span id='topic+head.Seurat'></span><span id='topic+tail.Seurat'></span>

<h3>Description</h3>

<p>The <code>[[</code> operator pulls either subobjects
(eg. <a href="#topic+Assay">v3</a> or <a href="#topic+Assay5">v5</a> assays,
<a href="#topic+DimReduc">dimensional reduction</a> information,
or <a href="#topic+Graph">nearest-neighbor graphs</a>) or cell-level
meta data from a <code><a href="#topic+Seurat">Seurat</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
x[[i = missing_arg(), ..., drop = FALSE, na.rm = FALSE]]

## S3 method for class 'Seurat'
head(x, n = 10L, ...)

## S3 method for class 'Seurat'
tail(x, n = 10L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B.Seurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Seurat_+3A_i">i</code></td>
<td>
<p>Name of cell-level meta data</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Seurat_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Seurat_+3A_drop">drop</code></td>
<td>
<p>See <code><a href="base.html#topic+drop">drop</a></code></p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Seurat_+3A_na.rm">na.rm</code></td>
<td>
<p>Remove cells where meta data is all <code>NA</code></p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.Seurat_+3A_n">n</code></td>
<td>
<p>Number of meta data rows to show</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Varies based on the value of <code>i</code>:
</p>

<ul>
<li><p> If <code>i</code> is missing, a data frame with cell-level meta data
</p>
</li>
<li><p> If <code>i</code> is a vector with cell-level meta data names, a data frame
(or vector of <code>drop = TRUE</code>) with cell-level meta data requested
</p>
</li>
<li><p> If <code>i</code> is a one-length character with the
<a href="#topic+names.Seurat">name of a subobject</a>, the
subobject specified by <code>i</code>
</p>
</li></ul>

<p><code>head</code>: The first <code>n</code> rows of cell-level metadata
</p>
<p><code>tail</code>: The last <code>n</code> rows of cell-level metadata
</p>


<h3>See Also</h3>

<p>See <a href="#topic++24.Seurat">here</a> for adding meta data with <code>[[&lt;-</code>,
<a href="#topic++5B+5B+3C-+2CSeurat">here</a> for adding subobjects with <code>[[&lt;-</code>, and
<a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">here</a> for removing subobjects and cell-level meta
data with <code>[[&lt;-</code>
</p>
<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the cell-level metadata data frame
head(pbmc_small[[]])

# Pull specific metadata information
head(pbmc_small[[c("letter.idents", "groups")]])
head(pbmc_small[["groups", drop = TRUE]])

# Get a sub-object (eg. an `Assay` or `DimReduc`)
pbmc_small[["RNA"]]
pbmc_small[["pca"]]

# Get the first 10 rows of cell-level metadata
head(pbmc_small)

# Get the last 10 rows of cell-level metadata
tail(pbmc_small)

</code></pre>

<hr>
<h2 id='+5B+5B.StdAssay'>Feature-Level Meta Data</h2><span id='topic++5B+5B.StdAssay'></span><span id='topic+head.StdAssay'></span><span id='topic+tail.StdAssay'></span><span id='topic++5B+5B+3C-+2CStdAssay+2Ccharacter+2Cmissing+2Cdata.frame-method'></span><span id='topic++5B+5B+3C-+2CStdAssay+2Cmissing+2Cmissing+2Cdata.frame-method'></span><span id='topic++5B+5B+3C-+2CStdAssay+2Ccharacter+2Cmissing+2Cfactor-method'></span><span id='topic++5B+5B+3C-+2CStdAssay+2Ccharacter+2Cmissing+2CNULL-method'></span><span id='topic++5B+5B+3C-+2CStdAssay+2Ccharacter+2Cmissing+2Cvector-method'></span><span id='topic++5B+5B+3C-+2CStdAssay+2Cnumeric+2Cmissing+2CANY-method'></span><span id='topic++5B+5B+3C-+2CStdAssay+2Cmissing+2Cmissing+2CNULL-method'></span>

<h3>Description</h3>

<p>Get and set feature-level meta data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
x[[i, j, ..., drop = FALSE]]

## S3 method for class 'StdAssay'
head(x, n = 10L, ...)

## S3 method for class 'StdAssay'
tail(x, n = 10L, ...)

## S4 replacement method for signature 'StdAssay,character,missing,data.frame'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'StdAssay,missing,missing,data.frame'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'StdAssay,character,missing,factor'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'StdAssay,character,missing,'NULL''
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'StdAssay,character,missing,vector'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'StdAssay,numeric,missing,ANY'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'StdAssay,missing,missing,'NULL''
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.StdAssay_+3A_i">i</code></td>
<td>
<p>Name of feature-level meta data to fetch or add</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.StdAssay_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.StdAssay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.StdAssay_+3A_drop">drop</code></td>
<td>
<p>See <code><a href="base.html#topic+drop">drop</a></code></p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.StdAssay_+3A_n">n</code></td>
<td>
<p>Number of meta data rows to show</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.StdAssay_+3A_value">value</code></td>
<td>
<p>Feature-level meta data to add</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>[[</code>: The feature-level meta data for <code>i</code>
</p>
<p><code>[[&lt;-</code>: <code>x</code> with <code>value</code> added as <code>i</code>
in feature-level meta data
</p>
<p><code>head</code>: The first <code>n</code> rows of feature-level meta data
</p>
<p><code>tail</code>: the last <code>n</code> rows of feature-level meta data
</p>


<h3>See Also</h3>

<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2CSeurat'>Add Subobjects</h2><span id='topic++5B+5B+3C-+2CSeurat'></span><span id='topic++5B+5B+3C-.Seurat'></span><span id='topic++5CS4method+7B+5B+5B+3C-+7D+7BSeurat+2Ccharacter+2Cmissing+2CAssay+7D'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CAssay-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CAssay5-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CDimReduc-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CGraph-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CNeighbor-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CSeuratCommand-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CSpatialImage-method'></span>

<h3>Description</h3>

<p>Add subobjects containing expression, dimensional reduction, or other
containerized data to a <code><a href="#topic+Seurat">Seurat</a></code> object. Subobjects can be
accessed with <code><a href="#topic++5B+5B.Seurat">[[</a></code> and manipulated directly within
the <code>Seurat</code> object or used independently
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'Seurat,character,missing,Assay'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,Assay5'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,DimReduc'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,Graph'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,Neighbor'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,SeuratCommand'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,SpatialImage'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat_+3A_i">i</code></td>
<td>
<p>Name to add subobject as</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat_+3A_value">value</code></td>
<td>
<p>A valid subobject (eg. a <a href="#topic+Assay">v3</a> or <a href="#topic+Assay5">v5</a>
assay, or a <a href="#topic+DimReduc">dimensional reduction</a>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with <code>value</code> added as <code>i</code>
</p>


<h3>See Also</h3>

<p>See <a href="#topic++5B+5B.Seurat">here</a> for pulling subobjects using <code>[[</code>,
<a href="#topic++24.Seurat">here</a> for adding metadata with <code>[[&lt;-</code>, and
<a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">here</a> for removing subobjects and cell-level meta
data with <code>[[&lt;-</code>
</p>
<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2CSeurat+2Ccharacter+2Cmissing+2CStdAssay-method'>Add Subobjects</h2><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CStdAssay-method'></span>

<h3>Description</h3>

<p>Add subobjects containing expression, dimensional reduction, or other
containerized data to a <code><a href="#topic+Seurat">Seurat</a></code> object. Subobjects can be
accessed with <code><a href="#topic++5B+5B.Seurat">[[</a></code> and manipulated directly within
the <code>Seurat</code> object or used independently
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'Seurat,character,missing,StdAssay'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2Ccharacter+2B2Cmissing+2B2CStdAssay-method_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2Ccharacter+2B2Cmissing+2B2CStdAssay-method_+3A_i">i</code></td>
<td>
<p>Name to add subobject as</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2Ccharacter+2B2Cmissing+2B2CStdAssay-method_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2Ccharacter+2B2Cmissing+2B2CStdAssay-method_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2Ccharacter+2B2Cmissing+2B2CStdAssay-method_+3A_value">value</code></td>
<td>
<p>A valid subobject (eg. a <a href="#topic+Assay">v3</a> or <a href="#topic+Assay5">v5</a>
assay, or a <a href="#topic+DimReduc">dimensional reduction</a>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with <code>value</code> added as <code>i</code>
</p>


<h3>See Also</h3>

<p>See <a href="#topic++5B+5B.Seurat">here</a> for pulling subobjects using <code>[[</code>,
<a href="#topic++24.Seurat">here</a> for adding metadata with <code>[[&lt;-</code>, and
<a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">here</a> for removing subobjects and cell-level meta
data with <code>[[&lt;-</code>
</p>
<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2CSeurat+2CNULL'>Remove Subobjects and Cell-Level Meta Data</h2><span id='topic++5B+5B+3C-+2CSeurat+2CNULL'></span><span id='topic+remove-object'></span><span id='topic+remove-objects'></span><span id='topic++5CS4method+7B+5B+5B+3C-+7D+7BSeurat+2Ccharacter+2Cmissing+2CNULL+7D'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2CNULL-method'></span>

<h3>Description</h3>

<p>Remove Subobjects and Cell-Level Meta Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'Seurat,character,missing,'NULL''
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2CNULL_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2CNULL_+3A_i">i</code></td>
<td>
<p>Name(s) of subobject(s) or cell-level meta data to remove</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2CNULL_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2CNULL_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2CSeurat+2B2CNULL_+3A_value">value</code></td>
<td>
<p>NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with <code>i</code> removed from the object
</p>


<h3>See Also</h3>

<p>See <a href="#topic++5B+5B.Seurat">here</a> for pulling subobjects using <code>[[</code>,
<a href="#topic++24.Seurat">here</a> for adding metadata with <code>[[&lt;-</code>, and
<a href="#topic++5B+5B+3C-+2CSeurat">here</a> for adding subobjects with <code>[[&lt;-</code>
</p>
<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>

<hr>
<h2 id='+24.Assay'>Layer Data</h2><span id='topic++24.Assay'></span><span id='topic++24+3C-.Assay'></span>

<h3>Description</h3>

<p>Get and set layer data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
x$i

## S3 replacement method for class 'Assay'
x$i &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id="+2B24.Assay_+3A_i">i</code></td>
<td>
<p>Name of layer data to get or set</p>
</td></tr>
<tr><td><code id="+2B24.Assay_+3A_value">value</code></td>
<td>
<p>A matrix-like object to add as a new layer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>$: Layer data for layer <code>i</code>
</p>
<p><code>$&lt;-</code>: <code>x</code> with layer data <code>value</code> saved as <code>i</code>
</p>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]

# Fetch a layer with `$`
rna$data[1:10, 1:4]

# Add a layer with `$`
rna$data &lt;- rna$counts
rna$data[1:10, 1:4]

</code></pre>

<hr>
<h2 id='+24.Assay5'>Layer Data</h2><span id='topic++24.Assay5'></span><span id='topic++24+3C-.Assay5'></span>

<h3>Description</h3>

<p>Get and set layer data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
x$i

## S3 replacement method for class 'Assay5'
x$i &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="+2B24.Assay5_+3A_i">i</code></td>
<td>
<p>Name of layer data to get or set</p>
</td></tr>
<tr><td><code id="+2B24.Assay5_+3A_value">value</code></td>
<td>
<p>A matrix-like object to add as a new layer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>$: Layer data for layer <code>i</code>
</p>
<p><code>$&lt;-</code>: <code>x</code> with layer data <code>value</code> saved as <code>i</code>
</p>


<h3>See Also</h3>

<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>

<hr>
<h2 id='+24.Seurat'>Cell-Level Meta Data</h2><span id='topic++24.Seurat'></span><span id='topic++24+3C-.Seurat'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2Cdata.frame-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Cmissing+2Cmissing+2Cdata.frame-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2Cfactor-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2Clist-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Cmissing+2Cmissing+2Clist-method'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2Cmissing+2Cvector-method'></span>

<h3>Description</h3>

<p>Get and set cell-level meta data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
x$i

## S3 replacement method for class 'Seurat'
x$i, ... &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,data.frame'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,missing,missing,data.frame'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,factor'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,list'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,missing,missing,list'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'Seurat,character,missing,vector'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24.Seurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="+2B24.Seurat_+3A_i">i</code></td>
<td>
<p>Name of cell-level meta data</p>
</td></tr>
<tr><td><code id="+2B24.Seurat_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="+2B24.Seurat_+3A_value">value</code></td>
<td>
<p>A vector to add as cell-level meta data</p>
</td></tr>
<tr><td><code id="+2B24.Seurat_+3A_j">j</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>$: Metadata column <code>i</code> for object <code>x</code>;
<strong>note</strong>: unlike <code>[[</code>, <code>$</code> drops the shape of the metadata
to return a vector instead of a data frame
</p>
<p><code>$&lt;-</code>: <code>x</code> with metadata <code>value</code> saved as <code>i</code>
</p>


<h3>See Also</h3>

<p>Seurat object, validity, and interaction methods
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get metadata using `$'
head(pbmc_small$groups)

# Add metadata using the `$' operator
set.seed(42)
pbmc_small$value &lt;- sample(1:3, size = ncol(pbmc_small), replace = TRUE)
head(pbmc_small[["value"]])

</code></pre>

<hr>
<h2 id='+24.SeuratCommand'>Command Log Parameter Access</h2><span id='topic++24.SeuratCommand'></span>

<h3>Description</h3>

<p>Pull parameter values from a <code><a href="#topic+SeuratCommand">SeuratCommand</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SeuratCommand'
x$i
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24.SeuratCommand_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+SeuratCommand">SeuratCommand</a></code> object</p>
</td></tr>
<tr><td><code id="+2B24.SeuratCommand_+3A_i">i</code></td>
<td>
<p>A parameter name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The value for parameter <code>i</code>
</p>


<h3>See Also</h3>

<p>Command log object and interaction methods
<code><a href="#topic+.DollarNames.SeuratCommand">.DollarNames.SeuratCommand</a>()</code>,
<code><a href="#topic+LogSeuratCommand">LogSeuratCommand</a>()</code>,
<code><a href="#topic+SeuratCommand-class">SeuratCommand-class</a></code>,
<code><a href="#topic++5B.SeuratCommand">[.SeuratCommand</a>()</code>,
<code><a href="#topic+as.list.SeuratCommand">as.list.SeuratCommand</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cmd &lt;- pbmc_small[["NormalizeData.RNA"]]
cmd$normalization.method

</code></pre>

<hr>
<h2 id='+24.StdAssay'>Layer Data</h2><span id='topic++24.StdAssay'></span><span id='topic++24+3C-.StdAssay'></span>

<h3>Description</h3>

<p>Get and set layer data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
x$i

## S3 replacement method for class 'StdAssay'
x$i &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="+2B24.StdAssay_+3A_i">i</code></td>
<td>
<p>Name of layer data to get or set</p>
</td></tr>
<tr><td><code id="+2B24.StdAssay_+3A_value">value</code></td>
<td>
<p>A matrix-like object to add as a new layer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>$: Layer data for layer <code>i</code>
</p>
<p><code>$&lt;-</code>: <code>x</code> with layer data <code>value</code> saved as <code>i</code>
</p>


<h3>See Also</h3>

<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='AddMetaData'>Add in metadata associated with either cells or features.</h2><span id='topic+AddMetaData'></span><span id='topic+SeuratAccess'></span><span id='topic+AddMetaData.Assay'></span><span id='topic+AddMetaData.Assay5'></span><span id='topic+AddMetaData.Seurat'></span>

<h3>Description</h3>

<p>Adds additional data to the object. Can be any piece of information
associated with a cell (examples include read depth, alignment rate,
experimental batch, or subpopulation identity) or feature (ENSG name,
variance). To add cell level information, add to the Seurat object. If adding
feature-level metadata, add to the Assay object (e.g. <code>object[["RNA"]]</code>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddMetaData(object, metadata, col.name = NULL)

## S3 method for class 'Assay'
AddMetaData(object, metadata, col.name = NULL)

## S3 method for class 'Assay5'
AddMetaData(object, metadata, col.name = NULL)

## S3 method for class 'Seurat'
AddMetaData(object, metadata, col.name = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddMetaData_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="AddMetaData_+3A_metadata">metadata</code></td>
<td>
<p>A vector, list, or data.frame with metadata to add</p>
</td></tr>
<tr><td><code id="AddMetaData_+3A_col.name">col.name</code></td>
<td>
<p>A name for meta data if not a named list or data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>object</code> with metadata added
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cluster_letters &lt;- LETTERS[Idents(object = pbmc_small)]
names(cluster_letters) &lt;- colnames(x = pbmc_small)
pbmc_small &lt;- AddMetaData(
  object = pbmc_small,
  metadata = cluster_letters,
  col.name = 'letter.idents'
)
head(x = pbmc_small[[]])

</code></pre>

<hr>
<h2 id='AddMetaData-StdAssay'>Add in metadata associated with either cells or features.</h2><span id='topic+AddMetaData-StdAssay'></span><span id='topic+AddMetaData.StdAssay'></span>

<h3>Description</h3>

<p>Adds additional data to the object. Can be any piece of information
associated with a cell (examples include read depth, alignment rate,
experimental batch, or subpopulation identity) or feature (ENSG name,
variance). To add cell level information, add to the Seurat object. If adding
feature-level metadata, add to the Assay object (e.g. <code>object[["RNA"]]</code>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
AddMetaData(object, metadata, col.name = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddMetaData-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="AddMetaData-StdAssay_+3A_metadata">metadata</code></td>
<td>
<p>A vector, list, or data.frame with metadata to add</p>
</td></tr>
<tr><td><code id="AddMetaData-StdAssay_+3A_col.name">col.name</code></td>
<td>
<p>A name for meta data if not a named list or data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>object</code> with metadata added
</p>

<hr>
<h2 id='aggregate'>Aggregate Molecules into an Expression Matrix</h2><span id='topic+aggregate'></span><span id='topic+aggregate.FOV'></span><span id='topic+aggregate.Molecules'></span>

<h3>Description</h3>

<p>Aggregate Molecules into an Expression Matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FOV'
aggregate(x, by = NULL, set = NULL, drop = TRUE, ...)

## S3 method for class 'Molecules'
aggregate(x, by, drop = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aggregate_+3A_x">x</code></td>
<td>
<p>An object with spatially-resolved molecule information</p>
</td></tr>
<tr><td><code id="aggregate_+3A_by">by</code></td>
<td>
<p>Name of a
<code><a href="#topic+Segmentation-class">Segmentation</a></code> within
<code>object</code> or a
<code><a href="#topic+Segmentation-class">Segmentation</a></code> object</p>
</td></tr>
<tr><td><code id="aggregate_+3A_set">set</code></td>
<td>
<p>Name of molecule set to aggregate</p>
</td></tr>
<tr><td><code id="aggregate_+3A_drop">drop</code></td>
<td>
<p>Drop molecules not present in a segmentation; if <code>FALSE</code>,
adds a column called &ldquo;<code>boundless</code>&rdquo; consisting of molecule counts
not in a segmentation</p>
</td></tr>
<tr><td><code id="aggregate_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An expression matrix
</p>


<h3>Progress Updates with <span class="pkg">progressr</span></h3>

<p>This function uses
<a href="https://cran.r-project.org/package=progressr"><span class="pkg">progressr</span></a> to
render status updates and progress bars. To enable progress updates, wrap
the function call in <code><a href="progressr.html#topic+with_progress">with_progress</a></code> or run
<code><a href="progressr.html#topic+handlers">handlers(global = TRUE)</a></code> before running
this function. For more details about <span class="pkg">progressr</span>, please read
<a href="https://progressr.futureverse.org/articles/progressr-intro.html"><code>vignette("progressr-intro")</code></a>
</p>


<h3>Parallelization with <span class="pkg">future</span></h3>

<p>This function uses
<a href="https://cran.r-project.org/package=future"><span class="pkg">future</span></a> to enable
parallelization. Parallelization strategies can be set using
<code><a href="future.html#topic+plan">plan</a></code>. Common plans include &ldquo;<code>sequential</code>&rdquo;
for non-parallelized processing or &ldquo;<code>multisession</code>&rdquo; for parallel
evaluation using multiple <span class="rlang"><b>R</b></span> sessions; for other plans, see the
&ldquo;Implemented evaluation strategies&rdquo; section of
<code><a href="future.html#topic+plan">?future::plan</a></code>. For a more thorough introduction
to <span class="pkg">future</span>, see
<a href="https://future.futureverse.org/articles/future-1-overview.html"><code>vignette("future-1-overview")</code></a>
</p>

<hr>
<h2 id='Angles'>Radian/Degree Conversions</h2><span id='topic+Angles'></span><span id='topic+Degrees'></span><span id='topic+Radians'></span>

<h3>Description</h3>

<p>Convert degrees to radians and vice versa
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Degrees(rad)

Radians(deg)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Angles_+3A_rad">rad</code></td>
<td>
<p>Angle in radians</p>
</td></tr>
<tr><td><code id="Angles_+3A_deg">deg</code></td>
<td>
<p>Angle in degrees</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>Degrees</code>: <code>rad</code> in degrees
</p>
<p><code>Radians</code>: <code>deg</code> in radians
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PolyVtx">PolyVtx</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Degrees(pi)

Radians(180)

</code></pre>

<hr>
<h2 id='as.Centroids'>Convert Segmentation Layers</h2><span id='topic+as.Centroids'></span><span id='topic+as.Segmentation'></span><span id='topic+as.Centroids.Segmentation'></span><span id='topic+as.Segmentation.Centroids'></span>

<h3>Description</h3>

<p>Convert Segmentation Layers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.Centroids(x, nsides = NULL, radius = NULL, theta = NULL, ...)

as.Segmentation(x, ...)

## S3 method for class 'Segmentation'
as.Centroids(x, nsides = NULL, radius = NULL, theta = NULL, ...)

## S3 method for class 'Centroids'
as.Segmentation(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.Centroids_+3A_x">x</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="as.Centroids_+3A_nsides">nsides</code></td>
<td>
<p>The number of sides to represent cells/spots; pass
<code><a href="base.html#topic+Inf">Inf</a></code> to plot as circles</p>
</td></tr>
<tr><td><code id="as.Centroids_+3A_radius">radius</code></td>
<td>
<p>Radius of shapes when plotting</p>
</td></tr>
<tr><td><code id="as.Centroids_+3A_theta">theta</code></td>
<td>
<p>Angle to adjust shapes when plotting</p>
</td></tr>
<tr><td><code id="as.Centroids_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>as.Centroids</code>: A
<code><a href="#topic+Centroids-class">Centroids</a></code> object
</p>
<p><code>as.Segmentation</code>: A
<code><a href="#topic+Segmentation-class">Segmentation</a></code> object
</p>

<hr>
<h2 id='as.Graph'>Coerce to a <code>Graph</code> Object</h2><span id='topic+as.Graph'></span><span id='topic+as.Graph.Matrix'></span><span id='topic+as.Graph.matrix'></span><span id='topic+as.Graph.Neighbor'></span>

<h3>Description</h3>

<p>Convert a <code><a href="base.html#topic+matrix">matrix</a></code> (or <code><a href="Matrix.html#topic+Matrix">Matrix</a></code>) to
a <code><a href="#topic+Graph">Graph</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.Graph(x, ...)

## S3 method for class 'Matrix'
as.Graph(x, ...)

## S3 method for class 'matrix'
as.Graph(x, ...)

## S3 method for class 'Neighbor'
as.Graph(x, weighted = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.Graph_+3A_x">x</code></td>
<td>
<p>The matrix to convert</p>
</td></tr>
<tr><td><code id="as.Graph_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="as.Graph_+3A_weighted">weighted</code></td>
<td>
<p>If TRUE, fill entries in Graph matrix with value from the
nn.dist slot of the Neighbor object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+Graph">Graph</a></code> object
</p>


<h3>See Also</h3>

<p>Other graph: 
<code><a href="#topic+Graph-class">Graph-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># converting sparse matrix
mat &lt;- Matrix::rsparsematrix(nrow = 10, ncol = 10, density = 0.1)
rownames(x = mat) &lt;- paste0("feature_", 1:10)
colnames(x = mat) &lt;- paste0("cell_", 1:10)
g &lt;- as.Graph(x = mat)

# converting dense matrix
mat &lt;- matrix(data = 1:16, nrow = 4)
rownames(x = mat) &lt;- paste0("feature_", 1:4)
colnames(x = mat) &lt;- paste0("cell_", 1:4)
g &lt;- as.Graph(x = mat)

</code></pre>

<hr>
<h2 id='as.list.SeuratCommand'>Coerce a <code>SeuratCommand</code> to a list</h2><span id='topic+as.list.SeuratCommand'></span>

<h3>Description</h3>

<p>Coerce a <code>SeuratCommand</code> to a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SeuratCommand'
as.list(x, complete = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.list.SeuratCommand_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+SeuratCommand">SeuratCommand</a></code> object</p>
</td></tr>
<tr><td><code id="as.list.SeuratCommand_+3A_complete">complete</code></td>
<td>
<p>Include slots besides just parameters
(eg. call string, name, timestamp)</p>
</td></tr>
<tr><td><code id="as.list.SeuratCommand_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the parameters and, if <code>complete = TRUE</code>,
the call string, name, and timestamp
</p>


<h3>See Also</h3>

<p>Command log object and interaction methods
<code><a href="#topic++24.SeuratCommand">$.SeuratCommand</a>()</code>,
<code><a href="#topic+.DollarNames.SeuratCommand">.DollarNames.SeuratCommand</a>()</code>,
<code><a href="#topic+LogSeuratCommand">LogSeuratCommand</a>()</code>,
<code><a href="#topic+SeuratCommand-class">SeuratCommand-class</a></code>,
<code><a href="#topic++5B.SeuratCommand">[.SeuratCommand</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cmd &lt;- pbmc_small[["NormalizeData.RNA"]]
as.list(cmd)
as.list(cmd, complete = TRUE)

</code></pre>

<hr>
<h2 id='as.matrix.LogMap'>Coerce Logical Maps to Matrices</h2><span id='topic+as.matrix.LogMap'></span>

<h3>Description</h3>

<p>Coerce a logical map to a matrix; this removes all
<a href="#topic+LogMap">logical map</a> class capabilities from
the object and returns a base-R matrix object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'LogMap'
as.matrix(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.matrix.LogMap_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+LogMap">LogMap</a></code> object</p>
</td></tr>
<tr><td><code id="as.matrix.LogMap_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A base-R matrix created from <code>x</code>
</p>


<h3>See Also</h3>

<p>Logical map objects, validity, and interaction methods:
<code><a href="#topic+LogMap-validity">LogMap-validity</a></code>,
<code><a href="#topic+LogMap">LogMap</a></code>,
<code><a href="#topic+droplevels.LogMap">droplevels.LogMap</a>()</code>,
<code><a href="#topic+intersect.LogMap">intersect.LogMap</a>()</code>,
<code><a href="#topic+labels.LogMap">labels.LogMap</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map &lt;- LogMap(letters[1:10])
map[['obs']] &lt;- c(1, 3, 7)
mat &lt;- as.matrix(map)
mat
class(mat)

</code></pre>

<hr>
<h2 id='as.Neighbor'>Coerce to a <code>Neighbor</code> Object</h2><span id='topic+as.Neighbor'></span><span id='topic+as.Neighbor.Graph'></span>

<h3>Description</h3>

<p>Convert objects to <code><a href="#topic+Neighbor">Neighbor</a></code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.Neighbor(x, ...)

## S3 method for class 'Graph'
as.Neighbor(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.Neighbor_+3A_x">x</code></td>
<td>
<p>An object to convert to <code><a href="#topic+Neighbor">Neighbor</a></code></p>
</td></tr>
<tr><td><code id="as.Neighbor_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+Neighbor">Neighbor</a></code> object
</p>

<hr>
<h2 id='as.Seurat'>Coerce to a <code>Seurat</code> Object</h2><span id='topic+as.Seurat'></span>

<h3>Description</h3>

<p>Convert objects to Seurat objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.Seurat(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.Seurat_+3A_x">x</code></td>
<td>
<p>An object to convert to class <code>Seurat</code></p>
</td></tr>
<tr><td><code id="as.Seurat_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+Seurat">Seurat</a></code> object generated from <code>x</code>
</p>

<hr>
<h2 id='as.sparse'>Cast to Sparse</h2><span id='topic+as.sparse'></span><span id='topic+as.sparse.data.frame'></span><span id='topic+as.sparse.Matrix'></span><span id='topic+as.sparse.matrix'></span><span id='topic+as.sparse.ngCMatrix'></span>

<h3>Description</h3>

<p>Convert dense objects to sparse representations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.sparse(x, ...)

## S3 method for class 'data.frame'
as.sparse(x, row.names = NULL, ...)

## S3 method for class 'Matrix'
as.sparse(x, ...)

## S3 method for class 'matrix'
as.sparse(x, ...)

## S3 method for class 'ngCMatrix'
as.sparse(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.sparse_+3A_x">x</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="as.sparse_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="as.sparse_+3A_row.names">row.names</code></td>
<td>
<p><code>NULL</code> or a character vector giving the row names for
the data; missing values are not allowed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sparse representation of the input data
</p>

<hr>
<h2 id='Assay-class'>The Assay Class</h2><span id='topic+Assay-class'></span><span id='topic+Assay'></span>

<h3>Description</h3>

<p>The Assay object is the basic unit of Seurat; each Assay stores raw,
normalized, and scaled data as well as cluster information, variable
features, and any other assay-specific metadata. Assays should contain single
cell expression data such as RNA-seq, protein, or imputed expression data.
</p>


<h3>Slots</h3>


<dl>
<dt><code>counts</code></dt><dd><p>Unnormalized data such as raw counts or TPMs</p>
</dd>
<dt><code>data</code></dt><dd><p>Normalized expression data</p>
</dd>
<dt><code>scale.data</code></dt><dd><p>Scaled expression data</p>
</dd>
<dt><code>assay.orig</code></dt><dd><p>Original assay that this assay is based off of. Used to
track assay provenance</p>
</dd>
<dt><code>var.features</code></dt><dd><p>Vector of features exhibiting high variance across
single cells</p>
</dd>
<dt><code>meta.features</code></dt><dd><p>Feature-level metadata</p>
</dd>
<dt><code>misc</code></dt><dd><p>A named list of unstructured miscellaneous data</p>
</dd>
<dt><code>key</code></dt><dd><p>A one-length character vector with the object's key; keys must
be one or more alphanumeric characters followed by an underscore
&ldquo;<code>_</code>&rdquo; (regex pattern
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;)</p>
</dd>
</dl>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>

<hr>
<h2 id='Assay-validity'>V3 Assay Validity</h2><span id='topic+Assay-validity'></span>

<h3>Description</h3>

<p>Validation of <code>Assay</code> objects is handled by
<code><a href="methods.html#topic+validObject">validObject</a></code>
</p>


<h3><code>data</code> Validation</h3>

<p>blah
</p>


<h3><code>counts</code> Validation</h3>

<p>blah
</p>


<h3><code>scale.data</code> Validation</h3>

<p>blah
</p>


<h3>Feature-Level Meta Data Validation</h3>

<p>blah
</p>


<h3>Variable Feature Validation</h3>

<p>blah
</p>


<h3>Key Validation</h3>

<p>Keys must be a one-length character vector; a key must be composed of one
of the following:
</p>

<ul>
<li><p> An empty string (eg. &ldquo;<code>''</code>&rdquo;) where <code>nzchar() == 0</code>
</p>
</li>
<li><p> An string composed of one or more alphanumeric values
(both lower- and upper-case) that ends with an underscore
(&ldquo;<code>_</code>&rdquo;); the first character must be a letter
</p>
</li></ul>

<p>Keys that are not empty strings are validated with the regex
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;
</p>


<h3>See Also</h3>

<p><code><a href="methods.html#topic+validObject">validObject</a></code>
</p>
<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]
validObject(rna)

</code></pre>

<hr>
<h2 id='Assay5-class'>The v5 <code>Assay</code> Object</h2><span id='topic+Assay5-class'></span><span id='topic+Assay5'></span>

<h3>Description</h3>

<p>The v5 <code>Assay</code> is the typical <code>Assay</code> class used in <span class="pkg">Seurat</span>
v5; ...
</p>


<h3>Slots</h3>


<dl>
<dt><code>layers</code></dt><dd><p>A named list containing expression matrices; each matrix should
be a two-dimensional object containing some subset of cells and features
defined in the <code>cells</code> and <code>features</code> slots. Cell and feature
membership is recorded in the <code>cells</code> and <code>features</code> slots,
respectively</p>
</dd>
<dt><code>cells</code></dt><dd><p>A <a href="#topic+LogMap">logical mapping</a> of cell names
and layer membership; this map contains all the possible cells that this
assay can contain. New layers must have some subset of cells present
in this map</p>
</dd>
<dt><code>features</code></dt><dd><p>A <a href="#topic+LogMap">logical mapping</a> of feature
names and layer membership; this map contains all the possible features that
this assay can contain. New layers must have some subset of features
present in this map</p>
</dd>
<dt><code>default</code></dt><dd><p>A one-length integer with the end index of the
<a href="#topic+DefaultLayer">default layer</a>; the default layer be
all layers up to and including the layer at index <code>default</code></p>
</dd>
<dt><code>assay.orig</code></dt><dd><p>Original assay that this assay is based off of;
used to track assay provenance</p>
</dd>
<dt><code>meta.data</code></dt><dd><p>A <a href="base.html#topic+data.frame">data frame</a> with feature-level
meta data; should have the same number of rows as <code>features</code></p>
</dd>
<dt><code>misc</code></dt><dd><p>A named list of unstructured miscellaneous data</p>
</dd>
<dt><code>key</code></dt><dd><p>A one-length character vector with the object's key; keys must
be one or more alphanumeric characters followed by an underscore
&ldquo;<code>_</code>&rdquo; (regex pattern
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;)</p>
</dd>
</dl>


<h3>See Also</h3>

<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>

<hr>
<h2 id='Assay5-validity'>V5 Assay Validity</h2><span id='topic+Assay5-validity'></span>

<h3>Description</h3>

<p>Validation of <code>Assay5</code> objects is handled by
<code><a href="methods.html#topic+validObject">validObject</a></code>
</p>


<h3>Layer Validation</h3>

<p>blah
</p>


<h3>Key Validation</h3>

<p>Keys must be a one-length character vector; a key must be composed of one
of the following:
</p>

<ul>
<li><p> An empty string (eg. &ldquo;<code>''</code>&rdquo;) where <code>nzchar() == 0</code>
</p>
</li>
<li><p> An string composed of one or more alphanumeric values
(both lower- and upper-case) that ends with an underscore
(&ldquo;<code>_</code>&rdquo;); the first character must be a letter
</p>
</li></ul>

<p>Keys that are not empty strings are validated with the regex
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;
</p>


<h3>See Also</h3>

<p><code><a href="methods.html#topic+validObject">validObject</a></code>
</p>
<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>

<hr>
<h2 id='Assay5T-class'>The Transposed v5 <code>Assay</code> Object</h2><span id='topic+Assay5T-class'></span><span id='topic+Assay5T'></span>

<h3>Description</h3>

<p>The Transposed v5 <code>Assay</code> Object
</p>


<h3>Slots</h3>


<dl>
<dt><code>layers</code></dt><dd><p>A named list containing expression matrices; each matrix should
be a two-dimensional object containing some subset of cells and features
defined in the <code>cells</code> and <code>features</code> slots. Cell and feature
membership is recorded in the <code>cells</code> and <code>features</code> slots,
respectively</p>
</dd>
<dt><code>cells</code></dt><dd><p>A <a href="#topic+LogMap">logical mapping</a> of cell names
and layer membership; this map contains all the possible cells that this
assay can contain. New layers must have some subset of cells present
in this map</p>
</dd>
<dt><code>features</code></dt><dd><p>A <a href="#topic+LogMap">logical mapping</a> of feature
names and layer membership; this map contains all the possible features that
this assay can contain. New layers must have some subset of features
present in this map</p>
</dd>
<dt><code>default</code></dt><dd><p>A one-length integer with the end index of the
<a href="#topic+DefaultLayer">default layer</a>; the default layer be
all layers up to and including the layer at index <code>default</code></p>
</dd>
<dt><code>assay.orig</code></dt><dd><p>Original assay that this assay is based off of;
used to track assay provenance</p>
</dd>
<dt><code>meta.data</code></dt><dd><p>A <a href="base.html#topic+data.frame">data frame</a> with feature-level
meta data; should have the same number of rows as <code>features</code></p>
</dd>
<dt><code>misc</code></dt><dd><p>A named list of unstructured miscellaneous data</p>
</dd>
<dt><code>key</code></dt><dd><p>A one-length character vector with the object's key; keys must
be one or more alphanumeric characters followed by an underscore
&ldquo;<code>_</code>&rdquo; (regex pattern
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;)</p>
</dd>
</dl>


<h3>Lifecycle</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p><strong>Warning</strong>: functionality described here is experimental and prone to
change without notice
</p>

<hr>
<h2 id='AssayData'>Get and Set Assay Data</h2><span id='topic+AssayData'></span><span id='topic+GetAssayData'></span><span id='topic+SetAssayData'></span><span id='topic+GetAssayData.Seurat'></span><span id='topic+SetAssayData.Seurat'></span><span id='topic+GetAssayData.Assay'></span><span id='topic+SetAssayData.Assay'></span>

<h3>Description</h3>

<p>General accessor and setter functions for <code><a href="#topic+Assay">Assay</a></code> objects.
<code>GetAssayData</code> can be used to pull information from any of the
expression matrices (eg. &ldquo;counts&rdquo;, &ldquo;data&rdquo;, or
&ldquo;scale.data&rdquo;). <code>SetAssayData</code> can be used to replace one of these
expression matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetAssayData(object, ...)

SetAssayData(object, layer, new.data, slot = deprecated(), ...)

## S3 method for class 'Seurat'
GetAssayData(object, assay = NULL, layer = NULL, slot = deprecated(), ...)

## S3 method for class 'Seurat'
SetAssayData(
  object,
  layer = "data",
  new.data,
  slot = deprecated(),
  assay = NULL,
  ...
)

## S3 method for class 'Assay'
GetAssayData(
  object,
  layer = c("data", "scale.data", "counts"),
  slot = deprecated(),
  ...
)

## S3 method for class 'Assay'
SetAssayData(
  object,
  layer = c("data", "scale.data", "counts"),
  new.data,
  slot = deprecated(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AssayData_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="AssayData_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="AssayData_+3A_layer">layer</code></td>
<td>
<p>Name of layer to get or set</p>
</td></tr>
<tr><td><code id="AssayData_+3A_new.data">new.data</code></td>
<td>
<p>New assay data to add</p>
</td></tr>
<tr><td><code id="AssayData_+3A_slot">slot</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> Specific assay data to get or set</p>
</td></tr>
<tr><td><code id="AssayData_+3A_assay">assay</code></td>
<td>
<p>Specific assay to get data from or set data for;
defaults to the <a href="#topic+DefaultAssay">default assay</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>GetAssayData</code>: returns the specified assay data
</p>
<p><code>SetAssayData</code>: <code>object</code> with the assay data set
</p>


<h3>Lifecycle</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#superseded"><img src="../help/figures/lifecycle-superseded.svg" alt='[Superseded]' /></a>
</p>
<p><code>GetAssayData</code> and <code>SetAssayData</code> have been superseded. To fetch
expression matrices, use <code><a href="#topic+LayerData">LayerData</a></code>; to set expression data,
use <code><a href="#topic+LayerData+3C-">LayerData&lt;-</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get assay data from the default assay in a Seurat object
GetAssayData(object = pbmc_small, layer = "data")[1:5,1:5]

# Set an Assay layer through the Seurat object
count.data &lt;- GetAssayData(object = pbmc_small[["RNA"]], layer = "counts")
count.data &lt;- as.matrix(x = count.data + 1)
new.seurat.object &lt;- SetAssayData(
    object = pbmc_small,
    layer = "counts",
    new.data = count.data,
    assay = "RNA"
)

# Get the data directly from an Assay object
GetAssayData(pbmc_small[["RNA"]], layer = "data")[1:5,1:5]

# Set an Assay layer directly
count.data &lt;- GetAssayData(pbmc_small[["RNA"]], layer = "counts")
count.data &lt;- as.matrix(x = count.data + 1)
new.assay &lt;- SetAssayData(pbmc_small[["RNA"]], layer = "counts", new.data = count.data)

</code></pre>

<hr>
<h2 id='AssayData-StdAssay'>Get and Set Assay Data</h2><span id='topic+AssayData-StdAssay'></span><span id='topic+GetAssayData.StdAssay'></span><span id='topic+SetAssayData.StdAssay'></span>

<h3>Description</h3>

<p>General accessor and setter functions for <code><a href="#topic+Assay">Assay</a></code> objects.
<code>GetAssayData</code> can be used to pull information from any of the
expression matrices (eg. &ldquo;counts&rdquo;, &ldquo;data&rdquo;, or
&ldquo;scale.data&rdquo;). <code>SetAssayData</code> can be used to replace one of these
expression matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
GetAssayData(object, layer = NULL, slot = deprecated(), ...)

## S3 method for class 'StdAssay'
SetAssayData(object, layer, new.data, slot = deprecated(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AssayData-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="AssayData-StdAssay_+3A_layer">layer</code></td>
<td>
<p>Name of layer to get or set</p>
</td></tr>
<tr><td><code id="AssayData-StdAssay_+3A_slot">slot</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> Specific assay data to get or set</p>
</td></tr>
<tr><td><code id="AssayData-StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="AssayData-StdAssay_+3A_new.data">new.data</code></td>
<td>
<p>New assay data to add</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>GetAssayData</code>: returns the specified assay data
</p>
<p><code>SetAssayData</code>: <code>object</code> with the assay data set
</p>


<h3>Lifecycle</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#superseded"><img src="../help/figures/lifecycle-superseded.svg" alt='[Superseded]' /></a>
</p>
<p><code>GetAssayData</code> and <code>SetAssayData</code> have been superseded. To fetch
expression matrices, use <code><a href="#topic+LayerData">LayerData</a></code>; to set expression data,
use <code><a href="#topic+LayerData+3C-">LayerData&lt;-</a></code>
</p>

<hr>
<h2 id='Assays'>Query Specific Object Types</h2><span id='topic+Assays'></span><span id='topic+Graphs'></span><span id='topic+Neighbors'></span><span id='topic+Reductions'></span><span id='topic+Assays.Seurat'></span>

<h3>Description</h3>

<p>List the names of <code><a href="#topic+Assay">Assay</a></code>, <code><a href="#topic+DimReduc">DimReduc</a></code>,
<code><a href="#topic+Graph">Graph</a></code>, <code><a href="#topic+Neighbor">Neighbor</a></code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Assays(object, ...)

Graphs(object, slot = NULL)

Neighbors(object, slot = NULL)

Reductions(object, slot = NULL)

## S3 method for class 'Seurat'
Assays(object, slot = deprecated(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Assays_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="Assays_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="Assays_+3A_slot">slot</code></td>
<td>
<p>Name of component object to return</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>slot</code> is <code>NULL</code>, the names of all component objects
in this <code>Seurat</code> object. Otherwise, the specific object specified
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Assays(pbmc_small)

Graphs(pbmc_small)

Reductions(object = pbmc_small)

</code></pre>

<hr>
<h2 id='AttachDeps'>Attach Required Packages</h2><span id='topic+AttachDeps'></span>

<h3>Description</h3>

<p>Helper function to attach required packages. Detects if a package is already
attached and if so, skips it. Should be called in <code><a href="base.html#topic+.onAttach">.onAttach</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AttachDeps(deps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AttachDeps_+3A_deps">deps</code></td>
<td>
<p>A character vector of packages to attach</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>NULL</code>
</p>


<h3>Lifecycle</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#superseded"><img src="../help/figures/lifecycle-superseded.svg" alt='[Superseded]' /></a>
</p>
<p><code>AttachDeps</code> has been superseded as of <span class="pkg">SeuratObject</span> v5.0.0;
as an alternative, list dependencies in the <code>Depends</code> section of
<code>DESCRIPTION</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Use in your .onAttach hook
if (FALSE) {
  .onAttach &lt;- function(libname, pkgname) {
    AttachDeps(c("SeuratObject", "rlang"))
  }
}

</code></pre>

<hr>
<h2 id='Boundaries'>Get, Set, and Query Segmentation Boundaries</h2><span id='topic+Boundaries'></span><span id='topic+DefaultBoundary'></span><span id='topic+DefaultBoundary+3C-'></span><span id='topic+Molecules'></span><span id='topic+Boundaries.FOV'></span><span id='topic+DefaultBoundary.FOV'></span><span id='topic+DefaultBoundary+3C-.FOV'></span><span id='topic+Molecules.FOV'></span>

<h3>Description</h3>

<p>Get, Set, and Query Segmentation Boundaries
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Boundaries(object, ...)

DefaultBoundary(object)

DefaultBoundary(object, ...) &lt;- value

Molecules(object, ...)

## S3 method for class 'FOV'
Boundaries(object, ...)

## S3 method for class 'FOV'
DefaultBoundary(object)

## S3 replacement method for class 'FOV'
DefaultBoundary(object, ...) &lt;- value

## S3 method for class 'FOV'
Molecules(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Boundaries_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Boundaries_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Boundaries_+3A_value">value</code></td>
<td>
<p>The name of a segmentation boundary to set as default</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>Boundaries</code>: The names of all segmentation boundaries present
within <code>object</code>
</p>
<p><code>DefaultBoundary</code>: The name of the default
segmentation boundary
</p>
<p><code>DefaultBoundary&lt;-</code>: <code>object</code> with the default
segmentation boundary set to <code>value</code>
</p>
<p><code>Molecules</code>: The names of all molecule sets present within
<code>object</code>
</p>

<hr>
<h2 id='CastAssay'>Cast Assay Layers</h2><span id='topic+CastAssay'></span><span id='topic+CastAssay.Assay5'></span>

<h3>Description</h3>

<p>Cast layers in v5 assays to other classes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CastAssay(object, to, ...)

## S3 method for class 'Assay5'
CastAssay(object, to, layers = NA, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CastAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="CastAssay_+3A_to">to</code></td>
<td>
<p>Either a class name or a function that takes a layer and returns
the same layer as a new class</p>
</td></tr>
<tr><td><code id="CastAssay_+3A_...">...</code></td>
<td>
<p>If <code>to</code> is a function, arguments passed to <code>to</code></p>
</td></tr>
<tr><td><code id="CastAssay_+3A_layers">layers</code></td>
<td>
<p>A vector of layers to cast; defaults to all layers</p>
</td></tr>
<tr><td><code id="CastAssay_+3A_verbose">verbose</code></td>
<td>
<p>Show progress updates</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>object</code> with the layers cast to class specified by <code>to</code>
</p>

<hr>
<h2 id='CastAssay-StdAssay'>Cast Assay Layers</h2><span id='topic+CastAssay-StdAssay'></span><span id='topic+CastAssay.StdAssay'></span>

<h3>Description</h3>

<p>Cast layers in v5 assays to other classes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
CastAssay(object, to, layers = NA, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CastAssay-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="CastAssay-StdAssay_+3A_to">to</code></td>
<td>
<p>Either a class name or a function that takes a layer and returns
the same layer as a new class</p>
</td></tr>
<tr><td><code id="CastAssay-StdAssay_+3A_layers">layers</code></td>
<td>
<p>A vector of layers to cast; defaults to all layers</p>
</td></tr>
<tr><td><code id="CastAssay-StdAssay_+3A_verbose">verbose</code></td>
<td>
<p>Show progress updates</p>
</td></tr>
<tr><td><code id="CastAssay-StdAssay_+3A_...">...</code></td>
<td>
<p>If <code>to</code> is a function, arguments passed to <code>to</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>object</code> with the layers cast to class specified by <code>to</code>
</p>

<hr>
<h2 id='Cells'>Cell and Feature Names</h2><span id='topic+Cells'></span><span id='topic+Features'></span><span id='topic+Cells.default'></span><span id='topic+Cells.Assay5'></span><span id='topic+Features.Assay5'></span><span id='topic+Cells.DimReduc'></span><span id='topic+Cells.Neighbor'></span>

<h3>Description</h3>

<p>Get the cell and feature names of an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cells(x, ...)

Features(x, ...)

## Default S3 method:
Cells(x, ...)

## S3 method for class 'Assay5'
Cells(x, layer = NULL, simplify = TRUE, ...)

## S3 method for class 'Assay5'
Features(x, layer = NULL, simplify = TRUE, ...)

## S3 method for class 'DimReduc'
Cells(x, ...)

## S3 method for class 'Neighbor'
Cells(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Cells_+3A_x">x</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Cells_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Cells_+3A_layer">layer</code></td>
<td>
<p>Layer to pull cells/features for; defaults to default layer;
if <code>NA</code>, returns all cells for the assay</p>
</td></tr>
<tr><td><code id="Cells_+3A_simplify">simplify</code></td>
<td>
<p>Simplify the cell/feature names into a single vector; if
<code>FALSE</code>, separates each cell/feature names by layer</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>Cell</code>: A vector of cell names
</p>
<p><code>Features</code>: A vector of feature names
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Cells(x = pbmc_small)

</code></pre>

<hr>
<h2 id='Cells-StdAssay'>Cell and Feature Names</h2><span id='topic+Cells-StdAssay'></span><span id='topic+Cells.StdAssay'></span><span id='topic+Features.StdAssay'></span>

<h3>Description</h3>

<p>Get the cell and feature names of an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
Cells(x, layer = NULL, simplify = TRUE, ...)

## S3 method for class 'StdAssay'
Features(x, layer = NULL, simplify = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Cells-StdAssay_+3A_x">x</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Cells-StdAssay_+3A_layer">layer</code></td>
<td>
<p>Layer to pull cells/features for; defaults to default layer;
if <code>NA</code>, returns all cells for the assay</p>
</td></tr>
<tr><td><code id="Cells-StdAssay_+3A_simplify">simplify</code></td>
<td>
<p>Simplify the cell/feature names into a single vector; if
<code>FALSE</code>, separates each cell/feature names by layer</p>
</td></tr>
<tr><td><code id="Cells-StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>Cell</code>: A vector of cell names
</p>
<p><code>Features</code>: A vector of feature names
</p>

<hr>
<h2 id='CellsByIdentities'>Get cell names grouped by identity class</h2><span id='topic+CellsByIdentities'></span>

<h3>Description</h3>

<p>Get cell names grouped by identity class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CellsByIdentities(object, idents = NULL, cells = NULL, return.null = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CellsByIdentities_+3A_object">object</code></td>
<td>
<p>A Seurat object</p>
</td></tr>
<tr><td><code id="CellsByIdentities_+3A_idents">idents</code></td>
<td>
<p>A vector of identity class levels to limit resulting list to;
defaults to all identity class levels</p>
</td></tr>
<tr><td><code id="CellsByIdentities_+3A_cells">cells</code></td>
<td>
<p>A vector of cells to grouping to</p>
</td></tr>
<tr><td><code id="CellsByIdentities_+3A_return.null">return.null</code></td>
<td>
<p>If no cells are requested, return a <code>NULL</code>;
by default, throws an error</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list where names are identity classes and values are vectors
of cells belonging to that class
</p>


<h3>Examples</h3>

<pre><code class='language-R'>CellsByIdentities(object = pbmc_small)

</code></pre>

<hr>
<h2 id='CellsByImage'>Get a vector of cell names associated with an image (or set of images)</h2><span id='topic+CellsByImage'></span>

<h3>Description</h3>

<p>Get a vector of cell names associated with an image (or set of images)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CellsByImage(object, images = NULL, unlist = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CellsByImage_+3A_object">object</code></td>
<td>
<p>Seurat object</p>
</td></tr>
<tr><td><code id="CellsByImage_+3A_images">images</code></td>
<td>
<p>Vector of image names</p>
</td></tr>
<tr><td><code id="CellsByImage_+3A_unlist">unlist</code></td>
<td>
<p>Return as a single vector of cell names as opposed to a list,
named by image name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of cell names
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
CellsByImage(object = object, images = "slice1")

## End(Not run)

</code></pre>

<hr>
<h2 id='Centroids-class'>The <code>Centroids</code> Class</h2><span id='topic+Centroids-class'></span>

<h3>Description</h3>

<p>The <code>Centroids</code> Class
</p>


<h3>Slots</h3>


<dl>
<dt><code>cells</code></dt><dd><p>(<code><a href="base.html#topic+character">character [n]</a></code>) A vector of cell
names; there should be as many cell names as there are points and no
duplicate names</p>
</dd>
<dt><code>nsides</code></dt><dd><p>(<code><a href="base.html#topic+integer">integer [1L]</a></code>) The number of sides
to draw when plotting centroids; must be either <code>0L</code> for circles or
greater than 3</p>
</dd>
<dt><code>radius</code></dt><dd><p>(<code><a href="base.html#topic+numeric">numeric [1L]</a></code>) The radius of the
shape when plotting the centroids</p>
</dd>
<dt><code>theta</code></dt><dd><p>(<code><a href="base.html#topic+numeric">numeric [1L]</a></code>) The angle in degrees
to adjust the shape when plotting the centroids</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code>Centroids</code> methods: <code><a href="#topic+Centroids-methods">Centroids-methods</a></code>
</p>
<p>Segmentation layer classes:
<code><a href="#topic+Centroids-methods">Centroids-methods</a></code>,
<code><a href="#topic+Molecules-class">Molecules-class</a></code>,
<code><a href="#topic+Molecules-methods">Molecules-methods</a></code>,
<code><a href="#topic+Segmentation-class">Segmentation-class</a></code>,
<code><a href="#topic+Segmentation-methods">Segmentation-methods</a></code>
</p>

<hr>
<h2 id='Centroids-methods'><code>Centroids</code> Methods</h2><span id='topic+Centroids-methods'></span><span id='topic+Cells.Centroids'></span><span id='topic+GetTissueCoordinates.Centroids'></span><span id='topic+Radius.Centroids'></span><span id='topic+RenameCells.Centroids'></span><span id='topic+Theta.Centroids'></span><span id='topic+is.finite.Centroids'></span><span id='topic+is.infinite.Centroids'></span><span id='topic+length.Centroids'></span><span id='topic+lengths.Centroids'></span><span id='topic+subset.Centroids'></span><span id='topic++5B+2CCentroids+2Ccharacter+2CANY+2CANY-method'></span><span id='topic++5B+2CCentroids+2Cnumeric+2CANY+2CANY-method'></span><span id='topic+show+2CCentroids-method'></span>

<h3>Description</h3>

<p>Methods for <code><a href="#topic+Centroids-class">Centroids</a></code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Centroids'
Cells(x, ...)

## S3 method for class 'Centroids'
GetTissueCoordinates(object, full = TRUE, ...)

## S3 method for class 'Centroids'
Radius(object)

## S3 method for class 'Centroids'
RenameCells(object, new.names = NULL, ...)

## S3 method for class 'Centroids'
Theta(object)

## S3 method for class 'Centroids'
is.finite(x)

## S3 method for class 'Centroids'
is.infinite(...)

## S3 method for class 'Centroids'
length(x)

## S3 method for class 'Centroids'
lengths(x, use.names = TRUE)

## S3 method for class 'Centroids'
subset(x, cells = NULL, ...)

## S4 method for signature 'Centroids,character,ANY,ANY'
x[i, j, ..., drop = TRUE]

## S4 method for signature 'Centroids,numeric,ANY,ANY'
x[i, j, ..., drop = TRUE]

## S4 method for signature 'Centroids'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Centroids-methods_+3A_x">x</code>, <code id="Centroids-methods_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+Centroids-class">Centroids</a></code>
object</p>
</td></tr>
<tr><td><code id="Centroids-methods_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Centroids-methods_+3A_full">full</code></td>
<td>
<p>Expand the coordinates to the full polygon</p>
</td></tr>
<tr><td><code id="Centroids-methods_+3A_new.names">new.names</code></td>
<td>
<p>vector of new cell names</p>
</td></tr>
<tr><td><code id="Centroids-methods_+3A_use.names">use.names</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="Centroids-methods_+3A_i">i</code>, <code id="Centroids-methods_+3A_cells">cells</code></td>
<td>
<p>A vector of cells to keep; if <code>NULL</code>, defaults
to all cells</p>
</td></tr>
<tr><td><code id="Centroids-methods_+3A_j">j</code>, <code id="Centroids-methods_+3A_drop">drop</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>GetTissueCoordinates</code>: Get cell spatial coordinates
</p>
<p><code>Radius</code>: Get the centroid radius
</p>
<p><code>RenameCells</code>: Update cell names
</p>
<p><code>Theta</code>: Get the offset angle
</p>
<p><code>is.finite</code>, <code>is.infinite</code>: Test to see if the centroids
are circular or polygonal
</p>
<p><code>length</code>: Get the number of sides for the polygonal centroid
</p>
<p><code>lengths</code>: Generate a run-length encoding of the cells present
</p>
<p><code>subset</code>, <code>[</code>: Subset a <code>Centroids</code> object to
certain cells
</p>
<p><code>show</code>: Display an object summary to stdout
</p>


<h3>Value</h3>

<p><code>GetTissueCoordinates</code>: A data frame with three columns:
</p>

<ul>
<li> <p>&ldquo;<code>x</code>&rdquo;: the x-coordinate
</p>
</li>
<li> <p>&ldquo;<code>y</code>&rdquo;: the y-coordinate
</p>
</li>
<li> <p>&ldquo;<code>cell</code>&rdquo;: the cell name
</p>
</li></ul>

<p>If <code>full</code> is <code>TRUE</code>, then each coordinate will indicate a vertex
for the cell polygon; otherwise, each coordinate will indicate a centroid
for the cell
</p>
<p><code>Radius</code> The radius of the centroids
</p>
<p><code>RenameCells</code>: <code>object</code> with the cells renamed to
<code>new.names</code>
</p>
<p><code>Theta</code>: The offset angle in degrees
</p>
<p><code>is.finite</code>: <code>TRUE</code> if the centroids are polygonal,
<code>FALSE</code> if circular
</p>
<p><code>is.infinite</code>: The opposite of <code>is.finite</code>
</p>
<p><code>length</code>: <code>0</code> if the centroids are circular, otherwise the
number of sides of the polygonal centroid
</p>
<p><code>lengths</code>: An <code><a href="base.html#topic+rle">rle</a></code> object for the cells
</p>
<p><code>subset</code>, <code>[</code>: <code>x</code> subsetted to the cells specified
by <code>cells</code>/<code>i</code>
</p>
<p><code>show</code>: Invisibly returns <code>NULL</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Centroids-class">Centroids-class</a></code>
</p>
<p>Segmentation layer classes:
<code><a href="#topic+Centroids-class">Centroids-class</a></code>,
<code><a href="#topic+Molecules-class">Molecules-class</a></code>,
<code><a href="#topic+Molecules-methods">Molecules-methods</a></code>,
<code><a href="#topic+Segmentation-class">Segmentation-class</a></code>,
<code><a href="#topic+Segmentation-methods">Segmentation-methods</a></code>
</p>

<hr>
<h2 id='CheckDots'>Check the Use of Dots</h2><span id='topic+CheckDots'></span>

<h3>Description</h3>

<p>Function to check the use of unused arguments passed to <code>...</code>; this
function is designed to be called from another function to see if an
argument passed to <code>...</code> remains unused and alert the user if so. Also
accepts a vector of function or function names to see if <code>...</code> can be
used in a downstream function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckDots(..., fxns = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckDots_+3A_...">...</code></td>
<td>
<p>Arguments passed to a function that fall under <code>...</code></p>
</td></tr>
<tr><td><code id="CheckDots_+3A_fxns">fxns</code></td>
<td>
<p>A list/vector of functions or function names</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Behavior of <code>CheckDots</code> can be controlled by the following option(s):
</p>

<dl>
<dt>&ldquo;<code>Seurat.checkdots</code>&rdquo;</dt><dd><p>Control how to alert the presence
of unused arguments in <code>...</code>; choose from
</p>

<ul>
<li> <p>&ldquo;<code>warn</code>&rdquo;: emit a warning (default)
</p>
</li>
<li> <p>&ldquo;<code>error</code>&rdquo;: throw an error
</p>
</li>
<li> <p>&ldquo;<code>silent</code>&rdquo;: no not alert the presence of unused
arguments in <code>...</code>
</p>
</li></ul>

</dd>
</dl>



<h3>Value</h3>

<p>Emits either an error or warning if an argument passed is unused;
invisibly returns <code>NULL</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
f &lt;- function(x, ...) {
  CheckDots(...)
  return(x ^ 2)
}
f(x = 3, y = 9)

## End(Not run)

</code></pre>

<hr>
<h2 id='CheckFeaturesNames'>Check features names format</h2><span id='topic+CheckFeaturesNames'></span>

<h3>Description</h3>

<p>Check features names format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckFeaturesNames(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckFeaturesNames_+3A_data">data</code></td>
<td>
<p>a matrix input, rownames(data) are feature names</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data</code> with update feature names
</p>

<hr>
<h2 id='CheckGC'>Conditional Garbage Collection</h2><span id='topic+CheckGC'></span>

<h3>Description</h3>

<p>Call <code>gc</code> only when desired
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckGC(option = "SeuratObject.memsafe")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckGC_+3A_option">option</code></td>
<td>
<p>...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>NULL</code>
</p>

<hr>
<h2 id='CheckLayersName'>Check layers names for the input list</h2><span id='topic+CheckLayersName'></span>

<h3>Description</h3>

<p>Check layers names for the input list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckLayersName(matrix.list, layers.type = c("counts", "data"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckLayersName_+3A_matrix.list">matrix.list</code></td>
<td>
<p>A list of matrices</p>
</td></tr>
<tr><td><code id="CheckLayersName_+3A_layers.type">layers.type</code></td>
<td>
<p>layers type, such as counts or data</p>
</td></tr>
</table>

<hr>
<h2 id='CheckMatrix'>Check Matrix Validity</h2><span id='topic+CheckMatrix'></span><span id='topic+CheckMatrix.default'></span><span id='topic+CheckMatrix.dMatrix'></span><span id='topic+CheckMatrix.lMatrix'></span>

<h3>Description</h3>

<p>Check Matrix Validity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckMatrix(object, checks, ...)

## Default S3 method:
CheckMatrix(object, checks, ...)

## S3 method for class 'dMatrix'
CheckMatrix(object, checks = c("infinite", "logical", "integer", "na"), ...)

## S3 method for class 'lMatrix'
CheckMatrix(object, checks = c("infinite", "logical", "integer", "na"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckMatrix_+3A_object">object</code></td>
<td>
<p>A matrix</p>
</td></tr>
<tr><td><code id="CheckMatrix_+3A_checks">checks</code></td>
<td>
<p>Type of checks to perform, choose one or more from:
</p>

<ul>
<li> <p>&ldquo;<code>infinite</code>&rdquo;: Emit a warning if any value is infinite
</p>
</li>
<li> <p>&ldquo;<code>logical</code>&rdquo;: Emit a warning if any value is a logical
</p>
</li>
<li> <p>&ldquo;<code>integer</code>&rdquo;: Emit a warning if any value is <em>not</em>
an integer
</p>
</li>
<li> <p>&ldquo;<code>na</code>&rdquo;: Emit a warning if any value is an <code>NA</code>
or <code>NaN</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="CheckMatrix_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Emits warnings for each test and invisibly returns <code>NULL</code>
</p>

<hr>
<h2 id='ClassKey'>Generate a Class Key</h2><span id='topic+ClassKey'></span>

<h3>Description</h3>

<p>Generate class keys for S4 classes. A class key follows the following
structure: &ldquo;<code>package:class</code>&rdquo;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClassKey(class, package = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClassKey_+3A_class">class</code></td>
<td>
<p>Class name</p>
</td></tr>
<tr><td><code id="ClassKey_+3A_package">package</code></td>
<td>
<p>Optional name of package; by default, will search namespaces
of loaded packages to determine the providing package</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The class key (&ldquo;<code>package:class</code>&rdquo;)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+s4list">s4list</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ClassKey("Seurat")

</code></pre>

<hr>
<h2 id='colMeans+2CAssay-method'>Row and Column Sums and Means</h2><span id='topic+colMeans+2CAssay-method'></span><span id='topic+colSums+2CAssay-method'></span><span id='topic+rowMeans+2CAssay-method'></span><span id='topic+rowSums+2CAssay-method'></span>

<h3>Description</h3>

<p>Calculate <code><a href="Matrix.html#topic+rowSums">rowSums</a></code>, <code><a href="Matrix.html#topic+colSums">colSums</a></code>,
<code><a href="Matrix.html#topic+rowMeans">rowMeans</a></code>, and <code><a href="Matrix.html#topic+colMeans">colMeans</a></code> on <code>Assay</code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Assay'
colMeans(x, na.rm = FALSE, dims = 1, ..., slot = "data")

## S4 method for signature 'Assay'
colSums(x, na.rm = FALSE, dims = 1, ..., slot = "data")

## S4 method for signature 'Assay'
rowMeans(x, na.rm = FALSE, dims = 1, ..., slot = "data")

## S4 method for signature 'Assay'
rowSums(x, na.rm = FALSE, dims = 1, ..., slot = "data")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colMeans+2B2CAssay-method_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id="colMeans+2B2CAssay-method_+3A_na.rm">na.rm</code></td>
<td>
<p>logical.  Should missing values (including <code>NaN</code>)
be omitted from the calculations?</p>
</td></tr>
<tr><td><code id="colMeans+2B2CAssay-method_+3A_dims">dims</code></td>
<td>
<p>completely ignored by the <code>Matrix</code> methods.</p>
</td></tr>
<tr><td><code id="colMeans+2B2CAssay-method_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="colMeans+2B2CAssay-method_+3A_slot">slot</code></td>
<td>
<p>Name of assay expression matrix to calculate column/row
means/sums on</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>colMeans</code>: The column (cell-wise) means of <code>slot</code>
</p>
<p><code>colSums</code>: The column (cell-wise) sums of <code>slot</code>
</p>
<p><code>rowMeans</code>: The row (feature-wise) means of <code>slot</code>
</p>
<p><code>rowSums</code>: The row (feature-wise) sums of <code>slot</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Assay">Assay</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]

colMeans(rna)

colSums(rna)

rowMeans(rna)

rowSums(rna)

</code></pre>

<hr>
<h2 id='colMeans+2CSeurat-method'>Row and Column Sums and Means</h2><span id='topic+colMeans+2CSeurat-method'></span><span id='topic+colSums+2CSeurat-method'></span><span id='topic+rowMeans+2CSeurat-method'></span><span id='topic+rowSums+2CSeurat-method'></span>

<h3>Description</h3>

<p>Calculate <code><a href="Matrix.html#topic+rowSums">rowSums</a></code>, <code><a href="Matrix.html#topic+colSums">colSums</a></code>,
<code><a href="Matrix.html#topic+rowMeans">rowMeans</a></code>, and <code><a href="Matrix.html#topic+colMeans">colMeans</a></code> on
<code><a href="#topic+Seurat">Seurat</a></code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Seurat'
colMeans(x, na.rm = FALSE, dims = 1, ..., slot = "data")

## S4 method for signature 'Seurat'
colSums(x, na.rm = FALSE, dims = 1, ..., slot = "data")

## S4 method for signature 'Seurat'
rowMeans(x, na.rm = FALSE, dims = 1, ..., slot = "data")

## S4 method for signature 'Seurat'
rowSums(x, na.rm = FALSE, dims = 1, ..., slot = "data")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colMeans+2B2CSeurat-method_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="colMeans+2B2CSeurat-method_+3A_na.rm">na.rm</code></td>
<td>
<p>logical.  Should missing values (including <code>NaN</code>)
be omitted from the calculations?</p>
</td></tr>
<tr><td><code id="colMeans+2B2CSeurat-method_+3A_dims">dims</code></td>
<td>
<p>completely ignored by the <code>Matrix</code> methods.</p>
</td></tr>
<tr><td><code id="colMeans+2B2CSeurat-method_+3A_...">...</code></td>
<td>
<p>potentially further arguments, for method <code>&lt;-&gt;</code>
generic compatibility.</p>
</td></tr>
<tr><td><code id="colMeans+2B2CSeurat-method_+3A_slot">slot</code></td>
<td>
<p>Name of assay expression matrix to calculate column/row
means/sums on</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>colMeans</code>: the column (cell-wise) means of <code>slot</code>
</p>
<p><code>colSums</code>: the column (cell-wise) sums of <code>slot</code>
</p>
<p><code>rowMeans</code>: the row (feature-wise) means of <code>slot</code>
</p>
<p><code>rowSums</code>: the row (feature-wise) sums of <code>slot</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Seurat">Seurat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(colMeans(pbmc_small))

head(colSums(pbmc_small))

head(rowMeans(pbmc_small))

head(rowSums(pbmc_small))

</code></pre>

<hr>
<h2 id='Command'>Get SeuratCommands</h2><span id='topic+Command'></span><span id='topic+Command.Seurat'></span>

<h3>Description</h3>

<p>Pull information on previously run commands in the Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Command(object, ...)

## S3 method for class 'Seurat'
Command(object, command = NULL, value = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Command_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Command_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Command_+3A_command">command</code></td>
<td>
<p>Name of the command to pull, pass <code>NULL</code> to get the
names of all commands run</p>
</td></tr>
<tr><td><code id="Command_+3A_value">value</code></td>
<td>
<p>Name of the parameter to pull the value for</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a SeuratCommand object or the requested parameter value
</p>

<hr>
<h2 id='CreateAssay5Object'>Create a v5 Assay object</h2><span id='topic+CreateAssay5Object'></span>

<h3>Description</h3>

<p>Create an <code><a href="#topic+Assay5">Assay5</a></code> object from a feature expression matrix;
the expected format of the matrix is features x cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateAssay5Object(
  counts = NULL,
  data = NULL,
  min.cells = 0,
  min.features = 0,
  csum = NULL,
  fsum = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateAssay5Object_+3A_counts">counts</code></td>
<td>
<p>A two-dimensional expression matrix</p>
</td></tr>
<tr><td><code id="CreateAssay5Object_+3A_data">data</code></td>
<td>
<p>Optional prenormalized data matrix</p>
</td></tr>
<tr><td><code id="CreateAssay5Object_+3A_min.cells">min.cells</code></td>
<td>
<p>Include features detected in at least this many cells;
will subset the counts matrix as well. To reintroduce excluded features,
create a new object with a lower cutoff</p>
</td></tr>
<tr><td><code id="CreateAssay5Object_+3A_min.features">min.features</code></td>
<td>
<p>Include cells where at least this many features
are detected</p>
</td></tr>
<tr><td><code id="CreateAssay5Object_+3A_csum">csum</code></td>
<td>
<p>Function for calculating cell sums</p>
</td></tr>
<tr><td><code id="CreateAssay5Object_+3A_fsum">fsum</code></td>
<td>
<p>Function for calculating feature sums</p>
</td></tr>
<tr><td><code id="CreateAssay5Object_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code><a href="#topic+Assay5">Assay5</a></code> object
</p>

<hr>
<h2 id='CreateAssayObject'>Create an Assay object</h2><span id='topic+CreateAssayObject'></span>

<h3>Description</h3>

<p>Create an Assay object from a feature (e.g. gene) expression matrix. The
expected format of the input matrix is features x cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateAssayObject(
  counts,
  data,
  min.cells = 0,
  min.features = 0,
  key = NULL,
  check.matrix = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateAssayObject_+3A_counts">counts</code></td>
<td>
<p>Unnormalized data such as raw counts or TPMs</p>
</td></tr>
<tr><td><code id="CreateAssayObject_+3A_data">data</code></td>
<td>
<p>Prenormalized data; if provided, do not pass <code>counts</code></p>
</td></tr>
<tr><td><code id="CreateAssayObject_+3A_min.cells">min.cells</code></td>
<td>
<p>Include features detected in at least this many cells. Will
subset the counts matrix as well. To reintroduce excluded features, create a
new object with a lower cutoff</p>
</td></tr>
<tr><td><code id="CreateAssayObject_+3A_min.features">min.features</code></td>
<td>
<p>Include cells where at least this many features are
detected</p>
</td></tr>
<tr><td><code id="CreateAssayObject_+3A_key">key</code></td>
<td>
<p>Optional key to initialize assay with</p>
</td></tr>
<tr><td><code id="CreateAssayObject_+3A_check.matrix">check.matrix</code></td>
<td>
<p>Check counts matrix for NA, NaN, Inf, and
non-integer values</p>
</td></tr>
<tr><td><code id="CreateAssayObject_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+as.sparse">as.sparse</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Non-unique cell or feature names are not allowed. Please make unique before
calling this function.
</p>


<h3>Value</h3>

<p>A <code><a href="#topic+Assay">Assay</a></code> object
</p>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pbmc_raw &lt;- read.table(
  file = system.file('extdata', 'pbmc_raw.txt', package = 'Seurat'),
  as.is = TRUE
)
pbmc_rna &lt;- CreateAssayObject(counts = pbmc_raw)
pbmc_rna

## End(Not run)

</code></pre>

<hr>
<h2 id='CreateCentroids'>Create a <code><a href="#topic+Centroids-class">Centroids</a></code> Objects</h2><span id='topic+CreateCentroids'></span>

<h3>Description</h3>

<p>Create a <code><a href="#topic+Centroids-class">Centroids</a></code> Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateCentroids(coords, nsides, radius, theta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateCentroids_+3A_coords">coords</code></td>
<td>
<p>The coordinates of cell/spot centroids</p>
</td></tr>
<tr><td><code id="CreateCentroids_+3A_nsides">nsides</code></td>
<td>
<p>The number of sides to represent cells/spots; pass
<code><a href="base.html#topic+Inf">Inf</a></code> to plot as circles</p>
</td></tr>
<tr><td><code id="CreateCentroids_+3A_radius">radius</code></td>
<td>
<p>Radius of shapes when plotting</p>
</td></tr>
<tr><td><code id="CreateCentroids_+3A_theta">theta</code></td>
<td>
<p>Angle to adjust shapes when plotting</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+Centroids-class">Centroids</a></code> object
</p>

<hr>
<h2 id='CreateDimReducObject'>Create a DimReduc object</h2><span id='topic+CreateDimReducObject'></span><span id='topic+SetDimReduction'></span>

<h3>Description</h3>

<p>Create a DimReduc object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateDimReducObject(
  embeddings = new(Class = "matrix"),
  loadings = new(Class = "matrix"),
  projected = new(Class = "matrix"),
  assay = NULL,
  stdev = numeric(),
  key = NULL,
  global = FALSE,
  jackstraw = NULL,
  misc = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateDimReducObject_+3A_embeddings">embeddings</code></td>
<td>
<p>A matrix with the cell embeddings</p>
</td></tr>
<tr><td><code id="CreateDimReducObject_+3A_loadings">loadings</code></td>
<td>
<p>A matrix with the feature loadings</p>
</td></tr>
<tr><td><code id="CreateDimReducObject_+3A_projected">projected</code></td>
<td>
<p>A matrix with the projected feature loadings</p>
</td></tr>
<tr><td><code id="CreateDimReducObject_+3A_assay">assay</code></td>
<td>
<p>Assay used to calculate this dimensional reduction</p>
</td></tr>
<tr><td><code id="CreateDimReducObject_+3A_stdev">stdev</code></td>
<td>
<p>Standard deviation (if applicable) for the dimensional reduction</p>
</td></tr>
<tr><td><code id="CreateDimReducObject_+3A_key">key</code></td>
<td>
<p>A character string to facilitate looking up features from a
specific DimReduc</p>
</td></tr>
<tr><td><code id="CreateDimReducObject_+3A_global">global</code></td>
<td>
<p>Specify this as a global reduction (useful for visualizations)</p>
</td></tr>
<tr><td><code id="CreateDimReducObject_+3A_jackstraw">jackstraw</code></td>
<td>
<p>Results from the JackStraw function</p>
</td></tr>
<tr><td><code id="CreateDimReducObject_+3A_misc">misc</code></td>
<td>
<p>list for the user to store any additional information associated
with the dimensional reduction</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+DimReduc">DimReduc</a></code> object
</p>


<h3>See Also</h3>

<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+DimReduc-class">DimReduc-class</a></code>,
<code><a href="#topic+DimReduc-validity">DimReduc-validity</a></code>,
<code><a href="#topic++5B.DimReduc">[.DimReduc</a>()</code>,
<code><a href="#topic++5B+5B.DimReduc">[[.DimReduc</a>()</code>,
<code><a href="#topic+dim.DimReduc">dim.DimReduc</a>()</code>,
<code><a href="#topic+merge.DimReduc">merge.DimReduc</a>()</code>,
<code><a href="#topic+print.DimReduc">print.DimReduc</a>()</code>,
<code><a href="#topic+subset.DimReduc">subset.DimReduc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- GetAssayData(pbmc_small[["RNA"]], slot = "scale.data")
pcs &lt;- prcomp(x = data)
pca.dr &lt;- CreateDimReducObject(
  embeddings = pcs$rotation,
  loadings = pcs$x,
  stdev = pcs$sdev,
  key = "PC",
  assay = "RNA"
)

</code></pre>

<hr>
<h2 id='CreateFOV'>Create Spatial Coordinates</h2><span id='topic+CreateFOV'></span><span id='topic+CreateFOV.Centroids'></span><span id='topic+CreateFOV.data.frame'></span><span id='topic+CreateFOV.list'></span><span id='topic+CreateFOV.Segmentation'></span>

<h3>Description</h3>

<p>Create Spatial Coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateFOV(coords, ...)

## S3 method for class 'Centroids'
CreateFOV(
  coords,
  molecules = NULL,
  assay = "Spatial",
  key = NULL,
  name = NULL,
  ...
)

## S3 method for class 'data.frame'
CreateFOV(
  coords,
  type = c("segmentation", "centroids"),
  nsides = Inf,
  radius = NULL,
  theta = 0L,
  molecules = NULL,
  assay = "Spatial",
  key = NULL,
  name = NULL,
  ...
)

## S3 method for class 'list'
CreateFOV(coords, molecules = NULL, assay = "Spatial", key = NULL, ...)

## S3 method for class 'Segmentation'
CreateFOV(
  coords,
  molecules = NULL,
  assay = "Spatial",
  key = NULL,
  name = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateFOV_+3A_coords">coords</code></td>
<td>
<p>Spatial coordinates</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_molecules">molecules</code></td>
<td>
<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code> with spatially-resolved
molecule information or a
<code><a href="#topic+Molecules-class">Molecules</a></code> object</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_assay">assay</code></td>
<td>
<p>Name of associated assay</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_key">key</code></td>
<td>
<p>Key for these spatial coordinates</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_name">name</code></td>
<td>
<p>When <code>coords</code> is a <code><a href="base.html#topic+data.frame">data.frame</a></code>,
<code><a href="#topic+Centroids-class">Centroids</a></code>, or
<code><a href="#topic+Segmentation-class">Segmentation</a></code>, name
to store coordinates as</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_type">type</code></td>
<td>
<p>When providing a <code><a href="base.html#topic+data.frame">data.frame</a></code>, specify if
the coordinates represent a cell segmentation or voxel centroids</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_nsides">nsides</code></td>
<td>
<p>The number of sides to represent cells/spots; pass
<code><a href="base.html#topic+Inf">Inf</a></code> to plot as circles</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_radius">radius</code></td>
<td>
<p>Radius of shapes when plotting</p>
</td></tr>
<tr><td><code id="CreateFOV_+3A_theta">theta</code></td>
<td>
<p>Angle to adjust shapes when plotting</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+FOV">FOV</a></code> object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FOV-class">FOV-class</a></code>
</p>

<hr>
<h2 id='CreateMolecules'>Create a <code><a href="#topic+Molecules">Molecules</a></code> Object</h2><span id='topic+CreateMolecules'></span><span id='topic+CreateMolecules.data.frame'></span><span id='topic+CreateMolecules.Molecules'></span><span id='topic+CreateMolecules.NULL'></span>

<h3>Description</h3>

<p>Create a <code><a href="#topic+Molecules">Molecules</a></code> Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateMolecules(coords, ...)

## S3 method for class 'data.frame'
CreateMolecules(coords, key = "", ...)

## S3 method for class 'Molecules'
CreateMolecules(coords, ...)

## S3 method for class ''NULL''
CreateMolecules(coords, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateMolecules_+3A_coords">coords</code></td>
<td>
<p>Spatial coordinates for molecules; should be a data frame
with three columns:
</p>

<ul>
<li> <p>&ldquo;<code>x</code>&rdquo;: x-coordinates for each molecule
</p>
</li>
<li> <p>&ldquo;<code>y</code>&rdquo;: y-coordinates for each molecule
</p>
</li>
<li> <p>&ldquo;<code>gene</code>&rdquo;: gene name for each molecule
</p>
</li></ul>
</td></tr>
<tr><td><code id="CreateMolecules_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="CreateMolecules_+3A_key">key</code></td>
<td>
<p>A key to set for the molecules</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+Molecules">Molecules</a></code> object
</p>

<hr>
<h2 id='CreateSegmentation'>Create a <code><a href="#topic+Segmentation-class">Segmentation</a></code> Objects</h2><span id='topic+CreateSegmentation'></span><span id='topic+CreateSegmentation.data.frame'></span><span id='topic+CreateSegmentation.Segmentation'></span>

<h3>Description</h3>

<p>Create a <code><a href="#topic+Segmentation-class">Segmentation</a></code> Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateSegmentation(coords)

## S3 method for class 'data.frame'
CreateSegmentation(coords)

## S3 method for class 'Segmentation'
CreateSegmentation(coords)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateSegmentation_+3A_coords">coords</code></td>
<td>
<p>The coordinates of cell segmentations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+Segmentation-class">Segmentation</a></code> object
</p>

<hr>
<h2 id='CreateSeuratObject'>Create a <code>Seurat</code> object</h2><span id='topic+CreateSeuratObject'></span><span id='topic+CreateSeuratObject.default'></span><span id='topic+CreateSeuratObject.Assay'></span><span id='topic+CreateSeuratObject.Assay5'></span>

<h3>Description</h3>

<p>Create a <code>Seurat</code> object from raw data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateSeuratObject(
  counts,
  assay = "RNA",
  names.field = 1,
  names.delim = "_",
  meta.data = NULL,
  project = "CreateSeuratObject",
  ...
)

## Default S3 method:
CreateSeuratObject(
  counts,
  assay = "RNA",
  names.field = 1L,
  names.delim = "_",
  meta.data = NULL,
  project = "SeuratProject",
  min.cells = 0,
  min.features = 0,
  ...
)

## S3 method for class 'Assay'
CreateSeuratObject(
  counts,
  assay = "RNA",
  names.field = 1L,
  names.delim = "_",
  meta.data = NULL,
  project = "SeuratProject",
  ...
)

## S3 method for class 'Assay5'
CreateSeuratObject(
  counts,
  assay = "RNA",
  names.field = 1L,
  names.delim = "_",
  meta.data = NULL,
  project = "SeuratProject",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateSeuratObject_+3A_counts">counts</code></td>
<td>
<p>Either a <code><a href="base.html#topic+matrix">matrix</a></code>-like object with
unnormalized data with cells as columns and features as rows or an
<code><a href="#topic+Assay">Assay</a></code>-derived object</p>
</td></tr>
<tr><td><code id="CreateSeuratObject_+3A_assay">assay</code></td>
<td>
<p>Name of the initial assay</p>
</td></tr>
<tr><td><code id="CreateSeuratObject_+3A_names.field">names.field</code></td>
<td>
<p>For the initial identity class for each cell, choose this
field from the cell's name. E.g. If your cells are named as
BARCODE_CLUSTER_CELLTYPE in the input matrix, set <code>names.field</code> to 3 to
set the initial identities to CELLTYPE.</p>
</td></tr>
<tr><td><code id="CreateSeuratObject_+3A_names.delim">names.delim</code></td>
<td>
<p>For the initial identity class for each cell, choose this
delimiter from the cell's column name. E.g. If your cells are named as
BARCODE-CLUSTER-CELLTYPE, set this to &ldquo;-&rdquo; to separate the cell name
into its component parts for picking the relevant field.</p>
</td></tr>
<tr><td><code id="CreateSeuratObject_+3A_meta.data">meta.data</code></td>
<td>
<p>Additional cell-level metadata to add to the Seurat object.
Should be a <code><a href="base.html#topic+data.frame">data.frame</a></code> where the rows are cell names and
the columns are additional metadata fields. Row names in the metadata need
to match the column names of the counts matrix.</p>
</td></tr>
<tr><td><code id="CreateSeuratObject_+3A_project">project</code></td>
<td>
<p><a href="#topic+Project">Project</a> name for the <code>Seurat</code> object</p>
</td></tr>
<tr><td><code id="CreateSeuratObject_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="CreateSeuratObject_+3A_min.cells">min.cells</code></td>
<td>
<p>Include features detected in at least this many cells. Will
subset the counts matrix as well. To reintroduce excluded features, create a
new object with a lower cutoff</p>
</td></tr>
<tr><td><code id="CreateSeuratObject_+3A_min.features">min.features</code></td>
<td>
<p>Include cells where at least this many features are
detected</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+Seurat">Seurat</a></code> object
</p>


<h3>Note</h3>

<p>In previous versions (&lt;3.0), this function also accepted a parameter to
set the expression threshold for a &lsquo;detected&rsquo; feature (gene). This
functionality has been removed to simplify the initialization
process/assumptions. If you would still like to impose this threshold for
your particular dataset, simply filter the input expression matrix before
calling this function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pbmc_raw &lt;- read.table(
  file = system.file('extdata', 'pbmc_raw.txt', package = 'Seurat'),
  as.is = TRUE
)
pbmc_small &lt;- CreateSeuratObject(counts = pbmc_raw)
pbmc_small

## End(Not run)

</code></pre>

<hr>
<h2 id='Crop'>Crop Coordinates</h2><span id='topic+Crop'></span><span id='topic+Crop.FOV'></span>

<h3>Description</h3>

<p>Crop Coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Crop(object, x = NULL, y = NULL, coords = c("plot", "tissue"), ...)

## S3 method for class 'FOV'
Crop(object, x = NULL, y = NULL, coords = c("plot", "tissue"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Crop_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Crop_+3A_x">x</code>, <code id="Crop_+3A_y">y</code></td>
<td>
<p>Range to crop x/y limits to; if <code>NULL</code>, uses full range of
<code>x</code>/<code>y</code></p>
</td></tr>
<tr><td><code id="Crop_+3A_coords">coords</code></td>
<td>
<p>Coordinate system to execute crop; choose from:
</p>

<ul>
<li> <p>&ldquo;<code>plot</code>&rdquo;: Coordinates as shown when plotting
</p>
</li>
<li> <p>&ldquo;<code>tissue</code>&rdquo;: Coordinates from
<code><a href="#topic+GetTissueCoordinates">GetTissueCoordinates</a></code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="Crop_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>object</code> cropped to the region specified by <code>x</code>
and <code>y</code>
</p>

<hr>
<h2 id='DefaultAssay'>Default Assay</h2><span id='topic+DefaultAssay'></span><span id='topic+DefaultAssay+3C-'></span><span id='topic+DefaultAssay.Graph'></span><span id='topic+DefaultAssay+3C-.Graph'></span><span id='topic+DefaultAssay.Assay'></span><span id='topic+DefaultAssay+3C-.Assay'></span><span id='topic+DefaultAssay.Assay5'></span><span id='topic+DefaultAssay+3C-.Assay5'></span><span id='topic+DefaultAssay.SeuratCommand'></span><span id='topic+DefaultAssay.DimReduc'></span><span id='topic+DefaultAssay+3C-.DimReduc'></span><span id='topic+DefaultAssay.Seurat'></span><span id='topic+DefaultAssay+3C-.Seurat'></span>

<h3>Description</h3>

<p>Get and set the default assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DefaultAssay(object, ...)

DefaultAssay(object, ...) &lt;- value

## S3 method for class 'Graph'
DefaultAssay(object, ...)

## S3 replacement method for class 'Graph'
DefaultAssay(object, ...) &lt;- value

## S3 method for class 'Assay'
DefaultAssay(object, ...)

## S3 replacement method for class 'Assay'
DefaultAssay(object, ...) &lt;- value

## S3 method for class 'Assay5'
DefaultAssay(object, ...)

## S3 replacement method for class 'Assay5'
DefaultAssay(object, ...) &lt;- value

## S3 method for class 'SeuratCommand'
DefaultAssay(object, ...)

## S3 method for class 'DimReduc'
DefaultAssay(object, ...)

## S3 replacement method for class 'DimReduc'
DefaultAssay(object, ...) &lt;- value

## S3 method for class 'Seurat'
DefaultAssay(object, ...)

## S3 replacement method for class 'Seurat'
DefaultAssay(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DefaultAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="DefaultAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="DefaultAssay_+3A_value">value</code></td>
<td>
<p>Name of assay to set as default</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>DefaultAssay</code>: The name of the default assay
</p>
<p><code>DefaultAssay&lt;-</code>: An object with the default assay updated
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get current default assay
DefaultAssay(object = pbmc_small)

# Create dummy new assay to demo switching default assays
new.assay &lt;- pbmc_small[["RNA"]]
Key(object = new.assay) &lt;- "RNA2_"
pbmc_small[["RNA2"]] &lt;- new.assay
# switch default assay to RNA2
DefaultAssay(object = pbmc_small) &lt;- "RNA2"
DefaultAssay(object = pbmc_small)

</code></pre>

<hr>
<h2 id='DefaultAssay-StdAssay'>Default Assay</h2><span id='topic+DefaultAssay-StdAssay'></span><span id='topic+DefaultAssay.StdAssay'></span><span id='topic+DefaultAssay+3C-.StdAssay'></span>

<h3>Description</h3>

<p>Get and set the default assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
DefaultAssay(object, ...)

## S3 replacement method for class 'StdAssay'
DefaultAssay(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DefaultAssay-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="DefaultAssay-StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="DefaultAssay-StdAssay_+3A_value">value</code></td>
<td>
<p>Name of assay to set as default</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>DefaultAssay</code>: The name of the default assay
</p>
<p><code>DefaultAssay&lt;-</code>: An object with the default assay updated
</p>

<hr>
<h2 id='DefaultDimReduc'>Find the default <code><a href="#topic+DimReduc">DimReduc</a></code></h2><span id='topic+DefaultDimReduc'></span>

<h3>Description</h3>

<p>Searches for <code><a href="#topic+DimReduc">DimReduc</a>s</code> matching &ldquo;umap&rdquo;, &ldquo;tsne&rdquo;,
or &ldquo;pca&rdquo;, case-insensitive, and in that order. Priority given to
<code><a href="#topic+DimReduc">DimReduc</a>s</code> matching the <code>DefaultAssay</code> or assay specified
(eg. &ldquo;pca&rdquo; for the default assay weights higher than &ldquo;umap&rdquo;
for a non-default assay)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DefaultDimReduc(object, assay = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DefaultDimReduc_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="DefaultDimReduc_+3A_assay">assay</code></td>
<td>
<p>Name of assay to use; defaults to the default assay of the object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The default <code><a href="#topic+DimReduc">DimReduc</a></code>, if possible
</p>


<h3>Examples</h3>

<pre><code class='language-R'>DefaultDimReduc(pbmc_small)

</code></pre>

<hr>
<h2 id='DefaultFOV'>Get and Set the Default FOV</h2><span id='topic+DefaultFOV'></span><span id='topic+DefaultFOV+3C-'></span><span id='topic+DefaultFOV.Seurat'></span><span id='topic+DefaultFOV+3C-.Seurat'></span>

<h3>Description</h3>

<p>Get and Set the Default FOV
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DefaultFOV(object, ...)

DefaultFOV(object, ...) &lt;- value

## S3 method for class 'Seurat'
DefaultFOV(object, assay = NULL, ...)

## S3 replacement method for class 'Seurat'
DefaultFOV(object, assay = NA, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DefaultFOV_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> Object</p>
</td></tr>
<tr><td><code id="DefaultFOV_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="DefaultFOV_+3A_value">value</code></td>
<td>
<p>The name of the <code><a href="#topic+FOV">FOV</a></code> to set as the default</p>
</td></tr>
<tr><td><code id="DefaultFOV_+3A_assay">assay</code></td>
<td>
<p>Name of assay to get or set default <code><a href="#topic+FOV">FOV</a></code> for;
pass <code>NA</code> to get or set the global default <code><a href="#topic+FOV">FOV</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>DefaultFOV</code>: The name of the default <code><a href="#topic+FOV">FOV</a></code>
</p>
<p><code>DefaultFOV&lt;-</code>: <code>object</code> with the default FOV set
to <code>value</code>
</p>

<hr>
<h2 id='DefaultLayer'>Default Layer</h2><span id='topic+DefaultLayer'></span><span id='topic+DefaultLayer+3C-'></span><span id='topic+DefaultLayer.Assay'></span><span id='topic+DefaultLayer.Assay5'></span><span id='topic+DefaultLayer+3C-.Assay5'></span>

<h3>Description</h3>

<p>Get and set the default layer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DefaultLayer(object, ...)

DefaultLayer(object, ...) &lt;- value

## S3 method for class 'Assay'
DefaultLayer(object, ...)

## S3 method for class 'Assay5'
DefaultLayer(object, ...)

## S3 replacement method for class 'Assay5'
DefaultLayer(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DefaultLayer_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="DefaultLayer_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="DefaultLayer_+3A_value">value</code></td>
<td>
<p>Name of layer to set as default</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>DefaultLayer</code>: The name of the default layer
</p>
<p><code>DefaultLayer&lt;-</code>: An object with the default layer updated
</p>

<hr>
<h2 id='DefaultLayer-StdAssay'>Default Layer</h2><span id='topic+DefaultLayer-StdAssay'></span><span id='topic+DefaultLayer.StdAssay'></span><span id='topic+DefaultLayer+3C-.StdAssay'></span>

<h3>Description</h3>

<p>Get and set the default layer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
DefaultLayer(object, ...)

## S3 replacement method for class 'StdAssay'
DefaultLayer(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DefaultLayer-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="DefaultLayer-StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="DefaultLayer-StdAssay_+3A_value">value</code></td>
<td>
<p>Name of layer to set as default</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>DefaultLayer</code>: The name of the default layer
</p>
<p><code>DefaultLayer&lt;-</code>: An object with the default layer updated
</p>

<hr>
<h2 id='dim.Assay'>Feature and Cell Numbers</h2><span id='topic+dim.Assay'></span>

<h3>Description</h3>

<p>Feature and Cell Numbers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
dim(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-length numeric vector with the total number of
features and cells in <code>x</code>
</p>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]
dim(rna)

</code></pre>

<hr>
<h2 id='dim.Assay5'>Feature and Cell Numbers</h2><span id='topic+dim.Assay5'></span>

<h3>Description</h3>

<p>Feature and Cell Numbers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
dim(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-length numeric vector with the total number of
features and cells in <code>x</code>
</p>


<h3>See Also</h3>

<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>

<hr>
<h2 id='dim.DimReduc'>Dimensional Reduction Meta-Information</h2><span id='topic+dim.DimReduc'></span><span id='topic+dimnames.DimReduc'></span><span id='topic+length.DimReduc'></span><span id='topic+names.DimReduc'></span>

<h3>Description</h3>

<p>Pull meta-information about cells and dimensions for a given
<a href="#topic+DimReduc">dimensional reduction</a>; cell meta-information is stored
as row meta-information (eg. <code>nrow</code>, <code>rownames</code>) and dimension
meta-information is stored as column meta-information (eg. <code>ncol</code>,
<code>colnames</code>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DimReduc'
dim(x)

## S3 method for class 'DimReduc'
dimnames(x)

## S3 method for class 'DimReduc'
length(x)

## S3 method for class 'DimReduc'
names(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim.DimReduc_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+DimReduc">DimReduc</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dim</code>: The number of cells (<code>nrow</code>) and dimensions
(<code>ncol</code>)
</p>
<p><code>dimnames</code>: The cell (row) and dimension (column) names
</p>
<p><code>length</code>: The number of dimensions
</p>
<p><code>names</code>: The dimension identifiers
</p>


<h3>See Also</h3>

<p><code>Cells</code>
</p>
<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+CreateDimReducObject">CreateDimReducObject</a>()</code>,
<code><a href="#topic+DimReduc-class">DimReduc-class</a></code>,
<code><a href="#topic+DimReduc-validity">DimReduc-validity</a></code>,
<code><a href="#topic++5B.DimReduc">[.DimReduc</a>()</code>,
<code><a href="#topic++5B+5B.DimReduc">[[.DimReduc</a>()</code>,
<code><a href="#topic+merge.DimReduc">merge.DimReduc</a>()</code>,
<code><a href="#topic+print.DimReduc">print.DimReduc</a>()</code>,
<code><a href="#topic+subset.DimReduc">subset.DimReduc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- pbmc_small[["pca"]]
pca
dim(pca)

# nrow is number of cells
nrow(pca)

# rownames pulls cell names
head(rownames(pca))

# ncol and length are number of dimensions
ncol(pca)
length(pca)

# colnames and names pull dimension identifiers
head(colnames(pca))
head(names(pca))

</code></pre>

<hr>
<h2 id='dim.Seurat'>Feature and Cell Numbers</h2><span id='topic+dim.Seurat'></span>

<h3>Description</h3>

<p>Feature and Cell Numbers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
dim(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim.Seurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-length numeric vector with the total number of
features and cells in <code>x</code>
</p>


<h3>See Also</h3>

<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the number of features in an object
nrow(pbmc_small)

# Get the number of cells in an object
ncol(pbmc_small)

</code></pre>

<hr>
<h2 id='dim.StdAssay'>Feature and Cell Numbers</h2><span id='topic+dim.StdAssay'></span>

<h3>Description</h3>

<p>Feature and Cell Numbers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
dim(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-length numeric vector with the total number of
features and cells in <code>x</code>
</p>


<h3>See Also</h3>

<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='dimnames.Assay'>Assay-Level Feature and Cell Names</h2><span id='topic+dimnames.Assay'></span><span id='topic+dimnames+3C-.Assay'></span>

<h3>Description</h3>

<p>Get and set feature and cell names in v5 Assays
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
dimnames(x)

## S3 replacement method for class 'Assay'
dimnames(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dimnames.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id="dimnames.Assay_+3A_value">value</code></td>
<td>
<p>A two-length list where the first entry is the existing feature
names for <code>x</code> and the second entry is the <em>updated</em> cell names
for <code>x</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dimnames</code>: A two-length list with the following values:
</p>

<ul>
<li><p> A character vector will all features in <code>x</code>
</p>
</li>
<li><p> A character vector will all cells in <code>x</code>
</p>
</li></ul>

<p><code>dimnames&lt;-</code>: <code>x</code> with the cell names updated to those
in <code>value[[2L]]</code>
</p>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>
<p><code><a href="#topic+Cells">Cells</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]

# Feature and cell names can be acquired with `rownames` and `colnames`
head(rownames(rna))
head(colnames(rna))

# Cell names can be updated with `colnames&lt;-`
colnames(rna)[1] &lt;- "newcell"
head(colnames(rna))

</code></pre>

<hr>
<h2 id='dimnames.Assay5'>Assay-Level Feature and Cell Names</h2><span id='topic+dimnames.Assay5'></span><span id='topic+dimnames+3C-.Assay5'></span>

<h3>Description</h3>

<p>Get and set feature and cell names in v5 Assays
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
dimnames(x)

## S3 replacement method for class 'Assay5'
dimnames(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dimnames.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="dimnames.Assay5_+3A_value">value</code></td>
<td>
<p>A two-length list with updated feature and/or cells names</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dimnames</code>: A two-length list with the following values:
</p>

<ul>
<li><p> A character vector with all features in <code>x</code>
</p>
</li>
<li><p> A character vector with all cells in <code>x</code>
</p>
</li></ul>

<p><code>dimnames&lt;-</code>: <code>x</code> with the feature and/or cell
names updated to <code>value</code>
</p>


<h3>See Also</h3>

<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>
<p><code><a href="#topic+Cells">Cells</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>
</p>

<hr>
<h2 id='dimnames.Seurat'>Feature and Cell Names</h2><span id='topic+dimnames.Seurat'></span><span id='topic+dimnames+3C-.Seurat'></span>

<h3>Description</h3>

<p>Get and set feature and cell inames in <code><a href="#topic+Seurat">Seurat</a></code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
dimnames(x)

## S3 replacement method for class 'Seurat'
dimnames(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dimnames.Seurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="dimnames.Seurat_+3A_value">value</code></td>
<td>
<p>A two-length list with updated feature and/or cells names</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dimnames</code>: A two-length list with the following values:
</p>

<ul>
<li><p> A character vector with all features in the
<a href="#topic+DefaultAssay">default assay</a>
</p>
</li>
<li><p> A character vector with all cells in <code>x</code>
</p>
</li></ul>

<p><code>dimnames&lt;-</code>: <code>x</code> with the feature and/or cell
names updated to <code>value</code>
</p>


<h3>See Also</h3>

<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>
<p><code><a href="#topic+Cells">Cells</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the feature names of an object
head(rownames(pbmc_small))

# Get the cell names of an object
head(colnames(pbmc_small))

colnames(pbmc_small)[1] &lt;- "newcell"
head(colnames(pbmc_small))

</code></pre>

<hr>
<h2 id='dimnames.StdAssay'>Assay-Level Feature and Cell Names</h2><span id='topic+dimnames.StdAssay'></span><span id='topic+dimnames+3C-.StdAssay'></span>

<h3>Description</h3>

<p>Get and set feature and cell names in v5 Assays
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
dimnames(x)

## S3 replacement method for class 'StdAssay'
dimnames(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dimnames.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="dimnames.StdAssay_+3A_value">value</code></td>
<td>
<p>A two-length list with updated feature and/or cells names</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dimnames</code>: A two-length list with the following values:
</p>

<ul>
<li><p> A character vector with all features in <code>x</code>
</p>
</li>
<li><p> A character vector with all cells in <code>x</code>
</p>
</li></ul>

<p><code>dimnames&lt;-</code>: <code>x</code> with the feature and/or cell
names updated to <code>value</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Cells">Cells</a></code> <code><a href="#topic+Features">Features</a></code>
</p>
<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='DimReduc-class'>The Dimensional Reduction Class</h2><span id='topic+DimReduc-class'></span><span id='topic+DimReduc'></span>

<h3>Description</h3>

<p>The DimReduc object stores a dimensionality reduction taken out in Seurat;
each DimReduc consists of a cell embeddings matrix, a feature loadings
matrix, and a projected feature loadings matrix.
</p>


<h3>Slots</h3>


<dl>
<dt><code>cell.embeddings</code></dt><dd><p>Cell embeddings matrix (required)</p>
</dd>
<dt><code>feature.loadings</code></dt><dd><p>Feature loadings matrix (optional)</p>
</dd>
<dt><code>feature.loadings.projected</code></dt><dd><p>Projected feature loadings matrix (optional)</p>
</dd>
<dt><code>assay.used</code></dt><dd><p>Name of assay used to generate <code>DimReduc</code> object</p>
</dd>
<dt><code>global</code></dt><dd><p>Is this <code>DimReduc</code> global/persistent? If so, it will not be
removed when removing its associated assay</p>
</dd>
<dt><code>stdev</code></dt><dd><p>A vector of standard deviations</p>
</dd>
<dt><code>jackstraw</code></dt><dd><p>A <code><a href="#topic+JackStrawData-class">JackStrawData-class</a></code> object associated with
this <code>DimReduc</code></p>
</dd>
<dt><code>misc</code></dt><dd><p>A named list of unstructured miscellaneous data</p>
</dd>
<dt><code>key</code></dt><dd><p>A one-length character vector with the object's key; keys must
be one or more alphanumeric characters followed by an underscore
&ldquo;<code>_</code>&rdquo; (regex pattern
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;)</p>
</dd>
</dl>


<h3>See Also</h3>

<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+CreateDimReducObject">CreateDimReducObject</a>()</code>,
<code><a href="#topic+DimReduc-validity">DimReduc-validity</a></code>,
<code><a href="#topic++5B.DimReduc">[.DimReduc</a>()</code>,
<code><a href="#topic++5B+5B.DimReduc">[[.DimReduc</a>()</code>,
<code><a href="#topic+dim.DimReduc">dim.DimReduc</a>()</code>,
<code><a href="#topic+merge.DimReduc">merge.DimReduc</a>()</code>,
<code><a href="#topic+print.DimReduc">print.DimReduc</a>()</code>,
<code><a href="#topic+subset.DimReduc">subset.DimReduc</a>()</code>
</p>

<hr>
<h2 id='DimReduc-validity'>Dimensional Reduction Validity</h2><span id='topic+DimReduc-validity'></span>

<h3>Description</h3>

<p>Validation of <code>DimReduc</code> objects is handled by
<code><a href="methods.html#topic+validObject">validObject</a></code>
</p>


<h3>Cell Embeddings Validation</h3>

<p>The cell embeddings matrix must be a numeric matrix of dimensions
<code class="reqn">n_{cells}</code> by <code class="reqn">d_{dimensions}</code>; row names must be the cell names
and column names must be the dimension identifier. The dimension identifier
must be &ldquo;<code>key_dimension</code>&rdquo; (eg. &ldquo;<code>PC_1</code>&rdquo;). Dimension
identifiers must be in order and cannot be skipped
</p>


<h3>Feature and Projected Feature Loadings Validation</h3>

<p>blah
</p>


<h3>Standard Deviations Validation</h3>

<p>blah
</p>


<h3>Key Validation</h3>

<p>Keys must be a one-length character vector; a key must be composed of one
of the following:
</p>

<ul>
<li><p> An empty string (eg. &ldquo;<code>''</code>&rdquo;) where <code>nzchar() == 0</code>
</p>
</li>
<li><p> An string composed of one or more alphanumeric values
(both lower- and upper-case) that ends with an underscore
(&ldquo;<code>_</code>&rdquo;); the first character must be a letter
</p>
</li></ul>

<p>Keys that are not empty strings are validated with the regex
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;
</p>


<h3>See Also</h3>

<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+CreateDimReducObject">CreateDimReducObject</a>()</code>,
<code><a href="#topic+DimReduc-class">DimReduc-class</a></code>,
<code><a href="#topic++5B.DimReduc">[.DimReduc</a>()</code>,
<code><a href="#topic++5B+5B.DimReduc">[[.DimReduc</a>()</code>,
<code><a href="#topic+dim.DimReduc">dim.DimReduc</a>()</code>,
<code><a href="#topic+merge.DimReduc">merge.DimReduc</a>()</code>,
<code><a href="#topic+print.DimReduc">print.DimReduc</a>()</code>,
<code><a href="#topic+subset.DimReduc">subset.DimReduc</a>()</code>
</p>

<hr>
<h2 id='Distances'>Get the Neighbor nearest neighbors distance matrix</h2><span id='topic+Distances'></span><span id='topic+Distances.Neighbor'></span>

<h3>Description</h3>

<p>Get the Neighbor nearest neighbors distance matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Distances(object, ...)

## S3 method for class 'Neighbor'
Distances(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Distances_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Distances_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The distance matrix
</p>

<hr>
<h2 id='droplevels.LogMap'>Drop Unused Logical Map Values</h2><span id='topic+droplevels.LogMap'></span>

<h3>Description</h3>

<p>Remove any unused values from a <a href="#topic+LogMap">logical map</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'LogMap'
droplevels(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="droplevels.LogMap_+3A_x">x</code></td>
<td>
<p>A <code>LogMap</code> object</p>
</td></tr>
<tr><td><code id="droplevels.LogMap_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with values not present in any
observation removed
</p>


<h3>See Also</h3>

<p>Logical map objects, validity, and interaction methods:
<code><a href="#topic+LogMap-validity">LogMap-validity</a></code>,
<code><a href="#topic+LogMap">LogMap</a></code>,
<code><a href="#topic+as.matrix.LogMap">as.matrix.LogMap</a>()</code>,
<code><a href="#topic+intersect.LogMap">intersect.LogMap</a>()</code>,
<code><a href="#topic+labels.LogMap">labels.LogMap</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map &lt;- LogMap(letters[1:10])
map[['obs']] &lt;- c(1, 3, 7)
map[['entry']] &lt;- c(2, 7, 10)

# Remove unused values
map &lt;- droplevels(map)
map
map[[]]

</code></pre>

<hr>
<h2 id='Embeddings'>Get Cell Embeddings</h2><span id='topic+Embeddings'></span><span id='topic+Embeddings.DimReduc'></span><span id='topic+Embeddings.Seurat'></span>

<h3>Description</h3>

<p>Get Cell Embeddings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Embeddings(object, ...)

## S3 method for class 'DimReduc'
Embeddings(object, ...)

## S3 method for class 'Seurat'
Embeddings(object, reduction = "pca", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Embeddings_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Embeddings_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Embeddings_+3A_reduction">reduction</code></td>
<td>
<p>Name of reduction to pull cell embeddings for</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The embeddings matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the embeddings directly from a DimReduc object
Embeddings(object = pbmc_small[["pca"]])[1:5, 1:5]

# Get the embeddings from a specific DimReduc in a Seurat object
Embeddings(object = pbmc_small, reduction = "pca")[1:5, 1:5]

</code></pre>

<hr>
<h2 id='EmptyDF'>Empty Data Frames</h2><span id='topic+EmptyDF'></span>

<h3>Description</h3>

<p>Create an empty <a href="base.html#topic+data.frame">data frame</a> with no row names and
zero columns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EmptyDF(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EmptyDF_+3A_n">n</code></td>
<td>
<p>Number of rows for the data frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="base.html#topic+data.frame">data frame</a> with <code>n</code> rows and
zero columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>EmptyDF(4L)

</code></pre>

<hr>
<h2 id='ExtractField'>Extract delimiter information from a string.</h2><span id='topic+ExtractField'></span>

<h3>Description</h3>

<p>Parses a string (usually a cell name) and extracts fields based
on a delimiter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractField(string, field = 1, delim = "_")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractField_+3A_string">string</code></td>
<td>
<p>String to parse.</p>
</td></tr>
<tr><td><code id="ExtractField_+3A_field">field</code></td>
<td>
<p>Integer(s) indicating which field(s) to extract. Can be a
vector multiple numbers.</p>
</td></tr>
<tr><td><code id="ExtractField_+3A_delim">delim</code></td>
<td>
<p>Delimiter to use, set to underscore by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new string, that parses out the requested fields, and
(if multiple), rejoins them with the same delimiter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ExtractField('Hello World', field = 1, delim = '_')

</code></pre>

<hr>
<h2 id='FetchData'>Access cellular data</h2><span id='topic+FetchData'></span><span id='topic+FetchData.DimReduc'></span><span id='topic+FetchData.Seurat'></span>

<h3>Description</h3>

<p>Retrieves data (feature expression, PCA scores, metrics, etc.) for a set
of cells in a Seurat object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FetchData(object, ...)

## S3 method for class 'DimReduc'
FetchData(object, vars, cells = NULL, ...)

## S3 method for class 'Seurat'
FetchData(
  object,
  vars,
  cells = NULL,
  layer = NULL,
  clean = TRUE,
  slot = deprecated(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FetchData_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="FetchData_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="FetchData_+3A_vars">vars</code></td>
<td>
<p>List of all variables to fetch, use keyword &ldquo;ident&rdquo; to
pull identity classes</p>
</td></tr>
<tr><td><code id="FetchData_+3A_cells">cells</code></td>
<td>
<p>Cells to collect data for (default is all cells)</p>
</td></tr>
<tr><td><code id="FetchData_+3A_layer">layer</code></td>
<td>
<p>Layer to pull feature data for</p>
</td></tr>
<tr><td><code id="FetchData_+3A_clean">clean</code></td>
<td>
<p>Remove cells that are missing data; choose from:
</p>

<ul>
<li> <p>&ldquo;<code>all</code>&rdquo;: consider all columns for cleaning
</p>
</li>
<li> <p>&ldquo;<code>ident</code>&rdquo;: consider all columns except the identity
class for cleaning
</p>
</li>
<li> <p>&ldquo;<code>project</code>&rdquo;: consider all columns except the identity
class for cleaning; fill missing identity values with the object's project
</p>
</li>
<li> <p>&ldquo;<code>none</code>&rdquo;: do not clean
</p>
</li></ul>

<p>Passing <code>TRUE</code> is a shortcut for &ldquo;<code>ident</code>&rdquo;; passing
<code>FALSE</code> is a shortcut for &ldquo;<code>none</code>&rdquo;</p>
</td></tr>
<tr><td><code id="FetchData_+3A_slot">slot</code></td>
<td>
<p>Deprecated in favor of <code>layer</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with cells as rows and cellular data as columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pc1 &lt;- FetchData(object = pbmc_small, vars = 'PC_1')
head(x = pc1)
head(x = FetchData(object = pbmc_small, vars = c('groups', 'ident')))

</code></pre>

<hr>
<h2 id='FilterObjects'>Find Sub-objects of a Certain Class</h2><span id='topic+FilterObjects'></span>

<h3>Description</h3>

<p>Get the names of objects within a <code>Seurat</code> object that are of a
certain class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FilterObjects(object, classes.keep = c("Assay", "StdAssay", "DimReduc"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FilterObjects_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="FilterObjects_+3A_classes.keep">classes.keep</code></td>
<td>
<p>A vector of names of classes to get</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with the names of objects within the <code>Seurat</code> object
that are of class <code>classes.keep</code>
</p>


<h3>Lifecycle</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p><code>FilterObjects</code> was deprecated in version 5.0.0; use <code><a href="#topic+.FilterObjects">.FilterObjects</a></code> instead
</p>


<h3>Examples</h3>

<pre><code class='language-R'>FilterObjects(pbmc_small)

</code></pre>

<hr>
<h2 id='FOV-class'>The Field of View Object</h2><span id='topic+FOV-class'></span><span id='topic+FOV'></span>

<h3>Description</h3>

<p>A modern container for storing coordinates of spatially-resolved single
cells. Capable of storing multiple cell segmentation boundary masks.
Supports coordinates for spatially-resolved molecule (FISH) data.
Compatible with <code><a href="#topic+SpatialImage">SpatialImage</a></code>
</p>


<h3>Slots</h3>


<dl>
<dt><code>molecules</code></dt><dd><p>A named list of
<code><a href="#topic+Molecules-class">Molecules</a></code> objects defining
spatially-resolved molecular coordinates</p>
</dd>
<dt><code>boundaries</code></dt><dd><p>A named list of
<code><a href="#topic+Segmentation-class">Segmentation</a></code> and
<code><a href="#topic+Centroids-class">Centroids</a></code> objects defining
spatially-resolved boundaries</p>
</dd>
<dt><code>assay</code></dt><dd><p>A character naming the associated assay
of the spatial coordinates</p>
</dd>
<dt><code>key</code></dt><dd><p>A one-length character vector with the object's key; keys must
be one or more alphanumeric characters followed by an underscore
&ldquo;<code>_</code>&rdquo; (regex pattern
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;)</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="#topic+FOV-methods">FOV-methods</a></code>
</p>

<hr>
<h2 id='FOV-methods'><code>FOV</code> Methods</h2><span id='topic+FOV-methods'></span><span id='topic+Cells.FOV'></span><span id='topic+Features.FOV'></span><span id='topic+FetchData.FOV'></span><span id='topic+GetTissueCoordinates.FOV'></span><span id='topic+Keys.FOV'></span><span id='topic+RenameCells.FOV'></span><span id='topic++24.FOV'></span><span id='topic++5B.FOV'></span><span id='topic++5B+5B.FOV'></span><span id='topic+length.FOV'></span><span id='topic+names.FOV'></span><span id='topic+subset.FOV'></span><span id='topic++5B+5B+3C-+2CFOV+2Ccharacter+2Cmissing+2CCentroids-method'></span><span id='topic++5B+5B+3C-+2CFOV+2Ccharacter+2Cmissing+2CMolecules-method'></span><span id='topic++5B+5B+3C-+2CFOV+2Ccharacter+2Cmissing+2CNULL-method'></span><span id='topic++5B+5B+3C-+2CFOV+2Ccharacter+2Cmissing+2CSegmentation-method'></span><span id='topic+show+2CFOV-method'></span>

<h3>Description</h3>

<p>Methods for <code><a href="#topic+FOV">FOV</a></code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FOV'
Cells(x, boundary = NULL, ...)

## S3 method for class 'FOV'
Features(x, set = NULL, ...)

## S3 method for class 'FOV'
FetchData(object, vars, cells = NULL, simplify = TRUE, ...)

## S3 method for class 'FOV'
GetTissueCoordinates(object, which = NULL, ...)

## S3 method for class 'FOV'
Keys(object, ...)

## S3 method for class 'FOV'
RenameCells(object, new.names = NULL, ...)

## S3 method for class 'FOV'
x$i, ...

## S3 method for class 'FOV'
x[i, j, ...]

## S3 method for class 'FOV'
x[[i, ...]]

## S3 method for class 'FOV'
length(x)

## S3 method for class 'FOV'
names(x)

## S3 method for class 'FOV'
subset(x, cells = NULL, features = NULL, ...)

## S4 replacement method for signature 'FOV,character,missing,Centroids'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'FOV,character,missing,Molecules'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'FOV,character,missing,'NULL''
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'FOV,character,missing,Segmentation'
x[[i, j, ...]] &lt;- value

## S4 method for signature 'FOV'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FOV-methods_+3A_x">x</code>, <code id="FOV-methods_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+FOV">FOV</a></code> object</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_boundary">boundary</code>, <code id="FOV-methods_+3A_set">set</code></td>
<td>
<p>Name of segmentation boundary or molecule set  to
extract cell or feature names for; pass <code>NA</code> to return all
cells or feature names</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_vars">vars</code></td>
<td>
<p>A vector of variables to fetch; can be the name of a
segmentation boundary, to get tissue coordinates, or molecule names,
to get molecule coordinates</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_simplify">simplify</code></td>
<td>
<p>If only returning either boundary or molecule coordinates,
return a single data frame instead of a list</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_which">which</code></td>
<td>
<p>Name of segmentation boundary or molecule set</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_new.names">new.names</code></td>
<td>
<p>vector of new cell names</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_i">i</code>, <code id="FOV-methods_+3A_cells">cells</code></td>
<td>
<p>For <code>[[</code> and <code>[[&lt;-</code>, the name of a segmentation or
&ldquo;molecules&rdquo;; for <code>FetchData</code>, <code>subset</code>. and <code>[</code>, a
vector of cells to keep</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_j">j</code>, <code id="FOV-methods_+3A_features">features</code></td>
<td>
<p>For <code>subset</code> and <code>[</code>, a vector of features to
keep; for <code>[[&lt;-</code>, not used</p>
</td></tr>
<tr><td><code id="FOV-methods_+3A_value">value</code></td>
<td>
<p>For <code>[[&lt;-</code>, a replacement
<code><a href="#topic+Molecules-class">Molecules</a></code>,
<code><a href="#topic+Centroids-class">Centroids</a></code>, or
<code><a href="#topic+Segmentation-class">Segmentation</a></code> object;
otherwise <code>NULL</code> to remove the boundary stored at <code>i</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following methods are defined for interacting with a
<code>FOV</code> object:
</p>
<p><code>Cells</code>: Get cell names
</p>
<p><code>Features</code>: Get spatially-resolved molecule names
</p>
<p><code>FetchData</code>: Fetch boundary and/or molecule coordinates from
a <code>FOV</code> object
</p>
<p><code>GetTissueCoordinates</code>: Get boundary or molecule
coordinates from a <code>FOV</code> object
</p>
<p><code>Keys</code>: Get the keys of molecule sets contained within a
<code>FOV</code> object
</p>
<p><code>RenameCells</code>: Update cell names
</p>
<p><code>$</code>, <code>[[</code>: Extract a segmentation boundary
</p>
<p><code>length</code>: Get the number of segmentation layers in a
<code>FOV</code> object
</p>
<p><code>names</code>: Get the names of segmentation layers and molecule sets
</p>
<p><code>subset</code>, <code>[</code>: Subset a <code>FOV</code> object
</p>
<p><code>[[&lt;-</code>: Add or remove segmentation layers and molecule
information to/from a <code>FOV</code> object
</p>
<p><code>show</code>: Display an object summary to stdout
</p>


<h3>Value</h3>

<p><code>Cells</code>: A vector of cell names
</p>
<p><code>Features</code>: A vector of spatially-resolved molecule names;
if no molecular information present, returns <code>NULL</code>
</p>
<p><code>FetchData</code>: If both molecule and boundary coordinates are
requested, then a two-length list:
</p>

<ul>
<li> <p>&ldquo;<code>molecules</code>&rdquo;: A data frame with the molecule coordinates
requested. If molecules requested are keyed, the keys are preserved in the
data frame
</p>
</li>
<li> <p>&ldquo;<code>coordinates</code>&rdquo;: A data frame with coordinates from the
segmentation boundaries requested
</p>
</li></ul>

<p>If <code>simplify</code> is <code>TRUE</code> and only one data frame is generated, then
only the data frame is returned. Otherwise, a one-length list is returned
with the single data frame generated
</p>
<p><code>GetTissueCoordinates</code>: ...
</p>
<p><code>Keys</code>: A named vector of molecule set keys; names are the
names of the molecule sets and values are the keys for the respective
molecule set
</p>
<p><code>RenameCells</code>: <code>object</code> with the cells renamed to
<code>new.names</code>
</p>
<p><code>$</code>, <code>[[</code>: The segmentation boundary or spatially-resolved
molecule information stored at <code>i</code>
</p>
<p><code>length</code>: The number of segmentation layers
(<code><a href="#topic+Segmentation-class">Segmentation</a></code> or
<code><a href="#topic+Centroids-class">Centroids</a></code> objects)
</p>
<p><code>names</code>: A vector of segmentation boundary and molecule set names
</p>
<p><code>subset</code>: <code>x</code> with just the cells and features specified
</p>
<p><code>[[&lt;-</code>: Varies depending on the class of <code>value</code>:
</p>

<ul>
<li><p> If <code>value</code> is <code>NULL</code>, returns <code>x</code> with the boundary
<code>i</code> removed; also allows removing <code>molecules</code>; does not allow
removing the default segmentation
</p>
</li>
<li><p> If <code>value</code> is a <code>Molecules</code>, returns <code>x</code> with
<code>value</code> stored in <code>molecules</code>; requires that <code>i</code> is
&ldquo;molecules&rdquo;
</p>
</li>
<li><p> Otherwise, stores <code>value</code> as a segmentation boundary named <code>i</code>
</p>
</li></ul>

<p><code>show</code>: Invisibly returns <code>NULL</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FOV-class">FOV-class</a></code>
</p>

<hr>
<h2 id='FOV-validity'>FOV Validity</h2><span id='topic+FOV-validity'></span>

<h3>Description</h3>

<p>Validation of <code>FOV</code> objects is handled by
<code><a href="methods.html#topic+validObject">validObject</a></code>
</p>


<h3>Boundary Validation</h3>

<p>blah
</p>


<h3>Molecule Validation</h3>

<p>blah
</p>


<h3>See Also</h3>

<p><code><a href="methods.html#topic+validObject">validObject</a></code>
</p>

<hr>
<h2 id='GetImage'>Get image data</h2><span id='topic+GetImage'></span><span id='topic+GetImage.Seurat'></span>

<h3>Description</h3>

<p>Get image data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetImage(object, mode = c("grob", "raster", "plotly", "raw"), ...)

## S3 method for class 'Seurat'
GetImage(
  object,
  mode = c("grob", "raster", "plotly", "raw"),
  image = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetImage_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="GetImage_+3A_mode">mode</code></td>
<td>
<p>How to return the image; should accept one of &ldquo;grob&rdquo;,
&ldquo;raster&rdquo;, &ldquo;plotly&rdquo;, or &ldquo;raw&rdquo;</p>
</td></tr>
<tr><td><code id="GetImage_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="GetImage_+3A_image">image</code></td>
<td>
<p>Name of <code>SpatialImage</code> object to pull image data for; if
<code>NULL</code>, will attempt to select an image automatically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Image data, varying depending on the value of <code>mode</code>:
</p>

<dl>
<dt>&ldquo;grob&rdquo;</dt><dd>
<p>An object representing image data inheriting from <code>grob</code> objects
(eg. <code>rastergrob</code>)
</p>
</dd>
<dt>&ldquo;raster&rdquo;</dt><dd><p>An object of class <code>raster</code></p>
</dd>
<dt>&ldquo;plotly&rdquo;</dt><dd>
<p>A list with image data suitable for Plotly rendering, see
<code><a href="plotly.html#topic+layout">plotly::layout</a></code> for more details
</p>
</dd>
<dt>&ldquo;raw&rdquo;</dt><dd><p>The raw image data as stored in the object</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="plotly.html#topic+layout">layout</a></code>
</p>

<hr>
<h2 id='GetTissueCoordinates'>Get tissue coordinates</h2><span id='topic+GetTissueCoordinates'></span><span id='topic+GetTissueCoordinates.Seurat'></span>

<h3>Description</h3>

<p>Get tissue coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetTissueCoordinates(object, ...)

## S3 method for class 'Seurat'
GetTissueCoordinates(object, image = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetTissueCoordinates_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="GetTissueCoordinates_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="GetTissueCoordinates_+3A_image">image</code></td>
<td>
<p>Name of <code>SpatialImage</code> object to get coordinates for; if
<code>NULL</code>, will attempt to select an image automatically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with tissue coordinates
</p>

<hr>
<h2 id='Graph-class'>The Graph Class</h2><span id='topic+Graph-class'></span><span id='topic+Graph'></span>

<h3>Description</h3>

<p>The Graph class inherits from <code><a href="Matrix.html#topic+sparseMatrix">dgCMatrix</a></code>.
We do this to enable future expandability of graphs.
</p>


<h3>Slots</h3>


<dl>
<dt><code>assay.used</code></dt><dd><p>Optional name of assay used to generate <code>Graph</code> object</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="Matrix.html#topic+dgCMatrix-class">dgCMatrix-class</a></code>
</p>
<p>Other graph: 
<code><a href="#topic+as.Graph">as.Graph</a>()</code>
</p>

<hr>
<h2 id='HVFInfo'>Highly Variable Features</h2><span id='topic+HVFInfo'></span><span id='topic+VariableFeatures'></span><span id='topic+VariableFeatures+3C-'></span><span id='topic+SVFInfo'></span><span id='topic+SpatiallyVariableFeatures'></span><span id='topic+HVFInfo.Seurat'></span><span id='topic+VariableFeatures.Seurat'></span><span id='topic+VariableFeatures+3C-.Seurat'></span><span id='topic+SVFInfo.Seurat'></span><span id='topic+SpatiallyVariableFeatures.Seurat'></span><span id='topic+HVFInfo.Assay'></span><span id='topic+SpatiallyVariableFeatures.Assay'></span><span id='topic+SVFInfo.Assay'></span><span id='topic+VariableFeatures.Assay'></span><span id='topic+VariableFeatures+3C-.Assay'></span><span id='topic+HVFInfo.Assay5'></span><span id='topic+VariableFeatures.Assay5'></span><span id='topic+VariableFeatures+3C-.Assay5'></span>

<h3>Description</h3>

<p>Get and set variable feature information for an <code><a href="#topic+Assay">Assay</a></code> object.
<code>HVFInfo</code> and <code>VariableFeatures</code> utilize generally variable
features, while <code>SVFInfo</code> and <code>SpatiallyVariableFeatures</code> are
restricted to spatially variable features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HVFInfo(object, method, status = FALSE, ...)

VariableFeatures(object, method = NULL, ...)

VariableFeatures(object, ...) &lt;- value

SVFInfo(object, method, status, ...)

SpatiallyVariableFeatures(object, method, ...)

## S3 method for class 'Seurat'
HVFInfo(
  object,
  method = NULL,
  status = FALSE,
  assay = NULL,
  selection.method = deprecated(),
  ...
)

## S3 method for class 'Seurat'
VariableFeatures(
  object,
  method = NULL,
  assay = NULL,
  nfeatures = NULL,
  layer = NA,
  simplify = TRUE,
  selection.method = deprecated(),
  ...
)

## S3 replacement method for class 'Seurat'
VariableFeatures(object, assay = NULL, ...) &lt;- value

## S3 method for class 'Seurat'
SVFInfo(
  object,
  method = c("markvariogram", "moransi"),
  status = FALSE,
  assay = NULL,
  selection.method = deprecated(),
  ...
)

## S3 method for class 'Seurat'
SpatiallyVariableFeatures(
  object,
  method = "moransi",
  assay = NULL,
  decreasing = TRUE,
  selection.method = deprecated(),
  ...
)

## S3 method for class 'Assay'
HVFInfo(object, method, status = FALSE, selection.method = deprecated(), ...)

## S3 method for class 'Assay'
SpatiallyVariableFeatures(
  object,
  method = "moransi",
  decreasing = TRUE,
  selection.method = deprecated(),
  ...
)

## S3 method for class 'Assay'
SVFInfo(
  object,
  method = c("markvariogram", "moransi"),
  status = FALSE,
  selection.method = deprecated(),
  ...
)

## S3 method for class 'Assay'
VariableFeatures(object, method = NULL, selection.method = deprecated(), ...)

## S3 replacement method for class 'Assay'
VariableFeatures(object, ...) &lt;- value

## S3 method for class 'Assay5'
HVFInfo(object, method = NULL, status = FALSE, layer = NULL, strip = TRUE, ...)

## S3 method for class 'Assay5'
VariableFeatures(
  object,
  method = NULL,
  layer = NA,
  simplify = TRUE,
  nfeatures = Inf,
  selection.method = deprecated(),
  ...
)

## S3 replacement method for class 'Assay5'
VariableFeatures(object, method = "custom", layer = NULL, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HVFInfo_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_method">method</code></td>
<td>
<p>Which method to pull. For <code>HVFInfo</code> and
<code>VariableFeatures</code>, choose one from one of the
following:
</p>

<ul>
<li> <p>&ldquo;vst&rdquo;
</p>
</li>
<li> <p>&ldquo;sctransform&rdquo; or &ldquo;sct&rdquo;
</p>
</li>
<li> <p>&ldquo;mean.var.plot&rdquo;, &ldquo;dispersion&rdquo;, &ldquo;mvp&rdquo;, or
&ldquo;disp&rdquo;
</p>
</li></ul>

<p>For <code>SVFInfo</code> and <code>SpatiallyVariableFeatures</code>, choose from:
</p>

<ul>
<li> <p>&ldquo;markvariogram&rdquo;
</p>
</li>
<li> <p>&ldquo;moransi&rdquo;
</p>
</li></ul>
</td></tr>
<tr><td><code id="HVFInfo_+3A_status">status</code></td>
<td>
<p>Add variable status to the resulting data frame</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_value">value</code></td>
<td>
<p>A character vector of variable features</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_assay">assay</code></td>
<td>
<p>Name of assay to pull highly variable feature information for</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_selection.method">selection.method</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a></p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_nfeatures">nfeatures</code></td>
<td>
<p>Maximum number of features to select when simplifying</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_layer">layer</code></td>
<td>
<p>Layer to pull variable features for</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_simplify">simplify</code></td>
<td>
<p>When pulling for multiple layers, combine into a single
vector and select a common set of variable features for all layers</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_decreasing">decreasing</code></td>
<td>
<p>Return features in decreasing order (most spatially
variable first).</p>
</td></tr>
<tr><td><code id="HVFInfo_+3A_strip">strip</code></td>
<td>
<p>Remove method/layer identifiers from highly variable data frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>HVFInfo</code>: A data frame with feature means, dispersion, and
scaled dispersion
</p>
<p><code>VariableFeatures</code>: a vector of the variable features
</p>
<p><code>SVFInfo</code>: a data frame with the spatially variable features
</p>
<p><code>SpatiallyVariableFeatures</code>: a character vector of the spatially
variable features
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the HVF info from a specific Assay in a Seurat object
HVFInfo(object = pbmc_small, assay = "RNA")[1:5, ]

# Get the HVF info directly from an Assay object
HVFInfo(pbmc_small[["RNA"]], method = 'vst')[1:5, ]

</code></pre>

<hr>
<h2 id='Idents'>Get, set, and manipulate an object's identity classes</h2><span id='topic+Idents'></span><span id='topic+Idents+3C-'></span><span id='topic+RenameIdents'></span><span id='topic+RenameIdent'></span><span id='topic+ReorderIdent'></span><span id='topic+SetIdent'></span><span id='topic+StashIdent'></span><span id='topic+Idents.Seurat'></span><span id='topic+Idents+3C-.Seurat'></span><span id='topic+ReorderIdent.Seurat'></span><span id='topic+RenameIdents.Seurat'></span><span id='topic+SetIdent.Seurat'></span><span id='topic+StashIdent.Seurat'></span><span id='topic+droplevels.Seurat'></span><span id='topic+levels.Seurat'></span><span id='topic+levels+3C-.Seurat'></span>

<h3>Description</h3>

<p>Get, set, and manipulate an object's identity classes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Idents(object, ...)

Idents(object, ...) &lt;- value

RenameIdents(object, ...)

ReorderIdent(object, var, ...)

SetIdent(object, ...)

StashIdent(object, save.name, ...)

## S3 method for class 'Seurat'
Idents(object, ...)

## S3 replacement method for class 'Seurat'
Idents(object, cells = NULL, drop = FALSE, replace = FALSE, ...) &lt;- value

## S3 method for class 'Seurat'
ReorderIdent(
  object,
  var,
  reverse = FALSE,
  afxn = mean,
  reorder.numeric = FALSE,
  ...
)

## S3 method for class 'Seurat'
RenameIdents(object, ...)

## S3 method for class 'Seurat'
SetIdent(object, cells = NULL, value, ...)

## S3 method for class 'Seurat'
StashIdent(object, save.name = "orig.ident", ...)

## S3 method for class 'Seurat'
droplevels(x, ...)

## S3 method for class 'Seurat'
levels(x)

## S3 replacement method for class 'Seurat'
levels(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Idents_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods; for <code>RenameIdents</code>: named
arguments as <code>old.ident = new.ident</code>; for <code>ReorderIdent</code>: arguments
passed on to <code><a href="#topic+FetchData">FetchData</a></code></p>
</td></tr>
<tr><td><code id="Idents_+3A_value">value</code></td>
<td>
<p>The name of the identities to pull from object metadata or the
identities themselves</p>
</td></tr>
<tr><td><code id="Idents_+3A_var">var</code></td>
<td>
<p>Feature or variable to order on</p>
</td></tr>
<tr><td><code id="Idents_+3A_save.name">save.name</code></td>
<td>
<p>Store current identity information under this name</p>
</td></tr>
<tr><td><code id="Idents_+3A_cells">cells</code></td>
<td>
<p>Set cell identities for specific cells</p>
</td></tr>
<tr><td><code id="Idents_+3A_drop">drop</code></td>
<td>
<p>Drop unused levels</p>
</td></tr>
<tr><td><code id="Idents_+3A_replace">replace</code></td>
<td>
<p>Replace identities for unset cells with <code>NA</code></p>
</td></tr>
<tr><td><code id="Idents_+3A_reverse">reverse</code></td>
<td>
<p>Reverse ordering</p>
</td></tr>
<tr><td><code id="Idents_+3A_afxn">afxn</code></td>
<td>
<p>Function to evaluate each identity class based on; default is
<code><a href="base.html#topic+mean">mean</a></code></p>
</td></tr>
<tr><td><code id="Idents_+3A_reorder.numeric">reorder.numeric</code></td>
<td>
<p>Rename all identity classes to be increasing numbers
starting from 1 (default is FALSE)</p>
</td></tr>
<tr><td><code id="Idents_+3A_x">x</code>, <code id="Idents_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>Idents</code>: The cell identities
</p>
<p><code>Idents&lt;-</code>: <code>object</code> with the cell identities changed
</p>
<p><code>RenameIdents</code>: An object with selected identity classes renamed
</p>
<p><code>ReorderIdent</code>: An object with
</p>
<p><code>SetIdent</code>: An object with new identity classes set
</p>
<p><code>StashIdent</code>: An object with the identities stashed
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get cell identity classes
Idents(pbmc_small)

# Set cell identity classes
# Can be used to set identities for specific cells to a new level
Idents(pbmc_small, cells = 1:4) &lt;- 'a'
head(Idents(pbmc_small))

# Can also set idents from a value in object metadata
colnames(pbmc_small[[]])
Idents(pbmc_small) &lt;- 'RNA_snn_res.1'
levels(pbmc_small)

# Rename cell identity classes
# Can provide an arbitrary amount of idents to rename
levels(pbmc_small)
pbmc_small &lt;- RenameIdents(pbmc_small, '0' = 'A', '2' = 'C')
levels(pbmc_small)

## Not run: 
head(Idents(pbmc_small))
pbmc_small &lt;- ReorderIdent(pbmc_small, var = 'PC_1')
head(Idents(pbmc_small))

## End(Not run)

# Set cell identity classes using SetIdent
cells.use &lt;- WhichCells(pbmc_small, idents = '1')
pbmc_small &lt;- SetIdent(pbmc_small, cells = cells.use, value = 'B')

head(pbmc_small[[]])
pbmc_small &lt;- StashIdent(pbmc_small, save.name = 'idents')
head(pbmc_small[[]])

# Get the levels of identity classes of a Seurat object
levels(x = pbmc_small)

# Reorder identity classes
levels(x = pbmc_small)
levels(x = pbmc_small) &lt;- c('C', 'A', 'B')
levels(x = pbmc_small)

</code></pre>

<hr>
<h2 id='Images'>Pull spatial image names</h2><span id='topic+Images'></span>

<h3>Description</h3>

<p>List the names of <code>SpatialImage</code> objects present in a <code>Seurat</code>
object. If <code>assay</code> is provided, limits search to images associated with
that assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Images(object, assay = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Images_+3A_object">object</code></td>
<td>
<p>A <code>Seurat</code> object</p>
</td></tr>
<tr><td><code id="Images_+3A_assay">assay</code></td>
<td>
<p>Name of assay to limit search to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of image names
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Images(object)

## End(Not run)

</code></pre>

<hr>
<h2 id='Index'>Get Neighbor algorithm index</h2><span id='topic+Index'></span><span id='topic+Index+3C-'></span><span id='topic+Index.Neighbor'></span><span id='topic+Index+3C-.Neighbor'></span>

<h3>Description</h3>

<p>Get Neighbor algorithm index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Index(object, ...)

Index(object, ...) &lt;- value

## S3 method for class 'Neighbor'
Index(object, ...)

## S3 replacement method for class 'Neighbor'
Index(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Index_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Index_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Index_+3A_value">value</code></td>
<td>
<p>The index to store</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the value in the alg.idx slot of the Neighbor object
</p>
<p><code>Idents&lt;-</code>: A Neighbor object with the index stored
</p>

<hr>
<h2 id='Indices'>Get Neighbor nearest neighbor index matrices</h2><span id='topic+Indices'></span><span id='topic+Indices.Neighbor'></span>

<h3>Description</h3>

<p>Get Neighbor nearest neighbor index matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Indices(object, ...)

## S3 method for class 'Neighbor'
Indices(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Indices_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Indices_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with the nearest neighbor indices
</p>

<hr>
<h2 id='intersect.LogMap'>Find Common Logical Map Values</h2><span id='topic+intersect.LogMap'></span>

<h3>Description</h3>

<p>Identify values in a <a href="#topic+LogMap">logical map</a> that are
common to every observation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'LogMap'
intersect(x, y = missing_arg(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intersect.LogMap_+3A_x">x</code></td>
<td>
<p>A <code>LogMap</code> object</p>
</td></tr>
<tr><td><code id="intersect.LogMap_+3A_y">y</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="intersect.LogMap_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The values of <code>x</code> that are present in <strong>every</strong> observation
</p>


<h3>See Also</h3>

<p>Logical map objects, validity, and interaction methods:
<code><a href="#topic+LogMap-validity">LogMap-validity</a></code>,
<code><a href="#topic+LogMap">LogMap</a></code>,
<code><a href="#topic+as.matrix.LogMap">as.matrix.LogMap</a>()</code>,
<code><a href="#topic+droplevels.LogMap">droplevels.LogMap</a>()</code>,
<code><a href="#topic+labels.LogMap">labels.LogMap</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map &lt;- LogMap(letters[1:10])
map[['obs']] &lt;- c(1, 3, 7)
map[['entry']] &lt;- c(2, 7, 10)

# Identify values that are present in every observation
intersect(map)

</code></pre>

<hr>
<h2 id='IsGlobal'>Is an object global/persistent?</h2><span id='topic+IsGlobal'></span><span id='topic+IsGlobal.default'></span><span id='topic+IsGlobal.DimReduc'></span>

<h3>Description</h3>

<p>Typically, when removing <code>Assay</code> objects from an <code>Seurat</code> object,
all associated objects (eg. <code>DimReduc</code>, <code>Graph</code>, and
<code>SeuratCommand</code> objects)
are removed as well. If an associated object is marked as global/persistent,
the associated object will remain even if its original assay was deleted
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsGlobal(object, ...)

## Default S3 method:
IsGlobal(object, ...)

## S3 method for class 'DimReduc'
IsGlobal(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsGlobal_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="IsGlobal_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>TRUE</code> if the object is global/persistent otherwise <code>FALSE</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>IsGlobal(pbmc_small[['pca']])

</code></pre>

<hr>
<h2 id='IsMatrixEmpty'>Check if a matrix is empty</h2><span id='topic+IsMatrixEmpty'></span><span id='topic+IsMatrixEmpty.default'></span>

<h3>Description</h3>

<p>Takes a matrix and asks if it's empty (either 0x0 or 1x1 with a value of NA)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsMatrixEmpty(x)

## Default S3 method:
IsMatrixEmpty(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsMatrixEmpty_+3A_x">x</code></td>
<td>
<p>A matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Whether or not <code>x</code> is empty
</p>


<h3>Examples</h3>

<pre><code class='language-R'>IsMatrixEmpty(new("matrix"))
IsMatrixEmpty(matrix())
IsMatrixEmpty(matrix(1:3))

</code></pre>

<hr>
<h2 id='IsNamedList'>Check List Names</h2><span id='topic+IsNamedList'></span>

<h3>Description</h3>

<p>Check to see if a list has names; also check to enforce that all names are
present and unique
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsNamedList(x, all.unique = TRUE, allow.empty = FALSE, pass.zero = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsNamedList_+3A_x">x</code></td>
<td>
<p>A list</p>
</td></tr>
<tr><td><code id="IsNamedList_+3A_all.unique">all.unique</code></td>
<td>
<p>Require that all names are unique from one another</p>
</td></tr>
<tr><td><code id="IsNamedList_+3A_allow.empty">allow.empty</code></td>
<td>
<p>Allow empty (<code>nchar = 0</code>) names</p>
</td></tr>
<tr><td><code id="IsNamedList_+3A_pass.zero">pass.zero</code></td>
<td>
<p>Pass on zero-length lists</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>TRUE</code> if ..., otherwise <code>FALSE</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>IsNamedList(list())
IsNamedList(list(), pass.zero = TRUE)
IsNamedList(list(1, 2, 3))
IsNamedList(list(a = 1, b = 2, c = 3))
IsNamedList(list(a = 1, 2, c = 3))
IsNamedList(list(a = 1, 2, c = 3), allow.empty = TRUE)
IsNamedList(list(a = 1, a = 2, a = 3))
IsNamedList(list(a = 1, a = 2, a = 3), all.unique = FALSE)

</code></pre>

<hr>
<h2 id='IsSparse'>Is a Matrix Sparse</h2><span id='topic+IsSparse'></span>

<h3>Description</h3>

<p>Is a Matrix Sparse
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsSparse(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsSparse_+3A_x">x</code></td>
<td>
<p>A matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>...
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.SparseSlots">.SparseSlots</a>()</code>,
<code><a href="#topic+RegisterSparseMatrix">RegisterSparseMatrix</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>IsSparse(matrix())
IsSparse(LayerData(pbmc_small, "counts"))

</code></pre>

<hr>
<h2 id='JackStrawData-class'>The JackStrawData Class</h2><span id='topic+JackStrawData-class'></span><span id='topic+JackStrawData'></span>

<h3>Description</h3>

<p>The JackStrawData is used to store the results of a JackStraw computation.
</p>


<h3>Slots</h3>


<dl>
<dt><code>empirical.p.values</code></dt><dd><p>Empirical p-values</p>
</dd>
<dt><code>fake.reduction.scores</code></dt><dd><p>Fake reduction scores</p>
</dd>
<dt><code>empirical.p.values.full</code></dt><dd><p>Empirical p-values on full</p>
</dd>
<dt><code>overall.p.values</code></dt><dd><p>Overall p-values from ScoreJackStraw</p>
</dd>
</dl>

<hr>
<h2 id='JackStrawData-methods'><code>JackStrawData</code> Methods</h2><span id='topic+JackStrawData-methods'></span><span id='topic+.DollarNames.JackStrawData'></span><span id='topic++24.JackStrawData'></span><span id='topic+as.logical.JackStrawData'></span><span id='topic+show+2CJackStrawData-method'></span>

<h3>Description</h3>

<p>Methods for <code><a href="#topic+JackStrawData">JackStrawData</a></code> objects for generics defined in
other packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'JackStrawData'
.DollarNames(x, pattern = "")

## S3 method for class 'JackStrawData'
x$i, ...

## S3 method for class 'JackStrawData'
as.logical(x, ...)

## S4 method for signature 'JackStrawData'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JackStrawData-methods_+3A_x">x</code>, <code id="JackStrawData-methods_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+JackStrawData">JackStrawData</a></code> object</p>
</td></tr>
<tr><td><code id="JackStrawData-methods_+3A_pattern">pattern</code></td>
<td>
<p> A regular expression.  Only matching names are
returned.
</p>
</td></tr>
<tr><td><code id="JackStrawData-methods_+3A_i">i</code></td>
<td>
<p>A <code>JackStrawData</code> slot name</p>
</td></tr>
<tr><td><code id="JackStrawData-methods_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>$</code>: Slot <code>i</code> from <code>x</code>
</p>
<p><code>as.logical</code>: <code>TRUE</code> if empirical p-values have been
calculated otherwise <code>FALSE</code>
</p>
<p><code>show</code>: Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and
invisibly returns <code>NULL</code>
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>.DollarNames(JackStrawData)</code>: Autocompletion for <code>$</code> access on a
<code>JackStrawData</code> object
</p>
</li>
<li> <p><code>$</code>: Access data from a <code>JackStrawData</code>
object
</p>
</li>
<li> <p><code>as.logical(JackStrawData)</code>: Have empirical p-values for a
<code>JackStrawData</code> object been calculated
</p>
</li>
<li> <p><code>show(JackStrawData)</code>: Overview of a <code>JackStrawData</code> object
</p>
</li></ul>

<hr>
<h2 id='JoinLayers'>Split and Join Layers Together</h2><span id='topic+JoinLayers'></span><span id='topic+JoinLayers.Assay5'></span><span id='topic+JoinLayers.Seurat'></span>

<h3>Description</h3>

<p>Split and Join Layers Together
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JoinLayers(object, ...)

## S3 method for class 'Assay5'
JoinLayers(object, layers = NULL, new = NULL, ...)

## S3 method for class 'Seurat'
JoinLayers(object, assay = NULL, layers = NULL, new = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JoinLayers_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="JoinLayers_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="JoinLayers_+3A_layers">layers</code></td>
<td>
<p>Names of layers to split or join</p>
</td></tr>
<tr><td><code id="JoinLayers_+3A_new">new</code></td>
<td>
<p>Name of new layers</p>
</td></tr>
<tr><td><code id="JoinLayers_+3A_assay">assay</code></td>
<td>
<p>Name of assay to split layers</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>object</code> with the layers specified joined
</p>

<hr>
<h2 id='JS'>Get and set JackStraw information</h2><span id='topic+JS'></span><span id='topic+JS+3C-'></span><span id='topic+JS.JackStrawData'></span><span id='topic+JS+3C-.JackStrawData'></span><span id='topic+JS.DimReduc'></span><span id='topic+JS+3C-.DimReduc'></span>

<h3>Description</h3>

<p>Get and set JackStraw information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JS(object, ...)

JS(object, ...) &lt;- value

## S3 method for class 'JackStrawData'
JS(object, slot, ...)

## S3 replacement method for class 'JackStrawData'
JS(object, slot, ...) &lt;- value

## S3 method for class 'DimReduc'
JS(object, slot = NULL, ...)

## S3 replacement method for class 'DimReduc'
JS(object, slot = NULL, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JS_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="JS_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="JS_+3A_value">value</code></td>
<td>
<p>JackStraw information</p>
</td></tr>
<tr><td><code id="JS_+3A_slot">slot</code></td>
<td>
<p>Name of slot to store JackStraw scores to
Can shorten to 'empirical', 'fake', 'full', or 'overall'</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>JS</code>: either a <code><a href="#topic+JackStrawData">JackStrawData</a></code> object or the
specified jackstraw data
</p>
<p><code>JS&lt;-</code>: <code>object</code> with the update jackstraw information
</p>

<hr>
<h2 id='Key'>Get and set object keys</h2><span id='topic+Key'></span><span id='topic+Keys'></span><span id='topic+Key+3C-'></span><span id='topic+Key.Assay'></span><span id='topic+Key+3C-.Assay'></span><span id='topic+Key.Assay5'></span><span id='topic+Key+3C-.Assay5'></span><span id='topic+Key.DimReduc'></span><span id='topic+Key+3C-.DimReduc'></span><span id='topic+Key.Seurat'></span><span id='topic+Keys.Seurat'></span>

<h3>Description</h3>

<p>Get and set object keys
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Key(object, ...)

Keys(object, ...)

Key(object, ...) &lt;- value

## S3 method for class 'Assay'
Key(object, ...)

## S3 replacement method for class 'Assay'
Key(object, ...) &lt;- value

## S3 method for class 'Assay5'
Key(object, ...)

## S3 replacement method for class 'Assay5'
Key(object, ...) &lt;- value

## S3 method for class 'DimReduc'
Key(object, ...)

## S3 replacement method for class 'DimReduc'
Key(object, ...) &lt;- value

## S3 method for class 'Seurat'
Key(object, ...)

## S3 method for class 'Seurat'
Keys(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Key_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Key_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Key_+3A_value">value</code></td>
<td>
<p>Key value</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>Key</code>: the object key
</p>
<p><code>Keys</code>: a named vector of keys of sub-objects
</p>
<p><code>Key&lt;-</code>: <code>object</code> with an updated key
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get an Assay key
Key(pbmc_small[["RNA"]])

# Set the key for an Assay
Key(pbmc_small[["RNA"]]) &lt;- "newkey_"
Key(pbmc_small[["RNA"]])

# Get a DimReduc key
Key(object = pbmc_small[["pca"]])

# Set the key for DimReduc
Key(object = pbmc_small[["pca"]]) &lt;- "newkey2_"
Key(object = pbmc_small[["pca"]])

# Show all keys associated with a Seurat object
Key(object = pbmc_small)
Keys(object = pbmc_small)

</code></pre>

<hr>
<h2 id='Key-validity'>Key Validity</h2><span id='topic+Key-validity'></span>

<h3>Description</h3>

<p>Validation of <code><a href="#topic+KeyMixin">KeyMixin</a></code> objects is handled by
<code><a href="methods.html#topic+validObject">validObject</a></code>
</p>


<h3>Key Validation</h3>

<p>Keys must be a one-length character vector; a key must be composed of one
of the following:
</p>

<ul>
<li><p> An empty string (eg. &ldquo;<code>''</code>&rdquo;) where <code>nzchar() == 0</code>
</p>
</li>
<li><p> An string composed of one or more alphanumeric values
(both lower- and upper-case) that ends with an underscore
(&ldquo;<code>_</code>&rdquo;); the first character must be a letter
</p>
</li></ul>

<p>Keys that are not empty strings are validated with the regex
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.KeyPattern">.KeyPattern</a>()</code>,
<code><a href="#topic+.RandomKey">.RandomKey</a>()</code>,
<code><a href="#topic+KeyMixin-class">KeyMixin-class</a></code>
</p>

<hr>
<h2 id='KeyMixin-class'>A Mixin for Keyed objects</h2><span id='topic+KeyMixin-class'></span><span id='topic+KeyMixin'></span><span id='topic+Key.character'></span><span id='topic+Key.KeyMixin'></span><span id='topic+Key+3C-.KeyMixin'></span>

<h3>Description</h3>

<p>A mixin (virtual class) for enabling keyed objects; provides consistent
behavior for getting, setting, and validating keys
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'character'
Key(object, quiet = FALSE, ...)

## S3 method for class 'KeyMixin'
Key(object, ...)

## S3 replacement method for class 'KeyMixin'
Key(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KeyMixin-class_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="KeyMixin-class_+3A_quiet">quiet</code></td>
<td>
<p>Suppress warnings when updating characters to keys</p>
</td></tr>
<tr><td><code id="KeyMixin-class_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="KeyMixin-class_+3A_value">value</code></td>
<td>
<p>A key to set</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Key.character</code>: Update a character to a key
</p>
<p><code>Key.KeyMixin</code>: Get the key of a keyed object
</p>
<p><code>Key&lt;-</code>: Set the key of a keyed object
</p>


<h3>Value</h3>

<p><code>Key.character</code>: <code>object</code> but as a syntactically-valid key
</p>
<p><code>Key.KeyMixin</code>: The key from <code>object</code>; if no key set,
returns <code>NULL</code>
</p>
<p><code>Key&lt;-</code>: <code>object</code> with the key set to <code>value</code>
</p>


<h3>Slots</h3>


<dl>
<dt><code>key</code></dt><dd><p>A one-length character vector with the object's key; keys must
be one or more alphanumeric characters followed by an underscore
&ldquo;<code>_</code>&rdquo; (regex pattern
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;)</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="#topic+.KeyPattern">.KeyPattern</a>()</code>,
<code><a href="#topic+.RandomKey">.RandomKey</a>()</code>,
<code><a href="#topic+Key-validity">Key-validity</a></code>
</p>

<hr>
<h2 id='labels.LogMap'>Find Observations by Value</h2><span id='topic+labels.LogMap'></span>

<h3>Description</h3>

<p>Identify the observations that contain a specific
value in a <a href="#topic+LogMap">logical map</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'LogMap'
labels(
  object,
  values,
  select = c("first", "last", "common", "all"),
  simplify = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="labels.LogMap_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+LogMap">LogMap</a></code> object</p>
</td></tr>
<tr><td><code id="labels.LogMap_+3A_values">values</code></td>
<td>
<p>A vector of values to find observations for</p>
</td></tr>
<tr><td><code id="labels.LogMap_+3A_select">select</code></td>
<td>
<p>Observation selection method; choose from:
</p>

<ul>
<li> <p>&ldquo;<code>first</code>&rdquo;: the first observation the value is found in
</p>
</li>
<li> <p>&ldquo;<code>last</code>&rdquo;: the last observation the value is found in
</p>
</li>
<li> <p>&ldquo;<code>common</code>&rdquo;: the first most-common observation the value
is found in; most-common is determined by the observation that contains
the most of the values requested
</p>
</li>
<li> <p>&ldquo;<code>all</code>&rdquo;: all observations the value is found in
</p>
</li></ul>
</td></tr>
<tr><td><code id="labels.LogMap_+3A_simplify">simplify</code></td>
<td>
<p>Simplify the resulting list to a vector</p>
</td></tr>
<tr><td><code id="labels.LogMap_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>labels</code>: A list, or vector if <code>simplify</code> is <code>TRUE</code>,
of all values and the observations they're found in, according
to the value of <code>select</code>
</p>


<h3>See Also</h3>

<p>Logical map objects, validity, and interaction methods:
<code><a href="#topic+LogMap-validity">LogMap-validity</a></code>,
<code><a href="#topic+LogMap">LogMap</a></code>,
<code><a href="#topic+as.matrix.LogMap">as.matrix.LogMap</a>()</code>,
<code><a href="#topic+droplevels.LogMap">droplevels.LogMap</a>()</code>,
<code><a href="#topic+intersect.LogMap">intersect.LogMap</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map &lt;- LogMap(letters[1:10])
map[['obs']] &lt;- c(1, 3, 7)
map[['entry']] &lt;- c(2, 7, 10)

# Find observations for a set of values
labels(map, c('a', 'b', 'g'))

</code></pre>

<hr>
<h2 id='LayerData'>Query and Manipulate Assay Layers</h2><span id='topic+LayerData'></span><span id='topic+LayerData+3C-'></span><span id='topic+Layers'></span><span id='topic+LayerData.Assay'></span><span id='topic+LayerData+3C-.Assay'></span><span id='topic+Layers.Assay'></span><span id='topic+LayerData.Assay5'></span><span id='topic+LayerData+3C-.Assay5'></span><span id='topic+Layers.Assay5'></span><span id='topic+LayerData.Seurat'></span><span id='topic+LayerData+3C-.Seurat'></span><span id='topic+Layers.Seurat'></span>

<h3>Description</h3>

<p>Query and Manipulate Assay Layers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LayerData(object, layer, ...)

LayerData(object, layer, ...) &lt;- value

Layers(object, ...)

## S3 method for class 'Assay'
LayerData(
  object,
  layer = NULL,
  cells = NULL,
  features = NULL,
  slot = deprecated(),
  ...
)

## S3 replacement method for class 'Assay'
LayerData(object, layer, ...) &lt;- value

## S3 method for class 'Assay'
Layers(object, search = NA, ...)

## S3 method for class 'Assay5'
LayerData(
  object,
  layer = NULL,
  cells = NULL,
  features = NULL,
  fast = FALSE,
  slot = deprecated(),
  ...
)

## S3 replacement method for class 'Assay5'
LayerData(object, layer, features = NULL, cells = NULL, ...) &lt;- value

## S3 method for class 'Assay5'
Layers(object, search = NA, ...)

## S3 method for class 'Seurat'
LayerData(object, layer = NULL, assay = NULL, slot = deprecated(), ...)

## S3 replacement method for class 'Seurat'
LayerData(object, layer, assay = NULL, ...) &lt;- value

## S3 method for class 'Seurat'
Layers(object, search = NA, assay = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LayerData_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="LayerData_+3A_layer">layer</code></td>
<td>
<p>Name of layer to fetch or set</p>
</td></tr>
<tr><td><code id="LayerData_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="LayerData_+3A_value">value</code></td>
<td>
<p>New two-dimensional data to be added as a layer</p>
</td></tr>
<tr><td><code id="LayerData_+3A_features">features</code>, <code id="LayerData_+3A_cells">cells</code></td>
<td>
<p>Vectors of features/cells to include</p>
</td></tr>
<tr><td><code id="LayerData_+3A_slot">slot</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a></p>
</td></tr>
<tr><td><code id="LayerData_+3A_search">search</code></td>
<td>
<p>A pattern to search layer names for; pass one of:
</p>

<ul>
<li> <p>&ldquo;<code>NA</code>&rdquo; to pull all layers
</p>
</li>
<li> <p>&ldquo;<code>NULL</code>&rdquo; to pull the default layer(s)
</p>
</li>
<li><p> a <a href="base.html#topic+grep">regular expression</a> that matches layer names
</p>
</li></ul>
</td></tr>
<tr><td><code id="LayerData_+3A_fast">fast</code></td>
<td>
<p>Determine how to return the layer data; choose from:
</p>

<dl>
<dt><code>FALSE</code></dt><dd><p>Apply any transpositions and attempt to add
feature/cell names (if supported) back to the layer data</p>
</dd>
<dt><code>NA</code></dt><dd><p>Attempt to add feature/cell names back to the layer data,
skip any transpositions</p>
</dd>
<dt><code>TRUE</code></dt><dd><p>Do not apply any transpositions or add feature/cell
names to the layer data</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="LayerData_+3A_assay">assay</code></td>
<td>
<p>Name of assay to fetch layer data from or assign layer data to</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>LayerData</code>: the layer data for <code>layer</code> from <code>object</code>
</p>
<p><code>Layer&lt;-</code>: <code>object</code> with <code>value</code> added as a layer
named <code>layer</code>
</p>
<p><code>Layers</code>: the names of the layers present in <code>object</code>
</p>

<hr>
<h2 id='Layers-StdAssay'>Query and Manipulate Assay Layers</h2><span id='topic+Layers-StdAssay'></span><span id='topic+LayerData.StdAssay'></span><span id='topic+LayerData+3C-.StdAssay'></span><span id='topic+Layers.StdAssay'></span>

<h3>Description</h3>

<p>Query and Manipulate Assay Layers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
LayerData(
  object,
  layer = NULL,
  cells = NULL,
  features = NULL,
  fast = FALSE,
  slot = deprecated(),
  ...
)

## S3 replacement method for class 'StdAssay'
LayerData(object, layer, features = NULL, cells = NULL, ...) &lt;- value

## S3 method for class 'StdAssay'
Layers(object, search = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Layers-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Layers-StdAssay_+3A_layer">layer</code></td>
<td>
<p>Name of layer to fetch or set</p>
</td></tr>
<tr><td><code id="Layers-StdAssay_+3A_features">features</code>, <code id="Layers-StdAssay_+3A_cells">cells</code></td>
<td>
<p>Vectors of features/cells to include</p>
</td></tr>
<tr><td><code id="Layers-StdAssay_+3A_fast">fast</code></td>
<td>
<p>Determine how to return the layer data; choose from:
</p>

<dl>
<dt><code>FALSE</code></dt><dd><p>Apply any transpositions and attempt to add
feature/cell names (if supported) back to the layer data</p>
</dd>
<dt><code>NA</code></dt><dd><p>Attempt to add feature/cell names back to the layer data,
skip any transpositions</p>
</dd>
<dt><code>TRUE</code></dt><dd><p>Do not apply any transpositions or add feature/cell
names to the layer data</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="Layers-StdAssay_+3A_slot">slot</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a></p>
</td></tr>
<tr><td><code id="Layers-StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Layers-StdAssay_+3A_value">value</code></td>
<td>
<p>New two-dimensional data to be added as a layer</p>
</td></tr>
<tr><td><code id="Layers-StdAssay_+3A_search">search</code></td>
<td>
<p>A pattern to search layer names for; pass one of:
</p>

<ul>
<li> <p>&ldquo;<code>NA</code>&rdquo; to pull all layers
</p>
</li>
<li> <p>&ldquo;<code>NULL</code>&rdquo; to pull the default layer(s)
</p>
</li>
<li><p> a <a href="base.html#topic+grep">regular expression</a> that matches layer names
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p><code>LayerData</code>: the layer data for <code>layer</code> from <code>object</code>
</p>
<p><code>Layer&lt;-</code>: <code>object</code> with <code>value</code> added as a layer
named <code>layer</code>
</p>
<p><code>Layers</code>: the names of the layers present in <code>object</code>
</p>

<hr>
<h2 id='Loadings'>Get and set feature loadings</h2><span id='topic+Loadings'></span><span id='topic+Loadings+3C-'></span><span id='topic+Loadings.DimReduc'></span><span id='topic+Loadings+3C-.DimReduc'></span><span id='topic+Loadings.Seurat'></span>

<h3>Description</h3>

<p>Get and set feature loadings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Loadings(object, ...)

Loadings(object, ...) &lt;- value

## S3 method for class 'DimReduc'
Loadings(object, projected = FALSE, ...)

## S3 replacement method for class 'DimReduc'
Loadings(object, projected = TRUE, ...) &lt;- value

## S3 method for class 'Seurat'
Loadings(object, reduction = "pca", projected = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Loadings_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Loadings_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Loadings_+3A_value">value</code></td>
<td>
<p>Feature loadings to add</p>
</td></tr>
<tr><td><code id="Loadings_+3A_projected">projected</code></td>
<td>
<p>Pull the projected feature loadings?</p>
</td></tr>
<tr><td><code id="Loadings_+3A_reduction">reduction</code></td>
<td>
<p>Name of reduction to pull feature loadings for</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>Loadings</code>: the feature loadings for <code>object</code>
</p>
<p><code>Loadings&lt;-</code>: <code>object</code> with the updated loadings
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the feature loadings for a given DimReduc
Loadings(object = pbmc_small[["pca"]])[1:5,1:5]

# Set the feature loadings for a given DimReduc
new.loadings &lt;- Loadings(object = pbmc_small[["pca"]])
new.loadings &lt;- new.loadings + 0.01
Loadings(object = pbmc_small[["pca"]]) &lt;- new.loadings

# Get the feature loadings for a specified DimReduc in a Seurat object
Loadings(object = pbmc_small, reduction = "pca")[1:5,1:5]

</code></pre>

<hr>
<h2 id='LogMap'>A Logical Map</h2><span id='topic+LogMap'></span><span id='topic++5B+5B+2CLogMap+2Ccharacter+2Cmissing-method'></span><span id='topic++5B+5B+2CLogMap+2Cmissing+2Cmissing-method'></span><span id='topic++5B+5B+2CLogMap+2CNULL+2Cmissing-method'></span><span id='topic++5B+5B+3C-+2CLogMap+2Ccharacter+2Cmissing+2Ccharacter-method'></span><span id='topic++5B+5B+3C-+2CLogMap+2Ccharacter+2Cmissing+2Cinteger-method'></span><span id='topic++5B+5B+3C-+2CLogMap+2Ccharacter+2Cmissing+2CNULL-method'></span><span id='topic++5B+5B+3C-+2CLogMap+2Ccharacter+2Cmissing+2Cnumeric-method'></span><span id='topic+LogMap-class'></span>

<h3>Description</h3>

<p>A simple container for storing mappings of values using logical matrices.
Keeps track of which values (rows) are present in which observations
(columns). <code>LogMap</code> objects can be created with <code>LogMap()</code>;
queries can be performed with <code>[[</code> and observations can be added
or removed with <code>[[&lt;-</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LogMap(y)

## S4 method for signature 'LogMap,character,missing'
x[[i, j, ...]]

## S4 method for signature 'LogMap,missing,missing'
x[[i, j, ...]]

## S4 method for signature 'LogMap,'NULL',missing'
x[[i, j, ...]]

## S4 replacement method for signature 'LogMap,character,missing,character'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'LogMap,character,missing,integer'
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'LogMap,character,missing,'NULL''
x[[i, j, ...]] &lt;- value

## S4 replacement method for signature 'LogMap,character,missing,numeric'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LogMap_+3A_y">y</code></td>
<td>
<p>A character vector</p>
</td></tr>
<tr><td><code id="LogMap_+3A_x">x</code></td>
<td>
<p>A <code>LogMap</code> object</p>
</td></tr>
<tr><td><code id="LogMap_+3A_i">i</code></td>
<td>
<p>A character vector of length 1, or <code>NULL</code></p>
</td></tr>
<tr><td><code id="LogMap_+3A_j">j</code></td>
<td>
<p>Not used</p>
</td></tr>
<tr><td><code id="LogMap_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="LogMap_+3A_value">value</code></td>
<td>
<p>A character or integer vector of values to record in the map
for <code>i</code>, or <code>NULL</code> to remove the record for <code>i</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>LogMap</code>: A new <code>LogMap</code> object with zero columns and
<code>length(x = x)</code> rows; rownames are set to <code>x</code>
</p>
<p><code>[[</code>: if <code>i</code> is a character vector, the rownames that are
mapped to <code>i</code>; otherwise the rownames of <code>x</code>
</p>
<p><code>[[&lt;-</code>: If <code>value</code> is <code>NULL</code>, then <code>x</code> without
the observations for <code>i</code>; otherwise, <code>x</code> with a new column for
<code>i</code> recording a <code>TRUE</code> for all values present in <code>value</code>
</p>


<h3>Slots</h3>


<dl>
<dt><code>.Data</code></dt><dd><p>A logical matrix with at least one row</p>
</dd>
</dl>


<h3>See Also</h3>

<p>Logical map objects, validity, and interaction methods:
<code><a href="#topic+LogMap-validity">LogMap-validity</a></code>,
<code><a href="#topic+as.matrix.LogMap">as.matrix.LogMap</a>()</code>,
<code><a href="#topic+droplevels.LogMap">droplevels.LogMap</a>()</code>,
<code><a href="#topic+intersect.LogMap">intersect.LogMap</a>()</code>,
<code><a href="#topic+labels.LogMap">labels.LogMap</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a LogMap
map &lt;- LogMap(letters[1:10])
map

# Get the names of values in the LogMap
map[[NULL]]
rownames(map)

# Add an observation to the LogMap
map[['obs']] &lt;- c(1, 3, 7)
map[['entry']] &lt;- c(2, 7, 10)
map

# Get the names of observations in the LogMap
colnames(map)

# Fetch an observation from the LogMap
map[['obs']]

# Get the full logical matrix
map[[]]

# Remove an observation from the LogMap
map[['obs']] &lt;- NULL
map[['entry']] &lt;- NULL
map

</code></pre>

<hr>
<h2 id='LogMap-validity'>Logical Map Validity</h2><span id='topic+LogMap-validity'></span>

<h3>Description</h3>

<p>Validation of <code>LogMap</code> objects is handled by
<code><a href="methods.html#topic+validObject">validObject</a></code>
</p>


<h3>Data Validation</h3>

<p>Logical maps must be a logical matrix containing only TRUE or FALSE values
</p>


<h3>Value Validation</h3>

<p>All values must be named within the rownames of the object. Duplicate or
empty (<code>""</code>) values are not allowed
</p>


<h3>Observation Validation</h3>

<p>All observations must be named within the column names of the object.
Duplicate or empty (<code>""</code>) observations are not allowed
</p>


<h3>See Also</h3>

<p><code><a href="methods.html#topic+validObject">validObject</a></code>
</p>
<p>Logical map objects, validity, and interaction methods:
<code><a href="#topic+LogMap">LogMap</a></code>,
<code><a href="#topic+as.matrix.LogMap">as.matrix.LogMap</a>()</code>,
<code><a href="#topic+droplevels.LogMap">droplevels.LogMap</a>()</code>,
<code><a href="#topic+intersect.LogMap">intersect.LogMap</a>()</code>,
<code><a href="#topic+labels.LogMap">labels.LogMap</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map &lt;- LogMap(letters[1:10])
map[['obs']] &lt;- c(1, 3, 7)
map[['entry']] &lt;- c(2, 7, 10)
validObject(map)

</code></pre>

<hr>
<h2 id='LogSeuratCommand'>Log a command</h2><span id='topic+LogSeuratCommand'></span>

<h3>Description</h3>

<p>Logs command run, storing the name, timestamp, and argument list. Stores in
the Seurat object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LogSeuratCommand(object, return.command = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LogSeuratCommand_+3A_object">object</code></td>
<td>
<p>Name of Seurat object</p>
</td></tr>
<tr><td><code id="LogSeuratCommand_+3A_return.command">return.command</code></td>
<td>
<p>Return a <code><a href="#topic+SeuratCommand">SeuratCommand</a></code> object instead</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>return.command</code>, returns a <code><a href="#topic+SeuratCommand">SeuratCommand</a></code>
object; otherwise, returns the Seurat object with command stored
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Command">Command</a></code>
</p>
<p>Command log object and interaction methods
<code><a href="#topic++24.SeuratCommand">$.SeuratCommand</a>()</code>,
<code><a href="#topic+.DollarNames.SeuratCommand">.DollarNames.SeuratCommand</a>()</code>,
<code><a href="#topic+SeuratCommand-class">SeuratCommand-class</a></code>,
<code><a href="#topic++5B.SeuratCommand">[.SeuratCommand</a>()</code>,
<code><a href="#topic+as.list.SeuratCommand">as.list.SeuratCommand</a>()</code>
</p>

<hr>
<h2 id='MatchCells'>Match Cells</h2><span id='topic+MatchCells'></span><span id='topic+MatchCells.character'></span><span id='topic+MatchCells.NULL'></span><span id='topic+MatchCells.numeric'></span>

<h3>Description</h3>

<p>Match Cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MatchCells(new, orig, ordered = FALSE)

## S3 method for class 'character'
MatchCells(new, orig, ordered = FALSE)

## S3 method for class ''NULL''
MatchCells(new, orig, ordered = FALSE)

## S3 method for class 'numeric'
MatchCells(new, orig, ordered = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MatchCells_+3A_new">new</code></td>
<td>
<p>A vector of new cells</p>
</td></tr>
<tr><td><code id="MatchCells_+3A_orig">orig</code></td>
<td>
<p>A vector of existing cells</p>
</td></tr>
<tr><td><code id="MatchCells_+3A_ordered">ordered</code></td>
<td>
<p>Sort the result to the same order as <code>orig</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector with new cells in order of the original cells; if
no match can be found, returns <code>NULL</code>
</p>

<hr>
<h2 id='merge.Assay'>Merge Assays</h2><span id='topic+merge.Assay'></span>

<h3>Description</h3>

<p>Merge one or more v3 assays together
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
merge(
  x = NULL,
  y = NULL,
  add.cell.ids = NULL,
  merge.data = TRUE,
  labels = NULL,
  collapse = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id="merge.Assay_+3A_y">y</code></td>
<td>
<p>One or more <code><a href="#topic+Assay">Assay</a></code> objects</p>
</td></tr>
<tr><td><code id="merge.Assay_+3A_add.cell.ids">add.cell.ids</code></td>
<td>
<p>A character vector of <code>length(x = c(x, y))</code>;
appends the corresponding values to the start of each objects' cell names</p>
</td></tr>
<tr><td><code id="merge.Assay_+3A_merge.data">merge.data</code></td>
<td>
<p>Merge the data slots instead of just merging the counts
(which requires renormalization); this is recommended if the same
normalization approach was applied to all objects</p>
</td></tr>
<tr><td><code id="merge.Assay_+3A_labels">labels</code>, <code id="merge.Assay_+3A_collapse">collapse</code></td>
<td>
<p>Currently unused</p>
</td></tr>
<tr><td><code id="merge.Assay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new assay with data merged from <code>c(x, y)</code>
</p>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>

<hr>
<h2 id='merge.Assay5'>Merge Assays</h2><span id='topic+merge.Assay5'></span>

<h3>Description</h3>

<p>Merge one or more v5 assays together
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
merge(x, y, labels = NULL, add.cell.ids = NULL, collapse = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="merge.Assay5_+3A_y">y</code></td>
<td>
<p>One or more <code><a href="#topic+Assay5">Assay5</a></code> objects</p>
</td></tr>
<tr><td><code id="merge.Assay5_+3A_labels">labels</code></td>
<td>
<p>A character vector equal to the number of objects; defaults to
<code>as.character(seq_along(c(x, y)))</code></p>
</td></tr>
<tr><td><code id="merge.Assay5_+3A_add.cell.ids">add.cell.ids</code></td>
<td>
<p>A character vector equal to the number of objects
provided to append to all cell names; if <code>TRUE</code>, uses <code>labels</code> as
<code>add.cell.ids</code></p>
</td></tr>
<tr><td><code id="merge.Assay5_+3A_collapse">collapse</code></td>
<td>
<p>If <code>TRUE</code>, merge layers of the same name together; if
<code>FALSE</code>, appends <code>labels</code> to the layer name</p>
</td></tr>
<tr><td><code id="merge.Assay5_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Note</strong>: collapsing layers is currently not supported
</p>


<h3>Value</h3>

<p>A new v5 assay with data merged from <code>c(x, y)</code>
</p>


<h3>See Also</h3>

<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>

<hr>
<h2 id='merge.DimReduc'>Merge Dimensional Reductions</h2><span id='topic+merge.DimReduc'></span>

<h3>Description</h3>

<p>Merge two or more <a href="#topic+DimReduc">dimensional reduction</a> together
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DimReduc'
merge(x = NULL, y = NULL, add.cell.ids = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge.DimReduc_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+DimReduc">DimReduc</a></code> object</p>
</td></tr>
<tr><td><code id="merge.DimReduc_+3A_y">y</code></td>
<td>
<p>One or more <code><a href="#topic+DimReduc">DimReduc</a></code> objects</p>
</td></tr>
<tr><td><code id="merge.DimReduc_+3A_add.cell.ids">add.cell.ids</code></td>
<td>
<p>A character vector equal to the number of objects
provided to append to all cell names; if <code>TRUE</code>, uses <code>labels</code> as
<code>add.cell.ids</code></p>
</td></tr>
<tr><td><code id="merge.DimReduc_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>DimReduc</code> object with data merged from <code>c(x, y)</code>
</p>


<h3>See Also</h3>

<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+CreateDimReducObject">CreateDimReducObject</a>()</code>,
<code><a href="#topic+DimReduc-class">DimReduc-class</a></code>,
<code><a href="#topic+DimReduc-validity">DimReduc-validity</a></code>,
<code><a href="#topic++5B.DimReduc">[.DimReduc</a>()</code>,
<code><a href="#topic++5B+5B.DimReduc">[[.DimReduc</a>()</code>,
<code><a href="#topic+dim.DimReduc">dim.DimReduc</a>()</code>,
<code><a href="#topic+print.DimReduc">print.DimReduc</a>()</code>,
<code><a href="#topic+subset.DimReduc">subset.DimReduc</a>()</code>
</p>

<hr>
<h2 id='merge.Seurat'>Merge Seurat Objects</h2><span id='topic+merge.Seurat'></span><span id='topic+merge'></span><span id='topic+MergeSeurat'></span><span id='topic+AddSamples'></span>

<h3>Description</h3>

<p>Merge Seurat Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
merge(
  x = NULL,
  y = NULL,
  add.cell.ids = NULL,
  collapse = FALSE,
  merge.data = TRUE,
  merge.dr = FALSE,
  project = getOption(x = "Seurat.object.project", default = "SeuratProject"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge.Seurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="merge.Seurat_+3A_y">y</code></td>
<td>
<p>A single <code>Seurat</code> object or a list of <code>Seurat</code> objects</p>
</td></tr>
<tr><td><code id="merge.Seurat_+3A_add.cell.ids">add.cell.ids</code></td>
<td>
<p>A character vector of <code>length(x = c(x, y))</code>;
appends the corresponding values to the start of each objects' cell names</p>
</td></tr>
<tr><td><code id="merge.Seurat_+3A_collapse">collapse</code></td>
<td>
<p>If <code>TRUE</code>, merge layers of the same name together; if
<code>FALSE</code>, appends <code>labels</code> to the layer name</p>
</td></tr>
<tr><td><code id="merge.Seurat_+3A_merge.data">merge.data</code></td>
<td>
<p>Merge the data slots instead of just merging the counts
(which requires renormalization); this is recommended if the same
normalization approach was applied to all objects</p>
</td></tr>
<tr><td><code id="merge.Seurat_+3A_merge.dr">merge.dr</code></td>
<td>
<p>Choose how to handle merging dimensional reductions:
</p>

<ul>
<li> <p>&ldquo;<code>TRUE</code>&rdquo;: merge dimensional reductions with the same name
across objects; dimensional reductions with different names are added as-is
</p>
</li>
<li> <p>&ldquo;<code>NA</code>&rdquo;: keep dimensional reductions from separate objects
separate; will append the project name for duplicate reduction names
</p>
</li>
<li> <p>&ldquo;<code>FALSE</code>&rdquo;: do not add dimensional reductions
</p>
</li></ul>
</td></tr>
<tr><td><code id="merge.Seurat_+3A_project">project</code></td>
<td>
<p><a href="#topic+Project">Project</a> name for the <code>Seurat</code> object</p>
</td></tr>
<tr><td><code id="merge.Seurat_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>merge</code>: Merged object
</p>


<h3>Merge Details</h3>

<p>When merging Seurat objects, the merge procedure will merge the Assay level
counts and potentially the data slots (depending on the merge.data parameter).
It will also merge the cell-level meta data that was stored with each object
and preserve the cell identities that were active in the objects pre-merge.
The merge will optionally merge reductions depending on the values passed to
<code>merge.dr</code> if they have the same name across objects. Here the
embeddings slots will be merged and if there are differing numbers of
dimensions across objects, only the first N shared dimensions will be merged.
The feature loadings slots will be filled by the values present in the first
object.The merge will not preserve graphs, logged commands, or feature-level
metadata that were present in the original objects. If add.cell.ids isn't
specified and any cell names are duplicated, cell names will be appended
with _X, where X is the numeric index of the object in c(x, y).
</p>


<h3>See Also</h3>

<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># `merge' examples
# merge two objects
merge(pbmc_small, y = pbmc_small)
# to merge more than two objects, pass one to x and a list of objects to y
merge(pbmc_small, y = c(pbmc_small, pbmc_small))

</code></pre>

<hr>
<h2 id='merge.StdAssay'>Merge Assays</h2><span id='topic+merge.StdAssay'></span>

<h3>Description</h3>

<p>Merge one or more v5 assays together
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
merge(x, y, labels = NULL, add.cell.ids = NULL, collapse = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="merge.StdAssay_+3A_y">y</code></td>
<td>
<p>One or more <code><a href="#topic+Assay5">Assay5</a></code> objects</p>
</td></tr>
<tr><td><code id="merge.StdAssay_+3A_labels">labels</code></td>
<td>
<p>A character vector equal to the number of objects; defaults to
<code>as.character(seq_along(c(x, y)))</code></p>
</td></tr>
<tr><td><code id="merge.StdAssay_+3A_add.cell.ids">add.cell.ids</code></td>
<td>
<p>A character vector equal to the number of objects
provided to append to all cell names; if <code>TRUE</code>, uses <code>labels</code> as
<code>add.cell.ids</code></p>
</td></tr>
<tr><td><code id="merge.StdAssay_+3A_collapse">collapse</code></td>
<td>
<p>If <code>TRUE</code>, merge layers of the same name together; if
<code>FALSE</code>, appends <code>labels</code> to the layer name</p>
</td></tr>
<tr><td><code id="merge.StdAssay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Note</strong>: collapsing layers is currently not supported
</p>


<h3>Value</h3>

<p>A new v5 assay with data merged from <code>c(x, y)</code>
</p>


<h3>Note</h3>

<p>All assays must be of the same type; merging different v5 assays (eg.
<code><a href="#topic+Assay5">Assay5</a></code> and <code><a href="#topic+Assay5T">Assay5T</a></code>) is currently unsupported
</p>

<hr>
<h2 id='Misc'>Get and set miscellaneous data</h2><span id='topic+Misc'></span><span id='topic+Misc+3C-'></span><span id='topic+Misc.Assay'></span><span id='topic+Misc+3C-.Assay'></span><span id='topic+Misc.Assay5'></span><span id='topic+Misc+3C-.Assay5'></span><span id='topic+Misc.DimReduc'></span><span id='topic+Misc+3C-.DimReduc'></span><span id='topic+Misc.Seurat'></span><span id='topic+Misc+3C-.Seurat'></span>

<h3>Description</h3>

<p>Get and set miscellaneous data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Misc(object, ...)

Misc(object, ...) &lt;- value

## S3 method for class 'Assay'
Misc(object, slot = NULL, ...)

## S3 replacement method for class 'Assay'
Misc(object, slot, ...) &lt;- value

## S3 method for class 'Assay5'
Misc(object, slot = NULL, ...)

## S3 replacement method for class 'Assay5'
Misc(object, slot, ...) &lt;- value

## S3 method for class 'DimReduc'
Misc(object, slot = NULL, ...)

## S3 replacement method for class 'DimReduc'
Misc(object, slot, ...) &lt;- value

## S3 method for class 'Seurat'
Misc(object, slot = NULL, ...)

## S3 replacement method for class 'Seurat'
Misc(object, slot, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Misc_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Misc_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Misc_+3A_value">value</code></td>
<td>
<p>Data to add</p>
</td></tr>
<tr><td><code id="Misc_+3A_slot">slot</code></td>
<td>
<p>Name of specific bit of meta data to pull</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Miscellaneous data
</p>
<p>An object with miscellaneous data added
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the misc info
Misc(object = pbmc_small, slot = "example")

# Add misc info
Misc(object = pbmc_small, slot = "example") &lt;- "testing_misc"

</code></pre>

<hr>
<h2 id='Misc-StdAssay'>Get and set miscellaneous data</h2><span id='topic+Misc-StdAssay'></span><span id='topic+Misc.StdAssay'></span><span id='topic+Misc+3C-.StdAssay'></span>

<h3>Description</h3>

<p>Get and set miscellaneous data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
Misc(object, slot = NULL, ...)

## S3 replacement method for class 'StdAssay'
Misc(object, slot, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Misc-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Misc-StdAssay_+3A_slot">slot</code></td>
<td>
<p>Name of specific bit of meta data to pull</p>
</td></tr>
<tr><td><code id="Misc-StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Misc-StdAssay_+3A_value">value</code></td>
<td>
<p>Data to add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Miscellaneous data
</p>
<p>An object with miscellaneous data added
</p>

<hr>
<h2 id='Molecules-class'>The Spatial Molecules Class</h2><span id='topic+Molecules-class'></span>

<h3>Description</h3>

<p>The Spatial Molecules Class
</p>


<h3>Slots</h3>


<dl>
<dt><code>.Data</code></dt><dd><p>A list of <code><a href="sp.html#topic+SpatialPoints">SpatialPoints</a></code> objects</p>
</dd>
<dt><code>key</code></dt><dd><p>The key for the <code>Molecules</code></p>
</dd>
</dl>


<h3>See Also</h3>

<p><code>Molecules</code> methods: <code><a href="#topic+Molecules-methods">Molecules-methods</a></code>
</p>
<p>Segmentation layer classes:
<code><a href="#topic+Centroids-class">Centroids-class</a></code>,
<code><a href="#topic+Centroids-methods">Centroids-methods</a></code>,
<code><a href="#topic+Molecules-methods">Molecules-methods</a></code>,
<code><a href="#topic+Segmentation-class">Segmentation-class</a></code>,
<code><a href="#topic+Segmentation-methods">Segmentation-methods</a></code>
</p>

<hr>
<h2 id='Molecules-methods'><code>Molecules</code> Methods</h2><span id='topic+Molecules-methods'></span><span id='topic+Features.Molecules'></span><span id='topic+GetTissueCoordinates.Molecules'></span><span id='topic+subset.Molecules'></span><span id='topic+show+2CMolecules-method'></span>

<h3>Description</h3>

<p>Methods for <code><a href="#topic+Molecules">Molecules</a></code> objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Molecules'
Features(x, ...)

## S3 method for class 'Molecules'
GetTissueCoordinates(object, features = NULL, ...)

## S3 method for class 'Molecules'
subset(x, features = NULL, ...)

## S4 method for signature 'Molecules'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Molecules-methods_+3A_x">x</code>, <code id="Molecules-methods_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+Molecules">Molecules</a></code> object</p>
</td></tr>
<tr><td><code id="Molecules-methods_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Molecules-methods_+3A_features">features</code></td>
<td>
<p>A vector of molecule names to keep; if <code>NULL</code>, defaults
to all molecules</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Features</code>: Get spatially-resolved molecule names
</p>
<p><code>GetTissueCoordinates</code>: Get spatially-resolved
molecule coordinates
</p>
<p><code>subset</code>: Subset a <code>Molecules</code> object to certain molecules
</p>
<p><code>show</code>: Display an object summary to stdout
</p>


<h3>Value</h3>

<p><code>Features</code>: A vector of spatially-resolved molecule names;
if no molecular information present, returns <code>NULL</code>
</p>
<p><code>GetTissueCoordinates</code>: A data frame with three columns:
</p>

<ul>
<li> <p>&ldquo;<code>x</code>&rdquo;: the x-coordinate of a molecule
</p>
</li>
<li> <p>&ldquo;<code>y</code>&rdquo;: the y-coordinate of a molecule
</p>
</li>
<li> <p>&ldquo;<code>molecule</code>&rdquo;: the molecule name
</p>
</li></ul>

<p><code>subset</code>: <code>x</code> subsetted to the features specified
by <code>features</code>
</p>
<p><code>show</code>: Invisibly returns <code>NULL</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Molecules-class">Molecules-class</a></code>
</p>
<p>Segmentation layer classes:
<code><a href="#topic+Centroids-class">Centroids-class</a></code>,
<code><a href="#topic+Centroids-methods">Centroids-methods</a></code>,
<code><a href="#topic+Molecules-class">Molecules-class</a></code>,
<code><a href="#topic+Segmentation-class">Segmentation-class</a></code>,
<code><a href="#topic+Segmentation-methods">Segmentation-methods</a></code>
</p>

<hr>
<h2 id='names.Seurat'>Subobject Names</h2><span id='topic+names.Seurat'></span>

<h3>Description</h3>

<p>Get the names of subobjects within a <code><a href="#topic+Seurat">Seurat</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
names(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="names.Seurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The names of all of the following subobjects within <code>x</code>:
</p>

<ul>
<li> <p><a href="#topic+Assay">v3</a> and <a href="#topic+Assay5">v5</a> assays
</p>
</li>
<li> <p><a href="#topic+DimReduc">dimensional reductions</a>
</p>
</li>
<li> <p><a href="#topic+SpatialImage">images</a> and <a href="#topic+FOV">FOVs</a>
</p>
</li>
<li> <p><a href="#topic+Graph">nearest-neighbor graphs</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>names(pbmc_small)

</code></pre>

<hr>
<h2 id='Neighbor-class'>The Neighbor class</h2><span id='topic+Neighbor-class'></span><span id='topic+Neighbor'></span>

<h3>Description</h3>

<p>The Neighbor class is used to store the results of neighbor finding
algorithms
</p>


<h3>Slots</h3>


<dl>
<dt><code>nn.idx</code></dt><dd><p>Matrix containing the nearest neighbor indices</p>
</dd>
<dt><code>nn.dist</code></dt><dd><p>Matrix containing the nearest neighbor distances</p>
</dd>
<dt><code>alg.idx</code></dt><dd><p>The neighbor finding index (if applicable). E.g. the annoy
index</p>
</dd>
<dt><code>alg.info</code></dt><dd><p>Any information associated with the algorithm that may be
needed downstream (e.g. distance metric used with annoy is needed when
reading in from stored file).</p>
</dd>
<dt><code>cell.names</code></dt><dd><p>Names of the cells for which the neighbors have been
computed.</p>
</dd>
</dl>

<hr>
<h2 id='Neighbor-methods'><code>Neighbor</code> Methods</h2><span id='topic+Neighbor-methods'></span><span id='topic+dim.Neighbor'></span><span id='topic+show+2CNeighbor-method'></span>

<h3>Description</h3>

<p>Methods for <code><a href="#topic+Neighbor">Neighbor</a></code> objects for generics defined in
other packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Neighbor'
dim(x)

## S4 method for signature 'Neighbor'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Neighbor-methods_+3A_x">x</code>, <code id="Neighbor-methods_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+Neighbor">Neighbor</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dim</code> Dimensions of the indices matrix
</p>
<p><code>show</code>: Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and
invisibly returns <code>NULL</code>
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>dim(Neighbor)</code>: Dimensions of the neighbor indices
</p>
</li>
<li> <p><code>show(Neighbor)</code>: Overview of a <code>Neighbor</code> object
</p>
</li></ul>

<hr>
<h2 id='old-assign'>Original double-bracket assign</h2><span id='topic+old-assign'></span><span id='topic++5B+5B+3C-+2CSeurat+2Ccharacter+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>This function has been replaced with value-specific double-bracket
assign methods and should generally not be called
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'Seurat,character,ANY,ANY'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="old-assign_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="old-assign_+3A_i">i</code></td>
<td>
<p>The name to store a subobject or various cell-level meta data as</p>
</td></tr>
<tr><td><code id="old-assign_+3A_value">value</code></td>
<td>
<p>New subobject or cell-level meta data</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with <code>value</code> stored as <code>i</code>
</p>


<h3>See Also</h3>

<p>See <a href="#topic++24.Seurat">here</a> for adding metadata with <code>[[&lt;-</code>, and
<a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">here</a> for removing subobjects and cell-level meta
data with <code>[[&lt;-</code>
</p>

<hr>
<h2 id='Overlay'>Overlay <code>Spatial</code> Objects Over One Another</h2><span id='topic+Overlay'></span><span id='topic+Overlay+2CCentroids+2CSpatialPolygons-method'></span><span id='topic+Overlay+2CSegmentation+2CSpatialPolygons-method'></span><span id='topic+Overlay+2CMolecules+2CSpatialPolygons-method'></span><span id='topic+Overlay+2CFOV+2CSpatial-method'></span><span id='topic+Overlay+2CFOV+2CSpatialPolygons-method'></span><span id='topic+Overlay+2CFOV+2CFOV-method'></span>

<h3>Description</h3>

<p>Create an overlay of some query spatial object (<code>x</code>) against some
target object (<code>y</code>). Basically, find all components of a query that
fall within the bounds of a target spatial region
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Overlay(x, y, invert = FALSE, ...)

## S4 method for signature 'Centroids,SpatialPolygons'
Overlay(x, y, invert = FALSE, ...)

## S4 method for signature 'Segmentation,SpatialPolygons'
Overlay(x, y, invert = FALSE, ...)

## S4 method for signature 'Molecules,SpatialPolygons'
Overlay(x, y, invert = FALSE, ...)

## S4 method for signature 'FOV,Spatial'
Overlay(x, y, invert = FALSE, ...)

## S4 method for signature 'FOV,SpatialPolygons'
Overlay(x, y, invert = FALSE, ...)

## S4 method for signature 'FOV,FOV'
Overlay(x, y, invert = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Overlay_+3A_x">x</code></td>
<td>
<p>Query <code>Spatial</code> object</p>
</td></tr>
<tr><td><code id="Overlay_+3A_y">y</code></td>
<td>
<p>Target <code>Spatial</code> object</p>
</td></tr>
<tr><td><code id="Overlay_+3A_invert">invert</code></td>
<td>
<p>Invert the overlay and return only the components of <code>x</code>
that fall <em>outside</em> the bounds of <code>y</code></p>
</td></tr>
<tr><td><code id="Overlay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with only the components that fall within the
bounds of <code>y</code>
</p>


<h3>Note</h3>

<p>This function requires the
<a href="https://cran.r-project.org/package=sf"><span class="pkg">sf</span></a> package
to be installed
</p>

<hr>
<h2 id='PackageCheck'>Check the existence of a package</h2><span id='topic+PackageCheck'></span>

<h3>Description</h3>

<p>Check the existence of a package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PackageCheck(..., error = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PackageCheck_+3A_...">...</code></td>
<td>
<p>Package names</p>
</td></tr>
<tr><td><code id="PackageCheck_+3A_error">error</code></td>
<td>
<p>If true, throw an error if the package doesn't exist</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns boolean denoting if the package is installed
</p>


<h3>Lifecycle</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p><code>PackageCheck</code> was deprecated in version 5.0.0; please use
<code><a href="rlang.html#topic+check_installed">rlang::check_installed</a>()</code> instead
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PackageCheck("SeuratObject", error = FALSE)

</code></pre>

<hr>
<h2 id='pbmc_small'>A small example version of the PBMC dataset</h2><span id='topic+pbmc_small'></span>

<h3>Description</h3>

<p>A subsetted version of 10X Genomics' 3k PBMC dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbmc_small
</code></pre>


<h3>Format</h3>

<p>A Seurat object with the following slots filled
</p>

<dl>
<dt>assays</dt><dd>
<p>Currently only contains one assay (&quot;RNA&quot; - scRNA-seq expression data)
</p>
<ul>
<li><p>counts - Raw expression data
</p>
</li>
<li><p>data - Normalized expression data
</p>
</li>
<li><p>scale.data - Scaled expression data
</p>
</li>
<li><p>var.features - names of the current features selected as variable
</p>
</li>
<li><p>meta.features - Assay level metadata such as mean and variance
</p>
</li></ul>
</dd>
<dt>meta.data</dt><dd><p>Cell level metadata</p>
</dd>
<dt>active.assay</dt><dd><p>Current default assay</p>
</dd>
<dt>active.ident</dt><dd><p>Current default idents</p>
</dd>
<dt>graphs</dt><dd><p>Neighbor graphs computed, currently stores the SNN</p>
</dd>
<dt>reductions</dt><dd><p>Dimensional reductions: currently PCA and tSNE</p>
</dd>
<dt>version</dt><dd><p>Seurat version used to create the object</p>
</dd>
<dt>commands</dt><dd><p>Command history</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/pbmc3k">https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/pbmc3k</a>
</p>

<hr>
<h2 id='PolyVtx'>Polygon Vertices</h2><span id='topic+PolyVtx'></span>

<h3>Description</h3>

<p>Calculate the vertices of a regular polygon given the number of sides and
its radius (distance from center to vertex). Also permits transforming the
resulting coordinates by moving the origin and altering the initial angle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PolyVtx(n, r = 1L, xc = 0L, yc = 0L, t1 = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PolyVtx_+3A_n">n</code></td>
<td>
<p>Number of sides of the polygon</p>
</td></tr>
<tr><td><code id="PolyVtx_+3A_r">r</code></td>
<td>
<p>Radius of the polygon</p>
</td></tr>
<tr><td><code id="PolyVtx_+3A_xc">xc</code>, <code id="PolyVtx_+3A_yc">yc</code></td>
<td>
<p>X/Y coordinates for the center of the polygon</p>
</td></tr>
<tr><td><code id="PolyVtx_+3A_t1">t1</code></td>
<td>
<p>Angle of the first vertex in degrees</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code> with <code>n</code> rows and two columns:
</p>

<dl>
<dt><code>x</code></dt><dd><p>X positions of each coordinate</p>
</dd>
<dt><code>y</code></dt><dd><p>Y positions of each coordinate</p>
</dd>
</dl>



<h3>References</h3>

<p><a href="https://stackoverflow.com/questions/3436453/calculate-coordinates-of-a-regular-polygons-vertices">https://stackoverflow.com/questions/3436453/calculate-coordinates-of-a-regular-polygons-vertices</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Angles">Angles</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coords &lt;- PolyVtx(5, t1 = 90)
coords
if (requireNamespace("ggplot2", quietly = TRUE)) {
  ggplot2::ggplot(coords, ggplot2::aes(x = x, y = y)) + ggplot2::geom_polygon()
}

</code></pre>

<hr>
<h2 id='print.DimReduc'>Print Top Feature Loadings</h2><span id='topic+print.DimReduc'></span><span id='topic+print'></span>

<h3>Description</h3>

<p>Prints a set of features that most strongly define a set of components;
<strong>note</strong>: requires feature loadings to be present in order to work
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DimReduc'
print(x, dims = 1:5, nfeatures = 20, projected = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.DimReduc_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+DimReduc">DimReduc</a></code> object</p>
</td></tr>
<tr><td><code id="print.DimReduc_+3A_dims">dims</code></td>
<td>
<p>Number of dimensions to display</p>
</td></tr>
<tr><td><code id="print.DimReduc_+3A_nfeatures">nfeatures</code></td>
<td>
<p>Number of genes to display</p>
</td></tr>
<tr><td><code id="print.DimReduc_+3A_projected">projected</code></td>
<td>
<p>Use projected slot</p>
</td></tr>
<tr><td><code id="print.DimReduc_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Displays set of features defining the components and
invisibly returns <code>x</code>
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+cat">cat</a></code>
</p>
<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+CreateDimReducObject">CreateDimReducObject</a>()</code>,
<code><a href="#topic+DimReduc-class">DimReduc-class</a></code>,
<code><a href="#topic+DimReduc-validity">DimReduc-validity</a></code>,
<code><a href="#topic++5B.DimReduc">[.DimReduc</a>()</code>,
<code><a href="#topic++5B+5B.DimReduc">[[.DimReduc</a>()</code>,
<code><a href="#topic+dim.DimReduc">dim.DimReduc</a>()</code>,
<code><a href="#topic+merge.DimReduc">merge.DimReduc</a>()</code>,
<code><a href="#topic+subset.DimReduc">subset.DimReduc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- pbmc_small[["pca"]]
print(pca)

</code></pre>

<hr>
<h2 id='Project'>Get and set project information</h2><span id='topic+Project'></span><span id='topic+Project+3C-'></span><span id='topic+Project.Seurat'></span><span id='topic+Project+3C-.Seurat'></span>

<h3>Description</h3>

<p>Get and set project information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Project(object, ...)

Project(object, ...) &lt;- value

## S3 method for class 'Seurat'
Project(object, ...)

## S3 replacement method for class 'Seurat'
Project(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Project_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Project_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Project_+3A_value">value</code></td>
<td>
<p>Project information to set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Project information
</p>
<p>An object with project information added
</p>

<hr>
<h2 id='Radius'>Get the spot radius from an image</h2><span id='topic+Radius'></span>

<h3>Description</h3>

<p>Get the spot radius from an image
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Radius(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Radius_+3A_object">object</code></td>
<td>
<p>An image object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The radius size
</p>

<hr>
<h2 id='RandomName'>Generate a random name</h2><span id='topic+RandomName'></span>

<h3>Description</h3>

<p>Make a name from randomly sampled characters, pasted together with no spaces
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RandomName(length = 5L, chars = letters, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RandomName_+3A_length">length</code></td>
<td>
<p>How long should the name be</p>
</td></tr>
<tr><td><code id="RandomName_+3A_chars">chars</code></td>
<td>
<p>A vector of 1-length characters to use to generate the name</p>
</td></tr>
<tr><td><code id="RandomName_+3A_...">...</code></td>
<td>
<p>Extra parameters passed to <code><a href="base.html#topic+sample">sample</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character with <code>nchar == length</code> of randomly sampled letters
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+sample">sample</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(42L)
RandomName()
RandomName(7L, replace = TRUE)

</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+plan'></span><span id='topic+intersect'></span><span id='topic+handlers'></span><span id='topic+with_progress'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>future</dt><dd><p><code><a href="future.html#topic+plan">plan</a></code></p>
</dd>
<dt>generics</dt><dd><p><code><a href="generics.html#topic+setops">intersect</a></code></p>
</dd>
<dt>progressr</dt><dd><p><code><a href="progressr.html#topic+handlers">handlers</a></code>, <code><a href="progressr.html#topic+with_progress">with_progress</a></code></p>
</dd>
</dl>

<hr>
<h2 id='RegisterSparseMatrix'>Register Sparse Matrix Classes</h2><span id='topic+RegisterSparseMatrix'></span>

<h3>Description</h3>

<p>Register Sparse Matrix Classes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RegisterSparseMatrix(class, package = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RegisterSparseMatrix_+3A_class">class</code></td>
<td>
<p>Class name</p>
</td></tr>
<tr><td><code id="RegisterSparseMatrix_+3A_package">package</code></td>
<td>
<p>Optional name of package; by default, will search namespaces
of loaded packages to determine the providing package</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>NULL</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+.SparseSlots">.SparseSlots</a>()</code>,
<code><a href="#topic+IsSparse">IsSparse</a>()</code>
</p>

<hr>
<h2 id='RenameAssays'>Rename assays in a <code>Seurat</code> object</h2><span id='topic+RenameAssays'></span>

<h3>Description</h3>

<p>Rename assays in a <code>Seurat</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RenameAssays(
  object,
  assay.name = NULL,
  new.assay.name = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RenameAssays_+3A_object">object</code></td>
<td>
<p>A <code>Seurat</code> object</p>
</td></tr>
<tr><td><code id="RenameAssays_+3A_assay.name">assay.name</code></td>
<td>
<p>original name of assay</p>
</td></tr>
<tr><td><code id="RenameAssays_+3A_new.assay.name">new.assay.name</code></td>
<td>
<p>new name of assay</p>
</td></tr>
<tr><td><code id="RenameAssays_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print messages</p>
</td></tr>
<tr><td><code id="RenameAssays_+3A_...">...</code></td>
<td>
<p>Named arguments as <code>old.assay = new.assay</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>object</code> with assays renamed
</p>


<h3>Examples</h3>

<pre><code class='language-R'>RenameAssays(object = pbmc_small, RNA = 'rna')

</code></pre>

<hr>
<h2 id='RenameCells'>Rename cells</h2><span id='topic+RenameCells'></span><span id='topic+RenameCells.Assay'></span><span id='topic+RenameCells.Assay5'></span><span id='topic+RenameCells.DimReduc'></span><span id='topic+RenameCells.Neighbor'></span><span id='topic+RenameCells.Seurat'></span>

<h3>Description</h3>

<p>Change the cell names in all the different parts of an object. Can be useful
before combining multiple objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RenameCells(object, ...)

## S3 method for class 'Assay'
RenameCells(object, new.names = NULL, ...)

## S3 method for class 'Assay5'
RenameCells(object, new.names = NULL, ...)

## S3 method for class 'DimReduc'
RenameCells(object, new.names = NULL, ...)

## S3 method for class 'Neighbor'
RenameCells(object, old.names = NULL, new.names = NULL, ...)

## S3 method for class 'Seurat'
RenameCells(
  object,
  add.cell.id = missing_arg(),
  new.names = missing_arg(),
  for.merge = deprecated(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RenameCells_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="RenameCells_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="RenameCells_+3A_new.names">new.names</code></td>
<td>
<p>vector of new cell names</p>
</td></tr>
<tr><td><code id="RenameCells_+3A_old.names">old.names</code></td>
<td>
<p>vector of old cell names</p>
</td></tr>
<tr><td><code id="RenameCells_+3A_add.cell.id">add.cell.id</code></td>
<td>
<p>prefix to add cell names</p>
</td></tr>
<tr><td><code id="RenameCells_+3A_for.merge">for.merge</code></td>
<td>
<p>Deprecated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>add.cell.id</code> is set a prefix is added to existing cell names. If
<code>new.names</code> is set these will be used to replace existing names.
</p>


<h3>Value</h3>

<p>An object with new cell names
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Rename cells in an Assay
head(x = colnames(x = pbmc_small[["RNA"]]))
renamed.assay &lt;- RenameCells(
    pbmc_small[["RNA"]],
    new.names = paste0("A_", colnames(x = pbmc_small[["RNA"]]))
)
head(x = colnames(x = renamed.assay))

# Rename cells in a DimReduc
head(x = Cells(x = pbmc_small[["pca"]]))
renamed.dimreduc &lt;- RenameCells(
    object = pbmc_small[["pca"]],
    new.names = paste0("A_", Cells(x = pbmc_small[["pca"]]))
)
head(x = Cells(x = renamed.dimreduc))

# Rename cells in a Seurat object
head(x = colnames(x = pbmc_small))
pbmc_small &lt;- RenameCells(object = pbmc_small, add.cell.id = "A")
head(x = colnames(x = pbmc_small))

</code></pre>

<hr>
<h2 id='RenameCells-StdAssay'>Rename cells</h2><span id='topic+RenameCells-StdAssay'></span><span id='topic+RenameCells.StdAssay'></span>

<h3>Description</h3>

<p>Change the cell names in all the different parts of an object. Can be useful
before combining multiple objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
RenameCells(object, new.names = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RenameCells-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="RenameCells-StdAssay_+3A_new.names">new.names</code></td>
<td>
<p>vector of new cell names</p>
</td></tr>
<tr><td><code id="RenameCells-StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>add.cell.id</code> is set a prefix is added to existing cell names. If
<code>new.names</code> is set these will be used to replace existing names.
</p>


<h3>Value</h3>

<p>An object with new cell names
</p>

<hr>
<h2 id='RowMergeSparseMatrices'>Merge Sparse Matrices by Row</h2><span id='topic+RowMergeSparseMatrices'></span>

<h3>Description</h3>

<p>Merge two or more sparse matrices by rowname.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RowMergeSparseMatrices(mat1, mat2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RowMergeSparseMatrices_+3A_mat1">mat1</code></td>
<td>
<p>First matrix</p>
</td></tr>
<tr><td><code id="RowMergeSparseMatrices_+3A_mat2">mat2</code></td>
<td>
<p>Second matrix or list of matrices</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Shared matrix rows (with the same row name) will be merged, and unshared
rows (with different names) will be filled with zeros in the matrix not
containing the row.
</p>


<h3>Value</h3>

<p>Returns a sparse matrix
</p>

<hr>
<h2 id='s4list'>S4/List Conversion</h2><span id='topic+s4list'></span><span id='topic+S4ToList'></span><span id='topic+IsS4List'></span><span id='topic+ListToS4'></span><span id='topic+S4ToList.default'></span><span id='topic+S4ToList.list'></span>

<h3>Description</h3>

<p>Convert S4 objects to lists and vice versa. Useful for declassing an S4
object while keeping track of it's class using attributes (see section
<strong>S4 Class Definition Attributes</strong> below for more details). Both
<code>ListToS4</code> and <code>S4ToList</code> are recursive functions, affecting
all lists/S4 objects contained as sub-lists/sub-objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>S4ToList(object)

IsS4List(x)

ListToS4(x)

## Default S3 method:
S4ToList(object)

## S3 method for class 'list'
S4ToList(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="s4list_+3A_object">object</code></td>
<td>
<p>An S4 object</p>
</td></tr>
<tr><td><code id="s4list_+3A_x">x</code></td>
<td>
<p>A list with an S4 class definition
(&ldquo;<code>classDef</code>&rdquo;) attribute</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>S4ToList</code>: A list with an S4 class definition attribute
</p>
<p><code>IsS4List</code>: <code>TRUE</code> if <code>x</code> is a list with an S4 class
definition attribute
</p>
<p><code>ListToS4</code>: An S4 object as defined by the S4 class definition
attribute
</p>


<h3>S4 Class Definition Attributes</h3>

<p>S4 classes are scoped to the package and class name. In order to properly
track which class a list is generated from in order to build a new one,
these function use an <code><a href="base.html#topic+attr">attribute</a></code> to denote the
class name and package of origin. This attribute is stored as 1-length
character vector named &ldquo;<code>classDef</code>&rdquo; and takes the form
of &ldquo;<code>package:class</code>&rdquo;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ClassKey">ClassKey</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Turn an S4 object into a list
pbmc.list &lt;- S4ToList(pbmc_small)
class(pbmc.list)
attributes(pbmc.list)

str(pbmc.list$reductions)

IsS4List(pbmc.list)

pbmc2 &lt;- ListToS4(pbmc.list)
pbmc2
class(pbmc2)
Reductions(pbmc2)
validObject(pbmc2)

</code></pre>

<hr>
<h2 id='SaveSeuratRds'>Save and Load <code>Seurat</code> Objects from Rds files</h2><span id='topic+SaveSeuratRds'></span><span id='topic+LoadSeuratRds'></span>

<h3>Description</h3>

<p>Save and Load <code>Seurat</code> Objects from Rds files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SaveSeuratRds(
  object,
  file = NULL,
  move = TRUE,
  destdir = deprecated(),
  relative = FALSE,
  ...
)

LoadSeuratRds(file, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SaveSeuratRds_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="SaveSeuratRds_+3A_file">file</code></td>
<td>
<p>Path to save <code>object</code> to; defaults to
<code>file.path(getwd(), paste0(Project(object), ".Rds"))</code></p>
</td></tr>
<tr><td><code id="SaveSeuratRds_+3A_move">move</code></td>
<td>
<p>Move on-disk layers into <code>dirname(file)</code></p>
</td></tr>
<tr><td><code id="SaveSeuratRds_+3A_destdir">destdir</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a></p>
</td></tr>
<tr><td><code id="SaveSeuratRds_+3A_relative">relative</code></td>
<td>
<p>Save relative paths instead of absolute ones</p>
</td></tr>
<tr><td><code id="SaveSeuratRds_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="base.html#topic+readRDS">base::saveRDS</a></code>, <code><a href="base.html#topic+readRDS">base::readRDS</a></code>
</p>

<dl>
<dt><code>ascii</code></dt><dd><p>a logical.  If <code>TRUE</code> or <code>NA</code>, an ASCII
representation is written; otherwise (default), a binary one is used.
See the comments in the help for <code><a href="base.html#topic+save">save</a></code>.</p>
</dd>
<dt><code>version</code></dt><dd><p>the workspace format version to use.  <code>NULL</code>
specifies the current default version (3). The only other supported
value is 2, the default from <span class="rlang"><b>R</b></span> 1.4.0 to <span class="rlang"><b>R</b></span> 3.5.0.</p>
</dd>
<dt><code>compress</code></dt><dd><p>a logical specifying whether saving to a named file is
to use <code>"gzip"</code> compression, or one of <code>"gzip"</code>,
<code>"bzip2"</code> or <code>"xz"</code> to indicate the type of compression to
be used.  Ignored if <code>file</code> is a connection.</p>
</dd>
<dt><code>refhook</code></dt><dd><p>a hook function for handling reference objects.</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>file</code>
</p>


<h3>Progress Updates with <span class="pkg">progressr</span></h3>

<p>This function uses
<a href="https://cran.r-project.org/package=progressr"><span class="pkg">progressr</span></a> to
render status updates and progress bars. To enable progress updates, wrap
the function call in <code><a href="progressr.html#topic+with_progress">with_progress</a></code> or run
<code><a href="progressr.html#topic+handlers">handlers(global = TRUE)</a></code> before running
this function. For more details about <span class="pkg">progressr</span>, please read
<a href="https://progressr.futureverse.org/articles/progressr-intro.html"><code>vignette("progressr-intro")</code></a>
</p>


<h3>Note</h3>

<p>This function requires the
<a href="https://cran.r-project.org/package=fs"><span class="pkg">fs</span></a> package
to be installed
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+saveRDS">saveRDS</a>()</code> <code><a href="base.html#topic+readRDS">readRDS</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("fs", quietly = TRUE)) {
  # Write out with DelayedArray
  if (requireNamespace("HDF5Array", quietly = TRUE)) {
    pbmc &lt;- pbmc_small

    pbmc[["disk"]] &lt;- CreateAssay5Object(list(
      mem = LayerData(pbmc, "counts"),
      disk = as(LayerData(pbmc, "counts"), "HDF5Array")
    ))

    # Save `pbmc` to an Rds file
    out &lt;- tempfile(fileext = ".Rds")
    SaveSeuratRds(pbmc, file = out)

    # Object cache
    obj &lt;- readRDS(out)
    Tool(obj, "SaveSeuratRds")

    # Load the saved object with on-disk layers back into memory
    pbmc2 &lt;- LoadSeuratRds(out)
    pbmc2
    pbmc2[["disk"]]
  }

  # Write out with BPCells
  if (requireNamespace("BPCells", quietly = TRUE)) {
    pbmc &lt;- pbmc_small

    bpm &lt;- BPCells::write_matrix_dir(LayerData(pbmc, "counts"), dir = tempfile())
    bph &lt;- BPCells::write_matrix_hdf5(
      LayerData(pbmc, "counts"),
      path = tempfile(fileext = ".h5"),
      group = "counts"
    )
    pbmc[["disk"]] &lt;- CreateAssay5Object(list(dir = bpm, h5 = bph))

    # Save `pbmc` to an Rds file
    out &lt;- tempfile(fileext = ".Rds")
    SaveSeuratRds(pbmc, file = out)

    # Object cache
    obj &lt;- readRDS(out)
    Tool(obj, "SaveSeuratRds")

    # Load the saved object with on-disk layers back into memory
    pbmc2 &lt;- LoadSeuratRds(out)
    pbmc2
    pbmc2[["disk"]]
  }
}

</code></pre>

<hr>
<h2 id='Segmentation-class'>The <code>Segmentation</code> Class</h2><span id='topic+Segmentation-class'></span>

<h3>Description</h3>

<p>The <code>Segmentation</code> Class
</p>


<h3>See Also</h3>

<p><code>Segmentation</code> methods: <code><a href="#topic+Segmentation-methods">Segmentation-methods</a></code>
</p>
<p>Segmentation layer classes:
<code><a href="#topic+Centroids-class">Centroids-class</a></code>,
<code><a href="#topic+Centroids-methods">Centroids-methods</a></code>,
<code><a href="#topic+Molecules-class">Molecules-class</a></code>,
<code><a href="#topic+Molecules-methods">Molecules-methods</a></code>,
<code><a href="#topic+Segmentation-methods">Segmentation-methods</a></code>
</p>

<hr>
<h2 id='Segmentation-methods'><code>Segmentation</code> Methods</h2><span id='topic+Segmentation-methods'></span><span id='topic+Cells.Segmentation'></span><span id='topic+GetTissueCoordinates.Segmentation'></span><span id='topic+RenameCells.Segmentation'></span><span id='topic+lengths.Segmentation'></span><span id='topic+subset.Segmentation'></span><span id='topic++5B+2CSegmentation+2CANY+2CANY+2CANY-method'></span><span id='topic+coordinates+2CSegmentation-method'></span><span id='topic+show+2CSegmentation-method'></span>

<h3>Description</h3>

<p>Methods for <code><a href="#topic+Segmentation-class">Segmentation</a></code>
objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Segmentation'
Cells(x, ...)

## S3 method for class 'Segmentation'
GetTissueCoordinates(object, full = TRUE, ...)

## S3 method for class 'Segmentation'
RenameCells(object, new.names = NULL, ...)

## S3 method for class 'Segmentation'
lengths(x, use.names = TRUE)

## S3 method for class 'Segmentation'
subset(x, cells = NULL, ...)

## S4 method for signature 'Segmentation,ANY,ANY,ANY'
x[i, j, ..., drop = TRUE]

## S4 method for signature 'Segmentation'
coordinates(obj, full = TRUE, ...)

## S4 method for signature 'Segmentation'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Segmentation-methods_+3A_x">x</code>, <code id="Segmentation-methods_+3A_object">object</code>, <code id="Segmentation-methods_+3A_obj">obj</code></td>
<td>
<p>A
<code><a href="#topic+Segmentation-class">Segmentation</a></code> object</p>
</td></tr>
<tr><td><code id="Segmentation-methods_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Segmentation-methods_+3A_full">full</code></td>
<td>
<p>Expand the coordinates to the full polygon</p>
</td></tr>
<tr><td><code id="Segmentation-methods_+3A_new.names">new.names</code></td>
<td>
<p>vector of new cell names</p>
</td></tr>
<tr><td><code id="Segmentation-methods_+3A_use.names">use.names</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="Segmentation-methods_+3A_i">i</code>, <code id="Segmentation-methods_+3A_cells">cells</code></td>
<td>
<p>A vector of cells to keep; if <code>NULL</code>, defaults
to all cells</p>
</td></tr>
<tr><td><code id="Segmentation-methods_+3A_j">j</code>, <code id="Segmentation-methods_+3A_drop">drop</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Cells</code>: Get cell names
</p>
<p><code>GetTissueCoordinates</code>, <code>coordinates</code>: Get
tissue coordinates
</p>
<p><code>RenameCells</code>: Update cell names
</p>
<p><code>lengths</code>: Generate a run-length encoding of the cells present
</p>
<p><code>subset</code>, <code>[</code>: Subset a <code>Segmentation</code> object to
certain cells
</p>
<p><code>show</code>: Display an object summary to stdout
</p>


<h3>Value</h3>

<p><code>Cells</code>: A vector of cell names
</p>
<p><code>GetTissueCoordinates</code>, <code>coordinates</code>: A data frame with
three columns:
</p>

<ul>
<li> <p>&ldquo;<code>x</code>&rdquo;: the x-coordinate
</p>
</li>
<li> <p>&ldquo;<code>y</code>&rdquo;: the y-coordinate
</p>
</li>
<li> <p>&ldquo;<code>cell</code>&rdquo; or &ldquo;<code>ID</code>&rdquo;: the cell name
</p>
</li></ul>

<p>If <code>full</code> is <code>TRUE</code>, then each coordinate will indicate a vertex
for the cell polygon; otherwise, each coordinate will indicate a centroid
for the cell. Note: <code>GetTissueCoordinates</code> ....
</p>
<p><code>RenameCells</code>: <code>object</code> with the cells renamed to
<code>new.names</code>
</p>
<p><code>lengths</code>: An <code><a href="base.html#topic+rle">rle</a></code> object for the cells
</p>
<p><code>subset</code>, <code>[</code>: <code>x</code> subsetted to the cells specified
by <code>cells</code>/<code>i</code>
</p>
<p><code>show</code>: Invisibly returns <code>NULL</code>
</p>


<h3>Progress Updates with <span class="pkg">progressr</span></h3>

<p>The following methods use
<a href="https://cran.r-project.org/package=progressr"><span class="pkg">progressr</span></a> to
render status updates and progress bars:
</p>

<ul>
<li> <p><code>RenameCells</code>
</p>
</li></ul>

<p>To enable progress updates, wrap
the function call in <code><a href="progressr.html#topic+with_progress">with_progress</a></code> or run
<code><a href="progressr.html#topic+handlers">handlers(global = TRUE)</a></code> before running
this function. For more details about <span class="pkg">progressr</span>, please read
<a href="https://progressr.futureverse.org/articles/progressr-intro.html"><code>vignette("progressr-intro")</code></a>
</p>


<h3>Parallelization with <span class="pkg">future</span></h3>

<p>The following methods use
<a href="https://cran.r-project.org/package=future"><span class="pkg">future</span></a> to enable
parallelization:
</p>

<ul>
<li> <p><code>RenameCells</code>
</p>
</li></ul>

<p>Parallelization strategies can be set using
<code><a href="future.html#topic+plan">plan</a></code>. Common plans include &ldquo;<code>sequential</code>&rdquo;
for non-parallelized processing or &ldquo;<code>multisession</code>&rdquo; for parallel
evaluation using multiple <span class="rlang"><b>R</b></span> sessions; for other plans, see the
&ldquo;Implemented evaluation strategies&rdquo; section of
<code><a href="future.html#topic+plan">?future::plan</a></code>. For a more thorough introduction
to <span class="pkg">future</span>, see
<a href="https://future.futureverse.org/articles/future-1-overview.html"><code>vignette("future-1-overview")</code></a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Segmentation-class">Segmentation-class</a></code>
</p>
<p>Segmentation layer classes:
<code><a href="#topic+Centroids-class">Centroids-class</a></code>,
<code><a href="#topic+Centroids-methods">Centroids-methods</a></code>,
<code><a href="#topic+Molecules-class">Molecules-class</a></code>,
<code><a href="#topic+Molecules-methods">Molecules-methods</a></code>,
<code><a href="#topic+Segmentation-class">Segmentation-class</a></code>
</p>

<hr>
<h2 id='set-if-na'>Set If or If Not <code>NA</code></h2><span id='topic+set-if-na'></span><span id='topic++25NA+25'></span><span id='topic++25na+25'></span><span id='topic++25+21NA+25'></span><span id='topic++25+21na+25'></span>

<h3>Description</h3>

<p>Set a default value depending on if an object is <code><a href="base.html#topic+NA">NA</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>x %NA% y

x %na% y

x %!NA% y

x %!na% y
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set-if-na_+3A_x">x</code></td>
<td>
<p>An object to test</p>
</td></tr>
<tr><td><code id="set-if-na_+3A_y">y</code></td>
<td>
<p>A default value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For <code>%NA%</code>: <code>y</code> if <code>x</code> is <code><a href="base.html#topic+NA">NA</a></code>;
otherwise <code>x</code>
</p>
<p>For <code>%!NA%</code>: <code>y</code> if <code>x</code> is <strong>not</strong>
<code><a href="base.html#topic+NA">NA</a></code>; otherwise <code>x</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set if NA
1 %NA% 2
NA %NA% 2

# Set if *not* NA
1 %!NA% 2
NA %!NA% 2

</code></pre>

<hr>
<h2 id='set-if-null'>Set If or If Not <code>NULL</code></h2><span id='topic+set-if-null'></span><span id='topic++25+7C+7C+25'></span><span id='topic++25iff+25'></span>

<h3>Description</h3>

<p>Set a default value depending on if an object is <code>NULL</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>x %||% y

x %iff% y
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set-if-null_+3A_x">x</code></td>
<td>
<p>An object to test</p>
</td></tr>
<tr><td><code id="set-if-null_+3A_y">y</code></td>
<td>
<p>A default value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For <code>%||%</code>: <code>y</code> if <code>x</code> is <code>NULL</code>;
otherwise <code>x</code>
</p>
<p>For <code>%iff%</code>: <code>y</code> if <code>x</code> is <strong>not</strong>
<code>NULL</code>; otherwise <code>x</code>
</p>


<h3>Author(s)</h3>

<p>For <code>%||%</code>: <span class="pkg">rlang</span> developers
</p>


<h3>See Also</h3>

<p><code><a href="rlang.html#topic+op-null-default">rlang::%||%</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set if NULL
1 %||% 2
NULL %||% 2

# Set if *not* NULL
1 %iff% 2
NULL %iff% 2

</code></pre>

<hr>
<h2 id='seurat-class'>The Seurat Class</h2><span id='topic+seurat-class'></span><span id='topic+oldseurat'></span>

<h3>Description</h3>

<p>The Seurat object is the center of each single cell analysis. It stores all
information associated with the dataset, including data, annotations,
analyses, etc. All that is needed to construct a Seurat object is an
expression matrix (rows are genes, columns are cells), which should
be log-scale
</p>


<h3>Details</h3>

<p>Each Seurat object has a number of slots which store information. Key slots
to access are listed below.
</p>


<h3>Slots</h3>


<dl>
<dt><code>raw.data</code></dt><dd><p>The raw project data</p>
</dd>
<dt><code>data</code></dt><dd><p>The normalized expression matrix (log-scale)</p>
</dd>
<dt><code>scale.data</code></dt><dd><p>scaled (default is z-scoring each gene) expression matrix;
used for dimensional reduction and heatmap visualization</p>
</dd>
<dt><code>var.genes</code></dt><dd><p>Vector of genes exhibiting high variance across single cells</p>
</dd>
<dt><code>is.expr</code></dt><dd><p>Expression threshold to determine if a gene is expressed
(0 by default)</p>
</dd>
<dt><code>ident</code></dt><dd><p>THe 'identity class' for each cell</p>
</dd>
<dt><code>meta.data</code></dt><dd><p>Contains meta-information about each cell, starting with
number of genes detected (nFeature) and the original identity class
(orig.ident); more information is added using <code>AddMetaData</code></p>
</dd>
<dt><code>project.name</code></dt><dd><p>Name of the project (for record keeping)</p>
</dd>
<dt><code>dr</code></dt><dd><p>List of stored dimensional reductions; named by technique</p>
</dd>
<dt><code>assay</code></dt><dd><p>List of additional assays for multimodal analysis; named by
technique</p>
</dd>
<dt><code>hvg.info</code></dt><dd><p>The output of the mean/variability analysis for all genes</p>
</dd>
<dt><code>imputed</code></dt><dd><p>Matrix of imputed gene scores</p>
</dd>
<dt><code>cell.names</code></dt><dd><p>Names of all single cells
(column names of the expression matrix)</p>
</dd>
<dt><code>cluster.tree</code></dt><dd><p>List where the first element is a phylo object containing
the phylogenetic tree relating different identity classes</p>
</dd>
<dt><code>snn</code></dt><dd><p>Spare matrix object representation of the SNN graph</p>
</dd>
<dt><code>calc.params</code></dt><dd><p>Named list to store all calculation-related
parameter choices</p>
</dd>
<dt><code>kmeans</code></dt><dd><p>Stores output of gene-based clustering from <code>DoKMeans</code></p>
</dd>
<dt><code>spatial</code></dt><dd><p>Stores internal data and calculations for spatial mapping of
single cells</p>
</dd>
<dt><code>misc</code></dt><dd><p>Miscellaneous spot to store any data alongside the object
(for example, gene lists)</p>
</dd>
<dt><code>version</code></dt><dd><p>Version of package used in object creation</p>
</dd>
</dl>

<hr>
<h2 id='Seurat-class'>The Seurat Class</h2><span id='topic+Seurat-class'></span><span id='topic+Seurat'></span>

<h3>Description</h3>

<p>The Seurat object is a representation of single-cell expression data for R;
each Seurat object revolves around a set of cells and consists of one or more
<code><a href="#topic+Assay">Assay</a></code> objects, or individual representations of
expression data (eg. RNA-seq, ATAC-seq, etc). These assays can be reduced
from their high-dimensional state to a lower-dimension state and stored as
<code><a href="#topic+DimReduc">DimReduc</a></code> objects. Seurat objects also
store additional metadata, both at the cell and feature level (contained
within individual assays). The object was designed to be as self-contained as
possible, and easily extendable to new methods.
</p>


<h3>Slots</h3>


<dl>
<dt><code>assays</code></dt><dd><p>A list of assays for this project</p>
</dd>
<dt><code>meta.data</code></dt><dd><p>Contains meta-information about each cell, starting with
number of features detected (nFeature) and the original identity class
(orig.ident); more information is added using <code><a href="#topic+AddMetaData">AddMetaData</a></code></p>
</dd>
<dt><code>active.assay</code></dt><dd><p>Name of the active, or default, assay; settable using
<code><a href="#topic+DefaultAssay">DefaultAssay</a></code></p>
</dd>
<dt><code>active.ident</code></dt><dd><p>The active cluster identity for this Seurat object;
settable using <code><a href="#topic+Idents">Idents</a></code></p>
</dd>
<dt><code>graphs</code></dt><dd><p>A list of <code><a href="#topic+Graph">Graph</a></code> objects</p>
</dd>
<dt><code>neighbors</code></dt><dd><p>...</p>
</dd>
<dt><code>reductions</code></dt><dd><p>A list of dimensional reduction objects for this object</p>
</dd>
<dt><code>images</code></dt><dd><p>A list of spatial image objects</p>
</dd>
<dt><code>project.name</code></dt><dd><p>Name of the project</p>
</dd>
<dt><code>misc</code></dt><dd><p>A list of miscellaneous information</p>
</dd>
<dt><code>version</code></dt><dd><p>Version of Seurat this object was built under</p>
</dd>
<dt><code>commands</code></dt><dd><p>A list of logged commands run on this <code>Seurat</code> object</p>
</dd>
<dt><code>tools</code></dt><dd><p>A list of miscellaneous data generated by other tools, should be
filled by developers only using <code><a href="#topic+Tool">Tool</a>&lt;-</code></p>
</dd>
</dl>


<h3>See Also</h3>

<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>

<hr>
<h2 id='Seurat-validity'>Seurat Object Validity</h2><span id='topic+Seurat-validity'></span>

<h3>Description</h3>

<p>Validation of <code>Seurat</code> objects is handled by
<code><a href="methods.html#topic+validObject">validObject</a></code>
</p>


<h3>See Also</h3>

<p><code><a href="methods.html#topic+validObject">validObject</a></code>
</p>
<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>,
<code><a href="#topic+subset.Seurat">subset.Seurat</a>()</code>
</p>

<hr>
<h2 id='SeuratCommand-class'>The <code>SeuratCommand</code> Class</h2><span id='topic+SeuratCommand-class'></span><span id='topic+SeuratCommand'></span>

<h3>Description</h3>

<p>The <code>SeuratCommand</code> is used for logging commands that are run
on a <code>Seurat</code> object; it stores parameters and timestamps
</p>


<h3>Slots</h3>


<dl>
<dt><code>name</code></dt><dd><p>Command name</p>
</dd>
<dt><code>time.stamp</code></dt><dd><p>Timestamp of when command was tun</p>
</dd>
<dt><code>assay.used</code></dt><dd><p>Optional name of assay used to generate
<code>SeuratCommand</code> object</p>
</dd>
<dt><code>call.string</code></dt><dd><p>String of the command call</p>
</dd>
<dt><code>params</code></dt><dd><p>List of parameters used in the command call</p>
</dd>
</dl>


<h3>See Also</h3>

<p>Command log object and interaction methods
<code><a href="#topic++24.SeuratCommand">$.SeuratCommand</a>()</code>,
<code><a href="#topic+.DollarNames.SeuratCommand">.DollarNames.SeuratCommand</a>()</code>,
<code><a href="#topic+LogSeuratCommand">LogSeuratCommand</a>()</code>,
<code><a href="#topic++5B.SeuratCommand">[.SeuratCommand</a>()</code>,
<code><a href="#topic+as.list.SeuratCommand">as.list.SeuratCommand</a>()</code>
</p>

<hr>
<h2 id='SeuratObject-options'><span class="pkg">Seurat</span> Options</h2><span id='topic+SeuratObject-options'></span>

<h3>Description</h3>

<p>Various options used in <span class="pkg">Seurat</span>
</p>


<h3>Package Options</h3>



<h4>Seurat.coords.short_range</h4>

<p>Defaults to
&ldquo;max&rdquo;<br />
Currently set to &ldquo;&rdquo;
</p>



<h4>Seurat.input.sparse_ratio</h4>

<p>Defaults to
&ldquo;0.4&rdquo;<br />
Currently set to &ldquo;&rdquo;
</p>



<h4>Seurat.io.rds.strict</h4>

<p>Defaults to
&ldquo;FALSE&rdquo;<br />
Currently set to &ldquo;&rdquo;
</p>



<h4>Seurat.object.assay.calcn</h4>

<p>Run <code>CalcN</code> when adding assay data to a <code>Seurat</code> object<br />
Defaults to
&ldquo;&rdquo;<br />
Currently set to &ldquo;&rdquo;
</p>



<h4>Seurat.object.assay.version</h4>

<p>Defaults to
&ldquo;v5&rdquo;<br />
Currently set to &ldquo;&rdquo;
</p>



<h4>Seurat.object.assay.v3.missing_layer</h4>

<p>Defaults to
&ldquo;matrix&rdquo;<br />
Currently set to &ldquo;&rdquo;
</p>



<h4>Seurat.object.project</h4>

<p>Default project for new <code><a href="#topic+Seurat">Seurat</a></code> objects<br />
Defaults to
&ldquo;SeuratProject&rdquo;<br />
Currently set to &ldquo;&rdquo;
</p>


<hr>
<h2 id='show+2CAssay-method'>V3 Assay Overview</h2><span id='topic+show+2CAssay-method'></span>

<h3>Description</h3>

<p>Overview of an <code><a href="#topic+Assay">Assay</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Assay'
show(object)
</code></pre>


<h3>Value</h3>

<p>Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and invisibly
returns <code>NULL</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Assay">Assay</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]
rna

</code></pre>

<hr>
<h2 id='show+2CDimReduc-method'>Dimensional Reduction Overview</h2><span id='topic+show+2CDimReduc-method'></span>

<h3>Description</h3>

<p>Overview of a <code><a href="#topic+DimReduc">DimReduc</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'DimReduc'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CDimReduc-method_+3A_object">object</code></td>
<td>
<p>A dimensional reduction</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and invisibly
returns <code>NULL</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+DimReduc">DimReduc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- pbmc_small[["pca"]]
pca

</code></pre>

<hr>
<h2 id='show+2CGraph-method'>Graph Object Overview</h2><span id='topic+show+2CGraph-method'></span>

<h3>Description</h3>

<p>Overview of a <code><a href="#topic+Graph">Graph</a></code> Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Graph'
show(object)
</code></pre>


<h3>Value</h3>

<p>Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and invisibly
returns <code>NULL</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc_small[["RNA_snn"]]

</code></pre>

<hr>
<h2 id='show+2CLogMap-method'><code><a href="#topic+LogMap">LogMap</a></code> Object Overview</h2><span id='topic+show+2CLogMap-method'></span>

<h3>Description</h3>

<p>Overview of a <code><a href="#topic+LogMap">LogMap</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'LogMap'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CLogMap-method_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+LogMap">LogMap</a></code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and invisibly
returns <code>NULL</code>
</p>

<hr>
<h2 id='show+2Cseurat-method'>Old Seurat Object Overview</h2><span id='topic+show+2Cseurat-method'></span>

<h3>Description</h3>

<p>Overview of a <code><a href="#topic+oldseurat">seurat</a></code> object overview
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'seurat'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2Cseurat-method_+3A_object">object</code></td>
<td>
<p>An old seurat object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and invisibly
returns <code>NULL</code>
</p>

<hr>
<h2 id='show+2CSeurat-method'>Seurat Object Overview</h2><span id='topic+show+2CSeurat-method'></span>

<h3>Description</h3>

<p>Overview of a <code><a href="#topic+Seurat">Seurat</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Seurat'
show(object)
</code></pre>


<h3>Value</h3>

<p>Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and invisibly
returns <code>NULL</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pbmc_small

</code></pre>

<hr>
<h2 id='show+2CSeuratCommand-method'>Command Log Overview</h2><span id='topic+show+2CSeuratCommand-method'></span>

<h3>Description</h3>

<p>Overview of a <code><a href="#topic+SeuratCommand">SeuratCommand</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'SeuratCommand'
show(object)
</code></pre>


<h3>Value</h3>

<p>Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and invisibly
returns <code>NULL</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cmd &lt;- pbmc_small[["NormalizeData.RNA"]]
cmd

</code></pre>

<hr>
<h2 id='show+2CStdAssay-method'>V5 Assay Overview</h2><span id='topic+show+2CStdAssay-method'></span>

<h3>Description</h3>

<p>Overview of a <code><a href="#topic+StdAssay">StdAssay</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'StdAssay'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CStdAssay-method_+3A_object">object</code></td>
<td>
<p>A v5 Assay</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and invisibly
returns <code>NULL</code>
</p>


<h3>See Also</h3>

<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='Simplify'>Simplify Geometry</h2><span id='topic+Simplify'></span><span id='topic+Simplify.Spatial'></span>

<h3>Description</h3>

<p>Simplify Geometry
</p>
<p>Simplify segmentations by reducing the number of vertices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Simplify(coords, tol, topologyPreserve = TRUE)

## S3 method for class 'Spatial'
Simplify(coords, tol, topologyPreserve = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Simplify_+3A_coords">coords</code></td>
<td>
<p>A 'Segmentation' object</p>
</td></tr>
<tr><td><code id="Simplify_+3A_tol">tol</code></td>
<td>
<p>Numerical tolerance value to be used by the Douglas-Peuker algorithm</p>
</td></tr>
<tr><td><code id="Simplify_+3A_topologypreserve">topologyPreserve</code></td>
<td>
<p>Logical determining if the algorithm should attempt to preserve the topology of the original geometry</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simplified version of <code>coords</code>
</p>
<p>A 'Segmentation' object with simplified segmentation vertices
</p>

<hr>
<h2 id='SparseEmptyMatrix'>Generate empty dgC sparse matrix</h2><span id='topic+SparseEmptyMatrix'></span>

<h3>Description</h3>

<p>Generate empty dgC sparse matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SparseEmptyMatrix(nrow, ncol, rownames = NULL, colnames = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SparseEmptyMatrix_+3A_ncol">ncol</code>, <code id="SparseEmptyMatrix_+3A_nrow">nrow</code></td>
<td>
<p>Number of columns and rows in matrix</p>
</td></tr>
<tr><td><code id="SparseEmptyMatrix_+3A_rownames">rownames</code>, <code id="SparseEmptyMatrix_+3A_colnames">colnames</code></td>
<td>
<p>Optional row- and column names for the matrix</p>
</td></tr>
</table>

<hr>
<h2 id='SpatialImage-class'>The SpatialImage class</h2><span id='topic+SpatialImage-class'></span><span id='topic+SpatialImage'></span>

<h3>Description</h3>

<p>The <code>SpatialImage</code> class is a virtual class representing spatial
information for Seurat. All spatial image information must inherit from this
class for use with <code>Seurat</code> objects
</p>


<h3>Slots</h3>


<dl>
<dt><code>assay</code></dt><dd><p>Name of assay to associate image data with; will give this image
priority for visualization when the assay is set as the active/default assay
in a <code>Seurat</code> object</p>
</dd>
<dt><code>key</code></dt><dd><p>A one-length character vector with the object's key; keys must
be one or more alphanumeric characters followed by an underscore
&ldquo;<code>_</code>&rdquo; (regex pattern
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;)</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="#topic+SpatialImage-methods">SpatialImage-methods</a></code> for a list of required and
provided methods
</p>

<hr>
<h2 id='SpatialImage-methods'><code>SpatialImage</code> methods</h2><span id='topic+SpatialImage-methods'></span><span id='topic+Cells.SpatialImage'></span><span id='topic+DefaultAssay.SpatialImage'></span><span id='topic+DefaultAssay+3C-.SpatialImage'></span><span id='topic+GetImage.SpatialImage'></span><span id='topic+GetTissueCoordinates.SpatialImage'></span><span id='topic+IsGlobal.SpatialImage'></span><span id='topic+Key.SpatialImage'></span><span id='topic+Key+3C-.SpatialImage'></span><span id='topic+Radius.SpatialImage'></span><span id='topic+RenameCells.SpatialImage'></span><span id='topic++5B.SpatialImage'></span><span id='topic+dim.SpatialImage'></span><span id='topic+subset.SpatialImage'></span><span id='topic+show+2CSpatialImage-method'></span>

<h3>Description</h3>

<p>Methods defined on the <code><a href="#topic+SpatialImage">SpatialImage</a></code> class. Some of these
methods must be overridden in order to ensure proper functionality of the
derived classes (see <strong>Required methods</strong> below). Other methods are
designed to work across all <code>SpatialImage</code>-derived subclasses, and
should only be overridden if necessary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SpatialImage'
Cells(x, ...)

## S3 method for class 'SpatialImage'
DefaultAssay(object, ...)

## S3 replacement method for class 'SpatialImage'
DefaultAssay(object, ...) &lt;- value

## S3 method for class 'SpatialImage'
GetImage(object, mode = c("grob", "raster", "plotly", "raw"), ...)

## S3 method for class 'SpatialImage'
GetTissueCoordinates(object, ...)

## S3 method for class 'SpatialImage'
IsGlobal(object, ...)

## S3 method for class 'SpatialImage'
Key(object, ...)

## S3 replacement method for class 'SpatialImage'
Key(object, ...) &lt;- value

## S3 method for class 'SpatialImage'
Radius(object)

## S3 method for class 'SpatialImage'
RenameCells(object, new.names = NULL, ...)

## S3 method for class 'SpatialImage'
x[i, ...]

## S3 method for class 'SpatialImage'
dim(x)

## S3 method for class 'SpatialImage'
subset(x, cells, ...)

## S4 method for signature 'SpatialImage'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SpatialImage-methods_+3A_x">x</code>, <code id="SpatialImage-methods_+3A_object">object</code></td>
<td>
<p>A <code>SpatialImage</code>-derived object</p>
</td></tr>
<tr><td><code id="SpatialImage-methods_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="SpatialImage-methods_+3A_value">value</code></td>
<td>
<p>Depends on the method:
</p>

<dl>
<dt><code>DefaultAssay&lt;-</code></dt><dd><p>Assay that the image should be
associated with</p>
</dd>
<dt><code>Key&lt;-</code></dt><dd><p>New key for the image</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="SpatialImage-methods_+3A_mode">mode</code></td>
<td>
<p>How to return the image; should accept one of &ldquo;grob&rdquo;,
&ldquo;raster&rdquo;, &ldquo;plotly&rdquo;, or &ldquo;raw&rdquo;</p>
</td></tr>
<tr><td><code id="SpatialImage-methods_+3A_new.names">new.names</code></td>
<td>
<p>vector of new cell names</p>
</td></tr>
<tr><td><code id="SpatialImage-methods_+3A_i">i</code>, <code id="SpatialImage-methods_+3A_cells">cells</code></td>
<td>
<p>A vector of cells to keep</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong>[Override]</strong> <code>Cells</code>: should return cell names
</p>
<p><code>DefaultAssay</code>: The associated assay of a
<code>SpatialImage</code>-derived object
</p>
<p><code>DefaultAssay&lt;-</code>: <code>object</code> with the associated assay
updated
</p>
<p><strong>[Override]</strong> <code>GetImage</code>: The image data from a
<code>SpatialImage</code>-derived object
</p>
<p><strong>[Override]</strong> <code>GetTissueCoordinates</code>: ...
</p>
<p><code>IsGlobal</code>: returns <code>TRUE</code> as images are, by default,
global
</p>
<p><code>Key</code>: The key for a <code>SpatialImage</code>-derived object
</p>
<p><code>Key&lt;-</code>: <code>object</code> with the key set to <code>value</code>
</p>
<p><code>Radius</code>: The spot radius size; by default, returns <code>NULL</code>
</p>
<p><strong>[Override]</strong> <code>RenameCells</code>: <code>object</code> with the new
cell names
</p>
<p><code>[</code>, <code>subset</code>: <code>x</code>/<code>object</code> for only the cells
requested
</p>
<p><strong>[Override]</strong> <code>dim</code>: The dimensions of the image data in
(Y, X) format
</p>
<p><code>show</code>: Prints summary to <code><a href="base.html#topic+stdout">stdout</a></code> and
invisibly returns <code>NULL</code>
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>Cells(SpatialImage)</code>: Get the cell names from an image
(<strong>[Override]</strong>)
</p>
</li>
<li> <p><code>DefaultAssay(SpatialImage)</code>: Get the associated assay of a
<code>SpatialImage</code>-derived object
</p>
</li>
<li> <p><code>DefaultAssay(SpatialImage) &lt;- value</code>: Set the associated assay of a
<code>SpatialImage</code>-derived object
</p>
</li>
<li> <p><code>GetImage(SpatialImage)</code>: Get the image data from a
<code>SpatialImage</code>-derived object
</p>
</li>
<li> <p><code>GetTissueCoordinates(SpatialImage)</code>: Get tissue coordinates for a
<code>SpatialImage</code>-derived object (<strong>[Override]</strong>)
</p>
</li>
<li> <p><code>IsGlobal(SpatialImage)</code>: Globality test for
<code>SpatialImage</code>-derived object
</p>
</li>
<li> <p><code>Key(SpatialImage)</code>: Get the key for a
<code>SpatialImage</code>-derived object
</p>
</li>
<li> <p><code>Key(SpatialImage) &lt;- value</code>: Set the key for a
<code>SpatialImage</code>-derived object
</p>
</li>
<li> <p><code>Radius(SpatialImage)</code>: Get the spot radius size
</p>
</li>
<li> <p><code>RenameCells(SpatialImage)</code>: Rename cells in a
<code>SpatialImage</code>-derived object (<strong>[Override]</strong>)
</p>
</li>
<li> <p><code>[</code>: Subset a <code>SpatialImage</code>-derived object
</p>
</li>
<li> <p><code>dim(SpatialImage)</code>: Get the plotting dimensions of an image
(<strong>[Override]</strong>)
</p>
</li>
<li> <p><code>subset(SpatialImage)</code>: Subset a <code>SpatialImage</code>-derived object
(<strong>[Override]</strong>)
</p>
</li>
<li> <p><code>show(SpatialImage)</code>: Overview of a <code>SpatialImage</code>-derived
object
</p>
</li></ul>


<h3>Provided methods</h3>

<p>These methods are defined on the <code>SpatialImage</code> object and should not
be overridden without careful thought
</p>

<ul>
<li> <p><code><a href="#topic+DefaultAssay">DefaultAssay</a></code> and <code><a href="#topic+DefaultAssay+3C-">DefaultAssay&lt;-</a></code>
</p>
</li>
<li> <p><code><a href="#topic+Key">Key</a></code> and <code><a href="#topic+Key+3C-">Key&lt;-</a></code>
</p>
</li>
<li> <p><code><a href="#topic+GetImage">GetImage</a></code>; this method <em>can</em> be overridden to
provide image data, normally returns empty image data. If overridden,
should default to returning a  <code><a href="grid.html#topic+grob">grob</a></code> object
</p>
</li>
<li> <p><code><a href="#topic+IsGlobal">IsGlobal</a></code>
</p>
</li>
<li> <p><code><a href="#topic+Radius">Radius</a></code>; this method <em>can</em> be overridden to
provide a spot radius for image objects
</p>
</li>
<li> <p><code><a href="base.html#topic+Extract">[</a></code>; this method <em>can</em> be overridden
to change default subset behavior, normally returns
<code>subset(x = x, cells = i)</code>. If overridden, should only accept <code>i</code>
</p>
</li></ul>



<h3>Required methods</h3>

<p>All subclasses of the <code>SpatialImage</code> class must define the following
methods; simply relying on the <code>SpatialImage</code> method will result in
errors. For required parameters and their values, see the <code>Usage</code> and
<code>Arguments</code> sections
</p>

<dl>
<dt><code><a href="#topic+Cells">Cells</a></code></dt><dd>
<p>Return the cell/spot barcodes associated with each position
</p>
</dd>
<dt><code><a href="base.html#topic+dim">dim</a></code></dt><dd>
<p>Return the dimensions of the image for plotting in <code>(Y, X)</code> format
</p>
</dd>
<dt><code><a href="#topic+GetTissueCoordinates">GetTissueCoordinates</a></code></dt><dd>
<p>Return tissue coordinates; by default, must return a two-column
<code>data.frame</code> with x-coordinates in the first column and
y-coordinates in the second
</p>
</dd>
<dt><code><a href="#topic+Radius">Radius</a></code></dt><dd>
<p>Return the spot radius; returns <code>NULL</code> by default for use with
non-spot image technologies
</p>
</dd>
<dt><code><a href="#topic+RenameCells">RenameCells</a></code></dt><dd>
<p>Rename the cell/spot barcodes for this image
</p>
</dd>
<dt><code><a href="#topic+subset">subset</a></code></dt><dd>
<p>Subset the image data by cells/spots
</p>
</dd>
</dl>

<p>These methods are used throughout Seurat, so defining them and setting the
proper defaults will allow subclasses of <code>SpatialImage</code> to work
seamlessly
</p>


<h3>See Also</h3>

<p><code><a href="#topic+DefaultAssay">DefaultAssay</a></code>
</p>
<p><code><a href="#topic+GetImage">GetImage</a></code>
</p>
<p><code><a href="#topic+GetTissueCoordinates">GetTissueCoordinates</a></code>
</p>
<p><code><a href="#topic+IsGlobal">IsGlobal</a></code>
</p>
<p><code><a href="#topic+Key">Key</a></code>
</p>
<p><code><a href="#topic+RenameCells">RenameCells</a></code>
</p>

<hr>
<h2 id='split.Assay'>Split an Assay</h2><span id='topic+split.Assay'></span>

<h3>Description</h3>

<p>Split an Assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
split(x, f, drop = FALSE, layers = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id="split.Assay_+3A_f">f</code></td>
<td>
<p>a &lsquo;factor&rsquo; in the sense that <code><a href="base.html#topic+as.factor">as.factor</a>(f)</code>
defines the grouping, or a list of such factors in which case their
interaction is used for the grouping. If <code>x</code> is a data frame,
<code>f</code> can also be a formula of the form <code> ~ g</code> to split by
the variable <code>g</code>, or more generally of the form <code> ~ g1 +
      ... + gk</code> to split by the interaction of the variables
<code>g1</code>, ..., <code>gk</code>, where these variables are evaluated in
the data frame <code>x</code> using the usual non-standard evaluation
rules.</p>
</td></tr>
<tr><td><code id="split.Assay_+3A_drop">drop</code></td>
<td>
<p>logical indicating if levels that do not occur should be dropped
(if <code>f</code> is a <code>factor</code> or a list).</p>
</td></tr>
<tr><td><code id="split.Assay_+3A_layers">layers</code></td>
<td>
<p>Names of layers to include in the split; pass <code>NA</code> for
all layers; pass <code>NULL</code> for the <a href="#topic+DefaultLayer">default layer</a></p>
</td></tr>
<tr><td><code id="split.Assay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a v5 assay with splitted layers
</p>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+subset.Assay">subset.Assay</a>()</code>
</p>

<hr>
<h2 id='split.Assay5'>Split an Assay</h2><span id='topic+split.Assay5'></span>

<h3>Description</h3>

<p>Split an Assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
split(
  x,
  f,
  drop = FALSE,
  layers = c("counts", "data"),
  ret = c("assay", "multiassays", "layers"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="split.Assay5_+3A_f">f</code></td>
<td>
<p>a &lsquo;factor&rsquo; in the sense that <code><a href="base.html#topic+as.factor">as.factor</a>(f)</code>
defines the grouping, or a list of such factors in which case their
interaction is used for the grouping. If <code>x</code> is a data frame,
<code>f</code> can also be a formula of the form <code> ~ g</code> to split by
the variable <code>g</code>, or more generally of the form <code> ~ g1 +
      ... + gk</code> to split by the interaction of the variables
<code>g1</code>, ..., <code>gk</code>, where these variables are evaluated in
the data frame <code>x</code> using the usual non-standard evaluation
rules.</p>
</td></tr>
<tr><td><code id="split.Assay5_+3A_drop">drop</code></td>
<td>
<p>logical indicating if levels that do not occur should be dropped
(if <code>f</code> is a <code>factor</code> or a list).</p>
</td></tr>
<tr><td><code id="split.Assay5_+3A_layers">layers</code></td>
<td>
<p>Names of layers to include in the split; pass <code>NA</code> for
all layers; pass <code>NULL</code> for the <a href="#topic+DefaultLayer">default layer</a></p>
</td></tr>
<tr><td><code id="split.Assay5_+3A_ret">ret</code></td>
<td>
<p>Type of return value; choose from:
</p>

<ul>
<li> <p>&ldquo;<code>assay</code>&rdquo;: a single <code><a href="#topic+Assay5">Assay5</a></code> object
</p>
</li>
<li> <p>&ldquo;<code>multiassay</code>&rdquo;: a list of <code><a href="#topic+Assay5">Assay5</a></code> objects
</p>
</li>
<li> <p>&ldquo;<code>layers</code>&rdquo;: a list of layer matrices
</p>
</li></ul>
</td></tr>
<tr><td><code id="split.Assay5_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depends on the value of <code>ret</code>:
</p>

<ul>
<li> <p>&ldquo;<code>assay</code>&rdquo;: <code>x</code> with the layers requested in
<code>layers</code> split based on <code>f</code>; all other layers are left as-is
</p>
</li>
<li> <p>&ldquo;<code>multiassay</code>&rdquo;: a list of <code><a href="#topic+Assay5">Assay5</a></code> objects;
the list contains one value per split and each assay contains only the
layers requested in <code>layers</code> with the <a href="#topic+Key">key</a> set to the split
</p>
</li>
<li> <p>&ldquo;<code>layers</code>&rdquo;: a list of matrices of length
<code>length(assays) * length(unique(f))</code>; the list is named as
&ldquo;<code>layer.split</code>&rdquo;
</p>
</li></ul>



<h3>Progress Updates with <span class="pkg">progressr</span></h3>

<p>This function uses
<a href="https://cran.r-project.org/package=progressr"><span class="pkg">progressr</span></a> to
render status updates and progress bars. To enable progress updates, wrap
the function call in <code><a href="progressr.html#topic+with_progress">with_progress</a></code> or run
<code><a href="progressr.html#topic+handlers">handlers(global = TRUE)</a></code> before running
this function. For more details about <span class="pkg">progressr</span>, please read
<a href="https://progressr.futureverse.org/articles/progressr-intro.html"><code>vignette("progressr-intro")</code></a>
</p>


<h3>See Also</h3>

<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+subset.Assay5">subset.Assay5</a>()</code>
</p>

<hr>
<h2 id='split.Seurat'>Split an Assay</h2><span id='topic+split.Seurat'></span>

<h3>Description</h3>

<p>Split an Assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
split(x, f, drop = FALSE, assay = NULL, layers = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split.Seurat_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="split.Seurat_+3A_f">f</code></td>
<td>
<p>a &lsquo;factor&rsquo; in the sense that <code><a href="base.html#topic+as.factor">as.factor</a>(f)</code>
defines the grouping, or a list of such factors in which case their
interaction is used for the grouping. If <code>x</code> is a data frame,
<code>f</code> can also be a formula of the form <code> ~ g</code> to split by
the variable <code>g</code>, or more generally of the form <code> ~ g1 +
      ... + gk</code> to split by the interaction of the variables
<code>g1</code>, ..., <code>gk</code>, where these variables are evaluated in
the data frame <code>x</code> using the usual non-standard evaluation
rules.</p>
</td></tr>
<tr><td><code id="split.Seurat_+3A_drop">drop</code></td>
<td>
<p>logical indicating if levels that do not occur should be dropped
(if <code>f</code> is a <code>factor</code> or a list).</p>
</td></tr>
<tr><td><code id="split.Seurat_+3A_layers">layers</code></td>
<td>
<p>Names of layers to include in the split; pass <code>NA</code> for
all layers; pass <code>NULL</code> for the <a href="#topic+DefaultLayer">default layer</a></p>
</td></tr>
<tr><td><code id="split.Seurat_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depends on the value of <code>ret</code>:
</p>

<ul>
<li> <p>&ldquo;<code>assay</code>&rdquo;: <code>x</code> with the layers requested in
<code>layers</code> split based on <code>f</code>; all other layers are left as-is
</p>
</li>
<li> <p>&ldquo;<code>multiassay</code>&rdquo;: a list of <code><a href="#topic+Assay5">Assay5</a></code> objects;
the list contains one value per split and each assay contains only the
layers requested in <code>layers</code> with the <a href="#topic+Key">key</a> set to the split
</p>
</li>
<li> <p>&ldquo;<code>layers</code>&rdquo;: a list of matrices of length
<code>length(assays) * length(unique(f))</code>; the list is named as
&ldquo;<code>layer.split</code>&rdquo;
</p>
</li></ul>



<h3>Progress Updates with <span class="pkg">progressr</span></h3>

<p>This function uses
<a href="https://cran.r-project.org/package=progressr"><span class="pkg">progressr</span></a> to
render status updates and progress bars. To enable progress updates, wrap
the function call in <code><a href="progressr.html#topic+with_progress">with_progress</a></code> or run
<code><a href="progressr.html#topic+handlers">handlers(global = TRUE)</a></code> before running
this function. For more details about <span class="pkg">progressr</span>, please read
<a href="https://progressr.futureverse.org/articles/progressr-intro.html"><code>vignette("progressr-intro")</code></a>
</p>

<hr>
<h2 id='split.StdAssay'>Split an Assay</h2><span id='topic+split.StdAssay'></span><span id='topic+split+2CStdAssay-method'></span>

<h3>Description</h3>

<p>Split an Assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
split(
  x,
  f,
  drop = FALSE,
  layers = c("counts", "data"),
  ret = c("assay", "multiassays", "layers"),
  ...
)

## S4 method for signature 'StdAssay'
split(
  x,
  f,
  drop = FALSE,
  layers = c("counts", "data"),
  ret = c("assay", "multiassays", "layers"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="split.StdAssay_+3A_f">f</code></td>
<td>
<p>a &lsquo;factor&rsquo; in the sense that <code><a href="base.html#topic+as.factor">as.factor</a>(f)</code>
defines the grouping, or a list of such factors in which case their
interaction is used for the grouping. If <code>x</code> is a data frame,
<code>f</code> can also be a formula of the form <code> ~ g</code> to split by
the variable <code>g</code>, or more generally of the form <code> ~ g1 +
      ... + gk</code> to split by the interaction of the variables
<code>g1</code>, ..., <code>gk</code>, where these variables are evaluated in
the data frame <code>x</code> using the usual non-standard evaluation
rules.</p>
</td></tr>
<tr><td><code id="split.StdAssay_+3A_drop">drop</code></td>
<td>
<p>logical indicating if levels that do not occur should be dropped
(if <code>f</code> is a <code>factor</code> or a list).</p>
</td></tr>
<tr><td><code id="split.StdAssay_+3A_layers">layers</code></td>
<td>
<p>Names of layers to include in the split; pass <code>NA</code> for
all layers; pass <code>NULL</code> for the <a href="#topic+DefaultLayer">default layer</a></p>
</td></tr>
<tr><td><code id="split.StdAssay_+3A_ret">ret</code></td>
<td>
<p>Type of return value; choose from:
</p>

<ul>
<li> <p>&ldquo;<code>assay</code>&rdquo;: a single <code><a href="#topic+Assay5">Assay5</a></code> object
</p>
</li>
<li> <p>&ldquo;<code>multiassay</code>&rdquo;: a list of <code><a href="#topic+Assay5">Assay5</a></code> objects
</p>
</li>
<li> <p>&ldquo;<code>layers</code>&rdquo;: a list of layer matrices
</p>
</li></ul>
</td></tr>
<tr><td><code id="split.StdAssay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depends on the value of <code>ret</code>:
</p>

<ul>
<li> <p>&ldquo;<code>assay</code>&rdquo;: <code>x</code> with the layers requested in
<code>layers</code> split based on <code>f</code>; all other layers are left as-is
</p>
</li>
<li> <p>&ldquo;<code>multiassay</code>&rdquo;: a list of <code><a href="#topic+Assay5">Assay5</a></code> objects;
the list contains one value per split and each assay contains only the
layers requested in <code>layers</code> with the <a href="#topic+Key">key</a> set to the split
</p>
</li>
<li> <p>&ldquo;<code>layers</code>&rdquo;: a list of matrices of length
<code>length(assays) * length(unique(f))</code>; the list is named as
&ldquo;<code>layer.split</code>&rdquo;
</p>
</li></ul>



<h3>Progress Updates with <span class="pkg">progressr</span></h3>

<p>This function uses
<a href="https://cran.r-project.org/package=progressr"><span class="pkg">progressr</span></a> to
render status updates and progress bars. To enable progress updates, wrap
the function call in <code><a href="progressr.html#topic+with_progress">with_progress</a></code> or run
<code><a href="progressr.html#topic+handlers">handlers(global = TRUE)</a></code> before running
this function. For more details about <span class="pkg">progressr</span>, please read
<a href="https://progressr.futureverse.org/articles/progressr-intro.html"><code>vignette("progressr-intro")</code></a>
</p>


<h3>See Also</h3>

<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='StdAssay-class'>Core Assay Infrastructure</h2><span id='topic+StdAssay-class'></span><span id='topic+StdAssay'></span>

<h3>Description</h3>

<p>The <code>StdAssay</code> class is a virtual class that provides core
infrastructure for assay data in <span class="pkg">Seurat</span>. Assays contain expression
data (layers) and associated feature-level meta data. Derived classes
(eg. <a href="#topic+Assay5">the v5 Assay</a>) may optionally
define additional functionality
</p>


<h3>Slots</h3>


<dl>
<dt><code>layers</code></dt><dd><p>A named list containing expression matrices; each matrix should
be a two-dimensional object containing some subset of cells and features
defined in the <code>cells</code> and <code>features</code> slots. Cell and feature
membership is recorded in the <code>cells</code> and <code>features</code> slots,
respectively</p>
</dd>
<dt><code>cells</code></dt><dd><p>A <a href="#topic+LogMap">logical mapping</a> of cell names
and layer membership; this map contains all the possible cells that this
assay can contain. New layers must have some subset of cells present
in this map</p>
</dd>
<dt><code>features</code></dt><dd><p>A <a href="#topic+LogMap">logical mapping</a> of feature
names and layer membership; this map contains all the possible features that
this assay can contain. New layers must have some subset of features
present in this map</p>
</dd>
<dt><code>default</code></dt><dd><p>A one-length integer with the end index of the
<a href="#topic+DefaultLayer">default layer</a>; the default layer be
all layers up to and including the layer at index <code>default</code></p>
</dd>
<dt><code>assay.orig</code></dt><dd><p>Original assay that this assay is based off of;
used to track assay provenance</p>
</dd>
<dt><code>meta.data</code></dt><dd><p>A <a href="base.html#topic+data.frame">data frame</a> with feature-level
meta data; should have the same number of rows as <code>features</code></p>
</dd>
<dt><code>misc</code></dt><dd><p>A named list of unstructured miscellaneous data</p>
</dd>
<dt><code>key</code></dt><dd><p>A one-length character vector with the object's key; keys must
be one or more alphanumeric characters followed by an underscore
&ldquo;<code>_</code>&rdquo; (regex pattern
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;)</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="#topic+Assay5-class">Assay5-class</a></code> <code><a href="#topic+Assay5T-class">Assay5T-class</a></code>
</p>
<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='StdAssay-validity'>V5 Assay Validity</h2><span id='topic+StdAssay-validity'></span>

<h3>Description</h3>

<p>Validation of <code>StdAssay</code> objects is handled by
<code><a href="methods.html#topic+validObject">validObject</a></code>
</p>


<h3>Layer Validation</h3>

<p>blah
</p>


<h3>Key Validation</h3>

<p>Keys must be a one-length character vector; a key must be composed of one
of the following:
</p>

<ul>
<li><p> An empty string (eg. &ldquo;<code>''</code>&rdquo;) where <code>nzchar() == 0</code>
</p>
</li>
<li><p> An string composed of one or more alphanumeric values
(both lower- and upper-case) that ends with an underscore
(&ldquo;<code>_</code>&rdquo;); the first character must be a letter
</p>
</li></ul>

<p>Keys that are not empty strings are validated with the regex
&ldquo;<code>^[a-zA-Z][a-zA-Z0-9]*_$</code>&rdquo;
</p>


<h3>See Also</h3>

<p><code><a href="methods.html#topic+validObject">validObject</a></code>
</p>
<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>,
<code><a href="#topic+subset.StdAssay">subset.StdAssay</a>()</code>
</p>

<hr>
<h2 id='Stdev'>Get the standard deviations for an object</h2><span id='topic+Stdev'></span><span id='topic+Stdev.DimReduc'></span><span id='topic+Stdev.Seurat'></span>

<h3>Description</h3>

<p>Get the standard deviations for an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Stdev(object, ...)

## S3 method for class 'DimReduc'
Stdev(object, ...)

## S3 method for class 'Seurat'
Stdev(object, reduction = "pca", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Stdev_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Stdev_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Stdev_+3A_reduction">reduction</code></td>
<td>
<p>Name of reduction to use</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The standard deviations
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the standard deviations for each PC from the DimReduc object
Stdev(object = pbmc_small[["pca"]])

# Get the standard deviations for each PC from the Seurat object
Stdev(object = pbmc_small, reduction = "pca")

</code></pre>

<hr>
<h2 id='StitchMatrix'>Stitch Matrices Together</h2><span id='topic+StitchMatrix'></span>

<h3>Description</h3>

<p>Stitch Matrices Together
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StitchMatrix(x, y, rowmap, colmap, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="StitchMatrix_+3A_x">x</code></td>
<td>
<p>A matrix</p>
</td></tr>
<tr><td><code id="StitchMatrix_+3A_y">y</code></td>
<td>
<p>One or more matrices of the same class or coercible to the
same class as <code>x</code></p>
</td></tr>
<tr><td><code id="StitchMatrix_+3A_rowmap">rowmap</code>, <code id="StitchMatrix_+3A_colmap">colmap</code></td>
<td>
<p><code><a href="#topic+LogMap">LogMap</a>s</code> describing the row and cell
membership of each matrix; the <code>LogMap</code> entries are assumed to be in
the order of <code>c(x, y)</code></p>
</td></tr>
<tr><td><code id="StitchMatrix_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single matrix of type <code>class(x)</code> consisting of all values
in component matrices
</p>

<hr>
<h2 id='subset.Assay'>Subset an Assay</h2><span id='topic+subset.Assay'></span>

<h3>Description</h3>

<p>Subset an Assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay'
subset(x, cells = NULL, features = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset.Assay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay">Assay</a></code> object</p>
</td></tr>
<tr><td><code id="subset.Assay_+3A_cells">cells</code></td>
<td>
<p>Cell names</p>
</td></tr>
<tr><td><code id="subset.Assay_+3A_features">features</code></td>
<td>
<p>Feature names</p>
</td></tr>
<tr><td><code id="subset.Assay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with just the cells and features specified by
<code>cells</code> and <code>features</code>
</p>


<h3>See Also</h3>

<p>v3 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay">$.Assay</a>()</code>,
<code><a href="#topic+Assay-class">Assay-class</a></code>,
<code><a href="#topic+Assay-validity">Assay-validity</a></code>,
<code><a href="#topic+CreateAssayObject">CreateAssayObject</a>()</code>,
<code><a href="#topic++5B.Assay">[.Assay</a>()</code>,
<code><a href="#topic++5B+5B.Assay">[[.Assay</a>()</code>,
<code><a href="#topic+dim.Assay">dim.Assay</a>()</code>,
<code><a href="#topic+dimnames.Assay">dimnames.Assay</a>()</code>,
<code><a href="#topic+merge.Assay">merge.Assay</a>()</code>,
<code><a href="#topic+split.Assay">split.Assay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rna &lt;- pbmc_small[["RNA"]]
rna2 &lt;- subset(rna, features = VariableFeatures(rna))
rna2

</code></pre>

<hr>
<h2 id='subset.Assay5'>Subset an Assay</h2><span id='topic+subset.Assay5'></span>

<h3>Description</h3>

<p>Subset an Assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Assay5'
subset(x, cells = NULL, features = NULL, layers = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset.Assay5_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="subset.Assay5_+3A_cells">cells</code></td>
<td>
<p>Cell names</p>
</td></tr>
<tr><td><code id="subset.Assay5_+3A_features">features</code></td>
<td>
<p>Feature names</p>
</td></tr>
<tr><td><code id="subset.Assay5_+3A_layers">layers</code></td>
<td>
<p>Layer to keep; defaults to all layers</p>
</td></tr>
<tr><td><code id="subset.Assay5_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with just the cells and features specified by
<code>cells</code> and <code>features</code> for the layers specified by <code>layers</code>
</p>


<h3>See Also</h3>

<p>v5 Assay object, validity, and interaction methods:
<code><a href="#topic++24.Assay5">$.Assay5</a>()</code>,
<code><a href="#topic+Assay5-class">Assay5-class</a></code>,
<code><a href="#topic+Assay5-validity">Assay5-validity</a></code>,
<code><a href="#topic++5B.Assay5">[.Assay5</a>()</code>,
<code><a href="#topic++5B+5B.Assay5">[[.Assay5</a>()</code>,
<code><a href="#topic+dim.Assay5">dim.Assay5</a>()</code>,
<code><a href="#topic+dimnames.Assay5">dimnames.Assay5</a>()</code>,
<code><a href="#topic+merge.Assay5">merge.Assay5</a>()</code>,
<code><a href="#topic+split.Assay5">split.Assay5</a>()</code>
</p>

<hr>
<h2 id='subset.DimReduc'>Subset a Dimensional Reduction</h2><span id='topic+subset.DimReduc'></span>

<h3>Description</h3>

<p>Subset a <code><a href="#topic+DimReduc">DimReduc</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DimReduc'
subset(x, cells = NULL, features = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset.DimReduc_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+DimReduc">DimReduc</a></code> object</p>
</td></tr>
<tr><td><code id="subset.DimReduc_+3A_cells">cells</code>, <code id="subset.DimReduc_+3A_features">features</code></td>
<td>
<p>Cells and features to keep during the subset</p>
</td></tr>
<tr><td><code id="subset.DimReduc_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> for cells <code>cells</code> and features <code>features</code>
</p>


<h3>See Also</h3>

<p>Dimensional reduction object, validity, and interaction methods
<code><a href="#topic+CreateDimReducObject">CreateDimReducObject</a>()</code>,
<code><a href="#topic+DimReduc-class">DimReduc-class</a></code>,
<code><a href="#topic+DimReduc-validity">DimReduc-validity</a></code>,
<code><a href="#topic++5B.DimReduc">[.DimReduc</a>()</code>,
<code><a href="#topic++5B+5B.DimReduc">[[.DimReduc</a>()</code>,
<code><a href="#topic+dim.DimReduc">dim.DimReduc</a>()</code>,
<code><a href="#topic+merge.DimReduc">merge.DimReduc</a>()</code>,
<code><a href="#topic+print.DimReduc">print.DimReduc</a>()</code>
</p>

<hr>
<h2 id='subset.Seurat'>Subset <code>Seurat</code> Objects</h2><span id='topic+subset.Seurat'></span><span id='topic+subset'></span><span id='topic++5B.Seurat'></span>

<h3>Description</h3>

<p>Subset <code>Seurat</code> Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Seurat'
subset(
  x,
  subset,
  cells = NULL,
  features = NULL,
  idents = NULL,
  return.null = FALSE,
  ...
)

## S3 method for class 'Seurat'
x[i, j, ...]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset.Seurat_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+Seurat">Seurat</a></code> object</p>
</td></tr>
<tr><td><code id="subset.Seurat_+3A_subset">subset</code></td>
<td>
<p>Logical expression indicating features/variables to keep</p>
</td></tr>
<tr><td><code id="subset.Seurat_+3A_cells">cells</code>, <code id="subset.Seurat_+3A_j">j</code></td>
<td>
<p>A vector of cell names or indices to keep</p>
</td></tr>
<tr><td><code id="subset.Seurat_+3A_features">features</code>, <code id="subset.Seurat_+3A_i">i</code></td>
<td>
<p>A vector of feature names or indices to keep</p>
</td></tr>
<tr><td><code id="subset.Seurat_+3A_idents">idents</code></td>
<td>
<p>A vector of identity classes to keep</p>
</td></tr>
<tr><td><code id="subset.Seurat_+3A_return.null">return.null</code></td>
<td>
<p>If no cells are requested, return a <code>NULL</code>;
by default, throws an error</p>
</td></tr>
<tr><td><code id="subset.Seurat_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+WhichCells">WhichCells</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>subset</code>: A subsetted <code>Seurat</code> object
</p>
<p><code>[</code>: object <code>x</code> with features <code>i</code> and cells <code>j</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+WhichCells">WhichCells</a></code>
</p>
<p>Seurat object, validity, and interaction methods
<code><a href="#topic++24.Seurat">$.Seurat</a>()</code>,
<code><a href="#topic+Seurat-class">Seurat-class</a></code>,
<code><a href="#topic+Seurat-validity">Seurat-validity</a></code>,
<code><a href="#topic++5B+5B.Seurat">[[.Seurat</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat+2CNULL">[[&lt;-,Seurat,NULL</a></code>,
<code><a href="#topic++5B+5B+3C-+2CSeurat">[[&lt;-,Seurat</a></code>,
<code><a href="#topic+dim.Seurat">dim.Seurat</a>()</code>,
<code><a href="#topic+dimnames.Seurat">dimnames.Seurat</a>()</code>,
<code><a href="#topic+merge.Seurat">merge.Seurat</a>()</code>,
<code><a href="#topic+names.Seurat">names.Seurat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># `subset` examples
subset(pbmc_small, subset = MS4A1 &gt; 4)
subset(pbmc_small, subset = `DLGAP1-AS1` &gt; 2)
subset(pbmc_small, idents = '0', invert = TRUE)
subset(pbmc_small, subset = MS4A1 &gt; 3, slot = 'counts')
subset(pbmc_small, features = VariableFeatures(object = pbmc_small))

# `[` examples
pbmc_small[VariableFeatures(object = pbmc_small), ]
pbmc_small[, 1:10]

</code></pre>

<hr>
<h2 id='subset.StdAssay'>Subset an Assay</h2><span id='topic+subset.StdAssay'></span>

<h3>Description</h3>

<p>Subset an Assay
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
subset(x, cells = NULL, features = NULL, layers = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset.StdAssay_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+Assay5">Assay5</a></code> object</p>
</td></tr>
<tr><td><code id="subset.StdAssay_+3A_cells">cells</code></td>
<td>
<p>Cell names</p>
</td></tr>
<tr><td><code id="subset.StdAssay_+3A_features">features</code></td>
<td>
<p>Feature names</p>
</td></tr>
<tr><td><code id="subset.StdAssay_+3A_layers">layers</code></td>
<td>
<p>Layer to keep; defaults to all layers</p>
</td></tr>
<tr><td><code id="subset.StdAssay_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>x</code> with just the cells and features specified by
<code>cells</code> and <code>features</code> for the layers specified by <code>layers</code>
</p>


<h3>See Also</h3>

<p>v5 Standard Assay object, validity, and interaction methods
<code><a href="#topic++24.StdAssay">$.StdAssay</a>()</code>,
<code><a href="#topic+.DollarNames.StdAssay">.DollarNames.StdAssay</a>()</code>,
<code><a href="#topic+StdAssay-class">StdAssay-class</a></code>,
<code><a href="#topic+StdAssay-validity">StdAssay-validity</a></code>,
<code><a href="#topic++5B.StdAssay">[.StdAssay</a>()</code>,
<code><a href="#topic++5B+5B.StdAssay">[[.StdAssay</a>()</code>,
<code><a href="#topic+dim.StdAssay">dim.StdAssay</a>()</code>,
<code><a href="#topic+dimnames.StdAssay">dimnames.StdAssay</a>()</code>,
<code><a href="#topic+show+2CStdAssay-method">show,StdAssay-method</a></code>,
<code><a href="#topic+split.StdAssay">split.StdAssay</a>()</code>
</p>

<hr>
<h2 id='Theta'>Get the offset angle</h2><span id='topic+Theta'></span>

<h3>Description</h3>

<p>Get the offset angle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Theta(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Theta_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
</table>

<hr>
<h2 id='Tool'>Get and Set Additional Tool Data</h2><span id='topic+Tool'></span><span id='topic+Tools'></span><span id='topic+Tool+3C-'></span><span id='topic+Tool.Seurat'></span><span id='topic+Tool+3C-.Seurat'></span>

<h3>Description</h3>

<p>Use <code>Tool</code> to get tool data. If no additional arguments are provided,
will return a vector with the names of tools in the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tool(object, ...)

Tool(object, ...) &lt;- value

## S3 method for class 'Seurat'
Tool(object, slot = NULL, ...)

## S3 replacement method for class 'Seurat'
Tool(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Tool_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Tool_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="Tool_+3A_value">value</code></td>
<td>
<p>Information to be added to tool list</p>
</td></tr>
<tr><td><code id="Tool_+3A_slot">slot</code></td>
<td>
<p>Name of tool to pull</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If no additional arguments, returns the names of the tools in the
object; otherwise returns the data placed by the tool requested
</p>


<h3>Note</h3>

<p>For developers: set tool data using <code>Tool&lt;-</code>. <code>Tool&lt;-</code> will
automatically set the name of the tool to the function that called
<code>Tool&lt;-</code>, so each function gets one entry in the tools list and cannot
overwrite another function's entry. The automatic naming will also remove any
method identifiers (eg. <code>RunPCA.Seurat</code> will become <code>RunPCA</code>);
please plan accordingly
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example function that adds unstructured data to tools
MyTool &lt;- function(object) {
  sample.tool.output &lt;- matrix(rnorm(n = 16), nrow = 4)
  # Note: `Tool&lt;-` must be called from within a function
  # and the name of the tool will be generated from the function name
  Tool(object) &lt;- sample.tool.output
  return(object)
}

# Run our tool
set.seed(42L)
pbmc_small &lt;- MyTool(pbmc_small)

# Get a list of tools run
Tool(pbmc_small)

# Access specific tool data
Tool(pbmc_small, slot = "MyTool")

</code></pre>

<hr>
<h2 id='UpdateSeuratObject'>Update old Seurat object to accommodate new features</h2><span id='topic+UpdateSeuratObject'></span>

<h3>Description</h3>

<p>Updates Seurat objects to new structure for storing data/calculations.
For Seurat v3 objects, will validate object structure ensuring all keys
and feature names are formed properly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UpdateSeuratObject(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UpdateSeuratObject_+3A_object">object</code></td>
<td>
<p>Seurat object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Seurat object compatible with latest changes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
updated_seurat_object = UpdateSeuratObject(object = old_seurat_object)

## End(Not run)

</code></pre>

<hr>
<h2 id='UpdateSlots'>Update slots in an object</h2><span id='topic+UpdateSlots'></span>

<h3>Description</h3>

<p>Update slots in an object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UpdateSlots(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UpdateSlots_+3A_object">object</code></td>
<td>
<p>An object to update</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>object</code> with the latest slot definitions
</p>

<hr>
<h2 id='v5-assay-summaries'>V5 Assay Summaries</h2><span id='topic+v5-assay-summaries'></span><span id='topic+colMeans+2CStdAssay-method'></span><span id='topic+colSums+2CStdAssay-method'></span><span id='topic+rowMeans+2CStdAssay-method'></span><span id='topic+rowSums+2CStdAssay-method'></span>

<h3>Description</h3>

<p>Summary maths for <code><a href="#topic+StdAssay">StdAssay</a></code> Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'StdAssay'
colMeans(x, na.rm = FALSE, dims = 1, layer = NULL, ...)

## S4 method for signature 'StdAssay'
colSums(x, na.rm = FALSE, dims = 1, layer = NULL, ...)

## S4 method for signature 'StdAssay'
rowMeans(x, na.rm = FALSE, dims = 1, layer = NULL, ...)

## S4 method for signature 'StdAssay'
rowSums(x, na.rm = FALSE, dims = 1, layer = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="v5-assay-summaries_+3A_x">x</code></td>
<td>
<p>an array of two or more dimensions, containing numeric,
complex, integer or logical values, or a numeric data frame.  For
<code>.colSums()</code> etc, a numeric, integer or logical matrix (or
vector of length <code>m * n</code>).</p>
</td></tr>
<tr><td><code id="v5-assay-summaries_+3A_na.rm">na.rm</code></td>
<td>
<p>logical.  Should missing values (including <code>NaN</code>)
be omitted from the calculations?</p>
</td></tr>
<tr><td><code id="v5-assay-summaries_+3A_dims">dims</code></td>
<td>
<p>integer: Which dimensions are regarded as &lsquo;rows&rsquo; or
&lsquo;columns&rsquo; to sum over.  For <code>row*</code>, the sum or mean is
over dimensions <code>dims+1, ...</code>; for <code>col*</code> it is over
dimensions <code>1:dims</code>.</p>
</td></tr>
<tr><td><code id="v5-assay-summaries_+3A_layer">layer</code></td>
<td>
<p>Name of layer to run function on</p>
</td></tr>
<tr><td><code id="v5-assay-summaries_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The results of the summary math function for the layer specified
</p>

<hr>
<h2 id='VariableFeatures-StdAssay'>Highly Variable Features</h2><span id='topic+VariableFeatures-StdAssay'></span><span id='topic+HVFInfo.StdAssay'></span><span id='topic+VariableFeatures.StdAssay'></span><span id='topic+VariableFeatures+3C-.StdAssay'></span>

<h3>Description</h3>

<p>Get and set variable feature information for an <code><a href="#topic+Assay">Assay</a></code> object.
<code>HVFInfo</code> and <code>VariableFeatures</code> utilize generally variable
features, while <code>SVFInfo</code> and <code>SpatiallyVariableFeatures</code> are
restricted to spatially variable features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StdAssay'
HVFInfo(object, method = NULL, status = FALSE, layer = NULL, strip = TRUE, ...)

## S3 method for class 'StdAssay'
VariableFeatures(
  object,
  method = NULL,
  layer = NA,
  simplify = TRUE,
  nfeatures = Inf,
  selection.method = deprecated(),
  ...
)

## S3 replacement method for class 'StdAssay'
VariableFeatures(object, method = "custom", layer = NULL, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VariableFeatures-StdAssay_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_method">method</code></td>
<td>
<p>Which method to pull. For <code>HVFInfo</code> and
<code>VariableFeatures</code>, choose one from one of the
following:
</p>

<ul>
<li> <p>&ldquo;vst&rdquo;
</p>
</li>
<li> <p>&ldquo;sctransform&rdquo; or &ldquo;sct&rdquo;
</p>
</li>
<li> <p>&ldquo;mean.var.plot&rdquo;, &ldquo;dispersion&rdquo;, &ldquo;mvp&rdquo;, or
&ldquo;disp&rdquo;
</p>
</li></ul>

<p>For <code>SVFInfo</code> and <code>SpatiallyVariableFeatures</code>, choose from:
</p>

<ul>
<li> <p>&ldquo;markvariogram&rdquo;
</p>
</li>
<li> <p>&ldquo;moransi&rdquo;
</p>
</li></ul>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_status">status</code></td>
<td>
<p>Add variable status to the resulting data frame</p>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_layer">layer</code></td>
<td>
<p>Layer to pull variable features for</p>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_strip">strip</code></td>
<td>
<p>Remove method/layer identifiers from highly variable data frame</p>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_simplify">simplify</code></td>
<td>
<p>When pulling for multiple layers, combine into a single
vector and select a common set of variable features for all layers</p>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_nfeatures">nfeatures</code></td>
<td>
<p>Maximum number of features to select when simplifying</p>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_selection.method">selection.method</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a></p>
</td></tr>
<tr><td><code id="VariableFeatures-StdAssay_+3A_value">value</code></td>
<td>
<p>A character vector of variable features</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>HVFInfo</code>: A data frame with feature means, dispersion, and
scaled dispersion
</p>
<p><code>VariableFeatures</code>: a vector of the variable features
</p>
<p><code>SVFInfo</code>: a data frame with the spatially variable features
</p>
<p><code>SpatiallyVariableFeatures</code>: a character vector of the spatially
variable features
</p>

<hr>
<h2 id='Version'>Get Version Information</h2><span id='topic+Version'></span><span id='topic+Version.Seurat'></span>

<h3>Description</h3>

<p>Get Version Information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Version(object, ...)

## S3 method for class 'Seurat'
Version(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Version_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="Version_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>Version(pbmc_small)

</code></pre>

<hr>
<h2 id='WhichCells'>Identify cells matching certain criteria</h2><span id='topic+WhichCells'></span><span id='topic+WhichCells.Assay'></span><span id='topic+WhichCells.Seurat'></span>

<h3>Description</h3>

<p>Returns a list of cells that match a particular set of criteria such as
identity class, high/low values for particular PCs, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WhichCells(object, ...)

## S3 method for class 'Assay'
WhichCells(object, cells = NULL, expression, invert = FALSE, ...)

## S3 method for class 'Seurat'
WhichCells(
  object,
  cells = NULL,
  idents = NULL,
  expression,
  slot = "data",
  invert = FALSE,
  downsample = Inf,
  seed = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WhichCells_+3A_object">object</code></td>
<td>
<p>An object</p>
</td></tr>
<tr><td><code id="WhichCells_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="#topic+CellsByIdentities">CellsByIdentities</a></code>
</p>

<dl>
<dt><code>return.null</code></dt><dd><p>If no cells are requested, return a <code>NULL</code>;
by default, throws an error</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="WhichCells_+3A_cells">cells</code></td>
<td>
<p>Subset of cell names</p>
</td></tr>
<tr><td><code id="WhichCells_+3A_expression">expression</code></td>
<td>
<p>A predicate expression for feature/variable expression,
can evaluate anything that can be pulled by <code>FetchData</code>; please note,
you may need to wrap feature names in backticks (<code>``</code>) if dashes
between numbers are present in the feature name</p>
</td></tr>
<tr><td><code id="WhichCells_+3A_invert">invert</code></td>
<td>
<p>Invert the selection of cells</p>
</td></tr>
<tr><td><code id="WhichCells_+3A_idents">idents</code></td>
<td>
<p>A vector of identity classes to keep</p>
</td></tr>
<tr><td><code id="WhichCells_+3A_slot">slot</code></td>
<td>
<p>Slot to pull feature data for</p>
</td></tr>
<tr><td><code id="WhichCells_+3A_downsample">downsample</code></td>
<td>
<p>Maximum number of cells per identity class, default is
<code>Inf</code>; downsampling will happen after all other operations, including
inverting the cell selection</p>
</td></tr>
<tr><td><code id="WhichCells_+3A_seed">seed</code></td>
<td>
<p>Random seed for downsampling. If NULL, does not set a seed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of cell names
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FetchData">FetchData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>WhichCells(pbmc_small, idents = 2)
WhichCells(pbmc_small, expression = MS4A1 &gt; 3)
levels(pbmc_small)
WhichCells(pbmc_small, idents = c(1, 2), invert = TRUE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
