<!DOCTYPE html><html><head><title>Help for package sgof</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sgof}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Bayesian.SGoF'><p>Bayesian SGoF multiple testing procedure</p></a></li>
<li><a href='#BBSGoF'><p>BBSGoF multiple testing procedure.</p></a></li>
<li><a href='#BH'><p>Benjamini-Hochberg (BH) multiple testing procedure</p></a></li>
<li><a href='#Binomial.SGoF'><p>Binomial SGoF multiple testing procedure</p></a></li>
<li><a href='#BY'><p>Benjamini-Yekutieli (BY) multiple testing procedure</p></a></li>
<li><a href='#Discrete.SGoF'><p>Discrete SGoF multiple testing procedure</p></a></li>
<li><a href='#Hedenfalk'><p>Hedenfalk data</p></a></li>
<li><a href='#plot.BBSGoF'><p>Plot of a BBSGoF object.</p></a></li>
<li><a href='#plot.BH'><p>Plot of a BH object</p></a></li>
<li><a href='#plot.Binomial.SGoF'><p>Plot of a Binomial.SGoF object</p></a></li>
<li><a href='#plot.BY'><p>Plot of a BY object</p></a></li>
<li><a href='#plot.SGoF'><p>Plot of a SGoF object</p></a></li>
<li><a href='#SGoF'><p>Conservative SGoF multiple testing procedure</p></a></li>
<li><a href='#sgof-package'><p>Multiple hypothesis testing</p></a></li>
<li><a href='#summary.Bayesian.SGoF'><p>Summary of a Bayesian.SGoF object</p></a></li>
<li><a href='#summary.BBSGoF'><p>Summary of a BBSGoF object</p></a></li>
<li><a href='#summary.BH'><p>Summary of a BH object</p></a></li>
<li><a href='#summary.Binomial.SGoF'><p>Summary of a Binomial.SGoF object</p></a></li>
<li><a href='#summary.BY'><p>Summary of a BY object</p></a></li>
<li><a href='#summary.SGoF'><p>Summary of a SGoF object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multiple Hypothesis Testing</td>
</tr>
<tr>
<td>Version:</td>
<td>2.3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-05</td>
</tr>
<tr>
<td>Author:</td>
<td>Irene Castro Conde and Jacobo de Una Alvarez</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Irene Castro Conde &lt;irene.castro@uvigo.es&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6), stats, poibin</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DiscreteFDR</td>
</tr>
<tr>
<td>Description:</td>
<td>Seven different methods for multiple testing problems. The SGoF-type methods (see for example, Carvajal Rodríguez et al., 2009 &lt;<a href="https://doi.org/10.1186%2F1471-2105-10-209">doi:10.1186/1471-2105-10-209</a>&gt;;  de Uña Álvarez, 2012 &lt;<a href="https://doi.org/10.1515%2F1544-6115.1812">doi:10.1515/1544-6115.1812</a>&gt;; Castro Conde et al., 2015 &lt;<a href="https://doi.org/10.1177%2F0962280215597580">doi:10.1177/0962280215597580</a>&gt;) and the BH and BY false discovery rate controlling procedures.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-05 10:20:49 UTC; erea_</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-05 12:40:24 UTC</td>
</tr>
</table>
<hr>
<h2 id='Bayesian.SGoF'>Bayesian SGoF multiple testing procedure</h2><span id='topic+Bayesian.SGoF'></span><span id='topic+print.Bayesian.SGoF'></span>

<h3>Description</h3>

<p>Performs the Bayesian SGoF method (<cite>Castro Conde and de Uña Álvarez , 2013 13/06</cite>) for multiple hypothesis testing.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bayesian.SGoF(u, alpha = 0.05, gamma = 0.05, P0 = 0.5, a0 = 1, b0 = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Bayesian.SGoF_+3A_u">u</code></td>
<td>
<p>A (non-empty) numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="Bayesian.SGoF_+3A_alpha">alpha</code></td>
<td>
<p>Numeric value. The significance level of the metatest.</p>
</td></tr>
<tr><td><code id="Bayesian.SGoF_+3A_gamma">gamma</code></td>
<td>
<p>Numeric value. The p-value threshold, so Bayesian SGoF looks for significance in the amount of p-values below gamma.</p>
</td></tr>
<tr><td><code id="Bayesian.SGoF_+3A_p0">P0</code></td>
<td>
<p>The a priori probability of the null hypothesis.</p>
</td></tr>
<tr><td><code id="Bayesian.SGoF_+3A_a0">a0</code></td>
<td>
<p>The first parameter of the a priori beta distribution.</p>
</td></tr>
<tr><td><code id="Bayesian.SGoF_+3A_b0">b0</code></td>
<td>
<p>The second parameter of the a priori beta distribution.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Bayesian SGoF (<cite>Castro Conde  and de Uña Álvarez, 2013 13/06</cite>) is an adaptation of SGoF method to the Bayesian paradigm, in which the proportion of p-values falling below <code>gamma</code> is random. This method has two main steps. First, Bayesian SGoF performs a pre-test at level <code>alpha</code> which decides if the complete null hypothesis should be rejected or not. 
This Bayesian pre-test is based on lower bounds of the a posteriori probability of H0 (computed using the  default a priori probability <code>P0</code>=.5, unless otherwise is indicated,  and a family of a priori beta distributions located at the null and indexed by a correlation factor). Second, the number of rejections is computed constructing an interval for the 'excess of significant cases', analogously to the SGoF procedure. For this, the posterior distribution of the proportion of p-values falling below <code>gamma</code> is used; this posterior distribution is calculated on the basis of the default priors  <code>a0</code>=<code>b0</code>=1, unless otherwise is indicated. Besides, the posterior probability that the complete null hypothesis is true is computed using P0, a0 and b0.
One important difference between the frequentist and the Bayesian SGoF is that the Bayesian setting induces (and hence allows for) a dependence structure among the p-values; this is very interesting for real problems where correlation is present. From a less philosophical point of view, in practice Bayesian SGoF may be more conservative than frequentist SGoF, particularly when the number of tests is small; this is due to the fact that Bayesian testing of point nulls is much more conservative than its frequentist counterpart and, therefore, the pre-test part of Bayesian SGoF may play a very important role.
Typically the choice <code>alpha</code>=<code>gamma</code> will be used for Bayesian.SGoF; this common value will be set as one of the usual significance levels (0.001, 0.01, 0.05, 0.1). Note however that <code>alpha</code> and <code>gamma</code> have different roles.
The false discovery rate is estimated by the simple method proposed by: <cite>Dalmasso , Broet , Moreau  (2005)</cite>, by taking n=1 in their formula.
</p>


<h3>Value</h3>

<p>A list containing the following values:
</p>
<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by Bayesian SGoF.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Posterior</code></td>
<td>
<p>The posterior probability that the complete null hypothesis is true depending on a0, b0 and P0.</p>
</td></tr>
<tr><td><code>s</code></td>
<td>
<p>The amount of p-values falling below gamma.</p>
</td></tr>
<tr><td><code>s.alpha</code></td>
<td>
<p>Critical point at level alpha of the Bayesian pre-test for the complete null depending on P0.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>The original p-values.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The specified significance level for the metatest.</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>The specified p-value threshold.</p>
</td></tr>
<tr><td><code>P0</code></td>
<td>
<p>The specified a priori probability of the null hypothesis.</p>
</td></tr>
<tr><td><code>a0</code></td>
<td>
<p>The first  specified parameter of the a priori beta distribution.</p>
</td></tr>
<tr><td><code>b0</code></td>
<td>
<p>The second specified parameter of the a priori beta distribution.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The matched call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Castro Conde I and de Uña Álvarez J (2013). SGoF multitesting method under the Bayesian paradigm. <em>Discussion Papers in Statistics and Operation Research</em>. Report 13/06. Statistics and OR Department. University of Vigo. 
</p>
<p>Dalmasso C, Broet P and Moreau T (2005) A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.Bayesian.SGoF">summary.Bayesian.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>

res&lt;-Bayesian.SGoF(Hedenfalk$x)
summary(res)   

</code></pre>

<hr>
<h2 id='BBSGoF'>BBSGoF multiple testing procedure.</h2><span id='topic+BBSGoF'></span><span id='topic+print.BBSGoF'></span>

<h3>Description</h3>

<p>BB-SGoF (<cite>de Uña-Álvarez, 2012</cite>; <cite>Castro-Conde and de Uña-Álvarez, in press</cite>) is an adaptation of SGoF method for possibly dependent tests. It is initially assumed that the provided vector of p-values are correlated
in k blocks of equal size (following the given sequence), where k is unknown.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BBSGoF(u, alpha = 0.05, gamma = 0.05, kmin = 2, kmax = min(length(u)%/%10, 100),
 tol = 10, adjusted.pvalues = FALSE, blocks = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BBSGoF_+3A_u">u</code></td>
<td>
<p>A (non-empty) numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="BBSGoF_+3A_alpha">alpha</code></td>
<td>
<p>Numeric value. The significance level of the metatest.</p>
</td></tr>
<tr><td><code id="BBSGoF_+3A_gamma">gamma</code></td>
<td>
<p>Numeric value. The p-value threshold, so SGoF looks for significance in the amount of p-values below gamma.</p>
</td></tr>
<tr><td><code id="BBSGoF_+3A_kmin">kmin</code></td>
<td>
<p>Numeric value. The smallest allowed number of blocks of correlated tests.</p>
</td></tr>
<tr><td><code id="BBSGoF_+3A_kmax">kmax</code></td>
<td>
<p>Numeric value. The largest allowed number of blocks of correlated tests.</p>
</td></tr>
<tr><td><code id="BBSGoF_+3A_tol">tol</code></td>
<td>
<p>Numeric value. The tolerance in model fitting (see Details).</p>
</td></tr>
<tr><td><code id="BBSGoF_+3A_adjusted.pvalues">adjusted.pvalues</code></td>
<td>
<p>Logical. Default is FALSE. A variable indicating whether to compute the adjusted p-values.</p>
</td></tr>
<tr><td><code id="BBSGoF_+3A_blocks">blocks</code></td>
<td>
<p>Numeric value. The number of existing blocks (see Details).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>BB-SGoF (<cite>de Uña-Álvarez, 2012</cite>; <cite>Castro-Conde and de Uña-Álvarez, in press</cite>) is an adaptation of SGoF method for possibly dependent tests. It is initially assumed that the provided vector of p-values are correlated in k blocks of equal size (following the given sequence), where k is unknown. Inference on the number of existing effects is performed following SGoF principles, but replacing the binomial distribution for a beta-binomial in the metatest. The beta-binomial distribution is approximated by the normal distribution; therefore, some caution is needed when the number of tests is small. It is implicitly assumed that the probability p for a p-value to fall below <code>gamma</code> is random, following a beta distribution, Beta(a,b); as a consequence, the number of p-values below gamma in each block generates a random sample from a Betabinomial(p,rho) model, where p=E(p)=a/(a+b) and rho=Var(p)/p(1-p)=1/(a+b+1) are respectively the mean of p and the within-block correlation between two indicators of type I(pi&lt;<code>gamma</code>), I(pj&lt;<code>gamma</code>). The parameters are estimated by maximum likelihood, and the asymptotic normal distribution of the estimated parameters is used to perform the inferences (so caution is needed when the number of p-values is small). Since k is unknown, the method is fitted for each integer ranging from k=<code>kmin</code> to k=<code>kmax</code>, and results for each k are saved. Automatic (conservative) choice of k is also performed; the automatic k is the value of k leading to the smallest amount of declared effects (by effects it is meant null hypotheses to be rejected).
The excess of observed significant cases in the beta-binomial metatest are reported as number of existing effects N. Finally, the effects are identified by considering the smallest N p-values.
BB-SGoF procedure weakly controls the family-wise error rate (FWER) and the false discovery rate (FDR) at level <code>alpha</code>. That is, the probability of commiting one or more than one type I errors along the multiple tests is bounded by <code>alpha</code> when all the null hypotheses are true. SGoF does not control for FWER nor FDR in the presence of effects. It has been quoted that BB-SGoF provides a good balance between FDR and power, particularly when the number of tests is large, and the effect level is weak to moderate. It is also known that the number of effects declared by BB-SGoF is a 100(1-<code>alpha</code>)% lower bound for the true number of existing effects with p-value below the initial threshold <code>gamma</code> so, interestingly, at probability 1-<code>alpha</code>, the number of false discoveries of BB-SGoF does not exceed the number of false non-discoveries (<cite>de Uña-Álvarez, 2012</cite>).
As for SGoF method, typically the choice <code>alpha</code>=<code>gamma</code> will be used for BB-SGoF; this common value will be set as one of the usual significance levels (0.001, 0.01, 0.05, 0.1). Note however that <code>alpha</code> and <code>gamma</code> have different roles.
When <code>adjusted.pvalues</code>=TRUE adjusted p-values are calculated. This are defined in the same spirit of SGoF method, but a guessed value for k must be supplied in the argument <code>blocks</code>. Once k is supplied, the adjusted p-value of a given p-value pi is defined as the smallest alpha0 for which the null hypothesis attached to pi is rejected by BB-SGoF (based on the given k) with <code>alpha</code>=<code>gamma</code>=alpha0. Actually, BBSGoF function provides an approximation of these adjusted p-values by restricting alpha0 to the set of original p-values.
The argument <code>tol</code> allows for a stronger (small <code>tol</code>) or weaker (large <code>tol</code>) criterion when removing poor fits of the beta-binomial model. When the variance of the estimated beta-binomial parameters for a given k is larger than <code>tol</code> times the median variance along k=<code>kmin</code>,...,<code>kmax</code>, the particular value of k is discarded.  
The false discovery rate is estimated by the simple method proposed by: <cite>Dalmasso, Broet, Moreau (2005)</cite>, by taking n=1 in their formula.
</p>


<h3>Value</h3>

<p>A list containing the following values:
</p>
<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by BB-SGoF with automatic k.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p> The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p> The adjusted p-values.</p>
</td></tr>
<tr><td><code>effects</code></td>
<td>
<p> A vector with the number of effects declared by BB-SGoF for each value of k.</p>
</td></tr>
<tr><td><code>SGoF</code></td>
<td>
<p>The number of effects declared by Conservative SGoF.</p>
</td></tr>
<tr><td><code>automatic.blocks</code></td>
<td>
<p> The automatic number of blocks.</p>
</td></tr>
<tr><td><code>deleted.blocks</code></td>
<td>
<p> A vector with the values of k for which the model gave a poor fit.</p>
</td></tr>
<tr><td><code>n.blocks</code></td>
<td>
<p>A vector with the values of k for which the model fitted well.</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>The average ratio of p-values below gamma.</p>
</td></tr>
<tr><td><code>cor</code></td>
<td>
<p>A vector with the estimated within-block correlation.</p>
</td></tr>
<tr><td><code>Tarone.pvalues</code></td>
<td>
<p>A vector with the p-values of Tarone’s test for no correlation.</p>
</td></tr>
<tr><td><code>Tarone.pvalue.auto</code></td>
<td>
<p>The p-values of Tarone’s test for the automatic k.</p>
</td></tr>
<tr><td><code>beta.parameters</code></td>
<td>
<p>The estimated parameters of the Beta(a,b) model for the automatic k.</p>
</td></tr>
<tr><td><code>betabinomial.parameters</code></td>
<td>
<p>The estimated parameters of the Betabinomial(p,rho) model for the automatic k.</p>
</td></tr>
<tr><td><code>sd.betabinomial.parameters</code></td>
<td>
<p>The standard deviation of the estimated parameters of the Betabinomial(p,rho) model for the automatic k.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>The original p-values.</p>
</td></tr>
<tr><td><code>adjusted.pvalues</code></td>
<td>
<p>A logical value indicating whether the adjusted p-values have been ordered.</p>
</td></tr>
<tr><td><code>blocks</code></td>
<td>
<p>Guessed value of k.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>The length of x.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The specified significance level for the metatest.</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>The specified p-value threshold.</p>
</td></tr>
<tr><td><code>kmin</code></td>
<td>
<p>The smallest allowed number of blocks of correlated tests.</p>
</td></tr>
<tr><td><code>kmax</code></td>
<td>
<p>The largest allowed number of blocks of correlated tests.</p>
</td></tr>
<tr><td><code>tol</code></td>
<td>
<p>Tolerance in model fitting (see Details).</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The matched call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Castro Conde I and de Uña Álvarez J. Power, FDR and conservativeness of BB-SGoF
method. <em>Computational Statistics</em>; Volume 30, Issue 4, pp 1143-1161 DOI: 10.1007/s00180-015-0553-2.
</p>
<p>Dalmasso C, Broet P and Moreau T (2005). A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>
<p>de Uña Álvarez J (2012). The Beta-Binomial SGoF method for multiple dependent tests. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 11, Iss. 3, Article 14.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.BBSGoF">summary.BBSGoF</a></code>,<code><a href="#topic+plot.BBSGoF">plot.BBSGoF</a></code>,<code><a href="#topic+BY">BY</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
p&lt;-runif(387)^2  #387 independent p-values, non-uniform intersection null violated

res&lt;-BBSGoF(p)
summary(res)    #automatic number of blocks, number of rejected nulls, 
		#estimated FDR, beta and beta-binomial parameters,
		#Tarone test of no correlation 

par(mfrow=c(2,2))
plot(res)   #Tarone test, within-block correlation, beta density (for automatic k),
	    #and decision plot (number of rejected nulls)


</code></pre>

<hr>
<h2 id='BH'>Benjamini-Hochberg (BH) multiple testing procedure</h2><span id='topic+BH'></span><span id='topic+print.BH'></span><span id='topic+Benjamini-Hochberg'></span>

<h3>Description</h3>

<p>Performs the Benjamini-Hochberg FDR-controlling method for multiple hypothesis testing. </p>


<h3>Usage</h3>

<pre><code class='language-R'>BH(u, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BH_+3A_u">u</code></td>
<td>
<p> A (non-empty) numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="BH_+3A_alpha">alpha</code></td>
<td>
<p> Numeric value. The significance level of the test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function BH allows for the application of the <cite>Benjamini and Hochberg (1995)</cite> false discovery rate controlling procedure.
The false discovery rate is estimated by the simple method proposed by: <cite>Dalmasso, Broet, Moreau (2005)</cite>, by taking n=1 in their formula.
</p>


<h3>Value</h3>

<p>A list containing the following values:
</p>
<table>
<tr><td><code>Rejections</code></td>
<td>
<p> The number of effects declared by the BH procedure.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p> The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>     The adjusted p-values.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p> The original p-values.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p> The specified significance level for the test.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p> The matched call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Benjamini Y and Hochberg Y (1995). Controlling the False Discovery Rate: A
Practical and Powerful Approach to Multiple Testing. <em>Journal of the Royal Statistical Society Series B (Methodological)</em> 57, 289&ndash;300.
</p>
<p>Dalmasso C, Broet P and Moreau T (2005). A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.BH">plot.BH</a></code>,<code><a href="#topic+summary.BH">summary.BH</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>


res&lt;-BH(Hedenfalk$x)
summary(res)   #number of rejected nulls, estimated FDR
plot(res)   #adjusted p-values
</code></pre>

<hr>
<h2 id='Binomial.SGoF'>Binomial SGoF multiple testing procedure</h2><span id='topic+Binomial.SGoF'></span><span id='topic+print.Binomial.SGoF'></span>

<h3>Description</h3>

<p>Performs the Binomial SGoF method (<cite>Carvajal Rodríguez et al., 2009</cite>) for multiple hypothesis testing.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Binomial.SGoF(u, alpha = 0.05, gamma = 0.05)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Binomial.SGoF_+3A_u">u</code></td>
<td>
<p>A (non-empty) numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="Binomial.SGoF_+3A_alpha">alpha</code></td>
<td>
<p>Numeric value. The significance level of the metatest.</p>
</td></tr>
<tr><td><code id="Binomial.SGoF_+3A_gamma">gamma</code></td>
<td>
<p>Numeric value. The p-value threshold, so Binomial SGoF looks for significance in the amount of p-values below gamma.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Binomial SGoF starts by counting the amount of p-values below <code>gamma</code>. This amount is compared to the expected one under the intersection or complete null hypothesis (all the nulls are true) in a metatest, performed at level <code>alpha</code>. Note that, under the intersection null, the p-values will be uniformly distributed on the (0,1) interval, so one expects <code>gamma</code> times the length of <code>u</code> p-values falling below <code>gamma</code>. If the intersection null is accepted, Binomial  SGoF reports no effects. If the intersection null is rejected, the excess of observed significant cases are reported as number of existing effects N (by effects it is meant null hypotheses to be rejected). Finally, the effects are identified by considering the smallest N p-values. The only input you need is the set of p-values.
Binomial SGoF procedure weakly controls the family-wise error rate (FWER) and the false discovery rate (FDR) at level <code>alpha</code>. That is, the probability of committing one or more than one type I errors along the multiple tests is bounded by <code>alpha</code> when all the null hypotheses are true. SGoF does not control for FWER nor FDR in the presence of effects. It has been quoted that Binomial SGoF provides a good balance between FDR and power, particularly when the number of tests is large, and the effect level is weak to moderate. It is also known that the number of effects declared by Binomial SGoF is a 100(1-<code>alpha</code>)% lower bound for the true number of existing effects with p-value below the initial threshold <code>gamma</code> so, interestingly, at probability 1-<code>alpha</code>, the number of false discoveries of SGoF does not exceed the number of false non-discoveries (<cite>de Uña Álvarez, 2012</cite>).
Typically, the choice <code>alpha</code>=<code>gamma</code> will be used; this common value will be set as one of the usual significance levels (0.001, 0.01, 0.05, 0.1). Note however that <code>alpha</code> and <code>gamma</code> have different roles.
The FDR is estimated by the simple method proposed by: <cite>Dalmasso, Broet, Moreau (2005)</cite>, by taking n=1 in their formula.
The adjusted p-value of a given p-value pi is defined as the smallest alpha0 for which the null hypothesis attached to pi is rejected by SGoF with <code>alpha</code>=<code>gamma</code>=alpha0. Actually, Binomial.SGoF function provides  these adjusted p-values by restricting alpha0 to the set of original p-values. <cite>Castro-Conde and de Uña-Álvarez  (2015)</cite>   proved that this restriction does not change the adjusted p-values, while reducing the computational time.
</p>


<h3>Value</h3>

<p>A list containing the following values:
</p>
<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by Binomial SGoF.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>The adjusted p-values.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>The original p-values.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The specified significance level for the metatest.</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>The specified p-value threshold.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The matched call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Carvajal Rodríguez A, de Uña Álvarez J and Rolán Álvarez E (2009). A new
multitest correction (SGoF) that increases its statistical power when increasing the number of tests. <em>BMC Bioinformatics</em> 10:209.
</p>
<p>Castro Conde I and de Uña Álvarez J (2015). Adjusted p-values for SGoF multiple test procedure.  <em>Biometrical Journal</em>,  57(1): 108-122. DOI: 10.1002/bimj.201300238
</p>
<p>Dalmasso C, Broet P and Moreau T (2005) A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>
<p>de Uña Álvarez J (2011). On the statistical properties of SGoF multitesting method. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 10, Iss. 1, Article 18.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.Binomial.SGoF">plot.Binomial.SGoF</a></code>,<code><a href="#topic+summary.Binomial.SGoF">summary.Binomial.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
p&lt;-runif(387)^2  #387 independent p-values, non-uniform intersection null violated

res&lt;-Binomial.SGoF(p)
summary(res)   #number of rejected nulls, estimated FDR
plot(res)   #adjusted p-values
</code></pre>

<hr>
<h2 id='BY'>Benjamini-Yekutieli (BY) multiple testing procedure</h2><span id='topic+BY'></span><span id='topic+print.BY'></span><span id='topic+Benjamini-Yekutieli'></span>

<h3>Description</h3>

<p>Performs the Benjamini-Yekutieli FDR-controlling method for multiple hypothesis testing. </p>


<h3>Usage</h3>

<pre><code class='language-R'>BY(u, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BY_+3A_u">u</code></td>
<td>
<p> A (non-empty) numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="BY_+3A_alpha">alpha</code></td>
<td>
<p> Numeric value. The significance level of the test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function BY allows for the application of the <cite>Benjamini and Yekutieli (2001)</cite> false discovery rate controlling procedure under dependence assumptions.
The false discovery rate is estimated by the simple method proposed by: <cite>Dalmasso, Broet, Moreau (2005)</cite>, by taking n=1 in their formula.
</p>


<h3>Value</h3>

<p>A list containing the following values:
</p>
<table>
<tr><td><code>Rejections</code></td>
<td>
<p> The number of effects declared by the BY procedure.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p> The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>The adjusted p-values.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p> The original p-values.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p> The specified significance level for the test.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p> The matched call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Benjamini Y and Yekutieli D (2001). The control of the false discovery rate in multiple testing under dependency. <em>Annals of Statistics</em> 29, 1165&ndash;1188.
</p>
<p>Dalmasso C, Broet P and Moreau T (2005). A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.BY">plot.BY</a></code>,<code><a href="#topic+summary.BY">summary.BY</a></code>,<code><a href="#topic+BBSGoF">BBSGoF</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>


res&lt;-BY(Hedenfalk$x)
summary(res)   #number of rejected nulls, estimated FDR
plot(res)   #adjusted p-values
</code></pre>

<hr>
<h2 id='Discrete.SGoF'>Discrete SGoF multiple testing procedure</h2><span id='topic+Discrete.SGoF'></span><span id='topic+print.Discrete.SGoF'></span>

<h3>Description</h3>

<p>Performs the Discrete SGoF method (<cite>Castro-Conde, Döhler et al., 2015</cite>) for multiple hypothesis testing.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Discrete.SGoF(u,pCDFlist=NA, K=NA, alpha = 0.05, gamma = 0.05, method=NA, 
              Discrete=TRUE, Sides=1,...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Discrete.SGoF_+3A_u">u</code></td>
<td>
<p>A (non-empty) numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="Discrete.SGoF_+3A_pcdflist">pCDFlist</code></td>
<td>
<p>A (non-empty) list with the empirical cumulative function of each discrete p-value.</p>
</td></tr>
<tr><td><code id="Discrete.SGoF_+3A_k">K</code></td>
<td>
<p>Numeric value. The number of continuous tests.</p>
</td></tr>
<tr><td><code id="Discrete.SGoF_+3A_alpha">alpha</code></td>
<td>
<p>Numeric value. The significance level of the metatest.</p>
</td></tr>
<tr><td><code id="Discrete.SGoF_+3A_gamma">gamma</code></td>
<td>
<p>Numeric value. The p-value threshold, so Discrete SGoF looks for significance in the amount of p-values below gamma.</p>
</td></tr>
<tr><td><code id="Discrete.SGoF_+3A_method">method</code></td>
<td>
<p>Method used in the computation of the Poisson binomial quantile.  &quot;DFT-CF&quot; for the exact method and &quot;RNA&quot; for the refined normal approximation.</p>
</td></tr>
<tr><td><code id="Discrete.SGoF_+3A_discrete">Discrete</code></td>
<td>
<p>Logical. Default is TRUE. A variable indicating if the tests are discrete or continuous in order to estimate the FDR.</p>
</td></tr>
<tr><td><code id="Discrete.SGoF_+3A_sides">Sides</code></td>
<td>
<p>Numeric value indicating if the tests are one-sided (default), <code>Sides=1</code>, or two-sided, <code>Sides=2</code> in order to estimate the FDR.</p>
</td></tr>
<tr><td><code id="Discrete.SGoF_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to <code>robust.fdr</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Discrete SGoF is an extension of Binomial SGoF, based on the generalized or Poisson binomial distribution (Hong, 2013),  which takes into account the discreteness of the p-values. If all the tests are continuous Discrete SGoF reduces to Binomial SGoF method. 
In particular, if the p-values are continuous, the number of rejections given by Discrete.SGoF will be the number of effects declared by Binomial SGoF.  For computing the Poisson Binomial quantile, the <code>poibin</code> package (Hong, 2019) is used. The exact method (&quot;DFT-CF&quot;) and the &quot;RNA&quot; approximation are used by default to compute the quantile depending on whether the number of tests is smaller than 2000  or not, respectively (see Hong 2013a for more information). However, the user can specified which of the two methods to use.
Discrete SGoF works the same like Binomial SGoF but it uses the quantiles of the generalized binomial distribution, as mentioned, instead of the binomial quantiles.
Discrete SGoF maintains the theoretical properties of Binomial SGoF, e.g. weak control of FDR(FWER) and increasing power when the number of tests increases (de Uña Álvarez, 2011).
The FDR is estimated by using the method proposed by: <cite>Pounds and Cheng (2006)</cite> using the <code>robust.fdr</code> function provided by the authors.
</p>


<h3>Value</h3>

<p>A list containing the following values:
</p>
<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by Discrete SGoF.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>pvalues</code></td>
<td>
<p>The original p-values.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The specified significance level for the metatest.</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>The specified p-value threshold.</p>
</td></tr>
<tr><td><code>K</code></td>
<td>
<p>The specified number of continuous tests.</p>
</td></tr>
<tr><td><code>Method</code></td>
<td>
<p>The specified method used in the computation of the Poisson binomial quantile. </p>
</td></tr>
<tr><td><code>Discrete</code></td>
<td>
<p>The specified type of tests.</p>
</td></tr>
<tr><td><code>Sides</code></td>
<td>
<p>Numeric value indicating if the tests are one-sided (default), <code>Sides=1</code>, or two-sided, <code>Sides=2</code>.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The matched call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Sebastian Döhler</p>


<h3>References</h3>

<p>Carvajal Rodríguez A, de Uña Álvarez J and Rolán Álvarez E (2009). A new
multitest correction (SGoF) that increases its statistical power when increasing the number of tests. <em>BMC Bioinformatics</em> 10:209.
</p>
<p>Castro Conde I, Döhler S and de Uña Álvarez J (2015). An extended sequential goodness-of-fit multiple testing method for discrete data. <em>Statistical Methods in Medical Research</em>, doi: 10.1177/0962280215597580.
</p>
<p>de Uña Álvarez J (2011). On the statistical properties of SGoF multitesting method. 
<em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 10, Iss. 1, Article 18.
</p>
<p>Durand G, Junge F (2021). DiscreteFDR: Multiple Testing Procedures with Adaptation for Discrete Tests. 
R package version  1.3.6
</p>
<p>Hong Y. (2013). On computing the distribution functions for the Poisson binomial
distribution. <em>Computational Statistics and Data Analysis</em> 59, 41-51.
</p>
<p>Hong Y. (2019). poibin: The Poisson Binomial Distribution. R package version 1.4
</p>
<p>Pounds, S. and C. Cheng (2006). Robust estimation of the false discovery rate.
<em>Bioinformatics</em> 22 (16), 1979-1987.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Binomial.SGoF">Binomial.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>

require(DiscreteFDR)

data(amnesia) #discrete data

AllAdverseCases&lt;-amnesia$OtherAdverseCases + amnesia$AmnesiaCases
A11 &lt;- amnesia$AmnesiaCases
A21 &lt;- sum(AllAdverseCases) - A11
A12 &lt;- AllAdverseCases - A11
A22 &lt;- sum(AllAdverseCases) - sum(amnesia$AmnesiaCases) - A12

A1. &lt;- sum(amnesia$AmnesiaCases)
A2. &lt;- sum(AllAdverseCases) - A1.
  
n &lt;- A11 + A12
k &lt;- pmin(n,A1.)

pCDFlist &lt;- list()
pvec &lt;- numeric(nrow(amnesia))

## Calculation of the p-values and the p-values CDFs: 

for (i in 1:nrow(amnesia))
{
  x &lt;- 0:k[i]
  pCDFlist[[i]] &lt;- dhyper(x ,A1., A2. ,n[i]) + phyper(x ,A1. ,A2. ,n[i] ,lower.tail = FALSE)
  pCDFlist[[i]] &lt;- rev(pCDFlist[[i]])
  pvec[i] &lt;- dhyper(A11[i] ,A1. ,A2. ,n[i]) + phyper(A11[i] ,A1. ,A2. ,n[i] ,lower.tail = FALSE)
}

res&lt;-Discrete.SGoF(u=pvec,pCDFlist=pCDFlist,alpha=0.05,gamma=0.05,Discrete=TRUE,Sides=1)
res


#continuous p-values

res2&lt;-Discrete.SGoF(u=Hedenfalk$x,K=3170,Discrete=FALSE, method="DFT-CF",Sides=2)
res2

</code></pre>

<hr>
<h2 id='Hedenfalk'>Hedenfalk data</h2><span id='topic+Hedenfalk'></span>

<h3>Description</h3>

<p>The data include information of the micro array study of hereditary breast cancer
of <cite>Hedenfalk et al. (2001)</cite>. Many cases of hereditary breast cancer are due to mutations in either the BRCA1 or the BRCA2 gene. The histopathological changes in these cancers are often characteristic of the mutant gene. They hypothesized that the genes expressed by these two types of tumors are also distinctive, perhaps allowing to identify cases of hereditary breast cancer on the basis of gene-expression profiles.
</p>
<p>The patients consisted of 23 with BRCA1 mutations, 17 with BRCA2 mutations, 20 with familial breast cancer, 19 with possibly familial breast cancer and 34 with sporadic breast cancer to determine whether there are distinctive patterns of global gene expression in these three kinds of tumors.
</p>
<p>One of the goals of this study was to find genes
differentially expressed between BRCA1- and BRCA2-mutation positive tumors.
Thus, the data included here are p-values obtained from a two- sample t-test analysis on a subset of 3170 genes, as described in <cite>Storey and Tibshirani (2003)</cite>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hedenfalk</code></pre>


<h3>Format</h3>


<dl>
<dt><code>x</code></dt><dd><p>A numeric vector of 3170 p-values of tests comparing BRCA1 to BRCA2.</p>
</dd>
</dl>



<h3>References</h3>

<p>Hedenfalk I, Duggan D, Chen Y, Radmacher M, Bittner M et al. (2001). Gene-Expression Profiles in Hereditary Breast Cancer. <em>New England Journal of Medicine</em> 344, 539&ndash;548.
</p>
<p>Storey JD and Tibshirani R (2003). Statistical significance for genome-wide studies. <em>Proceedings of the National Academy of Sciences</em>, 100: 9440&ndash;9445.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
hist(Hedenfalk$x)
</code></pre>

<hr>
<h2 id='plot.BBSGoF'>Plot of a BBSGoF object.</h2><span id='topic+plot.BBSGoF'></span>

<h3>Description</h3>

<p>Plot of a BBSGoF object</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BBSGoF'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.BBSGoF_+3A_x">x</code></td>
<td>
<p>A BBSGoF object.</p>
</td></tr>
<tr><td><code id="plot.BBSGoF_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>de Uña Álvarez J (2012). The Beta-Binomial SGoF method for multiple dependent tests. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 11, Iss. 3, Article 14.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BBSGoF">BBSGoF</a></code>,<code><a href="#topic+summary.BBSGoF">summary.BBSGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
p&lt;-runif(387)^2  #387 independent p-values, non-uniform intersection null violated

res&lt;-BBSGoF(p)
	
par(mfrow=c(2,2))
plot(res)   #Tarone test, within-block correlation, beta density (for automatic k),
	        #and decision plot (number of rejected nulls)

</code></pre>

<hr>
<h2 id='plot.BH'>Plot of a BH object</h2><span id='topic+plot.BH'></span>

<h3>Description</h3>

<p>Plot of the Adjusted p-values given by the BH method. </p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BH'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.BH_+3A_x">x</code></td>
<td>
<p>A BH object.</p>
</td></tr>
<tr><td><code id="plot.BH_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Benjamini Y and Hochberg Y (1995). Controlling the False Discovery Rate: A
Practical and Powerful Approach to Multiple Testing. <em>Journal of the Royal Statistical Society Series B (Methodological)</em> 57, 289&ndash;300.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BH">BH</a></code>,<code><a href="#topic+summary.BH">summary.BH</a></code>,<code><a href="#topic+SGoF">SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>


res&lt;-BH(Hedenfalk$x)
plot(res)  
</code></pre>

<hr>
<h2 id='plot.Binomial.SGoF'>Plot of a Binomial.SGoF object</h2><span id='topic+plot.Binomial.SGoF'></span>

<h3>Description</h3>

<p>Plot the Adjusted p-values given by the Binomial SGoF method. </p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Binomial.SGoF'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Binomial.SGoF_+3A_x">x</code></td>
<td>
<p>A Binomial.SGoF object.</p>
</td></tr>
<tr><td><code id="plot.Binomial.SGoF_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Carvajal Rodríguez A, de Uña Álvarez J and Rolán Álvarez E (2009) A new
multitest correction (Binomial.SGoF) that increases its statistical power when increasing the number of tests. <em>BMC Bioinformatics</em> 10:209.
</p>
<p>Castro Conde I and de Uña Álvarez J (2015). Adjusted p-values for SGoF multiple test procedure.  <em>Biometrical Journal</em> 57(1): 108-122. DOI: 10.1002/bimj.201300238
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Binomial.SGoF">Binomial.SGoF</a></code>,<code><a href="#topic+summary.Binomial.SGoF">summary.Binomial.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
p&lt;-runif(387)^2  #387 independent p-values, non-uniform intersection null violated

res&lt;-Binomial.SGoF(p)
plot(res)  
</code></pre>

<hr>
<h2 id='plot.BY'>Plot of a BY object</h2><span id='topic+plot.BY'></span>

<h3>Description</h3>

<p>Plot of the Adjusted p-values given by the BY method. </p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BY'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.BY_+3A_x">x</code></td>
<td>
<p>A BY object.</p>
</td></tr>
<tr><td><code id="plot.BY_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Benjamini Y and Yekutieli D (2001). The control of the false discovery rate in multiple testing under dependency. <em>Annals of Statistics</em> 29, 1165&ndash;1188.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BY">BY</a></code>,<code><a href="#topic+summary.BY">summary.BY</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>


res&lt;-BY(Hedenfalk$x)
plot(res)  
</code></pre>

<hr>
<h2 id='plot.SGoF'>Plot of a SGoF object</h2><span id='topic+plot.SGoF'></span>

<h3>Description</h3>

<p>Plot the Adjusted p-values given by the Conservative SGoF method. </p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SGoF'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.SGoF_+3A_x">x</code></td>
<td>
<p>A SGoF object.</p>
</td></tr>
<tr><td><code id="plot.SGoF_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Castro Conde I and de Uña Álvarez J (2015). Adjusted p-values for SGoF multiple test procedure.
<em>Biometrical Journal</em>; 57(1): 108-122. DOI:DOI:10.1002/bimj.201300238.
</p>
<p>de Uña Álvarez J (2011). On the statistical properties of SGoF multitesting method. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 10, Iss. 1, Article 18.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SGoF">SGoF</a></code>,<code><a href="#topic+summary.SGoF">summary.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
p&lt;-runif(387)^2  #387 independent p-values, non-uniform intersection null violated

res&lt;-SGoF(p)
plot(res)  
</code></pre>

<hr>
<h2 id='SGoF'>Conservative SGoF multiple testing procedure</h2><span id='topic+SGoF'></span><span id='topic+print.SGoF'></span>

<h3>Description</h3>

<p>Performs Conservative SGoF method (<cite>de Uña Álvarez, 2011</cite>) for multiple hypothesis testing.</p>


<h3>Usage</h3>

<pre><code class='language-R'>SGoF(u, alpha = 0.05, gamma = 0.05)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SGoF_+3A_u">u</code></td>
<td>
<p>A (non-empty) numeric vector of p-values.</p>
</td></tr>
<tr><td><code id="SGoF_+3A_alpha">alpha</code></td>
<td>
<p>Numeric value. The significance level of the metatest.</p>
</td></tr>
<tr><td><code id="SGoF_+3A_gamma">gamma</code></td>
<td>
<p>Numeric value. The p-value threshold, so Conservative SGoF looks for significance in the amount of p-values below gamma.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Conservative SGoF is an asymptotic version (large number of tests) of the  Binomial SGoF  procedure, where the binomial quantiles are approximated by the normal ones. Besides, the variance of the number of p-values below <code>gamma</code> is estimated without assuming that all the null hypotheses are true, which typically results in a more conservative decision (from this the method’s name). When the number of tests is large, Conservative SGoF and Binomial SGoF report approximately the same result. This method should no be used when the number of tests is small, because the binomial-normal approximation will perform poorly. Conservative SGoF method has the main properties of  Binomial SGoF like weak control of  the family-wise error rate (FWER) and the false discovery rate (FDR) at level <code>alpha</code> and  a good balance between FDR and power, particularly when the number of tests is large, and the effect level is weak to moderate. See <code>Binomial.SGoF</code> for more details.
Typically, the choice <code>alpha</code>=<code>gamma</code> will be used; this common value will be set as one of the usual significance levels (0.001, 0.01, 0.05, 0.1). Note however that <code>alpha</code> and <code>gamma</code> have different roles.
The FDR is estimated by the simple method proposed by: <cite>Dalmasso, Broet, Moreau  (2005)</cite>, by taking n=1 in their formula.
The adjusted p-value of a given p-value pi is defined as the smallest alpha0 for which the null hypothesis attached to pi is rejected by Conservative SGoF with <code>alpha</code>=<code>gamma</code>=alpha0. Actually, Conservative.SGoF function provides  these adjusted p-values by restricting alpha0 to the set of original p-values (<cite>Castro Conde  and de Uña Álvarez , 2015</cite>).
</p>


<h3>Value</h3>

<p>A list containing the following values:
</p>
<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by SGoF.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>The adjusted p-values.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>The original p-values.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The specified significance level for the metatest.</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>The specified p-value threshold.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The matched call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Castro Conde I and de Uña Álvarez J (2015). Adjusted p-values for SGoF multiple test procedure.   <em>Biometrical Journal</em>; 57(1): 108-122. DOI:10.1002/bimj.201300238.
</p>
<p>Dalmasso C, Broet P and Moreau T (2005). A simple procedure for estimating the false discovery rate. 
<em>Bioinformatics</em> 21:660&ndash;668
</p>
<p>de Uña-Álvarez J (2011). On the statistical properties of SGoF multitesting method. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 10, Iss. 1, Article 18.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.SGoF">summary.SGoF</a></code>,<code><a href="#topic+plot.SGoF">plot.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
p&lt;-runif(387)^2  #387 independent p-values, non-uniform intersection null violated

res&lt;-SGoF(p)
summary(res)   #number of rejected nulls, estimated FDR
plot(res)   #adjusted p-values
</code></pre>

<hr>
<h2 id='sgof-package'>Multiple hypothesis testing</h2><span id='topic+sgof-package'></span><span id='topic+sgof'></span>

<h3>Description</h3>

<p>This package implements seven different methods for multiple testing problems. 
The <cite>Benjamini and Hochberg (1995)</cite> false discovery rate controlling procedure and its modification for  dependent tests <cite>Benjamini and Yekutieli (2001)</cite>, 
the method called Binomial SGoF  proposed in <cite>Carvajal Rodríguez et al. (2009)</cite> and its conservative and bayesian versions called Conservative SGoF (<cite>de Uña Álvarez, 2011</cite>) and Bayesian SGoF (<cite>Castro Conde  and de Uña Álvarez, 2013 13/06</cite>), respectively,
and the BB-SGoF  (Beta-Binomial SGoF, <cite>de Uña Álvarez, 2012</cite>) and Discrete SGoF (<cite>Castro Conde et al., 2015</cite>) procedures which are  adaptations of SGoF method for possibly correlated tests and for discrete tests, respectively. Number of rejections, FDR and adjusted p-values are computed among other things.
</p>


<h3>Details</h3>

<p>This package incorporates the functions <code>BH</code>,<code>BY</code>, <code>SGoF</code>, <code>Binomial.SGoF</code>, <code>Bayesian.SGoF</code>, <code>Discrete.SGoF</code>  and <code>BBSGoF</code>, which call the methods aforementioned. For a complete list of functions, use library(help=&quot;sgof&quot;).
</p>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez
</p>
<p>Maintainer:Irene Castro Conde <a href="mailto:irene.castro@uvigo.es">irene.castro@uvigo.es</a>
</p>


<h3>References</h3>

<p>Benjamini Y and Hochberg Y (1995). Controlling the False Discovery Rate: A
Practical and Powerful Approach to Multiple Testing. <em>Journal of the Royal Statistical Society Series B (Methodological)</em> 57, 289&ndash;300.
</p>
<p>Benjamini Y and Yekutieli D (2001). The control of the false discovery rate in multiple testing under dependency. <em>Annals of Statistics</em> 29, 1165–-1188.
</p>
<p>Carvajal Rodríguez A, de Uña Álvarez J and Rolán Álvarez E (2009). A new
multitest correction (SGoF) that increases its statistical power when increasing the number of tests. <em>BMC Bioinformatics</em> 10:209.
</p>
<p>Castro Conde I and de Uña Álvarez J. Power, FDR and conservativeness of BB-SGoF
method. <em>Computational Statistics</em>; Volume 30, Issue 4, pp 1143-1161 DOI: 10.1007/s00180-015-0553-2.
</p>
<p>Castro Conde I and de Uña Álvarez J (2015). Adjusted p-values for SGoF multiple test procedure.  <em>Biometrical Journal</em>; 57(1): 108-122. DOI: 10.1002/bimj.201300238
</p>
<p>Castro Conde I, Döhler S and de Uña Álvarez J (2015). An extended sequential goodness-of-fit multiple testing method for discrete data. <em>Statistical Methods in Medical Research</em>. doi: 10.1177/0962280215597580.
</p>
<p>Castro Conde I and de Uña Álvarez J (2014). sgof: An R package for multiple testing problems. <em>The
R Journal</em>; Vol. 6/2 December: 96-113.
</p>
<p>Castro Conde I and de Uña Álvarez J (2013). SGoF multitesting method under the Bayesian paradigm. <em>Discussion Papers in Statistics and Operation Research</em>. Report 13/06. Statistics and OR Department. University of Vigo. 
</p>
<p>Dalmasso C, Broet P and Moreau T (2005). A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>
<p>de Uña Álvarez J (2011). On the statistical properties of SGoF multitesting method. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 10, Iss. 1, Article 18.
</p>
<p>de Uña Álvarez J (2012). The Beta-Binomial SGoF method for multiple dependent tests. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 11, Iss. 3, Article 14.
</p>
<p>Durand G, Junge F (2021). DiscreteFDR: Multiple Testing Procedures with Adaptation for Discrete Tests. 
R package version  1.3.6
</p>
<p>Hong Y. (2013). On computing the distribution functions for the Poisson binomial
distribution. <em>Computational Statistics and Data Analysis</em> 59, 41-51.
</p>
<p>Hong Y. (2019). poibin: The Poisson Binomial Distribution. R package version 1.4
</p>
<p>Pounds, S. and C. Cheng (2006). Robust estimation of the false discovery rate.
<em>Bioinformatics</em> 22 (16), 1979-1987.
</p>

<hr>
<h2 id='summary.Bayesian.SGoF'>Summary of a Bayesian.SGoF object</h2><span id='topic+summary.Bayesian.SGoF'></span>

<h3>Description</h3>

<p>Summary of the most important results given by the Bayesian SGoF procedure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Bayesian.SGoF'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.Bayesian.SGoF_+3A_object">object</code></td>
<td>
<p>A Bayesian.SGoF object.</p>
</td></tr>
<tr><td><code id="summary.Bayesian.SGoF_+3A_...">...</code></td>
<td>
<p>Additional arguments affecting the summary produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by the Bayesian SGoF procedure.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Posterior</code></td>
<td>
<p>The posterior probability that the complete null hypothesis is true considering the non informative election.</p>
</td></tr>
<tr><td><code>s</code></td>
<td>
<p>The proportion of p-values falling below gamma.</p>
</td></tr>
<tr><td><code>s.alpha</code></td>
<td>
<p>The first integer from 0 to n such that the Bayesian pre-test rejects the complete null hypothesis.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Castro Conde I and de Uña Álvarez J (2013). SGoF multitesting method under the Bayesian paradigm. <em>Discussion Papers in Statistics and Operation Research</em>. Report 13/06. Statistics and OR Department. University of Vigo. 
</p>
<p>Dalmasso C, Broet P and Moreau T (2005) A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Bayesian.SGoF">Bayesian.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>


res&lt;-Bayesian.SGoF(Hedenfalk$x)
summary(res)   


</code></pre>

<hr>
<h2 id='summary.BBSGoF'>Summary of a BBSGoF object</h2><span id='topic+summary.BBSGoF'></span>

<h3>Description</h3>

<p>Summary of the most important results given by the BBSGoF procedure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BBSGoF'
summary(object,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.BBSGoF_+3A_object">object</code></td>
<td>
<p>A BBSGoF object.</p>
</td></tr>
<tr><td><code id="summary.BBSGoF_+3A_...">...</code></td>
<td>
<p>Additional arguments affecting the summary produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by BB-SGoF with automatic k.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>Table of adjusted p-values falling under gamma.</p>
</td></tr>
<tr><td><code>Tarone.pvalue.auto</code></td>
<td>
<p>The p-values of Tarone’s test for the automatic k.</p>
</td></tr>
<tr><td><code>beta.parameters</code></td>
<td>
<p>The estimated parameters of the Beta(a,b) model for the automatic k.</p>
</td></tr>
<tr><td><code>betabinomial.parameters</code></td>
<td>
<p>The estimated parameters of the Betabinomial(p,ro) model for the automatic k.</p>
</td></tr>
<tr><td><code>sd.betabinomial.parameters</code></td>
<td>
<p>The standard deviation of the estimated parameters of the Betabinomial(p,ro) model for the automatic k.</p>
</td></tr>
<tr><td><code>automatic.blocks</code></td>
<td>
<p>The automatic number of blocks.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Dalmasso C, Broet P and Moreau T (2005) A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>
<p>de Uña Álvarez J (2012). The Beta-Binomial SGoF method for multiple dependent tests. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 11, Iss. 3, Article 14.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BBSGoF">BBSGoF</a></code>,<code><a href="#topic+plot.BBSGoF">plot.BBSGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
p&lt;-runif(387)^2  #387 p-values, intersection null violated

res&lt;-BBSGoF(p)
summary(res)    #automatic number of blocks, number of rejected nulls, 
		#estimated FDR, beta and beta-binomial parameters,
		#Tarone test of no correlation 


</code></pre>

<hr>
<h2 id='summary.BH'>Summary of a BH object</h2><span id='topic+summary.BH'></span>

<h3>Description</h3>

<p>Summary of the most important results given by the BH procedure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BH'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.BH_+3A_object">object</code></td>
<td>
<p>A BH object.</p>
</td></tr>
<tr><td><code id="summary.BH_+3A_...">...</code></td>
<td>
<p>Additional arguments affecting the summary produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by the BH procedure.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>Table of adjusted p-values falling under alpha.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Benjamini Y and Hochberg Y (1995): Controlling the False Discovery Rate: A
Practical and Powerful Approach to Multiple Testing. <em>Journal of the Royal Statistical Society Series B (Methodological)</em> 57, 289&ndash;300.
</p>
<p>Dalmasso C, Broet P and Moreau T (2005) A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BH">BH</a></code>,<code><a href="#topic+plot.BH">plot.BH</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>

res&lt;-BH(Hedenfalk$x)
summary(res) 
</code></pre>

<hr>
<h2 id='summary.Binomial.SGoF'>Summary of a Binomial.SGoF object</h2><span id='topic+summary.Binomial.SGoF'></span>

<h3>Description</h3>

<p>Summary of the most important results given by the Binomial SGoF procedure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Binomial.SGoF'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.Binomial.SGoF_+3A_object">object</code></td>
<td>
<p>A Binomial.SGoF object.</p>
</td></tr>
<tr><td><code id="summary.Binomial.SGoF_+3A_...">...</code></td>
<td>
<p>Additional arguments affecting the summary produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by the Binomial SGoF procedure .</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>Table of adjusted p-values falling under gamma.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Carvajal Rodríguez A, de Uña Álvarez J and Rolán Álvarez E (2009). A new
multitest correction (Binomial.SGoF) that increases its statistical power when increasing the number of tests. <em>BMC Bioinformatics</em> 10:209.
</p>
<p>Castro Conde I and de Uña Álvarez J (2015). Adjusted p-values for SGoF multiple test procedure.  <em>Biometrical Journal</em>; 57(1): 108-122. DOI: 10.1002/bimj.201300238
</p>
<p>Dalmasso C, Broet P and Moreau T (2005) A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Binomial.SGoF">Binomial.SGoF</a></code>,<code><a href="#topic+plot.Binomial.SGoF">plot.Binomial.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>


p&lt;-runif(387)^2  #387 independent p-values, non-uniform intersection null violated

res&lt;-Binomial.SGoF(p)
summary(res)   #number of rejected nulls, estimated FDR


</code></pre>

<hr>
<h2 id='summary.BY'>Summary of a BY object</h2><span id='topic+summary.BY'></span>

<h3>Description</h3>

<p>Summary of the most important results given by the BY procedure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BY'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.BY_+3A_object">object</code></td>
<td>
<p>A BY object.</p>
</td></tr>
<tr><td><code id="summary.BY_+3A_...">...</code></td>
<td>
<p>Additional arguments affecting the summary produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by the BY method.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>Table of adjusted p-values falling under alpha.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Benjamini Y and Yekutieli D (2001). The control of the false discovery rate in multiple testing under dependency. <em>Annals of Statistics</em> 29, 1165&ndash;1188.
</p>
<p>Dalmasso C, Broet P and Moreau T (2005) A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BY">BY</a></code>,<code><a href="#topic+plot.BY">plot.BY</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>

res&lt;-BY(Hedenfalk$x)
summary(res) 
</code></pre>

<hr>
<h2 id='summary.SGoF'>Summary of a SGoF object</h2><span id='topic+summary.SGoF'></span>

<h3>Description</h3>

<p>Summary of the most important results given by the Conservative SGoF procedure.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SGoF'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.SGoF_+3A_object">object</code></td>
<td>
<p>A SGoF object.</p>
</td></tr>
<tr><td><code id="summary.SGoF_+3A_...">...</code></td>
<td>
<p>Additional arguments affecting the summary produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Rejections</code></td>
<td>
<p>The number of effects declared by SGoF.</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>The estimated false discovery rate.</p>
</td></tr>
<tr><td><code>Adjusted.pvalues</code></td>
<td>
<p>Table of adjusted p-values falling under gamma.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Irene Castro Conde and Jacobo de Uña Álvarez</p>


<h3>References</h3>

<p>Castro Conde I and de Uña Álvarez J (2015). Adjusted p-values for SGoF multiple test procedure.  <em>Biometrical Journal</em>; 57(1): 108-122. DOI: 10.1002/bimj.201300238
</p>
<p>Dalmasso C, Broet P and Moreau T (2005) A simple procedure for estimating the false discovery rate. <em>Bioinformatics</em> 21:660&ndash;668
</p>
<p>de Uña Álvarez J (2011). On the statistical properties of SGoF multitesting method. <em>Statistical Applications in Genetics and Molecular Biology</em>, Vol. 10, Iss. 1, Article 18.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SGoF">SGoF</a></code>,<code><a href="#topic+plot.SGoF">plot.SGoF</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
p&lt;-runif(387)^2  #387 independent p-values, non-uniform intersection null violated

res&lt;-SGoF(p)
summary(res)   #number of rejected nulls, estimated FDR


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
