<!DOCTYPE html><html><head><title>Help for package sharpr2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sharpr2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#sharpr2-package'>
<p>Estimating regularoty scores and identifying high resolution driver elements for ATAC-STARR data</p></a></li>
<li><a href='#call_fdr'><p>call_fdr</p></a></li>
<li><a href='#call_sig_reg'><p>call_sig_reg</p></a></li>
<li><a href='#call_tile_reg'><p>call_tile_reg</p></a></li>
<li><a href='#find_reg'><p>find_reg</p></a></li>
<li><a href='#hidra_ex'>
<p>An example dataset including a region of one chromosome from an ATAC-STARR library</p></a></li>
<li><a href='#plot.sharpr2'><p>plot.sharpr2</p></a></li>
<li><a href='#sharpr2'><p>sharpr2</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Estimating Regulatory Scores and Identifying ATAC-STARR Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Liang He</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Liang He &lt;lianghe@mit.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An algorithm for identifying high-resolution driver elements for datasets from a high-definition reporter assay library. Xinchen Wang, Liang He, Sarah Goggin, Alham Saadat, Li Wang, Melina Claussnitzer, Manolis Kellis (2017) &lt;<a href="https://doi.org/10.1101%2F193136">doi:10.1101/193136</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), methods</td>
</tr>
<tr>
<td>Imports:</td>
<td>mvtnorm (&ge; 1.0), Matrix (&ge; 1.2)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1.9000</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Repository/R-Forge/Project:</td>
<td>sharpr2</td>
</tr>
<tr>
<td>Repository/R-Forge/Revision:</td>
<td>17</td>
</tr>
<tr>
<td>Repository/R-Forge/DateTimeStamp:</td>
<td>2018-05-15 23:53:57</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-05-16 05:06:15 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-05-16 00:15:11 UTC; rforge</td>
</tr>
</table>
<hr>
<h2 id='sharpr2-package'>
Estimating regularoty scores and identifying high resolution driver elements for ATAC-STARR data
</h2><span id='topic+sharpr2-package'></span>

<h3>Description</h3>

<p>The package develops an algorithm for identifying high-resolution driver elements for datasets from an ATAC-STARR library. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> sharpr2</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1.1.0000 </td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-05-12</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Liang He
</p>
<p>Maintainer: Liang He &lt;lianghe@mit.edu&gt;
</p>


<h3>References</h3>

<p>High-resolution genome-wide functional dissection of transcriptional regulatory regions in human.
Xinchen Wang, Liang He, Sarah Goggin, Alham Saadat, Li Wang, Melina Claussnitzer, Manolis Kellis.
bioRxiv 193136; doi: https://doi.org/10.1101/193136
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hidra_ex)
re &lt;- sharpr2(hidra_ex[1:2000,], l_min = 150, l_max = 600, f_dna = 5, f_rna = 0, sig=FALSE)

</code></pre>

<hr>
<h2 id='call_fdr'>call_fdr</h2><span id='topic+call_fdr'></span>

<h3>Description</h3>

<p>Calculate FDR-adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>call_fdr(whole_re, thres_tr = 10, method = 'BH')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="call_fdr_+3A_whole_re">whole_re</code></td>
<td>
<p>A list of the objects obtained from sharpr2 for each chromosome.</p>
</td></tr>
<tr><td><code id="call_fdr_+3A_thres_tr">thres_tr</code></td>
<td>
<p>The threshold for the size of tiled reigons used for calculate FDR-adjusted p-values. The default value is 10.</p>
</td></tr>
<tr><td><code id="call_fdr_+3A_method">method</code></td>
<td>
<p>The method for calculating FDR-adjusted p-values. See the function 'p.adjust' for more details abount the method. The default is 'BH'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gfdr: a result table (data.frame) containing FDR-adjusted p-values, chromosome, region, the size and the index of the tiled region it is located.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(hidra_ex)
whole_re &lt;- sharpr2(hidra_ex, l_min = 150, l_max = 600, f_dna = 5, f_rna = 0, sig=FALSE)
call_fdr(list(whole_re))


</code></pre>

<hr>
<h2 id='call_sig_reg'>call_sig_reg</h2><span id='topic+call_sig_reg'></span>

<h3>Description</h3>

<p>Given an object returned from the sharpr2 function, this function calls significant regions that contain driver elements for a specific tiled region based on a user-defined threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>call_sig_reg(res, nr, threshold = 3.5, win = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="call_sig_reg_+3A_res">res</code></td>
<td>
<p>An object obtained from the sharpr2 function.</p>
</td></tr>
<tr><td><code id="call_sig_reg_+3A_nr">nr</code></td>
<td>
<p>An integer indicating the number of tiled region in res for which driver elements will be called.</p>
</td></tr>
<tr><td><code id="call_sig_reg_+3A_threshold">threshold</code></td>
<td>
<p>The cutoff to identify driver elements in the tiled region. The positions with a z-score larger than the threshold will be called. The default is 3.5.</p>
</td></tr>
<tr><td><code id="call_sig_reg_+3A_win">win</code></td>
<td>
<p>A window size for removing sporadic significant regions. If a significant consecutive region is small than win, it will be treated as false signals. The default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sig_reg: identified regions containing driver elements.
</p>
<p>motif: predicted 20bp core driver elements
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(hidra_ex)
re  &lt;- sharpr2(hidra_ex[1:2000,], l_min = 150, l_max = 600, f_dna = 5, f_rna = 0, sig=TRUE)
call_sig_reg(re,850, threshold=2.5)

</code></pre>

<hr>
<h2 id='call_tile_reg'>call_tile_reg</h2><span id='topic+call_tile_reg'></span>

<h3>Description</h3>

<p>For a HiDRA dataset on a given chromosome, this function calls tile regions (the regions covered by at least one read).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>call_tile_reg(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="call_tile_reg_+3A_data">data</code></td>
<td>
<p>A data.frame for a HiDRA dataset for one chromosome. The data.frame must contain four columns: 'start', 'end', 'PLASMID', 'RNA', and is sorted by 'start'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tile_reg: A list containing the row ids in the data for each tiled region.
</p>
<p>size: The number of reads in each tiled region.
</p>
<p>num_r: The total number of tiled regions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hidra_ex)
tiled &lt;- call_tile_reg(hidra_ex)
</code></pre>

<hr>
<h2 id='find_reg'>find_reg</h2><span id='topic+find_reg'></span>

<h3>Description</h3>

<p>Given an object from sharpr2 and a position, this function finds the tiled region containing the position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_reg(re,pos)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_reg_+3A_re">re</code></td>
<td>
<p>An object obtained from sharpr2.</p>
</td></tr>
<tr><td><code id="find_reg_+3A_pos">pos</code></td>
<td>
<p>A position for which the tiled region is searched.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ind: the index of the tiled region in the object from sharpr2. If no such tile region is found, NA is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hidra_ex)
re &lt;- sharpr2(hidra_ex[1:2000,], l_min = 150, l_max = 600, f_dna = 5, f_rna = 0, sig=FALSE)
find_reg(re,1000000)
</code></pre>

<hr>
<h2 id='hidra_ex'>
An example dataset including a region of one chromosome from an ATAC-STARR library  
</h2><span id='topic+hidra_ex'></span>

<h3>Description</h3>

<p>This is an example dataset containing 10000 fragments with four columns 'start', 'end', 'PLASMID', 'RNA'. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hidra_ex)</code></pre>


<h3>Format</h3>

<p>The format is a data.frame with the columns:
start: the start position of the fragment.
end: the end position of the fragment.
PLASMID: the count of PLASMID for this fragment.
RNA: the count of RNA for this fragment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hidra_ex)
</code></pre>

<hr>
<h2 id='plot.sharpr2'>plot.sharpr2</h2><span id='topic+plot.sharpr2'></span>

<h3>Description</h3>

<p>Given an object returned from the sharpr2 function, this function plots the estimated scores (with s.e. if available) for a tiled region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sharpr2'
plot(x, tr, unc = "CI", loess = FALSE, add = FALSE,
  xlab = "Position", ylab = "Regulatory Score", cicol = 'orange', cimcol = 'grey', 
  sreg = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.sharpr2_+3A_x">x</code></td>
<td>
<p>An object returned from the sharpr2 function.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_tr">tr</code></td>
<td>
<p>An integer indicating which tiled region to be plotted.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_unc">unc</code></td>
<td>
<p>'MSE' or 'CI', indicating whether to plot sqrt(MSE) or 95%CI for uncertainty. The default is 'CI'.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_loess">loess</code></td>
<td>
<p>An indicator for whether the loess method is used for smoothing in plotting the scores from sharpr2. The standard errors are not plotted when loess is used.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_add">add</code></td>
<td>
<p>An indicator for whether to add the new plot to the existing one.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_xlab">xlab</code></td>
<td>
<p>The label for the x-axis. The default is 'Position'.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_ylab">ylab</code></td>
<td>
<p>The label for the y-axis. The default is 'Regulatory Score'.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_cicol">cicol</code></td>
<td>
<p>The color for CIs. The default is 'orange'.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_cimcol">cimcol</code></td>
<td>
<p>The color for filling the regions within CIs. The default is 'grey'.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_sreg">sreg</code></td>
<td>
<p>An indicator for whether to highlight the identified driver element regions. The default is TRUE.</p>
</td></tr>
<tr><td><code id="plot.sharpr2_+3A_...">...</code></td>
<td>
<p>Other parameters for plot.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(hidra_ex)
re &lt;- sharpr2(hidra_ex[1:2000,], l_min = 150, l_max = 600, f_dna = 5, f_rna = 0, sig=FALSE)
plot(re,584)
</code></pre>

<hr>
<h2 id='sharpr2'>sharpr2</h2><span id='topic+sharpr2'></span>

<h3>Description</h3>

<p>For a HiDRA dataset on a given chromosome, this function calls tiled regions (the regions covered by at least one fragment), and calculates regulatory scores for each tiled region. The regulatory scores are based on standardized log(RNA/PLASMID).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sharpr2(data, l_min = 150, l_max = 600, f_rna = 10, f_dna = 0,
  s_a = 300, verbose = FALSE, auto = TRUE, sig = TRUE, len = FALSE, 
  alpha = 0.05, win = 5, mse = FALSE, max_t = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sharpr2_+3A_data">data</code></td>
<td>
<p>A data.frame containing an ATAC-STARR dataset for one chromosome. The data.frame must contain four columns: 'start', 'end', 'PLASMID', 'RNA'. 'PLASMID' and 'RNA' are the values for DNA and RNA, which should be non-negative real numbers (average value over multiple replicates) or integers (counts).</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_l_min">l_min</code></td>
<td>
<p>The fragments with a length smaller than l_min will not be processed. The default is 150.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_l_max">l_max</code></td>
<td>
<p>The fragments with a length larger than l_max will not be processed. The default is 600.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_f_rna">f_rna</code></td>
<td>
<p>The fragments with an RNA count smaller than f_rna will not be processed. The default is 10.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_f_dna">f_dna</code></td>
<td>
<p>The fragments with an DNA count smaller than f_rna will not be processed. The default is 0.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_s_a">s_a</code></td>
<td>
<p>A variance hyperparameter in the prior for the latent regulatory scores. The default is 1000.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_verbose">verbose</code></td>
<td>
<p>An indicator of whether to show processing information. The default is FALSE.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_auto">auto</code></td>
<td>
<p>An indicator of whether to automatically estimate the ridge coefficient <code class="reqn">\lambda</code> from the data for each tiled region using a data-driven way described in the reference. The default is TRUE. If <em>auto</em> is TRUE, <em>s_a</em> is ignored and a ridge coefficient is estimated for each tiled region separately. If <em>auto</em> is FALSE, a global user-defined ridge coefficient (1/s_a) is used.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_sig">sig</code></td>
<td>
<p>An indicator of whether to identify significant motif regions for the estimated scores. Only valid if auto=TRUE. The default is TRUE.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_len">len</code></td>
<td>
<p>An indicator of whether to model log(RNA/PLASMID) of each fragment as the average or the sum of the latent regulatory scores. The default is FALSE, which is the sum.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_alpha">alpha</code></td>
<td>
<p>A regional FWER to call high resolution driver elements (the significant regulatory region). The default is 0.05.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_win">win</code></td>
<td>
<p>A window size for removing sporadic identified significant regions. If a significant consecutive region is small than <em>win</em>, it will be treated as false signals. The default is 5.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_mse">mse</code></td>
<td>
<p>An indicator of whether mean square errors are included in the output results. The default is FALSE.</p>
</td></tr>
<tr><td><code id="sharpr2_+3A_max_t">max_t</code></td>
<td>
<p>A value between 0 and 1, indicating the proportion of non-zero eigenvectors used to calculate <code class="reqn">\lambda</code> when auto=TRUE. The default is 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default value of s_a is set to be 300, which is equivalent to a ridge coefficient of 0.0033. This default ridge coefficient value is selected by the median of the estimated <code class="reqn">\lambda</code> from the first library. 
</p>


<h3>Value</h3>

<p>score: the regulatory scores for each tiled region. This list contains four components: est_a (the regulatory scores at each locus), sd_e (the sqare root of the mean square error), var_nb (the variance of the esitmate at each locus), <code class="reqn">\lambda</code> (the ridge coefficient).
</p>
<p>region: the start and end positions for each tiled region.
</p>
<p>n_reg: total number of tiled regions.
</p>
<p>n_read: the number of reads in each tiled region.
</p>
<p>sig_reg: identified high resolution driver elements based on the cutoff.
</p>
<p>motif: predicted 20bp motifs 
</p>
<p>cutoff: the cutoff used to call high resolution driver elements for the tiled region.
</p>


<h3>References</h3>

<p>Xinchen Wang, Liang He, Sarah Goggin, Alham Saadat, Li Wang, Melina Claussnitzer, Manolis Kellis. 
High-resolution genome-wide functional dissection of transcriptional regulatory regions in human.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hidra_ex)
re &lt;- sharpr2(hidra_ex[1:2000,], l_min = 150, l_max = 600, f_dna = 5, f_rna = 0, sig=FALSE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
