<!DOCTYPE html><html lang="en"><head><title>Help for package shinyWGD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {shinyWGD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#analysisEachCluster'><p>Perform synteny analysis for identified clusters</p></a></li>
<li><a href='#bootStrapPeaks'><p>Bootstrap Peaks in the Ks Distribution</p></a></li>
<li><a href='#calculateKsDistribution4wgd_multiple'><p>Calculate the Ks Distribution for Multiple Speices</p></a></li>
<li><a href='#CalHomoConcentration'><p>Compute the -log10 of Poisson Distribution</p></a></li>
<li><a href='#CalPvalue'><p>Compute the P-value of a Cluster using the Poisson Distribution</p></a></li>
<li><a href='#check_gff_from_file'><p>Check and Process GFF Input File from a Specific Path</p></a></li>
<li><a href='#check_gff_input'><p>Check and Prepare GFF/GTF Input File</p></a></li>
<li><a href='#check_proteome_from_file'><p>Check and Process Proteome Input File From a Special Path</p></a></li>
<li><a href='#check_proteome_input'><p>Check and Process Proteome Input File</p></a></li>
<li><a href='#checkFileExistence'><p>Check File Existence in a Data Table</p></a></li>
<li><a href='#cluster_synteny'><p>Cluster Synteny Data and Generate Trees</p></a></li>
<li><a href='#computing_depth'><p>Compute the Depth of Anchored Points</p></a></li>
<li><a href='#computing_depth_paranome'><p>Compute the Depth of Anchored Points in a Paranome Comparison</p></a></li>
<li><a href='#CountOrthologs'><p>Count Ortholog Genes in a Species</p></a></li>
<li><a href='#create_ksrates_cmd'><p>Create Ksrates Command Files from Shiny Input</p></a></li>
<li><a href='#create_ksrates_cmd_from_table'><p>Create Ksrates Command Files from Data Table</p></a></li>
<li><a href='#create_ksrates_configure_file_based_on_table'><p>Create Ksrates Configuration File Based on Data Table</p></a></li>
<li><a href='#create_ksrates_configure_file_v2'><p>Create Ksrates Configuration File</p></a></li>
<li><a href='#create_ksrates_expert_parameter_file'><p>Create ksrates Expert Parameter File</p></a></li>
<li><a href='#dfltBWrange'><p>dfltBWrange</p></a></li>
<li><a href='#dfltCounts'><p>dfltCounts</p></a></li>
<li><a href='#downloadButton_custom'><p>Creating a Custom Download Button</p></a></li>
<li><a href='#drvkde'><p>drvkde</p></a></li>
<li><a href='#extract_first_part'><p>Extract the first part of a string by splitting it at tab characters.</p></a></li>
<li><a href='#extractCluster'><p>Extract clusters based on specified scaffolds</p></a></li>
<li><a href='#find_peaks'><p>Find Peaks in a Numeric Vector</p></a></li>
<li><a href='#generate_ksd'><p>Generate Kernel Density Estimates (KDE) for Ks Distribution</p></a></li>
<li><a href='#generateKsDistribution'><p>Generate the Ks Distribution</p></a></li>
<li><a href='#get_segments'><p>Get Segmented Data from Anchorpoints and Ks Values</p></a></li>
<li><a href='#is_fasta_cds'><p>Check if a file is in FASTA format with cds sequences.</p></a></li>
<li><a href='#is.ksv'><p>Check if an object is of class &quot;ksv&quot;</p></a></li>
<li><a href='#is.not.null'><p>Check if an Object is Not NULL</p></a></li>
<li><a href='#ks_mclust_v2'><p>ks_mclust_v2</p></a></li>
<li><a href='#map_informal_name_to_latin_name'><p>Map Informal Names to Latin Names</p></a></li>
<li><a href='#mix_logNormal_Ks'><p>Log-Normal mixturing analyses of a Ks distributions for the whole paranome</p></a></li>
<li><a href='#modeFinder'><p>modeFinder</p></a></li>
<li><a href='#obtain_chromosome_length'><p>obtain_chromosome_length</p></a></li>
<li><a href='#obtain_chromosome_length_filter'><p>obtain_chromosome_length_filter</p></a></li>
<li><a href='#obtain_coordiantes_for_anchorpoints'><p>Obtain coordinates for anchorpoints from GFF files</p></a></li>
<li><a href='#obtain_coordiantes_for_anchorpoints_ks'><p>Obtain Coordinates and Ks Values for Anchorpoints</p></a></li>
<li><a href='#obtain_coordiantes_for_segments'><p>Obtain coordinates for segments in a comparison</p></a></li>
<li><a href='#obtain_coordinates_for_segments_multiple'><p>Obtain Coordinates for Segments in Multiple Synteny Blocks</p></a></li>
<li><a href='#obtain_mean_ks_for_each_multiplicon'><p>Compute the Mean of Ks values for Each Multiplicon</p></a></li>
<li><a href='#parse_EMMIX'><p>Read the EMMIX output for a range of components</p></a></li>
<li><a href='#parse_one_EMMIX'><p>Read the EMMIX output for a specify number of components</p></a></li>
<li><a href='#PeaksInKsDistributionValues'><p>Find Peaks in the Ks Distribution</p></a></li>
<li><a href='#read_data_file'><p>Read Data from Uploaded File</p></a></li>
<li><a href='#read.wgd_ksd'><p>Read the output file of wgd ksd</p></a></li>
<li><a href='#relativeRate'><p>relativeRate</p></a></li>
<li><a href='#remove_inner_stop_codon_sequence'><p>Remove Genes Contain Stop Codons within the Sequence</p></a></li>
<li><a href='#remove_old_dirs'><p>Remove directories older than a specified day</p></a></li>
<li><a href='#replace_informal_name_to_latin_name'><p>Replace Informal Names with Latin Names</p></a></li>
<li><a href='#resampleKsDistribution'><p>Resample a Ks Distribution</p></a></li>
<li><a href='#run_emmix_kmeas'><p>A wrapper to run EM analysis of \(ln\) Ks values with k-means</p></a></li>
<li><a href='#runshinyWGD'><p>The main code to run shinyWGD</p></a></li>
<li><a href='#SignifFeatureRegion'><p>SignifFeatureRegion</p></a></li>
<li><a href='#SiZer'><p>SiZer (Significant Zero Crossings)</p></a></li>
<li><a href='#symconv.ks'><p>symconv.ks</p></a></li>
<li><a href='#symconv2D.ks'><p>symconv2D.ks</p></a></li>
<li><a href='#symconv3D.ks'><p>symconv3D.ks</p></a></li>
<li><a href='#symconv4D.ks'><p>symconv4D.ks</p></a></li>
<li><a href='#TimeTreeFecher'><p>Extracts a timetree from TimeTree.org based on species names.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>'Shiny' Application for Whole Genome Duplication Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jia Li &lt;li081766@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a comprehensive 'Shiny' application for analyzing 
    Whole Genome Duplication ('WGD') events.
    This package provides a user-friendly 'Shiny' web application for 
    non-experienced researchers to prepare input data and execute command lines 
    for several well-known 'WGD' analysis tools, including 'wgd', 'ksrates', 
    'i-ADHoRe', 'OrthoFinder', and 'Whale'. This package also provides the source code 
    for experienced researchers to adjust and install the package to their own server. 
    Key Features
    1) Input Data Preparation 
    This package allows users to conveniently upload and format their data, 
	making it compatible with various 'WGD' analysis tools.
    2) Command Line Generation
    This package automatically generates the necessary command lines for 
	selected 'WGD' analysis tools, reducing manual errors and saving time.
    3) Visualization
    This package offers interactive visualizations to explore and interpret 'WGD' results, 
	facilitating in-depth 'WGD' analysis.
    4) Comparative Genomics
    Users can study and compare 'WGD' events across different species, 
	aiding in evolutionary and comparative genomics studies.
    5) User-Friendly Interface 
	This 'Shiny' web application provides an intuitive and accessible interface, 
	making 'WGD' analysis accessible to researchers and 'bioinformaticians' of all levels.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>pandoc (&gt;= 1.12.3), pandoc-citeproc</td>
</tr>
<tr>
<td>Imports:</td>
<td>shiny, shinyalert, stringr, vroom, fs, tidyr, data.table,
dplyr, ape, ks, mclust, htmltools, seqinr, httr, jsonlite</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tidyverse, knitr, rmarkdown, DT, argparse, bslib, bsplus,
english, fontawesome, igraph, shinyBS, shinyFiles,
shinyWidgets, shinyjs, stringi, tools, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-13 01:08:49 UTC; jiali</td>
</tr>
<tr>
<td>Author:</td>
<td>Jia Li [aut, cre],
  Zhen Li [ctb],
  Arthur Zwaenepoel [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-13 15:10:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='analysisEachCluster'>Perform synteny analysis for identified clusters</h2><span id='topic+analysisEachCluster'></span>

<h3>Description</h3>

<p>This function performs synteny analysis for clusters identified by hierarchical clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analysisEachCluster(
  segmented_file,
  segmented_anchorpoints_file,
  genes_file,
  cluster_info_file,
  identified_cluster_file,
  hcheight = 0.3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analysisEachCluster_+3A_segmented_file">segmented_file</code></td>
<td>
<p>The path to the segmented chromosome file.</p>
</td></tr>
<tr><td><code id="analysisEachCluster_+3A_segmented_anchorpoints_file">segmented_anchorpoints_file</code></td>
<td>
<p>The path to the segmented anchorpoints file.</p>
</td></tr>
<tr><td><code id="analysisEachCluster_+3A_genes_file">genes_file</code></td>
<td>
<p>genes.txt created by i-ADHoRe.</p>
</td></tr>
<tr><td><code id="analysisEachCluster_+3A_cluster_info_file">cluster_info_file</code></td>
<td>
<p>The path to the clustering information file.</p>
</td></tr>
<tr><td><code id="analysisEachCluster_+3A_identified_cluster_file">identified_cluster_file</code></td>
<td>
<p>The path to the output file for identified clusters.</p>
</td></tr>
<tr><td><code id="analysisEachCluster_+3A_hcheight">hcheight</code></td>
<td>
<p>The cutoff height for cluster identification (default: 0.3).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing information about identified clusters and their p-values.
</p>

<hr>
<h2 id='bootStrapPeaks'>Bootstrap Peaks in the Ks Distribution</h2><span id='topic+bootStrapPeaks'></span>

<h3>Description</h3>

<p>This function performs bootstrapping on a given Ks (synonymous substitution rates) distribution
to estimate peaks within the distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootStrapPeaks(
  ksRaw,
  binWidth = 0.1,
  maxK = 5,
  m = 3,
  peak.index = 1,
  peak.maxK = 2,
  spar = 0.25,
  rep = 1000,
  from = 0,
  to = maxK
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bootStrapPeaks_+3A_ksraw">ksRaw</code></td>
<td>
<p>A numeric vector representing the raw Ks distribution to be bootstrapped.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_binwidth">binWidth</code></td>
<td>
<p>A numeric value indicating the bin width for histogram calculation.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_maxk">maxK</code></td>
<td>
<p>A numeric value indicating the maximum Ks value to consider in the distribution.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_m">m</code></td>
<td>
<p>An integer specifying the parameter for peak detection.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_peak.index">peak.index</code></td>
<td>
<p>An integer indicating the index of the peak to be estimated.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_peak.maxk">peak.maxK</code></td>
<td>
<p>A numeric value indicating the maximum Ks value for peak estimation.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_spar">spar</code></td>
<td>
<p>A numeric value controlling the smoothness of spline fitting.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_rep">rep</code></td>
<td>
<p>An integer specifying the number of bootstrap repetitions.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_from">from</code></td>
<td>
<p>A numeric value indicating the lower bound for peak estimation.</p>
</td></tr>
<tr><td><code id="bootStrapPeaks_+3A_to">to</code></td>
<td>
<p>A numeric value indicating the upper bound for peak estimation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector containing bootstrapped peak estimates.
</p>

<hr>
<h2 id='calculateKsDistribution4wgd_multiple'>Calculate the Ks Distribution for Multiple Speices</h2><span id='topic+calculateKsDistribution4wgd_multiple'></span>

<h3>Description</h3>

<p>This function takes a list of data files, calculates the Ks distribution, and returns the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateKsDistribution4wgd_multiple(
  files_list,
  binWidth = 0.1,
  maxK = 5,
  plot.mode = "weighted",
  include.outliers = FALSE,
  minK = 0,
  minAlnLen = 0,
  minIdn = 0,
  minCov = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_files_list">files_list</code></td>
<td>
<p>A list of file paths containing Ks data.</p>
</td></tr>
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_binwidth">binWidth</code></td>
<td>
<p>The width of Ks bins for the distribution.</p>
</td></tr>
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_maxk">maxK</code></td>
<td>
<p>The maximum Ks value to consider.</p>
</td></tr>
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_plot.mode">plot.mode</code></td>
<td>
<p>The mode for plotting (&quot;weighted&quot;, &quot;average&quot;, &quot;min&quot;, or &quot;pairwise&quot;).</p>
</td></tr>
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_include.outliers">include.outliers</code></td>
<td>
<p>Whether to include outliers in the calculation.</p>
</td></tr>
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_mink">minK</code></td>
<td>
<p>The minimum Ks value to include in the distribution.</p>
</td></tr>
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_minalnlen">minAlnLen</code></td>
<td>
<p>The minimum alignment length to include in the distribution.</p>
</td></tr>
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_minidn">minIdn</code></td>
<td>
<p>The minimum alignment identity to include in the distribution.</p>
</td></tr>
<tr><td><code id="calculateKsDistribution4wgd_multiple_+3A_mincov">minCov</code></td>
<td>
<p>The minimum alignment coverage to include in the distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two data frames: &quot;bar&quot; for Ks distribution and &quot;density&quot; for density data.
</p>

<hr>
<h2 id='CalHomoConcentration'>Compute the -log10 of Poisson Distribution</h2><span id='topic+CalHomoConcentration'></span>

<h3>Description</h3>

<p>This function calculates the -log10 of the p-value of a Poisson distribution given the parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalHomoConcentration(m, n, q, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CalHomoConcentration_+3A_m">m</code></td>
<td>
<p>The total number of trials.</p>
</td></tr>
<tr><td><code id="CalHomoConcentration_+3A_n">n</code></td>
<td>
<p>The total number of possible outcomes.</p>
</td></tr>
<tr><td><code id="CalHomoConcentration_+3A_q">q</code></td>
<td>
<p>The observed number of successful outcomes.</p>
</td></tr>
<tr><td><code id="CalHomoConcentration_+3A_k">k</code></td>
<td>
<p>The expected number of successful outcomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The -log10 of the p-value.
</p>

<hr>
<h2 id='CalPvalue'>Compute the P-value of a Cluster using the Poisson Distribution</h2><span id='topic+CalPvalue'></span>

<h3>Description</h3>

<p>This function computes the P-value of a cluster using the Poisson distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalPvalue(m, n, q, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CalPvalue_+3A_m">m</code></td>
<td>
<p>The total number of all anchored points.</p>
</td></tr>
<tr><td><code id="CalPvalue_+3A_n">n</code></td>
<td>
<p>The product of the remapped gene number of the query species and subject species.</p>
</td></tr>
<tr><td><code id="CalPvalue_+3A_q">q</code></td>
<td>
<p>The number of anchored points in the cluster.</p>
</td></tr>
<tr><td><code id="CalPvalue_+3A_k">k</code></td>
<td>
<p>The product of the remapped gene number of the segmented chromosomes
of the query species and subject species.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed P-value.
</p>

<hr>
<h2 id='check_gff_from_file'>Check and Process GFF Input File from a Specific Path</h2><span id='topic+check_gff_from_file'></span>

<h3>Description</h3>

<p>This function checks the type of GFF input file specified by its path and processes it accordingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gff_from_file(gff_input_name, gff_input_path, working_wd)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_gff_from_file_+3A_gff_input_name">gff_input_name</code></td>
<td>
<p>The informal name of the GFF input file.</p>
</td></tr>
<tr><td><code id="check_gff_from_file_+3A_gff_input_path">gff_input_path</code></td>
<td>
<p>The path to the GFF input file.</p>
</td></tr>
<tr><td><code id="check_gff_from_file_+3A_working_wd">working_wd</code></td>
<td>
<p>A character string specifying the working directory to be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string containing the processed GFF file's path.
</p>

<hr>
<h2 id='check_gff_input'>Check and Prepare GFF/GTF Input File</h2><span id='topic+check_gff_input'></span>

<h3>Description</h3>

<p>This function checks the file format of a GFF/GTF input file and prepares it for analysis. It can handle both uncompressed and compressed formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gff_input(gff_input_name, gff_input_path, working_wd)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_gff_input_+3A_gff_input_name">gff_input_name</code></td>
<td>
<p>A descriptive name for the GFF/GTF file.</p>
</td></tr>
<tr><td><code id="check_gff_input_+3A_gff_input_path">gff_input_path</code></td>
<td>
<p>The file path to the GFF/GTF file.</p>
</td></tr>
<tr><td><code id="check_gff_input_+3A_working_wd">working_wd</code></td>
<td>
<p>A character string specifying the working directory to be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path to the prepared GFF file for analysis.
</p>

<hr>
<h2 id='check_proteome_from_file'>Check and Process Proteome Input File From a Special Path</h2><span id='topic+check_proteome_from_file'></span>

<h3>Description</h3>

<p>This function checks the type of proteome input file and processes it accordingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_proteome_from_file(proteome_name, proteome_input, working_wd)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_proteome_from_file_+3A_proteome_name">proteome_name</code></td>
<td>
<p>The informal name of the proteome input file.</p>
</td></tr>
<tr><td><code id="check_proteome_from_file_+3A_proteome_input">proteome_input</code></td>
<td>
<p>The proteome input data.</p>
</td></tr>
<tr><td><code id="check_proteome_from_file_+3A_working_wd">working_wd</code></td>
<td>
<p>A character string specifying the working directory to be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string containing the processed proteome file's path.
</p>

<hr>
<h2 id='check_proteome_input'>Check and Process Proteome Input File</h2><span id='topic+check_proteome_input'></span>

<h3>Description</h3>

<p>This function checks the type of proteome input file and processes it accordingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_proteome_input(proteome_name, proteome_input, working_wd)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_proteome_input_+3A_proteome_name">proteome_name</code></td>
<td>
<p>The informal name of the proteome input file.</p>
</td></tr>
<tr><td><code id="check_proteome_input_+3A_proteome_input">proteome_input</code></td>
<td>
<p>The proteome input data.</p>
</td></tr>
<tr><td><code id="check_proteome_input_+3A_working_wd">working_wd</code></td>
<td>
<p>A character string specifying the working directory to be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string containing the processed proteome file's path.
</p>

<hr>
<h2 id='checkFileExistence'>Check File Existence in a Data Table</h2><span id='topic+checkFileExistence'></span>

<h3>Description</h3>

<p>This function checks the existence of files specified in a data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkFileExistence(data_table, working_wd)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkFileExistence_+3A_data_table">data_table</code></td>
<td>
<p>A data table with file paths in columns V2 and V3.</p>
</td></tr>
<tr><td><code id="checkFileExistence_+3A_working_wd">working_wd</code></td>
<td>
<p>A path of the working directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function has no return value. It prints messages to the console.
</p>

<hr>
<h2 id='cluster_synteny'>Cluster Synteny Data and Generate Trees</h2><span id='topic+cluster_synteny'></span>

<h3>Description</h3>

<p>This function clusters synteny data based on calculated p-values and generates trees
for both column-based and row-based clustering. It then saves the cluster information and
trees to output files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_synteny(
  segmented_file,
  segmented_anchorpoints_file,
  genes_file,
  out_file
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cluster_synteny_+3A_segmented_file">segmented_file</code></td>
<td>
<p>A character string specifying the file path for segmented data.</p>
</td></tr>
<tr><td><code id="cluster_synteny_+3A_segmented_anchorpoints_file">segmented_anchorpoints_file</code></td>
<td>
<p>A character string specifying the file path for segmented anchorpoints.</p>
</td></tr>
<tr><td><code id="cluster_synteny_+3A_genes_file">genes_file</code></td>
<td>
<p>A character string specifying the file path for genes information created by i-ADHoRe.</p>
</td></tr>
<tr><td><code id="cluster_synteny_+3A_out_file">out_file</code></td>
<td>
<p>A character string specifying the output file path for saving cluster information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL (output files are generated with the specified information).
</p>

<hr>
<h2 id='computing_depth'>Compute the Depth of Anchored Points</h2><span id='topic+computing_depth'></span>

<h3>Description</h3>

<p>This function calculates the depth of anchored points based on the provided parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computing_depth(
  anchorpoint_ks_file,
  multiplicon_id,
  selected_query_chr,
  selected_subject_chr = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="computing_depth_+3A_anchorpoint_ks_file">anchorpoint_ks_file</code></td>
<td>
<p>The file containing anchorpoint and Ks data.</p>
</td></tr>
<tr><td><code id="computing_depth_+3A_multiplicon_id">multiplicon_id</code></td>
<td>
<p>The ID of the multiplicon to consider.</p>
</td></tr>
<tr><td><code id="computing_depth_+3A_selected_query_chr">selected_query_chr</code></td>
<td>
<p>A list of selected query chromosomes.</p>
</td></tr>
<tr><td><code id="computing_depth_+3A_selected_subject_chr">selected_subject_chr</code></td>
<td>
<p>A list of selected subject chromosomes (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing depth data frames, including &quot;query_depth&quot; and &quot;subject_depth&quot; if subject chromosomes are specified, or &quot;depth&quot; if not.
</p>

<hr>
<h2 id='computing_depth_paranome'>Compute the Depth of Anchored Points in a Paranome Comparison</h2><span id='topic+computing_depth_paranome'></span>

<h3>Description</h3>

<p>This function computes the depth of anchored points in a paranome comparison based on the provided parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computing_depth_paranome(
  anchorpoint_ks_file,
  multiplicon_id,
  selected_query_chr
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="computing_depth_paranome_+3A_anchorpoint_ks_file">anchorpoint_ks_file</code></td>
<td>
<p>The file containing anchor point and Ks value data.</p>
</td></tr>
<tr><td><code id="computing_depth_paranome_+3A_multiplicon_id">multiplicon_id</code></td>
<td>
<p>The IDs of the multiplicons to consider.</p>
</td></tr>
<tr><td><code id="computing_depth_paranome_+3A_selected_query_chr">selected_query_chr</code></td>
<td>
<p>The list of selected query chromosomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the depth dataframe.
</p>

<hr>
<h2 id='CountOrthologs'>Count Ortholog Genes in a Species</h2><span id='topic+CountOrthologs'></span>

<h3>Description</h3>

<p>This function counts ortholog genes in a given species based on input data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CountOrthologs(atomic.df, species)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CountOrthologs_+3A_atomic.df">atomic.df</code></td>
<td>
<p>A data frame containing information about ortholog genes.
It should have the following columns:
- multiplicon: The multiplicon identifier.
- geneX: The gene identifier in speciesX.
- speciesX: The species name for geneX.
- listX: The chromosome or list identifier for geneX.
- coordX: The coordinate information for geneX.
- geneY: The gene identifier in speciesY.
- speciesY: The species name for geneY.
- listY: The chromosome or list identifier for geneY.
- coordY: The coordinate information for geneY.
- level: The orthology level.
- num_anchors: The number of anchors.
- is_real: A flag indicating if the data is real.
- Ks: The Ks value.</p>
</td></tr>
<tr><td><code id="CountOrthologs_+3A_species">species</code></td>
<td>
<p>The species for which ortholog gene counts should be computed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame summarizing the counts of ortholog genes for each chromosome.
</p>

<hr>
<h2 id='create_ksrates_cmd'>Create Ksrates Command Files from Shiny Input</h2><span id='topic+create_ksrates_cmd'></span>

<h3>Description</h3>

<p>Create Ksrates Command Files from Shiny Input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ksrates_cmd(input, ksratesconf, cmd_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_ksrates_cmd_+3A_input">input</code></td>
<td>
<p>The Input object of Shiny.</p>
</td></tr>
<tr><td><code id="create_ksrates_cmd_+3A_ksratesconf">ksratesconf</code></td>
<td>
<p>The path to the Ksrates configuration file.</p>
</td></tr>
<tr><td><code id="create_ksrates_cmd_+3A_cmd_file">cmd_file</code></td>
<td>
<p>The path to the main Ksrates command file to be generated.</p>
</td></tr>
</table>

<hr>
<h2 id='create_ksrates_cmd_from_table'>Create Ksrates Command Files from Data Table</h2><span id='topic+create_ksrates_cmd_from_table'></span>

<h3>Description</h3>

<p>This function generates command files for running Ksrates and related analyses based on a data table and configuration file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ksrates_cmd_from_table(data_table, ksratesconf, cmd_file, focal_species)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_ksrates_cmd_from_table_+3A_data_table">data_table</code></td>
<td>
<p>The data table containing information about species.</p>
</td></tr>
<tr><td><code id="create_ksrates_cmd_from_table_+3A_ksratesconf">ksratesconf</code></td>
<td>
<p>The path to the Ksrates configuration file.</p>
</td></tr>
<tr><td><code id="create_ksrates_cmd_from_table_+3A_cmd_file">cmd_file</code></td>
<td>
<p>The path to the main Ksrates command file to be generated.</p>
</td></tr>
<tr><td><code id="create_ksrates_cmd_from_table_+3A_focal_species">focal_species</code></td>
<td>
<p>The name of the focal species.</p>
</td></tr>
</table>

<hr>
<h2 id='create_ksrates_configure_file_based_on_table'>Create Ksrates Configuration File Based on Data Table</h2><span id='topic+create_ksrates_configure_file_based_on_table'></span>

<h3>Description</h3>

<p>This function generates a Ksrates configuration file based on a data table and other parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ksrates_configure_file_based_on_table(
  data_table,
  focal_species,
  newick_tree_file,
  ksrates_conf_file,
  species_info_file,
  working_wd
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_ksrates_configure_file_based_on_table_+3A_data_table">data_table</code></td>
<td>
<p>The data table containing information about species, proteomes, and GFF files.</p>
</td></tr>
<tr><td><code id="create_ksrates_configure_file_based_on_table_+3A_focal_species">focal_species</code></td>
<td>
<p>The name of the focal species.</p>
</td></tr>
<tr><td><code id="create_ksrates_configure_file_based_on_table_+3A_newick_tree_file">newick_tree_file</code></td>
<td>
<p>The path to the Newick tree file.</p>
</td></tr>
<tr><td><code id="create_ksrates_configure_file_based_on_table_+3A_ksrates_conf_file">ksrates_conf_file</code></td>
<td>
<p>The path to the Ksrates configuration file to be generated.</p>
</td></tr>
<tr><td><code id="create_ksrates_configure_file_based_on_table_+3A_species_info_file">species_info_file</code></td>
<td>
<p>The path to the species information file.</p>
</td></tr>
<tr><td><code id="create_ksrates_configure_file_based_on_table_+3A_working_wd">working_wd</code></td>
<td>
<p>A character string specifying the working directory to be used.</p>
</td></tr>
</table>

<hr>
<h2 id='create_ksrates_configure_file_v2'>Create Ksrates Configuration File</h2><span id='topic+create_ksrates_configure_file_v2'></span>

<h3>Description</h3>

<p>This function generates a configuration file for the Ksrates pipeline based on Shiny input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ksrates_configure_file_v2(input, ksrates_conf_file, species_info_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_ksrates_configure_file_v2_+3A_input">input</code></td>
<td>
<p>The Input object of Shiny.</p>
</td></tr>
<tr><td><code id="create_ksrates_configure_file_v2_+3A_ksrates_conf_file">ksrates_conf_file</code></td>
<td>
<p>The path to the Ksrates configuration file.</p>
</td></tr>
<tr><td><code id="create_ksrates_configure_file_v2_+3A_species_info_file">species_info_file</code></td>
<td>
<p>The path to the species information file.</p>
</td></tr>
</table>

<hr>
<h2 id='create_ksrates_expert_parameter_file'>Create ksrates Expert Parameter File</h2><span id='topic+create_ksrates_expert_parameter_file'></span>

<h3>Description</h3>

<p>Create ksrates Expert Parameter File
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ksrates_expert_parameter_file(ksrates_expert_parameter_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_ksrates_expert_parameter_file_+3A_ksrates_expert_parameter_file">ksrates_expert_parameter_file</code></td>
<td>
<p>The file is used to store the ksrates expert parameter</p>
</td></tr>
</table>

<hr>
<h2 id='dfltBWrange'>dfltBWrange</h2><span id='topic+dfltBWrange'></span>

<h3>Description</h3>

<p>This function computes the default bandwidth range for kernel density estimation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfltBWrange(x, tau)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dfltBWrange_+3A_x">x</code></td>
<td>
<p>The input data, which can be a numeric vector or matrix.</p>
</td></tr>
<tr><td><code id="dfltBWrange_+3A_tau">tau</code></td>
<td>
<p>A parameter used in bandwidth calculation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of bandwidth ranges for each dimension of the input data.
</p>

<hr>
<h2 id='dfltCounts'>dfltCounts</h2><span id='topic+dfltCounts'></span>

<h3>Description</h3>

<p>This function bins the input data into a regular grid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfltCounts(
  x,
  gridsize = rep(64, NCOL(x)),
  h = rep(0, NCOL(x)),
  supp = 3.7,
  range.x,
  w
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dfltCounts_+3A_x">x</code></td>
<td>
<p>The input data, which should be a numeric matrix.</p>
</td></tr>
<tr><td><code id="dfltCounts_+3A_gridsize">gridsize</code></td>
<td>
<p>A vector specifying the number of bins along each dimension.</p>
</td></tr>
<tr><td><code id="dfltCounts_+3A_h">h</code></td>
<td>
<p>A vector specifying the bandwidth (smoothing parameter) along each dimension.</p>
</td></tr>
<tr><td><code id="dfltCounts_+3A_supp">supp</code></td>
<td>
<p>A parameter for determining the range of the bins.</p>
</td></tr>
<tr><td><code id="dfltCounts_+3A_range.x">range.x</code></td>
<td>
<p>A list specifying the range of values for each dimension.</p>
</td></tr>
<tr><td><code id="dfltCounts_+3A_w">w</code></td>
<td>
<p>A vector of weights for the data points.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the binned counts and the range of values for each dimension.
</p>

<hr>
<h2 id='downloadButton_custom'>Creating a Custom Download Button</h2><span id='topic+downloadButton_custom'></span>

<h3>Description</h3>

<p>Use this function to create a custom download button or link. When clicked, it will initiate a browser download. The filename and contents are specified by the corresponding downloadHandler() defined in the server function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downloadButton_custom(
  outputId,
  label = "Download",
  class = NULL,
  status = "primary",
  ...,
  icon = shiny::icon("download")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downloadButton_custom_+3A_outputid">outputId</code></td>
<td>
<p>The name of the output slot that the downloadHandler is assigned to.</p>
</td></tr>
<tr><td><code id="downloadButton_custom_+3A_label">label</code></td>
<td>
<p>The label that should appear on the button.</p>
</td></tr>
<tr><td><code id="downloadButton_custom_+3A_class">class</code></td>
<td>
<p>Additional CSS classes to apply to the tag, if any. Default NULL.</p>
</td></tr>
<tr><td><code id="downloadButton_custom_+3A_status">status</code></td>
<td>
<p>The status of the button; default is &quot;primary.&quot;</p>
</td></tr>
<tr><td><code id="downloadButton_custom_+3A_...">...</code></td>
<td>
<p>Other arguments to pass to the container tag function.</p>
</td></tr>
<tr><td><code id="downloadButton_custom_+3A_icon">icon</code></td>
<td>
<p>An icon() to appear on the button; default is icon(&quot;download&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An HTML tag to allow users to download the object.
</p>

<hr>
<h2 id='drvkde'>drvkde</h2><span id='topic+drvkde'></span>

<h3>Description</h3>

<p>Compute the mth derivative of a binned d-variate kernel density estimate based on grid counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drvkde(x, drv, bandwidth, gridsize, range.x, binned = FALSE, se = TRUE, w)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="drvkde_+3A_x">x</code></td>
<td>
<p>The input data.</p>
</td></tr>
<tr><td><code id="drvkde_+3A_drv">drv</code></td>
<td>
<p>The order of the derivative to compute.</p>
</td></tr>
<tr><td><code id="drvkde_+3A_bandwidth">bandwidth</code></td>
<td>
<p>The bandwidth (smoothing parameter) along each dimension.</p>
</td></tr>
<tr><td><code id="drvkde_+3A_gridsize">gridsize</code></td>
<td>
<p>The size of the grid.</p>
</td></tr>
<tr><td><code id="drvkde_+3A_range.x">range.x</code></td>
<td>
<p>A list specifying the range of values for each dimension.</p>
</td></tr>
<tr><td><code id="drvkde_+3A_binned">binned</code></td>
<td>
<p>A logical indicating whether the input data is already binned.</p>
</td></tr>
<tr><td><code id="drvkde_+3A_se">se</code></td>
<td>
<p>A logical indicating whether to compute standard errors.</p>
</td></tr>
<tr><td><code id="drvkde_+3A_w">w</code></td>
<td>
<p>A vector of weights for the data points.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the estimated density or derivative, and optionally, standard errors.
</p>

<hr>
<h2 id='extract_first_part'>Extract the first part of a string by splitting it at tab characters.</h2><span id='topic+extract_first_part'></span>

<h3>Description</h3>

<p>This function takes a string and splits it at tab characters. It then
returns the first part of the resulting character vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_first_part(name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_first_part_+3A_name">name</code></td>
<td>
<p>The input string to be split.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the first part of the input string.
</p>

<hr>
<h2 id='extractCluster'>Extract clusters based on specified scaffolds</h2><span id='topic+extractCluster'></span>

<h3>Description</h3>

<p>This function extracts clusters based on the specified scaffolds for both query and subject species.
It filters the data frames containing segment information and atomic anchorpoints to retain only the relevant clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCluster(segs.df, atomic.df, scaf.bycol, scaf.byrow)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extractCluster_+3A_segs.df">segs.df</code></td>
<td>
<p>A data frame containing segment information.</p>
</td></tr>
<tr><td><code id="extractCluster_+3A_atomic.df">atomic.df</code></td>
<td>
<p>A data frame containing atomic anchorpoints.</p>
</td></tr>
<tr><td><code id="extractCluster_+3A_scaf.bycol">scaf.bycol</code></td>
<td>
<p>A character vector specifying scaffolds for the query species.</p>
</td></tr>
<tr><td><code id="extractCluster_+3A_scaf.byrow">scaf.byrow</code></td>
<td>
<p>A character vector specifying scaffolds for the subject species.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two data frames: &quot;segs&quot; for segment information and &quot;atomic&quot; for atomic anchorpoints.
</p>

<hr>
<h2 id='find_peaks'>Find Peaks in a Numeric Vector</h2><span id='topic+find_peaks'></span>

<h3>Description</h3>

<p>This function identifies peaks in a numeric vector by analyzing the shape of the curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_peaks(x, m = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_peaks_+3A_x">x</code></td>
<td>
<p>A numeric vector in which peaks will be identified.</p>
</td></tr>
<tr><td><code id="find_peaks_+3A_m">m</code></td>
<td>
<p>An integer indicating the half-width of the neighborhood to consider when identifying peaks. A larger value of <code>m</code> makes peak detection less sensitive.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector containing the indices of the identified peaks in the input vector <code>x</code>.
</p>

<hr>
<h2 id='generate_ksd'>Generate Kernel Density Estimates (KDE) for Ks Distribution</h2><span id='topic+generate_ksd'></span>

<h3>Description</h3>

<p>This function generates Kernel Density Estimates (KDE) for the Ks (synonymous substitution rates) distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_ksd(ks_df, bin_width = 0.01, maxK = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_ksd_+3A_ks_df">ks_df</code></td>
<td>
<p>A data frame containing Ks values.</p>
</td></tr>
<tr><td><code id="generate_ksd_+3A_bin_width">bin_width</code></td>
<td>
<p>The width of each bin for KDE calculation.</p>
</td></tr>
<tr><td><code id="generate_ksd_+3A_maxk">maxK</code></td>
<td>
<p>The maximum Ks value for the distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>

<ul>
<li> <p><code>Ks</code>: A numeric vector representing the KDE values.
</p>
</li>
<li> <p><code>bin_width</code>: The width of each bin used for KDE calculation.
</p>
</li>
<li> <p><code>maxK</code>: The maximum Ks value for the distribution.
</p>
</li></ul>


<hr>
<h2 id='generateKsDistribution'>Generate the Ks Distribution</h2><span id='topic+generateKsDistribution'></span>

<h3>Description</h3>

<p>This function generates a Ks (synonymous substitution rates) distribution from raw Ks values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateKsDistribution(ksraw, speciesName = NULL, maxK = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generateKsDistribution_+3A_ksraw">ksraw</code></td>
<td>
<p>A numeric vector containing raw Ks values.</p>
</td></tr>
<tr><td><code id="generateKsDistribution_+3A_speciesname">speciesName</code></td>
<td>
<p>(Optional) A character string specifying the species name associated with the Ks values.</p>
</td></tr>
<tr><td><code id="generateKsDistribution_+3A_maxk">maxK</code></td>
<td>
<p>A numeric value indicating the maximum Ks value to consider in the distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector containing the binned Ks distribution.
</p>

<hr>
<h2 id='get_segments'>Get Segmented Data from Anchorpoints and Ks Values</h2><span id='topic+get_segments'></span>

<h3>Description</h3>

<p>This function extracts segmented data from anchorpoints and Ks (synonymous substitution rate) values,
based on specified criteria, and writes the results to output files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_segments(
  genes_file,
  anchors_ks_file,
  multiplicons_file,
  segmented_file,
  segmented_anchorpoints_file,
  num_anchors = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_segments_+3A_genes_file">genes_file</code></td>
<td>
<p>A character string specifying the file path for genes information created by i-ADHoRe.</p>
</td></tr>
<tr><td><code id="get_segments_+3A_anchors_ks_file">anchors_ks_file</code></td>
<td>
<p>A character string specifying the file path for anchorpoints Ks values data.</p>
</td></tr>
<tr><td><code id="get_segments_+3A_multiplicons_file">multiplicons_file</code></td>
<td>
<p>A character string specifying the file path for multiplicons information created by i-ADHoRe.</p>
</td></tr>
<tr><td><code id="get_segments_+3A_segmented_file">segmented_file</code></td>
<td>
<p>A character string specifying the output file path for segmented data.</p>
</td></tr>
<tr><td><code id="get_segments_+3A_segmented_anchorpoints_file">segmented_anchorpoints_file</code></td>
<td>
<p>A character string specifying the output file path for segmented anchorpoints.</p>
</td></tr>
<tr><td><code id="get_segments_+3A_num_anchors">num_anchors</code></td>
<td>
<p>An integer specifying the minimum number of anchorpoints required.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL (output files are generated with the specified information).
</p>

<hr>
<h2 id='is_fasta_cds'>Check if a file is in FASTA format with cds sequences.</h2><span id='topic+is_fasta_cds'></span>

<h3>Description</h3>

<p>This function checks whether a given file is in FASTA format with cds sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_fasta_cds(file_path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_fasta_cds_+3A_file_path">file_path</code></td>
<td>
<p>The path to the input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the file is in FASTA format with cds sequences, FALSE otherwise.
</p>

<hr>
<h2 id='is.ksv'>Check if an object is of class &quot;ksv&quot;</h2><span id='topic+is.ksv'></span>

<h3>Description</h3>

<p>This function checks if the provided object is of class &quot;ksv.&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.ksv(x)

is.ksv(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.ksv_+3A_x">x</code></td>
<td>
<p>The object to be checked.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns TRUE if the object is of class &quot;ksv&quot;; otherwise, returns FALSE.
</p>

<hr>
<h2 id='is.not.null'>Check if an Object is Not NULL</h2><span id='topic+is.not.null'></span>

<h3>Description</h3>

<p>This function checks if an object is not NULL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.not.null(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.not.null_+3A_x">x</code></td>
<td>
<p>An R object to check.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value indicating whether the object is not NULL.
</p>

<hr>
<h2 id='ks_mclust_v2'>ks_mclust_v2</h2><span id='topic+ks_mclust_v2'></span>

<h3>Description</h3>

<p>A wrapper to run emmix modeling using the mclust package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ks_mclust_v2(input_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ks_mclust_v2_+3A_input_data">input_data</code></td>
<td>
<p>The input data for clustering and modeling.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing clustering and modeling results.
</p>

<hr>
<h2 id='map_informal_name_to_latin_name'>Map Informal Names to Latin Names</h2><span id='topic+map_informal_name_to_latin_name'></span>

<h3>Description</h3>

<p>This function reads information from an Excel file (XLS) containing columns &quot;latin_name,&quot; &quot;informal_name,&quot; and &quot;gff.&quot; It extracts the &quot;latin_name&quot; and &quot;informal_name&quot; columns, performs some data manipulation, and returns a data frame with these two columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map_informal_name_to_latin_name(sp_gff_info_xls)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="map_informal_name_to_latin_name_+3A_sp_gff_info_xls">sp_gff_info_xls</code></td>
<td>
<p>The path to the Excel file containing species information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with &quot;latin_name&quot; and &quot;informal_name&quot; columns.
</p>

<hr>
<h2 id='mix_logNormal_Ks'>Log-Normal mixturing analyses of a Ks distributions for the whole paranome</h2><span id='topic+mix_logNormal_Ks'></span>

<h3>Description</h3>

<p>Log-Normal mixturing analyses of a Ks distributions for the whole paranome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mix_logNormal_Ks(ksv, G = 1:5, k.nstart = 500, maxK = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mix_logNormal_Ks_+3A_ksv">ksv</code></td>
<td>
<p>A <code>ksv</code> object.</p>
</td></tr>
<tr><td><code id="mix_logNormal_Ks_+3A_g">G</code></td>
<td>
<p>An integer vector specifying the range of the mixtured components.
A BIC is calculated for each component. The default is G=1:5.
For a formal analysis, it is recommended to use 1:10.</p>
</td></tr>
<tr><td><code id="mix_logNormal_Ks_+3A_k.nstart">k.nstart</code></td>
<td>
<p>How many random sets should be chosen in the k-means
clustering.
For a formal analysis, it is recommended to use 500.</p>
</td></tr>
<tr><td><code id="mix_logNormal_Ks_+3A_maxk">maxK</code></td>
<td>
<p>Maximum Ks values used in the mixture modeling analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with seven variables.
</p>

<hr>
<h2 id='modeFinder'>modeFinder</h2><span id='topic+modeFinder'></span>

<h3>Description</h3>

<p>Find the mode (peak) of a univariate distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modeFinder(x, bw = 0.1, from = 0, to = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="modeFinder_+3A_x">x</code></td>
<td>
<p>A numeric vector or a kernel density estimate (KDE).</p>
</td></tr>
<tr><td><code id="modeFinder_+3A_bw">bw</code></td>
<td>
<p>Bandwidth for the KDE. Default is 0.1.</p>
</td></tr>
<tr><td><code id="modeFinder_+3A_from">from</code></td>
<td>
<p>Starting point for mode search. Default is 0.</p>
</td></tr>
<tr><td><code id="modeFinder_+3A_to">to</code></td>
<td>
<p>Ending point for mode search. Default is 5.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The mode (peak) of the distribution.
</p>

<hr>
<h2 id='obtain_chromosome_length'>obtain_chromosome_length</h2><span id='topic+obtain_chromosome_length'></span>

<h3>Description</h3>

<p>Process species information file and extract chromosome lengths and mRNA counts from GFF files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obtain_chromosome_length(species_info_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="obtain_chromosome_length_+3A_species_info_file">species_info_file</code></td>
<td>
<p>A character string specifying the path to the species information file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two data frames: len_df for chromosome lengths and num_df for mRNA counts.
</p>

<hr>
<h2 id='obtain_chromosome_length_filter'>obtain_chromosome_length_filter</h2><span id='topic+obtain_chromosome_length_filter'></span>

<h3>Description</h3>

<p>Process a data frame containing species information and extract chromosome lengths and mRNA counts from GFF files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obtain_chromosome_length_filter(species_info_df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="obtain_chromosome_length_filter_+3A_species_info_df">species_info_df</code></td>
<td>
<p>A data frame containing species information with columns &quot;sp,&quot; &quot;cds,&quot; and &quot;gff.&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two data frames: len_df for chromosome lengths and num_df for mRNA counts.
</p>

<hr>
<h2 id='obtain_coordiantes_for_anchorpoints'>Obtain coordinates for anchorpoints from GFF files</h2><span id='topic+obtain_coordiantes_for_anchorpoints'></span>

<h3>Description</h3>

<p>This function takes a file containing anchorpoints, GFF files for two species, and species names,
and retrieves the coordinates of anchorpoints and associated genes from the GFF files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obtain_coordiantes_for_anchorpoints(
  anchorpoints,
  species1,
  gff_file1,
  out_file,
  species2 = NULL,
  gff_file2 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="obtain_coordiantes_for_anchorpoints_+3A_anchorpoints">anchorpoints</code></td>
<td>
<p>A file containing anchorpoints information with columns like gene_x, gene_y, and other relevant data.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_+3A_species1">species1</code></td>
<td>
<p>The name of the first species.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_+3A_gff_file1">gff_file1</code></td>
<td>
<p>The path to the GFF file for the first species.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_+3A_out_file">out_file</code></td>
<td>
<p>The output file where the results will be saved.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_+3A_species2">species2</code></td>
<td>
<p>(Optional) The name of the second species. Specify this parameter and gff_file2 if working with two species.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_+3A_gff_file2">gff_file2</code></td>
<td>
<p>(Optional) The path to the GFF file for the second species.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None. The function saves the results to the specified out_file.
</p>

<hr>
<h2 id='obtain_coordiantes_for_anchorpoints_ks'>Obtain Coordinates and Ks Values for Anchorpoints</h2><span id='topic+obtain_coordiantes_for_anchorpoints_ks'></span>

<h3>Description</h3>

<p>This function extracts coordinates and Ks (synonymous substitution rate) values for anchorpoints
from input data and merges them into a single output file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obtain_coordiantes_for_anchorpoints_ks(
  anchorpoints,
  anchorpoints_ks,
  genes_file,
  out_file,
  out_ks_file,
  species
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="obtain_coordiantes_for_anchorpoints_ks_+3A_anchorpoints">anchorpoints</code></td>
<td>
<p>A character string specifying the file path for anchorpoints data.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_ks_+3A_anchorpoints_ks">anchorpoints_ks</code></td>
<td>
<p>A character string specifying the file path for anchorpoints Ks values data.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_ks_+3A_genes_file">genes_file</code></td>
<td>
<p>A character string specifying the file path for genes information.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_ks_+3A_out_file">out_file</code></td>
<td>
<p>A character string specifying the output file path for coordinates.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_ks_+3A_out_ks_file">out_ks_file</code></td>
<td>
<p>A character string specifying the output file path for Ks values.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_anchorpoints_ks_+3A_species">species</code></td>
<td>
<p>A character string specifying the species name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL (output files are generated with the specified information).
</p>

<hr>
<h2 id='obtain_coordiantes_for_segments'>Obtain coordinates for segments in a comparison</h2><span id='topic+obtain_coordiantes_for_segments'></span>

<h3>Description</h3>

<p>This function retrieves the coordinates for segments in a comparison based on the provided parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obtain_coordiantes_for_segments(
  seg_file,
  sp1,
  gff_file1,
  out_file,
  sp2 = NULL,
  gff_file2 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="obtain_coordiantes_for_segments_+3A_seg_file">seg_file</code></td>
<td>
<p>The file containing segment data.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_segments_+3A_sp1">sp1</code></td>
<td>
<p>The species name for the first genome.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_segments_+3A_gff_file1">gff_file1</code></td>
<td>
<p>The GFF file for the first genome.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_segments_+3A_out_file">out_file</code></td>
<td>
<p>The output file to store the merged position data.</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_segments_+3A_sp2">sp2</code></td>
<td>
<p>The species name for the second genome (optional).</p>
</td></tr>
<tr><td><code id="obtain_coordiantes_for_segments_+3A_gff_file2">gff_file2</code></td>
<td>
<p>The GFF file for the second genome (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL (the results are saved in the output file).
</p>

<hr>
<h2 id='obtain_coordinates_for_segments_multiple'>Obtain Coordinates for Segments in Multiple Synteny Blocks</h2><span id='topic+obtain_coordinates_for_segments_multiple'></span>

<h3>Description</h3>

<p>This function extracts coordinates for segments within multiple synteny blocks based on input dataframes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obtain_coordinates_for_segments_multiple(seg_df, gff_df, input, out_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="obtain_coordinates_for_segments_multiple_+3A_seg_df">seg_df</code></td>
<td>
<p>A dataframe containing information about synteny segments.</p>
</td></tr>
<tr><td><code id="obtain_coordinates_for_segments_multiple_+3A_gff_df">gff_df</code></td>
<td>
<p>A dataframe containing GFF (General Feature Format) information.</p>
</td></tr>
<tr><td><code id="obtain_coordinates_for_segments_multiple_+3A_input">input</code></td>
<td>
<p>A list containing input data, typically multiple synteny query chromosomes.</p>
</td></tr>
<tr><td><code id="obtain_coordinates_for_segments_multiple_+3A_out_file">out_file</code></td>
<td>
<p>A character string specifying the output file path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with coordinates for segments within multiple synteny blocks.
</p>

<hr>
<h2 id='obtain_mean_ks_for_each_multiplicon'>Compute the Mean of Ks values for Each Multiplicon</h2><span id='topic+obtain_mean_ks_for_each_multiplicon'></span>

<h3>Description</h3>

<p>This function takes as input a multiplicon file, an anchorpoint file, Ks values, and other relevant information.
It calculates the mean of Ks values for each multiplicon and associates them with the corresponding data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obtain_mean_ks_for_each_multiplicon(
  multiplicon_file,
  anchorpoint_file,
  species1,
  ks_file,
  outfile,
  anchorpointout_file,
  species2 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="obtain_mean_ks_for_each_multiplicon_+3A_multiplicon_file">multiplicon_file</code></td>
<td>
<p>A file containing multiplicon information.</p>
</td></tr>
<tr><td><code id="obtain_mean_ks_for_each_multiplicon_+3A_anchorpoint_file">anchorpoint_file</code></td>
<td>
<p>A file containing anchorpoints information with columns like geneX, geneY, and other relevant data.</p>
</td></tr>
<tr><td><code id="obtain_mean_ks_for_each_multiplicon_+3A_species1">species1</code></td>
<td>
<p>The name of the first species.</p>
</td></tr>
<tr><td><code id="obtain_mean_ks_for_each_multiplicon_+3A_ks_file">ks_file</code></td>
<td>
<p>A file containing Ks values.</p>
</td></tr>
<tr><td><code id="obtain_mean_ks_for_each_multiplicon_+3A_outfile">outfile</code></td>
<td>
<p>The output file where the results will be saved.</p>
</td></tr>
<tr><td><code id="obtain_mean_ks_for_each_multiplicon_+3A_anchorpointout_file">anchorpointout_file</code></td>
<td>
<p>The output file for anchorpoint data with Ks values.</p>
</td></tr>
<tr><td><code id="obtain_mean_ks_for_each_multiplicon_+3A_species2">species2</code></td>
<td>
<p>(Optional) The name of the second species. Specify this parameter and ks_file if working with two species.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None. The function saves the results to the specified outfile and anchorpointout_file.
</p>

<hr>
<h2 id='parse_EMMIX'>Read the EMMIX output for a range of components</h2><span id='topic+parse_EMMIX'></span>

<h3>Description</h3>

<p>Read the EMMIX output for a range of components
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_EMMIX(emmix.out, G = 1:3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parse_EMMIX_+3A_emmix.out">emmix.out</code></td>
<td>
<p>The output file from EMMIX software.</p>
</td></tr>
<tr><td><code id="parse_EMMIX_+3A_g">G</code></td>
<td>
<p>An integer vector specifying the range of the mixture components.
The default is G=1:3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with seven variables.
</p>

<hr>
<h2 id='parse_one_EMMIX'>Read the EMMIX output for a specify number of components</h2><span id='topic+parse_one_EMMIX'></span>

<h3>Description</h3>

<p>Read the EMMIX output for a specify number of components
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_one_EMMIX(emmix.out, ncomponent = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parse_one_EMMIX_+3A_emmix.out">emmix.out</code></td>
<td>
<p>The output file from EMMIX software.</p>
</td></tr>
<tr><td><code id="parse_one_EMMIX_+3A_ncomponent">ncomponent</code></td>
<td>
<p>Number of components to read from the file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with seven variables.
</p>

<hr>
<h2 id='PeaksInKsDistributionValues'>Find Peaks in the Ks Distribution</h2><span id='topic+PeaksInKsDistributionValues'></span>

<h3>Description</h3>

<p>This function identifies peaks in a distribution of Ks (synonymous substitution rates) values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PeaksInKsDistributionValues(
  ks,
  binWidth = 0.1,
  maxK = 5,
  m = 3,
  peak.maxK = 2,
  spar = 0.25
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PeaksInKsDistributionValues_+3A_ks">ks</code></td>
<td>
<p>A numeric vector containing Ks values for which peaks will be identified.</p>
</td></tr>
<tr><td><code id="PeaksInKsDistributionValues_+3A_binwidth">binWidth</code></td>
<td>
<p>A numeric value specifying the bin width for creating the histogram.</p>
</td></tr>
<tr><td><code id="PeaksInKsDistributionValues_+3A_maxk">maxK</code></td>
<td>
<p>A numeric value indicating the maximum Ks value to consider.</p>
</td></tr>
<tr><td><code id="PeaksInKsDistributionValues_+3A_m">m</code></td>
<td>
<p>An integer indicating the half-width of the neighborhood to consider when identifying peaks. A larger value of <code>m</code> makes peak detection less sensitive.</p>
</td></tr>
<tr><td><code id="PeaksInKsDistributionValues_+3A_peak.maxk">peak.maxK</code></td>
<td>
<p>A numeric value specifying the maximum Ks value to consider when identifying peaks.</p>
</td></tr>
<tr><td><code id="PeaksInKsDistributionValues_+3A_spar">spar</code></td>
<td>
<p>A numeric value controlling the smoothness of the spline fit. Higher values make the fit smoother.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector containing the identified peaks in the Ks distribution.
</p>

<hr>
<h2 id='read_data_file'>Read Data from Uploaded File</h2><span id='topic+read_data_file'></span>

<h3>Description</h3>

<p>This function reads data from an uploaded file in a Shiny application and returns it as a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_data_file(uploadfile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_data_file_+3A_uploadfile">uploadfile</code></td>
<td>
<p>The object representing the uploaded file obtained through the Shiny upload function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the data from the uploaded file.
</p>

<hr>
<h2 id='read.wgd_ksd'>Read the output file of wgd ksd</h2><span id='topic+read.wgd_ksd'></span>

<h3>Description</h3>

<p>Read the output file of wgd ksd
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.wgd_ksd(
  file,
  include_outliers = FALSE,
  min_ks = 0,
  min_aln_len = 0,
  min_idn = 0,
  min_cov = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.wgd_ksd_+3A_file">file</code></td>
<td>
<p>The output file of <code style="white-space: pre;">&#8288;wgd ksd&#8288;</code></p>
</td></tr>
<tr><td><code id="read.wgd_ksd_+3A_include_outliers">include_outliers</code></td>
<td>
<p>Include outliers or not, default FALSE.</p>
</td></tr>
<tr><td><code id="read.wgd_ksd_+3A_min_ks">min_ks</code></td>
<td>
<p>Minimum Ks value, default 0.</p>
</td></tr>
<tr><td><code id="read.wgd_ksd_+3A_min_aln_len">min_aln_len</code></td>
<td>
<p>Minimum alignment length, default 0.</p>
</td></tr>
<tr><td><code id="read.wgd_ksd_+3A_min_idn">min_idn</code></td>
<td>
<p>Minimum alignment identity, default 0.</p>
</td></tr>
<tr><td><code id="read.wgd_ksd_+3A_min_cov">min_cov</code></td>
<td>
<p>Minimum alignment coverage, default 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ksv</code> object, which is a list including:
</p>

<ul>
<li> <p><code>ks_df</code>: the data frame that used for following analysis
</p>
</li>
<li> <p><code>ks_dist</code>: a list including a vector of Ks values in the distribution
</p>
</li>
<li> <p><code>raw_df</code>: raw data
</p>
</li>
<li> <p><code>filters</code>: filters that applied to the raw data
</p>
</li></ul>


<hr>
<h2 id='relativeRate'>relativeRate</h2><span id='topic+relativeRate'></span>

<h3>Description</h3>

<p>Compute relative rates using input data files and statistical computations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relativeRate(
  ksv2out_1_file,
  ksv2out_2_file,
  ksv_between_file,
  KsMax,
  low = 0.025,
  up = 0.975,
  bs = 1000
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="relativeRate_+3A_ksv2out_1_file">ksv2out_1_file</code></td>
<td>
<p>A character string specifying the path to the first input data file.</p>
</td></tr>
<tr><td><code id="relativeRate_+3A_ksv2out_2_file">ksv2out_2_file</code></td>
<td>
<p>A character string specifying the path to the second input data file.</p>
</td></tr>
<tr><td><code id="relativeRate_+3A_ksv_between_file">ksv_between_file</code></td>
<td>
<p>A character string specifying the path to the third input data file.</p>
</td></tr>
<tr><td><code id="relativeRate_+3A_ksmax">KsMax</code></td>
<td>
<p>A numeric value representing a maximum threshold for Ks values.</p>
</td></tr>
<tr><td><code id="relativeRate_+3A_low">low</code></td>
<td>
<p>A numeric value specifying the lower quantile for bootstrapping. Default is 0.025.</p>
</td></tr>
<tr><td><code id="relativeRate_+3A_up">up</code></td>
<td>
<p>A numeric value specifying the upper quantile for bootstrapping. Default is 0.975.</p>
</td></tr>
<tr><td><code id="relativeRate_+3A_bs">bs</code></td>
<td>
<p>An integer specifying the number of bootstrap iterations. Default is 1000.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing computed relative rates and their confidence intervals.
</p>

<hr>
<h2 id='remove_inner_stop_codon_sequence'>Remove Genes Contain Stop Codons within the Sequence</h2><span id='topic+remove_inner_stop_codon_sequence'></span>

<h3>Description</h3>

<p>This function removes the gene contains stop codons (TAA, TAG, TGA, taa, tag, tga)
within its sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_inner_stop_codon_sequence(sequence)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="remove_inner_stop_codon_sequence_+3A_sequence">sequence</code></td>
<td>
<p>A nucleotide sequence as a character string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string or NULL.
</p>

<hr>
<h2 id='remove_old_dirs'>Remove directories older than a specified day</h2><span id='topic+remove_old_dirs'></span>

<h3>Description</h3>

<p>This function removes directories in the specified base directory that are
older than a specified maximum age in days. It logs the removed directories
and any errors encountered during removal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_old_dirs(
  base_dir,
  max_age_in_days = 3,
  log_file = "remove_old_dirs.log",
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="remove_old_dirs_+3A_base_dir">base_dir</code></td>
<td>
<p>The base directory to search for old directories.</p>
</td></tr>
<tr><td><code id="remove_old_dirs_+3A_max_age_in_days">max_age_in_days</code></td>
<td>
<p>The maximum age (in days) for directories to be considered old.</p>
</td></tr>
<tr><td><code id="remove_old_dirs_+3A_log_file">log_file</code></td>
<td>
<p>The name of the log file to store information about removed directories and errors.</p>
</td></tr>
<tr><td><code id="remove_old_dirs_+3A_verbose">verbose</code></td>
<td>
<p>A logical value indicating whether to print messages to the console.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function does not return anything. It logs information about removed directories and errors.
</p>

<hr>
<h2 id='replace_informal_name_to_latin_name'>Replace Informal Names with Latin Names</h2><span id='topic+replace_informal_name_to_latin_name'></span>

<h3>Description</h3>

<p>This function takes a data frame <code>names_df</code> containing &quot;latin_name&quot; and &quot;informal_name&quot; columns and an <code>input</code> string as input. It replaces informal species names in the <code>input</code> string with their corresponding Latin names based on the information in <code>names_df</code>. If the <code>input</code> string contains underscores (&quot;_&quot;), it assumes a comparison between two species and replaces both informal names. Otherwise, it replaces the informal name in the <code>input</code> string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replace_informal_name_to_latin_name(names_df, input)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="replace_informal_name_to_latin_name_+3A_names_df">names_df</code></td>
<td>
<p>A data frame with &quot;latin_name&quot; and &quot;informal_name&quot; columns.</p>
</td></tr>
<tr><td><code id="replace_informal_name_to_latin_name_+3A_input">input</code></td>
<td>
<p>The input string that may contain informal species names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified input string with informal names replaced by Latin names.
</p>

<hr>
<h2 id='resampleKsDistribution'>Resample a Ks Distribution</h2><span id='topic+resampleKsDistribution'></span>

<h3>Description</h3>

<p>This function resamples a given Ks (synonymous substitution rates) distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resampleKsDistribution(ks, maxK = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="resampleKsDistribution_+3A_ks">ks</code></td>
<td>
<p>A numeric vector representing the Ks distribution to be resampled.</p>
</td></tr>
<tr><td><code id="resampleKsDistribution_+3A_maxk">maxK</code></td>
<td>
<p>A numeric value indicating the maximum Ks value to consider in the distribution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector containing a resampled Ks distribution.
</p>

<hr>
<h2 id='run_emmix_kmeas'>A wrapper to run EM analysis of \(ln\) Ks values with k-means</h2><span id='topic+run_emmix_kmeas'></span>

<h3>Description</h3>

<p>A wrapper to run EM analysis of \(ln\) Ks values with k-means
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_emmix_kmeas(v, k.centers = 2, k.nstart = 500)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_emmix_kmeas_+3A_v">v</code></td>
<td>
<p>A list include a vector of Ks values namely <code>ks_value</code>,
and a bolean variable namely <code>log</code></p>
</td></tr>
<tr><td><code id="run_emmix_kmeas_+3A_k.centers">k.centers</code></td>
<td>
<p>Number of k-means centers, default 2.</p>
</td></tr>
<tr><td><code id="run_emmix_kmeas_+3A_k.nstart">k.nstart</code></td>
<td>
<p>Number of random start of k-means clustering, default 10.
For a formal analysis, it is recommended to use 500.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, i.e., the original output of mclust::emV
</p>

<hr>
<h2 id='runshinyWGD'>The main code to run shinyWGD</h2><span id='topic+runshinyWGD'></span>

<h3>Description</h3>

<p>The main function to launch the Shiny application for whole genome duplication analysis.
This function initializes the app and opens a Shiny interface that allows users to interactively analyze whole-genome duplication data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runshinyWGD()
</code></pre>


<h3>Value</h3>

<p>No return value. This function is called for side effects, which include starting the Shiny application.
The function launches a Shiny app in a web browser, where users can interact with the whole genome duplication analysis.
</p>

<hr>
<h2 id='SignifFeatureRegion'>SignifFeatureRegion</h2><span id='topic+SignifFeatureRegion'></span>

<h3>Description</h3>

<p>This function computes the significance of features based on gradient and curvature analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SignifFeatureRegion(
  n,
  d,
  gcounts,
  gridsize,
  dest,
  bandwidth,
  signifLevel,
  range.x,
  grad = TRUE,
  curv = TRUE,
  neg.curv.only = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SignifFeatureRegion_+3A_n">n</code></td>
<td>
<p>The sample size.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_d">d</code></td>
<td>
<p>The dimensionality of the data.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_gcounts">gcounts</code></td>
<td>
<p>A numeric vector representing data counts.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_gridsize">gridsize</code></td>
<td>
<p>A numeric vector specifying the grid size.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_dest">dest</code></td>
<td>
<p>A kernel density estimate.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_bandwidth">bandwidth</code></td>
<td>
<p>The bandwidth parameter.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_signiflevel">signifLevel</code></td>
<td>
<p>The significance level.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_range.x">range.x</code></td>
<td>
<p>The range of x values.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_grad">grad</code></td>
<td>
<p>A logical value indicating whether to compute the gradient significance.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_curv">curv</code></td>
<td>
<p>A logical value indicating whether to compute the curvature significance.</p>
</td></tr>
<tr><td><code id="SignifFeatureRegion_+3A_neg.curv.only">neg.curv.only</code></td>
<td>
<p>A logical value indicating whether to consider negative curvature only.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the significance results for gradient and curvature.
</p>

<hr>
<h2 id='SiZer'>SiZer (Significant Zero Crossings)</h2><span id='topic+SiZer'></span>

<h3>Description</h3>

<p>The SiZer (Significant Zero Crossings) method is a technique used for assessing the statistical significance of zero crossings in data density estimation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SiZer(x, bw, gridsize, signifLevel = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SiZer_+3A_x">x</code></td>
<td>
<p>A numeric vector containing the data for which you want to calculate SiZer.</p>
</td></tr>
<tr><td><code id="SiZer_+3A_bw">bw</code></td>
<td>
<p>Bandwidth parameter for kernel density estimation. If not provided, default values are used.</p>
</td></tr>
<tr><td><code id="SiZer_+3A_gridsize">gridsize</code></td>
<td>
<p>A vector specifying the grid size for SiZer. Default is c(401, 151).</p>
</td></tr>
<tr><td><code id="SiZer_+3A_signiflevel">signifLevel</code></td>
<td>
<p>The significance level for SiZer. Default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing SiZer results, including the SiZer curve, the SiZer map, and the bandwidth.
</p>

<hr>
<h2 id='symconv.ks'>symconv.ks</h2><span id='topic+symconv.ks'></span>

<h3>Description</h3>

<p>Perform symmetric convolution using FFT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symconv.ks(rr, ss, skewflag)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="symconv.ks_+3A_rr">rr</code></td>
<td>
<p>The first input vector.</p>
</td></tr>
<tr><td><code id="symconv.ks_+3A_ss">ss</code></td>
<td>
<p>The second input vector.</p>
</td></tr>
<tr><td><code id="symconv.ks_+3A_skewflag">skewflag</code></td>
<td>
<p>A scalar value to apply skew correction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector representing the result of the symmetric convolution.
</p>

<hr>
<h2 id='symconv2D.ks'>symconv2D.ks</h2><span id='topic+symconv2D.ks'></span>

<h3>Description</h3>

<p>Perform symmetric 2D convolution using FFT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symconv2D.ks(rr, ss, skewflag = rep(1, 2))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="symconv2D.ks_+3A_rr">rr</code></td>
<td>
<p>The first input matrix.</p>
</td></tr>
<tr><td><code id="symconv2D.ks_+3A_ss">ss</code></td>
<td>
<p>The second input matrix.</p>
</td></tr>
<tr><td><code id="symconv2D.ks_+3A_skewflag">skewflag</code></td>
<td>
<p>A vector of two scalar values for skew correction along each dimension.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix representing the result of the symmetric 2D convolution.
</p>

<hr>
<h2 id='symconv3D.ks'>symconv3D.ks</h2><span id='topic+symconv3D.ks'></span>

<h3>Description</h3>

<p>Perform symmetric 3D convolution using FFT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symconv3D.ks(rr, ss, skewflag = rep(1, 3))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="symconv3D.ks_+3A_rr">rr</code></td>
<td>
<p>The first input 3D array.</p>
</td></tr>
<tr><td><code id="symconv3D.ks_+3A_ss">ss</code></td>
<td>
<p>The second input 3D array.</p>
</td></tr>
<tr><td><code id="symconv3D.ks_+3A_skewflag">skewflag</code></td>
<td>
<p>A vector of three scalar values for skew correction along each dimension.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 3D array representing the result of the symmetric 3D convolution.
</p>

<hr>
<h2 id='symconv4D.ks'>symconv4D.ks</h2><span id='topic+symconv4D.ks'></span>

<h3>Description</h3>

<p>Perform symmetric 4D convolution using FFT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symconv4D.ks(rr, ss, skewflag = rep(1, 4), fftflag = rep(TRUE, 2))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="symconv4D.ks_+3A_rr">rr</code></td>
<td>
<p>The first input 4D array.</p>
</td></tr>
<tr><td><code id="symconv4D.ks_+3A_ss">ss</code></td>
<td>
<p>The second input 4D array.</p>
</td></tr>
<tr><td><code id="symconv4D.ks_+3A_skewflag">skewflag</code></td>
<td>
<p>A vector of four scalar values for skew correction along each dimension.</p>
</td></tr>
<tr><td><code id="symconv4D.ks_+3A_fftflag">fftflag</code></td>
<td>
<p>A vector of two Boolean values for FFT flag.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 4D array representing the result of the symmetric 4D convolution.
</p>

<hr>
<h2 id='TimeTreeFecher'>Extracts a timetree from TimeTree.org based on species names.</h2><span id='topic+TimeTreeFecher'></span>

<h3>Description</h3>

<p>This function takes a file with species names as input and a prefix to define the output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TimeTreeFecher(input_file, prefix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TimeTreeFecher_+3A_input_file">input_file</code></td>
<td>
<p>A character string specifying the path to the file containing species names.</p>
</td></tr>
<tr><td><code id="TimeTreeFecher_+3A_prefix">prefix</code></td>
<td>
<p>A character string providing the prefix for the output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A timetree object representing the estimated divergence times between species.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
