<!DOCTYPE html><html lang="en"><head><title>Help for package SIBERG</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SIBERG}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fitGP'><p>Fit Generalized Poisson Mixture Model</p></a></li>
<li><a href='#fitLN'><p>Fit Log Normal Mixture Model</p></a></li>
<li><a href='#fitNB'><p>Fit Negative Binomial Mixture Model</p></a></li>
<li><a href='#fitNL'><p>Fit Negative Binomial Mixture Model</p></a></li>
<li><a href='#SIBER'><p>Fit Mixture Model on The RNAseq Data and Calculates Bimodality Index</p></a></li>
<li><a href='#simDat'><p>Simulated Data From 2-component Mixture Models</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-02</td>
</tr>
<tr>
<td>Title:</td>
<td>Systematic Identification of Bimodally Expressed Genes Using
RNAseq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Pan Tong, Kevin R. Coombes</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin R. Coombes &lt;krc@silicovore.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides models to identify bimodally expressed genes from
 RNAseq data based on the Bimodality Index. SIBERG models the  RNAseq data in
 the finite mixture modeling framework and incorporates mechanisms  for
 dealing with RNAseq normalization. Three types of mixture models are
 implemented,  namely, the mixture of log normal, negative binomial, or 
 generalized Poisson distribution. See Tong et al. (2013)
 &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbts713">doi:10.1093/bioinformatics/bts713</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>mclust</td>
</tr>
<tr>
<td>Suggests:</td>
<td>edgeR, doParallel</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://oompa.r-forge.r-project.org/">http://oompa.r-forge.r-project.org/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-03 01:10:52 UTC; KRC</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-03 06:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='fitGP'>Fit Generalized Poisson Mixture Model</h2><span id='topic+fitGP'></span>

<h3>Description</h3>

<p>The function fits a two-component Generalized Poisson mixture model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitGP(y, d=NULL, inits=NULL, model='V', zeroPercentThr=0.2)
</code></pre>


<h3>Arguments</h3>

  
<table role = "presentation">
<tr><td><code id="fitGP_+3A_y">y</code></td>
<td>
<p>A vector representing the RNAseq raw count.</p>
</td></tr>
<tr><td><code id="fitGP_+3A_d">d</code></td>
<td>
<p>A vector of the same length as y representing the normalization constant to be applied to the data. </p>
</td></tr>
<tr><td><code id="fitGP_+3A_inits">inits</code></td>
<td>
<p>Initial value to fit the mixture model. A vector with elements mu1, mu2, phi1, phi2 and pi1. </p>
</td></tr> 
<tr><td><code id="fitGP_+3A_model">model</code></td>
<td>
<p>Character specifying E or V model. E model fits the mixture model with equal dispersion phi 
while V model doesn't put any constraint. </p>
</td></tr> 
<tr><td><code id="fitGP_+3A_zeropercentthr">zeroPercentThr</code></td>
<td>
<p>A scalar specifying the minimum percent of zero counts needed when fitting a zero-inflated 
Generalized Poisson model. This parameter is used to deal with zero-inflation in RNAseq count data. When the percent of zero exceeds this threshold, 
rather than fitting a 2-component Generalized Poisson mixture, a mixture of point mass at 0 
and Generalized Poisson is fitted. </p>
</td></tr> 
</table>


<h3>Details</h3>

<p>This function directly maximize the log likelihood function through optimization. 
With this function, three models can be fitted: (1) Generalized Poisson mixture with equal dispersion (E model); 
(2) Generalized Poisson mixture with unequal dispersion (V model); (3) 0-inflated Generalized Poisson model.
The 0-inflated Generalized Poisson has the following density function:
</p>
<p><code class="reqn">P(Y=y)=\pi D(y) + (1-\pi)GP(\mu, \phi)</code> where D is the point mass at 0 while <code class="reqn">GP(\mu, \phi)</code> is the density
of Generalized Poisson distribution with mean <code class="reqn">\mu</code> and dispersion <code class="reqn">\phi</code>. The variance is <code class="reqn">\phi \mu</code>.
</p>
<p>The rule to fit 0-inflated model is that the observed percentage of count exceeds the user specified threshold. This
rule overrides the model argument when observed percentae of zero count exceeds the threshold.
</p>


<h3>Value</h3>

<p>A vector consisting parameter estimates of mu1, mu2, phi1, phi2, pi1, logLik and BIC. 
For 0-inflated model, mu1=phi1=0.
</p>


<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>References</h3>

<p>Tong, P., Chen, Y., Su, X. and Coombes, K. R. (2012). Systematic Identification of Bimodally Expressed Genes
Using RNAseq Data. Bioinformatics, 2013 Mar 1;29(5):605-13. 
</p>


<h3>See Also</h3>

<p><a href="#topic+SIBER">SIBER</a>
<a href="#topic+fitLN">fitLN</a>
<a href="#topic+fitNB">fitNB</a>
<a href="#topic+fitNL">fitNL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># artificial RNAseq data from negative binomial distribution
set.seed(1000)
dat &lt;- rnbinom(100, mu=1000, size=1/0.2)
fitGP(y=dat)
</code></pre>

<hr>
<h2 id='fitLN'>Fit Log Normal Mixture Model</h2><span id='topic+fitLN'></span>

<h3>Description</h3>

<p>The function fits a two-component log normal mixture model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitLN(y, base=10, eps=10, d=NULL, model='E', zeroPercentThr=0.2, logLikToLN=TRUE)
</code></pre>


<h3>Arguments</h3>

  
<table role = "presentation">
<tr><td><code id="fitLN_+3A_y">y</code></td>
<td>
<p>A vector representing the RNAseq raw count. </p>
</td></tr>
<tr><td><code id="fitLN_+3A_base">base</code></td>
<td>
<p>The logarithm base defining the parameter estimates in the logarithm scale. This is also the base of log transformation applied to the data. </p>
</td></tr> 
<tr><td><code id="fitLN_+3A_eps">eps</code></td>
<td>
<p>A scalar to be added to the count data to avoid taking logarithm of zero. </p>
</td></tr> 
<tr><td><code id="fitLN_+3A_d">d</code></td>
<td>
<p>A vector of the same length as y representing the normalization constant to be applied to the data. For the LN model, the original data would be devided by this vector. </p>
</td></tr>
<tr><td><code id="fitLN_+3A_model">model</code></td>
<td>
<p>Character specifying E or V model. E model fits the mixture model with equal variance while V model doesn't put any constraint. </p>
</td></tr> 
<tr><td><code id="fitLN_+3A_zeropercentthr">zeroPercentThr</code></td>
<td>
<p>A scalar specifying the minimum percent of zero counts needed when fitting a zero-inflated 
log normal model. This
parameter is used to deal with zero-inflation in RNAseq count data. When the percent of zero exceeds this threshold, 
1-comp mixture LN model is used to estimate mu and sigma from nonzero count.</p>
</td></tr> 
<tr><td><code id="fitLN_+3A_logliktoln">logLikToLN</code></td>
<td>
<p>logical indicating if the log likelihod is defined on the transformed value or the orginal value from log noral distribution. </p>
</td></tr> 
</table>


<h3>Details</h3>

<p>The parameter estimates from log normal mixture is obtained by taking logarithm and fit normal mixture. We use
mclust package to obtain parameter estimates of normal mixture model. In particular, <code class="reqn">log_{base}(\frac{y+eps}{d})</code> is used to
fit to normal mixture model. 
</p>
<p>With this function, three models can be fitted: (1) log normal mixture with equal variance (E model); 
(2) Generalized Poisson mixture with unequal variance (V model); (3) 0-inflated log normal model.
The 0-inflated log normal has the following density function:
</p>
<p><code class="reqn">P(Y=y)=\pi D(y) + (1-\pi)LN(\mu, \sigma)</code> where D is the point mass at 0 while <code class="reqn">LN(\mu, \sigma)</code> is the density
of log normal distribution with mean <code class="reqn">\mu</code> and variance <code class="reqn">\sigma^2</code>. 
</p>
<p>The rule to fit 0-inflated model is that the observed percentage of count exceeds the user specified threshold. This
rule overrides the model argument (E or V) when observed percentae of zero count exceeds the threshold.
</p>


<h3>Value</h3>

<p>A vector consisting parameter estimates of mu1, mu2, sigma1, sigma2, pi1, logLik and BIC. 
For 0-inflated model, mu1=sigma1=0.
</p>


<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>References</h3>

<p>Wang, J.,Wen, S., Symmans,W., Pusztai, L., and Coombes, K. (2009). The bimodality
index: a criterion for discovering and ranking bimodal signatures from cancer gene
expression profiling data. Cancer informatics, 7, 199.
</p>
<p>Tong, P., Chen, Y., Su, X. and Coombes, K. R. (2012). Systematic
Identification of Bimodally Expressed Genes Using RNAseq
Data. Bioinformatics, 2013 Mar 1;29(5):605-13. 
</p>


<h3>See Also</h3>

<p><a href="#topic+SIBER">SIBER</a>
<a href="#topic+fitNB">fitNB</a>
<a href="#topic+fitGP">fitGP</a>
<a href="#topic+fitNL">fitNL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># artificial RNAseq data from negative binomial distribution
set.seed(1000)
dat &lt;- rnbinom(100, mu=1000, size=1/0.2)
fitLN(y=dat)
</code></pre>

<hr>
<h2 id='fitNB'>Fit Negative Binomial Mixture Model</h2><span id='topic+fitNB'></span>

<h3>Description</h3>

<p>The function fits a two-component Negative Binomial mixture model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitNB(y, d=NULL, inits=NULL, model='V', zeroPercentThr=0.2)
</code></pre>


<h3>Arguments</h3>

  
<table role = "presentation">
<tr><td><code id="fitNB_+3A_y">y</code></td>
<td>
<p>A vector representing the RNAseq raw count.</p>
</td></tr>
<tr><td><code id="fitNB_+3A_d">d</code></td>
<td>
<p>A vector of the same length as y representing the normalization constant to be applied to the data. </p>
</td></tr>
<tr><td><code id="fitNB_+3A_inits">inits</code></td>
<td>
<p>Initial value to fit the mixture model. A vector with elements mu1, mu2, phi1, phi2 and pi1. For 0-inflated model,
only mu2, phi2, pi1 are used while the other elements can be arbitrary. </p>
</td></tr> 
<tr><td><code id="fitNB_+3A_model">model</code></td>
<td>
<p>Character specifying E or V model. E model fits the mixture model with equal dispersion phi while V model doesn't put any constraint. </p>
</td></tr> 
<tr><td><code id="fitNB_+3A_zeropercentthr">zeroPercentThr</code></td>
<td>
<p>A scalar specifying the minimum percent of zero counts needed when fitting a zero-inflated Negative Binomial model. This
parameter is used to deal with zero-inflation in RNAseq count data. When the percent of zero exceeds this threshold, 
rather than fitting a 2-component negative binomial mixture, a mixture of point mass at 0 and negative binomial is fitted. </p>
</td></tr> 
</table>


<h3>Details</h3>

<p>This function directly maximize the log likelihood function through optimization. 
With this function, three models can be fitted: (1) negative binomial mixture with equal dispersion (E model); 
(2) negative binomial mixture with unequal dispersion (V model); (3) 0-inflated negative binomial model.
The 0-inflated negative binomial has the following density function:
</p>
<p><code class="reqn">P(Y=y)=\pi D(y) + (1-\pi)NB(\mu, \phi)</code> where D is the point mass at 0 while <code class="reqn">NB(\mu, \phi)</code> is the density
of negative binomial distribution with mean <code class="reqn">\mu</code> and dispersion <code class="reqn">\phi</code>. The variance is <code class="reqn">\mu+\phi \mu^2</code>.
</p>
<p>The rule to fit 0-inflated model is that the observed percentage of count exceeds the user specified threshold. This
rule overrides the model argument when observed percentae of zero count exceeds the threshold.
</p>


<h3>Value</h3>

<p>A vector consisting parameter estimates of mu1, mu2, phi1, phi2, pi1, logLik and BIC. 
For 0-inflated model, mu1=phi1=0.
</p>


<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>References</h3>

<p>Tong, P., Chen, Y., Su, X. and Coombes, K. R. (2012). Systematic
Identification of Bimodally Expressed Genes Using RNAseq
Data. Bioinformatics, 2013 Mar 1;29(5):605-13. 
</p>


<h3>See Also</h3>

<p><a href="#topic+SIBER">SIBER</a>
<a href="#topic+fitLN">fitLN</a>
<a href="#topic+fitGP">fitGP</a>
<a href="#topic+fitNL">fitNL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># artificial RNAseq data from negative binomial distribution
set.seed(1000)
dat &lt;- rnbinom(100, mu=1000, size=1/0.2)
fitNB(y=dat)
</code></pre>

<hr>
<h2 id='fitNL'>Fit Negative Binomial Mixture Model</h2><span id='topic+fitNL'></span>

<h3>Description</h3>

<p>The function fits a two-component Negative Binomial mixture model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitNL(y, d=NULL, model='E')
</code></pre>


<h3>Arguments</h3>

  
<table role = "presentation">
<tr><td><code id="fitNL_+3A_y">y</code></td>
<td>
<p>A vector representing the transformed data that follows the normal mixture distribution.</p>
</td></tr>
<tr><td><code id="fitNL_+3A_d">d</code></td>
<td>
<p>A vector of the same length as y representing the normalization constant to be applied to the data. </p>
</td></tr>
<tr><td><code id="fitNL_+3A_model">model</code></td>
<td>
<p>Character specifying E or V model. E model fits the mixture model with equal variance while V model doesn't put any constraint. </p>
</td></tr> 
</table>


<h3>Details</h3>

<p>This function calls the mclust package to fit the 2-component normal mixture. 
</p>


<h3>Value</h3>

<p>A vector consisting parameter estimates of mu1, mu2, phi1, phi2, pi1, logLik and BIC. 
</p>


<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>References</h3>

<p>Wang, J.,Wen, S., Symmans,W., Pusztai, L., and Coombes, K. (2009). The bimodality
index: a criterion for discovering and ranking bimodal signatures from cancer gene
expression profiling data. Cancer informatics, 7, 199.
</p>
<p>Fraley, C. and Raftery, A. (2002). Model-based clustering, discriminant analysis, and
density estimation. Journal of the american statistical association,
97(458), 611:631.
</p>
<p>Tong, P., Chen, Y., Su, X. and Coombes, K. R. (2012). Systematic
Identification of Bimodally Expressed Genes Using RNAseq
Data. Bioinformatics, 2013 Mar 1;29(5):605-13. 
</p>


<h3>See Also</h3>

<p><a href="#topic+SIBER">SIBER</a>
<a href="#topic+fitLN">fitLN</a>
<a href="#topic+fitNB">fitNB</a>
<a href="#topic+fitGP">fitGP</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># artificial microarray data from normal distribution
set.seed(1000)
dat &lt;- rnorm(100, mean=5, sd=1)
fitNL(y=dat)
</code></pre>

<hr>
<h2 id='SIBER'>Fit Mixture Model on The RNAseq Data and Calculates Bimodality Index</h2><span id='topic+SIBER'></span>

<h3>Description</h3>

<p>The function fits a two-component mixture model and calculate BI from the parameter estimates. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SIBER(y, d=NULL, model=c('LN', 'NB', 'GP', 'NL'), zeroPercentThr=0.2, base=exp(1), eps=10)
</code></pre>


<h3>Arguments</h3>

  
<table role = "presentation">
<tr><td><code id="SIBER_+3A_y">y</code></td>
<td>
<p>A vector representing the RNAseq raw count or the transformed values if model=NL. </p>
</td></tr>
<tr><td><code id="SIBER_+3A_d">d</code></td>
<td>
<p>A vector of the same length as y representing the normalization constant to be applied to the data.</p>
</td></tr>
<tr><td><code id="SIBER_+3A_model">model</code></td>
<td>
<p>Character string specifying the mixture model type. It can be any of LN, NB, GP and NL.</p>
</td></tr> 
<tr><td><code id="SIBER_+3A_zeropercentthr">zeroPercentThr</code></td>
<td>
<p>A scalar specifying the minimum percent of zero to detect using log normal mixture. This
parameter is used to deal with zero-inflation in RNAseq count data. When the percent of zero exceeds this threshold, 
1-comp mixture LN model is used to estimate mu and sigma from nonzero count. This parameter is relevant only if model='LN'.
</p>
</td></tr> 
<tr><td><code id="SIBER_+3A_base">base</code></td>
<td>
<p>The logarithm base defining the parameter estimates in the logarithm scale from LN model . It is relevant only if model='LN'.</p>
</td></tr> 
<tr><td><code id="SIBER_+3A_eps">eps</code></td>
<td>
<p>A scalar to be added to the count data when model='LN'. This parameter is relevant only when model='LN'. </p>
</td></tr> 
</table>


<h3>Details</h3>

<p>SIBER proceeds in two steps. The first step fits a two-component mixture model. 
The second step calculates the Bimodality Index corresponding to the assumed mixture distribution.
Four types of mixture models are implemented: log normal (LN), Negative Binomial (NB),  Generalized Poisson (GP) and normal mixture (NL). The normal mixture model was developed to identify bimodal genes from microarray data in Wang et al. It is incorporated here
in case the user has already transformed the RNAseq data.
</p>
<p>Behind the scene, SIBER calls the fitNB, fitGP, fitLN and fitNL function with model=E depending on which
distribution model is specified. When the observed percentage of count exceeds the user specified threshold
zeroPercentThr, the 0-inflated model overrides the E model and will be fitted. 
</p>
<p>Type vignette('SIBER') in the R console to pull out the user manual in pdf format. 
</p>


<h3>Value</h3>

<p>A vector consisting estimates of mu1, mu2, sigma1, sigma2, p1, delta and BI. 
</p>


<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>References</h3>

<p>Wang J, Wen S, Symmans WF, Pusztai L, Coombes KR.<br />
<em>The bimodality index: a criterion for discovering and ranking
bimodal signatures from cancer gene expression profiling data.</em><br />
Cancer Inform. 2009 Aug 5;7:199-216.
</p>
<p>Tong P, Chen Y, Su X, Coombes KR.<br />
<em>SIBER: systematic identification of bimodally expressed genes
using RNAseq data.</em><br />
Bioinformatics. 2013 Mar 1;29(5):605-13.
</p>


<h3>See Also</h3>

<p><a href="#topic+fitLN">fitLN</a>
<a href="#topic+fitNB">fitNB</a>
<a href="#topic+fitGP">fitGP</a>
<a href="#topic+fitNL">fitNL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># artificial RNAseq data from negative binomial distribution
set.seed(1000)
dat &lt;- rnbinom(100, mu=1000, size=1/0.2)
# fit SIBER with the 4 mixture models
SIBER(y=dat, model='LN')
SIBER(y=dat, model='NB')
SIBER(y=dat, model='GP')
SIBER(y=log(dat+1), model='NL')

</code></pre>

<hr>
<h2 id='simDat'>Simulated Data From 2-component Mixture Models</h2><span id='topic+simDat'></span><span id='topic+parList'></span><span id='topic+dataList'></span>

<h3>Description</h3>

<p>Data from 2-component mixture models (NB, GP and LN) is simulated
with the true parameters given for testing and illustration purpose.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(simDat)
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following data objects:
</p>

<dl>
<dt>parList</dt><dd><p>A list of true parameters. There are three named elements 
(NB, GP and LN) corresponding to
the parameters used to simulate gene expression data from NB, GP and LN mixture models. Each
element is a 6 by 5 matrix giving the true parameters generating the simulated data. </p>
</dd>
<dt>dataList</dt><dd><p>A list of matrices for simulated gene expression data. 
There are three named elements (NB, GP and LN) corresponding to
the simulate gene expression data from NB, GP and LN mixture models. Each
element is a 6 by 200 matrix. That is, 6 genes (rows) are simulated with 200 samples (columns). 
The first 3 genes in each matrix are from 2-component mixture model while the last 3 genes
are from 0-inflated models. </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>See Also</h3>

<p><a href="#topic+SIBER">SIBER</a>
<a href="#topic+fitNB">fitNB</a>
<a href="#topic+fitGP">fitGP</a>
<a href="#topic+fitLN">fitLN</a>
<a href="#topic+fitNL">fitNL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SIBERG)
data(simDat)
sapply(parList, dim)
sapply(dataList, dim)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
