<!DOCTYPE html><html lang="en"><head><title>Help for package signalHsmm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {signalHsmm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#signalHsmm'><p>signalHsmm - prediction of signal peptides</p></a></li>
<li><a href='#aaaggregation'><p>Reduced amino acid alphabet</p></a></li>
<li><a href='#add_k_mer_state'><p>Adds k-mer hidden state to signalHsmm model</p></a></li>
<li><a href='#benchmark_dat'><p>Benchmark data set for signalHsmm</p></a></li>
<li><a href='#duration_viterbi'><p>Compute most probable path with extended Viterbi algorithm.</p></a></li>
<li><a href='#find_nhc'><p>Localize n-, h- and c-region in signal peptide</p></a></li>
<li><a href='#gui_signalHsmm'><p>GUI for signalHsmm</p></a></li>
<li><a href='#hsmm_pred'><p>hsmm_pred class</p></a></li>
<li><a href='#hsmm_pred_list'><p>hsmm_pred_list class</p></a></li>
<li><a href='#is_protein'><p>Protein test</p></a></li>
<li><a href='#plot.hsmm_pred'><p>Plot single signalHsmm prediction</p></a></li>
<li><a href='#pred2df'><p>Convert list of signalHsmm predictions</p></a></li>
<li><a href='#predict.sighsmm_model'><p>Predict sighsmm_model object</p></a></li>
<li><a href='#read_txt'><p>Read sequences from .txt file</p></a></li>
<li><a href='#read_uniprot'><p>Read data from UniProt database</p></a></li>
<li><a href='#run_signalHsmm'><p>Predict presence of signal peptide in protein</p></a></li>
<li><a href='#summary.hsmm_pred'><p>Summarize single signalHsmm prediction</p></a></li>
<li><a href='#summary.hsmm_pred_list'><p>Summarize list of signalHsmm predictions</p></a></li>
<li><a href='#train_hsmm'><p>Train sighsmm_model object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Predict Presence of Signal Peptides</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-11-11</td>
</tr>
<tr>
<td>Description:</td>
<td>Predicts the presence of signal peptides in eukaryotic protein
    using hidden semi-Markov models. The implemented algorithm can be accessed from
    both the command line and GUI.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/michbur/signalhsmm">https://github.com/michbur/signalhsmm</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, seqinr, shiny, stats, utils</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DT, rmarkdown, shinythemes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-11-15 08:53:01 UTC; michal</td>
</tr>
<tr>
<td>Author:</td>
<td>Michal Burdukiewicz
    <a href="https://orcid.org/0000-0001-8926-582X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre, aut],
  Piotr Sobczyk [aut],
  Jaroslaw Chilimoniuk
    <a href="https://orcid.org/0000-0001-5467-018X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michal Burdukiewicz &lt;michalburdukiewicz@gmail.com&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-11-15 09:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='signalHsmm'>signalHsmm - prediction of signal peptides</h2><span id='topic+signalHsmm'></span><span id='topic+signalHsmm-package'></span>

<h3>Description</h3>

<p>Using hidden semi-Markov models as a probabilistic framework, signalHsmm 
is new, highly accurate signal peptide predictor for eukaryotic proteins.
</p>


<h3>Details</h3>

<p>Secretory signal peptides are short (20-30 residues) N-terminal amino acid sequences 
tagging among others tag among others hormons, immune system proteins, structural 
proteins, and metabolic enzymes. They direct a protein to the endomembrane system and 
next to the extracellular localization. All signal peptides possess three distinct 
domains with variable length 
and characteristic amino acid composition. Despite their variability, signal peptides 
are universal enough to direct properly proteins in different secretory systems. 
For example, artifically introduced bacterial signal peptides can guide proteins in 
mammals and plants. 
</p>
<p>The development of signalHsmm was funded by National Science Center 
(2015/17/N/NZ2/01845).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>few_predictions &lt;- run_signalHsmm(benchmark_dat[1:3])
#see all predictions
pred2df(few_predictions)
#summary one prediction
summary(few_predictions[[1]])
#plot one prediction
plot(few_predictions[[1]])

#have fun with GUI
## Not run: 
gui_signalHsmm()

## End(Not run)
</code></pre>

<hr>
<h2 id='aaaggregation'>Reduced amino acid alphabet</h2><span id='topic+aaaggregation'></span>

<h3>Description</h3>

<p>Amino acids are grouped together in larger sets based on their 
physicochemical properties important in  the recognition of signal peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aaaggregation
</code></pre>


<h3>Format</h3>

<p>a list of length four. Each element contains a <code>character</code> vector 
of amino acid names (one-letter abbreviations).</p>

<hr>
<h2 id='add_k_mer_state'>Adds k-mer hidden state to signalHsmm model</h2><span id='topic+add_k_mer_state'></span>

<h3>Description</h3>

<p>Changes parameters for Hidden Semi-Markov Model to
add k-mer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_k_mer_state(kMer, pipar, tpmpar, od, params, pState, nState, pTrans, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_k_mer_state_+3A_kmer">kMer</code></td>
<td>
<p><code>character</code> vector representing k-mer aminoacid sequence.</p>
</td></tr>
<tr><td><code id="add_k_mer_state_+3A_pipar">pipar</code></td>
<td>
<p>Probabilities of initial state in Markov Model.</p>
</td></tr>
<tr><td><code id="add_k_mer_state_+3A_tpmpar">tpmpar</code></td>
<td>
<p>Matrix with transition probabilities between states.</p>
</td></tr>
<tr><td><code id="add_k_mer_state_+3A_od">od</code></td>
<td>
<p>Matrix of response probabilities. Eg. od[1,2] is a 
probability of signal 2 in state 1.</p>
</td></tr>
<tr><td><code id="add_k_mer_state_+3A_params">params</code></td>
<td>
<p>Matrix of probability distribution for duration.
Eg. params[10,2] is probability of duration of time 10 in state 2.</p>
</td></tr>
<tr><td><code id="add_k_mer_state_+3A_pstate">pState</code></td>
<td>
<p>number denoting hidden state right before k-mer.</p>
</td></tr>
<tr><td><code id="add_k_mer_state_+3A_nstate">nState</code></td>
<td>
<p>number denoting hidden state right after k-mer.</p>
</td></tr>
<tr><td><code id="add_k_mer_state_+3A_ptrans">pTrans</code></td>
<td>
<p>Probability of change from pState to k-mer hidden state.</p>
</td></tr>
<tr><td><code id="add_k_mer_state_+3A_d">d</code></td>
<td>
<p>Duration of the state.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length four:
</p>

<ul>
<li><p>pipar a vector of new probabilities of initial state in Markov Model,
</p>
</li>
<li><p>tpmpar a matrix with new transition probabilities between states,
</p>
</li>
<li><p>od matrix of new response probabilities,
</p>
</li>
<li><p>params matrix of new probability distributions for duration.
</p>
</li></ul>



<h3>Note</h3>

<p>Currently add only k-mers without distance.
</p>

<hr>
<h2 id='benchmark_dat'>Benchmark data set for signalHsmm</h2><span id='topic+benchmark_dat'></span>

<h3>Description</h3>

<p>Lists eukaryotic proteins added to UniProt database release 
2015_06 between 1.01.2010 and 1.06.2015 (140 proteins with signal peptide and 280 randomly 
sampled proteins without signal peptide).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>benchmark_dat
</code></pre>


<h3>Format</h3>

<p>a list of <code><a href="seqinr.html#topic+SeqFastaAA">SeqFastaAA</a></code> objects. 
Slot <code>sig</code> contains the range of signal peptide (if any).</p>


<h3>Source</h3>

<p><a href="http://www.uniprot.org/">UniProt</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>summary(benchmark_dat)
</code></pre>

<hr>
<h2 id='duration_viterbi'>Compute most probable path with extended Viterbi algorithm.</h2><span id='topic+duration_viterbi'></span>

<h3>Description</h3>

<p>Viterbi algorithm for Hidden Markov Model with duration
</p>


<h3>Usage</h3>

<pre><code class='language-R'>duration_viterbi(aa_sample, pipar, tpmpar, od, params)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="duration_viterbi_+3A_aa_sample">aa_sample</code></td>
<td>
<p><code>character</code> vector representing single aminoacid sequence.</p>
</td></tr>
<tr><td><code id="duration_viterbi_+3A_pipar">pipar</code></td>
<td>
<p>probabilities of initial state in Markov Model.</p>
</td></tr>
<tr><td><code id="duration_viterbi_+3A_tpmpar">tpmpar</code></td>
<td>
<p>matrix of transition probabilities between states.</p>
</td></tr>
<tr><td><code id="duration_viterbi_+3A_od">od</code></td>
<td>
<p>matrix of response probabilities. Eg. od[1,2] is a probability of signal 2 in state 1.</p>
</td></tr>
<tr><td><code id="duration_viterbi_+3A_params">params</code></td>
<td>
<p>matrix of probability distribution for duration. Eg. params[10,2] is probability of duration of time 10 in state 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length four:
</p>

<ul>
<li><p>path a vector of most probable path
</p>
</li>
<li><p>viterbi values of probability in all intermediate points,
</p>
</li>
<li><p>psi matrix that gives for every signal and state the previous state in viterbi path,
</p>
</li>
<li><p>duration matrix that gives for every signal and state gives the duration in that state on viterbi path.
</p>
</li></ul>



<h3>Note</h3>

<p>All computations are on logarithms of probabilities.
</p>

<hr>
<h2 id='find_nhc'>Localize n-, h- and c-region in signal peptide</h2><span id='topic+find_nhc'></span>

<h3>Description</h3>

<p>Finds borders between distinct regions constituting signal peptides using a heuristic
algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_nhc(protein, signal = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_nhc_+3A_protein">protein</code></td>
<td>
<p>a vector of amino acids or object of class 
<code><a href="seqinr.html#topic+SeqFastaAA">SeqFastaAA</a></code>.</p>
</td></tr>
<tr><td><code id="find_nhc_+3A_signal">signal</code></td>
<td>
<p>range of signal peptide. If <code>NULL</code>, the attribute <code>sig</code>
of <code>protein</code> will be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of length 4 containing positions of:
</p>

<ol>
<li><p> start of n-region,
</p>
</li>
<li><p> start of h-region,
</p>
</li>
<li><p> start of c-region,
</p>
</li>
<li><p> cleavage site.
</p>
</li></ol>



<h3>References</h3>

<p>Henrik Nielsen, Anders Krogh (1998). Prediction of signal peptides
and signal anchors by a hidden Markov model. <em>Proc. Sixth Int. Conf. on 
Intelligent Systems for Molecular Biology</em>.
</p>

<hr>
<h2 id='gui_signalHsmm'>GUI for signalHsmm</h2><span id='topic+gui_signalHsmm'></span>

<h3>Description</h3>

<p>A graphical user interface for predicting presence of signal peptides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gui_signalHsmm()
</code></pre>


<h3>Value</h3>

<p>null.
</p>


<h3>Note</h3>

<p>Any ad-blocking software may be cause of malfunctions.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_signalHsmm">run_signalHsmm</a></code>
</p>

<hr>
<h2 id='hsmm_pred'>hsmm_pred class</h2><span id='topic+hsmm_pred'></span>

<h3>Description</h3>

<p>A single prediction of <code>signalHsmm</code>.
</p>
<p>A stochastic model of signal peptide produced by <code>signalHsmm</code>.
</p>


<h3>Details</h3>

<p>Always a named list of five elements
</p>

<ol>
<li> <p><code>sp_probability</code> is a probability of signal peptide presence.
</p>
</li>
<li> <p><code>sp_start</code> is a start of potential signal peptide (naively 1 aminoacid).
</p>
</li>
<li> <p><code>sp_end</code> is a position of last amino acid of signal peptide.
</p>
</li>
<li> <p><code>struc</code> is numeric vector representing predicted structure of input 
protein.
</p>
</li>
<li> <p><code>prot</code> is character vector containing input sequence of amino acids.
</p>
</li>
<li> <p><code>str_approx</code> has value bigger than 0 if the predicted signal peptide 
structure was approximated (usually in case of sequences that have no signal peptides).
</p>
</li></ol>

<p>Always a named list of five elements
</p>

<ol>
<li> <p><code>aa_group</code> encoding of amino acids. See <code><a href="#topic+aaaggregation">aaaggregation</a></code> 
for an example.
</p>
</li>
<li> <p><code>pipar</code> probabilities of initial state in Markov Model.
</p>
</li>
<li> <p><code>tpmpar</code> matrix of transition probabilities between states.
</p>
</li>
<li> <p><code>od</code> matrix of response probabilities. Eg. od[1,2] is a probability 
of signal 2 in state 1.
</p>
</li>
<li> <p><code>overall_probs_log</code> probabilities of amino acids in mature protein.
</p>
</li>
<li> <p><code>params</code> matrix of probability distribution for duration. Eg. params[10,2] 
is probability of duration of time 10 in state 2.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+summary.hsmm_pred">summary.hsmm_pred</a></code> <code><a href="#topic+plot.hsmm_pred">plot.hsmm_pred</a></code>
</p>
<p><code><a href="#topic+train_hsmm">train_hsmm</a></code> <code><a href="#topic+predict.sighsmm_model">predict.sighsmm_model</a></code>
</p>

<hr>
<h2 id='hsmm_pred_list'>hsmm_pred_list class</h2><span id='topic+hsmm_pred_list'></span>

<h3>Description</h3>

<p>A list of prediction(s) generated by <code><a href="#topic+run_signalHsmm">run_signalHsmm</a></code> function.
</p>


<h3>Details</h3>

<p>A named list. Each element belongs to the <code><a href="#topic+hsmm_pred">hsmm_pred</a></code> class.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.hsmm_pred_list">summary.hsmm_pred_list</a></code>, <code><a href="#topic+pred2df">pred2df</a></code>
</p>

<hr>
<h2 id='is_protein'>Protein test</h2><span id='topic+is_protein'></span>

<h3>Description</h3>

<p>Checks if an object is a protein (contains letters from one-letter amino acid code).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_protein(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_protein_+3A_object">object</code></td>
<td>
<p><code>character</code> vector where each elemenents represent one amino acid.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>TRUE</code> or <code>FALSE</code>.
</p>

<hr>
<h2 id='plot.hsmm_pred'>Plot single signalHsmm prediction</h2><span id='topic+plot.hsmm_pred'></span>

<h3>Description</h3>

<p>Plots objects of class <code><a href="#topic+hsmm_pred">hsmm_pred</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hsmm_pred'
plot(x, add_legend = TRUE, only_sure = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.hsmm_pred_+3A_x">x</code></td>
<td>
<p>object of class <code><a href="#topic+hsmm_pred">hsmm_pred</a></code>.</p>
</td></tr>
<tr><td><code id="plot.hsmm_pred_+3A_add_legend">add_legend</code></td>
<td>
<p><code>logical</code>, if <code>TRUE</code>, legend is added to the plot.</p>
</td></tr>
<tr><td><code id="plot.hsmm_pred_+3A_only_sure">only_sure</code></td>
<td>
<p><code>logical</code>, if <code>FALSE</code> does not draw signal peptide structure
when probability is smaller than 0.5.</p>
</td></tr>
<tr><td><code id="plot.hsmm_pred_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>

<hr>
<h2 id='pred2df'>Convert list of signalHsmm predictions</h2><span id='topic+pred2df'></span>

<h3>Description</h3>

<p>Converts objects of class <code><a href="#topic+hsmm_pred_list">hsmm_pred_list</a></code> to data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred2df(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred2df_+3A_object">object</code></td>
<td>
<p>of class <code><a href="#topic+hsmm_pred_list">hsmm_pred_list</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame which columns contain respectively the probability of signal
peptide presence as well as the start and the end of predicted signal peptide.
</p>

<hr>
<h2 id='predict.sighsmm_model'>Predict sighsmm_model object</h2><span id='topic+predict.sighsmm_model'></span>

<h3>Description</h3>

<p>Predicts the presence of signal peptides using signalHsmm models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sighsmm_model'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.sighsmm_model_+3A_object">object</code></td>
<td>
<p><code>sighsmm_model</code> object.</p>
</td></tr>
<tr><td><code id="predict.sighsmm_model_+3A_newdata">newdata</code></td>
<td>
<p>unknown sequence of class <code>character</code> or <code>character</code>.
Alternatively, a <code>list</code> of sequences in mentioned formats.</p>
</td></tr>
<tr><td><code id="predict.sighsmm_model_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#remember to remove it
## Not run: 
pos_train_ultrahard &lt;- read_uniprot("pos_ultrahard_data.txt", euk = TRUE)
model1 &lt;- train_hsmm(pos_train_ultrahard, aa_group = aaaggregation)
predict(model1, benchmark_dat[1L:5])

## End(Not run)
</code></pre>

<hr>
<h2 id='read_txt'>Read sequences from .txt file</h2><span id='topic+read_txt'></span>

<h3>Description</h3>

<p>Read sequence data saved in text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_txt(connection)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_txt_+3A_connection">connection</code></td>
<td>
<p>a <code><a href="base.html#topic+connection">connection</a></code> to the text (.txt) file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input file should contain one or more amino acid sequences separated by 
empty line(s).
</p>


<h3>Value</h3>

<p>a list of sequences. Each element has class <code><a href="seqinr.html#topic+SeqFastaAA">SeqFastaAA</a></code>. If
connection contains no characters, function prompts warning and returns <code>NULL</code>.
</p>

<hr>
<h2 id='read_uniprot'>Read data from UniProt database</h2><span id='topic+read_uniprot'></span>

<h3>Description</h3>

<p>Read data saved in UniProt original flat text format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_uniprot(connection, ft_names, kwds = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_uniprot_+3A_connection">connection</code></td>
<td>
<p>a <code><a href="base.html#topic+connection">connection</a></code> to UniProt data in text format.</p>
</td></tr>
<tr><td><code id="read_uniprot_+3A_ft_names">ft_names</code></td>
<td>
<p>a character vector of UuniProt features to be extracted, for example 
<code>"signal"</code>, <code>"transit"</code>, <code>"propep"</code>. The case is not matched.</p>
</td></tr>
<tr><td><code id="read_uniprot_+3A_kwds">kwds</code></td>
<td>
<p>a <code>NULL</code> or character vector of keywords (not UniProt keywords, 
but words of interest, that may occur in the protein description).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of sequences. Each element has a class <code><a href="seqinr.html#topic+SeqFastaAA">SeqFastaAA</a></code>.
Attributes <code>OS</code> and <code>OC</code> represents respectively OS and OC fields in the 
protein description. A value of each feature is preserved as an attribute named 
after the feature.
</p>

<hr>
<h2 id='run_signalHsmm'>Predict presence of signal peptide in protein</h2><span id='topic+run_signalHsmm'></span>

<h3>Description</h3>

<p>Using the hidden semi-Markov model predict presence of signal peptide in 
eukaryotic proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_signalHsmm(test_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_signalHsmm_+3A_test_data">test_data</code></td>
<td>
<p>single protein sequence (<code>character</code> vector) or list of 
sequences. It may be an object of class <code><a href="seqinr.html#topic+SeqFastaAA">SeqFastaAA</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>signalHsmm</code> returns respectively probability of presence of 
signal peptide, start of signal peptide and the probable cleavage site localization.
If input consists of more than one sequence, result is a data.frame where each column
contains above values for different proteins.
</p>


<h3>Value</h3>

<p>An object of class <code>hsmm_pred_list</code>.
</p>


<h3>Note</h3>

<p>Currently start of signal peptide is naively set as 1 amino acid. The prediction 
of a cleavage site is still an experimental feature, use on your own risk.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hsmm_pred_list">hsmm_pred_list</a></code> <code><a href="#topic+hsmm_pred">hsmm_pred</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#run signalHsmm on one sequence
x1 &lt;- run_signalHsmm(benchmark_dat[[1]])

#run signalHsmm on one sequence, but input is a character vector
x2 &lt;- run_signalHsmm(c("M", "A", "G", "K", "E", "V", "I", "F", "I", "M", "A", "L", 
"F", "I", "A", "V", "E", "S", "S", "P", "I", "F", "S", "F", "D", 
"D", "L", "V", "C", "P", "S", "V", "T", "S", "L", "R", "V", "N", 
"V", "E", "K", "N", "E", "C", "S", "T", "K", "K", "D", "C", "G", 
"R", "N", "L", "C", "C", "E", "N", "Q", "N", "K", "I", "N", "V", 
"C", "V", "G", "G", "I", "M", "P", "L", "P", "K", "P", "N", "L", 
"D", "V", "N", "N", "I", "G", "G", "A", "V", "S", "E", "S", "V", 
"K", "Q", "K", "R", "E", "T", "A", "E", "S", "L"))

#run signalHsmm on list of sequences
x3 &lt;- run_signalHsmm(benchmark_dat[1:3])
#see summary of results
summary(x3)
#print results as data frame
pred2df(x3)
#summary one result
summary(x3[[1]])
plot(x3[[1]])
</code></pre>

<hr>
<h2 id='summary.hsmm_pred'>Summarize single signalHsmm prediction</h2><span id='topic+summary.hsmm_pred'></span>

<h3>Description</h3>

<p>Summarizes objects of class <code><a href="#topic+hsmm_pred">hsmm_pred</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hsmm_pred'
summary(object, only_sure = TRUE,
  double_linebreak = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.hsmm_pred_+3A_object">object</code></td>
<td>
<p>of class <code><a href="#topic+hsmm_pred">hsmm_pred</a></code>.</p>
</td></tr>
<tr><td><code id="summary.hsmm_pred_+3A_only_sure">only_sure</code></td>
<td>
<p><code>logical</code>, if <code>FALSE</code> does not draw signal peptide structure
when probability is smaller than 0.5.</p>
</td></tr>
<tr><td><code id="summary.hsmm_pred_+3A_double_linebreak">double_linebreak</code></td>
<td>
<p><code>logical</code>, if <code>TRUE</code> adds aestically pleasing (in .Rmd)
double linebreaks between the elements of summary.</p>
</td></tr>
<tr><td><code id="summary.hsmm_pred_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>

<hr>
<h2 id='summary.hsmm_pred_list'>Summarize list of signalHsmm predictions</h2><span id='topic+summary.hsmm_pred_list'></span>

<h3>Description</h3>

<p>Summarizes objects of class <code><a href="#topic+hsmm_pred_list">hsmm_pred_list</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'hsmm_pred_list'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.hsmm_pred_list_+3A_object">object</code></td>
<td>
<p>of class <code><a href="#topic+hsmm_pred_list">hsmm_pred_list</a></code>.</p>
</td></tr>
<tr><td><code id="summary.hsmm_pred_list_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing.
</p>

<hr>
<h2 id='train_hsmm'>Train sighsmm_model object</h2><span id='topic+train_hsmm'></span>

<h3>Description</h3>

<p>Train sighsmm_model object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>train_hsmm(train_data, aa_group, max_length = 32,
  region_fun = find_nhc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="train_hsmm_+3A_train_data">train_data</code></td>
<td>
<p>training data.</p>
</td></tr>
<tr><td><code id="train_hsmm_+3A_aa_group">aa_group</code></td>
<td>
<p>method of aggregating amino acids.</p>
</td></tr>
<tr><td><code id="train_hsmm_+3A_max_length">max_length</code></td>
<td>
<p>maximum length of signal peptide.</p>
</td></tr>
<tr><td><code id="train_hsmm_+3A_region_fun">region_fun</code></td>
<td>
<p>function defining borders of regions (see <code><a href="#topic+find_nhc">find_nhc</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>sighsmm_model</code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
