<!DOCTYPE html><html><head><title>Help for package sigora</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sigora}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#genesFromRandomPathways'><p>Function to randomly select genes associated with randomly pathways.</p></a></li>
<li><a href='#getGenes'><p>List genes involved in present GPS for a specific pathway in the</p>
summary_results</a></li>
<li><a href='#getURL'><p>Highlight the relevant genes for a specific pathway in its pathway diagram</p></a></li>
<li><a href='#idmap'><p>Identifier mappings for protein coding genes.</p></a></li>
<li><a href='#kegH'><p>Pathway GPS data, extracted from KEGG repository (Human).</p></a></li>
<li><a href='#kegM'><p>Pathway GPS data, extracted from KEGG repository (Mouse).</p></a></li>
<li><a href='#load_data'><p>load and return data when lazyLoad false insted of using data(datastr)</p></a></li>
<li><a href='#makeGPS'><p>Create your own Signature Object.</p></a></li>
<li><a href='#nciTable'><p>NCI human gene-pathway associations.</p></a></li>
<li><a href='#ora'><p>Traditional Overrepresentation Analysis.</p></a></li>
<li><a href='#reaH'><p>Pathway GPS data, extracted from the Reactome repository (Human).</p></a></li>
<li><a href='#reaM'><p>Pathway GPS data, extracted from Reactome repository (Mouse).</p></a></li>
<li><a href='#sigora'><p>Sigora's main function.</p></a></li>
<li><a href='#sigora-package'><p>sigora: Signature Overrepresentation Analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Signature Overrepresentation Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>3.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-29</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/wolski/sigora">https://github.com/wolski/sigora</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/wolski/sigora/issues">https://github.com/wolski/sigora/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, slam</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>Description:</td>
<td>
     Pathway Analysis is statistically linking observations on 
     the molecular level to biological processes or pathways on 
     the systems(i.e., organism, organ, tissue, cell) level.  
     Traditionally, pathway analysis methods regard pathways 
     as collections of single genes and treat all genes in a pathway
     as equally informative. However, this can lead to identifying
     spurious pathways as statistically significant since components
     are often shared amongst pathways. SIGORA seeks to avoid this
     pitfall by focusing on genes or gene pairs that are (as a combination)
     specific to a single pathway.
     In relying on such pathway gene-pair signatures (Pathway-GPS),
     SIGORA inherently uses the status of other genes in the
     experimental context to identify the most relevant pathways.
     The current version allows for pathway analysis of human
     and mouse datasets. In addition, it contains pre-computed
     Pathway-GPS data for pathways in the KEGG and  Reactome
     pathway repositories and mechanisms for extracting GPS
     for user-supplied repositories.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>biocViews:</td>
<td>GeneSetEnrichment, GO, Software, Pathways, KEGG</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-29 12:47:12 UTC; wolski</td>
</tr>
<tr>
<td>Author:</td>
<td>Amir Foroushani <a href="https://orcid.org/0000-0003-2748-3009"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Fiona Brinkman [aut],
  David Lynn [aut],
  Witold Wolski <a href="https://orcid.org/0000-0002-6468-120X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Witold Wolski &lt;wew@fgcz.ethz.ch&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-30 07:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='genesFromRandomPathways'>Function to randomly select genes associated with randomly pathways.</h2><span id='topic+genesFromRandomPathways'></span>

<h3>Description</h3>

<p>This function first randomly selects a number (np) of pathways, then
randomly selects a number (ng) of genes that are associated with at least
one of the selected pathways.  The function can be used to compare Sigora's
performance to traditional overrepresentation tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genesFromRandomPathways(GPSrepo, np, ng)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genesFromRandomPathways_+3A_gpsrepo">GPSrepo</code></td>
<td>
<p>A signature repository (created by ..) or one of the
precompiled options.</p>
</td></tr>
<tr><td><code id="genesFromRandomPathways_+3A_np">np</code></td>
<td>
<p>How many pathways to select.</p>
</td></tr>
<tr><td><code id="genesFromRandomPathways_+3A_ng">ng</code></td>
<td>
<p>Number of genes to be selected.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>selectedPathways</code></td>
<td>
<p>A vector containing the &quot;np&quot; originally
selected pathways.</p>
</td></tr> <tr><td><code>genes</code></td>
<td>
<p> A vector containing the &quot;ng&quot; selected
genes from selectedPathways.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Foroushani AB, Brinkman FS and Lynn DJ
(2013).&ldquo;Pathway-GPS and SIGORA: identifying relevant pathways based
on the over-representation of their gene-pair signatures.&rdquo;<em>PeerJ</em>,
<b>1</b>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sigora-package">sigora-package</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data('kegH')
## select 50 genes from 3 human KEGG pathways
seed=1234
set.seed(seed)
a1 &lt;- genesFromRandomPathways(kegH,3,50)
## originally selected pathways:
a1[["selectedPathways"]]
## what are the genes
a1[["genes"]]
## sigora's results
sigoraRes &lt;- sigora(GPSrepo =kegH, queryList = a1[["genes"]],
        level = 4)
## compare to traditional methods results:
oraRes &lt;- ora(a1[["genes"]],kegH)
dim(oraRes)
oraRes

</code></pre>

<hr>
<h2 id='getGenes'>List genes involved in present GPS for a specific pathway in the
summary_results</h2><span id='topic+getGenes'></span>

<h3>Description</h3>

<p>This function lists the genes involved in the present GPS for a pathway of
interest, odered by their contribution to the significance of the pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGenes(yy, i, idmap = load_data("idmap"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGenes_+3A_yy">yy</code></td>
<td>
<p>A sigora analysis result object (created by <code>sigora</code>).</p>
</td></tr>
<tr><td><code id="getGenes_+3A_i">i</code></td>
<td>
<p>The rank position of the pathway of interest in summary_results.</p>
</td></tr>
<tr><td><code id="getGenes_+3A_idmap">idmap</code></td>
<td>
<p>A dataframe for converting between different gene-identifier
types (e.g. ENSEMBL, ENTREZ and HGNC-Symbols of genes). Most users do not
need to set this argument, as there is a built-in conversion table.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table (dataframe) with ids of the relevant genes, ranked by their
contribution to the statistical significance of the pathway.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sigora">sigora</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data('kegH')
set.seed(seed=12345)
a1 &lt;- genesFromRandomPathways(kegH,3,50)
## originally selected pathways:\cr
a1[["selectedPathways"]]
## what are the genes
a1[["genes"]]
## sigora's results with this input:\cr
sigoraRes &lt;- sigora(GPSrepo = kegH, queryList = a1[["genes"]],level = 2)
## Genes related to the second most significant result:
head(getGenes(sigoraRes,2))

</code></pre>

<hr>
<h2 id='getURL'>Highlight the relevant genes for a specific pathway in its pathway diagram</h2><span id='topic+getURL'></span>

<h3>Description</h3>

<p>This function highlights the genes involved in the present GPS for a pathway
of interest in its diagram. Please note that this functionality is only
implemented for results of Reactome or KEGG based analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getURL(yy, i)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getURL_+3A_yy">yy</code></td>
<td>
<p>A sigora analysis result object (created by <code>sigora</code>).</p>
</td></tr>
<tr><td><code id="getURL_+3A_i">i</code></td>
<td>
<p>The rank position of the pathway of interest in summary_results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The URL of the pathway diagram, where the relevant genes from your
original query list are highlighted.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sigora">sigora</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data('kegH')
set.seed(seed=12345)
a1&lt;-genesFromRandomPathways(kegH,3,50)
## originally selected pathways:\cr
a1[["selectedPathways"]]
## what are the genes
a1[["genes"]]
## sigora's results with this input:\cr
sigoraRes &lt;- sigora(GPSrepo =kegH, queryList = a1[["genes"]],level = 2)
## Diagram for the most significant result, where the genes from our list are highlighted in red:
getURL(sigoraRes,1)

</code></pre>

<hr>
<h2 id='idmap'>Identifier mappings for protein coding genes.</h2><span id='topic+idmap'></span>

<h3>Description</h3>

<p>A mapping table for ENSEMBL, ENTREZ and gene names(HGNC/MGI symbols) of
Human and mouse protein coding gene.
</p>


<h3>Source</h3>

<p>www.ensembl.org/biomart/martview
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(idmap)
head(idmap)

</code></pre>

<hr>
<h2 id='kegH'>Pathway GPS data, extracted from KEGG repository (Human).</h2><span id='topic+kegH'></span>

<h3>Description</h3>

<p>KEGG human pathway GPS data, extracted by <code>makeGPS</code>, default settings.
This data can be used by <code>sigora</code> to preform signature
overrepresenation.
</p>


<h3>Source</h3>

<p>&lt;http://www.genome.jp/kegg/pathway.html&gt;
</p>


<h3>References</h3>

<p>Kanehisa, M., Goto, S., Sato, Y., Furumichi, M., &amp; Tanabe, M.
2012. &ldquo;KEGG for integration and interpretation of large-scale
molecular data sets.&rdquo; <em>Nucleic Acids Research</em> <b>40</b>(D1).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makeGPS">makeGPS</a></code>, <code><a href="#topic+sigora">sigora</a></code> , <code><a href="#topic+reaH">reaH</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(kegH)
str(kegH)

</code></pre>

<hr>
<h2 id='kegM'>Pathway GPS data, extracted from KEGG repository (Mouse).</h2><span id='topic+kegM'></span>

<h3>Description</h3>

<p>KEGG mouse pathway GPS data, extracted by <code>makeGPS</code>, default settings.
This data can be used by <code>sigora</code> to preform signature
overrepresenation.
</p>


<h3>Source</h3>

<p>&lt;http://www.genome.jp/kegg/pathway.html&gt;
</p>


<h3>References</h3>

<p>Kanehisa, M., Goto, S., Sato, Y., Furumichi, M., &amp; Tanabe, M.
2012. &ldquo;KEGG for integration and interpretation of large-scale
molecular data sets.&rdquo; <em>Nucleic Acids Research</em> <b>40</b>(D1).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(kegM)
## maybe str(kegM) ; plot(kegM) ...

</code></pre>

<hr>
<h2 id='load_data'>load and return data when lazyLoad false insted of using data(datastr)</h2><span id='topic+load_data'></span>

<h3>Description</h3>

<p>load and return data when lazyLoad false insted of using data(datastr)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_data(datastr, package = "sigora")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_data_+3A_datastr">datastr</code></td>
<td>
<p>name of datasets</p>
</td></tr>
<tr><td><code id="load_data_+3A_package">package</code></td>
<td>
<p>default sigora</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>idmap &lt;- load_data("idmap")

</code></pre>

<hr>
<h2 id='makeGPS'>Create your own Signature Object.</h2><span id='topic+makeGPS'></span>

<h3>Description</h3>

<p>Given a repository of gene-pathway associations either in a tab delimited
file with three columns (pathwayID,pathway Description,Gene) or a
corresponding dataframe, this function identifies all Gene Pair Signatures
(pairs of genes that are as a combination unique to a single pathway) and
Pathway Unique Genes (genes that are uniquely associated with a single
pathway) and stores them in a format that is usable by <code>sigora</code>.
Please also see the &quot;details&quot; and &quot;note&quot; sections below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeGPS(
  pathwayTable = NULL,
  fn = NULL,
  maxLevels = 5,
  saveFile = NULL,
  repoName = "userrepo",
  maxFunperGene = 100,
  maxGenesperPathway = 500,
  minGenesperPathway = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeGPS_+3A_pathwaytable">pathwayTable</code></td>
<td>
<p>A data frame describing gene-pathway associations in
following format: pathwayID,pathwayName,Gene. Either pathwayTable or fn
should be provided.</p>
</td></tr>
<tr><td><code id="makeGPS_+3A_fn">fn</code></td>
<td>
<p>Where to find the repository.Either pathwayTable or fn should be
provided.</p>
</td></tr>
<tr><td><code id="makeGPS_+3A_maxlevels">maxLevels</code></td>
<td>
<p>For hierarchical repositories, the number of levels to
consider.</p>
</td></tr>
<tr><td><code id="makeGPS_+3A_savefile">saveFile</code></td>
<td>
<p>Where to save the object as an rda file.</p>
</td></tr>
<tr><td><code id="makeGPS_+3A_reponame">repoName</code></td>
<td>
<p>Repository name.</p>
</td></tr>
<tr><td><code id="makeGPS_+3A_maxfunpergene">maxFunperGene</code></td>
<td>
<p>A cutoff threshold, genes with more than this number of
associated pathways are excluded to speed up the GPS identification process.</p>
</td></tr>
<tr><td><code id="makeGPS_+3A_maxgenesperpathway">maxGenesperPathway</code></td>
<td>
<p>A cutoff threshold, pathways with more than this
number of associated genes are excluded to speed up the GPS identification
process.</p>
</td></tr>
<tr><td><code id="makeGPS_+3A_mingenesperpathway">minGenesperPathway</code></td>
<td>
<p>A cutoff threshold, pathways with less than this
number of associated genes are excluded to speed up the GPS identification
process.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The primary purpose of <code>makeGPS</code> is to convert a user-supplied
gene-pathway association table to a repository of weighted Gene Pair
Signatures (GPS) that are unique features of pathways. Such GPS can than be
used for signature (gene-pair) based analyses using <code>sigora</code>.
Additionally, the resulting object also retains the original &quot;single
gene&quot;-&quot;pathway&quot; associations for the purpose of followup analyses, such as
comparison of sigora-results to traditional methods. <code>ora</code> is an
implementation of the traditional (individual gene) Overrepresentation
Analysis.
</p>


<h3>Value</h3>

<p>A GPS repository, to be used by <code>sigora</code> and <code>ora.</code>
</p>


<h3>Note</h3>

<p>This function relies on package <code>slam</code>, which should be installed
from CRAN.  It is fairly memory intensive, and it is recommended to be run
on a machine with at least 6GB of RAM. Also, make sure to save and reuse the
resulting GPS repository in future analyses!
</p>


<h3>References</h3>

<p>Foroushani AB, Brinkman FS and Lynn DJ
(2013).&ldquo;Pathway-GPS and SIGORA: identifying relevant pathways based
on the over-representation of their gene-pair signatures.&rdquo;<em>PeerJ</em>,
<b>1</b>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sigora">sigora</a></code>, <code><a href="#topic+sigora-package">sigora-package</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nciTable); data(idmap)
## what the input looks like:
head(nciTable)
## create a SigObject. use the saveFile parameter for reuse.

nciH&lt;-makeGPS(pathwayTable=load_data('nciTable'))
ils&lt;-grep("^IL",idmap[,"Symbol"],value=TRUE)
ilnci&lt;-sigora(queryList=ils,GPSrepo=nciH,level=3)

</code></pre>

<hr>
<h2 id='nciTable'>NCI human gene-pathway associations.</h2><span id='topic+nciTable'></span>

<h3>Description</h3>

<p>PID-NCI human pathway repository, as a data frame with three columns
corresponding to : pathwayId , pathwayName, gene. This is an example of the
expected format for the input data to <code>makeGPS</code>.
</p>


<h3>Details</h3>

<p>This dataset is provided to illustrate how to create your own GPS
repositories.<code>nciTable</code> is a dataframe with threecolumns corresponding
to pathwayId, pathwayName and gene.  Each row describes the association
between an individual gene and a PID-NCI pathway.  As you see in the
<code>examples</code>, section, one can convert this dataframe to a <code>GPS
repository</code> using <code>makeGPS</code>, and save the results for future reuse.
Using the thus created <code>GPS repository</code> one can preform Signature
Overrepresentation Analysis on lists of genes of interest.
</p>


<h3>Source</h3>

<p>&lt;https://github.com/NCIP/pathway-interaction-database/tree/master/download&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nciTable)
nciH&lt;-makeGPS(pathwayTable=load_data('nciTable'))
data(idmap)
ils&lt;-grep("^IL",idmap[,"Symbol"],value=TRUE)
ilnci&lt;-sigora(queryList=ils,GPSrepo=nciH,level=3)

</code></pre>

<hr>
<h2 id='ora'>Traditional Overrepresentation Analysis.</h2><span id='topic+ora'></span>

<h3>Description</h3>

<p>Traditional Overrepresentation Analysis by hypergeometric test: pathways are
treated as collections of individual genes and all genes are treated as
equally informative. This function is provided for comparison of the results
of traditional methods to Sigora.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ora(geneList, GPSrepo, idmap = load_data("idmap"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ora_+3A_genelist">geneList</code></td>
<td>
<p>A vector containing the list of genes of interest (e.g.
differentially expressed genes). Following Identifier types are supported:
Gene Symbols, ENTREZ-IDs or ENSEMBL-IDs.</p>
</td></tr>
<tr><td><code id="ora_+3A_gpsrepo">GPSrepo</code></td>
<td>
<p>A GPS-repository (either one of the provided precomputed GPS
repositories) or one created by <code>makeGPS</code>.</p>
</td></tr>
<tr><td><code id="ora_+3A_idmap">idmap</code></td>
<td>
<p>A dataframe for converting between different gene-identifier
types (e.g. ENSEMBL, ENTREZ and HGNC-Symbols of genes). Most users do not
need to set this argument, as there is a built-in conversion table.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The primary purpose of <code>makeGPS</code> is to create a GPS repository. It
does, however, retain the original &quot;single gene&quot;-&quot;pathway&quot; associations for
the purpose of followup analyses, such as comparison of sigora-results to
traditional methods. <code>ora</code> is an implementation of the traditional
(individual gene) Overrepresentation Analysis.
</p>


<h3>Value</h3>

<p>A dataframe with individual gene ORA results.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sigora-package">sigora-package</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(kegM)
## select 50 genes from 3 mouse pathways
set.seed(seed=345)
a1&lt;-genesFromRandomPathways(kegM,3,50)
## originally selected pathways:
a1[["selectedPathways"]]
## compare to traditional methods results:
oraRes &lt;- ora(a1[["genes"]],kegM)
dim(oraRes)
oraRes

</code></pre>

<hr>
<h2 id='reaH'>Pathway GPS data, extracted from the Reactome repository (Human).</h2><span id='topic+reaH'></span>

<h3>Description</h3>

<p>Reactome human pathway GPS data, extracted by <code>makeGPS</code>, default
settings. This data can be used by <code>sigora</code> to preform signature
overrepresenation.
</p>


<h3>Source</h3>

<p>&lt;http://www.reactome.org/&gt;
</p>


<h3>References</h3>

<p>Matthews, L., Gopinath, G., Gillespie, M., Caudy, M., Croft, D.,
et al. 2009. &ldquo;Reactome knowledgebase of human biological pathways and
processes.&rdquo; <em>Nucleic acids research</em> <b>37</b>(Database issue).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(reaH)
## maybe str(reaH) ;  ...

</code></pre>

<hr>
<h2 id='reaM'>Pathway GPS data, extracted from Reactome repository (Mouse).</h2><span id='topic+reaM'></span>

<h3>Description</h3>

<p>Reactome mouse pathway GPS data, extracted by <code>makeGPS</code>, default
settings. This data can be used by <code>sigora</code> to preform signature
overrepresenation.
</p>


<h3>Source</h3>

<p>&lt;http://www.reactome.org/&gt;
</p>


<h3>References</h3>

<p>Matthews, L., Gopinath, G., Gillespie, M., Caudy, M., Croft, D.,
et al. 2009. &ldquo;Reactome knowledgebase of human biological pathways and
processes.&rdquo; <em>Nucleic acids research</em> <b>37</b>(Database issue).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makeGPS">makeGPS</a></code>, <code><a href="#topic+sigora">sigora</a></code> , <code><a href="#topic+kegM">kegM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(reaM)
str(reaM)

</code></pre>

<hr>
<h2 id='sigora'>Sigora's main function.</h2><span id='topic+sigora'></span>

<h3>Description</h3>

<p>This function determines which Signatures (GPS) from a collection of GPS
data (<code>GPSrepo</code> argument) for the specified pathway repository are
present in the specified list of genes of interest (<code>queryList</code>
argument)). It then uses the distribution function of hypergeometric
probabilities to identify the pathways whose GPS are over-represented among
the present GPS and saves the results to the file specified in the
<code>saveFile</code> argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigora(
  GPSrepo,
  level,
  markers = FALSE,
  queryList = NULL,
  saveFile = NULL,
  weighting.method = "invhm",
  idmap = load_data("idmap")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sigora_+3A_gpsrepo">GPSrepo</code></td>
<td>
<p>An object created by <code>makeGPS</code> or one of the precompiled
GPS data collections that are provided with this package (currently for KEGG
and Reactome). e.g. <code>reaH</code> for human Reactome GPS, <code>kegH</code> for
human KEGG GPS, and <code>reaM</code> and <code>kegM</code> for corresponding mouse GPS.
See the examples section for creating and using your own GPS.</p>
</td></tr>
<tr><td><code id="sigora_+3A_level">level</code></td>
<td>
<p>In hierarchical repositories (e.g. Reactome) number of levels
to consider. Recommended value for KEGG: 2, for Reactome: 4.</p>
</td></tr>
<tr><td><code id="sigora_+3A_markers">markers</code></td>
<td>
<p>Whether to take single genes that are uniquely associated
with only one pathway into account (i.e. should pathway unique genes/PUGs be
considered GPS?). Recommended value: TRUE (1).</p>
</td></tr>
<tr><td><code id="sigora_+3A_querylist">queryList</code></td>
<td>
<p>A user specified list of genes of interest ('query list'),
as a vector of ENSEMBL/ ENTREZ IDs or gene symbols (HGNC/MGI).</p>
</td></tr>
<tr><td><code id="sigora_+3A_savefile">saveFile</code></td>
<td>
<p>If provided, the results are saved here as a tab delimited
File (including , for each pathway, a list of genes ordered by their
contribution to the statistical significance of the pathway).</p>
</td></tr>
<tr><td><code id="sigora_+3A_weighting.method">weighting.method</code></td>
<td>
<p>The weighting method or GPS. The default weighting
scheme for the GPS is the reciproc of the harmonic mean of the degrees of
the two component genes of a GPS. A wide range of alternative weighting
schemes are pre-implemented (see below). Additional user defined weighting
schemes are also supported. Currently, the following alternatives are
pre-implemented:<br />
'noweights','cosine','topov','reciprod','jac','justPUGs'and'invhm'.<br />
Additional user defined weighting schemes are also supported (see section
examples).<br /> 'noweights': assigns a constant of 1 to all GPS.<br /> 'cosine':
all GPS are weighted by the cosine of the degrees of their consituent
genes.<br /> 'topov': all GPS are weighted by topological overlap of their
consituent genes.<br /> 'reciprod': all GPS are weighted by reciproc of product
of the number of pathway annotations of their consituent genes.<br /> 'jac':all
GPS are weighted by the jaccard similarity of the pathway annotations
consituent genes.<br /> 'justPUGs': Analysis is performed using PUGs only.<br />
'invhm': all GPS are weighted by the reciproc of the harmonic mean of the
degrees of their consituent genes (default).<br /></p>
</td></tr>
<tr><td><code id="sigora_+3A_idmap">idmap</code></td>
<td>
<p>A dataframe for converting between different gene-identifier
types (e.g. ENSEMBL, ENTREZ and HGNC-Symbols of genes). Most users do not
need to set this argument, as there is a built-in conversion table.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>summary_results</code></td>
<td>
<p>A dataframe listing the analysis results.</p>
</td></tr>
<tr><td><code>detailed_results</code></td>
<td>
<p>A dataframe describing the detailed evidence
(present Gene-Pair Signatures) for each pathway.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Foroushani AB, Brinkman FS and Lynn DJ
(2013).&ldquo;Pathway-GPS and SIGORA: identifying relevant pathways based
on the over-representation of their gene-pair signatures.&rdquo;<em>PeerJ</em>,
<b>1</b>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sigora-package">sigora-package</a></code> , <code><a href="#topic+makeGPS">makeGPS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##query list
ils &lt;- grep("^IL",load_data('idmap')[["Symbol"]],value=TRUE)
## using precompiled GPS repositories:
sigRes.ilreact &lt;- sigora(queryList=ils,GPSrepo=load_data('reaH'),level=4)

sigRes.ilkeg &lt;- sigora(queryList=ils,GPSrepo=load_data('kegH'),level=2)
## user created GPS repository:
nciH&lt;-makeGPS(pathwayTable=load_data('nciTable'))
sigRes.ilnci&lt;-sigora(queryList=ils,GPSrepo=nciH,level=2)
## user defined weighting schemes :
myfunc&lt;-function(a,b){1/log(a+b)}
sigora(queryList=ils,GPSrepo=nciH,level=2, weighting.method = myfunc)

</code></pre>

<hr>
<h2 id='sigora-package'>sigora: Signature Overrepresentation Analysis</h2><span id='topic+sigora-package'></span><span id='topic+_PACKAGE'></span>

<h3>Description</h3>

<p>Pathway Analysis is statistically linking observations on the molecular level to biological processes or pathways on the systems(i.e., organism, organ, tissue, cell) level. Traditionally, pathway analysis methods regard pathways as collections of single genes and treat all genes in a pathway as equally informative. However, this can lead to identifying spurious pathways as statistically significant since components are often shared amongst pathways. SIGORA seeks to avoid this pitfall by focusing on genes or gene pairs that are (as a combination) specific to a single pathway. In relying on such pathway gene-pair signatures (Pathway-GPS), SIGORA inherently uses the status of other genes in the experimental context to identify the most relevant pathways. The current version allows for pathway analysis of human and mouse datasets. In addition, it contains pre-computed Pathway-GPS data for pathways in the KEGG and Reactome pathway repositories and mechanisms for extracting GPS for user-supplied repositories.
</p>


<h3>Details</h3>

<p>Signature Overrepresentation Analysis
</p>
<p>This package implements the pathway analysis method SIGORA. For an in depth
description of the method, please see our manuscript in PeerJ.  In short: a
<em>GPS</em> (gene pair signature) is a (weighted) pair of genes that <em>as
a combination</em> occurs only in a single pathway within a pathway repository.
A query list is a vector containing a gene list of interest (e.g. genes that
are differentially expressed in a particular condition).  A <em>present</em>
GPS is a GPS for which both components are in the query list.  SIGORA
identifies relevant pathways based on the over-representation analysis of
their (present) GPS.
</p>
<p>For more details see the package vignette.
</p>


<h3>Author(s)</h3>

<p>Amir B.K. Foroushani, Fiona S.L. Brinkman, David J. Lynn
</p>
<p>Maintainer: Witold Wolski &lt;wew@fgcz.ethz.ch&gt;
</p>


<h3>References</h3>

<p>Foroushani AB, Brinkman FS and Lynn DJ
(2013).&ldquo;Pathway-GPS and SIGORA: identifying relevant pathways based
on the over-representation of their gene-pair signatures.&rdquo;<em>PeerJ</em>,
<b>1</b>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sigora">sigora</a></code>, <code><a href="#topic+makeGPS">makeGPS</a></code>, <code><a href="#topic+ora">ora</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
