<!DOCTYPE html><html><head><title>Help for package sim1000G</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sim1000G}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#computePairIBD1'><p>Computes pairwise IBD1 for a specific pair of individuals.</p>
See function computePairIBD12 for description.</a></li>
<li><a href='#computePairIBD12'><p>Computes pairwise IBD1/2 for a specific pair of individuals</p></a></li>
<li><a href='#computePairIBD2'><p>Computes pairwise IBD2 for a specific pair of individuals</p></a></li>
<li><a href='#createVCF'><p>Creates a regional vcf file using bcftools to extract a region from 1000 genomes vcf files</p></a></li>
<li><a href='#crossoverCDFvector'><p>Contains recombination model information.</p></a></li>
<li><a href='#downloadGeneticMap'><p>Downloads a genetic map for a particular chromosome under GRCh37 coordinates for use with sim1000G.</p></a></li>
<li><a href='#generateChromosomeRecombinationPositions'><p>Generates a recombination vector arising from one meiotic event.</p>
The origin of segments is coded as (0 - haplotype1 ,  1 - haplotype2 )</a></li>
<li><a href='#generateFakeWholeGenomeGeneticMap'><p>Generates a fake genetic map that spans the whole genome.</p></a></li>
<li><a href='#generateRecombinationDistances'><p>Generate inter-recombination distances using a chi-square model. Note this are the distances between two succesive recombination events and not</p>
the absolute positions of the events. To generate the locations of the recombination events see the example below.</a></li>
<li><a href='#generateRecombinationDistances_noInterference'><p>Generate recombination distances using a no-interference model.</p></a></li>
<li><a href='#generateSingleRecombinationVector'><p>Genetates a recombination vector arising from one meiotic event.</p>
The origin of segments is coded as (0 - haplotype1 ,  1 - haplotype2 )</a></li>
<li><a href='#generateUniformGeneticMap'><p>Generates a uniform genetic map.</p></a></li>
<li><a href='#generateUnrelatedIndividuals'><p>Generates variant data for n unrelated individuals</p></a></li>
<li><a href='#geneticMap'><p>Holds the genetic map information that is used for simulations.</p></a></li>
<li><a href='#getCMfromBP'><p>Converts centimorgan position to base-pair. Return a list of centimorgan positions that correspond</p>
to the bp vector (in basepairs).</a></li>
<li><a href='#loadSimulation'><p>Load some previously saved simulation data by function saveSimulation</p></a></li>
<li><a href='#newFamily3generations'><p>Generates genotype data for a family of 3 generations</p></a></li>
<li><a href='#newFamilyWithOffspring'><p>Simulates genotypes for 1 family with n offspring</p></a></li>
<li><a href='#newNuclearFamily'><p>Simulates genotypes for 1 family with 1 offspring</p></a></li>
<li><a href='#pkg.opts'><p>Holds general package options</p></a></li>
<li><a href='#plotRegionalGeneticMap'><p>Generates a plot of the genetic map for a specified region.</p></a></li>
<li><a href='#printMatrix'><p>Utility function that prints a matrix. Useful for IBD12 matrices.</p></a></li>
<li><a href='#readGeneticMap'><p>Reads a genetic map downloaded from the function downloadGeneticMap or reads a genetic map from a specified file. If the argument filename is used</p>
then the genetic map is read from the corresponding file. Otherwise, if a chromosome is specified, the genetic map is downloaded for human chromosome
using grch37 coordinates.</a></li>
<li><a href='#readGeneticMapFromFile'><p>Reads a genetic map to be used for simulations. The genetic map should be</p>
of a single chromosome and covering the extent of the region to be simulated.
Whole chromosome genetic maps can also be used.</a></li>
<li><a href='#readVCF'><p>Read a vcf file, with options to filter out low or high frequency markers.</p></a></li>
<li><a href='#resetSimulation'><p>Removes all individuals that have been simulated and resets the simulator.</p></a></li>
<li><a href='#retrieveGenotypes'><p>Retrieve a matrix of simulated genotypes for a specific set of individual IDs</p></a></li>
<li><a href='#saveSimulation'><p>Save the data for a simulation for later use. When simulating multiple populations it</p>
allows saving and restoring of simulation data for each population.</a></li>
<li><a href='#setRecombinationModel'><p>Set recombination model to either poisson (no interference) or chi-square.</p></a></li>
<li><a href='#SIM'><p>Holds data necessary for a simulation.</p></a></li>
<li><a href='#sim1000G-package'><p>Simulations of rare/common variants using haplotype data from 1000 genomes</p></a></li>
<li><a href='#startSimulation'><p>Starts and initializes the data structures required for a simulation. A VCF file</p>
should be read beforehand with the function readVCF.</a></li>
<li><a href='#subsetVCF'><p>Generate a market subset of a vcf file</p></a></li>
<li><a href='#writePED'><p>Writes a plink compatible PED/MAP file from the simulated genotypes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genotype Simulations for Rare or Common Variants Using
Haplotypes from 1000 Genomes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.40</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-05-30</td>
</tr>
<tr>
<td>Author:</td>
<td>Apostolos Dimitromanolakis &lt;apostolis@live.ca&gt;, 
    Jingxiong Xu &lt;jingxiong.xu@mail.utoronto.ca&gt;,
    Agnieszka Krol &lt;krol@lunenfeld.ca&gt;,
    Laurent Briollais &lt;laurent@lunenfeld.ca&gt; </td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Apostolos Dimitromanolakis &lt;apostolis@live.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Generates realistic simulated genetic data in families or unrelated individuals.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.2), stats, hapsim, MASS, stringr, readr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, prettydoc, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, prettydoc, testthat, gplots, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-06-09 19:28:35 UTC; apo</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-06-09 19:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='computePairIBD1'>Computes pairwise IBD1 for a specific pair of individuals.
See function computePairIBD12 for description.</h2><span id='topic+computePairIBD1'></span>

<h3>Description</h3>

<p>Computes pairwise IBD1 for a specific pair of individuals.
See function computePairIBD12 for description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computePairIBD1(i, j)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computePairIBD1_+3A_i">i</code></td>
<td>
<p>Index of first individual</p>
</td></tr>
<tr><td><code id="computePairIBD1_+3A_j">j</code></td>
<td>
<p>Index of second individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Mean IBD1 as computed from shared haplotypes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)

# For realistic data use the function downloadGeneticMap
generateUniformGeneticMap()

startSimulation(vcf, totalNumberOfIndividuals = 200)

ped1 = newNuclearFamily(1)

v = computePairIBD1(1, 3)

cat("IBD1 of pair = ", v, "\n");

</code></pre>

<hr>
<h2 id='computePairIBD12'>Computes pairwise IBD1/2 for a specific pair of individuals</h2><span id='topic+computePairIBD12'></span>

<h3>Description</h3>

<p>Computes pairwise IBD1/2 for a specific pair of individuals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computePairIBD12(i, j)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computePairIBD12_+3A_i">i</code></td>
<td>
<p>Index of first individual</p>
</td></tr>
<tr><td><code id="computePairIBD12_+3A_j">j</code></td>
<td>
<p>Index of second individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Mean IBD1 and IBD2 as computed from shared haplotypes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")

vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)

generateUniformGeneticMap()

startSimulation(vcf, totalNumberOfIndividuals = 200)

ped1 = newNuclearFamily(1)

v = computePairIBD12(1, 3)

cat("IBD1 of pair = ", v[1], "\n");
cat("IBD2 of pair = ", v[2], "\n");


</code></pre>

<hr>
<h2 id='computePairIBD2'>Computes pairwise IBD2 for a specific pair of individuals</h2><span id='topic+computePairIBD2'></span>

<h3>Description</h3>

<p>Computes pairwise IBD2 for a specific pair of individuals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computePairIBD2(i, j)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computePairIBD2_+3A_i">i</code></td>
<td>
<p>Index of first individual</p>
</td></tr>
<tr><td><code id="computePairIBD2_+3A_j">j</code></td>
<td>
<p>Index of second individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Mean IBD2 as computed from shared haplotypes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)

# For realistic data use the function downloadGeneticMap
generateUniformGeneticMap()

startSimulation(vcf, totalNumberOfIndividuals = 200)

ped1 = newNuclearFamily(1)

v = computePairIBD2(1, 3)

cat("IBD2 of pair = ", v, "\n");

</code></pre>

<hr>
<h2 id='createVCF'>Creates a regional vcf file using bcftools to extract a region from 1000 genomes vcf files</h2><span id='topic+createVCF'></span>

<h3>Description</h3>

<p>Creates a regional vcf file using bcftools to extract a region from 1000 genomes vcf files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createVCF()
</code></pre>


<h3>Value</h3>

<p>none
</p>

<hr>
<h2 id='crossoverCDFvector'>Contains recombination model information.</h2><span id='topic+crossoverCDFvector'></span>

<h3>Description</h3>

<p>This vector contains the density between two recombination events, as a cumulative density function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crossoverCDFvector
</code></pre>


<h3>Format</h3>

<p>An object of class <code>logical</code> of length 1.</p>

<hr>
<h2 id='downloadGeneticMap'>Downloads a genetic map for a particular chromosome under GRCh37 coordinates for use with sim1000G.</h2><span id='topic+downloadGeneticMap'></span>

<h3>Description</h3>

<p>Downloads a genetic map for a particular chromosome under GRCh37 coordinates for use with sim1000G.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downloadGeneticMap(chromosome, dir = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="downloadGeneticMap_+3A_chromosome">chromosome</code></td>
<td>
<p>Chromosome number to download recombination distances from.</p>
</td></tr>
<tr><td><code id="downloadGeneticMap_+3A_dir">dir</code></td>
<td>
<p>Directory to save the genetic map to (default: temporary directory)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>


downloadGeneticMap(22, dir=tempdir() )


</code></pre>

<hr>
<h2 id='generateChromosomeRecombinationPositions'>Generates a recombination vector arising from one meiotic event.
The origin of segments is coded as (0 - haplotype1 ,  1 - haplotype2 )</h2><span id='topic+generateChromosomeRecombinationPositions'></span>

<h3>Description</h3>

<p>Generates a recombination vector arising from one meiotic event.
The origin of segments is coded as (0 - haplotype1 ,  1 - haplotype2 )
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateChromosomeRecombinationPositions(chromosomeLength = 500)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateChromosomeRecombinationPositions_+3A_chromosomelength">chromosomeLength</code></td>
<td>
<p>The length of the region in cm.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

# generate a recombination events for chromosome 4
readGeneticMap(4)
generateChromosomeRecombinationPositions(500)

</code></pre>

<hr>
<h2 id='generateFakeWholeGenomeGeneticMap'>Generates a fake genetic map that spans the whole genome.</h2><span id='topic+generateFakeWholeGenomeGeneticMap'></span>

<h3>Description</h3>

<p>Generates a fake genetic map that spans the whole genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateFakeWholeGenomeGeneticMap(vcf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateFakeWholeGenomeGeneticMap_+3A_vcf">vcf</code></td>
<td>
<p>A vcf file read by function readVCF.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = sprintf("%s/region.vcf.gz", examples_dir)
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)

# For realistic data use the function
# downloadGeneticMap
generateFakeWholeGenomeGeneticMap(vcf)

pdf(file=tempfile())
plotRegionalGeneticMap(seq(1e6,100e6,by=1e6/2))
dev.off()

</code></pre>

<hr>
<h2 id='generateRecombinationDistances'>Generate inter-recombination distances using a chi-square model. Note this are the distances between two succesive recombination events and not
the absolute positions of the events. To generate the locations of the recombination events see the example below.</h2><span id='topic+generateRecombinationDistances'></span>

<h3>Description</h3>

<p>Generate inter-recombination distances using a chi-square model. Note this are the distances between two succesive recombination events and not
the absolute positions of the events. To generate the locations of the recombination events see the example below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateRecombinationDistances(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateRecombinationDistances_+3A_n">n</code></td>
<td>
<p>Number of distances to generate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of distances between two recombination events.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

distances = generateRecombinationDistances(20)


positions_of_recombination = cumsum(distances)

if(0) hist(generateRecombinationDistances(20000),n=100)

</code></pre>

<hr>
<h2 id='generateRecombinationDistances_noInterference'>Generate recombination distances using a no-interference model.</h2><span id='topic+generateRecombinationDistances_noInterference'></span>

<h3>Description</h3>

<p>Generate recombination distances using a no-interference model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateRecombinationDistances_noInterference(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateRecombinationDistances_noInterference_+3A_n">n</code></td>
<td>
<p>Number of distances to generate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>recombination distances in centimorgan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")
mean ( generateRecombinationDistances_noInterference ( 200 ) )

</code></pre>

<hr>
<h2 id='generateSingleRecombinationVector'>Genetates a recombination vector arising from one meiotic event.
The origin of segments is coded as (0 - haplotype1 ,  1 - haplotype2 )</h2><span id='topic+generateSingleRecombinationVector'></span>

<h3>Description</h3>

<p>Genetates a recombination vector arising from one meiotic event.
The origin of segments is coded as (0 - haplotype1 ,  1 - haplotype2 )
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateSingleRecombinationVector(cm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateSingleRecombinationVector_+3A_cm">cm</code></td>
<td>
<p>The length of the region that we want to generate recombination distances.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)

# For realistic data use the function downloadGeneticMap
generateUniformGeneticMap()
generateSingleRecombinationVector( 1:100 )

</code></pre>

<hr>
<h2 id='generateUniformGeneticMap'>Generates a uniform genetic map.</h2><span id='topic+generateUniformGeneticMap'></span>

<h3>Description</h3>

<p>Generates a uniform genetic map by approximating 1 cm / Mbp. Only used for examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateUniformGeneticMap()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = sprintf("%s/region.vcf.gz", examples_dir)
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)

# For realistic data use the function readGeneticMap
generateUniformGeneticMap()

pdf(file=tempfile())
plotRegionalGeneticMap(seq(1e6,100e6,by=1e6/2))
dev.off()

</code></pre>

<hr>
<h2 id='generateUnrelatedIndividuals'>Generates variant data for n unrelated individuals</h2><span id='topic+generateUnrelatedIndividuals'></span>

<h3>Description</h3>

<p>Generates variant data for n unrelated individuals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateUnrelatedIndividuals(N = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateUnrelatedIndividuals_+3A_n">N</code></td>
<td>
<p>how many individuals to generate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>IDs of the generated individuals
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 , min_maf = 0.12)

genetic_map_of_region =
   system.file("examples",
     "chr4-geneticmap.txt",
     package = "sim1000G")

readGeneticMapFromFile(genetic_map_of_region)

startSimulation(vcf, totalNumberOfIndividuals = 1200)
ids = generateUnrelatedIndividuals(20)

# See also the documentation on our github page

</code></pre>

<hr>
<h2 id='geneticMap'>Holds the genetic map information that is used for simulations.</h2><span id='topic+geneticMap'></span>

<h3>Description</h3>

<p>Holds the genetic map information that is used for simulations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneticMap
</code></pre>


<h3>Format</h3>

<p>An object of class <code>environment</code> of length 0.</p>

<hr>
<h2 id='getCMfromBP'>Converts centimorgan position to base-pair. Return a list of centimorgan positions that correspond
to the bp vector (in basepairs).</h2><span id='topic+getCMfromBP'></span>

<h3>Description</h3>

<p>Converts centimorgan position to base-pair. Return a list of centimorgan positions that correspond
to the bp vector (in basepairs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCMfromBP(bp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCMfromBP_+3A_bp">bp</code></td>
<td>
<p>vector of base-pair positions</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = sprintf("%s/region.vcf.gz", examples_dir)
vcf = readVCF( vcf_file, maxNumberOfVariants = 100,
  min_maf = 0.12)

# For realistic data use the function downloadGeneticMap
generateUniformGeneticMap()
getCMfromBP(seq(1e6,100e6,by=1e6))


</code></pre>

<hr>
<h2 id='loadSimulation'>Load some previously saved simulation data by function saveSimulation</h2><span id='topic+loadSimulation'></span>

<h3>Description</h3>

<p>Load some previously saved simulation data by function saveSimulation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadSimulation(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadSimulation_+3A_id">id</code></td>
<td>
<p>Name the simulation to load which was previously saved by saveSimulation</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")

vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
           min_maf = 0.12 ,max_maf = NA)

# For a realistic genetic map
# use the function readGeneticMap
generateUniformGeneticMap()

startSimulation(vcf, totalNumberOfIndividuals = 200)

ped1 = newNuclearFamily(1)

saveSimulation("sim1")

vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.02 ,max_maf = 0.5)

startSimulation(vcf, totalNumberOfIndividuals = 200)
saveSimulation("sim2")

loadSimulation("sim1")



</code></pre>

<hr>
<h2 id='newFamily3generations'>Generates genotype data for a family of 3 generations</h2><span id='topic+newFamily3generations'></span>

<h3>Description</h3>

<p>Generates genotype data for a family of 3 generations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newFamily3generations(familyid, noffspring2 = 2, noffspring3 = c(1, 1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="newFamily3generations_+3A_familyid">familyid</code></td>
<td>
<p>What will be the family_id (for example: 100)</p>
</td></tr>
<tr><td><code id="newFamily3generations_+3A_noffspring2">noffspring2</code></td>
<td>
<p>Number of offspring in generation 2</p>
</td></tr>
<tr><td><code id="newFamily3generations_+3A_noffspring3">noffspring3</code></td>
<td>
<p>Number of offspring in generation 3 (vector of length noffspring2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>family structure object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)

generateUniformGeneticMap()

startSimulation(vcf, totalNumberOfIndividuals = 200)

ped_line = newFamily3generations(12, 3, c(3,3,2) )
</code></pre>

<hr>
<h2 id='newFamilyWithOffspring'>Simulates genotypes for 1 family with n offspring</h2><span id='topic+newFamilyWithOffspring'></span>

<h3>Description</h3>

<p>Simulates genotypes for 1 family with n offspring
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newFamilyWithOffspring(family_id, noffspring = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="newFamilyWithOffspring_+3A_family_id">family_id</code></td>
<td>
<p>What will be the family_id (for example: 100)</p>
</td></tr>
<tr><td><code id="newFamilyWithOffspring_+3A_noffspring">noffspring</code></td>
<td>
<p>Number of offsprings that this family will have</p>
</td></tr>
</table>


<h3>Value</h3>

<p>family structure object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ped_line = newFamilyWithOffspring(10,3)


</code></pre>

<hr>
<h2 id='newNuclearFamily'>Simulates genotypes for 1 family with 1 offspring</h2><span id='topic+newNuclearFamily'></span>

<h3>Description</h3>

<p>Simulates genotypes for 1 family with 1 offspring
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newNuclearFamily(family_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="newNuclearFamily_+3A_family_id">family_id</code></td>
<td>
<p>What will be the family_id (for example: 100)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>family structure object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
   min_maf = 0.12 ,max_maf = NA)

genetic_map_of_region = system.file("examples",
   "chr4-geneticmap.txt",
   package = "sim1000G")
readGeneticMapFromFile(genetic_map_of_region)

startSimulation(vcf, totalNumberOfIndividuals = 1200)
fam1 = newNuclearFamily(1)
fam2 = newNuclearFamily(2)

# See also the documentation on our github page

</code></pre>

<hr>
<h2 id='pkg.opts'>Holds general package options</h2><span id='topic+pkg.opts'></span>

<h3>Description</h3>

<p>Holds general package options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pkg.opts
</code></pre>


<h3>Format</h3>

<p>An object of class <code>environment</code> of length 1.</p>

<hr>
<h2 id='plotRegionalGeneticMap'>Generates a plot of the genetic map for a specified region.</h2><span id='topic+plotRegionalGeneticMap'></span>

<h3>Description</h3>

<p>The plot shows the centimorgan vs base-pair positions.
The position of markers that have been read is also depicted as vertical lines
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRegionalGeneticMap(bp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRegionalGeneticMap_+3A_bp">bp</code></td>
<td>
<p>Vector of base-pair positions to generate a plot for
library(&quot;sim1000G&quot;)
</p>
<p>examples_dir = system.file(&quot;examples&quot;, package = &quot;sim1000G&quot;)
vcf_file = sprintf(&quot;
vcf = readVCF( vcf_file, maxNumberOfVariants = 100,
min_maf = 0.12)
</p>
<p># For realistic data use the function readGeneticMap
generateUniformGeneticMap()
</p>
<p>pdf(file=tempfile())
plotRegionalGeneticMap(seq(1e6,100e6,by=1e6/2))
dev.off()</p>
</td></tr>
</table>

<hr>
<h2 id='printMatrix'>Utility function that prints a matrix. Useful for IBD12 matrices.</h2><span id='topic+printMatrix'></span>

<h3>Description</h3>

<p>Utility function that prints a matrix. Useful for IBD12 matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printMatrix(m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printMatrix_+3A_m">m</code></td>
<td>
<p>Matrix to be printed</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
printMatrix (  matrix(runif(16), nrow=4) )
</code></pre>

<hr>
<h2 id='readGeneticMap'>Reads a genetic map downloaded from the function downloadGeneticMap or reads a genetic map from a specified file. If the argument filename is used
then the genetic map is read from the corresponding file. Otherwise, if a chromosome is specified, the genetic map is downloaded for human chromosome
using grch37 coordinates.</h2><span id='topic+readGeneticMap'></span>

<h3>Description</h3>

<p>The map must contains a complete chromosome or enough markers to cover the area that
will be simulated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readGeneticMap(chromosome, filename = NA, dir = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readGeneticMap_+3A_chromosome">chromosome</code></td>
<td>
<p>Chromosome number to download a genetic map for , or</p>
</td></tr>
<tr><td><code id="readGeneticMap_+3A_filename">filename</code></td>
<td>
<p>A filename of an existing genetic map to read from (default NA).</p>
</td></tr>
<tr><td><code id="readGeneticMap_+3A_dir">dir</code></td>
<td>
<p>Directory the map file will be saved (only if chromosome is specified).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>



readGeneticMap(chromosome = 22)



</code></pre>

<hr>
<h2 id='readGeneticMapFromFile'>Reads a genetic map to be used for simulations. The genetic map should be
of a single chromosome and covering the extent of the region to be simulated.
Whole chromosome genetic maps can also be used.</h2><span id='topic+readGeneticMapFromFile'></span>

<h3>Description</h3>

<p>The file must be contain the following columns in the same order: chromosome, basepaire, rate(not used), centimorgan
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readGeneticMapFromFile(filelocation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readGeneticMapFromFile_+3A_filelocation">filelocation</code></td>
<td>
<p>Filename containing the genetic map</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

fname = downloadGeneticMap(10)

cat("genetic map downloaded at :", fname, "\n")
readGeneticMapFromFile(fname)


## End(Not run)
</code></pre>

<hr>
<h2 id='readVCF'>Read a vcf file, with options to filter out low or high frequency markers.</h2><span id='topic+readVCF'></span>

<h3>Description</h3>

<p>Read a vcf file, with options to filter out low or high frequency markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVCF(filename = "data.vcf", thin = NA, maxNumberOfVariants = 400,
  min_maf = 0.02, max_maf = NA, region_start = NA, region_end = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readVCF_+3A_filename">filename</code></td>
<td>
<p>Input VCF file</p>
</td></tr>
<tr><td><code id="readVCF_+3A_thin">thin</code></td>
<td>
<p>How much to thin markers</p>
</td></tr>
<tr><td><code id="readVCF_+3A_maxnumberofvariants">maxNumberOfVariants</code></td>
<td>
<p>Maximum number of variants to keep from region</p>
</td></tr>
<tr><td><code id="readVCF_+3A_min_maf">min_maf</code></td>
<td>
<p>Minimum allele frequency of markers to keep. If NA skip min_maf filtering.</p>
</td></tr>
<tr><td><code id="readVCF_+3A_max_maf">max_maf</code></td>
<td>
<p>Maximum allele frequency of markers to keep. If NA skip max_maf filtering.</p>
</td></tr>
<tr><td><code id="readVCF_+3A_region_start">region_start</code></td>
<td>
<p>Extract a region from a vcf files with this starting basepair position</p>
</td></tr>
<tr><td><code id="readVCF_+3A_region_end">region_end</code></td>
<td>
<p>Extract a region from a vcf files with this ending basepair position</p>
</td></tr>
</table>


<h3>Value</h3>

<p>VCF object to be used by startSimulation function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir,
  "region-chr4-93-TMEM156.vcf.gz")

vcf = readVCF( vcf_file, maxNumberOfVariants = 500 ,
               min_maf = 0.02 ,max_maf = NA)

str(as.list(vcf))
</code></pre>

<hr>
<h2 id='resetSimulation'>Removes all individuals that have been simulated and resets the simulator.</h2><span id='topic+resetSimulation'></span>

<h3>Description</h3>

<p>Removes all individuals that have been simulated and resets the simulator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resetSimulation()
</code></pre>


<h3>Value</h3>

<p>nothing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
resetSimulation()

</code></pre>

<hr>
<h2 id='retrieveGenotypes'>Retrieve a matrix of simulated genotypes for a specific set of individual IDs</h2><span id='topic+retrieveGenotypes'></span>

<h3>Description</h3>

<p>Retrieve a matrix of simulated genotypes for a specific set of individual IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retrieveGenotypes(ids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="retrieveGenotypes_+3A_ids">ids</code></td>
<td>
<p>Vector of ids of individuals to retrieve.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library("sim1000G")

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)

# For realistic data use the function downloadGeneticMap
generateUniformGeneticMap()

startSimulation(vcf, totalNumberOfIndividuals = 200)

ped1 = newNuclearFamily(1)

retrieveGenotypes(ped1$gtindex)

</code></pre>

<hr>
<h2 id='saveSimulation'>Save the data for a simulation for later use. When simulating multiple populations it
allows saving and restoring of simulation data for each population.</h2><span id='topic+saveSimulation'></span>

<h3>Description</h3>

<p>Save the data for a simulation for later use. When simulating multiple populations it
allows saving and restoring of simulation data for each population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saveSimulation(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saveSimulation_+3A_id">id</code></td>
<td>
<p>Name the simulation will be saved as.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>


examples_dir = system.file("examples", package = "sim1000G")

vcf_file = file.path(examples_dir, "region.vcf.gz")
vcf = readVCF( vcf_file, maxNumberOfVariants = 100 ,
               min_maf = 0.12 ,max_maf = NA)


# For realistic data use the functions downloadGeneticMap
generateUniformGeneticMap()

startSimulation(vcf, totalNumberOfIndividuals = 200)

ped1 = newNuclearFamily(1)

saveSimulation("sim1")

</code></pre>

<hr>
<h2 id='setRecombinationModel'>Set recombination model to either poisson (no interference) or chi-square.</h2><span id='topic+setRecombinationModel'></span>

<h3>Description</h3>

<p>Set recombination model to either poisson (no interference) or chi-square.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setRecombinationModel(model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setRecombinationModel_+3A_model">model</code></td>
<td>
<p>Either &quot;poisson&quot; or &quot;chisq&quot;</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>

generateUniformGeneticMap()

do_plots = 0

setRecombinationModel("chisq")
if(do_plots == 1)
 hist(generateRecombinationDistances(100000),n=200)

setRecombinationModel("poisson")
if(do_plots == 1)
 hist(generateRecombinationDistances(100000),n=200)

</code></pre>

<hr>
<h2 id='SIM'>Holds data necessary for a simulation.</h2><span id='topic+SIM'></span>

<h3>Description</h3>

<p>Holds data necessary for a simulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SIM
</code></pre>


<h3>Format</h3>

<p>An object of class <code>environment</code> of length 7.</p>

<hr>
<h2 id='sim1000G-package'>Simulations of rare/common variants using haplotype data from 1000 genomes</h2><span id='topic+sim1000G'></span><span id='topic+sim1000G-package'></span>

<h3>Description</h3>

<p>Documentation and examples can be found at the package directory folder inst / doc
or at our github url:
https://adimitromanolakis.github.io/sim1000G/
inst/doc/SimulatingFamilyData.html
</p>


<h3>Details</h3>

<p>See also our github repository page at:
https://github.com/adimitromanolakis/sim1000G
</p>

<hr>
<h2 id='startSimulation'>Starts and initializes the data structures required for a simulation. A VCF file
should be read beforehand with the function readVCF.</h2><span id='topic+startSimulation'></span>

<h3>Description</h3>

<p>Starts and initializes the data structures required for a simulation. A VCF file
should be read beforehand with the function readVCF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>startSimulation(vcf, totalNumberOfIndividuals = 2000, subset = NA,
  randomdata = 0, typeOfGeneticMap = "download")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="startSimulation_+3A_vcf">vcf</code></td>
<td>
<p>Input vcf file of a region (can be .gz). Must contain phased data.</p>
</td></tr>
<tr><td><code id="startSimulation_+3A_totalnumberofindividuals">totalNumberOfIndividuals</code></td>
<td>
<p>Maximum Number of individuals to allocate memory for. Set it above the number of individuals you want to simulate.</p>
</td></tr>
<tr><td><code id="startSimulation_+3A_subset">subset</code></td>
<td>
<p>A subset of individual IDs to use for simulation</p>
</td></tr>
<tr><td><code id="startSimulation_+3A_randomdata">randomdata</code></td>
<td>
<p>If 1, disregards the genotypes in the vcf file and generates independent markers that are not in LD.</p>
</td></tr>
<tr><td><code id="startSimulation_+3A_typeofgeneticmap">typeOfGeneticMap</code></td>
<td>
<p>Specify whether to download a genetic map for this chromosome</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library("sim1000G")
library(gplots)

examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir, "region.vcf.gz")

vcf = readVCF( vcf_file, maxNumberOfVariants = 100)


genetic_map_of_region = system.file(
   "examples",
   "chr4-geneticmap.txt",
   package = "sim1000G"
)

readGeneticMapFromFile(genetic_map_of_region)

pdf(file=tempfile())
plotRegionalGeneticMap(vcf$vcf[,2]+1)
dev.off()

startSimulation(vcf, totalNumberOfIndividuals = 200)

</code></pre>

<hr>
<h2 id='subsetVCF'>Generate a market subset of a vcf file</h2><span id='topic+subsetVCF'></span>

<h3>Description</h3>

<p>Generate a market subset of a vcf file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetVCF(vcf, var_index = NA, var_id = NA, individual_id = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetVCF_+3A_vcf">vcf</code></td>
<td>
<p>VCF data as created by function readVCF</p>
</td></tr>
<tr><td><code id="subsetVCF_+3A_var_index">var_index</code></td>
<td>
<p>index of number to subset. Should be in the range 1..length(vcf$varid)</p>
</td></tr>
<tr><td><code id="subsetVCF_+3A_var_id">var_id</code></td>
<td>
<p>id  of markers to subset. Should be a selection from vcf$varid. NA if no filtering on id to be performed.</p>
</td></tr>
<tr><td><code id="subsetVCF_+3A_individual_id">individual_id</code></td>
<td>
<p>IDs of individuals to subset. Should be a selection from vcf$individual_id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>VCF object to be used by startSimulation function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
examples_dir = system.file("examples", package = "sim1000G")

vcf_file = file.path(examples_dir, "region-chr4-93-TMEM156.vcf.gz")

vcf = readVCF( vcf_file, maxNumberOfVariants = 500 ,
               min_maf = 0.02 ,max_maf = NA)

vcf2 = subsetVCF(vcf, var_index = 1:50)

</code></pre>

<hr>
<h2 id='writePED'>Writes a plink compatible PED/MAP file from the simulated genotypes</h2><span id='topic+writePED'></span>

<h3>Description</h3>

<p>Writes a plink compatible PED/MAP file from the simulated genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writePED(vcf, fam, filename = "out")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writePED_+3A_vcf">vcf</code></td>
<td>
<p>vcf object used in simulation</p>
</td></tr>
<tr><td><code id="writePED_+3A_fam">fam</code></td>
<td>
<p>Individuals / families to be written</p>
</td></tr>
<tr><td><code id="writePED_+3A_filename">filename</code></td>
<td>
<p>Basename of output files (.ped/.map will be added automatically)</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
