<!DOCTYPE html><html><head><title>Help for package simboot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {simboot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#simboot-package'>
<p>Simultaneous inference for diversity indices.</p></a></li>
<li><a href='#asht'>
<p>Internal function for simultaneous asymptotic intervals</p></a></li>
<li><a href='#Bacteria'>
<p>Relative Abundances of Soil Bacteria</p></a></li>
<li><a href='#Boutrp'>
<p>Internal function</p></a></li>
<li><a href='#CCdrp'>
<p>Internal function</p></a></li>
<li><a href='#contrMat'><p> Contrast Matrices</p></a></li>
<li><a href='#corrmatgen'>
<p>Internal function.</p></a></li>
<li><a href='#estShannon'>
<p>Estimator for Shannon's index</p></a></li>
<li><a href='#estShannonf'>
<p>Estimator for Shannon's index odered by a factorial variable f.</p></a></li>
<li><a href='#estShannonWY'>
<p>Estimator for Shannon's index row wise.</p></a></li>
<li><a href='#estSimpson'>
<p>Estimator for Simpson's index</p></a></li>
<li><a href='#estSimpsonf'>
<p>Estimator for Simpson's index odered by a factorial variable f.</p></a></li>
<li><a href='#estThetaRow'>
<p>Internal function</p></a></li>
<li><a href='#mcpHill'><p> Multiplicity-adjusted p-values for comparing biodiversity via</p>
simultaneous inference of a user-defined selection of diversity
indices</a></li>
<li><a href='#predatGM'>
<p>Abundance data of predatory insects</p></a></li>
<li><a href='#rpht'>
<p>Internal function for simultaneous bayesian bootstrap intervals</p></a></li>
<li><a href='#saproDipGM'>
<p>Abundance data of Diptera with saprophagous larvae</p></a></li>
<li><a href='#sbdiv'>
<p>Perform simultaneous confidence intervals or adjusted p&ndash;values</p>
for the Shannon and the Simpson index.</a></li>
<li><a href='#SCIrp'>
<p>Internal function</p></a></li>
<li><a href='#Simpson'>
<p>Internal function for Simpson estimator</p></a></li>
<li><a href='#tsht'>
<p>Internal function for simultaenous bootstrap intervals</p></a></li>
<li><a href='#waldci'>
<p>Internal function for Wald intervals</p></a></li>
<li><a href='#WYht'>
<p>Internal function for simultaneous bootstrap confidence intervals</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simultaneous Inference for Diversity Indices</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Ralph Scherer [cre, aut],
  Philip Pallmann [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ralph Scherer &lt;shearer.ra76@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides estimation of simultaneous bootstrap and asymptotic confidence intervals for diversity indices, namely the Shannon and the Simpson index. Several pre&ndash;specified multiple comparison types are available to choose. Further user&ndash;defined contrast matrices are applicable. In addition, simboot estimates adjusted as well as unadjusted p&ndash;values for two of the three proposed bootstrap methods. Further simboot allows for comparing biological diversities of two or more groups while simultaneously testing a user-defined selection of Hill numbers of orders q, which are considered as appropriate and useful indices for measuring diversity.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/shearer/simboot">https://github.com/shearer/simboot</a>,
<a href="http://shearer.github.io/simboot/">http://shearer.github.io/simboot/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/shearer/simboot/issues">https://github.com/shearer/simboot/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>boot, mvtnorm</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-09 13:22:07 UTC; rps</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-09 13:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='simboot-package'>
Simultaneous inference for diversity indices.
</h2><span id='topic+simboot-package'></span><span id='topic+simboot'></span>

<h3>Description</h3>

<p>Package <span class="pkg">simboot</span> provides estimation of simultaneous bootstrap and
asymptotic confidence intervals for diversity indices, namely the
Shannon and the Simpson index. Several pre-specified
multiple-comparison types are available. Further user-defined
contrast matrices are applicable. In addition, <span class="pkg">simboot</span> estimates
adjusted as well as unadjusted <code class="reqn">p</code>&ndash;values for two of the three
proposed bootstrap methods. Further simboot allows for comparing
biological diversities of two or more groups with simultaneously
testing a user-defined selection of Hill numbers of orders q, which
are considered appropriate and useful indices for measuring diversity.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
  Package: </td><td style="text-align: left;"> simboot</td>
</tr>
<tr>
 <td style="text-align: left;">
  Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
  Version: </td><td style="text-align: left;"> 0.2-8</td>
</tr>
<tr>
 <td style="text-align: left;">
  Date: </td><td style="text-align: left;"> 2024-02-09</td>
</tr>
<tr>
 <td style="text-align: left;">
  License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
  LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Ralph Scherer\
Philip Pallmann\
</p>


<h3>References</h3>

<p>Scherer, R. and Schaarschmidt, F. (2013) Simultaneous confidence intervals
for comparing biodiversity indices estimated from overdispersed count
data. <em>Biometrical Journal</em> 55, 246&ndash;263.
</p>
<p>Evaluation of the methods in <code><a href="#topic+sbdiv">sbdiv</a></code>
</p>
<p>Pallmann, P. et al. (2012) Assessing group differences in
biodiversity by simultaneously testing a user-defined selection of
diversity indices. <em>Molecular ecology resources</em> 12, 1068&ndash;??78.
</p>
<p>Evaluation of the methods in <code><a href="#topic+mcpHill">mcpHill</a></code>
</p>
<p>Westfall, P. H. and Young, S. S. (1993) Resampling-Based
Multiple Testing: Examples and Methods for <code class="reqn">p</code>&ndash;Value
Adjustment. New York: Wiley.
</p>
<p>Corresponding method <code><a href="#topic+sbdiv">sbdiv</a></code> with method <code><a href="#topic+WYht">WYht</a></code>
</p>
<p>Besag, J., Green, P. J., Higdon, D., Mengersen, K. (1995)
Bayesian computation and stochastic systems (with discussion)
. <em>Statistical Science</em>, 10, 3&ndash;66.
</p>
<p>Corresponding method <code><a href="#topic+sbdiv">sbdiv</a></code> with method <code><a href="#topic+rpht">rpht</a></code>
</p>
<p>Beran, R. (1988) Balanced simultaneous confidence
sets. <em>Journal of the American Statistical Association</em>, 83, 679&ndash;686.
</p>
<p>Corresponding method <code><a href="#topic+sbdiv">sbdiv</a></code> with method <code><a href="#topic+tsht">tsht</a></code>
</p>
<p>Fritsch, K. S., Hsu, J. C. (1999) Multiple comparison of
entropies with application to dinosaur biodiversity. <em>Biometrics</em>,
55, 4, 1300&ndash;1305.
</p>
<p>Rogers, J. A., Hsu, J. C. (2001) Multiple comparisons of
biodiversity. <em>Biometrical Journal</em>, 43, 5, 617&ndash;625.
</p>
<p>Corresponding method <code><a href="#topic+sbdiv">sbdiv</a></code> with method <code><a href="#topic+asht">asht</a></code>
</p>
<p>Jost, L. (2008) G(ST) and its relatives do not measure
differentiation. <em>Molecular Ecology</em>, 17, 4015-4026.
</p>
<p>Corresponding method <code><a href="#topic+mcpHill">mcpHill</a></code>
</p>

<hr>
<h2 id='asht'>
Internal function for simultaneous asymptotic intervals
</h2><span id='topic+asht'></span>

<h3>Description</h3>

<p>Internal function for simultaneous asymptotic intervals
</p>


<h3>Note</h3>

<p>Only internal function. Use function <code><a href="#topic+sbdiv">sbdiv</a></code> instead
</p>


<h3>References</h3>

<p>Fritsch, K. S., Hsu, J. C. (1999) Multiple comparison of
entropies with application to dinosaur biodiversity. <em>Biometrics</em>,
55, 4, 1300&ndash;1305.
</p>
<p>Rogers, J. A., Hsu, J. C. (2001) Multiple comparisons of
biodiversity. <em>Biometrical Journal</em>, 43, 5, 617&ndash;625.
</p>

<hr>
<h2 id='Bacteria'>
Relative Abundances of Soil Bacteria
</h2><span id='topic+Bacteria'></span>

<h3>Description</h3>

<p>Relative abundances of soil bacteria from 27 samples collected
in nine forest and 18 grassland sites in Germany.  The data set
includes abundances of 18 bacterial phyla (including three
candidate phyla) and five proteobacterial classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Bacteria)</code></pre>


<h3>Format</h3>

<p>A data frame with 27 observations on the following 24 variables.
</p>

<dl>
<dt><code>Land use type</code></dt><dd><p>a factor with levels <code>forest</code> <code>grassland</code></p>
</dd>
<dt><code>Acidobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Actinobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bacteroidetes</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Chloroflexi</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cyanobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Deinococcus-Thermus</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Fibrobacteres</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Firmicutes</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Fusobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Gemmatimonadetes</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Nitrospira</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>OP11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Planctomycetes</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Spirochaetes</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Tenericutes</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TM7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Verrucomicrobia</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>WS3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Alphaproteobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Betaproteobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Deltaproteobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Gammaproteobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Epsilonproteobacteria</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Relative abundances of 18 bacterial phyla (including three
candidate phyla) and five proteobacterial classes (alpha, beta,
gamma, delta and epsilon) from two ecological metagenomics
studies (Will et al. 2010, Nacke et al. 2011).  There are 27
observations altogether, nine of which stem from forest and 18
from grassland plots in Germany.
</p>
<p>One goal of these investigations was to unravel differences in
bacterial diversity and community composition between the land
use types forest and grassland. 
</p>
<p>The bacteria's relative abundances were determined by analyzing
the V2-V3 region of the 16S rRNA gene via pyrosequencing-based
DNA techniques.
</p>


<h3>Source</h3>

<p>Will, C., Thuermer, A., Wollherr, A., et al. (2010) Horizon-
specific bacterial community composition of German grassland
soils, as revealed by pyrosequencing-based analysis of 16S rRNA
genes. <em>Applied and Environmental Microbiology</em>, 76, 6751&ndash;6759.
</p>
<p>Nacke, H., Thuermer, A., Wollherr, A., et al. (2011) Pyrosequencing-
based assessment of bacterial community structure along different
management types in German forest and grassland soils. <em>PLoS One</em>, 
6, e17000.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Bacteria)
str(Bacteria)

### Assess whether there is a difference in biodiversity and
### community composition species richness (Shannon index,
### Simpson index) between grassland and forest. 
### Bootstrap times set to 50 due to example time settings

library(simboot)
mcpHill(dataf=Bacteria[,2:24], fact=Bacteria[,1], boots=50, qval=c(0,1,2))
</code></pre>

<hr>
<h2 id='Boutrp'>
Internal function
</h2><span id='topic+Boutrp'></span>

<h3>Description</h3>

<p>Internal function for method <code><a href="#topic+rpht">rpht</a></code> in function <code><a href="#topic+sbdiv">sbdiv</a></code>
</p>


<h3>Note</h3>

<p>Only for internal use.
</p>

<hr>
<h2 id='CCdrp'>
Internal function
</h2><span id='topic+CCdrp'></span>

<h3>Description</h3>

<p>Internal function for method rpht in <code><a href="#topic+sbdiv">sbdiv</a></code>
</p>

<hr>
<h2 id='contrMat'> Contrast Matrices </h2><span id='topic+contrMat'></span>

<h3>Description</h3>

<p>Computes contrast matrices for several multiple comparison procedures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contrMat(n, type = c("Dunnett", "Tukey", "Sequen", "AVE", 
                     "Changepoint", "Williams", "Marcus", 
                     "McDermott", "UmbrellaWilliams", "GrandMean"), 
         base = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contrMat_+3A_n">n</code></td>
<td>
<p> a (possibly named) vector of sample sizes for each group.</p>
</td></tr>
<tr><td><code id="contrMat_+3A_type">type</code></td>
<td>
<p> type of contrast. </p>
</td></tr>
<tr><td><code id="contrMat_+3A_base">base</code></td>
<td>
<p> an integer specifying which group is considered the baseline
group for Dunnett contrasts.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Computes the requested matrix of contrasts for comparisons of mean levels.
</p>


<h3>Value</h3>

<p>The matrix of contrasts with appropriate row names is returned. 
</p>


<h3>Note</h3>

<p>Function <code>contrMat</code> is adapted from package <span class="pkg">multcomp</span>
</p>


<h3>References</h3>

<p>Frank Bretz, Alan Genz and Ludwig A. Hothorn (2001), On the numerical 
availability of multiple comparison procedures. <em>Biometrical Journal</em>,
<b>43</b>(5), 645&ndash;656.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> n &lt;- c(10,20,30,40)
 names(n) &lt;- paste("group", 1:4, sep="")
 contrMat(n)	# Dunnett is default
 contrMat(n, base = 2)	# use second level as baseline
 contrMat(n, type = "Tukey")
 contrMat(n, type = "Sequen")
 contrMat(n, type = "AVE")
 contrMat(n, type = "Changepoint")
 contrMat(n, type = "Williams")
 contrMat(n, type = "Marcus")
 contrMat(n, type = "McDermott")
 ### Umbrella-protected Williams contrasts, i.e. a sequence of 
 ### Williams-type contrasts with groups of higher order 
 ### stepwise omitted
 contrMat(n, type = "UmbrellaWilliams")
 ### comparison of each group with grand mean of all groups
 contrMat(n, type = "GrandMean")

</code></pre>

<hr>
<h2 id='corrmatgen'>
Internal function. 
</h2><span id='topic+corrmatgen'></span>

<h3>Description</h3>

<p>Correlation matrix for confidence intervals assuming multivariate
standard normal distribution. Calculates the correlation matrix for method
asci in function <code><a href="#topic+sbdiv">sbdiv</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corrmatgen(CM, varp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corrmatgen_+3A_cm">CM</code></td>
<td>
<p> a matrix of contrast coefficients, dimension MxI, where M=number of contrasts, and I=number of groups in a oneway layout </p>
</td></tr>
<tr><td><code id="corrmatgen_+3A_varp">varp</code></td>
<td>
<p> a numeric vector of groupwise variance estimates (length = I) </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of dimension MxM.
</p>

<hr>
<h2 id='estShannon'>
Estimator for Shannon's index
</h2><span id='topic+estShannon'></span>

<h3>Description</h3>

<p>Estimation function for Shannon's index. Internal use in
<code><a href="#topic+estShannonf">estShannonf</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estShannon(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estShannon_+3A_x">x</code></td>
<td>

<p>Vector of discrete-scaled numerical values.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimator of Shannon-Wiener index with bias correction. Number of
Species S in the bias correction does not take zeros into account.
</p>


<h3>Value</h3>

<p>Shannon-Wiener index with bias correction
</p>

<hr>
<h2 id='estShannonf'>
Estimator for Shannon's index odered by a factorial variable f.
</h2><span id='topic+estShannonf'></span>

<h3>Description</h3>

<p>Estimation function for Shannon's index. Internal use in
<code><a href="#topic+sbdiv">sbdiv</a></code> for methods <code><a href="#topic+rpht">rpht</a></code>, <code><a href="#topic+tsht">tsht</a></code>,
<code><a href="#topic+asht">asht</a></code>. Sums up species counts in each columns for every
treatment group and estimates Shannon's index with bias correction on
the resulting vectors of summed up species counts.
</p>
<p style="text-align: center;"><code class="reqn">\widehat{HBC}_{i} = \hat{H}_{i} + (S_i -1)/(2N_{i\bullet}) -
    (1-\sum(1/\hat{p}_{i\bullet s}))/(12N_{i\bullet}^2) -
    \sum((1/\hat{p}_{i\bullet s})-(1/(\hat{p}_{i\bullet
	s}^2)))/(12N_{i\bullet}^3);</code>
</p>

<p><code class="reqn">i=1,...,k;s=1,...,S;p_{i \bullet
      s}=\frac{\sum_{j=1}^{n}x_{sj}}{N_{i\bullet}}</code>;
</p>
<p style="text-align: center;"><code class="reqn">\hat{H}_i=(-1)\sum_{s=1}^{S}(\hat{p}_{i \bullet s} log(\hat{p}_{i \bullet s}))</code>
</p>

<p><code class="reqn">N_{i\bullet}= \sum_{j=1}^{n}N_{ij}</code> Number of observed individuals in treatment <code class="reqn">i</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estShannonf(X, f)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estShannonf_+3A_x">X</code></td>
<td>

<p><code class="reqn">n</code> times <code class="reqn">p</code> matrix containing species in <code class="reqn">p</code> columns and replicates in <code class="reqn">n</code> rows.
</p>
</td></tr>
<tr><td><code id="estShannonf_+3A_f">f</code></td>
<td>

<p>Factor variable containing treatment groups. Must be of length: replicates
times treatment groups.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>estimate</code></td>
<td>
<p>Estimated Shannon-Wiener index for treatment groups</p>
</td></tr>
<tr><td><code>varest</code></td>
<td>
<p>Estimated variance of Shannon-Wiener index for
treatment groups</p>
</td></tr>
</table>

<hr>
<h2 id='estShannonWY'>
Estimator for Shannon's index row wise.
</h2><span id='topic+estShannonWY'></span>

<h3>Description</h3>

<p>Estimation function for Shannon's index. Internal use in
<code><a href="#topic+WYht">WYht</a></code>. Calculates Shannon-Wiener index with bias correction
</p>
<p style="text-align: center;"><code class="reqn">\widehat{HBC}_{ij} = \hat{H}_{ij} + (S_{ij} -1)/(2N_{ij}) -
    (1-\sum_{s=1}^{S}(1/\hat{p}_{ijs}))/(12N_{ij}^2) -
    \sum_{s=1}^{S}((1/\hat{p}_{ijs})-(1/(\hat{p}_{ijs}^2)))/(12N_{ij}^3);</code>
</p>

<p style="text-align: center;"><code class="reqn">\hat{H}_{ij}=(-1)\sum_{s=1}^{S}(\hat{p}_{ijs} log(\hat{p}_{ijs}))</code>
</p>

<p><code class="reqn">i=1,...,k;j=1,...,n;s=1,...,S;</code>
</p>
<p><code class="reqn">S_j = </code> Number of observed species in replicate <code class="reqn">j</code>;
</p>
<p><code class="reqn">N_j= </code> Number of observed individuals in replicate <code class="reqn">j</code>
</p>
<p>for every row in a <code class="reqn">n \times p</code> matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estShannonWY(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estShannonWY_+3A_x">x</code></td>
<td>

<p>Vector of <code class="reqn">p</code> numerical species counts.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Shannon-Wiener index with bias correction
</p>

<hr>
<h2 id='estSimpson'>
Estimator for Simpson's index
</h2><span id='topic+estSimpson'></span>

<h3>Description</h3>

<p>Estimation function for Simpson's index <code class="reqn">1-p^2 * n/(n-1)</code>. Internal use in
<code><a href="#topic+estSimpsonf">estSimpsonf</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estSimpson(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estSimpson_+3A_x">x</code></td>
<td>

<p>Vector of discrete-scaled numerical values.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimator of Simpson's index
</p>

<hr>
<h2 id='estSimpsonf'>
Estimator for Simpson's index odered by a factorial variable f.
</h2><span id='topic+estSimpsonf'></span>

<h3>Description</h3>

<p>Estimation function for Simpson's index. Internal use in
<code><a href="#topic+sbdiv">sbdiv</a></code> for methods <code><a href="#topic+rpht">rpht</a></code>, <code><a href="#topic+tsht">tsht</a></code>,
<code><a href="#topic+asht">asht</a></code>. Sums up species counts in each columns for every
treatment group and estimates Simpson's index on the resulting vectors
of summed up species counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estSimpsonf(X, f)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estSimpsonf_+3A_x">X</code></td>
<td>

<p><code class="reqn">n</code> times <code class="reqn">p</code> matrix containing species in <code class="reqn">p</code> columns
and replicates in <code class="reqn">n</code>rows.</p>
</td></tr>
<tr><td><code id="estSimpsonf_+3A_f">f</code></td>
<td>

<p>Factor variable comtaining treatment groups. Must be of length: replicates
times treatment groups.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>estimate</code></td>
<td>
<p>Estimated Simpson index for treatment groups</p>
</td></tr>
<tr><td><code>varest</code></td>
<td>
<p>Estimated variance of Simpson's index for
treatment groups</p>
</td></tr>
</table>

<hr>
<h2 id='estThetaRow'>
Internal function
</h2><span id='topic+estThetaRow'></span>

<h3>Description</h3>

<p>Internal function for method <code><a href="#topic+WYht">WYht</a></code> in function
<code><a href="#topic+sbdiv">sbdiv</a></code>. Calculates the specified diversity index for every
replicated sample in each treatment group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estThetaRow(X, f, theta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estThetaRow_+3A_x">X</code></td>
<td>

<p>Matrix with dimension <code class="reqn">n \times p</code>.
</p>
</td></tr>
<tr><td><code id="estThetaRow_+3A_f">f</code></td>
<td>

<p>Factorial variable containing treatment groups.
</p>
</td></tr>
<tr><td><code id="estThetaRow_+3A_theta">theta</code></td>
<td>

<p>Shannon or Simpson index
</p>
</td></tr>
</table>

<hr>
<h2 id='mcpHill'> Multiplicity-adjusted p-values for comparing biodiversity via
simultaneous inference of a user-defined selection of diversity
indices </h2><span id='topic+mcpHill'></span>

<h3>Description</h3>

<p> The function mcpHill allows for comparing
biological diversities of two or more groups. It simultaneously tests
a user-defined selection of Hill numbers of orders q, which are
considered appropriate and useful indices for measuring diversity
(Jost 2008). As an output mcpHill gives p-values adjusted for
multiplicity according to the method of Westfall &amp; Young (1993).  </p>


<h3>Usage</h3>

<pre><code class='language-R'> mcpHill(dataf, fact, align = FALSE, block, boots = 5000, udmat
  = FALSE, usermat, mattype = "Dunnett", dunbase = 1, qval = seq(-1, 3),
  opt = "two.sided") </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcpHill_+3A_dataf">dataf</code></td>
<td>
<p>	Data frame containing numerical values (e.g. species
counts or relative abundances).  Rows represent repeated
observations of the (two or more) groups, columns
represent taxonomic units (usually species, or phyla,
classes etc.).</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_fact">fact</code></td>
<td>
<p> Vector assigning (two or
more) factor levels to the observations, i.e. the groups
to be compared. The length of fact must equal the number
of rows in dataf.</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_align">align</code></td>
<td>
<p>	Logical
indicating whether a block alignment should be carried
out. If TRUE, the blocks must be specified as a vector
in block. Default is FALSE.</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_block">block</code></td>
<td>
<p>	Vector
assigning which block an observation belongs to. Only
required if align=TRUE.  The length of block must equal
the number of rows in dataf.</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_boots">boots</code></td>
<td>
<p>	Number
of bootstrap replications. Values lower than 999 are
rejected. Default is 5000.</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_udmat">udmat</code></td>
<td>
<p>	Logical
indicating whether used-defined contrasts are applied
for multiple testing. If TRUE, a contrast matrix has to
be specified via usermat. Default is FALSE, meaning that
the contrast matrix is specified by a catchword
(e.g. &quot;Tukey&quot;, &quot;Dunnett&quot; etc.).</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_usermat">usermat</code></td>
<td>

<p>Matrix specifying user-defined multiple testing
contrasts. Only required if udmat=TRUE.  The row sums in
the matrix must equal zero.</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_mattype">mattype</code></td>
<td>
<p>	Type of
contrast matrix for multiple comparisons of
groups. Hence only required for comparisons of more than
two groups. Can be specified by the catchwords used in
function <code><a href="#topic+contrMat">contrMat</a></code> (e.g. &quot;Dunnett&quot;,
&quot;Tukey&quot;, &quot;GrandMean&quot;, &quot;AVE&quot;, &quot;Williams&quot;, &quot;Changepoint&quot;
etc.). Default is &quot;Dunnett&quot;.</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_dunbase">dunbase</code></td>
<td>
<p>	Integer determining the factor group (in alphanumerical order)
to be considered the baseline or control and therefore
only needed for Dunnett-type multiple contrasts.
Default is 1.</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_qval">qval</code></td>
<td>
<p>	Vector containing the
requested selection of q-values in order to specify the
Hill numbers of orders q to be investigated. Default is
seq(-1,3).</p>
</td></tr>
<tr><td><code id="mcpHill_+3A_opt">opt</code></td>
<td>
<p>	&quot;greater&quot; performs an
upper-tailed test, &quot;less&quot; a lower-tailed test and
&quot;two.sided&quot; a two-tailed test. Default is &quot;two.sided&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> The output of mcpHill is a matrix containing the chosen
selection of Hill numbers (their orders q) in the first column. The
multiplicity-adjusted p-values for each hypothesis tested are in the
second column.  The names of the rows denote which groups are being
compared.  </p>


<h3>Author(s)</h3>

<p> Philip Pallmann </p>


<h3>References</h3>

<p><em>Pallmann, P. et al. (2012)</em> Assessing group differences in
biodiversity by simultaneously testing a user-defined selection of
diversity indices. Molecular ecology resources 12, 1068&ndash;78.
</p>
<p><em>Jost, L. (2008)</em> G(ST) and its relatives do not measure
differentiation. Molecular Ecology, 17, 4015&ndash;4026.
</p>
<p><em>Westfall, P.H. and Young S.S. (1993)</em> Resampling-based multiple
testing: examples and methods for p-value adjustment. New York: Wiley.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Multiple testing with user-defined contrasts after block alignment

data(predatGM)

mymat &lt;- rbind( "GM - S1" = c(1,-1,0,0), "GM - S2" = c(1,0,-1,0), "GM -
  S3" = c(1,0,0,-1), "S1 - S2" = c(0,1,-1,0), "S1 - S3" = c(0,1,0,-1) )

# example runs with only 100 bootstrap steps. For estimation use 2000 or more.
mcpHill(dataf=predatGM[,3:35], fact=predatGM[,2], align=TRUE,
block=predatGM[,1], boots=100, udmat=TRUE, usermat=mymat, qval=seq(-1,
3, by=0.5))

# with Dunnett-type contrast matrix
mcpHill(dataf=predatGM[,3:35], fact=predatGM[,2], align=TRUE,
block=predatGM[,1], boots=100, udmat=FALSE, mattype = "Dunnett", qval=seq(-1,
3, by=0.5))

</code></pre>

<hr>
<h2 id='predatGM'>
Abundance data of predatory insects
</h2><span id='topic+predatGM'></span>

<h3>Description</h3>

<p>In a field trial with 8 complete blocks, one genetically modified crop variety and three varieties without genetical modification (S1, S2, S3) have been cultivated. 
Note that S1 is genetically closely related to the GM variety, and mainly differs from GM by not containing the transformation, while S2 and S3 are conventional varieties, which are genetically not closely related to GM and S1. In each of the 24 plots, a certain taxonomic group of predatory insects has been trapped. Trapped individuals have been classified to the species level. A total of 33 different species has been observed. For each plot, the summed counts of each species over one cultivation period is given in the variables Sp1, Sp2,...Sp33.
Among others, one question in research was: Does the genetic modified variety effect biodiversity of the (ecologically important, non-target) species?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(predatGM)</code></pre>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 35 variables.
</p>

<dl>
<dt><code>Block</code></dt><dd><p>a numeric vector, values 1,...,8 indicate the blocks of the trial </p>
</dd>
<dt><code>Variety</code></dt><dd><p>a factor distinguishing the four varieties in the field trial, with levels <code>GM</code> (the genetically modified variety), <code>S1</code> (the near-isogenic, conventional variety), <code>S2</code> and <code>S3</code> (further conventional varieties)</p>
</dd>
<dt><code>Sp1</code></dt><dd><p>a numeric vector, observed counts of species 1</p>
</dd>
<dt><code>Sp2</code></dt><dd><p>a numeric vector, ...</p>
</dd>
<dt><code>Sp3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp32</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sp33</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data set provided by Kai U. Priesnitz, Bavarian State Research Center for Agriculture, Institute for Plant
Protection, Freising, Germany.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(predatGM)

str(predatGM)

# Display data as a mosaicplot

# load("D:/Mueller/Biodiv/data/predatGM.rda")

# Matrix of counts with appropriate names
COUNTS&lt;-as.matrix(predatGM[,3:35])
SPECNAM&lt;-names(predatGM)[3:35]
colnames(COUNTS)&lt;-SPECNAM
rownames(COUNTS)&lt;-predatGM[,"Variety"]

# Assign colors and order by decreasing total abundance
COL&lt;-grey(c(0,2,4,6,8,1,3,5,7)/8)
DMO&lt;-COUNTS[,order(colSums(COUNTS), decreasing=TRUE)]
colnames(DMO)[15:33]&lt;-"."

# Mosaicplot
par(mar=c(4,2,1,1))
mosaicplot(DMO, col=COL, las=2, off=15, main="", cex=1.1)
mtext("A", side=3, line=-1.5, adj=0, cex=2)


</code></pre>

<hr>
<h2 id='rpht'>
Internal function for simultaneous bayesian bootstrap intervals
</h2><span id='topic+rpht'></span>

<h3>Description</h3>

<p>Internal function for simultaneous bayesian bootstrap intervals
</p>


<h3>Note</h3>

<p>Only internal function. Use function <code><a href="#topic+sbdiv">sbdiv</a></code> instead
</p>


<h3>References</h3>

<p>Besag, J., Green, P. J., Higdon, D., Mengersen, K. (1995)
Bayesian computation and stochastic systems (with discussion)
. <em>Statistical Science</em>, 10, 3&ndash;66.
</p>

<hr>
<h2 id='saproDipGM'>
Abundance data of Diptera with saprophagous larvae
</h2><span id='topic+saproDipGM'></span>

<h3>Description</h3>

<p>In a field trial with 6 complete blocks, three treatments have been applied: a genetically modified crop variety was cultivated without insecticide treatment (GM), its near-isogenic counterpart (i.e. not genetically modified but otherwise genetically closely related to the GM crop) has been cultivated without insecticide treatment (Iso), and the near-isogenic variety has been cultivated with insecticide treatment (Ins).
In each of the 18 plots, two emergence traps have been placed and Diptera with saprophagous larvae were classified to the species level and counted.
A total number of 25 different species has been observed and included in the present data set. For each plot, the summed counts of each species over one cultivation period (in 2002) and the two traps is given in the columns Acor, ..., Tnud.
Among others, one question in this trial was: Does the genetic modified variety effect biodiversity of the (ecologically important, non-target) species in comparison to the isogenic variety (as a negative control) and in comparison to the insecticide treated plants (as a positive control)?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(saproDipGM)</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations on the following 27 variables.
</p>

<dl>
<dt><code>Block</code></dt><dd><p>a numeric vector, values 1,...,6 indicate the blocks of the trial</p>
</dd>
<dt><code>Variety</code></dt><dd><p>a factor, distinguishing the 3 treatment levels: <code>GM</code> (genetically modified, no insecticide), <code>Ins</code> (not genetically modified, insecticide treatment) , and <code>Iso</code> (not genetically modified, no insecticide)</p>
</dd>
<dt><code>Acor</code></dt><dd><p>a numeric vector of counts of the first species</p>
</dd>
<dt><code>Arub</code></dt><dd><p>a numeric vector...</p>
</dd>
<dt><code>Aaph</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bbre</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Btri</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Burt</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bvag</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bill</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ccru</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cmir</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cvag</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dnit</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dand</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Lcin</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Lcas</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Malt</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Moli</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Mluc</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Mtox</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ppha</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sato</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Spal</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sate</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sleu</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Tnud</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data set provided by Dr. Sabine Prescher,Institute for Biosafety of Genetically Modified Plants, Julius-Kuehn-Institut, Braunschweig, Germany
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(saproDipGM)

str(saproDipGM)

# load("D:/Mueller/Biodiv/data/saproDipGM.rda")

# Display data as a mosaicplot

# Matrix of counts with appropriate names
COUNTS&lt;-as.matrix(saproDipGM[,3:27])
SPECNAM&lt;-names(saproDipGM)[3:27]
colnames(COUNTS)&lt;-SPECNAM
rownames(COUNTS)&lt;-saproDipGM[,"Variety"]

# Assign colors and order by decreasing total abundance
COL&lt;-grey(c(0,2,4,6,8,1,3,5,7)/8)
DMO&lt;-COUNTS[,order(colSums(COUNTS), decreasing=TRUE)]

# Mosaicplot
par(mar=c(4,2,1,1))
mosaicplot(DMO, col=COL, las=2, off=15, main="", cex=1.1)
mtext("A", side=3, line=-1.5, adj=0, cex=2)

</code></pre>

<hr>
<h2 id='sbdiv'>
Perform simultaneous confidence intervals or adjusted p&ndash;values
for the Shannon and the Simpson index.
</h2><span id='topic+sbdiv'></span>

<h3>Description</h3>

<p>Function <code>sbdiv</code> estimates simultaneous confidence intervals for the
Shannon or the Simpson index. This function provides calculation of
several pre&ndash;defined contrasts for confidence intervals.Further
self-defined contrast are applicable. Simultaneous resampling confidence
intervals are estimated according to the Algorithm of Besag et
al. (1995) using method <code><a href="#topic+rpht">rpht</a></code>, Westfall et al. (1993) using
method <code><a href="#topic+WYht">WYht</a></code> or similar to Beran (1988) using method
<code><a href="#topic+tsht">tsht</a></code>. Further estimation of simultaneous asymptotic
intervals adjusting for heterogeneous variances is provided by method
<code><a href="#topic+asht">asht</a></code> according to Fritsch and Hsu (1999) and Rogers and
Hsu (2001). However, estimation of asymptotic intervals may make
no sense in data sets with replicated samples due to overdispersion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sbdiv(X, f, theta = c("Shannon", "Simpson"),
type = c("Dunnett", "Tukey", "Sequen", "AVE", 
                     "Changepoint", "Williams", "Marcus", 
                     "McDermott", "UmbrellaWilliams", "GrandMean"),
cmat = NULL, method = c("WYht", "tsht", "rpht", "asht"), conf.level =
0.95, alternative = c("two.sided", "less", "greater"), R = 2000, base =
1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sbdiv_+3A_x">X</code></td>
<td>

<p>Data frame containing numerical values for counts in columns. Every
column represents on species.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_f">f</code></td>
<td>

<p>Vector of factorial variables for treatment groups. Vector length
must be equal to the length of treatment groups multiplicated with
sample replications.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_theta">theta</code></td>
<td>

<p>Biodiversity index. Options are Shannon and Simpson index.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_type">type</code></td>
<td>

<p>Type of comparison. Options are Dunnett, Tukey, Sequen, AVE,
Changepoint, Williams, Marcus, McDermott, UmbrellaWilliams,
GrandMean intervals. We tested only Dunnett and Tukey contrasts in
simulations.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_cmat">cmat</code></td>
<td>

<p>Optional self-defined contrast matrix. In case of using this
argument, the type argument is not considered.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_method">method</code></td>
<td>

<p>Possible methods are simultaneous bootstrap confidence intervals:
<code><a href="#topic+WYht">WYht</a></code>, <code><a href="#topic+tsht">tsht</a></code>, <code><a href="#topic+rpht">rpht</a></code> and asymptotic
simultaneous confidence intervals: <code><a href="#topic+asht">asht</a></code>. Adjusted and
unadjusted <code class="reqn">p</code>&ndash;values are estimated with method
<code><a href="#topic+WYht">WYht</a></code> and method <code><a href="#topic+tsht">tsht</a></code>.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_conf.level">conf.level</code></td>
<td>

<p>Pre-defined overall confidence level. Default is 0.95, while
two-sided inference is estimated with <code class="reqn">(1-conf.level)/2</code> for
each side and one-sided inference is estimated with
<code class="reqn">1-conf.level</code> for the side of interest.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_alternative">alternative</code></td>
<td>

<p>Specified type of interval. Could be &quot;one-sided&quot; or &quot;two.sided&quot;.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_r">R</code></td>
<td>

<p>Number of bootstrap steps. Default is 2000, which is a good
compromise between accuracy and computing time
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_base">base</code></td>
<td>

<p>Control group. base = 1 uses the first group in alphabetical order.
</p>
</td></tr>
<tr><td><code id="sbdiv_+3A_...">...</code></td>
<td>

<p>Further optional arguments for the internal used function <code>boot</code> from
package <span class="pkg">boot</span>. Most importantly, the number of Bootstrap samples can
be chosen via the parameter <code>R</code> (default is <code>R=2000</code>); see
<code>?boot</code> for further options.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>sbdiv</code> is the main function for estimating the different
multiplicity adjusted confidence intervals. Different methods are
called from internal functions.
</p>


<h3>Value</h3>

<table>
<tr><td><code>conf.int</code></td>
<td>
<p>estimate: Estimated difference between groups. Estimators
differ between the methods due to calculation. lower: Lower bounds
of estimated intervals. upper: Upper bounds of estimated intervals.
</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>adj. p: multiplicity adjusted p-values. raw p:
unadjusted p-values</p>
</td></tr>
<tr><td><code>conf.level</code></td>
<td>
<p>Pre-specified confidence level</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p>Pre-specified alternative</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ralph Scherer
</p>


<h3>References</h3>

<p>Scherer, R. and Schaarschmidt, F. (2013) Simultaneous confidence intervals
for comparing biodiversity indices estimated from overdispersed count
data. <em>Biometrical Journal</em> 55, 246&ndash;263.
</p>
<p>Evaluation of the methods in <code><a href="#topic+sbdiv">sbdiv</a></code>
</p>
<p>Westfall, P. H. and Young, S. S. (1993) Resampling-Based
Multiple Testing: Examples and Methods for <code class="reqn">p</code>&ndash;Value
Adjustment. New York: Wiley.
</p>
<p>Corresponding method <code>sbdiv</code> with method <code><a href="#topic+WYht">WYht</a></code>
</p>
<p>Besag, J., Green, P. J., Higdon, D., Mengersen, K. (1995)
Bayesian computation and stochastic systems (with discussion)
. <em>Statistical Science</em>, 10, 3&ndash;66.
</p>
<p>Corresponding method <code>sbdiv</code> with method <code><a href="#topic+rpht">rpht</a></code>
</p>
<p>Beran, R. (1988) Balanced simultaneous confidence
sets. <em>Journal of the American Statistical Association</em>, 83, 679&ndash;686.
</p>
<p>Corresponding method <code>sbdiv</code> with method <code><a href="#topic+tsht">tsht</a></code>
</p>
<p>Fritsch, K. S., Hsu, J. C. (1999) Multiple comparison of
entropies with application to dinosaur biodiversity. <em>Biometrics</em>,
55, 4, 1300&ndash;1305.
</p>
<p>Rogers, J. A., Hsu, J. C. (2001) Multiple comparisons of
biodiversity. <em>Biometrical Journal</em>, 43, 5, 617&ndash;625.
</p>
<p>Corresponding method <code>sbdiv</code> with method <code><a href="#topic+asht">asht</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## For plots of the datasets see the help files for the data sets.

## First dataset
data(predatGM)

## structure of data
str(predatGM)

## remove block variable
datspec_1 &lt;- predatGM[, -1]
str(datspec_1)

## Order of factorial variable
datspec_1$Variety

## argument base = 1 uses GM as control group. Not directly executable
## due to intensive computing time
# sbdiv(X = datspec_1[, 2:length(datspec_1)], f = datspec_1[, 1], theta =
# "Shannon", type = "Dunnett", method = "WYht", conf.level = 0.95,
# alternative = "two.sided", R = 2000, base = 1)

## Directly executable but senseless value for boot steps R
sbdiv(X = datspec_1[, 2:length(datspec_1)], f = datspec_1[, 1], theta =
"Shannon", type = "Dunnett", method = "WYht", conf.level = 0.95,
alternative = "two.sided", R = 100, base = 1)


## Second dataset
data(saproDipGM)

## structure
str(saproDipGM)

## remove block variable
datspec_2 &lt;- saproDipGM[, -1]
str(datspec_2)

## Order of factor variable
datspec_2$Variety

## argument base = 2 uses Ins as control group. Not directly executable
## due to intensive computing time
# sbdiv(X = datspec_2[, 2:length(datspec_2)], f = datspec_2[, 1], theta =
# "Shannon", type = "Dunnett", method = "rpht", conf.level = 0.95,
# alternative = "two.sided", R = 2000, base = 2)

## Directly executable but senseless value for boot steps R
sbdiv(X = datspec_2[, 2:length(datspec_2)], f = datspec_2[, 1], theta =
"Shannon", type = "Dunnett", method = "rpht", conf.level = 0.95,
alternative = "two.sided", R = 100, base = 2)

</code></pre>

<hr>
<h2 id='SCIrp'>
Internal function 
</h2><span id='topic+SCIrp'></span>

<h3>Description</h3>

<p>Interval estimation in method rpci in function sbci
</p>


<h3>Note</h3>

<p>Internal function. Use <code><a href="#topic+sbdiv">sbdiv</a></code> instead.
</p>

<hr>
<h2 id='Simpson'>
Internal function for Simpson estimator
</h2><span id='topic+Simpson'></span>

<h3>Description</h3>

<p>Calculates Simpson's index on probability vector <code class="reqn">p</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Simpson(p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Simpson_+3A_p">p</code></td>
<td>

<p>Probability vector <code class="reqn">x_s/n</code>
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Simpson's index
</p>


<h3>Note</h3>

<p>Only for internal use
</p>

<hr>
<h2 id='tsht'>
Internal function for simultaenous bootstrap intervals
</h2><span id='topic+tsht'></span>

<h3>Description</h3>

<p>Internal function for simultaenous bootstrap intervals based on summed
up counts for every species.
</p>


<h3>Note</h3>

<p>Only internal function. Use function <code><a href="#topic+sbdiv">sbdiv</a></code> instead
</p>


<h3>References</h3>

<p>Beran, R. (1988) Balanced simultaneous confidence
sets. <em>Journal of the American Statistical Association</em>, 83, 679&ndash;686.
</p>

<hr>
<h2 id='waldci'>
Internal function for Wald intervals
</h2><span id='topic+waldci'></span>

<h3>Description</h3>

<p>Internal function for wald intervals in method <code><a href="#topic+asht">asht</a></code> in
function <code><a href="#topic+sbdiv">sbdiv</a></code>
</p>


<h3>Note</h3>

<p>Internal function. Use function <code><a href="#topic+sbdiv">sbdiv</a></code> instead.
</p>

<hr>
<h2 id='WYht'>
Internal function for simultaneous bootstrap confidence intervals
</h2><span id='topic+WYht'></span>

<h3>Description</h3>

<p>Internal function for simultaneous bootstrap confidence intervals
based on resampled residuals
</p>


<h3>Note</h3>

<p>Only internal function. Use function <code><a href="#topic+sbdiv">sbdiv</a></code> instead
</p>


<h3>References</h3>

<p>Westfall, P. H. and Young, S. S. (1993) Resampling-Based
Multiple Testing: Examples and Methods for <code class="reqn">p</code>&ndash;Value
Adjustment. New York: Wiley.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
