<!DOCTYPE html><html><head><title>Help for package simDNAmixtures</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {simDNAmixtures}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#allele_specific_stutter_model'><p>Stutter model where the expected stutter rate depends on the allele and locus</p></a></li>
<li><a href='#gamma_model'><p>Defines a gamma model for peak height variability</p></a></li>
<li><a href='#gf'><p>Stutters and size regressions for a GlobalFiler 3500 kit</p></a></li>
<li><a href='#global_stutter_model'><p>Global stutter model where the expected stutter rate is constant across alleles and loci</p></a></li>
<li><a href='#kits'><p>Properties such as loci, dye, sizes for most standard forensic kits</p></a></li>
<li><a href='#log_normal_model'><p>Defines a log normal model for peak height variability</p></a></li>
<li><a href='#read_allele_freqs'><p>Read allele frequencies in FSIgen format (.csv)</p></a></li>
<li><a href='#read_size_regression'><p>Reads a size regression file</p></a></li>
<li><a href='#read_stutter_exceptions'><p>Reads a stutter exceptions file with overrides for expected stutter ratios</p></a></li>
<li><a href='#read_stutter_regression'><p>Reads a stutter regression file</p></a></li>
<li><a href='#read_wide_table'><p>Read wide table (.txt) with Allele1, Allele2, ... columns as is</p></a></li>
<li><a href='#sample_contributor_genotypes'><p>Sample genotypes for mixture contributors according to allele frequencies</p></a></li>
<li><a href='#sample_gamma_model'><p>Sample gamma model(s) with parameters according to priors</p></a></li>
<li><a href='#sample_genotype'><p>Sample a genotype according to allele frequencies</p></a></li>
<li><a href='#sample_log_normal_model'><p>Sample log normal model(s) with parameters according to priors</p></a></li>
<li><a href='#sample_log_normal_stutter_variance'><p>Sample log normal stutter variance parameters according to priors</p></a></li>
<li><a href='#sample_LSAE'><p>Sample Locus Specific Amplification Efficiency (LSAE) according to prior</p></a></li>
<li><a href='#sample_mixture_from_genotypes'><p>Sample mixture profile with provided genotypes</p></a></li>
<li><a href='#sample_mixtures'><p>Sample mixtures with random genotypes and random parameters according to priors</p></a></li>
<li><a href='#sample_offspring'><p>Sample offspring from two parental genotypes</p></a></li>
<li><a href='#sample_pedigree_genotypes'><p>Sample genotypes for pedigree according to allele frequencies by allele dropping.</p></a></li>
<li><a href='#SMASH_to_wide_table'><p>Converts SMASH (SampleName, Marker, Allele, Size, Height) data to a wide table</p></a></li>
<li><a href='#stutter_type'><p>Defines a stutter type to be used in the allele specific stutter model.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Simulate Forensic DNA Mixtures</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Mixed DNA profiles can be sampled according to models for probabilistic genotyping. Peak height variability is modelled using a log normal distribution or a gamma distribution. Sample contributors may be related according to a pedigree.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, readr, pedtools</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-03 18:36:58 UTC; mkruijver</td>
</tr>
<tr>
<td>Author:</td>
<td>Maarten Kruijver <a href="https://orcid.org/0000-0002-6890-7632"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Maarten Kruijver &lt;maarten.kruijver@esr.cri.nz&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-04 12:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='allele_specific_stutter_model'>Stutter model where the expected stutter rate depends on the allele and locus</h2><span id='topic+allele_specific_stutter_model'></span>

<h3>Description</h3>

<p>Stutter model where the expected stutter rate depends on the allele and locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allele_specific_stutter_model(stutter_types, size_regression)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allele_specific_stutter_model_+3A_stutter_types">stutter_types</code></td>
<td>
<p>List. See <a href="#topic+stutter_type">stutter_type</a>.</p>
</td></tr>
<tr><td><code id="allele_specific_stutter_model_+3A_size_regression">size_regression</code></td>
<td>
<p>Function, see <a href="#topic+read_size_regression">read_size_regression</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When a pg_model is constructed (see <a href="#topic+gamma_model">gamma_model</a>), a stutter model can optionally be applied. The allele specific stutter model is commonly used with a log normal model. The expected stutter ratio for a parent allele at a locus is obtained from a linear regression of observed stutter ratios against allele length. For some loci or alleles the linear model may not be satisfactory. To override the expected stutter rates for specific alleles, a list of exceptions can be used. See <a href="#topic+stutter_type">stutter_type</a> for more detail.
</p>


<h3>Value</h3>

<p>Object of class <code>stutter_model</code> to be used by e.g. <a href="#topic+log_normal_model">log_normal_model</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+global_stutter_model">global_stutter_model</a> for a stutter model where the expected stutter ratio does not depend on the locus or parent allele.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># we will define an allele specific stutter model for back stutter only

# prepare stutter regression
filename_bs_regression &lt;- system.file("extdata",
"GlobalFiler_Stutter_3500.txt",package = "simDNAmixtures")
bs_regression &lt;- read_stutter_regression(filename_bs_regression)

# prepare exceptions, i.e. where does the regression not apply?
filename_bs_exceptions &lt;- system.file("extdata",
"GlobalFiler_Stutter_Exceptions_3500.csv",package = "simDNAmixtures")
bs_exceptions &lt;- read_stutter_exceptions(filename_bs_exceptions)

# prepare a stutter type
backstutter &lt;- stutter_type(name = "BackStutter", delta = -1,
                            stutter_regression = bs_regression,
                            stutter_exceptions = bs_exceptions)

# assign stutter model
size_regression &lt;- read_size_regression(system.file("extdata",
"GlobalFiler_SizeRegression.csv",package = "simDNAmixtures"))
bs_model &lt;- allele_specific_stutter_model(list(backstutter), size_regression)
bs_model
</code></pre>

<hr>
<h2 id='gamma_model'>Defines a gamma model for peak height variability</h2><span id='topic+gamma_model'></span>

<h3>Description</h3>

<p>Defines a gamma model for peak height variability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_model(
  mixture_proportions,
  mu,
  cv,
  degradation_beta = rep(1, length(mixture_proportions)),
  LSAE = stats::setNames(rep(1, length(model_settings$locus_names)),
    model_settings$locus_names),
  model_settings
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamma_model_+3A_mixture_proportions">mixture_proportions</code></td>
<td>
<p>Numeric vector with the mixture proportion for each contributor.</p>
</td></tr>
<tr><td><code id="gamma_model_+3A_mu">mu</code></td>
<td>
<p>Numeric. Expectation of a full heterozygote contributing allele peak height.</p>
</td></tr>
<tr><td><code id="gamma_model_+3A_cv">cv</code></td>
<td>
<p>Numeric. Coefficient of variation of a full heterozygote contributing allele peak height</p>
</td></tr>
<tr><td><code id="gamma_model_+3A_degradation_beta">degradation_beta</code></td>
<td>
<p>Numeric Vector of same length as mixture_proportions. Degradation slope parameters for each contributor. Defaults to 1 for each contributor (i.e. not degraded)</p>
</td></tr>
<tr><td><code id="gamma_model_+3A_lsae">LSAE</code></td>
<td>
<p>Numeric vector (named) with Locus Specific Amplification Efficiencies. See <a href="#topic+sample_LSAE">sample_LSAE</a>. Defaults to 1 for each locus.</p>
</td></tr>
<tr><td><code id="gamma_model_+3A_model_settings">model_settings</code></td>
<td>
<p>List. Possible parameters: </p>

<ul>
<li><p> locus_names. Character vector.
</p>
</li>
<li><p> detection_threshold. Numeric vector (named) with Detection Thresholds.
</p>
</li>
<li><p> size_regression. Function, see <a href="#topic+read_size_regression">read_size_regression</a>.
</p>
</li>
<li><p> stutter_model. Optionally a stutter_model object that gives expected stutter heights. See <a href="#topic+global_stutter_model">global_stutter_model</a>.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>Define a gamma model for peak height variability with the parametrisation as described by Bleka et al. The model may then be used to sample DNA profiles using the <a href="#topic+sample_mixture_from_genotypes">sample_mixture_from_genotypes</a> function. Alternatively, to sample many models and profiles in one go with parameters according to a specified distribution, the <a href="#topic+sample_mixtures">sample_mixtures</a> function can be used.
</p>


<h3>Value</h3>

<p>Object of class <code>pg_model</code>.
</p>


<h3>References</h3>

<p>Bleka, Ø., Storvik, G., &amp; Gill, P. (2016). EuroForMix: An open source software based on a continuous model to evaluate STR DNA profiles from a mixture of contributors with artefacts. Forensic Science International: Genetics, 21, 35-44. doi: <a href="https://doi.org/10.1016/j.fsigen.2015.11.008">10.1016/j.fsigen.2015.11.008</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+log_normal_model">log_normal_model</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read allele frequencies
freqs &lt;- read_allele_freqs(system.file("extdata","FBI_extended_Cauc.csv",
                                       package = "simDNAmixtures"))

data(gf)

# define the gamma model for peak heights
model &lt;- gamma_model(mixture_proportions = 1, mu = 1000.,
                    cv = 0.1, model_settings = gf$gamma_settings_no_stutter)

# sample a single source profile (1-person 'mixture')
u1 &lt;- sample_contributor_genotypes("U1", freqs, loci = gf$autosomal_markers)
sample &lt;- sample_mixture_from_genotypes(u1, model)

# peaks follow a gamma distribution with an expected height of
# 1,000 for heterozygous alleles; 2,000 for homozygotes
hist(sample$Height)

# the gamma distribution is more obvious if many samples are taken
many_samples &lt;- replicate(n = 1e2,
                          sample_mixture_from_genotypes(u1, model),
                          simplify = FALSE)

hist(sapply(many_samples, function(x) x$Height))
</code></pre>

<hr>
<h2 id='gf'>Stutters and size regressions for a GlobalFiler 3500 kit</h2><span id='topic+gf'></span>

<h3>Description</h3>

<p>A dataset containing default parameters and settings for a GlobalFiler 3500 kit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gf
</code></pre>


<h3>Format</h3>

<p>A list of:
</p>

<dl>
<dt>autosomal_markers</dt><dd><p>Names of autosomal markers in the GlobalFiler kit</p>
</dd>
<dt>repeat_length_by_marker</dt><dd><p>Named numeric with STR repeat length by locus name</p>
</dd>
<dt>size_regression</dt><dd><p>See <a href="#topic+read_size_regression">read_size_regression</a></p>
</dd>
<dt>stutters</dt><dd><p>List of 4 stutter types, to be used with <a href="#topic+allele_specific_stutter_model">allele_specific_stutter_model</a></p>
</dd>
<dt>stutter_model</dt><dd><p>For convenience, a pre-defined allele_specific_stutter_model</p>
</dd>
<dt>log_normal_settings</dt><dd><p>Settings corresponding to a log normal model with all stutter types</p>
</dd>
<dt>log_normal_settings_fwbw</dt><dd><p>Settings corresponding to a log normal model with backward and forward stutter only</p>
</dd>
<dt>gamma_settings</dt><dd><p>Settings corresponding to a gamma model with all stutter types</p>
</dd>
<dt>gamma_settings_no_stutter</dt><dd><p>Settings for a gamma model without stutter</p>
</dd>
</dl>


<hr>
<h2 id='global_stutter_model'>Global stutter model where the expected stutter rate is constant across alleles and loci</h2><span id='topic+global_stutter_model'></span>

<h3>Description</h3>

<p>Global stutter model where the expected stutter rate is constant across alleles and loci
</p>


<h3>Usage</h3>

<pre><code class='language-R'>global_stutter_model(back_stutter_rate, forward_stutter_rate, size_regression)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="global_stutter_model_+3A_back_stutter_rate">back_stutter_rate</code></td>
<td>
<p>Numeric. (Optional)</p>
</td></tr>
<tr><td><code id="global_stutter_model_+3A_forward_stutter_rate">forward_stutter_rate</code></td>
<td>
<p>Numeric. (Optional)</p>
</td></tr>
<tr><td><code id="global_stutter_model_+3A_size_regression">size_regression</code></td>
<td>
<p>Function, see <a href="#topic+read_size_regression">read_size_regression</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When a pg_model is constructed (see <a href="#topic+gamma_model">gamma_model</a>), a stutter model can optionally be applied. In the global stutter model, the expected stutter rate is constant across all loci and for all parent alleles.
</p>


<h3>Value</h3>

<p>Object of class <code>stutter_model</code> to be used by e.g. <a href="#topic+gamma_model">gamma_model</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+allele_specific_stutter_model">allele_specific_stutter_model</a> for a stutter model where the expected stutter rate depends on the allele and locus.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># the stutter model needs a size regression to determine fragment length
# of stutter products
size_regression &lt;- read_size_regression(system.file("extdata",
"GlobalFiler_SizeRegression.csv",package = "simDNAmixtures"))

# define a stutter model with an expected back stutter rate of 10%
stutter_model &lt;- global_stutter_model(back_stutter_rate = 0.1,
                                     size_regression = size_regression)

stutter_model
</code></pre>

<hr>
<h2 id='kits'>Properties such as loci, dye, sizes for most standard forensic kits</h2><span id='topic+kits'></span>

<h3>Description</h3>

<p>A dataset containing the properties of forensic DNA kits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kits
</code></pre>


<h3>Format</h3>

<p>A list of data frames containing variables such as:
</p>

<dl>
<dt>Panel</dt><dd><p>Kit name</p>
</dd>
<dt>Marker</dt><dd></dd>
<dt>Allele</dt><dd></dd>
<dt>Size</dt><dd><p>Fragment length</p>
</dd>
<dt>Color</dt><dd><p>Dye colour</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://github.com/oyvble/euroformix/blob/master/inst/extdata/kit.txt">https://github.com/oyvble/euroformix/blob/master/inst/extdata/kit.txt</a>
</p>

<hr>
<h2 id='log_normal_model'>Defines a log normal model for peak height variability</h2><span id='topic+log_normal_model'></span>

<h3>Description</h3>

<p>Defines a log normal model for peak height variability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log_normal_model(
  template,
  degradation = rep(0, length(template)),
  LSAE = stats::setNames(rep(1, length(model_settings$locus_names)),
    model_settings$locus_names),
  c2,
  k2,
  model_settings
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="log_normal_model_+3A_template">template</code></td>
<td>
<p>Numeric vector</p>
</td></tr>
<tr><td><code id="log_normal_model_+3A_degradation">degradation</code></td>
<td>
<p>Numeric vector of same length as template. Degradation parameters for each contributor.</p>
</td></tr>
<tr><td><code id="log_normal_model_+3A_lsae">LSAE</code></td>
<td>
<p>Numeric vector (named) with Locus Specific Amplification Efficiencies. See <a href="#topic+sample_LSAE">sample_LSAE</a>. Defaults to 1 for each locus.</p>
</td></tr>
<tr><td><code id="log_normal_model_+3A_c2">c2</code></td>
<td>
<p>Numeric. Allele variance parameter.</p>
</td></tr>
<tr><td><code id="log_normal_model_+3A_k2">k2</code></td>
<td>
<p>Optionally a numeric vector with stutter variance parameters. See <a href="#topic+sample_log_normal_stutter_variance">sample_log_normal_stutter_variance</a>.</p>
</td></tr>
<tr><td><code id="log_normal_model_+3A_model_settings">model_settings</code></td>
<td>
<p>List. Possible parameters: </p>

<ul>
<li><p> locus_names. Character vector.
</p>
</li>
<li><p> degradation_parameter_cap. Numeric.
</p>
</li>
<li><p> c2_prior. Numeric of length two with shape and scale.
</p>
</li>
<li><p> LSAE_variance_prior. Numeric of length one.
</p>
</li>
<li><p> detection_threshold. Numeric vector (named) with Detection Thresholds. Defaults to 50 for each locus.
</p>
</li>
<li><p> size_regression. Function, see <a href="#topic+read_size_regression">read_size_regression</a>.
</p>
</li>
<li><p> stutter_model. Optionally a stutter_model object that gives expected stutter heights. See <a href="#topic+global_stutter_model">global_stutter_model</a>.
</p>
</li>
<li><p> stutter_variability. Optionally peak height variability parameters for stutters. Required when stutter_model is supplied.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>Define a log normal model for peak height variability with the parametrisation as described by Bright et al. The model may then be used to sample DNA profiles using the <a href="#topic+sample_mixture_from_genotypes">sample_mixture_from_genotypes</a> function. Alternatively, to sample many models and profiles in one go with parameters according to a specified distribution, the <a href="#topic+sample_mixtures">sample_mixtures</a> function can be used.
</p>


<h3>Value</h3>

<p>Object of class <code>pg_model</code>.
</p>


<h3>References</h3>

<p>Bright, J.A. et al. (2016). Developmental validation of STRmix™, expert software for the interpretation of forensic DNA profiles. Forensic Science International: Genetics, 23, 226-239. doi: <a href="https://doi.org/10.1016/j.fsigen.2016.05.007">10.1016/j.fsigen.2016.05.007</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+gamma_model">gamma_model</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gf)
freqs &lt;- read_allele_freqs(system.file("extdata","FBI_extended_Cauc.csv",
                           package = "simDNAmixtures"))

k2 &lt;- sample_log_normal_stutter_variance(gf$log_normal_settings$stutter_variability)

model &lt;- log_normal_model(template = 1e3, c2 = 15, k2 = k2,
                          model_settings = gf$log_normal_settings)
model
</code></pre>

<hr>
<h2 id='read_allele_freqs'>Read allele frequencies in FSIgen format (.csv)</h2><span id='topic+read_allele_freqs'></span>

<h3>Description</h3>

<p>Read allele frequencies in FSIgen format (.csv)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_allele_freqs(filename, remove_zeroes = TRUE, normalise = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_allele_freqs_+3A_filename">filename</code></td>
<td>
<p>Path to csv file.</p>
</td></tr>
<tr><td><code id="read_allele_freqs_+3A_remove_zeroes">remove_zeroes</code></td>
<td>
<p>Logical. Should frequencies of 0 be removed from the return value? Default is TRUE.</p>
</td></tr>
<tr><td><code id="read_allele_freqs_+3A_normalise">normalise</code></td>
<td>
<p>Logical. Should frequencies be normalised to sum to 1? Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads allele frequencies from a .csv file. The file should be in FSIgen format, i.e.
comma separated with the first column specifying the allele labels and
one column per locus. The last row should be the number of observations.
No error checking is done since the file format is only loosely defined,
e.g. we do not restrict the first column name or the last row name.
</p>


<h3>Value</h3>

<p>Named list with frequencies by locus. The frequencies at a locus are returned as a named numeric vector with names corresponding to alleles.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># below we read an allele freqs file that comes with the package
filename &lt;- system.file("extdata","FBI_extended_Cauc.csv",package = "simDNAmixtures")
freqs &lt;- read_allele_freqs(filename)
freqs # the output is a list with an attribute named \code{N} giving the sample size.
</code></pre>

<hr>
<h2 id='read_size_regression'>Reads a size regression file</h2><span id='topic+read_size_regression'></span>

<h3>Description</h3>

<p>Reads a size regression file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_size_regression(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_size_regression_+3A_filename">filename</code></td>
<td>
<p>Path to file (character).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Read a regression file from disk and returns a function that provides the fragment length (bp) for a given locus and allele.
</p>
<p>DNA profiles consist of the observed peaks (alleles or stutter products) at several loci as well as the peak heights and sizes. The size refers to the fragment length (bp). A linear relationship exists between the size of a peak and the size. When peaks are sampled in the <a href="#topic+sample_mixture_from_genotypes">sample_mixture_from_genotypes</a> function, a size is assigned using a size regression. The <code>read_size_regression</code> function reads such a regression from disk.
</p>


<h3>Value</h3>

<p>A function that takes a locus name and allele as arguments and returns the size.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filename &lt;- system.file("extdata",
                        "GlobalFiler_SizeRegression.csv",
                        package = "simDNAmixtures")

regression &lt;- read_size_regression(filename)

# obtain size for the 12 allele at the vWA locus
regression("vWA", 12)
</code></pre>

<hr>
<h2 id='read_stutter_exceptions'>Reads a stutter exceptions file with overrides for expected stutter ratios</h2><span id='topic+read_stutter_exceptions'></span>

<h3>Description</h3>

<p>Reads a stutter exceptions file with overrides for expected stutter ratios
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_stutter_exceptions(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_stutter_exceptions_+3A_filename">filename</code></td>
<td>
<p>Character. Path to file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads the file from disk and returns a named numeric vector with stutter ratio exceptions for a given locus and allele.
</p>


<h3>Value</h3>

<p>A named list with the stutter exceptions by locus. For each loucs, the exceptions are given as a named numeric with the names corresponding to the parent alleles and the expected stutter rates given as the values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filename &lt;- system.file("extdata","GlobalFiler_Stutter_Exceptions_3500.csv",
                        package = "simDNAmixtures")
exceptions &lt;- read_stutter_exceptions(filename)
exceptions$TH01["9.3"]
</code></pre>

<hr>
<h2 id='read_stutter_regression'>Reads a stutter regression file</h2><span id='topic+read_stutter_regression'></span>

<h3>Description</h3>

<p>Reads a stutter regression file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_stutter_regression(filename, min_stutter_ratio = 0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_stutter_regression_+3A_filename">filename</code></td>
<td>
<p>Character. Path to file.</p>
</td></tr>
<tr><td><code id="read_stutter_regression_+3A_min_stutter_ratio">min_stutter_ratio</code></td>
<td>
<p>Numeric.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads the file from disk and returns a function that provides the expected stutter ratio for a given locus and allele.
</p>


<h3>Value</h3>

<p>A function that takes a locus name and allele as arguments and returns the expected stutter ratio.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filename &lt;- system.file("extdata","GlobalFiler_Stutter_3500.txt",
                        package = "simDNAmixtures")
regression &lt;- read_stutter_regression(filename)

# obtain the expected stutter ratio for a 12 parent allele at the vWA locus
regression("vWA", 12)
</code></pre>

<hr>
<h2 id='read_wide_table'>Read wide table (.txt) with Allele1, Allele2, ... columns as is</h2><span id='topic+read_wide_table'></span>

<h3>Description</h3>

<p>Read wide table (.txt) with Allele1, Allele2, ... columns as is
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_wide_table(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_wide_table_+3A_filename">filename</code></td>
<td>
<p>Path to txt file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe
</p>

<hr>
<h2 id='sample_contributor_genotypes'>Sample genotypes for mixture contributors according to allele frequencies</h2><span id='topic+sample_contributor_genotypes'></span>

<h3>Description</h3>

<p>Sample genotypes for mixture contributors according to allele frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_contributor_genotypes(
  contributors,
  freqs,
  pedigree,
  loci = names(freqs),
  return_non_contributors = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_contributor_genotypes_+3A_contributors">contributors</code></td>
<td>
<p>Character vector with unique names of contributors. Valid names are &quot;U1&quot;, &quot;U2&quot;, ... for unrelated contributors or the names of pedigree members for related contributors.</p>
</td></tr>
<tr><td><code id="sample_contributor_genotypes_+3A_freqs">freqs</code></td>
<td>
<p>Allele frequencies (see <a href="#topic+read_allele_freqs">read_allele_freqs</a>)</p>
</td></tr>
<tr><td><code id="sample_contributor_genotypes_+3A_pedigree">pedigree</code></td>
<td>
<p>(optionally) <a href="pedtools.html#topic+ped">ped</a> object</p>
</td></tr>
<tr><td><code id="sample_contributor_genotypes_+3A_loci">loci</code></td>
<td>
<p>Character vector of locus names (defaults to <code>names</code> attribute of <code>freqs</code>)</p>
</td></tr>
<tr><td><code id="sample_contributor_genotypes_+3A_return_non_contributors">return_non_contributors</code></td>
<td>
<p>Logical. Should genotypes of non-contributing pedigree members also be returned?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each founder or unrelated person, a genotype is sampled randomly by drawing two alleles from allele frequencies. The non-founders get genotypes by allele dropping, see <a href="#topic+sample_pedigree_genotypes">sample_pedigree_genotypes</a> for details.
</p>


<h3>Value</h3>

<p>List of DataFrames with genotypes for each pedigree member. See <a href="#topic+sample_genotype">sample_genotype</a> for the DataFrame format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read allele frequencies
freqs &lt;- read_allele_freqs(system.file("extdata","FBI_extended_Cauc.csv",
                           package = "simDNAmixtures"))

# define a pedigree of siblings S1 and S2 (and their parents)
ped_sibs &lt;- pedtools::nuclearPed(children = c("S1", "S2"))

# sample genotypes for a mixture of S1 + U1 + S2
# where U1 is an unrelated person

sample_contributor_genotypes(contributors = c("S1","U1","S2"), freqs, ped_sibs)
</code></pre>

<hr>
<h2 id='sample_gamma_model'>Sample gamma model(s) with parameters according to priors</h2><span id='topic+sample_gamma_model'></span>

<h3>Description</h3>

<p>Sample gamma model(s) with parameters according to priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_gamma_model(number_of_contributors, sampling_parameters, model_settings)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_gamma_model_+3A_number_of_contributors">number_of_contributors</code></td>
<td>
<p>Integer</p>
</td></tr>
<tr><td><code id="sample_gamma_model_+3A_sampling_parameters">sampling_parameters</code></td>
<td>
<p>List. Needs to contain:
</p>

<ul>
<li><p> min_mu. Numeric of length one.
</p>
</li>
<li><p> max_mu. Numeric of length one.
</p>
</li>
<li><p> min_cv. Numeric of length one.
</p>
</li>
<li><p> max_cv. Numeric of length one.
</p>
</li>
<li><p> degradation_shape1. Numeric of length one.
</p>
</li>
<li><p> degradation_shape2. Numeric of length one.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sample_gamma_model_+3A_model_settings">model_settings</code></td>
<td>
<p>List. See <a href="#topic+gamma_model">gamma_model</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In simulation studies involving many mixed DNA profiles, one often needs to generate various samples with different model parameters. This function samples a gamma model with parameters according to prior distributions. The mean peak height parameter <code>mu</code> is sampled uniformly between <code>min_mu</code> and <code>max_mu</code>. Likewise, the variability parameter <code>cv</code> is sampled uniformly between <code>min_cv</code> and <code>max_cv</code>. The degradation slope parameter <code>beta</code> is sampled according to a Beta distribution with parameters <code>degradation_shape1</code> and <code>degradation_shape2</code>.
</p>


<h3>Value</h3>

<p>When <code>length(number_of_contributors)==1</code>, a single <a href="#topic+gamma_model">gamma_model</a> of class <code>pg_model</code>. Otherwise, a list of these.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gf)

sampling_parameters &lt;- list(min_mu = 50., max_mu = 5e3,
                           min_cv = 0.05, max_cv = 0.35,
                           degradation_shape1 = 10, degradation_shape2 = 1)

model_no_stutter &lt;- sample_gamma_model(number_of_contributors = 2,
                                      sampling_parameters = sampling_parameters,
                                      model_settings = gf$gamma_settings_no_stutter)

model_no_stutter$parameters

</code></pre>

<hr>
<h2 id='sample_genotype'>Sample a genotype according to allele frequencies</h2><span id='topic+sample_genotype'></span>

<h3>Description</h3>

<p>Sample a genotype according to allele frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_genotype(freqs, loci = names(freqs), label = "U")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_genotype_+3A_freqs">freqs</code></td>
<td>
<p>Allele frequencies (see <a href="#topic+read_allele_freqs">read_allele_freqs</a>)</p>
</td></tr>
<tr><td><code id="sample_genotype_+3A_loci">loci</code></td>
<td>
<p>Character vector of locus names (defaults to <code>names</code> attribute of <code>freqs</code>)</p>
</td></tr>
<tr><td><code id="sample_genotype_+3A_label">label</code></td>
<td>
<p>Sample name</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A genotype is sampled randomly by drawing two alleles from allele frequencies for each locus.
</p>


<h3>Value</h3>

<p>DataFrame with columns <code>Sample Name</code>, <code>Locus</code>, <code>Allele1</code> and <code>Allele2</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># below we read an allele freqs and sample a genotype
filename &lt;- system.file("extdata","FBI_extended_Cauc.csv",
                        package = "simDNAmixtures")
freqs &lt;- read_allele_freqs(filename)
sample_genotype(freqs, loci = c("D3S1358", "vWA"))
</code></pre>

<hr>
<h2 id='sample_log_normal_model'>Sample log normal model(s) with parameters according to priors</h2><span id='topic+sample_log_normal_model'></span>

<h3>Description</h3>

<p>Sample log normal model(s) with parameters according to priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_log_normal_model(
  number_of_contributors,
  sampling_parameters,
  model_settings
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_log_normal_model_+3A_number_of_contributors">number_of_contributors</code></td>
<td>
<p>Integer</p>
</td></tr>
<tr><td><code id="sample_log_normal_model_+3A_sampling_parameters">sampling_parameters</code></td>
<td>
<p>List. Needs to contain:
</p>

<ul>
<li><p> min_template. Numeric of length one.
</p>
</li>
<li><p> max_template. Numeric of length one.
</p>
</li>
<li><p> degradation_shape. Numeric of length one.
</p>
</li>
<li><p> degradation_scale. Numeric of length one.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sample_log_normal_model_+3A_model_settings">model_settings</code></td>
<td>
<p>List. See <a href="#topic+log_normal_model">log_normal_model</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In simulation studies involving many mixed DNA profiles, one often needs to generate various samples with different model parameters. This function samples a log normal model with parameters according to prior distributions. The template parameter for each contributor is sampled uniformly between <code>min_template</code> and <code>max_template</code>. The degradation parameter for each contributor is sampled from a gamma distribution with parameters <code>degradation_shape</code> and <code>degradation_scale</code>.
</p>


<h3>Value</h3>

<p>When <code>length(number_of_contributors)==1</code>, a single <a href="#topic+log_normal_model">log_normal_model</a> of class <code>pg_model</code>. Otherwise, a list of these.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gf)

sampling_parameters &lt;- list(min_template = 50., max_template = 1000.,
                            degradation_shape = 2.5, degradation_scale = 1e-3)

model_no_stutter &lt;- sample_log_normal_model(number_of_contributors = 1,
                                            sampling_parameters = sampling_parameters,
                                            model_settings = gf$log_normal_settings)
</code></pre>

<hr>
<h2 id='sample_log_normal_stutter_variance'>Sample log normal stutter variance parameters according to priors</h2><span id='topic+sample_log_normal_stutter_variance'></span>

<h3>Description</h3>

<p>Sample log normal stutter variance parameters according to priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_log_normal_stutter_variance(log_normal_stutter_variability)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_log_normal_stutter_variance_+3A_log_normal_stutter_variability">log_normal_stutter_variability</code></td>
<td>
<p>List of variability parameters. See <a href="#topic+gf">gf</a> for an example.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named numeric with stutter variance parameter for all stutter types. Names are <code>k2</code> concatenated with the name of the stuter type. See example.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(gf)
 log_normal_stutter_variability &lt;- gf$log_normal_settings$stutter_variability
 k2 &lt;- sample_log_normal_stutter_variance(log_normal_stutter_variability)
</code></pre>

<hr>
<h2 id='sample_LSAE'>Sample Locus Specific Amplification Efficiency (LSAE) according to prior</h2><span id='topic+sample_LSAE'></span>

<h3>Description</h3>

<p>Sample Locus Specific Amplification Efficiency (LSAE) according to prior
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_LSAE(LSAE_variance, locus_names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_LSAE_+3A_lsae_variance">LSAE_variance</code></td>
<td>
<p>Numeric. See <a href="#topic+gf">gf</a> for an example.</p>
</td></tr>
<tr><td><code id="sample_LSAE_+3A_locus_names">locus_names</code></td>
<td>
<p>Character vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the Bright et al. log normal model, the expected peak height
includes a multiplicative factor for the locus (marker). These factors are
called the LSAEs (Locus Specific Amplification Efficiencies). In the model,
the prior for the log10 of LSAE is normal with mean 0. The variance
can be specified.
</p>


<h3>Value</h3>

<p>Named numeric with LSAEs for each locus (names).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gf)
lsae &lt;- sample_LSAE(LSAE_variance = gf$log_normal_settings$LSAE_variance_prior,
                     locus_names = gf$autosomal_markers)

# the barplot shows that some loci amplify better than others
barplot(lsae, las=2)
</code></pre>

<hr>
<h2 id='sample_mixture_from_genotypes'>Sample mixture profile with provided genotypes</h2><span id='topic+sample_mixture_from_genotypes'></span>

<h3>Description</h3>

<p>Sample mixture profile with provided genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_mixture_from_genotypes(genotypes, model, sample_name = "mixture")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_mixture_from_genotypes_+3A_genotypes">genotypes</code></td>
<td>
<p>List of contributor genotypes. See <a href="#topic+sample_contributor_genotypes">sample_contributor_genotypes</a>.</p>
</td></tr>
<tr><td><code id="sample_mixture_from_genotypes_+3A_model">model</code></td>
<td>
<p>pg_model object.</p>
</td></tr>
<tr><td><code id="sample_mixture_from_genotypes_+3A_sample_name">sample_name</code></td>
<td>
<p>Character. Defaults to &quot;mixture&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A mixture profile is sampled according to the provided <code>pg_model</code> (see <a href="#topic+gamma_model">gamma_model</a>, <a href="#topic+log_normal_model">log_normal_model</a> and genotypes (see <a href="#topic+sample_contributor_genotypes">sample_contributor_genotypes</a>).
</p>


<h3>Value</h3>

<p>DataFrame with at least SMASH columns (see <a href="#topic+SMASH_to_wide_table">SMASH_to_wide_table</a>). Depending on the chosen <code>pg_model</code> (e.g. <a href="#topic+gamma_model">gamma_model</a> or <a href="#topic+log_normal_model">log_normal_model</a>), other columns with further details about the simulation are returned as well.
</p>


<h3>See Also</h3>

<p><a href="#topic+sample_mixtures">sample_mixtures</a> for a function that samples many mixtures in one go.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read allele frequencies and kit data
freqs &lt;- read_allele_freqs(system.file("extdata","FBI_extended_Cauc.csv",
                           package = "simDNAmixtures"))
data(gf)

# define a pedigree of siblings S1 and S2 (and their parents)
ped_sibs &lt;- pedtools::nuclearPed(children = c("S1", "S2"))

# sample genotypes for a mixture of S1 + U1 + S2
# where U1 is an unrelated person
genotypes &lt;- sample_contributor_genotypes(contributors = c("S1","U1","S2"),
freqs, ped_sibs, loci = gf$autosomal_markers)

# define a gamma model for peak heights
gamma_model &lt;- gamma_model(mixture_proportions = c(0.5, 0.3, 0.2), mu = 1000.,
                    cv = 0.1, model_settings = gf$gamma_settings_no_stutter)

# sample mixture from genotypes
mix &lt;- sample_mixture_from_genotypes(genotypes, gamma_model)
</code></pre>

<hr>
<h2 id='sample_mixtures'>Sample mixtures with random genotypes and random parameters according to priors</h2><span id='topic+sample_mixtures'></span>

<h3>Description</h3>

<p>Sample mixtures with random genotypes and random parameters according to priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_mixtures(
  n,
  contributors,
  freqs,
  sampling_parameters,
  model_settings,
  sample_model,
  pedigree,
  results_directory,
  seed,
  write_non_contributors = FALSE,
  tag = "simulation"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_mixtures_+3A_n">n</code></td>
<td>
<p>Integer. Number of samples.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_contributors">contributors</code></td>
<td>
<p>Character vector with unique names of contributors. Valid names are &quot;U1&quot;, &quot;U2&quot;, ... for unrelated contributors or the names of pedigree members for related contributors.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_freqs">freqs</code></td>
<td>
<p>Allele frequencies (see <a href="#topic+read_allele_freqs">read_allele_freqs</a>)</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_sampling_parameters">sampling_parameters</code></td>
<td>
<p>List. Passed to the sample_model function.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_model_settings">model_settings</code></td>
<td>
<p>List. Passed to the sample_model function.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_sample_model">sample_model</code></td>
<td>
<p>Function such as <a href="#topic+sample_log_normal_model">sample_log_normal_model</a>.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_pedigree">pedigree</code></td>
<td>
<p>(optionally) <a href="pedtools.html#topic+ped">ped</a> object. Contributors can be named pedigree members.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_results_directory">results_directory</code></td>
<td>
<p>(optionally) Character with path to directory where results are written to disk.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_seed">seed</code></td>
<td>
<p>(optionally) Integer seed value that can be used to get reproducible runs. If results are written to disk, the 'Run details.txt' file will contain a seed that can be used for reproducing the result.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_write_non_contributors">write_non_contributors</code></td>
<td>
<p>Logical. If TRUE, sampled genotypes for non-contributing pedigree members will also be written to disk. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="sample_mixtures_+3A_tag">tag</code></td>
<td>
<p>Character. Used for sub directory name when results_directory is provided.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>results_directory</code> is provided, this function has the side effect of writing results to disk.
</p>
<p>Return value is a list with simulation results:</p>

<ul>
<li> <p><code>call</code> matched call
</p>
</li>
<li> <p><code>smash</code> DataFrame with all samples in SMASH format (see <a href="#topic+SMASH_to_wide_table">SMASH_to_wide_table</a>)
</p>
</li>
<li> <p><code>samples</code> Detailed results for each sample
</p>
</li>
<li> <p><code>parameter_summary</code> DataFrame with parameters for each sample
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>freqs &lt;- read_allele_freqs(system.file("extdata","FBI_extended_Cauc.csv",
                           package = "simDNAmixtures"))
data(gf)

sampling_parameters &lt;- list(min_mu = 50., max_mu = 5e3,
                           min_cv = 0.05, max_cv = 0.35,
                           degradation_shape1 = 10, degradation_shape2 = 1)

mixtures &lt;- sample_mixtures(n = 2, contributors = c("U1", "U2"), freqs = freqs,
                            sampling_parameters = sampling_parameters,
                            model_settings = gf$gamma_settings_no_stutter,
                            sample_model = sample_gamma_model)

</code></pre>

<hr>
<h2 id='sample_offspring'>Sample offspring from two parental genotypes</h2><span id='topic+sample_offspring'></span>

<h3>Description</h3>

<p>Sample offspring from two parental genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_offspring(father, mother, label = "Child")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_offspring_+3A_father">father</code></td>
<td>
<p>DataFrame (see <a href="#topic+sample_genotype">sample_genotype</a>)</p>
</td></tr>
<tr><td><code id="sample_offspring_+3A_mother">mother</code></td>
<td>
<p>DataFrame (see <a href="#topic+sample_genotype">sample_genotype</a>)</p>
</td></tr>
<tr><td><code id="sample_offspring_+3A_label">label</code></td>
<td>
<p>SampleName of child (character)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A genotype is sampled according to Mendelian inheritance. That is, one of two alleles of a parent is passed down to the offspring.
</p>


<h3>Value</h3>

<p>DataFrame (see <a href="#topic+sample_genotype">sample_genotype</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># below we read an allele freqs and sample a genotype
filename &lt;- system.file("extdata","FBI_extended_Cauc.csv",
                        package = "simDNAmixtures")
freqs &lt;- read_allele_freqs(filename)

# sample parents
father &lt;- sample_genotype(freqs, loci = c("D3S1358", "vWA"))
mother &lt;- sample_genotype(freqs, loci = c("D3S1358", "vWA"))

# sample child
child &lt;- sample_offspring(father, mother)
</code></pre>

<hr>
<h2 id='sample_pedigree_genotypes'>Sample genotypes for pedigree according to allele frequencies by allele dropping.</h2><span id='topic+sample_pedigree_genotypes'></span>

<h3>Description</h3>

<p>Sample genotypes for pedigree according to allele frequencies by allele dropping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_pedigree_genotypes(pedigree, freqs, loci = names(freqs))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_pedigree_genotypes_+3A_pedigree">pedigree</code></td>
<td>
<p><a href="pedtools.html#topic+ped">ped</a> object</p>
</td></tr>
<tr><td><code id="sample_pedigree_genotypes_+3A_freqs">freqs</code></td>
<td>
<p>Allele frequencies (see <a href="#topic+read_allele_freqs">read_allele_freqs</a>)</p>
</td></tr>
<tr><td><code id="sample_pedigree_genotypes_+3A_loci">loci</code></td>
<td>
<p>Character vector of locus names (defaults to <code>names</code> attribute of <code>freqs</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each founder, a genotype is sampled randomly by drawing two alleles according to allele frequencies. Alleles for the rest of the pedigree are then obtained by allele dropping: <a href="#topic+sample_offspring">sample_offspring</a> is invoked for each non-founder.
</p>


<h3>Value</h3>

<p>List of DataFrames with genotypes for each pedigree member. See <a href="#topic+sample_genotype">sample_genotype</a> for the DataFrame format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>freqs &lt;- read_allele_freqs(system.file("extdata","FBI_extended_Cauc.csv",
                           package = "simDNAmixtures"))
data(gf)

ped_sibs &lt;- pedtools::nuclearPed(children = c("S1", "S2"))

sibs_genotypes &lt;- sample_pedigree_genotypes(ped = ped_sibs,
freqs = freqs, loci = gf$autosomal_markers)
</code></pre>

<hr>
<h2 id='SMASH_to_wide_table'>Converts SMASH (SampleName, Marker, Allele, Size, Height) data to a wide table</h2><span id='topic+SMASH_to_wide_table'></span>

<h3>Description</h3>

<p>Converts SMASH (SampleName, Marker, Allele, Size, Height) data to a wide table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMASH_to_wide_table(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SMASH_to_wide_table_+3A_x">x</code></td>
<td>
<p>DataFrame with SampleName, Marker, Allele, Size, Height columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame with columns: Sample Name, Marker, Allele 1, Allele 2, ..., Size 1, Size 2, ..., Height 1, Height 2, ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'># generate example data in SMASH form
freqs &lt;- read_allele_freqs(system.file("extdata","FBI_extended_Cauc.csv",
package = "simDNAmixtures"))
data(gf)

sampling_parameters &lt;- list(min_mu = 50., max_mu = 5e3,
                            min_cv = 0.05, max_cv = 0.35,
                            degradation_shape1 = 10, degradation_shape2 = 1)

mixtures &lt;- sample_mixtures(n = 2, contributors = c("U1"), freqs = freqs,
                            sampling_parameters = sampling_parameters,
                            model_settings = gf$gamma_settings_no_stutter,
                            sample_model = sample_gamma_model)

# convert from SMASH to wide table
wide_table &lt;- SMASH_to_wide_table(mixtures$smash)
</code></pre>

<hr>
<h2 id='stutter_type'>Defines a stutter type to be used in the allele specific stutter model.</h2><span id='topic+stutter_type'></span>

<h3>Description</h3>

<p>Defines a stutter type to be used in the allele specific stutter model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stutter_type(
  name,
  delta,
  applies_to_all_loci = TRUE,
  stutter_regression,
  stutter_exceptions,
  applies_to_loci,
  repeat_length_by_marker
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stutter_type_+3A_name">name</code></td>
<td>
<p>Character. Name of the stutter, e.g. &quot;BackStutter&quot;</p>
</td></tr>
<tr><td><code id="stutter_type_+3A_delta">delta</code></td>
<td>
<p>Numeric. When length one, repeat units gained (lost when negative). When length two, the second element is the number of base pairs gained (lost).</p>
</td></tr>
<tr><td><code id="stutter_type_+3A_applies_to_all_loci">applies_to_all_loci</code></td>
<td>
<p>Logical. Defaults to TRUE.</p>
</td></tr>
<tr><td><code id="stutter_type_+3A_stutter_regression">stutter_regression</code></td>
<td>
<p>Function. See <a href="#topic+read_stutter_regression">read_stutter_regression</a>.</p>
</td></tr>
<tr><td><code id="stutter_type_+3A_stutter_exceptions">stutter_exceptions</code></td>
<td>
<p>Optionally a list. See <a href="#topic+read_stutter_exceptions">read_stutter_exceptions</a>.</p>
</td></tr>
<tr><td><code id="stutter_type_+3A_applies_to_loci">applies_to_loci</code></td>
<td>
<p>Optionally a character vector of locus names to which this stutter type applies.</p>
</td></tr>
<tr><td><code id="stutter_type_+3A_repeat_length_by_marker">repeat_length_by_marker</code></td>
<td>
<p>Optionally a named integer vector with repeat lengths by marker. Only needed when delta is of length two.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When a pg_model is constructed (see <a href="#topic+log_normal_model">log_normal_model</a>), a stutter model can optionally be applied.
</p>


<h3>Value</h3>

<p>Object of class <code>stutter_type</code> to be passed to <a href="#topic+allele_specific_stutter_model">allele_specific_stutter_model</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filename_bs_exceptions &lt;- system.file("extdata",
"GlobalFiler_Stutter_Exceptions_3500.csv",package = "simDNAmixtures")
bs_exceptions &lt;- read_stutter_exceptions(filename_bs_exceptions)

filename_bs_regression &lt;- system.file("extdata",
"GlobalFiler_Stutter_3500.txt",package = "simDNAmixtures")
bs_regression &lt;- read_stutter_regression(filename_bs_regression)

backstutter &lt;- stutter_type(name = "BackStutter", delta = -1,
                            stutter_regression = bs_regression,
                            stutter_exceptions = bs_exceptions)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
