<!DOCTYPE html><html><head><title>Help for package simer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {simer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annotation'><p>Annotation simulation</p></a></li>
<li><a href='#build.cov'><p>Correlation building</p></a></li>
<li><a href='#cal.eff'><p>QTN genetic effects</p></a></li>
<li><a href='#checkEnv'><p>Environmental factor checking</p></a></li>
<li><a href='#format_time'><p>Time formating</p></a></li>
<li><a href='#generate.map'><p>Marker information</p></a></li>
<li><a href='#generate.pop'><p>Population generator</p></a></li>
<li><a href='#geno.cvt1'><p>Genotype code convertor 1</p></a></li>
<li><a href='#geno.cvt2'><p>Genotype code convertor 2</p></a></li>
<li><a href='#genotype'><p>Genotype simulation</p></a></li>
<li><a href='#getfam'><p>Family index and within-family index</p></a></li>
<li><a href='#GxG.network'><p>Genetic interaction network</p></a></li>
<li><a href='#IndPerGen'><p>Individual number per generation</p></a></li>
<li><a href='#load_if_installed'><p>Installation checking</p></a></li>
<li><a href='#logging.initialize'><p>Logging initialization</p></a></li>
<li><a href='#logging.log'><p>Logging</p></a></li>
<li><a href='#logging.print'><p>Logging printer</p></a></li>
<li><a href='#make_line'><p>Line making</p></a></li>
<li><a href='#mate'><p>Mate</p></a></li>
<li><a href='#mate.2waycro'><p>Two-way cross</p></a></li>
<li><a href='#mate.3waycro'><p>Three-way cross</p></a></li>
<li><a href='#mate.4waycro'><p>Four-way cross process</p></a></li>
<li><a href='#mate.assort'><p>Assortative mating</p></a></li>
<li><a href='#mate.backcro'><p>Back cross</p></a></li>
<li><a href='#mate.clone'><p>Clone</p></a></li>
<li><a href='#mate.dh'><p>Doubled haploid</p></a></li>
<li><a href='#mate.disassort'><p>Disassortative mating</p></a></li>
<li><a href='#mate.randexself'><p>Random mating excluding self-pollination</p></a></li>
<li><a href='#mate.randmate'><p>Random mating</p></a></li>
<li><a href='#mate.selfpol'><p>Self-pollination</p></a></li>
<li><a href='#mate.userped'><p>User-specified pedigree mating</p></a></li>
<li><a href='#mkl_env'><p>MKL environment</p></a></li>
<li><a href='#param.annot'><p>Annotation parameters generator</p></a></li>
<li><a href='#param.geno'><p>Genotype parameters generator</p></a></li>
<li><a href='#param.global'><p>Global parameters generator</p></a></li>
<li><a href='#param.pheno'><p>Phenotype parameters generator</p></a></li>
<li><a href='#param.reprod'><p>Reproduction parameters generator</p></a></li>
<li><a href='#param.sel'><p>Selection parameters generator</p></a></li>
<li><a href='#param.simer'><p>Parameter generator</p></a></li>
<li><a href='#paste_label'><p>Pasting label</p></a></li>
<li><a href='#phenotype'><p>Phenotype simulation</p></a></li>
<li><a href='#pop.geno'><p>Raw genotype matrix from outside in simdata</p></a></li>
<li><a href='#pop.map'><p>Map file from outside in simdata</p></a></li>
<li><a href='#print_accomplished'><p>Accomplishment</p></a></li>
<li><a href='#print_bar'><p>Progress bar</p></a></li>
<li><a href='#print_info'><p>Simer information</p></a></li>
<li><a href='#remove_bigmatrix'><p>Big.matrix removing</p></a></li>
<li><a href='#reproduces'><p>Reproduction</p></a></li>
<li><a href='#rule_wrap'><p>Accomplishment</p></a></li>
<li><a href='#selects'><p>Selection</p></a></li>
<li><a href='#simer'><p>Simer</p></a></li>
<li><a href='#simer.Data'><p>Data handling</p></a></li>
<li><a href='#simer.Data.Bfile2MVP'><p>simer.Data.Bfile2MVP: To transform plink binary data to MVP package</p></a></li>
<li><a href='#simer.Data.cHIBLUP'><p>Genetic evaluation</p></a></li>
<li><a href='#simer.Data.Env'><p>Environmental factor selection</p></a></li>
<li><a href='#simer.Data.Geno'><p>Genotype data quality control</p></a></li>
<li><a href='#simer.Data.Impute'><p>Genotype data imputation</p></a></li>
<li><a href='#simer.Data.Json'><p>Data quality control</p></a></li>
<li><a href='#simer.Data.Kin'><p>simer.Data.EMMA: To construct EMMA kinship matrix</p></a></li>
<li><a href='#simer.Data.Map'><p>simer.Data.Map: To check map file</p></a></li>
<li><a href='#simer.Data.MVP2Bfile'><p>simer.Data.MVP2Bfile: To transform MVP data to binary format</p></a></li>
<li><a href='#simer.Data.MVP2MVP'><p>Genotype data conversion</p></a></li>
<li><a href='#simer.Data.Ped'><p>Pedigree data quality control</p></a></li>
<li><a href='#simer.Data.Pheno'><p>Phenotype data quality control</p></a></li>
<li><a href='#simer.Data.SELIND'><p>Selection index construction</p></a></li>
<li><a href='#simer.Version'><p>Simer version</p></a></li>
<li><a href='#write.file'><p>File writing</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Data Simulation for Life Science and Breeding</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.0.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-28</td>
</tr>
<tr>
<td>Description:</td>
<td>Data simulator including genotype, phenotype, pedigree, 
    selection and reproduction in R. It simulates most of reproduction process
    of animals or plants and provides data for GS (Genomic Selection),
    GWAS (Genome-Wide Association Study), and Breeding.
    For ADI model, please see Kao C and Zeng Z (2002) &lt;<a href="https://doi.org/10.1093%2Fgenetics%2F160.3.1243">doi:10.1093/genetics/160.3.1243</a>&gt;.
    For build.cov, please see B. D. Ripley (1987) &lt;ISBN:9780470009604&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License 2.0</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/xiaolei-lab/SIMER">https://github.com/xiaolei-lab/SIMER</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/xiaolei-lab/SIMER/issues">https://github.com/xiaolei-lab/SIMER/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, stats, Matrix, methods, MASS, Rcpp, jsonlite, igraph</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppProgress, BH, bigmemory</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), bigmemory</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaolei Liu &lt;xll198708@gmail.com&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-28 09:28:51 UTC; Administrator</td>
</tr>
<tr>
<td>Author:</td>
<td>Dong Yin [aut],
  Xuanning Zhang [aut],
  Lilin Yin [aut],
  Haohao Zhang [aut],
  Zhenshuang Tang [aut],
  Jingya Xu [aut],
  Xiaohui Yuan [aut],
  Xiang Zhou [aut],
  Xinyun Li [aut],
  Shuhong Zhao [aut],
  Xiaolei Liu [cre, aut, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-29 11:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='annotation'>Annotation simulation</h2><span id='topic+annotation'></span>

<h3>Description</h3>

<p>Generating a map with annotation information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotation(SP, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotation_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="annotation_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Jul 10, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$map$pop.map</dt><dd><p>the map data with annotation information.</p>
</dd>
<dt>$map$species</dt><dd><p>the species of genetic map, which can be &quot;arabidopsis&quot;, &quot;cattle&quot;, &quot;chicken&quot;, &quot;dog&quot;, &quot;horse&quot;, &quot;human&quot;, &quot;maize&quot;, &quot;mice&quot;, &quot;pig&quot;, and &quot;rice&quot;.</p>
</dd>
<dt>$map$pop.marker</dt><dd><p>the number of markers.</p>
</dd>
<dt>$map$num.chr</dt><dd><p>the number of chromosomes.</p>
</dd>
<dt>$map$len.chr</dt><dd><p>the length of chromosomes.</p>
</dd>
<dt>$map$qtn.model</dt><dd><p>the genetic model of QTN such as 'A + D'.</p>
</dd>
<dt>$map$qtn.index</dt><dd><p>the QTN index for each trait.</p>
</dd>
<dt>$map$qtn.num</dt><dd><p>the QTN number for (each group in) each trait.</p>
</dd>
<dt>$map$qtn.dist</dt><dd><p>the QTN distribution containing 'norm', 'geom', 'gamma' or 'beta'.</p>
</dd>
<dt>$map$qtn.var</dt><dd><p>the variances for normal distribution.</p>
</dd>
<dt>$map$qtn.prob</dt><dd><p>the probability of success for geometric distribution.</p>
</dd>
<dt>$map$qtn.shape</dt><dd><p>the shape parameter for gamma distribution.</p>
</dd>
<dt>$map$qtn.scale</dt><dd><p>the scale parameter for gamma distribution.</p>
</dd>
<dt>$map$qtn.shape1</dt><dd><p>the shape1 parameter for beta distribution.</p>
</dd>
<dt>$map$qtn.shape2</dt><dd><p>the shape2 parameter for beta distribution.</p>
</dd>
<dt>$map$qtn.ncp</dt><dd><p>the ncp parameter for beta distribution.</p>
</dd>
<dt>$map$qtn.spot</dt><dd><p>the QTN distribution probability in each block.</p>
</dd>
<dt>$map$len.block</dt><dd><p>the block length.</p>
</dd>
<dt>$map$maf</dt><dd><p>the maf threshold, markers less than this threshold will be exclude.</p>
</dd>
<dt>$map$recom.spot</dt><dd><p>whether to generate recombination events.</p>
</dd>
<dt>$map$range.hot</dt><dd><p>the recombination times range in the hot spot.</p>
</dd>
<dt>$map$range.cold</dt><dd><p>the recombination times range in the cold spot.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))

# Run annotation simulation
SP &lt;- annotation(SP)
</code></pre>

<hr>
<h2 id='build.cov'>Correlation building</h2><span id='topic+build.cov'></span>

<h3>Description</h3>

<p>To bulid correlation of variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build.cov(df = NULL, mu = rep(0, nrow(Sigma)), Sigma = diag(2), tol = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build.cov_+3A_df">df</code></td>
<td>
<p>a data frame needing building correlation.</p>
</td></tr>
<tr><td><code id="build.cov_+3A_mu">mu</code></td>
<td>
<p>means of the variables.</p>
</td></tr>
<tr><td><code id="build.cov_+3A_sigma">Sigma</code></td>
<td>
<p>covariance matrix of variables.</p>
</td></tr>
<tr><td><code id="build.cov_+3A_tol">tol</code></td>
<td>
<p>tolerance (relative to largest variance) for numerical lack of positive-definiteness in Sigma.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Oct 10, 2019
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>a data frame with expected correlation
</p>


<h3>Author(s)</h3>

<p>Dong Yin and R
</p>


<h3>References</h3>

<p>B. D. Ripley (1987) Stochastic Simulation. Wiley. Page 98
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- data.frame(tr1 = rnorm(100), tr2 = rnorm(100))
df.cov &lt;- build.cov(df)
var(df.cov)
</code></pre>

<hr>
<h2 id='cal.eff'>QTN genetic effects</h2><span id='topic+cal.eff'></span>

<h3>Description</h3>

<p>Calculate for genetic effects vector of selected markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cal.eff(
  qtn.num = 10,
  qtn.dist = "norm",
  qtn.var = 1,
  qtn.prob = 0.5,
  qtn.shape = 1,
  qtn.scale = 1,
  qtn.shape1 = 1,
  qtn.shape2 = 1,
  qtn.ncp = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cal.eff_+3A_qtn.num">qtn.num</code></td>
<td>
<p>integer: the QTN number of single trait; vector: the multiple group QTN number of single trait; matrix: the QTN number of multiple traits.</p>
</td></tr>
<tr><td><code id="cal.eff_+3A_qtn.dist">qtn.dist</code></td>
<td>
<p>the QTN distribution containing 'norm', 'geom', 'gamma' or 'beta'.</p>
</td></tr>
<tr><td><code id="cal.eff_+3A_qtn.var">qtn.var</code></td>
<td>
<p>the standard deviations for normal distribution.</p>
</td></tr>
<tr><td><code id="cal.eff_+3A_qtn.prob">qtn.prob</code></td>
<td>
<p>the probability of success for geometric distribution.</p>
</td></tr>
<tr><td><code id="cal.eff_+3A_qtn.shape">qtn.shape</code></td>
<td>
<p>the shape parameter for gamma distribution.</p>
</td></tr>
<tr><td><code id="cal.eff_+3A_qtn.scale">qtn.scale</code></td>
<td>
<p>the scale parameter for gamma distribution.</p>
</td></tr>
<tr><td><code id="cal.eff_+3A_qtn.shape1">qtn.shape1</code></td>
<td>
<p>the shape1 parameter for beta distribution.</p>
</td></tr>
<tr><td><code id="cal.eff_+3A_qtn.shape2">qtn.shape2</code></td>
<td>
<p>the shape2 parameter for beta distribution.</p>
</td></tr>
<tr><td><code id="cal.eff_+3A_qtn.ncp">qtn.ncp</code></td>
<td>
<p>the ncp parameter for beta distribution.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>a vector of genetic effect.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>eff &lt;- cal.eff(qtn.num = 10)
str(eff)
</code></pre>

<hr>
<h2 id='checkEnv'>Environmental factor checking</h2><span id='topic+checkEnv'></span>

<h3>Description</h3>

<p>Check the levels of environmental factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkEnv(data, envName, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkEnv_+3A_data">data</code></td>
<td>
<p>data needing check.</p>
</td></tr>
<tr><td><code id="checkEnv_+3A_envname">envName</code></td>
<td>
<p>the environmental factor name within the data.</p>
</td></tr>
<tr><td><code id="checkEnv_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Sep 10, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>data without environmental factors of wrong level.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- data.frame(a = c(1, 1, 2), b = c(2, 2, 3), c = c(3, 3, 4))
envName &lt;- c("a", "b", "c")
data &lt;- checkEnv(data = data, envName = envName)
</code></pre>

<hr>
<h2 id='format_time'>Time formating</h2><span id='topic+format_time'></span>

<h3>Description</h3>

<p>Format the time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>format_time(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="format_time_+3A_x">x</code></td>
<td>
<p>the total seconds.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Oct 22, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>running time.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>format_time(x = 7200)
</code></pre>

<hr>
<h2 id='generate.map'>Marker information</h2><span id='topic+generate.map'></span>

<h3>Description</h3>

<p>Generate map data with marker information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate.map(
  species = NULL,
  pop.marker = NULL,
  num.chr = 18,
  len.chr = 1.5e+08
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate.map_+3A_species">species</code></td>
<td>
<p>the species of genetic map, which can be &quot;arabidopsis&quot;, &quot;cattle&quot;, &quot;chicken&quot;, &quot;dog&quot;, &quot;horse&quot;, &quot;human&quot;, &quot;maize&quot;, &quot;mice&quot;, &quot;pig&quot;, and &quot;rice&quot;.</p>
</td></tr>
<tr><td><code id="generate.map_+3A_pop.marker">pop.marker</code></td>
<td>
<p>the number of markers.</p>
</td></tr>
<tr><td><code id="generate.map_+3A_num.chr">num.chr</code></td>
<td>
<p>the number of chromosomes.</p>
</td></tr>
<tr><td><code id="generate.map_+3A_len.chr">len.chr</code></td>
<td>
<p>the length of chromosomes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Mar 19, 2022
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>a data frame with marker information.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pop.map &lt;- generate.map(pop.marker = 1e4)
str(pop.map)
</code></pre>

<hr>
<h2 id='generate.pop'>Population generator</h2><span id='topic+generate.pop'></span>

<h3>Description</h3>

<p>Generate population according to the number of individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate.pop(pop.ind = 100, from = 1, ratio = 0.5, gen = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate.pop_+3A_pop.ind">pop.ind</code></td>
<td>
<p>the number of the individuals in a population.</p>
</td></tr>
<tr><td><code id="generate.pop_+3A_from">from</code></td>
<td>
<p>initial index of the population.</p>
</td></tr>
<tr><td><code id="generate.pop_+3A_ratio">ratio</code></td>
<td>
<p>sex ratio of males in a population.</p>
</td></tr>
<tr><td><code id="generate.pop_+3A_gen">gen</code></td>
<td>
<p>generation ID of the population.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>a data frame of population information.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pop &lt;- generate.pop(pop.ind = 100)
head(pop)
</code></pre>

<hr>
<h2 id='geno.cvt1'>Genotype code convertor 1</h2><span id='topic+geno.cvt1'></span>

<h3>Description</h3>

<p>Convert genotype matrix from (0, 1) to (0, 1, 2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geno.cvt1(pop.geno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geno.cvt1_+3A_pop.geno">pop.geno</code></td>
<td>
<p>genotype matrix of (0, 1).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>genotype matrix of (0, 1, 2).
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
SP &lt;- param.geno(pop.marker = 1e4, pop.ind = 1e2, incols = 2)
SP &lt;- genotype(SP)
geno1 &lt;- SP$geno$pop.geno$gen1
geno2 &lt;- geno.cvt1(geno1)
geno1[1:6, 1:4]
geno2[1:6, 1:2]

</code></pre>

<hr>
<h2 id='geno.cvt2'>Genotype code convertor 2</h2><span id='topic+geno.cvt2'></span>

<h3>Description</h3>

<p>Convert genotype matrix from (0, 1, 2) to (0, 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geno.cvt2(pop.geno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geno.cvt2_+3A_pop.geno">pop.geno</code></td>
<td>
<p>genotype matrix of (0, 1, 2).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Jul 11, 2020
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>genotype matrix of (0, 1).
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
SP &lt;- param.geno(pop.marker = 1e4, pop.ind = 1e2, incols = 1)
SP &lt;- genotype(SP)
geno1 &lt;- SP$geno$pop.geno$gen1
geno2 &lt;- geno.cvt2(geno1)
geno1[1:6, 1:2]
geno2[1:6, 1:4]

</code></pre>

<hr>
<h2 id='genotype'>Genotype simulation</h2><span id='topic+genotype'></span>

<h3>Description</h3>

<p>Generating and editing genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotype(SP = NULL, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genotype_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="genotype_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="genotype_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$geno$pop.geno</dt><dd><p>the genotype data.</p>
</dd>
<dt>$geno$incols</dt><dd><p>'1': one-column genotype represents an individual; '2': two-column genotype represents an individual.</p>
</dd>
<dt>$geno$pop.marker</dt><dd><p>the number of markers.</p>
</dd>
<dt>$geno$pop.ind</dt><dd><p>the number of individuals in the base population.</p>
</dd>
<dt>$geno$prob</dt><dd><p>the genotype code probability.</p>
</dd>
<dt>$geno$rate.mut</dt><dd><p>the mutation rate of the genotype data.</p>
</dd>
<dt>$geno$cld</dt><dd><p>whether to generate a complete LD genotype data when 'incols == 2'.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate genotype simulation parameters
SP &lt;- param.geno(pop.marker = 1e4, pop.ind = 1e2)

# Run genotype simulation
SP &lt;- genotype(SP)

</code></pre>

<hr>
<h2 id='getfam'>Family index and within-family index</h2><span id='topic+getfam'></span>

<h3>Description</h3>

<p>Get indice of family and within-family
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getfam(sir, dam, fam.op, mode = c("pat", "mat", "pm"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getfam_+3A_sir">sir</code></td>
<td>
<p>the indice of sires.</p>
</td></tr>
<tr><td><code id="getfam_+3A_dam">dam</code></td>
<td>
<p>the indice of dams.</p>
</td></tr>
<tr><td><code id="getfam_+3A_fam.op">fam.op</code></td>
<td>
<p>the initial index of family indice.</p>
</td></tr>
<tr><td><code id="getfam_+3A_mode">mode</code></td>
<td>
<p>&quot;pat&quot;: paternal mode; &quot;mat&quot;: maternal mode; &quot;pm&quot;: paternal and maternal mode.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>a matrix with family indice and within-family indice.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>s &lt;- c(0, 0, 0, 0, 1, 3, 3, 1, 5, 7, 5, 7, 1, 3, 5, 7)
d &lt;- c(0, 0, 0, 0, 2, 4, 4, 2, 6, 8, 8, 6, 6, 8, 4, 8)
fam &lt;- getfam(sir = s, dam = d, fam.op = 1, mode = "pm")
fam
</code></pre>

<hr>
<h2 id='GxG.network'>Genetic interaction network</h2><span id='topic+GxG.network'></span>

<h3>Description</h3>

<p>Generate genetic interaction effect combination network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GxG.network(pop.map = NULL, qtn.pos = 1:10, qtn.model = "A:D")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GxG.network_+3A_pop.map">pop.map</code></td>
<td>
<p>the map data with annotation information.</p>
</td></tr>
<tr><td><code id="GxG.network_+3A_qtn.pos">qtn.pos</code></td>
<td>
<p>the index of QTNs in the map data.</p>
</td></tr>
<tr><td><code id="GxG.network_+3A_qtn.model">qtn.model</code></td>
<td>
<p>the genetic model of QTN such as 'A:D'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Mar 19, 2022
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>a data frame of genetic interaction effect.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pop.map &lt;- generate.map(pop.marker = 1e4)
GxG.net &lt;- GxG.network(pop.map)
head(GxG.net)
</code></pre>

<hr>
<h2 id='IndPerGen'>Individual number per generation</h2><span id='topic+IndPerGen'></span>

<h3>Description</h3>

<p>Calculate the individual number per generation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IndPerGen(
  pop,
  pop.gen = 2,
  ps = c(0.8, 0.8),
  reprod.way = "randmate",
  sex.rate = 0.5,
  prog = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IndPerGen_+3A_pop">pop</code></td>
<td>
<p>the population information containing environmental factors and other effects.</p>
</td></tr>
<tr><td><code id="IndPerGen_+3A_pop.gen">pop.gen</code></td>
<td>
<p>the generations of simulated population.</p>
</td></tr>
<tr><td><code id="IndPerGen_+3A_ps">ps</code></td>
<td>
<p>if ps &lt;= 1, fraction selected in selection of males and females; if ps &gt; 1, ps is number of selected males and females.</p>
</td></tr>
<tr><td><code id="IndPerGen_+3A_reprod.way">reprod.way</code></td>
<td>
<p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</td></tr>
<tr><td><code id="IndPerGen_+3A_sex.rate">sex.rate</code></td>
<td>
<p>the sex ratio of simulated population.</p>
</td></tr>
<tr><td><code id="IndPerGen_+3A_prog">prog</code></td>
<td>
<p>the progeny number of an individual.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 12, 2022
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the vector containing the individual number per generation.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pop &lt;- generate.pop(pop.ind = 100)
count.ind &lt;- IndPerGen(pop)
</code></pre>

<hr>
<h2 id='load_if_installed'>Installation checking</h2><span id='topic+load_if_installed'></span>

<h3>Description</h3>

<p>Check if the software is installed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_if_installed(package)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_if_installed_+3A_package">package</code></td>
<td>
<p>the package name.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Oct 22, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>

<hr>
<h2 id='logging.initialize'>Logging initialization</h2><span id='topic+logging.initialize'></span>

<h3>Description</h3>

<p>Initialize the logging process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logging.initialize(module, outpath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logging.initialize_+3A_module">module</code></td>
<td>
<p>the module name.</p>
</td></tr>
<tr><td><code id="logging.initialize_+3A_outpath">outpath</code></td>
<td>
<p>the path of output files, Simer writes files only if outpath is not 'NULL'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Jul 11, 2020
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>

<hr>
<h2 id='logging.log'>Logging</h2><span id='topic+logging.log'></span>

<h3>Description</h3>

<p>Print or write log.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logging.log(
  ...,
  file = NULL,
  sep = " ",
  fill = FALSE,
  labels = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logging.log_+3A_...">...</code></td>
<td>
<p>R objects.</p>
</td></tr>
<tr><td><code id="logging.log_+3A_file">file</code></td>
<td>
<p>a connection or a character string naming the file to print to. If &quot;&quot; (the default), cat prints to the standard output connection, the console unless redirected by sink. If it is &quot;|cmd&quot;, the output is piped to the command given by ‘cmd’, by opening a pipe connection.</p>
</td></tr>
<tr><td><code id="logging.log_+3A_sep">sep</code></td>
<td>
<p>a character vector of strings to append after each element.</p>
</td></tr>
<tr><td><code id="logging.log_+3A_fill">fill</code></td>
<td>
<p>a logical or (positive) numeric controlling how the output is broken into successive lines.</p>
</td></tr>
<tr><td><code id="logging.log_+3A_labels">labels</code></td>
<td>
<p>a character vector of labels for the lines printed. Ignored if fill is FALSE.</p>
</td></tr>
<tr><td><code id="logging.log_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Jul 11, 2020
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logging.log('simer')
</code></pre>

<hr>
<h2 id='logging.print'>Logging printer</h2><span id='topic+logging.print'></span>

<h3>Description</h3>

<p>Print R object information into file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logging.print(x, file = NULL, append = TRUE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logging.print_+3A_x">x</code></td>
<td>
<p>a matrix or a list.</p>
</td></tr>
<tr><td><code id="logging.print_+3A_file">file</code></td>
<td>
<p>the filename of output file.</p>
</td></tr>
<tr><td><code id="logging.print_+3A_append">append</code></td>
<td>
<p>logical. If TRUE, output will be appended to file; otherwise, it will overwrite the contents of file.</p>
</td></tr>
<tr><td><code id="logging.print_+3A_verbose">verbose</code></td>
<td>
<p>whether to print details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Feb 7, 2020
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- list(a = "a", b = "b")
logging.print(x)
</code></pre>

<hr>
<h2 id='make_line'>Line making</h2><span id='topic+make_line'></span>

<h3>Description</h3>

<p>Add a line to the screen.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_line(string, width, linechar = " ", align = "center", margin = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_line_+3A_string">string</code></td>
<td>
<p>a string.</p>
</td></tr>
<tr><td><code id="make_line_+3A_width">width</code></td>
<td>
<p>the width of the message.</p>
</td></tr>
<tr><td><code id="make_line_+3A_linechar">linechar</code></td>
<td>
<p>char in every line.</p>
</td></tr>
<tr><td><code id="make_line_+3A_align">align</code></td>
<td>
<p>the position of string.</p>
</td></tr>
<tr><td><code id="make_line_+3A_margin">margin</code></td>
<td>
<p>the margin information, default 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Dec 12, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>

<hr>
<h2 id='mate'>Mate</h2><span id='topic+mate'></span>

<h3>Description</h3>

<p>Mating according to the indice of sires and dams.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate(pop.geno, index.sir, index.dam, ncpus = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate_+3A_pop.geno">pop.geno</code></td>
<td>
<p>the genotype data.</p>
</td></tr>
<tr><td><code id="mate_+3A_index.sir">index.sir</code></td>
<td>
<p>the indice of sires.</p>
</td></tr>
<tr><td><code id="mate_+3A_index.dam">index.dam</code></td>
<td>
<p>the indice of dams.</p>
</td></tr>
<tr><td><code id="mate_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>a genotype matrix after mating
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate the genotype data
SP &lt;- param.geno(pop.marker = 1e4, pop.ind = 1e2)
SP &lt;- genotype(SP)
pop.geno &lt;- SP$geno$pop.geno$gen1

# The mating design
index.sir &lt;- rep(1:50, each = 2)
index.dam &lt;- rep(51:100, each = 2)

# Mate according to mating design
geno.curr &lt;- mate(pop.geno = pop.geno, index.sir = index.sir,
                  index.dam = index.dam)
geno.curr[1:5, 1:5]

</code></pre>

<hr>
<h2 id='mate.2waycro'>Two-way cross</h2><span id='topic+mate.2waycro'></span>

<h3>Description</h3>

<p>Produce individuals by two-way cross.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.2waycro(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.2waycro_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.2waycro_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.2waycro_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "2waycro")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Two different breeds are cut by sex
SP$pheno$pop$gen1$sex &lt;- rep(c(1, 2), c(50, 50))
# Run selection
SP &lt;- selects(SP)
# Run two-way cross
SP &lt;- mate.2waycro(SP)

</code></pre>

<hr>
<h2 id='mate.3waycro'>Three-way cross</h2><span id='topic+mate.3waycro'></span>

<h3>Description</h3>

<p>Produce individuals by three-way cross.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.3waycro(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.3waycro_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.3waycro_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.3waycro_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 11, 2022
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "3waycro")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Three different breeds are cut by sex
SP$pheno$pop$gen1$sex &lt;- rep(c(1, 2, 1), c(30, 30, 40))
# Run selection
SP &lt;- selects(SP)
# Run three-way cross
SP &lt;- mate.3waycro(SP)

</code></pre>

<hr>
<h2 id='mate.4waycro'>Four-way cross process</h2><span id='topic+mate.4waycro'></span>

<h3>Description</h3>

<p>Produce individuals by four-way cross.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.4waycro(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.4waycro_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.4waycro_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.4waycro_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 11, 2022
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "4waycro")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Four different breeds are cut by sex
SP$pheno$pop$gen1$sex &lt;- rep(c(1, 2, 1, 2), c(25, 25, 25, 25))
# Run selection
SP &lt;- selects(SP)
# Run four-way cross
SP &lt;- mate.4waycro(SP)

</code></pre>

<hr>
<h2 id='mate.assort'>Assortative mating</h2><span id='topic+mate.assort'></span>

<h3>Description</h3>

<p>Produce individuals by assortative mating.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.assort(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.assort_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.assort_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.assort_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Sep 30, 2022
Last update: Sep 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "assort")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)
# Run random mating
SP &lt;- mate.assort(SP)

</code></pre>

<hr>
<h2 id='mate.backcro'>Back cross</h2><span id='topic+mate.backcro'></span>

<h3>Description</h3>

<p>Produce individuals by back cross.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.backcro(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.backcro_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.backcro_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.backcro_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 12, 2022
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "backcro")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Two different breeds are cut by sex
SP$pheno$pop$gen1$sex &lt;- rep(c(1, 2), c(50, 50))
# Run selection
SP &lt;- selects(SP)
# Run back cross
SP &lt;- mate.backcro(SP)

</code></pre>

<hr>
<h2 id='mate.clone'>Clone</h2><span id='topic+mate.clone'></span>

<h3>Description</h3>

<p>Produce individuals by clone.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.clone(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.clone_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.clone_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.clone_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "clone")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)
# Run clone
SP &lt;- mate.clone(SP)

</code></pre>

<hr>
<h2 id='mate.dh'>Doubled haploid</h2><span id='topic+mate.dh'></span>

<h3>Description</h3>

<p>Produce individuals by doubled haploid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.dh(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.dh_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.dh_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.dh_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "dh")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)
# Run doubled haploid
SP &lt;- mate.dh(SP)

</code></pre>

<hr>
<h2 id='mate.disassort'>Disassortative mating</h2><span id='topic+mate.disassort'></span>

<h3>Description</h3>

<p>Produce individuals by disassortative mating.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.disassort(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.disassort_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.disassort_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.disassort_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Sep 30, 2022
Last update: Sep 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "disassort")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)
# Run random mating
SP &lt;- mate.assort(SP)

</code></pre>

<hr>
<h2 id='mate.randexself'>Random mating excluding self-pollination</h2><span id='topic+mate.randexself'></span>

<h3>Description</h3>

<p>Produce individuals by random mating excluding self-pollination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.randexself(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.randexself_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.randexself_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.randexself_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "randexself")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)
# Run random mating excluding self-pollination
SP &lt;- mate.randexself(SP)

</code></pre>

<hr>
<h2 id='mate.randmate'>Random mating</h2><span id='topic+mate.randmate'></span>

<h3>Description</h3>

<p>Produce individuals by random-mating.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.randmate(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.randmate_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.randmate_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.randmate_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "randmate")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)
# Run random mating
SP &lt;- mate.randmate(SP)

</code></pre>

<hr>
<h2 id='mate.selfpol'>Self-pollination</h2><span id='topic+mate.selfpol'></span>

<h3>Description</h3>

<p>Produce individuals by self-pollination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.selfpol(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.selfpol_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.selfpol_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.selfpol_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "selfpol")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)
# Run self-pollination
SP &lt;- mate.selfpol(SP)

</code></pre>

<hr>
<h2 id='mate.userped'>User-specified pedigree mating</h2><span id='topic+mate.userped'></span>

<h3>Description</h3>

<p>Produce individuals by user-specified pedigree mating.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mate.userped(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mate.userped_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="mate.userped_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="mate.userped_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 12, 2022
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.sel</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the sex ratio of simulated population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "userped")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run user-specified pedigree mating
SP &lt;- mate.userped(SP)

</code></pre>

<hr>
<h2 id='mkl_env'>MKL environment</h2><span id='topic+mkl_env'></span>

<h3>Description</h3>

<p>Run code in the MKL environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mkl_env(exprs, threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mkl_env_+3A_exprs">exprs</code></td>
<td>
<p>the expression.</p>
</td></tr>
<tr><td><code id="mkl_env_+3A_threads">threads</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Oct 22, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>

<hr>
<h2 id='param.annot'>Annotation parameters generator</h2><span id='topic+param.annot'></span>

<h3>Description</h3>

<p>Generate parameters for annotation data simulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.annot(SP = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.annot_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="param.annot_+3A_...">...</code></td>
<td>
<p>one or more parameter(s) for map simulation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Feb 24, 2022
Last update: Jul 10, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$map$pop.map</dt><dd><p>the map data with annotation information.</p>
</dd>
<dt>$map$species</dt><dd><p>the species of genetic map, which can be &quot;arabidopsis&quot;, &quot;cattle&quot;, &quot;chicken&quot;, &quot;dog&quot;, &quot;horse&quot;, &quot;human&quot;, &quot;maize&quot;, &quot;mice&quot;, &quot;pig&quot;, and &quot;rice&quot;.</p>
</dd>
<dt>$map$pop.marker</dt><dd><p>the number of markers.</p>
</dd>
<dt>$map$num.chr</dt><dd><p>the number of chromosomes.</p>
</dd>
<dt>$map$len.chr</dt><dd><p>the length of chromosomes.</p>
</dd>
<dt>$map$qtn.model</dt><dd><p>the genetic model of QTN such as 'A + D'.</p>
</dd>
<dt>$map$qtn.index</dt><dd><p>the QTN index for each trait.</p>
</dd>
<dt>$map$qtn.num</dt><dd><p>the QTN number for (each group in) each trait.</p>
</dd>
<dt>$map$qtn.dist</dt><dd><p>the QTN distribution containing 'norm', 'geom', 'gamma' or 'beta'.</p>
</dd>
<dt>$map$qtn.var</dt><dd><p>the standard deviations for normal distribution.</p>
</dd>
<dt>$map$qtn.prob</dt><dd><p>the probability of success for geometric distribution.</p>
</dd>
<dt>$map$qtn.shape</dt><dd><p>the shape parameter for gamma distribution.</p>
</dd>
<dt>$map$qtn.scale</dt><dd><p>the scale parameter for gamma distribution.</p>
</dd>
<dt>$map$qtn.shape1</dt><dd><p>the shape1 parameter for beta distribution.</p>
</dd>
<dt>$map$qtn.shape2</dt><dd><p>the shape2 parameter for beta distribution.</p>
</dd>
<dt>$map$qtn.ncp</dt><dd><p>the ncp parameter for beta distribution.</p>
</dd>
<dt>$map$qtn.spot</dt><dd><p>the QTN distribution probability in each block.</p>
</dd>
<dt>$map$len.block</dt><dd><p>the block length.</p>
</dd>
<dt>$map$maf</dt><dd><p>the maf threshold, markers less than this threshold will be exclude.</p>
</dd>
<dt>$map$recom.spot</dt><dd><p>whether to generate recombination events.</p>
</dd>
<dt>$map$range.hot</dt><dd><p>the recombination times range in the hot spot.</p>
</dd>
<dt>$map$range.cold</dt><dd><p>the recombination times range in the cold spot.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SP &lt;- param.annot(qtn.num = list(tr1 = 10))
str(SP)
</code></pre>

<hr>
<h2 id='param.geno'>Genotype parameters generator</h2><span id='topic+param.geno'></span>

<h3>Description</h3>

<p>Generate parameters for genotype data simulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.geno(SP = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.geno_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="param.geno_+3A_...">...</code></td>
<td>
<p>one or more parameter(s) for genotype simulation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Feb 21, 2022
Last update: Jul 4, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$geno$pop.geno</dt><dd><p>the genotype data.</p>
</dd>
<dt>$geno$incols</dt><dd><p>'1':one-column genotype represents an individual; '2': two-column genotype represents an individual.</p>
</dd>
<dt>$geno$pop.marker</dt><dd><p>the number of markers.</p>
</dd>
<dt>$geno$pop.ind</dt><dd><p>the number of individuals in the base population.</p>
</dd>
<dt>$geno$prob</dt><dd><p>the genotype code probability.</p>
</dd>
<dt>$geno$rate.mut</dt><dd><p>the mutation rate of the genotype data.</p>
</dd>
<dt>$geno$cld</dt><dd><p>whether to generate a complete LD genotype data when 'incols == 2'.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SP &lt;- param.geno(pop.marker = 1e4, pop.ind = 1e2)
str(SP)
</code></pre>

<hr>
<h2 id='param.global'>Global parameters generator</h2><span id='topic+param.global'></span>

<h3>Description</h3>

<p>Generate parameters for global options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.global(SP = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.global_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="param.global_+3A_...">...</code></td>
<td>
<p>one or more parameter(s) for global options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 16, 2022
Last update: Jul 4, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$replication</dt><dd><p>the replication times of simulation.</p>
</dd>
<dt>$seed.sim</dt><dd><p>simulation random seed.</p>
</dd>
<dt>$out</dt><dd><p>the prefix of output files.</p>
</dd>
<dt>$outpath</dt><dd><p>the path of output files, Simer writes files only if outpath is not 'NULL'.</p>
</dd>
<dt>$out.format</dt><dd><p>'numeric' or 'plink', the data format of output files.</p>
</dd>
<dt>$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$out.geno.gen</dt><dd><p>the output generations of genotype data.</p>
</dd>
<dt>$out.pheno.gen</dt><dd><p>the output generations of phenotype data.</p>
</dd>
<dt>$useAllGeno</dt><dd><p>whether to use all genotype data to simulate phenotype.</p>
</dd>
<dt>$ncpus</dt><dd><p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</dd>
<dt>$verbose</dt><dd><p>whether to print detail.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SP &lt;- param.global(out = "simer")
str(SP)
</code></pre>

<hr>
<h2 id='param.pheno'>Phenotype parameters generator</h2><span id='topic+param.pheno'></span>

<h3>Description</h3>

<p>Generate parameters for phenotype data simulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.pheno(SP = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.pheno_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="param.pheno_+3A_...">...</code></td>
<td>
<p>one or more parameter(s) for phenotype simulation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Feb 21, 2022
Last update: Jul 4, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$pheno$pop</dt><dd><p>the population information containing environmental factors and other effects.</p>
</dd>
<dt>$pheno$pop.ind</dt><dd><p>the number of individuals in the base population.</p>
</dd>
<dt>$pheno$pop.rep</dt><dd><p>the repeated times of repeated records.</p>
</dd>
<dt>$pheno$pop.rep.bal</dt><dd><p>whether repeated records are balanced.</p>
</dd>
<dt>$pheno$pop.env</dt><dd><p>a list of environmental factors setting.</p>
</dd>
<dt>$pheno$phe.type</dt><dd><p>a list of phenotype types.</p>
</dd>
<dt>$pheno$phe.model</dt><dd><p>a list of genetic model of phenotype such as &quot;T1 = A + E&quot;.</p>
</dd>
<dt>$pheno$phe.h2A</dt><dd><p>a list of additive heritability.</p>
</dd>
<dt>$pheno$phe.h2D</dt><dd><p>a list of dominant heritability.</p>
</dd>
<dt>$pheno$phe.h2GxG</dt><dd><p>a list of GxG interaction heritability.</p>
</dd>
<dt>$pheno$phe.h2GxE</dt><dd><p>a list of GxE interaction heritability.</p>
</dd>
<dt>$pheno$phe.h2PE</dt><dd><p>a list of permanent environmental heritability.</p>
</dd>
<dt>$pheno$phe.var</dt><dd><p>a list of phenotype variance.</p>
</dd>
<dt>$pheno$phe.corA</dt><dd><p>the additive genetic correlation matrix.</p>
</dd>
<dt>$pheno$phe.corD</dt><dd><p>the dominant genetic correlation matrix.</p>
</dd>
<dt>$pheno$phe.corGxG</dt><dd><p>the GxG genetic correlation matrix.</p>
</dd>
<dt>$pheno$phe.corPE</dt><dd><p>the permanent environmental correlation matrix.</p>
</dd>
<dt>$pheno$phe.corE</dt><dd><p>the residual correlation matrix.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SP &lt;- param.pheno(phe.model = list(tr1 = "T1 = A + E"))
str(SP)
</code></pre>

<hr>
<h2 id='param.reprod'>Reproduction parameters generator</h2><span id='topic+param.reprod'></span>

<h3>Description</h3>

<p>Generate parameters for reproduction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.reprod(SP = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.reprod_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="param.reprod_+3A_...">...</code></td>
<td>
<p>one or more parameter(s) for reproduction.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 6, 2022
Last update: Jul 4, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the male rate in the population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SP &lt;- param.reprod(reprod.way = "randmate")
str(SP)
</code></pre>

<hr>
<h2 id='param.sel'>Selection parameters generator</h2><span id='topic+param.sel'></span>

<h3>Description</h3>

<p>Generate parameters for selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.sel(SP = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.sel_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="param.sel_+3A_...">...</code></td>
<td>
<p>one or more parameter(s) for selection.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 6, 2022
Last update: Jul 4, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$sel$pop.sel</dt><dd><p>the selected males and females.</p>
</dd>
<dt>$sel$ps</dt><dd><p>if ps &lt;= 1, fraction selected in selection of males and females; if ps &gt; 1, ps is number of selected males and females.</p>
</dd>
<dt>$sel$decr</dt><dd><p>whether the sort order is decreasing.</p>
</dd>
<dt>$sel$sel.crit</dt><dd><p>the selection criteria, it can be 'TBV', 'TGV', and 'pheno'.</p>
</dd>
<dt>$sel$sel.single</dt><dd><p>the single-trait selection method, it can be 'ind', 'fam', 'infam', and 'comb'.</p>
</dd>
<dt>$sel$sel.multi</dt><dd><p>the multiple-trait selection method, it can be 'index', 'indcul', and 'tmd'.</p>
</dd>
<dt>$sel$index.wt</dt><dd><p>the weight of each trait for multiple-trait selection.</p>
</dd>
<dt>$sel$index.tdm</dt><dd><p>the index of tandem selection for multiple-trait selection.</p>
</dd>
<dt>$sel$goal.perc</dt><dd><p>the percentage of goal more than the mean of scores of individuals.</p>
</dd>
<dt>$sel$pass.perc</dt><dd><p>the percentage of expected excellent individuals.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SP &lt;- param.sel(sel.single = "ind")
str(SP)
</code></pre>

<hr>
<h2 id='param.simer'>Parameter generator</h2><span id='topic+param.simer'></span>

<h3>Description</h3>

<p>Generate parameters for Simer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.simer(SP = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.simer_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="param.simer_+3A_...">...</code></td>
<td>
<p>one or more parameter(s) for simer.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 17, 2022
Last update: Jul 4, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$global</dt><dd><p>a list of global parameters.</p>
</dd>
<dt>$map</dt><dd><p>a list of marker information parameters.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
<dt>$sel</dt><dd><p>a list of selection parameters.</p>
</dd>
<dt>$reprod</dt><dd><p>a list of reproduction parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SP &lt;- param.simer(out = "simer")
str(SP)
</code></pre>

<hr>
<h2 id='paste_label'>Pasting label</h2><span id='topic+paste_label'></span>

<h3>Description</h3>

<p>Paste label to a line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>paste_label(line, label, side = "right", margin = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="paste_label_+3A_line">line</code></td>
<td>
<p>long text.</p>
</td></tr>
<tr><td><code id="paste_label_+3A_label">label</code></td>
<td>
<p>short label.</p>
</td></tr>
<tr><td><code id="paste_label_+3A_side">side</code></td>
<td>
<p>&quot;right&quot; or &quot;left&quot;.</p>
</td></tr>
<tr><td><code id="paste_label_+3A_margin">margin</code></td>
<td>
<p>the margin information, default 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Oct 22, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>

<hr>
<h2 id='phenotype'>Phenotype simulation</h2><span id='topic+phenotype'></span>

<h3>Description</h3>

<p>Generate single-trait or multiple-trait phenotype by mixed model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phenotype(SP = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phenotype_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="phenotype_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$pheno$pop</dt><dd><p>the population information containing environmental factors and other effects.</p>
</dd>
<dt>$pheno$pop.ind</dt><dd><p>the number of individuals in the base population.</p>
</dd>
<dt>$pheno$pop.rep</dt><dd><p>the repeated times of repeated records.</p>
</dd>
<dt>$pheno$pop.rep.bal</dt><dd><p>whether repeated records are balanced.</p>
</dd>
<dt>$pheno$pop.env</dt><dd><p>a list of environmental factors setting.</p>
</dd>
<dt>$pheno$phe.type</dt><dd><p>a list of phenotype types.</p>
</dd>
<dt>$pheno$phe.model</dt><dd><p>a list of genetic model of phenotype such as &quot;T1 = A + E&quot;.</p>
</dd>
<dt>$pheno$phe.h2A</dt><dd><p>a list of additive heritability.</p>
</dd>
<dt>$pheno$phe.h2D</dt><dd><p>a list of dominant heritability.</p>
</dd>
<dt>$pheno$phe.h2GxG</dt><dd><p>a list of GxG interaction heritability.</p>
</dd>
<dt>$pheno$phe.h2GxE</dt><dd><p>a list of GxE interaction heritability.</p>
</dd>
<dt>$pheno$phe.h2PE</dt><dd><p>a list of permanent environmental heritability.</p>
</dd>
<dt>$pheno$phe.var</dt><dd><p>a list of phenotype variance.</p>
</dd>
<dt>$pheno$phe.corA</dt><dd><p>the additive genetic correlation matrix.</p>
</dd>
<dt>$pheno$phe.corD</dt><dd><p>the dominant genetic correlation matrix.</p>
</dd>
<dt>$pheno$phe.corGxG</dt><dd><p>the GxG genetic correlation matrix.</p>
</dd>
<dt>$pheno$phe.corPE</dt><dd><p>the permanent environmental correlation matrix.</p>
</dd>
<dt>$pheno$phe.corE</dt><dd><p>the residual correlation matrix.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>References</h3>

<p>Kao C and Zeng Z (2002) &lt;https://www.genetics.org/content/160/3/1243.long&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Prepare environmental factor list
pop.env &lt;- list(
  F1 = list( # fixed effect 1
    level = c("1", "2"),
    effect = list(tr1 = c(50, 30), tr2 = c(50, 30))
  ), 
  F2 = list( # fixed effect 2
    level = c("d1", "d2", "d3"),
    effect = list(tr1 = c(10, 20, 30), tr2 = c(10, 20, 30))
  ),
  C1 = list( # covariate 1
    level = c(70, 80, 90),
    slope = list(tr1 = 1.5, tr2 = 1.5)
  ),
  R1 = list( # random effect 1
    level = c("l1", "l2", "l3"),
    ratio = list(tr1 = 0.1, tr2 = 0.1)
  )
)

# Generate genotype simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = c(2, 8), tr2 = 10),
                  qtn.model = "A + D + A:D")
# Generate annotation simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(
  SP = SP, 
  pop.ind = 100,
  pop.rep = 2, # 2 repeated record
  pop.rep.bal = TRUE, # balanced repeated record
  pop.env = pop.env,
  phe.type = list(
    tr1 = "continuous",
    tr2 = list(case = 0.01, control = 0.99)
  ),
  phe.model = list(
    tr1 = "T1 = A + D + A:D + F1 + F2 + C1 + R1 + A:F1 + E",
    tr2 = "T2 = A + D + A:D + F1 + F2 + C1 + R1 + A:F1 + E"
  ),
  phe.var = list(tr1 = 100, tr2 = 100)
)

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)

</code></pre>

<hr>
<h2 id='pop.geno'>Raw genotype matrix from outside in simdata</h2><span id='topic+pop.geno'></span>

<h3>Description</h3>

<p>Raw genotype matrix from outside in simdata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(simdata)
</code></pre>


<h3>Format</h3>

<p>matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simdata)
dim(pop.geno)
head(pop.geno)
</code></pre>

<hr>
<h2 id='pop.map'>Map file from outside in simdata</h2><span id='topic+pop.map'></span>

<h3>Description</h3>

<p>Map file from outside in simdata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(simdata)
</code></pre>


<h3>Format</h3>

<p>list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simdata)
dim(pop.map)
head(pop.map)
</code></pre>

<hr>
<h2 id='print_accomplished'>Accomplishment</h2><span id='topic+print_accomplished'></span>

<h3>Description</h3>

<p>Print accomplishment information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_accomplished(width = 60, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_accomplished_+3A_width">width</code></td>
<td>
<p>the width of the message.</p>
</td></tr>
<tr><td><code id="print_accomplished_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Aug 30, 2017
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>

<hr>
<h2 id='print_bar'>Progress bar</h2><span id='topic+print_bar'></span>

<h3>Description</h3>

<p>Print progress bar.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_bar(
  i,
  n,
  type = c("type1", "type3"),
  symbol = "-",
  tmp.file = NULL,
  symbol.head = "&gt;&gt;&gt;",
  symbol.tail = "&gt;",
  fixed.points = TRUE,
  points = seq(0, 100, 1),
  symbol.len = 48,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_bar_+3A_i">i</code></td>
<td>
<p>the current loop number.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_n">n</code></td>
<td>
<p>the max loop number.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_type">type</code></td>
<td>
<p>type1 for &quot;for&quot; function.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_symbol">symbol</code></td>
<td>
<p>the symbol for the rate of progress.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_tmp.file">tmp.file</code></td>
<td>
<p>the opened file of &quot;fifo&quot; function.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_symbol.head">symbol.head</code></td>
<td>
<p>the head for the bar.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_symbol.tail">symbol.tail</code></td>
<td>
<p>the tail for the bar.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_fixed.points">fixed.points</code></td>
<td>
<p>whether use the setted points which will be printed.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_points">points</code></td>
<td>
<p>the setted points which will be printed.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_symbol.len">symbol.len</code></td>
<td>
<p>the total length of progress bar.</p>
</td></tr>
<tr><td><code id="print_bar_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Aug 30, 2017
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>

<hr>
<h2 id='print_info'>Simer information</h2><span id='topic+print_info'></span>

<h3>Description</h3>

<p>Print R Package information, include title, short_title, logo, version, authors, contact.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_info(
  welcome = NULL,
  title = NULL,
  short_title = NULL,
  logo = NULL,
  version = NULL,
  authors = NULL,
  contact = NULL,
  linechar = "=",
  width = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_info_+3A_welcome">welcome</code></td>
<td>
<p>welcome text, for example: &quot;Welcom to &lt;Packagename&gt;&quot;.</p>
</td></tr>
<tr><td><code id="print_info_+3A_title">title</code></td>
<td>
<p>long text to introduct package.</p>
</td></tr>
<tr><td><code id="print_info_+3A_short_title">short_title</code></td>
<td>
<p>short label, top-left of logo.</p>
</td></tr>
<tr><td><code id="print_info_+3A_logo">logo</code></td>
<td>
<p>logo.</p>
</td></tr>
<tr><td><code id="print_info_+3A_version">version</code></td>
<td>
<p>short label, bottom-right of logo.</p>
</td></tr>
<tr><td><code id="print_info_+3A_authors">authors</code></td>
<td>
<p>authors of software.</p>
</td></tr>
<tr><td><code id="print_info_+3A_contact">contact</code></td>
<td>
<p>email or website.</p>
</td></tr>
<tr><td><code id="print_info_+3A_linechar">linechar</code></td>
<td>
<p>1, 2, or char.</p>
</td></tr>
<tr><td><code id="print_info_+3A_width">width</code></td>
<td>
<p>banner width.</p>
</td></tr>
<tr><td><code id="print_info_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Oct 22, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>welcome information.
</p>


<h3>Author(s)</h3>

<p>Dong Yin and Haohao Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>welcome &lt;- "Welcome to SIMER"
title   &lt;- "Data Simulation for Life Science and Breeding"
authors &lt;- c("Designed and Maintained by Dong Yin, Xuanning Zhang,
              Lilin Yin, Haohao Zhang, and Xiaolei Liu", 
             "Contributors: Zhenshuang Tang, Jingya Xu, Xinyun Li, 
              Mengjin Zhu, Xiaohui Yuan, Shuhong Zhao")
contact &lt;- "Contact: xiaoleiliu@mail.hzau.edu.cn"
logo_s  &lt;- c(" ____ ___ __  __ _____ ____  ", 
             "/ ___|_ _|  \\/  | ____|  _ \\ ", 
             "\\___ \\| || |\\/| |  _| | |_) |", 
             " ___) | || |  | | |___|  _ &lt; ", 
             "|____/___|_|  |_|_____|_| \\_\\")
print_info(welcome = welcome, title = title, logo = logo_s, authors = authors, 
           contact = contact, linechar = '=', width = 70)
</code></pre>

<hr>
<h2 id='remove_bigmatrix'>Big.matrix removing</h2><span id='topic+remove_bigmatrix'></span>

<h3>Description</h3>

<p>Remove big.matrix safely.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_bigmatrix(x, desc_suffix = ".geno.desc", bin_suffix = ".geno.bin")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_bigmatrix_+3A_x">x</code></td>
<td>
<p>the filename of big.matrix.</p>
</td></tr>
<tr><td><code id="remove_bigmatrix_+3A_desc_suffix">desc_suffix</code></td>
<td>
<p>the suffix of description file of big.matrix.</p>
</td></tr>
<tr><td><code id="remove_bigmatrix_+3A_bin_suffix">bin_suffix</code></td>
<td>
<p>the suffix of binary file of big.matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Aug 8, 2019
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>TRUE or FALSE
</p>


<h3>Author(s)</h3>

<p>Haohao Zhang and Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(bigmemory)
mat &lt;- filebacked.big.matrix(
     nrow = 10,
     ncol = 10,
     init = 0,
     type = 'char',
     backingpath = ".",
     backingfile = 'simer.geno.bin',
     descriptorfile = 'simer.geno.desc')

remove_bigmatrix(x = "simer")

</code></pre>

<hr>
<h2 id='reproduces'>Reproduction</h2><span id='topic+reproduces'></span>

<h3>Description</h3>

<p>Population reproduction by different mate design.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reproduces(SP, ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reproduces_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="reproduces_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="reproduces_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Nov 14, 2018
Last update: Apr 29, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$reprod$pop.gen</dt><dd><p>the generations of simulated population.</p>
</dd>
<dt>$reprod$reprod.way</dt><dd><p>reproduction method, it consists of 'clone', 'dh', 'selfpol', 'randmate', 'randexself', 'assort', 'disassort', '2waycro', '3waycro', '4waycro', 'backcro', and 'userped'.</p>
</dd>
<dt>$reprod$sex.rate</dt><dd><p>the male rate in the population.</p>
</dd>
<dt>$reprod$prog</dt><dd><p>the progeny number of an individual.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")
# Generate reproduction parameters
SP &lt;- param.reprod(SP = SP, reprod.way = "randmate")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)
# Run reproduction
SP &lt;- reproduces(SP)

</code></pre>

<hr>
<h2 id='rule_wrap'>Accomplishment</h2><span id='topic+rule_wrap'></span>

<h3>Description</h3>

<p>Print accomplishment information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rule_wrap(string, width, align = "center", linechar = " ")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rule_wrap_+3A_string">string</code></td>
<td>
<p>a string.</p>
</td></tr>
<tr><td><code id="rule_wrap_+3A_width">width</code></td>
<td>
<p>the width of the message.</p>
</td></tr>
<tr><td><code id="rule_wrap_+3A_align">align</code></td>
<td>
<p>the position of string.</p>
</td></tr>
<tr><td><code id="rule_wrap_+3A_linechar">linechar</code></td>
<td>
<p>char in every line.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Oct 22, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>

<hr>
<h2 id='selects'>Selection</h2><span id='topic+selects'></span>

<h3>Description</h3>

<p>Select individuals by combination of selection method and criterion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selects(SP = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selects_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
<tr><td><code id="selects_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Sep 8, 2018
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$sel$pop.sel</dt><dd><p>the selected males and females.</p>
</dd>
<dt>$sel$ps</dt><dd><p>if ps &lt;= 1, fraction selected in selection of males and females; if ps &gt; 1, ps is number of selected males and females.</p>
</dd>
<dt>$sel$decr</dt><dd><p>whether the sort order is decreasing.</p>
</dd>
<dt>$sel$sel.crit</dt><dd><p>the selection criteria, it can be 'TBV', 'TGV', and 'pheno'.</p>
</dd>
<dt>$sel$sel.single</dt><dd><p>the single-trait selection method, it can be 'ind', 'fam', 'infam', and 'comb'.</p>
</dd>
<dt>$sel$sel.multi</dt><dd><p>the multiple-trait selection method, it can be 'index', 'indcul', and 'tmd'.</p>
</dd>
<dt>$sel$index.wt</dt><dd><p>the weight of each trait for multiple-trait selection.</p>
</dd>
<dt>$sel$index.tdm</dt><dd><p>the index of tandem selection for multiple-trait selection.</p>
</dd>
<dt>$sel$goal.perc</dt><dd><p>the percentage of goal more than the mean of scores of individuals.</p>
</dd>
<dt>$sel$pass.perc</dt><dd><p>the percentage of expected excellent individuals.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate annotation simulation parameters
SP &lt;- param.annot(qtn.num = list(tr1 = 10))
# Generate genotype simulation parameters
SP &lt;- param.geno(SP = SP, pop.marker = 1e4, pop.ind = 1e2)
# Generate phenotype simulation parameters
SP &lt;- param.pheno(SP = SP, pop.ind = 100)
# Generate selection parameters
SP &lt;- param.sel(SP = SP, sel.single = "ind")

# Run annotation simulation
SP &lt;- annotation(SP)
# Run genotype simulation
SP &lt;- genotype(SP)
# Run phenotype simulation
SP &lt;- phenotype(SP)
# Run selection
SP &lt;- selects(SP)

</code></pre>

<hr>
<h2 id='simer'>Simer</h2><span id='topic+simer'></span>

<h3>Description</h3>

<p>Main function of Simer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer(SP)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Jan 7, 2019
Last update: Apr 29, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$global</dt><dd><p>a list of global parameters.</p>
</dd>
<dt>$map</dt><dd><p>a list of marker information parameters.</p>
</dd>
<dt>$geno</dt><dd><p>a list of genotype simulation parameters.</p>
</dd>
<dt>$pheno</dt><dd><p>a list of phenotype simulation parameters.</p>
</dd>
<dt>$sel</dt><dd><p>a list of selection parameters.</p>
</dd>
<dt>$reprod</dt><dd><p>a list of reproduction parameters.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate all simulation parameters
SP &lt;- param.simer(out = "simer")

# Run Simer
SP &lt;- simer(SP)

</code></pre>

<hr>
<h2 id='simer.Data'>Data handling</h2><span id='topic+simer.Data'></span>

<h3>Description</h3>

<p>Make data quality control for genotype, phenotype, and pedigree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data(jsonList = NULL, out = "simer.qc", ncpus = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data_+3A_jsonlist">jsonList</code></td>
<td>
<p>a list of data quality control parameters.</p>
</td></tr>
<tr><td><code id="simer.Data_+3A_out">out</code></td>
<td>
<p>the prefix of output files.</p>
</td></tr>
<tr><td><code id="simer.Data_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: May 26, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$genotype</dt><dd><p>the path of genotype data.</p>
</dd>
<dt>$pedigree</dt><dd><p>the filename of pedigree data.</p>
</dd>
<dt>$selection_index</dt><dd><p>the selection index for all traits.</p>
</dd>
<dt>$breeding_value_index</dt><dd><p>the breeding value index for all traits.</p>
</dd>
<dt>$quality_control_plan</dt><dd><p>a list of parameters for data quality control.</p>
</dd>
<dt>$breeding_plan</dt><dd><p>a list of parameters for genetic evaluation.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read JSON file
jsonFile &lt;- system.file("extdata", "04breeding_plan", "plan1.json", package = "simer")
jsonList &lt;- jsonlite::fromJSON(txt = jsonFile, simplifyVector = FALSE)

## Not run: 
# It needs 'plink' and 'hiblup' software
jsonList &lt;- simer.Data(jsonList = jsonList)

## End(Not run)
</code></pre>

<hr>
<h2 id='simer.Data.Bfile2MVP'>simer.Data.Bfile2MVP: To transform plink binary data to MVP package</h2><span id='topic+simer.Data.Bfile2MVP'></span>

<h3>Description</h3>

<p>transforming plink binary data to MVP package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Bfile2MVP(
  bfile,
  out = "simer",
  maxLine = 10000,
  priority = "speed",
  type.geno = "char",
  threads = 10,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Bfile2MVP_+3A_bfile">bfile</code></td>
<td>
<p>Genotype in binary format (.bed, .bim, .fam).</p>
</td></tr>
<tr><td><code id="simer.Data.Bfile2MVP_+3A_out">out</code></td>
<td>
<p>the name of output file.</p>
</td></tr>
<tr><td><code id="simer.Data.Bfile2MVP_+3A_maxline">maxLine</code></td>
<td>
<p>the max number of line to write to big matrix for each loop.</p>
</td></tr>
<tr><td><code id="simer.Data.Bfile2MVP_+3A_priority">priority</code></td>
<td>
<p>'memory' or 'speed'.</p>
</td></tr>
<tr><td><code id="simer.Data.Bfile2MVP_+3A_type.geno">type.geno</code></td>
<td>
<p>the type of genotype elements.</p>
</td></tr>
<tr><td><code id="simer.Data.Bfile2MVP_+3A_threads">threads</code></td>
<td>
<p>number of thread for transforming.</p>
</td></tr>
<tr><td><code id="simer.Data.Bfile2MVP_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Sep 12, 2018
Last update: July 25, 2022
</p>


<h3>Value</h3>

<p>number of individuals and markers.
Output files:
genotype.desc, genotype.bin: genotype file in bigmemory format
phenotype.phe: ordered phenotype file, same taxa order with genotype file
map.map: SNP information
</p>


<h3>Author(s)</h3>

<p>Haohao Zhang and Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get bfile path
bfilePath &lt;- file.path(system.file("extdata", "02plinkb", package = "simer"), "demo")

# Data converting
simer.Data.Bfile2MVP(bfilePath, tempfile("outfile"))

</code></pre>

<hr>
<h2 id='simer.Data.cHIBLUP'>Genetic evaluation</h2><span id='topic+simer.Data.cHIBLUP'></span>

<h3>Description</h3>

<p>The function of calling HIBLUP software of C version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.cHIBLUP(
  jsonList = NULL,
  hiblupPath = "",
  mode = "A",
  vc.method = "AI",
  ncpus = 10,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.cHIBLUP_+3A_jsonlist">jsonList</code></td>
<td>
<p>the list of genetic evaluation parameters.</p>
</td></tr>
<tr><td><code id="simer.Data.cHIBLUP_+3A_hibluppath">hiblupPath</code></td>
<td>
<p>the path of HIBLUP software.</p>
</td></tr>
<tr><td><code id="simer.Data.cHIBLUP_+3A_mode">mode</code></td>
<td>
<p>'A' or 'AD', Additive effect model or Additive and Dominance model.</p>
</td></tr>
<tr><td><code id="simer.Data.cHIBLUP_+3A_vc.method">vc.method</code></td>
<td>
<p>default is 'AI', the method of calculating variance components in HIBLUP software.</p>
</td></tr>
<tr><td><code id="simer.Data.cHIBLUP_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data.cHIBLUP_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: June 28, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$randList</dt><dd><p>a list of estimated random effects.</p>
</dd>
<dt>$varList</dt><dd><p>a list of variance components.</p>
</dd>
<dt>$covA</dt><dd><p>the genetic covariance matrix for all traits.</p>
</dd>
<dt>$corA</dt><dd><p>the genetic correlation matrix for all traits.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read JSON file
jsonFile &lt;- system.file("extdata", "04breeding_plan", "plan1.json", package = "simer")
jsonList &lt;- jsonlite::fromJSON(txt = jsonFile, simplifyVector = FALSE)

## Not run: 
# It needs 'hiblup' software
gebvs &lt;- simer.Data.cHIBLUP(jsonList = jsonList)

## End(Not run)
</code></pre>

<hr>
<h2 id='simer.Data.Env'>Environmental factor selection</h2><span id='topic+simer.Data.Env'></span>

<h3>Description</h3>

<p>To find appropriate fixed effects, covariates, and random effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Env(
  jsonList = NULL,
  hiblupPath = "",
  header = TRUE,
  sep = "\t",
  ncpus = 10,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Env_+3A_jsonlist">jsonList</code></td>
<td>
<p>the list of environmental factor selection parameters.</p>
</td></tr>
<tr><td><code id="simer.Data.Env_+3A_hibluppath">hiblupPath</code></td>
<td>
<p>the path of HIBLUP software.</p>
</td></tr>
<tr><td><code id="simer.Data.Env_+3A_header">header</code></td>
<td>
<p>the header of file.</p>
</td></tr>
<tr><td><code id="simer.Data.Env_+3A_sep">sep</code></td>
<td>
<p>the separator of file.</p>
</td></tr>
<tr><td><code id="simer.Data.Env_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data.Env_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: July 17, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$genotype</dt><dd><p>the path of genotype data.</p>
</dd>
<dt>$pedigree</dt><dd><p>the filename of pedigree data.</p>
</dd>
<dt>$selection_index</dt><dd><p>the selection index for all traits.</p>
</dd>
<dt>$breeding_value_index</dt><dd><p>the breeding value index for all traits.</p>
</dd>
<dt>$quality_control_plan</dt><dd><p>a list of parameters for data quality control.</p>
</dd>
<dt>$breeding_plan</dt><dd><p>a list of parameters for genetic evaluation.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read JSON file
jsonFile &lt;- system.file("extdata", "04breeding_plan", "plan1.json", package = "simer")
jsonList &lt;- jsonlite::fromJSON(txt = jsonFile, simplifyVector = FALSE)

## Not run: 
# It needs 'hiblup' solfware
jsonList &lt;- simer.Data.Env(jsonList = jsonList)

## End(Not run)
</code></pre>

<hr>
<h2 id='simer.Data.Geno'>Genotype data quality control</h2><span id='topic+simer.Data.Geno'></span>

<h3>Description</h3>

<p>Data quality control for genotype data in MVP format and PLINK format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Geno(
  fileMVP = NULL,
  fileBed = NULL,
  filePlinkPed = NULL,
  filePed = NULL,
  filePhe = NULL,
  out = "simer.qc",
  genoType = "char",
  filter = NULL,
  filterGeno = NULL,
  filterHWE = NULL,
  filterMind = NULL,
  filterMAF = NULL,
  ncpus = 0,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Geno_+3A_filemvp">fileMVP</code></td>
<td>
<p>genotype in MVP format.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_filebed">fileBed</code></td>
<td>
<p>genotype in PLINK binary format.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_fileplinkped">filePlinkPed</code></td>
<td>
<p>genotype in PLINK numeric format.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_fileped">filePed</code></td>
<td>
<p>the filename of pedigree data.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_filephe">filePhe</code></td>
<td>
<p>the filename of phenotype data, it can be a vector.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_out">out</code></td>
<td>
<p>the prefix of output files.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_genotype">genoType</code></td>
<td>
<p>type parameter in bigmemory, genotype data. The default is char, it is highly recommended *NOT* to modify this parameter.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_filter">filter</code></td>
<td>
<p>filter of genotyped individual.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_filtergeno">filterGeno</code></td>
<td>
<p>threshold of sample miss rate.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_filterhwe">filterHWE</code></td>
<td>
<p>threshold of Hardy-Weinberg Test.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_filtermind">filterMind</code></td>
<td>
<p>threshold of variant miss rate.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_filtermaf">filterMAF</code></td>
<td>
<p>threshold of Minor Allele Frequency.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data.Geno_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: May 26, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns files
</p>

<dl>
<dt>&lt;out&gt;.bed</dt><dd><p>the .bed file of PLINK binary format.</p>
</dd>
<dt>&lt;out&gt;.bim</dt><dd><p>the .bim file of PLINK binary format.</p>
</dd>
<dt>&lt;out&gt;.fam</dt><dd><p>the .fam file of PLINK binary format.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the prefix of genotype data
fileBed &lt;- system.file("extdata", "02plinkb", "demo", package = "simer")

## Not run: 
# It needs 'plink' software
simer.Data.Geno(fileBed=fileBed)

## End(Not run)
</code></pre>

<hr>
<h2 id='simer.Data.Impute'>Genotype data imputation</h2><span id='topic+simer.Data.Impute'></span>

<h3>Description</h3>

<p>Impute the missing value within genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Impute(
  fileMVP = NULL,
  fileBed = NULL,
  out = NULL,
  maxLine = 10000,
  ncpus = 0,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Impute_+3A_filemvp">fileMVP</code></td>
<td>
<p>genotype in MVP format.</p>
</td></tr>
<tr><td><code id="simer.Data.Impute_+3A_filebed">fileBed</code></td>
<td>
<p>genotype in PLINK binary format.</p>
</td></tr>
<tr><td><code id="simer.Data.Impute_+3A_out">out</code></td>
<td>
<p>the name of output file.</p>
</td></tr>
<tr><td><code id="simer.Data.Impute_+3A_maxline">maxLine</code></td>
<td>
<p>number of SNPs, only used for saving memory when calculate kinship matrix.</p>
</td></tr>
<tr><td><code id="simer.Data.Impute_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data.Impute_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: May 26, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns files
</p>

<dl>
<dt>&lt;out&gt;.geno.desc</dt><dd><p>the description file of genotype data.</p>
</dd>
<dt>&lt;out&gt;.geno.bin</dt><dd><p>the binary file of genotype data.</p>
</dd>
<dt>&lt;out&gt;.geno.ind</dt><dd><p>the genotyped individual file.</p>
</dd>
<dt>&lt;out&gt;.geno.map</dt><dd><p>the marker information data file.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the prefix of genotype data
fileMVP &lt;- system.file("extdata", "02plinkb", "demo", package = "simer")

## Not run: 
# It needs 'beagle' software
fileMVPimp &lt;- simer.Data.Impute(fileBed = fileBed)

## End(Not run)
</code></pre>

<hr>
<h2 id='simer.Data.Json'>Data quality control</h2><span id='topic+simer.Data.Json'></span>

<h3>Description</h3>

<p>Make data quality control by JSON file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Json(
  jsonFile,
  hiblupPath = "",
  out = "simer.qc",
  dataQC = TRUE,
  buildModel = TRUE,
  buildIndex = TRUE,
  ncpus = 10,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Json_+3A_jsonfile">jsonFile</code></td>
<td>
<p>the path of JSON file.</p>
</td></tr>
<tr><td><code id="simer.Data.Json_+3A_hibluppath">hiblupPath</code></td>
<td>
<p>the path of HIBLUP software.</p>
</td></tr>
<tr><td><code id="simer.Data.Json_+3A_out">out</code></td>
<td>
<p>the prefix of output files.</p>
</td></tr>
<tr><td><code id="simer.Data.Json_+3A_dataqc">dataQC</code></td>
<td>
<p>whether to make data quality control.</p>
</td></tr>
<tr><td><code id="simer.Data.Json_+3A_buildmodel">buildModel</code></td>
<td>
<p>whether to build EBV model.</p>
</td></tr>
<tr><td><code id="simer.Data.Json_+3A_buildindex">buildIndex</code></td>
<td>
<p>whether to build Selection Index.</p>
</td></tr>
<tr><td><code id="simer.Data.Json_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data.Json_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Oct 19, 2020
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$genotype</dt><dd><p>the path of genotype data.</p>
</dd>
<dt>$pedigree</dt><dd><p>the filename of pedigree data.</p>
</dd>
<dt>$selection_index</dt><dd><p>the selection index for all traits.</p>
</dd>
<dt>$breeding_value_index</dt><dd><p>the breeding value index for all traits.</p>
</dd>
<dt>$quality_control_plan</dt><dd><p>a list of parameters for data quality control.</p>
</dd>
<dt>$breeding_plan</dt><dd><p>a list of parameters for genetic evaluation.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get JSON file
jsonFile &lt;- system.file("extdata", "04breeding_plan", "plan1.json", package = "simer")

## Not run: 
# It needs 'plink' and 'hiblup' software
jsonList &lt;- simer.Data.Json(jsonFile = jsonFile)

## End(Not run)
</code></pre>

<hr>
<h2 id='simer.Data.Kin'>simer.Data.EMMA: To construct EMMA kinship matrix</h2><span id='topic+simer.Data.Kin'></span>

<h3>Description</h3>

<p>constructing EMMA kinship matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Kin(
  fileKin = TRUE,
  fileMVP = "simer",
  out = NULL,
  method = "EMMA",
  sep = "\t",
  threads = 10,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Kin_+3A_filekin">fileKin</code></td>
<td>
<p>kinship that represents relationship among individuals, n * n matrix, n is sample size.</p>
</td></tr>
<tr><td><code id="simer.Data.Kin_+3A_filemvp">fileMVP</code></td>
<td>
<p>prefix for mvp format files.</p>
</td></tr>
<tr><td><code id="simer.Data.Kin_+3A_out">out</code></td>
<td>
<p>prefix of output file name.</p>
</td></tr>
<tr><td><code id="simer.Data.Kin_+3A_method">method</code></td>
<td>
<p>only&quot;EMMA&quot; method for now.</p>
</td></tr>
<tr><td><code id="simer.Data.Kin_+3A_sep">sep</code></td>
<td>
<p>seperator for Kinship file.</p>
</td></tr>
<tr><td><code id="simer.Data.Kin_+3A_threads">threads</code></td>
<td>
<p>the number of cpu.</p>
</td></tr>
<tr><td><code id="simer.Data.Kin_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Apr 19, 2023
Last update: Apr 19, 2023
</p>


<h3>Value</h3>

<p>Output file:
&lt;out&gt;.kin.bin
&lt;out&gt;.kin.desc
</p>


<h3>Author(s)</h3>

<p>Haohao Zhang and Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get the prefix of genotype data
fileMVP &lt;- system.file("extdata", "01bigmemory", "demo", package = "simer")

# Check map data
simer.Data.Kin(fileKin = TRUE, fileMVP = fileMVP, out = tempfile("outfile"))

</code></pre>

<hr>
<h2 id='simer.Data.Map'>simer.Data.Map: To check map file</h2><span id='topic+simer.Data.Map'></span>

<h3>Description</h3>

<p>checking map file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Map(
  map,
  out = "simer",
  cols = 1:5,
  header = TRUE,
  sep = "\t",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Map_+3A_map">map</code></td>
<td>
<p>the name of map file or map object(data.frame or matrix).</p>
</td></tr>
<tr><td><code id="simer.Data.Map_+3A_out">out</code></td>
<td>
<p>the name of output file.</p>
</td></tr>
<tr><td><code id="simer.Data.Map_+3A_cols">cols</code></td>
<td>
<p>selected columns.</p>
</td></tr>
<tr><td><code id="simer.Data.Map_+3A_header">header</code></td>
<td>
<p>whether the file contains header.</p>
</td></tr>
<tr><td><code id="simer.Data.Map_+3A_sep">sep</code></td>
<td>
<p>seperator of the file.</p>
</td></tr>
<tr><td><code id="simer.Data.Map_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Sep 12, 2018
Last update: July 25, 2022
</p>


<h3>Value</h3>

<p>Output file:
&lt;out&gt;.map
</p>


<h3>Author(s)</h3>

<p>Haohao Zhang and Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get map path
mapPath &lt;- system.file("extdata", "01bigmemory", "demo.geno.map", package = "simer")

# Check map data
simer.Data.Map(mapPath, tempfile("outfile"))
</code></pre>

<hr>
<h2 id='simer.Data.MVP2Bfile'>simer.Data.MVP2Bfile: To transform MVP data to binary format</h2><span id='topic+simer.Data.MVP2Bfile'></span>

<h3>Description</h3>

<p>transforming MVP data to binary format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.MVP2Bfile(
  bigmat,
  map,
  pheno = NULL,
  out = "simer",
  threads = 10,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.MVP2Bfile_+3A_bigmat">bigmat</code></td>
<td>
<p>Genotype in bigmatrix format (0,1,2).</p>
</td></tr>
<tr><td><code id="simer.Data.MVP2Bfile_+3A_map">map</code></td>
<td>
<p>the map file.</p>
</td></tr>
<tr><td><code id="simer.Data.MVP2Bfile_+3A_pheno">pheno</code></td>
<td>
<p>the phenotype file.</p>
</td></tr>
<tr><td><code id="simer.Data.MVP2Bfile_+3A_out">out</code></td>
<td>
<p>the name of output file.</p>
</td></tr>
<tr><td><code id="simer.Data.MVP2Bfile_+3A_threads">threads</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data.MVP2Bfile_+3A_verbose">verbose</code></td>
<td>
<p>whether to print the reminder.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Sep 12, 2018
Last update: July 20, 2022
</p>


<h3>Value</h3>

<p>NULL
Output files:
.bed, .bim, .fam
</p>


<h3>Author(s)</h3>

<p>Haohao Zhang and Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate bigmat and map
bigmat &lt;- as.big.matrix(matrix(1:6, 3, 2))
map &lt;- generate.map(pop.marker = 3)

# Data converting
simer.Data.MVP2Bfile(bigmat, map, out=tempfile("outfile"))

</code></pre>

<hr>
<h2 id='simer.Data.MVP2MVP'>Genotype data conversion</h2><span id='topic+simer.Data.MVP2MVP'></span>

<h3>Description</h3>

<p>Convert genotype data from MVP format to MVP format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.MVP2MVP(fileMVP, genoType = "char", out = "simer", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.MVP2MVP_+3A_filemvp">fileMVP</code></td>
<td>
<p>the prefix of MVP file.</p>
</td></tr>
<tr><td><code id="simer.Data.MVP2MVP_+3A_genotype">genoType</code></td>
<td>
<p>type parameter in bigmemory data. The default is 'char', it is highly recommended *NOT* to modify this parameter.</p>
</td></tr>
<tr><td><code id="simer.Data.MVP2MVP_+3A_out">out</code></td>
<td>
<p>the prefix of output files.</p>
</td></tr>
<tr><td><code id="simer.Data.MVP2MVP_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: May 26, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns files
</p>

<dl>
<dt>&lt;out&gt;.geno.desc</dt><dd><p>the description file of genotype data.</p>
</dd>
<dt>&lt;out&gt;.geno.bin</dt><dd><p>the binary file of genotype data.</p>
</dd>
<dt>&lt;out&gt;.geno.ind</dt><dd><p>the genotyped individual file.</p>
</dd>
<dt>&lt;out&gt;.geno.map</dt><dd><p>the marker information data file.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get the prefix of genotype data
fileMVP &lt;- system.file("extdata", "01bigmemory", "demo", package = "simer")

# Convert genotype data from MVP to MVP
simer.Data.MVP2MVP(fileMVP, out = tempfile("outfile"))

</code></pre>

<hr>
<h2 id='simer.Data.Ped'>Pedigree data quality control</h2><span id='topic+simer.Data.Ped'></span>

<h3>Description</h3>

<p>Data quality control for pedigree data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Ped(
  filePed,
  fileMVP = NULL,
  out = NULL,
  standardID = FALSE,
  fileSir = NULL,
  fileDam = NULL,
  exclThres = 0.1,
  assignThres = 0.05,
  header = TRUE,
  sep = "\t",
  ncpus = 0,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Ped_+3A_fileped">filePed</code></td>
<td>
<p>the filename of pedigree need correcting.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_filemvp">fileMVP</code></td>
<td>
<p>genotype in MVP format.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_out">out</code></td>
<td>
<p>the prefix of output file.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_standardid">standardID</code></td>
<td>
<p>whether kid id is 15-character standard.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_filesir">fileSir</code></td>
<td>
<p>the filename of candidate sires.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_filedam">fileDam</code></td>
<td>
<p>the filename of candidate dams.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_exclthres">exclThres</code></td>
<td>
<p>if conflict ratio is more than exclThres, exclude this parent.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_assignthres">assignThres</code></td>
<td>
<p>if conflict ratio is less than assignThres, assign this parent to the individual.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_header">header</code></td>
<td>
<p>whether the file contains header.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_sep">sep</code></td>
<td>
<p>separator of the file.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data.Ped_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: May 6, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns files
</p>

<dl>
<dt>&lt;out&gt;.ped.report</dt><dd><p>the report file containing correction condition.</p>
</dd>
<dt>&lt;out&gt;.ped.error</dt><dd><p>the file containing pedigree error.</p>
</dd>
<dt>&lt;out&gt;.ped</dt><dd><p>the pedigree file after correction.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Lilin Yin and Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get the filename of pedigree data
filePed &lt;- system.file("extdata", "05others", "pedigree.txt", package = "simer")

# Get the prefix of genotype data
fileMVP &lt;- system.file("extdata", "01bigmemory", "demo", package = "simer")

# Run pedigree correction
simer.Data.Ped(filePed = filePed, fileMVP = fileMVP, out = tempfile("outfile"))

</code></pre>

<hr>
<h2 id='simer.Data.Pheno'>Phenotype data quality control</h2><span id='topic+simer.Data.Pheno'></span>

<h3>Description</h3>

<p>Data quality control for phenotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.Pheno(
  filePhe = NULL,
  filePed = NULL,
  out = NULL,
  planPhe = NULL,
  pheCols = NULL,
  header = TRUE,
  sep = "\t",
  missing = c(NA, "NA", "Na", ".", "-", "NAN", "nan", "na", "N/A", "n/a", "&lt;NA&gt;", "",
    "-9", 9999),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.Pheno_+3A_filephe">filePhe</code></td>
<td>
<p>the phenotype files, it can be a vector.</p>
</td></tr>
<tr><td><code id="simer.Data.Pheno_+3A_fileped">filePed</code></td>
<td>
<p>the pedigree files, it can be a vector.</p>
</td></tr>
<tr><td><code id="simer.Data.Pheno_+3A_out">out</code></td>
<td>
<p>the prefix of output file.</p>
</td></tr>
<tr><td><code id="simer.Data.Pheno_+3A_planphe">planPhe</code></td>
<td>
<p>the plans for phenotype quality control.</p>
</td></tr>
<tr><td><code id="simer.Data.Pheno_+3A_phecols">pheCols</code></td>
<td>
<p>the column needing extracting.</p>
</td></tr>
<tr><td><code id="simer.Data.Pheno_+3A_header">header</code></td>
<td>
<p>the header of file.</p>
</td></tr>
<tr><td><code id="simer.Data.Pheno_+3A_sep">sep</code></td>
<td>
<p>the separator of file.</p>
</td></tr>
<tr><td><code id="simer.Data.Pheno_+3A_missing">missing</code></td>
<td>
<p>the missing value.</p>
</td></tr>
<tr><td><code id="simer.Data.Pheno_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: June 13, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns files
</p>

<dl>
<dt>&lt;out&gt;.phe</dt><dd><p>the phenotype file after correction.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Haohao Zhang and Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the filename of phenotype data
filePhe &lt;- system.file("extdata", "05others", "phenotype.txt", package = "simer")

# Run phenotype correction
simer.Data.Pheno(filePhe = filePhe, out = tempfile("outfile"))
</code></pre>

<hr>
<h2 id='simer.Data.SELIND'>Selection index construction</h2><span id='topic+simer.Data.SELIND'></span>

<h3>Description</h3>

<p>The function of General Selection Index.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Data.SELIND(jsonList = NULL, hiblupPath = "", ncpus = 10, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Data.SELIND_+3A_jsonlist">jsonList</code></td>
<td>
<p>the list of selection index construction parameters.</p>
</td></tr>
<tr><td><code id="simer.Data.SELIND_+3A_hibluppath">hiblupPath</code></td>
<td>
<p>the path of HIBLUP software.</p>
</td></tr>
<tr><td><code id="simer.Data.SELIND_+3A_ncpus">ncpus</code></td>
<td>
<p>the number of threads used, if NULL, (logical core number - 1) is automatically used.</p>
</td></tr>
<tr><td><code id="simer.Data.SELIND_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Aug 26, 2021
Last update: Apr 28, 2022
</p>


<h3>Value</h3>

<p>the function returns a list containing
</p>

<dl>
<dt>$genotype</dt><dd><p>the path of genotype data.</p>
</dd>
<dt>$pedigree</dt><dd><p>the filename of pedigree data.</p>
</dd>
<dt>$selection_index</dt><dd><p>the selection index for all traits.</p>
</dd>
<dt>$breeding_value_index</dt><dd><p>the breeding value index for all traits.</p>
</dd>
<dt>$quality_control_plan</dt><dd><p>a list of parameters for data quality control.</p>
</dd>
<dt>$breeding_plan</dt><dd><p>a list of parameters for genetic evaluation.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>References</h3>

<p>Y. S. Chen, Z. L. Sheng (1988) The Theory of General Selection Index. Genetic Report, 15(3): P185-P190
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read JSON file
jsonFile &lt;- system.file("extdata", "04breeding_plan", "plan1.json", package = "simer")
jsonList &lt;- jsonlite::fromJSON(txt = jsonFile, simplifyVector = FALSE)

## Not run: 
# It needs 'hiblup' software
jsonList &lt;- simer.Data.SELIND(jsonList = jsonList)

## End(Not run)
</code></pre>

<hr>
<h2 id='simer.Version'>Simer version</h2><span id='topic+simer.Version'></span>

<h3>Description</h3>

<p>Print simer version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simer.Version(width = 60, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simer.Version_+3A_width">width</code></td>
<td>
<p>the width of the message.</p>
</td></tr>
<tr><td><code id="simer.Version_+3A_verbose">verbose</code></td>
<td>
<p>whether to print detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Aug 30, 2017
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>version number.
</p>


<h3>Author(s)</h3>

<p>Dong Yin, Lilin Yin, Haohao Zhang, and Xiaolei Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simer.Version()
</code></pre>

<hr>
<h2 id='write.file'>File writing</h2><span id='topic+write.file'></span>

<h3>Description</h3>

<p>Write files of Simer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.file(SP)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.file_+3A_sp">SP</code></td>
<td>
<p>a list of all simulation parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Build date: Jan 7, 2019
Last update: Apr 30, 2022
</p>


<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>Dong Yin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
outpath &lt;- tempdir()
SP &lt;- param.simer(out = "simer")
SP &lt;- simer(SP)
SP$global$outpath &lt;- outpath
write.file(SP)
unlink(file.path(outpath, "180_Simer_Data_numeric"), recursive = TRUE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
