<!DOCTYPE html><html lang="en"><head><title>Help for package SIMICO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SIMICO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fd_term'>
<p>d/d_theta_l</p></a></li>
<li><a href='#gamma_fd'>
<p>d/d_gamma_l</p></a></li>
<li><a href='#gamma_off'>
<p>d^2/d_gamma_ldgamma_m</p></a></li>
<li><a href='#gamma_on'>
<p>d^2/d_gamma_ldgamma_l</p></a></li>
<li><a href='#gammasigma'>
<p>d^2/d_gamma_ldsigma^2</p></a></li>
<li><a href='#gammatheta'>
<p>d^2/d_gamma_kdtheta_k</p></a></li>
<li><a href='#gammatheta_off'>
<p>d^2/d_gamma_ldtheta_m</p></a></li>
<li><a href='#get_A'>
<p>Get A vector</p></a></li>
<li><a href='#Get_CausalSNPs_bynum'>
<p>Get_CausalSNPs_bynum()</p></a></li>
<li><a href='#haz_left'>
<p>H_ik(L_ik)</p></a></li>
<li><a href='#haz_right'>
<p>H_ik(R_ik)</p></a></li>
<li><a href='#sd_off'>
<p>d^2/d_theta_kdsigma^2</p></a></li>
<li><a href='#sd_on'>
<p>d^2/dsigma^2^2</p></a></li>
<li><a href='#sim_gmat'>
<p>Simulate genetic matrix.</p></a></li>
<li><a href='#simico_fit_null'>
<p>simico_fit_null()</p></a></li>
<li><a href='#simico_gen_dat'>
<p>simico_gen_dat()</p></a></li>
<li><a href='#simico_out'>
<p>Get P-Values</p></a></li>
<li><a href='#ss_fd'>
<p>d/d_sigma^2</p></a></li>
<li><a href='#ss_sd'>
<p>d^2/d_sigma^2^2</p></a></li>
<li><a href='#st_off'>
<p>d^2/d_theta_ldsigma^2</p></a></li>
<li><a href='#surv_diff'>
<p>S_ik(L_ik) - S_ik(R_ik)</p></a></li>
<li><a href='#surv_left'>
<p>S_ik(L_ik)</p></a></li>
<li><a href='#surv_right'>
<p>S_ik(R_ik)</p></a></li>
<li><a href='#without_one_phen'>
<p>Survival Difference Product without Outcome l</p></a></li>
<li><a href='#without_two_phen'>
<p>Survival Difference Product without Outcomes l and m</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Set-Based Inference for Multiple Interval-Censored Outcomes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains tests for association between a set of genetic variants and multiple correlated outcomes that are interval censored. Interval-censored data arises when the exact time of the onset of an outcome of interest is unknown but known to fall between two time points.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>bindata, fastGHQuad, CompQuadForm, stats, ICSKAT</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-25 22:49:29 UTC; jaihe2</td>
</tr>
<tr>
<td>Author:</td>
<td>Jaihee Choi [aut, cre],
  Ryan Sun [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jaihee Choi &lt;jaiheechoi01@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-25 23:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='fd_term'>
d/d_theta_l
</h2><span id='topic+fd_term'></span>

<h3>Description</h3>

<p>Calculate the first derivative of the theta terms for outcome l.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fd_term(l, temp_beta, phen,d, apply_diffs,
   A_i, no_l_all,HL_array, HR_array)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fd_term_+3A_l">l</code></td>
<td>
<p>Outcome of interest.</p>
</td></tr>
<tr><td><code id="fd_term_+3A_temp_beta">temp_beta</code></td>
<td>
<p>Vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="fd_term_+3A_phen">phen</code></td>
<td>
<p>list containing the covariate design matrices.</p>
</td></tr>
<tr><td><code id="fd_term_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
<tr><td><code id="fd_term_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="fd_term_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes k summed over all quadrature nodes d.</p>
</td></tr>
<tr><td><code id="fd_term_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="fd_term_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="fd_term_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a 1 x (p + 2) vector of the first derivative terms for outcome l.
</p>

<hr>
<h2 id='gamma_fd'>
d/d_gamma_l
</h2><span id='topic+gamma_fd'></span>

<h3>Description</h3>

<p>Calculates the gradient term for U_g for the score statistic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_fd(l, HL_array, HR_array, tpos_all, obs_all,
   temp_beta, A_i, no_l_all, gMat, a1, a2, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gamma_fd_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_tpos_all">tpos_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is left-censored or not.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_obs_all">obs_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is right-censored or not.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_temp_beta">temp_beta</code></td>
<td>
<p>Vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes k summed over all quadrature nodes d.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (K - 1) matrix containing the product of apply_diffs across all outcomes K excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_gmat">gMat</code></td>
<td>
<p>n x q matrix of genetic information.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_a1">a1</code></td>
<td>
<p>First shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_a2">a2</code></td>
<td>
<p>Second shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gamma_fd_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector containing the first derivative with respect to gamma.
</p>

<hr>
<h2 id='gamma_off'>
d^2/d_gamma_ldgamma_m
</h2><span id='topic+gamma_off'></span>

<h3>Description</h3>

<p>Calculates the [off-diagonal] Information matrix term for I_gamma gamma with respect to outcome l and outcome m.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_off(l, m, HL_array, HR_array,
   tpos_all, obs_all, temp_beta, A_i,
   no_l_all, no_two_all, gMat, a1, a2, k, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gamma_off_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_m">m</code></td>
<td>
<p>Index of second outcome of interest.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_tpos_all">tpos_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is left-censored or not.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_obs_all">obs_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is right-censored or not.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_temp_beta">temp_beta</code></td>
<td>
<p>Vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes k summed over all quadrature nodes d.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_no_two_all">no_two_all</code></td>
<td>
<p>n x (k - 2) matrix containing the product of apply_diffs across all outcomes k excluding the outcomes l and m.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_gmat">gMat</code></td>
<td>
<p>n x q matrix of genetic information.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_a1">a1</code></td>
<td>
<p>First shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_a2">a2</code></td>
<td>
<p>Second shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="gamma_off_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a matrix containing the component of the information matrix of the gamma parameter for outcomes l and m.
</p>

<hr>
<h2 id='gamma_on'>
d^2/d_gamma_ldgamma_l
</h2><span id='topic+gamma_on'></span>

<h3>Description</h3>

<p>Calculates the [on-diagonal] Information matrix term for I_gamma gamma with respect to outcome l.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamma_on(l, HL_array, HR_array, tpos_all, obs_all,
   temp_beta, A_i, no_l_all, gMat, a1, a2, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gamma_on_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_tpos_all">tpos_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is left-censored or not.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_obs_all">obs_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is right-censored or not.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_temp_beta">temp_beta</code></td>
<td>
<p>Vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes k summed over all quadrature nodes d.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_gmat">gMat</code></td>
<td>
<p>n x q matrix of genetic information.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_a1">a1</code></td>
<td>
<p>First shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_a2">a2</code></td>
<td>
<p>Second shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gamma_on_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a matrix containing the component of the information matrix of the gamma parameter for outcome l.
</p>

<hr>
<h2 id='gammasigma'>
d^2/d_gamma_ldsigma^2
</h2><span id='topic+gammasigma'></span>

<h3>Description</h3>

<p>Calculates the Information matrix term of I_eta gamma for one outcome of interest l.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gammasigma(
  l, HL_array, HR_array, tpos_all, obs_all,
  apply_diffs, temp_beta, A_i, xDats, no_l_all,
  no_two_all, gMat, a1, a2, k, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gammasigma_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_hl_array">HL_array</code></td>
<td>
<p>n x K matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_hr_array">HR_array</code></td>
<td>
<p>n x K matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_tpos_all">tpos_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is left-censored or not.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_obs_all">obs_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is right-censored or not.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes K summed over all quadrature nodes D.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_xdats">xDats</code></td>
<td>
<p>List of design matrices.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes K excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_no_two_all">no_two_all</code></td>
<td>
<p>n x (k - 2) matrix containing the product of apply_diffs across all outcomes k excluding the outcomes l and m.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_gmat">gMat</code></td>
<td>
<p>n x q matrix of genetic information.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_a1">a1</code></td>
<td>
<p>First shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_a2">a2</code></td>
<td>
<p>Second shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="gammasigma_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a matrix containing the component of the information matrix of the gamma and sigma^2 parameters for outcome l.
</p>

<hr>
<h2 id='gammatheta'>
d^2/d_gamma_kdtheta_k
</h2><span id='topic+gammatheta'></span>

<h3>Description</h3>

<p>Calculates the Information matrix term of I_eta gamma for outcome k.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gammatheta(l, HL_array, HR_array, tpos_all, obs_all, apply_diffs,
   temp_beta, A_i, xDats, no_l_all, gMat, a1, a2, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gammatheta_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_tpos_all">tpos_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is left-censored or not.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_obs_all">obs_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is right-censored or not.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes k summed over all quadrature nodes d.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_xdats">xDats</code></td>
<td>
<p>List of design matrices for all outcomes.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_gmat">gMat</code></td>
<td>
<p>n x q matrix of genetic information.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_a1">a1</code></td>
<td>
<p>First shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_a2">a2</code></td>
<td>
<p>Second shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gammatheta_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a matrix containing the component of the information matrix of the gamma and theta parameters for outcome l.
</p>

<hr>
<h2 id='gammatheta_off'>
d^2/d_gamma_ldtheta_m
</h2><span id='topic+gammatheta_off'></span>

<h3>Description</h3>

<p>Calculates the Information matrix term of I_eta gamma for outcomes l and m
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gammatheta_off(l,m, HL_array, HR_array, xAll, apply_diffs, temp_beta,
   A_i, no_l_all, no_two_all, gMat, a1, a2, k, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gammatheta_off_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_m">m</code></td>
<td>
<p>Index of second outcome of interest.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_xall">xAll</code></td>
<td>
<p>List of design matrices and censoring terms.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes K summed over all quadrature nodes d.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_no_two_all">no_two_all</code></td>
<td>
<p>n x (k - 2) matrix containing the product of apply_diffs across all outcomes k excluding the outcomes l and m.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_gmat">gMat</code></td>
<td>
<p>n x q matrix of genetic information.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_a1">a1</code></td>
<td>
<p>First shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_a2">a2</code></td>
<td>
<p>Second shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="gammatheta_off_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a matrix containing the component of the information matrix of the gamma and theta parameters for outcomes l and m.
</p>

<hr>
<h2 id='get_A'>
Get A vector
</h2><span id='topic+get_A'></span>

<h3>Description</h3>

<p>Product of difference of survival terms of the left and right interval times, across all outcomes k, summed over all quadrature nodes d.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_A(store, weights, d, n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_A_+3A_store">store</code></td>
<td>
<p>Matrix of difference of survival values of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="get_A_+3A_weights">weights</code></td>
<td>
<p>Gaussian quadrature weights.</p>
</td></tr>
<tr><td><code id="get_A_+3A_d">d</code></td>
<td>
<p>Total number of Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="get_A_+3A_n">n</code></td>
<td>
<p>Total number of observations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector used to compute the derivative terms.
</p>

<hr>
<h2 id='Get_CausalSNPs_bynum'>
Get_CausalSNPs_bynum()
</h2><span id='topic+Get_CausalSNPs_bynum'></span>

<h3>Description</h3>

<p>Matrix of subsetted genetic information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Get_CausalSNPs_bynum(gMat, num, Causal.MAF.Cutoff)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Get_CausalSNPs_bynum_+3A_gmat">gMat</code></td>
<td>
<p>Matrix of SNPs.</p>
</td></tr>
<tr><td><code id="Get_CausalSNPs_bynum_+3A_num">num</code></td>
<td>
<p>Number of causal variants.</p>
</td></tr>
<tr><td><code id="Get_CausalSNPs_bynum_+3A_causal.maf.cutoff">Causal.MAF.Cutoff</code></td>
<td>
<p>Minor allele frequency value cutoff for causal SNPs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output is a vector of indices to subset the full genetic matrix.
</p>

<hr>
<h2 id='haz_left'>
H_ik(L_ik)
</h2><span id='topic+haz_left'></span>

<h3>Description</h3>

<p>Calculates the hazard function of the left time interval for outcome l.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haz_left(l, d, temp_beta, phen, r1, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haz_left_+3A_l">l</code></td>
<td>
<p>Outcome of interest.</p>
</td></tr>
<tr><td><code id="haz_left_+3A_d">d</code></td>
<td>
<p>Total number of Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="haz_left_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="haz_left_+3A_phen">phen</code></td>
<td>
<p>list of data matrices containing both left and right information.</p>
</td></tr>
<tr><td><code id="haz_left_+3A_r1">r1</code></td>
<td>
<p>Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="haz_left_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector of the hazard values of the left times.
</p>

<hr>
<h2 id='haz_right'>
H_ik(R_ik)
</h2><span id='topic+haz_right'></span>

<h3>Description</h3>

<p>Calculates the hazard function of the right time interval for outcome l.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haz_right(l, d, temp_beta, phen, r1, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haz_right_+3A_l">l</code></td>
<td>
<p>Outcome of interest.</p>
</td></tr>
<tr><td><code id="haz_right_+3A_d">d</code></td>
<td>
<p>Total number of Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="haz_right_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="haz_right_+3A_phen">phen</code></td>
<td>
<p>list of data matrices containing both left and right information.</p>
</td></tr>
<tr><td><code id="haz_right_+3A_r1">r1</code></td>
<td>
<p>Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="haz_right_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector of the hazard values of the right times.
</p>

<hr>
<h2 id='sd_off'>
d^2/d_theta_kdsigma^2
</h2><span id='topic+sd_off'></span>

<h3>Description</h3>

<p>Calculates the Information matrix term of I_theta sigma^2 for outcomes l and m.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sd_off(l, m, phen_l, phen_m, temp_beta, d, apply_diffs, A_i,
   HL_array, HR_array, no_l_all, no_two_all, tpos_all, obs_all, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sd_off_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_m">m</code></td>
<td>
<p>Index of second outcome of interest.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_phen_l">phen_l</code></td>
<td>
<p>List containing the left and right design matrices and interval times for outcome l.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_phen_m">phen_m</code></td>
<td>
<p>List containing the left and right design matrices and interval times for outcome m.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_d">d</code></td>
<td>
<p>Total number of quadrature nodes.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes k summed over all quadrature nodes d.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_no_two_all">no_two_all</code></td>
<td>
<p>n x (k - 2) matrix containing the product of apply_diffs across all outcomes K excluding outcomes l and m.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_tpos_all">tpos_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is left-censored or not.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_obs_all">obs_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is right-censored or not.</p>
</td></tr>
<tr><td><code id="sd_off_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a matrix containing the component of the information matrix of the sigma and theta parameters.</p>

<hr>
<h2 id='sd_on'>
d^2/dsigma^2^2
</h2><span id='topic+sd_on'></span>

<h3>Description</h3>

<p>Calculates the Information matrix term of I_sigma^2 sigma^2 for outcome l.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sd_on(l, k, temp_beta, phen, d, apply_diffs, A_i,
   no_l_all, HL_array, HR_array)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sd_on_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_phen">phen</code></td>
<td>
<p>List containing the left and right design matrices and interval times for outcome l.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_d">d</code></td>
<td>
<p>Total number of quadrature nodes.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes K summed over all quadrature nodes D.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="sd_on_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a single value for the second derivative with respect to sigma.</p>

<hr>
<h2 id='sim_gmat'>
Simulate genetic matrix.
</h2><span id='topic+sim_gmat'></span>

<h3>Description</h3>

<p>Simulates a n x q genetic matrix with the option to specify the common pairwise correlation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_gmat(n,q,rho)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_gmat_+3A_n">n</code></td>
<td>
<p>Total number of observations.</p>
</td></tr>
<tr><td><code id="sim_gmat_+3A_q">q</code></td>
<td>
<p>Total number of SNPs.</p>
</td></tr>
<tr><td><code id="sim_gmat_+3A_rho">rho</code></td>
<td>
<p>Common pairwise correlation parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of a n x q genetic matrix of q SNPs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set sample size
n = 100

# Set number of SNPs
q = 5

# Set common pairwise correlation
rho = 0.1

# Simulate genetic matrix
gMat &lt;- sim_gmat(n, q, rho)
</code></pre>

<hr>
<h2 id='simico_fit_null'>
simico_fit_null()
</h2><span id='topic+simico_fit_null'></span>

<h3>Description</h3>

<p>Fit the null model via newton raphson for multiple outcomes interval-censored skat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simico_fit_null(init_beta, epsilon, xDats, lt_all, rt_all, k, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simico_fit_null_+3A_init_beta">init_beta</code></td>
<td>
<p>Starting values for NR.</p>
</td></tr>
<tr><td><code id="simico_fit_null_+3A_epsilon">epsilon</code></td>
<td>
<p>Stopping criterion for NR.</p>
</td></tr>
<tr><td><code id="simico_fit_null_+3A_xdats">xDats</code></td>
<td>
<p>List of left and right design matrices.</p>
</td></tr>
<tr><td><code id="simico_fit_null_+3A_lt_all">lt_all</code></td>
<td>
<p>n x k matrix of left times.</p>
</td></tr>
<tr><td><code id="simico_fit_null_+3A_rt_all">rt_all</code></td>
<td>
<p>n x k matrix of right times.</p>
</td></tr>
<tr><td><code id="simico_fit_null_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="simico_fit_null_+3A_d">d</code></td>
<td>
<p>Total number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>beta_fit</code></td>
<td>
<p>Vector of fitted coefficients.</p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p>Number of iterations needed for the Newton-Raphson to converge.</p>
</td></tr>
<tr><td><code>diff</code></td>
<td>
<p>Difference between the current values of temp_beta and the previous iteration of temp_beta.</p>
</td></tr>
<tr><td><code>jmat</code></td>
<td>
<p>Information matrix of the theta parameters.</p>
</td></tr>
<tr><td><code>grad</code></td>
<td>
<p>Vector of the first derivaive of the theta parameters.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Set number of outcomes
k = 2

# Set number of observations
n = 100

# Set number of covariates
p = 2

# Set number of SNPs
q = 50

# Set number of causal SNPs
num = 5

# Set number of quadrature nodes
d = 100

# Variance of subject-specific random effect
tauSq = 1

# Define the effect sizes
effectSizes &lt;- c(.03, .15)

# Set MAF cutoff for causal SNPs
Causal.MAF.Cutoff = 0.1

# the baseline cumulative hazard function
bhFunInv &lt;- function(x) {x}

set.seed(1)

# Generate covariate matrix
xMat &lt;- cbind(rnorm(n), rbinom(n=n, size=1, prob=0.5))

# Generate genetic matrix
gMat &lt;- matrix(data=rbinom(n=n*q, size=2, prob=0.1), nrow=n)

# Get indices to specific select causal variants
idx &lt;- Get_CausalSNPs_bynum(gMat, num, Causal.MAF.Cutoff)

# Subset the gMat
gMatCausal &lt;- gMat[,idx]

# Generate the multiple outcomes
exampleDat &lt;- simico_gen_dat(bhFunInv = bhFunInv, obsTimes = 1:3,
                             windowHalf = 0.1, n, p, k, tauSq, gMatCausal,
                             xMat, effectSizes)

# Set the initial estimate values
init_beta &lt;-c (rep(c(0, 0, 0, 1, 0), k), 1)

# Run the Newton-Raphson
nullFit &lt;- simico_fit_null(init_beta = init_beta,
   epsilon = 10^-5, xDats = exampleDat$fullDat$xDats,
   lt_all = exampleDat$leftTimesMat,
   rt_all = exampleDat$rightTimesMat,
   k = k, d = d)

</code></pre>

<hr>
<h2 id='simico_gen_dat'>
simico_gen_dat()
</h2><span id='topic+simico_gen_dat'></span>

<h3>Description</h3>

<p>Generate multiple interval-censored data under proportional hazards model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simico_gen_dat(bhFunInv, obsTimes = 1:3, windowHalf = 0.1,
   n, p, k, tauSq, gMatCausal, xMat, effectSizes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simico_gen_dat_+3A_bhfuninv">bhFunInv</code></td>
<td>
<p>The inverse of the baseline hazard function.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_obstimes">obsTimes</code></td>
<td>
<p>Vector of the intended observation times.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_windowhalf">windowHalf</code></td>
<td>
<p>The amount of time before or after the intended obsTimes that a visit might take place.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_n">n</code></td>
<td>
<p>Total number of observations.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_p">p</code></td>
<td>
<p>Total number of covariates.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_tausq">tauSq</code></td>
<td>
<p>Variance of the subject specific random effect.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_gmatcausal">gMatCausal</code></td>
<td>
<p>Matrix of subsetted genetic information for only a select causal SNPs.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_xmat">xMat</code></td>
<td>
<p>Matrix of covariates.</p>
</td></tr>
<tr><td><code id="simico_gen_dat_+3A_effectsizes">effectSizes</code></td>
<td>
<p>Vector of genetic effect sizes. Should be entered as a vector the same length as the number of outcomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>exactTimesMat</code></td>
<td>
<p>n x k matrix containing the simulated exact times that the event occurred.</p>
</td></tr>
<tr><td><code>leftTimesMat</code></td>
<td>
<p>n x k matrix containing the left time interval that is observed.</p>
</td></tr>
<tr><td><code>rightTimesMat</code></td>
<td>
<p>n x k matrix containing the right time interval that is observed.</p>
</td></tr>
<tr><td><code>obsInd</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is right-censored or not.</p>
</td></tr>
<tr><td><code>tposInd</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is left-censored or not.</p>
</td></tr>
<tr><td><code>fullDat</code></td>
<td>
<p>Data in complete form to enter into SIMICO functions.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Set number of outcomes
k = 2

# Set number of observations
n = 100

# Set number of covariates
p = 2

# Set number of SNPs
q = 50

# Set number of causal SNPs
num = 5

# Set number of quadrature nodes
d = 100

# Variance of subject-specific random effect
tauSq = 1

# Define the effect sizes
effectSizes &lt;- c(.03, .15)

# Set MAF cutoff for causal SNPs
Causal.MAF.Cutoff = 0.1

# the baseline cumulative hazard function
bhFunInv &lt;- function(x) {x}

set.seed(1)

# Generate covariate matrix
xMat &lt;- cbind(rnorm(n), rbinom(n=n, size=1, prob=0.5))

# Generate genetic matrix
gMat &lt;- matrix(data=rbinom(n=n*q, size=2, prob=0.1), nrow=n)

# Get indices to specific select causal variants
idx &lt;- Get_CausalSNPs_bynum(gMat, num, Causal.MAF.Cutoff)

# Subset the gMat
gMatCausal &lt;- gMat[,idx]

# Generate the multiple outcomes
exampleDat &lt;- simico_gen_dat(bhFunInv = bhFunInv, obsTimes = 1:3,
                             windowHalf = 0.1, n, p, k, tauSq, gMatCausal,
                             xMat, effectSizes)
</code></pre>

<hr>
<h2 id='simico_out'>
Get P-Values
</h2><span id='topic+simico_out'></span>

<h3>Description</h3>

<p>Calculate test statistic and p-values for multiple outcome test and multiple burden test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simico_out(nullFit, xDats, lt_all, rt_all, Itt, a1, a2, G, k, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simico_out_+3A_nullfit">nullFit</code></td>
<td>
<p>Results of the Newton-Raphson: estimates of the beta coefficients.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_xdats">xDats</code></td>
<td>
<p>List of design matrices.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_lt_all">lt_all</code></td>
<td>
<p>Matrix containing the generated left interval times.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_rt_all">rt_all</code></td>
<td>
<p>Matrix containing the generated right interval times.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_itt">Itt</code></td>
<td>
<p>I_theta theta - Information matrix of theta.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_g">G</code></td>
<td>
<p>n x q matrix of genetic information.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_a1">a1</code></td>
<td>
<p>First shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_a2">a2</code></td>
<td>
<p>Second shape parameter of beta parameter.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="simico_out_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>multQ</code></td>
<td>
<p>Score statistic for multiple outcome test.</p>
</td></tr>
<tr><td><code>multP</code></td>
<td>
<p>P-value for multiple outcome test.</p>
</td></tr>
<tr><td><code>burdQ</code></td>
<td>
<p>Score statistic for multiple burden test.</p>
</td></tr>
<tr><td><code>burdP</code></td>
<td>
<p>P-value for multiple burden test.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Set number of outcomes
k = 2

# Set number of observations
n = 100

# Set number of covariates
p = 2

# Set number of SNPs
q = 50

# Set number of causal SNPs
num = 5

# Set number of quadrature nodes
d = 100

# Variance of subject-specific random effect
tauSq = 1

# Define the effect sizes
effectSizes &lt;- c(.03, .15)

# Set MAF cutoff for causal SNPs
Causal.MAF.Cutoff = 0.1

# the baseline cumulative hazard function
bhFunInv &lt;- function(x) {x}

set.seed(1)

# Generate covariate matrix
xMat &lt;- cbind(rnorm(n), rbinom(n=n, size=1, prob=0.5))

# Generate genetic matrix
gMat &lt;- matrix(data=rbinom(n=n*q, size=2, prob=0.1), nrow=n)

# Get indices to specific select causal variants
idx &lt;- Get_CausalSNPs_bynum(gMat, num, Causal.MAF.Cutoff)

# Subset the gMat
gMatCausal &lt;- gMat[,idx]

# Generate the multiple outcomes
exampleDat &lt;- simico_gen_dat(bhFunInv = bhFunInv, obsTimes = 1:3,
                             windowHalf = 0.1, n, p, k, tauSq, gMatCausal,
                             xMat, effectSizes)

# Set the initial estimate values
init_beta &lt;-c (rep(c(0, 0, 0, 1, 0), k), 1)

# Run the newton-raphson
nullFit &lt;- simico_fit_null(init_beta = init_beta,
   epsilon = 10^-5, xDats = exampleDat$fullDat$xDats,
   lt_all = exampleDat$leftTimesMat,
   rt_all = exampleDat$rightTimesMat,
   k = k, d = d)

# Get the test statistics p-values
out &lt;- simico_out(nullFit = nullFit$beta_fit,
   xDats = exampleDat$fullDat$xDats,
   lt_all = exampleDat$leftTimesMat,
   rt_all = exampleDat$rightTimesMat,
   Itt = nullFit$jmat, a1 = 1, a2 = 25,
   G = gMat, k  = k, d = d)

# Print results
# Score statistic
out$multQ

# P-values
out$multP
</code></pre>

<hr>
<h2 id='ss_fd'>
d/d_sigma^2
</h2><span id='topic+ss_fd'></span>

<h3>Description</h3>

<p>Calculates the first derivative term with respect to sigma^2.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss_fd(l, phen, HL_array, HR_array, tpos_all, obs_all,
   apply_diffs, temp_beta, A_i, no_l_all, k, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ss_fd_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_phen">phen</code></td>
<td>
<p>List containing all the left and right design matrices.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_tpos_all">tpos_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is left-censored or not.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_obs_all">obs_all</code></td>
<td>
<p>n x k matrix containing a indictor for whether that time is right-censored or not.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes k summed over all quadrature nodes d.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="ss_fd_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a single value for the first derivative with respect to sigma.</p>

<hr>
<h2 id='ss_sd'>
d^2/d_sigma^2^2
</h2><span id='topic+ss_sd'></span>

<h3>Description</h3>

<p>Calculates the second derivative term with respect to sigma^2.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ss_sd(HL_array, HR_array, xAll, apply_diffs, temp_beta,
   A_i, no_l_all, no_two_all, k, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ss_sd_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_xall">xAll</code></td>
<td>
<p>List containing the left and right matrices and event times.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes K summed over all quadrature nodes D.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_no_two_all">no_two_all</code></td>
<td>
<p>n x (k - 2) matrix containing the product of apply_diffs across all outcomes k excluding outcomes l and m.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="ss_sd_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a single value for the second derivative with respect to sigma^2.</p>

<hr>
<h2 id='st_off'>
d^2/d_theta_ldsigma^2
</h2><span id='topic+st_off'></span>

<h3>Description</h3>

<p>Calculates the Information matrix term of I_eta theta for one outcome of interest l.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>st_off(l, HL_array, HR_array, xAll, apply_diffs,
   temp_beta, A_i, no_l_all, no_two_all, k, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="st_off_+3A_l">l</code></td>
<td>
<p>Index of first outcome of interest.</p>
</td></tr>
<tr><td><code id="st_off_+3A_hl_array">HL_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the left times.</p>
</td></tr>
<tr><td><code id="st_off_+3A_hr_array">HR_array</code></td>
<td>
<p>n x k matrix containing all the hazard values for the right times.</p>
</td></tr>
<tr><td><code id="st_off_+3A_xall">xAll</code></td>
<td>
<p>List containing the left and right matrices and event times.</p>
</td></tr>
<tr><td><code id="st_off_+3A_apply_diffs">apply_diffs</code></td>
<td>
<p>Matrix containing the differences between survival functions of the left and right time intervals.</p>
</td></tr>
<tr><td><code id="st_off_+3A_temp_beta">temp_beta</code></td>
<td>
<p>vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="st_off_+3A_a_i">A_i</code></td>
<td>
<p>Product of apply_diffs across all outcomes K summed over all quadrature nodes D.</p>
</td></tr>
<tr><td><code id="st_off_+3A_no_l_all">no_l_all</code></td>
<td>
<p>n x (k - 1) matrix containing the product of apply_diffs across all outcomes k excluding the current outcome l.</p>
</td></tr>
<tr><td><code id="st_off_+3A_no_two_all">no_two_all</code></td>
<td>
<p>n x (k - 2) matrix containing the product of apply_diffs across all outcomes k excluding outcomes l and m.</p>
</td></tr>
<tr><td><code id="st_off_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="st_off_+3A_d">d</code></td>
<td>
<p>Number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a matrix containing the component of the information matrix of the theta eta parameters for outcome l.</p>

<hr>
<h2 id='surv_diff'>
S_ik(L_ik) - S_ik(R_ik)
</h2><span id='topic+surv_diff'></span>

<h3>Description</h3>

<p>Calculates the difference between the survival functions of the left and right time intervals for outcome k for quadrature node d.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_diff(l, d, temp_beta, phen, r1, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="surv_diff_+3A_l">l</code></td>
<td>
<p>Outcome of interest.</p>
</td></tr>
<tr><td><code id="surv_diff_+3A_d">d</code></td>
<td>
<p>Total number of Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="surv_diff_+3A_temp_beta">temp_beta</code></td>
<td>
<p>Vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="surv_diff_+3A_phen">phen</code></td>
<td>
<p>List of data matrices containing both left and right information.</p>
</td></tr>
<tr><td><code id="surv_diff_+3A_r1">r1</code></td>
<td>
<p>Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="surv_diff_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector of the difference of the survival values of the left times and right times.
</p>

<hr>
<h2 id='surv_left'>
S_ik(L_ik)
</h2><span id='topic+surv_left'></span>

<h3>Description</h3>

<p>Calculates the survival function of the left time interval for outcome k for quadrature node d.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_left(l, d, temp_beta, phen, r1, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="surv_left_+3A_l">l</code></td>
<td>
<p>Outcome of interest.</p>
</td></tr>
<tr><td><code id="surv_left_+3A_d">d</code></td>
<td>
<p>Total number of Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="surv_left_+3A_temp_beta">temp_beta</code></td>
<td>
<p>Vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="surv_left_+3A_phen">phen</code></td>
<td>
<p>List of data matrices containing both left and right information.</p>
</td></tr>
<tr><td><code id="surv_left_+3A_r1">r1</code></td>
<td>
<p>Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="surv_left_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector of the survival values of the left times.
</p>

<hr>
<h2 id='surv_right'>
S_ik(R_ik)
</h2><span id='topic+surv_right'></span>

<h3>Description</h3>

<p>Calculates the survival function of the right time interval for outcome k for quadrature node d.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_right(l, d, temp_beta, phen, r1, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="surv_right_+3A_l">l</code></td>
<td>
<p>Outcome of interest.</p>
</td></tr>
<tr><td><code id="surv_right_+3A_d">d</code></td>
<td>
<p>Total number of Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="surv_right_+3A_temp_beta">temp_beta</code></td>
<td>
<p>Vector of fitted coefficients.</p>
</td></tr>
<tr><td><code id="surv_right_+3A_phen">phen</code></td>
<td>
<p>List of data matrices containing both left and right information.</p>
</td></tr>
<tr><td><code id="surv_right_+3A_r1">r1</code></td>
<td>
<p>Gaussian quadrature nodes.</p>
</td></tr>
<tr><td><code id="surv_right_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector of the survival values of the left times.
</p>

<hr>
<h2 id='without_one_phen'>
Survival Difference Product without Outcome l
</h2><span id='topic+without_one_phen'></span>

<h3>Description</h3>

<p>Calculate the product of the difference between survival terms excluding that of the outcome of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>without_one_phen(l, k, store)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="without_one_phen_+3A_l">l</code></td>
<td>
<p>Outcome of interest.</p>
</td></tr>
<tr><td><code id="without_one_phen_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="without_one_phen_+3A_store">store</code></td>
<td>
<p>Array of difference between left and right survival values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A n x (k-1) matrix where each column is the product of all the differences of left and right survival values across all outcomes excluding the column index outcome.
</p>

<hr>
<h2 id='without_two_phen'>
Survival Difference Product without Outcomes l and m
</h2><span id='topic+without_two_phen'></span>

<h3>Description</h3>

<p>Differnence of survival functions multiplied across all outcomes excluding outcomes l and m.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>without_two_phen(l, m, k, store, n, d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="without_two_phen_+3A_l">l</code></td>
<td>
<p>The first outcome of interest.</p>
</td></tr>
<tr><td><code id="without_two_phen_+3A_m">m</code></td>
<td>
<p>The second outcome of interest.</p>
</td></tr>
<tr><td><code id="without_two_phen_+3A_k">k</code></td>
<td>
<p>Total number of outcomes.</p>
</td></tr>
<tr><td><code id="without_two_phen_+3A_store">store</code></td>
<td>
<p>Array of difference between left and right survival values.</p>
</td></tr>
<tr><td><code id="without_two_phen_+3A_n">n</code></td>
<td>
<p>Total number of observation.</p>
</td></tr>
<tr><td><code id="without_two_phen_+3A_d">d</code></td>
<td>
<p>Total number of quadrature nodes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A n x (k-2) matrix containing the product of all the differences of left and right survival values across all outcomes excluding outcomes l and m.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
