<!DOCTYPE html><html><head><title>Help for package simMP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {simMP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mutDistriWGS'>
<p>Distribution of single base substitutions</p></a></li>
<li><a href='#simSBS'>
<p>Simulate single base substitutions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulate Somatic Mutations in Cancer Genomes from Mutational
Processes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.17.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Nan Zhou</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nan Zhou &lt;zhnanx@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulates somatic single base substitutions carried in cancer genomes. By only providing a human reference genome, substitutions that result from mutational processes operative in every cancer genome can be generated.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, BSgenome, GenomeInfoDb, GenomicRanges, IRanges,
XVector</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0), doParallel, utils, foreach, parallel, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BSgenome.Hsapiens.UCSC.hg38</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-04-21 03:42:12 UTC; mq44741340</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-04-21 06:03:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='mutDistriWGS'>
Distribution of single base substitutions
</h2><span id='topic+mutDistriWGS'></span>

<h3>Description</h3>

<p>Distribution of single base substitutions over all currently available WGS genomes in ICGC data realse 23.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mutDistriWGS")</code></pre>


<h3>Format</h3>

<p>A data frame with 3543 observations on the following variable.
</p>

<dl>
<dt><code>X0</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Zhou, Nan, et al. &quot;Pan-cancer scale landscape of simple somatic mutations.&quot; bioRxiv (2017): 112367.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mutDistriWGS)
head(mutDistriWGS)

## Not run: 
plot(1:nrow(mutDistriWGS), sort(c(t(mutDistriWGS)), decreasing = TRUE))

## End(Not run)
</code></pre>

<hr>
<h2 id='simSBS'>
Simulate single base substitutions
</h2><span id='topic+simSBS'></span>

<h3>Description</h3>

<p>Given the number of genomes to be created, generate single base substitutions in those genomes from simulated mutational processes, by referring to a human reference genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simSBS(nSigs = NULL, nGenomes = NULL, refGenome = NULL,
        similarity = 0.6, noise = 0,
        presetSigs = NULL, chrs = NULL, nMutPerGenome = NULL,
        sigPrevalence = NULL, chrDistribution = NULL,
        parallel = TRUE, saveDir = './')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simSBS_+3A_nsigs">nSigs</code></td>
<td>

<p><strong>Required</strong>. The number of mutational processes to be created.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_ngenomes">nGenomes</code></td>
<td>

<p><strong>Required</strong>. The number of genomes in which to simulate single base substitutions.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_refgenome">refGenome</code></td>
<td>

<p><strong>Required</strong>. A <code>BSgenome</code> object of human reference genome.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_similarity">similarity</code></td>
<td>

<p>Optional. Limit the similarity between any two mutational proccesses. 0 indicates no similarity while 1 indicates the opposite. Lower similarity may require more time to simulate.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_noise">noise</code></td>
<td>

<p>Optional. The value should between 0 and 1, indicating the amount of random mutations (noise) added to each simulated genome. 0 indicates no noise while 1 indicates the amount of noise is equal to the amount of mutation.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_presetsigs">presetSigs</code></td>
<td>

<p>Optional. Use user defined mutational processes to simulate mutations in the genome. It should be a 96-by-n matrix, where 96 denotes the number of mutation motifs while n denotes the number of mutational processes. If presetSigs is given, nSigs = n.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_chrs">chrs</code></td>
<td>

<p>Optional. On wich chromosome(s) mutations simulated from. Default is <code>c(1:22, 'X', 'Y')</code>. This argument accepts a vector that indicates chromosomes, which should be a vector created by manual input or, for example, using R code like <code>c(1:22, 'X', 'Y', 'M')</code>, where 'X', 'Y', 'M' are case sensitive (upper case) and indicate chromosome X, Y and mitochondrial chromosome. Incompatible input could cause fatal errors cause of unidentifiable chromosome name.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_nmutpergenome">nMutPerGenome</code></td>
<td>

<p>Optional. <em>NULL</em> or a numerical vector whose length equals nGenomes. Number of mutations on each genome to simulate. If not defined, Default will use the <a href="https://doi.org/10.1101/112367">distribution</a> of number of single base substitutions in all WGS projects of <a href="https://dcc.icgc.org">ICGC</a> release 23.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_sigprevalence">sigPrevalence</code></td>
<td>

<p>Optional. Acceptable values are either <em>NULL</em> or <em>a numerical vector</em>. The prevalence of mutational processes in wild. The default uses known prevalances of 21 processes from <a href="http://dx.doi.org/10.1038/nature12477">Alexandrov et al.</a>'s work.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_chrdistribution">chrDistribution</code></td>
<td>

<p>Optional. <em>NULL</em> or <em>a numerical vector</em> are acceptable. The percentage of mutations assigned to each chromosome in a genome. The default uses the distribution of length of chromosomes (chr1 to chr22 and chrX and chrY). If a numerical vector was given, its length should equal the length of <em>chrs</em> and values should sum up to 1.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_parallel">parallel</code></td>
<td>

<p>Optional. <code>TRUE</code> or <code>FALSE</code>. Whether enable or disable parallel computing ability.
</p>
</td></tr>
<tr><td><code id="simSBS_+3A_savedir">saveDir</code></td>
<td>

<p>Optional. The directory where to save simulation output. Default is the current working directory. Other paths should also be relative to the current working directory.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If succeed, the return value is 1. Simulation results are saved in saveDir.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(BSgenome.Hsapiens.UCSC.hg38)){
  simSBS(nSigs = 2, nGenomes = 2,
    refGenome = BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38,
    nMutPerGenome = sample(10:50, 2),
    parallel = FALSE)
}else{
  message('Cannot proceed withoud a valid reference genome.')
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
