<!DOCTYPE html><html><head><title>Help for package simphony</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {simphony}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#defaultDispFunc'><p>Default function for mapping expected counts to dispersion.</p></a></li>
<li><a href='#getExpectedAbund'><p>Calculate expected abundance</p></a></li>
<li><a href='#getSampledAbund'><p>Sample abundance values</p></a></li>
<li><a href='#mergeSimData'><p>Merge abundance data, feature metadata, and sample metadata</p></a></li>
<li><a href='#simphony'><p>Simulate feature abundance data</p></a></li>
<li><a href='#splitDiffFeatureGroups'><p>Split differential featureGroups</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulating Large-Scale, Rhythmic Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Description:</td>
<td>A tool for simulating rhythmic data: transcriptome data using
  Gaussian or negative binomial distributions, and behavioral activity data
  using Bernoulli or Poisson distributions. See Singer et al. (2019)
  &lt;<a href="https://doi.org/10.7717%2Fpeerj.6985">doi:10.7717/peerj.6985</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://simphony.hugheylab.org">https://simphony.hugheylab.org</a>,
<a href="https://github.com/hugheylab/simphony">https://github.com/hugheylab/simphony</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table (&ge; 1.11.4), foreach (&ge; 1.4.4)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2 (&ge; 3.0.0), kableExtra (&ge; 0.9.0), knitr (&ge; 1.20),
limma (&ge; 3.34.9), precrec (&ge; 0.9.1), rmarkdown (&ge; 1.9),
testthat (&ge; 2.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-08 21:05:10 UTC; jakehughey</td>
</tr>
<tr>
<td>Author:</td>
<td>Jake Hughey [aut, cre],
  Jordan Singer [aut],
  Darwin Fu [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jake Hughey &lt;jakejhughey@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-09 15:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='defaultDispFunc'>Default function for mapping expected counts to dispersion.</h2><span id='topic+defaultDispFunc'></span>

<h3>Description</h3>

<p>The function was estimated from circadian RNA-seq data from mouse liver
(PRJNA297287), using local regression in <code>DESeq2</code>. In a negative binomial
distribution, <code class="reqn">variance = mean + mean^2 * dispersion</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>defaultDispFunc(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="defaultDispFunc_+3A_x">x</code></td>
<td>
<p>Numeric vector of mean counts.</p>
</td></tr>
</table>


<h3>Format</h3>

<p>An object of class <code>function</code> of length 1.
</p>


<h3>Value</h3>

<p>Numeric vector of dispersions.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simphony">simphony()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>means = 2^(6:10)
dispersions = defaultDispFunc(means)

</code></pre>

<hr>
<h2 id='getExpectedAbund'>Calculate expected abundance</h2><span id='topic+getExpectedAbund'></span>

<h3>Description</h3>

<p>Calculate expected abundance for multiple features at multiple timepoints in
multiple conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExpectedAbund(
  featureMetadata,
  times = NULL,
  sampleMetadata = NULL,
  byCondGroup = is.null(times)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getExpectedAbund_+3A_featuremetadata">featureMetadata</code></td>
<td>
<p><code>data.table</code> with columns <code>feature</code>, <code>base</code>,
<code>rhyFunc</code>, <code>amp</code>, <code>period</code>, and <code>phase</code>, where every row corresponds to a
gene. If <code>byCondGroup</code> is <code>TRUE</code>, then must also have columns <code>cond</code> and
<code>group</code>.</p>
</td></tr>
<tr><td><code id="getExpectedAbund_+3A_times">times</code></td>
<td>
<p>Numeric vector of the times at which to calculate expected
abundance for each row in <code>featureMetadata</code>.</p>
</td></tr>
<tr><td><code id="getExpectedAbund_+3A_samplemetadata">sampleMetadata</code></td>
<td>
<p><code>data.table</code> with columns <code>sample</code>, <code>cond</code>, and
<code>time</code>. Either <code>times</code> or <code>sampleMetadata</code> must be provided, and the former
takes precedence.</p>
</td></tr>
<tr><td><code id="getExpectedAbund_+3A_bycondgroup">byCondGroup</code></td>
<td>
<p>Logical for whether to speed up the calculation by
grouping by the columns <code>cond</code> and <code>group</code>. Primarily for internal use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.table</code> derived from <code>featureMetadata</code> (but with more rows),
with additional columns <code>time</code> and <code>mu</code> and possibly others. If sampling
will use the negative binomial family, <code>mu</code> corresponds to log2 counts.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simphony">simphony()</a></code>, <code><a href="#topic+getSampledAbund">getSampledAbund()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')
featureMetadata = data.table(feature = c('feature_1', 'feature_2'),
                             base = function(x) 0,
                             amp = c(function(x) 0, function(x) 1),
                             period = 24,
                             phase = 0, rhyFunc = sin)
abundDt = getExpectedAbund(featureMetadata, times = 6:17)

</code></pre>

<hr>
<h2 id='getSampledAbund'>Sample abundance values</h2><span id='topic+getSampledAbund'></span>

<h3>Description</h3>

<p>Sample feature abundance values from the given distributions. This function
is used internally by <code><a href="#topic+simphony">simphony()</a></code>, and should not usually need to be
called directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSampledAbund(
  abundDt,
  logOdds = FALSE,
  family = c("gaussian", "negbinom", "bernoulli", "poisson"),
  inplace = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSampledAbund_+3A_abunddt">abundDt</code></td>
<td>
<p><code>data.table</code> of expected abundance. If <code>family</code> is 'gaussian',
required columns are <code>feature</code>, <code>sample</code>, <code>mu</code>, and <code>sd</code>. If <code>family</code> is
'negbinom', required columns are <code>feature</code>, <code>sample</code>, <code>mu</code>, <code>dispFunc</code>,
<code>cond</code>, and <code>group</code>. If <code>family</code> is 'bernoulli' or 'poisson', required
columns are <code>feature</code>, <code>sample</code>, and <code>mu</code>.</p>
</td></tr>
<tr><td><code id="getSampledAbund_+3A_logodds">logOdds</code></td>
<td>
<p>Logical for whether <code>mu</code> corresponds to log-odds. Only used if
<code>family</code> is 'bernoulli'.</p>
</td></tr>
<tr><td><code id="getSampledAbund_+3A_family">family</code></td>
<td>
<p>Character string for the family of distributions from which
to sample the abundance values. <code>simphony</code> will give a warning if it tries
to sample from a distribution outside the region in which the distribution
is defined: <code class="reqn">\mu &lt; 0</code> for negative binomial and Poisson, and
<code class="reqn">\mu &lt; 0</code> or <code class="reqn">\mu &gt; 1</code> for Bernoulli.</p>
</td></tr>
<tr><td><code id="getSampledAbund_+3A_inplace">inplace</code></td>
<td>
<p>Logical for whether to modify <code>abundDt</code> in-place, adding a
column <code>abund</code> containing the abundance values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of abundance values, where rows correspond to features and
columns correspond to samples.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simphony">simphony()</a></code>, <code><a href="#topic+getExpectedAbund">getExpectedAbund()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')
set.seed(6022)
abundDt = data.table(feature = 'feature_1', sample = c('sample_1', 'sample_2'),
                    mu = c(0, 5), sd = 1)
abundMat = getSampledAbund(abundDt)

</code></pre>

<hr>
<h2 id='mergeSimData'>Merge abundance data, feature metadata, and sample metadata</h2><span id='topic+mergeSimData'></span>

<h3>Description</h3>

<p>Merge a simulation's abundance data, feature metadata, and sample metadata
into one <code>data.table</code>. This function is useful for making plots using
ggplot2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeSimData(simData, features = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mergeSimData_+3A_simdata">simData</code></td>
<td>
<p>List with the following elements, such as returned by
<code><a href="#topic+simphony">simphony()</a></code>:
</p>

<dl>
<dt>abundData</dt><dd><p>Matrix of abundance values, with rownames for features and
colnames for samples.</p>
</dd>
<dt>sampleMetadata</dt><dd><p><code>data.table</code> with columns <code>sample</code> and <code>cond</code>.</p>
</dd>
<dt>featureMetadata</dt><dd><p><code>data.table</code> with columns <code>feature</code> and <code>cond</code>.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="mergeSimData_+3A_features">features</code></td>
<td>
<p>Character vector of features for which to get abundance data.
If <code>NULL</code>, then all features.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.table</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simphony">simphony()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')
featureGroups = data.table(amp = c(0, 1))
simData = simphony(featureGroups)
mergedSimData = mergeSimData(simData, simData$featureMetadata$feature[1:2])

</code></pre>

<hr>
<h2 id='simphony'>Simulate feature abundance data</h2><span id='topic+simphony'></span>

<h3>Description</h3>

<p>Simulate experiments in which abundances of rhythmic and non-rhythmic
features are measured at multiple timepoints in one or more conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simphony(
  featureGroupsList,
  fracFeatures = NULL,
  nFeatures = 10,
  timepointsType = c("auto", "specified", "random"),
  timeRange = c(0, 48),
  interval = 2,
  nReps = 1,
  timepoints = NULL,
  nSamplesPerCond = NULL,
  rhyFunc = sin,
  dispFunc = NULL,
  logOdds = FALSE,
  family = c("gaussian", "negbinom", "bernoulli", "poisson")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simphony_+3A_featuregroupslist">featureGroupsList</code></td>
<td>
<p><code>data.frame</code> or <code>data.table</code> (for a single
condition) or list of <code>data.frame</code>s or <code>data.table</code>s (for multiple
conditions), where each row corresponds to a group of features to simulate.
The following columns are all optional:
</p>

<dl>
<dt>fracFeatures</dt><dd><p>Fraction of simulated features to allocate to each
group. Defaults to 1/(number of groups).</p>
</dd>
<dt>rhyFunc</dt><dd><p>Function to generate rhythmic abundance. Must have a
period of <code class="reqn">2\pi</code>. Defaults to <code>sin</code>.</p>
</dd>
<dt>amp</dt><dd><p>Amplitude of rhythm. Defaults to 0. Corresponds to
multiplicative term in front of <code>rhyFunc</code>. Can be numeric (constant
over time) or a function (time-dependent). See vignette for examples.</p>
</dd>
<dt>period</dt><dd><p>Period of rhythm. Defaults to 24.</p>
</dd>
<dt>phase</dt><dd><p>Phase of rhythm, in the same units as <code>period</code>. Defaults to
0. Corresponds to an additive term in <code>rhyFunc</code>.</p>
</dd>
<dt>base</dt><dd><p>Baseline abundance, i.e., abundance when <code>rhyFunc</code> term is 0.
Depending on <code>family</code>, defaults to 0 ('gaussian'), 8 ('negbinom',
mean log2 counts), 0 ('bernoulli' with <code>logOdds</code> as <code>TRUE</code>),
0.5 ('bernoulli' if <code>logOdds</code> as <code>FALSE</code>), or 1 ('poisson'). Can be
numeric (constant over time) or a function (time-dependent). See
vignette for examples.</p>
</dd>
<dt>sd</dt><dd><p>Standard deviation of sampled abundance values. Defaults to 1.
Only used if <code>family</code> is 'gaussian'.</p>
</dd>
<dt>dispFunc</dt><dd><p>Function to calculate dispersion of sampled abundance
values, given expected abundance in counts. Only used if <code>family</code> is
'negbinom'.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="simphony_+3A_fracfeatures">fracFeatures</code></td>
<td>
<p>Fraction of simulated features to allocate to each group.
Defaults to 1/(number of groups). Only used if the first
<code>featureGroupsList</code> <code>data.frame</code> lacks a <code>fracFeatures</code> column.</p>
</td></tr>
<tr><td><code id="simphony_+3A_nfeatures">nFeatures</code></td>
<td>
<p>Integer for the total number of features to simulate.</p>
</td></tr>
<tr><td><code id="simphony_+3A_timepointstype">timepointsType</code></td>
<td>
<p>Character string for how to set the timepoints
for the simulation. Must be 'auto' (default), 'specified', or 'random'.</p>
</td></tr>
<tr><td><code id="simphony_+3A_timerange">timeRange</code></td>
<td>
<p>Numeric vector for the range of timepoints to use for the
simulation. Defaults to c(0, 48). Only used if <code>timepointsType</code> is 'auto'
or 'random'.</p>
</td></tr>
<tr><td><code id="simphony_+3A_interval">interval</code></td>
<td>
<p>Number for the amount of time between consecutive
timepoints, in the same units as <code>period</code>. The first timepoint is 0. Only
used if <code>timepointsType</code> is 'auto'.</p>
</td></tr>
<tr><td><code id="simphony_+3A_nreps">nReps</code></td>
<td>
<p>Integer for the number of replicates per timepoint. Only used
if <code>timepointsType</code> is 'auto'.</p>
</td></tr>
<tr><td><code id="simphony_+3A_timepoints">timepoints</code></td>
<td>
<p>Numeric vector of exact timepoints to simulate, including
any replicates. Only used if <code>timepointsType</code> is 'specified'.</p>
</td></tr>
<tr><td><code id="simphony_+3A_nsamplespercond">nSamplesPerCond</code></td>
<td>
<p>Integer for the number of samples per condition,
which will be randomly uniformly spaced between 0 and <code>period</code> and
different for each condition. Only used if timepointsType is 'random'.</p>
</td></tr>
<tr><td><code id="simphony_+3A_rhyfunc">rhyFunc</code></td>
<td>
<p>Function to generate rhythmic abundance. Must have a period
of <code class="reqn">2\pi</code>. Defaults to <code>sin</code>. Only used if a <code>data.frame</code> in
<code>featureGroupsList</code> lacks a <code>rhyFunc</code> column.</p>
</td></tr>
<tr><td><code id="simphony_+3A_dispfunc">dispFunc</code></td>
<td>
<p>Function to calculate dispersion of sampled abundance
values, given expected abundance in counts. Defaults to <code>defaultDispFunc</code>.
Only used if <code>family</code> is 'negbinom' and a <code>data.frame</code> in
<code>featureGroupsList</code> lacks a <code>dispFunc</code> column.</p>
</td></tr>
<tr><td><code id="simphony_+3A_logodds">logOdds</code></td>
<td>
<p>Logical for whether the rhythmic function corresponds to
log-odds. Only used if <code>family</code> is 'bernoulli'.</p>
</td></tr>
<tr><td><code id="simphony_+3A_family">family</code></td>
<td>
<p>Character string for the family of distributions from which to
sample the abundance values. <code>simphony</code> will give a warning if it tries to
sample from a distribution outside the region in which the distribution is
defined: <code class="reqn">\mu &lt; 0</code> for negative binomial and Poisson, and <code class="reqn">\mu &lt; 0</code>
or <code class="reqn">\mu &gt; 1</code> for Bernoulli.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with the following elements:
</p>

<dl>
<dt>abundData</dt><dd><p>Matrix of abundance values (counts, if <code>family</code> is
'negbinom'), with features as rownames and samples as colnames.</p>
</dd>
<dt>sampleMetadata</dt><dd><p><code>data.table</code> with one row per sample.</p>
</dd>
<dt>featureMetadata</dt><dd><p><code>data.table</code> with one row per feature per condition.
Columns <code>amp</code> and <code>base</code> are functions of time. Columns <code>amp0</code> and
<code>base0</code> are numeric and correspond to the amplitude and baseline
abundance at time 0, respectively.</p>
</dd>
<dt>experMetadata</dt><dd><p>List of arguments that were passed to <code>simphony</code>.</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+defaultDispFunc">defaultDispFunc()</a></code>, <code><a href="#topic+getExpectedAbund">getExpectedAbund()</a></code>, <code><a href="#topic+getSampledAbund">getSampledAbund()</a></code>,
<code><a href="#topic+mergeSimData">mergeSimData()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

# Simulate data for features having one of three sets of rhythmic parameters.
featureGroups = data.table(amp = c(0, 1, 1), phase = c(0, 0, 6),
                           rhyFunc = c(cos, cos, sin))
simData = simphony(featureGroups)

# Simulate data for an experiment with specified timepoints and replicates.
featureGroups = data.table(amp = c(0, 1))
simData = simphony(featureGroups, timepointsType = 'specified',
                   timepoints = c(0, 2, 2, 4, 12, 16, 21))

# Simulate data for an experiment with random timepoints between 0 and 24.
featureGroups = data.table(amp = c(0, 2))
simData = simphony(featureGroups, timepointsType = 'random',
                   timeRange = c(0, 24), nSamplesPerCond = 20)

# Simulate data with time-dependent rhythm amplitude or baseline abundance
featureGroups = data.table(amp = c(function(x) 1, function(x) 2^(-x / 24)),
                           base = c(function(x) x / 12, function(x) 0))
simData = simphony(featureGroups)

# Simulate data for features whose rhythmicity varies between two conditions.
featureGroupsList = list(
  data.table(amp = c(1, 2, 2), phase = c(0, -3, 0), period = c(24, 24, 22)),
  data.table(amp = c(3, 2, 2), phase = c(0, 3, 0), period = c(24, 24, 26)))
simData = simphony(featureGroupsList)

# Simulate data from a negative binomial distribution with a higher variance.
featureGroups = data.table(amp = 1, base = 6:8)
dispFunc = function(x) 3 * defaultDispFunc(x)
simData = simphony(featureGroups, family = 'negbinom', dispFunc = dispFunc)

# Simulate data at high temporal resolution from a Poisson distribution that
# alternates between two states.
featureGroups = data.table(amp = 1, base = 0,
                           rhyFunc = function(x) ifelse(x %% (2 * pi) &lt; pi, 0.5, 4))

simData = simphony(featureGroups, timeRange = c(0, 24 * 4), interval = 0.1,
                   nReps = 1, family = 'poisson')

# Simulate data for 100 features, half non-rhythmic and half rhythmic, with
# amplitudes for rhythmic features sampled from a log-normal distribution.
nFeatures = 100
rhyFrac = 0.5
nRhyFeatures = round(rhyFrac * nFeatures)
rhyAmps = exp(rnorm(nRhyFeatures, mean = 0, sd = 0.25))
fracFeatures = c(1 - rhyFrac, rep(rhyFrac / nRhyFeatures, nRhyFeatures))
featureGroups = data.table(amp = c(0, rhyAmps), fracFeatures = fracFeatures)
simData = simphony(featureGroups, nFeatures = nFeatures)

# Simulate data for 100 rhythmic features, with baseline log2 expected counts
# and residual log dispersion sampled from distributions whose parameters
# were estimated, using DESeq2 and fitdistrplus, from circadian RNA-seq data
# from mouse liver (PRJNA297287).
nFeatures = 100
baseLog2Counts = rnorm(nFeatures, mean = 8.63, sd = 2.73)
dispFactors = exp(rnorm(nFeatures, sd = 0.819))
dispFuncs = sapply(dispFactors, function(z) {function(x) defaultDispFunc(x) * z})
featureGroups = data.table(base = baseLog2Counts, dispFunc = dispFuncs, amp = 1)
simData = simphony(featureGroups, nFeatures = nFeatures, family = 'negbinom')

</code></pre>

<hr>
<h2 id='splitDiffFeatureGroups'>Split differential featureGroups</h2><span id='topic+splitDiffFeatureGroups'></span>

<h3>Description</h3>

<p>Split a diffFeatureGroups data.frame into a list of two featureGroups
data.frames, which can then be passed to <code><a href="#topic+simphony">simphony()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitDiffFeatureGroups(diffFeatureGroups, checkValid = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitDiffFeatureGroups_+3A_difffeaturegroups">diffFeatureGroups</code></td>
<td>
<p><code>data.frame</code> with optional columns <code>meanBase</code>,
<code>dBase</code>, <code>meanSd</code>, <code>dSd</code>, <code>meanAmp</code>, <code>dAmp</code>, <code>meanPhase</code>, and <code>dPhase</code>
describing the changes in abundance between two conditions. Each row
corresponds to a group of features.</p>
</td></tr>
<tr><td><code id="splitDiffFeatureGroups_+3A_checkvalid">checkValid</code></td>
<td>
<p>Logical for whether to only return rows for which both
amplitudes are greater than or equal to zero and both standard deviations
are greater than zero.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of two <code>data.table</code>s with possible columns <code>base</code>, <code>sd</code>, <code>amp</code>,
and <code>phase</code>, depending on the columns in <code>diffFeatureGroups</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simphony">simphony()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dGroups = data.frame(meanAmp = c(1, 1, 1, 1), dAmp = c(1, 1, 2, 2),
                     meanPhase = c(0, 0, 0, 0), dPhase = c(0, 3, 0, 3))
featureGroups = splitDiffFeatureGroups(dGroups)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
