<!DOCTYPE html><html><head><title>Help for package simplePHENOTYPES</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {simplePHENOTYPES}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#base_line_multi_traits'><p>Calculate genetic value based on QTN objects.</p></a></li>
<li><a href='#base_line_single_trait'><p>Calculate genetic value based on QTN objects.</p></a></li>
<li><a href='#constraint'><p>Select SNPs to be assigned as QTNs: To be included LD, FST</p></a></li>
<li><a href='#create_phenotypes'><p>Simulation of single/multiple traits under different models and genetic</p>
architectures.</a></li>
<li><a href='#file_loader'><p>Function used by create_phenotypes to load marker dataset.</p></a></li>
<li><a href='#genetic_effect'><p>Calculate genetic value based on QTN objects.</p></a></li>
<li><a href='#genotypes'><p>Generate a numeric (dosage) HapMap file</p></a></li>
<li><a href='#make_pd'><p>obtain a positive definite matrix</p></a></li>
<li><a href='#numericalization'><p>Converts character SNP genotype to numerical (-1, 0, 1) were 1 is the major</p>
allele.
Modified by Samuel Fernandes from:
Alexander E. Lipka, Feng Tian, Qishan Wang, Jason Peiffer, Meng Li,
Peter J. Bradbury, Michael A. Gore, Edward S. Buckler, Zhiwu Zhang
GAPIT: genome association and prediction integrated tool, Bioinformatics,
Volume 28, Issue 18, 15 September 2012, Pages 2397–2399.</a></li>
<li><a href='#phenotypes'><p>Generate environmental effects based on a given heritability</p></a></li>
<li><a href='#qtn_from_user'><p>Select SNPs to be assigned as QTNs</p></a></li>
<li><a href='#qtn_linkage'><p>Select SNPs to be assigned as QTNs</p></a></li>
<li><a href='#qtn_partially_pleiotropic'><p>Select SNPs to be assigned as QTNs</p></a></li>
<li><a href='#qtn_pleiotropic'><p>Select SNPs to be assigned as QTNs.</p></a></li>
<li><a href='#SNP55K_maize282_maf04'><p>Genotypic data for 282 inbred association panel using the 55K SNP array</p>
(filtered by MAF&gt;0.4 to reduce file size).
Reference: Cook et al. 2012. Genetic architecture of maize kernel
composition in the nested association mapping and inbred association panels.
Plant Physiology 158:824-834</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Date:</td>
<td>2021-01-19</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulation of Pleiotropic, Linked and Epistatic Phenotypes</td>
</tr>
<tr>
<td>Description:</td>
<td>The number of studies involving correlated traits and the availability of tools to handle this type of data has increased considerably in the last decade. With such a demand, we need tools for testing hypotheses related to single and multi-trait (correlated) phenotypes based on many genetic settings. Thus, we implemented various options for simulation of pleiotropy and Linkage Disequilibrium under additive, dominance and epistatic models. The simulation currently takes a marker data set as an input and then uses it for simulating multiple traits as described in Fernandes and Lipka (2020) &lt;<a href="https://doi.org/10.1186%2Fs12859-020-03804-y">doi:10.1186/s12859-020-03804-y</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, mvtnorm, stats, utils, SNPRelate, gdsfmt</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/samuelbfernandes/simplePHENOTYPES">https://github.com/samuelbfernandes/simplePHENOTYPES</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/samuelbfernandes/simplePHENOTYPES/issues">https://github.com/samuelbfernandes/simplePHENOTYPES/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-20 16:11:49 UTC; samuelf</td>
</tr>
<tr>
<td>Author:</td>
<td>Samuel Fernandes <a href="https://orcid.org/0000-0001-8269-535X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Alexander Lipka <a href="https://orcid.org/0000-0003-1571-8528"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samuel Fernandes &lt;samuelf@illinois.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-20 16:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='base_line_multi_traits'>Calculate genetic value based on QTN objects.</h2><span id='topic+base_line_multi_traits'></span>

<h3>Description</h3>

<p>Calculate genetic value based on QTN objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>base_line_multi_traits(
  add_obj = NULL,
  dom_obj = NULL,
  epi_obj = NULL,
  add_effect = NULL,
  dom_effect = NULL,
  epi_effect = NULL,
  epi_interaction = NULL,
  ntraits = NULL,
  cor = NULL,
  architecture = NULL,
  rep = NULL,
  rep_by = NULL,
  add = NULL,
  dom = NULL,
  epi = NULL,
  sim_method = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="base_line_multi_traits_+3A_add_obj">add_obj</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_dom_obj">dom_obj</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_epi_obj">epi_obj</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_add_effect">add_effect</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_dom_effect">dom_effect</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_epi_effect">epi_effect</code></td>
<td>
<p>kkkk</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_epi_interaction">epi_interaction</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_ntraits">ntraits</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_cor">cor</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_architecture">architecture</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_rep">rep</code></td>
<td>
<p>= 1,</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_rep_by">rep_by</code></td>
<td>
<p>= 'QTN',</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_add">add</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_dom">dom</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_epi">epi</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_sim_method">sim_method</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_multi_traits_+3A_verbose">verbose</code></td>
<td>
<p>= TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of Genetic values for multiple traits
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes
Last update: Apr 20, 2020
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-base_line_multi_traits&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;
</p>

<hr>
<h2 id='base_line_single_trait'>Calculate genetic value based on QTN objects.</h2><span id='topic+base_line_single_trait'></span>

<h3>Description</h3>

<p>Calculate genetic value based on QTN objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>base_line_single_trait(
  add_obj = NULL,
  dom_obj = NULL,
  epi_obj = NULL,
  add_effect = NULL,
  dom_effect = NULL,
  epi_effect = NULL,
  epi_interaction = NULL,
  rep = NULL,
  rep_by = "experiment",
  ntraits = 1,
  add = NULL,
  dom = NULL,
  epi = NULL,
  sim_method = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="base_line_single_trait_+3A_add_obj">add_obj</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_dom_obj">dom_obj</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_epi_obj">epi_obj</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_add_effect">add_effect</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_dom_effect">dom_effect</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_epi_effect">epi_effect</code></td>
<td>
<p>hhh</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_epi_interaction">epi_interaction</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_rep">rep</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_rep_by">rep_by</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_ntraits">ntraits</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_add">add</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_dom">dom</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_epi">epi</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="base_line_single_trait_+3A_sim_method">sim_method</code></td>
<td>
<p>= NULL,</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of Genetic values
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes and Alexander Lipka
Last update: Apr 20, 2020
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;Base_line_single_trait&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;
</p>

<hr>
<h2 id='constraint'>Select SNPs to be assigned as QTNs: To be included LD, FST</h2><span id='topic+constraint'></span>

<h3>Description</h3>

<p>Select SNPs to be assigned as QTNs: To be included LD, FST
</p>


<h3>Usage</h3>

<pre><code class='language-R'>constraint(
  genotypes = NULL,
  maf_above = NULL,
  maf_below = NULL,
  hets = NULL,
  verbose = verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="constraint_+3A_genotypes">genotypes</code></td>
<td>
<p>a numericalized genotype object (geno_obj).</p>
</td></tr>
<tr><td><code id="constraint_+3A_maf_above">maf_above</code></td>
<td>
<p>Threshold for the minimum value of minor allele frequency.</p>
</td></tr>
<tr><td><code id="constraint_+3A_maf_below">maf_below</code></td>
<td>
<p>Threshold for the maximum value of minor allele frequency.</p>
</td></tr>
<tr><td><code id="constraint_+3A_hets">hets</code></td>
<td>
<p>Option of including (\'include\') and removing (\'remove\') only
heterozygotes.</p>
</td></tr>
<tr><td><code id="constraint_+3A_verbose">verbose</code></td>
<td>
<p>= verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a filtered dataset to be used when selecting QTNs.
Last update: Apr 20, 2020
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash; constrain &mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>

<hr>
<h2 id='create_phenotypes'>Simulation of single/multiple traits under different models and genetic
architectures.</h2><span id='topic+create_phenotypes'></span>

<h3>Description</h3>

<p>Simulation of single/multiple traits under different models and genetic
architectures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_phenotypes(
  geno_obj = NULL,
  geno_file = NULL,
  geno_path = NULL,
  QTN_list = list(add = list(NULL), dom = list(NULL), epi = list(NULL)),
  prefix = NULL,
  rep = NULL,
  ntraits = 1,
  h2 = NULL,
  mean = NULL,
  model = NULL,
  architecture = "pleiotropic",
  add_QTN_num = NULL,
  dom_QTN_num = NULL,
  epi_QTN_num = NULL,
  epi_type = NULL,
  epi_interaction = 2,
  pleio_a = NULL,
  pleio_d = NULL,
  pleio_e = NULL,
  trait_spec_a_QTN_num = NULL,
  trait_spec_d_QTN_num = NULL,
  trait_spec_e_QTN_num = NULL,
  add_effect = NULL,
  big_add_QTN_effect = NULL,
  same_add_dom_QTN = FALSE,
  dom_effect = NULL,
  degree_of_dom = 1,
  epi_effect = NULL,
  type_of_ld = "indirect",
  ld_min = 0.2,
  ld_max = 0.8,
  ld_method = "composite",
  sim_method = "geometric",
  vary_QTN = FALSE,
  cor = NULL,
  cor_res = NULL,
  QTN_variance = FALSE,
  seed = NULL,
  home_dir = NULL,
  output_dir = NULL,
  export_gt = FALSE,
  output_format = "long",
  to_r = FALSE,
  out_geno = NULL,
  chr_prefix = "chr",
  remove_QTN = FALSE,
  warning_file_saver = TRUE,
  constraints = list(maf_above = NULL, maf_below = NULL, hets = NULL),
  maf_cutoff = NULL,
  nrows = Inf,
  na_string = "NA",
  SNP_effect = "Add",
  SNP_impute = "Middle",
  quiet = FALSE,
  verbose = TRUE,
  RNGversion = "3.5.1"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_phenotypes_+3A_geno_obj">geno_obj</code></td>
<td>
<p>Marker data set loaded as an R object.
Currently either HapMap or numericalized files
(code as aa = -1, Aa = 0 and AA = 1, e.g. 'data(&quot;SNP55K_maize282_maf04&quot;)')
are accepted. These and other file formats (VCF, GDS, and Plink Bed/Ped
files) may be read from file with 'geno_file' or 'geno_path'. Only one of
'geno_obj', 'geno_file' or 'geno_path' should be provided.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_geno_file">geno_file</code></td>
<td>
<p>Name of a marker data set to be read from a file. If in a 
different folder, the whole path should be provided. Formats accepted are
Numeric, HapMap, VCF, GDS, and Plink Bed/Ped files. Notice that the major
allele will always be 1 and the minor allele -1. Thus, when using Plink Bed
files, the dosage information will be converted to the opposite value.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_geno_path">geno_path</code></td>
<td>
<p>Path to a folder containing the marker data set
file/files (e.g., separated by chromosome). Formats accepted are:
Numeric, HapMap, VCF, GDS, and Plink Bed/Ped files</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_qtn_list">QTN_list</code></td>
<td>
<p>A list of specific markers to be used as QTNs. If one wants to specify the QTNs instead of selecting them randomly, at least one of the following elements should be provided: 'QTN_list$add', 'QTN_list$dom', and/or 'QTN_list$epi'. The element '$add', '$dom', and '$epi' are lists containing a vector of markers for each of the traits to be simulated. For example, to simulate 2 traits controlled by 1 pleiotropic and 2 trait-specific additive QTNs, the user would create a list of marker names 'marker_list &lt;- list(add = list(trait1 = c(&quot;marker1&quot;, &quot;marker2&quot;, &quot;marker3&quot;), trait2 = c(&quot;marker1&quot;, &quot;marker4&quot;, &quot;marker5&quot;)))' and set 'QTN_list = marker_list'. On the other hand, to simulate a single trait controlled by 1 additive and 2 dominance QTNs, the marker list would be 'marker_list &lt;- list(add = list(&quot;marker1&quot;), dom = list(c(&quot;marker2&quot;, &quot;marker3&quot;)))'. Notice that these vectors with maker names is used in the order they appear. For instance, in the list 'marker_list &lt;- list(add = list(trait9 = c(&quot;marker1&quot;), trait4 = c(&quot;marker5&quot;)))', the vector names itself (&quot;trait9&quot; and &quot;trait4&quot;) are ignored and &quot;trait9&quot; will be the vector of markers used to simulate the first trait and &quot;trait4&quot; will be the vector of markers used to simulate the second trait. Also, when using 'QTN_list', many parameters used for selecting QTNs will be ignored (e.g., 'constraints').</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_prefix">prefix</code></td>
<td>
<p>If 'geno_path' points to a folder with files other than the
marker data set, a part of the data set name may be used to select the desired
files (e.g., prefix = &quot;Chr&quot; would read files Chr1.hmp.txt, ..., Chr10.hmp.txt
but not HapMap.hmp.txt).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_rep">rep</code></td>
<td>
<p>The number of experiments (replicates of a trait with the same
genetic architecture) to be simulated.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_ntraits">ntraits</code></td>
<td>
<p>The number of multi-trait phenotypes to simulate under
pleiotropic, partially [pleiotropic], and LD (spurious pleiotropy)
architectures (see 'architecture'). If not assigned, a single trait will be
simulated. Currently, the only option for the LD architecture is
'ntraits = 2'.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_h2">h2</code></td>
<td>
<p>The heritability for each traits being simulated.
It could be either a vector with length equals to 'ntraits',
or a matrix with ncol equals to 'ntraits'. If the later is used, the simulation
will loop over the number of rows and will generate a result for each row.
If a single trait is being simulated and h2 is a vector,
one simulation of each heritability value will be conducted. Either none or
all traits are expected to have 'h2 = 0'.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_mean">mean</code></td>
<td>
<p>A vector with the mean (intercept) value for each of the simulated traits. If omitted, the simulated traits will be centered to zero.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_model">model</code></td>
<td>
<p>The genetic model to be assumed. The options are
&quot;A&quot; (additive), &quot;D&quot; (dominance), &quot;E&quot; (epistatic)
as well as any combination of those models such as &quot;AE&quot;, &quot;DE&quot; or &quot;ADE&quot;.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_architecture">architecture</code></td>
<td>
<p>The genetic architecture to be simulated. Should be provided
if &lsquo;ntraits' &gt; 1. Possible options are: &rsquo;pleiotropic' (default), for traits being
controlled by the same QTNs; 'partially', for traits being controlled by
pleiotropic and trait-specific QTNs; 'LD', for traits being exclusively
controlled by different QTNs in &quot;direct&quot; or &quot;indirect&quot; (See 'type_of_ld', 'ld_min', and
'ld_max' below) linkage disequilibrium. Currently the
only option for 'architecture = &quot;LD&quot;' is 'ntraits = 2'.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_add_qtn_num">add_QTN_num</code></td>
<td>
<p>The number of additive quantitative trait nucleotides
(QTNs) to be simulated.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_dom_qtn_num">dom_QTN_num</code></td>
<td>
<p>The number of dominance QTNs to be simulated.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_epi_qtn_num">epi_QTN_num</code></td>
<td>
<p>The number of epistatic (Currently, only additive x
additive epistasis are simulated) QTNs to be simulated.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_epi_type">epi_type</code></td>
<td>
<p>to be implemented...</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_epi_interaction">epi_interaction</code></td>
<td>
<p>Number of markers that compose an epistatic QTN. 
If 'epi_interaction = 2' (default), a 2-way interaction (marker1 x marker2) will be 
used to simulate epistatic QTNs. If 'epi_interaction = 3' a 3-way interaction (marker1 x marker2 x marker3) will be used instead.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_pleio_a">pleio_a</code></td>
<td>
<p>The number of pleiotropic additive QTNs to be
used if 'architecture = &quot;partially&quot;'. When 'sim_method = custom' (see below),
the first effects will be assigned to the pleiotropic QTNs and the last to
the trait-specific ones. For instance, in a scenario where ntraits = 2,
pleio_a = 2, trait_spec_a_QTN_num = 1, and add_effect = list(
trait1 = c(0.1, 0.2, 0.3), trait2 = c(0.4, 0.5, 0.6)), the trait-specific
QTNs for trait 1 and trait 2 will be 0.3 and 0.6, respectively. The first
two allelic effects will be assigned to the pleiotropic QTNs.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_pleio_d">pleio_d</code></td>
<td>
<p>The number of pleiotropic dominance QTNs to be
used if 'architecture = &quot;partially&quot;' (See pleio_a for details).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_pleio_e">pleio_e</code></td>
<td>
<p>The number of pleiotropic epistatic QTNs to be
used if 'architecture = &quot;partially&quot;' (See pleio_a for details).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_trait_spec_a_qtn_num">trait_spec_a_QTN_num</code></td>
<td>
<p>The number of trait-specific additive QTNs if
'architecture = &quot;partially&quot;'. It should be a vector of length equals
to 'ntraits'.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_trait_spec_d_qtn_num">trait_spec_d_QTN_num</code></td>
<td>
<p>The number of trait-specific dominance QTNs if
'architecture = &quot;partially&quot;'. It should be a vector of length equals
to 'ntraits'.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_trait_spec_e_qtn_num">trait_spec_e_QTN_num</code></td>
<td>
<p>The number of trait-specific epistatic QTNs if
'architecture = &quot;partially&quot;'. It should be a vector of length equals
to 'ntraits'.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_add_effect">add_effect</code></td>
<td>
<p>Additive effect size to be simulated. It may be either
a vector (assuming 'ntraits' = 1 or one allelic effect per trait to create a
geometric series ['sim_method = &quot;geometric&quot;']) or a list
of length = 'ntraits', i.e., if 'ntraits' &gt; 1, a list with one vector of
additive effects should be provided for each trait. Unless
'big_add_QTN_effect' is provided, the length of each vector
should be equal to the number of additive QTNs being simulated.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_big_add_qtn_effect">big_add_QTN_effect</code></td>
<td>
<p>Additive effect size for one possible major
effect quantitative trait nucleotide. If 'ntraits' &gt; 1,
big_add_QTN_effect should have length equals 'ntraits'.
If 'add_QTN_num' &gt; 1, this large effect will be assigned to the fist QTN.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_same_add_dom_qtn">same_add_dom_QTN</code></td>
<td>
<p>A boolean for selecting markers to be both additive
and dominance QTNs.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_dom_effect">dom_effect</code></td>
<td>
<p>Similar to the 'add_effect', it could be either
a vector or a list. Optional if 'same_add_dom_QTN = TRUE'.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_degree_of_dom">degree_of_dom</code></td>
<td>
<p>If the same set of QTNs are being used for simulating
additive and dominance effects, the dominance allelic effect could be a
proportion of the additive allelic effect.
In other words, 'degree_of_dom' equals to 0.5, 1, 1.5 will simulate,
partial dominance, complete dominance and overdominance, respectively.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_epi_effect">epi_effect</code></td>
<td>
<p>Epistatic (additive x additive) effect size to be
simulated. Similar to the 'add_effect', it could be either a vector or
a list.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_type_of_ld">type_of_ld</code></td>
<td>
<p>Type of LD used to simulate spurious pleiotropy. If
&quot;indirect&quot; (default), an intermediate marker is selected from which two
adjacent markers (one upstream and another downstream) will be chosen based
on its LD with the intermediate marker to be the QTNs. Optionally,
in the &quot;direct&quot; method, one marker is selected to be a QTN for trait 1, and
a second marker is selected based on its LD with the first selected marker to
be the QTN for trait 2.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_ld_min">ld_min</code></td>
<td>
<p>Minimum Linkage disequilibrium for selecting QTNs when
'architecture = LD'. The default is 'ld_min = 0.2' (markers should have a minimum LD of
0.2 to be used as QTNs).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_ld_max">ld_max</code></td>
<td>
<p>Maximum Linkage disequilibrium for selecting QTNs when
'architecture = LD'. The default is 'ld_max = 0.8' (markers should have an LD of
at maximum 0.8 to be used as QTNs).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_ld_method">ld_method</code></td>
<td>
<p>Four methods can be used to calculate linkage disequilibrium values: &quot;composite&quot; for LD composite measure (Default), &quot;r&quot; for R coefficient (by EM algorithm assuming HWE, it could be negative), &quot;dprime&quot; for D', and &quot;corr&quot; for correlation coefficient (see snpgdsLDpair from package SNPRelate).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_sim_method">sim_method</code></td>
<td>
<p>Provide the method of simulating allelic effects.
The options available are &quot;geometric&quot; and &quot;custom&quot;. For multiple QTNs,
a geometric series may be simulated, i.e., if add_effect = 0.5,
the effect size of the first QTNs will be 0.5, the effect size of the second
QTN will be 0.5^2, and the effect of the n^th QTN will be 0.5^n.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_vary_qtn">vary_QTN</code></td>
<td>
<p>A boolean that determines if the same set of quantitative trait
nucleotide (QTN) should be used to generate genetic effects for each
experiment ('vary_QTN = FALSE') or  if a different set of QTNs should be
used for each replication ('vary_QTN = TRUE').</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_cor">cor</code></td>
<td>
<p>Option to simulate traits with a predefined genetic correlation.
It should be a correlation matrix with a number of rows = 'ntraits'.
Default = NULL. Notice that when opting for controlling the correlation, the
genetic effects are transformed using Cholesky decomposition. In this case,
the correlation of genetic effects for different traits will be as provided, 
but due to the transformation, the actual allelic effects of correlated
traits may be different than the input allelic effect.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_cor_res">cor_res</code></td>
<td>
<p>Option to simulate traits with a predefined residual
correlation. It should be a correlation matrix with number of
rows = 'ntraits'. If NULL, an identity matrix (independent residuals)
will be used.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_qtn_variance">QTN_variance</code></td>
<td>
<p>Whether or not the percentage of the phenotypic variance
explained by each QTN (QTN variance / phenotypic variance) should be
exported. The default is FALSE. Notice that this is calculated prior to any
transformation, such as the whitening/coloring transformation used to assign
user-specified correlation to the genetic effect. In may not reflect the
actual variance explained when the data is transformed.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_seed">seed</code></td>
<td>
<p>Value to be used by set.seed. If NULL (default),
runif(1, 0, 1000000) will be used. Notice that at each sampling step,
a different seed generated based on the 'seed' parameter used.
For example, if one uses 'seed = 123', when simulating the 10th replication
of trait 1, the seed to be used is 'round( (123 * 10 * 10) * 1)'. On the
other hand, for simulating the 21st replication of trait 2, the seed to be
used will be 'round( (123 * 21 * 21) * 2)'. The master seed (unique value required to reproduce  results) is saved at the top of the log file. Unless verbose = FALSE the actual seed used in every  simulation is exported along with simulated phenotypes.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_home_dir">home_dir</code></td>
<td>
<p>Directory where files should be saved. It may be
home_dir = getwd().</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_output_dir">output_dir</code></td>
<td>
<p>Name to be used to create a folder inside 'home_dir' and
save output files.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_export_gt">export_gt</code></td>
<td>
<p>If TRUE genotypes of selected QTNs will be saved at file.
If FALSE (default), only the QTN information will be saved.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_output_format">output_format</code></td>
<td>
<p>Four options are available for saving simulated
phenotypes: 'multi-file', saves each simulation in a separate file;
'long' (default for multiple traits), appends each experiment (rep) to the
last one (by row); 'wide', saves experiments by column (default for single
trait) and 'gemma', saves .fam files to be used by gemma or other software
that uses plink bed files. (renaming .fam file with the same name of the bim
and bed files is necessary).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_to_r">to_r</code></td>
<td>
<p>Option for outputting the simulated results as an R data.frame in
addition to saving it to file. If TRUE, results need to be assigned to an
R object (see vignette).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_out_geno">out_geno</code></td>
<td>
<p>Optionally saves the numericalized genotype either as &quot;numeric&quot; (see
vignettes for an example data), &quot;BED&quot; or &quot;gds&quot;. The default is NULL.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_chr_prefix">chr_prefix</code></td>
<td>
<p>If input file format is VCF and out_geno = &quot;BED&quot;, and a prefix
is used in the chromosomes names, chr_prefix may be used to avoid issues in
converting to bed files (e.g., chr_prefix = &quot;chr&quot; in &quot;chr01&quot;).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_remove_qtn">remove_QTN</code></td>
<td>
<p>Whether or not a copy of the genotypic file should be saved
without the simulated QTNs. The default is FALSE. If 'vary_QTN = TRUE', the
question &quot;Are you sure that you want to save one genotypic file/rep
(remove_QTN = TRUE and vary_QTN = TRUE) [type yes or no] ?&quot; will pop up to
avoid saving multiple large files unintentionally</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_warning_file_saver">warning_file_saver</code></td>
<td>
<p>Skips the interactive question and saves all files
when 'remove_QTN = TRUE' and 'vary_QTN = TRUE'.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_constraints">constraints</code></td>
<td>
<p>Set constraints for QTN selection. Currently, the options
are maf_above (the minimum value of minor allele frequency, a double between
0 - 0.5), maf_below (the maximum value of minor allele frequency, a double
between 0 - 0.5),and hets ('include' and 'remove'). All of these options
are NULL by default ('list(maf_above = NULL, maf_below = NULL, hets = NULL )'
). For instance, if the parameters used are 
'constraints = list(maf_above = 0.3, maf_below = 0.44, hets = &quot;include&quot;)',
only heterozygote markers with minor allele frequency between 0.3 and 0.44 will
be selected to be QTNs. The option &quot;remove&quot; would only select homozygote
markers to be QTNs.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_maf_cutoff">maf_cutoff</code></td>
<td>
<p>Option for filtering the data set based on minor allele
frequency (Not to be confounded with the constraints option which will only
filter possible QTNs). It may be useful when outputting the genotypic data
set.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_nrows">nrows</code></td>
<td>
<p>Option for loading only part of a data set. Used when marker
data is in numeric or HapMap format. Please see data.table::fread for details.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_na_string">na_string</code></td>
<td>
<p>Tell create_phenotypes what character represents missing
data (default is &quot;NA&quot;). Used when the input marker data is numeric or
HapMap.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_snp_effect">SNP_effect</code></td>
<td>
<p>Parameter used for numericalization. The options are: Add
(AA = 1, Aa = 0, aa = -1),  Dom (AA = -1, Aa = 0, aa = -1), Left (AA = 1,
Aa = -1, aa = -1), Right (AA = 1, Aa = 1, aa = -1). The default option is Add.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_snp_impute">SNP_impute</code></td>
<td>
<p>Naive imputation for HapMap numericalization. The options
are: Major (NA &lt;- 1), Middle (NA &lt;- 0), and Minor (NA &lt;- -1).</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_quiet">quiet</code></td>
<td>
<p>Whether or not the log file should pop up into R once the
simulation is done.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_verbose">verbose</code></td>
<td>
<p>If FALSE, suppress all prints and suppress individual seed numbers from being saved to file. The master seed (unique value required to reproduce results) is saved at the top of the log file.</p>
</td></tr>
<tr><td><code id="create_phenotypes_+3A_rngversion">RNGversion</code></td>
<td>
<p>Parameter to set the random number generator. Different R versions may be selected, the default value is '3.5.1'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Single or multi-trait phenotypes in one of many formats.
Numericalized marker data set with or without the selected QTNs.
Diagnostic files (log, QTN information, summary of LD between QTNs,
proportion of phenotypic variance explained by each QTN).
</p>


<h3>Author(s)</h3>

<p>Samuel B Fernandes and Alexander E Lipka
Last update: Jan 19, 2021
</p>


<h3>References</h3>

<p>Fernandes, S.B., and Lipka, A.E., 2020 simplePHENOTYPES: SIMulation of pleiotropic, linked and epistatic
SIMulation of Pleiotropic, Linked and Epistatic PHENOTYPES. BMC Bioinformatics 21(1):491,
doi: <a href="https://doi.org/10.1186/s12859-020-03804-y">10.1186/s12859-020-03804-y</a> <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate 50 replications of a single phenotype.
data("SNP55K_maize282_maf04")
pheno &lt;- 
  create_phenotypes(
    geno_obj = SNP55K_maize282_maf04,
    add_QTN_num = 3,
    add_effect = 0.2,
    big_add_QTN_effect = 0.9,
    rep = 10,
    h2 = 0.7,
    model = "A",
    to_r = TRUE,
    home_dir = tempdir(),
    quiet = T
    )
# For more examples, please run the following:
# vignette("simplePHENOTYPES")

</code></pre>

<hr>
<h2 id='file_loader'>Function used by create_phenotypes to load marker dataset.</h2><span id='topic+file_loader'></span>

<h3>Description</h3>

<p>Function used by create_phenotypes to load marker dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>file_loader(
  geno_obj = NULL,
  geno_file = NULL,
  geno_path = NULL,
  nrows = Inf,
  na_string = "NA",
  prefix = NULL,
  SNP_impute = "Middle",
  SNP_effect = "Add",
  verbose = TRUE,
  chr_prefix = "chr"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="file_loader_+3A_geno_obj">geno_obj</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="file_loader_+3A_geno_file">geno_file</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="file_loader_+3A_geno_path">geno_path</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="file_loader_+3A_nrows">nrows</code></td>
<td>
<p>= Inf,</p>
</td></tr>
<tr><td><code id="file_loader_+3A_na_string">na_string</code></td>
<td>
<p>= &quot;NA&quot;,</p>
</td></tr>
<tr><td><code id="file_loader_+3A_prefix">prefix</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="file_loader_+3A_snp_impute">SNP_impute</code></td>
<td>
<p>= &quot;Middle&quot;,</p>
</td></tr>
<tr><td><code id="file_loader_+3A_snp_effect">SNP_effect</code></td>
<td>
<p>= &quot;Add&quot;,</p>
</td></tr>
<tr><td><code id="file_loader_+3A_verbose">verbose</code></td>
<td>
<p>= TRUE</p>
</td></tr>
<tr><td><code id="file_loader_+3A_chr_prefix">chr_prefix</code></td>
<td>
<p>= &quot;chrm&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genotype data sampled
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes
Last update: Apr 20, 2020
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;
</p>

<hr>
<h2 id='genetic_effect'>Calculate genetic value based on QTN objects.</h2><span id='topic+genetic_effect'></span>

<h3>Description</h3>

<p>Calculate genetic value based on QTN objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genetic_effect(
  add_obj = NULL,
  dom_obj = NULL,
  epi_obj = NULL,
  add_effect = NULL,
  dom_effect = NULL,
  epi_effect = NULL,
  epi_interaction = NULL,
  sim_method = NULL,
  add = NULL,
  dom = NULL,
  epi = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genetic_effect_+3A_add_obj">add_obj</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_dom_obj">dom_obj</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_epi_obj">epi_obj</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_add_effect">add_effect</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_dom_effect">dom_effect</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_epi_effect">epi_effect</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_epi_interaction">epi_interaction</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_sim_method">sim_method</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_add">add</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_dom">dom</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genetic_effect_+3A_epi">epi</code></td>
<td>
<p>= NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of Genetic values
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes
Last update: Apr 20, 2020
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;genetic_effect&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-
</p>

<hr>
<h2 id='genotypes'>Generate a numeric (dosage) HapMap file</h2><span id='topic+genotypes'></span>

<h3>Description</h3>

<p>Generate a numeric (dosage) HapMap file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotypes(
  geno_obj = NULL,
  geno_file = NULL,
  geno_path = NULL,
  nrows = Inf,
  na_string = "NA",
  prefix = NULL,
  maf_cutoff = NULL,
  SNP_effect = "Add",
  SNP_impute = "Middle",
  verbose = TRUE,
  chr_prefix = "chr"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genotypes_+3A_geno_obj">geno_obj</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genotypes_+3A_geno_file">geno_file</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genotypes_+3A_geno_path">geno_path</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genotypes_+3A_nrows">nrows</code></td>
<td>
<p>= Inf,</p>
</td></tr>
<tr><td><code id="genotypes_+3A_na_string">na_string</code></td>
<td>
<p>= &quot;NA&quot;,</p>
</td></tr>
<tr><td><code id="genotypes_+3A_prefix">prefix</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genotypes_+3A_maf_cutoff">maf_cutoff</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="genotypes_+3A_snp_effect">SNP_effect</code></td>
<td>
<p>= 'Add',</p>
</td></tr>
<tr><td><code id="genotypes_+3A_snp_impute">SNP_impute</code></td>
<td>
<p>= 'Middle',</p>
</td></tr>
<tr><td><code id="genotypes_+3A_verbose">verbose</code></td>
<td>
<p>= verbose</p>
</td></tr>
<tr><td><code id="genotypes_+3A_chr_prefix">chr_prefix</code></td>
<td>
<p>= &quot;chrm&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric HapMap
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes and Alexander Lipka
Last update: Apr 20, 2020
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>

<hr>
<h2 id='make_pd'>obtain a positive definite matrix</h2><span id='topic+make_pd'></span>

<h3>Description</h3>

<p>obtain a positive definite matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_pd(m, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_pd_+3A_m">m</code></td>
<td>
<p>square matrix</p>
</td></tr>
<tr><td><code id="make_pd_+3A_verbose">verbose</code></td>
<td>
<p>= TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>positive definite matrix
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes
Last update: Jan 5, 2021
</p>

<hr>
<h2 id='numericalization'>Converts character SNP genotype to numerical (-1, 0, 1) were 1 is the major
allele.
Modified by Samuel Fernandes from:
Alexander E. Lipka, Feng Tian, Qishan Wang, Jason Peiffer, Meng Li,
Peter J. Bradbury, Michael A. Gore, Edward S. Buckler, Zhiwu Zhang
GAPIT: genome association and prediction integrated tool, Bioinformatics,
Volume 28, Issue 18, 15 September 2012, Pages 2397–2399.</h2><span id='topic+numericalization'></span>

<h3>Description</h3>

<p>Converts character SNP genotype to numerical (-1, 0, 1) were 1 is the major
allele.
Modified by Samuel Fernandes from:
Alexander E. Lipka, Feng Tian, Qishan Wang, Jason Peiffer, Meng Li,
Peter J. Bradbury, Michael A. Gore, Edward S. Buckler, Zhiwu Zhang
GAPIT: genome association and prediction integrated tool, Bioinformatics,
Volume 28, Issue 18, 15 September 2012, Pages 2397–2399.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>numericalization(x, bit = NULL, effect = "Add", impute = "None")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="numericalization_+3A_x">x</code></td>
<td>
<p>ggg</p>
</td></tr>
<tr><td><code id="numericalization_+3A_bit">bit</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="numericalization_+3A_effect">effect</code></td>
<td>
<p>= 'Add',</p>
</td></tr>
<tr><td><code id="numericalization_+3A_impute">impute</code></td>
<td>
<p>= 'None',</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Corresponding numerical value
Last update: Apr 20, 2020
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;numericalization&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>

<hr>
<h2 id='phenotypes'>Generate environmental effects based on a given heritability</h2><span id='topic+phenotypes'></span>

<h3>Description</h3>

<p>Generate environmental effects based on a given heritability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phenotypes(
  base_line_trait = NULL,
  h2 = NULL,
  rep = NULL,
  seed = NULL,
  ntraits = NULL,
  output_format = NULL,
  fam = NULL,
  to_r = NULL,
  rep_by = NULL,
  hets = NULL,
  verbose = TRUE,
  QTN_variance = FALSE,
  add = NULL,
  dom = NULL,
  epi = NULL,
  cor_res = NULL,
  mean = NULL,
  cor = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phenotypes_+3A_base_line_trait">base_line_trait</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_h2">h2</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_rep">rep</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_seed">seed</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_ntraits">ntraits</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_output_format">output_format</code></td>
<td>
<p>= 'multi-file'</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_fam">fam</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_to_r">to_r</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_rep_by">rep_by</code></td>
<td>
<p>= 'QTN'</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_hets">hets</code></td>
<td>
<p>= 'QTN'</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_verbose">verbose</code></td>
<td>
<p>= TRUE</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_qtn_variance">QTN_variance</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="phenotypes_+3A_cor_res">cor_res</code></td>
<td>
<p>= NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Phenotypes for ntraits traits
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes and Alexander Lipka
Last update: Apr 20, 2020
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-phenotypes&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>

<hr>
<h2 id='qtn_from_user'>Select SNPs to be assigned as QTNs</h2><span id='topic+qtn_from_user'></span>

<h3>Description</h3>

<p>Select SNPs to be assigned as QTNs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtn_from_user(
  genotypes = NULL,
  QTN_list,
  export_gt = NULL,
  architecture = NULL,
  same_add_dom_QTN = NULL,
  add = NULL,
  dom = NULL,
  epi = NULL,
  epi_type = NULL,
  epi_interaction = 2,
  ntraits = NULL,
  type_of_ld = NULL,
  ld_method = "composite",
  gdsfile = NULL,
  ld_min = NULL,
  ld_max = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtn_from_user_+3A_genotypes">genotypes</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_qtn_list">QTN_list</code></td>
<td>
<p>...</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_export_gt">export_gt</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_architecture">architecture</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_same_add_dom_qtn">same_add_dom_QTN</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_add">add</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_dom">dom</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_epi">epi</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_epi_type">epi_type</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_epi_interaction">epi_interaction</code></td>
<td>
<p>= 2,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_ntraits">ntraits</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_type_of_ld">type_of_ld</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_ld_method">ld_method</code></td>
<td>
<p>= &quot;composite&quot;,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_gdsfile">gdsfile</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_ld_min">ld_min</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_from_user_+3A_ld_max">ld_max</code></td>
<td>
<p>= NULL,</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype of selected SNPs
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes and Alexander Lipka
Last update: Jan 7, 2021
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash; qtn_from_user &mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-
</p>

<hr>
<h2 id='qtn_linkage'>Select SNPs to be assigned as QTNs</h2><span id='topic+qtn_linkage'></span>

<h3>Description</h3>

<p>Select SNPs to be assigned as QTNs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtn_linkage(
  genotypes = NULL,
  seed = NULL,
  add_QTN_num = NULL,
  dom_QTN_num = NULL,
  ld_max = NULL,
  ld_min = NULL,
  ld_method = "composite",
  gdsfile = NULL,
  constraints = list(maf_above = NULL, maf_below = NULL),
  rep = NULL,
  rep_by = NULL,
  export_gt = NULL,
  same_add_dom_QTN = NULL,
  add = NULL,
  dom = NULL,
  type_of_ld = NULL,
  verbose = verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtn_linkage_+3A_genotypes">genotypes</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_seed">seed</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_add_qtn_num">add_QTN_num</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_dom_qtn_num">dom_QTN_num</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_ld_max">ld_max</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_ld_min">ld_min</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_ld_method">ld_method</code></td>
<td>
<p>Four methods can be used to calculate linkage disequilibrium values: &quot;composite&quot; for LD composite measure, &quot;r&quot; for R coefficient (by EM algorithm assuming HWE, it could be negative), &quot;dprime&quot; for D', and &quot;corr&quot; for correlation coefficient.</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_gdsfile">gdsfile</code></td>
<td>
<p>NULL</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_constraints">constraints</code></td>
<td>
<p>= list(maf_above = NULL, maf_below = NULL)</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_rep">rep</code></td>
<td>
<p>= 1,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_rep_by">rep_by</code></td>
<td>
<p>= 'QTN',</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_export_gt">export_gt</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_same_add_dom_qtn">same_add_dom_QTN</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_add">add</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_dom">dom</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_type_of_ld">type_of_ld</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="qtn_linkage_+3A_verbose">verbose</code></td>
<td>
<p>= verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype of selected SNPs
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes
Last update: Apr 20, 2020
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash; QTN_linkage &mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>

<hr>
<h2 id='qtn_partially_pleiotropic'>Select SNPs to be assigned as QTNs</h2><span id='topic+qtn_partially_pleiotropic'></span>

<h3>Description</h3>

<p>Select SNPs to be assigned as QTNs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtn_partially_pleiotropic(
  genotypes = NULL,
  seed = NULL,
  pleio_a = NULL,
  pleio_d = NULL,
  pleio_e = NULL,
  trait_spec_a_QTN_num = NULL,
  trait_spec_d_QTN_num = NULL,
  trait_spec_e_QTN_num = NULL,
  epi_type = NULL,
  epi_interaction = 2,
  ntraits = NULL,
  constraints = list(maf_above = NULL, maf_below = NULL, hets = NULL),
  rep = NULL,
  rep_by = NULL,
  export_gt = NULL,
  same_add_dom_QTN = NULL,
  add = NULL,
  dom = NULL,
  epi = NULL,
  verbose = verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtn_partially_pleiotropic_+3A_genotypes">genotypes</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_seed">seed</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_pleio_a">pleio_a</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_pleio_d">pleio_d</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_pleio_e">pleio_e</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_trait_spec_a_qtn_num">trait_spec_a_QTN_num</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_trait_spec_d_qtn_num">trait_spec_d_QTN_num</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_trait_spec_e_qtn_num">trait_spec_e_QTN_num</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_epi_type">epi_type</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_epi_interaction">epi_interaction</code></td>
<td>
<p>= 2,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_ntraits">ntraits</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_constraints">constraints</code></td>
<td>
<p>= list(maf_above = NULL, maf_below = NULL)</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_rep">rep</code></td>
<td>
<p>= 1,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_rep_by">rep_by</code></td>
<td>
<p>= 'QTN',</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_export_gt">export_gt</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_same_add_dom_qtn">same_add_dom_QTN</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_add">add</code></td>
<td>
<p>null</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_dom">dom</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_epi">epi</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="qtn_partially_pleiotropic_+3A_verbose">verbose</code></td>
<td>
<p>= verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype of selected SNPs
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes and Alexander Lipka
Last update: Apr 20, 2020
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash; QTN_partially_pleiotropic &mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-
</p>

<hr>
<h2 id='qtn_pleiotropic'>Select SNPs to be assigned as QTNs.</h2><span id='topic+qtn_pleiotropic'></span>

<h3>Description</h3>

<p>Select SNPs to be assigned as QTNs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtn_pleiotropic(
  genotypes = NULL,
  seed = NULL,
  same_add_dom_QTN = NULL,
  add_QTN_num = NULL,
  dom_QTN_num = NULL,
  epi_QTN_num = NULL,
  epi_type = NULL,
  epi_interaction = 2,
  constraints = list(maf_above = NULL, maf_below = NULL),
  rep = NULL,
  rep_by = NULL,
  export_gt = NULL,
  add = NULL,
  dom = NULL,
  epi = NULL,
  verbose = verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtn_pleiotropic_+3A_genotypes">genotypes</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_seed">seed</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_same_add_dom_qtn">same_add_dom_QTN</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_add_qtn_num">add_QTN_num</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_dom_qtn_num">dom_QTN_num</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_epi_qtn_num">epi_QTN_num</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_epi_type">epi_type</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_epi_interaction">epi_interaction</code></td>
<td>
<p>= 2,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_constraints">constraints</code></td>
<td>
<p>= list(maf_above = NULL, maf_below = NULL)</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_rep">rep</code></td>
<td>
<p>= 1,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_rep_by">rep_by</code></td>
<td>
<p>= 'QTN',</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_export_gt">export_gt</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_add">add</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_dom">dom</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_epi">epi</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="qtn_pleiotropic_+3A_verbose">verbose</code></td>
<td>
<p>= verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype of selected SNPs
</p>


<h3>Author(s)</h3>

<p>Samuel Fernandes and Alexander Lipka
Last update: Apr 20, 2020
</p>

<hr>
<h2 id='SNP55K_maize282_maf04'>Genotypic data for 282 inbred association panel using the 55K SNP array
(filtered by MAF&gt;0.4 to reduce file size).
Reference: Cook et al. 2012. Genetic architecture of maize kernel
composition in the nested association mapping and inbred association panels.
Plant Physiology 158:824-834</h2><span id='topic+SNP55K_maize282_maf04'></span>

<h3>Description</h3>

<p>Genotypic data for 282 inbred association panel using the 55K SNP array
(filtered by MAF&gt;0.4 to reduce file size).
Reference: Cook et al. 2012. Genetic architecture of maize kernel
composition in the nested association mapping and inbred association panels.
Plant Physiology 158:824-834
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNP55K_maize282_maf04
</code></pre>


<h3>Format</h3>

<p>A data frame with 10650 rows and 285 variables:
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.1104/pp.111.185033">10.1104/pp.111.185033</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
