<!DOCTYPE html><html><head><title>Help for package simulMGF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {simulMGF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#simGeno'>
<p>Function to simulate SNP matrix</p></a></li>
<li><a href='#simPheno'>
<p>Function to simulate phenotypes</p></a></li>
<li><a href='#simulFS'>
<p>Function to simulate full sib progeny genotype</p></a></li>
<li><a href='#simulHS'>
<p>Function to simulate half sib progeny genotypes</p></a></li>
<li><a href='#simulN'>
<p>Function to simulate a SNP matrix, a phenotypic trait and the effects of associated SNPs.</p></a></li>
<li><a href='#simulU'>
<p>Function to simulate a random SNP matrix, phenotype and QTLs with their effects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulate SNP Matrix, Phenotype and Genotypic Effects</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Martin Nahuel Garcia [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Martin Nahuel Garcia &lt;garcia.martin@inta.gob.ar&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulate genotypes in SNP (single nucleotide polymorphisms) Matrix as random numbers from an uniform distribution, for diploid organisms (coded by 0, 1, 2), Sikorska et al., (2013) &lt;<a href="https://doi.org/10.1186%2F1471-2105-14-166">doi:10.1186/1471-2105-14-166</a>&gt;, or half-sib/full-sib SNP matrix from real or simulated parents SNP data, assuming mendelian segregation. Simulate phenotypic traits for real or simulated SNP data, controlled by a specific number of quantitative trait loci  and their effects, sampled from a Normal or an Uniform distributions, assuming a pure additive model. This is useful for testing association and genomic prediction models or for educational purposes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mngar/simulMGF">https://github.com/mngar/simulMGF</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-28 17:10:40 UTC; Martin</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-02 22:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='simGeno'>
Function to simulate SNP matrix
</h2><span id='topic+simGeno'></span>

<h3>Description</h3>

<p>Simulate SNP matrix coded 0, 1 and 2; with random genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simGeno(Nind, Nmarkers)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simGeno_+3A_nind">Nind</code></td>
<td>

<p>number of individuals to simulate.
</p>
</td></tr>
<tr><td><code id="simGeno_+3A_nmarkers">Nmarkers</code></td>
<td>

<p>Nmarkers number of SNP markers to generate.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of dimensions Nind x Nmarkers.
</p>


<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#simulate 100 individuals and 1000 SNPs
set.seed(123)
simGeno(100, 1000)
#[1] "simG was generated"
dim(simG);simG[1:5,1:5]
#[1]  100 1000
#[,1] [,2] [,3] [,4] [,5]
#[1,]    0    1    0    2    2
#[2,]    2    0    2    0    0
#[3,]    1    1    1    2    2
#[4,]    2    2    1    2    1
#[5,]    2    1    1    1    1
</code></pre>

<hr>
<h2 id='simPheno'>
Function to simulate phenotypes
</h2><span id='topic+simPheno'></span>

<h3>Description</h3>

<p>Simulate a phenotype from a genotype matrix with QTLs with random effects sampled from a Normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simPheno(x, Nqtl, Esigma, Pmean, Perror)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simPheno_+3A_x">x</code></td>
<td>

<p>SNP matrix coded like 0 homozygous; 1 heterozygous; 2 homozygous
</p>
</td></tr>
<tr><td><code id="simPheno_+3A_nqtl">Nqtl</code></td>
<td>

<p>number of QTLs to simulate
</p>
</td></tr>
<tr><td><code id="simPheno_+3A_esigma">Esigma</code></td>
<td>

<p>standard deviation of effects with distribution N~(0,Esigma^2)
</p>
</td></tr>
<tr><td><code id="simPheno_+3A_pmean">Pmean</code></td>
<td>

<p>phenotype mean
</p>
</td></tr>
<tr><td><code id="simPheno_+3A_perror">Perror</code></td>
<td>

<p>standard deviation of error (portion of phenotype not explained by genomic information)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class list containing the trait, the markers associated and their effects.
</p>
<table>
<tr><td><code>pheno</code></td>
<td>
<p>vector with the trait values simulated.</p>
</td></tr>
<tr><td><code>QTN</code></td>
<td>
<p>column in the SNP matrix with the SNP associated.</p>
</td></tr>
<tr><td><code>Meffects</code></td>
<td>
<p>effects of the associated SNPs.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
simGeno(100, 1000)
#' #[1] "simG was generated"
simPheno(simG, 50, .8, 12, .5)
#[1] "simP was generated"
str(simP)
#List of 3
#$ pheno   : num [1:100, 1] 24 20.5 15.6 13.6 18.5 ...
#$ QTN     : int [1:50] 568 474 529 349 45 732 416 51 413 514 ...
#$ Meffects: num [1:50] 0.2396 -0.138 0.906 0.0186 1.0687 ...
</code></pre>

<hr>
<h2 id='simulFS'>
Function to simulate full sib progeny genotype
</h2><span id='topic+simulFS'></span>

<h3>Description</h3>

<p>Simulate full sib progeny genotypes from the genotype of the parents (matrixes with the same dimensions). Pair of parents mating will be in the order of the matrixes. We assume that these are diploid organisms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulFS(x, y, Nprogeny)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulFS_+3A_x">x</code></td>
<td>

<p>genotype matrix of a set of moms
</p>
</td></tr>
<tr><td><code id="simulFS_+3A_y">y</code></td>
<td>

<p>genotype matrix of a set of dads
</p>
</td></tr>
<tr><td><code id="simulFS_+3A_nprogeny">Nprogeny</code></td>
<td>

<p>Nprogeny number of progeny's genotypes to generate from each pair of parents
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of dimensions (nrow(x)*Nprogeny) x ncol(x)
</p>


<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#simulate 100 individuals and 1000 SNPs
set.seed(123)
simGeno(100, 1000)
#[1] "simG was generated"
#simulate the genotype of 5 FS from 3 pairs of parents
simulFS(simG[1:3,],simG[4:6,],5)
#[1] "simulatedFS was generated"
 dim(simulatedFS)
#[1]   15 1000
# The first 5 individuals are progeny of mom 1 and dad 1, the second 5 individuals
# are progeny of mom 2 and dad 2, and so on.
</code></pre>

<hr>
<h2 id='simulHS'>
Function to simulate half sib progeny genotypes
</h2><span id='topic+simulHS'></span>

<h3>Description</h3>

<p>Simulate half sib progeny from one genotyped parent assuming a random genotype for the other parental. We assume that these are diploid organisms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulHS(x, Nprogeny)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulHS_+3A_x">x</code></td>
<td>

<p>genotype matrix of a set of moms
</p>
</td></tr>
<tr><td><code id="simulHS_+3A_nprogeny">Nprogeny</code></td>
<td>

<p>number of progeny's genotypes to simulate for each mom
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function assume: a diploid organism; mendelian segregation of alleles; and independent segregation.
</p>


<h3>Value</h3>

<p>a matrix of dimensions (nrow(x)*Nprogeny) x ncol(x)
</p>


<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' #simulate 100 individuals and 1000 SNPs
set.seed(123)
simGeno(100, 1000)
#[1] "simG was generated"
#simulate the genotype of 3 sets 5 HS (one set by mom)
simulHS(simG[1:3,],5)
#[1] "simulatedHS was generated"
dim(simulatedHS)
#[1]   15 1000
</code></pre>

<hr>
<h2 id='simulN'>
Function to simulate a SNP matrix, a phenotypic trait and the effects of associated SNPs.
</h2><span id='topic+simulN'></span>

<h3>Description</h3>

<p>This function simulate a SNP matrix (coded as 0, 1, 2) and traits with a selected number of QTLs and their effects that will be sampled from a Normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulN(Nind, Nmarkers, Nqtl, Esigma, Pmean, Perror)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulN_+3A_nind">Nind</code></td>
<td>

<p>number of individuals to simulate.
</p>
</td></tr>
<tr><td><code id="simulN_+3A_nmarkers">Nmarkers</code></td>
<td>

<p>number of SNP markers to generate.
</p>
</td></tr>
<tr><td><code id="simulN_+3A_nqtl">Nqtl</code></td>
<td>

<p>number of QTLs controlling the trait.
</p>
</td></tr>
<tr><td><code id="simulN_+3A_esigma">Esigma</code></td>
<td>

<p>standard deviation of effects with distribution N~(0,Esigma^2).
</p>
</td></tr>
<tr><td><code id="simulN_+3A_pmean">Pmean</code></td>
<td>

<p>phenotype mean.
</p>
</td></tr>
<tr><td><code id="simulN_+3A_perror">Perror</code></td>
<td>

<p>standard deviation of error (portion of phenotype not explained by genomic information).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Genotypic data is simulated as the round value sampled from an uniform distribution with interval (-.5,2.5). Phenotypic data are obtained as a linear function defined by:
</p>
<p style="text-align: center;"><code class="reqn">y = Pmean + \sum QTN*Meffects + Perror </code>
</p>



<h3>Value</h3>

<p>An object of class list containing the SNP matrix, the trait, the markers associated and their effects.
</p>
<table>
<tr><td><code>geno</code></td>
<td>
<p>SNP matrix generated.</p>
</td></tr>
<tr><td><code>pheno</code></td>
<td>
<p>vector with the trait values simulated.</p>
</td></tr>
<tr><td><code>QTN</code></td>
<td>
<p>column in the SNP matrix with the SNP associated.</p>
</td></tr>
<tr><td><code>Meffects</code></td>
<td>
<p>effects of the associated SNPs.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The genotype is simulated in the same way of simGeno function.
The trait, QTLs and their effects are simulated in the same way of simPheno function.
</p>


<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>See Also</h3>

<p>simGeno, simPheno, simulU
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
simulN(100, 1000, 50, .9, 12, .5)
#[1] "nsimout was generated"
str(nsimout)
#List of 4
#$ geno    : num [1:100, 1:1000] 0 2 1 2 2 0 1 2 1 1 ...
#$ pheno   : num [1:100, 1] 25.4 21.6 16 13.8 19.4 ...
#$ QTN     : int [1:50] 568 474 529 349 45 732 416 51 413 514 ...
#$ Meffects: num [1:50] 0.2696 -0.1552 1.0192 0.0209 1.2023 ...
</code></pre>

<hr>
<h2 id='simulU'>
Function to simulate a random SNP matrix, phenotype and QTLs with their effects
</h2><span id='topic+simulU'></span>

<h3>Description</h3>

<p>This function simulate a SNP matrix (coded as 0, 1, 2) and traits with a selected number of QTLs and their effects that will be sampled from a Uniform distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulU(Nind, Nmarkers, Nqtl, Pmean, Perror)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulU_+3A_nind">Nind</code></td>
<td>

<p>number of individuals to simulate.
</p>
</td></tr>
<tr><td><code id="simulU_+3A_nmarkers">Nmarkers</code></td>
<td>

<p>number of SNP markers to generate.
</p>
</td></tr>
<tr><td><code id="simulU_+3A_nqtl">Nqtl</code></td>
<td>

<p>number of QTLs controlling the trait.
</p>
</td></tr>
<tr><td><code id="simulU_+3A_pmean">Pmean</code></td>
<td>

<p>phenotype mean.
</p>
</td></tr>
<tr><td><code id="simulU_+3A_perror">Perror</code></td>
<td>

<p>standard deviation of error (portion of phenotype not explained by genomic information).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class list containing the SNP matrix, the trait, the markers associated and their effects.
</p>
<table>
<tr><td><code>geno</code></td>
<td>
<p>SNP matrix generated.</p>
</td></tr>
<tr><td><code>pheno</code></td>
<td>
<p>vector with the trait values simulated.</p>
</td></tr>
<tr><td><code>QTN</code></td>
<td>
<p>column in the SNP matrix with the SNP associated.</p>
</td></tr>
<tr><td><code>Meffects</code></td>
<td>
<p>effects of the associated SNPs.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>See Also</h3>

<p>simGeno, simulN
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
simulU(100, 1000, 50, 12, .5)
#[1] "usimout was generated"
str(usimout)
#List of 4
#$ geno    : num [1:100, 1:1000] 0 2 1 2 2 0 1 2 1 1 ...
#$ pheno   : num [1:100, 1] 10.3 14.7 11.8 10.2 13.1 ...
#$ QTN     : int [1:50] 568 474 529 349 45 732 416 51 413 514 ...
#$ Meffects: num [1:50] 0.2355 0.0158 -0.1369 -0.1246 0.7426 ...
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
