<!DOCTYPE html><html lang="en"><head><title>Help for package sinaplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sinaplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#blood'><p>Expression data from 2095 AML/ALL and healthy bone marrow cells.</p></a></li>
<li><a href='#sinaplot'><p>sinaplot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>An Enhanced Chart for Simple and Truthful Representation of
Single Observations over Multiple Classes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-04-10</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nikos Sidiropoulos &lt;nikos.sidiro@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The sinaplot is a data visualization chart suitable for plotting
    any single variable in a multiclass data set. It is an enhanced jitter strip
    chart, where the width of the jitter is controlled by the density
    distribution of the data within each class.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.1.0), plyr(&ge; 1.8.4)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, RColorBrewer</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-04-21 14:28:00 UTC; bfh139</td>
</tr>
<tr>
<td>Author:</td>
<td>Nikos Sidiropoulos [aut, cre],
  Sina Hadi Sohi [aut],
  Nicolas Rapin [aut],
  Frederik Otzen Bagger [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-04-21 17:58:31 UTC</td>
</tr>
</table>
<hr>
<h2 id='blood'>Expression data from 2095 AML/ALL and healthy bone marrow cells.</h2><span id='topic+blood'></span>

<h3>Description</h3>

<p>Expression data from 2095 AML/ALL and healthy bone marrow cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(blood)
</code></pre>


<h3>Format</h3>

<p>A data frame with 2095 rows and 2
columns (Class (AML/ALL subtype), Gene expression values).</p>


<h3>Source</h3>

<p><a href="http://servers.binf.ku.dk/bloodspot/">http://servers.binf.ku.dk/bloodspot/</a>
</p>
<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13159">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13159</a>
</p>
<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15434">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15434</a>
</p>
<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE61804">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE61804</a>
</p>
<p><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE14468">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE14468</a>
</p>
<p><a href="http://cancergenome.nih.gov/">http://cancergenome.nih.gov/</a>
</p>

<hr>
<h2 id='sinaplot'>sinaplot</h2><span id='topic+sinaplot'></span><span id='topic+sinaplot.default'></span><span id='topic+sinaplot.formula'></span>

<h3>Description</h3>

<p>The SinaPlot is a data visualization chart suitable for plotting
any single variable in a multiclass dataset. It is an enhanced jitter strip
chart, where the width of the jitter is controlled by the density
distribution of the data within each class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sinaplot(x, ...)

## Default S3 method:
sinaplot(x, groups = NULL, method = c("density",
  "counts"), scale = TRUE, adjust = 0.75, bins = 50, bin_limit = 1,
  maxwidth = 1, seed = NULL, plot = TRUE, add = FALSE, log = FALSE,
  labels = NULL, xlab = "", ylab = "", col = NULL, pch = NULL, ...)

## S3 method for class 'formula'
sinaplot(formula, data = NULL, ..., subset,
  na.action = NULL, xlab, ylab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sinaplot_+3A_x">x</code></td>
<td>
<p>numeric vector or a data frame or a list of numeric vectors to be
plotted.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_groups">groups</code></td>
<td>
<p>optional vector of <code>length(x)</code>.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_method">method</code></td>
<td>
<p>choose the method to spread the samples within the same
bin along the x-axis. Available methods: &quot;density&quot; and &quot;counts&quot;.
See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_scale">scale</code></td>
<td>
<p>a logical that indicates whether the width of each group should
be scaled relative to the group with the highest density.
Default: <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_adjust">adjust</code></td>
<td>
<p>adjusts the bandwidth of the density kernel when
<code>method == "density"</code> (see <code><a href="stats.html#topic+density">density</a></code>).</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_bins">bins</code></td>
<td>
<p>number of bins to divide the y-axis into when
<code>method == "counts"</code>. Default: 50.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_bin_limit">bin_limit</code></td>
<td>
<p>if the samples within the same y-axis bin are more
than <code>bin_limit</code>, the samples's X coordinates will be adjusted.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_maxwidth">maxwidth</code></td>
<td>
<p>control the maximum width the points can spread into. Values
between 0 and 1.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_seed">seed</code></td>
<td>
<p>a single value that controls the random sample jittering. Set to
an integer to enable plot reproducibility. Default NULL.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_plot">plot</code></td>
<td>
<p>logical. When <code>TRUE</code> the sinaplot is produced, otherwise the
function returns the new sample coordinates. Default: <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_add">add</code></td>
<td>
<p>logical. If true add boxplot to current plot.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_log">log</code></td>
<td>
<p>logical. If true it uses a logarihmic scale on the y-axis.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_labels">labels</code></td>
<td>
<p>labels for each group. Recycled if necessary. By default,
these are inferred from the data.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_xlab">xlab</code>, <code id="sinaplot_+3A_ylab">ylab</code></td>
<td>
<p>axis labels.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_pch">pch</code>, <code id="sinaplot_+3A_col">col</code></td>
<td>
<p>plotting characters and colors, specified by group.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_formula">formula</code></td>
<td>
<p>a formula, such as y ~ grp, where y is a numeric vector of
data values to be split into groups according to the grouping variable grp
(usually a factor).</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_data">data</code></td>
<td>
<p>a data.frame (or list) from which the variables in formula should
be taken.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to be
used for plotting.</p>
</td></tr>
<tr><td><code id="sinaplot_+3A_na.action">na.action</code></td>
<td>
<p>a function which indicates what should happen when the data
contain NAs. The default is to ignore missing values in either the response
or the group.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are two available ways to define the x-axis borders for the
samples to spread within:
</p>

<ul>
<li><p><code>method = "density"</code>
</p>
<p>A density kernel is estimated along the y-axis for every sample group. The
borders are then defined by the density curve. Tuning parameter
<code>adjust</code> can be used to control the density bandwidth in the same way
it is used in <code><a href="stats.html#topic+density">density</a></code>. 
</p>
</li>
<li><p><code>method = "counts"</code>:
</p>
<p>The borders are defined by the number of samples that occupy the same
bin and the parameter <code>maxwidth</code> in the following fashion:
</p>
<p><code>xBorder = nsamples * maxwidth</code>
</p>

</li></ul>



<h3>Value</h3>

<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p>discrete x-coordinates, split by group</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>input values</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>input groups</p>
</td></tr>
<tr><td><code>scaled</code></td>
<td>
<p>final x-coordinates, adjusted by sinaplot</p>
</td></tr>
</table>
<p><code>NULL</code>
</p>
<p><code>NULL</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## sinaplot on a formula:

data("blood", package = "sinaplot")
boxplot(Gene ~ Class, data = blood)
sinaplot(Gene ~ Class, data = blood, pch = 20, add = TRUE)

## sinaplot on a data.frame:

df &lt;- data.frame(Uni05 = (1:100)/21, Norm = rnorm(100),
                  `5T` = rt(100, df = 5), Gam2 = rgamma(100, shape = 2))
boxplot(df)
sinaplot(df, add = TRUE, pch = 20)

## sinaplot on a list:

bimodal &lt;- c(rnorm(300, -2, 0.6), rnorm(300, 2, 0.6))
uniform &lt;- runif(500, -4, 4)
normal &lt;- rnorm(800,0,3)

distributions &lt;- list(uniform = uniform, bimodal = bimodal, normal = normal)
boxplot(distributions, col = 2:4)
sinaplot(distributions, add = TRUE, pch = 20)

## sinaplot on a vector:

x &lt;- c(rnorm(200, 4, 1), rnorm(200, 5, 2), rnorm(400, 6, 1.5))
groups &lt;- c(rep("Cond1", 200), rep("Cond2", 200), rep("Cond3", 400))

sinaplot(x, groups)

par(mfrow = c(2, 2))

sinaplot(x, groups, pch = 20, col = 2:4)
sinaplot(x, groups, scale = FALSE, pch = 20, col = 2:4)
sinaplot(x, groups, scale = FALSE, adjust = 1/6, pch = 20, col = 2:4)
sinaplot(x, groups, scale = FALSE, adjust = 3, pch = 20, col = 2:4)

#blood

par(mfrow = c(1,1))
sinaplot(blood$Gene, blood$Class)

old.mar &lt;- par()$mar
par(mar = c(9,4,4,2) + 0.1)
groups &lt;- levels(blood$Class)

sinaplot(blood$Gene, blood$Class, pch = 20, xaxt = "n", col = rainbow(18))
axis(1, at = 1:length(groups), labels = FALSE)
text(1:length(groups), y = par()$usr[3] - 0.1 * (par()$usr[4] - par()$usr[3]),
     xpd = TRUE, srt = 45, adj = 1, labels = groups)
par(mar = old.mar)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
