<!DOCTYPE html><html><head><title>Help for package singleCellHaystack</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {singleCellHaystack}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dat.expression'><p>Single cell RNA-seq dataset.</p></a></li>
<li><a href='#dat.tsne'><p>Single cell tSNE coordingates.</p></a></li>
<li><a href='#default_bandwidth.nrd'><p>Default function given by function bandwidth.nrd in MASS. No changes were made to this function.</p></a></li>
<li><a href='#extract_row_dgRMatrix'><p>Returns a row of a sparse matrix of class dgRMatrix.</p>
Function made by Ben Bolker and Ott Toomet (see https://stackoverflow.com/questions/47997184/)</a></li>
<li><a href='#extract_row_lgRMatrix'><p>Returns a row of a sparse matrix of class lgRMatrix.</p>
Function made by Ben Bolker and Ott Toomet (see https://stackoverflow.com/questions/47997184/)</a></li>
<li><a href='#get_D_KL'><p>Calculates the Kullback-Leibler divergence between distributions.</p></a></li>
<li><a href='#get_D_KL_continuous_highD'><p>Calculates the Kullback-Leibler divergence between distributions for the high-dimensional continuous version of haystack.</p></a></li>
<li><a href='#get_D_KL_highD'><p>Calculates the Kullback-Leibler divergence between distributions for the high-dimensional version of haystack().</p></a></li>
<li><a href='#get_density'><p>Function to get the density of points with value TRUE in the (x,y) plot</p></a></li>
<li><a href='#get_dist_two_sets'><p>Calculate the pairwise Euclidean distances between the rows of 2 matrices.</p></a></li>
<li><a href='#get_euclidean_distance'><p>Calculate the Euclidean distance between x and y.</p></a></li>
<li><a href='#get_grid_points'><p>A function to decide grid points in a higher-dimensional space</p></a></li>
<li><a href='#get_log_p_D_KL'><p>Estimates the significance of the observed Kullback-Leibler divergence by comparing to randomizations.</p></a></li>
<li><a href='#get_log_p_D_KL_continuous'><p>Estimates the significance of the observed Kullback-Leibler divergence by comparing to randomizations for the continuous version of haystack.</p></a></li>
<li><a href='#get_parameters_haystack'><p>Function that decides most of the parameters that will be used during the &quot;Haystack&quot; analysis.</p></a></li>
<li><a href='#get_reference'><p>Get reference distribution</p></a></li>
<li><a href='#haystack'><p>The main Haystack function</p></a></li>
<li><a href='#haystack_2D'><p>The main Haystack function, for 2-dimensional spaces.</p></a></li>
<li><a href='#haystack_continuous_highD'><p>The main Haystack function, for higher-dimensional spaces and continuous expression levels.</p></a></li>
<li><a href='#haystack_highD'><p>The main Haystack function, for higher-dimensional spaces.</p></a></li>
<li><a href='#hclust_haystack'><p>Function for hierarchical clustering of genes according to their expression distribution in 2D or multi-dimensional space</p></a></li>
<li><a href='#hclust_haystack_highD'><p>Function for hierarchical clustering of genes according to their distribution in a higher-dimensional space.</p></a></li>
<li><a href='#hclust_haystack_raw'><p>Function for hierarchical clustering of genes according to their distribution on a 2D plot.</p></a></li>
<li><a href='#kde2d_faster'><p>Based on the MASS kde2d() function, but heavily simplified; it's just tcrossprod() now.</p></a></li>
<li><a href='#kmeans_haystack'><p>Function for k-means clustering of genes according to their expression distribution in 2D or multi-dimensional space</p></a></li>
<li><a href='#kmeans_haystack_highD'><p>Function for k-means clustering of genes according to their distribution in a higher-dimensional space.</p></a></li>
<li><a href='#kmeans_haystack_raw'><p>Function for k-means clustering of genes according to their distribution on a 2D plot.</p></a></li>
<li><a href='#plot_compare_ranks'><p>plot_compare_ranks</p></a></li>
<li><a href='#plot_gene_haystack'><p>Visualizing the detection/expression of a gene in a 2D plot</p></a></li>
<li><a href='#plot_gene_haystack_raw'><p>Visualizing the detection/expression of a gene in a 2D plot</p></a></li>
<li><a href='#plot_gene_set_haystack'><p>Visualizing the detection/expression of a set of genes in a 2D plot</p></a></li>
<li><a href='#plot_gene_set_haystack_raw'><p>Visualizing the detection/expression of a set of genes in a 2D plot</p></a></li>
<li><a href='#plot_rand_fit'><p>plot_rand_fit</p></a></li>
<li><a href='#plot_rand_KLD'><p>plot_rand_KLD</p></a></li>
<li><a href='#read_haystack'><p>Function to read haystack results from file.</p></a></li>
<li><a href='#show_result_haystack'><p>show_result_haystack</p></a></li>
<li><a href='#singleCellHaystack-package'><p>singleCellHaystack: A Universal Differential Expression Prediction Tool for Single-Cell and Spatial Genomics Data</p></a></li>
<li><a href='#write_haystack'><p>Function to write haystack result data to file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Universal Differential Expression Prediction Tool for
Single-Cell and Spatial Genomics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>One key exploratory analysis step in single-cell genomics data analysis
    is the prediction of features with different activity levels. For example, we want 
    to predict differentially expressed genes (DEGs) in single-cell RNA-seq data, 
    spatial DEGs in spatial transcriptomics data, or differentially accessible 
    regions (DARs) in single-cell ATAC-seq data. 'singleCellHaystack' predicts differentially
    active features in single cell omics datasets without relying on the clustering
    of cells into arbitrary clusters. 'singleCellHaystack' uses Kullback-Leibler 
    divergence to find features (e.g., genes, genomic regions, etc) that are active
    in subsets of cells that are non-randomly positioned inside an input space (such as 
    1D trajectories, 2D tissue sections, multi-dimensional embeddings, etc). For 
    the theoretical background of 'singleCellHaystack' we refer to our original paper
    Vandenbon and Diez (Nature Communications, 2020) &lt;<a href="https://doi.org/10.1038%2Fs41467-020-17900-3">doi:10.1038/s41467-020-17900-3</a>&gt;
    and our update Vandenbon and Diez (Scientific Reports, 2023) &lt;<a href="https://doi.org/10.1038%2Fs41598-023-38965-2">doi:10.1038/s41598-023-38965-2</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, Matrix, splines, ggplot2, reshape2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, SummarizedExperiment,
SingleCellExperiment, SeuratObject, cowplot, wrswoR,
sparseMatrixStats, ComplexHeatmap, patchwork</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://alexisvdb.github.io/singleCellHaystack/">https://alexisvdb.github.io/singleCellHaystack/</a>,
<a href="https://github.com/alexisvdb/singleCellHaystack">https://github.com/alexisvdb/singleCellHaystack</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/alexisvdb/singleCellHaystack/issues">https://github.com/alexisvdb/singleCellHaystack/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-11 09:29:58 UTC; alex</td>
</tr>
<tr>
<td>Author:</td>
<td>Alexis Vandenbon <a href="https://orcid.org/0000-0003-2180-5732"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Diego Diez <a href="https://orcid.org/0000-0002-2325-4893"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alexis Vandenbon &lt;alexis.vandenbon@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-11 10:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='dat.expression'>Single cell RNA-seq dataset.</h2><span id='topic+dat.expression'></span>

<h3>Description</h3>

<p>Single cell RNA-seq dataset.
</p>

<hr>
<h2 id='dat.tsne'>Single cell tSNE coordingates.</h2><span id='topic+dat.tsne'></span>

<h3>Description</h3>

<p>Single cell tSNE coordingates.
</p>

<hr>
<h2 id='default_bandwidth.nrd'>Default function given by function bandwidth.nrd in MASS. No changes were made to this function.</h2><span id='topic+default_bandwidth.nrd'></span>

<h3>Description</h3>

<p>Default function given by function bandwidth.nrd in MASS. No changes were made to this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_bandwidth.nrd(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="default_bandwidth.nrd_+3A_x">x</code></td>
<td>
<p>A numeric vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A suitable bandwith.
</p>

<hr>
<h2 id='extract_row_dgRMatrix'>Returns a row of a sparse matrix of class dgRMatrix.
Function made by Ben Bolker and Ott Toomet (see https://stackoverflow.com/questions/47997184/)</h2><span id='topic+extract_row_dgRMatrix'></span>

<h3>Description</h3>

<p>Returns a row of a sparse matrix of class dgRMatrix.
Function made by Ben Bolker and Ott Toomet (see https://stackoverflow.com/questions/47997184/)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_row_dgRMatrix(m, i = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_row_dgRMatrix_+3A_m">m</code></td>
<td>
<p>a sparse matrix of class dgRMatrix</p>
</td></tr>
<tr><td><code id="extract_row_dgRMatrix_+3A_i">i</code></td>
<td>
<p>the index of the row to return</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A row (numerical vector) of the sparse matrix
</p>

<hr>
<h2 id='extract_row_lgRMatrix'>Returns a row of a sparse matrix of class lgRMatrix.
Function made by Ben Bolker and Ott Toomet (see https://stackoverflow.com/questions/47997184/)</h2><span id='topic+extract_row_lgRMatrix'></span>

<h3>Description</h3>

<p>Returns a row of a sparse matrix of class lgRMatrix.
Function made by Ben Bolker and Ott Toomet (see https://stackoverflow.com/questions/47997184/)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_row_lgRMatrix(m, i = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_row_lgRMatrix_+3A_m">m</code></td>
<td>
<p>a sparse matrix of class lgRMatrix</p>
</td></tr>
<tr><td><code id="extract_row_lgRMatrix_+3A_i">i</code></td>
<td>
<p>the index of the row to return</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A row (logical vector) of the sparse matrix
</p>

<hr>
<h2 id='get_D_KL'>Calculates the Kullback-Leibler divergence between distributions.</h2><span id='topic+get_D_KL'></span>

<h3>Description</h3>

<p>Calculates the Kullback-Leibler divergence between distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_D_KL(classes, parameters, reference.prob, pseudo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_D_KL_+3A_classes">classes</code></td>
<td>
<p>A logical vector. Values are T is the gene is expressed in a cell, F is not.</p>
</td></tr>
<tr><td><code id="get_D_KL_+3A_parameters">parameters</code></td>
<td>
<p>Parameters of the analysis, as set by function 'get_parameters_haystack'</p>
</td></tr>
<tr><td><code id="get_D_KL_+3A_reference.prob">reference.prob</code></td>
<td>
<p>A reference distribution to calculate the divergence against.</p>
</td></tr>
<tr><td><code id="get_D_KL_+3A_pseudo">pseudo</code></td>
<td>
<p>A pseudocount, used to avoid log(0) problems.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical value, the Kullback-Leibler divergence
</p>

<hr>
<h2 id='get_D_KL_continuous_highD'>Calculates the Kullback-Leibler divergence between distributions for the high-dimensional continuous version of haystack.</h2><span id='topic+get_D_KL_continuous_highD'></span>

<h3>Description</h3>

<p>Calculates the Kullback-Leibler divergence between distributions for the high-dimensional continuous version of haystack.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_D_KL_continuous_highD(
  weights,
  density.contributions,
  reference.prob,
  pseudo = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_D_KL_continuous_highD_+3A_weights">weights</code></td>
<td>
<p>A numerical vector with expression values of a gene.</p>
</td></tr>
<tr><td><code id="get_D_KL_continuous_highD_+3A_density.contributions">density.contributions</code></td>
<td>
<p>A matrix of density contributions of each cell (rows) to each center point (columns).</p>
</td></tr>
<tr><td><code id="get_D_KL_continuous_highD_+3A_reference.prob">reference.prob</code></td>
<td>
<p>A reference distribution to calculate the divergence against.</p>
</td></tr>
<tr><td><code id="get_D_KL_continuous_highD_+3A_pseudo">pseudo</code></td>
<td>
<p>A pseudocount, used to avoid log(0) problems.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical value, the Kullback-Leibler divergence
</p>

<hr>
<h2 id='get_D_KL_highD'>Calculates the Kullback-Leibler divergence between distributions for the high-dimensional version of haystack().</h2><span id='topic+get_D_KL_highD'></span>

<h3>Description</h3>

<p>Calculates the Kullback-Leibler divergence between distributions for the high-dimensional version of haystack().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_D_KL_highD(classes, density.contributions, reference.prob, pseudo = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_D_KL_highD_+3A_classes">classes</code></td>
<td>
<p>A logical vector. Values are T is the gene is expressed in a cell, F is not.</p>
</td></tr>
<tr><td><code id="get_D_KL_highD_+3A_density.contributions">density.contributions</code></td>
<td>
<p>A matrix of density contributions of each cell (rows) to each center point (columns).</p>
</td></tr>
<tr><td><code id="get_D_KL_highD_+3A_reference.prob">reference.prob</code></td>
<td>
<p>A reference distribution to calculate the divergence against.</p>
</td></tr>
<tr><td><code id="get_D_KL_highD_+3A_pseudo">pseudo</code></td>
<td>
<p>A pseudocount, used to avoid log(0) problems.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical value, the Kullback-Leibler divergence
</p>

<hr>
<h2 id='get_density'>Function to get the density of points with value TRUE in the (x,y) plot</h2><span id='topic+get_density'></span>

<h3>Description</h3>

<p>Function to get the density of points with value TRUE in the (x,y) plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_density(
  x,
  y,
  detection,
  rows.subset = 1:nrow(detection),
  high.resolution = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_density_+3A_x">x</code></td>
<td>
<p>x-axis coordinates of cells in a 2D representation (e.g. resulting from PCA or t-SNE)</p>
</td></tr>
<tr><td><code id="get_density_+3A_y">y</code></td>
<td>
<p>y-axis coordinates of cells in a 2D representation</p>
</td></tr>
<tr><td><code id="get_density_+3A_detection">detection</code></td>
<td>
<p>A logical matrix or dgRMatrix showing which gens (rows) are detected in which cells (columns)</p>
</td></tr>
<tr><td><code id="get_density_+3A_rows.subset">rows.subset</code></td>
<td>
<p>Indices of the rows of 'detection' for which to get the densities. Default: all.</p>
</td></tr>
<tr><td><code id="get_density_+3A_high.resolution">high.resolution</code></td>
<td>
<p>Logical: should high resolution be used? Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 3-dimensional array (dim 1: genes/rows of expression, dim 2 and 3: x and y grid points) with density data
</p>

<hr>
<h2 id='get_dist_two_sets'>Calculate the pairwise Euclidean distances between the rows of 2 matrices.</h2><span id='topic+get_dist_two_sets'></span>

<h3>Description</h3>

<p>Calculate the pairwise Euclidean distances between the rows of 2 matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_dist_two_sets(set1, set2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_dist_two_sets_+3A_set1">set1</code></td>
<td>
<p>A numerical matrix.</p>
</td></tr>
<tr><td><code id="get_dist_two_sets_+3A_set2">set2</code></td>
<td>
<p>A numerical matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of pairwise distances between the rows of 2 matrices.
</p>

<hr>
<h2 id='get_euclidean_distance'>Calculate the Euclidean distance between x and y.</h2><span id='topic+get_euclidean_distance'></span>

<h3>Description</h3>

<p>Calculate the Euclidean distance between x and y.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_euclidean_distance(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_euclidean_distance_+3A_x">x</code></td>
<td>
<p>A numerical vector.</p>
</td></tr>
<tr><td><code id="get_euclidean_distance_+3A_y">y</code></td>
<td>
<p>A numerical vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical value, the Euclidean distance.
</p>

<hr>
<h2 id='get_grid_points'>A function to decide grid points in a higher-dimensional space</h2><span id='topic+get_grid_points'></span>

<h3>Description</h3>

<p>A function to decide grid points in a higher-dimensional space
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_grid_points(input, method = "centroid", grid.points = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_grid_points_+3A_input">input</code></td>
<td>
<p>A numerical matrix with higher-dimensional coordinates (columns) of points (rows)</p>
</td></tr>
<tr><td><code id="get_grid_points_+3A_method">method</code></td>
<td>
<p>The method to decide grid points. Should be &quot;centroid&quot; (default) or &quot;seeding&quot;.</p>
</td></tr>
<tr><td><code id="get_grid_points_+3A_grid.points">grid.points</code></td>
<td>
<p>The number of grid points to return. Default is 100.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Coordinates of grid points in the higher-dimensonal space.
</p>

<hr>
<h2 id='get_log_p_D_KL'>Estimates the significance of the observed Kullback-Leibler divergence by comparing to randomizations.</h2><span id='topic+get_log_p_D_KL'></span>

<h3>Description</h3>

<p>Estimates the significance of the observed Kullback-Leibler divergence by comparing to randomizations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_log_p_D_KL(T.counts, D_KL.observed, D_KL.randomized, output.dir = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_log_p_D_KL_+3A_t.counts">T.counts</code></td>
<td>
<p>The number of cells in which a gene is detected.</p>
</td></tr>
<tr><td><code id="get_log_p_D_KL_+3A_d_kl.observed">D_KL.observed</code></td>
<td>
<p>A vector of observed Kullback-Leibler divergences.</p>
</td></tr>
<tr><td><code id="get_log_p_D_KL_+3A_d_kl.randomized">D_KL.randomized</code></td>
<td>
<p>A matrix of Kullback-Leibler divergences of randomized datasets.</p>
</td></tr>
<tr><td><code id="get_log_p_D_KL_+3A_output.dir">output.dir</code></td>
<td>
<p>Optional parameter. Default is NULL. If not NULL, some files will be written to this directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of log10 p values, not corrected for multiple testing using the Bonferroni correction.
</p>

<hr>
<h2 id='get_log_p_D_KL_continuous'>Estimates the significance of the observed Kullback-Leibler divergence by comparing to randomizations for the continuous version of haystack.</h2><span id='topic+get_log_p_D_KL_continuous'></span>

<h3>Description</h3>

<p>Estimates the significance of the observed Kullback-Leibler divergence by comparing to randomizations for the continuous version of haystack.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_log_p_D_KL_continuous(
  D_KL.observed,
  D_KL.randomized,
  all.coeffVar,
  train.coeffVar,
  output.dir = NULL,
  spline.method = "ns"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_log_p_D_KL_continuous_+3A_d_kl.observed">D_KL.observed</code></td>
<td>
<p>A vector of observed Kullback-Leibler divergences.</p>
</td></tr>
<tr><td><code id="get_log_p_D_KL_continuous_+3A_d_kl.randomized">D_KL.randomized</code></td>
<td>
<p>A matrix of Kullback-Leibler divergences of randomized datasets.</p>
</td></tr>
<tr><td><code id="get_log_p_D_KL_continuous_+3A_all.coeffvar">all.coeffVar</code></td>
<td>
<p>Coefficients of variation of all genes. Used for fitting the Kullback-Leibler divergences.</p>
</td></tr>
<tr><td><code id="get_log_p_D_KL_continuous_+3A_train.coeffvar">train.coeffVar</code></td>
<td>
<p>Coefficients of variation of genes that will be used for fitting the Kullback-Leibler divergences.</p>
</td></tr>
<tr><td><code id="get_log_p_D_KL_continuous_+3A_output.dir">output.dir</code></td>
<td>
<p>Optional parameter. Default is NULL. If not NULL, some files will be written to this directory.</p>
</td></tr>
<tr><td><code id="get_log_p_D_KL_continuous_+3A_spline.method">spline.method</code></td>
<td>
<p>Method to use for fitting splines &quot;ns&quot; (default): natural splines, &quot;bs&quot;: B-splines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of log10 p values, not corrected for multiple testing using the Bonferroni correction.
</p>

<hr>
<h2 id='get_parameters_haystack'>Function that decides most of the parameters that will be used during the &quot;Haystack&quot; analysis.</h2><span id='topic+get_parameters_haystack'></span>

<h3>Description</h3>

<p>Function that decides most of the parameters that will be used during the &quot;Haystack&quot; analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_parameters_haystack(x, y, high.resolution = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_parameters_haystack_+3A_x">x</code></td>
<td>
<p>x-axis coordinates of cells in a 2D representation (e.g. resulting from PCA or t-SNE)</p>
</td></tr>
<tr><td><code id="get_parameters_haystack_+3A_y">y</code></td>
<td>
<p>y-axis coordinates of cells in a 2D representation</p>
</td></tr>
<tr><td><code id="get_parameters_haystack_+3A_high.resolution">high.resolution</code></td>
<td>
<p>Logical: should high resolution be used? Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing various parameters to use in the analysis.
</p>

<hr>
<h2 id='get_reference'>Get reference distribution</h2><span id='topic+get_reference'></span>

<h3>Description</h3>

<p>Get reference distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_reference(param, use.advanced.sampling = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_reference_+3A_param">param</code></td>
<td>
<p>Parameters of the analysis, as set by function 'get_parameters_haystack'</p>
</td></tr>
<tr><td><code id="get_reference_+3A_use.advanced.sampling">use.advanced.sampling</code></td>
<td>
<p>If NULL naive sampling is used. If a vector is given (of length = no. of cells) sampling is done according to the values in the vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two components, Q for the reference distribution and pseudo.
</p>

<hr>
<h2 id='haystack'>The main Haystack function</h2><span id='topic+haystack'></span><span id='topic+haystack.matrix'></span><span id='topic+haystack.data.frame'></span><span id='topic+haystack.Seurat'></span><span id='topic+haystack.SingleCellExperiment'></span>

<h3>Description</h3>

<p>The main Haystack function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haystack(x, ...)

## S3 method for class 'matrix'
haystack(
  x,
  expression,
  weights.advanced.Q = NULL,
  dir.randomization = NULL,
  scale = TRUE,
  grid.points = 100,
  grid.method = "centroid",
  ...
)

## S3 method for class 'data.frame'
haystack(
  x,
  expression,
  weights.advanced.Q = NULL,
  dir.randomization = NULL,
  scale = TRUE,
  grid.points = 100,
  grid.method = "centroid",
  ...
)

## S3 method for class 'Seurat'
haystack(
  x,
  coord,
  assay = "RNA",
  slot = "data",
  dims = NULL,
  cutoff = 1,
  method = NULL,
  weights.advanced.Q = NULL,
  ...
)

## S3 method for class 'SingleCellExperiment'
haystack(
  x,
  assay = "counts",
  coord = "TSNE",
  dims = NULL,
  cutoff = 1,
  method = NULL,
  weights.advanced.Q = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haystack_+3A_x">x</code></td>
<td>
<p>a matrix or other object from which coordinates of cells can be extracted.</p>
</td></tr>
<tr><td><code id="haystack_+3A_...">...</code></td>
<td>
<p>further parameters passed down to methods.</p>
</td></tr>
<tr><td><code id="haystack_+3A_expression">expression</code></td>
<td>
<p>a matrix with expression data of genes (rows) in cells (columns)</p>
</td></tr>
<tr><td><code id="haystack_+3A_weights.advanced.q">weights.advanced.Q</code></td>
<td>
<p>If NULL naive sampling is used. If a vector is given (of length = no. of cells) sampling is done according to the values in the vector.</p>
</td></tr>
<tr><td><code id="haystack_+3A_dir.randomization">dir.randomization</code></td>
<td>
<p>If NULL, no output is made about the random sampling step. If not NULL, files related to the randomizations are printed to this directory.</p>
</td></tr>
<tr><td><code id="haystack_+3A_scale">scale</code></td>
<td>
<p>Logical (default=TRUE) indicating whether input coordinates in x should be scaled to mean 0 and standard deviation 1.</p>
</td></tr>
<tr><td><code id="haystack_+3A_grid.points">grid.points</code></td>
<td>
<p>An integer specifying the number of centers (gridpoints) to be used for estimating the density distributions of cells. Default is set to 100.</p>
</td></tr>
<tr><td><code id="haystack_+3A_grid.method">grid.method</code></td>
<td>
<p>The method to decide grid points for estimating the density in the high-dimensional space. Should be &quot;centroid&quot; (default) or &quot;seeding&quot;.</p>
</td></tr>
<tr><td><code id="haystack_+3A_coord">coord</code></td>
<td>
<p>name of coordinates slot for specific methods.</p>
</td></tr>
<tr><td><code id="haystack_+3A_assay">assay</code></td>
<td>
<p>name of assay data for Seurat method.</p>
</td></tr>
<tr><td><code id="haystack_+3A_slot">slot</code></td>
<td>
<p>name of slot for assay data for Seurat method.</p>
</td></tr>
<tr><td><code id="haystack_+3A_dims">dims</code></td>
<td>
<p>dimensions from coord to use. By default, all.</p>
</td></tr>
<tr><td><code id="haystack_+3A_cutoff">cutoff</code></td>
<td>
<p>cutoff for detection.</p>
</td></tr>
<tr><td><code id="haystack_+3A_method">method</code></td>
<td>
<p>choose between highD (default) and 2D haystack.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;haystack&quot;
</p>

<hr>
<h2 id='haystack_2D'>The main Haystack function, for 2-dimensional spaces.</h2><span id='topic+haystack_2D'></span>

<h3>Description</h3>

<p>The main Haystack function, for 2-dimensional spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haystack_2D(
  x,
  y,
  detection,
  use.advanced.sampling = NULL,
  dir.randomization = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haystack_2D_+3A_x">x</code></td>
<td>
<p>x-axis coordinates of cells in a 2D representation (e.g. resulting from PCA or t-SNE)</p>
</td></tr>
<tr><td><code id="haystack_2D_+3A_y">y</code></td>
<td>
<p>y-axis coordinates of cells in a 2D representation</p>
</td></tr>
<tr><td><code id="haystack_2D_+3A_detection">detection</code></td>
<td>
<p>A logical matrix showing which genes (rows) are detected in which cells (columns)</p>
</td></tr>
<tr><td><code id="haystack_2D_+3A_use.advanced.sampling">use.advanced.sampling</code></td>
<td>
<p>If NULL naive sampling is used. If a vector is given (of length = no. of cells) sampling is done according to the values in the vector.</p>
</td></tr>
<tr><td><code id="haystack_2D_+3A_dir.randomization">dir.randomization</code></td>
<td>
<p>If NULL, no output is made about the random sampling step. If not NULL, files related to the randomizations are printed to this directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;haystack&quot;
</p>

<hr>
<h2 id='haystack_continuous_highD'>The main Haystack function, for higher-dimensional spaces and continuous expression levels.</h2><span id='topic+haystack_continuous_highD'></span>

<h3>Description</h3>

<p>The main Haystack function, for higher-dimensional spaces and continuous expression levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haystack_continuous_highD(
  x,
  expression,
  grid.points = 100,
  weights.advanced.Q = NULL,
  dir.randomization = NULL,
  scale = TRUE,
  grid.method = "centroid",
  randomization.count = 100,
  n.genes.to.randomize = 100,
  selection.method.genes.to.randomize = "heavytails",
  grid.coord = NULL,
  spline.method = "ns"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haystack_continuous_highD_+3A_x">x</code></td>
<td>
<p>Coordinates of cells in a 2D or higher-dimensional space. Rows represent cells, columns the dimensions of the space.</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_expression">expression</code></td>
<td>
<p>a matrix with expression data of genes (rows) in cells (columns)</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_grid.points">grid.points</code></td>
<td>
<p>An integer specifying the number of centers (grid points) to be used for estimating the density distributions of cells. Default is set to 100.</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_weights.advanced.q">weights.advanced.Q</code></td>
<td>
<p>(Default: NULL) Optional weights of cells for calculating a weighted distribution of expression.</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_dir.randomization">dir.randomization</code></td>
<td>
<p>If NULL, no output is made about the random sampling step. If not NULL, files related to the randomizations are printed to this directory.</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_scale">scale</code></td>
<td>
<p>Logical (default=TRUE) indicating whether input coordinates in x should be scaled to mean 0 and standard deviation 1.</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_grid.method">grid.method</code></td>
<td>
<p>The method to decide grid points for estimating the density in the high-dimensional space. Should be &quot;centroid&quot; (default) or &quot;seeding&quot;.</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_randomization.count">randomization.count</code></td>
<td>
<p>Number of randomizations to use. Default: 100</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_n.genes.to.randomize">n.genes.to.randomize</code></td>
<td>
<p>Number of genes to use in randomizations. Default: 100</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_selection.method.genes.to.randomize">selection.method.genes.to.randomize</code></td>
<td>
<p>Method used to select genes for randomization.</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_grid.coord">grid.coord</code></td>
<td>
<p>matrix of grid coordinates.</p>
</td></tr>
<tr><td><code id="haystack_continuous_highD_+3A_spline.method">spline.method</code></td>
<td>
<p>Method to use for fitting splines &quot;ns&quot; (default): natural splines, &quot;bs&quot;: B-splines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;haystack&quot;, including the results of the analysis, and the coordinates of the grid points used to estimate densities.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># using the toy example of the singleCellHaystack package

# running haystack
res &lt;- haystack(dat.tsne, dat.expression)
# list top 10 biased genes
show_result_haystack(res, n=10)
</code></pre>

<hr>
<h2 id='haystack_highD'>The main Haystack function, for higher-dimensional spaces.</h2><span id='topic+haystack_highD'></span>

<h3>Description</h3>

<p>The main Haystack function, for higher-dimensional spaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haystack_highD(
  x,
  detection,
  grid.points = 100,
  use.advanced.sampling = NULL,
  dir.randomization = NULL,
  scale = TRUE,
  grid.method = "centroid"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haystack_highD_+3A_x">x</code></td>
<td>
<p>Coordinates of cells in a 2D or higher-dimensional space. Rows represent cells, columns the dimensions of the space.</p>
</td></tr>
<tr><td><code id="haystack_highD_+3A_detection">detection</code></td>
<td>
<p>A logical matrix showing which genes (rows) are detected in which cells (columns)</p>
</td></tr>
<tr><td><code id="haystack_highD_+3A_grid.points">grid.points</code></td>
<td>
<p>An integer specifying the number of centers (grid points) to be used for estimating the density distributions of cells. Default is set to 100.</p>
</td></tr>
<tr><td><code id="haystack_highD_+3A_use.advanced.sampling">use.advanced.sampling</code></td>
<td>
<p>If NULL naive sampling is used. If a vector is given (of length = no. of cells) sampling is done according to the values in the vector.</p>
</td></tr>
<tr><td><code id="haystack_highD_+3A_dir.randomization">dir.randomization</code></td>
<td>
<p>If NULL, no output is made about the random sampling step. If not NULL, files related to the randomizations are printed to this directory.</p>
</td></tr>
<tr><td><code id="haystack_highD_+3A_scale">scale</code></td>
<td>
<p>Logical (default=TRUE) indicating whether input coordinates in x should be scaled to mean 0 and standard deviation 1.</p>
</td></tr>
<tr><td><code id="haystack_highD_+3A_grid.method">grid.method</code></td>
<td>
<p>The method to decide grid points for estimating the density in the high-dimensional space. Should be &quot;centroid&quot; (default) or &quot;seeding&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;haystack&quot;, including the results of the analysis, and the coordinates of the grid points used to estimate densities.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># I need to add some examples.
# A toy example will be added too.
</code></pre>

<hr>
<h2 id='hclust_haystack'>Function for hierarchical clustering of genes according to their expression distribution in 2D or multi-dimensional space</h2><span id='topic+hclust_haystack'></span><span id='topic+hclust_haystack.matrix'></span><span id='topic+hclust_haystack.data.frame'></span>

<h3>Description</h3>

<p>Function for hierarchical clustering of genes according to their expression distribution in 2D or multi-dimensional space
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hclust_haystack(
  x,
  expression,
  grid.coordinates,
  hclust.method = "ward.D",
  cor.method = "spearman",
  ...
)

## S3 method for class 'matrix'
hclust_haystack(
  x,
  expression,
  grid.coordinates,
  hclust.method = "ward.D",
  cor.method = "spearman",
  ...
)

## S3 method for class 'data.frame'
hclust_haystack(
  x,
  expression,
  grid.coordinates,
  hclust.method = "ward.D",
  cor.method = "spearman",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hclust_haystack_+3A_x">x</code></td>
<td>
<p>a matrix or other object from which coordinates of cells can be extracted.</p>
</td></tr>
<tr><td><code id="hclust_haystack_+3A_expression">expression</code></td>
<td>
<p>expression matrix.</p>
</td></tr>
<tr><td><code id="hclust_haystack_+3A_grid.coordinates">grid.coordinates</code></td>
<td>
<p>coordinates of the grid points.</p>
</td></tr>
<tr><td><code id="hclust_haystack_+3A_hclust.method">hclust.method</code></td>
<td>
<p>method used with hclust.</p>
</td></tr>
<tr><td><code id="hclust_haystack_+3A_cor.method">cor.method</code></td>
<td>
<p>method used with cor.</p>
</td></tr>
<tr><td><code id="hclust_haystack_+3A_...">...</code></td>
<td>
<p>further parameters passed down to methods.</p>
</td></tr>
</table>

<hr>
<h2 id='hclust_haystack_highD'>Function for hierarchical clustering of genes according to their distribution in a higher-dimensional space.</h2><span id='topic+hclust_haystack_highD'></span>

<h3>Description</h3>

<p>Function for hierarchical clustering of genes according to their distribution in a higher-dimensional space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hclust_haystack_highD(
  x,
  detection,
  genes,
  method = "ward.D",
  grid.coordinates = NULL,
  scale = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hclust_haystack_highD_+3A_x">x</code></td>
<td>
<p>Coordinates of cells in a 2D or higher-dimensional space. Rows represent cells, columns the dimensions of the space.</p>
</td></tr>
<tr><td><code id="hclust_haystack_highD_+3A_detection">detection</code></td>
<td>
<p>A logical matrix showing which genes (rows) are detected in which cells (columns)</p>
</td></tr>
<tr><td><code id="hclust_haystack_highD_+3A_genes">genes</code></td>
<td>
<p>A set of genes (of the 'detection' data) which will be clustered.</p>
</td></tr>
<tr><td><code id="hclust_haystack_highD_+3A_method">method</code></td>
<td>
<p>The method to use for hierarchical clustering. See '?hclust' for more information. Default: &quot;ward.D&quot;.</p>
</td></tr>
<tr><td><code id="hclust_haystack_highD_+3A_grid.coordinates">grid.coordinates</code></td>
<td>
<p>Coordinates of grid points in the same space as 'x', to be used to estimate densities for clustering.</p>
</td></tr>
<tr><td><code id="hclust_haystack_highD_+3A_scale">scale</code></td>
<td>
<p>whether to scale data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class hclust, describing a hierarchical clustering tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># to be added
</code></pre>

<hr>
<h2 id='hclust_haystack_raw'>Function for hierarchical clustering of genes according to their distribution on a 2D plot.</h2><span id='topic+hclust_haystack_raw'></span>

<h3>Description</h3>

<p>Function for hierarchical clustering of genes according to their distribution on a 2D plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hclust_haystack_raw(x, y, detection, genes, method = "ward.D")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hclust_haystack_raw_+3A_x">x</code></td>
<td>
<p>x-axis coordinates of cells in a 2D representation (e.g. resulting from PCA or t-SNE)</p>
</td></tr>
<tr><td><code id="hclust_haystack_raw_+3A_y">y</code></td>
<td>
<p>y-axis coordinates of cells in a 2D representation</p>
</td></tr>
<tr><td><code id="hclust_haystack_raw_+3A_detection">detection</code></td>
<td>
<p>A logical matrix showing which genes (rows) are detected in which cells (columns)</p>
</td></tr>
<tr><td><code id="hclust_haystack_raw_+3A_genes">genes</code></td>
<td>
<p>A set of genes (of the 'detection' data) which will be clustered.</p>
</td></tr>
<tr><td><code id="hclust_haystack_raw_+3A_method">method</code></td>
<td>
<p>The method to use for hierarchical clustering. See '?hclust' for more information. Default: &quot;ward.D&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class hclust, describing a hierarchical clustering tree.
</p>

<hr>
<h2 id='kde2d_faster'>Based on the MASS kde2d() function, but heavily simplified; it's just tcrossprod() now.</h2><span id='topic+kde2d_faster'></span>

<h3>Description</h3>

<p>Based on the MASS kde2d() function, but heavily simplified; it's just tcrossprod() now.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kde2d_faster(dens.x, dens.y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kde2d_faster_+3A_dens.x">dens.x</code></td>
<td>
<p>Contribution of all cells to densities of the x-axis grid points.</p>
</td></tr>
<tr><td><code id="kde2d_faster_+3A_dens.y">dens.y</code></td>
<td>
<p>Contribution of all cells to densities of the y-axis grid points.</p>
</td></tr>
</table>

<hr>
<h2 id='kmeans_haystack'>Function for k-means clustering of genes according to their expression distribution in 2D or multi-dimensional space</h2><span id='topic+kmeans_haystack'></span><span id='topic+kmeans_haystack.matrix'></span><span id='topic+kmeans_haystack.data.frame'></span>

<h3>Description</h3>

<p>Function for k-means clustering of genes according to their expression distribution in 2D or multi-dimensional space
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmeans_haystack(x, expression, grid.coordinates, k, ...)

## S3 method for class 'matrix'
kmeans_haystack(x, expression, grid.coordinates, k, ...)

## S3 method for class 'data.frame'
kmeans_haystack(x, expression, grid.coordinates, k, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kmeans_haystack_+3A_x">x</code></td>
<td>
<p>a matrix or other object from which coordinates of cells can be extracted.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_+3A_expression">expression</code></td>
<td>
<p>expression matrix.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_+3A_grid.coordinates">grid.coordinates</code></td>
<td>
<p>coordinates of the grid points.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_+3A_k">k</code></td>
<td>
<p>number of clusters.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_+3A_...">...</code></td>
<td>
<p>further parameters passed down to methods.</p>
</td></tr>
</table>

<hr>
<h2 id='kmeans_haystack_highD'>Function for k-means clustering of genes according to their distribution in a higher-dimensional space.</h2><span id='topic+kmeans_haystack_highD'></span>

<h3>Description</h3>

<p>Function for k-means clustering of genes according to their distribution in a higher-dimensional space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmeans_haystack_highD(
  x,
  detection,
  genes,
  grid.coordinates = NULL,
  k,
  scale = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kmeans_haystack_highD_+3A_x">x</code></td>
<td>
<p>Coordinates of cells in a 2D or higher-dimensional space. Rows represent cells, columns the dimensions of the space.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_highD_+3A_detection">detection</code></td>
<td>
<p>A logical matrix showing which genes (rows) are detected in which cells (columns)</p>
</td></tr>
<tr><td><code id="kmeans_haystack_highD_+3A_genes">genes</code></td>
<td>
<p>A set of genes (of the 'detection' data) which will be clustered.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_highD_+3A_grid.coordinates">grid.coordinates</code></td>
<td>
<p>Coordinates of grid points in the same space as 'x', to be used to estimate densities for clustering.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_highD_+3A_k">k</code></td>
<td>
<p>The number of clusters to return.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_highD_+3A_scale">scale</code></td>
<td>
<p>whether to scale data.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_highD_+3A_...">...</code></td>
<td>
<p>Additional parameters which will be passed on to the kmeans function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class kmeans, describing a clustering into 'k' clusters
</p>


<h3>Examples</h3>

<pre><code class='language-R'># to be added
</code></pre>

<hr>
<h2 id='kmeans_haystack_raw'>Function for k-means clustering of genes according to their distribution on a 2D plot.</h2><span id='topic+kmeans_haystack_raw'></span>

<h3>Description</h3>

<p>Function for k-means clustering of genes according to their distribution on a 2D plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmeans_haystack_raw(x, y, detection, genes, k, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kmeans_haystack_raw_+3A_x">x</code></td>
<td>
<p>x-axis coordinates of cells in a 2D representation (e.g. resulting from PCA or t-SNE)</p>
</td></tr>
<tr><td><code id="kmeans_haystack_raw_+3A_y">y</code></td>
<td>
<p>y-axis coordinates of cells in a 2D representation</p>
</td></tr>
<tr><td><code id="kmeans_haystack_raw_+3A_detection">detection</code></td>
<td>
<p>A logical matrix showing which genes (rows) are detected in which cells (columns)</p>
</td></tr>
<tr><td><code id="kmeans_haystack_raw_+3A_genes">genes</code></td>
<td>
<p>A set of genes (of the 'detection' data) which will be clustered.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_raw_+3A_k">k</code></td>
<td>
<p>The number of clusters to return.</p>
</td></tr>
<tr><td><code id="kmeans_haystack_raw_+3A_...">...</code></td>
<td>
<p>Additional parameters which will be passed on to the kmeans function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class kmeans, describing a clustering into 'k' clusters
</p>

<hr>
<h2 id='plot_compare_ranks'>plot_compare_ranks</h2><span id='topic+plot_compare_ranks'></span>

<h3>Description</h3>

<p>plot_compare_ranks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_compare_ranks(res1, res2, sort_by = "log.p.vals")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_compare_ranks_+3A_res1">res1</code></td>
<td>
<p>haystack result.</p>
</td></tr>
<tr><td><code id="plot_compare_ranks_+3A_res2">res2</code></td>
<td>
<p>haystack result.</p>
</td></tr>
<tr><td><code id="plot_compare_ranks_+3A_sort_by">sort_by</code></td>
<td>
<p>column to sort results (default: log.p.vals).</p>
</td></tr>
</table>

<hr>
<h2 id='plot_gene_haystack'>Visualizing the detection/expression of a gene in a 2D plot</h2><span id='topic+plot_gene_haystack'></span><span id='topic+plot_gene_haystack.matrix'></span><span id='topic+plot_gene_haystack.data.frame'></span><span id='topic+plot_gene_haystack.SingleCellExperiment'></span><span id='topic+plot_gene_haystack.Seurat'></span>

<h3>Description</h3>

<p>Visualizing the detection/expression of a gene in a 2D plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gene_haystack(x, ...)

## S3 method for class 'matrix'
plot_gene_haystack(x, dim1 = 1, dim2 = 2, ...)

## S3 method for class 'data.frame'
plot_gene_haystack(x, dim1 = 1, dim2 = 2, ...)

## S3 method for class 'SingleCellExperiment'
plot_gene_haystack(
  x,
  dim1 = 1,
  dim2 = 2,
  assay = "counts",
  coord = "TSNE",
  ...
)

## S3 method for class 'Seurat'
plot_gene_haystack(
  x,
  dim1 = 1,
  dim2 = 2,
  assay = "RNA",
  slot = "data",
  coord = "tsne",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_gene_haystack_+3A_x">x</code></td>
<td>
<p>a matrix or other object from which coordinates of cells can be extracted.</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_+3A_...">...</code></td>
<td>
<p>further parameters passed to plot_gene_haystack_raw().</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_+3A_dim1">dim1</code></td>
<td>
<p>column index or name of matrix for x-axis coordinates.</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_+3A_dim2">dim2</code></td>
<td>
<p>column index or name of matrix for y-axis coordinates.</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_+3A_assay">assay</code></td>
<td>
<p>name of assay data for Seurat method.</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_+3A_coord">coord</code></td>
<td>
<p>name of coordinates slot for specific methods.</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_+3A_slot">slot</code></td>
<td>
<p>name of slot for assay data for Seurat method.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_gene_haystack_raw'>Visualizing the detection/expression of a gene in a 2D plot</h2><span id='topic+plot_gene_haystack_raw'></span>

<h3>Description</h3>

<p>Visualizing the detection/expression of a gene in a 2D plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gene_haystack_raw(
  x,
  y,
  gene,
  expression,
  detection = NULL,
  high.resolution = FALSE,
  point.size = 1,
  order.by.signal = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_gene_haystack_raw_+3A_x">x</code></td>
<td>
<p>x-axis coordinates of cells in a 2D representation (e.g. resulting from PCA or t-SNE)</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_raw_+3A_y">y</code></td>
<td>
<p>y-axis coordinates of cells in a 2D representation</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_raw_+3A_gene">gene</code></td>
<td>
<p>name of a gene that is present in the input expression data, or a numerical index</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_raw_+3A_expression">expression</code></td>
<td>
<p>a logical/numerical matrix showing detection/expression of genes (rows) in cells (columns)</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_raw_+3A_detection">detection</code></td>
<td>
<p>an optional logical matrix showing detection of genes (rows) in cells (columns). If left as NULL, the density distribution of the gene is not plotted.</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_raw_+3A_high.resolution">high.resolution</code></td>
<td>
<p>logical (default: FALSE). If set to TRUE, the density plot will be of a higher resolution</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_raw_+3A_point.size">point.size</code></td>
<td>
<p>numerical value to set size of points in plot. Default is 1.</p>
</td></tr>
<tr><td><code id="plot_gene_haystack_raw_+3A_order.by.signal">order.by.signal</code></td>
<td>
<p>If TRUE, cells with higher signal will be put on the foreground in the plot. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>

<hr>
<h2 id='plot_gene_set_haystack'>Visualizing the detection/expression of a set of genes in a 2D plot</h2><span id='topic+plot_gene_set_haystack'></span><span id='topic+plot_gene_set_haystack.matrix'></span><span id='topic+plot_gene_set_haystack.data.frame'></span><span id='topic+plot_gene_set_haystack.SingleCellExperiment'></span><span id='topic+plot_gene_set_haystack.Seurat'></span>

<h3>Description</h3>

<p>Visualizing the detection/expression of a set of genes in a 2D plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gene_set_haystack(x, ...)

## S3 method for class 'matrix'
plot_gene_set_haystack(x, dim1 = 1, dim2 = 2, ...)

## S3 method for class 'data.frame'
plot_gene_set_haystack(x, dim1 = 1, dim2 = 2, ...)

## S3 method for class 'SingleCellExperiment'
plot_gene_set_haystack(
  x,
  dim1 = 1,
  dim2 = 2,
  assay = "counts",
  coord = "TSNE",
  ...
)

## S3 method for class 'Seurat'
plot_gene_set_haystack(
  x,
  dim1 = 1,
  dim2 = 2,
  assay = "RNA",
  slot = "data",
  coord = "tsne",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_gene_set_haystack_+3A_x">x</code></td>
<td>
<p>a matrix or other object from which coordinates of cells can be extracted.</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_+3A_...">...</code></td>
<td>
<p>further parameters passed to plot_gene_haystack_raw().</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_+3A_dim1">dim1</code></td>
<td>
<p>column index or name of matrix for x-axis coordinates.</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_+3A_dim2">dim2</code></td>
<td>
<p>column index or name of matrix for y-axis coordinates.</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_+3A_assay">assay</code></td>
<td>
<p>name of assay data for Seurat method.</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_+3A_coord">coord</code></td>
<td>
<p>name of coordinates slot for specific methods.</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_+3A_slot">slot</code></td>
<td>
<p>name of slot for assay data for Seurat method.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_gene_set_haystack_raw'>Visualizing the detection/expression of a set of genes in a 2D plot</h2><span id='topic+plot_gene_set_haystack_raw'></span>

<h3>Description</h3>

<p>Visualizing the detection/expression of a set of genes in a 2D plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gene_set_haystack_raw(
  x,
  y,
  genes = NA,
  detection,
  high.resolution = TRUE,
  point.size = 1,
  order.by.signal = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_gene_set_haystack_raw_+3A_x">x</code></td>
<td>
<p>x-axis coordinates of cells in a 2D representation (e.g. resulting from PCA or t-SNE)</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_raw_+3A_y">y</code></td>
<td>
<p>y-axis coordinates of cells in a 2D representation</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_raw_+3A_genes">genes</code></td>
<td>
<p>Gene names that are present in the input expression data, or a numerical indeces. If NA, all genes will be used.</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_raw_+3A_detection">detection</code></td>
<td>
<p>a logical matrix showing detection of genes (rows) in cells (columns)</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_raw_+3A_high.resolution">high.resolution</code></td>
<td>
<p>logical (default: TRUE). If set to FALSE, the density plot will be of a lower resolution</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_raw_+3A_point.size">point.size</code></td>
<td>
<p>numerical value to set size of points in plot. Default is 1.</p>
</td></tr>
<tr><td><code id="plot_gene_set_haystack_raw_+3A_order.by.signal">order.by.signal</code></td>
<td>
<p>If TRUE, cells with higher signal will be put on the foreground in the plot. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>

<hr>
<h2 id='plot_rand_fit'>plot_rand_fit</h2><span id='topic+plot_rand_fit'></span><span id='topic+plot_rand_fit.haystack'></span>

<h3>Description</h3>

<p>plot_rand_fit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_rand_fit(x, type = c("mean", "sd"))

## S3 method for class 'haystack'
plot_rand_fit(x, type = c("mean", "sd"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_rand_fit_+3A_x">x</code></td>
<td>
<p>haystack object.</p>
</td></tr>
<tr><td><code id="plot_rand_fit_+3A_type">type</code></td>
<td>
<p>whether to plot mean or sd.</p>
</td></tr>
</table>

<hr>
<h2 id='plot_rand_KLD'>plot_rand_KLD</h2><span id='topic+plot_rand_KLD'></span>

<h3>Description</h3>

<p>Plots the distribution of randomized KLD for each of the genes, together with
the mean and standard deviation, the 0.95 quantile and the 0.95 quantile from
a normal distribution with mean and standard deviations from the distribution
of KLDs. The logCV is indicated in the subtitle of each plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_rand_KLD(x, n = 12, log = TRUE, tail = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_rand_KLD_+3A_x">x</code></td>
<td>
<p>haystack result.</p>
</td></tr>
<tr><td><code id="plot_rand_KLD_+3A_n">n</code></td>
<td>
<p>number of genes from randomization set to plot.</p>
</td></tr>
<tr><td><code id="plot_rand_KLD_+3A_log">log</code></td>
<td>
<p>whether to use log of KLD.</p>
</td></tr>
<tr><td><code id="plot_rand_KLD_+3A_tail">tail</code></td>
<td>
<p>whether the genes are chosen from the tail of randomized genes.</p>
</td></tr>
</table>

<hr>
<h2 id='read_haystack'>Function to read haystack results from file.</h2><span id='topic+read_haystack'></span>

<h3>Description</h3>

<p>Function to read haystack results from file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_haystack(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_haystack_+3A_file">file</code></td>
<td>
<p>A file containing 'haystack' results to read</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;haystack&quot;
</p>

<hr>
<h2 id='show_result_haystack'>show_result_haystack</h2><span id='topic+show_result_haystack'></span><span id='topic+show_result_haystack.haystack'></span>

<h3>Description</h3>

<p>Shows the results of the 'haystack' analysis in various ways, sorted by significance. Priority of params is genes &gt; p.value.threshold &gt; n.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_result_haystack(
  res.haystack,
  n = NULL,
  p.value.threshold = NULL,
  gene = NULL
)

## S3 method for class 'haystack'
show_result_haystack(
  res.haystack,
  n = NULL,
  p.value.threshold = NULL,
  gene = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show_result_haystack_+3A_res.haystack">res.haystack</code></td>
<td>
<p>A 'haystack' result object.</p>
</td></tr>
<tr><td><code id="show_result_haystack_+3A_n">n</code></td>
<td>
<p>If defined, the top &quot;n&quot; significant genes will be returned. Default: NA, which shows all results.</p>
</td></tr>
<tr><td><code id="show_result_haystack_+3A_p.value.threshold">p.value.threshold</code></td>
<td>
<p>If defined, genes passing this p-value threshold will be returned.</p>
</td></tr>
<tr><td><code id="show_result_haystack_+3A_gene">gene</code></td>
<td>
<p>If defined, the results of this (these) gene(s) will be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output is a data.frame with the following columns:
* D_KL the calculated KL divergence.
* log.p.vals log10 p.values calculated from randomization.
* log.p.adj log10 p.values adjusted by Bonferroni correction.
</p>


<h3>Value</h3>

<p>A data.frame with 'haystack' results sorted by log.p.vals.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># using the toy example of the singleCellHaystack package

# running haystack
res &lt;- haystack(dat.tsne, dat.expression)

# below are variations for showing the results in a table
# 1. list top 10 biased genes
show_result_haystack(res.haystack = res, n =10)
# 2. list genes with p value below a certain threshold
show_result_haystack(res.haystack = res, p.value.threshold=1e-10)
# 3. list a set of specified genes
set &lt;- c("gene_497","gene_386", "gene_275")
show_result_haystack(res.haystack = res, gene = set)
</code></pre>

<hr>
<h2 id='singleCellHaystack-package'>singleCellHaystack: A Universal Differential Expression Prediction Tool for Single-Cell and Spatial Genomics Data</h2><span id='topic+singleCellHaystack'></span><span id='topic+singleCellHaystack-package'></span>

<h3>Description</h3>

<p>One key exploratory analysis step in single-cell genomics data analysis is the prediction of features with different activity levels. For example, we want to predict differentially expressed genes (DEGs) in single-cell RNA-seq data, spatial DEGs in spatial transcriptomics data, or differentially accessible regions (DARs) in single-cell ATAC-seq data. 'singleCellHaystack' predicts differentially active features in single cell omics datasets without relying on the clustering of cells into arbitrary clusters. 'singleCellHaystack' uses Kullback-Leibler divergence to find features (e.g., genes, genomic regions, etc) that are active in subsets of cells that are non-randomly positioned inside an input space (such as 1D trajectories, 2D tissue sections, multi-dimensional embeddings, etc). For the theoretical background of 'singleCellHaystack' we refer to our original paper Vandenbon and Diez (Nature Communications, 2020) <a href="https://doi.org/10.1038/s41467-020-17900-3">doi:10.1038/s41467-020-17900-3</a> and our update Vandenbon and Diez (Scientific Reports, 2023) <a href="https://doi.org/10.1038/s41598-023-38965-2">doi:10.1038/s41598-023-38965-2</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Alexis Vandenbon <a href="mailto:alexis.vandenbon@gmail.com">alexis.vandenbon@gmail.com</a> (<a href="https://orcid.org/0000-0003-2180-5732">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Diego Diez <a href="mailto:diego10ruiz@gmail.com">diego10ruiz@gmail.com</a> (<a href="https://orcid.org/0000-0002-2325-4893">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://alexisvdb.github.io/singleCellHaystack/">https://alexisvdb.github.io/singleCellHaystack/</a>
</p>
</li>
<li> <p><a href="https://github.com/alexisvdb/singleCellHaystack">https://github.com/alexisvdb/singleCellHaystack</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/alexisvdb/singleCellHaystack/issues">https://github.com/alexisvdb/singleCellHaystack/issues</a>
</p>
</li></ul>


<hr>
<h2 id='write_haystack'>Function to write haystack result data to file.</h2><span id='topic+write_haystack'></span>

<h3>Description</h3>

<p>Function to write haystack result data to file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_haystack(res.haystack, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_haystack_+3A_res.haystack">res.haystack</code></td>
<td>
<p>A 'haystack' result variable</p>
</td></tr>
<tr><td><code id="write_haystack_+3A_file">file</code></td>
<td>
<p>A file to write to</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
