<!DOCTYPE html><html><head><title>Help for package SIPmg</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SIPmg}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#atomX'><p>Atom fraction excess table</p></a></li>
<li><a href='#calc_atom_excess_MAGs'><p>Calculate atom fraction excess</p></a></li>
<li><a href='#calc_Mheavymax_MAGs'><p>Calculate Mheavymax</p></a></li>
<li><a href='#coverage_normalization'><p>Normalize feature coverages to estimate absolute abundance or relative coverage using MAG/contig coverage values with or without multiplying total DNA concentration of the fraction</p></a></li>
<li><a href='#DESeq2_l2fc'><p>Calculating log2 fold change for HTS-SIP data.</p></a></li>
<li><a href='#df_atomX_boot'><p>Bootstrapped atom fraction excess table</p></a></li>
<li><a href='#f_tibble'><p>Coverage table</p></a></li>
<li><a href='#filter_l2fc'><p>Filter l2fc table</p></a></li>
<li><a href='#filter_na'><p>Remove MAGs with NAs from atomX table</p></a></li>
<li><a href='#fractions'><p>Fractions table</p></a></li>
<li><a href='#GC_content'><p>GC_content table</p></a></li>
<li><a href='#HRSIP'><p>(MW-)HR-SIP analysis</p></a></li>
<li><a href='#incorporators_taxonomy'><p>Isotope incorporator list with GTDB taxonomy</p></a></li>
<li><a href='#mag.table'><p>MAG abundance table in phyloseq format</p></a></li>
<li><a href='#phylo.qSIP'><p>Master phyloseq object</p></a></li>
<li><a href='#phylo.table'><p>Master phyloseq object using the MAG phyloseq objects</p></a></li>
<li><a href='#qSIP_atom_excess_format_MAGs'><p>Reformat a phyloseq object of qSIP_atom_excess_MAGs analysis</p></a></li>
<li><a href='#qSIP_atom_excess_MAGs'><p>Calculate atom fraction excess using q-SIP method</p></a></li>
<li><a href='#qSIP_bootstrap_fcr'><p>Calculate adjusted bootstrap CI after for multiple testing for atom fraction excess using q-SIP method. Multiple hypothesis tests are corrected by</p></a></li>
<li><a href='#sample.table'><p>phyloseq-styled sample table</p></a></li>
<li><a href='#samples.object'><p>Fractions table in phyloseq format</p></a></li>
<li><a href='#scale_features_lm'><p>Scale feature coverage values to estimate their absolute abundance</p></a></li>
<li><a href='#scale_features_rlm'><p>Scale feature coverage values to estimate their absolute abundance</p></a></li>
<li><a href='#seq_dil'><p>Sequins dilution table</p></a></li>
<li><a href='#sequins'><p>Sequins table</p></a></li>
<li><a href='#tax.table'><p>phyloseq taxa table from GTDB taxonomy input</p></a></li>
<li><a href='#taxonomy_tibble'><p>Taxonomy table</p></a></li>
<li><a href='#taxonomy.object'><p>Taxonomy table in phyloseq format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Statistical Analysis to Identify Isotope Incorporating MAGs</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical analysis as part of a stable isotope probing (SIP) metagenomics study to identify isotope incorporating taxa recovered as metagenome-assembled genomes (MAGs). 
    Helpful reading and a vignette in bookdown format is provided on the package site <a href="https://zielslab.github.io/SIPmg.github.io/">https://zielslab.github.io/SIPmg.github.io/</a>.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://zielslab.github.io/SIPmg.github.io/">https://zielslab.github.io/SIPmg.github.io/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ZielsLab/SIPmg">https://github.com/ZielsLab/SIPmg</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>HTSSIP, dplyr, lazyeval, phyloseq, plyr, stringr, tibble,
tidyr, magrittr, ggplot2, ggpubr, purrr, rlang, MASS, DESeq2,
data.table, utils</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-22 00:15:16 UTC; devalshah</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, EBImage, readr, BiocManager</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Author:</td>
<td>Pranav Sampara [aut, cre],
  Kate Waring [ctb],
  Ryan Ziels [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Pranav Sampara &lt;pranav.sai.4@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-22 00:40:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='atomX'>Atom fraction excess table</h2><span id='topic+atomX'></span>

<h3>Description</h3>

<p>Data table generated from the &quot;qSIP_atom_excess_MAGs&quot; function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(atomX)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;list&quot;
</p>

<hr>
<h2 id='calc_atom_excess_MAGs'>Calculate atom fraction excess</h2><span id='topic+calc_atom_excess_MAGs'></span>

<h3>Description</h3>

<p>See Hungate et al., 2015 for more details
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_atom_excess_MAGs(Mlab, Mlight, Mheavymax, isotope = "13C")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_atom_excess_MAGs_+3A_mlab">Mlab</code></td>
<td>
<p>The molecular wight of labeled DNA</p>
</td></tr>
<tr><td><code id="calc_atom_excess_MAGs_+3A_mlight">Mlight</code></td>
<td>
<p>The molecular wight of unlabeled DNA</p>
</td></tr>
<tr><td><code id="calc_atom_excess_MAGs_+3A_mheavymax">Mheavymax</code></td>
<td>
<p>The theoretical maximum molecular weight of fully-labeled DNA</p>
</td></tr>
<tr><td><code id="calc_atom_excess_MAGs_+3A_isotope">isotope</code></td>
<td>
<p>The isotope for which the DNA is labeled with ('13C' or '18O')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric value: atom fraction excess (A)
</p>

<hr>
<h2 id='calc_Mheavymax_MAGs'>Calculate Mheavymax</h2><span id='topic+calc_Mheavymax_MAGs'></span>

<h3>Description</h3>

<p>This script was adapted from https://github.com/buckleylab/HTSSIP/blob/master/R/qSIP_atom_excess.R
for use with genome-centric metagenomics. See Hungate et al., 2015 for more details
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_Mheavymax_MAGs(Mlight, isotope = "13C", Gi = Gi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_Mheavymax_MAGs_+3A_mlight">Mlight</code></td>
<td>
<p>The molecular wight of unlabeled DNA</p>
</td></tr>
<tr><td><code id="calc_Mheavymax_MAGs_+3A_isotope">isotope</code></td>
<td>
<p>The isotope for which the DNA is labeled with ('13C' or '18O')</p>
</td></tr>
<tr><td><code id="calc_Mheavymax_MAGs_+3A_gi">Gi</code></td>
<td>
<p>The G+C content of unlabeled DNA</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric value: maximum molecular weight of fully-labeled DNA
</p>

<hr>
<h2 id='coverage_normalization'>Normalize feature coverages to estimate absolute abundance or relative coverage using MAG/contig coverage values with or without multiplying total DNA concentration of the fraction</h2><span id='topic+coverage_normalization'></span>

<h3>Description</h3>

<p>Normalize feature coverages to estimate absolute abundance or relative coverage using MAG/contig coverage values with or without multiplying total DNA concentration of the fraction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coverage_normalization(
  f_tibble,
  contig_coverage,
  sequencing_yield,
  fractions_df,
  approach = "relative_coverage"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coverage_normalization_+3A_f_tibble">f_tibble</code></td>
<td>
<p>Can be either of
(1) a tibble with first column &quot;Feature&quot; that contains bin IDs, and the rest of the columns represent samples with bins' coverage values.
(2) a tibble as outputted by the program &quot;checkm coverage&quot; from the tool CheckM. Please check CheckM documentation - https://github.com/Ecogenomics/CheckM on the usage for &quot;checkm coverage&quot; program</p>
</td></tr>
<tr><td><code id="coverage_normalization_+3A_contig_coverage">contig_coverage</code></td>
<td>
<p>tibble with contig ID names (&quot;Feature&quot; column), sample columns with <strong>same sample names as in f_tibble</strong> containing coverage values of each contig, contig length in bp (&quot;contig_length&quot; column), and the MAG the contig is associated (&quot;MAG&quot; column) <strong>with same MAGs as in Feature column of f_tibble dataset</strong>.</p>
</td></tr>
<tr><td><code id="coverage_normalization_+3A_sequencing_yield">sequencing_yield</code></td>
<td>
<p>tibble containing sample ID (&quot;sample&quot; column) with <strong>same sample names as in f_tibble</strong> and number of reads in bp recovered in that sample (&quot;yield&quot; column).</p>
</td></tr>
<tr><td><code id="coverage_normalization_+3A_fractions_df">fractions_df</code></td>
<td>
<p>fractions data frame
A fractions file with the following columns
</p>

<ul>
<li><p> Replicate: Depends on how many replicates the study has
</p>
</li>
<li><p> Fractions: Typically in the range of 2-24
</p>
</li>
<li><p> Buoyant_density: As calculated from the refractometer for each fraction and replicate
</p>
</li>
<li><p> Isotope: &quot;12C&quot;, &quot;13C&quot;, &quot;14N&quot;, &quot;15N&quot; etc.
</p>
</li>
<li><p> DNA_concentration
</p>
</li>
<li><p> Sample: In the format &quot;'isotope'<em>rep</em>#<em>fraction</em>#&quot;.
For instance, &quot;12C_rep_1_fraction_1&quot;
</p>
</li></ul>
</td></tr>
<tr><td><code id="coverage_normalization_+3A_approach">approach</code></td>
<td>
<p>Please choose the method for coverage normalization as &quot;relative_coverage&quot;, &quot;greenlon&quot;, &quot;starr&quot; to estimate only relative coverage without multiplying DNA concentration of fraction, or as per methods in Greenlon et al. - https://journals.asm.org/doi/full/10.1128/msystems.00417-22 or Starr et al. - https://journals.asm.org/doi/10.1128/mSphere.00085-21</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tibble containing normalized coverage in required format with MAG name as first column and the normalized coverage values in each sample as the rest of the columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(f_tibble)

rel.cov = coverage_normalization(f_tibble)


</code></pre>

<hr>
<h2 id='DESeq2_l2fc'>Calculating log2 fold change for HTS-SIP data.</h2><span id='topic+DESeq2_l2fc'></span>

<h3>Description</h3>

<p>The 'use_geo_mean' parameter uses geometric means on all non-zero abundances
for estimateSizeFactors instead of using the default log-tranformed geometric means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DESeq2_l2fc(
  physeq,
  density_min,
  density_max,
  design,
  l2fc_threshold = 0.25,
  sparsity_threshold = 0.25,
  sparsity_apply = "all",
  size_factors = "geoMean"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DESeq2_l2fc_+3A_physeq">physeq</code></td>
<td>
<p>Phyloseq object</p>
</td></tr>
<tr><td><code id="DESeq2_l2fc_+3A_density_min">density_min</code></td>
<td>
<p>Minimum buoyant density of the 'heavy' gradient fractions</p>
</td></tr>
<tr><td><code id="DESeq2_l2fc_+3A_density_max">density_max</code></td>
<td>
<p>Maximum buoyant density of the 'heavy' gradient fractions</p>
</td></tr>
<tr><td><code id="DESeq2_l2fc_+3A_design">design</code></td>
<td>
<p><code>design</code> parameter used for DESeq2 analysis.
See <code>DESeq2::DESeq</code> for more details.</p>
</td></tr>
<tr><td><code id="DESeq2_l2fc_+3A_l2fc_threshold">l2fc_threshold</code></td>
<td>
<p>log2 fold change (l2fc) values must be significantly above this
threshold in order to reject the hypothesis of equal counts.</p>
</td></tr>
<tr><td><code id="DESeq2_l2fc_+3A_sparsity_threshold">sparsity_threshold</code></td>
<td>
<p>All OTUs observed in less than this portion (fraction: 0-1)
of gradient fraction samples are pruned. A a form of indepedent filtering,
The sparsity cutoff with the most rejected hypotheses is used.</p>
</td></tr>
<tr><td><code id="DESeq2_l2fc_+3A_sparsity_apply">sparsity_apply</code></td>
<td>
<p>Apply sparsity threshold to all gradient fraction samples ('all')
or just heavy fraction samples ('heavy')</p>
</td></tr>
<tr><td><code id="DESeq2_l2fc_+3A_size_factors">size_factors</code></td>
<td>
<p>Method of estimating size factors.
'geoMean' is from (Pepe-Ranney et. al., 2016) and removes all zero-abundances from the calculation.
'default' is the default for estimateSizeFactors.
'iterate' is an alternative when every OTU has a zero in &gt;=1 sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe of HRSIP results
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylo.qSIP)


df_l2fc = DESeq2_l2fc(phylo.qSIP, density_min=1.71, density_max=1.75, design=~Isotope)


</code></pre>

<hr>
<h2 id='df_atomX_boot'>Bootstrapped atom fraction excess table</h2><span id='topic+df_atomX_boot'></span>

<h3>Description</h3>

<p>Data table generated from bostrapping the AFE table using the &quot;qSIP_bootstrap_fcr&quot; function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(df_atomX_boot)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;data.frame&quot;
</p>

<hr>
<h2 id='f_tibble'>Coverage table</h2><span id='topic+f_tibble'></span>

<h3>Description</h3>

<p>Coverage data used for many functions in the package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(f_tibble)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;data.frame&quot;
</p>

<hr>
<h2 id='filter_l2fc'>Filter l2fc table</h2><span id='topic+filter_l2fc'></span>

<h3>Description</h3>

<p><code>filter_l2fc</code> filters a l2fc table to 'best' sparsity cutoffs &amp;
bouyant density windows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_l2fc(df_l2fc, padj_cutoff = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_l2fc_+3A_df_l2fc">df_l2fc</code></td>
<td>
<p>data.frame of log2 fold change values</p>
</td></tr>
<tr><td><code id="filter_l2fc_+3A_padj_cutoff">padj_cutoff</code></td>
<td>
<p>Adjusted p-value cutoff for rejecting the null hypothesis
that l2fc values were not greater than the l2fc_threshold.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>filtered df_l2fc object
</p>

<hr>
<h2 id='filter_na'>Remove MAGs with NAs from atomX table</h2><span id='topic+filter_na'></span>

<h3>Description</h3>

<p>This function enables removing NAs from the atomX table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_na(atomX)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_na_+3A_atomx">atomX</code></td>
<td>
<p>A list object created by <code>qSIP_atom_excess_MAGs()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 2 data.frame objects without MAGs which have NAs. 'W' contains the weighted mean buoyant density (W) values for each OTU in each treatment/control. 'A' contains the atom fraction excess values for each OTU. For the 'A' table, the 'Z' column is buoyant density shift, and the 'A' column is atom fraction excess.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(atomX)



### Remove NAs in atomX table
atomx_no_na = filter_na(atomX)


</code></pre>

<hr>
<h2 id='fractions'>Fractions table</h2><span id='topic+fractions'></span>

<h3>Description</h3>

<p>Fractions data used for many functions in the package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fractions)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;data.frame&quot;
</p>

<hr>
<h2 id='GC_content'>GC_content table</h2><span id='topic+GC_content'></span>

<h3>Description</h3>

<p>GC_content data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GC_content)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;data.frame&quot;
</p>

<hr>
<h2 id='HRSIP'>(MW-)HR-SIP analysis</h2><span id='topic+HRSIP'></span>

<h3>Description</h3>

<p>Conduct (multi-window) high resolution stable isotope probing (HR-SIP) analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HRSIP(
  physeq,
  design,
  density_windows = data.frame(density_min = c(1.7), density_max = c(1.75)),
  sparsity_threshold = seq(0, 0.3, 0.1),
  sparsity_apply = "all",
  l2fc_threshold = 0.25,
  padj_method = "BH",
  padj_cutoff = NULL,
  parallel = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HRSIP_+3A_physeq">physeq</code></td>
<td>
<p>Phyloseq object</p>
</td></tr>
<tr><td><code id="HRSIP_+3A_design">design</code></td>
<td>
<p><code>design</code> parameter used for DESeq2 analysis.
This is usually used to differentiate labeled-treatment and unlabeld-control samples.
See <code>DESeq2::DESeq</code> for more details on the option.</p>
</td></tr>
<tr><td><code id="HRSIP_+3A_density_windows">density_windows</code></td>
<td>
<p>The buoyant density window(s) used for for calculating log2
fold change values. Input can be a vector (length 2) or a data.frame with a 'density_min'
and a 'density_max' column (each row designates a density window).</p>
</td></tr>
<tr><td><code id="HRSIP_+3A_sparsity_threshold">sparsity_threshold</code></td>
<td>
<p>All OTUs observed in less than this portion (fraction: 0-1)
of gradient fraction samples are pruned. This is a form of indepedent filtering.
The sparsity cutoff with the most rejected hypotheses is used.</p>
</td></tr>
<tr><td><code id="HRSIP_+3A_sparsity_apply">sparsity_apply</code></td>
<td>
<p>Apply sparsity threshold to all gradient fraction samples ('all')
or just 'heavy' fraction samples ('heavy'), where 'heavy' samples are designated
by the <code>density_windows</code>.</p>
</td></tr>
<tr><td><code id="HRSIP_+3A_l2fc_threshold">l2fc_threshold</code></td>
<td>
<p>log2 fold change (l2fc) values must be significantly above this
threshold in order to reject the hypothesis of equal counts.
See <code>DESeq2</code> for more information.</p>
</td></tr>
<tr><td><code id="HRSIP_+3A_padj_method">padj_method</code></td>
<td>
<p>Method for global p-value adjustment (See <code>p.adjust()</code>).</p>
</td></tr>
<tr><td><code id="HRSIP_+3A_padj_cutoff">padj_cutoff</code></td>
<td>
<p>Adjusted p-value cutoff for rejecting the null hypothesis
that l2fc values were not greater than the l2fc_threshold.
Set to <code>NULL</code> to skip filtering of results to the sparsity cutoff with most
rejected hypotheses and filtering each OTU to the buoyant density window with the
greatest log2 fold change.</p>
</td></tr>
<tr><td><code id="HRSIP_+3A_parallel">parallel</code></td>
<td>
<p>Process each parameter combination in parallel.
See <code>plyr::mdply()</code> for more information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The (MW-)HR-SIP workflow is as follows:
</p>

<ol>
<li><p> For each sparsity threshold &amp; BD window: calculate log2 fold change values (with DESeq2) for each OTU
</p>
</li>
<li><p> Globally adjust p-values with a user-defined method (see p.adjust())
</p>
</li>
<li><p> Select the sparsity cutoff with the most rejected hypotheses
</p>
</li>
<li><p> For each OTU, select the BD window with the greatest log2 fold change value
</p>
</li></ol>



<h3>Value</h3>

<p>dataframe of HRSIP results
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylo.qSIP)


## HR-SIP
### Note: treatment-control samples differentiated with 'design=~Isotope'
df_l2fc = HRSIP(phylo.qSIP, design=~Isotope)
## Same, but multiple BD windows (MW-HR-SIP). For parallel processing change to parallel = TRUE
### Windows = 1.7-1.74, 1.72-1.75, and 1.73 - 1.76
windows = data.frame(density_min=c(1.71,1.72, 1.73), density_max=c(1.74,1.75,1.76))
df_l2fc = HRSIP(phylo.qSIP,
                design=~Isotope,
                density_windows = windows,
                padj_cutoff = 0.05,
                parallel=FALSE)



</code></pre>

<hr>
<h2 id='incorporators_taxonomy'>Isotope incorporator list with GTDB taxonomy</h2><span id='topic+incorporators_taxonomy'></span>

<h3>Description</h3>

<p>This function provides a table with MAGs and their corresponding GTDB taxonomy
as an output. This would be useful in identifying the taxa that have incorporation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>incorporators_taxonomy(taxonomy, bootstrapped_AFE_table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="incorporators_taxonomy_+3A_taxonomy">taxonomy</code></td>
<td>
<p>A taxonomy tibble obtained in the markdown. This taxonomy tibble is
typically a concatenated list of archaeal and bacterial taxonomy from GTDB-Tk
Please check GTDB-Tk documentation for running the tool</p>
</td></tr>
<tr><td><code id="incorporators_taxonomy_+3A_bootstrapped_afe_table">bootstrapped_AFE_table</code></td>
<td>
<p>A data frame indicating bootstrapped atom fraction excess values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with two columns, OTU and Taxonomy, with taxonomy of the incorporator MAGs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(taxonomy_tibble,df_atomX_boot)



### Making incorporator taxonomy list
incorporator_list = incorporators_taxonomy(taxonomy = taxonomy_tibble,
         bootstrapped_AFE_table = df_atomX_boot)


</code></pre>

<hr>
<h2 id='mag.table'>MAG abundance table in phyloseq format</h2><span id='topic+mag.table'></span>

<h3>Description</h3>

<p>MAG abundances in the format of phyloseq object to be used in the qSIP and (MW-)HR-SIP pipeline
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mag.table)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;phyloseq&quot;
</p>

<hr>
<h2 id='phylo.qSIP'>Master phyloseq object</h2><span id='topic+phylo.qSIP'></span>

<h3>Description</h3>

<p>Master phyloseq object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phylo.qSIP)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;phyloseq&quot;
</p>

<hr>
<h2 id='phylo.table'>Master phyloseq object using the MAG phyloseq objects</h2><span id='topic+phylo.table'></span>

<h3>Description</h3>

<p>Creates a phyloseq-style object using processed phyloseq objects for otu
table (here, MAG table), taxa table, and sample table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo.table(mag, taxa, samples)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylo.table_+3A_mag">mag</code></td>
<td>
<p>phyloseq-styled MAG table</p>
</td></tr>
<tr><td><code id="phylo.table_+3A_taxa">taxa</code></td>
<td>
<p>phyloseq-styled taxa table</p>
</td></tr>
<tr><td><code id="phylo.table_+3A_samples">samples</code></td>
<td>
<p>sample information table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phyloseq object for MAGs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(mag.table,taxonomy.object,samples.object,fractions,taxonomy_tibble)
###Making phyloseq table from fractions metadata
samples.object = sample.table(fractions)
taxonomy.object = tax.table(taxonomy_tibble)




### Making master phyloseq table from scaled MAG data, taxa and fractions phyloseq data
phylo.qSIP = phylo.table(mag.table,taxonomy.object,samples.object)


</code></pre>

<hr>
<h2 id='qSIP_atom_excess_format_MAGs'>Reformat a phyloseq object of qSIP_atom_excess_MAGs analysis</h2><span id='topic+qSIP_atom_excess_format_MAGs'></span>

<h3>Description</h3>

<p>Reformat a phyloseq object of qSIP_atom_excess_MAGs analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qSIP_atom_excess_format_MAGs(physeq, control_expr, treatment_rep)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qSIP_atom_excess_format_MAGs_+3A_physeq">physeq</code></td>
<td>
<p>A phyloseq object</p>
</td></tr>
<tr><td><code id="qSIP_atom_excess_format_MAGs_+3A_control_expr">control_expr</code></td>
<td>
<p>An expression for identifying unlabeled control
samples in the phyloseq object (eg., &quot;Substrate=='12C-Con'&quot;)</p>
</td></tr>
<tr><td><code id="qSIP_atom_excess_format_MAGs_+3A_treatment_rep">treatment_rep</code></td>
<td>
<p>Which column in the phyloseq sample data designates
replicate treatments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric value: atom fraction excess (A)
</p>

<hr>
<h2 id='qSIP_atom_excess_MAGs'>Calculate atom fraction excess using q-SIP method</h2><span id='topic+qSIP_atom_excess_MAGs'></span>

<h3>Description</h3>

<p>Calculate atom fraction excess using q-SIP method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qSIP_atom_excess_MAGs(
  physeq,
  control_expr,
  treatment_rep = NULL,
  isotope = "13C",
  df_OTU_W = NULL,
  Gi
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qSIP_atom_excess_MAGs_+3A_physeq">physeq</code></td>
<td>
<p>A phyloseq object</p>
</td></tr>
<tr><td><code id="qSIP_atom_excess_MAGs_+3A_control_expr">control_expr</code></td>
<td>
<p>Expression used to identify control samples based on sample_data.</p>
</td></tr>
<tr><td><code id="qSIP_atom_excess_MAGs_+3A_treatment_rep">treatment_rep</code></td>
<td>
<p>Which column in the phyloseq sample data designates
replicate treatments</p>
</td></tr>
<tr><td><code id="qSIP_atom_excess_MAGs_+3A_isotope">isotope</code></td>
<td>
<p>The isotope for which the DNA is labeled with ('13C' or '18O')</p>
</td></tr>
<tr><td><code id="qSIP_atom_excess_MAGs_+3A_df_otu_w">df_OTU_W</code></td>
<td>
<p>Keep NULL</p>
</td></tr>
<tr><td><code id="qSIP_atom_excess_MAGs_+3A_gi">Gi</code></td>
<td>
<p>GC content of the MAG</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 2 data.frame objects. 'W' contains the weighted mean buoyant density (W) values for each OTU in each treatment/control. 'A' contains the atom fraction excess values for each OTU. For the 'A' table, the 'Z' column is buoyant density shift, and the 'A' column is atom fraction excess.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylo.qSIP,GC_content)
### Making atomx table

## Not run::
### BD shift (Z) &amp; atom excess (A)
atomX = qSIP_atom_excess_MAGs(phylo.qSIP,
                         control_expr='Isotope=="12C"',
                         treatment_rep='Replicate',
                         Gi = GC_content)

</code></pre>

<hr>
<h2 id='qSIP_bootstrap_fcr'>Calculate adjusted bootstrap CI after for multiple testing for atom fraction excess using q-SIP method. Multiple hypothesis tests are corrected by</h2><span id='topic+qSIP_bootstrap_fcr'></span>

<h3>Description</h3>

<p>Calculate adjusted bootstrap CI after for multiple testing for atom fraction excess using q-SIP method. Multiple hypothesis tests are corrected by
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qSIP_bootstrap_fcr(
  atomX,
  isotope = "13C",
  n_sample = c(3, 3),
  ci_adjust_method = "fcr",
  n_boot = 10,
  parallel = FALSE,
  a = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qSIP_bootstrap_fcr_+3A_atomx">atomX</code></td>
<td>
<p>A list object created by <code>qSIP_atom_excess_MAGs()</code></p>
</td></tr>
<tr><td><code id="qSIP_bootstrap_fcr_+3A_isotope">isotope</code></td>
<td>
<p>The isotope for which the DNA is labeled with ('13C' or '18O')</p>
</td></tr>
<tr><td><code id="qSIP_bootstrap_fcr_+3A_n_sample">n_sample</code></td>
<td>
<p>A vector of length 2. The sample size for data resampling (with replacement) for 1) control samples and 2) treatment samples.</p>
</td></tr>
<tr><td><code id="qSIP_bootstrap_fcr_+3A_ci_adjust_method">ci_adjust_method</code></td>
<td>
<p>Confidence interval adjustment method. Please choose 'FCR', 'Bonferroni', or 'none' (if no adjustment is needed). Default is FCR and also provides unadjusted CI.</p>
</td></tr>
<tr><td><code id="qSIP_bootstrap_fcr_+3A_n_boot">n_boot</code></td>
<td>
<p>Number of bootstrap replicates.</p>
</td></tr>
<tr><td><code id="qSIP_bootstrap_fcr_+3A_parallel">parallel</code></td>
<td>
<p>Parallel processing. See <code>.parallel</code> option in <code>dplyr::mdply()</code> for more details.</p>
</td></tr>
<tr><td><code id="qSIP_bootstrap_fcr_+3A_a">a</code></td>
<td>
<p>A numeric value. The alpha for calculating confidence intervals.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame of atom fraction excess values (A) and atom fraction excess confidence intervals adjusted for multiple testing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phylo.qSIP,GC_content)

### BD shift (Z) &amp; atom excess (A)
atomX = qSIP_atom_excess_MAGs(phylo.qSIP,
                        control_expr='Isotope=="12C"',
                        treatment_rep='Replicate', Gi = GC_content)

### Add doParallel::registerDoParallel(num_cores) if parallel bootstrapping is to be done
df_atomX_boot = qSIP_bootstrap_fcr(atomX, n_boot=5, parallel = FALSE)


</code></pre>

<hr>
<h2 id='sample.table'>phyloseq-styled sample table</h2><span id='topic+sample.table'></span>

<h3>Description</h3>

<p>Creates a phyloseq-styled sample table from fractions metadata
containing data on fraction number, number of replicates, buoyant density
calculated from a refractometer, type of isotope, and DNA concentration
of each fraction, and isotope type. See below for information on &quot;fractions&quot; file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample.table(fractions_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample.table_+3A_fractions_df">fractions_df</code></td>
<td>
<p>fractions data frame
A fractions file with the following columns
</p>

<ul>
<li><p> Replicate: Depends on how many replicates the study has
</p>
</li>
<li><p> Fractions: Typically in the range of 2-24
</p>
</li>
<li><p> Buoyant_density: As calculated from the refractometer for each fraction and replicate
</p>
</li>
<li><p> Isotope: &quot;12C&quot;, &quot;13C&quot;, &quot;14N&quot;, &quot;15N&quot; etc.
</p>
</li>
<li><p> DNA_concentration
</p>
</li>
<li><p> Sample: In the format &quot;'isotope'<em>rep</em>#<em>fraction</em>#&quot;.
For instance, &quot;12C_rep_1_fraction_1&quot;
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>data  frame: phyloseq-style sample table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fractions)



### Making phyloseq table from fractions metadata
samples.object = sample.table(fractions)

</code></pre>

<hr>
<h2 id='samples.object'>Fractions table in phyloseq format</h2><span id='topic+samples.object'></span>

<h3>Description</h3>

<p>Fractions metadata in the format of phyloseq object to be used in the qSIP and (MW-)HR-SIP pipeline
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(samples.object)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;phyloseq&quot;
</p>

<hr>
<h2 id='scale_features_lm'>Scale feature coverage values to estimate their absolute abundance</h2><span id='topic+scale_features_lm'></span>

<h3>Description</h3>

<p>Calculates global scaling factors for features (contigs or bins),based on linear regression of sequin coverage. Options include log-transformations of coverage, as well as filtering features based on limit of detection. This function must be called first, before the feature abundance table, feature detection table, and plots are retrieved.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_features_lm(
  f_tibble,
  sequin_meta,
  seq_dilution,
  log_trans = TRUE,
  coe_of_variation = 250,
  lod_limit = 0,
  save_plots = TRUE,
  plot_dir = tempdir(),
  cook_filtering = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scale_features_lm_+3A_f_tibble">f_tibble</code></td>
<td>
<p>Can be either of
(1) a tibble with first column &quot;Feature&quot; that contains bin IDs, and the rest of the columns represent samples with bins' coverage values.
(2) a tibble as outputted by the program &quot;checkm coverage&quot; from the tool CheckM. Please check CheckM documentation - https://github.com/Ecogenomics/CheckM on the usage for &quot;checkm coverage&quot; program</p>
</td></tr>
<tr><td><code id="scale_features_lm_+3A_sequin_meta">sequin_meta</code></td>
<td>
<p>tibble containing sequin names (&quot;Feature column&quot;) and concentrations in attamoles/uL (&quot;Concentration&quot;) column.</p>
</td></tr>
<tr><td><code id="scale_features_lm_+3A_seq_dilution">seq_dilution</code></td>
<td>
<p>tibble with first column &quot;Sample&quot; with <strong>same sample names as in f_tibble</strong>, and a second column &quot;Dilution&quot; showing ratio of sequins added to final sample volume (e.g. a value of 0.01 for a dilution of 1 volume sequin to 99 volumes sample)</p>
</td></tr>
<tr><td><code id="scale_features_lm_+3A_log_trans">log_trans</code></td>
<td>
<p>Boolean (TRUE or FALSE), should coverages and sequin concentrations be log-scaled?</p>
</td></tr>
<tr><td><code id="scale_features_lm_+3A_coe_of_variation">coe_of_variation</code></td>
<td>
<p>Acceptable coefficient of variation for coverage and detection (eg. 20 - for 20 % threshold of coefficient of variation). Coverages above the threshold value will be flagged in the plots.</p>
</td></tr>
<tr><td><code id="scale_features_lm_+3A_lod_limit">lod_limit</code></td>
<td>
<p>(Decimal range 0-1) Threshold for the percentage of minimum detected sequins per concentration group. Default = 0</p>
</td></tr>
<tr><td><code id="scale_features_lm_+3A_save_plots">save_plots</code></td>
<td>
<p>Boolean (TRUE or FALSE), should sequin scaling be saved? Default = TRUE</p>
</td></tr>
<tr><td><code id="scale_features_lm_+3A_plot_dir">plot_dir</code></td>
<td>
<p>Directory where plots are to be saved. Will create a directory &quot;sequin_scaling_plots_lm&quot; if it does not exist.</p>
</td></tr>
<tr><td><code id="scale_features_lm_+3A_cook_filtering">cook_filtering</code></td>
<td>
<p>Boolean (TRUE or FALSE), should data points be filtered based on Cook's distance metric. Cooks distance can be useful in detecting influential outliers in an ordinary least squareâ€™s regression model, which can negatively influence the model. A threshold of Cooks distance of 4/n (where n is the sample size) is chosen, and any data point with Cooks distance &gt; 4/n is filtered out. It is typical to choose 4/n as the threshold in detecting the outliers in the data. Default = TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of tibbles containing
</p>

<ul>
<li><p> mag_tab: a tibble with first column &quot;Feature&quot; that contains bin (or contig IDs), and the rest of the columns represent samples with features' scaled abundances (attamoles/uL)
</p>
</li>
<li><p> mag_det: a tibble with first column &quot;Feature&quot; that contains bin (or contig IDs),
</p>
</li>
<li><p> plots: linear regression plots for scaling MAG coverage values to absolute abundance
</p>
</li>
<li><p> scale_fac: a master tibble with all of the intermediate values in above calculations
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(f_tibble, sequins, seq_dil)



### scaling sequins from coverage values
scaled_features_lm = scale_features_lm(f_tibble,sequin_meta, seq_dil)


</code></pre>

<hr>
<h2 id='scale_features_rlm'>Scale feature coverage values to estimate their absolute abundance</h2><span id='topic+scale_features_rlm'></span>

<h3>Description</h3>

<p>Calculates global scaling factors for features (contigs or bins),based on linear regression of sequin coverage. Options include log-transformations of coverage, as well as filtering features based on limit of detection. This function must be called first, before the feature abundance table, feature detection table, and plots are retrieved.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_features_rlm(
  f_tibble,
  sequin_meta,
  seq_dilution,
  log_trans = TRUE,
  coe_of_variation = 250,
  lod_limit = 0,
  save_plots = TRUE,
  plot_dir = tempdir()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scale_features_rlm_+3A_f_tibble">f_tibble</code></td>
<td>
<p>Can be either of
(1) a tibble with first column &quot;Feature&quot; that contains bin IDs, and the rest of the columns represent samples with bins' coverage values.
(2) a tibble as outputted by the program &quot;checkm coverage&quot; from the tool CheckM. Please check CheckM documentation - https://github.com/Ecogenomics/CheckM on the usage for &quot;checkm coverage&quot; program</p>
</td></tr>
<tr><td><code id="scale_features_rlm_+3A_sequin_meta">sequin_meta</code></td>
<td>
<p>tibble containing sequin names (&quot;Feature column&quot;) and concentrations in attamoles/uL (&quot;Concentration&quot;) column.</p>
</td></tr>
<tr><td><code id="scale_features_rlm_+3A_seq_dilution">seq_dilution</code></td>
<td>
<p>tibble with first column &quot;Sample&quot; with <strong>same sample names as in f_tibble</strong>, and a second column &quot;Dilution&quot; showing ratio of sequins added to final sample volume (e.g. a value of 0.01 for a dilution of 1 volume sequin to 99 volumes sample)</p>
</td></tr>
<tr><td><code id="scale_features_rlm_+3A_log_trans">log_trans</code></td>
<td>
<p>Boolean (TRUE or FALSE), should coverages and sequin concentrations be log-scaled? Default = TRUE</p>
</td></tr>
<tr><td><code id="scale_features_rlm_+3A_coe_of_variation">coe_of_variation</code></td>
<td>
<p>Acceptable coefficient of variation for coverage and detection (eg. 20 - for 20 % threshold of coefficient of variation). Coverages above the threshold value will be flagged in the plots. Default = 250</p>
</td></tr>
<tr><td><code id="scale_features_rlm_+3A_lod_limit">lod_limit</code></td>
<td>
<p>(Decimal range 0-1) Threshold for the percentage of minimum detected sequins per concentration group. Default = 0</p>
</td></tr>
<tr><td><code id="scale_features_rlm_+3A_save_plots">save_plots</code></td>
<td>
<p>Boolean (TRUE or FALSE), should sequin scaling be saved? Default = TRUE</p>
</td></tr>
<tr><td><code id="scale_features_rlm_+3A_plot_dir">plot_dir</code></td>
<td>
<p>Directory where plots are to be saved. Will create a directory &quot;sequin_scaling_plots_rlm&quot; if it does not exist.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of tibbles containing
</p>

<ul>
<li><p> mag_tab: a tibble with first column &quot;Feature&quot; that contains bin (or contig IDs), and the rest of the columns represent samples with features' scaled abundances (attamoles/uL)
</p>
</li>
<li><p> mag_det: a tibble with first column &quot;Feature&quot; that contains bin (or contig IDs),
</p>
</li>
<li><p> plots: linear regression plots for scaling MAG coverage values to absolute abundance (optional)
</p>
</li>
<li><p> scale_fac: a master tibble with all of the intermediate values in above calculations
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(f_tibble, sequins, seq_dil)



### scaling sequins from coverage values
scaled_features_rlm = scale_features_rlm(f_tibble,sequins, seq_dil)


</code></pre>

<hr>
<h2 id='seq_dil'>Sequins dilution table</h2><span id='topic+seq_dil'></span>

<h3>Description</h3>

<p>Sequins dilution data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(seq_dil)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;data.frame&quot;
</p>

<hr>
<h2 id='sequins'>Sequins table</h2><span id='topic+sequins'></span>

<h3>Description</h3>

<p>Sequins metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sequins)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;data.frame&quot;
</p>

<hr>
<h2 id='tax.table'>phyloseq taxa table from GTDB taxonomy input</h2><span id='topic+tax.table'></span>

<h3>Description</h3>

<p>A MAG table, similar to OTU table in phyloseq, will be generated from a
concantenated GTDB taxa table for bacteria and archaea
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax.table(taxonomy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tax.table_+3A_taxonomy">taxonomy</code></td>
<td>
<p>GTDB taxonomy data frame.  A taxonomy file in the GTDB output format. Load the bacteria and archaea taxonomy outputs separately.
The markdown requires loading the standard output files from GTDB-Tk separately for bacteria and archaea</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phyloseq-style taxonomy table, but for MAGs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(taxonomy_tibble)



### Making phyloseq table from taxonomy metadata
taxonomy.object = tax.table(taxonomy_tibble)


</code></pre>

<hr>
<h2 id='taxonomy_tibble'>Taxonomy table</h2><span id='topic+taxonomy_tibble'></span>

<h3>Description</h3>

<p>Taxonomy table from GTDB-Tk output - combining both bacterial and archaeal taxonomy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(taxonomy_tibble)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;data.frame&quot;
</p>

<hr>
<h2 id='taxonomy.object'>Taxonomy table in phyloseq format</h2><span id='topic+taxonomy.object'></span>

<h3>Description</h3>

<p>Taxonomy table in the format of phyloseq object to be used in the qSIP and (MW-)HR-SIP pipeline
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(taxonomy.object)
</code></pre>


<h3>Format</h3>

<p>An object of class &quot;phyloseq&quot;
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
