<!DOCTYPE html><html><head><title>Help for package SIS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SIS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#leukemia.test'><p>Gene expression Leukemia testing data set from Golub et al. (1999)</p></a></li>
<li><a href='#leukemia.train'><p>Gene expression Leukemia training data set from Golub et al. (1999)</p></a></li>
<li><a href='#predict.SIS'><p>Model prediction based on a fitted SIS object.</p></a></li>
<li><a href='#prostate.test'><p>Gene expression Prostate Cancer testing data set from Singh et al. (2002)</p></a></li>
<li><a href='#prostate.train'><p>Gene expression Prostate Cancer training data set from Singh et al. (2002)</p></a></li>
<li><a href='#SIS'><p>(Iterative) Sure Independence Screening ((I)SIS) and Fitting in Generalized</p>
Linear Models and Cox's Proportional Hazards Models</a></li>
<li><a href='#standardize'><p>Standardization of High-Dimensional Design Matrices</p></a></li>
<li><a href='#tune.fit'><p>Using the <span class="pkg">glmnet</span> and <span class="pkg">ncvreg</span> packages, fits a Generalized Linear</p>
Model or Cox Proportional Hazards Model using various methods for choosing
the regularization parameter <code class="reqn">\lambda</code></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.8-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-01-27</td>
</tr>
<tr>
<td>Title:</td>
<td>Sure Independence Screening</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet, ncvreg, survival</td>
</tr>
<tr>
<td>Description:</td>
<td>Variable selection techniques are essential tools for model
    selection and estimation in high-dimensional statistical models. Through this
    publicly available package, we provide a unified environment to carry out
    variable selection using iterative sure independence screening (SIS) (Fan and Lv (2008)&lt;<a href="https://doi.org/10.1111%2Fj.1467-9868.2008.00674.x">doi:10.1111/j.1467-9868.2008.00674.x</a>&gt;) and all
    of its variants in generalized linear models (Fan and Song (2009)&lt;<a href="https://doi.org/10.1214%2F10-AOS798">doi:10.1214/10-AOS798</a>&gt;) and the Cox proportional hazards
    model (Fan, Feng and Wu (2010)&lt;<a href="https://doi.org/10.1214%2F10-IMSCOLL606">doi:10.1214/10-IMSCOLL606</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-01-27 16:03:49 UTC; yangfeng</td>
</tr>
<tr>
<td>Author:</td>
<td>Yang Feng [aut, cre],
  Jianqing Fan [aut],
  Diego Franco Saldana [aut],
  Yichao Wu [aut],
  Richard Samworth [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yang Feng &lt;yangfengstat@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-01-27 17:10:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='leukemia.test'>Gene expression Leukemia testing data set from Golub et al. (1999)</h2><span id='topic+leukemia.test'></span>

<h3>Description</h3>

<p>Gene expression testing data of 7129 genes from 34 patients 
with acute leukemias (20 in class Acute Lymphoblastic Leukemia and 14 in 
class Acute Myeloid Leukemia) from the microarray study of Golub et al. (1999).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(leukemia.test)</code></pre>


<h3>Format</h3>

<p>A data frame with 34 observations on 7129 variables.</p>


<h3>Source</h3>

<p>http://wwwprod.broadinstitute.org/cgi-bin/cancer/datasets.cgi</p>


<h3>References</h3>

<p>Golub et al. (1999) Molecular Classification of Cancer: Class Discovery and 
Class Prediction by Gene Expression Monitoring. <em>Science</em>, <b>286</b>, 
531-537.
</p>

<hr>
<h2 id='leukemia.train'>Gene expression Leukemia training data set from Golub et al. (1999)</h2><span id='topic+leukemia.train'></span>

<h3>Description</h3>

<p>Gene expression training data of 7129 genes from 38 patients 
with acute leukemias (27 in class Acute Lymphoblastic Leukemia and 11 in 
class Acute Myeloid Leukemia) from the microarray study of Golub et al. (1999).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(leukemia.train)</code></pre>


<h3>Format</h3>

<p>A data frame with 38 observations on 7129 variables.</p>


<h3>Source</h3>

<p>http://wwwprod.broadinstitute.org/cgi-bin/cancer/datasets.cgi</p>


<h3>References</h3>

<p>Golub et al. (1999) Molecular Classification of Cancer: Class Discovery and 
Class Prediction by Gene Expression Monitoring. <em>Science</em>, <b>286</b>, 
531-537.
</p>

<hr>
<h2 id='predict.SIS'>Model prediction based on a fitted SIS object.</h2><span id='topic+predict.SIS'></span>

<h3>Description</h3>

<p>Similar to the usual predict methods, this function returns predictions from
a fitted <code>'SIS'</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SIS'
predict(
  object,
  newx,
  lambda = object$lambda,
  which = NULL,
  type = c("response", "link", "class"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.SIS_+3A_object">object</code></td>
<td>
<p>Fitted <code>'SIS'</code> model object.</p>
</td></tr>
<tr><td><code id="predict.SIS_+3A_newx">newx</code></td>
<td>
<p>Matrix of new values for <code>x</code> at which predictions are to be
made, without the intercept term.</p>
</td></tr>
<tr><td><code id="predict.SIS_+3A_lambda">lambda</code></td>
<td>
<p>Penalty parameter <code>lambda</code> of the final fitted
model by (I)SIS at which predictions are required. By default, only the
lambda minimizing the criterion <code>tune</code> is returned.</p>
</td></tr>
<tr><td><code id="predict.SIS_+3A_which">which</code></td>
<td>
<p>Indices of the penalty parameter <code>lambda</code> of the final fitted
model by (I)SIS at which predictions are required. If supplied, will overwrite 
the default <code>lambda</code> value.</p>
</td></tr>
<tr><td><code id="predict.SIS_+3A_type">type</code></td>
<td>
<p>Type of prediction required. Type <code>'response'</code> gives the
fitted values for <code>'gaussian'</code>, fitted probabilities for
<code>'binomial'</code>, fitted mean for <code>'poisson'</code>, and the fitted relative
risk for <code>'cox'</code>. Type <code>'link'</code> returns the linear predictors for
<code>'binomial'</code>, <code>'poisson'</code> and <code>'cox'</code> models; for
<code>'gaussian'</code> models it is equivalent to type <code>'response'</code>. Type
<code>'class'</code> applies only to <code>'binomial'</code> models, and produces the
class label corresponding to the maximum probability (0-1 labels).</p>
</td></tr>
<tr><td><code id="predict.SIS_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments to predict.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on type.
</p>


<h3>Author(s)</h3>

<p>Jianqing Fan, Yang Feng, Diego Franco Saldana, Richard Samworth, and
Yichao Wu
</p>


<h3>References</h3>

<p>Diego Franco Saldana and Yang Feng (2018) SIS: An R package for Sure Independence Screening in
Ultrahigh Dimensional Statistical Models, <em>Journal of Statistical Software</em>, <b>83</b>, 2, 1-25.
</p>
<p>Jianqing Fan and Jinchi Lv (2008) Sure Independence Screening
for Ultrahigh Dimensional Feature Space (with discussion). <em>Journal of
Royal Statistical Society B</em>, <b>70</b>, 849-911.
</p>
<p>Jianqing Fan and Rui Song (2010) Sure Independence Screening in Generalized
Linear Models with NP-Dimensionality.  <em>The Annals of Statistics</em>,
<b>38</b>, 3567-3604.
</p>
<p>Jianqing Fan, Richard Samworth, and Yichao Wu (2009) Ultrahigh Dimensional
Feature Selection: Beyond the Linear Model. <em>Journal of Machine
Learning Research</em>, <b>10</b>, 2013-2038.
</p>
<p>Jianqing Fan, Yang Feng, and Yichao Wu (2010) High-dimensional Variable
Selection for Cox Proportional Hazards Model. <em>IMS Collections</em>,
<b>6</b>, 70-86.
</p>
<p>Jianqing Fan, Yang Feng, and Rui Song (2011) Nonparametric Independence
Screening in Sparse Ultrahigh Dimensional Additive Models. <em>Journal of
the American Statistical Association</em>, <b>106</b>, 544-557.
</p>
<p>Diego Franco Saldana and Yang Feng (2014) SIS: An R package for Sure Independence Screening in
Ultrahigh Dimensional Statistical Models, <em>Journal of Statistical Software</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SIS">SIS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

set.seed(0)
n = 400; p = 50; rho = 0.5
corrmat = diag(rep(1-rho, p)) + matrix(rho, p, p)
corrmat[,4] = sqrt(rho)
corrmat[4, ] = sqrt(rho)
corrmat[4,4] = 1
corrmat[,5] = 0
corrmat[5, ] = 0
corrmat[5,5] = 1
cholmat = chol(corrmat)
x = matrix(rnorm(n*p, mean=0, sd=1), n, p)
x = x%*%cholmat
testX = matrix(rnorm(10*p, mean=0, sd=1), nrow=10, ncol=p)

# gaussian response 
set.seed(1)
b = c(4,4,4,-6*sqrt(2),4/3)
y=x[, 1:5]%*%b + rnorm(n)
model1=SIS(x, y, family='gaussian', tune='bic', varISIS='aggr', seed=11)

predict(model1, testX, type='response')
predict(model1, testX, which=1:10, type='response')

## Not run: 
# binary response 
set.seed(2)
feta = x[, 1:5]%*%b; fprob = exp(feta)/(1+exp(feta))
y = rbinom(n, 1, fprob)
model2=SIS(x, y, family='binomial', tune='bic', varISIS='aggr', seed=21)

predict(model2, testX, type='response')
predict(model2, testX, type='link')
predict(model2, testX, type='class')

predict(model2, testX, which=1:10, type='response')
predict(model2, testX, which=1:10, type='link')
predict(model2, testX, which=1:10, type='class')

# poisson response
set.seed(3)
b = c(0.6,0.6,0.6,-0.9*sqrt(2))
myrates = exp(x[, 1:4]%*%b)
y = rpois(n, myrates)
model3=SIS(x, y, family='poisson', penalty = 'lasso',tune='bic', varISIS='aggr', seed=31)

predict(model3, testX, type='response')
predict(model3, testX, type='link')

## End(Not run)



</code></pre>

<hr>
<h2 id='prostate.test'>Gene expression Prostate Cancer testing data set from Singh et al. (2002)</h2><span id='topic+prostate.test'></span>

<h3>Description</h3>

<p>Gene expression testing data of 12600 genes from 25 patients with 
prostate tumors and 9 normal specimens from the microarray study of 
Singh et al. (2002).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(prostate.test)</code></pre>


<h3>Format</h3>

<p>A data frame with 34 observations on 12600 variables.</p>


<h3>Source</h3>

<p>http://wwwprod.broadinstitute.org/cgi-bin/cancer/datasets.cgi</p>


<h3>References</h3>

<p>Singh et al. (2002) Gene Expression Correlates of Clinical Prostate Cancer 
Behavior. <em>Cancer Cell</em>, <b>1</b>, 203-209.
</p>

<hr>
<h2 id='prostate.train'>Gene expression Prostate Cancer training data set from Singh et al. (2002)</h2><span id='topic+prostate.train'></span>

<h3>Description</h3>

<p>Gene expression training data of 12600 genes from 52 patients with 
prostate tumors and 50 normal specimens from the microarray study of 
Singh et al. (2002).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(prostate.train)</code></pre>


<h3>Format</h3>

<p>A data frame with 102 observations on 12600 variables.</p>


<h3>Source</h3>

<p>http://wwwprod.broadinstitute.org/cgi-bin/cancer/datasets.cgi</p>


<h3>References</h3>

<p>Singh et al. (2002) Gene Expression Correlates of Clinical Prostate Cancer 
Behavior. <em>Cancer Cell</em>, <b>1</b>, 203-209.
</p>

<hr>
<h2 id='SIS'>(Iterative) Sure Independence Screening ((I)SIS) and Fitting in Generalized
Linear Models and Cox's Proportional Hazards Models</h2><span id='topic+SIS'></span>

<h3>Description</h3>

<p>This function first implements the Iterative Sure Independence Screening for
different variants of (I)SIS, and then fits the final regression model using
the R packages <span class="pkg">ncvreg</span> and <span class="pkg">glmnet</span> for the SCAD/MCP/LASSO
regularized loglikelihood for the variables picked by (I)SIS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SIS(
  x,
  y,
  family = c("gaussian", "binomial", "poisson", "cox"),
  penalty = c("SCAD", "MCP", "lasso"),
  concavity.parameter = switch(penalty, SCAD = 3.7, 3),
  tune = c("bic", "ebic", "aic", "cv"),
  nfolds = 10,
  type.measure = c("deviance", "class", "auc", "mse", "mae"),
  gamma.ebic = 1,
  nsis = NULL,
  iter = TRUE,
  iter.max = ifelse(greedy == FALSE, 10, floor(nrow(x)/log(nrow(x)))),
  varISIS = c("vanilla", "aggr", "cons"),
  perm = FALSE,
  q = 1,
  greedy = FALSE,
  greedy.size = 1,
  seed = NULL,
  standardize = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SIS_+3A_x">x</code></td>
<td>
<p>The design matrix, of dimensions n * p, without an intercept. Each
row is an observation vector.  <code>SIS</code> standardizes the data and includes
an intercept by default.</p>
</td></tr>
<tr><td><code id="SIS_+3A_y">y</code></td>
<td>
<p>The response vector of dimension n * 1. Quantitative for
<code>family='gaussian'</code>, non-negative counts for <code>family='poisson'</code>,
binary (0-1) for <code>family='binomial'</code>. For <code>family='cox'</code>, <code>y</code>
should be an object of class <code>Surv</code>, as provided by the function
<code>Surv()</code> in the package <span class="pkg">survival</span>.</p>
</td></tr>
<tr><td><code id="SIS_+3A_family">family</code></td>
<td>
<p>Response type (see above).</p>
</td></tr>
<tr><td><code id="SIS_+3A_penalty">penalty</code></td>
<td>
<p>The penalty to be applied in the regularized likelihood
subproblems. 'SCAD' (the default), 'MCP', or 'lasso' are provided.</p>
</td></tr>
<tr><td><code id="SIS_+3A_concavity.parameter">concavity.parameter</code></td>
<td>
<p>The tuning parameter used to adjust the concavity
of the SCAD/MCP penalty. Default is 3.7 for SCAD and 3 for MCP.</p>
</td></tr>
<tr><td><code id="SIS_+3A_tune">tune</code></td>
<td>
<p>Method for tuning the regularization parameter of the penalized
likelihood subproblems and of the final model selected by (I)SIS. Options
include <code>tune='bic'</code>, <code>tune='ebic'</code>, <code>tune='aic'</code>, and
<code>tune='cv'</code>.</p>
</td></tr>
<tr><td><code id="SIS_+3A_nfolds">nfolds</code></td>
<td>
<p>Number of folds used in cross-validation. The default is 10.</p>
</td></tr>
<tr><td><code id="SIS_+3A_type.measure">type.measure</code></td>
<td>
<p>Loss to use for cross-validation. Currently five
options, not all available for all models. The default is
<code>type.measure='deviance'</code>, which uses squared-error for gaussian models
(also equivalent to <code>type.measure='mse'</code> in this case), deviance for
logistic and poisson regression, and partial-likelihood for the Cox model.
Both <code>type.measure='class'</code> and <code>type.measure='auc'</code> apply only to
logistic regression and give misclassification error and area under the ROC
curve, respectively. <code>type.measure='mse'</code> or <code>type.measure='mae'</code>
(mean absolute error) can be used by all models except the <code>'cox'</code>;
they measure the deviation from the fitted mean to the response. For
<code>penalty='SCAD'</code> and <code>penalty='MCP'</code>, only
<code>type.measure='deviance'</code> is available.</p>
</td></tr>
<tr><td><code id="SIS_+3A_gamma.ebic">gamma.ebic</code></td>
<td>
<p>Specifies the parameter in the Extended BIC criterion
penalizing the size of the corresponding model space. The default is
<code>gamma.ebic=1</code>. See references at the end for details.</p>
</td></tr>
<tr><td><code id="SIS_+3A_nsis">nsis</code></td>
<td>
<p>Number of pedictors recuited by (I)SIS.</p>
</td></tr>
<tr><td><code id="SIS_+3A_iter">iter</code></td>
<td>
<p>Specifies whether to perform iterative SIS. The default is
<code>iter=TRUE</code>.</p>
</td></tr>
<tr><td><code id="SIS_+3A_iter.max">iter.max</code></td>
<td>
<p>Maximum number of iterations for (I)SIS and its variants.</p>
</td></tr>
<tr><td><code id="SIS_+3A_varisis">varISIS</code></td>
<td>
<p>Specifies whether to perform any of the two ISIS variants
based on randomly splitting the sample into two groups. The variant
<code>varISIS='aggr'</code> is an aggressive variable screening procedure, while
<code>varISIS='cons'</code> is a more conservative approach. The default is
<code>varISIS='vanilla'</code>, which performs the traditional vanilla version of
ISIS. See references at the end for details.</p>
</td></tr>
<tr><td><code id="SIS_+3A_perm">perm</code></td>
<td>
<p>Specifies whether to impose a data-driven threshold in the size
of the active sets calculated during the ISIS procedures. The threshold is
calculated by first decoupling the predictors <code class="reqn">x_i</code> and response
<code class="reqn">y_i</code> through a random permutation <code class="reqn">\pi</code> of <code class="reqn">(1,...,n)</code> to form
a null model.  For this newly permuted data, marginal regression
coefficients for each predictor are recalculated.  As the marginal
regression coeffcients of the original data should be larger than most
recalculated coefficients in the null model, the data-driven threshold is
given by the <code class="reqn">q</code>th quantile of the null coefficients. This data-driven
threshold only allows a <code class="reqn">1-q</code> proportion of inactive variables to enter
the model when <code class="reqn">x_i</code> and <code class="reqn">y_i</code> are not related (in the null model).
The default is here is <code>perm=FALSE</code>. See references at the end for
details.</p>
</td></tr>
<tr><td><code id="SIS_+3A_q">q</code></td>
<td>
<p>Quantile for calculating the data-driven threshold in the
permutation-based ISIS. The default is <code>q=1</code> (i.e., the maximum
absolute value of the permuted estimates).</p>
</td></tr>
<tr><td><code id="SIS_+3A_greedy">greedy</code></td>
<td>
<p>Specifies whether to run the greedy modification of the
permutation-based ISIS. The default is <code>greedy=FALSE</code>.</p>
</td></tr>
<tr><td><code id="SIS_+3A_greedy.size">greedy.size</code></td>
<td>
<p>Maximum size of the active sets in the greedy
modification of the permutation-based ISIS. The default is
<code>greedy.size=1</code>.</p>
</td></tr>
<tr><td><code id="SIS_+3A_seed">seed</code></td>
<td>
<p>Random seed used for sample splitting, random permutation, and
cross-validation sampling of training and test sets.</p>
</td></tr>
<tr><td><code id="SIS_+3A_standardize">standardize</code></td>
<td>
<p>Logical flag for x variable standardization, prior to
performing (iterative) variable screening.  The resulting coefficients are
always returned on the original scale. Default is <code>standardize=TRUE</code>.
If variables are in the same units already, you might not wish to
standardize.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object with </p>
<table>
<tr><td><code>sis.ix0</code></td>
<td>
<p>The vector of indices selected by
only SIS.</p>
</td></tr> <tr><td><code>ix</code></td>
<td>
<p> The vector of indices selected by
(I)SIS with regularization step.  </p>
</td></tr> <tr><td><code>coef.est</code></td>
<td>
<p> The vector of coefficients of the final model
selected by (I)SIS.  </p>
</td></tr> <tr><td><code>fit</code></td>
<td>
<p> A fitted object of type <code>ncvreg</code>,
<code>cv.ncvreg</code>, <code>glmnet</code>, or <code>cv.glmnet</code> for the final model
selected by the (I)SIS procedure. If <code>tune='cv'</code>, the returned fitted
object is of type <code>cv.ncvreg</code> if <code>penalty='SCAD'</code> or
<code>penalty='MCP'</code>; otherwise, the returned fitted object is of type
<code>cv.glmnet</code>. For the remaining options of <code>tune</code>, the returned
object is of type <code>glmnet</code> if <code>penalty='lasso'</code>, and <code>ncvreg</code>
otherwise.  </p>
</td></tr> <tr><td><code>path.index</code></td>
<td>
<p> The index along the solution path of
<code>fit</code> for which the criterion specified in <code>tune</code> is minimized.  </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jianqing Fan, Yang Feng, Diego Franco Saldana, Richard Samworth, and
Yichao Wu
</p>


<h3>References</h3>

<p>Diego Franco Saldana and Yang Feng (2018) SIS: An R package for Sure Independence Screening in
Ultrahigh Dimensional Statistical Models, <em>Journal of Statistical Software</em>, <b>83</b>, 2, 1-25.
</p>
<p>Jianqing Fan and Jinchi Lv (2008) Sure Independence Screening
for Ultrahigh Dimensional Feature Space (with discussion). <em>Journal of
Royal Statistical Society B</em>, <b>70</b>, 849-911.
</p>
<p>Jianqing Fan and Rui Song (2010) Sure Independence Screening in Generalized
Linear Models with NP-Dimensionality.  <em>The Annals of Statistics</em>,
<b>38</b>, 3567-3604.
</p>
<p>Jianqing Fan, Richard Samworth, and Yichao Wu (2009) Ultrahigh Dimensional
Feature Selection: Beyond the Linear Model. <em>Journal of Machine
Learning Research</em>, <b>10</b>, 2013-2038.
</p>
<p>Jianqing Fan, Yang Feng, and Yichao Wu (2010) High-dimensional Variable
Selection for Cox Proportional Hazards Model. <em>IMS Collections</em>,
<b>6</b>, 70-86.
</p>
<p>Jianqing Fan, Yang Feng, and Rui Song (2011) Nonparametric Independence
Screening in Sparse Ultrahigh Dimensional Additive Models. <em>Journal of
the American Statistical Association</em>, <b>106</b>, 544-557.
</p>
<p>Jiahua Chen and Zehua Chen (2008) Extended Bayesian Information Criteria for
Model Selection with Large Model Spaces. <em>Biometrika</em>, <b>95</b>,
759-771.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict.SIS">predict.SIS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

set.seed(0)
n = 400; p = 50; rho = 0.5
corrmat = diag(rep(1-rho, p)) + matrix(rho, p, p)
corrmat[,4] = sqrt(rho)
corrmat[4, ] = sqrt(rho)
corrmat[4,4] = 1
corrmat[,5] = 0
corrmat[5, ] = 0
corrmat[5,5] = 1
cholmat = chol(corrmat)
x = matrix(rnorm(n*p, mean=0, sd=1), n, p)
x = x%*%cholmat

# gaussian response 
set.seed(1)
b = c(4,4,4,-6*sqrt(2),4/3)
y=x[, 1:5]%*%b + rnorm(n)
# SIS without regularization
model10 = SIS(x, y, family='gaussian', iter = FALSE)
model10$sis.ix0
# ISIS with regularization
model11=SIS(x, y, family='gaussian', tune='bic')
model12=SIS(x, y, family='gaussian', tune='bic', varISIS='aggr', seed=11)
model11$ix
model12$ix

## Not run: 
# binary response 
set.seed(2)
feta = x[, 1:5]%*%b; fprob = exp(feta)/(1+exp(feta))
y = rbinom(n, 1, fprob)
model21=SIS(x, y, family='binomial', tune='bic')
model22=SIS(x, y, family='binomial', tune='bic', varISIS='aggr', seed=21)
model21$ix
model22$ix

# poisson response
set.seed(3)
b = c(0.6,0.6,0.6,-0.9*sqrt(2))
myrates = exp(x[, 1:4]%*%b)
y = rpois(n, myrates)
model31=SIS(x, y, family='poisson', penalty = 'lasso', tune='bic', perm=TRUE, q=0.9, 
            greedy=TRUE, seed=31)
model32=SIS(x, y, family='poisson', penalty = 'lasso',  tune='bic', varISIS='aggr', 
            perm=TRUE, q=0.9, seed=32)
model31$ix
model32$ix

# Cox model
set.seed(4)
b = c(4,4,4,-6*sqrt(2),4/3)
myrates = exp(x[, 1:5]%*%b)
Sur = rexp(n,myrates); CT = rexp(n,0.1)
Z = pmin(Sur,CT); ind = as.numeric(Sur&lt;=CT)
y = survival::Surv(Z,ind)
model41=SIS(x, y, family='cox', penalty='lasso', tune='bic', 
           varISIS='aggr', seed=41)
model42=SIS(x, y, family='cox', penalty='lasso', tune='bic', 
             varISIS='cons', seed=41)
model41$ix
model42$ix

## End(Not run)

</code></pre>

<hr>
<h2 id='standardize'>Standardization of High-Dimensional Design Matrices</h2><span id='topic+standardize'></span>

<h3>Description</h3>

<p>Standardizes the columns of a high-dimensional design matrix to mean zero
and unit Euclidean norm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardize(X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="standardize_+3A_x">X</code></td>
<td>
<p>A design matrix to be standardized.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs a location and scale transform to the columns of the original
design matrix, so that the resulting design matrix with <code class="reqn">p</code>-dimensional
observations <code class="reqn">\{x_i : i=1,...,n\}</code> of the form
<code class="reqn">x_i=(x_{i1},x_{i2},...,x_{ip})</code> satisfies <code class="reqn">\sum_{i=1}^{n} x_{ij} =
0</code> and <code class="reqn">\sum_{i=1}^{n} x_{ij}^{2} = 1</code> for <code class="reqn">j=1,...,p</code>.
</p>


<h3>Value</h3>

<p>A design matrix with standardized predictors or columns.
</p>


<h3>Author(s)</h3>

<p>Jianqing Fan, Yang Feng, Diego Franco Saldana, Richard Samworth, and
Yichao Wu
</p>


<h3>References</h3>

<p>Diego Franco Saldana and Yang Feng (2018) SIS: An R package for Sure Independence Screening in
Ultrahigh Dimensional Statistical Models, <em>Journal of Statistical Software</em>, <b>83</b>, 2, 1-25.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(0)
n = 400; p = 50; rho = 0.5
corrmat = diag(rep(1-rho, p)) + matrix(rho, p, p)
corrmat[,4] = sqrt(rho)
corrmat[4, ] = sqrt(rho)
corrmat[4,4] = 1
corrmat[,5] = 0
corrmat[5, ] = 0
corrmat[5,5] = 1
cholmat = chol(corrmat)
x = matrix(rnorm(n*p, mean=15, sd=9), n, p)
x = x%*%cholmat

x.standard = standardize(x)

</code></pre>

<hr>
<h2 id='tune.fit'>Using the <span class="pkg">glmnet</span> and <span class="pkg">ncvreg</span> packages, fits a Generalized Linear
Model or Cox Proportional Hazards Model using various methods for choosing
the regularization parameter <code class="reqn">\lambda</code></h2><span id='topic+tune.fit'></span>

<h3>Description</h3>

<p>This function fits a generalized linear model or a Cox proportional hazards
model via penalized maximum likelihood, with available penalties as
indicated in the <span class="pkg">glmnet</span> and <span class="pkg">ncvreg</span> packages. Instead of
providing the whole regularization solution path, the function returns the
solution at a unique value of <code class="reqn">\lambda</code>, the one optimizing the
criterion specified in <code>tune</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tune.fit(
  x,
  y,
  family = c("gaussian", "binomial", "poisson", "cox"),
  penalty = c("SCAD", "MCP", "lasso"),
  concavity.parameter = switch(penalty, SCAD = 3.7, 3),
  tune = c("cv", "aic", "bic", "ebic"),
  nfolds = 10,
  type.measure = c("deviance", "class", "auc", "mse", "mae"),
  gamma.ebic = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tune.fit_+3A_x">x</code></td>
<td>
<p>The design matrix, of dimensions n * p, without an intercept. Each
row is an observation vector.</p>
</td></tr>
<tr><td><code id="tune.fit_+3A_y">y</code></td>
<td>
<p>The response vector of dimension n * 1. Quantitative for
<code>family='gaussian'</code>, non-negative counts for <code>family='poisson'</code>,
binary (0-1) for <code>family='binomial'</code>. For <code>family='cox'</code>, <code>y</code>
should be an object of class <code>Surv</code>, as provided by the function
<code>Surv()</code> in the package <span class="pkg">survival</span>.</p>
</td></tr>
<tr><td><code id="tune.fit_+3A_family">family</code></td>
<td>
<p>Response type (see above).</p>
</td></tr>
<tr><td><code id="tune.fit_+3A_penalty">penalty</code></td>
<td>
<p>The penalty to be applied in the regularized likelihood
subproblems. 'SCAD' (the default), 'MCP', or 'lasso' are provided.</p>
</td></tr>
<tr><td><code id="tune.fit_+3A_concavity.parameter">concavity.parameter</code></td>
<td>
<p>The tuning parameter used to adjust the concavity
of the SCAD/MCP penalty. Default is 3.7 for SCAD and 3 for MCP.</p>
</td></tr>
<tr><td><code id="tune.fit_+3A_tune">tune</code></td>
<td>
<p>Method for selecting the regularization parameter along the
solution path of the penalized likelihood problem. Options to provide a
final model include <code>tune='cv'</code>, <code>tune='aic'</code>, <code>tune='bic'</code>,
and <code>tune='ebic'</code>. See references at the end for details.</p>
</td></tr>
<tr><td><code id="tune.fit_+3A_nfolds">nfolds</code></td>
<td>
<p>Number of folds used in cross-validation. The default is 10.</p>
</td></tr>
<tr><td><code id="tune.fit_+3A_type.measure">type.measure</code></td>
<td>
<p>Loss to use for cross-validation. Currently five
options, not all available for all models. The default is
<code>type.measure='deviance'</code>, which uses squared-error for gaussian models
(also equivalent to <code>type.measure='mse'</code> in this case), deviance for
logistic and poisson regression, and partial-likelihood for the Cox model.
Both <code>type.measure='class'</code> and <code>type.measure='auc'</code> apply only to
logistic regression and give misclassification error and area under the ROC
curve, respectively. <code>type.measure='mse'</code> or <code>type.measure='mae'</code>
(mean absolute error) can be used by all models except the <code>'cox'</code>;
they measure the deviation from the fitted mean to the response. For
<code>penalty='SCAD'</code> and <code>penalty='MCP'</code>, only
<code>type.measure='deviance'</code> is available.</p>
</td></tr>
<tr><td><code id="tune.fit_+3A_gamma.ebic">gamma.ebic</code></td>
<td>
<p>Specifies the parameter in the Extended BIC criterion
penalizing the size of the corresponding model space. The default is
<code>gamma.ebic=1</code>. See references at the end for details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object with </p>
<table>
<tr><td><code>ix</code></td>
<td>
<p> The vector of indices of the
nonzero coefficients selected by the maximum penalized likelihood procedure
with <code>tune</code> as the method for choosing the regularization parameter.  </p>
</td></tr>
<tr><td><code>a0</code></td>
<td>
<p>The intercept of the final model selected by <code>tune</code>.  </p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>The vector of coefficients of the final model selected by
<code>tune</code>.  </p>
</td></tr>
<tr><td><code>fit</code></td>
<td>
<p>The fitted penalized regression object.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>The corresponding lambda in the final model.</p>
</td></tr>
<tr><td><code>lambda.ind</code></td>
<td>
<p>The index on the solution path for the final model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jianqing Fan, Yang Feng, Diego Franco Saldana, Richard Samworth, and
Yichao Wu
</p>


<h3>References</h3>

<p>Jerome Friedman and Trevor Hastie and Rob Tibshirani (2010)
Regularization Paths for Generalized Linear Models Via Coordinate Descent.
<em>Journal of Statistical Software</em>, <b>33</b>(1), 1-22.
</p>
<p>Noah Simon and Jerome Friedman and Trevor Hastie and Rob Tibshirani (2011)
Regularization Paths for Cox's Proportional Hazards Model Via Coordinate
Descent. <em>Journal of Statistical Software</em>, <b>39</b>(5), 1-13.
</p>
<p>Patrick Breheny and Jian Huang (2011) Coordiante Descent Algorithms for
Nonconvex Penalized Regression, with Applications to Biological Feature
Selection. <em>The Annals of Applied Statistics</em>, <b>5</b>, 232-253.
</p>
<p>Hirotogu Akaike (1973) Information Theory and an Extension of the Maximum
Likelihood Principle. In <em>Proceedings of the 2nd International
Symposium on Information Theory</em>, BN Petrov and F Csaki (eds.), 267-281.
</p>
<p>Gideon Schwarz (1978) Estimating the Dimension of a Model. <em>The Annals
of Statistics</em>, <b>6</b>, 461-464.
</p>
<p>Jiahua Chen and Zehua Chen (2008) Extended Bayesian Information Criteria for
Model Selection with Large Model Spaces. <em>Biometrika</em>, <b>95</b>,
759-771.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

set.seed(0)
data('leukemia.train', package = 'SIS')
y.train = leukemia.train[,dim(leukemia.train)[2]]
x.train = as.matrix(leukemia.train[,-dim(leukemia.train)[2]])
x.train = standardize(x.train)
model = tune.fit(x.train[,1:3500], y.train, family='binomial', tune='bic')
model$ix
model$a0
model$beta


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
