<!DOCTYPE html><html><head><title>Help for package SITH</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SITH}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SITH-package'><p>Visualize and analyze intratumor heterogeneity using a spatial model of tumor growth</p></a></li>
<li><a href='#bulkSample'><p>Simulate bulk sampling</p></a></li>
<li><a href='#plotSlice'><p>2D cross section of the simulated tumor</p></a></li>
<li><a href='#progressionChain'><p>Create a linear chain graph to describe the order of mutations</p></a></li>
<li><a href='#progressionDAG_from_igraph'><p>Define the progression of mutations from an <code>igraph</code> object</p></a></li>
<li><a href='#randomBulkSamples'><p>Simulate multi-region bulk sampling</p></a></li>
<li><a href='#randomNeedles'><p>Simulate fine needle aspiration</p></a></li>
<li><a href='#randomSingleCells'><p>Simulate single cell sequencing data</p></a></li>
<li><a href='#simulateTumor'><p>Spatial simulation of tumor growth</p></a></li>
<li><a href='#singleCell'><p>Simulate single cell sequencing data</p></a></li>
<li><a href='#spatialDistribution'><p>Quantify the spatial distribution of mutants</p></a></li>
<li><a href='#visualizeTumor'><p>Interactive visualization of the simulated tumor</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Spatial Model of Intra-Tumor Heterogeneity</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-01-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Phillip B. Nicol</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Phillip B. Nicol &lt;philnicol740@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements a three-dimensional stochastic model of cancer growth and mutation similar to the one described in Waclaw et al. (2015) &lt;<a href="https://doi.org/10.1038%2Fnature14971">doi:10.1038/nature14971</a>&gt;. Allows for interactive 3D visualizations of the simulated tumor. Provides a comprehensive summary of the spatial distribution of mutants within the tumor. Contains functions which create synthetic sequencing datasets from the generated tumor. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.4), scatterplot3d, stats, graphics, grDevices</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rgl, igraph, knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/phillipnicol/SITH">https://github.com/phillipnicol/SITH</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/phillipnicol/SITH/issues">https://github.com/phillipnicol/SITH/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-04 18:31:46 UTC; phillipnicol</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-05 15:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='SITH-package'>Visualize and analyze intratumor heterogeneity using a spatial model of tumor growth</h2><span id='topic+SITH-package'></span><span id='topic+SITH'></span>

<h3>Description</h3>

<p>The <span class="pkg">SITH</span> (spatial model of intratumor heterogeneity) package implements a lattice based spatial model
of tumor growth and mutation. Interactive 3D visualization of the tumor are possible using <span class="pkg">rgl</span>. Additional functions for visualization and investigating the spatial distribution of mutants are provided. <span class="pkg">SITH</span> also
provides functions to simulate single cell sequencing and bulk sampling data sets from the simulated tumor. 
</p>


<h3>Background</h3>

<p>On-lattice models of tumor growth and mutation are computationally efficient and provide a simple setting to study how spatially constrained growth impacts intratumor heterogeneity. While this model has been studied extensively in literature (see Waclaw (2015), Chkhaidze (2019), Opasic (2019)), existing software is either not publicly available or inconvenient to use with <span class="rlang"><b>R</b></span>. 
</p>
<p>The motivation for creating the <span class="pkg">SITH</span> package was to provide a spatial simulator that is both easy to use and can be used entirely with <span class="rlang"><b>R</b></span>. The core function in the package is <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>, which wraps a C++ implementation of the model into <span class="rlang"><b>R</b></span> using the <span class="pkg">Rcpp</span> package. Once the results of the simulation are saved as an <span class="rlang"><b>R</b></span> object, <span class="pkg">SITH</span> provides several other useful functions for studying this model. 
</p>
<p>See the package vignette for more information on the model and the algorithm used. 
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol</p>


<h3>References</h3>

<p>B. Waclaw, I. Bozic, M. Pittman, R. Hruban, B. Vogelstein and M. Nowak. A spatial model predicts that dispersal and cell turnover limit intratumor heterogeneity. <em>Nature</em>, pages 261-264, 2015. https://doi.org/10.1038/nature14971. 
</p>
<p>K. Chkhaidze, T. Heide, B. Werner, M. Williams, W. Huang, G. Caravagna, T. Graham, and A. Sottoriva. Spatially constrained tumour growth affects the patterns of clonal selection and neutral drift in cancer genomic data. <em>PLOS Computational Biology</em>, 2019. https://doi.org/10.1371/journal.pcbi.1007243. 
</p>
<p>L. Opasic, D. Zhou, B. Wener, D. Dingli and A. Traulsen. How many samples are needed to infer truly clonal mutations
from heterogeneous tumours? <em>BMC Cancer</em>, https://doi.org/10.1186/s12885-019-5597-1. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Simulate tumor growth
out &lt;- simulateTumor()
  
#3d interactive visualization using rgl 
visualizeTumor(out)
#or see regions with lots of mutants
visualizeTumor(out, plot.type = "heat")
  
#get a summary on the spatial dist. of mutants
sp &lt;- spatialDistribution(out)
  
#simulate single cell sequencing
Scs &lt;- randomSingleCells(tumor = out, ncells = 5, fnr = 0.1)
  
#simulate bulk sampling
Bulks &lt;- randomBulkSamples(tumor = out, nsamples = 5)

</code></pre>

<hr>
<h2 id='bulkSample'>Simulate bulk sampling</h2><span id='topic+bulkSample'></span>

<h3>Description</h3>

<p>Simulate bulk sequencing data by taking a local sample from the tumor
and computing the variant allele frequencies of the various mutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bulkSample(tumor, pos, cube.length = 5, threshold = 0.05, coverage = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bulkSample_+3A_tumor">tumor</code></td>
<td>
<p>A list which is the output of <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.</p>
</td></tr>
<tr><td><code id="bulkSample_+3A_pos">pos</code></td>
<td>
<p>The center point of the sample.</p>
</td></tr>
<tr><td><code id="bulkSample_+3A_cube.length">cube.length</code></td>
<td>
<p>The side length of the cube of cells to be sampled.</p>
</td></tr>
<tr><td><code id="bulkSample_+3A_threshold">threshold</code></td>
<td>
<p>Only mutations with an allele frequency greater than the threshold will be included in the sample.</p>
</td></tr>
<tr><td><code id="bulkSample_+3A_coverage">coverage</code></td>
<td>
<p>If nonzero then deep sequencing with specified coverage is performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A local region of the tumor is sampled by constructing a cube with side length <code>cube.length</code> around
the center point <code>pos</code>. Each cell within the cube is sampled, and the reported quantity is variant (or mutation) 
allele frequency. Lattice sites without cells are assumed to be normal tissue, and thus the reported MAF may be less than
1.0 even if the mutation is present in all cancerous cells. 
</p>
<p>If <code>coverage</code> is non-zero then deep sequencing can be simulated. For a chosen coverage <code class="reqn">C</code>, it is known
that the number of times the base is read follows a <code class="reqn">Pois(C)</code> distribution (see Illumina's website). 
Let <code class="reqn">d</code> be the true coverage
sampled from this distribution. Then the estimated VAF is drawn from a <code class="reqn">Bin(d,p)/d</code> distribution. 
</p>
<p>Note that <code>cube.length</code> is required to be an odd integer (in order to have a well-defined center point).
</p>


<h3>Value</h3>

<p>A data frame with 1 row and columns corresponding to the mutations. The entries are the mutation allele frequency.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol
</p>


<h3>References</h3>

<p>K. Chkhaidze, T. Heide, B. Werner, M. Williams, W. Huang, G. Caravagna, T. Graham, and 
A. Sottoriva. Spatially con- strained tumour growth affects the 
patterns of clonal selection and neutral drift in cancer genomic data. PLOS Computational Biology, 2019.
https://doi.org/10.1371/journal.pcbi.1007243.
Lander ES, Waterman MS.(1988) Genomic mapping by fingerprinting random clones: 
a mathematical analysis, Genomics 2(3): 231-239.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(116776544, kind = "Mersenne-Twister", normal.kind = "Inversion")
out &lt;- simulateTumor(max_pop = 1000)
df &lt;- bulkSample(tumor = out, pos = c(0,0,0))

</code></pre>

<hr>
<h2 id='plotSlice'>2D cross section of the simulated tumor</h2><span id='topic+plotSlice'></span>

<h3>Description</h3>

<p>2D cross section of the simulated tumor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSlice(tumor, slice.dim = "x", level = 0, plot.type = "normal")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSlice_+3A_tumor">tumor</code></td>
<td>
<p>A list which is the output of <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.</p>
</td></tr>
<tr><td><code id="plotSlice_+3A_slice.dim">slice.dim</code></td>
<td>
<p>One of &quot;x&quot;, &quot;y&quot; or &quot;z&quot;, which denotes the dimension which will be fixed to obtain a 2D cross section.</p>
</td></tr>
<tr><td><code id="plotSlice_+3A_level">level</code></td>
<td>
<p>Which value will the dimension given in <code>slice.dim</code> be fixed at?</p>
</td></tr>
<tr><td><code id="plotSlice_+3A_plot.type">plot.type</code></td>
<td>
<p>Which type of plot to draw. &quot;Normal&quot; assigns a random rgb value to each genotype while
&quot;heat&quot; colors cells with more mutations red and cells with fewer mutations blue. This is exactly the same as <code>plot.type</code>
in <code>visualizeTumor</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol
</p>

<hr>
<h2 id='progressionChain'>Create a linear chain graph to describe the order of mutations</h2><span id='topic+progressionChain'></span>

<h3>Description</h3>

<p>A helper function for <code><a href="#topic+simulateTumor">simulateTumor</a>()</code> which returns to the user
the edge list for a linear chain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>progressionChain(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="progressionChain_+3A_n">n</code></td>
<td>
<p>Number of vertices in the chain</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with 4 columns and n-1 rows which will be accepted as input to
<code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol &lt;philnicol740@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G &lt;- progressionChain(3)
</code></pre>

<hr>
<h2 id='progressionDAG_from_igraph'>Define the progression of mutations from an <code>igraph</code> object</h2><span id='topic+progressionDAG_from_igraph'></span>

<h3>Description</h3>

<p>A helper function for <code><a href="#topic+simulateTumor">simulateTumor</a>()</code> which returns to the user
the edge list for a DAG which is defined as an <code>igraph</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>progressionDAG_from_igraph(iG)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="progressionDAG_from_igraph_+3A_ig">iG</code></td>
<td>
<p>An igraph object for a directed acyclic graph.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with 4 columns which contains the edges of the graph as well as the rate of 
crossing each edge and the selective advantage/disadvantage obtained by crossing each edge.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol &lt;philnicol740@gmail.com&gt;
</p>

<hr>
<h2 id='randomBulkSamples'>Simulate multi-region bulk sampling</h2><span id='topic+randomBulkSamples'></span>

<h3>Description</h3>

<p>Simulate bulk sequencing data by taking a local sample from the tumor
and computing the variant allele frequencies of the various mutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>randomBulkSamples(
  tumor,
  nsamples,
  cube.length = 5,
  threshold = 0.05,
  coverage = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="randomBulkSamples_+3A_tumor">tumor</code></td>
<td>
<p>A list which is the output of <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.</p>
</td></tr>
<tr><td><code id="randomBulkSamples_+3A_nsamples">nsamples</code></td>
<td>
<p>The number of bulk samples to take.</p>
</td></tr>
<tr><td><code id="randomBulkSamples_+3A_cube.length">cube.length</code></td>
<td>
<p>The side length of the cube of cells to be sampled.</p>
</td></tr>
<tr><td><code id="randomBulkSamples_+3A_threshold">threshold</code></td>
<td>
<p>Only mutations with an allele frequency greater than the threshold will be included in the sample.</p>
</td></tr>
<tr><td><code id="randomBulkSamples_+3A_coverage">coverage</code></td>
<td>
<p>If nonzero then deep sequencing with specified coverage is performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the same as <code><a href="#topic+bulkSample">bulkSample</a>()</code>, except multiple samples are taken 
with random center points.
</p>


<h3>Value</h3>

<p>A data frame with <code>nsamples</code> rows and columns corresponding to the mutations. 
The entries are the mutation allele frequency.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- simulateTumor(max_pop = 1000)
df &lt;- randomBulkSamples(tumor = out, nsamples = 5, cube.length = 5, threshold = 0.05)

</code></pre>

<hr>
<h2 id='randomNeedles'>Simulate fine needle aspiration</h2><span id='topic+randomNeedles'></span>

<h3>Description</h3>

<p>Simulate a sampling procedure which takes a fine needle through the simulated tumor and
reports the mutation allele frequency of the sampled cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>randomNeedles(tumor, nsamples, threshold = 0.05, coverage = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="randomNeedles_+3A_tumor">tumor</code></td>
<td>
<p>A list which is the output of <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.</p>
</td></tr>
<tr><td><code id="randomNeedles_+3A_nsamples">nsamples</code></td>
<td>
<p>The number of samples to take.</p>
</td></tr>
<tr><td><code id="randomNeedles_+3A_threshold">threshold</code></td>
<td>
<p>Only mutations with an allele frequency greater than the threshold will be included in the sample.</p>
</td></tr>
<tr><td><code id="randomNeedles_+3A_coverage">coverage</code></td>
<td>
<p>If nonzero then deep sequencing with specified coverage is performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This sampling procedure is inspired by Chkhaidze et. al. (2019) and simulates 
fine needle aspiration. A random one-dimensional cross-section 
of the tumor is chosen, and the cells within this cross section are sampled, reporting mutation allele frequency.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol
</p>


<h3>References</h3>

<p>K. Chkhaidze, T. Heide, B. Werner, M. Williams, W. Huang, G. Caravagna, T. Graham, and 
A. Sottoriva. Spatially con- strained tumour growth affects the 
patterns of clonal selection and neutral drift in cancer genomic data. PLOS Computational Biology, 2019.
https://doi.org/10.1371/journal.pcbi.1007243.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- simulateTumor(max_pop = 1000)
df &lt;- randomNeedles(tumor = out, nsamples = 5)

</code></pre>

<hr>
<h2 id='randomSingleCells'>Simulate single cell sequencing data</h2><span id='topic+randomSingleCells'></span>

<h3>Description</h3>

<p>Simulate single cell sequencing data by random selecting cells from the tumor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>randomSingleCells(tumor, ncells, fpr = 0, fnr = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="randomSingleCells_+3A_tumor">tumor</code></td>
<td>
<p>A list which is the output of <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.</p>
</td></tr>
<tr><td><code id="randomSingleCells_+3A_ncells">ncells</code></td>
<td>
<p>The number of cells to sample.</p>
</td></tr>
<tr><td><code id="randomSingleCells_+3A_fpr">fpr</code></td>
<td>
<p>The false positive rate</p>
</td></tr>
<tr><td><code id="randomSingleCells_+3A_fnr">fnr</code></td>
<td>
<p>The false negative rate</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The procedure is exactly the same as <code><a href="#topic+singleCell">singleCell</a>()</code> except that it allows multiple cells
to be sequenced at once (chosen randomly throughout the entire tumor).
</p>


<h3>Value</h3>

<p>A data frame with sample names on the row and mutation ID on the column. 
A 1 indicates that the mutation is present in the cell and a 0 indicates the mutation is not present.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol &lt;philnicol740@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- simulateTumor(max_pop = 1000)
df &lt;- randomSingleCells(tumor = out, ncells = 5, fnr = 0.1) 
</code></pre>

<hr>
<h2 id='simulateTumor'>Spatial simulation of tumor growth</h2><span id='topic+simulateTumor'></span>

<h3>Description</h3>

<p>Simulate the spatial growth of a tumor with a multi-type branching
process on the three-dimensional integer lattice.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateTumor(
  max_pop = 250000,
  div_rate = 0.25,
  death_rate = 0.18,
  mut_rate = 0.01,
  driver_prob = 0.003,
  selective_adv = 1.05,
  disease_model = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateTumor_+3A_max_pop">max_pop</code></td>
<td>
<p>Number of cells in the tumor.</p>
</td></tr>
<tr><td><code id="simulateTumor_+3A_div_rate">div_rate</code></td>
<td>
<p>Cell division rate.</p>
</td></tr>
<tr><td><code id="simulateTumor_+3A_death_rate">death_rate</code></td>
<td>
<p>Cell death rate.</p>
</td></tr>
<tr><td><code id="simulateTumor_+3A_mut_rate">mut_rate</code></td>
<td>
<p>Mutation rate. When a cell divides, both daughter cell acquire <code class="reqn">Pois(u)</code> genetic alterations</p>
</td></tr>
<tr><td><code id="simulateTumor_+3A_driver_prob">driver_prob</code></td>
<td>
<p>The probability that a genetic alteration is a driver mutation.</p>
</td></tr>
<tr><td><code id="simulateTumor_+3A_selective_adv">selective_adv</code></td>
<td>
<p>The selective advantage conferred to a driver mutation. A cell with k
driver mutations is given birth rate <code class="reqn">bs^k</code>.</p>
</td></tr>
<tr><td><code id="simulateTumor_+3A_disease_model">disease_model</code></td>
<td>
<p>Edge list for a directed acyclic graph describing possible transitions between states. See
<code><a href="#topic+progressionChain">progressionChain</a>()</code> for an example of a valid input matrix.</p>
</td></tr>
<tr><td><code id="simulateTumor_+3A_verbose">verbose</code></td>
<td>
<p>Whether or not to print simulation details to the R console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model is based upon Waclaw et. al. (2015), although the simulation algorithm used is different. A growth of a cancerous tumor
is modeled using an exponential birth-death process on the three-dimensional integer lattice. Each cell is given a birth rate
<code class="reqn">b</code> and a death rate <code class="reqn">d</code> such that the time until cell division or cell death is exponentially distributed with 
parameters <code class="reqn">b</code> and <code class="reqn">d</code>, respectively. A cell can replicate if at least one of the six sites adjacent to it is
unoccupied. Each time cell replication occurs, both daughter cells receive <code class="reqn">Pois(u)</code> genetic alterations. Each 
alteration is a driver mutation with some probability <code class="reqn">du</code>. A cell with k driver mutations is given birth rate 
<code class="reqn">bs^k</code>. The simulation begins with a single cell at the origin at time <code class="reqn">t = 0</code>. 
</p>
<p>The model is simulated using a Gillespie algorithm. See the package vignette for details on how the algorithm is implemented.
</p>


<h3>Value</h3>

<p>A list with components 
</p>

<ul>
<li> <p><code>cell_ids</code> - A data frame containing the information for the simulated cells. (x,y,z) position, allele ID number
(note that 0 is the wild-type allele),
number of genetic alterations, and Euclidean distance from origin are included. 
</p>
</li>
<li> <p><code>muts</code> - A data frame consisting of the mutation ID number, the count of the mutation within the population, 
and the mutation allele frequency (which is the count divided by N). 
</p>
</li>
<li> <p><code>phylo_tree</code> - A data frame giving all of the information necessary to determine the order of mutations. The parent
of a mutation is defined to be the most recent mutation that precedes it. Since the ID 0 corresponds to the initial mutation,
0 does not have any parents and is thus the root of the tree. 
</p>
</li>
<li> <p><code>genotypes</code> - A data frame containing the information about the mutations that make up each allele. The <code class="reqn">i</code>-th 
row of this data frame corresponds to the allele ID <code class="reqn">i-1</code>. The positive numbers in each row correspond to the IDs of the 
mutations present in that allele, while a -1 is simply a placeholder and indicates no mutation. The count column gives
the number of cells which have the specific allele. 
</p>
</li>
<li> <p><code>color_scheme</code> - A vector containing an assignment of a color to each allele.
</p>
</li>
<li> <p><code>drivers</code> - A vector containing the ID numbers for the driver mutations.
</p>
</li>
<li> <p><code>time</code> - The simulated time (in days). 
</p>
</li>
<li> <p><code>params</code> - The parameters used for the simulation. 
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Phillip B. Nicol &lt;philnicol740@gmail.com&gt;
</p>


<h3>References</h3>

<p>B. Waclaw, I. Bozic, M. Pittman, R. Hruban, B. Vogelstein and M. Nowak. A spatial model predicts
that dispersal and cell turnover limit intratumor heterogeneity. <em>Nature</em>, pages 261-264, 2015. 
</p>
<p>D. Gillespie. Exact stochastic simulation of coupled chemical reactions. <em>The Journal of Physical Chemistry</em>,
volume 81, pages 2340-2361, 1970.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- simulateTumor(max_pop = 1000)
#Take a look at mutants in order of decreasing MAF
sig_muts &lt;- out$muts[order(out$muts$MAF, decreasing = TRUE),]

#Specify the disease model
out &lt;- simulateTumor(max_pop = 1000, disease_model = progressionChain(3))

</code></pre>

<hr>
<h2 id='singleCell'>Simulate single cell sequencing data</h2><span id='topic+singleCell'></span>

<h3>Description</h3>

<p>Simulate single cell sequencing data by selecting a cell at a specified position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>singleCell(tumor, pos, noise = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="singleCell_+3A_tumor">tumor</code></td>
<td>
<p>A list which is the output of <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.</p>
</td></tr>
<tr><td><code id="singleCell_+3A_pos">pos</code></td>
<td>
<p>A vector of length 3 giving the (x,y,z) coordinates of the cell to sample.</p>
</td></tr>
<tr><td><code id="singleCell_+3A_noise">noise</code></td>
<td>
<p>The false negative rate.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function selects the cell at <code>pos</code> (error if no cell at specified position exists) and returns
the list of mutations present in the cell. Due to technological artifacts, the false negative rate can be quite higher
(10-20 percent). To account for this,
the <code>noise</code> parameter introduces false negatives into the data set at the specified rate.
</p>


<h3>Value</h3>

<p>A data frame with 1 row and columns corresponding to the mutations present in the cell. A 1 indicates that
the mutation is detected while a 0 indicates the mutation is not detected.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol &lt;philnicol740@gmail.com&gt;
</p>


<h3>References</h3>

<p>K. Jahn, J. Kupiers and N. Beerenwinkel. Tree inference for single-cell data. Genome Biology, volume 17, 2016. 
https://doi.org/10.1186/s13059-016-0936-x.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1126490984)
out &lt;- simulateTumor(max_pop = 1000)
df &lt;- singleCell(tumor = out, pos = c(0,0,0), noise = 0.1)

</code></pre>

<hr>
<h2 id='spatialDistribution'>Quantify the spatial distribution of mutants</h2><span id='topic+spatialDistribution'></span>

<h3>Description</h3>

<p>Provides a summary the spatial distribution of mutants within the simulated
tumor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spatialDistribution(tumor, N = 500, cutoff = 0.01, make.plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spatialDistribution_+3A_tumor">tumor</code></td>
<td>
<p>A list which is the output of <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.</p>
</td></tr>
<tr><td><code id="spatialDistribution_+3A_n">N</code></td>
<td>
<p>The number of pairs to sample.</p>
</td></tr>
<tr><td><code id="spatialDistribution_+3A_cutoff">cutoff</code></td>
<td>
<p>For a plot of clone sizes, all mutations with a MAF below <code>cutoff</code> are ignored.</p>
</td></tr>
<tr><td><code id="spatialDistribution_+3A_make.plot">make.plot</code></td>
<td>
<p>Whether or not to make plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genotype of a cell can be interpreted as a binary vector where the <code class="reqn">i</code>-th component is 1 if mutation
<code class="reqn">i</code> is present in the cell and is 0 otherwise. Then a natural comparison of the similarity between two cells is the 
Jaccard index <code class="reqn">J(A,B) = |I(A,B)|/|U(A,B)|</code>, where <code class="reqn">I(A,B)</code> is the intersection of <code class="reqn">A</code> and <code class="reqn">B</code> and 
<code class="reqn">U(A,B)</code> is the union. This function estimates the Jaccard index as a function of Euclidean distance between the 
cells by randomly sampling <code class="reqn">N</code> pairs of cells.
</p>


<h3>Value</h3>

<p>A list with the following components
</p>

<ul>
<li> <p><code>mean_mutant</code> - A data frame with 2 columns giving the mean number of mutants
as a function of Euclidean distance from the lattice origin (Euclid. distance rounded to nearest integer). 
</p>
</li>
<li> <p><code>mean_driver</code> - The same as <code>mean_mutant</code> except for driver mutations only. Will be <code>NULL</code> if 
no drivers are present in the simulated tumor. 
</p>
</li>
<li> <p><code>jaccard</code> A data frame with two columns giving mean jaccard index as a function of Euclidean distance 
between pairs of cells (rounded to nearest integer). 
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Phillip B. Nicol
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1126490984)
out &lt;- simulateTumor(max_pop = 1000, driver_prob = 0.1)
sp &lt;- spatialDistribution(tumor = out, make.plot = FALSE)

</code></pre>

<hr>
<h2 id='visualizeTumor'>Interactive visualization of the simulated tumor</h2><span id='topic+visualizeTumor'></span>

<h3>Description</h3>

<p>Interactive visualization of the simulated tumor using the <code>rgl</code> package (if available).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizeTumor(tumor, plot.type = "normal", background = "black", axes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizeTumor_+3A_tumor">tumor</code></td>
<td>
<p>A list which is the output of <code><a href="#topic+simulateTumor">simulateTumor</a>()</code>.</p>
</td></tr>
<tr><td><code id="visualizeTumor_+3A_plot.type">plot.type</code></td>
<td>
<p>Which type of plot to draw. &quot;Normal&quot; assigns a random rgb value to each genotype while
&quot;heat&quot; colors cells with more mutations red and cells with fewer mutations blue.</p>
</td></tr>
<tr><td><code id="visualizeTumor_+3A_background">background</code></td>
<td>
<p>If rgl is installed, this will set the color of the background</p>
</td></tr>
<tr><td><code id="visualizeTumor_+3A_axes">axes</code></td>
<td>
<p>Will include axes (rgl only).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <span class="pkg">rgl</span> is installed, then the plots will be interactive. If <span class="pkg">rgl</span> is unavailable, static plots will
be created with <span class="pkg">scatterplot3d</span>. Since plotting performance with <span class="pkg">scatterplot3d</span> is reduced, it is strongly
recommended that <span class="pkg">rgl</span> is installed for optimal use of this function.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
