<!DOCTYPE html><html lang="en"><head><title>Help for package SlideCNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SlideCNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bin'><p>Subfunction of bin_metadata() for expression/positional binning</p></a></li>
<li><a href='#bin_metadata'><p>Spatio-molecular binning of relative expression intensities</p></a></li>
<li><a href='#center_rm'><p>Center expression intensities</p></a></li>
<li><a href='#clone_so'><p>Add clone information to meta data of seurat object and bin the beads</p></a></li>
<li><a href='#cnv_heatmap'><p>Plot CNV scores on a heat map</p></a></li>
<li><a href='#dat_to_long'><p>Convert data to long format and add in metadata</p></a></li>
<li><a href='#find_cluster_markers'><p>Find and plot top n DEGs per cluster</p></a></li>
<li><a href='#find_go_terms'><p>Find and plot top n GO-enriched terms per cluster</p></a></li>
<li><a href='#get_num_clust'><p>Find optimal number of clusters</p></a></li>
<li><a href='#long_to_bin'><p>Convert to wide bin x genes + metadata format</p></a></li>
<li><a href='#make_seurat_annot'><p>Creation of Seurat object</p></a></li>
<li><a href='#make_so_bin'><p>Make a binned version of a Seurat object</p></a></li>
<li><a href='#mean_cnv_plot'><p>Plot mean CNV scores per bin and per chromosome</p></a></li>
<li><a href='#mode'><p>Subfunction of long_to_bin() that finds mode of vector/column</p></a></li>
<li><a href='#plot_clones'><p>Plot cluster/clone information</p></a></li>
<li><a href='#prep'><p>Infercnv-based preparation of relative gene expression intensities</p></a></li>
<li><a href='#prep_cnv_dat'><p>Prepare data for CNV heat map</p></a></li>
<li><a href='#quantile_plot'><p>Plot CNV score quantiles per bin and per chromosome</p></a></li>
<li><a href='#reexports'><p>Pipe</p></a></li>
<li><a href='#ref_adj'><p>Adjust for Reference (Normal) Beads</p></a></li>
<li><a href='#run_enrichr'><p>Subfunction to get significantly enriched GO terms given a set of signfiicant beads and genes</p></a></li>
<li><a href='#run_slide_cna'><p>Run SlideCNA workflow</p></a></li>
<li><a href='#scale_nUMI'><p>Scale for nUMI (UMI Count) to generate CNV scores</p></a></li>
<li><a href='#scalefit'><p>Subfunction for scale_nUMI that normalizes a given bin for UMI count and centers the  mean CNV score at 1</p></a></li>
<li><a href='#SpatialPlot'><p>Spatial plots of meta data</p></a></li>
<li><a href='#weight_rollmean'><p>Expressional smoothing along a chromosome using a weighted pyramidal moving average</p></a></li>
<li><a href='#weight_rollmean_sub'><p>Subfunction of weight_rollmean</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Calls Copy Number Alterations from Slide-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>This takes spatial single-cell-type RNA-seq data (specifically designed for Slide-seq v2) that calls copy number alterations (CNAs) using pseudo-spatial binning, clusters cellular units (e.g. beads) based on CNA profile, and visualizes spatial CNA patterns. Documentation about 'SlideCNA' is included in the the pre-print by Zhang et al. (2022, &lt;<a href="https://doi.org/10.1101%2F2022.11.25.517982">doi:10.1101/2022.11.25.517982</a>&gt;). The package 'enrichR' (&gt;= 3.0), conditionally used to annotate SlideCNA-determined clusters with gene ontology terms, can be installed at <a href="https://github.com/wjawaid/enrichR">https://github.com/wjawaid/enrichR</a> or with install_github("wjawaid/enrichR").</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, reshape2, dplyr, ggplot2, scales, pheatmap,
cluster, factoextra, dendextend, Seurat, tidyselect, stringr,
magrittr, tibble, futile.logger, mltools, utils</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), enrichR (&ge; 3.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-22 04:05:22 UTC; dzhang</td>
</tr>
<tr>
<td>Author:</td>
<td>Diane Zhang <a href="https://orcid.org/0000-0002-0268-4413"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Johanna Klughammer
    <a href="https://orcid.org/0000-0002-3628-9278"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Jan Watter <a href="https://orcid.org/0000-0001-7519-731X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Broad Institute of MIT and Harvard [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Diane Zhang &lt;dkzhang711@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-23 17:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='bin'>Subfunction of bin_metadata() for expression/positional binning</h2><span id='topic+bin'></span>

<h3>Description</h3>

<p>This function computes a pseudospatial distance between beads that combines spatial distance and distance from the expression space, then using the silhouette score and hierarchical clustering, segregates beads into bins
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bin(
  dat,
  md,
  k,
  pos = TRUE,
  pos_k = 55,
  ex_k = 1,
  hc_function = "ward.D2",
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bin_+3A_dat">dat</code></td>
<td>
<p>data.table of smoothed relative expression intensities</p>
</td></tr>
<tr><td><code id="bin_+3A_md">md</code></td>
<td>
<p>data.table of metadata of each bead</p>
</td></tr>
<tr><td><code id="bin_+3A_k">k</code></td>
<td>
<p>number of malignant bins to set</p>
</td></tr>
<tr><td><code id="bin_+3A_pos">pos</code></td>
<td>
<p>TRUE if doing spatial and expressional binning, FALSE if just expressional binning</p>
</td></tr>
<tr><td><code id="bin_+3A_pos_k">pos_k</code></td>
<td>
<p>positional weight</p>
</td></tr>
<tr><td><code id="bin_+3A_ex_k">ex_k</code></td>
<td>
<p>expressional weight</p>
</td></tr>
<tr><td><code id="bin_+3A_hc_function">hc_function</code></td>
<td>
<p>hierarchical clustering function</p>
</td></tr>
<tr><td><code id="bin_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table of bead metadata combined with bin designations
</p>

<hr>
<h2 id='bin_metadata'>Spatio-molecular binning of relative expression intensities</h2><span id='topic+bin_metadata'></span>

<h3>Description</h3>

<p>This function combines metadata with binned relative expression intensities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bin_metadata(
  md,
  dat,
  avg_bead_per_bin = 12,
  pos = TRUE,
  pos_k = 55,
  ex_k = 1,
  hc_function = "ward.D2",
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bin_metadata_+3A_md">md</code></td>
<td>
<p>data.table of metadata of each bead</p>
</td></tr>
<tr><td><code id="bin_metadata_+3A_dat">dat</code></td>
<td>
<p>data.table of smoothed relative expression intensities</p>
</td></tr>
<tr><td><code id="bin_metadata_+3A_avg_bead_per_bin">avg_bead_per_bin</code></td>
<td>
<p>integer of average number of beads there should be per bin</p>
</td></tr>
<tr><td><code id="bin_metadata_+3A_pos">pos</code></td>
<td>
<p>TRUE if doing spatial and expressional binning, FALSE if just expressional binning</p>
</td></tr>
<tr><td><code id="bin_metadata_+3A_pos_k">pos_k</code></td>
<td>
<p>positional weight</p>
</td></tr>
<tr><td><code id="bin_metadata_+3A_ex_k">ex_k</code></td>
<td>
<p>expressional weight</p>
</td></tr>
<tr><td><code id="bin_metadata_+3A_hc_function">hc_function</code></td>
<td>
<p>hierarchical clustering function</p>
</td></tr>
<tr><td><code id="bin_metadata_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table of bead metadata combined with binned expression intensities for all genes for all beads
</p>

<hr>
<h2 id='center_rm'>Center expression intensities</h2><span id='topic+center_rm'></span>

<h3>Description</h3>

<p>Take in a data.table of genomic positions and smoothed expression intensities counts and center by
subtracting average intensity across all beads for each gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>center_rm(rm)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="center_rm_+3A_rm">rm</code></td>
<td>
<p>data.table of smoothed expression intensities counts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>centered_rm data.table of smoothed, centered expression intensities
</p>

<hr>
<h2 id='clone_so'>Add clone information to meta data of seurat object and bin the beads</h2><span id='topic+clone_so'></span>

<h3>Description</h3>

<p>This function adds another column for cluster designation to a seurat object's meta data and bins beads
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clone_so(so, hcl_sub, md, mal = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clone_so_+3A_so">so</code></td>
<td>
<p>Seurat object of beads and their meta data</p>
</td></tr>
<tr><td><code id="clone_so_+3A_hcl_sub">hcl_sub</code></td>
<td>
<p>hierarchical clustering object of cluster assignemnt as outputted from SlideCNA::plot_clones()</p>
</td></tr>
<tr><td><code id="clone_so_+3A_md">md</code></td>
<td>
<p>data.table of metadata of each bead</p>
</td></tr>
<tr><td><code id="clone_so_+3A_mal">mal</code></td>
<td>
<p>TRUE if only using malignant beads</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A seurat object updated with clone information
</p>

<hr>
<h2 id='cnv_heatmap'>Plot CNV scores on a heat map</h2><span id='topic+cnv_heatmap'></span>

<h3>Description</h3>

<p>This function prepares data for plotting and makes a heat map of CNV scores per bead across all genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cnv_heatmap(
  cnv_data,
  md,
  chrom_colors,
  hc_function = "ward.D2",
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cnv_heatmap_+3A_cnv_data">cnv_data</code></td>
<td>
<p>list object of cnv data from SlideCNA::prep_cnv_dat()</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_md">md</code></td>
<td>
<p>data.table of metadata of each bead</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_chrom_colors">chrom_colors</code></td>
<td>
<p>vector of colors labeled by which chromosome they correspond to</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_hc_function">hc_function</code></td>
<td>
<p>character for which hierarchical clustering function to use</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

<hr>
<h2 id='dat_to_long'>Convert data to long format and add in metadata</h2><span id='topic+dat_to_long'></span>

<h3>Description</h3>

<p>This function will create rows for each bead and gene combination, adding in new metadata with bin designations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat_to_long(dat, md)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dat_to_long_+3A_dat">dat</code></td>
<td>
<p>data.table of smoothed relative expression intensities</p>
</td></tr>
<tr><td><code id="dat_to_long_+3A_md">md</code></td>
<td>
<p>data.table of metadata per bead</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table of bead expression intensities per gene with metadata in long format
</p>

<hr>
<h2 id='find_cluster_markers'>Find and plot top n DEGs per cluster</h2><span id='topic+find_cluster_markers'></span>

<h3>Description</h3>

<p>This function uses Seurat's marker finding capability to find DEGs of each cluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_cluster_markers(
  so_clone,
  type,
  logfc.threshold = 0.2,
  min.pct = 0,
  only.pos = TRUE,
  n_markers = 5,
  value = "log2_expr",
  text_size = 16,
  title_size = 18,
  legend_size_pt = 4,
  p_val_thresh = 0.05,
  bin = TRUE,
  plot_directory = None
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_cluster_markers_+3A_so_clone">so_clone</code></td>
<td>
<p>seurat object with 'clone' (SlideCNA-designated cluster) and bin annotations</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_type">type</code></td>
<td>
<p>character string that is 'all' if using malignant and normal clusters and 'malig'
if just using malignant clusters</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_logfc.threshold">logfc.threshold</code></td>
<td>
<p>numeric float that is seurat parameter,
representing the minimum log2 fold change for DEGs to be significant</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_min.pct">min.pct</code></td>
<td>
<p>numeric Seurat function parameter</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_only.pos">only.pos</code></td>
<td>
<p>TRUE if only using DEGs with positive log2 fold change</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_n_markers">n_markers</code></td>
<td>
<p>integer of number of top DEGs to plot/use</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_value">value</code></td>
<td>
<p>expression value of DEGs;  one of (&quot;log2_expr&quot;, &quot;avg_expr&quot;, and &quot;avg_log2FC&quot;) for log2-normalized aerage epxression, average expression, or log2 fold change</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_text_size">text_size</code></td>
<td>
<p>Ggplot2 text size</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_title_size">title_size</code></td>
<td>
<p>Ggplot2 title size</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_legend_size_pt">legend_size_pt</code></td>
<td>
<p>Ggplot2 legend_size_pt</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_p_val_thresh">p_val_thresh</code></td>
<td>
<p>value for p value cutoff for DEGs</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_bin">bin</code></td>
<td>
<p>TRUE if using binned beads</p>
</td></tr>
<tr><td><code id="find_cluster_markers_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with cluster marker information
markers_clone = data.table of all cluster markers
top_markers_clone = data.table of just top cluster markers
top_clone_vis = data.frame formatted for plot visualization of top cluster markers
</p>

<hr>
<h2 id='find_go_terms'>Find and plot top n GO-enriched terms per cluster</h2><span id='topic+find_go_terms'></span>

<h3>Description</h3>

<p>This function utilizes cluster-specific DEGs to identify cluster-specifc GO biological processes
and plots these if they occur
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_go_terms(
  cluster_markers_obj,
  type,
  n_terms = 5,
  text_size,
  title_size,
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_go_terms_+3A_cluster_markers_obj">cluster_markers_obj</code></td>
<td>
<p>list object with cluster marker information</p>
</td></tr>
<tr><td><code id="find_go_terms_+3A_type">type</code></td>
<td>
<p>character string that is 'all' if using malignant and normal clusters and 'malig'
if just using malignant clusters</p>
</td></tr>
<tr><td><code id="find_go_terms_+3A_n_terms">n_terms</code></td>
<td>
<p>integer of number of top DEGs to plot/use</p>
</td></tr>
<tr><td><code id="find_go_terms_+3A_text_size">text_size</code></td>
<td>
<p>integer of text size for ggplot</p>
</td></tr>
<tr><td><code id="find_go_terms_+3A_title_size">title_size</code></td>
<td>
<p>integer of title size for ggplot</p>
</td></tr>
<tr><td><code id="find_go_terms_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with cluster GO term information
en_clone = data.table of cluster GO terms
top_en_clone = data.table of just top cluster GO terms
</p>

<hr>
<h2 id='get_num_clust'>Find optimal number of clusters</h2><span id='topic+get_num_clust'></span>

<h3>Description</h3>

<p>This function uses the Silhouette Method applied to CNV scores to determine the
best number of clusters to divide the binned beads into
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_num_clust(
  data,
  hc_func = "ward.D2",
  max_k = 10,
  plot = TRUE,
  malig = FALSE,
  k = NA,
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_num_clust_+3A_data">data</code></td>
<td>
<p>cnv_data list object of cnv data from SlideCNA::prep_cnv_dat()</p>
</td></tr>
<tr><td><code id="get_num_clust_+3A_hc_func">hc_func</code></td>
<td>
<p>character string for which hierarchical clustering function to use</p>
</td></tr>
<tr><td><code id="get_num_clust_+3A_max_k">max_k</code></td>
<td>
<p>integer of number max number of clusters to evaluate (2:max_k)</p>
</td></tr>
<tr><td><code id="get_num_clust_+3A_plot">plot</code></td>
<td>
<p>TRUE if plotting silhoutte scores per cluster</p>
</td></tr>
<tr><td><code id="get_num_clust_+3A_malig">malig</code></td>
<td>
<p>TRUE if only using malignant bins and FALSE if using all bins</p>
</td></tr>
<tr><td><code id="get_num_clust_+3A_k">k</code></td>
<td>
<p>integer of optimal number of clusters, if known, and NA if not known</p>
</td></tr>
<tr><td><code id="get_num_clust_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer representing the number of clusters that optimizes the silhouette score
</p>

<hr>
<h2 id='long_to_bin'>Convert to wide bin x genes + metadata format</h2><span id='topic+long_to_bin'></span>

<h3>Description</h3>

<p>This function will combine beads into bins, taking the average expression intensities, average positions,
most common cluster seurat cluster, and most common cluster/tissue type of constituent beads
</p>


<h3>Usage</h3>

<pre><code class='language-R'>long_to_bin(dat_long, plot_directory, spatial = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="long_to_bin_+3A_dat_long">dat_long</code></td>
<td>
<p>data.table of bead expression intensities per gene with metadata in long format</p>
</td></tr>
<tr><td><code id="long_to_bin_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
<tr><td><code id="long_to_bin_+3A_spatial">spatial</code></td>
<td>
<p>True if using spatial information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table of expression intensities at aggregated bin level
</p>

<hr>
<h2 id='make_seurat_annot'>Creation of Seurat object</h2><span id='topic+make_seurat_annot'></span>

<h3>Description</h3>

<p>This function takes in raw counts (and potentially meta data) to make a Seurat object
and process it
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_seurat_annot(
  cb,
  md = NULL,
  seed_FindClusters = 0,
  seed_RunTSNE = 1,
  seed_RunUMAP = 42
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_seurat_annot_+3A_cb">cb</code></td>
<td>
<p>sparse counts matrix (genes x cells/beads)</p>
</td></tr>
<tr><td><code id="make_seurat_annot_+3A_md">md</code></td>
<td>
<p>data.frame of meta data for cells/beads if specific annotations known</p>
</td></tr>
<tr><td><code id="make_seurat_annot_+3A_seed_findclusters">seed_FindClusters</code></td>
<td>
<p>seed number for FindCLusters</p>
</td></tr>
<tr><td><code id="make_seurat_annot_+3A_seed_runtsne">seed_RunTSNE</code></td>
<td>
<p>seed number for RunTSNE</p>
</td></tr>
<tr><td><code id="make_seurat_annot_+3A_seed_runumap">seed_RunUMAP</code></td>
<td>
<p>seed number for RunUMAP</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object with specific Seurat features run
</p>

<hr>
<h2 id='make_so_bin'>Make a binned version of a Seurat object</h2><span id='topic+make_so_bin'></span>

<h3>Description</h3>

<p>Aggregate Seurat object counts by bin to create a new Seurat object with binned beads
as units instead of beads
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_so_bin(so, md, hcl_sub, mal = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_so_bin_+3A_so">so</code></td>
<td>
<p>Seurat object of beads and their meta data</p>
</td></tr>
<tr><td><code id="make_so_bin_+3A_md">md</code></td>
<td>
<p>data.frame of metadata for Seurat object</p>
</td></tr>
<tr><td><code id="make_so_bin_+3A_hcl_sub">hcl_sub</code></td>
<td>
<p>hierarchical clustering object of cluster assignemnt as outputted from SlideCNA::plot_clones()</p>
</td></tr>
<tr><td><code id="make_so_bin_+3A_mal">mal</code></td>
<td>
<p>TRUE if using malignant beads only</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object with binned beads as units and corresponding binned metadata
</p>

<hr>
<h2 id='mean_cnv_plot'>Plot mean CNV scores per bin and per chromosome</h2><span id='topic+mean_cnv_plot'></span>

<h3>Description</h3>

<p>This function colors and plots each bin by its mean CNV score
on spatial coordinates for each chromosome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mean_cnv_plot(
  cnv_data,
  text_size,
  title_size,
  legend_height_bar,
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mean_cnv_plot_+3A_cnv_data">cnv_data</code></td>
<td>
<p>list object of cnv data from SlideCNA::prep_cnv_dat()</p>
</td></tr>
<tr><td><code id="mean_cnv_plot_+3A_text_size">text_size</code></td>
<td>
<p>integer of text size for ggplot</p>
</td></tr>
<tr><td><code id="mean_cnv_plot_+3A_title_size">title_size</code></td>
<td>
<p>integer of title size for ggplot</p>
</td></tr>
<tr><td><code id="mean_cnv_plot_+3A_legend_height_bar">legend_height_bar</code></td>
<td>
<p>integer of bar height of legend for ggplot</p>
</td></tr>
<tr><td><code id="mean_cnv_plot_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

<hr>
<h2 id='mode'>Subfunction of long_to_bin() that finds mode of vector/column</h2><span id='topic+mode'></span>

<h3>Description</h3>

<p>This function finds the mode of a vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mode(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mode_+3A_x">x</code></td>
<td>
<p>vector (column in data.table) to calculate the mode from</p>
</td></tr>
</table>


<h3>Value</h3>

<p>mode of the vector
</p>

<hr>
<h2 id='plot_clones'>Plot cluster/clone information</h2><span id='topic+plot_clones'></span>

<h3>Description</h3>

<p>This function plots cluster dendrograms, spatial assignment, and the CNV heat map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_clones(
  cnv_data,
  md,
  k,
  type,
  chrom_colors,
  text_size,
  title_size,
  legend_size_pt,
  legend_height_bar,
  hc_function = "ward.D2",
  plot_directory,
  spatial = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_clones_+3A_cnv_data">cnv_data</code></td>
<td>
<p>list object of cnv data from SlideCNA::prep_cnv_dat()</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_md">md</code></td>
<td>
<p>data.table of metadata of each bead</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_k">k</code></td>
<td>
<p>integer of number of clusters/clones</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_type">type</code></td>
<td>
<p>character string, being &quot;all&quot; if using all binned beads, or &quot;malig&quot; if just malignant binned beads</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_chrom_colors">chrom_colors</code></td>
<td>
<p>vector of colors labeled by which chromosome they correspond to</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_text_size">text_size</code></td>
<td>
<p>Ggplot2 text size</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_title_size">title_size</code></td>
<td>
<p>Ggplot2 title size</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_legend_size_pt">legend_size_pt</code></td>
<td>
<p>Ggplot2 legend_size_pt</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_legend_height_bar">legend_height_bar</code></td>
<td>
<p>Ggplot2 legend_height_bar</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_hc_function">hc_function</code></td>
<td>
<p>character string for which hierarchical clustering function to use</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
<tr><td><code id="plot_clones_+3A_spatial">spatial</code></td>
<td>
<p>TRUE if using spatial information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A hierarchical clustering object of the clusters
</p>

<hr>
<h2 id='prep'>Infercnv-based preparation of relative gene expression intensities</h2><span id='topic+prep'></span>

<h3>Description</h3>

<p>This function takes in a data table of raw counts and a vector of reference/normal beads
to normalize counts and adjust for reference expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep(so, normal_beads, gene_pos, chrom_ord, logTPM = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_+3A_so">so</code></td>
<td>
<p>Seurat object of Slide-seq data with raw counts</p>
</td></tr>
<tr><td><code id="prep_+3A_normal_beads">normal_beads</code></td>
<td>
<p>vector of names of normal beads</p>
</td></tr>
<tr><td><code id="prep_+3A_gene_pos">gene_pos</code></td>
<td>
<p>data.table with columns for GENE, chr, start, end, rel_gene_pos (1 : # of genes on chromosome)</p>
</td></tr>
<tr><td><code id="prep_+3A_chrom_ord">chrom_ord</code></td>
<td>
<p>vector of the names of chromosomes in order</p>
</td></tr>
<tr><td><code id="prep_+3A_logtpm">logTPM</code></td>
<td>
<p>TRUE if performing adjustment with logTPM</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table of normalized, capped, and ref-adjusted counts with genomic psoition info
</p>

<hr>
<h2 id='prep_cnv_dat'>Prepare data for CNV heat map</h2><span id='topic+prep_cnv_dat'></span>

<h3>Description</h3>

<p>This function caps CNV scores, adds annotation columns for plotting,
performs hierarchical clustering of bins based on similar CNV score, and plots nUMI per bin
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_cnv_dat(
  dat_bin,
  lower = 0.6,
  upper = 1.4,
  hc_function = "ward.D2",
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_cnv_dat_+3A_dat_bin">dat_bin</code></td>
<td>
<p>data.table of CNV scores per bin</p>
</td></tr>
<tr><td><code id="prep_cnv_dat_+3A_lower">lower</code></td>
<td>
<p>numeric float to represent the lower cap for CNV scores</p>
</td></tr>
<tr><td><code id="prep_cnv_dat_+3A_upper">upper</code></td>
<td>
<p>numeric float to represent the upper cap for CNV scores</p>
</td></tr>
<tr><td><code id="prep_cnv_dat_+3A_hc_function">hc_function</code></td>
<td>
<p>character for which hierarchical clustering function to use</p>
</td></tr>
<tr><td><code id="prep_cnv_dat_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object for downstream cnv plotting and analysis
all = data.table of CNV scores of all bins x (metadata + genes)
malig = data.table of CNV scores of just malignant bins x (metadata + genes)
all_wide = data.frame in wide format of CNV scores of all bins x (metadata + genes)
malig_wide = data.frame in wide format of CNV scores of just malignant bins x (metadata + genes)
hcl = hclust object that describes the hierarchical clustering for malignant bins
hcl_all = hclust object that describes the hierarchical clustering for all bins
</p>

<hr>
<h2 id='quantile_plot'>Plot CNV score quantiles per bin and per chromosome</h2><span id='topic+quantile_plot'></span>

<h3>Description</h3>

<p>This function colors and plots each bin by its CNV score quantiles (min, 1st quartile, median, 3rd quartile, max)
on spatial coordinates for each chromosome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quantile_plot(
  cnv_data,
  cluster_label = "seurat_clusters",
  text_size,
  title_size,
  legend_height_bar,
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="quantile_plot_+3A_cnv_data">cnv_data</code></td>
<td>
<p>list object of cnv data from SlideCNA::prep_cnv_dat()</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_cluster_label">cluster_label</code></td>
<td>
<p>character string of which column name to keep</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_text_size">text_size</code></td>
<td>
<p>integer of text size for ggplot</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_title_size">title_size</code></td>
<td>
<p>integer of title size for ggplot</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_legend_height_bar">legend_height_bar</code></td>
<td>
<p>integer of bar height of legend for ggplot</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

<hr>
<h2 id='reexports'>Pipe</h2><span id='topic+reexports'></span><span id='topic++25+3E+25'></span><span id='topic++3A+3D'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>data.table</dt><dd><p><code><a href="data.table.html#topic+assign">:=</a></code></p>
</dd>
<dt>magrittr</dt><dd><p><code><a href="magrittr.html#topic+pipe">%&gt;%</a></code></p>
</dd>
</dl>

<hr>
<h2 id='ref_adj'>Adjust for Reference (Normal) Beads</h2><span id='topic+ref_adj'></span>

<h3>Description</h3>

<p>Take in a data.table of genomic positions and smoothed, centered expression intensities counts and
adjust for reference beads by subtracting average intensities of reference beads for each gene.
This is the second reference adjustment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ref_adj(centered_rm, normal_beads)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ref_adj_+3A_centered_rm">centered_rm</code></td>
<td>
<p>data.table of smoothed, centered expression intensities counts</p>
</td></tr>
<tr><td><code id="ref_adj_+3A_normal_beads">normal_beads</code></td>
<td>
<p>vector of names of normal beads</p>
</td></tr>
</table>


<h3>Value</h3>

<p>rm_adj data.table of smoothed relative expression intensities
</p>

<hr>
<h2 id='run_enrichr'>Subfunction to get significantly enriched GO terms given a set of signfiicant beads and genes</h2><span id='topic+run_enrichr'></span>

<h3>Description</h3>

<p>This function finds the GO biological processes associated with the top n genes using enrichR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_enrichr(genes, n_genes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_enrichr_+3A_genes">genes</code></td>
<td>
<p>vector of differentially expressed genes</p>
</td></tr>
<tr><td><code id="run_enrichr_+3A_n_genes">n_genes</code></td>
<td>
<p>number of the most significantly enriched DEGs to base gene enrichment from</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table of the most significant GO terms and their meta data
</p>

<hr>
<h2 id='run_slide_cna'>Run SlideCNA workflow</h2><span id='topic+run_slide_cna'></span>

<h3>Description</h3>

<p>Take a raw expression counts, cell type annotations, and positional cooridnates to identify CNA patterns across space and CNA-based clustering patterns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_slide_cna(
  counts,
  beads_df,
  gene_pos,
  output_directory,
  plot_directory,
  spatial = TRUE,
  roll_mean_window = 101,
  avg_bead_per_bin = 12,
  pos = TRUE,
  pos_k = 55,
  ex_k = 1,
  hc_function_bin = "ward.D2",
  spatial_vars_to_plot = c("seurat_clusters", "bin_all", "N_bin", "umi_bin",
    "cluster_type"),
  scale_bin_thresh_hard = TRUE,
  lower_bound_cnv = 0.6,
  upper_bound_cnv = 1.4,
  hc_function_cnv = "ward.D2",
  hc_function_cnv_heatmap = "ward.D2",
  quantile_plot_cluster_label = "seurat_clusters",
  hc_function_silhouette = "ward.D2",
  max_k_silhouette = 10,
  plot_silhouette = TRUE,
  hc_function_plot_clones = "ward.D2",
  use_GO_terms = TRUE,
  chrom_ord = c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9",
    "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18",
    "chr19", "chr20", "chr21", "chr22", "chr23", "chrX", "chrY", "chrM"),
  chrom_colors = c(chr1 = "#8DD3C7", chr2 = "#FFFFB3", chr3 = "#BEBADA", chr4 =
    "#FB8072", chr5 = "#80B1D3", chr6 = "#FDB462", chr7 = "#B3DE69", chr8 = "#FCCDE5",
    chr9 = "#D9D9D9", chr10 = "#BC80BD", chr11 = "#CCEBC5", chr12 = "#FFED6F", chr13 =
    "#1B9E77", chr14 = "#D95F02", chr15 = "#7570B3", chr16 = "#E7298A", chr17 =
    "#66A61E", chr18 = "#E6AB02", chr19 = "#A6761D", chr20 = "#666666", chr21 =
    "#A6CEE3", chr22 = "#1F78B4", chrX = "#B2DF8A"),
  text_size = 16,
  title_size = 18,
  legend_size_pt = 4,
  legend_height_bar = 1.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_slide_cna_+3A_counts">counts</code></td>
<td>
<p>data.frame of raw counts (genes x beads)</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_beads_df">beads_df</code></td>
<td>
<p>data.frame of annotation of each bead (beads x annotations); contains columns 'bc' for bead names, 'cluster_type' for annotations of 'Normal' or 'Malignant', 'pos_x' for x-coordinate bead positions, and 'pos_y' for y-coordinate bead positions</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_gene_pos">gene_pos</code></td>
<td>
<p>data.frame with columns for GENE, chr, start, end, rel_gene_pos (1 : # of genes on chromosome)</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_output_directory">output_directory</code></td>
<td>
<p>output directory path</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_spatial">spatial</code></td>
<td>
<p>TRUE if using spatial information FALSE if not</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_roll_mean_window">roll_mean_window</code></td>
<td>
<p>integer number of adjacent genes for which to average over in pyramidal weighting scheme</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_avg_bead_per_bin">avg_bead_per_bin</code></td>
<td>
<p>integer of average number of beads there should be per bin</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_pos">pos</code></td>
<td>
<p>TRUE if doing spatial and expressional binning, FALSE if just expressional binning</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_pos_k">pos_k</code></td>
<td>
<p>positional weight</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_ex_k">ex_k</code></td>
<td>
<p>expressional weight</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_hc_function_bin">hc_function_bin</code></td>
<td>
<p>hierarchical clustering function for binning; to feed hclust's method argument, one of &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot;, &quot;mcquitty&quot;, &quot;median&quot; or &quot;centroid&quot;</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_spatial_vars_to_plot">spatial_vars_to_plot</code></td>
<td>
<p>character vector of features to plot/columns of metadata</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_scale_bin_thresh_hard">scale_bin_thresh_hard</code></td>
<td>
<p>TRUE if using strict thresholds for expression thresholds and FALSE if adjusting
thresholds based on 1 + or - the mean of absolute min and max vlaues</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_lower_bound_cnv">lower_bound_cnv</code></td>
<td>
<p>numeric float to represent the lower cap for CNV scores</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_upper_bound_cnv">upper_bound_cnv</code></td>
<td>
<p>numeric float to represent the upper cap for CNV scores</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_hc_function_cnv">hc_function_cnv</code></td>
<td>
<p>character for which hierarchical clustering function to use for CNV-calling; to feed hclust's method argument, one of &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot;, &quot;mcquitty&quot;, &quot;median&quot; or &quot;centroid&quot;</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_hc_function_cnv_heatmap">hc_function_cnv_heatmap</code></td>
<td>
<p>character for which hierarchical clustering function to use for visualzing CNV heat map; to feed hclust's method argument, one of &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot;, &quot;mcquitty&quot;, &quot;median&quot; or &quot;centroid&quot;</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_quantile_plot_cluster_label">quantile_plot_cluster_label</code></td>
<td>
<p>character string of which column name to keep in quantile plot</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_hc_function_silhouette">hc_function_silhouette</code></td>
<td>
<p>character string for which hierarchical clustering function to use for
the Silhouette method; to feed hclust's method argument, one of &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot;, &quot;mcquitty&quot;, &quot;median&quot; or &quot;centroid&quot;</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_max_k_silhouette">max_k_silhouette</code></td>
<td>
<p>integer of number max number of clusters to evaluate (2:max_k_silhouette)
.       in Silhouette method</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_plot_silhouette">plot_silhouette</code></td>
<td>
<p>TRUE if plotting silhouette scores for clustering</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_hc_function_plot_clones">hc_function_plot_clones</code></td>
<td>
<p>character string for which hierarchical clustering function to use
in plotting clones</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_use_go_terms">use_GO_terms</code></td>
<td>
<p>TRUE if using enrichR to get Gene Ontology terms for SlideCNA-defined clusters</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_chrom_ord">chrom_ord</code></td>
<td>
<p>character vector of order and names of chromosomes</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_chrom_colors">chrom_colors</code></td>
<td>
<p>character vector of which colors each chromosome should be in heat map</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_text_size">text_size</code></td>
<td>
<p>integer of size of text in some ggplots</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_title_size">title_size</code></td>
<td>
<p>integer of size of title in some ggplots</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_legend_size_pt">legend_size_pt</code></td>
<td>
<p>integer of size of legend text size in some ggplots</p>
</td></tr>
<tr><td><code id="run_slide_cna_+3A_legend_height_bar">legend_height_bar</code></td>
<td>
<p>integer of height of legend bar in some ggplots</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

<hr>
<h2 id='scale_nUMI'>Scale for nUMI (UMI Count) to generate CNV scores</h2><span id='topic+scale_nUMI'></span>

<h3>Description</h3>

<p>This function re-scales expression intensities to be in a smaller range, normalizes for nUMI per bin,
and subtracts reference bead signal
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_nUMI(dat_bin, thresh_hard = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_nUMI_+3A_dat_bin">dat_bin</code></td>
<td>
<p>data.table of relative expression intensities per bin</p>
</td></tr>
<tr><td><code id="scale_nUMI_+3A_thresh_hard">thresh_hard</code></td>
<td>
<p>TRUE if using strict thresholds for expression thresholds and FALSE if adjusting
thresholds based on 1 + or - the mean of absolute min and max values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table of CNV scores per bin
</p>

<hr>
<h2 id='scalefit'>Subfunction for scale_nUMI that normalizes a given bin for UMI count and centers the  mean CNV score at 1</h2><span id='topic+scalefit'></span>

<h3>Description</h3>

<p>This function re-scales expression intensities to be in a smaller range, normalizes for nUMI per bin,
and centers the CNV scores to have a mean of 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scalefit(obj, nbin, start, end)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scalefit_+3A_obj">obj</code></td>
<td>
<p>data.table of relative expression intensities per bin</p>
</td></tr>
<tr><td><code id="scalefit_+3A_nbin">nbin</code></td>
<td>
<p>nUMIs in that specific bin</p>
</td></tr>
<tr><td><code id="scalefit_+3A_start">start</code></td>
<td>
<p>lower bound of CNV scores</p>
</td></tr>
<tr><td><code id="scalefit_+3A_end">end</code></td>
<td>
<p>upper bound of CNV scores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of adjusted CNV scores for that bins with nbin number of nUMIs within the range (inclusive) of start to end
</p>

<hr>
<h2 id='SpatialPlot'>Spatial plots of meta data</h2><span id='topic+SpatialPlot'></span>

<h3>Description</h3>

<p>This function will plot information about beads and bins on x and y coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SpatialPlot(
  dat_long,
  vars = NULL,
  text_size,
  title_size,
  legend_size_pt,
  legend_height_bar,
  plot_directory
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SpatialPlot_+3A_dat_long">dat_long</code></td>
<td>
<p>data.table of bead expression intensities per gene with metadata in long format</p>
</td></tr>
<tr><td><code id="SpatialPlot_+3A_vars">vars</code></td>
<td>
<p>character vector of features to plot/columns of metadata</p>
</td></tr>
<tr><td><code id="SpatialPlot_+3A_text_size">text_size</code></td>
<td>
<p>Ggplot2 text size</p>
</td></tr>
<tr><td><code id="SpatialPlot_+3A_title_size">title_size</code></td>
<td>
<p>Ggplot2 title size</p>
</td></tr>
<tr><td><code id="SpatialPlot_+3A_legend_size_pt">legend_size_pt</code></td>
<td>
<p>Ggplot2 legend_size_pt</p>
</td></tr>
<tr><td><code id="SpatialPlot_+3A_legend_height_bar">legend_height_bar</code></td>
<td>
<p>Ggplot2 legend_height_bar</p>
</td></tr>
<tr><td><code id="SpatialPlot_+3A_plot_directory">plot_directory</code></td>
<td>
<p>output plot directory path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

<hr>
<h2 id='weight_rollmean'>Expressional smoothing along a chromosome using a weighted pyramidal moving average</h2><span id='topic+weight_rollmean'></span>

<h3>Description</h3>

<p>Take in a data.table of genomic positions and bead normalized/modified counts and apply pyramidal weighting
with a window size k to create smoothed expression intensities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weight_rollmean(dat, k = 101)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="weight_rollmean_+3A_dat">dat</code></td>
<td>
<p>data.table of normalized/adjusted counts</p>
</td></tr>
<tr><td><code id="weight_rollmean_+3A_k">k</code></td>
<td>
<p>size of window for weighting</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table of expression intensities
</p>

<hr>
<h2 id='weight_rollmean_sub'>Subfunction of weight_rollmean</h2><span id='topic+weight_rollmean_sub'></span>

<h3>Description</h3>

<p>Take in a counts matrix and apply pyramidal weighting
with a window size k to create smoothed expression intensities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weight_rollmean_sub(mat, k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="weight_rollmean_sub_+3A_mat">mat</code></td>
<td>
<p>matrix of normalized/adjusted counts</p>
</td></tr>
<tr><td><code id="weight_rollmean_sub_+3A_k">k</code></td>
<td>
<p>size of window for weighting</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of smoothed counts
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
