<!DOCTYPE html><html lang="en-GB"><head><title>Help for package smartsnp</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {smartsnp}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#read_packedancestrymap'><p>Read Files in PACKEDANCESTRYMAP format</p></a></li>
<li><a href='#smart_mva'><p>Smart Multivariate Analyses (wrapper of PCA, PERMANOVA and PERMDISP)</p></a></li>
<li><a href='#smart_pca'><p>Smart Principal Component Analysis</p></a></li>
<li><a href='#smart_permanova'><p>Smart Permutational Multivariate Analysis of Variance</p></a></li>
<li><a href='#smart_permdisp'><p>Smart Permutational Multivariate Analysis of Dispersion</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Fast Multivariate Analyses of Big Genomic Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-03-01</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Christian Huber &lt;christian.domitian.huber@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fast computation of multivariate analyses of small (10s to 100s markers) to big (1000s to 100000s) genotype data. Runs Principal Component Analysis allowing for centering, z-score standardization and scaling for genetic drift, projection of ancient samples to modern genetic space and multivariate tests for differences in group location (Permutation-Based Multivariate Analysis of Variance) and dispersion (Permutation-Based Multivariate Analysis of Dispersion).</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>bootSVD, data.table, foreach, Rfast, RSpectra, vegan, vroom,
Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-03-03 12:11:40 UTC; christianhuber</td>
</tr>
<tr>
<td>Author:</td>
<td>Salvador Herrando-Perez
    <a href="https://orcid.org/0000-0001-6052-6854"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Ray Tobler <a href="https://orcid.org/0000-0002-4603-1473"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Christian Huber <a href="https://orcid.org/0000-0002-2267-2604"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-03-04 09:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='read_packedancestrymap'>Read Files in PACKEDANCESTRYMAP format</h2><span id='topic+read_packedancestrymap'></span>

<h3>Description</h3>

<p>This function loads genotype data in <code>PACKEDANCESTRYMAP</code> format (binary or compressed).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_packedancestrymap(pref)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_packedancestrymap_+3A_pref">pref</code></td>
<td>
<p>The prefix of the file name that contains the genotype data (i.e., without the <code>*.geno</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list containing a single element:
</p>

<ul>
<li> <p><code>geno</code> Genotype data as R matrix.<br />
</p>
</li></ul>


<hr>
<h2 id='smart_mva'>Smart Multivariate Analyses (wrapper of PCA, PERMANOVA and PERMDISP)</h2><span id='topic+smart_mva'></span>

<h3>Description</h3>

<p>Computes Principal Component Analysis (PCA) for variable x sample genotype data, such as Single Nucleotide Polymorphisms (SNP), in combination with Permutational Multivariate Analysis of Variance (PERMANOVA) and Permutational Multivariate Analysis of Dispersion (PERMDISP).
A wrapper of functions <code>smart_pca</code>, <code>smart_permanova</code> and <code>smart_permdisp</code>.
Genetic markers such as SNPs can be scaled by <code>centering</code>, z-scores and genetic drift-based dispersion.
The latter follows the SMARTPCA implementation of Patterson, Price and Reich (2006).
Optimized to run fast computation for big datasets.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smart_mva_+3A_snp_data">snp_data</code></td>
<td>
<p>snp_data</p>
</td></tr></table>
<p>File name read from working directory.
SNP = rows, samples = columns without row names or column headings.
SNP values must be count data (no decimals allowed).
File extension detected automatically whether text or <code>EIGENSTRAT</code>.
See details.
</p>
<table role = "presentation">
<tr><td><code id="smart_mva_+3A_packed_data">packed_data</code></td>
<td>
<p>Logical value for <code>EIGENSTRAT</code>, irrelevant for text data.
Default <code>packed_data = FALSE</code> assumes uncompressed <code>EIGENSTRAT</code>.
<code>packed_data = TRUE</code> for compressed or binary <code>EIGENSTRAT</code> (<code>PACKENDANCESTRYMAP</code>).</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_sample_group">sample_group</code></td>
<td>
<p>Character or numeric vector assigning samples to groups.
Coerced to factor.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_sample_remove">sample_remove</code></td>
<td>
<p>Logical <code>FALSE</code> or numeric vector indicating column numbers (samples) to be removed from computations.
Default <code>sample_remove =  FALSE</code> keeps all samples.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_snp_remove">snp_remove</code></td>
<td>
<p>Logical <code>FALSE</code> or numeric vector indicating row numbers (SNPs) to be removed from computations.
Default <code>snp_remove =  FALSE</code> keeps all SNPs.
See details.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_pca">pca</code></td>
<td>
<p>Logical indicating if PCA is computed.
Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_permanova">permanova</code></td>
<td>
<p>Logical indicating if PERMANOVA is computed.
Default <code>TRUE</code></p>
</td></tr>
<tr><td><code id="smart_mva_+3A_permdisp">permdisp</code></td>
<td>
<p>Logical indicating if PERMDISP is computed.
Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_missing_value">missing_value</code></td>
<td>
<p>Number <code>9</code> or string <code>NA</code> indicating missing value.
Default <code>missing_value = 9</code> as in <code>EIGENSTRAT</code>.
If no missing values present, no effect on computation.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_missing_impute">missing_impute</code></td>
<td>
<p>String handling missing values.
Default <code>missing_impute = "mean"</code> replaces missing values of each SNP by mean of non-missing values across samples.
<code>missing_impute = "remove"</code> removes SNPs with at least one missing value.
If no missing values present, no effect on computation.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_scaling">scaling</code></td>
<td>
<p>String. Default <code>scaling = "drift"</code> scales SNPs to control for expected allele frequency dispersion caused by genetic drift (SMARTPCA).
<code>scaling = "center"</code> for <code>centering</code> (covariance-based PCA).
<code>scaling = "sd"</code> for <code>centered</code> SNPs divided by standard deviation (correlation-based PCA).
<code>scaling = "none"</code> for no scaling.
See details.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_program_svd">program_svd</code></td>
<td>
<p>String indicating R package computing single value decomposition (SVD).
Default <code>program_svd = "Rspectra"</code> for <code><a href="RSpectra.html#topic+svds">svds</a></code>.
<code>program_svd = "bootSVD"</code> for <code><a href="bootSVD.html#topic+fastSVD">fastSVD</a></code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_sample_project">sample_project</code></td>
<td>
<p>Numeric vector indicating column numbers (ancient samples) projected onto (modern) PCA space.
Default <code>sample_project =  FALSE</code> implements no projection.
See details.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_pc_project">pc_project</code></td>
<td>
<p>Numeric vector indicating the ranks of the PCA axes ancient samples are projected onto. Default <code>pc_ancient = c(1, 2)</code> for PCA axes 1 and 2.
If <code>program_svd = "RSpectra"</code>, <code>length(pc_ancient)</code> must be smaller than or equal to <code>pc_axes</code>.
No effect on computation, if no ancient samples present.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_sample_distance">sample_distance</code></td>
<td>
<p>Type of inter-sample proximity computed (distance, similarity, dissimilarity).
Default is <code>Euclidean distance</code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_program_distance">program_distance</code></td>
<td>
<p>A string value indicating R package to estimate proximities between pairs of samples.
Default <code>program_distance = "Rfast"</code> uses function <code><a href="Rfast.html#topic+Dist">Dist</a></code>; <code>program_distance = "vegan"</code> uses <code><a href="vegan.html#topic+vegdist">vegdist</a></code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_target_space">target_space</code></td>
<td>
<p>String.
Default <code>target_space = "multidimensional"</code> applies PERMANOVA and/or PERMDISP to sample-by-sample triangular matrix computed from variable-by-sample data, <code>pc_axes</code> has no effect on computation. <code>target_space = "pca"</code> applies PERMANOVA and/or PERMDISP to sample-by-sample data in PCA space, <code>pc_axes</code> determines number of PCA axes for testing.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_pc_axes">pc_axes</code></td>
<td>
<p>Number of PCA axes computed always starting with PCA axis 1.
Default <code>pc_axes = 2</code> computes PCA axes 1 and 2 if <code>target_space = "pca"</code>.
No effect on computation if <code>target_space = "multidimensional"</code>.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_pairwise">pairwise</code></td>
<td>
<p>Logical.
Default <code>pairwise = FALSE</code> computes global test. <code>pairwise = TRUE</code> computes global and pairwise tests.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_pairwise_method">pairwise_method</code></td>
<td>
<p>String specifying type of correction for multiple testing.
Default <code>"holm"</code>.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_permutation_n">permutation_n</code></td>
<td>
<p>Number of permutations resulting in PERMANOVA/PERMDISP test <em>p value</em>.
Default <code>9999</code>.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_permutation_seed">permutation_seed</code></td>
<td>
<p>Number fixing random generator of permutations.
Default <code>1</code>.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_dispersion_type">dispersion_type</code></td>
<td>
<p>String indicating quantification of group dispersion whether relative to spatial <code>"median"</code> or <code>"centroid"</code> in PERMDISP.
Default <code>"median"</code>.</p>
</td></tr>
<tr><td><code id="smart_mva_+3A_samplesize_bias">samplesize_bias</code></td>
<td>
<p>Logical. <code>samplesize_bias = TRUE</code> for dispersion weighted by number of samples per group in PERMDISP.
Default <code>pairwise = FALSE</code> for no weighting.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See details in other functions for conceptualization of PCA (<code>smart_pca</code>) (Hotelling 1993), SMARTPCA (Patterson, Price and Reich 2006), PERMANOVA (<code>smart_permanova</code>) (Anderson 2001) and PERMDISP (<code>smart_permdisp</code> (Anderson 2006), types of scaling, ancient projection, and correction for multiple testing.<br />
</p>
<p>Users can compute any combination of the three analyses by assigning <code>TRUE</code> or <code>FALSE</code> to <code>pca</code> and/or <code>permanova</code> and/or <code>permdisp</code>.<br />
</p>
<p>PERMANOVA and PERMDISP exclude samples (columns) specified in either <code>sample_remove</code> or <code>sample_project</code>.
Projected samples are not used for testing as their PCA coordinates are derived from, and therefore depend on, the coordinates of non-projected samples.<br />
</p>
<p>Data read from working directory with SNPs as rows and samples as columns. Two alternative formats: (1) text file of SNPs by samples (file extension and column separators recognized automatically) read using <code><a href="data.table.html#topic+fread">fread</a></code>; or (2) duet of <code>EIGENSTRAT</code> files (see <a href="https://reich.hms.harvard.edu/software">https://reich.hms.harvard.edu/software</a>) using <code><a href="vroom.html#topic+vroom_fwf">vroom_fwf</a></code>, including a genotype file of SNPs by samples (<code>*.geno</code>), and a sample file (<code>*.ind</code>) containing three vectors assigning individual samples to unique user-predefined groups (populations), sexes (or other user-defined descriptor) and alphanumeric identifiers.
For <code>EIGENSTRAT</code>, vector <code>sample_group</code> assigns samples to groups retrievable from column 3 of file <code>*.ind</code>.
SNPs with zero variance removed prior to SVD to optimize computation time and avoid undefined values if <code>scaling = "sd"</code> or <code>"drift"</code>.<br />
</p>
<p>Users can select subsets of samples or SNPs by introducing a vector including column numbers for samples (<code>sample_remove</code>) and/or row numbers for SNPs (<code>snp_remove</code>) to be removed from computations.
Function stops if the final number of SNPs is 1 or 2.
<code>EIGENSOFT</code> was conceived for the analysis of human genes and its SMARTPCA suite so accepts 22 (autosomal) chromosomes by default.
If &gt;22 chromosomes are provided and the internal parameter <code>numchrom</code> is not set to the target number chromosomes of interest, SMARTPCA automatically subsets chromosomes 1 to 22.
In contrast, <code>smart_mva</code> accepts any number of autosomes with or without the sex chromosomes from an <code>EIGENSTRAT</code> file.<br />
</p>


<h3>Value</h3>

<p>Returns a list containing the following elements:
</p>

<ul>
<li><p>pca.snp_loadingsDataframe of principal coefficients of SNPs.
One set of coefficients per PCA axis computed.
</p>
</li>
<li><p>pca.eigenvaluesDataframe of eigenvalues, variance and cumulative variance explained.
One eigenvalue per PCA axis computed.
</p>
</li>
<li><p>pca_sample_coordinatesDataframe showing PCA sample summary. Column <em>Group</em> assigns samples to groups. Column <em>Class</em> specifies if samples &quot;Removed&quot; from PCA or &quot;Projected&quot; onto PCA space.
Sequence of additional columns shows principal components (coordinates) of samples in PCA space (1 column per PCA computed named PC1, PC2, ...).
</p>
</li>
<li><p>test_samplesDataframe showing test sample summary.
Column <em>Group</em> assigns samples to tested groups.
Column <em>Class</em> specifies if samples were used in, or removed from, testing (PERMANOVA and/or PERMDISP).
Column <em>Sample_dispersion</em> shows dispersion of individual samples relative to spatial <code>"median"</code> or <code>"centroid"</code> used in PERMDISP.
</p>
</li>
<li><p>permanova.global_testList showing PERMANOVA table with degrees of freedom, sum of squares, mean sum of squares, <em>F</em> statistic, variance explained (<em>R2</em>) and <em>p</em> value.
</p>
</li>
<li><p>permanova.pairwise_testList showing PERMANOVA table with <em>F</em> statistic, variance explained (<em>R2</em>), <em>p</em> value and corrected <em>p</em> value per pair of groups.
</p>
</li>
<li><p>permdisp.global_testList showing PERMDISP table with degrees of freedoms, sum of squares, mean sum of squares, <em>F</em> statistic and <em>p</em> value.
</p>
</li>
<li><p>permdisp.pairwise_testList showing PERMDISP table with <em>F</em> statistic, <em>p</em> value and corrected <em>p</em> value per pair of groups.
Obtained only if <code>pairwise = TRUE</code>.
</p>
</li>
<li><p>permdisp.biasString indicating if PERMDISP dispersion corrected for number of samples per group.
</p>
</li>
<li><p>permdisp.group_locationDataframe showing coordinates of spatial <code>"median"</code> or <code>"centroid"</code> per group in PERMDISP.
</p>
</li>
<li><p>test.pairwise_correctionString indicating type of correction for multiple testing in PERMANOVA and/or PERMDISP.
</p>
</li>
<li><p>test.permutation_numberNumber of permutations applied to obtain the distribution of <em>F</em> statistic of PERMANOVA and/or PERMDISP.
</p>
</li>
<li><p>test.permutation_seedNumber fixing random generator of permutations of PERMANOVA and/or PERMDISP for reproducibility of results.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+smart_pca">smart_pca</a></code>,
<code><a href="#topic+smart_permanova">smart_permanova</a></code>,
<code><a href="#topic+smart_permdisp">smart_permdisp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Path to example genotype matrix "dataSNP"
pathToGenoFile = system.file("extdata", "dataSNP", package = "smartsnp")

# Assign 50 samples to each of two groups and colors
my_groups &lt;- as.factor(c(rep("A", 50), rep("B", 50))); cols = c("red", "blue")

# Run PCA, PERMANOVA and PERMDISP
mvaR &lt;- smart_mva(snp_data = pathToGenoFile, sample_group = my_groups)
mvaR$pca$pca.eigenvalues # extract PCA eigenvalues
mvaR$pca$pca.snp_loadings # extract principal coefficients (SNP loadings)
mvaR$pca$pca.sample_coordinates # extract PCA principal components (sample position in PCA space)

# plot PCA
plot(mvaR$pca$pca.sample_coordinates[,c("PC1","PC2")], cex = 2,
     pch = 19, col = cols[my_groups], main = "genotype smartpca")
legend("topleft", legend = levels(my_groups), cex = 1,
       pch = 19, col = cols, text.col = cols)

# Extract PERMANOVA table
mvaR$test$permanova.global_test

# Extract PERMDISP table
mvaR$test$permdisp.global_test # extract PERMDISP table

# Extract sample summary and dispersion of individual samples used in PERMDISP
mvaR$test$test_samples

</code></pre>

<hr>
<h2 id='smart_pca'>Smart Principal Component Analysis</h2><span id='topic+smart_pca'></span>

<h3>Description</h3>

<p>Compute Principal Component Analysis (PCA) for variable x sample genotype data including covariance (<code>centered</code>), correlation (z-score) and SMARTPCA scaling,
and implements projection of ancient samples onto modern PCA space. SMARTPCA scaling controls for genetic drift when variables are bi-allelic genetic markers
such as single nucleotide polymorphisms (SNP) following Patterson, Price and Reich (2006).
Optimized to run fast single value decomposition for big datasets.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smart_pca_+3A_snp_data">snp_data</code></td>
<td>
<p>File name read from working directory.
SNP = rows, samples = columns without row names or column headings.
SNP values must be count data (no decimals allowed). File extension detected automatically whether text or <code>EIGENSTRAT</code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_packed_data">packed_data</code></td>
<td>
<p>Logical value for <code>EIGENSTRAT</code>, irrelevant for text data.
Default <code>packed_data = FALSE</code> assumes uncompressed <code>EIGENSTRAT</code>.
<code>packed_data = TRUE</code> for compressed or binary <code>EIGENSTRAT</code> (<code>PACKENDANCESTRYMAP</code>).</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_sample_group">sample_group</code></td>
<td>
<p>Character or numeric vector assigning samples to groups.
Coerced to factor.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_sample_remove">sample_remove</code></td>
<td>
<p>Logical <code>FALSE</code> or numeric vector indicating column numbers (samples) to be removed from computations.
Default <code>sample_remove =  FALSE</code> keeps all samples.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_snp_remove">snp_remove</code></td>
<td>
<p>Logical <code>FALSE</code> or numeric vector indicating row numbers (SNPs) to be removed from computations.
Default <code>snp_remove =  FALSE</code> keeps all SNPs.
See details.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_missing_value">missing_value</code></td>
<td>
<p>Number <code>9</code> or string <code>NA</code> indicating missing value.
Default <code>missing_value = 9</code> as in <code>EIGENSTRAT</code>.
If no missing values present, no effect on computation.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_missing_impute">missing_impute</code></td>
<td>
<p>String handling missing values.
Default <code>missing_impute = "mean"</code> replaces missing values of each SNP by mean of non-missing values across samples.
<code>missing_impute = "remove"</code> removes SNPs with at least one missing value.
If no missing values present, no effect on computation.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_scaling">scaling</code></td>
<td>
<p>String. Default <code>scaling = "drift"</code> scales SNPs to control for expected allele frequency dispersion caused by genetic drift (SMARTPCA).
<code>scaling = "center"</code> for <code>centering</code> (covariance-based PCA).
<code>scaling = "sd"</code> for <code>centered</code> SNPs divided by standard deviation (correlation-based PCA).
<code>scaling = "none"</code> for no scaling.
See details.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_program_svd">program_svd</code></td>
<td>
<p>String indicating R package computing single value decomposition (SVD).
Default <code>program_svd = "Rspectra"</code> for <code><a href="RSpectra.html#topic+svds">svds</a></code>.
<code>program_svd = "bootSVD"</code> for <code><a href="bootSVD.html#topic+fastSVD">fastSVD</a></code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_pc_axes">pc_axes</code></td>
<td>
<p>A numeric value.
If <code>program_svd = "Rspectra"</code> this argument indicates number of PCA axes computed starting with PCA axis 1.
Default <code>pc_axes = 2</code> computes PCA axes 1 and 2.
No effect on computation if <code>program_svd = "bootSVD"</code> since all PCA axes are computed.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_sample_project">sample_project</code></td>
<td>
<p>Numeric vector indicating column numbers (ancient samples) projected onto (modern) PCA space.
Default <code>sample_project =  FALSE</code> indicates no samples will be used for projection.
See details.</p>
</td></tr>
<tr><td><code id="smart_pca_+3A_pc_project">pc_project</code></td>
<td>
<p>Numeric vector indicating the ranks of the PCA axes ancient samples are projected onto.
Default <code>pc_ancient = c(1, 2)</code> for PCA axes 1 and 2. If <code>program_svd = "RSpectra"</code>, <code>length(pc_ancient)</code> must be smaller than or equal to <code>pc_axes</code>.
No effect on computation, if no ancient samples present.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PCA is a rigid rotation of a Cartesian coordinate system (samples = points, axes = variables or SNPs) that maximizes the dispersion of points along a new system of axes (Pearson 1901; Hotelling 1933; Jolliffe 2002).
In rotated space (ordination), axes are <code>principal axes</code> (PCA axes), <code>eigenvalues</code> measure variance explained, and <code>principal coefficients</code> measure importance of SNPs (eigenvectors), <code>principal components</code> are coordinates of samples (i.e., linear combinations of scaled variables weighted by eigenvectors).
Principal coefficients are direction cosines between original and PCA axes (Legendre &amp; Legendre 2012). PCA can be computed by <code>eigenanalysis</code> or, as implemented here, single value decomposition (SVD). <br />
</p>
<p>SNPs can be scaled in four different ways prior to SVD: (1) no scaling; (2) covariance: SNPs <code>centered</code> such that <em>M(i,j)</em> = <em>C(i,j)</em> minus <em>mean(j)</em>) where <em>C(i,j)</em> is the number of variant alleles for SNP <em>j</em> and sample <em>i</em>, and <em>M(i,j)</em> is the <code>centered</code> value of each data point; (3) correlation (z-scores): SNPs <code>centered</code> then divided by standard deviation <em>sd(j)</em>, (4) SMARTPCA: SNPs <code>centered</code> then divided by <em>sqrt(p(j)(1-p(j)))</em>, where <em>p(j)</em> equals <em>mean(j)</em> divided by <em>2</em>, quantifies the underlying allele frequency (autosomal chromosomes) and conceptualizes that SNP frequency changes at rate proportional to <em>sqrt(p(j)(1-p(j)))</em> per generation due to genetic drift (Patterson, Price and Reich 2006).
SMARTPCA standardization results in all SNPs that comply with Hardy-Weinberg equilibrium having identical variance.
SMARTPCA (Patterson, Price and Reich 2006) and <code>EIGENSTRAT</code> (Price, Patterson, Plenge, Weinblatt, Shadick and Reich 2006) are the computing suites of software <code>EIGENSOFT</code> (<a href="https://reich.hms.harvard.edu/software">https://reich.hms.harvard.edu/software</a>).<br />
</p>
<p><code><a href="RSpectra.html#topic+svds">svds</a></code> runs single value decomposition much faster than <code><a href="bootSVD.html#topic+fastSVD">fastSVD</a></code>. With <code><a href="RSpectra.html#topic+svds">svds</a></code>, <code>pc_axes</code> indicates number of eigenvalues and eigenvectors computed starting from PCA axis 1. <code><a href="bootSVD.html#topic+fastSVD">fastSVD</a></code> computes all eigenvalues and eigenvectors. Eigenvalues calculated from singular values divided by number of samples minus 1. If number of samples equals number of SNPS, <code><a href="bootSVD.html#topic+fastSVD">fastSVD</a></code> prints message alert that no computing efficiency is achieved for square matrices.<br />
</p>
<p>Ancient samples (with many missing values) can be projected onto modern PCA space derived from modern samples.
Following Nelson Taylor and MacGregor (1996), the projected coordinates of a given ancient sample equal the slope coefficient of linear fit through the origin of (scaled) non-missing SNP values of that sample (response) versus principal coefficients of same SNPs in modern samples.
Number of projected coordinates per ancient sample given by <code>length(pc_ancient)</code>.
With <code><a href="RSpectra.html#topic+svds">svds</a></code>, <code>pc_axes</code> must be larger or equal to <code>length(pc_ancient)</code>.<br />
</p>
<p>Data read from working directory with SNPs as rows and samples as columns.
Two alternative formats: (1) text file of SNPs by samples (file extension and column separators recognized automatically) read using <code><a href="data.table.html#topic+fread">fread</a></code>; or (2) duet of <code>EIGENSTRAT</code> files (see <a href="https://reich.hms.harvard.edu/software">https://reich.hms.harvard.edu/software</a>) using <code><a href="vroom.html#topic+vroom_fwf">vroom_fwf</a></code>, including a genotype file of SNPs by samples (<code>*.geno</code>), and a sample file (<code>*.ind</code>) containing three vectors assigning individual samples to unique user-predefined groups (populations), sexes (or other user-defined descriptor) and alphanumeric identifiers.
For <code>EIGENSTRAT</code>, vector <code>sample_group</code> assigns samples to groups retrievable from column of file <code>*.ind</code>. SNPs with zero variance removed prior to SVD to optimize computation time and avoid undefined values if <code>scaling = "sd"</code> or <code>"drift"</code>.<br />
</p>
<p>Users can select subsets of samples or SNPs by introducing a vector including column numbers for samples (<code>sample_remove</code>) and/or row numbers for SNPs (<code>snp_remove</code>) to be removed from computations.
Function stops if the final number of SNPs is 1 or 2.
<code>EIGENSOFT</code> was conceived for the analysis of human genes and its SMARTPCA suite so accepts 22 (autosomal) chromosomes by default.
If &gt;22 chromosomes are provided and the internal parameter <code>numchrom</code> is not set to the target number chromosomes of interest, SMARTPCA automatically subsets chromosomes 1 to 22.
In contrast, <code>smart_pca</code> accepts any number of autosomes with or without the sex chromosomes from an <code>EIGENSTRAT</code> file.<br />
</p>


<h3>Value</h3>

<p>Returns a list containing the following elements:
</p>

<ul>
<li> <p><code>pca.snp_loadings</code> Dataframe of principal coefficients of SNPs. One set of coefficients per PCA axis computed.<br />
</p>
</li>
<li> <p><code>pca.eigenvalues</code> Dataframe of eigenvalues, variance and cumulative variance explained. One eigenvalue per PCA axis computed.<br />
</p>
</li>
<li> <p><code>pca_sample_coordinates</code> Dataframe showing PCA sample summary.
Column <em>Group</em> assigns samples to groups.
Column <em>Class</em> specifies if samples &quot;Removed&quot; from PCA or &quot;Projected&quot; onto PCA space.
Sequence of additional columns shows principal components (coordinates) of samples in PCA space (1 column per PCA computed named PC1, PC2, ...).
</p>
</li></ul>



<h3>References</h3>

<p>Hotelling, H. (1933) Analysis of a complex of statistical variables into principal components. Journal of Educational Psychology, 24, 417-441.<br />
</p>
<p>Jolliffe, I.T. (2002) Principal Component Analysis (Springer, New York, USA).<br />
</p>
<p>Legendre, P. &amp; L. F. J. Legendre (2012). Numerical ecology.  Developments in environmental modelling (Elsevier, Oxford, UK).<br />
</p>
<p>Nelson, P.R.C., P.A. Taylor, and J.F. MacGregor (1996) Missing data methods in PCA and PLS: score calculations with incomplete observations. Chemometrics and Intelligent Laboratory Systems, 35, 45-65.<br />
</p>
<p>Patterson, N.J., A. L. Price and D. Reich (2006) Population structure and eigenanalysis. PLoS Genetics, 2, e190.<br />
</p>
<p>Pearson, K. (1901) On lines and planes of closest fit to systems of points in space. Philosophical Magazine, 2, 559-572.<br />
</p>
<p>Price, A.L., N.J. Patterson, R.M. Plenge, M.E. Weinblatt, N.A. Shadick and David Reich (2006). Principal components analysis corrects for stratification in genome-wide association studies. Nature Genetics, 38, 904-909.
</p>


<h3>See Also</h3>

<p><code><a href="bootSVD.html#topic+fastSVD">fastSVD</a></code> (package <b>bootSVD</b>),
<code><a href="foreach.html#topic+foreach">foreach</a></code> (package <b>foreach</b>),
<code><a href="data.table.html#topic+fread">fread</a></code> (package <b>data.table</b>),
<code><a href="Rfast.html#topic+rowVars">rowVars</a></code> (package <b>Rfast</b>),
<code><a href="RSpectra.html#topic+svds">svds</a></code> (package <b>RSpectra</b>),
<code><a href="vroom.html#topic+vroom_fwf">vroom_fwf</a></code> (package <b>vroom</b>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Path to example genotype matrix "dataSNP"
pathToGenoFile = system.file("extdata", "dataSNP", package = "smartsnp")

# Example 1: modern samples
#assign 50 samples to each of two groups and colors
my_groups &lt;- c(rep("A", 50), rep("B", 50)); cols = c("red", "blue")
#run PCA with truncated SVD (PCA 1 x PCA 2)
pcaR1 &lt;- smart_pca(snp_data = pathToGenoFile, sample_group = my_groups)
pcaR1$pca.eigenvalues # extract eigenvalues
pcaR1$pca.snp_loadings # extract principal coefficients (SNP loadings)
pcaR1$pca.sample_coordinates # extract principal components (sample position in PCA space)
#plot PCA
plot(pcaR1$pca.sample_coordinates[,c("PC1","PC2")], cex = 2,
     pch = 19, col = cols[as.factor(my_groups)], main = "genotype smartpca")
legend("topleft", legend = levels(as.factor(my_groups)), cex =1,
     pch = 19, col = cols, text.col = cols)

# Example 2: modern and ancient samples (ancient samples projected onto modern PCA space)
#assign samples 1st to 10th per group to ancient
my_ancient &lt;- c(1:10, 51:60)
#run PCA with truncated SVD (PCA 1 x PCA 2)
pcaR2 &lt;- smart_pca(snp_data = pathToGenoFile, sample_group = my_groups, sample_project = my_ancient)
pcaR2$pca.eigenvalues # extract eigenvalues
pcaR2$pca.snp_loadings # extract principal coefficients (SNP loading)
pcaR2$pca.sample_coordinates # extract principal components (sample position in PCA space)
#assign samples to groups (A, ancient, B) and colors
my_groups[my_ancient] &lt;- "ancient"; cols = c("red", "black", "blue")
#plot PCA
plot(pcaR2$pca.sample_coordinates[,c("PC1","PC2")],
     cex = 2, col = cols[as.factor(my_groups)], pch = 19, main = "genotype smartpca")
legend("topleft", legend = levels(as.factor(my_groups)),  cex = 1,
     pch = 19, col = cols, text.col = cols)

</code></pre>

<hr>
<h2 id='smart_permanova'>Smart Permutational Multivariate Analysis of Variance</h2><span id='topic+smart_permanova'></span>

<h3>Description</h3>

<p>Computes Permutational Multivariate Analysis of Variance (PERMANOVA) for testing differences in group location using multivariate data. Variance partitioning computed on a sample-by-sample triangular matrix obtained from variable-by-sample data following Anderson (2001).
Calculates a range of inter-sample distances, similarities and dissimilarities.
Includes control for genetic drift for bi-allelic genetic markers such as single nucleotide polymorphisms (SNP) following Patterson, Price and Reich (2006) that can be combined with SMART Principal Component Analysis (PCA). Optimized to run fast matrix building and permutations for big datasets in ecological, evolutionary and genomic research.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smart_permanova_+3A_snp_data">snp_data</code></td>
<td>
<p>File name read from working directory.
SNP = rows, samples = columns without row names or column headings.
SNP values must be count data (no decimals allowed).
File extension detected automatically whether text or <code>EIGENSTRAT</code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_packed_data">packed_data</code></td>
<td>
<p>Logical value for <code>EIGENSTRAT</code>, irrelevant for text data.
Default <code>packed_data = FALSE</code> assumes uncompressed <code>EIGENSTRAT</code>.
<code>packed_data = TRUE</code> for compressed or binary <code>EIGENSTRAT</code> (<code>PACKENDANCESTRYMAP</code>).</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_sample_group">sample_group</code></td>
<td>
<p>Character or numeric vector assigning samples to groups. Coerced to factor.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_sample_remove">sample_remove</code></td>
<td>
<p>Logical <code>FALSE</code> or numeric vector indicating column numbers (samples) to be removed from computations.
Default <code>sample_remove =  FALSE</code> keeps all samples.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_snp_remove">snp_remove</code></td>
<td>
<p>Logical <code>FALSE</code> or numeric vector indicating row numbers (SNPs) to be removed from computations.
Default <code>snp_remove =  FALSE</code> keeps all SNPs. See details.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_missing_value">missing_value</code></td>
<td>
<p>Number <code>9</code> or string <code>NA</code> indicating missing value.
Default <code>missing_value = 9</code> as in <code>EIGENSTRAT</code>.
If no missing values present, no effect on computation.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_missing_impute">missing_impute</code></td>
<td>
<p>String handling missing values.
Default <code>missing_impute = "mean"</code> replaces missing values of each SNP by mean of non-missing values across samples.
<code>missing_impute = "remove"</code> removes SNPs with at least one missing value.
If no missing values present, no effect on computation.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_scaling">scaling</code></td>
<td>
<p>String. Default <code>scaling = "drift"</code> scales SNPs to control for expected allele frequency dispersion caused by genetic drift (SMARTPCA).
<code>scaling = "center"</code> for <code>centering</code> (covariance-based PCA).
<code>scaling = "sd"</code> for <code>centered</code> SNPs divided by standard deviation (correlation-based PCA).
<code>scaling = "none"</code> for no scaling.
See details.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_sample_distance">sample_distance</code></td>
<td>
<p>Type of inter-sample proximity computed (distance, similarity, dissimilarity).
Default is <code>Euclidean distance</code>. See details.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_program_distance">program_distance</code></td>
<td>
<p>A string value indicating R package to estimate proximities between pairs of samples.
Default <code>program_distance = "Rfast"</code> uses function <code><a href="Rfast.html#topic+Dist">Dist</a></code>; <code>program_distance = "vegan"</code> uses <code><a href="vegan.html#topic+vegdist">vegdist</a></code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_target_space">target_space</code></td>
<td>
<p>String.
Default <code>target_space = "multidimensional"</code> applies PERMANOVA to sample-by-sample triangular matrix computed from variable-by-sample data, <code>pc_axes</code> has no effect on computation.
<code>target_space = "pca"</code> applies PERMANOVA to sample-by-sample data in PCA space, <code>pc_axes</code> determines number of PCA axes for testing.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_pc_axes">pc_axes</code></td>
<td>
<p>Number of PCA axes computed always starting with PCA axis 1. Default <code>pc_axes = 2</code> computes PCA axes 1 and 2 if <code>target_space = "pca"</code>.
No effect on computation if <code>target_space = "multidimensional"</code>.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_pairwise">pairwise</code></td>
<td>
<p>Logical.
Default <code>pairwise = FALSE</code> computes global test.
<code>pairwise = TRUE</code> computes global and pairwise tests.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_pairwise_method">pairwise_method</code></td>
<td>
<p>String specifying type of correction for multiple testing.
Default <code>"holm"</code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_permutation_n">permutation_n</code></td>
<td>
<p>Number of permutations resulting in PERMANOVA test <em>p value</em>.
Default <code>9999</code>.</p>
</td></tr>
<tr><td><code id="smart_permanova_+3A_permutation_seed">permutation_seed</code></td>
<td>
<p>Number fixing random generator of permutations.
Default <code>1</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PERMANOVA is a form of linear modelling that partitions variation in a triangular matrix of inter-sample proximities obtained from variable-by-sample data.
Uses permutations to estimate the probability of observed group differences in SNP composition given a null hypothesis of no differences between groups (Anderson 2001).
Proximity between samples can be any type of distance, similarity or dissimilarity.
Original acronym <code>NPMANOVA</code> (Non-Parametric MANOVA) replaced with PERMANOVA (Anderson 2004, 2017).<br />
</p>
<p>Univariate ANOVA captures differences in mean and variance referred to as location and dispersion in PERMANOVA's multivariate context (Anderson &amp; Walsh 2013, Warton, Wright and Wang 2012).
To attribute group differences to location (position of sample groups) and/or dispersion (spread of sample groups), PERMANOVA must be combined with PERMDISP as implemented through <code>smart_permdisp</code>.<br />
</p>
<p>Function <code>smart_permanova</code> uses <code><a href="vegan.html#topic+adonis">adonis</a></code> to fit formula <code>snp_eucli ~ sample_group</code>, where <code>snp_eucli</code> is the sample-by-sample triangular matrix in Principal Coordinate Analysis (Gower 1966) space.
Current version restricted to one-way designs (one categorical predictor) though PERMANOVA can handle &gt;1 crossed and/or nested factors (Anderson 2001) and continuous predictors (McArdle &amp; Anderson 2001).
If &gt;2 sample groups tested, <code>pairwise = TRUE</code> allows pairwise testing and correction for multiple testing by <code>holm (Holm)</code> [default], <code>hochberg (Hochberg)</code>, <code>hommel (Hommel)</code>, <code>bonferroni (Bonferroni)</code>, <code>BY (Benjamini-Yekuieli)</code>, <code>BH (Benjamini-Hochberg)</code> or <code>fdr (False Discovery Rate)</code>.<br />
</p>
<p>For big data, <code><a href="Rfast.html#topic+Dist">Dist</a></code> builds sample-by-sample triangular matrix much faster than <code><a href="vegan.html#topic+vegdist">vegdist</a></code>.
<code><a href="Rfast.html#topic+Dist">Dist</a></code> computes proximities <code>euclidean</code>, <code>manhattan</code>, <code>canberra1</code>, <code>canberra2</code>, <code>minimum</code>, <code>maximum</code>, <code>minkowski</code>, <code>bhattacharyya</code>, <code>hellinger</code>, <code>kullback_leibler</code> and <code>jensen_shannon</code>. <code><a href="vegan.html#topic+vegdist">vegdist</a></code> computes <code>manhattan</code>, <code>euclidean</code>, <code>canberra</code>, <code>clark</code>, <code>bray</code>, <code>kulczynski</code>, <code>jaccard</code>, <code>gower</code>, <code>altGower</code>, <code>morisita</code>, <code>horn</code>, <code>mountford</code>, <code>raup</code>, <code>binomial</code>, <code>chao</code>, <code>cao</code> and <code>mahalanobis</code>.
Euclidean distance required for SMARTPCA scaling.<br />
</p>
<p><code>sample_remove</code> should include both samples removed from PCA and ancient samples projected onto PCA space (if any).<br />
</p>
<p>Data read from working directory with SNPs as rows and samples as columns.
Two alternative formats: (1) text file of SNPs by samples (file extension and column separators recognized automatically) read using <code><a href="data.table.html#topic+fread">fread</a></code>; or (2) duet of <code>EIGENSTRAT</code> files (see <a href="https://reich.hms.harvard.edu/software">https://reich.hms.harvard.edu/software</a>) using <code><a href="vroom.html#topic+vroom_fwf">vroom_fwf</a></code>, including a genotype file of SNPs by samples (<code>*.geno</code>), and a sample file (<code>*.ind</code>) containing three vectors assigning individual samples to unique user-predefined groups (populations), sexes (or other user-defined descriptor) and alphanumeric identifiers.
For <code>EIGENSTRAT</code>, vector <code>sample_group</code> assigns samples to groups retrievable from column 3 of file <code>*.ind</code>.
SNPs with zero variance removed prior to SVD to optimize computation time and avoid undefined values if <code>scaling = "sd"</code> or <code>"drift"</code>.<br />
</p>
<p>Users can select subsets of samples or SNPs by introducing a vector including column numbers for samples (<code>sample_remove</code>) and/or row numbers for SNPs (<code>snp_remove</code>) to be removed from computations.
Function stops if the final number of SNPs is 1 or 2.
<code>EIGENSOFT</code> was conceived for the analysis of human genes and its SMARTPCA suite so accepts 22 (autosomal) chromosomes by default.
If &gt;22 chromosomes are provided and the internal parameter <code>numchrom</code> is not set to the target number chromosomes of interest, SMARTPCA automatically subsets chromosomes 1 to 22.
In contrast, <code>smart_permanova</code> accepts any number of autosomes with or without the sex chromosomes from an <code>EIGENSTRAT</code> file.<br />
</p>


<h3>Value</h3>

<p>Returns a list containing the following elements:
</p>

<ul>
<li><p>permanova.samplesDataframe showing sample summary.
Column <em>Group</em> assigns samples to tested groups.
Column <em>Class</em> specifies if samples were used in, or removed from, testing.
</p>
</li>
<li><p>permanova.global_testList showing table with degrees of freedom, sum of squares, mean sum of squares, <em>F</em> statistic, variance explained (<em>R2</em>) and <em>p</em> value.
</p>
</li>
<li><p>permanova.pairwise_testList showing table <em>F</em> statistic, variance explained (<em>R2</em>), <em>p</em> value and corrected <em>p</em> value per pair of groups.
Obtained only if <code>pairwise = TRUE</code>.
</p>
</li>
<li><p>permanova.pairwise_correctionString indicating type of correction for multiple testing.
</p>
</li>
<li><p>permanova.permutation_numberNumber of permutations applied to obtain the distribution of <em>p value</em>.
</p>
</li>
<li><p>permanova.permutation_seedNumber fixing random generator of permutations for reproducibility of results.
</p>
</li></ul>



<h3>References</h3>

<p>Anderson, M. J. (2001) A new method for non-parametric multivariate analysis of variance. Austral Ecology, 26, 32-46.<br />
Anderson, M. J. (2004). PERMANOVA_2factor: a FORTRAN computer program for permutational multivariate analysis of variance (for any two-factor ANOVA design) using permutation tests  (Department of Statistics, University of Auckland, New Zealand).<br />
Anderson, M. J. &amp; D. C. I. Walsh (2013) PERMANOVA, ANOSIM, and the Mantel test in the face of heterogeneous dispersions: What null hypothesis are you testing? Ecological Monographs, 83, 557-574.<br />
Gower, J. C. (1966) Some distance properties of latent root and vector methods used in multivariate analysis. Biometrika, 53, 325-338.<br />
McArdle, B. H. &amp; M. J. Anderson (2001) Fitting multivariate models to community data: a comment on distance-based redundancy analysis. Ecology, 82, 290-297.<br />
Patterson, N., A. L. Price and D. Reich (2006) Population structure and eigenanalysis. PLoS Genetics, 2, e190.<br />
Warton, D. I., S. T. Wright and Y. Wang (2012) Distance-based multivariate analyses confound location and dispersion effects. Methods in Ecology and Evolution, 3, 89-101.
</p>


<h3>See Also</h3>

<p><code><a href="vegan.html#topic+adonis">adonis</a></code> (package <b>vegan</b>),
<code><a href="Rfast.html#topic+Dist">Dist</a></code> (package <b>Rfast</b>),
<code><a href="data.table.html#topic+fread">fread</a></code> (package <b>data.table</b>),
<code><a href="vegan.html#topic+vegdist">vegdist</a></code> (package <b>vegan</b>),
<code><a href="vroom.html#topic+vroom_fwf">vroom_fwf</a></code> (package <b>vroom</b>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Path to example genotype matrix "dataSNP"
pathToGenoFile = system.file("extdata", "dataSNP", package = "smartsnp")

# Assign 50 samples to each of two groups
my_groups &lt;- as.factor(c(rep("A", 50), rep("B", 50)))

# Run PERMANOVA
permanovaR &lt;- smart_permanova(snp_data = pathToGenoFile, sample_group = my_groups)

# Extract summary table assigning samples to groups
permanovaR$permanova.samples

# Extract PERMANOVA table
permanovaR$permanova.global_test

# Plot means of squares per group
#run pca with truncated SVD (PCA 1 x PCA 2)
pcaR1 &lt;- smart_pca(snp_data = pathToGenoFile, sample_group = my_groups)
#compute Euclidean inter-sample distances in PCA space (triangular matrix)
snp_eucli &lt;- vegan::vegdist(pcaR1$pca.sample_coordinates[,c("PC1","PC2")], method = "euclidean")
#run PERMANOVA
permanova &lt;- vegan::adonis(formula = snp_eucli ~ my_groups, permutations = 9999)
#extract meanSqs (groups versus residuals)
meanSqs &lt;- as.matrix(t(permanova$aov.tab$MeanSqs[1:2]))
colnames(meanSqs) &lt;- c("Groups", "Residuals")
#two horizontal plots
oldpar &lt;- par(mfrow = c(2,1), oma = c(0,5,0.1,0.1), lwd = 2)
barplot(meanSqs, horiz = TRUE, main = "PERMANOVA mean of squares",
  cex.names = 2, cex.main = 2, col = c("grey40"))
#run ANOSIM
anosimD &lt;- vegan::anosim(snp_eucli, my_groups, permutations = 999)
#remove outputs for clean plotting
#anosimD[2] &lt;- ""; anosimD[5] &lt;- ""
par(mar = c(5, 0.1, 3.5, 0.1))
plot(anosimD, xlab = "", ylab = "distance/similarity ranks",
  main = "Inter-sample proximity ranks", cex.main =2, cex.axis = 2,
  col = c("cyan", "red", "blue"))
par(oldpar)

</code></pre>

<hr>
<h2 id='smart_permdisp'>Smart Permutational Multivariate Analysis of Dispersion</h2><span id='topic+smart_permdisp'></span>

<h3>Description</h3>

<p>Computes Permutational Multivariate Analysis of Dispersion (PERMDISP) in group dispersion using multivariate data.
Variance partitioning computed on a sample-by-sample triangular matrix obtained from variable-by-sample data following Anderson (2006).
Calculates a range of inter-sample distances, similarities and dissimilarities. Includes control for genetic drift for bi-allelic genetic markers including single nucleotide polymorphisms (SNP) following Patterson, Price and Reich (2006) that can be combined with SMART Principal Component Analysis (PCA).
Optimized to run fast matrix building and permutations for big datasets in ecological, evolutionary and genomic research.<br />
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smart_permdisp_+3A_snp_data">snp_data</code></td>
<td>
<p>File name read from working directory. SNP = rows, samples = columns without row names or column headings.
SNP values must be count data (no decimals allowed).
File extension detected automatically whether text or <code>EIGENSTRAT</code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_packed_data">packed_data</code></td>
<td>
<p>Logical value for <code>EIGENSTRAT</code>, irrelevant for text data.
Default <code>packed_data = FALSE</code> assumes uncompressed <code>EIGENSTRAT</code>.
<code>packed_data = TRUE</code> for compressed or binary <code>EIGENSTRAT</code> (<code>PACKENDANCESTRYMAP</code>).</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_sample_group">sample_group</code></td>
<td>
<p>Character or numeric vector assigning samples to groups. Coerced to factor.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_sample_remove">sample_remove</code></td>
<td>
<p>Logical <code>FALSE</code> or numeric vector indicating column numbers (samples) to be removed from computations.
Default <code>sample_remove =  FALSE</code> keeps all samples.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_snp_remove">snp_remove</code></td>
<td>
<p>Logical <code>FALSE</code> or numeric vector indicating row numbers (SNPs) to be removed from computations.
Default <code>snp_remove =  FALSE</code> keeps all SNPs.
See details.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_missing_value">missing_value</code></td>
<td>
<p>Number <code>9</code> or string <code>NA</code> indicating missing value.
Default <code>missing_value = 9</code> as in <code>EIGENSTRAT</code>.
If no missing values present, no effect on computation.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_missing_impute">missing_impute</code></td>
<td>
<p>String handling missing values. Default <code>missing_impute = "mean"</code> replaces missing values of each SNP by mean of non-missing values across samples.
<code>missing_impute = "remove"</code> removes SNPs with at least one missing value.
If no missing values present, no effect on computation.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_scaling">scaling</code></td>
<td>
<p>String. Default <code>scaling = "drift"</code> scales SNPs to control for expected allele frequency dispersion caused by genetic drift (SMARTPCA). <code>scaling = "center"</code> for <code>centering</code> (covariance-based PCA). <code>scaling = "sd"</code> for <code>centered</code> SNPs divided by standard deviation (correlation-based PCA).
<code>scaling = "none"</code> for no scaling.
See details.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_sample_distance">sample_distance</code></td>
<td>
<p>Type of inter-sample proximity computed (distance, similarity, dissimilarity).
Default is <code>Euclidean distance</code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_program_distance">program_distance</code></td>
<td>
<p>A string value indicating R package to estimate proximities between pairs of samples.
Default <code>program_distance = "Rfast"</code> uses function <code><a href="Rfast.html#topic+Dist">Dist</a></code>; <code>program_distance = "vegan"</code> uses <code><a href="vegan.html#topic+vegdist">vegdist</a></code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_target_space">target_space</code></td>
<td>
<p>String.
Default <code>target_space = "multidimensional"</code> applies PERMANOVA to sample-by-sample triangular matrix computed from variable-by-sample data, <code>pc_axes</code> has no effect on computation.
<code>target_space = "pca"</code> applies PERMANOVA to sample-by-sample data in PCA space, <code>pc_axes</code> determines number of PCA axes for testing.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_pc_axes">pc_axes</code></td>
<td>
<p>Number of PCA axes computed always starting with PCA axis 1.
Default <code>pc_axes = 2</code> computes PCA axes 1 and 2 if <code>target_space = "pca"</code>.
No effect on computation if <code>target_space = "multidimensional"</code>.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_pairwise">pairwise</code></td>
<td>
<p>Logical.
Default <code>pairwise = FALSE</code> computes global test.
<code>pairwise = TRUE</code> computes global and pairwise tests.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_pairwise_method">pairwise_method</code></td>
<td>
<p>String specifying type of correction for multiple testing.
Default <code>"holm"</code>.
See details.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_permutation_n">permutation_n</code></td>
<td>
<p>Number of permutations resulting in PERMDISP test <em>p value</em>.
Default <code>9999</code>.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_permutation_seed">permutation_seed</code></td>
<td>
<p>Number fixing random generator of permutations.
Default <code>1</code>.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_dispersion_type">dispersion_type</code></td>
<td>
<p>String indicating quantification of group dispersion whether relative to spatial <code>"median"</code> or <code>"centroid"</code>.
Default <code>"median"</code>. See details.</p>
</td></tr>
<tr><td><code id="smart_permdisp_+3A_samplesize_bias">samplesize_bias</code></td>
<td>
<p>Logical.
<code>samplesize_bias = TRUE</code> for dispersion weighted by number of samples per group.
Default <code>pairwise = FALSE</code> for no weighting.
See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PERMDISP is a form of <code>homoscedasticity</code> test analogous to univariate Levene's (1960) and, more closely, Brown &amp; Forsythe's (1974) tests. Applies PERMANOVA test (Anderson 2001) for differences in Euclidean dispersion among groups (Anderson 2006).
Proximity between samples can be any type of distance, similarity or dissimilarity.
Group dispersion estimated relative to group centroids (central point) or to spatial (geometric) medians (point minimizing distance to group samples) in Principal Coordinate Analysis (<code>PCoA</code>, Gower 1966) space.
Acronym PERMDISP originates from Marti Anderson's FORTRAN program (Anderson 2004).
</p>
<p>Control for unequal number of samples among groups optionally done by weighting sample distance to group spatial <em>median</em> or  <em>centroid</em> by <em>sqrt(n/(n-1))</em> (O'Neill &amp; Mathews 2000), and the null hypothesis then being tested changes from <em>d_1 = d_2  = ... = d_t</em> (balanced design) to  <em>((n_1-1)/n_1) x d_1 = ((n_2-1)/n_2) x d_2  = ... =  ((n_t-1)/n_t) x d_t</em> (unbalanced design) where <em>d</em> represents dispersion of groups <em>1</em> to <em>t</em>, and <em>n</em> represents number of samples per group.
To attribute group differences to location (position of sample groups) and/or dispersion (spread of sample groups), PERMDISP must be combined with PERMANOVA as implemented through <code>smart_permanova</code>.<br />
</p>
<p><code>smart_permdisp</code> uses <code><a href="vegan.html#topic+betadisper">betadisper</a></code> to estimate an ANOVA <em>F</em> statistic and group dispersions using formula <code>snp_eucli ~ sample_group</code>, where <code>snp_eucli</code> is the sample-by-sample triangular matrix in <code>PCoA</code> space.
If &gt;2 sample groups tested, <code>pairwise = TRUE</code> allows pairwise testing and correction for multiple testing by <code>holm (Holm)</code> [default], <code>hochberg (Hochberg)</code>, <code>hommel (Hommel)</code>, <code>bonferroni (Bonferroni)</code>, <code>BY (Benjamini-Yekuieli)</code>, <code>BH (Benjamini-Hochberg)</code> or <code>fdr (False Discovery Rate)</code>.<br />
</p>
<p>For big data, <code><a href="Rfast.html#topic+Dist">Dist</a></code> builds sample-by-sample triangular matrix much faster than <code><a href="vegan.html#topic+vegdist">vegdist</a></code>. <code><a href="Rfast.html#topic+Dist">Dist</a></code> computes proximities <code>euclidean</code>, <code>manhattan</code>, <code>canberra1</code>, <code>canberra2</code>, <code>minimum</code>, <code>maximum</code>, <code>minkowski</code>, <code>bhattacharyya</code>, <code>hellinger</code>, <code>kullback_leibler</code> and <code>jensen_shannon</code>. <code><a href="vegan.html#topic+vegdist">vegdist</a></code> computes <code>manhattan</code>, <code>euclidean</code>, <code>canberra</code>, <code>clark</code>, <code>bray</code>, <code>kulczynski</code>, <code>jaccard</code>, <code>gower</code>, <code>altGower</code>, <code>morisita</code>, <code>horn</code>, <code>mountford</code>, <code>raup</code>, <code>binomial</code>, <code>chao</code>, <code>cao</code> and <code>mahalanobis</code>.
Euclidean distance required for SMARTPCA scaling.<br />
</p>
<p><code>sample_remove</code> should include both samples removed from PCA and ancient samples projected onto PCA space (if any).<br />
</p>
<p>Data read from working directory with SNPs as rows and samples as columns. Two alternative formats: (1) text file of SNPs by samples (file extension and column separators recognized automatically) read using <code><a href="data.table.html#topic+fread">fread</a></code>; or (2) duet of <code>EIGENSTRAT</code> files (see <a href="https://reich.hms.harvard.edu/software">https://reich.hms.harvard.edu/software</a>) using <code><a href="vroom.html#topic+vroom_fwf">vroom_fwf</a></code>, including a genotype file of SNPs by samples <code>*.geno</code>, and a sample file (<code>*.ind</code>) containing three vectors assigning individual samples to unique user-predefined groups (populations), sexes (or other user-defined descriptor) and alphanumeric identifiers. For <code>EIGENSTRAT</code>, vector <code>sample_group</code> assigns samples to groups retrievable from column 3 of file <code>*.ind</code>.
SNPs with zero variance removed prior to SVD to optimize computation time and avoid undefined values if <code>scaling = "sd"</code> or <code>"drift"</code>.<br />
</p>
<p>Users can select subsets of samples or SNPs by introducing a vector including column numbers for samples (<code>sample_remove</code>) and/or row numbers for SNPs (<code>snp_remove</code>) to be removed from computations.
Function stops if the final number of SNPs is 1 or 2.
<code>EIGENSOFT</code> was conceived for the analysis of human genes and its <code>SMARTPCA</code> suite so accepts 22 (autosomal) chromosomes by default.
If &gt;22 chromosomes are provided and the internal parameter <code>numchrom</code> is not set to the target number chromosomes of interest, SMARTPCA automatically subsets chromosomes 1 to 22.
In contrast, <code>smart_permdisp</code> accepts any number of autosomes with or without the sex chromosomes from an <code>EIGENSTRAT</code> file.<br />
</p>


<h3>Value</h3>

<p>Returns a list containing the following elements:
</p>

<ul>
<li><p>permdisp.samplesDataframe showing sample summary.
Column <em>Group</em> assigns samples to tested groups.
Column <em>Class</em> specifies if samples were used in, or removed from, testing (PERMDISP).
Column <em>Sample_dispersion</em> shows dispersion of individual samples relative to spatial <code>"median"</code> or <code>"centroid"</code>.
</p>
</li>
<li><p>permdisp.biasString indicating if PERMDISP dispersions corrected for number of samples per group.
</p>
</li>
<li><p>permdisp.group_locationDataframe showing coordinates of spatial <code>"median"</code> or <code>"centroid"</code> per group.
</p>
</li>
<li><p>permdisp.global_testList showing table with degrees of freedom, sum of squares, mean sum of squares, <em>F</em> statistic and <em>p</em> value.
</p>
</li>
<li><p>permdisp.pairwise_testList showing table with <em>F</em> statistic, <em>p</em> value and corrected <em>p</em> value per pair of groups.
Obtained only if <code>pairwise = TRUE</code>.
</p>
</li>
<li><p>permdisp.pairwise_correctionString indicating type of correction for multiple testing.
</p>
</li>
<li><p>permdisp.permutation_numberNumber of permutations applied to obtain the distribution of <em>F</em> statistic.
</p>
</li>
<li><p>permdisp.permutation_seedNumber fixing random generator of permutations for reproducibility of results.
</p>
</li></ul>



<h3>References</h3>

<p>Anderson, M. J. (2001) A new method for non-parametric multivariate analysis of variance. Austral Ecology, 26, 32-46.<br />
Anderson, M. J. (2004). PERMANOVA_2factor: a FORTRAN computer program for permutational multivariate analysis of variance (for any two-factor ANOVA design) using permutation tests  (Department of Statistics, University of Auckland, New Zealand).<br />
Anderson, M. J. &amp; D. C. I. Walsh (2013) PERMANOVA, ANOSIM, and the Mantel test in the face of heterogeneous dispersions: What null hypothesis are you testing? Ecological Monographs, 83, 557-574.<br />
Gower, J. C. (1966) Some distance properties of latent root and vector methods used in multivariate analysis. Biometrika, 53, 325-338.<br />
McArdle, B. H. &amp; M. J. Anderson (2001) Fitting multivariate models to community data: a comment on distance-based redundancy analysis. Ecology, 82, 290-297.<br />
Patterson, N., A. L. Price and D. Reich (2006) Population structure and eigenanalysis. PLoS Genetics, 2, e190.<br />
Warton, D. I., S. T. Wright and Y. Wang (2012) Distance-based multivariate analyses confound location and dispersion effects. Methods in Ecology and Evolution, 3, 89-101.
</p>


<h3>See Also</h3>

<p><code><a href="vegan.html#topic+adonis">adonis</a></code> (package <b>vegan</b>),
<code><a href="Rfast.html#topic+Dist">Dist</a></code> (package <b>Rfast</b>),
<code><a href="data.table.html#topic+fread">fread</a></code> (package <b>data.table</b>),
<code><a href="vegan.html#topic+vegdist">vegdist</a></code> (package <b>vegan</b>),
<code><a href="vroom.html#topic+vroom_fwf">vroom_fwf</a></code> (package <b>vroom</b>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Path to example genotype matrix "dataSNP"
pathToGenoFile = system.file("extdata", "dataSNP", package = "smartsnp")

# Assign 50 samples to each of two groups and colours
my_groups &lt;- as.factor(c(rep("A", 50), rep("B", 50))); cols = c("red", "blue")

# Run PERMDISP
permdispR &lt;- smart_permdisp(snp_data = pathToGenoFile, sample_group = my_groups)

# Extract summary table assigning samples to groups and dispersion of individual samples
permdispR$permdisp.samples

# Extract PERMDISP table
permdispR$permdisp.global_test

# Plot sample distances to group central medians
#run pca with truncated SVD (PCA 1 x PCA 2)
pcaR1 &lt;- smart_pca(snp_data = pathToGenoFile, sample_group = my_groups)
#compute Euclidean inter-sample distances in PCA space (triangular matrix)
snp_eucli &lt;- vegan::vegdist(pcaR1$pca.sample_coordinates[,c("PC1","PC2")], method = "euclidean")
#calculate spatial medians
disMed &lt;- vegan::betadisper(d = snp_eucli, group = my_groups); disMed
#plot
oldpar &lt;- par(mar = c(4, 4, 5, 0.1), lwd = 2)
boxplot(disMed, las =2, cex.axis = 2, cex.main = 1.5, horizontal = TRUE, varwidth = TRUE,
   col = cols, xlab = "", ylab = "", main = "Sample distance to group spatial medians")
par(oldpar)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
