<!DOCTYPE html><html><head><title>Help for package SMDIC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SMDIC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SMDIC-package'><p>Identification of somatic mutation-driven immune cells</p></a></li>
<li><a href='#cell24'><p>A data.frame of 24 immune cells name from Bindea et al</p></a></li>
<li><a href='#cell64'><p>A data.frame of 64 immune cells name from xCell method</p></a></li>
<li><a href='#envData'>
<p>envData</p></a></li>
<li><a href='#exp2cell'><p>exp2cell</p></a></li>
<li><a href='#gene2cellsummary'><p>gene2cellsummary</p></a></li>
<li><a href='#GetExampleData'><p>Get the example data</p></a></li>
<li><a href='#heatmapcell'><p>heatmapcell</p></a></li>
<li><a href='#immunelist'><p>A large list of 24 immune cells type-specific gene signatures from Bindea et al</p></a></li>
<li><a href='#maf2matrix'><p>maf2matrix</p></a></li>
<li><a href='#mutcellsummary'><p>mutcellsummary</p></a></li>
<li><a href='#mutcorcell'><p>mutcorcell</p></a></li>
<li><a href='#plotCoocMutex'><p>plotCoocMutex</p></a></li>
<li><a href='#plotwaterfall'><p>plotwaterfall</p></a></li>
<li><a href='#survcell'><p>survcell</p></a></li>
<li><a href='#xCell.data'><p>xCell datasets</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identification of Somatic Mutation-Driven Immune Cells</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.5</td>
</tr>
<tr>
<td>Author:</td>
<td>Baotong Zheng, Junwei Han</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A computing tool is developed to automated identify somatic mutation-driven immune cells. The operation modes including: i) inferring the relative abundance matrix of tumor-infiltrating immune cells and integrating it with a particular gene mutation status, ii) detecting differential immune cells with respect to the gene mutation status and converting the abundance matrix of significant differential immune cell into two binary matrices (one for up-regulated and one for down-regulated), iii) identifying somatic mutation-driven immune cells by comparing the gene mutation status with each immune cell in the binary matrices across all samples, and iv) visualization of immune cell abundance of samples in different mutation status..</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>GSVA, samr, e1071, parallel, preprocessCore, pheatmap,
maftools, grDevices, survival, survminer, MASS, pracma, stats,
RColorBrewer, backports</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, R.utils</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-07 11:10:19 UTC; 20687</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-12 04:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='SMDIC-package'>Identification of somatic mutation-driven immune cells</h2><span id='topic+SMDIC-package'></span>

<h3>Description</h3>

<p>With the use of functions in this packages, users could identify the immune cells driven by somatic mutations in tumor microenvironment.
</p>

<hr>
<h2 id='cell24'>A data.frame of 24 immune cells name from Bindea et al</h2><span id='topic+cell24'></span>

<h3>Description</h3>

<p>It's a built-in data. The first column represents the abbreviation of 24 immune cells, the second column represents the full name of 24 immune cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cell24
</code></pre>


<h3>Format</h3>

<p>A data.frame with 24 rows and 2 column
</p>


<h3>References</h3>

<p>Bindea G, Mlecnik B, Tosolini M, Kirilovsky A, Waldner M, Obenauf AC, et al. Spatiotemporal dynamics of intratumoral immune cells reveal the immune landscape in human cancer. Immunity. 2013;39:782–95.
</p>

<hr>
<h2 id='cell64'>A data.frame of 64 immune cells name from xCell method</h2><span id='topic+cell64'></span>

<h3>Description</h3>

<p>It's a built-in data. The first column represents the abbreviation of 64 immune cells, the second column represents the full name of 64 immune cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cell64
</code></pre>


<h3>Format</h3>

<p>A data.frame with 64 rows and 2 column
</p>


<h3>References</h3>

<p>Aran D , Hu Z , Butte A J . xCell: digitally portraying the tissue cellular heterogeneity landscape[J]. Genome Biology, 2017, 18(1):220.
</p>

<hr>
<h2 id='envData'>
envData
</h2><span id='topic+envData'></span><span id='topic+exp.example'></span><span id='topic+cellmatrix'></span><span id='topic+mutcell'></span><span id='topic+mutmatrix'></span>

<h3>Description</h3>

<p>The variables in the environment include an example expression profiles, a cell abundance matrix, a binary numerical matrix which shows the immune cells driven by somatic mutation, a binary mutations matrix.
</p>


<h3>Format</h3>

<p>An environment variable</p>


<h3>Details</h3>

<p>The environment variable includes the variable <code>exp.example</code>, <code>cellmatrix</code>,<code>mutcell</code>,<code>mutmatrix</code>
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Baotong Zheng<a href="mailto:btzheng1116@163.com">btzheng1116@163.com</a></p>

<hr>
<h2 id='exp2cell'>exp2cell</h2><span id='topic+exp2cell'></span>

<h3>Description</h3>

<p>Function 'exp2cell' use gene expression profiles to quantify cell abundance matrix.
'exp2cell' provides three methods for estimating the relative infiltration abundance of different cell types in the tumor microenvironment (TME),
which including xCell, ssGSEA estimated method proposed by Şenbabaoğlu et al. and CIBERSORT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp2cell(exp, method = "xCell", perm = 100, QN = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exp2cell_+3A_exp">exp</code></td>
<td>
<p>The gene expression data set. A matrix with row names as symbols and columns as samples.
Gene expression profiles were used to quantify cell abundance matrix.</p>
</td></tr>
<tr><td><code id="exp2cell_+3A_method">method</code></td>
<td>
<p>Method must be one of &quot;xCell&quot;, &quot;ssGSEA&quot; and &quot;CIBERSORT&quot;.</p>
</td></tr>
<tr><td><code id="exp2cell_+3A_perm">perm</code></td>
<td>
<p>No. permutations; set to &gt;=100 to calculate p-values (default = 100)</p>
</td></tr>
<tr><td><code id="exp2cell_+3A_qn">QN</code></td>
<td>
<p>Quantile normalization of input mixture (default = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cell abundance matrix.
</p>


<h3>References</h3>

<p>1. Aaron, M, Newman, et al. Robust enumeration of cell subsets from tissue expression profiles.[J]. Nature Methods, 2015.
2. Aran D , Hu Z , Butte A J . xCell: digitally portraying the tissue cellular heterogeneity landscape[J]. Genome Biology, 2017, 18(1):220.
3. Şenbabaoğlu, Yasin, Gejman R S , Winer A G , et al. Tumor immune microenvironment characterization in clear cell renal cell carcinoma identifies prognostic and immunotherapeutically relevant messenger RNA signatures[J]. Genome biology, 2016, 17(1).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get breast cancer gene expression profile.
exp.example&lt;-GetExampleData("exp.example")

#perform the exp2cell method. Method must be one of "xCell","ssGSEA" and "CIBERSORT".
cellmatrix&lt;-exp2cell(exp=exp.example,method="ssGSEA") #cell abundance matrix
</code></pre>

<hr>
<h2 id='gene2cellsummary'>gene2cellsummary</h2><span id='topic+gene2cellsummary'></span>

<h3>Description</h3>

<p>Function 'gene2cellsummary' is a generic function used to produce result summaries of the immune cells driven by a somatic mutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene2cellsummary(gene, method = "xCell", mutcell)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene2cellsummary_+3A_gene">gene</code></td>
<td>
<p>Somatic mutant gene name</p>
</td></tr>
<tr><td><code id="gene2cellsummary_+3A_method">method</code></td>
<td>
<p>Method must be one of &quot;xCell&quot;,&quot;ssGSEA&quot; and &quot;CIBERSORT&quot;.</p>
</td></tr>
<tr><td><code id="gene2cellsummary_+3A_mutcell">mutcell</code></td>
<td>
<p>The result of 'mutcorcell' funtion.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix shows the short name, full name, pvalue, fdr, cell responses(up or down) of the cells driven by a somatic mutation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the result of `mutcorcell` funtion.
mutcell&lt;-GetExampleData("mutcell")

# perform the function gene2cellsummary
genecellsummary&lt;-gene2cellsummary(gene="TP53",mutcell=mutcell)
</code></pre>

<hr>
<h2 id='GetExampleData'>Get the example data</h2><span id='topic+GetExampleData'></span>

<h3>Description</h3>

<p>Get the example data from SMDIC package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetExampleData(exampleData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetExampleData_+3A_exampledata">exampleData</code></td>
<td>
<p>A character, should be one of &quot;exp.example&quot;, &quot;cellmatrix&quot;, &quot;mutcell&quot;, &quot;mutmatrix&quot;, &quot;surv&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function 'GetExampleData(ExampleData = &quot;mutmatrix)&quot;)' obtains the mutations matrix
</p>


<h3>References</h3>

<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>

<hr>
<h2 id='heatmapcell'>heatmapcell</h2><span id='topic+heatmapcell'></span>

<h3>Description</h3>

<p>A function to draw clustered heatmaps for the cells driven by a somatic mutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmapcell(
  gene,
  mutcell,
  cellmatrix,
  mutmatrix,
  title = NA,
  show_rownames = TRUE,
  show_colnames = FALSE,
  annotation_colors = NA,
  annotation_row = NA,
  annotation_col = NA,
  color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmapcell_+3A_gene">gene</code></td>
<td>
<p>Somatic mutant gene name</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_mutcell">mutcell</code></td>
<td>
<p>A list, mutcell is the result of function 'mutcorcell'.</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_cellmatrix">cellmatrix</code></td>
<td>
<p>Cell abundance matrix, cellmatrix is the result of function 'exp2cell'.</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_mutmatrix">mutmatrix</code></td>
<td>
<p>A binary mutations matrix, which can not only come from the maf2matrix function, but also any binary mutations matrix, in which 1 represents any mutation occurs in a particular gene in a particular sample, otherwise the element is 0.</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_title">title</code></td>
<td>
<p>The title of the plot</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_show_rownames">show_rownames</code></td>
<td>
<p>boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_show_colnames">show_colnames</code></td>
<td>
<p>boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_annotation_colors">annotation_colors</code></td>
<td>
<p>list for specifying annotation_row and annotation_col track colors manually. It is possible to define the colors for only some of the features. Check examples for details.</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_annotation_row">annotation_row</code></td>
<td>
<p>data frame that specifies the annotations shown on left side of the heatmap. Each row defines the features for a specific row. The rows in the data and in the annotation are matched using corresponding row names. Note that color schemes takes into account if variable is continuous or discrete.</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_annotation_col">annotation_col</code></td>
<td>
<p>similar to annotation_row, but for columns.</p>
</td></tr>
<tr><td><code id="heatmapcell_+3A_color">color</code></td>
<td>
<p>vector of colors used in heatmap.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#get the result of `mutcorcell` function.
mutcell&lt;-GetExampleData("mutcell")

#get cell abundance matrix which is the result of exp2cell function
cellmatrix&lt;-GetExampleData("cellmatrix")

#get the binary mutations matrix
mutmatrix&lt;-GetExampleData("mutmatrix")

# plot significant up-regulation or down-regulation cells heat map specific for breast cancer
heatmapcell(gene = "TP53",mutcell = mutcell,cellmatrix = cellmatrix,mutmatrix = mutmatrix)
</code></pre>

<hr>
<h2 id='immunelist'>A large list of 24 immune cells type-specific gene signatures from Bindea et al</h2><span id='topic+immunelist'></span>

<h3>Description</h3>

<p>It's a built-in data. The name of the list represent 24 immune cells, the value of the list are 24 immune cells type-specific gene signatures from Bindea et al
</p>


<h3>Usage</h3>

<pre><code class='language-R'>immunelist
</code></pre>


<h3>Format</h3>

<p>A list
</p>


<h3>References</h3>

<p>Bindea G, Mlecnik B, Tosolini M, Kirilovsky A, Waldner M, Obenauf AC, et al. Spatiotemporal dynamics of intratumoral immune cells reveal the immune landscape in human cancer. Immunity. 2013;39:782–95.
</p>

<hr>
<h2 id='maf2matrix'>maf2matrix</h2><span id='topic+maf2matrix'></span>

<h3>Description</h3>

<p>Function 'maf2matrix' use mutation annotation file (MAF) format data to build a binary mutations matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maf2matrix(maffile, percent = 0.01, nonsynonymous = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maf2matrix_+3A_maffile">maffile</code></td>
<td>
<p>The name of mutation annotation file (MAF) format data. It must be an absolute path or the name relatived to the current working directory.</p>
</td></tr>
<tr><td><code id="maf2matrix_+3A_percent">percent</code></td>
<td>
<p>A threshold value(one percent as the default value). The genes with a given mutation frequency equal or greater than the threshold value are retained for the following analysis.</p>
</td></tr>
<tr><td><code id="maf2matrix_+3A_nonsynonymous">nonsynonymous</code></td>
<td>
<p>Logical, tell if extract the non-silent somatic mutations (nonsense mutation, missense mutation, frame-shif indels, splice site, nonstop mutation, translation start site, inframe indels).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A binary mutations matrix, in which 1 represents any mutation occurs in a particular gene in a particular sample, otherwise the element is 0.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get path of the mutation annotation file.
maf = system.file('extdata', 'example.maf.gz', package = 'SMDIC')

# perform function `maf2matrix`.
mutmatrix.example&lt;-maf2matrix(maf)
</code></pre>

<hr>
<h2 id='mutcellsummary'>mutcellsummary</h2><span id='topic+mutcellsummary'></span>

<h3>Description</h3>

<p>Function 'mutcellsummary' is a generic function used to produce summaries of the results of 'mutcorcell' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutcellsummary(mutcell, mutmatrix, cellmatrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutcellsummary_+3A_mutcell">mutcell</code></td>
<td>
<p>The result of 'mutcorcell' funtion.</p>
</td></tr>
<tr><td><code id="mutcellsummary_+3A_mutmatrix">mutmatrix</code></td>
<td>
<p>A binary mutations matrix, which can not only come from the maf2matrix function, but also any binary mutations matrix, in which 1 represents any mutation occurs in a particular gene in a particular sample, otherwise the element is 0.</p>
</td></tr>
<tr><td><code id="mutcellsummary_+3A_cellmatrix">cellmatrix</code></td>
<td>
<p>Cell abundance matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result summaries have four columns. The first column is somatic mutant gene names, the second column is the immune cell names driven by the somatic mutation, the third column is the number of the immune cell, the fourth column is the mutation rate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get result of `mutcorcell` funtion
mutcell&lt;-GetExampleData("mutcell")

#get cell abundance matrix which is the result of exp2cell function
cellmatrix&lt;-GetExampleData("cellmatrix")

# get the binary mutations matrix
mutmatrix&lt;-GetExampleData("mutmatrix") # A binary mutations matrix

#perform the function mutcellsummary
summary&lt;-mutcellsummary(mutcell = mutcell,mutmatrix = mutmatrix,cellmatrix=cellmatrix)
</code></pre>

<hr>
<h2 id='mutcorcell'>mutcorcell</h2><span id='topic+mutcorcell'></span>

<h3>Description</h3>

<p>Function 'mutcorcell' identifies somatic mutation-driven immune cells by comparing the cell abundance matrix and binary mutations matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutcorcell(
  cellmatrix = cellmatrix,
  mutmatrix = mutmatrix,
  samfdr.cutoff = 0.05,
  nperms = 100,
  fisher.cutoff = 0.05,
  fisher.adjust = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutcorcell_+3A_cellmatrix">cellmatrix</code></td>
<td>
<p>Cell abundance matrix.</p>
</td></tr>
<tr><td><code id="mutcorcell_+3A_mutmatrix">mutmatrix</code></td>
<td>
<p>A binary mutations matrix, which can not only come from the maf2matrix function, but also any binary mutations matrix, in which 1 represents any mutation occurs in a particular gene in a particular sample, otherwise the element is 0.</p>
</td></tr>
<tr><td><code id="mutcorcell_+3A_samfdr.cutoff">samfdr.cutoff</code></td>
<td>
<p>False Discovery Rate cutoff for output in significant immune cells</p>
</td></tr>
<tr><td><code id="mutcorcell_+3A_nperms">nperms</code></td>
<td>
<p>Number of permutations used by SAM to estimate False Discovery Rates</p>
</td></tr>
<tr><td><code id="mutcorcell_+3A_fisher.cutoff">fisher.cutoff</code></td>
<td>
<p>False Discovery Rate(fisher.adjust=TRUE) or P-Value(fisher.adjust=FALSE) cutoff for Fisher's exact test</p>
</td></tr>
<tr><td><code id="mutcorcell_+3A_fisher.adjust">fisher.adjust</code></td>
<td>
<p>Logical,tell if corrects p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of four matrices: a binary numerical matrix which shows the immune cells driven by somatic mutant gene;
two numerical matrix which show the pvalue and fdr of the immune cells driven by somatic mutant gene;
a character matrix which shows the cell responses of the immune cells driven by somatic mutant gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get cell abundance matrix which is the result of exp2cell function
cellmatrix&lt;-GetExampleData("cellmatrix")

#get the binary mutations matrix,
mutmatrix&lt;-GetExampleData("mutmatrix")

#perform the function `mutcorcell`.
mutcell&lt;-mutcorcell(cellmatrix = cellmatrix,mutmatrix = mutmatrix)

# The summary for somatic mutations are produced by function `mutcellsummary`.
#summary&lt;-mutcellsummary(mutcell = mutcell,mutmatrix = mutmatrix,cellmatrix=cellmatrix)

# The summary of the immune cells driven by a mutation are produced by function `gene2cellsummary`.
#genecellsummary&lt;-gene2cellsummary(gene="TP53",mutcell=mutcell)
</code></pre>

<hr>
<h2 id='plotCoocMutex'>plotCoocMutex</h2><span id='topic+plotCoocMutex'></span>

<h3>Description</h3>

<p>Function 'plotCoocMutex' plots the co-occurrence and mutual exclusivity plots for mutation genes which drive immune cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCoocMutex(maffile, mutcell.summary, cellnumcuoff = 3, fontSize = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotCoocMutex_+3A_maffile">maffile</code></td>
<td>
<p>The name of mutation annotation file (MAF) format data. It must be an absolute path or the name  relatived to the current working directory.</p>
</td></tr>
<tr><td><code id="plotCoocMutex_+3A_mutcell.summary">mutcell.summary</code></td>
<td>
<p>The result of 'mutcellsummary' function</p>
</td></tr>
<tr><td><code id="plotCoocMutex_+3A_cellnumcuoff">cellnumcuoff</code></td>
<td>
<p>A threshold value (4 as the default value). The mutation genes which drive at least &quot;cellnumcuoff&quot; cells are retained for drawing a co-occurrence and mutual exclusivity plots.</p>
</td></tr>
<tr><td><code id="plotCoocMutex_+3A_fontsize">fontSize</code></td>
<td>
<p>cex for gene names. Default 0.8</p>
</td></tr>
</table>


<h3>References</h3>

<p>Gerstung M, Pellagatti A, Malcovati L, et al. Combining gene mutation with gene expression data improves outcome prediction in myelodysplastic syndromes. Nature Communications. 2015;6:5901. doi:10.1038/ncomms6901.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the result of `exp2cell` funtion
cellmatrix&lt;-GetExampleData("cellmatrix")

#get the binary mutations matrix,
mutmatrix&lt;-GetExampleData("mutmatrix")

# get the result of `mutcorcell` funtion
mutcell&lt;-GetExampleData("mutcell")

#perform the function mutcellsummary
summary&lt;-mutcellsummary(mutcell = mutcell,mutmatrix = mutmatrix,cellmatrix=cellmatrix)

#dir is the name of mutation annotation file (MAF) format data.
#It must be an absolute path or the name relatived to the current working directory.
maf&lt;-system.file("extdata", "example.maf.gz", package = "SMDIC") #MAF file
#plot the co-occurrence and mutual exclusivity plots for mutation genes which drive immune cells.
plotCoocMutex(maffile = maf,mutcell.summary = summary,cellnumcuoff =0)
</code></pre>

<hr>
<h2 id='plotwaterfall'>plotwaterfall</h2><span id='topic+plotwaterfall'></span>

<h3>Description</h3>

<p>Function 'plotwaterfall' plots the waterfall for mutation genes which drive immune cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotwaterfall(
  maffile,
  mutcell.summary,
  cellnumcuoff = 3,
  fontSize = 0.8,
  showTumorSampleBarcodes = F,
  showTitle = TRUE,
  colors = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotwaterfall_+3A_maffile">maffile</code></td>
<td>
<p>The name of mutation annotation file (MAF) format data. It must be an absolute path or the name  relatived to the current working directory.</p>
</td></tr>
<tr><td><code id="plotwaterfall_+3A_mutcell.summary">mutcell.summary</code></td>
<td>
<p>The result of 'mutcellsummary' function</p>
</td></tr>
<tr><td><code id="plotwaterfall_+3A_cellnumcuoff">cellnumcuoff</code></td>
<td>
<p>a threshold value (3 as the default value). The mutation genes which drive at least &quot;cellnumcuoff&quot; cells are retained for drawing an waterfall.</p>
</td></tr>
<tr><td><code id="plotwaterfall_+3A_fontsize">fontSize</code></td>
<td>
<p>font size for gene names. Default 0.8.</p>
</td></tr>
<tr><td><code id="plotwaterfall_+3A_showtumorsamplebarcodes">showTumorSampleBarcodes</code></td>
<td>
<p>logical to include sample names.</p>
</td></tr>
<tr><td><code id="plotwaterfall_+3A_showtitle">showTitle</code></td>
<td>
<p>Default TRUE</p>
</td></tr>
<tr><td><code id="plotwaterfall_+3A_colors">colors</code></td>
<td>
<p>named vector of colors for each Variant_Classification.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># get result of `exp2cell` funtion
cellmatrix&lt;-GetExampleData("cellmatrix")

#get the binary mutations matrix,
mutmatrix&lt;-GetExampleData("mutmatrix")

# get the result of `mutcorcell` funtion
mutcell&lt;-GetExampleData("mutcell")

#perform the function mutcellsummary
summary&lt;-mutcellsummary(mutcell = mutcell,mutmatrix = mutmatrix,cellmatrix=cellmatrix)

#dir is the name of mutation annotation file (MAF) format data.
#It must be an absolute path or the name relatived to the current working directory.
maf&lt;-system.file("extdata", "example.maf.gz", package = "SMDIC") #MAF file
# mutcell.summary is the result of function mutcellsummary

#plot the waterfall for mutation genes which drive immune cells
plotwaterfall(maffile = maf,mutcell.summary = summary,cellnumcuoff =3)
</code></pre>

<hr>
<h2 id='survcell'>survcell</h2><span id='topic+survcell'></span>

<h3>Description</h3>

<p>Function 'survcell' draws Kaplan–Meier curves for survival in the above-median and below-median groups for cell risk score. The cell risk score is calaulated by the weighted mean of cells driven by a gene mutation, where the  weight of cells is estimated by the &quot;Univariate&quot; or &quot;Multivariate&quot; cox.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survcell(
  gene,
  mutcell,
  cellmatrix,
  surv,
  method = "Multivariate",
  legend.title = "Strata",
  legend.labs = c("group=0", "group=1"),
  palette = c("#E7B800", "#2E9FDF"),
  color = NULL,
  pval = TRUE,
  title = NULL,
  ggtheme = theme_survminer()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survcell_+3A_gene">gene</code></td>
<td>
<p>Somatic mutant gene name</p>
</td></tr>
<tr><td><code id="survcell_+3A_mutcell">mutcell</code></td>
<td>
<p>The result of 'mutcorcell' function</p>
</td></tr>
<tr><td><code id="survcell_+3A_cellmatrix">cellmatrix</code></td>
<td>
<p>Cell abundance matrix</p>
</td></tr>
<tr><td><code id="survcell_+3A_surv">surv</code></td>
<td>
<p>Surv is the survival data, the first column is the sample name, the second column is the survival time, and the third is the survival event.</p>
</td></tr>
<tr><td><code id="survcell_+3A_method">method</code></td>
<td>
<p>Method must be one of &quot;Univariate&quot; and &quot;Multivariate&quot;. The coefficient of cells for risk score are estimated by &quot;Univariate&quot; or &quot;Multivariate&quot; cox proportional risk regression model on cell abundance matrix and overall survival data..</p>
</td></tr>
<tr><td><code id="survcell_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title.</p>
</td></tr>
<tr><td><code id="survcell_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels. Used to replace the names of the strata from the fit. Should be given in the same order as those strata.</p>
</td></tr>
<tr><td><code id="survcell_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used. Allowed values include &quot;hue&quot; for the default hue color scale; &quot;grey&quot; for grey color palettes; brewer palettes e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;. See details section for more information. Can be also a numeric vector of length(groups); in this case a basic color palette is created using the function palette.</p>
</td></tr>
<tr><td><code id="survcell_+3A_color">color</code></td>
<td>
<p>color to be used for the survival curves.If the number of strata/group (n.strata) = 1, the expected value is the color name. For example color = &quot;blue&quot;.If n.strata &gt; 1, the expected value is the grouping variable name. By default, survival curves are colored by strata using the argument color = &quot;strata&quot;, but you can also color survival curves by any other grouping variables used to fit the survival curves. In this case, it's possible to specify a custom color palette by using the argument palette.</p>
</td></tr>
<tr><td><code id="survcell_+3A_pval">pval</code></td>
<td>
<p>logical value, a numeric or a string. If logical and TRUE, the p-value is added on the plot. If numeric, than the computet p-value is substituted with the one passed with this parameter. If character, then the customized string appears on the plot.</p>
</td></tr>
<tr><td><code id="survcell_+3A_title">title</code></td>
<td>
<p>the title of the survival curve</p>
</td></tr>
<tr><td><code id="survcell_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_survminer. Allowed values include ggplot2 official themes: see theme.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Kaplan–Meier curves
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get the result of `mutcorcell` function.
mutcell&lt;-GetExampleData("mutcell")

# get cell abundance matrix which is the result of exp2cell function
cellmatrix&lt;-GetExampleData("cellmatrix")

# get survival data
surv&lt;-GetExampleData("surv")

#draw Kaplan–Meier curves
survcell(gene ="TP53",mutcell=mutcell,cellmatrix=cellmatrix,surv=surv)
</code></pre>

<hr>
<h2 id='xCell.data'>xCell datasets</h2><span id='topic+xCell.data'></span>

<h3>Description</h3>

<p>xCell datasets. It's a built-in data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xCell.data
</code></pre>


<h3>Format</h3>

<p>list:
</p>

<dl>
<dt>spill</dt><dd><p>spillover matrix and calibration parameters</p>
</dd>
<dt>signatures</dt><dd><p>the signatures for calculating scores</p>
</dd>
<dt>genes</dt><dd><p>genes to use to calculate xCell</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
