<!DOCTYPE html><html lang="en"><head><title>Help for package smer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {smer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#approximate_memory_requirements'><p>Estimate Memory Requirements for SME Routine</p></a></li>
<li><a href='#create_hdf5_file'><p>Create an HDF5 File</p></a></li>
<li><a href='#getting_started'><p>Simulated Dataset for Genome-Wide Interaction Analysis</p></a></li>
<li><a href='#read_hdf5_dataset'><p>Read Dataset from an HDF5 File</p></a></li>
<li><a href='#simulate_traits'><p>Simulate Quantitative Traits from PLINK Genotypes</p></a></li>
<li><a href='#sme'><p>Sparse Marginal Epistasis Test (SME)</p></a></li>
<li><a href='#write_hdf5_dataset'><p>Write Data to an HDF5 Dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Sparse Marginal Epistasis Test</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/lcrawlab/sme">https://github.com/lcrawlab/sme</a>, <a href="https://lcrawlab.github.io/sme/">https://lcrawlab.github.io/sme/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/lcrawlab/sme/issues">https://github.com/lcrawlab/sme/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>The Sparse Marginal Epistasis Test is a computationally efficient 
    genetics method which detects statistical epistasis in complex traits;
    see Stamp et al. (2025, &lt;<a href="https://doi.org/10.1101%2F2025.01.11.632557">doi:10.1101/2025.01.11.632557</a>&gt;) for details.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>BH, Rcpp, RcppEigen, Rhdf5lib, testthat</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, genio, logging, mvMAPIT, Rcpp, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>GenomicRanges, ggplot2, knitr, rmarkdown, testthat (&ge;
3.0.0), xml2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.4.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>biocViews:</td>
<td>GenomeWideAssociation, Epistasis, Genetics, SNP,
LinearMixedModel</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-16 07:09:10 UTC; jds</td>
</tr>
<tr>
<td>Author:</td>
<td>Julian Stamp <a href="https://orcid.org/0000-0003-3014-6249"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut],
  Lorin Crawford <a href="https://orcid.org/0000-0003-0178-8242"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  sriramlab [cph] (Author of included mailman algorithm),
  Blue Brain Project/EPFL [cph] (Author of included HighFive library)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Julian Stamp &lt;julian.d.stamp@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-16 15:50:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='approximate_memory_requirements'>Estimate Memory Requirements for SME Routine</h2><span id='topic+approximate_memory_requirements'></span>

<h3>Description</h3>

<p>This function provides an approximate estimate of the memory requirements
(in gigabytes) for running the Sparse Marginal Epistasis (SME) routine
based on input parameters such as the number of samples, SNPs, and other configurations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>approximate_memory_requirements(
  n_samples,
  n_snps,
  n_blocks,
  n_randvecs,
  chunksize
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="approximate_memory_requirements_+3A_n_samples">n_samples</code></td>
<td>
<p>Integer. The number of samples in the dataset.</p>
</td></tr>
<tr><td><code id="approximate_memory_requirements_+3A_n_snps">n_snps</code></td>
<td>
<p>Integer. The total number of SNPs in the dataset.</p>
</td></tr>
<tr><td><code id="approximate_memory_requirements_+3A_n_blocks">n_blocks</code></td>
<td>
<p>Integer. The number of genotype blocks used to partition SNPs.
Affects the size of encoded genotype segments.</p>
</td></tr>
<tr><td><code id="approximate_memory_requirements_+3A_n_randvecs">n_randvecs</code></td>
<td>
<p>Integer. The number of random vectors used for stochastic
trace estimation. Affects memory for operations involving random vectors.</p>
</td></tr>
<tr><td><code id="approximate_memory_requirements_+3A_chunksize">chunksize</code></td>
<td>
<p>Integer. The number of focal SNPs processed per chunk.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates memory usage by summing the contributions from
various components used in the SME routine, including:
</p>

<ul>
<li><p> Variance component estimates (<code>vc_estimates</code>)
</p>
</li>
<li><p> Phenotype-related matrices
</p>
</li>
<li><p> Random vector-based computations
</p>
</li>
<li><p> Genotype objects and block statistics
</p>
</li>
<li><p> Gene-by-gene interaction masks
</p>
</li></ul>

<p>The estimated memory requirement is derived from the data dimensions
and operational needs, and it provides a guideline for configuring resources
for the analysis.
</p>


<h3>Value</h3>

<p>Numeric. The approximate memory requirement (in gigabytes) for the
SME routine.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n_samples &lt;- 1e5
n_snps &lt;- 1e6
n_blocks &lt;- 100
n_randvecs &lt;- 100
chunksize &lt;- 10
approximate_memory_requirements(n_samples,
                                n_snps,
                                n_blocks,
                                n_randvecs,
                                chunksize)

</code></pre>

<hr>
<h2 id='create_hdf5_file'>Create an HDF5 File</h2><span id='topic+create_hdf5_file'></span>

<h3>Description</h3>

<p>This function creates a new, empty HDF5 file at the specified location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_hdf5_file(hdf5_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_hdf5_file_+3A_hdf5_file">hdf5_file</code></td>
<td>
<p>A character string specifying the path and name of the HDF5
file to be created.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value; the function creates the HDF5 file at the specified
location.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create an empty HDF5 file
hdf5_file &lt;- tempfile()
create_hdf5_file(hdf5_file)


</code></pre>

<hr>
<h2 id='getting_started'>Simulated Dataset for Genome-Wide Interaction Analysis</h2><span id='topic+getting_started'></span>

<h3>Description</h3>

<p><code>getting_started</code> is a simulated dataset created to demonstrate the use of
the <code>sme()</code> function for genome-wide interaction analyses. It contains
results from a simulated analysis involving additive genetic effects and
gene-by-gene (GxG) interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("getting_started")
</code></pre>


<h3>Format</h3>

<p>A list with results from <code>sme()</code>, including the following components:
</p>

<dl>
<dt><code>summary</code></dt><dd><p>A data frame summarizing the analysis results, including
p-values for SNP associations (<code>p</code>).</p>
</dd>
<dt><code>pve</code></dt><dd><p>A data frame containing the per SNP variance component
estimates normalized to phenotypic variance explained (PVE).</p>
</dd>
<dt><code>vc</code></dt><dd><p>A data frame containing the per SNP variance component
estimates.</p>
</dd>
<dt><code>gxg_snps</code></dt><dd><p>A vector containing the indices of the SNPs assigned to
have epistatic interactions in the trait simulations.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The dataset was generated as follows:
</p>

<ul>
<li> <p><strong>Genotype Simulation</strong>:
Genotype data for 5000 individuals and 6,000 SNPs was simulated with
synthetic allele counts.
</p>
</li>
<li> <p><strong>Phenotype Simulation</strong>:
Phenotypic values were simulated with an additive heritability of 0.3 and a
GxG interaction heritability of 0.25. A set of 100 SNPs were selected for
additive effects, and two groups of 5 SNPs each were used for GxG
interactions.
</p>
</li>
<li> <p><strong>PLINK-Compatible Files</strong>:
The simulated data was saved in PLINK-compatible <code>.bed</code>, <code>.fam</code>,
and <code>.bim</code> files.
</p>
</li>
<li> <p><strong>Interaction Analysis</strong>:
The <code>sme()</code> function was used to perform genome-wide interaction analyses
on a subset of SNP indices, including the GxG SNP groups and 100 additional
additive SNPs. Memory-efficient computation parameters
(e.g., <code>chun_ksize</code>, <code>n_randvecs</code>, and <code>n_blocks</code>) were applied.
</p>
</li></ul>



<h3>Key Parameters</h3>


<ul>
<li> <p><strong>Additive Heritability</strong>: 0.3
</p>
</li>
<li> <p><strong>GxG Heritability</strong>: 0.25
</p>
</li>
<li> <p><strong>Number of Samples</strong>: 5000
</p>
</li>
<li> <p><strong>Number of SNPs</strong>: 6,000
</p>
</li>
<li> <p><strong>Selected Additive SNPs</strong>: 100
</p>
</li>
<li> <p><strong>Selected GxG SNP Groups</strong>:
</p>

<ul>
<li><p> Group 1: 5 SNPs
</p>
</li>
<li><p> Group 2: 5 SNPs
</p>
</li></ul>

</li></ul>



<h3>Source</h3>

<p>data-raw/getting_started.R
</p>


<h3>See Also</h3>

<p><a href="#topic+sme">sme</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("getting_started")
head(getting_started$summary)

</code></pre>

<hr>
<h2 id='read_hdf5_dataset'>Read Dataset from an HDF5 File</h2><span id='topic+read_hdf5_dataset'></span>

<h3>Description</h3>

<p>This function reads a dataset from an existing HDF5 file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_hdf5_dataset(file_name, dataset_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_hdf5_dataset_+3A_file_name">file_name</code></td>
<td>
<p>A character string specifying the path to the HDF5 file.</p>
</td></tr>
<tr><td><code id="read_hdf5_dataset_+3A_dataset_name">dataset_name</code></td>
<td>
<p>A character string specifying the name of the dataset
within the HDF5 file to read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The content of the dataset from the HDF5 file, typically in the form
of an R object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data_to_write &lt;- 1:10

# Create an empty HDF5 file
hdf5_file &lt;- tempfile()
create_hdf5_file(hdf5_file)

# Write new data to a dataset in the HDF5 file
write_hdf5_dataset(hdf5_file, "group/dataset", data_to_write)

# Read a dataset from an HDF5 file
hdf5_data &lt;- read_hdf5_dataset(hdf5_file, "group/dataset")
print(hdf5_data)

</code></pre>

<hr>
<h2 id='simulate_traits'>Simulate Quantitative Traits from PLINK Genotypes</h2><span id='topic+simulate_traits'></span>

<h3>Description</h3>

<p>This function simulates a quantitative trait based on additive and epistatic
genetic effects using genotype data from a PLINK dataset. The simulated trait
is saved to a specified output file in a phenotype format compatible with
PLINK.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_traits(
  plink_file,
  output_file,
  additive_heritability,
  gxg_heritability,
  additive_indices,
  gxg_indices_1,
  gxg_indices_2,
  log_level = "WARNING"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulate_traits_+3A_plink_file">plink_file</code></td>
<td>
<p>Character. Path to the PLINK dataset (without file
extension). The function will append <code>.bed</code>, <code>.bim</code>, and <code>.fam</code> extensions
as needed.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_output_file">output_file</code></td>
<td>
<p>Character. Path to the output file where the simulated
trait will be saved.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_additive_heritability">additive_heritability</code></td>
<td>
<p>Numeric. A value between 0 and 1 specifying the
proportion of trait variance due to additive genetic effects.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_gxg_heritability">gxg_heritability</code></td>
<td>
<p>Numeric. A value between 0 and 1 specifying the
proportion of trait variance due to gene-by-gene (epistatic) interactions.
The sum of <code>additive_heritability</code> and <code>gxg_heritability</code> must not exceed 1.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_additive_indices">additive_indices</code></td>
<td>
<p>Integer vector. Indices of SNPs contributing to
additive genetic effects.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_gxg_indices_1">gxg_indices_1</code></td>
<td>
<p>Integer vector. Indices of SNPs in the first group for
epistatic interactions.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_gxg_indices_2">gxg_indices_2</code></td>
<td>
<p>Integer vector. Indices of SNPs in the second group for
epistatic interactions.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_log_level">log_level</code></td>
<td>
<p>Character. Logging level for messages
(e.g., &quot;DEBUG&quot;, &quot;INFO&quot;, &quot;WARNING&quot;). Default is &quot;WARNING&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function uses the following components to simulate the trait:
</p>

<ul>
<li><p> Additive genetic effects: Determined by <code>additive_indices</code> and the
specified <code>additive_heritability</code>.
</p>
</li>
<li><p> Epistatic interactions: Simulated using pairs of SNPs from <code>gxg_indices_1</code>
and <code>gxg_indices_2</code>, contributing to the <code>gxg_heritability</code>.
</p>
</li>
<li><p> Environmental effects: Any remaining variance not explained by genetic
effects is assigned to random environmental noise.
</p>
</li></ul>

<p>The output file is in PLINK-compatible phenotype format with three columns:
Family ID (<code>FID</code>), Individual ID (<code>IID</code>), and the simulated trait (<code>TRAIT</code>).
</p>


<h3>Value</h3>

<p>None. The simulated trait is written to the specified <code>output_file</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plink_file &lt;- gsub("\\.bed", "", system.file("testdata", "test.bed", package = "smer"))
out_file &lt;- tempfile()
additive_heritability &lt;- 0.3
gxg_heritability &lt;- 0.1
additive_snps &lt;- sort(sample(1:100, 50, replace = FALSE))
gxg_group_1 &lt;- sort(sample(additive_snps, 10, replace = FALSE))
gxg_group_2 &lt;- sort(sample(setdiff(additive_snps, gxg_group_1), 10, replace = FALSE))
n_samples &lt;- 200
simulate_traits(
  plink_file,
  out_file,
  additive_heritability,
  gxg_heritability,
  additive_snps,
  gxg_group_1,
  gxg_group_2
)
from_file &lt;- read.table(out_file, header = TRUE)
head(from_file)

</code></pre>

<hr>
<h2 id='sme'>Sparse Marginal Epistasis Test (SME)</h2><span id='topic+sme'></span>

<h3>Description</h3>

<p>SME fits a linear mixed model in order to test for marginal epistasis. It
concentrates the scans for epistasis to regions of the genome that have known
functional enrichment for a trait of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sme(
  plink_file,
  pheno_file,
  mask_file = NULL,
  gxg_indices = NULL,
  chunk_size = NULL,
  n_randvecs = 10,
  n_blocks = 100,
  n_threads = 1,
  gxg_h5_group = "gxg",
  ld_h5_group = "ld",
  rand_seed = -1,
  log_level = "WARNING"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sme_+3A_plink_file">plink_file</code></td>
<td>
<p>Character. File path to the PLINK dataset
(without *.bed extension).
The function will append <code>.bim</code>, <code>.bed</code>, and <code>.fam</code> extensions
automatically.
The genotype data must not have any missing genotypes. Use PLINK to remove
variants with missing genotypes or impute them.</p>
</td></tr>
<tr><td><code id="sme_+3A_pheno_file">pheno_file</code></td>
<td>
<p>Character. File path to a phenotype file in PLINK format.
The file should contain exactly one phenotype column.</p>
</td></tr>
<tr><td><code id="sme_+3A_mask_file">mask_file</code></td>
<td>
<p>Character or NULL. File path to an HDF5 file specifying
per-SNP masks for gene-by-gene interaction tests. This file informs which
SNPs are tested for marginal epistasis. Defaults to <code>NULL</code>, indicating no
masking. Masking impacts the scaling of memory and time.</p>
</td></tr>
<tr><td><code id="sme_+3A_gxg_indices">gxg_indices</code></td>
<td>
<p>Integer vector or NULL. List of indices corresponding to
SNPs to test for marginal epistasis.
If <code>NULL</code>, all SNPs in the dataset will be tested.
These indices are <strong>1-based</strong>.</p>
</td></tr>
<tr><td><code id="sme_+3A_chunk_size">chunk_size</code></td>
<td>
<p>Integer or NULL. Number of SNPs processed per chunk.
This influences memory
usage and can be left <code>NULL</code> to automatically determine the chunk size
based on <code>gxg_indices</code> and number of threads.</p>
</td></tr>
<tr><td><code id="sme_+3A_n_randvecs">n_randvecs</code></td>
<td>
<p>Integer. Number of random vectors used for stochastic trace
estimation.
Higher values yield more accurate estimates but increase computational
cost. Default is 10.</p>
</td></tr>
<tr><td><code id="sme_+3A_n_blocks">n_blocks</code></td>
<td>
<p>Integer. Number of blocks into which SNPs are divided for
processing.
This parameter affects memory requirements. Default is 100.</p>
</td></tr>
<tr><td><code id="sme_+3A_n_threads">n_threads</code></td>
<td>
<p>Integer. Number of threads for OpenMP parallel processing.
Default is 1.</p>
</td></tr>
<tr><td><code id="sme_+3A_gxg_h5_group">gxg_h5_group</code></td>
<td>
<p>Character. Name of the HDF5 group within the mask file
containing gene-by-gene
interaction masks. SNPs in this group will be included in the gene-by-gene
interactions. Defaults to &quot;gxg&quot;.</p>
</td></tr>
<tr><td><code id="sme_+3A_ld_h5_group">ld_h5_group</code></td>
<td>
<p>Character. Name of the HDF5 group within the mask file
containing linkage disequilibrium
masks. SNPs in this group are excluded from analysis. Defaults to &quot;ld&quot;.</p>
</td></tr>
<tr><td><code id="sme_+3A_rand_seed">rand_seed</code></td>
<td>
<p>Integer. Seed for random vector generation. If <code>-1</code>, no seed
is set. Default is -1.</p>
</td></tr>
<tr><td><code id="sme_+3A_log_level">log_level</code></td>
<td>
<p>Character. Logging level for messages. Must be in uppercase
(e.g., &quot;DEBUG&quot;, &quot;INFO&quot;, &quot;WARNING&quot;, &quot;ERROR&quot;). Default is &quot;WARNING&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function integrates PLINK-formatted genotype and phenotype data to
perform marginal epistasis tests on a set of SNPs. Using stochastic trace
estimation, the method computes variance components for gene-by-gene
interaction and genetic relatedness using the MQS estimator. The process is
parallelized using OpenMP when <code>n_threads &gt; 1</code>.
</p>
<p>The memory requirements and computation time scaling can be optimized through
the parameters <code>chunk_size</code>, <code>n_randvecs</code>, and <code>n_blocks</code>.
</p>
<p><strong>Mask Format Requirements</strong>
</p>
<p>The mask file format is an HDF5 file used for storing index data for
the masking process. This format supports data retrieval by index.
Below are the required groups and datasets within the HDF5 file:
</p>
<p>The required group names can be configured as input parameters.
The defaults are described below.
</p>

<ul>
<li> <p><strong>Groups</strong>:
</p>

<ul>
<li> <p><code>ld</code>: Stores SNPs in LD with the focal SNP. These SNPs will be <strong>excluded</strong>.
</p>
</li>
<li> <p><code>gxg</code>: Stores indices of SNPs that the marginal epistasis test is conditioned on. These SNPs will be <strong>included</strong>.
</p>
</li></ul>

</li>
<li> <p><strong>Datasets</strong>:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;ld/&lt;j&gt;&#8288;</code>: For each focal SNP <code style="white-space: pre;">&#8288;&lt;j&gt;&#8288;</code>, this dataset contains indices of SNPs
in the same LD block as that SNP. These SNPs will be <strong>excluded</strong> from the gene-by-gene interaction covariance matrix.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;gxg/&lt;j&gt;&#8288;</code>: For each focal SNP <code style="white-space: pre;">&#8288;&lt;j&gt;&#8288;</code>, this dataset contains indices of SNPs to <strong>include</strong> in the
the gene-by-gene interaction covariance matrix for focal SNP <code style="white-space: pre;">&#8288;&lt;j&gt;&#8288;</code>.
</p>
</li></ul>

</li></ul>

<p><strong>Important</strong>: All indices in the mask file data are <strong>zero-based</strong>, matching the zero-based indices of the PLINK <code>.bim</code> file.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li> <p><code>summary</code>: A tibble summarizing results for each tested SNP, including:
</p>

<ul>
<li> <p><code>id</code>: Variant ID.
</p>
</li>
<li> <p><code>index</code>: Index of the SNP in the dataset.
</p>
</li>
<li> <p><code>chromosome</code>: Chromosome number.
</p>
</li>
<li> <p><code>position</code>: Genomic position of the SNP.
</p>
</li>
<li> <p><code>p</code>: P value for the gene-by-gene interaction test.
</p>
</li>
<li> <p><code>pve</code>: Proportion of variance explained (PVE) by gene-by-gene interactions.
</p>
</li>
<li> <p><code>vc</code>: Variance component estimate.
</p>
</li>
<li> <p><code>se</code>: Standard error of the variance component.
</p>
</li></ul>

</li>
<li> <p><code>pve</code>: A long-format tibble of PVE for all variance components.
</p>
</li>
<li> <p><code>vc_estimate</code>: A long-format tibble of variance component estimates.
</p>
</li>
<li> <p><code>vc_se</code>: A long-format tibble of standard errors for variance components.
</p>
</li>
<li> <p><code>average_duration</code>: Average computation time per SNP.
</p>
</li></ul>



<h3>References</h3>

<p>Stamp, J., Pattillo Smith, S., Weinreich, D., &amp; Crawford, L. (2025).
Sparse modeling of interactions enables fast detection of genome-wide
epistasis in biobank-scale studies. bioRxiv, 2025.01.11.632557.
</p>
<p>Stamp, J., DenAdel, A., Weinreich, D., &amp; Crawford, L. (2023).
Leveraging the genetic correlation between traits improves the detection of
epistasis in genome-wide association studies.
G3: Genes, Genomes, Genetics, 13(8), jkad118.
</p>
<p>Crawford, L., Zeng, P., Mukherjee, S., &amp; Zhou, X. (2017).
Detecting epistasis with the marginal epistasis test in genetic mapping
studies of quantitative traits. PLoS genetics, 13(7), e1006869.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plink_file &lt;- gsub("\\.bed", "", system.file("testdata", "test.bed", package="smer"))
pheno_file &lt;- system.file("testdata", "test_h2_0.5.pheno", package="smer")
mask_file &lt;- ""

# Parameter inputs
chunk_size &lt;- 10
n_randvecs &lt;- 10
n_blocks &lt;- 10
n_threads &lt;- 1

# 1-based Indices of SNPs to be analyzed
n_snps &lt;- 100
snp_indices &lt;- 1:n_snps

sme_result &lt;- sme(
  plink_file,
  pheno_file,
  mask_file,
  snp_indices,
  chunk_size,
  n_randvecs,
  n_blocks,
  n_threads
)
head(sme_result$summary)

</code></pre>

<hr>
<h2 id='write_hdf5_dataset'>Write Data to an HDF5 Dataset</h2><span id='topic+write_hdf5_dataset'></span>

<h3>Description</h3>

<p>This function writes new data to an existing HDF5 file. If the dataset
already exists, it will be replaced with the new data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_hdf5_dataset(file_name, dataset_name, new_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_hdf5_dataset_+3A_file_name">file_name</code></td>
<td>
<p>A character string specifying the path to the HDF5 file.</p>
</td></tr>
<tr><td><code id="write_hdf5_dataset_+3A_dataset_name">dataset_name</code></td>
<td>
<p>A character string specifying the name of the dataset
to be written in the HDF5 file.</p>
</td></tr>
<tr><td><code id="write_hdf5_dataset_+3A_new_data">new_data</code></td>
<td>
<p>The new data to write to the dataset. The data must be
compatible with the dataset's structure.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value; the function modifies the specified dataset in the
HDF5 file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data_to_write &lt;- 1:10

# Create an empty HDF5 file
hdf5_file &lt;- tempfile()
create_hdf5_file(hdf5_file)

# Write new data to a dataset in the HDF5 file
write_hdf5_dataset(hdf5_file, "group/dataset", data_to_write)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
