<!DOCTYPE html><html lang="en"><head><title>Help for package SMITIDstruct</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SMITIDstruct}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SMITIDstruct-package'><p>Data Structure and Manipulation Tool for Host and Viral Population</p></a></li>
<li><a href='#addcode'><p>addcode</p></a></li>
<li><a href='#addHost'><p>addHost</p></a></li>
<li><a href='#addIndex'><p>addIndex</p></a></li>
<li><a href='#addViralObs'><p>addViralObs</p></a></li>
<li><a href='#alleleCount'><p>alleleCount</p></a></li>
<li><a href='#concatViralPop'><p>concatViralPop</p></a></li>
<li><a href='#createAViralPop'><p>createAViralPop</p></a></li>
<li><a href='#createHost'><p>createHost</p></a></li>
<li><a href='#createIndex'><p>createIndex</p></a></li>
<li><a href='#demo.SMITIDstruct.run'><p>demo.SMITIDstruct.run</p></a></li>
<li><a href='#diversity.pDistance'><p>diversity.pDistance</p></a></li>
<li><a href='#diversity.sfs'><p>diversity.sfs</p></a></li>
<li><a href='#getCov'><p>getCov</p></a></li>
<li><a href='#getDate'><p>getDate</p></a></li>
<li><a href='#getDiversity.pDistance'><p>getDiversity.pDistance</p></a></li>
<li><a href='#getDiversity.sfs'><p>getDiversity.sfs</p></a></li>
<li><a href='#getInfosByHostAndTime'><p>getInfosByHostAndTime</p></a></li>
<li><a href='#getStates'><p>getStates</p></a></li>
<li><a href='#getTimeLine'><p>getTimeLine</p></a></li>
<li><a href='#getTimestamp'><p>getTimestamp</p></a></li>
<li><a href='#getTransmissionTree'><p>getTransmissionTree</p></a></li>
<li><a href='#Host'><p>Class Host</p></a></li>
<li><a href='#is.juliendate'><p>is.juliendate</p></a></li>
<li><a href='#is.StringDate'><p>is.StringDate</p></a></li>
<li><a href='#is.timestamp'><p>is.timestamp</p></a></li>
<li><a href='#isInCode'><p>isInCode</p></a></li>
<li><a href='#loadCoords'><p>loadCoords</p></a></li>
<li><a href='#loadCovs'><p>loadCovs</p></a></li>
<li><a href='#loadHost'><p>loadHost</p></a></li>
<li><a href='#loadStates'><p>loadStates</p></a></li>
<li><a href='#loadTree'><p>loadTree</p></a></li>
<li><a href='#loadTreeDF'><p>loadTreeDF</p></a></li>
<li><a href='#loadViralObs'><p>loadViralObs</p></a></li>
<li><a href='#loadViralPop'><p>loadViralPop</p></a></li>
<li><a href='#loadViralPopSet'><p>loadViralPopSet</p></a></li>
<li><a href='#mergeCode'><p>mergeCode</p></a></li>
<li><a href='#plotDiversity.pDistance'><p>plotDiversity.pDistance</p></a></li>
<li><a href='#plotDiversity.sfs'><p>plotDiversity.sfs</p></a></li>
<li><a href='#setStates'><p>setStates</p></a></li>
<li><a href='#simulateStates'><p>simulateStates</p></a></li>
<li><a href='#ViralPop'><p>Class ViralPop</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Structure and Manipulations Tool for Host and Viral
Population</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-06-14</td>
</tr>
<tr>
<td>Author:</td>
<td>Jean-Francois Rey [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jean-Francois Rey &lt;jean-francois.rey@inra.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical Methods for Inferring Transmissions of Infectious Diseases from deep sequencing data (SMITID).
  It allow sequence-space-time host and viral population data storage, indexation and querying.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE [expanded from: GPL (&ge; 2) | file LICENSE]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>BuildVignettes:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Biarch:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://informatique-mia.inra.fr/biosp/anr-smitid-project">https://informatique-mia.inra.fr/biosp/anr-smitid-project</a>,
<a href="https://gitlab.paca.inra.fr/SMITID/structR">https://gitlab.paca.inra.fr/SMITID/structR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://gitlab.paca.inra.fr/SMITID/structR/issues">https://gitlab.paca.inra.fr/SMITID/structR/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>methods, utils, grDevices (&ge; 3.0.0), graphics (&ge; 3.0.0), R
(&ge; 3.3.0)</td>
</tr>
<tr>
<td>DependsNote:</td>
<td>BioC (&gt;= 3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, sf (&ge; 0.6.3), stats (&ge; 3.0.2), Biostrings (&ge;
2.0.0)</td>
</tr>
<tr>
<td>ImportsNote:</td>
<td>BioC (&gt;= 3.0), Recommended: Biostrings</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.0)</td>
</tr>
<tr>
<td>Collate:</td>
<td>'Class-Host.R' 'Class-ViralPop.R' 'Methods-Host.R'
'Methods-ViralPop.R' 'Methods-time.R' 'SMITIDstruct.R' 'demo.R'
'diversity.R' 'index.R'</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-06-14 10:05:06 UTC; jfrey</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-06-14 11:30:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='SMITIDstruct-package'>Data Structure and Manipulation Tool for Host and Viral Population</h2><span id='topic+SMITIDstruct-package'></span><span id='topic+_PACKAGE'></span><span id='topic+SMITIDstruct'></span>

<h3>Description</h3>

<p>Statistical Methods for Inferring Transmissions of Infectious Diseases from deep sequencing data (SMITID).
It allow sequence-space-time host and viral population data storage, indexation and querying.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
         Package: </td><td style="text-align: left;"> SMITIDstruct</td>
</tr>
<tr>
 <td style="text-align: left;">
         Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
         Version: </td><td style="text-align: left;"> 0.0.5</td>
</tr>
<tr>
 <td style="text-align: left;">
         Date: </td><td style="text-align: left;"> 2019-06-14</td>
</tr>
<tr>
 <td style="text-align: left;">
         License: </td><td style="text-align: left;"> GPL (&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
         </td>
</tr>

</table>

<p>The SMITIDstruct package contains functions and methods for manipulating Host and Viral population genotico-space-time data.
</p>


<h3>Author(s)</h3>

<p>Jean-Francois Rey <a href="mailto:jean-francois.rey@inra.fr">jean-francois.rey@inra.fr</a>
</p>
<p>Maintainer: Jean-Francois Rey <a href="mailto:jean-francois.rey@inra.fr">jean-francois.rey@inra.fr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+demo.SMITIDstruct.run">demo.SMITIDstruct.run</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Run a simulation
library("SMITIDstruct")
demo.SMITIDstruct.run() 


</code></pre>

<hr>
<h2 id='addcode'>addcode</h2><span id='topic+addcode'></span>

<h3>Description</h3>

<p>add a code event to an another
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addcode(code, code.add)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addcode_+3A_code">code</code></td>
<td>
<p>an existing code</p>
</td></tr>
<tr><td><code id="addcode_+3A_code.add">code.add</code></td>
<td>
<p>the code to add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>merge of the two code
</p>

<hr>
<h2 id='addHost'>addHost</h2><span id='topic+addHost'></span>

<h3>Description</h3>

<p>add an Host to a HostSet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addHost(lhost, id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addHost_+3A_lhost">lhost</code></td>
<td>
<p>a hostSet Object</p>
</td></tr>
<tr><td><code id="addHost_+3A_id">id</code></td>
<td>
<p>a character of host ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a HostSet of host object with there ID
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lhost &lt;- list()
lhost &lt;- addHost(lhost,"42")

</code></pre>

<hr>
<h2 id='addIndex'>addIndex</h2><span id='topic+addIndex'></span>

<h3>Description</h3>

<p>add to an index a new eventcode
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addIndex(index, id_host, time, code)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addIndex_+3A_index">index</code></td>
<td>
<p>an index</p>
</td></tr>
<tr><td><code id="addIndex_+3A_id_host">id_host</code></td>
<td>
<p>an host index in HostSet</p>
</td></tr>
<tr><td><code id="addIndex_+3A_time">time</code></td>
<td>
<p>a time</p>
</td></tr>
<tr><td><code id="addIndex_+3A_code">code</code></td>
<td>
<p>an event code</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the index updated (add a row or update one)
</p>

<hr>
<h2 id='addViralObs'>addViralObs</h2><span id='topic+addViralObs'></span>

<h3>Description</h3>

<p>load Viral pop observation in Host object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addViralObs(lhost, lvpop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addViralObs_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="addViralObs_+3A_lvpop">lvpop</code></td>
<td>
<p>a ViralPopSet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lhost update with viral population observed
</p>

<hr>
<h2 id='alleleCount'>alleleCount</h2><span id='topic+alleleCount'></span>

<h3>Description</h3>

<p>count allele at each position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alleleCount(mat, seq.char = c("A", "T", "G", "C"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="alleleCount_+3A_mat">mat</code></td>
<td>
<p>a genomique seq list as matrix by row</p>
</td></tr>
<tr><td><code id="alleleCount_+3A_seq.char">seq.char</code></td>
<td>
<p>allele alphabet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix, each row as a unique seq and col as allele count by position
</p>

<hr>
<h2 id='concatViralPop'>concatViralPop</h2><span id='topic+concatViralPop'></span>

<h3>Description</h3>

<p>concat several Viral population in one ViralPop object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concatViralPop(lvpop, lid)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="concatViralPop_+3A_lvpop">lvpop</code></td>
<td>
<p>a ViralPop Set</p>
</td></tr>
<tr><td><code id="concatViralPop_+3A_lid">lid</code></td>
<td>
<p>vector of viralpop id to concat</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ViralPop object with ID concatenation from all IDs and time at 0.
</p>

<hr>
<h2 id='createAViralPop'>createAViralPop</h2><span id='topic+createAViralPop'></span>

<h3>Description</h3>

<p>Create a new ViralPop object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createAViralPop(host_id, obs_time, seq, id_seq = "seq_ID",
  seq_value = "seq", prop = "prop", compact = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createAViralPop_+3A_host_id">host_id</code></td>
<td>
<p>host ID which viral pop is observed</p>
</td></tr>
<tr><td><code id="createAViralPop_+3A_obs_time">obs_time</code></td>
<td>
<p>time of the observation (numeric or date)</p>
</td></tr>
<tr><td><code id="createAViralPop_+3A_seq">seq</code></td>
<td>
<p>a data.frame of sequences ID, sequences and counts</p>
</td></tr>
<tr><td><code id="createAViralPop_+3A_id_seq">id_seq</code></td>
<td>
<p>column name containing the sequences ID</p>
</td></tr>
<tr><td><code id="createAViralPop_+3A_seq_value">seq_value</code></td>
<td>
<p>column name containing the sequences</p>
</td></tr>
<tr><td><code id="createAViralPop_+3A_prop">prop</code></td>
<td>
<p>column name containing the count of each sequences</p>
</td></tr>
<tr><td><code id="createAViralPop_+3A_compact">compact</code></td>
<td>
<p>boolean, default FALSE, if TRUE will try group identicals sequences (not implemented yet)</p>
</td></tr>
</table>

<hr>
<h2 id='createHost'>createHost</h2><span id='topic+createHost'></span>

<h3>Description</h3>

<p>create a list of Host class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createHost(list_host)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createHost_+3A_list_host">list_host</code></td>
<td>
<p>a character vector of host ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a HostSet of host object with there ID
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lh &lt;- seq(1,30,1)
lhost &lt;- createHost(lh)

</code></pre>

<hr>
<h2 id='createIndex'>createIndex</h2><span id='topic+createIndex'></span>

<h3>Description</h3>

<p>create an index of time id_host and event code
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createIndex(hostlist)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createIndex_+3A_hostlist">hostlist</code></td>
<td>
<p>a Hostset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with TIME, ID_HOST and EVENTCODE as columns
</p>

<hr>
<h2 id='demo.SMITIDstruct.run'>demo.SMITIDstruct.run</h2><span id='topic+demo.SMITIDstruct.run'></span>

<h3>Description</h3>

<p>run a demo to load HostSet, ViralPopSet and index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo.SMITIDstruct.run()
</code></pre>

<hr>
<h2 id='diversity.pDistance'>diversity.pDistance</h2><span id='topic+diversity.pDistance'></span>

<h3>Description</h3>

<p>diversity calculation using Mean Pairwise Distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diversity.pDistance(vpop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diversity.pDistance_+3A_vpop">vpop</code></td>
<td>
<p>a ViralPop object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>result
</p>

<hr>
<h2 id='diversity.sfs'>diversity.sfs</h2><span id='topic+diversity.sfs'></span>

<h3>Description</h3>

<p>Allele frequency spectrum or Site frequency spectra : the distribution of alternative allele frequencies across all sites of genetic sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diversity.sfs(vpop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diversity.sfs_+3A_vpop">vpop</code></td>
<td>
<p>a viralPop class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the site frequency spectra
</p>

<hr>
<h2 id='getCov'>getCov</h2><span id='topic+getCov'></span>

<h3>Description</h3>

<p>get Host(s) covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCov(lhost, id = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCov_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="getCov_+3A_id">id</code></td>
<td>
<p>a vector of host id (default NA : all lhost)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame
</p>

<hr>
<h2 id='getDate'>getDate</h2><span id='topic+getDate'></span>

<h3>Description</h3>

<p>Converte timestamp to Date (string)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDate(time, format = "%Y-%m-%dT%H:%M:%S")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDate_+3A_time">time</code></td>
<td>
<p>a timestamp or vector of</p>
</td></tr>
<tr><td><code id="getDate_+3A_format">format</code></td>
<td>
<p>Date format output (default %Y-%m-%dT%H:%M:%S)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>time as string date
</p>

<hr>
<h2 id='getDiversity.pDistance'>getDiversity.pDistance</h2><span id='topic+getDiversity.pDistance'></span>

<h3>Description</h3>

<p>get pairwise distance of an host over viral population observated
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDiversity.pDistance(host, lvpop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDiversity.pDistance_+3A_host">host</code></td>
<td>
<p>an Host object</p>
</td></tr>
<tr><td><code id="getDiversity.pDistance_+3A_lvpop">lvpop</code></td>
<td>
<p>a ViralPopSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with col as time of observation and p_distance
</p>

<hr>
<h2 id='getDiversity.sfs'>getDiversity.sfs</h2><span id='topic+getDiversity.sfs'></span>

<h3>Description</h3>

<p>get Allele Frequency Spectrum or Site Frequency spectra for observated viral pop of an host
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDiversity.sfs(host, lvpop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDiversity.sfs_+3A_host">host</code></td>
<td>
<p>an Host object</p>
</td></tr>
<tr><td><code id="getDiversity.sfs_+3A_lvpop">lvpop</code></td>
<td>
<p>an ViralPopSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list indexed by time that contains allele.time and count
</p>

<hr>
<h2 id='getInfosByHostAndTime'>getInfosByHostAndTime</h2><span id='topic+getInfosByHostAndTime'></span>

<h3>Description</h3>

<p>get hosts informations, status, infectedby, coordinates and time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getInfosByHostAndTime(index, lhost)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getInfosByHostAndTime_+3A_index">index</code></td>
<td>
<p>an index</p>
</td></tr>
<tr><td><code id="getInfosByHostAndTime_+3A_lhost">lhost</code></td>
<td>
<p>a hosts list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with colnames (id, time, infectedby, status, probabilities, X ,Y)
</p>

<hr>
<h2 id='getStates'>getStates</h2><span id='topic+getStates'></span>

<h3>Description</h3>

<p>get Host(s) states
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getStates(lhost, id = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getStates_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="getStates_+3A_id">id</code></td>
<td>
<p>a vector of host id (default NA : all lhost)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame
</p>

<hr>
<h2 id='getTimeLine'>getTimeLine</h2><span id='topic+getTimeLine'></span>

<h3>Description</h3>

<p>get the time line of an host
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTimeLine(lhost, id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTimeLine_+3A_lhost">lhost</code></td>
<td>
<p>a hostSet</p>
</td></tr>
<tr><td><code id="getTimeLine_+3A_id">id</code></td>
<td>
<p>a host ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame
</p>

<hr>
<h2 id='getTimestamp'>getTimestamp</h2><span id='topic+getTimestamp'></span>

<h3>Description</h3>

<p>Get the timestamp of Date
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTimestamp(date, format = "%Y-%m-%dT%H:%M:%S")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTimestamp_+3A_date">date</code></td>
<td>
<p>a date (as string) or vector of</p>
</td></tr>
<tr><td><code id="getTimestamp_+3A_format">format</code></td>
<td>
<p>the date format (default %Y-%m-%dT%H:%M:%S)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>timestamp of the date(s)
</p>

<hr>
<h2 id='getTransmissionTree'>getTransmissionTree</h2><span id='topic+getTransmissionTree'></span>

<h3>Description</h3>

<p>get a transmission tree as a data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTransmissionTree(lhost, id = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getTransmissionTree_+3A_lhost">lhost</code></td>
<td>
<p>a hostSet</p>
</td></tr>
<tr><td><code id="getTransmissionTree_+3A_id">id</code></td>
<td>
<p>a vector of hosts ids (default NA : all host)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame as source|target|time in columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path = system.file("extdata", "data-simul/", package="SMITIDstruct")
lhost &lt;- list()
lhost &lt;- loadTree(lhost,paste(path,"/tree.txt",sep=''))
print(getTransmissionTree(lhost))
</code></pre>

<hr>
<h2 id='Host'>Class Host</h2><span id='topic+Host'></span><span id='topic+Host-class'></span>

<h3>Description</h3>

<p>Spatio-temporal information about Host.
</p>


<h3>Details</h3>

<p>Object can be created by calling ...
</p>
<p>rdname Host-class
</p>


<h3>Slots</h3>


<dl>
<dt><code>ID</code></dt><dd><p>Host identifier</p>
</dd>
<dt><code>coordinates</code></dt><dd><p>Host coordinates in time (as sf)</p>
</dd>
<dt><code>states</code></dt><dd><p>Host States/Status (dob, Inf...)</p>
</dd>
<dt><code>sources</code></dt><dd><p>data.frame of time and host id who infected this host</p>
</dd>
<dt><code>offsprings</code></dt><dd><p>data.frame of time and host id who has been contamined by this host</p>
</dd>
<dt><code>ID_V_POP</code></dt><dd><p>data.frame of time and index of Viral population Observation</p>
</dd>
<dt><code>covariates</code></dt><dd><p>data.frame of time, cavariate and value of this host.</p>
</dd>
</dl>

<hr>
<h2 id='is.juliendate'>is.juliendate</h2><span id='topic+is.juliendate'></span>

<h3>Description</h3>

<p>Chekc if a numeric is not a timestamp
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.juliendate(time)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.juliendate_+3A_time">time</code></td>
<td>
<p>a numeric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if time is a julien day, otherwise FALSE
</p>

<hr>
<h2 id='is.StringDate'>is.StringDate</h2><span id='topic+is.StringDate'></span>

<h3>Description</h3>

<p>Check if a string represent a date
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.StringDate(date)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.StringDate_+3A_date">date</code></td>
<td>
<p>a string or a vector of string (without NA)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if date contains date format
</p>

<hr>
<h2 id='is.timestamp'>is.timestamp</h2><span id='topic+is.timestamp'></span>

<h3>Description</h3>

<p>Check if a numeric represent a timestamp
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.timestamp(time)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.timestamp_+3A_time">time</code></td>
<td>
<p>a numeric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if time &gt;= 1971
</p>

<hr>
<h2 id='isInCode'>isInCode</h2><span id='topic+isInCode'></span>

<h3>Description</h3>

<p>check a code contains a specific code
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isInCode(code, thecode)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isInCode_+3A_code">code</code></td>
<td>
<p>list of code to test</p>
</td></tr>
<tr><td><code id="isInCode_+3A_thecode">thecode</code></td>
<td>
<p>the real code</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if code contain thecode otherwise FLASE
</p>

<hr>
<h2 id='loadCoords'>loadCoords</h2><span id='topic+loadCoords'></span>

<h3>Description</h3>

<p>Load Hosts states
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadCoords(lhost, dfCoords, id = "ID")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadCoords_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="loadCoords_+3A_dfcoords">dfCoords</code></td>
<td>
<p>a data.frame with host ID, time and longitude latitude values</p>
</td></tr>
<tr><td><code id="loadCoords_+3A_id">id</code></td>
<td>
<p>colname for host ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lhost updated
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path = system.file("extdata", "data-simul/", package="SMITIDstruct")
lhost &lt;- list()
lhost &lt;- loadTree(lhost,paste(path,"/tree.txt",sep=''))
coords &lt;- read.table(file=paste(path,"/hosts_coords.txt",sep=''), header=TRUE, check.names=FALSE)
lhost &lt;- loadCoords(lhost,coords)
</code></pre>

<hr>
<h2 id='loadCovs'>loadCovs</h2><span id='topic+loadCovs'></span>

<h3>Description</h3>

<p>Load Hosts covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadCovs(lhost, dfCovs, id = "ID", colCovs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadCovs_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="loadCovs_+3A_dfcovs">dfCovs</code></td>
<td>
<p>a data.frame with host ID in rows and covariates in columns</p>
</td></tr>
<tr><td><code id="loadCovs_+3A_id">id</code></td>
<td>
<p>colname for host ID</p>
</td></tr>
<tr><td><code id="loadCovs_+3A_colcovs">colCovs</code></td>
<td>
<p>colnames of covariates columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lhost updated with covariates
</p>

<hr>
<h2 id='loadHost'>loadHost</h2><span id='topic+loadHost'></span>

<h3>Description</h3>

<p>load host object from a file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadHost(file = "host.txt")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadHost_+3A_file">file</code></td>
<td>
<p>a file containing hosts data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of Host object (HostSet)
include Class-Host.R
</p>

<hr>
<h2 id='loadStates'>loadStates</h2><span id='topic+loadStates'></span>

<h3>Description</h3>

<p>Load Hosts states
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadStates(lhost, dfStates, id = "ID", colStates)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadStates_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="loadStates_+3A_dfstates">dfStates</code></td>
<td>
<p>a data.frame with host ID and states in columns and time as value</p>
</td></tr>
<tr><td><code id="loadStates_+3A_id">id</code></td>
<td>
<p>colname for host ID</p>
</td></tr>
<tr><td><code id="loadStates_+3A_colstates">colStates</code></td>
<td>
<p>colnames of States columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lhost updated
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path = system.file("extdata", "data-simul/", package="SMITIDstruct")
lhost &lt;- list()
class(lhost) &lt;- "hostSet"
lhost &lt;- loadTree(lhost,paste(path,"/tree.txt",sep='')) 
obs &lt;- read.table(paste(path,"/obs.txt",sep=''),header=TRUE, check.names=FALSE)
obs.states &lt;- c(colnames(obs[-grep("ID|Tobs.*",colnames(obs))]))
lhost &lt;- loadStates(lhost, obs, colStates=obs.states)
</code></pre>

<hr>
<h2 id='loadTree'>loadTree</h2><span id='topic+loadTree'></span>

<h3>Description</h3>

<p>load sources and offsprings from file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadTree(lhost = list(), file = "tree.txt", source = "ID-source",
  receptor = "ID-receptor", tinf = "Tinf", weight = "Weight")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadTree_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="loadTree_+3A_file">file</code></td>
<td>
<p>a file containing tree data</p>
</td></tr>
<tr><td><code id="loadTree_+3A_source">source</code></td>
<td>
<p>column name for source ID</p>
</td></tr>
<tr><td><code id="loadTree_+3A_receptor">receptor</code></td>
<td>
<p>column name for receptor ID</p>
</td></tr>
<tr><td><code id="loadTree_+3A_tinf">tinf</code></td>
<td>
<p>column name for infection Time</p>
</td></tr>
<tr><td><code id="loadTree_+3A_weight">weight</code></td>
<td>
<p>column name of infection weight</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the lhost param update with sources and offsprings
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path = system.file("extdata", "data-simul/", package="SMITIDstruct")
lhost &lt;- list()
class(lhost) &lt;- "hostSet"
lhost &lt;- loadTree(lhost,paste(path,"/tree.txt",sep=''))
</code></pre>

<hr>
<h2 id='loadTreeDF'>loadTreeDF</h2><span id='topic+loadTreeDF'></span>

<h3>Description</h3>

<p>load sources and offsprings from a data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadTreeDF(lhost = list(), df = data.frame(), source = "ID-source",
  receptor = "ID-receptor", tinf = "Tinf", weight = "Weight")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadTreeDF_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="loadTreeDF_+3A_df">df</code></td>
<td>
<p>a data.frame containing tree data</p>
</td></tr>
<tr><td><code id="loadTreeDF_+3A_source">source</code></td>
<td>
<p>column name for source ID</p>
</td></tr>
<tr><td><code id="loadTreeDF_+3A_receptor">receptor</code></td>
<td>
<p>column name for receptor ID</p>
</td></tr>
<tr><td><code id="loadTreeDF_+3A_tinf">tinf</code></td>
<td>
<p>column name for infection Time</p>
</td></tr>
<tr><td><code id="loadTreeDF_+3A_weight">weight</code></td>
<td>
<p>infection links probability</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the lhost param update with sources and offsprings
</p>

<hr>
<h2 id='loadViralObs'>loadViralObs</h2><span id='topic+loadViralObs'></span>

<h3>Description</h3>

<p>load a ViralPop object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadViralObs(id, time, file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadViralObs_+3A_id">id</code></td>
<td>
<p>host pathogen ID</p>
</td></tr>
<tr><td><code id="loadViralObs_+3A_time">time</code></td>
<td>
<p>time of the observation (numeric or Date)</p>
</td></tr>
<tr><td><code id="loadViralObs_+3A_file">file</code></td>
<td>
<p>a fasta file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a new ViralPop object
</p>

<hr>
<h2 id='loadViralPop'>loadViralPop</h2><span id='topic+loadViralPop'></span>

<h3>Description</h3>

<p>Load all ViralPop observated in the file.obs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadViralPop(directory, listFiles, listCol = list(id = "id", timeObs =
  "time", filename = "filename"), file.extension = "fasta")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadViralPop_+3A_directory">directory</code></td>
<td>
<p>path where is data</p>
</td></tr>
<tr><td><code id="loadViralPop_+3A_listfiles">listFiles</code></td>
<td>
<p>a dataframe with host ID, time observation and file name (filename.fasta)</p>
</td></tr>
<tr><td><code id="loadViralPop_+3A_listcol">listCol</code></td>
<td>
<p>a list of listFiles colomns names (&quot;id&quot;, &quot;timeObs&quot;, &quot;filename&quot;)</p>
</td></tr>
<tr><td><code id="loadViralPop_+3A_file.extension">file.extension</code></td>
<td>
<p>genotype file extension</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of VirlaPop object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
path = system.file("extdata", "data-simul/", package="SMITIDstruct")
files &lt;- list.files(path, pattern = ".*.fasta" ,full.names=FALSE)
lfileinfo &lt;- sapply(files,function(x){return(substr(x,1,nchar(x)-6))})
splitFiles &lt;- strsplit(lfileinfo, "_");
listF &lt;- cbind(data.frame(matrix(unlist(splitFiles),nrow=length(splitFiles), byrow=TRUE),
               stringsAsFactors = FALSE), names(splitFiles))
colnames(listF) &lt;- c("id", "time", "filename")
lvpop &lt;- loadViralPop(path,listF)

</code></pre>

<hr>
<h2 id='loadViralPopSet'>loadViralPopSet</h2><span id='topic+loadViralPopSet'></span>

<h3>Description</h3>

<p>load a list of viral populations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadViralPopSet(lvpop = list(), list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadViralPopSet_+3A_lvpop">lvpop</code></td>
<td>
<p>a viralPopSet (default new one)</p>
</td></tr>
<tr><td><code id="loadViralPopSet_+3A_list">list</code></td>
<td>
<p>a list (see details)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The list have to be on this format:  
list$HOST_ID$TIME$list$seq_id
$seq
$prop
A list indexed by host ID, follow by a list indexed by time (of observation).
The last list contains an array of seq_ID (sequence ID), an array of seq (sequence as characters), and an array of the count of seq.
example : 
$'HOST_42'$'2014-01-01T00:00:00'$seq_ID [&quot;SEQ_1&quot;,&quot;SEQ_2&quot;]
$'HOST_42'$'2014-01-01T00:00:00'$seq [&quot;ACGT&quot;,&quot;TGCA&quot;]
$'HOST_42'$'2014-01-01T00:00:00'$seq_ID [&quot;46&quot;,&quot;6&quot;]
</p>

<hr>
<h2 id='mergeCode'>mergeCode</h2><span id='topic+mergeCode'></span>

<h3>Description</h3>

<p>merge a list of event code
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeCode(listcode)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mergeCode_+3A_listcode">listcode</code></td>
<td>
<p>a list of event code*</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a code
</p>

<hr>
<h2 id='plotDiversity.pDistance'>plotDiversity.pDistance</h2><span id='topic+plotDiversity.pDistance'></span>

<h3>Description</h3>

<p>plot Mean Pairwise Distance for an host viralpop over time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDiversity.pDistance(host, lvpop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotDiversity.pDistance_+3A_host">host</code></td>
<td>
<p>an Host object</p>
</td></tr>
<tr><td><code id="plotDiversity.pDistance_+3A_lvpop">lvpop</code></td>
<td>
<p>a ViralPopSet object</p>
</td></tr>
</table>

<hr>
<h2 id='plotDiversity.sfs'>plotDiversity.sfs</h2><span id='topic+plotDiversity.sfs'></span>

<h3>Description</h3>

<p>plot Allele frequency spetrum for an host viralpop over time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDiversity.sfs(host, lvpop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotDiversity.sfs_+3A_host">host</code></td>
<td>
<p>an Host object</p>
</td></tr>
<tr><td><code id="plotDiversity.sfs_+3A_lvpop">lvpop</code></td>
<td>
<p>an ViralPopSet object</p>
</td></tr>
</table>

<hr>
<h2 id='setStates'>setStates</h2><span id='topic+setStates'></span>

<h3>Description</h3>

<p>set hosts states from a data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setStates(lhost, dfStates, colStates = c(id = "ID", time = "time", states
  = "value"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setStates_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
<tr><td><code id="setStates_+3A_dfstates">dfStates</code></td>
<td>
<p>a data.frame with host ID and states and time in columns</p>
</td></tr>
<tr><td><code id="setStates_+3A_colstates">colStates</code></td>
<td>
<p>vector of the columns name, id, time and states</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the HostSet updated
</p>

<hr>
<h2 id='simulateStates'>simulateStates</h2><span id='topic+simulateStates'></span>

<h3>Description</h3>

<p>simulate states from sources infection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateStates(lhost)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulateStates_+3A_lhost">lhost</code></td>
<td>
<p>a HostSet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>lhost update with states from sources time ~
</p>

<hr>
<h2 id='ViralPop'>Class ViralPop</h2><span id='topic+ViralPop'></span><span id='topic+ViralPop-class'></span>

<h3>Description</h3>

<p>Viral population data containing genotypes
</p>


<h3>Slots</h3>


<dl>
<dt><code>ID</code></dt><dd><p>Host identifier</p>
</dd>
<dt><code>time</code></dt><dd><p>Observation time as numeric since 1970/01/01</p>
</dd>
<dt><code>size</code></dt><dd><p>Qt of variants</p>
</dd>
<dt><code>names</code></dt><dd><p>list of variants id with same sequence</p>
</dd>
<dt><code>genotypes</code></dt><dd><p>all variants genotypes (as DNAStringSet)</p>
</dd>
<dt><code>proportions</code></dt><dd><p>proportions of each variants</p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
