<!DOCTYPE html><html lang="en"><head><title>Help for package SMLE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SMLE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SMLE-package'>
<p>Joint SMLE-screening for generalized linear models</p></a></li>
<li><a href='#coef.smle'><p>Extract coefficients from fitted model</p></a></li>
<li><a href='#Gen_Data'><p>Data simulator for high-dimensional GLMs</p></a></li>
<li><a href='#logLik.smle'><p>Extract log-likelihood</p></a></li>
<li><a href='#plot.selection'><p>Plots to visualize the post-screening selection</p></a></li>
<li><a href='#plot.smle'><p>Plots to visualize SMLE screening</p></a></li>
<li><a href='#predict.smle'><p>Prediction based on SMLE screening and selection</p></a></li>
<li><a href='#print.smle'><p>Print an object</p></a></li>
<li><a href='#pvals'><p>p values of synthetic genetic association study data set</p></a></li>
<li><a href='#SMLE'><p>Joint feature screening via sparse maximum likelihood estimation for GLMs</p></a></li>
<li><a href='#smle_select'><p>Elaborative post-screening selection with SMLE</p></a></li>
<li><a href='#summary.smle'><p>Summarize SMLE-screening and selection</p></a></li>
<li><a href='#synSNP'><p>Synthetic genetic association study data set</p></a></li>
<li><a href='#vote_update'><p>Extract and adjust voting from SMLE selection</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Joint Feature Screening via Sparse MLE</td>
</tr>
<tr>
<td>Version:</td>
<td>2.2-2</td>
</tr>
<tr>
<td>Description:</td>
<td>Feature screening is a powerful tool in processing ultrahigh dimensional data. It attempts to screen out most irrelevant features in preparation for a more elaborate analysis. Xu and Chen (2014)&lt;<a href="https://doi.org/10.1080%2F01621459.2013.879531">doi:10.1080/01621459.2013.879531</a>&gt; proposed an effective screening method SMLE, which naturally incorporates the joint effects among features in the screening process. This package provides an efficient implementation of SMLE-screening for high-dimensional linear, logistic, and Poisson models. The package also provides a function for conducting accurate post-screening feature selection based on an iterative hard-thresholding procedure and a user-specified selection criterion.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet, matrixcalc, mvnfast</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Qianxiang Zang [aut, cre],
  Chen Xu [aut],
  Kelly Burkett [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qianxiang Zang &lt;SMLEmaintainer@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-28 22:31:07 UTC; mac</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-29 00:10:06 UTC</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
</table>
<hr>
<h2 id='SMLE-package'>
Joint SMLE-screening for generalized linear models
</h2><span id='topic+SMLE-package'></span>

<h3>Description</h3>

<p>Feature screening is a powerful tool in processing ultrahigh dimensional data. It attempts to screen
out most irrelevant features in preparation for a more elaborate analysis. This package provides an efficient implementation of SMLE-screening for linear, logistic, and Poisson models, where the joint effects among features are naturally incorporated in the screening process. The package also provides a function for conducting accurate post-screening feature selection based on an iterative hard-thresholding procedure and a user-specified selection criterion.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> smle</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.1-1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-02-12</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Input a <code class="reqn">n \times 1</code> response vector Y and a <code class="reqn">n \times p</code> predictor (feature) matrix X. The package outputs a set of <code class="reqn">k &lt; n</code> features that seem to be most relevant for joint regression. Moreover, the package provides a data simulator that generates synthetic datasets from high-dimensional GLMs, which accommodate both numerical and categorical features with commonly used correlation structures. <br />
</p>
<p>Key functions:<br />
<code>Gen_Data</code><br />
<code>SMLE</code><br />
<code>smle_select</code><br />
</p>


<h3>Author(s)</h3>

<p>Qianxiang Zang, Chen Xu, Kelly Burkett<br />
Maintainer: Qianxiang Zang  &lt;qzang023@uottawa.ca&gt;
</p>


<h3>References</h3>

<p>Xu, C. and Chen, J. (2014)
The Sparse MLE for Ultrahigh-Dimensional Feature Screening
<em>Journal of the American Statistical Association</em>, <b>109</b>(507), 1257â€“1269.<br />
</p>
<p>Friedman, J., Hastie, T. and Tibshirani, R. (2010)
Regularization Paths for Generalized Linear Models via Coordinate
Descent
<em>Journal of Statistical Software</em>, <b>33</b>(1), 1-22.<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
#Generate correlated data
Data &lt;- Gen_Data(n = 200, p = 5000, correlation = "MA",family = "gaussian")
print(Data)

# joint feature screening via SMLE
fit &lt;- SMLE(Y = Data$Y, X = Data$X, k = 10, family = "gaussian")
print(fit)
summary(fit)
plot(fit)

#Are there any features missed after screening?
setdiff(Data$subset_true, fit$ID_retained)

# Elaborative selection after screening
fit_s &lt;- smle_select(fit, gamma_ebic = 0.5, vote = FALSE)

#Are there any features missed after selection? 
setdiff(Data$subset_true, fit_s$ID_selected)
print(fit_s)
summary(fit_s)
plot(fit_s)
</code></pre>

<hr>
<h2 id='coef.smle'>Extract coefficients from fitted model</h2><span id='topic+coef.smle'></span><span id='topic+coef.selection'></span>

<h3>Description</h3>

<p>Extract coefficients from fitted model for either a <code>'smle'</code> or <code>'selection'</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'smle'
coef(object, refit = TRUE, ...)

## S3 method for class 'selection'
coef(object, refit = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coef.smle_+3A_object">object</code></td>
<td>
<p>Returned object from either the function <code><a href="#topic+SMLE">SMLE</a>()</code> or <code><a href="#topic+smle_select">smle_select</a>()</code>.</p>
</td></tr>
<tr><td><code id="coef.smle_+3A_refit">refit</code></td>
<td>
<p>A logical flag that controls what coefficients are being return. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="coef.smle_+3A_...">...</code></td>
<td>
<p>This argument is not used and listed for method consistency.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fitted coefficients based on the screened or selected model specified in the object. 
If <code>refit = TRUE</code>, the coefficients are estimated by re-fitting the final 
screened/selected model with <code><a href="stats.html#topic+glm">glm</a>()</code>. If <code>refit = FALSE</code> the coefficients estimated by the IHT algorithm are returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)
Data&lt;-Gen_Data(n=100, p=5000, family = "gaussian", correlation="ID")
fit&lt;-SMLE(Y = Data$Y, X = Data$X, k=15, family = "gaussian")
coef(fit)
fit_s&lt;-smle_select(fit)
coef(fit_s)
</code></pre>

<hr>
<h2 id='Gen_Data'>Data simulator for high-dimensional GLMs</h2><span id='topic+Gen_Data'></span>

<h3>Description</h3>

<p>This function generates synthetic datasets from GLMs with a user-specified correlation structure.
It permits both numerical and categorical features, whose quantity can be larger than the sample size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gen_Data(
  n = 200,
  p = 1000,
  sigma = 1,
  num_ctgidx = NULL,
  pos_ctgidx = NULL,
  num_truecoef = NULL,
  pos_truecoef = NULL,
  level_ctgidx = NULL,
  effect_truecoef = NULL,
  correlation = c("ID", "AR", "MA", "CS"),
  rho = 0.2,
  family = c("gaussian", "binomial", "poisson")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Gen_Data_+3A_n">n</code></td>
<td>
<p>Sample size, number of rows for the feature matrix to be generated.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_p">p</code></td>
<td>
<p>Number of columns for the feature matrix to be generated.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_sigma">sigma</code></td>
<td>
<p>Parameter for noise level.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_num_ctgidx">num_ctgidx</code></td>
<td>
<p>The number of features that are categorical. Set to <code>FALSE</code> for only numerical features. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_pos_ctgidx">pos_ctgidx</code></td>
<td>
<p>Vector of indices denoting which columns are categorical.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_num_truecoef">num_truecoef</code></td>
<td>
<p>The number of features (columns) that affect response. Default is 5.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_pos_truecoef">pos_truecoef</code></td>
<td>
<p>Vector of indices denoting which features (columns) affect the response variable. 
If not specified, positions are randomly sampled. See Details for more information.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_level_ctgidx">level_ctgidx</code></td>
<td>
<p>Vector to indicate the number of levels for the categorical features in <code>pos_ctgidx</code>. Default is 2.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_effect_truecoef">effect_truecoef</code></td>
<td>
<p>Effect size corresponding to the features in <code>pos_truecoef</code>. 
If not specified, effect size is sampled based on a uniform distribution and direction is randomly sampled. See Details.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_correlation">correlation</code></td>
<td>
<p>Correlation structure among features. <code>correlation = "ID"</code> for independent,
<code>correlation = 'MA'</code> for moving average, <code>correlation = "CS"</code> for compound symmetry, <code>correlation = "AR"</code> 
for auto regressive. Default is <code>"ID"</code>. For more information see Details.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_rho">rho</code></td>
<td>
<p>Parameter controlling the correlation strength, default is <code>0.2</code>. See Details.</p>
</td></tr>
<tr><td><code id="Gen_Data_+3A_family">family</code></td>
<td>
<p>Model type for the response variable.
<code>"gaussian"</code> for normally distributed data, <code>poisson</code> for non-negative counts,
<code>"binomial"</code> for binary (0-1).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simulated data <code class="reqn">(y_i , x_i)</code> where <code class="reqn"> x_i = (x_{i1},x_{i2} , . . . , x_{ip})</code> are generated as follows:
First, we generate a <code class="reqn">p</code> by <code class="reqn">1</code> model coefficient vector beta with all 
entries being zero, except for the positions specified in <code>pos_truecoef</code>,
on which <code>effect_truecoef</code> is used. When <code>pos_truecoef</code> is not specified, 
we randomly choose <code>num_truecoef</code> positions from the coefficient
vector. When <code>effect_truecoef</code> is not specified, we randomly set the strength 
of the true model coefficients as follow:
</p>
<p style="text-align: center;"><code class="reqn">(0.5+U) Z,</code>
</p>

<p>where <code class="reqn">U</code> is sampled from a uniform distribution from 0 to 1,  
and <code class="reqn">Z</code> is sampled from a binomial distribution <code class="reqn">P(Z=1)=1/2,P(Z=-1)=1/2</code>.
</p>
<p>Next, we generate a <code class="reqn">n</code> by <code class="reqn">p</code> feature matrix <code class="reqn">X</code> according to the model selected with
<code>correlation</code> and specified as follows.
</p>
<p>Independent (ID):  all features are independently generated from <code class="reqn">N( 0, 1)</code>.
</p>
<p>Moving average (MA): candidate features <code class="reqn">x_1,..., x_p</code> are joint normal,
marginally <code class="reqn">N( 0, 1)</code>, with
</p>
<p><code class="reqn">cov(x_j, x_{j-1}) = \rho</code>, <code class="reqn">cov(x_j, x_{j-2}) = \rho/2</code> and <code class="reqn">cov(x_j, x_h) = 0</code> for <code class="reqn">|j-h|&gt;3</code>.
</p>
<p>Compound symmetry (CS): candidate features <code class="reqn">x_1,..., x_p</code> are joint normal,
marginally <code class="reqn">N( 0, 1)</code>, with <code class="reqn">cov(x_j, x_h) =\rho/2</code> if <code class="reqn">j</code>, <code class="reqn">h</code>
are both in the set of important features and <code class="reqn">cov(x_j, x_h)=\rho</code> when only
one of <code class="reqn">j</code> or <code class="reqn">h</code> are in the set of important features.
</p>
<p>Auto-regressive (AR): candidate features <code class="reqn">x_1,..., x_p</code> are joint normal, marginally <code class="reqn">N( 0, 1)</code>, with
</p>
<p><code class="reqn">cov(x_j, x_h) = \rho^{|j-h|}</code> for all <code class="reqn">j</code> and <code class="reqn">h</code>. The correlation strength <code class="reqn">\rho</code> is controlled by the argument <code>rho</code>. 
</p>
<p>Then, we generate the response variable <code class="reqn">Y</code> according to its response type, which is controlled by the argument <code>family</code>
For the Gaussian model, <code class="reqn">y_i =x_i\beta + \epsilon_i</code> where <code class="reqn">\epsilon_i</code> is <code class="reqn">N( 0, 1)</code> for <code class="reqn">i</code> from <code class="reqn">1</code> to <code class="reqn">n</code>. 
For the binary model let <code class="reqn">\pi_i = P(Y = 1|x_i)</code>. We sample <code class="reqn">y_i</code> from Bernoulli(<code class="reqn">\pi_i</code>) where logit<code class="reqn">(\pi_i) = x_i \beta</code>.
Finally, for the Poisson model, <code class="reqn">y_i</code> is generated from the Poisson distribution with the link <code class="reqn">\pi_i</code> = exp<code class="reqn">(x_i\beta )</code>.
For more details see the reference below.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Response variable vector of length <code class="reqn">n</code>.</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>Feature matrix or data.frame (matrix if <code>num_ctgidx =FALSE</code> and data.frame otherwise).</p>
</td></tr>
<tr><td><code>subset_true</code></td>
<td>
<p>Vector of column indices of X for the features that affect the response variables (relevant features).</p>
</td></tr>
<tr><td><code>coef_true</code></td>
<td>
<p>Vector of effects for the features that affect the response variables.</p>
</td></tr>
<tr><td><code>categorical</code></td>
<td>
<p>Logical flag whether the model contains categorical features.</p>
</td></tr>
<tr><td><code>CI</code></td>
<td>
<p>Indices of categorical features when <code>categorical = TRUE</code>.</p>
</td></tr>
</table>
<p>rho,family,correlation are return of arguments passed in the function call.
</p>


<h3>References</h3>

<p>Xu, C. and Chen, J. (2014). The Sparse MLE for Ultrahigh-Dimensional Feature
Screening, <em>Journal of the American Statistical Association</em>, <b>109</b>(507), 1257-1269
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Simulating data with binomial response and auto-regressive structure.
set.seed(1)
Data &lt;- Gen_Data(n = 500, p = 2000, family = "binomial", correlation = "AR")
cor(Data$X[,1:5])
print(Data)


</code></pre>

<hr>
<h2 id='logLik.smle'>Extract log-likelihood</h2><span id='topic+logLik.smle'></span><span id='topic+logLik.selection'></span>

<h3>Description</h3>

<p>This is a method written to extract the log-likelihood from <code>'smle'</code> and <code>'selection'</code> objects. 
It refits the model by <code><a href="stats.html#topic+glm">glm</a>()</code> based on the response and the features selected after screening or selection, 
and returns an object of <code>'logLik'</code> from the generic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'smle'
logLik(object, ...)

## S3 method for class 'selection'
logLik(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="logLik.smle_+3A_object">object</code></td>
<td>
<p>An object of class <code>'smle'</code> or <code>'sdata'</code>.</p>
</td></tr>
<tr><td><code id="logLik.smle_+3A_...">...</code></td>
<td>
<p>Forwarded arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>'logLik'</code>. This is a number with at least one attribute,
<code>"df"</code> (degrees of freedom), giving the number of (estimated) parameters in the model. For more details, see the generic <code><a href="stats.html#topic+logLik">logLik</a>()</code> in <span class="pkg">stats</span>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
Data&lt;-Gen_Data(n=100, p=5000, family = "gaussian", correlation="ID")
fit&lt;-SMLE(Y=Data$Y, X=Data$X, k=9, family = "gaussian")
logLik(fit)

</code></pre>

<hr>
<h2 id='plot.selection'>Plots to visualize the post-screening selection</h2><span id='topic+plot.selection'></span>

<h3>Description</h3>

<p>This function constructs a sparsity vs. selection criterion curve for a <code>'selection'</code> object.
When EBIC is used with voting, it also constructs a histogram showing the voting result.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'selection'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.selection_+3A_x">x</code></td>
<td>
<p>A <code>'selection'</code> object as the output from <code><a href="#topic+smle_select">smle_select</a>()</code>.</p>
</td></tr>
<tr><td><code id="plot.selection_+3A_...">...</code></td>
<td>
<p>Additional arguments to the <code><a href="base.html#topic+plot">plot</a>()</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
Data &lt;- Gen_Data(correlation = "MA", family = "gaussian")
fit &lt;- SMLE(Y = Data$Y, X = Data$X, k = 20, family = "gaussian")
fit_s &lt;- smle_select(fit, vote = TRUE)
plot(fit_s)

</code></pre>

<hr>
<h2 id='plot.smle'>Plots to visualize SMLE screening</h2><span id='topic+plot.smle'></span>

<h3>Description</h3>

<p>This function returns two plot windows. By default, the first shows 
1) the solution path (estimated coefficient by iteration step) for 
the retained features.
By default, the second plot contains 4 plots to assess convergence:
2) log-likelihood,  
3) Euclidean distance between the current and the previous coefficient estimates,   
4) the number of tries in u-search (see details of <code><a href="#topic+SMLE">SMLE</a>()</code>),  
and 5) the number of features changed in the current active set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'smle'
plot(x, num_path = NULL, label = TRUE, which_path = NULL, out_plot = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.smle_+3A_x">x</code></td>
<td>
<p>A <code>'smle'</code> object as the output from <code><a href="#topic+SMLE">SMLE</a>()</code>.</p>
</td></tr>
<tr><td><code id="plot.smle_+3A_num_path">num_path</code></td>
<td>
<p>The number of top coefficients to be shown.
Default is equal to the number of features retained in the model.</p>
</td></tr>
<tr><td><code id="plot.smle_+3A_label">label</code></td>
<td>
<p>Logical flag for whether to label each curve with the feature index. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot.smle_+3A_which_path">which_path</code></td>
<td>
<p>A vector to control which features are shown in addition to the paths for the most significant coefficients.</p>
</td></tr>
<tr><td><code id="plot.smle_+3A_out_plot">out_plot</code></td>
<td>
<p>A number from 1 to 5 indicating which plot is to be shown in the separate window; the default for solution path plot is &quot;1&quot;.
See Description for plot labels 2-5.</p>
</td></tr>
<tr><td><code id="plot.smle_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the second plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)
Data &lt;- Gen_Data(correlation = "CS")
fit &lt;- SMLE(Y = Data$Y,X = Data$X, k = 20, family = "gaussian")
plot(fit)

</code></pre>

<hr>
<h2 id='predict.smle'>Prediction based on SMLE screening and selection</h2><span id='topic+predict.smle'></span><span id='topic+predict.selection'></span>

<h3>Description</h3>

<p>For a model object of class <code>'smle'</code> or <code>'selection'</code>, this function returns the predicted response values after re-fitting the model using <code><a href="stats.html#topic+glm">glm</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'smle'
predict(object, newdata = NULL, type = c("link", "response", "terms"), ...)

## S3 method for class 'selection'
predict(object, newdata = NULL, type = c("link", "response", "terms"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.smle_+3A_object">object</code></td>
<td>
<p>A <code>'smle'</code> or <code>'selection'</code> object.</p>
</td></tr>
<tr><td><code id="predict.smle_+3A_newdata">newdata</code></td>
<td>
<p>Matrix of new values for the features at which predictions are to be made. If omitted, the fitted linear predictors are used.</p>
</td></tr>
<tr><td><code id="predict.smle_+3A_type">type</code></td>
<td>
<p>The type of prediction required by <code><a href="stats.html#topic+predict.glm">predict.glm</a>()</code>.</p>
</td></tr>
<tr><td><code id="predict.smle_+3A_...">...</code></td>
<td>
<p>Further arguments passed to <code><a href="stats.html#topic+predict.glm">predict.glm</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A prediction vector with length equal to the number of rows of <code>newdata</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)
Data_sim &lt;- Gen_Data(n = 420, p = 1000, sigma = 0.5, family = "gaussian")
train_X &lt;- Data_sim$X[1:400,]; test_X &lt;- Data_sim$X[401:420,]
train_Y &lt;- Data_sim$Y[1:400]; test_Y &lt;- Data_sim$Y[401:420]
fit1 &lt;- SMLE(Y = train_Y, X = train_X, family = "gaussian", k = 10)

#Fitted responses vs true responses in training data
predict(fit1)[1:10]
train_Y[1:10]

#Predicted responses vs true responses in testing data
predict(fit1, newdata = test_X)
test_Y



</code></pre>

<hr>
<h2 id='print.smle'>Print an object</h2><span id='topic+print.smle'></span><span id='topic+print.selection'></span><span id='topic+print.summary.smle'></span><span id='topic+print.summary.selection'></span><span id='topic+print.sdata'></span>

<h3>Description</h3>

<p>This function prints information about the fitted model from a call to <code><a href="#topic+SMLE">SMLE</a>()</code> or <code><a href="#topic+smle_select">smle_select</a>()</code>,
or about the simulated data from a call to <code><a href="#topic+Gen_Data">Gen_Data</a>()</code>. The object passed as an argument to print is returned invisibly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'smle'
print(x, ...)

## S3 method for class 'selection'
print(x, ...)

## S3 method for class 'summary.smle'
print(x, ...)

## S3 method for class 'summary.selection'
print(x, ...)

## S3 method for class 'sdata'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.smle_+3A_x">x</code></td>
<td>
<p>Fitted object.</p>
</td></tr>
<tr><td><code id="print.smle_+3A_...">...</code></td>
<td>
<p>This argument is not used and listed for method consistency.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return argument invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
Data&lt;-Gen_Data(correlation = "MA", family = "gaussian")
Data
fit&lt;-SMLE(Y = Data$Y, X = Data$X, k = 20, family = "gaussian")
print(fit)
summary(fit)
</code></pre>

<hr>
<h2 id='pvals'>p values of synthetic genetic association study data set</h2><span id='topic+pvals'></span>

<h3>Description</h3>

<p>The first column is the chromosome number. The second columns is SNP name. 
The third column is the genomic position of the SNP on the whole data set.
The marginal p-values of each SNPs is pre-calculated and saved in the fourth column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pvals)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 10031 rows and 4 columns.
</p>

<hr>
<h2 id='SMLE'>Joint feature screening via sparse maximum likelihood estimation for GLMs</h2><span id='topic+SMLE'></span><span id='topic+SMLE.default'></span><span id='topic+SMLE.formula'></span>

<h3>Description</h3>

<p>Input a <code class="reqn">n</code> by <code class="reqn">1</code> response <code class="reqn">Y</code> and a <code class="reqn">n</code> by <code class="reqn">p</code> feature matrix <code class="reqn">X</code>;
the function uses SMLE to retain only a set of <code class="reqn">k&lt;n</code> features that seem
to be most related to the response variable. It thus serves as a pre-processing step for an
elaborative analysis. In SMLE, the joint effects between features are naturally
accounted for; this makes the screening more reliable. The function uses the
efficient iterative hard thresholding (IHT) algorithm with step parameter
adaptively tuned for fast convergence. Users can choose to further conduct
an elaborative selection after SMLE-screening. See <code><a href="#topic+smle_select">smle_select</a>()</code> for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMLE(formula = NULL, ...)

## Default S3 method:
SMLE(
  formula = NULL,
  X = NULL,
  Y = NULL,
  data = NULL,
  k = NULL,
  family = c("gaussian", "binomial", "poisson"),
  keyset = NULL,
  intercept = TRUE,
  categorical = TRUE,
  group = TRUE,
  codingtype = NULL,
  coef_initial = NULL,
  max_iter = 500,
  tol = 10^(-3),
  selection = F,
  standardize = TRUE,
  fast = FALSE,
  U = 1,
  U_rate = 0.5,
  penalize_mod = TRUE,
  ...
)

## S3 method for class 'formula'
SMLE(formula, data, k = NULL, keyset = NULL, categorical = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SMLE_+3A_formula">formula</code></td>
<td>
<p>An object of class <code>'formula'</code> (or one that can be coerced to that class): a symbolic description of the model to be fitted. It should be <code>NULL</code> when <code>X</code> and <code>Y</code> are used.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to <code><a href="#topic+smle_select">smle_select</a>()</code> if <code>selection = TRUE</code>. 
See <code><a href="#topic+smle_select">smle_select</a>()</code> documentation for more details.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_x">X</code></td>
<td>
<p>The <code class="reqn">n</code> by <code class="reqn">p</code> feature matrix <code class="reqn">X</code> with each column denoting a feature
(covariate) and each row denoting an observation vector. The input should be
a <code>'matrix'</code> object for numerical data, and <code>'data.frame'</code> for categorical
data (or a mixture of numerical and categorical data). The algorithm will
treat covariates having class <code>'factor'</code> as categorical data and extend the data
frame dimension by the dummy columns needed for coding the categorical features.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_y">Y</code></td>
<td>
<p>The response vector <code class="reqn">Y</code> of dimension <code class="reqn">n</code> by <code class="reqn">1</code>. Quantitative for
<code>family = "gaussian"</code>, non-negative counts for <code>family = "poisson"</code>,
binary (0-1) for <code>family = "binomial"</code>. Input <code class="reqn">Y</code> should be <code>'numeric'</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_data">data</code></td>
<td>
<p>An optional data frame, list or environment (or object coercible by <code><a href="base.html#topic+as.data.frame">as.data.frame</a>()</code> to a <code>'data.frame'</code>) containing the features in the model. It is required if <code>'formula'</code> is used.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_k">k</code></td>
<td>
<p>Total number of features (including <code>keyset</code>) to be retained 
after screening. Default is the largest integer not 
exceeding <code class="reqn">0.5</code>log<code class="reqn">(n) n^{1/3}</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_family">family</code></td>
<td>
<p>Model assumption between <code class="reqn">Y</code> and <code class="reqn">X</code>; 
either a character string representing one of the built-in families, or else a glm() family object. 
The default model is Gaussian linear.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_keyset">keyset</code></td>
<td>
<p>A numeric vector with column indices for the key features that 
do not participate in feature screening and are forced to remain in the model.
The column indices for the key features should be from <code>data</code> if <code>'formula'</code> is used
or in <code>X</code> if <code>X</code> and <code>Y</code> are provided. The class of <code>keyset</code> 
can be <code>'numeric'</code>,<code>'integer'</code> or <code>'character'</code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_intercept">intercept</code></td>
<td>
<p>A logical flag to indicate whether to an intercept be used in
the model. An intercept will not participate in screening.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_categorical">categorical</code></td>
<td>
<p>A logical flag for whether the input feature matrix includes 
categorical features( either <code>'factor'</code> or <code>'character'</code>). <code>FALSE</code> 
treats all features as numerical and not check for whether there are categorical features; 
<code>TRUE</code> treats the data as having some categorical features and the algorithm 
determines which columns contain the categorical features. If all features are 
known to be numerical, it will be faster to run SMLE with this argument set to 
<code>FALSE</code>. we will need to find which columns are the categorical features.
Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_group">group</code></td>
<td>
<p>Logical flag for whether to treat the dummy covariates of a
categorical feature as a group. (Only for categorical data, see Details).
Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_codingtype">codingtype</code></td>
<td>
<p>Coding types for categorical features; default is <code>"DV"</code>.
<code>codingtype = "all"</code> convert each level to a 0-1 vector.
<code>codingtype = "DV"</code> conducts deviation coding for each level in
comparison with the grand mean.
<code>codingtype = "standard"</code> conducts standard dummy coding for each level
in comparison with the reference level (first level).</p>
</td></tr>
<tr><td><code id="SMLE_+3A_coef_initial">coef_initial</code></td>
<td>
<p>A <code class="reqn">p</code>-dimensional vector for the initial coefficient 
value of the IHT algorithm.  The default is to use Lasso with the sparsity 
closest to <code class="reqn">n-1</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_max_iter">max_iter</code></td>
<td>
<p>Maximum number of iteration steps. Default is 500.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_tol">tol</code></td>
<td>
<p>A tolerance level to stop the iterations, when the squared sum of
differences between two successive coefficient updates is below it.
Default is <code class="reqn">10^{-3}</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_selection">selection</code></td>
<td>
<p>A logical flag to indicate whether an elaborate selection
is to be conducted by <code><a href="#topic+smle_select">smle_select</a>()</code> after screening.
If <code>TRUE</code>, the function will return a <code>'selection'</code> object, 
see <code><a href="#topic+smle_select">smle_select</a>()</code> documentation. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_standardize">standardize</code></td>
<td>
<p>A logical flag for feature standardization, prior to
performing feature screening. The resulting coefficients are
always returned on the original scale. 
If features are in the same units already, you might not wish to
standardize. Default is <code>standardize = TRUE</code>.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_fast">fast</code></td>
<td>
<p>Set to <code>TRUE</code> to enable early stop for SMLE-screening. It may help
to boost the screening efficiency with a little sacrifice of accuracy.
Default is <code>FALSE</code>, see Details.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_u">U</code></td>
<td>
<p>A numerical multiplier of initial tuning step parameter in 
IHT algorithm. Default is 1. For binomial model, a larger initial value is recommended; a
smaller one is recommended for poisson model.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_u_rate">U_rate</code></td>
<td>
<p>Decreasing rate in tuning step parameter <code class="reqn">1/u</code> in IHT
algorithm. See Details.</p>
</td></tr>
<tr><td><code id="SMLE_+3A_penalize_mod">penalize_mod</code></td>
<td>
<p>A logical flag to indicate whether adjustment is used in
ranking groups of features. This argument is applicable only when
<code>categorical = TRUE</code> with <code>group = TRUE</code>. When <code>penalize_mod = TRUE</code>, 
a factor of <code class="reqn">\sqrt J</code> is divided from the <code class="reqn">L_2</code> effect of a group with <code class="reqn">J</code> members. 
Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>With the input <code class="reqn">Y</code> and <code class="reqn">X</code>, <code><a href="#topic+SMLE">SMLE</a>()</code> conducts joint feature screening by running
iterative hard thresholding algorithm (IHT), where the default initial value is set to
be the Lasso estimate with the sparsity closest to the sample size minus one.
</p>
<p>In <code><a href="#topic+SMLE">SMLE</a>()</code>, the initial value for step size parameter <code class="reqn">1/u</code> is 
determined as follows. When <code>coef_initial = 0</code>, we set <code class="reqn">1/u = U / \sqrt{p}</code>.
When <code>coef_initial != 0</code>, we generate a sub-matrix <code class="reqn">X_0</code> using the columns of <code class="reqn">X</code> 
corresponding to the non-zero positions of <code>coef_initial</code> and set
<code class="reqn">1/u = U/\sqrt{p}||X||^2_{\infty}</code> and recursively decrease the value of step size by 
<code>U_rate</code> to guarantee the likelihood increment. This strategy is called <code class="reqn">u</code>-search.
</p>
<p><code><a href="#topic+SMLE">SMLE</a>()</code> terminates IHT iterations when either <code>tol</code> or <code>max_iter</code> is
satisfied. When <code>fast = TRUE</code>, the algorithm also stops when the non-zero
members of the coefficient estimates remain the same for 10 successive
iterations or the log-likelihood difference between coefficient estimates is less
than <code class="reqn">0.01</code> times the log-likelihood increase of the first step, or 
<code>tol</code><code class="reqn">\sqrt k</code> is satisfied.
</p>
<p>In <code><a href="#topic+SMLE">SMLE</a>()</code>, categorical features are coded by dummy covariates with the
method specified in <code>codingtype</code>. Users can use <code>group</code> to specify
whether to treat those dummy covariates as a single group feature or as
individual features.
When <code>group = TRUE</code> with <code>penalize_mod = TRUE</code>, the effect for a group
of <code class="reqn">J</code> dummy covariates is computed by
</p>
<p style="text-align: center;"><code class="reqn"> \beta_i = \sqrt{(\beta_1)^2+...+(\beta_J)^2}/\sqrt J,</code>
</p>

<p>which will be treated as a single feature in IHT iterations. When <code>group = FALSE</code>, 
a group of <code class="reqn">J</code> dummy covariates will be treated as <code class="reqn">J</code> individual features in the IHT iterations; in this case, 
a categorical feature is retained after screening when at least one of the corresponding dummy covariates is retained.
</p>
<p>Since feature screening is usually a preprocessing step, users may wish to
further conduct an elaborative feature selection after screening. This can
be done by setting <code>selection = TRUE</code> in <code><a href="#topic+SMLE">SMLE</a>()</code> or applying any existing
selection method on the output of <code><a href="#topic+SMLE">SMLE</a>()</code>.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>ID_retained</code></td>
<td>
<p>A vector indicating the features retained after SMLE-screening.
The output includes both features retained by <code><a href="#topic+SMLE">SMLE</a>()</code> and the features specified in <code>keyset</code>.</p>
</td></tr>
<tr><td><code>coef_retained</code></td>
<td>
<p>The vector of coefficients estimated by IHT for the retained features. When the 
retained set contains a categorical feature, the value returns a group effect if
<code>group = TRUE</code>, or returns the strongest dummy covariate effect if <code>group = FALSE</code>.</p>
</td></tr>
<tr><td><code>path_retained</code></td>
<td>
<p>IHT iteration path with columns recording the coefficient updates.</p>
</td></tr>
<tr><td><code>num_retained</code></td>
<td>
<p>Number of retained features after screening.</p>
</td></tr>
<tr><td><code>intercept</code></td>
<td>
<p>The estimated intercept value by IHT, if <code>intercept = TRUE</code>.</p>
</td></tr>
<tr><td><code>steps</code></td>
<td>
<p>Number of IHT iterations.</p>
</td></tr>
<tr><td><code>likelihood_iter</code></td>
<td>
<p>A list of log-likelihood updates over the IHT iterations. </p>
</td></tr>
<tr><td><code>Usearch</code></td>
<td>
<p>A vector giving the number of attempts to find a proper <code class="reqn">1/u</code> at each iteration step.</p>
</td></tr>
<tr><td><code>modified_data</code></td>
<td>
<p>A list containing data objects generated by SMLE. 
</p>
<p><code>CM</code>: Design matrix of class <code>'matrix'</code> for numeric features (or <code>'data.frame'</code> with categorical features).
</p>
<p><code>DM</code>: A matrix with dummy variable features added. (only if there are categorical features).
</p>
<p><code>dum_col</code>: Number of levels for all categorical features.
</p>
<p><code>CI</code>: Indices of categorical features in <code>CM</code>.
</p>
<p><code>DFI</code>: Indices of categorical features in <code>IM</code>.
</p>
</td></tr>
<tr><td><code>iteration_data</code></td>
<td>
<p>A list containing data objects that track the coefficients over iterations.
</p>
<p><code>IM</code>: Iteration path matrix with columns recording IHT coefficient updates.
</p>
<p><code>beta0</code>: Inital value of regression coefficient for IHT.
</p>
<p><code>feature_name</code>: A list contains the names of selected features.
</p>
<p><code>FD</code>: A matrix that contains feature indices retained at each iteration step.
</p>
</td></tr> 
</table>
<p><code>X</code>, <code>Y</code>, <code>data</code>, <code>family</code>, <code>categorical</code> and <code>codingtype</code> are return of arguments passed in the function call.
</p>


<h3>References</h3>

<p>UCLA Statistical Consulting Group. <em>coding systems for categorical
variables in regression analysis</em>. <a href="https://stats.oarc.ucla.edu/r/library/r-library-contrast-coding-systems-for-categorical-variables/">https://stats.oarc.ucla.edu/r/library/r-library-contrast-coding-systems-for-categorical-variables/</a>.
Retrieved May 28, 2020.
</p>
<p>Xu, C. and Chen, J. (2014). The Sparse MLE for Ultrahigh-Dimensional Feature
Screening, <em>Journal of the American Statistical Association</em>, <b>109</b>(507), 1257-1269.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example 1:
set.seed(1)
Data &lt;- Gen_Data( n= 200, p = 5000, family = "gaussian", correlation = "ID")
fit &lt;- SMLE( Y = Data$Y , X = Data$X, k = 9,family = "gaussian")
summary(fit)
Data$subset_true %in% fit$ID_retained # Sure screening check.
plot(fit)

# Example 2:
set.seed(1)
Data_sim2 &lt;- Gen_Data(n = 420, p = 1000, family = "gaussian", num_ctgidx = 5, 
                      pos_ctgidx = c(1,3,5,7,9), effect_truecoef= c(1,2,3,-4,-5),
                      pos_truecoef = c(1,3,5,7,8), level_ctgidx = c(3,3,3,4,5))
train_X &lt;- Data_sim2$X[1:400,]; test_X &lt;- Data_sim2$X[401:420,]
train_Y &lt;- Data_sim2$Y[1:400]; test_Y &lt;- Data_sim2$Y[401:420]
fit &lt;- SMLE(Y = train_Y, X = train_X, family = "gaussian", group = TRUE, k = 15)
predict(fit, newdata = test_X)
test_Y

# Example 3:
library(datasets)
data("attitude")
set.seed(1)
noise &lt;- matrix(rnorm(30*100, mean = mean(attitude$rating) , sd = 1), ncol = 100)
colnames(noise) &lt;- paste("Noise", seq(100), sep = ".")
df &lt;- data.frame(cbind(attitude, noise))
fit &lt;- SMLE(rating ~., data = df)
fit



</code></pre>

<hr>
<h2 id='smle_select'>Elaborative post-screening selection with SMLE</h2><span id='topic+smle_select'></span><span id='topic+smle_select.sdata'></span><span id='topic+smle_select.default'></span><span id='topic+smle_select.smle'></span>

<h3>Description</h3>

<p>The features retained after screening are still likely to contain some that 
are not related to the response. The function <code><a href="#topic+smle_select">smle_select</a>()</code> is designed to 
further identify the relevant features using <code><a href="#topic+SMLE">SMLE</a>()</code>.
Given a response and a set of <code class="reqn">K</code> features, this function
first runs <code><a href="#topic+SMLE">SMLE</a>(fast = TRUE)</code> to generate a series of sub-models with
sparsity k varying from <code>k_min</code> to <code>k_max</code>.
It then selects the best model from the series based on a selection criterion.
</p>
<p>When criterion EBIC is used, users can choose to repeat the selection with
different values of the tuning parameter <code class="reqn">\gamma</code>, and
conduct importance voting for each feature. When <code>vote = T</code>, this function 
fits all the models with <code class="reqn">\gamma</code> specified in <code>gamma_seq</code> and features 
with frequency higher than <code>vote_threshold</code> will be selected in <code>ID_voted</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smle_select(object, ...)

## S3 method for class 'sdata'
smle_select(
  object,
  k_min = 1,
  k_max = NULL,
  subset = NULL,
  gamma_ebic = 0.5,
  vote = FALSE,
  keyset = NULL,
  criterion = "ebic",
  codingtype = c("DV", "standard", "all"),
  gamma_seq = c(seq(0, 1, 0.2)),
  vote_threshold = 0.6,
  parallel = FALSE,
  num_clusters = NULL,
  ...
)

## Default S3 method:
smle_select(
  object = NULL,
  Y = NULL,
  X = NULL,
  family = "gaussian",
  keyset = NULL,
  ...
)

## S3 method for class 'smle'
smle_select(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smle_select_+3A_object">object</code></td>
<td>
<p>Object of class <code>'smle'</code> or <code>'sdata'</code>. Users can also
input a response vector and a feature matrix.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_k_min">k_min</code></td>
<td>
<p>The lower bound of candidate model sparsity. Default is 1.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_k_max">k_max</code></td>
<td>
<p>The upper bound of candidate model sparsity. Default is 
the number of columns in feature matrix.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_subset">subset</code></td>
<td>
<p>An index vector indicating which features (columns of the
feature matrix) are to be selected.  Not applicable if a <code>'smle'</code>
object is the input.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_gamma_ebic">gamma_ebic</code></td>
<td>
<p>The EBIC tuning parameter, in <code class="reqn">[0 , 1]</code>. Default is 0.5.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_vote">vote</code></td>
<td>
<p>The logical flag for whether to perform the voting procedure. Only available when <code>criterion = "ebic"</code>.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_keyset">keyset</code></td>
<td>
<p>A numeric vector with column indices for the key features that 
do not participate in feature screening and are forced to remain in the model. See SMLE for details.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_criterion">criterion</code></td>
<td>
<p>Selection criterion. One of &quot;<code>ebic</code>&quot;,&quot;<code>bic</code>&quot;,&quot;<code>aic</code>&quot;. Default is &quot;<code>ebic</code>&quot;.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_codingtype">codingtype</code></td>
<td>
<p>Coding types for categorical features; for more details see <code><a href="#topic+SMLE">SMLE</a>()</code> documentation.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_gamma_seq">gamma_seq</code></td>
<td>
<p>The sequence of values for <code>gamma_ebic</code> when <code>vote = TRUE</code>.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_vote_threshold">vote_threshold</code></td>
<td>
<p>A relative voting threshold in percentage. A feature is
considered to be important when it receives votes passing the threshold. Default is 0.6.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_parallel">parallel</code></td>
<td>
<p>A logical flag to use parallel computing to do voting selection.
Default is <code>FALSE</code>. See Details.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_num_clusters">num_clusters</code></td>
<td>
<p>The number of compute clusters to use when 
<code>parallel = TRUE</code>. The default will be 2 times cores detected.</p>
</td></tr>
<tr><td><code id="smle_select_+3A_y">Y</code></td>
<td>
<p>Input response vector (when <code>object = NULL</code>).</p>
</td></tr>
<tr><td><code id="smle_select_+3A_x">X</code></td>
<td>
<p>Input features matrix (when <code>object = NULL</code>).</p>
</td></tr>
<tr><td><code id="smle_select_+3A_family">family</code></td>
<td>
<p>Model assumption; see <code><a href="#topic+SMLE">SMLE</a>()</code> documentation. Default is Gaussian linear.
</p>
<p>When input is a <code>'smle'</code> or <code>'sdata'</code> object, the same
model will be used in the selection.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function accepts three types of input objects; 
1) <code>'smle'</code> object, as the output from <code><a href="#topic+SMLE">SMLE</a>()</code>; 
2) <code>'sdata'</code> object, as the output from <code><a href="#topic+Gen_Data">Gen_Data</a>()</code>; 
3) other response and feature matrix input by users.
</p>
<p>Note that this function is mainly designed to conduct an elaborative selection
after feature screening. We do not recommend using it directly for
ultra-high-dimensional data without screening.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>The call that produced this object.</p>
</td></tr>
<tr><td><code>ID_selected</code></td>
<td>
<p>A list of selected features.</p>
</td></tr>
<tr><td><code>coef_selected</code></td>
<td>
<p>Fitted model coefficients.</p>
</td></tr>
<tr><td><code>intercept</code></td>
<td>
<p>Fitted model intercept.</p>
</td></tr>
<tr><td><code>criterion_value</code></td>
<td>
<p>Values of selection criterion for the candidate models with various sparsity.</p>
</td></tr>
<tr><td><code>categorical</code></td>
<td>
<p>A logical flag whether the input feature matrix includes categorical features</p>
</td></tr>
<tr><td><code>ID_pool</code></td>
<td>
<p>A vector containing all features selected during voting. </p>
</td></tr>
<tr><td><code>ID_voted</code></td>
<td>
<p>A vector containing the features selected when <code>vote = T</code>.</p>
</td></tr>
<tr><td><code>CI</code></td>
<td>
<p>Indices of categorical features when <code>categorical = TRUE</code>.</p>
</td></tr>
</table>
<p><code>X</code>, <code>Y</code>, <code>family</code>, <code>gamma_ebic</code>, <code>gamma_seq</code>, <code>criterion</code>, <code>vote</code>,
<code>codyingtype</code>, <code>vote_threshold</code> are return of arguments passed in the function call.
</p>


<h3>References</h3>

<p>Chen. J. and Chen. Z. (2012). &quot;Extended BIC for small-n-large-p sparse GLM.&quot;
<em>Statistica Sinica</em>, <b>22</b>(2), 555-574.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)
Data&lt;-Gen_Data(correlation = "MA", family = "gaussian")
fit&lt;-SMLE(Y = Data$Y, X = Data$X, k = 20, family = "gaussian")

fit_bic&lt;-smle_select(fit, criterion = "bic")
summary(fit_bic)

fit_ebic&lt;-smle_select(fit, criterion = "ebic", vote = TRUE)
summary(fit_ebic)
plot(fit_ebic)


</code></pre>

<hr>
<h2 id='summary.smle'>Summarize SMLE-screening and selection</h2><span id='topic+summary.smle'></span><span id='topic+summary.selection'></span>

<h3>Description</h3>

<p>This function prints a summary of a <code>'smle'</code> (or a <code>'selection'</code>) object.
In particular, it shows the features retained after SMLE-screening (or selection) with the related convergence information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'smle'
summary(object, ...)

## S3 method for class 'selection'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.smle_+3A_object">object</code></td>
<td>
<p>A <code>'smle'</code> or <code>'selection'</code> object.</p>
</td></tr>
<tr><td><code id="summary.smle_+3A_...">...</code></td>
<td>
<p>This argument is not used and listed for method consistency.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
Data &lt;- Gen_Data(correlation = "MA", family = "gaussian")
fit &lt;- SMLE(Y = Data$Y, X = Data$X, k = 20, family = "gaussian")
summary(fit)
fit_s &lt;- smle_select(fit)
summary(fit_s)

</code></pre>

<hr>
<h2 id='synSNP'>Synthetic genetic association study data set</h2><span id='topic+synSNP'></span>

<h3>Description</h3>

<p>This simulated data set consists of 10,031 genetic variants (SNPs) and a 
continuous response variable measured on 800 individuals. The 
genotypes were sampled from genotypic distributions derived from the  
1000 Genomes project using the <span class="rlang"><b>R</b></span> package <span class="pkg">sim1000G</span>. The genotype 
is coded as 0, 1, or 2 by counting the number of minor alleles (the allele 
that is less common in the sample). The continuous response variable was 
simulated from a normal distribution with mean that depends additively 
on the causal SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(synSNP)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>'data.frame'</code> with 800 rows and 10,032 columns.
</p>


<h3>References</h3>

<p>The 1000 Genomes Project Consortium (2015). Global reference for human genetic variation,
<em>Nature</em>, <b>526</b>(7571), 68-74.s
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(synSNP)
Y_SNP &lt;- synSNP[,1]
X_SNP &lt;- synSNP[,-1]
fit &lt;- SMLE(Y = Y_SNP, X = X_SNP, k = 40)
summary(fit)
plot(fit)

</code></pre>

<hr>
<h2 id='vote_update'>Extract and adjust voting from SMLE selection</h2><span id='topic+vote_update'></span><span id='topic+vote_update.selection'></span>

<h3>Description</h3>

<p>When <code><a href="#topic+smle_select">smle_select</a>()</code> is used with <code>criterion = "ebic"</code> and <code>vote = TRUE</code>, users 
can use <code><a href="#topic+vote_update">vote_update</a>()</code> to adjust the voting threshold without a need 
of rerun <code><a href="#topic+smle_select">smle_select</a>()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vote_update(object, ...)

## S3 method for class 'selection'
vote_update(object, vote_threshold = 0.6, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vote_update_+3A_object">object</code></td>
<td>
<p>A <code>'selection'</code> object as the output from <code><a href="#topic+smle_select">smle_select</a>()</code>.</p>
</td></tr>
<tr><td><code id="vote_update_+3A_...">...</code></td>
<td>
<p>This argument is not used and listed for method consistency.</p>
</td></tr>
<tr><td><code id="vote_update_+3A_vote_threshold">vote_threshold</code></td>
<td>
<p>A voting threshold in percentage. A feature is
considered to be important when it receives votes passing the threshold.
Default is 0.6.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a vector indicating the features selected by
EBIC voting with the specified <code>vote_threhold</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
Data &lt;- Gen_Data(n = 100, p = 3000, correlation = "MA", rho = 0.7, family = "gaussian")
colnames(Data$X)&lt;- paste("X.",seq(3000) , sep = "")
fit &lt;- SMLE(Y = Data$Y, X = Data$X, k = 20, family = "gaussian")
fit_s &lt;- smle_select(fit, criterion = "ebic", vote = TRUE)
plot(fit_s)
fit_s
vote_update(fit_s, vote_threshold = 0.4)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
