<!DOCTYPE html><html lang="en"><head><title>Help for package SMPracticals</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SMPracticals}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add.exp.lines'><p>Add Exponential Lines in Practical 11.3</p></a></li>
<li><a href='#alofi'><p>Daily Rainfall at Alofi</p></a></li>
<li><a href='#aml'><p>Remission Times for Acute Myelogenous Leukaemia</p></a></li>
<li><a href='#arithmetic'><p>Teaching Arithmetic Data</p></a></li>
<li><a href='#bard'><p>Shakespeare's Word Type Frequencies</p></a></li>
<li><a href='#barley'><p>Spring Barley Data</p></a></li>
<li><a href='#beaver'><p>Body Temperatures for a Female Beaver</p></a></li>
<li><a href='#beaver.gibbs'><p> Gibbs Sampler for Normal Changepoint Model, Practical 11.7</p></a></li>
<li><a href='#beetle'><p>Japanese beetle data</p></a></li>
<li><a href='#bike'><p>Bicycling Times</p></a></li>
<li><a href='#births'><p>Birth Times</p></a></li>
<li><a href='#blalock'><p>Blalock&ndash;Taussig Shunt Data</p></a></li>
<li><a href='#bliss'><p>Bliss data on deaths of flour beetles</p></a></li>
<li><a href='#blood'><p>Blood Group Data</p></a></li>
<li><a href='#breast'><p>Breast Cancer Data</p></a></li>
<li><a href='#burt'><p>IQs of identical twins</p></a></li>
<li><a href='#cake'><p>Breaking of Chocolate Cakes</p></a></li>
<li><a href='#calcium'><p>Calcium Uptake Data</p></a></li>
<li><a href='#cardiac'><p>Mortality Rates for Cardiac Surgery on Babies at 12 Hospitals</p></a></li>
<li><a href='#cat.heart'><p>Cat Heart Data</p></a></li>
<li><a href='#cement'><p>Hald Cement Data</p></a></li>
<li><a href='#chicks'><p>Chick Bone Data</p></a></li>
<li><a href='#chimps'><p>Chimpanzee Learning Data</p></a></li>
<li><a href='#cloth'><p>Numbers of Flaws in Lengths of Cloth</p></a></li>
<li><a href='#coal'><p>Data on UK coal mining disasters</p></a></li>
<li><a href='#coin.spin'><p>Function for Coin Spinning, Practical 11.1</p></a></li>
<li><a href='#danish'><p>Danish Fire Insurance Claims</p></a></li>
<li><a href='#darwin'><p>Darwin's Maize Data</p></a></li>
<li><a href='#exp.gibbs'><p>Gibbs Sampling for Two Truncated Exponential Variables, Practical 11.3</p></a></li>
<li><a href='#eyes'><p>Visual Impairment Data</p></a></li>
<li><a href='#field.concrete'><p>Field Concrete Mixture Data</p></a></li>
<li><a href='#fir'><p>Counts of Balsam-fir Seedlings</p></a></li>
<li><a href='#forbes'><p>Atmospheric Pressure and Boiling Point in the Alps</p></a></li>
<li><a href='#frets'><p>Head Dimensions in Brothers</p></a></li>
<li><a href='#ftse'><p>FTSE Daily Returns</p></a></li>
<li><a href='#galaxy'><p>Galaxy Velocity Data</p></a></li>
<li><a href='#get.alpha'><p>Estimate Alpha from Data</p></a></li>
<li><a href='#glm.diag'><p>Generalized Linear Model Diagnostics</p></a></li>
<li><a href='#hus'><p>Haemolytic Uraemic Syndrome</p></a></li>
<li><a href='#hus.gibbs'><p>Gibbs Sampler for Poisson Changepoint Model, Practical 11.6</p></a></li>
<li><a href='#ihess'><p>Inverse Hessian</p></a></li>
<li><a href='#intron'><p>Intron Gene Sequence</p></a></li>
<li><a href='#jacamar'><p>Jacamar Learning Ability Data</p></a></li>
<li><a href='#jelinski'><p>Jelinski and Moranda Data on Software Failures</p></a></li>
<li><a href='#leuk'><p>Survival Times and White Blood Counts for Leukaemia Patients</p></a></li>
<li><a href='#lik.ci'><p>Likelihood Confidence Intervals for Scalar Parameter</p></a></li>
<li><a href='#limits'><p>Swedish Speed Limit Data</p></a></li>
<li><a href='#lizards'><p>Lizard Count Data</p></a></li>
<li><a href='#lung.cancer'><p>Lung Cancer Deaths among UK Physicians</p></a></li>
<li><a href='#magnesium'><p>Magnesium Treatment for Heart Attack Patients</p></a></li>
<li><a href='#manaus'><p>Average Heights of the Rio Negro river at Manaus</p></a></li>
<li><a href='#marking'><p>Examination Marking Data</p></a></li>
<li><a href='#mathmarks'><p>Math Marks Data</p></a></li>
<li><a href='#MClik'><p>Likelihood Estimation for Markov Chains</p></a></li>
<li><a href='#mice'><p>Mice Deaths from Radiation</p></a></li>
<li><a href='#millet'><p>Millet Data</p></a></li>
<li><a href='#motorette'><p>Motorette Failure Data</p></a></li>
<li><a href='#nematode'><p>Nematode Data</p></a></li>
<li><a href='#nodal'><p>Nodal Involvement in Prostate Cancer</p></a></li>
<li><a href='#nuclear'><p>Nuclear Power Station Construction Data</p></a></li>
<li><a href='#old.age'><p>Estimates of Hazard Function for Old Age</p></a></li>
<li><a href='#pairs.mod'><p>Modified Scatterplot Matrix</p></a></li>
<li><a href='#paulsen'><p>Neurotransmission in Guinea Pig Brains</p></a></li>
<li><a href='#pbc'><p>Mayo Clinic Primary Biliary Cirrhosis Data</p></a></li>
<li><a href='#pigeon'><p>Homing Pigeon Data</p></a></li>
<li><a href='#pigs'><p>Pig Diet Data</p></a></li>
<li><a href='#pipette'><p>Red Blood Cell Data</p></a></li>
<li><a href='#plot.glm.diag'><p>Diagnostic plots for generalized linear models</p></a></li>
<li><a href='#pneu'><p>Pneumoconiosis amongst Coalminers</p></a></li>
<li><a href='#poi.beta.laplace'><p>Laplace Approximation for Posterior Density, Practical 11.2</p></a></li>
<li><a href='#poi.gibbs'><p>Gibbs Sampler for Hierarchical Poisson Model, Practical 11.5</p></a></li>
<li><a href='#poisons'><p>Survival Times for Poisoned Animals</p></a></li>
<li><a href='#pollution'><p>Air Pollution and Mortality</p></a></li>
<li><a href='#pumps'><p>Pump Failure Data</p></a></li>
<li><a href='#qqexp'><p>Exponential Quantile-Quantile Plots</p></a></li>
<li><a href='#quake'><p>Japanese Earthquake Data</p></a></li>
<li><a href='#rat.growth'><p>Rat Growth Data</p></a></li>
<li><a href='#salinity'><p>Water Salinity and River Discharge</p></a></li>
<li><a href='#seeds'><p>Germination of seeds</p></a></li>
<li><a href='#shoe'><p>Shoe Wear Data</p></a></li>
<li><a href='#shuttle'><p>O-ring Thermal Distress Data for Space Shuttle</p></a></li>
<li><a href='#smoking'><p>Survival and Smoking</p></a></li>
<li><a href='#soccer'><p>Soccer Scores from English Premier League, 2000-2001 Season</p></a></li>
<li><a href='#springs'><p>Spring Failure Data</p></a></li>
<li><a href='#sticky'><p>Stickiness of blood data</p></a></li>
<li><a href='#survival'><p>Survival of Rats After Radiation Doses</p></a></li>
<li><a href='#teak'><p>Teak Plant Data</p></a></li>
<li><a href='#tide'><p>Annual Maximum Sea Levels</p></a></li>
<li><a href='#toxo'><p>Toxoplasmosis Data</p></a></li>
<li><a href='#ulcer'><p>Recurrent Bleeding from Ulcers</p></a></li>
<li><a href='#urine'><p> Urine Analysis Data</p></a></li>
<li><a href='#venice'><p>Extreme Sea Levels at Venice</p></a></li>
<li><a href='#yahoo'><p>Yahoo Closing Prices</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.4-3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-13</td>
</tr>
<tr>
<td>Title:</td>
<td>Practicals for Use with Davison (2003) Statistical Models</td>
</tr>
<tr>
<td>Author:</td>
<td>Anthony Davison &lt;Anthony.Davison@epfl.ch&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alessandra R. Brazzale &lt;brazzale@stat.unipd.it&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 1.14.0), ellipse</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, nlme, survival</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains the datasets and a few functions for use with 
        the practicals outlined in Appendix A of the book
        Statistical Models (Davison, 2003, Cambridge University Press), 
        which can be found at &lt;<a href="https://doi.org/10.1017%2FCBO9780511815850">doi:10.1017/CBO9780511815850</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td>doi:10.1017/CBO9780511815850</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-05 10:29:08 UTC; brazzale</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-05 14:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='add.exp.lines'>Add Exponential Lines in Practical 11.3 </h2><span id='topic+add.exp.lines'></span>

<h3>Description</h3>

<p>Adds lines to density plot used in Practical 11.3
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.exp.lines(exp.out, i, B = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add.exp.lines_+3A_exp.out">exp.out</code></td>
<td>
<p>Gibbs sampler output</p>
</td></tr>
<tr><td><code id="add.exp.lines_+3A_i">i</code></td>
<td>
<p>Variable index (=1, 2)</p>
</td></tr>
<tr><td><code id="add.exp.lines_+3A_b">B</code></td>
<td>
<p>Upper bound for truncated exponential density</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anthony Davison</p>


<h3>Examples</h3>

<pre><code class='language-R'>B &lt;-10; I &lt;- 15; S &lt;- 500

exp.out &lt;- exp.gibbs(B=B,I=I,S=S)

hist(exp.out[1,,I],prob=TRUE,nclass=15,xlab="u1",ylab="PDF",xlim=c(0,B),ylim=c(0,1))

add.exp.lines(exp.out,1)

</code></pre>

<hr>
<h2 id='alofi'>Daily Rainfall at Alofi</h2><span id='topic+alofi'></span>

<h3>Description</h3>

<p>Three-state data derived from daily rainfall over three years at Alofi
in the Niue Island group in the Pacific Ocean.  The states are
1 (no rain), 2 (up to 5mm rain), 3 (over 5mm).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(alofi)</code></pre>


<h3>Source</h3>

<p>Avery, P. J. and Henderson, D. A. (1999) Fitting Markov chain models to
discrete state series such as DNA sequences.
<em>Applied Statistics</em>, <b>48</b>, 53&ndash;61.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 294.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alofi)
fit &lt;- MClik(alofi)
fit$df
fit$AIC
plot(fit$df,fit$AIC)  # best model has minimal AIC?
</code></pre>

<hr>
<h2 id='aml'>Remission Times for Acute Myelogenous Leukaemia</h2><span id='topic+aml'></span>

<h3>Description</h3>

<p>A clinical trial to evaluate the efficacy of maintenance
chemotherapy for acute myelogenous leukaemia was  conducted by
Embury et al. (1977) at Stanford University.  After reaching a
stage of remission through treatment by chemotherapy, patients
were randomized into two groups. The first group received
maintenance chemotherapy and the  second group did not.  The aim
of the study was to see if maintenance  chemotherapy increased the
length of the remission.  The data here formed a preliminary
analysis which was conducted in October 1974.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(aml)</code></pre>


<h3>Format</h3>

<p>A data frame with 23 observations on the following 3 variables.
</p>

<dl>
<dt>time</dt><dd><p>The length of the complete remission (in weeks).</p>
</dd>
<dt>cens</dt><dd><p>An indicator of right censoring.  1 indicates that the
patient had a relapse and so 'time' is the length of the
remission. 0 indicates that the patient had left the study or
was still in remission in October 1974, that is the length of
remission is right-censored.</p>
</dd>
<dt>group</dt><dd><p>The group into which the patient was randomized.  Group 1
received maintenance chemotherapy, group 2 did not.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Miller, R.G. (1981) <em>Survival Analysis</em>. John Wiley: New York.  Page 49.
</p>


<h3>References</h3>

<p>Embury, S.H, Elias, L., Heller, P.H., Hood, C.E., Greenberg, P.L.
and Schrier, S.L. (1977) Remission maintenance therapy in  acute
myelogenous leukaemia. <em>Western Journal of Medicine</em>, <b>126</b>,
267&ndash;272.
</p>

<hr>
<h2 id='arithmetic'>Teaching Arithmetic Data</h2><span id='topic+arithmetic'></span>

<h3>Description</h3>

<p>45 school pupils were divided at random into 5 groups of size 9.
Groups A and B were taught arithmetic in separate classes by the
usual method.  Groups C, D, and E were taught together for several
days.  On each day group C were publically praised, group D were 
publically reproved, and group E were ignored.  The responses are
from a standard test taken by all pupils at the end of the period.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(arithmetic)</code></pre>


<h3>Format</h3>

<p>A data frame with 45 observations on the following 2 variables.
</p>

<dl>
<dt>group</dt><dd><p>a factor with levels <code>A</code> <code>B</code> <code>C</code> <code>D</code> <code>E</code></p>
</dd>
<dt>y</dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College, London.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 427.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(arithmetic)
attach(arithmetic)
plot(y~group)
anova(lm(y~group,data=arithmetic))
summary(lm(y~group,data=arithmetic))  # two different parametrisations
summary(lm(y~group-1,data=arithmetic)) # for ANOVA
</code></pre>

<hr>
<h2 id='bard'>Shakespeare's Word Type Frequencies</h2><span id='topic+bard'></span>

<h3>Description</h3>

<p>These are the frequencies with which Shakespeare used word types.
There are 846 word types which appear more than 100 times in his
total works, giving an overall total of 31534 word types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bard)</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on the following 2 variables.
</p>

<dl>
<dt>r</dt><dd><p>Number of times a word type is used</p>
</dd>
<dt>n</dt><dd><p>Number of word types used r times</p>
</dd>
</dl>



<h3>Details</h3>

<p>The canon of Shakespeare's accepted works contains 884,647 words, with
31,534 distinct word types.  A word type is a distinguishable arrangement 
of letters, so &lsquo;king&rsquo; is different from &lsquo;kings&rsquo; and &lsquo;alehouse&rsquo; different 
from both &lsquo;ale&rsquo; and &lsquo;house&rsquo;. 
</p>


<h3>Source</h3>

<p>Efron, B. and Thisted, R. (1976) Estimating the number of 
unseen species: How many words did Shakespeare know? <em>Biometrika</em>, 
<b>63</b>,  435&ndash;448.  
</p>
<p>Thisted, R. and Efron, B. (1987 ) Did Shakespeare write a newly-discovered poem? 
<em>Biometrika</em>, <b>74</b>, 445&ndash;455. 
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 629.
</p>

<hr>
<h2 id='barley'>Spring Barley Data</h2><span id='topic+barley'></span>

<h3>Description</h3>

<p>The spatial layout and plot yield at harvest in a final assessment trial of 
75 varietes of spring barley.  The varieties are sown in three blocks, each 
with 75 plots, and each variety is replicated thrice.  The yield for variety
27 is missing in block 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(barley)</code></pre>


<h3>Format</h3>

<p>A data frame with 225 observations on the following 4 variables.
</p>

<dl>
<dt>Block</dt><dd><p>a factor with three levels</p>
</dd>
<dt>Location</dt><dd><p>a numeric vector with 75 values giving the plot</p>
</dd>
<dt>Variety</dt><dd><p>a factor with 75 levels giving the variety of barley sown in the plot</p>
</dd>
<dt>y</dt><dd><p>yield at harvest, standardised to have unit crude variance</p>
</dd>
</dl>



<h3>Source</h3>

<p>Besag, J. E., Green, P. J., Higdon, D. and Mengersen, K. (1995)  
Bayesian computation and stochastic systems (with Discussion).  
<em>Statistical Science</em>, <b>10</b>, 3&ndash;66.  
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Pages 534&ndash;535.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(barley)
</code></pre>

<hr>
<h2 id='beaver'>Body Temperatures for a Female Beaver</h2><span id='topic+beaver'></span>

<h3>Description</h3>

<p>Data comprise 100 consecutive elemetric measurements of the body
temperature of a female beaver, at 10-minute intervals.
The animal remained in its lodge for the first 38 recordings, and then
went outside.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(beaver)</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on the following 4 variables.
</p>

<dl>
<dt>day</dt><dd><p>Day number</p>
</dd>
<dt>time</dt><dd><p>Time of day (hhmm)</p>
</dd>
<dt>temp</dt><dd><p>Body temperature (degrees Celsius)</p>
</dd>
<dt>activ</dt><dd><p>Indicator of activity outside the lodge</p>
</dd>
</dl>



<h3>Source</h3>

<p>Reynolds, P. S. (1994) Time-series analyses of beaver body temperatures.
In <em>Case Studies in Biometry</em>, eds N. Lange, L. Ryan,
L. Billard, D. R. Brillinger, L. Conquest and J. Greenhouse, pp. 211&ndash;228.
New York: Wiley.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 266.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(beaver)
plot(beaver$temp,type="l",xlab="Time",ylab="Temperature")
</code></pre>

<hr>
<h2 id='beaver.gibbs'> Gibbs Sampler for Normal Changepoint Model, Practical 11.7</h2><span id='topic+beaver.gibbs'></span>

<h3>Description</h3>

<p>This function implements a Gibbs sampler for the normal changepoint
model applied to the beaver temperature data used in Example 6.22 
and Practical 11.7 of Davison (2003), which should be consulted for details.</p>


<h3>Usage</h3>

<pre><code class='language-R'>beaver.gibbs(init, y, R = 10, a = 1, b = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="beaver.gibbs_+3A_init">init</code></td>
<td>
<p>Initial values for parameters </p>
</td></tr>
<tr><td><code id="beaver.gibbs_+3A_y">y</code></td>
<td>
<p> A series of normal observations</p>
</td></tr>
<tr><td><code id="beaver.gibbs_+3A_r">R</code></td>
<td>
<p>Number of iterations of sampler</p>
</td></tr>
<tr><td><code id="beaver.gibbs_+3A_a">a</code></td>
<td>
<p>Value of a hyperparameter</p>
</td></tr>
<tr><td><code id="beaver.gibbs_+3A_b">b</code></td>
<td>
<p> Value of a hyperparameter</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is provided simply so that  readers spend less time typing. It is not intended to be
robust and general code.
</p>


<h3>Value</h3>

<p>A matrix of size R x 6, whose first four columns contain the values of the
parameters for the iterations.  Columns 5 and 6 contain the log likelihood and log prior
for that iteration.
</p>


<h3>Author(s)</h3>

<p>Anthony Davison (<code>anthony.davison@epfl.ch</code>)</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press. Practical 11.7.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## From Example 11.7:
data(beaver)
system.time( gibbs.out &lt;- beaver.gibbs(c(36, 40, 3, 38), beaver$temp, R=1000))
par(mfrow=c(2,3))
plot.ts(gibbs.out[,1],main="mu1") # time series plot for mu1
plot.ts(gibbs.out[,2],main="mu2") # time series plot for mu2
plot.ts(gibbs.out[,3],main="lambda") # time series plot for lambda
plot.ts(gibbs.out[,4],main="gamma") # time series plot for gamma
plot.ts(gibbs.out[,5],main="log likelihood")  # and of log likelihood
</code></pre>

<hr>
<h2 id='beetle'>Japanese beetle data</h2><span id='topic+beetle'></span>

<h3>Description</h3>

<p>Numbers of Japanese beetle larvae per square found in the top foot of soil
of an 18 x 8 foot area of a field planted with maize.  The columns of the matrix
correspond to the direction of cultivation of the field; the maize rows were sown
4 feet apart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(beetle)</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:18, 1:8] 0 2 3 1 5 3 5 3 2 3 ...
</p>


<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College, London</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(beetle)
</code></pre>

<hr>
<h2 id='bike'>Bicycling Times</h2><span id='topic+bike'></span>

<h3>Description</h3>

<p>The times taken to cycle up a hill, as function of the bicycle
seat height, use of dynamo, and tyre pressure.  16 runs were made
using a factorial design.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bike)</code></pre>


<h3>Format</h3>

<p>A data frame with 16 observations on the following 11 variables.
</p>

<dl>
<dt>day</dt><dd><p>Day of run</p>
</dd>
<dt>run</dt><dd><p>Order of run</p>
</dd>
<dt>seat</dt><dd><p>Seat height: -1 indicates 26 inches, 1 indicates 30 inches</p>
</dd>
<dt>dynamo</dt><dd><p>Use of dynamo: -1 indicates not used</p>
</dd>
<dt>tyre</dt><dd><p>Tyre pressure: -1 indicates 40 psi, 1 indicates 55 psi</p>
</dd>
<dt>dayf</dt><dd><p>factor corresponding to day</p>
</dd>
<dt>runf</dt><dd><p>factor corresponding to run</p>
</dd>
<dt>seatf</dt><dd><p>factor corresponding to seat height</p>
</dd>
<dt>dynamof</dt><dd><p>factor corresponding to use of dynamo</p>
</dd>
<dt>tyref</dt><dd><p>factor corresponding to tyre pressure</p>
</dd>
<dt>time</dt><dd><p>Run time (seconds)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Box, G. E. P., Hunter, W. G. and Hunter, J. S. (1978) 
<em>Statistics for Experimenters</em>.  New York: Wiley.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 357.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bike)
anova(lm(time~dayf+runf+seat+dynamo+tyre,data=bike))
</code></pre>

<hr>
<h2 id='births'>Birth Times</h2><span id='topic+births'></span>

<h3>Description</h3>

<p>Times spent in delivery suite by 95 women giving birth at the John Radcliffe Hospital,
Oxford.  The data were kindly provided by Ethel Burns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(births)</code></pre>


<h3>Format</h3>

<p>A data frame with 95 observations on the following 2 variables.
</p>

<dl>
<dt>day</dt><dd><p>Day on which woman arrived</p>
</dd>
<dt>time</dt><dd><p>Time (hours) spent on delivery suite</p>
</dd>
</dl>



<h3>Source</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press. Page 18.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(births)
</code></pre>

<hr>
<h2 id='blalock'>Blalock&ndash;Taussig Shunt Data</h2><span id='topic+blalock'></span>

<h3>Description</h3>

<p>The Blalock&ndash;Taussig shunt is an operative procedure for infants with 
congenital cyanotic heart disease.  The data are the survival times
in months for shunts in 81 infants, divided into two age groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(blalock)</code></pre>


<h3>Format</h3>

<p>A data frame with 81 observations on the following 3 variables.
</p>

<dl>
<dt>group</dt><dd><p>1 indicates infants aged over 1 month at time of the 
operation.  2 indicates those aged 30 or fewer days at time of operation.</p>
</dd>
<dt>months</dt><dd><p>survival time in months</p>
</dd>
<dt>cens</dt><dd><p>censoring indicator: 1 indicates observed failure time</p>
</dd>
</dl>



<h3>Source</h3>

<p>Oakes, D. (1991) Life-table analysis.
In <em>Statistical Theory and Modelling: In Honour of Sir David
Cox, FRS</em>, eds D. V. Hinkley, N. Reid and E. J. Snell, pp. 107&ndash;128. 
London: Chapman and Hall/CRC Press.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 192.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(blalock)
library(survival)
plot(survfit(Surv(months,cens)~group,data=blalock),conf.int=TRUE,col=c(2,3))
</code></pre>

<hr>
<h2 id='bliss'>Bliss data on deaths of flour beetles</h2><span id='topic+bliss'></span>

<h3>Description</h3>

<p>These are the number of adult flour beetles which died following a
5-hour exposure to gaseous carbon disulphide.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bliss)</code></pre>


<h3>Format</h3>

<p>A data frame with 8 observations on the following 3 variables.
</p>

<dl>
<dt>dose</dt><dd><p>concentration of carbon disulphide(mg. per litre)</p>
</dd>
<dt>m</dt><dd><p>Numbers of beetles exposed</p>
</dd>
<dt>r</dt><dd><p>Numbers of beetles dying</p>
</dd>
</dl>



<h3>Source</h3>

<p>Bliss, C. I. (1935).The calculation of the dosage-mortality curve. 
<em>Annals of Applied BIology</em>, <b>22</b>, 134-167.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bliss)
attach(bliss)
plot(log(dose),r/m,ylim=c(0,1),ylab="Proportion dead")
fit &lt;- glm(cbind(r,m-r)~log(dose),binomial)
summary(fit)
</code></pre>

<hr>
<h2 id='blood'>Blood Group Data</h2><span id='topic+blood'></span>

<h3>Description</h3>

<p>Data on the incidence of blood groups O, A, B, and AB
in 12 studies on people living in Britain or of British
origin living elsewhere.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(blood)</code></pre>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 4 variables.
</p>

<dl>
<dt>O</dt><dd><p>Number of persons with blood group O</p>
</dd>
<dt>A</dt><dd><p>Number of persons with blood group A</p>
</dd>
<dt>B</dt><dd><p>Number of persons with blood group B</p>
</dd>
<dt>AB</dt><dd><p>Number of persons with blood group AB</p>
</dd>
</dl>



<h3>Source</h3>

<p>Taylor, G. L. and Prior, A. M. (1938)
Blood groups in England.  <em>Annals of Eugenics</em>, <b>8</b>, 343&ndash;355.
</p>

<hr>
<h2 id='breast'>Breast Cancer Data</h2><span id='topic+breast'></span>

<h3>Description</h3>

<p>Initial and follow-up status for 37 breast cancer patients treated for
spinal metastases.  The status is able to walk unaided (1), unable to
walk unaided (2), dead (3).  The follow-up times are 0, 3, 6, 12, 24, 
and 60 months after treatment began.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(breast)</code></pre>


<h3>Format</h3>

<p>A data frame with 37 observations on the following 8 variables.
</p>

<dl>
<dt>j</dt><dd><p>Case number</p>
</dd>
<dt>init</dt><dd><p>Initial status</p>
</dd>
<dt>x0</dt><dd><p>Status immediately after treatment started</p>
</dd>
<dt>x1</dt><dd><p>Status after 3 months</p>
</dd>
<dt>x2</dt><dd><p>Status after 6 months</p>
</dd>
<dt>x3</dt><dd><p>Status after 12 months</p>
</dd>
<dt>x4</dt><dd><p>Status after 24 months</p>
</dd>
<dt>x5</dt><dd><p>Status after 60 months</p>
</dd>
</dl>



<h3>Details</h3>

<p>Woman 24 was alive after 6 months
but her ability to walk was not recorded (she was in state 1 or 2).
</p>
<p>NA indicates that a woman has previously died, or that her status is
unknown.
</p>


<h3>Source</h3>

<p>de Stavola, B. L. (1988) Testing departures from time homogeneity in
multistate Markov processes.
<em>Applied Statistics</em>, <b>37</b>, 242&ndash;250.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 227.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(breast)
</code></pre>

<hr>
<h2 id='burt'>IQs of identical twins</h2><span id='topic+burt'></span>

<h3>Description</h3>

<p>These are said to be measurements of IQ scores for pairs of identical 
twins, the first raised by foster parents and the second raised 
by natural parents, published by Sir Cyril Burt. Cases are divided into groups according to 
parents' social class, A-C, labelled 1-3.
The general objective 
is to assess the impact of social class, and in particular the 
effect of environment,  on IQ.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(burt)</code></pre>


<h3>Format</h3>

<p>A data frame with 27 observations on the following 3 variables.
</p>

<dl>
<dt>y</dt><dd><p>IQ score for twin raised in foster home</p>
</dd>
<dt>x</dt><dd><p>IQ score for twin raised by natural parents</p>
</dd>
<dt>class</dt><dd><p>Social class of twins</p>
</dd>
</dl>



<h3>Details</h3>

<p>Burt used these and similar data to argue that IQ was largely inherited, a view which strongly
influenced British education through the creation of the 11+ exam, which was used to decide
which children should be given different forms of education.  However after his death it was
suggested that the data were fake, a view accepted by some and strongly rebutted by others.
</p>


<h3>Source</h3>

<p>Unpublished lecture notes of David Hinkley.
</p>


<h3>References</h3>

<p>For information about Burt, see www.indiana.edu/~intell/burt.shtml  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(burt)
attach(burt)
par(pty="s")
plot(x,y,type="n",xlim=c(60,140),ylim=c(60,140))
text(x,y,class,cex=0.8)
abline(0,1,lty=2)
</code></pre>

<hr>
<h2 id='cake'>Breaking of Chocolate Cakes</h2><span id='topic+cake'></span>

<h3>Description</h3>

<p>Data on breaking angles of chocolate cakes made using different recipes,
mixes, and cooking temperatures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cake)</code></pre>


<h3>Format</h3>

<p>A data frame with 270 observations on the following 4 variables.
</p>

<dl>
<dt>recipe</dt><dd><p>Recipe used</p>
</dd>
<dt>mix</dt><dd><p>mix, a factor with 15 levels</p>
</dd>
<dt>temp</dt><dd><p>temperature (degrees Fahrenheit) at which cake baked</p>
</dd>
<dt>y</dt><dd><p>breaking angle (degrees)</p>
</dd>
</dl>



<h3>Details</h3>

<p>These are data from an experiment in which six different
temperatures for cooking three recipes for chocolate cake were compared.  Each time a mix
was made using one of the recipes, enough batter was prepared for six cakes, which were
then randomly allocated to be cooked at the different temperatures.  
The response is the breaking angle, found by fixing one half of a slab of cake, then
pivoting the other half about the middle until breakage occurs.
</p>


<h3>Source</h3>

<p>Cochran, W. G. and Cox, G. M. (1959) <em>Experimental Designs</em>. Second
edition.  New York: Wiley.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 454.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cake)
</code></pre>

<hr>
<h2 id='calcium'>Calcium Uptake Data</h2><span id='topic+calcium'></span>

<h3>Description</h3>

<p>These are data on the uptake of calcium by cells suspended in a
radioactive solution, as a function of time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(calcium)</code></pre>


<h3>Format</h3>

<p>A data frame with 27 observations on the following 2 variables.
</p>

<dl>
<dt>time</dt><dd><p>The time (in minutes) that the cells were suspended in the
solution</p>
</dd>
<dt>cal</dt><dd><p>The amount of calcium uptake (nmoles/mg)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Howard Grimes from the Botany Department, North Carolina State
University, conducted an experiment for biochemical analysis of
intracellular storage and transport of calcium across plasma
membrane.  Cells were suspended in a solution of radioactive
calcium for a certain length of time and then the amount of
radioactive calcium that was absorbed by the cells was measured. 
The experiment was repeated independently with 9 different times
of suspension each replicated 3 times.
</p>


<h3>Source</h3>

<p>Rawlings, J.O. (1988) <em>Applied Regression Analysis</em>.  Wadsworth and
Brooks/Cole Statistics/Probability Series.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 469.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(calcium)
summary(nls(cal~beta0*(1-exp(-time/beta1)),data=calcium,start=list(beta0=5,beta1=5)))
</code></pre>

<hr>
<h2 id='cardiac'>Mortality Rates for Cardiac Surgery on Babies at 12 Hospitals</h2><span id='topic+cardiac'></span>

<h3>Description</h3>

<p>The title should be self-explanatory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cardiac)</code></pre>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 2 variables.
</p>

<dl>
<dt>r</dt><dd><p>Number of deaths</p>
</dd>
<dt>m</dt><dd><p>Number of operations</p>
</dd>
</dl>



<h3>Source</h3>

<p>Spiegelhalter, D. J., Thomas, A., Best, N. G. and W. R. Gilks
(1996)  <em>BUGS 0.5 Examples Volume 1 (Version ii)</em>.  
Cambridge: MRC Biostatistics Unit.  Page 15.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 579.</p>

<hr>
<h2 id='cat.heart'>Cat Heart Data</h2><span id='topic+cat.heart'></span>

<h3>Description</h3>

<p>Data from a Latin square experiment on the potencies of cardiac drugs given to
anesthetized cats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cat.heart)</code></pre>


<h3>Format</h3>

<p>A data frame with 64 observations on the following 6 variables.
</p>

<dl>
<dt>Day</dt><dd><p>on which experiment performed</p>
</dd>
<dt>Time</dt><dd><p>morning or afternoon</p>
</dd>
<dt>Observer</dt><dd><p>four observers took part</p>
</dd>
<dt>Drug</dt><dd><p>cardiac drug given to cat</p>
</dd>
<dt>y</dt><dd><p>100 times log dose in micrograms at which cat died</p>
</dd>
<dt>x</dt><dd><p>100 times log cat heart weight in grams</p>
</dd>
</dl>



<h3>Details</h3>

<p>These are results from an experiment to determine the relative potencies of 
eight similar cardiac drugs, labelled A&ndash;H, where A is a standard.  The method 
used was to infuse slowly a suitable dilution of the drug into an anaesthetized cat.  
The dose at which death occurred and the weight of the cat's heart were recorded.  
Four observers each made two determinations on each of eight days, 
with a Latin square design used to eliminate observer and time differences.  
The heart weight cannot be known at the start of the experiment, but might 
be  expected to affect comparisons
among the treatments; it is assumed that heart weight is unaffected by the treatments.</p>


<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College, London.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 447.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cat.heart)
anova(lm(y~Observer+Time+Day+Drug+Observer:Time,data=cat.heart))
</code></pre>

<hr>
<h2 id='cement'>Hald Cement Data</h2><span id='topic+cement'></span>

<h3>Description</h3>

<p>Heat evolved in setting of cement, as a function of its chemical
composition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cement)</code></pre>


<h3>Format</h3>

<p>A data frame with 13 observations on the following 5 variables.
</p>

<dl>
<dt>x1</dt><dd><p>percentage weight in clinkers of 3CaO.Al2O3</p>
</dd>
<dt>x2</dt><dd><p>percentage weight in clinkers of 3CaO.SiO2</p>
</dd>
<dt>x3</dt><dd><p>percentage weight in clinkers of 4CaO.Al2O3.Fe2O3</p>
</dd>
<dt>x4</dt><dd><p>percentage weight in clinkers of 2CaO.SiO2</p>
</dd>
<dt>y</dt><dd><p>heat evolved (calories/gram)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Woods, H., Steinour, H. H. and Starke, H. R. (1932) 
Effect of composition of Portland cement on heat evolved during hardening.
<em>Industrial Engineering and Chemistry</em>, <b>24</b>, 1207&ndash;1214.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 355.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cement)
lm(y~x1+x2+x3+x4,data=cement)
</code></pre>

<hr>
<h2 id='chicks'>Chick Bone Data</h2><span id='topic+chicks'></span>

<h3>Description</h3>

<p>Balanced incomplete block design on the effect of amino
acids on growth of chick bones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(chicks)</code></pre>


<h3>Format</h3>

<p>A data frame with 30 observations on the following 3 variables.
</p>

<dl>
<dt>Treat</dt><dd><p>Treatment with levels <code>All</code> (all amino acids present), 
<code>Arg-</code>(all acids present except Arg), etc.</p>
</dd>
<dt>Pair</dt><dd><p>bones were taken in pairs from 15 chicks</p>
</dd>
<dt>y</dt><dd><p>Log10 dry weight of bones at end of experiment</p>
</dd>
</dl>



<h3>Details</h3>

<p>Bones from 7-day-old chick embryos were cultivated over a nutrient
chemical medium.  Two bones were available from each chick, and the
experiment was set out in a balanced incomplete block design with two
units per block.  The treatments were growth in the complete medium,
with about 30 nutrients in carefully controlled quantities, and growth
in five other media, each with a single amino acid omitted.  Thus His-,
Arg-, and so forth denote media without particular amino acids.  
</p>


<h3>Source</h3>

<p>Cox, D. R. and Snell, E. J. (1981) 
<em>Applied Statistics: Principles and Examples</em>.
London: Chapman and Hall/CRC Press.  Page 95.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 432.
</p>

<hr>
<h2 id='chimps'>Chimpanzee Learning Data</h2><span id='topic+chimps'></span>

<h3>Description</h3>

<p>These are the times in minutes taken for four chimpanzees to learn each of four 
words.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(chimps)</code></pre>


<h3>Format</h3>

<p>A data frame with 40 observations on the following 3 variables.
</p>

<dl>
<dt>chimp</dt><dd><p>a factor with levels 1-4</p>
</dd>
<dt>word</dt><dd><p>a factor with 1-10</p>
</dd>
<dt>y</dt><dd><p>learning time (minutes)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Brown, B. W. and Hollander, M. (1977) <em>Statistics: A Biomedical
Introduction</em>. New York: Wiley.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 485.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chimps)
anova(glm(y~chimp+word,Gamma(log),data=chimps),test="F")
anova(glm(y~word+chimp,Gamma(log),data=chimps),test="F")
</code></pre>

<hr>
<h2 id='cloth'>Numbers of Flaws in Lengths of Cloth</h2><span id='topic+cloth'></span>

<h3>Description</h3>

<p>The data comprise lengths of cloth samples and the numbers of flaws found in
them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cloth)</code></pre>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 2 variables.
</p>

<dl>
<dt>x</dt><dd><p>The length of the roll of cloth.</p>
</dd>
<dt>y</dt><dd><p>The number of flaws found in the roll.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Bissell, A. F. (1972) A negative binomial model with varying element size. 
<em>Biometrika</em>, <b>59</b>, 435&ndash;441.</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 515.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cloth)
attach(cloth)
plot(x,y)
# Comparison of Poisson and quasilikelihood fits
summary(glm(y~x-1,family=poisson(identity)))
summary(glm(y~x-1,family=quasipoisson(identity)))
</code></pre>

<hr>
<h2 id='coal'>Data on UK coal mining disasters</h2><span id='topic+coal'></span>

<h3>Description</h3>

<p>The 'coal' data frame has 191 rows and 1 column.
</p>
<p>This data frame gives the dates of 191 explosions in UK coal mines
which resulted in 10 or more fatalities.  The time span of the
data is from March 15, 1851 until March 22 1962.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(coal)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following column:
</p>

<dl>
<dt>date</dt><dd><p>The date of the disaster.  The integer part of 'date' gives
the year.  The day is represented as the fraction of the year
that had elapsed on that day.
</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from
</p>
<p>Hand, D.J., Daly, F., Lunn, A.D., McConway, K.J. and Ostrowski, E.
(1994) <em>A Handbook of Small Data Sets</em>, Chapman and Hall.
</p>


<h3>References</h3>

<p>Jarrett, R.G. (1979) A note on the intervals between coal-mining
disasters. <em>Biometrika</em>, <b>66</b>, 191&ndash;193.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(coal)
plot(density(coal$date))
rug(coal$date)
</code></pre>

<hr>
<h2 id='coin.spin'>Function for Coin Spinning, Practical 11.1</h2><span id='topic+coin.spin'></span>

<h3>Description</h3>

<p>This function computes the posterior distribution of the success
probability theta when a coin is spun on its edge (or tossed), when the
prior density for that probability is a mixture of beta densities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coin.spin(para, r = 0, n = 0, n.points = 199)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coin.spin_+3A_para">para</code></td>
<td>
<p>A matrix with 3 columns and k rows, where k is the number of components of
the mixture.  The first column contains the probabilities, and the next two the shape parameters
a and b for the components.</p>
</td></tr>
<tr><td><code id="coin.spin_+3A_r">r</code></td>
<td>
<p> Number of successes</p>
</td></tr>
<tr><td><code id="coin.spin_+3A_n">n</code></td>
<td>
<p>Number of trials</p>
</td></tr>
<tr><td><code id="coin.spin_+3A_n.points">n.points</code></td>
<td>
<p>The number of values of theta, equally-spaced between 0 and 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is provided simply so that  readers spend less time typing. It is not intended to be
robust and general code.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p>Values of theta</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>Values of posterior density for theta</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anthony Davison (<code>anthony.davison@epfl.ch</code>)</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press. Practical 11.1.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## From Practical 11.1:
para &lt;- matrix( c(0.5, 10, 20, 0.5, 20, 10), nrow=2, ncol=3, byrow=TRUE)
prior &lt;- coin.spin(para)
plot(prior, xlab="theta",ylab="PDF", type="l",ylim=c(0,6))
post &lt;- coin.spin(para, r=4, n=10)
</code></pre>

<hr>
<h2 id='danish'>Danish Fire Insurance Claims</h2><span id='topic+danish'></span>

<h3>Description</h3>

<p>Data on major insurance claims due to fires in Denmark, 
1980&ndash;1990.  The values of the claims have been rescaled 
for commercial reasons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(beaver)</code></pre>


<h3>Format</h3>

<p>An irregular time series.
</p>


<h3>Source</h3>

<p>Embrechts, P., Kluppelberg, C. and Mikosch, T. (1997) 
<em>Modelling Extremal Events for Insurance and Finance</em>.
Berlin: Springer.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press.
Page 278.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(danish)
plot(danish,type="h")
</code></pre>

<hr>
<h2 id='darwin'>Darwin's Maize Data</h2><span id='topic+darwin'></span>

<h3>Description</h3>

<p>The heights in eighths of inches of young maize plants put by Charles Darwin in
four pots.  He planted 15 pairs of plants together, one of each pair being
cross-fertilised, and the other being self-fertilised.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(darwin)</code></pre>


<h3>Format</h3>

<p>A data frame with 30 observations on the following 4 variables.
</p>

<dl>
<dt>pot</dt><dd><p>a factor giving the pot</p>
</dd>
<dt>pair</dt><dd><p>a factor giving the pair</p>
</dd>
<dt>type</dt><dd><p>a factor giving the type of fertilisation</p>
</dd>
<dt>height</dt><dd><p>height of plant in eighths of inches</p>
</dd>
</dl>



<h3>Source</h3>

<p>Fisher, R. A. (1935)  <em>Design of Experiments</em>.  Edinburgh: Oliver and Boyd.  Page 30.
</p>


<h3>References</h3>

<p>The original book is reprinted as part of 
Fisher, R. A. (1990)  <em>Statistical Methods, Experimental Design, 
and Scientific Inference</em>.  Oxford University Press.  
</p>
<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 2.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(darwin)
attach(darwin)
plot(height~type)
anova(lm(height~pot+pair+type,data=darwin))</code></pre>

<hr>
<h2 id='exp.gibbs'>Gibbs Sampling for Two Truncated Exponential Variables, Practical 11.3</h2><span id='topic+exp.gibbs'></span>

<h3>Description</h3>

<p>Performs Gibbs sampling for problem with two truncated exponential variables.
See Practical 11.3 of Davison (2003) for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp.gibbs(u1 = NULL, u2 = NULL, B, I = 100, S = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exp.gibbs_+3A_u1">u1</code></td>
<td>
<p>Initial values for variable 1 </p>
</td></tr>
<tr><td><code id="exp.gibbs_+3A_u2">u2</code></td>
<td>
<p>Initial values for variable 2</p>
</td></tr>
<tr><td><code id="exp.gibbs_+3A_b">B</code></td>
<td>
<p>Value at which exponential distribution is truncated</p>
</td></tr>
<tr><td><code id="exp.gibbs_+3A_i">I</code></td>
<td>
<p>Number of iterations of sampler</p>
</td></tr>
<tr><td><code id="exp.gibbs_+3A_s">S</code></td>
<td>
<p>Number of replicates of sampler</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is provided simply so that  readers spend less time typing. It is not intended to be
robust and general code.
</p>


<h3>Value</h3>

<p>A 2 x S x I array containing the values of the variables for the successive
iterations
</p>


<h3>Author(s)</h3>

<p>Anthony Davison (<code>anthony.davison@epfl.ch</code>)</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press.Practical 11.3.</p>


<h3>Examples</h3>

<pre><code class='language-R'>add.exp.lines &lt;- function( exp.out, i, B=10)
{
  dexp.trunc &lt;- function( u, lambda, B ) 
     dexp(u, rate=lambda)/(1-exp(-lambda*B))
  S &lt;- dim(exp.out)[2]
  I &lt;- dim(exp.out)[3]
  u &lt;- seq(0.0001,B,length=1000)
  fu &lt;- rep(0,1000)
  for (s in 1:S) fu &lt;- fu + dexp.trunc(u,exp.out[3-i,s,I],B)/S
  lines(u,fu,col="red")
  invisible()
}
par(mfrow=c(3,2))
B &lt;-10; I &lt;- 15; S &lt;- 500
exp.out &lt;- exp.gibbs(B=B,I=I,S=S)
hist(exp.out[1,,I],prob=TRUE,nclass=15,xlab="u1",ylab="PDF",xlim=c(0,B),ylim=c(0,1))
add.exp.lines(exp.out,1)
hist(exp.out[2,,I],prob=TRUE,nclass=15,xlab="u2",ylab="PDF",xlim=c(0,B),ylim=c(0,1))
add.exp.lines(exp.out,2)
</code></pre>

<hr>
<h2 id='eyes'>Visual Impairment Data</h2><span id='topic+eyes'></span>

<h3>Description</h3>

<p>Joint distribution of visual impairment on both eyes, by race and age.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eyes)</code></pre>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 6 variables.
</p>

<dl>
<dt>L</dt><dd><p>Impairment (+) or not (-) for left eye.</p>
</dd>
<dt>R</dt><dd><p>Impairment (+) or not (-) for right eye.</p>
</dd>
<dt>age</dt><dd><p>a factor with levels <code>40-50</code> <code>51-60</code> 
<code>61-70</code> <code>70+</code></p>
</dd>
<dt>colour</dt><dd><p>White (W) or black (B)</p>
</dd>
<dt>a</dt><dd><p>mid-point for age groups, as numeric vector</p>
</dd>
<dt>y</dt><dd><p>Number of individuals in each class</p>
</dd>
</dl>



<h3>Source</h3>

<p>K.-Y. Liang, S. L. Zeger and B. Qaqish (1992)  
Multivariate regression analyses for categorical data (with Discussion).
<em>Journal of the Royal Statistical Society, series B</em>, <b>54</b>, 3&ndash;40.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 505.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eyes)
eyes.glm &lt;- glm(y~age*colour+L*R+(L+R):poly(a,2)+colour:(L+R),poisson,data=eyes)
anova(eyes.glm,test="Chi")   # analysis of deviance for loglinear model
</code></pre>

<hr>
<h2 id='field.concrete'>Field Concrete Mixture Data</h2><span id='topic+field.concrete'></span>

<h3>Description</h3>

<p>Data from a 4x4 Latin square experiment on the efficiency of
a field concrete mixer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(field.concrete)</code></pre>


<h3>Format</h3>

<p>A data frame with 16 observations on the following 4 variables.
</p>

<dl>
<dt>efficiency</dt><dd><p>a numeric vector</p>
</dd>
<dt>speed</dt><dd><p>a factor with levels 4, 8, 12, 16 mph.</p>
</dd>
<dt>run</dt><dd><p>order in which runs were performed each day</p>
</dd>
<dt>day</dt><dd><p>day on which runs were performed</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College, London.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 435.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(field.concrete)
fit &lt;- lm(efficiency~run+day+speed,data=field.concrete)
anova(fit)
summary(fit)
fit &lt;- lm(efficiency~run+day+poly(as.numeric(speed),3),data=field.concrete)
summary(fit)
</code></pre>

<hr>
<h2 id='fir'>Counts of Balsam-fir Seedlings</h2><span id='topic+fir'></span>

<h3>Description</h3>

<p>The number of balsam-fir seedlings in each quadrant of a grid of
50 five foot  square quadrants were counted.  The grid consisted
of 5 rows of 10 quadrants in each row.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fir)</code></pre>


<h3>Format</h3>

<p>A data frame with 50 observations on the following 3 variables.
</p>

<dl>
<dt>count</dt><dd><p>The number of seedlings in the quadrant</p>
</dd>
<dt>row</dt><dd><p>The row number of the quadrant</p>
</dd>
<dt>col</dt><dd><p>The quadrant number within the row</p>
</dd>
</dl>



<h3>Source</h3>

<p>Davison, A.C. and Hinkley, D.V. (1997)  <em>Bootstrap Methods and
Their Application</em>. Cambridge University Press.</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 641.</p>

<hr>
<h2 id='forbes'>Atmospheric Pressure and Boiling Point in the Alps</h2><span id='topic+forbes'></span>

<h3>Description</h3>

<p>James Forbes measured the atmospheric pressure and boiling
point of water at 17 locations in the Alps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(forbes)</code></pre>


<h3>Format</h3>

<p>A data frame with 17 observations on the following 2 variables.
</p>

<dl>
<dt>bp</dt><dd><p>Boiling point (Fahrenheit)</p>
</dd>
<dt>pres</dt><dd><p>Pressure (inches of mercury)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Atkinson, A. C. (1985)  <em>Plots, Transformations, and Regression</em>.  
Oxford University Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(forbes)
plot(forbes)
fit &lt;- lm(bp~pres,data=forbes)
fit
plot(forbes$pres,resid(fit))  # model OK?
# try refitting with transformation
fit &lt;- lm(log(bp)~log(pres),data=forbes)
</code></pre>

<hr>
<h2 id='frets'>Head Dimensions in Brothers</h2><span id='topic+frets'></span>

<h3>Description</h3>

<p>The 'frets' data frame has 25 rows and 4 columns.
</p>
<p>The data consist of measurements of the length and breadth of the
heads of  pairs of adult brothers in 25 randomly sampled families.
All measurements are expressed in millimetres.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(frets)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>l1</dt><dd><p>The head length of the eldest son.</p>
</dd>
<dt>b1</dt><dd><p>The head breadth of the eldest son.</p>
</dd>
<dt>l2</dt><dd><p>The head length of the second son.</p>
</dd>
<dt>b2</dt><dd><p>The head breadth of the second son.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Frets, G.P. (1921) Heredity of head form in man. <em>Genetica</em>, <b>3</b>, 193.
</p>


<h3>References</h3>

<p>Whittaker, J. (1990) <em>Graphical Models in Applied Multivariate
Statistics</em>. John Wiley.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(frets)
## maybe str(frets) ; plot(frets) ...
</code></pre>

<hr>
<h2 id='ftse'>FTSE Daily Returns</h2><span id='topic+ftse'></span>

<h3>Description</h3>

<p>Daily returns (
index, 1991&ndash;1998.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ftse)</code></pre>


<h3>Format</h3>

<p>A time series.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press. Page 266.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ftse)
plot(ftse,type="l",xlab="Time",ylab="Percent return")
plot(exp(cumsum(ftse/100)),type="l",xlab="Time",ylab="Relative closing value")
</code></pre>

<hr>
<h2 id='galaxy'>Galaxy Velocity Data</h2><span id='topic+galaxy'></span>

<h3>Description</h3>

<p>Velocities (km/second) of 82 galaxies in a survey of the Corona
Borealis region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(galaxy)</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:82] 9.17 9.35 9.48 9.56 9.78 ...
</p>


<h3>Source</h3>

<p>Roeder, K. (1990) Density estimation with confidence sets exemplified by
superclusters and voids in galaxies.
<em>Journal of the American Statistical Association</em>, <b>85</b>, 617&ndash;624.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 214.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(galaxy)
plot(density(galaxy))
rug(galaxy)
</code></pre>

<hr>
<h2 id='get.alpha'>Estimate Alpha from Data</h2><span id='topic+get.alpha'></span>

<h3>Description</h3>

<p>Estimate prior value of a parameter from the data.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.alpha(d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.alpha_+3A_d">d</code></td>
<td>
<p>A data frame with vector components <code>y</code> and <code>x</code> of the same length.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prior value of a parameter, estimated from the data.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+poi.beta.laplace">poi.beta.laplace</a></code></p>

<hr>
<h2 id='glm.diag'>Generalized Linear Model Diagnostics</h2><span id='topic+glm.diag'></span>

<h3>Description</h3>

<p>Calculates jackknife deviance residuals, standardized deviance residuals, standardized Pearson residuals, approximate Cook statistic, leverage and estimated dispersion.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'diag'
glm(glmfit)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glm.diag_+3A_glmfit">glmfit</code></td>
<td>
 <p><code>glmfit</code> is a <code>glm.object</code> or a <code>lm.object</code> - the result of a call to <code>glm()</code> or <code>lm()</code> </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following items:
</p>
<table role = "presentation">
<tr><td><code>res</code></td>
<td>
<p>The vector of jackknife deviance residuals.</p>
</td></tr>
<tr><td><code>rd</code></td>
<td>
<p>The vector of standardized deviance residuals.</p>
</td></tr>
<tr><td><code>rp</code></td>
<td>
<p>The vector of standardized Pearson residuals.</p>
</td></tr>
<tr><td><code>cook</code></td>
<td>
<p>The vector of approximate Cook statistics.</p>
</td></tr>
<tr><td><code>h</code></td>
<td>
<p>The vector of leverages of the observations.</p>
</td></tr>
<tr><td><code>sd</code></td>
<td>
<p>The value used to standardize the residuals. This is the the estimate of residual standard deviation in the Gaussian family and is the square root of the estimated shape parameter in the Gamma family. In all other cases it is 1.</p>
</td></tr>
</table>


<h3>Note</h3>

<p> See the helpfile for <code>glm.diag.plots</code> for an example of the use of <code>glm.diag</code>. </p>


<h3>Author(s)</h3>

<p> Anthony Davison &lt;anthony.davison@epfl.ch&gt; </p>


<h3>References</h3>

<p> Davison, A.C. and Snell, E.J. (1991) Residuals and diagnostics. In <em>Statistical Theory and Modelling: In Honour of Sir David Cox</em>. D.V. Hinkley, N. Reid and E.J. Snell (editors), 83-106. Chapman and Hall.</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">glm</a></code>,<code><a href="stats.html#topic+lm">lm</a></code>,<code><a href="#topic+plot.glm.diag">plot.glm.diag</a></code>,<code><a href="stats.html#topic+summary.glm">summary.glm</a></code> </p>

<hr>
<h2 id='hus'>Haemolytic Uraemic Syndrome</h2><span id='topic+hus'></span>

<h3>Description</h3>

<p>Annual numbers of cases of 'diarrhoea-associated haemolytic
uraemic syndrome' treated in clinics in Birmingham and Newcastle
from 1970&ndash;1989.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hus)</code></pre>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 3 variables.
</p>

<dl>
<dt>year</dt><dd><p>a numeric vector</p>
</dd>
<dt>birmingham</dt><dd><p>Number of cases treated in Birmingham</p>
</dd>
<dt>newcastle</dt><dd><p>Number of cases treated in Newcastle</p>
</dd>
</dl>



<h3>Source</h3>

<p>Henderson, R. and Matthews, J. N. S. (1993) An investigation of 
changepoints in the annual number of cases of haemolytic 
uraemic syndrome.  <em>Applied Statistics</em>, <b>42</b>, 461&ndash;471.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 142.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hus)
plot(hus$year,hus$birmingham,ylab="Annual number of cases",type="s")
</code></pre>

<hr>
<h2 id='hus.gibbs'>Gibbs Sampler for Poisson Changepoint Model, Practical 11.6</h2><span id='topic+hus.gibbs'></span>

<h3>Description</h3>

<p>This function implements a Gibbs sampler for the Poisson changepoint
model applied to the HUS data used in Example 4.40 and Practical 11.6
of Davison (2003), which should be consulted for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hus.gibbs(init, y, R = 10, a1 = 1, a2 = 1, c = 0.01, d = 0.01)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hus.gibbs_+3A_init">init</code></td>
<td>
<p>Initial values for parameters</p>
</td></tr>
<tr><td><code id="hus.gibbs_+3A_y">y</code></td>
<td>
<p>A series of Poisson counts</p>
</td></tr>
<tr><td><code id="hus.gibbs_+3A_r">R</code></td>
<td>
<p>Number of iterations of sampler</p>
</td></tr>
<tr><td><code id="hus.gibbs_+3A_a1">a1</code></td>
<td>
<p>Value of a hyperparameter</p>
</td></tr>
<tr><td><code id="hus.gibbs_+3A_a2">a2</code></td>
<td>
<p>Value of a hyperparameter</p>
</td></tr>
<tr><td><code id="hus.gibbs_+3A_c">c</code></td>
<td>
<p>Value of a hyperparameter</p>
</td></tr>
<tr><td><code id="hus.gibbs_+3A_d">d</code></td>
<td>
<p>Value of a hyperparameter</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is provided simply so that  readers spend less time typing. It is not intended to be
robust and general code.
</p>


<h3>Value</h3>

<p>A matrix of size R x 7, whose first five columns contain the values of the
parameters for the iterations.  Columns 6 and 7 contain the log likelihood and log prior
for that iteration.
</p>


<h3>Author(s)</h3>

<p>Anthony Davison (<code>anthony.davison@epfl.ch</code>)</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press. Practical 11.6.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## From Example 11.6:
hus &lt;- c(1,5,3,2,2,1,0,0,2,1,1,7,11,4,7,10,16,16,9,15)
system.time( gibbs.out &lt;- hus.gibbs(c(5, 5, 1, 1, 2), hus, R=1000))
plot.ts(gibbs.out[,1], main="lambda1") # time series plot for lam1
plot.ts(gibbs.out[,2], main="lambda1") # time series plot for lam2
plot.ts(gibbs.out[,6], main="log lik") # and of log likelihood
table(gibbs.out[,5])  # tabulate observed values of tau
rm(hus)
</code></pre>

<hr>
<h2 id='ihess'>Inverse Hessian</h2><span id='topic+ihess'></span>

<h3>Description</h3>

<p>Inverse Hessian matrix, useful for obtaining standard errors</p>


<h3>Usage</h3>

<pre><code class='language-R'>ihess(f, x, ep = 1e-04, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ihess_+3A_f">f</code></td>
<td>
<p>Usually a negative log likelihood</p>
</td></tr>
<tr><td><code id="ihess_+3A_x">x</code></td>
<td>
<p>Usually maximum likelihood estimates for f</p>
</td></tr>
<tr><td><code id="ihess_+3A_ep">ep</code></td>
<td>
<p>Step length used to compute numerical second derivatives</p>
</td></tr>
<tr><td><code id="ihess_+3A_...">...</code></td>
<td>
<p>Extra arguments for f, if any</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of dimension dim(x) times dim(x), containing inverse Hessian
matrix of f at x.
</p>


<h3>Note</h3>

<p>This is not needed in R, where hessian matrices are obtained by setting hessian=T in
calls to optimisation functions.</p>


<h3>Author(s)</h3>

<p>Anthony Davison</p>


<h3>References</h3>

<p>Based on code written by Stuart Coles of Padova University</p>


<h3>Examples</h3>

<pre><code class='language-R'># ML fit of t distribution
nlogL &lt;- function(x, data) # negative log likelihood
{ mu &lt;- x[1]
  sig &lt;- x[2]
  df &lt;- x[3]
  -sum(log( dt((data-mu)/sig, df=df)/sig )) }
y &lt;- rt(n=100, df=10) # generate t data
# this is Splus code.....so remove the #'s for it to work in R
# fit &lt;- nlminb(c(1,1,4), nlogL, upper=c(Inf,Inf,Inf), lower=c(-Inf,0,0),
#               data=y)
# fit$parameters # maximum likelihood estimates
# J &lt;- ihess(nlogL, fit$parameters, data=y)
#  sqrt(diag(J)) # standard errors based on observed information
# 
# In this example the standard error can be a bad measure of
# uncertainty for the df.
</code></pre>

<hr>
<h2 id='intron'>Intron Gene Sequence</h2><span id='topic+intron'></span>

<h3>Description</h3>

<p>Sequence of 1572 bases from first human preproglucagon gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(intron)</code></pre>


<h3>Format</h3>

<p>This is a factor with 4 levels,  &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;
</p>


<h3>Source</h3>

<p>Avery, P. J. and Henderson, D. A. (1999) Fitting Markov chain models to
discrete state series such as DNA sequences.
<em>Applied Statistics</em>, <b>48</b>, 53&ndash;61.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 226.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(intron)
## maybe str(intron) ; plot(intron) ...
</code></pre>

<hr>
<h2 id='jacamar'>Jacamar Learning Ability Data</h2><span id='topic+jacamar'></span>

<h3>Description</h3>

<p>Response of a rufous-tailed jacamar to butterflies, by not attacking them,
by attacking but not eating them, and by attacking and eating them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(jacamar)</code></pre>


<h3>Format</h3>

<p>A data frame with 48 observations on the following 5 variables.
</p>

<dl>
<dt>species</dt><dd><p>Butterfly species: Aphrissa boisduvalli (Ab), 
Phoebis argante (Pa), Dryas iulia (Di),  Pierella luna (Pl), Consul fabius (Cf), 
Siproeta stelenes (Ss)</p>
</dd>
<dt>colour</dt><dd><p>colour butterfly wings were painted: Unpainted, Brown, 
Yellow, Blue, Green, Red, Orange, Black</p>
</dd>
<dt>N</dt><dd><p>Number not attacked</p>
</dd>
<dt>S</dt><dd><p>Number attacked but rejected</p>
</dd>
<dt>E</dt><dd><p>Number eaten</p>
</dd>
</dl>



<h3>Details</h3>

<p>As part of a study of the learning ability of tropical birds, Peng Chai 
of the University of Texas at
Austin collected data on the response of a rufous-tailed jacamar
to butterflies.  He used marker pens to paint the underside 
of the wings of eight species of butterflies, and then 
released each butterfly in the cage where the bird was confined.  
The bird responded in three ways: by not attacking the butterfly
(N); by attacking the butterfly, then sampling but rejecting it 
(S); or by attacking and eating the butterfly, usually after 
removing some or all of the wings (E).
</p>


<h3>Source</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 470.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jacamar)
</code></pre>

<hr>
<h2 id='jelinski'>Jelinski and Moranda Data on Software Failures</h2><span id='topic+jelinski'></span>

<h3>Description</h3>

<p>These are times in days between sucessive failures of a piece of
software developed as part of a large data system.  The software
was released after the first 31 failures.  The last three failures
occurred after release.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(jelinski)</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:34] 9 12 11 4 7 2 5 8 5 7 ...
</p>


<h3>Source</h3>

<p>Jelinski, Z. and Moranda, P. B. (1972)  
Software reliability research.  In W. Freiberger (ed), 
<em>Statistical Computer Performance Evaluation</em>.  London: 
Academic Press.  Pages 465&ndash;484.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 299.</p>

<hr>
<h2 id='leuk'>Survival Times and White Blood Counts for Leukaemia Patients</h2><span id='topic+leuk'></span>

<h3>Description</h3>

<p>A data frame of data from 33 leukaemia patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(leuk)</code></pre>


<h3>Format</h3>

<p>A data frame with 33 observations on the following 3 variables.
</p>

<dl>
<dt>wbc</dt><dd><p>white blood cell count</p>
</dd>
<dt>ag</dt><dd><p>a test result, '&quot;present&quot;' or '&quot;absent&quot;'</p>
</dd>
<dt>time</dt><dd><p>survival time in weeks</p>
</dd>
</dl>



<h3>Details</h3>

<p>Survival times are given for 33 patients who died from acute
myelogenous leukaemia.  Also measured was the patient's white
blood cell count at the time of diagnosis.  The patients were also
factored into 2 groups according to the presence or absence of a
morphologic characteristic of white blood cells. Patients termed
AG positive were identified by the presence of Auer rods and/or
significant granulation of the leukaemic cells in the bone marrow
at the time of diagnosis.
</p>


<h3>Source</h3>

<p>Feigl, P. and Zelen, M. (1965) Estimation of exponential survival
probabilities with concomitant information. Biometrics, 21,
826&ndash;838.
</p>


<h3>References</h3>

<p>Cox, D. R. and Oakes, D. (1984) <em>Analysis of Survival Data</em>.    
Chapman &amp; Hall, p. 9.
</p>
<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 542.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leuk)
library(survival)
plot(survfit(Surv(time) ~ ag, data = leuk), lty = 2:3, col = 2:3)
# fit of exponential model
summary(glm(time~ag+log10(wbc),data=leuk,family=Gamma(log)),dispersion=1)
# now Cox models
leuk.cox &lt;- coxph(Surv(time) ~ ag + log(wbc), leuk)
summary(leuk.cox)
</code></pre>

<hr>
<h2 id='lik.ci'>Likelihood Confidence Intervals for Scalar Parameter</h2><span id='topic+lik.ci'></span>

<h3>Description</h3>

<p>A simple function for computing confidence intervals from the
values of a likelihood function for a scalar parameter.  
It prints the maximum likelihood estimate (MLE) and its standard error, 
and confidence intervals based on normal approximation to the
distribution of the MLE and on the chi-squared approximation
to the distribution of the likelihood ratio statistic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lik.ci(psi, logL, conf = c(0.975, 0.025))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lik.ci_+3A_psi">psi</code></td>
<td>
<p>Vector containing parameter values, the range of which contains the MLE</p>
</td></tr>
<tr><td><code id="lik.ci_+3A_logl">logL</code></td>
<td>
<p>Vector containing corresponding log likelihood values</p>
</td></tr>
<tr><td><code id="lik.ci_+3A_conf">conf</code></td>
<td>
<p>Vector containing levels for which confidence interval limits needed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See above
</p>


<h3>Note</h3>

<p>This uses the spline functions in library(modreg).</p>


<h3>Author(s)</h3>

<p>Anthony Davison (Anthony.Davison@epfl.ch)</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge 
University Press.  Sections 4.4.2, 4.5.1.</p>


<h3>Examples</h3>

<pre><code class='language-R'># likelihood analysis for mean of truncated Poisson data
y &lt;- c(1:6)
n &lt;- c(1486,694,195,37,10,1)
logL &lt;- function(x, y, n.obs)      # x is theta
{  f &lt;- dpois(y,x)/(1-dpois(0,x))  # dpois is Poisson PDF
   sum(n*log(f))  }                # log likelihood
theta &lt;- seq(from=0.8, to=1, length=200)
L &lt;- rep(NA, 200)
for (i in 1:200) L[i] &lt;- logL(theta[i], y, n)
plot(theta, L, type="l", ylab="Log likelihood")
lik.ci(theta, L)
</code></pre>

<hr>
<h2 id='limits'>Swedish Speed Limit Data</h2><span id='topic+limits'></span>

<h3>Description</h3>

<p>The data are numbers of traffic accidents with personal injuries, reported to the police, on Swedish 
roads on 92 days in 1961 and 92 matching days in 1962.  On some of these days a general speed limit
of 90 or 100 km/hour was imposed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(limits)</code></pre>


<h3>Format</h3>

<p>A data frame with 92 observations on the following 5 variables.
</p>

<dl>
<dt>day</dt><dd><p>A factor indicating the day, coded <code>1</code>-<code>92</code>.</p>
</dd>
<dt>lim1</dt><dd><p>1 indicates a limit imposed in 1961, 0 not.</p>
</dd>
<dt>lim2</dt><dd><p>1 indicates a limit imposed in 1962, 0 not.</p>
</dd>
<dt>y1</dt><dd><p>Number of accidents on this day in 1961.</p>
</dd>
<dt>y2</dt><dd><p>Number of accidents on this day in 1962.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Svensson, A. (1981)  On a goodness-of-fit test for multiplicative Poisson models.  
<em>Annals of Statistics</em>, <b>9</b>, 697&ndash;704.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(limits)
## maybe str(limits) ; plot(limits) ...
</code></pre>

<hr>
<h2 id='lizards'>Lizard Count Data</h2><span id='topic+lizards'></span>

<h3>Description</h3>

<p>These are data on the structural habitat of two species of lizards in 
Whitehouse, Jamaica.  They comprise
observed counts for perch height, perch diameter, insolation, 
and time of day, for both species.  The
data can be represented as a 2 x 2 x 2 x 3 x 2 contingency table. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lizards)</code></pre>


<h3>Format</h3>

<p>A data frame with 48 observations on the following 6 variables.
</p>

<dl>
<dt>height</dt><dd><p><code>high</code> indicates perch at height 
5 or more feet, <code>low</code> indicates perch below 5 feet.</p>
</dd>
<dt>diameter</dt><dd><p><code>large</code> indicates perch diameter 2 inches or more, 
<code>small</code> indicates perch diameter less than 2 inches.</p>
</dd>
<dt>sun</dt><dd><p>Is the perch in a <code>shady</code> or a <code>sunny</code> location?</p>
</dd>
<dt>time</dt><dd><p>Time of day when lizard observed: <code>early</code>, 
<code>late</code> or <code>midday</code>.</p>
</dd>
<dt>species</dt><dd><p>Species of lizard: <code>grahami</code> or <code>opalinus</code>.</p>
</dd>
<dt>y</dt><dd><p>Number of lizards seen.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Bishop, Y. M. M., Fienberg, S. E. and Holland, P. W. (1975)  
<em>Discrete Multivariate Analysis</em>.  
Cambridge, Mass.: MIT Press.  Page 164.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lizards)
## maybe str(lizards) ; plot(lizards) ...
</code></pre>

<hr>
<h2 id='lung.cancer'>Lung Cancer Deaths among UK Physicians</h2><span id='topic+lung.cancer'></span>

<h3>Description</h3>

<p>The data give the number of deaths due to lung cancer in British
male physicians, as a function of years of smoking and cigarette consumption.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lung.cancer)</code></pre>


<h3>Format</h3>

<p>A data frame with 63 observations on the following 4 variables.
</p>

<dl>
<dt>years.smok</dt><dd><p>a factor giving the number of years smoking</p>
</dd>
<dt>cigarettes</dt><dd><p>a factor giving cigarette consumption</p>
</dd>
<dt>Time</dt><dd><p>man-years at risk</p>
</dd>
<dt>y</dt><dd><p>number of deaths</p>
</dd>
</dl>



<h3>Source</h3>

<p>Frome, E. L. (1983)  The analysis of rates using Poisson regression models.
<em>Biometrics</em>, <b>39</b>, 665&ndash;674.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.  Page 8.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lung.cancer)
</code></pre>

<hr>
<h2 id='magnesium'>Magnesium Treatment for Heart Attack Patients</h2><span id='topic+magnesium'></span>

<h3>Description</h3>

<p>Data from 11 clinical trials to compare magnesium treatment for
heart attacks with control.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(magnesium)</code></pre>


<h3>Format</h3>

<p>A data frame with 22 observations on the following 4 variables.
</p>

<dl>
<dt>trial</dt><dd><p>a factor with levels <code>1</code>&ndash;<code>11</code></p>
</dd>
<dt>group</dt><dd><p>Treatment indicator (factor)</p>
</dd>
<dt>m</dt><dd><p>Total patients in group</p>
</dd>
<dt>r</dt><dd><p>Number of deaths in group</p>
</dd>
</dl>



<h3>Source</h3>

<p>Copas, J. B. (1999) What works?: Selectivity models and meta-analysis.
<em>Journal of the Royal Statistical Society series A</em>, <b>162</b>, 96&ndash;109.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  Statistical Models.  Cambridge University Press.
Page 208.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(magnesium)
fit &lt;- glm(cbind(r,m-r)~trial+group,binomial,data=magnesium[1:20,])
anova(fit,test="Chi")
summary(fit)
</code></pre>

<hr>
<h2 id='manaus'>Average Heights of the Rio Negro river at Manaus</h2><span id='topic+manaus'></span>

<h3>Description</h3>

<p>The 'manaus' time series is of class '&quot;ts&quot;' and has 1080
observations on one variable, which is the monthly average
of the daily stages (heights) of the Rio Negro at Manaus,
from January 1903 until December 1992. The units are metres.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(manaus)</code></pre>


<h3>Format</h3>

<p>A time series
</p>


<h3>Details</h3>

<p>The data values are monthly averages of the daily stages (heights)
of the Rio Negro at Manaus.  Manaus is 18km upstream from the
confluence of the Rio Negro with the Amazon but because of the
tiny slope of the water surface and the lower courses of its
flatland affluents, they may be regarded as a good approximation
of the water level in the Amazon at the confluence.  The data here
cover 90 years from January 1903 until December 1992.
</p>
<p>The Manaus gauge is tied in with an arbitrary bench mark of 100m
set in the steps of the Municipal Prefecture; gauge readings are
usually referred to sea level, on the basis of a mark on the steps
leading to the Parish Church (Matriz), which is assumed to lie at
an altitude of 35.874 m according to observations made many years
ago under the direction of Samuel Pereira, an engineer in charge
of the Manaus Sanitation Committee Whereas such an altitude
cannot, by any means, be considered to be a precise datum point,
observations have been provisionally referred to it.  The
measurements are in metres.
</p>


<h3>Source</h3>

<p>The data were kindly made available by Professors H. O'Reilly
Sternberg and  D. R. Brillinger of the University of California at
Berkeley.
</p>


<h3>References</h3>

<p>Davison, A.C. and Hinkley, D.V. (1997) <em>Bootstrap Methods and
Their Application</em>. Cambridge University Press.
</p>
<p>Sternberg, H. O'R. (1987) Aggravation of floods in the Amazon
river as a consequence of deforestation? <em>Geografiska Annaler</em>,
69A, 201-219.
</p>
<p>Sternberg, H. O'R. (1995) Waters and wetlands of Brazilian
Amazonia: An uncertain future. In <em>The Fragile Tropics of Latin
America: Sustainable Management of Changing Environments</em>,
Nishizawa, T. and Uitto, J.I. (editors), United Nations University
Press, 113-179.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(manaus)
plot(manaus)
acf(manaus)
pacf(manaus)
</code></pre>

<hr>
<h2 id='marking'>Examination Marking Data</h2><span id='topic+marking'></span>

<h3>Description</h3>

<p>The data are from an experiment to compare how different markers assess examination
scripts, some of which were original and others of which were photocopies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(marking)</code></pre>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 5 variables.
</p>

<dl>
<dt>Exam</dt><dd><p>Two exams were marked</p>
</dd>
<dt>Script</dt><dd><p>Scripts from 8 persons were marked, coded <code>1</code>-<code>8</code>.</p>
</dd>
<dt>Marker</dt><dd><p>Coded <code>1</code>-<code>4</code></p>
</dd>
<dt>Original</dt><dd><p>Is the script an original (1) or a photocopy (0)?</p>
</dd>
<dt>y</dt><dd><p>The mark out of 80 attributed by the marker.</p>
</dd>
</dl>



<h3>Details</h3>

<p>Normally each marker had a different batch of
scripts, but for the experiment one script was taken at random from each batch
and replaced after three copies of it had been made.  The three copies
were sent to the other three markers who assessed them, while the 
original was replaced and assessed in the usual way.  Each of the four
copies was therefore assessed by a single marker, but the three markers
who had a copy knew that the script was part of the experiment, while
the person marking the original did not know it to be part of the
experiment.  The experiment was repeated at another examination, with
the same examiners, but different scripts.  </p>


<h3>Source</h3>

<p>Lindley, D. V. (1961) An experiment in the marking of an examination (with Discussion).
<em>Journal of the Royal Statistical Society, series A</em>, <b>124</b>, 285&ndash;313.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(marking)
## maybe str(marking) ; plot(marking) ...
</code></pre>

<hr>
<h2 id='mathmarks'>Math Marks Data</h2><span id='topic+mathmarks'></span>

<h3>Description</h3>

<p>Marks out of 100 for 88 students taking examinations in
mechanics (C), vectors (C), algebra (O), analysis (O),
statistics (O), where C indicates closed and O indicates 
open book examination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mathmarks)</code></pre>


<h3>Format</h3>

<p>A data frame with 88 observations on the following 5 variables.
</p>

<dl>
<dt>mechanics</dt><dd><p>mark out of 100 for mechanics</p>
</dd>
<dt>vectors</dt><dd><p>mark out of 100 for vectors</p>
</dd>
<dt>algebra</dt><dd><p>mark out of 100 for algebra</p>
</dd>
<dt>analysis</dt><dd><p>mark out of 100 for analysis</p>
</dd>
<dt>statistics</dt><dd><p>mark out of 100 for statistics</p>
</dd>
</dl>



<h3>Source</h3>

<p>Mardia, K. V., Kent, J. T. and Bibby, J. M. (1979) 
<em>Multivariate  Analysis</em>.  London: Academic Press.  Pages 3&ndash;4.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 256.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mathmarks)
pairs(mathmarks)
var(mathmarks)
</code></pre>

<hr>
<h2 id='MClik'>Likelihood Estimation for Markov Chains</h2><span id='topic+MClik'></span>

<h3>Description</h3>

<p>Computes maximum likelihood estimates of transition probabilities
for stationary Markov chain models, of order 0 (independence) to 3.
</p>
<p>This is intended for use with Practical 6.1 of Davison (2003), not as production
code.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MClik(d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MClik_+3A_d">d</code></td>
<td>
<p>A sequence containing successive states of the chain</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>order</code></td>
<td>
<p>order of fitted chain</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>degrees of freedom using in fitting</p>
</td></tr>
<tr><td><code>L</code></td>
<td>
<p>maximum log likelihood for each order</p>
</td></tr>
<tr><td><code>AIC</code></td>
<td>
<p>Akaike information criterion for each order</p>
</td></tr>
<tr><td><code>one</code></td>
<td>
<p>one-way marginal table of counts</p>
</td></tr>
<tr><td><code>two</code></td>
<td>
<p>two-way margin table of transitions</p>
</td></tr>
<tr><td><code>three</code></td>
<td>
<p>three-way marginal table of transitions</p>
</td></tr>
<tr><td><code>four</code></td>
<td>
<p>four-way marginal table of transitions</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>A. C. Davison (Anthony.Davison@epfl.ch)</p>


<h3>References</h3>

<p>Avery, P. J. and Henderson, D. A. (1999) Fitting Markov chain models to
discrete state series such as DNA sequences. <em>Applied Statistics</em>, <b>48</b>, 53&ndash;61.
</p>
<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Section 6.1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(intron)

fit &lt;- MClik(intron)
</code></pre>

<hr>
<h2 id='mice'>Mice Deaths from Radiation</h2><span id='topic+mice'></span>

<h3>Description</h3>

<p>RFM male mice were exposed to 300 rads of x-radiation at 5&ndash;6
weeks of age.  The causes of death were thymic lymphoma, reticulum
cell sarcoma, and other.  Some of the mice were kept in a 
conventional environment, and the others in a germ-free environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mice)</code></pre>


<h3>Format</h3>

<p>A data frame with 177 observations on the following 4 variables.
</p>

<dl>
<dt>type</dt><dd><p>Environment type (factor)</p>
</dd>
<dt>cause</dt><dd><p>Cause of death</p>
</dd>
<dt>status</dt><dd><p>Censoring indicator, with 1 indicating death</p>
</dd>
<dt>y</dt><dd><p>Age at death (weeks)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Hoel, D. G. and Walburg, H. E. (1972) Statistical analysis of survival
experiments.
<em>Journal of the National Cancer Institute</em>, <b>49</b>, 361&ndash;372.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 200.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mice)
library(survival)
fit &lt;- survfit(Surv(y,status)~cause,data=mice[1:95,]) # first group
plot(fit,lty=c(3,2,1))
</code></pre>

<hr>
<h2 id='millet'>Millet Data</h2><span id='topic+millet'></span>

<h3>Description</h3>

<p>Data from an experiment conducted to determine the optimal planting distance
between plants in rows of millet.  The rows were 1 foot apart.  The design was a
5 x 5 Latin square.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(millet)</code></pre>


<h3>Format</h3>

<p>A data frame with 25 observations on the following 4 variables.
</p>

<dl>
<dt>row</dt><dd><p>Row label, coded <code>1</code>-<code>5</code>.</p>
</dd>
<dt>col</dt><dd><p>Column label, coded <code>1</code>-<code>5</code>.</p>
</dd>
<dt>dist</dt><dd><p>distances between plants:<code>2</code>, <code>4</code>, <code>6</code>, <code>8</code>, or <code>10</code>
inches.</p>
</dd>
<dt>y</dt><dd><p>Average yield (grams) of three central rows, 15 feet long after allowing for discards,
from each plot.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College, London.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(millet)
## maybe str(millet) ; plot(millet) ...
</code></pre>

<hr>
<h2 id='motorette'>Motorette Failure Data</h2><span id='topic+motorette'></span>

<h3>Description</h3>

<p>Times to failure of motorettes tested at different
temperatures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(motorette)</code></pre>


<h3>Format</h3>

<p>A data frame with 40 observations on the following 3 variables.
</p>

<dl>
<dt>x</dt><dd><p>Temperature in degrees Fahrenheit</p>
</dd>
<dt>cens</dt><dd><p>Censoring indicator</p>
</dd>
<dt>y</dt><dd><p>Failure time in hours</p>
</dd>
</dl>



<h3>Source</h3>

<p>Nelson, W. D. and Hahn, G. J. (1972)  Linear estimation of a 
regression relationship from censored data.  Part 1 &mdash; simple methods 
and their application (with Discussion).  <em>Technometrics</em>, <b>14</b>, 247&ndash;276.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 615.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(motorette)
library(survival)
motor.fit &lt;- survreg(Surv(y,cens)~log(x),dist="weibull",data=motorette)
summary(motor.fit)
</code></pre>

<hr>
<h2 id='nematode'>Nematode Data</h2><span id='topic+nematode'></span>

<h3>Description</h3>

<p>Numbers of nematodes invading individual fly larvae for
various numbers of initial challengers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nematode)</code></pre>


<h3>Format</h3>

<p>A data frame with 29 observations on the following 3 variables.
</p>

<dl>
<dt>m</dt><dd><p>Number of challengers</p>
</dd>
<dt>r</dt><dd><p>Number of invading nematodes</p>
</dd>
<dt>y</dt><dd><p>Number of fly larvae</p>
</dd>
</dl>



<h3>Source</h3>

<p>Faddy, M. J. and Fenlon, J. S. (1999) Stochastic modelling of the invasion
process of nematodes in fly larvae.
<em>Applied Statistics</em>, <b>48</b>, 31&ndash;37.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 295.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nematode)
</code></pre>

<hr>
<h2 id='nodal'>Nodal Involvement in Prostate Cancer</h2><span id='topic+nodal'></span>

<h3>Description</h3>

<p>The &lsquo;nodal&rsquo; data frame has 53 rows and 7 columns.
</p>
<p>The treatment strategy for a patient diagnosed with cancer of the
prostate depend highly on whether the cancer has spread to the
surrounding lymph nodes. It is common to operate on the patient to
get samples from the nodes which can then be analysed under a
microscope but clearly it would be preferable if an accurate
assessment of nodal involvement could be made without surgery.
</p>
<p>For a sample of 53 prostate cancer patients, a number of possible
predictor variables were measured before surgery.  The patients
then had surgery to  determine nodal involvement.  It was required
to see if nodal involvement could  be accurately predicted from
the predictor variables and which ones were most important.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nodal)</code></pre>


<h3>Format</h3>

<p>A data frame with 53 observations on the following 7 variables.
</p>

<dl>
<dt>m</dt><dd><p>A column of ones.</p>
</dd>
<dt>r</dt><dd><p>An indicator of nodal involvement. </p>
</dd>
<dt>aged</dt><dd><p>The patients age dichotomized into less than 60 (&lsquo;0&rsquo;) and
60 or over &lsquo;1&rsquo;.</p>
</dd>
<dt>stage</dt><dd><p>A measurement of the size and position of the tumour
observed by palpatation with the fingers via the rectum.  A
value of &lsquo;1&rsquo; indicates a more serious case of the cancer.</p>
</dd>
<dt>grade</dt><dd><p>Another indicator of the seriousness of the cancer, this
one is determined by a pathology reading of a biopsy taken by
needle before surgery. A value of &lsquo;1&rsquo; indicates a more
serious case of the cancer.</p>
</dd>
<dt>xray</dt><dd><p>A third measure of the seriousness of the cancer taken from
an X-ray reading. A value of &lsquo;1&rsquo; indicates a more serious
case of the cancer.</p>
</dd>
<dt>acid</dt><dd><p>The level of acid phosphatase in the blood serum.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Brown, B.W. (1980) Prediction analysis for binary data. In 
<em>Biostatistics Casebook</em>. R.G. Miller, B. Efron, B.W. Brown and L.E.
Moses (editors), 3-18. John Wiley.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 491.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nodal)
nodal.glm &lt;- glm(r~aged+stage+grade+xray+acid,binomial,data=nodal)
summary(nodal.glm,correlation=FALSE)

</code></pre>

<hr>
<h2 id='nuclear'>Nuclear Power Station Construction Data</h2><span id='topic+nuclear'></span>

<h3>Description</h3>

<p>The data relate to the construction of 32 light water reactor
(LWR) plants  constructed in the U.S.A in the late 1960's and
early 1970's.  The data was collected with the aim of predicting
the cost of construction of further LWR plants.  6 of the power
plants had partial turnkey guarantees and it is possible that, for
these plants, some manufacturers' subsidies may be hidden  in the
quoted capital costs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nuclear)</code></pre>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 11 variables.
</p>

<dl>
<dt>cost</dt><dd><p>The capital cost of construction in millions of dollars
adjusted to 1976 base.</p>
</dd>
<dt>date</dt><dd><p>The date on which the construction permit was issued.  The
data are measured in years since January 1 1990 to the
nearest month.</p>
</dd>
<dt>t1</dt><dd><p>The time between application for and issue of the
construction permit.</p>
</dd>
<dt>t2</dt><dd><p>The time between issue of operating license and construction
permit.</p>
</dd>
<dt>cap</dt><dd><p>The net capacity of the power plant (MWe).</p>
</dd>
<dt>pr</dt><dd><p>A binary variable where &lsquo;1&rsquo; indicates the prior existence of
a LWR plant at the same site.</p>
</dd>
<dt>ne</dt><dd><p>A binary variable where &lsquo;1&rsquo; indicates that the plant was
constructed in the  north-east region of the U.S.A.</p>
</dd>
<dt>ct</dt><dd><p>A binary variable where &lsquo;1&rsquo; indicates the use of a cooling
tower in the plant.</p>
</dd>
<dt>bw</dt><dd><p>A binary variable where &lsquo;1&rsquo; indicates that the nuclear steam
supply system was manufactured by Babcock-Wilcox.</p>
</dd>
<dt>cum.n</dt><dd><p>The cumulative number of power plants constructed by each
architect-engineer.</p>
</dd>
<dt>pt</dt><dd><p>A binary variable where &lsquo;1&rsquo; indicates those plants with
partial turnkey guarantees.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Cox, D.R. and Snell, E.J. (1981)  <em>Applied Statistics: Principles
and Examples</em>. Chapman and Hall.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 401.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nuclear)
pairs(nuclear)
fit &lt;- lm(log(cost)~date+t1+t2+cap+pr+ne+ct+bw+cum.n+pr,data=nuclear)
step(fit)  # stepwise model selection
</code></pre>

<hr>
<h2 id='old.age'>Estimates of Hazard Function for Old Age</h2><span id='topic+old.age'></span>

<h3>Description</h3>

<p>Historical estimates of the force of mortality (hazard function)
averaged for 5-year age groups.  The data are taken from various
historical sources.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(old.age)</code></pre>


<h3>Format</h3>

<p>A data frame with 14 observations on the following 8 variables.
</p>

<dl>
<dt>age</dt><dd><p>Age group (5-year intervals)</p>
</dd>
<dt>hungary</dt><dd><p>Data estimated from Hungarian graveyards, 900&ndash;1100</p>
</dd>
<dt>eng.1640</dt><dd><p>Data estimated from England, 1640&ndash;1689</p>
</dd>
<dt>breslau</dt><dd><p>Data estimated from Breslau, 1687&ndash;1691</p>
</dd>
<dt>engm.1841</dt><dd><p>Data from England and Wales, males, 1841</p>
</dd>
<dt>engf.1841</dt><dd><p>Data from England and Wales, females, 1841</p>
</dd>
<dt>engm.1980</dt><dd><p>Data from England and Wales, males, 1980&ndash;1982</p>
</dd>
<dt>engf.1980</dt><dd><p>Data from England and Wales, females, 1980&ndash;1982</p>
</dd>
</dl>



<h3>Details</h3>

<p>The estimated numbers of people on which the data in the columns are
based are 2300, 3133, 2675, 71,000, 74,000, 834,0000, and 828,000.
</p>


<h3>Source</h3>

<p>Thatcher, A. R. (1999) The long-term pattern of adult mortality and the
highest attained age (with Discussion).
<em>Journal of the Royal Statistical Society series A</em>, <b>16</b>, 5&ndash;43.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 194.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(old.age)
</code></pre>

<hr>
<h2 id='pairs.mod'>Modified Scatterplot Matrix</h2><span id='topic+pairs.mod'></span>

<h3>Description</h3>

<p>Plots a scatterplot matrix in which panels below the diagonal
show ordinary scatterplots of pairs of variables, and those above
the diagonal show scatterplots of residuals after regression on all
the other variables.   
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mod'
pairs(x, format = "MC", labelnames = names(x), highlight = NULL, level = 0.9, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pairs.mod_+3A_x">x</code></td>
<td>
<p>A matrix whose rows correspond to units and whose columns
correspond to variables measured on those units.</p>
</td></tr>
<tr><td><code id="pairs.mod_+3A_format">format</code></td>
<td>
<p>'MM' for marginal (that is, standard) scatterplots above and below the diagonal, 'MC' for marginal below and conditional (= partial) above, etc.  'MC' by default.</p>
</td></tr>
<tr><td><code id="pairs.mod_+3A_labelnames">labelnames</code></td>
<td>
<p>Names of the variables.</p>
</td></tr>
<tr><td><code id="pairs.mod_+3A_highlight">highlight</code></td>
<td>
<p>Indexes of observations (rows) to be highlighted.</p>
</td></tr>
<tr><td><code id="pairs.mod_+3A_level">level</code></td>
<td>
<p>Scalar giving the level for the contour, 0.9 by default.</p>
</td></tr>
<tr><td><code id="pairs.mod_+3A_...">...</code></td>
<td>
<p>The plotting symbol and other arguments for the points can be controlled by &lsquo;pch=&rsquo;, etc.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The diagonal shows histograms of the original data, and (in black) histograms of the partial
residuals after adjustment on all the other variables, shifted to have
the same mean as the original data.   Also given are the original
</p>
<p>The below-diagonal panels contain the numerical value of the correlation, and those above the diagonal contain the partial
correlation, that is, the correlation of the residuals after linear regression on the remaining variables. The panels show ellipses which
would contain 90 percent of the observations in a large normal sample
with the same mean and covariance matrix as the data.
</p>


<h3>Value</h3>

<p>Produces the scatterplot matrix, and prints the marginal and partial
standard deviations of the variables.
</p>


<h3>Note</h3>

<p>pairs.mod calls library(ellipse) and will give an error if this is
unavailable.
</p>


<h3>Author(s)</h3>

<p>Sylvain Sardy (Sylvain.Sardy@epfl.ch)
</p>


<h3>References</h3>

<p>Davison, A. C. and Sardy, S. (2000)  The partial scatterplot matrix.
<em>Journal of Computational and Graphical Statistics</em>, <b>9</b>, 750&ndash;758.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ellipse)
data(mathmarks)
pairs.mod(mathmarks)
</code></pre>

<hr>
<h2 id='paulsen'>Neurotransmission in Guinea Pig Brains</h2><span id='topic+paulsen'></span>

<h3>Description</h3>

<p>The 'paulsen' data frame has 346 rows and 1 columns.  
</p>
<p>Sections were prepared from the brain of adult guinea pigs.
Spontaneous currents that flowed into individual brain cells were
then recorded and the peak amplitude of each current measured. 
The aim of the experiment  was to see if the current flow was
quantal in nature (i.e. that it is not a single burst but instead
is built up of many smaller bursts of current).  If the current
was indeed quantal then it would be expected that the distribution
of the current amplitude would be multimodal with modes at regular
intervals.  The modes would be expected to decrease in magnitude
for higher current amplitudes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(paulsen)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following column:
</p>

<dl>
<dt>y</dt><dd><p> The current flowing into individual brain cells. The currents
are measured in pico-amperes.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were kindly made available by Dr. O. Paulsen of the
Department of Pharmacology, University of Oxford. 
</p>
<p>Paulsen, O. and Heggelund, P. (1994) The quantal size at
retinogeniculate  synapses determined from spontaneous and evoked
EPSCs in guinea-pig thalamic  slices. <em>Journal of Physiology</em>,
<b>480</b>, 505-511.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(paulsen)
hist(paulsen$y,prob=TRUE)
</code></pre>

<hr>
<h2 id='pbc'>Mayo Clinic Primary Biliary Cirrhosis Data</h2><span id='topic+pbc'></span>

<h3>Description</h3>

<p>Followup of 312 randomised and 108 unrandomised patients with
primary biliary cirrhosis, a rare autoimmune liver disease, at
Mayo Clinic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pbc)</code></pre>


<h3>Format</h3>

<p>A data frame with 418 observations on the following 20 variables.
</p>

<dl>
<dt>age</dt><dd><p>in years</p>
</dd>
<dt>alb</dt><dd><p>serum albumin</p>
</dd>
<dt>alkphos</dt><dd><p>alkaline phosphotase</p>
</dd>
<dt>ascites</dt><dd><p>presence of ascites</p>
</dd>
<dt>bili</dt><dd><p>serum bilirubin</p>
</dd>
<dt>chol</dt><dd><p>serum cholesterol</p>
</dd>
<dt>edema</dt><dd><p>presence of edema</p>
</dd>
<dt>edtrt</dt><dd><p>0 no edema, 0.5 untreated or successfully treated
1 unsuccessfully treated edema</p>
</dd>
<dt>hepmeg</dt><dd><p>enlarged liver</p>
</dd>
<dt>time</dt><dd><p>survival time</p>
</dd>
<dt>platelet</dt><dd><p>platelet count</p>
</dd>
<dt>protime</dt><dd><p>standardised blood clotting time</p>
</dd>
<dt>sex</dt><dd><p>1=male</p>
</dd>
<dt>sgot</dt><dd><p>liver enzyme (now called AST)</p>
</dd>
<dt>spiders</dt><dd><p>blood vessel malformations in the skin</p>
</dd>
<dt>stage</dt><dd><p>histologic stage of disease (needs biopsy)</p>
</dd>
<dt>status</dt><dd><p>censoring status</p>
</dd>
<dt>trt</dt><dd><p>1/2/-9 for control, treatment, not randomised</p>
</dd>
<dt>trig</dt><dd><p>triglycerides</p>
</dd>
<dt>copper</dt><dd><p>urine copper</p>
</dd>
</dl>



<h3>Source</h3>

<p>Fleming, T. R. and Harrington, D. P. (1991)   
<em>Counting Processes and Survival Analysis</em>.  Wiley: New York.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 549.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pbc)  
# to make version of dataset used in book
pbcm &lt;- pbc[(pbc$trt!=-9),]
pbcm$copper[(pbcm$copper==-9)] &lt;- median(pbcm$copper[(pbcm$copper!=-9)])
pbcm$platelet[(pbcm$platelet==-9)] &lt;- median(pbcm$platelet[(pbcm$platelet!=-9)])
attach(pbcm)

library(survival)
par(mfrow=c(1,2),pty="s")
plot(survfit(Surv(time,status)~trt),ylim=c(0,1),lty=c(1,2),
   ylab="Survival probability",xlab="Time (days)")
plot(survfit(coxph(Surv(time,status)~trt+strata(sex))),ylim=c(0,1),lty=c(1,2),
   ylab="Survival probability",xlab="Time (days)")
lines(survfit(coxph(Surv(time,status)~trt)),lwd=2)
# proportional hazards model fit
fit &lt;- coxph(formula = Surv(time, status) ~ age + alb + alkphos + ascites + 
      bili + edtrt + hepmeg + platelet + protime + sex + spiders, data=pbcm)
summary(fit)
step.fit &lt;- step(fit,direction="backward")
</code></pre>

<hr>
<h2 id='pigeon'>Homing Pigeon Data</h2><span id='topic+pigeon'></span>

<h3>Description</h3>

<p>Bearings (degrees) of 29 homing pigeons 30, 60, 90 after their
release, and on vanishing from sight.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pigeon)</code></pre>


<h3>Format</h3>

<p>A data frame with 29 observations on the following 4 variables.
</p>

<dl>
<dt>s30</dt><dd><p>Bearing after 30 seconds</p>
</dd>
<dt>s60</dt><dd><p>Bearing after 60 seconds</p>
</dd>
<dt>s90</dt><dd><p>Bearing after 90 seconds</p>
</dd>
<dt>van</dt><dd><p>Bearing on vanishing from sight</p>
</dd>
</dl>



<h3>Source</h3>

<p>Artes, R. (1997) Extensoes da Teoria das Equacoes de
Estimacao Generalizadas a Dados Circulares e Modelos de
Dispersao. Ph.D. thesis, University of Sao Paulo.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press. Page 173.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pigeon)
plt &lt;- function( ang, r=c(1,2,3,4), lty=1,... )
{
  si &lt;- sin(2*pi*ang/360)
  co &lt;- cos(2*pi*ang/360)
  points( r*si,r*co )
  lines( c(0,r*si),c(0,r*co),...)
}
par(pty="s")
plot(c(0,0),c(0,0),xlim=c(-4,4),ylim=c(-4,4),
    xlab="Easting",ylab="Northing")
for (i in 1:nrow(pigeon)) plt( pigeon[i,],col=i )

</code></pre>

<hr>
<h2 id='pigs'>Pig Diet Data</h2><span id='topic+pigs'></span>

<h3>Description</h3>

<p>Data on weight gains in 32 pigs, divided into eight groups
of four, and with 4 different diets allocated to the group
members.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pigs)</code></pre>


<h3>Format</h3>

<p>A data frame with 32 observations on the following 3 variables.
</p>

<dl>
<dt>group</dt><dd><p>a factor with 8 levels</p>
</dd>
<dt>diet</dt><dd><p>a factor with levels <code>I</code>&ndash;<code>IV</code></p>
</dd>
<dt>gain</dt><dd><p>weight gain (units unknown)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College, 
London
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 431.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pigs)
anova(lm(gain~group+diet,data=pigs))
</code></pre>

<hr>
<h2 id='pipette'>Red Blood Cell Data</h2><span id='topic+pipette'></span>

<h3>Description</h3>

<p>Numbers of red blood cells counted by five doctors using ten sets of apparatus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pipette)</code></pre>


<h3>Format</h3>

<p>A data frame with 50 observations on the following 3 variables.
</p>

<dl>
<dt>apparatus</dt><dd><p>Factor with ten levels</p>
</dd>
<dt>doctor</dt><dd><p>Factor with five levels</p>
</dd>
<dt>y</dt><dd><p>Number of red blood cells</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College, London.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 462.
</p>

<hr>
<h2 id='plot.glm.diag'>Diagnostic plots for generalized linear models</h2><span id='topic+plot.glm.diag'></span>

<h3>Description</h3>

<p>Makes plot of jackknife deviance residuals against linear predictor, normal scores plots of standardized deviance residuals, plot of approximate Cook statistics against leverage/(1-leverage), and case plot of Cook statistic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'glm.diag'
plot(x, glmdiag = glm.diag(x), subset = NULL, iden = FALSE, 
                        labels = NULL, ret = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.glm.diag_+3A_x">x</code></td>
<td>
<p><code>glm.object</code> or <code>lm.object</code> : the result of a call to <code>glm()</code> or <code>lm()</code></p>
</td></tr>
<tr><td><code id="plot.glm.diag_+3A_glmdiag">glmdiag</code></td>
<td>
<p>Diagnostics of <code>x</code> obtained from a call to <code>glm.diag</code>. If it is not supplied then it is calculated.</p>
</td></tr>
<tr><td><code id="plot.glm.diag_+3A_subset">subset</code></td>
<td>
<p> Subset of <code>data</code> for which glm fitting performed: should be the same as the  <code>subset</code> option used in the call to <code>glm()</code> or <code>lm()</code> which generated <code>x</code>. Needed only if the <code>subset=</code> option was used in the call to <code>glm</code> or <code>lm</code>. </p>
</td></tr>
<tr><td><code id="plot.glm.diag_+3A_iden">iden</code></td>
<td>
<p> A logical argument. If <code>TRUE</code> then, after the plots are drawn, the user will be prompted for an integer between 0 and 4. A positive integer will select a plot and invoke <code>identify()</code> on that plot. After exiting <code>identify()</code>, the user is again prompted, this loop continuing until the user responds to the prompt with 0. If <code>iden</code> is <code>FALSE</code> (default) the user cannot interact with the plots.</p>
</td></tr>
<tr><td><code id="plot.glm.diag_+3A_labels">labels</code></td>
<td>
<p>A vector of labels for use with <code>identify()</code> if <code>iden</code> is <code>TRUE</code>. If it is not supplied then the labels are derived from <code>x</code>. </p>
</td></tr>
<tr><td><code id="plot.glm.diag_+3A_ret">ret</code></td>
<td>
<p>A logical argument indicating if <code>glmdiag</code> should be returned. The default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot.glm.diag_+3A_...">...</code></td>
<td>
<p>Other arguments, which are ignored.  This is included only for compatibility with S3 methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot on the top left is a plot of the jackknife deviance residuals against the fitted values.
</p>
<p>The plot on the top right is a normal QQ plot of the standardized deviance residuals. The dotted line is the expected line if the standardized residuals are normally distributed, i.e. it is the line with intercept 0 and slope 1.
</p>
<p>The bottom two panels are plots of the Cook statistics. On the left is a plot of the Cook statistics against the standardized leverages. In general there will be two dotted lines on this plot. The horizontal line is at 8/(n-2p) where n is the number of observations and p is the number of parameters estimated. Points above this line may be points with high influence on the model. The vertical line is at 2p/(n-2p) and points to the right of this line have high leverage compared to the variance of the raw residual at that point. If all points are below the horizontal line or to the left of the vertical line then the line is not shown.
</p>
<p>The final plot again shows the Cook statistic this time plotted against case number enabling us to find which observations are influential.
</p>
<p>Use of <code>iden=T</code> is encouraged for proper exploration of these four plots as a guide to how well the model fits the data and whether certain observations have an unduly large effect on parameter estimates.</p>


<h3>Value</h3>

<p>If ret is <code>TRUE</code> then the value of <code>glmdiag</code> is returned otherwise there is no returned value.</p>


<h3>Author(s)</h3>

<p> Angelo Canty </p>


<h3>References</h3>

<p> Davison, A.C. and Snell, E.J. (1991) Residuals and diagnostics. In  <em>Statistical Theory and Modelling: In Honour of Sir David Cox</em>.  D.V. Hinkley, N. Reid, and E.J. Snell (editors), 83-106. Chapman and Hall. </p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">glm</a></code>,<code><a href="#topic+glm.diag">glm.diag</a></code>,<code><a href="graphics.html#topic+identify">identify</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>  # leukaemia data
require(MASS)
data(leuk, package="MASS")
leuk.mod &lt;- glm(time~ag-1+log10(wbc),family=Gamma(log),data=leuk)
leuk.diag &lt;- glm.diag(leuk.mod)
plot.glm.diag(leuk.mod,leuk.diag)
</code></pre>

<hr>
<h2 id='pneu'>Pneumoconiosis amongst Coalminers</h2><span id='topic+pneu'></span>

<h3>Description</h3>

<p>This gives the degree of pneumoconiosis (normal, present, or severe)
in a group of coalminers as a function of the number of years worked
at the coalface.  The degree of the disease was assessed radiologically and is qualitative.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pneu)</code></pre>


<h3>Format</h3>

<p>A data frame with 8 observations on the following 4 variables.
</p>

<dl>
<dt>Years</dt><dd><p>Period of exposure (years worked at the coalface)</p>
</dd>
<dt>Normal</dt><dd><p>Number of miners with normal lungs</p>
</dd>
<dt>Present</dt><dd><p>Number of miners with disease present</p>
</dd>
<dt>Severe</dt><dd><p>Number of miners with severe disease</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ashford, J. R. (1959)  An approach to the analysis of data for semi-quantal 
responses in biological assay.  <em>Biometrics</em>, <b>15</b>, 573&ndash;581.</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 509.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pneu)
summary(glm(cbind(Present+Severe,Normal)~log(Years),data=pneu,binomial))
summary(glm(cbind(Severe,Normal+Present)~log(Years),data=pneu,binomial))
</code></pre>

<hr>
<h2 id='poi.beta.laplace'>Laplace Approximation for Posterior Density, Practical 11.2</h2><span id='topic+poi.beta.laplace'></span>

<h3>Description</h3>

<p>This function computes the Laplace approximation to the posterior density
of the parameter beta in a Poisson regression model.  For more details see
Practical 11.2 of Davison (2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poi.beta.laplace(data, alpha = get.alpha(data), phi = 1, nu = 0.1, 
                 beta = seq(from = 0, to = 7, length = 1000))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="poi.beta.laplace_+3A_data">data</code></td>
<td>
<p>A data frame with vector components <code>y</code> and <code>x</code> of the same length.  
<code>y</code> contains the numbers of counts, and <code>x</code> the corresponding time intervals.</p>
</td></tr>
<tr><td><code id="poi.beta.laplace_+3A_alpha">alpha</code></td>
<td>
<p>Prior value of a parameter, estimated from the data by default.</p>
</td></tr>
<tr><td><code id="poi.beta.laplace_+3A_phi">phi</code></td>
<td>
<p>Prior value of a parameter.</p>
</td></tr>
<tr><td><code id="poi.beta.laplace_+3A_nu">nu</code></td>
<td>
<p> Prior value of a parameter.</p>
</td></tr>
<tr><td><code id="poi.beta.laplace_+3A_beta">beta</code></td>
<td>
<p>Values for which posterior density of beta should be provided. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is provided simply so that  readers spend less time typing. It is not intended to be
robust and general code.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>int</code></td>
<td>
<p>Estimated integral of posterior density.</p>
</td></tr>
<tr><td><code>conv</code></td>
<td>
<p>Did the routine for the Laplace optimization converge?</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>Values of beta</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>Values of posterior density</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anthony Davison (<code>anthony.davison@epfl.ch</code>)</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press. Practical 11.2.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## From Practical 11.2:
get.alpha &lt;- function(d)
{  # estimate alpha from data
  rho &lt;- d$y/d$x
  n &lt;- length(d$y)
  mean(rho)^2/( (n-1)*var(rho)/n - mean(rho)*mean(1/d$x) )
}
data(cloth)
attach(cloth)
plot(x,y)
beta &lt;- seq(from=0,to=10,length=1000)
beta.post &lt;- poi.beta.laplace(cloth,beta=beta,nu=1)
plot(beta.post,type="l",xlab="beta",ylab="Posterior density")
beta.post &lt;- poi.beta.laplace(cloth,beta=beta,nu=5)
lines(beta.post,lty=2)
</code></pre>

<hr>
<h2 id='poi.gibbs'>Gibbs Sampler for Hierarchical Poisson Model, Practical 11.5</h2><span id='topic+poi.gibbs'></span>

<h3>Description</h3>

<p>This function implements Gibbs sampling for the hierarchical Poisson model
described in Example 11.19 and Practical 11.5 of Davison (2003), which 
should be consulted for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poi.gibbs(d, alpha, gamma, delta, I, S)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="poi.gibbs_+3A_d">d</code></td>
<td>
<p>A data frame with vector components <code>y</code> containing the numbers of
counts and <code>x</code> the period for which the <code>n</code> Poisson processes are observed.</p>
</td></tr>
<tr><td><code id="poi.gibbs_+3A_alpha">alpha</code></td>
<td>
<p>A hyperparameter of the prior density</p>
</td></tr>
<tr><td><code id="poi.gibbs_+3A_gamma">gamma</code></td>
<td>
<p>A hyperparameter of the prior density</p>
</td></tr>
<tr><td><code id="poi.gibbs_+3A_delta">delta</code></td>
<td>
<p> A hyperparameter of the prior density</p>
</td></tr>
<tr><td><code id="poi.gibbs_+3A_i">I</code></td>
<td>
<p>Number of iterations for which sampler is run</p>
</td></tr>
<tr><td><code id="poi.gibbs_+3A_s">S</code></td>
<td>
<p>Number of independent replicates of sampler</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is provided simply so that  readers spend less time typing. It is not intended to be
robust and general code.
</p>


<h3>Value</h3>

<p>An I x S x (n+1) array containing the successive iterations of the samplers, 
for the I iterations, S independent replicates, and <code>n</code> rate parameters plus the parameter
<code>beta</code> of the prior distribution.
</p>


<h3>Author(s)</h3>

<p>Anthony Davison (<code>anthony.davison@epfl.ch</code>)</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press. Practical 11.5.</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## From Practical 11.5:
data(pumps)
system.time( pumps.sim &lt;- poi.gibbs(pumps, alpha=1.8, delta=0.1, gamma=1, 
             I=1000, S=5) )
par(mfrow=c(2,3))
plot.ts(pumps.sim[,1,1])
acf(pumps.sim[,1,1])
pacf(pumps.sim[,1,1])
plot.ts(pumps.sim[,1,11])
acf(pumps.sim[,1,11])
pacf(pumps.sim[,1,11])
</code></pre>

<hr>
<h2 id='poisons'>Survival Times for Poisoned Animals</h2><span id='topic+poisons'></span>

<h3>Description</h3>

<p>In an experiment to assess the usefulness of
treatments for poisons, 
48 animals were split randomly into 12 groups of 4.
Each group was administered one of three poisons,
and one of four treatments, giving a 
3x4 factorial design with 4 replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(poisons)</code></pre>


<h3>Format</h3>

<p>A data frame with 48 observations on the following 3 variables.
</p>

<dl>
<dt>time</dt><dd><p>Survival time (units of 10 hours)</p>
</dd>
<dt>poison</dt><dd><p>Factor giving poison</p>
</dd>
<dt>treat</dt><dd><p>Factor giving treatment</p>
</dd>
</dl>



<h3>Source</h3>

<p>Box, G. E. P. and Cox, D. R. (1964) An analysis of transformations
(with  Discussion).
<em>Journal of the Royal Statistical Society series B</em>, <b>26</b>, 211&ndash;246.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 391.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(poisons)
fit &lt;- lm(time~poison+treat,data=poisons)
library(MASS)
boxcox(time~poison+treat,data=poisons)
</code></pre>

<hr>
<h2 id='pollution'>Air Pollution and Mortality</h2><span id='topic+pollution'></span>

<h3>Description</h3>

<p>Data on the relation between weather, socioeconomic, and air pollution variables
and mortality rates in 60 Standard Metropolitan Statistical Areas (SMSAs) of the
USA, for the years 1959-1961.  Some of the variables are highly collinear.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pollution)</code></pre>


<h3>Format</h3>

<p>A data frame with 60 observations on the following variables.
</p>

<dl>
<dt>prec</dt><dd><p>Average annual precipitation in inches</p>
</dd>
<dt>jant</dt><dd><p>Average January temperature in degrees F</p>
</dd>
<dt>jult</dt><dd><p>Average July temperature in degrees F</p>
</dd>
<dt>ovr95</dt><dd><p>Percentage of 1960 SMSA population aged 65 or older</p>
</dd>
<dt>popn</dt><dd><p>Average household size</p>
</dd>
<dt>educ</dt><dd><p>Median school years completed by those over 22</p>
</dd>
<dt>hous</dt><dd><p>percentage of housing units which are sound and with all facilities</p>
</dd>
<dt>dens</dt><dd><p>Population per square mile in urbanized areas, 1960</p>
</dd>
<dt>nonw</dt><dd><p>Percentage non-white population in urbanized areas, 1960</p>
</dd>
<dt>wwdrk</dt><dd><p>Percentage  employed in white collar occupations</p>
</dd>
<dt>poor</dt><dd><p>Percentage  of families with income &lt; 3000 dollars</p>
</dd>
<dt>hc</dt><dd><p>Relative hydrocarbon pollution potential</p>
</dd>
<dt>nox</dt><dd><p>Same for nitric oxides</p>
</dd>
<dt>so</dt><dd><p>Same for sulphur dioxide</p>
</dd>
<dt>humid</dt><dd><p>Annual average percentage relative humidity at 1pm</p>
</dd>
<dt>mort</dt><dd><p>Total age-adjusted mortality rate per 100,000</p>
</dd>
</dl>



<h3>Source</h3>

<p>McDonald, G. C. and Schwing, R. C. (1973) Instabilities of regression
estimates relating air pollution to mortality, <em>Technometrics</em>, <b>15</b>, 463-482.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pollution)
## maybe str(pollution) ; plot(pollution) ...
</code></pre>

<hr>
<h2 id='pumps'>Pump Failure Data</h2><span id='topic+pumps'></span>

<h3>Description</h3>

<p>The data give numbers of failures of 
ten pumps from several systems
in the nuclear plant Farley 1.  Pumps 1, 3, 4, and 6 operate continuously, 
while the rest operate only intermittantly or on standby.  </p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pumps)</code></pre>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 2 variables.
</p>

<dl>
<dt>x</dt><dd><p>Operating time (in thousands of operatin hours)</p>
</dd>
<dt>y</dt><dd><p>Number of failures</p>
</dd>
</dl>



<h3>Source</h3>

<p>Gaver, D. P. and O'Muircheartaigh, I. G. (1987) 
Robust empirical Bayes analysis of event rates.  <em>Technometrics</em>, 
<b>29</b>, 1&ndash;15.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 600.</p>

<hr>
<h2 id='qqexp'>Exponential Quantile-Quantile Plots</h2><span id='topic+qqexp'></span>

<h3>Description</h3>

<p>Exponential probability plot of data.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqexp(y, line = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qqexp_+3A_y">y</code></td>
<td>
<p>Vector for which plot is required</p>
</td></tr>
<tr><td><code id="qqexp_+3A_line">line</code></td>
<td>
<p>Add line to plot (no line by default)</p>
</td></tr>
<tr><td><code id="qqexp_+3A_...">...</code></td>
<td>
<p>Other options for plot command</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A exponential probablity plot of the data in y; that is, a plot of
the ordered values of y against the quantiles of the standard exponential
distribution.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+qqnorm">qqnorm</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>qqexp(rexp(50))
qqexp(rgamma(50,shape=2),line=TRUE)
</code></pre>

<hr>
<h2 id='quake'>Japanese Earthquake Data</h2><span id='topic+quake'></span>

<h3>Description</h3>

<p>Times and magnitudes (Richter scale) of 483 shallow earthquakes in an
offshore region east of Honshu and south of Hokkaido, for the period
1885&ndash;1980.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(quake)</code></pre>


<h3>Format</h3>

<p>An irregular time series with earthquake
</p>

<dl>
<dt>time</dt><dd><p>in days since start of 1885</p>
</dd>
<dt>mag</dt><dd><p>magnitude (Richter scale)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ogata, Y. (1988) Statistical models for earthquake occurrences and 
residual analysis for point processes.
<em>Journal of the American Statistical Association</em>, <b>83</b>, 9&ndash;27.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 289.
</p>

<hr>
<h2 id='rat.growth'>Rat Growth Data</h2><span id='topic+rat.growth'></span>

<h3>Description</h3>

<p>Data on the weights of 30 rats each week for 5 weeks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(rat.growth)</code></pre>


<h3>Format</h3>

<p>A data frame with 150 observations on the following 3 variables.
</p>

<dl>
<dt>rat</dt><dd><p>a factor with levels 1-30</p>
</dd>
<dt>week</dt><dd><p>takes values 0-4</p>
</dd>
<dt>y</dt><dd><p>rat weight (units unspecified)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Gelfand, A. E., Hills, S. E., Racine-Poon, A. and Smith, A. F. M. (1990)
Illustration of Bayesian inference in normal data models using Gibbs sampling.
<em>Journal of the American Statistical Association</em>, <b>85</b>, 972&ndash;985.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 460.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rat.growth)
library(nlme)
rat.fit &lt;- groupedData( y~poly(week,2) | rat,
                   data = rat.growth,
                   labels = list( x = "Week",
                     y = "Weight" ),
                   units = list( x = "", y = "(?)") )

summary(lme(rat.fit))
</code></pre>

<hr>
<h2 id='salinity'>Water Salinity and River Discharge</h2><span id='topic+salinity'></span>

<h3>Description</h3>

<p>The 'salinity' data frame has 28 rows and 4 columns.
</p>
<p>Biweekly averages of the water salinity and river discharge in
Pamlico Sound, North Carolina were recorded between the years 1972
and 1977. The data in this set consists only of those measurements
in March, April and May.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(salinity)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>sal</dt><dd><p>The average salinity of the water over two weeks.</p>
</dd>
<dt>lag</dt><dd><p>The average salinity of the water lagged two weeks.  Since
only spring is used, the value of 'lag' is not always equal
to the previous value of 'sal'.</p>
</dd>
<dt>trend</dt><dd><p>A factor indicating in which of the 6 biweekly periods
between March and May, the observations were taken. The
levels of the factor are from 0 to 5 with 0 being the first
two weeks in March.</p>
</dd>
<dt>dis</dt><dd><p>The amount of river discharge during the two weeks for which
'sal' is the average salinity.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from
</p>
<p>Ruppert, D. and Carroll, R.J. (1980) Trimmed least squares
estimation in the  linear model.  <em>Journal of the American
Statistical Association</em>, <b>75</b>, 828&ndash;838.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(salinity)
## maybe str(salinity) ; plot(salinity) ...
</code></pre>

<hr>
<h2 id='seeds'>Germination of seeds</h2><span id='topic+seeds'></span>

<h3>Description</h3>

<p>These are the number of seeds germinating when subjected to extracts of certain roots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(seeds)</code></pre>


<h3>Format</h3>

<p>A data frame with 21 observations on the following 4 variables.
</p>

<dl>
<dt>r</dt><dd><p>Number of seeds germinating</p>
</dd>
<dt>m</dt><dd><p>Total number of seeds</p>
</dd>
<dt>seed</dt><dd><p>Seed type: O. aegyptiaco 75 or O. aegyptiaco 73</p>
</dd>
<dt>root</dt><dd><p>Root extract</p>
</dd>
</dl>



<h3>Source</h3>

<p>Crowder, M. J.  (1978)  Beta-binomial ANOVA for proportions.  <em>Applied Statistics</em>,  
<b>27</b>, 34&ndash;37.
</p>


<h3>References</h3>

<p>Cox, D. R. and Snell, E. J. (1989)  <em>Analysis of Binary Data</em>, second edition.  
London: Chapman and Hall.  Section 3.2.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(seeds)
## maybe str(seeds) ; plot(seeds) ...
</code></pre>

<hr>
<h2 id='shoe'>Shoe Wear Data</h2><span id='topic+shoe'></span>

<h3>Description</h3>

<p>Amount of wear in a paired comparison of two materials used
for soling the shoes of 10 boys.  The materials were allocated randomly
to the left and right feet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(shoe)</code></pre>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 4 variables.
</p>

<dl>
<dt>material</dt><dd><p>factor giving the shoe sole material</p>
</dd>
<dt>boy</dt><dd><p>factor with 10 levels</p>
</dd>
<dt>foot</dt><dd><p>factor giving left or right foot</p>
</dd>
<dt>y</dt><dd><p>amount of shoe wear</p>
</dd>
</dl>



<h3>Source</h3>

<p>Box, G. E. P., Hunter, W. G. and Hunter, J. S. (1978) 
<em>Statistics for  Experimenters</em>.  New York: Wiley.  Page 100.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 422.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(shoe)
attach(shoe)
d &lt;- y[material=="B"]-y[material=="A"]  # difference
t.test(d)  # t test of hypothesis that B wears quicker 
</code></pre>

<hr>
<h2 id='shuttle'>O-ring Thermal Distress Data for Space Shuttle</h2><span id='topic+shuttle'></span>

<h3>Description</h3>

<p>Data on the number of rubber O-rings showing thermal distress for
23 flights of the space shuttle, with the ambient temperature and
pressure at which tests on the putty next to the rings were performed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(shuttle)</code></pre>


<h3>Format</h3>

<p>A data frame with 23 observations on the following 4 variables.
</p>

<dl>
<dt>m</dt><dd><p>Number of rings</p>
</dd>
<dt>r</dt><dd><p>Number of rings showing thermal distress</p>
</dd>
<dt>temperature</dt><dd><p>ambient temperature (degrees Fahrenheit)</p>
</dd>
<dt>pressure</dt><dd><p>pressure (pounds per square inch)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Dalal, S. R., Fowlkes, E. B. and Hoadley, B. (1989)  Risk analysis of the space shuttle:
Pre-Challenger prediction of failure.  <em>Journal of the American Statistical
Association</em>, <b>84</b>, 945&ndash;957.</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.  Page 7.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(shuttle)
attach(shuttle)
plot(temperature, r/m,ylab="Proportion of failures")
</code></pre>

<hr>
<h2 id='smoking'>Survival and Smoking</h2><span id='topic+smoking'></span>

<h3>Description</h3>

<p>Twenty-year survival and smoking status for 1314 women from
Whickham, near Newcastle-upon-Tyne.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(smoking)</code></pre>


<h3>Format</h3>

<p>A data frame with 14 observations on the following 4 variables.
</p>

<dl>
<dt>age</dt><dd><p>Age group (factor)</p>
</dd>
<dt>smoker</dt><dd><p>Smoking status (1=smoker, 0=non-smoker)</p>
</dd>
<dt>alive</dt><dd><p>Number alive after 20 years</p>
</dd>
<dt>dead</dt><dd><p>Number dead after 20 years</p>
</dd>
</dl>



<h3>Source</h3>

<p>Appleton, D. R., French, J. M. and Vanderpump, M. P. J. (1996) 
Ignoring a covariate: An example of Simpson's paradox.
<em>The American Statistician</em>, <b>50</b>, 340&ndash;341.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 258.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(smoking)
summary(glm(cbind(dead,alive)~smoker,data=smoking,binomial))
# note sign change for smoker covariate, due to Simpson's paradox
summary(glm(cbind(dead,alive)~age+smoker,data=smoking,binomial))
</code></pre>

<hr>
<h2 id='soccer'>Soccer Scores from English Premier League, 2000-2001 Season</h2><span id='topic+soccer'></span>

<h3>Description</h3>

<p>These are scores for the 380 fixtures in the English Premier League,
2000&ndash;2001.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(soccer)</code></pre>


<h3>Format</h3>

<p>A data frame with 380 observations on the following 7 variables.
</p>

<dl>
<dt>month</dt><dd><p>Month of match</p>
</dd>
<dt>day</dt><dd><p>Day of match</p>
</dd>
<dt>year</dt><dd><p>Year of match</p>
</dd>
<dt>team1</dt><dd><p>Home team</p>
</dd>
<dt>team2</dt><dd><p>Away team</p>
</dd>
<dt>score1</dt><dd><p>Goals scored by home team</p>
</dd>
<dt>score2</dt><dd><p>Goals scored by away team</p>
</dd>
</dl>



<h3>Source</h3>

<p>http://www.soccerbase.com/footballlive/
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 499.</p>

<hr>
<h2 id='springs'>Spring Failure Data</h2><span id='topic+springs'></span>

<h3>Description</h3>

<p>Failure times of 60 springs divided into 6 groups of 10, with each group
subject to a different level of stress.  Some of the times are right-censored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(springs)</code></pre>


<h3>Format</h3>

<p>A data frame with 60 observations on the following 3 variables.
</p>

<dl>
<dt>cycles</dt><dd><p>failure times (in units of $10^3$ cycles of loading)</p>
</dd>
<dt>cens</dt><dd><p>censoring indicator, with 0 indicating right-censoring</p>
</dd>
<dt>stress</dt><dd><p>a factor giving the stress (N/mm$^3$)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Cox, D. R. and Oakes, D. (1984)  <em>Analysis of Survival Data</em>.  
London: Chapman and Hall/CRC Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(springs)
attach(springs)
plot(cycles~stress)
plot(cycles~stress,log="y")
</code></pre>

<hr>
<h2 id='sticky'>Stickiness of blood data</h2><span id='topic+sticky'></span>

<h3>Description</h3>

<p>Data on stickiness of blood for six subjects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sticky)</code></pre>


<h3>Format</h3>

<p>A data frame with 42 observations on the following 2 variables.
</p>

<dl>
<dt>subject</dt><dd><p>factor with levels 1&ndash;6</p>
</dd>
<dt>y</dt><dd><p>measurement of a property related to stickiness of blood</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College London.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 450.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sticky)
anova(lm(y~subject,data=sticky))
</code></pre>

<hr>
<h2 id='survival'>Survival of Rats After Radiation Doses</h2><span id='topic+survival'></span>

<h3>Description</h3>

<p>The &lsquo;survival&rsquo; data frame has 14 rows and 2 columns.
</p>
<p>The data measured the survival percentages of batches of rats who
were given  varying doses of radiation.  At each of 6 doses there
were two or three replications of the experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(survival)</code></pre>


<h3>Format</h3>

<p>A data frame with 14 observations on the following 2 variables.
</p>

<dl>
<dt>dose</dt><dd><p>The dose of radiation administered (rads).</p>
</dd>
<dt>surv</dt><dd><p>The survival rate of the batches expressed as a percentage.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Efron, B. (1988) Computer-intensive methods in statistical
regression. <em>SIAM Review</em>, <b>30</b>, 421-449.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 376.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(survival)
plot(survival$dose,survival$surv,log="y")  # note the obvious outlier
lm(log(surv)~dose,data=survival)
</code></pre>

<hr>
<h2 id='teak'>Teak Plant Data</h2><span id='topic+teak'></span>

<h3>Description</h3>

<p>These are data from an experiment on the growth of teak plants after one season, using
two planting methods and three root lengths.  Plants were laid out in four randomised 
blocks, each consisting of 6 plots with 50 plants in each plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(teak)</code></pre>


<h3>Format</h3>

<p>A data frame with 24 observations on the following 4 variables.
</p>

<dl>
<dt>Block</dt><dd><p>Block labels.</p>
</dd>
<dt>Plant</dt><dd><p><code>A</code> indicates planting using pits, <code>B</code> using crowbar.</p>
</dd>
<dt>Root</dt><dd><p>length,  <code>4</code>, <code>6</code> or <code>8</code> inches.</p>
</dd>
<dt>y</dt><dd><p>mean height (inches) of the 50 plants grown on each plot.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Unpublished lecture notes, Imperial College, London.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(teak)
anova(lm(y~Block*Plant*Root,data=teak),test="F")
</code></pre>

<hr>
<h2 id='tide'>Annual Maximum Sea Levels</h2><span id='topic+tide'></span>

<h3>Description</h3>

<p>Annual maximum sea levels (m) at seven locations near to or in
south-east England, between 1819&ndash;1986.  There are many missing
values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tide)</code></pre>


<h3>Format</h3>

<p>A data frame with 168 observations on the following 8 variables.
</p>

<dl>
<dt>year</dt><dd><p>Year</p>
</dd>
<dt>Yarmouth</dt><dd><p>Annual maximum high tide at Yarmouth</p>
</dd>
<dt>Lowestoft</dt><dd><p>Annual maximum high tide at Lowestoft</p>
</dd>
<dt>Harwich</dt><dd><p>Annual maximum high tide at Harwich</p>
</dd>
<dt>Walton</dt><dd><p>Annual maximum high tide at Walton</p>
</dd>
<dt>Holland</dt><dd><p>Annual maximum high tide at a site in Holland</p>
</dd>
<dt>Southend</dt><dd><p>Annual maximum high tide at Southend</p>
</dd>
<dt>Sheerness</dt><dd><p>Annual maximum high tide at Sheerness</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were kindly provided by Professor Jonathan Tawn of 
Lancaster University.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 281.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tide)
plot(tide$year,tide$Yarmouth,type="l")
</code></pre>

<hr>
<h2 id='toxo'>Toxoplasmosis Data</h2><span id='topic+toxo'></span>

<h3>Description</h3>

<p>Data on the relation between rainfall and the numbers of people testing
positive for toxoplasmosis in 34 cities in El Salvador.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(toxo)</code></pre>


<h3>Format</h3>

<p>A data frame with 34 observations on the following 3 variables.
</p>

<dl>
<dt>rain</dt><dd><p>Annual rainfall (mm)</p>
</dd>
<dt>m</dt><dd><p>Number of persons tested</p>
</dd>
<dt>r</dt><dd><p>Number of persons testing positive for toxoplasmosis</p>
</dd>
</dl>



<h3>Source</h3>

  
<p>Efron, B. (1986)  Double exponential families and their use in generalized 
linear regression.  <em>Journal of the American Statistical Association</em>, 
<b>82</b>, 
171&ndash;200.</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 516.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(toxo)
anova(glm(cbind(r,m-r)~poly(rain,3),data=toxo,family=binomial),test="Chi")
fit &lt;- glm(cbind(r,m-r)~poly(rain,3),data=toxo,family=quasibinomial)
anova(fit,test="F")
summary(fit)
</code></pre>

<hr>
<h2 id='ulcer'>Recurrent Bleeding from Ulcers</h2><span id='topic+ulcer'></span>

<h3>Description</h3>

<p>Data from 40 experiments to compare a new surgery for stomach ulcer
with an older surgery.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ulcer)</code></pre>


<h3>Format</h3>

<p>A data frame with 80 observations on the following 9 variables.
</p>

<dl>
<dt>author</dt><dd><p>Author of study from which data taken</p>
</dd>
<dt>year</dt><dd><p>Year of publication</p>
</dd>
<dt>quality</dt><dd><p>Assessment of quality of trial on which data based</p>
</dd>
<dt>age</dt><dd><p>Mean age of patients</p>
</dd>
<dt>r</dt><dd><p>Number of patients without recurrent bleeding</p>
</dd>
<dt>m</dt><dd><p>Total number of patients</p>
</dd>
<dt>bleed</dt><dd><p>a numeric vector</p>
</dd>
<dt>treat</dt><dd><p>Factor giving control (C) or variants of new treatment</p>
</dd>
<dt>table</dt><dd><p>Factor giving 2x2 table corresponding to each trial</p>
</dd>
</dl>



<h3>Source</h3>

<p>Efron, B. (1996) Empirical Bayes methods for combining likelihoods (with
Discussion).
<em>Journal of the American Statistical Association</em>, <b>91</b>,  538&ndash;565.
</p>
<p>Errors in the data given in the paper have been corrected 
here.</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  Cambridge University Press.
Page 496.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ulcer)
glm(cbind(r,m-r)~table+treat,data=ulcer,family=binomial)

</code></pre>

<hr>
<h2 id='urine'> Urine Analysis Data</h2><span id='topic+urine'></span>

<h3>Description</h3>

<p>The 'urine' data frame has 79 rows and 7 columns.
</p>
<p>79 urine specimens were analyzed in an effort to determine if
certain physical characteristics of the urine might be related to
the formation of calcium oxalate crystals.  Cases 1 and 55 have
missing covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(urine)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>r</dt><dd><p>Indicator of the presence of calcium oxalate crystals.</p>
</dd>
<dt>gravity</dt><dd><p>The specific gravity of the urine.</p>
</dd>
<dt>ph</dt><dd><p>The pH reading of the urine.</p>
</dd>
<dt>osmo</dt><dd><p>The osmolarity of the urine. Osmolarity is proportional to
the concentration of molecules in solution.</p>
</dd>
<dt>cond</dt><dd><p>The conductivity of the urine.  Conductivity is
proportional to the  concentration of charged ions in
solution.</p>
</dd>
<dt>urea</dt><dd><p>The urea concentration in millimoles per litre.</p>
</dd>
<dt>calc</dt><dd><p>The calcium concentration in millimoles per litre.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Andrews, D.F. and Herzberg, A.M. (1985)  <em>Data: A Collection of
Problems from Many Fields for the Student and Research Worker</em>. 
Springer-Verlag.  Pages 249&ndash;251.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(urine)
glm(r~gravity+ph+osmo+cond+urea+calc,binomial,data=urine,subset=-c(1,55))
</code></pre>

<hr>
<h2 id='venice'>Extreme Sea Levels at Venice</h2><span id='topic+venice'></span>

<h3>Description</h3>

<p>The ten highest annual sea levels (cm) at Venice, from 1887&ndash;1981.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(venice)</code></pre>


<h3>Format</h3>

<p>A data frame with 95 observations on the following 11 variables.
</p>

<dl>
<dt>year</dt><dd><p>1887&ndash;1981</p>
</dd>
<dt>y1</dt><dd><p>Annual maximum sea level (cm)</p>
</dd>
<dt>y2</dt><dd><p>Second largest sea level (cm)</p>
</dd>
<dt>y3</dt><dd><p>Third largest sea level (cm)</p>
</dd>
<dt>y4</dt><dd><p>Fourth largest sea level (cm)</p>
</dd>
<dt>y5</dt><dd><p>Fifth largest sea level (cm)</p>
</dd>
<dt>y6</dt><dd><p>Sixth largest sea level (cm)</p>
</dd>
<dt>y7</dt><dd><p>Seventh largest sea level (cm)</p>
</dd>
<dt>y8</dt><dd><p>Eighth largest sea level (cm)</p>
</dd>
<dt>y9</dt><dd><p>Ninth largest sea level (cm)</p>
</dd>
<dt>y10</dt><dd><p>Tenth largest sea level (cm)</p>
</dd>
</dl>



<h3>Details</h3>

<p>There are missing values in 1922 and 1935.  
</p>


<h3>Source</h3>

<p>Pirazzoli, P. A. (1982) Maree estreme a Venezia (periodo 1872&ndash;1981).
<em>Acqua Aria</em>, <b>10</b>, 1023&ndash;1039.
</p>


<h3>References</h3>

<p>Davison, A. C. (2003)  <em>Statistical Models</em>.  
Cambridge University Press.  Page 162.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(venice)
attach(venice)
y &lt;- y1[year&gt;1930]  # for analysis in Section 5 of Davison (2003)
x &lt;- year[year&gt;1930]-1956
plot(x+1956,y,ylab="Sea level (cm)",xlab="Year")
lm(y~x)
</code></pre>

<hr>
<h2 id='yahoo'>Yahoo Closing Prices</h2><span id='topic+yahoo'></span>

<h3>Description</h3>

<p>Daily closing prices (US dollars) of <code>Yahoo.com</code> shares from 12 April 1996 to 
26 April 2000.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(yahoo)</code></pre>


<h3>Format</h3>

<p>An irregular time series with 1017 values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(yahoo)
plot(yahoo,type="l",ylab="Yahoo closing prices")
plot(diff(100*log(yahoo)),type="l",ylab="Yahoo log returns (percent)")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
