<!DOCTYPE html><html><head><title>Help for package SMUT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SMUT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#eigenMapMatMult'>
<p>Matrix multiplication using RcppEigen</p></a></li>
<li><a href='#Generalized_Testing_coefficient_of_mediator'>
<p>Testing coefficient of mediator in the outcome model for an outcome following an exponential family distribution or a survival outcome</p></a></li>
<li><a href='#Genotype_data'><p>Example genotype data for SMUT</p></a></li>
<li><a href='#GSMUT'>
<p>Generalized Multi-SNP Mediation Intersection-Union Test</p></a></li>
<li><a href='#SMUT'>
<p>Multi-SNP Mediation Intersection-Union Test</p></a></li>
<li><a href='#Testing_coefficient_of_mediator'>
<p>Testing coefficient of mediator in the outcome model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multi-SNP Mediation Intersection-Union Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Wujuan Zhong</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wujuan Zhong &lt;zhongwujuan@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Testing the mediation effect of multiple SNPs on an outcome through a mediator.</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.14), SKAT, MASS</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-09-23 23:03:50 UTC; Fox</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-09-24 04:40:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='eigenMapMatMult'>
Matrix multiplication using RcppEigen
</h2><span id='topic+eigenMapMatMult'></span>

<h3>Description</h3>

<p>Matrix multiplication using RcppEigen.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eigenMapMatMult(A,B)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eigenMapMatMult_+3A_a">A</code>, <code id="eigenMapMatMult_+3A_b">B</code></td>
<td>

<p>numeric (double) complex matrices or vectors.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The matrix product. The value is the same as A %*% B
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SMUT)
A=matrix(1:9,3,3)
A=A+0
B=as.matrix(c(5.0, 2.0, 0.0))
eigenMapMatMult(A,B)
# the result is the same as A %*% B

# Thanks for using our R package SMUT 
</code></pre>

<hr>
<h2 id='Generalized_Testing_coefficient_of_mediator'>
Testing coefficient of mediator in the outcome model for an outcome following an exponential family distribution or a survival outcome
</h2><span id='topic+Generalized_Testing_coefficient_of_mediator'></span>

<h3>Description</h3>

<p>Testing coefficient of mediator, namely theta, in the outcome model. The outcome model is the following. <br />
outcome ~ intercept + G*gamma + mediator*theta + error
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Generalized_Testing_coefficient_of_mediator(G,mediator,outcome,
covariates=NULL,outcome_type,
approxi=TRUE,verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Generalized_Testing_coefficient_of_mediator_+3A_g">G</code></td>
<td>

<p>n by p matrix (n rows and p columns). Each row is one individual; each column is one SNP.
</p>
</td></tr>
<tr><td><code id="Generalized_Testing_coefficient_of_mediator_+3A_mediator">mediator</code></td>
<td>

<p>a vector length of n. It is the mediator variable.
</p>
</td></tr>
<tr><td><code id="Generalized_Testing_coefficient_of_mediator_+3A_outcome">outcome</code></td>
<td>

<p>a vector length of n. It is the outcome variable.
</p>
</td></tr>
<tr><td><code id="Generalized_Testing_coefficient_of_mediator_+3A_covariates">covariates</code></td>
<td>

<p>n by r matrix (n rows and r columns). Each row is one individual; each column is one covariate.
</p>
</td></tr>
<tr><td><code id="Generalized_Testing_coefficient_of_mediator_+3A_outcome_type">outcome_type</code></td>
<td>

<p>Type of the outcome variable. &quot;continuous&quot; for a continuous outcome; &quot;binary&quot; for a binary outcome; &quot;count&quot; for a count outcome; &quot;survival&quot; for a survival outcome.
</p>
</td></tr>
<tr><td><code id="Generalized_Testing_coefficient_of_mediator_+3A_approxi">approxi</code></td>
<td>

<p>a boolean value. This is an indicator whether the approximation of computing derivatives is applied to save computing time. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="Generalized_Testing_coefficient_of_mediator_+3A_verbose">verbose</code></td>
<td>

<p>a boolean value. If TRUE a lot of computing details is printed. Default is FALSE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p_value</code></td>
<td>

<p>P value for testing the coefficient of mediator in the outcome model.
</p>
</td></tr>
<tr><td><code>theta_hat</code></td>
<td>

<p>The point estimate of theta (coefficient of mediator) in the outcome model. 
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wujuan Zhong
</p>

<hr>
<h2 id='Genotype_data'>Example genotype data for SMUT</h2><span id='topic+Genotype_data'></span>

<h3>Description</h3>

<p>Example genotype data for SMUT. It is a matrix with 100 rows and 200 columns. Each row is an individual; each column is a SNP.
</p>


<h3>Format</h3>

<p>It is a matrix with 100 rows and 200 columns. Each row is an individual; each column is a SNP.
</p>

<hr>
<h2 id='GSMUT'>
Generalized Multi-SNP Mediation Intersection-Union Test
</h2><span id='topic+GSMUT'></span>

<h3>Description</h3>

<p>Testing the mediation effect of multiple SNPs on an outcome following an exponential family distribution or a survival outcome through a continuous mediator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSMUT(G,mediator,outcome,covariates=NULL,outcome_type,
approxi=TRUE,verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSMUT_+3A_g">G</code></td>
<td>

<p>n by p matrix (n rows and p columns). Each row is one individual; each column is one SNP.
</p>
</td></tr>
<tr><td><code id="GSMUT_+3A_mediator">mediator</code></td>
<td>

<p>a vector length of n. It is the mediator variable.
</p>
</td></tr>
<tr><td><code id="GSMUT_+3A_outcome">outcome</code></td>
<td>

<p>a vector length of n. It is the outcome variable.
</p>
</td></tr>
<tr><td><code id="GSMUT_+3A_covariates">covariates</code></td>
<td>

<p>n by r matrix (n rows and r columns). Each row is one individual; each column is one covariate.
</p>
</td></tr>
<tr><td><code id="GSMUT_+3A_outcome_type">outcome_type</code></td>
<td>

<p>Type of the outcome variable. &quot;continuous&quot; for a continuous outcome; &quot;binary&quot; for a binary outcome; &quot;count&quot; for a count outcome; &quot;survival&quot; for a survival outcome.
</p>
</td></tr>
<tr><td><code id="GSMUT_+3A_approxi">approxi</code></td>
<td>

<p>a boolean value. This is an indicator whether the approximation of computing derivatives is applied to save computing time. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="GSMUT_+3A_verbose">verbose</code></td>
<td>

<p>a boolean value. If TRUE a lot of computing details is printed. Default is FALSE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p_value_IUT</code></td>
<td>

<p>The p value for testing the mediation effect (theta*beta) based on intersection-union test.
</p>
</td></tr>
<tr><td><code>p_value_theta</code></td>
<td>

<p>The p value for testing theta in the outcome model.
The outcome model is the following. <br />
outcome ~ intercept + covariates*iota + G*gamma + mediator*theta 
</p>
</td></tr>
<tr><td><code>theta_hat</code></td>
<td>

<p>The point estimate of theta (coefficient of mediator) in the outcome model. 
</p>
</td></tr>
<tr><td><code>p_value_beta</code></td>
<td>

<p>The p value for testing beta in the mediator model.
The mediator model is the following. <br />
mediator ~ intercept + covariates*iota + G*beta + error
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wujuan Zhong
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SMUT)
# load the Genotype data included in this R package
data("Genotype_data")


##### for a binary outcome #####
set.seed(1)

# generate two covariates
covariate_1=rnorm(nrow(Genotype_data),0,1)
covariate_2=sample(c(0,1),size=nrow(Genotype_data),replace = TRUE)
covariates=cbind(covariate_1,covariate_2)

# generate a mediator 
beta=rnorm(ncol(Genotype_data),0,0.5)
tau_M=c(-0.3,0.2)
e1 = rnorm(nrow(Genotype_data), 0, 1)
mediator = 1 + eigenMapMatMult(Genotype_data,beta) + 
           eigenMapMatMult(covariates, tau_M) + e1

#### generate a binary outcome ####
theta=1
gamma=rnorm(ncol(Genotype_data),0,0.5)
tau=c(-0.2,0.2)
eta=1 + eigenMapMatMult(Genotype_data, gamma) + 
    eigenMapMatMult(covariates, tau) + theta * mediator
pi=1/(1+exp( -(eta ) ))
outcome=rbinom(length(pi),size=1,prob=pi) 
result=GSMUT(G=Genotype_data,mediator=mediator,outcome=outcome,
covariates=covariates,outcome_type="binary")
print(result)
# p_value_IUT is the p value for the mediation effect.

## Not run: 

##### generate a count outcome #####
theta=1
gamma=rnorm(ncol(Genotype_data),0,0.5)
tau=c(-0.2,0.2)
eta=1 + eigenMapMatMult(Genotype_data, gamma) + 
    eigenMapMatMult(covariates, tau) + theta * mediator
mu_param=exp(eta) # the mean parameter
phi_param=10  # the shape parameter
outcome=rnbinom(length(mu_param),size=phi_param,mu=mu_param)
result=GSMUT(G=Genotype_data,mediator=mediator,outcome=outcome,
covariates=covariates,outcome_type="count")
print(result)
# p_value_IUT is the p value for the mediation effect.

##### generate a survival outcome #####
theta=2
gamma=rnorm(ncol(Genotype_data),0,0.5)
tau=c(-0.2,0.2)
eta=1 + eigenMapMatMult(Genotype_data, gamma) + 
    eigenMapMatMult(covariates, tau) + theta * mediator
v=runif(nrow(Genotype_data))
lambda=0.01; rho=1; rateC=0.001
Tlat=(- log(v) / (lambda * exp( eta  )))^(1 / rho)
# censoring times
C= rexp(nrow(Genotype_data), rate=rateC)
# follow-up times and event indicators
time= pmin(Tlat, C)
status= as.numeric(Tlat &lt;= C)
outcome=cbind(time,status)      
colnames(outcome)=c("time","status")
result=GSMUT(G=Genotype_data,mediator=mediator,outcome=outcome,
covariates=covariates,outcome_type="survival")
print(result)
# p_value_IUT is the p value for the mediation effect.


## End(Not run) 


</code></pre>

<hr>
<h2 id='SMUT'>
Multi-SNP Mediation Intersection-Union Test
</h2><span id='topic+SMUT'></span>

<h3>Description</h3>

<p>Testing the mediation effect of multiple SNPs on an outcome through a mediator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMUT(G, mediator, outcome, 
outcome_type="continuous", method="score",approxi=TRUE, debug=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SMUT_+3A_g">G</code></td>
<td>

<p>n by p matrix (n rows and p columns). Each row is one individual; each column is one SNP.
</p>
</td></tr>
<tr><td><code id="SMUT_+3A_mediator">mediator</code></td>
<td>

<p>a vector length of n. It is the mediator variable.
</p>
</td></tr>
<tr><td><code id="SMUT_+3A_outcome">outcome</code></td>
<td>

<p>a vector length of n. It is the outcome variable.
</p>
</td></tr>
<tr><td><code id="SMUT_+3A_outcome_type">outcome_type</code></td>
<td>

<p>Type of the outcome variable. For now, this package only deals with continuous outcome. Default is &quot;continuous&quot;.
</p>
</td></tr>
<tr><td><code id="SMUT_+3A_method">method</code></td>
<td>

<p>The method of testing coefficient of mediator in the outcome model. The score test is used. Default is &quot;score&quot;.
</p>
</td></tr>
<tr><td><code id="SMUT_+3A_approxi">approxi</code></td>
<td>

<p>a boolean value. This is an indicator whether the approximation of the score statistic is applied to save computing time. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="SMUT_+3A_debug">debug</code></td>
<td>

<p>a boolean value. If TRUE a lot of computing details is printed; otherwise the
function is completely silent. Default is FALSE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p_value_IUT</code></td>
<td>

<p>The p value for testing the mediation effect (theta*beta) based on intersection-union test.
</p>
</td></tr>
<tr><td><code>p_value_theta</code></td>
<td>

<p>The p value for testing theta in the outcome model.
The outcome model is the following. <br />
outcome ~ intercept + G*gamma + mediator*theta + error.
</p>
</td></tr>
<tr><td><code>p_value_beta</code></td>
<td>

<p>The p value for testing beta in the mediator model.
The mediator model is the following. <br />
mediator ~ intercept + G*beta + error
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wujuan Zhong
</p>


<h3>References</h3>

<p>Zhong, W., Spracklen, C. N., Mohlke, K. L., Zheng, X., Fine, J., &amp; Li, Y. (2019). Multi-SNP mediation intersection-union test. <em>Bioinformatics</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SMUT)
# load the Genotype data included in this R package
data("Genotype_data")

# generate one mediator and one outcome

# first example, the mediation effect is significant
set.seed(1)
beta=rnorm(ncol(Genotype_data),1,2)
e1 = rnorm(nrow(Genotype_data), 0, 1)
mediator = 1 + eigenMapMatMult(Genotype_data,beta) + e1

theta=0.8
gamma=rnorm(ncol(Genotype_data),0.5,2)
e2 = rnorm(nrow(Genotype_data), 0, 1)
outcome = 2 + eigenMapMatMult(Genotype_data,gamma) + theta*mediator + e2

p_value=SMUT(G=Genotype_data,mediator=mediator,outcome=outcome)
print(p_value)

# p_value_IUT is the p value for the mediation effect.
# we have significant(at alpha level 0.05) mediation effects (p_value_IUT = 0.001655787).

# second example, the mediation effect is non-significant
set.seed(1)
beta=rnorm(ncol(Genotype_data),1,2)
e1 = rnorm(nrow(Genotype_data), 0, 1)
mediator = 1 + eigenMapMatMult(Genotype_data,beta) + e1

theta=0
gamma=rnorm(ncol(Genotype_data),0.5,2)
e2 = rnorm(nrow(Genotype_data), 0, 1)
outcome = 2 + eigenMapMatMult(Genotype_data,gamma) + theta*mediator + e2

p_value=SMUT(G=Genotype_data,mediator=mediator,outcome=outcome)
print(p_value)

# p_value_IUT is the p value for the mediation effect.
# we have non-significant(at alpha level 0.05) mediation effects (p_value_IUT = 0.3281677).

# third example, the mediation effect is non-significant
set.seed(1)
beta=rep(0,ncol(Genotype_data))
e1 = rnorm(nrow(Genotype_data), 0, 1)
mediator = 1 + eigenMapMatMult(Genotype_data,beta) + e1

theta=0.8
gamma=rnorm(ncol(Genotype_data),0.5,2)
e2 = rnorm(nrow(Genotype_data), 0, 1)
outcome = 2 + eigenMapMatMult(Genotype_data,gamma) + theta*mediator + e2

p_value=SMUT(G=Genotype_data,mediator=mediator,outcome=outcome)
print(p_value)

# p_value_IUT is the p value for the mediation effect.
# we have non-significant(at alpha level 0.05) mediation effects (p_value_IUT = 0.5596977).

# Thanks for using our R package SMUT 
</code></pre>

<hr>
<h2 id='Testing_coefficient_of_mediator'>
Testing coefficient of mediator in the outcome model
</h2><span id='topic+Testing_coefficient_of_mediator'></span>

<h3>Description</h3>

<p>Testing coefficient of mediator, namely theta, in the outcome model. The outcome model is the following. <br />
outcome ~ intercept + G*gamma + mediator*theta + error
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Testing_coefficient_of_mediator(G, mediator, outcome, 
outcome_type="continuous", method="score", approxi=TRUE, debug=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Testing_coefficient_of_mediator_+3A_g">G</code></td>
<td>

<p>n by p matrix (n rows and p columns). Each row is one individual; each column is one SNP.
</p>
</td></tr>
<tr><td><code id="Testing_coefficient_of_mediator_+3A_mediator">mediator</code></td>
<td>

<p>a vector length of n. It is the mediator variable.
</p>
</td></tr>
<tr><td><code id="Testing_coefficient_of_mediator_+3A_outcome">outcome</code></td>
<td>

<p>a vector length of n. It is the outcome variable.
</p>
</td></tr>
<tr><td><code id="Testing_coefficient_of_mediator_+3A_outcome_type">outcome_type</code></td>
<td>

<p>Type of the outcome variable. For now, this package only deals with continuous outcome. Default is &quot;continuous&quot;.
</p>
</td></tr>
<tr><td><code id="Testing_coefficient_of_mediator_+3A_method">method</code></td>
<td>

<p>The method of testing coefficient of mediator in the outcome model. The score test is used. Default is &quot;score&quot;.
</p>
</td></tr>
<tr><td><code id="Testing_coefficient_of_mediator_+3A_approxi">approxi</code></td>
<td>

<p>a boolean value. This is an indicator whether the approximation of the score statistic is applied to save computing time. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="Testing_coefficient_of_mediator_+3A_debug">debug</code></td>
<td>

<p>a boolean value. If TRUE a lot of computing details is printed; otherwise the
function is completely silent. Default is FALSE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P value for testing the coefficient of mediator in the outcome model.
</p>


<h3>Author(s)</h3>

<p>Wujuan Zhong
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SMUT)
# load the Genotype data included in this R package
data("Genotype_data")

# generate one mediator and one outcome

set.seed(1)
beta=rnorm(ncol(Genotype_data),1,2)
e1 = rnorm(nrow(Genotype_data), 0, 1)
mediator = 1 + eigenMapMatMult(Genotype_data,beta) + e1

theta=0.8
gamma=rnorm(ncol(Genotype_data),0.5,2)
e2 = rnorm(nrow(Genotype_data), 0, 1)
outcome = 2 + eigenMapMatMult(Genotype_data,gamma) + theta*mediator + e2

p_value=Testing_coefficient_of_mediator(G=Genotype_data,mediator=mediator,outcome=outcome)
print(p_value)

# Thanks for using our R package SMUT 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
