<!DOCTYPE html><html lang="en"><head><title>Help for package SMVar</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SMVar}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ApoAIdata'><p> ApoAIdata</p></a></li>
<li><a href='#SMVar-package'><p> Structural Model for Variances</p></a></li>
<li><a href='#SMVar.paired'><p> Structural model for variances with paired data</p></a></li>
<li><a href='#SMVar.unpaired'><p> Structural model for variances with unpaired data</p></a></li>
<li><a href='#Spleendata'><p> Spleendata</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Structural Model for Variances</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-04-12</td>
</tr>
<tr>
<td>Author:</td>
<td>Guillemette Marot [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samuel Blanck &lt;samuel.blanck@univ-lille.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.6.0)</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of the structural model for variances in order to
        detect differentially expressed genes from gene expression data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-12 08:03:40 UTC; sblanck</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-12 16:12:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='ApoAIdata'> ApoAIdata </h2><span id='topic+ApoAIdata'></span>

<h3>Description</h3>

 
<p>Example dataset for unpaired data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ApoAIdata)</code></pre>


<h3>Format</h3>

<p>ApoAIdata is a list with 3 elements
</p>

<dl>
<dt>ApoAIGeneId</dt><dd><p> vector of fictive gene names)</p>
</dd>
<dt>ApoAICond1</dt><dd><p> matrix with 6226 rows and 8 columns with normalized normal mice measurements</p>
</dd>
<dt>ApoAICond2</dt><dd><p> matrix with 6226 rows and 8 columns with normalized KO mice measurements</p>
</dd>
</dl>



<h3>Source</h3>

<p>Similar to the example dataset used in the package Varmixt
</p>


<h3>References</h3>

<p>M.J. Callow, S. Dudoit, E.L. Gong, T.P. Speed, and E.M. Rubin. Microarray expression profiling identifies genes with altered expression in hdl-deficien mice. Genome Res., 10(12) : 2022-9, 2000
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ApoAIdata)
attach(ApoAIdata)
</code></pre>

<hr>
<h2 id='SMVar-package'> Structural Model for Variances  </h2><span id='topic+SMVar'></span>

<h3>Description</h3>

<p>Package containing moderated t-tests to detect differentially expressed genes for paired and unpaired data</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> SMVar</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.3.3</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2011-08-03</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>SMVar.unpaired and SMVar.paired are the most important functions. 
</p>


<h3>Author(s)</h3>

<p>Guillemette Marot &lt;guillemette.marot@inria.fr&gt;
</p>


<h3>References</h3>

<p>F. Jaffrezic, Marot, G., Degrelle, S., Hue, I. and Foulley, J. L. (2007) A structural mixed model for variances in differential gene expression studies. Genetical Research (89) 19:25 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SMVar)
data(ApoAIdata)
attach(ApoAIdata)
SMVar.unpaired(ApoAIGeneId,list(ApoAICond1,ApoAICond2))
</code></pre>

<hr>
<h2 id='SMVar.paired'> Structural model for variances with paired data </h2><span id='topic+SMVar.paired'></span>

<h3>Description</h3>

<p>Function to detect differentially expressed genes when data are paired
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMVar.paired(geneNumbers, logratio, fileexport = NULL, 
            minrep = 2, method = "BH", threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SMVar.paired_+3A_genenumbers">geneNumbers</code></td>
<td>
<p> Vector with gene names or dataframe which contains all information about spots on the chip </p>
</td></tr>
<tr><td><code id="SMVar.paired_+3A_logratio">logratio</code></td>
<td>
<p> matrix with one row by gene and one column by replicate giving the logratio </p>
</td></tr>
<tr><td><code id="SMVar.paired_+3A_fileexport">fileexport</code></td>
<td>
<p> file to export the list of differentially expressed genes </p>
</td></tr>
<tr><td><code id="SMVar.paired_+3A_minrep">minrep</code></td>
<td>
<p> minimum number of replicates to take a gene into account, <code>minrep</code> must be higher than 2 </p>
</td></tr>
<tr><td><code id="SMVar.paired_+3A_method">method</code></td>
<td>
<p> method of multiple tests adjustment for p.values </p>
</td></tr>
<tr><td><code id="SMVar.paired_+3A_threshold">threshold</code></td>
<td>
<p> threshold of False Discovery Rate </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements the structural model for variances described in (Jaffrezic et al., 2007).
Data must be normalized before calling the function. Matrix <code>geneNumbers</code> must have one of
the following formats: &quot;matrix&quot;,&quot;data.frame&quot;,&quot;vector&quot;,&quot;character&quot;,&quot;numeric&quot;,&quot;integer&quot;.
</p>


<h3>Value</h3>

<p>Only the number of differentially expressed genes is printed. 
If asked, the file giving the list of differentially expressed genes is created
</p>
<p>If the user creates an object when calling the function (for example &quot;Stat=SMVar.paired(...)&quot;)
then Stat contains the information for all genes, is sorted by ascending p-values and
</p>
<table role = "presentation">
<tr><td><code>Stat$TestStat</code></td>
<td>
<p>gives the test statistics as described in the paper </p>
</td></tr>
<tr><td><code>Stat$StudentPValue</code></td>
<td>
<p> gives the raw p-values</p>
</td></tr>
<tr><td><code>Stat$DegOfFreedom</code></td>
<td>
<p>gives the number of degrees of freedom for the Student distribution for the test statistics</p>
</td></tr>
<tr><td><code>Stat$LogRatio</code></td>
<td>
<p>  gives the logratios</p>
</td></tr>
<tr><td><code>Stat$AdjPValue</code></td>
<td>
<p>gives the adjusted p-values</p>
</td></tr>
</table>


<h3>Note</h3>

<p> If the first column of the file geneNumbers contains identical names for two different spots,
these two spots are only counted once if they are both differentially expressed.
By default, the correction for multiple testing is Benjamini Hochberg with a threshold of False Discovery Rate (FDR) of  5%.
The FDR threshold can be changed, and it is also possible to choose the multiple test correction method 
(&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;).
To see the references for these methods, use the R-help ?p.adjust.  
</p>


<h3>Author(s)</h3>

<p> Guillemette Marot with contributions from Anne de la Foye </p>


<h3>References</h3>

<p>  F. Jaffrezic, Marot, G., Degrelle, S., Hue, I. and Foulley, J. L. (2007) A structural mixed model for variances in differential gene expression studies. Genetical Research (89) 19:25 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SMVar)
data(Spleendata)
attach(Spleendata)
SMVar.paired(SpleenGeneId,SpleenLogRatio)
</code></pre>

<hr>
<h2 id='SMVar.unpaired'> Structural model for variances with unpaired data </h2><span id='topic+SMVar.unpaired'></span>

<h3>Description</h3>

<p>Function to detect differentially expressed genes when data are unpaired
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMVar.unpaired(geneNumbers, listcond, fileexport = NULL,
               minrep = 2, method = "BH", threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SMVar.unpaired_+3A_genenumbers">geneNumbers</code></td>
<td>
<p> Vector with gene names or dataframe which contains all information about spots on the chip </p>
</td></tr>
<tr><td><code id="SMVar.unpaired_+3A_listcond">listcond</code></td>
<td>
<p> list of the different conditions to be compared</p>
</td></tr>
<tr><td><code id="SMVar.unpaired_+3A_fileexport">fileexport</code></td>
<td>
<p> file to export the list of differentially expressed genes </p>
</td></tr>
<tr><td><code id="SMVar.unpaired_+3A_minrep">minrep</code></td>
<td>
<p> minimum number of replicates to take a gene into account, <code>minrep</code> must be higher than 2 </p>
</td></tr>
<tr><td><code id="SMVar.unpaired_+3A_method">method</code></td>
<td>
<p> method of multiple tests adjustment for p.values </p>
</td></tr>
<tr><td><code id="SMVar.unpaired_+3A_threshold">threshold</code></td>
<td>
<p> threshold of False Discovery Rate </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements the structural model for variances described in (Jaffrezic et al., 2007).
Data must be normalized before calling the function. Matrix <code>geneNumbers</code> must have one of
the following formats: &quot;matrix&quot;,&quot;data.frame&quot;,&quot;vector&quot;,&quot;character&quot;,&quot;numeric&quot;,&quot;integer&quot;.
</p>


<h3>Value</h3>

<p>Only the number of differentially expressed genes is printed. 
If asked, the file giving the list of differentially expressed genes is created.
</p>
<p>If the user creates an object when calling the function (for example &quot;Stat=SMVar.paired(...)&quot;)
then Stat contains the information for all genes, is sorted by ascending p-values and
</p>
<table role = "presentation">
<tr><td><code>Stat$TestStat</code></td>
<td>
<p>gives the test statistics as described in the paper</p>
</td></tr>
<tr><td><code>Stat$StudentPValue</code></td>
<td>
<p> gives the raw p-values</p>
</td></tr>
<tr><td><code>Stat$DegOfFreedom</code></td>
<td>
<p>gives the number of degrees of freedom for the Student distribution for the test statistics</p>
</td></tr>
<tr><td><code>Stat$Cond1</code></td>
<td>
<p>  gives the first condition considered in the log-ratio</p>
</td></tr>
<tr><td><code>Stat$Cond2</code></td>
<td>
<p>  gives the second condition considered in the log-ratio</p>
</td></tr> 
<tr><td><code>Stat$LogRatio</code></td>
<td>
<p>  gives the logratios (listcond[[Cond2]]-listcond[[Cond1]])</p>
</td></tr>
<tr><td><code>Stat$AdjPValue</code></td>
<td>
<p>gives the adjusted p-values </p>
</td></tr>
</table>


<h3>Note</h3>

<p> If the first column of the file geneNumbers contains identical names for two different spots,
these two spots are only counted once if they are both differentially expressed.
By default, the correction for multiple testing is Benjamini Hochberg with a threshold of False Discovery Rate (FDR) of  5%.
The FDR threshold can be changed, and it is also possible to choose the multiple test correction method 
(&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;).
To see the references for these methods, use the R-help ?p.adjust.
</p>


<h3>Author(s)</h3>

<p> Guillemette Marot with contributions from Anne de la Foye </p>


<h3>References</h3>

<p>  F. Jaffrezic, Marot, G., Degrelle, S., Hue, I. and Foulley, J. L. (2007) A structural mixed model for variances in differential gene expression studies. Genetical Research (89) 19:25 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SMVar)
data(ApoAIdata)
attach(ApoAIdata)
SMVar.unpaired(ApoAIGeneId,list(ApoAICond1,ApoAICond2))
</code></pre>

<hr>
<h2 id='Spleendata'> Spleendata </h2><span id='topic+Spleendata'></span>

<h3>Description</h3>

<p>Example dataset for paired data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Spleendata)</code></pre>


<h3>Format</h3>

<p>Spleendata is a list with 2 elements
</p>

<dl>
<dt> SpleenGeneId</dt><dd><p>Gene names)</p>
</dd>
<dt> SpleenLogRatio</dt><dd><p> Matrix with 4360 rows and 6 columns with normalized log-ratio</p>
</dd>
</dl>



<h3>Source</h3>

<p>Similar to the example dataset used in the package Varmixt
</p>


<h3>References</h3>

<p>P. Delmar, Robin, S., Tronik-Le Roux S. and Daudin J.-J. (2005) Mixture model on the variance for the differential analysis of gene expression data, JRSS series C, 54(1), 31:50</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Spleendata)
attach(Spleendata)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
