<!DOCTYPE html><html><head><title>Help for package SNPassoc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SNPassoc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#association'><p>Association analysis between a single SNP and a given phenotype</p></a></li>
<li><a href='#asthma'><p>SNP data on asthma case-control study</p></a></li>
<li><a href='#Bonferroni.sig'><p> Bonferroni correction of p values</p></a></li>
<li><a href='#GenomicControl'><p>Population substructure</p></a></li>
<li><a href='#getGeneSymbol'><p>Get gene symbol from a list of SNPs</p></a></li>
<li><a href='#getNiceTable'><p>Get Latex output</p></a></li>
<li><a href='#getSignificantSNPs'><p>Extract significant SNPs from an object of class 'WGassociation'</p></a></li>
<li><a href='#haplo.interaction'><p>Haplotype interaction with a covariate</p></a></li>
<li><a href='#HapMap'><p> SNPs from HapMap project</p></a></li>
<li><a href='#HapMap.SNPs.pos'><p> SNPs from HapMap project</p></a></li>
<li><a href='#inheritance'><p>Collapsing (or recoding) genotypes into different categories (generally two) depending on a given genetic mode of inheritance</p></a></li>
<li><a href='#int'>
<p>Identify interaction term</p></a></li>
<li><a href='#interactionPval'><p> Two-dimensional SNP analysis for association studies</p></a></li>
<li><a href='#intervals'><p>Print ORs and 95% confidence intervals for an object of class 'haplo.glm'</p></a></li>
<li><a href='#is.Monomorphic'><p>Check whether a SNP is Monomorphic</p></a></li>
<li><a href='#LD'><p>max-statistic for a 2x3 table</p></a></li>
<li><a href='#make.geno'><p>Create a group of locus objects from some SNPs, assign to 'model.matrix' class.</p></a></li>
<li><a href='#maxstat'><p>max-statistic for a 2x3 table</p></a></li>
<li><a href='#odds'><p>Extract odds ratios, 95% CI and pvalues</p></a></li>
<li><a href='#permTest'><p>Permutation test analysis</p></a></li>
<li><a href='#plot.WGassociation'><p>Function to plot -log p values from an object of class 'WGassociation'</p></a></li>
<li><a href='#plotMissing'><p> Plot of missing genotypes</p></a></li>
<li><a href='#qqpval'><p>Functions for inspecting population substructure</p></a></li>
<li><a href='#related'><p>Get related samples</p></a></li>
<li><a href='#resHapMap'><p> SNPs from HapMap project</p></a></li>
<li><a href='#scanWGassociation'><p>Whole genome association analysis</p></a></li>
<li><a href='#setupSNP'><p>Convert columns in a dataframe to class 'snp'</p></a></li>
<li><a href='#snp'><p>SNP object</p></a></li>
<li><a href='#SNPassoc-internal'><p>Internal SNPstat functions</p></a></li>
<li><a href='#SNPs'><p>SNPs in a case-control study</p></a></li>
<li><a href='#SNPs.info.pos'><p>SNPs in a case-control study</p></a></li>
<li><a href='#sortSNPs'><p>Sort a vector of SNPs by genomic position</p></a></li>
<li><a href='#Table.mean.se'><p>Descriptive sample size, mean, and standard error</p></a></li>
<li><a href='#Table.N.Per'><p> Descriptive sample size and percentage</p></a></li>
<li><a href='#tableHWE'><p>Test for Hardy-Weinberg Equilibrium</p></a></li>
<li><a href='#WGassociation'><p>Whole genome association analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.1-0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-12-13</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>haplo.stats, mvtnorm, parallel, survival, tidyr, plyr,
ggplot2, poisbinom</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, biomaRt, VariantAnnotation,
GenomicRanges, IRanges, S4Vectors, org.Hs.eg.db,
TxDb.Hsapiens.UCSC.hg19.knownGene, BiocStyle</td>
</tr>
<tr>
<td>Title:</td>
<td>SNPs-Based Whole Genome Association Studies</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions to perform most of the common analysis in genome 
    association studies are implemented. These analyses include descriptive 
    statistics and exploratory analysis of missing values, calculation of 
    Hardy-Weinberg equilibrium, analysis of association based on generalized
    linear models (either for quantitative or binary traits), and analysis
    of multiple SNPs (haplotype and epistasis analysis). Permutation test 
    and related tests (sum statistic and truncated product) are also 
    implemented. Max-statistic and genetic risk-allele score exact 
    distributions are also possible to be estimated. The methods are 
    described in Gonzalez JR et al., 2007 &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtm025">doi:10.1093/bioinformatics/btm025</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/isglobal-brge/SNPassoc">https://github.com/isglobal-brge/SNPassoc</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-14 19:53:28 UTC; mailos</td>
</tr>
<tr>
<td>Author:</td>
<td>Victor Moreno [aut],
  Juan R Gonzalez <a href="https://orcid.org/0000-0003-3267-2146"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Dolors Pelegri <a href="https://orcid.org/0000-0002-5993-3003"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dolors Pelegri &lt;dolors.pelegri@isglobal.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-14 20:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='association'>Association analysis between a single SNP and a given phenotype</h2><span id='topic+association'></span><span id='topic+print.snpOut'></span>

<h3>Description</h3>

<p>This function carries out an association analysis between a single SNP and
a dependent variable (phenotype) under five different genetic models (inheritance patterns): 
codominant, dominant, recessive, overdominant and log-additive. The phenotype may be quantitative 
or categorical. In the second case (e.g. case-control studies) this variable must be of class 'factor' 
with two levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
association(formula, data, model=c("all"), model.interaction=
       c("codominant"), subset, name.snp = NULL, quantitative = 
       is.quantitative(formula,data), genotypingRate= 0, 
       level = 0.95, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="association_+3A_formula">formula</code></td>
<td>
<p>a symbolic description of the model to be fited (a formula object). 
It might have either a continuous variable (quantitative traits) or a 
factor variable (case-control studies) as the response on the left of the <code>~</code> 
operator and a term corresponding to the SNP on the right. This term must be
of class <code>snp</code> (e.g. <code>~</code>snp(var), where var is a given SNP), and it is 
required. Terms with additional covariates on the right of the <code>~</code> operator may be 
added to fit an adjusted model (e.g., <code>~</code>var1+var2+...+varN+SNP). The formula 
allows to incorporate more than one object of class <code>snp</code>. In that case, 
the analysis is done for the first SNP which appears in the formula adjusted by the others 
covariates and other additional SNPs. </p>
</td></tr>
<tr><td><code id="association_+3A_data">data</code></td>
<td>
<p>a required dataframe of class 'setupSNP' containing the variables in the model.</p>
</td></tr>
<tr><td><code id="association_+3A_model">model</code></td>
<td>
<p>a character string specifying the type of genetic model (mode of inheritance) for the SNP. 
This indicates how the genotypes should be collapsed. Possible
values are &quot;codominant&quot;, &quot;dominant&quot;, &quot;recessive&quot;, &quot;overdominant&quot;, &quot;additive&quot; or &quot;all&quot;. The default
is &quot;all&quot; that fits the 5 possible genetic models. Only the first words are required, e.g &quot;co&quot;, &quot;do&quot;, etc.</p>
</td></tr>              
<tr><td><code id="association_+3A_model.interaction">model.interaction</code></td>
<td>
<p>a character string specifying the type of genetic model (mode of inheritance) assumed for
the SNP when it is included in a interaction term. Possible values are &quot;codominant&quot;, 
&quot;dominant&quot;, &quot;recessive&quot;, &quot;overdominant&quot;. The default is &quot;codominant&quot;.</p>
</td></tr>
<tr><td><code id="association_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to be used in the fitting process</p>
</td></tr>              
<tr><td><code id="association_+3A_name.snp">name.snp</code></td>
<td>
<p>optional label of the SNP variable to be printed.</p>
</td></tr>
<tr><td><code id="association_+3A_quantitative">quantitative</code></td>
<td>
<p>logical value indicating whether the phenotype (that which is in the
left of the operator ~ in 'formula' argument) is quantitative. The function 
'is.quantitative' returns FALSE when the phenotype
is a variable with two categories (i.e. indicating case-control status). Thus,
it is not a required argument but it may be modified by the user.</p>
</td></tr>
<tr><td><code id="association_+3A_genotypingrate">genotypingRate</code></td>
<td>
<p> minimum percentage of genotype rate for the SNP to be analyzed.
Default is 0% (e.g. all SNPs are analyzed). This parameter should not be
changed. It is used in the function 'WGassociation'.</p>
</td></tr>
<tr><td><code id="association_+3A_level">level</code></td>
<td>
<p>signification level for confidence intervals.</p>
</td></tr>
<tr><td><code id="association_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed through glm function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function should be called by the user when we are interested in analyzing an unique SNP.
It is recommended to use <code><a href="#topic+WGassociation">WGassociation</a></code> function when more than one SNP is studied. <br />
</p>


<h3>Value</h3>

<p>For each genetic model (codominant, dominant, recessive, overdominant, and log-additive) the function gives 
a matrix with sample size and percentages for each genotype, the Odds Ratio and its 95% confidence interval
(taking the most frequent homozygous genotype as the reference), the p-value corresponding to the likelihood ratio test obtained 
from a comparison with the null model, and the Akaike Information Criterion (AIC) of each genetic model. In the case 
of analyzing a quantitative trait, the function returns a matrix with sample size, mean and standard errors for each genotype, 
mean difference and its 95% confidence interval with respect to the most frequent homozygous genotype, 
the p-value obtained from an overall gene effect and the Akaike Information Criterion (AIC) of each genetic model. 
</p>
<p>When an interaction term (a categorical covariate with an SNP) is included in the model, three different tables are given. 
The first one correponds to the full interaction matrix where the ORs (or mean differences if a quantitative trait is analyzed) 
are expressed with respect to the non variant genotype and the first category of the covariate. The other two tables show the ORs 
and their 95% confidence intervals for both marginal models. P values for interaction and trend are also showed in the output. 
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>
<p>Iniesta R, Guino E, Moreno V. Statistical analysis of genetic polymorphisms in epidemiological studies.
Gac Sanit. 2005;19(4):333-41.
</p>
<p>Elston RC. Introduction and overview. Statistical methods in genetic epidemiology. Stat Methods Med Res.
2000;9:527-41.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+WGassociation">WGassociation</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)

# first, we create an object of class 'setupSNP'
datSNP&lt;-setupSNP(SNPs,6:40,sep="")

# case-control study, crude analysis
association(casco~snp10001, data=datSNP)

# case-control study, adjusted by sex and arterial blood pressure
association(casco~sex+snp10001+blood.pre, data=datSNP)


# quantitative trait, crude analysis
association(log(protein)~snp10001,data=datSNP)
# quantitative trait, adjusted by sex
association(log(protein)~snp10001+sex,data=datSNP)


#
# Interaction analysis
#

# Interaction SNP and factor
association(log(protein)~snp10001*sex+blood.pre, data=datSNP, 
          model="codominant")

# Interaction SNP and SNP (codominant and codominant)
association(log(protein)~snp10001*factor(snp10002)+blood.pre, 
          data=datSNP, model="codominant")

# Interaction SNP and SNP (dominant and recessive)
association(log(protein)~snp10001*factor(recessive(snp100019))+blood.pre, 
          data=datSNP, model="dominant")


</code></pre>

<hr>
<h2 id='asthma'>SNP data on asthma case-control study</h2><span id='topic+asthma'></span>

<h3>Description</h3>

<p><code>data.frame</code> with 51 SNPs and 6 epidemiological variables: country, gender, age, bmi, smoke ans case/control status.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("asthma")
</code></pre>


<h3>Format</h3>

<p>The <code>asthma</code> data frame has 1578 rows (individuals) and 57 columns (variables) of data from a genetic study on asthma.</p>


<h3>Value</h3>

<p>An <code>data.frame</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(asthma)
dim(asthma)
str(asthma)
</code></pre>

<hr>
<h2 id='Bonferroni.sig'> Bonferroni correction of p values</h2><span id='topic+Bonferroni.sig'></span>

<h3>Description</h3>

<p>This function shows the SNPs that are statistically significant after correcting
for the number of tests performed (Bonferroni correction) for an object of class
&quot;WGassociation&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'> Bonferroni.sig(x, model = "codominant", alpha = 0.05, 
      include.all.SNPs=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Bonferroni.sig_+3A_x">x</code></td>
<td>
<p>an object of class 'WGassociation'.</p>
</td></tr>
<tr><td><code id="Bonferroni.sig_+3A_model">model</code></td>
<td>
<p>a character string specifying the type of genetic model (mode of inheritance). This 
indicantes how the genotypes should be collapsed when 'plot.summary' is TRUE. Possible
values are &quot;codominant&quot;, &quot;dominant&quot;, &quot;recessive&quot;, &quot;overdominant&quot;, or &quot;log-additive&quot;. 
The default is &quot;codominant&quot;. Only the first words are required, e.g &quot;co&quot;, &quot;do&quot;, ... . </p>
</td></tr>
<tr><td><code id="Bonferroni.sig_+3A_alpha">alpha</code></td>
<td>
<p>nominal level of significance. Default is 0.05</p>
</td></tr>
<tr><td><code id="Bonferroni.sig_+3A_include.all.snps">include.all.SNPs</code></td>
<td>
<p>logical value indicating whether all SNPs are considered in the Bonferroni
correction. That is, the number of performed tests is equal to the number of SNPs or equal to the
number of SNPs where a p value may be computed. The default value is FALSE indicating that the 
number of tests is equal to the number of SNPs that are non Monomorphic and the rate of genotyping
is greater than the percentage indicated in the <code>GeneticModel.pval</code> function. </p>
</td></tr>     
</table>


<h3>Details</h3>

<p>After deciding the genetic model, the function shows the SNPs that are statistically significant at 
<code>alpha</code> level corrected by the number of performed tests. 
</p>


<h3>Value</h3>

<p>A data frame with the SNPs and the p values for those SNPs that are statistically significant
after Bonferroni correction
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+WGassociation">WGassociation</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
datSNP&lt;-setupSNP(SNPs,6:40,sep="")
ans&lt;-WGassociation(protein~1,data=datSNP,model="all")
Bonferroni.sig(ans, model="codominant", alpha=0.05, include.all.SNPs=FALSE)

</code></pre>

<hr>
<h2 id='GenomicControl'>Population substructure</h2><span id='topic+GenomicControl'></span>

<h3>Description</h3>

<p>This function estimates an inflation (or deflation) factor, lambda, as indicated in the paper
by Devlin et al. (2001) and corrects the p-values using this factor. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenomicControl(x, snp.sel)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenomicControl_+3A_x">x</code></td>
<td>
<p>an object of class 'WGassociation'. </p>
</td></tr>
<tr><td><code id="GenomicControl_+3A_snp.sel">snp.sel</code></td>
<td>
<p>SNPs used to compute lambda. Not required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method is only valid for 2x2 tables. This means that the object of class 'WGassociation' might not
have fitted the codominant model.
</p>
<p>See reference for further details.
</p>


<h3>Value</h3>

<p>The same object of class 'WGassociation' where the p-values have been corrected for genomic control.
</p>


<h3>References</h3>

<p>B Devlin, K Roeder, and S.A. Bacanu. Unbiased Methods for Population Based Association Studies. 
Genetic Epidemiology (2001) 21:273-84
</p>


<h3>See Also</h3>

<p><code><a href="#topic+qqpval">qqpval</a></code>, <code><a href="#topic+WGassociation">WGassociation</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(SNPs) 
datSNP&lt;-setupSNP(SNPs,6:40,sep="")
res&lt;-WGassociation(casco,datSNP,model=c("do","re","log-add"))

# Genomic Control 
resCorrected&lt;-GenomicControl(res)
</code></pre>

<hr>
<h2 id='getGeneSymbol'>Get gene symbol from a list of SNPs</h2><span id='topic+getGeneSymbol'></span>

<h3>Description</h3>

<p>Get gene symbol from a list of SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGeneSymbol(
  x,
  snpCol = 1,
  chrCol = 2,
  posCol = 3,
  db = TxDb.Hsapiens.UCSC.hg19.knownGene
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getGeneSymbol_+3A_x">x</code></td>
<td>
<p>data.frame containing: SNP name, chromosome and genomic position.</p>
</td></tr>
<tr><td><code id="getGeneSymbol_+3A_snpcol">snpCol</code></td>
<td>
<p>column of x having the SNP name. Default is 1.</p>
</td></tr>
<tr><td><code id="getGeneSymbol_+3A_chrcol">chrCol</code></td>
<td>
<p>column of x having the SNP chromosome. Default is 2.</p>
</td></tr>
<tr><td><code id="getGeneSymbol_+3A_poscol">posCol</code></td>
<td>
<p>column of x having the SNP position. Default is 3.</p>
</td></tr>
<tr><td><code id="getGeneSymbol_+3A_db">db</code></td>
<td>
<p>reference genome. Default is 'TxDb.Hsapiens.UCSC.hg19.knownGene'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame having initial information and gene symbol
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    
        snps = c('rs58108140','rs189107123','rs180734498','rs144762171')
        chr = c('chr1','chr1','chr1','chr1')
        pos = c(10583, 10611, 13302, 13327)
        
        x &lt;- data.frame(snps, chr, pos )
        
        getGeneSymbol(x)
    
</code></pre>

<hr>
<h2 id='getNiceTable'>Get Latex output</h2><span id='topic+getNiceTable'></span>

<h3>Description</h3>

<p>Create Latex output from association analyses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNiceTable( x )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getNiceTable_+3A_x">x</code></td>
<td>
<p>WGassociation object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The R output of specific association analyses exported into LaTeX 
</p>

<hr>
<h2 id='getSignificantSNPs'>Extract significant SNPs from an object of class 'WGassociation' </h2><span id='topic+getSignificantSNPs'></span>

<h3>Description</h3>

<p>Extract significant SNPs from an object of class 'WGassociation' when genomic 
information is available</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSignificantSNPs(x, chromosome, model, sig = 1e-15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSignificantSNPs_+3A_x">x</code></td>
<td>
<p>an object of class 'WGassociation'</p>
</td></tr>
<tr><td><code id="getSignificantSNPs_+3A_chromosome">chromosome</code></td>
<td>
<p> chromosome from which SNPs are extracted</p>
</td></tr>
<tr><td><code id="getSignificantSNPs_+3A_model">model</code></td>
<td>
<p> genetic model from which SNPs are extracted</p>
</td></tr>
<tr><td><code id="getSignificantSNPs_+3A_sig">sig</code></td>
<td>
<p> statistical significance level. The default is 1e-15</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>names</code></td>
<td>
<p>the name of SNPs</p>
</td></tr>
<tr><td><code>column</code></td>
<td>
<p>the columns corresponding to the SNPs in the original data frame</p>
</td></tr>
</table>
<p>...
</p>


<h3>See Also</h3>

  <p><code><a href="#topic+WGassociation">WGassociation</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(resHapMap)
# resHapMap contains the results for a log-additive genetic model

# to get the significant SNPs for chromosome 12
getSignificantSNPs(resHapMap,chromosome=12)
# to get the significant SNPs for chromosome 5
getSignificantSNPs(resHapMap,5)
# to get the significant SNPs for chromosome X at level 1e-8
getSignificantSNPs(resHapMap,5,sig=1e-8)

</code></pre>

<hr>
<h2 id='haplo.interaction'>Haplotype interaction with a covariate</h2><span id='topic+haplo.interaction'></span><span id='topic+print.haploOut'></span>

<h3>Description</h3>

<p>This function computes the ORs (or mean differences if a quantitative trait is analyzed) and their 95% confidence intervals corresponding to
an interaction between the haplotypes and a categorical covariate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplo.interaction(formula, data, SNPs.sel, quantitative = 
  is.quantitative(formula, data), haplo.freq.min = 0.05, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplo.interaction_+3A_formula">formula</code></td>
<td>
<p> a symbolic description of the model to be fitted (a formula object). 
It might have either a continuous variable (quantitative traits) or a 
factor variable (case-control studies) as the response on the left of the <code>~</code> 
operator and a term corresponding to the interaction variable on the right indicated using 
'interaction' function (e.g. <code>~</code>int(var), where var is a factor variable) and it is 
required. Terms with additional covariates on the the right of the ~ operator may be 
added to fit an adjusted model (e.g., <code>~</code>var1+var2+...+varN+int(var)). </p>
</td></tr>
<tr><td><code id="haplo.interaction_+3A_data">data</code></td>
<td>
<p> an object of class 'setupSNP' containing the variables in the model and the SNPs that will be used to estimate the
haplotypes. </p>
</td></tr>
<tr><td><code id="haplo.interaction_+3A_snps.sel">SNPs.sel</code></td>
<td>
<p>a vector indicating the names of SNPs that are used to estimate the haplotypes</p>
</td></tr>
<tr><td><code id="haplo.interaction_+3A_quantitative">quantitative</code></td>
<td>
<p>logical value indicating whether the phenotype (which is on the
left of the operator <code>~</code> in 'formula' argument) is quantitative. The function 
'is.quantitative' returns FALSE when the phenotype
is a variable with two categories (i.e. indicating case-control status). Thus,
it is not a required argument but it may be modified by the user. </p>
</td></tr>
<tr><td><code id="haplo.interaction_+3A_haplo.freq.min">haplo.freq.min</code></td>
<td>
<p>control parameter for haplo.glm included in 'haplo.glm.control'. This parameter corresponds to the minimum 
haplotype frequency for a haplotype to be included in the regression model as its own effect. The
haplotype frequency is based on the EM algorithm that estimates haplotype frequencies independently
of any trait. </p>
</td></tr>
<tr><td><code id="haplo.interaction_+3A_...">...</code></td>
<td>
<p>additional parameters for 'haplo.glm.control'.</p>
</td></tr>                      
</table>


<h3>Details</h3>

<p>The function estimates the haplotypes for the SNPs indicated in the 'SNPs.sel' argument. Then, usign 'haplo.glm' function (from 'haplo.stats'
library) estimates the interaction between these haplotypes and the covariate indicated in the formula by means of 'interaction' function. 
</p>


<h3>Value</h3>

<p>Three different tables are given. The first one corresponds to the full interaction matrix where the ORs (or mean differences if a quantitative 
trait is analyzed)  are expressed with respect to the most frequent haplotype and the first category of the covariate. The other two tables 
show the ORs (or mean differences if a quantitative trait is analyzed) and their 95% confidence intervals for both marginal models. 
P values for interaction are also showed in the output. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not Run
library(SNPassoc)
library(haplo.stats)

data(SNPs)
datSNP&lt;-setupSNP(SNPs,6:40,sep="")
res &lt;- haplo.interaction(log(protein)~int(sex), data=datSNP,
                    SNPs.sel=c("snp100019","snp10001","snp100029"))
res
</code></pre>

<hr>
<h2 id='HapMap'> SNPs from HapMap project</h2><span id='topic+HapMap'></span>

<h3>Description</h3>

<p>Information about 9307 SNPs from the HapMap project
belonging to 22 chromosomes. Information about two different population is available:
European population (CEU) and Yoruba (YRI). The genomic
information (names of SNPs, chromosomes and genetic position) is
also available in a data frame called 'HapMap.SNPs.pos'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HapMap)</code></pre>


<h3>Format</h3>

<p>A data frame with 120 observations on the 9808 variables (SNPs) and one variable
called 'group' indicating the population.
</p>


<h3>Source</h3>

<p>HapMap project (http://www.hapmap.org)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(HapMap)
</code></pre>

<hr>
<h2 id='HapMap.SNPs.pos'> SNPs from HapMap project</h2><span id='topic+HapMap.SNPs.pos'></span>

<h3>Description</h3>

<p>Information about 9307 SNPs from the HapMap project
belonging to 22 chromosomes. Information about two different population is available:
European population (CEU) and Yoruba (YRI). The genomic
information (names of SNPs, chromosomes and genetic position) is
also available in a data frame called 'HapMap.SNPs.pos'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(HapMap.SNPs.pos)</code></pre>


<h3>Format</h3>

<p>A data frame with 120 observations on the 9808 variables (SNPs) and one variable
called 'group' indicating the population.
</p>


<h3>Source</h3>

<p>HapMap project (http://www.hapmap.org)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(HapMap.SNPs.pos)
</code></pre>

<hr>
<h2 id='inheritance'>Collapsing (or recoding) genotypes into different categories (generally two) depending on a given genetic mode of inheritance</h2><span id='topic+inheritance'></span><span id='topic+geneticModel'></span><span id='topic+codominant'></span><span id='topic+dominant'></span><span id='topic+recessive'></span><span id='topic+overdominant'></span><span id='topic+additive'></span>

<h3>Description</h3>

<p>codominant function recodifies a variable having genotypes depending on the allelic frequency
in descending order. <br />
</p>
<p>dominant, recessive, and overdominant functions collapse the three categories of a given SNP 
into two categories as follows: Let 'AA', 'Aa', and 'aa' be the three genotypes. After determining
the most frequent allele (let's suppose that 'A' is the major allele) the functions return a vector
with to categories as follows. dominant: 'AA' and 'Aa-aa'; recessive: 'AA-Aa' and 'aa'; 
overdominant: 'AA-aa' vs 'Aa'. <br />
</p>
<p>additive function creates a numerical variable, 1, 2, 3 corresponding to the three genotypes sorted
out by descending allelic frequency (this model is referred as log-additive).</p>


<h3>Usage</h3>

<pre><code class='language-R'>codominant(o)
dominant(o)
recessive(o)
overdominant(o)
additive(o)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="inheritance_+3A_o">o</code></td>
<td>
<p>categorical covariate having genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A snp object collapsing genotypes into different categories depending on a given genetic mode of inheritance</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
dominant(snp(SNPs$snp10001,sep=""))
overdominant(snp(SNPs$snp10001,sep=""))
</code></pre>

<hr>
<h2 id='int'>
Identify interaction term
</h2><span id='topic+int'></span>

<h3>Description</h3>

<p>This is a special function used for 'haplo.interaction' function.  It
identifies the variable that will interact with the haplotype estimates. 
Using <code>int()</code> in a formula implies that
the interaction term between this variable and haplotypes is included in 'haplo.glm' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>int(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="int_+3A_x">x</code></td>
<td>

<p>A factor variable.
</p>
</td></tr></table>


<h3>Value</h3>

<p><code>x</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+haplo.interaction">haplo.interaction</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(SNPassoc)
library(haplo.stats)

data(SNPs)
datSNP&lt;-setupSNP(SNPs, 6:40, sep = "")
mod &lt;- haplo.interaction(casco~int(sex)+blood.pre, data = datSNP, 
                         SNPs.sel = c("snp10001","snp10004","snp10005"))

</code></pre>

<hr>
<h2 id='interactionPval'> Two-dimensional SNP analysis for association studies</h2><span id='topic+interactionPval'></span><span id='topic+print.SNPinteraction'></span><span id='topic+plot.SNPinteraction'></span>

<h3>Description</h3>

<p>Perform a two-dimensional SNP analysis (interaction) for association studies with
possible allowance for covariate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interactionPval(formula, data, quantitative = 
          is.quantitative(formula, data), model = "codominant")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="interactionPval_+3A_formula">formula</code></td>
<td>
<p>a formula object. It might have either a continuous variable (quantitative traits) or a 
factor variable (case-control study) as the response on the left of the <code>~</code> 
operator and the terms corresponding to the covariates to be adjusted. A crude analysis
is performed indicating <code>~</code>1 </p>
</td></tr>
<tr><td><code id="interactionPval_+3A_data">data</code></td>
<td>
<p> a required object of class 'setupSNP'. </p>
</td></tr>
<tr><td><code id="interactionPval_+3A_quantitative">quantitative</code></td>
<td>
<p> logical value indicating whether the phenotype (those which is in the
left of the operator ~ in 'formula' argument) is quantitative. The function 
'is.quantitative' returns FALSE when the phenotype
is a variable with two categories (i.e. indicating case-control status). Thus,
it is not a required argument but it may be modified by the user.</p>
</td></tr>
<tr><td><code id="interactionPval_+3A_model">model</code></td>
<td>
<p>a character string specifying the type of genetic model (mode of inheritance). This 
indicates how the genotypes should be collapsed. Possible
value are &quot;codominant&quot;, &quot;dominant&quot;, &quot;recessive&quot;, &quot;overdominant&quot; or &quot;log-additive&quot;. The default
is &quot;codominant&quot;. Only the first words are required, e.g &quot;co&quot;, &quot;do&quot;, &quot;re&quot;, &quot;ov&quot;, &quot;log&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'interactionPval' function calculates, for each pair of SNPs (i,j), the likelihood underling the null model L0, 
the likelihood under each of the single-SNP, L(i) and L(j), the likelihood under an additive SNP
model La(i,j), and the likelihood under a full SNP model (including SNP-SNP interaction), Lf(i,j).
</p>
<p>The upper triangle in matrix from this function contains the p values for the interaction (epistasis) 
log-likelihood ratio test, LRT, LRTij = -2 (log Lf(i,j) - log La(i,j))
</p>
<p>The diagonal contains the p values from LRT for the crude effect of each SNP, LRTii = -2 (log L(i) - log L0) 
</p>
<p>The lower triangle contains the p values from LRT comparing the two-SNP additive likelihood to the best
of the single-SNP models, LRTji = -2 (log La(i,j) - log max(L(i),L(j))) 
</p>
<p>In all cases the models including the SNPs are adjusted by the covariates indicated in the 'formula'
argument. This method is used either for quantitative traits and dicotomous variables (case-control studies). 
</p>


<h3>Value</h3>

<p>The 'interactionPval' function returns a matrix of class 'SNPinteraction' containing the p values corresponding to the 
different likelihood ratio tests above describe. 
</p>
<p>Methods defined for 'SNPinteraction' objects are provided for print and plot. The plot method uses 'image' to plot a grid 
of p values.  The upper triangle contains the interaction (epistasis) p values from LRT. The content in the lower triangle 
is the p values from the LRT comparing the additive model with the best single model. The diagonal contains the main effects 
pvalues from LRT. The 'plot.SNPinteraction' function also allows the user to plot the SNPs sorted by genomic position and 
with the information about chromosomes as in the 'plotMissing' function.
</p>


<h3>Note</h3>

<p>  two-dimensional SNP analysis on a dense grid can take a great deal of computer
time and memory.
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setupSNP">setupSNP</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(SNPs)
datSNP&lt;-setupSNP(SNPs,6:40,sep="")

ansCod&lt;-interactionPval(log(protein)~sex,datSNP)
print(ansCod)
plot(ansCod)

</code></pre>

<hr>
<h2 id='intervals'>Print ORs and 95% confidence intervals for an object of class 'haplo.glm'</h2><span id='topic+intervals'></span><span id='topic+intervals.haplo.glm'></span><span id='topic+print.intervals'></span><span id='topic+summary.haplo.glm'></span>

<h3>Description</h3>

<p>Print ORs and confidence intervals for an object of class 'haplo.glm'</p>


<h3>Usage</h3>

<pre><code class='language-R'>intervals(o, level=.95, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intervals_+3A_o">o</code></td>
<td>
<p>object of class 'haplo.glm'</p>
</td></tr>
<tr><td><code id="intervals_+3A_level">level</code></td>
<td>
<p>significance level. Default is 95 percent</p>
</td></tr>
<tr><td><code id="intervals_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p> intervals object with ORs and 95% confidence intervals for an object of class 'haplo.glm'</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not Run
library(SNPassoc)
library(haplo.stats)

data(asthma, package = "SNPassoc")

asthma.s &lt;- setupSNP(data=asthma, colSNPs=7:ncol(asthma), sep="")
trait &lt;- asthma.s$casecontrol
snpsH &lt;- c("rs714588", "rs1023555",  "rs898070")
genoH &lt;- make.geno(asthma.s, snpsH)

mod &lt;- haplo.stats:: haplo.glm( trait ~ genoH,           
                                family="binomial", 
                                locus.label=snpsH,
                                allele.lev=attributes(genoH)$unique.alleles,
                                control = haplo.glm.control(haplo.freq.min=0.05))   
intervals(mod)
summary(mod)

</code></pre>

<hr>
<h2 id='is.Monomorphic'>Check whether a SNP is Monomorphic</h2><span id='topic+is.Monomorphic'></span>

<h3>Description</h3>

<p>This function verifies when a SNP is Monomorphic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.Monomorphic(x) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.Monomorphic_+3A_x">x</code></td>
<td>
<p>any R object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value TRUE if the SNP is Monomorphic, otherwise a FALSE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(SNPs)
is.Monomorphic(SNPs$snp10001)
is.Monomorphic(SNPs$snp100020)
apply(SNPs[,20:30],2,is.Monomorphic)
</code></pre>

<hr>
<h2 id='LD'>max-statistic for a 2x3 table</h2><span id='topic+LD'></span><span id='topic+LD.snp'></span><span id='topic+LD.setupSNP'></span><span id='topic+LDplot'></span><span id='topic+LDtable'></span>

<h3>Description</h3>

<p>Compute pairwise linkage disequilibrium between genetic markers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LD(g1, ...)

## S3 method for class 'snp'
LD(g1, g2, ...)

## S3 method for class 'setupSNP'
LD(g1, SNPs, ...)


LDplot(x, digits = 3, marker, distance, which = c("D", "D'",
    "r", "X^2", "P-value", "n", " "), ...)

LDtable(x, colorcut = c(0, 0.01, 0.025, 0.05, 0.1, 1), 
       colors = heat.colors(length(colorcut)),
       textcol = "black", digits = 3, show.all = FALSE, 
       which = c("D", "D'", "r", "X^2", "P-value", "n"), 
       colorize = "P-value", cex, ...)


</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LD_+3A_g1">g1</code></td>
<td>
<p>genotype object or dataframe containing genotype objects</p>
</td></tr>
<tr><td><code id="LD_+3A_g2">g2</code></td>
<td>
<p>genotype object (ignored if g1 is a dataframe)</p>
</td></tr>  
<tr><td><code id="LD_+3A_snps">SNPs</code></td>
<td>
<p>columns containing SNPs</p>
</td></tr>  
<tr><td><code id="LD_+3A_x">x</code></td>
<td>
<p>LD or LD.data.frame object</p>
</td></tr> 
<tr><td><code id="LD_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to display</p>
</td></tr>
<tr><td><code id="LD_+3A_which">which</code></td>
<td>
<p>Name(s) of LD information items to be displayed</p>
</td></tr>
<tr><td><code id="LD_+3A_colorcut">colorcut</code></td>
<td>
<p>P-value cutoffs points for colorizing LDtable</p>
</td></tr>
<tr><td><code id="LD_+3A_colors">colors</code></td>
<td>
<p>Colors for each P-value cutoff given in 'colorcut' for LDtable</p>
</td></tr>
<tr><td><code id="LD_+3A_textcol">textcol</code></td>
<td>
<p>Color for text labels for LDtable</p>
</td></tr>
<tr><td><code id="LD_+3A_marker">marker</code></td>
<td>
<p>Marker used as 'comparator' on LDplot.  If omitted separate
lines for each marker will be displayed</p>
</td></tr>
<tr><td><code id="LD_+3A_distance">distance</code></td>
<td>
<p>Marker location, used for locating of markers on LDplot.</p>
</td></tr>
<tr><td><code id="LD_+3A_show.all">show.all</code></td>
<td>
<p>If TRUE, show all rows/columns of matrix. Otherwise omit
completely blank rows/columns.</p>
</td></tr>
<tr><td><code id="LD_+3A_colorize">colorize</code></td>
<td>
<p>LD parameter used for determining table cell colors</p>
</td></tr>
<tr><td><code id="LD_+3A_cex">cex</code></td>
<td>
<p>Scaling factor for table text. If absent, text will be scaled
to fit within the table cells.</p>
</td></tr>
<tr><td><code id="LD_+3A_...">...</code></td>
<td>
<p>Optional arguments ('plot.LD.data.frame' passes these to
'LDtable' and 'LDplot').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>functions adapted from LD, LDtable and LDplot in package genetics by Gregory Warnes et al.  (warnes@bst.rochester.edu)
</p>


<h3>References</h3>

<p>genetics R package by Gregory Warnes et al.  (warnes@bst.rochester.edu)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setupSNP">setupSNP</a></code>   <code><a href="#topic+snp">snp</a></code>
</p>

<hr>
<h2 id='make.geno'>Create a group of locus objects from some SNPs, assign to 'model.matrix' class.
</h2><span id='topic+make.geno'></span>

<h3>Description</h3>

<p>This function prepares the CRITICAL element corresponding to matrix of genotypes necessary to be included in 'haplo.glm' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.geno(data, SNPs.sel)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make.geno_+3A_data">data</code></td>
<td>
<p>an object of class 'setupSNP' containing the the SNPs that will be used to estimate the haplotypes.</p>
</td></tr>
<tr><td><code id="make.geno_+3A_snps.sel">SNPs.sel</code></td>
<td>
<p> a vector indicating the names of SNPs that are used to estimate the haplotypes  </p>
</td></tr>
</table>


<h3>Value</h3>

<p>the same as 'setupGeno' function, from 'haplo.stats' library, returns
</p>


<h3>See Also</h3>

<p><code><a href="#topic+snp">snp</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:
data(SNPs)
# first, we create an object of class 'setupSNP'
datSNP&lt;-setupSNP(SNPs,6:40,sep="")
geno&lt;-make.geno(datSNP,c("snp10001","snp10002","snp10003"))
## End(Not run)


</code></pre>

<hr>
<h2 id='maxstat'>max-statistic for a 2x3 table</h2><span id='topic+maxstat'></span><span id='topic+maxstat.default'></span><span id='topic+maxstat.table'></span><span id='topic+maxstat.setupSNP'></span><span id='topic+maxstat.matrix'></span><span id='topic+print.maxstat'></span>

<h3>Description</h3>

<p>Computes the asymptotic p-value for max-statistic for a 2x3 table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxstat(x, ...)

## Default S3 method:
maxstat(x, y, ...)

## S3 method for class 'table'
maxstat(x, ...)

## S3 method for class 'setupSNP'
maxstat(x, y, colSNPs=attr(x,"colSNPs"), ...)

## S3 method for class 'matrix'
maxstat(x, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maxstat_+3A_x">x</code></td>
<td>
<p>a numeric matrix with 2 rows (cases/controls) and 3 colums (genotypes) or a vector with case/control status
or an object of class 'setupSNP'.</p>
</td></tr>
<tr><td><code id="maxstat_+3A_y">y</code></td>
<td>
<p>an optional numeric vector containing the information for a given SNP. In this
case 'x' argument must contain a vector indicarting case/control status. If 'x' argument is an object of 
class 'setupSNP' this argument migth be the name of the variable containing case/control information.</p>
</td></tr>
<tr><td><code id="maxstat_+3A_colsnps">colSNPs</code></td>
<td>
<p>a vector indicating which columns contain those SNPs to compute max-statistic. By default
max-statistic is computed for those SNPs specified when the object of class 'setupSNP' was
created.</p>
</td></tr>
<tr><td><code id="maxstat_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with the chi-square statistic for dominant, recessive, log-additive and max-statistic and 
its asymptotic p-value. 
</p>


<h3>References</h3>

<p>Gonzalez JR, Carrasco JL, Dudbridge F, Armengol L, Estivill X, Moreno V. Maximizing association statistics over 
genetic models (2007). Submitted   
</p>
<p>Sladek R, Rocheleau G, Rung J et al. A genome-wide association study identifies novel risk loci for type 2 diabetes (2007). Nature 445, 881-885 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setupSNP">setupSNP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# example from Sladek et al. (2007) for the SNP rs1111875 
 tt&lt;-matrix(c(77,298,310,122,316,231),nrow=2,ncol=3,byrow=TRUE)
 maxstat(tt)

 data(SNPs)
 maxstat(SNPs$casco,SNPs$snp10001) 
 myDat&lt;-setupSNP(SNPs,6:40,sep="")
 maxstat(myDat,casco)

 

</code></pre>

<hr>
<h2 id='odds'>Extract odds ratios, 95% CI and pvalues</h2><span id='topic+odds'></span>

<h3>Description</h3>

<p>Extract odds ratios, 95
</p>


<h3>Usage</h3>

<pre><code class='language-R'> odds(x, model=c("log-additive", "dominant", "recessive", "overdominant", "codominant"), 
      sorted=c("no","p-value","or"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="odds_+3A_x">x</code></td>
<td>
<p>an object of class 'WGassociation' output of WGassociation</p>
</td></tr>
<tr><td><code id="odds_+3A_model">model</code></td>
<td>
<p>model to be extracted. Only first one is used. The first letter is enough, low or upper case.</p>
</td></tr>
<tr><td><code id="odds_+3A_sorted">sorted</code></td>
<td>
<p>Sort the output by P value or OR.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with OR 95% CI (lower, upper) and P value for the selected model. For codominant model, the OR and 95%CI are given for heterozygous and homozigous.
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(SNPs)
 datSNP&lt;-setupSNP(SNPs,6:40,sep="")
 ans&lt;-WGassociation(casco~1,data=datSNP,model="all")
 odds(ans)
</code></pre>

<hr>
<h2 id='permTest'>Permutation test analysis</h2><span id='topic+permTest'></span><span id='topic+print.permTest'></span><span id='topic+plot.permTest'></span>

<h3>Description</h3>

<p>This function extract the p values for permutation approach performed using scanWGassociation function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>   permTest(x, method="minimum", K)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="permTest_+3A_x">x</code></td>
<td>
<p>a required object of class 'WGassociation' with the attribute 'permTest'. See details</p>
</td></tr>
<tr><td><code id="permTest_+3A_method">method</code></td>
<td>
<p>statistic used in the permutation test. The default is 'minimum' but 'rtp' (rank truncated product) is
also available.</p>
</td></tr>
<tr><td><code id="permTest_+3A_k">K</code></td>
<td>
<p>number of the K most significant p values from the total number of test performed (e.g number of SNPs) used
to compute the rank truncated product. This argument is only required when method='rtp'. See references </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function extract the p values from an object of class 'WGassociation'. This object migth be obtained using 
the funcion called 'scanWGassociation' indicating the number of permutations in the argument 'nperm'. 
</p>


<h3>Value</h3>

<p>An object of class 'permTest'. 
</p>
<p>'print' returns a summary indicating the number of SNPs analyzed, the number of valid SNPs (those non-Monomorphic and 
that pass the calling rate), the p value after Bonferroni correction, and the p values based on permutation approach. 
One of them is based on considering the empirical percentil for the minimum p values, and the another one on assuming 
that the minimum p values follow a beta distribution.
</p>
<p>'plot' produces a plot of the empirical distribution for the minimum p values (histogram) and the expected distribution
assuming a beta distribution. The corrected p value is also showed in the plot. 
</p>
<p>See examples for further illustration about all previous issues. 
</p>


<h3>References</h3>

<p>Dudbridge F, Gusnanto A and Koeleman BPC. Detecting multiple associations in genome-wide studies.  
Human Genomics, 2006;2:310-317.
</p>
<p>Dudbridge F and Koeleman BPC. Efficient computation of significance levels for multiple associations in large studies of 
correlated data, including genomewide association studies. Am J Hum Genet, 2004;75:424-435.
</p>
<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+scanWGassociation">scanWGassociation</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(SNPassoc)

data(asthma, package = "SNPassoc")
asthma.s &lt;- setupSNP(data=asthma, colSNPs=7:ncol(asthma), sep="")

ans &lt;- WGassociation(casecontrol, data=asthma.s)

</code></pre>

<hr>
<h2 id='plot.WGassociation'>Function to plot -log p values from an object of class 'WGassociation'</h2><span id='topic+plot.WGassociation'></span>

<h3>Description</h3>

<p>Function to plot -log p values from an object of class 'WGassociation'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'WGassociation'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.WGassociation_+3A_x">x</code></td>
<td>
<p> an object of class 'WGassociation'</p>
</td></tr>
<tr><td><code id="plot.WGassociation_+3A_...">...</code></td>
<td>
<p>other graphical parameters </p>
</td></tr>
</table>


<h3>Details</h3>

<p>A panel with different plots (one for each mode of inheritance) are plotted. Each of them represents 
the -log(p value) for each SNP. Two horizontal lines are also plotted. One one them indicates the nominal 
statistical significance level whereas the other one indicates the statistical 
significance level after Bonferroni correction.
</p>


<h3>Value</h3>

<p>No return value, just the plot
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+association">association</a></code> <code><a href="#topic+setupSNP">setupSNP</a></code> <code><a href="#topic+WGassociation">WGassociation</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(SNPassoc)

data(asthma, package = "SNPassoc")
asthma.s &lt;- setupSNP(data=asthma, colSNPs=7:ncol(asthma), sep="")

ans &lt;- WGassociation(casecontrol, data=asthma.s)

plot(ans)

</code></pre>

<hr>
<h2 id='plotMissing'> Plot of missing genotypes</h2><span id='topic+plotMissing'></span>

<h3>Description</h3>

<p>Plot a grid showing which genotypes are missing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMissing(x, print.labels.SNPs = TRUE, 
        main = "Genotype missing data", ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMissing_+3A_x">x</code></td>
<td>
<p>an object of class 'setupSNP'</p>
</td></tr>
<tr><td><code id="plotMissing_+3A_print.labels.snps">print.labels.SNPs</code></td>
<td>
<p>should labels of SNPs be printed?</p>
</td></tr>
<tr><td><code id="plotMissing_+3A_main">main</code></td>
<td>
<p>title to place on plot</p>
</td></tr>
<tr><td><code id="plotMissing_+3A_...">...</code></td>
<td>
<p>extra arguments of 'image' function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses 'image' function to plot a grid with black pixels where the genotypes
are missing.  
</p>


<h3>Value</h3>

<p>No return value, just the plot
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+setupSNP">setupSNP</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> data(SNPs)
 data(SNPs.info.pos) 
 ans&lt;-setupSNP(SNPs,colSNPs=6:40,sep="")
 plotMissing(ans)
 
 # The same plot with the SNPs sorted by genomic position and 
 # showing the information about chromosomes

 ans&lt;-setupSNP(SNPs,colSNPs=6:40,sort=TRUE,SNPs.info.pos,sep="") 
 plotMissing(ans)
</code></pre>

<hr>
<h2 id='qqpval'>Functions for inspecting population substructure</h2><span id='topic+qqpval'></span>

<h3>Description</h3>

<p>This function plots ranked observed p values against the corresponding expected p values in -log scale. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqpval(p, pch=16, col=4, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qqpval_+3A_p">p</code></td>
<td>
<p>a vector of p values</p>
</td></tr>
<tr><td><code id="qqpval_+3A_pch">pch</code></td>
<td>
<p>symbol to use for points</p>
</td></tr>
<tr><td><code id="qqpval_+3A_col">col</code></td>
<td>
<p>color for points</p>
</td></tr>
<tr><td><code id="qqpval_+3A_...">...</code></td>
<td>
<p>other plot arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, just the plot
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+GenomicControl">GenomicControl</a></code>, <code><a href="#topic+WGassociation">WGassociation</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
datSNP&lt;-setupSNP(SNPs,6:40,sep="")
res&lt;-WGassociation(casco,datSNP,model=c("do","re","log-add"))

# observed vs expected p values for recessive model
qqpval(recessive(res))

</code></pre>

<hr>
<h2 id='related'>Get related samples</h2><span id='topic+related'></span>

<h3>Description</h3>

<p>Get related samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>related(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="related_+3A_x">x</code></td>
<td>
<p>An object obtained from SNPrelate package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with related individuals.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(SNPassoc)
  data(SNPs)

</code></pre>

<hr>
<h2 id='resHapMap'> SNPs from HapMap project</h2><span id='topic+resHapMap'></span>

<h3>Description</h3>

<p>Information about 9307 SNPs from the HapMap project
belonging to 22 chromosomes. Information about two different population is available:
European population (CEU) and Yoruba (YRI). The genomic
information (names of SNPs, chromosomes and genetic position) is
also available in a data frame called 'HapMap.SNPs.pos'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(resHapMap)</code></pre>


<h3>Format</h3>

<p>A data frame with 120 observations on the 9808 variables (SNPs) and one variable
called 'group' indicating the population.
</p>


<h3>Source</h3>

<p>HapMap project (http://www.hapmap.org)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(resHapMap)
</code></pre>

<hr>
<h2 id='scanWGassociation'>Whole genome association analysis</h2><span id='topic+scanWGassociation'></span>

<h3>Description</h3>

<p>This function is obsolete due to some problems with gfotran compiler. 
Use 'WGassociation' function instead or send an e-mail to the maintainer 
for receiving a version including this function
</p>

<hr>
<h2 id='setupSNP'>Convert columns in a dataframe to class 'snp'</h2><span id='topic+setupSNP'></span><span id='topic+summary.setupSNP'></span><span id='topic+plot.setupSNP'></span><span id='topic++5B.setupSNP'></span><span id='topic++5B+5B+3C-.setupSNP'></span><span id='topic++5B+3C-.setupSNP'></span><span id='topic++24+3C-.setupSNP'></span><span id='topic+labels.setupSNP'></span>

<h3>Description</h3>

<p><code>setupSNP</code> Convert columns in a dataframe to class 'snp'
</p>
<p><code>summary.setupSNP</code> gives a summary for an object of class 'setupSNP' including 
allele names, major allele frequencie, an exact thest of Hardy-Weinberg 
equilibrium and percentage of missing genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setupSNP(data, colSNPs, sort = FALSE, info, sep = "/", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setupSNP_+3A_data">data</code></td>
<td>
<p>dataframe containing columns with the SNPs to be converted</p>
</td></tr>
<tr><td><code id="setupSNP_+3A_colsnps">colSNPs</code></td>
<td>
<p>Vector specifying which columns contain SNPs data</p>
</td></tr>
<tr><td><code id="setupSNP_+3A_sort">sort</code></td>
<td>
<p>should SNPs be sorted. Default is FALSE</p>
</td></tr>
<tr><td><code id="setupSNP_+3A_info">info</code></td>
<td>
<p>if sort is TRUE a dataframe containing information 
about the SNPs regarding their genomic position and the gene where 
they are located</p>
</td></tr>
<tr><td><code id="setupSNP_+3A_sep">sep</code></td>
<td>
<p>character separator used to divide alleles in the genotypes</p>
</td></tr>
<tr><td><code id="setupSNP_+3A_...">...</code></td>
<td>
<p>optional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of class 'setupSNP' containing converted SNP variables. 
All other variables will be unchanged.
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+snp">snp</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
 data(SNPs)
 myDat&lt;-setupSNP(SNPs,6:40,sep="")


#sorted SNPs and having genomic information
 data(SNPs.info.pos)
 myDat.o&lt;-setupSNP(SNPs,6:40,sep="",sort=TRUE, info=SNPs.info.pos)

# summary
 summary(myDat.o)

# plot one SNP
  plot(myDat,which=2)

</code></pre>

<hr>
<h2 id='snp'>SNP object</h2><span id='topic+snp'></span><span id='topic+is.snp'></span><span id='topic+as.snp'></span><span id='topic+reorder.snp'></span><span id='topic+summary.snp'></span><span id='topic+plot.snp'></span><span id='topic+dominant.snp'></span><span id='topic+codominant.snp'></span><span id='topic+recessive.snp'></span><span id='topic+additive.snp'></span><span id='topic+print.snp'></span><span id='topic++5B.snp'></span><span id='topic+print.summary.snp'></span>

<h3>Description</h3>

<p><code>snp</code> creates an snp object
</p>
<p><code>is</code> returns <code>TRUE</code> if <code>x</code> is of class 'snp'
</p>
<p><code>as</code> attempts to coerce its argument into an object of class 'snp'
</p>
<p><code>reorder</code> change the reference genotype
</p>
<p><code>summary</code> gives a summary for an object of class 'snp' including genotype
and allele frequencies and an exact thest of Hardy-Weinberg 
equilibrium 
</p>
<p><code>plot</code>    gives a summary for an object of class 'snp' including genotype
and allele frequencies and an exact thest of Hardy-Weinberg 
equilibrium in a plot. Barplot or pie are allowed
</p>
<p><code>[.snp</code>   is a copy of [.factor modified to preserve all attributes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
  snp(x, sep = "/", name.genotypes, reorder="common", 
    remove.spaces = TRUE, allow.partial.missing = FALSE) 

  is.snp(x)
 
  as.snp(x, ...)

  ## S3 method for class 'snp'
additive(o)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snp_+3A_x">x</code></td>
<td>
<p>either an object of class 'snp' or an object to be converted to class 'snp'</p>
</td></tr>
<tr><td><code id="snp_+3A_sep">sep</code></td>
<td>
<p>character separator used to divide alleles when <code>x</code> is a vector of strings where 
each string holds both alleles. The default is &quot;/&quot;. See below for details.</p>
</td></tr>         
<tr><td><code id="snp_+3A_name.genotypes">name.genotypes</code></td>
<td>
<p>the codes for the genotypes. This argument may be useful when genotypes 
are coded using three different codes (e.g., 0,1,2 or hom1, het, hom2)</p>
</td></tr>
<tr><td><code id="snp_+3A_reorder">reorder</code></td>
<td>
<p>how should genotypes within an individual be reordered. Possible values are
'common' or 'minor'. The default is 
<code>reorder="common"</code>. In that case, alleles are sorted within each individual by
common homozygous.</p>
</td></tr>
<tr><td><code id="snp_+3A_remove.spaces">remove.spaces</code></td>
<td>
<p>logical indicating whether spaces and tabs will be
removed from the genotypes before processing</p>
</td></tr>
<tr><td><code id="snp_+3A_allow.partial.missing">allow.partial.missing</code></td>
<td>
<p>logical indicating whether one allele is
permitted to be missing.  When set to 'FALSE' both alleles
are set to 'NA' when either is missing.</p>
</td></tr>
<tr><td><code id="snp_+3A_o">o</code></td>
<td>
<p>an object of class 'snp' to be coded as a linear covariate: 0,1,2</p>
</td></tr>
<tr><td><code id="snp_+3A_...">...</code></td>
<td>
<p>optional arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>SNP objects hold information on which gene or marker alleles were
observed for different individuals.  For each individual, two alleles
are recorded.
</p>
<p>The snp class considers the stored alleles to be unordered , i.e., &quot;C/T&quot;
is equivalent to &quot;T/C&quot;. It assumes that the order of the alleles is not important.
</p>
<p>When <code>snp</code> is called, <code>x</code> is a character vector, and it is
assumed that each element encodes both alleles. In this case, if
<code>sep</code> is a character string, <code>x</code> is assumed to be coded
as &quot;Allele1&lt;sep&gt;Allele2&quot;.  If <code>sep</code> is a numeric value, it is
assumed that character locations <code>1:sep</code> contain allele 1 and
that remaining locations contain allele 2.
</p>
<p><code>additive.snp</code> recodes the SNPs for being analyzed as a linear covariate (codes 0,1,2) 
</p>


<h3>Value</h3>

<p>The snp class extends &quot;factor&quot; where the levels is a character vector of possible 
genotype values stored coded by <code>paste( allele1, "", allele2, sep="/")</code>
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+association">association</a></code>  
</p>


<h3>Examples</h3>

<pre><code class='language-R'># some examples of snp data in different formats

dat1  &lt;- c("21", "21", "11", "22", "21",
                    "22", "22", "11", "11", NA)
ans1  &lt;- snp(dat1,sep="")
ans1

dat2 &lt;- c("A/A","A/G","G/G","A/G","G/G",
                    "A/A","A/A","G/G",NA)
ans2  &lt;- snp(dat2,sep="/")
ans2

dat3 &lt;- c("C-C","C-T","C-C","T-T","C-C",
                    "C-C","C-C","C-C","T-T",NA)
ans3 &lt;- snp(dat3,sep="-")
ans3


dat4 &lt;- c("het","het","het","hom1","hom2",
                    "het","het","hom1","hom1",NA)
ans4 &lt;- snp(dat4,name.genotypes=c("hom1","het","hom2"))
ans4


# summary 
summary(ans3)

# plots

plot(ans3)
plot(ans3,type=pie)
plot(ans3,type=pie,label="SNP 10045")

</code></pre>

<hr>
<h2 id='SNPassoc-internal'>Internal SNPstat functions</h2><span id='topic+SNPassoc-internal'></span><span id='topic+association.fit'></span><span id='topic+extractPval'></span><span id='topic+extractPval.i'></span><span id='topic+SNPHWE'></span><span id='topic+GenotypeRate'></span><span id='topic+intervals.dif'></span><span id='topic+intervals.or'></span><span id='topic+is.quantitative'></span><span id='topic+haplo.inter.fit'></span><span id='topic+table.corner'></span><span id='topic+table.interaction'></span><span id='topic+crea.lab'></span><span id='topic+togeno'></span><span id='topic+orderChromosome'></span><span id='topic+expandsetupSNP'></span><span id='topic+pvalTest'></span><span id='topic+modelTest'></span><span id='topic+assoc'></span><span id='topic+codominant.default'></span><span id='topic+dominant.default'></span><span id='topic+recessive.default'></span><span id='topic+overdominant.default'></span><span id='topic+additive.default'></span><span id='topic+trim'></span><span id='topic+interleave'></span>

<h3>Description</h3>

<p>Internal SNPassoc functions</p>


<h3>Usage</h3>

<pre><code class='language-R'>association.fit(var, dep, adj, quantitative, type, level, 
       nIndiv, genotypingRate = 0, ...) 
extractPval(x)
extractPval.i(i,x,pos,models)
SNPHWE(x)
GenotypeRate(x)
haplo.inter.fit(geno, var2, dep, adj = NULL, fam, 
          haplo.freq.min, ...)
crea.lab(x,pos.ini,cex,dist)
orderChromosome(x)
togeno(f,sep=sep,lab=lab)
expandsetupSNP(o)
pvalTest(dataX,Y,quantitative,type,genotypingRate)
modelTest(X,Y,quantitative,type,genotypingRate)
assoc(y,x,test="lrt",quantitative)
trim(s)
interleave(..., append.source=TRUE, sep=": ", drop=FALSE)
## Default S3 method:
codominant(o)
## Default S3 method:
dominant(o)
## Default S3 method:
recessive(o)
## Default S3 method:
overdominant(o)
## Default S3 method:
additive(o)
</code></pre>


<h3>Details</h3>

<p>These are not to be called by the user</p>


<h3>Value</h3>

<p>No return value, internal calls </p>

<hr>
<h2 id='SNPs'>SNPs in a case-control study</h2><span id='topic+SNPs'></span>

<h3>Description</h3>

<p>SNPs data.frame contains selected SNPs and other clinical covariates for cases and controls 
in a case-control study
</p>
<p>SNPs.info.pos data.frame contains the names of the SNPs included in the data set 'SNPs' including
their chromosome and their genomic position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SNPs)</code></pre>


<h3>Format</h3>

<p>'SNPs' data.frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
	id </td><td style="text-align: left;"> identifier of each subject </td>
</tr>
<tr>
 <td style="text-align: left;">
	casco </td><td style="text-align: left;"> case or control status: 0-control, 1-case </td>
</tr>
<tr>
 <td style="text-align: left;">
	sex </td><td style="text-align: left;"> gender: Male and Female </td>
</tr>
<tr>
 <td style="text-align: left;">
	blood.pre </td><td style="text-align: left;"> arterial blood presure </td>
</tr>
<tr>
 <td style="text-align: left;">
	protein </td><td style="text-align: left;"> protein levels </td>
</tr>
<tr>
 <td style="text-align: left;">
	snp10001 </td><td style="text-align: left;"> SNP 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
	snp10002 </td><td style="text-align: left;"> SNP 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
	  ...        </td><td style="text-align: left;">  ...      </td>
</tr>
<tr>
 <td style="text-align: left;">
	snp100036 </td><td style="text-align: left;"> SNP 36 </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>



<h3>Source</h3>

<p>Data obtained from our department. The reference and details will be supplied after being published.
</p>

<hr>
<h2 id='SNPs.info.pos'>SNPs in a case-control study</h2><span id='topic+SNPs.info.pos'></span>

<h3>Description</h3>

<p>SNPs data.frame contains selected SNPs and other clinical covariates for cases and controls 
in a case-control study
</p>
<p>SNPs.info.pos data.frame contains the names of the SNPs included in the data set 'SNPs' including
their chromosome and their genomic position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SNPs.info.pos)</code></pre>


<h3>Format</h3>

<p>'SNPs.info.pos' data.frame contains the following columns:
A data frame with 35 observations on the following 3 variables.
</p>

<dl>
<dt><code>snp</code></dt><dd><p>name of SNP</p>
</dd>
<dt><code>chr</code></dt><dd><p>name of chromosome</p>
</dd>
<dt><code>pos</code></dt><dd><p>genomic position</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data obtained from our department. The reference and details will be supplied after being published.
</p>

<hr>
<h2 id='sortSNPs'>Sort a vector of SNPs by genomic position</h2><span id='topic+sortSNPs'></span>

<h3>Description</h3>

<p>This function sorts a vector with the position of SNPs in a data frame
using another data frame which contains information about SNPs, their 
chromosome, and their genomic position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sortSNPs(data, colSNPs, info)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sortSNPs_+3A_data">data</code></td>
<td>
<p>a required data frame with the SNPs</p>
</td></tr>
<tr><td><code id="sortSNPs_+3A_colsnps">colSNPs</code></td>
<td>
<p>a required vector indicating which columns of 'data' contains genotype data</p>
</td></tr>
<tr><td><code id="sortSNPs_+3A_info">info</code></td>
<td>
<p>a required data frame with genomic information for the SNPs (chromosome and position).
The first column must have the SNPs, the second one the chromosome and the third one the genomic 
position. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>First of all, the function obtains a vector with the SNPs sorted using the data frame with the genomic 
positions (see 'dataSNPs.pos' argument).
Then, the columns which indicate where the information about the 
genotypes is in our data frame, are sorted using this vector. 
</p>
<p>This information is useful when <code><a href="#topic+WGassociation">WGassociation</a></code> function is called since it 
allow the user to summaryze the results with the SNPs sorted by genomic position
</p>


<h3>Value</h3>

<p>a vector indicating the colums where the SNPs are recorded in our data frame ('data' argument), sorted
using the genomic positions listed in 'dataSNPs.pos' argument)  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# data(SNPs)
# data(SNPs.info.pos)
# colSNPs.order&lt;-sortSNPs(SNPs,c(6:40),SNPs.info.pos)
#
</code></pre>

<hr>
<h2 id='Table.mean.se'>Descriptive sample size, mean, and standard error</h2><span id='topic+Table.mean.se'></span>

<h3>Description</h3>

<p>This function computes sample size, mean and standard error of a quantitative trait for
each genotype (or combination of genotypes)  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Table.mean.se(var, dep, subset = !is.na(var))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Table.mean.se_+3A_var">var</code></td>
<td>
<p>quantitative trait</p>
</td></tr>
<tr><td><code id="Table.mean.se_+3A_dep">dep</code></td>
<td>
<p>variable with genotypes or any combination of them</p>
</td></tr>
<tr><td><code id="Table.mean.se_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to be used in the descriptive analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>tp</code></td>
<td>
<p>A matrix giving sample size (n), median (me) and standard error (se) for each genotype</p>
</td></tr>
</table>


<h3>See Also</h3>

  <p><code><a href="#topic+Table.N.Per">Table.N.Per</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
# sample size, mean age and standard error for each genotype
Table.mean.se(SNPs$snp10001,SNPs$protein)

# The same table for a subset (males)
Table.mean.se(SNPs$snp10001,SNPs$protein,SNPs$sex=="Male")

# The same table assuming a dominant model
Table.mean.se(dominant(snp(SNPs$snp10001,sep="")),SNPs$protein,SNPs$sex=="Male")


</code></pre>

<hr>
<h2 id='Table.N.Per'> Descriptive sample size and percentage</h2><span id='topic+Table.N.Per'></span>

<h3>Description</h3>

<p>This function computes sample size and percentage for each category of a categorical trait (e.g.
case-control status) for each genotype (or combination of genotypes).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Table.N.Per(var, dep, subset = !is.na(var))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Table.N.Per_+3A_var">var</code></td>
<td>
<p>categorical trait.</p>
</td></tr>
<tr><td><code id="Table.N.Per_+3A_dep">dep</code></td>
<td>
<p>variable with genotypes or any combination of them</p>
</td></tr>
<tr><td><code id="Table.N.Per_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to be used in the descriptive analysis. </p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>tp</code></td>
<td>
<p>A matrix giving sample size (n),and the percentage (%) for each level of the categorical trait
for each genotype</p>
</td></tr>
</table>


<h3>See Also</h3>

 <p><code><a href="#topic+Table.mean.se">Table.mean.se</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
#sample size and percentage of cases and controls for each genotype 
Table.N.Per(SNPs$snp10001,SNPs$casco)

# The same table for a subset (males)
Table.N.Per(SNPs$snp10001,SNPs$casco,SNPs$sex=="Male")

# The same table assuming a dominant model
Table.N.Per(dominant(snp(SNPs$snp10001,sep="")),SNPs$casco,SNPs$sex=="Male")


</code></pre>

<hr>
<h2 id='tableHWE'>Test for Hardy-Weinberg Equilibrium</h2><span id='topic+tableHWE'></span><span id='topic+print.tableHWE'></span>

<h3>Description</h3>

<p>Test the null hypothesis that Hardy-Weinberg equilibrium holds in cases, 
controls and both populations.
</p>
<p><code>print</code> print the information. Number of digits, and significance
level can be changed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableHWE(x, strata, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tableHWE_+3A_x">x</code></td>
<td>
<p>an object of class 'setupSNP'</p>
</td></tr>
<tr><td><code id="tableHWE_+3A_strata">strata</code></td>
<td>
<p>a factor variable for a stratified analysis</p>
</td></tr>
<tr><td><code id="tableHWE_+3A_...">...</code></td>
<td>
<p>optional arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the HWE test for those variables of class 'snp' in the object x of 
class 'setupSNP'
</p>


<h3>Value</h3>

<p>A matrix with p values for Hardy-Weinberg Equilibrium 
</p>


<h3>Author(s)</h3>

<p>This function is based on an R function which computes an exact SNP test of Hardy-Weinberg Equilibrium 
written by Wigginton JE, Cutler DJ and Abecasis GR available at <a href="http://csg.sph.umich.edu/abecasis/Exact/r_instruct.html">http://csg.sph.umich.edu/abecasis/Exact/r_instruct.html</a></p>


<h3>References</h3>

<p>Wigginton JE, Cutler DJ and Abecasis GR (2005). A note on exact tests of Hardy-Weinberg equilibrium. Am J Hum Genet 76:887-93 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+setupSNP">setupSNP</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
ans&lt;-setupSNP(SNPs,6:40,sep="")
res&lt;-tableHWE(ans)
print(res)
#change the significance level showed in the flag column
print(res,sig=0.001)


#stratified analysis
res&lt;-tableHWE(ans,ans$sex)
print(res)


</code></pre>

<hr>
<h2 id='WGassociation'>Whole genome association analysis</h2><span id='topic+WGassociation'></span><span id='topic+WGstats'></span><span id='topic+print.WGassociation'></span><span id='topic+summary.WGassociation'></span><span id='topic+labels.WGassociation'></span><span id='topic+pvalues'></span><span id='topic+pvalues.WGassociation'></span><span id='topic+codominant.WGassociation'></span><span id='topic+dominant.WGassociation'></span><span id='topic+recessive.WGassociation'></span><span id='topic+overdominant.WGassociation'></span><span id='topic+additive.WGassociation'></span><span id='topic++5B.WGassociation'></span>

<h3>Description</h3>

<p>This function carries out a whole genome association analysis between the SNPs and
a dependent variable (phenotype) under five different genetic models (inheritance patterns): 
codominant, dominant, recessive, overdominant and log-additive. The phenotype may be quantitative 
or categorical. In the second case (e.g. case-control studies) this variable must be of class 'factor' 
with two levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>   WGassociation(formula, data, model = c("all"), 
                 quantitative = is.quantitative(formula, data),
                 genotypingRate = 80, level = 0.95, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WGassociation_+3A_formula">formula</code></td>
<td>
<p>either a symbolic description of the model to be fited (a formula object) without the SNP
or the name of response variable in the case of fitting single models (e.g. unadjusted models).
It might have either a continuous variable (quantitative traits) or a 
factor variable (case-control studies) as the response on the left of the <code>~</code> 
operator and terms with additional covariates on the right of the <code>~</code> operator may be 
added to fit an adjusted model (e.g., <code>~</code>var1+var2+...+varN+SNP). See details</p>
</td></tr>
<tr><td><code id="WGassociation_+3A_data">data</code></td>
<td>
<p>a required dataframe of class 'setupSNP' containing the variables in the model and the SNPs</p>
</td></tr>
<tr><td><code id="WGassociation_+3A_model">model</code></td>
<td>
<p>a character string specifying the type of genetic model (mode of inheritance) for the SNP. 
This indicates how the genotypes should be collapsed. Possible
values are &quot;codominant&quot;, &quot;dominant&quot;, &quot;recessive&quot;, &quot;overdominant&quot;, &quot;log-additive&quot; or &quot;all&quot;. The default
is &quot;all&quot; that fits the 5 possible genetic models. Only the first words are required, e.g &quot;co&quot;, &quot;do&quot;, etc.</p>
</td></tr>
<tr><td><code id="WGassociation_+3A_quantitative">quantitative</code></td>
<td>
<p>logical value indicating whether the phenotype (that which is in the
left of the operator ~ in 'formula' argument) is quantitative. The function 
'is.quantitative' returns FALSE when the phenotype
is a variable with two categories (i.e. indicating case-control status). Thus,
it is not a required argument but it may be modified by the user.</p>
</td></tr>
<tr><td><code id="WGassociation_+3A_genotypingrate">genotypingRate</code></td>
<td>
<p> minimum percentage of genotype rate for a given SNP to be included in the analysis.
Default is 80%.</p>
</td></tr>
<tr><td><code id="WGassociation_+3A_level">level</code></td>
<td>
<p>signification level for confidence intervals. Defaul 95%.</p>
</td></tr>
<tr><td><code id="WGassociation_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed through glm function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function assesses the association between the response variable included in the left side in 
the 'formula' and the SNPs included in the 'data' argument adjusted by those variables included 
in the right side of the 'formula'. Different genetic models may be analyzed using 'model' argument.
</p>


<h3>Value</h3>

<p>An object of class 'WGassociation'. 
</p>
<p>'summary' returns a summary table by groups defined in info (genes/chromosomes).
</p>
<p>'WGstats' returns a detailed output, similar to the produced by <code><a href="#topic+association">association</a></code>.
</p>
<p>'pvalues' and 'print' return a table of p-values for each genetic model for each SNP.
The first column indicates whether a problem with genotyping is present.
</p>
<p>'plot' produces a plot of p values in the -log scale. See <code><a href="#topic+plot.WGassociation">plot.WGassociation</a></code> for
further details.
</p>
<p>'labels' returns the names of the SNPs analyzed.
</p>
<p>The functions 'codominat', 'dominant', 'recessive', 'overdominant' and 'additive'
are used to obtain the p values under these genetic models.
</p>
<p>See examples for further illustration about all previous issues. 
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+getSignificantSNPs">getSignificantSNPs</a></code> <code><a href="#topic+association">association</a></code> 
<code><a href="#topic+WGstats">WGstats</a></code> <code><a href="#topic+setupSNP">setupSNP</a></code> 
<code><a href="#topic+plot.WGassociation">plot.WGassociation</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
datSNP&lt;-setupSNP(SNPs,6:40,sep="")
ansAll&lt;-WGassociation(protein~1,data=datSNP,model="all")

# In that case the formula is not required. You can also write:
# ansAll&lt;-WGassociation(protein,data=datSNP,model="all")


#only codominant and log-additive
ansCoAd&lt;-WGassociation(protein~1,data=datSNP,model=c("co","log-add"))

#for printing p values
print(ansAll)
print(ansCoAd)

#for obtaining a matrix with the p palues
pvalAll&lt;-pvalues(ansAll)
pvalCoAd&lt;-pvalues(ansCoAd)

# when all models are fitted and we are interested in obtaining 
# p values for different genetic models

# codominant model
pvalCod&lt;-codominant(ansAll)

# recessive model
pvalRec&lt;-recessive(ansAll)

# and the same for additive, dominant or overdominant


#summary
summary(ansAll)

#for a detailed report
WGstats(ansAll)

#for plotting the p values
plot(ansAll)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
