<!DOCTYPE html><html><head><title>Help for package SNPannotator</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SNPannotator}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annotate'><p>Run the annotation pipeline on a list of variants</p></a></li>
<li><a href='#LDlist'><p>Find variants in high LD with the lead SNP.</p></a></li>
<li><a href='#listDatabases'><p>List population from human database (1000 Genomes project)</p></a></li>
<li><a href='#pingServer'><p>Checks if the service is alive</p></a></li>
<li><a href='#releaseVersion'><p>Shows the data releases available on this REST server.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Investigating the Functional Characteristics of Selected SNPs
and Their Vicinity Genomic Region</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.6.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-01-12</td>
</tr>
<tr>
<td>Author:</td>
<td>Alireza Ani [aut, cre],
  Zoha Kamali [aut],
  Ahmad Vaez [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alireza Ani &lt;a.ani@umcg.nl&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, httr, jsonlite, xml2, openxlsx, progress,
doParallel, foreach, ggplot2, kableExtra, methods, rmarkdown</td>
</tr>
<tr>
<td>Description:</td>
<td>To investigate the functional characteristics of selected SNPs and their vicinity genomic region. Linked SNPs in moderate to high linkage disequilibrium (e.g. r2&gt;0.50) with the corresponding index SNPs will be selected for further analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-12 09:33:50 UTC; Alireza</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-12 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='annotate'>Run the annotation pipeline on a list of variants</h2><span id='topic+annotate'></span>

<h3>Description</h3>

<p>This function receives a list of variants and checks their information on Ensembl website via
the Ensembl REST API server.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotate(
  rslist,
  server,
  db,
  outputPath,
  window_size = 500,
  r2 = 0.5,
  LDlist = TRUE,
  cadd = FALSE,
  geneNames.file = NULL,
  regulatoryType.file = NULL,
  cores = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotate_+3A_rslist">rslist</code></td>
<td>
<p>A vector of rs numbers.</p>
</td></tr>
<tr><td><code id="annotate_+3A_server">server</code></td>
<td>
<p>Name of the server. &quot;https://rest.ensembl.org&quot; can be used for GRCh38
and &quot;https://grch37.rest.ensembl.org&quot; for GRCh37.</p>
</td></tr>
<tr><td><code id="annotate_+3A_db">db</code></td>
<td>
<p>The population database for calculating LD scores. This can be found using <code>listDatabases</code> function.</p>
</td></tr>
<tr><td><code id="annotate_+3A_outputpath">outputPath</code></td>
<td>
<p>The report file will be saved in this path as an Excel file (.xlsx)</p>
</td></tr>
<tr><td><code id="annotate_+3A_window_size">window_size</code></td>
<td>
<p>Number of base pairs around the variant for checking LD scores (max = 500kb)</p>
</td></tr>
<tr><td><code id="annotate_+3A_r2">r2</code></td>
<td>
<p>The LD threshold for selecting variants around the target SNP.</p>
</td></tr>
<tr><td><code id="annotate_+3A_ldlist">LDlist</code></td>
<td>
<p>If set to TRUE, variants in high LD will be found and added to the output.</p>
</td></tr>
<tr><td><code id="annotate_+3A_cadd">cadd</code></td>
<td>
<p>If set to TRUE, the CADD scores will be added to variant information.</p>
</td></tr>
<tr><td><code id="annotate_+3A_genenames.file">geneNames.file</code></td>
<td>
<p>path the gene information file (*.rds). Default value is NULL and ENSEMBL website will be checked if no file is provided.</p>
</td></tr>
<tr><td><code id="annotate_+3A_regulatorytype.file">regulatoryType.file</code></td>
<td>
<p>path the variants regulatory type information file (*.rds). Default value is NULL and this step will be skipped if no file is provided.</p>
</td></tr>
<tr><td><code id="annotate_+3A_cores">cores</code></td>
<td>
<p>set to a value above 0 for parallel processing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data table with all variant information is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# select the required server
server &lt;- "https://grch37.rest.ensembl.org"

# select the database for population data
# this can be selected from listDatabases() function
db &lt;- "1000GENOMES:phase_3:EUR"

# create a vector of required SNPs
rslist=c('rs236349')

output &lt;- annotate(rslist,server,db,
  outputPath = paste(tempdir(),'sampleOutput.xlsx',sep="/"),
  window_size = 500,
  r2 = .9,
  cadd = FALSE)
  
## End(Not run)

</code></pre>

<hr>
<h2 id='LDlist'>Find variants in high LD with the lead SNP.</h2><span id='topic+LDlist'></span>

<h3>Description</h3>

<p>This function returns a list of variables that are in high LD with the lead variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDlist(rslist, server, db, window_size, r2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDlist_+3A_rslist">rslist</code></td>
<td>
<p>A vector of rs numbers.</p>
</td></tr>
<tr><td><code id="LDlist_+3A_server">server</code></td>
<td>
<p>Name of the server. &quot;https://rest.ensembl.org&quot; can be used for GRCh38
and &quot;https://grch37.rest.ensembl.org&quot; for GRCh37.</p>
</td></tr>
<tr><td><code id="LDlist_+3A_db">db</code></td>
<td>
<p>The population database for calculating LD scores. This can be found using <code>listDatabases</code> function.</p>
</td></tr>
<tr><td><code id="LDlist_+3A_window_size">window_size</code></td>
<td>
<p>Number of base pairs around the variant for checking LD scores (max = 500kb)</p>
</td></tr>
<tr><td><code id="LDlist_+3A_r2">r2</code></td>
<td>
<p>The LD threshold for selecting variants around the target SNP.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data table with variant information.
</p>

<hr>
<h2 id='listDatabases'>List population from human database (1000 Genomes project)</h2><span id='topic+listDatabases'></span>

<h3>Description</h3>

<p>This function list the name, description and size of the available populations
in 1000 Genomes project database. This database will be used for returning variables in high LD
with the target SNP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listDatabases(server)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listDatabases_+3A_server">server</code></td>
<td>
<p>name of the server. &quot;https://rest.ensembl.org&quot; can be used for GRCh38
and &quot;https://grch37.rest.ensembl.org&quot; for GRCh37.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table is returned which includes the name, description and size of the available populations
in 1000 Genomes project database.
</p>

<hr>
<h2 id='pingServer'>Checks if the service is alive</h2><span id='topic+pingServer'></span>

<h3>Description</h3>

<p>This function test whether the Ensembl server is accessible or not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pingServer(server)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pingServer_+3A_server">server</code></td>
<td>
<p>name of the server. &quot;https://rest.ensembl.org&quot; can be used for GRCh38
and &quot;https://grch37.rest.ensembl.org&quot; for GRCh37.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a message is displayed to the user
</p>

<hr>
<h2 id='releaseVersion'>Shows the data releases available on this REST server.</h2><span id='topic+releaseVersion'></span>

<h3>Description</h3>

<p>Shows the data releases available on this REST server.
May return more than one release (unfrequent non-standard Ensembl configuration).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>releaseVersion(server)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="releaseVersion_+3A_server">server</code></td>
<td>
<p>name of the server.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a message is displayed to the user
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
