<!DOCTYPE html><html><head><title>Help for package snpAIMeR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {snpAIMeR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#snpAIMeR'><p>Assess the Diagnostic Power of Genomic Marker Combinations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Assess the Diagnostic Power of Genomic Marker Combinations</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Population genetics package for designing diagnostic panels. 
    Candidate markers, marker combinations, and different panel sizes are 
    assessed for how well they can predict the source population of known 
    samples. Requires a genotype file of candidate markers in STRUCTURE format. 
    Methods for population cross-validation are described in Jombart (2008) 
    &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn129">doi:10.1093/bioinformatics/btn129</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Contact:</td>
<td>Kim Vertacnik &lt;kim.vertacnik@mailbox.org&gt; or Julian Dupuis
&lt;julian.dupuis@uky.edu&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/OksanaVe/snpAIMeR">https://github.com/OksanaVe/snpAIMeR</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>adegenet, doParallel, dplyr, forcats, foreach, ggplot2,
graphics, magrittr, parallel, readr, tidyr, utils, withr, yaml</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-20 20:01:29 UTC; kim</td>
</tr>
<tr>
<td>Author:</td>
<td>Kim Vertacnik <a href="https://orcid.org/0009-0006-7097-5091"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut],
  Oksana Vernygora <a href="https://orcid.org/0000-0001-6444-3888"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Julian Dupuis <a href="https://orcid.org/0000-0002-6989-9179"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kim Vertacnik &lt;kim.vertacnik@mailbox.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-20 20:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='snpAIMeR'>Assess the Diagnostic Power of Genomic Marker Combinations</h2><span id='topic+snpAIMeR'></span>

<h3>Description</h3>

<p>Population genetics package for optimizing diagnostic panels. User-selected
candidate markers are assessed individually and in combination for how
well they can predict the source population of known samples. Requires a
genotype file in STRUCTURE format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpAIMeR(run_mode, config_file = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snpAIMeR_+3A_run_mode">run_mode</code></td>
<td>
<p>Modes are &quot;interactive&quot;, &quot;non-interactive&quot;, or &quot;example&quot;; mode
must be in quotes.</p>
</td></tr>
<tr><td><code id="snpAIMeR_+3A_config_file">config_file</code></td>
<td>
<p>Yaml file required for &quot;non-interactive&quot; mode; filename/path
must be in quotes.</p>
</td></tr>
<tr><td><code id="snpAIMeR_+3A_verbose">verbose</code></td>
<td>
<p>Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Yaml file format for &quot;non-interactive&quot; mode (do not include bullet points):
</p>

<ul>
<li><p> min_range: &lt;minimum panel size&gt;
</p>
</li>
<li><p> max_range: &lt;maximum panel size; we recommend no more than 15 markers&gt;
</p>
</li>
<li><p> assignment_rate_threshold: &lt;value from 0 to 1&gt;
</p>
</li>
<li><p> cross_validation_replicates: &lt;we recommend 100 minimum&gt;
</p>
</li>
<li><p> working_directory: &lt;path name in quotes&gt;
</p>
</li>
<li><p> structure_file: &lt;path name in quotes&gt;
</p>
</li>
<li><p> number_of_individuals: &lt;same as adegenet's &quot;n.ind&quot;&gt;
</p>
</li>
<li><p> number_of_loci: &lt;same as adegenet's &quot;n.loc&quot;&gt;
</p>
</li>
<li><p> one_data_row_per_individual: &lt;TRUE or FALSE&gt;
</p>
</li>
<li><p> column_sample_IDs: &lt;column number&gt;
</p>
</li>
<li><p> column_population_assignments: &lt;column number&gt;
</p>
</li>
<li><p> column_other_info: &lt;column number&gt;
</p>
</li>
<li><p> row_markernames: &lt;row number&gt;
</p>
</li>
<li><p> no_genotype_character: &lt;default is &quot;-9&quot;&gt;
</p>
</li>
<li><p> optional_population_info: &lt;optional&gt;
</p>
</li>
<li><p> genotype_character_separator: &lt;optional&gt;
</p>
</li></ul>

<p>Minimizing run time: Because of the number of possible combinations, we
recommend testing no more than 15 markers. For example, testing 15 markers
in panel sizes of 1 to 15 (32,767 total combinations) with 1,000
cross-validation replicates on a system with 48 processor cores took about 5
hours and 20 GB RAM. Reducing the number of cross-validation replicates will
reduce run time, however, we recommend no less than 100 replicates.
</p>


<h3>Value</h3>

<p>Cross-validation assignment rates for individual markers, marker
combinations, and panel sizes. Outputs three .csv and two .pdf files to a
user-specified directory.
</p>


<h3>See Also</h3>

<p>https://github.com/OksanaVe/snpAIMeR
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("adegenet", quietly = TRUE)) {
  data(nancycats, package = "adegenet")
  snpAIMeR("example", verbose = TRUE)
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
