<!DOCTYPE html><html><head><title>Help for package snplist</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {snplist}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#exportPLINKSet'>
<p>exportPLINKSet</p></a></li>
<li><a href='#getBioMartData'>
<p>getBioMartData</p></a></li>
<li><a href='#makeGeneSet'>
<p>makeGeneSet</p></a></li>
<li><a href='#setGeneTable'>
<p>setGeneTable</p></a></li>
<li><a href='#setSNPTable'>
<p>setSNPTable</p></a></li>
<li><a href='#snplist-package'>
<p>Tools to Create Gene Sets</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools to Create Gene Sets</td>
</tr>
<tr>
<td>Version:</td>
<td>0.18.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-10-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Chanhee Yi, Alexander Sibley, and Kouros Owzar</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alexander Sibley &lt;dcibioinformatics@duke.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A set of functions to create SQL tables of gene and SNP information and compose them into a SNP Set, for example to export to a PLINK set.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0), RSQLite (&ge; 1.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>biomaRt (&ge; 2.16.0), Rcpp (&ge; 0.10.5), R.utils (&ge; 1.27.1),
DBI (&ge; 0.3.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-10-08 16:52:47 UTC; alex</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-10-09 15:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='exportPLINKSet'>
exportPLINKSet
</h2><span id='topic+exportPLINKSet'></span>

<h3>Description</h3>

<p>Simple function using Rcpp to write the gene set to a file in the PLINK set format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  exportPLINKSet(geneSets, fname)	
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exportPLINKSet_+3A_genesets">geneSets</code></td>
<td>
<p>An object created by the <code>makeGeneSet()</code> function.
</p>
</td></tr>
<tr><td><code id="exportPLINKSet_+3A_fname">fname</code></td>
<td>
<p>The name of the PLINK file to be created.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Boolean indicating if the file was successfully written.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makeGeneSet">makeGeneSet</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Please see the vignette or the package description 
    # for an example of using this function.
</code></pre>

<hr>
<h2 id='getBioMartData'>
getBioMartData
</h2><span id='topic+getBioMartData'></span>

<h3>Description</h3>

<p>A function leveraging the <code>biomaRt</code> package to retrieve gene chromosome and start and end positions from Ensembl.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    getBioMartData(genes,verbose=FALSE,...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBioMartData_+3A_genes">genes</code></td>
<td>

<p>A vector of gene names matching <code>hgnc_symbol</code> in the Ensembl database.
</p>
</td></tr>
<tr><td><code id="getBioMartData_+3A_verbose">verbose</code></td>
<td>

<p>A Boolean indicating whether to output the funcitons progress in terms of the dimensions of the <code>data.frame</code> being constructed. Default is FALSE.
</p>
</td></tr>
<tr><td><code id="getBioMartData_+3A_...">...</code></td>
<td>

<p>Additional arguments passed on to the internal call to <code>biomaRt::useMart(...)</code>. If no such arguments are provided, <code>useMart("ensembl", dataset="hsapiens_gene_ensembl")</code> is run by default.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> object with columns 'gene','chr','start', and 'end', suitable for input to the <code>setGeneTable</code> function.
</p>


<h3>Note</h3>

<p>At the time of package release, the BioMart community portal is temporarily unavailable. See www.biomart.org for updated status or more information. To access alternative hosts, pass additional arguments to the internal call to <code>biomaRt::useMart(...)</code>, as in the second example below.
</p>


<h3>References</h3>

<p>Durinck S., Spellman P.T., Birney E. and Huber W. (2009) Mapping identifiers for the integration of genomic datasets with the R/Bioconductor package biomaRt, <em>Nature Protocols</em>, <b>4</b>, 1184&ndash;1191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+setGeneTable">setGeneTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getBioMartData(c("BRCA1","BRCA2"))
getBioMartData(c("BRCA1","BRCA2"), 
               host="www.ensembl.org", 
               biomart="ENSEMBL_MART_ENSEMBL", 
               dataset="hsapiens_gene_ensembl")

## End(Not run)
</code></pre>

<hr>
<h2 id='makeGeneSet'>
makeGeneSet
</h2><span id='topic+makeGeneSet'></span>

<h3>Description</h3>

<p>This function uses existing SQLite tables (from <code>setGeneTable</code> and <code>setSNPTable</code>) to make SNP sets. The SNP Set for each gene is the collection of SNPs located either between the start and end locations of the gene, or within a specified neighborhood around the gene. The SNP Sets are stored in the SQLite database, and returned as a list object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    makeGeneSet(annoInfo=NULL,margin=0,annoTable='anno',geneTable='gene',
                        allTable='allchrpos',db='snplistdb',dbCleanUp=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeGeneSet_+3A_annoinfo">annoInfo</code></td>
<td>

<p>A <code>vector</code> of rsIDs, a <code>data.frame</code> with an 'rsid' column, or a file with one rsID per line. The SNP sets will be restricted to contain only the SNPs listed here. Default is NULL, in which case all SNPs present in the SNP table in the SQLite database will be used.
</p>
</td></tr>
<tr><td><code id="makeGeneSet_+3A_margin">margin</code></td>
<td>

<p>A number, indicating the size of the neighborhood (in base pairs) surrounding a genes start and end positions in which a SNP will be included in that genes SNP set. Default is 0.
</p>
</td></tr>
<tr><td><code id="makeGeneSet_+3A_annotable">annoTable</code></td>
<td>

<p>A string indicating the name of the SQLite table for the rsIDs from <code>annoInfo</code>. Also used in naming the resulting table of SNP sets ('&lt;name&gt;ToGene').  Default is 'anno'.
</p>
</td></tr>
<tr><td><code id="makeGeneSet_+3A_genetable">geneTable</code></td>
<td>

<p>Name of the SQLite table containg chromosome, start and end positions for each gene, as previously created by <code>setGeneTable</code>. Default is 'gene'.
</p>
</td></tr>
<tr><td><code id="makeGeneSet_+3A_alltable">allTable</code></td>
<td>

<p>Name of the SQLite table containg chromosome and position for each SNP, as previously created by <code>setSNPTable</code>. Default is 'allchrpos'.
</p>
</td></tr>
<tr><td><code id="makeGeneSet_+3A_db">db</code></td>
<td>

<p>Name of the SQLite database in which to find the gene and SNP tables and create the SNP set table. Default is 'snplistdb'.
</p>
</td></tr>
<tr><td><code id="makeGeneSet_+3A_dbcleanup">dbCleanUp</code></td>
<td>

<p>Boolean indicating if the tables and views created by the function should be dropped after the SNP set is returned. Default is FALSE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: This function relies on the prior execution of the <code>setGeneTable</code> and <code>setSNPTable</code> functions and the SQLite databes and tables they create. If the <code>table</code> or <code>db</code> argument in either of those functions is changed from the default value, it must also be changed here.
</p>


<h3>Value</h3>

<p>Returns a <code>list</code> of SNP sets of the form:
</p>
<table>
<tr><td><code>&lt;gene name&gt;</code></td>
<td>
<p>Vector of rsIDs of SNPs within &lt;gene&gt; (or the neighborhood around it)</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+setGeneTable">setGeneTable</a></code>, <code><a href="#topic+setSNPTable">setSNPTable</a></code>, <code><a href="#topic+snplist-package">snplist-package</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Please see the vignette or the package description 
    # for an example of using this function.
</code></pre>

<hr>
<h2 id='setGeneTable'>
setGeneTable
</h2><span id='topic+setGeneTable'></span>

<h3>Description</h3>

<p>Takes a <code>data.frame</code> object with columns 'gene','chr','start', and 'end', and creates an SQLite table of the information. Returns a count of the number of genes in the table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    setGeneTable(geneInfo,table='gene',db='snplistdb') 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setGeneTable_+3A_geneinfo">geneInfo</code></td>
<td>

<p>A <code>data.frame</code> object of gene location info with columns 'gene','chr','start', and 'end'.
</p>
</td></tr>
<tr><td><code id="setGeneTable_+3A_table">table</code></td>
<td>

<p>Name of the SQLite table to be created. Default is 'gene'.
</p>
</td></tr>
<tr><td><code id="setGeneTable_+3A_db">db</code></td>
<td>

<p>Name of the SQLite database in which to create <code>table</code>. Default is 'snplistdb'.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Count of genes included in <code>table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    geneInfo &lt;- cbind(c('BRCA1','BRCA2'),c(17,13),c(41196312,32889611),c(41277500,32973805))
    colnames(geneInfo) &lt;- c('gene','chr','start','end')
    ## Not run: 
    setGeneTable(as.data.frame(geneInfo))
    
## End(Not run)
</code></pre>

<hr>
<h2 id='setSNPTable'>
setSNPTable
</h2><span id='topic+setSNPTable'></span>

<h3>Description</h3>

<p>Takes a file or <code>data.frame</code> object with columns 'chr','pos', and 'rsid', and creates an SQLite table of the information. Returns a count of the number of SNPs in the table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    setSNPTable(snpInfo,table='allchrpos',db='snplistdb')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setSNPTable_+3A_snpinfo">snpInfo</code></td>
<td>

<p>A <code>data.frame</code> object of SNP location info with columns 'chr','pos', and 'rsid', or a tab-delimited file with those columns and one record per row.
</p>
</td></tr>
<tr><td><code id="setSNPTable_+3A_table">table</code></td>
<td>

<p>Name of the SQLite table to be created. Default is 'allchrpos'.
</p>
</td></tr>
<tr><td><code id="setSNPTable_+3A_db">db</code></td>
<td>

<p>Name of the SQLite database in which to create <code>table</code>. Default is 'snplistdb'.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Count of genes included in <code>table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    snpInfo &lt;- cbind(c(17,17,13,13),
		     c(41211653, 41213996, 32890026,32890572),
		     c("rs8176273","rs8176265","rs9562605","rs1799943") )
    colnames(snpInfo) &lt;- c('chr','pos','rsid')
    ## Not run: 
    setSNPTable(as.data.frame(snpInfo))
    
## End(Not run)
</code></pre>

<hr>
<h2 id='snplist-package'>
Tools to Create Gene Sets
</h2><span id='topic+snplist-package'></span><span id='topic+snplist'></span>

<h3>Description</h3>

<p>A set of functions to create SQL tables of gene and SNP information and compose them into a SNP Set, for example for use with the <code>RSNPset</code> package, or to export to a PLINK set.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> snplist</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.18.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-12-11</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Please see the example function calls below, or refer to the individual function documentation or the included vignette for more information.
</p>


<h3>Author(s)</h3>

<p>Authors: Chanhee Yi, Alexander Sibley, and Kouros Owzar
Maintainer: Alexander Sibley &lt;alexander.sibley@dm.duke.edu&gt;
</p>


<h3>See Also</h3>

<p><code>RSQLite</code>, <code><a href="Rcpp.html#topic+Rcpp">Rcpp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>chromosome &lt;- c(1,5,22,"X","Y","MT")

geneNum &lt;- 5
snpNum &lt;- 1200
annoDataNum &lt;- 500

chrLength &lt;- 1000
geneLength &lt;- 100

gene &lt;- paste("gene",1:geneNum,sep="")
chr &lt;- sample(chromosome,geneNum,replace=TRUE)
start &lt;- sample(chrLength,geneNum,replace=TRUE)
d &lt;- sample(geneLength,geneNum,replace=TRUE)
end &lt;- start+d
geneInfo &lt;- data.frame(gene,chr,start,end)

rsid &lt;- paste("rs",1:snpNum,sep="")
chr &lt;- sample(chromosome,snpNum,replace=TRUE)
pos &lt;- sample(chrLength+geneLength,snpNum,replace=TRUE)
snpInfo &lt;- data.frame(rsid,chr,pos)

annoInfo &lt;- data.frame("rsid"=sample(rsid,annoDataNum))

dim(geneInfo)
dim(snpInfo)
dim(annoInfo)

## Not run: 
setGeneTable(geneInfo)
setSNPTable(snpInfo)
geneset &lt;- makeGeneSet(annoInfo)
exportPLINKSet(geneset,"geneSet.set")
file.show("geneSet.set")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
