<!DOCTYPE html><html><head><title>Help for package snplinkage</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {snplinkage}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe</p></a></li>
<li><a href='#%&lt;&gt;%'><p>Assignment pipe</p></a></li>
<li><a href='#%$%'><p>Exposition pipe</p></a></li>
<li><a href='#chisq_pvalues'><p>Compute Chi-squared p-values</p></a></li>
<li><a href='#chisq_pvalues_gdata'><p>Compute Chi-squared p-values on a Genotype data object</p></a></li>
<li><a href='#diamond_annots'><p>Get diamond ggplot layer.</p></a></li>
<li><a href='#gdata_add_gene_annots'><p>gdata_add_gene_annots</p></a></li>
<li><a href='#gdata_add_gene_annots_aim_example'><p>gdata_add_gene_annots_aim_example</p></a></li>
<li><a href='#gdata_add_gene_annots_hladr_example'><p>gdata_add_gene_annots_hladr_example</p></a></li>
<li><a href='#gdata_scans_annots'><p>gdata_scan_annots</p></a></li>
<li><a href='#gdata_snps_annots'><p>gdata_snp_annots</p></a></li>
<li><a href='#get_biomart_metadb'><p>get_biomart_metadb</p></a></li>
<li><a href='#ggplot_associations'><p>Ggplot associations</p></a></li>
<li><a href='#ggplot_ld'><p>Ggplot linkage disequilibrium</p></a></li>
<li><a href='#ggplot_snp_pos'><p>Ggplot SNPs position</p></a></li>
<li><a href='#gtable_ld'><p>Gtable of linkage disequilibrium and chromosomic positions</p></a></li>
<li><a href='#gtable_ld_associations'><p>Gtable of linkage disequilibrium and associations</p></a></li>
<li><a href='#gtable_ld_associations_gdata'><p>Gtable of linkage disequilibrium and associations using a GenotypeData object</p></a></li>
<li><a href='#gtable_ld_gdata'><p>Gtable of linkage disequilibrium and positions using a GenotypeData object</p></a></li>
<li><a href='#load_gds_as_genotype_data'><p>Load GDS as Genotype Data</p></a></li>
<li><a href='#parallel_apply'><p>Separate a matrix in a list of matrices of length the number of cores and</p>
apply a function on the columns in parallel</a></li>
<li><a href='#print_qc_as_tex_table'><p>print_qc_as_tex_table</p></a></li>
<li><a href='#save_hgdp_as_gds'><p>save_hgdp_as_gds</p></a></li>
<li><a href='#select_region_idxs'><p>select_region_idxs</p></a></li>
<li><a href='#snplinkage-package'><p>snplinkage: Single Nucleotide Polymorphisms Linkage Disequilibrium Visualizations</p></a></li>
<li><a href='#snprelate_allele_frequencies'><p>Compute allele frequencie and snp missing rate</p></a></li>
<li><a href='#snprelate_ld'><p>Wrapper for snpgdsLDMat to compute r2</p></a></li>
<li><a href='#snprelate_ld_select'><p>Wrapper for snpgdsLDpruning to select Tag SNPs</p></a></li>
<li><a href='#snprelate_qc'><p>snprelate_qc</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Single Nucleotide Polymorphisms Linkage Disequilibrium
Visualizations</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Linkage disequilibrium visualizations of up to several hundreds of single nucleotide polymorphisms (SNPs), annotated with chromosomic positions and gene names. Two types of plots are available for small numbers of SNPs (&lt;40) and for large numbers (tested up to 500). Both can be extended by combining other ggplots, e.g. association studies results, and functions enable to directly visualize the effect of SNP selection methods, as minor allele frequency filtering and TagSNP selection, with a second correlation heatmap. The SNPs correlations are computed on Genotype Data objects from the 'GWASTools' package using the 'SNPRelate' package, and the plots are customizable 'ggplot2' and 'gtable' objects and are annotated using the 'biomaRt' package. Usage is detailed in the vignette with example data and results from up to 500 SNPs of 1,200 scans are in Charlon T. (2019) &lt;<a href="https://doi.org/10.13097%2Farchive-ouverte%2Funige%3A161795">doi:10.13097/archive-ouverte/unige:161795</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>gdsfmt, ggplot2, gtable, magrittr, stats, utils</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15), GWASTools (&ge; 1.10.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>biomaRt, cowplot, data.table, dplyr, ggrepel, grid,
grDevices, knitr, methods, plyr, SNPRelate, testthat</td>
</tr>
<tr>
<td>biocViews:</td>
<td>GeneticVariability, MicroArray, SNP</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://gitlab.com/thomaschln/snplinkage">https://gitlab.com/thomaschln/snplinkage</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://gitlab.com/thomaschln/snplinkage/-/issues">https://gitlab.com/thomaschln/snplinkage/-/issues</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-03 06:43:57 UTC; root</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Charlon <a href="https://orcid.org/0000-0001-7497-0470"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Karl Forner [aut],
  Alessandro Di Cara [aut],
  Jérôme Wojcik [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Charlon &lt;charlon@protonmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-04 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>Pipe an object forward into a function or call expression.
Magrittr imported function, see details and examples in the magrittr package.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Result of rhs applied to lhs, see details in magrittr package.
</p>

<hr>
<h2 id='+25+26lt+3B+26gt+3B+25'>Assignment pipe</h2><span id='topic++25+3C+3E+25'></span>

<h3>Description</h3>

<p>Pipe an object forward into a function or call expression and update the
'lhs' object with the resulting value.
Magrittr imported function, see details and examples in the magrittr package.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26lt+2B3B+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>An object which serves both as the initial value and as target.</p>
</td></tr>
<tr><td><code id="+2B25+2B26lt+2B3B+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>a function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None, used to update the value of lhs.
</p>

<hr>
<h2 id='+25+24+25'>Exposition pipe</h2><span id='topic++25+24+25'></span>

<h3>Description</h3>

<p>Expose the names in 'lhs' to the 'rhs' expression.
Magrittr imported function, see details and examples in the magrittr package.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B24+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A list, environment, or a data.frame.</p>
</td></tr>
<tr><td><code id="+2B25+2B24+2B25_+3A_rhs">rhs</code></td>
<td>
<p>An expression where the names in lhs is available.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Result of rhs applied to one or several names of lhs.
</p>

<hr>
<h2 id='chisq_pvalues'>Compute Chi-squared p-values</h2><span id='topic+chisq_pvalues'></span>

<h3>Description</h3>

<p>Compute Chi-squared p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chisq_pvalues(
  m_data,
  response,
  adjust_method = "fdr",
  mlog10_transform = TRUE,
  n_cores = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chisq_pvalues_+3A_m_data">m_data</code></td>
<td>
<p>Data matrix of observations by variables</p>
</td></tr>
<tr><td><code id="chisq_pvalues_+3A_response">response</code></td>
<td>
<p>Response vector of length the number of observations</p>
</td></tr>
<tr><td><code id="chisq_pvalues_+3A_adjust_method">adjust_method</code></td>
<td>
<p>Multiple testing p-value adjustment method.
Passed to stats::p.adjust. 'fdr' by default.</p>
</td></tr>
<tr><td><code id="chisq_pvalues_+3A_mlog10_transform">mlog10_transform</code></td>
<td>
<p>Logical, transform p-values by minus log10.
True by default.</p>
</td></tr>
<tr><td><code id="chisq_pvalues_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of cores</p>
</td></tr>
<tr><td><code id="chisq_pvalues_+3A_...">...</code></td>
<td>
<p>Passed to stats::chisq.test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Chi-squared p-values
</p>

<hr>
<h2 id='chisq_pvalues_gdata'>Compute Chi-squared p-values on a Genotype data object</h2><span id='topic+chisq_pvalues_gdata'></span>

<h3>Description</h3>

<p>Compute Chi-squared p-values on a Genotype data object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chisq_pvalues_gdata(
  gdata,
  snp_idxs,
  response_column = "region",
  response_value = "Europe",
  threshold = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chisq_pvalues_gdata_+3A_gdata">gdata</code></td>
<td>
<p>Genotype data object</p>
</td></tr>
<tr><td><code id="chisq_pvalues_gdata_+3A_snp_idxs">snp_idxs</code></td>
<td>
<p>SNPs indexes</p>
</td></tr>
<tr><td><code id="chisq_pvalues_gdata_+3A_response_column">response_column</code></td>
<td>
<p>Response column in gdata scans annotations data frame</p>
</td></tr>
<tr><td><code id="chisq_pvalues_gdata_+3A_response_value">response_value</code></td>
<td>
<p>Response value. The response vector will be a logical,
true if equal to the value, false otherwise.</p>
</td></tr>
<tr><td><code id="chisq_pvalues_gdata_+3A_threshold">threshold</code></td>
<td>
<p>Keep only associations greater than the threshold</p>
</td></tr>
<tr><td><code id="chisq_pvalues_gdata_+3A_...">...</code></td>
<td>
<p>Passed to chisq_pvalues</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SNPs annotation data frame, chi-squared p-values in column pvalues
</p>

<hr>
<h2 id='diamond_annots'>Get diamond ggplot layer.</h2><span id='topic+diamond_annots'></span>

<h3>Description</h3>

<p>Diamond ggplot layer for ggplot_ld
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diamond_annots(data, x = "x", y = "y", color = "color", size = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diamond_annots_+3A_data">data</code></td>
<td>
<p>Data frame of 3 columns defining the diamonds</p>
</td></tr>
<tr><td><code id="diamond_annots_+3A_x">x</code></td>
<td>
<p>Name of the column for horizontal positions</p>
</td></tr>
<tr><td><code id="diamond_annots_+3A_y">y</code></td>
<td>
<p>Name of the column for vertical positions</p>
</td></tr>
<tr><td><code id="diamond_annots_+3A_color">color</code></td>
<td>
<p>Name of the column for color values</p>
</td></tr>
<tr><td><code id="diamond_annots_+3A_size">size</code></td>
<td>
<p>Radius of the diamonds</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gglayers
</p>

<hr>
<h2 id='gdata_add_gene_annots'>gdata_add_gene_annots</h2><span id='topic+gdata_add_gene_annots'></span>

<h3>Description</h3>

<p>Add biomaRt gene annotations to Genotype Data object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdata_add_gene_annots(
  gdata,
  snp_idxs,
  rsids_colname = "probe_id",
  biomart_metadb = get_biomart_metadb()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdata_add_gene_annots_+3A_gdata">gdata</code></td>
<td>
<p>Genotype Data object</p>
</td></tr>
<tr><td><code id="gdata_add_gene_annots_+3A_snp_idxs">snp_idxs</code></td>
<td>
<p>SNP indexes</p>
</td></tr>
<tr><td><code id="gdata_add_gene_annots_+3A_rsids_colname">rsids_colname</code></td>
<td>
<p>Column of SNP annotation data frame with rs identifiers</p>
</td></tr>
<tr><td><code id="gdata_add_gene_annots_+3A_biomart_metadb">biomart_metadb</code></td>
<td>
<p>List with slots snpmart and ensembl, corresponding to 
the biomart databases to query for SNP identifiers
and gene names, respectively. See get_biomart_metadb
function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype Data object
</p>

<hr>
<h2 id='gdata_add_gene_annots_aim_example'>gdata_add_gene_annots_aim_example</h2><span id='topic+gdata_add_gene_annots_aim_example'></span>

<h3>Description</h3>

<p>Add ancestry informative markers gene annotations to Genotype Data object.
Convenience function for the vignette to avoid querying biomaRt on build.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdata_add_gene_annots_aim_example(gdata, aim_idxs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdata_add_gene_annots_aim_example_+3A_gdata">gdata</code></td>
<td>
<p>Genotype Data object</p>
</td></tr>
<tr><td><code id="gdata_add_gene_annots_aim_example_+3A_aim_idxs">aim_idxs</code></td>
<td>
<p>AIM indexes in the example Genotype data object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype Data object
</p>

<hr>
<h2 id='gdata_add_gene_annots_hladr_example'>gdata_add_gene_annots_hladr_example</h2><span id='topic+gdata_add_gene_annots_hladr_example'></span>

<h3>Description</h3>

<p>Add HLA-DR gene annotations to Genotype Data object.
Convenience function for the vignette to avoid querying biomaRt on build.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdata_add_gene_annots_hladr_example(gdata, hla_dr_idxs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdata_add_gene_annots_hladr_example_+3A_gdata">gdata</code></td>
<td>
<p>Genotype Data object</p>
</td></tr>
<tr><td><code id="gdata_add_gene_annots_hladr_example_+3A_hla_dr_idxs">hla_dr_idxs</code></td>
<td>
<p>HLA-DR indexes in the example Genotype data object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype Data object
</p>

<hr>
<h2 id='gdata_scans_annots'>gdata_scan_annots</h2><span id='topic+gdata_scans_annots'></span>

<h3>Description</h3>

<p>Get scans annotations from a Genotype Data object or a subset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdata_scans_annots(gdata, scan_ids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdata_scans_annots_+3A_gdata">gdata</code></td>
<td>
<p>Genotype Data object</p>
</td></tr>
<tr><td><code id="gdata_scans_annots_+3A_scan_ids">scan_ids</code></td>
<td>
<p>Scan identifiers to subset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Scans annotations data frame
</p>

<hr>
<h2 id='gdata_snps_annots'>gdata_snp_annots</h2><span id='topic+gdata_snps_annots'></span>

<h3>Description</h3>

<p>Get SNPs annotations from a Genotype Data object or a subset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gdata_snps_annots(gdata, snp_ids = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gdata_snps_annots_+3A_gdata">gdata</code></td>
<td>
<p>Genotype Data object</p>
</td></tr>
<tr><td><code id="gdata_snps_annots_+3A_snp_ids">snp_ids</code></td>
<td>
<p>SNP identifiers to subset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SNP annotation data frame
</p>

<hr>
<h2 id='get_biomart_metadb'>get_biomart_metadb</h2><span id='topic+get_biomart_metadb'></span>

<h3>Description</h3>

<p>To query gene names of SNPs, it is necessary to retrieve two objects using
biomaRt::useMart. First, the object required to map SNP rs identifiers to
ENSEMBL identifiers. Second, the object required to map ENSEMBL identifiers
to common gene names. The function returns a list of two slots named
snpmart and ensembl corresponding to each one, respectively.
Once obtained it is saved to a local file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_biomart_metadb(
  filepath = extdata_filepath("bmart_meta.rds"),
  host = "https://grch37.ensembl.org"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_biomart_metadb_+3A_filepath">filepath</code></td>
<td>
<p>Path to save the biomaRt objects</p>
</td></tr>
<tr><td><code id="get_biomart_metadb_+3A_host">host</code></td>
<td>
<p>BiomaRt Ensembl host, by default https://grch37.ensembl.org</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of slots snpmart and ensembl as detailed above
</p>

<hr>
<h2 id='ggplot_associations'>Ggplot associations</h2><span id='topic+ggplot_associations'></span>

<h3>Description</h3>

<p>Get SNPs associations ggplot, either as points or as a linked area.
Optionally add labels to most associated points using ggrepel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_associations(
  df_snp,
  pvalue_colname = "pvalues",
  labels_colname = "probe_id",
  n_labels = 10,
  nudge = c(0, 1),
  linked_area = FALSE,
  byindex = linked_area,
  colors = if (linked_area) snp_position_colors(nrow(df_snp)) else "black"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_associations_+3A_df_snp">df_snp</code></td>
<td>
<p>SNP annotation data frame with columns chromosome,
position, and as specified by parameters pvalue_colname
and optionally labels_colname.</p>
</td></tr>
<tr><td><code id="ggplot_associations_+3A_pvalue_colname">pvalue_colname</code></td>
<td>
<p>Column name of df_snp with association values</p>
</td></tr>
<tr><td><code id="ggplot_associations_+3A_labels_colname">labels_colname</code></td>
<td>
<p>Optional column name of df_snp with labels.
Set to NULL to remove.</p>
</td></tr>
<tr><td><code id="ggplot_associations_+3A_n_labels">n_labels</code></td>
<td>
<p>Number of labels of most associated points to display.</p>
</td></tr>
<tr><td><code id="ggplot_associations_+3A_nudge">nudge</code></td>
<td>
<p>Nudge parameter passed to ggrepel::geom_label_repel.</p>
</td></tr>
<tr><td><code id="ggplot_associations_+3A_linked_area">linked_area</code></td>
<td>
<p>Add a linked area to associations points, default FALSE</p>
</td></tr>
<tr><td><code id="ggplot_associations_+3A_byindex">byindex</code></td>
<td>
<p>Display by SNP index or chromosomic position (default)</p>
</td></tr>
<tr><td><code id="ggplot_associations_+3A_colors">colors</code></td>
<td>
<p>Colors of SNPs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>

<hr>
<h2 id='ggplot_ld'>Ggplot linkage disequilibrium</h2><span id='topic+ggplot_ld'></span>

<h3>Description</h3>

<p>Display SNP r2 correlations using points or diamonds with text.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_ld(
  df_ld,
  diamonds = length(unique(df_ld$SNP_A)) &lt; 40,
  point_size = 120/sqrt(nrow(df_ld)),
  reverse = FALSE,
  reindex = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_ld_+3A_df_ld">df_ld</code></td>
<td>
<p>Data frame with columns SNP_A, SNP_B, and R2.
As returned by the snprelate_ld function.</p>
</td></tr>
<tr><td><code id="ggplot_ld_+3A_diamonds">diamonds</code></td>
<td>
<p>Should the values be displayed as diamonds or points ?
Default is TRUE for less than 40 SNPs.</p>
</td></tr>
<tr><td><code id="ggplot_ld_+3A_point_size">point_size</code></td>
<td>
<p>Size for geom_point. Ignored if diamonds is TRUE.</p>
</td></tr>
<tr><td><code id="ggplot_ld_+3A_reverse">reverse</code></td>
<td>
<p>Reverse the display (horizontal symmetry)</p>
</td></tr>
<tr><td><code id="ggplot_ld_+3A_reindex">reindex</code></td>
<td>
<p>If FALSE, SNPs are positionned following their IDs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>

<hr>
<h2 id='ggplot_snp_pos'>Ggplot SNPs position</h2><span id='topic+ggplot_snp_pos'></span>

<h3>Description</h3>

<p>Get SNPs position ggplot with mappings to combine with other ggplots.
Optionally add labels and an upper subset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_snp_pos(
  df_snp,
  upper_subset = NULL,
  labels_colname = NULL,
  colors = snp_position_colors(nrow(df_snp))
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_snp_pos_+3A_df_snp">df_snp</code></td>
<td>
<p>SNP annotation data frame with a column named position
and, if specified, one named as the labels_colname
parameter.</p>
</td></tr>
<tr><td><code id="ggplot_snp_pos_+3A_upper_subset">upper_subset</code></td>
<td>
<p>Subset of df_snp for the positions on the upper side</p>
</td></tr>
<tr><td><code id="ggplot_snp_pos_+3A_labels_colname">labels_colname</code></td>
<td>
<p>Optional column name of df_snp to use as SNP labels.</p>
</td></tr>
<tr><td><code id="ggplot_snp_pos_+3A_colors">colors</code></td>
<td>
<p>Colors for each SNP</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>

<hr>
<h2 id='gtable_ld'>Gtable of linkage disequilibrium and chromosomic positions</h2><span id='topic+gtable_ld'></span>

<h3>Description</h3>

<p>Creates a gtable of linkage disequilibrium and chromosomic positions
ggplots. A biplot_subset parameter is available to add a second linkage
disequibrium ggplot to visualize the effect of a SNP selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtable_ld(
  df_ld,
  df_snp,
  biplot_subset = NULL,
  labels_colname = NULL,
  diamonds = length(unique(df_ld$SNP_A)) &lt; 40,
  point_size = ifelse(is.null(biplot_subset), 120, 80)/sqrt(nrow(df_ld)),
  title = "",
  title_biplot = "",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtable_ld_+3A_df_ld">df_ld</code></td>
<td>
<p>Data frame returned by snprelate_ld</p>
</td></tr>
<tr><td><code id="gtable_ld_+3A_df_snp">df_snp</code></td>
<td>
<p>SNP annotations with columns snpID and position</p>
</td></tr>
<tr><td><code id="gtable_ld_+3A_biplot_subset">biplot_subset</code></td>
<td>
<p>SNP indexes of the subset for the second ld plot</p>
</td></tr>
<tr><td><code id="gtable_ld_+3A_labels_colname">labels_colname</code></td>
<td>
<p>Column name of df_snp to use as SNP labels</p>
</td></tr>
<tr><td><code id="gtable_ld_+3A_diamonds">diamonds</code></td>
<td>
<p>Display the values as diamonds or as points
Default is TRUE for less than 40 SNPs.</p>
</td></tr>
<tr><td><code id="gtable_ld_+3A_point_size">point_size</code></td>
<td>
<p>Size for geom_point. Ignored if diamonds is TRUE.</p>
</td></tr>
<tr><td><code id="gtable_ld_+3A_title">title</code></td>
<td>
<p>Plot title</p>
</td></tr>
<tr><td><code id="gtable_ld_+3A_title_biplot">title_biplot</code></td>
<td>
<p>Optional biplot title</p>
</td></tr>
<tr><td><code id="gtable_ld_+3A_...">...</code></td>
<td>
<p>Passed to ggplot_ld</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gtable of ggplots
</p>

<hr>
<h2 id='gtable_ld_associations'>Gtable of linkage disequilibrium and associations</h2><span id='topic+gtable_ld_associations'></span>

<h3>Description</h3>

<p>Creates a gtable of a linkage disequilibrium, chromosomic positions, and
association scores ggplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtable_ld_associations(
  df_assocs,
  df_ld,
  pvalue_colname = "pvalues",
  labels_colname = "probe_id",
  n_labels = 5,
  diamonds = nrow(df_assocs) &lt;= 40,
  linked_area = diamonds,
  point_size = 150/nrow(df_assocs),
  colors = snp_position_colors(nrow(df_assocs)),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtable_ld_associations_+3A_df_assocs">df_assocs</code></td>
<td>
<p>SNP annotation data frame with columns chromosome,
position, and as specified by parameters pvalue_colname
and optionally labels_colname.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_df_ld">df_ld</code></td>
<td>
<p>Data frame with columns SNP_A, SNP_B, and R2, as
returned by the snprelate_ld function.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_pvalue_colname">pvalue_colname</code></td>
<td>
<p>Column name of df_snp with association values</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_labels_colname">labels_colname</code></td>
<td>
<p>Optional column name of df_snp with labels.
Set NULL to remove labels.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_n_labels">n_labels</code></td>
<td>
<p>Number of labels of most associated SNPs to display.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_diamonds">diamonds</code></td>
<td>
<p>Should the values be displayed as diamonds or points ?
Default is TRUE for up to 40 SNPs.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_linked_area">linked_area</code></td>
<td>
<p>Add a linked area to associations points.
Default same as diamonds.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_point_size">point_size</code></td>
<td>
<p>Point size for ggplot_ld, ignored if diamonds is TRUE.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_colors">colors</code></td>
<td>
<p>Colors of SNPs</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_+3A_...">...</code></td>
<td>
<p>Passed to ggplot_associations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gtable
</p>

<hr>
<h2 id='gtable_ld_associations_gdata'>Gtable of linkage disequilibrium and associations using a GenotypeData object</h2><span id='topic+gtable_ld_associations_gdata'></span>

<h3>Description</h3>

<p>Compute linkage disequilibrium using snprelate_ld on the set of SNPs in the
associations data frame and call gtable_ld_associations.
Creates a gtable of a linkage disequilibrium, chromosomic positions, and
association scores ggplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtable_ld_associations_gdata(
  df_assocs,
  gdata,
  pvalue_colname = "pvalues",
  labels_colname = "probe_id",
  diamonds = nrow(df_assocs) &lt;= 40,
  window = 15,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtable_ld_associations_gdata_+3A_df_assocs">df_assocs</code></td>
<td>
<p>SNP annotation data frame with columns chromosome,
position, and as specified by parameters pvalue_colname
and optionally labels_colname.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_gdata_+3A_gdata">gdata</code></td>
<td>
<p>GenotypeData object, as returned by
load_gds_as_genotype_data</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_gdata_+3A_pvalue_colname">pvalue_colname</code></td>
<td>
<p>Column name of df_snp with association values</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_gdata_+3A_labels_colname">labels_colname</code></td>
<td>
<p>Optional column name of df_snp with labels.
Set NULL to remove labels.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_gdata_+3A_diamonds">diamonds</code></td>
<td>
<p>Should the values be displayed as diamonds or points ?
Default is TRUE for up to 40 SNPs.</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_gdata_+3A_window">window</code></td>
<td>
<p>Window size for snprelate_ld.
Forced to the total number of SNPs if diamonds is FALSE</p>
</td></tr>
<tr><td><code id="gtable_ld_associations_gdata_+3A_...">...</code></td>
<td>
<p>Passed to gtable_ld_associations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gtable
</p>

<hr>
<h2 id='gtable_ld_gdata'>Gtable of linkage disequilibrium and positions using a GenotypeData object</h2><span id='topic+gtable_ld_gdata'></span>

<h3>Description</h3>

<p>Compute linkage disequilibrium using snprelate_ld on a set of SNP indexes and
call gtable_ld.
Two parameters are available to compute and compare minor allele frequency
filtering and TagSNP selection by displaying two LD plots with their
positions in the center.
The maf and r2 parameters are used similarly and as follows:
- compare baseline with MAF 5
gtable_ld(gdata, snps_idx, maf = 0.05)
- compare baseline with TagSNP r2 = 0.8
gtable_ld(gdata, snps_idx, r2 = 0.8)
- compare 5
gtable_ld(gdata, snps_idx, maf = c(0.05, 0.05), r2 = 0.8)
- compare MAF 5
gtable_ld(gdata, snps_idx, maf = c(0.05, 0.1), r2 = c(0.8, 0.6))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtable_ld_gdata(
  gdata,
  snps_idx,
  maf = NULL,
  r2 = NULL,
  diamonds = length(snps_idx) &lt; 40,
  window = 15,
  autotitle = TRUE,
  autotitle_bp = TRUE,
  double_title = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtable_ld_gdata_+3A_gdata">gdata</code></td>
<td>
<p>GenotypeData object returned by load_gds_as_genotype_data</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_snps_idx">snps_idx</code></td>
<td>
<p>SNPs indexes to select</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_maf">maf</code></td>
<td>
<p>Minor allele frequency threshold(s), see description</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_r2">r2</code></td>
<td>
<p>TagSNP r2 threshold(s), see description</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_diamonds">diamonds</code></td>
<td>
<p>Display the values as diamonds or as points
Default is TRUE for less than 40 SNPs.</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_window">window</code></td>
<td>
<p>Window size for snprelate_ld.
Forced to the total number of SNPs if diamonds is FALSE</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_autotitle">autotitle</code></td>
<td>
<p>Set title to feature selection method(s), number of SNPs
and chromosome</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_autotitle_bp">autotitle_bp</code></td>
<td>
<p>Set biplot title to feature selection method(s), number
of SNPs and chromosome</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_double_title">double_title</code></td>
<td>
<p>Logical, if false (default) keep only biplot title</p>
</td></tr>
<tr><td><code id="gtable_ld_gdata_+3A_...">...</code></td>
<td>
<p>Passed to gtable_ld</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gtable of ggplots
</p>

<hr>
<h2 id='load_gds_as_genotype_data'>Load GDS as Genotype Data</h2><span id='topic+load_gds_as_genotype_data'></span>

<h3>Description</h3>

<p>Open a connection to a snpgds file (cf. SNPRelate package) as a Genotype
Data object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_gds_as_genotype_data(
  gds_file,
  read_snp_annot = TRUE,
  read_scan_annot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_gds_as_genotype_data_+3A_gds_file">gds_file</code></td>
<td>
<p>Path of snpgds file</p>
</td></tr>
<tr><td><code id="load_gds_as_genotype_data_+3A_read_snp_annot">read_snp_annot</code></td>
<td>
<p>Read the SNPs' annotations</p>
</td></tr>
<tr><td><code id="load_gds_as_genotype_data_+3A_read_scan_annot">read_scan_annot</code></td>
<td>
<p>Read the scans' annotations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Genotype Data object
</p>

<hr>
<h2 id='parallel_apply'>Separate a matrix in a list of matrices of length the number of cores and
apply a function on the columns in parallel</h2><span id='topic+parallel_apply'></span>

<h3>Description</h3>

<p>Separate a matrix in a list of matrices of length the number of cores and
apply a function on the columns in parallel
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parallel_apply(m_data, apply_fun, n_cores = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parallel_apply_+3A_m_data">m_data</code></td>
<td>
<p>Data matrix</p>
</td></tr>
<tr><td><code id="parallel_apply_+3A_apply_fun">apply_fun</code></td>
<td>
<p>Function to apply</p>
</td></tr>
<tr><td><code id="parallel_apply_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of cores</p>
</td></tr>
<tr><td><code id="parallel_apply_+3A_...">...</code></td>
<td>
<p>Passed to apply_fun</p>
</td></tr>
</table>


<h3>Value</h3>

<p>apply_fun return
</p>

<hr>
<h2 id='print_qc_as_tex_table'>print_qc_as_tex_table</h2><span id='topic+print_qc_as_tex_table'></span>

<h3>Description</h3>

<p>Print information about quality control performed by the snprelate_qc
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_qc_as_tex_table(
  gdata_qc,
  label = "qc",
  caption = paste("Quality control and feature selection of the subset of the",
    "human genome diversity project dataset.")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_qc_as_tex_table_+3A_gdata_qc">gdata_qc</code></td>
<td>
<p>Genotype Data object object returned by snprelate_qc</p>
</td></tr>
<tr><td><code id="print_qc_as_tex_table_+3A_label">label</code></td>
<td>
<p>Label of the Tex table</p>
</td></tr>
<tr><td><code id="print_qc_as_tex_table_+3A_caption">caption</code></td>
<td>
<p>Caption of the Tex table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints knitr::kable object using cat
</p>

<hr>
<h2 id='save_hgdp_as_gds'>save_hgdp_as_gds</h2><span id='topic+save_hgdp_as_gds'></span>

<h3>Description</h3>

<p>Save the HGDP SNP data text file as a Genomic Data Structure file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save_hgdp_as_gds(paths = hgdp_filepaths(), outpath = tempfile(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save_hgdp_as_gds_+3A_paths">paths</code></td>
<td>
<p>Paths of the zip, txt, and gds files</p>
</td></tr>
<tr><td><code id="save_hgdp_as_gds_+3A_outpath">outpath</code></td>
<td>
<p>Output GDS file path</p>
</td></tr>
<tr><td><code id="save_hgdp_as_gds_+3A_...">...</code></td>
<td>
<p>Passed to save_genotype_data_as_gds</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path of the saved gds file
</p>

<hr>
<h2 id='select_region_idxs'>select_region_idxs</h2><span id='topic+select_region_idxs'></span>

<h3>Description</h3>

<p>Select SNP indexes corresponding to a specific genomic region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_region_idxs(
  gdata,
  chromosome,
  position_min = -Inf,
  position_max = Inf,
  n_snps = 0,
  offset = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_region_idxs_+3A_gdata">gdata</code></td>
<td>
<p>Genotype Data object</p>
</td></tr>
<tr><td><code id="select_region_idxs_+3A_chromosome">chromosome</code></td>
<td>
<p>Chromosome to select</p>
</td></tr>
<tr><td><code id="select_region_idxs_+3A_position_min">position_min</code></td>
<td>
<p>Minimum base pair position to select</p>
</td></tr>
<tr><td><code id="select_region_idxs_+3A_position_max">position_max</code></td>
<td>
<p>Maximum base pair position to select</p>
</td></tr>
<tr><td><code id="select_region_idxs_+3A_n_snps">n_snps</code></td>
<td>
<p>Maximum number of SNPs to return</p>
</td></tr>
<tr><td><code id="select_region_idxs_+3A_offset">offset</code></td>
<td>
<p>Number of SNPs to offset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SNP indexes of Genotype Data object
</p>

<hr>
<h2 id='snplinkage-package'>snplinkage: Single Nucleotide Polymorphisms Linkage Disequilibrium Visualizations</h2><span id='topic+snplinkage'></span><span id='topic+snplinkage-package'></span>

<h3>Description</h3>

<p>Linkage disequilibrium visualizations of up to several hundreds of single nucleotide polymorphisms (SNPs), annotated with chromosomic positions and gene names. Two types of plots are available for small numbers of SNPs (&lt;40) and for large numbers (tested up to 500). Both can be extended by combining other ggplots, e.g. association studies results, and functions enable to directly visualize the effect of SNP selection methods, as minor allele frequency filtering and TagSNP selection, with a second correlation heatmap. The SNPs correlations are computed on Genotype Data objects from the 'GWASTools' package using the 'SNPRelate' package, and the plots are customizable 'ggplot2' and 'gtable' objects and are annotated using the 'biomaRt' package. Usage is detailed in the vignette with example data and results from up to 500 SNPs of 1,200 scans are in Charlon T. (2019) <a href="https://doi.org/10.13097/archive-ouverte/unige%3A161795">doi:10.13097/archive-ouverte/unige:161795</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Thomas Charlon <a href="mailto:charlon@protonmail.com">charlon@protonmail.com</a> (<a href="https://orcid.org/0000-0001-7497-0470">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Karl Forner
</p>
</li>
<li><p> Alessandro Di Cara
</p>
</li>
<li><p> Jérôme Wojcik
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://gitlab.com/thomaschln/snplinkage">https://gitlab.com/thomaschln/snplinkage</a>
</p>
</li>
<li><p> Report bugs at <a href="https://gitlab.com/thomaschln/snplinkage/-/issues">https://gitlab.com/thomaschln/snplinkage/-/issues</a>
</p>
</li></ul>


<hr>
<h2 id='snprelate_allele_frequencies'>Compute allele frequencie and snp missing rate</h2><span id='topic+snprelate_allele_frequencies'></span>

<h3>Description</h3>

<p>Wrapper over SNPRelate::snpgdsSNPRateFreq
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snprelate_allele_frequencies(
  gdata,
  snps_idx = NULL,
  scans_idx = NULL,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snprelate_allele_frequencies_+3A_gdata">gdata</code></td>
<td>
<p>A GenotypeData object</p>
</td></tr>
<tr><td><code id="snprelate_allele_frequencies_+3A_snps_idx">snps_idx</code></td>
<td>
<p>Vector of snps indices</p>
</td></tr>
<tr><td><code id="snprelate_allele_frequencies_+3A_scans_idx">scans_idx</code></td>
<td>
<p>Vector of scans indices</p>
</td></tr>
<tr><td><code id="snprelate_allele_frequencies_+3A_quiet">quiet</code></td>
<td>
<p>Whether to be quiet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of snps_idx, snps_ids, allele1, allele2, maf, missing
where allele1 and allele2 are the rates of the alleles, and maf the minimum
of the 2. Missing is the missing rate. N.B: the allele rates are computed
on the non missing genotypes, i.e. their sum equals 1.
</p>

<hr>
<h2 id='snprelate_ld'>Wrapper for snpgdsLDMat to compute r2</h2><span id='topic+snprelate_ld'></span>

<h3>Description</h3>

<p>Wrapper for snpgdsLDMat to compute r2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snprelate_ld(
  gdata,
  window_size = 0,
  min_r2 = 0,
  snps_idx = NULL,
  scans_idx = NULL,
  threads = 1,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snprelate_ld_+3A_gdata">gdata</code></td>
<td>
<p>A GenotypeData object</p>
</td></tr>
<tr><td><code id="snprelate_ld_+3A_window_size">window_size</code></td>
<td>
<p>Max number of SNPs in LD window, 0 for no window</p>
</td></tr>
<tr><td><code id="snprelate_ld_+3A_min_r2">min_r2</code></td>
<td>
<p>Minimum r2 value to report</p>
</td></tr>
<tr><td><code id="snprelate_ld_+3A_snps_idx">snps_idx</code></td>
<td>
<p>Indices of snps to use</p>
</td></tr>
<tr><td><code id="snprelate_ld_+3A_scans_idx">scans_idx</code></td>
<td>
<p>Indices of scans to use</p>
</td></tr>
<tr><td><code id="snprelate_ld_+3A_threads">threads</code></td>
<td>
<p>The number of threads to use</p>
</td></tr>
<tr><td><code id="snprelate_ld_+3A_quiet">quiet</code></td>
<td>
<p>Whether to be quiet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns SNP_A, SNP_B, R2 for r2 &gt;= min_r2
</p>

<hr>
<h2 id='snprelate_ld_select'>Wrapper for snpgdsLDpruning to select Tag SNPs</h2><span id='topic+snprelate_ld_select'></span>

<h3>Description</h3>

<p>The tagged snp set is (by sliding window) representative and strongly not
redundant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snprelate_ld_select(
  gdata,
  window_length = 500L,
  min_r2,
  window_size = NA,
  snps_idx = NULL,
  scans_idx = NULL,
  remove.monosnp = FALSE,
  autosome.only = FALSE,
  method = "r",
  threads = 1,
  quiet = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snprelate_ld_select_+3A_gdata">gdata</code></td>
<td>
<p>A GenotypeData object</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_window_length">window_length</code></td>
<td>
<p>Max length in kb of the window</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_min_r2">min_r2</code></td>
<td>
<p>Minimum r2 value to report</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_window_size">window_size</code></td>
<td>
<p>Max number of SNPs in LD window</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_snps_idx">snps_idx</code></td>
<td>
<p>Indices of snps to use</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_scans_idx">scans_idx</code></td>
<td>
<p>Indices of scans to use</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_remove.monosnp">remove.monosnp</code></td>
<td>
<p>if TRUE, remove monomorphic SNPs</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_autosome.only">autosome.only</code></td>
<td>
<p>if <code>TRUE</code>, use autosomal SNPs only; if it is a
numeric or character value, keep SNPs according to the specified
chromosome</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_method">method</code></td>
<td>
<p>&quot;composite&quot;, &quot;r&quot;, &quot;dprime&quot;, &quot;corr&quot;, see details</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_threads">threads</code></td>
<td>
<p>The number of threads to use, currently ignored</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_quiet">quiet</code></td>
<td>
<p>Whether to be quiet</p>
</td></tr>
<tr><td><code id="snprelate_ld_select_+3A_...">...</code></td>
<td>
<p>Forwarded to SNPRelate::snpgdsLDpruning</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of SNP IDs stratified by chromosomes.
</p>

<hr>
<h2 id='snprelate_qc'>snprelate_qc</h2><span id='topic+snprelate_qc'></span>

<h3>Description</h3>

<p>Quality control using SNPRelate functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snprelate_qc(
  gdata,
  samples_nas = 0.03,
  ibs = 0.99,
  keep_ids = NULL,
  snps_nas = 0.01,
  maf = 0.05,
  tagsnp = 0.8,
  n_cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snprelate_qc_+3A_gdata">gdata</code></td>
<td>
<p>Genotype data object</p>
</td></tr>
<tr><td><code id="snprelate_qc_+3A_samples_nas">samples_nas</code></td>
<td>
<p>NA threshold for samples, default 3 pct</p>
</td></tr>
<tr><td><code id="snprelate_qc_+3A_ibs">ibs</code></td>
<td>
<p>Samples identity by state threshold, default 99 pct</p>
</td></tr>
<tr><td><code id="snprelate_qc_+3A_keep_ids">keep_ids</code></td>
<td>
<p>Samples ids to keep even if IBS is higher than threshold.
Used for monozygotic twins.</p>
</td></tr>
<tr><td><code id="snprelate_qc_+3A_snps_nas">snps_nas</code></td>
<td>
<p>NA threshold for SNPs, default 1 pct</p>
</td></tr>
<tr><td><code id="snprelate_qc_+3A_maf">maf</code></td>
<td>
<p>Minor allele frequency threshold, default 5 pct</p>
</td></tr>
<tr><td><code id="snprelate_qc_+3A_tagsnp">tagsnp</code></td>
<td>
<p>TagSNP r2 correlation threshold, default 0.8</p>
</td></tr>
<tr><td><code id="snprelate_qc_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of cores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of gdata, Genotype data object, and df_qc, QC info data frame
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
