<!DOCTYPE html><html><head><title>Help for package snpReady</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {snpReady}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#G.matrix'><p>Estimation of Genomic Relationship Matrix</p></a></li>
<li><a href='#maize.hyb'><p>maize hybrids</p></a></li>
<li><a href='#maize.line'><p>maize lines</p></a></li>
<li><a href='#popgen'><p>Population genetics from genomic data</p></a></li>
<li><a href='#raw.data'><p>Preparation of genomic data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.9.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-04-11</td>
</tr>
<tr>
<td>Title:</td>
<td>Preparing Genotypic Datasets in Order to Run Genomic Analysis</td>
</tr>
<tr>
<td>Description:</td>
<td>Three functions to clean, summarize and prepare genomic datasets to Genome
    Selection and Genome Association analysis and to estimate population genetic parameters.</td>
</tr>
<tr>
<td>Depends:</td>
<td>Matrix, matrixcalc, stringr, rgl, impute</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, reshape2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-04-11 12:58:20 UTC; Italo Granato</td>
</tr>
<tr>
<td>Author:</td>
<td>Italo Granato [aut, cre],
  Roberto Fritsche-Neto [aut, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Italo Granato &lt;italo.granato@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-04-11 13:11:48 UTC</td>
</tr>
</table>
<hr>
<h2 id='G.matrix'>Estimation of Genomic Relationship Matrix</h2><span id='topic+G.matrix'></span>

<h3>Description</h3>

<p>It generates four different types of Genomic Relationship Matrix (GRM)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>G.matrix(M, method=c("VanRaden", "UAR", "UARadj", "GK"), format=c("wide", "long"), 
         plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="G.matrix_+3A_m">M</code></td>
<td>
<p><code>matrix</code>. Matrix of markers in which <code class="reqn">n</code> individuals are in rows and <code class="reqn">p</code> markers in columns. This matrix do not need to be centered.</p>
</td></tr>
<tr><td><code id="G.matrix_+3A_method">method</code></td>
<td>
<p>Method to built the GRM. Four methods are currently supported. <code>"VanRaden"</code> indicates the method proposed by Vanraden (2008) for additive
genomic relationship and its counterpart for dominance genomic relationship. <code>"UAR"</code> and <code>"UARadj"</code> are methods proposed by Yang et al. (2010) for additive genomic relationship. <code>"GK"</code> represents the Gaussian kernel for additive genomic. See <code>Details</code></p>
</td></tr>
<tr><td><code id="G.matrix_+3A_format">format</code></td>
<td>
<p>Type of object to be returned. <code>wide</code> returns a <code class="reqn">n \times n</code> matrix. <code>long</code> returns the low diagonal from GRM as a table with 3 columns. See <code>Details</code></p>
</td></tr>
<tr><td><code id="G.matrix_+3A_plot">plot</code></td>
<td>
<p>If <code>TRUE</code>, a graphical output is produced. See <code>Details</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>G.matrix provides four different types of relationship matrix. The <code>VanRaden</code> represents the relationship matrix estimated as proposed by Vanraden (2008):
</p>
<p style="text-align: center;"><code class="reqn">G = \frac{XX'}{trace(XX')/n}</code>
</p>

<p><code class="reqn">X</code> is the centered marker matrix. For any marker locus <code class="reqn">i</code>, <code class="reqn">x_i = m_i - 2p_{i}</code> where <code class="reqn">m_i</code> is the vector of SNP genotypes coded as allele couting (0, 1 and 2).
</p>
<p><code>UAR</code> is the genomic relationship matrices proposed by Yang et al. (2010) and named as Unified Additive Relationship (UAR). This matrix is then obtained by
</p>
<p style="text-align: center;"><code class="reqn">
G_{UAR} = A_{jk} = \frac{1}{N} \sum_i{A_{ijk}} = \left \{
\begin{array}{ll}
\frac{1}{N} \sum_i{\frac{(x_{ij} - 2p_{i})(x_{ik} - 2p_i)}{2p_i(1-p_i)}}, j \neq k \\
1 +  \frac{1}{N} \sum_i{\frac{x_{ij}^{2}(1 + 2p_{i})x_{ij} + 2p_i^{2}}{2p_i(1-p_i)}}, j = k
\end{array}
\right.
</code>
</p>
  
<p>where <code class="reqn">p_i</code> is the allele frequency at SNP <code class="reqn">i</code> and <code class="reqn">x_{ij}</code> is the SNP genotype that takes a value of 0, 1 or 2 for the genotype of the <code class="reqn">j^{th}</code> individual at SNP <code class="reqn">i</code>.
The same authors proposed an adjustment in the original UAR matrix (<code>UARadj</code>) to reduce the bias in estimation of variance in the relationship in causal loci. Thus:
</p>
<p style="text-align: center;"><code class="reqn">
G_{UARadj} = \left \{
\begin{array}{ll}
 \beta A_{jk}, j \neq k \\
1 + \beta(A_{jk} - 1), j = k
\end{array}
\right.
</code>
</p>
  
<p>where <code class="reqn">\beta = 1 - frac{c + 1/N}{var(A_{jk}}</code>, c is a constant dependent on MAF of causal variants assumed as 0. 
<code>GK</code> represents the Gaussian kernel, obtained by
</p>
<p style="text-align: center;"><code class="reqn"> K (x_i, x_{i'}) = \frac{exp(-d_{ii'}^2)}{quantile(d^2, 0.5)}</code>
</p>

<p>where <code class="reqn">d_{ii'}^2</code> is the square of euclidian distance between two individuals
The <code>format</code> argument is the desired output format. For <code>"wide"</code>, the relationship output produced is in matrix format, with <code class="reqn">n \times n</code> dimension. 
If <code>"long"</code> is the chosen format, the inverse of the relationship matrix is computed and converted to a table. In this case, the low triangular part of the relationship matrix
is changed to a table with three columns representing the respective rows, columns, and values (Used mainly by ASReml)
</p>
<p>If the relationship matrix is not positive definite, a near positive definite matrix is created and solved, followed by a warning message.
</p>


<h3>Value</h3>

<p>It returns the GRM. If the method is <code>VanRaden</code>, additive and dominance matrices are produced. Otherwise, only the additive form. 
If <code>plot</code> is <code>TRUE</code> a heat map of the pairwise relationship is save as pdf into the working directory . Also, a 3D plot with the three first principal components is generated. 
</p>


<h3>References</h3>

<p>Pérez-Elizalde, S.,Cuevas, J.; Pérez-Rodríguez, P.; Crossa, J. (2015) Selection of The Bandwidth Parameter in a Bayesian Kernel Regression Model for Genomic-Enabled Prediction. J Agr Biol Envir S, 20-4:512-532
</p>
<p>Yang, J., Benyamin, B., McEvoy, B.P., et al (2010) Common SNPs explain a large proportion of the heritability for human height. Nature Genetics 42:565-569
</p>
<p>VanRaden, P.M. (2008) Efficient Methods to Compute Genomic Predictions. Journal of Dairy Science, 91:4414-4423
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#(1) Additive and dominance relationship matrix 
data(maize.hyb)
x &lt;- G.matrix(maize.hyb, method = "VanRaden", format = "wide")
A &lt;- x$Ga
D &lt;- x$Gd
</code></pre>

<hr>
<h2 id='maize.hyb'>maize hybrids</h2><span id='topic+maize.hyb'></span>

<h3>Description</h3>

<p>50 hybrids of maize genotyped with 492 SNP markers
</p>


<h3>Format</h3>

<p>A matrix with 50 rows and 492 columns, where hybrids are in rows and SNP markers in columns</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' data(maize.hyb)


</code></pre>

<hr>
<h2 id='maize.line'>maize lines</h2><span id='topic+maize.line'></span>

<h3>Description</h3>

<p>A raw dataset of maize lines genotyped with 768 markers
</p>


<h3>Format</h3>

<p>A matrix with 70656 observations on the following 4 variables.
</p>
 
<ul>
<li><p>sample: identification of samples (name of individuals)
</p>
</li>
<li><p>marker: identification of SNP markers
</p>
</li>
<li><p>allele.1: Allele 1
</p>
</li>
<li><p>allele.2: Allele 2</p>
</li></ul>


<h3>Source</h3>

<p>Lines genotyped from allogamous breeding laboratory - ESALQ/USP <a href="http://www.genetica.esalq.usp.br/alogamas/index2.html">http://www.genetica.esalq.usp.br/alogamas/index2.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(maize.line)
## str(maize.line)

</code></pre>

<hr>
<h2 id='popgen'>Population genetics from genomic data</h2><span id='topic+popgen'></span>

<h3>Description</h3>

<p>Allows for estimating parameters of population genetics from genomic data. Besides, it also allows the estimate of same parameters considering subpopulations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>popgen(M, subgroups, plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="popgen_+3A_m">M</code></td>
<td>
<p>Object of class <code>matrix</code>. A (non-empty) matrix of molecular markers, considering the count of reference alleles per loci (0, 1 or 2). Markers must be in columns and individuals in rows. Missing data should be assigned as <code>NA</code></p>
</td></tr>
<tr><td><code id="popgen_+3A_subgroups">subgroups</code></td>
<td>
<p>A <code>vector</code> with information for subgroups or subpopulations.</p>
</td></tr>
<tr><td><code id="popgen_+3A_plot">plot</code></td>
<td>
<p>If <code>TRUE</code>, a graphical output is produced. See <code>details</code> </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The number of subgroups is defined by the user and accepts any data type (<code>character</code>, <code>integer</code> ...) to distinguish subpopulations.
These two inputs must have the same sort for rows (genotypes).
</p>


<h3>Value</h3>

<p>Two-level lists are returned (<code>whole</code> and <code>bygroup</code>), one with general information for markers and individuals and another by subgroups (if applicable).
</p>
<p>For <code>whole</code>, a list containing estimates parameters for
</p>

<dl>
<dt>$Markers</dt><dd><p>For each marker it presents the allelic frequency (<code class="reqn">p</code> and <code class="reqn">q</code>),
Minor Allele Frequency (<code class="reqn">MAF</code>), expected heterozygosity (<code class="reqn">H_e</code>), observed
heterozygosity (<code class="reqn">H_o</code>), Nei's Genetic Diversity (<code class="reqn">DG</code>) and Polymorphism Informative Content(<code class="reqn">PIC</code>), proportion of missing (<code class="reqn">Miss</code>), <code class="reqn">\chi^2</code> statistic for the Hardy-Weinberg equilibrium test and its pvalue</p>
</dd> 
<dt>$Genotypes</dt><dd><p>It presents observed heterozygosity (<code class="reqn">H_o</code>) and coefficient of inbreeding (<code class="reqn">F_i</code>) estimated as excess of homozygous relative to the expected (Keller et al. (2011))</p>
</dd>
<dt>$Population</dt><dd><p>The same parameters as those for markers except PIC are returned for general population along with lower and upper boundaries</p>
</dd>
<dt>$Variability</dt><dd><p>shows estimates of effective population size (<code class="reqn">Ne</code>), additive (<code class="reqn">Va</code>) and dominance (<code class="reqn">Vd</code>) variances components, and a summary of number of groups, genotypes and markers</p>
</dd>
</dl>
<p>In the presence of <code>subgroups</code>, the same populational parameters are estimated considering each subpopulation accompanied by its exclusive and fixed alleles. Moreover, a list with the F-statistics (<code class="reqn">F_IT</code>, <code class="reqn">F_IS</code> and <code class="reqn">F_ST</code>) for genotypes and markers are exhibited. For genotypes, it shows the statistics considering all subpopulations and a pairwise framework, and for markers loci, the parameters are presented only considering all subpopulations.
</p>
<p>The <code>plot</code> produces a histogram for the estimates of <code class="reqn">MAF</code>, <code class="reqn">GD</code>, <code class="reqn">PIC</code> and <code class="reqn">He</code> considering the whole population and subpopulations, when it is available. Also, a heat map of the pairwise <code class="reqn">F_ST</code> between populations is produced.
</p>



<h3>References</h3>

<p>Weir, B.S. and C.C. Cockerham. (1984). Estimating F-Statistics for the Analysis of Population Structure. Evolution 38: 1358-1370. doi:10.2307/2408641.
</p>
<p>Keller M.C., Visscher P.M., Goddard M.E. (2011) Quantification of inbreeding due to distant ancestors and its detection using dense single nucleotide polymorphism data. Genetics 189:237-249. doi: 10.1534/genetics.111.130922
</p>


<h3>Examples</h3>

<pre><code class='language-R'># hybrid maize data
data(maize.hyb)
x &lt;- popgen(maize.hyb) 

# using subpopulations
PS&lt;-c(rep(1,25), rep(2,25))
x &lt;- popgen(maize.hyb, subgroups=PS)
</code></pre>

<hr>
<h2 id='raw.data'>Preparation of genomic data</h2><span id='topic+raw.data'></span>

<h3>Description</h3>

<p>This function gets genomic data ready to be used in packages or softwares that
perform genomic predictions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>raw.data(data, frame = c("long","wide"), hapmap=NULL, base=TRUE, sweep.sample=1,
        call.rate=0.95, maf=0.05, imput=TRUE, imput.type=c("wright", "mean","knni"), 
        outfile=c("012","-101","structure"), plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="raw.data_+3A_data">data</code></td>
<td>
<p>object of class <code>matrix</code></p>
</td></tr>
<tr><td><code id="raw.data_+3A_frame">frame</code></td>
<td>
<p>Format of genomic data to be imputed. Two formats are currently supported. <code>"long"</code> is used for objects
with sample ID (1st column), marker ID (2nd column), fist allele (3rd column) and second allele (4th column). While <code>"wide"</code> is for objects with dimension <code class="reqn">n \times m</code> where markers must be in columns and individuals in rows</p>
</td></tr>
<tr><td><code id="raw.data_+3A_hapmap">hapmap</code></td>
<td>
<p><code>matrix</code>. Object with information on SNPs, chromosome and position</p>
</td></tr>
<tr><td><code id="raw.data_+3A_base">base</code></td>
<td>
<p><code>logical</code>. Are genotypes coded as nitrogenous bases? if <code>TRUE</code>, data are converted to numeric. If <code>FALSE</code>, it follows to clean up</p>
</td></tr>
<tr><td><code id="raw.data_+3A_sweep.sample">sweep.sample</code></td>
<td>
<p><code>numeric</code>. Threshold for removing samples from data by missing rate. Samples with missing rate above the defined threshold are
removed from dataset</p>
</td></tr>
<tr><td><code id="raw.data_+3A_call.rate">call.rate</code></td>
<td>
<p><code>numeric</code>. Threshold for removing marker by missing genotype rate. SNP with <code>call rate</code> below threshold are removed from dataset. Must be between 0, 1</p>
</td></tr>
<tr><td><code id="raw.data_+3A_maf">maf</code></td>
<td>
<p>Threshold for removing SNP by minor allele frequency. Must be between 0, 1</p>
</td></tr>
<tr><td><code id="raw.data_+3A_imput">imput</code></td>
<td>
<p>Should imputation of missing data be performed?. Default is <code>TRUE</code> </p>
</td></tr>
<tr><td><code id="raw.data_+3A_imput.type">imput.type</code></td>
<td>
<p>Type of imputation. It can be &quot;wright&quot;, &quot;mean&quot; or &quot;knni&quot;. See <code>details</code> </p>
</td></tr>
<tr><td><code id="raw.data_+3A_outfile">outfile</code></td>
<td>
<p><code>character</code>. Type of output to be produced. <code>"012"</code> outputs matrix coded as 0 to <code>AA</code>, 1 to <code>Aa</code> and 2 to <code>aa</code>. <code>"-101"</code>
presents marker matrix coded as -1, 0 and 1 to <code>aa</code>, <code>Aa</code> and <code>AA</code>, respectively. <code>"structure"</code> returns a matrix suitable for STRUCTURE Software.
For this, each individual is splited in two rows, one for each allele. Nitrogenous bases are then recoded to a specific number, so A is 1, C is 2, G is 3 and T is 4. This format is only acceptable if base are <code>TRUE</code></p>
</td></tr>
<tr><td><code id="raw.data_+3A_plot">plot</code></td>
<td>
<p>If <code>TRUE</code>, a graphical output of quality control is produced. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function allows flexible input of genomic data. Data might be in long format with 4 columns or in wide 
format where markers are in columns and individuals in rows. Both numeric and nitrogenous bases are accepted. Samples and markers can be eliminated based on missing data rate. Markers can also be eliminated based on
the frequency of the minor allele.  
Three methods of imputation are currently implemented. One is carried out through combination of allele frequency and individual observed heterozygosity estimated from markers. 
</p>
<p style="text-align: center;"><code class="reqn"> 
p(x_{ij}) = \left \{
          \begin{array}{ll}
          0 = (1 - p_j)^2 + p_j (1 - p_j) F_i \\
          1 = 2 p_j (1 - p_j) - 2 p_j (1 - p_j) F_i\\
          2 = p_j^2 + p_j (1 - p_j) F_i        
                \end{array}
              \right.
</code>
</p>

<p>Hence, for missing values, genotypes are imputed based on their probability of occurrence. This probability depends both on genotype frequency and inbreeding of the individual a specific locus.
The second method is based on mean of SNP. Thus, each missing point in a SNP <code class="reqn">j</code> is replaced by mean of SNP <code class="reqn">j</code>  
</p>
<p style="text-align: center;"><code class="reqn">x_{ij} = 2p_j</code>
</p>

<p>The &quot;knni&quot; imputes missing markers using the mean of the k-nearest markers. Nearest markers are found by computing the Euclidian distance between markers. If you use this option, please refer to the package impute (Hastie et al. 2017) in publications.
</p>


<h3>Value</h3>

<p>Returns a properly coded marker matrix output and a report specifying which individuals are removed by <code>sweep.sample</code> and which markers are removed by <code>"call.rate"</code>
and <code>maf</code>.
Also, a plot with proportion of removed markers and imputed data, for each chromosome, when the map is included, is produced when <code>plot</code> is TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(maize.line)
M &lt;- as.matrix(maize.line)
mrc &lt;- raw.data(M, frame="long", base=TRUE, sweep.sample= 0.8, 
         call.rate=0.95, maf=0.05, imput=FALSE, outfile="-101")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
