<!DOCTYPE html><html lang="en"><head><title>Help for package snpsettest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {snpsettest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#snpsettest-package'><p>snpsettest: A Set-Based Association Test using GWAS Summary Statistics</p></a></li>
<li><a href='#exGWAS'><p>An example file of GWAS summary statistics</p></a></li>
<li><a href='#gene.curated.GRCh37'><p>Human gene information from the GENCODE GRCh37 version</p></a></li>
<li><a href='#gene.curated.GRCh38'><p>Human gene information from the GENCODE GRCh38 version</p></a></li>
<li><a href='#harmonize_sumstats'><p>Harmonizing GWAS summary to reference data</p></a></li>
<li><a href='#map_snp_to_gene'><p>Map SNPs to genes</p></a></li>
<li><a href='#read_reference_bed'><p>Read a PLINK bed file for reference data</p></a></li>
<li><a href='#snpset_test'><p>Set-based association tests</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>A Set-Based Association Test using GWAS Summary Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>The goal of 'snpsettest' is to provide simple tools that perform
             set-based association tests (e.g., gene-based association tests)
             using GWAS (genome-wide association study) summary statistics. A
             set-based association test in this package is based on the
             statistical model described in VEGAS (versatile gene-based
             association study), which combines the effects of a set of SNPs
             accounting for linkage disequilibrium between markers. This package
             uses a different approach from the original VEGAS implementation to
             compute set-level p values more efficiently, as described in
             <a href="https://github.com/HimesGroup/snpsettest/wiki/Statistical-test-in-snpsettest">https://github.com/HimesGroup/snpsettest/wiki/Statistical-test-in-snpsettest</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>gaston, data.table, Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tidyr, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/HimesGroup/snpsettest">https://github.com/HimesGroup/snpsettest</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HimesGroup/snpsettest/issues">https://github.com/HimesGroup/snpsettest/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-09 17:04:05 UTC; jaehyun</td>
</tr>
<tr>
<td>Author:</td>
<td>Jaehyun Joo [aut, cre],
  Blanca Himes [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jaehyun Joo &lt;jaehyunjoo@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-09 17:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='snpsettest-package'>snpsettest: A Set-Based Association Test using GWAS Summary Statistics</h2><span id='topic+snpsettest'></span><span id='topic+snpsettest-package'></span>

<h3>Description</h3>

<p>The goal of 'snpsettest' is to provide simple tools that perform set-based association tests (e.g., gene-based association tests) using GWAS (genome-wide association study) summary statistics. A set-based association test in this package is based on the statistical model described in VEGAS (versatile gene-based association study), which combines the effects of a set of SNPs accounting for linkage disequilibrium between markers. This package uses a different approach from the original VEGAS implementation to compute set-level p values more efficiently, as described in <a href="https://github.com/HimesGroup/snpsettest/wiki/Statistical-test-in-snpsettest">https://github.com/HimesGroup/snpsettest/wiki/Statistical-test-in-snpsettest</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Jaehyun Joo <a href="mailto:jaehyunjoo@outlook.com">jaehyunjoo@outlook.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Blanca Himes <a href="mailto:bhimes@pennmedicine.upenn.edu">bhimes@pennmedicine.upenn.edu</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/HimesGroup/snpsettest">https://github.com/HimesGroup/snpsettest</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/HimesGroup/snpsettest/issues">https://github.com/HimesGroup/snpsettest/issues</a>
</p>
</li></ul>


<hr>
<h2 id='exGWAS'>An example file of GWAS summary statistics</h2><span id='topic+exGWAS'></span>

<h3>Description</h3>

<p>An example file of GWAS summary statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exGWAS
</code></pre>


<h3>Format</h3>

<p>Data frame with columns
</p>

<dl>
<dt>id</dt><dd><p>SNP ID.</p>
</dd>
<dt>chr</dt><dd><p>chromosome.</p>
</dd>
<dt>pos</dt><dd><p>base-pair position.</p>
</dd>
<dt>A1, A2</dt><dd><p>allele codes.</p>
</dd>
<dt>pvalue</dt><dd><p>p value.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>head(exGWAS)
</code></pre>

<hr>
<h2 id='gene.curated.GRCh37'>Human gene information from the GENCODE GRCh37 version</h2><span id='topic+gene.curated.GRCh37'></span>

<h3>Description</h3>

<p>Human gene information was extracted from the GENCODE release 19. This data
only contains 'KNOWN' status genes with the following gene biotypes:
protein-coding, Immunoglobulin (Ig) variable chain and T-cell receptor (TcR)
genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene.curated.GRCh37
</code></pre>


<h3>Format</h3>

<p>Data frame with columns
</p>

<dl>
<dt>gene.id</dt><dd><p>SNP ID.</p>
</dd>
<dt>chr</dt><dd><p>chromosome.</p>
</dd>
<dt>start</dt><dd><p>genomic start location (1-based).</p>
</dd>
<dt>end</dt><dd><p>genomic end location.</p>
</dd>
<dt>strand</dt><dd><p>genomic strand.</p>
</dd>
<dt>gene.name</dt><dd><p>gene symbols mapped to the GENCODE genes.</p>
</dd>
<dt>gene.type</dt><dd><p>gene biotypes in the GENCODE genes.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.gencodegenes.org/human/release_19.html">https://www.gencodegenes.org/human/release_19.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(gene.curated.GRCh37)
</code></pre>

<hr>
<h2 id='gene.curated.GRCh38'>Human gene information from the GENCODE GRCh38 version</h2><span id='topic+gene.curated.GRCh38'></span>

<h3>Description</h3>

<p>Human gene information was extracted from the GENCODE release 37. This data
only contains genes with the following gene biotypes: protein-coding,
Immunoglobulin (Ig) variable chain and T-cell receptor (TcR) genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene.curated.GRCh38
</code></pre>


<h3>Format</h3>

<p>Data frame with columns
</p>

<dl>
<dt>gene.id</dt><dd><p>SNP ID.</p>
</dd>
<dt>chr</dt><dd><p>chromosome.</p>
</dd>
<dt>start</dt><dd><p>genomic start location (1-based).</p>
</dd>
<dt>end</dt><dd><p>genomic end location.</p>
</dd>
<dt>strand</dt><dd><p>genomic strand.</p>
</dd>
<dt>gene.name</dt><dd><p>gene symbols mapped to the GENCODE genes.</p>
</dd>
<dt>gene.type</dt><dd><p>gene biotypes in the GENCODE genes.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.gencodegenes.org/human/release_37.html">https://www.gencodegenes.org/human/release_37.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(gene.curated.GRCh38)
</code></pre>

<hr>
<h2 id='harmonize_sumstats'>Harmonizing GWAS summary to reference data</h2><span id='topic+harmonize_sumstats'></span>

<h3>Description</h3>

<p>Finds an intersection of variants between GWAS summary and reference data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harmonize_sumstats(
  sumstats,
  x,
  match_by_id = TRUE,
  check_strand_flip = FALSE,
  return_indice = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="harmonize_sumstats_+3A_sumstats">sumstats</code></td>
<td>
<p>A data frame with two columns: &quot;id&quot; and &quot;pvalue&quot;.
</p>

<ul>
<li><p> id = SNP ID (e.g., rs numbers)
</p>
</li>
<li><p> pvalue = SNP-level p value
</p>
</li></ul>

<p>If <code>match_by_id = FALSE</code>, it requires additional columns: &quot;chr&quot;, &quot;pos&quot;, &quot;A1&quot;
and &quot;A2&quot;.
</p>

<ul>
<li><p> chr =  chromosome
</p>
</li>
<li><p> pos =  base-pair position (must be integer)
</p>
</li>
<li><p> A1, A2 = allele codes (allele order is not important)
</p>
</li></ul>
</td></tr>
<tr><td><code id="harmonize_sumstats_+3A_x">x</code></td>
<td>
<p>A <code>bed.matrix</code> object created using the reference data.</p>
</td></tr>
<tr><td><code id="harmonize_sumstats_+3A_match_by_id">match_by_id</code></td>
<td>
<p>If <code>TRUE</code>, SNP matching will be performed by SNP IDs
instead of genomic position and allele codes. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="harmonize_sumstats_+3A_check_strand_flip">check_strand_flip</code></td>
<td>
<p>Only applies when <code>match_by_id = FALSE</code>. If <code>TRUE</code>,
the function 1) removes ambiguous A/T and G/C SNPs for which the strand is
not obvious, and 2) attempts to find additional matching entries by
flipping allele codes (i.e., A-&gt;T, T-&gt;A, C-&gt;G, G-&gt;A). If the GWAS genotype
data itself is used as the reference data, it would be safe to set
<code>FALSE</code>. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="harmonize_sumstats_+3A_return_indice">return_indice</code></td>
<td>
<p>Only applied when <code>match_by_id = FALSE</code>. If <code>TRUE</code>, the
function provides an additional column indicating whether the match is
with swapped alleles. If <code>check_strand_flip = TRUE</code>, the function also
provides an additional column indicating whether the match is with flipped
strand. Unnecessary for gene-based tests in this package, but may be
useful for other purposes (e.g., harmonization for meta-analysis that
needs to flip the sign of beta for a match with swapped alleles).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pre-processing of GWAS summary data is required because the sets of variants
available in a particular GWAS might be poorly matched to the variants in
reference data. SNP matching can be performed either 1) by SNP ID or 2) by
chromosome code, base-pair position, and allele codes, while taking into
account possible strand flips and reference allele swap. For matched
entries, the SNP IDs in GWAS summary data are replaced with the ones in the
reference data.
</p>


<h3>Value</h3>

<p>A data frame with columns: &quot;id&quot;, &quot;chr&quot;, &quot;pos&quot;, &quot;A1&quot;, &quot;A2&quot; and
&quot;pvalue&quot;. If <code>return_indice = TRUE</code>, the data frame includes additional
columns <code>key_</code>, <code>swapped_</code>, and <code>flipped_</code>. <code>key_</code> is &quot;chr_pos_A1_A2&quot; in
<code>sumstat</code> (the original input before harmonization). <code>swapped_</code> contains a
logical vector indicating reference allele swap. <code>flipped_</code> contains a
logical vector indicating strand flip.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## GWAS summary statistics
head(exGWAS)

## Load reference genotype data
bfile &lt;- system.file("extdata", "example.bed", package = "snpsettest")
x &lt;- read_reference_bed(path = bfile)

## Harmonize by SNP IDs
hsumstats1 &lt;- harmonize_sumstats(exGWAS, x)

## Harmonize by genomic position and allele codes
## Reference allele swap will be taken into account
hsumstats2 &lt;- harmonize_sumstats(exGWAS, x, match_by_id = FALSE)

## Check matching entries by flipping allele codes
## Ambiguous SNPs will be excluded from harmonization
hsumstats3 &lt;- harmonize_sumstats(exGWAS, x, match_by_id = FALSE,
                                 check_strand_flip = TRUE)
</code></pre>

<hr>
<h2 id='map_snp_to_gene'>Map SNPs to genes</h2><span id='topic+map_snp_to_gene'></span>

<h3>Description</h3>

<p>Annotate SNPs onto their neighboring genes (or arbitrary genomic regions) to
perform set-based association tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map_snp_to_gene(
  info_snp,
  info_gene,
  extend_start = 20L,
  extend_end = 20L,
  only_sets = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="map_snp_to_gene_+3A_info_snp">info_snp</code></td>
<td>
<p>A data frame with columns: &quot;id&quot;, &quot;chr&quot;, and &quot;pos&quot;.
</p>

<ul>
<li><p> id = SNP ID (e.g., rs numbers)
</p>
</li>
<li><p> chr = chromosome
</p>
</li>
<li><p> pos = base-pair position
</p>
</li></ul>
</td></tr>
<tr><td><code id="map_snp_to_gene_+3A_info_gene">info_gene</code></td>
<td>
<p>A data frame with columns: &quot;gene.id&quot;, &quot;chr&quot;, &quot;start&quot;,
and &quot;end&quot;.
</p>

<ul>
<li><p> gene.id = gene ID (or identifier for genomic regions)
</p>
</li>
<li><p> chr = chromosome (must be the same chromosome coding scheme in <code>info_snp</code>)
</p>
</li>
<li><p> start = genomic start position
</p>
</li>
<li><p> end = genomic end position
</p>
</li></ul>

<p>If a gene has multiple intervals, SNPs mapped to any of them will be merged
into a single set. Please assign unique IDs if you don't want this behavior.</p>
</td></tr>
<tr><td><code id="map_snp_to_gene_+3A_extend_start">extend_start</code></td>
<td>
<p>A single non-negative integer, allowing for a certain kb
window before the gene to be included. Default is 20 (= 20kb).</p>
</td></tr>
<tr><td><code id="map_snp_to_gene_+3A_extend_end">extend_end</code></td>
<td>
<p>A single non-negative integer, allowing for a certain kb
window after the gene to be included. Default is 20 (= 20kb).</p>
</td></tr>
<tr><td><code id="map_snp_to_gene_+3A_only_sets">only_sets</code></td>
<td>
<p>If <code>TRUE</code>, only sets of SNPs for individual genes are
returned. Otherwise, both sets and mapping information are returned.
Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A nested list containing following components:
</p>

<ul>
<li><p> sets: a named list where each index represents a separate set of SNPs
</p>
</li>
<li><p> map: a data frame containing SNP mapping information
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## GWAS summary statistics
head(exGWAS)

## Gene information data
head(gene.curated.GRCh37)

## Map SNPs to genes
snp_sets &lt;- map_snp_to_gene(exGWAS, gene.curated.GRCh37)

## Better to use harmonized GWAS data for gene mapping
bfile &lt;- system.file("extdata", "example.bed", package = "snpsettest")
x &lt;- read_reference_bed(path = bfile)
hsumstats &lt;- harmonize_sumstats(exGWAS, x)
snp_sets &lt;- map_snp_to_gene(hsumstats, gene.curated.GRCh37)

</code></pre>

<hr>
<h2 id='read_reference_bed'>Read a PLINK bed file for reference data</h2><span id='topic+read_reference_bed'></span>

<h3>Description</h3>

<p>Create a <code>bed.matrix</code> object from a .bed file. The function expects
.fam and .bim files under the same directory. See <a href="gaston.html#topic+read.bed.matrix">gaston::read.bed.matrix</a>
for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_reference_bed(path, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_reference_bed_+3A_path">path</code></td>
<td>
<p>A path to the .bed file</p>
</td></tr>
<tr><td><code id="read_reference_bed_+3A_...">...</code></td>
<td>
<p>Further arguments used in <a href="gaston.html#topic+read.bed.matrix">gaston::read.bed.matrix</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="gaston.html#topic+bed.matrix-class">gaston::bed.matrix</a> object with a Z-standardized genotype matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Get a path to the example .bed file
bfile &lt;- system.file("extdata", "example.bed", package = "snpsettest")

## Read a .bed file
x &lt;- read_reference_bed(path = bfile)
</code></pre>

<hr>
<h2 id='snpset_test'>Set-based association tests</h2><span id='topic+snpset_test'></span>

<h3>Description</h3>

<p>Perform set-based association tests between multiple sets of SNPs and a
phenotype using GWAS summary statistics. If the function encounters missing
genotypes in the reference data, they will be imputed with genotype means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpset_test(hsumstats, x, snp_sets, method = c("saddle", "davies"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpset_test_+3A_hsumstats">hsumstats</code></td>
<td>
<p>A data frame processed by <code><a href="#topic+harmonize_sumstats">harmonize_sumstats</a></code>.</p>
</td></tr>
<tr><td><code id="snpset_test_+3A_x">x</code></td>
<td>
<p>A <code>bed.matrix</code> object created from the reference data.</p>
</td></tr>
<tr><td><code id="snpset_test_+3A_snp_sets">snp_sets</code></td>
<td>
<p>A named list where each index represents a separate set of
SNPs.</p>
</td></tr>
<tr><td><code id="snpset_test_+3A_method">method</code></td>
<td>
<p>A method to compute a set-level p value. &quot;saddle&quot; uses
Kuonen's saddlepoint approximation (1999) and &quot;davies&quot; uses the algorithm
of Davies (1980). When &quot;davies&quot; method failed to produce a meaningful
result, &quot;saddle&quot; method is used as a fallback. Default is &quot;saddle&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with columns: &quot;set.id&quot;, &quot;pvalue&quot;, &quot;n.snp&quot;, &quot;top.snp.id&quot;
and &quot;top.snp.pvalue&quot;
</p>

<ul>
<li><p> set.id = a name of SNP set
</p>
</li>
<li><p> tstat = a test statistic
</p>
</li>
<li><p> pvalue = a set-level p value
</p>
</li>
<li><p> n.snp = the number of SNPs used in a test
</p>
</li>
<li><p> top.snp.id = SNP ID with the smallest p-value within a set of SNPs
</p>
</li>
<li><p> top.snp.pvalue = The smallest p-value within a set of SNPs
</p>
</li></ul>



<h3>References</h3>

<p>Kuonen, D. Saddlepoint Approximations for Distributions of Quadratic Forms
in Normal Variables. Biometrika 86, 929–935 (1999).
</p>
<p>Davies, R. B. Algorithm AS 155: The Distribution of a Linear Combination of
Chi-Square Random Variables. Journal of the Royal Statistical Society.
Series C (Applied Statistics) 29, 323–333 (1980).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## GWAS summary statistics
head(exGWAS)

## Load reference genotype data
bfile &lt;- system.file("extdata", "example.bed", package = "snpsettest")
x &lt;- read_reference_bed(path = bfile)

## GWAS harmonization with reference data
hsumstats &lt;- harmonize_sumstats(exGWAS, x)

## Perform a set-based test with an arbitrary SNP set
snpset_test(hsumstats, x, list(test = c("SNP_880", "SNP_1533", "SNP_4189")))

## Gene information data
head(gene.curated.GRCh37)

## Map SNPs to genes
snp_sets &lt;- map_snp_to_gene(hsumstats, gene.curated.GRCh37)

## Perform gene-based association tests
out &lt;- snpset_test(hsumstats, x, snp_sets$sets)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
