<!DOCTYPE html><html lang="en"><head><title>Help for package SNVLFDR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SNVLFDR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#get_LFDRs'><p>Estimates LFDR values per genomic site</p></a></li>
<li><a href='#get_LFDRs_given_caller'><p>Estimates LFDR values per mutant site detected by a desired variant caller</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Empirical Bayes Single Nucleotide Variant Calling</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Ali Karimnezhad [aut, cre, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ali Karimnezhad &lt;ali.karimnezhad@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Identifies single nucleotide variants in next-generation sequencing data by estimating their local false discovery rates. For more details, see Karimnezhad, A. and Perkins, T. J. (2024) &lt;<a href="https://doi.org/10.1038%2Fs41598-024-51958-z">doi:10.1038/s41598-024-51958-z</a>&gt;.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-24 16:32:08 UTC; alikarimnezhad</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-25 13:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='get_LFDRs'>Estimates LFDR values per genomic site</h2><span id='topic+get_LFDRs'></span>

<h3>Description</h3>

<p>Based on a given read count matrix, identifies single nucleotide variants (SNVs) by estimating local false discovery rates (LFDRs). Users can set an initial value for the proportion of non-mutant sites and specify thresholds for allele frequency, read depth and LFDR cut-off value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_LFDRs(
  bam_input,
  bedfile,
  BQ.T,
  MQ.T,
  pi0.initial,
  AF.T,
  DP.T,
  LFDR.T,
  error,
  method,
  epsilon
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_LFDRs_+3A_bam_input">bam_input</code></td>
<td>
<p>Path to an input BAM file. The file must be in the format of a csv file generated by bam-readcount (https://github.com/genome/bam-readcount). See Examples.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_bedfile">bedfile</code></td>
<td>
<p>Path to a bed file containing genomic regions of interest. This file has to have three columns with chr# in the first column and start and end positions in the second and third columns respectively. No headers.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_bq.t">BQ.T</code></td>
<td>
<p>Minimum base call quality threshold. It can be set to 0 to include all sites. Otherwise, this threshold excludes sites with  a base call quality below the specified threshold. It is recomended to set it to 20.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_mq.t">MQ.T</code></td>
<td>
<p>Minimum mapping  quality threshold. It can be set to 0 to include all sites. Otherwise, this threshold excludes sites with a mapping quality below the specified threshold. It is recomended to set it to 20.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_pi0.initial">pi0.initial</code></td>
<td>
<p>Initial value for the proportion of non-mutant sites. It can be any number between 0 and 1. However it is recommended to set it to a number between 0.95 and 0.99 for more accuracy. If no value is specified, it will be set to 0.95 by default.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_af.t">AF.T</code></td>
<td>
<p>Allele frequency threshold. It can be set to 0 to include all sites. Otherwise, this threshold excludes sites with an allele frequency below the specified threshold.  If no value is specified, it will be set to 0.01 by default.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_dp.t">DP.T</code></td>
<td>
<p>Read depth threshold. It can be set to 0 to include all sites. Otherwise, this threshold excludes sites with a read depth below the specified threshold.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_lfdr.t">LFDR.T</code></td>
<td>
<p>A number between 0 and 1. It can be set to 0 to include all sites. Otherwise, this threshold excludes sites with an estimated LFDR below the specified threshold.  If no value is specified, it will be set to 0.01 by default.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_error">error</code></td>
<td>
<p>Error rate between 0 and 1. If it is set to NULL, a weighted average of average base call quality and average mapping quality per site will be calculated. Otherwise, it may be set to 0.01 or a desired error vector can be introduced by the user.</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_method">method</code></td>
<td>
<p>Method used to estimate pi0 and LFDRs. It can be &quot;empirical&quot;, &quot;uniform_empirical&quot; or &quot;uniform&quot;. If no method is specified, it will be set to &quot;empirical&quot; by default (recommended).</p>
</td></tr>
<tr><td><code id="get_LFDRs_+3A_epsilon">epsilon</code></td>
<td>
<p>The difference between old and new estimates of pi0 used for convergence. If no value is specified, it will be set to 0.01 by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list. Slot <strong>estimated.pi0</strong> returns estimated proportion of non-mutant sites. Slot <strong>estimated.LFDRs</strong> returns estimated LFDRs for genomic sites that were not filtered out.
Slot <strong>filtered.bam</strong> adds estimated LFDRs, model errors and a mutant variable (indicating whether each site is detected to be a mutant (1) or non-mutant (0) site) to the filtered input file .
</p>


<h3>References</h3>

<p>Karimnezhad, A. and Perkins, T.J. (2024). Empirical Bayes single nucleotide variant calling for next-generation sequencing data. Scientific Reports 14,  1550, &lt;doi:10.1038/s41598-024-51958-z&gt;</p>


<h3>Examples</h3>

<pre><code class='language-R'>bam_input &lt;- system.file("extdata", "bam_input.csv", package="SNVLFDR")
bedfile &lt;- system.file("extdata", "regions.bed", package="SNVLFDR")
BQ.T=20
MQ.T=20
pi0.initial=0.95
AF.T=0.01
DP.T=10
LFDR.T=0.01
error=NULL
method='empirical'
epsilon=0.01
output=get_LFDRs(bam_input,bedfile,BQ.T,MQ.T,pi0.initial,AF.T,DP.T,LFDR.T,error,method,epsilon)

</code></pre>

<hr>
<h2 id='get_LFDRs_given_caller'>Estimates LFDR values per mutant site detected by a desired variant caller</h2><span id='topic+get_LFDRs_given_caller'></span>

<h3>Description</h3>

<p>Based on a given read count matrix and a list of calls made by a desired variant caller, estimates LFDRs that corresponds to each genomic site. It further
classifies sites to either mutant or non-mutant sites by comparing their estimated LFDRs with an LFDR cut-off value. The cut-off value as well as error rates can be defined by users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_LFDRs_given_caller(bam_input, calls, LFDR.T, error)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_LFDRs_given_caller_+3A_bam_input">bam_input</code></td>
<td>
<p>Path to an original BAM file used to call variants. The file must be in the format of a csv file generated by bam-readcount (https://github.com/genome/bam-readcount). See Examples.</p>
</td></tr>
<tr><td><code id="get_LFDRs_given_caller_+3A_calls">calls</code></td>
<td>
<p>Path to a vcf file generated by a variant caller. The first and second columns of this file have to be CHR names and positions, respectively.</p>
</td></tr>
<tr><td><code id="get_LFDRs_given_caller_+3A_lfdr.t">LFDR.T</code></td>
<td>
<p>A number between 0 and 1. It can be set to 0 to include all sites. Otherwise, this threshold excludes sites with an LFDR below the specified threshold.  If no value is specified, it will be set to 0.01 by default.</p>
</td></tr>
<tr><td><code id="get_LFDRs_given_caller_+3A_error">error</code></td>
<td>
<p>Error rate between 0 and 1. If it is set to NULL, a weighted average of average base call quality and average mapping quality per site will be calculated. Otherwise, it may be set to 0.01 or a desired error vector can be introduced by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list. Slot <strong>estimated.LFDRs</strong> returns estimated LFDRs for all sites in the input file.
Slot <strong>updated.bam</strong> adds estimated LFDRs, model errors and a mutant variable (indicating whether each site is detected to be a mutant (1) or non-mutant (0) site) to the input bam file.
</p>


<h3>References</h3>

<p>Karimnezhad, A. and Perkins, T.J. (2024). Empirical Bayes single nucleotide variant calling for next-generation sequencing data. Scientific Reports 14,  1550, &lt;doi:10.1038/s41598-024-51958-z&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bam_path &lt;- system.file("extdata", "bam_input.csv", package="SNVLFDR")
calls_path &lt;- system.file("extdata", "calls.vcf", package="SNVLFDR")
output=get_LFDRs_given_caller(bam_input=bam_path,calls=calls_path,LFDR.T=0.01,error=NULL)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
