<!DOCTYPE html><html><head><title>Help for package som</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {som}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#filtering'>
<p>Filter data before feeding som algorithm for gene expression data</p></a></li>
<li><a href='#normalize'>
<p>normalize data before feeding som algorithm</p></a></li>
<li><a href='#plot.som'><p> Visualizing a SOM</p></a></li>
<li><a href='#qerror'><p> quantization accuracy</p></a></li>
<li><a href='#som'>
<p>Function to train a Self-Organizing Map</p></a></li>
<li><a href='#som-internal'><p>Internal som functions</p></a></li>
<li><a href='#summary.som'><p> summarize a som object</p></a></li>
<li><a href='#yeast'><p> yeast cell cycle</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.3-5.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2010-04-08</td>
</tr>
<tr>
<td>Title:</td>
<td>Self-Organizing Map</td>
</tr>
<tr>
<td>Author:</td>
<td>Jun Yan &lt;jyan@stat.uconn.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jun Yan &lt;jyan@stat.uconn.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Description:</td>
<td>Self-Organizing Map (with application in gene clustering).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-07-06 08:24:46 UTC; hornik</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-07-06 10:26:15</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
</table>
<hr>
<h2 id='filtering'>
Filter data before feeding som algorithm for gene expression data
</h2><span id='topic+filtering'></span>

<h3>Description</h3>

<p>Filtering data by certain floor, ceiling, max/min ratio, and max - min difference. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filtering(x, lt=20, ut=16000, mmr=3, mmd=200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filtering_+3A_x">x</code></td>
<td>

<p>a data frame or matrix of input data.
</p>
</td></tr>
<tr><td><code id="filtering_+3A_lt">lt</code></td>
<td>

<p>floor value replaces those less than it with the value
</p>
</td></tr>
<tr><td><code id="filtering_+3A_ut">ut</code></td>
<td>

<p>ceiling value replaced those greater than it with the value
</p>
</td></tr>
<tr><td><code id="filtering_+3A_mmr">mmr</code></td>
<td>

<p>the max/min ratio, rows with max/min &lt; mmr will be removed
</p>
</td></tr>
<tr><td><code id="filtering_+3A_mmd">mmd</code></td>
<td>

<p>the max - min difference, rows with (max - min) &lt; mmd will be removed
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An dataframe or matrix after the filtering
</p>


<h3>Author(s)</h3>

<p>Jun Yan &lt;jyan@stat.uiowa.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+normalize">normalize</a></code>.
</p>

<hr>
<h2 id='normalize'>
normalize data before feeding som algorithm
</h2><span id='topic+normalize'></span>

<h3>Description</h3>

<p>Normalize the data so that each row has mean 0 and variance 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize(x, byrow=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_+3A_x">x</code></td>
<td>

<p>a data frame or matrix of input data.
</p>
</td></tr>
<tr><td><code id="normalize_+3A_byrow">byrow</code></td>
<td>

<p>whether normalizing by row or by column, default is byrow.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An dataframe or matrix after the normalizing.
</p>


<h3>Author(s)</h3>

<p>Jun Yan &lt;jyan@stat.uiowa.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+filtering">filtering</a></code>.
</p>

<hr>
<h2 id='plot.som'> Visualizing a SOM </h2><span id='topic+plot.som'></span>

<h3>Description</h3>

<p>Plot the SOM in a 2-dim map with means and sd bars.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'som'
plot(x, sdbar=1, ylim=c(-3, 3), color=TRUE,
ntik=3, yadj=0.1, xlab="", ylab="", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.som_+3A_x">x</code></td>
<td>
<p> a som object </p>
</td></tr>
<tr><td><code id="plot.som_+3A_sdbar">sdbar</code></td>
<td>
<p> the length of sdbar in sd, no sdbar if sdbar=0 </p>
</td></tr>
<tr><td><code id="plot.som_+3A_ylim">ylim</code></td>
<td>
<p> the range of y axies in each cell of the map </p>
</td></tr>
<tr><td><code id="plot.som_+3A_color">color</code></td>
<td>
<p> whether or not use color plotting </p>
</td></tr>
<tr><td><code id="plot.som_+3A_ntik">ntik</code></td>
<td>
<p> the number of tiks of the vertical axis </p>
</td></tr>
<tr><td><code id="plot.som_+3A_yadj">yadj</code></td>
<td>
<p> the proportion used to put the number of obs </p>
</td></tr>
<tr><td><code id="plot.som_+3A_xlab">xlab</code></td>
<td>
<p> x label </p>
</td></tr>
<tr><td><code id="plot.som_+3A_ylab">ylab</code></td>
<td>
<p> y label </p>
</td></tr>
<tr><td><code id="plot.som_+3A_...">...</code></td>
<td>
<p> other options to plot </p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is not cleanly written. The original purpose was to
mimic what GENECLUSTER does. The ylim is hardcoded so that only
standardized data could be properly plotted.
</p>
<p>There are visualization methods like umat and sammon in SOM\_PAK3.1,
but not implemented here. 
</p>


<h3>Author(s)</h3>

<p> Jun Yan &lt;jyan@stat.uiowa.edu&gt; </p>


<h3>Examples</h3>

<pre><code class='language-R'>foo &lt;- som(matrix(rnorm(1000), 250), 3, 5)
plot(foo, ylim=c(-1, 1))
</code></pre>

<hr>
<h2 id='qerror'> quantization accuracy </h2><span id='topic+qerror'></span>

<h3>Description</h3>

<p>get the average distortion measure
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qerror(obj, err.radius=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qerror_+3A_obj">obj</code></td>
<td>
<p> a &lsquo;som&rsquo; object </p>
</td></tr>
<tr><td><code id="qerror_+3A_err.radius">err.radius</code></td>
<td>
<p> radius used calculating qerror </p>
</td></tr>
</table>


<h3>Value</h3>

<p>An average of the following quantity (weighted distance measure) over all x in the sample,
</p>
<p style="text-align: center;"><code class="reqn">\sum ||x - m_i|| h_{ci}</code>
</p>

<p>where <code class="reqn">h_{ci}</code> is the neighbourhood kernel for the ith code.
</p>


<h3>Author(s)</h3>

<p>Jun Yan &lt;jyan@stat.uiowa.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>foo &lt;- som(matrix(rnorm(1000), 100), 2, 4)
qerror(foo, 3)
</code></pre>

<hr>
<h2 id='som'>
Function to train a Self-Organizing Map
</h2><span id='topic+som.init'></span><span id='topic+som'></span><span id='topic+som.train'></span><span id='topic+som.update'></span><span id='topic+som.project'></span>

<h3>Description</h3>

<p>Produces an object of class &quot;som&quot; which is a Self-Organizing Map fit of the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>som.init(data, xdim, ydim, init="linear")
som(data, xdim, ydim, init="linear", alpha=NULL, alphaType="inverse",
neigh="gaussian", topol="rect", radius=NULL, rlen=NULL, err.radius=1,
inv.alp.c=NULL)
som.train(data, code, xdim, ydim, alpha=NULL, alphaType="inverse",
neigh="gaussian", topol="rect", radius=NULL, rlen=NULL, err.radius=1, inv.alp.c=NULL)
som.update(obj, alpha = NULL, radius = NULL, rlen = NULL, err.radius =
1, inv.alp.c = NULL)
som.project(obj, newdat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="som_+3A_obj">obj</code></td>
<td>
<p>a &lsquo;som&rsquo; object.</p>
</td></tr>
<tr><td><code id="som_+3A_newdat">newdat</code></td>
<td>
<p>a new dataset needs to be projected onto the map.</p>
</td></tr>
<tr><td><code id="som_+3A_code">code</code></td>
<td>
<p>a matrix of initial code vector in the map.</p>
</td></tr>
<tr><td><code id="som_+3A_data">data</code></td>
<td>

<p>a data frame or matrix of input data.
</p>
</td></tr>
<tr><td><code id="som_+3A_xdim">xdim</code></td>
<td>

<p>an integer specifying the x-dimension of the map.
</p>
</td></tr>
<tr><td><code id="som_+3A_ydim">ydim</code></td>
<td>

<p>an integer specifying the y-dimension of the map.
</p>
</td></tr>
<tr><td><code id="som_+3A_init">init</code></td>
<td>

<p>a character string specifying the initializing method. The following
are permitted:
<code>"sample"</code> uses a radom sample from the data;
<code>"random"</code> uses random draws from N(0,1);
<code>"linear"</code> uses the linear grids upon the first two principle
components directin.
</p>
</td></tr>
<tr><td><code id="som_+3A_alpha">alpha</code></td>
<td>

<p>a vector of initial learning rate parameter for the two training
phases. Decreases linearly to zero during training.
</p>
</td></tr>
<tr><td><code id="som_+3A_alphatype">alphaType</code></td>
<td>

<p>a character string specifying learning rate funciton type. Possible
choices are linear function (<code>"linear"</code>) and inverse-time type
function (<code>"inverse"</code>).
</p>
</td></tr>
<tr><td><code id="som_+3A_neigh">neigh</code></td>
<td>

<p>a character string specifying the neighborhood function type. The
following are permitted:
</p>
<p><code>"bubble"</code>
<code>"gaussian"</code>
</p>
</td></tr>
<tr><td><code id="som_+3A_topol">topol</code></td>
<td>

<p>a character string specifying the topology type when measuring
distance in the map. The following are permitted:
</p>
<p><code>"hexa"</code>
<code>"rect"</code>
</p>
</td></tr>
<tr><td><code id="som_+3A_radius">radius</code></td>
<td>

<p>a vector of initial radius of the training area in som-algorithm for
the two training phases. Decreases linearly to one during training.
</p>
</td></tr>
<tr><td><code id="som_+3A_rlen">rlen</code></td>
<td>

<p>a vector of running length (number of steps) in the two training phases.
</p>
</td></tr>
<tr><td><code id="som_+3A_err.radius">err.radius</code></td>
<td>

<p>a numeric value specifying the radius when calculating average
distortion measure.
</p>
</td></tr>
<tr><td><code id="som_+3A_inv.alp.c">inv.alp.c</code></td>
<td>

<p>the constant C in the inverse learning rate function:
alpha0 * C / (C + t);
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>&lsquo;som.init&rsquo; initializes a map and returns the code matrix. &lsquo;som&rsquo; does
the two-step som training in a batch fashion and return a &lsquo;som&rsquo;
object. &lsquo;som.train&rsquo; takes data, code, and traing parameters and
perform the requested som training. &lsquo;som.update&rsquo; takes a &lsquo;som&rsquo; object
and further train it with updated paramters. &lsquo;som.project&rsquo; projects
new data onto the map.
</p>
<p>An object of class <code>"som"</code> representing the fit, which is a list
containing the following components:
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>the dataset on which som was applied.</p>
</td></tr>
<tr><td><code>init</code></td>
<td>
<p>a character string indicating the initializing method.</p>
</td></tr>
<tr><td><code>xdim</code></td>
<td>
<p>an integer specifying the x-dimension of the map.</p>
</td></tr>
<tr><td><code>ydim</code></td>
<td>
<p>an integer specifying the y-dimension of the map.</p>
</td></tr>
<tr><td><code>code</code></td>
<td>
<p>a metrix with nrow = xdim*ydim, each row corresponding to
a code vector of a cell in the map. The mapping from cell coordinate
(x, y) to the row index in the code matrix is:
rownumber = x + y * xdim
</p>
</td></tr>
<tr><td><code>visual</code></td>
<td>
<p>a data frame of three columns, with the same number of
rows as in data: x and y are the coordinate of the corresponding
observation in the map, and qerror is the quantization error
computed as the squared distance (depends topol) between the observation
vector and its coding vector.
</p>
</td></tr>
<tr><td><code>alpha0</code></td>
<td>
<p>a vector of initial learning rate parameter for the two
training phases.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>a character string specifying learning rate funciton
type.</p>
</td></tr>
<tr><td><code>neigh</code></td>
<td>
<p>a character string specifying the neighborhood function
type.</p>
</td></tr>
<tr><td><code>topol</code></td>
<td>
<p>a character string specifying the topology type when measuring
distance in the map.</p>
</td></tr>
<tr><td><code>radius0</code></td>
<td>
<p>a vector of initial radius of the training area in
som-algorithm for the two training phases.</p>
</td></tr>
<tr><td><code>rlen</code></td>
<td>
<p>a vector of running length in the two training phases.</p>
</td></tr>
<tr><td><code>qerror</code></td>
<td>
<p>a numeric value of average distortion measure.</p>
</td></tr>
<tr><td><code>code.sum</code></td>
<td>
<p>a dataframe summaries the number of observations in
each map cell.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Yan &lt;jyan@stat.uiowa.edu&gt;
</p>


<h3>References</h3>

<p>Kohonen, Hynninen, Kangas, and Laaksonen (1995), SOM-PAK, the
Self-Organizing Map Program Package (version 3.1).
http://www.cis.hut.fi/research/papers/som\_tr96.ps.Z</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(yeast)
yeast &lt;- yeast[, -c(1, 11)]
yeast.f &lt;- filtering(yeast)
yeast.f.n &lt;- normalize(yeast.f)
foo &lt;- som(yeast.f.n, xdim=5, ydim=6)
foo &lt;- som(yeast.f.n, xdim=5, ydim=6, topol="hexa", neigh="gaussian")
plot(foo)
</code></pre>

<hr>
<h2 id='som-internal'>Internal som functions</h2><span id='topic+ciplot'></span><span id='topic+inrange'></span><span id='topic+plotcell'></span><span id='topic+somgrids'></span><span id='topic+somsum'></span><span id='topic+som.par'></span>

<h3>Description</h3>

<p>Internal som Functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ciplot(x, y, se, n=1, d, ywindow)
inrange(x, xlim)
plotcell(x, y, dat, code, n, sdbar=1, ylim, yadj)
somgrids(xdim, ydim, color, yadj=0.1, hexa, ntik, ylim)
somsum(obj)
som.par(obj)
</code></pre>


<h3>Details</h3>

<p>These are not to be called directly by the user.
</p>

<hr>
<h2 id='summary.som'> summarize a som object </h2><span id='topic+summary.som'></span><span id='topic+print.som'></span>

<h3>Description</h3>

<p>print out the configuration parameters of a som object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'som'
summary(object, ...)
## S3 method for class 'som'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.som_+3A_object">object</code>, <code id="summary.som_+3A_x">x</code></td>
<td>
<p> a &lsquo;som&rsquo; object </p>
</td></tr>
<tr><td><code id="summary.som_+3A_...">...</code></td>
<td>
<p>nothing yet</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jun Yan &lt;jyan@stat.uiowa.edu&gt;
</p>

<hr>
<h2 id='yeast'> yeast cell cycle </h2><span id='topic+yeast'></span>

<h3>Description</h3>

<p>The <code>yeast</code> data frame has 6601 rows and 18 columns, i.e., 6601
genes, measured at 18 time points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(yeast)
</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Gene</dt><dd><p>a character vector of gene names</p>
</dd>
<dt>zero</dt><dd>
<p>a numeric vector    </p>
</dd>
<dt>ten</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>twenty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>thirty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>fourty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>fifty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>sixty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>seventy</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>eighty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>ninety</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>hundred</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>one.ten</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>one.twenty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>one.thirty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>one.fourty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>one.fifty</dt><dd>
<p>a numeric vector
</p>
</dd>
<dt>one.sixty</dt><dd>
<p>a numeric vector
</p>
</dd>
</dl>



<h3>Source</h3>

<p>http://genomics.stanford.edu
</p>


<h3>References</h3>

<p>Tamayo et. al. (1999), Interpreting patterns of gene expression with
self-organizing maps: Methods and application to hematopoietic
differentiation, PNAS V96, pp2907-2912, March 1999.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
