<!DOCTYPE html><html lang="en"><head><title>Help for package sonicLength</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sonicLength}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#sonicLength-package'>
<p>Calibration of Number of Length Variants</p></a></li>
<li><a href='#A1'>
<p>HTLV-1 Fragment Length Data</p></a></li>
<li><a href='#estAbund'>
<p>estimate Abundances from sonicated samples</p></a></li>
<li><a href='#Ey.given.x'>
<p>E-Step for Abundance Estimation</p></a></li>
<li><a href='#Fragment+20Distribution'><p>Simple Fragment Length Distribution</p></a></li>
<li><a href='#maxEM'><p> maximum likelihood estimates relative abundances</p></a></li>
<li><a href='#maxEM.iter.control'>
<p>iteration controls</p></a></li>
<li><a href='#mstep'>
<p>M-step to maximize theta</p></a></li>
<li><a href='#phi.update.lframe'>
<p>estimate phi</p></a></li>
<li><a href='#simSonic'><p>Simulate Sonicated Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimating Abundance of Clones from DNA Fragmentation Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-09-19</td>
</tr>
<tr>
<td>Author:</td>
<td>Charles Berry &lt;ccberry@ucsd.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Charles Berry &lt;ccberry@ucsd.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.14.0), splines</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimate the abundance of cell clones from the
	     distribution of lengths of DNA fragments (as created by
	     sonication, whence &lsquo;sonicLength&rsquo;).  The algorithm in
	     "Estimating abundances of retroviral insertion sites from
	     DNA fragment length data" by Berry CC, Gillet NA, Melamed
	     A, Gormley N, Bangham CR, Bushman FD. Bioinformatics;
	     2012 Mar 15;28(6):755-62 is implemented.  The
	     experimental setting and estimation details are described
	     in detail there. Briefly, integration of new DNA in a
	     host genome (due to retroviral infection or gene therapy)
	     can be tracked using DNA sequencing, potentially allowing
	     characterization of the abundance of individual cell
	     clones bearing distinct integration sites. The locations
	     of integration sites can be determined by fragmenting the
	     host DNA (via sonication or fragmentase), breaking the
	     newly integrated DNA at a known sequence, amplifying the
	     fragments containing both host and integrated DNA,
	     sequencing those amplicons, then mapping the host
	     sequences to positions on the reference genome. The
	     relative number of fragments containing a given position
	     in the host genome estimates the relative abundance of
	     cells hosting the corresponding integration site, but
	     that number is not available and the count of amplicons
	     per fragment varies widely.  However, the expected number
	     of distinct fragment lengths is a function of the
	     abundance of cells hosting an integration site at a given
	     position and a certain nuisance parameter. The algorithm
	     implicitly estimates that function to estimate the
	     relative abundance.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Repository/R-Forge/Project:</td>
<td>soniclength</td>
</tr>
<tr>
<td>Repository/R-Forge/Revision:</td>
<td>12</td>
</tr>
<tr>
<td>Repository/R-Forge/DateTimeStamp:</td>
<td>2021-09-19 22:55:50</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-20 04:10:02 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-19 23:08:17 UTC; rforge</td>
</tr>
</table>
<hr>
<h2 id='sonicLength-package'>
Calibration of Number of Length Variants
</h2><span id='topic+sonicLength-package'></span><span id='topic+sonicLength'></span>

<h3>Description</h3>

<p>Estimate the number of fragments produced by sonication of integration
sites in DNA (as arise in gene therapy or in retroviral infections)
from tallies of the number of distinct lengths. See the Berry et al. reference
where the experimental setting and algorithm are spelled out in detail.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> sonicLength</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> see file LICENSE distributed with this software</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package has the necessary utilities to provide a function that will
do calibration of the number of distinct lengths (for an integration
site recovered via sonication) to estimate the number of distinct
sonication fragments that underlie the observed number of lengths.
</p>


<h3>Author(s)</h3>

<p>Charles C. Berry
</p>
<p>Maintainer: &lt;ccberry@ucsd.edu&gt;
</p>


<h3>References</h3>

<p>The reference to the algorithm is:
</p>
<p>Estimating abundances of retroviral insertion sites 
from DNA fragment length data. Berry CC, Gillet NA, 
Melamed A, Gormley N, Bangham CR, Bushman FD.
Bioinformatics; 2012 Mar 15;28(6):755-62.  
</p>
<p>Some applications that illustrate use of this method are found here:
</p>
<p>Gillet, Nicolas A., et al. &quot;The host genomic environment of the provirus
determines the abundance of HTLV-1 infected T-cell clones.&quot; Blood 117.11
(2011): 3113-3122.
</p>
<p>Maldarelli, F., et al. &quot;Specific HIV integration sites are linked to
clonal expansion and persistence of infected cells.&quot; Science 345.6193
(2014): 179-183.
</p>

<hr>
<h2 id='A1'>
HTLV-1 Fragment Length Data
</h2><span id='topic+A1'></span>

<h3>Description</h3>

<p>A sample from a patient with HTLV-1 was analyzed as described in
Gillet et al (2001). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(A1)</code></pre>


<h3>Format</h3>

<p>A data frame with 6176 observations on the following 3 variables.
</p>

<dl>
<dt><code>locations</code></dt><dd><p>a factor with levels <code>01 F 101480450</code>
... <code>X R 99972891</code> indicating the genomic sites at which
insertions were detected</p>
</dd>
<dt><code>lengths</code></dt><dd><p>a numeric vector of fragment lengths</p>
</dd>
<dt><code>replicates</code></dt><dd><p>a numeric vector giving replicate IDs</p>
</dd>
</dl>



<h3>Details</h3>

<p>For each replicate, the unique combinations of the genome location at which an insertion
site was detected and the length(s) of the fragment(s) sequenced were
noted. There is one such row in he data.frame for each such combination.
</p>


<h3>Source</h3>

<p>See Gillet et al 2011 for more details.
</p>


<h3>References</h3>

<p>Gillet, N.A., Malani, N., Melamed, A., Gormley, N., Carter, R., Bentley, D.,
Berry, C., Bushman, F.D., Taylor, G.P., and Bangham, C.R. (2011).
The host genomic environment of the provirus determines the
abundance of HTLV-1-infected T-cell clones.
<em> Blood </em> 117, 3113-3122.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(A1)
str(A1)
</code></pre>

<hr>
<h2 id='estAbund'>
estimate Abundances from sonicated samples
</h2><span id='topic+estAbund'></span>

<h3>Description</h3>

<p>When one or more replicate samples are sonicated and the lengths for each
integration site are recorded, it is of interest to estimate actual
number of sonicants allowing for coincidentally equal lengths. A
likelihood approach is used  here. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estAbund(locations, lengths, replicates = NULL, jackknife = F, kmax=0,
min.length = 20, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estAbund_+3A_locations">locations</code></td>
<td>

<p>a vector of IDs for distinct locations
</p>
</td></tr>
<tr><td><code id="estAbund_+3A_lengths">lengths</code></td>
<td>

<p>a vector of the corresponding lengths
</p>
</td></tr>
<tr><td><code id="estAbund_+3A_replicates">replicates</code></td>
<td>

<p>an optional vector of the replicate sample ID
</p>
</td></tr>
<tr><td><code id="estAbund_+3A_jackknife">jackknife</code></td>
<td>

<p>Whether to do leave-one-out jackknife samples over the replicates
</p>
</td></tr>
<tr><td><code id="estAbund_+3A_kmax">kmax</code></td>
<td>
<p>highest count to bother with when computing mass function
for counts. All higher values are
globbed together in the result. If <code>kmax==0</code> don't compute </p>
</td></tr>
<tr><td><code id="estAbund_+3A_min.length">min.length</code></td>
<td>
<p>Least length expected for a fragment.  Shorter
lengths will not be permitted</p>
</td></tr>
<tr><td><code id="estAbund_+3A_...">...</code></td>
<td>

<p>Other arguments that may be passed along to functions that do the
real work.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a wrapper function for <code><a href="#topic+maxEM">maxEM</a></code> so study that
function to see what is going on.
</p>


<h3>Value</h3>

<p>a list with components
</p>
<table role = "presentation">
<tr><td><code>theta</code></td>
<td>
<p> estimated abundances </p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p>  estimated bin probabilities</p>
</td></tr>
<tr><td><code>var.theta</code></td>
<td>
<p> variances of <code>theta</code> </p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p> number of iterations till convergence was achieved </p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p> the function call </p>
</td></tr>
<tr><td><code>lframe</code></td>
<td>
<p> the <code>data.frame</code> used to construct the estimate of
<code>phi</code></p>
</td></tr>
<tr><td><code>obs</code></td>
<td>
<p> the observed counts of distinct lengths </p>
</td></tr>
<tr><td><code>jackknife</code></td>
<td>
<p> a list of calls to <code> maxEM </code> with one replicate
omitted in each</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p> a <code>data.frame</code> with columns
<code>locations</code>,<code>lengths</code>, and <code>replicates</code> (if supplied) </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Charles C Berry <a href="mailto:ccberry@users.r-forge.r-project.org">ccberry@users.r-forge.r-project.org</a>
</p>


<h3>References</h3>

<p>Estimating abundances of retroviral insertion sites 
from DNA fragment length data. Berry CC, Gillet NA, 
Melamed A, Gormley N, Bangham CR, Bushman FD.
Bioinformatics; 2012 Mar 15;28(6):755-62.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maxEM">maxEM</a></code>
<code><a href="#topic+phi.update.lframe">phi.update.lframe</a></code>
The package vignette:Estimating Abundances with sonicLength
</p>

<hr>
<h2 id='Ey.given.x'>
E-Step for Abundance Estimation
</h2><span id='topic+Ey.given.x'></span><span id='topic+pr.y.given.x'></span>

<h3>Description</h3>

<p>Take the E-step of the EM algorithm for estimating the number of
sonicants in a sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ey.given.x(x, theta, phi)
pr.y.given.x(x, theta, phi, kmax=20 )
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Ey.given.x_+3A_x">x</code></td>
<td>

<p>a zero-one indicator matrix whose rows correspond to unique lengths with rownames
indicating those lengths
</p>
</td></tr>
<tr><td><code id="Ey.given.x_+3A_theta">theta</code></td>
<td>
<p> a vector of the abundance estimates. <code>length(theta)==ncol(x)</code></p>
</td></tr>
<tr><td><code id="Ey.given.x_+3A_phi">phi</code></td>
<td>
<p> a vector of the probabilities of sonicant
lengths. <code>length(phi)==nrow(x)</code></p>
</td></tr>
<tr><td><code id="Ey.given.x_+3A_kmax">kmax</code></td>
<td>
<p> highest count to bother with (all higher values are
globbed together in the result) </p>
</td></tr>
</table>


<h3>Details</h3>

<p> Supposing Poisson sampling of sonicants,
<code>Ey.given.x(x,theta,phi)[i,j]</code> gives the expected value of the
number of sonicants given that <code>x[i,j]</code> distinct sonicant lengths
were observed. <code>pr.y.given.x(x,theta,phi.kmax)[k,j]</code> gives the
probability of <code>k</code> sonicants (censored at <code>kmax+1</code>) of
insertion site <code>j</code> </p>


<h3>Value</h3>

<p><code>Ey.given.x()</code> is a double matrix of the same dimension as
<code>x</code>. <code>pr.y.given.x(...,kmax)</code> is a double matrix of
dimension <code>c( kmax+1, ncol(x) )</code> 
</p>


<h3>Author(s)</h3>

<p>Charles C. Berry <a href="mailto:ccberry@users.r-forge.r-project.org">ccberry@users.r-forge.r-project.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
mat &lt;- diag(10)
mat[ ,10 ] &lt;- 1.0
phi1 &lt;- prop.table( rep(1,10))
theta1 &lt;- 1:10
Ey.given.x( mat, theta1, phi1 )
pr.mat &lt;- pr.y.given.x( mat, theta1, phi1 )
## Estimate Seen plus Unseen sites
popsize.chao &lt;- function(tab) sum(tab)+tab[1]*(tab[1]-1)/(2*(tab[2]+1))
## evaluate at expected counts
popsize.chao( rowSums(pr.mat) ) 
## average randomly sampled counts
cnt.samp &lt;- function(x) sample( seq_along(x) , 1 ,pr=x )
mean(replicate(100,popsize.chao( table(apply(pr.mat,2, cnt.samp) ))))
</code></pre>

<hr>
<h2 id='Fragment+20Distribution'>Simple Fragment Length Distribution</h2><span id='topic+dfrag'></span><span id='topic+qfrag'></span><span id='topic+rfrag'></span><span id='topic+pfrag'></span>

<h3>Description</h3>

<p> Density, distribution function, quantile function and random
generation for a simple parametric distribution for fragments lengths
(given by conditioning a geometric distribution on whether the length
can be recovered).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfrag( x, loc=45, lscale=2.5, rate=0.02, maxx=qgeom(1-1e-7,rate) )
pfrag( q, loc=45, lscale=2.5, rate=0.02, maxx=qgeom(1-1e-7,rate), lower.tail=TRUE )
qfrag( p, loc=45, lscale=2.5, rate=0.02, maxx=qgeom(1-1e-7,rate), lower.tail=TRUE )
rfrag( n, loc=45, lscale=2.5, rate=0.02, maxx=qgeom(1-1e-7,rate) )
  </code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Fragment+2B20Distribution_+3A_x">x</code>, <code id="Fragment+2B20Distribution_+3A_q">q</code></td>
<td>
<p> vector of quantile (of lengths).</p>
</td></tr>
<tr><td><code id="Fragment+2B20Distribution_+3A_n">n</code></td>
<td>
<p> number of lengths to sample.</p>
</td></tr>
<tr><td><code id="Fragment+2B20Distribution_+3A_p">p</code></td>
<td>
<p>vector of probabilities </p>
</td></tr>
<tr><td><code id="Fragment+2B20Distribution_+3A_loc">loc</code></td>
<td>
<p> vector of locations of logistic for prob of recovery.</p>
</td></tr>
<tr><td><code id="Fragment+2B20Distribution_+3A_lscale">lscale</code></td>
<td>
<p> vector of scales of logistic for prob of recovery.</p>
</td></tr>
<tr><td><code id="Fragment+2B20Distribution_+3A_rate">rate</code></td>
<td>
<p> probability for geometric probability for fragment lengths.</p>
</td></tr>
<tr><td><code id="Fragment+2B20Distribution_+3A_maxx">maxx</code></td>
<td>
<p> integer; largest value of x to bother with.</p>
</td></tr>
<tr><td><code id="Fragment+2B20Distribution_+3A_lower.tail">lower.tail</code></td>
<td>
<p> logical; if TRUE (default), probabilities are
<code class="reqn">P[X \le x]</code> otherwise, <code class="reqn">P[X &gt; x]</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p> The mass function is given by
<code>plogis(x,loc,lscale)*dgeom(x,rate) / denom </code>, where <code>denom</code>
scales the result to sum to 1.0 in the range <code>0:maxx</code>. The other
functions all depend on this in the obvious manner. If <code>maxx</code> is
not large enough a warning may be issued, but even without this warning
the results may be slightly innaccurate if
<code>pgeom(maxx,rate,lower.tail=FALSE)</code> is non-negligible.
</p>


<h3>Value</h3>

<p><code>dfrag</code> gives the mass function,
<code>pfrag</code> gives the distribution function,
<code>qfrag</code> gives the quantile function, and
<code>rfrag</code> generates random deviates.
</p>


<h3>Author(s)</h3>

<p>Charles C. Berry <a href="mailto:ccberry@users.r-forge.r-project.org">ccberry@users.r-forge.r-project.org</a></p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+pgeom">pgeom</a></code>, <code><a href="base.html#topic+sample">sample</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
  plot( 0:300, table(factor(rfrag(2000),0:300)) )
  lines( 0:300, 2000*dfrag(0:300) )
  
  </code></pre>

<hr>
<h2 id='maxEM'> maximum likelihood estimates relative abundances</h2><span id='topic+maxEM'></span>

<h3>Description</h3>

<p>From information about the lengths of sonicants of integration sites,
infer the relative abundances of different clones and the distributon
of sonicant lengths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxEM(slmat, theta.var=FALSE, phi.update=NULL,
                  phi.deriv=NULL, lframe = NULL, glm.frm = NULL, iter.control=NULL, ... ) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="maxEM_+3A_slmat">slmat</code></td>
<td>
    
<p>a matrix whose rows correspond to unique lengths with rownames
indicating those lengths
</p>
</td></tr>
<tr><td><code id="maxEM_+3A_theta.var">theta.var</code></td>
<td>
<p>logical, return variance of theta estimates?</p>
</td></tr>
<tr><td><code id="maxEM_+3A_phi.update">phi.update</code></td>
<td>

<p>a function of an object like <code>slmat</code> that returns estimtes of
phi - a default vesrion is invoked if none is provided
</p>
</td></tr>
<tr><td><code id="maxEM_+3A_phi.deriv">phi.deriv</code></td>
<td>
<p> function of theta and phi that returns derivatives
of phi wrt beta (its parameters) </p>
</td></tr>
<tr><td><code id="maxEM_+3A_lframe">lframe</code></td>
<td>
<p> a <code>data.frame</code> which will be used to estimate
<code>phi</code>, the supplied function <code>phi.update</code> must know what to do with this
arg or it will be ignored</p>
</td></tr>
<tr><td><code id="maxEM_+3A_glm.frm">glm.frm</code></td>
<td>
<p> a formula like <code>y ~ bs(x,knots=c(50,100))</code> to
fit <code>phi</code> </p>
</td></tr>
<tr><td><code id="maxEM_+3A_iter.control">iter.control</code></td>
<td>

<p>a list of default values for iteration control - see <code><a href="#topic+maxEM.iter.control">maxEM.iter.control</a></code>
</p>
</td></tr>
<tr><td><code id="maxEM_+3A_...">...</code></td>
<td>
<p> possibly other args to pass to <code>phi.update</code> </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The EM method is used to infer the relative abundances of different sites.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>theta</code></td>
<td>
<p> a vector of the abundance estimates</p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p> a vector of the probabilities of sonicant lengths</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p> the call used </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Charles C. Berry <a href="mailto:ccberry@users.r-forge.r-project.org">ccberry@users.r-forge.r-project.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Ey.given.x">Ey.given.x</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mat0 &lt;- matrix(0,nr=48,nc=140)
vals &lt;- c(rep(1,100),2:40,100)
mat1 &lt;- sapply( vals, function(x) as.numeric(is.element(1:200 ,rgeom(x,.02))))
mat &lt;- rbind(mat0,mat1)
posVals &lt;- colSums(mat) &gt; 0
vals &lt;- vals[ posVals ]
mat &lt;- mat[, posVals ]
rownames(mat) &lt;- 1:nrow(mat)
res &lt;- maxEM(mat)
matplot( vals, cbind(res$theta, colSums(mat)), pch=1:2,
        xlab='original values', ylab='estimated values',
        main='Simulated Sonicants and Estimates')
legend( "bottomright", pch=1:2, col=1:2,
        legend=c(expression(hat(theta)[j]),expression(sum(y[ij],i))))
abline(a=0,b=1,col='gray')
</code></pre>

<hr>
<h2 id='maxEM.iter.control'>
iteration controls
</h2><span id='topic+maxEM.iter.control'></span>

<h3>Description</h3>

<p>Set parameters for control of iteration in <code>maxEM</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxEM.iter.control(min.reps = 3, max.reps = 2000, max.abs.le = 0.1,
max.rel.le = 1e-05, phi.min=.Machine$double.eps)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="maxEM.iter.control_+3A_min.reps">min.reps</code></td>
<td>

<p>a positive integer
</p>
</td></tr>
<tr><td><code id="maxEM.iter.control_+3A_max.reps">max.reps</code></td>
<td>

<p>a positive integer
</p>
</td></tr>
<tr><td><code id="maxEM.iter.control_+3A_max.abs.le">max.abs.le</code></td>
<td>

<p>a smallish double
</p>
</td></tr>
<tr><td><code id="maxEM.iter.control_+3A_max.rel.le">max.rel.le</code></td>
<td>

<p>a tiny double
</p>
</td></tr>
<tr><td><code id="maxEM.iter.control_+3A_phi.min">phi.min</code></td>
<td>
<p> a smallish double </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Most users should not need to tinker with these settings
</p>


<h3>Value</h3>

<p>a list such as <code>formals(maxEM.iter.control)</code>
</p>


<h3>Author(s)</h3>

<p>Charles C. Berry <a href="mailto:ccberry@users.r-forge.r-project.org">ccberry@users.r-forge.r-project.org</a>
</p>

<hr>
<h2 id='mstep'>
M-step to maximize theta
</h2><span id='topic+mstep'></span>

<h3>Description</h3>

<p>This is a utility function for <code><a href="#topic+maxEM">maxEM</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mstep(x, theta, phi)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mstep_+3A_x">x</code></td>
<td>

<p>a zero-one indicator matrix whose rows correspond to unique lengths with rownames
indicating those lengths
</p>
</td></tr>
<tr><td><code id="mstep_+3A_theta">theta</code></td>
<td>

<p>a vector of the abundance estimates. <code>length(theta)==ncol(x)</code>
</p>
</td></tr>
<tr><td><code id="mstep_+3A_phi">phi</code></td>
<td>
<p>a vector of the probabilities of sonicant lengths. <code>length(phi)==nrow(x)</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The M-step for theta is computed. Probably, there is no need to use
this function directly, but just in case it is here. 
</p>


<h3>Value</h3>

<p>a vector like the input <code>theta</code>
</p>


<h3>Author(s)</h3>

<p>Charles C. Berry
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maxEM">maxEM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
mat &lt;- diag(10)
mat[ ,10 ] &lt;- 1.0
phi1 &lt;- prop.table( rep(1,10))
theta1 &lt;- 1:10
sonicLength:::mstep( mat, theta1,phi1)
</code></pre>

<hr>
<h2 id='phi.update.lframe'>
estimate phi 
</h2><span id='topic+phi.update.lframe'></span>

<h3>Description</h3>

<p>Estimate phi in a flexible manner as guided by <code>lframe</code> - usually
involving multiple strata, and allowing for formulas other than the
default choice
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phi.update.lframe(obj, return.fit = FALSE, lframe, glm.frm = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phi.update.lframe_+3A_obj">obj</code></td>
<td>

<p>a matrix whose rowSums are used to estimate phi
</p>
</td></tr>
<tr><td><code id="phi.update.lframe_+3A_return.fit">return.fit</code></td>
<td>
<p> logical - 
the result of the fit should be returned instead of phi
</p>
</td></tr>
<tr><td><code id="phi.update.lframe_+3A_lframe">lframe</code></td>
<td>

<p>A <code>data.frame</code> with columns <code>y</code>, <code>x</code>, <code>strata</code>,
<code>orig</code> (or matching those in <code>glm.frm</code>) and for which
<code>rownames(lframe)[lframe$orig]</code> match the rownames of <code>obj</code>.
</p>
</td></tr>
<tr><td><code id="phi.update.lframe_+3A_glm.frm">glm.frm</code></td>
<td>

<p>a formula. If omitted <code> y ~ bs( x, knots= c(50,100) ) </code> will be used.
</p>
</td></tr>
<tr><td><code id="phi.update.lframe_+3A_...">...</code></td>
<td>

<p>curently not used
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>fitting phi - the probabilities that a sonicant lands in a particular
bin - is crucial to estimating theta, then number of sonicants when
more than one lans in a bin. bins may be defined by sonicant lengths
when there is just one sample. When there are multiple samples, then
one may wish to estimate phi separately for each
one. <code>lframe$strata</code> indicates separate sample and the default
<code>glm.frm</code> will fit each one separately. 
</p>


<h3>Value</h3>

<p>a vector whose sum is 1.0 with one element for each row of obj
</p>


<h3>Author(s)</h3>

<p>Charles C. Berry <a href="mailto:ccberry@users.r-forge.r-project.org">ccberry@users.r-forge.r-project.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maxEM">maxEM</a></code>
</p>

<hr>
<h2 id='simSonic'>Simulate Sonicated Data</h2><span id='topic+simSonic'></span>

<h3>Description</h3>

<p> Sonication and Processing of DNA containing retroviral
sequences yields genomic locations of retroviral insertion sites and a
set of associated fragment lengths. The simulation depends on the
expected number of fragments from each site and the distribution of
lengths. </p>


<h3>Usage</h3>

<pre><code class='language-R'>simSonic( theta, phi )
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simSonic_+3A_theta">theta</code></td>
<td>
<p>a vector of expected number of fragments from each
site. If the vector has unqiue names, these will be used to label the
locations</p>
</td></tr>
<tr><td><code id="simSonic_+3A_phi">phi</code></td>
<td>
<p> a vector of probabilities (or a list containing a vector
named <code>phi</code>) whose names are like <code>'rep len'</code>, where
<code>'rep'</code> defines the replicate and <code>'len'</code> defines the
sonicant length . More than one replicate can be used, In any case
<code>1-exp(-phi%o%theta)</code> determines the probability of
observations in the different locaitons, of the different lengths, and
in different replicates</p>
</td></tr>
</table>


<h3>Details</h3>

<p> This object can provide the arguments used by <code>estAbund</code>
</p>


<h3>Value</h3>

<p>a <code>data.frame</code> with columns <code>locations</code>,
<code>lengths</code>, and <code>replicates</code>. See <code><a href="#topic+estAbund">estAbund</a></code> for
more details.
</p>


<h3>Author(s)</h3>

<p>Charles C. Berry <a href="mailto:ccberry@users.r-forge.r-project.org">ccberry@users.r-forge.r-project.org</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>
theta &lt;- seq(0.5,20.5,by=0.5)
phi &lt;- prop.table(1:10)
names(phi) &lt;- paste( 1 , 51:60 )
res &lt;- simSonic( theta, phi )
head(res)
tail(res)
summary(res)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
