<!DOCTYPE html><html><head><title>Help for package SoyNAM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SoyNAM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BLUP'>
<p>Best Linear Unbias Predictor</p></a></li>
<li><a href='#Dataset'>
<p>Datasets</p></a></li>
<li><a href='#SoyNAM-package'>
<p>Soybean Nested Association Mapping Dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Soybean Nested Association Mapping Dataset</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-01-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Alencar Xavier, William Beavis, James Specht, Brian Diers, Rouf Mian, Reka Howard,
	George Graef, Randall Nelson, William Schapaugh, Dechun Wang, Grover Shannon, Leah McHale,
	Perry Cregan, Qijian Song, Miguel Lopez, William Muir, Katy Rainey.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alencar Xavier &lt;alenxav@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Genomic and multi-environmental soybean data. Soybean Nested
    Association Mapping (SoyNAM) project dataset funded by the United Soybean Board
    (USB). BLUP function formats data for genome-wide prediction and association analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>NAM, lme4, reshape2, Rcpp, RcppEigen</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-04 15:39:16 UTC; rd7564</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-04 17:20:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='BLUP'>
Best Linear Unbias Predictor
</h2><span id='topic+BLUP'></span><span id='topic+ENV'></span><span id='topic+gwnam'></span><span id='topic+mrr'></span>

<h3>Description</h3>

<p>Genetic values for a given trait computed by REML.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BLUP(trait="yield",family="all",env="all",dereg=FALSE,
     MAF=0.05,use.check=TRUE,impute="FM",rm.rep=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BLUP_+3A_trait">trait</code></td>
<td>

<p>Character. Trait of interest. The options are: &quot;yield&quot; (grain yield in Kg/ha), &quot;maturity&quot; (days to maturity), &quot;height&quot; (plant height in cm), &quot;lodging&quot; (lodging score from 1 to 5), &quot;protein&quot; (protein percentage in the grain), &quot;oil&quot; (oil percentage in the grain), &quot;size&quot; (seed size = mass of 100 seeds in grams) and &quot;fiber&quot; (fiber percentage in the grain).
</p>
</td></tr>
<tr><td><code id="BLUP_+3A_family">family</code></td>
<td>

<p>Numberic vector or &quot;all&quot;. Which SoyNAM families to use.
</p>
</td></tr>
<tr><td><code id="BLUP_+3A_env">env</code></td>
<td>

<p>Numberic vector or &quot;all&quot;. Which environments to use. The environments are coded as follows: 1 (IA_2012), 2 (IA_2013), 3 (IL_2011), 4 (IL_2012), 5 (IL_2013), 6 (IN_2012), 7 (IN_2013), 8 (KS_2012), 9 (KS_2013), 10 (MI_2012), 11 (MO_2012), 12 (MO_2013), 13 (NE_2011), 14 (NE_2012), 15 (OHmc_2012), 16 (OHmc_2013), 17 (OHmi_2012) and 18 (OHmi_2013).
</p>
</td></tr>
<tr><td><code id="BLUP_+3A_dereg">dereg</code></td>
<td>

<p>Logic. If true, deregress BLUPs (remove shrinkage).
</p>
</td></tr>
<tr><td><code id="BLUP_+3A_maf">MAF</code></td>
<td>

<p>Numeric. Minor allele frequency threshold for the markers.
</p>
</td></tr>
<tr><td><code id="BLUP_+3A_use.check">use.check</code></td>
<td>

<p>Logical. If TRUE, it includes a control term as fixed effect in the model.
</p>
</td></tr>
<tr><td><code id="BLUP_+3A_impute">impute</code></td>
<td>

<p>NULL, 'EXP' of 'FM'. If 'EXP', it imputes missing genotypes using expectation (allele frequency). If 'FM' is imputes missing genotypes using a forward Markov Chain algorithm, filling missing loci with the most likely genotype based on the previous marker.
</p>
</td></tr>
<tr><td><code id="BLUP_+3A_rm.rep">rm.rep</code></td>
<td>

<p>Logical. If TRUE, it removes replicated genotypes. Genotypes are treated as identical when the genotypes are more than 95 percent identical. This argument requires imputed genotypes.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses the raw dataset (<code class="reqn">data(soynam)</code>), allowing user-defined data quality control for genotypes and BLUPs of genetic values.
</p>
<p>The algorithm start from selecting the chosen families and environment that will be used for the best linear unbias predictor (BLUP). The BLUP values are calculates based on the following model: 
</p>
<p><code class="reqn">Trait = Control + Environment + Genotype</code>)
</p>
<p>Where control is a covariate set as fixed effect based on the  checks of each set (microenvironment); Environment is a random effect that represents the combination of location and year; and Genotype is the random effect associated to the lines. The BLUP values are the regression coefficients corresponding to the Genotype effect. The BLUP is calculated using the R package lme4 (Bates 2010) using REML.
</p>
<p>If checks are used as covariate (use.check=TRUE), then the best linear unbias estimator (BLUE) of the check effects is assigned to each set as a micro-environmental control. Each set had between one and five controls, including the SoyNAM parents and five other cultivars. These genotypes are normalized by environment and the BLUE of each set is calculated. All genotypes in a same set will have the same check effect.
</p>


<h3>Value</h3>

<p>This function returns a list with four objects. A numeric vector with the  BLUP solution of the phenotyes (&quot;Phen&quot;); the corresponding genotypes (&quot;Gen&quot;); a vector with the respective family (&quot;Fam&quot;); and a numeric vector with the number of SNPs per cromosome (&quot;Chrom&quot;). The output of this fuction has the exact input format for the NAM package (Xavier et al. 2015) to perform genome-wide association analysis.
</p>


<h3>Author(s)</h3>

<p>Alencar Xavier
</p>


<h3>References</h3>

<p>Bates, D. M. (2010). lme4: Mixed-effects modeling with R. URL http://lme4.r-forge.r-project.org/book.
</p>
<p>Xavier, A., Xu, S., Muir, W. M., &amp; Rainey, K. M. (2015). NAM: association studies in multiple populations. Bioinformatics, 31(23), 3862-3864.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Test=BLUP(trait="yield",family=2:3,env=1:2)
</code></pre>

<hr>
<h2 id='Dataset'>
Datasets
</h2><span id='topic+soynam'></span><span id='topic+data.check'></span><span id='topic+data.line'></span><span id='topic+gen.raw'></span><span id='topic+soybase'></span><span id='topic+data.check.qa'></span><span id='topic+data.line.qa'></span><span id='topic+gen.qa'></span><span id='topic+data.check.in'></span><span id='topic+data.line.in'></span><span id='topic+gen.in'></span>

<h3>Description</h3>

<p>Genotypes and phenotypes from quality assured (soybase) or original (soynam) datasets. An additional dataset containing yield components (soyin) collected at Purdue University is also available. See the section &quot;details&quot; for the description of data objects.
</p>
<p>The SoyNAM population (<a href="https://soybase.org/SoyNAM/index.php">https://soybase.org/SoyNAM/index.php</a>) is a nested association mapping panel that comprises more than 5000 recombinant inbred lines (RILs), including determinate, indeterminate, and semi-determinate genotypes from maturity groups (MG) ranging from late MG II to early MG IV, derived from 40 biparental populations, where progenies were not exposed to selection. Each biparental population approximately contains 140 individuals and all families share the cultivar IA3023 as the standard parent. From the other 40 founder parents, 17 lines are elite public germplasm from different regions, 15 have diverse ancestry and 8 are plant introductions. The SoyNAM population was designed to dissect the genetic architecture of complex traits and to map yield-associated quantitative trait loci (QTL) using a diverse panel.
</p>
<p>Parental lines were sequenced to derive the SNP allele calls. A total of 5303 SNP loci were selected with the criterion of maximizing the number of families segregating for those loci. The SNPs were used to build the SoyNAM 6K BreadChip SNP array using the Illumina Infinium HD Assay platform (Song et al. 2017). Among those SNPs, a subset of 4312 markers were selected by the SoyNAM group as quality-assured based on proportion of missing loci and correct segregation patterns. Both raw and quality assured genotypes are available in the R package SoyNAM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(soybase)
data(soynam)
data(soyin)
</code></pre>


<h3>Details</h3>

<p>Datasets of the SoyNAM project, original and quality assured (QA) versions. Data was downloaded on November 16th 2015 from <a href="https://soybase.org/SoyNAM/index.php">https://soybase.org/SoyNAM/index.php</a>. The data collected in Indiana was collected in 2013-2014 and made available on January 2018. Studies performed on the entire dataset with additional detail about the experimental settings include Diers et al. (2018) and Xavier et al. (2018).
</p>
<p>Genotypic matrices are named &quot;gen.raw&quot; and &quot;gen.qa&quot; for the raw and QA versions, respectively. Markers that were entirely missing were removed from the raw dataset. In each dataset, phenotypes are allocated into two objects, one with the lines (&quot;data.line&quot;) and one with checks and parents (&quot;data.checks&quot;). Information on data objects include year, location, environment (combination of year and location), strain, family, set (set in each environment), spot (combination of set and environment), height (in centimeters), R8 (number of days to maturity), planting date (501 represents may 1), flowering (701 represents july 1), maturity (901 represents september 1), lodging (score from 1 to 5), yield (in Kg/ha), moisture, protein (percentage in the seed), oil (percentage in the seed), fiber (percentage in the seed), seed size (in grams of 100 seeds).
</p>
<p>The dataset including yield components collected at Purdue University (West Lafayette, Indiana) was used to investigate genomic prediction (Xavier et al. 2016) and interaction among traits (Xavier et al. 2017). This dataset contains the genotypic information in the matrix &quot;gen.in&quot;, with missing values imputed using the software MaCH (Li et al. 2010). Similar to the datasets previously described, phenotypes are allocated into two objects, lines (&quot;data.line.in&quot;) and checks (&quot;data.checks.in&quot;). Information on these data objects include year, location, environment (combination of year and location), strain, family, set (set in each environment), spot (combination of set and environment), the spatial coordinates of the field plots (BLOCK, ROW and COLUMN), plant height (in centimeters), R1 (number of days to flowering), R8 (number of days to maturity), lodging (score from 1 to 5), yield (in bu/ac), leaf shape (ratio length:width), number of nodes in the main stem, number of pods in the main stem, number of pods per node, average canopy coverage, rate of canopy coverage, growing degree day to flowering (GDD_R1), growing degree day to maturity (GDD_R1), and length of reproductive period in terms of growing degree day (GDD_REP).
</p>


<h3>Author(s)</h3>

<p>Alencar Xavier
</p>


<h3>References</h3>

<p>Diers, B. W., Specht, J., Rainey, K. M., Cregan, P., Song, Q., Ramasubramanian, V., ... &amp; Shannon, G. (2018). Genetic Architecture of Soybean Yield and Agronomic Traits. G3: Genes, Genomes, Genetics, g3-200332.
</p>
<p>Li, Y., Willer, C. J., Ding, J., Scheet, P., &amp; Abecasis, G. R. (2010). MaCH: using sequence and genotype data to estimate haplotypes and unobserved genotypes. Genetic epidemiology, 34(8), 816-834.
</p>
<p>Song, Q., Yan, L., Quigley, C., Jordan, B.D., Fickus, E., Schroeder, S., Song, B., An, Y. Q. C., Hyten, D., Nelson, R., Rainey, K. M., Beavis, W. D., Specht, J. E., Diers, B. W., Cregan, P. (2017). Development and Genetic Characterization of the Soybean Nested Association Mapping (NAM) Population. Plant Genome. 10(2):1-14.
</p>
<p>Xavier, A., Muir, W. M., &amp; Rainey, K. M. (2016). Assessing predictive properties of genome-wide selection in soybeans. G3: Genes, Genomes, Genetics, 6(8), 2611-2616.
</p>
<p>Xavier, A., Hall, B., Casteel, S., Muir, W., &amp; Rainey, K. M. (2017). Using unsupervised learning techniques to assess interactions among complex traits in soybeans. Euphytica, 213(8), 200.
</p>
<p>Xavier, A., Jarquin, D., Howard, R., Ramasubramanian, V., Specht, J. E., Graef, G. L., ... &amp; Nelson, R. (2017). Genome-Wide analysis of grain yield stability and environmental interactions in a multiparental soybean population. G3: Genes, Genomes, Genetics, g3-300300.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soynam)
data(soybase)
data(soyin)
</code></pre>

<hr>
<h2 id='SoyNAM-package'>
Soybean Nested Association Mapping Dataset
</h2><span id='topic+SoyNAM-package'></span>

<h3>Description</h3>

<p>Genomic and multi-environmental soybean data. Soybean Nested
Association Mapping (SoyNAM) project dataset funded by the United Soybean Board
(USB). BLUP function formats data for genome-wide prediction and association analysis.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> NAM</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.6.2 </td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-01-03</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Alencar Xavier, William Beavis, James Specht, Brian Diers, Rouf Mian, Reka Howard,
George Graef, Randall Nelson, William Schapaugh, Dechun Wang, Grover Shannon, Leah McHale,
Perry Cregan, Qijian Song, Miguel Lopez, William Muir, Katy Rainey.
</p>
<p>Maintainer: Alencar Xavier &lt;alenxav@gmail.com&gt;
</p>


<h3>See Also</h3>

<p>Functions: BLUP, data(soybase), data(soynam) and data(soyin).
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
