<!DOCTYPE html><html lang="en"><head><title>Help for package SpaCCI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SpaCCI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Find_regional_IDs'><p>Find the spatial neighborhood Spot IDs</p></a></li>
<li><a href='#GetShuffledCT'><p>Perform a Deranged Shuffle of Cell Types</p></a></li>
<li><a href='#Global_Permutations'><p>Perform Global Permutations</p></a></li>
<li><a href='#Local_Regional_Permutations'><p>Perform Local and Regional Permutations</p></a></li>
<li><a href='#LR_database'><p>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</p></a></li>
<li><a href='#plot_localized'><p>Plot Localized Hotspot Pattern</p></a></li>
<li><a href='#plot_localized_Seurat'><p>Plot Localized Hotspot Pattern on Seurat Object</p></a></li>
<li><a href='#plot_SpaCCI_chordDiagram'><p>Plot SpaCCI Results on Chord Diagram</p></a></li>
<li><a href='#plot_SpaCCI_heatmap'><p>Plot SpaCCI Results on the heatmap</p>
Visualize inferred significant cell-cell interactions using a heatmap</a></li>
<li><a href='#plot_SpaCCI_local'><p>Plot SpaCCI Localized Interaction Results</p></a></li>
<li><a href='#plot_SpaCCI_local_Strength'><p>Plot SpaCCI Localized Interaction Strength Results</p></a></li>
<li><a href='#plot_Strength_localized'><p>Plot Localized Hotspot Strength Pattern</p></a></li>
<li><a href='#plot_Strength_Seurat'><p>Plot Localized Hotspot Strength Pattern on Seurat Object</p></a></li>
<li><a href='#possible_L_R_pairs_cellchat'><p>CellChat Database: Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</p></a></li>
<li><a href='#possible_L_R_pairs_Cellinker'><p>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</p></a></li>
<li><a href='#possible_L_R_pairs_cellphoneDB'><p>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</p></a></li>
<li><a href='#possible_L_R_pairs_connectome'><p>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</p></a></li>
<li><a href='#possible_L_R_pairs_ICELLNET'><p>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</p></a></li>
<li><a href='#random_region'><p>Select Closest Spatial IDs to a Center Point: this is used for permutation</p></a></li>
<li><a href='#result_global'><p>result_global data for SpaCCI</p></a></li>
<li><a href='#result_local'><p>result_local data and spatial coordinates for SpaCCI</p>
Local Regional Data</a></li>
<li><a href='#result_local_spatial_coords_df'><p>Local Spatial Coordinates Data Frame</p></a></li>
<li><a href='#result_regional'><p>result_regional data for SpaCCI</p></a></li>
<li><a href='#run_SpaCCI'><p>Run SpaCCI Analysis</p></a></li>
<li><a href='#scPalette'><p>Generate a Color Palette</p></a></li>
<li><a href='#SpaCCI_global'><p>Infer Cell-Cell Interactions on a Global Scale</p></a></li>
<li><a href='#SpaCCI_local'><p>Infer Cell-Cell Interactions on a Local Scale</p></a></li>
<li><a href='#SpaCCI_region'><p>Infer Cell-Cell Interactions in a Specified Region</p></a></li>
<li><a href='#test_data'><p>Test data for SpaCCI</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Spatially Aware Cell-Cell Interaction Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-01-18</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Li-Ting Ku &lt;lku@mdanderson.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides tools for analyzing spatial cell-cell interactions based on ligand-receptor pairs, including functions for local, regional, and global analysis using spatial transcriptomics data. Integrates with databases like 'CellChat' <a href="http://www.cellchat.org/">http://www.cellchat.org/</a>, 'CellPhoneDB' <a href="https://www.cellphonedb.org/">https://www.cellphonedb.org/</a>, 'Cellinker' <a href="https://www.rna-society.org/cellinker/">https://www.rna-society.org/cellinker/</a>, 'ICELLNET' <a href="https://github.com/soumelis-lab/ICELLNET">https://github.com/soumelis-lab/ICELLNET</a>, and 'ConnectomeDB' <a href="https://humanconnectome.org/software/connectomedb/">https://humanconnectome.org/software/connectomedb/</a> to identify ligand-receptor pairs, visualize interactions through heatmaps, chord diagrams, and infer interactions on different spatial scales. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.13), Seurat (&ge; 4.0.0), nnls, ggrepel, pheatmap,
circlize (&ge; 0.4.12), Matrix, dplyr, patchwork, grDevices,
reshape2, graphics, ggplot2, FNN</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-18 22:31:41 UTC; lku</td>
</tr>
<tr>
<td>Author:</td>
<td>Li-Ting Ku [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-18 23:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Find_regional_IDs'>Find the spatial neighborhood Spot IDs</h2><span id='topic+Find_regional_IDs'></span>

<h3>Description</h3>

<p>This function identifies the spatial neighborhood Spot IDs around a given center Spot ID within a specified radius.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Find_regional_IDs(
  object,
  spatial_coord,
  centerID,
  enhanced = FALSE,
  radius,
  avern = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Find_regional_IDs_+3A_object">object</code></td>
<td>
<p>An object that could be either 1. A Seurat object, or 2. A data frame where the columns are Spot_IDs (i.e., the gene*spot expression matrix).</p>
</td></tr>
<tr><td><code id="Find_regional_IDs_+3A_spatial_coord">spatial_coord</code></td>
<td>
<p>A data frame of the spatial coordinates. The column names should include 'c(&quot;Spot_ID&quot;, &quot;imagerow&quot;, &quot;imagecol&quot;)', and the row names must be the Spot_ID, which is the same as the row names in the cell type proportion data frame or the column names of the gene*spot expression data frame.</p>
</td></tr>
<tr><td><code id="Find_regional_IDs_+3A_centerid">centerID</code></td>
<td>
<p>A vector of length 1, representing a single Spot_ID that serves as the center for the neighborhood.</p>
</td></tr>
<tr><td><code id="Find_regional_IDs_+3A_enhanced">enhanced</code></td>
<td>
<p>Logical; if 'TRUE', enhances the Seurat object by marking the neighborhood in a special way. Defaults to 'FALSE'.</p>
</td></tr>
<tr><td><code id="Find_regional_IDs_+3A_radius">radius</code></td>
<td>
<p>The radius of the spatial neighborhood, specified as a numeric value.</p>
</td></tr>
<tr><td><code id="Find_regional_IDs_+3A_avern">avern</code></td>
<td>
<p>Numeric; the number of samples to average over when determining the unit distance. Defaults to 5.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>centerID</dt><dd><p>The input center Spot_ID.</p>
</dd>
<dt>closeID</dt><dd><p>The Spot_IDs of the neighboring spots within the specified radius.</p>
</dd>
<dt>unit</dt><dd><p>The unit distance used to determine the neighborhood.</p>
</dd>
</dl>


<hr>
<h2 id='GetShuffledCT'>Perform a Deranged Shuffle of Cell Types</h2><span id='topic+GetShuffledCT'></span>

<h3>Description</h3>

<p>This function takes a vector of cell types and returns a shuffled version where no element remains in its original position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetShuffledCT(CellType)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetShuffledCT_+3A_celltype">CellType</code></td>
<td>
<p>A character vector representing the cell types to be shuffled.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the same length as 'CellType', with elements shuffled such that no element remains in its original position.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
original &lt;- c("B_cell", "T_cell", "NK_cell", "Macrophage")
shuffled &lt;- GetShuffledCT(original)
print(shuffled)


</code></pre>

<hr>
<h2 id='Global_Permutations'>Perform Global Permutations</h2><span id='topic+Global_Permutations'></span>

<h3>Description</h3>

<p>This function performs global permutations on the spatial transcriptomics data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Global_Permutations(
  permutationMatrix,
  permut_col,
  cellPropMatrix,
  spotGeneMatrix,
  LigandVectorIndex,
  ReceptorVectorIndex,
  null_expression,
  nBoot
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Global_Permutations_+3A_permutationmatrix">permutationMatrix</code></td>
<td>
<p>A matrix containing permutations.</p>
</td></tr>
<tr><td><code id="Global_Permutations_+3A_permut_col">permut_col</code></td>
<td>
<p>A column matrix of permutations.</p>
</td></tr>
<tr><td><code id="Global_Permutations_+3A_cellpropmatrix">cellPropMatrix</code></td>
<td>
<p>A matrix of cell type proportions.</p>
</td></tr>
<tr><td><code id="Global_Permutations_+3A_spotgenematrix">spotGeneMatrix</code></td>
<td>
<p>A matrix of gene expressions at spots.</p>
</td></tr>
<tr><td><code id="Global_Permutations_+3A_ligandvectorindex">LigandVectorIndex</code></td>
<td>
<p>A vector of ligand indices.</p>
</td></tr>
<tr><td><code id="Global_Permutations_+3A_receptorvectorindex">ReceptorVectorIndex</code></td>
<td>
<p>A vector of receptor indices.</p>
</td></tr>
<tr><td><code id="Global_Permutations_+3A_null_expression">null_expression</code></td>
<td>
<p>A matrix of null expression values.</p>
</td></tr>
<tr><td><code id="Global_Permutations_+3A_nboot">nBoot</code></td>
<td>
<p>Number of bootstrap iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with the results of the global permutations.
</p>

<hr>
<h2 id='Local_Regional_Permutations'>Perform Local and Regional Permutations</h2><span id='topic+Local_Regional_Permutations'></span>

<h3>Description</h3>

<p>This function performs local and regional permutations on the spatial transcriptomics data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Local_Regional_Permutations(
  permutationMatrix,
  permut_col,
  cellPropMatrix,
  spotGeneMatrix,
  LigandVectorIndex,
  ReceptorVectorIndex,
  null_expression,
  nBoot
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Local_Regional_Permutations_+3A_permutationmatrix">permutationMatrix</code></td>
<td>
<p>A matrix containing permutations.</p>
</td></tr>
<tr><td><code id="Local_Regional_Permutations_+3A_permut_col">permut_col</code></td>
<td>
<p>A column matrix of permutations.</p>
</td></tr>
<tr><td><code id="Local_Regional_Permutations_+3A_cellpropmatrix">cellPropMatrix</code></td>
<td>
<p>A matrix of cell type proportions.</p>
</td></tr>
<tr><td><code id="Local_Regional_Permutations_+3A_spotgenematrix">spotGeneMatrix</code></td>
<td>
<p>A matrix of gene expressions at spots.</p>
</td></tr>
<tr><td><code id="Local_Regional_Permutations_+3A_ligandvectorindex">LigandVectorIndex</code></td>
<td>
<p>A vector of ligand indices.</p>
</td></tr>
<tr><td><code id="Local_Regional_Permutations_+3A_receptorvectorindex">ReceptorVectorIndex</code></td>
<td>
<p>A vector of receptor indices.</p>
</td></tr>
<tr><td><code id="Local_Regional_Permutations_+3A_null_expression">null_expression</code></td>
<td>
<p>A matrix of null expression values.</p>
</td></tr>
<tr><td><code id="Local_Regional_Permutations_+3A_nboot">nBoot</code></td>
<td>
<p>Number of bootstrap iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with the results of the local and regional permutations.
</p>

<hr>
<h2 id='LR_database'>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</h2><span id='topic+LR_database'></span>

<h3>Description</h3>

<p>This function identifies possible ligand-receptor (L-R) pairs for cell-cell communication analysis using a selected database. It checks for the presence of all genes involved in each L-R pair within the provided gene expression matrix, filtering based on a specified expression percentage threshold. The function supports multiple databases including CellChat, CellPhoneDB, Cellinker, ICELLNET, and ConnectomeDB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LR_database(
  species,
  database_name,
  gene_spot_expression_dataframe,
  percentage = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LR_database_+3A_species">species</code></td>
<td>
<p>A string specifying the species (<code>"Human"</code> or <code>"Mouse"</code>).</p>
</td></tr>
<tr><td><code id="LR_database_+3A_database_name">database_name</code></td>
<td>
<p>A string specifying the L-R database to use. Options include <code>"CellChat"</code>, <code>"CellPhoneDB"</code>, <code>"Cellinker"</code>, <code>"ICELLNET"</code>, and <code>"ConnectomeDB"</code>.</p>
</td></tr>
<tr><td><code id="LR_database_+3A_gene_spot_expression_dataframe">gene_spot_expression_dataframe</code></td>
<td>
<p>A gene expression data frame with genes as row names and Spot IDs as column names. This data frame is used to verify the presence of all genes involved in the L-R pairs.</p>
</td></tr>
<tr><td><code id="LR_database_+3A_percentage">percentage</code></td>
<td>
<p>A numeric value specifying the minimum percentage of spots in which a gene must be expressed to be considered. The default is <code>10</code>, meaning the gene express over 10% of spots.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt><code>possible_LR_pairs</code></dt><dd><p>A data frame of L-R pairs where all genes are present in the <code>gene_spot_expression_dataframe</code> and meet the expression threshold. The data frame includes the ligand and receptor vectors, and the combined gene vectors.</p>
</dd>
<dt><code>possible_LR_pairs_info</code></dt><dd><p>A data frame with detailed information about the identified L-R pairs, including their original annotations from the selected database.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>library(SpaCCI)
#Load the example data
data(test_data)
gene_spot_df &lt;- test_data$gene_spot_df
result &lt;- LR_database(species = "Human",
            database_name = "CellChat",
            gene_spot_expression_dataframe = gene_spot_df)
</code></pre>

<hr>
<h2 id='plot_localized'>Plot Localized Hotspot Pattern</h2><span id='topic+plot_localized'></span>

<h3>Description</h3>

<p>Visualize the inferred cell-cell interaction localized pattern if NOT using Seurat Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_localized(
  spatial_coord,
  resultdf_list,
  RegionIDs_matrix,
  celltype_ligand,
  celltype_receptor,
  plot_size,
  L_R_pair_name = NULL,
  alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_localized_+3A_spatial_coord">spatial_coord</code></td>
<td>
<p>A data frame of the spatial coordinates. The columns should include <code>"Spot_ID"</code>, <code>"imagerow"</code>, and <code>"imagecol"</code>. And the row names must be the names of <code>"Spot_ID"</code>, which is the same as the rownames in cell type proportion data frame or the colnames of the gene* spot expression data frame</p>
</td></tr>
<tr><td><code id="plot_localized_+3A_resultdf_list">resultdf_list</code></td>
<td>
<p>A result of data frame list from the output of <code>run_SpaCCI(..., analysis_scale = "local",...)</code> <code>`dataframelist`</code></p>
</td></tr>
<tr><td><code id="plot_localized_+3A_regionids_matrix">RegionIDs_matrix</code></td>
<td>
<p>A result of matrix list from the output of <code>run_SpaCCI(..., analysis_scale = "local",...)</code> <code>`RegionIDs_matrix`</code></p>
</td></tr>
<tr><td><code id="plot_localized_+3A_celltype_ligand">celltype_ligand</code></td>
<td>
<p>Ligand cell type string inputted by user, the name of the cell type should match the names in the <code>`spot_cell_proportion_dataframe`</code> during the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_localized_+3A_celltype_receptor">celltype_receptor</code></td>
<td>
<p>Receptor cell type string inputted by user, the name of the cell type should match the names in the <code>`spot_cell_proportion_dataframe`</code> during the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_localized_+3A_plot_size">plot_size</code></td>
<td>
<p>As this function incorporate with <code>Seurat</code>'s <code>`SpatialFeaturePlot`</code>, this parameter could control the plotting size of the each spot.</p>
</td></tr>
<tr><td><code id="plot_localized_+3A_l_r_pair_name">L_R_pair_name</code></td>
<td>
<p>Initially this is set to <code>NULL</code>, if one is interested in a specific Ligand-Receptor pair, then one could specify the L_R_pair_name here. Note: the input name should match the L-R pair name exists in the dataframe in the output of SpaCCI_local &quot;dataframelist&quot;.</p>
</td></tr>
<tr><td><code id="plot_localized_+3A_alpha">alpha</code></td>
<td>
<p>This is the significant cutoff for the adjusted-p-value of thr permutation test. Initially this is set to <code>0.05</code>, one could adjust the cutoff.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The localized plot from the inferred cell-cell interaction on the local scale.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Run localized hotspot plot
Result &lt;- run_SpaCCI(..., analysis_scale = "local",...)
local_plot &lt;- plot_localized(spatial_coord = spatial_coords_df,
                             resultdf_list = Result$dataframelist,
                             RegionIDs_matrix = Result$RegionIDs_matrix,
                             celltype_ligand = "Beta_cells",
                             celltype_receptor = "T_ells",
                             plot_size = 3)


</code></pre>

<hr>
<h2 id='plot_localized_Seurat'>Plot Localized Hotspot Pattern on Seurat Object</h2><span id='topic+plot_localized_Seurat'></span>

<h3>Description</h3>

<p>Visualize the inferred cell-cell interaction localized pattern on the tissue image with Seurat_object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_localized_Seurat(
  Seurat_object,
  resultdf_list,
  RegionIDs_matrix,
  celltype_ligand,
  celltype_receptor,
  plot_size,
  L_R_pair_name = NULL,
  alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_localized_Seurat_+3A_seurat_object">Seurat_object</code></td>
<td>
<p>A Seurat object</p>
</td></tr>
<tr><td><code id="plot_localized_Seurat_+3A_resultdf_list">resultdf_list</code></td>
<td>
<p>A result of data frame list from the output of <code>run_SpaCCI(..., analysis_scale = "local",...)</code> <code>`dataframelist`</code></p>
</td></tr>
<tr><td><code id="plot_localized_Seurat_+3A_regionids_matrix">RegionIDs_matrix</code></td>
<td>
<p>A result of matrix list from the output of <code>run_SpaCCI(..., analysis_scale = "local",...)</code> <code>`RegionIDs_matrix`</code></p>
</td></tr>
<tr><td><code id="plot_localized_Seurat_+3A_celltype_ligand">celltype_ligand</code></td>
<td>
<p>Ligand cell type string inputted by user, the name of the cell type should match the names in the <code>`spot_cell_proportion_dataframe`</code> during the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_localized_Seurat_+3A_celltype_receptor">celltype_receptor</code></td>
<td>
<p>Receptor cell type string inputted by user, the name of the cell type should match the names in the <code>`spot_cell_proportion_dataframe`</code> during the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_localized_Seurat_+3A_plot_size">plot_size</code></td>
<td>
<p>As this function incorporate with <code>Seurat</code>'s <code>`SpatialFeaturePlot`</code>, this parameter could control the plotting size of the each spot.</p>
</td></tr>
<tr><td><code id="plot_localized_Seurat_+3A_l_r_pair_name">L_R_pair_name</code></td>
<td>
<p>Initially this is set to <code>NULL</code>, if one is interested in a specific Ligand-Receptor pair, then one could specify the L_R_pair_name here. Note: the input name should match the L-R pair name exists in the dataframe in the output of SpaCCI_local &quot;dataframelist&quot;.</p>
</td></tr>
<tr><td><code id="plot_localized_Seurat_+3A_alpha">alpha</code></td>
<td>
<p>This is the significant cutoff for the adjusted-p-value of thr permutation test. Initially this is set to <code>0.05</code>, one could adjust the cutoff.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The localized plot from the inferred cell-cell interaction on the local scale.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Not Run

# Run localized hotspot plot
Result &lt;- run_SpaCCI(..., analysis_scale = "local",...)
local_plot &lt;- plot_localized_Seurat(Seurat_object = gene_spot_df,
                                    resultdf_list = Result$dataframelist,
                                    RegionIDs_matrix = Result$RegionIDs_matrix,
                                    celltype_ligand = "Beta_cells",
                                    celltype_receptor = "T_ells",
                                    plot_size = 3)


</code></pre>

<hr>
<h2 id='plot_SpaCCI_chordDiagram'>Plot SpaCCI Results on Chord Diagram</h2><span id='topic+plot_SpaCCI_chordDiagram'></span>

<h3>Description</h3>

<p>This function generates a chord diagram to visualize cell-cell interactions based on ligand-receptor pairs. The interactions can be filtered by specific cell types, pathways, or interaction names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_SpaCCI_chordDiagram(
  SpaCCI_Result_List,
  specific_celltypes = NULL,
  pathway_name = NULL,
  L_R_pair_name = NULL,
  color = NULL,
  alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_SpaCCI_chordDiagram_+3A_spacci_result_list">SpaCCI_Result_List</code></td>
<td>
<p>A list containing the results from a SpaCCI <code>"regional"</code> or <code>"global"</code> analysis. This list should include <code>pvalue_df</code>, which are the outputs from <code>run_SpaCCI(..., analysis_scale = "regional",...)</code> or  <code>run_SpaCCI(..., analysis_scale = "global",...)</code>.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_chordDiagram_+3A_specific_celltypes">specific_celltypes</code></td>
<td>
<p>A character vector specifying the cell types to include in the plot, RECOMMEND using colnames of cell type proportion matrix to include all cell types. If <code>NULL</code>, cell types that involved in significant interactions are included.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_chordDiagram_+3A_pathway_name">pathway_name</code></td>
<td>
<p>A single character string specifying the pathway name to filter the interactions. If <code>NULL</code>, all pathways are included.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_chordDiagram_+3A_l_r_pair_name">L_R_pair_name</code></td>
<td>
<p>A character vector specifying the ligand-receptor pair names to include in the plot. If <code>NULL</code>, all interactions are included.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_chordDiagram_+3A_color">color</code></td>
<td>
<p>A named vector of colors to use for the cell types. If <code>NULL</code>, a default color palette is used.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_chordDiagram_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value specifying the significance threshold for adjusted P-values. Initially, set to <code>0.05</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A chord diagram plot visualizing the significant cell-cell interactions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SpaCCI)
library(dplyr)
library(circlize)
data(result_global)
celltypes &lt;- c("beta" , "delta" , "ductal","macrophage",
                "activated_stellate", "quiescent_stellate")
# Run the result chordDiagram for global analysis
plot_SpaCCI_chordDiagram(SpaCCI_Result_List = result_global,
                         specific_celltypes = c(celltypes),
                         L_R_pair_name  = "AREG_EGFR")

</code></pre>

<hr>
<h2 id='plot_SpaCCI_heatmap'>Plot SpaCCI Results on the heatmap
Visualize inferred significant cell-cell interactions using a heatmap</h2><span id='topic+plot_SpaCCI_heatmap'></span>

<h3>Description</h3>

<p>Plot SpaCCI Results on the heatmap
Visualize inferred significant cell-cell interactions using a heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_SpaCCI_heatmap(
  SpaCCI_Result_List,
  specific_celltypes = NULL,
  pathways = NULL,
  interaction = NULL,
  log1p_transform = FALSE,
  show_rownames = TRUE,
  show_colnames = TRUE,
  scale = "none",
  cluster_cols = TRUE,
  cluster_rows = TRUE,
  border_color = "white",
  fontsize_row = 11,
  fontsize_col = 11,
  family = "Arial",
  main = "",
  treeheight_col = 0,
  treeheight_row = 0,
  low_col = "dodgerblue4",
  mid_col = "peachpuff",
  high_col = "deeppink4",
  alpha = 0.05,
  return_tables = FALSE,
  symmetrical = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_SpaCCI_heatmap_+3A_spacci_result_list">SpaCCI_Result_List</code></td>
<td>
<p>A list containing the results from a SpaCCI <code>"regional"</code> or <code>"global"</code> analysis. This list should include <code>pvalue_df</code>, which are the outputs from <code>run_SpaCCI(..., analysis_scale = "regional",...)</code> or  <code>run_SpaCCI(..., analysis_scale = "global",...)</code>.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_specific_celltypes">specific_celltypes</code></td>
<td>
<p>A vector of cell types to include in the heatmap, i.e c(&quot;Celltype_A&quot;,&quot;Celltype_B&quot;). NOTE: the cell type names should match the names input in the SpaCCI analysis.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_pathways">pathways</code></td>
<td>
<p>A vector of pathways to filter the interactions. Initially set to <code>NULL</code>, if not, then it will aggregate the results of the selected pathways.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_interaction">interaction</code></td>
<td>
<p>A vector of interactions to filter. Initially set to <code>NULL</code>, if not, then it will aggregate the results of the selected interactions.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_log1p_transform">log1p_transform</code></td>
<td>
<p>Logical; whether to apply a log(1 + x) transformation to the count matrix.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_show_rownames">show_rownames</code></td>
<td>
<p>Logical; whether to show row names in the heatmap.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_show_colnames">show_colnames</code></td>
<td>
<p>Logical; whether to show column names in the heatmap.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_scale">scale</code></td>
<td>
<p>Character; whether to scale the data (&quot;row&quot;, &quot;column&quot;, &quot;none&quot;).</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>Logical; whether to cluster columns.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>Logical; whether to cluster rows.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_border_color">border_color</code></td>
<td>
<p>Character; color of the heatmap borders.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_fontsize_row">fontsize_row</code></td>
<td>
<p>Numeric; font size for row names.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_fontsize_col">fontsize_col</code></td>
<td>
<p>Numeric; font size for column names.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_family">family</code></td>
<td>
<p>Character; font family for text in the heatmap.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_main">main</code></td>
<td>
<p>Character; title of the heatmap.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_treeheight_col">treeheight_col</code></td>
<td>
<p>Numeric; height of the column dendrogram.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_treeheight_row">treeheight_row</code></td>
<td>
<p>Numeric; height of the row dendrogram.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_low_col">low_col</code></td>
<td>
<p>Character; color for low values in the heatmap.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_mid_col">mid_col</code></td>
<td>
<p>Character; color for mid values in the heatmap.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_high_col">high_col</code></td>
<td>
<p>Character; color for high values in the heatmap.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_alpha">alpha</code></td>
<td>
<p>Numeric; significance threshold for p-values, initailly set to <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_return_tables">return_tables</code></td>
<td>
<p>Logical; whether to return the count matrix and summary tables.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_symmetrical">symmetrical</code></td>
<td>
<p>Logical; whether to make the heatmap symmetrical.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_heatmap_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to 'pheatmap'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If 'return_tables' is <code>FALSE</code> (default), the function returns a heatmap object created by <code>pheatmap</code>, showing the count of significant cell-cell interactions. If 'return_tables' is <code>TRUE</code>, the function returns a list containing:
</p>

<dl>
<dt><strong>heatmap</strong></dt><dd><p>The heatmap object showing the significant cell-cell interactions.</p>
</dd>
<dt><strong>heatmap_countmatrix</strong></dt><dd><p>The matrix used to generate the heatmap, with cell types as rows and columns, and counts of significant interactions as values.</p>
</dd>
<dt><strong>table</strong></dt><dd><p>A data frame summarizing the counts of significant interactions between each ligand and receptor cell type combination.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>library(SpaCCI)
library(dplyr)
library(reshape2)
library(grDevices)
library(pheatmap)
data(result_global)
celltypes &lt;- c("beta" , "delta" , "ductal","macrophage",
                "activated_stellate", "quiescent_stellate")
plot_SpaCCI_heatmap(SpaCCI_Result_List = result_global,
                    symmetrical = FALSE, cluster_cols = FALSE, return_tables = FALSE,
                    cluster_rows = FALSE, #cellheight = 10, cellwidth = 10,
                    specific_celltypes = c(celltypes),
                    main= "Cell-Cell Interaction Count")

</code></pre>

<hr>
<h2 id='plot_SpaCCI_local'>Plot SpaCCI Localized Interaction Results</h2><span id='topic+plot_SpaCCI_local'></span>

<h3>Description</h3>

<p>This function provides a unified interface to visualize the localized cell-cell interaction patterns inferred by SpaCCI, either using a Seurat object with a spatial image or a spatial coordinates data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_SpaCCI_local(
  Seurat_Object = NULL,
  spatial_coordinates_dataframe = NULL,
  SpaCCI_local_Result_List,
  Ligand_cell_type,
  Receptor_cell_type,
  spot_plot_size,
  specific_LR_pair_name = NULL,
  significant_cutoff = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_SpaCCI_local_+3A_seurat_object">Seurat_Object</code></td>
<td>
<p>Optional. A Seurat object containing spatial data. If provided, the function will plot the interaction patterns on the tissue image.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_+3A_spatial_coordinates_dataframe">spatial_coordinates_dataframe</code></td>
<td>
<p>Optional. A data frame containing the spatial coordinates of the spots. The columns should include <code>"Spot_ID"</code>, <code>"imagerow"</code>, and <code>"imagecol"</code>. The row names must be the names of <code>"Spot_ID"</code>, matching those in the cell type proportion data frame or the gene expression data frame.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_+3A_spacci_local_result_list">SpaCCI_local_Result_List</code></td>
<td>
<p>A list containing the results from a SpaCCI local analysis. This list should include <code>dataframelist</code> and <code>RegionIDs_matrix</code>, which are the outputs from <code>run_SpaCCI(..., analysis_scale = "local",...)</code>.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_+3A_ligand_cell_type">Ligand_cell_type</code></td>
<td>
<p>The name of the ligand cell type to plot. This should match the cell type names used in the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_+3A_receptor_cell_type">Receptor_cell_type</code></td>
<td>
<p>The name of the receptor cell type to plot. This should match the cell type names used in the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_+3A_spot_plot_size">spot_plot_size</code></td>
<td>
<p>A numeric value controlling the size of the spots in the plot.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_+3A_specific_lr_pair_name">specific_LR_pair_name</code></td>
<td>
<p>Optional. The name of a specific ligand-receptor pair to plot. If provided, the plot will focus on this interaction. The name should match those in the <code>SpaCCI_local_Result_List$dataframelist</code>.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_+3A_significant_cutoff">significant_cutoff</code></td>
<td>
<p>A numeric value specifying the significance cutoff for the adjusted P-values from the permutation test. Default is <code>0.05</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot object showing the localized interaction patterns. The plot will be generated using either the Seurat object or the spatial coordinates data frame, depending on the input provided.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot localized SpaCCI results using Seurat object
library(SpaCCI)
library(dplyr)
data(result_local)
data(result_local_spatial_coords_df)
spatial_coords_df &lt;- result_local_spatial_coords_df
#plot_SpaCCI_local(Seurat_Object = seurat_object,.....)

# Plot localized SpaCCI results using spatial coordinates
plot_SpaCCI_local(spatial_coordinates_dataframe = spatial_coords_df,
                  SpaCCI_local_Result_List = result_local,
                  Ligand_cell_type = "ductal",
                  Receptor_cell_type = "activated_stellate",
                  spot_plot_size = 3)

</code></pre>

<hr>
<h2 id='plot_SpaCCI_local_Strength'>Plot SpaCCI Localized Interaction Strength Results</h2><span id='topic+plot_SpaCCI_local_Strength'></span>

<h3>Description</h3>

<p>This function provides a unified interface to visualize the localized cell-cell interaction strength patterns inferred by SpaCCI, either using a Seurat object with a spatial image or a spatial coordinates data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_SpaCCI_local_Strength(
  Seurat_Object = NULL,
  spatial_coordinates_dataframe = NULL,
  SpaCCI_local_Result_List,
  Ligand_cell_type,
  Receptor_cell_type,
  spot_plot_size,
  specific_LR_pair_name = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_SpaCCI_local_Strength_+3A_seurat_object">Seurat_Object</code></td>
<td>
<p>Optional. A Seurat object containing spatial data. If provided, the function will plot the interaction patterns on the tissue image.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_Strength_+3A_spatial_coordinates_dataframe">spatial_coordinates_dataframe</code></td>
<td>
<p>Optional. A data frame containing the spatial coordinates of the spots. The columns should include <code>"Spot_ID"</code>, <code>"imagerow"</code>, and <code>"imagecol"</code>. The row names must be the names of <code>"Spot_ID"</code>, matching those in the cell type proportion data frame or the gene expression data frame.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_Strength_+3A_spacci_local_result_list">SpaCCI_local_Result_List</code></td>
<td>
<p>A list containing the results from a SpaCCI local analysis. This list should include <code>dataframelist</code> and <code>RegionIDs_matrix</code>, which are the outputs from <code>run_SpaCCI(..., analysis_scale = "local",...)</code>.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_Strength_+3A_ligand_cell_type">Ligand_cell_type</code></td>
<td>
<p>The name of the ligand cell type to plot. This should match the cell type names used in the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_Strength_+3A_receptor_cell_type">Receptor_cell_type</code></td>
<td>
<p>The name of the receptor cell type to plot. This should match the cell type names used in the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_Strength_+3A_spot_plot_size">spot_plot_size</code></td>
<td>
<p>A numeric value controlling the size of the spots in the plot.</p>
</td></tr>
<tr><td><code id="plot_SpaCCI_local_Strength_+3A_specific_lr_pair_name">specific_LR_pair_name</code></td>
<td>
<p>Optional. The name of a specific ligand-receptor pair to plot. If provided, the plot will focus on this interaction. The name should match those in the <code>SpaCCI_local_Result_List$dataframelist</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot object showing the localized interaction strength patterns. The plot will be generated using either the Seurat object or the spatial coordinates data frame, depending on the input provided.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot localized SpaCCI results using Seurat object
library(SpaCCI)
library(dplyr)
data(result_local)
data(result_local_spatial_coords_df)
spatial_coords_df &lt;- result_local_spatial_coords_df
#plot_SpaCCI_local(Seurat_Object = seurat_object,.....)

# Plot localized SpaCCI results using spatial coordinates
plot_SpaCCI_local_Strength(spatial_coordinates_dataframe = spatial_coords_df,
                           SpaCCI_local_Result_List = result_local,
                           Ligand_cell_type = "ductal",
                           Receptor_cell_type = "activated_stellate",
                           spot_plot_size = 3)

</code></pre>

<hr>
<h2 id='plot_Strength_localized'>Plot Localized Hotspot Strength Pattern</h2><span id='topic+plot_Strength_localized'></span>

<h3>Description</h3>

<p>Visualize the inferred cell-cell interaction localized pattern if NOT using Seurat Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_Strength_localized(
  spatial_coord,
  resultdf_list,
  RegionIDs_matrix,
  celltype_ligand,
  celltype_receptor,
  plot_size,
  L_R_pair_name = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_Strength_localized_+3A_spatial_coord">spatial_coord</code></td>
<td>
<p>A data frame of the spatial coordinates. The columns should include <code>"Spot_ID"</code>, <code>"imagerow"</code>, and <code>"imagecol"</code>. And the row names must be the names of <code>"Spot_ID"</code>, which is the same as the rownames in cell type proportion data frame or the colnames of the gene* spot expression data frame</p>
</td></tr>
<tr><td><code id="plot_Strength_localized_+3A_resultdf_list">resultdf_list</code></td>
<td>
<p>A result of data frame list from the output of <code>run_SpaCCI(..., analysis_scale = "local",...)</code> <code>`dataframelist`</code></p>
</td></tr>
<tr><td><code id="plot_Strength_localized_+3A_regionids_matrix">RegionIDs_matrix</code></td>
<td>
<p>A result of matrix list from the output of <code>run_SpaCCI(..., analysis_scale = "local",...)</code> <code>`RegionIDs_matrix`</code></p>
</td></tr>
<tr><td><code id="plot_Strength_localized_+3A_celltype_ligand">celltype_ligand</code></td>
<td>
<p>Ligand cell type string inputted by user, the name of the cell type should match the names in the <code>`spot_cell_proportion_dataframe`</code> during the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_Strength_localized_+3A_celltype_receptor">celltype_receptor</code></td>
<td>
<p>Receptor cell type string inputted by user, the name of the cell type should match the names in the <code>`spot_cell_proportion_dataframe`</code> during the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_Strength_localized_+3A_plot_size">plot_size</code></td>
<td>
<p>As this function incorporate with <code>Seurat</code>'s <code>`SpatialFeaturePlot`</code>, this parameter could control the plotting size of the each spot.</p>
</td></tr>
<tr><td><code id="plot_Strength_localized_+3A_l_r_pair_name">L_R_pair_name</code></td>
<td>
<p>Initially this is set to <code>NULL</code>, if one is interested in a specific Ligand-Receptor pair, then one could specify the L_R_pair_name here. Note: the input name should match the L-R pair name exists in the dataframe in the output of SpaCCI_local &quot;dataframelist&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The localized plot from the inferred cell-cell interaction on the local scale.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Run localized hotspot plot
Result &lt;- run_SpaCCI(..., analysis_scale = "local",...)
local_plot &lt;- plot_Strength_localized(spatial_coord = spatial_coords_df,
                                      resultdf_list = Result$dataframelist,
                                      RegionIDs_matrix = Result$RegionIDs_matrix,
                                      celltype_ligand = "Beta_cells",
                                      celltype_receptor = "T_ells",
                                      plot_size = 3)


</code></pre>

<hr>
<h2 id='plot_Strength_Seurat'>Plot Localized Hotspot Strength Pattern on Seurat Object</h2><span id='topic+plot_Strength_Seurat'></span>

<h3>Description</h3>

<p>Visualize the inferred cell-cell interaction localized pattern on the tissue image with Seurat_object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_Strength_Seurat(
  Seurat_object,
  resultdf_list,
  RegionIDs_matrix,
  celltype_ligand,
  celltype_receptor,
  plot_size,
  L_R_pair_name = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_Strength_Seurat_+3A_seurat_object">Seurat_object</code></td>
<td>
<p>A Seurat object</p>
</td></tr>
<tr><td><code id="plot_Strength_Seurat_+3A_resultdf_list">resultdf_list</code></td>
<td>
<p>A result of data frame list from the output of <code>run_SpaCCI(..., analysis_scale = "local",...)</code> <code>`dataframelist`</code></p>
</td></tr>
<tr><td><code id="plot_Strength_Seurat_+3A_regionids_matrix">RegionIDs_matrix</code></td>
<td>
<p>A result of matrix list from the output of <code>run_SpaCCI(..., analysis_scale = "local",...)</code> <code>`RegionIDs_matrix`</code></p>
</td></tr>
<tr><td><code id="plot_Strength_Seurat_+3A_celltype_ligand">celltype_ligand</code></td>
<td>
<p>Ligand cell type string inputted by user, the name of the cell type should match the names in the <code>`spot_cell_proportion_dataframe`</code> during the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_Strength_Seurat_+3A_celltype_receptor">celltype_receptor</code></td>
<td>
<p>Receptor cell type string inputted by user, the name of the cell type should match the names in the <code>`spot_cell_proportion_dataframe`</code> during the <code>run_SpaCCI</code> analysis.</p>
</td></tr>
<tr><td><code id="plot_Strength_Seurat_+3A_plot_size">plot_size</code></td>
<td>
<p>As this function incorporate with <code>Seurat</code>'s <code>`SpatialFeaturePlot`</code>, this parameter could control the plotting size of the each spot.</p>
</td></tr>
<tr><td><code id="plot_Strength_Seurat_+3A_l_r_pair_name">L_R_pair_name</code></td>
<td>
<p>Initially this is set to <code>NULL</code>, if one is interested in a specific Ligand-Receptor pair, then one could specify the L_R_pair_name here. Note: the input name should match the L-R pair name exists in the dataframe in the output of SpaCCI_local &quot;dataframelist&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The localized plot from the inferred cell-cell interaction on the local scale.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Not Run

# Run localized hotspot plot
Result &lt;- run_SpaCCI(..., analysis_scale = "local",...)
local_plot &lt;- plot_Strength_Seurat(Seurat_object = gene_spot_df,
                                    resultdf_list = Result$dataframelist,
                                    RegionIDs_matrix = Result$RegionIDs_matrix,
                                    celltype_ligand = "Beta_cells",
                                    celltype_receptor = "T_ells",
                                    plot_size = 3)


</code></pre>

<hr>
<h2 id='possible_L_R_pairs_cellchat'>CellChat Database: Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</h2><span id='topic+possible_L_R_pairs_cellchat'></span>

<h3>Description</h3>

<p>This function identifies possible ligand-receptor (L-R) pairs for cell-cell communication analysis using a subset of the CellChat database. It checks for the presence of all genes involved in each L-R pair within the provided gene expression matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possible_L_R_pairs_cellchat(
  species,
  gene_spot_expression_dataframe,
  percentage
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="possible_L_R_pairs_cellchat_+3A_species">species</code></td>
<td>
<p>A string specifying the species (&quot;Human&quot; or &quot;Mouse&quot;). The function selects the appropriate CellChatDB object, typically 'CellChatDB.human' or 'CellChatDB.mouse', which contains information on ligand-receptor interactions.</p>
</td></tr>
<tr><td><code id="possible_L_R_pairs_cellchat_+3A_gene_spot_expression_dataframe">gene_spot_expression_dataframe</code></td>
<td>
<p>A gene expression data frame with genes as row names and Spot IDs as column names. This data frame is used to verify the presence of all genes involved in the L-R pairs.</p>
</td></tr>
<tr><td><code id="possible_L_R_pairs_cellchat_+3A_percentage">percentage</code></td>
<td>
<p>A numeric value specifying the minimum percentage of spots in which a gene must be expressed to be considered. The default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>possible_L_R_pairs</dt><dd><p>A data frame of L-R pairs where all genes are present in the 'gene_spot_expression_dataframe'. The data frame includes the ligand and receptor vectors, and the combined gene vectors.</p>
</dd>
<dt>possible_L_R_pairs_details</dt><dd><p>A data frame with detailed information about the L-R pairs, including the original annotations from the CellChatDB.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>

library(SpaCCI)
#Load the example data
load(system.file("extdata", "Tutorial_example_data.rda", package = "SpaCCI"))
Example_Seurat &lt;- NormalizeData(Example_Seurat)
gene_spot_df &lt;- as.data.frame(Example_Seurat@assays$Spatial@data)
result &lt;- possible_L_R_pairs_cellchat(CellChatDB.human,
      gene_spot_expression_dataframe = gene_spot_df)

</code></pre>

<hr>
<h2 id='possible_L_R_pairs_Cellinker'>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</h2><span id='topic+possible_L_R_pairs_Cellinker'></span>

<h3>Description</h3>

<p>Cellinker Database: This function identifies possible ligand-receptor (L-R) pairs for cell-cell communication analysis using data from the Cellinker database. It checks for the presence of all genes involved in each L-R pair within the provided gene expression matrix, filtering based on a specified expression percentage threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possible_L_R_pairs_Cellinker(
  species,
  gene_spot_expression_dataframe,
  percentage
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="possible_L_R_pairs_Cellinker_+3A_species">species</code></td>
<td>
<p>A string specifying the species (&quot;Human&quot; or &quot;Mouse&quot;). The function selects the appropriate Cellinker interaction file based on this input.</p>
</td></tr>
<tr><td><code id="possible_L_R_pairs_Cellinker_+3A_gene_spot_expression_dataframe">gene_spot_expression_dataframe</code></td>
<td>
<p>A gene expression data frame with genes as row names and Spot IDs as column names. This data frame is used to verify the presence of all genes involved in the L-R pairs.</p>
</td></tr>
<tr><td><code id="possible_L_R_pairs_Cellinker_+3A_percentage">percentage</code></td>
<td>
<p>A numeric value specifying the minimum percentage of spots in which a gene must be expressed to be considered. The default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>possible_L_R_pairs</dt><dd><p>A data frame of L-R pairs where all genes are present in the 'gene_spot_expression_dataframe' and meet the expression threshold. The data frame includes the ligand and receptor vectors, and the combined gene vectors.</p>
</dd>
<dt>possible_L_R_pairs_details</dt><dd><p>A data frame with detailed information about the identified L-R pairs, including their original annotations from the Cellinker dataset.</p>
</dd>
</dl>


<hr>
<h2 id='possible_L_R_pairs_cellphoneDB'>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</h2><span id='topic+possible_L_R_pairs_cellphoneDB'></span>

<h3>Description</h3>

<p>CellPhone Database: This function identifies possible ligand-receptor (L-R) pairs for cell-cell communication analysis using data from a CellPhoneDB dataset. It checks for the presence of all genes involved in each L-R pair within the provided gene expression matrix and filters based on a specified expression percentage threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possible_L_R_pairs_cellphoneDB(gene_spot_expression_dataframe, percentage)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="possible_L_R_pairs_cellphoneDB_+3A_gene_spot_expression_dataframe">gene_spot_expression_dataframe</code></td>
<td>
<p>A gene expression data frame with genes as row names and Spot IDs as column names. This data frame is used to verify the presence of all genes involved in the L-R pairs.</p>
</td></tr>
<tr><td><code id="possible_L_R_pairs_cellphoneDB_+3A_percentage">percentage</code></td>
<td>
<p>A numeric value specifying the minimum percentage of spots in which a gene must be expressed to be considered. The default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>possible_L_R_pairs</dt><dd><p>A data frame of L-R pairs where all genes are present in the 'gene_spot_expression_dataframe' and meet the expression threshold. The data frame includes the ligand and receptor vectors, and the combined gene vectors.</p>
</dd>
<dt>possible_L_R_pairs_details</dt><dd><p>A data frame with detailed information about the identified L-R pairs, including their original annotations from the CellPhoneDB dataset.</p>
</dd>
</dl>


<hr>
<h2 id='possible_L_R_pairs_connectome'>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</h2><span id='topic+possible_L_R_pairs_connectome'></span>

<h3>Description</h3>

<p>ConnectomeDB 2020 Database: This function identifies possible ligand-receptor (L-R) pairs based on gene expression data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possible_L_R_pairs_connectome(gene_spot_expression_dataframe, percentage)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="possible_L_R_pairs_connectome_+3A_gene_spot_expression_dataframe">gene_spot_expression_dataframe</code></td>
<td>
<p>A gene expression data frame with genes as row names and Spot IDs as column names. This data frame is used to verify the presence of all genes involved in the L-R pairs.</p>
</td></tr>
<tr><td><code id="possible_L_R_pairs_connectome_+3A_percentage">percentage</code></td>
<td>
<p>A numeric value specifying the minimum percentage of spots in which a gene must be expressed to be considered. The default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>possible_L_R_pairs</dt><dd><p>A data frame of L-R pairs where all genes are present in the 'gene_spot_expression_dataframe' and meet the expression threshold. The data frame includes the ligand and receptor vectors, and the combined gene vectors.</p>
</dd>
<dt>possible_L_R_pairs_details</dt><dd><p>A data frame with detailed information about the identified L-R pairs, including their original annotations from the ConnectomeDB 2020 dataset.</p>
</dd>
</dl>


<hr>
<h2 id='possible_L_R_pairs_ICELLNET'>Identify Possible Ligand-Receptor Pairs for Cell-Cell Communication</h2><span id='topic+possible_L_R_pairs_ICELLNET'></span>

<h3>Description</h3>

<p>ICELLENT Database: This function identifies possible ligand-receptor (L-R) pairs for cell-cell communication analysis using data from the ICELLNET database. It checks for the presence of all genes involved in each L-R pair within the provided gene expression matrix, filtering based on a specified expression percentage threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possible_L_R_pairs_ICELLNET(gene_spot_expression_dataframe, percentage)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="possible_L_R_pairs_ICELLNET_+3A_gene_spot_expression_dataframe">gene_spot_expression_dataframe</code></td>
<td>
<p>A gene expression data frame with genes as row names and Spot IDs as column names. This data frame is used to verify the presence of all genes involved in the L-R pairs.</p>
</td></tr>
<tr><td><code id="possible_L_R_pairs_ICELLNET_+3A_percentage">percentage</code></td>
<td>
<p>A numeric value specifying the minimum percentage of spots in which a gene must be expressed to be considered. The default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>possible_L_R_pairs</dt><dd><p>A data frame of L-R pairs where all genes are present in the 'gene_spot_expression_dataframe' and meet the expression threshold. The data frame includes the ligand and receptor vectors, and the combined gene vectors.</p>
</dd>
<dt>possible_L_R_pairs_details</dt><dd><p>A data frame with detailed information about the identified L-R pairs, including their original annotations from the ICELLNET dataset.</p>
</dd>
</dl>


<hr>
<h2 id='random_region'>Select Closest Spatial IDs to a Center Point: this is used for permutation</h2><span id='topic+random_region'></span>

<h3>Description</h3>

<p>This function identifies and returns the IDs of the closest spatial points to a specified center point based on Euclidean distance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_region(spatial_coord, center_id, n_ids)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_region_+3A_spatial_coord">spatial_coord</code></td>
<td>
<p>A data frame of the spatial coordinates. The column names should include 'c(&quot;Spot_ID&quot;, &quot;imagerow&quot;, &quot;imagecol&quot;)', and the row names must be the Spot_IDs, which is the same as the row names in the cell type proportion data frame or the column names of the gene*spot expression data frame.</p>
</td></tr>
<tr><td><code id="random_region_+3A_center_id">center_id</code></td>
<td>
<p>A character string specifying the ID of the center spot from which distances are calculated.</p>
</td></tr>
<tr><td><code id="random_region_+3A_n_ids">n_ids</code></td>
<td>
<p>An integer specifying the number of closest IDs to select.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the 'n_ids' closest IDs to the specified center ID.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
spatial_coord &lt;- data.frame(
  imagecol = c(1, 2, 3, 4, 5),
  imagerow = c(5, 4, 3, 2, 1),
  row.names = c("Spot1", "Spot2", "Spot3", "Spot4", "Spot5")
)
center_id &lt;- "Spot3"
closest_ids &lt;- random_region(spatial_coord, center_id, 3)
print(closest_ids)

</code></pre>

<hr>
<h2 id='result_global'>result_global data for SpaCCI</h2><span id='topic+result_global'></span>

<h3>Description</h3>

<p>This example dataset is the result of running the <code>run_SpaCCI</code> function.
It contains the inferred cell-cell interactions across the global scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(result_global)
</code></pre>


<h3>Format</h3>

<p>A list containing:
</p>

<dl>
<dt>pvalue_df</dt><dd><p>A data frame of p-values and adjusted p-values for cell-cell interactions.</p>
</dd>
</dl>



<h3>Details</h3>

<p>These objects can be used for testing and running example analyses with the SpaCCI package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(result_global)
  print(result_global)
</code></pre>

<hr>
<h2 id='result_local'>result_local data and spatial coordinates for SpaCCI
Local Regional Data</h2><span id='topic+result_local'></span>

<h3>Description</h3>

<p>This dataset contains the results of spatial neighborhood analyses.
These objects can be used for testing and running example analyses with the SpaCCI package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(result_local)
</code></pre>


<h3>Format</h3>

<p>A list containing:
</p>

<dl>
<dt>dataframelist</dt><dd><p>A list of data frames, each containing p-value results for spatial neighborhoods.</p>
</dd>
<dt>RegionIDs_matrix</dt><dd><p>A list of matrices, each containing the spot IDs for specific neighborhoods.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>  data(result_local)
  print(result_local)
</code></pre>

<hr>
<h2 id='result_local_spatial_coords_df'>Local Spatial Coordinates Data Frame</h2><span id='topic+result_local_spatial_coords_df'></span>

<h3>Description</h3>

<p>A dataset containing spatial coordinates for local analysis, typically used in spatial transcriptomics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(result_local)
</code></pre>


<h3>Format</h3>

<p>A data frame with X rows and 3 columns:
</p>

<dl>
<dt>imagecol</dt><dd><p>The x-coordinate of the spot in the image.</p>
</dd>
<dt>imagerow</dt><dd><p>The y-coordinate of the spot in the image.</p>
</dd>
<dt>Spot_ID</dt><dd><p>The unique identifier for each spatial spot.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(result_local_spatial_coords_df)
head(result_local_spatial_coords_df)
</code></pre>

<hr>
<h2 id='result_regional'>result_regional data for SpaCCI</h2><span id='topic+result_regional'></span>

<h3>Description</h3>

<p>This example dataset is the result of running the <code>run_SpaCCI</code> function.
It contains the inferred cell-cell interactions across the regional scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(result_regional)
</code></pre>


<h3>Format</h3>

<p>A list containing:
</p>

<dl>
<dt>pvalue_df</dt><dd><p>A data frame of p-values and adjusted p-values for cell-cell interactions.</p>
</dd>
</dl>



<h3>Details</h3>

<p>These objects can be used for testing and running example analyses with the SpaCCI package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(result_regional)
  print(result_regional)
</code></pre>

<hr>
<h2 id='run_SpaCCI'>Run SpaCCI Analysis</h2><span id='topic+run_SpaCCI'></span>

<h3>Description</h3>

<p>This function runs the SpaCCI analysis to infer cell-cell interactions based on ligand-receptor pairs at global, regional, or local spatial scales. It integrates gene expression data, cell type proportions, and spatial coordinates with a user-specified ligand-receptor database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_SpaCCI(
  gene_spot_expression_dataframe,
  spot_cell_proportion_dataframe,
  spatial_coordinates_dataframe,
  LR_database_list,
  specific_LR_pair = NULL,
  analysis_scale,
  region_spot_IDs = NULL,
  local_scale_proportion = 1,
  neighborhood_radius = 2.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_SpaCCI_+3A_gene_spot_expression_dataframe">gene_spot_expression_dataframe</code></td>
<td>
<p>A data frame of gene expression values, where row names are genes and column names are spot IDs.</p>
</td></tr>
<tr><td><code id="run_SpaCCI_+3A_spot_cell_proportion_dataframe">spot_cell_proportion_dataframe</code></td>
<td>
<p>A data frame of cell type proportions, where row names are spot IDs and column names are cell types.</p>
</td></tr>
<tr><td><code id="run_SpaCCI_+3A_spatial_coordinates_dataframe">spatial_coordinates_dataframe</code></td>
<td>
<p>A data frame containing the spatial coordinates of the spots. The columns should include <code>"Spot_ID"</code>, <code>"imagerow"</code>, and <code>"imagecol"</code>. And the row names must be the names of <code>"Spot_ID"</code>.</p>
</td></tr>
<tr><td><code id="run_SpaCCI_+3A_lr_database_list">LR_database_list</code></td>
<td>
<p>A list containing ligand-receptor pairs and additional information, generated by functions using <code>`LR_database()`</code>.</p>
</td></tr>
<tr><td><code id="run_SpaCCI_+3A_specific_lr_pair">specific_LR_pair</code></td>
<td>
<p>Required if <code>analysis_scale</code> is <code>"local"</code>. A vector of ligand-receptor pair names for localized analysis. The names should match those in row names in <code>`LR_database_list$possible_LR_pairs_info`</code>.</p>
</td></tr>
<tr><td><code id="run_SpaCCI_+3A_analysis_scale">analysis_scale</code></td>
<td>
<p>A string specifying the scale of analysis: <code>"global"</code>, <code>"regional"</code>, or <code>"local"</code>.</p>
</td></tr>
<tr><td><code id="run_SpaCCI_+3A_region_spot_ids">region_spot_IDs</code></td>
<td>
<p>Required if <code>analysis_scale</code> is &quot;regional&quot;. A vector of spot IDs defining the region for regional analysis.</p>
</td></tr>
<tr><td><code id="run_SpaCCI_+3A_local_scale_proportion">local_scale_proportion</code></td>
<td>
<p>Optional. A numeric value ranging from 0 to 1, (0,1] specifying the proportion of spots to use for localized analysis. Default is <code>1</code>, meaning using <code>100</code>% proportion of spots. One could modified if want to reducing computing time.</p>
</td></tr>
<tr><td><code id="run_SpaCCI_+3A_neighborhood_radius">neighborhood_radius</code></td>
<td>
<p>Optional. A numeric value specifying the radius of the neighborhood for localized analysis. Default is <code>2.5</code>, according to the 10X Visium ST data accounting for 200-250 <code class="reqn">\mu</code>m interacting distance.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function supports three scales of analysis:
</p>

<dl>
<dt><code>global</code></dt><dd><p>Analyzes interactions across the entire dataset.</p>
</dd>
<dt><code>regional</code></dt><dd><p>Analyzes interactions within a specified region of spots. Requires <code>region_spot_IDs</code>.</p>
</dd>
<dt><code>local</code></dt><dd><p>Analyzes localized hotspot of interactions for specific ligand-receptor pairs on the entire slides. Requires <code>specific_LR_pair</code>.</p>
</dd>
</dl>



<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>If <code>analysis_scale</code> is <code>"local"</code>:</dt><dd>
<p>A list containing:
</p>

<dl>
<dt><code>dataframelist</code></dt><dd><p>A list of data frames, each representing the inferred interactions for a specific center spot. Each data frame includes information on ligand and receptor cell types, P-values, and adjusted P-values.</p>
</dd>
<dt><code>RegionIDs_matrix</code></dt><dd><p>A list of matrices, each containing the IDs of the spots within the specified radius of each center spot.</p>
</dd>
</dl>

</dd>
<dt>If <code>analysis_scale</code> is <code>"regional"</code> or <code>"global"</code>:</dt><dd>
<p>A list containing:
</p>

<dl>
<dt><code>pvalue_df</code></dt><dd><p>A data frame of inferred interactions within the specified region or globally, including information on ligand and receptor cell types, P-values, and adjusted P-values.</p>
</dd>
</dl>

</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>library(SpaCCI)
library(nnls)
library(FNN)
library(Matrix)
#Load the example data
data(test_data)
gene_spot_df &lt;- test_data$gene_spot_df
cell_prop_df &lt;- test_data$cell_prop_df
spatial_coords_df &lt;- test_data$spatial_coords_df

result &lt;- LR_database(species = "Human",
                      database_name = "CellChat",
                      gene_spot_expression_dataframe = gene_spot_df)
# global
result_global &lt;- run_SpaCCI(gene_spot_expression_dataframe = gene_spot_df,
                            spot_cell_proportion_dataframe = cell_prop_df,
                            spatial_coordinates_dataframe = spatial_coords_df,
                            LR_database_list = result,
                            analysis_scale = "global")

# local
result_local &lt;- run_SpaCCI(gene_spot_expression_dataframe = gene_spot_df,
                          spot_cell_proportion_dataframe = cell_prop_df,
                          spatial_coordinates_dataframe = spatial_coords_df,
                          LR_database_list = result,
                          specific_LR_pair = "EDN2_EDNRA",
                          analysis_scale = "local",
                          local_scale_proportion = 0.1,
                          neighborhood_radius = 2.5)

</code></pre>

<hr>
<h2 id='scPalette'>Generate a Color Palette</h2><span id='topic+scPalette'></span>

<h3>Description</h3>

<p>This function generates a color palette. It selects colors from a predefined color space, and if more colors are needed than are available in the predefined set, it generates a palette using color interpolation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scPalette(n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scPalette_+3A_n">n</code></td>
<td>
<p>An integer specifying the number of colors needed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of colors in hexadecimal format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate a palette with 5 colors
palette &lt;- scPalette(5)
print(palette)

# Generate a palette with 30 colors
large_palette &lt;- scPalette(30)
print(large_palette)

</code></pre>

<hr>
<h2 id='SpaCCI_global'>Infer Cell-Cell Interactions on a Global Scale</h2><span id='topic+SpaCCI_global'></span>

<h3>Description</h3>

<p>This function infers cell-cell interactions on a global scale using spatial transcriptomics data. It applies permutation testing to identify significant ligand-receptor interactions across all spots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SpaCCI_global(
  gene_spot_df,
  spot_cell_prop_df,
  spatial_coord,
  matching_L_R_pairs,
  matching_L_R_pairs_info
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SpaCCI_global_+3A_gene_spot_df">gene_spot_df</code></td>
<td>
<p>A data frame where the rows are genes and the columns are spots (Spot_IDs), representing gene expression levels across spatial spots.</p>
</td></tr>
<tr><td><code id="SpaCCI_global_+3A_spot_cell_prop_df">spot_cell_prop_df</code></td>
<td>
<p>A data frame of cell type proportions for each spot. The rows represent spots (Spot_IDs), and the columns represent different cell types.</p>
</td></tr>
<tr><td><code id="SpaCCI_global_+3A_spatial_coord">spatial_coord</code></td>
<td>
<p>A data frame of the spatial coordinates. The column names should include 'c(&quot;Spot_ID&quot;, &quot;imagerow&quot;, &quot;imagecol&quot;)', and the row names must be the Spot_IDs, which is the same as the row names in the cell type proportion data frame or the column names of the gene*spot expression data frame.</p>
</td></tr>
<tr><td><code id="SpaCCI_global_+3A_matching_l_r_pairs">matching_L_R_pairs</code></td>
<td>
<p>A data frame containing matching ligand-receptor pairs. Each row corresponds to a ligand-receptor pair, with columns for <code>ligand_vector</code> and <code>receptor_vector</code>.</p>
</td></tr>
<tr><td><code id="SpaCCI_global_+3A_matching_l_r_pairs_info">matching_L_R_pairs_info</code></td>
<td>
<p>A data frame providing additional information for each ligand-receptor pair, such as pathway information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>pvalue_df</dt><dd><p>A data frame of inferred interactions across the global scale, including information on ligand and receptor cell types, interaction strength, P-values, and adjusted P-values.</p>
</dd>
</dl>


<hr>
<h2 id='SpaCCI_local'>Infer Cell-Cell Interactions on a Local Scale</h2><span id='topic+SpaCCI_local'></span>

<h3>Description</h3>

<p>This function infers cell-cell interactions on a local scale using spatial transcriptomics data. It utilizes permutation testing to identify significant ligand-receptor interactions within specified neighborhoods around randomly selected center spots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SpaCCI_local(
  gene_spot_df,
  spot_cell_prop_df,
  spatial_coord,
  prop,
  radius,
  matching_L_R_pairs,
  matching_L_R_pairs_info
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SpaCCI_local_+3A_gene_spot_df">gene_spot_df</code></td>
<td>
<p>A data frame where the rows are genes and the columns are spots (Spot_IDs), representing gene expression levels across spatial spots.</p>
</td></tr>
<tr><td><code id="SpaCCI_local_+3A_spot_cell_prop_df">spot_cell_prop_df</code></td>
<td>
<p>A data frame of cell type proportions for each spot. The rows represent spots (Spot_IDs), and the columns represent different cell types.</p>
</td></tr>
<tr><td><code id="SpaCCI_local_+3A_spatial_coord">spatial_coord</code></td>
<td>
<p>A data frame of the spatial coordinates. The column names should include 'c(&quot;Spot_ID&quot;, &quot;imagerow&quot;, &quot;imagecol&quot;)', and the row names must be the Spot_IDs, which is the same as the row names in the cell type proportion data frame or the column names of the gene*spot expression data frame.</p>
</td></tr>
<tr><td><code id="SpaCCI_local_+3A_prop">prop</code></td>
<td>
<p>A numeric value representing the proportion of spots to randomly sample as center spots for local neighborhood analysis.</p>
</td></tr>
<tr><td><code id="SpaCCI_local_+3A_radius">radius</code></td>
<td>
<p>A numeric value specifying the radius of the spatial neighborhood around each center spot.</p>
</td></tr>
<tr><td><code id="SpaCCI_local_+3A_matching_l_r_pairs">matching_L_R_pairs</code></td>
<td>
<p>A data frame containing matching ligand-receptor pairs. Each row corresponds to a ligand-receptor pair, with columns for <code>ligand_vector</code> and <code>receptor_vector</code>.</p>
</td></tr>
<tr><td><code id="SpaCCI_local_+3A_matching_l_r_pairs_info">matching_L_R_pairs_info</code></td>
<td>
<p>A data frame providing additional information for each ligand-receptor pair, such as pathway information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>dataframelist</dt><dd><p>A list of data frames, each representing the inferred interactions for a specific center spot. Each data frame includes information on ligand and receptor cell types, P-values, and adjusted P-values.</p>
</dd>
<dt>RegionIDs_matrix</dt><dd><p>A list of matrices, each containing the IDs of the spots within the specified radius of each center spot.</p>
</dd>
</dl>


<hr>
<h2 id='SpaCCI_region'>Infer Cell-Cell Interactions in a Specified Region</h2><span id='topic+SpaCCI_region'></span>

<h3>Description</h3>

<p>This function infers cell-cell interactions within a specified region using spatial transcriptomics data. It applies permutation testing to identify significant ligand-receptor interactions in the region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SpaCCI_region(
  gene_spot_df,
  spot_cell_prop_df,
  spatial_coord,
  region_spot_IDs,
  matching_L_R_pairs,
  matching_L_R_pairs_info
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SpaCCI_region_+3A_gene_spot_df">gene_spot_df</code></td>
<td>
<p>A data frame where the rows are genes and the columns are spots (Spot_IDs), representing gene expression levels across spatial spots.</p>
</td></tr>
<tr><td><code id="SpaCCI_region_+3A_spot_cell_prop_df">spot_cell_prop_df</code></td>
<td>
<p>A data frame of cell type proportions for each spot. The rows represent spots (Spot_IDs), and the columns represent different cell types.</p>
</td></tr>
<tr><td><code id="SpaCCI_region_+3A_spatial_coord">spatial_coord</code></td>
<td>
<p>A data frame of the spatial coordinates. The column names should include 'c(&quot;Spot_ID&quot;, &quot;imagerow&quot;, &quot;imagecol&quot;)', and the row names must be the Spot_IDs, which is the same as the row names in the cell type proportion data frame or the column names of the gene*spot expression data frame.</p>
</td></tr>
<tr><td><code id="SpaCCI_region_+3A_region_spot_ids">region_spot_IDs</code></td>
<td>
<p>A vector of Spot_IDs representing the spots included in the region of interest.</p>
</td></tr>
<tr><td><code id="SpaCCI_region_+3A_matching_l_r_pairs">matching_L_R_pairs</code></td>
<td>
<p>A data frame containing matching ligand-receptor pairs. Each row corresponds to a ligand-receptor pair, with columns for <code>ligand_vector</code> and <code>receptor_vector</code>.</p>
</td></tr>
<tr><td><code id="SpaCCI_region_+3A_matching_l_r_pairs_info">matching_L_R_pairs_info</code></td>
<td>
<p>A data frame providing additional information for each ligand-receptor pair, such as pathway information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>pvalue_df</dt><dd><p>A data frame of inferred interactions within the specified region, including information on ligand and receptor cell types, P-values, and adjusted P-values.</p>
</dd>
</dl>


<hr>
<h2 id='test_data'>Test data for SpaCCI</h2><span id='topic+test_data'></span>

<h3>Description</h3>

<p>This dataset includes:
</p>

<ul>
<li><p>gene_spt_df: A data frame of spot-level gene expression.
</p>
</li>
<li><p>cell_prop_df: A data frame with cell type proportions.
</p>
</li>
<li><p>spatial_coords_df: A data frame of spatial coordinates.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(test_data)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(SpaCCI)
data(test_data)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
