<!DOCTYPE html><html lang="en"><head><title>Help for package SpadeR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SpadeR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SpadeR-package'><p>Species-richness prediction and diversity estimation with R</p></a></li>
<li><a href='#ChaoShared'><p>Estimation of the number of shared species between two communities/assemblages</p></a></li>
<li><a href='#ChaoSharedData'>
<p>Data for Function ChaoShared</p></a></li>
<li><a href='#ChaoSpecies'><p>Estimation of species richness in a community</p></a></li>
<li><a href='#ChaoSpeciesData'>
<p>Data for Function ChaoSpecies</p></a></li>
<li><a href='#Diversity'><p>Estimation of species diversity (Hill numbers)</p></a></li>
<li><a href='#DiversityData'>
<p>Data for Function Diversity</p></a></li>
<li><a href='#Genetics'><p>Estimation of genetic differentiation measures</p></a></li>
<li><a href='#GeneticsDataAbu'>
<p>Human allele frequency data for  Function Genetics</p></a></li>
<li><a href='#SimilarityMult'><p>Estimation of multiple-community similarity measures</p></a></li>
<li><a href='#SimilarityMultData'>
<p>Data for Function SimilarityMult</p></a></li>
<li><a href='#SimilarityPair'><p>Estimation of two-assemblage similarity measures</p></a></li>
<li><a href='#SimilarityPairData'>
<p>Data for Function SimilarityPair</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Species-Richness Prediction and Diversity Estimation with R</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-09-06</td>
</tr>
<tr>
<td>Author:</td>
<td>Anne Chao, K. H. Ma, T. C. Hsieh and Chun-Huo Chiu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anne Chao &lt;chao@stat.nthu.edu.tw&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimation of various biodiversity indices and related (dis)similarity measures based on individual-based (abundance) data or sampling-unit-based (incidence) data taken from one or multiple communities/assemblages.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.14)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-09-06 05:40:30 UTC; USER</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-09-06 14:53:57</td>
</tr>
</table>
<hr>
<h2 id='SpadeR-package'>Species-richness prediction and diversity estimation with R
</h2><span id='topic+SpadeR-package'></span><span id='topic+SpadeR'></span>

<h3>Description</h3>

<p>Provides simple functions to compute various biodiversity indices and related (dis)similarity measures based on individual-based (abundance) data or sampling-unit-based (incidence) data taken from one or multiple communities/assemblages. <br /><br />
</p>
<p>This package contains six main functions: <br /><br />
1. <code>ChaoSpecies</code> (estimating species richness for one community). <br /><br />
2. <code>Diversity</code> (estimating a continuous diversity profile and various diversity indices in one community including species richness, Shannon
diversity and Simpson diversity). This function also features plots of empirical and estimated continuous diversity profiles. <br /><br />
3. <code>ChaoShared</code> (estimating the number of shared species between two communities). <br /><br />
4. <code>SimilartyPair</code> (estimating various similarity indices between two assemblages). Both richness- and abundance-based two-community similarity indices are included. <br /><br />
5. <code>SimilarityMult</code> (estimating various similarity indices among <code class="reqn">N</code> communities). Both richness- and abundance-based <code class="reqn">N</code>-community similarity indices are included. <br /><br />
6. <code>Genetics</code> (estimating allelic dissimilarity/differentiation among sub-populations based on multiple-subpopulation genetics data). <br /><br />
Except for the <code>Genetics</code> function, there are at least three types of data are supported for each function. <br /><br />
</p>


<h3>Details</h3>

<p>Data are generally classified as abundance data and incidence data and there are five types of data input formats options (datatype=&quot;abundance&quot;, &quot;abundance_freq_count&quot;, &quot;incidence_freq&quot;, &quot;incidence_freq_count&quot;, &quot;incidence_raw&quot;). <br /><br />
</p>

<dl>
<dt>A.</dt><dd><p>Individual-based abundance data when a sample of individuals is taken from each community.</p>
</dd>
</dl>
<p><b>Type (1) abundance data</b> (datatype = &quot;abundance&quot;): Input data consist of species (in rows) by community (in columns) matrix. The entries of each row are the observed abundances of a species in <code class="reqn">N</code> communities. <br /><br />
<b>Type (1A) abundance-frequency counts data</b> only for a single community (datatype = &quot;abundance_freq_count&quot;): input data are arranged as (1 <code class="reqn">f_1 \ 2 \ f_2 \ ... \ r \ f_r</code>)(each number needs to be separated by at least one blank space or separated by rows), where <code class="reqn">r</code> denotes the maximum frequency and <code class="reqn">f_k</code> denotes the number of species represented by exactly <code class="reqn">k</code> individuals/times in the sample. Here the data (<code class="reqn">f_1, f_2, ..., f_r</code>) are referred to as &quot;abundance-frequency counts&quot;.
</p>
<dl>
<dt>B.</dt><dd><p>Sampling-unit-based incidence data when a number of sampling units are randomly taken from each community. Only the incidence (detection/non-detection) of species is recorded in each sampling unit. There are three data formats options.</p>
</dd> </dl>
<p><br /><br />
<b>Type (2) incidence-frequency data</b> (datatype=&quot;incidence_freq&quot;): The first row of the input data must be the number of sampling units in each community. Beginning with the second row, input data consist of species (in rows) by community (in columns) matrix. The entries of each row are the observed incidence frequencies (the number of detections or the number of sampling units in which a species are detected) of a species in <code class="reqn">N</code> communities. <br /><br />
<b>Type (2A) incidence-frequency counts data</b> only for a single community (datatype=&quot;incidence <br /> _freq_count&quot;): input data are arranged as  (<code class="reqn">T \ 1 \ Q_1 \ 2 \ Q_2 \ ... \ r \ Q_r</code>) (each number needs to be separated by at least one blank space or separated by rows), where <code class="reqn">Q_k</code> denotes the number of species that were detected in exactly <code class="reqn">k</code> sampling units, while <code class="reqn">r</code> denotes the number of sampling units in which the most frequent species were found. The first entry must be the total number of sampling units, <code class="reqn">T</code>. The data (<code class="reqn">Q_1, Q_2, ..., Q_r</code>) are referred to as &quot;incidence frequency counts&quot;. <br /><br />
<b>Type (2B) incidence-raw data</b> (datatype=&quot;incidence_raw&quot;): Data consist of a species-by-sampling-unit incidence (detection/non-detection) matrix; typically &quot;1&quot; means a detection and &quot;0&quot; means a non-detection. Each row refers to the detection/non-detection record of a species in <code class="reqn">T</code> sampling units. Users must specify the number of sampling units in the function argument &quot;units&quot;. The first <code class="reqn">T_1</code> columns of the input matrix denote species detection/non-detection data based on the <code class="reqn">T_1</code> sampling units from Community 1, and the next <code class="reqn">T_2</code> columns denote the detection/non-detection data based on the <code class="reqn">T_2</code> sampling units from Community 2, and so on, and the last <code class="reqn">T_N</code> columns denote the detection/non-detection data based on <code class="reqn">T_N</code> sampling units from Community <code class="reqn">N</code>, <code class="reqn">T_1 + T_2 + ... + T_N = T</code>. <br />

</p>
<p>An Online version of SpadeR is also available for users without an R background: <br />
http://chao.stat.nthu.edu.tw/wordpress/software_download/softwarespader_online/. <br />
In the detailed Online SpadeR User's Guide, we illustrate all the running procedures in an easily
accessible way through numerical examples with proper interpretations of portions of the output.
All the data of those illustrative examples are included in this package. <br /><br />
</p>
<p>functions: ChaoSpecies, Diversity, ChaoShared, SimilarityPair, SimilarityMult, Genetics
</p>


<h3>Author(s)</h3>

<p>Anne Chao, K. H. Ma, T. C. Hsieh and Chun-Huo Chiu 
</p>
<p>Maintainer: Anne Chao &lt;chao@stat.nthu.edu.tw&gt;
</p>

<hr>
<h2 id='ChaoShared'>Estimation of the number of shared species between two communities/assemblages</h2><span id='topic+ChaoShared'></span>

<h3>Description</h3>

<p><code>ChaoShared</code>: Estimation of shared species richness between two communities/assemblages based on
three types of data: Type (1) abundance data (datatype=&quot;abundance&quot;), Type (2) incidence-frequency
data (datatype=&quot;incidence_freq&quot;), and Type (2B) incidence-raw data (datatype=&quot;incidence<br /> 
_raw&quot;); see <code>SpadeR-package</code> details for data input formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChaoShared(data, datatype = c("abundance", "incidence_freq", "incidence_raw"),
  units, se = TRUE, nboot = 200, conf = 0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ChaoShared_+3A_data">data</code></td>
<td>
<p>a matrix/data.frame of species abundances/incidences.<br /></p>
</td></tr>
<tr><td><code id="ChaoShared_+3A_datatype">datatype</code></td>
<td>
<p>type of input data, &quot;abundance&quot;, &quot;incidence_freq&quot; or &quot;incidence_raw&quot;. <br /></p>
</td></tr>
<tr><td><code id="ChaoShared_+3A_units">units</code></td>
<td>
<p>number of sampling units in each community. For <code>datatype = "incidence_raw"</code>, users must specify the number of sampling units taken from each community. This argument is not needed for &quot;abundance&quot; and &quot;incidence_freq&quot; data.<br /></p>
</td></tr>
<tr><td><code id="ChaoShared_+3A_se">se</code></td>
<td>
<p>a logical variable to calculate the bootstrap standard error and the associated confidence interval. <br /></p>
</td></tr>
<tr><td><code id="ChaoShared_+3A_nboot">nboot</code></td>
<td>
<p>an integer specifying the number of bootstrap replications. <br /></p>
</td></tr>
<tr><td><code id="ChaoShared_+3A_conf">conf</code></td>
<td>
<p>a positive number <code class="reqn">\le</code> 1 specifying the level of confidence interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of two objects: <br /><br />
<code>$Basic_data_information</code> for summarizing data information. <br /><br />
<code>$Estimation_results</code> for showing a table of various shared richess estimates, standard errors, and the associated confidence intervals. <br /><br />
</p>


<h3>References</h3>

<p>Chao, A., Hwang, W.-H., Chen, Y.-C. and Kuo. C.-Y. (2000). Estimating the  number of shared species in two communities. Statistica Sinica, 10, 227-246.<br /><br />
Pan, H.-Y., Chao, A. and Foissner, W. (2009). A non-parametric lower bound for the number of species shared by multiple communities. Journal of Agricultural, Biological and Environmental Statistics, 14, 452-468.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ChaoSharedData)
# Type (1) abundance data
ChaoShared(ChaoSharedData$Abu,"abundance",se=TRUE,nboot=200,conf=0.95)
# Type (2) incidence-frequency data 
ChaoShared(ChaoSharedData$Inci,"incidence_freq",se=TRUE,nboot=200,conf=0.95)
# Type (2B) incidence-raw data   
ChaoShared(ChaoSharedData$Inci_raw,"incidence_raw",units=c(16,17),se=TRUE,nboot=200,conf=0.95)
</code></pre>

<hr>
<h2 id='ChaoSharedData'>
Data for Function ChaoShared
</h2><span id='topic+ChaoSharedData'></span>

<h3>Description</h3>

<p>There are three data sets: <br /><br />
1. Type (1) abundance data (<code>ChaoSharedData$Abu</code>) <br /><br />
The data consist of the observed bird abundances/frequencies collected from two estuaries (Chao et al. 2000). For each species (row), the entry of the first column is the observed species frequency from Estuary I, and the second column is the observed species frequency from Estuary II. The species checklist includes 201 species, so the entry data includes a matrix of 201 rows and 2 columns. <br /><br />
</p>
<p>2. Type (2) incidence-frequency data (<code>ChaoSharedData$Inci</code>) <br /><br />
The data consist of bird incidence (detection/non-detection) frequencies observed in 2015 (by 16 teams) and 2016 (by 17 teams) in the Hong Kong Bird Race. Each team is regarded as a sampling unit. Unlike the abundance data, the numbers of sampling units (16 and 17 for these data) are specified in the first row. Beginning with the second row, the entry of the first column is the observed incidence frequency (the total number of detections among all teams) of a given species in 2015, and the entry of the second column is the observed incidence frequency of the same species in 2016. A 280-species checklist was used, thus the input data consist of 281 rows (the first entry records the number of sampling units) and 2 columns. <br /><br />
3. Type (2B) incidence-raw data (<code>ChaoSharedData$Inci_raw</code>) <br /><br />
The data consist of raw detection/non-detection records of bird species in 2015 (by 16 teams) and 2016 (by 17 teams) in the Hong Kong Bird Race. A 280-species checklist was used. The raw data consist of a 280 x 33 (species-by-sampling-unit) matrix with element 1's (detection) or 0's (non-detection). Each row of the matrix refers to the detection/non-detection records of the same species so that the information about shared species can be computed. The first 16 columns of the matrix denote the species detection/non-detection data by 16 teams in 2015, and the next 17 columns denote the species detection/non-detection data by 17 teams in 2016.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ChaoSharedData)</code></pre>


<h3>References</h3>

<p>Chao, A., Hwang, W.-H., Chen, Y.-C. and Kuo. C.-Y. (2000). Estimating the number of shared species in two communities. Statistica Sinica, 10, 227-246. <br /><br />
World Wildlife Fund (WWF) for Nature, Hong Kong. Bird Bird Race. <br />
http://www.wwf.org.hk/en/getinvolved/hkbbr/. Assessed on July 26, 2016
</p>

<hr>
<h2 id='ChaoSpecies'>Estimation of species richness in a community</h2><span id='topic+ChaoSpecies'></span>

<h3>Description</h3>

<p><code>ChaoSpecies</code>: Estimation of species richness in a single community based on five types of data:
Type (1) abundance data (datatype=&quot;abundance&quot;), Type (1A) abundance-frequency counts <br />
(datatype=&quot;abundance_freq_count&quot;), Type (2) incidence-frequency data (datatype =
&quot;incidence_freq&quot;), Type (2A) incidence-frequency counts (datatype=&quot;incidence_freq_count&quot;), and
Type (2B) incidence-raw data (datatype=&quot;incidence_raw&quot;); see <code>SpadeR-package</code> details for data input formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ChaoSpecies(data, datatype = c("abundance", "abundance_freq_count",
  "incidence_freq", "incidence_freq_count", "incidence_raw"), k = 10,
  conf = 0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ChaoSpecies_+3A_data">data</code></td>
<td>
<p>a matrix/data.frame of species abundances/incidences.<br /></p>
</td></tr>
<tr><td><code id="ChaoSpecies_+3A_datatype">datatype</code></td>
<td>
<p>type of input data, &quot;abundance&quot;, &quot;abundance_freq_count&quot;, &quot;incidence_freq&quot;, &quot;incidence_freq_count&quot; or &quot;incidence_raw&quot;. <br /></p>
</td></tr>
<tr><td><code id="ChaoSpecies_+3A_k">k</code></td>
<td>
<p>the cut-off point (default = 10), which separates species into &quot;abundant&quot; and &quot;rare&quot; groups for abundance data for the estimator ACE; it separates species into &quot;frequent&quot; and
&quot;infrequent&quot; groups for incidence data for the estimator ICE.</p>
</td></tr>
<tr><td><code id="ChaoSpecies_+3A_conf">conf</code></td>
<td>
<p>a positive number <code class="reqn">\le</code> 1 specifying the level of confidence interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of three objects: <br /><br />
<code>$Basic_data_information</code> and <code>$Rare_species_group</code>/<code>$Infreq_species_group</code> for summarizing data information. <br /><br />
<code>$Species_table</code> for showing a table of various species richness estimates, standard errors, and the associated confidence intervals. <br /><br />
</p>


<h3>References</h3>

<p>Chao, A., and Chiu, C. H. (2012). Estimation of species richness and shared species richness. In N. Balakrishnan (ed). Methods and Applications of Statistics in the Atmospheric and Earth Sciences. p.76-111, Wiley, New York.<br /><br />
Chao, A., and Chiu, C. H. (2016). Nonparametric estimation and comparison of species richness. Wiley Online Reference in the Life Science. In: eLS. John Wiley and Sons, Ltd: Chichester. DOI: 10.1002/9780470015902.a0026329.<br /><br />
Chao, A., and Chiu, C. H. (2016). Species richness: estimation and comparison. Wiley StatsRef: Statistics Reference Online. 1-26.<br /><br />
Chiu, C. H., Wang Y. T., Walther B. A. and Chao A. (2014). An improved non-parametric lower bound of species richness via the Good-Turing frequency formulas. Biometrics, 70, 671-682. <br /><br />
Gotelli, N. G. and Chao, A. (2013). Measuring and estimating species richness, species diver- sity, and biotic similarity from sampling data. Encyclopedia of Biodiversity, 2nd Edition, Vol. 5, 195-211, Waltham, MA. <br /><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ChaoSpeciesData)
# Type (1) abundance data
ChaoSpecies(ChaoSpeciesData$Abu,"abundance",k=10,conf=0.95)
# Type (1A) abundance-frequency counts data
ChaoSpecies(ChaoSpeciesData$Abu_count,"abundance_freq_count",k=10,conf=0.95)
# Type (2) incidence-frequency data
ChaoSpecies(ChaoSpeciesData$Inci,"incidence_freq",k=10,conf=0.95)
# Type (2A) incidence-frequency counts data
ChaoSpecies(ChaoSpeciesData$Inci_count,"incidence_freq_count",k=10,conf=0.95)
# Type (2B) incidence-raw data 
ChaoSpecies(ChaoSpeciesData$Inci_raw,"incidence_raw",k=10,conf=0.95)
</code></pre>

<hr>
<h2 id='ChaoSpeciesData'>
Data for Function ChaoSpecies
</h2><span id='topic+ChaoSpeciesData'></span>

<h3>Description</h3>

<p>There are five data sets: <br /><br />
1. Type (1) abundance data (<code>ChaoSpeciesData$Abu</code>) <br /><br />
The data consist of 25 birds abundances/frequencies in a sample (Magurran, 1988, p.152). Their observed frequencies are respectively 752, 276, 194, 126, 121, 97, 95, 83, 72, 44, 39, 16, 15, 13, 9, 9, 9, 8, 7, 4, 2, 2, 1, 1, 1. <br /><br />
2. Type (1A) abundance-frequency counts data (<code>ChaoSpeciesData$Abu_count</code>) <br /><br />
The data consist of the observed species abundance distribution of endangered and rare vascular plant species in the central portion of the southern Appalachian region (Miller and Wiegert, 1989). A total of 188 species were recorded out of 1008 individuals compiled over a span of 150 years of field observations. The data are read as: (1 61 2 35 3 18 4 12 ... 67 1); each number needs to be separated by at least one blank space or by separated by rows. Here the first pair (1, 61) indicates that there are 61 singletons, the second pair (2, 35) indicates there are 35 doubletons, and so on, with the last pair (67, 1) indicating that there is one species that is represented by 67 individuals. <br /><br />
3. Type (2) incidence-frequency data (<code>ChaoSpeciesData$Inci</code>) <br /><br />
The data include seed-bank records taken from Butler and Chazdon (1998). There were 121 soil samples (each soil sample is regarded as a sampling unit) and species of seedlings that germinated from each soil sample were recorded. A total of 34 species of seedlings were found in the 121 soil samples. In the input data, the entry in the first row denotes the number of sampling units. Then, beginning with the second row, each row records the species incidence frequency (i.e., the number of soil samples in which the seedlings were found) of a given species in all 121 soil samples. The ordering of data entries does not affect the analysis. <br /><br />
4. Type (2A) incidence-frequency counts data (<code>ChaoSpeciesData$Inci_freq_count</code>) <br /><br />
The data consist of cottontail capture-recapture data provided in Edwards and Eberhardt (1967) to illustrate that species richness estimation can be applied to estimate the size of a population. An &quot;individual&quot; animal in capture-recapture studies corresponds to a &quot;species&quot; in the richness estimation. A total of 142 captures were recorded for 76 distinct rabbits in 18 trapping nights. For these data, the incidence frequency counts (<code class="reqn">Q_1</code> to <code class="reqn">Q_7</code>) were 43, 16, 8, 6, 0, 2, 1. The input data are read as follows: 
(18 1 43 2 16 3 8 4 6 6 2 7 1); each number needs to be separated by at least one blank space or separated by rows. Here the pair (1, 43) indicates that there are 43 unique species, the next pair (2, 16) indicates there are 16 duplicate species, and so on. <br /><br />
5. Type (2B) incidence-raw data (<code>ChaoSpeciesData$Inci_raw</code>) <br /><br />
In the cottontail capture-recapture experiments conducted by Edwards and Eberhardt (1967), a total of 76 distinct individuals (regarded as 76 &quot;species&quot;) were found in 18 trapping nights. The incidence-raw data consist of a capture/non-capture matrix (where &quot;1&quot; means a capture and &quot;0&quot; means a non-capture) with 76 rows and 18 columns. If we regard this capture-recapture matrix as a species-by-sampling-unit matrix, then species richness estimation can be applied to estimate the size of the cottontail population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ChaoSpeciesData)</code></pre>


<h3>References</h3>

<p>Magurran, A. E. (1988). Ecological Diversity and Its Measurement. 
Princeton University Press, Princeton, New Jersey. <br /><br />
Miller, R. I. and Wiegert, R. G. (1989). Documenting completeness, species-area relations, and the species-abundance distribution of a regional     flora. Ecology, 70, 16-22. <br /><br />
Butler, B. J., and Chazdon, R. L. (1998). Species richness, spatial variation, and abundance of
the soil seed bank of a secondary tropical rain forest. Biotropica, 30, 214-222.<br /><br />
Edwards, W. R. and Eberhardt, L. (1967). Estimating cottontail abundance from live trapping data. The Journal of Wildlife Management, 31, 87-96. 
</p>

<hr>
<h2 id='Diversity'>Estimation of species diversity (Hill numbers)</h2><span id='topic+Diversity'></span>

<h3>Description</h3>

<p><code>Diversity</code>: Estimating a continuous diversity profile in one community including species rich-
ness, Shannon diversity and Simpson diversity). This function also supplies plots of empirical and
estimated continuous diversity profiles. Various estimates for Shannon entropy and the Gini-
Simpson index are also computed. All five types of data are supported: Type (1) abundance data
(datatype=&quot;abundance&quot;), Type (1A) abundance-frequency counts
(datatype=&quot;abundance_freq_count&quot;), Type (2) incidence-frequency data (datatype =
&quot;incidence_freq&quot;), Type (2A) incidence-frequency counts (datatype=&quot;incidence_freq_count&quot;), and
Type (2B) incidence-raw data (datatype=&quot;incidence_raw&quot;); see <code>SpadeR-package</code> details for data input formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Diversity(data, datatype = c("abundance", "abundance_freq_count",
  "incidence_freq", "incidence_freq_count", "incidence_raw"), q = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Diversity_+3A_data">data</code></td>
<td>
<p>a matrix/data.frame of species abundances/incidences.<br /></p>
</td></tr>
<tr><td><code id="Diversity_+3A_datatype">datatype</code></td>
<td>
<p>type of input data, &quot;abundance&quot;, &quot;abundance_freq_count&quot;, &quot;incidence_freq&quot;, &quot;incidence_freq_count&quot; or &quot;incidence_raw&quot;. <br /></p>
</td></tr>
<tr><td><code id="Diversity_+3A_q">q</code></td>
<td>
<p>a vector of nonnegative numbers specifying the diversity orders for which Hill numbers will be estimated. If <code>NULL</code>, then
Hill numbers will be estimated at order q from 0 to 3 with increments of 0.25.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of seven objects: <br /><br />
<code>$Basic_data</code> for summarizing data information. <br /><br />
<code>$Species_richness</code> for showing various species richness estimates along with related statistics. <br /><br />
<code>$Shannon_index</code> and <code>$Shannon_diversity</code> for showing various Shannon index/diversity estimates. <br /><br />
<code>$Simpson_index</code> and <code>$Simpson_diversity</code> for showing two Simpson index/diversity estimates. <br /><br />
<code>$Hill_numbers</code> for showing Hill number (diversity) estimates of diversity orders specified in the argument <code>q</code>. <br /><br />
</p>


<h3>References</h3>

<p>Chao, A., and Chiu, C. H. (2012). Estimation of species richness and shared species richness. In N. Balakrishnan (ed). Methods and Applications of Statistics in the Atmospheric and Earth Sciences. p.76-111, Wiley, New York.<br /><br />
Chao, A. and Jost, L. (2015). Estimating diversity and entropy profiles via discovery rates of new species. Methods in Ecology and Evolution, 6, 873-882.<br /><br />
Chao, A., Wang, Y. T. and Jost, L. (2013). Entropy and the species accumulation curve: a novel estimator of entropy via discovery rates of new species. Methods in Ecology and Evolution 4, 1091-1110.<br /><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(DiversityData)
# Type (1) abundance data 
Diversity(DiversityData$Abu,"abundance",q=c(0,0.5,1,1.5,2))
# Type (1A) abundance-frequency counts data 
Diversity(DiversityData$Abu_count,"abundance_freq_count",q=seq(0,3,by=0.5))
# Type (2) incidence-frequency data 
Diversity(DiversityData$Inci,"incidence_freq",q=NULL)
# Type (2A) incidence-frequency counts data 
Diversity(DiversityData$Inci_freq_count,"incidence_freq_count",q=NULL)
# Type (2B) incidence-raw data 
Diversity(DiversityData$Inci_raw,"incidence_raw",q=NULL)

## End(Not run)
</code></pre>

<hr>
<h2 id='DiversityData'>
Data for Function Diversity
</h2><span id='topic+DiversityData'></span>

<h3>Description</h3>

<p>There are five data sets: <br /><br />
1. Type (1) abundance data (<code>DiversityData$Abu</code>) <br /><br />
The data include a column of the observed tree abundances/frequencies from an old-growth rain forest in Costa Rica (Chao et al. 2005, 2008). There were 69 tree species among 557 individuals. <br /><br />
2. Type (1A) abundance-frequency counts data (<code>DiversityData$Abu_count</code>) <br /><br />
The data consist of the observed beetles species abundance-frequency counts collected from the Osa old-growth forest site in Costa Rica (Janzen, 1973). There were 112 species among 237 individuals. The input abundance-frequency counts data are arranged as = (1 84 2 10 3 4 4 3 ... 42 1); each number needs to be separated by at least one blank space or separated by rows. Here the first pair (1, 84) indicates that there are 84 singletons, the second pair (2, 10) indicates there are 10 doubletons, and so on, with the last pair (42, 1) indicating that there is one species that is represented by 42 individuals. <br /><br />
3. Type (2) incidence-frequency data (<code>DiversityData$Inci</code>) <br /><br />
The single-column data include the observed incidence-based frequencies of tropical rainforest ants collected by Berlese extraction of soil samples (217 sampling units) in Costa Rica (Longino et al. 2002). In the input data, the entry in the first row denotes the number of sampling units (217); the subsequent 117 rows denote species incidence frequencies of the observed species.  <br /><br />
4. Type (2A) incidence-frequency counts data (<code>DiversityData$Inci_freq_count</code>) <br /><br />
The seed-bank data consist of the observed species incidence-based frequency counts of seedlings that germinated from soil samples (Butler and Chazdon, 1998); here each soil sample is regarded as a sampling unit. A total of 34 species of seedlings were found in the 121 soil samples. The incidence frequency counts are read as = (121 1 3 2 2 3 3 ... 61 1); each number needs to be separated by at least one blank space or by separated by rows. The first entry, indicating that there are 121 soil samples, is followed by the 18 pairs (1, 3), (2, 2), (3, 3), (4, 3), (5, 1), (6, 5), and so on, up to (61, 1). Here (1, 3) indicates that there are 3 unique species, (2, 2) indicates there are 2 duplicate species, and so on, with 
(61, 1) indicating that there is one species found in 61 soil samples. <br /><br />
5. Type (2B) incidence-raw data (<code>DiversityData$Inci_raw</code>) <br /><br />
The data consist of raw incidence data of the seed-bank records, described above for the incidence frequency counts data. The input data include a 34 x 121 (species-by-sampling-unit) matrix. For each element of the matrix, &quot;1&quot; means a detection and &quot;0&quot; means a non-detection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(DiversityData)</code></pre>


<h3>References</h3>

<p>Chao, A., Chazdon, R. L., Colwell, R. K. and Shen, T.-J. (2005). A new statistical approach for assessing similarity of species composition with incidence and abundance data. Ecology Letters, 8, 148-159. <br /><br />
Chao, A., Jost, L., Chiang, S.-C., Jiang, Y.-H. and Chazdon, R. L. (2008). A Two-stage probabilistic approach to multiple-community similarity indices. Biometrics, 64, 1178-1186. <br /><br />
Janzen, D. H. (1973) Sweep samples of tropical foliage insects: description of study sites, with data on species abundances and size distributions   . Ecology, 54, 659-686. <br /><br />
Longino, J. T., Coddington, J. A. and Colwell, R. K. (2002). The ant fauna of a tropical rain forest: estimating species richness three different   ways. Ecology, 83, 689-702. <br /><br />
Butler, B. J., and Chazdon, R. L. (1998). Species richness, spatial variation, and abundance of the
soil seed bank of a secondary tropical rain forest. Biotropica, 30, 214-222. <br /><br />
</p>

<hr>
<h2 id='Genetics'>Estimation of genetic differentiation measures</h2><span id='topic+Genetics'></span>

<h3>Description</h3>

<p><code>Genetics</code>: Estimation allelic differentiation among subpopulations based on multiple-subpopulation
genetics data. The richness-based indices include the classic Jaccard and Sorensen dissimilarity
indices; the abundance-based indices include the conventional Gst measure, Horn, Morisita-Horn
and regional species-differentiation indices. <br /><br />
Only Type (1) abundance data (datatype=&quot;abundance&quot;) is supported; input data for each sub-population 
include sample frequencies in an empirical sample of individuals. When there are multiple subpopulations, input data consist of an allele-by-subpopulation frequency matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Genetics(X, q = 2, nboot = 200)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Genetics_+3A_x">X</code></td>
<td>
<p>a matrix, or a data.frame of allele frequencies.</p>
</td></tr>
<tr><td><code id="Genetics_+3A_q">q</code></td>
<td>
<p>a specified order to use to compute pairwise dissimilarity measures. If <code>q = 0</code>, this function computes the estimated pairwise Jaccard and Sorensen dissimilarity indices.
If <code>q = 1</code>, this function computes the estimated pairwise equal-weighted and size-weighted Horn indices;
If <code>q = 2</code>, this function computes the estimated pairwise Morisita-Horn and regional species-diffrentiation indices.</p>
</td></tr>
<tr><td><code id="Genetics_+3A_nboot">nboot</code></td>
<td>
<p>an integer specifying the number of bootstrap replications.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of ten objects: <br /><br />
<code>$info</code> for summarizing data information.<br /><br /> 
<code>$Empirical_richness</code> for showing the observed values of the richness-based dis-similarity indices
including the classic Jaccard and Sorensen indices. <br /><br />
<code>$Empirical_relative</code> for showing the observed values of the equal-weighted dis-similarity
indices for comparing allele relative abundances including Gst, Horn, Morisita-Horn and regional differentiation measures. <br /> <br />
<code>$Empirical_WtRelative</code> for showing the observed value of the dis-similarity index for
comparing size-weighted allele relative abundances, i.e., Horn size-weighted measure based on Shannon-entropy under equal-effort sampling. <br /><br />
The corresponding three objects for showing the estimated dis-similarity indies are: <br />
<code>$estimated_richness</code>,  <code>$estimated_relative</code> and <code>$estimated_WtRelative</code>. <br /><br />
<code>$pairwise</code> and <code>$dissimilarity.matrix</code> for showing respectively the pairwise dis-similarity
estimates (with related statistics) and the dissimilarity matrix for various measures depending on
the diversity order <code>q</code> specified in the function argument. <br /><br />
<code>$q</code> for showing which diversity order <code>q</code> to compute pairwise dissimilarity.
</p>


<h3>References</h3>

<p>Chao, A., and Chiu, C. H. (2016). Bridging the variance and diversity decomposition approaches to beta diversity via similarity and differentiation measures. Methods in Ecology and Evolution, 7, 919-928. <br /><br />
Chao, A., Jost, L., Hsieh, T. C., Ma, K. H., Sherwin, W. B. and Rollins, L. A. (2015). Expected Shannon entropy and Shannon differentiation between subpopulations for neutral genes under the finite island model. Plos One, 10:e0125471.<br /><br />
Jost, L. (2008). <code class="reqn">G_{ST}</code> and its relatives do not measure differentiation. Molecular Ecology, 17, 4015-4026.<br /><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Type (1) abundance data 
data(GeneticsDataAbu)
Genetics(GeneticsDataAbu,q=2,nboot=200)

## End(Not run)
</code></pre>

<hr>
<h2 id='GeneticsDataAbu'>
Human allele frequency data for  Function Genetics
</h2><span id='topic+GeneticsDataAbu'></span>

<h3>Description</h3>

<p>The data taken from Rosenberg et al. (2002) consist of allele frequencies from four human
subpopulations (BiakaPyg, Palestin, Bedouin and Druze). The data are formatted as an allele
(row) by subpopulation (column) matrix file. Entries in each row denote the frequencies of each
allele found in the four subpopulations. The data include an observed allele frequency table with 27 rows and 4 columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GeneticsDataAbu)</code></pre>


<h3>References</h3>

<p>Rosenberg, N. A., Pritchard, J. K., Weber, J. L., Cann, H. M., Kidd, K. K., Zhivotovsky, L. A. and Feldman, M. W. (2002). Genetic structure of human populations. Science, 298, 2381-2385.
</p>

<hr>
<h2 id='SimilarityMult'>Estimation of multiple-community similarity measures</h2><span id='topic+SimilarityMult'></span>

<h3>Description</h3>

<p><code>SimilarityMult</code>: Estimation various <code class="reqn">N</code>-community similarity indices. The richness-based indices
include the classic <code class="reqn">N</code>-community Jaccard and Sorensen indices; the abundance-based indices include the Horn, Morisita-Horn, regional species-overlap, and the <code class="reqn">N</code>-community Bray-Curtis indices.
Three types of data are supported: Type (1) abundance data (datatype=&quot;abundance&quot;), Type (2)
incidence-frequency data (datatype=&quot;incidence_freq&quot;), and Type (2B) incidence-raw data
(datatype=&quot;incidence_raw&quot;); see <code>SpadeR-package</code> details for data input formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SimilarityMult(X, datatype = c("abundance", "incidence_freq",
  "incidence_raw"), units, q = 2, nboot = 200, goal = "relative")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SimilarityMult_+3A_x">X</code></td>
<td>
<p>a matrix/data.frame of species abundances/incidences.<br /></p>
</td></tr>
<tr><td><code id="SimilarityMult_+3A_datatype">datatype</code></td>
<td>
<p>type of input data, &quot;abundance&quot;, &quot;incidence_freq&quot; or &quot;incidence_raw&quot;. <br /></p>
</td></tr>
<tr><td><code id="SimilarityMult_+3A_units">units</code></td>
<td>
<p>number of sampling units in each community. For <code>datatype = "incidence_raw"</code>, users must specify the number of sampling units taken from each community. This argument is not needed for &quot;abundance&quot; and &quot;incidence_freq&quot; data. <br /></p>
</td></tr>
<tr><td><code id="SimilarityMult_+3A_q">q</code></td>
<td>
<p>a specified order to use to compute pairwise similarity measures. If <code>q = 0</code>, this function computes the estimated pairwise richness-based Jaccard and
Sorensen similarity indices.
If <code>q = 1</code> and <code>goal=relative</code>, this function computes the estimated pairwise equal-weighted and size-weighted Horn indices based on Shannon entropy;
If <code>q = 1</code> and <code>goal=absolute</code>, this function computes the estimated pairwise Shannon-entropy-based measure for comparing absolute abundances. If <code>q = 2</code> and <code>goal=relative</code>, 
this function computes the estimated pairwise Morisita-Horn and regional species-overlap indices based on species relative abundances.
If <code>q = 2</code> and <code>goal=absolute</code>, 
this function computes the estimated pairwise Morisita-Horn and regional species-overlap indices based on species absolute abundances.</p>
</td></tr>
<tr><td><code id="SimilarityMult_+3A_nboot">nboot</code></td>
<td>
<p>an integer specifying the number of bootstrap replications.</p>
</td></tr>
<tr><td><code id="SimilarityMult_+3A_goal">goal</code></td>
<td>
<p>a specified estimating goal to use to compute pairwise similarity measures:comparing species relative abundances (<code>goal=relative</code>) or comparing species absolute abundances (<code>goal=absolute</code>). <br /><br /></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of fourteen objects: <br /><br />
<code>$datatype</code> for showing the specified data types (abundance or incidence).<br /><br />
<code>$info</code> for summarizing data information.<br /><br /> 
<code>$Empirical_richness</code> for showing the observed values of the richness-based similarity indices
include the classic <code class="reqn">N</code>-community Jaccard and Sorensen indices. <br /><br />
<code>$Empirical_relative</code> for showing the observed values of the equal-weighted similarity indices
for comparing species relative abundances including Horn, Morisita-Horn and regional overlap measures. <br /> <br />
<code>$Empirical_WtRelative</code> for showing the observed value of the Horn similarity index for comparing
size-weighted species relative abundances based on Shannon entropy under equal-effort sampling. <br /><br />
<code>$Empirical_absolute</code> for showing the observed values of the similarity indices for comparing
absolute abundances. These measures include the Shannon-entropy-based measure, Morisita-Horn and the regional species-overlap measures based on species absolute abundance, as well as the <code class="reqn">N</code>-community Bray-Curtis index.
All measures are valid only under equal-effort sampling. <br /><br />
The corresponding four objects for showing the estimated similarity indices are:
<code>$estimated_richness</code>, <code>$estimated_relative</code>, <code>$estimated_WtRelative</code> and <code>$estimated_absolute</code>. <br /><br />
<code>$pairwise</code> and <code>$similarity.matrix</code> for showing respectively the pairwise dis-similarity
estimates (with related statistics) and the similarity matrix for various measures depending on the
diversity order <code>q</code> and the <code>goal</code> aspecified in the function arguments. <br /><br />
<code>$goal</code> for showing the goal specified in the argument goal (absolute or relative) used to compute pairwise similarity.<br /><br />
<code>$q</code> for showing which diversity order <code>q</code> specified to compute pairwise similarity. <br /><br />
</p>


<h3>References</h3>

<p>Chao, A., and Chiu, C. H. (2016). Bridging the variance and diversity decomposition approaches to beta diversity via similarity and differentiation measures. Methods in Ecology and Evolution, 7, 919-928. <br /><br />
Chao, A., Jost, L., Hsieh, T. C., Ma, K. H., Sherwin, W. B. and Rollins, L. A. (2015). Expected Shannon entropy and Shannon differentiation between subpopulations for neutral genes under the finite island model. Plos One, 10:e0125471.<br /><br />
Chiu, C. H., Jost, L. and Chao, A. (2014). Phylogenetic beta diversity, similarity, and differentiation measures based on Hill numbers. Ecological Monographs, 84, 21-44.<br /><br />
Gotelli, N. G. and Chao, A. (2013). Measuring and estimating species richness, species diver- sity,
and biotic similarity from sampling data. Encyclopedia of Biodiversity, 2nd Edition, Vol. 5, 195-211, Waltham, MA.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(SimilarityMultData)
# Type (1) abundance data 
SimilarityMult(SimilarityMultData$Abu,"abundance",q=2,nboot=200,"relative")
# Type (2) incidence-frequency data 
SimilarityMult(SimilarityMultData$Inci,"incidence_freq",q=2,nboot=200,"relative")
# Type (2B) incidence-raw data 
SimilarityMult(SimilarityMultData$Inci_raw,"incidence_raw",
units=c(19,17,15),q=2,nboot=200,"relative")

## End(Not run)
</code></pre>

<hr>
<h2 id='SimilarityMultData'>
Data for Function SimilarityMult
</h2><span id='topic+SimilarityMultData'></span>

<h3>Description</h3>

<p>There are three data sets: <br /><br />
1. Type (1) abundance data (<code>SimilarityMultData$Abu</code>) <br /><br />
The data include the observed species frequencies of three communities: seedlings (column 1), saplings (column 2) and trees (column 3) collected from an old-growth rain forest; see Chao et al. (2005, 2008). The three entries in each row are the observed frequency (or abundance) of each species from the three communities. <br /><br />
2. Type (2) incidence-frequency data (<code>SimilarityMultData$Inci</code>) <br /><br />
The data include the observed incidence frequencies of tropical rainforest ants using three sampling techniques: (a) Berlese extraction of soil samples (217 samples), (b) fogging samples from canopy fogging (459 samples), and (c) Malaise trap samples for flying and crawling insects (62 samples); The data were collected in Costa-Rica (Longino et al. 2002). The three entries in the first row of the input data denote the number of sampling units (217, 459 and 62). Beginning with the second row, the three numbers in each row denotes incidence frequencies (the total number of detections) in the samples based on three sampling techniques. <br /><br />
3. Type (2B) incidence-raw data (<code>SimilarityMultData$Inci_raw</code>) <br /><br />
The data include the observed soil ciliate species detection/non-detection data for a total of 51 soil samples from three areas of Namibia, Africa: Etosha Pan (19 samples), Central Namib Desert (17 samples) and Southern Namib Desert (15 samples). The raw detection/non-detection data include 365 x 51 matrix of 0's and 1's (0 denotes a non-detection and 1 denotes a detection).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SimilarityMultData)</code></pre>


<h3>References</h3>

<p>Chao, A., Chazdon, R. L., Colwell, R. K. and Shen, T.-J. (2005). A new statistical approach for assessing similarity of species composition with incidence and abundance data. Ecology Letters, 8, 148-159.<br /><br />
Chao, A., Jost, L., Chiang, S.-C., Jiang, Y.-H. and Chazdon, R. L. (2008). A Two-stage probabilistic approach to multiple-community similarity indices. Biometrics, 64, 1178-1186. <br /><br />
Longino, J. T., Coddington, J. A. and Colwell, R. K. (2002). The ant fauna of a tropical rain forest: estimating species richness three different ways. Ecology, 83, 689-702. <br /><br />
Foissner, W., Agatha, S. and Berger, H. (2002). Soil Ciliates (Protozoa, Ciliophora) from Namibia (Southwest Africa), with emphasis on two contrasting environments, the Etosha Region and the Namib Desert. Denisia, 5, 1-1459.
</p>

<hr>
<h2 id='SimilarityPair'>Estimation of two-assemblage similarity measures</h2><span id='topic+SimilarityPair'></span>

<h3>Description</h3>

<p><code>SimilarityPair</code>: Estimation various similarity indices for two assemblages. The richness-based
indices include the classic two-community Jaccard and Sorensen indices; the abundance-based
indices include the Horn, Morisita-Horn, regional species-overlap, two-community Bray-Curtis and the
abundance-based Jaccard and Sorensen indices. Three types of data are supported: Type (1)
abundance data (datatype=&quot;abundance&quot;), Type (2) incidence-frequency data
(datatype=&quot;incidence_freq&quot;), and Type (2B) incidence-raw data (datatype=&quot;incidence_raw&quot;); see
<code>SpadeR-package</code> details for data input formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SimilarityPair(X, datatype = c("abundance", "incidence_freq",
  "incidence_raw"), units, nboot = 200)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SimilarityPair_+3A_x">X</code></td>
<td>
<p>a matrix/data.frame of species abundances/incidences.<br /></p>
</td></tr>
<tr><td><code id="SimilarityPair_+3A_datatype">datatype</code></td>
<td>
<p>type of input data, &quot;abundance&quot;, &quot;incidence_freq&quot; or &quot;incidence_raw&quot;. <br /></p>
</td></tr>
<tr><td><code id="SimilarityPair_+3A_units">units</code></td>
<td>
<p>number of sampling units in each community. For <code>datatype = "incidence_raw"</code>, users must specify the number of sampling units taken from each community. This argument is not needed for &quot;abundance&quot; and &quot;incidence_freq&quot; data. <br /></p>
</td></tr>
<tr><td><code id="SimilarityPair_+3A_nboot">nboot</code></td>
<td>
<p>an integer specifying the number of replications.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of ten objects: <br /><br />
<code>$datatype</code> for showing the specified data types (abundance or incidence). <br /><br />
<code>$info</code> for summarizing data information. <br /><br />
<code>$Empirical_richness</code> for showing the observed values of the richness-based similarity indices
include the classic two-community Jaccard and Sorensen indices. <br /><br />
<code>$Empirical_relative</code> for showing the observed values of the equal-weighted similarity indices
for comparing species relative abundances including Horn, Morisita-Horn, regional overlap,
Chao-Jaccard and Chao-Sorensen abundance (or incidence) measures based on species relative abundances. <br /> <br />
<code>$Empirical_WtRelative</code> for showing the observed value of the Horn similarity index for comparing
size-weighted species relative abundances based on Shannon entropy under equal-effort sampling. <br /><br />
<code>$Empirical_absolute</code> for showing the observed values of the similarity indices for comparing
absolute abundances. These measures include the Shannon-entropy-based measure, 
Morisita-Horn and the regional overlap measures based on species absolute abundances, as well as the Bray-Curtis index.
All measures are valid only under equal-effort sampling. <br /><br />
The corresponding four objects for showing the estimated similarity indices are:
<code>$estimated_richness</code>, <code>$estimated_relative</code>, <code>$estimated_WtRelative</code> and <code>$estimated_Absolute</code>. <br /><br />
</p>


<h3>References</h3>

<p>Chao, A., Chazdon, R. L., Colwell, R. K. and Shen, T.-J. (2005). A new statistical approach for assessing similarity of species composition with incidence and abundance data. Ecology Letters, 8, 148-159.<br /><br />
Chao, A., and Chiu, C. H. (2016). Bridging the variance and diversity decomposition approaches to beta diversity via similarity and differentiation measures. Methods in Ecology and Evolution, 7, 919-928. <br /><br />
Chao, A., Jost, L., Hsieh, T. C., Ma, K. H., Sherwin, W. B. and Rollins, L. A. (2015). Expected
Shannon entropy and Shannon differentiation between subpopulations for neutral genes under the finite island model. Plos One, 10:e0125471. <br /><br />
Chiu, C. H., Jost, L. and Chao, A. (2014). Phylogenetic beta diversity, similarity, and differentiation measures based on Hill numbers. Ecological Monographs, 84, 21-44.<br /><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(SimilarityPairData)
# Type (1) abundance data 
SimilarityPair(SimilarityPairData$Abu,"abundance",nboot=200)
# Type (2) incidence-frequency data 
SimilarityPair(SimilarityPairData$Inci,"incidence_freq",nboot=200)
# Type (2B) incidence-raw data 
SimilarityPair(SimilarityPairData$Inci_raw,"incidence_raw",units=c(19,17),nboot=200)

## End(Not run)
</code></pre>

<hr>
<h2 id='SimilarityPairData'>
Data for Function SimilarityPair
</h2><span id='topic+SimilarityPairData'></span>

<h3>Description</h3>

<p>There are three data sets: <br /><br />
1. Type (1) abundance data (<code>SimilarityPairData$Abu</code>)  <br /><br />
The data include the observed species frequencies of two communities: seedlings (column 1), and trees (column 2) collected from an old-growth rain forest; see Chao et al. (2005, 2008). The two entries in each row are the observed frequency (or abundance) of each species from the two communities. (These data are subset of <code>SimilarityMultData$Abu</code> used in the function SimilarityMult.) <br /><br />
2. Type (2) incidence-frequency data (<code>SimilarityPairData$Inci</code>) <br /><br />
The data include the observed incidence frequencies of tropical rainforest ants based on two sampling techniques: (a) Berlese extraction of soil samples (217 samples), and (b) Malaise trap samples for flying and crawling insects (62 samples); see Longino et al. (2002). The two entries in first row of the input data denote the number of sampling units (217 and 62). Beginning with the second row, the two numbers in each row denotes incidence frequencies (the total number of detections) in the soil samples based on the two sampling techniques. (These data are subset of <code>SimilarityMultData$Inci</code> used in the function SimilarityMult.)  <br /><br />
3. Type (2B) incidence-raw data (<code>SimilarityPairData$Inci_raw</code>) <br /><br />
The data include the observed soil ciliate species detection/non-detection data for a total of 36 soil samples from two areas of Namibia, Africa: Etosha Pan (19 samples), and Central Namib Desert (17 samples). The raw detection/non-detection data include 365 x 36 matrix of 0's and 1's (0 denotes a non-detection and 1 denotes a detection). (These data are subset of <code>SimilarityMultData$Inci_raw</code> used in the function SimilarityMult.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SimilarityPairData)</code></pre>


<h3>References</h3>

<p>Chao, A., Chazdon, R. L., Colwell, R. K. and Shen, T.-J. (2005). A new statistical approach for assessing similarity of species composition with incidence and abundance data. Ecology Letters, 8, 148-159.<br /><br />
Chao, A., Jost, L., Chiang, S.-C., Jiang, Y.-H. and Chazdon, R. L. (2008). A Two-stage probabilistic approach to multiple-community similarity indices. Biometrics, 64, 1178-1186.<br /><br />
Longino, J. T., Coddington, J. A. and Colwell, R. K. (2002). The ant fauna of a tropical rain forest: estimating species richness three different ways. Ecology, 83, 689-702. <br /><br />
Foissner, W., Agatha, S. and Berger, H. (2002) Soil Ciliates (Protozoa, Ciliophora) from Namibia (Southwest Africa), with emphasis on two Contrasting environments, the Etosha Region and the Namib Desert. Denisia, 5, 1-1459.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
