<!DOCTYPE html><html><head><title>Help for package SparseDC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SparseDC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cell_type_biase'><p>Biase Data Cell Type</p></a></li>
<li><a href='#condition_biase'><p>Biase Data Conditions</p></a></li>
<li><a href='#data_biase'><p>Biase Data</p></a></li>
<li><a href='#generate_uni_dat'><p>Uniform data generator</p>
For use with the gap statistic. Generates datasets drawn from the reference
distribution where each reference feature is generated uniformly over the
range of observed values for that feature.</a></li>
<li><a href='#lambda1_calculator'><p>Lambda 1 Calculator.</p></a></li>
<li><a href='#lambda2_calculator'><p>Lambda 2 Calculator.</p></a></li>
<li><a href='#pre_proc_data'><p>Pre-process Data</p></a></li>
<li><a href='#S_func'><p>The soft thresholding operator</p></a></li>
<li><a href='#sim_data'><p>Data Simulator</p></a></li>
<li><a href='#sparsedc_cluster'><p>Sparse Differential Clustering</p></a></li>
<li><a href='#sparsedc_gap'><p>Gap Statistic Calculator</p></a></li>
<li><a href='#update_c'><p>Update Clusters</p></a></li>
<li><a href='#update_mu'><p>Update the Center Values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Implementation of SparseDC Algorithm</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.17</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the algorithm described in 
    Barron, M., Zhang, S. and Li, J. 2017, "A sparse differential
    clustering algorithm for tracing cell type changes via single-cell
    RNA-sequencing data", Nucleic Acids Research, gkx1113,
    &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkx1113">doi:10.1093/nar/gkx1113</a>&gt;. This algorithm clusters samples from two
    different populations, links the clusters across the conditions and
    identifies marker genes for these changes. The package was designed for
    scRNA-Seq data but is also applicable to many other data types, just
    replace cells with samples and genes with variables. The package also
    contains functions for estimating the parameters for SparseDC as outlined
    in the paper. We recommend that users further select their marker genes
    using the magnitude of the cluster centers.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-01-04 17:55:17 UTC; Administrator</td>
</tr>
<tr>
<td>Author:</td>
<td>Jun Li [aut, cre],
  Martin Barron [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jun Li &lt;jun.li@nd.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-01-04 19:11:30 UTC</td>
</tr>
</table>
<hr>
<h2 id='cell_type_biase'>Biase Data Cell Type</h2><span id='topic+cell_type_biase'></span>

<h3>Description</h3>

<p>The cell type of each of the cells in the Biase data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cell_type_biase
</code></pre>


<h3>Format</h3>

<p>An R.Data object containing a vector with the cell type of each of the
cells in the Biase Data.</p>

<hr>
<h2 id='condition_biase'>Biase Data Conditions</h2><span id='topic+condition_biase'></span>

<h3>Description</h3>

<p>The condition for each sample in the Biase data. To be used when splitting
the data to demonstrate SparseDC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>condition_biase
</code></pre>


<h3>Format</h3>

<p>An R.Data object containing a vector with the conditon of the 49 cells
in the Biase data.</p>

<hr>
<h2 id='data_biase'>Biase Data</h2><span id='topic+data_biase'></span>

<h3>Description</h3>

<p>This dataset was created by Biase et al. to study cell fat inclination in
mouse embryos. It contains FPKM gene expression measurements for 49 cells
and 16,514 genes. There are three cell types in the dataset, zygote,
two-cell embryo, and four-cell embryo cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_biase
</code></pre>


<h3>Format</h3>

<p>An R.Data object storing FPKM gene expression
measurements for each of the samples.</p>

<hr>
<h2 id='generate_uni_dat'>Uniform data generator
For use with the gap statistic. Generates datasets drawn from the reference
distribution where each reference feature is generated uniformly over the
range of observed values for that feature.</h2><span id='topic+generate_uni_dat'></span>

<h3>Description</h3>

<p>Uniform data generator
For use with the gap statistic. Generates datasets drawn from the reference
distribution where each reference feature is generated uniformly over the
range of observed values for that feature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_uni_dat(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_uni_dat_+3A_data">data</code></td>
<td>
<p>A dataset with rows as features and columns as samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataset drawn from the reference distribution for use internally
with the gap statistic.
</p>

<hr>
<h2 id='lambda1_calculator'>Lambda 1 Calculator.</h2><span id='topic+lambda1_calculator'></span>

<h3>Description</h3>

<p>Calculates the lambda 1 value for the sparseDC algorithm. The lambda 1
value controls the number of marker genes selected for each
cluster in the output from SparseDC. It is calculated as the value of
lambda 1 that results in no marker genes being selected when then are no
meaningful clusters present in the data. Please see the original manuscript
for further details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lambda1_calculator(pdat1, pdat2, ncluster, alpha1 = 0.5, nboot1 = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lambda1_calculator_+3A_pdat1">pdat1</code></td>
<td>
<p>The centered data from condition 1, columns should be
samples (cells) and rows should be features (genes).</p>
</td></tr>
<tr><td><code id="lambda1_calculator_+3A_pdat2">pdat2</code></td>
<td>
<p>The centered data from condition 2, columns should be
samples (cells) and rows should be features (genes). The number of genes
should be the same as <code>pdat1</code>.
as in pdat1.</p>
</td></tr>
<tr><td><code id="lambda1_calculator_+3A_ncluster">ncluster</code></td>
<td>
<p>The number of clusters present in the data.</p>
</td></tr>
<tr><td><code id="lambda1_calculator_+3A_alpha1">alpha1</code></td>
<td>
<p>The quantile of the bootstrapped lambda 1 values to use,
range is (0,1). The default value is 0.5, the median of the calculated
lambda 1 values.</p>
</td></tr>
<tr><td><code id="lambda1_calculator_+3A_nboot1">nboot1</code></td>
<td>
<p>The number of bootstrap repetitions used for estimating lambda 1,
the default value is 1000.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The calculated value of lambda 1 to use in the main SparseDC
algorithm.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lambda2_calculator">lambda2_calculator</a></code> for how to calculate the lambda 2
parameter. <code><a href="#topic+sparsedc_cluster">sparsedc_cluster</a></code> for the main sparse differential
clustering function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(10)
# Select small dataset for example
data_test &lt;- data_biase[1:100,]
# Split data into conditions A and B
data_A &lt;- data_test[ , which(condition_biase == "A")]
data_B &lt;- data_test[ , which(condition_biase == "B")]
# Pre-process the data
pre_data &lt;- pre_proc_data(data_A, data_B, norm = FALSE, log = TRUE,
center = TRUE)
# Calculate the lambda 1 value
lambda1_calculator(pdat1 = pre_data[[1]], pdat2 = pre_data[[2]], ncluster=3,
 alpha1 = 0.5, nboot1 = 1000)

# Can also run

# Pre-process the data
pre_data &lt;- pre_proc_data(data_A, data_B, norm = FALSE, log = TRUE,
center = TRUE)
pdata_A &lt;- pre_data[[1]]
pdata_B &lt;- pre_data[[2]]
# Calculate the lambda 1 value
lambda1_calculator(pdat1 = pdata_A, pdat2 = pdata_B , ncluster=3,
 alpha1 = 0.5, nboot1 = 1000)

</code></pre>

<hr>
<h2 id='lambda2_calculator'>Lambda 2 Calculator.</h2><span id='topic+lambda2_calculator'></span>

<h3>Description</h3>

<p>Calculates the lambda 2 values for use in the main SparseDC algorithm, the
lambda 2 value controls the number of genes that show condition-dependent
expression within each cell type. That is it controls the number of
different mean values across the conditions for each cluster. It is
calculated by estimating the value of lambda 2 that would result in no
difference in mean values across conditions when there are no meaningful
differences across between the conditions. For further details please see
the original manuscript.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lambda2_calculator(pdat1, pdat2, ncluster, alpha2 = 0.5, nboot2 = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lambda2_calculator_+3A_pdat1">pdat1</code></td>
<td>
<p>The centered data from condition 1, columns should be
samples (cells) and rows should be features (genes).</p>
</td></tr>
<tr><td><code id="lambda2_calculator_+3A_pdat2">pdat2</code></td>
<td>
<p>The centered data from condition 2, columns should be
samples (cells) and rows should be features (genes). The number of genes
should be the same as <code>pdat1</code>.
as in pdat1.</p>
</td></tr>
<tr><td><code id="lambda2_calculator_+3A_ncluster">ncluster</code></td>
<td>
<p>The number of clusters present in the data.</p>
</td></tr>
<tr><td><code id="lambda2_calculator_+3A_alpha2">alpha2</code></td>
<td>
<p>The quantile of the bootstrapped lambda 2 values to use,
range is (0,1). The default value is 0.5, the median of the calculated
lambda 2 values.</p>
</td></tr>
<tr><td><code id="lambda2_calculator_+3A_nboot2">nboot2</code></td>
<td>
<p>The number of bootstrap repetitions for estimating lambda 2,
the default value is 1000.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The calculated value of lambda 2 to use in the main SparseDC
algorithm.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lambda1_calculator">lambda1_calculator</a></code>  <code><a href="#topic+sparsedc_cluster">sparsedc_cluster</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(10)
# Select small dataset for example
data_test &lt;- data_biase[1:100,]
# Split data into conditions A and B
data_A &lt;- data_test[ , which(condition_biase == "A")]
data_B &lt;- data_test[ , which(condition_biase == "B")]
# Pre-process the data
pre_data &lt;- pre_proc_data(data_A, data_B, norm = FALSE, log = TRUE,
center = TRUE)
# Calculate the lambda 2 value
lambda2_calculator(pdat1 = pre_data[[1]], pdat2 = pre_data[[2]], ncluster = 3,
 alpha2 = 0.5, nboot2 = 1000)

 # Can also run
 pdata_A &lt;- pre_data[[1]]
 pdata_B &lt;- pre_data[[2]]
lambda2_calculator(pdat1 = pdata_A, pdat2 = pdata_B, ncluster = 3,
 alpha2 = 0.5, nboot2 = 1000)

</code></pre>

<hr>
<h2 id='pre_proc_data'>Pre-process Data</h2><span id='topic+pre_proc_data'></span>

<h3>Description</h3>

<p>This function pre-process the data so that SparseDC can be applied.
SparseDC requires data that have been normalized for sequencing depth,
log-transformed and centralized on a gene-by-gene basis. For the sequencing
depth normalization we recommend that users use one of the many methods
developed for normalizing scRNA-Seq data prior to using SparseDC and so
can set <code>norm = FALSE</code>. However, here we normalize the data by dividing
by the total number of reads. This function log transforms the data by
applying <code>log(x + 1)</code> to each of the data sets. By far the most
important pre-processing step for SparseDC is the centralization of the data.
Having centralized data is a core component of the SparseDC algorithm and is
necessary for both accurate clustering of the cells and identifying marker
genes. We therefore recommend that all users centralize their data using
this function and that only experienced users set <code>center = FALSE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre_proc_data(dat1, dat2, norm = TRUE, log = TRUE, center = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pre_proc_data_+3A_dat1">dat1</code></td>
<td>
<p>The data for the first condition with samples (cells) as columns
and features (genes) as rows.</p>
</td></tr>
<tr><td><code id="pre_proc_data_+3A_dat2">dat2</code></td>
<td>
<p>The data for the second condition with samples (cells) as
columns and features (genes) as rows.</p>
</td></tr>
<tr><td><code id="pre_proc_data_+3A_norm">norm</code></td>
<td>
<p>This parameter controls whether the data is normalized for
sequencing depth by dividing each column by the total number of reads for
that sample. We recommend that user use one of the many methods for
normalizing scRNA-Seq data and so set this as <code>FALSE</code>. The default
value is <code>TRUE</code></p>
</td></tr>
<tr><td><code id="pre_proc_data_+3A_log">log</code></td>
<td>
<p>This parameter controls whether the data is transformed using
<code>log(x + 1)</code>. The default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="pre_proc_data_+3A_center">center</code></td>
<td>
<p>This parameter controls whether the data is centered on a gene
by gene basis. We recommend all users center their data prior to applying
SparseDC and only experienced users should set this as <code>FALSE</code>. The
default value is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns the two pre-processed datasets stored as a
list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(10)
# Select small dataset for example
data_test &lt;- data_biase[1:100,]
# Split data into condition A and B
data_A &lt;- data_test[ , which(condition_biase == "A")]
data_B &lt;- data_test[ , which(condition_biase == "B")]
# Pre-process the data
pre_data &lt;- pre_proc_data(data_A, data_B, norm = FALSE, log = TRUE,
center = TRUE)
# Extract Data
pdata_A &lt;- pre_data[[1]]
pdata_B &lt;- pre_data[[2]]

</code></pre>

<hr>
<h2 id='S_func'>The soft thresholding operator</h2><span id='topic+S_func'></span>

<h3>Description</h3>

<p>Function to solve the soft thresholding problem
</p>


<h3>Usage</h3>

<pre><code class='language-R'>S_func(x, a)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="S_func_+3A_x">x</code></td>
<td>
<p>The data value</p>
</td></tr>
<tr><td><code id="S_func_+3A_a">a</code></td>
<td>
<p>The lambda value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The solution to the soft thresholding operator.
</p>

<hr>
<h2 id='sim_data'>Data Simulator</h2><span id='topic+sim_data'></span>

<h3>Description</h3>

<p>Simulates two condition data for a range of conditions depending on the
parameters used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_data(genes, cells, sig.genes, sig.genes.s, clus.t1, clus.t2,
  same.sig = FALSE, u.l = 1, u.h = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_data_+3A_genes">genes</code></td>
<td>
<p>The number of genes to be simulated.</p>
</td></tr>
<tr><td><code id="sim_data_+3A_cells">cells</code></td>
<td>
<p>The number of cells to be simulated per condition.</p>
</td></tr>
<tr><td><code id="sim_data_+3A_sig.genes">sig.genes</code></td>
<td>
<p>The number of marker genes for each cluster.</p>
</td></tr>
<tr><td><code id="sim_data_+3A_sig.genes.s">sig.genes.s</code></td>
<td>
<p>The number of marker genes shared across conditions for
each cluster. Should be less than or equal to <code>sig.genes</code></p>
</td></tr>
<tr><td><code id="sim_data_+3A_clus.t1">clus.t1</code></td>
<td>
<p>A vector of clusters present in the first condition. Start
at 1, e.g. <code>c(1,2,3,4)</code></p>
</td></tr>
<tr><td><code id="sim_data_+3A_clus.t2">clus.t2</code></td>
<td>
<p>A vector of clusters present in the second condition. Does
not have to match clus.t1, e.g <code>c(3,4,5)</code></p>
</td></tr>
<tr><td><code id="sim_data_+3A_same.sig">same.sig</code></td>
<td>
<p>TRUE or FALSE. Should each cluster have a unique set of
marker genes. default is FALSE.</p>
</td></tr>
<tr><td><code id="sim_data_+3A_u.l">u.l</code></td>
<td>
<p>Lower bound for the cluster gene means, default is 1.</p>
</td></tr>
<tr><td><code id="sim_data_+3A_u.h">u.h</code></td>
<td>
<p>Upper bound for the cluster gene means, default is 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the two simulated data matrices, <code>dat.1</code> and
<code>dat.2</code>, true clusters for the cells in the first and second
conditions, <code>clusters1</code> and <code>clusters2</code>, a matrix indicating
marker genes for the first and second condition, <code>sig.gene.mat.1</code>
and <code>sig.gene.mat.2</code>, the base mean values for each gene
<code>gene.means</code> and the cluster specific additions for each gene
<code>clus.gene.means</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(10)
genes &lt;- 1000  # Simulate 1,000 genes
cells &lt;- 100  # Simulate 100 cells per condition
clus.t1 &lt;- c(1,2,3)  # Generate 3 clusters present in condition A
clus.t2 &lt;- c(1,2,3)  # Generate 3 clusters present in condition B
sig.genes &lt;- 30  # Generate 30 marker genes per cluster
sig.genes.s &lt;- 15  # Let half of the 30 marker genes be shared.
temp_sim_dat &lt;- sim_data(genes, cells, sig.genes, sig.genes.s,
clus.t1, clus.t2)

</code></pre>

<hr>
<h2 id='sparsedc_cluster'>Sparse Differential Clustering</h2><span id='topic+sparsedc_cluster'></span>

<h3>Description</h3>

<p>The main SparseDC function. This function clusters the samples from the two
conditions and links the clusters across the conditions. It also identifies
marker genes for each of the clusters. There are three types of marker gene
which SparseDC identifies. Please see the original manuscript for further
details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sparsedc_cluster(pdat1, pdat2, ncluster, lambda1, lambda2, nitter = 20,
  nstarts = 50, init_iter = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sparsedc_cluster_+3A_pdat1">pdat1</code></td>
<td>
<p>The centered data from condition 1, columns should be
samples (cells) and rows should be features (genes).</p>
</td></tr>
<tr><td><code id="sparsedc_cluster_+3A_pdat2">pdat2</code></td>
<td>
<p>The centered data from condition 2, columns should be
samples (cells) and rows should be features (genes). The number of genes
should be the same as <code>pdat1</code>.
as in pdat1.</p>
</td></tr>
<tr><td><code id="sparsedc_cluster_+3A_ncluster">ncluster</code></td>
<td>
<p>The number of clusters present in the data.</p>
</td></tr>
<tr><td><code id="sparsedc_cluster_+3A_lambda1">lambda1</code></td>
<td>
<p>The lambda 1 value to use in the SparseDC function. This
value controls the number of marker genes detected for each of the clusters
in the final result. This can be calculated using the &quot;lambda1_calculator&quot;
function or supplied by the user.</p>
</td></tr>
<tr><td><code id="sparsedc_cluster_+3A_lambda2">lambda2</code></td>
<td>
<p>The lambda 2 value to use in the SparseDC function. This
value controls the number of genes that show condition-dependent
expression within each cell type. This can be calculated using the
&quot;lambda2_calculator&quot; function or supplied by the user.</p>
</td></tr>
<tr><td><code id="sparsedc_cluster_+3A_nitter">nitter</code></td>
<td>
<p>The max number of iterations for each of the start values, the
default value is 20.</p>
</td></tr>
<tr><td><code id="sparsedc_cluster_+3A_nstarts">nstarts</code></td>
<td>
<p>The number of start values to use for SparseDC. The default
value is 50.</p>
</td></tr>
<tr><td><code id="sparsedc_cluster_+3A_init_iter">init_iter</code></td>
<td>
<p>The number of iterations used to generate the starting
center values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the clustering solution, cluster centers  and the
score of each of the starts.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lambda1_calculator">lambda1_calculator</a></code>  <code><a href="#topic+lambda2_calculator">lambda2_calculator</a></code>
<code><a href="#topic+update_c">update_c</a></code> <code><a href="#topic+update_mu">update_mu</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(10)
# Select small dataset for example
data_test &lt;- data_biase[1:100,]
# Split data into conditions 1 and 2
data_1 &lt;- data_test[ , which(condition_biase == "A")]
data_2 &lt;- data_test[ , which(condition_biase == "B")]
# Preprocess data (log transform and center)
pre_data &lt;- pre_proc_data(data_1, data_2, norm = FALSE, log = TRUE,
center = TRUE)
# Calculate lambda 1 parameter
lambda1 &lt;- lambda1_calculator(pdat1 = pre_data[[1]], pdat2 = pre_data[[2]],
 ncluster=3, alpha1 = 0.5, nboot1 = 1000)
# Calculate lambda 2 parameter
lambda2 &lt;- lambda2_calculator(pdat1 = pre_data[[1]], pdat2 = pre_data[[2]],
 ncluster = 3, alpha2 = 0.5, nboot2 = 1000)
# Run sparse DC
sdc_res &lt;- sparsedc_cluster(pdat1 = pre_data[[1]], pdat2 = pre_data[[2]], ncluster = 3,
lambda1 = lambda1, lambda2 = lambda2, nitter = 20, nstarts =50)
# Extract results
clusters_1 &lt;- sdc_res$clusters1  # Clusters for condition 1 data
clusters_2 &lt;- sdc_res$clusters2  # Clusters for condition 2 data
centers_1 &lt;- sdc_res$centers1  # Centers for condition 1 data
centers_2 &lt;- sdc_res$centers2  # Centers for condition 2 data
# View clusters
summary(as.factor(clusters_1))
summary(as.factor(clusters_2))
# View Marker genes
gene_names &lt;- row.names(data_test)
m_gene_c1_up1 &lt;- gene_names[which(centers_1[,1] &gt; 0)]
m_gene_c1_up2 &lt;- gene_names[which(centers_2[,1] &gt; 0)]
m_gene_c1_down1 &lt;- gene_names[which(centers_1[,1] &lt; 0)]
m_gene_c1_down2 &lt;- gene_names[which(centers_2[,1] &lt; 0)]
m_gene_c2_cond &lt;- gene_names[which(centers_1[,2] != centers_2[,2])]

# Can also run

pre_data &lt;- pre_proc_data(data_1, data_2, norm = FALSE, log = TRUE,
center = TRUE)
pdata_A &lt;- pre_data[[1]]
pdata_B &lt;- pre_data[[2]]
lambda1 &lt;- lambda1_calculator(pdat1 = pdata_A , pdat2 = pdata_B,
ncluster=3, alpha1 = 0.5, nboot1 = 1000)
lambda2 &lt;- lambda2_calculator(pdat1 = pdata_A, pdat2 = pdata_B,
 ncluster = 3, alpha2 = 0.5, nboot2 = 1000)
# Run sparse DC
sdc_res &lt;- sparsedc_cluster(pdat1 = pdata_A, pdat2 = pdata_B, ncluster = 3,
lambda1 = lambda1, lambda2 = lambda2, nitter = 20, nstarts =50)

</code></pre>

<hr>
<h2 id='sparsedc_gap'>Gap Statistic Calculator</h2><span id='topic+sparsedc_gap'></span>

<h3>Description</h3>

<p>This function calculates the gap statistic for SparseDC. For use
when the number of clusters in the data is unknown. We recommend
using alternate methods to infer the number of clusters in the
data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sparsedc_gap(pdat1, pdat2, min_clus, max_clus, nboots = 200, nitter = 20,
  nstarts = 10, l1_boot = 50, l2_boot = 50)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sparsedc_gap_+3A_pdat1">pdat1</code></td>
<td>
<p>The centered data from condition 1, columns should be
samples (cells) and rows should be features (genes).</p>
</td></tr>
<tr><td><code id="sparsedc_gap_+3A_pdat2">pdat2</code></td>
<td>
<p>The centered data from condition 2, columns should be
samples (cells) and rows should be features (genes). The number of genes
should be the same as <code>pdat1</code>.
as in pdat1.</p>
</td></tr>
<tr><td><code id="sparsedc_gap_+3A_min_clus">min_clus</code></td>
<td>
<p>The minimum number of clusters to try, minimum value is 2.</p>
</td></tr>
<tr><td><code id="sparsedc_gap_+3A_max_clus">max_clus</code></td>
<td>
<p>The maximum number of clusters to try.</p>
</td></tr>
<tr><td><code id="sparsedc_gap_+3A_nboots">nboots</code></td>
<td>
<p>The number of bootstrap repetitions to use, default = 200.</p>
</td></tr>
<tr><td><code id="sparsedc_gap_+3A_nitter">nitter</code></td>
<td>
<p>The max number of iterations for each of the start values, the
default value is 20.</p>
</td></tr>
<tr><td><code id="sparsedc_gap_+3A_nstarts">nstarts</code></td>
<td>
<p>The number of start values to use for SparseDC. The default
value is 10.</p>
</td></tr>
<tr><td><code id="sparsedc_gap_+3A_l1_boot">l1_boot</code></td>
<td>
<p>The number of bootstrap repetitions used for estimating
lambda 1.</p>
</td></tr>
<tr><td><code id="sparsedc_gap_+3A_l2_boot">l2_boot</code></td>
<td>
<p>The number of bootstrap repetitions used for estimating
lambda 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the optimal number of clusters, as well as gap
statistics and the calculated standard error for each number of clusters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load a small dataset
data_test &lt;- data_biase[1:50,]
# Split data into conditions 1 and 2
data_1 &lt;- data_test[ , which(condition_biase == "A")]
data_2 &lt;- data_test[ , which(condition_biase == "B")]
# Preprocess data (log transform and center)
pre_data &lt;- pre_proc_data(data_1, data_2, norm = FALSE, log = TRUE,
center = TRUE)
# Run with one bootstrap sample for example
gap_stat &lt;- sparsedc_gap(pre_data[[1]], pre_data[[2]],
 min_clus &lt;- 2, max_clus &lt;- 3, nboots &lt;- 2)


</code></pre>

<hr>
<h2 id='update_c'>Update Clusters</h2><span id='topic+update_c'></span>

<h3>Description</h3>

<p>Updates the cluster membership for each iteration of SparseDC. Runs inside
<code>sparse_dc_fun</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_c(mu_1, mu_2, pdat1, pdat2, ncluster)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update_c_+3A_mu_1">mu_1</code></td>
<td>
<p>The center values for each cluster in condition 1.</p>
</td></tr>
<tr><td><code id="update_c_+3A_mu_2">mu_2</code></td>
<td>
<p>The center values for each cluster in condition 2.</p>
</td></tr>
<tr><td><code id="update_c_+3A_pdat1">pdat1</code></td>
<td>
<p>The centered data from condition 1, columns should be
samples (cells) and rows should be features (genes).</p>
</td></tr>
<tr><td><code id="update_c_+3A_pdat2">pdat2</code></td>
<td>
<p>The centered data from condition 2, columns should be
samples (cells) and rows should be features (genes). The number of genes
should be the same as <code>pdat1</code>.
as in pdat1.</p>
</td></tr>
<tr><td><code id="update_c_+3A_ncluster">ncluster</code></td>
<td>
<p>The number of clusters present in the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the cluster membership for condition 1 and
condition 2.
</p>

<hr>
<h2 id='update_mu'>Update the Center Values</h2><span id='topic+update_mu'></span>

<h3>Description</h3>

<p>This fucntion updates the center values for each cluster for each iteration
of SparseDC. This function runs inside <code>sparse_dc_fun</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_mu(C_1, C_2, pdat1, pdat2, ncluster, lambda1, lambda2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update_mu_+3A_c_1">C_1</code></td>
<td>
<p>The cluster membership for samples in condition 1</p>
</td></tr>
<tr><td><code id="update_mu_+3A_c_2">C_2</code></td>
<td>
<p>The cluster membership for samples in condition 2</p>
</td></tr>
<tr><td><code id="update_mu_+3A_pdat1">pdat1</code></td>
<td>
<p>The centered data from condition 1, columns should be
samples (cells) and rows should be features (genes).</p>
</td></tr>
<tr><td><code id="update_mu_+3A_pdat2">pdat2</code></td>
<td>
<p>The centered data from condition 2, columns should be
samples (cells) and rows should be features (genes). The number of genes
should be the same as <code>pdat1</code>.
as in pdat1.</p>
</td></tr>
<tr><td><code id="update_mu_+3A_ncluster">ncluster</code></td>
<td>
<p>The number of clusters present in the data.</p>
</td></tr>
<tr><td><code id="update_mu_+3A_lambda1">lambda1</code></td>
<td>
<p>The lambda 1 value to use in the SparseDC function. This
value controls the number of marker genes detected for each of the clusters
in the final result. This can be calculated using the &quot;lambda1_calculator&quot;
function or supplied by the user.</p>
</td></tr>
<tr><td><code id="update_mu_+3A_lambda2">lambda2</code></td>
<td>
<p>The lambda 2 value to use in the SparseDC function. This
value controls the number of genes that show condition-dependent
expression within each cell type. This can be calculated using the
&quot;lambda2_calculator&quot; function or supplied by the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list containing the center values for each of the
clusters in condition 1 and 2.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
