<!DOCTYPE html><html><head><title>Help for package SPAtest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SPAtest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Saddle_Prob'>
<p>Calculate Saddlepoint p values (for external libraries)</p></a></li>
<li><a href='#ScoreTest_SPA'>
<p>Score test based on saddlepoint approximation</p></a></li>
<li><a href='#ScoreTest_SPA_wMeta'>
<p>Prepare summary statistics for meta-analysis and perform SPA test.</p></a></li>
<li><a href='#ScoreTest_wSaddleApprox_NULL_Model'>
<p>Preparing the null model</p></a></li>
<li><a href='#SPAmeta'>
<p>Meta-analysis using SPA test summary statistics</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Score Test and Meta-Analysis Based on Saddlepoint Approximation</td>
</tr>
<tr>
<td>Version:</td>
<td>3.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-09-04</td>
</tr>
<tr>
<td>Author:</td>
<td>Rounak Dey, Seunggeun Lee</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rounak Dey &lt;deyrnk@umich.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs score test using saddlepoint approximation to estimate the null distribution. Also prepares summary statistics for meta-analysis and performs meta-analysis to combine multiple association results. For the latest version, please check <a href="https://github.com/leeshawn/SPAtest">https://github.com/leeshawn/SPAtest</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-09-04 12:41:13 UTC; Rounak</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-09-04 13:52:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='Saddle_Prob'>
Calculate Saddlepoint p values (for external libraries)
</h2><span id='topic+Saddle_Prob'></span><span id='topic+Saddle_Prob_fast'></span>

<h3>Description</h3>

<p>Function to calculate the SPA p value from score function. Intended to be used by external libraries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Saddle_Prob(q, mu, g, Cutoff=2,alpha,output="P",nodes.fixed,nodes.init,log.p=FALSE)
Saddle_Prob_fast(q, g,mu,gNA,gNB,muNA,muNB,Cutoff=2,alpha,output,nodes.fixed,
nodes.init,log.p=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Saddle_Prob_+3A_q">q</code></td>
<td>

<p>Numeric scalar, score statistic.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_mu">mu</code></td>
<td>

<p>Numeric vector, fitted probabilities from the null model.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_g">g</code></td>
<td>

<p>Numeric vector, covariate adjusted genotypes.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_gna">gNA</code></td>
<td>

<p>Numeric vector, covariate adjusted genotypes where the observed genotype is zero.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_gnb">gNB</code></td>
<td>

<p>Numeric vector, covariate adjusted genotypes where the observed genotype is non-zero.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_muna">muNA</code></td>
<td>

<p>Numeric vector, fitted probabilities from the null model where the observed genotype is zero.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_munb">muNB</code></td>
<td>

<p>Numeric vector, fitted probabilities from the null model where the observed genotype is non-zero.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_cutoff">Cutoff</code></td>
<td>

<p>An integer or the string &quot;BE&quot; denoting the standard deviation cutoff to be used. If <code>Cutoff</code> = &quot;BE&quot;, the level-specific cutoff based on Berry-Esseen theorem is calculated. If the test statistic lies within the standard deviation cutoff of the mean, p-value based on traditional score test is returned. Default value is 2.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_alpha">alpha</code></td>
<td>

<p>Significance level for the test(s), default value is <code class="reqn">5\times 10^{-8}</code>. Used only if <code>Cutoff</code> = &quot;BE&quot;.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_output">output</code></td>
<td>

<p>String specifying the output required. Possible values are &quot;<code>P</code>&quot; (default), &quot;<code>metaZ</code>&quot;, &quot;<code>metaGC</code>&quot;, and &quot;<code>metaspline</code>&quot;.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_nodes.fixed">nodes.fixed</code></td>
<td>

<p>Vector denoting the spline nodes for the spline based summary statistics, if you do not want to provide a fixed set of nodes instead of estimating the optimal set of nodes. Only applicable when the output is &quot;<code>metaspline</code>&quot;.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_nodes.init">nodes.init</code></td>
<td>

<p>Vector denoting the initial values of the spline nodes when you want to estimate the optimal set of spline nodes using the coordinate descent algorithm. Only applicable when the output is &quot;<code>metaspline</code>&quot;. Ignored if  <code>nodes.fixed</code> is provided. The node at 0 will be automatically added, no need to provide that. The number of finally selected nodes will be the same as in <code>nodes.init</code>.
</p>
</td></tr>
<tr><td><code id="Saddle_Prob_+3A_log.p">log.p</code></td>
<td>

<p>Whether to return natural log-transformed p-values, default value is <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>
<p>p-value or natural log-transformed p-value based on the saddlepoint approximation. If <code>output</code> = &quot;P&quot;, it is unsigned. For all other choice of <code>output</code>, it is signed.</p>
</td></tr>
<tr><td><code>p.value.NA</code></td>
<td>
<p>p-value or natural log-transformed p-value based on the normal approximation (traditional score test).  If <code>output</code> = &quot;P&quot;, it is unsigned. For all other choice of <code>output</code>, it is signed.</p>
</td></tr>
<tr><td><code>Is.converge</code></td>
<td>
<p>&quot;TRUE&quot; or &quot;FALSE&quot; denoting whether the root-finding algorithm for the saddlepoint equation has converged.</p>
</td></tr>
<tr><td><code>Score</code></td>
<td>
<p>Centered score statistic.</p>
</td></tr>
<tr><td><code>splfun</code></td>
<td>
<p>Outputs from fitting the CGF-Spline method, if <code>output</code> = &quot;metaspline&quot;.</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>Variance of the score test statistic, if <code>output</code> = &quot;metaspline&quot;.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ScoreTest_SPA">ScoreTest_SPA</a></code>
</p>
<p><code><a href="#topic+ScoreTest_SPA_wMeta">ScoreTest_SPA_wMeta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
Saddle_Prob(q, mu, g, Cutoff=2,alpha,output="P",nodes.fixed,nodes.init,log.p=FALSE)
Saddle_Prob_fast(q, g,mu,gNA,gNB,muNA,muNB,Cutoff=2,alpha,output,nodes.fixed,nodes.init,log.p=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ScoreTest_SPA'>
Score test based on saddlepoint approximation
</h2><span id='topic+ScoreTest_SPA'></span>

<h3>Description</h3>

<p>Performs score test using saddlepoint approximation to estimate the null distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScoreTest_SPA(genos,pheno,cov,obj.null,method=c("fastSPA","SPA"),minmac=5,
Cutoff=2,alpha=5*10^-8,missing.id=NA,beta.out=FALSE,beta.Cutoff=5*10^-7,log.p=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ScoreTest_SPA_+3A_genos">genos</code></td>
<td>

<p>A vector or matrix containing the genotypes or dosages. If matrix is provided then rows should correspond to SNPs and columns should correspond to subjects. Optional, but needed if <code>obj.null</code> is missing.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_pheno">pheno</code></td>
<td>

<p>A vector containing the outcomes (phenotypes). Optional, but needed if <code>obj.null</code> is missing.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_cov">cov</code></td>
<td>

<p>A matrix or data frame containing the covariates. Optional, but needed if <code>obj.null</code> is missing.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_obj.null">obj.null</code></td>
<td>

<p>An object of class &quot;<code>SA_NULL</code>&quot;. (Optional)
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_method">method</code></td>
<td>

<p>String specifying the p-value calculation method. Possible values are &quot;<code>fastSPA</code>&quot; (default) and &quot;<code>SPA</code>&quot;.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_minmac">minmac</code></td>
<td>

<p>Minimum minor allele count threshold to run SPA test, default value is <code class="reqn">5</code>.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_cutoff">Cutoff</code></td>
<td>

<p>An integer or the string &quot;BE&quot; denoting the standard deviation cutoff to be used. If <code>Cutoff</code> = &quot;BE&quot;, the level-specific cutoff based on Berry-Esseen theorem is calculated. If the test statistic lies within the standard deviation cutoff of the mean, p-value based on traditional score test is returned. Default value is 2.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_alpha">alpha</code></td>
<td>

<p>Significance level for the test(s), default value is <code class="reqn">5\times 10^{-8}</code>. Used only if <code>Cutoff</code> = &quot;BE&quot;.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_missing.id">missing.id</code></td>
<td>

<p>Missing value indicator. Numeric or <code>NA</code>, default value is <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_beta.out">beta.out</code></td>
<td>

<p>Logical indicating whether log odds ratios (beta parameters) are to be estimated, default value is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_beta.cutoff">beta.Cutoff</code></td>
<td>

<p>Maximum p-value threshold for beta parameters to be estimated, default value is <code class="reqn">5\times 10^{-7}</code>.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_+3A_log.p">log.p</code></td>
<td>

<p>Whether to return natural log-transformed p-values, default value is <code>FALSE</code>. NOTE: log-transformed p-values are not compatible with Meta-analysis using this package.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>genos</code> can have discrete <code>0, 1, 2</code> values or continuous values between <code>[0,2]</code>. The genotype or dosage values can represent any of the major allele, minor allele, reference allele or alternate allele counts (or dosages), as long as it is consistent throughout the subjects.
</p>
<p><code>genos</code> can have missing values denoted by the <code>missing.id</code> argument. Such missing values will be imputed using mean imputation. <code>pheno</code> or <code>cov</code> cannot have missing values. 
</p>
<p><code>pheno</code> and <code>cov</code> are ignored if <code>obj.null</code> is provided. If both <code>obj.null</code> and <code>cov</code> is missing, or <code>obj.null</code> is missing and <code>cov=NULL</code>, then the vector <code>rep(1,n)</code> is assigned to <code>cov</code>, where <code>n</code> is the number of subjects.
</p>
<p><code>method</code> = &quot;SPA&quot; is the basic saddlepoint approximation based test without the partially normal approximation improvement.
<code>method</code> = &quot;fastSPA&quot; utilizes the partially normal approximation approach for improved efficiency, especially for rare variants.
</p>
<p>Beta parameters are estimated using Firth's method, and thus computationally expensive. Therefore, it is recommended that beta parameters are only to be estimated when the p-value is very small (denoted by <code>beta.Cutoff</code>). The code for beta estimation is as implemented by Clement Ma in the EPACTS software.
</p>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>
<p>p-value or natural log-transformed p-value based on the saddlepoint approximation.</p>
</td></tr>
<tr><td><code>p.value.NA</code></td>
<td>
<p>p-value or natural log-transformed p-value based on the normal approximation (traditional score test).</p>
</td></tr>
<tr><td><code>Is.converge</code></td>
<td>
<p>&quot;TRUE&quot; or &quot;FALSE&quot; denoting whether the root-finding algorithm for the saddlepoint equation has converged.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Genotype log-odds ratio estimate.</p>
</td></tr>
<tr><td><code>SEbeta</code></td>
<td>
<p>Standard error for the genotype log-odds ratio.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Rounak Dey, <a href="mailto:deyrnk@umich.edu">deyrnk@umich.edu</a>
</p>


<h3>References</h3>

<p>Dey, R. et al., 2017. A Fast and Accurate Algorithm to Test for Binary Phenotypes and Its Application to PheWAS.
The American Journal of Human Genetics, Vol 101 (1), 37-49.
</p>
<p>Ma, C. et al., 2013. Recommended Joint and Meta-Analysis Strategies for Case-Control Association Testing of Single Low-Count Variants.
Genetic Epidemiology, Vol 37 (6), 539-550.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ScoreTest_wSaddleApprox_NULL_Model">ScoreTest_wSaddleApprox_NULL_Model</a></code>
</p>
<p><code><a href="#topic+ScoreTest_SPA_wMeta">ScoreTest_SPA_wMeta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ScoreTest_SPA(genos,pheno,cov,obj.null,method=c("fastSPA","SPA"),
minmac=5,Cutoff=2,alpha=5*10^-8,missing.id=NA,beta.out=FALSE,beta.Cutoff=5*10^-7,log.p=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ScoreTest_SPA_wMeta'>
Prepare summary statistics for meta-analysis and perform SPA test.
</h2><span id='topic+ScoreTest_SPA_wMeta'></span>

<h3>Description</h3>

<p>Performs score test using saddlepoint approximation and prepares summary statistics for meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScoreTest_SPA_wMeta(genos,pheno,cov,obj.null,method=c("fastSPA","SPA"),minmac=5,
Cutoff=2,alpha=5*10^-8,missing.id=NA,beta.out=FALSE,beta.Cutoff=5*10^-7,
output=c("P","metaZ","metaGC","metaspline"),nodes.fixed=NULL,
nodes.init=c(-100,-10,-1,1,10,100))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_genos">genos</code></td>
<td>

<p>A vector or matrix containing the genotypes or dosages. If matrix is provided then rows should correspond to SNPs and columns should correspond to subjects. Optional, but needed if <code>obj.null</code> is missing.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_pheno">pheno</code></td>
<td>

<p>A vector containing the outcomes (phenotypes). Optional, but needed if <code>obj.null</code> is missing.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_cov">cov</code></td>
<td>

<p>A matrix or data frame containing the covariates. Optional, but needed if <code>obj.null</code> is missing.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_obj.null">obj.null</code></td>
<td>

<p>An object of class &quot;<code>SA_NULL</code>&quot;. (Optional)
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_method">method</code></td>
<td>

<p>String specifying the p-value calculation method. Possible values are &quot;<code>fastSPA</code>&quot; (default) and &quot;<code>SPA</code>&quot;.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_minmac">minmac</code></td>
<td>

<p>Minimum minor allele count threshold to run SPA test, default value is <code class="reqn">5</code>.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_cutoff">Cutoff</code></td>
<td>

<p>An integer or the string &quot;BE&quot; denoting the standard deviation cutoff to be used. If <code>Cutoff</code> = &quot;BE&quot;, the level-specific cutoff based on Berry-Esseen theorem is calculated. If the test statistic lies within the standard deviation cutoff of the mean, p-value based on traditional score test is returned. Default value is 2.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_alpha">alpha</code></td>
<td>

<p>Significance level for the test(s), default value is <code class="reqn">5\times 10^{-8}</code>. Used only if <code>Cutoff</code> = &quot;BE&quot;.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_missing.id">missing.id</code></td>
<td>

<p>Missing value indicator. Numeric or <code>NA</code>, default value is <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_beta.out">beta.out</code></td>
<td>

<p>Logical indicating whether log odds ratios (beta parameters) are to be estimated, default value is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_beta.cutoff">beta.Cutoff</code></td>
<td>

<p>Maximum p-value threshold for beta parameters to be estimated, default value is <code class="reqn">5\times 10^{-7}</code>.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_output">output</code></td>
<td>

<p>String specifying the output required. Possible values are &quot;<code>P</code>&quot; (default), &quot;<code>metaZ</code>&quot;, &quot;<code>metaGC</code>&quot;, and &quot;<code>metaspline</code>&quot;.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_nodes.fixed">nodes.fixed</code></td>
<td>

<p>Vector denoting the spline nodes for the spline based summary statistics, if you do not want to provide a fixed set of nodes instead of estimating the optimal set of nodes. Only applicable when the output is &quot;<code>metaspline</code>&quot;.
</p>
</td></tr>
<tr><td><code id="ScoreTest_SPA_wMeta_+3A_nodes.init">nodes.init</code></td>
<td>

<p>Vector denoting the initial values of the spline nodes when you want to estimate the optimal set of spline nodes using the coordinate descent algorithm. Only applicable when the output is &quot;<code>metaspline</code>&quot;. Ignored if  <code>nodes.fixed</code> is provided. The node at 0 will be automatically added, no need to provide that. The number of finally selected nodes will be the same as in <code>nodes.init</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>genos</code> can have discrete <code>0, 1, 2</code> values or continuous values between <code>[0,2]</code>. The genotype or dosage values can represent any of the major allele, minor allele, reference allele or alternate allele counts (or dosages), as long as it is consistent throughout the subjects.
</p>
<p><code>genos</code> can have missing values denoted by the <code>missing.id</code> argument. Such missing values will be imputed using mean imputation. <code>pheno</code> or <code>cov</code> cannot have missing values. 
</p>
<p><code>pheno</code> and <code>cov</code> are ignored if <code>obj.null</code> is provided. If both <code>obj.null</code> and <code>cov</code> is missing, or <code>obj.null</code> is missing and <code>cov=NULL</code>, then the vector <code>rep(1,n)</code> is assigned to <code>cov</code>, where <code>n</code> is the number of subjects.
</p>
<p><code>method</code> = &quot;SPA&quot; is the basic saddlepoint approximation based test without the partially normal approximation improvement.
<code>method</code> = &quot;fastSPA&quot; utilizes the partially normal approximation approach for improved efficiency, especially for rare variants.
</p>
<p>Beta parameters are estimated using Firth's method, and thus computationally expensive. Therefore, it is recommended that beta parameters are only to be estimated when the p-value is very small (denoted by <code>beta.Cutoff</code>). The code for beta estimation is as implemented by Clement Ma in the EPACTS software.
</p>
<p>If <code>output</code> = &quot;P&quot;, the output will be the same as running <code><a href="#topic+ScoreTest_SPA">ScoreTest_SPA</a></code>. No summary statistics for meta-analysis is provided.
</p>
<p>If <code>output</code> = &quot;metaZ&quot;, the output will be the same as with <code>output</code> = &quot;P&quot;, except the p values in the output will be signed, and minor allele frequencies are also provided. This choice of <code>output</code> provide required summary statistics for Z score-based meta-analysis.
</p>
<p>If <code>output</code> = &quot;metaGC&quot;, the output will be the same as with <code>output</code> = &quot;P&quot;, except the p values in the output will be signed, and genotype counts of homozygous minor and heterozygous genotypes are also provided.  This choice of <code>output</code> provide required summary statistics for genotype count-based meta-analysis.
</p>
<p>If <code>output</code> = &quot;metaspline&quot;, the output will be the same as with <code>output</code> = &quot;metaGC&quot;, additionally spline-based summary statistics are also provided.  This choice of <code>output</code> provide required summary statistics for spline-based as well as genotype count-based meta-analysis.
</p>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>
<p>p-value based on the saddlepoint approximation. If <code>output</code> = &quot;P&quot;, it is unsigned. For all other choice of <code>output</code>, it is signed.</p>
</td></tr>
<tr><td><code>p.value.NA</code></td>
<td>
<p>p-value based on the normal approximation (traditional score test).  If <code>output</code> = &quot;P&quot;, it is unsigned. For all other choice of <code>output</code>, it is signed.</p>
</td></tr>
<tr><td><code>Is.converge</code></td>
<td>
<p>&quot;TRUE&quot; or &quot;FALSE&quot; denoting whether the root-finding algorithm for the saddlepoint equation has converged.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Genotype log-odds ratio estimate.</p>
</td></tr>
<tr><td><code>SEbeta</code></td>
<td>
<p>Standard error for the genotype log-odds ratio.</p>
</td></tr>
<tr><td><code>MAF</code></td>
<td>
<p>Minor allele frequencies. Only provided when <code>output</code> = &quot;metaZ&quot;.</p>
</td></tr>
<tr><td><code>GCmat</code></td>
<td>
<p>Genotype counts for homozygous minor (column 1) and heterozygous genotypes (column 2). Only provided when <code>output</code> = &quot;metaGC&quot; or &quot;metaspline&quot;.</p>
</td></tr>
<tr><td><code>spldata</code></td>
<td>
<p>Spline-based summary statistics on the CGF. Column 1 represents the raw score values, column 2 the variance of those scores, and the next columns represent nodes, first and second derivatives of the CGF in equal sizes. Only provided when <code>output</code> = &quot;metaspline&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Rounak Dey, <a href="mailto:deyrnk@umich.edu">deyrnk@umich.edu</a>
</p>


<h3>References</h3>

<p>Dey, R. et al., 2017. A Fast and Accurate Algorithm to Test for Binary Phenotypes and Its Application to PheWAS.
The American Journal of Human Genetics, Vol 101 (1), 37-49.
</p>
<p>Ma, C. et al., 2013. Recommended Joint and Meta-Analysis Strategies for Case-Control Association Testing of Single Low-Count Variants.
Genetic Epidemiology, Vol 37 (6), 539-550.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ScoreTest_wSaddleApprox_NULL_Model">ScoreTest_wSaddleApprox_NULL_Model</a></code>
</p>
<p><code><a href="#topic+ScoreTest_SPA">ScoreTest_SPA</a></code>
</p>
<p><code><a href="#topic+SPAmeta">SPAmeta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ScoreTest_SPA_wMeta(genos,pheno,cov,obj.null,method=c("fastSPA","SPA"),
minmac=5,Cutoff=2,alpha=5*10^-8,missing.id=NA,beta.out=FALSE,beta.Cutoff=5*10^-7,
output=c("P","metaZ","metaGC","metaspline"),nodes.fixed=NULL,
nodes.init=c(-100,-10,-1,1,10,100))

## End(Not run)
</code></pre>

<hr>
<h2 id='ScoreTest_wSaddleApprox_NULL_Model'>
Preparing the null model
</h2><span id='topic+ScoreTest_wSaddleApprox_NULL_Model'></span>

<h3>Description</h3>

<p>Prepares the null model object <code>SA_NULL</code> to be used in <code>ScoreTest_SPA</code> based on the outcome and the covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ScoreTest_wSaddleApprox_NULL_Model(formula, data=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ScoreTest_wSaddleApprox_NULL_Model_+3A_formula">formula</code></td>
<td>

<p>An object of class &quot;formula&quot; (as used in the function &quot;glm&quot;).
</p>
</td></tr>
<tr><td><code id="ScoreTest_wSaddleApprox_NULL_Model_+3A_data">data</code></td>
<td>

<p>An optional data frame, list or environment containing the variables in the model.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>ScoreTest_wSaddleApprox_NULL_Model</code> returns an object of class <code>SA_NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Rounak Dey, <a href="mailto:deyrnk@umich.edu">deyrnk@umich.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ScoreTest_SPA">ScoreTest_SPA</a></code>
</p>
<p><code><a href="#topic+ScoreTest_SPA_wMeta">ScoreTest_SPA_wMeta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ScoreTest_wSaddleApprox_NULL_Model(formula, data=NULL)

## End(Not run)
</code></pre>

<hr>
<h2 id='SPAmeta'>
Meta-analysis using SPA test summary statistics
</h2><span id='topic+SPAmeta'></span>

<h3>Description</h3>

<p>Performs meta-analysis using summary statistics obtained from SPA test within individual studies. It can be used to hybridize different meta-analysis techniques based on the available summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SPAmeta(pvalue.Z=NULL,MAF.Z=NULL,CCsize.Z=NULL,
	pvalue.GC=NULL,GCmat=NULL,CCsize.GC=NULL,Cutoff.GC=2,
	spldata=NULL,
	Cutoff.meta=2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SPAmeta_+3A_pvalue.z">pvalue.Z</code></td>
<td>

<p>Vector denoting signed p values for which no other summary statistic is available and Z score conversion is suited.
</p>
</td></tr>
<tr><td><code id="SPAmeta_+3A_maf.z">MAF.Z</code></td>
<td>

<p>Vector denoting the minor allele frequencies in different studies corresponding to <code>pvalue.Z</code>. If missing, the different studies will be assumed to have no heterogeneity in the variants. Optional, if only Z score-based meta-analysis is being performed. Required, if being hybridized with other meta-analysis methods.
</p>
</td></tr>
<tr><td><code id="SPAmeta_+3A_ccsize.z">CCsize.Z</code></td>
<td>

<p>Matrix denoting the case-control sample sizes of the studies corresponding to <code>pvalue.Z</code>. Column 1 represents cases, column 2 controls.
</p>
</td></tr>
<tr><td><code id="SPAmeta_+3A_pvalue.gc">pvalue.GC</code></td>
<td>

<p>Vector denoting signed p values for which genotype counts of the homozygous minor and heterozygous genotypes are available and the p values are calculated using the SPA test. These p values will be used in the meta-analysis using the genotype count-based meta-analysis strategy.
</p>
</td></tr>
<tr><td><code id="SPAmeta_+3A_gcmat">GCmat</code></td>
<td>

<p>Matrix denoting the genotype counts, column 1 represents the homozygous minor and column 2 the heterozygous genotypes.
</p>
</td></tr>
<tr><td><code id="SPAmeta_+3A_ccsize.gc">CCsize.GC</code></td>
<td>

<p>Matrix denoting the case-control sample sizes of the studies corresponding to <code>pvalue.GC</code>. Column 1 represents cases, column 2 controls.
</p>
</td></tr>
<tr><td><code id="SPAmeta_+3A_cutoff.gc">Cutoff.GC</code></td>
<td>

<p>An integer or vector specifying the Cutoffs used to calculate the SPA test p values within individual studies.
</p>
</td></tr>
<tr><td><code id="SPAmeta_+3A_spldata">spldata</code></td>
<td>

<p>Vector, matrix or a list denoting the spline-based summary statistics as obtained from running <code><a href="#topic+ScoreTest_SPA_wMeta">ScoreTest_SPA_wMeta</a></code> with <code>output</code> = &quot;metaspline&quot;. Provide only for the studies with this information available.
</p>
</td></tr>
<tr><td><code id="SPAmeta_+3A_cutoff.meta">Cutoff.meta</code></td>
<td>

<p>An integer denoting the fixed standard deviation cutoff to be used for the combined meta-analysis test. Default value is 2.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If all studies with genotype count information used the same cutoff, then providing only a single number as <code>Cutoff.GC</code> is sufficient. Otherwise, a vector denoting the cutoff within each study is required. Currently, &quot;BE&quot; as a within-study cutoff is not accepted.
</p>
<p>If provided as a matrix, the rows of <code>spldata</code> should correspond to the individual studies for which spline-based meta-analysis is to be applied. Column 1 should denote the raw within-study scores, column 2 the corresponding variances. Out of the next 3k columns, first k columns should denote the nodes, next k columns the functional values of the first derivative of the CGF, and the last k columns the functional values of the second derivative of the CGF. None of the elements of the matrix can be missing or NA. If different number of nodes were used in different studies, please provide them in list format.
If spline-based summary statistic is available for only one study, then it can also be provided as a vector with 3k+2 elements where the elements denote similar statistics as denoted above in the case of a matrix.
If provided as a list, the elements of the list correspond to the different studies. Each element is a vector which denote the score, variance, nodes, and first and second derivative values in this order for that particular study. The list type input is more suited when the number of nodes in different studies are different.
</p>
<p>If only Z score-based meta-analysis is needed, provide only <code>pvalue.Z</code>, <code>MAF.Z</code>, and <code>CCsize.Z</code>. <code>MAF.Z</code> is used to properly weight the Z scores when there is between study heterogeneity present. It is optional to provide the <code>MAF.Z</code> information if only Z score-based meta-analysis is being performed. Then the studies will be assumed to be homogeneous if <code>MAF.Z</code> is missing. However, <code>MAF.Z</code> is required if the Z score-based meta-analysis is being hybridized with other meta-analysis methods.
If only genotype count-based meta-analysis is needed, provide only <code>pvalue.GC</code>,<code>GCmat</code>,<code>CCsize.GC</code>, and <code>Cutoff.GC</code>.
If only spline-based meta-analysis is needed, provide only <code>spldata</code>.
If different studies provide different types of summary information, hybridize the methods accordingly.
</p>
<p>For now, this package does not support having different minor alleles in different studies. If the minor alleles are different in different studies, some manual pre-processing is required, such as changing the signs of the p values, adjusting the MAF and genotype counts accordingly, and changing the signs of the scores, nodes and first derivatives in the spline-based summary statistics. In later implementations, we will try to provide a function for doing that pre-processing.
</p>


<h3>Value</h3>

<p>The signed meta-analysis p value is returned.
</p>


<h3>Author(s)</h3>

<p>Rounak Dey, <a href="mailto:deyrnk@umich.edu">deyrnk@umich.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ScoreTest_wSaddleApprox_NULL_Model">ScoreTest_wSaddleApprox_NULL_Model</a></code>
</p>
<p><code><a href="#topic+ScoreTest_SPA_wMeta">ScoreTest_SPA_wMeta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
SPAmeta(pvalue.Z=NULL,MAF.Z=NULL,CCsize.Z=NULL,
	pvalue.GC=NULL,GCmat=NULL,CCsize.GC=NULL,Cutoff.GC=2,
	spldata=NULL,
	Cutoff.meta=2)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
