<!DOCTYPE html><html><head><title>Help for package SPCompute</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SPCompute}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#check_parameters'><p>Check if the parameter list contains all the parameters required for the computation.</p></a></li>
<li><a href='#Compute_Power'><p>Compute the Power of an association study, at a given sample size.</p></a></li>
<li><a href='#Compute_Power_multi'><p>Compute the Power of an association study at a given sample size, accommodating more than one covariates, using the Semi-Simulation method.</p></a></li>
<li><a href='#Compute_Size'><p>Compute the sample size of an association study, to achieve a target power.</p></a></li>
<li><a href='#Compute_Size_multi'><p>Compute the sample size of an association study to achieve a target power for multiple E's, using semi-sim.</p></a></li>
<li><a href='#convert_preva_to_intercept'><p>Convert the prevalence value to the intercept value beta0.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Compute Power or Sample Size for GWAS with Covariate Effect</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Ziang Zhang, Lei Sun</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ziang Zhang &lt;aguero.zhang@mail.utoronto.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fast computation of the required sample size or the
    achieved power, for GWAS studies with different types of covariate
    effects and different types of covariate-gene dependency structure.
    For the detailed description of the methodology, see Zhang (2022)
    "Power and Sample Size Computation for Genetic Association Studies
     of Binary Traits: Accounting for Covariate Effects" &lt;<a href="https://doi.org/10.48550/arXiv.2203.15641">doi:10.48550/arXiv.2203.15641</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-24 17:20:28 UTC; ziangzhang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-24 17:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='check_parameters'>Check if the parameter list contains all the parameters required for the computation.</h2><span id='topic+check_parameters'></span>

<h3>Description</h3>

<p>Check if the parameter list contains all the parameters required for the computation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_parameters(parameters, response, covariate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_parameters_+3A_parameters">parameters</code></td>
<td>
<p>A list of parameters that contains all the required parameters in the model. If response is &quot;binary&quot;, this list needs to contain &quot;preva&quot; which denotes the prevalence of the disease (or case to control ratio for case-control sampling). If response is continuous, the list needs to contain &quot;TraitSD&quot; and &quot;TraitMean&quot; which represent the standard deviation and mean of the continuous trait.
#' If covariate is not &quot;none&quot;, a parameter &quot;gammaG&quot; needs to be defined to capture the dependence between the SNP and the covariate (through linear regression model if covariate is continuous, and logistic model if covariate is binary). If covariate is &quot;binary&quot;, list needs to contains &quot;pE&quot; that defines the frequency of the covariate. If it is continuous, list needs to contain &quot;muE&quot; and &quot;sigmaE&quot; to define
#' its mean and standard deviation. The MAF is defined as &quot;pG&quot;, with HWE assumed to hold.</p>
</td></tr>
<tr><td><code id="check_parameters_+3A_response">response</code></td>
<td>
<p>A string of either &quot;binary&quot; or &quot;continuous&quot;, indicating the type of response/trait variable in the model.</p>
</td></tr>
<tr><td><code id="check_parameters_+3A_covariate">covariate</code></td>
<td>
<p>A string of either &quot;binary&quot;, &quot;continuous&quot; or &quot;none&quot; indicating the type of covariate E in the model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE or FALSE if all the parameters are correctly defined.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>parameters &lt;- list(TraitMean = 0.3, TraitSD = 1, pG = 0.2, betaG = log(1.1),
betaE = log(1.1), muE = 0, sigmaE = 3, gammaG = log(2.1))

SPCompute::check_parameters(parameters, "continuous", "continuous")
</code></pre>

<hr>
<h2 id='Compute_Power'>Compute the Power of an association study, at a given sample size.</h2><span id='topic+Compute_Power'></span>

<h3>Description</h3>

<p>Compute the Power of an association study, at a given sample size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compute_Power(
  parameters,
  n,
  response = "binary",
  covariate = "binary",
  mode = "additive",
  alpha = 0.05,
  seed = 123,
  LargePowerApproxi = FALSE,
  searchSizeGamma0 = 100,
  searchSizeBeta0 = 100,
  B = 10000,
  method = "semi-sim"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compute_Power_+3A_parameters">parameters</code></td>
<td>
<p>A list of parameters that contains all the required parameters in the model. If response is &quot;binary&quot;, this list needs to contain &quot;prev&quot; which denotes the prevalence of the disease (or case to control ratio for case-control sampling). If response is continuous, the list needs to contain &quot;traitSD&quot; and &quot;traitMean&quot; which represent the standard deviation and mean of the continuous trait.
If covariate is not &quot;none&quot;, a parameter &quot;gammaG&quot; needs to be defined to capture the dependence between the SNP and the covariate (through linear regression model if covariate is continuous, and logistic model if covariate is binary). If covariate is &quot;binary&quot;, list needs to contains &quot;pE&quot; that defines the frequency of the covariate. If it is continuous, list needs to contain &quot;muE&quot; and &quot;sigmaE&quot; to define
its mean and standard deviation. The MAF is defined as &quot;pG&quot;, with HWE assumed to hold.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_n">n</code></td>
<td>
<p>An integer number that indicates the sample size.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_response">response</code></td>
<td>
<p>A string of either &quot;binary&quot; or &quot;continuous&quot;, indicating the type of response/trait variable in the model, by default is &quot;binary&quot;</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_covariate">covariate</code></td>
<td>
<p>A string of either &quot;binary&quot;, &quot;continuous&quot; or &quot;none&quot; indicating the type of covariate E in the model, by default is &quot;binary&quot;.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_mode">mode</code></td>
<td>
<p>A string of either &quot;additive&quot;, &quot;dominant&quot; or &quot;recessive&quot;, indicating the genetic mode, by default is &quot;additive&quot;.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value that denotes the significance level used in the study, by default is 0.05.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_seed">seed</code></td>
<td>
<p>An integer number that indicates the seed used for the simulation to compute the approximate fisher information matrix, by default is 123.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_largepowerapproxi">LargePowerApproxi</code></td>
<td>
<p>TRUE or FALSE indicates whether to use the large power approximation formula.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_searchsizegamma0">searchSizeGamma0</code></td>
<td>
<p>The interval radius for the numerical search of gamma0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_searchsizebeta0">searchSizeBeta0</code></td>
<td>
<p>The interval radius for the numerical search of beta0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_b">B</code></td>
<td>
<p>An integer number that indicates the number of simulated sample to approximate the fisher information matrix, by default is 10000 (Should only be changed if computation uses semi-simulation method).</p>
</td></tr>
<tr><td><code id="Compute_Power_+3A_method">method</code></td>
<td>
<p>An character that is either &quot;semi-sim&quot; (default) or &quot;expand&quot; using the idea of representative dataset. This specifies the method being used to compute the power/sample size when the trait is binary using logistic regression. The default method will be faster for large sample size computation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The power that can be achieved at the given sample size.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>parameters &lt;- list(TraitMean = 0.3, TraitSD = 1, pG = 0.2, betaG = log(1.1),
betaE = log(1.1), muE = 0, sigmaE = 3, gammaG = log(2.1))

Compute_Power(parameters, n = 1000, response = "continuous",
covariate = "continuous", method = "semi-sim")
</code></pre>

<hr>
<h2 id='Compute_Power_multi'>Compute the Power of an association study at a given sample size, accommodating more than one covariates, using the Semi-Simulation method.</h2><span id='topic+Compute_Power_multi'></span>

<h3>Description</h3>

<p>Compute the Power of an association study at a given sample size, accommodating more than one covariates, using the Semi-Simulation method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compute_Power_multi(
  parameters,
  n,
  response = "binary",
  covariate,
  mode = "additive",
  alpha = 0.05,
  seed = 123,
  searchSizeBeta0 = 8,
  searchSizeGamma0 = 8,
  LargePowerApproxi = FALSE,
  B = 10000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compute_Power_multi_+3A_parameters">parameters</code></td>
<td>
<p>A list of parameters that contains all the required parameters in the model. If response is &quot;binary&quot;, this list needs to contain &quot;prev&quot; which denotes the prevalence of the disease (or case to control ratio for case-control sampling). If response is continuous, the list needs to contain &quot;traitSD&quot; and &quot;traitMean&quot; which represent the standard deviation and mean of the continuous trait.
If covariate is not &quot;none&quot;, a parameter &quot;gammaG&quot; needs to be defined to capture the dependence between the SNP and the covariate (through linear regression model if covariate is continuous, and logistic model if covariate is binary). If covariate is &quot;binary&quot;, list needs to contains &quot;pE&quot; that defines the frequency of the covariate. If it is continuous, list needs to contain &quot;muE&quot; and &quot;sigmaE&quot; to define
its mean and standard deviation. The MAF is defined as &quot;pG&quot;, with HWE assumed to hold.
Without specifying the parameter &quot;gammaE&quot;, by default it is assumed the two covariates are conditionally independent given G. The parameter &quot;gammaE&quot; when specified, should be interpreted as the regression coefficient of the first covariate when regressing the second covariate on it conditional on the SNP G.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_n">n</code></td>
<td>
<p>An integer number that indicates the sample size.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_response">response</code></td>
<td>
<p>A string of either &quot;binary&quot; or &quot;continuous&quot;, indicating the type of response/trait variable in the model, by default is &quot;binary&quot;</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_covariate">covariate</code></td>
<td>
<p>A vector of length two with elements being either c(&quot;binary&quot;, &quot;continuous&quot;),c(&quot;binary&quot;, &quot;binary&quot;) or c(&quot;continuous&quot;, &quot;continuous&quot;),  indicating the type of covariate E in the model.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_mode">mode</code></td>
<td>
<p>A string of either &quot;additive&quot;, &quot;dominant&quot; or &quot;recessive&quot;, indicating the genetic mode, by default is &quot;additive&quot;.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value that denotes the significance level used in the study, by default is 0.05.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_seed">seed</code></td>
<td>
<p>An integer number that indicates the seed used for the simulation to compute the approximate fisher information matrix, by default is 123.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_searchsizebeta0">searchSizeBeta0</code></td>
<td>
<p>The interval radius for the numerical search of beta0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_searchsizegamma0">searchSizeGamma0</code></td>
<td>
<p>The interval radius for the numerical search of gamma0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_largepowerapproxi">LargePowerApproxi</code></td>
<td>
<p>TRUE or FALSE indicates whether to use the large power approximation formula.</p>
</td></tr>
<tr><td><code id="Compute_Power_multi_+3A_b">B</code></td>
<td>
<p>An integer number that indicates the number of simulated sample to approximate the fisher information matrix, by default is 10000 (Should only be changed if computation uses semi-simulation method).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The power that can be achieved at the given sample size.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>parameters &lt;- list(TraitMean = 0.3, TraitSD = 1, pG = 0.2, betaG = log(1.1),
betaE = c(log(1.1), log(1.2)),
muE = 0, sigmaE = 3, gammaG = c(log(2.1), log(2.2)), pE = 0.4)
SPCompute::Compute_Power_multi(parameters, n = 1000, response = "continuous",
covariate = c("binary", "continuous"))
</code></pre>

<hr>
<h2 id='Compute_Size'>Compute the sample size of an association study, to achieve a target power.</h2><span id='topic+Compute_Size'></span>

<h3>Description</h3>

<p>Compute the sample size of an association study, to achieve a target power.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compute_Size(
  parameters,
  PowerAim,
  response = "binary",
  covariate = "binary",
  mode = "additive",
  alpha = 0.05,
  seed = 123,
  LargePowerApproxi = FALSE,
  searchSizeGamma0 = 100,
  searchSizeBeta0 = 100,
  B = 10000,
  method = "semi-sim",
  lower.lim.n = 1000,
  upper.lim.n = 8e+05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compute_Size_+3A_parameters">parameters</code></td>
<td>
<p>A list of parameters that contains all the required parameters in the model. If response is &quot;binary&quot;, this list needs to contain &quot;prev&quot; which denotes the prevalence of the disease (or case to control ratio for case-control sampling). If response is continuous, the list needs to contain &quot;traitSD&quot; and &quot;traitMean&quot; which represent the standard deviation and mean of the continuous trait.
If covariate is not &quot;none&quot;, a parameter &quot;gammaG&quot; needs to be defined to capture the dependence between the SNP and the covariate (through linear regression model if covariate is continuous, and logistic model if covariate is binary). If covariate is &quot;binary&quot;, list needs to contains &quot;pE&quot; that defines the frequency of the covariate. If it is continuous, list needs to contain &quot;muE&quot; and &quot;sigmaE&quot; to define
its mean and standard deviation. The MAF is defined as &quot;pG&quot;, with HWE assumed to hold.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_poweraim">PowerAim</code></td>
<td>
<p>An numeric value between 0 and 1 that indicates the aimed power of the study.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_response">response</code></td>
<td>
<p>A string of either &quot;binary&quot; or &quot;continuous&quot;, indicating the type of response/trait variable in the model, by default is &quot;binary&quot;</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_covariate">covariate</code></td>
<td>
<p>A string of either &quot;binary&quot;, &quot;continuous&quot; or &quot;none&quot; indicating the type of covariate E in the model, by default is &quot;binary&quot;.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_mode">mode</code></td>
<td>
<p>A string of either &quot;additive&quot;, &quot;dominant&quot; or &quot;recessive&quot;, indicating the genetic mode, by default is &quot;additive&quot;.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value that denotes the significance level used in the study, by default is 0.05.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_seed">seed</code></td>
<td>
<p>An integer number that indicates the seed used for the simulation to compute the approximate fisher information matrix, by default is 123.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_largepowerapproxi">LargePowerApproxi</code></td>
<td>
<p>TRUE or FALSE indicates whether to use the large power approximation formula.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_searchsizegamma0">searchSizeGamma0</code></td>
<td>
<p>The interval radius for the numerical search of gamma0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_searchsizebeta0">searchSizeBeta0</code></td>
<td>
<p>The interval radius for the numerical search of beta0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_b">B</code></td>
<td>
<p>An integer number that indicates the number of simulated sample to approximate the fisher information matrix, by default is 10000 (Should only be changed if computation uses semi-simulation method).</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_method">method</code></td>
<td>
<p>An character that is either &quot;semi-sim&quot; (default) or &quot;expand&quot; using the idea of representative dataset. This specifies the method being used to compute the power/sample size when the trait is binary using logistic regression. The default method will be faster for large sample size computation.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_lower.lim.n">lower.lim.n</code></td>
<td>
<p>An integer number that indicates the smallest sample size to be considered, only for &quot;expand&quot; method.</p>
</td></tr>
<tr><td><code id="Compute_Size_+3A_upper.lim.n">upper.lim.n</code></td>
<td>
<p>An integer number that indicates the largest sample size to be considered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The required sample size.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>parameters &lt;- list(TraitMean = 0.3, TraitSD = 1, pG = 0.2, betaG = log(1.1),
betaE = log(1.1), muE = 0, sigmaE = 3, gammaG = log(2.1))

Compute_Size(parameters, PowerAim = 0.8, response = "continuous",
covariate = "continuous", method = "semi-sim")
</code></pre>

<hr>
<h2 id='Compute_Size_multi'>Compute the sample size of an association study to achieve a target power for multiple E's, using semi-sim.</h2><span id='topic+Compute_Size_multi'></span>

<h3>Description</h3>

<p>Compute the sample size of an association study to achieve a target power for multiple E's, using semi-sim.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compute_Size_multi(
  parameters,
  PowerAim,
  response = "binary",
  covariate,
  mode = "additive",
  alpha = 0.05,
  seed = 123,
  LargePowerApproxi = FALSE,
  searchSizeGamma0 = 100,
  searchSizeBeta0 = 100,
  B = 10000,
  upper.lim.n = 8e+05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compute_Size_multi_+3A_parameters">parameters</code></td>
<td>
<p>A list of parameters that contains all the required parameters in the model. If response is &quot;binary&quot;, this list needs to contain &quot;prev&quot; which denotes the prevalence of the disease (or case to control ratio for case-control sampling). If response is continuous, the list needs to contain &quot;traitSD&quot; and &quot;traitMean&quot; which represent the standard deviation and mean of the continuous trait.
If covariate is not &quot;none&quot;, a parameter &quot;gammaG&quot; needs to be defined to capture the dependence between the SNP and the covariate (through linear regression model if covariate is continuous, and logistic model if covariate is binary). If covariate is &quot;binary&quot;, list needs to contains &quot;pE&quot; that defines the frequency of the covariate. If it is continuous, list needs to contain &quot;muE&quot; and &quot;sigmaE&quot; to define
its mean and standard deviation. The MAF is defined as &quot;pG&quot;, with HWE assumed to hold.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_poweraim">PowerAim</code></td>
<td>
<p>An numeric value between 0 and 1 that indicates the aimed power of the study.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_response">response</code></td>
<td>
<p>A string of either &quot;binary&quot; or &quot;continuous&quot;, indicating the type of response/trait variable in the model, by default is &quot;binary&quot;</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_covariate">covariate</code></td>
<td>
<p>Same as in Compute_Power_multi.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_mode">mode</code></td>
<td>
<p>A string of either &quot;additive&quot;, &quot;dominant&quot; or &quot;recessive&quot;, indicating the genetic mode, by default is &quot;additive&quot;.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_alpha">alpha</code></td>
<td>
<p>A numeric value that denotes the significance level used in the study, by default is 0.05.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_seed">seed</code></td>
<td>
<p>An integer number that indicates the seed used for the simulation to compute the approximate fisher information matrix, by default is 123.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_largepowerapproxi">LargePowerApproxi</code></td>
<td>
<p>TRUE or FALSE indicates whether to use the large power approximation formula.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_searchsizegamma0">searchSizeGamma0</code></td>
<td>
<p>The interval radius for the numerical search of gamma0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_searchsizebeta0">searchSizeBeta0</code></td>
<td>
<p>The interval radius for the numerical search of beta0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_b">B</code></td>
<td>
<p>An integer number that indicates the number of simulated sample to approximate the fisher information matrix, by default is 10000 (Should only be changed if computation uses semi-simulation method).</p>
</td></tr>
<tr><td><code id="Compute_Size_multi_+3A_upper.lim.n">upper.lim.n</code></td>
<td>
<p>An integer number that indicates the largest sample size to be considered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The required sample size.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>parameters &lt;- list(TraitMean = 0.3, TraitSD = 1, pG = 0.2,
betaG = log(1.1), betaE = c(log(1.1), log(1.2)),
muE = 0, sigmaE = 3, gammaG = c(log(2.1), log(2.2)), pE = 0.4)
SPCompute::Compute_Size_multi(parameters, PowerAim = 0.8,
response = "continuous", covariate = c("binary", "continuous"))
</code></pre>

<hr>
<h2 id='convert_preva_to_intercept'>Convert the prevalence value to the intercept value beta0.</h2><span id='topic+convert_preva_to_intercept'></span>

<h3>Description</h3>

<p>Convert the prevalence value to the intercept value beta0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_preva_to_intercept(
  parameters,
  mode = "additive",
  covariate = "binary",
  seed = 123,
  B = 10000,
  searchSizeBeta0 = 8,
  searchSizeGamma0 = 8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_preva_to_intercept_+3A_parameters">parameters</code></td>
<td>
<p>A list of parameters that contains all the required parameters in the model. If response is &quot;binary&quot;, this list needs to contain &quot;prev&quot; which denotes the prevalence of the disease (or case to control ratio for case-control sampling). If response is continuous, the list needs to contain &quot;traitSD&quot; and &quot;traitMean&quot; which represent the standard deviation and mean of the continuous trait.
If covariate is not &quot;none&quot;, a parameter &quot;gammaG&quot; needs to be defined to capture the dependence between the SNP and the covariate (through linear regression model if covariate is continuous, and logistic model if covariate is binary). If covariate is &quot;binary&quot;, list needs to contains &quot;pE&quot; that defines the frequency of the covariate. If it is continuous, list needs to contain &quot;muE&quot; and &quot;sigmaE&quot; to define
its mean and standard deviation. The MAF is defined as &quot;pG&quot;, with HWE assumed to hold.</p>
</td></tr>
<tr><td><code id="convert_preva_to_intercept_+3A_mode">mode</code></td>
<td>
<p>A string of either &quot;additive&quot;, &quot;dominant&quot; or &quot;recessive&quot;, indicating the genetic mode, by default is &quot;additive&quot;.</p>
</td></tr>
<tr><td><code id="convert_preva_to_intercept_+3A_covariate">covariate</code></td>
<td>
<p>A string of either &quot;binary&quot;, &quot;continuous&quot; or &quot;none&quot; indicating the type of covariate E in the model, by default is &quot;binary&quot;.</p>
</td></tr>
<tr><td><code id="convert_preva_to_intercept_+3A_seed">seed</code></td>
<td>
<p>An integer number that indicates the seed used for the simulation if needed, by default is 123.</p>
</td></tr>
<tr><td><code id="convert_preva_to_intercept_+3A_b">B</code></td>
<td>
<p>An integer number that indicates the number of simulated sample to use if needed, by default is 10000.</p>
</td></tr>
<tr><td><code id="convert_preva_to_intercept_+3A_searchsizebeta0">searchSizeBeta0</code></td>
<td>
<p>The interval radius for the numerical search of beta0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
<tr><td><code id="convert_preva_to_intercept_+3A_searchsizegamma0">searchSizeGamma0</code></td>
<td>
<p>The interval radius for the numerical search of gamma0, by default is 8. Setting to higher values may solve some numerical problems at the cost of longer runtime.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The corresponding gamma0, beta0 and residual variance of E (if applicable).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>convert_preva_to_intercept(parameters = list(preva = 0.2, betaG = 0.6, betaE = c(0.9),
gammaG = c(0.2), muE = c(0), sigmaE = c(1), pG = 0.3), covariate = "continuous")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
