<!DOCTYPE html><html><head><title>Help for package speedytax</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {speedytax}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#speedytax-package'><p>speedytax: Rapidly Import Classifier Results into 'phyloseq'</p></a></li>
<li><a href='#import_qiime2_tax_table'><p>Import QIIME2 Classification Table</p></a></li>
<li><a href='#import_rdp_tax_table'><p>Import RDP Classifier Taxonomy Table</p></a></li>
<li><a href='#import_sintax_tax_table'><p>Import SINTAX Taxonomy Table</p></a></li>
<li><a href='#read_tax_example'><p>Get path to example taxonomy files</p></a></li>
<li><a href='#simple_cap'><p>Simple Capitalization</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Rapidly Import Classifier Results into 'phyloseq'</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Import classification results from the 'RDP Classifier' (Ribosomal Database Project),' 'USEARCH sintax,' 'vsearch sintax' and the 'QIIME2' (Quantitative Insights into Microbial Ecology) classifiers into 'phyloseq' tax_table objects.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, phyloseq, Rcpp, stringr, tibble, tidyr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-14 00:37:49 UTC; johnq</td>
</tr>
<tr>
<td>Author:</td>
<td>John Quensen <a href="https://orcid.org/0000-0002-7669-1595"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John Quensen &lt;quensenj@msu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-14 02:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='speedytax-package'>speedytax: Rapidly Import Classifier Results into 'phyloseq'</h2><span id='topic+speedytax'></span><span id='topic+speedytax-package'></span>

<h3>Description</h3>

<p>Import classification results from the 'RDP Classifier' (Ribosomal Database Project),' 'USEARCH sintax,' 'vsearch sintax' and the 'QIIME2' (Quantitative Insights into Microbial Ecology) classifiers into 'phyloseq' tax_table objects.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: John Quensen <a href="mailto:quensenj@msu.edu">quensenj@msu.edu</a> (<a href="https://orcid.org/0000-0002-7669-1595">ORCID</a>) [copyright holder]
</p>

<hr>
<h2 id='import_qiime2_tax_table'>Import QIIME2 Classification Table</h2><span id='topic+import_qiime2_tax_table'></span>

<h3>Description</h3>

<p>Import taxonomy results exported from the QIIME2 Bayesian classifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_qiime2_tax_table(in_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_qiime2_tax_table_+3A_in_file">in_file</code></td>
<td>
<p>A tab-delimited classification table output by QIIME2</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function expects up to 7 ranks (Domain, Phylum, Class, Order, Family, Genus and Species) but determines the number actually in the file.
</p>
<p>Depending on how the QIIME2 classifier is built, the ranks in the Taxon field of the classification result may be separated by a semicolon or by a semicolon plus a space. This function correctly parses the Taxon field in both cases.
</p>


<h3>Value</h3>

<p>A phyloseq tax_table object
</p>


<h3>References</h3>

<p>Bolyen E, Rideout JR, Dillon MR, Bokulich NA, <em>et al</em>. 2019. Reproducible, interactive, scalable and extensible microbiome data science using QIIME 2. Nat Biotechnol 37:852-857.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example with ranks in Taxon field separated by semicolon plus a space.
taxonomy_file &lt;- system.file("extdata", "qiime2_table.tsv", package = "speedytax")
example_tax_table &lt;- import_qiime2_tax_table(in_file = taxonomy_file)
example_tax_table
## Example with ranks in Taxon field separated by semicolon only.
taxonomy_file &lt;- system.file("extdata", "qiime2_table_a.tsv", package = "speedytax")
example_tax_table &lt;- import_qiime2_tax_table(in_file = taxonomy_file)
example_tax_table
</code></pre>

<hr>
<h2 id='import_rdp_tax_table'>Import RDP Classifier Taxonomy Table</h2><span id='topic+import_rdp_tax_table'></span>

<h3>Description</h3>

<p>Imports fixed rank taxonomy files created with the RDP Classifier
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_rdp_tax_table(in_file, confidence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_rdp_tax_table_+3A_in_file">in_file</code></td>
<td>
<p>A fix-rank tab-delimited text file output by the RDP Classifier</p>
</td></tr>
<tr><td><code id="import_rdp_tax_table_+3A_confidence">confidence</code></td>
<td>
<p>The confidence level for filtering the taxonomy</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The RDP Classifier must be given the option -f fixrank (or &ndash;format fixrank) in order for this importer to work correctly.
</p>
<p>A confidence value of 0.8 for is recommended for full-length 16S rRNA gene sequences and a value of 0.5 is recommended for shorter amplicons.
</p>


<h3>Value</h3>

<p>A phyloseq tax_table object
</p>


<h3>References</h3>

<p>Wang Q, Garrity GM, Tiedje JM, Cole JR. 2007. Naive Bayesian classifier for rapid assignment of rRNA sequences into the new bacterial taxonomy. Appl Environ Microbiol 73:5261-5267.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>taxonomy_file &lt;- system.file("extdata", "rdp_table.tsv", package = "speedytax")
example_tax_table &lt;- import_rdp_tax_table(in_file = taxonomy_file)
example_tax_table
</code></pre>

<hr>
<h2 id='import_sintax_tax_table'>Import SINTAX Taxonomy Table</h2><span id='topic+import_sintax_tax_table'></span>

<h3>Description</h3>

<p>Imports taxonomy files created with USEARCH and vsearch sintax commands
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_sintax_tax_table(in_file, confidence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_sintax_tax_table_+3A_in_file">in_file</code></td>
<td>
<p>A fix-rank tab-delimited text file output by SINTAX</p>
</td></tr>
<tr><td><code id="import_sintax_tax_table_+3A_confidence">confidence</code></td>
<td>
<p>The confidence level for filtering the taxonomy (0.8 by default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function works with both vsearch and USEARCH sintax results.
</p>
<p>A confidence value of 0.8 is recommended for full-length 16S rRNA gene sequences and a value of 0.5 is recommended for shorter amplicons.
</p>


<h3>Value</h3>

<p>A phyloseq tax_table object
</p>


<h3>References</h3>

<p>Rognes T, Flouri T, Nichols B, Quince C, Mah√© F. (2016) VSEARCH: a versatile open source tool for metagenomics. PeerJ 4:e2584. doi: 10.7717/peerj.2584
</p>
<p>Edgar RC (2016) SINTAX: a simple non-Bayesian taxonomy classifier for 16S and ITS sequences. bioRxiv. doi:10.1101/074161
</p>


<h3>Examples</h3>

<pre><code class='language-R'># With a vsearch sintax result
taxonomy_file &lt;- system.file("extdata", "vsearch_sintax_table.tsv", package = "speedytax")
example_tax_table &lt;- import_sintax_tax_table(in_file = taxonomy_file)
example_tax_table
# With a USEARCH sintax result
taxonomy_file &lt;- system.file("extdata", "usearch_sintax_table.tsv", package = "speedytax")
example_tax_table &lt;- import_sintax_tax_table(in_file = taxonomy_file)
example_tax_table
</code></pre>

<hr>
<h2 id='read_tax_example'>Get path to example taxonomy files</h2><span id='topic+read_tax_example'></span>

<h3>Description</h3>

<p>Access the files in speedyseq's inst/extdata sub-directory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_tax_example(file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_tax_example_+3A_file">file</code></td>
<td>
<p>Name of file to return.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the full path to the requested file from the package sub-directory inst/extdata if it exists, or a list of the files in the directory if no file is specified.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>read_tax_example()
read_tax_example("rdp_table.tsv")
</code></pre>

<hr>
<h2 id='simple_cap'>Simple Capitalization</h2><span id='topic+simple_cap'></span>

<h3>Description</h3>

<p>Simple Capitalization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simple_cap(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simple_cap_+3A_x">x</code></td>
<td>
<p>A string or vector of strings</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The string with every word capitalized
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simple_cap("an old dog learning new tricks")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
