<!DOCTYPE html><html lang="en"><head><title>Help for package spinBayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {spinBayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#spinBayes-package'><p>spinBayes: Semi-Parametric Gene-Environment Interaction via Bayesian Variable Selection</p></a></li>
<li><a href='#BVCfit'><p>fit a Semi-parametric Bayesian variable selection</p></a></li>
<li><a href='#BVSelection'><p>Variable selection for a BVCfit object</p></a></li>
<li><a href='#data'><p>simulated data for demonstrating the features of BVCfit</p></a></li>
<li><a href='#plot.BVCfit'><p>plot a BVCfit object</p></a></li>
<li><a href='#predict.BVCfit'><p>make predictions from a BVCfit object</p></a></li>
<li><a href='#print.BVCfit'><p>print a BVCfit object</p></a></li>
<li><a href='#print.BVCfit.pred'><p>print a BVCfit.pred object</p></a></li>
<li><a href='#print.BVSelection'><p>print a BVSelection object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Semi-Parametric Gene-Environment Interaction via Bayesian
Variable Selection</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jie Ren &lt;renjie0910@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Many complex diseases are known to be affected by the interactions between 
   genetic variants and environmental exposures beyond the main genetic and environmental 
   effects. Existing Bayesian methods for gene-environment (G×E) interaction studies are 
   challenged by the high-dimensional nature of the study and the complexity of environmental 
   influences. We have developed a novel and powerful semi-parametric Bayesian variable 
   selection method that can accommodate linear and nonlinear G×E interactions simultaneously 
   (Ren et al. (2020) &lt;<a href="https://doi.org/10.1002%2Fsim.8434">doi:10.1002/sim.8434</a>&gt;). Furthermore, the proposed method can conduct 
   structural identification by distinguishing nonlinear interactions from main effects only 
   case within Bayesian framework. Spike-and-slab priors are incorporated on both individual 
   and group level to shrink coefficients corresponding to irrelevant main and interaction 
   effects to zero exactly. The Markov chain Monte Carlo algorithms of the proposed and 
   alternative methods are  efficiently implemented in C++. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, testthat</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, splines, glmnet, utils, stats, ggplot2</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jrhub/spinBayes">https://github.com/jrhub/spinBayes</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jrhub/spinBayes/issues">https://github.com/jrhub/spinBayes/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), covr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-10 00:29:41 UTC; 17854</td>
</tr>
<tr>
<td>Author:</td>
<td>Jie Ren [aut, cre],
  Fei Zhou [aut],
  Xiaoxi Li [aut],
  Cen Wu [aut],
  Yu Jiang [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-10 07:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='spinBayes-package'>spinBayes: Semi-Parametric Gene-Environment Interaction via Bayesian Variable Selection</h2><span id='topic+spinBayes'></span><span id='topic+spinBayes-package'></span>

<h3>Description</h3>

<p>Many complex diseases are known to be affected by the interactions between genetic variants and environmental exposures beyond the main genetic and environmental effects. Existing Bayesian methods for gene-environment (G×E) interaction studies are challenged by the high-dimensional nature of the study and the complexity of environmental influences. We have developed a novel and powerful semi-parametric Bayesian variable selection method that can accommodate linear and nonlinear G×E interactions simultaneously (Ren et al. (2020) <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>). Furthermore, the proposed method can conduct structural identification by distinguishing nonlinear interactions from main effects only case within Bayesian framework. Spike-and-slab priors are incorporated on both individual and group level to shrink coefficients corresponding to irrelevant main and interaction effects to zero exactly. The Markov chain Monte Carlo algorithms of the proposed and alternative methods are efficiently implemented in C++.
</p>
<p>Within the Bayesian framework, we propose a partially linear varying coefficient model (PLVC) for G×E interactions.  The varying coefficient functions capture the possible non-linear G×E interaction, and the linear component  models the G×E interactions with linear assumptions. The changing of basis with B splines is adopted to separate the coefficient functions with varying, non-zero constant and zero forms, corresponding to cases of nonlinear interaction, main effect only (no interaction) and no genetic interaction at all.
</p>


<h3>Details</h3>

<p>The user friendly, integrated interface BVCfit() allows users to flexibly choose the fitting methods they prefer. There are three arguments in BVCfit() that control the fitting method
</p>

<table>
<tr>
 <td style="text-align: right;">
sparse: </td><td style="text-align: left;"> whether to use the spike-and-slab priors to achieve sparsity. </td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
VC: </td><td style="text-align: left;"> whether to separate the coefficient functions with varying effects </td>
</tr>
<tr>
 <td style="text-align: right;"> </td><td style="text-align: left;"> and non-zero constant (main) effects.</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
structural: </td><td style="text-align: left;"> whether to use varying coefficient functions for modeling </td>
</tr>
<tr>
 <td style="text-align: right;"> </td><td style="text-align: left;"> non-linear GxE interactions.
</td>
</tr>

</table>

<p>BVCfit() returns a BVCfit object that contains the posterior estimates of each coefficients.
S3 generic functions BVSelection(), predict(), plot() and print() are implemented for BVCfit objects.
BVSelection() takes a BVCfit object and returns the variable selection results.
predict() takes a BVCfit object and returns the predicted values for new observations.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Jie Ren <a href="mailto:renjie0910@gmail.com">renjie0910@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Fei Zhou
</p>
</li>
<li><p> Xiaoxi Li
</p>
</li>
<li><p> Cen Wu
</p>
</li>
<li><p> Yu Jiang
</p>
</li></ul>



<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y., Wu, C. (2020). Semiparametric Bayesian variable selection
for gene-environment interactions. <em>Statistics in Medicine</em>, 39(5): 617–638. <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>.
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S., and Wu, C. (2021). Gene-Environment Interaction: A Variable Selection Perspective.
<em>Methods in Molecular Biology</em>, 2212:191-223. <a href="https://doi.org/10.1007/978-1-0716-0947-7_13">doi:10.1007/978-1-0716-0947-7_13</a>. PMID: 33733358.
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y., Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach
accounting for hierarchical structures. <em>Statistics in Medicine</em>, 37:437–456. <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>.
</p>
<p>Wu, C., Zhong, P.-S., and Cui, Y. (2018). Additive varying–coefficient model for nonlinear gene–environment interactions.
<em>Statistical Applications in Genetics and Molecular Biology</em>, 17(2). <a href="https://doi.org/10.1515/sagmb-2017-0008">doi:10.1515/sagmb-2017-0008</a>.
</p>
<p>Jiang, Y., Huang, Y., Du, Y., Zhao, Y., Ren, J., Ma, S., Wu, C. (2017). Identification of prognostic genes and pathways in
lung adenocarcinoma using a Bayesian Approach. <em>Cancer Informatics</em>, 1(7).
</p>
<p>Wu, C., Shi, X., Cui, Y., and Ma, S. (2015). A penalized robust semiparametric approach for gene-environment interactions.
<em>Statistics in Medicine</em>, 34(30): 4016–4030. <a href="https://doi.org/10.1002/sim.6609">doi:10.1002/sim.6609</a>.
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883. <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a>.
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially
linear varying coefficient model. <em>Statistics in Medicine</em>, 33(28), 4988–4998. <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>.
</p>
<p>Wu, C., and Cui, Y. (2013). Boosting signals in gene–based association studies via efficient SNP selection.
<em>Briefings in Bioinformatics</em>, 15(2):279–291. <a href="https://doi.org/10.1093/bib/bbs087">doi:10.1093/bib/bbs087</a>.
</p>
<p>Wu, C., and Cui, Y. (2013). A novel method for identifying nonlinear gene–environment interactions in case–control
association studies. <em>Human Genetics</em>, 132(12):1413–1425. <a href="https://doi.org/10.1007/s00439-013-1350-z">doi:10.1007/s00439-013-1350-z</a>.
</p>
<p>Wu, C., Zhong, P.S., and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<em>Technical Report. Michigan State University</em>.
</p>
<p>Wu, C., Li, S., and Cui, Y. (2012). Genetic Association Studies: An Information Content Perspective.
<em>Current Genomics</em>, 13(7), 566–573. <a href="https://doi.org/10.2174/138920212803251382">doi:10.2174/138920212803251382</a>.
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/jrhub/spinBayes">https://github.com/jrhub/spinBayes</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/jrhub/spinBayes/issues">https://github.com/jrhub/spinBayes/issues</a>
</p>
</li></ul>

<p><code><a href="#topic+BVCfit">BVCfit</a></code>
</p>

<hr>
<h2 id='BVCfit'>fit a Semi-parametric Bayesian variable selection</h2><span id='topic+BVCfit'></span>

<h3>Description</h3>

<p>fit a Bayesian semi-parametric model for both linear and non-linear G×E interactions. Users can also specify all the interactions as linear and fit a Bayesian LASSO type of model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BVCfit(
  X,
  Y,
  Z,
  E = NULL,
  clin = NULL,
  iterations = 10000,
  burn.in = NULL,
  sparse = TRUE,
  structural = TRUE,
  VC = TRUE,
  kn = 2,
  degree = 2,
  hyper = NULL,
  debugging = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BVCfit_+3A_x">X</code></td>
<td>
<p>the matrix of predictors (genetic factors) without intercept. Each row should be an observation vector. A column of 1 will be added to the X matrix
as the intercept.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_y">Y</code></td>
<td>
<p>the response variable. The current version of BVCfit only supports continuous response.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_z">Z</code></td>
<td>
<p>a vector of environmental factor for non-linear G×E interactions.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_e">E</code></td>
<td>
<p>a vector of environmental factor for linear G×E interactions.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_clin">clin</code></td>
<td>
<p>a matrix of clinical variables. Clinical variables are not subject to penalty.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_iterations">iterations</code></td>
<td>
<p>the number of MCMC iterations.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_burn.in">burn.in</code></td>
<td>
<p>the number of iterations for burn-in.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly. 'sparse' has effect only when VC=TRUE.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_structural">structural</code></td>
<td>
<p>logical flag. If TRUE, the coefficient functions with varying effects and constant effects will be penalized separately. 'structural' has effect only when VC=TRUE.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_vc">VC</code></td>
<td>
<p>logical flag. If TRUE, varying coefficient functions will be used for modeling the interactions between Z and X.
If FALSE, interactions between Z and X will be modeled as linear interactions.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_kn">kn</code></td>
<td>
<p>the number of interior knots for B-spline.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_degree">degree</code></td>
<td>
<p>the degree of B spline basis.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_hyper">hyper</code></td>
<td>
<p>a named list of hyperparameters.</p>
</td></tr>
<tr><td><code id="BVCfit_+3A_debugging">debugging</code></td>
<td>
<p>logical flag. If TRUE, progress will be output to the console and extra information will be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, varying coefficient functions are used for modeling the nonlinear interactions between Z and X. Assuming both E and clin are NULL, the model can be expressed as
</p>
<p style="text-align: center;"><code class="reqn">Y = \beta_{0}(Z)+\sum\beta_{j}(Z)X_{j} + \epsilon </code>
</p>

<p>The basis expansion and changing of basis with B splines will be done automatically:
</p>
<p style="text-align: center;"><code class="reqn">\beta_{j}(\cdot)\approx \gamma_{j1} + \sum_{k=2}^{q}{B}_{jk}(\cdot)\gamma_{jk}</code>
</p>

<p>where <code class="reqn">B_{jk}(\cdot)</code> represents B spline basis. <code class="reqn">\gamma_{j1}</code> and <code class="reqn">(\gamma_{j2}, \ldots, \gamma_{jq})^\top</code> correspond to the constant and varying parts of the coefficient functional, respectively.
q=kn+degree+1 is the number of basis functions. By default, kn=degree=2. User can change the values of kn and degree to any other positive integers.
If E is provided, the linear interactions between E and X will be added modeled as pairwise-products:
</p>
<p style="text-align: center;"><code class="reqn">Y = \beta_{0}(Z)+\sum\beta_{j}(Z)X_{j} + \zeta_{0}E + \sum \zeta_{j}EX_{j} + \epsilon</code>
</p>

<p>If clin is provided, clinical variables
will be added to the model.
</p>
<p>If VC=FALSE, all interactions are treated as linear and a Bayesian LASSO model will be used. With non-null values of E and clin, the full linear model is:
</p>
<p style="text-align: center;"><code class="reqn">Y \sim Z + ZX + clin + E + EX</code>
</p>

<p>Please check the references for more details about the model.
</p>
<p>Users can modify the hyper-parameters by providing a named list of hyper-parameters via the argument 'hyper'.
The list can have the following named components
</p>

<ul>
<li><p> a.c, a.v, a.e: shape parameters of the Gamma priors on <code class="reqn">\lambda_{c}</code>, <code class="reqn">\lambda_{v}</code> and <code class="reqn">\lambda_{e}</code>, respectively.
</p>
</li>
<li><p> b.c, b.v, b.e: rate parameters of the Gamma priors on <code class="reqn">\lambda_{c}</code>, <code class="reqn">\lambda_{v}</code> and <code class="reqn">\lambda_{e}</code>, respectively.
</p>
</li>
<li><p> r.c, r.v, r.e: shape parameters of the Beta priors (<code class="reqn">\pi^{r-1}(1-\pi)^{w-1}</code>) on <code class="reqn">\pi_{c}</code>, <code class="reqn">\pi_{v}</code> and <code class="reqn">\pi_{e}</code>, respectively.
</p>
</li>
<li><p> w.c, w.v, w.e: shape parameters of the Beta priors on <code class="reqn">\pi_{c}</code>, <code class="reqn">\pi_{v}</code> and <code class="reqn">\pi_{e}</code>, respectively.
</p>
</li>
<li><p> s: shape parameters of the Inverse-gamma prior on <code class="reqn">\sigma^{2}</code>.
</p>
</li>
<li><p> h: scale parameters of the Inverse-gamma prior on <code class="reqn">\sigma^{2}</code>.
</p>
</li></ul>

<p>Please check the references for more details about the prior distributions.
</p>


<h3>Value</h3>

<p>an object of class &quot;BVCfit&quot; is returned, which is a list with components:
</p>

<ul>
<li><p> posterior: posterior samples from the MCMC
</p>
</li>
<li><p> coefficients: a list of posterior estimates of coefficients
</p>
</li>
<li><p> burn.in: the number of iterations for burn-in
</p>
</li>
<li><p> iterations: the number of MCMC iterations.
</p>
</li></ul>



<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y., Wu, C. (2020) Semiparametric Bayesian variable selection for gene-environment interactions.
<em>Statistics in Medicine</em>, 39(5): 617– 638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gExp)

## default method
spbayes=BVCfit(X, Y, Z, E, clin)
spbayes


## non-structural
structural=FALSE
spbayes=BVCfit(X, Y, Z, E, clin, structural=structural)
spbayes

## non-sparse
sparse=FALSE
spbayes=BVCfit(X, Y, Z, E, clin, sparse=sparse)
spbayes


</code></pre>

<hr>
<h2 id='BVSelection'>Variable selection for a BVCfit object</h2><span id='topic+BVSelection'></span><span id='topic+BVSelection.BVCNonSparse'></span><span id='topic+BVSelection.BVCSparse'></span>

<h3>Description</h3>

<p>Variable selection for a BVCfit object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BVSelection(obj, ...)

## S3 method for class 'BVCNonSparse'
BVSelection(obj, burn.in = obj$burn.in, prob = 0.95, ...)

## S3 method for class 'BVCSparse'
BVSelection(obj, burn.in = obj$burn.in, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BVSelection_+3A_obj">obj</code></td>
<td>
<p>BVCfit object.</p>
</td></tr>
<tr><td><code id="BVSelection_+3A_...">...</code></td>
<td>
<p>other BVSelection arguments</p>
</td></tr>
<tr><td><code id="BVSelection_+3A_burn.in">burn.in</code></td>
<td>
<p>MCMC burn-in.</p>
</td></tr>
<tr><td><code id="BVSelection_+3A_prob">prob</code></td>
<td>
<p>probability for credible interval, between 0 and 1. e.g. prob=0.95 leads to 95% credible interval</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For class 'BVCSparse', the median probability model (MPM) (Barbieri and Berger 2004) is used to identify predictors that are significantly associated
with the response variable. For class 'BVCNonSparse', variable selection is based on 95% credible interval.
Please check the references for more details about the variable selection.
</p>


<h3>Value</h3>

<p>an object of class &quot;BVSelection&quot; is returned, which is a list with components:
</p>

<ul>
<li><p> method: method used for identifying important effects
</p>
</li>
<li><p> indices: a list of indices and names of selected variables
</p>
</li>
<li><p> summary: a summary of selected variables
</p>
</li></ul>



<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y., Wu, C. (2020) Semiparametric Bayesian variable selection for gene-environment interactions.
<em>Statistics in Medicine</em>, 39(5): 617– 638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>
<p>Barbieri, M.M. and Berger, J.O. (2004). Optimal predictive model selection
<em>Ann. Statist</em>, 32(3):870–897
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BVCfit">BVCfit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gExp)
## sparse
spbayes=BVCfit(X, Y, Z, E, clin)
spbayes

selected = BVSelection(spbayes)
selected$indices

## non-sparse
spbayes=BVCfit(X, Y, Z, E, clin, sparse=FALSE)
spbayes

selected = BVSelection(spbayes)
selected

</code></pre>

<hr>
<h2 id='data'>simulated data for demonstrating the features of BVCfit</h2><span id='topic+data'></span><span id='topic+gExp.new'></span><span id='topic+gExp.L'></span><span id='topic+spbayes'></span><span id='topic+X'></span><span id='topic+Y'></span><span id='topic+Z'></span><span id='topic+E'></span><span id='topic+clin'></span><span id='topic+X2'></span><span id='topic+Y2'></span><span id='topic+Z2'></span><span id='topic+E2'></span><span id='topic+clin2'></span><span id='topic+X.new'></span><span id='topic+Y.new'></span><span id='topic+Z.new'></span><span id='topic+E.new'></span><span id='topic+clin.new'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of BVCfit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gExp")
data("gExp.new")
data("gExp.L")
</code></pre>


<h3>Format</h3>

<p>gExp consists of five components: X, Y, Z, E and clin. gExp.new contains the data of new observations
(X.new, Y.new, Z.new, E.new and clin.new) which can be used for evaluating the prediction performance.
</p>
<p>gExp.L contains larger datasets: X2, Y2, Z2, E2 and clin2
</p>


<h3>Details</h3>

<p>the same true model is used for generating Y, Y.new and Y2
</p>
<p style="text-align: center;"><code class="reqn">Y = \beta_{0}(Z)+\beta_{1}(Z)X_{1} + \beta_{2}(Z)X_{2} + 1.5X_{3} - X_{5} + 1.3E - 1.2 EX_{2}+1.3 EX_{4} - clin_{1} + 1.5 clin_{2} + \epsilon</code>
</p>

<p>where <code class="reqn">\epsilon\sim N(0,1)</code>, <code class="reqn">\beta_{0}=2\sin(0.2\pi*Z)</code>, <code class="reqn">\beta_{1}=2\exp(0.2Z-1) </code> and <code class="reqn">\beta_{2}=-0.6Z(1-0.1Z) </code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BVCfit">BVCfit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gExp)
dim(X)

data(gExp.L)
dim(X)

</code></pre>

<hr>
<h2 id='plot.BVCfit'>plot a BVCfit object</h2><span id='topic+plot.BVCfit'></span>

<h3>Description</h3>

<p>plot the identified varying effects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BVCfit'
plot(x, prob=0.95, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.BVCfit_+3A_x">x</code></td>
<td>
<p>BVCfit object.</p>
</td></tr>
<tr><td><code id="plot.BVCfit_+3A_prob">prob</code></td>
<td>
<p>probability for credible interval, between 0 and 1. e.g. prob=0.95 leads to 95% credible interval</p>
</td></tr>
<tr><td><code id="plot.BVCfit_+3A_...">...</code></td>
<td>
<p>other plot arguments</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+BVCfit">BVCfit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gExp)
spbayes=BVCfit(X, Y, Z, E, clin)
plot(spbayes)

</code></pre>

<hr>
<h2 id='predict.BVCfit'>make predictions from a BVCfit object</h2><span id='topic+predict.BVCfit'></span><span id='topic+predict.VarLin'></span><span id='topic+predict.VarOnly'></span><span id='topic+predict.LinOnly'></span>

<h3>Description</h3>

<p>make predictions from a BVCfit object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BVCfit'
predict(object, X.new, Z.new, E.new = NULL, clin.new = NULL, Y.new = NULL, ...)

## S3 method for class 'VarLin'
predict(object, X.new, Z.new, E.new, clin.new = NULL, Y.new = NULL, ...)

## S3 method for class 'VarOnly'
predict(object, X.new, Z.new, clin.new = NULL, Y.new = NULL, ...)

## S3 method for class 'LinOnly'
predict(object, X.new, Z.new, E.new = NULL, clin.new = NULL, Y.new = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.BVCfit_+3A_object">object</code></td>
<td>
<p>BVCfit object.</p>
</td></tr>
<tr><td><code id="predict.BVCfit_+3A_x.new">X.new</code></td>
<td>
<p>a matrix of new values for X at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.BVCfit_+3A_z.new">Z.new</code></td>
<td>
<p>a vector of new values for Z at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.BVCfit_+3A_e.new">E.new</code></td>
<td>
<p>a vector of new values for E at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.BVCfit_+3A_clin.new">clin.new</code></td>
<td>
<p>a vector or matrix of new values for clin at which predictions are to be made.</p>
</td></tr>
<tr><td><code id="predict.BVCfit_+3A_y.new">Y.new</code></td>
<td>
<p>a vector of the response of new observations. If provided, the prediction mean squared error (PMSE) will be computed based on Y.new.</p>
</td></tr>
<tr><td><code id="predict.BVCfit_+3A_...">...</code></td>
<td>
<p>other predict arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>X.new (clin.new) must have the same number of columns as X (clin) used for fitting the model. If E and clin are provided when fit the model, E.new and clin.new
must not be NULL, and vice versa. The predictions are made based on the posterior estimates of coefficients in the BVCfit object.
Note that the main effects of environmental exposures Z and E are not subject to selection.
</p>


<h3>Value</h3>

<p>an object of class &quot;BVCfit.pred&quot; is returned, which is a list with components:
</p>
<table role = "presentation">
<tr><td><code>pmse</code></td>
<td>
<p>predictions mean squared error. pmse is NULL is Y.new=NULL.</p>
</td></tr>
<tr><td><code>y.pred</code></td>
<td>
<p>predicted values of the new observations.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+BVCfit">BVCfit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gExp)
spbayes=BVCfit(X, Y, Z, E, clin)
spbayes

data(gExp.new)
pred = predict(spbayes, X.new, Z.new, E.new, clin.new, Y.new)
pred$pmse
# pred$y.pred

</code></pre>

<hr>
<h2 id='print.BVCfit'>print a BVCfit object</h2><span id='topic+print.BVCfit'></span>

<h3>Description</h3>

<p>Print a summary of a BVCfit object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BVCfit'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.BVCfit_+3A_x">x</code></td>
<td>
<p>BVCfit object.</p>
</td></tr>
<tr><td><code id="print.BVCfit_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.BVCfit_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+BVCfit">BVCfit</a></code>
</p>

<hr>
<h2 id='print.BVCfit.pred'>print a BVCfit.pred object</h2><span id='topic+print.BVCfit.pred'></span>

<h3>Description</h3>

<p>Print a summary of a BVCfit.pred object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BVCfit.pred'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.BVCfit.pred_+3A_x">x</code></td>
<td>
<p>BVCfit object.</p>
</td></tr>
<tr><td><code id="print.BVCfit.pred_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.BVCfit.pred_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+predict.BVCfit">predict.BVCfit</a></code>
</p>

<hr>
<h2 id='print.BVSelection'>print a BVSelection object</h2><span id='topic+print.BVSelection'></span>

<h3>Description</h3>

<p>Print a summary of a BVSelection object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BVSelection'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.BVSelection_+3A_x">x</code></td>
<td>
<p>BVSelection object.</p>
</td></tr>
<tr><td><code id="print.BVSelection_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.BVSelection_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+BVSelection">BVSelection</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
