<!DOCTYPE html><html lang="en"><head><title>Help for package SPmlficmcm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SPmlficmcm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SPmlficmcm-package'>
<p>SemiParametric Maximum Likelihood for interaction in case-mother control-mother designs</p></a></li>
<li><a href='#Est.Inpar'>
<p>Computes the initial values</p></a></li>
<li><a href='#FtSmlrmCMCM'>
<p>Generates the logistic model data</p></a></li>
<li><a href='#SeltcEch'>
<p>Resampling</p></a></li>
<li><a href='#Spmlficmcm'>
<p>Semiparametric maximum likelihood for interaction in case-mother control-mother</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-12-10</td>
</tr>
<tr>
<td>Title:</td>
<td>Semiparametric Maximum Likelihood Method for Interactions
Gene-Environment in Case-Mother Control-Mother Designs</td>
</tr>
<tr>
<td>Author:</td>
<td>Moliere Nguile-Makao [aut, cre],
  Alexandre Bureau [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Moliere Nguile-Makao &lt;moliere.nguile@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>nleqslv</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the method of general semiparametric maximum likelihood estimation for logistic models in case-mother control-mother designs.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-12-11 10:46:12 UTC; molierenguilemakao</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-12-11 19:53:05</td>
</tr>
</table>
<hr>
<h2 id='SPmlficmcm-package'>
SemiParametric Maximum Likelihood for interaction in case-mother control-mother designs
</h2><span id='topic+SPmlficmcm-package'></span><span id='topic+SPmlficmcm'></span>

<h3>Description</h3>

<p>Implementation of a method of general semiparametric maximum likelihood estimation for logistic models in case-mother control-mother designs. The method was proposed by <em>Chen et al., (2012)</em> for the complete data and <em>Nguile-Makao et al., (2015)</em> proposed an extension of the method allowing missing offspring genotype.
</p>


<h3>Details</h3>

<p>The package <span class="pkg">SPmlficmcm</span> implements the semiparametric maximum likelihood estimation method published by <em>Chen et al., (2012)</em>. This method permits to analyze the interaction effects involving genetic variants and environmental exposures on the risk of adverse obstetric and early-life outcomes. <em>Nguile-Makao et al., (2015)</em> proposed an extension of this method allowing missing offspring genotype. The package performs the analysis the following way: it builds the nonlinear system from the database, resolves the nonlinear system using the <code><a href="nleqslv.html#topic+nleqslv">nleqslv</a></code> function of package <span class="pkg">nleqslv</span>. It estimates the model parameters and the standard errors using the log profile likelihood function and the one-step method estimation. All this procedure may be done for complete data and also for missing offspring genotype. For more details see <em>Chen et al., (2012)</em>, and <em>Nguile-Makao et al., (2015)</em> . 
The modeling supposes that the distribution of maternal genotype and offspring genotype satisfy the following assumptions: <em>random mating</em>, <em>Hardy-Weinberg equilibrium</em> and <em>Mendelian inheritance</em>. The package also permits to treat the missing offspring genotype data.
</p>
<p>Index of help topics:
</p>
<pre>
Est.Inpar               Computes the initial values
FtSmlrmCMCM             Generates the logistic model data
SPmlficmcm-package      SemiParametric Maximum Likelihood for
                        interaction in case-mother control-mother
                        designs
SeltcEch                Resampling
Spmlficmcm              Semiparametric maximum likelihood for
                        interaction in case-mother control-mother
</pre>


<h3>Author(s)</h3>

<p>Moliere Nguile-Makao and Alexandre Bureau
</p>
<p>Maintainer:  Moliere Nguile-Makao &lt;moliere.nguile@gmail.com&gt;
</p>


<h3>References</h3>

<p>Jinbo Chen,  Dongyu Lin  and Hagit Hochner (2012) Semiparametric Maximum Likelihood Methods for Analyzing Genetic and Environmental Effects with Case-Control Mother-Child Pair Data. <em>Biometrics</em> DOI: 10.1111/j.1541-0420.2011.01728.
</p>
<p>Moliere Nguile-Makao, Alexandre Bureau (2015), Semi-Parametric Maximum likelihood Method for interaction in Case-Mother Control-Mother designs: Package SPmlficmcm. <em>Journal of Statistical Software DOI: 10.18637/jss.v068.i10</em>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Est.Inpar">Est.Inpar</a></code>, <code><a href="#topic+FtSmlrmCMCM">FtSmlrmCMCM</a></code>, <code><a href="#topic+SeltcEch">SeltcEch</a></code>, <code><a href="#topic+Spmlficmcm">Spmlficmcm</a></code>
</p>

<hr>
<h2 id='Est.Inpar'>
Computes the initial values 
</h2><span id='topic+Est.Inpar'></span>

<h3>Description</h3>

<p>Computes initial values of the model parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Est.Inpar(fl, N, gnma, gnch, tab1, typ, p = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Est.Inpar_+3A_fl">fl</code></td>
<td>
<p>Model formula.
</p>
</td></tr>
<tr><td><code id="Est.Inpar_+3A_n">N</code></td>
<td>
<p>Numeric vector containing the number of eligible cases( N1 ) and controls ( N0) in the population N = ( N0, N1 ). 
</p>
</td></tr>
<tr><td><code id="Est.Inpar_+3A_gnma">gnma</code></td>
<td>
<p>Name of mother genotype variable.
</p>
</td></tr>
<tr><td><code id="Est.Inpar_+3A_gnch">gnch</code></td>
<td>
<p>Name of offspring genotype variable.
</p>
</td></tr>
<tr><td><code id="Est.Inpar_+3A_tab1">tab1</code></td>
<td>
<p><code>data.frame</code> of the database.
</p>
</td></tr>
<tr><td><code id="Est.Inpar_+3A_typ">typ</code></td>
<td>
<p>Argument that specifies whether the data are complete (1) or there are missing offspring genotypes (2). 
</p>
</td></tr>
<tr><td><code id="Est.Inpar_+3A_p">p</code></td>
<td>
<p>Disease prevalence. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function use logistic regression to evaluate the initial values of the equation parameters given by the formula and uses empirical estimation to compute the initial values of the nonlinear system. For details, see <em>Nguile-Makao et al., (2015)</em>.
</p>


<h3>Value</h3>

<p>A list containing components
</p>
<table role = "presentation">
<tr><td><code>parms</code></td>
<td>
<p>Initial values of the model parameters given by the formula and the allelic frequency parameter.</p>
</td></tr>
<tr><td><code>ma.u</code></td>
<td>
<p>Initial values of nonlinear system.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Moliere Nguile-Makao, Alexandre Bureau (2015).Semi-Parametric Maximum Likelihood Method for Interaction in Case-Mother Control-Mother Designs: Package SPmlficmcm. <em>Journal of Statistical Software</em>, 68(10), 1-17. doi:10.18637/jss.v068.i10
</p>

<hr>
<h2 id='FtSmlrmCMCM'>
Generates the logistic model data 
</h2><span id='topic+FtSmlrmCMCM'></span>

<h3>Description</h3>

<p>The function generates data from a logistic regression model. The data obtained contain: an outcome variable, the mother and child genotype coded as the number of minor allele and the environmental factors. For simulation of each environmental variable, the user can specify the coefficients of linear dependency between the mother genotype and the environmental factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FtSmlrmCMCM(fl, N, theta, beta, interc, vpo, vprob, vcorr)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FtSmlrmCMCM_+3A_fl">fl</code></td>
<td>
<p>Model formula.
</p>
</td></tr>
<tr><td><code id="FtSmlrmCMCM_+3A_n">N</code></td>
<td>
<p>Sample size.
</p>
</td></tr>
<tr><td><code id="FtSmlrmCMCM_+3A_theta">theta</code></td>
<td>
<p>Minor allele frequency.
</p>
</td></tr>
<tr><td><code id="FtSmlrmCMCM_+3A_beta">beta</code></td>
<td>
<p>Parameter vector of the effects.
</p>
</td></tr>
<tr><td><code id="FtSmlrmCMCM_+3A_interc">interc</code></td>
<td>
<p>Intercept of the model.
</p>
</td></tr>
<tr><td><code id="FtSmlrmCMCM_+3A_vpo">vpo</code></td>
<td>
<p>Numeric vector containing the positions of the terms corresponding to the mother and child genotypes in the left-hand side of the formula.
</p>
</td></tr>
<tr><td><code id="FtSmlrmCMCM_+3A_vprob">vprob</code></td>
<td>
<p>Numeric vector containing the prevalence (success probability) of each environmental factor.
</p>
</td></tr>
<tr><td><code id="FtSmlrmCMCM_+3A_vcorr">vcorr</code></td>
<td>
<p>Numeric vector containing the coefficients of linear dependency between the mother genotype and environmental factors. The value 0 corresponds to independence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function generates data, where the outcome variable is associated with the explanatory variables by a logistic regression model. 
</p>
<p>Ex: log(P/(1-P))=B0+B1*X1+B2*X2+Bm*Gm+Bc*Gc+B2m*X2:Gm.
</p>
<p>Where P=Pr(Y=1|X), X=(X1,X2) and Y is the outcome variable. The environmental factors are generated the following way: for each variable, a temporary variable is generated with a binomial law of success probability equal to vprob[i] plus vcorr[i]*Gm, i is the factor position. The genotypes of the mother and her child are coded as the number of minor alleles, i.e. under an additive model of the alleles on the log odds. The data generated suppose that the assumptions of <em>Hardy-Weinberg equilibrium</em>, <em>random mating type</em> and <em>Mendelian inheritance</em> are satisfied. The function uses the formula f(x)=1/(1+exp(-x)) to generated the outcome variable. The data.frame returned by the function contains the variables whose names correspond to terms labels of the formula. The particularity of this function is to generate the genotype of a mother and her child taking into account the parental link.
</p>


<h3>Value</h3>

<p>The function returns a <code>data.frame</code> containing an outcome variable, the environmental factors and two genotypes of the mother and her child.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1-Creation of database
  set.seed(13200)
   M=5000
   fl=outc~X1+X2+gm+gc+X2:gm
   vpo=c(3,4)
   vprob=c(0.35,0.55)
   vcorr=c(2,1)
   theta=0.3;
   beta=c(-0.916,0.857,0.405,-0.693,0.573)
   interc=-2.23
   Dataf&lt;-FtSmlrmCMCM(fl,M,theta,beta,interc,vpo,vprob,vcorr)
   Dataf[1:10,]
</code></pre>

<hr>
<h2 id='SeltcEch'>
Resampling
</h2><span id='topic+SeltcEch'></span>

<h3>Description</h3>

<p>The function draws randomly n1+n0 individuals among N1 + N0 individuals with N0 &gt; n0 representing the numbers of non-cases and N1 &gt; n1 representing the number of cases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeltcEch(outc, n1, n0, id, datf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SeltcEch_+3A_outc">outc</code></td>
<td>
<p>Outcome variable (0,1).
</p>
</td></tr>
<tr><td><code id="SeltcEch_+3A_n1">n1</code></td>
<td>
<p>Numeric value representing the number of cases.
</p>
</td></tr>
<tr><td><code id="SeltcEch_+3A_n0">n0</code></td>
<td>
<p>Numeric value representing the number of controls.
</p>
</td></tr>
<tr><td><code id="SeltcEch_+3A_id">id</code></td>
<td>
<p>Identifying number of the mother-child pair.
</p>
</td></tr>
<tr><td><code id="SeltcEch_+3A_datf">datf</code></td>
<td>
<p><code>data.frame</code> of the database.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function uses the <code><a href="base.html#topic+sample">sample</a></code> function to resample the database.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with n0 + n1 rows.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+sample">sample</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(13200)
   M=5000;
   fl1=outc~Z1+Z2+Gm+Gc+Z2:Gm;
   vpo=c(3,4)
   vprob=c(0.35,0.55)
   vcorr=c(2,1)
   theta=0.3
   beta=c(-0.916,0.857,0.405,-0.693,0.573)
   interc=-2.23
   Dataf&lt;-FtSmlrmCMCM(fl1,M,theta,beta,interc,vpo,vprob,vcorr)        
   # Number of subjects eligible to the study in the population
   N0=dim(Dataf[Dataf["outc"]==0,])[1];
   N1=dim(Dataf[Dataf["outc"]==1,])[1]
   N=c(N0,N1)          
   # Sampling of the study database  
   n0=308
   n1=83 
   DatfE1&lt;-SeltcEch("outc",n1,n0,"obs",Dataf)
   DatfE1[1:10,] 
</code></pre>

<hr>
<h2 id='Spmlficmcm'>
Semiparametric maximum likelihood for interaction in case-mother control-mother
</h2><span id='topic+Spmlficmcm'></span>

<h3>Description</h3>

<p>The function builds the nonlinear system from the data, solves the system and assesses the effect of each factor of the model, computes the variance - covariance matrix and deduces from it the standard deviations of each factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Spmlficmcm(fl, N, gmname, gcname, DatfE, typ, start, p=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Spmlficmcm_+3A_fl">fl</code></td>
<td>
<p>Model formula.
</p>
</td></tr>
<tr><td><code id="Spmlficmcm_+3A_n">N</code></td>
<td>
<p>Numeric vector containing eligible number cases and controls in the study population N=(N0, N1).
</p>
</td></tr>
<tr><td><code id="Spmlficmcm_+3A_gmname">gmname</code></td>
<td>
<p>Name of mother genotype variable.
</p>
</td></tr>
<tr><td><code id="Spmlficmcm_+3A_gcname">gcname</code></td>
<td>
<p>Name of offspring genotype variable.
</p>
</td></tr>
<tr><td><code id="Spmlficmcm_+3A_datfe">DatfE</code></td>
<td>
<p><code>data.frame</code> in long format containing the following variables:<em>outcome</em> variable, <em>mother genotype</em>, <em>offspring genotype</em> and <em>environmental factors</em>.
</p>
</td></tr>
<tr><td><code id="Spmlficmcm_+3A_typ">typ</code></td>
<td>
<p>Argument indicating whether the data are complete (1) or contain missing offspring genotypes (2).
</p>
</td></tr>
<tr><td><code id="Spmlficmcm_+3A_start">start</code></td>
<td>
<p>Vector of the initial values of the model parameters.
</p>
</td></tr>
<tr><td><code id="Spmlficmcm_+3A_p">p</code></td>
<td>
<p>Disease prevalence 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>Spmlficmcm</code> builds the nonlinear system from the data and solves the nonlinear system. Then, it uses the log profile likelihood function and the one-step method to estimate the parameters of each factor of the model formula and their standard errors. The programme computes the gradient of the profile likelihood using the analytical formula and the Hessian matrix numerically from the gradient. The genotype is coded as the number of minor alleles. The model supposes that the distribution of maternal genotype and offspring genotype satisfy the following assumptions: <em>random mating</em>, <em>Hardy-Weinberg equilibrium</em> and <em>Mendelian inheritance</em>. When the data contains missing offspring genotypes, the profile likelihood is summed over the possible genotypes of each child whose genotype is missing. The argument <code>typ</code> allows the user to specify whether the data is complete or not. Argument <code>start</code> permits to the user to give the initials values of model parameter. 
Ex: in the following equation log(P/(1-P))=B0+B1*X1+B2*X2+Bm*Gm+Bc*Gc+B2m*X2:Gm, <code>start</code>=(B0, B1, B2, Bm, Bc, B2m, fp) where fp is the log of the odds of the minor allelic frequency. However, if the user provides no values, the function uses logistic regression to compute the initial B=(B0, B1, B2, Bm, Bc, B2m) and takes 0.1 as the initial value of fp. If the argument <code>N</code> is unavailable, it is possible to specify the disease population prevalence in the argument <code>p</code> instead of <code>N</code>. In that casse, <code>N1</code> is set equal to 5 n1, in order to avoid observing N1&lt;n1 when prevalence is small. We then set N0=[(1-p)/p]*N1. 
</p>


<h3>Value</h3>

<p>A list containing components
</p>
<table role = "presentation">
<tr><td><code>Uim</code></td>
<td>
<p>Nonlinear system solution</p>
</td></tr>
<tr><td><code>MatR</code></td>
<td>
<p>Matrix containing the estimates and their standard errors</p>
</td></tr>
<tr><td><code>Matv</code></td>
<td>
<p>Variance - covariance matrix</p>
</td></tr>
<tr><td><code>Lhft</code></td>
<td>
<p>Log-likelihood function. It takes as argument a vector of the model parameters</p>
</td></tr>
<tr><td><code>Value_loglikh</code></td>
<td>
<p>Value of the Log-likelihood function computed at the parameters estimated</p>
</td></tr>
</table>


<h3>References</h3>

<p>Jinbo Chen, Dongyu Lin and Hagit Hochner (2012) Semiparametric Maximum Likelihood Methods for Analyzing Genetic and Environmental Effects with Case-Control Mother-Child Pair Data. <em>Biometrics</em> DOI: 10.1111/j.1541-0420.2011.01728.
</p>
<p>Moliere Nguile-Makao, Alexandre Bureau (2015), Semi-Parametric Maximum likelihood Method for interaction in Case-Mother Control-Mother designs: Package SPmlficmcm. <em>Journal of Statistical Software DOI: 10.18637/jss.v068.i10</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1-Creation of database
## Not run: 
  set.seed(13200)
  M=20000;
  fl=outc~X1+X2+gm+gnch+X1:gnch+X2:gm;
  theta=0.3
  beta=c(-0.916,0.857,0.588,0.405,-0.693,0.488)
  interc=-2.23
  vpo=c(3,4)
  vprob=c(0.35,0.55)
  vcorr=c(2,1)
  Dataf&lt;-FtSmlrmCMCM(fl,M,theta,beta,interc,vpo,vprob,vcorr)
  rho&lt;-table(Dataf$outc)[2]/20000 # Disease prevalence
         
  # Number of subjects eligible to the study in the population 
  N=c(dim(Dataf[Dataf$outc==0,])[1],dim(Dataf[Dataf$outc==1,])[1])
         
  # Sampling of the study database  
  n0=1232;n1=327; 
  DatfE1&lt;-SeltcEch("outc",n1,n0,"obs",Dataf)


# 2 Creation of missing data on the offspring genotype 
        DatfE=DatfE1 
        gnch&lt;-DatfE["gnch"]
        gnch&lt;-as.vector(gnch[,1])
        gnch1&lt;-sample(c(0,1),length(gnch),replace=TRUE,prob=c(0.91,0.09))
        gnch[gnch1==1]&lt;-NA
        DatfE=DatfE1
        DatfE$gnch&lt;-NULL;DatfE$gnch&lt;-gnch
# 3 Creation of the two databases 
      # DatfEcd :complete data
      # DatfEmd :data with missing genotypes for a subset of children.
        DatfEcd&lt;-DatfE[is.na(DatfE["gnch"])!=TRUE,]
        DatfEmd&lt;-DatfE
        rm(gnch);rm(gnch1) 
# data obtained
DatfEcd[26:30,]
DatfEmd[26:30,]

##4 Estimation of parameters=======================================================
## model equation         
fl=outc~X1+X2+gm+gnch+X1:gnch+X2:gm;
## Estimation of the parameters (no missing data)
        # N = (N0,N1) is available
        Rsnm1&lt;-Spmlficmcm(fl,N,"gm","gnch",DatfEcd,1)
        #solution of the nonlinear system
        round(Rsnm1$Uim,digits=3)
        #estimates
        round(Rsnm1$MatR,digits=3)
        #variance - covariance matrix
        round(Rsnm1$Matv,digits=5)
        # N = (N0,N1) is not available
        Rsnm2&lt;-Spmlficmcm(fl=fl,gmname="gm",gcname="gnch",DatfE=DatfEcd,typ=1,p=rho)
        #solution of the nonlinear system
        round(Rsnm2$Uim,digits=3)
        #estimates
        round(Rsnm2$MatR,digits=3)
        #variance - covariance matrix
        round(Rsnm2$Matv,digits=5)
## Estimation of the parameters (with missing data)
        # N = (N0,N1) is available
        Rswm1&lt;-Spmlficmcm(fl,N,"gm","gnch",DatfEmd,typ=2)
        #solution of the nonlinear system
        round(Rswm1$Uim,digits=3)
        #estimates
        round(Rswm1$MatR,digits=3)
        #variance - covariance matrix
        round(Rswm1$Matv,digits=5)
        # N = (N0,N1) is not available
        Rswm2&lt;-Spmlficmcm(fl=fl,gmname="gm",gcname="gnch",DatfE=DatfEmd,typ=2,p=rho)
        #solution of the nonlinear system
        round(Rswm2$Uim,digits=3)
        #estimates
        round(Rswm2$MatR,digits=3)
        #variance - covariance matrix
        round(Rswm2$Matv,digits=5)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
