<!DOCTYPE html><html lang="en"><head><title>Help for package springer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {springer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#springer-package'><p>Sparse Group Variable Selection for Gene-Environment Interactions in the Longitudinal Study</p></a></li>
<li><a href='#cv.springer'><p>k-folds cross-validation for springer</p></a></li>
<li><a href='#dat'><p>simulated data for demonstrating the usage of springer</p></a></li>
<li><a href='#dmcp'><p>The first order derivative function of MCP (Minimax Concave Penalty)</p></a></li>
<li><a href='#penalty'><p>This function provides the penalty functions.  Users can choose one of the three penalties: sparse group MCP, group MCP and MCP.</p></a></li>
<li><a href='#print.springer'><p>print a springer result</p></a></li>
<li><a href='#reformat'><p>This function changes the format of the longitudinal data from wide format to long format</p></a></li>
<li><a href='#springer'><p>fit the model with given tuning parameters</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Sparse Group Variable Selection for Gene-Environment
Interactions in the Longitudinal Study</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.9</td>
</tr>
<tr>
<td>Author:</td>
<td>Fei Zhou, Yuwen Liu, Xi Lu, Jie Ren, Cen Wu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fei Zhou &lt;fei.zhou@outlook.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Recently, regularized variable selection has emerged as a powerful tool to identify and dissect gene-environment interactions. Nevertheless, in longitudinal studies with high dimensional genetic factors, regularization methods for G×E interactions have not been systematically developed. In this package, we provide the implementation of sparse group variable selection, based on both the quadratic inference function (QIF) and generalized estimating equation (GEE), to accommodate the bi-level selection for longitudinal G×E studies with high dimensional genomic features. Alternative methods conducting only the group or individual level selection have also been included. The core modules of the package have been developed in C++. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/feizhoustat/springer">https://github.com/feizhoustat/springer</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/feizhoustat/springer/issues">https://github.com/feizhoustat/springer/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS,Rcpp</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-01 07:33:20 UTC; feizh</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-01 07:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='springer-package'>Sparse Group Variable Selection for Gene-Environment Interactions in the Longitudinal Study</h2><span id='topic+springer-package'></span>

<h3>Description</h3>

<p>In this package, we provide a set of regularized variable selection methods tailored for longitudinal studies of gene- environment interactions. The proposed method conducts sparse group variable selection by accounting for bi-level sparsity. Specifically, the individual and group level penalties have been simultaneously imposed to identify important main and interaction effects under three working correlation structures (exchangeable , AR-1 and independence), based on either the quadratic inference function (QIF) or generalized estimating equation (GEE). In addition, only the individual or group level selection in the longitudinal setting can also be conducted using springer. In total, springer provides 18 (=3<code class="reqn">\times</code>3<code class="reqn">\times</code>2) methods. Among them, sparse group variable selection for longitudinal studies have been developed for the first time. Please read the Details below for how to configure the method used.
</p>


<h3>Details</h3>

<p>Users can flexibly choose the methods to fit the model by specifying the three arguments in the user interface <strong>springer()</strong>:
</p>

<table>
<tr>
 <td style="text-align: right;">
func: </td><td style="text-align: left;"> the framework to obtain the score equation.  Two choices are available: </td>
</tr>
<tr>
 <td style="text-align: right;"> </td><td style="text-align: left;"> "GEE" and "QIF". </td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
corr: </td><td style="text-align: left;"> working correlation.  Three choices are available: </td>
</tr>
<tr>
 <td style="text-align: right;"> </td><td style="text-align: left;"> "exchangeable", "AR-1" and "independence". </td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
structure: </td><td style="text-align: left;"> structural identification. Three choices are available: </td>
</tr>
<tr>
 <td style="text-align: right;"> </td><td style="text-align: left;"> "bilevel", "group" and "individual".
</td>
</tr>

</table>

<p>The function springer() returns a springer object that contains the estimated coefficients.
</p>


<h3>References</h3>

<p>Zhou, F., Liu, Y., Ren, J., Wang, W., and Wu, C. (2023). Springer: An R package for bi-level variable selection of high-dimensional longitudinal data. 
<em>Frontiers in Genetics</em>, 14, 1088223 <a href="https://doi.org/10.3389/fgene.2023.1088223">doi:10.3389/fgene.2023.1088223</a>
</p>
<p>Zhou, F., Lu, X., Ren, J., Fan, K., Ma, S. and Wu, C. (2022). Sparse Group Variable Selection for Gene-Environment Interactions in the Longitudinal Study.
<em>Genetic Epidemiology</em>, 46(5-6), 317-340 <a href="https://doi.org/10.1002/gepi.22461">doi:10.1002/gepi.22461</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2021). Gene-Environment Interaction: a Variable Selection Perspective.
<em>Epistasis: Methods and Protocols, Springer US</em> <a href="https://doi.org/10.1007/978-1-0716-0947-7_13">doi:10.1007/978-1-0716-0947-7_13</a>
</p>
<p>Zhou, F., Ren, J.,  Li, G., Jiang, Y., Li, X., Wang, W. and Wu, C. (2019). Penalized Variable Selection for Lipid-Environment Interactions in a Longitudinal Lipidomics Study.
<em>Genes</em>, 10(12), 1002 <a href="https://doi.org/10.3390/genes10121002">doi:10.3390/genes10121002</a>
</p>
<p>Zhou, F., Ren, J., Li, X., Wu, C. and Jiang, Y. (2019) interep: Interaction Analysis of Repeated Measure Data.
R package version 0.3.1. <a href="https://CRAN.R-project.org/package=interep">https://CRAN.R-project.org/package=interep</a>
</p>
<p>Ren, J., Du, Y., Li, S., Ma, S., Jiang, Y. and Wu, C. (2019). Robust network-based regularization and variable selection for high-dimensional genomic data in cancer prognosis.
<em>Genetic epidemiology</em>, 43(3), 276-291 <a href="https://doi.org/10.1002/gepi.22194">doi:10.1002/gepi.22194</a>
</p>
<p>Wu, C., Zhang, Q., Jiang, Y. and Ma, S. (2018). Robust network-based analysis of the associations between (epi) genetic measurements.
<em>Journal of multivariate analysis</em>, 168, 119-130 <a href="https://doi.org/10.1016/j.jmva.2018.06.009">doi:10.1016/j.jmva.2018.06.009</a>
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y. and Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach accounting for hierarchical structures.
<em>Statistics in Medicine</em>, 37:437-456 <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2018). Additive varying-coefficient model for nonlinear gene-environment interactions.
<em>Statistical Applications in Genetics and Molecular Biology</em>, 17(2) <a href="https://doi.org/10.1515/sagmb-2017-0008">doi:10.1515/sagmb-2017-0008</a>
</p>
<p>Ren, J., He, T., Li, Y., Liu, S., Du, Y., Jiang, Y. and Wu, C. (2017). Network-based regularization for high dimensional SNP data in the case-control study of Type 2 diabetes.
<em>BMC genetics</em>, 18(1), 44 <a href="https://doi.org/10.1186/s12863-017-0495-5">doi:10.1186/s12863-017-0495-5</a>
</p>
<p>Wu, C., Shi, X., Cui, Y. and Ma, S. (2015). A penalized robust semiparametric approach for gene-environment interactions.
<em>Statistics in Medicine</em>, 34 (30): 4016-4030 <a href="https://doi.org/10.1002/sim.6609">doi:10.1002/sim.6609</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene-environment interactions under a multi-response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988-4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Wu, C. and Cui, Y. (2014). Boosting signals in gene-based association studies via efficient SNP selection.
<em>Briefings in bioinformatics</em>, 15(2), 279-291 <a href="https://doi.org/10.1093/bib/bbs087">doi:10.1093/bib/bbs087</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<em>Technical Report. Michigan State University.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+springer">springer</a></code>
</p>

<hr>
<h2 id='cv.springer'>k-folds cross-validation for springer</h2><span id='topic+cv.springer'></span>

<h3>Description</h3>

<p>This function conducts k-fold cross-validation for springer and returns the optimal values of the tuning parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv.springer(
  clin = NULL,
  e,
  g,
  y,
  beta0,
  lambda1,
  lambda2,
  nfolds,
  func,
  corr,
  structure,
  maxits = 30,
  tol = 0.001
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cv.springer_+3A_clin">clin</code></td>
<td>
<p>a matrix of clinical covariates. The default value is NULL. Whether to include the clinical covariates is decided by user.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_e">e</code></td>
<td>
<p>a matrix of environment factors.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_g">g</code></td>
<td>
<p>a matrix of genetic factors.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_y">y</code></td>
<td>
<p>the longitudinal response.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_beta0">beta0</code></td>
<td>
<p>the initial value for the coefficient vector.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_lambda1">lambda1</code></td>
<td>
<p>a user-supplied sequence of <code class="reqn">\lambda_{1}</code> values, which serves as a tuning parameter for the individual-level penalty.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_lambda2">lambda2</code></td>
<td>
<p>a user-supplied sequence of <code class="reqn">\lambda_{2}</code> values, which serves as a tuning parameter for the group-level penalty.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_nfolds">nfolds</code></td>
<td>
<p>the number of folds for cross-validation.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_func">func</code></td>
<td>
<p>the framework to obtain the score equation.  Two choices are available: &quot;GEE&quot; and &quot;QIF&quot;.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_corr">corr</code></td>
<td>
<p>the working correlation structure adopted in the estimation algorithm. The springer provides three choices for the
working correlation structure: exchangeable, AR-1,and independence.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_structure">structure</code></td>
<td>
<p>Three choices are available for structured variable selection. &quot;bilevel&quot; for sparse-group selection on both group-level and individual-level. &quot;group&quot; for selection on group-level only. &quot;individual&quot; for selection on individual-level only.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_maxits">maxits</code></td>
<td>
<p>the maximum number of iterations that is used in the estimation algorithm. The default value is 30.</p>
</td></tr>
<tr><td><code id="cv.springer_+3A_tol">tol</code></td>
<td>
<p>The tolerance level. Coefficients with absolute values that are smaller than the tolerance level will be set to zero. The adhoc value can be chosen as 0.001.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For bi-level sparse group selection, cv.springer returns two optimal tuning parameters,
<code class="reqn">\lambda_{1}</code> and <code class="reqn">\lambda_{2}</code>; for group-level selection, this function returns the optimal <code class="reqn">\lambda_{2}</code> with <code class="reqn">\lambda_{1}</code>=0;
for individual-level selection, this function returns the optimal <code class="reqn">\lambda_{1}</code> with <code class="reqn">\lambda_{2}</code>=0.
</p>


<h3>Value</h3>

<p>an object of class &quot;cv.springer&quot; is returned, with is a list with components below:
</p>
<table role = "presentation">
<tr><td><code>lam1</code></td>
<td>
<p>the optimal <code class="reqn">\lambda_{1}</code>.</p>
</td></tr>
<tr><td><code>lam2</code></td>
<td>
<p>the optimal <code class="reqn">\lambda_{2}</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='dat'>simulated data for demonstrating the usage of springer</h2><span id='topic+dat'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the usage of springer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dat")
</code></pre>


<h3>Format</h3>

<p>The dat file consists of five components: e, g, y, clin and coeff. The coefficients are the true values of parameters used for generating Y.
</p>


<h3>Details</h3>

<p><strong>The data model for generating Y</strong>
</p>
<p>Consider a longitudinal case study with <code class="reqn">n</code> subjects and <code class="reqn">k_i</code> measurements over time for the <code class="reqn">i</code>th subject (<code class="reqn">i=1,\ldots,n</code>).
Let <code class="reqn">Y_{ij}</code> be the response of the <code class="reqn">j</code>th observation for the <code class="reqn">i</code>th subject (<code class="reqn">i=1,\ldots,n</code>, <code class="reqn">j=1,\ldots,k_i</code>),
<code class="reqn">X_{ij}=(X_{ij1},...,X_{ijp})^\top</code> be a <code class="reqn">p</code>-dimensional vector of covariates denoting <code class="reqn">p</code> genetic factors, <code class="reqn">E_{ij}=(E_{ij1},...,E_{ijq})^\top</code>
be a <code class="reqn">q</code>-dimensional environmental factor and <code class="reqn">Clin_{ij}=(Clin_{ij1},...,Clin_{ijt})^\top</code> be a <code class="reqn">t</code>-dimensional clinical factor.  There is time dependence among measurements on the same subject, but we assume that the measurements
between different subjects are independent.  The model we used for hierarchical variable selection for gene&ndash;environment interactions is given as:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij}= \alpha_0 + \sum_{m=1}^{t}\theta_m Clin_{ijm} + \sum_{u=1}^{q}\alpha_u E_{iju} + \sum_{v=1}^{p}(\gamma_v X_{ijv} + \sum_{u=1}^{q}h_{uv} E_{iju} X_{ijv})+\epsilon_{ij},</code>
</p>

<p>where <code class="reqn">\alpha_{0}</code> is the intercept and the marginal density of <code class="reqn">Y_{ij}</code> belongs to a canonical exponential family defined in Liang and Zeger (1986).
Define <code class="reqn">\eta_v=(\gamma_v, h_{1v}, ..., h_{qv})^\top</code>, which is a vector of length q+1 and <code class="reqn">Z_{ijv}=(X_{ijv}, E_{ij1}X_{ijv}, ..., E_{ijq}X_{ijv})^\top</code>,
which contains the main genetic effect of the <code class="reqn">v</code>th SNP from the <code class="reqn">j</code>th measurement on the <code class="reqn">i</code>th subject and its interactions with all the <code class="reqn">q</code>
environmental factors. The model can be written as:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij}= \alpha_0 + \sum_{m=1}^{t}\theta_m Clin_{ijm} + \sum_{u=1}^{q}\alpha_u E_{iju} + \sum_{v=1}^{p}\eta_v^\top Z_{ijv}+\epsilon_{ij},</code>
</p>

<p>where <code class="reqn">Z_{ijv}</code> is the <code class="reqn">v</code>th genetic factor and its interactions with the <code class="reqn">q</code> environment factors for the <code class="reqn">j</code>th measurement on the <code class="reqn">i</code>th subject,
and <code class="reqn">\eta_{v}</code> is the corresponding coefficient vector of length <code class="reqn">1+q</code>. The random error <code class="reqn">\epsilon_{i}=(\epsilon_{i1},...,\epsilon_{ik_i})^{T}</code>, which is assumed to follow a multivariate normal distribution with <code class="reqn">\Sigma_i</code>
as the covariance matrix for the repeated measurements of the <code class="reqn">ith</code> subject among the <code class="reqn">k_i</code> time points.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+springer">springer</a></code>
</p>

<hr>
<h2 id='dmcp'>The first order derivative function of MCP (Minimax Concave Penalty)</h2><span id='topic+dmcp'></span>

<h3>Description</h3>

<p>The first order derivative function of MCP (Minimax Concave Penalty)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dmcp(theta, lambda, gamma = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dmcp_+3A_theta">theta</code></td>
<td>
<p>a coefficient vector.</p>
</td></tr>
<tr><td><code id="dmcp_+3A_lambda">lambda</code></td>
<td>
<p>the tuning parameter.</p>
</td></tr>
<tr><td><code id="dmcp_+3A_gamma">gamma</code></td>
<td>
<p>the regularization parameter for MCP (Minimax Concave Penalty).
It balances the unbiasedness and concavity of MCP.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The regularization parameter <code class="reqn">\gamma</code> for MCP should be obtained via a data-driven approach in a rigorous way.
Among the published studies, it is suggested to check several choices, such as 1.4, 3, 4.2, 5.8, 6.9, and 10, then fix the value.
We examined this sequence in our study and found that the results are not sensitive to the choice of value for <code class="reqn">\gamma</code>.  Therefore,
we set the value to 3. To be prudent, other values should also be examined in practice.
</p>


<h3>Value</h3>

<p>the first order derivative of the MCP function.
</p>


<h3>References</h3>

<p>Ren, J., Du, Y., Li, S., Ma, S., Jiang, Y. and Wu, C. (2019). Robust network-based regularization and variable selection for high-dimensional genomic data in cancer prognosis.
<em>Genetic epidemiology</em>, 43(3), 276-291 <a href="https://doi.org/10.1002/gepi.22194">doi:10.1002/gepi.22194</a>
</p>
<p>Wu, C., Zhang, Q., Jiang, Y. and Ma, S. (2018). Robust network-based analysis of the associations between (epi) genetic measurements.
<em>Journal of multivariate analysis</em>, 168, 119-130 <a href="https://doi.org/10.1016/j.jmva.2018.06.009">doi:10.1016/j.jmva.2018.06.009</a>
</p>
<p>Ren, J., He, T., Li, Y., Liu, S., Du, Y., Jiang, Y. and Wu, C. (2017). Network-based regularization for high dimensional SNP data in the case-control study of Type 2 diabetes.
<em>BMC genetics</em>, 18(1), 44 <a href="https://doi.org/10.1186/s12863-017-0495-5">doi:10.1186/s12863-017-0495-5</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>theta=runif(30,-4,4)
lambda=1
dmcp(theta,lambda,gamma=3)
</code></pre>

<hr>
<h2 id='penalty'>This function provides the penalty functions.  Users can choose one of the three penalties: sparse group MCP, group MCP and MCP.</h2><span id='topic+penalty'></span>

<h3>Description</h3>

<p>This function provides the penalty functions.  Users can choose one of the three penalties: sparse group MCP, group MCP and MCP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penalty(x, n, t, p, q, beta, lam1, structure, p1, lam2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="penalty_+3A_x">x</code></td>
<td>
<p>the matrix of predictors, consisting of the clinical covariates, environmental factors, genetic factors and gene-environment interactions.</p>
</td></tr>
<tr><td><code id="penalty_+3A_n">n</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code id="penalty_+3A_t">t</code></td>
<td>
<p>the number of clinical covariates.</p>
</td></tr>
<tr><td><code id="penalty_+3A_p">p</code></td>
<td>
<p>the number of predictors, which consists of the clinical covariates, environmental factors, genetic factors and gene-environment interactions.</p>
</td></tr>
<tr><td><code id="penalty_+3A_q">q</code></td>
<td>
<p>the number of environment factors.</p>
</td></tr>
<tr><td><code id="penalty_+3A_beta">beta</code></td>
<td>
<p>the coefficient vector.</p>
</td></tr>
<tr><td><code id="penalty_+3A_lam1">lam1</code></td>
<td>
<p>the tuning parameter <code class="reqn">\lambda_1</code> for individual-level penalty.</p>
</td></tr>
<tr><td><code id="penalty_+3A_structure">structure</code></td>
<td>
<p>Three choices are available for structured variable selection. &quot;bilevel&quot; for sparse-group selection on both group-level and individual-level. &quot;group&quot; for selection on group-level only. &quot;individual&quot; for selection on individual-level only.</p>
</td></tr>
<tr><td><code id="penalty_+3A_p1">p1</code></td>
<td>
<p>the number of genetic factors.</p>
</td></tr>
<tr><td><code id="penalty_+3A_lam2">lam2</code></td>
<td>
<p>the tuning parameter <code class="reqn">\lambda_2</code> for group-level penalty.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When structure=&quot;bilevel&quot;, sparse group MCP is adopted and variable selection for longitudinal data including both genetic main effects and gene-environment interactions will be conducted on both individual and group levels (bi-level selection):
</p>

<ul>
<li> <p><strong>Group-level selection:</strong> If the <code class="reqn">v</code>th genetic factor has any effect at all (associated with the response or not) can be determined by whether <code class="reqn">||\eta_{v}||_{2}=0</code>.
</p>
</li>
<li> <p><strong>Individual-level selection:</strong> whether the <code class="reqn">v</code>th genetic variant has main effect, G<code class="reqn">\times</code>E interaction or both can be determined by the nonzero componet.
</p>
</li></ul>

<p>If structure=&quot;group&quot;, group MCP will be used and only group-level selection will be conducted on <code class="reqn">||\eta_{v}||_{2}</code>; if structure=&quot;individual&quot;, MCP will be adopted and only individual-level selection will be conducted on each <code class="reqn">\eta_{vu}</code>, (<code class="reqn">u=1,\ldots,q</code>).
</p>
<p>The minimax concave penalty (MCP) is adopted as the baseline penalty function in the springer package. Methods based on other popular choices, such as SCAD and LASSO, will be examined in the future.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>H</code></td>
<td>
<p>the penalty function.</p>
</td></tr>
</table>

<hr>
<h2 id='print.springer'>print a springer result</h2><span id='topic+print.springer'></span>

<h3>Description</h3>

<p>Print a springer result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'springer'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.springer_+3A_x">x</code></td>
<td>
<p>springer result</p>
</td></tr>
<tr><td><code id="print.springer_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.springer_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+springer">springer</a></code>
</p>

<hr>
<h2 id='reformat'>This function changes the format of the longitudinal data from wide format to long format</h2><span id='topic+reformat'></span>

<h3>Description</h3>

<p>This function changes the format of the longitudinal data from wide format to long format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reformat(k, y, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reformat_+3A_k">k</code></td>
<td>
<p>the number of repeated measurements/time points.</p>
</td></tr>
<tr><td><code id="reformat_+3A_y">y</code></td>
<td>
<p>the longitudinal response.</p>
</td></tr>
<tr><td><code id="reformat_+3A_x">x</code></td>
<td>
<p>a matrix of predictors, consisting of clinical covariates, genetic and environment factors, as well as gene-environment interactions.</p>
</td></tr>
</table>

<hr>
<h2 id='springer'>fit the model with given tuning parameters</h2><span id='topic+springer'></span>

<h3>Description</h3>

<p>This function performs penalized variable selection for longitudinal data based on generalized estimating equation (GEE) or quadratic inference functions (QIF) with a given value of lambda.
Typical usage is to first obtain the optimal lambda using cross validation, then provide it to the springer function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>springer(
  clin = NULL,
  e,
  g,
  y,
  beta0,
  func,
  corr,
  structure,
  lam1,
  lam2,
  maxits = 30,
  tol = 0.001
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="springer_+3A_clin">clin</code></td>
<td>
<p>a matrix of clinical covariates. The default value is NULL. Whether to include the clinical covariates is decided by user.</p>
</td></tr>
<tr><td><code id="springer_+3A_e">e</code></td>
<td>
<p>a matrix of environment factors.</p>
</td></tr>
<tr><td><code id="springer_+3A_g">g</code></td>
<td>
<p>a matrix of genetic factors.</p>
</td></tr>
<tr><td><code id="springer_+3A_y">y</code></td>
<td>
<p>the longitudinal response.</p>
</td></tr>
<tr><td><code id="springer_+3A_beta0">beta0</code></td>
<td>
<p>the initial coefficient vector</p>
</td></tr>
<tr><td><code id="springer_+3A_func">func</code></td>
<td>
<p>the framework to obtain the score equation.  Two choices are available: &quot;GEE&quot; and &quot;QIF&quot;.</p>
</td></tr>
<tr><td><code id="springer_+3A_corr">corr</code></td>
<td>
<p>the working correlation structure adopted in the estimation algorithm. The springer provides three choices for the
working correlation structure: exchangeable, AR-1,and independence.</p>
</td></tr>
<tr><td><code id="springer_+3A_structure">structure</code></td>
<td>
<p>Three choices are available for structured variable selection. &quot;bilevel&quot; for sparse-group selection on both group-level and individual-level. &quot;group&quot; for selection on group-level only. &quot;individual&quot; for selection on individual-level only.</p>
</td></tr>
<tr><td><code id="springer_+3A_lam1">lam1</code></td>
<td>
<p>the tuning parameter <code class="reqn">\lambda_1</code> for individual-level penalty applied to genetic factors.</p>
</td></tr>
<tr><td><code id="springer_+3A_lam2">lam2</code></td>
<td>
<p>the tuning parameter <code class="reqn">\lambda_2</code> for group-level penalty applied to gene-environment interactions.</p>
</td></tr>
<tr><td><code id="springer_+3A_maxits">maxits</code></td>
<td>
<p>the maximum number of iterations that is used in the estimation algorithm. The default value is 30.</p>
</td></tr>
<tr><td><code id="springer_+3A_tol">tol</code></td>
<td>
<p>The tolerance level. Coefficients with absolute values that are smaller than the tolerance level will be set to zero. The adhoc value can be chosen as 0.001.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Look back to the data model described in &quot;<code><a href="#topic+dat">dat</a></code>&quot;:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij}= \alpha_0 + \sum_{m=1}^{t}\theta_m Clin_{ijm} + \sum_{u=1}^{q}\alpha_u E_{iju} + \sum_{v=1}^{p}\eta_v^\top Z_{ijv}+\epsilon_{ij},</code>
</p>

<p>where <code class="reqn">Z_{ijv}</code> contains the <code class="reqn">v</code>th genetic main factor and its interactions with the <code class="reqn">q</code> environment factors for the <code class="reqn">j</code>th measurement on the <code class="reqn">i</code>th subject
and <code class="reqn">\eta_{v}</code> is the corresponding coefficient vector of length <code class="reqn">1+q</code>.
</p>
<p>When structure=&quot;bilevel&quot;, variable selection for genetic main effects and gene-environment interactions under the longitudinal response will be conducted on both individual and group levels (bi-level selection):
</p>

<ul>
<li> <p><strong>Group-level selection:</strong> by determining whether <code class="reqn">||\eta_{v}||_{2}=0</code>, we can know if the <code class="reqn">v</code>th genetic variant has any effect at all.
</p>
</li>
<li> <p><strong>Individual-level selection:</strong> investigate whether the <code class="reqn">v</code>th genetic variant has main effect, G<code class="reqn">\times</code>E interaction or both, by determining which components in <code class="reqn">\eta_{v}</code> has non-zero values.
</p>
</li></ul>

<p>If structure=&quot;group&quot;, only group-level selection will be conducted on <code class="reqn">||\eta_{v}||_{2}</code>; if structure=&quot;individual&quot;, only individual-level selection will be conducted on each <code class="reqn">\eta_{vu}</code>, (<code class="reqn">u=1,\ldots,q</code>).
</p>
<p>This function also provides choices for the framework that is used.  If func=&quot;QIF&quot;, variable selection will be conducted within the quadratic inference functions framework; if func=&quot;GEE&quot;, variable selection will be
conducted within the generalized estimating equation framework.
</p>
<p>There are three options for the choice of the working correlation.  If corr=&quot;exchangeable&quot;, the exchangeable working correlation will be applied; if corr=&quot;AR-1&quot;, the AR-1 working correlation will be adopted; if corr=&quot;independence&quot;,
the independence working correlation will be used.
Please check the references for more details.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>coef</code></td>
<td>
<p>the coefficient vector.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data("dat")
##load the clinical covariates, environment factors, genetic factors and response from the
##"dat" file
clin=dat$clin
if(is.null(clin)){t=0} else{t=dim(clin)[2]}
e=dat$e
u=dim(e)[2]
g=dat$g
y=dat$y
##initial coefficient
beta0=dat$coef
##true nonzero coefficients
index=dat$index
beta = springer(clin=clin, e, g, y,beta0,func="GEE",corr="independence",structure="bilevel",
lam1=dat$lam1, lam2=dat$lam2,maxits=30,tol=0.01)
##only focus on the genetic main effects and gene-environment interactions
beta[1:(1+t+u)]=0
##effects that have nonzero coefficients
pos = which(beta != 0)
##true positive and false positive
tp = length(intersect(index, pos))
fp = length(pos) - tp
list(tp=tp, fp=fp)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
