<!DOCTYPE html><html><head><title>Help for package SQMtools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SQMtools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#combineSQM'><p>Combine several SQM objects</p></a></li>
<li><a href='#combineSQMlite'><p>Combine several SQM or SQMlite objects</p></a></li>
<li><a href='#exportKrona'><p>Export the taxonomy of a SQM object into a Krona Chart</p></a></li>
<li><a href='#exportPathway'><p>Export the functions of a SQM object into KEGG pathway maps</p></a></li>
<li><a href='#exportTable'><p>Export results in tabular format</p></a></li>
<li><a href='#Hadza'><p>Hadza hunter-gatherer gut metagenomes</p></a></li>
<li><a href='#loadSQM'><p>Load a SqueezeMeta project into R</p></a></li>
<li><a href='#loadSQMlite'><p>Load tables generated by <code>sqm2tables.py</code>, <code>sqmreads2tables.py</code> or <code>combine-sqm-tables.py</code> into R.</p></a></li>
<li><a href='#MGKOs'><p>Single Copy Phylogenetic Marker Genes from Sunagawa's group (KOs)</p></a></li>
<li><a href='#MGOGs'><p>Single Copy Phylogenetic Marker Genes from Sunagawa's group (OGs)</p></a></li>
<li><a href='#mostAbundant'><p>Get the N most abundant rows (or columns) from a numeric table</p></a></li>
<li><a href='#mostVariable'><p>Get the N most variable rows (or columns) from a numeric table</p></a></li>
<li><a href='#plotBars'><p>Plot a barplot using ggplot2</p></a></li>
<li><a href='#plotBins'><p>Barplot of the most abundant bins in a SQM object</p></a></li>
<li><a href='#plotFunctions'><p>Heatmap of the most abundant functions in a SQM object</p></a></li>
<li><a href='#plotHeatmap'><p>Plot a heatmap using ggplot2</p></a></li>
<li><a href='#plotTaxonomy'><p>Barplot of the most abundant taxa in a SQM object</p></a></li>
<li><a href='#RecA'><p>RecA/RadA recombinase</p></a></li>
<li><a href='#rowMaxs'><p>Return a vector with the row-wise maxima of a matrix or dataframe.</p></a></li>
<li><a href='#rowMins'><p>Return a vector with the row-wise minima of a matrix or dataframe.</p></a></li>
<li><a href='#seqvec2fasta'><p>Print a named vector of sequences as a fasta-formatted string</p></a></li>
<li><a href='#subsetBins'><p>Create a SQM object containing only the requested bins, and the contigs and ORFs contained in them.</p></a></li>
<li><a href='#subsetContigs'><p>Select contigs</p></a></li>
<li><a href='#subsetFun'><p>Filter results by function</p></a></li>
<li><a href='#subsetORFs'><p>Select ORFs</p></a></li>
<li><a href='#subsetRand'><p>Select random ORFs</p></a></li>
<li><a href='#subsetSamples'><p>Filter results by sample</p></a></li>
<li><a href='#subsetTax'><p>Filter results by taxonomy</p></a></li>
<li><a href='#summary.SQM'><p>summary method for class SQM</p></a></li>
<li><a href='#summary.SQMlite'><p>summary method for class SQMlite</p></a></li>
<li><a href='#USiCGs'><p>Universal Single-Copy Genes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Analyze Results Generated by the 'SqueezeMeta' Pipeline</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.3</td>
</tr>
<tr>
<td>Description:</td>
<td>'SqueezeMeta' is a versatile pipeline for the automated analysis of metagenomics/metatranscriptomics data (<a href="https://github.com/jtamames/SqueezeMeta">https://github.com/jtamames/SqueezeMeta</a>). This package provides functions loading 'SqueezeMeta' results into R, filtering them based on different criteria, and visualizing the results using basic plots. The 'SqueezeMeta' project (and any subsets of it generated by the different filtering functions) is parsed into a single object, whose different components (e.g. tables with the taxonomic or functional composition across samples, contig/gene abundance profiles) can be easily analyzed using other R packages such as 'vegan' or 'DESeq2'. The methods in this package are further described in Puente-Sánchez et al., (2020) &lt;<a href="https://doi.org/10.1186%2Fs12859-020-03703-2">doi:10.1186/s12859-020-03703-2</a>&gt;.</td>
</tr>
<tr>
<td>Author:</td>
<td>Fernando Puente-Sánchez [aut, cre],
  Natalia García-García [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fernando Puente-Sánchez &lt;fernando.puente.sanchez@slu.se&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>Imports:</td>
<td>reshape2, ggplot2, pathview, data.table</td>
</tr>
<tr>
<td>Suggests:</td>
<td>vegan, DESeq2</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jtamames/SqueezeMeta/issues">https://github.com/jtamames/SqueezeMeta/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jtamames/SqueezeMeta">https://github.com/jtamames/SqueezeMeta</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-15 13:50:40 UTC; fer</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-18 09:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='combineSQM'>Combine several SQM objects</h2><span id='topic+combineSQM'></span>

<h3>Description</h3>

<p>Combine an arbitrary number of SQM objects into a single SQM object. The input objects must be subsets of the same original SQM object (i.e. from the same SqueezeMeta run). For combining results from different runs please check <code><a href="#topic+combineSQMlite">combineSQMlite</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combineSQM(
  ...,
  tax_source = "orfs",
  trusted_functions_only = FALSE,
  ignore_unclassified_functions = FALSE,
  rescale_tpm = TRUE,
  rescale_copy_number = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combineSQM_+3A_...">...</code></td>
<td>
<p>an arbitrary number of SQM objects. Alternatively, a single list containing an arbitrary number of SQM objects.</p>
</td></tr>
<tr><td><code id="combineSQM_+3A_tax_source">tax_source</code></td>
<td>
<p>character. Features used for calculating aggregated abundances at the different taxonomic ranks. Either <code>"orfs"</code> or <code>"contigs"</code> (default <code>"orfs"</code>). If the objects being combined contain a subset of taxa or bins, this parameter can be set to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="combineSQM_+3A_trusted_functions_only">trusted_functions_only</code></td>
<td>
<p>logical. If <code>TRUE</code>, only highly trusted functional annotations (best hit + best average) will be considered when generating aggregated function tables. If <code>FALSE</code>, best hit annotations will be used (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="combineSQM_+3A_ignore_unclassified_functions">ignore_unclassified_functions</code></td>
<td>
<p>logical. If <code>FALSE</code>, ORFs with no functional classification will be aggregated together into an &quot;Unclassified&quot; category. If <code>TRUE</code>, they will be ignored (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="combineSQM_+3A_rescale_tpm">rescale_tpm</code></td>
<td>
<p>logical. If <code>TRUE</code>, TPMs for KEGGs, COGs, and PFAMs will be recalculated (so that the TPMs in the subset actually add up to 1 million). Otherwise, per-function TPMs will be calculated by aggregating the TPMs of the ORFs annotated with that function, and will thus keep the scaling present in the parent object (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="combineSQM_+3A_rescale_copy_number">rescale_copy_number</code></td>
<td>
<p>logical. If <code>TRUE</code>, copy numbers with be recalculated using the RecA/RadA coverages in the subset. Otherwise, RecA/RadA coverages will be taken from the parent object with the highest RecA/RadA coverages. By default it is set to <code>TRUE</code>, which means that the returned copy numbers will represent the average copy number per function <em>in the genomes of the selected bins or contigs</em>. If any SQM objects that are being combined contain a functional subset rather than a contig/bins subset, this parameter should be set to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A SQM object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsetFun">subsetFun</a></code>, <code><a href="#topic+subsetTax">subsetTax</a></code>, <code><a href="#topic+combineSQMlite">combineSQMlite</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
# Select Carbohydrate metabolism ORFs in Bacteroidetes,
#  and Amino acid metabolism ORFs in Proteobacteria
bact = subsetTax(Hadza, "phylum", "Bacteroidetes")
bact.carb = subsetFun(bact, "Carbohydrate metabolism")
proteo = subsetTax(Hadza, "phylum", "Proteobacteria")
proteo.amins = subsetFun(proteo, "Amino acid metabolism")
bact.carb_proteo.amins = combineSQM(bact.carb, proteo.amins, rescale_copy_number=FALSE)
</code></pre>

<hr>
<h2 id='combineSQMlite'>Combine several SQM or SQMlite objects</h2><span id='topic+combineSQMlite'></span>

<h3>Description</h3>

<p>Combine an arbitrary number of SQM or SQMlite objects into a single SQMlite object. This function accepts objects originating from different projects (i.e. different SqueezeMeta runs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combineSQMlite(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combineSQMlite_+3A_...">...</code></td>
<td>
<p>an arbitrary number of SQM or SQMlite objects. Alternatively, a single list containing an arbitrary number of SQMlite objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A SQMlite object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+combineSQM">combineSQM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(Hadza)
# Load data coming from a different run
other = loadSQMlite("/path/to/other/project/tables") # e.g. if the project was run using sqm_reads
# (We could also use loadSQM to load the data as long as the data comes from a SqueezeMeta run)
combined = combineSQMlite(Hadza, other)
# Now we can plot together the samples from Hadza and the second project
plotTaxonomy(combined, 'family')

## End(Not run)
</code></pre>

<hr>
<h2 id='exportKrona'>Export the taxonomy of a SQM object into a Krona Chart</h2><span id='topic+exportKrona'></span>

<h3>Description</h3>

<p>Generate a krona chart containing the full taxonomy from a SQM object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportKrona(SQM, output_name = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exportKrona_+3A_sqm">SQM</code></td>
<td>
<p>A SQM or SQMlite object.</p>
</td></tr>
<tr><td><code id="exportKrona_+3A_output_name">output_name</code></td>
<td>
<p>character. Name of the output file containing the Krona charts in html format (default <code>"&lt;project_name&gt;.krona.html")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Original code was kindly provided by Giuseppe D'Auria (dauria_giu@gva.es).
</p>


<h3>Value</h3>

<p>No return value, but a krona chart is produced in the current working directory.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotTaxonomy">plotTaxonomy</a></code> for plotting the most abundant taxa of a SQM object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Hadza)
# Check that kronatools is present.
ecode = system('ktImportText', ignore.stdout = TRUE, ignore.stderr = TRUE)
# If so, run.
if(ecode==0) { exportKrona(Hadza) }

</code></pre>

<hr>
<h2 id='exportPathway'>Export the functions of a SQM object into KEGG pathway maps</h2><span id='topic+exportPathway'></span>

<h3>Description</h3>

<p>This function is a wrapper for the pathview package (Luo <em>et al.</em>, 2017. <em>Nucleic acids research</em>, 45:W501-W508). It will generate annotated KEGG pathway maps showing which reactions are present in the different samples. It will also generate legends with the color scales for each sample in separate png files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportPathway(
  SQM,
  pathway_id,
  count = "tpm",
  samples = NULL,
  split_samples = FALSE,
  sample_colors = NULL,
  log_scale = FALSE,
  fold_change_groups = NULL,
  fold_change_colors = NULL,
  max_scale_value = NULL,
  color_bins = 10,
  output_suffix = "pathview"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exportPathway_+3A_sqm">SQM</code></td>
<td>
<p>A SQM or SQMlite object.</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_pathway_id">pathway_id</code></td>
<td>
<p>character. The five-number KEGG pathway identifier. A list of all pathway identifiers can be found in <a href="https://www.genome.jp/kegg/pathway.html">https://www.genome.jp/kegg/pathway.html</a>.</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_count">count</code></td>
<td>
<p>character. Either <code>"abund"</code> for raw abundances, <code>"percent"</code> for percentages, <code>"bases"</code> for raw base counts, <code>"tpm"</code> for TPM normalized values or <code>"copy_number"</code> for copy numbers (default <code>"tpm"</code>). Note that a given count type might not available in this object (e.g. TPM or copy number in SQMlite objects originating from a SQM reads project).</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_samples">samples</code></td>
<td>
<p>character. An optional vector with the names of the samples to export. If absent, all samples will be exported (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_split_samples">split_samples</code></td>
<td>
<p>logical. Generate a different output file for each sample (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_sample_colors">sample_colors</code></td>
<td>
<p>character. An optional vector with the plotting colors for each sample (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_log_scale">log_scale</code></td>
<td>
<p>logical. Use a base 10 logarithmic transformation for the color scale. Will have no effect if <code>fold_change_groups</code> is provided (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_fold_change_groups">fold_change_groups</code></td>
<td>
<p>list. An optional list containing two vectors of samples. If provided, the function will generate a single plot displaying the log2 fold-change between the average abundances of both groups of samples ( log(second group / first group) ) (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_fold_change_colors">fold_change_colors</code></td>
<td>
<p>character. An optional vector with the plotting colors of both groups in the fold-change plot. Will be ignored if <code>fold_change_group</code> is not provided.</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_max_scale_value">max_scale_value</code></td>
<td>
<p>numeric. Maximum value to include in the color scale. By default it is the maximum value in the selected samples (if plotting abundances in samples) or the maximum absolute log2 fold-change (if plotting fold changes) (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_color_bins">color_bins</code></td>
<td>
<p>numeric. Number of bins used to generate the gradient in the color scale (default <code>10</code>).</p>
</td></tr>
<tr><td><code id="exportPathway_+3A_output_suffix">output_suffix</code></td>
<td>
<p>character. Suffix to be added to the output files (default <code>"pathview"</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, but Pathview figures are produced in the current working directory.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotFunctions">plotFunctions</a></code> for plotting the most functions taxa of a SQM object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Hadza)
exportPathway(Hadza, "00910", count = 'copy_number', 
              output_suffix = "nitrogen_metabolism", sample_colors = c("red", "blue"))
exportPathway(Hadza, "00250", count = 'tpm', 
              output_suffix = "ala_asp_glu_metabolism_FoldChange", 
              fold_change_groups = list(c("H1"), c("H12")), max_scale_value=2)

</code></pre>

<hr>
<h2 id='exportTable'>Export results in tabular format</h2><span id='topic+exportTable'></span>

<h3>Description</h3>

<p>This function is a wrapper for R's write.table function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportTable(table, output_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exportTable_+3A_table">table</code></td>
<td>
<p>vector, matrix or data.frame. The table to be written.</p>
</td></tr>
<tr><td><code id="exportTable_+3A_output_name">output_name</code></td>
<td>
<p>character. Name of the output file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, but a table is produced in the current working directory.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Hadza)
Hadza.iron = subsetFun(Hadza, "iron")
# Write the taxonomic distribution at the genus level of all the genes related to iron.
exportTable(Hadza.iron$taxa$genus$percent, "Hadza.ironGenes.genus.tsv")
# Now write the distribution of the different iron-related COGs
#  (Clusters of Orthologous Groups) across samples.
exportTable(Hadza.iron$functions$COG$tpm, "Hadza.ironGenes.COG.tsv")
# Now write all the information contained in the ORF table.
exportTable(Hadza.iron$orfs$table, "Hadza.ironGenes.orftable.tsv")

</code></pre>

<hr>
<h2 id='Hadza'>Hadza hunter-gatherer gut metagenomes</h2><span id='topic+Hadza'></span>

<h3>Description</h3>

<p>Subset of two bins (and the associated contigs and genes)
generated by running SqueezeMeta on two gut metagenomic samples
obtained from two hunter-gatherers of the Hadza ethnic group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Hadza)
</code></pre>


<h3>Format</h3>

<p>A SQM object; see <code><a href="#topic+loadSQM">loadSQM</a></code>.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR1927149">SRR1927149</a>, <a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR1929485">SRR1929485</a>.
</p>


<h3>References</h3>

<p>Rampelli <em>et al.</em>, 2015. Metagenome Sequencing of the Hadza
Hunter-Gatherer Gut Microbiota. <em>Curr. biol.</em> <b>25</b>:1682-93
(<a href="https://pubmed.ncbi.nlm.nih.gov/25981789/">PubMed</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
plotTaxonomy(Hadza, "genus", rescale=TRUE)
plotFunctions(Hadza, "COG")

</code></pre>

<hr>
<h2 id='loadSQM'>Load a SqueezeMeta project into R</h2><span id='topic+loadSQM'></span>

<h3>Description</h3>

<p>This function takes the path to a project directory generated by <a href="https://github.com/jtamames/SqueezeMeta">SqueezeMeta</a> (whose name is specified in the <code>-p</code> parameter of the SqueezeMeta.pl script) and parses the results into a SQM object. Alternatively, it can load the project data from a zip file produced by <code>sqm2zip.py</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadSQM(
  project_path,
  tax_mode = "prokfilter",
  trusted_functions_only = FALSE,
  engine = "data.table"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadSQM_+3A_project_path">project_path</code></td>
<td>
<p>character, project directory generated by SqueezeMeta, or zip file generated by <code>sqm2zip.py</code>.</p>
</td></tr>
<tr><td><code id="loadSQM_+3A_tax_mode">tax_mode</code></td>
<td>
<p>character, which taxonomic classification should be loaded? SqueezeMeta applies the identity thresholds described in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4005636/">Luo <em>et al.</em>, 2014</a>. Use <code>allfilter</code> for applying the minimum identity threshold to all taxa, <code>prokfilter</code> for applying the threshold to Bacteria and Archaea, but not to Eukaryotes, and <code>nofilter</code> for applying no thresholds at all (default <code>prokfilter</code>).</p>
</td></tr>
<tr><td><code id="loadSQM_+3A_trusted_functions_only">trusted_functions_only</code></td>
<td>
<p>logical. If <code>TRUE</code>, only highly trusted functional annotations (best hit + best average) will be considered when generating aggregated function tables. If <code>FALSE</code>, best hit annotations will be used (default <code>FALSE</code>). Will only have an effect if the <code>project_dir/results/tables</code> is not already present.</p>
</td></tr>
<tr><td><code id="loadSQM_+3A_engine">engine</code></td>
<td>
<p>character. Engine used to load the ORFs and contigs tables. Either <code>data.frame</code> or <code>data.table</code> (significantly faster if your project is large). Default <code>data.table</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQM object containing the parsed project.
</p>


<h3>Prerequisites</h3>


<p>Run <a href="https://github.com/jtamames/SqueezeMeta">SqueezeMeta</a>! An example call for running it would be:
</p>
<p><code>/path/to/SqueezeMeta/scripts/SqueezeMeta.pl</code><br />
<code>-m coassembly -f fastq_dir -s samples_file -p project_dir</code>

</p>


<h3>The SQM object structure</h3>


<p>The SQM object is a nested list which contains the following information:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>lvl1</b>         </td><td style="text-align: left;"> <b>lvl2</b>               </td><td style="text-align: left;"> <b>lvl3</b>          </td><td style="text-align: left;"> <b>type</b>             </td><td style="text-align: left;"> <b>rows/names</b> </td><td style="text-align: left;"> <b>columns</b> </td><td style="text-align: left;"> <b>data</b>        </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$orfs</b>        </td><td style="text-align: left;"> <b>$table</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>dataframe</em>        </td><td style="text-align: left;"> orfs              </td><td style="text-align: left;"> misc. data     </td><td style="text-align: left;"> misc. data         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$abund</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orfs              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$bases</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orfs              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$cov</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orfs              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> coverages          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$cpm</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orfs              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> covs. / 10^6 reads </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tpm</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orfs              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> tpm                </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$seqs</b>              </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> orfs              </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> sequences          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tax</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character matrix</em> </td><td style="text-align: left;"> orfs              </td><td style="text-align: left;"> tax. ranks     </td><td style="text-align: left;"> taxonomy           </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$contigs</b>     </td><td style="text-align: left;"> <b>$table</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>dataframe</em>        </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> misc. data     </td><td style="text-align: left;"> misc. data         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$abund</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$bases</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$cov</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> coverages          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$cpm</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> covs. / 10^6 reads </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tpm</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> tpm                </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$seqs</b>              </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> sequences          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tax</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character matrix</em> </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> tax. ranks     </td><td style="text-align: left;"> taxonomies         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$bins</b>              </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character matrix</em> </td><td style="text-align: left;"> contigs           </td><td style="text-align: left;"> bin. methods   </td><td style="text-align: left;"> bins               </td>
</tr>
<tr>
 <td style="text-align: left;">
$bins               </td><td style="text-align: left;"> <b>$table</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>dataframe</em>        </td><td style="text-align: left;"> bins              </td><td style="text-align: left;"> misc. data     </td><td style="text-align: left;"> misc. data         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$length</b>            </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric vector</em>   </td><td style="text-align: left;"> bins              </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> length             </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$abund</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> bins              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$percent</b>           </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> bins              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$bases</b>             </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> bins              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$cov</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> bins              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> coverages          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$cpm</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> bins              </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> covs. / 10^6 reads </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tax</b>               </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character matrix</em> </td><td style="text-align: left;"> bins              </td><td style="text-align: left;"> tax. ranks     </td><td style="text-align: left;"> taxonomy           </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$taxa</b>        </td><td style="text-align: left;"> <b>$superkingdom</b>      </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> superkingdoms     </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> superkingdoms     </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$phylum</b>            </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> phyla             </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> phyla             </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$class</b>             </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> classes           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> classes           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$order</b>             </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orders            </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orders            </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$family</b>            </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> families          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> families          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$genus</b>             </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> genera            </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> genera            </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$species</b>           </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> species           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> species           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$functions</b>   </td><td style="text-align: left;"> <b>$KEGG</b>              </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$bases</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$cov</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> coverages          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$cpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> covs. / 10^6 reads </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$tpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> tpm                </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$copy_number</b>  </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> avg. copies        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$COG</b>               </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$bases</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$cov</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> coverages          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$cpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> covs. / 10^6 reads </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$tpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> tpm                </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$copy_number</b>  </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> avg. copies        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$PFAM</b>              </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$bases</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$cov</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> coverages          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$cpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> covs. / 10^6 reads </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$tpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> tpm                </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$copy_number</b>  </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> avg. copies        </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$total_reads</b> </td><td style="text-align: left;">                           </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric vector</em>   </td><td style="text-align: left;"> samples           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> total reads        </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$misc</b>        </td><td style="text-align: left;"> <b>$project_name</b>      </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> (empty)           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> project name       </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$samples</b>           </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> (empty)           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> samples            </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tax_names_long</b>    </td><td style="text-align: left;"> <b>$superkingdom</b> </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$phylum</b>       </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$class</b>        </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$order</b>        </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$family</b>       </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$genus</b>        </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$species</b>      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tax_names_short</b>   </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> full names        </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> short names        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$KEGG_names</b>        </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> KEGG names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$KEGG_paths</b>        </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> KEGG hiararchy     </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$COG_names</b>         </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> COG names          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$COG_paths</b>         </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> COG hierarchy      </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$ext_annot_sources</b> </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> external databases </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>If external databases for functional classification were provided to SqueezeMeta via the <code>-extdb</code> argument, the corresponding abundance (reads and bases), coverages, tpm and copy number profiles will be present in <code>SQM$functions</code> (e.g. results for the CAZy database would be present in <code>SQM$functions$CAZy</code>). Additionally, the extended names of the features present in the external database will be present in <code>SQM$misc</code> (e.g. <code>SQM$misc$CAZy_names</code>).

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## (outside R)
## Run SqueezeMeta on the test data.
 /path/to/SqueezeMeta/scripts/SqueezeMeta.pl -p Hadza -f raw -m coassembly -s test.samples
## Now go into R.
library(SQMtools)
Hadza = loadSQM("Hadza") # Where Hadza is the path to the SqueezeMeta output directory.

## End(Not run)

data(Hadza) # We will illustrate the structure of the SQM object on the test data
# Which are the ten most abundant KEGG IDs in our data?
topKEGG = names(sort(rowSums(Hadza$functions$KEGG$tpm), decreasing=TRUE))[1:11]
topKEGG = topKEGG[topKEGG!="Unclassified"]
# Which functions do those KEGG IDs represent?
Hadza$misc$KEGG_names[topKEGG]
# What is the relative abundance of the Gammaproteobacteria class across samples?
Hadza$taxa$class$percent["Gammaproteobacteria",]
# Which information is stored in the orf, contig and bin tables?
colnames(Hadza$orfs$table)
colnames(Hadza$contigs$table)
colnames(Hadza$bins$table)
# What is the GC content distribution of my metagenome?
boxplot(Hadza$contigs$table[,"GC perc"]) # Not weighted by contig length or abundance!
</code></pre>

<hr>
<h2 id='loadSQMlite'>Load tables generated by <code>sqm2tables.py</code>, <code>sqmreads2tables.py</code> or <code>combine-sqm-tables.py</code> into R.</h2><span id='topic+loadSQMlite'></span>

<h3>Description</h3>

<p>This function takes the path to the output directory generated by <code>sqm2tables.py</code>, <code>sqmreads2tables.py</code> or <code>combine-sqm-tables.py</code> a SQMlite object.
The SQMlite object will contain taxonomic and functional profiles, but no detailed information on ORFs, contigs or bins. However, it will also have a much smaller memory footprint.
A SQMlite object can be used for plotting and exporting, but it can not be subsetted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadSQMlite(tables_path, tax_mode = "allfilter")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadSQMlite_+3A_tables_path">tables_path</code></td>
<td>
<p>character, tables directory generated by <code>sqm2table.py</code>, <code>sqmreads2tables.py</code> or <code>combine-sqm-tables.py</code>.</p>
</td></tr>
<tr><td><code id="loadSQMlite_+3A_tax_mode">tax_mode</code></td>
<td>
<p>character, which taxonomic classification should be loaded? SqueezeMeta applies the identity thresholds described in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4005636/">Luo <em>et al.</em>, 2014</a>. Use <code>allfilter</code> for applying the minimum identity threshold to all taxa (default), <code>prokfilter</code> for applying the threshold to Bacteria and Archaea, but not to Eukaryotes, and <code>nofilter</code> for applying no thresholds at all.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQMlite object containing the parsed tables.
</p>


<h3>The SQMlite object structure</h3>


<p>The SQMlite object is a nested list which contains the following information:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>lvl1</b>         </td><td style="text-align: left;"> <b>lvl2</b>               </td><td style="text-align: left;"> <b>lvl3</b>          </td><td style="text-align: left;"> <b>type</b>             </td><td style="text-align: left;"> <b>rows/names</b> </td><td style="text-align: left;"> <b>columns</b> </td><td style="text-align: left;"> <b>data</b>        </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$taxa</b>        </td><td style="text-align: left;"> <b>$superkingdom</b>      </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> superkingdoms     </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> superkingdoms     </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$phylum</b>            </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> phyla             </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> phyla             </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$class</b>             </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> classes           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> classes           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$order</b>             </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orders            </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> orders            </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$family</b>            </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> families          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> families          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$genus</b>             </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> genera            </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> genera            </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$species</b>           </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> species           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$percent</b>      </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> species           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> percentages        </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$functions</b>   </td><td style="text-align: left;"> <b>$KEGG</b>              </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$bases</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$tpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> tpm                </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$copy_number</b>  </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> avg. copies        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$COG</b>               </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$bases</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$tpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> tpm                </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$copy_number</b>  </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> avg. copies        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$PFAM</b>              </td><td style="text-align: left;"> <b>$abund</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (reads) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$bases</b>        </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> abundances (bases) </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$tpm</b>          </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> tpm                </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$copy_number</b>  </td><td style="text-align: left;"> <em>numeric matrix</em>   </td><td style="text-align: left;"> PFAM ids          </td><td style="text-align: left;"> samples        </td><td style="text-align: left;"> avg. copies        </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$total_reads</b> </td><td style="text-align: left;">                           </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>numeric vector</em>   </td><td style="text-align: left;"> samples           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> total reads        </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>$misc</b>        </td><td style="text-align: left;"> <b>$project_name</b>      </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> (empty)           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> project name       </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$samples</b>           </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> (empty)           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> samples            </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tax_names_long</b>    </td><td style="text-align: left;"> <b>$superkingdom</b> </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$phylum</b>       </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$class</b>        </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$order</b>        </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$family</b>       </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$genus</b>        </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;">                           </td><td style="text-align: left;"> <b>$species</b>      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> short names       </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> full names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$tax_names_short</b>   </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> full names        </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> short names        </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$KEGG_names</b>        </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> KEGG names         </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$KEGG_paths</b>        </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> KEGG ids          </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> KEGG hiararchy     </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$COG_names</b>         </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> COG names          </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$COG_paths</b>         </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> COG ids           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> COG hierarchy      </td>
</tr>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: left;"> <b>$ext_annot_sources</b> </td><td style="text-align: left;">                      </td><td style="text-align: left;"> <em>character vector</em> </td><td style="text-align: left;"> (empty)           </td><td style="text-align: left;"> (n/a)          </td><td style="text-align: left;"> external databases </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>If external databases for functional classification were provided to SqueezeMeta or SqueezeMeta_reads via the <code>-extdb</code> argument, the corresponding abundance, tpm and copy number profiles will be present in <code>SQM$functions</code> (e.g. results for the CAZy database would be present in <code>SQM$functions$CAZy</code>). Additionally, the extended names of the features present in the external database will be present in <code>SQM$misc</code> (e.g. <code>SQM$misc$CAZy_names</code>). Note that results generated by SqueezeMeta_reads will contain only read abundances, but not bases, tpm or copy number estimations.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotBars">plotBars</a></code> and <code><a href="#topic+plotFunctions">plotFunctions</a></code> will plot the most abundant taxa and functions in a SQMlite object. <code><a href="#topic+exportKrona">exportKrona</a></code> will generate Krona charts reporting the taxonomy in a SQMlite object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## (outside R)
## Run SqueezeMeta on the test data.
/path/to/SqueezeMeta/scripts/SqueezeMeta.pl -p Hadza -f raw -m coassembly -s test.samples
## Generate the tabular outputs!
/path/to/SqueezeMeta/utils/sqm2tables.py Hadza Hadza/results/tables
## Now go into R.
library(SQMtools)
Hadza = loadSQMlite("Hadza/results/tables")
# Where Hadza is the path to the SqueezeMeta output directory.
# Note that this is not the whole SQM project, just the directory containing the tables.
# It would also work with tables generated by sqmreads2tables.py, or combine-sqm-tables.py
plotTaxonomy(Hadza)
plotFunctions(Hadza)
exportKrona(Hadza, 'myKronaTest.html')

## End(Not run)
</code></pre>

<hr>
<h2 id='MGKOs'>Single Copy Phylogenetic Marker Genes from Sunagawa's group (KOs)</h2><span id='topic+MGKOs'></span>

<h3>Description</h3>

<p>Lists of Single Copy Phylogenetic Marker Genes.
These are useful for transforming coverages or tpms into copy numbers.
This is an alternative way of normalizing data in order to be able to
compare functional profiles in samples with different sequencing depths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MGKOs)
</code></pre>


<h3>Format</h3>

<p>Character vector with the KEGG identifiers for 10 Single Copy Phylogenetic Marker Genes.
</p>


<h3>References</h3>

<p>Salazar, G <em>et al.</em> (2019). 
Gene Expression Changes and Community Turnover Differentially Shape the Global Ocean Metatranscriptome
<em>Cell</em> <b>179</b>:1068-1083.
(<a href="https://pubmed.ncbi.nlm.nih.gov/31730850/">PubMed</a>).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MGOGs">MGOGs</a></code> for an equivalent list using OGs instead of KOs; <code><a href="#topic+USiCGs">USiCGs</a></code> for an alternative set of single copy genes, and for examples on how to generate copy numbers.
</p>

<hr>
<h2 id='MGOGs'>Single Copy Phylogenetic Marker Genes from Sunagawa's group (OGs)</h2><span id='topic+MGOGs'></span>

<h3>Description</h3>

<p>Lists of Single Copy Phylogenetic Marker Genes.
These are useful for transforming coverages or tpms into copy numbers.
This is an alternative way of normalizing data in order to be able to
compare functional profiles in samples with different sequencing depths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MGOGs)
</code></pre>


<h3>Format</h3>

<p>Character vector with the COG identifiers for 10 Single Copy Phylogenetic Marker Genes.
</p>


<h3>References</h3>

<p>Salazar, G <em>et al.</em> (2019). 
Gene Expression Changes and Community Turnover Differentially Shape the Global Ocean Metatranscriptome
<em>Cell</em> <b>179</b>:1068-1083.
(<a href="https://pubmed.ncbi.nlm.nih.gov/31730850/">PubMed</a>).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MGKOs">MGKOs</a></code> for an equivalent list using KOs instead of OGs; <code><a href="#topic+USiCGs">USiCGs</a></code> for an alternative set of single copy genes, and for examples on how to generate copy numbers.
</p>

<hr>
<h2 id='mostAbundant'>Get the N most abundant rows (or columns) from a numeric table</h2><span id='topic+mostAbundant'></span>

<h3>Description</h3>

<p>Return a subset of an input matrix or data frame, containing only the N most abundant rows (or columns), sorted. Alternatively, a custom set of rows can be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mostAbundant(
  data,
  N = 10,
  items = NULL,
  others = FALSE,
  rescale = FALSE,
  bycol = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mostAbundant_+3A_data">data</code></td>
<td>
<p>numeric matrix or data frame</p>
</td></tr>
<tr><td><code id="mostAbundant_+3A_n">N</code></td>
<td>
<p>integer Number of rows to return (default <code>10</code>).</p>
</td></tr>
<tr><td><code id="mostAbundant_+3A_items">items</code></td>
<td>
<p>Character vector. Custom row names to return. If provided, it will override <code>N</code> (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="mostAbundant_+3A_others">others</code></td>
<td>
<p>logical. If <code>TRUE</code>, an extra row will be returned containing the aggregated abundances of the elements not selected with <code>N</code> or <code>items</code> (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="mostAbundant_+3A_rescale">rescale</code></td>
<td>
<p>logical. Scale result to percentages column-wise (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="mostAbundant_+3A_bycol">bycol</code></td>
<td>
<p>logical. Operate on columns instead of rows (default <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix or data frame (same as input) with the selected rows (or columns).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
Hadza.carb = subsetFun(Hadza, "Carbohydrate metabolism")
# Which are the 20 most abundant KEGG functions in the ORFs related to carbohydrate metabolism?
topCarb = mostAbundant(Hadza.carb$functions$KEGG$tpm, N=20)
# Now print them with nice names.
rownames(topCarb) = paste(rownames(topCarb),
                          Hadza.carb$misc$KEGG_names[rownames(topCarb)], sep="; ")
topCarb
# We can pass this to any R function.
heatmap(topCarb)
# But for convenience we provide wrappers for plotting ggplot2 heatmaps and barplots.
plotHeatmap(topCarb, label_y="TPM")
plotBars(topCarb, label_y="TPM")
</code></pre>

<hr>
<h2 id='mostVariable'>Get the N most variable rows (or columns) from a numeric table</h2><span id='topic+mostVariable'></span>

<h3>Description</h3>

<p>Return a subset of an input matrix or data frame, containing only the N most variable rows (or columns), sorted. Variability is calculated as the Coefficient of Variation (sd/mean).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mostVariable(data, N = 10, bycol = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mostVariable_+3A_data">data</code></td>
<td>
<p>numeric matrix or data frame</p>
</td></tr>
<tr><td><code id="mostVariable_+3A_n">N</code></td>
<td>
<p>integer Number of rows to return (default <code>10</code>).</p>
</td></tr>
<tr><td><code id="mostVariable_+3A_bycol">bycol</code></td>
<td>
<p>logical. Operate on columns instead of rows (default <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix or data frame (same as input) with the selected rows or columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
Hadza.carb = subsetFun(Hadza, "Carbohydrate metabolism")
# Which are the 20 most variable KEGG functions in the ORFs related to carbohydrate metabolism?
topCarb = mostVariable(Hadza.carb$functions$KEGG$tpm, N=20)
# Now print them with nice names
rownames(topCarb) = paste(rownames(topCarb),
                          Hadza.carb$misc$KEGG_names[rownames(topCarb)], sep="; ")
topCarb
# We can pass this to any R function
heatmap(topCarb)
# But for convenience we provide wrappers for plotting ggplot2 heatmaps and barplots
plotHeatmap(topCarb, label_y="TPM")
plotBars(topCarb, label_y="TPM")
</code></pre>

<hr>
<h2 id='plotBars'>Plot a barplot using ggplot2</h2><span id='topic+plotBars'></span>

<h3>Description</h3>

<p>Plot a ggplot2 barplot from a matrix or data frame. The data should be in tabular format (e.g. features in rows and samples in columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBars(
  data,
  label_x = "Samples",
  label_y = "Abundances",
  label_fill = "Features",
  color = NULL,
  base_size = 11,
  max_scale_value = NULL,
  metadata_groups = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBars_+3A_data">data</code></td>
<td>
<p>Numeric matrix or data frame.</p>
</td></tr>
<tr><td><code id="plotBars_+3A_label_x">label_x</code></td>
<td>
<p>character Label for the x axis (default <code>"Samples"</code>).</p>
</td></tr>
<tr><td><code id="plotBars_+3A_label_y">label_y</code></td>
<td>
<p>character Label for the y axis (default <code>"Abundances"</code>).</p>
</td></tr>
<tr><td><code id="plotBars_+3A_label_fill">label_fill</code></td>
<td>
<p>character Label for color categories (default <code>"Features"</code>).</p>
</td></tr>
<tr><td><code id="plotBars_+3A_color">color</code></td>
<td>
<p>Vector with custom colors for the different features. If empty, the default ggplot2 palette will be used (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotBars_+3A_base_size">base_size</code></td>
<td>
<p>numeric. Base font size (default <code>11</code>).</p>
</td></tr>
<tr><td><code id="plotBars_+3A_max_scale_value">max_scale_value</code></td>
<td>
<p>numeric. Maximum value to include in the y axis. By default it is handled automatically by ggplot2 (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotBars_+3A_metadata_groups">metadata_groups</code></td>
<td>
<p>list. Split the plot into groups defined by the user: list('G1' = c('sample1', sample2'), 'G2' = c('sample3', 'sample4')) default <code>NULL</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 plot object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotTaxonomy">plotTaxonomy</a></code> for plotting the most abundant taxa of a SQM object; <code><a href="#topic+plotHeatmap">plotHeatmap</a></code> for plotting a heatmap with arbitrary data; <code><a href="#topic+mostAbundant">mostAbundant</a></code> for selecting the most abundant rows in a dataframe or matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
sk = Hadza$taxa$superkingdom$abund
plotBars(sk, label_y = "Raw reads", label_fill = "Superkingdom")
</code></pre>

<hr>
<h2 id='plotBins'>Barplot of the most abundant bins in a SQM object</h2><span id='topic+plotBins'></span>

<h3>Description</h3>

<p>This function selects the most abundant bins across all samples in a SQM object and represents their abundances in a barplot. Alternatively, a custom set of bins can be represented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBins(
  SQM,
  count = "percent",
  N = 15,
  bins = NULL,
  others = TRUE,
  samples = NULL,
  ignore_unmapped = FALSE,
  ignore_nobin = FALSE,
  rescale = FALSE,
  color = NULL,
  base_size = 11,
  max_scale_value = NULL,
  metadata_groups = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBins_+3A_sqm">SQM</code></td>
<td>
<p>A SQM or a SQMlite object.</p>
</td></tr>
<tr><td><code id="plotBins_+3A_count">count</code></td>
<td>
<p>character. Either <code>"abund"</code> for raw abundances, <code>"percent"</code> for percentages, <code>"cov"</code> for coverages, or <code>"cpm"</code> for coverages per million reads (default <code>"percent"</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_n">N</code></td>
<td>
<p>integer Plot the <code>N</code> most abundant bins (default <code>15</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_bins">bins</code></td>
<td>
<p>character. Custom bins to plot. If provided, it will override <code>N</code> (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_others">others</code></td>
<td>
<p>logical. Collapse the abundances of least abundant bins, and include the result in the plot (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_samples">samples</code></td>
<td>
<p>character. Character vector with the names of the samples to include in the plot. Can also be used to plot the samples in a custom order. If not provided, all samples will be plotted (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_ignore_unmapped">ignore_unmapped</code></td>
<td>
<p>logical. Don't include unmapped reads in the plot (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_ignore_nobin">ignore_nobin</code></td>
<td>
<p>logical. Don't include reads which are not in a bin in the plot (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_rescale">rescale</code></td>
<td>
<p>logical. Re-scale results to percentages (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_color">color</code></td>
<td>
<p>Vector with custom colors for the different features. If empty, we will use our own hand-picked pallete if N&lt;=15, and the default ggplot2 palette otherwise (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_base_size">base_size</code></td>
<td>
<p>numeric. Base font size (default <code>11</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_max_scale_value">max_scale_value</code></td>
<td>
<p>numeric. Maximum value to include in the y axis. By default it is handled automatically by ggplot2 (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotBins_+3A_metadata_groups">metadata_groups</code></td>
<td>
<p>list. Split the plot into groups defined by the user: list('G1' = c('sample1', sample2'), 'G2' = c('sample3', 'sample4')) default <code>NULL</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 plot object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotBins">plotBins</a></code> for plotting the most abundant bins of a SQM object; <code><a href="#topic+plotBars">plotBars</a></code> and <code><a href="#topic+plotHeatmap">plotHeatmap</a></code> for plotting barplots or heatmaps with arbitrary data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
# Bins distribution.
plotBins(Hadza)
</code></pre>

<hr>
<h2 id='plotFunctions'>Heatmap of the most abundant functions in a SQM object</h2><span id='topic+plotFunctions'></span>

<h3>Description</h3>

<p>This function selects the most abundant functions across all samples in a SQM object and represents their abundances in a heatmap. Alternatively, a custom set of functions can be represented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotFunctions(
  SQM,
  fun_level = "KEGG",
  count = "tpm",
  N = 25,
  fun = NULL,
  samples = NULL,
  ignore_unmapped = TRUE,
  ignore_unclassified = TRUE,
  gradient_col = c("ghostwhite", "dodgerblue4"),
  base_size = 11,
  metadata_groups = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotFunctions_+3A_sqm">SQM</code></td>
<td>
<p>A SQM or SQMlite object.</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_fun_level">fun_level</code></td>
<td>
<p>character. Either <code>"KEGG"</code>, <code>"COG"</code>, <code>"PFAM"</code> or any other custom database used for annotation (default <code>"KEGG"</code>).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_count">count</code></td>
<td>
<p>character. Either <code>"abund"</code> for raw abundances, <code>"percent"</code> for percentages, <code>"bases"</code> for raw base counts, <code>"cpm"</code> for coverages per million reads, <code>"tpm"</code> for TPM normalized values or <code>"copy_number"</code> for copy numbers (default <code>"tpm"</code>). Note that a given count type might not available in this object (e.g. TPM or copy number in SQMlite objects originating from a SQM reads project).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_n">N</code></td>
<td>
<p>integer Plot the <code>N</code> most abundant functions (default <code>25</code>).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_fun">fun</code></td>
<td>
<p>character. Custom functions to plot. If provided, it will override <code>N</code> (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_samples">samples</code></td>
<td>
<p>character. Character vector with the names of the samples to include in the plot. Can also be used to plot the samples in a custom order. If not provided, all samples will be plotted (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_ignore_unmapped">ignore_unmapped</code></td>
<td>
<p>logical. Don't include unmapped reads in the plot (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_ignore_unclassified">ignore_unclassified</code></td>
<td>
<p>logical. Don't include unclassified ORFs in the plot (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_gradient_col">gradient_col</code></td>
<td>
<p>A vector of two colors representing the low and high ends of the color gradient (default <code>c("ghostwhite", "dodgerblue4")</code>).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_base_size">base_size</code></td>
<td>
<p>numeric. Base font size (default <code>11</code>).</p>
</td></tr>
<tr><td><code id="plotFunctions_+3A_metadata_groups">metadata_groups</code></td>
<td>
<p>list. Split the plot into groups defined by the user: list('G1' = c('sample1', sample2'), 'G2' = c('sample3', 'sample4')) default <code>NULL</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 plot object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotTaxonomy">plotTaxonomy</a></code> for plotting the most abundant taxa of a SQM object; <code><a href="#topic+plotBars">plotBars</a></code> and <code><a href="#topic+plotHeatmap">plotHeatmap</a></code> for plotting barplots or heatmaps with arbitrary data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
plotFunctions(Hadza)
</code></pre>

<hr>
<h2 id='plotHeatmap'>Plot a heatmap using ggplot2</h2><span id='topic+plotHeatmap'></span>

<h3>Description</h3>

<p>Plot a ggplot2 heatmap from a matrix or data frame. The data should be in tabular format (e.g. features in rows and samples in columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHeatmap(
  data,
  label_x = "Samples",
  label_y = "Features",
  label_fill = "Abundance",
  gradient_col = c("ghostwhite", "dodgerblue4"),
  base_size = 11,
  metadata_groups = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotHeatmap_+3A_data">data</code></td>
<td>
<p>numeric matrix or data frame.</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_label_x">label_x</code></td>
<td>
<p>character Label for the x axis (default <code>"Samples"</code>).</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_label_y">label_y</code></td>
<td>
<p>character Label for the y axis (default <code>"Features"</code>).</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_label_fill">label_fill</code></td>
<td>
<p>character Label for color scale (default <code>"Abundance"</code>).</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_gradient_col">gradient_col</code></td>
<td>
<p>A vector of two colors representing the low and high ends of the color gradient (default <code>c("ghostwhite", "dodgerblue4")</code>).</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_base_size">base_size</code></td>
<td>
<p>numeric. Base font size (default <code>11</code>).</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_metadata_groups">metadata_groups</code></td>
<td>
<p>list. Split the plot into groups defined by the user: list('G1' = c('sample1', sample2'), 'G2' = c('sample3', 'sample4')) default <code>NULL</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 plot object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotFunctions">plotFunctions</a></code> for plotting the top functional categories of a SQM object; <code><a href="#topic+plotBars">plotBars</a></code> for plotting a barplot with arbitrary data; <code><a href="#topic+mostAbundant">mostAbundant</a></code> for selecting the most abundant rows in a dataframe or matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
topPFAM = mostAbundant(Hadza$functions$PFAM$tpm)
topPFAM = topPFAM[rownames(topPFAM) != "Unclassified",] # Take out the Unclassified ORFs.
plotHeatmap(topPFAM, label_x = "Samples", label_y = "PFAMs", label_fill = "TPM")
</code></pre>

<hr>
<h2 id='plotTaxonomy'>Barplot of the most abundant taxa in a SQM object</h2><span id='topic+plotTaxonomy'></span>

<h3>Description</h3>

<p>This function selects the most abundant taxa across all samples in a SQM object and represents their abundances in a barplot. Alternatively, a custom set of taxa can be represented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTaxonomy(
  SQM,
  rank = "phylum",
  count = "percent",
  N = 15,
  tax = NULL,
  others = TRUE,
  samples = NULL,
  nocds = "treat_separately",
  ignore_unmapped = FALSE,
  ignore_unclassified = FALSE,
  no_partial_classifications = FALSE,
  rescale = FALSE,
  color = NULL,
  base_size = 11,
  max_scale_value = NULL,
  metadata_groups = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTaxonomy_+3A_sqm">SQM</code></td>
<td>
<p>A SQM or a SQMlite object.</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_rank">rank</code></td>
<td>
<p>Taxonomic rank to plot (default <code>phylum</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_count">count</code></td>
<td>
<p>character. Either <code>"percent"</code> for percentages, or <code>"abund"</code> for raw abundances (default <code>"percent"</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_n">N</code></td>
<td>
<p>integer Plot the <code>N</code> most abundant taxa (default <code>15</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_tax">tax</code></td>
<td>
<p>character. Custom taxa to plot. If provided, it will override <code>N</code> (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_others">others</code></td>
<td>
<p>logical. Collapse the abundances of least abundant taxa, and include the result in the plot (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_samples">samples</code></td>
<td>
<p>character. Character vector with the names of the samples to include in the plot. Can also be used to plot the samples in a custom order. If not provided, all samples will be plotted (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_nocds">nocds</code></td>
<td>
<p>character. Either <code>"treat_separately"</code> to treat reads annotated as No CDS separately, <code>"treat_as_unclassified"</code> to treat them as Unclassified or <code>"ignore"</code> to ignore them in the plot (default <code>"treat_separately"</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_ignore_unmapped">ignore_unmapped</code></td>
<td>
<p>logical. Don't include unmapped reads in the plot (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_ignore_unclassified">ignore_unclassified</code></td>
<td>
<p>logical. Don't include unclassified reads in the plot (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_no_partial_classifications">no_partial_classifications</code></td>
<td>
<p>logical. Treat reads not fully classified at the requested level (e.g. &quot;Unclassified bacteroidetes&quot; at the class level or below) as fully unclassified. This takes effect before <code>ignore_unclassified</code>, so if both are <code>TRUE</code> the plot will only contain fully classified contigs (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_rescale">rescale</code></td>
<td>
<p>logical. Re-scale results to percentages (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_color">color</code></td>
<td>
<p>Vector with custom colors for the different features. If empty, we will use our own hand-picked pallete if N&lt;=15, and the default ggplot2 palette otherwise (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_base_size">base_size</code></td>
<td>
<p>numeric. Base font size (default <code>11</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_max_scale_value">max_scale_value</code></td>
<td>
<p>numeric. Maximum value to include in the y axis. By default it is handled automatically by ggplot2 (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plotTaxonomy_+3A_metadata_groups">metadata_groups</code></td>
<td>
<p>list. Split the plot into groups defined by the user: list('G1' = c('sample1', sample2'), 'G2' = c('sample3', 'sample4')) default <code>NULL</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 plot object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotFunctions">plotFunctions</a></code> for plotting the most abundant functions of a SQM object; <code><a href="#topic+plotBars">plotBars</a></code> and <code><a href="#topic+plotHeatmap">plotHeatmap</a></code> for plotting barplots or heatmaps with arbitrary data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
Hadza.amin = subsetFun(Hadza, "Amino acid metabolism")
# Taxonomic distribution of amino acid metabolism ORFs at the family level.
plotTaxonomy(Hadza.amin, "family")
</code></pre>

<hr>
<h2 id='RecA'>RecA/RadA recombinase</h2><span id='topic+RecA'></span>

<h3>Description</h3>

<p>The recombination protein RecA/RadA is essential for the repair and
maintenance of DNA, and has homologs in every bacteria and archaea.
By dividing the coverage of functions by the coverage of RecA, abundances
can be transformed into copy numbers, which can be used to compare 
functional profiles in samples with different sequencing depths.
RecA-derived copy numbers are available in the SQM object
(<code>SQM$functions$&lt;annotation_type&gt;$copy_number</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RecA)
</code></pre>


<h3>Format</h3>

<p>Character vector with the COG identifier for RecA/RadA.
</p>


<h3>Source</h3>

<p><a href="http://eggnogdb.embl.de/#/app/results?seqid=P0A7G6&amp;target_nogs=COG0468#COG0468_datamenu">EggNOG Database</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
data(RecA)
### Let's calculate the average copy number of each function in our samples.
# We do it for COG annotations here, but we could also do it for KEGG or PFAMs.
COG.coverage = Hadza$functions$COG$cov
COG.copynumber = t(t(COG.coverage) / COG.coverage[RecA,]) # Sample-wise division by RecA coverage.
</code></pre>

<hr>
<h2 id='rowMaxs'>Return a vector with the row-wise maxima of a matrix or dataframe.</h2><span id='topic+rowMaxs'></span>

<h3>Description</h3>

<p>Return a vector with the row-wise maxima of a matrix or dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rowMaxs(table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rowMaxs_+3A_table">table</code></td>
<td>
<p>matrix or dataframe.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with the row-wise maxima.
</p>

<hr>
<h2 id='rowMins'>Return a vector with the row-wise minima of a matrix or dataframe.</h2><span id='topic+rowMins'></span>

<h3>Description</h3>

<p>Return a vector with the row-wise minima of a matrix or dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rowMins(table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rowMins_+3A_table">table</code></td>
<td>
<p>matrix or dataframe.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with the row-wise minima.
</p>

<hr>
<h2 id='seqvec2fasta'>Print a named vector of sequences as a fasta-formatted string</h2><span id='topic+seqvec2fasta'></span>

<h3>Description</h3>

<p>Print a named vector of sequences as a fasta-formatted string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seqvec2fasta(seqvec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seqvec2fasta_+3A_seqvec">seqvec</code></td>
<td>
<p>vector. The vector to be written as a fasta string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character string with the sequence/s written in the fasta format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
seqvec2fasta(Hadza$orfs$seqs[1:10])
</code></pre>

<hr>
<h2 id='subsetBins'>Create a SQM object containing only the requested bins, and the contigs and ORFs contained in them.</h2><span id='topic+subsetBins'></span>

<h3>Description</h3>

<p>Create a SQM object containing only the requested bins, and the contigs and ORFs contained in them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetBins(
  SQM,
  bins,
  trusted_functions_only = FALSE,
  ignore_unclassified_functions = FALSE,
  rescale_tpm = TRUE,
  rescale_copy_number = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetBins_+3A_sqm">SQM</code></td>
<td>
<p>SQM object to be subsetted.</p>
</td></tr>
<tr><td><code id="subsetBins_+3A_bins">bins</code></td>
<td>
<p>character. Vector of bins to be selected.</p>
</td></tr>
<tr><td><code id="subsetBins_+3A_trusted_functions_only">trusted_functions_only</code></td>
<td>
<p>logical. If <code>TRUE</code>, only highly trusted functional annotations (best hit + best average) will be considered when generating aggregated function tables. If <code>FALSE</code>, best hit annotations will be used (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetBins_+3A_ignore_unclassified_functions">ignore_unclassified_functions</code></td>
<td>
<p>logical. If <code>FALSE</code>, ORFs with no functional classification will be aggregated together into an &quot;Unclassified&quot; category. If <code>TRUE</code>, they will be ignored (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetBins_+3A_rescale_tpm">rescale_tpm</code></td>
<td>
<p>logical. If <code>TRUE</code>, TPMs for KEGGs, COGs, and PFAMs will be recalculated (so that the TPMs in the subset actually add up to 1 million). Otherwise, per-function TPMs will be calculated by aggregating the TPMs of the ORFs annotated with that function, and will thus keep the scaling present in the parent object. By default it is set to <code>TRUE</code>, which means that the returned TPMs will be scaled <em>by million of reads of the selected bins</em>.</p>
</td></tr>
<tr><td><code id="subsetBins_+3A_rescale_copy_number">rescale_copy_number</code></td>
<td>
<p>logical. If <code>TRUE</code>, copy numbers with be recalculated using the RecA/RadA coverages in the subset. Otherwise, RecA/RadA coverages will be taken from the parent object. By default it is set to <code>TRUE</code>, which means that the returned copy numbers for each function will represent the average copy number of that function <em>per genome of the selected bins</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQM object containing only the requested bins.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsetContigs">subsetContigs</a></code>, <code><a href="#topic+subsetORFs">subsetORFs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
# Which are the most complete bins?
topBinNames = rownames(Hadza$bins$table)[order(Hadza$bins$table[,"Completeness"],
                                         decreasing=TRUE)][1:2]
# Subset with the most complete bin.
topBin = subsetBins(Hadza, topBinNames[1])
</code></pre>

<hr>
<h2 id='subsetContigs'>Select contigs</h2><span id='topic+subsetContigs'></span>

<h3>Description</h3>

<p>Create a SQM object containing only the requested contigs, the ORFs contained in them and the bins that contain them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetContigs(
  SQM,
  contigs,
  trusted_functions_only = FALSE,
  ignore_unclassified_functions = FALSE,
  rescale_tpm = FALSE,
  rescale_copy_number = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetContigs_+3A_sqm">SQM</code></td>
<td>
<p>SQM object to be subsetted.</p>
</td></tr>
<tr><td><code id="subsetContigs_+3A_contigs">contigs</code></td>
<td>
<p>character. Vector of contigs to be selected.</p>
</td></tr>
<tr><td><code id="subsetContigs_+3A_trusted_functions_only">trusted_functions_only</code></td>
<td>
<p>logical. If <code>TRUE</code>, only highly trusted functional annotations (best hit + best average) will be considered when generating aggregated function tables. If <code>FALSE</code>, best hit annotations will be used (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetContigs_+3A_ignore_unclassified_functions">ignore_unclassified_functions</code></td>
<td>
<p>logical. If <code>FALSE</code>, ORFs with no functional classification will be aggregated together into an &quot;Unclassified&quot; category. If <code>TRUE</code>, they will be ignored (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetContigs_+3A_rescale_tpm">rescale_tpm</code></td>
<td>
<p>logical. If <code>TRUE</code>, TPMs for KEGGs, COGs, and PFAMs will be recalculated (so that the TPMs in the subset actually add up to 1 million). Otherwise, per-function TPMs will be calculated by aggregating the TPMs of the ORFs annotated with that function, and will thus keep the scaling present in the parent object (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetContigs_+3A_rescale_copy_number">rescale_copy_number</code></td>
<td>
<p>logical. If <code>TRUE</code>, copy numbers with be recalculated using the RecA/RadA coverages in the subset. Otherwise, RecA/RadA coverages will be taken from the parent object. By default it is set to <code>FALSE</code>, which means that the returned copy numbers for each function will represent the average copy number of that function per genome in the parent object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQM object containing only the selected contigs.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsetORFs">subsetORFs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
# Which contigs have a GC content below 40?
lowGCcontigNames = rownames(Hadza$contigs$table[Hadza$contigs$table[,"GC perc"]&lt;40,])
lowGCcontigs = subsetContigs(Hadza, lowGCcontigNames)
hist(lowGCcontigs$contigs$table[,"GC perc"])
</code></pre>

<hr>
<h2 id='subsetFun'>Filter results by function</h2><span id='topic+subsetFun'></span>

<h3>Description</h3>

<p>Create a SQM object containing only the ORFs with a given function, and the contigs and bins that contain them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetFun(
  SQM,
  fun,
  columns = NULL,
  ignore_case = TRUE,
  fixed = FALSE,
  trusted_functions_only = FALSE,
  ignore_unclassified_functions = FALSE,
  rescale_tpm = FALSE,
  rescale_copy_number = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetFun_+3A_sqm">SQM</code></td>
<td>
<p>SQM object to be subsetted.</p>
</td></tr>
<tr><td><code id="subsetFun_+3A_fun">fun</code></td>
<td>
<p>character. Pattern to search for in the different functional classifications.</p>
</td></tr>
<tr><td><code id="subsetFun_+3A_columns">columns</code></td>
<td>
<p>character. Restrict the search to the provided column names from <code>SQM$orfs$table</code>. If not provided the search will be performed in all the columns containing functional information (default <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="subsetFun_+3A_ignore_case">ignore_case</code></td>
<td>
<p>logical Make pattern matching case-insensitive (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="subsetFun_+3A_fixed">fixed</code></td>
<td>
<p>logical. If <code>TRUE</code>, pattern is a string to be matched as is. If <code>FALSE</code> the pattern is treated as a regular expression (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetFun_+3A_trusted_functions_only">trusted_functions_only</code></td>
<td>
<p>logical. If <code>TRUE</code>, only highly trusted functional annotations (best hit + best average) will be considered when generating aggregated function tables. If <code>FALSE</code>, best hit annotations will be used (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetFun_+3A_ignore_unclassified_functions">ignore_unclassified_functions</code></td>
<td>
<p>logical. If <code>FALSE</code>, ORFs with no functional classification will be aggregated together into an &quot;Unclassified&quot; category. If <code>TRUE</code>, they will be ignored (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetFun_+3A_rescale_tpm">rescale_tpm</code></td>
<td>
<p>logical. If <code>TRUE</code>, TPMs for KEGGs, COGs, and PFAMs will be recalculated (so that the TPMs in the subset actually add up to 1 million). Otherwise, per-function TPMs will be calculated by aggregating the TPMs of the ORFs annotated with that function, and will thus keep the scaling present in the parent object (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetFun_+3A_rescale_copy_number">rescale_copy_number</code></td>
<td>
<p>logical. If <code>TRUE</code>, copy numbers with be recalculated using the RecA/RadA coverages in the subset. Otherwise, RecA/RadA coverages will be taken from the parent object. By default it is set to <code>FALSE</code>, which means that the returned copy numbers for each function will represent the average copy number of that function per genome in the parent object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQM object containing only the requested function.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsetTax">subsetTax</a></code>, <code><a href="#topic+subsetORFs">subsetORFs</a></code>, <code><a href="#topic+subsetSamples">subsetSamples</a></code>, <code><a href="#topic+combineSQM">combineSQM</a></code>. The most abundant items of a particular table contained in a SQM object can be selected with <code><a href="#topic+mostAbundant">mostAbundant</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
Hadza.iron = subsetFun(Hadza, "iron")
Hadza.carb = subsetFun(Hadza, "Carbohydrate metabolism")
# Search for multiple patterns using regular expressions
Hadza.twoKOs = subsetFun(Hadza, "K00812|K00813", fixed=FALSE)
</code></pre>

<hr>
<h2 id='subsetORFs'>Select ORFs</h2><span id='topic+subsetORFs'></span>

<h3>Description</h3>

<p>Create a SQM object containing only the requested ORFs, and the contigs and bins that contain them. Internally, all the other subset functions in this package end up calling <code>subsetORFs</code> to do the work for them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetORFs(
  SQM,
  orfs,
  tax_source = "orfs",
  trusted_functions_only = FALSE,
  ignore_unclassified_functions = FALSE,
  rescale_tpm = FALSE,
  rescale_copy_number = FALSE,
  contigs_override = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetORFs_+3A_sqm">SQM</code></td>
<td>
<p>SQM object to be subsetted.</p>
</td></tr>
<tr><td><code id="subsetORFs_+3A_orfs">orfs</code></td>
<td>
<p>character. Vector of ORFs to be selected.</p>
</td></tr>
<tr><td><code id="subsetORFs_+3A_tax_source">tax_source</code></td>
<td>
<p>character. Features used for calculating aggregated abundances at the different taxonomic ranks. Either <code>"orfs"</code> or <code>"contigs"</code> (default <code>"orfs"</code>).</p>
</td></tr>
<tr><td><code id="subsetORFs_+3A_trusted_functions_only">trusted_functions_only</code></td>
<td>
<p>logical. If <code>TRUE</code>, only highly trusted functional annotations (best hit + best average) will be considered when generating aggregated function tables. If <code>FALSE</code>, best hit annotations will be used (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetORFs_+3A_ignore_unclassified_functions">ignore_unclassified_functions</code></td>
<td>
<p>logical. If <code>FALSE</code>, ORFs with no functional classification will be aggregated together into an &quot;Unclassified&quot; category. If <code>TRUE</code>, they will be ignored (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetORFs_+3A_rescale_tpm">rescale_tpm</code></td>
<td>
<p>logical. If <code>TRUE</code>, TPMs for KEGGs, COGs, and PFAMs will be recalculated (so that the TPMs in the subset actually add up to 1 million). Otherwise, per-function TPMs will be calculated by aggregating the TPMs of the ORFs annotated with that function, and will thus keep the scaling present in the parent object (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetORFs_+3A_rescale_copy_number">rescale_copy_number</code></td>
<td>
<p>logical. If <code>TRUE</code>, copy numbers with be recalculated using the RecA/RadA coverages in the subset. Otherwise, RecA/RadA coverages will be taken from the parent object. By default it is set to <code>FALSE</code>, which means that the returned copy numbers for each function will represent the average copy number of that function per genome in the parent object.</p>
</td></tr>
<tr><td><code id="subsetORFs_+3A_contigs_override">contigs_override</code></td>
<td>
<p>character. Optional vector of contigs to be included in the subsetted object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQM object containing the requested ORFs.
</p>


<h3>A note on contig/bins subsetting</h3>

<p>While this function selects the contigs and bins that contain the desired orfs, it DOES NOT recalculate contig/bin abundance and statistics based on the selected ORFs only. This means that the abundances presented in tables such as <code>SQM$contig$abund</code> or <code>SQM$bins$tpm</code> will still refer to the complete contigs and bins, regardless of whether only a fraction of their ORFs are actually present in the returned SQM object. This is also true for the statistics presented in <code>SQM$contigs$table</code> and <code>SQM$bins$table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
# Select the 100 most abundant ORFs in our dataset.
mostAbundantORFnames = names(sort(rowSums(Hadza$orfs$tpm), decreasing=TRUE))[1:100]
mostAbundantORFs = subsetORFs(Hadza, mostAbundantORFnames)
</code></pre>

<hr>
<h2 id='subsetRand'>Select random ORFs</h2><span id='topic+subsetRand'></span>

<h3>Description</h3>

<p>Create a random subset of a SQM object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetRand(SQM, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetRand_+3A_sqm">SQM</code></td>
<td>
<p>SQM object to be subsetted.</p>
</td></tr>
<tr><td><code id="subsetRand_+3A_n">N</code></td>
<td>
<p>numeric. number of random ORFs to select.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQM object containing a random subset of ORFs.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsetORFs">subsetORFs</a></code>
</p>

<hr>
<h2 id='subsetSamples'>Filter results by sample</h2><span id='topic+subsetSamples'></span>

<h3>Description</h3>

<p>Create a SQM object containing only samples specified by the user, and the ORFs, contigs, bins, taxa and functions present in those samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetSamples(SQM, samples, remove_missing = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetSamples_+3A_sqm">SQM</code></td>
<td>
<p>SQM object to be subsetted.</p>
</td></tr>
<tr><td><code id="subsetSamples_+3A_samples">samples</code></td>
<td>
<p>character. Samples to be included in the subset.</p>
</td></tr>
<tr><td><code id="subsetSamples_+3A_remove_missing">remove_missing</code></td>
<td>
<p>bool. If <code>TRUE</code>, ORFs, contigs, bins, taxa and functions absent from the selected samples will be removed from the subsetted object (default <code>TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQM object containing only the requested samples.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsetTax">subsetTax</a></code>, <code><a href="#topic+subsetFun">subsetFun</a></code>, <code><a href="#topic+subsetORFs">subsetORFs</a></code>, <code><a href="#topic+combineSQM">combineSQM</a></code>. The most abundant items of a particular table contained in a SQM object can be selected with <code><a href="#topic+mostAbundant">mostAbundant</a></code>.
</p>

<hr>
<h2 id='subsetTax'>Filter results by taxonomy</h2><span id='topic+subsetTax'></span>

<h3>Description</h3>

<p>Create a SQM object containing only the contigs with a given consensus taxonomy, the ORFs contained in them and the bins that contain them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetTax(
  SQM,
  rank,
  tax,
  trusted_functions_only = FALSE,
  ignore_unclassified_functions = FALSE,
  rescale_tpm = TRUE,
  rescale_copy_number = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetTax_+3A_sqm">SQM</code></td>
<td>
<p>SQM object to be subsetted.</p>
</td></tr>
<tr><td><code id="subsetTax_+3A_rank">rank</code></td>
<td>
<p>character. The taxonomic rank from which to select the desired taxa (<code>superkingdom</code>, <code>phylum</code>, <code>class</code>, <code>order</code>, <code>family</code>, <code>genus</code>, <code>species</code>)</p>
</td></tr>
<tr><td><code id="subsetTax_+3A_tax">tax</code></td>
<td>
<p>character. The taxon to select.</p>
</td></tr>
<tr><td><code id="subsetTax_+3A_trusted_functions_only">trusted_functions_only</code></td>
<td>
<p>logical. If <code>TRUE</code>, only highly trusted functional annotations (best hit + best average) will be considered when generating aggregated function tables. If <code>FALSE</code>, best hit annotations will be used (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetTax_+3A_ignore_unclassified_functions">ignore_unclassified_functions</code></td>
<td>
<p>logical. If <code>FALSE</code>, ORFs with no functional classification will be aggregated together into an &quot;Unclassified&quot; category. If <code>TRUE</code>, they will be ignored (default <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="subsetTax_+3A_rescale_tpm">rescale_tpm</code></td>
<td>
<p>logical. If <code>TRUE</code>, TPMs for KEGGs, COGs, and PFAMs will be recalculated (so that the TPMs in the subset actually add up to 1 million). Otherwise, per-function TPMs will be calculated by aggregating the TPMs of the ORFs annotated with that function, and will thus keep the scaling present in the parent object. By default it is set to <code>TRUE</code>, which means that the returned TPMs will be scaled <em>by million of reads of the selected taxon</em>.</p>
</td></tr>
<tr><td><code id="subsetTax_+3A_rescale_copy_number">rescale_copy_number</code></td>
<td>
<p>logical. If <code>TRUE</code>, copy numbers with be recalculated using the RecA/RadA coverages in the subset. Otherwise, RecA/RadA coverages will be taken from the parent object. By default it is set to <code>TRUE</code>, which means that the returned copy numbers for each function will represent the average copy number of that function <em>per genome of the selected taxon</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SQM object containing only the requested taxon.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+subsetFun">subsetFun</a></code>, <code><a href="#topic+subsetContigs">subsetContigs</a></code>, <code><a href="#topic+subsetSamples">subsetSamples</a></code>, <code><a href="#topic+combineSQM">combineSQM</a></code>. The most abundant items of a particular table contained in a SQM object can be selected with <code><a href="#topic+mostAbundant">mostAbundant</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
Hadza.Prevotella = subsetTax(Hadza, "genus", "Prevotella")
Hadza.Proteobacteria = subsetTax(Hadza, "phylum", "Proteobacteria")
</code></pre>

<hr>
<h2 id='summary.SQM'>summary method for class SQM</h2><span id='topic+summary.SQM'></span>

<h3>Description</h3>

<p>Computes different statistics of the data contained in the SQM object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SQM'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.SQM_+3A_object">object</code></td>
<td>
<p>SQM object to be summarized.</p>
</td></tr>
<tr><td><code id="summary.SQM_+3A_...">...</code></td>
<td>
<p>Additional parameters (ignored).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of summary statistics.
</p>

<hr>
<h2 id='summary.SQMlite'>summary method for class SQMlite</h2><span id='topic+summary.SQMlite'></span>

<h3>Description</h3>

<p>Computes different statistics of the data contained in the SQMlite object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'SQMlite'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.SQMlite_+3A_object">object</code></td>
<td>
<p>SQMlite object to be summarized.</p>
</td></tr>
<tr><td><code id="summary.SQMlite_+3A_...">...</code></td>
<td>
<p>Additional parameters (ignored).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of summary statistics.
</p>

<hr>
<h2 id='USiCGs'>Universal Single-Copy Genes</h2><span id='topic+USiCGs'></span>

<h3>Description</h3>

<p>Lists of Universal Single Copy Genes for Bacteria and Archaea.
These are useful for transforming coverages or tpms into copy numbers.
This is an alternative way of normalizing data in order to be able to
compare functional profiles in samples with different sequencing depths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(USiCGs)
</code></pre>


<h3>Format</h3>

<p>Character vector with the KEGG identifiers for 15 Universal Single Copy Genes.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3798274/bin/pcbi.1003292.s016.xls">Carr <em>et al.</em>, 2013. Table S1</a>.
</p>


<h3>References</h3>

<p>Carr, Shen-Orr &amp; Borenstein (2013). 
Reconstructing the Genomic Content of Microbiome Taxa through Shotgun Metagenomic Deconvolution
<em>PLoS Comput. Biol.</em> <b>9</b>:e1003292.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3798274/">PubMed</a>).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MGOGs">MGOGs</a></code> and <code><a href="#topic+MGKOs">MGKOs</a></code> for an alternative set of single copy genes, and for examples on how to generate copy numbers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hadza)
data(USiCGs)
### Let's look at the Universal Single Copy Gene distribution in our samples.
KEGG.tpm = Hadza$functions$KEGG$tpm
all(USiCGs %in% rownames(KEGG.tpm)) # Are all the USiCGs present in our dataset?
# Plot a boxplot of USiCGs tpms and calculate median USiCGs tpm.
# This looks weird in the test dataset because it contains only a small subset of the metagenomes.
# In a set of complete metagenomes USiCGs should have fairly similar TPM averages
# and low dispersion across samples.
boxplot(t(KEGG.tpm[USiCGs,]), names=USiCGs, ylab="TPM", col="slateblue2")
 
### Now let's calculate the average copy numbers of each function.
# We do it for KEGG annotations here, but we could also do it for COGs or PFAMs.
USiCGs.cov = apply(Hadza$functions$KEGG$cov[USiCGs,], 2, median)
# Sample-wise division by the median USiCG coverage.
KEGG.copynumber = t(t(Hadza$functions$KEGG$cov) / USiCGs.cov)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
