<!DOCTYPE html><html><head><title>Help for package squid</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {squid}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#squid'><p>SQUID - Statistical Quantification of Individual Differences: an educational and statistical tool for understanding multi-level phenotypic data in linear mixed models</p></a></li>
<li><a href='#squidApp'><p>SQuID application</p></a></li>
<li><a href='#squidR'><p>Run SQuID without graphical user interface</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Statistical Quantification of Individual Differences</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A simulation-based tool made to help researchers to become familiar with
    multilevel variations, and to build up sampling designs for their study. 
    This tool has two main objectives: First, it provides an educational tool useful for students, 
    teachers and researchers who want to learn to use mixed-effects models. 
    Users can experience how the mixed-effects model framework can be used to understand 
    distinct biological phenomena by interactively exploring simulated multilevel data. 
    Second, it offers research opportunities to those who are already familiar with 
    mixed-effects models, as it enables the generation of data sets that users may download 
    and use for a range of simulation-based statistical analyses such as power 
    and sensitivity analysis of multilevel and multivariate data [Allegue, H., Araya-Ajoy, Y.G., Dingemanse, 
    N.J., Dochtermann N.A., Garamszegi, L.Z., Nakagawa, S., Reale, D., Schielzeth, H. and Westneat, D.F. (2016) 
    &lt;<a href="https://doi.org/10.1111%2F2041-210X.12659">doi:10.1111/2041-210X.12659</a>&gt;].</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/squid-group/squid">https://github.com/squid-group/squid</a>,
<a href="https://squid-group.github.io/squid/">https://squid-group.github.io/squid/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/squid-group/squid/issues">https://github.com/squid-group/squid/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, shiny (&ge; 1.6.0), shinyMatrix (&ge; 0.4.0), ggplot2 (&ge;
3.3.5), plotly (&ge; 4.9.3), grid (&ge; 4.1.1), MASS (&ge; 7.3-53.1),
lme4 (&ge; 1.1-21), arm (&ge; 1.10-1), data.table (&ge; 1.1-27.1),
brms (&ge; 2.15.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-21 20:20:50 UTC; HASSEN</td>
</tr>
<tr>
<td>Author:</td>
<td>Hassen Allegue [aut, cre],
  Yimen G. Araya-Ajoy [aut],
  Niels J. Dingemanse [aut],
  Ned A. Dochtermann [aut],
  Laszlo Z. Garamszegi [aut],
  Shinichi Nakagawa [aut],
  Denis Reale [aut],
  Holger Schielzeth [aut],
  David F. Westneat [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hassen Allegue &lt;h.all@disroot.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-21 20:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='squid'>SQUID - Statistical Quantification of Individual Differences: an educational and statistical tool for understanding multi-level phenotypic data in linear mixed models</h2><span id='topic+squid'></span><span id='topic+squid-package'></span>

<h3>Description</h3>

<p>SQuID stands for <strong>S</strong>tatistical <strong>Qu</strong>antification of <strong>I</strong>ndividual <strong>D</strong>ifferences and is the product of the SQuID working group. The package aims to help scholars who, like us, are interested in understanding patterns of phenotypic variance. Individual differences are the raw material for natural selection to act on and hence the basis of evolutionary adaptation. Understanding the sources of phenotypic variance is thus a most essential feature of biological investigation and mixed effects models offer a great, albeit challenging tool. Disseminating the properties, potentials and interpretational challenges in the research community is thus a foremost goal of SQuID.
</p>
<p>The SQuID package has two main objectives: First, it provides an educational tool useful for students, teachers and researchers who want to learn to use mixed-effects models. Users can experience how the mixed-effects model framework can be used to understand distinct biological phenomena by interactively exploring simulated multilevel data. Second, SQuID offers research opportunities to those who are already familiar with mixed-effects models, as SQuID enables the generation of datasets that users may download and use for a range of simulation-based statistical analyses such as power and sensitivity analysis of multilevel and multivariate data.
</p>


<h3>Background</h3>

<p>The phenotype of a trait in an individual results from a sum of genetic and environmental influences. Phenotypic variation is structured in a hierarchical way and the hierarchical modeling in mixed effect models is great tool to analyze and decompose such variation. Phenotypes vary across species, across populations of the same species, across individuals of the same population, and across repeated observations of the same individual. We focused on the individual level because it represents one of the most important biological levels to both ecological and evolutionary processes. Different sources of variation are at the origin of the phenotype of an individual. Individuals may differ in their phenotypes because they carry different gene variants (i.e. alleles). But individuals also experience different environments during their lifetime. Some environmental influences impose a lasting mark on the phenotype, while others are more ephemerous. The former tend to produce long-lasting, among-individual variation, while the latter causes within-individual variation. However, this depends on the time scale at which the measurements of the phenotypes are done relative that of the environmental influences. Furthermore, individuals differ not only in their average phenotypes but also in how they respond to changes in their environment (i.e. differences in individual phenotypic plasticity). This represents an interaction between the among- and the within-individual levels of variation. The patterns of variation can, thus, be very complex. Selection can act differently on these different components of variance in the phenotypes of a trait, and this is why it is important to quantify their magnitude.
</p>
<p>Mixed models are very flexible statistical tools that provide a way to estimate the variation at these different levels, and represent the general statistical framework for evolutionary biology. Because of the progress in computational capacities mixed models have become increasingly popular among ecologists and evolutionary biologists over the last decade. However, fitting mixed model is not a straightforward exercise, and the way data are sampled among and within individuals can have strong implications on the outcome of the model. This is why we created the SQuID simulation tool that could help new users interested in decomposing phenotypic variance to get more familiar with the concept of hierarchical organization of traits, with mixed models and to avoid pitfalls caused by inappropriate sampling. 
</p>


<h3>Details</h3>

<p>SQuID is a simulation-based tool that can be used for research and educational purposes. SQuID creates a world inhabited by individuals whose phenotypes are generated by a user-defined phenotypic equation, which allows easy translation of biological hypotheses into mathematically quantifiable parameters. The framework is suitable for performing simulation studies, determining optimal sampling designs for user-specific biological problems, and making simulation based inferences to aid in the interpretation of empirical studies. SQuID is also a teaching tool for biologists interested in learning, or teaching others, how to implement and interpret mixed-effects models, when studying the processes causing phenotypic variation. SQuID is based on a mathematical model that creates a group of individuals (i.e. study population) repeatedly expressing phenotypes, for one or two different traits, in uniform time. Phenotypic values of traits are generated following the general principle of the phenotypic equation (<a href="https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/1365-2656.12013">Dingemanse &amp; Dochtermann 2013, Journal of Animal Ecology</a>): phenotypes are assumed to be the summed effects of a series of components and the phenotypic variance (Vp) is the sum of the respective variances in theses causal components. The user has thus the flexibility to add different variance components that will form the phenotype of the individual at each time step, and to set up the relative importance of each component through the definition of environmental effects. SQuID then allows the user to collect a subsample of phenotypes for each simulated individual (i.e. operational data set), according to a specific sampling design. The major difference between SQuID and other R packages that also allow performance analysis through data simulation (e.g. <a href="https://cran.r-project.org/package=pamm">pamm</a>, <a href="https://cran.r-project.org/package=odprism">odprism</a>, <a href="https://cran.r-project.org/package=simr">simr</a>), is that only SQUID allows separate steps for generating the world first and then model a sampling process from it. SQuID is subject to evolution and is designed to adapt to more complex scenarios in the future.
</p>
<p>SQuID has two main functions; <code><a href="#topic+squidApp">squidApp</a></code> and <code><a href="#topic+squidR">squidR</a></code>.
</p>


<h4>squidApp</h4>

<p>This function runs the SQuID application which is a browser-based interface created with the <a href="https://shiny.rstudio.com/">shiny</a> package. SQuID is built up as a series of modules that guide the user into situations of increasing complexity to explore the phenotypic equation model and the dynamics between the way phenotypes are sampled and the estimation of parameters of specific interest; The last module is the full model simulation that allows the user to generate data sets that can then be used to run analyses in the statistical package of their choice for specific research questions. For most of the modules, the simulated data set is automatically fed into a statistical model in R and the main results of the analysis shown in an output. For the full model the user has the opportunity to download the operational data set for further analyses. The SQuID application also has a tab (Full model (Step by step)) describing in details the SQuID full model.
</p>



<h4>squidR</h4>

<p>This function is a traditional R function that allows data generation and sampling without the browser-based interface. This function can be used for more advanced and efficient simulations once you understand how SQuID works. squidR can be easily included in R scripts.
</p>



<h3>References</h3>

<p>Allegue, H., Araya-Ajoy, Y.G., Dingemanse, N.J., Dochtermann N.A., Garamszegi, L.Z., Nakagawa, S., Reale, D., Schielzeth, H. and Westneat, D.F. (2016) SQuID - Statistical Quantification of Individual Differences: an educational and statistical tool for understanding multi-level phenotypic data in linear mixed models. <em>Methods in Ecology and Evolution</em>, <b>8</b>:257-267.
</p>
<p>Dingemanse, N.J. and Dochtermann N.A. (2013) Quantifying individual variation in behaviour: mixed-effect modelling approaches. <em>Journal of Animal Ecology</em>, <b>82</b>:39-54.
</p>

<hr>
<h2 id='squidApp'>SQuID application</h2><span id='topic+squidApp'></span>

<h3>Description</h3>

<p><code>squidApp</code> runs SQuID application as a browser interface that has been developed with the package shiny.
</p>
<p>SQuID web interface is made to help researchers to become familiar with multilevel phenotypic variation, and to design optimal sampling schemes for their studies. SQuID is built up as a series of modules that guide the user into situations of increasing complexity to explore the dynamics between the way phenotypes are sampled and the estimation of parameters of specific interest. The complete simulations the full model simulation tab provides functionality that allows the user to generate data sets that can then be used to run analyses in the statistical package of their choice and address specific research questions.
</p>
<p>SQuID is based on a mathematical model that creates a group of individuals (i.e. study population) repeatedly expressing phenotypes, for one or two different traits, through time. Phenotypic values of traits are generated following the general principle of the phenotypic equation (<a href="https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/1365-2656.12013">Dingemanse &amp; Dochtermann 2013, Journal of Animal Ecology</a>): phenotypic variance (Vp) is assumed to be the sum of a series of components (see the full model tab on the web interface). The user has the flexibility to add different causal variance components that will form the phenotype of the individual at each time step, and to set up the relative importance of each component through the definition of environmental effects. SQuID then allows the user to collect a subsample of phenotypes for each simulated individual (i.e. operational data set), according to a specific sampling design. For most of the modules, the simulated data set is automatically fed into a statistical model in R and the main results of the analysis shown in an output. For the full model the user has the opportunity to download the operational data set for further analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>squidApp(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="squidApp_+3A_...">...</code></td>
<td>
<p>any argument that could be passed to the function <code><a href="shiny.html#topic+runApp">runApp</a></code> from the <a href="https://cran.r-project.org/package=shiny">shiny</a> package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>SQuID application will by default show up on your default web browser (we recommend to update your web browser to its latest version). In order to use properly the SQuID application, we encourage you to start by reading the documentation on the portal page. From there you will be guided through SQuID application according to your experience and knowledge of multilevel modelling.
</p>
<p>For more advanced and efficient simulations, SQuID is also available as an R function <code><a href="#topic+squidR">squidR</a></code>.
</p>
<p>Note that while the SQuID application is running, your <code>R</code> session will be busy. Then, in order to exit the SQuID application mode, you have to press the &quot;Esc&quot; key while you are on your <code>R</code> session window.
</p>


<h3>References</h3>

<p>Allegue, H., Araya-Ajoy, Y.G., Dingemanse, N.J., Dochtermann N.A., Garamszegi, L.Z., Nakagawa, S., Reale, D., Schielzeth, H. and Westneat, D.F. (2016) SQuID - Statistical Quantification of Individual Differences: an educational and statistical tool for understanding multi-level phenotypic data in linear mixed models. <em>Methods in Ecology and Evolution</em>, <b>8</b>:257-267.
</p>
<p>Dingemanse, N.J. and Dochtermann N.A. (2013) Quantifying individual variation in behaviour: mixed-effect modelling approaches. <em>Journal of Animal Ecology</em>, <b>82</b>:39-54.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+squidR">squidR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># run SQuID application
## Not run: squidApp()

# press the "Esc" key to exit SQuID application mode. 
</code></pre>

<hr>
<h2 id='squidR'>Run SQuID without graphical user interface</h2><span id='topic+squidR'></span>

<h3>Description</h3>

<p>The function simulates a world inhabited by individuals whose phenotypes are generated by a user-defined phenotypic equation and then samples from this world according to a user-defined sampling design. If this is your first experience with SQuID, we highly recommend to start with the interactive SQuID application with graphical interface in order to learn and get more familiarized with the SQuID model and its sampling design (use the function <code><a href="#topic+squidApp">squidApp</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>squidR(input=list(), plot=FALSE, data=NULL, module=NULL, 
       X_previsualization=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="squidR_+3A_input">input</code></td>
<td>
<p>A <code>list</code> of the model input parameters (see Details).</p>
</td></tr>
<tr><td><code id="squidR_+3A_plot">plot</code></td>
<td>
<p><code>logical</code>; If <code>FALSE</code> (default), <code>squidR</code> does not include the simulation plots into the output list (see Value).</p>
</td></tr>
<tr><td><code id="squidR_+3A_data">data</code></td>
<td>
<p>A <code>data.frame</code> of the full data (the &quot;world&quot;) returned by <code>squidR</code> (<code>output$full_data</code>).</p>
</td></tr>
<tr><td><code id="squidR_+3A_module">module</code></td>
<td>
<p>A <code>character string</code> of the module name. This argument is only used by SQuID app.</p>
</td></tr>
<tr><td><code id="squidR_+3A_x_previsualization">X_previsualization</code></td>
<td>
<p>A <code>character string</code> of the environment name. This argument is only used by SQuID app.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A detailed description of the SQuID model and its sampling design is provided on the SQuID app (run SQuID app with <code><a href="#topic+squidApp">squidApp</a></code>). Reading the SQuID documentation on the SQuID app is prerequisite to use properly the SQuID R function.
</p>
<p>The argument <code>input</code> is a list that contains the different input parameters to the SQuID model. All the parameters are listed below including their default value and a short description. If a parameter is not declared within the <code>input</code> list, the default value will be used.
</p>

<ul>
<li> <p><code>Tmax</code> (default 1): positive <code>integer</code>; the duration of the simulation (number of time steps).
</p>
</li>
<li> <p><code>NP</code> (default 1): positive <code>integer</code>; the number of replicated populations. The replicated populations are generated independently using the same simulation design that has been initially inputted by the user. Note that data from different populations will be saved in the same output file that you could later use to run statistical analyses.
</p>
</li>
<li> <p><code>NI</code> (default 1): positive <code>integer</code>; the number of individuals in each sampled population. <code>NI</code> must be divisible by the number of higher-level groups (<code>NG</code>).
</p>
</li>
<li> <p><code>NT</code> (default 1): positive <code>integer</code>; the number of traits for each individual. SQuID allows a maximum of 2 traits.
</p>
</li>
<li> <p><code>NG</code> (default 1): positive <code>integer</code>; the number of higher-level groups. <code>NG</code> must be lower than the number of individuals <code>NI</code>.
</p>
</li>
<li> <p><code>X1_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, an environmental effect x1 will be added to the model equation.
</p>
</li>
<li> <p><code>X1_sto_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, a stochastic environmental effect will be added to the environmental effect x1. The stochastic environmental effect values follow a normal distribution with a mean of 0 and a variance of <code>X1_sto_V</code>.
</p>
</li>
<li> <p><code>X1_sto_shared</code> (default <code>TRUE</code>): logical; if <code>TRUE</code>, the stochastic environmental effect included in the environmental effect x1 will be shared between individuals. If <code>FALSE</code>, each individual will experience a different stochastic environmental effect x1 (i.e. a new stochastic environmental effect is generated for each individual).
</p>
</li>
<li> <p><code>X1_sto_V</code> (default 1): <code>numeric</code>; the variance used to generate the normally distributed stochastic environmental effect x1.
</p>
</li>
<li> <p><code>X1_sto_autocor_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, an autocorrelation effect will be added to the stochastic environmental effect x1.
</p>
</li>
<li> <p><code>X1_sto_corr</code> (default 0): <code>numeric</code>; the correlation value ranging from 0 to 1 that characterizes the magnitude of the temporal autocorrelation between two consecutive time steps.
</p>
</li>
<li> <p><code>X1_lin_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, a linear trend will be added to the environmental effect x1.
</p>
</li>
<li> <p><code>X1_lin_shared</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, the linear trend included in the environmental effect x1 will be shared between individuals. If <code>FALSE</code>, each individual will experience a different linear trend within the environmental effect x1. A different linear trend is generated for each individual by varying the intercept and the slope of the linear equation. The variation in intercepts and slopes following a normal distribution with means <code>X1_lin_intercept</code>, <code>X1_lin_slope</code> and variance <code>X1_lin_V</code>.
</p>
</li>
<li> <p><code>X1_lin_intercept</code> (default 0): <code>numeric</code>; the intercept of the linear trend included into the environmental effect x1.
</p>
</li>
<li> <p><code>X1_lin_slope</code> (default 1): <code>numeric</code>; the slope of the linear trend included into the environmental effect x1.
</p>
</li>
<li> <p><code>X1_lin_V</code> (default 1): <code>numeric</code>; the variance used to generate different (unshared) linear trend effect for each individual within the environmental effect x1.
</p>
</li>
<li> <p><code>X1_cyc_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, a cyclic trend will be added the environmental effect x1.
</p>
</li>
<li> <p><code>X1_cyc_shared</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, the cyclic trend for environmental effect x1 will be shared between individuals. If <code>FALSE</code>, each individual will experience a different cyclic trend for environmental effect x1. A new cyclic trend is generated for each individual by varying the amplitude, the period, the horizontal shift and the vertical shift of the cyclic equation following a normal distribution where the mean is the parameter value itself and the variance is <code>X1_cyc_V</code>.
</p>
</li>
<li> <p><code>X1_cyc_amplitude</code> (default 10): positive <code>numeric</code>; the amplitude of the cyclic trend for environmental effect x1.
</p>
</li>
<li> <p><code>X1_cyc_period</code> (default 10): positive <code>numeric</code>; the period of the cyclic trend for the environmental effect x1.
</p>
</li>
<li> <p><code>X1_cyc_Hshift</code> (default 0): positive <code>numeric</code>; the horizontal shift of the cyclic trend for environmental effect x1.
</p>
</li>
<li> <p><code>X1_cyc_Vshift</code> (default 0): positive <code>numeric</code>; the vertical shift of the cyclic trend for environmental effect x1.
</p>
</li>
<li> <p><code>X1_cyc_V</code> (default 0): <code>numeric</code>; the variance used to generate different (unshared) cyclic trends for each individual in the environmental effect x1.
</p>
</li>
<li> <p><code>X2_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, an environmental effect x2 will be added to the model equation.
</p>
</li>
<li> <p><code>X2_sto_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, a stochastic environmental effect will be added to the environmental effect x2. The stochastic environmental effect values follow a normal distribution with mean of 0 and a variance of <code>X2_sto_V</code>.
</p>
</li>
<li> <p><code>X2_sto_shared</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, the stochastic environmental effect included in the environmental effect x2 will be shared between individuals. If <code>FALSE</code>, each individual will experience a different stochastic environmental effect  x2 (i.e. a new stochastic environmental effect is generated for each individual).
</p>
</li>
<li> <p><code>X2_sto_V</code> (default 1): <code>numeric</code>; the variance for the normally distributed stochastic environmental effect  x2.
</p>
</li>
<li> <p><code>X2_sto_autocor_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, an autocorrelation effect will be added to the stochastic environmental effect x2.
</p>
</li>
<li> <p><code>X2_sto_corr</code> (default 0): <code>numeric</code>; the correlation value ranging from 0 to 1 that characterizes the magnitude of the temporal autocorrelation between two consecutive time steps.
</p>
</li>
<li> <p><code>X2_lin_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, a linear trend will be added to the environmental effect x2.
</p>
</li>
<li> <p><code>X2_lin_shared</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, the linear trend included in the environmental effect x2 will be shared between individuals. If <code>FALSE</code>, each individual will experience a different linear trend within the environmental effect x2. A new linear trend is generated for each individual by varying the intercept and the slope of the linear equation. The variation in intercepts and slopes follow a normal distribution with means <code>X2_lin_intercept</code>, <code>X2_lin_slope</code> and variance <code>X2_lin_V</code>.
</p>
</li>
<li> <p><code>X2_lin_intercept</code> (default 0): <code>numeric</code>; the intercept of the linear trend for environmental effect x2.
</p>
</li>
<li> <p><code>X2_lin_slope</code> (default 1): <code>numeric</code>; the slope of the linear trend for environmental effect x2.
</p>
</li>
<li> <p><code>X2_lin_V</code> (default 1): <code>numeric</code>; the variance used to generate different (unshared) linear trend effect for each individual's environmental effect x2.
</p>
</li>
<li> <p><code>X2_cyc_state</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, a cyclic trend will be added the environmental effect x2.
</p>
</li>
<li> <p><code>X2_cyc_shared</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, the cyclic trend included in the environmental effect x2 will be shared between individuals. If <code>FALSE</code>, each individual will experience a different cyclic trend for the environmental effect x2. A new cyclic trend is generated for each individual by varying the amplitude, the period, the horizontal shift and the vertical shift of the cyclic equation following a normal distribution where the mean is the parameter value itself and the variance is <code>X2_cyc_V</code>.
</p>
</li>
<li> <p><code>X2_cyc_amplitude</code> (default 10): positive <code>numeric</code>; the amplitude of the cyclic trend included into the environmental effect x2.
</p>
</li>
<li> <p><code>X2_cyc_period</code> (default 10): positive <code>numeric</code>; the period of the cyclic trend for environmental effect x2.
</p>
</li>
<li> <p><code>X2_cyc_Hshift</code> (default 0): positive <code>numeric</code>; the horizontal shift of the cyclic trend for environmental effect x2.
</p>
</li>
<li> <p><code>X2_cyc_Vshift</code> (default 0): positive <code>numeric</code>; the vertical shift of the cyclic trend for environmental effect x2.
</p>
</li>
<li> <p><code>X2_cyc_V</code> (default 0): <code>numeric</code>; the variance used to generate different (unshared) cyclic trend effect for each individual within the environmental effect x2.
</p>
</li>
<li> <p><code>X_Interaction</code> (default <code>FALSE</code>): <code>logical</code>; if <code>TRUE</code>, the environmental effect x1*x2 representing the interaction between the environmental effect x1 and the environmental effect x2 will be added to the model equation.
</p>
</li>
<li> <p><code>B</code> (default <code>c(0,0,0,0)</code>): <code>numeric vector</code>; the population mean values. When the number of trait is set to 1, the length of the vector is 4. The vector elements are respectively the population intercept (b0), the population  response to environmental influences x1 (fixed slope; b1), the population  response to environmental influences x2 (fixed slope; b2) and the population  response interaction between environmental influences x1 and x2 (fixed slope; b12). If the number of trait is set to 2, the length of the vector is doubled (len=8) and the second half corresponds to the population mean values associated with the second trait.
</p>
</li>
<li> <p><code>Vind</code> (default <code>matrix(0, nrow=4, ncol=4)</code>): square, positive definite <code>matrix</code> or lower triangular <code>matrix</code>; the individual variance/correlation matrix. Variances are positive <code>numeric</code> numbers and correlations are <code>numeric</code> numbers ranged between -1 and 1. Variances are along the matrix diagonal and correlation values below the matrix diagonal. When the number of trait is set to 1, the dimension of the matrix is 4 rows by 4 columns (4x4) and the diagonal elements are respectively the individual-specific  (random) intercept variance (VI), the individual-specific response to an environmental effect x1 (random slope) variance (VS1), the individual-specific response to an environmental effect x2 (random slope) variance (VS2) and the individual-specific response interaction to two environmental effects x1 and x2 (random slopes) variance (VS12). When the number of trait is set to 2, the dimension of the matrix is doubled (8x8) where the second half of the diagonal is the individual variances (intercept and slopes) of the second trait. Note that it is possible to set correlation relationships between random effects within and/or among traits.
</p>
</li>
<li> <p><code>Ve</code> (default 0): positive <code>numeric</code>; the measurement error variance.
</p>
</li>
<li> <p><code>VG</code> (default 0): positive <code>numeric</code>; the high-level grouping variance.
</p>
</li>
<li> <p><code>NR</code> (default 1): positive <code>integer</code>; the average number of records sampled per individual.
</p>
</li>
<li> <p><code>Vhsi</code> (default 0): <code>numeric</code> (between 0 and 0.95); the among-individual variance in timing of sampling.
</p>
</li>
<li> <p><code>NR_ind</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, individuals will be sampled the same number of times.
</p>
</li>
<li> <p><code>NR_trait</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, traits within individuals will be sampled the same number of times.
</p>
</li>
<li> <p><code>ST_ind</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, individuals will be sampled at the same time (i.e. recorded simultaneously).
</p>
</li>
<li> <p><code>ST_trait</code> (default <code>TRUE</code>): <code>logical</code>; if <code>TRUE</code>, traits within individuals will be sampled at the same times (i.e. recorded simultaneously).
</p>
</li></ul>

<p>When the full data of a previous squid simulation is passed to <code>squidR</code> through the argument <code>data</code>, <code>squidR</code> does not generate a new world and samples directly into the already existing world.
</p>


<h3>Value</h3>

<p><code>squidR</code> returns a <code>list</code> that includes a <code>data.frame</code> of the full data generated (the &quot;world&quot;), a <code>data.frame</code> of the sampled data and six <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> plots of the results (only if <code>plot</code> is set to <code>TRUE</code>).
</p>
<p>The full and the sampled data could be accessed respectively by <code>output$full_data</code> and <code>output$sampled_data</code>.
A description of the columns of the full and the sampled <code>data.frame</code> is provided below:
</p>

<ul>
<li> <p><code>Replicate</code>: (factor) the replicated population identifier.
</p>
</li>
<li> <p><code>Individual</code>: (factor) the individual identifier.
</p>
</li>
<li> <p><code>Group</code>: (factor) the high-level grouping identifier.
</p>
</li>
<li> <p><code>Individual_Trait</code>: (factor) the identifier of each individual/trait combination.
</p>
</li>
<li> <p><code>Trait</code>: (factor) the trait identifier.
</p>
</li>
<li> <p><code>Time</code>: (integer) the time step value.
</p>
</li>
<li> <p><code>Phenotype</code>: (numeric) the individual phenotype.
</p>
</li>
<li> <p><code>Beta0</code>: (numeric) the population phenotypic mean.
</p>
</li>
<li> <p><code>Beta1</code>: (numeric) the population mean response to environmental influences x1.
</p>
</li>
<li> <p><code>Beta2</code>: (numeric) the population mean response to environmental influences x2.
</p>
</li>
<li> <p><code>Beta12</code>: (numeric) the population mean response to environmental influences x1x2 (the interaction between the environmental effect x1 and the environmental effect x2).
</p>
</li>
<li> <p><code>I</code>: (numeric) the individual-specific deviations (random-intercepts) from the population phenotypic mean <code>Beta0</code>.
</p>
</li>
<li> <p><code>S1</code>: (numeric) the individual-specific response to the environmental influence x1 (random-slope).
</p>
</li>
<li> <p><code>S2</code>: (numeric) the individual-specific response to the environmental influence x2 (random-slope).
</p>
</li>
<li> <p><code>S12</code>: (numeric) the individual-specific response to the environmental influence x1x2 (the interaction between the environmental effect x1 and the environmental effect x2; random-slope).
</p>
</li>
<li> <p><code>X1</code>: (numeric) the environmental gradient x1.
</p>
</li>
<li> <p><code>X2</code>: (numeric) the environmental gradient x2.
</p>
</li>
<li> <p><code>X1X2</code>: (numeric) the environmental gradient x1x2 (the interaction between the environmental effect x1 and the environmental effect x2).
</p>
</li>
<li> <p><code>G</code>: (numeric) the higher-level grouping value.
</p>
</li>
<li> <p><code>e</code>: (numeric) the measurement error.
</p>
</li></ul>

<p><code>squidR</code> returns a list that also includes a list (<code>plots</code>) of six <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> plots of the results. Note that the plots display only the data of the first trait, the first replicate and the first 20 individuals of the simulation.
</p>

<ul>
<li> <p><code>$plots$X1</code>: a plot of the environmental gradient x1. 
</p>
</li>
<li> <p><code>$plots$X2</code>: a plot of the environmental gradient x2. 
</p>
</li>
<li> <p><code>$plots$X1X2</code>: a plot of the environmental gradient x1x2 (the interaction between the environmental effect x1 and the environmental effect x2). 
</p>
</li>
<li> <p><code>$plots$totPhen</code>: a plot of the raw individual phenotype over time.
</p>
</li>
<li> <p><code>$plots$sampPhen</code>: a plot of the sampled individual phenotype over time.
</p>
</li>
<li> <p><code>$plots$sampTime</code>: a plot of the sampling time of each individual.
</p>
</li></ul>



<h3>References</h3>

<p>Allegue, H., Araya-Ajoy, Y.G., Dingemanse, N.J., Dochtermann N.A., Garamszegi, L.Z., Nakagawa, S., Reale, D., Schielzeth, H. and Westneat, D.F. (2016) SQuID - Statistical Quantification of Individual Differences: an educational and statistical tool for understanding multi-level phenotypic data in linear mixed models. <em>Methods in Ecology and Evolution</em>, <b>8</b>:257-267.
</p>
<p>Dingemanse, N.J. and Dochtermann N.A. (2013) Quantifying individual variation in behaviour: mixed-effect modelling approaches. <em>Journal of Animal Ecology</em>, <b>82</b>:39-54.
</p>


<h3>Note</h3>

<p>If this is your first experience with SQuID, we highly recommend to start with the interactive SQuID application with graphical interface in order to learn and get more familiarized with the SQuID model (phenotypic equation) and its sampling design (use the function <code><a href="#topic+squidApp">squidApp</a></code>).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+squidApp">squidApp</a></code> to run the SQuID app which includes detailed documentation on the SQuID model and its sampling design.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## create the input list ##
input &lt;- list()

## define the model input parameters ##

input$Tmax &lt;- 100 # 100 time steps
input$NP   &lt;- 10  # 10 replicated populations
input$NI   &lt;- 20  # 20 individuals per replicate population
input$NT   &lt;- 1   # 1 trait per individual

# in this simulation model, only one environmental gradient (x1) is added

# turn on the environmental gradient x1
input$X1_state      &lt;- TRUE  

# add a stochastic environmental effect to the environmental effect x1
input$X1_sto_state  &lt;- TRUE  

# the stochastic environmental effect included 
# into the environmental effect x1 is not shared among individual. 
# Each individual will experience a  different stochastic environmental effect. 
input$X1_sto_shared &lt;- FALSE                  

#  add an autocorrelation effect to the stochastic 
#  environmental effect within the environmental effect x1.
input$X1_sto_autocor_state &lt;- FALSE 

# define the autocorrelation value between two consecutive time steps.
input$X1_sto_corr          &lt;- 0.5 

# add a linear trend to the environmental effect x1. 
# the default parameters for the linear trend will be used 
# (an intercept of 0, a slope of 1 and the linear trend will be shared among individuals)
input$X1_lin_state &lt;- TRUE 

# define the population mean values vector.
# the population mean (fixed intercept) and
# the population mean response to environmental influences x1 (fixed slope) 
# are set to 0.1.
input$B    &lt;- c(0.1, 0.1, 0, 0)
                                                  
# define the individual variance/correlation matrix (squared matrix; 4x4)
# the individual-specific deviation (random intercept; VI) variance is set to 0.7.
# the individual-specific response to an environmental effect x1 (random slope; VS1) is set to 0.5.
# the correlation between the individual-specific deviation (I) 
# and the individual-specific response 
# to an environmental effect x1 
# (S1) is set to -0.7.                                                     
input$Vind &lt;- matrix(c(0.7  , 0   , 0 , 0,
                       -0.7 , 0.5 , 0 , 0,
                       0    , 0   , 0 , 0,
                       0    , 0   , 0 , 0), 
			nrow=4, byrow=TRUE)
                       
input$Ve  &lt;- 0.05 # the measurement error variance

## define the sampling design ##

# the mean number of records per individual
input$NR     &lt;- 10
# the among-individual variance in timing of sampling
input$Vhsi   &lt;- 0.2
# individuals are not sampled the same number of times.
input$NR_ind &lt;- FALSE 
# individuals are not sampled at the time
input$ST_ind &lt;- FALSE 

## run the simulation ##
output &lt;- squidR(input, plot=TRUE)

# plot the individual phenotype values over time
print(output$plot$totPhen) 

# see first records of the simulated data set
head(output$full_data)

## run a second simulation with the identical world but with a different sampling design 
## where individuals are sampled the same number of times and at the same time    

input$NR_ind &lt;- TRUE # individuals are sampled the same number of times.
input$ST_ind &lt;- TRUE # individuals are sampled at the time

output2 &lt;- squidR(input, plot=TRUE, data=output$full_data)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
