<!DOCTYPE html><html><head><title>Help for package sRDA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sRDA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#generate_data'><p>Generate data sets for sparse multivariate analysis</p></a></li>
<li><a href='#sCCA'><p>Sparse Canonical Correlation analysis</p></a></li>
<li><a href='#sRDA'><p>sRDA.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Sparse Redundancy Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-12-12</td>
</tr>
<tr>
<td>Author:</td>
<td>Attila Csala [aut, cre],
  Koos Zwinderman [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Attila Csala &lt;a@csala.me&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Sparse redundancy analysis
	for high dimensional (biomedical) data. Directional multivariate analysis 
	to express the maximum variance in the predicted data set by a linear 
	combination of variables of the predictive data set. Implemented in a 
	partial least squares framework, for more details see Csala et al. (2017) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx374">doi:10.1093/bioinformatics/btx374</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.7), Matrix, doParallel, elasticnet, foreach, mvtnorm</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-12-14 10:24:28 UTC; acsala</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-12-14 13:36:54 UTC</td>
</tr>
</table>
<hr>
<h2 id='generate_data'>Generate data sets for sparse multivariate analysis</h2><span id='topic+generate_data'></span>

<h3>Description</h3>

<p>Generate two data sets with highly correlated and noise variables modeled
in a multiple latent variable structure. The latent variables are orthogonal
to each other thus capture a different portion of association between the
involved data sets. Thu function generates data that can be used to verify
sRDA's ability of finding the highly correlated variables accross multiple
latent variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_data(nr_LVs = 1, n = 50, nr_correlated_Xs = c(5),
  nr_uncorrelated_Xs = 250, mean_reg_weights_assoc_X = c(0.7),
  sd_reg_weights_assoc_X = c(0.05), Xnoise_min = -0.3, Xnoise_max = 0.3,
  nr_correlated_Ys = c(5), nr_uncorrelated_Ys = 350,
  mean_reg_weights_assoc_Y = c(0.7), sd_reg_weights_assoc_Y = c(0.05),
  Ynoise_min = -0.3, Ynoise_max = 0.3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_data_+3A_nr_lvs">nr_LVs</code></td>
<td>
<p>The number of latent variables between the predicitve
and predicted data sets. The latent variables model the association between
data sets.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_n">n</code></td>
<td>
<p>The number of observations (rows) in the data sets.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_nr_correlated_xs">nr_correlated_Xs</code></td>
<td>
<p>Number of variables of the
predictive data set that are associated with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_nr_uncorrelated_xs">nr_uncorrelated_Xs</code></td>
<td>
<p>Number of variables of the predictive data
set that is not associated with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_mean_reg_weights_assoc_x">mean_reg_weights_assoc_X</code></td>
<td>
<p>Mean of the
regression weights of the predictive varaibles that are associated with the
latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_sd_reg_weights_assoc_x">sd_reg_weights_assoc_X</code></td>
<td>
<p>Standard deviation
of the regression weights of the predictive varaibles that are associated
with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_xnoise_min">Xnoise_min</code></td>
<td>
<p>The lower bound of the unifrom distribution that is used to
sample the values for the regression weights of the predictive varaibles that
are not associated with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_xnoise_max">Xnoise_max</code></td>
<td>
<p>The upper bound of the unifrom distribution that is used to
sample the values for the regression weights of the predictive varaibles that
are not associated with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_nr_correlated_ys">nr_correlated_Ys</code></td>
<td>
<p>Number of variables of the
predictive data set that are associated with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_nr_uncorrelated_ys">nr_uncorrelated_Ys</code></td>
<td>
<p>Number of variables of the predicted data
set that is not associated with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_mean_reg_weights_assoc_y">mean_reg_weights_assoc_Y</code></td>
<td>
<p>Mean of the
regression weights of the predicted varaibles that are associated with the
latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_sd_reg_weights_assoc_y">sd_reg_weights_assoc_Y</code></td>
<td>
<p>Standard deviation
of the regression weights of the predicted varaibles that are associated
with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_ynoise_min">Ynoise_min</code></td>
<td>
<p>The lower bound of the unifrom distribution that is used to
sample the values for the regression weights of the predicted varaibles that
are not associated with the latent variables.</p>
</td></tr>
<tr><td><code id="generate_data_+3A_ynoise_max">Ynoise_max</code></td>
<td>
<p>The upper bound of the unifrom distribution that is used to
sample the values for the regression weights of the prediced varaibles that
are not associated with the latent variables.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># generate data with few highly correlated variahbles
dataXY &lt;- generate_data(nr_LVs = 2,
                           n = 250,
                           nr_correlated_Xs = c(5,20),
                           nr_uncorrelated_Xs = 250,
                           mean_reg_weights_assoc_X =
                             c(0.9,0.5),
                           sd_reg_weights_assoc_X =
                             c(0.05, 0.05),
                           Xnoise_min = -0.3,
                           Xnoise_max = 0.3,
                           nr_correlated_Ys = c(10,15),
                           nr_uncorrelated_Ys = 350,
                           mean_reg_weights_assoc_Y =
                             c(0.9,0.6),
                           sd_reg_weights_assoc_Y =
                             c(0.05, 0.05),
                           Ynoise_min = -0.3,
                           Ynoise_max = 0.3)

# seperate predictor and predicted sets
X &lt;- dataXY$X
Y &lt;- dataXY$Y

dim(X);dim(Y)

</code></pre>

<hr>
<h2 id='sCCA'>Sparse Canonical Correlation analysis</h2><span id='topic+sCCA'></span>

<h3>Description</h3>

<p>Sparse Canonical Correlation analysis for high dimensional (biomedical) data.
The function takes two datasets and returns a linear combination of
maximally correlated canonical correlate pairs.  Elastic net penalization
(with its variants, UST, Ridge and Lasso penalization) is implemented for
sparsity and smoothnesswith a  built in cross validation procedure to obtain
the optimal penalization parameters. It is possible to obtain multiple
canonical variate pairs that are orthogonal to each other.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sCCA(predictor, predicted, penalization = "enet", ridge_penalty = 1,
  nonzero = 1, max_iterations = 100, tolerance = 1 * 10^-20,
  cross_validate = FALSE, parallel_CV = TRUE, nr_subsets = 10,
  multiple_LV = FALSE, nr_LVs = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sCCA_+3A_predictor">predictor</code></td>
<td>
<p>The n*p matrix of the predictor data set</p>
</td></tr>
<tr><td><code id="sCCA_+3A_predicted">predicted</code></td>
<td>
<p>The n*q matrix of the predicted data set</p>
</td></tr>
<tr><td><code id="sCCA_+3A_penalization">penalization</code></td>
<td>
<p>The penalization method applied during the analysis (none, enet or ust)</p>
</td></tr>
<tr><td><code id="sCCA_+3A_ridge_penalty">ridge_penalty</code></td>
<td>
<p>The ridge penalty parameter of the predictor set's latent variable used
for enet or ust (an integer if cross_validate = FALE, a list otherwise)</p>
</td></tr>
<tr><td><code id="sCCA_+3A_nonzero">nonzero</code></td>
<td>
<p>The number of non-zero weights of the predictor set's latent variable
(an integer if cross_validate = FALE, a list otherwise)</p>
</td></tr>
<tr><td><code id="sCCA_+3A_max_iterations">max_iterations</code></td>
<td>
<p>The maximum number of iterations of the algorithm</p>
</td></tr>
<tr><td><code id="sCCA_+3A_tolerance">tolerance</code></td>
<td>
<p>Convergence criteria</p>
</td></tr>
<tr><td><code id="sCCA_+3A_cross_validate">cross_validate</code></td>
<td>
<p>K-fold cross validation to find best optimal penalty parameters
(TRUE or FALSE)</p>
</td></tr>
<tr><td><code id="sCCA_+3A_parallel_cv">parallel_CV</code></td>
<td>
<p>Run the cross validation parallel (TRUE or FALSE)</p>
</td></tr>
<tr><td><code id="sCCA_+3A_nr_subsets">nr_subsets</code></td>
<td>
<p>Number of subsets for k-fold cross validation</p>
</td></tr>
<tr><td><code id="sCCA_+3A_multiple_lv">multiple_LV</code></td>
<td>
<p>Obtain multiple latent variable pairs (TRUE or FALSE)</p>
</td></tr>
<tr><td><code id="sCCA_+3A_nr_lvs">nr_LVs</code></td>
<td>
<p>Number of latent variables to be obtained</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"sRDA"</code>.
</p>
<table>
<tr><td><code>XI</code></td>
<td>

<p>Predictor set's latent variable scores</p>
</td></tr>
<tr><td><code>ETA</code></td>
<td>

<p>Predictive set's latent variable scores</p>
</td></tr>
<tr><td><code>ALPHA</code></td>
<td>

<p>Weights of the predictor set's latent variable</p>
</td></tr>
<tr><td><code>BETA</code></td>
<td>

<p>Weights of the predicted set's latent variable</p>
</td></tr>
<tr><td><code>nr_iterations</code></td>
<td>

<p>Number of iterations ran before convergence (or max number of iterations)</p>
</td></tr>
<tr><td><code>SOLVE_XIXI</code></td>
<td>

<p>Inverse of the predictor set's latent variable variance matrix</p>
</td></tr>
<tr><td><code>iterations_crts</code></td>
<td>

<p>The convergence criterion value (a small positive tolerance)</p>
</td></tr>
<tr><td><code>sum_absolute_betas</code></td>
<td>

<p>Sum of the absolute values of beta weights</p>
</td></tr>
<tr><td><code>ridge_penalty</code></td>
<td>

<p>The ridge penalty parameter used for the model</p>
</td></tr>
<tr><td><code>nr_nonzeros</code></td>
<td>

<p>The number of nonzero alpha weights in the model</p>
</td></tr>
<tr><td><code>nr_latent_variables</code></td>
<td>

<p>The number of latient variable pairs in the model</p>
</td></tr>
<tr><td><code>CV_results</code></td>
<td>

<p>The detailed results of cross validations (if cross_validate is TRUE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Attila Csala
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# generate data with few highly correlated variahbles
dataXY &lt;- generate_data(nr_LVs = 2,
                           n = 250,
                           nr_correlated_Xs = c(5,20),
                           nr_uncorrelated_Xs = 250,
                           mean_reg_weights_assoc_X =
                             c(0.9,0.5),
                           sd_reg_weights_assoc_X =
                             c(0.05, 0.05),
                           Xnoise_min = -0.3,
                           Xnoise_max = 0.3,
                           nr_correlated_Ys = c(10,15),
                           nr_uncorrelated_Ys = 350,
                           mean_reg_weights_assoc_Y =
                             c(0.9,0.6),
                           sd_reg_weights_assoc_Y =
                             c(0.05, 0.05),
                           Ynoise_min = -0.3,
                           Ynoise_max = 0.3)

# seperate predictor and predicted sets
X &lt;- dataXY$X
Y &lt;- dataXY$Y

# run sRDA
CCA.res &lt;- sCCA(predictor = X, predicted = Y, nonzero = 5,
ridge_penalty = 1, penalization = "ust")


# check first 10 weights of X
CCA.res$ALPHA[1:10]

## Not run: 
# run sRDA with cross-validation to determine best penalization parameters
CCA.res &lt;- sCCA(predictor = X, predicted = Y, nonzero = c(5,10,15),
ridge_penalty = c(0.1,1), penalization = "enet", cross_validate = TRUE,
parallel_CV = TRUE)

# check first 10 weights of X
CCA.res$ALPHA[1:10]
CCA.res$ridge_penalty
CCA.res$nr_nonzeros

# obtain multiple latent variables
CCA.res &lt;- sCCA(predictor = X, predicted = Y, nonzero = c(5,10,15),
ridge_penalty = c(0.1,1), penalization = "enet", cross_validate = TRUE,
parallel_CV = TRUE, multiple_LV = TRUE, nr_LVs = 2, max_iterations = 5)

# check first 10 weights of X in first two component
CCA.res$ALPHA[[1]][1:10]
CCA.res$ALPHA[[2]][1:10]

# latent variables are orthogonal to each other
t(CCA.res$XI[[1]]) %*% CCA.res$XI[[2]]


## End(Not run)
</code></pre>

<hr>
<h2 id='sRDA'>sRDA.</h2><span id='topic+sRDA'></span><span id='topic+sRDA-package'></span><span id='topic+sRDA'></span>

<h3>Description</h3>

<p>sRDA.
</p>
<p>Sparse Redundancy Analysis (sRDA) to express the maximum variance in the
predicted data set by a linear combination of variables (latent variable)
of the predictive data set. Elastic net penalization (with its variants, UST,
Ridge and Lasso penalization) is implemented for sparsity and smoothness
with a  built in cross validation procedure to obtain the optimal
penalization parameters. It is possible to obtain multiple latent variables
which are orthogonal to each other, thus each explaining a different protion
of variance in the predicted data set. sRDA is implemented in a Partial Least
Squares framework, for more details see Csala et al. (2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sRDA(predictor, predicted, penalization = "enet", ridge_penalty = 1,
  nonzero = 1, max_iterations = 100, tolerance = 1 * 10^-20,
  cross_validate = FALSE, parallel_CV = FALSE, nr_subsets = 10,
  multiple_LV = FALSE, nr_LVs = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sRDA_+3A_predictor">predictor</code></td>
<td>
<p>The n*p matrix of the predictor data set</p>
</td></tr>
<tr><td><code id="sRDA_+3A_predicted">predicted</code></td>
<td>
<p>The n*q matrix of the predicted data set</p>
</td></tr>
<tr><td><code id="sRDA_+3A_penalization">penalization</code></td>
<td>
<p>The penalization method applied during the analysis (none, enet or ust)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_ridge_penalty">ridge_penalty</code></td>
<td>
<p>The ridge penalty parameter of the predictor set's latent variable used
for enet (an integer if cross_validate = FALSE, a list otherwise)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_nonzero">nonzero</code></td>
<td>
<p>The number of non-zero weights of the predictor set's latent variable used
for enet or ust (an integer if cross_validate = FALSE, a list otherwise)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_max_iterations">max_iterations</code></td>
<td>
<p>The maximum number of iterations of the algorithm (integer)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_tolerance">tolerance</code></td>
<td>
<p>Convergence criteria (number, a small positive tolerance)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_cross_validate">cross_validate</code></td>
<td>
<p>K-fold cross validation to find best optimal penalty parameters
(TRUE or FALSE)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_parallel_cv">parallel_CV</code></td>
<td>
<p>Run the cross validation parallel (TRUE or FALSE)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_nr_subsets">nr_subsets</code></td>
<td>
<p>Number of subsets for k-fold cross validation (integer, the value for k)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_multiple_lv">multiple_LV</code></td>
<td>
<p>Obtain multiple latent variable pairs (TRUE or FALSE)</p>
</td></tr>
<tr><td><code id="sRDA_+3A_nr_lvs">nr_LVs</code></td>
<td>
<p>Number of latent variable pairs (components) to be obtained (integer)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"sRDA"</code>.
</p>
<table>
<tr><td><code>XI</code></td>
<td>

<p>Predictor set's latent variable scores</p>
</td></tr>
<tr><td><code>ETA</code></td>
<td>

<p>Predictive set's latent variable scores</p>
</td></tr>
<tr><td><code>ALPHA</code></td>
<td>

<p>Weights of the predictor set's latent variable</p>
</td></tr>
<tr><td><code>BETA</code></td>
<td>

<p>Weights of the predicted set's latent variable</p>
</td></tr>
<tr><td><code>nr_iterations</code></td>
<td>

<p>Number of iterations ran before convergence (or max number of iterations)</p>
</td></tr>
<tr><td><code>SOLVE_XIXI</code></td>
<td>

<p>Inverse of the predictor set's latent variable variance matrix</p>
</td></tr>
<tr><td><code>iterations_crts</code></td>
<td>

<p>The convergence criterion value (a small positive tolerance)</p>
</td></tr>
<tr><td><code>sum_absolute_betas</code></td>
<td>

<p>Sum of the absolute values of beta weights</p>
</td></tr>
<tr><td><code>ridge_penalty</code></td>
<td>

<p>The ridge penalty parameter used for the model</p>
</td></tr>
<tr><td><code>nr_nonzeros</code></td>
<td>

<p>The number of nonzero alpha weights in the model</p>
</td></tr>
<tr><td><code>nr_latent_variables</code></td>
<td>

<p>The number of latient variable pairs (components) in the model</p>
</td></tr>
<tr><td><code>CV_results</code></td>
<td>

<p>The detailed results of cross validations (if cross_validate is TRUE)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Attila Csala
</p>


<h3>References</h3>

<p>Csala A., Voorbraak F.P.J.M., Zwinderman A.H., and Hof M.H. (2017) Sparse
redundancy analysis of high-dimensional genetic and genomic data.
<em>Bioinformatics</em>, <b>33</b>, pp.3228-3234.
<a href="https://doi.org/10.1093/bioinformatics/btx374">https://doi.org/10.1093/bioinformatics/btx374</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># generate data with few highly correlated variahbles
dataXY &lt;- generate_data(nr_LVs = 2,
                           n = 250,
                           nr_correlated_Xs = c(5,20),
                           nr_uncorrelated_Xs = 250,
                           mean_reg_weights_assoc_X =
                             c(0.9,0.5),
                           sd_reg_weights_assoc_X =
                             c(0.05, 0.05),
                           Xnoise_min = -0.3,
                           Xnoise_max = 0.3,
                           nr_correlated_Ys = c(10,15),
                           nr_uncorrelated_Ys = 350,
                           mean_reg_weights_assoc_Y =
                             c(0.9,0.6),
                           sd_reg_weights_assoc_Y =
                             c(0.05, 0.05),
                           Ynoise_min = -0.3,
                           Ynoise_max = 0.3)



# seperate predictor and predicted sets
X &lt;- dataXY$X
Y &lt;- dataXY$Y

# run sRDA
RDA.res &lt;- sRDA(predictor = X, predicted = Y, nonzero = 5,
ridge_penalty = 1, penalization = "ust")


# check first 10 weights of X
RDA.res$ALPHA[1:10]

## Not run: 
# run sRDA with cross-validation to determine best penalization parameters
RDA.res &lt;- sRDA(predictor = X, predicted = Y, nonzero = c(5,10,15),
ridge_penalty = c(0.1,1), penalization = "enet", cross_validate = TRUE,
parallel_CV = TRUE)

# check first 10 weights of X
RDA.res$ALPHA[1:10]

# check the Ridge parameter and the number of nonzeros included in the model
RDA.res$ridge_penalty
RDA.res$nr_nonzeros

# check how much time the cross validation did take
RDA.res$CV_results$stime

# obtain multiple latent variables (components)
RDA.res &lt;- sRDA(predictor = X, predicted = Y, nonzero = c(5,10,15),
ridge_penalty = c(0.1,1), penalization = "enet", cross_validate = TRUE,
parallel_CV = TRUE, multiple_LV = TRUE, nr_LVs = 2, max_iterations = 5)

# check first 20 weights of X in first two component
RDA.res$ALPHA[[1]][1:20]
RDA.res$ALPHA[[2]][1:20]

# components are orthogonal to each other
t(RDA.res$XI[[1]]) %*% RDA.res$XI[[2]]


## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
