<!DOCTYPE html><html><head><title>Help for package sRNAGenetic</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sRNAGenetic}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#sRNAGenetic-package'><p>sRNAGenetic: Analysis of Small RNA Expression Changes in Hybrid Plants</p></a></li>
<li><a href='#basepreplot'><p>Generate the base frequency plot of miRNA</p></a></li>
<li><a href='#Countfiliter'><p>Filitering low expressed miRNAs based on count: Countfiliter</p></a></li>
<li><a href='#F1_miRNA_count'><p>Sequences of miRNAs from one species</p></a></li>
<li><a href='#F1_miRNA_rpm'><p>Sequences of miRNAs from one species</p></a></li>
<li><a href='#F1_sRNA_seq'><p>Sequences of sRNAs from one species</p></a></li>
<li><a href='#genetic'><p>Classification for 12 expression patterns</p></a></li>
<li><a href='#Get12Bins'><p>Genetic effects analysis: Twelve bins of expression analysis (method2)</p></a></li>
<li><a href='#GetDAtable'><p>Genetic effects analysis of miRNA: |d/a| (method 1)</p></a></li>
<li><a href='#lenplot'><p>Generate the sRNA length distribution plot</p></a></li>
<li><a href='#mirnapredata'><p>Generate the data of miRNA base frequence in each position</p></a></li>
<li><a href='#miVennData'><p>Species specific expression analysis: miVennData</p></a></li>
<li><a href='#miVennPlot'><p>Species specific expression analysis: miVennPlot</p></a></li>
<li><a href='#P1_miRNA_count'><p>Sequences of miRNAs from one species</p></a></li>
<li><a href='#P1_miRNA_rpm'><p>Sequences of miRNAs from one species</p></a></li>
<li><a href='#P1_sRNA_seq'><p>Sequences of sRNAs from one species</p></a></li>
<li><a href='#P2_miRNA_count'><p>Sequences of miRNAs from one species</p></a></li>
<li><a href='#P2_miRNA_rpm'><p>Sequences of miRNAs from one species</p></a></li>
<li><a href='#P2_sRNA_seq'><p>Sequences of sRNAs from one species</p></a></li>
<li><a href='#polyDESeq'><p>Differential expression analysis</p></a></li>
<li><a href='#Rpmfiliter'><p>Filitering low expressed miRNAs based on RPM: Rpmfiliter</p></a></li>
<li><a href='#srnapredata'><p>Generate the data of sRNA length distribution</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Analysis of Small RNA Expression Changes in Hybrid Plants</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>The most important function of the R package is the genetic effects analysis of small RNA in hybrid plants via two methods, and at the same time, it provides various forms of graph related to data characteristics and expression analysis. In terms of two classification methods, one is the calculation of the additive (a) and dominant (d), the other is the evaluation of expression level dominance by comparing the total expression of the small RNA in progeny with the expression level in the parent species.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>DESeq2, futile.logger, ggplot2, ggsci, plyr, VennDiagram</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-03-17 11:12:19 UTC; wuyuqing</td>
</tr>
<tr>
<td>Author:</td>
<td>Yu qing Wu <a href="https://orcid.org/0000-0002-6333-0926"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yu qing Wu &lt;wuyuqing0104@163.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-03-21 08:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='sRNAGenetic-package'>sRNAGenetic: Analysis of Small RNA Expression Changes in Hybrid Plants</h2><span id='topic+sRNAGenetic'></span><span id='topic+sRNAGenetic-package'></span>

<h3>Description</h3>

<p>The most important function of the R package sRNAGenetic is the genetic effects analysis of miRNA after plant polyploidization via two methods, and at the same time, it provides various forms of graph related to data characteristics and expression analysis. In terms of two classification methods, one is the calculation of the additive (a) and dominant (d), the other is the evaluation of ELD by comparing the total expression of the miRNA in allotetraploid with the expression level in the parent species.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yu qing Wu <a href="mailto:wuyuqing0104@163.com">wuyuqing0104@163.com</a> (<a href="https://orcid.org/0000-0002-6333-0926">ORCID</a>)
</p>

<hr>
<h2 id='basepreplot'>Generate the base frequency plot of miRNA</h2><span id='topic+basepreplot'></span>

<h3>Description</h3>

<p>Generate the base frequency plot of miRNA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basepreplot(file_dataframe, width = 0.6, size = 12)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basepreplot_+3A_file_dataframe">file_dataframe</code></td>
<td>
<p>A dataframe. The output result after running mirnapredata.</p>
</td></tr>
<tr><td><code id="basepreplot_+3A_width">width</code></td>
<td>
<p>A numeric. The width of the output bar plot, and default is 0.6.</p>
</td></tr>
<tr><td><code id="basepreplot_+3A_size">size</code></td>
<td>
<p>A numeric. The size of axis text, and default is 0.6.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The miRNA base frequency plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##P1
P1_miRNA_data &lt;- mirnapredata(mirnaseq_dataframe = P1_miRNA_count)
##Drawing
basepreplot(file_dataframe = P1_miRNA_data)
</code></pre>

<hr>
<h2 id='Countfiliter'>Filitering low expressed miRNAs based on count: Countfiliter</h2><span id='topic+Countfiliter'></span>

<h3>Description</h3>

<p>Filitering low expressed miRNAs based on count: Countfiliter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Countfiliter(P1_count, P2_count, F1_count, count_threshold = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Countfiliter_+3A_p1_count">P1_count</code></td>
<td>
<p>A dataframe. The count data of miRNA from the P1 species. The first column must be the miRNA sequence. Others are listed as the count of miRNA, and each column denotes one biological replicate of the sample.</p>
</td></tr>
<tr><td><code id="Countfiliter_+3A_p2_count">P2_count</code></td>
<td>
<p>A dataframe. Similar with P1_count, the count data of miRNA from the P2 species.</p>
</td></tr>
<tr><td><code id="Countfiliter_+3A_f1_count">F1_count</code></td>
<td>
<p>A dataframe. Similar with P1_count, the count data of miRNA from the F1 species.</p>
</td></tr>
<tr><td><code id="Countfiliter_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A numeric. In all samples, there is at least one sample whose count value is more than or equal to count_threshold to be retained. By default, the count value more than or equal to 5 is retained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe. The result includes all miRNAs that fulfill the count value requirement (count &gt;= count_threshold) in at least one sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Get the filitered mirna count table (default: Count &gt;= 5 in at least one sample)
Count5result &lt;- Countfiliter(P1_count = P1_miRNA_count,
                             P2_count = P2_miRNA_count,
                             F1_count = F1_miRNA_count,count_threshold = 5)
</code></pre>

<hr>
<h2 id='F1_miRNA_count'>Sequences of miRNAs from one species</h2><span id='topic+F1_miRNA_count'></span>

<h3>Description</h3>

<p>The first column of this data set is all miRNA sequences from &quot;F1&quot;, and other columns are the corresponding count values from different samples
</p>


<h3>Format</h3>

<p>A dataframe containing all miRNAs with count value in &quot;F1&quot; (the S3 generations of Parents (Maternal parent: Triticum turgidum; Male parent: Aegilops tauschii)).
</p>


<h3>Source</h3>

<p>Generated from the S3 generations of Triticum turgidum (AABB) and Aegilops tauschii (DD).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F1_miRNA_count)
</code></pre>

<hr>
<h2 id='F1_miRNA_rpm'>Sequences of miRNAs from one species</h2><span id='topic+F1_miRNA_rpm'></span>

<h3>Description</h3>

<p>The first column of this data set is all miRNA sequences from &quot;F1&quot;, and other columns are the corresponding RPM values of different samples
</p>


<h3>Format</h3>

<p>A dataframe containing all miRNAs with RPM value in &quot;F1&quot; (the S3 generations of Parents (Maternal parent: Triticum turgidum; Male parent: Aegilops tauschii)).
</p>


<h3>Source</h3>

<p>Generated from the S3 generations of Triticum turgidum (AABB) and Aegilops tauschii (DD).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F1_miRNA_rpm)
</code></pre>

<hr>
<h2 id='F1_sRNA_seq'>Sequences of sRNAs from one species</h2><span id='topic+F1_sRNA_seq'></span>

<h3>Description</h3>

<p>The first column of this data set is all sequences from one speceie for the data statistics
</p>


<h3>Format</h3>

<p>A dataframe containing numerous sequences of all sRNAs in &quot;F1&quot; (the S3 generations of Parents (Maternal parent: Triticum turgidum; Male parent: Aegilops tauschii)). However, only 400 sRNAs are selected as test data due to the large data of sRNA.
</p>


<h3>Details</h3>

<p>This <code>data.frame</code> is very useful for the functional demonstration of &quot;srnapredata&quot;
</p>


<h3>Source</h3>

<p>Generated from the S3 generations of Triticum turgidum (AABB) and Aegilops tauschii (DD).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(F1_sRNA_seq)
</code></pre>

<hr>
<h2 id='genetic'>Classification for 12 expression patterns</h2><span id='topic+genetic'></span>

<h3>Description</h3>

<p>The input data is generated from the analysis result of DESeq2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genetic(pv11, pv12, pv21, fc11, fc12, fc21, Pvalue)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genetic_+3A_pv11">pv11</code></td>
<td>
<p>A numeric. The P value of F1_vs_P1 (Treatment:F1; Control:P1).</p>
</td></tr>
<tr><td><code id="genetic_+3A_pv12">pv12</code></td>
<td>
<p>A numeric. The P value of F1_vs_P2 (Treatment:F1; Control:P2)</p>
</td></tr>
<tr><td><code id="genetic_+3A_pv21">pv21</code></td>
<td>
<p>A numeric. The P value of P2_vs_P1 (Treatment:P2; Control:P1)</p>
</td></tr>
<tr><td><code id="genetic_+3A_fc11">fc11</code></td>
<td>
<p>A numeric. The Log2(FoldChange) value of F1_vs_P1 (Treatment:F1; Control:P1)</p>
</td></tr>
<tr><td><code id="genetic_+3A_fc12">fc12</code></td>
<td>
<p>A numeric. The Log2(FoldChange) value of F1_vs_P2 (Treatment:F1; Control:P2)</p>
</td></tr>
<tr><td><code id="genetic_+3A_fc21">fc21</code></td>
<td>
<p>A numeric. The Log2(FoldChange) value of P2_vs_P1 (Treatment:P2; Control:P1)</p>
</td></tr>
<tr><td><code id="genetic_+3A_pvalue">Pvalue</code></td>
<td>
<p>A numeric. Filtration criteria of P value for Classification.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe.
</p>

<hr>
<h2 id='Get12Bins'>Genetic effects analysis: Twelve bins of expression analysis (method2)</h2><span id='topic+Get12Bins'></span>

<h3>Description</h3>

<p>Genetic effects analysis: Twelve bins of expression analysis (method2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Get12Bins(P1_count, P2_count, F1_count, count_threshold = 5, Pvalue = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Get12Bins_+3A_p1_count">P1_count</code></td>
<td>
<p>A dataframe. The count data of miRNA from the P1 species. The first column must be the miRNA sequence. Others are listed as the count of miRNA, and each column denotes one biological replicate of the sample.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_p2_count">P2_count</code></td>
<td>
<p>A dataframe. Similar with P1_count, the count data of miRNA from the P2 species.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_f1_count">F1_count</code></td>
<td>
<p>A dataframe. Similar with P1_count, the count data of miRNA from the F1 species.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A numeric. In all samples, there is at least one sample whose count value is more than or equal to count_threshold to be retained. By default, the count value more than or equal to 5 is retained.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_pvalue">Pvalue</code></td>
<td>
<p>A numeric. The threshold of significance test among different groups. Default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe. The output results contain the P value, log2FoldChange and grouping information for each miRNA expressed in all species (count &gt;= count_threshold). F1_vs_P1(P value: pv11,log2FoldChange: fc11), F1_vs_P2(P value: pv12,log2FoldChange: fc12), P2_vs_P1(P value: pv21,log2FoldChange: fc21)
</p>

<hr>
<h2 id='GetDAtable'>Genetic effects analysis of miRNA: |d/a| (method 1)</h2><span id='topic+GetDAtable'></span>

<h3>Description</h3>

<p>The additive (a) and dominant (d) values were calculated by the expression level of each miRNA. Edwards et al. proposed that the &quot;|d/a|&quot; can be used as the criterion to estimate the expression patterns of miRNAs. Specific classification criteria are as follows, |d/a| &lt;= 0.2, additivity; |d/a| &gt; 0.2 and |d/a| &lt;= 0.8, partial dominance; |d/a| &gt; 0.8 and |d/a| &lt;= 1.2, dominance; |d/a| &gt; 1.2, overdominance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetDAtable(P1_RPM, P2_RPM, F1_RPM, rpm_threshold = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetDAtable_+3A_p1_rpm">P1_RPM</code></td>
<td>
<p>A dataframe. The rpm data of miRNA from the P1 species. The first column must be the miRNA sequence. Others are listed as the rpm of miRNA, and each column denotes one biological replicate of the sample.</p>
</td></tr>
<tr><td><code id="GetDAtable_+3A_p2_rpm">P2_RPM</code></td>
<td>
<p>A dataframe. Similar with P1_RPM, the rpm data of miRNA from the P2 species.</p>
</td></tr>
<tr><td><code id="GetDAtable_+3A_f1_rpm">F1_RPM</code></td>
<td>
<p>A dataframe. Similar with P1_RPM, the rpm data of miRNA from the F1 species.</p>
</td></tr>
<tr><td><code id="GetDAtable_+3A_rpm_threshold">rpm_threshold</code></td>
<td>
<p>A numeric. the average of rpm value among all the biological replicates. By default, the average of rpm more than or equal to 1 is retained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe. The output results contain the value of &quot;|d/a|&quot; and grouping results for each miRNA expressed in all species (average_rpm &gt;= rpm_threshold).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Get the classification results based on the value of |d/a|
DAresult &lt;- GetDAtable(P1_RPM = P1_miRNA_rpm,
                       P2_RPM = P2_miRNA_rpm,
                       F1_RPM = F1_miRNA_rpm,rpm_threshold = 1)
</code></pre>

<hr>
<h2 id='lenplot'>Generate the sRNA length distribution plot</h2><span id='topic+lenplot'></span>

<h3>Description</h3>

<p>Generate the sRNA length distribution plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lenplot(file_dataframe, width = 0.6, size = 12)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lenplot_+3A_file_dataframe">file_dataframe</code></td>
<td>
<p>A dataframe. The output result after running &quot;srnapredata&quot;.</p>
</td></tr>
<tr><td><code id="lenplot_+3A_width">width</code></td>
<td>
<p>A numeric. The width of the output bar plot, and default is 0.6.</p>
</td></tr>
<tr><td><code id="lenplot_+3A_size">size</code></td>
<td>
<p>A numeric. The size of text in the outpot plot, and default is 12.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The sRNA length distribution plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##F1
F1_sRNA &lt;- srnapredata(srnaseq_dataframe = F1_sRNA_seq, group = "F1")
##P1
P1_sRNA &lt;- srnapredata(srnaseq_dataframe = P1_sRNA_seq, group = "P1")
##P2
P2_sRNA &lt;- srnapredata(srnaseq_dataframe = P2_sRNA_seq, group = "P2")
##integrate all sRNA data from P1, P2, and F1
sRNA_data &lt;- rbind(F1_sRNA,P1_sRNA,P2_sRNA)
##plot
lenplot(file_dataframe = sRNA_data)
</code></pre>

<hr>
<h2 id='mirnapredata'>Generate the data of miRNA base frequence in each position</h2><span id='topic+mirnapredata'></span>

<h3>Description</h3>

<p>Generally, the &quot;T&quot; base account for the highest percentage of miRNA in the first position.The function of &quot;mirnapredata&quot; can provide the input data for the next drawing of miRNA base distribution in each position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mirnapredata(mirnaseq_dataframe)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mirnapredata_+3A_mirnaseq_dataframe">mirnaseq_dataframe</code></td>
<td>
<p>A dataframe. The first column must be the sRNA sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe. About the output results, the first column is the base, the second column is the base frequency, the third column is the position.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##P1
P1_miRNA_data &lt;- mirnapredata(mirnaseq_dataframe = P1_miRNA_count)
##P2
P2_miRNA_data &lt;- mirnapredata(mirnaseq_dataframe = P2_miRNA_count)
##F1
F1_miRNA_data &lt;- mirnapredata(mirnaseq_dataframe = F1_miRNA_count)
</code></pre>

<hr>
<h2 id='miVennData'>Species specific expression analysis: miVennData</h2><span id='topic+miVennData'></span>

<h3>Description</h3>

<p>miVennData: Extract the species-specific miRNAs and the shared miRNAs among parents and offspring.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>miVennData(
  P1_RPM,
  P2_RPM,
  F1_RPM,
  rpm_threshold = 1,
  output_file = "venn_list"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="miVennData_+3A_p1_rpm">P1_RPM</code></td>
<td>
<p>A dataframe. The rpm data of miRNA from the P1 species. The first column must be the miRNA sequence. Others are listed as the rpm of miRNA, and each column denotes one biological replicate of the sample.</p>
</td></tr>
<tr><td><code id="miVennData_+3A_p2_rpm">P2_RPM</code></td>
<td>
<p>A dataframe. Similar with P1_RPM, the rpm data of miRNA from the P2 species.</p>
</td></tr>
<tr><td><code id="miVennData_+3A_f1_rpm">F1_RPM</code></td>
<td>
<p>A dataframe. Similar with P1_RPM, the rpm data of miRNA from the F1 species.</p>
</td></tr>
<tr><td><code id="miVennData_+3A_rpm_threshold">rpm_threshold</code></td>
<td>
<p>A numeric. the average of rpm value among all the biological replicates. By default, the average of rpm more than or equal to 1 is retained.</p>
</td></tr>
<tr><td><code id="miVennData_+3A_output_file">output_file</code></td>
<td>
<p>Specify the output file. &quot;venn_list&quot; is the default option, which outputs all the information of the Venn diagram. &quot;all_common&quot; is one of options, which outputs the miRNAs shared by parents and offspring. &quot;P1_specific&quot; is one of options, which outputs P1 specific expression miRNA. &quot;P2_specific&quot; is one of options, which outputs P2 specific expression miRNA. &quot;F1_specific&quot; is one of options, which outputs F1 specific expression miRNA.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe. The output results is based on your selection (output_file).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Extract the species-specific miRNAs and the shared miRNAs among parents and offspring.
##output_file = "venn_list"
venn_list &lt;- miVennData(P1_RPM = P1_miRNA_rpm,
                        P2_RPM = P2_miRNA_rpm,
                        F1_RPM = F1_miRNA_rpm,
                        rpm_threshold = 1,output_file = "venn_list")
##output_file = "P1_specific"
P1_specific &lt;- miVennData(P1_RPM = P1_miRNA_rpm,
                          P2_RPM = P2_miRNA_rpm,
                          F1_RPM = F1_miRNA_rpm,
                          rpm_threshold = 1,output_file = "P1_specific")
##output_file = "P2_specific"
P2_specific &lt;- miVennData(P1_RPM = P1_miRNA_rpm,
                          P2_RPM = P2_miRNA_rpm,
                          F1_RPM = F1_miRNA_rpm,
                          rpm_threshold = 1,output_file = "P2_specific")
##output_file = "F1_specific"
F1_specific &lt;- miVennData(P1_RPM = P1_miRNA_rpm,
                          P2_RPM = P2_miRNA_rpm,
                          F1_RPM = F1_miRNA_rpm,
                          rpm_threshold = 1,output_file = "F1_specific")
##output_file = "all_common"
all_common &lt;- miVennData(P1_RPM = P1_miRNA_rpm,
                         P2_RPM = P2_miRNA_rpm,
                         F1_RPM = F1_miRNA_rpm,
                         rpm_threshold = 1,output_file = "all_common")
</code></pre>

<hr>
<h2 id='miVennPlot'>Species specific expression analysis: miVennPlot</h2><span id='topic+miVennPlot'></span>

<h3>Description</h3>

<p>miVennPlot: generate the Venn diagram with the specific expression information of miRNAs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>miVennPlot(P1_RPM, P2_RPM, F1_RPM, rpm_threshold = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="miVennPlot_+3A_p1_rpm">P1_RPM</code></td>
<td>
<p>A dataframe. The rpm data of miRNA from the P1 species. The first column must be the miRNA sequence. Others are listed as the rpm of miRNA, and each column denotes one biological replicate of the sample.</p>
</td></tr>
<tr><td><code id="miVennPlot_+3A_p2_rpm">P2_RPM</code></td>
<td>
<p>A dataframe. Similar with P1_RPM, the rpm data of miRNA from the P2 species.</p>
</td></tr>
<tr><td><code id="miVennPlot_+3A_f1_rpm">F1_RPM</code></td>
<td>
<p>A dataframe. Similar with P1_RPM, the rpm data of miRNA from the F1 species.</p>
</td></tr>
<tr><td><code id="miVennPlot_+3A_rpm_threshold">rpm_threshold</code></td>
<td>
<p>A numeric. the average of rpm value among all the biological replicates. By default, the average of rpm more than or equal to 1 is retained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Venn diagram with the specific expression information of miRNAs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Drawing
miVennPlot(P1_RPM = P1_miRNA_rpm,
           P2_RPM = P2_miRNA_rpm,
           F1_RPM = F1_miRNA_rpm,rpm_threshold = 1)
</code></pre>

<hr>
<h2 id='P1_miRNA_count'>Sequences of miRNAs from one species</h2><span id='topic+P1_miRNA_count'></span>

<h3>Description</h3>

<p>The first column of this data set is all miRNA sequences from &quot;P1&quot;, and other columns are the corresponding count values from different samples
</p>


<h3>Format</h3>

<p>A dataframe containing all miRNAs with count value in &quot;P1&quot; (Male parent: Aegilops tauschii).
</p>


<h3>Source</h3>

<p>Generated from the Aegilops tauschii (DD).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(P1_miRNA_count)
</code></pre>

<hr>
<h2 id='P1_miRNA_rpm'>Sequences of miRNAs from one species</h2><span id='topic+P1_miRNA_rpm'></span>

<h3>Description</h3>

<p>The first column of this data set is all miRNA sequences from &quot;P1&quot;, and other columns are the corresponding RPM values of different samples
</p>


<h3>Format</h3>

<p>A dataframe containing all miRNAs with RPM value in &quot;P1&quot; (Male parent: Aegilops tauschii).
</p>


<h3>Source</h3>

<p>Generated from the Aegilops tauschii (DD).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(P1_miRNA_rpm)
</code></pre>

<hr>
<h2 id='P1_sRNA_seq'>Sequences of sRNAs from one species</h2><span id='topic+P1_sRNA_seq'></span>

<h3>Description</h3>

<p>The first column of this data set is all sequences from one speceie for the data statistics
</p>


<h3>Format</h3>

<p>A dataframe containing numerous sequences of all sRNAs in &quot;P1&quot; (Male parent: Aegilops tauschii). However, only 400 sRNAs are selected as test data due to the large data of sRNA.
</p>


<h3>Details</h3>

<p>This <code>data.frame</code> is very useful for the functional demonstration of &quot;srnapredata&quot;
</p>


<h3>Source</h3>

<p>Generated from the Aegilops tauschii (DD).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(P1_sRNA_seq)
</code></pre>

<hr>
<h2 id='P2_miRNA_count'>Sequences of miRNAs from one species</h2><span id='topic+P2_miRNA_count'></span>

<h3>Description</h3>

<p>The first column of this data set is all miRNA sequences from &quot;P2&quot;, and other columns are the corresponding count values from different samples
</p>


<h3>Format</h3>

<p>A dataframe containing all miRNAs with count value in &quot;P2&quot; (Maternal parent: Triticum turgidum).
</p>


<h3>Source</h3>

<p>Generated from the Triticum turgidum (AABB).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(P2_miRNA_count)
</code></pre>

<hr>
<h2 id='P2_miRNA_rpm'>Sequences of miRNAs from one species</h2><span id='topic+P2_miRNA_rpm'></span>

<h3>Description</h3>

<p>The first column of this data set is all miRNA sequences from &quot;P2&quot;, and other columns are the corresponding RPM values of different samples
</p>


<h3>Format</h3>

<p>A dataframe containing all miRNAs with RPM value in &quot;P2&quot; (Maternal parent: Triticum turgidum).
</p>


<h3>Source</h3>

<p>Generated from the Triticum turgidum (AABB).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(P2_miRNA_rpm)
</code></pre>

<hr>
<h2 id='P2_sRNA_seq'>Sequences of sRNAs from one species</h2><span id='topic+P2_sRNA_seq'></span>

<h3>Description</h3>

<p>The first column of this data set is all sequences from one species for the data statistics
</p>


<h3>Format</h3>

<p>A dataframe containing numerous sequences of all sRNAs in &quot;P2&quot; (Maternal parent: Triticum turgidum). However, only 400 sRNAs are selected as test data due to the large data of sRNA.
</p>


<h3>Details</h3>

<p>This <code>data.frame</code> is very useful for the functional demonstration of &quot;srnapredata&quot;
</p>


<h3>Source</h3>

<p>Generated from the Triticum turgidum (AABB).
</p>


<h3>References</h3>

<p>Li, A., et al., mRNA and Small RNA Transcriptomes Reveal Insights into Dynamic Homoeolog Regulation of Allopolyploid Heterosis in Nascent Hexaploid Wheat. Plant Cell, 2014. 26(5): p. 1878-1900.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(P2_sRNA_seq)
</code></pre>

<hr>
<h2 id='polyDESeq'>Differential expression analysis</h2><span id='topic+polyDESeq'></span>

<h3>Description</h3>

<p>Differential expression analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polyDESeq(P1_count, P2_count, F1_count, count_threshold = 5, Pvalue = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="polyDESeq_+3A_p1_count">P1_count</code></td>
<td>
<p>A dataframe. The count data of miRNA from the P1 species. The first column must be the miRNA sequence. Others are listed as the count of miRNA, and each column denotes one biological replicate of the sample.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_p2_count">P2_count</code></td>
<td>
<p>A dataframe. Similar with P1_count, the count data of miRNA from the P2 species.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_f1_count">F1_count</code></td>
<td>
<p>A dataframe. Similar with P1_count, the count data of miRNA from the F1 species.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A numeric. In all samples, there is at least one sample whose count value is more than or equal to count_threshold to be retained. By default, the count value more than or equal to 5 is retained.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_pvalue">Pvalue</code></td>
<td>
<p>A numeric. The threshold of significance test among different groups. Default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe. Differential expression analysis results of miRNA expressed in each two species (count &gt;= count_threshold).
</p>

<hr>
<h2 id='Rpmfiliter'>Filitering low expressed miRNAs based on RPM: Rpmfiliter</h2><span id='topic+Rpmfiliter'></span>

<h3>Description</h3>

<p>Filitering low expressed miRNAs based on RPM: Rpmfiliter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Rpmfiliter(P1_RPM, P2_RPM, F1_RPM, rpm_threshold = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Rpmfiliter_+3A_p1_rpm">P1_RPM</code></td>
<td>
<p>A dataframe. The rpm data of miRNA from the P1 species. The first column must be the miRNA sequence. Others are listed as the rpm of miRNA, and each column denotes one biological replicate of the sample.</p>
</td></tr>
<tr><td><code id="Rpmfiliter_+3A_p2_rpm">P2_RPM</code></td>
<td>
<p>A dataframe. Similar with P1_RPM, the rpm data of miRNA from the P2 species.</p>
</td></tr>
<tr><td><code id="Rpmfiliter_+3A_f1_rpm">F1_RPM</code></td>
<td>
<p>A dataframe. Similar with P1_RPM, the rpm data of miRNA from the F1 species.</p>
</td></tr>
<tr><td><code id="Rpmfiliter_+3A_rpm_threshold">rpm_threshold</code></td>
<td>
<p>A numeric. the average of rpm value among all the biological replicates. By default, the average of rpm more than or equal to 1 is retained.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe. The result includes all miRNAs that fulfill the average rpm value requirement (Average rpm &gt;= rpm_threshold) among all species.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Get the filitered mirna rpm table (default: the average rpm &gt;= 1 in three species)
Rpm1result &lt;- Rpmfiliter(P1_RPM = P1_miRNA_rpm,
                         P2_RPM = P2_miRNA_rpm,
                         F1_RPM = F1_miRNA_rpm,rpm_threshold = 1)
</code></pre>

<hr>
<h2 id='srnapredata'>Generate the data of sRNA length distribution</h2><span id='topic+srnapredata'></span>

<h3>Description</h3>

<p>Generally, the length interval of sRNA is 21-24. The function of &quot;srnapredata&quot; can provide the input data for the next drawing of sRNA length distribution among different species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srnapredata(srnaseq_dataframe, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srnapredata_+3A_srnaseq_dataframe">srnaseq_dataframe</code></td>
<td>
<p>A dataframe. The first column must be the sRNA sequence.</p>
</td></tr>
<tr><td><code id="srnapredata_+3A_group">group</code></td>
<td>
<p>A character. You an select a representative group name for next drawing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe. The output results are consist of three columns, the first column is the length of sRNA, the second column id the frequency, and the third column is the group name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Only 400 sRNAs are selected as test data due to the large data of sRNA.
##Recommended to use the "data.table" package for reading data quickly.
##F1
F1_sRNA &lt;- srnapredata(srnaseq_dataframe = F1_sRNA_seq, group = "F1")
##P1
P1_sRNA &lt;- srnapredata(srnaseq_dataframe = P1_sRNA_seq, group = "P1")
##P2
P2_sRNA &lt;- srnapredata(srnaseq_dataframe = P2_sRNA_seq, group = "P2")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
