<!DOCTYPE html><html><head><title>Help for package SRS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SRS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Scaling with ranked subsampling (SRS)'>
<p>Scaling with ranked subsampling (SRS)</p></a></li>
<li><a href='#Scaling with ranked subsampling (SRS) Shiny app'>
<p>Shiny app for scaling with ranked subsampling (SRS)</p></a></li>
<li><a href='#Scaling with ranked subsampling curve (SRScurve)'>
<p>Scaling with ranked subsampling curve (SRScurve)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Scaling with Ranked Subsampling</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of species count data in ecology often requires normalization to an identical sample size. Rarefying (random subsampling without replacement), which is a popular method for normalization, has been widely criticized for its poor reproducibility and potential distortion of the community structure. In the context of microbiome count data, researchers explicitly advised against the use of rarefying. An alternative to rarefying is scaling with ranked subsampling (SRS). SRS consists of two steps. In the first step, the total counts for all OTUs (operational taxonomic units) or species in each sample are divided by a scaling factor chosen in such a way that the sum of the scaled counts Cscaled equals Cmin. In the second step, the non-integer Cscaled values are converted into integers by an algorithm that we dub ranked subsampling. The Cscaled value for each OTU or species is split into the integer part Cint  (Cint = floor(Cscaled)) and the fractional part Cfrac (Cfrac = Cscaled - Cints). Since the sum of Cint is smaller or equal to Cmin, additional  delta C = Cmin - the sum of Cint counts have to be added to the library to reach the total count of Cmin. This is achieved as follows. OTUs are ranked in the descending order of their Cfrac values. Beginning with the OTU of the highest rank, single count per OTU is added to the normalized library until the total number of added counts reaches delta C and the sum of all counts in the normalized library equals Cmin. When the lowest Cfrag involved in picking delta C counts is shared by several OTUs, the OTUs used for adding a single count to the library are selected in the order of their Cint values. This selection minimizes the effect of normalization on the relative frequencies of OTUs. OTUs with identical Cfrag as well as Cint are sampled randomly without replacement. See Beule &amp; Karlovsky (2020) &lt;<a href="https://doi.org/10.7717%2Fpeerj.9593">doi:10.7717/peerj.9593</a>&gt; for details.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0), vegan (&ge; 2.5-6), shiny (&ge; 1.5.0), DT (&ge;
0.16), shinycssloaders (&ge; 1.0.0), shinybusy (&ge; 0.2.2)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by-sa/4.0">CC BY-SA 4.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Author:</td>
<td>Lukas Beule [aut, cre],
  Vitor Heidrich [aut],
  Petr Karlovsky [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lukas Beule &lt;lukas.beule@julius-kuehn.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-03-20 07:30:36 UTC; lukasbeule</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-03-27 14:30:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='Scaling+20with+20ranked+20subsampling+20+28SRS+29'>
Scaling with ranked subsampling (SRS)
</h2><span id='topic+SRS'></span>

<h3>Description</h3>

<p>Scaling with ranked subsampling (SRS) for the normalization of ecological count data.
It is recommended to use <a href="#topic+SRS.shiny.app">SRS.shiny.app</a> for the determination of Cmin.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SRS(data, Cmin, set_seed = TRUE, seed = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20+2B28SRS+2B29_+3A_data">data</code></td>
<td>

<p>Data frame (species count or OTU table) in which columns are samples and rows are the counts of species or OTUs. Only integers are accepted as data.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20+2B28SRS+2B29_+3A_cmin">Cmin</code></td>
<td>

<p>The number of counts to which all samples will be normalized. Typically, the total OTU count of the sample with the lowest sequencing depth is chosen as Cmin. Samples with sequencing depth lower than the chosen Cmin will be discarded.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20+2B28SRS+2B29_+3A_set_seed">set_seed</code></td>
<td>

<p>Logical, if TRUE, a seed is set to enable reproducibility of SRS if OTUs with identical Cfrag as well as Cint are sampled randomly without replacement. See <a href="base.html#topic+set.seed">set.seed</a> for details. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20+2B28SRS+2B29_+3A_seed">seed</code></td>
<td>

<p>Integer, specifying the seed. See <a href="base.html#topic+set.seed">set.seed</a> for details. Default is 1.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is recommended to use <a href="#topic+SRS.shiny.app">SRS.shiny.app</a> for the determination of Cmin.
SRS consists of two steps. In the first step, the total counts for all OTUs (operational taxonomic units) or species in each sample are divided by a scaling factor chosen in such a way that the sum of the scaled counts Cscaled equals Cmin. In the second step, the non-integer Cscaled values are converted into integers by an algorithm that we dub ranked subsampling. The Cscaled value for each OTU or species is split into the integer part Cint  (<code class="reqn">Cint = floor(Cscaled)</code>) and the fractional part Cfrac (<code class="reqn">Cfrac = Cscaled - Cint</code>). Since <code class="reqn">\Sigma Cint \le Cmin</code> , additional <code class="reqn">\Delta C = Cmin - \Sigma Cint</code> counts have to be added to the library to reach the total count of Cmin. This is achieved as follows. OTUs are ranked in the descending order of their Cfrac values. Beginning with the OTU of the highest rank, single count per OTU is added to the normalized library until the total number of added counts reaches <code class="reqn">\Delta C</code> and the sum of all counts in the normalized library equals Cmin. When the lowest Cfrag involved in picking <code class="reqn">\Delta C</code> counts is shared by several OTUs, the OTUs used for adding a single count to the library are selected in the order of their Cint values. This selection minimizes the effect of normalization on the relative frequencies of OTUs. OTUs with identical Cfrag as well as Cint are sampled randomly without replacement.
</p>


<h3>Value</h3>

<p>Data frame normalized to Cmin.
</p>


<h3>Author(s)</h3>

<p>Lukas Beule, Vitor Heidrich, Devon O'rourke, Petr Karlovsky
</p>


<h3>References</h3>

<p>Beule L, Karlovsky P. 2020. Improved normalization of species count data in ecology by scaling with ranked subsampling (SRS): application to microbial communities. PeerJ 8:e9593
</p>
<p>&lt;https://doi.org/10.7717/peerj.9593&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Samples should be arranged columnwise.
##Input data should not contain any categorial
##data such as taxonomic assignment or barcode sequences.
##An example of the input data can be found below:

example_input_data &lt;- matrix(c(sample(1:20, 100, replace = TRUE),
sample(1:30, 100, replace = TRUE),sample(1:40, 100, replace = TRUE)), nrow = 100)
colnames(example_input_data) &lt;- c("sample_1","sample_2","sample_3")
example_input_data &lt;- as.data.frame(example_input_data)
example_input_data

##Selection of the desired number of counts
##(e.g., total OTU counts of the sample with the lowest sequencing depth):

Cmin &lt;- min(colSums(example_input_data))
Cmin

##Running the SRS function
SRS_output &lt;- SRS(example_input_data, Cmin)
SRS_output

##Samples that have a total number of counts &lt; Cmin will be discarded:
SRS_output &lt;- SRS(example_input_data, Cmin+1)
SRS_output
</code></pre>

<hr>
<h2 id='Scaling+20with+20ranked+20subsampling+20+28SRS+29+20Shiny+20app'>
Shiny app for scaling with ranked subsampling (SRS) 
</h2><span id='topic+SRS.shiny.app'></span>

<h3>Description</h3>

<p>Shiny app for the determination of Cmin for scaling with ranked subsampling (SRS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SRS.shiny.app(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20+2B28SRS+2B29+2B20Shiny+2B20app_+3A_data">data</code></td>
<td>

<p>Data frame (species count or OTU table) in which columns are samples and rows are the counts of species or OTUs. Only integers are accepted as data.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Shiny app that generates a visualization of retained samples, summary statistics, SRS curves, and an interactive table in response to varying minimum sample size (Cmin).
</p>


<h3>Value</h3>

<p>Launches Shiny app for SRS in the default web browser.
</p>


<h3>Author(s)</h3>

<p>Vitor Heidrich, Devon O'rourke, Petr Karlovsky, Lukas Beule
</p>


<h3>References</h3>

<p>Beule L, Karlovsky P. 2020. Improved normalization of species count data in ecology by scaling with ranked subsampling (SRS): application to microbial communities. PeerJ 8:e9593
</p>
<p>&lt;https://doi.org/10.7717/peerj.9593&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Samples should be arranged columnwise.
##Input data should not contain any categorial
##data such as taxonomic assignment or barcode sequences.
##An example of the input data can be found below:

example_input_data &lt;- matrix(c(sample(1:20, 100, replace = TRUE),
sample(1:30, 100, replace = TRUE),sample(1:40, 100, replace = TRUE)), nrow = 100)
colnames(example_input_data) &lt;- c("sample_1","sample_2","sample_3")
example_input_data &lt;- as.data.frame(example_input_data)
example_input_data

##Launching the SRS shiny app with example_input_data as input
if (interactive()) {SRS.shiny.app(example_input_data)}
</code></pre>

<hr>
<h2 id='Scaling+20with+20ranked+20subsampling+20curve+20+28SRScurve+29'>
Scaling with ranked subsampling curve (SRScurve)
</h2><span id='topic+SRScurve'></span>

<h3>Description</h3>

<p>For each column of the input data, draws a line plot of alpha diversity indices (see <a href="#topic+SRScurve">metric</a>) at different sample sizes (specified by <a href="#topic+SRScurve">step</a>) normalized by scaling with ranked subsampling (using <a href="#topic+SRS">SRS</a>). Minimum sample size (cutoff-level) can be evaluated by specifying <a href="#topic+SRScurve">sample</a>. The function further allows to visualize trade-offs between cutoff-level and alpha diversity and enables direct comparison of SRS and repeated rarefying.
</p>
<p>See Beule &amp; Karlovsky (2020) &lt;doi:10.7717/peerj.9593&gt; for details regarding SRS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SRScurve(data, metric = "richness", step = 50, sample = 0, max.sample.size = 0,
        rarefy.comparison = FALSE, rarefy.repeats = 10, 
        rarefy.comparison.legend = FALSE, xlab = "sample size", 
        ylab = "richness", label = FALSE, col, lty, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_data">data</code></td>
<td>

<p>Data frame (species count or OTU table) in which columns are samples and rows are the counts of species or OTUs. Only integers are accepted as data.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_metric">metric</code></td>
<td>

<p>Character, &quot;richness&quot; (using <a href="vegan.html#topic+specnumber">specnumber</a>) for species richness or &quot;<a href="vegan.html#topic+diversity">shannon</a>&quot;, &quot;<a href="vegan.html#topic+diversity">simpson</a>&quot; or &quot;<a href="vegan.html#topic+diversity">invsimpson</a>&quot; (using <a href="vegan.html#topic+diversity">diversity</a>) for common diversity indices. Default is &quot;richness&quot;.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_step">step</code></td>
<td>

<p>Numeric, specifying the step used to vary the sample size. Default is 50.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_sample">sample</code></td>
<td>

<p>Numeric, specifying the cutoff-level to visualize trade-offs between cutoff-level and alpha diversity.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_max.sample.size">max.sample.size</code></td>
<td>

<p>Numeric, specifying the maximum sample size to which SRS curves are drawn. Default is 0 which does not limit the maximum sample size.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_rarefy.comparison">rarefy.comparison</code></td>
<td>

<p>Logical, if TRUE, median values of rarefy with n repeats (specified by <a href="#topic+SRScurve">rarefy.repeats</a>) will be drawn for comparison. Default is FALSE.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_rarefy.repeats">rarefy.repeats</code></td>
<td>

<p>Numeric, specifying the number of repeats used to obtain median values for rarefying. Default is 10.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_rarefy.comparison.legend">rarefy.comparison.legend</code></td>
<td>

<p>Logical, if TRUE, a legend for the comparison between SRS and rarefy is plotted. Default is FALSE.
</p>
</td></tr>
<tr><td><code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_xlab">xlab</code>, <code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_ylab">ylab</code>, <code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_label">label</code>, <code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_col">col</code>, <code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_lty">lty</code>, <code id="Scaling+2B20with+2B20ranked+2B20subsampling+2B20curve+2B20+2B28SRScurve+2B29_+3A_...">...</code></td>
<td>

<p>Graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Beule &amp; Karlovsky (2020) &lt;doi:10.7717/peerj.9593&gt; for details regarding scaling with ranked subsampling.
</p>


<h3>Value</h3>

<p>Returns a line plot visualizing the change in alpha diversity indices with changing sample size.
</p>


<h3>Author(s)</h3>

<p>Vitor Heidrich, Petr Karlovsky, Lukas Beule
</p>


<h3>References</h3>

<p>Beule L, Karlovsky P. 2020. Improved normalization of species count data in ecology by scaling with ranked subsampling (SRS): application to microbial communities. PeerJ 8:e9593
</p>
<p>&lt;https://doi.org/10.7717/peerj.9593&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Samples should be arranged columnwise.
##Input data should not contain any categorial
##data such as taxonomic assignment or barcode sequences.
##An example of the input data can be found below:

example_input_data &lt;- matrix(c(sample(1:20, 100, replace = TRUE),
sample(1:30, 100, replace = TRUE),sample(1:40, 100, replace = TRUE)), nrow = 100)
colnames(example_input_data) &lt;- c("sample_1","sample_2","sample_3")
example_input_data &lt;- as.data.frame(example_input_data)
example_input_data

##Default settings of SRScurve.
SRScurve(example_input_data, metric = "richness", step = 50,
         ylab = "richness", 
         col = c("#000000", "#E69F00", "#56B4E9"))

##Limit the compution of SRS curves to a sample size of 200.
SRScurve(example_input_data, metric = "richness", step = 50,
         max.sample.size = 200, ylab = "richness", 
         col = c("#000000", "#E69F00", "#56B4E9"))

##SRScurve with comparison of SRS (solid lines) and repeated rarefying (dashed lines).
##Different colors correspond to indiviual samples. Cuttoff-level set to 200.
SRScurve(example_input_data, metric = "richness", step = 50,
         sample = 200, max.sample.size = 200,
         rarefy.comparison = TRUE, rarefy.repeats = 10, rarefy.comparison.legend = TRUE,
         ylab = "richness", 
         col = c(rep(c("#000000", "#E69F00", "#56B4E9"),2)),
         lty = c(1,2))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
