<!DOCTYPE html><html><head><title>Help for package SRTsim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SRTsim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compareSRT'><p>Summarize metrics for reference data and synthetic data</p></a></li>
<li><a href='#convert_grid'><p>Convert continuous coordinate into integer, essential for BayesSpace to determine the neighborhood info</p></a></li>
<li><a href='#createSRT'><p>Create simSRT object</p></a></li>
<li><a href='#EstParam'><p>Access Model Fitting Parameters</p></a></li>
<li><a href='#exampleLIBD'><p>Data used for creating vignettes</p></a></li>
<li><a href='#fit_pos_optim'><p>fitting data with poisson through optim function</p></a></li>
<li><a href='#get_metrics_pd'><p>Extracted summarized metrics for reference data and synthetic data</p></a></li>
<li><a href='#get_stats_gene'><p>Summarize gene-wise summary metrics</p></a></li>
<li><a href='#get_stats_loc'><p>Summarize location-wise summary metrics</p></a></li>
<li><a href='#metaParam'><p>Access User-Specified Parameters</p></a></li>
<li><a href='#refcolData'><p>Access reference colData</p></a></li>
<li><a href='#refCounts'><p>Access reference count matrix</p></a></li>
<li><a href='#refrowData'><p>Access reference rowData</p></a></li>
<li><a href='#reGenCountshiny'><p>ReSimulate Count Data with Parameters Specification from Shiny</p></a></li>
<li><a href='#Shiny2SRT'><p>Create a SRTsim object from reference-free shinyoutput</p></a></li>
<li><a href='#simcolData'><p>Access synthetic colData</p></a></li>
<li><a href='#simCounts'><p>Access synthetic count matrix</p></a></li>
<li><a href='#simNewLocs'><p>Fit the marginal distributions for single gene</p></a></li>
<li><a href='#simrowData'><p>Access synthetic rowData</p></a></li>
<li><a href='#srtsim_cci_free'><p>Generate Data with Cell-Cell Interaction Under Reference-Free Mode</p></a></li>
<li><a href='#srtsim_cci_ref'><p>Generate Data with Cell-Cell Interaction Under Reference-Based Mode</p></a></li>
<li><a href='#srtsim_count'><p>Generate Data with Estimated Parameters</p></a></li>
<li><a href='#srtsim_count_affine'><p>Generate Data with Estimated Parameters For A New Designed Pattern</p></a></li>
<li><a href='#srtsim_fit'><p>Fit the marginal distributions for each row of a count matrix</p></a></li>
<li><a href='#srtsim_newlocs'><p>Fit the marginal distributions for each row of a count matrix</p></a></li>
<li><a href='#SRTsim_shiny'><p>Run the SRTsim Shiny Application</p></a></li>
<li><a href='#subsetSRT'><p>Subset SRT object based on domain labels of interest</p></a></li>
<li><a href='#toyData'><p>A toyExample to showcase reference-based simulations</p></a></li>
<li><a href='#toyShiny'><p>A toyExample to showcase reference-free simulations</p></a></li>
<li><a href='#visualize_gene'><p>Visualize expression pattern for the gene of interest in reference data and synthetic data</p></a></li>
<li><a href='#visualize_metrics'><p>Visualize summarized metrics for reference data and synthetic data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulator for Spatially Resolved Transcriptomics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-12-23</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jiaqiang Zhu &lt;jiaqiang@umich.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An independent, reproducible, and flexible Spatially Resolved Transcriptomics (SRT) simulation framework that can be used to facilitate the development of SRT analytical methods for a wide variety of SRT-specific analyses. It utilizes spatial localization information to simulate SRT expression count data in a reproducible and scalable fashion. Two major simulation schemes are implemented in 'SRTsim': reference-based and reference-free.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>concaveman,sf,sp,spatstat.geom,parallel,pdist,MASS,S4Vectors,stats,Matrix,Morpho,matrixStats,plotly,shiny,viridis,shinydashboard,dashboardthemes,shinyBS,
ggplot2, ggpubr, spatstat.random,magrittr,FNN,dplyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr,rmarkdown,BiocStyle,RefManageR,BiocManager,sessioninfo</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), methods</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Collate:</td>
<td>'SRTaffine.R' 'SRTcount.R' 'SRTfit.R' 'SRTmodels.R'
'SRTsim_class.R' 'body.R' 'compareSRT.R' 'createSRT.R' 'data.R'
'globals.R' 'reGenCountshiny.R' 'runapp.R' 'server.R'
'shiny2srt.R' 'sidebar.R' 'simSRTLocs.R' 'subsetSRT.R'
'SRTcci.R' 'ui.R' 'utilies_func.R' 'visualize_gene.R'
'visualize_metrics.R'</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-02 23:00:01 UTC; jiaqiangzhu</td>
</tr>
<tr>
<td>Author:</td>
<td>Jiaqiang Zhu <a href="https://orcid.org/0000-0002-1455-0041"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb, cre],
  Lulu Shang <a href="https://orcid.org/0000-0002-2480-3065"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Xiang Zhou <a href="https://orcid.org/0000-0002-4331-7599"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-13 15:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='compareSRT'>Summarize metrics for reference data and synthetic data</h2><span id='topic+compareSRT'></span>

<h3>Description</h3>

<p>Summarize metrics for reference data and synthetic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareSRT(simsrt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareSRT_+3A_simsrt">simsrt</code></td>
<td>
<p>A SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object with summarized metrics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
## Compute metrics 
toySRT   &lt;- compareSRT(toySRT)
</code></pre>

<hr>
<h2 id='convert_grid'>Convert continuous coordinate into integer, essential for BayesSpace to determine the neighborhood info</h2><span id='topic+convert_grid'></span>

<h3>Description</h3>

<p>Convert continuous coordinate into integer, essential for BayesSpace to determine the neighborhood info
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_grid(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_grid_+3A_x">x</code></td>
<td>
<p>A numeric vector of continuous coordinate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a numeric vector oof integer coordinate
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)

## Create New Locations within Profile
toySRT2  &lt;- srtsim_newlocs(toySRT,new_loc_num=1000)

## Convert non-integer x-coordinates into an integer value
newGrid_x &lt;- convert_grid(simcolData(toySRT2)$x)
</code></pre>

<hr>
<h2 id='createSRT'>Create simSRT object</h2><span id='topic+createSRT'></span>

<h3>Description</h3>

<p>Create simSRT object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createSRT(count_in, loc_in, refID = "ref1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createSRT_+3A_count_in">count_in</code></td>
<td>
<p>A gene expression count <code>matrix</code></p>
</td></tr>
<tr><td><code id="createSRT_+3A_loc_in">loc_in</code></td>
<td>
<p>A location <code>dataframe</code> with colnames x,y,label</p>
</td></tr>
<tr><td><code id="createSRT_+3A_refid">refID</code></td>
<td>
<p>A <code>character</code> reference sample identifier. Default = <code>ref1</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a spatialExperiment-based object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)

## Explore the object
toySRT 
</code></pre>

<hr>
<h2 id='EstParam'>Access Model Fitting Parameters</h2><span id='topic+EstParam'></span>

<h3>Description</h3>

<p>Access Model Fitting Parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EstParam(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EstParam_+3A_x">x</code></td>
<td>
<p>SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of estimated parameters by fitting models
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
EstParam(toySRT)

</code></pre>

<hr>
<h2 id='exampleLIBD'>Data used for creating vignettes</h2><span id='topic+exampleLIBD'></span>

<h3>Description</h3>

<p>A data list containing the a gene expression matrix and a location dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exampleLIBD
</code></pre>


<h3>Format</h3>

<p>A data list
</p>

<dl>
<dt>example_count</dt><dd><p>A sparse matrix with 80 rows and 3611 columns.</p>
</dd>
<dt>example_loc</dt><dd><p>A data frame with 3611 rows and 6 columns.</p>
</dd>
</dl>



<h3>Source</h3>

<p>created based on a SpatialLIBD data (SampleID: 151673) to serve as an example
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleLIBD)       #Lazy loading. Data becomes visible as soon as called
</code></pre>

<hr>
<h2 id='fit_pos_optim'>fitting data with poisson through optim function</h2><span id='topic+fit_pos_optim'></span>

<h3>Description</h3>

<p>fitting data with poisson through optim function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_pos_optim(x, maxiter = 500)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_pos_optim_+3A_x">x</code></td>
<td>
<p>A vector of count values to be fitted</p>
</td></tr>
<tr><td><code id="fit_pos_optim_+3A_maxiter">maxiter</code></td>
<td>
<p>number of iteration</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a vector with mean paramter lambda, loglikelihood value llk, convergence
</p>

<hr>
<h2 id='get_metrics_pd'>Extracted summarized metrics for reference data and synthetic data</h2><span id='topic+get_metrics_pd'></span>

<h3>Description</h3>

<p>Extracted summarized metrics for reference data and synthetic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_metrics_pd(simsrt, metric = "GeneMean")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_metrics_pd_+3A_simsrt">simsrt</code></td>
<td>
<p>A SRTsim object</p>
</td></tr>
<tr><td><code id="get_metrics_pd_+3A_metric">metric</code></td>
<td>
<p>Specification of metrics to be plotted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a dataframe for ggplot
</p>

<hr>
<h2 id='get_stats_gene'>Summarize gene-wise summary metrics</h2><span id='topic+get_stats_gene'></span>

<h3>Description</h3>

<p>Summarize gene-wise summary metrics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_stats_gene(mat, group, log_trans = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_stats_gene_+3A_mat">mat</code></td>
<td>
<p>A count matrix</p>
</td></tr>
<tr><td><code id="get_stats_gene_+3A_group">group</code></td>
<td>
<p>A group label</p>
</td></tr>
<tr><td><code id="get_stats_gene_+3A_log_trans">log_trans</code></td>
<td>
<p>A logical constant indicating whether to log transform the gene mean and variance</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a n by 5 dataframe with location metrics
</p>

<hr>
<h2 id='get_stats_loc'>Summarize location-wise summary metrics</h2><span id='topic+get_stats_loc'></span>

<h3>Description</h3>

<p>Summarize location-wise summary metrics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_stats_loc(mat, group, log_trans = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_stats_loc_+3A_mat">mat</code></td>
<td>
<p>A count matrix</p>
</td></tr>
<tr><td><code id="get_stats_loc_+3A_group">group</code></td>
<td>
<p>A group label</p>
</td></tr>
<tr><td><code id="get_stats_loc_+3A_log_trans">log_trans</code></td>
<td>
<p>A logical constant indicating whether to log transform the libsize</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a n by 3 dataframe with location metrics
</p>

<hr>
<h2 id='metaParam'>Access User-Specified Parameters</h2><span id='topic+metaParam'></span>

<h3>Description</h3>

<p>Access User-Specified Parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaParam(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaParam_+3A_x">x</code></td>
<td>
<p>SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of user-specified parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
metaParam(toySRT)

</code></pre>

<hr>
<h2 id='refcolData'>Access reference colData</h2><span id='topic+refcolData'></span>

<h3>Description</h3>

<p>Access reference colData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refcolData(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refcolData_+3A_x">x</code></td>
<td>
<p>SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the colData of reference data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
refcolData(toySRT)

</code></pre>

<hr>
<h2 id='refCounts'>Access reference count matrix</h2><span id='topic+refCounts'></span>

<h3>Description</h3>

<p>Access reference count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refCounts(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refCounts_+3A_x">x</code></td>
<td>
<p>SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a reference count matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
refCounts(toySRT)[1:3,1:3]

</code></pre>

<hr>
<h2 id='refrowData'>Access reference rowData</h2><span id='topic+refrowData'></span>

<h3>Description</h3>

<p>Access reference rowData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refrowData(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refrowData_+3A_x">x</code></td>
<td>
<p>SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the rowData of reference data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
refrowData(toySRT)

</code></pre>

<hr>
<h2 id='reGenCountshiny'>ReSimulate Count Data with Parameters Specification from Shiny</h2><span id='topic+reGenCountshiny'></span>

<h3>Description</h3>

<p>ReSimulate Count Data with Parameters Specification from Shiny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reGenCountshiny(shinyOutput, NewSeed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reGenCountshiny_+3A_shinyoutput">shinyOutput</code></td>
<td>
<p>A list of Shiny Output. Including a simCount, simInfo,simcountParam,simLocParam</p>
</td></tr>
<tr><td><code id="reGenCountshiny_+3A_newseed">NewSeed</code></td>
<td>
<p>A new seed for data generation. Useful when multiple replicates are needed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a Count DataFrame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Re-generate Count Data based on ShinyOutput Parameters, should be same as simCount in ShinyOutput
cMat &lt;- reGenCountshiny(toyShiny)

## Generate Count Data with A New Seed based on ShinyOutput Parameters 
cMat2 &lt;- reGenCountshiny(toyShiny,NewSeed=2)

## Comparison across the output
toyShiny$simCount[1:3,1:3]
cMat[1:3,1:3]
cMat2[1:3,1:3]

</code></pre>

<hr>
<h2 id='Shiny2SRT'>Create a SRTsim object from reference-free shinyoutput</h2><span id='topic+Shiny2SRT'></span>

<h3>Description</h3>

<p>Create a SRTsim object from reference-free shinyoutput
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Shiny2SRT(shinyOutput)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Shiny2SRT_+3A_shinyoutput">shinyOutput</code></td>
<td>
<p>A list of Shiny Output. Including a simCount, simInfo,simcountParam,simLocParam</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a SRTsim object with user-specified parameters stored in metaParam slot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
shinySRT &lt;- Shiny2SRT(toyShiny)

## Explore the new SRT object
shinySRT@metaParam
shinySRT@simCounts[1:3,1:3]
shinySRT@simcolData
</code></pre>

<hr>
<h2 id='simcolData'>Access synthetic colData</h2><span id='topic+simcolData'></span>

<h3>Description</h3>

<p>Access synthetic colData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simcolData(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simcolData_+3A_x">x</code></td>
<td>
<p>SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the colData of synthetic data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
simcolData(toySRT)

</code></pre>

<hr>
<h2 id='simCounts'>Access synthetic count matrix</h2><span id='topic+simCounts'></span>

<h3>Description</h3>

<p>Access synthetic count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simCounts(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simCounts_+3A_x">x</code></td>
<td>
<p>SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a synthetic count matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
simCounts(toySRT)[1:3,1:3]

</code></pre>

<hr>
<h2 id='simNewLocs'>Fit the marginal distributions for single gene</h2><span id='topic+simNewLocs'></span>

<h3>Description</h3>

<p>Fit the marginal distributions for single gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simNewLocs(newN, lay_out = c("grid", "random"), preLoc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simNewLocs_+3A_newn">newN</code></td>
<td>
<p>A integer specifying the number of spatial locations in the synthetic data</p>
</td></tr>
<tr><td><code id="simNewLocs_+3A_lay_out">lay_out</code></td>
<td>
<p>A character string specifying arrangement of new generated spatial locations. Default is &quot;grid&quot;</p>
</td></tr>
<tr><td><code id="simNewLocs_+3A_preloc">preLoc</code></td>
<td>
<p>A data frame of shape n by 3 that x, y coodinates and domain label</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a n by 2 dataframe with newly generated spatial locations
</p>

<hr>
<h2 id='simrowData'>Access synthetic rowData</h2><span id='topic+simrowData'></span>

<h3>Description</h3>

<p>Access synthetic rowData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simrowData(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simrowData_+3A_x">x</code></td>
<td>
<p>SRTsim object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the rowData of synthetic data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)
simrowData(toySRT)

</code></pre>

<hr>
<h2 id='srtsim_cci_free'>Generate Data with Cell-Cell Interaction Under Reference-Free Mode</h2><span id='topic+srtsim_cci_free'></span>

<h3>Description</h3>

<p>Generate Data with Cell-Cell Interaction Under Reference-Free Mode
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srtsim_cci_free(
  zero_prop_in = 0,
  disper_in = Inf,
  mu_in = 1,
  numGene = 1000,
  location_in,
  region_cell_map,
  fc = 3,
  LR_in,
  sim_seed = 1,
  numKNN = 4,
  numSingleCellType = 2000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srtsim_cci_free_+3A_zero_prop_in">zero_prop_in</code></td>
<td>
<p>A number specifying zero proportion for the count model, default is 0</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_disper_in">disper_in</code></td>
<td>
<p>A number specifying dispersion for the count model, default is Inf. Same as the size parameter in rnbinom.</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_mu_in">mu_in</code></td>
<td>
<p>A number specifying mean for background count model, default is 1</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_numgene">numGene</code></td>
<td>
<p>An integer specifying the number of genes in the synthetic data, default is 1000</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_location_in">location_in</code></td>
<td>
<p>A dataframe with x, y, and region_label</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_region_cell_map">region_cell_map</code></td>
<td>
<p>A dataframe specifying the cell type proportion in each region. Row: region,Column: cell type.</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_fc">fc</code></td>
<td>
<p>A number specifying effect size for ligand-receptor pairs that mediate the cel-cell communication, default is 3</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_lr_in">LR_in</code></td>
<td>
<p>A dataframe specifying ligand and receptor pairs, containing four columns: protein_a, protein_b, celltypeA, and celltype B</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_sim_seed">sim_seed</code></td>
<td>
<p>A number for reproducible purpose</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_numknn">numKNN</code></td>
<td>
<p>A number specifying number of nearest neighbors with elevated gene expressin levels, default is 4</p>
</td></tr>
<tr><td><code id="srtsim_cci_free_+3A_numsinglecelltype">numSingleCellType</code></td>
<td>
<p>A number specifying number of spots in the background pool. Gene expression count are then sampled from this background pool.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a SRTsim object with a newly generated count matrix and correspoding parameters
</p>

<hr>
<h2 id='srtsim_cci_ref'>Generate Data with Cell-Cell Interaction Under Reference-Based Mode</h2><span id='topic+srtsim_cci_ref'></span>

<h3>Description</h3>

<p>Generate Data with Cell-Cell Interaction Under Reference-Based Mode
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srtsim_cci_ref(
  EstParam = NULL,
  numGene = 1000,
  location_in,
  region_cell_map,
  fc = 3,
  LR_in,
  sim_seed = 1,
  numKNN = 4,
  numSingleCellType = 2000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srtsim_cci_ref_+3A_estparam">EstParam</code></td>
<td>
<p>A list of estimated parameters from srtsim_fit function, EstParam slot if the simSRT object.</p>
</td></tr>
<tr><td><code id="srtsim_cci_ref_+3A_numgene">numGene</code></td>
<td>
<p>An integer specifying the number of genes in the synthetic data, default is 1000</p>
</td></tr>
<tr><td><code id="srtsim_cci_ref_+3A_location_in">location_in</code></td>
<td>
<p>A dataframe with x, y, and region_label</p>
</td></tr>
<tr><td><code id="srtsim_cci_ref_+3A_region_cell_map">region_cell_map</code></td>
<td>
<p>A dataframe specifying the cell type proportion in each region. Row: region,Column: cell type.</p>
</td></tr>
<tr><td><code id="srtsim_cci_ref_+3A_fc">fc</code></td>
<td>
<p>A number specifying effect size for ligand-receptor pairs that mediate the cel-cell communication, default is 3</p>
</td></tr>
<tr><td><code id="srtsim_cci_ref_+3A_lr_in">LR_in</code></td>
<td>
<p>A dataframe specifying ligand and receptor pairs, containing four columns: protein_a, protein_b, celltypeA, and celltype B</p>
</td></tr>
<tr><td><code id="srtsim_cci_ref_+3A_sim_seed">sim_seed</code></td>
<td>
<p>A number for reproducible purpose</p>
</td></tr>
<tr><td><code id="srtsim_cci_ref_+3A_numknn">numKNN</code></td>
<td>
<p>A number specifying number of nearest neighbors with elevated gene expressin levels, default is 4</p>
</td></tr>
<tr><td><code id="srtsim_cci_ref_+3A_numsinglecelltype">numSingleCellType</code></td>
<td>
<p>A number specifying number of spots in the background pool. Gene expression count are then sampled from this background pool.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a SRTsim object with a newly generated count matrix and correspoding parameters
</p>

<hr>
<h2 id='srtsim_count'>Generate Data with Estimated Parameters</h2><span id='topic+srtsim_count'></span>

<h3>Description</h3>

<p>Generate Data with Estimated Parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srtsim_count(
  simsrt,
  breaktie = "random",
  total_count_new = NULL,
  total_count_old = NULL,
  rrr = NULL,
  nn_num = 5,
  nn_func = c("mean", "median", "ransam"),
  numCores = 1,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srtsim_count_+3A_simsrt">simsrt</code></td>
<td>
<p>A object with estimated parameters from fitting step</p>
</td></tr>
<tr><td><code id="srtsim_count_+3A_breaktie">breaktie</code></td>
<td>
<p>A character string specifying how ties are treated.  Same as the &quot;tie.method&quot; in rank function</p>
</td></tr>
<tr><td><code id="srtsim_count_+3A_total_count_new">total_count_new</code></td>
<td>
<p>The (expected) total number of reads or UMIs in the simulated count matrix.</p>
</td></tr>
<tr><td><code id="srtsim_count_+3A_total_count_old">total_count_old</code></td>
<td>
<p>The total number of reads or UMIs in the original count matrix.</p>
</td></tr>
<tr><td><code id="srtsim_count_+3A_rrr">rrr</code></td>
<td>
<p>The ratio applies to the gene-specific mean estimate, used for the fixing average sequencing depth simulation. Default is null. Its specification will override the specification of total_count_new and total_count_old.</p>
</td></tr>
<tr><td><code id="srtsim_count_+3A_nn_num">nn_num</code></td>
<td>
<p>A integer of nearest neighbors, default is 5.</p>
</td></tr>
<tr><td><code id="srtsim_count_+3A_nn_func">nn_func</code></td>
<td>
<p>A character string specifying how the psedo-count to be generated. options include 'mean','median' and 'ransam'.</p>
</td></tr>
<tr><td><code id="srtsim_count_+3A_numcores">numCores</code></td>
<td>
<p>The number of cores to use</p>
</td></tr>
<tr><td><code id="srtsim_count_+3A_verbose">verbose</code></td>
<td>
<p>Whether to show running information for srtsim_count</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a SRTsim object with a newly generated count matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")
## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)

## Explore the synthetic count matrix
simCounts(toySRT)[1:3,1:3]
</code></pre>

<hr>
<h2 id='srtsim_count_affine'>Generate Data with Estimated Parameters For A New Designed Pattern</h2><span id='topic+srtsim_count_affine'></span>

<h3>Description</h3>

<p>Generate Data with Estimated Parameters For A New Designed Pattern
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srtsim_count_affine(
  simsrt,
  reflabel,
  targetlabel,
  breaktie = "random",
  nn_func = c("mean", "median", "ransam"),
  nn_num = 5,
  local_sid = NULL,
  numCores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srtsim_count_affine_+3A_simsrt">simsrt</code></td>
<td>
<p>A SRTsim object with estimated parameters from fitting step</p>
</td></tr>
<tr><td><code id="srtsim_count_affine_+3A_reflabel">reflabel</code></td>
<td>
<p>A character vector specifying labels for reference regions</p>
</td></tr>
<tr><td><code id="srtsim_count_affine_+3A_targetlabel">targetlabel</code></td>
<td>
<p>A character vector specifying labels for target regions</p>
</td></tr>
<tr><td><code id="srtsim_count_affine_+3A_breaktie">breaktie</code></td>
<td>
<p>A character string specifying how ties are treated.  Same as the &quot;tie.method&quot; in rank function</p>
</td></tr>
<tr><td><code id="srtsim_count_affine_+3A_nn_func">nn_func</code></td>
<td>
<p>A character string specifying how the psedo-count to be generated. options include 'mean','median' and 'ransam'.</p>
</td></tr>
<tr><td><code id="srtsim_count_affine_+3A_nn_num">nn_num</code></td>
<td>
<p>A integer of nearest neighbors, default is 5.</p>
</td></tr>
<tr><td><code id="srtsim_count_affine_+3A_local_sid">local_sid</code></td>
<td>
<p>A numberic seed used locally for the affine transformation. Default is NULL.</p>
</td></tr>
<tr><td><code id="srtsim_count_affine_+3A_numcores">numCores</code></td>
<td>
<p>A number of cores to use</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a SRTsim object with a newly generated count matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Prepare Data From LIBD Sample
subinfo &lt;- exampleLIBD$info[,c("imagecol","imagerow","layer")]
colnames(subinfo) &lt;- c("x","y","label")
gns 	&lt;- c("ENSG00000168314","ENSG00000183036", "ENSG00000132639" )

## Create a simSRT Object with Three Genes For a Fast Example
simSRT1 &lt;- createSRT(count_in= exampleLIBD$count[gns,],loc_in =subinfo)

## Estimate model parameters for data generation: domain-specific 
simSRT1 &lt;- srtsim_fit(simSRT1,sim_schem="domain")

## Define New Layer Structures
simSRT1@refcolData$target_label &lt;- "NL1"
simSRT1@refcolData$target_label[simSRT1@refcolData$label %in% paste0("Layer",4:5)] &lt;- "NL2"
simSRT1@refcolData$target_label[simSRT1@refcolData$label %in% c("Layer6","WM")] &lt;- "NL3"

## Perform Data Generation for New Defined Layer Structures
## Reference: WM --&gt; NL3, Layer5--&gt; NL2, Layer3 --&gt; NL1
simSRT1 &lt;- srtsim_count_affine(simSRT1,
								reflabel=c("Layer3","Layer5","WM"),
								targetlabel=c("NL1","NL2","NL3"),
								nn_func="ransam"
								)

## Visualize the Expression Pattern for Gene of Interest
visualize_gene(simsrt=simSRT1,plotgn = "ENSG00000168314",rev_y=TRUE,ptsizeCount=1)
</code></pre>

<hr>
<h2 id='srtsim_fit'>Fit the marginal distributions for each row of a count matrix</h2><span id='topic+srtsim_fit'></span>

<h3>Description</h3>

<p>Fit the marginal distributions for each row of a count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srtsim_fit(
  simsrt,
  marginal = c("auto_choose", "zinb", "nb", "poisson", "zip"),
  sim_scheme = c("tissue", "domain"),
  min_nonzero_num = 2,
  maxiter = 500
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srtsim_fit_+3A_simsrt">simsrt</code></td>
<td>
<p>A SRTsim object</p>
</td></tr>
<tr><td><code id="srtsim_fit_+3A_marginal">marginal</code></td>
<td>
<p>Specification of the types of marginal distribution.Default value is 'auto_choose' which chooses between ZINB, NB, ZIP, and Poisson by a likelihood ratio test (lrt),AIC and whether there is underdispersion.'zinb' will fit the ZINB model. If there is underdispersion, it will fit the Poisson model. If there is no zero at all or an error occurs, it will fit an NB model instead.'nb' fits the NB model and chooses between NB and Poisson depending on whether there is underdispersion. 'poisson' simply fits the Poisson model.'zip' fits the ZIP model and chooses between ZIP and Poisson by a likelihood ratio test</p>
</td></tr>
<tr><td><code id="srtsim_fit_+3A_sim_scheme">sim_scheme</code></td>
<td>
<p>a character string specifying simulation scheme. &quot;tissue&quot; stands for tissue-based
simulation; &quot;domain&quot; stands for domain-specific simulation. Default is &quot;tissue&quot;.</p>
</td></tr>
<tr><td><code id="srtsim_fit_+3A_min_nonzero_num">min_nonzero_num</code></td>
<td>
<p>The minimum number of non-zero values required for a gene to be fitted. Default is 2.</p>
</td></tr>
<tr><td><code id="srtsim_fit_+3A_maxiter">maxiter</code></td>
<td>
<p>The number of iterations for the model-fitting. Default is 500.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object with estimated parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")

</code></pre>

<hr>
<h2 id='srtsim_newlocs'>Fit the marginal distributions for each row of a count matrix</h2><span id='topic+srtsim_newlocs'></span>

<h3>Description</h3>

<p>Fit the marginal distributions for each row of a count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srtsim_newlocs(
  simsrt,
  new_loc_num = NULL,
  loc_lay_out = c("grid", "random"),
  voting_nn = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="srtsim_newlocs_+3A_simsrt">simsrt</code></td>
<td>
<p>A SRTsim object</p>
</td></tr>
<tr><td><code id="srtsim_newlocs_+3A_new_loc_num">new_loc_num</code></td>
<td>
<p>A integer specifying the number of spatial locations in the synthetic data</p>
</td></tr>
<tr><td><code id="srtsim_newlocs_+3A_loc_lay_out">loc_lay_out</code></td>
<td>
<p>a character string specifying arrangement of new generated spatial locations. Default is &quot;grid&quot;</p>
</td></tr>
<tr><td><code id="srtsim_newlocs_+3A_voting_nn">voting_nn</code></td>
<td>
<p>A integer of nearest neighbors used in label assignment for new generated locations. Default is 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a object with estimated parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)

## Create New Locations within Profile
toySRT2  &lt;- srtsim_newlocs(toySRT,new_loc_num=1000)

## Explore New Generated Locations
simcolData(toySRT2)
</code></pre>

<hr>
<h2 id='SRTsim_shiny'>Run the SRTsim Shiny Application</h2><span id='topic+SRTsim_shiny'></span>

<h3>Description</h3>

<p>Run the SRTsim Shiny Application
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SRTsim_shiny()
</code></pre>


<h3>Value</h3>

<p>A list that contains a count matrix, a location dataframe, and all parameter specifications.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
 ## Will Load an Interactive Session
shinyOutput &lt;- SRTsim_shiny()

## End(Not run)
</code></pre>

<hr>
<h2 id='subsetSRT'>Subset SRT object based on domain labels of interest</h2><span id='topic+subsetSRT'></span>

<h3>Description</h3>

<p>Subset SRT object based on domain labels of interest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetSRT(simsrt, sel_label = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetSRT_+3A_simsrt">simsrt</code></td>
<td>
<p>A SRTsim object</p>
</td></tr>
<tr><td><code id="subsetSRT_+3A_sel_label">sel_label</code></td>
<td>
<p>A vector of selected domain labels used for the data generation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a spatialExperiment-based object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Only Keep the Spatial Locations labelled as "A" in the reference data
subtoySRT &lt;- subsetSRT(toySRT,sel_label = "A")

</code></pre>

<hr>
<h2 id='toyData'>A toyExample to showcase reference-based simulations</h2><span id='topic+toyData'></span>

<h3>Description</h3>

<p>A data list containing the a gene expression matrix and a location dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toyData
</code></pre>


<h3>Format</h3>

<p>A data list
</p>

<dl>
<dt>toyCount</dt><dd><p>A sparse matrix with 100 rows and 251 columns.</p>
</dd>
<dt>toyInfo</dt><dd><p>A data frame with 251 rows and 3 columns.</p>
</dd>
</dl>



<h3>Source</h3>

<p>created based on a ST Human Breast Cancer data to serve as an example
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(toyData)       #Lazy loading. Data becomes visible as soon as called
</code></pre>

<hr>
<h2 id='toyShiny'>A toyExample to showcase reference-free simulations</h2><span id='topic+toyShiny'></span>

<h3>Description</h3>

<p>A list of shiny output
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toyShiny
</code></pre>


<h3>Format</h3>

<p>A list of shiny output
</p>

<dl>
<dt>simCount</dt><dd><p>A data frame with 150 rows and 980 columns.</p>
</dd>
<dt>simInfo</dt><dd><p>A data frame with 980 rows and 4 columns: x, y, group, foldchange</p>
</dd>
<dt>simcountParam</dt><dd><p>A list of user-specified parameters for count generation</p>
</dd>
<dt>simLocParam</dt><dd><p>A list of user-specified parameters for pattern design</p>
</dd>
</dl>



<h3>Source</h3>

<p>created based using the SRTsim_shiny()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(toyShiny)       #Lazy loading. Data becomes visible as soon as called
</code></pre>

<hr>
<h2 id='visualize_gene'>Visualize expression pattern for the gene of interest in reference data and synthetic data</h2><span id='topic+visualize_gene'></span>

<h3>Description</h3>

<p>Visualize expression pattern for the gene of interest in reference data and synthetic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_gene(
  simsrt,
  plotgn = NULL,
  ptsizeCount = 2,
  textsizeCount = 12,
  rev_y = FALSE,
  virOption = "D",
  virDirection = -1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_gene_+3A_simsrt">simsrt</code></td>
<td>
<p>A SRTsim object</p>
</td></tr>
<tr><td><code id="visualize_gene_+3A_plotgn">plotgn</code></td>
<td>
<p>A gene name selected for visualization</p>
</td></tr>
<tr><td><code id="visualize_gene_+3A_ptsizecount">ptsizeCount</code></td>
<td>
<p>Specification of point size. Default is 2.</p>
</td></tr>
<tr><td><code id="visualize_gene_+3A_textsizecount">textsizeCount</code></td>
<td>
<p>Specification of axis font size. Default is 12.</p>
</td></tr>
<tr><td><code id="visualize_gene_+3A_rev_y">rev_y</code></td>
<td>
<p>Logical indicating whether to reverse the y axis. Default is FALSE. Useful for Visualize the LIBD data.</p>
</td></tr>
<tr><td><code id="visualize_gene_+3A_viroption">virOption</code></td>
<td>
<p>Specification of <code>option</code> in the <code>scale_color_viridis</code>. Default is &quot;D&quot;. User can choose a letter from 'A' to 'H'.</p>
</td></tr>
<tr><td><code id="visualize_gene_+3A_virdirection">virDirection</code></td>
<td>
<p>Specification of <code>direction</code> in the <code>scale_color_viridis</code>. Default is &quot;-1&quot;.  User can choose '1' or '-1'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns two expression plots for the gene of interest
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)
## Create New Locations within Profile
toySRT2 &lt;- srtsim_newlocs(toySRT,new_loc_num=1000)

## Estimate model parameters for data generation
toySRT2 &lt;- srtsim_fit(toySRT2,sim_schem="tissue")

## Generate synthetic data with estimated parameters
toySRT2     &lt;- srtsim_count(toySRT2,rrr=1)

## compare the expression pattern of HLA-B in synthetic data and reference data
visualize_gene(simsrt=toySRT2,plotgn = "HLA-B")
</code></pre>

<hr>
<h2 id='visualize_metrics'>Visualize summarized metrics for reference data and synthetic data</h2><span id='topic+visualize_metrics'></span>

<h3>Description</h3>

<p>Visualize summarized metrics for reference data and synthetic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_metrics(
  simsrt,
  metric_type = c("all", "genewise", "locwise", "GeneMean", "GeneVar", "GeneCV",
    "GeneZeroProp", "LocZeroProp", "LocLibSize"),
  colorpalette = "Set3",
  axistextsize = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_metrics_+3A_simsrt">simsrt</code></td>
<td>
<p>A SRTsim object</p>
</td></tr>
<tr><td><code id="visualize_metrics_+3A_metric_type">metric_type</code></td>
<td>
<p>Specification of metrics to be plotted. Default value is 'all', which will plot all six metrics: including four gene-wise metrics and two location-wise metrics. &quot;genewise&quot; will produce violin plots for all four gene-wise metrics; &quot;locwise&quot; will produce violin plots for all two location-wise metrics; &quot;GeneMean&quot;, &quot;GeneVar&quot;, &quot;GeneCV&quot;, &quot;GeneZeroProp&quot;, &quot;LocZeroProp&quot;, and &quot;LocLibSize&quot; will produce single violin plot for the corresponding metric.</p>
</td></tr>
<tr><td><code id="visualize_metrics_+3A_colorpalette">colorpalette</code></td>
<td>
<p>Specification of color palette to be passed to <code>palette</code> in the <code>scale_fill_brewer</code>. Default is &quot;Set3&quot;</p>
</td></tr>
<tr><td><code id="visualize_metrics_+3A_axistextsize">axistextsize</code></td>
<td>
<p>Specification of axis font size. Default is 12.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of ggplots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

## Create a simSRT object
toySRT  &lt;- createSRT(count_in=toyData$toyCount,loc_in = toyData$toyInfo)
set.seed(1)

## Estimate model parameters for data generation
toySRT &lt;- srtsim_fit(toySRT,sim_schem="tissue")

## Generate synthetic data with estimated parameters
toySRT &lt;- srtsim_count(toySRT)

## Compute metrics 
toySRT   &lt;- compareSRT(toySRT)

## Visualize Metrics
visualize_metrics(toySRT)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
