<!DOCTYPE html><html lang="en"><head><title>Help for package ssdGSA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ssdGSA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#avg_expression'><p>Function to Calculate the Average of Gene Expressions</p></a></li>
<li><a href='#check_gene_name_match'><p>Function to Check if Gene IDs in the Input Gene Data Match Well</p></a></li>
<li><a href='#check_gene_name_match_noDir'><p>Function to Check if Gene IDs in the Input Gene Data Match Well When Direction Matrix Is Missing</p></a></li>
<li><a href='#check_genes_missing'><p>Function to Check if Genes in Gene Sets Have Missing Information in Data Matrix and Direction Matrix</p></a></li>
<li><a href='#check_genes_missing_noDir'><p>Function to Check if Genes in Gene Sets Have Missing Information in Data Matrix</p></a></li>
<li><a href='#check_genes_missing_total'><p>Function to Check if 100% of Genes in Gene Sets Have Missing Information in Data Matrix</p></a></li>
<li><a href='#data_matrix'><p>This is data to be included in package</p></a></li>
<li><a href='#data_matrix_entrezID'><p>This is data to be included in package</p></a></li>
<li><a href='#direction_matrix'><p>This is data to be included in package</p></a></li>
<li><a href='#gene_sets'><p>This is data to be included in package</p></a></li>
<li><a href='#median_expression'><p>Function to Calculate Median of Gene Expressions</p></a></li>
<li><a href='#ssdGSA'><p>Single Sample Directional Gene Set Analysis (ssdGSA)</p></a></li>
<li><a href='#ssdGSA_individual'><p>Single Sample Directional Gene Set Analysis Using Individual Weighted Scores</p></a></li>
<li><a href='#ssGSA'><p>Function to Calculate Single Sample Gene Set Scores without Direction Matrix</p></a></li>
<li><a href='#transform_ensembl_2_entrez'><p>Function to Transform ENSEMBL ID to ENTREZ ID</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single Sample Directional Gene Set Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A method that inherits the standard gene set variation analysis (GSVA) method and also provides the option to use summary statistics from any analysis (disease vs healthy, lesional side vs nonlesional side, etc..) input to define the direction of gene sets used for directional gene set score calculation for a given disease. Note to use this package, GSVA(&gt;= 1.52.1) is needed to pre-installed. 
    Hanzelmann, S., Castelo, R., and Guinney, J. (2013) &lt;<a href="https://doi.org/10.1186%2F1471-2105-14-7">doi:10.1186/1471-2105-14-7</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0),</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>GSVA, dplyr, purrr, stringr, tibble, vctrs, stats, tidyselect,
utils, clusterProfiler, org.Hs.eg.db</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-26 20:30:18 UTC; QIANQ09</td>
</tr>
<tr>
<td>Author:</td>
<td>Xingpeng Li [aut, cre],
  Qi Qian [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xingpeng Li &lt;xingpeng.li@pfizer.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-26 23:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='avg_expression'>Function to Calculate the Average of Gene Expressions</h2><span id='topic+avg_expression'></span>

<h3>Description</h3>

<p>This function is to calculate the average of gene expressions for genes in the given gene sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>avg_expression(Data, pathway.db)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="avg_expression_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ensembl ID as row names and columns corresponding to different samples.</p>
</td></tr>
<tr><td><code id="avg_expression_+3A_pathway.db">pathway.db</code></td>
<td>
<p>A list of gene sets.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Within the ssdGSA function, when GSA_method = &quot;avg.exprs&quot;, this function is used to calculate the average of gene
expressions for genes in the given gene sets.
</p>


<h3>Value</h3>

<p>Matrix of average gene expression in each gene set with rows corresponding to gene sets and columns corresponding to
samples will be returned.
</p>

<hr>
<h2 id='check_gene_name_match'>Function to Check if Gene IDs in the Input Gene Data Match Well</h2><span id='topic+check_gene_name_match'></span>

<h3>Description</h3>

<p>This function is to check if the gene IDs in the gene sets, data matrix and direction matrix match well.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gene_name_match(
  genes_in_Gene_sets,
  genes_in_Data,
  genes_in_Direction_matrix
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_gene_name_match_+3A_genes_in_gene_sets">genes_in_Gene_sets</code></td>
<td>
<p>A list of gene names from the gene sets.</p>
</td></tr>
<tr><td><code id="check_gene_name_match_+3A_genes_in_data">genes_in_Data</code></td>
<td>
<p>A list of gene names from the data matrix.</p>
</td></tr>
<tr><td><code id="check_gene_name_match_+3A_genes_in_direction_matrix">genes_in_Direction_matrix</code></td>
<td>
<p>A list of gene names from the direction matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Before single sample directional gene set analysis, it is necessary to check whether the gene ID types in the
gene sets, data matrix and direction matrix have the same gene ID type. If not, the ssdGSA and ssdGSA_individual would stop,
and users should double check to make gene ID types in different parts match one another.
</p>


<h3>Value</h3>

<p>If there are more than 10\
the single sample directional gene set analysis would stop.
</p>

<hr>
<h2 id='check_gene_name_match_noDir'>Function to Check if Gene IDs in the Input Gene Data Match Well When Direction Matrix Is Missing</h2><span id='topic+check_gene_name_match_noDir'></span>

<h3>Description</h3>

<p>This function is to check if the gene IDs in the gene sets and data matrix match well.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_gene_name_match_noDir(genes_in_Gene_sets, genes_in_Data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_gene_name_match_noDir_+3A_genes_in_gene_sets">genes_in_Gene_sets</code></td>
<td>
<p>A list of gene names from the gene sets.</p>
</td></tr>
<tr><td><code id="check_gene_name_match_noDir_+3A_genes_in_data">genes_in_Data</code></td>
<td>
<p>A list of gene names from the data matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Before single sample directional gene set analysis, it is necessary to check whether the gene ID types in the in the
gene sets, data matrix and direction matrix have the same ID type. If not, users should double check to make gene ID types
match one another.
</p>


<h3>Value</h3>

<p>If there are more than 10\
the single sample directional gene set analysis would stop.
</p>

<hr>
<h2 id='check_genes_missing'>Function to Check if Genes in Gene Sets Have Missing Information in Data Matrix and Direction Matrix</h2><span id='topic+check_genes_missing'></span>

<h3>Description</h3>

<p>This function is to check if genes in gene sets to be analyzed have missing information in data matrix and direction matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_genes_missing(Gene_sets, Data, Direction_matrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_genes_missing_+3A_gene_sets">Gene_sets</code></td>
<td>
<p>A list of gene sets to be analyzed, with gene set names as component names, and each component is a vector of gene entrez ID.</p>
</td></tr>
<tr><td><code id="check_genes_missing_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ensembl ID as row names and columns corresponding to different samples.</p>
</td></tr>
<tr><td><code id="check_genes_missing_+3A_direction_matrix">Direction_matrix</code></td>
<td>
<p>Matrix containing directionality information for each gene, such as effect size, p value of summary statistics.
Each row of the matrix is for one gene, and there should be at least two columns (with the 1st column containing
gene entrez ID, and 2nd column containing directionality information).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Before single sample directional gene set analysis, it is necessary to check if genes in the gene sets have missing information in
data matrix and direction matrix. If not, warning messages would be given such that users can double check whether the gene set analysis results
are reliable.
</p>


<h3>Value</h3>

<p>When at least one gene in the gene sets have information missing in data matrix or direction matrix, warning messages will be given, as
well as the percentages (missing number/total number) of gene sets. If less than 10 gene sets have missing information, percentages
(missing number/total number) of genes in each gene set that have missing information in data matrix and direction matrix will also be reported.
However, if more than 10 gene sets have missing information, no detailed individual gene set missing information will be reported. Also note that
if a gene set has 100% information missing in the data or direction matrix, the name of the gene set will be notated.
</p>

<hr>
<h2 id='check_genes_missing_noDir'>Function to Check if Genes in Gene Sets Have Missing Information in Data Matrix</h2><span id='topic+check_genes_missing_noDir'></span>

<h3>Description</h3>

<p>This function is to check if genes in gene sets to be analyzed have missing information in data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_genes_missing_noDir(Gene_sets, Data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_genes_missing_noDir_+3A_gene_sets">Gene_sets</code></td>
<td>
<p>A list of gene sets to be analyzed, with gene set names as component names, and each component is a vector
of gene entrez ID.</p>
</td></tr>
<tr><td><code id="check_genes_missing_noDir_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ensembl ID as row names and columns corresponding to different samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Before single sample directional gene set analysis, it is necessary to check if genes in the gene sets have missing information in
data matrix. If not, warning messages would be given such that users can double check whether the gene set analysis results
are reliable.
</p>


<h3>Value</h3>

<p>When at least one gene in the gene sets have information missing in data matrix, warning messages will be given, as
well as the percentages (missing number/total number) of gene sets. If less than 10 gene sets have missing information, percentages
(missing number/total number) of genes in each gene set that have missing information in data matrix and direction matrix will also be reported.
However, if more than 10 gene sets have missing information, no detailed individual gene set missing information will be reported. Also note that
if a gene set has 100% information missing in the data matrix, the name of the gene set will be notated.
</p>

<hr>
<h2 id='check_genes_missing_total'>Function to Check if 100% of Genes in Gene Sets Have Missing Information in Data Matrix</h2><span id='topic+check_genes_missing_total'></span>

<h3>Description</h3>

<p>This function is to check if 100% of genes in gene sets to be analyzed have missing information in data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_genes_missing_total(Gene_sets, Data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_genes_missing_total_+3A_gene_sets">Gene_sets</code></td>
<td>
<p>A list of gene sets to be analyzed, with gene set names as component names, and each component is a vector
of gene entrez ID.</p>
</td></tr>
<tr><td><code id="check_genes_missing_total_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ensembl ID as row names and columns corresponding to different samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Before single sample directional gene set analysis, it is necessary to check if genes in the gene sets have missing information in
data matrix. If a gene set has 100% information missing in the data matrix, the name of the gene set will be returned as a list named
'Total_missing_in_data_matrix'; If no such gene sets exist, nothing will be returned.
</p>


<h3>Value</h3>

<p>A list 'Total_missing_in_data_matrix' with names of the gene sets that have 100% of genes that have information missing in the data matrix will be
returned. If there are no such gene sets, NULL list will be returned.
</p>

<hr>
<h2 id='data_matrix'>This is data to be included in package</h2><span id='topic+data_matrix'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_matrix
</code></pre>


<h3>Format</h3>

<p>An example data matrix of gene expressions with gene ensembl ID as row names and
columns corresponding to different samples.
</p>

<dl>
<dt>S1</dt><dd><p>gene expression for this gene from the 1st sample</p>
</dd>
<dt>S2</dt><dd><p>gene expression for this gene from the 2nd sample</p>
</dd>
<dt>S3</dt><dd><p>gene expression for this gene from the 3rd sample</p>
</dd>
<dt>S4</dt><dd><p>gene expression for this gene from the 4th sample</p>
</dd>
<dt>S5</dt><dd><p>gene expression for this gene from the 5th sample</p>
</dd>
<dt>S6</dt><dd><p>gene expression for this gene from the 6th sample</p>
</dd>
<dt>S7</dt><dd><p>gene expression for this gene from the 7th sample</p>
</dd>
<dt>S8</dt><dd><p>gene expression for this gene from the 8th sample</p>
</dd>
<dt>S9</dt><dd><p>gene expression for this gene from the 9th sample</p>
</dd>
<dt>S10</dt><dd><p>gene expression for this gene from the 10th sample</p>
</dd>
</dl>


<hr>
<h2 id='data_matrix_entrezID'>This is data to be included in package</h2><span id='topic+data_matrix_entrezID'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_matrix_entrezID
</code></pre>


<h3>Format</h3>

<p>An example data matrix of gene expressions with gene entrez ID as row names and
columns corresponding to different samples.
</p>

<dl>
<dt>S1</dt><dd><p>gene expression for this gene from the 1st sample</p>
</dd>
<dt>S2</dt><dd><p>gene expression for this gene from the 2nd sample</p>
</dd>
<dt>S3</dt><dd><p>gene expression for this gene from the 3rd sample</p>
</dd>
<dt>S4</dt><dd><p>gene expression for this gene from the 4th sample</p>
</dd>
<dt>S5</dt><dd><p>gene expression for this gene from the 5th sample</p>
</dd>
<dt>S6</dt><dd><p>gene expression for this gene from the 6th sample</p>
</dd>
<dt>S7</dt><dd><p>gene expression for this gene from the 7th sample</p>
</dd>
<dt>S8</dt><dd><p>gene expression for this gene from the 8th sample</p>
</dd>
<dt>S9</dt><dd><p>gene expression for this gene from the 9th sample</p>
</dd>
<dt>S10</dt><dd><p>gene expression for this gene from the 10th sample</p>
</dd>
</dl>


<hr>
<h2 id='direction_matrix'>This is data to be included in package</h2><span id='topic+direction_matrix'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>direction_matrix
</code></pre>


<h3>Format</h3>

<p>An example direction matrix containing directionality information from summary statistics
such as effect size (ES) or p value, with each row for one gene.
</p>

<dl>
<dt>gene</dt><dd><p>gene entrez ID</p>
</dd>
<dt>ES</dt><dd><p>effect size (SE) for this gene from summary statistics</p>
</dd>
<dt>pval</dt><dd><p>p value for this gene from summary statistics</p>
</dd>
</dl>


<hr>
<h2 id='gene_sets'>This is data to be included in package</h2><span id='topic+gene_sets'></span>

<h3>Description</h3>

<p>This is data to be included in package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_sets
</code></pre>


<h3>Format</h3>

<p>An example disease gene sets in the form of a list, with gene set names as list component names,
and each component is a vector of gene entrez ID. In this sample gene sets list, there are 10 gene sets in total.
</p>

<dl>
<dt>TCELL.KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
<dt>TLR.KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
<dt>BCELL.KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
<dt>NEUROTROPHIN.KEGG_NEUROTROPHIN_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
<dt>ERBB.KEGG_ERBB_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
<dt>CALCIUM.KEGG_CALCIUM_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
<dt>CHEMOKINE.KEGG_CHEMOKINE_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
<dt>GNRH.KEGG_GNRH_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
<dt>VEGF.KEGG_VEGF_SIGNALING_PATHWAY</dt><dd><p>gene entrez ID related to this pathway</p>
</dd>
</dl>


<hr>
<h2 id='median_expression'>Function to Calculate Median of Gene Expressions</h2><span id='topic+median_expression'></span>

<h3>Description</h3>

<p>This function is to calculate the median of gene expressions for genes in the given gene sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>median_expression(Data, pathway.db)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="median_expression_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ensembl ID as row names and columns corresponding to different samples.</p>
</td></tr>
<tr><td><code id="median_expression_+3A_pathway.db">pathway.db</code></td>
<td>
<p>A list of gene sets.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Within the ssdGSA function, when GSA_method = &quot;median.exprs&quot;, this function is used to calculate the average of
gene expressions for genes in the given gene sets.
</p>


<h3>Value</h3>

<p>Matrix of average gene expression in each gene set with rows corresponding to gene sets and columns corresponding to
samples will be returned.
</p>

<hr>
<h2 id='ssdGSA'>Single Sample Directional Gene Set Analysis (ssdGSA)</h2><span id='topic+ssdGSA'></span>

<h3>Description</h3>

<p>This function is to calculate directional (disease weighted) gene set scores by incorporating each gene's
correlation to a disease or pathway in the gene set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssdGSA(
  Data,
  Gene_sets,
  Direction_matrix = NULL,
  GSA_weight = "equal_weighted",
  GSA_weighted_by = "sum.ES",
  GSA_method = "gsva",
  min.sz = 1,
  max.sz = 2000,
  mx.diff = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ssdGSA_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ID as row names and columns corresponding to different samples.</p>
</td></tr>
<tr><td><code id="ssdGSA_+3A_gene_sets">Gene_sets</code></td>
<td>
<p>A list of gene sets with gene set names as component names, and each component is a vector of gene ID.</p>
</td></tr>
<tr><td><code id="ssdGSA_+3A_direction_matrix">Direction_matrix</code></td>
<td>
<p>Matrix containing directionality information for each gene, such as effect size, t statistics,
p value of summary statistics. Each row of the direction matrix is for one gene, and there should be at least two columns
(with the 1st column containing gene entrez ID, and 2nd column containing directionality information). Note that the default is
&quot;Direction_matrix = NULL&quot;, meaning that no direction matrix is inputted, then the classic single sample gene set scores without
direction information would be calculated and returned.</p>
</td></tr>
<tr><td><code id="ssdGSA_+3A_gsa_weight">GSA_weight</code></td>
<td>
<p>Method to calculate weight in GSA. By default this is set to &quot;group_weighted&quot;. Other option is &quot;equal_weighted&quot;.</p>
</td></tr>
<tr><td><code id="ssdGSA_+3A_gsa_weighted_by">GSA_weighted_by</code></td>
<td>
<p>When &quot;group_weighted&quot; is chosen to calculate GSA_weight, further specifications are needed to specify how
group weights are calculated. By default, this is set to &quot;avg.ES&quot; (average of group ES). Other options are &quot;sum.ES&quot; (sum of group ES)
and &quot;median.ES&quot; (median of group ES).</p>
</td></tr>
<tr><td><code id="ssdGSA_+3A_gsa_method">GSA_method</code></td>
<td>
<p>Method to employ in the estimation of gene set enrichment scores per sample. By default this is set to &quot;gsva&quot;
(Hanzelmann et al, 2013). Other options are &quot;ssgsea&quot; (Barbie et al, 2009), &quot;zscore&quot; (Lee et al, 2008), &quot;avg.exprs&quot; (average value
of gene expressions in the gene set), and &quot;median.exprs&quot; (median of gene expressions in the gene set).</p>
</td></tr>
<tr><td><code id="ssdGSA_+3A_min.sz">min.sz</code></td>
<td>
<p>GSVA parameter to define the minimum size of the resulting gene sets. By default this is set to 1.</p>
</td></tr>
<tr><td><code id="ssdGSA_+3A_max.sz">max.sz</code></td>
<td>
<p>GSVA parameter to define the maximum size of the resulting gene sets. By default this is set to 2000.</p>
</td></tr>
<tr><td><code id="ssdGSA_+3A_mx.diff">mx.diff</code></td>
<td>
<p>GSVA parameter to offer two approaches to calculate the enrichment statistic from the KS random walk statistic.
mx.diff = FALSE: enrichment statistic is calculated as the maximum distance of the random walk from 0. mx.diff=TRUE (default):
enrichment statistic is calculated as the magnitude difference between the largest positive and negative random walk deviations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Single sample directional gene set analysis inherits the standard gene set variation analysis(GSVA) method, but also provides
the option to use summary statistics from any analysis (disease vs healthy, lesional side vs nonlesional side, etc..) input to define the direction of gene
sets used for directional gene set score calculation for a given disease or directional function. This function is specific for using group weighted scores.
</p>


<h3>Value</h3>

<p>Matrix of directional gene set scores with rows corresponding to gene sets and columns corresponding to different
samples will be return.
</p>


<h3>References</h3>

<p>Xingpeng Li, Qi Qian. ssdGSA - Single sample directional gene set analysis tool.
</p>
<p>Barbie, D.A. et al. Systematic RNA interference reveals that oncogenic KRAS-driven cancers require TBK1. Nature, 462(5):108-112, 2009.
</p>
<p>Hanzelmann, S., Castelo, R. and Guinney, J. GSVA: Gene set variation analysis for microarray and RNA-Seq data. BMC Bioinformatics, 14:7, 2013.
</p>
<p>Lee, E. et al. Inferring pathway activity toward precise disease classification. PLoS Comp Biol, 4(11):e1000217, 2008.
</p>
<p>Tomfohr, J. et al. Pathway level analysis of gene expression using singular value decomposition. BMC Bioinformatics, 6:225, 2005.
</p>


<h3>See Also</h3>

<p>ssdGSA_individual
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ssdGSA(Data = data_matrix_entrezID,
       Gene_sets = gene_sets[c(1,2,4)],
       Direction_matrix = direction_matrix,
       GSA_weight = "group_weighted",
       GSA_weighted_by = "sum.ES",
       GSA_method = "gsva",
       min.sz = 1,
       max.sz = 2000,
       mx.diff = TRUE
       )


</code></pre>

<hr>
<h2 id='ssdGSA_individual'>Single Sample Directional Gene Set Analysis Using Individual Weighted Scores</h2><span id='topic+ssdGSA_individual'></span>

<h3>Description</h3>

<p>This function is to calculate single sample directional (disease weighted) gene set scores for a given disease using individual weighted scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssdGSA_individual(Data, Gene_sets, Direction_matrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ssdGSA_individual_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ensembl ID as row names and columns corresponding to different samples.</p>
</td></tr>
<tr><td><code id="ssdGSA_individual_+3A_gene_sets">Gene_sets</code></td>
<td>
<p>A list of gene sets with gene set names as component names, and each component is a vector of gene entrez ID.</p>
</td></tr>
<tr><td><code id="ssdGSA_individual_+3A_direction_matrix">Direction_matrix</code></td>
<td>
<p>Matrix containing directionality information for each gene, such as effect size, t statistics,
p value of summary statistics. Each row of the direction matrix is for one gene, and there should be at least two columns
(with the 1st column containing gene entrez ID, and 2nd column containing directionality information).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Single sample directional gene set analysis using individual weighted scores inherits the standard gene set variation
analysis(GSVA) method, but also provides the option to use summary statistics from any analysis (disease vs healthy, lesional side vs nonlesional side, etc..)
input to define the direction of gene sets used for directional gene set score calculation for a given disease. This function is
specific for using individual weighted scores.
</p>


<h3>Value</h3>

<p>Matrix of directional gene set scores with rows corresponding to gene sets and columns corresponding to different
samples will be return.
</p>


<h3>See Also</h3>

<p>ssdGSA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ssdGSA_individual(Data = data_matrix_entrezID,
                  Gene_sets = gene_sets[c(1,2,4)],
                  Direction_matrix = direction_matrix
                  )


</code></pre>

<hr>
<h2 id='ssGSA'>Function to Calculate Single Sample Gene Set Scores without Direction Matrix</h2><span id='topic+ssGSA'></span>

<h3>Description</h3>

<p>This function is to calculate traditional single sample gene set scores without considering the direction
of each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssGSA(
  Data,
  Gene_sets,
  GSA_weight = "equal_weighted",
  GSA_weighted_by = "sum.ES",
  GSA_method = "gsva",
  min.sz = 1,
  max.sz = 2000,
  mx.diff = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ssGSA_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ID as row names and columns corresponding to different samples.</p>
</td></tr>
<tr><td><code id="ssGSA_+3A_gene_sets">Gene_sets</code></td>
<td>
<p>A list of gene sets with gene set names as component names, and each component is a vector of gene ID.</p>
</td></tr>
<tr><td><code id="ssGSA_+3A_gsa_weight">GSA_weight</code></td>
<td>
<p>Method to calculate weight in GSA. By default this is set to &quot;group_weighted&quot;. Other option is &quot;equal_weighted&quot;.</p>
</td></tr>
<tr><td><code id="ssGSA_+3A_gsa_weighted_by">GSA_weighted_by</code></td>
<td>
<p>When &quot;group_weighted&quot; is chosen to calculate GSA_weight, further specifications are need to specify how
group weights are calculated. By default this is set to &quot;avg.ES&quot; (average of group ES). Other options are &quot;sum.ES&quot; (sum of group ES)
and &quot;median.ES&quot; (median of group ES).</p>
</td></tr>
<tr><td><code id="ssGSA_+3A_gsa_method">GSA_method</code></td>
<td>
<p>Method to employ in the estimation of gene-set enrichment scores per sample. By default this is set to &quot;gsva&quot;
(Hanzelmann et al, 2013). Other options are &quot;ssgsea&quot; (Barbie et al, 2009), &quot;zscore&quot; (Lee et al, 2008), &quot;avg.exprs&quot; (average value
of gene expressions in the gene set), and &quot;median.exprs&quot; (median of gene expressions in the gene set).</p>
</td></tr>
<tr><td><code id="ssGSA_+3A_min.sz">min.sz</code></td>
<td>
<p>GSVA parameter to define the minimum size of the resulting gene sets. By default this is set to 1.</p>
</td></tr>
<tr><td><code id="ssGSA_+3A_max.sz">max.sz</code></td>
<td>
<p>GSVA parameter to define the maximum size of the resulting gene sets. By default this is set to 2000.</p>
</td></tr>
<tr><td><code id="ssGSA_+3A_mx.diff">mx.diff</code></td>
<td>
<p>GSVA parameter to offer two approaches to calculate the enrichment statistic from the KS random walk statistic.
mx.diff = FALSE: enrichment statistic is calculated as the maximum distance of the random walk from 0. mx.diff=TRUE (default):
enrichment statistic is calculated as the magnitude difference between the largest positive and negative random walk deviations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Single sample directional gene set analysis inherits the standard gene set variation analysis(GSVA) method, but also provides
the option to use summary statistics from any analysis (disease vs healthy, LS vs NL, etc..) input to define the direction of gene
sets used for directional gene set score calculation for a given disease or directional function. However, when the directionality
information is missing for genes, gene set scores from traditional single sample gene set analysis will be returned.
</p>


<h3>Value</h3>

<p>Matrix of gene set scores (without considering directionality information of each gene) with rows corresponding to gene sets
and columns corresponding to different samples will be return.
</p>


<h3>References</h3>

<p>Xingpeng Li, Qi Qian. ssdGSA - Single sample direction gene set analysis tool.
</p>
<p>Barbie, D.A. et al. Systematic RNA interference reveals that oncogenic KRAS-driven cancers require TBK1. Nature, 462(5):108-112, 2009.
</p>
<p>Hanzelmann, S., Castelo, R. and Guinney, J. GSVA: Gene set variation analysis for microarray and RNA-Seq data. BMC Bioinformatics, 14:7, 2013.
</p>
<p>Lee, E. et al. Inferring pathway activity toward precise disease classification. PLoS Comp Biol, 4(11):e1000217, 2008.
</p>
<p>Tomfohr, J. et al. Pathway level analysis of gene expression using singular value decomposition. BMC Bioinformatics, 6:225, 2005.
</p>


<h3>See Also</h3>

<p>ssdGSA, ssdGSA_individual
</p>

<hr>
<h2 id='transform_ensembl_2_entrez'>Function to Transform ENSEMBL ID to ENTREZ ID</h2><span id='topic+transform_ensembl_2_entrez'></span>

<h3>Description</h3>

<p>This function is to uniform gene ID types in data matrices, i.e., from ENSEMBL ID to ENTREZ ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transform_ensembl_2_entrez(Data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="transform_ensembl_2_entrez_+3A_data">Data</code></td>
<td>
<p>Data matrix of gene expressions with gene ensembl ID as row names and columns corresponding to different samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since gene IDs in data matrices from different sources may be in different formats (ensembl ID or entrez ID),
this function is to transform the gene IDs in the data matrix from ensembl ID to entrez ID, to assist the following
single sample directional gene set analysis.
</p>


<h3>Value</h3>

<p>Data matrix of gene expressions with ENSEMBL ID as row names and columns corresponding to samples will be return.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>transform_ensembl_2_entrez(Data = data_matrix)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
