<!DOCTYPE html><html lang="en"><head><title>Help for package SSHAARP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SSHAARP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#checkAlleleSyntax'><p>Check allele syntax</p></a></li>
<li><a href='#checkHaplotypeSyntax'><p>Check haplotype syntax</p></a></li>
<li><a href='#checkLocusANHIG'><p>Check locus validity</p></a></li>
<li><a href='#checkLocusDataset'><p>Check locus validity and if the locus is present in the user specified dataset</p></a></li>
<li><a href='#checkMotifSyntax'><p>Check motif syntax</p></a></li>
<li><a href='#checkPosition'><p>Checks if amino acid positions in motif exist</p></a></li>
<li><a href='#dataSubset'><p>Dataset manipulation for motifs and alleles</p></a></li>
<li><a href='#dataSubsetHaplo'><p>Dataset manipulation for haplotypes</p></a></li>
<li><a href='#findMotif'><p>Returns an alignment data frame of alleles that share a specific amino acid motif</p></a></li>
<li><a href='#getVariantInfo'><p>Locus and allele or motif extraction for motifs, allele, or haplotype mapping</p></a></li>
<li><a href='#isNamePresent'><p>Checks if name portion of entered variant is present</p></a></li>
<li><a href='#mock_haplotype_dataset'><p>Mock Haplotype Dataset</p></a></li>
<li><a href='#PALM'><p>Population Allele Locating Mapmaker</p></a></li>
<li><a href='#readFilename'><p>Designates dataset by either reading in file the user has provided, or using the Solberg dataset or mock haplotype dataset</p></a></li>
<li><a href='#solberg_dataset'><p>Solberg Dataset</p></a></li>
<li><a href='#verifyAlleleANHIG'><p>Verifies the allele entered is present in the IMGT protein alignments</p></a></li>
<li><a href='#verifyAlleleANHIGHaplo'><p>Verifies the alleles in entered haplotype are present in IMGTprotalignments</p></a></li>
<li><a href='#verifyAlleleDataset'><p>Verifies the allele entered is present in specified dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>2.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-12-10</td>
</tr>
<tr>
<td>Title:</td>
<td>Searching Shared HLA Amino Acid Residue Prevalence</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Livia Tran &lt;livia.tran@ucsf.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>Description:</td>
<td>Processes amino acid alignments produced by the 'IPD-IMGT/HLA (Immuno Polymorphism-ImMunoGeneTics/Human Leukocyte Antigen) Database' to identify user-defined amino acid residue motifs shared across HLA alleles, HLA alleles, or HLA haplotypes, and calculates frequencies based on HLA allele frequency data. 'SSHAARP' (Searching Shared HLA Amino Acid Residue Prevalence) uses 'Generic Mapping Tools (GMT)' software and the 'GMT' R package to generate global frequency heat maps that illustrate the distribution of each user-defined map around the globe. 'SSHAARP' analyzes the allele frequency data described by Solberg et al. (2008) &lt;<a href="https://doi.org/10.1016%2Fj.humimm.2008.05.001">doi:10.1016/j.humimm.2008.05.001</a>&gt;, a global set of 497 population samples from 185 published datasets, representing 66,800 individuals total. Users may also specify their own datasets, but file conventions must follow the prebundled Solberg dataset, or the mock haplotype dataset.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, stringr, gtools, BIGDAWG, gmt, DescTools, dplyr,
utils, filesstrings, purrr, stringi, HLAtools</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GMT (5 or 6), Ghostscript (&gt;=9.6)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-10 06:34:11 UTC; tran</td>
</tr>
<tr>
<td>Author:</td>
<td>Livia Tran [aut, cre],
  Steven Mack [aut],
  Josh Bredeweg [ctb],
  Dale Steinhardt [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-11 07:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='checkAlleleSyntax'>Check allele syntax</h2><span id='topic+checkAlleleSyntax'></span>

<h3>Description</h3>

<p>Checks if allele syntax is valid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkAlleleSyntax(allele, filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkAlleleSyntax_+3A_allele">allele</code></td>
<td>
<p>An allele name written in the IPD-IMGT/HLA Database format.</p>
</td></tr>
<tr><td><code id="checkAlleleSyntax_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the Solberg dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if allele syntax is correct. Otherwise,a vector containing FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='checkHaplotypeSyntax'>Check haplotype syntax</h2><span id='topic+checkHaplotypeSyntax'></span>

<h3>Description</h3>

<p>Checks if alleles in a haplotype have correct syntax and an appropriate number of fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkHaplotypeSyntax(haplotype, filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkHaplotypeSyntax_+3A_haplotype">haplotype</code></td>
<td>
<p>A haplotype where allele names are written in the IPD-IMGT/HLA Database format, and have 1-4 fields. Alleles in haplotypes may be delimited by &quot;-&quot; or &quot;~&quot;.</p>
</td></tr>
<tr><td><code id="checkHaplotypeSyntax_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or pre-bundled mock haplotype dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the mock haplotype dataset bundled with the package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if all alleles in entered haplotype have correct syntax and appropriate number of fields. Otherwise, a vector containing FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='checkLocusANHIG'>Check locus validity</h2><span id='topic+checkLocusANHIG'></span>

<h3>Description</h3>

<p>Checks if the locus in the entered variant is a protein-coding gene annotated by the IPD-IMGT/HLA Database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkLocusANHIG(variant)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkLocusANHIG_+3A_variant">variant</code></td>
<td>
<p>An amino acid motif or allele with an HLA locus name followed by an asterisk. This function ONLY evaluates if the locus in the entered variant is a protein-coding gene.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if locus in entered variant are in the IPD-IMGT/HLA Database. Otherwise, a vector with FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example of valid locus in a motif
checkLocusANHIG("DRB1*26F~28E")
#[1] TRUE

#Example of an invalid locus in an allele
checkLocusANHIG("BOO*01:01")
#[1] "FALSE"                     "BOO is not a valid locus."
</code></pre>

<hr>
<h2 id='checkLocusDataset'>Check locus validity and if the locus is present in the user specified dataset</h2><span id='topic+checkLocusDataset'></span>

<h3>Description</h3>

<p>Checks if the locus in the entered variant is a protein-coding gene annotated by the IPD-IMGT/HLA Database, and if it is in the user specified dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkLocusDataset(variant, filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkLocusDataset_+3A_variant">variant</code></td>
<td>
<p>An allele or an amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids. Haplotypes must contain alleles that follow the aforementioned format, and may be delimited by &quot;~&quot; or &quot;-&quot;.</p>
</td></tr>
<tr><td><code id="checkLocusDataset_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset or mock haplotype dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the datasets bundled with the package. Allele and motif datasets should follow the Solberg dataset format, and haplotype datasets should follow the SSHAARP haplotype mock data format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if locus is a protein-coding gene and is in the specified dataset. Otherwise, a vector with FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='checkMotifSyntax'>Check motif syntax</h2><span id='topic+checkMotifSyntax'></span>

<h3>Description</h3>

<p>Checks if motif syntax is valid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkMotifSyntax(motif, filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkMotifSyntax_+3A_motif">motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td></tr>
<tr><td><code id="checkMotifSyntax_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the Solberg dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the motif syntax is valid. Otherwise, a vector containing FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Example with correct motif syntax where user specified dataset is the Solberg dataset
checkMotifSyntax("DRB1*26F~28E~30Y", filename=SSHAARP::solberg_dataset)

#Example with incorrect motif syntax where user specified dataset is the Solberg dataset
checkMotifSyntax("DRB1****26F~28E", filename=SSHAARP::solberg_dataset)
</code></pre>

<hr>
<h2 id='checkPosition'>Checks if amino acid positions in motif exist</h2><span id='topic+checkPosition'></span>

<h3>Description</h3>

<p>Checks if amino acid positions in the entered motif exist in IMGTprotalignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkPosition(motif, filename, alignments)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkPosition_+3A_motif">motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids. This function ONLY checks if the entered amino acid positions exist in IMGTprotalignments.</p>
</td></tr>
<tr><td><code id="checkPosition_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the Solberg dataset.</p>
</td></tr>
<tr><td><code id="checkPosition_+3A_alignments">alignments</code></td>
<td>
<p>A list object of sub-lists of data frames of protein alignments for the HLA and HLA-region genes supported in the ANHIG/IMGTHLA GitHub Repository. Alignments will always be the most recent version IPD-IMGT/HLA Database version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if all of the amino acid positions in a motif exist. Otherwise, a vector with FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='dataSubset'>Dataset manipulation for motifs and alleles</h2><span id='topic+dataSubset'></span>

<h3>Description</h3>

<p>Returns a modified version of the user selected dataset that includes a column of locus*allele names, is sorted by by population name, and is reduced to the specified locus. Cardinal coordinates are converted to their Cartesian equivalents (i.e. 50S is converted to -50).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataSubset(variant, filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dataSubset_+3A_variant">variant</code></td>
<td>
<p>An allele or an amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td></tr>
<tr><td><code id="dataSubset_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the Solberg dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing a reformatted version of the user selected dataset, with rows ordered by population name, Cartesian coordinates in the latit and longit columns, and limited to populations with data for the specified locus. Otherwise, a vector containing FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>
<p>The Solberg dataset is the tab-delimited ‘1-locus-alleles.dat’ text file in the results.zip archive at http://pypop.org/popdata/.
</p>
<p>The Solberg dataset is also prepackaged into SSHAARP as 'solberg_dataset'.
</p>

<hr>
<h2 id='dataSubsetHaplo'>Dataset manipulation for haplotypes</h2><span id='topic+dataSubsetHaplo'></span>

<h3>Description</h3>

<p>Returns the user input dataset that contains the selected haplotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataSubsetHaplo(haplotype, filename, AFND, alignments)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dataSubsetHaplo_+3A_haplotype">haplotype</code></td>
<td>
<p>A haplotype where allele names are written in the IPD-IMGT/HLA Database format, and have 1-4 fields. Alleles in haplotypes may be delimited by &quot;-&quot; or &quot;~&quot;.</p>
</td></tr>
<tr><td><code id="dataSubsetHaplo_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or pre-bundled mock haplotype dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the mock haplotype dataset bundled with the package.</p>
</td></tr>
<tr><td><code id="dataSubsetHaplo_+3A_afnd">AFND</code></td>
<td>
<p>A logical parameter that determines whether the user specified dataset is data from AFND. This parameter is only relevant if haplotype maps are being made.</p>
</td></tr>
<tr><td><code id="dataSubsetHaplo_+3A_alignments">alignments</code></td>
<td>
<p>A list object of sub-lists of data frames of protein alignments for the HLA and HLA-region genes supported in the ANHIG/IMGTHLA GitHub Repository. Alignments will always be the most recent version IPD-IMGT/HLA Database version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two element list with 1) a subset data frame containing only haplotypes with alleles present in the user input haplotype, and 2) a data frame of the full dataset. Alleles with two fields will be evaluated with their three and four field allele equivalents, and alleles with three fields will be evaluated with their four field allele equivalent. Otherwise, a vector containing FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='findMotif'>Returns an alignment data frame of alleles that share a specific amino acid motif</h2><span id='topic+findMotif'></span>

<h3>Description</h3>

<p>Returns an alignment data frame of alleles that share a specific amino acid motif.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findMotif(motif, filename, alignments)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findMotif_+3A_motif">motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td></tr>
<tr><td><code id="findMotif_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the Solberg dataset.</p>
</td></tr>
<tr><td><code id="findMotif_+3A_alignments">alignments</code></td>
<td>
<p>A list object of sub-lists of data frames of protein alignments for the HLA and HLA-region genes supported in the ANHIG/IMGTHLA GitHub Repository. Alignments will always be the most recent version IPD-IMGT/HLA Database version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An amino acid alignment dataframe of alleles that share the specified motif. Otherwise, a vector containing FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='getVariantInfo'>Locus and allele or motif extraction for motifs, allele, or haplotype mapping</h2><span id='topic+getVariantInfo'></span>

<h3>Description</h3>

<p>Extracts locus and allele or motif information from variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getVariantInfo(variant)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getVariantInfo_+3A_variant">variant</code></td>
<td>
<p>An amino acid motif or allele. Amino acid motifs must be in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Alleles must have 1-4 fields.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with loci and allele or motif information.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='isNamePresent'>Checks if name portion of entered variant is present</h2><span id='topic+isNamePresent'></span>

<h3>Description</h3>

<p>Checks if the name portion of the entered variant is present. Names consist of information following the locus and asterisk of the entered variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isNamePresent(variant, variantType)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isNamePresent_+3A_variant">variant</code></td>
<td>
<p>An amino acid motif or allele. Amino acid motifs must be in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Alleles must have 1-4 fields.</p>
</td></tr>
<tr><td><code id="isNamePresent_+3A_varianttype">variantType</code></td>
<td>
<p>Identifies whether the variant is an allele or motif.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if name is present. Otherwise, a vector with FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='mock_haplotype_dataset'>Mock Haplotype Dataset</h2><span id='topic+mock_haplotype_dataset'></span>

<h3>Description</h3>

<p>A dataframe containing mock haplotype data modeled after the Allele Frequency Network Database (AFND) haplotype data structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mock_haplotype_dataset
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 24989 rows and 7 columns.
</p>

<hr>
<h2 id='PALM'>Population Allele Locating Mapmaker</h2><span id='topic+PALM'></span>

<h3>Description</h3>

<p>Produces a frequency heatmap for a specified allele, amino-acid motif, or haplotype based on the allele frequency data in the Solberg dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PALM(
  variant,
  variantType,
  filename,
  mask = FALSE,
  color = TRUE,
  filterMigrant = TRUE,
  mapScale = TRUE,
  direct = getwd(),
  AFND = TRUE,
  generateLowFreq = TRUE,
  resolution = 500
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PALM_+3A_variant">variant</code></td>
<td>
<p>An allele or an amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids. Haplotypes must contain alleles that follow the aforementioned format, and may be delimited by &quot;~&quot; or &quot;-&quot;.</p>
</td></tr>
<tr><td><code id="PALM_+3A_varianttype">variantType</code></td>
<td>
<p>Specifies whether the variant is an allele, motif, or haplotype.</p>
</td></tr>
<tr><td><code id="PALM_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset or mock haplotype dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the datasets bundled with the package. Allele and motif datasets should follow the Solberg dataset format, and haplotype datasets should follow the SSHAARP haplotype mock data format.</p>
</td></tr>
<tr><td><code id="PALM_+3A_mask">mask</code></td>
<td>
<p>A logical parameter that determines if areas with little to no population coverage should be masked. The default value is set to FALSE.</p>
</td></tr>
<tr><td><code id="PALM_+3A_color">color</code></td>
<td>
<p>A logical parameter that identifies if the heat maps should be made in color (TRUE) or gray scale (FALSE). The default value is TRUE.</p>
</td></tr>
<tr><td><code id="PALM_+3A_filtermigrant">filterMigrant</code></td>
<td>
<p>A logical parameter that determines if admixed populations (OTH) and migrant populations (i.e. any complexities with the 'mig') should be excluded from the dataset. The default value is TRUE.</p>
</td></tr>
<tr><td><code id="PALM_+3A_mapscale">mapScale</code></td>
<td>
<p>A logical parameter that determines if the max frequency value of the map scale should be 1 (FALSE), or if it should represent the maximum frequency of the chosen motif, allele, or haplotype (TRUE). The default value is TRUE.</p>
</td></tr>
<tr><td><code id="PALM_+3A_direct">direct</code></td>
<td>
<p>The directory into which the map produced is written. The default directory is the user's working directory.</p>
</td></tr>
<tr><td><code id="PALM_+3A_afnd">AFND</code></td>
<td>
<p>A logical parameter that determines whether the user specified dataset is data from AFND. This parameter is only relevant if haplotype maps are being made. Default is TRUE.</p>
</td></tr>
<tr><td><code id="PALM_+3A_generatelowfreq">generateLowFreq</code></td>
<td>
<p>A logical parameter that determines whether maps should be generated for a variant if the maximum frequency for the variant is low frequency. Low frequency populations are defined as those with a frequency of 0.000, indicating three zeros after the decimal point.</p>
</td></tr>
<tr><td><code id="PALM_+3A_resolution">resolution</code></td>
<td>
<p>An integer for raster resolution in dpi for the final map output. It is not recommended to go below 400. Default is set to 500.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The specified motif and the directory into which the heat map was written are returned in an invisible character vector. Otherwise, a warning message is returned.
</p>


<h3>Note</h3>

<p>IMGT protein alignments will be generated for the locus of the specified variant the first time PALM is executed for a given locus. The alignments will be saved to the temp directory and referenced by PALM. PALM checks if the locus specific alignment is present in the temp directory; if it is not, a protein alignment object will be built for the locus. Restarting the R session will remove existing alignments.
</p>
<p>The produced frequency heatmap is generated by using the Generic Mapping Tools (GMT) R Package, which is an interface between R and the GMT map making software.
</p>
<p>The Solberg dataset is the tab-delimited ‘1-locus-alleles.dat’ text file in the results.zip archive at http://pypop.org/popdata/.
</p>
<p>The Solberg dataset is also prepackaged into SSHAARP as 'solberg_dataset'.
</p>
<p>A mock haplotype dataset modeled after the AFND network's haplotype dataset structure is available for usage under &quot;mock_haplotype_dataset&quot;.
</p>
<p>While the map legend identifies the highest frequency value, values in this range may not be represented on the map due to frequency averaging over neighboring populations.
</p>


<h3>References</h3>

<p>Solberg et.al. (2008) &lt;doi: 10.1016/j.humimm.2008.05.001&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example to produce a motif color map where migrant populations are filtered out, mask is off
## Not run: PALM("DRB1*26F~28E~30Y",
variantType="motif",
mask = FALSE,
filterMigrant=TRUE,
filename = SSHAARP::solberg_dataset)
## End(Not run)

#Example to produce an allele greyscale map where migrant populations are not filtered, mask is on
## Not run: PALM("DRB1*01:01",
variantType="allele",
mask = TRUE, color=FALSE,
filterMigrant=FALSE,
filename = SSHAARP::solberg_dataset)
## End(Not run)

#Example to produce a color allele map with mapScale T and the allele has more than 2 fields
## Not run: PALM("DRB1*01:01:01",
variantType="allele",
filterMigrant=FALSE,
mapScale=TRUE,
filename = SSHAARP::solberg_dataset)
## End(Not run)

#Example to produce a color haplotype map with default parameters with the mock haplotype dataset
## Not run: PALM("DRB1*01:01~A*01:01",
variantType = "haplotype",
filename = SSHAARP::mock_haplotype_dataset)
## End(Not run)

</code></pre>

<hr>
<h2 id='readFilename'>Designates dataset by either reading in file the user has provided, or using the Solberg dataset or mock haplotype dataset</h2><span id='topic+readFilename'></span>

<h3>Description</h3>

<p>Returns the user specified dataset, either by reading in the file the user has provided, or by using the Solberg dataset or mock haplotype dataset. If the user provides a dataset and the filename is not found, an error will be returned. If the user provided dataset does not have the same number of columns or the same column names as the reference datasets, an error message will be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readFilename(filename, variant)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readFilename_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset or mock haplotype dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the datasets bundled with the package. Allele and motif datasets should follow the Solberg dataset format, and haplotype datasets should follow the SSHAARP haplotype mock data format.</p>
</td></tr>
<tr><td><code id="readFilename_+3A_variant">variant</code></td>
<td>
<p>An allele or an amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids. Haplotypes must contain alleles that follow the aforementioned format, and may be delimited by &quot;~&quot; or &quot;-&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of the user specified dataset.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='solberg_dataset'>Solberg Dataset</h2><span id='topic+solberg_dataset'></span>

<h3>Description</h3>

<p>A dataframe of the original Solberg dataset, which is a global dataset of 497 population samples from 185 published datasets, representing 66,800 individuals.
For more information on the Solberg dataset, please see the vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solberg_dataset
</code></pre>


<h3>Format</h3>

<p>A dataframe with 20163 rows and 13 columns.
</p>


<h3>Source</h3>

<p>results.zip file from http://pypop.org/popdata/
</p>


<h3>References</h3>

<p>Solberg et.al &quot;Balancing selection and heterogeneity across the classical human leukocyte antigen loci: A meta-analytic review of 497 population studies&quot;. Human Immunology (2008) 69, 443–464
</p>

<hr>
<h2 id='verifyAlleleANHIG'>Verifies the allele entered is present in the IMGT protein alignments</h2><span id='topic+verifyAlleleANHIG'></span>

<h3>Description</h3>

<p>Verifies the allele entered is present in IMGT protein alignments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verifyAlleleANHIG(allele, filename, alignments)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verifyAlleleANHIG_+3A_allele">allele</code></td>
<td>
<p>An allele name written in the IPD-IMGT/HLA Database format.</p>
</td></tr>
<tr><td><code id="verifyAlleleANHIG_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the Solberg dataset.</p>
</td></tr>
<tr><td><code id="verifyAlleleANHIG_+3A_alignments">alignments</code></td>
<td>
<p>A list object of sub-lists of data frames of protein alignments for the HLA and HLA-region genes supported in the ANHIG/IMGTHLA GitHub Repository. Alignments will always be the most recent version IPD-IMGT/HLA Database version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if allele is present in the IMGTprotalignment object. Otherwise, a vector containing FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='verifyAlleleANHIGHaplo'>Verifies the alleles in entered haplotype are present in IMGTprotalignments</h2><span id='topic+verifyAlleleANHIGHaplo'></span>

<h3>Description</h3>

<p>Verifies the alleles in entered haplotype are present in IMGTprotalignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verifyAlleleANHIGHaplo(haplotype, filename, alignments)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verifyAlleleANHIGHaplo_+3A_haplotype">haplotype</code></td>
<td>
<p>A haplotype where allele names are written in the IPD-IMGT/HLA Database format, and have 1-4 fields. Alleles in haplotypes may be delimited by &quot;-&quot; or &quot;~&quot;.</p>
</td></tr>
<tr><td><code id="verifyAlleleANHIGHaplo_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or pre-bundled mock haplotype dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the mock haplotype dataset bundled with the package.</p>
</td></tr>
<tr><td><code id="verifyAlleleANHIGHaplo_+3A_alignments">alignments</code></td>
<td>
<p>A list object of sub-lists of data frames of protein alignments for the HLA and HLA-region genes supported in the ANHIG/IMGTHLA GitHub Repository. Alignments will always be the most recent version IPD-IMGT/HLA Database version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if all alleles in a haplotype are present in the IMGTprotalignment object. Otherwise, a vector containing FALSE and an error message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

<hr>
<h2 id='verifyAlleleDataset'>Verifies the allele entered is present in specified dataset</h2><span id='topic+verifyAlleleDataset'></span>

<h3>Description</h3>

<p>Verifies the allele entered is present in the specified dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verifyAlleleDataset(allele, filename, alignments)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verifyAlleleDataset_+3A_allele">allele</code></td>
<td>
<p>An allele name written in the IPD-IMGT/HLA Database format.</p>
</td></tr>
<tr><td><code id="verifyAlleleDataset_+3A_filename">filename</code></td>
<td>
<p>The full file path of the user specified dataset if the user wishes to use their own file, or the pre-bundled Solberg dataset. User provided datasets must be a .dat, .txt, or.csv file, and must conform to the structure and format of the Solberg dataset.</p>
</td></tr>
<tr><td><code id="verifyAlleleDataset_+3A_alignments">alignments</code></td>
<td>
<p>A list object of sub-lists of data frames of protein alignments for the HLA and HLA-region genes supported in the ANHIG/IMGTHLA GitHub Repository. Alignments will always be the most recent version IPD-IMGT/HLA Database version.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the allele is present in the specified data set, and the filtered allele dataset. If a user enters an allele with more than two fields and has selected the Solberg dataset as the data source, a message informing the user that the allele has been truncated is appended to the output. If an allele entered is valid, but is not present in the user provided dataset, a warning message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
