<!DOCTYPE html><html><head><title>Help for package SSHAARP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SSHAARP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AA_atlas'><p>Exon boundaries for all exons in protein coding genes in the IPD-IMGT/HLA Database release v 3.39.0</p></a></li>
<li><a href='#BLAASD'><p>BLAASD - Build Loci Amino Acid Specific Dataframe</p></a></li>
<li><a href='#checkMotif'><p>Syntactic and semantic validation of HLA amino acid motifs</p></a></li>
<li><a href='#dataSubset'><p>Solberg dataset manipulation</p></a></li>
<li><a href='#findMotif'><p>Returns an alignment data frame of alleles that share a specific amino acid motif</p></a></li>
<li><a href='#IMGTprotalignments'><p>Protein alignments for all protein coding genes in the IPD-IMGT/HLA Database release v 3.39.0.</p></a></li>
<li><a href='#PALM'><p>Population Allele Locating Mapmaker</p></a></li>
<li><a href='#solberg_dataset'><p>Solberg Dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Searching Shared HLA Amino Acid Residue Prevalence</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Livia Tran &lt;livia.tran@ucsf.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Description:</td>
<td>Processes amino acid alignments produced by the 'IPD-IMGT/HLA (Immuno Polymorphism-ImMunoGeneTics/Human Leukocyte Antigen) Database' to identify user-defined amino acid residue motifs shared across HLA alleles, and calculates the frequencies of those motifs based on HLA allele frequency data. 'SSHAARP' (Searching Shared HLA Amino Acid Residue Prevalence) uses 'Generic Mapping Tools (GMT)' software and the 'GMT' R package to generate global frequency heat maps that illustrate the distribution of each user-defined map around the globe. 'SSHAARP' analyzes the allele frequency data described by Solberg et al. (2008) &lt;<a href="https://doi.org/10.1016%2Fj.humimm.2008.05.001">doi:10.1016/j.humimm.2008.05.001</a>&gt;, a global set of 497 population samples from 185 published datasets, representing 66,800 individuals total.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, stringr, gtools, BIGDAWG, gmt, DescTools, dplyr,
utils, filesstrings</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GMT (5 or 6), Ghostscript (&gt;=9.6)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-16 23:18:52 UTC; tran.livia</td>
</tr>
<tr>
<td>Author:</td>
<td>Livia Tran [aut, cre],
  Steven Mack [aut],
  Josh Bredeweg [ctb],
  Dale Steinhardt [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-17 08:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='AA_atlas'>Exon boundaries for all exons in protein coding genes in the IPD-IMGT/HLA Database release v 3.39.0</h2><span id='topic+AA_atlas'></span>

<h3>Description</h3>

<p>A list object containing exon boundaries for all exons in protein coding genes in the IPD-IMGT/HLA Database release v 3.39.0. Exon boundaries were determined from the nucleotide alignment files, which were downloaded from the ANHIG/IMGTHLA Github respository.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AA_atlas
</code></pre>


<h3>Format</h3>

<p>A list containing exon boundaries in a dataframe format for each locus.
</p>


<h3>Note</h3>

<p>For internal use only.
</p>


<h3>Source</h3>

<p>https://github.com/ANHIG/IMGTHLA/tree/Latest/alignments
</p>

<hr>
<h2 id='BLAASD'>BLAASD - Build Loci Amino Acid Specific Dataframe</h2><span id='topic+BLAASD'></span>

<h3>Description</h3>

<p>Extracts alignment sequence information for a given locus from the ANHIG/IMGTHLA GitHub repository to produce a dataframe of individual amino acid data for each amino acid position for all alleles, for a user-defined HLA locus or loci. The first 4 columns are locus, allele, trimmed allele, and allele_name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BLAASD(loci)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BLAASD_+3A_loci">loci</code></td>
<td>
<p>A vector of un-prefixed HLA locus names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object of data frames for each specified locus. Each list element is a data frame of allele names and the corresponding peptide sequence for each amino acid position. An error message is return if the loci input is not a locus for which petpide alignments are available in the ANHIG/IMGTHLA Github Repository.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#BLAASD with one locus as input
BLAASD("C")

#BLAASD with multiple loci as input
BLAASD(c("A", "B", "C"))
</code></pre>

<hr>
<h2 id='checkMotif'>Syntactic and semantic validation of HLA amino acid motifs</h2><span id='topic+checkMotif'></span>

<h3>Description</h3>

<p>Checks input motif for errors in format and amino acid positions not present in the locus alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkMotif(motif)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkMotif_+3A_motif">motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A warning message if the input motif is formatted incorrectly, or contains an amino acid position not present in the alignment. Otherwise, a list object with extracted locus information, a correctly formatted motif, and locus specific amino acid dataframe are returned. Note checkMotif() does not check amino acid variants in a specified motif; that is done by findMotif().
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example where a motif is formatted correctly
checkMotif("DRB1*26F~28E~30Y")

#Example where format is incorrect
checkMotif("DRB1**26F~28E~30Y")

#Example where an amino acid position does not exist
checkMotif("DRB1**26F~28E~300000Y")
</code></pre>

<hr>
<h2 id='dataSubset'>Solberg dataset manipulation</h2><span id='topic+dataSubset'></span>

<h3>Description</h3>

<p>Returns a modified version of the Solberg dataset that includes a column of locus*allele names, is sorted by by population name, and is reduced to the specified locus. Cardinal coordinates are converted to their Cartesian equivalents (i.e. 50S is converted to -50).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataSubset(motif, filename = SSHAARP::solberg_dataset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dataSubset_+3A_motif">motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td></tr>
<tr><td><code id="dataSubset_+3A_filename">filename</code></td>
<td>
<p>The filename of the local copy of the Solberg dataset - the defaulted filename is the solberg_dataset in the SSHAARP package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing a reformatted version of the Solberg dataset, with rows ordered by population name, Cartesian coordinates in the latit and longit columns, and limited to populations with data for the specified locus. If a motif has formatting errors, a warning message is returned.
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>
<p>The Solberg dataset is the tab-delimited ‘1-locus-alleles.dat’ text file in the results.zip archive at http://pypop.org/popdata/.
</p>
<p>The Solberg dataset is also prepackaged into SSHAARP as 'solberg_dataset'.
</p>

<hr>
<h2 id='findMotif'>Returns an alignment data frame of alleles that share a specific amino acid motif</h2><span id='topic+findMotif'></span>

<h3>Description</h3>

<p>Consumes the alignment data frame produced by BLAASD() and returns an alignment data frame of alleles that share a specific amino acid motif.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findMotif(motif)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findMotif_+3A_motif">motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An amino acid alignment dataframe of alleles that share the specified motif. If the motif is not found in any alleles, or the motif has formatting errors, a warning message is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#example with actual motif
findMotif("DRB1*26F~28E~30Y")
("DRB1*26F~28E")

#example with non-existent motif
findMotif("DRB1*26F~28E~30Z")

#extracting names of alleles with user-defined motif
findMotif("DRB1*26F~28E~30Y")[,4]
</code></pre>

<hr>
<h2 id='IMGTprotalignments'>Protein alignments for all protein coding genes in the IPD-IMGT/HLA Database release v 3.39.0.</h2><span id='topic+IMGTprotalignments'></span>

<h3>Description</h3>

<p>A list object containing protein alignments for all protein coding genes in the IPD-IMGT/HLA Database release. Alignments were downloaded from the ANHIG/IMGTHLA Github respository.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IMGTprotalignments
</code></pre>


<h3>Format</h3>

<p>A list containing protein alignments in a dataframe format for each locus.
</p>


<h3>Source</h3>

<p>https://github.com/ANHIG/IMGTHLA/tree/Latest/alignments
</p>

<hr>
<h2 id='PALM'>Population Allele Locating Mapmaker</h2><span id='topic+PALM'></span>

<h3>Description</h3>

<p>Produces a frequency heatmap for a specified amino-acid motif, based on the allele frequency data in the Solberg dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PALM(
  motif,
  filename = SSHAARP::solberg_dataset,
  direct = getwd(),
  color = TRUE,
  filterMigrant = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PALM_+3A_motif">motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td></tr>
<tr><td><code id="PALM_+3A_filename">filename</code></td>
<td>
<p>The filename of the local copy of the Solberg dataset - the defaulted filename is the solberg_dataset in the SSHAARP package.</p>
</td></tr>
<tr><td><code id="PALM_+3A_direct">direct</code></td>
<td>
<p>The directory into which the map produced is written. The default directory is the user's working directory.</p>
</td></tr>
<tr><td><code id="PALM_+3A_color">color</code></td>
<td>
<p>A logical parameter that identifies if the heat maps should be made in color (TRUE) or gray scale (FALSE). The default option is TRUE.</p>
</td></tr>
<tr><td><code id="PALM_+3A_filtermigrant">filterMigrant</code></td>
<td>
<p>A logical parameter that determines if admixed populations (OTH) and migrant populations (i.e. any complexities with the 'mig') should be excluded from the dataset. The default option is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The specified motif and the directory into which the heatmap was written are returned in an invisible character vector. If the user enters a motif that is not found in the Solberg dataset, or that does not exist, a warning message is returned. If an incorrectly formatted motif is entered, or the user does not have the GMT software installed on their operating system, a vector with a warning message is returned. The produced heatmap is written to the user's specified directory (default is user's working directory) as a .jpg file, where the filename is &quot;'motif'.jpg&quot;.
</p>


<h3>Note</h3>

<p>The produced frequency heatmap is generated by using the Generic Mapping Tools (GMT) R Package, which is an interface between R and the GMT Map-Making software.
</p>
<p>The Solberg dataset is the tab-delimited ‘1-locus-alleles.dat’ text file in the results.zip archive at http://pypop.org/popdata/.
</p>
<p>The Solberg dataset is also prepackaged into SSHAARP as 'solberg_dataset'.
</p>
<p>While the map legend identifies the highest frequency value, values in this range may not be represented on the map due to frequency averaging over neighboring populations.
</p>


<h3>References</h3>

<p>Solberg et.al. (2008) &lt;doi: 10.1016/j.humimm.2008.05.001&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#example to produce a color frequency heat map where migrant populations are filtered out
PALM("DRB1*26F~28E~30Y",filename = solberg_dataset[85:100,], filterMigrant=TRUE)
#example to produce a greyscale heat map where migrant populations are not filtered out
PALM("DRB1*26F~28E~30Y", filename = solberg_dataset[85:100,], color=FALSE, filterMigrant=FALSE)

</code></pre>

<hr>
<h2 id='solberg_dataset'>Solberg Dataset</h2><span id='topic+solberg_dataset'></span>

<h3>Description</h3>

<p>A dataframe of the original Solberg dataset, which is a global dataset of 497 population samples from 185 published datasets, representing 66,800 individuals.
For more information on the Solberg dataset, please see the vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solberg_dataset
</code></pre>


<h3>Format</h3>

<p>A dataframe with 20163 rows and 13 columns.
</p>


<h3>Source</h3>

<p>results.zip file from http://pypop.org/popdata/
</p>


<h3>References</h3>

<p>Solberg et.al &quot;Balancing selection and heterogeneity across the classical human leukocyte antigen loci: A meta-analytic review of 497 population studies&quot;. Human Immunology (2008) 69, 443–464
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
