<!DOCTYPE html><html><head><title>Help for package ssizeRNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ssizeRNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#check.power'><p>Average Power and True FDR Based on limma/voom RNAseq Analysis Pipeline</p></a></li>
<li><a href='#hammer.eset'>
<p>RNA-seq data from Hammer, P. et al., 2010</p></a></li>
<li><a href='#sim.counts'><p>RNA-seq Count Data Simulation from Negative-Binomial Distribution</p></a></li>
<li><a href='#ssize.twoSampVaryDelta'><p>Sample Size Calculations for Two-Sample Microarray Experiments with</p>
Differing Mean Expressions but fixed Standard Deviations Among Genes</a></li>
<li><a href='#ssizeRNA_single'><p>Sample Size Calculations for Two-Sample RNA-seq Experiments</p>
with Single Set of Parameters</a></li>
<li><a href='#ssizeRNA_vary'><p>Sample Size Calculations for Two-Sample RNA-seq Experiments with</p>
Differing Mean and Dispersion Among Genes</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Sample Size Calculation for RNA-Seq Experimental Design</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-08-15</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ran Bi &lt;biranpier@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We propose a procedure for sample size calculation while
    controlling false discovery rate for RNA-seq experimental design. Our
    procedure depends on the Voom method proposed for RNA-seq data analysis
    by Law et al. (2014) &lt;<a href="https://doi.org/10.1186%2Fgb-2014-15-2-r29">doi:10.1186/gb-2014-15-2-r29</a>&gt; and the sample size 
    calculation method proposed for microarray experiments by Liu and Hwang 
    (2007) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtl664">doi:10.1093/bioinformatics/btl664</a>&gt;. We develop a set of functions
    that calculates appropriate sample sizes for two-sample t-test for RNA-seq
    experiments with fixed or varied set of parameters. The outputs also contain a
    plot of power versus sample size, a table of power at different sample sizes,
    and a table of critical test values at different sample sizes. 
    To install this package, please use 
    'source("http://bioconductor.org/biocLite.R"); biocLite("ssizeRNA")'. 
    For R version 3.5 or greater, please use  
    'if(!requireNamespace("BiocManager", quietly = TRUE)){install.packages("BiocManager")}; BiocManager::install("ssizeRNA")'.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, Biobase, edgeR, limma, qvalue, ssize.fdr, graphics,
stats</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>biocViews:</td>
<td>GeneExpression, DifferentialExpression, ExperimentalDesign,
Sequencing, RNASeq, DNASeq, Microarray</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-08-16 06:31:20 UTC; Ran</td>
</tr>
<tr>
<td>Author:</td>
<td>Ran Bi [aut, cre],
  Peng Liu [aut],
  Tim Triche [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-08-20 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='check.power'>Average Power and True FDR Based on limma/voom RNAseq Analysis Pipeline</h2><span id='topic+check.power'></span>

<h3>Description</h3>

<p>For the limma/voom RNAseq analysis pipeline, when we control false discovery
rate by using the Benjamini and Hochberg step-up procedure (1995) 
and/or Storey and Tibshirani's q-value procedure (Storey et al, 2004),
<code>check.power</code> calculates average power and true FDR for given sample 
size, user-specified proportions of non-differentially expressed genes, 
number of iterations, FDR level to control, mean counts in control group, 
dispersion, and fold change.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check.power(nGenes = 10000, pi0 = 0.8, m, mu, disp, fc, up = 0.5,
  replace = TRUE, fdr = 0.05, sims = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check.power_+3A_ngenes">nGenes</code></td>
<td>
<p>total number of genes, the default value is <code>10000</code>.</p>
</td></tr>
<tr><td><code id="check.power_+3A_pi0">pi0</code></td>
<td>
<p>proportion of non-differentially expressed genes, 
the default value is <code>0.8</code>.</p>
</td></tr>
<tr><td><code id="check.power_+3A_m">m</code></td>
<td>
<p>sample size per treatment group.</p>
</td></tr>
<tr><td><code id="check.power_+3A_mu">mu</code></td>
<td>
<p>a vector (or scalar) of mean counts in control group 
from which to simulate.</p>
</td></tr>
<tr><td><code id="check.power_+3A_disp">disp</code></td>
<td>
<p>a vector (or scalar) of dispersion parameter 
from which to simulate.</p>
</td></tr>
<tr><td><code id="check.power_+3A_fc">fc</code></td>
<td>
<p>a vector (or scalar, or a function that takes an integer n 
and generates a vector of length n)
of fold change for differentially expressed (DE) genes.</p>
</td></tr>
<tr><td><code id="check.power_+3A_up">up</code></td>
<td>
<p>proportion of up-regulated genes among all DE genes, 
the default value is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="check.power_+3A_replace">replace</code></td>
<td>
<p>sample with or without replacement from given parameters. 
See Details for more information.</p>
</td></tr>
<tr><td><code id="check.power_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled.</p>
</td></tr>
<tr><td><code id="check.power_+3A_sims">sims</code></td>
<td>
<p>number of simulations to run when computing power and FDR.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>pow_bh_ave</code></td>
<td>
<p>average power when controlling FDR 
by Benjamini and Hochberg (1995) method.</p>
</td></tr>
<tr><td><code>fdr_bh_ave</code></td>
<td>
<p>true false discovery rate when controlling FDR 
by Benjamini and Hochberg (1995) method.</p>
</td></tr>
<tr><td><code>pow_bh_ave</code></td>
<td>
<p>average power when controlling FDR 
by q-value procedure (Storey et al., 2004).</p>
</td></tr>
<tr><td><code>fdr_bh_ave</code></td>
<td>
<p>true false discovery rate when controlling FDR
by q-value procedure (Storey et al., 2004).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>References</h3>

<p>Benjamini, Y. and Hochberg, Y. (1995) 
Controlling the false discovery rate: a practical and 
powerful approach to multiple testing. 
<em>J. R. Stat. Soc. B</em>, 57, 289-300.
</p>
<p>Storey, J. D., Taylor, J. E. and Siegmund, D. (2004)
Strong control, conservative point estimation and 
simultaneous rates: a unified approach. 
<em>J. R. Stat. Soc. B</em>, 66, 187- 205.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(limma)
library(qvalue)
m &lt;- 14                      ## sample size per treatment group
mu &lt;- 10                     ## mean read counts in control group
disp &lt;- 0.1                  ## dispersion for all genes
fc &lt;- 2                      ## 2-fold change for DE genes

check.power(m = m, mu = mu, disp = disp, fc = fc, sims = 2)

</code></pre>

<hr>
<h2 id='hammer.eset'>
RNA-seq data from Hammer, P. et al., 2010
</h2><span id='topic+hammer.eset'></span>

<h3>Description</h3>

<p>RNA-seq data structured as an expressionSet, from &quot;mRNA-seq with 
agnostic splice site discovery for nervous system transcriptomics tested in 
chronic pain&quot; by Hammer, P. et al. (Genome Res. 2010, 20(6):847-860), 
<a href="http://dx.doi.org/10.1101/gr.101204.109">http://dx.doi.org/10.1101/gr.101204.109</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hammer.eset)
</code></pre>


<h3>Value</h3>

<p>RNA-seq data structured as an expressionSet.
</p>


<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>

<hr>
<h2 id='sim.counts'>RNA-seq Count Data Simulation from Negative-Binomial Distribution</h2><span id='topic+sim.counts'></span>

<h3>Description</h3>

<p>This function simulates count data from Negative-Binomial distribution
for two-sample RNA-seq experiments with given mean, dispersion 
and fold change. 
A count data matrix is generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.counts(nGenes = 10000, pi0 = 0.8, m, mu, disp, fc, up = 0.5,
  replace = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.counts_+3A_ngenes">nGenes</code></td>
<td>
<p>total number of genes, the default value is <code>10000</code>.</p>
</td></tr>
<tr><td><code id="sim.counts_+3A_pi0">pi0</code></td>
<td>
<p>proportion of non-differentially expressed genes, 
the default value is <code>0.8</code>.</p>
</td></tr>
<tr><td><code id="sim.counts_+3A_m">m</code></td>
<td>
<p>sample size per treatment group.</p>
</td></tr>
<tr><td><code id="sim.counts_+3A_mu">mu</code></td>
<td>
<p>a vector (or scalar) of mean counts in control group 
from which to simulate.</p>
</td></tr>
<tr><td><code id="sim.counts_+3A_disp">disp</code></td>
<td>
<p>a vector (or scalar) of dispersion parameter 
from which to simulate.</p>
</td></tr>
<tr><td><code id="sim.counts_+3A_fc">fc</code></td>
<td>
<p>a vector (or scalar, or a function that takes an integer n 
and generates a vector of length n)
of fold change for differentially expressed (DE) genes.</p>
</td></tr>
<tr><td><code id="sim.counts_+3A_up">up</code></td>
<td>
<p>proportion of up-regulated genes among all DE genes, 
the default value is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="sim.counts_+3A_replace">replace</code></td>
<td>
<p>sample with or without replacement from given parameters. 
See Details for more information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the total number of genes <code>nGenes</code> is larger 
than length of <code>mu</code> or <code>disp</code>, 
<code>replace</code> always equals <code>TRUE</code>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>counts</code></td>
<td>
<p>RNA-seq count data matrix.</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>treatment group vector.</p>
</td></tr>
<tr><td><code>lambda0</code></td>
<td>
<p>mean counts in control group for each gene.</p>
</td></tr>
<tr><td><code>phi0</code></td>
<td>
<p>dispersion parameter for each gene.</p>
</td></tr>
<tr><td><code>de</code></td>
<td>
<p>differentially expressed genes indicator: 
<code>0</code> for non-differentially expressed genes, 
<code>1</code> for up-regulated genes, 
<code>-1</code> for down-regulated genes.</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>log2 fold change for each gene between 
treatment group and control group.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 3                    ## sample size per treatment group
mu &lt;- 10                  ## mean counts in control group for all genes 
disp &lt;- 0.1               ## dispersion for all genes
fc &lt;- 2                   ## 2-fold change for DE genes

sim &lt;- sim.counts(m = m, mu = mu, disp = disp, fc = fc)
sim$counts                ## count data matrix


## varying fold change
fc1 &lt;- function(x){exp(rnorm(x, log(2), 0.5*log(2)))}
sim1 &lt;- sim.counts(m = m, mu = mu, disp = disp, fc = fc1)

</code></pre>

<hr>
<h2 id='ssize.twoSampVaryDelta'>Sample Size Calculations for Two-Sample Microarray Experiments with 
Differing Mean Expressions but fixed Standard Deviations Among Genes</h2><span id='topic+ssize.twoSampVaryDelta'></span>

<h3>Description</h3>

<p>For given desired power, controlled false discovery rate, 
and user-specified proportions of non-differentially expressed genes, 
<code>ssize.twoSampVaryDelta</code> calculates appropriate sample sizes for 
two-sample microarray experiments in which the differences between mean 
treatment expression levels (<em>delta.g</em> for gene <em>g</em>) 
vary among genes. 
A plot of power versus sample size is generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssize.twoSampVaryDelta(deltaMean, deltaSE, sigma, fdr = 0.05,
  power = 0.8, pi0 = 0.95, maxN = 35, side = "two-sided",
  cex.title = 1.15, cex.legend = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_deltamean">deltaMean</code></td>
<td>
<p>location (mean) parameter of normal distribution 
followed by each <em>delta.g</em>.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_deltase">deltaSE</code></td>
<td>
<p>scale (standard deviation) parameter of normal distribution 
followed by each <em>delta.g</em>.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_sigma">sigma</code></td>
<td>
<p>the common standard deviation of expressions for all genes.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_pi0">pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially 
expressed genes.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_side">side</code></td>
<td>
<p>options are &quot;two-sided&quot;, &quot;upper&quot;, or &quot;lower&quot;.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles.</p>
</td></tr>
<tr><td><code id="ssize.twoSampVaryDelta_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each <em>delta.g</em> is assumed to follow a Normal distribution 
with mean <code>deltaMean</code> and standard deviation <code>deltaSE</code>. 
The standard deviations of expressions are assumed identical for all genes.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations 
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at 
which desired power is first reached.</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding 
sample sizes.</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations with 
corresponding sample sizes.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>References</h3>

<p>Liu, P. and Hwang, J. T. G. (2007) Quick calculation for 
sample size while controlling false discovery rate with application 
to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746. 
</p>
<p>Orr, M. and Liu, P. (2009) Sample size estimation while controlling 
false discovery rate for microarray experiments using ssize.fdr package. 
<em>The R Journal</em>, 1, 1, May 2009, 47-53.
</p>


<h3>See Also</h3>

<p><code>ssize.twoSamp</code>, <code>ssize.twoSampVary</code>, 
<code>ssize.oneSamp</code>, <code>ssize.oneSampVary</code>, 
<code>ssize.F</code>, <code>ssize.Fvary</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dm &lt;- 1.2; ds &lt;- 0.1  ## the delta.g's follow a Normal(1.2, 0.1) distribution
s &lt;- 1                ## common standard deviation
fdr &lt;- 0.05           ## false discovery rate to be controlled
pwr &lt;- 0.8            ## desired power
pi0 &lt;- c(0.5, 0.8, 0.99) ## proportions of non-differentially expressed genes
N &lt;- 35               ## maximum sample size for calculations

size &lt;- ssize.twoSampVaryDelta(deltaMean = dm, deltaSE = ds, sigma = s, 
                               fdr = fdr, power = pwr, pi0 = pi0, 
                               maxN = N, side = "two-sided")
size$ssize                ## first sample size(s) to reach desired power
size$power                ## calculated power for each sample size
size$crit.vals            ## calculated critical value for each sample size

</code></pre>

<hr>
<h2 id='ssizeRNA_single'>Sample Size Calculations for Two-Sample RNA-seq Experiments 
with Single Set of Parameters</h2><span id='topic+ssizeRNA_single'></span>

<h3>Description</h3>

<p>This function calculates appropriate sample sizes for two-sample 
RNA-seq experiments for a desired power in which mean and 
dispersion parameters are identical for all genes. 
Sample size calculations are performed at controlled false discovery rates, 
user-specified proportions of non-differentially expressed genes, 
mean counts in control group, dispersion, and fold change. 
A plot of power versus sample size is generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssizeRNA_single(nGenes = 10000, pi0 = 0.8, m = 200, mu, disp, fc,
  up = 0.5, replace = TRUE, fdr = 0.05, power = 0.8, maxN = 35,
  side = "two-sided", cex.title = 1.15, cex.legend = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssizeRNA_single_+3A_ngenes">nGenes</code></td>
<td>
<p>total number of genes, the default value is <code>10000</code>.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_pi0">pi0</code></td>
<td>
<p>proportion of non-differentially expressed genes, 
the default value is <code>0.8</code>.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_m">m</code></td>
<td>
<p>pseudo sample size for generated data.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_mu">mu</code></td>
<td>
<p>a vector (or scalar) of mean counts in control group 
from which to simulate.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_disp">disp</code></td>
<td>
<p>a vector (or scalar) of dispersion parameter 
from which to simulate.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_fc">fc</code></td>
<td>
<p>a vector (or scalar, or a function that takes an integer n 
and generates a vector of length n)
of fold change for differentially expressed (DE) genes.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_up">up</code></td>
<td>
<p>proportion of up-regulated genes among all DE genes, 
the default value is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_replace">replace</code></td>
<td>
<p>sample with or without replacement from given parameters. 
See Details for more information.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_side">side</code></td>
<td>
<p>options are &quot;two-sided&quot;, &quot;upper&quot;, or &quot;lower&quot;.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles.</p>
</td></tr>
<tr><td><code id="ssizeRNA_single_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a vector is input for <code>pi0</code>, sample size calculations 
are performed for each proportion.
</p>
<p>If the total number of genes is larger than length of <code>mu</code> or 
<code>disp</code>, <code>replace</code> always equals <code>TRUE</code>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which 
desired power is first reached.</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes.</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations with 
corresponding sample sizes.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>References</h3>

<p>Liu, P. and Hwang, J. T. G. (2007) Quick calculation for 
sample size while controlling false discovery rate with application 
to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746. 
</p>
<p>Orr, M. and Liu, P. (2009) Sample size estimation while controlling 
false discovery rate for microarray experiments using ssize.fdr package. 
<em>The R Journal</em>, 1, 1, May 2009, 47-53. 
</p>
<p>Law, C. W., Chen, Y., Shi, W., Smyth, G. K. (2014). Voom: precision weights 
unlock linear model analysis tools for RNA-seq read counts. 
<em>Genome Biology</em> 15, R29.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ssizeRNA_vary">ssizeRNA_vary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mu &lt;- 10                ## mean counts in control group for all genes
disp &lt;- 0.1             ## dispersion for all genes
fc &lt;- 2                 ## 2-fold change for DE genes

size &lt;- ssizeRNA_single(m = 30, mu = mu, disp = disp, fc = fc, 
                        maxN = 20)
size$ssize              ## first sample size to reach desired power
size$power              ## calculated power for each sample size
size$crit.vals          ## calculated critical value for each sample size

</code></pre>

<hr>
<h2 id='ssizeRNA_vary'>Sample Size Calculations for Two-Sample RNA-seq Experiments with 
Differing Mean and Dispersion Among Genes</h2><span id='topic+ssizeRNA_vary'></span>

<h3>Description</h3>

<p>This function calculates appropriate sample sizes for two-sample 
RNA-seq experiments for a desired power in which mean and 
dispersion vary among genes. 
Sample size calculations are performed at controlled false discovery rates, 
user-specified proportions of non-differentially expressed genes, 
mean counts in control group, dispersion, and fold change. 
A plot of power versus sample size is generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssizeRNA_vary(nGenes = 10000, pi0 = 0.8, m = 200, mu, disp, fc,
  up = 0.5, replace = TRUE, fdr = 0.05, power = 0.8, maxN = 35,
  side = "two-sided", cex.title = 1.15, cex.legend = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssizeRNA_vary_+3A_ngenes">nGenes</code></td>
<td>
<p>total number of genes, the default value is <code>10000</code>.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_pi0">pi0</code></td>
<td>
<p>proportion of non-differentially expressed genes, 
the default value is <code>0.8</code>.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_m">m</code></td>
<td>
<p>pseudo sample size for generated data.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_mu">mu</code></td>
<td>
<p>a vector (or scalar) of mean counts in control group 
from which to simulate.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_disp">disp</code></td>
<td>
<p>a vector (or scalar) of dispersion parameter 
from which to simulate.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_fc">fc</code></td>
<td>
<p>a vector (or scalar, or a function that takes an integer n 
and generates a vector of length n)
of fold change for differentially expressed (DE) genes.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_up">up</code></td>
<td>
<p>proportion of up-regulated genes among all DE genes, 
the default value is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_replace">replace</code></td>
<td>
<p>sample with or without replacement from given parameters. 
See Details for more information.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_side">side</code></td>
<td>
<p>options are &quot;two-sided&quot;, &quot;upper&quot;, or &quot;lower&quot;.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles.</p>
</td></tr>
<tr><td><code id="ssizeRNA_vary_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a vector is input for <code>pi0</code>, sample size calculations 
are performed for each proportion.
</p>
<p>If the total number of genes is larger than length of <code>mu</code> or 
<code>disp</code>, <code>replace</code> always equals <code>TRUE</code>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which 
desired power is first reached.</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes.</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations with 
corresponding sample sizes.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>References</h3>

<p>Liu, P. and Hwang, J. T. G. (2007) Quick calculation for 
sample size while controlling false discovery rate with application 
to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746. 
</p>
<p>Orr, M. and Liu, P. (2009) Sample size estimation while controlling 
false discovery rate for microarray experiments using ssize.fdr package. 
<em>The R Journal</em>, 1, 1, May 2009, 47-53. 
</p>
<p>Law, C. W., Chen, Y., Shi, W., Smyth, G. K. (2014). Voom: precision weights 
unlock linear model analysis tools for RNA-seq read counts. 
<em>Genome Biology</em> 15, R29.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ssizeRNA_single">ssizeRNA_single</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(edgeR)
library(Biobase)
data(hammer.eset)
## load hammer dataset (Hammer, P. et al., 2010)

counts &lt;- exprs(hammer.eset)[, phenoData(hammer.eset)$Time == "2 weeks"]
counts &lt;- counts[rowSums(counts) &gt; 0,]
trt &lt;- hammer.eset$protocol[which(hammer.eset$Time == "2 weeks")] 

mu &lt;- apply(counts[, trt == "control"], 1, mean)  
## average read count in control group for each gene

d &lt;- DGEList(counts)
d &lt;- calcNormFactors(d)
d &lt;- estimateCommonDisp(d)
d &lt;- estimateTagwiseDisp(d)
disp &lt;- d$tagwise.dispersion      
## dispersion for each gene

## fixed fold change
fc &lt;- 2
size &lt;- ssizeRNA_vary(mu = mu, disp = disp, fc = fc, 
                      m = 30, maxN = 15, replace = FALSE)
size$ssize         ## first sample size to reach desired power
size$power         ## calculated power for each sample size
size$crit.vals     ## calculated critical value for each sample size


## varying fold change
## fc1 &lt;- function(x){exp(rnorm(x, log(2), 0.5*log(2)))}
## size1 &lt;- ssizeRNA_vary(pi0 = 0.8, mu = mu, disp = disp, fc = fc1, 
##                        m = 30, maxN = 20, replace = FALSE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
