<!DOCTYPE html><html><head><title>Help for package ssize.fdr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ssize.fdr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ssize.F'><p>Sample Size Calculations for Multi-Sample Microarray Experiments</p></a></li>
<li><a href='#ssize.fdr-package'><p>Sample Size Calculations for Microarray Experiments</p></a></li>
<li><a href='#ssize.Fvary'><p>Sample Size Calculations for Multi-Sample Microarray Experiments</p>
with Differing Variances among Genes</a></li>
<li><a href='#ssize.oneSamp'><p>Sample Size Calculations for One-Sample Microarray Experiments</p></a></li>
<li><a href='#ssize.oneSampVary'><p>Sample Size Calculations for One-Sample Microarray Experiments with</p>
Differing Mean Expressions and Standard Deviations Among Genes</a></li>
<li><a href='#ssize.twoSamp'><p>Sample Size Calculations for Two-Sample Microarray Experiments</p></a></li>
<li><a href='#ssize.twoSampVary'><p>Sample Size Calculations for Two-Sample Microarray Experiments with</p>
Differing Mean Expressions and Standard Deviations Among Genes</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Sample Size Calculations for Microarray Experiments</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Megan Orr [aut, cre], Peng Liu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Megan Orr &lt;megan.orr@ndsu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions that calculate appropriate sample sizes for one-sample t-tests, two-sample t-tests, and F-tests for microarray experiments based on desired power while controlling for false discovery rates. For all tests, the standard deviations (variances) among genes can be assumed fixed or random.  This is also true for effect sizes among genes in one-sample and two sample experiments. Functions also output a chart of power versus sample size, a table of power at different sample sizes, and a table of critical test values at different sample sizes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-06 06:11:56 UTC; megan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-07 04:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ssize.F'>Sample Size Calculations for Multi-Sample Microarray Experiments</h2><span id='topic+ssize.F'></span>

<h3>Description</h3>

<p>Calculates appropriate sample sizes for multi-sample
microarray experiments for a desired power.  Sample size
calculations are performed at controlled false discovery rates
and user-specified proportions of non-differentially expressed
genes, design matrix, and standard deviation. A graph of power
versus sample size is created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssize.F(X, beta, L = NULL, dn, sigma, fdr = 0.05, power = 0.8, pi0 = 0.95, maxN = 20,
  cex.title=1.15, cex.legend=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssize.F_+3A_x">X</code></td>
<td>
<p>design matrix of experiment</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_beta">beta</code></td>
<td>
<p>parameter vector</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_l">L</code></td>
<td>
<p>coefficient matrix or vector for linear contrasts of
interest</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_dn">dn</code></td>
<td>
<p>a function of the degrees of freedom based on the design
of the experiment</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_sigma">sigma</code></td>
<td>
<p>the standard deviation for all genes</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_pi0">pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially
expressed genes</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles</p>
</td></tr>
<tr><td><code id="ssize.F_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Standard deviations are assumed to be identical for all genes.
See the function <code><a href="#topic+ssize.Fvary">ssize.Fvary</a></code> for sample size
calculations with varying standard deviations among genes.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which desired
power is first reached</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations with corresponding
sample sizes</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Powers calculated to be 0 may be negligibly conservative.
</p>
<p>Critical values calculated as &lsquo;NA&rsquo; are values &gt;100.
</p>


<h3>Author(s)</h3>

<p>Megan Orr <a href="mailto:megan.orr@ndsu.edu">megan.orr@ndsu.edu</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a></p>


<h3>References</h3>

<p> Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation
for sample size while controlling false discovery rate with
application to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746. </p>


<h3>See Also</h3>

<p><code><a href="#topic+ssize.twoSampVary">ssize.twoSampVary</a></code>, <code><a href="#topic+ssize.oneSamp">ssize.oneSamp</a></code>,
<code><a href="#topic+ssize.oneSampVary">ssize.oneSampVary</a></code>, <code><a href="#topic+ssize.F">ssize.F</a></code>,
<code><a href="#topic+ssize.Fvary">ssize.Fvary</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> ##Sample size calculation for three-treatment loop design microarray experiment
 ##See Figure S2 of Liu &amp; Hwang (2007)
 des&lt;-matrix(c(1,-1,0,0,1,-1),ncol=2,byrow=FALSE)	##design matrix of loop design experiment
 b&lt;-c(1,-0.5)			##difference between first two treatments is 1 and
  #second and third treatments is -0.5
 df&lt;-function(n){3*n-2}		##degrees of freedom for this design is 3n-2
 s&lt;-1				##standard deviation
 a&lt;-0.05			##false discovery rate to be controlled
 pwr1&lt;-0.8			##desired power
 p0&lt;-c(0.5,0.9,0.95,0.995)	##proportions of non-differentially expressed genes
 N1&lt;-20				##maximum sample size for calculations

 ft&lt;-ssize.F(X=des,beta=b,dn=df,sigma=s,fdr=a,power=pwr1,pi0=p0,maxN=N1)
 ft$ssize	##first sample sizes to reach desired power for each proportion of
  #non-differentially expressed genes
 ft$power	##power for each sample size
 ft$crit.vals	##critical value for each sample size

</code></pre>

<hr>
<h2 id='ssize.fdr-package'>Sample Size Calculations for Microarray Experiments</h2><span id='topic+ssize.fdr-package'></span><span id='topic+ssize.fdr'></span>

<h3>Description</h3>

<p>This package calculates appropriate sample sizes for one-sample,
two-sample, and multi-sample microarray experiments for a desired power
of the test.  Sample sizes are calculated under controlled false
discovery rates and fixed proportions of non-differentially expressed
genes. Outputs a graph of power versus sample size.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ssize.fdr</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.3</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-06-05</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>For all functions, the user inputs the desired power, the false
discovery rate to be controlled, the proportion(s) of non-
differentially expressed genes, and the maximum possible sample size
to be used in calculations. If the user inputs a vector of proportions
of non-differentially expressed genes, samples size calculations are
performed for each proportion. For the function <code><a href="#topic+ssize.twoSamp">ssize.twoSamp</a></code>,
the user must additionally input the common difference in mean treatment
expressions as well as the common standard deviation for all genes.
This becomes the common effect size and common standard deviation for
all genes when using the function <code><a href="#topic+ssize.oneSamp">ssize.oneSamp</a></code>. For the
function <code><a href="#topic+ssize.twoSampVary">ssize.twoSampVary</a></code> (<code><a href="#topic+ssize.oneSampVary">ssize.oneSampVary</a></code>)
the differences in mean treatment expressions (effect sizes) are assumed
to follow a normal distribution and the variances among genes are assumed
to follow an inverse gamma distribution, so parameters for these
distributions must be entered. For the function <code><a href="#topic+ssize.F">ssize.F</a></code>,
the design matrix of the experiment, the parameter vector, and an optional
coefficient matrix or vector of linear contrasts of interest must also be
entered. The function <code><a href="#topic+ssize.Fvary">ssize.Fvary</a></code> allows the variances of
the genes to follow an inverse gamma distribution, so the shape and scale
parameters must be specified by the user.
</p>


<h3>Author(s)</h3>

<p>Megan Orr &lt;megan.orr@ndsu.edu&gt;, Peng Liu &lt;pliu@iastate.edu&gt;</p>


<h3>References</h3>

<p>Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation for
sample size while controlling false discovery rate with application
to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746.</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 a&lt;-0.05	##false discovery rate to be controlled
 pwr&lt;-0.8	##desired power
 p0&lt;-c(0.5,0.9,0.95)	##proportions of non-differentially expressed genes
 N&lt;-20; N1&lt;-35	##maximum sample size for calculations


 ##Example of function ssize.oneSamp
 d&lt;-1		##effect size
 s&lt;-0.5  	##standard deviation
 os&lt;-ssize.oneSamp(delta=d,sigma=s,fdr=a,power=pwr,pi0=p0,maxN=N,side="two-sided")
 os$ssize	##first sample sizes to reach desired power
 os$power	##calculated power for each sample size
 os$crit.vals	##calculated critical value for each sample size


 ##Example of function ssize.oneSampVary
 dm&lt;-2;  ds&lt;-1	##the effect sizes of the genes follow a Normal(2,1) distribution
 alph&lt;-3;  beta&lt;-1	##the variances of the genes follow an Inverse Gamma(3,1) distribution.
 osv&lt;-ssize.oneSampVary(deltaMean=dm,deltaSE=ds,a=alph,b=beta,fdr=a,power=pwr,
 pi0=p0,maxN=N1,side="two-sided")
 osv$ssize	##first sample sizes to reach desired power
 osv$power	##calculated power for each sample size
 osv$crit.vals	##calculated critical value for each sample size


 ##Example of function ssize.twoSamp
 ##Calculates sample sizes for two-sample microarray experiments
 ##See Figure 1.(a) of Liu &amp; Hwang (2007)
 d1&lt;-1		##difference in differentially expressed genes to be detected
 s1&lt;-0.5  	##standard deviation
 ts&lt;-ssize.twoSamp(delta=d1,sigma=s1,fdr=a,power=pwr,pi0=pi,maxN=N,side="two-sided")
 ts$ssize	##first sample sizes to reach desired power
 ts$power	##calculated power for each sample size
 ts$crit.vals	##calculated critical value for each sample size


 ##Example of function ssize.twoSampVary
 ##Calculates sample sizes for multi-sample microarray experiments in which both the differences in
   ##expressions between treatments and the standard deviations vary among genes.
 ##See Figure 3.(a) of Liu &amp; Hwang (2007)
 dm&lt;-2		##mean parameter of normal distribution of differences
 ##between treatments among genes
 ds&lt;-1		##standard deviation parameter of normal distribution
 ##of differences between treatments among genes
 alph&lt;-3	##shape parameter of inverse gamma distribution followed
 ##by standard deviations of genes
 beta&lt;-1	##scale parameter of inverse gamma distribution followed
 ##by standard deviations of genes
 tsv&lt;-ssize.twoSampVary(deltaMean=dm,deltaSE=ds,a=alph,b=beta,
 fdr=a,power=pwr,pi0=p0,maxN=N1,side="two-sided")
 tsv$ssize	##first sample sizes to reach desired power
 tsv$power	##calculated power for each sample size
 tsv$crit.vals	##calculated critical value for each sample sizesv


 ##Example of function ssize.F
 ##Sample size calculation for three-treatment loop design microarray experiment
 ##See Figure S2. of Liu &amp; Hwang (2007)
 des&lt;-matrix(c(1,-1,0,0,1,-1),ncol=2,byrow=FALSE)	##design matrix of loop design experiment
 b&lt;-c(1,-0.5)			##difference between first two treatments is 1 and
 ##second and third treatments is -0.5
 df&lt;-function(n){3*n-2}		##degrees of freedom for this design is 3n-2
 s&lt;-1				##standard deviation
 p0.F&lt;-c(0.5,0.9,0.95,0.995)	##proportions of non-differentially expressed genes

 ft&lt;-ssize.F(X=des,beta=b,dn=df,sigma=s,fdr=a,power=pwr,pi0=p0.F,maxN=N)
 ft$ssize	##first sample sizes to reach desired power
 ft$power	##calculated power for each sample size
 ft$crit.vals	##calculated critical value for each sample sizeft$ssize


 ##Example of function ssize.Fvary
 ##Sample size calculation for three-treatment loop design microarray experiment
 des&lt;-matrix(c(1,-1,0,0,1,-1),ncol=2,byrow=FALSE)	##design matrix of loop design experiment
 b&lt;-c(1,-0.5)			##difference between first two treatments is 1 and
 ##second and third treatments is -0.5
 df&lt;-function(n){3*n-2}		##degrees of freedom for this design is 3n-2
 alph&lt;-3;beta&lt;-1	##variances among genes follow an Inverse Gamma(3,1)
 a1&lt;-0.05	##fdr to be fixed
 p0.F&lt;-c(0.9,0.95,0.995)		##proportions of non-differentially expressed genes

 ftv&lt;-ssize.Fvary(X=des,beta=b,dn=df,a=alph,b=beta,fdr=a1,power=pwr,pi0=p0,maxN=N1)
 ftv$ssize	##first sample sizes to reach desired power
 ftv$power	##calculated power for each sample size
 ftv$crit.vals	##calculated critical value for each sample sizeft$ssize

</code></pre>

<hr>
<h2 id='ssize.Fvary'>Sample Size Calculations for Multi-Sample Microarray Experiments
with Differing Variances among Genes</h2><span id='topic+ssize.Fvary'></span>

<h3>Description</h3>

<p>Calculates appropriate sample sizes for multi-sample
microarray experiments in which standard deviations vary among genes.
Sample sizes are determined based on a desired power, a  controlled
false discovery rate, and user-specified proportions of
non-differentially expressed genes and design matrix. A graph of
power versus sample size is created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssize.Fvary(X, beta, L = NULL, dn, a, b, fdr = 0.05, power = 0.8, pi0 = 0.95, maxN = 20,
  cex.title=1.15, cex.legend=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssize.Fvary_+3A_x">X</code></td>
<td>
<p>design matrix of experiment</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_beta">beta</code></td>
<td>
<p>parameter vector</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_l">L</code></td>
<td>
<p>coefficient matrix or vector for linear contrasts of
interest</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_dn">dn</code></td>
<td>
<p>a function of the degrees of freedom based on the design
of the experiment</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_a">a</code></td>
<td>
<p>shape parameter of inverse gamma distribution followed by
variances of genes</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_b">b</code></td>
<td>
<p>scale parameter of inverse gamma distribution followed by
variances of genes</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_pi0">pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially
expressed genes</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles</p>
</td></tr>
<tr><td><code id="ssize.Fvary_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The variances among genes are assumed to follow an Inverse Gamma
distribution with shape parameter <code>a</code> and scale parameter
<code>b</code>.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which desired power
is first reached</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations with corresponding sample
sizes</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Numerical integration used in calculations performed by the function
<code><a href="stats.html#topic+integrate">integrate</a></code>, which uses adaptive quadrature of functions.
</p>
<p>Powers calculated to be 0 may be negligibly conservative.
</p>
<p>Critical values calculated as &lsquo;NA&rsquo; are values &gt;100.
</p>


<h3>Author(s)</h3>

<p>Megan Orr <a href="mailto:megan.orr@ndsu.edu">megan.orr@ndsu.edu</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a></p>


<h3>References</h3>

<p> Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation for
sample size while controlling false discovery rate with application
to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746. </p>


<h3>See Also</h3>

<p><code><a href="#topic+ssize.twoSamp">ssize.twoSamp</a></code>, <code><a href="#topic+ssize.twoSampVary">ssize.twoSampVary</a></code>,
<code><a href="#topic+ssize.oneSamp">ssize.oneSamp</a></code>, <code><a href="#topic+ssize.oneSampVary">ssize.oneSampVary</a></code>,
<code><a href="#topic+ssize.F">ssize.F</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> ##Sample size calculation for three-treatment loop design microarray experiment
 des&lt;-matrix(c(1,-1,0,0,1,-1),ncol=2,byrow=FALSE)	##design matrix of loop design experiment
 b&lt;-c(1,-0.5)			##difference between first two treatments is 1 and
  #second and third treatments is -0.5
 df&lt;-function(n){3*n-2}		##degrees of freedom for this design is 3n-2
 alph&lt;-3;beta&lt;-1	##variances among genes follow an Inverse Gamma(3,1)
 a1&lt;-0.05	##fdr to be fixed
 pwr&lt;-0.8	##desired power
 p0&lt;-c(0.9,0.95,0.995)		##proportions of non-differentially expressed genes
 N1&lt;-35		##maximum sample size to be used in calculations

 ftv&lt;-ssize.Fvary(X=des,beta=b,dn=df,a=alph,b=beta,fdr=a1,power=pwr,pi0=p0,maxN=N1)
 ftv$ssize	##first sample sizes to reach desired power
 ftv$power	##calculated power for each sample size
 ftv$crit.vals	##calculated critical value for each sample sizeft$ssize

</code></pre>

<hr>
<h2 id='ssize.oneSamp'>Sample Size Calculations for One-Sample Microarray Experiments</h2><span id='topic+ssize.oneSamp'></span>

<h3>Description</h3>

<p>Calculates appropriate sample sizes for one-sample
microarray experiments for a desired power.  Sample size calculations
are performed at controlled false discovery rates and user-specified
proportions of non-differentially expressed genes, effect size, and
standard deviation.  A graph of power versus sample size is created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssize.oneSamp(delta, sigma, fdr = 0.05, power = 0.8, pi0 = 0.95, maxN = 35,
side = "two-sided", cex.title=1.15, cex.legend=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssize.oneSamp_+3A_delta">delta</code></td>
<td>
<p>the common effect size for all genes</p>
</td></tr>
<tr><td><code id="ssize.oneSamp_+3A_sigma">sigma</code></td>
<td>
<p>the standard deviation for all genes</p>
</td></tr>
<tr><td><code id="ssize.oneSamp_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled</p>
</td></tr>
<tr><td><code id="ssize.oneSamp_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved</p>
</td></tr>
<tr><td><code id="ssize.oneSamp_+3A_pi0">pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially
expressed genes</p>
</td></tr>
<tr><td><code id="ssize.oneSamp_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations</p>
</td></tr>
<tr><td><code id="ssize.oneSamp_+3A_side">side</code></td>
<td>
<p>options are &quot;two-sided&quot;, &quot;upper&quot;, or &quot;lower&quot;</p>
</td></tr>
<tr><td><code id="ssize.oneSamp_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles</p>
</td></tr>
<tr><td><code id="ssize.oneSamp_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Effect sizes and standard deviations are assumed to be identical
for all genes.  See the function <code><a href="#topic+ssize.oneSampVary">ssize.oneSampVary</a></code>
for sample size calculations with varying effects sizes and
standard deviations among genes.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes at which desired power is first reached</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations with corresponding sample sizes</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Powers calculated to be 0 may be negligibly conservative.
</p>
<p>Critical values calculated as &lsquo;NA&rsquo; are values &gt;20.
</p>
<p>Running this function with the <code>side</code> option of &ldquo;lower&rdquo; will
possibly result in multiple warnings. Calculating the probability
that an observation is less than the negative critical value under
a t-distribution with non-centrality parameter <em>delta/sigma</em>
(see argument section above) and the appropriate degrees of freedom
is a calculation that is performed many times while the function
runs. When the difference between the critical value and
<em>delta/sigma</em> is large, this probability is virtually zero.
This happens repeatedly while the function <code><a href="stats.html#topic+optimize">optimize</a></code>
finds the appropriate critical value for each sample size.  Because
of this, the function <code><a href="stats.html#topic+pt">pt</a></code> outputs a value &lt;1e-8 in
addition to a warning of &ldquo;full precision not achieved&rdquo;. This has no
impact on the accuracy of the resulting calculations of sample size.
</p>


<h3>Author(s)</h3>

<p>Megan Orr <a href="mailto:megan.orr@ndsu.edu">megan.orr@ndsu.edu</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a></p>


<h3>References</h3>

<p> Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation for
sample size while controlling false discovery rate with application
to microarray analysis. Bioinformatics 23(6): 739-746.</p>


<h3>See Also</h3>

<p><code><a href="#topic+ssize.twoSampVary">ssize.twoSampVary</a></code>, <code><a href="#topic+ssize.oneSamp">ssize.oneSamp</a></code>,
<code><a href="#topic+ssize.oneSampVary">ssize.oneSampVary</a></code>, <code><a href="#topic+ssize.F">ssize.F</a></code>,
<code><a href="#topic+ssize.Fvary">ssize.Fvary</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> d&lt;-2		##effect size
 s&lt;-1  		##standard deviation
 a&lt;-0.05	##false discovery rate to be controlled
 pwr&lt;-0.8	##desired power
 p0&lt;-c(0.5,0.9,0.95)	##proportions of non-differentially expressed genes
 N&lt;-20		##maximum sample size for calculations
 os&lt;-ssize.oneSamp(delta=d,sigma=s,fdr=a,power=pwr,pi0=p0,maxN=N,side="two-sided")
 os$ssize	##first sample sizes to reach desired power
 os$power	##calculated power for each sample size
 os$crit.vals	##calculated critical value for each sample size
</code></pre>

<hr>
<h2 id='ssize.oneSampVary'>Sample Size Calculations for One-Sample Microarray Experiments with
Differing Mean Expressions and Standard Deviations Among Genes</h2><span id='topic+ssize.oneSampVary'></span>

<h3>Description</h3>

<p>Calculates appropriate sample sizes for two-sample microarray
experiments in which effect sizes as well as variances vary among genes.
Sample sizes are determined based on a desired power, a controlled false
discovery rate, and user-specified proportions of non-differentially
expressed genes. Outputs a graph of power versus sample size. A graph of
power versus sample size is created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssize.oneSampVary(deltaMean, deltaSE, a, b, fdr = 0.05, power = 0.8, pi0 = 0.95,
maxN = 35, side = "two-sided", cex.title=1.15, cex.legend=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssize.oneSampVary_+3A_deltamean">deltaMean</code></td>
<td>
<p>mean of normal distribution followed
by effect sizes among genes</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_deltase">deltaSE</code></td>
<td>
<p>standard deviation of normal distribution
followed by effect sizes among genes</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_a">a</code></td>
<td>
<p>shape parameter of inverse gamma distribution followed by variances
of genes</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_b">b</code></td>
<td>
<p>scale parameter of inverse gamma distribution followed by variances
of genes</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_pi0">pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially expressed
genes</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_side">side</code></td>
<td>
<p>options are &quot;two-sided&quot;, &quot;upper&quot;, or &quot;lower&quot;</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles</p>
</td></tr>
<tr><td><code id="ssize.oneSampVary_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The effect sizes among genes are assumed to follow a Normal distribution
with mean specified by <code>deltaMean</code> and standard deviation specified by
<code>deltaSE</code>. The variances among genes are assumed to follow an
Inverse Gamma distribution with shape parameter <code>a</code> and scale parameter
<code>b</code>.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which desired power is first
reached</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations with corresponding sample sizes</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Numerical integration used in calculations performed by the function
<code><a href="stats.html#topic+integrate">integrate</a></code>, which uses adaptive quadrature of functions.
</p>
<p>Powers calculated to be 0 may be negligibly conservative.
</p>
<p>Critical values calculated as &lsquo;NA&rsquo; are values &gt;20.
</p>
<p>Running this function may result in many warnings. Probabilities under
different t-distributions with non-zero non-centrality parameters are
calculated many times while the function runs.  If these probabilities
are virtually zero, the function <code><a href="stats.html#topic+pt">pt</a></code> outputs a value &lt;1e-8
and outputs a warning of &ldquo;full precision not achieved&rdquo;.  These values
have no impact on the accuracy of the resulting calculations.
</p>


<h3>Author(s)</h3>

<p>Megan Orr <a href="mailto:megan.orr@ndsu.edu">megan.orr@ndsu.edu</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a></p>


<h3>References</h3>

<p>Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation for sample
size while controlling false discovery rate with application to microarray
analysis. <em>Bioinformatics</em> 23(6): 739-746. </p>


<h3>See Also</h3>

<p><code><a href="#topic+ssize.twoSampVary">ssize.twoSampVary</a></code>, <code><a href="#topic+ssize.oneSamp">ssize.oneSamp</a></code>,
<code><a href="#topic+ssize.oneSampVary">ssize.oneSampVary</a></code>, <code><a href="#topic+ssize.F">ssize.F</a></code>,
<code><a href="#topic+ssize.Fvary">ssize.Fvary</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> dm&lt;-2;  ds&lt;-1	##the effect sizes of the genes follow a Normal(2,1) distribution
 alph&lt;-3;  beta&lt;-1	##the variances of the genes follow an Inverse Gamma(3,1) distribution.
 a2&lt;-0.05	##false discovery rate to be controlled
 pwr2&lt;-0.8	##desired power
 p0&lt;-c(0.90,0.95,0.995)	##proportions of non-differentially expressed genes
 N1&lt;-35		##maximum sample size to be used in calculations

 osv&lt;-ssize.oneSampVary(deltaMean=dm,deltaSE=ds,a=alph,b=beta,fdr=a2,power=pwr2,pi0=p0,
 maxN=N1,side="two-sided")
 osv$ssize	##first sample sizes to reach desired power
 osv$power	##calculated power for each sample size
 osv$crit.vals	##calculated critical value for each sample size
</code></pre>

<hr>
<h2 id='ssize.twoSamp'>Sample Size Calculations for Two-Sample Microarray Experiments</h2><span id='topic+ssize.twoSamp'></span>

<h3>Description</h3>

<p>Calculates appropriate sample sizes for two-sample
microarray experiments for a desired power. Sample size calculations
are performed at controlled false discovery rates, user-specified
proportions of non-differentially expressed genes, effect size, and
standard deviation.  A graph of power versus sample size is created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssize.twoSamp(delta, sigma, fdr = 0.05, power = 0.8, pi0 = 0.95, maxN = 35,
side = "two-sided", cex.title=1.15, cex.legend=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssize.twoSamp_+3A_delta">delta</code></td>
<td>
<p>the common difference in mean expressions between the two
samples	for all genes</p>
</td></tr>
<tr><td><code id="ssize.twoSamp_+3A_sigma">sigma</code></td>
<td>
<p>the common standard deviation of expressions for all genes</p>
</td></tr>
<tr><td><code id="ssize.twoSamp_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled</p>
</td></tr>
<tr><td><code id="ssize.twoSamp_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved</p>
</td></tr>
<tr><td><code id="ssize.twoSamp_+3A_pi0">pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially
expressed genes</p>
</td></tr>
<tr><td><code id="ssize.twoSamp_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations</p>
</td></tr>
<tr><td><code id="ssize.twoSamp_+3A_side">side</code></td>
<td>
<p>options are &quot;two-sided&quot;, &quot;upper&quot;, or &quot;lower&quot;</p>
</td></tr>
<tr><td><code id="ssize.twoSamp_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles</p>
</td></tr>
<tr><td><code id="ssize.twoSamp_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The true difference between mean expressions of the two samples
as well as the standard deviations of expressions are assumed
identical for all genes.  See the function
<code><a href="#topic+ssize.twoSampVary">ssize.twoSampVary</a></code> for sample size calculations
with varying differences between sample mean expressions and
standard deviations among genes.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which desired power
is first reached</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations of two-sample t-test
with corresponding sample sizes</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Powers calculated to be 0 may be negligibly conservative.
</p>
<p>Critical values calculated as &lsquo;NA&rsquo; are values &gt;20.
</p>
<p>Running this function with the <code>side</code> option of &quot;lower&quot; will
possibly result in multiple warnings. Calculating the probability
that an observation is less than the negative critical value under
a t-distribution with non-centrality parameter <em>delta/sigma</em>
(see argument section above) and the appropriate degrees of freedom
is a calculation that is performed many times while the function
runs. When the difference between the critical value and
<em>delta/sigma</em> is large, this probability is virtually zero.
This happens repeatedly while the function <code><a href="stats.html#topic+optimize">optimize</a></code>
finds the appropriate critical value for each sample size.  Because
of this, the function <code><a href="stats.html#topic+pt">pt</a></code> outputs a value &lt;1e-8 in
addition to a warning of &ldquo;full precision not achieved&rdquo;. This has no
impact on the accuracy of the resulting calculations of sample size.
</p>


<h3>Author(s)</h3>

<p>Megan Orr <a href="mailto:megan.orr@ndsu.edu">megan.orr@ndsu.edu</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a></p>


<h3>References</h3>

<p> Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation for
sample size while controlling false discovery rate with application
to microarray analysis. Bioinformatics 23(6): 739-746. </p>


<h3>See Also</h3>

<p><code><a href="#topic+ssize.twoSampVary">ssize.twoSampVary</a></code>, <code><a href="#topic+ssize.oneSamp">ssize.oneSamp</a></code>,
<code><a href="#topic+ssize.oneSampVary">ssize.oneSampVary</a></code>, <code><a href="#topic+ssize.F">ssize.F</a></code>,
<code><a href="#topic+ssize.Fvary">ssize.Fvary</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> ##See Figure 1.(a) of Liu &amp; Hwang (2007)
 d&lt;-1		##difference in differentially expressed genes to be detected
 s&lt;-0.5  	##standard deviation
 a&lt;-0.05	##false discovery rate to be controlled
 pwr&lt;-0.8	##desired power
 p0&lt;-c(0.5,0.9,0.95)	##proportions of non-differentially expressed genes
 N&lt;-20		##maximum sample size for calculations

 ts&lt;-ssize.twoSamp(delta=d,sigma=s,fdr=a,power=pwr,pi0=p0,maxN=N,side="two-sided")
 ts$ssize	##first sample sizes to reach desired power for each proportion of
			 ##non-differentially expressed genes
 ts$power	##calculated power for each sample size
 ts$crit.vals	##calculated critical value for each sample size

</code></pre>

<hr>
<h2 id='ssize.twoSampVary'>Sample Size Calculations for Two-Sample Microarray Experiments with
Differing Mean Expressions and Standard Deviations Among Genes</h2><span id='topic+ssize.twoSampVary'></span>

<h3>Description</h3>

<p>Calculates appropriate sample sizes for two-sample microarray
experiments in which the differences between mean treatment expression
levels (<em>delta.g</em> for gene <em>g</em>) as well as standard deviations
vary among genes. Sample sizes are determined based on a desired power, a
controlled false discovery rate, and user-specified proportions of
non-differentially expressed genes. A graph of power versus sample size is
created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssize.twoSampVary(deltaMean, deltaSE, a, b, fdr = 0.05, power = 0.8, pi0 = 0.95,
maxN = 35, side = "two-sided", cex.title=1.15, cex.legend=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssize.twoSampVary_+3A_deltamean">deltaMean</code></td>
<td>
<p>location (mean) parameter of normal distribution followed
by each <em>delta.g</em></p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_deltase">deltaSE</code></td>
<td>
<p>scale (standard deviation) parameter of normal distribution
followed by each <em>delta.g</em></p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_a">a</code></td>
<td>
<p>shape parameter of inverse gamma distribution followed by variances
of genes</p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_b">b</code></td>
<td>
<p>scale parameter of inverse gamma distribution followed by variances
of genes</p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_fdr">fdr</code></td>
<td>
<p>the false discovery rate to be controlled</p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_power">power</code></td>
<td>
<p>the desired power to be achieved</p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_pi0">pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially
expressed genes</p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_maxn">maxN</code></td>
<td>
<p>the maximum sample size used for power calculations</p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_side">side</code></td>
<td>
<p>options are &quot;two-sided&quot;, &quot;upper&quot;, or &quot;lower&quot;</p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_cex.title">cex.title</code></td>
<td>
<p>controls size of chart titles</p>
</td></tr>
<tr><td><code id="ssize.twoSampVary_+3A_cex.legend">cex.legend</code></td>
<td>
<p>controls size of chart legend</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each <em>delta.g</em> is assumed to follow a Normal distribution
with mean specified by <code>deltaMean</code> and standard deviation specified
by <code>deltaSE</code>. The variances among genes are assumed to follow an
Inverse Gamma distribution with shape parameter <code>a</code> and scale
parameter <code>b</code>.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which desired power is
first reached</p>
</td></tr>
<tr><td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes</p>
</td></tr>
<tr><td><code>crit.vals</code></td>
<td>
<p>critical value calculations with corresponding sample sizes</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Numerical integration used in calculations performed by the function
<code><a href="stats.html#topic+integrate">integrate</a></code>, which uses adaptive quadrature of functions.
</p>
<p>Powers calculated to be 0 may be negligibly conservative.
</p>
<p>Critical values calculated as &lsquo;NA&rsquo; are values &gt;20.
</p>
<p>Running this function may result in many warnings. Probabilities under
different t-distributions with non-zero non-centrality parameters are
calculated many times while the function runs.  If these probabilities
are virtually zero, the function <code><a href="stats.html#topic+pt">pt</a></code> outputs a value &lt;1e-8
and outputs a warning of &ldquo;full precision not achieved&rdquo;.  These values
have no impact on the accuracy of the resulting calculations.
</p>


<h3>Author(s)</h3>

<p>Megan Orr <a href="mailto:megan.orr@ndsu.edu">megan.orr@ndsu.edu</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a></p>


<h3>References</h3>

<p> Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation for
sample size while controlling false discovery rate with application
to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746. </p>


<h3>See Also</h3>

<p><code><a href="#topic+ssize.twoSampVary">ssize.twoSampVary</a></code>, <code><a href="#topic+ssize.oneSamp">ssize.oneSamp</a></code>,
<code><a href="#topic+ssize.oneSampVary">ssize.oneSampVary</a></code>, <code><a href="#topic+ssize.F">ssize.F</a></code>,
<code><a href="#topic+ssize.Fvary">ssize.Fvary</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> ##See Figure 3.(a) of Liu &amp; Hwang (2007)
 dm&lt;-2; ds&lt;-1	##the delta.g's follow a Normal(2,1) distribution
 alph&lt;-3; beta&lt;-1	##the variances of genes follow an Inverse Gamma(a,b) distribution
 a2&lt;-0.05	##false discovery rate to be controlled
 pwr2&lt;-0.8	##desired power
 p0&lt;-c(0.90,0.95,0.995)	##proportions of non-differentially expressed genes
 N1&lt;-35		##maximum sample size to be used in calculations

 tsv&lt;-ssize.twoSampVary(deltaMean=dm,deltaSE=ds,a=alph,b=beta,fdr=a2,power=pwr2,pi0=p0,
 maxN=N1,side="two-sided")
 tsv$ssize	##first sample size(s) to reach desired power
 tsv$power	##calculated power for each sample size
 tsv$crit.vals	##calculated critical value for each sample size

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
