<!DOCTYPE html><html lang="en"><head><title>Help for package ssMutPA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ssMutPA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cox_data'><p>cox_data Univariate cox proportional hazards regression data.</p></a></li>
<li><a href='#dot_data'><p>dot_data A data frame.</p></a></li>
<li><a href='#dotplot'><p>Plotting the Dot plot.</p></a></li>
<li><a href='#FastSEAscore'><p>Calculate the enrichment score of the pathways .</p></a></li>
<li><a href='#get_heatmap'><p>Plotting a heatmap with subtype labels.</p></a></li>
<li><a href='#get_indicate'><p>Indicator function.</p></a></li>
<li><a href='#get_mut_status'><p>Converts MAF file or data in other formats into mutation matrix.</p></a></li>
<li><a href='#get_RWR_ES'><p>Calculate the single-sample mutation-based pathway enrichment score.</p></a></li>
<li><a href='#get_samp_class'><p>Determining cancer subtypes using unsupervised spectral clustering algorithm.</p></a></li>
<li><a href='#get_seeds_score'><p>Calculate the local wight of seed nodes.</p></a></li>
<li><a href='#get_univarCox_result'><p>Perform the univariate Cox proportional hazards regression analysis.</p></a></li>
<li><a href='#mountain_plot'><p>Plotting the density ridges plot.</p></a></li>
<li><a href='#MRWR'><p>A global propagation algorithm, random walk with restart (RWR), to predict probable influence of nodes in the network by seed nodes.</p></a></li>
<li><a href='#mut_onco'><p>mut_onco A binary mutation matrix.</p></a></li>
<li><a href='#mut_status'><p>mut_status A binary mutation matrix.</p></a></li>
<li><a href='#Oncoplot'><p>Drawing a waterfall plot of a particular pathway.</p></a></li>
<li><a href='#Path_ES'><p>Path_ES Single-sample mutation-based pathway enrichment score profiles.</p></a></li>
<li><a href='#Path_Name'><p>Path_Name</p></a></li>
<li><a href='#RWR_res'><p>RWR_res Random walk results based on mutated genes in a single sample.</p></a></li>
<li><a href='#samp_class_onco'><p>samp_class_onco</p></a></li>
<li><a href='#sample_class'><p>sample_class The subtype labels of samples.</p></a></li>
<li><a href='#Seeds_Score'><p>Seeds_Score The local weight of seed nodes.</p></a></li>
<li><a href='#sur_onco'><p>sur_onco</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single-Sample Mutation-Based Pathway Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>A systematic bioinformatics tool to perform single-sample mutation-based pathway analysis by integrating somatic mutation data with the Protein-Protein Interaction (PPI) network. In this method, we use local and global weighted strategies to evaluate the effects of network genes from mutations according to the network topology and then calculate the mutation-based pathway enrichment score (ssMutPES) to reflect the accumulated effect of mutations of each pathway. Subsequently, the ssMutPES profiles are used for unsupervised spectral clustering to identify cancer subtypes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, ggridges, grDevices, igraph, kernlab, maftools,
Matrix, NbClust, parallel, pheatmap, RColorBrewer, stats,
survival, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, qpdf, rmarkdown</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-15 12:34:47 UTC; DELL</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han [aut, cre, cph],
  Yalan He [aut],
  Qian Wang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-15 22:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cox_data'>cox_data Univariate cox proportional hazards regression data.</h2><span id='topic+cox_data'></span>

<h3>Description</h3>

<p>A data frame for implementing univariate cox proportional hazards regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cox_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 20 rows and 293 columns.
</p>

<hr>
<h2 id='dot_data'>dot_data A data frame.</h2><span id='topic+dot_data'></span>

<h3>Description</h3>

<p>The data frame was used to plot a dot plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dot_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 350 rows and 4 columns.
</p>

<hr>
<h2 id='dotplot'>Plotting the Dot plot.</h2><span id='topic+dotplot'></span>

<h3>Description</h3>

<p>The function is used to draw a graph to reflect the univariate HRs and P-values of the pathways in different cancer types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dotplot(data, low_col = "#6ADD26", high_col = "#AB2513", cut_point = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dotplot_+3A_data">data</code></td>
<td>
<p>A pathway activity score matrix, which rows represent the pathways and the columns are samples.</p>
</td></tr>
<tr><td><code id="dotplot_+3A_low_col">low_col</code>, <code id="dotplot_+3A_high_col">high_col</code></td>
<td>
<p>Colours for low and high ends of the gradient.</p>
</td></tr>
<tr><td><code id="dotplot_+3A_cut_point">cut_point</code></td>
<td>
<p>The threshold of HRs,when HR is greater than the cut_point,HR is assigned cut_point.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Dot plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
data(dot_data)
#perform the function `dotplot`.
dotplot(dot_data)
</code></pre>

<hr>
<h2 id='FastSEAscore'>Calculate the enrichment score of the pathways .</h2><span id='topic+FastSEAscore'></span>

<h3>Description</h3>

<p>The function 'FastSEAscore' is used to calculate the pathway enrichment score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FastSEAscore(labels.list, correl_vector)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FastSEAscore_+3A_labels.list">labels.list</code></td>
<td>
<p>The position of the genes in the pathway in the ranked gene list .</p>
</td></tr>
<tr><td><code id="FastSEAscore_+3A_correl_vector">correl_vector</code></td>
<td>
<p>A ranked list of all genes in the PPI network.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Enrichment scores of pathways based on predefined gene ranked lists.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
pathway_path&lt;-system.file("extdata","kegg_323_gmt.Rdata",package = "ssMutPA")
load(pathway_path)
pathway_list&lt;-split(kegg_323_gmt[,2],kegg_323_gmt[,1])
data(RWR_res)
gene_list&lt;-sort(RWR_res,decreasing=TRUE)
tag.indicator &lt;- sign(match(names(gene_list), pathway_list[[1]], nomatch = 0))
#perform the function `FastSEAscore`.
Path_ES&lt;-FastSEAscore(labels.list=tag.indicator,correl_vector = gene_list)
names(Path_ES)&lt;-names(pathway_list)[1]
</code></pre>

<hr>
<h2 id='get_heatmap'>Plotting a heatmap with subtype labels.</h2><span id='topic+get_heatmap'></span>

<h3>Description</h3>

<p>The function 'get_heatmap' is used to plot a heatmap with subtype labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_heatmap(
  Path_ES,
  Path_name,
  samp_class,
  scale = "row",
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  fontsize = 8,
  annotation_legend = TRUE,
  annotation_names_row = TRUE,
  annotation_names_col = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_heatmap_+3A_path_es">Path_ES</code></td>
<td>
<p>Single-sample mutation-based pathway enrichment score profiles.The file can be generated by the function 'get_RWR_ES'.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_path_name">Path_name</code></td>
<td>
<p>The names of the pathways that you want to show in the heatmap.The Path_name must be included in the row names of the Path_ES .</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_samp_class">samp_class</code></td>
<td>
<p>A vector containing information about the subtype labels.The vector can be generated by the function 'get_samp_class'.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_scale">scale</code></td>
<td>
<p>character indicating if the values should be centered and scaled in either the row direction or the column direction, or none. Corresponding values are &quot;row&quot;, &quot;column&quot; and &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>Boolean values determining if rows should be clustered or hclust object.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>Boolean values determining if columns should be clustered or hclust object.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_show_rownames">show_rownames</code></td>
<td>
<p>Boolean specifying if row names are be shown.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_show_colnames">show_colnames</code></td>
<td>
<p>Boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_fontsize">fontsize</code></td>
<td>
<p>base font size for the plot.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_annotation_legend">annotation_legend</code></td>
<td>
<p>Boolean value showing if the legend for annotation tracks should be drawn .</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_annotation_names_row">annotation_names_row</code></td>
<td>
<p>Boolean value showing if the names for row annotation tracks should be drawn.</p>
</td></tr>
<tr><td><code id="get_heatmap_+3A_annotation_names_col">annotation_names_col</code></td>
<td>
<p>Boolean value showing if the names for column annotation tracks should be drawn.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
data(Path_ES,sample_class,Path_Name)
#perform the function `get_heatmap`.
get_heatmap(Path_ES,Path_name=Path_Name,samp_class=sample_class)
</code></pre>

<hr>
<h2 id='get_indicate'>Indicator function.</h2><span id='topic+get_indicate'></span>

<h3>Description</h3>

<p>Indicator function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_indicate(vector, cutpoint)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_indicate_+3A_vector">vector</code></td>
<td>
<p>A numerical value.</p>
</td></tr>
<tr><td><code id="get_indicate_+3A_cutpoint">cutpoint</code></td>
<td>
<p>The threshold of the indicator function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer with the value 1 or 0.
</p>

<hr>
<h2 id='get_mut_status'>Converts MAF file or data in other formats into mutation matrix.</h2><span id='topic+get_mut_status'></span>

<h3>Description</h3>

<p>The function 'get_mut_status' is used to convert MAF file or data in other formats into a binary mutation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mut_status(maf_data, nonsynonymous = TRUE, TCGA = TRUE, mut_rate = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mut_status_+3A_maf_data">maf_data</code></td>
<td>
<p>The patients' somatic mutation data, which in MAF format or others.</p>
</td></tr>
<tr><td><code id="get_mut_status_+3A_nonsynonymous">nonsynonymous</code></td>
<td>
<p>Logical. Determine if extract the non-silent somatic mutations .</p>
</td></tr>
<tr><td><code id="get_mut_status_+3A_tcga">TCGA</code></td>
<td>
<p>Logical. Determine whether the file is in MAF format .</p>
</td></tr>
<tr><td><code id="get_mut_status_+3A_mut_rate">mut_rate</code></td>
<td>
<p>Used to filter genes with target mutation rate .</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A binary mutations matrix, in which 1 represents that a particular gene has mutated in a particular sample, and 0 represents that gene is wild type.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
mut_path &lt;- system.file("extdata","mutation_data.Rdata",package = "ssMutPA")
load(mut_path)
#perform the function `get_mut_status`.
mut_status&lt;-get_mut_status(mutation_data,nonsynonymous=TRUE,TCGA=TRUE,mut_rate=0)
</code></pre>

<hr>
<h2 id='get_RWR_ES'>Calculate the single-sample mutation-based pathway enrichment score.</h2><span id='topic+get_RWR_ES'></span>

<h3>Description</h3>

<p>The function 'get_RWR_ES' is used to calculate the single-sample mutation-based pathway enrichment score. Using somatic mutation data,PPI network and pathway data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_RWR_ES(
  mut_status,
  min_sample = 0,
  max_sample = dim(mut_status)[1],
  net_data,
  pathway_data,
  r = 0.7,
  Numcore = 2,
  BC_Num = length(V(net_data)$name),
  cut_point = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_RWR_ES_+3A_mut_status">mut_status</code></td>
<td>
<p>A binary mutation matrix.The file can be generated by the function 'get_mut_status'.</p>
</td></tr>
<tr><td><code id="get_RWR_ES_+3A_min_sample">min_sample</code></td>
<td>
<p>The minimum number of mutated genes contained in a sample,default to 0.</p>
</td></tr>
<tr><td><code id="get_RWR_ES_+3A_max_sample">max_sample</code></td>
<td>
<p>The maximum number of mutated genes contained in a sample.</p>
</td></tr>
<tr><td><code id="get_RWR_ES_+3A_net_data">net_data</code></td>
<td>
<p>A list of the PPI network information, including nodes and edges.</p>
</td></tr>
<tr><td><code id="get_RWR_ES_+3A_pathway_data">pathway_data</code></td>
<td>
<p>A data frame containing the pathways and their corresponding genes. The first column is the names of pathways and the second column is the genes included in the pathways.</p>
</td></tr>
<tr><td><code id="get_RWR_ES_+3A_r">r</code></td>
<td>
<p>A numeric value between 0 and 1. r is a certain probability of continuing the random walk or restarting from the restart set. Default to 0.7.</p>
</td></tr>
<tr><td><code id="get_RWR_ES_+3A_numcore">Numcore</code></td>
<td>
<p>The number of threads when running programs with multiple threads,default to 2 .</p>
</td></tr>
<tr><td><code id="get_RWR_ES_+3A_bc_num">BC_Num</code></td>
<td>
<p>Number of background genes required to calculate seed node weight.</p>
</td></tr>
<tr><td><code id="get_RWR_ES_+3A_cut_point">cut_point</code></td>
<td>
<p>The threshold of indicator function .</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single-sample mutation-based pathway enrichment score profiles, where each element represents the enrichment score of a pathway in a sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
data(mut_status)
net_path &lt;- system.file("extdata","ppi_network.Rdata",package = "ssMutPA")
load(net_path)
pathway_path&lt;-system.file("extdata","kegg_323_gmt.Rdata",package = "ssMutPA")
load(pathway_path)
samp_name&lt;-c("TCGA-06-0881-01A","TCGA-76-4934-01A")
examp_data&lt;-mut_status[,samp_name]
#perform the function `get_RWR_ES`.
Path_ES&lt;-get_RWR_ES(examp_data,net_data=ppi_network,pathway_data=kegg_323_gmt,BC_Num=12436)
</code></pre>

<hr>
<h2 id='get_samp_class'>Determining cancer subtypes using unsupervised spectral clustering algorithm.</h2><span id='topic+get_samp_class'></span>

<h3>Description</h3>

<p>The function 'get_samp_class' using spectral clustering algorithm to obtain cancer subtype labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_samp_class(
  Path_ES,
  sur,
  seed_num = 50,
  cox_pval = 0.05,
  min.nc = 2,
  max.nc = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_samp_class_+3A_path_es">Path_ES</code></td>
<td>
<p>Single-sample mutation-based pathway enrichment score(ssMutPES) profiles.The file can be generated by the function 'get_RWR_ES'.</p>
</td></tr>
<tr><td><code id="get_samp_class_+3A_sur">sur</code></td>
<td>
<p>A matrix containing the samples' survival time and survival states.</p>
</td></tr>
<tr><td><code id="get_samp_class_+3A_seed_num">seed_num</code></td>
<td>
<p>The number of seeds for iterating to select the optimal clustering result.</p>
</td></tr>
<tr><td><code id="get_samp_class_+3A_cox_pval">cox_pval</code></td>
<td>
<p>A custom threshold to filter characteristic pathways, default to 0.05.</p>
</td></tr>
<tr><td><code id="get_samp_class_+3A_min.nc">min.nc</code></td>
<td>
<p>Minimal number of clusters.</p>
</td></tr>
<tr><td><code id="get_samp_class_+3A_max.nc">max.nc</code></td>
<td>
<p>maximal number of clusters,greater or equal to min.nc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the filtered pathways,the best seed for clustering,and cancer subtyoe labels .
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data.
surv_path &lt;- system.file("extdata","sur.Rdata",package = "ssMutPA")
load(surv_path)
data(Path_ES)
#perform function `get_samp_class`.
res&lt;-get_samp_class(Path_ES,sur,seed_num=5,cox_pval=0.05,min.nc = 2,max.nc =5)
</code></pre>

<hr>
<h2 id='get_seeds_score'>Calculate the local wight of seed nodes.</h2><span id='topic+get_seeds_score'></span>

<h3>Description</h3>

<p>The function 'get_seeds_score' is used to calculate the local wight of seed nodes in a single sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_seeds_score(net_data, seed, mut_gene, BC_Num, cut_point = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_seeds_score_+3A_net_data">net_data</code></td>
<td>
<p>A list of the PPI network information, including nodes and edges.</p>
</td></tr>
<tr><td><code id="get_seeds_score_+3A_seed">seed</code></td>
<td>
<p>A vector containing the gene symbols of the seed nodes.</p>
</td></tr>
<tr><td><code id="get_seeds_score_+3A_mut_gene">mut_gene</code></td>
<td>
<p>A vector containing the gene symbols of the mutated genes in a single sample.</p>
</td></tr>
<tr><td><code id="get_seeds_score_+3A_bc_num">BC_Num</code></td>
<td>
<p>Number of background genes.</p>
</td></tr>
<tr><td><code id="get_seeds_score_+3A_cut_point">cut_point</code></td>
<td>
<p>The threshold of indicator function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the weight of seed nodes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
net_path &lt;- system.file("extdata","ppi_network.Rdata",package = "ssMutPA")
load(net_path)
data(mut_status)
seed&lt;-intersect(names(mut_status[,1])[which(mut_status[,1]!=0)],igraph::V(ppi_network)$name)
mut_gene&lt;-intersect(names(mut_status[,1])[which(mut_status[,1]!=0)],igraph::V(ppi_network)$name)
#perform the function `get_seeds_score`.
Seeds_Score&lt;-get_seeds_score(net_data=ppi_network,seed=seed,mut_gene,BC_Num=12436,cut_point=0)
</code></pre>

<hr>
<h2 id='get_univarCox_result'>Perform the univariate Cox proportional hazards regression analysis.</h2><span id='topic+get_univarCox_result'></span>

<h3>Description</h3>

<p>The function 'get_univarCox_result' is used to perform the univariate Cox proportional hazards regression analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_univarCox_result(DE_path_sur)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_univarCox_result_+3A_de_path_sur">DE_path_sur</code></td>
<td>
<p>A matrix containing the activity values of all pathways in each sample, along with the survival time and survival status of the samples.Note that the column names of survival time and survival status must be &quot;survival&quot; and &quot;event&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the pathways' coefficient, HR, confidence interval, and survival related difference p-value .
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get path of the mutation annotation file.
data(cox_data)
#perform function `get_univarCox_result`.
res&lt;-get_univarCox_result(cox_data)
</code></pre>

<hr>
<h2 id='mountain_plot'>Plotting the density ridges plot.</h2><span id='topic+mountain_plot'></span>

<h3>Description</h3>

<p>The function 'mountain_plot' is used to draw a graph to reflect the distribution of the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mountain_plot(data, sample_class, Path_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mountain_plot_+3A_data">data</code></td>
<td>
<p>A pathway activity score matrix,which row names represent the pathways and the column names are samples.</p>
</td></tr>
<tr><td><code id="mountain_plot_+3A_sample_class">sample_class</code></td>
<td>
<p>A vector containing subtype labels of the samples.</p>
</td></tr>
<tr><td><code id="mountain_plot_+3A_path_name">Path_name</code></td>
<td>
<p>The names of the pathways that you want to show in the graph.The 'Path_name' must be included in the row names of the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Density ridges plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
data(Path_ES,sample_class)
#perform the function `mountain_plot`.
mountain_plot(data=Path_ES,sample_class=sample_class,Path_name=rownames(Path_ES)[c(12,20,74)])
</code></pre>

<hr>
<h2 id='MRWR'>A global propagation algorithm, random walk with restart (RWR), to predict probable influence of nodes in the network by seed nodes.</h2><span id='topic+MRWR'></span>

<h3>Description</h3>

<p>The function 'MRWR' is used to predict probable influence of nodes in the network by seed nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MRWR(
  net_AdjMatrNorm,
  Seeds,
  net_data,
  mut_gene,
  r = 0.7,
  BC_Num = length(V(net_data)$name),
  cut_point = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MRWR_+3A_net_adjmatrnorm">net_AdjMatrNorm</code></td>
<td>
<p>Row normalized network adjacency matrix.</p>
</td></tr>
<tr><td><code id="MRWR_+3A_seeds">Seeds</code></td>
<td>
<p>A vector containing the gene symbols of the seed nodes.</p>
</td></tr>
<tr><td><code id="MRWR_+3A_net_data">net_data</code></td>
<td>
<p>A list of the PPI network information,including nodes and edges .</p>
</td></tr>
<tr><td><code id="MRWR_+3A_mut_gene">mut_gene</code></td>
<td>
<p>A vector containing the gene symbols of the mutated genes in a sample.</p>
</td></tr>
<tr><td><code id="MRWR_+3A_r">r</code></td>
<td>
<p>A numeric value between 0 and 1. r is a certain probability of continuing the random walk or restarting from the restart set. Default to 0.7.</p>
</td></tr>
<tr><td><code id="MRWR_+3A_bc_num">BC_Num</code></td>
<td>
<p>Number of background genes required to calculate seed node weight.</p>
</td></tr>
<tr><td><code id="MRWR_+3A_cut_point">cut_point</code></td>
<td>
<p>The threshold of indicator function .</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An matrix of global weight, where the row names are genes in the network and the column names are samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
net_path &lt;- system.file("extdata","ppi_network.Rdata",package = "ssMutPA")
load(net_path)
net_AdjMatr&lt;-as.matrix(igraph::get.adjacency(ppi_network))
net_AdjMatrNorm &lt;- t(t(net_AdjMatr)/(Matrix::colSums(net_AdjMatr, na.rm = FALSE, dims = 1)))
data(mut_status)
mut_gene&lt;-intersect(names(mut_status[,1])[which(mut_status[,1]!=0)],igraph::V(ppi_network)$name)
seed&lt;-intersect(names(mut_status[,1])[which(mut_status[,1]!=0)],igraph::V(ppi_network)$name)
#perform the function `MRWR`.
RWR_res&lt;-MRWR(net_AdjMatrNorm,Seeds=seed,net_data=ppi_network,mut_gene,BC_Num = 12436)
</code></pre>

<hr>
<h2 id='mut_onco'>mut_onco A binary mutation matrix.</h2><span id='topic+mut_onco'></span>

<h3>Description</h3>

<p>mut_onco is used to calculate the mutation frequency of genes..
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_onco
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 11968 rows and 44 columns.
</p>

<hr>
<h2 id='mut_status'>mut_status A binary mutation matrix.</h2><span id='topic+mut_status'></span>

<h3>Description</h3>

<p>The mut_status is a binary mutation matrix,in which 1 represents mutation and 0 represents wild type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_status
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 12802 rows and 50 columns.
</p>

<hr>
<h2 id='Oncoplot'>Drawing a waterfall plot of a particular pathway.</h2><span id='topic+Oncoplot'></span>

<h3>Description</h3>

<p>Load the data in MAF format and draw a waterfall plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Oncoplot(
  maf,
  samp_class,
  sur,
  mut_status,
  pathway,
  pathway_name,
  isTCGA = FALSE,
  top = 20,
  clinicalFeatures = c("sample_group", "event"),
  class_col = c("#00468B", "#ED0000"),
  event_col = c("#B3DE69", "#BC80BD"),
  sortByAnnotation = TRUE,
  gene_mar = 7,
  removeNonMutated = FALSE,
  drawRowBar = TRUE,
  drawColBar = TRUE,
  leftBarData = NULL,
  leftBarLims = NULL,
  rightBarData = NULL,
  rightBarLims = NULL,
  topBarData = NULL,
  logColBar = FALSE,
  draw_titv = FALSE,
  showTumorSampleBarcodes = FALSE,
  fill = TRUE,
  showTitle = TRUE,
  titleText = NULL,
  vc_cols = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Oncoplot_+3A_maf">maf</code></td>
<td>
<p>A data of MAF format.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_samp_class">samp_class</code></td>
<td>
<p>A vector containing subtype labels of the samples.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_sur">sur</code></td>
<td>
<p>A matrix containing the samples' survival time and survival status.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_mut_status">mut_status</code></td>
<td>
<p>A binary mutations matrix.The file can be generated by the function 'get_mut_status'.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_pathway">pathway</code></td>
<td>
<p>A list containing pathway information .</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_pathway_name">pathway_name</code></td>
<td>
<p>The names of the pathways that you want to visualize.For example &quot;JAK-STAT signaling pathway&quot;.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_istcga">isTCGA</code></td>
<td>
<p>Is input MAF file from TCGA source? If TRUE uses only first 12 characters from Tumor_Sample_Barcode.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_top">top</code></td>
<td>
<p>How many top genes to be drawn,genes are arranged from high to low depending on the frequency of mutations. defaults to 20.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_clinicalfeatures">clinicalFeatures</code></td>
<td>
<p>Columns names from 'clinical.data' slot of MAF to be drawn in the plot.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_class_col">class_col</code></td>
<td>
<p>The color of sample class .</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_event_col">event_col</code></td>
<td>
<p>The color of survival status .</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_sortbyannotation">sortByAnnotation</code></td>
<td>
<p>Logical sort oncomatrix (samples) by provided 'clinicalFeatures'. Sorts based on first 'clinicalFeatures'. Defaults to TRUE. column-sort.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_gene_mar">gene_mar</code></td>
<td>
<p>Margin width for gene names.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_removenonmutated">removeNonMutated</code></td>
<td>
<p>Logical. If TRUE removes samples with no mutations in the GenePathwayOncoplots for better visualization. Default FALSE.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_drawrowbar">drawRowBar</code></td>
<td>
<p>Logical. Plots righ barplot for each gene. Default TRUE.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_drawcolbar">drawColBar</code></td>
<td>
<p>Logical plots top barplot for each sample. Default TRUE.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_leftbardata">leftBarData</code></td>
<td>
<p>Data for leftside barplot. Must be a data.frame with two columns containing gene names and values. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_leftbarlims">leftBarLims</code></td>
<td>
<p>Limits for 'leftBarData'. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_rightbardata">rightBarData</code></td>
<td>
<p>Data for rightside barplot. Must be a data.frame with two columns containing to gene names and values. Default 'NULL' which draws distibution by variant classification. This option is applicable when only 'drawRowBar' is TRUE.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_rightbarlims">rightBarLims</code></td>
<td>
<p>Limits for 'rightBarData'. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_topbardata">topBarData</code></td>
<td>
<p>Default 'NULL' which draws absolute number of mutation load for each sample. Can be overridden by choosing one clinical indicator(Numeric) or by providing a two column data.frame contaning sample names and values for each sample. This option is applicable when only 'drawColBar' is TRUE.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_logcolbar">logColBar</code></td>
<td>
<p>Plot top bar plot on log10 scale. Default FALSE.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_draw_titv">draw_titv</code></td>
<td>
<p>Logical Includes TiTv plot. Default FALSE</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_showtumorsamplebarcodes">showTumorSampleBarcodes</code></td>
<td>
<p>Logical to include sample names.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_fill">fill</code></td>
<td>
<p>Logical. If TRUE draws genes and samples as blank grids even when they are not altered.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_showtitle">showTitle</code></td>
<td>
<p>Default TRUE.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_titletext">titleText</code></td>
<td>
<p>Custom title. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="Oncoplot_+3A_vc_cols">vc_cols</code></td>
<td>
<p>named vector of colors for each Variant_Classification.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A waterfall plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load the data
mut_path &lt;- system.file("extdata","maffile.txt",package = "ssMutPA")
maf&lt;-maftools::read.maf(mut_path ,isTCGA = FALSE)
pathway_path &lt;- system.file("extdata","kegg_323_gmt.Rdata",package = "ssMutPA")
load(pathway_path)
data(samp_class_onco,mut_onco,sur_onco)
samples &lt;- names(samp_class_onco)
samp_class_onco &lt;- paste0("class_",samp_class_onco)
names(samp_class_onco) &lt;- samples
sur_onco$event &lt;- ifelse(sur_onco$event%in%1,"Dead","Alive")
col &lt;- c("#8DD3C7", "#FFFFB3", "#BEBADA", "#FB8072", "#80B1D3")
##draw a waterfall plot
#win.graph()
Oncoplot(maf,samp_class_onco,sur_onco,mut_onco,kegg_323_gmt,"IL-17 signaling pathway",vc_cols=col)
</code></pre>

<hr>
<h2 id='Path_ES'>Path_ES Single-sample mutation-based pathway enrichment score profiles.</h2><span id='topic+Path_ES'></span>

<h3>Description</h3>

<p>The Path_ES is used for clustering and other analysis .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Path_ES
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 215 rows and 882 columns.
</p>

<hr>
<h2 id='Path_Name'>Path_Name</h2><span id='topic+Path_Name'></span>

<h3>Description</h3>

<p>The pathway names that the user wants to display.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Path_Name
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 50.
</p>

<hr>
<h2 id='RWR_res'>RWR_res Random walk results based on mutated genes in a single sample.</h2><span id='topic+RWR_res'></span>

<h3>Description</h3>

<p>The RWR_res is random walk results,used for enrichment analysis .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RWR_res
</code></pre>


<h3>Format</h3>

<p>An object of class <code>numeric</code> of length 12436.
</p>

<hr>
<h2 id='samp_class_onco'>samp_class_onco</h2><span id='topic+samp_class_onco'></span>

<h3>Description</h3>

<p>Sample subtype labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samp_class_onco
</code></pre>


<h3>Format</h3>

<p>An object of class <code>integer</code> of length 44.
</p>

<hr>
<h2 id='sample_class'>sample_class The subtype labels of samples.</h2><span id='topic+sample_class'></span>

<h3>Description</h3>

<p>The subtype labels of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_class
</code></pre>


<h3>Format</h3>

<p>An object of class <code>integer</code> of length 882.
</p>

<hr>
<h2 id='Seeds_Score'>Seeds_Score The local weight of seed nodes.</h2><span id='topic+Seeds_Score'></span>

<h3>Description</h3>

<p>Seeds_Score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Seeds_Score
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 35 rows and 2 columns.
</p>

<hr>
<h2 id='sur_onco'>sur_onco</h2><span id='topic+sur_onco'></span>

<h3>Description</h3>

<p>Patient survival-related information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sur_onco
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 44 rows and 2 columns.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
