<!DOCTYPE html><html lang="en"><head><title>Help for package sssc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sssc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#config_df'><p>Default parameters of config.</p></a></li>
<li><a href='#generate_feature'><p>Feature Generation for Contamination Detection Model</p></a></li>
<li><a href='#getAlt2'><p>Second alternative allele percentage</p></a></li>
<li><a href='#getAnnoRate'><p>Annotation rate</p></a></li>
<li><a href='#getAvgLL'><p>Calculate average log-likelihood</p></a></li>
<li><a href='#getLowDepth'><p>Low depth percentage</p></a></li>
<li><a href='#getRatio'><p>Get the ratio of allele frequencies with a region</p></a></li>
<li><a href='#getSkewness'><p>Get absolute value of skewness</p></a></li>
<li><a href='#getSNVRate'><p>SNV percentage</p></a></li>
<li><a href='#getVar'><p>Calculate zygosity variable</p></a></li>
<li><a href='#locateFile'><p>Check input filename</p></a></li>
<li><a href='#negll'><p>Negative Log Likelihood</p></a></li>
<li><a href='#read_vcf'><p>VCF Data Input</p></a></li>
<li><a href='#readGATK'><p>Read in input vcf data in GATK format for Contamination detection</p></a></li>
<li><a href='#readStrelka'><p>Read in input vcf data in strelka2 format for Contamination detection</p></a></li>
<li><a href='#readVarDict'><p>Read in input vcf data in VarDict format for Contamination detection</p></a></li>
<li><a href='#readVarPROWL'><p>Read in input vcf data in VarPROWL format</p></a></li>
<li><a href='#rho_est'><p>Estimate Rho for Alternative Allele Frequency</p></a></li>
<li><a href='#rmChangePoint'><p>Remove CNV regions within VCF files by changepoint method</p></a></li>
<li><a href='#rmCNVinVCF'><p>Remove CNV regions within VCF files given cnv file</p></a></li>
<li><a href='#sssc'><p>Same Species Sample Contamination</p></a></li>
<li><a href='#summary_vcf'><p>VCF Data Summary</p></a></li>
<li><a href='#svm_class_model'><p>Default svm classification model.</p></a></li>
<li><a href='#svm_regression_model'><p>Default svm regression model.</p></a></li>
<li><a href='#train_ct'><p>Train Contamination Detection Model</p></a></li>
<li><a href='#update_vcf'><p>Remove CNV regions within VCF files</p></a></li>
<li><a href='#vcf_example'><p>VCF example file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Same Species Sample Contamination Detection</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Imports Variant Calling Format file into R. It can detect
             whether a sample contains contaminant from the same species.
             In the first stage of the approach, a change-point detection
             method is used to identify copy number variations for filtering.
             Next, features are extracted from the data for a support vector
             machine model. For log-likelihood calculation, the deviation
             parameter is estimated by maximum likelihood method. Using
             a radial basis function kernel support vector machine, the
             contamination of a sample can be detected.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>changepoint, e1071, ggplot2, stats, VGAM</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-15 04:21:27 UTC; tjiang8</td>
</tr>
<tr>
<td>Author:</td>
<td>Tao Jiang [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tao Jiang &lt;tjiang8@ncsu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-15 11:22:54 UTC</td>
</tr>
</table>
<hr>
<h2 id='config_df'>Default parameters of config.</h2><span id='topic+config_df'></span>

<h3>Description</h3>

<p>A dataframe containing default parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>config_df
</code></pre>


<h3>Format</h3>

<p>A data frame with 12 variables:
</p>

<dl>
<dt><code>threshold</code></dt><dd><p>Threshold for allele frequency</p>
</dd>
<dt><code>skew</code></dt><dd><p>Skewness for allele frequency</p>
</dd>
<dt><code>lower</code></dt><dd><p>Lower bound for allele frequency region</p>
</dd>
<dt><code>upper</code></dt><dd><p>Upper bound for allele frequency region</p>
</dd>
<dt><code>ldpthred</code></dt><dd><p>Threshold to determine low depth</p>
</dd>
<dt><code>hom_mle</code></dt><dd><p>Hom MLE of p in Beta-Binomial model</p>
</dd>
<dt><code>het_mle</code></dt><dd><p>Het MLE of p in Beta-Binomial model</p>
</dd>
<dt><code>Hom_thred</code></dt><dd><p>Threshold between hom and high</p>
</dd>
<dt><code>High_thred</code></dt><dd><p>Threshold between high and het</p>
</dd>
<dt><code>Het_thred</code></dt><dd><p>Threshold between het and low</p>
</dd>
<dt><code>hom_rho</code></dt><dd><p>Hom MLE of rho in Beta-Binomial model</p>
</dd>
<dt><code>het_rho</code></dt><dd><p>Het MLE of rho in Beta-Binomial model</p>
</dd>
</dl>


<h3>Source</h3>

<p>Created by Tao Jiang
</p>

<hr>
<h2 id='generate_feature'>Feature Generation for Contamination Detection Model</h2><span id='topic+generate_feature'></span>

<h3>Description</h3>

<p>Generates features from each pair of input VCF objects for training contamination detection model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_feature(file, hom_p = 0.999, het_p = 0.5, hom_rho = 0.005,
  het_rho = 0.1, mixture, homcut = 0.99, highcut = 0.7, hetcut = 0.3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_feature_+3A_file">file</code></td>
<td>
<p>VCF input object</p>
</td></tr>
<tr><td><code id="generate_feature_+3A_hom_p">hom_p</code></td>
<td>
<p>The initial value for p in Homozygous Beta-Binomial model, default is 0.999</p>
</td></tr>
<tr><td><code id="generate_feature_+3A_het_p">het_p</code></td>
<td>
<p>The initial value for p in Heterozygous Beta-Binomial model, default is 0.5</p>
</td></tr>
<tr><td><code id="generate_feature_+3A_hom_rho">hom_rho</code></td>
<td>
<p>The initial value for rho in Homozygous Beta-Binomial model, default is 0.005</p>
</td></tr>
<tr><td><code id="generate_feature_+3A_het_rho">het_rho</code></td>
<td>
<p>The initial value for rho in Heterozygous Beta-Binomial model, default is 0.1</p>
</td></tr>
<tr><td><code id="generate_feature_+3A_mixture">mixture</code></td>
<td>
<p>A vector of whether the sample is contaminated: 0 for pure; 1 for contaminated</p>
</td></tr>
<tr><td><code id="generate_feature_+3A_homcut">homcut</code></td>
<td>
<p>Cutoff allele frequency value between hom and high, default is 0.99</p>
</td></tr>
<tr><td><code id="generate_feature_+3A_highcut">highcut</code></td>
<td>
<p>Cutoff allele frequency value between high and het, default is 0.7</p>
</td></tr>
<tr><td><code id="generate_feature_+3A_hetcut">hetcut</code></td>
<td>
<p>Cutoff allele frequency value between het and low, default is 0.3</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with all features for training model of contamination detection
</p>

<hr>
<h2 id='getAlt2'>Second alternative allele percentage</h2><span id='topic+getAlt2'></span>

<h3>Description</h3>

<p>Second alternative allele percentage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAlt2(f)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAlt2_+3A_f">f</code></td>
<td>
<p>Input raw file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Percent of the second alternative allele
</p>

<hr>
<h2 id='getAnnoRate'>Annotation rate</h2><span id='topic+getAnnoRate'></span>

<h3>Description</h3>

<p>Annotation rate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAnnoRate(f)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAnnoRate_+3A_f">f</code></td>
<td>
<p>Input raw file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Percentage of annotation locus
</p>

<hr>
<h2 id='getAvgLL'>Calculate average log-likelihood</h2><span id='topic+getAvgLL'></span>

<h3>Description</h3>

<p>Calculate average log-likelihood
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAvgLL(df, hom_mle, het_mle, hom_rho, het_rho)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAvgLL_+3A_df">df</code></td>
<td>
<p>Input modified file</p>
</td></tr>
<tr><td><code id="getAvgLL_+3A_hom_mle">hom_mle</code></td>
<td>
<p>Hom MLE of p in Beta-Binomial model, default is 0.9981416 from NA12878_1_L5</p>
</td></tr>
<tr><td><code id="getAvgLL_+3A_het_mle">het_mle</code></td>
<td>
<p>Het MLE of p in Beta-Binomial model, default is 0.4737897 from NA12878_1_L5</p>
</td></tr>
<tr><td><code id="getAvgLL_+3A_hom_rho">hom_rho</code></td>
<td>
<p>Hom MLE of rho in Beta-Binomial model, default is 0.04570275 from NA12878_1_L5</p>
</td></tr>
<tr><td><code id="getAvgLL_+3A_het_rho">het_rho</code></td>
<td>
<p>Het MLE of rho in Beta-Binomial model, default is 0.02224098 from NA12878_1_L5</p>
</td></tr>
</table>


<h3>Value</h3>

<p>meanLL
</p>

<hr>
<h2 id='getLowDepth'>Low depth percentage</h2><span id='topic+getLowDepth'></span>

<h3>Description</h3>

<p>Low depth percentage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLowDepth(f, ldpthred)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getLowDepth_+3A_f">f</code></td>
<td>
<p>Input raw file</p>
</td></tr>
<tr><td><code id="getLowDepth_+3A_ldpthred">ldpthred</code></td>
<td>
<p>Threshold to determine low depth, default is 20</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Percentage of low depth
</p>

<hr>
<h2 id='getRatio'>Get the ratio of allele frequencies with a region</h2><span id='topic+getRatio'></span>

<h3>Description</h3>

<p>Get the ratio of allele frequencies with a region
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRatio(subdf, lower, upper)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getRatio_+3A_subdf">subdf</code></td>
<td>
<p>Dataframe with calculated statistics</p>
</td></tr>
<tr><td><code id="getRatio_+3A_lower">lower</code></td>
<td>
<p>Lower bound for allele frequency region</p>
</td></tr>
<tr><td><code id="getRatio_+3A_upper">upper</code></td>
<td>
<p>Upper bound for allele frequency region</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Ratio of allele frequencies with a region
</p>

<hr>
<h2 id='getSkewness'>Get absolute value of skewness</h2><span id='topic+getSkewness'></span>

<h3>Description</h3>

<p>Get absolute value of skewness
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSkewness(subdf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getSkewness_+3A_subdf">subdf</code></td>
<td>
<p>Input dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Absolute value of skewness
</p>

<hr>
<h2 id='getSNVRate'>SNV percentage</h2><span id='topic+getSNVRate'></span>

<h3>Description</h3>

<p>SNV percentage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSNVRate(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getSNVRate_+3A_df">df</code></td>
<td>
<p>Input raw file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Percentage of SNV
</p>

<hr>
<h2 id='getVar'>Calculate zygosity variable</h2><span id='topic+getVar'></span>

<h3>Description</h3>

<p>Calculate zygosity variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getVar(df, state, hom_mle, het_mle)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getVar_+3A_df">df</code></td>
<td>
<p>Input modified file</p>
</td></tr>
<tr><td><code id="getVar_+3A_state">state</code></td>
<td>
<p>Zygosity state</p>
</td></tr>
<tr><td><code id="getVar_+3A_hom_mle">hom_mle</code></td>
<td>
<p>MLE in hom model</p>
</td></tr>
<tr><td><code id="getVar_+3A_het_mle">het_mle</code></td>
<td>
<p>MLE in het model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Zygosity variable
</p>

<hr>
<h2 id='locateFile'>Check input filename</h2><span id='topic+locateFile'></span>

<h3>Description</h3>

<p>Check input filename
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locateFile(fn, extension)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="locateFile_+3A_fn">fn</code></td>
<td>
<p>Exact full file name of input file, including directory</p>
</td></tr>
<tr><td><code id="locateFile_+3A_extension">extension</code></td>
<td>
<p>Expected input file extension: vcf &amp; txt</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Valid directory
</p>

<hr>
<h2 id='negll'>Negative Log Likelihood</h2><span id='topic+negll'></span>

<h3>Description</h3>

<p>Calculates negative log likelihood for beta binomial distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>negll(x, size, prob, rho)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="negll_+3A_x">x</code></td>
<td>
<p>Depth of alternative allele</p>
</td></tr>
<tr><td><code id="negll_+3A_size">size</code></td>
<td>
<p>Total depth</p>
</td></tr>
<tr><td><code id="negll_+3A_prob">prob</code></td>
<td>
<p>Theoretical probability for heterozygous is 0.5, for homozygous is 0.999</p>
</td></tr>
<tr><td><code id="negll_+3A_rho">rho</code></td>
<td>
<p>Rho parameter of Beta-Binomial distribution of alternative allele</p>
</td></tr>
</table>

<hr>
<h2 id='read_vcf'>VCF Data Input</h2><span id='topic+read_vcf'></span>

<h3>Description</h3>

<p>Reads a file in vcf or vcf.gz file and creates a list containing Content, Meta, VCF and file_sample_name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_vcf(fn, vcffor, dbOnly = FALSE, depCut = FALSE, thred = 20,
  metaline = 200, extnum = 10, keepall = T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_vcf_+3A_fn">fn</code></td>
<td>
<p>Input vcf file name</p>
</td></tr>
<tr><td><code id="read_vcf_+3A_vcffor">vcffor</code></td>
<td>
<p>Input vcf data format: 1) GATK; 2) VarPROWL; 3) VarDict; 4) strelka2</p>
</td></tr>
<tr><td><code id="read_vcf_+3A_dbonly">dbOnly</code></td>
<td>
<p>Use dbSNP as filter, default is FALSE</p>
</td></tr>
<tr><td><code id="read_vcf_+3A_depcut">depCut</code></td>
<td>
<p>Use a threshold for min depth , default is False</p>
</td></tr>
<tr><td><code id="read_vcf_+3A_thred">thred</code></td>
<td>
<p>Threshold for min depth, default is 20</p>
</td></tr>
<tr><td><code id="read_vcf_+3A_metaline">metaline</code></td>
<td>
<p>Number of head lines to read in (better to be large enough), the lines will be checked if they contain meta information, default is 200</p>
</td></tr>
<tr><td><code id="read_vcf_+3A_extnum">extnum</code></td>
<td>
<p>The column number to be extracted from vcf, default is 10; 0 for not extracting any column; extnum should be between 10 and total column number</p>
</td></tr>
<tr><td><code id="read_vcf_+3A_keepall">keepall</code></td>
<td>
<p>Keep unextracted column in output, default is TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing (1) Content: a vector showing what is contained; (2) Meta: a data frame containing meta-information of the file;
(3) VCF: a data frame, the main part of VCF file; (4) file_sample_name: the file name and sample name,
in case when multiple samples exist in one file, file and sample names might be different
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file.name &lt;- system.file("extdata", "example.vcf.gz", package = "sssc")
example &lt;- read_vcf(fn=file.name, vcffor="VarPROWL")
</code></pre>

<hr>
<h2 id='readGATK'>Read in input vcf data in GATK format for Contamination detection</h2><span id='topic+readGATK'></span>

<h3>Description</h3>

<p>Read in input vcf data in GATK format for Contamination detection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readGATK(dr, dbOnly, depCut, thred, content, extnum, keepall)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readGATK_+3A_dr">dr</code></td>
<td>
<p>A valid input object</p>
</td></tr>
<tr><td><code id="readGATK_+3A_dbonly">dbOnly</code></td>
<td>
<p>Use dbSNP as filter, default is FALSE, passed from read_vcf</p>
</td></tr>
<tr><td><code id="readGATK_+3A_depcut">depCut</code></td>
<td>
<p>Use a threshold for min depth , default is False</p>
</td></tr>
<tr><td><code id="readGATK_+3A_thred">thred</code></td>
<td>
<p>Threshold for min depth, default is 20</p>
</td></tr>
<tr><td><code id="readGATK_+3A_content">content</code></td>
<td>
<p>Column names in VCF files</p>
</td></tr>
<tr><td><code id="readGATK_+3A_extnum">extnum</code></td>
<td>
<p>The column number or numbers to be extracted from vcf, default is 10; 0 for not extracting any columns</p>
</td></tr>
<tr><td><code id="readGATK_+3A_keepall">keepall</code></td>
<td>
<p>Keep unextracted column in output, default is TRUE, passed from read_vcf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe from VCF file
</p>

<hr>
<h2 id='readStrelka'>Read in input vcf data in strelka2 format for Contamination detection</h2><span id='topic+readStrelka'></span>

<h3>Description</h3>

<p>Read in input vcf data in strelka2 format for Contamination detection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readStrelka(dr, dbOnly, depCut, thred, content, extnum, keepall)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readStrelka_+3A_dr">dr</code></td>
<td>
<p>A valid input object</p>
</td></tr>
<tr><td><code id="readStrelka_+3A_dbonly">dbOnly</code></td>
<td>
<p>Use dbSNP as filter, default is FALSE, passed from read_vcf</p>
</td></tr>
<tr><td><code id="readStrelka_+3A_depcut">depCut</code></td>
<td>
<p>Use a threshold for min depth , default is False</p>
</td></tr>
<tr><td><code id="readStrelka_+3A_thred">thred</code></td>
<td>
<p>Threshold for min depth, default is 20</p>
</td></tr>
<tr><td><code id="readStrelka_+3A_content">content</code></td>
<td>
<p>Column names in VCF files</p>
</td></tr>
<tr><td><code id="readStrelka_+3A_extnum">extnum</code></td>
<td>
<p>The column number or numbers to be extracted from vcf, default is 10; 0 for not extracting any columns</p>
</td></tr>
<tr><td><code id="readStrelka_+3A_keepall">keepall</code></td>
<td>
<p>Keep unextracted column in output, default is TRUE, passed from read_vcf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe from VCF file
</p>

<hr>
<h2 id='readVarDict'>Read in input vcf data in VarDict format for Contamination detection</h2><span id='topic+readVarDict'></span>

<h3>Description</h3>

<p>Read in input vcf data in VarDict format for Contamination detection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVarDict(dr, dbOnly, depCut, thred, content, extnum, keepall)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readVarDict_+3A_dr">dr</code></td>
<td>
<p>A valid input object</p>
</td></tr>
<tr><td><code id="readVarDict_+3A_dbonly">dbOnly</code></td>
<td>
<p>Use dbSNP as filter, default is FALSE, passed from read_vcf</p>
</td></tr>
<tr><td><code id="readVarDict_+3A_depcut">depCut</code></td>
<td>
<p>Use a threshold for min depth , default is False</p>
</td></tr>
<tr><td><code id="readVarDict_+3A_thred">thred</code></td>
<td>
<p>Threshold for min depth, default is 20</p>
</td></tr>
<tr><td><code id="readVarDict_+3A_content">content</code></td>
<td>
<p>Column names in VCF files</p>
</td></tr>
<tr><td><code id="readVarDict_+3A_extnum">extnum</code></td>
<td>
<p>The column number to be extracted from vcf, default is 10; 0 for not extracting any column</p>
</td></tr>
<tr><td><code id="readVarDict_+3A_keepall">keepall</code></td>
<td>
<p>Keep unextracted column in output, default is TRUE, passed from read_vcf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe from VCF file
</p>

<hr>
<h2 id='readVarPROWL'>Read in input vcf data in VarPROWL format</h2><span id='topic+readVarPROWL'></span>

<h3>Description</h3>

<p>Read in input vcf data in VarPROWL format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readVarPROWL(dr, dbOnly, depCut, thred, content, extnum, keepall)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readVarPROWL_+3A_dr">dr</code></td>
<td>
<p>A valid input object</p>
</td></tr>
<tr><td><code id="readVarPROWL_+3A_dbonly">dbOnly</code></td>
<td>
<p>Use dbSNP as filter, default is FALSE, passed from read_vcf</p>
</td></tr>
<tr><td><code id="readVarPROWL_+3A_depcut">depCut</code></td>
<td>
<p>Use a threshold for min depth , default is False</p>
</td></tr>
<tr><td><code id="readVarPROWL_+3A_thred">thred</code></td>
<td>
<p>Threshold for min depth, default is 20</p>
</td></tr>
<tr><td><code id="readVarPROWL_+3A_content">content</code></td>
<td>
<p>Column names in VCF files</p>
</td></tr>
<tr><td><code id="readVarPROWL_+3A_extnum">extnum</code></td>
<td>
<p>The column number or numbers to be extracted from vcf, default is 10; 0 for not extracting any columns</p>
</td></tr>
<tr><td><code id="readVarPROWL_+3A_keepall">keepall</code></td>
<td>
<p>Keep unextracted column in output, default is TRUE, passed from read_vcf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vcf Dataframe from VCF file
</p>

<hr>
<h2 id='rho_est'>Estimate Rho for Alternative Allele Frequency</h2><span id='topic+rho_est'></span>

<h3>Description</h3>

<p>Estimates Rho parameter in beta binomial distribution for alternative allele frequency
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rho_est(vl)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rho_est_+3A_vl">vl</code></td>
<td>
<p>A list of vcf objects from read_vcf function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing (1) het_rho: Rho parameter of heterozygous location; (2) hom_rho: Rho parameter homozygous location;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("vcf_example")
vcf_list &lt;- list()
vcf_list[[1]] &lt;- vcf_example$VCF
res &lt;- rho_est(vl = vcf_list)
res$het_rho[[1]]$par
res$hom_rho[[1]]$par
</code></pre>

<hr>
<h2 id='rmChangePoint'>Remove CNV regions within VCF files by changepoint method</h2><span id='topic+rmChangePoint'></span>

<h3>Description</h3>

<p>Remove CNV regions within VCF files by changepoint method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmChangePoint(vcf, threshold, skew, lower, upper)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rmChangePoint_+3A_vcf">vcf</code></td>
<td>
<p>Input VCF files</p>
</td></tr>
<tr><td><code id="rmChangePoint_+3A_threshold">threshold</code></td>
<td>
<p>Threshold for allele frequency</p>
</td></tr>
<tr><td><code id="rmChangePoint_+3A_skew">skew</code></td>
<td>
<p>Skewness for allele frequency</p>
</td></tr>
<tr><td><code id="rmChangePoint_+3A_lower">lower</code></td>
<td>
<p>Lower bound for allele frequency region</p>
</td></tr>
<tr><td><code id="rmChangePoint_+3A_upper">upper</code></td>
<td>
<p>Upper bound for allele frequency region</p>
</td></tr>
</table>


<h3>Value</h3>

<p>VCF object without changepoint region
</p>

<hr>
<h2 id='rmCNVinVCF'>Remove CNV regions within VCF files given cnv file</h2><span id='topic+rmCNVinVCF'></span>

<h3>Description</h3>

<p>Remove CNV regions within VCF files given cnv file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmCNVinVCF(vcf, cnvobj)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rmCNVinVCF_+3A_vcf">vcf</code></td>
<td>
<p>Input VCF files</p>
</td></tr>
<tr><td><code id="rmCNVinVCF_+3A_cnvobj">cnvobj</code></td>
<td>
<p>cnv object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>VCF object without changepoint region
</p>

<hr>
<h2 id='sssc'>Same Species Sample Contamination</h2><span id='topic+sssc'></span>

<h3>Description</h3>

<p>Detects whether a sample is contaminated another sample of its same species. The input file should be in vcf format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sssc(file, rmCNV = FALSE, cnvobj = NULL, config = NULL,
  class_model = NULL, regression_model = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sssc_+3A_file">file</code></td>
<td>
<p>VCF input object</p>
</td></tr>
<tr><td><code id="sssc_+3A_rmcnv">rmCNV</code></td>
<td>
<p>Remove CNV regions, default is FALSE</p>
</td></tr>
<tr><td><code id="sssc_+3A_cnvobj">cnvobj</code></td>
<td>
<p>cnv object, default is NULL</p>
</td></tr>
<tr><td><code id="sssc_+3A_config">config</code></td>
<td>
<p>config information of parameters. A default set is generated as part of the model and is included in a model object, which contains</p>
</td></tr>
<tr><td><code id="sssc_+3A_class_model">class_model</code></td>
<td>
<p>An SVM classification model</p>
</td></tr>
<tr><td><code id="sssc_+3A_regression_model">regression_model</code></td>
<td>
<p>An SVM regression model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing (1) stat: a data frame with all statistics for contamination estimation; (2) result: contamination estimation (Class = 0, pure; Class = 1, contaminated)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(vcf_example)
result &lt;- sssc(file = vcf_example)
</code></pre>

<hr>
<h2 id='summary_vcf'>VCF Data Summary</h2><span id='topic+summary_vcf'></span>

<h3>Description</h3>

<p>Summarizes allele frequency information in scatter and density plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_vcf(vcf, ZG = NULL, CHR = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary_vcf_+3A_vcf">vcf</code></td>
<td>
<p>VCF object from read_vcf function</p>
</td></tr>
<tr><td><code id="summary_vcf_+3A_zg">ZG</code></td>
<td>
<p>zygosity: (1) null, for both het and hom, default; (2) het; (3) hom</p>
</td></tr>
<tr><td><code id="summary_vcf_+3A_chr">CHR</code></td>
<td>
<p>chromosome number: (1) null, all chromosome, default; (2) any specific number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing (1) scatter: allele frequency scatter plot; (2) density: allele frequency density plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("vcf_example")
tmp &lt;- summary_vcf(vcf = vcf_example, ZG = 'het', CHR = c(1,2))
plot(tmp$scatter)
plot(tmp$density)
</code></pre>

<hr>
<h2 id='svm_class_model'>Default svm classification model.</h2><span id='topic+svm_class_model'></span>

<h3>Description</h3>

<p>An svm object containing default svm classification model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>svm_class_model
</code></pre>


<h3>Format</h3>

<p>An svm object:</p>


<h3>Source</h3>

<p>Created by Tao Jiang
</p>

<hr>
<h2 id='svm_regression_model'>Default svm regression model.</h2><span id='topic+svm_regression_model'></span>

<h3>Description</h3>

<p>An svm object containing default svm regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>svm_regression_model
</code></pre>


<h3>Format</h3>

<p>An svm object:</p>


<h3>Source</h3>

<p>Created by Tao Jiang
</p>

<hr>
<h2 id='train_ct'>Train Contamination Detection Model</h2><span id='topic+train_ct'></span>

<h3>Description</h3>

<p>Trains two SVM models (classification and regression) to detects whether a sample is contaminated another sample of its same species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>train_ct(feature)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="train_ct_+3A_feature">feature</code></td>
<td>
<p>Feature list objects from generate_feature()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contains two trained svm models: regression &amp; classification
</p>

<hr>
<h2 id='update_vcf'>Remove CNV regions within VCF files</h2><span id='topic+update_vcf'></span>

<h3>Description</h3>

<p>Remove CNV regions within VCF files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_vcf(rmCNV = FALSE, vcf, cnvobj = NULL, threshold = 0.1,
  skew = 0.5, lower = 0.45, upper = 0.55)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="update_vcf_+3A_rmcnv">rmCNV</code></td>
<td>
<p>Remove CNV regions, default is FALSE</p>
</td></tr>
<tr><td><code id="update_vcf_+3A_vcf">vcf</code></td>
<td>
<p>Input VCF files</p>
</td></tr>
<tr><td><code id="update_vcf_+3A_cnvobj">cnvobj</code></td>
<td>
<p>cnv object, default is NULL</p>
</td></tr>
<tr><td><code id="update_vcf_+3A_threshold">threshold</code></td>
<td>
<p>Threshold for allele frequency, default is 0.1</p>
</td></tr>
<tr><td><code id="update_vcf_+3A_skew">skew</code></td>
<td>
<p>Skewness for allele frequency, default is 0.5</p>
</td></tr>
<tr><td><code id="update_vcf_+3A_lower">lower</code></td>
<td>
<p>Lower bound for allele frequency region, default is 0.45</p>
</td></tr>
<tr><td><code id="update_vcf_+3A_upper">upper</code></td>
<td>
<p>Upper bound for allele frequency region, default is 0.55</p>
</td></tr>
</table>


<h3>Value</h3>

<p>VCF file without CNV region
</p>

<hr>
<h2 id='vcf_example'>VCF example file.</h2><span id='topic+vcf_example'></span>

<h3>Description</h3>

<p>An example containing a list of 4 data frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf_example
</code></pre>


<h3>Format</h3>

<p>A list of 4 data frames:</p>


<h3>Source</h3>

<p>Created by Tao Jiang
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
