<!DOCTYPE html><html lang="en"><head><title>Help for package Sstack</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Sstack}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BSHorizontalStack'><p>Horizontal stacking Random Forest models.</p></a></li>
<li><a href='#BSstack'><p>Bootstrap Stacking model builder.</p></a></li>
<li><a href='#BSstack_predict'><p>Predict using a set of Stacked Random Forest models.</p></a></li>
<li><a href='#BSVerticalStack'><p>Vertical stacking Random Forest models.</p></a></li>
<li><a href='#StackData'><p>Sample Stack Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bootstrap Stacking of Random Forest Models for Heterogeneous
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin Matlock, Raziur Rahman</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin Matlock &lt;kevin.matlock@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Generates and predicts a set of linearly stacked Random Forest models using bootstrap sampling. Individual datasets may be heterogeneous (not all samples have full sets of features). Contains support for parallelization but the user should register their cores before running. This is an extension of the method found in Matlock (2018) &lt;<a href="https://doi.org/10.1186%2Fs12859-018-2060-2">doi:10.1186/s12859-018-2060-2</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>randomForest, foreach, dplyr, parallel, doParallel</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-04-30 19:13:32 UTC; root</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-05-01 15:38:50 UTC</td>
</tr>
</table>
<hr>
<h2 id='BSHorizontalStack'>Horizontal stacking Random Forest models.</h2><span id='topic+BSHorizontalStack'></span>

<h3>Description</h3>

<p>Generate the weights for a horizontally stacked set of Random Forest (RF) models given a set of heterogeneous datasets.
For horizontal stacking some subset of samples must be common among all datasets. Subfunction of BSstack but can be used stand-alone.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BSHorizontalStack(T = 100, mtry = NA, nodesize = 5, iter = 100,
  Xn = NULL, ECHO = TRUE, Cf = NULL, Y, X1, X2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BSHorizontalStack_+3A_t">T</code></td>
<td>
<p>Number of trees for the individual RF models. (int)</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_mtry">mtry</code></td>
<td>
<p>Number of variables available for splitting at each tree node. If a scalar is given then all models use the given values. If a 1D array is given then each individual model uses the given value.</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_nodesize">nodesize</code></td>
<td>
<p>Minimum size of terminal nodes. If a scalar is given then all models use the given values. If a 1D array is given then each individual model uses the given value. By default all models use 5.</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_iter">iter</code></td>
<td>
<p>The number of time to bootstrap sample the data. (int)</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_xn">Xn</code></td>
<td>
<p>List containing each dataset to be stacked. If not supplied will be generated from X1, X2, ...</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_echo">ECHO</code></td>
<td>
<p>Bool, enable to provide output to the user in terms of overlapping samples and runtime for CV.</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_cf">Cf</code></td>
<td>
<p>Character vector listing set of samples common among all given datasets. If not found will generate on it's own.</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_y">Y</code></td>
<td>
<p>Nsample x 1 data table of responses for ALL samples. Must have matching rownames with each individual dataset.</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_x1">X1</code></td>
<td>
<p>Data table of first dataset to be stacked. Rownames should be contained within Y.</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_x2">X2</code></td>
<td>
<p>Data table of second dataset to be stacked. Rownames should be contained within Y.</p>
</td></tr>
<tr><td><code id="BSHorizontalStack_+3A_...">...</code></td>
<td>
<p>Further data tables, X3, X4, ..., Xl.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Required Packages:
dplyr, randomForest, foreach
</p>


<h3>Value</h3>

<p>Weights and offsets for each individual RF model.
</p>

<hr>
<h2 id='BSstack'>Bootstrap Stacking model builder.</h2><span id='topic+BSstack'></span>

<h3>Description</h3>

<p>Creates a bootstrapped linear stacked set of Random Forest (RF) models given a set of heterogeneous datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BSstack(T = 50, mtry = NULL, nodesize = 5, iter = 25, CV = NA,
  Xn = NULL, ECHO = TRUE, Y, X1, X2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BSstack_+3A_t">T</code></td>
<td>
<p>Number of trees for the individual RF models. (int)</p>
</td></tr>
<tr><td><code id="BSstack_+3A_mtry">mtry</code></td>
<td>
<p>Number of variables available for splitting at each tree node. If a scalar is given then all models use the given values. If a 1D array is given then each individual model uses the given value. If NA then for each model it will be set to Nfeats/3</p>
</td></tr>
<tr><td><code id="BSstack_+3A_nodesize">nodesize</code></td>
<td>
<p>Minimum size of terminal nodes. If a scalar is given then all models use the given values. If a 1D array is given then each individual model uses the given value. By default all models use 5.</p>
</td></tr>
<tr><td><code id="BSstack_+3A_iter">iter</code></td>
<td>
<p>The number of time to bootstrap sample the data. (int)</p>
</td></tr>
<tr><td><code id="BSstack_+3A_cv">CV</code></td>
<td>
<p>Cross validation (CV) to measure mean-absolute error and correlation coefficient, if NA (default) no CV is performed. Otherwise the value gives the number of folds for CV. If CV&lt;2 then leave-one-out CV is performed. CV is performed utilizing the samples that have full record.</p>
</td></tr>
<tr><td><code id="BSstack_+3A_xn">Xn</code></td>
<td>
<p>List containing each dataset to be stacked. If not supplied will be generated from X1, X2, ...</p>
</td></tr>
<tr><td><code id="BSstack_+3A_echo">ECHO</code></td>
<td>
<p>Bool, enable to provide output to the user in terms of overlapping samples and runtime for CV.</p>
</td></tr>
<tr><td><code id="BSstack_+3A_y">Y</code></td>
<td>
<p>Nsample x 1 data table of responses for ALL samples. Must have matching rownames with each individual dataset.</p>
</td></tr>
<tr><td><code id="BSstack_+3A_x1">X1</code></td>
<td>
<p>Data table of first dataset to be stacked. Rownames should be contained within Y.</p>
</td></tr>
<tr><td><code id="BSstack_+3A_x2">X2</code></td>
<td>
<p>Data table of second dataset to be stacked. Rownames should be contained within Y.</p>
</td></tr>
<tr><td><code id="BSstack_+3A_...">...</code></td>
<td>
<p>Further data tables, X3, X4, ..., Xl.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Required Packages:
dplyr, randomForest, foreach
</p>


<h3>Value</h3>

<p>If CV != null : A list composed of:
[1] List containing [1] individual RF models, [2] Nstack +1 weights and [3] feature names for full record samples. This argument is what is used for BSstack_predict
[2] Mean-absolute error calculated using cross validation (scalar).
[3] Pearson correlation coefficient between actual and predicted values through cross validation (scalar -1&lt;=r&lt;=1).
[4] Individual weights calculate for each fold (CV x Nstack+1 matrix).
[5] Out of fold predictions for the overlaping samples.
[6] Actual values for the overlaping samples.
If CV &gt; 1 : Also
[7] The fold assignments for the overlapping samples.
If CV = null : Only [1] is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Sstack)
library(doParallel)
data(StackData)

AUC=StackData[[1]]
GE=StackData[[2]]
RPPA=StackData[[3]]

X1 &lt;- GE[1:400,1:75]
X2 &lt;- GE[200:400,76:150]
Xt &lt;- GE[401:487,]

set.seed(1)

cl &lt;- makeCluster(2)
registerDoParallel(cl)

Hbs &lt;- BSstack(T = 25, iter = 20, Y = AUC, X1 = X1, X2 = X2)

stopCluster(cl)

Yp &lt;- BSstack_predict(Hbs[[1]],Xt)

maeH1 &lt;- mean(abs(AUC[401:487,]-Yp[,1]))
maeH2 &lt;- mean(abs(AUC[401:487,]-Yp[,2]))
maeHs &lt;- mean(abs(AUC[401:487,]-Yp[,3]))
</code></pre>

<hr>
<h2 id='BSstack_predict'>Predict using a set of Stacked Random Forest models.</h2><span id='topic+BSstack_predict'></span>

<h3>Description</h3>

<p>Gives predictions for a linear bootstrapped stacked Random Forest predictors. Gives the predictions
of each individual model as well as the linearlly combined predictions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BSstack_predict(BSmodel, Xi)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BSstack_predict_+3A_bsmodel">BSmodel</code></td>
<td>
<p>List containing the individual Random Forest models, their weights, and feature names. Generated using BSstack function.</p>
</td></tr>
<tr><td><code id="BSstack_predict_+3A_xi">Xi</code></td>
<td>
<p>NxM datatable containing input features to be predicted. Must contain all features used in the individual RF models.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Required Packages:
randomForest
</p>


<h3>Value</h3>

<p>NxL+1 matrix where L is the number of individual RF models. Predictions for the ith RF model is found in the ith column of this matrix while predictions for the stacked model is in the final column.
</p>

<hr>
<h2 id='BSVerticalStack'>Vertical stacking Random Forest models.</h2><span id='topic+BSVerticalStack'></span>

<h3>Description</h3>

<p>Generate the weights for a vertically stacked set of Random Forest (RF) models given a set of heterogeneous datasets.
For vertical stacking at least one dataset must contain full record (all features). Subfunction of BSstack but can be used stand-alone.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BSVerticalStack(T = 50, mtry = NULL, nodesize = 5, iter = 25,
  ECHO = TRUE, Y, Xfull = NULL, Xn = NULL, X1, X2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BSVerticalStack_+3A_t">T</code></td>
<td>
<p>Number of trees for the individual RF models. (int)</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_mtry">mtry</code></td>
<td>
<p>Number of variables available for splitting at each tree node. If a scalar is given then all models use the given values. If a 1D array is given then each individual model uses the given value.</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_nodesize">nodesize</code></td>
<td>
<p>Minimum size of terminal nodes. If a scalar is given then all models use the given values. If a 1D array is given then each individual model uses the given value. By default all models use 5.</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_iter">iter</code></td>
<td>
<p>The number of time to bootstrap sample the data. (int)</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_echo">ECHO</code></td>
<td>
<p>Bool, enable to provide output to the user in terms of overlapping samples and runtime for CV.</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_y">Y</code></td>
<td>
<p>Nsample x 1 data table of responses for ALL samples. Must have matching rownames with each individual dataset.</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_xfull">Xfull</code></td>
<td>
<p>Data table containing samples with full record. Used for generating the weights. Will attempt to find if not given.</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_xn">Xn</code></td>
<td>
<p>List containing each dataset to be stacked. If not supplied will be generated from X1, X2, ...</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_x1">X1</code></td>
<td>
<p>Data table of first dataset to be stacked. Rownames should be contained within Y.</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_x2">X2</code></td>
<td>
<p>Data table of second dataset to be stacked. Rownames should be contained within Y.</p>
</td></tr>
<tr><td><code id="BSVerticalStack_+3A_...">...</code></td>
<td>
<p>Further data tables, X3, X4, ..., Xl.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Required Packages:
dplyr, randomForest, foreach
</p>


<h3>Value</h3>

<p>Weights and offsets for each individual RF model.
</p>

<hr>
<h2 id='StackData'>Sample Stack Data</h2><span id='topic+StackData'></span>

<h3>Description</h3>

<p>A demo dataset containing Gene Expression (GE), Reverse Phase Protein Array (RPPA) and drug sensitivity measure (AUC) for cancer cell lines that have been tested for the drug 17-AAG. Used to demonstrate the benefits of stacking by building an integrated model that will more effectively predict the AUC values given both GE and RPPA data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StackData
</code></pre>


<h3>Format</h3>

<p>A list of 3 variables containing Gene Expression (GE), Reverse Phase Protein Array (RPPA) and drug sensitivity measure (AUC):
</p>

<dl>
<dt>GE</dt><dd><p>Gene Expression (GE) data corresponding to cell lines tested by the drug 17-AAG taken from the Cancer Cell Line Encyclopedia. Relieff has been run to select the top 150 genes. </p>
</dd>
<dt>RPPA</dt><dd><p>Reverse Phase Protein Array (RPPA) data corresponding to cell lines tested by the drug 17-AAG taken from the MD Anderson Cell Lines Project. Relieff has been run to select the top 150 proteins.</p>
</dd>
<dt>AUC</dt><dd><p>Area under the Dose-Responce Curve (AUC) corresponding to the drug 17-AAG applied to each cell line. Taken from the Cancer Cell Line Encyclopedia. </p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="http://bioinformatics.mdanderson.org/main/MCLP:Overview">http://bioinformatics.mdanderson.org/main/MCLP:Overview</a>
<a href="https://portals.broadinstitute.org/ccle">https://portals.broadinstitute.org/ccle</a>
</p>


<h3>References</h3>

<p>Barretina,J. and et al. (2012) The Cancer Cell Line Encyclopedia enables
predictive modelling of anticancer drug sensitivity. Nature, 483 (7391),
603â€“607.
Li, J. and et al. (2016) Characterization of Human Cancer Cell Lines by Reverse-Phase Protein Arrays.
Cancer Cell (In Press).
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
