<!DOCTYPE html><html lang="en"><head><title>Help for package ssw</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ssw}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ssw-package'><p>ssw: Striped Smith-Waterman Algorithm for Sequence Alignment using SIMD</p></a></li>
<li><a href='#align'><p>Perform Smith-Waterman alignment of a read against a reference sequence</p></a></li>
<li><a href='#force_align'><p>Perform forced alignment with increased gap open penalty</p></a></li>
<li><a href='#formatter'><p>Format and pretty-print SSW forced alignment results without truncation</p></a></li>
<li><a href='#install_ssw_py'><p>Install ssw-py and its dependencies</p></a></li>
<li><a href='#is_installed_ssw_py'><p>Is ssw-py installed?</p></a></li>
<li><a href='#print.ssw'><p>Print SSW alignment results</p></a></li>
<li><a href='#ssw_py'><p>Global reference to ssw-py</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Striped Smith-Waterman Algorithm for Sequence Alignment using
SIMD</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides an R interface for 'SSW' (Striped Smith-Waterman)
    via its 'Python' binding 'ssw-py'. 'SSW' is a fast 'C' and 'C++'
    implementation of the Smith-Waterman algorithm for pairwise sequence
    alignment using Single-Instruction-Multiple-Data (SIMD) instructions.
    'SSW' enhances the standard algorithm by efficiently returning alignment
    information and suboptimal alignment scores.
    The core 'SSW' library offers performance improvements for various
    bioinformatics tasks, including protein database searches,
    short-read alignments, primary and split-read mapping,
    structural variant detection, and read-overlap graph generation.
    These features make 'SSW' particularly useful for genomic applications.
    Zhao et al. (2013) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0082138">doi:10.1371/journal.pone.0082138</a>&gt; developed the
    original 'C' and 'C++' implementation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://nanx.me/ssw-r/">https://nanx.me/ssw-r/</a>, <a href="https://github.com/nanxstats/ssw-r">https://github.com/nanxstats/ssw-r</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/nanxstats/ssw-r/issues">https://github.com/nanxstats/ssw-r/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Python (&gt;= 3.6.0), ssw-py (&gt;= 1.0.0). Detailed
installation instructions can be found in the README file.</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>reticulate</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-18 00:26:03 UTC; nanx</td>
</tr>
<tr>
<td>Author:</td>
<td>Nan Xiao <a href="https://orcid.org/0000-0002-0250-5673"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nan Xiao &lt;me@nanx.me&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-18 02:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ssw-package'>ssw: Striped Smith-Waterman Algorithm for Sequence Alignment using SIMD</h2><span id='topic+ssw'></span><span id='topic+ssw-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Provides an R interface for 'SSW' (Striped Smith-Waterman) via its 'Python' binding 'ssw-py'. 'SSW' is a fast 'C' and 'C++' implementation of the Smith-Waterman algorithm for pairwise sequence alignment using Single-Instruction-Multiple-Data (SIMD) instructions. 'SSW' enhances the standard algorithm by efficiently returning alignment information and suboptimal alignment scores. The core 'SSW' library offers performance improvements for various bioinformatics tasks, including protein database searches, short-read alignments, primary and split-read mapping, structural variant detection, and read-overlap graph generation. These features make 'SSW' particularly useful for genomic applications. Zhao et al. (2013) <a href="https://doi.org/10.1371/journal.pone.0082138">doi:10.1371/journal.pone.0082138</a> developed the original 'C' and 'C++' implementation.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Nan Xiao <a href="mailto:me@nanx.me">me@nanx.me</a> (<a href="https://orcid.org/0000-0002-0250-5673">ORCID</a>) [copyright holder]
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://nanx.me/ssw-r/">https://nanx.me/ssw-r/</a>
</p>
</li>
<li> <p><a href="https://github.com/nanxstats/ssw-r">https://github.com/nanxstats/ssw-r</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/nanxstats/ssw-r/issues">https://github.com/nanxstats/ssw-r/issues</a>
</p>
</li></ul>


<hr>
<h2 id='align'>Perform Smith-Waterman alignment of a read against a reference sequence</h2><span id='topic+align'></span>

<h3>Description</h3>

<p>Perform Smith-Waterman alignment of a read against a reference sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align(
  read,
  reference,
  gap_open = 3L,
  gap_extension = 1L,
  start_idx = 0L,
  end_idx = 0L,
  match_score = 2L,
  mismatch_penalty = 2L
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_+3A_read">read</code></td>
<td>
<p>A character string of the read.</p>
</td></tr>
<tr><td><code id="align_+3A_reference">reference</code></td>
<td>
<p>A character string of the reference.</p>
</td></tr>
<tr><td><code id="align_+3A_gap_open">gap_open</code></td>
<td>
<p>Penalty for opening a gap. Default is 3.</p>
</td></tr>
<tr><td><code id="align_+3A_gap_extension">gap_extension</code></td>
<td>
<p>Penalty for extending a gap. Default is 1.</p>
</td></tr>
<tr><td><code id="align_+3A_start_idx">start_idx</code></td>
<td>
<p>Index to start the alignment search. Default is 0.</p>
</td></tr>
<tr><td><code id="align_+3A_end_idx">end_idx</code></td>
<td>
<p>Index to end the alignment search.
The default value 0 means using the entire reference length.</p>
</td></tr>
<tr><td><code id="align_+3A_match_score">match_score</code></td>
<td>
<p>An integer for scoring matches,
ranging from 0 to 255. Default is 2.</p>
</td></tr>
<tr><td><code id="align_+3A_mismatch_penalty">mismatch_penalty</code></td>
<td>
<p>An integer for mismatch penalties,
ranging from 0 to 255. Default is 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>ssw</code> containing the ssw aligner object
and the alignment results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
a &lt;- align("ACGT", "TTTTACGTCCCCC")
a
a$alignment$optimal_score
a$alignment$sub_optimal_score

</code></pre>

<hr>
<h2 id='force_align'>Perform forced alignment with increased gap open penalty</h2><span id='topic+force_align'></span>

<h3>Description</h3>

<p>Perform forced alignment with increased gap open penalty
</p>


<h3>Usage</h3>

<pre><code class='language-R'>force_align(
  read,
  reference,
  force_overhang = FALSE,
  match_score = 2L,
  mismatch_penalty = 2L
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="force_align_+3A_read">read</code></td>
<td>
<p>A character string of the read.</p>
</td></tr>
<tr><td><code id="force_align_+3A_reference">reference</code></td>
<td>
<p>A character string of the reference.</p>
</td></tr>
<tr><td><code id="force_align_+3A_force_overhang">force_overhang</code></td>
<td>
<p>Logical. If <code>TRUE</code>, ensures only one end of the
alignment overhangs. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="force_align_+3A_match_score">match_score</code></td>
<td>
<p>An integer for scoring matches,
ranging from 0 to 255. Default is 2.</p>
</td></tr>
<tr><td><code id="force_align_+3A_mismatch_penalty">mismatch_penalty</code></td>
<td>
<p>An integer for mismatch penalties,
ranging from 0 to 255. Default is 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class <code>ssw</code> containing the input sequences,
the ssw aligner object, and the alignment results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Results are truncated
a &lt;- force_align("ACTG", "TTTTCTGCCCCCACG")
a

# Format the results
b &lt;- a |&gt; formatter()
b

# Print the formatted results directly
a |&gt; formatter(print = TRUE)

</code></pre>

<hr>
<h2 id='formatter'>Format and pretty-print SSW forced alignment results without truncation</h2><span id='topic+formatter'></span>

<h3>Description</h3>

<p>Format and pretty-print SSW forced alignment results without truncation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatter(x, print = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="formatter_+3A_x">x</code></td>
<td>
<p>An object of class <code>ssw</code> containing the forced alignment results.</p>
</td></tr>
<tr><td><code id="formatter_+3A_print">print</code></td>
<td>
<p>Pretty-print the results? Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The formatted forced alignment results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
a &lt;- force_align("ACTG", "TTTTCTGCCCCCACG")
b &lt;- a |&gt; formatter()
b
a |&gt; formatter(print = TRUE)

</code></pre>

<hr>
<h2 id='install_ssw_py'>Install ssw-py and its dependencies</h2><span id='topic+install_ssw_py'></span>

<h3>Description</h3>

<p>Install ssw-py and its dependencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_ssw_py(
  ...,
  envname = "r-ssw-py",
  new_env = identical(envname, "r-ssw-py")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="install_ssw_py_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="reticulate.html#topic+py_install">reticulate::py_install()</a></code>.</p>
</td></tr>
<tr><td><code id="install_ssw_py_+3A_envname">envname</code></td>
<td>
<p>The name or full path of the environment in which
ssw-py is installed. Default is <code>r-ssw-py</code>.</p>
</td></tr>
<tr><td><code id="install_ssw_py_+3A_new_env">new_env</code></td>
<td>
<p>Logical. If <code>TRUE</code>, the specified Python environment
will be deleted and recreated if it already exists.
Defaults to <code>TRUE</code> only when using the default environment name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns <code>NULL</code>. Primarily used for its side effect
of installing the Python package in the specified environment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
install_ssw_py()

</code></pre>

<hr>
<h2 id='is_installed_ssw_py'>Is ssw-py installed?</h2><span id='topic+is_installed_ssw_py'></span>

<h3>Description</h3>

<p>Is ssw-py installed?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_installed_ssw_py()
</code></pre>


<h3>Value</h3>

<p><code>TRUE</code> if installed, <code>FALSE</code> if not.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_installed_ssw_py()
</code></pre>

<hr>
<h2 id='print.ssw'>Print SSW alignment results</h2><span id='topic+print.ssw'></span>

<h3>Description</h3>

<p>Print SSW alignment results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ssw'
print(x, start_idx = 0L, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.ssw_+3A_x">x</code></td>
<td>
<p>An object of class <code>ssw</code>.</p>
</td></tr>
<tr><td><code id="print.ssw_+3A_start_idx">start_idx</code></td>
<td>
<p>Index to start printing from.</p>
</td></tr>
<tr><td><code id="print.ssw_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code><a href="base.html#topic+print">print()</a></code> (not used).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the input object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
a &lt;- align("ACGT", "TTTTACGTCCCCC")
a

</code></pre>

<hr>
<h2 id='ssw_py'>Global reference to ssw-py</h2><span id='topic+ssw_py'></span>

<h3>Description</h3>

<p>Global reference to ssw-py which will be initialized in <code>.onLoad</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssw_py
</code></pre>


<h3>Format</h3>

<p>An object of class <code>python.builtin.module</code> (inherits from <code>python.builtin.object</code>) of length 0.
</p>


<h3>Value</h3>

<p>ssw-py reference object
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
