<!DOCTYPE html><html lang="en"><head><title>Help for package StAMPP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {StAMPP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#potato'><p>Example genotype input format</p></a></li>
<li><a href='#potato.mini'><p>Smaller example genotype input format</p></a></li>
<li><a href='#stampp2genlight'><p>Convert StAMPP genotype data to genlight object</p></a></li>
<li><a href='#stamppAmova'><p>Analysis of Molecular Variance</p></a></li>
<li><a href='#stamppConvert'><p>Import and Convert</p></a></li>
<li><a href='#stamppFst'><p>Fst Computation</p></a></li>
<li><a href='#stamppGmatrix'><p>Genomic Relationship Calculation</p></a></li>
<li><a href='#stamppNeisD'><p>Genetic Distance Calculation</p></a></li>
<li><a href='#stamppPhylip'><p>Export to Phylip Format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Analysis of Mixed Ploidy Populations</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0), pegas</td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel, doParallel, foreach, adegenet, methods, utils</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-07-30</td>
</tr>
<tr>
<td>Author:</td>
<td>LW Pembleton</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>LW Pembleton &lt;lwpembleton@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Allows users to calculate pairwise Nei's Genetic Distances (Nei 1972), pairwise Fixation
 Indexes (Fst) (Weir &amp; Cockerham 1984) and also Genomic Relationship matrixes following Yang et al. (2010) in mixed and single
 ploidy populations. Bootstrapping across loci is implemented during Fst calculation to generate confidence intervals and p-values
 around pairwise Fst values. StAMPP utilises SNP genotype data of any ploidy level (with the ability to handle missing data) and is coded to  
 utilise multithreading where available to allow efficient analysis of large datasets. StAMPP is able to handle genotype data from genlight objects 
 allowing integration with other packages such adegenet.
 Please refer to LW Pembleton, NOI Cogan &amp; JW Forster, 2013, Molecular Ecology Resources, 13(5), 946-952. &lt;<a href="https://doi.org/10.1111%2F1755-0998.12129">doi:10.1111/1755-0998.12129</a>&gt; for the appropriate citation and user manual. Thank you in advance.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/lpembleton/StAMPP">https://github.com/lpembleton/StAMPP</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-07 15:40:14 UTC; lp40</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-08 04:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='potato'>Example genotype input format</h2><span id='topic+potato'></span>

<h3>Description</h3>

<p>A data frame containing Solcap potato genotype data in tetraploid and diploid format as an small example of the input format required by StAMPP
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(potato)
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 48 variables:
</p>

<dl>
<dt>Sample</dt><dd><p>Sample names</p>
</dd>
<dt>Pop</dt><dd><p>Population name</p>
</dd>
<dt>Ploidy</dt><dd><p>Ploidy level</p>
</dd>
<dt>Format</dt><dd><p>Format of genotype data</p>
</dd>
<dt>solcap_snp_c1_1</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_1000</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10000</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10001</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10011</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10012</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10031</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10042</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10050</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10054</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10109</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10130</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10157</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10202</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10252</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10253</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10255</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_1029</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10295</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10297</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10351</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10384</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10397</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10457</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10491</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10492</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10494</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10579</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10646</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10669</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10715</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10737</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10743</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10762</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10855</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10873</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10879</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10900</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10932</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_1094</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_11137</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_11144</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_11196</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_11206</dt><dd><p>genotype data</p>
</dd>
</dl>



<h3>Source</h3>

<p>The example genotype data is a subset of data from the publically avaliable Solcap potato dataset which was re-scored in GenomeStudio in diploid and tetraploid formats
</p>

<hr>
<h2 id='potato.mini'>Smaller example genotype input format</h2><span id='topic+potato.mini'></span>

<h3>Description</h3>

<p>A data frame containing Solcap potato genotype data in tetraploid and diploid format as an small example of the input format required by StAMPP
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(potato.mini)
</code></pre>


<h3>Format</h3>

<p>A data frame with 6 rows and 48 variables:
</p>

<dl>
<dt>Sample</dt><dd><p>Sample names</p>
</dd>
<dt>Pop</dt><dd><p>Population name</p>
</dd>
<dt>Ploidy</dt><dd><p>Ploidy level</p>
</dd>
<dt>Format</dt><dd><p>Format of genotype data</p>
</dd>
<dt>solcap_snp_c1_1</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_1000</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10000</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10001</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10011</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10012</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10031</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10042</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10050</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10054</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10109</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10130</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10157</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10202</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10252</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10253</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10255</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_1029</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10295</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10297</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10351</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10384</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10397</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10457</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10491</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10492</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10494</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10579</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10646</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10669</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10715</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10737</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10743</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10762</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10855</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10873</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10879</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10900</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_10932</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_1094</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_11137</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_11144</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_11196</dt><dd><p>genotype data</p>
</dd>
<dt>solcap_snp_c1_11206</dt><dd><p>genotype data</p>
</dd>
</dl>



<h3>Source</h3>

<p>The example genotype data is a subset of data from the publically avaliable Solcap potato dataset which was re-scored in GenomeStudio in diploid and tetraploid formats
</p>

<hr>
<h2 id='stampp2genlight'>Convert StAMPP genotype data to genlight object</h2><span id='topic+stampp2genlight'></span>

<h3>Description</h3>

<p>Converts a StAMPP formated allele frequency data frame generated from the stamppConvert function to a genlight object for use in other packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stampp2genlight(geno, pop = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stampp2genlight_+3A_geno">geno</code></td>
<td>
<p>a data frame containing allele frequency data generated from stamppConvert</p>
</td></tr>
<tr><td><code id="stampp2genlight_+3A_pop">pop</code></td>
<td>
<p>logical. True if population IDs are present in the StAMPP genotype data, False if population IDs are absent.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>StAMPP only exports to genlight objects as they are able to handle mixed ploidy datasets unlike genpop and genloci objects.
The genlight object allows the intergration between StAMPP and other common R packages such as ADEGENET
</p>


<h3>Value</h3>

<p>A object of class genlight which contains genotype data, individual IDs, population IDs (if present) and ploidy levels
</p>


<h3>Author(s)</h3>

<p>Luke Pembleton &lt;luke.pembleton at agriculture.vic.gov.au&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># import genotype data and convert to allele frequecies
data(potato.mini, package="StAMPP")
potato.freq &lt;- stamppConvert(potato.mini, "r")
# Convert the StAMPP formatted allele frequency data frame to a genlight object
potato.genlight &lt;- stampp2genlight(potato.freq, TRUE)
</code></pre>

<hr>
<h2 id='stamppAmova'>Analysis of Molecular Variance</h2><span id='topic+stamppAmova'></span>

<h3>Description</h3>

<p>Calculates an AMOVA based on the genetic distance matrix from stamppNeisD() using the amova() function from the package PEGAS for exploring within and between population variation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stamppAmova(dist.mat, geno, perm = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stamppAmova_+3A_dist.mat">dist.mat</code></td>
<td>
<p>the matrix of genetic distances between individuals generated from stamppNeisD()</p>
</td></tr>
<tr><td><code id="stamppAmova_+3A_geno">geno</code></td>
<td>
<p>a data frame containing allele frequency data generated from stamppConvert, or a genlight object containing genotype data, individual IDs, population IDs and ploidy levels</p>
</td></tr>
<tr><td><code id="stamppAmova_+3A_perm">perm</code></td>
<td>
<p>the number of permutations for the tests of hypotheses</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses the formula distance ~ populations, to calculate an AMOVA for population differentiation and within &amp; between population variation.
This function uses the amova function from the PEGAS package.
</p>


<h3>Value</h3>

<p>An object of class &quot;amova&quot; which is a list containing a table of sum of square deviations (SSD), mean square deviations (MSD) and the number of degrees of freedom as well as the variance components
</p>


<h3>Author(s)</h3>

<p>Luke Pembleton &lt;luke.pembleton at agriculture.vic.gov.au&gt;
</p>


<h3>References</h3>

<p>Paradis E (2010) pegas: an R package for population genetics with an integrated-modular approach. Bioinformatics 26, 419-420. &lt;doi:10.1093/bioinformatics/btp696&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># import genotype data and convert to allele frequecies
data(potato.mini, package="StAMPP")
potato.freq &lt;- stamppConvert(potato.mini, "r")
# Calculate genetic distance between individuals
potato.D.ind &lt;- stamppNeisD(potato.freq, FALSE)
# Calculate AMOVA
stamppAmova(potato.D.ind, potato.freq, 100)
</code></pre>

<hr>
<h2 id='stamppConvert'>Import and Convert</h2><span id='topic+stamppConvert'></span>

<h3>Description</h3>

<p>Imports biallelic AB formated or allele A frequency genotype data. If the data is in imported in biallelic AB format this function also converts it to allele frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stamppConvert(genotype.file, type = "csv")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stamppConvert_+3A_genotype.file">genotype.file</code></td>
<td>
<p>the genotype input file. This should be a R matrix object or a file path for a csv file containing the genotype data in either bialleleic AB format or allele 'A' frequency format, or a genlight object containing genotype data</p>
</td></tr>
<tr><td><code id="stamppConvert_+3A_type">type</code></td>
<td>
<p>the type of file the genotype data is being imported from; &quot;csv&quot; = comma seperated file, &quot;r&quot; = data frame in the R workspace, &quot;genlight&quot; = genlight object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class data.frame which contains allele frequency data for use in other StAMPP functions
</p>


<h3>Author(s)</h3>

<p>Luke Pembleton &lt;luke.pembleton at agriculture.vic.gov.au&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Import example data into the R workspace
data(potato.mini, package="StAMPP")
# Convert to allele frequencies
potato.freq &lt;- stamppConvert(potato.mini, "r")
</code></pre>

<hr>
<h2 id='stamppFst'>Fst Computation</h2><span id='topic+stamppFst'></span>

<h3>Description</h3>

<p>This function calculates pairwise Fst values along with confidence intervals and p-values between populations according to the method proposed by Wright(1949) and updated by Weir and Cockerham (1984)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stamppFst(geno, nboots = 100, percent = 95, nclusters = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stamppFst_+3A_geno">geno</code></td>
<td>
<p>a data frame containing allele frequency data generated from stamppConvert, or a genlight object containing genotype data, individual IDs, population IDs and ploidy levels</p>
</td></tr>
<tr><td><code id="stamppFst_+3A_nboots">nboots</code></td>
<td>
<p>number of bootstraps to perform across loci to generate confidence intervals and p-values</p>
</td></tr>
<tr><td><code id="stamppFst_+3A_percent">percent</code></td>
<td>
<p>the percentile to calculate the confidence interval around</p>
</td></tr>
<tr><td><code id="stamppFst_+3A_nclusters">nclusters</code></td>
<td>
<p>number of proccesor treads or cores to use during calculations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If possible, using multiple processing threads or cores is recommended to assist in calculating Fst values over a large number of bootstraps.
</p>


<h3>Value</h3>

<p>An object list with the components:
<code>Fsts</code>a matrix of pairwise Fst values between populations
<code>Pvalues</code>a matrix of p-values for each of the pairwise Fst values containined in the 'Fsts' matrix
<code>Bootstraps</code>a dataframe of each Fst value generated during Bootstrapping and the associated confidence intervals
If nboots&lt;2, no bootstrapping is performed and therefore only a matrix of Fst values is returned.
</p>


<h3>Author(s)</h3>

<p>Luke Pembleton &lt;luke.pembleton at agriculture.vic.gov.au&gt;
</p>


<h3>References</h3>

<p>Wright S (1949) The Genetical Structure of Populations. Annals of Human Genetics 15, 323-354. &lt;doi:10.1111/j.1469-1809.1949.tb02451.x&gt;
Weir BS, Cockerham CC (1984) Estimating F Statistics for the ANalysis of Population Structure. Evolution 38, 1358-1370. &lt;doi:10.2307/2408641&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># import genotype data and convert to allele frequecies
data(potato.mini, package="StAMPP")
potato.freq &lt;- stamppConvert(potato.mini, "r")
# Calculate pairwise Fst values between each population
potato.fst &lt;- stamppFst(potato.freq, 100, 95, 1)
</code></pre>

<hr>
<h2 id='stamppGmatrix'>Genomic Relationship Calculation</h2><span id='topic+stamppGmatrix'></span>

<h3>Description</h3>

<p>This function calculates a genomic relationship matrix following the method decribed by Yang et al (2010)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stamppGmatrix(geno)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stamppGmatrix_+3A_geno">geno</code></td>
<td>
<p>a data frame containing allele frequency data generated from stamppConvert, or a genlight object containing genotype data, individual IDs, population IDs and ploidy levels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class matrix which contains the genomic relationship values between each individual
</p>


<h3>Author(s)</h3>

<p>Luke Pembleton &lt;luke.pembleton at agriculture.vic.gov.au&gt;
</p>


<h3>References</h3>

<p>Yang J, Benyamin B, McEvoy BP, et al (2010) Common SNPs explain a large proportion of the heritability for human height. Nat Genet 42, 565-569. &lt;doi:10.1038/ng.608&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># import genotype data and convert to allele frequecies
data(potato.mini, package="StAMPP")
potato.freq &lt;- stamppConvert(potato.mini, "r")
# Calculate genomic relationship values between each individual
potato.fst &lt;- stamppGmatrix(potato.freq)
</code></pre>

<hr>
<h2 id='stamppNeisD'>Genetic Distance Calculation</h2><span id='topic+stamppNeisD'></span>

<h3>Description</h3>

<p>This function calculates Nei's genetic distance (Nei 1972) between populations or individuals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stamppNeisD(geno, pop = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stamppNeisD_+3A_geno">geno</code></td>
<td>
<p>a data frame containing allele frequency data generated from stamppConvert, or a genlight object containing genotype data, individual IDs, population IDs and ploidy levels</p>
</td></tr>
<tr><td><code id="stamppNeisD_+3A_pop">pop</code></td>
<td>
<p>logical. True if genetic distance should be calculated between populations, false if it should be calculated between individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A object of class matrix which contains the genetic distance between each population or individual
</p>


<h3>Author(s)</h3>

<p>Luke Pembleton &lt;luke.pembleton at agriculture.vic.gov.au&gt;
</p>


<h3>References</h3>

<p>Nei M (1972) Genetic Distance between Populations. The American Naturalist 106, 283-292.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># import genotype data and convert to allele frequecies
data(potato.mini, package="StAMPP")
potato.freq &lt;- stamppConvert(potato.mini, "r")
# Calculate genetic distance between individuals
potato.D.ind &lt;- stamppNeisD(potato.freq, FALSE)
# Calculate genetic distance between populations
potato.D.pop &lt;- stamppNeisD(potato.freq, TRUE)
</code></pre>

<hr>
<h2 id='stamppPhylip'>Export to Phylip Format</h2><span id='topic+stamppPhylip'></span>

<h3>Description</h3>

<p>Converts the genetic distance matrix generated with stamppNeisD into Phylip format and exports it as a text file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stamppPhylip(distance.mat, file = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stamppPhylip_+3A_distance.mat">distance.mat</code></td>
<td>
<p>the matrix containing the genetic distances generated from stamppNeisD to be converted into Phylip format</p>
</td></tr>
<tr><td><code id="stamppPhylip_+3A_file">file</code></td>
<td>
<p>the file path and name to save the Phylip format matrix as</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The exported Phylip formated text file can be easily imported into sofware packages such as DARWin (Perrier &amp; Jacquemound-Collet 2006) to be used to generate neighbour joining trees
</p>


<h3>Author(s)</h3>

<p>Luke Pembleton &lt;luke.pembleton at agriculture.vic.gov.au&gt;
</p>


<h3>References</h3>

<p>Perrier X, Jacquemound-Collet JP (2006) DARWin - Dissimilarity Analysis and Representation for Windows. Agricultural Research for Development
</p>


<h3>Examples</h3>

<pre><code class='language-R'># import genotype data and convert to allele frequecies
data(potato.mini, package="StAMPP")
potato.freq &lt;- stamppConvert(potato.mini, "r")
# Calculate genetic distance between populations
potato.D.pop &lt;- stamppNeisD(potato.freq, TRUE)
# Export the genetic distance matrix in Phylip format
## Not run: stamppPhylip(potato.D.pop, file="potato_distance.txt")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
