<!DOCTYPE html><html><head><title>Help for package statgenQTLxT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {statgenQTLxT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#covUnstr'><p>Compute unstructured covariance</p></a></li>
<li><a href='#EMFA'><p>Factor analytic variation of EM algoritm</p></a></li>
<li><a href='#estEffsCPP'><p>Estimates for covariates</p></a></li>
<li><a href='#estEffTot'><p>Helper function for estimating effect</p></a></li>
<li><a href='#exclMarkers'><p>Select markers to be excluded from GWAS scan.</p></a></li>
<li><a href='#gDataDropsRestr'><p>Subset of DROPS data for use in examples</p></a></li>
<li><a href='#LLQuadFormDiagCPP'><p>Compute tYPY as in Zhou and Stephens eqn. 50.</p></a></li>
<li><a href='#runMultiTraitGwas'><p>Perform multi-trait GWAS</p></a></li>
<li><a href='#statgenQTLxT-package'><p>statgenQTLxT: Multi-Trait and Multi-Trial Genome Wide Association Studies</p></a></li>
<li><a href='#updateFA'><p>Update W and P in EMFA algorithm</p></a></li>
<li><a href='#updateFAHomVar'><p>Update W and P in EMFA algorithm for homogeneous variance.</p></a></li>
<li><a href='#updatePrec'><p>Helper function for updating precision matrix.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multi-Trait and Multi-Trial Genome Wide Association Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-23</td>
</tr>
<tr>
<td>Description:</td>
<td>Fast multi-trait and multi-trail Genome Wide Association Studies 
    (GWAS) following the method described in Zhou and Stephens. (2014), 
    &lt;<a href="https://doi.org/10.1038%2Fnmeth.2848">doi:10.1038/nmeth.2848</a>&gt;. One of a series of statistical genetic packages 
    for streamlining the analysis of typical plant breeding experiments 
    developed by Biometris.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, foreach, Rcpp, sommer (&ge; 4.2.0), statgenGWAS (&ge;
1.0.9)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, tinytest</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-23 16:10:00 UTC; rossu027</td>
</tr>
<tr>
<td>Author:</td>
<td>Bart-Jan van Rossum
    <a href="https://orcid.org/0000-0002-8673-2514"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Willem Kruijer <a href="https://orcid.org/0000-0001-7179-1733"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Fred van Eeuwijk <a href="https://orcid.org/0000-0003-3672-2921"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Martin Boer <a href="https://orcid.org/0000-0002-1879-4588"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Daniela Bustos-Korts
    <a href="https://orcid.org/0000-0003-3827-6726"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Emilie J Millet <a href="https://orcid.org/0000-0002-2913-4892"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Joao Paulo <a href="https://orcid.org/0000-0002-4180-0763"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Maikel Verouden <a href="https://orcid.org/0000-0002-4893-3323"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Ron Wehrens <a href="https://orcid.org/0000-0002-8798-5599"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Choazhi Zheng <a href="https://orcid.org/0000-0001-6030-3933"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bart-Jan van Rossum &lt;bart-jan.vanrossum@wur.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-23 16:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='covUnstr'>Compute unstructured covariance</h2><span id='topic+covUnstr'></span><span id='topic+covPW'></span>

<h3>Description</h3>

<p>Compute unstructured covariance pairwise using <code>covPairwise</code> or using a
single model using <code>covUnstr</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covUnstr(Y, K, X = NULL, fixDiag = FALSE, VeDiag = FALSE)

covPW(Y, K, X = NULL, fixDiag = FALSE, corMat = FALSE, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covUnstr_+3A_y">Y</code></td>
<td>
<p>An n x p matrix of observed phenotypes, on p traits or trials
for n individuals. No missing values are allowed.</p>
</td></tr>
<tr><td><code id="covUnstr_+3A_k">K</code></td>
<td>
<p>An n x n kinship matrix.</p>
</td></tr>
<tr><td><code id="covUnstr_+3A_x">X</code></td>
<td>
<p>An n x c covariate matrix, c being the number of covariates and n
being the number of genotypes.</p>
</td></tr>
<tr><td><code id="covUnstr_+3A_fixdiag">fixDiag</code></td>
<td>
<p>Should the diagonal of the covariate matrix be fixed during
calculations? &ndash; NOT YET IMPLEMENTED</p>
</td></tr>
<tr><td><code id="covUnstr_+3A_vediag">VeDiag</code></td>
<td>
<p>Should Ve be a diagonal matrix?</p>
</td></tr>
<tr><td><code id="covUnstr_+3A_cormat">corMat</code></td>
<td>
<p>Should the output be a correlation matrix instead of a
covariance matrix?</p>
</td></tr>
<tr><td><code id="covUnstr_+3A_parallel">parallel</code></td>
<td>
<p>Should the computation of variance components be done in
parallel?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two matrices <code>Vg</code> and <code>Ve</code> containing genotypic
and environmental variance components respectively.
</p>

<hr>
<h2 id='EMFA'>Factor analytic variation of EM algoritm</h2><span id='topic+EMFA'></span>

<h3>Description</h3>

<p>Implementation of the factor analytic variation of the EM algoritm as
proposed by Dahl et al. (2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EMFA(
  y,
  k,
  size_param_x = NULL,
  cmHet = TRUE,
  dmHet = TRUE,
  tolerance = 1e-06,
  maxIter = 300L,
  size_param_cmStart = NULL,
  size_param_dmStart = NULL,
  mG = 1L,
  mE = 1L,
  maxDiag = 10000,
  stopIfDecreasing = TRUE,
  traits = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EMFA_+3A_y">y</code></td>
<td>
<p>An n x p matrix of observed phenotypes, on p traits or environments
for n individuals. No missing values are allowed.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_k">k</code></td>
<td>
<p>An n x n kinship matrix.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_size_param_x">size_param_x</code></td>
<td>
<p>An n x c covariate matrix, c being the number of
covariates and n being the number of genotypes. c has to be at least one
(typically an intercept). No missing values are allowed. If not provided a
vector of 1s is used.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_cmhet">cmHet</code></td>
<td>
<p>Should an extra diagonal part be added in the model for the
precision matrix Cm?</p>
</td></tr>
<tr><td><code id="EMFA_+3A_dmhet">dmHet</code></td>
<td>
<p>Should an extra diagonal part be added in the model for the
precision matrix Dm?</p>
</td></tr>
<tr><td><code id="EMFA_+3A_tolerance">tolerance</code></td>
<td>
<p>A numerical value. The iterating process stops if the
difference in conditional log-likelihood between two consecutive iterations
drops below tolerance.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_maxiter">maxIter</code></td>
<td>
<p>A numerical value for the maximum number of iterations.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_size_param_cmstart">size_param_cmStart</code></td>
<td>
<p>A p x p matrix containing starting values for the
precision matrix Cm.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_size_param_dmstart">size_param_dmStart</code></td>
<td>
<p>A p x p matrix containing starting values for the
precision matrix Dm.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_mg">mG</code></td>
<td>
<p>An integer. The order of the genetic part of the model.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_me">mE</code></td>
<td>
<p>An integer. The order of the environmental part of the model.</p>
</td></tr>
<tr><td><code id="EMFA_+3A_maxdiag">maxDiag</code></td>
<td>
<p>A numical value. The maximal value of the diagonal elements
in the precision matrices Cm and Dm (ignoring the low-rank part W W^t)</p>
</td></tr>
<tr><td><code id="EMFA_+3A_stopifdecreasing">stopIfDecreasing</code></td>
<td>
<p>Should the iterating process stop if after 50
iterations the log-likelihood decreases between two consecutive iterations?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components
</p>

<ul>
<li><p><code>Vg</code> The genetic variance components matrix.
</p>
</li>
<li><p><code>Ve</code> The environmental variance components matrix.
</p>
</li></ul>



<h3>References</h3>

<p>Dahl et al. (2013). Network inference in matrix-variate Gaussian
models with non-independent noise. arXiv preprint arXiv:1312.1622.
</p>
<p>Zhou, X. and Stephens, M. (2014). Efficient multivariate linear
mixed model algorithms for genome-wide association studies. Nature Methods,
February 2014, Vol. 11, p. 407–409
</p>

<hr>
<h2 id='estEffsCPP'>Estimates for covariates</h2><span id='topic+estEffsCPP'></span>

<h3>Description</h3>

<p>Compute the estimates and standard errors for the covariates in the input
matrix W.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estEffsCPP(
  y0,
  w0,
  x0,
  vg,
  ve,
  k,
  returnSe = TRUE,
  estCom = FALSE,
  nCores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estEffsCPP_+3A_y0">y0</code></td>
<td>
<p>An n x p matrix of observed phenotypes, on p traits or environments
for n genotypes. No missing values are allowed.</p>
</td></tr>
<tr><td><code id="estEffsCPP_+3A_w0">w0</code></td>
<td>
<p>An n x c covariate matrix, c being the number of covariates and n
being the number of genotypes. c has to be at least one (typically an
intercept). No missing values are allowed.</p>
</td></tr>
<tr><td><code id="estEffsCPP_+3A_x0">x0</code></td>
<td>
<p>An n x ns matrix of marker scores. Neither missing values nor
non-segregating markers are allowed.</p>
</td></tr>
<tr><td><code id="estEffsCPP_+3A_vg">vg</code></td>
<td>
<p>A p x p matrix of genetic covariances.</p>
</td></tr>
<tr><td><code id="estEffsCPP_+3A_ve">ve</code></td>
<td>
<p>A p x p matrix of environmental covariances.</p>
</td></tr>
<tr><td><code id="estEffsCPP_+3A_k">k</code></td>
<td>
<p>An n x n genetic relatedness matrix.</p>
</td></tr>
<tr><td><code id="estEffsCPP_+3A_returnse">returnSe</code></td>
<td>
<p>Should standard errors and p-values be returned?</p>
</td></tr>
<tr><td><code id="estEffsCPP_+3A_estcom">estCom</code></td>
<td>
<p>Should the common SNP-effect model be fitted?</p>
</td></tr>
<tr><td><code id="estEffsCPP_+3A_ncores">nCores</code></td>
<td>
<p>An integer indicating the number of cores used for parallel
computation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the estimates, optionally the standard errors of
the estimates and corresponding p-values. If <code>estCom = TRUE</code> also
common SNP-effects, their standard errors and corresponding p-values and
the p-values for QtlxE are output.
</p>


<h3>References</h3>

<p>Zhou, X. and Stephens, M. (2014). Efficient multivariate linear
mixed model algorithms for genome-wide association studies. Nature Methods,
February 2014, Vol. 11, p. 407–409
</p>

<hr>
<h2 id='estEffTot'>Helper function for estimating effect</h2><span id='topic+estEffTot'></span>

<h3>Description</h3>

<p>Function is a wrapper around estEffsCPP, useful for usage in chromosome
specific calculations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estEffTot(
  markers,
  X,
  Y,
  K,
  XRed,
  Vg,
  Ve,
  VgRed,
  VeRed,
  snpCov,
  allFreq,
  MAF,
  estCom,
  nCores = NULL
)
</code></pre>

<hr>
<h2 id='exclMarkers'>Select markers to be excluded from GWAS scan.</h2><span id='topic+exclMarkers'></span>

<h3>Description</h3>

<p>Helper function for selecting markers to be excluded from GWAS scan.
Markers are excluded if they are identical to any of the snpCovariates
(including the snpCovariates themselves).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exclMarkers(snpCov, markers, allFreq, ref = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exclMarkers_+3A_snpcov">snpCov</code></td>
<td>
<p>A character vector of snpCovariates.</p>
</td></tr>
<tr><td><code id="exclMarkers_+3A_markers">markers</code></td>
<td>
<p>A matrix with marker information.</p>
</td></tr>
<tr><td><code id="exclMarkers_+3A_allfreq">allFreq</code></td>
<td>
<p>A numerical vector of allele frequencies of the markers in
<code>markers</code>. This could be computed from markers as well but it is
needed in the general algorithm so to not redo things unnecessarily it is
not redone here.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical vector of markers to be exluded from the GWAS scan.
</p>

<hr>
<h2 id='gDataDropsRestr'>Subset of DROPS data for use in examples</h2><span id='topic+gDataDropsRestr'></span>

<h3>Description</h3>

<p>A <code><a href="statgenGWAS.html#topic+gData">gData</a></code> object based on a subset of the DROPS data
set used in the statgenGWAS package. The data is restricted to 3 traits and
10% (just over 4000) of the available markers. For a full description of the
data set see <code><a href="statgenGWAS.html#topic+dropsData">dropsData</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gDataDropsRestr
</code></pre>


<h3>Format</h3>

<p>An object of class <code>gData</code> of length 5.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.15454/IASSTN">doi:10.15454/IASSTN</a>
</p>


<h3>References</h3>

<p>Millet, E. J., Pommier, C., et al. (2019). A multi-site
experiment in a network of European fields for assessing the maize yield
response to environmental scenarios - Data set.
<a href="https://doi.org/10.15454/IASSTN">doi:10.15454/IASSTN</a>
</p>
<p>Ganal MW, et al. (2011) A Large Maize (Zea mays L.) SNP
Genotyping Array: Development and Germplasm Genotyping, and Genetic Mapping
to Compare with the B73 Reference Genome. PLoS ONE 6(12): e28334.
<a href="https://doi.org/10.1371/journal.pone.0028334">doi:10.1371/journal.pone.0028334</a>
</p>

<hr>
<h2 id='LLQuadFormDiagCPP'>Compute tYPY as in Zhou and Stephens eqn. 50.</h2><span id='topic+LLQuadFormDiagCPP'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">t(y) * P * y</code>, part of the log-likelihood functions from
equation 26 and 27 in Zhou and Stephens using equation 50. Equation 56, 57
and 58 are used to do the actual computations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LLQuadFormDiagCPP(y, vInv, size_param_x = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LLQuadFormDiagCPP_+3A_vinv">vInv</code></td>
<td>
<p>A n x p x p cube containing for each genotype l the
p x p matrix <code class="reqn">v_l ^ {-1}</code> (in the notation of Zhou and Stephens).</p>
</td></tr>
<tr><td><code id="LLQuadFormDiagCPP_+3A_size_param_x">size_param_x</code></td>
<td>
<p>An optional c x n covariate matrix, c being the number
of covariates and n being the number of genotypes. c has to be at least one
(typically an intercept). No missing values are allowed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is assumed that X and Y have already been rotated by Uk, where Uk is such
that the kinship matrix K equals <code class="reqn">K = Uk * Dk * t(Uk)</code>.<br />
The original X and Y are right multiplied by Uk, e.g. <code>Y &lt;- Y * Uk</code>.
See Zhou and Stephens (2014), supplement.<br />
It is these rotated versions that are the input of this function.
</p>


<h3>Value</h3>

<p>A numerical value for the <code class="reqn">t(y) * P * y</code> part of the
log-likelihood function.
</p>


<h3>References</h3>

<p>Zhou, X. and Stephens, M. (2014). Efficient multivariate linear
mixed model algorithms for genome-wide association studies. Nature Methods,
February 2014, Vol. 11, p. 407–409
</p>

<hr>
<h2 id='runMultiTraitGwas'>Perform multi-trait GWAS</h2><span id='topic+runMultiTraitGwas'></span>

<h3>Description</h3>

<p><code>runMultiTraitGwas</code> performs multi-trait or multi-environment Genome
Wide Association mapping on phenotypic and genotypic data contained in a
<code>gData</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runMultiTraitGwas(
  gData,
  trials = NULL,
  traits = NULL,
  covar = NULL,
  snpCov = NULL,
  kin = NULL,
  kinshipMethod = c("astle", "IBS", "vanRaden", "identity"),
  GLSMethod = c("single", "multi"),
  estCom = FALSE,
  useMAF = TRUE,
  MAF = 0.01,
  MAC = 10,
  genomicControl = FALSE,
  fitVarComp = TRUE,
  covModel = c("unst", "pw", "fa"),
  VeDiag = TRUE,
  maxIter = 2e+05,
  mG = 1,
  mE = 1,
  Vg = NULL,
  Ve = NULL,
  thrType = c("bonf", "fixed", "small", "fdr"),
  alpha = 0.05,
  LODThr = 4,
  nSnpLOD = 10,
  pThr = 0.05,
  rho = 0.4,
  sizeInclRegion = 0,
  minR2 = 0.5,
  parallel = FALSE,
  nCores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runMultiTraitGwas_+3A_gdata">gData</code></td>
<td>
<p>An object of class <code>gData</code> containing at least <code>map</code>,
<code>markers</code> and <code>pheno</code>. The latter should not contain missing
values. Multi-trait or multi-environment GWAS is performed for all variables
in <code>pheno</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_trials">trials</code></td>
<td>
<p>A vector specifying the environment on which to run GWAS.
This can be either a numeric index or a character name of a list item in
<code>pheno</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_traits">traits</code></td>
<td>
<p>A vector of traits on which to run GWAS. These can be either
numeric indices or character names of columns in <code>pheno</code>. If
<code>NULL</code>, GWAS is run on all traits.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_covar">covar</code></td>
<td>
<p>An optional vector of covariates taken into account when
running GWAS. These can be either numeric indices or character names of
columns in <code>covar</code> in <code>gData</code>. If <code>NULL</code>, no covariates are
used. An intercept is included automatically (and should not be assigned as
covariate). SNP-covariates should be assigned using the snpCov parameter.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_snpcov">snpCov</code></td>
<td>
<p>An optional character vector of SNP-names to be included as
covariates. SNP-names should match those used in <code>gData</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_kin">kin</code></td>
<td>
<p>An optional kinship matrix or list of kinship matrices. These
matrices can be from the <code>matrix</code> class as defined in the base package
or from the <code>dsyMatrix</code> class, the class of symmetric matrices in the
Matrix package.<br />
If <code>GLSMethod</code> = &quot;single&quot; then one matrix should be provided, if
<code>GLSMethod</code> = &quot;multi&quot;, a list of chromosome specific matrices of length
equal to the number of chromosomes in <code>map</code> in <code>gData</code>.<br />
If <code>NULL</code> then matrix <code>kinship</code> in <code>gData</code> is used. <br />
If both <code>kin</code> is provided and <code>gData</code> contains a matrix
<code>kinship</code> then <code>kin</code> is used.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_kinshipmethod">kinshipMethod</code></td>
<td>
<p>An optional character indicating the method used for
calculating the kinship matrix(ces). Currently &quot;astle&quot; (Astle and Balding,
2009), &quot;IBS&quot;, &quot;vanRaden&quot; (VanRaden, 2008), and &quot;identity&quot; are supported.
If a kinship matrix is supplied either in <code>gData</code> or in parameter
<code>kin</code>, <code>kinshipMethod</code> is ignored.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_glsmethod">GLSMethod</code></td>
<td>
<p>A character string indicating the method used to estimate
the marker effects. Either <code>single</code> for using a single kinship matrix,
or <code>multi</code> for using chromosome specific kinship matrices.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_estcom">estCom</code></td>
<td>
<p>Should the common SNP-effect model be fitted? If <code>TRUE</code>
not only the SNP-effects but also the common SNP-effect and QTL x E effect
are estimated.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_usemaf">useMAF</code></td>
<td>
<p>Should the minor allele frequency be used for selecting SNPs
for the analysis. If <code>FALSE</code>, the minor allele count is used instead.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_maf">MAF</code></td>
<td>
<p>The minor allele frequency (MAF) threshold used in GWAS. A
numerical value between 0 and 1. SNPs with MAF below this value are not taken
into account in the analysis, i.e. p-values and effect sizes are put to
missing (<code>NA</code>). Ignored if <code>useMAF</code> is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_mac">MAC</code></td>
<td>
<p>A numerical value. SNPs with minor allele count below this value
are not taken into account for the analysis, i.e. p-values and effect sizes
are set to missing (<code>NA</code>). Ignored if <code>useMAF</code> is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_genomiccontrol">genomicControl</code></td>
<td>
<p>Should genomic control correction as in Devlin and
Roeder (1999) be applied?</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_fitvarcomp">fitVarComp</code></td>
<td>
<p>Should the variance components be fitted? If <code>FALSE</code>,
they should be supplied in <code>Vg</code> and <code>Ve</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_covmodel">covModel</code></td>
<td>
<p>A character string indicating the covariance model for the
genetic background (Vg) and residual effects (Ve); see details.
Either <code>unst</code> for unstructured for both Vg and
Ve (as in Zhou and Stephens (2014)), <code>pw</code> for unstructered for both Vg
and Ve (pairwise, as in Furlotte and Eskin (2013)) or <code>fa</code> for
factor-analytic for both Vg and Ve.<br />
Ignored if <code>fitVarComp</code> = <code>FALSE</code></p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_vediag">VeDiag</code></td>
<td>
<p>Should there be environmental correlations if covModel = &quot;unst&quot;
or &quot;pw&quot;? If traits are measured on the same individuals, put <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_maxiter">maxIter</code></td>
<td>
<p>An integer for the maximum number of iterations. Only used
when <code>covModel = "fa"</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_mg">mG</code></td>
<td>
<p>An integer. The order of the genetic part of the factor analytic
model. Only used when <code>covModel = "fa"</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_me">mE</code></td>
<td>
<p>An integer. The order of the environmental part of the factor
analytic model. Only used when <code>covModel = "fa"</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_vg">Vg</code></td>
<td>
<p>An optional matrix with genotypic variance components. <code>Vg</code>
should have row and column names corresponding to the column names of
<code>gData$pheno</code>. It may contain additional rows and columns which will be
ignored. Ignored if fitVarComp = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_ve">Ve</code></td>
<td>
<p>An optional matrix with environmental variance components.
<code>Ve</code> should have row names column names corresponding to the column
names of <code>gData$pheno</code>. It may contain additional rows and columns
which will be ignored. Ignored if fitVarComp = <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_thrtype">thrType</code></td>
<td>
<p>A character string indicating the type of threshold used for
the selection of candidate loci. Either <code>bonf</code> for using the
Bonferroni threshold, a LOD-threshold of <code class="reqn">-log10(alpha/p)</code>, where p is
the number of markers and alpha can be specified in <code>alpha</code>,
<code>fixed</code> for a self-chosen fixed LOD-threshold, specified in
<code>LODThr</code> or <code>small</code>, the LOD-threshold is chosen such as the SNPs
with the <code>nSnpLOD</code> smallest p-values are selected. <code>nSnpLOD</code> can
be specified.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_alpha">alpha</code></td>
<td>
<p>A numerical value used for calculating the LOD-threshold for
<code>thrType</code> = &quot;bonf&quot; and the significant p-Values for <code>thrType</code> =
&quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_lodthr">LODThr</code></td>
<td>
<p>A numerical value used as a LOD-threshold when
<code>thrType</code> = &quot;fixed&quot;.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_nsnplod">nSnpLOD</code></td>
<td>
<p>A numerical value indicating the number of SNPs with the
smallest p-values that are selected when <code>thrType</code> = &quot;small&quot;.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_pthr">pThr</code></td>
<td>
<p>A numerical value just as the cut off value for p-Values for
<code>thrType</code> = &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_rho">rho</code></td>
<td>
<p>A numerical value used a the minimum value for SNPs to be
considered correlated when using <code>thrType</code> = &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_sizeinclregion">sizeInclRegion</code></td>
<td>
<p>An integer. Should the results for SNPs close to
significant SNPs be included? If so, the size of the region in centimorgan
or base pairs. Otherwise 0.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_minr2">minR2</code></td>
<td>
<p>A numerical value between 0 and 1. Restricts the SNPs included
in the region close to significant SNPs to only those SNPs that are in
sufficient Linkage Disequilibrium (LD) with the significant snp, where LD
is measured in terms of <code class="reqn">R^2</code>. If for example <code>sizeInclRegion</code> =
200000 and <code>minR2</code> = 0.5, then for every significant SNP also those SNPs
whose LD (<code class="reqn">R^2</code>) with the significant SNP is at least 0.5 AND which are
at most 200000 away from this significant snp are included. Ignored if
<code>sizeInclRegion</code> = 0.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_parallel">parallel</code></td>
<td>
<p>Should the computation of variance components be done in
parallel? Only used if <code>covModel = "pw"</code>. A parallel computing
environment has to be setup by the user.</p>
</td></tr>
<tr><td><code id="runMultiTraitGwas_+3A_ncores">nCores</code></td>
<td>
<p>A numerical value indicating the number of cores to be used by
the parallel part of the algorithm. If <code>NULL</code> the number of cores used
will be equal to the number of cores available on the machine - 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>GWAS</code>.
</p>


<h3>Details</h3>

<p>runMultiTraitGwas estimates the effect of a SNP in
different trials or on different traits, one SNP at a time. Genetic
and residual covariances are fitted only once, for a model without SNPs.
Following the diagonalization scheme of Zhou and Stephens (2014), the
following model is fit
</p>
<p><code class="reqn">Y = \left(\begin{array}{c} Y_1 \\ \vdots \\ Y_p\end{array}\right) =
\left(\begin{array}{c} X_1\gamma_1 \\ \vdots \\ X_p\gamma_p\end{array}\right)
+ \left(\begin{array}{c} x_1\beta_1 \\ \vdots \\ x_p\beta_p\end{array}\right)
+ \left(\begin{array}{c} G_1 \\ \vdots \\ G_p\end{array}\right) +
\left(\begin{array}{c} E_1 \\ \vdots \\ E_p\end{array}\right)</code>
</p>
<p>where <code class="reqn">Y</code> is a <code class="reqn">np \times 1</code> vector of phenotypic values for <code class="reqn">n</code>
genotypes and <code class="reqn">p</code> traits or trials. <code class="reqn">x</code> is the <code class="reqn">n \times 1</code>
vector of scores for the marker under consideration, and <code class="reqn">X</code> the
<code class="reqn">n \times q</code> design matrix for the other covariates. By default only a
trait (environment) specific intercept is included. The vector of genetic
background effects
(<code class="reqn">\left(\begin{array}{c}G_1 \\ \vdots \\ G_p\end{array}\right)</code>) is
Gaussian with zero mean and covariance <code class="reqn">V_g \otimes K</code>, where <code class="reqn">V_g</code>
is a <code class="reqn">p \times p</code> matrix of genetic (co)variances, and <code class="reqn">K</code> an
<code class="reqn">n \times n</code> kinship matrix. Similarly, the residual errors
(<code class="reqn">\left(\begin{array}{c}E_1 \\ \vdots \\ E_p\end{array}\right)</code>)
have covariance <code class="reqn">V_e \otimes I_n</code>, for a <code class="reqn">p \times p</code> matrix
<code class="reqn">V_e</code> of residual (co)variances.
</p>


<h3>Hypotheses for the SNP-effects</h3>

<p>For each SNP, the null-hypothesis <code class="reqn">\beta_1 = \dots = \beta_p = 0</code> is
tested, using the likelihood ratio test (LRT) described in Zhou and
Stephens (2014). If <code>estCom = TRUE</code>, additional tests for a common
effect and for QTL x E are performed, using the parameterization
<code class="reqn">\beta_j = \alpha + \alpha_j (1 \leq j \leq p)</code>.
As in Korte et al (2012), we use likelihood ratio tests, but not restricted
to the bivariate case. For the common effect, we fit the reduced
model <code class="reqn">\beta_j = \alpha</code>, and test if <code class="reqn">\alpha = 0</code>.
For QTL-by-environment interaction, we test if <code class="reqn">\alpha_1 = \dots =
\alpha_p = 0</code>.
</p>


<h3>Models for the genetic and residual covariance</h3>

<p><code class="reqn">V_g</code> and <code class="reqn">V_e</code> can be provided by the user
(<code>fitVarComp = FALSE</code>);
otherwise one of the following models is used, depending on covModel.
If <code>covModel = "unst"</code>, an unstructured model is assumed, as in Zhou and
Stephens (2014): <code class="reqn">V_g</code> and <code class="reqn">V_e</code> can be any positive-definite matrix,
requiring a total of <code class="reqn">p(p + 1)/2</code> parameters per matrix.
If <code>covModel = "fa"</code>, a factor-analytic model is fitted using an
EM-algorithm, as in Millet et al (2016). <code class="reqn">V_g</code> and <code class="reqn">V_e</code> are assumed
to be of the form <code class="reqn">W W^t + D</code>, where <code class="reqn">W</code> is a <code class="reqn">p \times m</code> matrix
of factor loadings and <code class="reqn">D</code> a diagonal matrix with trait or environment
specific values. <code class="reqn">m</code> is the order of the model, and the parameters
<code>mG</code> and <code>mE</code> specify the order used for respectively <code class="reqn">V_g</code>
and <code class="reqn">V_e</code>. <code>maxIter</code> sets the maximum number of iterations used
in the EM-algorithm.
Finally, if <code>covModel = "pw"</code>, <code class="reqn">V_g</code> and <code class="reqn">V_e</code> are estimated
'pairwise', as in Furlotte and Eskin (2015). Looping over pairs of traits
or trials <code class="reqn">1 \leq j &lt; k \leq p</code>,
<code class="reqn">V_g[j,k] = V_g[k,j]</code> and <code class="reqn">V_e[j,k] = V_e[k,j]</code>
are estimated assuming a bivariate mixed model. The diagonals of
<code class="reqn">V_g</code> and <code class="reqn">V_e</code> are fitted assuming univariate mixed models. If the
resulting <code class="reqn">V_g</code> or <code class="reqn">V_e</code> is not positive-definite, they are
replaced by the nearest positive-definite matrix.
In case <code>covModel = "unst"</code> or <code>"pw"</code> it is possible to assume
that <code class="reqn">V_e</code> is diagonal (<code>VeDiag = TRUE</code>)
</p>


<h3>References</h3>

<p>Dahl et al. (2013). Network inference in matrix-variate Gaussian
models with non-independent noise. arXiv preprint arXiv:1312.1622.
</p>
<p>Furlotte, N.A. and Eskin, E. (2015). Efficient multiple-trait
association and estimation of genetic correlation using the matrix-variate
linear mixed model. Genetics, May 2015, Vol.200-1, p. 59-68.
</p>
<p>Korte et al. (2012). A mixed-model approach for genome-wide
association studies of correlated traits in structured populations.
Nature Genetics, 44(9), 1066–1071. <a href="https://doi.org/10.1038/ng.2376">doi:10.1038/ng.2376</a>
</p>
<p>Millet et al. (2016). Genome-wide analysis of yield in Europe:
allelic effects as functions of drought and heat scenarios. Plant Physiology,
pp.00621.2016. <a href="https://doi.org/10.1104/pp.16.00621">doi:10.1104/pp.16.00621</a>
</p>
<p>Thoen et al. (2016). Genetic architecture of plant stress
resistance: multi-trait genome-wide association mapping. New Phytologist,
213(3), 1346–1362. <a href="https://doi.org/10.1111/nph.14220">doi:10.1111/nph.14220</a>
</p>
<p>Zhou, X. and Stephens, M. (2014). Efficient multivariate linear
mixed model algorithms for genome-wide association studies. Nature Methods,
February 2014, Vol. 11, p. 407–409.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## First create a gData object.
## See the vignette for a detailed description.
## Here we use the gData object included in the package

## Run multi-trait GWAS
## Use a factor analytic model to estimate variance components.

mtg0 &lt;- runMultiTraitGwas(gDataDropsRestr,
                         trial = "Mur13W",
                         covModel = "fa")


## Plot the results.
## For details on the different plots see plot.GWAS

plot(mtg0, plotType = "qq")
plot(mtg0, plotType = "manhattan")
plot(mtg0, plotType = "qtl", yThr = 3.5)


## Run multi-trait GWAS
## Use a pairwise model to estimate variance components.
## Estimate common effects and set a fixed threshold for significant SNPs

mtg1 &lt;- runMultiTraitGwas(gDataDropsRestr,
                         trial = "Mur13W",
                         covModel = "pw",
                         estCom = TRUE,
                         thrType = "fixed",
                         LODThr = 3)


## Run multi-trait GWAS
## Use an unstructured model to estimate variance components.
## Identify the 5 SNPs with smallest p-values as significant SNPs.
## Compute the kinship matrix using the vanRaden method.

mtg2 &lt;- runMultiTraitGwas(gDataDropsRestr,
                         trial = "Mur13W",
                         kinshipMethod = "vanRaden",
                         covModel = "unst",
                         thrType = "small",
                         nSnpLOD = 5)


</code></pre>

<hr>
<h2 id='statgenQTLxT-package'>statgenQTLxT: Multi-Trait and Multi-Trial Genome Wide Association Studies</h2><span id='topic+statgenQTLxT'></span><span id='topic+statgenQTLxT-package'></span>

<h3>Description</h3>

<p>Fast multi-trait and multi-trail Genome Wide Association Studies (GWAS) following the method described in Zhou and Stephens. (2014), <a href="https://doi.org/10.1038/nmeth.2848">doi:10.1038/nmeth.2848</a>. One of a series of statistical genetic packages for streamlining the analysis of typical plant breeding experiments developed by Biometris.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Bart-Jan van Rossum <a href="mailto:bart-jan.vanrossum@wur.nl">bart-jan.vanrossum@wur.nl</a> (<a href="https://orcid.org/0000-0002-8673-2514">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Willem Kruijer (<a href="https://orcid.org/0000-0001-7179-1733">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Fred van Eeuwijk (<a href="https://orcid.org/0000-0003-3672-2921">ORCID</a>) [contributor]
</p>
</li>
<li><p> Martin Boer (<a href="https://orcid.org/0000-0002-1879-4588">ORCID</a>) [contributor]
</p>
</li>
<li><p> Daniela Bustos-Korts (<a href="https://orcid.org/0000-0003-3827-6726">ORCID</a>) [contributor]
</p>
</li>
<li><p> Emilie J Millet (<a href="https://orcid.org/0000-0002-2913-4892">ORCID</a>) [contributor]
</p>
</li>
<li><p> Joao Paulo (<a href="https://orcid.org/0000-0002-4180-0763">ORCID</a>) [contributor]
</p>
</li>
<li><p> Maikel Verouden (<a href="https://orcid.org/0000-0002-4893-3323">ORCID</a>) [contributor]
</p>
</li>
<li><p> Ron Wehrens (<a href="https://orcid.org/0000-0002-8798-5599">ORCID</a>) [contributor]
</p>
</li>
<li><p> Choazhi Zheng (<a href="https://orcid.org/0000-0001-6030-3933">ORCID</a>) [contributor]
</p>
</li></ul>


<hr>
<h2 id='updateFA'>Update W and P in EMFA algorithm</h2><span id='topic+updateFA'></span>

<h3>Description</h3>

<p>Update W and P used in the iteration process in the EMFA algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateFA(
  y,
  wStart,
  pStart,
  wNew,
  pNew,
  m0,
  hetVar = FALSE,
  maxDiag = 10000,
  tolerance = 1e-04,
  maxIter = 100L,
  printProgress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateFA_+3A_y">y</code></td>
<td>
<p>An n x p matrix or data.frame.</p>
</td></tr>
<tr><td><code id="updateFA_+3A_wstart">wStart</code></td>
<td>
<p>A p x p matrix or data.frame containing starting values for W.</p>
</td></tr>
<tr><td><code id="updateFA_+3A_pstart">pStart</code></td>
<td>
<p>A p x p matrix or data.frame containing starting values for P.</p>
</td></tr>
<tr><td><code id="updateFA_+3A_m0">m0</code></td>
<td>
<p>An integer. The order of the model.</p>
</td></tr>
<tr><td><code id="updateFA_+3A_hetvar">hetVar</code></td>
<td>
<p>Should an extra diagonal part be added in the model for the
precision matrix?</p>
</td></tr>
<tr><td><code id="updateFA_+3A_maxdiag">maxDiag</code></td>
<td>
<p>A numerical value for the maximum value of the diagonal of P.</p>
</td></tr>
<tr><td><code id="updateFA_+3A_tolerance">tolerance</code></td>
<td>
<p>A numerical value. The iterating process stops if the sum
of the difference for P and W between two steps gets lower than this value.</p>
</td></tr>
<tr><td><code id="updateFA_+3A_maxiter">maxIter</code></td>
<td>
<p>A numerical value for the maximum number of iterations.</p>
</td></tr>
<tr><td><code id="updateFA_+3A_printprogress">printProgress</code></td>
<td>
<p>Should progress be printed during iterations?</p>
</td></tr>
</table>

<hr>
<h2 id='updateFAHomVar'>Update W and P in EMFA algorithm for homogeneous variance.</h2><span id='topic+updateFAHomVar'></span>

<h3>Description</h3>

<p>Update W and P used in the iteration process in the EMFA algorithm in case
the variance is homogeneous.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateFAHomVar(s, wNew, pNew, m, maxDiag = 10000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateFAHomVar_+3A_s">s</code></td>
<td>
<p>A p x p sample covariance matrix.</p>
</td></tr>
<tr><td><code id="updateFAHomVar_+3A_m">m</code></td>
<td>
<p>An integer. The order of the model.</p>
</td></tr>
<tr><td><code id="updateFAHomVar_+3A_maxdiag">maxDiag</code></td>
<td>
<p>A numerical value for the maximum value of sigma2.</p>
</td></tr>
</table>

<hr>
<h2 id='updatePrec'>Helper function for updating precision matrix.</h2><span id='topic+updatePrec'></span>

<h3>Description</h3>

<p>Helper function for updating the precision matrices in the EMFA algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updatePrec(m, nc, omega, w, p, wNew, pNew, cNew, het, maxDiag)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updatePrec_+3A_m">m</code></td>
<td>
<p>An integer, the order of the model.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_nc">nc</code></td>
<td>
<p>An integer, the number of traits or genotypes.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_omega">omega</code></td>
<td>
<p>A computed matrix for the current step in the algoritm.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_w">w</code></td>
<td>
<p>A model matrix for the current step in the algorithm.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_p">p</code></td>
<td>
<p>A model matrix for the current step in the algorithm.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_wnew">wNew</code></td>
<td>
<p>A pointer to the updated model matrix for w.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_pnew">pNew</code></td>
<td>
<p>A pointer to the updated model matrix for p.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_cnew">cNew</code></td>
<td>
<p>A pointer to the updated matrix c.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_het">het</code></td>
<td>
<p>Should an extra diagonal part be added in the model for the
precision matrix.</p>
</td></tr>
<tr><td><code id="updatePrec_+3A_maxdiag">maxDiag</code></td>
<td>
<p>A numerical value for the maximum value of sigma2.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
