<!DOCTYPE html><html><head><title>Help for package statgenGWAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {statgenGWAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#codeMarkers'><p>Code and impute markers</p></a></li>
<li><a href='#dropsData'><p>DROPS data sets</p></a></li>
<li><a href='#gData'><p>S3 Class gData</p></a></li>
<li><a href='#kinship'><p>Functions for calculating kinship matrices</p></a></li>
<li><a href='#plot.gData'><p>Plot function for the class <code>gData</code></p></a></li>
<li><a href='#plot.GWAS'><p>Plot function for the class <code>GWAS</code></p></a></li>
<li><a href='#readPLINK'><p>Read PLINK binary data</p></a></li>
<li><a href='#runSingleTraitGwas'><p>Perform single-trait GWAS</p></a></li>
<li><a href='#statgenGWAS-package'><p>statgenGWAS: Genome Wide Association Studies</p></a></li>
<li><a href='#summary.gData'><p>Summary function for the class <code>gData</code></p></a></li>
<li><a href='#summary.GWAS'><p>Summary function for the class <code>GWAS</code></p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome Wide Association Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-10-13</td>
</tr>
<tr>
<td>Description:</td>
<td>Fast single trait Genome Wide Association Studies (GWAS) following 
    the method described in Kang et al. (2010), &lt;<a href="https://doi.org/10.1038%2Fng.548">doi:10.1038/ng.548</a>&gt;.        
    One of a series of statistical genetic packages for streamlining the 
    analysis of typical plant breeding experiments developed by Biometris.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, ggplot2 (&ge; 3.0.0), methods, sommer (&ge; 3.7.3),
Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, officer, tinytest, snpStats</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://biometris.github.io/statgenGWAS/index.html">https://biometris.github.io/statgenGWAS/index.html</a>,
<a href="https://github.com/Biometris/statgenGWAS/">https://github.com/Biometris/statgenGWAS/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Biometris/statgenGWAS/issues">https://github.com/Biometris/statgenGWAS/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-13 14:49:17 UTC; rossu027</td>
</tr>
<tr>
<td>Author:</td>
<td>Bart-Jan van Rossum
    <a href="https://orcid.org/0000-0002-8673-2514"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Willem Kruijer <a href="https://orcid.org/0000-0001-7179-1733"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Fred van Eeuwijk <a href="https://orcid.org/0000-0003-3672-2921"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Martin Boer <a href="https://orcid.org/0000-0002-1879-4588"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Marcos Malosetti <a href="https://orcid.org/0000-0002-8150-1397"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Daniela Bustos-Korts
    <a href="https://orcid.org/0000-0003-3827-6726"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Emilie Millet <a href="https://orcid.org/0000-0002-2913-4892"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Joao Paulo <a href="https://orcid.org/0000-0002-4180-0763"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Maikel Verouden <a href="https://orcid.org/0000-0002-4893-3323"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Ron Wehrens <a href="https://orcid.org/0000-0002-8798-5599"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Choazhi Zheng <a href="https://orcid.org/0000-0001-6030-3933"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bart-Jan van Rossum &lt;bart-jan.vanrossum@wur.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-13 15:30:43 UTC</td>
</tr>
</table>
<hr>
<h2 id='codeMarkers'>Code and impute markers</h2><span id='topic+codeMarkers'></span>

<h3>Description</h3>

<p><code>codeMarkers</code> codes markers in a <code>gData</code> object and optionally
performs imputation of missing values as well.<br />
The function performs the following steps:<br />
</p>

<ol>
<li><p>replace strings in <code>naStrings</code> by <code>NA</code>.
</p>
</li>
<li><p>remove genotypes with a fraction of missing values higher than
<code>nMissGeno</code>.
</p>
</li>
<li><p>remove SNPs with a fraction of missing values higher than
<code>nMiss</code>.
</p>
</li>
<li><p>recode SNPs to numerical values.
</p>
</li>
<li><p>remove SNPs with a minor allele frequency lower than <code>MAF</code>.
</p>
</li>
<li><p>optionally remove duplicate SNPs.
</p>
</li>
<li><p>optionally impute missing values.
</p>
</li>
<li><p>repeat steps 5. and 6. if missing values are imputed.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>codeMarkers(
  gData,
  refAll = "minor",
  nMissGeno = 1,
  nMiss = 1,
  MAF = NULL,
  MAC = NULL,
  removeDuplicates = TRUE,
  keep = NULL,
  impute = TRUE,
  imputeType = c("random", "fixed", "beagle"),
  fixedValue = NULL,
  naStrings = NA,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="codeMarkers_+3A_gdata">gData</code></td>
<td>
<p>An object of class <code>gData</code> containing at least
<code>markers</code>.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_refall">refAll</code></td>
<td>
<p>A character string indicating the reference allele used when
recoding markers.<br />
If &quot;minor&quot;, then the recoding is done using the minor allele as reference
allele. Alternatively a single character can be supplied as a reference
allele for the whole set of SNPs, or a character vector with a reference
allele per SNP.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_nmissgeno">nMissGeno</code></td>
<td>
<p>A numerical value between 0 and 1. Genotypes with a
fraction of missing values higher than <code>nMissGeno</code> will be removed.
Genotypes with only missing values will always be removed.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_nmiss">nMiss</code></td>
<td>
<p>A numerical value between 0 and 1. SNPs with a fraction of
missing values higher than <code>nMiss</code> will be removed. SNPs with only
missing values will always be removed.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_maf">MAF</code></td>
<td>
<p>A numerical value between 0 and 1. SNPs with a Minor Allele
Frequency (MAF) below this value will be removed. Only one of <code>MAF</code>
and <code>MAC</code> may be specified.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_mac">MAC</code></td>
<td>
<p>A numerical value. SNPs with Minor Allele Count (MAC) below this
value will be removed. Only one of <code>MAF</code> and <code>MAC</code> may be
specified.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_removeduplicates">removeDuplicates</code></td>
<td>
<p>Should duplicate SNPs be removed?</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_keep">keep</code></td>
<td>
<p>A vector of SNPs that should never be removed in the whole
process.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_impute">impute</code></td>
<td>
<p>Should imputation of missing values be done?</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_imputetype">imputeType</code></td>
<td>
<p>A character string indicating what kind of imputation of
values should be done.<br />
</p>

<ul>
<li><p>fixed - missing values will be replaced by a given fixed value.
</p>
</li>
<li><p>random - missing values will be replaced by a random value calculated
using allele frequencies per SNP.
</p>
</li>
<li><p>beagle - missing values will be imputed using beagle software, version
5.2. Beagle only accepts integers as map positions. If you use this option,
please cite the original papers in your publication (see references).
</p>
</li></ul>
</td></tr>
<tr><td><code id="codeMarkers_+3A_fixedvalue">fixedValue</code></td>
<td>
<p>A numerical value used for replacing missing values in
case <code>inputType</code> is fixed.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_nastrings">naStrings</code></td>
<td>
<p>A character vector of strings to be treated as NA.</p>
</td></tr>
<tr><td><code id="codeMarkers_+3A_verbose">verbose</code></td>
<td>
<p>Should a summary of the performed steps be printed?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A copy of the input <code>gData</code> object with markers replaced by
coded and imputed markers.
</p>


<h3>References</h3>

<p>S R Browning and B L Browning (2007) Rapid and accurate haplotype
phasing and missing data inference for whole genome association studies by
use of localized haplotype clustering. Am J Hum Genet 81:1084-1097.
<a href="https://doi.org/10.1086/521987">doi:10.1086/521987</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create markers
markers &lt;- matrix(c(
"AA",   "AB",   "AA",   "BB",   "BA",   "AB",   "AA",   "AA",   NA,  "AA",
"AA",   "AA",   "BB",   "BB",   "AA",   "AA",   "BB",   "AA",   NA,  "AA",
"AA",   "BA",   "AB",   "BB",   "AB",   "AB",   "AA",   "BB",   NA,  "AA",
"AA",   "AA",   "BB",   "BB",   "AA",   "AA",   "AA",   "AA",   NA,  "AA",
"AA",   "AA",   "BB",   "BB",   "AA",   "BB",   "BB",   "BB",  "AB", "AA",
"AA",   "AA",   "BB",   "BB",   "AA",    NA,    "BB",   "AA",   NA,  "AA",
"AB",   "AB",   "BB",   "BB",   "BB",   "AA",   "BB",   "BB",   NA,  "AB",
"AA",   "AA",    NA,    "BB",    NA,    "AA",   "AA",   "AA",  "AA", "AA",
"AA",    NA,     NA,    "BB",   "BB",   "BB",   "BB",   "BB",  "AA", "AA",
"AA",    NA,    "AA",   "BB",   "BB",   "BB",   "AA",   "AA",   NA,  "AA"),
ncol = 10, byrow = TRUE, dimnames = list(paste0("IND", 1:10),
paste0("SNP", 1:10)))

## create object of class 'gData'.
gData &lt;- createGData(geno = markers)

## Code markers by minor allele, no imputation.
gDataCoded1 &lt;- codeMarkers(gData = gData, impute = FALSE)

## Code markers by reference alleles, impute missings by fixed value.
gDataCoded2 &lt;- codeMarkers(gData = gData,
                           refAll = rep(x = c("A", "B"), times =  5),
                           impute = TRUE, imputeType = "fixed",
                           fixedValue = 1)

## Code markers by minor allele, impute by random value.
gDataCoded3 &lt;- codeMarkers(gData = gData, impute = TRUE,
                           imputeType = "random")

</code></pre>

<hr>
<h2 id='dropsData'>DROPS data sets</h2><span id='topic+dropsData'></span><span id='topic+dropsMap'></span><span id='topic+dropsMarkers'></span><span id='topic+dropsPheno'></span>

<h3>Description</h3>

<p>This dataset comes from the European Union project DROPS (DROught-tolerant
yielding PlantS). A panel of 256 maize hybrids was grown with two water
regimes (irrigated or rainfed), in seven fields in 2012 and 2013,
respectively, spread along a climatic transect from western to eastern
Europe, plus one site in Chile in 2013. This resulted in 28 experiments
defined as the combination of one year, one site and one water regime, with
two and three repetitions for rainfed and irrigated treatments, respectively.
A detailed environmental characterisation was carried out, with hourly
records of micrometeorological data and soil water status, and associated
with precise measurement of phenology. Grain yield and its components were
measured at the end of the experiment.<br />
10 experiments have been selected from the full data set, two for each of
the five main environmental scenarios that were identified in the data. The
scenarios have been added to the data as well as a classification of the
genotypes in four genetic groups.<br />
The main purpose of this dataset
consists in using the environmental characterization to quantify the genetic
variability of maize grain yield in response to the environmental drivers
for genotype-by-environment interaction. For instance, allelic effects at
QTLs identified over the field network are consistent within a scenario but
largely differ between scenarios.<br /><br />
</p>
<p>The data is split in three separate data.frames.
</p>

<dl>
<dt><strong>dropsMarkers</strong></dt><dd><p>This data.frame contains the 50K genotyping
matrix coded in allelic dose (012) filtered and imputed. Genotyping of
41,722 loci on 246 parental lines were obtained using 50K Illumina Infinium
HD arrays (Ganal et al., 2011). Genotype were coded in allelic dose with 0
for the minor allele, 1 for the heterozygote, and 2 for the major allele.
Genotype were filtered (MAF &gt; 1%) and missing data imputed using Beagle v3.<br />
A data.frame with 246 rows and 41723 columns.
</p>

<dl>
<dt>Ind</dt><dd><p>name of the genotype</p>
</dd>
<dt>SYMN83 to PZE-110111485</dt><dd><p>coded QTLs</p>
</dd>
</dl>

</dd>
<dt><strong>dropsMap</strong></dt><dd><p>This data.frame contains the description of the
41,722 loci genotyped by 50K Illumina Infinium Array on the 246 lines.<br />
A data.frame with 41722 rows and 5 columns.
</p>

<dl>
<dt>SNP.names</dt><dd><p>name of the SNP</p>
</dd>
<dt>Chromosome</dt><dd><p>number of the B73 reference genome V2</p>
</dd>
<dt>Position</dt><dd><p>position on the B73 reference genome V2 in basepairs</p>
</dd>
<dt>allele1</dt><dd><p>first original allele (A, T, G or C)</p>
</dd>
<dt>allele2</dt><dd><p>second original allele (A, T, G or C)</p>
</dd>
</dl>

</dd>
<dt><strong>dropsPheno</strong></dt><dd><p>This data.frame contains the genotypic means
(Best Linear Unbiased Estimators, BLUEs), with one value per experiment
(Location × year × water regime) per genotype.<br />
A data.frame with 2460 rows and 19 columns.<br />
</p>

<dl>
<dt>Experiment</dt><dd><p>experiments ID described by the three first letters of the
city’s name followed by the year of experiment and the water regime with W
for watered and R for rain-fed.</p>
</dd>
<dt>parent1</dt><dd><p>identifier of donor dent line</p>
</dd>
<dt>Code_ID, Variety_ID, Accession_ID</dt><dd><p>identifier of the genotype</p>
</dd>
<dt>geno.panel</dt><dd><p>project in which the genetic material was generated</p>
</dd>
<dt>grain.yield</dt><dd><p>genotypic mean for yield adjusted at 15% grain moisture,
in ton per hectare (t ha^-1)</p>
</dd>
<dt>grain.number</dt><dd><p>genotypic mean for number of grain per square meter</p>
</dd>
<dt>grain.weight</dt><dd><p>genotypic mean for individual grain weight in milligram
(mg)</p>
</dd>
<dt>anthesis</dt><dd><p>genotypic mean for male flowering (pollen shed), in thermal
time cumulated since emergence (d_20°C)</p>
</dd>
<dt>silking</dt><dd><p>genotypic mean for female flowering (silking emergence), in
thermal time cumulated since emergence (d_20°C)</p>
</dd>
<dt>plant.height</dt><dd><p>genotypic mean for plant height, from ground level to
the base of the flag leaf (highest) leaf in centimeter (cm)</p>
</dd>
<dt>tassel.height</dt><dd><p>genotypic mean for plant height including tassel, from
ground level to the highest point of the tassel in centimeter (cm)</p>
</dd>
<dt>ear.height</dt><dd><p>genotypic mean for ear insertion height, from ground level
to ligule of the highest ear leaf in centimeter (cm)</p>
</dd>
<dt>year</dt><dd><p>year in which the experiment was performed</p>
</dd>
<dt>loc</dt><dd><p>location where the experiment was performed, a three letter
abbreviation</p>
</dd>
<dt>scenarioWater</dt><dd><p>water scenario for the experiment, well watered (WW) or
water deficit (WD)</p>
</dd>
<dt>scenarioTemp</dt><dd><p>temperature scenario for the experiment, Cool, Hot or
Hot(Day)</p>
</dd>
<dt>scenarioFull</dt><dd><p>the full scenario for the experiment, a combination of
scenarioWater and scenarioTemp</p>
</dd>
<dt>geneticGroup</dt><dd><p>the genetic group to which the genotype belongs</p>
</dd>
</dl>

</dd>
</dl>



<h3>Note</h3>

<p>From the source data, the experiments from 2011 have been removed since
these do not contain all genotypes. Also the experiment Gra13W has been
removed.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.15454/IASSTN">doi:10.15454/IASSTN</a>
</p>


<h3>References</h3>

<p>Millet, E. J., Pommier, C., et al. (2019). A multi-site
experiment in a network of European fields for assessing the maize yield
response to environmental scenarios - Data set.
<a href="https://doi.org/10.15454/IASSTN">doi:10.15454/IASSTN</a>
</p>
<p>Ganal MW, et al. (2011) A Large Maize (Zea mays L.) SNP
Genotyping Array: Development and Germplasm Genotyping, and Genetic Mapping
to Compare with the B73 Reference Genome. PLoS ONE 6(12): e28334.
<a href="https://doi.org/10.1371/journal.pone.0028334">doi:10.1371/journal.pone.0028334</a>
</p>

<hr>
<h2 id='gData'>S3 Class gData</h2><span id='topic+gData'></span><span id='topic+createGData'></span>

<h3>Description</h3>

<p><code>createGData</code> creates an object of S3 class gData with genotypic and
phenotypic data for usage in further analysis. All input to the function is
optional, however at least one input should be provided. It is possible to
provide an existing <code>gData</code> object as additional input in which case
data is added to this object. Existing data will be overwritten with a
warning.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createGData(
  gData = NULL,
  geno = NULL,
  map = NULL,
  kin = NULL,
  pheno = NULL,
  covar = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gData_+3A_gdata">gData</code></td>
<td>
<p>An optional gData object to be modified. If <code>NULL</code>, a new
gData object is created.</p>
</td></tr>
<tr><td><code id="gData_+3A_geno">geno</code></td>
<td>
<p>A matrix or data.frame with genotypes in the rows and markers in
the columns. A matrix from the <code>matrix</code> in the base package may be
provided as well as as matrix from the Matrix package.<br />
If no row names are provided, they are taken from <code>pheno</code> (if supplied
and dimension matches). If no column names are provided, the row names
from <code>map</code> are used (if supplied and dimension matches).</p>
</td></tr>
<tr><td><code id="gData_+3A_map">map</code></td>
<td>
<p>A data.frame with columns <code>chr</code> for chromosome and
<code>pos</code> for position. Positions can be in base pair (bp) or centimorgan (cM). They
should not be cumulative over the chromosomes. Other columns are ignored.
Marker names should be in the row names. These should match the marker names
in <code>geno</code> (if supplied).</p>
</td></tr>
<tr><td><code id="gData_+3A_kin">kin</code></td>
<td>
<p>A kinship matrix or list of kinship matrices with genotype in
rows and colums. These matrices can be from the <code>matrix</code> class, as
defined in the base package, or from the <code>dsyMatrix</code> class, the class
of symmetric matrices in the Matrix package.<br />
The genotypes should be identical to the genotypes in <code>geno</code>.<br />
If a list of kinship matrices is provided these are supposed to be
chromosome specific matrices. In that case their names should match
the names of the chromosomes in <code>map</code>. If no names are
provided, the number of matrices should match the number of chromosomes
in <code>map</code> in which case default names are provided.</p>
</td></tr>
<tr><td><code id="gData_+3A_pheno">pheno</code></td>
<td>
<p>A data.frame or a list of data.frames with phenotypic data,
with genotypes in the first column <code>genotype</code> and traits in the
following columns. The trait columns should be numerical columns only.
A list of data.frames can be used for replications, i.e. different
trials.</p>
</td></tr>
<tr><td><code id="gData_+3A_covar">covar</code></td>
<td>
<p>A data.frame with extra covariates per genotype. Genotypes
should be in the rows.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>gData</code> with the following components:
</p>
<table>
<tr><td><code>map</code></td>
<td>
<p>a data.frame containing map data. Map is sorted by
chromosome and position.</p>
</td></tr>
<tr><td><code>markers</code></td>
<td>
<p>a matrix containing marker information.</p>
</td></tr>
<tr><td><code>pheno</code></td>
<td>
<p>a list of data.frames containing phenotypic data.</p>
</td></tr>
<tr><td><code>kinship</code></td>
<td>
<p>a kinship matrix.</p>
</td></tr>
<tr><td><code>covar</code></td>
<td>
<p>a data.frame with extra covariates.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Bart-Jan van Rossum
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.gData">summary.gData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
## Create genotypic data.
geno &lt;- matrix(sample(x = c(0, 1, 2), size = 15, replace = TRUE), nrow = 3)
dimnames(geno) &lt;- list(paste0("G", 1:3), paste0("M", 1:5))

## Construct map.
map &lt;- data.frame(chr = c(1, 1, 2, 2, 2), pos = 1:5,
                  row.names = paste0("M", 1:5))

## Compute kinship matrix.
kin &lt;- kinship(X = geno, method = "IBS")

## Create phenotypic data.
pheno &lt;- data.frame(paste0("G", 1:3),
                    matrix(rnorm(n = 12, mean = 50, sd = 5), nrow = 3),
                    stringsAsFactors = FALSE)
dimnames(pheno) = list(paste0("G", 1:3), c("genotype", paste0("T", 1:4)))

## Combine all data in gData object.
gData &lt;- createGData(geno = geno, map = map, kin = kin, pheno = pheno)
summary(gData)

## Construct covariate.
covar &lt;- data.frame(C1 = c("a", "a", "b"), row.names = paste0("G", 1:3))

## Compute alternative kinship matrix.
kin2 &lt;- kinship(X = geno, method = "astle")

## Add covariates to previously created gData object and overwrite
## current kinship matrix by newly computed one.
gData2 &lt;- createGData(gData = gData, kin = kin2, covar = covar)

</code></pre>

<hr>
<h2 id='kinship'>Functions for calculating kinship matrices</h2><span id='topic+kinship'></span>

<h3>Description</h3>

<p>A collection of functions for calculating kinship matrices using different
algorithms. The following algorithms are included: astle (Astle and Balding,
2009), Identity By State (IBS) and VanRaden (VanRaden, 2008) for
marker matrices. For method identity an identity kinship matrix is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinship(
  X,
  method = c("astle", "IBS", "vanRaden", "identity"),
  MAF = NULL,
  denominator = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kinship_+3A_x">X</code></td>
<td>
<p>An n x m marker matrix with genotypes in the rows (n) and markers in
the columns (m).</p>
</td></tr>
<tr><td><code id="kinship_+3A_method">method</code></td>
<td>
<p>The method used for computing the kinship matrix.</p>
</td></tr>
<tr><td><code id="kinship_+3A_maf">MAF</code></td>
<td>
<p>The minor allele frequency (MAF) threshold used in kinship
computation. A numerical value between 0 and 1. SNPs with MAF below this
value are not taken into account when computing the kinship. If NULL all
markers are used regardless of their allele frequency.</p>
</td></tr>
<tr><td><code id="kinship_+3A_denominator">denominator</code></td>
<td>
<p>A numerical value. See details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An n x n kinship matrix.
</p>


<h3>Marker matrices</h3>

<p>In all algorithms the input matrix <code>X</code> is first cleaned, i.e. markers
with a variance of 0 are excluded from the calculation of the kinship matrix.
Then some form of scaling is done which differs per algorithm. This gives a
scaled matrix <code>Z</code>. The matrix <code class="reqn">ZZ^t / denominator</code> is returned.
By default the denominator is equal to the number of columns in <code>Z</code> for
<code>astle</code> and <code>IBS</code> and <code class="reqn">2 * p * (1-p)</code> where
<code class="reqn">p = colSums(X) / (2 * nrow(X))</code> for <code>vanRaden</code>. This denominator
can be overwritten by the user, e.g. when computing kinship matrices by
splitting <code>X</code> in smaller matrices and then adding the results together
in the end.
</p>


<h3>References</h3>

<p>Astle, William, and David J. Balding. 2009. “Population Structure
and Cryptic Relatedness in Genetic Association Studies.” Statistical Science
24 (4): 451–71. <a href="https://doi.org/10.1214/09-sts307">doi:10.1214/09-sts307</a>.
</p>
<p>VanRaden P.M. (2008) Efficient methods to compute genomic
predictions. Journal of Dairy Science 91 (11): 4414–23.
<a href="https://doi.org/10.3168/jds.2007-0980">doi:10.3168/jds.2007-0980</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create example matrix.
M &lt;- matrix(c(1, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1), nrow = 4)

## Compute kinship matrices using different methods.
kinship(M, method = "astle")
kinship(M, method = "IBS")
kinship(M, method = "vanRaden")

## Only use markers with a Minor Allele Frequency of 0.3 or more.
kinship(M, method = "astle", MAF = 0.3)

## Compute kinship matrix using astle and balding method with denominator 2.
kinship(M, method = "astle", denominator = 2)

</code></pre>

<hr>
<h2 id='plot.gData'>Plot function for the class <code>gData</code></h2><span id='topic+plot.gData'></span>

<h3>Description</h3>

<p>Creates a plot of the genetic map in an object of S3 class <code>gData</code>. A
plot of the genetic map showing the length of the chromosomes and the
positions of the markers in the genetic map is created.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gData'
plot(x, ..., highlight = NULL, title = NULL, output = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.gData_+3A_x">x</code></td>
<td>
<p>An object of class <code>gData</code>.</p>
</td></tr>
<tr><td><code id="plot.gData_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="plot.gData_+3A_highlight">highlight</code></td>
<td>
<p>A data.frame with at least columns chr and pos,
containing marker positions that should be highlighted in the plot. If a
column &quot;name&quot; is present that is used for annotation, otherwise the
highlighted markers are annotated as chr\@pos#'</p>
</td></tr>
<tr><td><code id="plot.gData_+3A_title">title</code></td>
<td>
<p>A character string, the title of the plot.</p>
</td></tr>
<tr><td><code id="plot.gData_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code>, only a ggplot object is invisibly returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>ggplot</code> is invisibly returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
## Create genotypic data.
geno &lt;- matrix(sample(x = c(0, 1, 2), size = 15, replace = TRUE), nrow = 3)
dimnames(geno) &lt;- list(paste0("G", 1:3), paste0("M", 1:5))

## Construct map.
map &lt;- data.frame(chr = c(1, 1, 2, 2, 2), pos = 1:5,
                  row.names = paste0("M", 1:5))

## Compute kinship matrix.
kin &lt;- kinship(X = geno, method = "IBS")

## Create phenotypic data.
pheno &lt;- data.frame(paste0("G", 1:3),
                    matrix(rnorm(n = 12, mean = 50, sd = 5), nrow = 3),
                    stringsAsFactors = FALSE)
dimnames(pheno) = list(paste0("G", 1:3), c("genotype", paste0("T", 1:4)))

## Combine all data in gData object.
gData &lt;- createGData(geno = geno, map = map, kin = kin, pheno = pheno)

## Plot genetic map.
plot(gData)

## Plot genetic map. Highlight first marker in map.
plot(gData, highlight = map[1, ])
 
</code></pre>

<hr>
<h2 id='plot.GWAS'>Plot function for the class <code>GWAS</code></h2><span id='topic+plot.GWAS'></span>

<h3>Description</h3>

<p>Creates a plot of an object of S3 class <code>GWAS</code>. The following types of
plot can be made:
</p>

<ul>
<li><p>a manhattan plot, i.e. a plot of LOD-scores per SNP
</p>
</li>
<li><p>a QQ plot of observed LOD-scores versus expected LOD-scores
</p>
</li>
<li><p>a qtl plot of effect sizes and directions for multiple traits
</p>
</li></ul>

<p>Manhattan plots and QQ plots are made for a single trait which
should be indicated using the parameter <code>trait</code>. If the analysis was
done for only one trait, it is detected automatically. The qtl plot will plot
all traits analyzed.<br />
See details for a detailed description of the plots and the plot options
specific to the different plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GWAS'
plot(
  x,
  ...,
  plotType = c("manhattan", "qq", "qtl"),
  trial = NULL,
  trait = NULL,
  title = NULL,
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.GWAS_+3A_x">x</code></td>
<td>
<p>An object of class <code>GWAS</code>.</p>
</td></tr>
<tr><td><code id="plot.GWAS_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed on to the actual plotting
functions.</p>
</td></tr>
<tr><td><code id="plot.GWAS_+3A_plottype">plotType</code></td>
<td>
<p>A character string indicating the type of plot to be made.
One of &quot;manhattan&quot;, &quot;qq&quot; and &quot;qtl&quot;.</p>
</td></tr>
<tr><td><code id="plot.GWAS_+3A_trial">trial</code></td>
<td>
<p>A character string or numeric index indicating for which
trial the plot should be made. If <code>x</code> only contains results for
one trial, <code>trial</code> may be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plot.GWAS_+3A_trait">trait</code></td>
<td>
<p>A character string indicating for which trait the results
should be plotted. For <code>type</code> &quot;qtl&quot; all traits are plotted. If <code>x</code>
only contains results for one trait, <code>trait</code> may be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plot.GWAS_+3A_title">title</code></td>
<td>
<p>A character string, the title of the plot.</p>
</td></tr>
<tr><td><code id="plot.GWAS_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code>, only a list of ggplot objects is invisibly returned.</p>
</td></tr>
</table>


<h3>Manhattan Plot</h3>

<p>A LOD-profile of all marker positions and corresponding LOD-scores is
plotted. Significant markers are highlighted with red dots. By default these
are taken from the result of the GWAS analysis however the LOD-threshold for
significant parameters may be modified using the parameter <code>yThr</code>. The
threshold is plotted as a horizontal line. If there are previously known
marker effect, false positives and true negatives can also be marked.<br />
Extra parameter options:
</p>

<dl>
<dt><code>xLab</code></dt><dd><p>A character string, the x-axis label. Default =
<code>"Chromosomes"</code></p>
</dd>
<dt><code>yLab</code></dt><dd><p>A character string, the y-axis label. Default =
<code>-log10(p)</code></p>
</dd>
<dt><code>effects</code></dt><dd><p>A character vector, indicating which SNPs correspond
to a real (known) effect. Used for determining true/false positives and
false negatives. True positives are colored green, false positives orange and
false negatives yellow.</p>
</dd>
<dt><code>colPalette</code></dt><dd><p>A color palette used for plotting. Default
coloring is done by chromosome, using black and grey.</p>
</dd>
<dt><code>yThr</code></dt><dd><p>A numerical value for the LOD-threshold. The value from
the GWAS analysis is used as default.</p>
</dd>
<dt><code>signLwd</code></dt><dd><p>A numerical value giving the thickness of the
points that are false/true positives/negatives. Default = 0.6</p>
</dd>
<dt><code>lod</code></dt><dd><p>A positive numerical value. For the SNPs with a LOD-value
below this value, only 5% is plotted. The chance of a SNP being plotted is
proportional to its LOD-score. This option can be useful when plotting a
large number of SNPs. The 5% of SNPs plotted is selected randomly. For
reproducible results use set.seed before calling the function.</p>
</dd>
<dt><code>chr</code></dt><dd><p>A vector of chromosomes to be plotted. By default, all
chromosomes are plotted. Using this option allows restricting the plot to a
subset of chromosomes.</p>
</dd>
<dt><code>startPos</code></dt><dd><p>A numerical value indicating the start position for
the plot. Using this option allows restricting the plot to a part of a
selected chromosome. Only used if exactly one chromosome is specified in
<code>chr</code>.</p>
</dd>
<dt><code>endPos</code></dt><dd><p>A numerical value indicating the end position for
the plot. Using this option allows restricting the plot to a part of a
selected chromosome. Only used if exactly one chromosome is specified in
<code>chr</code>.</p>
</dd>
</dl>



<h3>QQ-Plot</h3>

<p>From the LOD-scores calculated in the GWAS analysis, a QQ-plot is generated
with observed LOD-scores versus expected LOD-scores. Code is adapted from
Segura et al. (2012).
</p>


<h3>QTL Plot</h3>

<p>A plot of effect sizes for the significant SNPs found in the GWAS analysis
is created. Each horizontal line contains QTLs of one trait,
phenotypic trait or trial. Optionally, vertical white lines can indicate
chromosome subdivision, genes of interest, known QTL, etc. Circle diameters
are proportional to the absolute value of allelic effect. Colors indicate the
direction of the effect: green when the allele increases the trait value,
and blue when it decreases the value.<br />
Extra parameter options:
</p>

<dl>
<dt><code>normalize</code></dt><dd><p>Should the snpEffect be normalized? Default =
<code>FALSE</code></p>
</dd>
<dt><code>sortData</code></dt><dd><p>Should the data be sorted before plotting? Either
<code>FALSE</code>, if no sorting should be done, or a character string indicating
the data column to use for sorting. This should be a numerical column.
Default = <code>FALSE</code></p>
</dd>
<dt><code>binPositions</code></dt><dd><p>An optional data.frame containing at leasts two
columns, chr(omosome) and pos(ition). Vertical lines are plotted at those
positions. Default = <code>NULL</code></p>
</dd>
<dt><code>printVertGrid</code></dt><dd><p>Should default vertical grid lines be plotted.
Default = <code>TRUE</code></p>
</dd>
<dt><code>yLab</code></dt><dd><p>A character string, the y-axis label. Default =
<code>"Traits"</code></p>
</dd>
<dt><code>yThr</code></dt><dd><p>A numerical value for the LOD-threshold. The value from
the GWAS analysis is used as default.</p>
</dd>
<dt><code>chr</code></dt><dd><p>A vector of chromosomes to be plotted. By default all
chromosomes are plotted. Using this option this can be restricted to a
subset of chromosomes.</p>
</dd>
<dt><code>exportPptx</code></dt><dd><p>Should the plot be exported to a .pptx file?
Default = <code>FALSE</code></p>
</dd>
<dt><code>pptxName</code></dt><dd><p>A character string, the name of the .pptx file to
which the plot is exported. Ignored if exportPptx = <code>FALSE</code>.</p>
</dd>
</dl>



<h3>References</h3>

<p>Millet et al. (2016) Genome-wide analysis of yield in Europe:
Allelic effects vary with drought and heat scenarios. Plant Physiology,
October 2016, Vol. 172, p. 749–764
</p>
<p>Segura et al. (2012) An efficient multi-locus mixed-model
approach for genome-wide association studies in structured populations.
Nature Genetics, June 2012, Vol. 44, p. 825–830.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create a gData object Using the data from the DROPS project.
## See the included vignette for a more extensive description on the steps.
data(dropsMarkers)
data(dropsMap)
data(dropsPheno)

## Add genotypes as row names of dropsMarkers and drop Ind column.
rownames(dropsMarkers) &lt;- dropsMarkers[["Ind"]]
dropsMarkers &lt;- dropsMarkers[colnames(dropsMarkers) != "Ind"]

## Add genotypes as row names of dropsMap.
rownames(dropsMap) &lt;- dropsMap[["SNP.names"]]

## Rename Chomosome and Position columns.
colnames(dropsMap)[match(c("Chromosome", "Position"), 
                   colnames(dropsMap))] &lt;- c("chr", "pos")
                   
## Convert phenotypic data to a list.
dropsPhenoList &lt;- split(x = dropsPheno, f = dropsPheno[["Experiment"]])

## Rename Variety_ID to genotype and select relevant columns.
dropsPhenoList &lt;- lapply(X = dropsPhenoList, FUN = function(trial) {
  colnames(trial)[colnames(trial) == "Variety_ID"] &lt;- "genotype"
  trial &lt;- trial[c("genotype", "grain.yield", "grain.number", "seed.size",
                 "anthesis", "silking", "plant.height", "tassel.height",
                 "ear.height")]
return(trial)
}) 

## Create gData object.
gDataDrops &lt;- createGData(geno = dropsMarkers, map = dropsMap, 
                          pheno = dropsPhenoList)
                          
## Run single trait GWAS for trait 'grain.yield' for trial Mur13W.

GWASDrops &lt;- runSingleTraitGwas(gData = gDataDrops,
                               trials = "Mur13W",
                               traits = "grain.yield")
                               
## Create a manhattan plot.
plot(GWASDrops)

## Manually set a threshold for significant snps and add a title.
plot(GWASDrops, 
    yThr = 3.5, 
    title = "Manhattan plot for Mur13W")
    
## Restrict plot to part of chr 6.
plot(GWASDrops, 
    yThr = 3.5, 
    chr = 6,
    startPos = 0,
    endPos = 110000000)
    
## Create a qq plot.
plot(GWASDrops, 
    plotType = "qq",
    title = "QQ plot for Mur13W")
    
## Create a QTL plot.
plot(GWASDrops,
    plotType = "qtl",
    title = "QTL plot for Mur13W")
    
## Manually set a threshold and don't show vertical lines.
plot(GWASDrops,     
    plotType = "qtl",
    yThr = 3.5,
    printVertGrid = FALSE,
    title = "QTL plot for Mur13W")          


</code></pre>

<hr>
<h2 id='readPLINK'>Read PLINK binary data</h2><span id='topic+readPLINK'></span>

<h3>Description</h3>

<p>Read PLINK binary data and save in gData format. This is a wrapper around
<code><a href="snpStats.html#topic+read.plink">read.plink</a></code> in the Bioconductor package
<code>snpStats</code>. This package needs to be installed for the function to
work.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readPLINK(bed, bim, fam, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readPLINK_+3A_bed">bed</code></td>
<td>
<p>The name of the file containing the packed binary SNP genotype
data. It should have the extension .bed; If it doesn't, then this extension
will be appended.</p>
</td></tr>
<tr><td><code id="readPLINK_+3A_bim">bim</code></td>
<td>
<p>The file containing the SNP descriptions. If not specified
<code>bed</code> is used with its file extension replaced by bim.</p>
</td></tr>
<tr><td><code id="readPLINK_+3A_fam">fam</code></td>
<td>
<p>The file containing subject (and, possibly, family) identifiers.
This is basically a tab-delimited &quot;pedfile&quot;. If not specified
<code>bed</code> is used with its file extension replaced by fam.</p>
</td></tr>
<tr><td><code id="readPLINK_+3A_...">...</code></td>
<td>
<p>Further arguments passed to <code><a href="snpStats.html#topic+read.plink">read.plink</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>gData</code>.
</p>

<hr>
<h2 id='runSingleTraitGwas'>Perform single-trait GWAS</h2><span id='topic+runSingleTraitGwas'></span>

<h3>Description</h3>

<p><code>runSingleTraitGwas</code> performs a single-trait Genome Wide Association
Study (GWAS) on phenotypic and genotypic data contained in a <code>gData</code>
object. A covariance matrix is computed using the EMMA algorithm (Kang et
al., 2008) or the Newton-Raphson algorithm (Tunnicliffe, 1989) in the
<code>sommer</code> package. Then a Generalized Least Squares (GLS) method is used
for estimating the marker effects and corresponding p-values. This is done
using either one kinship matrix for all chromosomes or different
chromosome-specific kinship matrices for each chromosome. Significant SNPs
are selected based on a user defined threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runSingleTraitGwas(
  gData,
  traits = NULL,
  trials = NULL,
  covar = NULL,
  snpCov = NULL,
  kin = NULL,
  kinshipMethod = c("astle", "IBS", "vanRaden"),
  remlAlgo = c("EMMA", "NR"),
  GLSMethod = c("single", "multi"),
  useMAF = TRUE,
  MAF = 0.01,
  MAC = 10,
  genomicControl = FALSE,
  thrType = c("bonf", "fixed", "small", "fdr"),
  alpha = 0.05,
  LODThr = 4,
  nSnpLOD = 10,
  pThr = 0.05,
  rho = 0.5,
  sizeInclRegion = 0,
  minR2 = 0.5,
  nCores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runSingleTraitGwas_+3A_gdata">gData</code></td>
<td>
<p>An object of class <code>gData</code> containing at least <code>map</code>,
<code>markers</code> and <code>pheno</code>.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_traits">traits</code></td>
<td>
<p>A vector of traits on which to run GWAS. These can be either
numeric indices or character names of columns in <code>pheno</code>. If <code>NULL</code>,
GWAS is run on all traits.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_trials">trials</code></td>
<td>
<p>A vector of trials on which to run GWAS. These can
be either numeric indices or character names of list items in <code>pheno</code>.
If <code>NULL</code>, GWAS is run for all trials. GWAS is run for the
selected trials in sequential order.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_covar">covar</code></td>
<td>
<p>An optional vector of covariates taken into account when
running GWAS. These can be either numeric indices or character names of
columns in <code>covar</code> in <code>gData</code>. If <code>NULL</code> no covariates are
used.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_snpcov">snpCov</code></td>
<td>
<p>An optional character vector of snps to be included as
covariates.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_kin">kin</code></td>
<td>
<p>An optional kinship matrix or list of kinship matrices. These
matrices can be from the <code>matrix</code> class as defined in the base package
or from the <code>dsyMatrix</code> class, the class of symmetric matrices in the
Matrix package.<br />
If <code>GLSMethod</code> = &quot;single&quot; then one matrix should be provided, if
<code>GLSMethod</code> = &quot;multi&quot;, a list of chromosome specific matrices of length
equal to the number of chromosomes in <code>map</code> in <code>gData</code>.<br />
If <code>NULL</code> then matrix <code>kinship</code> in <code>gData</code> is used. <br />
If both <code>kin</code> is provided and <code>gData</code> contains a matrix
<code>kinship</code> then <code>kin</code> is used.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_kinshipmethod">kinshipMethod</code></td>
<td>
<p>An optional character indicating the method used for
calculating the kinship matrix(ces). Currently &quot;astle&quot; (Astle and Balding,
2009), &quot;IBS&quot; and &quot;vanRaden&quot; (VanRaden, 2008) are supported. If a
kinship matrix is supplied either in <code>gData</code> or in parameter <code>kin</code>,
<code>kinshipMethod</code> is ignored.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_remlalgo">remlAlgo</code></td>
<td>
<p>A character string indicating the algorithm used to estimate
the variance components. Either <code>EMMA</code>, for the EMMA algorithm, or
<code>NR</code>, for the Newton-Raphson algorithm.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_glsmethod">GLSMethod</code></td>
<td>
<p>A character string indicating the method used to estimate
the marker effects. Either <code>single</code> for using a single kinship matrix,
or <code>multi</code> for using chromosome specific kinship matrices.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_usemaf">useMAF</code></td>
<td>
<p>Should the minor allele frequency be used for selecting SNPs
for the analysis. If <code>FALSE</code>, the minor allele count is used instead.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_maf">MAF</code></td>
<td>
<p>The minor allele frequency (MAF) threshold used in GWAS. A
numerical value between 0 and 1. SNPs with MAF below this value are not taken
into account in the analysis, i.e. p-values and effect sizes are put to
missing (<code>NA</code>). Ignored if <code>useMAF</code> is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_mac">MAC</code></td>
<td>
<p>A numerical value. SNPs with minor allele count below this value
are not taken into account for the analysis, i.e. p-values and effect sizes
are set to missing (<code>NA</code>). Ignored if <code>useMAF</code> is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_genomiccontrol">genomicControl</code></td>
<td>
<p>Should genomic control correction as in Devlin and
Roeder (1999) be applied?</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_thrtype">thrType</code></td>
<td>
<p>A character string indicating the type of threshold used for
the selection of candidate loci. See Details.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_alpha">alpha</code></td>
<td>
<p>A numerical value used for calculating the LOD-threshold for
<code>thrType</code> = &quot;bonf&quot; and the significant p-Values for <code>thrType</code> =
&quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_lodthr">LODThr</code></td>
<td>
<p>A numerical value used as a LOD-threshold when
<code>thrType</code> = &quot;fixed&quot;.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_nsnplod">nSnpLOD</code></td>
<td>
<p>A numerical value indicating the number of SNPs with the
smallest p-values that are selected when <code>thrType</code> = &quot;small&quot;.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_pthr">pThr</code></td>
<td>
<p>A numerical value just as the cut off value for p-Values for
<code>thrType</code> = &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_rho">rho</code></td>
<td>
<p>A numerical value used a the minimum value for SNPs to be
considered correlated when using <code>thrType</code> = &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_sizeinclregion">sizeInclRegion</code></td>
<td>
<p>An integer. Should the results for SNPs close to
significant SNPs be included? If so, the size of the region in centimorgan
or base pairs. Otherwise 0.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_minr2">minR2</code></td>
<td>
<p>A numerical value between 0 and 1. Restricts the SNPs included
in the region close to significant SNPs to only those SNPs that are in
sufficient Linkage Disequilibrium (LD) with the significant snp, where LD
is measured in terms of <code class="reqn">R^2</code>. If for example <code>sizeInclRegion</code> =
200000 and <code>minR2</code> = 0.5, then for every significant SNP also those SNPs
whose LD (<code class="reqn">R^2</code>) with the significant SNP is at least 0.5 AND which are
at most 200000 away from this significant snp are included. Ignored if
<code>sizeInclRegion</code> = 0.</p>
</td></tr>
<tr><td><code id="runSingleTraitGwas_+3A_ncores">nCores</code></td>
<td>
<p>A numerical value indicating the number of cores to be used by
the parallel part of the algorithm. If <code>NULL</code> the number of cores used
will be equal to the number of cores available on the machine - 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>GWAS</code>.
</p>


<h3>Threshold types for the selection of candidate loci</h3>

<p>For the selection of candidate loci from the GWAS output four different
methods can be used. The method used can be specified in the function
parameter <code>thrType</code>. Further parameters can be used to fine tune the
method.
</p>

<dl>
<dt>bonf</dt><dd><p>The Bonferroni threshold, a LOD-threshold of
<code class="reqn">-log10(alpha / m)</code>, where m is the number of SNPs and alpha can be
specified by the function parameter <code>alpha</code></p>
</dd>
<dt>fixed</dt><dd><p>A fixed LOD-threshold, specified by the function parameter
<code>LODThr</code></p>
</dd>
<dt>small</dt><dd><p>The n SNPs with with the smallest p-Values are selected. n can
be specified in <code>nSnpLOD</code>
</p>
</dd>
<dt>fdr</dt><dd><p>Following the algorithm proposed by Brzyski D. et al. (2017),
SNPs are selected in such a way that the False Discovery Rate (fdr) is
minimized. To do this, first the SNPs are restricted to the SNPs with a
p-Value below <code>pThr</code>. Then clusters of SNPs are created using a
two step iterative process in which first the SNP with the lowest p-Value is
selected as cluster representative. This SNP and all SNPs that have a
correlation with this SNP of <code class="reqn">\rho</code> or higher will form a cluster.
<code class="reqn">\rho</code> can be specified as an argument in the function and has a default
value of 0.5, which is a recommended starting value in practice.
The selected SNPs are removed from the list and the procedure repeated until
no SNPs are left. Finally to determine the number of significant clusters
the first cluster is determined for which the p-Value of the cluster
representative is not smaller than <code class="reqn">cluster_{number} * \alpha / m</code> where
m is the number of SNPs and alpha can be specified by the function parameter
<code>alpha</code>. All previous clusters are selected as significant.</p>
</dd>
</dl>



<h3>References</h3>

<p>Astle, William, and David J. Balding. 2009. Population Structure
and Cryptic Relatedness in Genetic Association Studies. Statistical Science
24 (4): 451–71. <a href="https://doi.org/10.1214/09-sts307">doi:10.1214/09-sts307</a>.
</p>
<p>Brzyski D. et al. (2017) Controlling the Rate of GWAS False
Discoveries. Genetics 205 (1): 61-75.
<a href="https://doi.org/10.1534/genetics.116.193987">doi:10.1534/genetics.116.193987</a>
</p>
<p>Devlin, B., and Kathryn Roeder. 1999. Genomic Control for
Association Studies. Biometrics 55 (4): 997–1004.
<a href="https://doi.org/10.1111/j.0006-341x.1999.00997.x">doi:10.1111/j.0006-341x.1999.00997.x</a>.
</p>
<p>Kang et al. (2008) Efficient Control of Population Structure in
Model Organism Association Mapping. Genetics 178 (3): 1709–23.
<a href="https://doi.org/10.1534/genetics.107.080101">doi:10.1534/genetics.107.080101</a>.
</p>
<p>Millet, E. J., Pommier, C., et al. (2019). A multi-site
experiment in a network of European fields for assessing the maize yield
response to environmental scenarios - Data set.
<a href="https://doi.org/10.15454/IASSTN">doi:10.15454/IASSTN</a>
</p>
<p>Rincent et al. (2014) Recovering power in association mapping
panels with variable levels of linkage disequilibrium. Genetics 197 (1):
375–87. <a href="https://doi.org/10.1534/genetics.113.159731">doi:10.1534/genetics.113.159731</a>.
</p>
<p>Segura et al. (2012) An efficient multi-locus mixed-model
approach for genome-wide association studies in structured populations.
Nature Genetics 44 (7): 825–30. <a href="https://doi.org/10.1038/ng.2314">doi:10.1038/ng.2314</a>.
</p>
<p>Sun et al. (2010) Variation explained in mixed-model association
mapping. Heredity 105 (4): 333–40. <a href="https://doi.org/10.1038/hdy.2010.11">doi:10.1038/hdy.2010.11</a>.
</p>
<p>Tunnicliffe W. (1989) On the use of marginal likelihood in time
series model estimation. JRSS 51 (1): 15–27.
</p>
<p>VanRaden P.M. (2008) Efficient methods to compute genomic
predictions. Journal of Dairy Science 91 (11): 4414–23.
<a href="https://doi.org/10.3168/jds.2007-0980">doi:10.3168/jds.2007-0980</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.GWAS">summary.GWAS</a></code>, <code><a href="#topic+plot.GWAS">plot.GWAS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create a gData object Using the data from the DROPS project.
## See the included vignette for a more extensive description on the steps.
data(dropsMarkers)
data(dropsMap)
data(dropsPheno)

## Add genotypes as row names of dropsMarkers and drop Ind column.
rownames(dropsMarkers) &lt;- dropsMarkers[["Ind"]]
dropsMarkers &lt;- dropsMarkers[colnames(dropsMarkers) != "Ind"]

## Add genotypes as row names of dropsMap.
rownames(dropsMap) &lt;- dropsMap[["SNP.names"]]

## Rename Chomosome and Position columns.
colnames(dropsMap)[match(c("Chromosome", "Position"), 
                   colnames(dropsMap))] &lt;- c("chr", "pos")
                   
## Convert phenotypic data to a list.
dropsPhenoList &lt;- split(x = dropsPheno, f = dropsPheno[["Experiment"]])

## Rename Variety_ID to genotype and select relevant columns.
dropsPhenoList &lt;- lapply(X = dropsPhenoList, FUN = function(trial) {
  colnames(trial)[colnames(trial) == "Variety_ID"] &lt;- "genotype"
  trial &lt;- trial[c("genotype", "grain.yield", "grain.number", "seed.size",
                 "anthesis", "silking", "plant.height", "tassel.height",
                 "ear.height")]
return(trial)
}) 

## Create gData object.
gDataDrops &lt;- createGData(geno = dropsMarkers, map = dropsMap, 
                          pheno = dropsPhenoList)
                          
## Run single trait GWAS for trait 'grain.yield' for trial Mur13W.

GWASDrops &lt;- runSingleTraitGwas(gData = gDataDrops,
                               trials = "Mur13W",
                               traits = "grain.yield")

                               
## Run single trait GWAS for trait 'grain.yield' for trial Mur13W.
## Use chromosome specific kinship matrices calculated using vanRaden method.

GWASDropsMult &lt;- runSingleTraitGwas(gData = gDataDrops,
                                    trials = "Mur13W",
                                    traits = "grain.yield",
                                    kinshipMethod = "vanRaden",
                                    GLSMethod = "multi")  


</code></pre>

<hr>
<h2 id='statgenGWAS-package'>statgenGWAS: Genome Wide Association Studies</h2><span id='topic+statgenGWAS'></span><span id='topic+statgenGWAS-package'></span>

<h3>Description</h3>

<p>Fast single trait Genome Wide Association Studies (GWAS) following the method described in Kang et al. (2010), <a href="https://doi.org/10.1038/ng.548">doi:10.1038/ng.548</a>. One of a series of statistical genetic packages for streamlining the analysis of typical plant breeding experiments developed by Biometris.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Bart-Jan van Rossum <a href="mailto:bart-jan.vanrossum@wur.nl">bart-jan.vanrossum@wur.nl</a> (<a href="https://orcid.org/0000-0002-8673-2514">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Willem Kruijer (<a href="https://orcid.org/0000-0001-7179-1733">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Fred van Eeuwijk (<a href="https://orcid.org/0000-0003-3672-2921">ORCID</a>) [contributor]
</p>
</li>
<li><p> Martin Boer (<a href="https://orcid.org/0000-0002-1879-4588">ORCID</a>) [contributor]
</p>
</li>
<li><p> Marcos Malosetti (<a href="https://orcid.org/0000-0002-8150-1397">ORCID</a>) [contributor]
</p>
</li>
<li><p> Daniela Bustos-Korts (<a href="https://orcid.org/0000-0003-3827-6726">ORCID</a>) [contributor]
</p>
</li>
<li><p> Emilie Millet (<a href="https://orcid.org/0000-0002-2913-4892">ORCID</a>) [contributor]
</p>
</li>
<li><p> Joao Paulo (<a href="https://orcid.org/0000-0002-4180-0763">ORCID</a>) [contributor]
</p>
</li>
<li><p> Maikel Verouden (<a href="https://orcid.org/0000-0002-4893-3323">ORCID</a>) [contributor]
</p>
</li>
<li><p> Ron Wehrens (<a href="https://orcid.org/0000-0002-8798-5599">ORCID</a>) [contributor]
</p>
</li>
<li><p> Choazhi Zheng (<a href="https://orcid.org/0000-0001-6030-3933">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://biometris.github.io/statgenGWAS/index.html">https://biometris.github.io/statgenGWAS/index.html</a>
</p>
</li>
<li> <p><a href="https://github.com/Biometris/statgenGWAS/">https://github.com/Biometris/statgenGWAS/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Biometris/statgenGWAS/issues">https://github.com/Biometris/statgenGWAS/issues</a>
</p>
</li></ul>


<hr>
<h2 id='summary.gData'>Summary function for the class <code>gData</code></h2><span id='topic+summary.gData'></span>

<h3>Description</h3>

<p>Gives a summary for an object of S3 class <code>gData</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gData'
summary(object, ..., trials = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.gData_+3A_object">object</code></td>
<td>
<p>An object of class <code>gData</code>.</p>
</td></tr>
<tr><td><code id="summary.gData_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="summary.gData_+3A_trials">trials</code></td>
<td>
<p>A vector of trials to include in the summary. These can
be either numeric indices or character names of list items in <code>pheno</code>.
If <code>NULL</code>, all trials are included.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with a most four components:
</p>

<dl>
<dt>mapSum</dt><dd><p>A list with number of markers and number of chromosomes in
the map.</p>
</dd>
<dt>markerSum</dt><dd><p>A list with number of markers, number of genotypes and
the distribution of the values within the markers.</p>
</dd>
<dt>phenoSum</dt><dd><p>A list of data.frames, one per trial with a summary of all
traits within the trial.</p>
</dd>
<dt>covarSum</dt><dd><p>A list of data.frames, one per trial with a summary of all
covariates within the trial.</p>
</dd>
</dl>

<p>All components are only present in the output if the corresponding content is
present in the gData object.
</p>

<hr>
<h2 id='summary.GWAS'>Summary function for the class <code>GWAS</code></h2><span id='topic+summary.GWAS'></span>

<h3>Description</h3>

<p>Gives a summary for an object of S3 class <code>GWAS</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GWAS'
summary(object, ..., trials = NULL, traits = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.GWAS_+3A_object">object</code></td>
<td>
<p>An object of class <code>GWAS</code>.</p>
</td></tr>
<tr><td><code id="summary.GWAS_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="summary.GWAS_+3A_trials">trials</code></td>
<td>
<p>A vector of strings or numeric indices indicating for
which trials the summary should be made. If <code>NULL</code>, a summary is
made for all trials.</p>
</td></tr>
<tr><td><code id="summary.GWAS_+3A_traits">traits</code></td>
<td>
<p>A vector of strings indicating which traits to include in the
summary. If <code>NULL</code>, all traits are included in the summary.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
