<!DOCTYPE html><html lang="en"><head><title>Help for package statgenIBD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {statgenIBD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#statgenIBD-package'><p>statgenIBD: Calculation of IBD Probabilities</p></a></li>
<li><a href='#allGenoPlot'><p>Helper function for plotting probabilities for all genotypes</p></a></li>
<li><a href='#c.IBDprob'><p>Concatenate function for objects of class IBDprob</p></a></li>
<li><a href='#calcIBD'><p>Calculate IBD probabilities</p></a></li>
<li><a href='#getProbs'><p>Extract Probabilities for markers</p></a></li>
<li><a href='#pedPlot'><p>Helper function for plotting the pedigree</p></a></li>
<li><a href='#plot.IBDprob'><p>Plot function for objects of class IBDprob</p></a></li>
<li><a href='#readIBDs'><p>Read IBD probabilities from file</p></a></li>
<li><a href='#readRABBIT'><p>Read IBD probabilities</p></a></li>
<li><a href='#singleGenoPlot'><p>Helper function for plotting probabilities for a single genotype</p></a></li>
<li><a href='#summary.IBDprob'><p>Summary function for objects of class IBDprob</p></a></li>
<li><a href='#writeFlapjack'><p>Write to Flapjack format</p></a></li>
<li><a href='#writeIBDs'><p>Write IBD probabilities to file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calculation of IBD Probabilities</td>
</tr>
<tr>
<td>Description:</td>
<td>For biparental, three and four-way crosses Identity by Descent 
    (IBD) probabilities can be calculated using Hidden Markov Models and 
    inheritance vectors following Lander and Green
    (<a href="https://www.jstor.org/stable/29713">https://www.jstor.org/stable/29713</a>) and Huang
    (&lt;<a href="https://doi.org/10.1073%2Fpnas.1100465108">doi:10.1073/pnas.1100465108</a>&gt;). One of a series of statistical genetic 
    packages for streamlining the analysis of typical plant breeding experiments
    developed by Biometris.</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-02-05</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, ggplot2, Matrix, rlang, Rcpp, statgenGWAS (&ge;
1.0.9), stringi</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, tinytest, R.utils</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://biometris.github.io/statgenIBD/index.html">https://biometris.github.io/statgenIBD/index.html</a>,
<a href="https://github.com/Biometris/statgenIBD/">https://github.com/Biometris/statgenIBD/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Biometris/statgenIBD/issues">https://github.com/Biometris/statgenIBD/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-05 11:22:46 UTC; rossu027</td>
</tr>
<tr>
<td>Author:</td>
<td>Martin Boer <a href="https://orcid.org/0000-0002-1879-4588"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Bart-Jan van Rossum
    <a href="https://orcid.org/0000-0002-8673-2514"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Wenhao Li <a href="https://orcid.org/0000-0001-5719-5775"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Johannes Kruisselbrink
    <a href="https://orcid.org/0000-0003-1673-5725"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bart-Jan van Rossum &lt;bart-jan.vanrossum@wur.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-05 12:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='statgenIBD-package'>statgenIBD: Calculation of IBD Probabilities</h2><span id='topic+statgenIBD'></span><span id='topic+statgenIBD-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>For biparental, three and four-way crosses Identity by Descent (IBD) probabilities can be calculated using Hidden Markov Models and inheritance vectors following Lander and Green (<a href="https://www.jstor.org/stable/29713">https://www.jstor.org/stable/29713</a>) and Huang (<a href="https://doi.org/10.1073/pnas.1100465108">doi:10.1073/pnas.1100465108</a>). One of a series of statistical genetic packages for streamlining the analysis of typical plant breeding experiments developed by Biometris.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Bart-Jan van Rossum <a href="mailto:bart-jan.vanrossum@wur.nl">bart-jan.vanrossum@wur.nl</a> (<a href="https://orcid.org/0000-0002-8673-2514">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Martin Boer <a href="mailto:martin.boer@wur.nl">martin.boer@wur.nl</a> (<a href="https://orcid.org/0000-0002-1879-4588">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Wenhao Li (<a href="https://orcid.org/0000-0001-5719-5775">ORCID</a>) [contributor]
</p>
</li>
<li><p> Johannes Kruisselbrink (<a href="https://orcid.org/0000-0003-1673-5725">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://biometris.github.io/statgenIBD/index.html">https://biometris.github.io/statgenIBD/index.html</a>
</p>
</li>
<li> <p><a href="https://github.com/Biometris/statgenIBD/">https://github.com/Biometris/statgenIBD/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Biometris/statgenIBD/issues">https://github.com/Biometris/statgenIBD/issues</a>
</p>
</li></ul>


<hr>
<h2 id='allGenoPlot'>Helper function for plotting probabilities for all genotypes</h2><span id='topic+allGenoPlot'></span>

<h3>Description</h3>

<p>Helper function for plotting probabilities for all genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allGenoPlot(markers, map, parents, title)
</code></pre>

<hr>
<h2 id='c.IBDprob'>Concatenate function for objects of class IBDprob</h2><span id='topic+c.IBDprob'></span>

<h3>Description</h3>

<p>Concatenates objects of class <code>IBDprob</code>. All objects that are
concatenated should have the same population type and the same map. The
function is mainly meant for combining information for multiple crosses
with overlapping parents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBDprob'
c(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="c.IBDprob_+3A_...">...</code></td>
<td>
<p>Objects of class <code>IBDprob</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>IBDprob</code> containing data for all
concatenated objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute IBD probabilties for AxB.
AB &lt;- calcIBD(popType = "F4DH",
             markerFile = system.file("extdata/multipop", "AxB.txt",
                                  package = "statgenIBD"),
             mapFile = system.file("extdata/multipop", "mapfile.txt",
                                  package = "statgenIBD"))
## Compute IBD probabilties for Axc.
AC &lt;- calcIBD(popType = "F4DH",
              markerFile = system.file("extdata/multipop", "AxC.txt",
                                      package = "statgenIBD"),
              mapFile = system.file("extdata/multipop", "mapfile.txt",
                                   package = "statgenIBD"))

## Combine results.
ABC &lt;- c(AB, AC)

## Check summary.
summary(ABC)

</code></pre>

<hr>
<h2 id='calcIBD'>Calculate IBD probabilities</h2><span id='topic+calcIBD'></span>

<h3>Description</h3>

<p>Calculate IBD probabilities for different types of populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcIBD(
  popType,
  markerFile,
  mapFile,
  evalPos = NULL,
  evalDist = NULL,
  grid = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calcIBD_+3A_poptype">popType</code></td>
<td>
<p>A character string indicating the type of population. One of
DH, Fx, FxDH, BCx, BCxDH, BC1Sx, BC1SxDH, C3, C3DH, C3Sx, C3SxDH, C4, C4DH,
C4Sx, C4SxDH (see Details).</p>
</td></tr>
<tr><td><code id="calcIBD_+3A_markerfile">markerFile</code></td>
<td>
<p>A character string indicating the location of the file with
genotypic information for the population. The file should be in
tab-delimited format with a header containing marker names.</p>
</td></tr>
<tr><td><code id="calcIBD_+3A_mapfile">mapFile</code></td>
<td>
<p>A character string indicating the location of the map file
for the population. The file should be in tab-delimited format. It should
consist of exactly three columns, marker, chromosome and position. There
should be no header. The positions in the file should be in centimorgan.</p>
</td></tr>
<tr><td><code id="calcIBD_+3A_evalpos">evalPos</code></td>
<td>
<p>A data.frame with evaluation positions to which the
calculations should be limited.</p>
</td></tr>
<tr><td><code id="calcIBD_+3A_evaldist">evalDist</code></td>
<td>
<p>An optional numerical value indicating the maximum
distance for marker positions. Extra markers will be added based on the
value of <code>grid</code>.</p>
</td></tr>
<tr><td><code id="calcIBD_+3A_grid">grid</code></td>
<td>
<p>Should the extra markers that are added to assure the a
maximum distince of <code>evalDist</code> be on a grid (<code>TRUE</code>) or in between
marker existing marker positions (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="calcIBD_+3A_verbose">verbose</code></td>
<td>
<p>Should messages indicating the progress of the process be
printed?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>IBD probabilities can be calculated for many different types of populations.
In the following table all supported populations are listed. Note that the
value of x in the population types is variable, with its maximum value
depicted in the last column.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Population type</strong> </td><td style="text-align: left;"> <strong>Cross</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>max. x</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   DH </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> doubled haploid population </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   Fx </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> Fx population (F1, followed by x-1 generations of selfing) </td><td style="text-align: left;"> 8 </td>
</tr>
<tr>
 <td style="text-align: left;">
   FxDH </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> Fx, followed by DH generation </td><td style="text-align: left;"> 8 </td>
</tr>
<tr>
 <td style="text-align: left;">
   BCx </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> backcross, second parent is recurrent parent </td><td style="text-align: left;"> 9 </td>
</tr>
<tr>
 <td style="text-align: left;">
   BCxDH </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> BCx, followed by DH generation </td><td style="text-align: left;"> 9 </td>
</tr>
<tr>
 <td style="text-align: left;">
   BC1Sx </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> BC1, followed by x generations of selfing </td><td style="text-align: left;"> 7 </td>
</tr>
<tr>
 <td style="text-align: left;">
   BC1SxDH </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> BC1, followed by x generations of selfing and DH </td><td style="text-align: left;"> 6 </td>
</tr>
<tr>
 <td style="text-align: left;">
   C3 </td><td style="text-align: left;"> three-way </td><td style="text-align: left;"> three way cross: (AxB) x C </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   C3DH </td><td style="text-align: left;"> three-way </td><td style="text-align: left;"> C3, followed by DH generation </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   C3Sx </td><td style="text-align: left;"> three-way </td><td style="text-align: left;"> C3, followed by x generations of selfing </td><td style="text-align: left;"> 7 </td>
</tr>
<tr>
 <td style="text-align: left;">
   C3SxDH </td><td style="text-align: left;"> three-way </td><td style="text-align: left;"> C3, followed by x generations of selfing and DH generation </td><td style="text-align: left;"> 6 </td>
</tr>
<tr>
 <td style="text-align: left;">
   C4 </td><td style="text-align: left;"> four-way </td><td style="text-align: left;"> four-way cross: (AxB) x (CxD) </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   C4DH </td><td style="text-align: left;"> four-way </td><td style="text-align: left;"> C4, followed by DH generation </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   C4Sx </td><td style="text-align: left;"> four-way </td><td style="text-align: left;"> C4, followed by x generations of selfing </td><td style="text-align: left;"> 6 </td>
</tr>
<tr>
 <td style="text-align: left;">
   C4SxDH </td><td style="text-align: left;"> four-way </td><td style="text-align: left;"> C4, followed by x generations of selfing and DH generation </td><td style="text-align: left;"> 6 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>An object of class <code>IBDprob</code>, a <code>list</code> with five elements,
</p>

<dl>
<dt>map</dt><dd><p>a <code>data.frame</code> with chromosome and position of the markers.</p>
</dd>
<dt>markers</dt><dd><p>a 3-dimensional <code>array</code> of IBD probabilities with
genotypes, markers and parents as array dimensions.</p>
</dd>
<dt>parents</dt><dd><p>the parents.</p>
</dd>
<dt>popType</dt><dd><p>the population type.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Compute IBD probabilities for Steptoe Morex.
SxMIBD &lt;- calcIBD(popType = "DH",
                  markerFile = system.file("extdata/SxM", "SxM_geno.txt",
                                        package = "statgenIBD"),
                  mapFile = system.file("extdata/SxM", "SxM_map.txt",
                                        package = "statgenIBD"))

## Check summary.
summary(SxMIBD)

## Compute IBD probabilities for Steptoe Morex.
## Add extra evaluation positions in between exiting marker positions
## to assure evaluation positions are at most 5 cM apart.
SxMIBD_Ext &lt;- calcIBD(popType = "DH",
                      markerFile = system.file("extdata/SxM", "SxM_geno.txt",
                                              package = "statgenIBD"),
                      mapFile = system.file("extdata/SxM", "SxM_map.txt",
                                            package = "statgenIBD"),
                      evalDist = 5)

## Check summary.
summary(SxMIBD_Ext)

</code></pre>

<hr>
<h2 id='getProbs'>Extract Probabilities for markers</h2><span id='topic+getProbs'></span>

<h3>Description</h3>

<p>Extract IBD probabilities for one or more markers from an object of class
<code>IBDprob</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getProbs(IBDprob, markers, sumProbs = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getProbs_+3A_ibdprob">IBDprob</code></td>
<td>
<p>An object of class <code>IBDprob</code>.</p>
</td></tr>
<tr><td><code id="getProbs_+3A_markers">markers</code></td>
<td>
<p>A character vector of markers that should be extracted.</p>
</td></tr>
<tr><td><code id="getProbs_+3A_sumprobs">sumProbs</code></td>
<td>
<p>Should the probabilities by summed per parent. If <code>TRUE</code>
the probability for e.g. parent A in a cross with parent B will be calculated
as pA + 0.5 * pAB. If <code>FALSE</code> both pA and pAB will be output without
further calculations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with IBD probabilities for the extracted markers in the
column and genotypes in the rows.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute IBD probabilities for Steptoe Morex.
SxMIBD &lt;- calcIBD(popType = "DH",
                  markerFile = system.file("extdata/SxM", "SxM_geno.txt",
                                        package = "statgenIBD"),
                  mapFile = system.file("extdata/SxM", "SxM_map.txt",
                                        package = "statgenIBD"))

## Get probabilities for a single marker.
probOne &lt;- getProbs(IBDprob = SxMIBD,
                    markers = "plc")
head(probOne)

## Get probabilities for a multiple markers.
probMult &lt;- getProbs(IBDprob = SxMIBD,
                     markers = c("plc", "tuba1"))
head(probMult)

</code></pre>

<hr>
<h2 id='pedPlot'>Helper function for plotting the pedigree</h2><span id='topic+pedPlot'></span>

<h3>Description</h3>

<p>Helper function for plotting the pedigree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedPlot(pedigree, offSpring, popType, genoCross, multiCross = NULL, title)
</code></pre>

<hr>
<h2 id='plot.IBDprob'>Plot function for objects of class IBDprob</h2><span id='topic+plot.IBDprob'></span>

<h3>Description</h3>

<p>Creates a plot for an object of class <code>IBDprob</code>. Six types of plot can
be made:
</p>

<ul>
<li> <p><code>singleGeno</code> A plot for a single genotype showing the IBD
probabilities for all parents across the genome.
</p>
</li>
<li> <p><code>allGeno</code> A plot showing for all genotypes the IBD
probabilities of the parent with the highest probability per marker.
</p>
</li>
<li> <p><code>pedigree</code> A plot showing the structure of the pedigree of
the population.
</p>
</li>
<li> <p><code>map</code> A plot of the genetic map showing the length of the
chromosomes and the positions of the markers.
</p>
</li>
<li> <p><code>meanProbs</code> A plot showing the coverage of each parent across
the population.
</p>
</li>
<li> <p><code>totalCoverage</code> A plot showing the total coverage of each
parent.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBDprob'
plot(
  x,
  ...,
  plotType = c("singleGeno", "allGeno", "pedigree", "map", "meanProbs", "totalCoverage"),
  genotype,
  chr = NULL,
  title = NULL,
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.IBDprob_+3A_x">x</code></td>
<td>
<p>An object of class <code>IBDprob</code>.</p>
</td></tr>
<tr><td><code id="plot.IBDprob_+3A_...">...</code></td>
<td>
<p>Further arguments. Unused.</p>
</td></tr>
<tr><td><code id="plot.IBDprob_+3A_plottype">plotType</code></td>
<td>
<p>A character string indicating the type of plot that should
be made.</p>
</td></tr>
<tr><td><code id="plot.IBDprob_+3A_genotype">genotype</code></td>
<td>
<p>A character string indicating the genotype for which the
plot should be made. Only for <code>plotType = "singleGeno"</code>.</p>
</td></tr>
<tr><td><code id="plot.IBDprob_+3A_chr">chr</code></td>
<td>
<p>A character vector indicating the chromosomes to which the
coverage should be restricted. Only for <code>plotType = "meanProbs"</code> and
<code>plotType = "totalCoverage"</code>. If <code>NULL</code> all chromosomes are
included.</p>
</td></tr>
<tr><td><code id="plot.IBDprob_+3A_title">title</code></td>
<td>
<p>A character string, the title of the plot.</p>
</td></tr>
<tr><td><code id="plot.IBDprob_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code>, only a ggplot object is invisibly returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object is invisibly returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Compute IBD probabilities for Steptoe Morex.
## Add extra evaluation positions in between exiting marker positions
## to assure evaluation positions are at most 2 cM apart.
SxMIBD_Ext &lt;- calcIBD(popType = "DH",
                      markerFile = system.file("extdata/SxM", "SxM_geno.txt",
                                              package = "statgenIBD"),
                      mapFile = system.file("extdata/SxM", "SxM_map.txt",
                                           package = "statgenIBD"),
                      evalDist = 2)

## Plot results for genotype dh005.
plot(SxMIBD_Ext,
     plotType = "singleGeno",
     genotype = "dh005")

## Plot results for all genotypes.
plot(SxMIBD_Ext,
     plotType = "allGeno")

## Plot structure of the pedigree.
plot(SxMIBD_Ext,
     plotType = "pedigree")

## Plot genetic map.
plot(SxMIBD_Ext,
     plotType = "map")

## Plot coverage across population.
plot(SxMIBD_Ext,
     plotType = "meanProbs")

## Plot total coverage.
plot(SxMIBD_Ext,
     plotType = "totalCoverage")

## End(Not run)

</code></pre>

<hr>
<h2 id='readIBDs'>Read IBD probabilities from file</h2><span id='topic+readIBDs'></span>

<h3>Description</h3>

<p>Reads IBD probabilities from a plain text, tab-delimited .txt or .ibd file.
Information about the file format can be found in the vignette (
<code>vignette("IBDFileFormat", package = "statgenIBD")</code>). A data.frame with
the map must be specified as well.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readIBDs(infile, map)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readIBDs_+3A_infile">infile</code></td>
<td>
<p>A character string specifying the path of the input file.
Compressed files with extension &quot;.gz&quot; or &quot;.bz2&quot; are supported as well.</p>
</td></tr>
<tr><td><code id="readIBDs_+3A_map">map</code></td>
<td>
<p>A data.frame with columns <code>chr</code> for chromosome and
<code>pos</code> for position. Positions should be in centimorgan (cM). They
should not be cumulative over the chromosomes. Other columns are ignored.
Marker names should be in the row names. These should match the marker names
in the input file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>IBDprob</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Read map for Steptoe Morex.
SxMmap &lt;- read.delim(system.file("extdata/SxM", "SxM_map.txt",
                               package = "statgenIBD"), header = FALSE)
rownames(SxMmap) &lt;- SxMmap$V1
SxMmap &lt;- SxMmap[, -1]
colnames(SxMmap) &lt;- c("chr", "pos")


## Read IBD probabilities for Steptoe Morex.
SxMIBD &lt;- readIBDs(system.file("extdata/SxM", "SxM_IBDs.txt",
                               package = "statgenIBD"),
                  map = SxMmap)

## Print summary.
summary(SxMIBD)

</code></pre>

<hr>
<h2 id='readRABBIT'>Read IBD probabilities</h2><span id='topic+readRABBIT'></span>

<h3>Description</h3>

<p>Read a file with IBD probabilities computed by the RABBIT software package,
see <a href="https://github.com/Biometris/RABBIT">RABBIT</a> for details.
It is possible to additionally read the pedigree file that is also used by
RABBIT. Reading this file allows for plotting the pedigree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readRABBIT(infile, pedFile = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readRABBIT_+3A_infile">infile</code></td>
<td>
<p>A character string, a link to a .csv file with IBD
probabilities. Compressed .csv files with extension &quot;.gz&quot; or &quot;.bz2&quot; are
supported as well.</p>
</td></tr>
<tr><td><code id="readRABBIT_+3A_pedfile">pedFile</code></td>
<td>
<p>A character string, a link to a .csv file with pedigree
information as used by RABBIT as input. Compressed .csv files with extension
&quot;.gz&quot; or &quot;.bz2&quot; are supported as well.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>IBDprob</code> object with map and markers corresponding to the
imported information in the imported .csv file.
</p>


<h3>References</h3>

<p>Zheng, Chaozhi, Martin P Boer, and Fred A Van Eeuwijk.
“Recursive Algorithms for Modeling Genomic Ancestral Origins in a Fixed
Pedigree.” G3 Genes|Genomes|Genetics 8 (10): 3231–45.
https://doi.org/10.1534/G3.118.200340.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Read RABBIT data for barley.
genoFile &lt;- system.file("extdata/barley", "barley_magicReconstruct.zip",
                       package = "statgenIBD")
barleyIBD &lt;- readRABBIT(unzip(genoFile, exdir = tempdir()))

## End(Not run)

</code></pre>

<hr>
<h2 id='singleGenoPlot'>Helper function for plotting probabilities for a single genotype</h2><span id='topic+singleGenoPlot'></span>

<h3>Description</h3>

<p>Helper function for plotting probabilities for a single genotype
</p>


<h3>Usage</h3>

<pre><code class='language-R'>singleGenoPlot(markers, map, parents, genotype, title)
</code></pre>

<hr>
<h2 id='summary.IBDprob'>Summary function for objects of class IBDprob</h2><span id='topic+summary.IBDprob'></span>

<h3>Description</h3>

<p>Prints a short summary for objects of class <code>IBDprob</code>. The
summary consists of the population type, number of evaluation points,
number of individuals and names of the parents in the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'IBDprob'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.IBDprob_+3A_object">object</code></td>
<td>
<p>An object of class <code>IBDprob</code>.</p>
</td></tr>
<tr><td><code id="summary.IBDprob_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, a summary is printed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute IBD probabilities for Steptoe Morex.
SxMIBD &lt;- calcIBD(popType = "DH",
                  markerFile = system.file("extdata/SxM", "SxM_geno.txt",
                                          package = "statgenIBD"),
                  mapFile = system.file("extdata/SxM", "SxM_map.txt",
                                        package = "statgenIBD"))

## Print summary
summary(SxMIBD)

</code></pre>

<hr>
<h2 id='writeFlapjack'>Write to Flapjack format</h2><span id='topic+writeFlapjack'></span>

<h3>Description</h3>

<p>Export the results of an IBD calculation to Flapjack format so it can be
visualized there.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeFlapjack(
  IBDprob,
  outFileMap = "ibd_map.txt",
  outFileGeno = "ibd_geno.txt"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeFlapjack_+3A_ibdprob">IBDprob</code></td>
<td>
<p>An object of class <code>IBDprob</code>.</p>
</td></tr>
<tr><td><code id="writeFlapjack_+3A_outfilemap">outFileMap</code></td>
<td>
<p>A character string, the full path to the output map file.</p>
</td></tr>
<tr><td><code id="writeFlapjack_+3A_outfilegeno">outFileGeno</code></td>
<td>
<p>A character string, the full path to the output genotype
file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No output. Output files are written to a folder.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute IBD probabilities for Steptoe Morex.
SxMIBD &lt;- calcIBD(popType = "DH",
                  markerFile = system.file("extdata/SxM", "SxM_geno.txt",
                                          package = "statgenIBD"),
                  mapFile = system.file("extdata/SxM", "SxM_map.txt",
                                       package = "statgenIBD"))

## Write output in Flapjack format to temporary files.
writeFlapjack(SxMIBD,
              outFileMap = tempfile(fileext = ".txt"),
              outFileGeno = tempfile(fileext = ".txt"))

</code></pre>

<hr>
<h2 id='writeIBDs'>Write IBD probabilities to file.</h2><span id='topic+writeIBDs'></span>

<h3>Description</h3>

<p>Writes IBD probabilities to a plain text, tab-delimited .txt or .ibd file.
Information about the file format can be found in the vignette (
<code>vignette("IBDFileFormat", package = "statgenIBD")</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeIBDs(IBDprob, outFile, decimals = 6, minProb = 0, compress = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeIBDs_+3A_ibdprob">IBDprob</code></td>
<td>
<p>An object of class <code>IBDprob</code> containing the IBD
probabilities.</p>
</td></tr>
<tr><td><code id="writeIBDs_+3A_outfile">outFile</code></td>
<td>
<p>A character string specifying the path of the output file.</p>
</td></tr>
<tr><td><code id="writeIBDs_+3A_decimals">decimals</code></td>
<td>
<p>An integer value specifying the number of decimals to include
in writing the output file.</p>
</td></tr>
<tr><td><code id="writeIBDs_+3A_minprob">minProb</code></td>
<td>
<p>A numerical value between zero and 1 / number of parents,
specifying the minimum probability cutoff value. Probabilities below this
cutoff are set to zero and other probabilities are rescaled to make sure that
the probabilities sum up to one.</p>
</td></tr>
<tr><td><code id="writeIBDs_+3A_compress">compress</code></td>
<td>
<p>Should the output be compressed to .gz format?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No output. The output file is created as a result of calling this
function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute IBD probabilities for Steptoe Morex.
SxMIBD &lt;- calcIBD(popType = "DH",
                 markerFile = system.file("extdata/SxM", "SxM_geno.txt",
                                         package = "statgenIBD"),
                 mapFile = system.file("extdata/SxM", "SxM_map.txt",
                                      package = "statgenIBD"))

## Write IBDs to temporary files.
writeIBDs(IBDprob = SxMIBD, outFile = tempfile(fileext = ".txt"))

## Write IBDs to file, set values &lt;0.05 to zero and only print 3 decimals.
writeIBDs(IBDprob = SxMIBD, outFile = tempfile(fileext = ".txt"),
         decimals = 3, minProb = 0.05)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
