<!DOCTYPE html><html><head><title>Help for package statVisual</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {statVisual}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#barPlot'>
<p>Compare Groups Based on Barplots Across Time</p></a></li>
<li><a href='#BiAxisErrBar'>
<p>Compare Patterns of Two Outcomes in One Scatter Plot</p></a></li>
<li><a href='#Box'>
<p>Compare Groups Based on Boxplots Across Time</p></a></li>
<li><a href='#BoxROC'>
<p>Compare Boxplots with ROC Curve</p></a></li>
<li><a href='#cv_glmnet_plot'>
<p>Plot the Cross-Validation Curve Produced by cv.glmnet</p></a></li>
<li><a href='#Den'>
<p>Compare Groups Based on Density Plots</p></a></li>
<li><a href='#Dendro'>
<p>Compare Groups Based on Dendrogram</p></a></li>
<li><a href='#diffCorDat'>
<p>A Dataset for Differential Correlation Analysis</p></a></li>
<li><a href='#ErrBar'>
<p>Compare Groups Based on dotplots Across Time</p></a></li>
<li><a href='#esSim'>
<p>A Simulated Gene Expression Dataset</p></a></li>
<li><a href='#genoSim'>
<p>An ExpressionSet Object Storing Simulated Genotype Data</p></a></li>
<li><a href='#Heat'>
<p>Heatmap with Row Names Colored by Group</p></a></li>
<li><a href='#Hist'>
<p>Compare Groups Based on Histograms</p></a></li>
<li><a href='#ImpPlot'>
<p>Plot of Variable Importance</p></a></li>
<li><a href='#iprcomp'>
<p>Improved Function for Obtaining Principal Components</p></a></li>
<li><a href='#LinePlot'>
<p>Compare Groups Based on Trajectory Plots</p></a></li>
<li><a href='#longDat'>
<p>A Simulated Dataset for Longitudinal Data Analysis</p></a></li>
<li><a href='#PCA_score'>
<p>Scatter Plot of 2 Specified Principal Components</p></a></li>
<li><a href='#PVCA'>
<p>Principal Variance Component Analysis (PVCA)</p></a></li>
<li><a href='#stackedBarPlot'>
<p>Draw Stacked Bar Plots</p></a></li>
<li><a href='#statVisual'>
<p>The Wrapper Function Incorporating All Wrapper Functions in statVisual</p></a></li>
<li><a href='#Volcano'>
<p>Volcano Plot</p></a></li>
<li><a href='#XYscatter'>
<p>Compare Groups Based on Scatter Plots</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Visualization Tools</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-02-19</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), Biobase, ggplot2, stats</td>
</tr>
<tr>
<td>Imports:</td>
<td>GGally, dplyr, reshape2, ggdendro, tibble, ggfortify, ggrepel,
gbm, forestplot, rpart.plot, grDevices, gridExtra, pROC,
factoextra, methods, pvca, limma, randomForest, glmnet, knitr,
rmarkdown, pheatmap, RColorBrewer, graphics, tidyverse,
magrittr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Description:</td>
<td>Visualization functions in the applications of translational medicine (TM) and biomarker (BM) development to compare groups by statistically visualizing data and/or results of analyses, such as visualizing data by displaying in one figure different groups' histograms, boxplots, densities, scatter plots, error-bar plots, or trajectory plots, by displaying scatter plots of top principal components or dendrograms with data points colored based on group information, or visualizing volcano plots to check the results of whole genome analyses for gene differential expression. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-02-20 18:59:53 UTC; zwf</td>
</tr>
<tr>
<td>Author:</td>
<td>Wenfei Zhang [aut, cre],
  Weiliang Qiu [aut, ctb],
  Xuan Lin [aut, ctb],
  Donghui Zhang [aut, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wenfei Zhang &lt;Wenfeizhang@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-02-20 19:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='barPlot'>
Compare Groups Based on Barplots Across Time
</h2><span id='topic+barPlot'></span>

<h3>Description</h3>

<p>This function is to compare groups using barplots at each time point. 
In addition, line segments are used to connect the mean/median of each barplot 
of the same group across time to show the differences between 
the mean trajectories. 
Also, for each barplot 
the barplot of mean <code class="reqn">+/-</code> standard error will be plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barPlot(
    data, 
    x = NULL, 
    y, 
    group = NULL,
    semFlag = TRUE,
    xFlag = FALSE, 
    bar.width = 0.5, 
    dodge.width = 0.8, 
    jitter = FALSE, 
    jitter.alpha = 0.7, 
    jitter.width = 0.1, 
    line = NULL, 
    line.color = "black", 
    xlab = x, 
    ylab = line, 
    theme_classic = TRUE, 
    group.lab = group, 
    title = "bar plots", 
    xLevel = NULL,
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="barPlot_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_x">x</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the first grouping variable
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_y">y</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable on y axis
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The barplots will be drawn for each of the subject group within each category of <code>x</code>. 
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_semflag">semFlag</code></td>
<td>

<p>logical. Indicate if sem or se should be used to draw error bar
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_xflag">xFlag</code></td>
<td>

<p>logical. Indicate if <code>x</code> should be treated as continuous (<code>xFlag=TRUE</code>)
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_bar.width">bar.width</code></td>
<td>

<p>numeric. error bar width
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_dodge.width">dodge.width</code></td>
<td>

<p>numeric. dodge width for error bar and jitter (prevent overlapping)
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_jitter">jitter</code></td>
<td>

<p>logical, plot jitter or not, default TRUE
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_jitter.alpha">jitter.alpha</code></td>
<td>

<p>numeric. jitter transparency
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_jitter.width">jitter.width</code></td>
<td>

<p>numeric. jitter width in error bar
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_line">line</code></td>
<td>

<p>character. line connect error bar, default uses mean, can be set as 'median', NULL (no line)
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_line.color">line.color</code></td>
<td>

<p>character. connection line color, only available when group = NULL
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_xlab">xlab</code></td>
<td>

<p>character. x axis label
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_ylab">ylab</code></td>
<td>

<p>character. y axis label
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_group.lab">group.lab</code></td>
<td>

<p>character. label of group variable
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_title">title</code></td>
<td>

<p>character. title of plot
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_xlevel">xLevel</code></td>
<td>

<p>character. A character vector indicating the order of the elements of <code>x</code> to be shown on x-axis if <code>is.null(x)==FALSE</code>.
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="barPlot_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the following 9 elements:
&ldquo;data&rdquo;, &ldquo;layers&rdquo;, &ldquo;scales&rdquo;, &ldquo;mapping&rdquo;,
&ldquo;theme&rdquo;, &ldquo;coordinates&rdquo;, &ldquo;facet&rdquo;, &ldquo;plot_env&rdquo;,
&ldquo;labels&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(longDat)

print(dim(longDat))
print(longDat[1:3,])

print(table(longDat$time, useNA = "ifany"))
print(table(longDat$grp, useNA = "ifany"))
print(table(longDat$sid, useNA = "ifany"))

print(table(longDat$time, longDat$grp))

statVisual(type = 'barPlot', 
  data = longDat, 
  x = 'time', 
  y = 'y', 
  group = 'grp',
  title = "Bar plots across time") 


barPlot(
  data = longDat, 
  x = 'time', 
  y = 'y', 
  group = 'grp',
  title = "Bar plots across time") 


</code></pre>

<hr>
<h2 id='BiAxisErrBar'>
Compare Patterns of Two Outcomes in One Scatter Plot
</h2><span id='topic+BiAxisErrBar'></span>

<h3>Description</h3>

<p>Compare patterns of two outcomes with different scales across the range of the common predictor using error bar plots. Each bar plot displays mean <code class="reqn">+/-</code> standard error. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BiAxisErrBar(dat, 
	     group, 
	     y.left, 
	     y.right, 
	     title = "Bi-Axis Error Bar Plot",
	     xlab = group, 
	     ylab.left = y.left, 
	     ylab.right = y.right, 
	     legendLabel = "y axis variables",
	     delta = NULL, 
	     cvThresh = 0.01, 
             Ntick = 5,
             semFlag = TRUE, #semFlag = FALSE if SE is required
	     GroupLevel = NULL,
	     addThemeFlag = FALSE
	     )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BiAxisErrBar_+3A_dat">dat</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_group">group</code></td>
<td>

<p>character. A categorical variable in <code>data</code> that indicates the predictor. 
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_y.left">y.left</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the first outcome variable, the error bar plot of which will be drawn on the left side.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_y.right">y.right</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the second outcome variable, the error bar plot of which will be drawn on the right side.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_title">title</code></td>
<td>

<p>character. title of the plot.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_xlab">xlab</code></td>
<td>

<p>character. Label for the x-axis.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_ylab.left">ylab.left</code></td>
<td>

<p>character. Label for the left-side y-axis.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_ylab.right">ylab.right</code></td>
<td>

<p>character. Label for the right-side y-axis.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_legendlabel">legendLabel</code></td>
<td>

<p>character. Legend label.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_delta">delta</code></td>
<td>

<p>numeric. A small number so that the second error bar plot will shift <code>delta</code> distance from the first error bar plot.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_cvthresh">cvThresh</code></td>
<td>

<p>numeric. A small positive number. If the coefficient of variation (CV)
is smaller than <code>cvThresh</code>, then the scaling factor will
be set to one.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_ntick">Ntick</code></td>
<td>

<p>integer. Number of ticks on the two y-axes.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_semflag">semFlag</code></td>
<td>

<p>logical. Indicating if standard error of the mean (<code>semFlag = TRUE</code>) or standard error (<code>semFlag = FALSE</code>) will be used to construct the error bars.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_grouplevel">GroupLevel</code></td>
<td>

<p>A vector of unique values of <code>group</code> indicating the order
of <code>group</code> shown in x-axis.
</p>
</td></tr>
<tr><td><code id="BiAxisErrBar_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>, <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(tidyverse)
library(ggplot2)

print(head(mtcars))

print(table(mtcars$gear, useNA="ifany"))

statVisual(type = "BiAxisErrBar",
  dat= mtcars,
  group = "gear",
  y.left = "mpg",
  y.right = "wt")



BiAxisErrBar(
  dat = mtcars,
  group = "gear",
  y.left = "mpg",
  y.right = "wt")

</code></pre>

<hr>
<h2 id='Box'>
Compare Groups Based on Boxplots Across Time
</h2><span id='topic+Box'></span>

<h3>Description</h3>

<p>This function is to compare groups using boxplots at each time point. 
In addition, line segments are used to connect the mean/median of each boxplot 
of the same group across time to show the differences between 
the mean trajectories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Box(
    data, 
    x = NULL, 
    y, 
    group = NULL, 
    fill = NULL, 
    theme_classic = TRUE, 
    fill.alpha = 0.7, 
    box.width = 0.5, 
    dodge.width = 0.8, 
    jitter = TRUE, 
    jitter.alpha = 0.7, 
    jitter.width = 0.2, 
    point.size = 1, 
    xlab = x, 
    ylab = y, 
    group.lab = group, 
    fill.lab = group, 
    title = "Boxplot", 
    line = "mean", 
    line.color = "black", 
    xLevel = NULL,
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Box_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="Box_+3A_x">x</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the first grouping variable (e.g. observation time)
</p>
</td></tr>
<tr><td><code id="Box_+3A_y">y</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable on y axis
</p>
</td></tr>
<tr><td><code id="Box_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups (e.g., treatment group). The boxplots will be drawn for each of the subject group within each category of <code>x</code>. 
</p>
</td></tr>
<tr><td><code id="Box_+3A_fill">fill</code></td>
<td>

<p>boxplot inside color indicated by the categories of <code>group</code>
</p>
</td></tr>
<tr><td><code id="Box_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="Box_+3A_fill.alpha">fill.alpha</code></td>
<td>

<p>boxplot transparency
</p>
</td></tr>
<tr><td><code id="Box_+3A_box.width">box.width</code></td>
<td>

<p>boxplot width
</p>
</td></tr>
<tr><td><code id="Box_+3A_dodge.width">dodge.width</code></td>
<td>

<p>dodge width for boxplot and jitter (prevent overlapping)
</p>
</td></tr>
<tr><td><code id="Box_+3A_jitter">jitter</code></td>
<td>

<p>logical. plot jitter or not, default TRUE
</p>
</td></tr>
<tr><td><code id="Box_+3A_jitter.alpha">jitter.alpha</code></td>
<td>

<p>jitter transparency
</p>
</td></tr>
<tr><td><code id="Box_+3A_jitter.width">jitter.width</code></td>
<td>

<p>jitter width in boxplot
</p>
</td></tr>
<tr><td><code id="Box_+3A_point.size">point.size</code></td>
<td>

<p>size of a jitter point
</p>
</td></tr>
<tr><td><code id="Box_+3A_xlab">xlab</code></td>
<td>

<p>character. x axis label
</p>
</td></tr>
<tr><td><code id="Box_+3A_ylab">ylab</code></td>
<td>

<p>character. y axis label
</p>
</td></tr>
<tr><td><code id="Box_+3A_group.lab">group.lab</code></td>
<td>

<p>label of group variable
</p>
</td></tr>
<tr><td><code id="Box_+3A_fill.lab">fill.lab</code></td>
<td>

<p>label of fill variable
</p>
</td></tr>
<tr><td><code id="Box_+3A_title">title</code></td>
<td>

<p>character. title of plot
</p>
</td></tr>
<tr><td><code id="Box_+3A_line">line</code></td>
<td>

<p>line connect boxes, default plot mean, can be set as 'median', or NULL (no line)
</p>
</td></tr>
<tr><td><code id="Box_+3A_line.color">line.color</code></td>
<td>

<p>connection line color, only available when group = NULL
</p>
</td></tr>
<tr><td><code id="Box_+3A_xlevel">xLevel</code></td>
<td>

<p>character. A character vector indicating the order of the elements of <code>x</code> to be shown on x-axis if <code>is.null(x)==FALSE</code>.
</p>
</td></tr>
<tr><td><code id="Box_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="Box_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following 9 elements: <code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>, <code>theme</code>, <code>coordinates</code>, <code>facet</code>,
<code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(dplyr)

data(longDat)

print(dim(longDat))
print(longDat[1:3,])

print(table(longDat$time, useNA = "ifany"))
print(table(longDat$grp, useNA = "ifany"))
print(table(longDat$sid, useNA = "ifany"))

print(table(longDat$time, longDat$grp))

statVisual(type = 'Box', 
           data = longDat, 
           x = 'time', 
           y = 'y', 
           group = 'grp',
	   title = "Boxplots across time") 

Box( 
    data = longDat, 
    x = 'time', 
    y = 'y', 
    group = 'grp',
    title = "Boxplots across time") 


</code></pre>

<hr>
<h2 id='BoxROC'>
Compare Boxplots with ROC Curve
</h2><span id='topic+BoxROC'></span>

<h3>Description</h3>

<p>Compare boxplots with ROC curve.
The value of the variable <code>y</code> will be jittered shown on the boxplots. The area under ROC curve will also be calculated and shown in the plot of ROC curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BoxROC(
    data, 
    group.var, 
    y, 
    box.xlab = group.var, 
    box.ylab = y, 
    box.group.lab = group.var, 
    jitter.alpha = 0.8, 
    jitter.width = 0.1, 
    point.size = 3, 
    roc.xlab = "Specificity", 
    roc.ylab = "Sensitivity",
    addThemeFlag = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BoxROC_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_group.var">group.var</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the two subject groups. It also indicates
the color of the two boxplots.
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_y">y</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable,
for which the box will be drawn. 
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_box.xlab">box.xlab</code></td>
<td>

<p>character. boxplot x axis label (default: <code>group.var</code>)
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_box.ylab">box.ylab</code></td>
<td>

<p>character. boxplot y axis label (default: <code>y</code>)
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_box.group.lab">box.group.lab</code></td>
<td>

<p>character. boxplot legend label (default: <code>group.var</code>)
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_jitter.alpha">jitter.alpha</code></td>
<td>

<p>numeric. transparency of jitters
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_jitter.width">jitter.width</code></td>
<td>

<p>numeric. width of jitters
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_point.size">point.size</code></td>
<td>

<p>size of a jitter point
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_roc.xlab">roc.xlab</code></td>
<td>

<p>character. roc curve x axis label (default: <code>Specificities</code>)
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_roc.ylab">roc.ylab</code></td>
<td>

<p>character. roc curve y axis label (default: <code>Sensitivities</code>)
</p>
</td></tr>
<tr><td><code id="BoxROC_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following 12 elements: <code>grobs</code>, <code>layout</code>,
<code>widths</code>, <code>heights</code>, <code>respect</code>, <code>rownames</code>,
<code>colnames</code>, <code>name</code>,
<code>gp</code>, <code>vp</code>, <code>children</code>, <code>childrenOrder</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(dplyr)
library(gridExtra)

data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first probe which is over-expressed in cases
pDat$probe1 = dat[1,]

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))

statVisual(type = 'BoxROC', 
           data = pDat, 
           group = 'grp', 
           y = 'probe1', 
           point.size = 1)

BoxROC(
  data = pDat,
  group = 'grp', 
  y = 'probe1', 
  point.size = 1)

</code></pre>

<hr>
<h2 id='cv_glmnet_plot'>
Plot the Cross-Validation Curve Produced by cv.glmnet
</h2><span id='topic+cv_glmnet_plot'></span>

<h3>Description</h3>

<p>Plots the cross-validation curve, and upper and lower standard error curves, as a function of the values of the tuning parameter lambda.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv_glmnet_plot(x, 
	       y, 
	       family = "binomial", 
	       addThemeFlag = TRUE, 
	       ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv_glmnet_plot_+3A_x">x</code></td>
<td>

<p>a matrix with rows are subjects and columns are numeric variables (predictors). No missing values are allowed.
</p>
</td></tr>
<tr><td><code id="cv_glmnet_plot_+3A_y">y</code></td>
<td>

<p>a vector of response. The number of elements of <code>y</code> is the same as the number of rows of <code>x</code>.
</p>
</td></tr>
<tr><td><code id="cv_glmnet_plot_+3A_family">family</code></td>
<td>

<p>character. Indicating response type. see the description in <code><a href="glmnet.html#topic+glmnet">glmnet</a></code>.   
</p>
</td></tr>
<tr><td><code id="cv_glmnet_plot_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="cv_glmnet_plot_+3A_...">...</code></td>
<td>

<p>other input parameters for <code><a href="glmnet.html#topic+glmnet">glmnet</a></code> function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>       <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(dplyr)
library(tibble)
library(glmnet)

data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first 6 probes (3 OE probes, 2 UE probes, and 1 NE probe)
pDat$probe1 = dat[1,]
pDat$probe2 = dat[2,]
pDat$probe3 = dat[3,]
pDat$probe4 = dat[4,]
pDat$probe5 = dat[5,]
pDat$probe6 = dat[6,]

print(pDat[1:2, ])

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))


statVisual(type = "cv_glmnet_plot",
           x = as.matrix(pDat[, c(3:8)]), 
           y = pDat$grp, 
           family = "binomial")

cv_glmnet_plot(x = as.matrix(pDat[, c(3:8)]), 
               y = pDat$grp, 
               family = "binomial")

</code></pre>

<hr>
<h2 id='Den'>
Compare Groups Based on Density Plots
</h2><span id='topic+Den'></span>

<h3>Description</h3>

<p>Compare groups based on density plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Den(
    data, 
    y, 
    group = NULL, 
    fill = group, 
    border.color = NULL, 
    inner.color = NULL, 
    theme_classic = TRUE, 
    xlab = y, 
    ylab = "density", 
    group.lab = group, 
    title = "Density plot", 
    alpha = 0.3, 
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Den_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="Den_+3A_y">y</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable,
for which the histogram will be drawn. The string <code>y</code> can also indicate a function of the variable, e.g., <code class="reqn">\log(y)</code>.
</p>
</td></tr>
<tr><td><code id="Den_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The density will be drawn for each of the subject group. It also indicates the border colors of the densities.
</p>
</td></tr>
<tr><td><code id="Den_+3A_fill">fill</code></td>
<td>

<p>grouping variable, density inside color
</p>
</td></tr>
<tr><td><code id="Den_+3A_border.color">border.color</code></td>
<td>

<p>density border color, only available when group &amp; fill are NULL
</p>
</td></tr>
<tr><td><code id="Den_+3A_inner.color">inner.color</code></td>
<td>

<p>density inside color, only available when group &amp; fill are NULL
</p>
</td></tr>
<tr><td><code id="Den_+3A_theme_classic">theme_classic</code></td>
<td>

<p>Use classic background without grids (default: FALSE)
</p>
</td></tr>
<tr><td><code id="Den_+3A_xlab">xlab</code></td>
<td>

<p>x axis label
</p>
</td></tr>
<tr><td><code id="Den_+3A_ylab">ylab</code></td>
<td>

<p>y axis label
</p>
</td></tr>
<tr><td><code id="Den_+3A_group.lab">group.lab</code></td>
<td>

<p>label of group variable
</p>
</td></tr>
<tr><td><code id="Den_+3A_title">title</code></td>
<td>

<p>title of plot
</p>
</td></tr>
<tr><td><code id="Den_+3A_alpha">alpha</code></td>
<td>

<p>transparency of density inside color
</p>
</td></tr>
<tr><td><code id="Den_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="Den_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>, <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first probe which is over-expressed in cases
pDat$probe1 = dat[1,]

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))

statVisual(type = 'Den', 
           data = pDat, 
           y = 'probe1', 
           group = 'grp') 

Den( 
    data = pDat, 
    y = 'probe1', 
    group = 'grp') 

</code></pre>

<hr>
<h2 id='Dendro'>
Compare Groups Based on Dendrogram
</h2><span id='topic+Dendro'></span>

<h3>Description</h3>

<p>Compare groups based on dendrogram. The nodes of the dendrogram will be colored by group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dendro(
    x, 
    group = NULL, 
    xlab = NULL, 
    ylab = NULL, 
    title = NULL, 
    cor.use = "pairwise.complete.obs", 
    cor.method = "pearson", 
    distance = "rawdata", 
    distance.method = "euclidean", 
    hclust.method = "complete", 
    yintercept = NULL, 
    theme_classic = TRUE, 
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Dendro_+3A_x">x</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The nodes of the dendrogram will be colored by info provided by <code>group</code>.
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_xlab">xlab</code></td>
<td>

<p>x axis label
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_ylab">ylab</code></td>
<td>

<p>y axis label
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_title">title</code></td>
<td>

<p>title of the plot
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_cor.use">cor.use</code></td>
<td>

<p>character. Indicate which data will be used to compute correlation coefficients. It can take values &ldquo;everything&rdquo;, &ldquo;all.obs&rdquo;, 
&ldquo;complete.obs&rdquo;, &ldquo;na.or.complete&rdquo;, &ldquo;pairwise.complete.obs&rdquo;.
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_cor.method">cor.method</code></td>
<td>

<p>character. Indicate which type of correlation coefficients will be calculated: &ldquo;pearson&rdquo;, &ldquo;kendall&rdquo;, &ldquo;spearman&rdquo;.
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_distance">distance</code></td>
<td>

<p>character. Indicate which type of data will be used to calculate distance: &ldquo;rawdata&rdquo; (i.e., using raw data to calculate distance),
&ldquo;cor&rdquo; (i.e., using correlation coefficients as distance),
&ldquo;1-cor&rdquo; (i.e., using (<code class="reqn">1-</code>correlation coefficients) 
as distance),
&ldquo;1-|cor|&rdquo; (i.e., using (<code class="reqn">1-</code>|correlation coefficients|) as distance).
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_distance.method">distance.method</code></td>
<td>

<p>character. Available when &lsquo;distance = &quot;rawdata&quot;&rsquo;. Indicate the definition of distance:
distance used in calculate dist
&ldquo;rawdata&rdquo; (i.e., using raw data to calculate distance),
&ldquo;cor&rdquo; (i.e., using correlation coefficients as distance),
&ldquo;1-cor&rdquo; (i.e., using (<code class="reqn">1-</code>correlation coefficients) as distance),
&ldquo;1-|cor|&rdquo; (i.e., using (<code class="reqn">1-</code>|correlation coefficients|) as distance).
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_hclust.method">hclust.method</code></td>
<td>

<p>character. Indicate which agglomeration method will be used to perform
hierarchical clustering.
This should be (an unambiguous abbreviation of) one of 
&ldquo;ward.D&rdquo;, &ldquo;ward.D2&rdquo;, &ldquo;single&rdquo;, 
&ldquo;complete&rdquo;, &ldquo;average&rdquo;, &ldquo;mcquitty&rdquo;, 
&ldquo;median&rdquo;, or &ldquo;centroid&rdquo;. Please refer to
<code><a href="stats.html#topic+hclust">hclust</a></code>.
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_yintercept">yintercept</code></td>
<td>

<p>numeric. A line indicating the height of the dendrogram, for example,
indicating where the dendrogram should be cut to obtain clusters.
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="Dendro_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>       <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first 6 probes (3 OE probes, 2 UE probes, and 1 NE probe)
pDat$probe1 = dat[1,]
pDat$probe2 = dat[2,]
pDat$probe3 = dat[3,]
pDat$probe4 = dat[4,]
pDat$probe5 = dat[5,]
pDat$probe6 = dat[6,]

print(pDat[1:2, ])

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))

pDat$grp = factor(pDat$grp)

statVisual(type = 'Dendro', 
           x = pDat[, c(3:8)], 
           group = pDat$grp)

Dendro(
       x = pDat[, c(3:8)], 
       group = pDat$grp)



</code></pre>

<hr>
<h2 id='diffCorDat'>
A Dataset for Differential Correlation Analysis
</h2><span id='topic+diffCorDat'></span>

<h3>Description</h3>

<p>A dataset for differential correlation analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("diffCorDat")</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on the following 3 variables.
</p>

<dl>
<dt><code>probe1</code></dt><dd><p>numeric. expression level for probe1</p>
</dd>
<dt><code>probe2</code></dt><dd><p>numeric. expression level for probe2</p>
</dd>
<dt><code>grp</code></dt><dd><p>character. a factor with levels <code>cases</code> <code>controls</code></p>
</dd>
</dl>



<h3>Details</h3>

<p>The simulated data set contains expression levels of 2 gene probes for 50 cases and 50 controls. The expression levels of probe1 are generated from <code class="reqn">N(0, 1)</code>. The expression levels of probe2 for controls are also generated from <code class="reqn">N(0, 1)</code>. The expression levels of probe 2 for cases are generated from the formula <code class="reqn">probe2_{i} = -probe1_{i} + e_i</code>, <code class="reqn">i=1, \ldots, nCases</code>, where <code class="reqn">e_i\sim N(0, 0.3^2)</code>.
</p>
<p>That is, the expression levels of probe 1 and probe 2 are negatively correlated in cases, but not correlated in controls.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diffCorDat)

print(dim(diffCorDat))
print(diffCorDat[1:2,])
</code></pre>

<hr>
<h2 id='ErrBar'>
Compare Groups Based on dotplots Across Time
</h2><span id='topic+ErrBar'></span>

<h3>Description</h3>

<p>This function is to compare groups using dotplots at each time point. 
In addition, line segments are used to connect the mean/median of each dotplot 
of the same group across time to show the differences between 
the mean trajectories. 
Also, for each dotplot 
the barplot of mean <code class="reqn">+/-</code> standard error will be plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ErrBar(
    data, 
    x = NULL, 
    y, 
    group = NULL,
    semFlag = TRUE,
    xFlag = FALSE,
    bar.width = 0.5, 
    dodge.width = 0.8, 
    jitter = TRUE, 
    jitter.alpha = 0.7, 
    jitter.width = 0.1, 
    line = "mean", 
    line.color = "black", 
    xlab = x, 
    ylab = line, 
    theme_classic = TRUE, 
    group.lab = group, 
    title = "Dot plots", 
    xLevel = NULL,
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ErrBar_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_x">x</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the first grouping variable
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_y">y</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable on y axis
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The dotplots will be drawn for each of the subject group within each category of <code>x</code>. 
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_semflag">semFlag</code></td>
<td>

<p>logical. Indicate if sem or se should be used to draw error bar
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_xflag">xFlag</code></td>
<td>

<p>logical. Indicate if <code>x</code> should be treated as continuous (<code>xFlag=TRUE</code>)
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_bar.width">bar.width</code></td>
<td>

<p>numeric. error bar width
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_dodge.width">dodge.width</code></td>
<td>

<p>numeric. dodge width for error bar and jitter (prevent overlapping)
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_jitter">jitter</code></td>
<td>

<p>logical, plot jitter or not, default TRUE
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_jitter.alpha">jitter.alpha</code></td>
<td>

<p>numeric. jitter transparency
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_jitter.width">jitter.width</code></td>
<td>

<p>numeric. jitter width in error bar
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_line">line</code></td>
<td>

<p>character. line connect error bar, default uses mean, can be set as 'median', NULL (no line)
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_line.color">line.color</code></td>
<td>

<p>character. connection line color, only available when group = NULL
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_xlab">xlab</code></td>
<td>

<p>character. x axis label
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_ylab">ylab</code></td>
<td>

<p>character. y axis label
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_group.lab">group.lab</code></td>
<td>

<p>character. label of group variable
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_title">title</code></td>
<td>

<p>character. title of plot
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_xlevel">xLevel</code></td>
<td>

<p>character. A character vector indicating the order of the elements of <code>x</code> to be shown on x-axis if <code>is.null(x)==FALSE</code>.
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="ErrBar_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the following 9 elements:
&ldquo;data&rdquo;, &ldquo;layers&rdquo;, &ldquo;scales&rdquo;, &ldquo;mapping&rdquo;,
&ldquo;theme&rdquo;, &ldquo;coordinates&rdquo;, &ldquo;facet&rdquo;, &ldquo;plot_env&rdquo;,
&ldquo;labels&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(longDat)

print(dim(longDat))
print(longDat[1:3,])

print(table(longDat$time, useNA = "ifany"))
print(table(longDat$grp, useNA = "ifany"))
print(table(longDat$sid, useNA = "ifany"))

print(table(longDat$time, longDat$grp))

statVisual(type = 'ErrBar', 
  data = longDat, 
  x = 'time', 
  y = 'y', 
  group = 'grp',
  title = "Dot plots across time") 


ErrBar(
  data = longDat, 
  x = 'time', 
  y = 'y', 
  group = 'grp',
  title = "Dot plots across time") 


</code></pre>

<hr>
<h2 id='esSim'>
A Simulated Gene Expression Dataset
</h2><span id='topic+esSim'></span>

<h3>Description</h3>

<p>A simulated gene expression dataset for differential expression analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("esSim")</code></pre>


<h3>Format</h3>

<p>The format is:
Formal class 'ExpressionSet' [package &quot;Biobase&quot;] with expression levels
of 100 probes for 20 samples.
</p>
<p>The phenotype data contain 2 phenotype variables:
sid (subject id) and grp (group indicator: 1 stands for case; 0 stands for control).
</p>
<p>The feature data contain 4 feature variables:
probeid (probe id), gene (fake gene symbol), chr (fake chromosome number),
and memProbes (probe significance indicator: 1 stands for probes over-expressed (OE) in cases; -1 stands for probes under-expressed (UE) in cases; and 0 stands for non-differentially expressed (NE) probes). There are 3 OE probes,
2 UE probes, and 95 NE probes.
</p>


<h3>Details</h3>

<p>The dataset was generated based on the R code in the manual
of the function <code>lmFit</code> of the R Bioconductor package <code>limma</code>.
There are 100 probes and 20 samples (10 controls and 10 cases). The first 3 probes are over-expressed in cases. The 4-th and 5-th probes are under-expressed in cases. The remaining 95 probes are non-differentially expressed between cases and controls. Expression levels for 100 probes were first generated from normal distribution with mean 0 and standard deviation varying between probes (<code class="reqn">sd=0.3\sqrt{4/\chi^2_4}</code>). For the 3 OE probes, we add 2 to the expression levels of the 10 cases. For the 2 UE probes, we subtract 2 from the expression levels of the 10 cases.
</p>


<h3>References</h3>

<p>Please see the example in the manual for the function <code>lmFit</code> in the R Bioconductor package <code>limma</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esSim)

print(esSim)

###
dat=exprs(esSim)
print(dim(dat))
print(dat[1:2,])

###
pDat=pData(esSim)
print(dim(pDat))
print(pDat)

# subject group status
print(table(esSim$grp))

###
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2, ])

# probe's status of differential expression
print(table(fDat$memProbes))

</code></pre>

<hr>
<h2 id='genoSim'>
An ExpressionSet Object Storing Simulated Genotype Data
</h2><span id='topic+genoSim'></span>

<h3>Description</h3>

<p>An ExpressionSet object storing simulated genotype data with 10 SNPs and 100 subjects. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("genoSim")</code></pre>


<h3>Details</h3>

<p>The simulated genotype data contain 50 cases and 50 controls. Each subject has genotype data for 10 SNPs. The first 2 SNPs have different minor allele frequencies (MAFs) between cases and controls (MAF for cases is 0.4 and MAF for controls is 0.2). We assume Hardy Weinberg Equilibrium.
The remaining 8 SNPs have the same MAF (<code class="reqn">MAF=0.2</code>) in both cases and controls.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genoSim)

print(genoSim)
</code></pre>

<hr>
<h2 id='Heat'>
Heatmap with Row Names Colored by Group
</h2><span id='topic+Heat'></span>

<h3>Description</h3>

<p>Heatmap with row names colored by group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Heat(data, 
     group = NULL, 
     fontsize_row=10,
     fontsize_col=10, 
     scale = "none",
     cluster_rows = TRUE,
     cluster_cols = TRUE,
     color = colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))(100),
     angle_col = c("270", "0", "45", "90", "315"), 
     ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Heat_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects. Except the column indicating subject group, all columns of <code>data</code> should be numeric.
</p>
</td></tr>
<tr><td><code id="Heat_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The row names of the heatmap will be colored based on <code>group</code>.
</p>
</td></tr>
<tr><td><code id="Heat_+3A_fontsize_col">fontsize_col</code></td>
<td>

<p>x axis label font size
</p>
</td></tr>
<tr><td><code id="Heat_+3A_fontsize_row">fontsize_row</code></td>
<td>

<p>y axis label font size
</p>
</td></tr>
<tr><td><code id="Heat_+3A_scale">scale</code></td>
<td>

<p>character. Indicate how data will be scaled: 
&ldquo;none&rdquo; (i.e., no scaling), &ldquo;row&rdquo; (i.e., row scaled), 
&ldquo;column&rdquo; (i.e., column scaled).
</p>
</td></tr>
<tr><td><code id="Heat_+3A_cluster_rows">cluster_rows</code></td>
<td>

<p>logic. Indicates if rows should be clustered.
</p>
</td></tr>
<tr><td><code id="Heat_+3A_cluster_cols">cluster_cols</code></td>
<td>

<p>logic. Indicates if columns should be clustered.
</p>
</td></tr>
<tr><td><code id="Heat_+3A_color">color</code></td>
<td>

<p>vector indicating colors used in heatmap
</p>
</td></tr>
<tr><td><code id="Heat_+3A_angle_col">angle_col</code></td>
<td>

<p>angle of the column labels. Please refer to the manual in <code>pheatmap</code>
</p>
</td></tr>
<tr><td><code id="Heat_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 10 elements:
&ldquo;rowInd&rdquo;, &ldquo;colInd&rdquo;, &ldquo;call&rdquo;, &ldquo;carpet&rdquo;,
&ldquo;rowDendrogram&rdquo;, &ldquo;colDendrogram&rdquo;, &ldquo;breaks&rdquo;,
&ldquo;col&rdquo;, &ldquo;colorTable&rdquo;, &ldquo;layout&rdquo;.
</p>


<h3>Note</h3>

<p>This function is based on the function <code>pheatmap</code> in <code>pheatmap</code> R package.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first 6 probes (3 OE probes, 2 UE probes, and 1 NE probe)
pDat$probe1 = dat[1,]
pDat$probe2 = dat[2,]
pDat$probe3 = dat[3,]
pDat$probe4 = dat[4,]
pDat$probe5 = dat[5,]
pDat$probe6 = dat[6,]

print(pDat[1:2, ])

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))

statVisual(type = 'Heat', 
           data = pDat[, c(2:8)], 
           group = 'grp')

Heat(
     data = pDat[, c(2:8)], 
     group = 'grp')

</code></pre>

<hr>
<h2 id='Hist'>
Compare Groups Based on Histograms
</h2><span id='topic+Hist'></span>

<h3>Description</h3>

<p>Compare groups based on histograms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hist(
    data, 
    y, 
    group = NULL, 
    fill = group, 
    border.color = NULL, 
    inner.color = NULL, 
    theme_classic = TRUE, 
    bins = NULL, 
    binwidth = NULL, 
    alpha = 0.8, 
    xlab = y, 
    ylab = "count", 
    group.lab = group, 
    title = "Histogram", 
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Hist_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_y">y</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable,
for which the histogram will be drawn. The string <code>y</code> can also indicate a function of the variable, e.g., <code class="reqn">\log(y)</code>.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The histogram will be drawn for each of the subject group. It also indicates the border colors of the histograms.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_fill">fill</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. It indicates the inside colors of the histograms.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_border.color">border.color</code></td>
<td>

<p>Histogram border color, only available when group &amp; fill are NULL.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_inner.color">inner.color</code></td>
<td>

<p>Histogram inside color, only available when group &amp; fill are NULL.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="Hist_+3A_bins">bins</code></td>
<td>

<p>integer. number of bins of histogram (default: 30).
</p>
</td></tr>
<tr><td><code id="Hist_+3A_binwidth">binwidth</code></td>
<td>

<p>Bin width of histogram.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_alpha">alpha</code></td>
<td>

<p>Transparency of histogram inside color.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_xlab">xlab</code></td>
<td>

<p>x axis label
</p>
</td></tr>
<tr><td><code id="Hist_+3A_ylab">ylab</code></td>
<td>

<p>y axis label
</p>
</td></tr>
<tr><td><code id="Hist_+3A_group.lab">group.lab</code></td>
<td>

<p>label of group variable
</p>
</td></tr>
<tr><td><code id="Hist_+3A_title">title</code></td>
<td>

<p>title of the plot
</p>
</td></tr>
<tr><td><code id="Hist_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="Hist_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>, <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first probe which is over-expressed in cases
pDat$probe1 = dat[1,]

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))

statVisual(type = 'Hist', 
       data = pDat, 
       y = 'probe1', 
       group = 'grp') 

Hist(
     data = pDat, 
     y = 'probe1', 
     group = 'grp') 


</code></pre>

<hr>
<h2 id='ImpPlot'>
Plot of Variable Importance
</h2><span id='topic+ImpPlot'></span>

<h3>Description</h3>

<p>Plot of variable importance based on results from <code>randomForest</code> or <code>gbm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ImpPlot(model, 
	theme_classic = TRUE, 
	n.trees = NULL, 
	addThemeFlag = TRUE,
	...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ImpPlot_+3A_model">model</code></td>
<td>

<p>An object returned by <code>randomForest</code> or <code>gbm</code>
</p>
</td></tr>
<tr><td><code id="ImpPlot_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="ImpPlot_+3A_n.trees">n.trees</code></td>
<td>

<p>integer. The number of trees used to generate the plot
used in the function <code>summary.gbm</code> in the R library <code>gbm</code>. 
Only the first n.trees trees will be used.
</p>
</td></tr>
<tr><td><code id="ImpPlot_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="ImpPlot_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>       <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(dplyr)
library(randomForest)
library(tibble)


data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first 6 probes (3 OE probes, 2 UE probes, and 1 NE probe)
pDat$probe1 = dat[1,]
pDat$probe2 = dat[2,]
pDat$probe3 = dat[3,]
pDat$probe4 = dat[4,]
pDat$probe5 = dat[5,]
pDat$probe6 = dat[6,]

print(pDat[1:2, ])

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))

pDat$grp = factor(pDat$grp)


rf_m = randomForest(
  x = pDat[, c(3:8)], 
  y = pDat$grp, 
  importance = TRUE, proximity = TRUE
)


statVisual(type = 'ImpPlot', model = rf_m)

ImpPlot(model = rf_m)

</code></pre>

<hr>
<h2 id='iprcomp'>
Improved Function for Obtaining Principal Components
</h2><span id='topic+iprcomp'></span>

<h3>Description</h3>

<p>Calculate principal components when data contains missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iprcomp(dat, center = TRUE, scale. = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iprcomp_+3A_dat">dat</code></td>
<td>

<p>n by p matrix. rows are subjects and columns are variables
</p>
</td></tr>
<tr><td><code id="iprcomp_+3A_center">center</code></td>
<td>

<p>logical. Indicates if each row of <code>dat</code> needs to be mean-centered
</p>
</td></tr>
<tr><td><code id="iprcomp_+3A_scale.">scale.</code></td>
<td>

<p>logical. Indicates if each row of <code>dat</code> needs to be scaled to have variance one
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We first set missing values as median of the corresponding variable, then call the function <code>prcomp</code>.
This is a very simple solution. The user can use their own imputation methods before calling <code>prcomp</code>.
</p>


<h3>Value</h3>

<p>A list of 3 elements
</p>
<table>
<tr><td><code>sdev</code></td>
<td>
<p>square root of the eigen values</p>
</td></tr>
<tr><td><code>rotation</code></td>
<td>
<p>a matrix with columns are eigen vectors, i.e., projection direction</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>a matrix with columns are principal components</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># generate simulated data
set.seed(1234567)
dat.x = matrix(rnorm(500), nrow = 100, ncol = 5)
dat.y = matrix(rnorm(500, mean = 2), nrow = 100, ncol = 5)
dat = rbind(dat.x, dat.y)
grp = c(rep(0, 100), rep(1, 100))
print(dim(dat))

res = iprcomp(dat, center = TRUE, scale.  =  FALSE)

# for each row, set one artificial missing value
dat.na=dat
nr=nrow(dat.na)
nc=ncol(dat.na)
for(i in 1:nr)
{
  posi=sample(x=1:nc, size=1)
  dat.na[i,posi]=NA
}

res.na = iprcomp(dat.na, center = TRUE, scale.  =  FALSE)

##
# pca plot
##
par(mfrow = c(3,1))
# original data without missing values
plot(x = res$x[,1], y = res$x[,2], xlab = "PC1", ylab  =  "PC2")
# perturbed data with one NA per probe 
# the pattern of original data is captured
plot(x = res.na$x[,1], y = res.na$x[,2], xlab = "PC1", ylab  =  "PC2", main = "with missing values")
par(mfrow = c(1,1))

</code></pre>

<hr>
<h2 id='LinePlot'>
Compare Groups Based on Trajectory Plots
</h2><span id='topic+LinePlot'></span>

<h3>Description</h3>

<p>Compare groups based on trajectory plots. Trajectories belonging to different groups will have different colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LinePlot(
    data, 
    x, 
    y, 
    sid,
    group = NULL, 
    xFlag = FALSE,
    points = TRUE, 
    point.size = 1, 
    theme_classic = TRUE, 
    xlab = x, 
    ylab = y, 
    title = "Trajectory plot", 
    xLevel = NULL,
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LinePlot_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_x">x</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the time.
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_y">y</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable on y axis
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_sid">sid</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject id.
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The trajectories of subjects
in the same group will have the same color.
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_xflag">xFlag</code></td>
<td>

<p>logical. Indicate if <code>x</code> should be treated as continuous (<code>xFlag=TRUE</code>)
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_points">points</code></td>
<td>

<p>logical. Indicates if points will be added to the trajectories on the coordinate (x, y).
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_point.size">point.size</code></td>
<td>

<p>numeric. size of the data points on the trajectories
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_xlab">xlab</code></td>
<td>

<p>character. x axis label
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_ylab">ylab</code></td>
<td>

<p>character. y axis label
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_title">title</code></td>
<td>

<p>character. title of plot
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_xlevel">xLevel</code></td>
<td>

<p>character. A character vector indicating the order of the elements of <code>x</code> to be shown on x-axis if <code>is.null(x)==FALSE</code>.
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="LinePlot_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following 9 elements: <code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>, <code>theme</code>, <code>coordinates</code>, <code>facet</code>,
<code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(longDat)

print(dim(longDat))
print(longDat[1:3,])

print(table(longDat$time, useNA = "ifany"))
print(table(longDat$grp, useNA = "ifany"))
print(table(longDat$sid, useNA = "ifany"))

print(table(longDat$time, longDat$grp))

statVisual(type = "LinePlot",
  data = longDat,
  x = 'time',
  y = 'y',
  sid = 'sid',
  group = 'grp')

LinePlot(
  data = longDat,
  x = 'time',
  y = 'y',
  sid = 'sid',
  group = 'grp')

</code></pre>

<hr>
<h2 id='longDat'>
A Simulated Dataset for Longitudinal Data Analysis
</h2><span id='topic+longDat'></span>

<h3>Description</h3>

<p>A simulated dataset for longitudinal data analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("longDat")</code></pre>


<h3>Format</h3>

<p>A data frame with 540 observations on the following 4 variables.
</p>

<dl>
<dt><code>sid</code></dt><dd><p>subject id</p>
</dd>
<dt><code>time</code></dt><dd><p>time points. A factor with levels <code>time1</code> <code>time2</code> <code>time3</code> <code>time4</code> <code>time5</code> <code>time6</code></p>
</dd>
<dt><code>y</code></dt><dd><p>numeric. outcome variable</p>
</dd>
<dt><code>grp</code></dt><dd><p>subject group. A factor with levels <code>grp1</code> <code>grp2</code> <code>grp3</code></p>
</dd>
</dl>



<h3>Details</h3>

<p>The dataset is generated from the following mixed effects model for repeated measures:
</p>
<p style="text-align: center;"><code class="reqn">y_{ij}=\beta_{0i}+\beta_1 t_{j} + \beta_2 grp_{2i} +
\beta_3 grp_{3i} + \beta_4 \times\left(t_{j}\times grp_{2i}\right)
+ \beta_5 \times\left(t_{j}\times grp_{3i}\right)
+\epsilon_{ij},
</code>
</p>
<p> where <code class="reqn">y_{ij}</code> is the outcome value for the <code class="reqn">i</code>-th subject
measured at <code class="reqn">j</code>-th time point <code class="reqn">t_{j}</code>, 
<code class="reqn">grp_{2i}</code> is a dummy variable indicating if the <code class="reqn">i</code>-th subject
is from group 2, 
<code class="reqn">grp_{3i}</code> is a dummy variable indicating if the <code class="reqn">i</code>-th subject
is from group 3, 
<code class="reqn">\beta_{0i}\sim N\left(\beta_0, \sigma_b^2\right)</code>,
<code class="reqn">\epsilon_{ij}\sim N\left(0, \sigma_e^2\right)</code>, <code class="reqn">i=1,\ldots, n, j=1, \ldots, m</code>,
<code class="reqn">n</code> is the number of subjects, and <code class="reqn">m</code> is the number of time points.
</p>
<p>When <code class="reqn">t_j=0</code>, the expected outcome value is
</p>
<p style="text-align: center;"><code class="reqn">
E\left(y_{ij}\right)=\beta_0+\beta_2 dose_{2i} + \beta_3 dose_{3i}.</code>
</p>

<p>Hence, we have at baseline
</p>
<p style="text-align: center;"><code class="reqn">
E\left(y_{ij}\right)=\beta_0,\; \mbox{for dose 1 group}.</code>
</p>

<p style="text-align: center;"><code class="reqn">
E\left(y_{ij}\right)=\beta_0 + \beta_2,\; \mbox{for dose 2 group}.</code>
</p>

<p style="text-align: center;"><code class="reqn">
E\left(y_{ij}\right)=\beta_0 + \beta_3,\; \mbox{for dose 3 group}.</code>
</p>

<p>For dose 1 group, the expected outcome values across time is
</p>
<p style="text-align: center;"><code class="reqn">
E\left(y_{ij}\right)=\beta_0+\beta_1 t_{j}.</code>
</p>

<p>We also can get the expected difference of outcome values between dose 2 group and dose 1 group, between dose 3 group and dose 1 group, and between
dose 3 group and dose 2 group:
</p>
<p style="text-align: center;"><code class="reqn">
E\left(y_{ij} - y_{i'j}\right) =\beta_2+\beta_4 t_{j},\;\mbox{for subject $i$ in dose 2 group and subject $i'$ in dose 1 group},</code>
</p>

<p style="text-align: center;"><code class="reqn">
E\left(y_{kj} - y_{i'j}\right) =\beta_3+\beta_5 t_{j},\;\mbox{for subject $k$ in dose 3 group and subject $i'$ in dose 1 group},</code>
</p>

<p style="text-align: center;"><code class="reqn">
E\left(y_{kj} - y_{ij}\right) =\left(\beta_3-\beta_2\right)+\left(\beta_5-\beta_4\right) t_{j},\;\mbox{for subject $i$ in dose 3 group and subject $i$ in dose 2 group}.</code>
</p>

<p>We set <code class="reqn">n=90</code>, <code class="reqn">m=6</code>, 
<code class="reqn">\beta_0=5</code>,
<code class="reqn">\beta_1=0</code>,
<code class="reqn">\beta_2=0</code>,
<code class="reqn">\beta_3=0</code>,
<code class="reqn">\beta_4=2</code>,
<code class="reqn">\beta_5=-2</code>,
<code class="reqn">\sigma_e=1</code>,
<code class="reqn">\sigma_b=0.5</code>,
and
<code class="reqn">t_{ij}=j, j=1, \ldots, m</code>.
</p>
<p>That is, the trajectories for dose 1 group are horizontal with mean intercept at <code class="reqn">5</code>, the trajectories for dose 2 group are linearly increasing with slope <code class="reqn">2</code> and mean intercept <code class="reqn">5</code>, and the trajectories for dose 3 group are linearly decreasing with slope <code class="reqn">-2</code> and mean intercept <code class="reqn">5</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(longDat)

print(dim(longDat))
print(longDat[1:3,])

print(table(longDat$time, useNA = "ifany"))
print(table(longDat$grp, useNA = "ifany"))
print(table(longDat$sid, useNA = "ifany"))

print(table(longDat$time, longDat$grp))
</code></pre>

<hr>
<h2 id='PCA_score'>
Scatter Plot of 2 Specified Principal Components
</h2><span id='topic+PCA_score'></span>

<h3>Description</h3>

<p>Scatter plot of 2 specified principal components. The size of the data points on the PCA plot indicates the Mahalanobis distance (distance between each point and mean value).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA_score(
    prcomp_obj, 
    data, 
    dims = c(1, 2),
    color = NULL, 
    MD = TRUE, 
    loadings = FALSE, 
    loadings.color = "black", 
    loadings.label = FALSE,
    title = "pca plot",
    addThemeFlag = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCA_score_+3A_prcomp_obj">prcomp_obj</code></td>
<td>

<p>the object returned by the function <code><a href="stats.html#topic+prcomp">prcomp</a></code>.
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects. The object <code>prcomp_obj</code> is based on <code>data</code>
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_dims">dims</code></td>
<td>

<p>a numeric vector with 2 elements indicating which two principal components
will be used to draw scatter plot.
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_color">color</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The data points on the PCA plot will be colored by the group info.
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_md">MD</code></td>
<td>

<p>logical. Indicate if the Mahalanobis distance (distance between each point and mean value) would be used to indicate the size of data points on the PCA plot
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_loadings">loadings</code></td>
<td>

<p>logical. Indicate if loading plot would be superimposed on the PCA plot. (default: FALSE)
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_loadings.color">loadings.color</code></td>
<td>

<p>character. Indicate the color of the loading axis.
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_loadings.label">loadings.label</code></td>
<td>

<p>logical. Indicating if loading labels should be added to the plot. (default: FALSE)
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_title">title</code></td>
<td>

<p>character. Figure title.
</p>
</td></tr>
<tr><td><code id="PCA_score_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>, <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(factoextra)

data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first 6 probes (3 OE probes, 2 UE probes, and 1 NE probe)
pDat$probe1 = dat[1,]
pDat$probe2 = dat[2,]
pDat$probe3 = dat[3,]
pDat$probe4 = dat[4,]
pDat$probe5 = dat[5,]
pDat$probe6 = dat[6,]

print(pDat[1:2, ])

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))

pDat$grp = factor(pDat$grp)

###

pca.obj = iprcomp(pDat[, c(3:8)], scale. = TRUE)

# scree plot
factoextra::fviz_eig(pca.obj, addlabels = TRUE)

# scatter plot of PC1 vs PC2
statVisual(type = 'PCA_score',
           prcomp_obj = pca.obj, 
           dims = c(1, 2),
           data = pDat, 
           color = 'grp',
           loadings = FALSE)

PCA_score(prcomp_obj = pca.obj, 
          dims = c(1, 3),
          data = pDat, 
          color = 'grp',
          loadings = FALSE)

</code></pre>

<hr>
<h2 id='PVCA'>
Principal Variance Component Analysis (PVCA)
</h2><span id='topic+PVCA'></span>

<h3>Description</h3>

<p>Plot of weighted average proportion variance versus effects in 
principal variance component analysis (PVCA).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PVCA(
    clin_data, 
    clin_subjid, 
    gene_data, 
    pct_threshold = 0.8, 
    batch.factors, 
    theme_classic = FALSE, 
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PVCA_+3A_clin_data">clin_data</code></td>
<td>

<p>A data frame containing clinical information, including  an id variable that corresponds to rownames of <code>gene_data</code>
</p>
</td></tr>
<tr><td><code id="PVCA_+3A_clin_subjid">clin_subjid</code></td>
<td>

<p>character. The column name of <code>clin_data</code> that indicates subject id. It corresponds to the rowname of <code>gene_data</code>.
</p>
</td></tr>
<tr><td><code id="PVCA_+3A_gene_data">gene_data</code></td>
<td>

<p>A data frame with genes as rows and subjects as columns.
</p>
</td></tr>
<tr><td><code id="PVCA_+3A_pct_threshold">pct_threshold</code></td>
<td>

<p>numeric. The percentile value of the minimum amount of the variabilities that the selected principal components need to explain
</p>
</td></tr>
<tr><td><code id="PVCA_+3A_batch.factors">batch.factors</code></td>
<td>

<p>character. A vector of factors that the mixed linear model will be fit on.
</p>
</td></tr>
<tr><td><code id="PVCA_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="PVCA_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="PVCA_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>, <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(pvca)


data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# create a fake Batch variable
esSim$Batch=c(rep("A", 4), rep("B", 6), rep("C", 10))
# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])


# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])


statVisual(type = 'PVCA',
           clin_data = pData(esSim), 
           clin_subjid = "sid", 
           gene_data = exprs(esSim), 
           batch.factors = c("grp", "Batch"))

PVCA( 
     clin_data = pData(esSim), 
     clin_subjid = "sid", 
     gene_data = exprs(esSim), 
     batch.factors = c("grp", "Batch"))

</code></pre>

<hr>
<h2 id='stackedBarPlot'>
Draw Stacked Bar Plots
</h2><span id='topic+stackedBarPlot'></span>

<h3>Description</h3>

<p>Draw stacked bar plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stackedBarPlot(dat, 
	       catVar, 
	       group, 
	       xlab = catVar, 
	       ylab = "Count", 
	       group.lab = group, 
	       title = "Stacked barplots of counts", 
	       catVarLevel = NULL, 
	       groupLevel = NULL, 
	       addThemeFlag = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stackedBarPlot_+3A_dat">dat</code></td>
<td>

<p>A data frame object. Rows are subjects and columns are variables.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_catvar">catVar</code></td>
<td>

<p>character. The name of the cateogrical variable to be shown in x-axis.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_group">group</code></td>
<td>

<p>character. The name of variable indicating groups of subjects.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_xlab">xlab</code></td>
<td>

<p>character. Label for x-axis.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_ylab">ylab</code></td>
<td>

<p>character. Label for y-axis.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_group.lab">group.lab</code></td>
<td>

<p>character. Label for group in legend.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_title">title</code></td>
<td>

<p>character. Figure title.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_catvarlevel">catVarLevel</code></td>
<td>

<p>character. A vector indicating the order of the unique elements of <code>catVar</code> should be shown in x-axis.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_grouplevel">groupLevel</code></td>
<td>

<p>character. A vector indicating the order of the unique elements of <code>group</code> should be shown in figure and in legend.
</p>
</td></tr>
<tr><td><code id="stackedBarPlot_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the following 9 elements:
&ldquo;data&rdquo;, &ldquo;layers&rdquo;, &ldquo;scales&rdquo;, &ldquo;mapping&rdquo;,
&ldquo;theme&rdquo;, &ldquo;coordinates&rdquo;, &ldquo;facet&rdquo;, &ldquo;plot_env&rdquo;,
&ldquo;labels&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genoSim)

pDat = pData(genoSim)
geno = exprs(genoSim)

pDat$snp1 = geno[1,]

print(table(pDat$snp1, pDat$grp, useNA="ifany"))

stackedBarPlot(dat = pDat, 
	       catVar = "snp1", 
	       group = "grp", 
               xlab = "snp1", 
	       ylab = "Count", 
	       group.lab = "grp",
               title = "Stacked barplots of counts",
               catVarLevel = NULL)

</code></pre>

<hr>
<h2 id='statVisual'>
The Wrapper Function Incorporating All Wrapper Functions in statVisual
</h2><span id='topic+statVisual'></span>

<h3>Description</h3>

<p>The wrapper function incorporating all wrapper functions in statVisual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>statVisual(type, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="statVisual_+3A_type">type</code></td>
<td>

<p>character. Indicate the functions to be called. It can take 
the following values:
&ldquo;BiAxisErrBar&rdquo;,
&ldquo;Box&rdquo;,
&ldquo;BoxROC&rdquo;,
&ldquo;cv_glmnet_plot&rdquo;,
&ldquo;Den&rdquo;,
&ldquo;Dendro&rdquo;,
&ldquo;ErrBar&rdquo;,
&ldquo;Heat&rdquo;,
&ldquo;Hist&rdquo;,
&ldquo;ImpPlot&rdquo;,
&ldquo;iprcomp&rdquo;,
&ldquo;LinePlot&rdquo;,
&ldquo;PCA_score&rdquo;,
&ldquo;PVCA&rdquo;,
&ldquo;statVisual&rdquo;,
&ldquo;Volcano&rdquo;,
&ldquo;XYscatter&rdquo;.
</p>
</td></tr>
<tr><td><code id="statVisual_+3A_...">...</code></td>
<td>

<p>input parameters for the functions specified by <code>type</code>.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BiAxisErrBar">BiAxisErrBar</a></code>,
<code><a href="#topic+Box">Box</a></code>,
<code><a href="#topic+BoxROC">BoxROC</a></code>,
<code><a href="#topic+cv_glmnet_plot">cv_glmnet_plot</a></code>,
<code><a href="#topic+Den">Den</a></code>,
<code><a href="#topic+Dendro">Dendro</a></code>,
<code><a href="#topic+ErrBar">ErrBar</a></code>,
<code><a href="#topic+Heat">Heat</a></code>,
<code><a href="#topic+Hist">Hist</a></code>,
<code><a href="#topic+ImpPlot">ImpPlot</a></code>,
<code><a href="#topic+iprcomp">iprcomp</a></code>,
<code><a href="#topic+LinePlot">LinePlot</a></code>,
<code><a href="#topic+PCA_score">PCA_score</a></code>,
<code><a href="#topic+PVCA">PVCA</a></code>,
<code><a href="#topic+statVisual">statVisual</a></code>,
<code><a href="#topic+Volcano">Volcano</a></code>,
<code><a href="#topic+XYscatter">XYscatter</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esSim)
print(esSim)

# expression data
dat = exprs(esSim)
print(dim(dat))
print(dat[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2,])

# choose the first probe which is over-expressed in cases
pDat$probe1 = dat[1,]

# check histograms of probe 1 expression in cases and controls
print(table(pDat$grp, useNA = "ifany"))

statVisual(type = 'Hist', 
       data = pDat, 
       y = 'probe1', 
       group = 'grp') 


</code></pre>

<hr>
<h2 id='Volcano'>
Volcano Plot
</h2><span id='topic+Volcano'></span>

<h3>Description</h3>

<p>Volcano plot with the option to label the significant results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Volcano(
    resFrame, 
    stats, 
    p.value, 
    group = NULL, 
    xlab = "logFC", 
    ylab = "-log10(p value)", 
    title = NULL, 
    vline.col = "orange", 
    hline.col = "dodgerblue", 
    vline = list(xintercept = c(-1, 1), label = c(-1, 1)), 
    hline = list(
        yintercept = c(-log10(0.05), 
                       -log10(0.05/nrow(resFrame)), 
                       -log10(max(resFrame[p.adjust(resFrame[, p.value], 
                                       method = "fdr") &lt;= 0.05, p.value]))), 
        label = c("p value: 0.05", "Bonferroni: 0.05", "FDR: 0.05")), 
    rowname.var = NULL, 
    point.size = 3, 
    theme_classic = TRUE, 
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Volcano_+3A_resframe">resFrame</code></td>
<td>

<p>A data frame stored information about the results, including gene id, statistic (e.g., log fold change, odds ratio), p-value, and significance of a gene. 
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_stats">stats</code></td>
<td>

<p>character. The column name of <code>resFrame</code> that indicates the effect of a gene.
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_p.value">p.value</code></td>
<td>

<p>character. The column name of <code>resFrame</code> that indicates the p-value.
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>resFrame</code> that indicates the significance of a gene.
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_xlab">xlab</code></td>
<td>

<p>x axis label
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_ylab">ylab</code></td>
<td>

<p>y axis label
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_title">title</code></td>
<td>

<p>title of the plot
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_vline.col">vline.col</code></td>
<td>

<p>color of the vertical lines (default: &ldquo;orange&rdquo;)
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_hline.col">hline.col</code></td>
<td>

<p>color of the horizontal lines (default: &ldquo;dodgerblue&rdquo;)
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_vline">vline</code></td>
<td>

<p>A list with two elements: &ldquo;xintercept&rdquo; and &ldquo;label&rdquo;,
where the former element is a numeric vector indicating the x-axis location to draw vertical color lines and the latter element is list of labels for the elements in &ldquo;xintercept&rdquo;.
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_hline">hline</code></td>
<td>

<p>A list with two elements: &ldquo;yintercept&rdquo; and &ldquo;label&rdquo;,
where the former element is a numeric vector indicating the y-axis location to draw horizontal color lines and the latter element is list of labels for the elements in &ldquo;xintercept&rdquo;.
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_rowname.var">rowname.var</code></td>
<td>

<p>character. The column name of <code>resFrame</code> that indicates which variable will be used to label the significant results in the volcano plot. The elements of this column for non-significant results should be set to be <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_point.size">point.size</code></td>
<td>

<p>size of data points in the plot.
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="Volcano_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>       <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggrepel)
library(limma)

# load the simulated dataset
data(esSim)
print(esSim)

# expression levels
y = exprs(esSim)
print(dim(y))
print(y[1:2,])

# phenotype data
pDat = pData(esSim)
print(dim(pDat))
print(pDat)

# design matrix
design = model.matrix(~grp, data = pDat)
print(design)

options(digits = 3)

# Ordinary fit
fit &lt;- lmFit(y, design)
fit2 &lt;- eBayes(fit)

# get result data frame
resFrame = topTable(fit2,coef = 2, number = nrow(esSim))
print(dim(resFrame))
print(resFrame[1:2,])
resFrame$sigFlag  =  resFrame$adj.P.Val &lt; 0.05

resFrame$probe  =  rownames(resFrame)
# make sure set NA to genes non-differentially expressed
resFrame$probe[which(resFrame$sigFlag == FALSE)] = NA

print(resFrame[1:2,])
print(table(resFrame$sigFlag, useNA = "ifany"))

statVisual(type = 'Volcano',
           resFrame = resFrame, 
           stats = 'logFC', 
           p.value = 'P.Value', 
           group = 'sigFlag', 
           rowname.var = 'probe', 
           point.size = 1)

Volcano(
  resFrame = resFrame, 
  stats = 'logFC', 
  p.value = 'P.Value', 
  group = 'sigFlag', 
  rowname.var = 'probe', 
  point.size = 1)


</code></pre>

<hr>
<h2 id='XYscatter'>
Compare Groups Based on Scatter Plots
</h2><span id='topic+XYscatter'></span>

<h3>Description</h3>

<p>Compare groups based on scatter plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XYscatter(
    data, 
    x, 
    y, 
    group = NULL, 
    alpha = 1, 
    point.size = 3, 
    xlab = x, 
    ylab = y, 
    group.lab = group, 
    title = "Scatter plot", 
    theme_classic = TRUE, 
    addThemeFlag = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XYscatter_+3A_data">data</code></td>
<td>

<p>A data frame. Rows are subjects; Columns are variables describing the subjects.
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_x">x</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable on the x axis of the scatter plot 
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_y">y</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the variable on the y axis of the scatter plot 
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_group">group</code></td>
<td>

<p>character. The column name of <code>data</code> that indicates the subject groups. The scatter plot will be drawn for each of the subject group. It also indicates the colors of the data points in the scatter plots.
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_alpha">alpha</code></td>
<td>

<p>Transparency of histogram inside color.
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_point.size">point.size</code></td>
<td>

<p>numeric. Indicate the size of the data points 
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_xlab">xlab</code></td>
<td>

<p>x axis label
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_ylab">ylab</code></td>
<td>

<p>y axis label
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_group.lab">group.lab</code></td>
<td>

<p>label of group variable
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_title">title</code></td>
<td>

<p>title of the plot
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_theme_classic">theme_classic</code></td>
<td>

<p>logical. Use classic background without grids (default: TRUE).
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_addthemeflag">addThemeFlag</code></td>
<td>

<p>logical. Indicates if light blue background and white grid should be added to the figure.
</p>
</td></tr>
<tr><td><code id="XYscatter_+3A_...">...</code></td>
<td>

<p>other input parameters for facet &amp; theme
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 9 elements.
<code>data</code>, <code>layers</code>, <code>scales</code>, <code>mapping</code>,
<code>theme</code>, <code>coordinates</code>,
<code>facet</code>       <code>plot_env</code>, and <code>labels</code>.
</p>


<h3>Author(s)</h3>

<p>Wenfei Zhang &lt;Wenfei.Zhang@sanofi.com&gt;,
Weiliang Qiu &lt;Weiliang.Qiu@sanofi.com&gt;,
Xuan Lin &lt;Xuan.Lin@sanofi.com&gt;,
Donghui Zhang &lt;Donghui.Zhang@sanofi.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diffCorDat)

print(dim(diffCorDat))
print(diffCorDat[1:2,])

statVisual(type = 'XYscatter',
  data = diffCorDat, 
  x = 'probe1', 
  y = 'probe2', 
  group = 'grp', 
  title = 'Scatter Plot: probe1 vs probe2')

XYscatter( 
  data = diffCorDat, 
  x = 'probe1', 
  y = 'probe2', 
  group = 'grp', 
  title = 'Scatter Plot: probe1 vs probe2')

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
