<!DOCTYPE html><html><head><title>Help for package statgenMPP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {statgenMPP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#barleyMQM'><p>Pre-computed MQM output barley</p></a></li>
<li><a href='#barleyPheno'><p>Phenotypic data for awn length in barley</p></a></li>
<li><a href='#calcDistance'><p>cofactor selection based on the QTL windowsize</p></a></li>
<li><a href='#calcIBDMPP'><p>IBD calculation for multi parental populations</p></a></li>
<li><a href='#createGDataMPP'><p>Create an object of class gDataMPP</p></a></li>
<li><a href='#effectPlot'><p>Create a matrix plot</p></a></li>
<li><a href='#kinshipIBD'><p>Compute kinship matrix for IBD probabilities</p></a></li>
<li><a href='#maizeMQM'><p>Pre-computed MQM output maize</p></a></li>
<li><a href='#maizeSQM'><p>Pre-computed SQM output maize</p></a></li>
<li><a href='#plot.gDataMPP'><p>Plot function for the class <code>gDataMPP</code></p></a></li>
<li><a href='#plot.QTLMPP'><p>Plot function for the class <code>QTLMPP</code></p></a></li>
<li><a href='#print.summary.QTLMPP'><p>Print summary of object of class <code>summary.QTLMPP</code></p></a></li>
<li><a href='#randomQTLmodel'><p>one position fitted in the mixed model</p></a></li>
<li><a href='#readRABBITMPP'><p>Read IBD probabilities</p></a></li>
<li><a href='#scanQTL'><p>one-round genome scan with specified cofactor(s)</p></a></li>
<li><a href='#selQTLMPP'><p>Multi round genome scans for QTL detection</p></a></li>
<li><a href='#statgenMPP-package'><p>statgenMPP: QTL Mapping for Multi Parent Populations</p></a></li>
<li><a href='#summary.gDataMPP'><p>Summary function for the class <code>gDataMPP</code></p></a></li>
<li><a href='#summary.QTLMPP'><p>Summary function for the class <code>QTLMPP</code></p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>QTL Mapping for Multi Parent Populations</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-12-2</td>
</tr>
<tr>
<td>Description:</td>
<td>For Multi Parent Populations (MPP) Identity By Descend (IBD) 
    probabilities are computed using Hidden Markov Models. These probabilities 
    are then used in a mixed model approach for QTL Mapping as described in 
    Li et al. (&lt;<a href="https://doi.org/10.1007%2Fs00122-021-03919-7">doi:10.1007/s00122-021-03919-7</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6), statgenGWAS (&ge; 1.0.8)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, foreach, ggplot2, gridExtra, LMMsolver, Matrix,
scales, spam, statgenIBD (&ge; 1.0.4), stringi</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, tinytest</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-02 15:25:10 UTC; rossu027</td>
</tr>
<tr>
<td>Author:</td>
<td>Wenhao Li <a href="https://orcid.org/0000-0001-5719-5775"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Martin Boer <a href="https://orcid.org/0000-0002-1879-4588"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Bart-Jan van Rossum
    <a href="https://orcid.org/0000-0002-8673-2514"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Fred van Eeuwijk <a href="https://orcid.org/0000-0003-3672-2921"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bart-Jan van Rossum &lt;bart-jan.vanrossum@wur.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-02 22:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='barleyMQM'>Pre-computed MQM output barley</h2><span id='topic+barleyMQM'></span>

<h3>Description</h3>

<p>Pre-computed MQM output for barley data used in vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barleyMQM
</code></pre>


<h3>Format</h3>

<p>An object of class <code>QTLMPP</code> (inherits from <code>GWAS</code>) of length 5.
</p>

<hr>
<h2 id='barleyPheno'>Phenotypic data for awn length in barley</h2><span id='topic+barleyPheno'></span>

<h3>Description</h3>

<p>Phenotypic data for awn length in barley
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barleyPheno
</code></pre>


<h3>Format</h3>

<p>A data.frame with 916 rows and 3 columns.
</p>

<dl>
<dt>genotype</dt><dd><p>Genotype</p>
</dd>
<dt>Awn_length</dt><dd><p>Awn_length in mm.</p>
</dd>
<dt>cross</dt><dd><p>The cross the genotype belongs to.</p>
</dd>
</dl>



<h3>References</h3>

<p>Fine mapping of a major QTL for awn length in barley using a
multiparent mapping population. Liller CB, Walla A, Boer MP, Hedley P,
Macaulay M, Effgen S, von Korff M, van Esse GW, Koornneef M.
Theor Appl Genet. 2017 Feb;130(2):269-281. <a href="https://doi.org/10.1007/s00122-016-2807-y">doi:10.1007/s00122-016-2807-y</a>.
</p>

<hr>
<h2 id='calcDistance'>cofactor selection based on the QTL windowsize</h2><span id='topic+calcDistance'></span>

<h3>Description</h3>

<p>cofactor selection based on the QTL windowsize
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcDistance(map, m1, m2)
</code></pre>

<hr>
<h2 id='calcIBDMPP'>IBD calculation for multi parental populations</h2><span id='topic+calcIBDMPP'></span>

<h3>Description</h3>

<p>IBD calculation for multi parental populations. Per cross IBD probabilities
are calculated using <code>calcIBD</code> in the statgenIBD package. These
probabilities are combined with optional phenotypic data and stored in a
single object of class <code>gDataMPP</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcIBDMPP(
  crossNames,
  markerFiles,
  pheno,
  popType,
  mapFile,
  evalDist,
  grid = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcIBDMPP_+3A_crossnames">crossNames</code></td>
<td>
<p>A character vector, the names of the crosses.</p>
</td></tr>
<tr><td><code id="calcIBDMPP_+3A_markerfiles">markerFiles</code></td>
<td>
<p>A character vector indicating the locations of the files
with genotypic information for the populations. The files should be in
tab-delimited format with a header containing marker names.</p>
</td></tr>
<tr><td><code id="calcIBDMPP_+3A_pheno">pheno</code></td>
<td>
<p>A data.frame or a list of data.frames with phenotypic data,
with genotypes in the first column <code>genotype</code> and traits in the
following columns. The trait columns should be numerical columns only.
A list of data.frames can be used for replications, i.e. different
trials.</p>
</td></tr>
<tr><td><code id="calcIBDMPP_+3A_poptype">popType</code></td>
<td>
<p>A character string indicating the type of population. One of
DH, Fx, FxDH, BCx, BCxDH, BC1Sx, BC1SxDH, C3, C3DH, C3Sx, C3SxDH, C4, C4DH,
C4Sx, C4SxDH (see Details).</p>
</td></tr>
<tr><td><code id="calcIBDMPP_+3A_mapfile">mapFile</code></td>
<td>
<p>A character string indicating the location of the map file
for the population. The file should be in tab-delimited format. It should
consist of exactly three columns, marker, chromosome and position. There
should be no header. The positions in the file should be in centimorgan.</p>
</td></tr>
<tr><td><code id="calcIBDMPP_+3A_evaldist">evalDist</code></td>
<td>
<p>A numeric value, the maximum distance in cM between
evaluation points.</p>
</td></tr>
<tr><td><code id="calcIBDMPP_+3A_grid">grid</code></td>
<td>
<p>Should the extra markers that are added to assure the a
maximum distince of <code>evalDist</code> be on a grid (<code>TRUE</code>) or in between
marker existing marker positions (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="calcIBDMPP_+3A_verbose">verbose</code></td>
<td>
<p>Should progress be printed?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>IBD probabilities can be calculated for many different types of populations.
In the following table all supported populations are listed. Note that the
value of x in the population types is variable, with its maximum value
depicted in the last column.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Population type</strong> </td><td style="text-align: left;"> <strong>Cross</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td><td style="text-align: left;"> <strong>max. x</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   DH </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> doubled haploid population </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   Fx </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> Fx population (F1, followed by x-1 generations of selfing) </td><td style="text-align: left;"> 8 </td>
</tr>
<tr>
 <td style="text-align: left;">
   FxDH </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> Fx, followed by DH generation </td><td style="text-align: left;"> 8 </td>
</tr>
<tr>
 <td style="text-align: left;">
   BCx </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> backcross, second parent is recurrent parent </td><td style="text-align: left;"> 9 </td>
</tr>
<tr>
 <td style="text-align: left;">
   BCxDH </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> BCx, followed by DH generation </td><td style="text-align: left;"> 9 </td>
</tr>
<tr>
 <td style="text-align: left;">
   BC1Sx </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> BC1, followed by x generations of selfing </td><td style="text-align: left;"> 7 </td>
</tr>
<tr>
 <td style="text-align: left;">
   BC1SxDH </td><td style="text-align: left;"> biparental </td><td style="text-align: left;"> BC1, followed by x generations of selfing and DH </td><td style="text-align: left;"> 6 </td>
</tr>
<tr>
 <td style="text-align: left;">
   C3 </td><td style="text-align: left;"> three-way </td><td style="text-align: left;"> three way cross: (AxB) x C </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   C3DH </td><td style="text-align: left;"> three-way </td><td style="text-align: left;"> C3, followed by DH generation </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   C3Sx </td><td style="text-align: left;"> three-way </td><td style="text-align: left;"> C3, followed by x generations of selfing </td><td style="text-align: left;"> 7 </td>
</tr>
<tr>
 <td style="text-align: left;">
   C3SxDH </td><td style="text-align: left;"> three-way </td><td style="text-align: left;"> C3, followed by x generations of selfing and DH generation </td><td style="text-align: left;"> 6 </td>
</tr>
<tr>
 <td style="text-align: left;">
   C4 </td><td style="text-align: left;"> four-way </td><td style="text-align: left;"> four-way cross: (AxB) x (CxD) </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   C4DH </td><td style="text-align: left;"> four-way </td><td style="text-align: left;"> C4, followed by DH generation </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   C4Sx </td><td style="text-align: left;"> four-way </td><td style="text-align: left;"> C4, followed by x generations of selfing </td><td style="text-align: left;"> 6 </td>
</tr>
<tr>
 <td style="text-align: left;">
   C4SxDH </td><td style="text-align: left;"> four-way </td><td style="text-align: left;"> C4, followed by x generations of selfing and DH generation </td><td style="text-align: left;"> 6 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>An object of class <code>gDataMPP</code> with the following components:
</p>
<table>
<tr><td><code>map</code></td>
<td>
<p>a data.frame containing map data. Map is sorted by
chromosome and position.</p>
</td></tr>
<tr><td><code>markers</code></td>
<td>
<p>a 3D matrix containing IBD probabilities.</p>
</td></tr>
<tr><td><code>pheno</code></td>
<td>
<p>data.frame or list of data.frames containing phenotypic
data.</p>
</td></tr>
<tr><td><code>kinship</code></td>
<td>
<p>a kinship matrix.</p>
</td></tr>
<tr><td><code>covar</code></td>
<td>
<p>a data.frame with extra covariates (including the
name of the cross).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Read phenotypic data.
pheno &lt;- read.delim(system.file("extdata/multipop", "AxBxCpheno.txt",
                               package = "statgenMPP"))
## Rename first column to genotype.
colnames(pheno)[1] &lt;- "genotype"


## Compute IBD probabilities for simulated population - AxB, AxC.
ABC &lt;- calcIBDMPP(crossNames = c("AxB", "AxC"),
                  markerFiles = c(system.file("extdata/multipop", "AxB.txt",
                                              package = "statgenMPP"),
                                  system.file("extdata/multipop", "AxC.txt",
                                              package = "statgenMPP")),
                  pheno = pheno,
                  popType = "F4DH",
                  mapFile = system.file("extdata/multipop", "mapfile.txt",
                                        package = "statgenMPP"),
                  evalDist = 5)

summary(ABC)

</code></pre>

<hr>
<h2 id='createGDataMPP'>Create an object of class gDataMPP</h2><span id='topic+createGDataMPP'></span>

<h3>Description</h3>

<p>Function for creating an object of class gDataMPP from an object of class
IBDprob (computed or imported using statgenIBD) and phenotypic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createGDataMPP(IBDprob, pheno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createGDataMPP_+3A_ibdprob">IBDprob</code></td>
<td>
<p>An object of class <code>IBDprob</code>.</p>
</td></tr>
<tr><td><code id="createGDataMPP_+3A_pheno">pheno</code></td>
<td>
<p>A data frame with at least columns &quot;genotype&quot; for the
&quot;genotype&quot; and one or more numerical columns containing phenotypic
information. A column &quot;cross&quot; can be used for indicating the cross the
genotype comes from. This column is ignored if the <code>IBDprob</code> has an
attribute genoCross containing this information (the default behaviour).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>gDataMPP</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Read phenotypic data.
pheno &lt;- read.delim(system.file("extdata/multipop", "AxBxCpheno.txt",
                               package = "statgenMPP"))
## Rename first column to genotype.
colnames(pheno)[1] &lt;- "genotype"

## Compute IBD probabilities for simulated population using statgenIBD - AxB
AB &lt;- statgenIBD::calcIBD(popType = "F4DH",
                         markerFile = system.file("extdata/multipop", "AxB.txt",
                                                 package = "statgenMPP"),
                         mapFile = system.file("extdata/multipop", "mapfile.txt",
                                              package = "statgenMPP"),
                         evalDist = 5)

## Create an object of class gDataMPP from IBD probabilities and
## phenotypic data.
ABMPP &lt;- createGDataMPP(IBDprob = AB, pheno = pheno)

</code></pre>

<hr>
<h2 id='effectPlot'>Create a matrix plot</h2><span id='topic+effectPlot'></span>

<h3>Description</h3>

<p>Create a matrix plot of effect sizes and directions per parent for
significant snps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effectPlot(
  effectDat,
  signSnp,
  map,
  xLab = "Chromosomes",
  yLab = "Parents",
  chrBoundaries,
  title = NULL,
  trait = NULL,
  ...,
  output = TRUE
)
</code></pre>

<hr>
<h2 id='kinshipIBD'>Compute kinship matrix for IBD probabilities</h2><span id='topic+kinshipIBD'></span>

<h3>Description</h3>

<p>Compute a kinship matrix or a list of chromosome specific kinship matrices
for a 3D array of IBD probabilities. The kinship matrix is computed by
averaging <code class="reqn">Z Z^t</code> over all markers, where <code class="reqn">Z</code> is the genotype x
parents matrix for the marker. If <code>chrSpecific = TRUE</code> chromosome
specific kinship matrices are computed for each chromosome based only on the
markers on all other chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinshipIBD(markers, map = NULL, chrSpecific = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kinshipIBD_+3A_markers">markers</code></td>
<td>
<p>An n x m x p array with IBD probabilities with genotypes in
the rows (n), markers in the columns (m), and parents in the 3rd dimension
(p).</p>
</td></tr>
<tr><td><code id="kinshipIBD_+3A_map">map</code></td>
<td>
<p>A data.frame with columns <code>chr</code> for chromosome and
<code>pos</code> for position. Positions should be in centimorgan (cM). They
should not be cumulative over the chromosomes. Other columns are ignored.
Marker names should be in the row names. These should match the marker names
in the input file. Only required if <code>chrSpecific = TRUE</code>.</p>
</td></tr>
<tr><td><code id="kinshipIBD_+3A_chrspecific">chrSpecific</code></td>
<td>
<p>Should chromosome specific kinship matrices be
computed?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A kinship matrix or a list of chromosome specific kinship matrices.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Read phenotypic data.
pheno &lt;- read.delim(system.file("extdata/multipop", "AxBxCpheno.txt",
                               package = "statgenMPP"))
## Rename first column to genotype.
colnames(pheno)[1] &lt;- "genotype"


## Compute IBD probabilities for simulated population - AxB, AxC.
ABC &lt;- calcIBDMPP(crossNames = c("AxB", "AxC"),
                  markerFiles = c(system.file("extdata/multipop", "AxB.txt",
                                              package = "statgenMPP"),
                                  system.file("extdata/multipop", "AxC.txt",
                                              package = "statgenMPP")),
                  pheno = pheno,
                  popType = "F4DH",
                  mapFile = system.file("extdata/multipop", "mapfile.txt",
                                        package = "statgenMPP"),
                  evalDist = 5)

## Compute chromosome specific kinship matrices.
KChrSpec &lt;- kinshipIBD(markers = ABC$markers, map = ABC$map)

</code></pre>

<hr>
<h2 id='maizeMQM'>Pre-computed MQM output maize</h2><span id='topic+maizeMQM'></span>

<h3>Description</h3>

<p>Pre-computed MQM output for maize data used in vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maizeMQM
</code></pre>


<h3>Format</h3>

<p>An object of class <code>QTLMPP</code> (inherits from <code>GWAS</code>) of length 5.
</p>

<hr>
<h2 id='maizeSQM'>Pre-computed SQM output maize</h2><span id='topic+maizeSQM'></span>

<h3>Description</h3>

<p>Pre-computed SQM output for maize data used in vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maizeSQM
</code></pre>


<h3>Format</h3>

<p>An object of class <code>QTLMPP</code> (inherits from <code>GWAS</code>) of length 5.
</p>

<hr>
<h2 id='plot.gDataMPP'>Plot function for the class <code>gDataMPP</code></h2><span id='topic+plot.gDataMPP'></span>

<h3>Description</h3>

<p>Creates a plot of an object of S3 class <code>gDataMPP</code>. The following types
of plot can be made:
</p>

<ul>
<li><p><code>genMap</code> A plot of the genetic map.
</p>
</li>
<li><p><code>allGeno</code> A plot showing for all genotypes the IBD
probabilities of the parent with the highest probability per marker.
</p>
</li>
<li><p><code>singleGeno</code> A plot for a single genotype showing the IBD
probabilities for all parents across the genome.
</p>
</li>
<li><p><code>pedigree</code> A plot showing the structure of the pedigree of
the population.
</p>
</li></ul>

<p>See the respective sections for more details on the plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gDataMPP'
plot(
  x,
  ...,
  plotType = c("genMap", "allGeno", "singleGeno", "pedigree"),
  genotype = NULL,
  title = NULL,
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.gDataMPP_+3A_x">x</code></td>
<td>
<p>An object of class <code>gDataMPP</code>.</p>
</td></tr>
<tr><td><code id="plot.gDataMPP_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed on to the actual plotting
functions.</p>
</td></tr>
<tr><td><code id="plot.gDataMPP_+3A_plottype">plotType</code></td>
<td>
<p>A character string indicating the type of plot to be made.
One of &quot;genMap&quot;, &quot;singleGeno&quot;, &quot;allGeno&quot; or &quot;pedigree&quot;.</p>
</td></tr>
<tr><td><code id="plot.gDataMPP_+3A_genotype">genotype</code></td>
<td>
<p>A character string indicating the genotype for which the
plot should be made. Only for <code>plotType = "singleGeno"</code>.</p>
</td></tr>
<tr><td><code id="plot.gDataMPP_+3A_title">title</code></td>
<td>
<p>A character string, the title of the plot.</p>
</td></tr>
<tr><td><code id="plot.gDataMPP_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code>, only a ggplot object is invisibly returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object is invisibly returned.
</p>


<h3>genMap</h3>

<p>A plot is made showing the lengths of the chromosomes and the position of
the markers that are present in the map. It is possible to highlight one
or more markers using the extra parameter <code>highlight</code>.
</p>


<h3>allGeno</h3>

<p>A plot is made showing all genotypes and markers. Each combination of
genotype and marker is colored according to the parent with the highest
probability. A darker color indicates a higher probability.
</p>


<h3>singleGeno</h3>

<p>A plot is made for a single genotype, specified by
<code>genotpye = "name_of_genotype"</code> showing the IBD probabilities for the
selected genotype for all parents across the genome.
</p>


<h3>pedigree</h3>

<p>A plot is made showing the structure of the pedigree for the population in
the <code>gDataMPP</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Read phenotypic data.
pheno &lt;- read.delim(system.file("extdata/multipop", "AxBxCpheno.txt",
                               package = "statgenMPP"))
## Rename first column to genotype.
colnames(pheno)[1] &lt;- "genotype"


## Compute IBD probabilities for simulated population - AxB, AxC.
ABC &lt;- calcIBDMPP(crossNames = c("AxB", "AxC"),
                  markerFiles = c(system.file("extdata/multipop", "AxB.txt",
                                              package = "statgenMPP"),
                                  system.file("extdata/multipop", "AxC.txt",
                                              package = "statgenMPP")),
                  pheno = pheno,
                  popType = "F4DH",
                  mapFile = system.file("extdata/multipop", "mapfile.txt",
                                        package = "statgenMPP"),
                  evalDist = 5)

## Plot the genetic map.
plot(ABC, plotType = "genMap")

## Plot the genetic map and highlight marker EXT_3_30.
plot(ABC, plotType = "genMap", highlight = "EXT_3_30")

## Plot the IBD probabilities across the genome for all genotypes.
plot(ABC, plotType = "allGeno")

## Plot the IBD probabilities for genotype AxB0001.
plot(ABC, plotType = "singleGeno", genotype = "AxB0001")

## Plot the pedigree.
plot(ABC, plotType = "pedigree")

</code></pre>

<hr>
<h2 id='plot.QTLMPP'>Plot function for the class <code>QTLMPP</code></h2><span id='topic+plot.QTLMPP'></span>

<h3>Description</h3>

<p>Creates a plot of an object of S3 class <code>QTLMPP</code>. The following types of
plot can be made:
</p>

<ul>
<li><p>QTLProfile A QTL profile plot, i.e. a plot of <code class="reqn">-log10(p)</code> values
per marker.
</p>
</li>
<li><p>parEffs  A plot of effect sizes and directions per parent.
</p>
</li>
<li><p>QTLRegion A plot highlighting the QTLs found on the genetic map.
</p>
</li>
<li><p>QTLProfileExt A combination of the QTL profile and parental effects
plotted above each other.
</p>
</li></ul>

<p>See details for a detailed description of the plots and the plot options
specific to the different plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'QTLMPP'
plot(
  x,
  ...,
  plotType = c("QTLProfile", "parEffs", "QTLRegion", "QTLProfileExt"),
  title = NULL,
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.QTLMPP_+3A_x">x</code></td>
<td>
<p>An object of class <code>QTLMPP</code>.</p>
</td></tr>
<tr><td><code id="plot.QTLMPP_+3A_...">...</code></td>
<td>
<p>further arguments to be passed on to the actual plotting
functions.</p>
</td></tr>
<tr><td><code id="plot.QTLMPP_+3A_plottype">plotType</code></td>
<td>
<p>A character string indicating the type of plot to be made.
One of &quot;QTLProfile&quot;, &quot;parEffs&quot;, and &quot;QTLRegion&quot;.</p>
</td></tr>
<tr><td><code id="plot.QTLMPP_+3A_title">title</code></td>
<td>
<p>A character string, the title of the plot.</p>
</td></tr>
<tr><td><code id="plot.QTLMPP_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code>, only a ggplot object is invisibly returned.</p>
</td></tr>
</table>


<h3>QTL Profile plot</h3>

<p>A profile of all marker positions and corresponding <code class="reqn">-log10(p)</code> values is
plotted. QTLs found are highlighted with red dots. The threshold is plotted
as a horizontal line. If there are previously known marker effects, false
positives and true negatives can also be marked.<br />
Extra parameter options:
</p>

<dl>
<dt><code>xLab</code></dt><dd><p>A character string, the x-axis label. Default =
<code>"Chromosomes"</code></p>
</dd>
<dt><code>yLab</code></dt><dd><p>A character string, the y-axis label. Default =
<code>-log10(p)</code></p>
</dd>
<dt><code>effects</code></dt><dd><p>A character vector, indicating which markers correspond
to a real (known) effect. Used for determining true/false positives and
false negatives. True positives are colored green, false positives orange and
false negatives yellow.</p>
</dd>
<dt><code>colPalette</code></dt><dd><p>A color palette used for plotting. Default
coloring is done by chromosome, using black and grey.</p>
</dd>
<dt><code>signLwd</code></dt><dd><p>A numerical value giving the thickness of the
points that are false/true positives/negatives. Default = 0.6</p>
</dd>
<dt><code>chr</code></dt><dd><p>A vector of chromosomes to be plotted. By default, all
chromosomes are plotted. Using this option allows restricting the plot to a
subset of chromosomes.</p>
</dd>
</dl>



<h3>Parental effects Plot</h3>

<p>A plot of effect sizes for each of the parents for the QTLs found is
created.<br />
Extra parameter options:
</p>

<dl>
<dt><code>xLab</code></dt><dd><p>A character string, the x-axis label. Default =
<code>"Chromosomes"</code></p>
</dd>
<dt><code>yLab</code></dt><dd><p>A character string, the y-axis label. Default =
<code>"Parents"</code></p>
</dd>
</dl>
<p>#' </p>
<dl>
<dt><code>chr</code></dt><dd><p>A vector of chromosomes to be plotted. By default, all
chromosomes are plotted. Using this option allows restricting the plot to a
subset of chromosomes.</p>
</dd>
</dl>



<h3>QTL Region Plot</h3>

<p>A plot of effect is created on which the positions of the QTLs found in
the QTL detection are highlighted in red.<br />
No extra parameter options.
</p>


<h3>Extended QTL Profile Plot</h3>

<p>An extended version of the QTL Profile Plot, in which the QLT profile plot is
combined with the parental effect plot to make it easier to assess the
effects for each specific QTL found.<br />
Extra parameter options:
</p>

<dl>
<dt><code>chr</code></dt><dd><p>A vector of chromosomes to be plotted. By default, all
chromosomes are plotted. Using this option allows restricting the plot to a
subset of chromosomes.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Read phenotypic data.
pheno &lt;- read.delim(system.file("extdata/multipop", "AxBxCpheno.txt",
                               package = "statgenMPP"))
## Rename first column to genotype.
colnames(pheno)[1] &lt;- "genotype"


## Compute IBD probabilities for simulated population - AxB, AxC.
ABC &lt;- calcIBDMPP(crossNames = c("AxB", "AxC"),
                  markerFiles = c(system.file("extdata/multipop", "AxB.txt",
                                              package = "statgenMPP"),
                                  system.file("extdata/multipop", "AxC.txt",
                                              package = "statgenMPP")),
                  pheno = pheno,
                  popType = "F4DH",
                  mapFile = system.file("extdata/multipop", "mapfile.txt",
                                        package = "statgenMPP"),
                  evalDist = 5)

## Multi-QTL Mapping.
ABC_MQM &lt;- selQTLMPP(ABC, trait = "yield")

## QTL Profile plot.
plot(ABC_MQM, plotType = "QTLProfile")

## Plot of parental effects for QTLs found.
plot(ABC_MQM, plotType = "parEffs")

## Plot of genetic map highlighting positions of QTLs found.
plot(ABC_MQM, plotType = "QTLRegion")

## Extended QTL Profile plot.
plot(ABC_MQM, plotType = "QTLProfileExt")

## End(Not run)

</code></pre>

<hr>
<h2 id='print.summary.QTLMPP'>Print summary of object of class <code>summary.QTLMPP</code></h2><span id='topic+print.summary.QTLMPP'></span>

<h3>Description</h3>

<p>Print summary of object of class <code>summary.QTLMPP</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'summary.QTLMPP'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.summary.QTLMPP_+3A_x">x</code></td>
<td>
<p>An object of class <code>summary.QTLMPP</code>.</p>
</td></tr>
<tr><td><code id="print.summary.QTLMPP_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>

<hr>
<h2 id='randomQTLmodel'>one position fitted in the mixed model</h2><span id='topic+randomQTLmodel'></span>

<h3>Description</h3>

<p>one position fitted in the mixed model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>randomQTLmodel(
  modDat,
  map,
  parents,
  trait = "yield",
  scanMrk = NULL,
  cofMrk = NULL,
  NULLmodel = FALSE
)
</code></pre>

<hr>
<h2 id='readRABBITMPP'>Read IBD probabilities</h2><span id='topic+readRABBITMPP'></span>

<h3>Description</h3>

<p>Read a file with IBD probabilities computed by the RABBIT software package.
It is possible to additionally read the pedigree file that is also used by
RABBIT. Reading this file allows for plotting the pedigree. Phenotypic data
can be added from a data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readRABBITMPP(infile, pedFile = NULL, pheno = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readRABBITMPP_+3A_infile">infile</code></td>
<td>
<p>A character string, a link to a .csv file with IBD
probabilities.</p>
</td></tr>
<tr><td><code id="readRABBITMPP_+3A_pedfile">pedFile</code></td>
<td>
<p>A character string, a link to a .csv file with pedigree
information as used by RABBIT as input.</p>
</td></tr>
<tr><td><code id="readRABBITMPP_+3A_pheno">pheno</code></td>
<td>
<p>A data frame with at least columns &quot;genotype&quot; for the
&quot;genotype&quot; and one or more numerical columns containing phenotypic
information. A column &quot;cross&quot; can be used for indicating the cross the
genotype comes from.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>gDataMPP</code> with map and markers
corresponding to the imported information in the imported .csv file.
</p>


<h3>References</h3>

<p>Zheng, Chaozhi, Martin P Boer, and Fred A Van Eeuwijk.
“Recursive Algorithms for Modeling Genomic Ancestral Origins in a Fixed
Pedigree.” G3 Genes|Genomes|Genetics 8 (10): 3231–45.
https://doi.org/10.1534/G3.118.200340.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Read RABBIT data for barley.
genoFile &lt;- system.file("extdata/barley", "barley_magicReconstruct.zip",
                       package = "statgenMPP")
barleyMPMPP &lt;- readRABBITMPP(unzip(genoFile, exdir = tempdir()))

</code></pre>

<hr>
<h2 id='scanQTL'>one-round genome scan with specified cofactor(s)</h2><span id='topic+scanQTL'></span>

<h3>Description</h3>

<p>one-round genome scan with specified cofactor(s)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scanQTL(
  modDat,
  map,
  markers,
  parents,
  QTLwindow = 10,
  cof = NULL,
  Usc = NULL,
  trait = NULL,
  maxIter = 100,
  parallel = FALSE,
  verbose = FALSE
)
</code></pre>

<hr>
<h2 id='selQTLMPP'>Multi round genome scans for QTL detection</h2><span id='topic+selQTLMPP'></span>

<h3>Description</h3>

<p>Multi round genome scans for QTL detection.<br /><br />
Several rounds of QTL detection are performed. First a model is fitted
without cofactors. If for at least one marker the <code class="reqn">-log10(p)</code> value is
above the threshold the marker with the lowest p-Value is added as cofactor
in the next round of QTL detection. This process continues until there are
no new markers with a <code class="reqn">-log10(p)</code> value above the threshold or until
the maximum number of cofactors is reached.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selQTLMPP(
  MPPobj,
  trait = NULL,
  QTLwindow = 10,
  threshold = 3,
  maxCofactors = NULL,
  K = NULL,
  computeKin = FALSE,
  parallel = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selQTLMPP_+3A_mppobj">MPPobj</code></td>
<td>
<p>An object of class gDataMPP, typically the output of either
<code><a href="#topic+calcIBDMPP">calcIBDMPP</a></code> or <code><a href="#topic+readRABBITMPP">readRABBITMPP</a></code>.</p>
</td></tr>
<tr><td><code id="selQTLMPP_+3A_trait">trait</code></td>
<td>
<p>A character string indicating the trait QTL mapping is done for.</p>
</td></tr>
<tr><td><code id="selQTLMPP_+3A_qtlwindow">QTLwindow</code></td>
<td>
<p>A numerical value indicating the window around a QTL that
is considered as part of that QTL.</p>
</td></tr>
<tr><td><code id="selQTLMPP_+3A_threshold">threshold</code></td>
<td>
<p>A numerical value indicating the threshold for the
<code class="reqn">-log10p</code> value of a marker to be considered a QTL.</p>
</td></tr>
<tr><td><code id="selQTLMPP_+3A_maxcofactors">maxCofactors</code></td>
<td>
<p>A numerical value, the maximum number of cofactors to
include in the model. If <code>NULL</code> cofactors are added until no new
cofactors are found.</p>
</td></tr>
<tr><td><code id="selQTLMPP_+3A_k">K</code></td>
<td>
<p>A list of chromosome specific kinship matrices. If
<code>NULL</code> and <code>computeKin = FALSE</code> no kinship matrix is included in
the models.</p>
</td></tr>
<tr><td><code id="selQTLMPP_+3A_computekin">computeKin</code></td>
<td>
<p>Should chromosome specific kinship matrices be computed?</p>
</td></tr>
<tr><td><code id="selQTLMPP_+3A_parallel">parallel</code></td>
<td>
<p>Should the computation of variance components be done in
parallel? This requires a parallel back-end to be registered. See examples.</p>
</td></tr>
<tr><td><code id="selQTLMPP_+3A_verbose">verbose</code></td>
<td>
<p>Should progress and intermediate plots be output?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default only family specific effects and residual variances and no
kinship relations are included in the model. It is possible to include
kinship relations by either specifying <code>computeKin = TRUE</code>. When doing
so the kinship matrix is computed by averaging <code class="reqn">Z Z^t</code> over all markers,
where <code class="reqn">Z</code> is the genotype x parents matrix for the marker. It is also
possible to specify a list of precomputed chromosome
specific kinship matrices in <code>K</code>. Note that adding a kinship matrix
to the model increases the computation time a lot, especially for large
populations.
</p>


<h3>Value</h3>

<p>An object of class <code>QTLMPP</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+kinshipIBD">kinshipIBD</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Read phenotypic data.
pheno &lt;- read.delim(system.file("extdata/multipop", "AxBxCpheno.txt",
                               package = "statgenMPP"))
## Rename first column to genotype.
colnames(pheno)[1] &lt;- "genotype"


## Compute IBD probabilities for simulated population - AxB, AxC.
ABC &lt;- calcIBDMPP(crossNames = c("AxB", "AxC"),
                  markerFiles = c(system.file("extdata/multipop", "AxB.txt",
                                              package = "statgenMPP"),
                                  system.file("extdata/multipop", "AxC.txt",
                                              package = "statgenMPP")),
                  pheno = pheno,
                  popType = "F4DH",
                  mapFile = system.file("extdata/multipop", "mapfile.txt",
                                        package = "statgenMPP"),
                  evalDist = 5)

## Single-QTL Mapping.
ABC_SQM &lt;- selQTLMPP(ABC, trait = "yield", maxCofactors = 0)

## Multi-QTL Mapping.
## Not run: 
## Register parallel back-end with 2 cores.
doParallel::registerDoParallel(cores = 2)

## Run multi-QTL mapping.
ABC_MQM &lt;- selQTLMPP(ABC, trait = "yield", parallel = TRUE)

## Run multi-QTL mapping - include kinship matrix.
ABC_MQM_kin &lt;- selQTLMPP(ABC, trait = "yield", parallel = TRUE,
                        computeKin = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='statgenMPP-package'>statgenMPP: QTL Mapping for Multi Parent Populations</h2><span id='topic+statgenMPP'></span><span id='topic+statgenMPP-package'></span>

<h3>Description</h3>

<p>For Multi Parent Populations (MPP) Identity By Descend (IBD) probabilities are computed using Hidden Markov Models. These probabilities are then used in a mixed model approach for QTL Mapping as described in Li et al. (<a href="https://doi.org/10.1007/s00122-021-03919-7">doi:10.1007/s00122-021-03919-7</a>).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Bart-Jan van Rossum <a href="mailto:bart-jan.vanrossum@wur.nl">bart-jan.vanrossum@wur.nl</a> (<a href="https://orcid.org/0000-0002-8673-2514">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Wenhao Li <a href="mailto:wenhao.li@wur.nl">wenhao.li@wur.nl</a> (<a href="https://orcid.org/0000-0001-5719-5775">ORCID</a>)
</p>
</li>
<li><p> Martin Boer <a href="mailto:martin.boer@wur.nl">martin.boer@wur.nl</a> (<a href="https://orcid.org/0000-0002-1879-4588">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Fred van Eeuwijk (<a href="https://orcid.org/0000-0003-3672-2921">ORCID</a>) [contributor]
</p>
</li></ul>


<hr>
<h2 id='summary.gDataMPP'>Summary function for the class <code>gDataMPP</code></h2><span id='topic+summary.gDataMPP'></span>

<h3>Description</h3>

<p>Gives a summary for an object of S3 class <code>gDataMPP</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'gDataMPP'
summary(object, ..., trials = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.gDataMPP_+3A_object">object</code></td>
<td>
<p>An object of class <code>gDataMPP</code>.</p>
</td></tr>
<tr><td><code id="summary.gDataMPP_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="summary.gDataMPP_+3A_trials">trials</code></td>
<td>
<p>A vector of trials to include in the summary. These can
be either numeric indices or character names of list items in <code>pheno</code>.
If <code>NULL</code>, all trials are included.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with a most four components:
</p>

<dl>
<dt>mapSum</dt><dd><p>A list with number of markers and number of chromosomes in
the map.</p>
</dd>
<dt>markerSum</dt><dd><p>A list with number of markers, number of genotypes and
the distribution of the values within the markers.</p>
</dd>
<dt>phenoSum</dt><dd><p>A list of data.frames, one per trial with a summary of all
traits within the trial.</p>
</dd>
<dt>covarSum</dt><dd><p>A list of data.frames, one per trial with a summary of all
covariates within the trial.</p>
</dd>
</dl>

<p>All components are only present in the output if the corresponding content is
present in the <code>gDataMPP</code> object.
</p>

<hr>
<h2 id='summary.QTLMPP'>Summary function for the class <code>QTLMPP</code></h2><span id='topic+summary.QTLMPP'></span>

<h3>Description</h3>

<p>Gives a summary for an object of S3 class <code>QTLMPP</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'QTLMPP'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.QTLMPP_+3A_object">object</code></td>
<td>
<p>An object of class <code>QTLMPP</code>.</p>
</td></tr>
<tr><td><code id="summary.QTLMPP_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Read phenotypic data.
pheno &lt;- read.delim(system.file("extdata/multipop", "AxBxCpheno.txt",
                               package = "statgenMPP"))
## Rename first column to genotype.
colnames(pheno)[1] &lt;- "genotype"

## Compute IBD probabilities for simulated population - AxB, AxC.
ABC &lt;- calcIBDMPP(crossNames = c("AxB", "AxC"),
                  markerFiles = c(system.file("extdata/multipop", "AxB.txt",
                                              package = "statgenMPP"),
                                  system.file("extdata/multipop", "AxC.txt",
                                              package = "statgenMPP")),
                  pheno = pheno,
                  popType = "F4DH",
                  mapFile = system.file("extdata/multipop", "mapfile.txt",
                                        package = "statgenMPP"),
                  evalDist = 5)

## Multi-QTL Mapping.
ABC_MQM &lt;- selQTLMPP(ABC, trait = "yield")

## Print summary.
summary(ABC_MQM)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
