<!DOCTYPE html><html><head><title>Help for package statgenGxE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {statgenGxE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AMMI'><p>S3 class AMMI</p></a></li>
<li><a href='#correlations'><p>Compute different types of correlations.</p></a></li>
<li><a href='#CRDR'><p>Calculate the correlated response to selection</p></a></li>
<li><a href='#createVarComp'><p>S3 class varComp</p></a></li>
<li><a href='#diagnostics'><p>Get diagnostics for an object of class varComp</p></a></li>
<li><a href='#dropsPheno'><p>DROPS data set</p></a></li>
<li><a href='#fitted.AMMI'><p>Extract fitted values.</p></a></li>
<li><a href='#fitted.FW'><p>Extract fitted values.</p></a></li>
<li><a href='#fitted.varCov'><p>Extract fitted values.</p></a></li>
<li><a href='#FW'><p>S3 class FW</p></a></li>
<li><a href='#gxeAmmi'><p>AMMI analysis</p></a></li>
<li><a href='#gxeFw'><p>Finlay-Wilkinson analysis</p></a></li>
<li><a href='#gxeGGE'><p>GGE analysis</p></a></li>
<li><a href='#gxeMegaEnv'><p>Form mega environments based on fitted values from an AMMI model</p></a></li>
<li><a href='#gxeStability'><p>Calculate stability coefficients for genotype-by-environment data</p></a></li>
<li><a href='#gxeVarComp'><p>Mixed model analysis of GxE table of means</p></a></li>
<li><a href='#gxeVarCov'><p>Selects the best variance-covariance model for a set of trials</p></a></li>
<li><a href='#herit'><p>Calculate heritability</p></a></li>
<li><a href='#megaEnv'><p>S3 class megaEnv</p></a></li>
<li><a href='#multMissing'><p>Estimate missing values in multivariate data</p></a></li>
<li><a href='#plot.AMMI'><p>Plot function for class AMMI</p></a></li>
<li><a href='#plot.FW'><p>Plot function for class FW</p></a></li>
<li><a href='#plot.megaEnv'><p>Plot function for class megaEnv</p></a></li>
<li><a href='#plot.stability'><p>Plot function for class stability</p></a></li>
<li><a href='#plot.varComp'><p>Plot function for class varComp</p></a></li>
<li><a href='#plot.varCov'><p>Plot function for class varCov</p></a></li>
<li><a href='#predict.megaEnv'><p>Compute BLUPS based on a set of mega environments</p></a></li>
<li><a href='#predict.varComp'><p>Predictions based on a fitted varComp model.</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#report.AMMI'><p>Report method for class AMMI</p></a></li>
<li><a href='#report.FW'><p>Report method for class FW</p></a></li>
<li><a href='#report.stability'><p>Report method for class stability</p></a></li>
<li><a href='#report.varCov'><p>Report method for class varCov</p></a></li>
<li><a href='#residuals.AMMI'><p>Extract residuals.</p></a></li>
<li><a href='#residuals.FW'><p>Extract residuals.</p></a></li>
<li><a href='#residuals.varCov'><p>Extract residuals.</p></a></li>
<li><a href='#stability'><p>S3 class stability</p></a></li>
<li><a href='#statgenGxE-package'><p>statgenGxE: Genotype by Environment (GxE) Analysis</p></a></li>
<li><a href='#TDMaize'><p>Field data for a maize experiment in Tlaltizapan, Mexico</p></a></li>
<li><a href='#varCov'><p>S3 class varCov</p></a></li>
<li><a href='#vc'><p>Extract variance components</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genotype by Environment (GxE) Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-3-18</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of multi environment data of plant breeding experiments
    following the analyses described in Malosetti, Ribaut, 
    and van Eeuwijk (2013), &lt;<a href="https://doi.org/10.3389%2Ffphys.2013.00044">doi:10.3389/fphys.2013.00044</a>&gt;.
    One of a series of statistical genetic packages for streamlining the analysis of 
    typical plant breeding experiments developed by Biometris.
    Some functions have been created to be used in conjunction with the R 
    package 'asreml' for the 'ASReml' software, which can be obtained upon 
    purchase from 'VSN' international (<a href="https://vsni.co.uk/software/asreml-r">https://vsni.co.uk/software/asreml-r</a>).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>emmeans, ggplot2 (&ge; 3.3.0), gridExtra, knitr, lme4, methods,
rlang, statgenSTA (&ge; 1.0.6), xtable</td>
</tr>
<tr>
<td>Suggests:</td>
<td>asreml(&ge; 3.0), rmarkdown, testthat</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>pdflatex</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://biometris.github.io/statgenGxE/index.html">https://biometris.github.io/statgenGxE/index.html</a>,
<a href="https://github.com/Biometris/statgenGxE/">https://github.com/Biometris/statgenGxE/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Biometris/statgenGxE/issues">https://github.com/Biometris/statgenGxE/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-19 07:05:47 UTC; rossu027</td>
</tr>
<tr>
<td>Author:</td>
<td>Bart-Jan van Rossum [aut, cre],
  Fred van Eeuwijk <a href="https://orcid.org/0000-0003-3672-2921"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Martin Boer [ctb],
  Marcos Malosetti <a href="https://orcid.org/0000-0002-8150-1397"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Daniela Bustos-Korts
    <a href="https://orcid.org/0000-0003-3827-6726"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Emilie J Millet <a href="https://orcid.org/0000-0002-2913-4892"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Joao Paulo <a href="https://orcid.org/0000-0002-4180-0763"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Maikel Verouden <a href="https://orcid.org/0000-0002-4893-3323"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Willem Kruijer <a href="https://orcid.org/0000-0001-7179-1733"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Ron Wehrens <a href="https://orcid.org/0000-0002-8798-5599"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Choazhi Zheng <a href="https://orcid.org/0000-0001-6030-3933"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bart-Jan van Rossum &lt;bart-jan.vanrossum@wur.nl&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-19 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='AMMI'>S3 class AMMI</h2><span id='topic+AMMI'></span><span id='topic+createAMMI'></span>

<h3>Description</h3>

<p>Function for creating objects of S3 class AMMI.<br />
<code><a href="base.html#topic+print">print</a></code>, <code><a href="base.html#topic+summary">summary</a></code>, <code><a href="graphics.html#topic+plot">plot</a></code> and
<code><a href="#topic+report">report</a></code> methods are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createAMMI(
  envScores,
  genoScores,
  importance,
  anova,
  fitted,
  trait,
  envMean,
  genoMean,
  overallMean,
  dat,
  GGE,
  byYear
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AMMI_+3A_envscores">envScores</code></td>
<td>
<p>A matrix containing environmental scores.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_genoscores">genoScores</code></td>
<td>
<p>A matrix containing genotypic scores.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_importance">importance</code></td>
<td>
<p>A data.frame containing the importance of the principal
components.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_anova">anova</code></td>
<td>
<p>A data.frame containing anova scores of the AMMI analysis.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_fitted">fitted</code></td>
<td>
<p>A matrix containing fitted values from the AMMI model.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_trait">trait</code></td>
<td>
<p>A character string indicating the analyzed trait.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_envmean">envMean</code></td>
<td>
<p>A numerical vector containing the environmental means.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_genomean">genoMean</code></td>
<td>
<p>A numerical vector containing the genotypic means.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_overallmean">overallMean</code></td>
<td>
<p>A numerical value containing the overall mean.</p>
</td></tr>
<tr><td><code id="AMMI_+3A_gge">GGE</code></td>
<td>
<p>Has a GGE analysis been performed?</p>
</td></tr>
<tr><td><code id="AMMI_+3A_byyear">byYear</code></td>
<td>
<p>Has the analysis been performed by year?</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+plot.AMMI">plot.AMMI</a></code>, <code><a href="#topic+report.AMMI">report.AMMI</a></code>
</p>

<hr>
<h2 id='correlations'>Compute different types of correlations.</h2><span id='topic+correlations'></span>

<h3>Description</h3>

<p>Compute three types of correlations for models fitted with a nesting factor.
</p>

<ul>
<li><p> correlation between scenarios or environment types:
</p>
<p style="text-align: center;"><code class="reqn">\sigma_G^2 / (\sigma_G^2 + \sigma_{GS}^2)</code>
</p>

</li>
<li><p> correlation between trials within scenarios or environment types:
</p>
<p style="text-align: center;"><code class="reqn">(\sigma_G^2 + \sigma_{GS}^2) / (\sigma_G^2 + \sigma_{GS}^2 +
\sigma_E^2)</code>
</p>

</li>
<li><p> correlation between trials that belong to different
scenarios/environment types:
</p>
<p style="text-align: center;"><code class="reqn">\sigma_G^2 / (\sigma_G^2 + \sigma_{GS}^2 + \sigma_E^2)</code>
</p>

</li></ul>

<p>In these formulas the <code class="reqn">\sigma</code> terms stand for the standard deviations of
the respective model terms. So <code class="reqn">\sigma_S</code> is the standard deviation for
the scenario term in the model, <code class="reqn">\sigma_{GS}</code> for the standard deviation
of the genotype by scenario term and <code class="reqn">\sigma_E</code> corresponds to the
residual standard deviation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correlations(varComp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correlations_+3A_varcomp">varComp</code></td>
<td>
<p>An object of class varComp.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with three correlations.
</p>


<h3>See Also</h3>

<p>Other Mixed model analysis: 
<code><a href="#topic+CRDR">CRDR</a>()</code>,
<code><a href="#topic+diagnostics">diagnostics</a>()</code>,
<code><a href="#topic+gxeVarComp">gxeVarComp</a>()</code>,
<code><a href="#topic+herit">herit</a>()</code>,
<code><a href="#topic+plot.varComp">plot.varComp</a>()</code>,
<code><a href="#topic+predict.varComp">predict.varComp</a>()</code>,
<code><a href="#topic+vc">vc</a>()</code>
</p>

<hr>
<h2 id='CRDR'>Calculate the correlated response to selection</h2><span id='topic+CRDR'></span>

<h3>Description</h3>

<p>Calculate the correlated response to selection (CRDR) based on the fitted
model. The CRDR is calculated as described by Atlin et al. E.g. for a model
with trials nested within scenarios, which has a random part that looks like
this: genotype + genotype:scenario + genotype:scenario:trial the CRDR is
calculated as:<br /><br />
</p>
<p style="text-align: center;"><code class="reqn">H1 = \sigma_G^2 / (\sigma_G^2 + \sigma_S^2 / s + \sigma_{ST}^2 / st +
\sigma_E^2 / str)</code>
</p>

<p style="text-align: center;"><code class="reqn">H2 = (\sigma_G^2 + \sigma_S^2) / (\sigma_G^2 + \sigma_S^2 +
\sigma_{ST}^2 / st + \sigma_E^2 / str)</code>
</p>

<p style="text-align: center;"><code class="reqn">CRDR = (\sigma_G^2 / (\sigma_G^2 + \sigma_S^2)) * sqrt(H1 / H2)</code>
</p>

<p>In these formulas the <code class="reqn">\sigma</code> terms stand for the standard deviations of
the respective model terms, and the lower case letters for the number of
levels for the respective model terms. So <code class="reqn">\sigma_S</code> is the standard
deviation for the scenario term in the model and <code class="reqn">s</code> is the number of
scenarios. <code class="reqn">\sigma_E</code> corresponds to the residual standard deviation and
<code class="reqn">r</code> to the number of replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CRDR(varComp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CRDR_+3A_varcomp">varComp</code></td>
<td>
<p>An object of class varComp.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Atlin, G. N., Baker, R. J., McRae, K. B., &amp; Lu, X. (2000).
Selection response in subdivided target regions. Crop Science, 40(1), 7–13.
<a href="https://doi.org/10.2135/cropsci2000.4017">doi:10.2135/cropsci2000.4017</a>
</p>


<h3>See Also</h3>

<p>Other Mixed model analysis: 
<code><a href="#topic+correlations">correlations</a>()</code>,
<code><a href="#topic+diagnostics">diagnostics</a>()</code>,
<code><a href="#topic+gxeVarComp">gxeVarComp</a>()</code>,
<code><a href="#topic+herit">herit</a>()</code>,
<code><a href="#topic+plot.varComp">plot.varComp</a>()</code>,
<code><a href="#topic+predict.varComp">predict.varComp</a>()</code>,
<code><a href="#topic+vc">vc</a>()</code>
</p>

<hr>
<h2 id='createVarComp'>S3 class varComp</h2><span id='topic+createVarComp'></span>

<h3>Description</h3>

<p>Function for creating objects of S3 class varComp.<br />
<code><a href="base.html#topic+print">print</a></code>, <code><a href="base.html#topic+summary">summary</a></code>, and <code><a href="graphics.html#topic+plot">plot</a></code> methods
are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createVarComp(
  fitMod,
  modDat,
  trait,
  nestingFactor,
  useLocYear,
  useRegionLocYear,
  fullRandVC,
  aovFullFixedMod,
  engine,
  diagTabs
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createVarComp_+3A_fitmod">fitMod</code></td>
<td>
<p>A fitted variance components model.</p>
</td></tr>
<tr><td><code id="createVarComp_+3A_moddat">modDat</code></td>
<td>
<p>A data.frame containing the data used in fitting the model.</p>
</td></tr>
</table>

<hr>
<h2 id='diagnostics'>Get diagnostics for an object of class varComp</h2><span id='topic+diagnostics'></span>

<h3>Description</h3>

<p>Get the diagnostics for the model fitted. This will print a table of
combinations missing in the data. For each random factor in the model a
table is printed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagnostics(varComp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diagnostics_+3A_varcomp">varComp</code></td>
<td>
<p>An object of class varComp.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of tables is invisibly returned.
</p>


<h3>See Also</h3>

<p>Other Mixed model analysis: 
<code><a href="#topic+CRDR">CRDR</a>()</code>,
<code><a href="#topic+correlations">correlations</a>()</code>,
<code><a href="#topic+gxeVarComp">gxeVarComp</a>()</code>,
<code><a href="#topic+herit">herit</a>()</code>,
<code><a href="#topic+plot.varComp">plot.varComp</a>()</code>,
<code><a href="#topic+predict.varComp">predict.varComp</a>()</code>,
<code><a href="#topic+vc">vc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Fit a mixed model.
geVarComp &lt;- gxeVarComp(TD = TDMaize, trait = "yld")

## Display diagnostics.
diagnostics(geVarComp)

</code></pre>

<hr>
<h2 id='dropsPheno'>DROPS data set</h2><span id='topic+dropsPheno'></span>

<h3>Description</h3>

<p>This dataset comes from the European Union project DROPS (DROught-tolerant
yielding PlantS). A panel of 256 maize hybrids was grown with two water
regimes (irrigated or rainfed), in seven fields in 2012 and 2013,
respectively, spread along a climatic transect from western to eastern
Europe, plus one site in Chile in 2013. This resulted in 28 experiments
defined as the combination of one year, one site and one water regime, with
two and three repetitions for rainfed and irrigated treatments, respectively.
A detailed environmental characterisation was carried out, with hourly
records of micrometeorological data and soil water status, and associated
with precise measurement of phenology. Grain yield and its components were
measured at the end of the experiment.<br />
10 experiments have been selected from the full data set, two for each of
the five main environmental scenarios that were identified in the data. The
scenarios have been added to the data as well as a classification of the
genotypes in four genetic groups.<br /><br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dropsPheno
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 2460 rows and 20 columns.
</p>


<h3>Details</h3>

<p>The data.frame contains the genotypic means  (Best Linear Unbiased Estimators,
BLUEs), with one value per experiment  (Location × year × water regime) per
genotype.<br />
A data.frame with 2460 rows and 19 columns.<br />
</p>

<dl>
<dt>Experiment</dt><dd><p>experiments ID described by the three first letters of the
city’s name followed by the year of experiment and the water regime with W
for watered and R for rain-fed.</p>
</dd>
<dt>parent1</dt><dd><p>identifier of donor dent line</p>
</dd>
<dt>Code_ID, Variety_ID, Accession_ID</dt><dd><p>identifier of the genotype</p>
</dd>
<dt>geno.panel</dt><dd><p>project in which the genetic material was generated</p>
</dd>
<dt>grain.yield</dt><dd><p>genotypic mean for yield adjusted at 15\
in ton per hectare (t ha^-1)</p>
</dd>
<dt>grain.number</dt><dd><p>genotypic mean for number of grain per square meter</p>
</dd>
<dt>grain.weight</dt><dd><p>genotypic mean for individual grain weight in milligram
(mg)</p>
</dd>
<dt>anthesis</dt><dd><p>genotypic mean for male flowering (pollen shed), in thermal
time cumulated since emergence (d_20°C)</p>
</dd>
<dt>silking</dt><dd><p>genotypic mean for female flowering (silking emergence), in
thermal time cumulated since emergence (d_20°C)</p>
</dd>
<dt>plant.height</dt><dd><p>genotypic mean for plant height, from ground level to
the base of the flag leaf (highest) leaf in centimeter (cm)</p>
</dd>
<dt>tassel.height</dt><dd><p>genotypic mean for plant height including tassel, from
ground level to the highest point of the tassel in centimeter (cm)</p>
</dd>
<dt>ear.height</dt><dd><p>genotypic mean for ear insertion height, from ground level
to ligule of the highest ear leaf in centimeter (cm)</p>
</dd>
<dt>year</dt><dd><p>year in which the experiment was performed</p>
</dd>
<dt>loc</dt><dd><p>location where the experiment was performed, a three letter
abbreviation</p>
</dd>
<dt>scenarioWater</dt><dd><p>water scenario for the experiment, well watered (WW) or
water deficit (WD)</p>
</dd>
<dt>scenarioTemp</dt><dd><p>temperature scenario for the experiment, Cool, Hot or
Hot(Day)</p>
</dd>
<dt>scenarioFull</dt><dd><p>the full scenario for the experiment, a combination of
scenarioWater and scenarioTemp</p>
</dd>
<dt>geneticGroup</dt><dd><p>the genetic group to which the genotype belongs</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://doi.org/10.15454/IASSTN">doi:10.15454/IASSTN</a>
</p>


<h3>References</h3>

<p>Millet, E. J., Pommier, C., et al. (2019). A multi-site
experiment in a network of European fields for assessing the maize yield
response to environmental scenarios (Data set).
<a href="https://doi.org/10.15454/IASSTN">doi:10.15454/IASSTN</a>
</p>

<hr>
<h2 id='fitted.AMMI'>Extract fitted values.</h2><span id='topic+fitted.AMMI'></span>

<h3>Description</h3>

<p>Extract the fitted values for an object of class AMMI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AMMI'
fitted(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitted.AMMI_+3A_object">object</code></td>
<td>
<p>An object of class AMMI</p>
</td></tr>
<tr><td><code id="fitted.AMMI_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with fitted values.
</p>


<h3>See Also</h3>

<p>Other AMMI: 
<code><a href="#topic+gxeAmmi">gxeAmmi</a>()</code>,
<code><a href="#topic+plot.AMMI">plot.AMMI</a>()</code>,
<code><a href="#topic+report.AMMI">report.AMMI</a>()</code>,
<code><a href="#topic+residuals.AMMI">residuals.AMMI</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run AMMI analysis on TDMaize.
geAmmi &lt;- gxeAmmi(TD = TDMaize, trait = "yld")

## Extract fitted values.
fitAmmi &lt;- fitted(geAmmi)
head(fitAmmi)

</code></pre>

<hr>
<h2 id='fitted.FW'>Extract fitted values.</h2><span id='topic+fitted.FW'></span>

<h3>Description</h3>

<p>Extract the fitted values for a fitted Finlay-Wilkinson model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FW'
fitted(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitted.FW_+3A_object">object</code></td>
<td>
<p>An object of class FW</p>
</td></tr>
<tr><td><code id="fitted.FW_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with fitted values.
</p>


<h3>See Also</h3>

<p>Other Finlay-Wilkinson: 
<code><a href="#topic+gxeFw">gxeFw</a>()</code>,
<code><a href="#topic+plot.FW">plot.FW</a>()</code>,
<code><a href="#topic+report.FW">report.FW</a>()</code>,
<code><a href="#topic+residuals.FW">residuals.FW</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run Finlay-Wilkinson analysis.
geFW &lt;- gxeFw(TD = TDMaize, trait = "yld")

## Extract fitted values.
fitFW &lt;- fitted(geFW)
head(fitFW)

</code></pre>

<hr>
<h2 id='fitted.varCov'>Extract fitted values.</h2><span id='topic+fitted.varCov'></span>

<h3>Description</h3>

<p>Extract the fitted values for an object of class varCov.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'varCov'
fitted(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitted.varCov_+3A_object">object</code></td>
<td>
<p>An object of class varCov</p>
</td></tr>
<tr><td><code id="fitted.varCov_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with fitted values.
</p>


<h3>See Also</h3>

<p>Other varCov: 
<code><a href="#topic+gxeVarCov">gxeVarCov</a>()</code>,
<code><a href="#topic+plot.varCov">plot.varCov</a>()</code>,
<code><a href="#topic+report.varCov">report.varCov</a>()</code>,
<code><a href="#topic+residuals.varCov">residuals.varCov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("asreml", quietly = TRUE)) {
  ## Select the best variance-covariance model using asreml for modeling.
  geVarCov &lt;- gxeVarCov(TD = TDMaize, trait = "yld", engine = "asreml")
  ## Extract fitted values from the model.

  fitVarCov &lt;- fitted(geVarCov)
  head(fitVarCov)
  }


</code></pre>

<hr>
<h2 id='FW'>S3 class FW</h2><span id='topic+FW'></span><span id='topic+createFW'></span>

<h3>Description</h3>

<p>Function for creating objects of S3 class FW (Finlay-Wilkinson).<br />
<code><a href="base.html#topic+print">print</a></code>, <code><a href="base.html#topic+summary">summary</a></code>, <code><a href="graphics.html#topic+plot">plot</a></code> and
<code><a href="#topic+report">report</a></code> methods are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createFW(
  estimates,
  anova,
  envEffs,
  trait,
  nGeno,
  nEnv,
  TD,
  fittedGeno,
  tol,
  iter
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FW_+3A_estimates">estimates</code></td>
<td>
<p>A data.frame containing the estimated values.</p>
</td></tr>
<tr><td><code id="FW_+3A_anova">anova</code></td>
<td>
<p>A data.frame containing anova scores of the FW analysis.</p>
</td></tr>
<tr><td><code id="FW_+3A_enveffs">envEffs</code></td>
<td>
<p>A data.frame containing the environmental effects.</p>
</td></tr>
<tr><td><code id="FW_+3A_trait">trait</code></td>
<td>
<p>A character value indicating the analysed trait.</p>
</td></tr>
<tr><td><code id="FW_+3A_ngeno">nGeno</code></td>
<td>
<p>A numerical value containing the number of genotypes in the
analysis.</p>
</td></tr>
<tr><td><code id="FW_+3A_nenv">nEnv</code></td>
<td>
<p>A numerical value containing the number of environments in the
analysis.</p>
</td></tr>
<tr><td><code id="FW_+3A_td">TD</code></td>
<td>
<p>The object of class <code><a href="statgenSTA.html#topic+TD">TD</a></code> on which the analysis was
performed.</p>
</td></tr>
<tr><td><code id="FW_+3A_fittedgeno">fittedGeno</code></td>
<td>
<p>The fitted values for the genotypes.</p>
</td></tr>
<tr><td><code id="FW_+3A_tol">tol</code></td>
<td>
<p>A numerical value containing the tolerance used during the
analysis.</p>
</td></tr>
<tr><td><code id="FW_+3A_iter">iter</code></td>
<td>
<p>A numerical value containing the number of iterations for the
analysis to converge.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+plot.FW">plot.FW</a></code>, <code><a href="#topic+report.FW">report.FW</a></code>
</p>

<hr>
<h2 id='gxeAmmi'>AMMI analysis</h2><span id='topic+gxeAmmi'></span>

<h3>Description</h3>

<p>The Additive Main Effects and Multiplicative Interaction (AMMI) model fits
a model which involves the Additive Main effects (i.e. genotype and trial)
along with Multiplicative Interaction effects. The additive effects are the
classical ANOVA main effects for genotype and environment, the
multiplicative effects follow from a principal component analysis on the
interaction residuals (= genotype by environment means after adjustment for
additive genotype and environment effects). This results in an interaction
characterized by Interaction Principal Components (IPCA) enabling
simultaneous plotting of genotypes and trials.<br /><br />
The parameter <code>nPC</code> is used to indicate the number of principal
components that is used in the principal component analysis (PCA). By setting
this parameter to <code>NULL</code> the algorithm determines the best number of
principal components (see Details).<br /><br />
By specifying the parameter <code>byYear = TRUE</code>, a separate analysis will be
done for every year in the data. Combining the option with <code>nPC = NULL</code>
may result in different numbers of principal components per year. The AMMI
estimates will still be returned as a single data.frame, but the other
results will be either lists or arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gxeAmmi(
  TD,
  trials = names(TD),
  trait,
  nPC = 2,
  byYear = FALSE,
  center = TRUE,
  excludeGeno = NULL,
  useWt = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gxeAmmi_+3A_td">TD</code></td>
<td>
<p>An object of class <code><a href="statgenSTA.html#topic+TD">TD</a></code>.</p>
</td></tr>
<tr><td><code id="gxeAmmi_+3A_trials">trials</code></td>
<td>
<p>A character string specifying the trials to be analyzed. If
not supplied, all trials are used in the analysis.</p>
</td></tr>
<tr><td><code id="gxeAmmi_+3A_trait">trait</code></td>
<td>
<p>A character string specifying the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="gxeAmmi_+3A_npc">nPC</code></td>
<td>
<p>An integer specifying the number of principal components used
as multiplicative term of genotype-by-trial interaction. If <code>NULL</code>, the
number of principal components is determined by the algorithm using
forward selection. See details.</p>
</td></tr>
<tr><td><code id="gxeAmmi_+3A_byyear">byYear</code></td>
<td>
<p>Should the analysis be done by year? If <code>TRUE</code> the data
is split by the variable year, analysis is performed and the results are
merged together and returned.</p>
</td></tr>
<tr><td><code id="gxeAmmi_+3A_center">center</code></td>
<td>
<p>Should the variables be shifted to be zero centered?</p>
</td></tr>
<tr><td><code id="gxeAmmi_+3A_excludegeno">excludeGeno</code></td>
<td>
<p>An optional character vector with names of genotypes to
be excluded from the analysis. If <code>NULL</code>, all genotypes are used.</p>
</td></tr>
<tr><td><code id="gxeAmmi_+3A_usewt">useWt</code></td>
<td>
<p>Should weighting be used when modeling? Requires a column
<code>wt</code> in <code>TD</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First a linear model <code class="reqn">trait = genotype + trial + \epsilon</code> is fitted with
both genotype and trial fixed components in the model.<br />
The residuals from the fitted model are then used in a PCA. If <code>nPC</code> is
not <code>NULL</code> a single PCA is done using <code><a href="stats.html#topic+prcomp">prcomp</a></code> with
maximum rank <code>nPC</code>.<br />
In case <code>nPC = NULL</code>, the PCA is first done with one PC. Then using
forward selection one by one the number of PCs is increased as long as the
added component is significant in the analysis.<br />
AMMI estimates are then computed using the results of the PCA.<br />
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+AMMI">AMMI</a></code>, a list containing:
</p>
<table>
<tr><td><code>envScores</code></td>
<td>
<p>A matrix with environmental scores.</p>
</td></tr>
<tr><td><code>genoScores</code></td>
<td>
<p>A matrix with genotypic scores.</p>
</td></tr>
<tr><td><code>importance</code></td>
<td>
<p>A data.frame containing the importance of the principal
components.</p>
</td></tr>
<tr><td><code>anova</code></td>
<td>
<p>A data.frame containing anova scores of the AMMI analysis.</p>
</td></tr>
<tr><td><code>fitted</code></td>
<td>
<p>A data.frame containing fitted values from the AMMI model.</p>
</td></tr>
<tr><td><code>trait</code></td>
<td>
<p>A character string containing the analyzed trait.</p>
</td></tr>
<tr><td><code>envMean</code></td>
<td>
<p>A numerical vector containing the environmental means.</p>
</td></tr>
<tr><td><code>genoMean</code></td>
<td>
<p>A numerical vector containing the genotypic means.</p>
</td></tr>
<tr><td><code>overallMean</code></td>
<td>
<p>A numerical value containing the overall mean.</p>
</td></tr>
</table>
<p>If <code>byYear</code> = <code>TRUE</code>, all returned items in the AMMI object except
<code>fitted</code> will consist of a list of results by year.
</p>


<h3>References</h3>

<p>Gauch H.G. (1992) Statistical Analysis of Regional Yield Trials:
AMMI Analysis of Factorial Designs. Elsevier, Amsterdam.
</p>
<p>Yan, W., Kang, M. (2002). GGE Biplot Analysis. Boca Raton: CRC
Press.
</p>


<h3>See Also</h3>

<p>Other AMMI: 
<code><a href="#topic+fitted.AMMI">fitted.AMMI</a>()</code>,
<code><a href="#topic+plot.AMMI">plot.AMMI</a>()</code>,
<code><a href="#topic+report.AMMI">report.AMMI</a>()</code>,
<code><a href="#topic+residuals.AMMI">residuals.AMMI</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run AMMI analysis on TDMaize.
geAmmi &lt;- gxeAmmi(TD = TDMaize, trait = "yld")

## Summarize results.
summary(geAmmi)

## Create a biplot of genotypes and environment interaction with PC1 and PC2.
plot(geAmmi, plotType = "AMMI2")

## Create a pdf report summarizing the results.

report(geAmmi, outfile = tempfile(fileext = ".pdf"))


</code></pre>

<hr>
<h2 id='gxeFw'>Finlay-Wilkinson analysis</h2><span id='topic+gxeFw'></span>

<h3>Description</h3>

<p>This function performs a Finlay-Wilkinson analysis of data classified by two
factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gxeFw(
  TD,
  trials = names(TD),
  trait,
  maxIter = 15,
  tol = 0.001,
  sorted = c("descending", "ascending", "none"),
  genotypes = NULL,
  useWt = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gxeFw_+3A_td">TD</code></td>
<td>
<p>An object of class <code><a href="statgenSTA.html#topic+TD">TD</a></code>.</p>
</td></tr>
<tr><td><code id="gxeFw_+3A_trials">trials</code></td>
<td>
<p>A character string specifying the trials to be analyzed. If
not supplied, all trials are used in the analysis.</p>
</td></tr>
<tr><td><code id="gxeFw_+3A_trait">trait</code></td>
<td>
<p>A character string specifying the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="gxeFw_+3A_maxiter">maxIter</code></td>
<td>
<p>An integer specifying the maximum number of iterations in
the algorithm.</p>
</td></tr>
<tr><td><code id="gxeFw_+3A_tol">tol</code></td>
<td>
<p>A positive numerical value specifying convergence tolerance of the
algorithm.</p>
</td></tr>
<tr><td><code id="gxeFw_+3A_sorted">sorted</code></td>
<td>
<p>A character string specifying the sorting order of the
estimated values in the output.</p>
</td></tr>
<tr><td><code id="gxeFw_+3A_genotypes">genotypes</code></td>
<td>
<p>An optional character string containing the genotypes to
which the analysis should be restricted. If <code>NULL</code>, all genotypes are
used.</p>
</td></tr>
<tr><td><code id="gxeFw_+3A_usewt">useWt</code></td>
<td>
<p>Should weighting be used when modeling? Requires a column
<code>wt</code> in <code>TD</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+FW">FW</a></code>, a list containing:
</p>
<table>
<tr><td><code>estimates</code></td>
<td>
<p>A data.frame containing the estimated values, with the
following columns:
</p>

<ul>
<li><p> genotype: The name of the genotype.
</p>
</li>
<li><p> sens: The estimate of the sensitivity.
</p>
</li>
<li><p> se_sens: The standard error of the estimate of the sensitivity.
</p>
</li>
<li><p> genMean: The estimate of the genotypic mean.
</p>
</li>
<li><p> se_genMean: The standard error of the estimate of the genotypic
mean.
</p>
</li>
<li><p> MSdeviation: The mean square deviation about the line fitted to
each genotype
</p>
</li>
<li><p> rank: The rank of the genotype based on its sensitivity.
</p>
</li></ul>

</td></tr>
<tr><td><code>anova</code></td>
<td>
<p>A data.frame containing anova scores of the FW analysis.</p>
</td></tr>
<tr><td><code>envEffs</code></td>
<td>
<p>A data.frame containing the environmental effects, with the
following columns:
</p>

<ul>
<li><p> trial: The name of the trial.
</p>
</li>
<li><p> envEff: The estimate of the environment effect.
</p>
</li>
<li><p> se_envEff: The standard error of the estimate of the environment
effect.
</p>
</li>
<li><p> envMean: The estimate of the environment mean.
</p>
</li>
<li><p> rank: The rank of the trial based on its mean.
</p>
</li></ul>

</td></tr>
<tr><td><code>TD</code></td>
<td>
<p>The object of class TD on which the analysis was performed.</p>
</td></tr>
<tr><td><code>fittedGeno</code></td>
<td>
<p>A numerical vector containing the fitted values for the
genotypes.</p>
</td></tr>
<tr><td><code>trait</code></td>
<td>
<p>A character string containing the analyzed trait.</p>
</td></tr>
<tr><td><code>nGeno</code></td>
<td>
<p>A numerical value containing the number of genotypes in the
analysis.</p>
</td></tr>
<tr><td><code>nEnv</code></td>
<td>
<p>A numerical value containing the number of environments in the
analysis.</p>
</td></tr>
<tr><td><code>tol</code></td>
<td>
<p>A numerical value containing the tolerance used during the
analysis.</p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p>A numerical value containing the number of iterations for the
analysis to converge.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Finlay, K.W. &amp; Wilkinson, G.N. (1963). The analysis of adaptation
in a plant-breeding programme. Australian Journal of Agricultural
Research, 14, 742-754.
</p>


<h3>See Also</h3>

<p>Other Finlay-Wilkinson: 
<code><a href="#topic+fitted.FW">fitted.FW</a>()</code>,
<code><a href="#topic+plot.FW">plot.FW</a>()</code>,
<code><a href="#topic+report.FW">report.FW</a>()</code>,
<code><a href="#topic+residuals.FW">residuals.FW</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run Finlay-Wilkinson analysis on TDMaize.
geFW &lt;- gxeFw(TDMaize, trait = "yld")

## Summarize results.
summary(geFW)

## Create a scatterplot of the results.
plot(geFW, plotType = "scatter")


## Create a report summarizing the results.
report(geFW, outfile = tempfile(fileext = ".pdf"))


</code></pre>

<hr>
<h2 id='gxeGGE'>GGE analysis</h2><span id='topic+gxeGGE'></span>

<h3>Description</h3>

<p>The Genotype plus Genotype by Environment interaction (GGE) model fits
a model with trial as main fixed effect. Then a principal component
analysis is done on the residuals. This results in an interaction
characterized by Interaction Principal Components (IPCA)
enabling simultaneous plotting of genotypes and trials.<br /><br />
The parameter <code>nPC</code> is used to indicate the number of principal
components that is used in the principal component analysis (PCA). By setting
this parameter to <code>NULL</code> the algorithm determines the best number of
principal components (see Details).<br /><br />
By specifying the parameter <code>byYear = TRUE</code>, a separate analysis will be
done for every year in the data. Combining the option with <code>nPC = NULL</code>
may result in different numbers of principal components per year. The GGE
estimates will still be returned as a single data.frame, but the other
results will be either lists or arrays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gxeGGE(
  TD,
  trials = names(TD),
  trait,
  nPC = 2,
  byYear = FALSE,
  center = TRUE,
  excludeGeno = NULL,
  useWt = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gxeGGE_+3A_td">TD</code></td>
<td>
<p>An object of class <code><a href="statgenSTA.html#topic+TD">TD</a></code>.</p>
</td></tr>
<tr><td><code id="gxeGGE_+3A_trials">trials</code></td>
<td>
<p>A character string specifying the trials to be analyzed. If
not supplied, all trials are used in the analysis.</p>
</td></tr>
<tr><td><code id="gxeGGE_+3A_trait">trait</code></td>
<td>
<p>A character string specifying the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="gxeGGE_+3A_npc">nPC</code></td>
<td>
<p>An integer specifying the number of principal components used
as multiplicative term of genotype-by-trial interaction. If <code>NULL</code>, the
number of principal components is determined by the algorithm using
forward selection. See details.</p>
</td></tr>
<tr><td><code id="gxeGGE_+3A_byyear">byYear</code></td>
<td>
<p>Should the analysis be done by year? If <code>TRUE</code> the data
is split by the variable year, analysis is performed and the results are
merged together and returned.</p>
</td></tr>
<tr><td><code id="gxeGGE_+3A_center">center</code></td>
<td>
<p>Should the variables be shifted to be zero centered?</p>
</td></tr>
<tr><td><code id="gxeGGE_+3A_excludegeno">excludeGeno</code></td>
<td>
<p>An optional character vector with names of genotypes to
be excluded from the analysis. If <code>NULL</code>, all genotypes are used.</p>
</td></tr>
<tr><td><code id="gxeGGE_+3A_usewt">useWt</code></td>
<td>
<p>Should weighting be used when modeling? Requires a column
<code>wt</code> in <code>TD</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First a linear model <code class="reqn">trait = trial + \epsilon</code> is fitted with trial a
fixed component in the model.<br />
The residuals from the fitted model are then used in a PCA. If <code>nPC</code> is
not <code>NULL</code> a single PCA is done using <code><a href="stats.html#topic+prcomp">prcomp</a></code> with
maximum rank <code>nPC</code>.<br />
In case <code>nPC = NULL</code>, the PCA is first done with one PC. Then using
forward selection one by one the number of PCs is increased as long as the
added component is significant in the analysis.<br />
GGE estimates are then computed using the results of the PCA.<br />
</p>

<hr>
<h2 id='gxeMegaEnv'>Form mega environments based on fitted values from an AMMI model</h2><span id='topic+gxeMegaEnv'></span>

<h3>Description</h3>

<p>This function fits an AMMI model and then using the fitted values produces
a new factor clustering the trials. This factor is added as a column megaEnv
to the input data. If a column megaEnv already exists this column is
overwritten with a warning.<br /><br />
Mega environments are created by grouping environments based on their best
performing genotype; i.e. environments that share the same best genotype
belong to the same mega environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gxeMegaEnv(
  TD,
  trials = names(TD),
  trait,
  method = c("max", "min"),
  byYear = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gxeMegaEnv_+3A_td">TD</code></td>
<td>
<p>An object of class <code><a href="statgenSTA.html#topic+TD">TD</a></code>.</p>
</td></tr>
<tr><td><code id="gxeMegaEnv_+3A_trials">trials</code></td>
<td>
<p>A character string specifying the trials to be analyzed. If
not supplied, all trials are used in the analysis.</p>
</td></tr>
<tr><td><code id="gxeMegaEnv_+3A_trait">trait</code></td>
<td>
<p>A character string specifying the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="gxeMegaEnv_+3A_method">method</code></td>
<td>
<p>A character string indicating the criterion to determine
the best genotype per environment, either <code>"max"</code> or <code>"min"</code>.</p>
</td></tr>
<tr><td><code id="gxeMegaEnv_+3A_byyear">byYear</code></td>
<td>
<p>Should the analysis be done by year? If <code>TRUE</code> the data
is split by the variable year, analysis is performed and the results are
merged together and returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class megaEnv, a list consisting of
</p>

<dl>
<dt>TD</dt><dd><p>An object of class TD, the TD object used as input to the function
with an extra column megaEnv.</p>
</dd>
<dt>summTab</dt><dd><p>A data.frame, a summary table containing information on the
trials in each mega environment.</p>
</dd>
<dt>trait</dt><dd><p>The trait used for calculating the mega environments.</p>
</dd>
</dl>



<h3>References</h3>

<p>Atlin, G. N., R. J. Baker, K. B. McRae, and X. Lu. 2000.
Selection Response in Subdivided Target Regions. Crop Sci. 40:7-13.
<a href="https://doi.org/10.2135/cropsci2000.4017">doi:10.2135/cropsci2000.4017</a>
</p>


<h3>See Also</h3>

<p>Other mega environments: 
<code><a href="#topic+plot.megaEnv">plot.megaEnv</a>()</code>,
<code><a href="#topic+predict.megaEnv">predict.megaEnv</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Calculate mega environments for TDMaize.
gemegaEnv &lt;- gxeMegaEnv(TD = TDMaize, trait = "yld")

## Calculate new mega environments based on the genotypes with the lowest
## value per environment.
gemegaEnv2 &lt;- gxeMegaEnv(TD = TDMaize, trait = "yld", method = "min")

</code></pre>

<hr>
<h2 id='gxeStability'>Calculate stability coefficients for genotype-by-environment data</h2><span id='topic+gxeStability'></span>

<h3>Description</h3>

<p>This function calculates different measures of stability, the
cultivar-superiority measure of Lin &amp; Binns (1988), Shukla's (1972) stability
variance and Wricke's (1962) ecovalence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gxeStability(
  TD,
  trials = names(TD),
  trait,
  method = c("superiority", "static", "wricke"),
  bestMethod = c("max", "min"),
  sorted = c("descending", "ascending", "none")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gxeStability_+3A_td">TD</code></td>
<td>
<p>An object of class <code><a href="statgenSTA.html#topic+TD">TD</a></code>.</p>
</td></tr>
<tr><td><code id="gxeStability_+3A_trials">trials</code></td>
<td>
<p>A character string specifying the trials to be analyzed. If
not supplied, all trials are used in the analysis.</p>
</td></tr>
<tr><td><code id="gxeStability_+3A_trait">trait</code></td>
<td>
<p>A character string specifying the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="gxeStability_+3A_method">method</code></td>
<td>
<p>A character vector specifying the measures of stability to be
calculated. Options are &quot;superiority&quot; (cultivar-superiority measure),
&quot;static&quot; (Shukla's stability variance) or &quot;wricke&quot; (wricke's ecovalence).</p>
</td></tr>
<tr><td><code id="gxeStability_+3A_bestmethod">bestMethod</code></td>
<td>
<p>A character string specifying the criterion to define
the best genotype. Either <code>"max"</code> or <code>"min"</code>.</p>
</td></tr>
<tr><td><code id="gxeStability_+3A_sorted">sorted</code></td>
<td>
<p>A character string specifying the sorting order of the results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+stability">stability</a></code>, a list containing:
</p>
<table>
<tr><td><code>superiority</code></td>
<td>
<p>A data.frame containing values for the
cultivar-superiority measure of Lin and Binns.</p>
</td></tr>
<tr><td><code>static</code></td>
<td>
<p>A data.frame containing values for Shukla's stability
variance.</p>
</td></tr>
<tr><td><code>wricke</code></td>
<td>
<p>A data.frame containing values for Wricke's ecovalence.</p>
</td></tr>
<tr><td><code>trait</code></td>
<td>
<p>A character string indicating the trait that has been analyzed.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lin, C. S. and Binns, M. R. 1988. A superiority measure of
cultivar performance for cultivar x location data. Can. J. Plant Sci. 68:
193-198
</p>
<p>Shukla, G.K. 1972. Some statistical aspects of partitioning
genotype-environmental components of variability. Heredity 29:237-245
</p>
<p>Wricke, G. Uber eine method zur erfassung der okologischen
streubreit in feldversuchen. Zeitschrift für Pflanzenzucht,
v. 47, p. 92-96, 1962
</p>


<h3>See Also</h3>

<p>Other stability: 
<code><a href="#topic+plot.stability">plot.stability</a>()</code>,
<code><a href="#topic+report.stability">report.stability</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute three stability measures for TDMaize.
geStab &lt;- gxeStability(TD = TDMaize, trait = "yld")

## Summarize results.
summary(geStab)

## Create plot of the computed stability measures against the means.
plot(geStab)


## Create a .pdf report summarizing the stability measures.
report(geStab, outfile = tempfile(fileext = ".pdf"))


## Compute Wricke's ecovalance for TDMaize with minimal values for yield as
## the best values. Sort results in ascending order.
geStab2 &lt;- gxeStability(TD = TDMaize, trait = "yld", method = "wricke",
                       bestMethod = "min", sorted = "ascending")
summary(geStab2)

</code></pre>

<hr>
<h2 id='gxeVarComp'>Mixed model analysis of GxE table of means</h2><span id='topic+gxeVarComp'></span>

<h3>Description</h3>

<p>This function fits a mixed model best fitting to the data in a TD object.
The exact model fitted is determined by both the structure of the genotype by
environment table of observations and the chosen parameters.<br /><br />
Six different types of models can be fitted depending on the structure of
the environments in the data. These models are described in the table below,
together with the function parameters used in <code>gxeVarComp</code> to fit the model.</p>

<table>
<tr>
 <td style="text-align: left;">
   Structure of environments </td><td style="text-align: left;"> Model </td><td style="text-align: left;"> Function parameters </td>
</tr>
<tr>
 <td style="text-align: left;">
   Environments correspond to trials </td><td style="text-align: left;"> <strong>trait</strong> = trial + <strong>genotype + genotype:trial</strong> </td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
   Trials form a factorial structure of locations x years </td><td style="text-align: left;"> <strong>trait</strong> = year + location + year:location + <strong>genotype + genotype:year + genotype:location + genotype:year:location</strong> </td><td style="text-align: left;"> <code>locationYear = TRUE</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Trials are nested within year </td><td style="text-align: left;"> <strong>trait</strong> = year + year:trial + <strong>genotype + genotype:year + genotype:year:trial</strong> </td><td style="text-align: left;"> <code>nestingFactor = "year"</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Trials are nested within locations </td><td style="text-align: left;"> <strong>trait</strong> = location + location:trial + <strong>genotype + genotype:location + genotype:location:trial</strong> </td><td style="text-align: left;"> <code>nestingFactor = "loc"</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Trials correspond to locations within regions across years </td><td style="text-align: left;"> <strong>trait</strong> = region + region:location + year + region:year + region:location:year + <strong>genotype + genotype:region + genotype:region:location + genotype:year + genotype:region:year + genotype:region:location:year</strong> </td><td style="text-align: left;"> <code>regionLocationYear = TRUE</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Trials are nested within scenarios </td><td style="text-align: left;"> <strong>trait</strong> = scenario + scenario:trial + <strong>genotype + genotype:scenario + genotype:scenario:trial</strong> </td><td style="text-align: left;"> <code>nestingFactor = "scenario"</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>In the models above the random part of the model is printed bold.<br />
For data in the form of GxE means, the last random term in all models above
will become a residual term. If the GxE means are provided together with
weights, then a residual term will be added to the models above.<br /><br />
The function first fits a model where all model terms are included as fixed
terms. Based on the ANOVA table of this model, terms in the fixed part of the
model that are likely to give a problem when fitting the mixed model are
removed because of the reduced connectivity and number of available
observations to estimate that model term. Also a warning is printed if the
mean sum of squares for a model term points to a possible zero variance
component in the mixed model.<br /><br />
Then a model is fitted where all model terms are included as random terms.
Based on the variance components in this model the percentage of variance
explained by each of the model components is determined. The percentages of
variance are printed in the model summary, together with the variance
components. The latter are presented on a standard deviation scale.<br /><br />
Finally a mixed model is fitted as specified in the overview above. Based on
this mixed model variance components can be computed using <code><a href="#topic+vc">vc</a></code>,
heritabilies can be computed using <code><a href="#topic+herit">herit</a></code> and predictions can be
made using <code><a href="#topic+predict.varComp">predict.varComp</a></code>. Predictions of genotypic
performance can be made at the level of individual trials, or for groups of
trials by using <code>predictLevel</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gxeVarComp(
  TD,
  trials = names(TD),
  trait,
  engine = c("lme4", "asreml"),
  locationYear = FALSE,
  nestingFactor = NULL,
  regionLocationYear = FALSE,
  useWt = FALSE,
  diagnostics = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gxeVarComp_+3A_td">TD</code></td>
<td>
<p>An object of class <code><a href="statgenSTA.html#topic+TD">TD</a></code>.</p>
</td></tr>
<tr><td><code id="gxeVarComp_+3A_trials">trials</code></td>
<td>
<p>A character string specifying the trials to be analyzed. If
not supplied, all trials are used in the analysis.</p>
</td></tr>
<tr><td><code id="gxeVarComp_+3A_trait">trait</code></td>
<td>
<p>A character string specifying the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="gxeVarComp_+3A_engine">engine</code></td>
<td>
<p>A character string specifying the engine used for modeling.
Either &quot;lme4&quot; or &quot;asreml&quot;.</p>
</td></tr>
<tr><td><code id="gxeVarComp_+3A_locationyear">locationYear</code></td>
<td>
<p>Should a model be fitted assuming a factorial structure
of locations x years?</p>
</td></tr>
<tr><td><code id="gxeVarComp_+3A_nestingfactor">nestingFactor</code></td>
<td>
<p>A character string specifying a column in TD specifying
the nesting structure of the trials.</p>
</td></tr>
<tr><td><code id="gxeVarComp_+3A_regionlocationyear">regionLocationYear</code></td>
<td>
<p>Should a model be fitted assuming locations within
regions across years?</p>
</td></tr>
<tr><td><code id="gxeVarComp_+3A_usewt">useWt</code></td>
<td>
<p>Should the model be fitted using weights? Doing so requires a
column wt in the data. If <code>useWt = FALSE</code>, the default, and the data
contains no replicates, the last model term will be dropped and used as
homogeneous residual.</p>
</td></tr>
<tr><td><code id="gxeVarComp_+3A_diagnostics">diagnostics</code></td>
<td>
<p>Should diagnostics on missing combinations of model
variables be printed?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>varComp</code>, a list containing:
</p>
<table>
<tr><td><code>fitMod</code></td>
<td>
<p>The fitted model.</p>
</td></tr>
<tr><td><code>modDat</code></td>
<td>
<p>A data.frame containing the data used when fitting the model.</p>
</td></tr>
<tr><td><code>nestingFactor</code></td>
<td>
<p>A name of the variable used as nesting variable in the
model.</p>
</td></tr>
<tr><td><code>useLocYear</code></td>
<td>
<p>A boolean specifying if a model containing location x year
interaction was fitted.</p>
</td></tr>
<tr><td><code>fullRandVC</code></td>
<td>
<p>A data.frame containing the variance components for the
fully random model.</p>
</td></tr>
<tr><td><code>aovFullMixedMod</code></td>
<td>
<p>A data.frame containing the ANOVA table for the fully
fixed model.</p>
</td></tr>
<tr><td><code>engine</code></td>
<td>
<p>The engine used for fitting the model.</p>
</td></tr>
<tr><td><code>diagTabs</code></td>
<td>
<p>A list of data.frame, one for each random model term,
containing the missing combinations in the data for that term.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other Mixed model analysis: 
<code><a href="#topic+CRDR">CRDR</a>()</code>,
<code><a href="#topic+correlations">correlations</a>()</code>,
<code><a href="#topic+diagnostics">diagnostics</a>()</code>,
<code><a href="#topic+herit">herit</a>()</code>,
<code><a href="#topic+plot.varComp">plot.varComp</a>()</code>,
<code><a href="#topic+predict.varComp">predict.varComp</a>()</code>,
<code><a href="#topic+vc">vc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Fit a mixed model.
geVarComp &lt;- gxeVarComp(TD = TDMaize, trait = "yld")

## Summarize results.
summary(geVarComp)

## Plot the standard deviations.
plot(geVarComp)

## Generate predictions
pred &lt;- predict(geVarComp, predictLevel = "trial")
head(pred)

</code></pre>

<hr>
<h2 id='gxeVarCov'>Selects the best variance-covariance model for a set of trials</h2><span id='topic+gxeVarCov'></span>

<h3>Description</h3>

<p>This function selects the best covariance structure for genetic correlations
between trials. It fits a range of variance-covariance models (identity,
compound symmetry (cs), diagonal, simple correlation with heterogeneous
variance (outside), heterogeneous compound symmetry (hcs),
first order factor analytic (fa), second order factor analytic (fa2) and
unstructured), and selects the best one using a goodness-of-fit criterion.
See details for the exact models fitted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gxeVarCov(
  TD,
  trials = names(TD),
  trait,
  models = c("identity", "cs", "diagonal", "hcs", "outside", "fa", "fa2", "unstructured"),
  engine = c("lme4", "asreml"),
  criterion = c("BIC", "AIC"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gxeVarCov_+3A_td">TD</code></td>
<td>
<p>An object of class <code><a href="statgenSTA.html#topic+TD">TD</a></code>.</p>
</td></tr>
<tr><td><code id="gxeVarCov_+3A_trials">trials</code></td>
<td>
<p>A character string specifying the trials to be analyzed. If
not supplied, all trials are used in the analysis.</p>
</td></tr>
<tr><td><code id="gxeVarCov_+3A_trait">trait</code></td>
<td>
<p>A character string specifying the trait to be analyzed.</p>
</td></tr>
<tr><td><code id="gxeVarCov_+3A_models">models</code></td>
<td>
<p>A character vector specifying the models to be fitted.</p>
</td></tr>
<tr><td><code id="gxeVarCov_+3A_engine">engine</code></td>
<td>
<p>A character string specifying the engine used for modeling.
Either &quot;lme4&quot; or &quot;asreml&quot;.</p>
</td></tr>
<tr><td><code id="gxeVarCov_+3A_criterion">criterion</code></td>
<td>
<p>A string specifying a goodness-of-fit criterion. Either
&quot;AIC&quot; or &quot;BIC&quot;.</p>
</td></tr>
<tr><td><code id="gxeVarCov_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to the modeling engine.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The models fitted are of the form <code class="reqn">y_{ij} = \mu_j + \epsilon_{ij}</code>, where
<code class="reqn">y_{ij}</code> is the phenotypic value of genotype <code class="reqn">i</code> in environment
<code class="reqn">j</code>, <code class="reqn">\mu_j</code>
is the environmental mean, and <code class="reqn">\epsilon_{ij}</code> represents mainly genetic
variation, although some non-genetic variation may be included as well.
The random term <code class="reqn">\epsilon_{ij}</code> is modeled in eight ways as described in
the table below.</p>

<table>
<tr>
 <td style="text-align: left;">
   Model </td><td style="text-align: left;"> Description </td><td style="text-align: left;"> var(<code class="reqn">g_{ij}</code>) </td><td style="text-align: left;"> cov(<code class="reqn">g_{ij}</code>;<code class="reqn">g_{ik}</code>) </td><td style="text-align: left;"> Number of parameters </td>
</tr>
<tr>
 <td style="text-align: left;">
   identity </td><td style="text-align: left;"> identity </td><td style="text-align: left;"> <code class="reqn">\sigma_G^2</code> </td><td style="text-align: left;"> 0 </td><td style="text-align: left;"> 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
   cs </td><td style="text-align: left;"> compound symmetry </td><td style="text-align: left;"> <code class="reqn">\sigma_G^2+\sigma_{GE}^2</code> </td><td style="text-align: left;"> <code class="reqn">\sigma_{GE}^2</code> </td><td style="text-align: left;"> 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
   diagonal </td><td style="text-align: left;"> diagonal matrix (heteroscedastic) </td><td style="text-align: left;"> <code class="reqn">\sigma_{GE_j}^2</code> </td><td style="text-align: left;"> 0 </td><td style="text-align: left;"> <code class="reqn">J</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   hcs </td><td style="text-align: left;"> heterogeneous compound symmetry </td><td style="text-align: left;"> <code class="reqn">\sigma_G^2+\sigma_{GE_j}^2</code> </td><td style="text-align: left;"> <code class="reqn">\sigma_G^2</code> </td><td style="text-align: left;"> <code class="reqn">J+1</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   outside </td><td style="text-align: left;"> heterogeneity outside </td><td style="text-align: left;"> <code class="reqn">\sigma_{G_j}^2</code> </td><td style="text-align: left;"> <code class="reqn">\theta</code> </td><td style="text-align: left;"> <code class="reqn">J+1</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   fa </td><td style="text-align: left;"> first order factor analytic </td><td style="text-align: left;"> <code class="reqn">\lambda_{1j}^2+\sigma_{GE_j}^2</code> </td><td style="text-align: left;"> <code class="reqn">\lambda_{1j}\lambda_{1k}</code> </td><td style="text-align: left;"> <code class="reqn">2J</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   fa2 </td><td style="text-align: left;"> second order factor analytic </td><td style="text-align: left;"> <code class="reqn">\lambda_{1j}^2+\lambda_{2j}^2+\sigma_{GE_j}^2</code> </td><td style="text-align: left;"> <code class="reqn">\lambda_{1j}\lambda_{1k}+\lambda_{2j}\lambda_{2k}</code> </td><td style="text-align: left;"> <code class="reqn">3J-1</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   unstructured </td><td style="text-align: left;"> unstructured </td><td style="text-align: left;"> <code class="reqn">\sigma_{G_j}^2</code> </td><td style="text-align: left;"> <code class="reqn">\sigma_{G_{j,k}}^2</code> </td><td style="text-align: left;"> <code class="reqn">J(J+1)/2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>In this table <code class="reqn">J</code> is the number of environments, <code class="reqn">\sigma_G^2</code> the
variance component for the genotype main effects, <code class="reqn">\sigma_{GE}^2</code> the
variance component for GxE interactions. <code class="reqn">\sigma_{G_j}^2</code> and
<code class="reqn">\sigma_{GE_j}^2</code> are the environment specific variance components for
the genotype main effects and GxE interaction in environment <code class="reqn">j</code>.
<code class="reqn">\sigma_{G_{j,k}}^2</code> is the genetic covariance between environments
<code class="reqn">j</code> and <code class="reqn">k</code>. <code class="reqn">\theta</code> is the common correlation between
environments and <code class="reqn">\lambda_{1j}</code> and <code class="reqn">\lambda_{2j}</code> are
environment specific multiplicative parameters.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+varCov">varCov</a></code>, a list object containing:
</p>
<table>
<tr><td><code>STA</code></td>
<td>
<p>An object of class STA containing the best fitted model.</p>
</td></tr>
<tr><td><code>choice</code></td>
<td>
<p>A character string indicating the best fitted model.</p>
</td></tr>
<tr><td><code>summary</code></td>
<td>
<p>A data.frame with a summary of the fitted models.</p>
</td></tr>
<tr><td><code>vcov</code></td>
<td>
<p>The covariance matrix of the best fitted model.</p>
</td></tr>
<tr><td><code>criterion</code></td>
<td>
<p>A character string indicating the goodness-of-fit criterion
used for determining the best model, either &quot;AIC&quot; or &quot;BIC&quot;.</p>
</td></tr>
<tr><td><code>engine</code></td>
<td>
<p>A character string containing the engine used for
the analysis.</p>
</td></tr>
<tr><td><code>trait</code></td>
<td>
<p>A character string containing the trait analyzed.</p>
</td></tr>
<tr><td><code>dat</code></td>
<td>
<p>A data.frame with the full data set used for the analysis.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>If <code>engine = "lme4"</code>, only the compound symmetry model can be
fitted.
</p>


<h3>See Also</h3>

<p>Other varCov: 
<code><a href="#topic+fitted.varCov">fitted.varCov</a>()</code>,
<code><a href="#topic+plot.varCov">plot.varCov</a>()</code>,
<code><a href="#topic+report.varCov">report.varCov</a>()</code>,
<code><a href="#topic+residuals.varCov">residuals.varCov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Select the best variance-covariance model using lme4 for modeling.
geVarCov &lt;- gxeVarCov(TD = TDMaize, trait = "yld")

## Summarize results.
summary(geVarCov)


## Create a pdf report summarizing the results.
report(geVarCov, outfile = tempfile(fileext = ".pdf"))



if (requireNamespace("asreml", quietly = TRUE)) {
  ## Select the best variance-covariance model using asreml for modeling.
  ## Use BIC as a goodness-of-fit criterion.
  geVarCov2 &lt;- gxeVarCov(TD = TDMaize, trait = "yld", engine = "asreml",
                         criterion = "BIC")

  summary(geVarCov2)

  ## Plot a heatmap of the correlation matrix for the best model.
  plot(geVarCov2)
  }


</code></pre>

<hr>
<h2 id='herit'>Calculate heritability</h2><span id='topic+herit'></span>

<h3>Description</h3>

<p>Calculate the heritability based on the fitted model. The heritability is
calculated as described by Atlin et al. E.g. for a model with trials nested
within locations, which has a random part that looks like this: genotype +
genotype:location + genotype:location:trial the heritability is computed
as<br /><br />
</p>
<p style="text-align: center;"><code class="reqn">\sigma_G^2 / (\sigma_G^2 + \sigma_L^2 / l + \sigma_{LT}^2 / lt +
\sigma_E^2 / ltr)</code>
</p>

<p>In this formula the <code class="reqn">\sigma</code> terms stand for the standard deviations of
the respective model terms, and the lower case letters for the number of
levels for the respective model terms. So <code class="reqn">\sigma_L</code> is the standard
deviation for the location term in the model and <code class="reqn">l</code> is the number of
locations. <code class="reqn">\sigma_E</code> corresponds to the residual standard deviation and
<code class="reqn">r</code> to the number of replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>herit(varComp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="herit_+3A_varcomp">varComp</code></td>
<td>
<p>An object of class varComp.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Atlin, G. N., Baker, R. J., McRae, K. B., &amp; Lu, X. (2000).
Selection response in subdivided target regions. Crop Science, 40(1), 7–13.
<a href="https://doi.org/10.2135/cropsci2000.4017">doi:10.2135/cropsci2000.4017</a>
</p>


<h3>See Also</h3>

<p>Other Mixed model analysis: 
<code><a href="#topic+CRDR">CRDR</a>()</code>,
<code><a href="#topic+correlations">correlations</a>()</code>,
<code><a href="#topic+diagnostics">diagnostics</a>()</code>,
<code><a href="#topic+gxeVarComp">gxeVarComp</a>()</code>,
<code><a href="#topic+plot.varComp">plot.varComp</a>()</code>,
<code><a href="#topic+predict.varComp">predict.varComp</a>()</code>,
<code><a href="#topic+vc">vc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Fit a mixed model.
geVarComp &lt;- gxeVarComp(TD = TDMaize, trait = "yld")

## Compute heritability.
herit(geVarComp)

</code></pre>

<hr>
<h2 id='megaEnv'>S3 class megaEnv</h2><span id='topic+megaEnv'></span><span id='topic+createMegaEnv'></span>

<h3>Description</h3>

<p>Function for creating objects of S3 class megaEnv.<br />
<code><a href="base.html#topic+print">print</a></code>, <code><a href="base.html#topic+summary">summary</a></code>, <code><a href="graphics.html#topic+plot">plot</a></code> and
<code><a href="#topic+report">report</a></code> methods are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createMegaEnv(TD, summTab, trait)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="megaEnv_+3A_td">TD</code></td>
<td>
<p>A data.frame containing values for the
cultivar-superiority measure of Lin and Binns.</p>
</td></tr>
<tr><td><code id="megaEnv_+3A_summtab">summTab</code></td>
<td>
<p>A data.frame containing values for Shukla's stability variance.</p>
</td></tr>
<tr><td><code id="megaEnv_+3A_trait">trait</code></td>
<td>
<p>A character string indicating the trait that has been analyzed.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+predict.megaEnv">predict.megaEnv</a></code>
</p>

<hr>
<h2 id='multMissing'>Estimate missing values in multivariate data</h2><span id='topic+multMissing'></span>

<h3>Description</h3>

<p>This function estimates missing values for units in a multivariate dataset,
using an iterative regression technique.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multMissing(Y, maxIter = 10, naStrings = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multMissing_+3A_y">Y</code></td>
<td>
<p>A matrix, data.frame or vector of multivariate data.</p>
</td></tr>
<tr><td><code id="multMissing_+3A_maxiter">maxIter</code></td>
<td>
<p>An integer specifying the maximum number of iterations.</p>
</td></tr>
<tr><td><code id="multMissing_+3A_nastrings">naStrings</code></td>
<td>
<p>A character vector of strings which are to be interpreted
as <code>NA</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Initial estimates of the missing values in each variate are formed from the
variate means using the values for units that have no missing values for
any variate. Estimates of the missing values for each variate are then
recalculated as the fitted values from the multiple regression of that
variate on all the other variates. When all the missing values have been
estimated the variate means are recalculated. If any of the means differs
from the previous mean by more than a tolerance (the initial standard error
divided by 1000) the process is repeated, subject to a maximum number of
repetitions defined by <code>maxIter</code> option. The default maximum number
of iterations (10) is usually sufficient when there are few missing
values, say two or three. If there are many more, 20 or so, it may be
necessary to increase the maximum number of iterations to around 30. The
method is similar to that of Orchard &amp; Woodbury (1972), but does not adjust
for bias in the variance-covariance matrix as suggested by Beale &amp;
Little (1975).
</p>


<h3>Value</h3>

<p>An object of the same class as the input <code>Y</code> with the
missing values replaced by their estimates.
</p>


<h3>References</h3>

<p>Beale, E.M.L. &amp; Little, R.J.A. (1975). Missing values in
multivariate analysis. Journal of the Royal Statistical Society, Series B,
37, 129-145.
</p>
<p>Orchard, T. &amp; Woodbury, M.A. (1972). A missing information principle:
theory and applications. In: Proceedings of the 6th Berkeley Symposium in
Mathematical Statistics and Probability, Vol I, 697-715.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>M &lt;- matrix(c("1", "2", "3", NA, "b", "5", "6",
              "6", "5", "b", NA, "3", "2", "1"), nrow = 7, ncol = 2)

## Estimate missing values treating "b" as NA.
multMissing(M, naStrings = "b")

</code></pre>

<hr>
<h2 id='plot.AMMI'>Plot function for class AMMI</h2><span id='topic+plot.AMMI'></span>

<h3>Description</h3>

<p>Two types of biplot can be made. A plot of genotype and environment
means vs PC1 (AMMI1) or a biplot of genotypes and environment interaction
with PC1 and PC2 (AMMI2).<br /><br />
If the AMMI analysis was done by year, a separate plot will be made for
every year in the data. For some years the number of principal components
may be lower than the number specified on the secondary axis. If this is the
case this year is skipped when plotting. If this happens for all years the
function returns an error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AMMI'
plot(
  x,
  ...,
  plotType = c("AMMI1", "AMMI2", "GGE2"),
  scale = 1,
  plotGeno = TRUE,
  colorGenoBy = NULL,
  colGeno = NULL,
  sizeGeno = 0,
  plotConvHull = FALSE,
  plotEnv = TRUE,
  colorEnvBy = NULL,
  colEnv = NULL,
  sizeEnv = 3,
  envFactor = 1,
  primAxis = "PC1",
  secAxis = "PC2",
  rotatePC = NULL,
  title = NULL,
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.AMMI_+3A_x">x</code></td>
<td>
<p>An object of class AMMI</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_plottype">plotType</code></td>
<td>
<p>A character string indicating which plot should be made.
Either &quot;AMMI1&quot; for an AMMI1 plot (genotype and
environment means vs PC1) or &quot;AMMI2&quot; for an AMMI2 biplot (genotypes and
environment interaction with PC1 and PC2) respectively. For results of a
GGE analysis only an GGE2 biplot can be made and plotType may be ignored.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_scale">scale</code></td>
<td>
<p>A numerical value. The variables are scaled by
<code>lambda ^ scale</code> and the observations by <code>lambda ^ (1 - scale)</code>
where <code>lambda</code> are the singular values computed by
<code><a href="stats.html#topic+princomp">princomp</a></code> in <code><a href="#topic+gxeAmmi">gxeAmmi</a></code>. Normally
<code>0 &lt;= scale &lt;= 1</code>, and a warning will be issued if the specified
scale is outside this range.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_plotgeno">plotGeno</code></td>
<td>
<p>Should genotypes be plotted?</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_colorgenoby">colorGenoBy</code></td>
<td>
<p>A character string indicating a column in the <code>TD</code>
used as input for the AMMI analysis by which the genotypes should be colored.
If <code>NULL</code> all genotypes will be colored in black.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_colgeno">colGeno</code></td>
<td>
<p>A character vector with plot colors for the genotypes. A
single color when <code>colorGenoBy = NULL</code>, a vector of colors otherwise.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_sizegeno">sizeGeno</code></td>
<td>
<p>An numerical value indicating the text size for plotting the
genotypes. Use <code>sizeGeno = 0</code> for plotting genotypes as points instead
of using their names.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_plotconvhull">plotConvHull</code></td>
<td>
<p>Should a convex hull be plotted around the genotypes. If
<code>TRUE</code> a convex hull is plotted. For GGE2 biplots lines from the origin
of the plot perpendicular to the edges of the hull are added. Only valid for
AMMI2 and GGE2 biplots.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_plotenv">plotEnv</code></td>
<td>
<p>Should environments be plotted?</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_colorenvby">colorEnvBy</code></td>
<td>
<p>A character string indicating a column in the <code>TD</code>
used as input for the AMMI analysis by which the environments should be
colored. If <code>NULL</code> all genotypes will be colored in red.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_colenv">colEnv</code></td>
<td>
<p>A character string with plot colors for the environments. A
single color when <code>colorEnvBy = NULL</code>, a vector of colors otherwise.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_sizeenv">sizeEnv</code></td>
<td>
<p>An integer indicating the text size for plotting the
environments.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_envfactor">envFactor</code></td>
<td>
<p>A positive numerical value giving a factor by which to blow
up the environmental scores. Providing a value between 0 and 1 will
effectively blow up the genotypic scores.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_primaxis">primAxis</code></td>
<td>
<p>A character string indicating the principal component to be
plotted on the primary axis of the AMMI2 plot. Has to be given as
<code>"PCn"</code> where n is the number of the principal component.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_secaxis">secAxis</code></td>
<td>
<p>A character string indicating the principal component to be
plotted on the secondary axis of the AMMI2 plot. Has to be given as
<code>"PCn"</code> where n is the number of the principal component. n Has to
differ from <code>primAxis</code>.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_rotatepc">rotatePC</code></td>
<td>
<p>A character string indicating a genotype or environment that
is to be aligned with the positive x-axis in the plot.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_title">title</code></td>
<td>
<p>A character string used a title for the plot.</p>
</td></tr>
<tr><td><code id="plot.AMMI_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code> only a list of ggplot objects is invisibly returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A biplot depending on <code>plotType</code>. The ggplot object for the
biplot is returned invisibly.
</p>


<h3>See Also</h3>

<p>Other AMMI: 
<code><a href="#topic+fitted.AMMI">fitted.AMMI</a>()</code>,
<code><a href="#topic+gxeAmmi">gxeAmmi</a>()</code>,
<code><a href="#topic+report.AMMI">report.AMMI</a>()</code>,
<code><a href="#topic+residuals.AMMI">residuals.AMMI</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run AMMI analysis.
geAmmi &lt;- gxeAmmi(TD = TDMaize, trait = "yld")

## Create an AMMI1 biplot.
plot(geAmmi)

## Create an AMMI2 biplot.
plot(geAmmi, plotType = "AMMI2", scale = 0.5)

## Create an AMMI2 biplot, with HN96b along the positive x-axis.
plot(geAmmi, plotType = "AMMI2", scale = 0.5, rotatePC = "HN96b")

## Run GGE analysis.
geGGE &lt;- gxeGGE(TD = TDMaize, trait = "yld")

## Create an GGE2 biplot.
## Add a convex hull.
plot(geGGE, plotType = "GGE2", scale = 0.5, plotConvHull = TRUE)

</code></pre>

<hr>
<h2 id='plot.FW'>Plot function for class FW</h2><span id='topic+plot.FW'></span>

<h3>Description</h3>

<p>Four types of plot can be made. A scatter plot for genotypic mean,
square root of mean squared deviation and sensitivity, a line plot with
fitted lines for each genotype, a trellis plot with individual slopes per
genotype and a scatter plot of fitted values in the worst and best trial.<br />
It is possible to select genotypes for the trellis plot using the
<code>genotypes</code> parameter. If there are more than 64 genotypes, only the
first 64 are plotted in the trellis plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FW'
plot(
  x,
  ...,
  plotType = c("scatter", "line", "trellis", "scatterFit"),
  order = c("ascending", "descending"),
  response = c("predicted", "observed"),
  colorGenoBy = NULL,
  colGeno = NULL,
  genotypes = NULL,
  title = paste("Finlay &amp; Wilkinson analysis for", x$trait),
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.FW_+3A_x">x</code></td>
<td>
<p>An object of class FW.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_plottype">plotType</code></td>
<td>
<p>A character string indicating which plot should be made.
Either &quot;scatter&quot;, &quot;line&quot; or &quot;trellis&quot; for creating a scatter
plot of genotypic means, mse and sensitivities, a plot of fitted lines for
each genotype or a trellis plot of the individual genotype slopes
respectively.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_order">order</code></td>
<td>
<p>A character string specifying whether the results in the line
plot should be ordered in an increasing (or decreasing) order of
sensitivities. Ignored if <code>plotType</code> is not &quot;line&quot;.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_response">response</code></td>
<td>
<p>A character string specifying whether in the line plot the
&quot;predicted&quot; or the &quot;observed&quot; data should be plotted. Ignored if
<code>plotType</code> is not &quot;line&quot;.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_colorgenoby">colorGenoBy</code></td>
<td>
<p>A character string indicating a column in the <code>TD</code>
used as input for the Finlay Wilkinson analysis by which the genotypes
should be colored. If <code>NULL</code> all genotypes will be colored differently.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_colgeno">colGeno</code></td>
<td>
<p>A character vector with plot colors for the genotypes. A
single color when <code>colorGenoBy = NULL</code>, a vector of colors otherwise.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_genotypes">genotypes</code></td>
<td>
<p>An optional character string containing the genotypes to
be plotted in the trellis plot. If <code>NULL</code> all genotypes are plotted.
If more than 64 genotypes are selected, only the first 64 are plotted.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_title">title</code></td>
<td>
<p>A character string used a title for the plot.</p>
</td></tr>
<tr><td><code id="plot.FW_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code>, only a list of ggplot objects is invisibly returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot depending on <code>plotType</code>.
</p>


<h3>See Also</h3>

<p>Other Finlay-Wilkinson: 
<code><a href="#topic+fitted.FW">fitted.FW</a>()</code>,
<code><a href="#topic+gxeFw">gxeFw</a>()</code>,
<code><a href="#topic+report.FW">report.FW</a>()</code>,
<code><a href="#topic+residuals.FW">residuals.FW</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run Finlay-Wilkinson analysis.
geFW &lt;- gxeFw(TD = TDMaize, trait = "yld")

## Create a scatter plot.
plot(geFW)

## Create a line plot.
plot(geFW, plotType = "line")

## Create a line plot showing observed data value for genotypes and fitted lines.
## Display trials in descending order.
plot(geFW, plotType = "line", order = "descending", response = "observed")


## Create a trellis plot.
plot(geFW, plotType = "trellis")

## Create a scatter plot of fitted values for the worst and best trials.
plot(geFW, plotType = "scatterFit")


</code></pre>

<hr>
<h2 id='plot.megaEnv'>Plot function for class megaEnv</h2><span id='topic+plot.megaEnv'></span>

<h3>Description</h3>

<p>Function for creating scatter plots of predicted values in computed mega
environments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'megaEnv'
plot(
  x,
  ...,
  engine = c("lme4", "asreml"),
  colorGenoBy = NULL,
  title = paste("Scatterplot of mega environments for", x$trait),
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.megaEnv_+3A_x">x</code></td>
<td>
<p>An object of class megaEnv.</p>
</td></tr>
<tr><td><code id="plot.megaEnv_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed on to underlying plot functions.</p>
</td></tr>
<tr><td><code id="plot.megaEnv_+3A_engine">engine</code></td>
<td>
<p>A character string specifying the engine used for making the
predictions on which the plots are based.</p>
</td></tr>
<tr><td><code id="plot.megaEnv_+3A_colorgenoby">colorGenoBy</code></td>
<td>
<p>A character string indicating a column in <code>TD</code> by
which the genotypes in the scatter plots are colored. If <code>NULL</code> all
genotypes are displayed in black.</p>
</td></tr>
<tr><td><code id="plot.megaEnv_+3A_title">title</code></td>
<td>
<p>A character string used a title for the plot.</p>
</td></tr>
<tr><td><code id="plot.megaEnv_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code> only a ggtable object is invisibly returned.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other mega environments: 
<code><a href="#topic+gxeMegaEnv">gxeMegaEnv</a>()</code>,
<code><a href="#topic+predict.megaEnv">predict.megaEnv</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute mega environments for TDMaize.
geMegaEnv &lt;- gxeMegaEnv(TD = TDMaize, trait = "yld")

## Create a scatter plot of predicted values.
plot(geMegaEnv)

</code></pre>

<hr>
<h2 id='plot.stability'>Plot function for class stability</h2><span id='topic+plot.stability'></span>

<h3>Description</h3>

<p>Function for creating scatter plots of the square roots of the computed
stability measures against the means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'stability'
plot(
  x,
  ...,
  colorGenoBy = NULL,
  colGeno = NULL,
  title = paste("Stability coefficients for", x$trait),
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.stability_+3A_x">x</code></td>
<td>
<p>An object of class stability.</p>
</td></tr>
<tr><td><code id="plot.stability_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="plot.stability_+3A_colorgenoby">colorGenoBy</code></td>
<td>
<p>A character string indicating a column in the <code>TD</code>
used as input for the stability analysis by which the genotypes should be
colored. If <code>NULL</code> all genotypes will be colored black.</p>
</td></tr>
<tr><td><code id="plot.stability_+3A_colgeno">colGeno</code></td>
<td>
<p>A character vector with plot colors for the genotypes. A
single color when <code>colorGenoBy = NULL</code>, a vector of colors otherwise.</p>
</td></tr>
<tr><td><code id="plot.stability_+3A_title">title</code></td>
<td>
<p>A character string used a title for the plot.</p>
</td></tr>
<tr><td><code id="plot.stability_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code> only a list of ggplot objects is invisibly returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of ggplot object is invisibly returned.
</p>


<h3>See Also</h3>

<p>Other stability: 
<code><a href="#topic+gxeStability">gxeStability</a>()</code>,
<code><a href="#topic+report.stability">report.stability</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute three stability measures for TDMaize.
geStab &lt;- gxeStability(TD = TDMaize, trait = "yld")

## Create scatter plots of the computed stability measures against the means.
plot(geStab)

</code></pre>

<hr>
<h2 id='plot.varComp'>Plot function for class varComp</h2><span id='topic+plot.varComp'></span>

<h3>Description</h3>

<p>A plot is created of either the standard deviations of each of the terms in
the fitted model or the percentage of variance explained by each of the
terms in the fitted model. Also the degrees of freedom for each of the
terms is shown in the plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'varComp'
plot(x, ..., plotType = c("sd", "percVar"), title = NULL, output = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.varComp_+3A_x">x</code></td>
<td>
<p>An object of class varComp</p>
</td></tr>
<tr><td><code id="plot.varComp_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="plot.varComp_+3A_plottype">plotType</code></td>
<td>
<p>A character string. Either &quot;sd&quot; to plot the standard
deviation of the variance components, or &quot;percVar&quot; to plot the percentage of
variance explained by each variance component.</p>
</td></tr>
<tr><td><code id="plot.varComp_+3A_title">title</code></td>
<td>
<p>A character string used a title for the plot.</p>
</td></tr>
<tr><td><code id="plot.varComp_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code> only a ggplot object is invisibly returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object is invisibly returned.
</p>


<h3>See Also</h3>

<p>Other Mixed model analysis: 
<code><a href="#topic+CRDR">CRDR</a>()</code>,
<code><a href="#topic+correlations">correlations</a>()</code>,
<code><a href="#topic+diagnostics">diagnostics</a>()</code>,
<code><a href="#topic+gxeVarComp">gxeVarComp</a>()</code>,
<code><a href="#topic+herit">herit</a>()</code>,
<code><a href="#topic+predict.varComp">predict.varComp</a>()</code>,
<code><a href="#topic+vc">vc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Fit a mixed model.
geVarComp &lt;- gxeVarComp(TD = TDMaize, trait = "yld")

## Plot the standard deviations.
plot(geVarComp)
## Plot the percentage of variance explained.
plot(geVarComp, plotType = "percVar")

</code></pre>

<hr>
<h2 id='plot.varCov'>Plot function for class varCov</h2><span id='topic+plot.varCov'></span>

<h3>Description</h3>

<p>Function for plotting a heatmap of the correlation matrix for objects of
class varCov.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'varCov'
plot(x, title = paste("Heatmap for model:", x$choice), ..., output = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.varCov_+3A_x">x</code></td>
<td>
<p>An object of class varCov</p>
</td></tr>
<tr><td><code id="plot.varCov_+3A_title">title</code></td>
<td>
<p>A character string used a title for the plot.</p>
</td></tr>
<tr><td><code id="plot.varCov_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="plot.varCov_+3A_output">output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code> only a ggplot object is invisibly returned.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other varCov: 
<code><a href="#topic+fitted.varCov">fitted.varCov</a>()</code>,
<code><a href="#topic+gxeVarCov">gxeVarCov</a>()</code>,
<code><a href="#topic+report.varCov">report.varCov</a>()</code>,
<code><a href="#topic+residuals.varCov">residuals.varCov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("asreml", quietly = TRUE)) {
  ## Select the best variance-covariance model using asreml for modeling.
  geVarCov &lt;- gxeVarCov(TD = TDMaize, trait = "yld", engine = "asreml")

  ## Create a heatmap of the correlation matrix for the best model.
  plot(geVarCov)
  }


</code></pre>

<hr>
<h2 id='predict.megaEnv'>Compute BLUPS based on a set of mega environments</h2><span id='topic+predict.megaEnv'></span>

<h3>Description</h3>

<p>This function calculates Best Linear Unbiased Predictors (BLUPS) and
associated standard errors based on a set of mega environments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'megaEnv'
predict(
  object,
  ...,
  trials = names(object$TD),
  useYear = FALSE,
  engine = c("lme4", "asreml")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.megaEnv_+3A_object">object</code></td>
<td>
<p>An object of class megaEnv.</p>
</td></tr>
<tr><td><code id="predict.megaEnv_+3A_...">...</code></td>
<td>
<p>Further parameters passed to either <code>asreml</code> or <code>lmer</code>.</p>
</td></tr>
<tr><td><code id="predict.megaEnv_+3A_trials">trials</code></td>
<td>
<p>A character string specifying the trials to be analyzed. If
not supplied, all trials are used in the analysis.</p>
</td></tr>
<tr><td><code id="predict.megaEnv_+3A_useyear">useYear</code></td>
<td>
<p>Should year be used for modeling (as years within
trials). If <code>TRUE</code>, <code>TD</code> should contain a column &quot;year&quot;.</p>
</td></tr>
<tr><td><code id="predict.megaEnv_+3A_engine">engine</code></td>
<td>
<p>A character string specifying the engine used for modeling.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list consisting of two data.frames, <code>predictedValue</code>
containing BLUPs per genotype per mega environment and <code>standardError</code>
containing standard errors for those BLUPs.
</p>


<h3>See Also</h3>

<p>Other mega environments: 
<code><a href="#topic+gxeMegaEnv">gxeMegaEnv</a>()</code>,
<code><a href="#topic+plot.megaEnv">plot.megaEnv</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute mega environments for TDMaize.
geMegaEnv &lt;- gxeMegaEnv(TD = TDMaize, trait = "yld")

## Compute BLUPS and standard errors for those mega environments.
megaEnvPred &lt;- predict(geMegaEnv)
head(megaEnvPred$predictedValue)
head(megaEnvPred$standardError)

</code></pre>

<hr>
<h2 id='predict.varComp'>Predictions based on a fitted varComp model.</h2><span id='topic+predict.varComp'></span>

<h3>Description</h3>

<p>Predictions are made based on the fitted model in the varComp object.
These predictions can be at genotype level, at genotype x trial level or at
the level of genotype x nestingFactor. If the model was fitted with trial as
year x location then genotype x trial level becomes genotype x year x location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'varComp'
predict(object, ..., predictLevel = "genotype")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.varComp_+3A_object">object</code></td>
<td>
<p>An object of class varComp.</p>
</td></tr>
<tr><td><code id="predict.varComp_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="predict.varComp_+3A_predictlevel">predictLevel</code></td>
<td>
<p>A character string, the level at which prediction should
be made. Either &quot;genotype&quot; for prediction at genotype level, &quot;trial&quot; for
predictions at genotype x trial level, the variable used as nesting factor
for predictions at the level of genotype x nestingFactor level, or one or
more of the extra terms used in the model. E.g. c(&quot;region&quot;, &quot;year&quot;) for a
model fitted with <code>regionLocationYear = TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with predictions.
</p>


<h3>See Also</h3>

<p>Other Mixed model analysis: 
<code><a href="#topic+CRDR">CRDR</a>()</code>,
<code><a href="#topic+correlations">correlations</a>()</code>,
<code><a href="#topic+diagnostics">diagnostics</a>()</code>,
<code><a href="#topic+gxeVarComp">gxeVarComp</a>()</code>,
<code><a href="#topic+herit">herit</a>()</code>,
<code><a href="#topic+plot.varComp">plot.varComp</a>()</code>,
<code><a href="#topic+vc">vc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Fit a mixed model.
geVarComp &lt;- gxeVarComp(TD = TDMaize, trait = "yld")

## Predictions at genotype level.
predGeno &lt;- predict(geVarComp)
head(predGeno)

## Predictions at genotype x trial level.
predGenoTrial &lt;- predict(geVarComp, predictLevel = "trial")
head(predGenoTrial)

</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+report'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>statgenSTA</dt><dd><p><code><a href="statgenSTA.html#topic+report">report</a></code></p>
</dd>
</dl>

<hr>
<h2 id='report.AMMI'>Report method for class AMMI</h2><span id='topic+report.AMMI'></span>

<h3>Description</h3>

<p>A pdf report will be created containing a summary of an AMMI object.
Simultaneously the same report will be created as a tex file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AMMI'
report(x, ..., outfile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report.AMMI_+3A_x">x</code></td>
<td>
<p>An object of class AMMI.</p>
</td></tr>
<tr><td><code id="report.AMMI_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="report.AMMI_+3A_outfile">outfile</code></td>
<td>
<p>A character string, the name and location of the output .pdf
and .tex file for the report. If <code>NULL</code>, a report with a default name
will be created in the current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pdf and tex report.
</p>


<h3>See Also</h3>

<p>Other AMMI: 
<code><a href="#topic+fitted.AMMI">fitted.AMMI</a>()</code>,
<code><a href="#topic+gxeAmmi">gxeAmmi</a>()</code>,
<code><a href="#topic+plot.AMMI">plot.AMMI</a>()</code>,
<code><a href="#topic+residuals.AMMI">residuals.AMMI</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run AMMI analysis on TDMaize.
geAmmi &lt;- gxeAmmi(TD = TDMaize, trait = "yld")

## Create a pdf report summarizing the results.
report(geAmmi, outfile = tempfile(fileext = ".pdf"))


</code></pre>

<hr>
<h2 id='report.FW'>Report method for class FW</h2><span id='topic+report.FW'></span>

<h3>Description</h3>

<p>A pdf report will be created containing a summary of an FW object.
Simultaneously the same report will be created as a tex file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FW'
report(x, sortBy = c("sens", "genMean", "mse"), ..., outfile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report.FW_+3A_x">x</code></td>
<td>
<p>An object of class FW.</p>
</td></tr>
<tr><td><code id="report.FW_+3A_sortby">sortBy</code></td>
<td>
<p>A character string indicating by which variable the estimates
should be sorted. Either <code>sens</code>(itivity), <code>genMean</code> (genotypic
Mean) or <code>mse</code> (mean squared error).</p>
</td></tr>
<tr><td><code id="report.FW_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="report.FW_+3A_outfile">outfile</code></td>
<td>
<p>A character string, the name and location of the output .pdf
and .tex file for the report. If <code>NULL</code>, a report with a default name
will be created in the current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pdf and tex report.
</p>


<h3>See Also</h3>

<p>Other Finlay-Wilkinson: 
<code><a href="#topic+fitted.FW">fitted.FW</a>()</code>,
<code><a href="#topic+gxeFw">gxeFw</a>()</code>,
<code><a href="#topic+plot.FW">plot.FW</a>()</code>,
<code><a href="#topic+residuals.FW">residuals.FW</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run Finlay-Wilkinson analysis on TDMaize.
geFW &lt;- gxeFw(TDMaize, trait = "yld")


## Create a report summarizing the results.
report(geFW, outfile = tempfile(fileext = ".pdf"))


</code></pre>

<hr>
<h2 id='report.stability'>Report method for class stability</h2><span id='topic+report.stability'></span>

<h3>Description</h3>

<p>A pdf report will be created containing a summary of an object of class
stability. Simultaneously the same report will be created as a tex file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'stability'
report(x, ..., outfile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report.stability_+3A_x">x</code></td>
<td>
<p>An object of class stability.</p>
</td></tr>
<tr><td><code id="report.stability_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="report.stability_+3A_outfile">outfile</code></td>
<td>
<p>A character string, the name and location of the output .pdf
and .tex file for the report. If <code>NULL</code>, a report with a default name
will be created in the current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pdf and tex report.
</p>


<h3>See Also</h3>

<p>Other stability: 
<code><a href="#topic+gxeStability">gxeStability</a>()</code>,
<code><a href="#topic+plot.stability">plot.stability</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute three stability measures for TDMaize.
geStab &lt;- gxeStability(TD = TDMaize, trait = "yld")

## Create a .pdf report summarizing the stability measures.
report(geStab, outfile = tempfile(fileext = ".pdf"))


</code></pre>

<hr>
<h2 id='report.varCov'>Report method for class varCov</h2><span id='topic+report.varCov'></span>

<h3>Description</h3>

<p>A pdf report will be created containing a summary of an object of class
varCov. Simultaneously the same report will be created as a tex
file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'varCov'
report(x, ..., outfile = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report.varCov_+3A_x">x</code></td>
<td>
<p>An object of class varCov.</p>
</td></tr>
<tr><td><code id="report.varCov_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
<tr><td><code id="report.varCov_+3A_outfile">outfile</code></td>
<td>
<p>A character string, the name and location of the output .pdf
and .tex file for the report. If <code>NULL</code>, a report with a default name
will be created in the current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pdf and tex report.
</p>


<h3>See Also</h3>

<p>Other varCov: 
<code><a href="#topic+fitted.varCov">fitted.varCov</a>()</code>,
<code><a href="#topic+gxeVarCov">gxeVarCov</a>()</code>,
<code><a href="#topic+plot.varCov">plot.varCov</a>()</code>,
<code><a href="#topic+residuals.varCov">residuals.varCov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Select the best variance-covariance model using asreml for modeling.
if (requireNamespace("asreml", quietly = TRUE)) {
  geVarCov &lt;- gxeVarCov(TD = TDMaize, trait = "yld", engine = "asreml")

  ## Create a pdf report summarizing the results.
  report(geVarCov, outfile = tempfile(fileext = ".pdf"))
  }


</code></pre>

<hr>
<h2 id='residuals.AMMI'>Extract residuals.</h2><span id='topic+residuals.AMMI'></span>

<h3>Description</h3>

<p>Extract the residuals for the fitted AMMI model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AMMI'
residuals(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="residuals.AMMI_+3A_object">object</code></td>
<td>
<p>An object of class AMMI</p>
</td></tr>
<tr><td><code id="residuals.AMMI_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with residuals.
</p>


<h3>See Also</h3>

<p>Other AMMI: 
<code><a href="#topic+fitted.AMMI">fitted.AMMI</a>()</code>,
<code><a href="#topic+gxeAmmi">gxeAmmi</a>()</code>,
<code><a href="#topic+plot.AMMI">plot.AMMI</a>()</code>,
<code><a href="#topic+report.AMMI">report.AMMI</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run AMMI analysis on TDMaize.
geAmmi &lt;- gxeAmmi(TD = TDMaize, trait = "yld")

## Extract residuals.
residAmmi &lt;- residuals(geAmmi)
head(residAmmi)

</code></pre>

<hr>
<h2 id='residuals.FW'>Extract residuals.</h2><span id='topic+residuals.FW'></span>

<h3>Description</h3>

<p>Extract the residuals for a fitted Finlay-Wilkinson model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FW'
residuals(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="residuals.FW_+3A_object">object</code></td>
<td>
<p>An object of class FW</p>
</td></tr>
<tr><td><code id="residuals.FW_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with residuals.
</p>


<h3>See Also</h3>

<p>Other Finlay-Wilkinson: 
<code><a href="#topic+fitted.FW">fitted.FW</a>()</code>,
<code><a href="#topic+gxeFw">gxeFw</a>()</code>,
<code><a href="#topic+plot.FW">plot.FW</a>()</code>,
<code><a href="#topic+report.FW">report.FW</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Run Finlay-Wilkinson analysis.
geFW &lt;- gxeFw(TD = TDMaize, trait = "yld")

## Extract residuals.
residFW &lt;- residuals(geFW)
head(residFW)

</code></pre>

<hr>
<h2 id='residuals.varCov'>Extract residuals.</h2><span id='topic+residuals.varCov'></span>

<h3>Description</h3>

<p>Extract the residuals for the best model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'varCov'
residuals(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="residuals.varCov_+3A_object">object</code></td>
<td>
<p>An object of class varCov</p>
</td></tr>
<tr><td><code id="residuals.varCov_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with residuals.
</p>


<h3>See Also</h3>

<p>Other varCov: 
<code><a href="#topic+fitted.varCov">fitted.varCov</a>()</code>,
<code><a href="#topic+gxeVarCov">gxeVarCov</a>()</code>,
<code><a href="#topic+plot.varCov">plot.varCov</a>()</code>,
<code><a href="#topic+report.varCov">report.varCov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Select the best variance-covariance model using asreml for modeling.
if (requireNamespace("asreml", quietly = TRUE)) {
  geVarCov &lt;- gxeVarCov(TD = TDMaize, trait = "yld", engine = "asreml")

  ## Extract residuals from the model.
  residVarCov &lt;- residuals(geVarCov)
  head(residVarCov)
  }


</code></pre>

<hr>
<h2 id='stability'>S3 class stability</h2><span id='topic+stability'></span><span id='topic+createStability'></span>

<h3>Description</h3>

<p>Function for creating objects of S3 class stability.<br />
<code><a href="base.html#topic+print">print</a></code>, <code><a href="base.html#topic+summary">summary</a></code>, <code><a href="graphics.html#topic+plot">plot</a></code> and
<code><a href="#topic+report">report</a></code> methods are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createStability(superiority = NULL, static = NULL, wricke = NULL, TD, trait)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stability_+3A_superiority">superiority</code></td>
<td>
<p>A data.frame containing values for the
cultivar-superiority measure of Lin and Binns.</p>
</td></tr>
<tr><td><code id="stability_+3A_static">static</code></td>
<td>
<p>A data.frame containing values for Shukla's stability variance.</p>
</td></tr>
<tr><td><code id="stability_+3A_wricke">wricke</code></td>
<td>
<p>A data.frame containing values for Wricke's ecovalence.</p>
</td></tr>
<tr><td><code id="stability_+3A_trait">trait</code></td>
<td>
<p>A character string indicating the trait that has been analyzed.</p>
</td></tr>
</table>

<hr>
<h2 id='statgenGxE-package'>statgenGxE: Genotype by Environment (GxE) Analysis</h2><span id='topic+statgenGxE'></span><span id='topic+statgenGxE-package'></span>

<h3>Description</h3>

<p>Analysis of multi environment data of plant breeding experiments following the analyses described in Malosetti, Ribaut, and van Eeuwijk (2013), <a href="https://doi.org/10.3389/fphys.2013.00044">doi:10.3389/fphys.2013.00044</a>. One of a series of statistical genetic packages for streamlining the analysis of typical plant breeding experiments developed by Biometris. Some functions have been created to be used in conjunction with the R package 'asreml' for the 'ASReml' software, which can be obtained upon purchase from 'VSN' international (<a href="https://vsni.co.uk/software/asreml-r">https://vsni.co.uk/software/asreml-r</a>).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Bart-Jan van Rossum <a href="mailto:bart-jan.vanrossum@wur.nl">bart-jan.vanrossum@wur.nl</a>
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Fred van Eeuwijk (<a href="https://orcid.org/0000-0003-3672-2921">ORCID</a>) [contributor]
</p>
</li>
<li><p> Martin Boer [contributor]
</p>
</li>
<li><p> Marcos Malosetti (<a href="https://orcid.org/0000-0002-8150-1397">ORCID</a>) [contributor]
</p>
</li>
<li><p> Daniela Bustos-Korts (<a href="https://orcid.org/0000-0003-3827-6726">ORCID</a>) [contributor]
</p>
</li>
<li><p> Emilie J Millet (<a href="https://orcid.org/0000-0002-2913-4892">ORCID</a>) [contributor]
</p>
</li>
<li><p> Joao Paulo (<a href="https://orcid.org/0000-0002-4180-0763">ORCID</a>) [contributor]
</p>
</li>
<li><p> Maikel Verouden (<a href="https://orcid.org/0000-0002-4893-3323">ORCID</a>) [contributor]
</p>
</li>
<li><p> Willem Kruijer (<a href="https://orcid.org/0000-0001-7179-1733">ORCID</a>) [contributor]
</p>
</li>
<li><p> Ron Wehrens (<a href="https://orcid.org/0000-0002-8798-5599">ORCID</a>) [contributor]
</p>
</li>
<li><p> Choazhi Zheng (<a href="https://orcid.org/0000-0001-6030-3933">ORCID</a>) [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://biometris.github.io/statgenGxE/index.html">https://biometris.github.io/statgenGxE/index.html</a>
</p>
</li>
<li> <p><a href="https://github.com/Biometris/statgenGxE/">https://github.com/Biometris/statgenGxE/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Biometris/statgenGxE/issues">https://github.com/Biometris/statgenGxE/issues</a>
</p>
</li></ul>


<hr>
<h2 id='TDMaize'>Field data for a maize experiment in Tlaltizapan, Mexico</h2><span id='topic+TDMaize'></span>

<h3>Description</h3>

<p>A dataset converted into a TD object containing data corresponding to an
F2 maize reference population from CIMMYT maize drought breeding program,
which was derived from the cross of a drought-tolerant line (P1) with a
drought susceptible line (P2) as described in detail by Ribaut et al. (1996,
1997).<br />
DNA from 211 F2 plants was extracted to produce information for 132
co-dominant markers on 10 linkage groups. Phenotypic evaluations were
performed on 211 F2:3 families, each one derived from an original F2 plant.
The families were evaluated under different water and nitrogen regimes
during 1992, 1994 and 1996. In the winter of 1992 three water regimes were
imposed on the trials: well watered (NS), intermediate stress (IS) and severe
stress (SS). In the winter of 1994, only the IS and SS trials were available.
Nitrogen availability varied in the 1996 trials, with two low nitrogen
treatments (LN, in winter and summer) and one high-nitrogen treatment
(HN in summer). In each of the trials, five traits were evaluated but only
grain yield is included in the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TDMaize
</code></pre>


<h3>Format</h3>

<p>A TD object, a list containing 8 data.frames, each with the following
columns:
</p>

<dl>
<dt>trial</dt><dd><p>trial, a combination of water regime, year and nitrogen
treatment</p>
</dd>
<dt>genotype</dt><dd><p>genotype</p>
</dd>
<dt>yld</dt><dd><p>grain yield in tons</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://link.springer.com/article/10.1007/BF00221905">https://link.springer.com/article/10.1007/BF00221905</a>
</p>


<h3>References</h3>

<p>Ribaut JM, Hoisington DA, Deutsch JA, Jiang C, Gonzalez de Leon D
(1996) Identification of quantitative trait loci under drought conditions in
tropical maize.1. Flowering parameters and the anthesis-silking interval.
Theor Appl Genet 92:905–914
</p>
<p>Ribaut JM, Jiang C, Gonzalez de Leon D, Edmeades GO, Hoisington
DA (1997) Identification of quantitative trait loci under drought conditions
in tropical maize.2. Yield components and marker-assisted selection
strategies. Theor Appl Genet 94:887–896
</p>

<hr>
<h2 id='varCov'>S3 class varCov</h2><span id='topic+varCov'></span><span id='topic+createVarCov'></span>

<h3>Description</h3>

<p>Function for creating objects of S3 class varCov.<br />
<code><a href="base.html#topic+print">print</a></code>, <code><a href="base.html#topic+summary">summary</a></code>, <code><a href="graphics.html#topic+plot">plot</a></code> and
<code><a href="#topic+report">report</a></code> methods are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createVarCov(STA, choice, summary, vcov, criterion, engine, dat, trait)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varCov_+3A_sta">STA</code></td>
<td>
<p>An object of class STA, the best fitted model.</p>
</td></tr>
<tr><td><code id="varCov_+3A_choice">choice</code></td>
<td>
<p>A character string indicating the best fitted model.</p>
</td></tr>
<tr><td><code id="varCov_+3A_summary">summary</code></td>
<td>
<p>A data.frame with a summary of the fitted models.</p>
</td></tr>
<tr><td><code id="varCov_+3A_vcov">vcov</code></td>
<td>
<p>The covariance matrix of the best fitted model.</p>
</td></tr>
<tr><td><code id="varCov_+3A_criterion">criterion</code></td>
<td>
<p>A character string indicating the goodness-of-fit criterion
used for determinening the best model.</p>
</td></tr>
<tr><td><code id="varCov_+3A_engine">engine</code></td>
<td>
<p>A character string containing the engine used for the analysis.</p>
</td></tr>
</table>

<hr>
<h2 id='vc'>Extract variance components</h2><span id='topic+vc'></span>

<h3>Description</h3>

<p>Extract variance components from an object of class varComp.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vc(varComp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vc_+3A_varcomp">varComp</code></td>
<td>
<p>An object of class varComp.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with variance components and standard errors for
the random components in the fitted model.
</p>


<h3>See Also</h3>

<p>Other Mixed model analysis: 
<code><a href="#topic+CRDR">CRDR</a>()</code>,
<code><a href="#topic+correlations">correlations</a>()</code>,
<code><a href="#topic+diagnostics">diagnostics</a>()</code>,
<code><a href="#topic+gxeVarComp">gxeVarComp</a>()</code>,
<code><a href="#topic+herit">herit</a>()</code>,
<code><a href="#topic+plot.varComp">plot.varComp</a>()</code>,
<code><a href="#topic+predict.varComp">predict.varComp</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Fit a mixed model.
geVarComp &lt;- gxeVarComp(TD = TDMaize, trait = "yld")

## Extract variance components.
vc(geVarComp)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
