<!DOCTYPE html><html><head><title>Help for package STREAK</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {STREAK}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#STREAK-package'><p>STREAK: Receptor Abundance Estimation using Feature Selection and Gene Set Scoring</p></a></li>
<li><a href='#receptorAbundanceEstimation'><p>Receptor abundance estimation for single cell RNA-sequencing (scRNA-seq) data using gene set scoring and thresholding.</p></a></li>
<li><a href='#receptorGeneSetConstruction'><p>Gene sets weights membership matrix construction for receptor abundance estimation.</p></a></li>
<li><a href='#target.malt.rna.mat'><p>Single cell RNA-sequencing (scRNA-seq) target subset of the 10X Genomics MALT counts.</p></a></li>
<li><a href='#train.malt.adt.mat'><p>Cellular Indexing of Transcriptomes and Epitopes by Sequencing (CITE-seq) training subset of the 10X Genomics MALT counts.</p></a></li>
<li><a href='#train.malt.rna.mat'><p>Single cell RNA-sequencing (scRNA-seq) training subset of the 10X Genomics MALT counts.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Receptor Abundance Estimation using Feature Selection and Gene
Set Scoring</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Azka Javaid &lt;azka.javaid.gr@dartmouth.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs receptor abundance estimation for single cell RNA-sequencing data using a supervised feature selection mechanism and a thresholded gene set scoring procedure. Seurat's normalization method is described in: Hao et al., (2021) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2021.04.048">doi:10.1016/j.cell.2021.04.048</a>&gt;, Stuart et al., (2019) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2019.05.031">doi:10.1016/j.cell.2019.05.031</a>&gt;, Butler et al., (2018) &lt;<a href="https://doi.org/10.1038%2Fnbt.4096">doi:10.1038/nbt.4096</a>&gt; and Satija et al., (2015) &lt;<a href="https://doi.org/10.1038%2Fnbt.3192">doi:10.1038/nbt.3192</a>&gt;. Method for reduced rank reconstruction and rank-k selection is detailed in: Javaid et al., (2022) &lt;<a href="https://doi.org/10.1101%2F2022.10.08.511197">doi:10.1101/2022.10.08.511197</a>&gt;. Gene set scoring procedure is described in: Frost et al., (2020) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkaa582">doi:10.1093/nar/gkaa582</a>&gt;. Clustering method is outlined in: Song et al., (2020) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtaa613">doi:10.1093/bioinformatics/btaa613</a>&gt; and Wang et al., (2011) &lt;<a href="https://doi.org/10.32614%2FRJ-2011-015">doi:10.32614/RJ-2011-015</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>Imports:</td>
<td>Ckmeans.1d.dp, Matrix, Seurat, SPECK, stats, VAM</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-17 20:31:39 UTC; azkajavaid</td>
</tr>
<tr>
<td>Author:</td>
<td>H. Robert Frost [aut],
  Azka Javaid [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-17 21:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='STREAK-package'>STREAK: Receptor Abundance Estimation using Feature Selection and Gene Set Scoring</h2><span id='topic+STREAK'></span><span id='topic+STREAK-package'></span>

<h3>Description</h3>

<p>Performs receptor abundance estimation for single cell RNA-sequencing data using a supervised feature selection mechanism and a thresholded gene set scoring procedure. Seurat's normalization method is described in: Hao et al., (2021) <a href="https://doi.org/10.1016/j.cell.2021.04.048">doi:10.1016/j.cell.2021.04.048</a>, Stuart et al., (2019) <a href="https://doi.org/10.1016/j.cell.2019.05.031">doi:10.1016/j.cell.2019.05.031</a>, Butler et al., (2018) <a href="https://doi.org/10.1038/nbt.4096">doi:10.1038/nbt.4096</a> and Satija et al., (2015) <a href="https://doi.org/10.1038/nbt.3192">doi:10.1038/nbt.3192</a>. Method for reduced rank reconstruction and rank-k selection is detailed in: Javaid et al., (2022) <a href="https://doi.org/10.1101/2022.10.08.511197">doi:10.1101/2022.10.08.511197</a>. Gene set scoring procedure is described in: Frost et al., (2020) <a href="https://doi.org/10.1093/nar/gkaa582">doi:10.1093/nar/gkaa582</a>. Clustering method is outlined in: Song et al., (2020) <a href="https://doi.org/10.1093/bioinformatics/btaa613">doi:10.1093/bioinformatics/btaa613</a> and Wang et al., (2011) <a href="https://doi.org/10.32614/RJ-2011-015">doi:10.32614/RJ-2011-015</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Azka Javaid <a href="mailto:azka.javaid.gr@dartmouth.edu">azka.javaid.gr@dartmouth.edu</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> H. Robert Frost <a href="mailto:hildreth.r.frost@dartmouth.edu">hildreth.r.frost@dartmouth.edu</a>
</p>
</li></ul>


<hr>
<h2 id='receptorAbundanceEstimation'>Receptor abundance estimation for single cell RNA-sequencing (scRNA-seq) data using gene set scoring and thresholding.</h2><span id='topic+receptorAbundanceEstimation'></span>

<h3>Description</h3>

<p>Performs receptor abundance estimation for <code class="reqn">m x n</code> scRNA-seq target data using gene set scoring and thresholding. scRNA-seq target counts are normalized and
reduced rank reconstructed (RRR) using the <code><a href="SPECK.html#topic+randomizedRRR">SPECK::randomizedRRR()</a></code> function. Gene set scoring is next performed leveraging expression from
the top most weighted genes based on the gene sets weights membership matrix with the <code><a href="VAM.html#topic+vam">VAM::vam()</a></code> function. The resulting cell-specific gene set scores
are then thresholded utilizing the <code><a href="Ckmeans.1d.dp.html#topic+Ckmeans.1d.dp">Ckmeans.1d.dp::Ckmeans.1d.dp()</a></code> function. Note that this function only performs normalization
and does not perform any quality control (QC) checks on the inputted target scRNA-seq counts matrix. Any QC needed can be performed on the target matrix before passing it
as an input to the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>receptorAbundanceEstimation(
  target.rnaseq,
  receptor.geneset.matrix,
  num.genes = 10,
  rank.range.end = 100,
  min.consec.diff = 0.01,
  rep.consec.diff = 2,
  manual.rank = NULL,
  seed.rsvd = 1,
  max.num.clusters = 4,
  seed.ckmeans = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="receptorAbundanceEstimation_+3A_target.rnaseq">target.rnaseq</code></td>
<td>
<p><code class="reqn">m x n</code> scRNA-seq counts matrix for <code class="reqn">m</code> cells and <code class="reqn">n</code> genes.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_receptor.geneset.matrix">receptor.geneset.matrix</code></td>
<td>
<p><code class="reqn">n x h</code> Gene sets weights membership matrix.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_num.genes">num.genes</code></td>
<td>
<p>Number of top most weighted genes for subsequent gene set scoring and thresholding.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_rank.range.end">rank.range.end</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_min.consec.diff">min.consec.diff</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_rep.consec.diff">rep.consec.diff</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_manual.rank">manual.rank</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_seed.rsvd">seed.rsvd</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_max.num.clusters">max.num.clusters</code></td>
<td>
<p>See documentation for the <code>ckmeansThreshold</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorAbundanceEstimation_+3A_seed.ckmeans">seed.ckmeans</code></td>
<td>
<p>See documentation for the <code>ckmeansThreshold</code> function from the <code>SPECK</code> package.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code>receptor.abundance.estimates</code> - A <code class="reqn">m x h</code> matrix consisting of abundance estimates for <code class="reqn">m</code> cells in <code class="reqn">h</code> receptors.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data("train.malt.rna.mat")
data("train.malt.adt.mat")
receptor.geneset.matrix.out &lt;- receptorGeneSetConstruction(train.rnaseq =
                                         train.malt.rna.mat[1:100,1:80],
                                         train.citeseq =
                                         train.malt.adt.mat[1:100,1:2],
                                         rank.range.end = 70,
                                         min.consec.diff = 0.01,
                                         rep.consec.diff = 2,
                                         manual.rank = NULL,
                                         seed.rsvd = 1)
dim(receptor.geneset.matrix.out)
head(receptor.geneset.matrix.out)
data("target.malt.rna.mat")
receptor.abundance.estimates.out &lt;- receptorAbundanceEstimation(target.rnaseq =
                                    target.malt.rna.mat[1:200,1:80],
                                    receptor.geneset.matrix =
                                    receptor.geneset.matrix.out,
                                    num.genes = 10, rank.range.end = 70,
                                    min.consec.diff = 0.01,
                                    rep.consec.diff = 2,
                                    manual.rank = NULL, seed.rsvd = 1,
                                    max.num.clusters = 4, seed.ckmeans = 2)
dim(receptor.abundance.estimates.out)
head(receptor.abundance.estimates.out)
</code></pre>

<hr>
<h2 id='receptorGeneSetConstruction'>Gene sets weights membership matrix construction for receptor abundance estimation.</h2><span id='topic+receptorGeneSetConstruction'></span>

<h3>Description</h3>

<p>Computes <code class="reqn">n x h</code> gene sets weights membership matrix using associations learned between log-normalized and reduced rank reconstructed (RRR)
<code class="reqn">m x n</code> scRNA-seq training data and <code class="reqn">m x h</code> CITE-seq ADT training counts normalized using the centered log ratio (CLR) transformation. scRNA-seq counts are normalized and
RRR using the <code><a href="SPECK.html#topic+randomizedRRR">SPECK::randomizedRRR()</a></code> function while CITE-seq counts are normalized using the
<code><a href="Seurat.html#topic+NormalizeData">Seurat::NormalizeData()</a></code> function with the <code>normalization.method</code> parameter set to <code>CLR</code>. Spearman rank correlations are computed between the
normalized CITE-seq data and the normalized and RRR scRNA-seq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>receptorGeneSetConstruction(
  train.rnaseq,
  train.citeseq,
  rank.range.end = 100,
  min.consec.diff = 0.01,
  rep.consec.diff = 2,
  manual.rank = NULL,
  seed.rsvd = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="receptorGeneSetConstruction_+3A_train.rnaseq">train.rnaseq</code></td>
<td>
<p><code class="reqn">m x n</code> scRNA-seq counts matrix for <code class="reqn">m</code> cells
and <code class="reqn">n</code> genes.</p>
</td></tr>
<tr><td><code id="receptorGeneSetConstruction_+3A_train.citeseq">train.citeseq</code></td>
<td>
<p><code class="reqn">m x h</code> CITE-seq ADT counts matrix for <code class="reqn">m</code> cells (same cells as the <code>train.rnaseq matrix</code>) and <code class="reqn">h</code> cell-surface proteins.</p>
</td></tr>
<tr><td><code id="receptorGeneSetConstruction_+3A_rank.range.end">rank.range.end</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorGeneSetConstruction_+3A_min.consec.diff">min.consec.diff</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorGeneSetConstruction_+3A_rep.consec.diff">rep.consec.diff</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorGeneSetConstruction_+3A_manual.rank">manual.rank</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
<tr><td><code id="receptorGeneSetConstruction_+3A_seed.rsvd">seed.rsvd</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code>receptor.geneset.matrix</code> - A <code class="reqn">n x h</code> gene sets weights membership matrix where a column <code class="reqn">i</code> from <code class="reqn">h</code> corresponds to the weights for <code class="reqn">n</code> genes from the scRNA-seq matrix trained against the corresponding CITE-seq ADT transcript <code class="reqn">h</code>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data("train.malt.rna.mat")
data("train.malt.adt.mat")
receptor.geneset.matrix.out &lt;- receptorGeneSetConstruction(train.rnaseq =
                                         train.malt.rna.mat[1:100,1:80],
                                         train.citeseq =
                                         train.malt.adt.mat[1:100,1:2],
                                         rank.range.end = 70,
                                         min.consec.diff = 0.01,
                                         rep.consec.diff = 2,
                                         manual.rank = NULL, seed.rsvd = 1)
dim(receptor.geneset.matrix.out)
head(receptor.geneset.matrix.out)
</code></pre>

<hr>
<h2 id='target.malt.rna.mat'>Single cell RNA-sequencing (scRNA-seq) target subset of the 10X Genomics MALT counts.</h2><span id='topic+target.malt.rna.mat'></span>

<h3>Description</h3>

<p>A random subset of joint scRNA-seq/CITE-seq 10X Genomics human extranodal marginal zone B-cell tumor/mucosa-associated lymphoid tissue (MALT) target data. See the <code>dataProcessing.R</code> file from <code>data-raw</code> folder for code to recreate data subset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>target.malt.rna.mat
</code></pre>


<h3>Format</h3>

<p>A scRNA-seq counts matrix of <code>dgCMatrix-class</code> from the <code>Matrix</code> package with 4000 cells and 33538 genes.
</p>


<h3>Source</h3>

<p>&lt;https://www.10xgenomics.com/resources/datasets/10-k-cells-from-a-malt-tumor-gene-expression-and-cell-surface-protein-3-standard-3-0-0&gt;
</p>

<hr>
<h2 id='train.malt.adt.mat'>Cellular Indexing of Transcriptomes and Epitopes by Sequencing (CITE-seq) training subset of the 10X Genomics MALT counts.</h2><span id='topic+train.malt.adt.mat'></span>

<h3>Description</h3>

<p>A random subset of joint scRNA-seq/CITE-seq 10X Genomics human extranodal marginal zone B-cell tumor/mucosa-associated lymphoid tissue (MALT) training data. See the <code>dataProcessing.R</code> file from <code>data-raw</code> folder for code to recreate data subset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>train.malt.adt.mat
</code></pre>


<h3>Format</h3>

<p>A CITE-seq counts matrix of <code>dgeMatrix-class</code> from the <code>Matrix</code> package with 1000 cells and 17 genes.
</p>


<h3>Source</h3>

<p>&lt;https://www.10xgenomics.com/resources/datasets/10-k-cells-from-a-malt-tumor-gene-expression-and-cell-surface-protein-3-standard-3-0-0&gt;
</p>

<hr>
<h2 id='train.malt.rna.mat'>Single cell RNA-sequencing (scRNA-seq) training subset of the 10X Genomics MALT counts.</h2><span id='topic+train.malt.rna.mat'></span>

<h3>Description</h3>

<p>A random subset of joint scRNA-seq/CITE-seq 10X Genomics human extranodal marginal zone B-cell tumor/mucosa-associated lymphoid tissue (MALT) training data. See the <code>dataProcessing.R</code> file from <code>data-raw</code> folder for code to recreate data subset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>train.malt.rna.mat
</code></pre>


<h3>Format</h3>

<p>A scRNA-seq counts matrix of <code>dgCMatrix-class</code> from the <code>Matrix</code> package with 1000 cells and 33538 genes.
</p>


<h3>Source</h3>

<p>&lt;https://www.10xgenomics.com/resources/datasets/10-k-cells-from-a-malt-tumor-gene-expression-and-cell-surface-protein-3-standard-3-0-0&gt;
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
