<!DOCTYPE html><html><head><title>Help for package strvalidator</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {strvalidator}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#addColor'><p>Add Color Information.</p></a></li>
<li><a href='#addData'><p>Adds New Data Columns to a Data Frame</p></a></li>
<li><a href='#addData_gui'><p>Add Data</p></a></li>
<li><a href='#addDye_gui'><p>Add Dye Information</p></a></li>
<li><a href='#addMarker'><p>Add Missing Markers.</p></a></li>
<li><a href='#addMarker_gui'><p>Add Missing Markers</p></a></li>
<li><a href='#addOrder'><p>Add Marker Order.</p></a></li>
<li><a href='#addSize'><p>Add Size Information.</p></a></li>
<li><a href='#addSize_gui'><p>Add Size Information</p></a></li>
<li><a href='#auditTrail'><p>Log Audit Trail.</p></a></li>
<li><a href='#calculateAllele'><p>Calculate Allele</p></a></li>
<li><a href='#calculateAllele_gui'><p>Calculate Allele</p></a></li>
<li><a href='#calculateAllT'><p>Calculate Stochastic Thresholds</p></a></li>
<li><a href='#calculateAllT_gui'><p>Calculate Stochastic Thresholds</p></a></li>
<li><a href='#calculateAT'><p>Calculate Analytical Threshold</p></a></li>
<li><a href='#calculateAT_gui'><p>Calculate Analytical Threshold</p></a></li>
<li><a href='#calculateAT6'><p>Calculate Analytical Threshold</p></a></li>
<li><a href='#calculateAT6_gui'><p>Calculate Analytical Threshold</p></a></li>
<li><a href='#calculateCapillary'><p>Calculate Capillary Balance</p></a></li>
<li><a href='#calculateCapillary_gui'><p>Calculate Capillary Balance</p></a></li>
<li><a href='#calculateConcordance'><p>Calculate Concordance.</p></a></li>
<li><a href='#calculateConcordance_gui'><p>Calculate Concordance</p></a></li>
<li><a href='#calculateCopies'><p>Calculate Allele Copies</p></a></li>
<li><a href='#calculateCopies_gui'><p>Calculate Allele Copies</p></a></li>
<li><a href='#calculateDropout'><p>Calculate Drop-out Events</p></a></li>
<li><a href='#calculateDropout_gui'><p>Calculate Dropout Events</p></a></li>
<li><a href='#calculateHb'><p>Calculate Heterozygote Balance</p></a></li>
<li><a href='#calculateHb_gui'><p>Calculate Heterozygote Balance</p></a></li>
<li><a href='#calculateHeight'><p>Calculate Peak Height.</p></a></li>
<li><a href='#calculateHeight_gui'><p>Calculate Peak Height</p></a></li>
<li><a href='#calculateLb'><p>Calculate Inter-locus Balance</p></a></li>
<li><a href='#calculateLb_gui'><p>Calculate Locus Balance</p></a></li>
<li><a href='#calculateMixture'><p>Calculate Mixture.</p></a></li>
<li><a href='#calculateMixture_gui'><p>Calculate Mixture</p></a></li>
<li><a href='#calculateOL'><p>Analyze Off-ladder Alleles</p></a></li>
<li><a href='#calculateOL_gui'><p>Analyze Off-ladder Alleles</p></a></li>
<li><a href='#calculateOverlap'><p>Calculate Bins Overlap</p></a></li>
<li><a href='#calculateOverlap_gui'><p>Calculate Bins Overlap</p></a></li>
<li><a href='#calculatePeaks'><p>Calculate Peaks</p></a></li>
<li><a href='#calculatePeaks_gui'><p>Calculate Peaks</p></a></li>
<li><a href='#calculatePullup'><p>Calculate Spectral Pull-up</p></a></li>
<li><a href='#calculatePullup_gui'><p>Calculate Spectral Pull-up</p></a></li>
<li><a href='#calculateRatio'><p>Calculate Ratio</p></a></li>
<li><a href='#calculateRatio_gui'><p>Calculate Ratio</p></a></li>
<li><a href='#calculateResultType'><p>Calculate Result Type</p></a></li>
<li><a href='#calculateResultType_gui'><p>Calculate Result Type</p></a></li>
<li><a href='#calculateSlope'><p>Calculate Profile Slope.</p></a></li>
<li><a href='#calculateSlope_gui'><p>Calculate Profile Slope</p></a></li>
<li><a href='#calculateSpike'><p>Detect Spike</p></a></li>
<li><a href='#calculateSpike_gui'><p>Detect Spike</p></a></li>
<li><a href='#calculateStatistics'><p>Summary Statistics</p></a></li>
<li><a href='#calculateStatistics_gui'><p>Calculate Statistics</p></a></li>
<li><a href='#calculateStutter'><p>Calculate Stutter</p></a></li>
<li><a href='#calculateStutter_gui'><p>Calculate Stutter</p></a></li>
<li><a href='#calculateT'><p>Calculate Stochastic Threshold</p></a></li>
<li><a href='#checkDataset'><p>Check Dataset</p></a></li>
<li><a href='#checkSubset'><p>Check Subset</p></a></li>
<li><a href='#checkSubset_gui'><p>Check Subset</p></a></li>
<li><a href='#colConvert'><p>Convert Columns</p></a></li>
<li><a href='#colNames'><p>Column Names</p></a></li>
<li><a href='#columns'><p>Column Actions</p></a></li>
<li><a href='#columns_gui'><p>Column Actions</p></a></li>
<li><a href='#combine_gui'><p>Combine Datasets</p></a></li>
<li><a href='#combineBinsAndPanels'><p>Combine Bins And Panels Files.</p></a></li>
<li><a href='#cropData_gui'><p>Crop Or Replace</p></a></li>
<li><a href='#detectKit'><p>Detect Kit</p></a></li>
<li><a href='#editData_gui'><p>Edit or View Data Frames</p></a></li>
<li><a href='#export'><p>Export</p></a></li>
<li><a href='#export_gui'><p>Export</p></a></li>
<li><a href='#filterProfile'><p>Filter Profile</p></a></li>
<li><a href='#filterProfile_gui'><p>Filter Profile</p></a></li>
<li><a href='#generateEPG'><p>Generate EPG</p></a></li>
<li><a href='#generateEPG_gui'><p>Generate EPG</p></a></li>
<li><a href='#getDb'><p>Get Allele Frequency Database</p></a></li>
<li><a href='#getKit'><p>Get Kit</p></a></li>
<li><a href='#getSetting'><p>Get Settings.</p></a></li>
<li><a href='#getStrings'><p>Get Language Strings</p></a></li>
<li><a href='#ggsave_gui'><p>Save Image</p></a></li>
<li><a href='#guessProfile'><p>Guess Profile</p></a></li>
<li><a href='#guessProfile_gui'><p>Guess Profile</p></a></li>
<li><a href='#heightToPeak'><p>Height To Peak.</p></a></li>
<li><a href='#import'><p>Import Data</p></a></li>
<li><a href='#import_gui'><p>Import Data</p></a></li>
<li><a href='#listObjects'><p>List Objects</p></a></li>
<li><a href='#makeKit_gui'><p>Make Kit</p></a></li>
<li><a href='#maskAT'><p>Mask And Prepare Data To Analyze Analytical Threshold</p></a></li>
<li><a href='#modelDropout_gui'><p>Model And Plot Drop-out Events</p></a></li>
<li><a href='#plotAT_gui'><p>Plot Analytical Threshold</p></a></li>
<li><a href='#plotBalance_gui'><p>Plot Balance</p></a></li>
<li><a href='#plotCapillary_gui'><p>Plot Capillary Balance</p></a></li>
<li><a href='#plotContamination_gui'><p>Plot Contamination</p></a></li>
<li><a href='#plotDistribution_gui'><p>Plot Distribution</p></a></li>
<li><a href='#plotDropout_gui'><p>Plot Drop-out Events</p></a></li>
<li><a href='#plotEPG2'><p>plotEPG2</p></a></li>
<li><a href='#plotEPG2_gui'><p>Plot EPG</p></a></li>
<li><a href='#plotGroups_gui'><p>Plot Empirical Cumulative Distributions</p></a></li>
<li><a href='#plotKit_gui'><p>Plot Kit Marker Ranges</p></a></li>
<li><a href='#plotPeaks_gui'><p>Plot Peaks</p></a></li>
<li><a href='#plotPrecision_gui'><p>Plot Precision</p></a></li>
<li><a href='#plotPullup_gui'><p>Plot Pull-up</p></a></li>
<li><a href='#plotRatio_gui'><p>Plot Ratio</p></a></li>
<li><a href='#plotResultType_gui'><p>Plot Result Type</p></a></li>
<li><a href='#plotSlope_gui'><p>Plot Profile Slope</p></a></li>
<li><a href='#plotStutter_gui'><p>Plot Stutter</p></a></li>
<li><a href='#readBinsFile'><p>Read Bins file</p></a></li>
<li><a href='#readPanelsFile'><p>Read Panels File</p></a></li>
<li><a href='#ref1'><p>ESX17 Positive Control Profile</p></a></li>
<li><a href='#ref11'><p>ESX17 Positive Control Profile</p></a></li>
<li><a href='#ref2'><p>SGMPlus example data</p></a></li>
<li><a href='#ref3'><p>ESX17 example data for dropout analysis.</p></a></li>
<li><a href='#ref4'><p>ESX17 example data for dropout analysis.</p></a></li>
<li><a href='#ref51'><p>ESX17 example data for mixture analysis.</p></a></li>
<li><a href='#ref52'><p>ESX17 example data for mixture analysis.</p></a></li>
<li><a href='#ref61'><p>Fusion example data for dropout analysis.</p></a></li>
<li><a href='#ref62'><p>Fusion example data for dropout analysis.</p></a></li>
<li><a href='#ref7'><p>ESSplex SE QS example data for inhibition analysis.</p></a></li>
<li><a href='#removeArtefact'><p>Remove Artefacts</p></a></li>
<li><a href='#removeArtefact_gui'><p>Remove Artefact</p></a></li>
<li><a href='#removeSpike'><p>Remove Spikes</p></a></li>
<li><a href='#removeSpike_gui'><p>Remove Spike</p></a></li>
<li><a href='#sample_tableToList'><p>sample_tableToList</p></a></li>
<li><a href='#saveObject'><p>Save Object</p></a></li>
<li><a href='#scrambleAlleles'><p>Scramble Alleles</p></a></li>
<li><a href='#set1'><p>Typing data in 'GeneMapper' format</p></a></li>
<li><a href='#set2'><p>SGMPlus example data</p></a></li>
<li><a href='#set3'><p>ESX17 example data for dropout analysis.</p></a></li>
<li><a href='#set4'><p>ESX17 example data for dropout analysis.</p></a></li>
<li><a href='#set5'><p>ESX17 example data for mixture analysis.</p></a></li>
<li><a href='#set6'><p>Fusion example data for dropout analysis.</p></a></li>
<li><a href='#set7'><p>ESSplex SE QS example data for inhibition analysis.</p></a></li>
<li><a href='#slim'><p>Slim Data Frames</p></a></li>
<li><a href='#slim_gui'><p>Slim Data Frames</p></a></li>
<li><a href='#sortMarker'><p>Sort Markers</p></a></li>
<li><a href='#strvalidator'><p>Graphical User Interface For The STR-validator Package</p></a></li>
<li><a href='#strvalidator-package'><p>Process Control and Internal Validation of Forensic STR Kits</p></a></li>
<li><a href='#trim'><p>Trim Data</p></a></li>
<li><a href='#trim_gui'><p>Trim Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Process Control and Validation of Forensic STR Kits</td>
</tr>
<tr>
<td>Version:</td>
<td>2.4.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Oskar Hansson</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Oskar Hansson &lt;oskhan@ous-hf.no&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://sites.google.com/site/forensicapps/strvalidator">https://sites.google.com/site/forensicapps/strvalidator</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/OskarHansson/strvalidator/issues">https://github.com/OskarHansson/strvalidator/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 2.0.0), gWidgets2, gWidgets2tcltk (&gt; 1.0.6),
gridExtra, grid, gtable, plyr, scales, data.table, DT, dplyr,
plotly, grDevices, graphics, stats, utils, MASS</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ResourceSelection, testthat</td>
</tr>
<tr>
<td>Description:</td>
<td>An open source platform for validation and process control.
    Tools to analyze data from internal validation of forensic short tandem
    repeat (STR) kits are provided. The tools are developed to provide
    the necessary data to conform with guidelines for internal validation
    issued by the European Network of Forensic Science Institutes (ENFSI)
    DNA Working Group, and the Scientific Working Group on DNA Analysis Methods
    (SWGDAM). A front-end graphical user interface is provided.
    More information about each function can be found in the
    respective help documentation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-16 13:36:31 UTC; oskar</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-16 14:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='addColor'>Add Color Information.</h2><span id='topic+addColor'></span>

<h3>Description</h3>

<p>Add color information 'Color', 'Dye' or 'R Color'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addColor(
  data,
  kit = NA,
  have = NA,
  need = NA,
  overwrite = FALSE,
  ignore.case = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addColor_+3A_data">data</code></td>
<td>
<p>data frame or vector.</p>
</td></tr>
<tr><td><code id="addColor_+3A_kit">kit</code></td>
<td>
<p>string representing the forensic STR kit used.
Default is NA, in which case 'have' must contain a valid column.</p>
</td></tr>
<tr><td><code id="addColor_+3A_have">have</code></td>
<td>
<p>character string to specify color column to be matched.
Default is NA, in which case color information is derived from 'kit' and added
to a column named 'Color'.
If 'data' is a vector 'have' must be a single string.</p>
</td></tr>
<tr><td><code id="addColor_+3A_need">need</code></td>
<td>
<p>character string or string vector to specify color columns to be added.
Default is NA, in which case all columns will be added.
If 'data' is a vector 'need' must be a single string.</p>
</td></tr>
<tr><td><code id="addColor_+3A_overwrite">overwrite</code></td>
<td>
<p>logical if TRUE and column exist it will be overwritten.</p>
</td></tr>
<tr><td><code id="addColor_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical if TRUE case in marker names will be ignored.</p>
</td></tr>
<tr><td><code id="addColor_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Primers in forensic STR typing kits are labeled with a fluorescent
dye. The dyes are represented with single letters (Dye) in exported result
files or with strings (Color) in 'panels' files.
For visualization in R the R color names are used (R.Color).
The function can add new color schemes matched to the existing, or
it can convert a vector containing one scheme to another.
</p>


<h3>Value</h3>

<p>data.frame with additional columns for added colors,
or vector with converted values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get marker and colors for SGM Plus.
df &lt;- getKit("SGMPlus", what = "Color")
# Add dye color.
dfDye &lt;- addColor(data = df, need = "Dye")
# Add all color alternatives.
dfAll &lt;- addColor(data = df)
# Convert a dye vector to R colors
addColor(data = c("R", "G", "Y", "B"), have = "dye", need = "r.color")
</code></pre>

<hr>
<h2 id='addData'>Adds New Data Columns to a Data Frame</h2><span id='topic+addData'></span>

<h3>Description</h3>

<p>Adds values from columns in 'new.data' to 'data' by keys.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addData(
  data,
  new.data,
  by.col,
  then.by.col = NULL,
  exact = TRUE,
  ignore.case = TRUE,
  what = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addData_+3A_data">data</code></td>
<td>
<p>Data frame containing your main data.</p>
</td></tr>
<tr><td><code id="addData_+3A_new.data">new.data</code></td>
<td>
<p>Data frame containing information you want to add to 'data'.</p>
</td></tr>
<tr><td><code id="addData_+3A_by.col">by.col</code></td>
<td>
<p>character, primary key column.</p>
</td></tr>
<tr><td><code id="addData_+3A_then.by.col">then.by.col</code></td>
<td>
<p>character, secondary key column.</p>
</td></tr>
<tr><td><code id="addData_+3A_exact">exact</code></td>
<td>
<p>logical, TRUE matches keys exact.</p>
</td></tr>
<tr><td><code id="addData_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical, TRUE ignore case.</p>
</td></tr>
<tr><td><code id="addData_+3A_what">what</code></td>
<td>
<p>character vector defining columns to add. Default is all new columns.</p>
</td></tr>
<tr><td><code id="addData_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Information in columns in data frame 'new.data' is added to data frame
'data' based on primary key value in column 'by.col', and optionally on
secondary key values in column 'then.by.col'.
</p>


<h3>Value</h3>

<p>data.frame the original data frame containing additional columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get marker names and alleles for Promega PowerPlex ESX 17.
x &lt;- getKit("ESX17", what = "Allele")
# Get marker names and colors for Promega PowerPlex ESX 17.
y &lt;- getKit("ESX17", what = "Color")
# Add color information to allele information.
z &lt;- addData(data = x, new.data = y, by.col = "Marker")
print(x)
print(y)
print(z)
</code></pre>

<hr>
<h2 id='addData_gui'>Add Data</h2><span id='topic+addData_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for <code><a href="#topic+addData">addData</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addData_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addData_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="addData_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="addData_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="addData_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+addData">addData</a></code> function by providing a graphical
user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addData">addData</a></code>
</p>

<hr>
<h2 id='addDye_gui'>Add Dye Information</h2><span id='topic+addDye_gui'></span>

<h3>Description</h3>

<p>GUI wrapper to the <code><a href="#topic+addColor">addColor</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addDye_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addDye_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="addDye_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="addDye_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="addDye_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Convenience GUI for the use of <code><a href="#topic+addColor">addColor</a></code> and
<code><a href="#topic+addOrder">addOrder</a></code> to add 'Dye', 'Color', 'R.Color', and marker 'Order'
to a dataset.
'Dye' is the one letter abbreviations for the fluorophores commonly used
to label primers in forensic STR typing kits (e.g. R and Y),
'Color' is the corresponding color name (e.g. red and yellow),
'R.Color' is the plot color used in R (e.g. red and black).
'Order' is the marker order in the selected kit.
NB! Existing columns will be overwritten.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addColor">addColor</a></code>
</p>

<hr>
<h2 id='addMarker'>Add Missing Markers.</h2><span id='topic+addMarker'></span>

<h3>Description</h3>

<p>Add missing markers to a dataset given a set of markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMarker(data, marker, ignore.case = FALSE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addMarker_+3A_data">data</code></td>
<td>
<p>data.frame or vector with sample names.</p>
</td></tr>
<tr><td><code id="addMarker_+3A_marker">marker</code></td>
<td>
<p>vector with marker names.</p>
</td></tr>
<tr><td><code id="addMarker_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical. TRUE ignores case in marker names.</p>
</td></tr>
<tr><td><code id="addMarker_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a dataset or a vector with sample names the function loops through
each sample and add any missing markers.
Returns a dataframe where each sample have at least one row per marker in
the specified marker vector. Use <code><a href="#topic+sortMarker">sortMarker</a></code> to sort the markers
according to a specified kit.
Required columns are: 'Sample.Name'.
</p>


<h3>Value</h3>

<p>data.frame.
</p>

<hr>
<h2 id='addMarker_gui'>Add Missing Markers</h2><span id='topic+addMarker_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+addMarker">addMarker</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMarker_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addMarker_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="addMarker_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="addMarker_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="addMarker_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+addMarker">addMarker</a></code> function by providing a graphical
user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addMarker">addMarker</a></code>
</p>

<hr>
<h2 id='addOrder'>Add Marker Order.</h2><span id='topic+addOrder'></span>

<h3>Description</h3>

<p>Add marker order to data frame containing a column 'Marker'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addOrder(
  data,
  kit = NULL,
  overwrite = FALSE,
  ignore.case = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addOrder_+3A_data">data</code></td>
<td>
<p>data frame or vector.</p>
</td></tr>
<tr><td><code id="addOrder_+3A_kit">kit</code></td>
<td>
<p>string representing the forensic STR kit used.
Default is NULL and automatic detection of kit will be attempted.</p>
</td></tr>
<tr><td><code id="addOrder_+3A_overwrite">overwrite</code></td>
<td>
<p>logical if TRUE and column exist it will be overwritten.</p>
</td></tr>
<tr><td><code id="addOrder_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical if TRUE case in marker names will be ignored.</p>
</td></tr>
<tr><td><code id="addOrder_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Markers in a kit appear in a certain order. Not all STR-validator functions
keep the original marker order in the result. A column indicating the marker
order is added to the dataset. This is especially useful when exporting the
data to an external spread-sheet software and allow to quickly sort the data
in the correct order.
</p>


<h3>Value</h3>

<p>data.frame with additional numeric column 'Order'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a dataset containing two samples.
data("set2")
# Add marker order when kit is known.
addOrder(data = set2, kit = "SGMPlus")
</code></pre>

<hr>
<h2 id='addSize'>Add Size Information.</h2><span id='topic+addSize'></span>

<h3>Description</h3>

<p>Add size information to alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addSize(data, kit = NA, bins = TRUE, ignore.case = FALSE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addSize_+3A_data">data</code></td>
<td>
<p>data.frame with at least columns 'Marker' and 'Allele'.</p>
</td></tr>
<tr><td><code id="addSize_+3A_kit">kit</code></td>
<td>
<p>data.frame with columns 'Marker', 'Allele', and 'Size' (for bins=TRUE) or
'Marker', 'Allele', 'Offset' and 'Repeat' (for bins=FALSE).</p>
</td></tr>
<tr><td><code id="addSize_+3A_bins">bins</code></td>
<td>
<p>logical TRUE alleles get size from corresponding bin.
If FALSE the size is calculated from the locus offset and repeat unit.</p>
</td></tr>
<tr><td><code id="addSize_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical TRUE case in marker names are ignored.</p>
</td></tr>
<tr><td><code id="addSize_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adds a column 'Size' with the fragment size in base pair (bp) for each allele as
estimated from kit bins OR calculated from offset and repeat. The bins
option return NA for alleles not in bin. The calculate option handles
all named alleles including micro variants (e.g. '9.3').
Handles 'X' and 'Y' by replacing them with '1' and '2'.
</p>


<h3>Value</h3>

<p>data.frame with additional columns for added size.
</p>

<hr>
<h2 id='addSize_gui'>Add Size Information</h2><span id='topic+addSize_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+addSize">addSize</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addSize_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addSize_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="addSize_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="addSize_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="addSize_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+addSize">addSize</a></code> function by providing a
graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addSize">addSize</a></code>
</p>

<hr>
<h2 id='auditTrail'>Log Audit Trail.</h2><span id='topic+auditTrail'></span>

<h3>Description</h3>

<p>Adds an audit trail to a dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auditTrail(
  obj,
  f.call = NULL,
  key = NULL,
  value = NULL,
  label = NULL,
  arguments = TRUE,
  exact = TRUE,
  remove = FALSE,
  package = NULL,
  rversion = TRUE,
  timestamp = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auditTrail_+3A_obj">obj</code></td>
<td>
<p>object to add or update the audit trail.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_f.call">f.call</code></td>
<td>
<p>the function call i.e. <code>match.call()</code>.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_key">key</code></td>
<td>
<p>list or vector of additional keys to log.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_value">value</code></td>
<td>
<p>list or vector of additional values to log.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_label">label</code></td>
<td>
<p>optional label used if <code>f.call=NULL</code>.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_arguments">arguments</code></td>
<td>
<p>logical. <code>TRUE</code> log function arguments.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_exact">exact</code></td>
<td>
<p>logical for exact matching of attribute name.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_remove">remove</code></td>
<td>
<p>logical. If <code>TRUE</code> the 'audit trail' attribute is removed.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_package">package</code></td>
<td>
<p>character to log the package version.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_rversion">rversion</code></td>
<td>
<p>logical to log the R version.</p>
</td></tr>
<tr><td><code id="auditTrail_+3A_timestamp">timestamp</code></td>
<td>
<p>logical to add or update timestamp.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Automatically add or updates an attribute 'audit trail' with arguments
and parameters extracted from the function call. To list the arguments
with the default set but not overridden <code>arguments=TRUE</code> must be set
(default). Additional custom key-value pairs can be added. The <code>label</code>
is extracted from the function name from <code>f.call</code>. Specify <code>package</code>
to include the version number of a package.
</p>


<h3>Value</h3>

<p>object with added or updated attribute 'audit trail'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A simple function with audit trail logging.
myFunction &lt;- function(x, a, b = 5) {
  x &lt;- x + a + b
  x &lt;- auditTrail(obj = x, f.call = match.call(), package = "strvalidator")
  return(x)
}
# Run the function.
myData &lt;- myFunction(x = 10, a = 2)
# Check the audit trail.
cat(attr(myData, "audit trail"))

# Remove the audit trail.
myData &lt;- auditTrail(myData, remove = TRUE)
# Confirm that the audit trail is removed.
cat(attr(myData, "audit trail"))
</code></pre>

<hr>
<h2 id='calculateAllele'>Calculate Allele</h2><span id='topic+calculateAllele'></span>

<h3>Description</h3>

<p>Calculates summary statistics for alleles per marker over the entire dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAllele(
  data,
  threshold = NULL,
  sex.rm = FALSE,
  kit = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAllele_+3A_data">data</code></td>
<td>
<p>data.frame including columns 'Marker' and 'Allele', and
optionally 'Height' and 'Size'.</p>
</td></tr>
<tr><td><code id="calculateAllele_+3A_threshold">threshold</code></td>
<td>
<p>numeric if not NULL only peak heights above 'threshold'
will be considered.</p>
</td></tr>
<tr><td><code id="calculateAllele_+3A_sex.rm">sex.rm</code></td>
<td>
<p>logical TRUE removes all sex markers. Requires 'kit'.</p>
</td></tr>
<tr><td><code id="calculateAllele_+3A_kit">kit</code></td>
<td>
<p>character for the DNA typing kit defining the sex markers.</p>
</td></tr>
<tr><td><code id="calculateAllele_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a table of the alleles in the dataset sorted by number of
observations.For each allele the proportion of total observations is
calculated. Using a threshold this can be used to separate likely artefacts
from likely drop-in peaks. In addition the observed allele frequency is
calculated. If columns 'Height' and/or 'Size' are available summary
statistics is calculated.
NB! The function removes NA's and OL's prior to analysis.
</p>


<h3>Value</h3>

<p>data.frame with columns 'Marker', 'Allele', 'Peaks', 'Size.Min',
'Size.Mean', 'Size.Max', 'Height.Min', 'Height.Mean', 'Height.Max',
'Total.Peaks', 'Allele.Proportion', 'Sum.Peaks', and 'Allele.Frequency'.
</p>


<h3>See Also</h3>

<p><code><a href="data.table.html#topic+data.table">data.table</a></code>
</p>

<hr>
<h2 id='calculateAllele_gui'>Calculate Allele</h2><span id='topic+calculateAllele_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateAllele">calculateAllele</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAllele_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAllele_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="calculateAllele_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateAllele_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateAllele_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateAllele">calculateAllele</a></code> function by providing a
graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='calculateAllT'>Calculate Stochastic Thresholds</h2><span id='topic+calculateAllT'></span>

<h3>Description</h3>

<p>Calculates point estimates for the stochastic threshold using multiple models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAllT(
  data,
  kit,
  p.dropout = 0.01,
  p.conservative = 0.05,
  rm.sex = TRUE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAllT_+3A_data">data</code></td>
<td>
<p>output from <code><a href="#topic+calculateDropout">calculateDropout</a></code>.</p>
</td></tr>
<tr><td><code id="calculateAllT_+3A_kit">kit</code></td>
<td>
<p>character string to define the kit which is required to remove sex markers.</p>
</td></tr>
<tr><td><code id="calculateAllT_+3A_p.dropout">p.dropout</code></td>
<td>
<p>numeric accepted risk of dropout at the stochastic threshold. Default=0.01.</p>
</td></tr>
<tr><td><code id="calculateAllT_+3A_p.conservative">p.conservative</code></td>
<td>
<p>numeric accepted risk that the actual probability of
dropout is &gt;p.dropout at the conservative estimate. Default=0.05.</p>
</td></tr>
<tr><td><code id="calculateAllT_+3A_rm.sex">rm.sex</code></td>
<td>
<p>logical default=TRUE removes sex markers defined for the given <code>kit</code>.</p>
</td></tr>
<tr><td><code id="calculateAllT_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Expects output from <code><a href="#topic+calculateDropout">calculateDropout</a></code> as input.
The function calls <code><a href="#topic+calculateT">calculateT</a></code> repeatedly to estimate the
stochastic threshold using different models. The output is a data.frame
summarizing the result. Use the <code><a href="#topic+modelDropout_gui">modelDropout_gui</a></code> to plot
individual models.
</p>
<p>Explanation of the result:
Explanatory_variable - Drop-out is the dependent variable.
An allele in heterozygous markers in the reference profile is chosen and
drop-out is scored if the other allele is not observed in the sample, i.e.
below the AT. The 'Random' method chose a random allele, while the 'LMW'
and 'HMW' method chose the low and high molecular weight allele, respectively.
The 'Locus' method score drop-out if any of the two alleles has dropped out.
As explanatory variable the peak height of the surviving allele '(Ph)',
average profile peak height '(H)', the logarithm of the surviving allele
'log(Ph)', and the logarithm of the average profile peak height 'log(H)'
is used.
P(dropout)=x.xx@T - is the point estimate for corresponding to
the specified accepted risk of drop-out.
P(dropout&gt;x.xx)&lt;0.05@T - is the conservative point estimate corresponding
to a stochastic threshold with a risk &lt;0.05 that the actual drop-out
probability is &gt;x.xx
Hosmer-Lemeshow_p - p-value from the Hosmer-Lemeshow test. A value &lt;0.05
indicates poor fit between the model and the observations.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateDropout">calculateDropout</a></code>, <code><a href="#topic+calculateT">calculateT</a></code>,
<code><a href="#topic+modelDropout_gui">modelDropout_gui</a></code>, <code><a href="#topic+plotDropout_gui">plotDropout_gui</a></code>
</p>

<hr>
<h2 id='calculateAllT_gui'>Calculate Stochastic Thresholds</h2><span id='topic+calculateAllT_gui'></span>

<h3>Description</h3>

<p>GUI wrapper to the <code><a href="#topic+calculateAllT">calculateAllT</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAllT_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAllT_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateAllT_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateAllT_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateAllT_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Convenience GUI for the use of <code><a href="#topic+calculateAllT">calculateAllT</a></code> to calculate
point estimates for the stochastic threshold using multiple models.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateAllT">calculateAllT</a></code>
</p>

<hr>
<h2 id='calculateAT'>Calculate Analytical Threshold</h2><span id='topic+calculateAT'></span>

<h3>Description</h3>

<p>Calculate analytical thresholds estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAT(
  data,
  ref = NULL,
  mask.height = TRUE,
  height = 500,
  mask.sample = TRUE,
  per.dye = TRUE,
  range.sample = 20,
  mask.ils = TRUE,
  range.ils = 10,
  k = 3,
  rank.t = 0.99,
  alpha = 0.01,
  ignore.case = TRUE,
  word = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAT_+3A_data">data</code></td>
<td>
<p>a data frame containing at least 'Dye.Sample.Peak',
'Sample.File.Name', 'Marker', 'Allele', 'Height', and 'Data.Point'.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_ref">ref</code></td>
<td>
<p>a data frame containing at least
'Sample.Name', 'Marker', 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_mask.height">mask.height</code></td>
<td>
<p>logical to indicate if high peaks should be masked.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_height">height</code></td>
<td>
<p>integer for global lower peak height threshold for peaks
to be excluded from the analysis. Active if 'mask.peak=TRUE.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_mask.sample">mask.sample</code></td>
<td>
<p>logical to indicate if sample allelic peaks should be masked.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_per.dye">per.dye</code></td>
<td>
<p>logical TRUE if sample peaks should be masked per dye channel.
FALSE if sample peaks should be masked globally across dye channels.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_range.sample">range.sample</code></td>
<td>
<p>integer to specify the masking range in (+/-) data points.
Active if mask.sample=TRUE.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_mask.ils">mask.ils</code></td>
<td>
<p>logical to indicate if internal lane standard peaks should be masked.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_range.ils">range.ils</code></td>
<td>
<p>integer to specify the masking range in (+/-) data points.
Active if mask.ils=TRUE.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_k">k</code></td>
<td>
<p>numeric factor for the desired confidence level (method AT1).</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_rank.t">rank.t</code></td>
<td>
<p>numeric percentile rank threshold (method AT2).</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_alpha">alpha</code></td>
<td>
<p>numeric one-sided confidence interval to obtain the
critical value from the t-distribution (method AT4).</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical to indicate if sample matching should ignore case.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_word">word</code></td>
<td>
<p>logical to indicate if word boundaries should be added before sample matching.</p>
</td></tr>
<tr><td><code id="calculateAT_+3A_debug">debug</code></td>
<td>
<p>logical to indicate if debug information should be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculate the analytical threshold (AT) according to method 1, 2, and 4 as
recommended in the reference by analyzing the background signal (noise).
In addition method 7, a log-normal version of method 1 has been implemented.
Method 1: The average signal + 'k' * the standard deviation.
Method 2: The percentile rank method. The percentage of noise peaks below 'rank.t'.
Method 4: Utilize the mean and standard deviation and the critical value obtained
from the t-distribution for confidence interval 'alpha' (one-sided) and observed
peaks analyzed (i.e. not masked) minus one as degrees of freedom, and the number
of samples.
Method 7: The average natural logarithm of the signal + k * the standard deviation.
</p>
<p>If samples containing DNA are used, a range around the allelic peaks can be masked
from the analysis to discard peaks higher than the noise. Masking can be within
each dye or across all dye channels.
Similarly a range around the peaks of the internal lane standard (ILS) can be
masked across all dye channels. Which can bleed-through in week samples
(i.e. negative controls)
The mean, standard deviation, and number of peaks are calculated per dye per sample,
per sample, globally across all samples, and globally across all samples per dye,
for each method to estimate AT. Also the complete percentile rank list is calculated.
</p>


<h3>Value</h3>

<p>list of three data frames. The first with result per dye per sample,
per sample, globally across all samples, and globally across all samples
per dye, for each method. The second is the complete percentile rank list.
The third is the masked raw data used for calculation to enable manual
check of the result.
</p>


<h3>References</h3>

<p>J. Bregu et.al.,
Analytical thresholds and sensitivity: establishing RFU thresholds for
forensic DNA analysis, J. Forensic Sci. 58 (1) (2013) 120-129,
ISSN 1556-4029, DOI: 10.1111/1556-4029.12008.
<a href="https://doi.org/10.1111/1556-4029.12008">doi:10.1111/1556-4029.12008</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maskAT">maskAT</a></code>, <code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='calculateAT_gui'>Calculate Analytical Threshold</h2><span id='topic+calculateAT_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+maskAT">maskAT</a></code> and <code><a href="#topic+calculateAT">calculateAT</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAT_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAT_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateAT_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateAT_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateAT_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateAT">calculateAT</a></code> and
<code><a href="#topic+calculateAT">calculateAT</a></code> function by providing a graphical user interface.
In addition there are integrated control functions.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateAT">calculateAT</a></code>, <code><a href="#topic+maskAT">maskAT</a></code>,
<code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='calculateAT6'>Calculate Analytical Threshold</h2><span id='topic+calculateAT6'></span>

<h3>Description</h3>

<p>Calculate analytical thresholds estimate using linear regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAT6(
  data,
  ref,
  amount = NULL,
  weighted = TRUE,
  alpha = 0.05,
  ignore.case = TRUE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAT6_+3A_data">data</code></td>
<td>
<p>data.frame containing at least columns 'Sample.Name', 'Marker',
'Allele', and 'Height'.</p>
</td></tr>
<tr><td><code id="calculateAT6_+3A_ref">ref</code></td>
<td>
<p>data.frame containing at least columns 'Sample.Name', 'Marker',
and 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateAT6_+3A_amount">amount</code></td>
<td>
<p>data.frame containing at least columns 'Sample.Name' and 'Amount'.
If NULL 'data' must contain a column 'Amount'.</p>
</td></tr>
<tr><td><code id="calculateAT6_+3A_weighted">weighted</code></td>
<td>
<p>logical to calculate weighted linear regression (weight=1/se^2).</p>
</td></tr>
<tr><td><code id="calculateAT6_+3A_alpha">alpha</code></td>
<td>
<p>numeric [0,1] significance level for the t-statistic.</p>
</td></tr>
<tr><td><code id="calculateAT6_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical to indicate if sample matching should ignore case.</p>
</td></tr>
<tr><td><code id="calculateAT6_+3A_debug">debug</code></td>
<td>
<p>logical to indicate if debug information should be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculate the analytical threshold (AT) according to method 6 as
outlined in the reference. In short serial dilutions are analyzed
and the average peak height is calculated. Linear regression or
Weighted linear regression with amount of DNA as the predictor for
the peak height is performed.
Method 6: A simplified version of the upper limit approach.
AT6 = y-intercept + t-statistic * standard error of the regression.
Assumes the y-intercept is not different from the mean blank signal.
The mean blank signal should be included in the confidence range
('Lower' to 'AT6' in the resulting data frame).
NB! This is an indirect method to estimate AT and should be verified
by other methods. From the reference: A way to determine the validity
of this approach is based on whether the y-intercept +- (1-a)100
contains the mean blank signal. If the mean blank signal is included
in the y-intercept band, the following relationship [i.e. AT6] can be used
to determine the AT. However, it should be noted that the ATs derived in
this manner need to be calculated for each color and for all preparations
(i.e., different injections, sample preparation volumes, post-PCR cleanup,
etc.).
NB! Quality sensors must be removed prior to analysis.
</p>


<h3>Value</h3>

<p>data.frame with columns 'Amount', 'Height', 'Sd', 'Weight', 'N',
'Alpha', 'Lower', 'Intercept', and 'AT6'.
</p>


<h3>References</h3>

<p>J. Bregu et.al.,
Analytical thresholds and sensitivity: establishing RFU thresholds for
forensic DNA analysis, J. Forensic Sci. 58 (1) (2013) 120-129,
ISSN 1556-4029, DOI: 10.1111/1556-4029.12008.
<a href="https://doi.org/10.1111/1556-4029.12008">doi:10.1111/1556-4029.12008</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateAT6_gui">calculateAT6_gui</a></code>, <code><a href="#topic+calculateAT">calculateAT</a></code>,
<code><a href="#topic+calculateAT_gui">calculateAT_gui</a></code>, <code><a href="stats.html#topic+lm">lm</a></code>
</p>

<hr>
<h2 id='calculateAT6_gui'>Calculate Analytical Threshold</h2><span id='topic+calculateAT6_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateAT6">calculateAT6</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateAT6_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateAT6_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateAT6_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateAT6_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateAT6_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Scores dropouts for a dataset.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateAT6">calculateAT6</a></code>, <code><a href="#topic+calculateAT">calculateAT</a></code>,
<code><a href="#topic+calculateAT_gui">calculateAT_gui</a></code>, <code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='calculateCapillary'>Calculate Capillary Balance</h2><span id='topic+calculateCapillary'></span>

<h3>Description</h3>

<p>Calculates the ILS inter capillary balance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateCapillary(samples.table, plot.table, sq = 0, run = "", debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateCapillary_+3A_samples.table">samples.table</code></td>
<td>
<p>data frame containing at least the columns
'Sample.File', 'Sample.Name', 'Size.Standard', 'Instrument.Type',
'Instrument.ID', 'Cap', 'Well', and 'SQ'.</p>
</td></tr>
<tr><td><code id="calculateCapillary_+3A_plot.table">plot.table</code></td>
<td>
<p>data frame containing at least the columns
'Sample.File.Name', 'Size', and 'Height'.</p>
</td></tr>
<tr><td><code id="calculateCapillary_+3A_sq">sq</code></td>
<td>
<p>numeric threshold for 'Sizing Quality' (SQ).</p>
</td></tr>
<tr><td><code id="calculateCapillary_+3A_run">run</code></td>
<td>
<p>character string for run name.</p>
</td></tr>
<tr><td><code id="calculateCapillary_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates the inter capillary balance for the internal lane standard (ILS).
Require information from both the 'samples.table' and the 'plot.table'.
</p>


<h3>Value</h3>

<p>data.frame with with columns 'Instrument', 'Instrument.ID', 'Run',
'Mean.Height', 'SQ', 'Injection', 'Capillary', 'Well', 'Comment'.
</p>

<hr>
<h2 id='calculateCapillary_gui'>Calculate Capillary Balance</h2><span id='topic+calculateCapillary_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateCapillary">calculateCapillary</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateCapillary_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateCapillary_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateCapillary_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateCapillary_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateCapillary_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateCapillary">calculateCapillary</a></code> function by providing
a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateCapillary">calculateCapillary</a></code>
</p>

<hr>
<h2 id='calculateConcordance'>Calculate Concordance.</h2><span id='topic+calculateConcordance'></span>

<h3>Description</h3>

<p>Calculates concordance and discordance for profiles in multiple datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateConcordance(
  data,
  kit.name = NA,
  no.marker = "NO MARKER",
  no.sample = "NO SAMPLE",
  delimeter = ",",
  list.all = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateConcordance_+3A_data">data</code></td>
<td>
<p>list of data frames in 'slim' format with at least columns
'Sample.Name', 'Marker', and 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateConcordance_+3A_kit.name">kit.name</code></td>
<td>
<p>character vector for DNA typing kit names in same order and
of same lengths as data sets in 'data' list. Default is NA in which case
they will be numbered.</p>
</td></tr>
<tr><td><code id="calculateConcordance_+3A_no.marker">no.marker</code></td>
<td>
<p>character vector for string when marker is missing.</p>
</td></tr>
<tr><td><code id="calculateConcordance_+3A_no.sample">no.sample</code></td>
<td>
<p>character vector for string when sample is missing.</p>
</td></tr>
<tr><td><code id="calculateConcordance_+3A_delimeter">delimeter</code></td>
<td>
<p>character to separate the alleles in a genotype.
Default is comma e.g '12,16'.</p>
</td></tr>
<tr><td><code id="calculateConcordance_+3A_list.all">list.all</code></td>
<td>
<p>logical TRUE to return missing samples.</p>
</td></tr>
<tr><td><code id="calculateConcordance_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes a list of datasets as input. It is assumed that each unique
sample name represent a result originating from the same source DNA and
thus is expected to give identical DNA profiles. The function first compare
the profiles for each sample across datasets and lists discordant results.
Then it performs a pair-wise comparison and compiles a concordance table.
The tables are returned as two data frames in a list.
NB! Typing and PCR artefacts (spikes, off-ladder peaks, stutters etc.)
must be removed before analysis.
NB! It is expected that the unique set of marker names across a dataset is
present in each sample for that dataset (a missing marker is a discordance).
</p>


<h3>Value</h3>

<p>list of data.frames (discordance table, and pair-wise comparison).
</p>

<hr>
<h2 id='calculateConcordance_gui'>Calculate Concordance</h2><span id='topic+calculateConcordance_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateConcordance">calculateConcordance</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateConcordance_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateConcordance_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateConcordance_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateConcordance_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateConcordance_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateConcordance">calculateConcordance</a></code> function by
providing a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateConcordance">calculateConcordance</a></code>
</p>

<hr>
<h2 id='calculateCopies'>Calculate Allele Copies</h2><span id='topic+calculateCopies'></span>

<h3>Description</h3>

<p>Calculates the number of alleles in each marker.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateCopies(
  data,
  observed = FALSE,
  copies = TRUE,
  heterozygous = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateCopies_+3A_data">data</code></td>
<td>
<p>Data frame containing at least columns 'Sample.Name', 'Marker,
and 'Allele*'.</p>
</td></tr>
<tr><td><code id="calculateCopies_+3A_observed">observed</code></td>
<td>
<p>logical indicating if a column 'Observed' should be
used to count the number of unique alleles.</p>
</td></tr>
<tr><td><code id="calculateCopies_+3A_copies">copies</code></td>
<td>
<p>logical indicating if a column 'Copies' should be used to
indicate the number of allele copies with 1 for heterozygotes and 2 for
homozygotes.</p>
</td></tr>
<tr><td><code id="calculateCopies_+3A_heterozygous">heterozygous</code></td>
<td>
<p>logical indicating if a column 'Heterozygous' should be
used to indicate heterozygotes with 1 and homozygotes with 0.</p>
</td></tr>
<tr><td><code id="calculateCopies_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates the number of unique values in the 'Allele*' columns
for each marker, the number of allele copies, or indicate heterozygous loci.
Observed - number of unique alleles.
Copies - number of allele copies, '1' for heterozygotes and '2' for homozygotes.
Heterozygous - '1' for heterozygous and '0' for homozygous loci.
NB! The 'copies' and 'heterozygous' option are intended for known complete
profiles, while 'observed' can be used for any samples to count the number
of peaks. Sample names must be unique. The result is per marker but repeated
for each row of that marker. Data in 'fat' format is auto slimmed.
</p>


<h3>Value</h3>

<p>data.frame the original data frame with optional columns 'Observed',
'Copies', and 'Heterozygous'.
</p>

<hr>
<h2 id='calculateCopies_gui'>Calculate Allele Copies</h2><span id='topic+calculateCopies_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code>link{calculateCopies}</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateCopies_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateCopies_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateCopies_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateCopies_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateCopies_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateCopies">calculateCopies</a></code> function by
providing a graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateCopies">calculateCopies</a></code>
</p>

<hr>
<h2 id='calculateDropout'>Calculate Drop-out Events</h2><span id='topic+calculateDropout'></span>

<h3>Description</h3>

<p>Calculate drop-out events (allele and locus) and records the surviving peak height.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateDropout(
  data,
  ref,
  threshold = NULL,
  method = c("1", "2", "X", "L"),
  ignore.case = TRUE,
  sex.rm = FALSE,
  qs.rm = TRUE,
  kit = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateDropout_+3A_data">data</code></td>
<td>
<p>data frame in GeneMapper format containing at least a column 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateDropout_+3A_ref">ref</code></td>
<td>
<p>data frame in GeneMapper format.</p>
</td></tr>
<tr><td><code id="calculateDropout_+3A_threshold">threshold</code></td>
<td>
<p>numeric, threshold in RFU defining a dropout event.
Default is 'NULL' and dropout is scored purely on the absence of a peak.</p>
</td></tr>
<tr><td><code id="calculateDropout_+3A_method">method</code></td>
<td>
<p>character vector, specifying which scoring method(s) to use.
Method 'X' for random allele, '1' or '2' for the low/high molecular weight allele,
and 'L' for the locus method (the option is case insensitive).</p>
</td></tr>
<tr><td><code id="calculateDropout_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical, default TRUE for case insensitive.</p>
</td></tr>
<tr><td><code id="calculateDropout_+3A_sex.rm">sex.rm</code></td>
<td>
<p>logical, default FALSE to include sex markers in the analysis.</p>
</td></tr>
<tr><td><code id="calculateDropout_+3A_qs.rm">qs.rm</code></td>
<td>
<p>logical, default TRUE to exclude quality sensors from the analysis.</p>
</td></tr>
<tr><td><code id="calculateDropout_+3A_kit">kit</code></td>
<td>
<p>character, required if sex.rm=TRUE or qs.rm=TRUE to define the kit.</p>
</td></tr>
<tr><td><code id="calculateDropout_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates drop-out events. In case of allele dropout the peak height of the
surviving allele is given. Homozygous alleles in the reference set can be
either single or double notation (X or X X). Markers present in the
reference set but not in the data set will be added to the result.
NB! 'Sample.Name' in 'ref' must be unique core name of replicate sample
names in 'data'.
Use <code>checkSubset</code> to make sure subsetting works as intended.
There are options to remove sex markers and quality sensors from analysis.
</p>
<p>NB! There are several methods of scoring drop-out events for regression.
Currently the 'MethodX', 'Method1', and 'Method2' are endorsed by the DNA
commission (see Appendix B in ref 1). However, an alternative method is to
consider the whole locus and score drop-out if any allele is missing.
</p>
<p>Explanation of the methods:
Dropout - all alleles are scored according to AT. This is pure observations
and is not used for modeling.
MethodX - a random reference allele is selected and drop-out is scored in
relation to the the partner allele.
Method1 - the low molecular weight allele is selected and drop-out is
scored in relation to the partner allele.
Method2 - the high molecular weight allele is selected and drop-out is
scored in relation to the partner allele.
MethodL - drop-out is scored per locus i.e. drop-out if any allele has
dropped out.
</p>
<p>Method X/1/2 records the peak height of the partner allele to be used as
the explanatory variable in the logistic regression. The locus method L also
do this when there has been a drop-out, if not the the mean peak height for
the locus is used. Peak heights for the locus method are stored in a
separate column.
</p>


<h3>Value</h3>

<p>data.frame with columns 'Sample.Name', 'Marker', 'Allele', 'Height', 'Dropout',
'Rfu', 'Heterozygous', and 'Model'.
Dropout: 0 indicate no dropout, 1 indicate allele dropout, and 2 indicate locus dropout.
Rfu: height of surviving allele.
Heterozygous: 1 for heterozygous and 0 for homozygous.
And any of the following containing the response (or explanatory) variable used
for modeling by logistic regression in function <code>modelDropout</code>:
'MethodX', 'Method1', 'Method2', 'MethodL' and 'MethodL.Ph'.
</p>


<h3>References</h3>

<p>Peter Gill et.al.,
DNA commission of the International Society of Forensic Genetics:
Recommendations on the evaluation of STR typing results that may
include drop-out and/or drop-in using probabilistic methods,
Forensic Science International: Genetics, Volume 6, Issue 6, December 2012,
Pages 679-688, ISSN 1872-4973, 10.1016/j.fsigen.2012.06.002.
<a href="https://doi.org/10.1016/j.fsigen.2012.06.002">doi:10.1016/j.fsigen.2012.06.002</a>
</p>
<p>Peter Gill, Roberto Puch-Solis, James Curran,
The low-template-DNA (stochastic) threshold-Its determination relative to
risk analysis for national DNA databases,
Forensic Science International: Genetics, Volume 3, Issue 2, March 2009,
Pages 104-111, ISSN 1872-4973, 10.1016/j.fsigen.2008.11.009.
<a href="https://doi.org/10.1016/j.fsigen.2008.11.009">doi:10.1016/j.fsigen.2008.11.009</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(set4)
data(ref4)
drop &lt;- calculateDropout(data = set4, ref = ref4, kit = "ESX17", ignore.case = TRUE)
</code></pre>

<hr>
<h2 id='calculateDropout_gui'>Calculate Dropout Events</h2><span id='topic+calculateDropout_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateDropout">calculateDropout</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateDropout_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateDropout_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateDropout_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateDropout_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateDropout_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Scores dropouts for a dataset.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateDropout">calculateDropout</a></code>, <code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='calculateHb'>Calculate Heterozygote Balance</h2><span id='topic+calculateHb'></span>

<h3>Description</h3>

<p>Calculates the heterozygote (intra-locus) peak balance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateHb(
  data,
  ref,
  hb = 1,
  kit = NULL,
  sex.rm = FALSE,
  qs.rm = FALSE,
  ignore.case = TRUE,
  exact = FALSE,
  word = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateHb_+3A_data">data</code></td>
<td>
<p>a data frame containing at least
'Sample.Name', 'Marker', 'Height', and 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_ref">ref</code></td>
<td>
<p>a data frame containing at least
'Sample.Name', 'Marker', 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_hb">hb</code></td>
<td>
<p>numerical, definition of heterozygote balance. Default is hb=1.
hb=1: HMW/LMW, hb=2: LMW/HMW, hb=3; min(Ph)/max(Ph).</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_kit">kit</code></td>
<td>
<p>character defining the kit used. If NULL automatic detection is attempted.</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_sex.rm">sex.rm</code></td>
<td>
<p>logical TRUE removes sex markers defined by 'kit'.</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_qs.rm">qs.rm</code></td>
<td>
<p>logical TRUE removes quality sensors defined by 'kit'.</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical indicating if sample matching should ignore case.</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_exact">exact</code></td>
<td>
<p>logical indicating if exact sample matching should be used.</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_word">word</code></td>
<td>
<p>logical indicating if word boundaries should be added before sample matching.</p>
</td></tr>
<tr><td><code id="calculateHb_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates the heterozygote (intra-locus) peak balance for a dataset.
Known allele peaks will be extracted using the reference prior to analysis.
Calculates the heterozygote balance (Hb), size difference between
heterozygous alleles (Delta), and mean peak height (MPH).
NB! 'X' and 'Y' will be handled as '1' and '2' respectively.
</p>


<h3>Value</h3>

<p>data.frame with with columns 'Sample.Name', 'Marker', 'Delta', 'Hb', 'MPH'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ref2)
data(set2)
# Calculate average balances.
calculateHb(data = set2, ref = ref2)
</code></pre>

<hr>
<h2 id='calculateHb_gui'>Calculate Heterozygote Balance</h2><span id='topic+calculateHb_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateHb">calculateHb</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateHb_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateHb_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateHb_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateHb_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateHb_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateHb">calculateHb</a></code> function
by providing a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code>link{calculateHb}</code>, <code>link{checkSubset}</code>
</p>

<hr>
<h2 id='calculateHeight'>Calculate Peak Height.</h2><span id='topic+calculateHeight'></span>

<h3>Description</h3>

<p>Calculate peak height metrics for samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateHeight(
  data,
  ref = NULL,
  na.replace = NULL,
  add = TRUE,
  exclude = NULL,
  sex.rm = FALSE,
  qs.rm = FALSE,
  kit = NULL,
  ignore.case = TRUE,
  exact = FALSE,
  word = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateHeight_+3A_data">data</code></td>
<td>
<p>data.frame with at least columns 'Sample.Name' and 'Height'.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_ref">ref</code></td>
<td>
<p>data.frame with at least columns 'Sample.Name' and 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_na.replace">na.replace</code></td>
<td>
<p>replaces NA values in the final result.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_add">add</code></td>
<td>
<p>logical default is TRUE which will add or overwrite columns
'TPH', 'Peaks', 'H', and 'Proportion' in the provided 'data'.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_exclude">exclude</code></td>
<td>
<p>character vector (case sensitive) e.g. &quot;OL&quot; excludes rows with
&quot;OL&quot; in the 'Allele' column (not necessary when a reference dataset is provided).</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_sex.rm">sex.rm</code></td>
<td>
<p>logical, default FALSE to include sex markers in the analysis.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_qs.rm">qs.rm</code></td>
<td>
<p>logical, default TRUE to exclude quality sensors from the analysis.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_kit">kit</code></td>
<td>
<p>character, required if sex.rm=TRUE or qs.rm=TRUE to define the kit.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical TRUE ignores case in sample name matching.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_exact">exact</code></td>
<td>
<p>logical TRUE for exact sample name matching.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_word">word</code></td>
<td>
<p>logical TRUE to add word boundaries to sample name matching.</p>
</td></tr>
<tr><td><code id="calculateHeight_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates the total peak height (TPH), and number of observed peaks (Peaks),
for each sample by default. If a reference dataset is provided average peak
height (H), and profile proportion (Proportion) are calculated.
</p>
<p>H is calculated according to the formula (references [1][2]):
<code class="reqn">H = sum(peak heights)/(n[het] + 2n[hom]</code>
Where:
n[het] = number of observed heterozygous alleles
n[hom] = number of observed homozygous alleles
</p>
<p>Important: The above formula has a drawback that when many alleles have
dropped out, i.e. when only few alleles are detected, H can be overestimated.
For example, if there are only 1 (homozygote) peak observed in the profile,
with a height of 100 RFU, then H=100 RFU. This means that the value of H will
always be between half the analytical threshold (AT/2) and the peak height of
the observed allele (if only one). For this reason Tvedebrink et al. actually
modified the estimate to take the number of expected alleles into account
when estimating the expected peak height (reference [3]). Basically, they adjust
the estimated peak height for the fact that they know how many alleles that
fall below the AT, such that the expected peak height could be estimated lower
than AT. In addition, they account for degradation using a log-linear
relationship on peak heights and fragment length.
</p>
<p>Tip: If it is known that all expected peaks are observed and no unexpected
peaks are present, the dataset can be used as a reference for itself.
</p>
<p>Note: If a reference dataset is provided the known alleles will be extracted
from the dataset.
</p>


<h3>Value</h3>

<p>data.frame with with at least columns 'Sample.Name', 'TPH', and 'Peaks'.
</p>


<h3>References</h3>

<p>[1] Torben Tvedebrink, Poul Svante Eriksen, Helle Smidt Mogensen, Niels Morling,
Evaluating the weight of evidence by using quantitative short tandem repeat data in DNA mixtures
Journal of the Royal Statistical Society: Series C (Applied Statistics),
Volume 59, Issue 5, 2010,
Pages 855-874, 10.1111/j.1467-9876.2010.00722.x.
<a href="https://doi.org/10.1111/j.1467-9876.2010.00722.x">doi:10.1111/j.1467-9876.2010.00722.x</a>
</p>
<p>[2] Torben Tvedebrink, Helle Smidt Mogensen, Maria Charlotte Stene, Niels Morling,
Performance of two 17 locus forensic identification STR kits-Applied Biosystems's AmpFlSTR NGMSElect and Promega's PowerPlex ESI17 kits
Forensic Science International: Genetics,
Volume 6, Issue 5, 2012,
Pages 523-531, 10.1016/j.fsigen.2011.12.006.
<a href="https://doi.org/10.1016/j.fsigen.2011.12.006">doi:10.1016/j.fsigen.2011.12.006</a>
</p>
<p>[3] Torben Tvedebrink, Maria Asplund, Poul Svante Eriksen, Helle Smidt Mogensen, Niels Morling,
Estimating drop-out probabilities of STR alleles accounting for stutters, detection threshold truncation and degradation
Forensic Science International: Genetics Supplement Series,
Volume 4, Issue 1, 2013,
Pages e51-e52, 10.1016/j.fsigss.2013.10.026.
<a href="https://doi.org/10.1016/j.fsigss.2013.10.026">doi:10.1016/j.fsigss.2013.10.026</a>
</p>

<hr>
<h2 id='calculateHeight_gui'>Calculate Peak Height</h2><span id='topic+calculateHeight_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateHeight">calculateHeight</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateHeight_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateHeight_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateHeight_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateHeight_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateHeight_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateHeight">calculateHeight</a></code> function by providing a graphical
user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>References</h3>

<p>Torben Tvedebrink, Poul Svante Eriksen, Helle Smidt Mogensen, Niels Morling,
Evaluating the weight of evidence by using quantitative short tandem repeat data in DNA mixtures
Journal of the Royal Statistical Society: Series C (Applied Statistics),
Volume 59, Issue 5, 2010,
Pages 855-874, 10.1111/j.1467-9876.2010.00722.x.
<a href="https://doi.org/10.1111/j.1467-9876.2010.00722.x">doi:10.1111/j.1467-9876.2010.00722.x</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateHeight">calculateHeight</a></code>
</p>

<hr>
<h2 id='calculateLb'>Calculate Inter-locus Balance</h2><span id='topic+calculateLb'></span>

<h3>Description</h3>

<p>Calculates the inter-locus balance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateLb(
  data,
  ref = NULL,
  option = "prop",
  by.dye = FALSE,
  ol.rm = TRUE,
  sex.rm = FALSE,
  qs.rm = FALSE,
  na = NULL,
  kit = NULL,
  ignore.case = TRUE,
  word = FALSE,
  exact = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateLb_+3A_data">data</code></td>
<td>
<p>data.frame containing at least
'Sample.Name', 'Marker', and 'Height'.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_ref">ref</code></td>
<td>
<p>data.frame containing at least 'Sample.Name', 'Marker', 'Allele'.
If provided alleles matching 'ref' will be extracted from 'data'
(see <code><a href="#topic+filterProfile">filterProfile</a></code>).</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_option">option</code></td>
<td>
<p>character: 'prop' for proportional Lb, 'norm' for normalized
LB, 'cent' for centred Lb, and 'peak' for the min and max peak height ratio.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_by.dye">by.dye</code></td>
<td>
<p>logical. Default is FALSE for global Lb, if TRUE Lb is calculated
within each dye channel.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_ol.rm">ol.rm</code></td>
<td>
<p>logical. Default is TRUE indicating that off-ladder 'OL' alleles
will be removed.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_sex.rm">sex.rm</code></td>
<td>
<p>logical. Default is FALSE indicating that all markers will be
considered. If TRUE sex markers will be removed.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_qs.rm">qs.rm</code></td>
<td>
<p>logical. Default is TRUE indicating that all quality sensors
will be removed.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_na">na</code></td>
<td>
<p>numeric. Numeric to replace NA values e.g. locus dropout can be
given a peak height equal to the limit of detection threshold, or zero.
Default is NULL indicating that NA will be treated as missing values.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_kit">kit</code></td>
<td>
<p>character providing the kit name. Attempt to auto detect if NULL.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical indicating if sample matching should ignore case.
Only used if 'ref' is provided and 'data' is filtered.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_word">word</code></td>
<td>
<p>logical indicating if word boundaries should be added before sample matching.
Only used if 'ref' is provided and 'data' is filtered.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_exact">exact</code></td>
<td>
<p>logical indicating if exact sample matching should be used.
Only used if 'ref' is provided and 'data' is filtered.</p>
</td></tr>
<tr><td><code id="calculateLb_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The inter-locus balance (Lb), or profile balance, can be calculated
as a proportion of the whole, normalized, or as centred quantities (as in
the reference but using the mean total marker peak height instead of H).
Lb can be calculated globally across the complete profile or within each dye
channel. All markers must be present in each sample. Data can be unfiltered
or filtered since the sum of peak heights by marker is used. A reference
dataset is required to filter the dataset, which also adds any missing
markers. A kit should be provided for filtering of known profile, sex
markers, or quality sensors. If not automatic detection will be attempted.
If missing, dye will be added according to kit. Off-ladder alleles and
quality sensors are by default removed from the dataset.
Sex markers are optionally removed.
Some columns in the result may vary:
TPH: Total (marker) Peak Height.
TPPH: Total Profile Peak Height.
MTPH: Maximum (sample) Total Peak Height.
MPH: Mean (marker) Peak Height.
</p>


<h3>Value</h3>

<p>data.frame with at least columns 'Sample.Name', 'Marker', 'TPH', 'Peaks', and 'Lb'.
See description for additional columns.
</p>


<h3>References</h3>

<p>Torben Tvedebrink et.al.,
Performance of two 17 locus forensic identification STR kits-Applied
Biosystems's AmpFlSTR NGMSElect and Promega's PowerPlex ESI17 kits,
Forensic Science International: Genetics, Volume 6, Issue 5, September 2012,
Pages 523-531, ISSN 1872-4973, 10.1016/j.fsigen.2011.12.006.
<a href="https://doi.org/10.1016/j.fsigen.2011.12.006">doi:10.1016/j.fsigen.2011.12.006</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data.
data(set2)

# Calculate inter-locus balance.
res &lt;- calculateLb(data = set2)
print(res)
</code></pre>

<hr>
<h2 id='calculateLb_gui'>Calculate Locus Balance</h2><span id='topic+calculateLb_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateLb">calculateLb</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateLb_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateLb_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateLb_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateLb_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateLb_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateLb">calculateLb</a></code> function
by providing a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code>link{calculateLb}</code>, <code>link{checkSubset}</code>
</p>

<hr>
<h2 id='calculateMixture'>Calculate Mixture.</h2><span id='topic+calculateMixture'></span>

<h3>Description</h3>

<p>Calculate Mx, drop-in, and 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateMixture(
  data,
  ref1,
  ref2,
  ol.rm = TRUE,
  ignore.dropout = TRUE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateMixture_+3A_data">data</code></td>
<td>
<p>list of data frames in 'slim' format with at least columns
'Sample.Name', 'Marker', and 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateMixture_+3A_ref1">ref1</code></td>
<td>
<p>data.frame with known genotypes for the major contributor.</p>
</td></tr>
<tr><td><code id="calculateMixture_+3A_ref2">ref2</code></td>
<td>
<p>data.frame with known genotypes for the minor contributor.</p>
</td></tr>
<tr><td><code id="calculateMixture_+3A_ol.rm">ol.rm</code></td>
<td>
<p>logical TRUE removes off-ladder alleles (OL), FALSE count
OL as drop-in.</p>
</td></tr>
<tr><td><code id="calculateMixture_+3A_ignore.dropout">ignore.dropout</code></td>
<td>
<p>logical TRUE calculate Mx also if there are
missing alleles.</p>
</td></tr>
<tr><td><code id="calculateMixture_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a set of mixture results, reference profiles for the
major component, and reference profile for the minor component the
function calculates the mixture proportion (Mx), the average Mx, the
absolute difference D=|Mx-AvgMx| for each marker, the percentage
profile for the minor component, number of drop-ins.
The observed and expected number of free alleles for the minor component
(used to calculate the profile percentage) is also given.
</p>
<p>NB! All sample names must be unique within and between each reference dataset.
NB! Samples in ref1 and ref2 must be in 'sync'. The first sample in
ref1 is combined with the first sample in ref2 to make a mixture sample.
For example: ref1 &quot;A&quot; and ref2 &quot;B&quot; match mixture samples &quot;A_B_1&quot;, &quot;A_B_2&quot;
and so on.
NB! If reference datasets have unequal number of unique samples the smaller
dataset will limit the calculation.
</p>
<p>Mixture proportion is calculated in accordance with:
<br /> Locus style (minor:MAJOR) | Mx
<br /> AA:AB | (A-B)/(A+B)
<br /> AB:AA | (2*B)/(A+B)
<br /> AB:AC | B/(B+C)
<br /> AA:BB | A/(A+B)
<br /> AB:CC | (A+B)/(A+B+C)
<br /> AB:CD | (A+B)/(A+B+C+D)
<br /> AB:AB | NA - cannot be calculated
<br /> AA:AA | NA - cannot be calculated
</p>


<h3>Value</h3>

<p>data.frame with columns 'Sample.Name', 'Marker', 'Style', 'Mx',
'Average', 'Difference', 'Observed', 'Expected', 'Profile', and 'Dropin'.
</p>


<h3>References</h3>

<p>Bright, Jo-Anne, Jnana Turkington, and John Buckleton.
&quot;Examination of the Variability in Mixed DNA Profile Parameters for the
Identifiler Multiplex.&quot;
Forensic Science International: Genetics 4, no. 2 (February 2010): 111-14.
doi:10.1016/j.fsigen.2009.07.002.
<a href="https://doi.org/10.1016/j.fsigen.2009.07.002">doi:10.1016/j.fsigen.2009.07.002</a>
</p>

<hr>
<h2 id='calculateMixture_gui'>Calculate Mixture</h2><span id='topic+calculateMixture_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateMixture">calculateMixture</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateMixture_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateMixture_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateMixture_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateMixture_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateMixture_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateMixture">calculateMixture</a></code> function by
providing a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateMixture">calculateMixture</a></code>, <code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='calculateOL'>Analyze Off-ladder Alleles</h2><span id='topic+calculateOL'></span>

<h3>Description</h3>

<p>Analyze the risk for off-ladder alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateOL(kit, db, virtual = TRUE, limit = TRUE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateOL_+3A_kit">kit</code></td>
<td>
<p>data.frame, providing kit information.</p>
</td></tr>
<tr><td><code id="calculateOL_+3A_db">db</code></td>
<td>
<p>data.frame, allele frequency database.</p>
</td></tr>
<tr><td><code id="calculateOL_+3A_virtual">virtual</code></td>
<td>
<p>logical default is TRUE, calculation includes virtual alleles.</p>
</td></tr>
<tr><td><code id="calculateOL_+3A_limit">limit</code></td>
<td>
<p>logical default is TRUE, limit small frequencies to 5/2N.</p>
</td></tr>
<tr><td><code id="calculateOL_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By analyzing the allelic ladders the risk for getting off-ladder
(OL) alleles are calculated. The frequencies from a provided population
database is used to calculate the risk per marker and in total for the given
kit(s). Virtual alleles can be excluded from the calculation.
Small frequencies can be limited to the estimate 5/2N.
</p>


<h3>Value</h3>

<p>data.frame with columns 'Kit', 'Marker', 'Database', 'Risk', and 'Total'.
</p>

<hr>
<h2 id='calculateOL_gui'>Analyze Off-ladder Alleles</h2><span id='topic+calculateOL_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateOL">calculateOL</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateOL_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = TRUE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateOL_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateOL_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateOL_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateOL_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By analysis of the allelic ladder the risk for getting off-ladder
(OL) alleles are calculated. The frequencies from a provided population
database is used to calculate the risk per marker and in total for the given
kit(s). Virtual alleles can be excluded from the calculation.
Small frequencies can be limited to the estimate 5/2N.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateOL">calculateOL</a></code>
</p>

<hr>
<h2 id='calculateOverlap'>Calculate Bins Overlap</h2><span id='topic+calculateOverlap'></span>

<h3>Description</h3>

<p>Analyses the bins overlap between colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateOverlap(
  data,
  db = NULL,
  penalty = NULL,
  virtual = TRUE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateOverlap_+3A_data">data</code></td>
<td>
<p>data frame providing kit information.</p>
</td></tr>
<tr><td><code id="calculateOverlap_+3A_db">db</code></td>
<td>
<p>data frame allele frequency database.</p>
</td></tr>
<tr><td><code id="calculateOverlap_+3A_penalty">penalty</code></td>
<td>
<p>vector with factors for reducing the impact from distant dye channels.
NB! Length must equal number of dyes in kit minus one.</p>
</td></tr>
<tr><td><code id="calculateOverlap_+3A_virtual">virtual</code></td>
<td>
<p>logical default is TRUE meaning that overlap calculation includes virtual bins.</p>
</td></tr>
<tr><td><code id="calculateOverlap_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By analyzing the bins overlap between dye channels a measure of
the risk for spectral pull-up artefacts can be obtain. The default result
is a matrix with the total bins overlap in number of base pairs. If an allele
frequency database is provided the overlap at each bin is multiplied with the
frequency of the corresponding allele. If no frequence exist for that allele
a frequency of 5/2N will be used. X and Y alleles is given the frequency 1.
A penalty matrix can be supplied to reduce the effect by spectral distance,
meaning that overlap with the neighboring dye can be counted in full (100
while a non neighbor dye get its overlap reduced (to e.g. 10
</p>


<h3>Value</h3>

<p>data.frame with columns 'Kit', 'Color', [dyes], 'Sum', and 'Score'.
</p>

<hr>
<h2 id='calculateOverlap_gui'>Calculate Bins Overlap</h2><span id='topic+calculateOverlap_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateOverlap">calculateOverlap</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateOverlap_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = TRUE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateOverlap_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateOverlap_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateOverlap_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateOverlap_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By analysis of the bins overlap between dye channels a measure of
the risk for spectral pull-up artefacts can be obtain. The default result
is a matrix with the total bins overlap in number of base pairs. If an allele
frequency database is provided the overlap at each bin is multiplied with the
frequency of the corresponding allele. If no frequence exist for that allele
a frequency of 5/2N will be used. X and Y alleles is given the frequency 1.
A scoring matrix can be supplied to reduce the effect by spectral distance,
meaning that overlap with the neighboring dye can be counted in full (100
while a non neighbor dye get its overlap reduced (to e.g. 10
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateOverlap">calculateOverlap</a></code>
</p>

<hr>
<h2 id='calculatePeaks'>Calculate Peaks</h2><span id='topic+calculatePeaks'></span>

<h3>Description</h3>

<p>Calculates the number of peaks in samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculatePeaks(
  data,
  bins = c(0, 2, 3),
  labels = NULL,
  ol.rm = FALSE,
  by.marker = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculatePeaks_+3A_data">data</code></td>
<td>
<p>data frame containing at least the columns
'Sample.Name' and 'Height'.</p>
</td></tr>
<tr><td><code id="calculatePeaks_+3A_bins">bins</code></td>
<td>
<p>numeric vector containing the cut-off points defined as
maximum number of peaks for all but the last label, which is anything
above final cut-off. Must be sorted in ascending order.</p>
</td></tr>
<tr><td><code id="calculatePeaks_+3A_labels">labels</code></td>
<td>
<p>character vector defining the group labels.
Length must be equal to number of bins + one label for anything above the
final cut-off.</p>
</td></tr>
<tr><td><code id="calculatePeaks_+3A_ol.rm">ol.rm</code></td>
<td>
<p>logical if TRUE, off-ladder alleles 'OL' peaks will be discarded.
if FALSE, all peaks will be included in the calculations.</p>
</td></tr>
<tr><td><code id="calculatePeaks_+3A_by.marker">by.marker</code></td>
<td>
<p>logical if TRUE, peaks will counted per marker.
if FALSE, peaks will counted per sample.</p>
</td></tr>
<tr><td><code id="calculatePeaks_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Count the number of peaks in a sample profile based on values in the
'Height' column. Each sample is labeled according to custom labels
defined by the number of peaks. Peaks can be counted by sample or by
marker within a sample.
There is an option to discard off-ladder peaks ('OL').
The default purpose for this function is to categorize contamination in
negative controls, but it can be used to simply calculating the number of
peaks in any sample.
NB! A column 'Peaks' for the number of peaks will be created.
If present it will be overwritten.
NB! A column 'Group' for the sample group will be created.
If present it will be overwritten.
NB! A column 'Id' will be created by combining the content in the
'Sample.Name' and 'File' column (if available).
The unique entries in the 'Id' column will be the definition of a unique sample.
If 'File' is present this allows for identical sample names in different
batches (files) to be identified as unique samples.
If 'Id' is present it will be overwritten.
</p>


<h3>Value</h3>

<p>data.frame with with additional columns 'Peaks', 'Group', and 'Id'.
</p>

<hr>
<h2 id='calculatePeaks_gui'>Calculate Peaks</h2><span id='topic+calculatePeaks_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculatePeaks">calculatePeaks</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculatePeaks_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculatePeaks_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="calculatePeaks_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculatePeaks_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculatePeaks_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Counts the number of peaks in samples and markers with option to
discard off-ladder peaks and to label groups according to maximum number
of peaks.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculatePeaks">calculatePeaks</a></code>
</p>

<hr>
<h2 id='calculatePullup'>Calculate Spectral Pull-up</h2><span id='topic+calculatePullup'></span>

<h3>Description</h3>

<p>Calculates possible pull-up peaks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculatePullup(
  data,
  ref,
  pullup.range = 6,
  block.range = 12,
  ol.rm = FALSE,
  ignore.case = TRUE,
  word = FALSE,
  discard = FALSE,
  limit = 1,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculatePullup_+3A_data">data</code></td>
<td>
<p>a data frame containing at least 'Sample.Name', 'Marker', 'Height',
'Allele', 'Dye', 'Data.Point' and 'Size'.</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_ref">ref</code></td>
<td>
<p>a data frame containing at least
'Sample.Name', 'Marker', 'Allele'.</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_pullup.range">pullup.range</code></td>
<td>
<p>numeric to set the analysis window to look for pull-up
peaks (known allele data point +- pullup.range/2)</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_block.range">block.range</code></td>
<td>
<p>numeric to set blocking range to check for known allele overlap
(known allele data point +- block.range/2).</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_ol.rm">ol.rm</code></td>
<td>
<p>logical TRUE if off-ladder peaks should be excluded from analysis.
Default is FALSE to include off-ladder peaks.</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical indicating if sample matching should ignore case.</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_word">word</code></td>
<td>
<p>logical indicating if word boundaries should be added before sample matching.</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_discard">discard</code></td>
<td>
<p>logical TRUE if known alleles with no detected pull-up should
be discarded from the result. Default is FALSE to include alleles not causing pull-up.</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_limit">limit</code></td>
<td>
<p>numeric remove ratios &gt; limit from the result. Default is 1 to remove
pull-up peaks that are higher than the source peak and hence likely not a real pull-up.</p>
</td></tr>
<tr><td><code id="calculatePullup_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates possible pull-up (aka. bleed-through) peaks in a dataset.
Known alleles are identified and the analysis window range is marked.
If the blocking range of known alleles overlap, they are excluded from the
analysis. Pull-up peaks within the data point analysis window, around
known alleles, are identified, the data point difference, and the ratio
is calculated.
Off-ladder ('OL') alleles are included by default but can be excluded.
All known peaks included in the analysis are by default written to the
result even if they did not cause any pull-up. These rows can be discarded
from the result.
</p>


<h3>Value</h3>

<p>data.frame with with columns 'Sample.Name', 'Marker', 'Dye',
'Allele', 'Height', 'Size', 'Data.Point', 'P.Marker', 'P.Dye', 'P.Allele',
'P.Height', 'P.Size', 'P.Data.Point', 'Delta', 'Ratio'.
</p>

<hr>
<h2 id='calculatePullup_gui'>Calculate Spectral Pull-up</h2><span id='topic+calculatePullup_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculatePullup">calculatePullup</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculatePullup_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculatePullup_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculatePullup_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculatePullup_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculatePullup_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculatePullup">calculatePullup</a></code> function by
providing a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculatePullup">calculatePullup</a></code>, <code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='calculateRatio'>Calculate Ratio</h2><span id='topic+calculateRatio'></span>

<h3>Description</h3>

<p>Calculates the peak height ratio between specified loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateRatio(
  data,
  ref = NULL,
  numerator = NULL,
  denominator = NULL,
  group = NULL,
  ol.rm = TRUE,
  ignore.case = TRUE,
  word = FALSE,
  exact = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateRatio_+3A_data">data</code></td>
<td>
<p>a data frame containing at least
'Sample.Name', 'Marker', 'Height', 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_ref">ref</code></td>
<td>
<p>a data frame containing at least 'Sample.Name', 'Marker', 'Allele'.
If provided alleles matching 'ref' will be extracted from 'data'
(see <code><a href="#topic+filterProfile">filterProfile</a></code>).</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_numerator">numerator</code></td>
<td>
<p>character vector with marker names.</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_denominator">denominator</code></td>
<td>
<p>character vector with marker names.</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_group">group</code></td>
<td>
<p>character column name to group by.</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_ol.rm">ol.rm</code></td>
<td>
<p>logical indicating if off-ladder 'OL' alleles should be removed.</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical indicating if sample matching should ignore case.</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_word">word</code></td>
<td>
<p>logical indicating if word boundaries should be added before sample matching.</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_exact">exact</code></td>
<td>
<p>logical indicating if exact sample matching should be used.</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Default is to calculate the ratio between all unique pairwise
combinations of markers/loci. If equal number of markers are provided in
the numerator and the denominator the provided pairwise ratios will be
calculated. If markers are provided in only the numerator or only the
denominator the ratio of all possible combinations of the provided markers
and the markers not provided will be calculated. If the number of markers
provided are different in the numerator and in the denominator the shorter
vector will be repeated to equal the longer vector in length.
Data can be unfiltered or filtered since the sum of
peak heights per marker is used. Off-ladder alleles is by default removed
from the dataset before calculations.
</p>


<h3>Value</h3>

<p>data.frame with with columns 'Sample.Name', 'Marker', 'Delta', 'Hb', 'Lb', 'MPH', 'TPH'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(set2)
# Calculate ratio between the shortest and longest marker in each dye.
numerator &lt;- c("D3S1358", "AMEL", "D19S433")
denominator &lt;- c("D2S1338", "D18S51", "FGA")
calculateRatio(data = set2, numerator = numerator, denominator = denominator)
calculateRatio(data = set2, numerator = NULL, denominator = "AMEL")
calculateRatio(data = set2, numerator = c("AMEL", "TH01"), denominator = NULL)
calculateRatio(data = set2, numerator = NULL, denominator = NULL)
</code></pre>

<hr>
<h2 id='calculateRatio_gui'>Calculate Ratio</h2><span id='topic+calculateRatio_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateRatio">calculateRatio</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateRatio_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateRatio_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateRatio_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateRatio_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateRatio_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateRatio">calculateRatio</a></code> function
by providing a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code>link{calculateRatio}</code>, <code>link{checkSubset}</code>
</p>

<hr>
<h2 id='calculateResultType'>Calculate Result Type</h2><span id='topic+calculateResultType'></span>

<h3>Description</h3>

<p>Calculate the result type for samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateResultType(
  data,
  kit = NULL,
  add.missing.marker = TRUE,
  threshold = NULL,
  mixture.limits = NULL,
  partial.limits = NULL,
  subset.name = NA,
  marker.subset = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateResultType_+3A_data">data</code></td>
<td>
<p>a data frame containing at least the column 'Sample.Name'.</p>
</td></tr>
<tr><td><code id="calculateResultType_+3A_kit">kit</code></td>
<td>
<p>character string or integer defining the kit.</p>
</td></tr>
<tr><td><code id="calculateResultType_+3A_add.missing.marker">add.missing.marker</code></td>
<td>
<p>logical, default is TRUE which adds missing markers.</p>
</td></tr>
<tr><td><code id="calculateResultType_+3A_threshold">threshold</code></td>
<td>
<p>integer indicating the dropout threshold.</p>
</td></tr>
<tr><td><code id="calculateResultType_+3A_mixture.limits">mixture.limits</code></td>
<td>
<p>integer or vector indicating subtypes of 'Mixture'.</p>
</td></tr>
<tr><td><code id="calculateResultType_+3A_partial.limits">partial.limits</code></td>
<td>
<p>integer or vector indicating subtypes of 'Partial'.</p>
</td></tr>
<tr><td><code id="calculateResultType_+3A_subset.name">subset.name</code></td>
<td>
<p>string naming the subset of 'Complete'.</p>
</td></tr>
<tr><td><code id="calculateResultType_+3A_marker.subset">marker.subset</code></td>
<td>
<p>string with marker names defining the subset of 'Complete'.</p>
</td></tr>
<tr><td><code id="calculateResultType_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates result types for samples in 'data'.
Defined types are: 'No result', 'Mixture', 'Partial', and 'Complete'.
Subtypes can be defined by parameters.
An integer passed to 'threshold' defines a subtype of 'Complete' &quot;Complete profile all peaks &gt;threshold&quot;.
An integer or vector passed to 'mixture.limits' define subtypes of 'Mixture' &quot;&gt; [mixture.limits] markers&quot;.
An integer or vector passed to 'partial.limits' define subtypes of 'Partial' &quot;&gt; [partial.limits] peaks&quot;.
A string with marker names separated by pipe (|) passed to 'marker.subset' and
a string 'subset.name' defines a subtype of 'Partial' &quot;Complete [subset.name]&quot;.
</p>


<h3>Value</h3>

<p>data.frame with columns 'Sample.Name','Type', and 'Subtype'.
</p>

<hr>
<h2 id='calculateResultType_gui'>Calculate Result Type</h2><span id='topic+calculateResultType_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateResultType">calculateResultType</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateResultType_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateResultType_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateResultType_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateResultType_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateResultType_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of <code><a href="#topic+calculateResultType">calculateResultType</a></code> by providing a
graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateResultType">calculateResultType</a></code>
</p>

<hr>
<h2 id='calculateSlope'>Calculate Profile Slope.</h2><span id='topic+calculateSlope'></span>

<h3>Description</h3>

<p>Calculate profile slope for samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateSlope(data, ref, conf = 0.975, kit = NULL, debug = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateSlope_+3A_data">data</code></td>
<td>
<p>data.frame with at least columns 'Sample.Name', 'Marker', and
'Height'.</p>
</td></tr>
<tr><td><code id="calculateSlope_+3A_ref">ref</code></td>
<td>
<p>data.frame with at least columns 'Sample.Name', 'Marker',
and 'Allele'</p>
</td></tr>
<tr><td><code id="calculateSlope_+3A_conf">conf</code></td>
<td>
<p>numeric confidence limit to calculate a confidence interval from
(Student t Distribution with 'Peaks'-2 degree of freedom). Default is 0.975
corresponding to a 95% confidence interval.</p>
</td></tr>
<tr><td><code id="calculateSlope_+3A_kit">kit</code></td>
<td>
<p>character string or vector specifying the analysis kits used
to produce the data. If length(kit) != number of groups, kit[1] will be
used for all groups.</p>
</td></tr>
<tr><td><code id="calculateSlope_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateSlope_+3A_...">...</code></td>
<td>
<p>additional arguments to the <code><a href="#topic+filterProfile">filterProfile</a></code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates the profile slope for each sample. The slope is calculated as a
linear model specified by the response (natural logarithm of peak height) by
the term size (in base pair). If 'Size' is not present in the dataset, one or
multiple kit names can be given as argument 'kit'. The specified kits will
be used to estimate the size of each allele. If 'kit' is NULL the kit(s)
will be automatically detected, and the 'Size' will be calculated.
</p>
<p>The column 'Group' can be used to separate datasets to be compared, and if
so 'kit' must be a vector of equal length as the number of groups, and in
the same order. If not the first 'kit' will be recycled for all groups.
</p>
<p>Data will be filtered using the reference profiles.
</p>


<h3>Value</h3>

<p>data.frame with with columns 'Sample.Name', 'Kit', 'Group', 'Slope',
'Error', 'Peaks', 'Lower', and 'Upper'.
</p>

<hr>
<h2 id='calculateSlope_gui'>Calculate Profile Slope</h2><span id='topic+calculateSlope_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateSlope">calculateSlope</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateSlope_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateSlope_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="calculateSlope_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateSlope_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateSlope_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateSlope">calculateSlope</a></code> function
by providing a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateSlope">calculateSlope</a></code>
</p>

<hr>
<h2 id='calculateSpike'>Detect Spike</h2><span id='topic+calculateSpike'></span>

<h3>Description</h3>

<p>Detect samples with possible spikes in the DNA profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateSpike(
  data,
  threshold = NULL,
  tolerance = 2,
  kit = NULL,
  quick = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateSpike_+3A_data">data</code></td>
<td>
<p>data.frame with including columns 'Sample.Name', 'Marker', 'Size'.</p>
</td></tr>
<tr><td><code id="calculateSpike_+3A_threshold">threshold</code></td>
<td>
<p>numeric number of peaks of similar size in different dye
channels to pass as a possible spike (NULL = number of dye channels
minus one to allow for one unlabeled peak).</p>
</td></tr>
<tr><td><code id="calculateSpike_+3A_tolerance">tolerance</code></td>
<td>
<p>numeric tolerance for Size. For the quick and dirty
rounding method e.g. 1.5 rounds Size to +/- 0.75 bp. For the slower but
more accurate method the value is the maximum allowed difference between
peaks in a spike.</p>
</td></tr>
<tr><td><code id="calculateSpike_+3A_kit">kit</code></td>
<td>
<p>string or numeric for the STR-kit used (NULL = auto detect).</p>
</td></tr>
<tr><td><code id="calculateSpike_+3A_quick">quick</code></td>
<td>
<p>logical TRUE for the quick and dirty method. Default is FALSE
which use a slower but more accurate method.</p>
</td></tr>
<tr><td><code id="calculateSpike_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a list of possible spikes by searching for peaks aligned
vertically (i.e. nearly identical size). There are two methods to search.
The default method (quick=FALSE) method that calculates the distance between
each peak in a sample, and the quick and dirty method (quick=TRUE) that
rounds the size and then group peaks with identical size. The rounding
method is faster because it uses the data.table package. The accurate method
is slower because it uses nested loops - the first through each sample to
calculate the distance between all peaks, and the second loops through the
distance matrix to identify which peaks lies within the tolerance.
NB! The quick method may not catch all spikes since two peaks can be
separated by rounding e.g. 200.5 and 200.6 becomes 200 and 201 respectively.
</p>


<h3>Value</h3>

<p>data.frame
</p>


<h3>See Also</h3>

<p><code><a href="data.table.html#topic+data.table">data.table</a></code>
</p>

<hr>
<h2 id='calculateSpike_gui'>Detect Spike</h2><span id='topic+calculateSpike_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateSpike">calculateSpike</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateSpike_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateSpike_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="calculateSpike_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateSpike_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateSpike_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateSpike">calculateSpike</a></code> function
by providing a graphical user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateSpike">calculateSpike</a></code>
</p>

<hr>
<h2 id='calculateStatistics'>Summary Statistics</h2><span id='topic+calculateStatistics'></span>

<h3>Description</h3>

<p>Calculate summary statistics for the selected target and scope.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateStatistics(
  data,
  target,
  quant = 0.95,
  group = NULL,
  count = NULL,
  decimals = -1,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateStatistics_+3A_data">data</code></td>
<td>
<p>data.frame containing the data of interest.</p>
</td></tr>
<tr><td><code id="calculateStatistics_+3A_target">target</code></td>
<td>
<p>character column to calculate summary statistics for.</p>
</td></tr>
<tr><td><code id="calculateStatistics_+3A_quant">quant</code></td>
<td>
<p>numeric quantile to calculate 0,1, default 0.95.</p>
</td></tr>
<tr><td><code id="calculateStatistics_+3A_group">group</code></td>
<td>
<p>character vector of column(s) to group by, if any.</p>
</td></tr>
<tr><td><code id="calculateStatistics_+3A_count">count</code></td>
<td>
<p>character column to count unique values in, if any.</p>
</td></tr>
<tr><td><code id="calculateStatistics_+3A_decimals">decimals</code></td>
<td>
<p>numeric number of decimals. Negative does not round.</p>
</td></tr>
<tr><td><code id="calculateStatistics_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculate summary statistics for the given target column ('X') across the
entire dataset or grouped by one or multiple columns, and counts the number
of unique values in the given count column ('Y'). Returns a data.frame
with the grouped columns, number of unique values 'Y.n', number of
observations 'X.n', the minimum value 'X.Min', the mean value 'X.Mean',
standard deviation 'X.Stdv', and the provided percentile 'X.Perc.##'.
For more details see <code>unique</code>, <code>min</code>, <code>mean</code>, <code>sd</code>,
<code>quantile</code>.
</p>


<h3>Value</h3>

<p>data.frame with summary statistics.
</p>

<hr>
<h2 id='calculateStatistics_gui'>Calculate Statistics</h2><span id='topic+calculateStatistics_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateStatistics">calculateStatistics</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateStatistics_gui(
  data = NULL,
  target = NULL,
  quant = 0.95,
  group = NULL,
  count = NULL,
  decimals = 4,
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateStatistics_gui_+3A_data">data</code></td>
<td>
<p>character preselected data.frame if provided and exist in environment.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_target">target</code></td>
<td>
<p>character vector preselected target column.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_quant">quant</code></td>
<td>
<p>numeric quantile to calculate. Default=0.95.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_group">group</code></td>
<td>
<p>character vector preselected column(s) to group by.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_count">count</code></td>
<td>
<p>character vector preselected column to count unique values in.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_decimals">decimals</code></td>
<td>
<p>numeric number of decimals. Negative does not round.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateStatistics_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateStatistics">calculateStatistics</a></code> function
by providing a graphical user interface. Preselected values can be provided
as arguments.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code>link{quantile}</code>, <code>link{min}</code>, <code>link{max}</code>, <code>link{mean}</code>, <code>link{sd}</code>
</p>

<hr>
<h2 id='calculateStutter'>Calculate Stutter</h2><span id='topic+calculateStutter'></span>

<h3>Description</h3>

<p>Calculate statistics for stutters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateStutter(
  data,
  ref,
  back = 2,
  forward = 1,
  interference = 0,
  replace.val = NULL,
  by.val = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateStutter_+3A_data">data</code></td>
<td>
<p>data frame with genotype data.
Requires columns 'Sample.Name', 'Marker', 'Allele', 'Height'.</p>
</td></tr>
<tr><td><code id="calculateStutter_+3A_ref">ref</code></td>
<td>
<p>data frame with the known profiles.
Requires columns 'Sample.Name', 'Marker', 'Allele'.</p>
</td></tr>
<tr><td><code id="calculateStutter_+3A_back">back</code></td>
<td>
<p>integer for the maximal number of backward stutters
(max size difference 2 = n-2 repeats).</p>
</td></tr>
<tr><td><code id="calculateStutter_+3A_forward">forward</code></td>
<td>
<p>integer for the maximal number of forward stutters
(max size difference 1 = n+1 repeats).</p>
</td></tr>
<tr><td><code id="calculateStutter_+3A_interference">interference</code></td>
<td>
<p>integer specifying accepted level of allowed overlap.</p>
</td></tr>
<tr><td><code id="calculateStutter_+3A_replace.val">replace.val</code></td>
<td>
<p>numeric vector with 'false' stutters to replace.</p>
</td></tr>
<tr><td><code id="calculateStutter_+3A_by.val">by.val</code></td>
<td>
<p>numeric vector with correct stutters.</p>
</td></tr>
<tr><td><code id="calculateStutter_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates stutter ratios based on the 'reference' data set
and a defined analysis range around the true allele.
</p>
<p>NB! Off-ladder alleles ('OL') is NOT included in the analysis.
NB! Labeled pull-ups or artefacts within stutter range IS included
in the analysis.
</p>
<p>There are three levels of allowed overlap (interference).
0 = no interference (default): calculate the ratio for a stutter only if
there are no overlap between the stutter or its allele with the analysis
range of another allele.
1 = stutter-stutter interference: calculate the ratio for a stutter even
if the stutter or its allele overlap with a stutter within the analysis
range of another allele.
2 = stutter-allele interference: calculate the ratio for a stutter even if
the stutter and its allele overlap with the analysis range of another allele.
</p>


<h3>Value</h3>

<p>data.frame with extracted result.
</p>

<hr>
<h2 id='calculateStutter_gui'>Calculate Stutter</h2><span id='topic+calculateStutter_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+calculateStutter">calculateStutter</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateStutter_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateStutter_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="calculateStutter_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="calculateStutter_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="calculateStutter_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+calculateStutter">calculateStutter</a></code> function by providing
a graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateStutter">calculateStutter</a></code>, <code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='calculateT'>Calculate Stochastic Threshold</h2><span id='topic+calculateT'></span>

<h3>Description</h3>

<p>Calculates point estimates for the stochastic threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateT(
  data,
  log.model = FALSE,
  p.dropout = 0.01,
  pred.int = 0.95,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateT_+3A_data">data</code></td>
<td>
<p>data.frame with dependent and explanatory values in columns named 'Dep' and 'Exp'.</p>
</td></tr>
<tr><td><code id="calculateT_+3A_log.model">log.model</code></td>
<td>
<p>logical indicating if data should be log transformed. Default=FALSE.</p>
</td></tr>
<tr><td><code id="calculateT_+3A_p.dropout">p.dropout</code></td>
<td>
<p>numeric accepted risk to calculate point estimate for. Default=0.01.</p>
</td></tr>
<tr><td><code id="calculateT_+3A_pred.int">pred.int</code></td>
<td>
<p>numeric prediction interval. Default=0.95.</p>
</td></tr>
<tr><td><code id="calculateT_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a data.frame with observed values for the dependent variable
(column 'Dep') and explanatory values (column 'Exp') point estimates
corresponding to a risk level of <code>p.dropout</code> are calculated
using logistic regression: <code>glm(Dep~Exp, family=binomial("logit")</code>.
A conservative estimate is calculated from the <code>pred.int</code>.
In addition the model parameters B0 (intercept) and B1 (slope),
Hosmer-Lemeshow test statistic (p-value), and the number of observed
and dropped out alleles is returned.
</p>


<h3>Value</h3>

<p>vector with named parameters
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateDropout">calculateDropout</a></code>, <code><a href="#topic+calculateAllT">calculateAllT</a></code>,
<code><a href="#topic+modelDropout_gui">modelDropout_gui</a></code>, <code><a href="#topic+plotDropout_gui">plotDropout_gui</a></code>
</p>

<hr>
<h2 id='checkDataset'>Check Dataset</h2><span id='topic+checkDataset'></span>

<h3>Description</h3>

<p>Check a data.frame before analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkDataset(
  name,
  reqcol = NULL,
  slim = FALSE,
  slimcol = NULL,
  string = NULL,
  stringcol = NULL,
  env = parent.frame(),
  parent = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkDataset_+3A_name">name</code></td>
<td>
<p>character name of data.frame.</p>
</td></tr>
<tr><td><code id="checkDataset_+3A_reqcol">reqcol</code></td>
<td>
<p>character vector with required column names.</p>
</td></tr>
<tr><td><code id="checkDataset_+3A_slim">slim</code></td>
<td>
<p>logical TRUE to check if 'slim' data.</p>
</td></tr>
<tr><td><code id="checkDataset_+3A_slimcol">slimcol</code></td>
<td>
<p>character vector with column names to check if 'slim' data.</p>
</td></tr>
<tr><td><code id="checkDataset_+3A_string">string</code></td>
<td>
<p>character vector with invalid strings in 'stringcol', return FALSE if found.</p>
</td></tr>
<tr><td><code id="checkDataset_+3A_stringcol">stringcol</code></td>
<td>
<p>character vector with column names to check for 'string'.</p>
</td></tr>
<tr><td><code id="checkDataset_+3A_env">env</code></td>
<td>
<p>environment where to look for the data frame.</p>
</td></tr>
<tr><td><code id="checkDataset_+3A_parent">parent</code></td>
<td>
<p>parent gWidget.</p>
</td></tr>
<tr><td><code id="checkDataset_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Check that the object exist, there are rows, the required columns exist,
if data.frame is 'fat', and if invalid strings exist. Show error message if not.
</p>

<hr>
<h2 id='checkSubset'>Check Subset</h2><span id='topic+checkSubset'></span>

<h3>Description</h3>

<p>Check the result of subsetting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkSubset(
  data,
  ref,
  console = TRUE,
  ignore.case = TRUE,
  word = FALSE,
  exact = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkSubset_+3A_data">data</code></td>
<td>
<p>a data frame in GeneMapper format containing column 'Sample.Name'.</p>
</td></tr>
<tr><td><code id="checkSubset_+3A_ref">ref</code></td>
<td>
<p>a data frame in GeneMapper format containing column 'Sample.Name',
OR an atomic vector e.g. a single sample name string.</p>
</td></tr>
<tr><td><code id="checkSubset_+3A_console">console</code></td>
<td>
<p>logical, if TRUE result is printed to R console,
if FALSE a string is returned.</p>
</td></tr>
<tr><td><code id="checkSubset_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical, if TRUE case insensitive matching is used.</p>
</td></tr>
<tr><td><code id="checkSubset_+3A_word">word</code></td>
<td>
<p>logical, if TRUE only word matching (regex).</p>
</td></tr>
<tr><td><code id="checkSubset_+3A_exact">exact</code></td>
<td>
<p>logical, if TRUE only exact match.</p>
</td></tr>
<tr><td><code id="checkSubset_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Check if ref and sample names are unique for subsetting.
Prints the result to the R-prompt.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+grep">grep</a></code>
</p>

<hr>
<h2 id='checkSubset_gui'>Check Subset</h2><span id='topic+checkSubset_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+checkSubset">checkSubset</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkSubset_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkSubset_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="checkSubset_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="checkSubset_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="checkSubset_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+checkSubset">checkSubset</a></code> function by providing
a graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='colConvert'>Convert Columns</h2><span id='topic+colConvert'></span>

<h3>Description</h3>

<p>Internal helper function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colConvert(
  data,
  columns = "Height|Size|Data.Point",
  ignore.case = TRUE,
  fixed = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colConvert_+3A_data">data</code></td>
<td>
<p>data.frame.</p>
</td></tr>
<tr><td><code id="colConvert_+3A_columns">columns</code></td>
<td>
<p>character string containing a regular expression
(or character string for fixed = TRUE) to be matched in the given
character vector (separate multiple column names by | in reg.exp).</p>
</td></tr>
<tr><td><code id="colConvert_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical TRUE to ignore case in matching.</p>
</td></tr>
<tr><td><code id="colConvert_+3A_fixed">fixed</code></td>
<td>
<p>logical TRUE if columns is a string to be matched as is.</p>
</td></tr>
<tr><td><code id="colConvert_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes a data frame as input and return it after converting known numeric
columns to numeric.
</p>


<h3>Value</h3>

<p>data.frame.
</p>

<hr>
<h2 id='colNames'>Column Names</h2><span id='topic+colNames'></span>

<h3>Description</h3>

<p>Internal helper function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colNames(data, slim = TRUE, concatenate = NULL, numbered = TRUE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colNames_+3A_data">data</code></td>
<td>
<p>data.frame.</p>
</td></tr>
<tr><td><code id="colNames_+3A_slim">slim</code></td>
<td>
<p>logical, TRUE returns column names occurring once,
FALSE returns column names occurring multiple times.</p>
</td></tr>
<tr><td><code id="colNames_+3A_concatenate">concatenate</code></td>
<td>
<p>string, if not NULL returns a single string with column
names concatenated by the provided string instead of a vector.</p>
</td></tr>
<tr><td><code id="colNames_+3A_numbered">numbered</code></td>
<td>
<p>logical indicating if repeated column names must have a number suffix.</p>
</td></tr>
<tr><td><code id="colNames_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes a data frame as input and return either column names
occurring once or multiple times. Matching is done by the 'base name'
(the substring to the left of the last period, if any). The return type
is a string vector by default, or a single string of column names separated
by a string 'concatenate' (see 'collapse' in <code>paste</code> for details).
There is an option to limit multiple names to those with a number suffix.
</p>


<h3>Value</h3>

<p>character, vector or string.
</p>

<hr>
<h2 id='columns'>Column Actions</h2><span id='topic+columns'></span>

<h3>Description</h3>

<p>Perform actions on columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>columns(
  data,
  col1 = NA,
  col2 = NA,
  operator = "&amp;",
  fixed = NA,
  target = NA,
  start = 1,
  stop = 1,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="columns_+3A_data">data</code></td>
<td>
<p>a data frame.</p>
</td></tr>
<tr><td><code id="columns_+3A_col1">col1</code></td>
<td>
<p>character column name to perform action on.</p>
</td></tr>
<tr><td><code id="columns_+3A_col2">col2</code></td>
<td>
<p>character optional second column name to perform action on.</p>
</td></tr>
<tr><td><code id="columns_+3A_operator">operator</code></td>
<td>
<p>character to indicate operator: '&amp;' concatenate, '+' add,
'*' multiply, '-' subtract, '/' divide, 'substr' extract a substring.</p>
</td></tr>
<tr><td><code id="columns_+3A_fixed">fixed</code></td>
<td>
<p>character or numeric providing the second operand if 'col2'
is not used.</p>
</td></tr>
<tr><td><code id="columns_+3A_target">target</code></td>
<td>
<p>character to specify column name for result. Default is to
overwrite 'col1'. If not present it will be added.</p>
</td></tr>
<tr><td><code id="columns_+3A_start">start</code></td>
<td>
<p>integer, the first position to be extracted.</p>
</td></tr>
<tr><td><code id="columns_+3A_stop">stop</code></td>
<td>
<p>integer, the last position to be extracted.</p>
</td></tr>
<tr><td><code id="columns_+3A_debug">debug</code></td>
<td>
<p>logical to indicate if debug information should be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Perform actions on columns in a data frame. There are five actions:
concatenate, add, multiply, subtract, divide. The selected action can be
performed on two columns, or one column and a fixed value, or a new column
can be added. A target column for the result is specified.
NB! if the target column already exist it will be overwritten,
else it will be created. A common use is to create a unique
Sample.Name from the existing Sample.Name column and e.g. the File.Name
or File.Time columns. It can also be used to calculate the Amount from
the Concentration.
</p>


<h3>Value</h3>

<p>data frame.
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+substr">substr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get a sample dataset.
data(set2)
# Add concatenate Sample.Name and Dye.
set2 &lt;- columns(data = set2, col1 = "Sample.Name", col2 = "Dye")
# Multiply Height by 4.
set2 &lt;- columns(data = set2, col1 = "Height", operator = "*", fixed = 4)
# Add a new column.
set2 &lt;- columns(data = set2, operator = "&amp;", fixed = "1234", target = "Batch")
</code></pre>

<hr>
<h2 id='columns_gui'>Column Actions</h2><span id='topic+columns_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+columns">columns</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>columns_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="columns_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="columns_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="columns_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="columns_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+columns">columns</a></code> function by providing a
graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='combine_gui'>Combine Datasets</h2><span id='topic+combine_gui'></span>

<h3>Description</h3>

<p>GUI for combining two datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_gui(env = parent.frame(), debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="combine_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="combine_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simple GUI to combine two datasets using the <code><a href="plyr.html#topic+rbind.fill">rbind.fill</a></code>
function.
NB! Datasets must have identical column names but not necessarily
in the same order.
</p>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='combineBinsAndPanels'>Combine Bins And Panels Files.</h2><span id='topic+combineBinsAndPanels'></span>

<h3>Description</h3>

<p>Combines useful information into one dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combineBinsAndPanels(bin, panel)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combineBinsAndPanels_+3A_bin">bin</code></td>
<td>
<p>data frame created from the 'bins' file.</p>
</td></tr>
<tr><td><code id="combineBinsAndPanels_+3A_panel">panel</code></td>
<td>
<p>data frame created from the 'panels' file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Combines information from two sources ('Bins' and 'Panels' file) to create
a dataset containing information about panel name, marker name, alleles in the
allelic ladder, their size and size range, a flag indicating virtual alleles,
fluorophore color, repeat size, marker range. The short name, full name,
and sex marker flag is populated through the <code>makeKit_gui</code> user interface.
In addition the function calculates an estimated offset for each marker,
which can be used for creating EPG like plots.
Note: offset is estimated by taking the smallest physical ladder fragment
e.g. 98.28 for D3 in ESX17. Round this to an integer (98) and finally
subtract the number of base pair for that repeat i.e. 4*9=36,
which gives an offset of 98-36 = 62 bp.
Microvariants are handled by taking the decimal part multiplied with 10
and adding this to the number of base pair e.g. 9.3 = 4*9 + 0.3*10 = 39 bp.
</p>


<h3>Value</h3>

<p>data frame containing columns 'Panel', 'Marker', 'Allele',
'Size', 'Size.Min', 'Size.Max', 'Virtual', 'Color', 'Repeat',
'Marker.Min', 'Marker.Max', 'Offset', 'Short.Name', 'Full.Name'
</p>

<hr>
<h2 id='cropData_gui'>Crop Or Replace</h2><span id='topic+cropData_gui'></span>

<h3>Description</h3>

<p>GUI simplifying cropping and replacing values in data frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cropData_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cropData_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="cropData_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="cropData_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="cropData_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select a data frame from the drop-down and a target column.
To remove rows with 'NA' check the appropriate box.
Select to discard or replace values and additional options.
Click button to 'Apply' changes.
Multiple actions can be performed on one dataset before saving as
a new dataframe.
NB! Check that data type is correct before click apply to avoid strange behavior.
If data type is numeric any string will become a numeric 'NA'.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trim_gui">trim_gui</a></code>, <code><a href="#topic+editData_gui">editData_gui</a></code>, <code><a href="#topic+combine_gui">combine_gui</a></code>
</p>

<hr>
<h2 id='detectKit'>Detect Kit</h2><span id='topic+detectKit'></span>

<h3>Description</h3>

<p>Finds the most likely STR kit for a dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detectKit(data, index = FALSE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="detectKit_+3A_data">data</code></td>
<td>
<p>data frame with column 'Marker' or vector with marker names.</p>
</td></tr>
<tr><td><code id="detectKit_+3A_index">index</code></td>
<td>
<p>logical, returns kit index if TRUE or short name if FALSE.</p>
</td></tr>
<tr><td><code id="detectKit_+3A_debug">debug</code></td>
<td>
<p>logical, prints debug information if TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function first check if there is a 'kit' attribute for the dataset.
If there was a 'kit' attribute, and a match is found in <code>getKit</code>
the corresponding kit or index is returned.
If an attribute does not exist the function looks at the markers
in the dataset and returns the most likely kit(s).
</p>


<h3>Value</h3>

<p>integer or string indicating the detected kit.
</p>

<hr>
<h2 id='editData_gui'>Edit or View Data Frames</h2><span id='topic+editData_gui'></span>

<h3>Description</h3>

<p>GUI to edit and view data frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>editData_gui(
  env = parent.frame(),
  savegui = NULL,
  data = NULL,
  name = NULL,
  edit = TRUE,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="editData_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="editData_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="editData_gui_+3A_data">data</code></td>
<td>
<p>data.frame for instant viewing.</p>
</td></tr>
<tr><td><code id="editData_gui_+3A_name">name</code></td>
<td>
<p>character string with the name of the provided dataset.</p>
</td></tr>
<tr><td><code id="editData_gui_+3A_edit">edit</code></td>
<td>
<p>logical TRUE to enable edit (uses <code>gdf</code>), FALSE to
view and enable sorting by clicking a column header (uses <code><a href="gtable.html#topic+gtable">gtable</a></code>).</p>
</td></tr>
<tr><td><code id="editData_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="editData_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select a data frame from the drop-down to view or edit a dataset.
It is possible to save as a new dataframe. To enable sorting by clicking the
column headers the view mode must be used (i.e. edit = FALSE). There is an
option to limit the number of rows shown that can be used to preview large
datasets that may otherwise cause performance problems. Attributes of the
dataset can be views in a separate window.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trim_gui">trim_gui</a></code>, <code><a href="#topic+cropData_gui">cropData_gui</a></code>, <code><a href="#topic+combine_gui">combine_gui</a></code>
</p>

<hr>
<h2 id='export'>Export</h2><span id='topic+export'></span>

<h3>Description</h3>

<p>Exports or saves various objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export(
  object,
  name = NA,
  use.object.name = is.na(name),
  env = parent.frame(),
  path = NA,
  ext = "auto",
  delim = "\t",
  width = 3000,
  height = 2000,
  res = 250,
  overwrite = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_+3A_object">object</code></td>
<td>
<p>string, list or vector containing object names to be exported.</p>
</td></tr>
<tr><td><code id="export_+3A_name">name</code></td>
<td>
<p>string, list or vector containing file names.
Multiple names as string must be separated by pipe '|' or comma ','.
If not equal number of names as objects, first name will be used to construct names.</p>
</td></tr>
<tr><td><code id="export_+3A_use.object.name">use.object.name</code></td>
<td>
<p>logical, if TRUE file name will be the same as object name.</p>
</td></tr>
<tr><td><code id="export_+3A_env">env</code></td>
<td>
<p>environment where the objects exists.</p>
</td></tr>
<tr><td><code id="export_+3A_path">path</code></td>
<td>
<p>string specifying the destination folder exported objects.</p>
</td></tr>
<tr><td><code id="export_+3A_ext">ext</code></td>
<td>
<p>string specifying file extension.
Default is 'auto' for automatic .txt or .png based on object class.
If .RData all objects will be exported as .RData files.</p>
</td></tr>
<tr><td><code id="export_+3A_delim">delim</code></td>
<td>
<p>string specifying the delimiter used as separator.</p>
</td></tr>
<tr><td><code id="export_+3A_width">width</code></td>
<td>
<p>integer specifying the width of the image.</p>
</td></tr>
<tr><td><code id="export_+3A_height">height</code></td>
<td>
<p>integer specifying the height of the image.</p>
</td></tr>
<tr><td><code id="export_+3A_res">res</code></td>
<td>
<p>integer specifying the resolution of the image.</p>
</td></tr>
<tr><td><code id="export_+3A_overwrite">overwrite</code></td>
<td>
<p>logical, TRUE if existing files should be overwritten.</p>
</td></tr>
<tr><td><code id="export_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Export objects to a directory on the file system. Currently only objects
of class data.frames or ggplot are supported. data.frame objects will
be exported as '.txt' and ggplot objects as '.png'.
.RData applies to all supported object types.
</p>


<h3>Value</h3>

<p>NA if all objects were exported OR,
data.frame with columns 'Object', 'Name', and 'New.Name' with objects
that were not exported.
</p>

<hr>
<h2 id='export_gui'>Export</h2><span id='topic+export_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+export">export</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_gui(
  obj = listObjects(env = env, obj.class = c("data.frame", "ggplot")),
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_gui_+3A_obj">obj</code></td>
<td>
<p>character vector with object names.</p>
</td></tr>
<tr><td><code id="export_gui_+3A_env">env</code></td>
<td>
<p>environment where the objects exist.
Default is the current environment.</p>
</td></tr>
<tr><td><code id="export_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="export_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="export_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+export">export</a></code> function by providing a
graphical user interface to it. Currently all available objects provided
are selected by default.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+export">export</a></code>
</p>

<hr>
<h2 id='filterProfile'>Filter Profile</h2><span id='topic+filterProfile'></span>

<h3>Description</h3>

<p>Filter peaks from profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterProfile(
  data,
  ref = NULL,
  add.missing.loci = FALSE,
  keep.na = FALSE,
  ignore.case = TRUE,
  exact = FALSE,
  word = FALSE,
  invert = FALSE,
  sex.rm = FALSE,
  qs.rm = FALSE,
  kit = NULL,
  filter.allele = TRUE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterProfile_+3A_data">data</code></td>
<td>
<p>data frame with genotype data in 'slim' format.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_ref">ref</code></td>
<td>
<p>data frame with reference profile in 'slim' format.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_add.missing.loci">add.missing.loci</code></td>
<td>
<p>logical. TRUE add loci present in ref but not in data.
Overrides keep.na=FALSE.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_keep.na">keep.na</code></td>
<td>
<p>logical. FALSE discards NA alleles.
TRUE keep loci/sample even if no matching allele.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical TRUE ignore case.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_exact">exact</code></td>
<td>
<p>logical TRUE use exact matching of sample names.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_word">word</code></td>
<td>
<p>logical TRUE adds word boundaries when matching sample names.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_invert">invert</code></td>
<td>
<p>logical TRUE filter peaks NOT matching the reference.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_sex.rm">sex.rm</code></td>
<td>
<p>logical TRUE removes sex markers defined by 'kit'.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_qs.rm">qs.rm</code></td>
<td>
<p>logical TRUE removes quality sensors defined by 'kit'.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_kit">kit</code></td>
<td>
<p>character string defining the kit used.
If NULL automatic detection will be attempted.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_filter.allele">filter.allele</code></td>
<td>
<p>logical TRUE filter known alleles. FALSE increase the
performance if only sex markers or quality sensors should be removed.</p>
</td></tr>
<tr><td><code id="filterProfile_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Filters out the peaks matching (or not matching) specified known profiles
from typing data containing 'noise' such as stutters.
If 'ref' does not contain a 'Sample.Name' column it will be used
as reference for all samples in 'data'. The 'invert' option filters out
peaks NOT matching the reference (e.g. drop-in peaks). Sex markers and
quality sensors can be removed.
NB! add.missing.loci overrides keep.na.
Returns data where allele names match/not match 'ref' allele names.
Required columns are: 'Sample.Name', 'Marker', and 'Allele'.
</p>


<h3>Value</h3>

<p>data.frame with extracted result.
</p>

<hr>
<h2 id='filterProfile_gui'>Filter Profile</h2><span id='topic+filterProfile_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+filterProfile">filterProfile</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterProfile_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterProfile_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="filterProfile_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="filterProfile_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="filterProfile_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+filterProfile">filterProfile</a></code> function
by providing a graphical user interface to it.
All data not matching/matching the reference will be discarded.
Useful for filtering stutters and artifacts from raw typing data or
to identify drop-ins.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+filterProfile">filterProfile</a></code>, <code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='generateEPG'>Generate EPG</h2><span id='topic+generateEPG'></span>

<h3>Description</h3>

<p>Visualizes an EPG from DNA profiling data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateEPG(
  data,
  kit,
  title = NULL,
  wrap = TRUE,
  boxplot = FALSE,
  peaks = TRUE,
  collapse = TRUE,
  silent = FALSE,
  ignore.case = TRUE,
  at = 0,
  scale = "free",
  limit.x = TRUE,
  label.size = 3,
  label.angle = 0,
  label.vjust = 1,
  label.hjust = 0.5,
  expand = 0.1,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateEPG_+3A_data">data</code></td>
<td>
<p>data frame containing at least columns 'Sample.Name', 'Allele', and 'Marker'.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_kit">kit</code></td>
<td>
<p>string or integer representing the STR typing kit.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_title">title</code></td>
<td>
<p>string providing the title for the EPG.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_wrap">wrap</code></td>
<td>
<p>logical TRUE to wrap by dye.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_boxplot">boxplot</code></td>
<td>
<p>logical TRUE to plot distributions of peak heights as boxplots.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_peaks">peaks</code></td>
<td>
<p>logical TRUE to plot peaks for distributions using mean peak height.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_collapse">collapse</code></td>
<td>
<p>logical TRUE to add the peak heights of identical alleles peaks within each marker.
NB! Removes off-ladder alleles.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_silent">silent</code></td>
<td>
<p>logical FALSE to show plot.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical FALSE for case sensitive marker names.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_at">at</code></td>
<td>
<p>numeric analytical threshold (Height &lt;= at will not be plotted).</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_scale">scale</code></td>
<td>
<p>character &quot;free&quot; free x and y scale, alternatively &quot;free_y&quot; or &quot;free_x&quot;.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_limit.x">limit.x</code></td>
<td>
<p>logical TRUE to fix x-axis to size range.
To get a common x scale set scale=&quot;free_y&quot; and limit.x=TRUE.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_label.size">label.size</code></td>
<td>
<p>numeric for allele label text size.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_label.angle">label.angle</code></td>
<td>
<p>numeric for allele label print angle.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_label.vjust">label.vjust</code></td>
<td>
<p>numeric for vertical justification of allele labels.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_label.hjust">label.hjust</code></td>
<td>
<p>numeric for horizontal justification of allele labels.</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_expand">expand</code></td>
<td>
<p>numeric for plot are expansion (to avoid clipping of labels).</p>
</td></tr>
<tr><td><code id="generateEPG_+3A_debug">debug</code></td>
<td>
<p>logical for printing debug information to the console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generates a electropherogram like plot from 'data' and 'kit'.
If 'Size' is not present it is estimated from kit information and allele values.
If 'Height' is not present a default of 1000 RFU is used.
Off-ladder alleles can be plotted if 'Size' is provided.
There are various options to customize the plot scale and labels.
It is also possible to plot 'distributions' of peak heights as boxplots.
</p>


<h3>Value</h3>

<p>ggplot object.
</p>

<hr>
<h2 id='generateEPG_gui'>Generate EPG</h2><span id='topic+generateEPG_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+generateEPG">generateEPG</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateEPG_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateEPG_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="generateEPG_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="generateEPG_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="generateEPG_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+generateEPG">generateEPG</a></code> function by providing a graphical
user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generateEPG">generateEPG</a></code>
</p>

<hr>
<h2 id='getDb'>Get Allele Frequency Database</h2><span id='topic+getDb'></span>

<h3>Description</h3>

<p>Gives access to allele frequency databases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDb(db.name.or.index = NULL, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDb_+3A_db.name.or.index">db.name.or.index</code></td>
<td>
<p>string or integer specifying the database.
If NULL a vector of available databases is returned.</p>
</td></tr>
<tr><td><code id="getDb_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function provides access to allele frequency databases stored in
the file database.txt in the package directory.
It returns the specified allele frequency database.
</p>


<h3>Value</h3>

<p>data.frame with allele frequency database information.
If no matching database or database index is found NA is returned.
If the database was not found NULL is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Show available allele frequency databases.
getDb()
</code></pre>

<hr>
<h2 id='getKit'>Get Kit</h2><span id='topic+getKit'></span>

<h3>Description</h3>

<p>Provides information about STR kits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKit(
  kit = NULL,
  what = NA,
  show.messages = FALSE,
  .kit.info = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKit_+3A_kit">kit</code></td>
<td>
<p>string or integer to specify the kit.</p>
</td></tr>
<tr><td><code id="getKit_+3A_what">what</code></td>
<td>
<p>string to specify which information to return. Default is 'NA' which return all info.
Not case sensitive. Possible values: &quot;Index&quot;, &quot;Panel&quot;, &quot;Short.Name&quot;, &quot;Full.Name&quot;,
&quot;Marker, &quot;Allele&quot;, &quot;Size&quot;, &quot;Virtual&quot;, &quot;Color&quot;, &quot;Repeat&quot;, &quot;Range&quot;, &quot;Offset&quot;, &quot;Sex.Marker&quot;,
&quot;Quality.Sensor&quot;. An unsupported value returns NA and a warning.</p>
</td></tr>
<tr><td><code id="getKit_+3A_show.messages">show.messages</code></td>
<td>
<p>logical, default TRUE for printing messages to the R prompt.</p>
</td></tr>
<tr><td><code id="getKit_+3A_.kit.info">.kit.info</code></td>
<td>
<p>data frame, run function on a data frame instead of the kits.txt file.</p>
</td></tr>
<tr><td><code id="getKit_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns the following information for a kit specified in kits.txt:
Panel name, short kit name (unique, user defined), full kit name (user defined),
marker names, allele names, allele sizes (bp),
minimum allele size, maximum allele size (bp), flag for virtual alleles,
marker color, marker repeat unit size (bp), minimum marker size,
maximum marker, marker offset (bp), flag for sex markers (TRUE/FALSE).
</p>
<p>If no matching kit or kit index is found NA is returned.
If kit='NULL' or '0' a vector of available kits is printed and NA returned.
</p>


<h3>Value</h3>

<p>data.frame with kit information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Show all information stored for kit with short name 'ESX17'.
getKit("ESX17")
</code></pre>

<hr>
<h2 id='getSetting'>Get Settings.</h2><span id='topic+getSetting'></span>

<h3>Description</h3>

<p>Accepts a key string and returns the corresponding value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSetting(key)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSetting_+3A_key">key</code></td>
<td>
<p>character key for value to return.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Accepts a key string and returns the corresponding value from the settings.txt
file located within the package folders exdata sub folder.
</p>


<h3>Value</h3>

<p>character the retrieved value or NA if not found.
</p>

<hr>
<h2 id='getStrings'>Get Language Strings</h2><span id='topic+getStrings'></span>

<h3>Description</h3>

<p>Accepts a language code and gui. Returns the corresponding language strings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getStrings(language = NA, gui = NA, key = NA, encoding = NA, about = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getStrings_+3A_language">language</code></td>
<td>
<p>character name of the language.</p>
</td></tr>
<tr><td><code id="getStrings_+3A_gui">gui</code></td>
<td>
<p>character the function name for the gui to 'translate'.</p>
</td></tr>
<tr><td><code id="getStrings_+3A_key">key</code></td>
<td>
<p>character the key to 'translate'. Only used in combination with 'gui'.</p>
</td></tr>
<tr><td><code id="getStrings_+3A_encoding">encoding</code></td>
<td>
<p>character encoding to be assumed for input strings.</p>
</td></tr>
<tr><td><code id="getStrings_+3A_about">about</code></td>
<td>
<p>logical FALSE (default) to read key value pairs,
TRUE to read about file as plain text.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Accepts a language code, gui, and key. Returns the corresponding language strings
for the specified gui function or key from a text file named as the language code.
Replaces backslash + n with new line character (only if 'gui' is specified).
</p>


<h3>Value</h3>

<p>character vector or data.table with the retrieved values.
NULL if file or gui was not found.
</p>

<hr>
<h2 id='ggsave_gui'>Save Image</h2><span id='topic+ggsave_gui'></span>

<h3>Description</h3>

<p>A simple GUI wrapper for <code><a href="ggplot2.html#topic+ggsave">ggsave</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsave_gui(
  ggplot = NULL,
  name = "",
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsave_gui_+3A_ggplot">ggplot</code></td>
<td>
<p>plot object.</p>
</td></tr>
<tr><td><code id="ggsave_gui_+3A_name">name</code></td>
<td>
<p>optional string providing a file name.</p>
</td></tr>
<tr><td><code id="ggsave_gui_+3A_env">env</code></td>
<td>
<p>environment where the objects exist.
Default is the current environment.</p>
</td></tr>
<tr><td><code id="ggsave_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="ggsave_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="ggsave_gui_+3A_parent">parent</code></td>
<td>
<p>object specifying the parent widget to center the message box,
and to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simple GUI wrapper for ggsave.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggsave">ggsave</a></code>
</p>

<hr>
<h2 id='guessProfile'>Guess Profile</h2><span id='topic+guessProfile'></span>

<h3>Description</h3>

<p>Guesses the correct profile based on peak height.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>guessProfile(
  data,
  ratio = 0.6,
  height = 50,
  na.rm = FALSE,
  ol.rm = TRUE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="guessProfile_+3A_data">data</code></td>
<td>
<p>a data frame containing at least 'Sample.Name', 'Marker', 'Allele', Height'.</p>
</td></tr>
<tr><td><code id="guessProfile_+3A_ratio">ratio</code></td>
<td>
<p>numeric giving the peak height ratio threshold.</p>
</td></tr>
<tr><td><code id="guessProfile_+3A_height">height</code></td>
<td>
<p>numeric giving the minimum peak height.</p>
</td></tr>
<tr><td><code id="guessProfile_+3A_na.rm">na.rm</code></td>
<td>
<p>logical indicating if rows with no peak should be discarded.</p>
</td></tr>
<tr><td><code id="guessProfile_+3A_ol.rm">ol.rm</code></td>
<td>
<p>logical indicating if off-ladder alleles should be discarded.</p>
</td></tr>
<tr><td><code id="guessProfile_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes typing data from single source samples and filters out the presumed
profile based on peak height and a ratio. Keeps the two highest peaks if
their ratio is above the threshold, or the single highest peak if below
the threshold.
</p>


<h3>Value</h3>

<p>data.frame 'data' with genotype rows only.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load an example dataset.
data(set2)
# Filter out probable profile with criteria at least 70% Hb.
guessProfile(data = set2, ratio = 0.7)
</code></pre>

<hr>
<h2 id='guessProfile_gui'>Guess Profile</h2><span id='topic+guessProfile_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+guessProfile">guessProfile</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>guessProfile_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="guessProfile_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="guessProfile_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="guessProfile_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="guessProfile_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+guessProfile">guessProfile</a></code> function by providing
a graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+guessProfile">guessProfile</a></code>, <code><a href="#topic+checkSubset">checkSubset</a></code>
</p>

<hr>
<h2 id='heightToPeak'>Height To Peak.</h2><span id='topic+heightToPeak'></span>

<h3>Description</h3>

<p>Helper function to convert a peak into a plotable polygon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heightToPeak(data, width = 1, keep.na = TRUE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heightToPeak_+3A_data">data</code></td>
<td>
<p>data frame containing at least columns 'Height' and 'Size'.</p>
</td></tr>
<tr><td><code id="heightToPeak_+3A_width">width</code></td>
<td>
<p>numeric specifying the width of the peak in bp.</p>
</td></tr>
<tr><td><code id="heightToPeak_+3A_keep.na">keep.na</code></td>
<td>
<p>logical. TRUE to keep empty markers.</p>
</td></tr>
<tr><td><code id="heightToPeak_+3A_debug">debug</code></td>
<td>
<p>logical. TRUE prints debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts a single height and size value to a plotable 0-height-0 triangle/peak value.
Makes 3 data points from each peak size for plotting a polygon representing a peak.
Factors in other columns might get converted to factor level.
</p>


<h3>Value</h3>

<p>data.frame with new values.
</p>

<hr>
<h2 id='import'>Import Data</h2><span id='topic+import'></span>

<h3>Description</h3>

<p>Import text files and apply post processing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import(
  folder = TRUE,
  extension = "txt",
  suffix = NA,
  prefix = NA,
  import.file = NA,
  folder.name = NA,
  file.name = TRUE,
  time.stamp = TRUE,
  separator = "\t",
  ignore.case = TRUE,
  auto.trim = FALSE,
  trim.samples = NULL,
  trim.invert = FALSE,
  auto.slim = FALSE,
  slim.na = TRUE,
  na.strings = c("NA", ""),
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_+3A_folder">folder</code></td>
<td>
<p>logical, TRUE all files in folder will be imported,
FALSE only selected file will be imported.</p>
</td></tr>
<tr><td><code id="import_+3A_extension">extension</code></td>
<td>
<p>string providing the file extension.</p>
</td></tr>
<tr><td><code id="import_+3A_suffix">suffix</code></td>
<td>
<p>string, only files with specified suffix will be imported.</p>
</td></tr>
<tr><td><code id="import_+3A_prefix">prefix</code></td>
<td>
<p>string, only files with specified prefix will be imported.</p>
</td></tr>
<tr><td><code id="import_+3A_import.file">import.file</code></td>
<td>
<p>string if file name is provided file will be imported
without showing the file open dialogue.</p>
</td></tr>
<tr><td><code id="import_+3A_folder.name">folder.name</code></td>
<td>
<p>string if folder name is provided files in folder
will be imported without showing the select folder dialogue.</p>
</td></tr>
<tr><td><code id="import_+3A_file.name">file.name</code></td>
<td>
<p>logical if TRUE the file name is written in a column 'File.Name'.
NB! Any existing 'File.Name' column is overwritten.</p>
</td></tr>
<tr><td><code id="import_+3A_time.stamp">time.stamp</code></td>
<td>
<p>logical if TRUE the file modified time stamp is written
in a column 'Time'.
NB! Any existing 'Time' column is overwritten.</p>
</td></tr>
<tr><td><code id="import_+3A_separator">separator</code></td>
<td>
<p>character for the delimiter used to separate columns
(see 'sep' in <code><a href="utils.html#topic+read.table">read.table</a></code> for details).</p>
</td></tr>
<tr><td><code id="import_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical indicating if case should be ignored. Only applies
to multiple file import option.</p>
</td></tr>
<tr><td><code id="import_+3A_auto.trim">auto.trim</code></td>
<td>
<p>logical indicating if dataset should be trimmed.</p>
</td></tr>
<tr><td><code id="import_+3A_trim.samples">trim.samples</code></td>
<td>
<p>character vector with sample names to trim.</p>
</td></tr>
<tr><td><code id="import_+3A_trim.invert">trim.invert</code></td>
<td>
<p>logical to keep (TRUE) or remove (FALSE) samples.</p>
</td></tr>
<tr><td><code id="import_+3A_auto.slim">auto.slim</code></td>
<td>
<p>logical indicating if dataset should be slimmed.</p>
</td></tr>
<tr><td><code id="import_+3A_slim.na">slim.na</code></td>
<td>
<p>logical indicating if rows without data should remain.</p>
</td></tr>
<tr><td><code id="import_+3A_na.strings">na.strings</code></td>
<td>
<p>character vector with strings to be replaced by NA.</p>
</td></tr>
<tr><td><code id="import_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Imports text files (e.g. GeneMapper results exported as text files)
as data frames. Options to import one or multiple files. For multiple
files it is possible to specify prefix, suffix, and file extension
to create a file name filter. The file name and/or file time stamp
can be imported.
NB! Empty strings (&quot;&quot;) and NA strings (&quot;NA&quot;) are converted to NA.
See <code><a href="base.html#topic+list.files">list.files</a></code> and <code><a href="utils.html#topic+read.table">read.table</a></code> for additional details.
</p>


<h3>Value</h3>

<p>data.frame with imported result.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trim">trim</a></code>, <code><a href="#topic+slim">slim</a></code>, <code><a href="base.html#topic+list.files">list.files</a></code>, <code><a href="utils.html#topic+read.table">read.table</a></code>
</p>

<hr>
<h2 id='import_gui'>Import Data</h2><span id='topic+import_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+import">import</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_gui_+3A_env">env</code></td>
<td>
<p>environment into which the object will be saved.
Default is the current environment.</p>
</td></tr>
<tr><td><code id="import_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="import_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="import_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+import">import</a></code> function by providing a graphical
user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+import">import</a></code>
</p>

<hr>
<h2 id='listObjects'>List Objects</h2><span id='topic+listObjects'></span>

<h3>Description</h3>

<p>Internal helper function to list objects in an environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listObjects(
  env = parent.frame(),
  obj.class = NULL,
  sort = NULL,
  decreasing = TRUE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listObjects_+3A_env">env</code></td>
<td>
<p>environment in which to search for objects.</p>
</td></tr>
<tr><td><code id="listObjects_+3A_obj.class">obj.class</code></td>
<td>
<p>character string or vector specifying the object class.</p>
</td></tr>
<tr><td><code id="listObjects_+3A_sort">sort</code></td>
<td>
<p>character string &quot;time&quot;, &quot;alpha&quot;, &quot;size&quot; specifying the sorting order. Default = NULL.</p>
</td></tr>
<tr><td><code id="listObjects_+3A_decreasing">decreasing</code></td>
<td>
<p>logical used to indicate order when sorting is not NULL. Default = TRUE.</p>
</td></tr>
<tr><td><code id="listObjects_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internal helper function to retrieve a list of objects from a workspace.
Take an environment as argument and optionally an object class.
Returns a list of objects of the specified class in the environment.
</p>


<h3>Value</h3>

<p>character vector with the object names or NULL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# List data frames in the workspace.
listObjects(obj.class = "data.frame")
# List functions in the workspace.
listObjects(obj.class = "function")

## End(Not run)

</code></pre>

<hr>
<h2 id='makeKit_gui'>Make Kit</h2><span id='topic+makeKit_gui'></span>

<h3>Description</h3>

<p>Add new kits or edit the kit file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeKit_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeKit_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="makeKit_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.
[Not currently used]</p>
</td></tr>
<tr><td><code id="makeKit_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="makeKit_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A graphical user interface for reading information from 'bins' and
'panels' file for the creation of additional kits. It is also possible to
edit the short and full name of existing kits or removing kits.
The gender marker of each kits is auto detected but can be changed manually.
#' NB! Short name must be unique.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+readBinsFile">readBinsFile</a></code>, <code><a href="#topic+readPanelsFile">readPanelsFile</a></code>, <code><a href="#topic+combineBinsAndPanels">combineBinsAndPanels</a></code>
</p>

<hr>
<h2 id='maskAT'>Mask And Prepare Data To Analyze Analytical Threshold</h2><span id='topic+maskAT'></span>

<h3>Description</h3>

<p>Break-out function to prepare data for the function <code>calculateAT</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maskAT(
  data,
  ref = NULL,
  mask.height = TRUE,
  height = 500,
  mask.sample = TRUE,
  per.dye = TRUE,
  range.sample = 20,
  mask.ils = TRUE,
  range.ils = 10,
  ignore.case = TRUE,
  word = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maskAT_+3A_data">data</code></td>
<td>
<p>a data frame containing at least 'Dye.Sample.Peak',
'Sample.File.Name', 'Marker', 'Allele', 'Height', and 'Data.Point'.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_ref">ref</code></td>
<td>
<p>a data frame containing at least
'Sample.Name', 'Marker', 'Allele'.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_mask.height">mask.height</code></td>
<td>
<p>logical to indicate if high peaks should be masked.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_height">height</code></td>
<td>
<p>integer for global lower peak height threshold for peaks
to be excluded from the analysis. Active if 'mask.peak=TRUE.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_mask.sample">mask.sample</code></td>
<td>
<p>logical to indicate if sample allelic peaks should be masked.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_per.dye">per.dye</code></td>
<td>
<p>logical TRUE if sample peaks should be masked per dye channel.
FALSE if sample peaks should be masked globally across dye channels.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_range.sample">range.sample</code></td>
<td>
<p>integer to specify the masking range in (+/-) data points.
Active if mask.sample=TRUE.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_mask.ils">mask.ils</code></td>
<td>
<p>logical to indicate if internal lane standard peaks should be masked.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_range.ils">range.ils</code></td>
<td>
<p>integer to specify the masking range in (+/-) data points.
Active if mask.ils=TRUE.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical to indicate if sample matching should ignore case.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_word">word</code></td>
<td>
<p>logical to indicate if word boundaries should be added before sample matching.</p>
</td></tr>
<tr><td><code id="maskAT_+3A_debug">debug</code></td>
<td>
<p>logical to indicate if debug information should be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prepares the 'SamplePlotSizingTable' for analysis of analytical threshold. It is needed
by the plot functions for control of masking. The preparation consist of
converting the 'Height' and 'Data.Point' column to numeric (if needed), then
dye channel information is extracted from the 'Dye.Sample.Peak' column and
added to its own 'Dye' column, known fragments of the internal lane standard
(marked with an asterisk '*') is flagged as 'TRUE' in a new column 'ILS'.
</p>


<h3>Value</h3>

<p>data.frame with added columns 'Dye' and 'ILS'.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateAT">calculateAT</a></code>
</p>

<hr>
<h2 id='modelDropout_gui'>Model And Plot Drop-out Events</h2><span id='topic+modelDropout_gui'></span>

<h3>Description</h3>

<p>Model the probability of drop-out and plot graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modelDropout_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="modelDropout_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="modelDropout_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="modelDropout_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="modelDropout_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+calculateDropout">calculateDropout</a></code> score drop-out events relative to a user
defined LDT in four different ways:
(1) by reference to the low molecular weight allele (Method1),
(2) by reference to the high molecular weight allele (Method2),
(3) by reference to a random allele (MethodX), and
(4) by reference to the locus (MethodL).
Options 1-3 are recommended by the DNA commission (see reference),
while option 4 is included for experimental purposes.
Options 1-3 may discard many dropout events while option 4 catches all
drop-out events. On the other hand options 1-3 can score events below
the LDT, while option 4 cannot, making accurate predictions possible
below the LDT. This is also why the number of observed drop-out events
may differ between model plots and heatmap, scatterplot, and ecdf.
</p>
<p>Method X/1/2 records the peak height of the partner allele to be used as
the explanatory variable in the logistic regression. The locus method L also
do this when there has been a drop-out, if not the the mean peak height for
the locus is used. Peak heights for the locus method are stored in a
separate column.
</p>
<p>Using the scored drop-out events and the peak heights of the surviving
alleles the probability of drop-out can be modeled by logistic regression
as described in Appendix B in reference [1].
P(dropout|H) = B0 + B1*H, where 'H' is the peak height or log(peak height).
This produces a plot with the predicted probabilities for a range of peak heights.
There are options to print the model parameters, mark the stochastic
threshold at a specified probability of drop-out, include the underlying
observations, and to calculate a specified prediction interval.
A conservative estimate of the stochastic threshold can be calculated
from the prediction interval: the risk of observing a drop-out probability
greater than the specified threshold limit, at the conservative peak height,
is less than a specified value (e.g. 1-0.95=0.05). By default the gender
marker is excluded from the dataset used for modeling, and the peak height
is used as explanatory variable. The logarithm of the average peak height 'H'
can be used instead of the allele/locus peak height [3] (The implementation
of 'H' has limitations when dropout is present. See <code><a href="#topic+calculateHeight">calculateHeight</a></code>).
To evaluate the goodness of fit for the logistic regression the
Hosmer-Lemeshow test is used [4]. A value below 0.05 indicates a poor fit.
Alternatives to the logistic regression method are discussed in reference [5]
and [6].
</p>
<p>Explanation of the result:
Dropout - all alleles are scored according to the limit of detection threshold (LDT).
This is the observations and is not used for modeling.
Rfu - peak height of the surviving allele.
MethodX - a random reference allele is selected and drop-out is scored in
relation to the the partner allele.
Method1 - the low molecular weight allele is selected and drop-out is
scored if the high molecular weight allele is missing.
Method2 - the high molecular weight allele is selected and drop-out is
scored if the low molecular weight allele is missing.
MethodL - drop-out is scored per locus i.e. drop-out if any allele is missing.
MethodL.Ph - peak height of the surviving allele if one allele has dropped out,
or the average peak height if no drop-out.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>References</h3>

<p>[1] Peter Gill et.al.,
DNA commission of the International Society of Forensic Genetics:
Recommendations on the evaluation of STR typing results that may
include drop-out and/or drop-in using probabilistic methods,
Forensic Science International: Genetics, Volume 6, Issue 6, December 2012,
Pages 679-688, ISSN 1872-4973, 10.1016/j.fsigen.2012.06.002.
<a href="https://doi.org/10.1016/j.fsigen.2012.06.002">doi:10.1016/j.fsigen.2012.06.002</a>
</p>
<p>[2] Peter Gill, Roberto Puch-Solis, James Curran,
The low-template-DNA (stochastic) threshold-Its determination relative to
risk analysis for national DNA databases,
Forensic Science International: Genetics, Volume 3, Issue 2, March 2009,
Pages 104-111, ISSN 1872-4973, 10.1016/j.fsigen.2008.11.009.
<a href="https://doi.org/10.1016/j.fsigen.2008.11.009">doi:10.1016/j.fsigen.2008.11.009</a>
</p>
<p>[3] Torben Tvedebrink, Poul Svante Eriksen, Helle Smidt Mogensen, Niels Morling,
Estimating the probability of allelic drop-out of STR alleles in forensic genetics,
Forensic Science International: Genetics, Volume 3, Issue 4, September 2009,
Pages 222-226, ISSN 1872-4973, 10.1016/j.fsigen.2009.02.002.
<a href="https://doi.org/10.1016/j.fsigen.2009.02.002">doi:10.1016/j.fsigen.2009.02.002</a>
</p>
<p>[4] H. DW Jr., S. Lemeshow, Applied Logistic Regression, John Wiley &amp; Sons, 2004.
</p>
<p>[5] A.A. Westen, L.J.W. Grol, J. Harteveld, A.S. Matai, P. de Knijff, T. Sijen,
Assessment of the stochastic threshold, back- and forward stutter filters
and low template techniques for NGM,
Forensic Science International: Genetetics, Volume 6, Issue 6 December 2012,
Pages 708-715, ISSN 1872-4973, 10.1016/j.fsigen.2012.05.001.
<a href="https://doi.org/10.1016/j.fsigen.2012.05.001">doi:10.1016/j.fsigen.2012.05.001</a>
</p>
<p>[6] R. Puch-Solis, A.J. Kirkham, P. Gill, J. Read, S. Watson, D. Drew,
Practical determination of the low template DNA threshold,
Forensic Science International: Genetetics, Volume 5, Issue 5, November 2011,
Pages 422-427, ISSN 1872-4973, 10.1016/j.fsigen.2010.09.001.
<a href="https://doi.org/10.1016/j.fsigen.2010.09.001">doi:10.1016/j.fsigen.2010.09.001</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateDropout">calculateDropout</a></code>, <code><a href="#topic+plotDropout_gui">plotDropout_gui</a></code>, <code><a href="ResourceSelection.html#topic+hoslem.test">hoslem.test</a></code>
</p>

<hr>
<h2 id='plotAT_gui'>Plot Analytical Threshold</h2><span id='topic+plotAT_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from analytical threshold data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAT_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAT_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="plotAT_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotAT_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotAT_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select data to plot in the drop-down menu. Plot regression data
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='plotBalance_gui'>Plot Balance</h2><span id='topic+plotBalance_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from balance data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBalance_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBalance_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotBalance_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotBalance_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotBalance_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select a dataset to plot and the typing kit used (if not automatically detected).
Plot heterozygote peak balance versus the average locus peak height,
the average profile peak height 'H', or by the difference in repeat units
(delta). Plot inter-locus balance versus the average locus peak height, or
the average profile peak height 'H'. Automatic plot titles can be replaced by
custom titles. Sex markers can be excluded. It is possible to plot
logarithmic ratios. A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='plotCapillary_gui'>Plot Capillary Balance</h2><span id='topic+plotCapillary_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from capillary balance data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCapillary_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotCapillary_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotCapillary_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotCapillary_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotCapillary_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select a dataset to plot from the drop-down menu.
Plot capillary balance as a dotplot, boxplot or as a distribution.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='plotContamination_gui'>Plot Contamination</h2><span id='topic+plotContamination_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from negative control data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotContamination_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotContamination_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="plotContamination_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotContamination_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotContamination_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select data to plot in the drop-down menu.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>References</h3>

<p>Duncan Taylor et.al.,
Validating multiplexes for use in conjunction with modern interpretation strategies,
Forensic Science International: Genetics, Volume 20, January 2016,
Pages 6-19, ISSN 1872-4973, 10.1016/j.fsigen.2015.09.011.
<a href="https://doi.org/10.1016/j.fsigen.2015.09.011">doi:10.1016/j.fsigen.2015.09.011</a>
</p>

<hr>
<h2 id='plotDistribution_gui'>Plot Distribution</h2><span id='topic+plotDistribution_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of distribution plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDistribution_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDistribution_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotDistribution_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotDistribution_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotDistribution_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot the distribution of data as cumulative distribution function,
probability density function, or count. First select a dataset, then select
a group (in column 'Group' if any), finally select a column to plot the distribution of.
It is possible to overlay a boxplot and to plot logarithms.
Various smoothing kernels and bandwidths can be specified.
The bandwidth or the number of bins can be specified for the histogram.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+log">log</a></code>, <code><a href="ggplot2.html#topic+geom_density">geom_density</a></code>
</p>

<hr>
<h2 id='plotDropout_gui'>Plot Drop-out Events</h2><span id='topic+plotDropout_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from dropout data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDropout_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDropout_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotDropout_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotDropout_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotDropout_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot dropout data as heatmap arranged by, average peak height,
amount, concentration, or sample name. It is also possible to plot the
empirical cumulative distribution (ecdp) of the peak heights of surviving heterozygote
alleles (with dropout of the partner allele), or a dotplot of all dropout events.
The peak height of homozygote alleles can be included in the ecdp.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>References</h3>

<p>Antoinette A. Westen, Laurens J.W. Grol, Joyce Harteveld, Anuska S.Matai,
Peter de Knijff, Titia Sijen, Assessment of the stochastic threshold, back- and
forward stutter filters and low template techniques for NGM,
Forensic Science International: Genetetics, Volume 6, Issue 6, December 2012,
Pages 708-715, ISSN 1872-4973, 10.1016/j.fsigen.2012.05.001.
<a href="https://doi.org/10.1016/j.fsigen.2012.05.001">doi:10.1016/j.fsigen.2012.05.001</a>
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='plotEPG2'>plotEPG2</h2><span id='topic+plotEPG2'></span>

<h3>Description</h3>

<p>EPG data visualizer (interactive)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEPG2(
  mixData,
  kit,
  refData = NULL,
  AT = NULL,
  ST = NULL,
  dyeYmax = TRUE,
  plotRepsOnly = TRUE,
  options = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotEPG2_+3A_mixdata">mixData</code></td>
<td>
<p>List of mixData[[ss]][[loc]] =list(adata,hdata), with samplenames ss, loci names loc, allele vector adata (can be strings or numeric), intensity vector hdata (must be numeric)</p>
</td></tr>
<tr><td><code id="plotEPG2_+3A_kit">kit</code></td>
<td>
<p>Short name of kit: See supported kits with getKit()</p>
</td></tr>
<tr><td><code id="plotEPG2_+3A_refdata">refData</code></td>
<td>
<p>List of refData[[rr]][[loc]] or refData[[loc]][[rr]] to label references (flexible). Visualizer will show dropout alleles.</p>
</td></tr>
<tr><td><code id="plotEPG2_+3A_at">AT</code></td>
<td>
<p>A detection threshold can be shown in a dashed line in the plot (constant). Possibly a vector with locus column names</p>
</td></tr>
<tr><td><code id="plotEPG2_+3A_st">ST</code></td>
<td>
<p>A stochastic threshold can be shown in a dashed line in the plot (constant). Possibly a vector with locus column names</p>
</td></tr>
<tr><td><code id="plotEPG2_+3A_dyeymax">dyeYmax</code></td>
<td>
<p>Whether Y-axis should be same for all markers (FALSE) or not (TRUE this is default)</p>
</td></tr>
<tr><td><code id="plotEPG2_+3A_plotrepsonly">plotRepsOnly</code></td>
<td>
<p>Whether only replicate-plot is shown in case of multiple samples (TRUE is default)</p>
</td></tr>
<tr><td><code id="plotEPG2_+3A_options">options</code></td>
<td>
<p>A list of possible plot configurations. See comments below</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots peak height with corresponding allele for sample(s) for a given kit.
</p>


<h3>Value</h3>

<p>sub A plotly widget
</p>


<h3>Author(s)</h3>

<p>Oyvind Bleka
</p>

<hr>
<h2 id='plotEPG2_gui'>Plot EPG</h2><span id='topic+plotEPG2_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+plotEPG2">plotEPG2</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEPG2_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotEPG2_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotEPG2_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotEPG2_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotEPG2_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+plotEPG2">plotEPG2</a></code> function by providing a graphical
user interface.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotEPG2">plotEPG2</a></code>
</p>

<hr>
<h2 id='plotGroups_gui'>Plot Empirical Cumulative Distributions</h2><span id='topic+plotGroups_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of empirical cumulative distribution plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGroups_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotGroups_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotGroups_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotGroups_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotGroups_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot the distribution of data as cumulative distribution function
for multiple groups. First select a dataset, then select columns to flat,
group, and plot by. For example, if a genotype dataset is selected and data
is flattened by Sample.Name the 'group by' and 'plot by' values must be
identical for all rows for a given sample.
Automatic plot titles can be replaced by custom titles.
Group names can be changed. A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+stat_ecdf">stat_ecdf</a></code>
</p>

<hr>
<h2 id='plotKit_gui'>Plot Kit Marker Ranges</h2><span id='topic+plotKit_gui'></span>

<h3>Description</h3>

<p>GUI for plotting marker ranges for kits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotKit_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotKit_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotKit_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotKit_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotKit_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Create an overview of the size range for markers in different kits.
It is possible to select multiple kits, specify titles, font size, distance
between two kits, distance between dye channels, and the transparency of dyes.
</p>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='plotPeaks_gui'>Plot Peaks</h2><span id='topic+plotPeaks_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from result type data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPeaks_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPeaks_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotPeaks_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotPeaks_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotPeaks_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot result type data. It is possible to customize titles and font
size. Data can be plotted as as frequency or proportion. The values can be
printed on the plot with custom number of decimals. There are several
color palettes to chose from.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='plotPrecision_gui'>Plot Precision</h2><span id='topic+plotPrecision_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from precision data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPrecision_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPrecision_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="plotPrecision_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotPrecision_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotPrecision_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot precision data for size, height, or data point as dotplot or
boxplot. Plot per marker or all in one. Use the mean value or the allele
designation as x-axis labels.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='plotPullup_gui'>Plot Pull-up</h2><span id='topic+plotPullup_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from pull-up data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPullup_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPullup_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotPullup_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotPullup_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotPullup_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select a dataset to plot and the typing kit used (if not automatically detected).
Plot pull-up peak ratio versus the peak height of the known allele
Automatic plot titles can be replaced by custom titles.
Sex markers can be excluded. A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='plotRatio_gui'>Plot Ratio</h2><span id='topic+plotRatio_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from marker ratio data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRatio_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRatio_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="plotRatio_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotRatio_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotRatio_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select data to plot in the drop-down menu.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='plotResultType_gui'>Plot Result Type</h2><span id='topic+plotResultType_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from result type data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotResultType_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotResultType_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotResultType_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotResultType_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotResultType_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot result type data. It is possible to customize titles and font
size. Data can be plotted as as frequency or proportion. The values can be
printed on the plot with custom number of decimals. There are several
color palettes to chose from.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='plotSlope_gui'>Plot Profile Slope</h2><span id='topic+plotSlope_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from slope data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSlope_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSlope_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="plotSlope_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotSlope_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotSlope_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select a dataset to plot. Plot slope by sample.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='plotStutter_gui'>Plot Stutter</h2><span id='topic+plotStutter_gui'></span>

<h3>Description</h3>

<p>GUI simplifying the creation of plots from stutter data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotStutter_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotStutter_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="plotStutter_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="plotStutter_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="plotStutter_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Select data to plot in the drop-down menu. Check that the correct
kit has been detected. Plot stutter data by parent allele or by peak height.
Automatic plot titles can be replaced by custom titles.
A name for the result is automatically suggested.
The resulting plot can be saved as either a plot object or as an image.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><a href="https://ggplot2.tidyverse.org/">https://ggplot2.tidyverse.org/</a> for details on plot settings.
</p>

<hr>
<h2 id='readBinsFile'>Read Bins file</h2><span id='topic+readBinsFile'></span>

<h3>Description</h3>

<p>Reads GeneMapper 'Bins' files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBinsFile(bin.files, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readBinsFile_+3A_bin.files">bin.files</code></td>
<td>
<p>string, complete path to Bins file.</p>
</td></tr>
<tr><td><code id="readBinsFile_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads useful information from 'Bins' files and save it as a data.frame.
</p>


<h3>Value</h3>

<p>data.frame containing the columns 'Panel', 'Marker', 'Allele', 'Size',
'Size.Min', 'Size.Max', 'Virtual'.
</p>

<hr>
<h2 id='readPanelsFile'>Read Panels File</h2><span id='topic+readPanelsFile'></span>

<h3>Description</h3>

<p>Reads GeneMapper Panels files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readPanelsFile(panel.files, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readPanelsFile_+3A_panel.files">panel.files</code></td>
<td>
<p>string, complete path to Panels file.</p>
</td></tr>
<tr><td><code id="readPanelsFile_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads useful information from Panels files and save it as a data.frame.
</p>


<h3>Value</h3>

<p>data.frame containing the columns 'Panel', 'Marker', 'Color',
'Marker.Min', 'Marker.Max', 'Repeat'.
</p>

<hr>
<h2 id='ref1'>ESX17 Positive Control Profile</h2><span id='topic+ref1'></span>

<h3>Description</h3>

<p>A dataset in 'GeneMapper' format containing the DNA profile of
the ESX17 positive control sample with homozygotes as one entry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref1)
</code></pre>


<h3>Format</h3>

<p>A data frame with 17 rows and 4 variables
</p>

<hr>
<h2 id='ref11'>ESX17 Positive Control Profile</h2><span id='topic+ref11'></span>

<h3>Description</h3>

<p>A dataset in 'GeneMapper' format containing the DNA profile of
the ESX17 positive control sample with homozygotes as two entries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref11)
</code></pre>


<h3>Format</h3>

<p>A data frame with 17 rows and 4 variables
</p>

<hr>
<h2 id='ref2'>SGMPlus example data</h2><span id='topic+ref2'></span>

<h3>Description</h3>

<p>A slimmed reference dataset containing an arbitrary SGMPlus DNA profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref2)
</code></pre>


<h3>Format</h3>

<p>A data frame with 16 rows and 3 variables
</p>

<hr>
<h2 id='ref3'>ESX17 example data for dropout analysis.</h2><span id='topic+ref3'></span>

<h3>Description</h3>

<p>Reference profiles for source samples.
Text file in GeneMapper format.
</p>


<h3>Format</h3>

<p>ASCII text file
</p>

<hr>
<h2 id='ref4'>ESX17 example data for dropout analysis.</h2><span id='topic+ref4'></span>

<h3>Description</h3>

<p>A slimmed dataset containing reference profiles for source samples in set4.
Reference 'A2' has double entries for homozygotes.
Reference 'F2' has single entries for homozygotes.
Reference 'bc' has double entries for homozygotes, and lower case sample name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref4)
</code></pre>


<h3>Format</h3>

<p>A data frame with 98 rows and 3 variables
</p>

<hr>
<h2 id='ref51'>ESX17 example data for mixture analysis.</h2><span id='topic+ref51'></span>

<h3>Description</h3>

<p>A slimmed dataset containing the reference profile for the major
component  in set5.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref51)
</code></pre>


<h3>Format</h3>

<p>A data frame with 34 rows and 3 variables
</p>

<hr>
<h2 id='ref52'>ESX17 example data for mixture analysis.</h2><span id='topic+ref52'></span>

<h3>Description</h3>

<p>A slimmed dataset containing the reference profile for the minor
component  in set5.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref52)
</code></pre>


<h3>Format</h3>

<p>A data frame with 34 rows and 3 variables
</p>

<hr>
<h2 id='ref61'>Fusion example data for dropout analysis.</h2><span id='topic+ref61'></span>

<h3>Description</h3>

<p>A slimmed dataset containing the reference profile for the
samples in set6.
NB! Marker order is different from set6.
NB! Reference R has a Y marker with NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref61)
</code></pre>


<h3>Format</h3>

<p>A data frame with 89 rows and 3 variables
</p>

<hr>
<h2 id='ref62'>Fusion example data for dropout analysis.</h2><span id='topic+ref62'></span>

<h3>Description</h3>

<p>A slimmed dataset containing the reference profile for the
samples in set6.
NB! Marker order is same as set6.
NB! Reference R has a Y marker with NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref62)
</code></pre>


<h3>Format</h3>

<p>A data frame with 89 rows and 3 variables
</p>

<hr>
<h2 id='ref7'>ESSplex SE QS example data for inhibition analysis.</h2><span id='topic+ref7'></span>

<h3>Description</h3>

<p>A slimmed dataset containing the reference profile for the
samples in set7.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ref7)
</code></pre>


<h3>Format</h3>

<p>A data frame with 35 rows and 4 variables
</p>

<hr>
<h2 id='removeArtefact'>Remove Artefacts</h2><span id='topic+removeArtefact'></span>

<h3>Description</h3>

<p>Remove artefact peaks from data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeArtefact(
  data,
  artefact = NULL,
  marker = NULL,
  allele = NULL,
  threshold = NULL,
  na.rm = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeArtefact_+3A_data">data</code></td>
<td>
<p>data.frame with data to remove spikes from.</p>
</td></tr>
<tr><td><code id="removeArtefact_+3A_artefact">artefact</code></td>
<td>
<p>data.frame that lists artefacts in columns 'Marker',
'Allele', optionally with 'Allele.Proportion'.
Alternatively artefacts can be provided using 'marker' and 'allele'.</p>
</td></tr>
<tr><td><code id="removeArtefact_+3A_marker">marker</code></td>
<td>
<p>character vector with marker names paired with values in 'allele'.</p>
</td></tr>
<tr><td><code id="removeArtefact_+3A_allele">allele</code></td>
<td>
<p>character vector with allele names paired with values in 'marker'.</p>
</td></tr>
<tr><td><code id="removeArtefact_+3A_threshold">threshold</code></td>
<td>
<p>numeric value defining a minimum proportion for artefacts.
Requires 'artefacts' including the column 'Allele.Proportion'.</p>
</td></tr>
<tr><td><code id="removeArtefact_+3A_na.rm">na.rm</code></td>
<td>
<p>logical TRUE to preserve Allele=NA in 'data'.</p>
</td></tr>
<tr><td><code id="removeArtefact_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Removes identified artefacts from the dataset. Likely artefacts can be
identified using the function <code><a href="#topic+calculateAllele">calculateAllele</a></code>. The output
should then be provided to the 'artefact'. Alternatively known artefacts
can be provided using the 'marker' and 'allele' arguments.
</p>


<h3>Value</h3>

<p>data.frame with spikes removed.
</p>

<hr>
<h2 id='removeArtefact_gui'>Remove Artefact</h2><span id='topic+removeArtefact_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+removeArtefact">removeArtefact</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeArtefact_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeArtefact_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="removeArtefact_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="removeArtefact_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="removeArtefact_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+removeArtefact">removeArtefact</a></code> function by providing a
graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='removeSpike'>Remove Spikes</h2><span id='topic+removeSpike'></span>

<h3>Description</h3>

<p>Remove spikes from data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeSpike(data, spike, invert = FALSE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeSpike_+3A_data">data</code></td>
<td>
<p>data.frame with data to remove spikes from.</p>
</td></tr>
<tr><td><code id="removeSpike_+3A_spike">spike</code></td>
<td>
<p>data.frame with list of spikes.</p>
</td></tr>
<tr><td><code id="removeSpike_+3A_invert">invert</code></td>
<td>
<p>logical FALSE to remove spikes, TRUE to keep spikes.</p>
</td></tr>
<tr><td><code id="removeSpike_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Removes identified spikes from the dataset. Spikes are identified using the
function <code><a href="#topic+calculateSpike">calculateSpike</a></code> and provided as a separate dataset.
NB! Samples must have unique identifiers.
Some laboratories use non-unique names for e.g. negative controls. To allow
identification of specific samples when multiple batches are imported into
one dataset an id is automatically created by combining the sample name and
the file name. This work well as long as there is at most 1 identically
named sample in each file (batch). To enable multiple identically named
samples in one file, the sample names can be prefixed with the lane or well
number before importing them to STR-validator.
</p>


<h3>Value</h3>

<p>data.frame with spikes removed.
</p>

<hr>
<h2 id='removeSpike_gui'>Remove Spike</h2><span id='topic+removeSpike_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+removeSpike">removeSpike</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeSpike_gui(
  env = parent.frame(),
  savegui = NULL,
  debug = FALSE,
  parent = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeSpike_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames.</p>
</td></tr>
<tr><td><code id="removeSpike_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="removeSpike_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="removeSpike_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+removeSpike">removeSpike</a></code> function by providing a
graphical user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>

<hr>
<h2 id='sample_tableToList'>sample_tableToList</h2><span id='topic+sample_tableToList'></span>

<h3>Description</h3>

<p>Converting table to list format (helpfunction)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_tableToList(table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_tableToList_+3A_table">table</code></td>
<td>
<p>A table with data (Evid or refs)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>outL A data list
</p>


<h3>Author(s)</h3>

<p>Oyvind Bleka
</p>

<hr>
<h2 id='saveObject'>Save Object</h2><span id='topic+saveObject'></span>

<h3>Description</h3>

<p>Save an object in the specified environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saveObject(
  name = NULL,
  object,
  parent = NULL,
  suggest = "",
  env = parent.frame(),
  remove = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saveObject_+3A_name">name</code></td>
<td>
<p>character giving the name of the object to save.
If NULL a dialogue asks for a name.</p>
</td></tr>
<tr><td><code id="saveObject_+3A_object">object</code></td>
<td>
<p>object to save.</p>
</td></tr>
<tr><td><code id="saveObject_+3A_parent">parent</code></td>
<td>
<p>object specifying the parent GUI object to center the message box.</p>
</td></tr>
<tr><td><code id="saveObject_+3A_suggest">suggest</code></td>
<td>
<p>character string for a suggested name for the object to save/re-name.</p>
</td></tr>
<tr><td><code id="saveObject_+3A_env">env</code></td>
<td>
<p>environment in which to save and search for existing objects.</p>
</td></tr>
<tr><td><code id="saveObject_+3A_remove">remove</code></td>
<td>
<p>character string for a named object to remove (e.g. the original object if re-naming).</p>
</td></tr>
<tr><td><code id="saveObject_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Saves an object with the given name in the specified environment
if it does not exist. If the object exist a message box ask if the object
should be overwritten. The function can also be used to re-name if 'name'
is not provided (NULL). The 'suggest' provides a suggested name for the
object to re-name.
</p>


<h3>Value</h3>

<p>logical TRUE if object was saved FALSE if not.
</p>

<hr>
<h2 id='scrambleAlleles'>Scramble Alleles</h2><span id='topic+scrambleAlleles'></span>

<h3>Description</h3>

<p>Scrambles alleles in a dataset to anonymize the profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scrambleAlleles(data, db = "ESX 17 Hill")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scrambleAlleles_+3A_data">data</code></td>
<td>
<p>data.frame with columns 'Sample.Name', 'Marker', and 'Allele'.</p>
</td></tr>
<tr><td><code id="scrambleAlleles_+3A_db">db</code></td>
<td>
<p>character defining the allele frequency database to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internal helper function to create example data.
Assumes data with unique alleles per marker i.e. no duplications.
This allow for sampling without replacement see <code><a href="base.html#topic+sample">sample</a></code>.
Sex markers are currently not scrambled i.e. they are kept intact.
Alleles in the dataset is replaced with random alleles sampled from the allele database.
If 'Size' is in the dataset it will be replaced by an estimated size.
If 'Data.Point' is present it will be removed.
</p>


<h3>Value</h3>

<p>data.frame with changes in 'Allele' column.
</p>

<hr>
<h2 id='set1'>Typing data in 'GeneMapper' format</h2><span id='topic+set1'></span>

<h3>Description</h3>

<p>A dataset containing ESX17 genotyping result for 8 replicates
of the positive control sample, a negative control and ladder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(set1)
</code></pre>


<h3>Format</h3>

<p>A data frame with 170 rows and 13 variables
</p>

<hr>
<h2 id='set2'>SGMPlus example data</h2><span id='topic+set2'></span>

<h3>Description</h3>

<p>A slimmed dataset containing SGM Plus genotyping result for 2 replicates
of 'sampleA'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(set2)
</code></pre>


<h3>Format</h3>

<p>A data frame with 32 rows and 5 variables
</p>

<hr>
<h2 id='set3'>ESX17 example data for dropout analysis.</h2><span id='topic+set3'></span>

<h3>Description</h3>

<p>Data from dilution experiment for dropout analysis.
Text file with exported GeneMapper genotypes table.
</p>


<h3>Format</h3>

<p>ASCII text file
</p>

<hr>
<h2 id='set4'>ESX17 example data for dropout analysis.</h2><span id='topic+set4'></span>

<h3>Description</h3>

<p>A slimmed dataset containing data from
dilution experiment for dropout analysis (from set3).
One sample replicate has lower case sample name (bc9).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(set4)
</code></pre>


<h3>Format</h3>

<p>A data frame with 1609 rows and 5 variables
</p>

<hr>
<h2 id='set5'>ESX17 example data for mixture analysis.</h2><span id='topic+set5'></span>

<h3>Description</h3>

<p>A slimmed dataset containing data from
mixture experiment for Mx analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(set5)
</code></pre>


<h3>Format</h3>

<p>A data frame with 1663 rows and 7 variables
</p>

<hr>
<h2 id='set6'>Fusion example data for dropout analysis.</h2><span id='topic+set6'></span>

<h3>Description</h3>

<p>A slimmed dataset containing data from
sensitivity experiment for dropout analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(set6)
</code></pre>


<h3>Format</h3>

<p>A data frame with 1848 rows and 7 variables
</p>

<hr>
<h2 id='set7'>ESSplex SE QS example data for inhibition analysis.</h2><span id='topic+set7'></span>

<h3>Description</h3>

<p>A slimmed dataset containing data from
inhibition experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(set7)
</code></pre>


<h3>Format</h3>

<p>A data frame with 883 rows and 7 variables
</p>

<hr>
<h2 id='slim'>Slim Data Frames</h2><span id='topic+slim'></span>

<h3>Description</h3>

<p>Slim data frames with repeated columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slim(data, fix = NULL, stack = NULL, keep.na = TRUE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slim_+3A_data">data</code></td>
<td>
<p>data.frame.</p>
</td></tr>
<tr><td><code id="slim_+3A_fix">fix</code></td>
<td>
<p>vector of strings with column names to keep fixed.</p>
</td></tr>
<tr><td><code id="slim_+3A_stack">stack</code></td>
<td>
<p>vector of strings with column names to slim.</p>
</td></tr>
<tr><td><code id="slim_+3A_keep.na">keep.na</code></td>
<td>
<p>logical, keep a row even if no data.</p>
</td></tr>
<tr><td><code id="slim_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Stack repeated columns into single columns.
For example, the following data frame:
Sample.Name|Marker|Allele.1|Allele.2|Size.1|Size.2|Data.Point..
using this command:
slim(data, fix=c(&quot;Sample.Name&quot;,&quot;Marker&quot;), stack=c(&quot;Allele&quot;,&quot;Size&quot;))
would result in this data frame (NB! 'Data.Point' is dropped):
Sample.Name|Marker|Allele|Size
</p>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='slim_gui'>Slim Data Frames</h2><span id='topic+slim_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+slim">slim</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slim_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slim_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="slim_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="slim_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="slim_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+slim">slim</a></code> function by providing a graphical
user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+slim">slim</a></code>
</p>

<hr>
<h2 id='sortMarker'>Sort Markers</h2><span id='topic+sortMarker'></span>

<h3>Description</h3>

<p>Sort markers and dye as they appear in the EPG.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sortMarker(data, kit, add.missing.levels = FALSE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sortMarker_+3A_data">data</code></td>
<td>
<p>data.frame containing a column 'Marker' and optionally 'Dye'.</p>
</td></tr>
<tr><td><code id="sortMarker_+3A_kit">kit</code></td>
<td>
<p>string or integer indicating kit.</p>
</td></tr>
<tr><td><code id="sortMarker_+3A_add.missing.levels">add.missing.levels</code></td>
<td>
<p>logical, TRUE missing markers are added,
FALSE missing markers are not added.</p>
</td></tr>
<tr><td><code id="sortMarker_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Change the order of factor levels for 'Marker' and 'Dye' according to 'kit'.
Levels in data must be identical with kit information.
</p>


<h3>Value</h3>

<p>data.frame with factor levels sorted according to 'kit'.
</p>

<hr>
<h2 id='strvalidator'>Graphical User Interface For The STR-validator Package</h2><span id='topic+strvalidator'></span>

<h3>Description</h3>

<p>GUI simplifying the use of the strvalidator package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strvalidator(debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strvalidator_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The graphical user interface give easy access to all graphical
versions of the functions available in the strvalidator package. It connects
functions 'under the hood' to allow a degree of automation not available
using the command based functions. In addition it provides a project based
workflow.<br /><br />
Click <code>Index</code> at the bottom of the help page to see a complete list
of functions.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'># To start the graphical user interface.
## Not run: 
strvalidator()

## End(Not run)

</code></pre>

<hr>
<h2 id='strvalidator-package'>Process Control and Internal Validation of Forensic STR Kits</h2><span id='topic+strvalidator-package'></span>

<h3>Description</h3>

<p>STR-validator is a free and open source R-package intended for
process control and internal validation of forensic STR DNA typing kit.
Its graphical user interface simplifies the analysis of data exported from e.g.
GeneMapper software, without extensive knowledge about R. It provides
functions to import, view, edit, and export data. After analysis the results,
generated plots, heat-maps, and data can be saved in a project for easy access.
Currently, analysis modules for stutter, balance, dropout, mixture, concordance,
typing result, precision, pull-up, and analytical thresholds are available.
In addition there are functions to analyze the GeneMapper bins- and panels files.
EPG like plots can be generate from data. STR-validator can greatly increase the speed of
validation by reducing the time and effort needed to analyze the validation
data. It allows exploration of the characteristics of DNA typing kits according
to ENFSI and SWGDAM recommendations. This facilitates the implementation of
probabilistic interpretation of DNA results.
</p>
<p>STR-validator was written and is maintained by Oskar Hansson, senior forensic
scientist at Oslo University Hospital (OUS), Section for Forensic Biology.
The work initially received external funding from the European
Union seventh Framework Programme (FP7/2007-2013) under grant agreement no
285487 (EUROFORGEN-NoE) but development and maintenance is now performed as
a part of my position at OUS, and on personal spare time.<br /><br />
</p>
<p>Effort has been made to assure correct results. Refer to the main website
for a list of functions specifically tested at build time.<br /><br />
</p>
<p>Click <code>Index</code> at the bottom of the page to see a complete list of
functions.<br /><br />
</p>
<p>Created and maintained by:<br />
Oskar Hansson, Section for Forensic Biology (OUS, Norway)<br /><br />
</p>
<p>More information can be found at:<br />
<a href="https://sites.google.com/site/forensicapps/strvalidator">https://sites.google.com/site/forensicapps/strvalidator</a><br /><br />
</p>
<p>Info and user community at Facebook:<br />
<a href="https://www.facebook.com/pages/STR-validator/240891279451450?ref=tn_tnmn">https://www.facebook.com/pages/STR-validator/240891279451450?ref=tn_tnmn</a><br /><br />
<a href="https://www.facebook.com/groups/strvalidator/">https://www.facebook.com/groups/strvalidator/</a><br /><br />
</p>
<p>The source code is hosted at GitHub:<br />
<a href="https://github.com/OskarHansson/strvalidator">https://github.com/OskarHansson/strvalidator</a><br /><br />
</p>
<p>Please report bugs to:<br />
<a href="https://github.com/OskarHansson/strvalidator/issues">https://github.com/OskarHansson/strvalidator/issues</a><br /><br />
</p>


<h3>Author(s)</h3>

<p>Oskar Hansson <a href="mailto:oskhan@ous-hf.no">oskhan@ous-hf.no</a>
</p>


<h3>References</h3>

<p>Recommended Minimum Criteria for the Validation of Various Aspects of the DNA Profiling Process
<a href="http://enfsi.eu/wp-content/uploads/2016/09/minimum_validation_guidelines_in_dna_profiling_-_v2010_0.pdf">http://enfsi.eu/wp-content/uploads/2016/09/minimum_validation_guidelines_in_dna_profiling_-_v2010_0.pdf</a>
Validation Guidelines for Forensic DNA Analysis Methods (2012)
<a href="http://media.wix.com/ugd/4344b0_cbc27d16dcb64fd88cb36ab2a2a25e4c.pdf">http://media.wix.com/ugd/4344b0_cbc27d16dcb64fd88cb36ab2a2a25e4c.pdf</a>
</p>

<hr>
<h2 id='trim'>Trim Data</h2><span id='topic+trim'></span>

<h3>Description</h3>

<p>Extract data from a dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trim(
  data,
  samples = NULL,
  columns = NULL,
  word = FALSE,
  ignore.case = TRUE,
  invert.s = FALSE,
  invert.c = FALSE,
  rm.na.col = TRUE,
  rm.empty.col = TRUE,
  missing = NA,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trim_+3A_data">data</code></td>
<td>
<p>data.frame with genotype data.</p>
</td></tr>
<tr><td><code id="trim_+3A_samples">samples</code></td>
<td>
<p>string giving sample names separated by pipe (|).</p>
</td></tr>
<tr><td><code id="trim_+3A_columns">columns</code></td>
<td>
<p>string giving column names separated by pipe (|).</p>
</td></tr>
<tr><td><code id="trim_+3A_word">word</code></td>
<td>
<p>logical indicating if a word boundary should be added to
<code>samples</code> and <code>columns</code>.</p>
</td></tr>
<tr><td><code id="trim_+3A_ignore.case">ignore.case</code></td>
<td>
<p>logical, TRUE ignore case in sample names.</p>
</td></tr>
<tr><td><code id="trim_+3A_invert.s">invert.s</code></td>
<td>
<p>logical, TRUE to remove matching samples from 'data',
FALSE to remove samples NOT matching (i.e. keep matching samples).</p>
</td></tr>
<tr><td><code id="trim_+3A_invert.c">invert.c</code></td>
<td>
<p>logical, TRUE to remove matching columns from 'data',
FALSE to remove columns NOT matching (i.e. keep matching columns).
while TRUE will remove columns NOT given.</p>
</td></tr>
<tr><td><code id="trim_+3A_rm.na.col">rm.na.col</code></td>
<td>
<p>logical, TRUE columns with only NA are removed from 'data'
while FALSE will preserve the columns.</p>
</td></tr>
<tr><td><code id="trim_+3A_rm.empty.col">rm.empty.col</code></td>
<td>
<p>logical, TRUE columns with no values are removed from 'data'
while FALSE will preserve the columns.</p>
</td></tr>
<tr><td><code id="trim_+3A_missing">missing</code></td>
<td>
<p>value to replace missing values with.</p>
</td></tr>
<tr><td><code id="trim_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies extraction of specific data from a larger dataset.
Look for samples in column named 'Sample.Name', 'Sample.File.Name', or
the first column containing the string 'Sample' in mentioned order
(not case sensitive). Remove unwanted columns.
</p>


<h3>Value</h3>

<p>data.frame with extracted result.
</p>

<hr>
<h2 id='trim_gui'>Trim Data</h2><span id='topic+trim_gui'></span>

<h3>Description</h3>

<p>GUI wrapper for the <code><a href="#topic+trim">trim</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trim_gui(env = parent.frame(), savegui = NULL, debug = FALSE, parent = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trim_gui_+3A_env">env</code></td>
<td>
<p>environment in which to search for data frames and save result.</p>
</td></tr>
<tr><td><code id="trim_gui_+3A_savegui">savegui</code></td>
<td>
<p>logical indicating if GUI settings should be saved in the environment.</p>
</td></tr>
<tr><td><code id="trim_gui_+3A_debug">debug</code></td>
<td>
<p>logical indicating printing debug information.</p>
</td></tr>
<tr><td><code id="trim_gui_+3A_parent">parent</code></td>
<td>
<p>widget to get focus when finished.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simplifies the use of the <code><a href="#topic+trim">trim</a></code> function by providing a graphical
user interface to it.
</p>


<h3>Value</h3>

<p>TRUE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trim">trim</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
