<!DOCTYPE html><html><head><title>Help for package subgxe</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {subgxe}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pasta'><p>pasta for multi-phenotype analysis</p></a></li>
<li><a href='#studies'><p>Synthetic data for subgxe</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Combine Multiple GWAS by Using Gene-Environment Interactions</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Classical methods for combining summary data from genome-wide
    association studies (GWAS) only use marginal genetic effects and power can
    be compromised in the presence of heterogeneity. 'subgxe' is a R package
    that implements p-value assisted subset testing for association (pASTA),
    a method developed by Yu et al. (2019) &lt;<a href="https://doi.org/10.1159%2F000496867">doi:10.1159/000496867</a>&gt;. pASTA
    generalizes association analysis based on subsets by incorporating
    gene-environment interactions into the testing procedure.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/umich-cphds/subgxe">https://github.com/umich-cphds/subgxe</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/umich-cphds/subgxe/issues">https://github.com/umich-cphds/subgxe/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>lmtest, knitr, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-06-14 11:26:19 UTC; alexrix</td>
</tr>
<tr>
<td>Author:</td>
<td>Youfei Yu [aut],
  Alexander Rix [cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alexander Rix &lt;alexrix@umich.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-06-14 13:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='pasta'>pasta for multi-phenotype analysis</h2><span id='topic+pasta'></span>

<h3>Description</h3>

<p>Search for the subset that yields the strongest evidence of
association and calculate the meta-analytic p-value, possibly in the
presence of gene-environmental interaction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pasta(p.values, study.sizes, cor)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pasta_+3A_p.values">p.values</code></td>
<td>
<p>The p.value of each study.</p>
</td></tr>
<tr><td><code id="pasta_+3A_study.sizes">study.sizes</code></td>
<td>
<p>The sample size of each study.</p>
</td></tr>
<tr><td><code id="pasta_+3A_cor">cor</code></td>
<td>
<p>The correlation matrix of the studies. For example, if each study
is independent, <code>cor</code> would be the identity matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the joint p value and the test statistic, which
contains the optimal subset.
</p>


<h3>References</h3>

<p>Yu Y, Xia L, Lee S, Zhou X, Stringham H, M, Boehnke M, Mukherjee
B: Subset-Based Analysis Using Gene-Environment Interactions for Discovery
of Genetic Associations across Multiple Studies or Phenotypes. Hum Hered
2019. doi: 10.1159/000496867
</p>


<h3>Examples</h3>

<pre><code class='language-R'># grab synthetic study for example
data("studies")
n.studies &lt;- 5
study.sizes &lt;- c(nrow(studies[[1]]), nrow(studies[[2]]), nrow(studies[[3]]),
                   nrow(studies[[4]]), nrow(studies[[5]]))
study.pvals &lt;- rep(0, n.studies)
# Correlations of p-values among the studies.
# In this case the studies were generated independently so its just I
cor.matrix &lt;- diag(1, n.studies)
# load the lrtest() function to conduct the likelihood ratio test
# Used just to generate the input p-values, not required in pasta itself.

library(lmtest)

for(i in 1:n.studies) {
 # model with gene(G) by environment(E) interaction
 model &lt;- glm(D ~ G + E + GbyE, data = studies[[i]], family = binomial)
 # model without G and GE interaction
 null.model &lt;- glm(D ~ E, data = studies[[i]], family = binomial)
 # likelihood ratio test from the package lmtest
 study.pvals[i] = lmtest::lrtest(null.model, model)[2, 5]
}

pasta &lt;- pasta(study.pvals, study.sizes, cor.matrix)

pasta$p.pasta
pasta$test.statistic$selected.subset
</code></pre>

<hr>
<h2 id='studies'>Synthetic data for subgxe</h2><span id='topic+studies'></span>

<h3>Description</h3>

<p>Synthetic data for subgxe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studies
</code></pre>


<h3>Format</h3>

<p>A list of 5 <code>data.frames</code> with 12000 observations
(6000 cases, 6000 controls) on 4 variables:
</p>

<dl>
<dt>D</dt><dd><p>Disease status. Numeric 0-1</p>
</dd>
<dt>G</dt><dd><p>Genetic variant. Numeric 0-1</p>
</dd>
<dt>E</dt><dd><p>Exposure. Numeric 0-1</p>
</dd>
<dt>GbyE</dt><dd><p><code>G * E</code>. Either 1 or 0.</p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
