<!DOCTYPE html><html><head><title>Help for package SubpathwayLNCE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SubpathwayLNCE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#envData'><p>The variables in the environment variable <code>envData</code> of the system</p></a></li>
<li><a href='#getBackgroundLnc'>
<p>Get the background of molecules</p></a></li>
<li><a href='#GetExampleData'>
<p>Get the environment data</p></a></li>
<li><a href='#getInteGraphList'><p>Get the reconstructed signal pathway graphs</p></a></li>
<li><a href='#getInteUMGraph'><p>Get the reconstructed condition-specific lncRNA-regulated signal pathways</p></a></li>
<li><a href='#getLncGenePairs'><p>Get the co-express lncRNA-mRNA interactions</p></a></li>
<li><a href='#getLocSubGraphLnc'><p>Get the located signal subpathways</p></a></li>
<li><a href='#identifyLncGraphW'><p>Annotate and identify subpathways</p></a></li>
<li><a href='#plotAnnGraph'><p>Visualize annotated pathway graphs</p></a></li>
<li><a href='#plotGraphL'><p>Visualize a pathway graph</p></a></li>
<li><a href='#printGraphW'><p>Print the results of graph annotation and identification</p></a></li>
<li><a href='#SubpathwayLNCE-internal'><p>SubpathwayLNCE internal functions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identify Signal Subpathways Competitively Regulated by LncRNAs
Based on ceRNA Theory</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-1-15</td>
</tr>
<tr>
<td>Author:</td>
<td>Xinrui Shi, Chunquan Li and Xia Li</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xinrui Shi &lt;xinrui103@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Identify dysfunctional subpathways competitively regulated by lncRNAs through integrating lncRNA-mRNA expression profile and pathway topologies. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), igraph, RBGL, utils, BiasedUrn, graph, graphics,
stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>XML</td>
</tr>
<tr>
<td>Collate:</td>
<td>ExpProcess.r getBackgroundLnc.r getEdgeLabel.r getEdgeLty.r
getEnvironmentData.r getInteGraphList.r getInteUMGraph.r
getKGeneFromGene.r getLayout.r getLncGenePairs.r
getExampleData.r getLocSubGraphLnc.r getNodeLabel.r
getOneNodePath.r getSymbolFromGene.r identifyLncGraphW.r
plotAnnGraph.r plotGraphL.r printGraphW.r</td>
</tr>
<tr>
<td>LazyData:</td>
<td>Yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Statistics, SuPathways, LncRNAs, enrichment analysis,ceRNA</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-01-19 01:50:31 UTC; Administrator</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-01-19 08:17:59</td>
</tr>
</table>
<hr>
<h2 id='envData'>The variables in the environment variable <code>envData</code> of the system</h2><span id='topic+envData'></span>

<h3>Description</h3>

<p>The variables in the environment variable <code>envData</code> of the system.
</p>


<h3>Format</h3>

<p>An environment variable</p>


<h3>Author(s)</h3>

<p>Xinrui Shi, Chunquan Li and Xia Li</p>

<hr>
<h2 id='getBackgroundLnc'>
Get the background of molecules
</h2><span id='topic+getBackgroundLnc'></span>

<h3>Description</h3>

<p>getBackgroundLnc attempts to get the background of user-specified molecules.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBackgroundLnc(type = "gene_lncRNA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBackgroundLnc_+3A_type">type</code></td>
<td>

<p>A character string. Should be one of &quot;gene&quot;, &quot;lncRNA&quot; or &quot;gene_lncRNA&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default background is obtained from the environment variable. For human, the reference gene background is all human genes in KEGG pathways. The reference lncRNA background is collected from GeneCode database. 
</p>


<h3>Value</h3>

<p>A character vector. 
</p>


<h3>Author(s)</h3>

<p>Xinrui Shi, Chunquan Li and Xia Li
</p>


<h3>See Also</h3>

<p><code><a href="#topic+identifyLncGraphW">identifyLncGraphW</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## get all background of genes
 bgGene &lt;- getBackgroundLnc(type="gene")

## get all background of lncRNAs
 bgLncRNA  &lt;- getBackgroundLnc(type="lncRNA")

## get all background of lncRNAs and genes
 bgGL &lt;- getBackgroundLnc(type="gene_lncRNA")

## End(Not run)
</code></pre>

<hr>
<h2 id='GetExampleData'>
Get the environment data
</h2><span id='topic+GetExampleData'></span>

<h3>Description</h3>

<p>Get variables in current environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetExampleData(exampleData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetExampleData_+3A_exampledata">exampleData</code></td>
<td>

<p>A character string. It must be one of them, including &quot;code&quot;, &quot;compound&quot;, &quot;g2&quot;, &quot;gene&quot;, &quot;gene2path&quot;, &quot;gene2symbol&quot;, &quot;GeneExp&quot;, &quot;keggGene2gene&quot;, &quot;lnc2Name&quot;, &quot;lncBackground&quot;, &quot;LncExp&quot;, &quot;LncGenePairs&quot;, &quot;mart&quot;, &quot;nocode&quot;, &quot;pp&quot;, &quot;resultT&quot; and &quot;sub&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parameter exampleData is &quot;code&quot;, which represents to obtain all human coding gene collected from Genecode.
</p>
<p>The parameter K2riData is &quot;compound&quot;, which represents to obtain all huamn compound.
</p>
<p>The parameter K2riData is &quot;g2&quot;, which represents to obtain undirect KEGG metabolic pathway graphs with genes as nodes.
</p>
<p>The parameter K2riData is &quot;gene&quot;, which represents to obtain interseted gene of example.
</p>
<p>The parameter K2riData is &quot;gene2symbol&quot;, which represents to obtain gene-symbol data.
</p>
<p>The parameter K2riData is &quot;gene2path&quot;, which represents to obtain gene-pathway data.
</p>
<p>The parameter K2riData is &quot;GeneExp&quot;, which represents to obtain gene expression profile of example.
</p>
<p>The parameter K2riData is &quot;keggGene2gene&quot;, which represents to obtain kegggene-gene data.
</p>
<p>The parameter K2riData is &quot;lnc2Name&quot;, which represents to obtain lncRNA Ensemble-Name data.
</p>
<p>The parameter K2riData is &quot;lncBackground&quot;, which represents to obtain background of lncRNA, gene or lncRNA-gene.
</p>
<p>The parameter K2riData is &quot;LncExp&quot;, which represents to obtain lncRNA expression profile of example.
</p>
<p>The parameter K2riData is &quot;LncGenePairs&quot;, which represents to obtain the cmepetitively regulated lncRNA-mRNA interactions.
</p>
<p>The parameter K2riData is &quot;mart&quot;, which represents to obtain gene ensemble-name-entrez data.
</p>
<p>The parameter K2riData is &quot;nocode&quot;, which represents to obtain all huaman lncRNA collected from Genecode.
</p>
<p>The parameter K2riData is &quot;resultT&quot;, which represents to obtain result of example.
</p>
<p>The parameter K2riData is &quot;sub&quot;, which represents to obtain subpahtway of example.
</p>
<p>The parameter K2riData is &quot;geneLnc&quot;, which represents to obtain user-interested genes and lncRNAs of example.
</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li and Xia Li</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 


# obtain lncRNA expression profile #
LncExp &lt;- GetExampleData(exampleData="LncExp")



## End(Not run)
</code></pre>

<hr>
<h2 id='getInteGraphList'>Get the reconstructed signal pathway graphs</h2><span id='topic+getInteGraphList'></span>

<h3>Description</h3>

<p>Get the reconstructed KEGG signal pathway graphs embedded by lncRNAs of competitive regulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getInteGraphList(graphList,relations)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getInteGraphList_+3A_graphlist">graphList</code></td>
<td>
<p>A graphList. There nodes must be represented by genes.</p>
</td></tr>
<tr><td><code id="getInteGraphList_+3A_relations">relations</code></td>
<td>
<p>A data frame. It contains two columns, the first is lncRNA names and the second is its target names.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument &quot;relations&quot; represents user-interested lncRNA-target interactions, which can be returned from the <code>GetExampleData</code>.
</p>


<h3>Value</h3>

<p>A graph list. 
</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li and Xia Li</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotGraphL">plotGraphL</a></code>, <code><a href="#topic+getLocSubGraphLnc">getLocSubGraphLnc</a></code>, <code><a href="#topic+GetExampleData">GetExampleData</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
### Integrate lncRNAs of competitive regulation into KEGG pathway graphs ###
  LncGenePairs&lt;-GetExampleData(exampleData="LncGenePairs")
  g2&lt;-GetExampleData(exampleData="g2")
  interUMGraph&lt;-getInteGraphList(g2,LncGenePairs)
  #interUMGraph&lt;-getInteUMGraph(LncGenePairs)
### get user-interested lncRNAs and genes sets.
  #geneLnc&lt;-GetExampleData(exampleData="geneLnc")
# get locate subpathways.
  #sub&lt;-getLocSubGraphLnc(geneLnc,interUMGraph,type="gene_lncRNA",n=1,s=8)
  #SubcodeLncResult&lt;-identifyLncGraphW(geneLnc,sub,type="gene_lncRNA",bet=1)
  #resultT&lt;-printGraphW(SubcodeLncResult,detail=TRUE)

 
## End(Not run)
</code></pre>

<hr>
<h2 id='getInteUMGraph'>Get the reconstructed condition-specific lncRNA-regulated signal pathways</h2><span id='topic+getInteUMGraph'></span>

<h3>Description</h3>

<p>Get the reconstructed condition-specific LncRNA-regulated signal pathways embedded by lncRNAs of competitive regulation through linking to their regulated-mRNAs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getInteUMGraph(LncGenePairs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getInteUMGraph_+3A_lncgenepairs">LncGenePairs</code></td>
<td>
<p>A data frame. It contains two columns, the first is lncRNA names and the second is mRNA names.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument &quot;LncGenePairs&quot; represents competitively regulated lncRNA-mRNA interactions, which can be returned from the <code>getLncGenePairs</code>.
</p>


<h3>Value</h3>

<p>A graph list. 
</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li and Xia Li</p>


<h3>See Also</h3>

<p><code><a href="#topic+getLncGenePairs">getLncGenePairs</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

### Integrate lncRNAs of competitive regulation into KEGG pathway graphs ###
  LncGenePairs&lt;-GetExampleData(exampleData="LncGenePairs")
  inteUMGraph&lt;-getInteUMGraph(LncGenePairs)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='getLncGenePairs'>Get the co-express lncRNA-mRNA interactions</h2><span id='topic+getLncGenePairs'></span>

<h3>Description</h3>

<p>Get the co-express lncRNA-mRNA interactions reached a significant positive threshold. These lncRNAs-mRNAs pairs are competitivly rugulated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLncGenePairs(GeneExp,LncExp,a=0.025)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLncGenePairs_+3A_geneexp">GeneExp</code></td>
<td>
<p>A gene expression profile. The first row of gene expression profile is names of mRNA, and the gene expression profile and the lncRNA expression profile must be mached. </p>
</td></tr>
<tr><td><code id="getLncGenePairs_+3A_lncexp">LncExp</code></td>
<td>
<p>A lncRNA expression profile. The first row of lncRNA expression profile is names of lncRNA, and the gene expression profile and the lncRNA expression profile must be mached.</p>
</td></tr>
<tr><td><code id="getLncGenePairs_+3A_a">a</code></td>
<td>
<p>A parameter,namely the significant positive threshold.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Co-express coefficient for any pair of relations in the candidate LncRNA-mRNA network is calculated basing on matched LncRNA and mRNA expression profiles, those relations had reached a significant positive threshold were retained.
</p>


<h3>Value</h3>

<p>A data frame. The data frame has two columns, the fist is lncRNA names, the other is mRNA names.
</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li and Xia Li</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 


 GeneExp&lt;-GetExampleData(exampleData="GeneExp")
 LncExp&lt;-GetExampleData(exampleData="LncExp")
 LncGenePairs&lt;-getLncGenePairs(GeneExp,LncExp,a=0.025)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='getLocSubGraphLnc'>Get the located signal subpathways
</h2><span id='topic+getLocSubGraphLnc'></span>

<h3>Description</h3>

<p>Locate signal subpathways competitively regulated by lncRNAs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLocSubGraphLnc(moleculeList, graphList, type="gene_lncRNA", 
   n=1, s=8, method = "shortestPaths")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLocSubGraphLnc_+3A_moleculelist">moleculeList</code></td>
<td>
<p>A character vector. Such as competitively regulated lncRNAs and/or differentially expressed genes under disease phenotypes. Gene should be geneid.</p>
</td></tr>
<tr><td><code id="getLocSubGraphLnc_+3A_graphlist">graphList</code></td>
<td>
<p>A graph list. There nodes must be represented by genes. </p>
</td></tr> 
<tr><td><code id="getLocSubGraphLnc_+3A_type">type</code></td>
<td>
<p>A character string. Should be one of &quot;gene&quot;, &quot;lncRNA&quot; or &quot;gene_lncRNA&quot;.</p>
</td></tr>
<tr><td><code id="getLocSubGraphLnc_+3A_n">n</code></td>
<td>
<p>An integer. The maximum acceptable quantities of non-signature node at the shortest path between each two interested molecules.</p>
</td></tr>  
<tr><td><code id="getLocSubGraphLnc_+3A_s">s</code></td>
<td>
<p>An integer. The minimum acceptable quantities of nodes in located subpathways.</p>
</td></tr>  
<tr><td><code id="getLocSubGraphLnc_+3A_method">method</code></td>
<td>
<p>A character string. In which the shorest path algorithms will be used.</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>We apply lenient distance similarity method to locate signal subpathways competitively regulated by miRNAs. We first map user competitively regulated lncRNAs and/or interested genes to pathways as signatures. For a given pathway, we compute the shortest path between any two signatures. In shortest path, if the number of non-signature nodes between two signatures is no more than n, then these two signature nodes and other nodes at the shortest path are added into the same node set. We extract the corresponding subgraph in the pathway graph according to each node set. We finally define these subgraphs with node number &gt;= s as the subpathway regions of the pathway. 
The argument <code>n</code> is maximum number of permitted non-signature nodes at the shortest path between signature nodes. The default parameter n=1.
The argument <code>s</code> is used to filter subpathways in which the number of nodes are less than the parameter s. The default parameter s=8.
The argument <code>method</code> determines which shortest path algorithms will be used. We set the default value as &quot;get.shortest.paths&quot;. 
</p>


<h3>Value</h3>

<p> A list of graphs.</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li and Xia Li</p>


<h3>See Also</h3>

<p><code><a href="#topic+getInteGraphList">getInteGraphList</a></code>, <code><a href="#topic+identifyLncGraphW">identifyLncGraphW</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

### Integrate lncRNAs of competitive regulation into KEGG pathway graphs ###
  LncGenePairs&lt;-GetExampleData(exampleData="LncGenePairs")
  interUMGraph&lt;-getInteUMGraph(LncGenePairs)
### get user-interested lncRNAs and genes sets.
##geneLnc&lt;-c(getBackground(type="gene")[1:3000],unique(LncGenePairs[1,]))
  geneLnc&lt;-GetExampleData(exampleData="geneLnc")
# get locate subpathways.
  sub&lt;-getLocSubGraphLnc(geneLnc,interUMGraph,type="gene_lncRNA",n=1,s=8)
  

## End(Not run)
</code></pre>

<hr>
<h2 id='identifyLncGraphW'>Annotate and identify subpathways</h2><span id='topic+identifyLncGraphW'></span>

<h3>Description</h3>

<p>Annotate user-interested molecules to pathways and identify significantly enriched subpathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  identifyLncGraphW(moleculeList,graphList,type="gene_lncRNA",
             background=getBackgroundLnc(type),
             order="pvalue",decreasing=FALSE,bet=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="identifyLncGraphW_+3A_moleculelist">moleculeList</code></td>
<td>
<p>A character vector. Such as user-interested lncRNAs and/or genes under disease phenotypes.</p>
</td></tr>
<tr><td><code id="identifyLncGraphW_+3A_graphlist">graphList</code></td>
<td>
<p>A graph list. There nodes must be represented by genes or lncRNAs and genes.</p>
</td></tr> 
<tr><td><code id="identifyLncGraphW_+3A_type">type</code></td>
<td>
<p>A character string. Should be one of &quot;gene&quot;, &quot;lncRNA&quot; or &quot;gene_lncRNA&quot;.</p>
</td></tr>   
<tr><td><code id="identifyLncGraphW_+3A_background">background</code></td>
<td>
<p>A character vector of molecules.</p>
</td></tr>
<tr><td><code id="identifyLncGraphW_+3A_order">order</code></td>
<td>
<p>A character string. Should be one of &quot;pvalue&quot;, &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="identifyLncGraphW_+3A_decreasing">decreasing</code></td>
<td>
<p>A logical. Should the sort be ordered by increasing or decreasing?</p>
</td></tr>
<tr><td><code id="identifyLncGraphW_+3A_bet">bet</code></td>
<td>
<p>A number. A parameter. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can support the annotation and identification of metabolic subpathways based on genes, lncRNAs or gene_lncRNAs sets. The argument <code>moleculeList</code> supports three kinds of molecular sets: &quot;genes&quot;, &quot;lncRNAs&quot; or &quot;gene_lncRNAs&quot;. 
</p>
<p>The argument <code>type</code> represent the type of input molecules, including one of &quot;genes&quot;, &quot;lncRNAs&quot; or &quot;gene_lncRNA&quot;. 
</p>
<p>Detailed <code>background</code> information is provided in the function <code><a href="#topic+getBackgroundLnc">getBackgroundLnc</a></code>.
</p>
<p>When many correlated subpathways are considered, the parameter <code>order</code> is used to order the pathways on the basis of &quot;pvalue&quot; or &quot;fdr&quot;. 
</p>
<p>The parameter <code>decreasing</code> is set TRUE that represent the order would be performed by decreasing.
</p>


<h3>Value</h3>

<p>A list. It include: 'pathwayId', 'pathwayName', 'annMoleculeList', 'annMoleculeNumber', 'annBgMoleculeList', 'annBgNumber', 'MoleculeNumber', 'bgNumber','annWeight', 'pvalue', and 'fdr',  corresponding to pathway identifier, pathway name, the submitted molecules annotated to a pathway, the number of submitted molecules annotated to a pathway, the background molecules annotated to a pathway, the number of background molecules annotated to a pathway, the number of submitted molecules, the number of background molecules, the weight of lncRNA competitively regulated subpathway p-value of the hypergeometric test, and Benjamini-Hochberg fdr values.
</p>
<p>The background molecules annotated to a pathway are equal to all molecules in the pathway. For example, if the submitted molecules are human genes, the background molecules annotated to a pathway are equal to all human genes in the pathway.
</p>
<p>The number of background molecules is the number of all molecules. For example, if the submitted molecules are human genes, the number of background molecules is equal to all human genes.
</p>
<p>To visualize and save the results, the <code>list</code> can be converted to the <code>data.frame</code> by the function <code><a href="#topic+printGraphW">printGraphW</a></code>. 
</p>
<p>Note that <code>moleculeList</code> must be a 'character' vector. The genes must be represented by NCBI gene ids, and lncRNAs must be represented by mature lncRNA Ensemble name.
</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li and Xia Li</p>


<h3>See Also</h3>

<p><code><a href="#topic+printGraphW">printGraphW</a></code>, <code><a href="#topic+getBackgroundLnc">getBackgroundLnc</a></code>, <code><a href="#topic+GetExampleData">GetExampleData</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

### Integrate lncRNAs of competitive regulation into KEGG pathway graphs ###
  #LncGenePairs&lt;-GetExampleData(exampleData="LncGenePairs")
  #inteUMGraph&lt;-getInteUMGraph(LncGenePairs)
### get user-interested lncRNAs and genes sets.
  geneLnc&lt;-GetExampleData(exampleData="geneLnc")
# get locate subpathways.
  #sub&lt;-getLocSubGraphLnc(geneLnc,interUMGraph,type="gene_lncRNA",n=1,s=8)
   sub&lt;-GetExampleData(exampleData="sub")
  SubcodeLncResult&lt;-identifyLncGraphW(geneLnc,sub,type="gene_lncRNA",bet=1)

## End(Not run)
</code></pre>

<hr>
<h2 id='plotAnnGraph'>Visualize annotated pathway graphs</h2><span id='topic+plotAnnGraph'></span>

<h3>Description</h3>

<p>Visualize annotated pathway graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	plotAnnGraph(pathwayId,graphList,ann,gotoKEGG=FALSE,orgSpecific=TRUE,
	displayInR=TRUE,match=TRUE,vertex.frame.color="red",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAnnGraph_+3A_pathwayid">pathwayId</code></td>
<td>
<p>A character vector of pathway identifier, e.g., c(&quot;path:00010&quot;,&quot;path:00020&quot;).</p>
</td></tr>
<tr><td><code id="plotAnnGraph_+3A_graphlist">graphList</code></td>
<td>
<p>A graph list.</p>
</td></tr>
<tr><td><code id="plotAnnGraph_+3A_ann">ann</code></td>
<td>
<p>A list, e.g., the return value of the function <code><a href="#topic+identifyLncGraphW">identifyLncGraphW</a></code>.</p>
</td></tr>
<tr><td><code id="plotAnnGraph_+3A_gotokegg">gotoKEGG</code></td>
<td>
<p>A logical. If TRUE, go to KEGG web site. The identified nodes are colored red.</p>
</td></tr>
<tr><td><code id="plotAnnGraph_+3A_orgspecific">orgSpecific</code></td>
<td>
<p>A logical. </p>
</td></tr>
<tr><td><code id="plotAnnGraph_+3A_match">match</code></td>
<td>
<p>A logical.</p>
</td></tr>
<tr><td><code id="plotAnnGraph_+3A_displayinr">displayInR</code></td>
<td>
<p>A logical. If TRUE, display pathway graphs on the device.</p>
</td></tr>
<tr><td><code id="plotAnnGraph_+3A_vertex.frame.color">vertex.frame.color</code></td>
<td>
<p>A character defining the frame color of the nodes which represent the annotated nodes in the pathway.</p>
</td></tr>
<tr><td><code id="plotAnnGraph_+3A_...">...</code></td>
<td>
<p> The arguments passed to or from methods. See <code><a href="#topic+plotGraphL">plotGraphL</a></code>, <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> and see <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>The function can visualize the annotated pathway (or subpathways) graphs. The red nodes in the result graph represent the nodes which include the submitted molecules.
</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li, Xia Li</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotGraphL">plotGraphL</a></code>,<code><a href="#topic+identifyLncGraphW">identifyLncGraphW</a></code>,<code><a href="#topic+printGraphW">printGraphW</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
### Integrate lncRNAs of competitive regulation into KEGG pathway graphs ###
  LncGenePairs&lt;-GetExampleData(exampleData="LncGenePairs")
  g2&lt;-GetExampleData(exampleData="g2")
  interUMGraph&lt;-getInteGraphList(g2,LncGenePairs)
  interUMGraph&lt;-getInteUMGraph(LncGenePairs)
### get user-interested lncRNAs and genes sets.
  geneLnc&lt;-GetExampleData(exampleData="geneLnc")
# get locate subpathways.
  sub&lt;-getLocSubGraphLnc(geneLnc,interUMGraph,type="gene_lncRNA",n=1,s=8)
  SubcodeLncResult&lt;-identifyLncGraphW(geneLnc,sub,type="gene_lncRNA",bet=1)
  resultT&lt;-printGraphW(SubcodeLncResult,detail=TRUE)
  plotAnnGraph("path:04020_1",sub,SubcodeLncResult,gotoKEGG=FALSE)


## End(Not run)
</code></pre>

<hr>
<h2 id='plotGraphL'>Visualize a pathway graph</h2><span id='topic+plotGraphL'></span>

<h3>Description</h3>

<p>Visualize a pathway graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGraphL(graph,margin=0,vertex.label.cex=0.6,vertex.label.font=1,
  vertex.size=8,vertex.size2=6,edge.arrow.size=0.2,
  edge.arrow.width=3,vertex.label=V(graph)$graphics_name,
  vertex.shape=V(graph)$graphics_type,layout=getLayout(graph),
  vertex.label.color="black",vertex.color=V(graph)$graphics_bgcolor,
  vertex.frame.color="dimgray",edge.color="dimgray",
  edge.label=getEdgeLabel(graph),edge.label.cex=0.6,
  edge.label.color="dimgray",edge.lty=getEdgeLty(graph),
  axes=FALSE,xlab="",ylab="",sub=NULL,main=NULL,...)
 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotGraphL_+3A_graph">graph</code></td>
<td>
<p> The igraph object of a pathway graph.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_margin">margin</code></td>
<td>
<p> A numeric. The value is usually between -0.5 and 0.5, which is able to zoom in or out a pathway graph. The default is 0.</p>
</td></tr> 
<tr><td><code id="plotGraphL_+3A_vertex.label.cex">vertex.label.cex</code></td>
<td>
<p> A numeric vector of node label size.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_vertex.label.font">vertex.label.font</code></td>
<td>
<p> A numeric vector of label font.</p>
</td></tr> 
<tr><td><code id="plotGraphL_+3A_vertex.size">vertex.size</code></td>
<td>
<p> A numeric vector of Node size. See <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code></p>
</td></tr> 
<tr><td><code id="plotGraphL_+3A_vertex.size2">vertex.size2</code></td>
<td>
<p> A numeric vector of Node size.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_edge.arrow.size">edge.arrow.size</code></td>
<td>
<p> Edge arrow size.The default is 0.2.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_edge.arrow.width">edge.arrow.width</code></td>
<td>
<p> Edge arrow width. The default is 3.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_vertex.label">vertex.label</code></td>
<td>
<p> A vector of node label. The default is graphics_name.</p>
</td></tr>  
<tr><td><code id="plotGraphL_+3A_vertex.shape">vertex.shape</code></td>
<td>
<p> A vector of node shape. The default is graphics_type.</p>
</td></tr> 
<tr><td><code id="plotGraphL_+3A_layout">layout</code></td>
<td>
<p> A matrix of x-y coordinates with two dims. Determine the placement of the nodes for drawing a graph. The default is the KEGG node coordinates that are originally obtained from the KGML file.</p>
</td></tr>  
<tr><td><code id="plotGraphL_+3A_vertex.label.color">vertex.label.color</code></td>
<td>
<p> A vector of node label colors. The default is black.</p>
</td></tr>    
<tr><td><code id="plotGraphL_+3A_vertex.color">vertex.color</code></td>
<td>
<p> A vector of node colors. The default is the KEGG node color.</p>
</td></tr>  
<tr><td><code id="plotGraphL_+3A_vertex.frame.color">vertex.frame.color</code></td>
<td>
<p> A vector of node frame color. The default is dimgray.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_edge.color">edge.color</code></td>
<td>
<p> A vector of edge color. The default is dimgray.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_edge.label">edge.label</code></td>
<td>
<p> A vector of edge label.</p>
</td></tr>  
<tr><td><code id="plotGraphL_+3A_edge.label.cex">edge.label.cex</code></td>
<td>
<p> Edge label size.</p>
</td></tr>  
<tr><td><code id="plotGraphL_+3A_edge.label.color">edge.label.color</code></td>
<td>
<p> A vector of edge label color. The default is dimgray.</p>
</td></tr>  
<tr><td><code id="plotGraphL_+3A_edge.lty">edge.lty</code></td>
<td>
<p> A vector of line type for the edges.</p>
</td></tr>    
<tr><td><code id="plotGraphL_+3A_axes">axes</code></td>
<td>
<p> A logical. whether to plot axes. The default is FALSE.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_xlab">xlab</code></td>
<td>
<p> A character string. The label of the horizontal axis. The default is the empty string.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_ylab">ylab</code></td>
<td>
<p> A character string. The label of the vertical axis. The default is the empty string.</p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_sub">sub</code></td>
<td>
<p> A character string of subtitle. </p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_main">main</code></td>
<td>
<p> A character string of main title. </p>
</td></tr>
<tr><td><code id="plotGraphL_+3A_...">...</code></td>
<td>
<p> The arguments passed to or from methods. See <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>The function <code>plotGraphL</code> is able to display a pathway graph. 
</p>
<p>The argument <code>layout</code> is used to determine the placement of the nodes for drawing a graph. There are mainly two preprocessed methods to determine the placement of the nodes for drawing a pathway graph: the KEGG pathway layout and <code><a href="graphics.html#topic+layout">layout</a></code> provided in the function <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code> of the <code>igraph</code> package. The default layout is the KEGG layout, for which the coordinates of nodes in KEGG is used to determine the placement of the nodes for drawing a graph. Therefore, the returned figure by the function may be very similar to the KEGG pathway graph when information in the pathway graph is complete relatively. The layouts provided in <code>igraph</code> include <code>layout.reingold.tilford</code>, <code>layout.random</code>, <code>layout.circle</code>, <code>layout.sphere</code>,..., .
</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li and Xia Li</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+plot">plot</a></code>, <code><a href="igraph.html#topic+layout.random">layout.random</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
###  get  signal pathway graphs   ###

g&lt;- GetExampleData(exampleData="g2")

# visualize the graph 
plotGraphL(g[[9]],layout=layout.random)


## End(Not run)
</code></pre>

<hr>
<h2 id='printGraphW'>Print the results of graph annotation and identification</h2><span id='topic+printGraphW'></span>

<h3>Description</h3>

<p>Print the results of graph annotation and identification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>     printGraphW(ann,detail=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printGraphW_+3A_ann">ann</code></td>
<td>
<p> A list. The value was returned from the function <code><a href="#topic+identifyLncGraphW">identifyLncGraphW</a></code>.</p>
</td></tr>
<tr><td><code id="printGraphW_+3A_detail">detail</code></td>
<td>
<p> A logical. If true, gene lists from the function <code><a href="#topic+identifyLncGraphW">identifyLncGraphW</a></code> are converted into strings, which are used to display and write results with genes. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame. Columns include pathwayId, pathwayName, annMoleculeRatio, annBgRatio,annWeight, pvalue, 'fdr', annMoleculeList, annBgMoleculeList. Detailed information is provided in <code><a href="#topic+identifyLncGraphW">identifyLncGraphW</a></code>.
</p>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li and Xia Li</p>


<h3>See Also</h3>

<p><code><a href="#topic+identifyLncGraphW">identifyLncGraphW</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
### Integrate lncRNAs of competitive regulation into KEGG pathway graphs ###
  LncGenePairs&lt;-GetExampleData(exampleData="LncGenePairs")
  inteUMGraph&lt;-getInteUMGraph(LncGenePairs)
### get user-interested lncRNAs and genes sets.
  geneLnc&lt;-GetExampleData(exampleData="geneLnc")
# get locate subpathways.
  sub&lt;-getLocSubGraphLnc(geneLnc,inteUMGraph,type="gene_lncRNA",n=1,s=8)
  SubcodeLncResult&lt;-identifyLncGraphW(geneLnc,sub,type="gene_lncRNA",bet=1)
  resultT&lt;-printGraphW(SubcodeLncResult,detail=TRUE)



## End(Not run)
</code></pre>

<hr>
<h2 id='SubpathwayLNCE-internal'>SubpathwayLNCE internal functions</h2><span id='topic+initialize'></span><span id='topic+Getenvir'></span><span id='topic+getEdgeLabel'></span><span id='topic+getEdgeLty'></span><span id='topic+getLayout'></span><span id='topic+ExpProcess'></span><span id='topic+getKGeneFromGene'></span><span id='topic+getNodeLabel'></span><span id='topic+getOneNodePath'></span><span id='topic+getSymbolFromGene'></span>

<h3>Description</h3>

<p>These functions internally called by other functions in the package <code>SupathwayLNCE</code> and not meant to be called by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
    initialize()
    Getenvir(envData)
	getEdgeLabel(graph)
    getEdgeLty(graph)
	getLayout(graph)
	ExpProcess(ExpMatrix,pert)
	getKGeneFromGene(geneList)
	getNodeLabel(graph,type="symbol",displayNumber=1)
	getOneNodePath(current_node,other_nodes,pathway,
	n,all_shortest_paths_length,directed,method="shortestPaths")
	getSymbolFromGene(geneList)
	
</code></pre>


<h3>Author(s)</h3>

<p> Xinrui Shi, Chunquan Li, Xia Li</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
