<!DOCTYPE html><html><head><title>Help for package SubtypeDrug</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SubtypeDrug}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AccumulateNormal'><p>SubtypeDrug internal function</p></a></li>
<li><a href='#CalculateSES'><p>SubtypeDrug internal function</p></a></li>
<li><a href='#Colork'><p>Color</p></a></li>
<li><a href='#Disease_drugs'><p>Simulated result data</p></a></li>
<li><a href='#Drugs_CID'><p>Correspondence table of drug label and drug ID in PubCham database</p></a></li>
<li><a href='#Geneexp'><p>Simulated gene expression data</p></a></li>
<li><a href='#GeneexpT'><p>Gene expression data for testing</p></a></li>
<li><a href='#getDrugMatrix'><p>SubtypeDrug internal function</p></a></li>
<li><a href='#getDrugSpw'><p>SubtypeDrug internal function</p></a></li>
<li><a href='#getDrugStructure'><p>Get drug chemical structure diagram data</p></a></li>
<li><a href='#isPackageLoaded'><p>SubtypeDrug internal function</p></a></li>
<li><a href='#plotDScoreHeatmap'><p>Plot a heat map of the normalized drug-disease reverse association scores for cancer samples</p></a></li>
<li><a href='#plotDSpwHeatmap'><p>Plot heat map of the drug regulated subpathway activity score</p></a></li>
<li><a href='#plotGlobalGraph'><p>Plot a global graph of the drug</p></a></li>
<li><a href='#plotSpwNetGraph'><p>Polt a subpathway network graph</p></a></li>
<li><a href='#PrioSubtypeDrug'><p>Prioritization of candidate cancer subtype-specific drugs (PrioSubtypeDrug)</p></a></li>
<li><a href='#ReadClsFile'><p>SubtypeDrug internal function</p></a></li>
<li><a href='#SpwNetworkData'><p>Subpathway network structure data</p></a></li>
<li><a href='#Subtype_drugs'><p>Simulation result data</p></a></li>
<li><a href='#UserDS'><p>Simulated user-defined drug regulator subpathway dataset</p></a></li>
<li><a href='#UserDST'><p>User-defined drug regulator subpathway dataset for testing</p></a></li>
<li><a href='#UserGS'><p>Simulated user-defined gene set data</p></a></li>
<li><a href='#UserGST'><p>User-defined gene set data for testing</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Prioritization of Candidate Cancer Subtype Specific Drugs</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.8</td>
</tr>
<tr>
<td>Author:</td>
<td>Xudong Han,
  Junwei Han,
  Chonghui Liu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A systematic biology tool was developed to prioritize cancer subtype-specific drugs by integrating genetic perturbation, drug action, biological pathway, and cancer subtype. 
    The capabilities of this tool include inferring patient-specific subpathway activity profiles in the context of gene expression profiles with subtype labels, calculating differentially 
    expressed subpathways based on cultured human cells treated with drugs in the 'cMap' (connectivity map) database, prioritizing cancer subtype specific drugs according to drug-disease 
    reverse association score based on subpathway, and visualization of results (Castelo (2013) &lt;<a href="https://doi.org/10.1186%2F1471-2105-14-7">doi:10.1186/1471-2105-14-7</a>&gt;; Han et al (2019) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtz894">doi:10.1093/bioinformatics/btz894</a>&gt;; Lamb and Justin (2006) &lt;<a href="https://doi.org/10.1126%2Fscience.1132939">doi:10.1126/science.1132939</a>&gt;). Please cite using &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtab011">doi:10.1093/bioinformatics/btab011</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/hanjunwei-lab/SubtypeDrug/issues">https://github.com/hanjunwei-lab/SubtypeDrug/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>BiocGenerics,GSVA,grDevices,graphics,igraph,parallel,pheatmap,rvest,stats,xml2,ChemmineR</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-18 03:07:51 UTC; 12859</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-18 06:00:02 UTC</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
</table>
<hr>
<h2 id='AccumulateNormal'>SubtypeDrug internal function</h2><span id='topic+AccumulateNormal'></span>

<h3>Description</h3>

<p>Infering patient-specific subpathway activity profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AccumulateNormal(x_matrix, control_index)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AccumulateNormal_+3A_x_matrix">x_matrix</code></td>
<td>
<p>A subpathway activity profile. rows are subpathwyas, columns are samples.</p>
</td></tr>
<tr><td><code id="AccumulateNormal_+3A_control_index">control_index</code></td>
<td>
<p>A vector. In the sample of the subpathway activity profile, the position of control samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>AccumulateNormal
</p>


<h3>Value</h3>

<p>A matrix.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x&lt;-matrix(c(1:10),ncol = 5)
x1&lt;-AccumulateNormal(x,c(3,5))
</code></pre>

<hr>
<h2 id='CalculateSES'>SubtypeDrug internal function</h2><span id='topic+CalculateSES'></span>

<h3>Description</h3>

<p>Calculate subpathway enrichment score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalculateSES(labels.list, correl.vector = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CalculateSES_+3A_labels.list">labels.list</code></td>
<td>
<p>A vector of 0 and 1.</p>
</td></tr>
<tr><td><code id="CalculateSES_+3A_correl.vector">correl.vector</code></td>
<td>
<p>A vector. The weight value used to calculate the enrichment score.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>CalculateSES
</p>


<h3>Value</h3>

<p>A vector.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x&lt;-CalculateSES(sample(c(0,1),10,replace = TRUE),c(1:10))
</code></pre>

<hr>
<h2 id='Colork'>Color</h2><span id='topic+Colork'></span>

<h3>Description</h3>

<p>This variable stores the color data required by the program.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Colork
</code></pre>


<h3>Format</h3>

<p>A vector containing 73 values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Colork)

</code></pre>

<hr>
<h2 id='Disease_drugs'>Simulated result data</h2><span id='topic+Disease_drugs'></span>

<h3>Description</h3>

<p>The simulated result data of only two sample types is generated by the functional PrioSubtypeDrug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Disease_drugs
</code></pre>


<h3>Format</h3>

<p>A list containing 8 variables.
The variables are as follows:
</p>

<ul>
<li><p> Cacner Results table for cacner
</p>
</li>
<li><p> SubpathwayMatrix Subpathway activity natrix
</p>
</li>
<li><p> SampleInformation Cancer sample phenotypic information
</p>
</li>
<li><p> Parameter Parameter of the function PrioSubtypeDrug
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># data(Disease_drugs)
</code></pre>

<hr>
<h2 id='Drugs_CID'>Correspondence table of drug label and drug ID in PubCham database</h2><span id='topic+Drugs_CID'></span>

<h3>Description</h3>

<p>A data frame for the drug and its corresponding PubCham database ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Drugs_CID
</code></pre>


<h3>Format</h3>

<p>A dataframe with drug label and CID.
The variables are as follows:
</p>

<ul>
<li><p> Drugs Drug label
</p>
</li>
<li><p> CID Drug ID in PubCham database
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(Drugs_CID)
</code></pre>

<hr>
<h2 id='Geneexp'>Simulated gene expression data</h2><span id='topic+Geneexp'></span>

<h3>Description</h3>

<p>Simulated normalized gene expression profile data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Geneexp
</code></pre>


<h3>Format</h3>

<p>A matrix with 3000 genes and 40 samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Geneexp)
</code></pre>

<hr>
<h2 id='GeneexpT'>Gene expression data for testing</h2><span id='topic+GeneexpT'></span>

<h3>Description</h3>

<p>Simulated normalized gene expression profile data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GeneexpT
</code></pre>


<h3>Format</h3>

<p>A matrix with 40 samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GeneexpT)
</code></pre>

<hr>
<h2 id='getDrugMatrix'>SubtypeDrug internal function</h2><span id='topic+getDrugMatrix'></span>

<h3>Description</h3>

<p>Obtaining drug-disease reverse association score matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDrugMatrix(spw_matrix, drug_target_data, weighted.score)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDrugMatrix_+3A_spw_matrix">spw_matrix</code></td>
<td>
<p>A subpathway activity profile. rows are subpathwyas, columns are samples.</p>
</td></tr>
<tr><td><code id="getDrugMatrix_+3A_drug_target_data">drug_target_data</code></td>
<td>
<p>A list. A list stores a collection of drug up- and down-regulated subpathways.</p>
</td></tr>
<tr><td><code id="getDrugMatrix_+3A_weighted.score">weighted.score</code></td>
<td>
<p>A binary value of 0 or 1. If the 'weighted.score' = 1, the drug reverse association score will be weighted by the subpathway activity.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>getDrugMatrix
</p>


<h3>Value</h3>

<p>A matrix.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(GSVA)
Geneexp&lt;-get("GeneexpT")
UserDS&lt;-get("UserDST")
UserGS&lt;-get("UserGST")
gsvapar &lt;- gsvaParam(Geneexp, UserGS)
spw_matrix&lt;-gsva(gsvapar)
x&lt;-getDrugMatrix(spw_matrix,UserDS,weighted.score=FALSE)
</code></pre>

<hr>
<h2 id='getDrugSpw'>SubtypeDrug internal function</h2><span id='topic+getDrugSpw'></span>

<h3>Description</h3>

<p>According to the parameters set by the user, the up-regulatory
and down-regulatory subpathway data of drug is obtained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDrugSpw(
  drug_target_data,
  spw_matrix_rnames,
  drug.P.value.threshold,
  drug.min.sz,
  drug.max.sz
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDrugSpw_+3A_drug_target_data">drug_target_data</code></td>
<td>
<p>A list. A list stores a collection of drug up- and down-regulated subpathways.</p>
</td></tr>
<tr><td><code id="getDrugSpw_+3A_spw_matrix_rnames">spw_matrix_rnames</code></td>
<td>
<p>A vector. A vector consisting of row names of subpathway activity profile.</p>
</td></tr>
<tr><td><code id="getDrugSpw_+3A_drug.p.value.threshold">drug.P.value.threshold</code></td>
<td>
<p>A value. According to the threshold of the significant P value set by parameter 'drug.p.val.threshold', the drug up-regulation and down-regulatory subpathways were screened.</p>
</td></tr>
<tr><td><code id="getDrugSpw_+3A_drug.min.sz">drug.min.sz</code></td>
<td>
<p>A numeric. The drug regulated subpathways intersects with the subpathways in the subpathway activity profile. Then drugs with less than 'drug.spw.min.sz' up- or down-regulated subpathways are removed.</p>
</td></tr>
<tr><td><code id="getDrugSpw_+3A_drug.max.sz">drug.max.sz</code></td>
<td>
<p>A numeric. Similar to parameter 'drug.spw.min.sz', drugs with more than 'drug.spw.max.sz' up- or down-regulated subpathways are removed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>getDrugSpw
</p>


<h3>Value</h3>

<p>a list.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(GSVA)
Geneexp&lt;-get("Geneexp")
UserGS&lt;-get("UserGS")
UserDS&lt;-get("UserDS")
gsvaPar &lt;- GSVA::ssgseaParam(Geneexp,UserGS,minSize=2)
spw_matrix&lt;-gsva(gsvaPar)
x&lt;-getDrugSpw(UserDS,row.names(spw_matrix),0.05,1,100)
</code></pre>

<hr>
<h2 id='getDrugStructure'>Get drug chemical structure diagram data</h2><span id='topic+getDrugStructure'></span>

<h3>Description</h3>

<p>'getDrugStructure()' outputs the chemical structure graph data of the
drug or compound based on the input drug label by the user. The results can be visualized by the 'plot' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDrugStructure(drug.label = "", main = "", sub = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDrugStructure_+3A_drug.label">drug.label</code></td>
<td>
<p>A character string of drug label to determine which drug to use for visualization.</p>
</td></tr>
<tr><td><code id="getDrugStructure_+3A_main">main</code></td>
<td>
<p>An overall title for the chemical structure graph.</p>
</td></tr>
<tr><td><code id="getDrugStructure_+3A_sub">sub</code></td>
<td>
<p>A sub title for the chemical structure graph.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>getDrugStructure
</p>


<h3>Value</h3>

<p>A sdfset object.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(rvest)
require(ChemmineR)
# Plot the chemical structure of drug pirenperone.
# Chem_str&lt;-getDrugStructure(drug.label="pirenperone.")
# plot(Chem_str)
</code></pre>

<hr>
<h2 id='isPackageLoaded'>SubtypeDrug internal function</h2><span id='topic+isPackageLoaded'></span>

<h3>Description</h3>

<p>Determine if the package is loaded. If the package is not loaded,
the program will prompt the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isPackageLoaded(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isPackageLoaded_+3A_name">name</code></td>
<td>
<p>A string. The name of the R package which determines whether it is loaded.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>isPackageLoaded
</p>


<h3>Value</h3>

<p>A string, TRUE or FALSE.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>isPackageLoaded("pheatmap")
</code></pre>

<hr>
<h2 id='plotDScoreHeatmap'>Plot a heat map of the normalized drug-disease reverse association scores for cancer samples</h2><span id='topic+plotDScoreHeatmap'></span>

<h3>Description</h3>

<p>According to the parameter setting, the function 'plotDScoreHeatmap()' displays the heat map of the normalized
drug-disease reverse association score for the significant drugs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDScoreHeatmap(
  data,
  subtype.label = "all",
  SDS = "all",
  E_Pvalue.th = 1,
  E_FDR.th = 0.05,
  S_Pvalue.th = 1,
  S_FDR.th = 0.001,
  show.rownames = TRUE,
  show.colnames = FALSE,
  color = colorRampPalette(c("#0A8D0A", "#F8F0EB", "red"))(190),
  subtype_colors = NA,
  drug_colors = NA,
  border_color = "grey60",
  cellwidth = NA,
  cellheight = NA,
  fontsize = 10,
  fontsize.row = 10,
  fontsize.col = 10,
  scale = "row"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDScoreHeatmap_+3A_data">data</code></td>
<td>
<p>A list of result data generated by function 'PrioSubtypeDrug()'.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_subtype.label">subtype.label</code></td>
<td>
<p>Character string indicates which sample of the cancer subtype was used to plot the heat map.
If subtype.label = &quot;all&quot; (default), all cancer samples will be shown in the heat map.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_sds">SDS</code></td>
<td>
<p>A string indicates that the range of SDS is used for the heat map. if SDS=&quot;all&quot; (default), the SDS will not be filtered.
SDS=&quot;negative&quot;, only drugs with SDS&lt;0 are used. SDS=&quot;positive&quot;, only drugs with SDS&gt;0 are used.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_e_pvalue.th">E_Pvalue.th</code></td>
<td>
<p>A numeric.A threshold is used to filter the drug effected P value (default: 1).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_e_fdr.th">E_FDR.th</code></td>
<td>
<p>A numeric.A threshold is used to filter the drug effected FDR (default: 0.05).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_s_pvalue.th">S_Pvalue.th</code></td>
<td>
<p>A numeric.A threshold is used to filter the Subtype specific P value (default: 1).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_s_fdr.th">S_FDR.th</code></td>
<td>
<p>A numeric.A threshold is used to filter the Subtype specific P value (default: 0.001).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_show.rownames">show.rownames</code></td>
<td>
<p>Boolean specifying if row names are be shown (default: TRUE).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_show.colnames">show.colnames</code></td>
<td>
<p>Boolean specifying if column names are be shown (default: FALSE).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_color">color</code></td>
<td>
<p>Vector of colors used in heatmap.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_subtype_colors">subtype_colors</code></td>
<td>
<p>Vector of colors is used to annotate the sample subtype. Its length should correspond to the number of sample subtypes.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_drug_colors">drug_colors</code></td>
<td>
<p>Vector of colors is used to label subtype-specific drugs.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_border_color">border_color</code></td>
<td>
<p>Color of cell borders on heatmap, use NA if no border should be drawn.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_cellwidth">cellwidth</code></td>
<td>
<p>Individual cell width in points. If left as NA, then the values depend on the size of plotting window.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_cellheight">cellheight</code></td>
<td>
<p>Individual cell height in points. If left as NA, then the values depend on the size of plotting window.</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_fontsize">fontsize</code></td>
<td>
<p>Base fontsize for the plot (default: 10).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_fontsize.row">fontsize.row</code></td>
<td>
<p>Fontsize for rownames (default: 10).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_fontsize.col">fontsize.col</code></td>
<td>
<p>Fontsize for colnames (default: 10).</p>
</td></tr>
<tr><td><code id="plotDScoreHeatmap_+3A_scale">scale</code></td>
<td>
<p>Character indicating if the values should be centered and scaled in either the row direction or the column direction, or none. Corresponding values are &quot;row&quot; (default), &quot;column&quot; and &quot;none&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotDScoreHeatmap
</p>


<h3>Value</h3>

<p>A heat map.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(pheatmap)
## Get the result data of PrioSubtypeDrug().
## The data is based on the simulated breast cancer subtype data.
Subtype_drugs&lt;-get("Subtype_drugs")
## Heat map of all subtype-specific drugs.
#plotDScoreHeatmap(data=Subtype_drugs,E_Pvalue.th=0.05,
#                           S_Pvalue.th=0.05)
## Plot only Basal subtype-specific drugs.
plotDScoreHeatmap(Subtype_drugs,subtype.label="Basal",SDS="all",E_Pvalue.th=0.05,
                  E_FDR.t=1,S_Pvalue.th=0.05,S_FDR.th=1)
</code></pre>

<hr>
<h2 id='plotDSpwHeatmap'>Plot heat map of the drug regulated subpathway activity score</h2><span id='topic+plotDSpwHeatmap'></span>

<h3>Description</h3>

<p>The 'plotDSpwHeatmap()' function plots a heat map of the subpathways that are regulated by specified drug and have
differential expression between specified cancer subtype and normal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDSpwHeatmap(
  data,
  drug.label = "",
  subtype.label = "",
  show.rownames = TRUE,
  show.colnames = TRUE,
  color = NA,
  phen_colors = NA,
  border_color = "grey60",
  cellwidth = NA,
  cellheight = NA,
  fontsize = 10,
  fontsize.row = 10,
  fontsize.col = 10,
  scale = "row"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDSpwHeatmap_+3A_data">data</code></td>
<td>
<p>A list of result data generated by function 'PrioSubtypeDrug()'.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_drug.label">drug.label</code></td>
<td>
<p>A character string of drug labels to determine which drug to use for visualization.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_subtype.label">subtype.label</code></td>
<td>
<p>Character string indicates which sample of the cancer subtype was used to plot the heat map.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_show.rownames">show.rownames</code></td>
<td>
<p>Boolean specifying if row names are be shown.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_show.colnames">show.colnames</code></td>
<td>
<p>Boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_color">color</code></td>
<td>
<p>Vector of colors used in heatmap.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_phen_colors">phen_colors</code></td>
<td>
<p>Vector of colors is used to annotate the sample subtype and control sample.It should be assigned two colors.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_border_color">border_color</code></td>
<td>
<p>Color of cell borders on heatmap, use NA if no border should be drawn.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_cellwidth">cellwidth</code></td>
<td>
<p>Individual cell width in points. If left as NA, then the values depend on the size of plotting window.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_cellheight">cellheight</code></td>
<td>
<p>Individual cell height in points. If left as NA, then the values depend on the size of plotting window.</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_fontsize">fontsize</code></td>
<td>
<p>Base fontsize for the plot (default: 10).</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_fontsize.row">fontsize.row</code></td>
<td>
<p>Fontsize for rownames (default: 10).</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_fontsize.col">fontsize.col</code></td>
<td>
<p>Fontsize for colnames (default: 10).</p>
</td></tr>
<tr><td><code id="plotDSpwHeatmap_+3A_scale">scale</code></td>
<td>
<p>Character indicating if the values should be centered and scaled in either the row direction or the column direction, or none. Corresponding values are &quot;row&quot;, &quot;column&quot; and &quot;none&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotDSpwHeatmap
</p>
<p>Based on the input cancer subtype, the program draws a heat map of the drug regulated subpathway activity
score. If the cancer subtype of input has sutype-specific drug score (SDS)&lt;0,
we can observe the drug upregulatory subpathway is lowly expressed in the cancer subtype samples and high in the normal samples;
the drug downregulatory subpathway is highly expressed in the cancer subtype samples and low in the normal samples. This indicates that after the drug action,
these subpathways activity is converted from the level of the cancer subtype into the level of normal. If the cancer subtype of input has sutype-specific drug
score (SDS)&gt;0, it is indicated that the drug action may promote the subpathway expression status of the cancer subtype.
</p>


<h3>Value</h3>

<p>A heat map.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(pheatmap)
## Get the result data of PrioSubtypeDrug().
## The data is based on the simulated breast cancer subtype data.
Subtype_drugs&lt;-get("Subtype_drugs")
plotDSpwHeatmap(data=Subtype_drugs,drug.label="pirenperone(1.02e-05M)",subtype.label="Basal")
##Visualize the results of only two types of samples.
Disease_drugs&lt;-get("Disease_drugs")
plotDSpwHeatmap(data=Disease_drugs,drug.label="W-13(1e-05M)",subtype.label="Cancer")
</code></pre>

<hr>
<h2 id='plotGlobalGraph'>Plot a global graph of the drug</h2><span id='topic+plotGlobalGraph'></span>

<h3>Description</h3>

<p>The 'plotGlobalGraph()' identifies the drug label entered by the user, and plots an
integrated diagram including box plot of the normalized drug-disease reverse association scores, null distribution curves
of significant P-value, and heat map of cancer subtype sample distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGlobalGraph(
  data,
  drug.label = "",
  overall.main = "",
  overall.cex.main = 1.5,
  cex.submap.axis = 1,
  cex.submap.lab = 1,
  cex.submap.main = 1,
  cex.submap.sub = 1,
  cex.legend = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotGlobalGraph_+3A_data">data</code></td>
<td>
<p>A list of result data generated by function 'PrioSubtypeDrug()'.</p>
</td></tr>
<tr><td><code id="plotGlobalGraph_+3A_drug.label">drug.label</code></td>
<td>
<p>A character string of drug labels to determine which drug to use for visualization.</p>
</td></tr>
<tr><td><code id="plotGlobalGraph_+3A_overall.main">overall.main</code></td>
<td>
<p>An overall title for the whole graph. If the user
does not make any input, the title will display a drug label.</p>
</td></tr>
<tr><td><code id="plotGlobalGraph_+3A_overall.cex.main">overall.cex.main</code></td>
<td>
<p>The magnification to be used for overall.main (default: 1.5).</p>
</td></tr>
<tr><td><code id="plotGlobalGraph_+3A_cex.submap.axis">cex.submap.axis</code></td>
<td>
<p>The magnification to be used for axis of each submap annotation relative to
the current setting of cex.</p>
</td></tr>
<tr><td><code id="plotGlobalGraph_+3A_cex.submap.lab">cex.submap.lab</code></td>
<td>
<p>The magnification to be used for x and y labels of each submap relative to
the current setting of cex.</p>
</td></tr>
<tr><td><code id="plotGlobalGraph_+3A_cex.submap.main">cex.submap.main</code></td>
<td>
<p>The magnification to be used for main titles of each submap relative to the
current setting of cex.</p>
</td></tr>
<tr><td><code id="plotGlobalGraph_+3A_cex.submap.sub">cex.submap.sub</code></td>
<td>
<p>The magnification to be used for sub titles of each submap relative to the
current setting of cex.</p>
</td></tr>
<tr><td><code id="plotGlobalGraph_+3A_cex.legend">cex.legend</code></td>
<td>
<p>fontsize of labels for legend.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotGlobalGraph
</p>


<h3>Value</h3>

<p>A plot.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Get the result data of PrioSubtypeDrug().
## The data is based on the simulated breast cancer subtype data.
Subtype_drugs&lt;-get("Subtype_drugs")
## Plot a global graph of the drug pirenperone(1.02e-05M).
plotGlobalGraph(data=Subtype_drugs,drug.label="pirenperone(1.02e-05M)")
</code></pre>

<hr>
<h2 id='plotSpwNetGraph'>Polt a subpathway network graph</h2><span id='topic+plotSpwNetGraph'></span>

<h3>Description</h3>

<p>Visualize a subpathway network graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSpwNetGraph(
  spwid,
  layout = NULL,
  margin = 0,
  vertex.label.cex = 0.6,
  vertex.label.font = 1,
  vertex.size = 8,
  vertex.size2 = 6,
  edge.arrow.size = 0.2,
  edge.arrow.width = 3,
  edge.label.cex = 0.6,
  vertex.label.color = "black",
  vertex.color = "#BFFFBF",
  vertex.frame.color = "dimgray",
  edge.color = "dimgray",
  edge.label.color = "dimgray",
  sub = NULL,
  main = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSpwNetGraph_+3A_spwid">spwid</code></td>
<td>
<p>The subpathway id which the user wants to plot.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_layout">layout</code></td>
<td>
<p>A matrix of x-y coordinates with two dims. Determine the placement
of the nodes for drawing a graph.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_margin">margin</code></td>
<td>
<p>A numeric. The value is usually between -0.5 and 0.5, which is able to zoom in or out a subpathway graph. The
default is 0.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_vertex.label.cex">vertex.label.cex</code></td>
<td>
<p>A numeric vector of node label size.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_vertex.label.font">vertex.label.font</code></td>
<td>
<p>A numeric vector of label font.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_vertex.size">vertex.size</code></td>
<td>
<p>A numeric vector of Node size. See <code><a href="igraph.html#topic+plot.igraph">plot.igraph</a></code>.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_vertex.size2">vertex.size2</code></td>
<td>
<p>A numeric vector of Node size.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_edge.arrow.size">edge.arrow.size</code></td>
<td>
<p>Edge arrow size.The default is 0.2.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_edge.arrow.width">edge.arrow.width</code></td>
<td>
<p>Edge arrow width. The default is 3.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_edge.label.cex">edge.label.cex</code></td>
<td>
<p>Edge label size.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_vertex.label.color">vertex.label.color</code></td>
<td>
<p>A vector of node label colors. The default is black.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_vertex.color">vertex.color</code></td>
<td>
<p>A vector of node colors. The default is the KEGG node color.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_vertex.frame.color">vertex.frame.color</code></td>
<td>
<p>A vector of node frame color. The default is dimgray.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_edge.color">edge.color</code></td>
<td>
<p>A vector of edge color. The default is dimgray.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_edge.label.color">edge.label.color</code></td>
<td>
<p>A vector of edge label color. The default is dimgray.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_sub">sub</code></td>
<td>
<p>A character string of subtitle.</p>
</td></tr>
<tr><td><code id="plotSpwNetGraph_+3A_main">main</code></td>
<td>
<p>A character string of main title.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>plotSpwNetGraph
</p>
<p>The function plotSpwNetGraph is able to display a subpathway graph.
The argument layout is used to determine the placement of the nodes
for drawing a graph.The layouts provided in igraph include 'layout_as_star', 'layout_as_tree',
'layout_in_circle', 'layout_nicely','layout_on_grid', 'layout_on_sphere', 'layout_randomly', 'layout_with_dh', 'layout_with_fr',
'layout_with_gem', 'layout_with_graphopt', 'layout_with_kk', 'layout_with_lgl', 'layout_with_mds'.
The 'layout_as_tree' generates a tree-like layout, so it is mainly for
trees. The 'layout_randomly' places the nodes randomly. The 'layout_in_circle' places
the nodes on a unit circle. Detailed information on the parameters can be found in <code><a href="igraph.html#topic+layout_">layout_</a></code>
</p>


<h3>Value</h3>

<p>a plot
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(igraph)
# plot network graph of the subpathway 00020_4.
plotSpwNetGraph(spwid="00020_4")
</code></pre>

<hr>
<h2 id='PrioSubtypeDrug'>Prioritization of candidate cancer subtype-specific drugs (PrioSubtypeDrug)</h2><span id='topic+PrioSubtypeDrug'></span>

<h3>Description</h3>

<p>Integrating drug, gene, and subpathway data to identify drugs specific to cancer subtypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrioSubtypeDrug(
  expr,
  input.cls = "",
  control.label = "",
  subpathway.list,
  spw.min.sz = 10,
  spw.max.sz = Inf,
  spw.score.method = "gsva",
  kcdf = "Gaussian",
  drug.spw.data,
  drug.spw.p.val.th = 0.05,
  drug.spw.min.sz = 10,
  drug.spw.max.sz = Inf,
  weighted.drug.score = TRUE,
  nperm = 1000,
  parallel.sz = 1,
  E_FDR = 0.05,
  S_FDR = 0.001
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PrioSubtypeDrug_+3A_expr">expr</code></td>
<td>
<p>Matrix of gene expression values (rows are genes, columns are samples).</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_input.cls">input.cls</code></td>
<td>
<p>Input sample subtype class vector file in CLS format.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_control.label">control.label</code></td>
<td>
<p>In the CLS file of 'input.cls', the label of the control sample.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_subpathway.list">subpathway.list</code></td>
<td>
<p>A list.  The subpathway list data is mined from KEGG data is stored in the package 'SubtypeDrugData' and can be downloaded through the connection <a href="https://github.com/hanjunwei-lab/SubtypeDrugData">https://github.com/hanjunwei-lab/SubtypeDrugData</a>.
The gene tags included in the subpathway list data should be consistent with those in the gene expression profile. The package 'SubtypeDrugData' provides two choices that include the Entrezid and Symbol tags of the gene.
Users can also enter their own pathway or gene set list data.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_spw.min.sz">spw.min.sz</code></td>
<td>
<p>Removes subpathways that contain fewer genes than 'spw.min.sz' (default: 10).</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_spw.max.sz">spw.max.sz</code></td>
<td>
<p>Removes subpathways that contain more genes than 'spw.max.sz' (default: Inf).</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_spw.score.method">spw.score.method</code></td>
<td>
<p>Method to employ in the estimation of subpathway
enrichment scores per sample. By default this is set to 'gsva' (Hänzelmann
et al, 2013) and other options are 'ssgsea' (Barbie et al, 2009).</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_kcdf">kcdf</code></td>
<td>
<p>Character string denoting the kernel to use during the
non-parametric estimation of the cumulative distribution function of
expression levels across samples when 'spw.score.method=&quot;gsva&quot;'. By default,
'kcdf=&quot;Gaussian&quot;' which is suitable when input expression values are
continuous, such as microarray fluorescent units in logarithmic scale,
RNA-seq log-CPMs, log-RPKMs or log-TPMs. When input expression values are
integer counts, such as those derived from RNA-seq experiments, then this
argument should be set to 'kcdf=&quot;Poisson&quot;'.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_drug.spw.data">drug.spw.data</code></td>
<td>
<p>A list data of drug regulation. The drug subpathway association data we constructed is stored in package 'SubtypeDrugData' and can be downloaded via connection <a href="https://github.com/hanjunwei-lab/SubtypeDrugData">https://github.com/hanjunwei-lab/SubtypeDrugData</a>.
If the input is user-defined drug regulation data, the data should be a list data with each drug as its element. Each drug also contains 'Target_upregulation' and 'Target_downregulation' subpathway or gene set.
Subpathway or gene set contained in drug regulation data should exist in input data of parameter 'subpathway.list'.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_drug.spw.p.val.th">drug.spw.p.val.th</code></td>
<td>
<p>Parameter used only when 'drug.spw.data=&quot;DrugSpwData&quot;'. According to the threshold of the significant P value
set by parameter 'drug.spw.p.val.th' (default: 0.05), the drug up-regulation and down-regulatory subpathways were screened.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_drug.spw.min.sz">drug.spw.min.sz</code></td>
<td>
<p>A numeric. The drug regulated subpathways intersects with the subpathways in the
subpathway activity profile. Then drugs with less than 'drug.spw.min.sz' (default: 10) up- or down-regulated subpathways are removed.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_drug.spw.max.sz">drug.spw.max.sz</code></td>
<td>
<p>A numeric. Similar to parameter 'drug.spw.min.sz', drugs with more
than 'drug.spw.max.sz' (default: Inf) up- or down-regulated subpathways are removed.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_weighted.drug.score">weighted.drug.score</code></td>
<td>
<p>A boolean values determines the method for
calculating the normalized drug-disease reverse association score of the drug for each sample.
'weighted.drug.score=TRUE' (default): KS random walk statistic with individualized subpathway activity aberrance score as weight was used to
calculate the normalized drug-disease reverse association score.
'weighted.drug.score=FALSE': Similar to 'CMap' (Lamb et al., 2006), no weight is needed, and the normalized drug-disease
reverse association score is calculated by the rank of the individualized subpathway activity aberrance score.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_nperm">nperm</code></td>
<td>
<p>Number of random permutations (default: 1000).</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_parallel.sz">parallel.sz</code></td>
<td>
<p>Number of processors to use when doing the calculations in
parallel (default value: 1). If parallel.sz=0,
then it will use all available core processors unless we set this argument
with a smaller number.</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_e_fdr">E_FDR</code></td>
<td>
<p>Significance threshold for E_FDR for drugs (default: 0.05)</p>
</td></tr>
<tr><td><code id="PrioSubtypeDrug_+3A_s_fdr">S_FDR</code></td>
<td>
<p>Significance threshold for S_FDR for drugs (default: 0.001)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PrioSubtypeDrug
</p>
<p>First, the function PrioSubtypeDrug uses the 'GSVA' or 'ssgsea' method to
convert the disease gene expression profile into subpathway activity profile. Parameters 'subpathway.list', 'spw.min.sz' and 'spw.max.sz' are used
to process the subpathway list data. 'spw.score.method' and 'kcdf' are used to control the method of constructing the subpathway activity score
profile.
Individualized subpathway activity aberrance score was estimated using the mean and standard deviation of the Control samples.
Subpathways of each cancer sample are ordered in a ranked list according to individualized subpathway activity aberrance score.
Next, we calculate the normalized drug-disease reverse association score by enriching drug regulated subpathway tags to the subpathway ranked list.
Finlly, all drug-regulated subpathways are enriched into each cancer sample to obtain a normalized drug-disease reverse association score matrix.
The 'drug.spw.p.val.th', 'drug.spw.min.sz' and 'drug.spw.max.sz' is used to screen the drug regulated subpathway set.
If user-defined drug targeting data is used, drug regulated 'Target_upregulation' and 'Target_downregulation' should already be defined in the data.
The 'weighted.drug.score' to control the method of calculating the normalized drug-disease reverse association score.
Finally, empirical sample-based permutation test procedure to obtain significative cancer subtype specific drugs.
For samples containing only cancer and Control, the subpathways are ranked according to the difference in activity between cancer and Control samples.
Subsequently, the subpathway set of drug up- and down-regulated is enriched to the ranking list of subpathway to evaluate the normalized drug-disease reverse association score and
subpathway-based permutation test procedure to calculate significance.
The subpathway list data and drug subpathway associated data set is stored in package 'SubtypeDrugData' and
can be obtained on <a href="https://github.com/hanjunwei-lab/SubtypeDrugData">https://github.com/hanjunwei-lab/SubtypeDrugData</a>.
</p>


<h3>Value</h3>

<p>A list contains the result table of drug scoring and significance, a subpathway activity score matrix, a normalized drug-disease reverse
association score matrix, sample information, and user set parameter information.
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(GSVA)
require(parallel)
## Get simulated breast cancer gene expression profile data.
Geneexp&lt;-get("Geneexp")
## Obtain sample subtype data and calculate breast cancer subtype-specific drugs.
Subtype&lt;-system.file("extdata", "Subtype_labels.cls", package = "SubtypeDrug")

## Subpathway list data and drug subpathway association data
## were stored in packet `SubtypeDrugData`.
## `SubtypeDrugData` has been uploaded to the github repository.
## If subpathway list data and drug subpathway association data are needed,
## users can download and install through `install_github` function and
## set parameter url=""hanjunwei-lab/SubtypeDrugData".
## After installing and loading package `SubtypeDrugData`,
## users can use the following command to get the data.
## Get subpathway list data.
## If the gene expression profile contains gene Symbol.
## data(SpwSymbolList)
## If the gene expression profile contains gene Entrezid.
## data(SpwEntrezidList)
## Get drug subpathway association data.
## data(DrugSpwData)

## Identify breast subtype-specific drugs.
## Subtype_drugs&lt;-PrioSubtypeDrug(Geneexp,Subtype,"Control",SpwSymbolList,drug.spw.data=DrugSpwData,
##                                       E_FDR=1,S_FDR=1)

## Identify breast cancer-related drugs in only two types of samples: breast cancer and control.
Cancer&lt;-system.file("extdata", "Cancer_normal_labels.cls", package = "SubtypeDrug")
## Disease_drugs&lt;-PrioSubtypeDrug(Geneexp,Cancer,"Control",SpwSymbolList,drug.spw.data=DrugSpwData,
##                                       E_FDR=1,S_FDR=1)

## The function PrioSubtypeDrug() can also support user-defined data.
Geneexp&lt;-get("GeneexpT")
## User-defined drug regulation data should resemble the structure below
UserDS&lt;-get("UserDST")
str(UserDS)
## Need to load gene set data consistent with drug regulation data.
UserGS&lt;-get("UserGST")
str(UserGS)
Drugs&lt;-PrioSubtypeDrug(Geneexp,Cancer,"Control",UserGS,spw.min.sz=1,
                       drug.spw.data=UserDS,drug.spw.min.sz=1,
                       nperm=10,E_FDR=1,S_FDR=1)
</code></pre>

<hr>
<h2 id='ReadClsFile'>SubtypeDrug internal function</h2><span id='topic+ReadClsFile'></span>

<h3>Description</h3>

<p>These are function read sample label file (.cls format).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadClsFile(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadClsFile_+3A_file">file</code></td>
<td>
<p>Input sample subtype class vector file in CLS format.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ReadClsFile
</p>


<h3>Value</h3>

<p>a list
</p>


<h3>Author(s)</h3>

<p>Xudong Han,
Junwei Han,
Chonghui Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Subtype&lt;-system.file("extdata", "Subtype_labels.cls", package = "SubtypeDrug")
x&lt;-ReadClsFile(Subtype)
</code></pre>

<hr>
<h2 id='SpwNetworkData'>Subpathway network structure data</h2><span id='topic+SpwNetworkData'></span>

<h3>Description</h3>

<p>A list to store the network data of the genes contained in the subpathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SpwNetworkData
</code></pre>


<h3>Format</h3>

<p>A list containing 1598 subpathway network.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SpwNetworkData)
</code></pre>

<hr>
<h2 id='Subtype_drugs'>Simulation result data</h2><span id='topic+Subtype_drugs'></span>

<h3>Description</h3>

<p>The result data of the simulation is generated by the functional OCSSD.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Subtype_drugs
</code></pre>


<h3>Format</h3>

<p>A list containing 8 variables.
The variables are as follows:
</p>

<ul>
<li><p> Basal Results table for basal subtype
</p>
</li>
<li><p> Her2 Results table for Her2 subtype
</p>
</li>
<li><p> LumA Results table for LumA subtype
</p>
</li>
<li><p> LumB Results table for LumB subtype
</p>
</li>
<li><p> DrugMatrix Drug disease reverse association matrix
</p>
</li>
<li><p> SubpathwayMatrix Subpathway activity natrix
</p>
</li>
<li><p> SampleInformation Cancer sample phenotypic information
</p>
</li>
<li><p> Parameter Parameter of the function OCSSD
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># data(Subtype_drugs)
</code></pre>

<hr>
<h2 id='UserDS'>Simulated user-defined drug regulator subpathway dataset</h2><span id='topic+UserDS'></span>

<h3>Description</h3>

<p>The drug regulator subpathway data set is modeled as a case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UserDS
</code></pre>


<h3>Format</h3>

<p>A list containing 5 drugs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(UserDS)
</code></pre>

<hr>
<h2 id='UserDST'>User-defined drug regulator subpathway dataset for testing</h2><span id='topic+UserDST'></span>

<h3>Description</h3>

<p>The drug regulator subpathway data set is modeled as a case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UserDST
</code></pre>


<h3>Format</h3>

<p>A list.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(UserDST)
</code></pre>

<hr>
<h2 id='UserGS'>Simulated user-defined gene set data</h2><span id='topic+UserGS'></span>

<h3>Description</h3>

<p>Gene set data is simulated for case studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UserGS
</code></pre>


<h3>Format</h3>

<p>A list containing 50 gene sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(UserDS)
</code></pre>

<hr>
<h2 id='UserGST'>User-defined gene set data for testing</h2><span id='topic+UserGST'></span>

<h3>Description</h3>

<p>Gene set data is simulated for case studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UserGST
</code></pre>


<h3>Format</h3>

<p>A list.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(UserGST)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
