<!DOCTYPE html><html lang="en"><head><title>Help for package sumFREGAT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {sumFREGAT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#sumFREGAT-package'><p>sumFREGAT: Fast REGional Association Tests on summary statistics</p></a></li>
<li><a href='#ACATO'><p>omnibus aggregated Cauchy association test</p></a></li>
<li><a href='#gene-based+20test+20functions'><p>Gene-based tests on summary statistics</p></a></li>
<li><a href='#prep.score.files'><p>Prepare score files</p></a></li>
<li><a href='#sumSTAAR'><p>variant-Set Test for Association using Annotation infoRmation on summary statistics</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Fast Region-Based Association Tests on Summary Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.5</td>
</tr>
<tr>
<td>Author:</td>
<td>Nadezhda M. Belonogova &lt;belon@bionet.nsc.ru&gt; and 
    Gulnara R. Svishcheva &lt;gulsvi@bionet.nsc.ru&gt;
    with contributions from:
    Seunggeun Lee (kernel functions),
    Pierre Lafaye de Micheaux ('davies' method),
    Thomas Lumley ('kuonen' method),
    Minghui Wang, Yiyuan Liu, Shizhong Han (simpleM function), 
    Yaowu Liu (ACAT function),
    James O. Ramsay (functional data analysis functions),
    Xihao Li, Zilin Li, and Han Chen (conception of the STAAR procedure)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nadezhda M. Belonogova &lt;belon@bionet.nsc.ru&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, Matrix, splines, seqminer, GBJ</td>
</tr>
<tr>
<td>Suggests:</td>
<td>data.table</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Description:</td>
<td>An adaptation of classical region/gene-based association analysis
    techniques to the use of summary statistics (P values and effect sizes)
    and correlations between genetic variants as input. It is a tool to perform
    the most popular and efficient gene-based tests using the results of genome-wide
    association (meta-)analyses without having the original genotypes and
    phenotypes at hand.
    See for details:
    Svishcheva et al (2019) Gene-based association tests using GWAS summary
    statistics. Bioinformatics.
    Belonogova et al (2022) SumSTAAR: A flexible framework for gene-based
    association studies using GWAS summary statistics. PLOS Comp Biol.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-06 12:19:21 UTC; lima</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-07 08:10:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='sumFREGAT-package'>sumFREGAT: Fast REGional Association Tests on summary statistics</h2><span id='topic+sumFREGAT-package'></span>

<h3>Description</h3>

<p>The sumFREGAT package computes the most popular and efficient
tests for the region-based association analysis using summary statistics data
(beta and P values) and correlations between genetic variants. It does not
require genotype or phenotype data. Methods implemented are SKAT and SKAT-O
(sequence kernel association tests), BT (burden test), ACAT (aggregated Cauchy
association test), FLM (functional linear model), PCA (principal components analysis),
and others.
</p>


<h3>Details</h3>

<p>Package: sumFREGAT<br />
Type: Package<br />
License: GPLv3
</p>


<h3>Author(s)</h3>

<p>Nadezhda Belonogova &lt;belon@bionet.nsc.ru&gt;<br />
Gulnara Svishcheva &lt;gulsvi@bionet.nsc.ru&gt;<br /><br />
The authors are grateful to Dr. Anatoly Kirichenko for technical support<br />
and Prof. Tatiana Axenovich for scientific guidance.<br />
</p>

<hr>
<h2 id='ACATO'>omnibus aggregated Cauchy association test</h2><span id='topic+ACATO'></span>

<h3>Description</h3>

<p>This test combines P values via aggregated Cauchy association test (Liu, Y. et al., 2019)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ACATO(p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ACATO_+3A_p">p</code></td>
<td>
<p>a vector of P values to combine</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ACATO is an omnibus aggregated Cauchy association test recently proposed by Liu, Y. et al. (2019).
This function can be used to combine P values obtained by different gene-based tests for the same gene
(see Examples below).
</p>


<h3>Value</h3>

<p>a single P value which is a combination of the P values given in input
</p>


<h3>References</h3>

<p>Liu Y. et al. (2019) ACAT: a fast and powerful p value combination method for rare-variant analysis in sequencing studies. Am. J. Hum. Genet. 104, 410-421.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cor.path &lt;- system.file("testfiles/", package = "sumFREGAT")
score.file &lt;- system.file("testfiles/CFH.full.vcf.gz", package = "sumFREGAT")

p.bt &lt;- BT(score.file, genes = "CFH", cor.path = cor.path)$pvalue
p.skat &lt;- SKAT(score.file, genes = "CFH", cor.path = cor.path)$pvalue
p.pca &lt;- PCA(score.file, genes = "CFH", cor.path = cor.path, n = 85)$pvalue

p.combined &lt;- ACATO(c(p.bt, p.skat, p.pca))

</code></pre>

<hr>
<h2 id='gene-based+20test+20functions'>Gene-based tests on summary statistics</h2><span id='topic+SKAT'></span><span id='topic+SKATO'></span><span id='topic+sumchi'></span><span id='topic+BT'></span><span id='topic+PCA'></span><span id='topic+FLM'></span><span id='topic+simpleM'></span><span id='topic+minp'></span><span id='topic+ACAT'></span>

<h3>Description</h3>

<p>A set of tests for gene-based association analysis on GWAS summary statistics:
sequence kernel association tests ('SKAT', 'SKATO'), sum of chi-squares ('sumchi'),
aggregated Cauchy association test ('ACAT'), burden test ('BT'), principal component
analysis-based test ('PCA'), functional linear model-based test ('FLM'),
Bonferroni correction test ('simpleM'), minimum P value test ('minp').
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SKAT(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", beta.par = c(1, 25),
weights.function = NULL, user.weights = FALSE, gen.var.weights = "se.beta",
method = "kuonen", acc = 1e-8, lim = 1e+6, rho = FALSE, p.threshold = 0.8,
write.file = FALSE, quiet = FALSE)

SKATO(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", beta.par = c(1, 25),
weights.function = NULL, user.weights = FALSE, method = "kuonen",
acc = 1e-8, lim = 1e+6, rho = TRUE, p.threshold = 0.8, write.file = FALSE,
quiet = FALSE)

sumchi(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", method = "kuonen", acc = 1e-8,
lim = 1e+6, write.file = FALSE, quiet = FALSE)

ACAT(score.file, gene.file, genes = "all", anno.type = "",
beta.par = c(1, 1), weights.function = NULL, user.weights = FALSE,
gen.var.weights = "none", mac.threshold = NA, n = NA,
write.file = FALSE, quiet = FALSE)

BT(score.file, gene.file, genes = "all", cor.path = "cor/",
anno.type = "", beta.par = c(1, 25), weights.function = NULL,
user.weights = FALSE, write.file = FALSE, quiet = FALSE)

PCA(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", n, beta.par = c(1, 1),
weights.function = NULL, user.weights = FALSE, reference.matrix = TRUE,
fun = "LH", var.fraction = 0.85, write.file = FALSE, quiet = FALSE)

FLM(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", n, beta.par = c(1, 1),
weights.function = NULL, user.weights = FALSE, basis.function = "fourier",
k = 25, order = 4, flip.genotypes = FALSE, Fan = TRUE,
reference.matrix = TRUE, fun = "LH", write.file = FALSE, quiet = FALSE)

simpleM(score.file, gene.file, genes = "all", cor.path = "cor/",
anno.type = "", var.fraction = .995, write.file = FALSE, quiet = FALSE)

minp(score.file, gene.file, genes = "all", cor.path = "cor/",
anno.type = "", write.file = FALSE, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_score.file">score.file</code></td>
<td>
<p>name of data file generated by <code>prep.score.files()</code>.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_gene.file">gene.file</code></td>
<td>
<p>can be a name of a custom text file listing genes in refFlat format. Values &quot;hg19&quot; and &quot;hg38&quot;
can be set to use default gene files containing protein-coding genes with start and stop positions from corresponding build.
Mind that the same build should be used in <code>score.file</code> and <code>gene.file</code>.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_genes">genes</code></td>
<td>
<p>character vector of gene names to be analyzed. Can be &quot;chr1&quot;, &quot;chr2&quot; etc. to analyze
all genes on a corresponding chromosome. If not set, function will
attempt to analyze all genes listed in <code>gene.file</code>.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_cor.path">cor.path</code></td>
<td>
<p>path to a folder with correlation matrix files (one file per each gene to be analyzed).
Correlation matrices in text format are allowed, though &quot;.RData&quot; is preferable as computationally efficient.
Each file should contain a square matrix with correlation coefficients (r) between genetic variants
of a gene. An example of correlation file format:<br />
&quot;snpname1&quot; &quot;snpname2&quot; &quot;snpname3&quot; ...<br />
&quot;snpname1&quot; 1 0.018 -0.003 ...<br />
&quot;snpname2&quot; 0.018 1 0.081 ...<br />
&quot;snpname3&quot; -0.003 0.081 1 ...<br />
...<br />
If genotypes are available, matrices can be generated as follows:<br />
<code>cor.matrix &lt;- cor(g)</code><br />
<code>save(cor.matrix, file = paste0(geneName, ".RData"))</code><br />
where <code>g</code> is a genotype matrix (nsample x nvariants) for a given gene with genotypes coded as 0, 1, 2
(coding should be exactly the same that was used to generate GWAS statistics).<br />
If genotypes are not available, correlations can be approximated through reference samples.
Reference matrices from 1KG European sample can be downloaded at http://mga.bionet.nsc.ru/sumFREGAT/.<br />
Names of correlation files should be constructed as &quot;geneName.RData&quot; (e.g. &quot;ABCG1.RData&quot;, &quot;ADAMTS1.RData&quot;, etc.)
for &quot;.RData&quot; format or &quot;geneName.txt&quot; for text format.<br />
Example corfiles can be found as:<br />
system.file(&quot;testfiles/CFH.cor&quot;, package = &quot;sumFREGAT&quot;)<br />
system.file(&quot;testfiles/CFH.RData&quot;, package = &quot;sumFREGAT&quot;)<br />
</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_approximation">approximation</code></td>
<td>
<p>a logical value indicating whether approximation for large genes (&gt;= 500 SNPs) should be used.
Applicable for SKAT, SKATO, sumchi, PCA, FLM (default = TRUE for these methods).</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_anno.type">anno.type</code></td>
<td>
<p>given (functional) annotations provided by user (see) <code>prep.score.files()</code>,
a character (or character vector) indicating annotation types to be used.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_n">n</code></td>
<td>
<p>size of the sample on which summary statistics were obtained.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_beta.par">beta.par</code></td>
<td>
<p>two positive numeric shape parameters in the beta distribution to assign weights 
for each genetic variant as a function of minor allele frequency (MAF) in the default weights function (see Details).</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_weights.function">weights.function</code></td>
<td>
<p>a function of MAF to assign weights
for each genetic variant. By default, the weights will be calculated using the beta distribution (see Details).</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_user.weights">user.weights</code></td>
<td>
<p>a logical value indicating whether weights from the input file should be applied. Default = FALSE.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_gen.var.weights">gen.var.weights</code></td>
<td>
<p>a character indicating whether scores should be weighted by the variance of genotypes:
&quot;none&quot; - no weights applied, resulting in a sum chi-square test.
&quot;se.beta&quot; - scores weighted by variance of genotypes estimated from P values and effect sizes (regression coefficients)
if provided by user.
&quot;af&quot; - scores weighted by variance of genotypes calculated as AF * (1 - AF), where AF is allele frequency. 
In <code>SKAT()</code>, &quot;se.beta&quot; weighting results in a test analogous to a standard SKAT, while &quot;af&quot; is a way to approximate
the standard test in case when betas are not available. Unweighted sum chi-square test gives more weights to common variants
compared with SKAT-like tests.
In <code>ACAT()</code>, the simplest default unweighted test combines P values (Z scores) without any additional info.
Scores can be weighted by via gen.var.weights = 'se.beta' or 'af', similarly to SKAT.
</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_mac.threshold">mac.threshold</code></td>
<td>
<p> an integer number. In ACAT, scores with MAC &lt;= 10 will be combined using Burden test. MACs are calculated
from MAFs, <code>n</code> must be set to apply <code>mac.threshold</code>.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_method">method</code></td>
<td>
<p>the method for computing P value in kernel-based tests. Available methods are &quot;kuonen&quot;, &quot;davies&quot; and &quot;hybrid&quot;
(see Details). Default = &quot;kuonen&quot;.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_acc">acc</code></td>
<td>
<p>accuracy parameter for &quot;davies&quot; method.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_lim">lim</code></td>
<td>
<p>limit parameter for &quot;davies&quot; method.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_rho">rho</code></td>
<td>
<p>if TRUE the optimal test (SKAT-O) is performed [Lee et al., 2012]. <code>rho</code> can be a vector of grid
values from 0 to 1. The default grid is c(0, 0.1^2, 0.2^2, 0.3^2, 0.4^2, 0.5^2, 0.5, 1).</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_p.threshold">p.threshold</code></td>
<td>
<p>the largest P value that will be considered as important when performing computational optimization in SKAT-O.
All P values larger than p.threshold will be processed via burden test.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_basis.function">basis.function</code></td>
<td>
<p>a basis function type for beta-smooth. Can be set to &quot;bspline&quot; (B-spline basis) or
&quot;fourier&quot; (Fourier basis, default).</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_k">k</code></td>
<td>
<p>the number of basis functions to be used for beta-smooth (default = 25).</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_order">order</code></td>
<td>
<p>a polynomial order to be used in &quot;bspline&quot;. Default = 4 corresponds to the cubic B-splines.
as no effect if only Fourier bases are used.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_flip.genotypes">flip.genotypes</code></td>
<td>
<p>a logical value indicating whether the genotypes of some genetic variants should be
flipped (relabelled) for their better functional representation [Vsevolozhskaya et al., 2014]. Default = FALSE.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_fan">Fan</code></td>
<td>
<p>if TRUE (default) then linearly dependent genetic variants will be omitted, as it
was done in the original realization of FLM test by Fan et al. (2013).</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_reference.matrix">reference.matrix</code></td>
<td>
<p>logical indicating whether the correlation matrices were generated using reference matrix.
If TRUE, regularization algorithms will be applied in order to ensure invertibility and numerical stability of the matrices.
Use 'reference.matrix = FALSE' ONLY IF you are sure that correlation matrices were generated using the same genotype data
as for GWAS summary statistics in input.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_fun">fun</code></td>
<td>
<p>one of two regularization algorithms, 'LH' (default) or 'derivLH'. Currently both give similar results.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_var.fraction">var.fraction</code></td>
<td>
<p>minimal proportion of genetic variance within region that should be explained by principal
components used (see Details for more info).</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_write.file">write.file</code></td>
<td>
<p>output file name. If specified, output (as it proceeds) will be written 
to the file.</p>
</td></tr>
<tr><td><code id="gene-based+2B20test+2B20functions_+3A_quiet">quiet</code></td>
<td>
<p>quiet = TRUE suppresses excessive output from reading vcf file genewise.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>'SKAT' uses the linear weighted kernel function to set the inter-individual
similarity matrix <code class="reqn">K = GWWG^T</code> for SKAT and <code class="reqn">K = GW(I\rho + (1 - \rho)ee^T)WG^T</code> for SKAT-O,
where <code class="reqn">\mathit{G}</code> is the <code class="reqn">\mathit{n\times p}</code>
genotype matrix for <code class="reqn">\mathit{n}</code> individuals and <code class="reqn">\mathit{p}</code> genetic variants in the region, 
<code class="reqn">\mathit{W}</code> is the <code class="reqn">\mathit{p\times p}</code> diagonal weight matrix, <code class="reqn">I</code> is the <code class="reqn">\mathit{p\times p}</code>
identity matrix, <code class="reqn">\rho</code> is pairwise correlation
coefficient between genetic effects (which will be adaptively selected from given <code>rho</code>),
and <code class="reqn">e</code> is the <code class="reqn">\mathit{p\times 1}</code> vector of ones. Given the shape parameters
of the beta function, <code>beta.par = c(a, b)</code>, 
the weights are defined using probability density function of the beta distribution:<br />
<br />
<code class="reqn">W_{i}=(B(a,b))^{^{-1}}MAF_{i}^{a-1}(1-MAF_{i})^{b-1} </code>,<br />
<br />
where <code class="reqn">MAF_{i}</code> is a minor allelic frequency for the <code class="reqn">i^{th}</code> genetic variant in the region,
which is estimated from genotypes, and <code class="reqn">B(a,b)</code> is the beta function. This way of defining weights
is the same as in original SKAT (see [Wu et al., 2011] for details). <code>beta.par = c(1, 1)</code> corresponds
to the unweighted SKAT. The same weighting principle can be applied in 'BT' (burden test, default weights <code>beta.par = c(1, 25)</code>),
as well as in 'PCA' and 'FLM' tests (unweighted by default).<br />
</p>
<p>Depending on the method option chosen, either Kuonen or Davies method is used to calculate P values from the
score statistics in SKAT. Both an Applied Statistics algorithm that inverts the characteristic function
of the mixture chisq [Davies, 1980] and a saddlepoint approximation [Kuonen, 1999] are nearly exact,
with the latter usually being a bit faster.<br />
</p>
<p>A hybrid approach has been proposed by Wu et al. [2016]. It uses the Davies' method with high accuracy,
and then switches to the saddlepoint approximation method when the Davies' method fails to converge.
This approach yields more accurate results in terms of type I errors, especially for small significance levels.
However, 'hybrid' method runs several times slower than the saddlepoint approximation method itself (i.e. 'kuonen' method).
We therefore recommend using the hybrid approach only for those regions that show significant (or nearly significant)
P values to ensure their accuracy.<br />
</p>
<p>'ACAT' is an adaptation of a set-based aggregated Cauchy association test (ACAT-V). It has been recently proposed
by Liu, Y. et al. (2019). This in the only gene-based test statistic that can be calculated from P values (Z scores) only,
and does not require SNP-SNP correlation info.
</p>
<p>Burden test ('BT', collapsing technique) suggests that the effects of causal genetic variants within a region have
the same direction and the majority of variants are causal. If this is not the case, other regional tests (SKAT and FLM)
are shown to have higher power compared to burden test [Svishcheva et al., 2015]. By default, 'BT' assigns weights
calculated using the beta distribution with shape parameters <code>beta.par = c(1, 25)</code>.<br />
</p>
<p>'PCA' test is based on the spectral decomposition of
correlation matrix among genetic variants. The number of top principal components will be chosen
in such a way that &gt;= <code>var.fraction</code> of region variance can be explained by these PCs.
By default, <code>var.fraction</code> = 0.85, i.e. PCs explain &gt;= 85% of region variance.<br />
</p>
<p>A similar principle is used in 'simpleM' to calculate the effective number of independent tests.<br />
</p>
<p>'FLM' test assumes that the effects of multiple genetic variants
can be described as a continuous function, which can be modelled through B-spline
or Fourier basis functions. When the number of basis functions
(set by <code class="reqn">k</code>) is less than the number of variants
within the region, the FLM test has an advantage of using
less degrees of freedom [Svishcheva, et al., 2015].<br />
</p>
<p>For genes with <code class="reqn">m &lt;= k</code>, functional linear models are equivalent to a
standard multiple linear regression, and the latter is used for these cases.
The ultimate model name is returned in output (the &quot;model&quot; column).<br />
</p>


<h3>Value</h3>

<p>A data frame with map info, P values, numbers of variants
and filtered variants for each of analyzed genes.
<br />
In addition:
<br />
- BT() returns gene-level estimates of effect sizes (betas) and their standard errors.
<br />
- FLM() returns the names of the functional models used for each region.
Names shortly describe the functional basis and the number of basis functions used.
E.g., &quot;F25&quot; means 25 Fourier basis functions, &quot;B15&quot; means 15 B-spline basis functions.
<br />
- PCA() returns the number of the principal components
used for each region and the proportion of genetic variance
they make up.
</p>


<h3>References</h3>

<p>Davies R.B. (1980) Algorithm AS 155: The Distribution of a Linear Combination of chi-2 Random Variables, Journal of the Royal Statistical Society. Series C (Applied Statistics), Vol. 29, N 3, P. 323-333.<br />
Kuonen D. (1999) Saddlepoint Approximations for Distributions of Quadratic Forms in Normal Variables. Biometrika, Vol. 86, No. 4, P. 929-935.<br />
Wu M.C., et al. (2011) Rare-variant association testing for sequencing data with the sequence kernel association test. Am. J. Hum. Genet., Vol. 89, P. 82-93.<br />
Lee S., et al. (2012) Optimal unified approach for rare variant association testing with application to small sample case-control whole-exome sequencing studies. American Journal of Human Genetics, 91, 224-237.<br />
Liu Y. et al. (2019) ACAT: a fast and powerful p value combination method for rare-variant analysis in sequencing studies. Am. J. Hum. Genet. 104, 410-421.
Svishcheva G.R., Belonogova N.M. and Axenovich T.I. (2015) Region-based association test for familial data under functional linear models. PLoS ONE 10(6): e0128999.<br />
Wu B., et al. (2016) On efficient and accurate calculation of significance p-values for sequence kernel association testing of variant set. Ann Hum Genet, 80(2): 123-135.<br />
Vsevolozhskaya O.A., et al. (2014) Functional Analysis of Variance for Association Studies. PLoS ONE 9(9): e105074.<br />
Fan R, Wang Y, Mills JL, Wilson AF, Bailey-Wilson JE, et al. (2013) Functional linear models for association analysis of quantitative traits. Genet Epidemiol 37: 726-42.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Using example score files "CFH.vcf.gz" and "CFH.full.vcf.gz"
# generated by prep.score.files() function (see examples for prep.score.files())

# Test available with minimal input (P values only):

score.file &lt;- system.file("testfiles/CFH.vcf.gz",
	package = "sumFREGAT")
ACAT(score.file, gene.file = "hg19", genes = "CFH")

# Tests that require P values, rsID, and SNP-SNP correlation info:

score.file &lt;- system.file("testfiles/CFH.vcf.gz",
	package = "sumFREGAT")
cor.path &lt;- system.file("testfiles/", package = "sumFREGAT")

sumchi(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)
simpleM(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)
minp(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)

# Tests available with full input including P values, rsID,
# betas, effect allele, allele frequencies (for default weighting),
# and correlation matrices:

score.file &lt;- system.file("testfiles/CFH.full.vcf.gz",
	package = "sumFREGAT")
cor.path &lt;- system.file("testfiles/", package = "sumFREGAT")

BT(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)
SKAT(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)
SKATO(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)

# Tests that require sample size to be provided as "n" argument:

PCA(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path, n = 85)
FLM(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path, n = 85)

</code></pre>

<hr>
<h2 id='prep.score.files'>Prepare score files</h2><span id='topic+prep.score.files'></span>

<h3>Description</h3>

<p>Calculates Z scores from P values and beta input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep.score.files(data, reference = "ref1KG.MAC5.EUR_AF.RData",
output.file.prefix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep.score.files_+3A_data">data</code></td>
<td>
<p>a file name or data.frame with two mandatory columns (case-insensitive header):<br /><br />
&quot;ID&quot;: unique names of genetic variants (rsIDs if 1000G correlation matrices will be used)<br />
&quot;P&quot;: P value<br /><br />
Chromosome and positions are desirable:<br /><br />
&quot;CHROM&quot;: chromosome<br />
&quot;POS&quot;: positions for the same build as in <code>gene.file</code> (see <code>gene-based test functions</code>)<br /><br />
Map data are required to assign genetic variants to genes. If not present in input file, the function will
attempt to link them from the reference file (see <code>reference</code> below).<br /><br />
Additional columns that can be present in input file to be used in gene-based tests:<br /><br />
&quot;EA&quot;: effect allele<br />
&quot;BETA&quot;: effect size (betas and genetic correlations should be calculated for the same genotype coding)<br />
&quot;EAF&quot;: effect allele frequency<br /><br />
Effect allele and size data can be proccesed with reference file only (see <code>reference</code> below).<br /><br />
&quot;REF&quot;: reference allele<br />
&quot;ALT&quot;: alternative allele<br /><br />
&quot;REF&quot; and &quot;ALT&quot; columns can be used to compare alleles with those in reference file and exclude genetic variants if
alleles do not match<br /><br />
Annotation columns:<br /><br />
&quot;ANNO&quot;: functional annotations (like &quot;intron_variant&quot;, &quot;synonymous&quot;, &quot;missense&quot; etc.)<br />
&quot;PROB&quot;, &quot;PROB1&quot;, &quot;PROB2&quot;, &quot;PROB3&quot; etc.: probabilities of a variant to be causal, can be passed to
<code>sumSTAAR()</code> (PHRED scale) or <code>FFGAS()</code><br /><br />
For example:<br /><br />
CHROM POS ID EA P BETA EAF<br />
1 196632134 1:196632134 T 0.80675 0.22946 0.00588<br />
1 196632386 1:196632386 A 0.48694 0.65208 0.00588<br />
1 196632470 1:196632470 G 0.25594 -0.19280 0.19412<br /><br />
Avoid rounding of betas and P values as this can affect the precision of regional tests.<br /><br />
The more data (columns) is present in input file, the more gene-based tests are available to run. ACAT test is available
with minimal input (rsIDs and P values). Together with correlation matrices (reference matrices calculated from 1000G data
are available at http://mga.bionet.nsc.ru/sumFREGAT/) it allows to run <code>minp()</code>, <code>simpleM</code>, and <code>sumchi()</code> tests.
Adding info on effect allele (&quot;EA&quot;) and effect size (&quot;BETA&quot;) enables essentially all sumFREGAT tests. Adding allele frequencies
enables standard weighting via beta distribution (see <code>gene-based test functions</code> for details).
</p>
</td></tr>
<tr><td><code id="prep.score.files_+3A_reference">reference</code></td>
<td>
<p>path to a reference file or data.frame with additional data needed to recode user <code>data</code> according to
correlation matrices that will be used. Reference file for 1000G correlation matrices is available at
http://mga.bionet.nsc.ru/sumFREGAT/. Reference file contains &quot;ID&quot; column with names of genetic variants that are used
in correlation matrices as well as &quot;REF&quot; and &quot;ALT&quot; columns with alleles that were coded during calculation of correlation
coefficients as 0 and 1, respectively. Effect sizes from <code>data</code> will be inverted for variants with effect alleles
different from &quot;ALT&quot; alleles in reference data. If presented, &quot;REF&quot; and &quot;ALT&quot; columns from the input data will be used to
sort out variants with alleles different from those in reference data. The reference file can also be a source of map data
and allele frequencies if they are not present in <code>data</code>. &quot;AF&quot; column in the reference file represents the allele
frequency of &quot;ALT&quot; allele.</p>
</td></tr>
<tr><td><code id="prep.score.files_+3A_output.file.prefix">output.file.prefix</code></td>
<td>
<p>if not set, the input file name will be used as output prefix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>does not return any value, writes output files with Z scores to be used in any type of
gene-based analysis in sumFREGAT (see <code>gene-based test functions</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

data &lt;- system.file("testfiles/CFH.dat", package = "sumFREGAT")
prep.score.files(data, output.file.prefix = "CFH")

# requires reference file "ref1KG.MAC5.EUR_AF.RData" (can be downloaded
# at http://mga.bionet.nsc.ru/sumFREGAT/)

data &lt;- system.file("testfiles/CFH.full.input.dat", package = "sumFREGAT")
prep.score.files(data, reference = "ref1KG.MAC5.EUR_AF.RData",
	output.file.prefix = "CFH.full")

data &lt;- system.file("testfiles/CFH.prob.dat", package = "sumFREGAT")
prep.score.files(data, reference = "ref1KG.MAC5.EUR_AF.RData",
	output.file.prefix = "CFH.prob")

## End(Not run)
</code></pre>

<hr>
<h2 id='sumSTAAR'>variant-Set Test for Association using Annotation infoRmation on summary statistics</h2><span id='topic+sumSTAAR'></span>

<h3>Description</h3>

<p>STAAR procedure developed by Li et al. (2020) adapted to use on summary statistics, with extentions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
sumSTAAR(score.file, gene.file, genes = 'all', cor.path = 'cor/',
tests = c('BT', 'SKAT', 'ACAT'), beta.par.matrix = rbind(c(1, 1), c(1, 25)), 
prob.causal = 'all', phred = TRUE, n = NA, mac.threshold = NA,
approximation = TRUE, write.file = FALSE, staar.output = TRUE, quiet = FALSE)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sumSTAAR_+3A_score.file">score.file</code></td>
<td>
<p>name of data file generated by <code>prep.score.files()</code>.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_gene.file">gene.file</code></td>
<td>
<p>can be a name of a custom text file listing genes in refFlat format. Values &quot;hg19&quot; and &quot;hg38&quot;
can be set to use default gene files containing protein-coding genes with start and stop positions from corresponding build.
Mind that the same build should be used in <code>score.file</code> and <code>gene.file</code>.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_genes">genes</code></td>
<td>
<p>character vector of gene names to be analyzed. Can be &quot;chr1&quot;, &quot;chr2&quot; etc. to analyze
all genes on a corresponding chromosome. If not set, function will
attempt to analyze all genes listed in <code>gene.file</code>.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_cor.path">cor.path</code></td>
<td>
<p>path to a folder with correlation matrix files (one file per each gene to be analyzed).
Correlation matrices in text format are allowed, though &quot;.RData&quot; is preferable as computationally efficient.
Each file should contain a square matrix with correlation coefficients (r) between genetic variants
of a gene. An example of correlation file format:<br />
&quot;snpname1&quot; &quot;snpname2&quot; &quot;snpname3&quot; ...<br />
&quot;snpname1&quot; 1 0.018 -0.003 ...<br />
&quot;snpname2&quot; 0.018 1 0.081 ...<br />
&quot;snpname3&quot; -0.003 0.081 1 ...<br />
...<br />
If genotypes are available, matrices can be generated as follows:<br />
<code>cor.matrix &lt;- cor(g)</code><br />
<code>save(cor.matrix, file = paste0(geneName, ".RData"))</code><br />
where <code>g</code> is a genotype matrix (nsample x nvariants) for a given gene with genotypes coded as 0, 1, 2
(coding should be exactly the same that was used to generate GWAS statistics).<br />
If genotypes are not available, correlations can be approximated through reference samples.
Reference matrices from 1KG European sample can be downloaded at http://mga.bionet.nsc.ru/sumFREGAT/.<br />
Names of correlation files should be constructed as &quot;geneName.RData&quot; (e.g. &quot;ABCG1.RData&quot;, &quot;ADAMTS1.RData&quot;, etc.)
for &quot;.RData&quot; format or &quot;geneName.txt&quot; for text format.<br />
Example corfiles can be found as:<br />
system.file(&quot;testfiles/CFH.cor&quot;, package = &quot;sumFREGAT&quot;)<br />
system.file(&quot;testfiles/CFH.RData&quot;, package = &quot;sumFREGAT&quot;)<br />
</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_tests">tests</code></td>
<td>
<p>a character vector with gene-based methods to be applied. By default, three methods are used: 'BT', 'SKAT',
and 'ACAT'. Other weighted tests ('SKATO', 'PCA', 'FLM') can be included. Tests that do not imply
weighting ('simpleM', 'minP', 'sumchi'), if listed, will be calculated once and combined along with other tests
into the total sumSTAAR p-value. ACAT performs with <code>gen.var.weights = "af"</code> for consistency with STAAR.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_beta.par.matrix">beta.par.matrix</code></td>
<td>
<p>an (n x 2) matrix with rows corresponding to n <code>beta.par</code> values used sequentially. Default value
<code>rbind(c(1, 1), c(1, 25))</code> means that <code>beta.par = c(1, 1)</code> and <code>beta.par = c(1, 25)</code> will be applied. <code>beta.par</code>
are two positive numeric shape parameters in the beta distribution to assign weights 
for each genetic variant as a function of minor allele frequency (MAF).</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_prob.causal">prob.causal</code></td>
<td>
<p>a character vector to define a set of annotations to be used. Annotations should be initially passed to
<code>prep.score.files</code> (see <code>prep.score.files()</code> function) as input file columns &quot;PROB&quot;, &quot;PROB1&quot;, &quot;PROB2&quot;, &quot;PROB3&quot; etc.
The same naming should be used for <code>prob.causal</code> argument. By default, all &quot;PROB&quot; columns will be used. Annotations 
are expected to be in PHRED scale, set <code>phred = FALSE</code> to use simple probabilities.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_phred">phred</code></td>
<td>
<p>a logical value indicating whether probabilities are in PHRED scale. If <code>TRUE</code> (default for <code>sumSTAAR()</code>), PHRED-to-probability
transformation will be performed for values indicated in <code>prob.causal</code>.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_n">n</code></td>
<td>
<p>size of the sample on which summary statistics were obtained. Should be assigned if 'PCA' or 'FLM'
are included in <code>tests</code>.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_mac.threshold">mac.threshold</code></td>
<td>
<p> an integer number. In ACAT, scores with MAC &lt;= 10 will be combined using Burden test. MACs are calculated
from MAFs, <code>n</code> must be set to apply <code>mac.threshold</code>. The original STAAR procedure performs with <code>mac.threshold = 10</code>.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_approximation">approximation</code></td>
<td>
<p>logical value indicating whether approximation should be used for SKAT, SKATO, PCA and FLM.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_write.file">write.file</code></td>
<td>
<p>output file name. If specified, output for all tests (as it proceeds) will be written 
to corresponding files.</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_staar.output">staar.output</code></td>
<td>
<p>logical value indicating whether extensive output format should be used (see Details).</p>
</td></tr>
<tr><td><code id="sumSTAAR_+3A_quiet">quiet</code></td>
<td>
<p><code>quiet = TRUE</code> suppresses excessive output from reading vcf file genewise.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The STAAR procedure has been recently proposed by Li et al. (2020) and described in detail therein. It calculates
a set of P values using a range of gene-based tests, beta distribution weights parameters, and weighting by each of
10 functional annotations. The P values are then combined using Cauchy method (see <code>ACATO()</code> function and Liu, Y. et al., (2019)).
</p>


<h3>Value</h3>

<p>With <code>staar.output = FALSE</code> returns a data table with a single P value for each test (combinations of differently weighted and unweighted
iterations of the same test) and a total sumSTAAR P value. ACAT-O method is used to combine P values (Liu, Y. et al., 2019).<br />
</p>
<p>If <code>staar.output = TRUE</code> the function returns a data frame of size (n.genes x (n.tests x n.beta.pars x (n.annotations + 1) + n.tests + 2)) containing P values
for combined tests and all individual tests. The output is analogous to that of original STAAR procedure. For example,
by default, the output will contain columns:<br /><br />
gene # gene symbol<br />
BT.1.1.PROB0 # P value for burden test with beta.par = c(1, 1) and no annotations<br />
BT.1.1.PROB1 # P value for burden test with beta.par = c(1, 1) and weighted by first annotation<br />
BT.1.1.PROB2 # P value for burden test with beta.par = c(1, 1) and weighted by second annotation<br />
...<br />
BT.1.1.PROB10 # P value for burden test with beta.par = c(1, 1) and weighted by tenth annotation<br />
BT.1.1.STAAR # combined P value of all burden tests with beta.par = c(1, 1)<br />
BT.1.25.PROB0 # P value for burden test with beta.par = c(1, 25) and no annotations<br />
BT.1.25.PROB1 # P value for burden test with beta.par = c(1, 25) and weighted by first annotation<br />
...<br />
BT.1.25.PROB10 # P value for burden test with beta.par = c(1, 25) and weighted by tenth annotation<br />
BT.1.25.STAAR # combined P value of all burden tests with beta.par = c(1, 25)<br />
SKAT.1.1.PROB0 # P value for SKAT with beta.par = c(1, 1) and no annotations<br />
SKAT.1.1.PROB1 # P value for SKAT with beta.par = c(1, 1) and weighted by first annotation<br />
...<br />
SKAT.1.1.PROB10 # P value for SKAT with beta.par = c(1, 1) and weighted by tenth annotation<br />
SKAT.1.1.STAAR # combined P value of all SKATs with beta.par = c(1, 1)<br />
SKAT.1.25.PROB0 # P value for SKAT with beta.par = c(1, 25) and no annotations<br />
...<br />
SKAT.1.25.PROB10 # P value for SKAT with beta.par = c(1, 25) and weighted by tenth annotation<br />
SKAT.1.25.STAAR # combined P value of all SKATs with beta.par = c(1, 25)<br />
ACAT.1.1.PROB0 # P value for ACAT with beta.par = c(1, 1) and no annotations<br />
ACAT.1.1.PROB1 # P value for ACAT with beta.par = c(1, 1) and weighted by first annotation<br />
...<br />
ACAT.1.1.PROB10 # P value for ACAT with beta.par = c(1, 1) and weighted by tenth annotation<br />
ACAT.1.1.STAAR # combined P value of all ACATs with beta.par = c(1, 1)<br />
ACAT.1.25.PROB0 # P value for ACAT with beta.par = c(1, 25) and no annotations<br />
...<br />
ACAT.1.25.PROB10 # P value for ACAT with beta.par = c(1, 25) and weighted by tenth annotation<br />
ACAT.1.25.STAAR # combined P value of all ACATs with beta.par = c(1, 25)<br />
sumSTAAR.ACAT_O # combined P value of all 'PROB0' gene-based tests (without weighting by annotations)<br />
sumSTAAR.STAAR_O # combined P value of all gene-based tests<br />
</p>


<h3>References</h3>

<p>Belonogova et al. (2022) SumSTAAR: A flexible framework for gene-based association studies using GWAS summary statistics. PLOS Comp Biol. https://doi.org/10.1371/journal.pcbi.1010172
Li X., et al. (2020) Dynamic incorporation of multiple in silico functional annotations empowers rare variant association analysis of large whole-genome sequencing studies at scale. Nature Genetics. DOI: 10.1038/s41588-020-0676-4.<br />
Liu Y. et al. (2019) ACAT: a fast and powerful p value combination method for rare-variant analysis in sequencing studies. Am. J. Hum. Genet. 104, 410-421.<br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cor.path &lt;- system.file("testfiles/", package = "sumFREGAT")
score.file &lt;- system.file("testfiles/CFH.prob.vcf.gz",
	package = "sumFREGAT")
sumSTAAR(score.file, prob.causal = "PROB", gene.file = "hg19",
	genes = "CFH", cor.path, quiet = TRUE)

## Not run: 

score.file &lt;- system.file("testfiles/CFH.prob.phred.vcf.gz",
	package = "sumFREGAT")
res &lt;- sumSTAAR(score.file, gene.file = "hg19", genes = "CFH",
	cor.path, quiet = TRUE)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
