<!DOCTYPE html><html lang="en"><head><title>Help for package SummaryLasso</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SummaryLasso}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#funcIndex'><p>Inputs for the functional annotations of SNPs.</p></a></li>
<li><a href='#gsfPEN'>
<p>SummaryLasso incorporating multiple traits and functional annotations of SNPs.</p></a></li>
<li><a href='#gsPEN'>
<p>SummaryLasso incorporating multiple traits</p></a></li>
<li><a href='#Nvec'><p>A vector of sample sizes for the q traits of the <code>summaryZ</code>.</p></a></li>
<li><a href='#plinkLD'><p>The LD info from output of the software (plink)</p></a></li>
<li><a href='#summaryZ'><p>The Z statistics from the univariate analysis of the association between 3614 SNPs and three traits respectively.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Building Polygenic Risk Score Using GWAS Summary Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Ting-Huei Chen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ting-Huei Chen &lt;tingstat22@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.5.0), gtools</td>
</tr>
<tr>
<td>Description:</td>
<td>Shrinkage estimator for polygenic risk prediction models based on summary statistics of genome-wide association studies.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-11-11 15:52:02 UTC; ting-huei</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-15 12:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='funcIndex'>Inputs for the functional annotations of SNPs.</h2><span id='topic+funcIndex'></span>

<h3>Description</h3>

<p>A 3614 x 3 matrix with (0,1) entry with 3614 SNPs and 
3 functional annotations. For the element at i-th row, j-th column, the entry 0 means 
SNP i without j-th functional annotation; entry 1 means otherwise.
follows: 
</p>

<ul>
<li><p> f1: The binary index for functional annotation 1.
</p>
</li>
<li><p> f2: The binary index for functional annotation 2.
</p>
</li>
<li><p> f3: The binary index for functional annotation 3.
</p>
</li></ul>


<h3>Usage</h3>

<pre><code class='language-R'>data(summaryZ)</code></pre>


<h3>Format</h3>

<p>A matrix with 3614 rows for the 3614 SNPs and 3 columns for functional annotations.</p>

<hr>
<h2 id='gsfPEN'>
SummaryLasso incorporating multiple traits and functional annotations of SNPs.
</h2><span id='topic+gsfPEN'></span>

<h3>Description</h3>

<p>SummaryLasso to model pleiotropy by introducing a group-Lasso type penalty, which is 
sensitive to select SNPs modestly associated with multiple traits and to incorporate functional 
annotations of SNPs simultaneously.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsfPEN(summaryZ, Nvec, plinkLD, NumIter = 1000, RupperVal = NULL, 
breaking = 1, numChrs = 22, ChrIndexBeta = 0, Init_summaryBetas = 0, 
Zscale = 1, tuningMatrix = NULL, penalty = "mixLOG", funcIndex, 
numfunc, p.Threshold = NULL, p.Thresholdpara = c(0.5, 10^-4, 4), 
taufactor = c(1/25, 1, 3), llim_length = 4, subtuning = 4, 
Lambda_limit = c(0.5, 0.9), Lenlam = 4, lambdavec_func = NULL, 
lambdavec_func_limit_len = c(1.5, 3), dfMax = NULL, outputAll = 0, 
warmStart = 0, customed = 0, AllTuningMatrix = NULL, SDvec = NULL, 
IniBeta = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gsfPEN_+3A_summaryz">summaryZ</code></td>
<td>

<p>The Z statistics of p SNPs from q GWA studies. A matrix with dimension p x q for p SNPs and q traits. 
The first column corresponds to the primary trait and the rest columns correspond to the secondary traits.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_nvec">Nvec</code></td>
<td>

<p>A vector of length q for the sample sizes of q GWA studies.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_plinkld">plinkLD</code></td>
<td>

<p>.ld file obtained from the LD calculation from plink.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_numiter">NumIter</code></td>
<td>

<p>The number of maximum iterations for the estimation procedure. 
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_rupperval">RupperVal</code></td>
<td>

<p>The maximum tolerable magnitude of the estimates of coefficients during the iterations. 
This is to avoid certain estimates of coefficients to diverge during the iterations. This may happen 
when the signs of the correlation coefficients were estimated incorrectly. The default value is 50 
times the maximum of coefficients from the input in absolute values.</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_breaking">breaking</code></td>
<td>

<p>A binary (0,1) variable to check if there are some certain estimates of coefficients to diverge 
during the iterations. This may happen when the signs of the correlation coefficients were 
estimated incorrectly. The default value is 1.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_numchrs">numChrs</code></td>
<td>

<p>The number of chromosomes used in the analysis. Current version of package does not use this argument. 
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_chrindexbeta">ChrIndexBeta</code></td>
<td>

<p>The chromosome index for each SNP. Current version of packge does not use this argument. 
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_init_summarybetas">Init_summaryBetas</code></td>
<td>

<p>Can be used to set the initial values of the coefficients for the iterative estimation. 
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_zscale">Zscale</code></td>
<td>

<p>A binary (0,1) variable to make the coefficients from different GWA studies with unequal sample sizes comparable.
The default value is 1.</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_tuningmatrix">tuningMatrix</code></td>
<td>

<p>Inputs for the tuning values of the tuning parameters. Default is null and it will be generated automatically.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_penalty">penalty</code></td>
<td>

<p>Current version of pacakge does not use this argument.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_funcindex">funcIndex</code></td>
<td>

<p>Inputs for the functional annotations of SNPs. A p x k matrix with (0,1) entry; p is the number of SNPs and 
k is the number of functional annotations. For the element at i-th row, j-th column, the entry 0 means 
SNP i without j-th functional annotation; entry 1 means otherwise.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_numfunc">numfunc</code></td>
<td>

<p>The number of functional annotations.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_p.threshold">p.Threshold</code></td>
<td>

<p>The p-values threshold to set up the tuning values of the baseline tuning parameter.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_p.thresholdpara">p.Thresholdpara</code></td>
<td>

<p>When p.Threshold is null, p.Threshold will be generated automatically based on the values of p.Thresholdpara.
The default values are c(0.5, 10^4, 5), where the first element is the maximum of the p-value threshold, 
the second element is the minimum, and the third element is total number of p-value thresholds to be 
generated from the minimum to the maximum.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_taufactor">taufactor</code></td>
<td>

<p>The weights to generate the tuning values for the tuning paramter &quot;tau&quot; and the default is c(1/25, 1, 10) 
times the median of the p summation of the coefficients for each SNP across q traits.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_llim_length">llim_length</code></td>
<td>

<p>The argument to set up the number of tuning values for lambdas between the lower and upper bound. 
The default value is 10.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_subtuning">subtuning</code></td>
<td>

<p>The argument to set up the number of tuning values for lambdas between the lower and upper bound. 
The default value is 50.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_lambda_limit">Lambda_limit</code></td>
<td>

<p>The quantiles to set up the tuning values of lambda. The default value is c(0.5, 0.9).
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_lenlam">Lenlam</code></td>
<td>

<p>The number of tuning values for lambda parameter without using the Log penalty. In other words, the 
initial <code>Lenlam</code> rows of the tuningMatrix are for summayLasso single trait analysis.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_lambdavec_func">lambdavec_func</code></td>
<td>

<p>The tuning values for the tuning parameters associated with the functional annotations.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_lambdavec_func_limit_len">lambdavec_func_limit_len</code></td>
<td>

<p>When lambdavec_func is null, lambdavec_func will be generated automatically based on the arguments of 
<code>lambdavec_func_limit_len</code>. The default values are c(1.5, 4). The first element is the maximum of the 
tuning value and the second element is the total number of the tuning values to be 
generated from 0 to the maximum.</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_dfmax">dfMax</code></td>
<td>

<p>The upper bound of the number of non-zero estimates of coefficients for the primary trait.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_outputall">outputAll</code></td>
<td>

<p>For internal checking usage. The default value is 0.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_warmstart">warmStart</code></td>
<td>

<p>For internal checking usage. The default value is 0.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_customed">customed</code></td>
<td>

<p>For internal checking usage. The default value is 0.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_alltuningmatrix">AllTuningMatrix</code></td>
<td>

<p>For internal checking usage. The default value is NULL.
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_sdvec">SDvec</code></td>
<td>

<p>The matrix of the standard error for regression coefficients. When the input of SummaryZ is at Z scale, 
let SDvec = NULL and it will be computed internally. 
</p>
</td></tr>
<tr><td><code id="gsfPEN_+3A_inibeta">IniBeta</code></td>
<td>

<p>A binary (0,1) variable to indicate if the regression coefficients need to be initialized or not. 1 is for yes.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the tuning values for the tuning parameters may need to be modified manually when the selected 
optimal tuning parameters are at the boundary of the inputs.</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>BetaMatrix</code></td>
<td>
<p>The output of the coefficients matrix with dimensions (total number of combinations 
of the tuning values times (pq)). Each column represents the vectorization of the p x q coefficients 
matrix given a particular combination of the tuning values (stacking its columns into a column vector). 
</p>
</td></tr> 
<tr><td><code>Numitervec</code></td>
<td>
<p>This vector shows the number of iterations to converge for each combination 
of the tuning values.</p>
</td></tr>
<tr><td><code>AllTuningMatrix</code></td>
<td>
<p>This matrix shows all combination of tuning values used in the estimation process. 
Its dimension is that total number of combinations 
of the tuning values times total number of tuning parameters.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ting-Huei Chen
</p>


<h3>References</h3>

<p>This R packages is based on the method introduced in the manuscript &quot;A comprehensive statistical framework for building polygenic risk prediction models based on summary statistics of genome-wide association studies.&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("summaryZ")
data("Nvec")
data("plinkLD")
data("funcIndex")
output = gsfPEN(summaryZ=summaryZ, Nvec=Nvec, plinkLD=plinkLD,funcIndex=funcIndex, 
numfunc=ncol(funcIndex))
</code></pre>

<hr>
<h2 id='gsPEN'>
SummaryLasso incorporating multiple traits
</h2><span id='topic+gsPEN'></span>

<h3>Description</h3>

<p>SummaryLasso to model pleiotropy by introducing a group-Lasso type penalty, 
which is sensitive to select SNPs modestly associated with multiple traits. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsPEN(summaryZ, Nvec, plinkLD, NumIter = 100, breaking = 1, numChrs = 22, 
ChrIndexBeta = 0, Init_summaryBetas = 0, Zscale = 1, RupperVal = NULL, 
tuningMatrix = NULL, penalty = c("mixLOG"), taufactor = c(1/25, 1, 10), 
llim_length = 10, subtuning = 50, Lambda_limit = c(0.5, 0.9), 
Lenlam_singleTrait = 200, dfMax = NULL, IniBeta = 0, inverseTuning = 0, 
outputAll = 0, warmStart = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gsPEN_+3A_summaryz">summaryZ</code></td>
<td>

<p>The Z statistics of p SNPs from q GWA studies. A matrix with dimension p x q for p SNPs and q traits.
The first column corresponds to the primary trait and the rest columns correspond to the secondary traits.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_nvec">Nvec</code></td>
<td>

<p>A vector of length q for the sample sizes of q GWA studies.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_plinkld">plinkLD</code></td>
<td>

<p>.ld file of the LD calculation from plink.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_numiter">NumIter</code></td>
<td>

<p>The number of maximum iteraions for the estimation procedure. 
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_breaking">breaking</code></td>
<td>

<p>A binary (0,1) variable to check if there are some certain estimates of coefficients to diverge during 
the iterations. This may happen when the signs of the correlation coefficinets were estimated incorrectly. 
The default value is 1.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_numchrs">numChrs</code></td>
<td>

<p>The number of chromosomes used in the analysis. Current version of pacakge does not use this argument.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_chrindexbeta">ChrIndexBeta</code></td>
<td>

<p>The chromosome index for each SNP. Current version of pacakge does not use this argument.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_init_summarybetas">Init_summaryBetas</code></td>
<td>

<p>Can be used to set the initial values of the coefficients for the iterative estimation. 
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_zscale">Zscale</code></td>
<td>

<p>A binary (0,1) variable to make the coefficients from different GWA studies with unequal sample sizes comparable.
The default value is 1.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_rupperval">RupperVal</code></td>
<td>

<p>The maximum tolerable magnitude of the estimates of coefficients during the iterations. 
This is to avoid a certain estimates of coefficients to diverge during the iterations. This may happen when 
the signs of the correlation coefficinets were estimated incorrectly. The default value is 50 times the 
maximum of coeffcients from the input in absolute values.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_tuningmatrix">tuningMatrix</code></td>
<td>

<p>Inputs for the tuning values of the tuning parameters. Default is null and it will be generated automatically. 
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_penalty">penalty</code></td>
<td>

<p>Current version of pacakge does not use this argument. 
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_taufactor">taufactor</code></td>
<td>

<p>The weights to generate the tuning values for the tuning paramter &quot;tau&quot; and the default is c(1/25, 1, 10) 
times the median of the p summation of the coefficients for each SNP across q traits.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_llim_length">llim_length</code></td>
<td>

<p>The argument to set up the number of tuning values for lambdas between the lower and upper bound. The default 
value is 10.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_subtuning">subtuning</code></td>
<td>

<p>The argument to set up the number of tuning values for lambdas between the lower and upper bound. The default 
value is 50.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_lambda_limit">Lambda_limit</code></td>
<td>

<p>The quantiles to set up the tuning values of lambda. The default value is c(0.5, 0.9).
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_lenlam_singletrait">Lenlam_singleTrait</code></td>
<td>

<p>The quantiles to set up the tuning values of lambda for single trait analysis.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_dfmax">dfMax</code></td>
<td>

<p>The upper bound of the number of non-zero estimates of coefficients for the primary trait.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_inibeta">IniBeta</code></td>
<td>

<p>A binary (0,1) variable to indicate if the regression coefficients need to be initialized or not. 1 is for yes.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_inversetuning">inverseTuning</code></td>
<td>

<p>For internal checking usage. The default value is 0.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_outputall">outputAll</code></td>
<td>

<p>For internal checking usage. The default value is 0.
</p>
</td></tr>
<tr><td><code id="gsPEN_+3A_warmstart">warmStart</code></td>
<td>

<p>For analysis with single trait or multiple traits without functional annotations, it is recommended to use 
warmStart = 1 to enhance computations. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the tuning values for the tuning parameters may need to be modified manually when the selected 
optimal tuning parameters are at the boundary of the inputs.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>BetaMatrix</code></td>
<td>
<p>The output of the coefficients matrix with dimensions (total number of combinations 
of the tuning values times (pq)). Each column represents the vectorization of the p x q coefficients 
matrix given a particular combination of the tuning values (stacking its columns into a column vector). 
</p>
</td></tr> 
<tr><td><code>Numitervec</code></td>
<td>
<p>This vector shows the number of iterations to converge for each combination 
of the tuning values.</p>
</td></tr>
<tr><td><code>AllTuningMatrix</code></td>
<td>
<p>This matrix shows all combination of tuning values used in the estimation process. 
Its dimension is that total number of combinations of the tuning values times total number of tuning parameters.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ting-Huei Chen
</p>


<h3>References</h3>

<p>This R packages is based on the method introduced in the manuscript &quot;A comprehensive statistical framework for building polygenic risk prediction models based on summary statistics of genome-wide association studies.&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("summaryZ")
data("Nvec")
data("plinkLD")
output = gsPEN(summaryZ=summaryZ, Nvec=Nvec, plinkLD=plinkLD)

</code></pre>

<hr>
<h2 id='Nvec'>A vector of sample sizes for the q traits of the <code>summaryZ</code>.</h2><span id='topic+Nvec'></span>

<h3>Description</h3>

<p>A vector of q sample sizes for the q set of Z statistics corresponding to the q columns of <code>summaryZ</code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Nvec)</code></pre>


<h3>Format</h3>

<p>A vector with q elements, where q is the number of columns of <code>summaryZ</code>.</p>

<hr>
<h2 id='plinkLD'>The LD info from output of the software (plink)</h2><span id='topic+plinkLD'></span>

<h3>Description</h3>

<p> The LD information is crucial for the analysis by SummaryLasso. 
The reference alleles used to obtained for the Z statsitics or the 
regression coefficients have to be the sames as those used for the LD calculation. 
This file can be obtained directly from the output of the LD calculation by the 
software (plink); for example the output can be like plink.ld.
On the other hand, the user can calcuate the LD based on their prefered tools.
The variables are as
follows: 
</p>

<ul>
<li><p> CHR_A: The chromosome of SNP_A 
</p>
</li>
<li><p> BP_A: The positions of SNP_A  
</p>
</li>
<li><p> SNP_A: The names of SNP_A 
</p>
</li>
<li><p> CHR_B: The chromosome of SNP_B 
</p>
</li>
<li><p> BP_B: The positions of SNP_B 
</p>
</li>
<li><p> SNP_B: The names of SNP_B    
</p>
</li>
<li><p> R: The correlation between SNP_A and SNP_B
</p>
</li></ul>


<h3>Usage</h3>

<pre><code class='language-R'>data(plinkLD)</code></pre>


<h3>Format</h3>

<p>A data frame with 205959 rows and 7 columns</p>


<h3>References</h3>


<ul>
<li><p> Purcell S, et al. (2007) 
PLINK: a toolset for whole-genome association and population-based 
linkage analysis. <em>American Journal of Human Genetics</em>, <strong>81</strong>.
</p>
</li></ul>


<hr>
<h2 id='summaryZ'>The Z statistics from the univariate analysis of the association between 3614 SNPs and three traits respectively.</h2><span id='topic+summaryZ'></span>

<h3>Description</h3>

<p>These Z statsitics are obtained from simulated datasets.  The variables are as
follows: 
</p>

<ul>
<li><p> Z1: The Z statistics from trait 1; the primary trait.
</p>
</li>
<li><p> Z2: The Z statistics from trait 2; the secondary trait.
</p>
</li>
<li><p> Z2: The Z statistics from trait 3; the secondary trait.
</p>
</li></ul>


<h3>Usage</h3>

<pre><code class='language-R'>data(summaryZ)</code></pre>


<h3>Format</h3>

<p>A matrix with 3614 rows for the 3614 SNPs and 3 columns for 3 traits.</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
