<!DOCTYPE html><html lang="en"><head><title>Help for package SUMO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SUMO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#SUMO'><p>SUMO: A Package for Simulating Multi-Omics Data</p></a></li>
<li><a href='#divide_features_one'><p>Dividing features to create vectors with signal in the first omic for single data</p></a></li>
<li><a href='#divide_features_two'><p>Dividing features to create vectors with signal in the second omic for single data</p></a></li>
<li><a href='#divide_samples'><p>Global Variable</p></a></li>
<li><a href='#divide_vector'><p>Divide features into randomized subsets based on factor Segments</p></a></li>
<li><a href='#feature_selection_one'><p>Dividing features to create vectors with signal in the first omic</p></a></li>
<li><a href='#feature_selection_two'><p>Dividing features to create vectors with signal in the second omic</p></a></li>
<li><a href='#OmixCraftHD'><p>Simulation of high-dimensional data with predefined single factor or multiple factors in multi-omics</p></a></li>
<li><a href='#plot_factor'><p>Visualization of factor scores</p></a></li>
<li><a href='#plot_simData'><p>Visualizing the simulated data using image map and 3D visualization</p></a></li>
<li><a href='#plot_weights'><p>Visualizing the loading of the features</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Generating Multi-Omics Datasets</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Designed to generate multi-omics datasets that closely reflect biological complexity, the package allows for testing, validation, and benchmarking of multi-omics integrative methods. The simulated data includes one or multiple predefined signals (latent/unobserved factors), giving users complete control over the data-generated characteristics. Tini, Giulia, et al (2019) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbx167">doi:10.1093/bib/bbx167</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by/4.0">CC BY 4.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, gridExtra, rlang</td>
</tr>
<tr>
<td>Collate:</td>
<td>'divide_vector.R' 'divide_features_two.R'
'divide_features_one.R' 'feature_selection_two.R'
'feature_selection_one.R' 'divide_samples.R' 'OmixCraftHD.R'
'SUMO.R' 'plot_factor.R' 'plot_simData.R' 'plot_weights.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-20 13:58:44 UTC; Lenovo</td>
</tr>
<tr>
<td>Author:</td>
<td>Bernard Isekah Osang'ir
    <a href="https://orcid.org/0000-0002-5557-3602"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Bernard Isekah Osang'ir [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bernard Isekah Osang'ir &lt;Bernard.Osangir@sckcen.be&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-22 13:10:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='SUMO'>SUMO: A Package for Simulating Multi-Omics Data</h2><span id='topic+SUMO-package'></span><span id='topic+SUMO'></span>

<h3>Description</h3>

<p>SUMO provides tools for simulating complex multi-omics data, allowing researchers to generate datasets that reflect the biological intricacies found in real-world data. This package aims to fill a gap in current omics research by providing users with flexible and customizable tools for generating synthetic data that can be used for method development, benchmarking of Multi-Omics methods.
</p>


<h3>Details</h3>

<p>Key features of SUMO include:
</p>

<ul>
<li> <p><strong>Simulating Multi-Omics Data</strong>: Generate multi-layered datasets with customizable structures.
</p>
</li>
<li> <p><strong>Flexible Data Generation</strong>: Control over various simulation parameters such as the sample, noise levels, and signal positions.
</p>
</li>
<li> <p><strong>Visualization Tools</strong>: Functions to visualize simulated data, including scatterplots, histogram, heatmaps and 3D visualization.
</p>
</li></ul>

<p>Key functions include:
</p>

<ul>
<li> <p><code>OmixCraftHD()</code>: Generates synthetic multi-omics datasets based on user-defined parameters.
</p>
</li>
<li> <p><code>plot_simData</code>: Visualizes the structure of the generated data
</p>
</li>
<li> <p><code>plot_factor</code>: Visualizes the raw factor scores, for visual identification of signal noise
</p>
</li>
<li> <p><code>plot_weights</code>: Visualizes the raw features loadings, for visual identification of signal noise
</p>
</li></ul>



<h3>Value</h3>

<p>A list containing synthetic multi-omics datasets, which may include several omics layers such as gene expression, proteomics, methylation, etc. Each layer is represented as a matrix or data frame where rows correspond to samples and columns correspond to features.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Bernard Isekah Osang'ir <a href="mailto:Bernard.Osangir@sckcen.be">Bernard.Osangir@sckcen.be</a> (<a href="https://orcid.org/0000-0002-5557-3602">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Bernard Isekah Osang'ir <a href="mailto:bernard.osangir@uhasselt.be">bernard.osangir@uhasselt.be</a>
</p>
</li></ul>


<hr>
<h2 id='divide_features_one'>Dividing features to create vectors with signal in the first omic for single data</h2><span id='topic+divide_features_one'></span>

<h3>Description</h3>

<p>Dividing features to create vectors with signal in the first omic for single data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divide_features_one(n_features_one, num.factor)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="divide_features_one_+3A_n_features_one">n_features_one</code></td>
<td>
<p>number of features of first omic</p>
</td></tr>
<tr><td><code id="divide_features_one_+3A_num.factor">num.factor</code></td>
<td>
<p>number of factors (should be set to '1')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric vectors. Each vector contains 80% of the features from one segment of the original feature set.
The number of segments is determined by the number of factors provided (<code>num.factor</code>).
</p>

<ul>
<li><p> Each vector is a subset of the original feature set, selected randomly to contain 80% of the elements from the segment.
</p>
</li></ul>

<p>If the minimum segment size constraint is too large for the given feature length and number of segments, the function retries using the <code>divide_vector()</code> function.
</p>

<hr>
<h2 id='divide_features_two'>Dividing features to create vectors with signal in the second omic for single data</h2><span id='topic+divide_features_two'></span>

<h3>Description</h3>

<p>Dividing features to create vectors with signal in the second omic for single data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divide_features_two(n_features_two, num.factor)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="divide_features_two_+3A_n_features_two">n_features_two</code></td>
<td>
<p>number of features of first omic</p>
</td></tr>
<tr><td><code id="divide_features_two_+3A_num.factor">num.factor</code></td>
<td>
<p>number of factors (should be set to '1')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric vectors. Each vector contains 80% of the features from one segment of the original feature set.
The number of segments is determined by the number of factors provided (<code>num.factor</code>).
</p>

<ul>
<li><p> Each vector is a subset of the original feature set, selected randomly to contain 80% of the elements from the segment.
</p>
</li></ul>

<p>If the minimum segment size constraint is too large for the given feature length and number of segments, the function retries using the <code>divide_vector()</code> function.
</p>

<hr>
<h2 id='divide_samples'>Global Variable</h2><span id='topic+divide_samples'></span>

<h3>Description</h3>

<p>A global variable used in multiple functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divide_samples(n_samples, num, min_size)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="divide_samples_+3A_n_samples">n_samples</code></td>
<td>
<p>number of samples</p>
</td></tr>
<tr><td><code id="divide_samples_+3A_num">num</code></td>
<td>
<p>number of factors</p>
</td></tr>
<tr><td><code id="divide_samples_+3A_min_size">min_size</code></td>
<td>
<p>Minimum length of any samples scores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric vectors.
If <code>num == 1</code>, the list contains a single vector representing a random selection of between 10% and 55% of the elements from the full dataset.
If <code>num &gt; 1</code>, the list contains <code>num</code> vectors, each representing 75% of the elements from one of the <code>num</code> segments of the dataset.
The segmentation ensures that all segments are at least the size of <code>min_size</code>.
If the segment sizes are too small, the function retries the segmentation process.
</p>

<hr>
<h2 id='divide_vector'>Divide features into randomized subsets based on factor Segments</h2><span id='topic+divide_vector'></span>

<h3>Description</h3>

<p>Divide features into randomized subsets based on factor Segments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divide_vector(n_samples, num, min_size)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="divide_vector_+3A_n_samples">n_samples</code></td>
<td>
<p>number of samples</p>
</td></tr>
<tr><td><code id="divide_vector_+3A_num">num</code></td>
<td>
<p>number of factors</p>
</td></tr>
<tr><td><code id="divide_vector_+3A_min_size">min_size</code></td>
<td>
<p>Minimum length of any samples scores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric vectors. Each vector contains 80% of the features from one segment of the original feature set.
The number of segments is determined by the number of factors provided (<code>num.factor</code>).
</p>

<ul>
<li><p> Each vector is a subset of the original feature set, selected randomly to contain 80% of the elements from the segment.
</p>
</li></ul>

<p>Only used when the minimum segment size constraint is too large for the given feature length and number of segments.
</p>

<hr>
<h2 id='feature_selection_one'>Dividing features to create vectors with signal in the first omic</h2><span id='topic+feature_selection_one'></span>

<h3>Description</h3>

<p>Dividing features to create vectors with signal in the first omic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>feature_selection_one(n_features_one, num.factor, no_factor)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="feature_selection_one_+3A_n_features_one">n_features_one</code></td>
<td>
<p>number of features of first omic</p>
</td></tr>
<tr><td><code id="feature_selection_one_+3A_num.factor">num.factor</code></td>
<td>
<p>type of factors - single or multiple</p>
</td></tr>
<tr><td><code id="feature_selection_one_+3A_no_factor">no_factor</code></td>
<td>
<p>number of factors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric vectors.
</p>

<ul>
<li><p> The first vector contains a consecutive subset of the first <code>num_elements</code> from the original vector.
</p>
</li>
<li><p> The subsequent vectors are sub-vectors derived from remaining segments, each containing 40% of the elements from the corresponding segment.
</p>
</li>
<li><p> If <code>num.factor == 'multiple'</code>, the segments are divided based on <code>no_factor</code>, and the function ensures the segments meet the size constraints.
</p>
</li>
<li><p> The function recursively retries segmentation if any segment size is smaller than the minimum constraint of 10 elements.
</p>
</li></ul>

<p>The function returns an error if the input parameters or constraints are invalid (e.g., <code>num.factor</code> is not &quot;multiple&quot; or <code>no_factor</code> is missing).
</p>

<hr>
<h2 id='feature_selection_two'>Dividing features to create vectors with signal in the second omic</h2><span id='topic+feature_selection_two'></span>

<h3>Description</h3>

<p>Dividing features to create vectors with signal in the second omic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>feature_selection_two(n_features_two, num.factor, no_factor)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="feature_selection_two_+3A_n_features_two">n_features_two</code></td>
<td>
<p>number of features of second omic</p>
</td></tr>
<tr><td><code id="feature_selection_two_+3A_num.factor">num.factor</code></td>
<td>
<p>type of factors - single or multiple</p>
</td></tr>
<tr><td><code id="feature_selection_two_+3A_no_factor">no_factor</code></td>
<td>
<p>number of factors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric vectors.
The first vector represents a random subset of between 10% and 60% of the elements from the original feature vector.
The remaining vectors represent 40% of the elements from each of the segments created from the rest of the feature vector.
If the segment sizes are too small or there are overlapping elements across the final vectors, the function retries and returns a new list of vectors.
Each vector is guaranteed to have no overlapping elements with the others.
If the input parameters are invalid, the function throws an error.
</p>

<hr>
<h2 id='OmixCraftHD'>Simulation of high-dimensional data with predefined single factor or multiple factors in multi-omics</h2><span id='topic+OmixCraftHD'></span>

<h3>Description</h3>

<p>Simulation of high-dimensional data with predefined single factor or multiple factors in multi-omics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OmixCraftHD(
  vector_features = c(2000, 2000),
  n_samples = 50,
  sigmas_vector = c(3, 5),
  n_factors = 3,
  num.factor = "multiple",
  advanced_dist = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="OmixCraftHD_+3A_vector_features">vector_features</code></td>
<td>
<p>Vector of features assigned to the two simulated datasets respectively '1' first dataset, '2' second dataset</p>
</td></tr>
<tr><td><code id="OmixCraftHD_+3A_n_samples">n_samples</code></td>
<td>
<p>The number of samples common between the two simulated datasets</p>
</td></tr>
<tr><td><code id="OmixCraftHD_+3A_sigmas_vector">sigmas_vector</code></td>
<td>
<p>Vector for the noise variability for the two simulated datasets respectively,  '1' first dataset, '2' second dataset</p>
</td></tr>
<tr><td><code id="OmixCraftHD_+3A_n_factors">n_factors</code></td>
<td>
<p>Number of predefined factors</p>
</td></tr>
<tr><td><code id="OmixCraftHD_+3A_num.factor">num.factor</code></td>
<td>
<p>Category of factors to be simulated specified as 'single', or 'multiple'.</p>
</td></tr>
<tr><td><code id="OmixCraftHD_+3A_advanced_dist">advanced_dist</code></td>
<td>
<p>Applicable only when num.factor = 'multiple'. Contains six possible arguments, &rdquo;, NULL, 'mixed', 'omic.one', or 'omic.two', 'exclusive'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li> <p><code>dataset_1</code>: A matrix or data frame representing the first simulated dataset with rows as samples and columns as features.
</p>
</li>
<li> <p><code>dataset_2</code>: A matrix or data frame representing the second simulated dataset with rows as samples and columns as features.
</p>
</li>
<li> <p><code>factors</code>: A matrix representing the predefined factors used in generating the datasets. If <code>num.factor</code> is 'single', this contains one set of factors. If <code>num.factor</code> is 'multiple', it contains multiple sets of factors.
</p>
</li>
<li> <p><code>noise</code>: A list containing the noise terms added to both datasets based on the <code>sigmas_vector</code>.
</p>
</li>
<li> <p><code>factor_assignment</code>: A vector indicating how factors are assigned to datasets, depending on the <code>num.factor</code> and <code>advanced_dist</code> settings.
</p>
</li></ul>

<p>The output provides simulated multi-omics datasets with predefined latent factors and noise, which can be used to model complex biological data structures.
</p>
<p>A list containing:
</p>

<ul>
<li> <p><code>dataset_1</code>: A matrix or data frame representing the first simulated dataset with rows as samples and columns as features.
</p>
</li>
<li> <p><code>dataset_2</code>: A matrix or data frame representing the second simulated dataset with rows as samples and columns as features.
</p>
</li>
<li> <p><code>factors</code>: A matrix representing the predefined factors used in generating the datasets. If <code>num.factor</code> is 'single', this contains one set of factors. If <code>num.factor</code> is 'multiple', it contains multiple sets of factors.
</p>
</li>
<li> <p><code>noise</code>: A list containing the noise terms added to both datasets based on the <code>sigmas_vector</code>.
</p>
</li>
<li> <p><code>factor_assignment</code>: A vector indicating how factors are assigned to datasets, depending on the <code>num.factor</code> and <code>advanced_dist</code> settings.
</p>
</li></ul>

<p>The output provides simulated multi-omics datasets with predefined latent factors and noise, which can be used to model complex biological data structures.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Examples
set.seed(1234)
output_obj &lt;- OmixCraftHD(
  vector_features = c(2000,3000),
  sigmas_vector=c(8,5),
  n_samples=100,
  n_factors=5,
  num.factor='multiple',
  advanced_dist='mixed'
)
output_obj &lt;- OmixCraftHD(
  vector_features = c(5000,3000),
  sigmas_vector=c(3,4),
  n_samples=30, n_factors=1
)

</code></pre>

<hr>
<h2 id='plot_factor'>Visualization of factor scores</h2><span id='topic+plot_factor'></span>

<h3>Description</h3>

<p>Visualization of factor scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_factor(sim_object = NULL, factor_num = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_factor_+3A_sim_object">sim_object</code></td>
<td>
<p>R object containing data to be plotted</p>
</td></tr>
<tr><td><code id="plot_factor_+3A_factor_num">factor_num</code></td>
<td>
<p>Factor to be plotted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the factor scores for the specified factor (or all factors) in <code>sim_object</code>.
If <code>factor_num = 'all'</code>, a combined plot of all factors is returned. If a specific <code>factor_num</code> is provided, the plot for that factor is returned.
The plot can be further customized or displayed using standard <code>ggplot2</code> functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Examples
output_obj &lt;- OmixCraftHD(
  vector_features = c(2000,3000),
  sigmas_vector=c(3,4),
  n_samples=30,
  n_factors=1
)
plot_factor(sim_object = output_obj, factor_num = 1)
plot_factor(sim_object = output_obj, factor_num = 'all')
</code></pre>

<hr>
<h2 id='plot_simData'>Visualizing the simulated data using image map and 3D visualization</h2><span id='topic+plot_simData'></span>

<h3>Description</h3>

<p>Visualizing the simulated data using image map and 3D visualization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_simData(sim_object, type = "heatmap")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_simData_+3A_sim_object">sim_object</code></td>
<td>
<p>R object containing simulated data to be plotted</p>
</td></tr>
<tr><td><code id="plot_simData_+3A_type">type</code></td>
<td>
<p>type of the plot. Heatmap for image plot and 3D for persp 3D plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function generates and displays a plot based on the specified <code>type</code>.
If <code>type</code> is &quot;heatmap&quot;, the function displays a 2D heatmap of the simulated data.
If <code>type</code> is &quot;3D&quot;, the function creates a 3D surface plot of the simulated data.
The function does not return any values but generates the requested plot as a side effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Examples
output_obj &lt;- OmixCraftHD(
  vector_features = c(2000,3000),
  sigmas_vector=c(8,5),
  n_samples=100,
  n_factors=5,
  num.factor='multiple',
  advanced_dist='mixed'
)
plot_simData(sim_object = output_obj, type = "heatmap")
plot_simData(sim_object = output_obj, type = "3D")
</code></pre>

<hr>
<h2 id='plot_weights'>Visualizing the loading of the features</h2><span id='topic+plot_weights'></span>

<h3>Description</h3>

<p>Visualizing the loading of the features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_weights(
  sim_object = NULL,
  factor_num = 1,
  data = "omic.one",
  type = "scatter"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_weights_+3A_sim_object">sim_object</code></td>
<td>
<p>R object containing data to be plotted</p>
</td></tr>
<tr><td><code id="plot_weights_+3A_factor_num">factor_num</code></td>
<td>
<p>Factor to be plotted.</p>
</td></tr>
<tr><td><code id="plot_weights_+3A_data">data</code></td>
<td>
<p>Section of the integrated data to be plotted, omic.one or omic.two are the options</p>
</td></tr>
<tr><td><code id="plot_weights_+3A_type">type</code></td>
<td>
<p>Type of plot. Scatter plot and histogram are the only allowed plots</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
If <code>type</code> is &quot;scatter&quot;, the function returns a scatter plot visualizing the loadings of features for the selected factor.
If <code>type</code> is &quot;histogram&quot;, the function returns a histogram displaying the distribution of the loadings for the selected factor.
The plot visualizes either <code>omic.one</code> or <code>omic.two</code> data based on the user input in the <code>data</code> parameter.
The ggplot object can be further modified or directly plotted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Examples
output_obj &lt;- OmixCraftHD(
  vector_features = c(2000,3000),
  sigmas_vector=c(8,5),
  n_samples=100,
  n_factors=4,
  num.factor='multiple',
  advanced_dist='mixed'
)
plot_weights(sim_object = output_obj, factor_num = 1, data = 'omic.one', type = 'scatter')
plot_weights(sim_object = output_obj, factor_num = 1, data = 'omic.one', type = 'histogram')
plot_weights(sim_object = output_obj, factor_num = 1, data = 'omic.two', type = 'scatter')
plot_weights(sim_object = output_obj, factor_num = 1, data = 'omic.two', type = 'histogram')
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
