<!DOCTYPE html><html><head><title>Help for package superbiclust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {superbiclust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BiclustSet-class'><p>Class <code>BiclustSet</code></p></a></li>
<li><a href='#BiclustSet-methods'><p>Constructor of BiclustSet object</p></a></li>
<li><a href='#combine'>
<p>Combine two Biclust objects into one</p></a></li>
<li><a href='#getStats'>
<p>Get frequency statistic for the columns and rows membership</p></a></li>
<li><a href='#HCLtree'>
<p>Hierarchical structure of bicluster output</p></a></li>
<li><a href='#jaccardMat'>
<p>Jaccard similarity Matrix for bicluster output</p></a></li>
<li><a href='#kulczynskiMat'>
<p>Kulczynski similarity Matrix for bicluster output</p></a></li>
<li><a href='#ochiaiMat'>
<p>Ochiai similarity Matrix for bicluster output</p></a></li>
<li><a href='#plotProfilesAcrossAllSamples'>
<p>Plot Gene Expression Profiles Across All Samples of the Original Data</p></a></li>
<li><a href='#plotProfilesWithinBicluster'>
<p>Plot Gene Expression Profiles within a (Core) Bicluster</p></a></li>
<li><a href='#plotSuper'>
<p>Plot gene profiles within biclusters</p></a></li>
<li><a href='#plotSuperAll'>
<p>Plot gene profiles for all samples in the data</p></a></li>
<li><a href='#sensitivityMat'>
<p>Sensitivity Matrix for bicluster output</p></a></li>
<li><a href='#similarity'>
<p>Similarity Matrix for bicluster output</p></a></li>
<li><a href='#sorensenMat'>
<p>Sorensen similarity Matrix for bicluster output</p></a></li>
<li><a href='#specificityMat'>
<p>Specificity Matrix for bicluster output</p></a></li>
<li><a href='#superbiclust-package'>
<p>generating robust biclusters form the set of biclusters</p>
</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Generating Robust Biclusters from a Bicluster Set (Ensemble
Biclustering)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-05-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Tatsiana Khamiakova</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tatsiana Khamiakova &lt;t.tavita@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>biclust, fabia</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, Matrix, graphics, stats</td>
</tr>
<tr>
<td>Description:</td>
<td>Biclusters are submatrices in the data matrix which
        satisfy certain conditions of homogeneity. Package contains
        functions for generating robust biclusters with respect to the
        initialization parameters for a given bicluster solution
        contained in a bicluster set in data, the procedure is also
        known as ensemble biclustering. The set of biclusters is
        evaluated based on the similarity of its elements (the
        overlap), and afterwards the hierarchical tree is constructed
        to obtain cut-off points for the classes of robust biclusters.
        The result is a number of robust (or super) biclusters with
        none or low overlap.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-27 07:22:06 UTC; tkhamiak</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-28 06:00:02 UTC</td>
</tr>
<tr>
<td>Repository/R-Forge/Project:</td>
<td>superbiclust</td>
</tr>
<tr>
<td>Repository/R-Forge/Revision:</td>
<td>14</td>
</tr>
<tr>
<td>Repository/R-Forge/DateTimeStamp:</td>
<td>2014-11-27 15:39:29</td>
</tr>
</table>
<hr>
<h2 id='BiclustSet-class'>Class <code>BiclustSet</code></h2><span id='topic+BiclustSet'></span><span id='topic+BiclustSet-class'></span><span id='topic+show+2CBiclustSet-method'></span><span id='topic+summary+2CBiclustSet-method'></span>

<h3>Description</h3>

<p>BiclustSet Class contains the biclustering result in a form: bicluster rows and bicluster columns
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("BiclustSet", ...)</code>. 
The variety of inputs variety of inputs (isa2, fabia, biclust,...) can be used.
</p>


<h3>Slots</h3>


<dl>
<dt><code>GenesMembership</code>:</dt><dd><p>logical, object of class <code>"matrix"</code>, with row membership within a bicluster </p>
</dd>
<dt><code>ColumnMembership</code>:</dt><dd><p>logical, object of class <code>"matrix"</code>, with column membership within a bicluster </p>
</dd>
<dt><code>Number</code>:</dt><dd><p>code&quot;numeric&quot;, number of biclusters in the set</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>

<hr>
<h2 id='BiclustSet-methods'>Constructor of BiclustSet object</h2><span id='topic+BiclustSet-methods'></span><span id='topic+BiclustSet+2CANY-method'></span><span id='topic+BiclustSet+2CBiclust-method'></span><span id='topic+BiclustSet+2CFactorization-method'></span><span id='topic+BiclustSet+2Clist-method'></span>

<h3>Description</h3>

<p>The method extract relevant information from a variety of biclustering input and constructs a BiclustSet object
</p>


<h3>Methods</h3>


<dl>
<dt><code>signature(x = "ANY")</code></dt><dd>
</dd>
<dt><code>signature(x = "Biclust")</code></dt><dd>
<p>Converts Biclust objects into BiclustSet object
</p>
</dd>
<dt><code>signature(x = "Factorization")</code></dt><dd>
<p>Converts FABIA Factorization object into BiClustSet
</p>
</dd>
<dt><code>signature(x = "list")</code></dt><dd>
<p>Converts  a list with biclustering results into BiClustSet
</p>
</dd>
</dl>


<h3>See Also</h3>

<p><code><a href="#topic+BiclustSet-class">BiclustSet</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 test &lt;- matrix(rnorm(5000), 100, 50)
 test[11:20,11:20] &lt;- rnorm(100, 3, 0.1)
 test[17:26,21:30] &lt;- rnorm(100, 3, 0.1)
#Run FABIA
 set.seed(1)
 FabiaRes1 &lt;- fabia(test)
#construct BiclustSet object from FABIA output
 FabiabiclustSet &lt;- BiclustSet(FabiaRes1)
 FabiabiclustSet 
</code></pre>

<hr>
<h2 id='combine'>
Combine two Biclust objects into one
</h2><span id='topic+combine'></span>

<h3>Description</h3>

<p>Combine two Biclust objects into one
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine(x,y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_+3A_x">x</code></td>
<td>

<p>1st Biclust object containing bicluster results
</p>
</td></tr>
<tr><td><code id="combine_+3A_y">y</code></td>
<td>

<p>2nd Biclust object containing bicluster results
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a biclust function returns empty set, joined result contains only results of non-empty object. 
Info and Parameters slots of a &quot;Biclust&quot; object contain information about both biclustering runs.
</p>


<h3>Value</h3>

<p>object of a class Biclust
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BiclustSet-class">BiclustSet</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#combine output of two biclust objects
 test &lt;- matrix(rnorm(5000), 100, 50)
 test[11:20,11:20] &lt;- rnorm(100, 3, 0.1)
 test[17:26,21:30] &lt;- rnorm(100, 3, 0.1)
 set.seed(1)
 PlaidRes1 &lt;- biclust(x=test, method=BCPlaid())
 set.seed(2)
 PlaidRes2 &lt;- biclust(x=test, method=BCPlaid())
 combinedRes &lt;- combine(PlaidRes1,PlaidRes2)
 summary(combinedRes)
</code></pre>

<hr>
<h2 id='getStats'>
Get frequency statistic for the columns and rows membership
</h2><span id='topic+getStats'></span>

<h3>Description</h3>

<p>For a given Bicluster set, for each row and column in data, compute frequency of apperance within a bicluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getStats(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getStats_+3A_x">x</code></td>
<td>

<p>Biclust object containing bicluster results
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of column and row frequencies
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>

<hr>
<h2 id='HCLtree'>
Hierarchical structure of bicluster output
</h2><span id='topic+HCLtree'></span>

<h3>Description</h3>

<p>Constructs and plots hierarchical tree of biclusters output based on the similarity matrix</p>


<h3>Usage</h3>

<pre><code class='language-R'>HCLtree(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HCLtree_+3A_x">x</code></td>
<td>

<p>Similarity object containing pairwise similarity indices for all biclusters in the output
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function operates on a similarity matrix, which is converted to the distance between biclusters according to 
<code class="reqn">dist(a,b)= 1-sim(a,b)</code>, where the smaller the distance, the higher is overlap in terms of rows and columns.
The tree is constructed using complete method and plotted. 
Further, the structure must be explored and robust or super-biclusters obtained after cutting the tree.
<code>identify</code> function can be applied to the hierarchical tree to see the partition and get the plots of biclusters. 
</p>


<h3>Value</h3>

<p>tree
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+similarity">similarity</a></code>, <code><a href="#topic+plotSuper">plotSuper</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#compute sensitivity for BiMAX biclusters
 test &lt;- matrix(rnorm(5000), 100, 50)
 test[11:20,11:20] &lt;- rnorm(100, 3, 0.1)
 test[17:26,21:30] &lt;- rnorm(100, 3, 0.1)
 testBin &lt;- binarize(test,2)
 res &lt;- biclust(x=testBin, method=BCBimax(), minr=4, minc=4, number=10)
 BiMaxBiclustSet &lt;-  BiclustSet(res)
 SensitivityMatr&lt;- similarity(BiMaxBiclustSet,index="sensitivity")
#construct hierarchical clustering based on the sensitivities
HCLsensitivity &lt;- HCLtree(SensitivityMatr) 
plot(HCLsensitivity, main="structure of bicluster solution")
</code></pre>

<hr>
<h2 id='jaccardMat'>
Jaccard similarity Matrix for bicluster output
</h2><span id='topic+jaccardMat'></span>

<h3>Description</h3>

<p>computes jaccard similarity matrix for biclusters in two bicluster sets</p>


<h3>Usage</h3>

<pre><code class='language-R'>jaccardMat(x, y, type=c("rows", "cols", "both"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jaccardMat_+3A_x">x</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="jaccardMat_+3A_y">y</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="jaccardMat_+3A_type">type</code></td>
<td>

<p>whether to compute Jaccard index in two dimensions, row dimension or column dimension 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function operates on BiclustSet objects and computes pairwise similarity based on the common elements between biclusters. 
The Jaccard similarity score <code class="reqn">ja</code> for two biclusters A and B is computed as </p>
<p style="text-align: center;"><code class="reqn">ja=\frac{|A\cap B|}{|A\cup B|}</code>
</p>


<h3>Value</h3>

<p>matrix of pairwise Jaccard indices
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+similarity">similarity</a></code>,<code><a href="#topic+kulczynskiMat">kulczynskiMat</a></code>, <code><a href="#topic+ochiaiMat">ochiaiMat</a></code>, <code><a href="#topic+sensitivityMat">sensitivityMat</a></code>,
<code><a href="#topic+specificityMat">specificityMat</a></code>,<code><a href="#topic+sorensenMat">sorensenMat</a></code> 
</p>

<hr>
<h2 id='kulczynskiMat'>
Kulczynski similarity Matrix for bicluster output
</h2><span id='topic+kulczynskiMat'></span>

<h3>Description</h3>

<p>computes Kulczynski similarity matrix for biclusters in two bicluster sets</p>


<h3>Usage</h3>

<pre><code class='language-R'>kulczynskiMat(x, y, type=c("rows", "cols", "both"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kulczynskiMat_+3A_x">x</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="kulczynskiMat_+3A_y">y</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="kulczynskiMat_+3A_type">type</code></td>
<td>

<p>whether to compute Kulczynski index in two dimensions, row dimension or column dimension 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function operates on BiclustSet objects and computes pairwise similarity based on the common elements between biclusters. 
Kulczynski similarity score <code class="reqn">ku</code> for two biclusters A and B is computed as 
</p>
<p style="text-align: center;"><code class="reqn">ku = 2|A\cap B|\left(\frac{1}{|A|} + \frac{1}{|B|}\right)</code>
</p>



<h3>Value</h3>

<p>matrix of pairwise Kulczynski indices
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+similarity">similarity</a></code>,<code><a href="#topic+jaccardMat">jaccardMat</a></code>, <code><a href="#topic+ochiaiMat">ochiaiMat</a></code>, <code><a href="#topic+sensitivityMat">sensitivityMat</a></code>,
<code><a href="#topic+specificityMat">specificityMat</a></code>,<code><a href="#topic+sorensenMat">sorensenMat</a></code> 
</p>

<hr>
<h2 id='ochiaiMat'>
Ochiai similarity Matrix for bicluster output
</h2><span id='topic+ochiaiMat'></span>

<h3>Description</h3>

<p>Computes Ochiai similarity matrix for biclusters in two bicluster sets</p>


<h3>Usage</h3>

<pre><code class='language-R'>ochiaiMat(x, y, type=c("rows", "cols", "both"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ochiaiMat_+3A_x">x</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="ochiaiMat_+3A_y">y</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="ochiaiMat_+3A_type">type</code></td>
<td>

<p>whether to compute Ochiai index in two dimensions, row dimension or column dimension 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function operates on BiclustSet objects and computes pairwise similarity based on the common elements between biclusters. 
The Ochiai similarity score <code class="reqn">ja</code> for two biclusters A and B is computed as 
</p>
<p style="text-align: center;"><code class="reqn">oc=\frac{|A\cap B|}{\sqrt{|A| |B|}}</code>
</p>



<h3>Value</h3>

<p>matrix of pairwise Ochiai indices
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+similarity">similarity</a></code>,<code><a href="#topic+jaccardMat">jaccardMat</a></code>, <code><a href="#topic+ochiaiMat">ochiaiMat</a></code>, <code><a href="#topic+sensitivityMat">sensitivityMat</a></code>,
<code><a href="#topic+specificityMat">specificityMat</a></code>,<code><a href="#topic+sorensenMat">sorensenMat</a></code> 
</p>

<hr>
<h2 id='plotProfilesAcrossAllSamples'>
Plot Gene Expression Profiles Across All Samples of the Original Data
</h2><span id='topic+plotProfilesAcrossAllSamples'></span>

<h3>Description</h3>

<p>Plot Gene Expression Profiles Across All Samples of the Original Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotProfilesAcrossAllSamples(x, coreBiclusterGenes, coreBiclusterSamples)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotProfilesAcrossAllSamples_+3A_x">x</code></td>
<td>

<p>data
</p>
</td></tr>
<tr><td><code id="plotProfilesAcrossAllSamples_+3A_corebiclustergenes">coreBiclusterGenes</code></td>
<td>

<p>vector of genes belonging to bicluster
</p>
</td></tr>
<tr><td><code id="plotProfilesAcrossAllSamples_+3A_corebiclustersamples">coreBiclusterSamples</code></td>
<td>

<p>vector of samples belonging to bicluster
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot re-sorts the samples by bicluster membership and highlights them in red. Only the genes of a bicluster are plotted.
</p>


<h3>Value</h3>

<p>no return value; a plot is drawn to the current device
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BiclustSet-class">BiclustSet</a></code> ,<code><a href="#topic+plotSuperAll">plotSuperAll</a></code>
</p>

<hr>
<h2 id='plotProfilesWithinBicluster'>
Plot Gene Expression Profiles within a (Core) Bicluster
</h2><span id='topic+plotProfilesWithinBicluster'></span>

<h3>Description</h3>

<p>Plot Gene Expression Profiles within a (Core) Bicluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotProfilesWithinBicluster(x, main = "", sampleNames, geneNames = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotProfilesWithinBicluster_+3A_x">x</code></td>
<td>

<p>expression matrix (of class 'matrix') for the subset of genes and samples corresponding to the bicluster under study
</p>
</td></tr>
<tr><td><code id="plotProfilesWithinBicluster_+3A_main">main</code></td>
<td>

<p>main title for the graph
</p>
</td></tr>
<tr><td><code id="plotProfilesWithinBicluster_+3A_samplenames">sampleNames</code></td>
<td>

<p>names of the samples to be used for annotating the x axis (character vector
of length equal to the number of columns of the expression matrix 'x' (representing the bicluster)
</p>
</td></tr>
<tr><td><code id="plotProfilesWithinBicluster_+3A_genenames">geneNames</code></td>
<td>

<p>names of the genes to be plotted in a legend (character vector of length equal
to the number of rows of the expression matrix 'x'); only suitable for biclusters containing a
small number of genes 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return value; a plot is drawn to the current device
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSuper">plotSuper</a></code> 
</p>

<hr>
<h2 id='plotSuper'>
Plot gene profiles within biclusters
</h2><span id='topic+plotSuper'></span>

<h3>Description</h3>

<p>Function for plotting gene profiles for compounds within constructed super-bicluster</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSuper(x, data, BiclustSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSuper_+3A_x">x</code></td>
<td>

<p>a vector, containing indices of biclusters, to be joined for obtaining the robust bicluster  
</p>
</td></tr>
<tr><td><code id="plotSuper_+3A_data">data</code></td>
<td>

<p>matrix, dataset, from which the bicluster results are obtained  
</p>
</td></tr>
<tr><td><code id="plotSuper_+3A_biclustset">BiclustSet</code></td>
<td>

<p>a BiclustSet object containing bicluster output  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function constructs a robust bicluster from a set of biclusters identified in a hierarchical tree and
plots gene profiles for columns in a robust bicluster. Each line represents a gene from a bicluster. 
The bicluster is saved as <code>Biclust</code> object which can be further plotetd by available functions from <code>biclust package</code>.
The information about the number of biclusters used to generate the resulting robust bicluster is saved in <code>Call</code> slot of the object.
This information is important to see how often the bicluster has been discovered under different parameter settings (e.g. initialization seeds) 
Indices used as an input can be obtained by identify function or by cutting the tree.
</p>


<h3>Value</h3>

<p>biclust object containing bicluster and the information about bicluster subset used to generate it
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HCLtree">HCLtree</a></code>, <code><a href="#topic+plotSuperAll">plotSuperAll</a></code>, <code><a href="#topic+plotProfilesWithinBicluster">plotProfilesWithinBicluster</a></code>
</p>

<hr>
<h2 id='plotSuperAll'>
Plot gene profiles for all samples in the data
</h2><span id='topic+plotSuperAll'></span>

<h3>Description</h3>

<p>Function for plotting bicluster gene profiles for all samples in the data</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSuperAll(x, data, BiclustSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSuperAll_+3A_x">x</code></td>
<td>

<p>a vector, containing indices of biclusters, to be joined for obtaining the robust bicluster  
</p>
</td></tr>
<tr><td><code id="plotSuperAll_+3A_data">data</code></td>
<td>

<p>matrix, dataset, from which the bicluster results are obtained  
</p>
</td></tr>
<tr><td><code id="plotSuperAll_+3A_biclustset">BiclustSet</code></td>
<td>

<p>a BiclustSet object containing bicluster output  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function constructs a robust bicluster from a subset of biclusters specified in x argument and plots the expression profiles 
</p>


<h3>Value</h3>

<p>biclust object
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HCLtree">HCLtree</a></code>, <code><a href="#topic+plotProfilesAcrossAllSamples">plotProfilesAcrossAllSamples</a></code>
</p>

<hr>
<h2 id='sensitivityMat'>
Sensitivity Matrix for bicluster output
</h2><span id='topic+sensitivityMat'></span>

<h3>Description</h3>

<p>Computes sensitivity matrix for biclusters in two bicluster sets</p>


<h3>Usage</h3>

<pre><code class='language-R'>sensitivityMat(x, y, type=c("rows", "cols", "both"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sensitivityMat_+3A_x">x</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="sensitivityMat_+3A_y">y</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="sensitivityMat_+3A_type">type</code></td>
<td>

<p>whether to compute sensitivity in two dimensions, row dimension or column dimension 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function operates on BiclustSet objects and computes pairwise similarity based on the common elements between biclusters. 
Sensitivity inclusion score <code class="reqn">sen</code> of biclusters A and B is computed as 
</p>
<p style="text-align: center;"><code class="reqn">sen=\frac{|A\cap B|}{|A|}</code>
</p>



<h3>Value</h3>

<p>matrix of pairwise sensitivities
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+similarity">similarity</a></code>,<code><a href="#topic+jaccardMat">jaccardMat</a></code>, <code><a href="#topic+ochiaiMat">ochiaiMat</a></code>, <code><a href="#topic+kulczynskiMat">kulczynskiMat</a></code>,
<code><a href="#topic+specificityMat">specificityMat</a></code>,<code><a href="#topic+sorensenMat">sorensenMat</a></code> 
</p>

<hr>
<h2 id='similarity'>
Similarity Matrix for bicluster output
</h2><span id='topic+similarity'></span>

<h3>Description</h3>

<p>computes similarity matrix for the biclustering output based on one of the pairwise similarity indices of biclusters in a given bicluster set</p>


<h3>Usage</h3>

<pre><code class='language-R'>similarity(x, index = "jaccard", type="rows")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="similarity_+3A_x">x</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="similarity_+3A_index">index</code></td>
<td>

<p>similarity index for the biclusters in output
</p>
</td></tr>
<tr><td><code id="similarity_+3A_type">type</code></td>
<td>

<p>whether to perform similarity in two dimensions, &quot;both&quot; (recommended for biclustering), row dimension,
&quot;rows&quot; (default, requires less computations) or column dimension &quot;cols&quot;
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function operates on BiclustSet object and computes pairwise similarity based on the common elements between biclusters. 
<code>type</code> variable controls whether similarity index is constructed for all elements, or in one dimension (rows or columns) only. 
In general, similarity indices for one dimension (row or column) are higher than for two-dimensions. 
Several options for similarity indices are offered: jaccard (default), kulczynski, sensitivity, specificity, sorensen  and ochiai indices.
</p>


<h3>Value</h3>

<p>a &quot;similarity&quot; object containing similarity matrix
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HCLtree">HCLtree</a></code>, <code><a href="#topic+plotSuper">plotSuper</a></code> , <code><a href="#topic+jaccardMat">jaccardMat</a></code>,<code><a href="#topic+kulczynskiMat">kulczynskiMat</a></code>, <code><a href="#topic+ochiaiMat">ochiaiMat</a></code>, <code><a href="#topic+sensitivityMat">sensitivityMat</a></code>,
<code><a href="#topic+specificityMat">specificityMat</a></code>,<code><a href="#topic+sorensenMat">sorensenMat</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#compute sensitivity for BiMAX biclusters
 test &lt;- matrix(rnorm(5000), 100, 50)
 test[11:20,11:20] &lt;- rnorm(100, 3, 0.1)
 test[17:26,21:30] &lt;- rnorm(100, 3, 0.1)
 testBin &lt;- binarize(test,2)
 res &lt;- biclust(x=testBin, method=BCBimax(), minr=4, minc=4, number=10)
 BiMaxBiclustSet &lt;-  BiclustSet(res)
 SensitivityMatr&lt;- similarity(BiMaxBiclustSet,index="sensitivity", type="rows")
 SensitivityMatr
</code></pre>

<hr>
<h2 id='sorensenMat'>
Sorensen similarity Matrix for bicluster output
</h2><span id='topic+sorensenMat'></span>

<h3>Description</h3>

<p>Computes Sorensen similarity matrix for biclusters in two bicluster sets</p>


<h3>Usage</h3>

<pre><code class='language-R'>sorensenMat(x, y, type=c("rows", "cols", "both"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sorensenMat_+3A_x">x</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="sorensenMat_+3A_y">y</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="sorensenMat_+3A_type">type</code></td>
<td>

<p>whether to compute Sorensen index in two dimensions, row dimension or column dimension 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function operates on BiclustSet objects and computes pairwise similarity based on the common elements between biclusters. 
Sorensen similarity score <code class="reqn">so</code> for two biclusters A and B is computed as 
</p>
<p style="text-align: center;"><code class="reqn">so = \frac{2|A\cap B|}{|A| + |B|}</code>
</p>



<h3>Value</h3>

<p>matrix of pairwise Sorensen indices
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+similarity">similarity</a></code>,<code><a href="#topic+jaccardMat">jaccardMat</a></code>, <code><a href="#topic+ochiaiMat">ochiaiMat</a></code>, <code><a href="#topic+sensitivityMat">sensitivityMat</a></code>,
<code><a href="#topic+specificityMat">specificityMat</a></code>,<code><a href="#topic+kulczynskiMat">kulczynskiMat</a></code> 
</p>

<hr>
<h2 id='specificityMat'>
Specificity Matrix for bicluster output
</h2><span id='topic+specificityMat'></span>

<h3>Description</h3>

<p>Computes specificity matrix for biclusters in two bicluster sets</p>


<h3>Usage</h3>

<pre><code class='language-R'>specificityMat(x, y, type=c("rows", "cols", "both"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="specificityMat_+3A_x">x</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="specificityMat_+3A_y">y</code></td>
<td>

<p>BiclustSet object containing row and column indicators of bicluster membership, number of biclusters
</p>
</td></tr>
<tr><td><code id="specificityMat_+3A_type">type</code></td>
<td>

<p>whether to compute specificity in two dimensions, row dimension or column dimension 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function operates on BiclustSet objects and computes pairwise similarity based on the common elements between biclusters. 
Sensitivity inclusion score <code class="reqn">spe</code> of biclusters A and B is computed as 
</p>
<p style="text-align: center;"><code class="reqn">spe=\frac{|A\cap B|}{|B|}</code>
</p>



<h3>Value</h3>

<p>matrix of pairwise specificities
</p>


<h3>Author(s)</h3>

<p>Tatsiana Khamiakova
</p>


<h3>See Also</h3>

<p><code><a href="#topic+similarity">similarity</a></code>,<code><a href="#topic+jaccardMat">jaccardMat</a></code>, <code><a href="#topic+ochiaiMat">ochiaiMat</a></code>, <code><a href="#topic+kulczynskiMat">kulczynskiMat</a></code>,
<code><a href="#topic+sensitivityMat">sensitivityMat</a></code>,<code><a href="#topic+sorensenMat">sorensenMat</a></code> 
</p>

<hr>
<h2 id='superbiclust-package'>
generating robust biclusters form the set of biclusters
</h2><span id='topic+superbiclust-package'></span><span id='topic+superbiclust'></span>

<h3>Description</h3>

<p>The package contains a number of functions for computing similarity matrix of the biclusters obtained by a variety of methods, initialization seeds or various parameter settings.
It uses biclustering output as generated by biclust or fabia. isa2 package can be used to generate the biclusters as well, however, a prior conversion is needed to a biclust object by using isa2.biclust() function. 
The matrix is used for the construction of hierarchical tree based on overall similarity, row similarity or column similarity to obtain cut-off points for the similarity metric of choice.
Various statistics are output per bicluster set: a number of a given gene(compound) or gene (compound) set has been present in any bicluster of output or per run.
After the tree is cut, the robiust or super biclusters are obtained in a form of biclust object, which can further be used for plotting of biclusters.
Biclusters are submatrices in the data which satisfy certain conditions of homogeneity. For more details on biclusters and biclustering see Madeira and Oliveira (2004).</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> superbiclust</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.99</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2012-08-23</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Tatsiana Khamiakova &lt;tatsiana.khamiakova@uhasselt.be&gt;
</p>


<h3>References</h3>

<p>Madeira and Oliveira (2004) Biclustering algorithms for biological data analysis: a survey. IEEE/ACM Trans Comput Biol Bioinform. 2004 Jan-Mar;1(1):24-45.
Shi et al. (2010) A bi-ordering approach to linking gene expression	with clinical annotations in gastric cancer. BMC Bioinformatics. 11. pages 477. 
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
