<!DOCTYPE html><html><head><title>Help for package SuperExactTest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SuperExactTest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Cancer'><p>Cancer Census Dataset</p></a></li>
<li><a href='#cis.eqtls'><p>cis-eQTLs</p></a></li>
<li><a href='#cpsets'><p>Multi-Set Intersection Probability</p></a></li>
<li><a href='#deBarcode'><p>Decrypt Barcode</p></a></li>
<li><a href='#GWAS'><p>GAWS Catalog Dataset</p></a></li>
<li><a href='#intersect'><p>Set Operations</p></a></li>
<li><a href='#intersectElements'><p>Find Intersection Membership</p></a></li>
<li><a href='#jaccard'><p>Calculate Jaccard Index</p></a></li>
<li><a href='#MSET'><p>Exact Test of Multi-Set Intersection</p></a></li>
<li><a href='#msets'><p>Class to Contain Multi-Set Intersections</p></a></li>
<li><a href='#plot.msets'><p>Draw Multi-Set Intersections</p></a></li>
<li><a href='#summary.msets'><p>Summarize an msets Object</p></a></li>
<li><a href='#SuperExactTest'><p>SuperExactTest Package</p></a></li>
<li><a href='#supertest'><p>Calculate Intersections Among Multiple Sets and Perform Statistical Tests</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Exact Test and Visualization of Multi-Set Intersections</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-03-29</td>
</tr>
<tr>
<td>Author:</td>
<td>Minghui Wang, Yongzhong Zhao and Bin Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Minghui Wang &lt;minghui.wang@mssm.edu&gt;</td>
</tr>
<tr>
<td>Contact:</td>
<td>Minghui Wang &lt;minghui.wang@mssm.edu&gt;, Bin Zhang
&lt;bin.zhang@mssm.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Identification of sets of objects with shared features is a common operation in all disciplines. Analysis of intersections among multiple sets is fundamental for in-depth understanding of their complex relationships. This package implements a theoretical framework for efficient computation of statistical distributions of multi-set intersections based upon combinatorial theory, and provides multiple scalable techniques for visualizing the intersection statistics. The statistical algorithm behind this package was published in Wang et al. (2015) &lt;<a href="https://doi.org/10.1038%2Fsrep16923">doi:10.1038/srep16923</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>grid (&ge; 3.1.0), methods, R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mw201608/SuperExactTest/">https://github.com/mw201608/SuperExactTest/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mw201608/SuperExactTest/issues">https://github.com/mw201608/SuperExactTest/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-03-23 15:58:48 UTC; mhwan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-03-23 16:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='Cancer'>Cancer Census Dataset</h2><span id='topic+Cancer'></span>

<h3>Description</h3>

<p>This example dataset contains a list of seven cancer predisposition gene sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Cancer)</code></pre>


<h3>Details</h3>

<p>The seven cancer predisposition gene sets are:
</p>

<ul>
<li><p> NRG (Rahman, N. Realizing the promise of cancer predisposition genes. Nature 2014, 505:302-308);
</p>
</li>
<li><p> NBG (Tamborero, D. et al. Comprehensive identification of mutational cancer driver genes across 12 tumor types. Scientific reports 2013, 3:2650);
</p>
</li>
<li><p> LDG (Kandoth, C. et al. Mutational landscape and significance across 12 major cancer types. Nature 2013, 502:333-339);
</p>
</li>
<li><p> GGG (Lawrence, M. S. et al. Discovery and saturation analysis of cancer genes across 21 tumour types. Nature 2014, 505:495-501);
</p>
</li>
<li><p> ELG (Garraway, L. A. &amp; Lander, E. S. Lessons from the cancer genome. Cell 2013, 153:17-37);
</p>
</li>
<li><p> CCG (Futreal, P. A. et al. A census of human cancer genes. Nature reviews. Cancer 2004, 4:177-183);
</p>
</li>
<li><p> BVG (Vogelstein, B. et al. Cancer genome landscapes. Science 2013, 339:1546-1558).
</p>
</li></ul>



<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+supertest">supertest</a></code></p>

<hr>
<h2 id='cis.eqtls'>cis-eQTLs</h2><span id='topic+cis.eqtls'></span>

<h3>Description</h3>

<p>This example dataset contains a list of cis-eQTL genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eqtls)</code></pre>


<h3>Details</h3>

<p>A list is included in this dataset: <code>cis.eqtls</code>, which contains four sets of cis-eQTL genes published by Gibbs et al (PLOS Genetics 2010, 6:e1000952) as deposited in the eQTL Browser (http://www.ncbi.nlm.nih.gov/projects/gap/eqtl/index.cgi).
The four sets of cis-eQTL genes were detected in four different brain regions from Gibbs: brain cerebellum (CB), brain frontal cortex region (FC), brain temporal cortex region (TC), and brain pons region (PONS) respectively.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+supertest">supertest</a></code></p>

<hr>
<h2 id='cpsets'>Multi-Set Intersection Probability</h2><span id='topic+cpsets'></span><span id='topic+dpsets'></span>

<h3>Description</h3>

<p>Density and distribution function of multi-set intersection test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dpsets(x,L,n,log.p =FALSE)
cpsets(x,L,n,lower.tail=TRUE,log.p=FALSE,
       simulation.p.value=FALSE,number.simulations=1000000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpsets_+3A_x">x</code></td>
<td>
<p>integer, number of elements overlap among all sets.</p>
</td></tr>
<tr><td><code id="cpsets_+3A_l">L</code></td>
<td>
<p>vector, set sizes.</p>
</td></tr> 
<tr><td><code id="cpsets_+3A_n">n</code></td>
<td>
<p>integer, background population size.</p>
</td></tr>
<tr><td><code id="cpsets_+3A_lower.tail">lower.tail</code></td>
<td>
<p>logical; if TRUE, probability is <code>P[overlap &lt;= x]</code>, otherwise, <code>P[overlap &gt; x]</code>.</p>
</td></tr>
<tr><td><code id="cpsets_+3A_log.p">log.p</code></td>
<td>
<p>logical; if TRUE, probability p is given as <code>log(p)</code>.</p>
</td></tr>
<tr><td><code id="cpsets_+3A_simulation.p.value">simulation.p.value</code></td>
<td>
<p>logical; if TRUE, probability p is computed from simulation.</p>
</td></tr>
<tr><td><code id="cpsets_+3A_number.simulations">number.simulations</code></td>
<td>
<p>integer; number of simulations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>dpsets</code> gives the density and <code>cpsets</code> gives the distribution function.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+supertest">supertest</a></code>, <code><a href="#topic+MSET">MSET</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#set up fake data
n=500; A=260; B=320; C=430; D=300; x=170
(d=dpsets(x,c(A,B,C,D),n))
(p=cpsets(x,c(A,B,C,D),n,lower.tail=FALSE))

## End(Not run)
</code></pre>

<hr>
<h2 id='deBarcode'>Decrypt Barcode</h2><span id='topic+deBarcode'></span>

<h3>Description</h3>

<p>Decrypt barcode information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deBarcode(barcode, setnames, collapse=' &amp; ')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deBarcode_+3A_barcode">barcode</code></td>
<td>
<p>a vector of character strings, encoding the intersection combination.</p>
</td></tr>
<tr><td><code id="deBarcode_+3A_setnames">setnames</code></td>
<td>
<p>set names.</p>
</td></tr>
<tr><td><code id="deBarcode_+3A_collapse">collapse</code></td>
<td>
<p>an optional character string to separate the results. See <code><a href="base.html#topic+paste">paste</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>barcode</code> are character strings of '0' and '1', indicating absence or presence of each set in a intersection combination.
</p>


<h3>Value</h3>

<p>A vector.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>deBarcode(c('01011','10100'), c('S1','S2','S3','S4','S5'))
</code></pre>

<hr>
<h2 id='GWAS'>GAWS Catalog Dataset</h2><span id='topic+GWAS'></span>

<h3>Description</h3>

<p>This example dataset contains a list of gene sets associated with six types of clinical traits curated in the GWAS Catalog.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GWAS)</code></pre>


<h3>Details</h3>

<p>The six clinical traits are:
</p>

<ul>
<li><p> NEU (Bipolar disorder and schizophrenia, Schizophrenia, Major depressive disorder, Alzheimer's disease, Parkinson's disease, Cognitive performance, Bipolar disorder);
</p>
</li>
<li><p> INF (Crohn's disease, Ulcerative colitis, Inflammatory bowel disease, Rheumatoid arthritis, Multiple sclerosis, Systemic lupus erythematosus);
</p>
</li>
<li><p> CVD (Type 2 diabetes, Coronary heart disease, Blood pressure, total Cholesterol, HDL cholesterol, Triglycerides);
</p>
</li>
<li><p> HT (height);
</p>
</li>
<li><p> IgG (IgG glycosylation);
</p>
</li>
<li><p> OB (obesity, obesity related traits).
</p>
</li></ul>



<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+supertest">supertest</a></code></p>

<hr>
<h2 id='intersect'>Set Operations</h2><span id='topic+intersect'></span><span id='topic+union'></span>

<h3>Description</h3>

<p>Performs set union and intersection on multiple input vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>union(x, y, ...)
intersect(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intersect_+3A_x">x</code>, <code id="intersect_+3A_y">y</code>, <code id="intersect_+3A_...">...</code></td>
<td>
<p>vectors (of the same mode) containing a sequence of items (conceptually) with no duplicated values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions extend the the same functions in the <code>base</code> package to handle more than two input vectors.
</p>


<h3>Value</h3>

<p>A vector of the same mode as x or y for intersect, and of a common mode for union. 
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##not run##
</code></pre>

<hr>
<h2 id='intersectElements'>Find Intersection Membership</h2><span id='topic+intersectElements'></span>

<h3>Description</h3>

<p>Find intersections and assign element to intersection combinations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intersectElements(x, mutual.exclusive=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intersectElements_+3A_x">x</code></td>
<td>
<p>list; a collection of sets.</p>
</td></tr>
<tr><td><code id="intersectElements_+3A_mutual.exclusive">mutual.exclusive</code></td>
<td>
<p>logical; see <code>Details</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See example below for the use of <code>mutual.exclusive</code>.
</p>


<h3>Value</h3>

<p>A data.frame with two columns:
</p>
<table>
<tr><td><code>Entry</code></td>
<td>
<p>set elements.</p>
</td></tr>
<tr><td><code>barcode</code></td>
<td>
<p>intersection combination that each entry belongs to.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
sets=list(S1=sample(letters,10), S2=sample(letters,5), S3=sample(letters,7))
intersectElements(sets,mutual.exclusive=TRUE)
intersectElements(sets,mutual.exclusive=FALSE)
</code></pre>

<hr>
<h2 id='jaccard'>Calculate Jaccard Index</h2><span id='topic+jaccard'></span>

<h3>Description</h3>

<p>This function calculates Jaccard indices between pairs of sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jaccard(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jaccard_+3A_x">x</code></td>
<td>
<p>list, a collect of sets.</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>A matrix of pairwise Jaccard indices.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#set up fake data
x=list(S1=letters[1:20], S2=letters[10:26], S3=sample(letters,10), S4=sample(letters,10))
jaccard(x)

## End(Not run)
</code></pre>

<hr>
<h2 id='MSET'>Exact Test of Multi-Set Intersection</h2><span id='topic+MSET'></span>

<h3>Description</h3>

<p>Calculate FE and significance of intersection among multiple sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MSET(x,n,lower.tail=TRUE,log.p=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MSET_+3A_x">x</code></td>
<td>
<p>list; a collection of sets.</p>
</td></tr> 
<tr><td><code id="MSET_+3A_n">n</code></td>
<td>
<p>integer; background population size.</p>
</td></tr>
<tr><td><code id="MSET_+3A_lower.tail">lower.tail</code></td>
<td>
<p>logical; if TRUE, probability is <code>P[overlap &lt; m]</code>, otherwise, 
<code>P[overlap &gt;= m]</code>, where <code>m</code> is the number of elements overlap between all sets.</p>
</td></tr>
<tr><td><code id="MSET_+3A_log.p">log.p</code></td>
<td>
<p>logical; if TRUE, probability p is given as log(p).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements an efficient statistical test for multi-set intersections. The algorithm behind this function was described in Wang et al 2015.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>
<table>
<tr><td><code>intersects</code></td>
<td>
<p>a vector of intersect items.</p>
</td></tr>
<tr><td><code>FE</code></td>
<td>
<p>fold enrichment of the intersection.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>one-tail probability of observing equal to or larger than the number of intersect items.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+supertest">supertest</a></code>, <code><a href="#topic+cpsets">cpsets</a></code>, <code><a href="#topic+dpsets">dpsets</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#set up fake data
x=list(S1=letters[1:20], S2=letters[10:26], S3=sample(letters,10), S4=sample(letters,10))
MSET(x, 26, FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='msets'>Class to Contain Multi-Set Intersections</h2><span id='topic+msets'></span>

<h3>Description</h3>

<p>This object contains data regarding the intersections between multiple sets. This object is usually created by the <code>supertest</code> function.
</p>


<h3>Details</h3>

<p>Intersection combination is denoted by a <code>barcode</code> string of '0' and '1', where a value of '1' in the <code>i</code>th position of the string indicates that the intersection is involved with the <code>i</code>th set, 0 otherwise. E.g., string '000101' indicates that the intersection is an overlap between the 4th and 6th sets. Function <code><a href="#topic+deBarcode">deBarcode</a></code> can be used to decrypt the barcode.
Generic <code>summary</code> and <code>plot</code> functions can be applied to extract and visualize the results.
</p>


<h3>Value</h3>

<table>
<tr><td><code>x</code></td>
<td>
<p>a list of sets from input.</p>
</td></tr>
<tr><td><code>set.names</code></td>
<td>
<p>names of the sets. If the input sets do not have names, they will be automatically named as SetX where X is an integer from 1 to the total number of sets.</p>
</td></tr>
<tr><td><code>set.sizes</code></td>
<td>
<p>a vector of set sizes.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>background population size.</p>
</td></tr>
<tr><td><code>overlap.sizes</code></td>
<td>
<p>a named vector of intersection sizes. Each intersection component is named by a barcoded character string of '0' and '1'. See <code>Details</code> for barcode.</p>
</td></tr>
<tr><td><code>overlap.expected</code></td>
<td>
<p>a named vector of expected intersection sizes when item <code>n</code> is available.</p>
</td></tr>
<tr><td><code>P.value</code></td>
<td>
<p>a vector of p values for the intersections when item <code>n</code> is available.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+supertest">supertest</a></code>, <code><a href="#topic+summary.msets">summary.msets</a></code>, <code><a href="#topic+plot.msets">plot.msets</a></code>, <code><a href="#topic+deBarcode">deBarcode</a></code></p>

<hr>
<h2 id='plot.msets'>Draw Multi-Set Intersections</h2><span id='topic+plot.msets'></span>

<h3>Description</h3>

<p>This function draws intersections among multiple sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'msets'
plot(x, Layout=c('circular','landscape'), degree=NULL,
	keep.empty.intersections=TRUE,
	sort.by=c('set','size','degree','p-value'),
	min.intersection.size=0, max.intersection.size=Inf,
	ylim=NULL, log.scale=FALSE, yfrac=0.8, margin=NULL,
	color.scale.pos=c(0.85, 0.9), legend.pos=c(0.85,0.25),
	legend.col=2, legend.text.cex=1, color.scale.cex=1, 
	color.scale.title=expression(paste(-Log[10],'(',italic(P),')')),
	color.on='#2EFE64', color.off='#EEEEEE', 
	show.overlap.size=TRUE, show.fold.enrichment=FALSE, show.set.size=TRUE,
	overlap.size.cex=0.9, track.area.range=0.3, bar.area.range=0.2,
	new.gridPage=TRUE, minMinusLog10PValue=0,
	maxMinusLog10PValue=NULL, show.elements=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.msets_+3A_x">x</code></td>
<td>
<p>a <code>msets</code> object.</p>
</td></tr> 
<tr><td><code id="plot.msets_+3A_layout">Layout</code></td>
<td>
<p>layout for plotting.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_degree">degree</code></td>
<td>
<p>a vector of intersection degrees for plotting. E.g., when <code>degree=c(2:3)</code>, only those intersections involving two or three sets will be plotted. By default, <code>degree=NULL</code>, all possible intersections are plotted.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_keep.empty.intersections">keep.empty.intersections</code></td>
<td>
<p>logical; if <code>FALSE</code>, empty intersection(s) will be discarded to save plotting space.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_min.intersection.size">min.intersection.size</code></td>
<td>
<p>Minimum size of an intersection to be plotted.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_max.intersection.size">max.intersection.size</code></td>
<td>
<p>Maximum size of an intersection to be plotted.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_sort.by">sort.by</code></td>
<td>
<p>how to sort intersections. It can be either one of the key words &quot;<code>set</code>&quot;, &quot;<code>size</code>&quot;, &quot;<code>degree</code>&quot;, and &quot;<code>p-value</code>&quot;, or a vector of custom ordered set combination strings.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_ylim">ylim</code></td>
<td>
<p>the limits c(y1, y2) of plotting overlap size.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_log.scale">log.scale</code></td>
<td>
<p>logical; whether to plot with log transformed intersection sizes.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_yfrac">yfrac</code></td>
<td>
<p>numeric; the fraction (0 to 1) of canvas used for plotting bars. Only used for <code>landscape</code> Layout.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_margin">margin</code></td>
<td>
<p>numeric; a vector of 4 numeric values specifying the margins (bottom, left, top, &amp; right) in unit of &quot;lines&quot;. Default c(1,1,1,1)+0.1 for <code>circular</code> Layout and c(0.5,5,1.5,2)+0.1 for <code>landscape</code> Layout. Increase margin if the plot area is cropped.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_color.scale.pos">color.scale.pos</code></td>
<td>
<p>numeric; x and y coordinates (0 to 1) for packing the color scale guide. It could be a keyword &quot;<code>topright</code>&quot; or &quot;<code>topleft</code>&quot; in the <code>landscape</code> layout, and one of &quot;<code>topright</code>&quot;, &quot;<code>topleft</code>&quot;, &quot;<code>bottomright</code>&quot; and &quot;<code>bottomleft</code>&quot; in the <code>circular</code> layout.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_legend.pos">legend.pos</code></td>
<td>
<p>numeric; x and y coordinates (0 to 1) for packing the legend in the <code>circular</code> layout. It could be one of the keywords &quot;bottomright&quot;, &quot;bottomleft&quot;, &quot;topleft&quot; and &quot;topright&quot;.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_legend.col">legend.col</code></td>
<td>
<p>integer; number of columns of the legend in the <code>circular</code> layout.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_legend.text.cex">legend.text.cex</code></td>
<td>
<p>numeric; specifying the amount by which legend text should be magnified relative to the default.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_color.scale.cex">color.scale.cex</code></td>
<td>
<p>numeric; specifying the amount by which color scale text should be magnified relative to the default.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_color.scale.title">color.scale.title</code></td>
<td>
<p>character or expression; a title for the color scale guide.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_color.on">color.on</code></td>
<td>
<p>color code; specifying the color for set(s) which are &quot;<code>present</code>&quot; for an intersection. Can be a vector of colors. When <code>NULL</code>, a predefined list of colors will be used.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_color.off">color.off</code></td>
<td>
<p>color code; specifying the color for set(s) which are &quot;absent&quot; for an intersection.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_show.overlap.size">show.overlap.size</code></td>
<td>
<p>logical; whether to show overlap size on top of the bars. This will be set to <code>FALSE</code> if <code>show.fold.enrichment = TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_show.fold.enrichment">show.fold.enrichment</code></td>
<td>
<p>logical; whether to show fold enrichment if available rather than overlap size. This will impact <code>show.overlap.size</code>.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_show.set.size">show.set.size</code></td>
<td>
<p>color code; whether to show set size in the <code>landscape</code> layout.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_overlap.size.cex">overlap.size.cex</code></td>
<td>
<p>numeric; specifying the amount by which overlap size text should be magnified relative to the default.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_track.area.range">track.area.range</code></td>
<td>
<p>the magnitude of track area from origin in the <code>circular</code> layout.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_bar.area.range">bar.area.range</code></td>
<td>
<p>the magnitude of bar area from edge of the track area in the <code>circular</code> layout. The sum of <code>track.area.range</code> and <code>track.area.range</code> should not be larger than 0.5.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_new.gridpage">new.gridPage</code></td>
<td>
<p>logic; whether to start a new grid page. Set <code>FALSE</code> to allow for customized arrangement of the grid layout.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_minminuslog10pvalue">minMinusLog10PValue</code></td>
<td>
<p>numeric; minimum minus log10 P value for capping the scale of color map. Default 0.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_maxminuslog10pvalue">maxMinusLog10PValue</code></td>
<td>
<p>numeric; maximum minus log10 P value for capping the scale of color map. Default maximum from the data.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_show.elements">show.elements</code></td>
<td>
<p>logical; whether to show the intersection elements on top of the bars with the <code>landscape</code> layout. See <code>Details</code> for more control options <code>elements.*</code>.</p>
</td></tr>
<tr><td><code id="plot.msets_+3A_...">...</code></td>
<td>
<p>additional arguments for the plot function. See <code>Details</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot canvas has coordinates 0~1 for both x and y axes. Additional optional plot parameters include:
</p>

<ul>
<li> <p><code>ylab</code>, a chracter string of y axis label.
</p>
</li>
<li> <p><code>circle.radii</code>, radii size of the circles in <code>landscape</code> Layout. Default 0.5.
</p>
</li>
<li> <p><code>heatmapColor</code>, a vector of customized heat colors.
</p>
</li>
<li> <p><code>show.expected.overlap</code>, whether to show expcted overlap in <code>landscape</code> Layout. Default 'FALSE'.
</p>
</li>
<li> <p><code>expected.overlap.style</code>, one of c(&quot;hatchedBox&quot;,&quot;horizBar&quot;,&quot;box&quot;). Default 'hatchedBox'.
</p>
</li>
<li> <p><code>expected.overlap.lwd</code>, line width for expected.overlap &quot;horizBar&quot; and &quot;box&quot;. Default 2.
</p>
</li>
<li> <p><code>color.expected.overlap</code>, color for showing expcted overlap in hatched lines. Default 'grey'.
</p>
</li>
<li> <p><code>alpha.expected.overlap</code>, alpha channel for transparency for showing expcted overlap hatched lines. Default 1 (normalized to  the range 0 to 1).
</p>
</li>
<li> <p><code>cex</code>, scale of text font size.
</p>
</li>
<li> <p><code>cex.lab</code>, scale of axis label text font size.
</p>
</li>
<li> <p><code>show.track.id</code>, logic, whether to show the track id in the <code>circular</code> layout. Default <code>TRUE</code>.
</p>
</li>
<li> <p><code>phantom.tracks</code>, number of phantom tracks in the middle in the <code>circular</code> layout. Default 2.
</p>
</li>
<li> <p><code>gap.within.track</code>, ratio of gap width over block width on the same track. Default 0.1.
</p>
</li>
<li> <p><code>gap.between.track</code>, ratio of gap width over track width. Default 0.1.
</p>
</li>
<li> <p><code>bar.split</code>, a vector of two values specifying a continuous range that will be cropped in the y axis with the <code>landscape</code> layout.
</p>
</li>
<li> <p><code>elements.list</code>, a data.frame or matrix such as the one generated by the <code>summary</code> function from a <code>msets</code> object, with row names matching the barcodes of intersection combinations and at least one column named &quot;Elements&quot; listing the elements to be displayed (the elements should be concatenated by separator &quot;, &quot;).
</p>
</li>
<li> <p><code>elements.cex</code>, numeric; specifying the amount by which intersection element text should be magnified. Default 0.9.
</p>
</li>
<li> <p><code>elements.rot</code>, numeric; the angle to rotate the text of intersection elements. Default 45.
</p>
</li>
<li> <p><code>elements.col</code>, colour for intersection element text. Default black.
</p>
</li>
<li> <p><code>elements.maximum</code>, maximum number of elements to show.
</p>
</li>
<li> <p><code>intersection.size.rotate</code>, logic, whether to rotate the text of intersection size.
</p>
</li>
<li> <p><code>flip.vertical</code>, logic, whether to flip the bars to downwards in <code>landscape</code> Layout. Default 'FALSE'.
</p>
</li>
<li> <p><code>title</code>, figure title. Default NULL.
</p>
</li>
<li> <p><code>cex.title</code>, scale of title text font size. Default 1.
</p>
</li></ul>



<h3>Value</h3>

<p>No return.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+msets">msets</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#set up fake data
x=list(S1=letters[1:20], S2=letters[10:26], S3=sample(letters,10), S4=sample(letters,10))
obj=supertest(x,n=26)
plot(obj)

## End(Not run)
</code></pre>

<hr>
<h2 id='summary.msets'>Summarize an msets Object</h2><span id='topic+summary.msets'></span>

<h3>Description</h3>

<p>This function outputs summary statistics of a msets object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'msets'
summary(object, degree=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.msets_+3A_object">object</code></td>
<td>
<p>a <code>msets</code> object.</p>
</td></tr> 
<tr><td><code id="summary.msets_+3A_degree">degree</code></td>
<td>
<p>a vector of intersection degrees to pull out.</p>
</td></tr>
<tr><td><code id="summary.msets_+3A_...">...</code></td>
<td>
<p>additional arguments (not implemented).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list:
</p>
<table>
<tr><td><code>Barcode</code></td>
<td>
<p>a vector of 0/1 character strings, representing the set composition of each intersection.</p>
</td></tr>
<tr><td><code>otab</code></td>
<td>
<p>a vector of observed intersection size between any combination of sets.</p>
</td></tr>
<tr><td><code>etab</code></td>
<td>
<p>a vector of expected intersection size between any combination of sets if background population size is specified.</p>
</td></tr>
<tr><td><code>set.names</code></td>
<td>
<p>set names.</p>
</td></tr>
<tr><td><code>set.sizes</code></td>
<td>
<p>set sizes.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>background population size.</p>
</td></tr>
<tr><td><code>P.value</code></td>
<td>
<p>upper tail p value for each intersection if background population size n is specified.</p>
</td></tr>
<tr><td><code>Table</code></td>
<td>
<p>a data.frame containing degree, otab, etab, fold change, p value and the overlap elements.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+msets">msets</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#set up fake data
x=list(S1=letters[1:20], S2=letters[10:26], S3=sample(letters,10), S4=sample(letters,10))
obj=supertest(x,n=26)
summary(obj)

## End(Not run)
</code></pre>

<hr>
<h2 id='SuperExactTest'>SuperExactTest Package</h2><span id='topic+SuperExactTest'></span>

<h3>Description</h3>

<p>Efficient Test and Visualization of Multi-set Intersections
</p>


<h3>Details</h3>

<p>The main functions that most users may need from this package are <code><a href="#topic+supertest">supertest</a></code> and <code><a href="#topic+MSET">MSET</a></code>. For a brief introduction of using this package, please see <code>vignette("set_html")</code>.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+supertest">supertest</a></code>, <code><a href="#topic+MSET">MSET</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#See a brieft instroduction of using this package
vignette("set_html")

## End(Not run)
</code></pre>

<hr>
<h2 id='supertest'>Calculate Intersections Among Multiple Sets and Perform Statistical Tests</h2><span id='topic+supertest'></span><span id='topic+supertest+2Clist-method'></span>

<h3>Description</h3>

<p>This function calculates intersection sizes among multiple sets and performs statistical tests of the intersections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>supertest(x, n=NULL, degree=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="supertest_+3A_x">x</code></td>
<td>
<p>list; a collection of sets.</p>
</td></tr>
<tr><td><code id="supertest_+3A_n">n</code></td>
<td>
<p>integer, background population size. Required for computing the statistical significance of intersections.</p>
</td></tr>
<tr><td><code id="supertest_+3A_degree">degree</code></td>
<td>
<p>a vector of intersection degrees for overlap analysis. E.g., when <code>degree=c(2:3)</code>, only those intersections involving two or three sets will be computed. By default, <code>degree=NULL</code>, all possible intersections are computed.</p>
</td></tr>
<tr><td><code id="supertest_+3A_...">...</code></td>
<td>
<p>additional arguments (not implemented).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates intersection sizes between multiple sets and, if background population size <code>n</code> is specified, performs statistical tests of the intersections.
For a brief introduction of using this package, please see <code>vignette("set_html")</code>.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+msets">msets</a></code>.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;minghui.wang@mssm.edu&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Minghui Wang, Yongzhong Zhao, and Bin Zhang (2015). Efficient Test and Visualization of Multi-Set Intersections. <em>Scientific Reports</em> 5: 16923.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+msets">msets</a></code>, <code><a href="#topic+MSET">MSET</a></code>, <code><a href="#topic+Cancer">Cancer</a></code>, <code><a href="#topic+cpsets">cpsets</a></code>, <code><a href="#topic+dpsets">dpsets</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Analyze the cancer gene sets
data(Cancer)
Result=supertest(Cancer, n=20687)
summary(Result)
plot(Result,degree=2:7,sort.by='size')

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
