<!DOCTYPE html><html><head><title>Help for package survminer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {survminer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_ggsurvplot'><p>Add Components to a ggsurvplot</p></a></li>
<li><a href='#arrange_ggsurvplots'><p>Arranging Multiple ggsurvplots</p></a></li>
<li><a href='#BMT'><p>Bone Marrow Transplant</p></a></li>
<li><a href='#BRCAOV.survInfo'><p>Breast and Ovarian Cancers Survival Information</p></a></li>
<li><a href='#ggadjustedcurves'><p>Adjusted Survival Curves for Cox Proportional Hazards Model</p></a></li>
<li><a href='#ggcompetingrisks'><p>Cumulative Incidence Curves for Competing Risks</p></a></li>
<li><a href='#ggcoxdiagnostics'><p>Diagnostic Plots for Cox Proportional Hazards Model with ggplot2</p></a></li>
<li><a href='#ggcoxfunctional'><p>Functional Form of Continuous Variable in Cox Proportional Hazards Model</p></a></li>
<li><a href='#ggcoxzph'><p>Graphical Test of Proportional Hazards with ggplot2</p></a></li>
<li><a href='#ggflexsurvplot'><p>Ggplots of Fitted Flexible Survival Models</p></a></li>
<li><a href='#ggforest'><p>Forest Plot for Cox Proportional Hazards Model</p></a></li>
<li><a href='#ggrisktable'><p>Plot Survival Tables</p></a></li>
<li><a href='#ggsurvevents'><p>Distribution of Events' Times</p></a></li>
<li><a href='#ggsurvplot'><p>Drawing Survival Curves Using ggplot2</p></a></li>
<li><a href='#ggsurvplot_add_all'><p>Add Survival Curves of Pooled Patients onto the Main Plot</p></a></li>
<li><a href='#ggsurvplot_arguments'><p>ggsurvplot Argument Descriptions</p></a></li>
<li><a href='#ggsurvplot_combine'><p>Combine a List of Survfit Objects on the Same Plot</p></a></li>
<li><a href='#ggsurvplot_df'><p>Plot Survival Curves from Survival Summary Data Frame</p></a></li>
<li><a href='#ggsurvplot_facet'><p>Facet Survival Curves into Multiple Panels</p></a></li>
<li><a href='#ggsurvplot_group_by'><p>Survival Curves of Grouped Data sets</p></a></li>
<li><a href='#ggsurvplot_list'><p>Plot a List of Survfit Objects</p></a></li>
<li><a href='#myeloma'><p>Multiple Myeloma Data</p></a></li>
<li><a href='#pairwise_survdiff'><p>Multiple Comparisons of Survival Curves</p></a></li>
<li><a href='#surv_cutpoint'><p>Determine the Optimal Cutpoint for Continuous Variables</p></a></li>
<li><a href='#surv_fit'><p>Create Survival Curves</p></a></li>
<li><a href='#surv_group_by'><p>Create a Grouped Dataset for Survival Analysis</p></a></li>
<li><a href='#surv_median'><p>Median of Survival Curves</p></a></li>
<li><a href='#surv_pvalue'><p>Compute P-value Comparing Survival Curves</p></a></li>
<li><a href='#surv_summary'><p>Nice Summary of a Survival Curve</p></a></li>
<li><a href='#theme_survminer'><p>Theme for Survminer Plots</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Drawing Survival Curves using 'ggplot2'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-03-09</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains the function 'ggsurvplot()' for drawing easily beautiful
    and 'ready-to-publish' survival curves with the 'number at risk' table
    and 'censoring count plot'. Other functions are also available to plot 
    adjusted curves for &lsquo;Cox' model and to visually examine &rsquo;Cox' model assumptions.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>ggplot2, ggpubr(&ge; 0.1.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>grid, gridExtra (&ge; 2.0), magrittr, maxstat, methods, scales,
survival, stats, broom, dplyr, tidyr, survMisc, purrr, tibble,
rlang, ggtext (&ge; 0.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, flexsurv, cmprsk, markdown, testthat, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://rpkgs.datanovia.com/survminer/index.html">https://rpkgs.datanovia.com/survminer/index.html</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kassambara/survminer/issues">https://github.com/kassambara/survminer/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Collate:</td>
<td>'BMT.R' 'BRCAOV.survInfo.R' 'add_ggsurvplot.R' 'utilities.R'
'surv_summary.R' 'ggsurvtable.R' 'surv_pvalue.R'
'ggsurvplot_df.R' 'ggsurvplot_core.R' 'ggsurvplot_add_all.R'
'ggsurvplot_list.R' 'ggsurvplot_group_by.R' 'ggsurvplot.R'
'arrange_ggsurvplots.R' 'ggadjustedcurves.R'
'ggcompetingrisks.R' 'ggcoxdiagnostics.R' 'ggcoxfunctional.R'
'ggcoxzph.R' 'ggflexsurvplot.R' 'ggforest.R' 'ggsurvevents.R'
'ggsurvplot_combine.R' 'ggsurvplot_facet.R' 'ggsurvtheme.R'
'ggurvplot_arguments.R' 'myeloma.R' 'pairwise_survdiff.R'
'surv_cutpoint.R' 'surv_group_by.R' 'surv_fit.R'
'surv_median.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-03-09 08:37:37 UTC; kassambara</td>
</tr>
<tr>
<td>Author:</td>
<td>Alboukadel Kassambara [aut, cre],
  Marcin Kosinski [aut],
  Przemyslaw Biecek [aut],
  Scheipl Fabian [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alboukadel Kassambara &lt;alboukadel.kassambara@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-03-09 09:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_ggsurvplot'>Add Components to a ggsurvplot</h2><span id='topic+add_ggsurvplot'></span><span id='topic++2B.ggsurv'></span><span id='topic++25+2B+2B+25'></span>

<h3>Description</h3>

<p>Allows to add ggplot components - theme(), labs(), ... - to an
object of class ggsurv, which is a list of ggplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggsurv'
e1 + e2

e1 %++% e2
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_ggsurvplot_+3A_e1">e1</code></td>
<td>
<p>an object of class ggsurv.</p>
</td></tr>
<tr><td><code id="add_ggsurvplot_+3A_e2">e2</code></td>
<td>
<p>a plot component such as theme and labs.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+theme_survminer">theme_survminer</a></code> and <code><a href="#topic+ggsurvplot">ggsurvplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Fit survival curves
require("survival")
fit&lt;- survfit(Surv(time, status) ~ sex, data = lung)

# Basic survival curves
p &lt;- ggsurvplot(fit, data = lung, risk.table = TRUE,
   main = "Survival curve",
   submain = "Based on Kaplan-Meier estimates",
   caption = "created with survminer"
   )
p

# Customizing the plots
p + theme_survminer(
     font.main = c(16, "bold", "darkblue"),
     font.submain = c(15, "bold.italic", "purple"),
     font.caption = c(14, "plain", "orange"),
     font.x = c(14, "bold.italic", "red"),
     font.y = c(14, "bold.italic", "darkred"),
     font.tickslab = c(12, "plain", "darkgreen")
)
</code></pre>

<hr>
<h2 id='arrange_ggsurvplots'>Arranging Multiple ggsurvplots</h2><span id='topic+arrange_ggsurvplots'></span>

<h3>Description</h3>

<p>Arranging multiple ggsurvplots on the same page.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arrange_ggsurvplots(
  x,
  print = TRUE,
  title = NA,
  ncol = 2,
  nrow = 1,
  surv.plot.height = NULL,
  risk.table.height = NULL,
  ncensor.plot.height = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="arrange_ggsurvplots_+3A_x">x</code></td>
<td>
<p>a list of ggsurvplots.</p>
</td></tr>
<tr><td><code id="arrange_ggsurvplots_+3A_print">print</code></td>
<td>
<p>logical value. If TRUE, the arranged plots are displayed.</p>
</td></tr>
<tr><td><code id="arrange_ggsurvplots_+3A_title">title</code></td>
<td>
<p>character vector specifying page title. Default is NA.</p>
</td></tr>
<tr><td><code id="arrange_ggsurvplots_+3A_ncol">ncol</code>, <code id="arrange_ggsurvplots_+3A_nrow">nrow</code></td>
<td>
<p>the number of columns and rows, respectively.</p>
</td></tr>
<tr><td><code id="arrange_ggsurvplots_+3A_surv.plot.height">surv.plot.height</code></td>
<td>
<p>the height of the survival plot on the grid. Default
is 0.75. Ignored when risk.table = FALSE. <code>1-risk.table.height -
ncensor.plot.height</code> when <code>risk.table = TRUE</code> and <code>ncensor.plot =
TRUE</code></p>
</td></tr>
<tr><td><code id="arrange_ggsurvplots_+3A_risk.table.height">risk.table.height</code></td>
<td>
<p>the height of the risk table on the grid. Increase
the value when you have many strata. Default is 0.25. Ignored when
risk.table = FALSE.</p>
</td></tr>
<tr><td><code id="arrange_ggsurvplots_+3A_ncensor.plot.height">ncensor.plot.height</code></td>
<td>
<p>The height of the censor plot. Used when
<code>ncensor.plot = TRUE</code>.</p>
</td></tr>
<tr><td><code id="arrange_ggsurvplots_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns an invisible object of class arrangelist (see
<a href="gridExtra.html#topic+marrangeGrob">marrangeGrob</a>), which can be saved into a pdf file using
the function <a href="ggplot2.html#topic+ggsave">ggsave</a>.
</p>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara, <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Fit survival curves
require("survival")
fit&lt;- survfit(Surv(time, status) ~ sex, data = lung)

# List of ggsurvplots
require("survminer")
splots &lt;- list()
splots[[1]] &lt;- ggsurvplot(fit, data = lung, risk.table = TRUE, ggtheme = theme_minimal())
splots[[2]] &lt;- ggsurvplot(fit, data = lung, risk.table = TRUE, ggtheme = theme_grey())

# Arrange multiple ggsurvplots and print the output
arrange_ggsurvplots(splots, print = TRUE,
  ncol = 2, nrow = 1, risk.table.height = 0.4)

## Not run: 
# Arrange and save into pdf file
res &lt;- arrange_ggsurvplots(splots, print = FALSE)
ggsave("myfile.pdf", res)

## End(Not run)


</code></pre>

<hr>
<h2 id='BMT'>Bone Marrow Transplant</h2><span id='topic+BMT'></span>

<h3>Description</h3>

<p>Bone marrow transplant data from L Scrucca et aL., Bone Marrow
Transplantation (2007). Data from 35 patients with acute leukaemia who
underwent HSCT. Used for competing risk analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("BMT")
</code></pre>


<h3>Format</h3>

<p>A data frame with 35 rows and 3 columns.
</p>
<pre>
- dis: disease; 0 = ALL; 1 = AML
- ftime: follow-up time
- status: 0 = censored (survival); 1 = Transplant-related mortality; 2 = relapse
</pre>


<h3>References</h3>

<p>Scrucca L, Santucci A, Aversa F. Competing risk analysis using R:
an easy guide for clinicians. Bone Marrow Transplant. 2007 Aug;40(4):381-7.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BMT)
if(require("cmprsk")){

# Data preparaion
#+++++++++++++++++++++
# Label diseases
BMT$dis &lt;- factor(BMT$dis, levels = c(0,1),
   labels = c("ALL", "AML"))
# Label status
BMT$status &lt;- factor(BMT$status, levels = c(0,1,2),
  labels = c("Censored","Mortality","Relapse"))

# Cumulative Incidence Function
# ++++++++++++++++++++++++++
fit &lt;- cmprsk::cuminc(
  ftime = BMT$ftime,      # Failure time variable
  fstatus = BMT$status,   # Codes for different causes of failure
  group = BMT$dis         # Estimates will calculated within groups
 )

# Visualize
# +++++++++++++++++++++++
ggcompetingrisks(fit)
ggcompetingrisks(fit, multiple_panels = FALSE,
  legend = "right")

}

</code></pre>

<hr>
<h2 id='BRCAOV.survInfo'>Breast and Ovarian Cancers Survival Information</h2><span id='topic+BRCAOV.survInfo'></span>

<h3>Description</h3>

<p>Breat and Ovarian cancers survival information from the
RTCGA.clinical R/Bioconductor package.http://rtcga.github.io/RTCGA/.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("BRCAOV.survInfo")
</code></pre>


<h3>Format</h3>

<p>A data frame with 1674 rows and 4 columns.
</p>
<pre>
 - times: follow-up time;
 - bcr_patient_barcode: Patient bar code;
 - patient.vital_status = survival status. 0 = alive, 1 = dead;
 - admin.disease_code: disease code. brca = breast cancer, ov = ovarian
 cancer.
 </pre>


<h3>Source</h3>

<p>From the RTCGA.clinical R/Bioconductor package. The data is generated as follow:
</p>
<pre>
# Installing RTCGA.clinical
source("https://bioconductor.org/biocLite.R")
biocLite("RTCGA.clinical")

# Generating the BRCAOV survival information
library(RTCGA.clinical)
survivalTCGA(BRCA.clinical, OV.clinical,
extract.cols = "admin.disease_code") -&gt; BRCAOV.survInfo
</pre>


<h3>Examples</h3>

<pre><code class='language-R'>data(BRCAOV.survInfo)
library(survival)
fit &lt;- survfit(Surv(times, patient.vital_status) ~ admin.disease_code,
           data = BRCAOV.survInfo)
ggsurvplot(fit, data = BRCAOV.survInfo, risk.table = TRUE)
</code></pre>

<hr>
<h2 id='ggadjustedcurves'>Adjusted Survival Curves for Cox Proportional Hazards Model</h2><span id='topic+ggadjustedcurves'></span><span id='topic+surv_adjustedcurves'></span>

<h3>Description</h3>

<p>The function <code>surv_adjustedcurves()</code> calculates while the function <code>ggadjustedcurves()</code> plots adjusted survival curves for the <code>coxph</code> model.
The main idea behind this function is to present expected survival curves calculated based on Cox model separately for subpopulations. The very detailed description and interesting discussion of adjusted curves is presented in 'Adjusted Survival Curves' by Terry Therneau, Cynthia Crowson, Elizabeth Atkinson (2015) <code>https://cran.r-project.org/web/packages/survival/vignettes/adjcurve.pdf</code>.
Many approaches are discussed in this article. Currently four approaches (two unbalanced, one conditional and one marginal) are implemented in the <code>ggadjustedcurves()</code> function. See the section Details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggadjustedcurves(
  fit,
  variable = NULL,
  data = NULL,
  reference = NULL,
  method = "conditional",
  fun = NULL,
  palette = "hue",
  ylab = "Survival rate",
  size = 1,
  ggtheme = theme_survminer(),
  ...
)

surv_adjustedcurves(
  fit,
  variable = NULL,
  data = NULL,
  reference = NULL,
  method = "conditional",
  size = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggadjustedcurves_+3A_fit">fit</code></td>
<td>
<p>an object of class <a href="survival.html#topic+coxph.object">coxph.object</a> - created with <a href="survival.html#topic+coxph">coxph</a> function.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_variable">variable</code></td>
<td>
<p>a character, name of the grouping variable to be plotted. If not supplied then it will be extracted from the model formula from the <code>strata()</code> component. If there is no <code>strata()</code> component then only a single curve will be plotted - average for the thole population.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_data">data</code></td>
<td>
<p>a dataset for predictions. If not supplied then data will be extracted from the <code>fit</code> object.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_reference">reference</code></td>
<td>
<p>a dataset for reference population, to which dependent variables should be balanced. If not specified, then the <code>data</code> will be used instead. Note that the <code>reference</code> dataset should contain all variables used in <code>fit</code> object.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_method">method</code></td>
<td>
<p>a character, describes how the expected survival curves shall be calculated. Possible options:
'single' (average for population), 'average' (averages for subpopulations), 'marginal', 'conditional' (averages for subpopulations after rebalancing). See the Details section  for further information.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_fun">fun</code></td>
<td>
<p>an arbitrary function defining a transformation of the survival
curve.  Often used transformations can be specified with a character
argument: &quot;event&quot; plots cumulative events (f(y) = 1-y), &quot;cumhaz&quot; plots the
cumulative hazard function (f(y) = -log(y)), and &quot;pct&quot; for survival
probability in percentage.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used. Allowed values include &quot;hue&quot; for
the default hue color scale; &quot;grey&quot; for grey color palettes; brewer palettes
e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;,
&quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;.
See details section for more information.  Can be also a numeric vector of
length(groups); in this case a basic color palette is created using the
function <a href="grDevices.html#topic+palette">palette</a>.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_ylab">ylab</code></td>
<td>
<p>a label for oy axis.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_size">size</code></td>
<td>
<p>the curve size.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Allowed values include ggplot2 official themes: see <code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggadjustedcurves_+3A_...">...</code></td>
<td>
<p>further arguments passed to the function <code><a href="ggpubr.html#topic+ggpar">ggpar</a></code> for customizing the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently four approaches are implemented in the <code>ggadjustedcurves()</code> function.
</p>
<p>For <code>method = "single"</code> a single survival curve is calculated and plotted. The curve presents an expected survival calculated for population <code>data</code> calculated based on the Cox model <code>fit</code>.
</p>
<p>For <code>method = "average"</code> a separate survival curve is plotted for each level of a variable listed as <code>variable</code>. If this argument is not specified, then it will be extracted from the <code>strata</code> component of <code>fit</code> argument.  Each curve presents an expected survival calculated for subpopulation from <code>data</code> based on a Cox model <code>fit</code>. Note that in this method subpopulations are NOT balanced.
</p>
<p>For <code>method = "marginal"</code> a survival curve is plotted for each level of a grouping variable selected by <code>variable</code> argument. If this argument is not specified, then it will be extracted from the <code>strata</code> component of <code>fit</code> object.  Subpopulations are balanced with respect to variables in the <code>fit</code> formula to keep distributions similar to these in the <code>reference</code> population. If no reference population is specified, then the whole <code>data</code> is used as a reference population instead. The balancing is performed in a following way: (1) for each subpopulation a logistic regression model is created to model the odds of being in the subpopulation against the reference population given the other variables listed in a <code>fit</code> object, (2) reverse probabilities of belonging to a specified subpopulation are used as weights in the Cox model, (3) the Cox model is refitted with weights taken into account, (4) expected survival curves are calculated for each subpopulation based on a refitted weighted model.
</p>
<p>For <code>method = "conditional"</code> a separate survival curve is plotted for each level of a grouping variable selected by <code>variable</code> argument. If this argument is not specified, then it will be extracted from the <code>strata</code> component of <code>fit</code> object.  Subpopulations are balanced in a following way: (1) the data is replicated as many times as many subpopulations are considered (say k), (2) for each row in original data a set of k copies are created and for every copy a different value of a grouping variable is assigned, this will create a new dataset balanced in terms of grouping variables, (3) expected survival is calculated for each subpopulation based on the new artificial dataset. Here the model <code>fit</code> is not refitted.
</p>
<p>Note that <code>surv_adjustedcurves</code> function calculates survival curves and based on this function one can calculate median survival.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>gg</code>.
</p>


<h3>Author(s)</h3>

<p>Przemyslaw Biecek, <a href="mailto:przemyslaw.biecek@gmail.com">przemyslaw.biecek@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)
fit2 &lt;- coxph( Surv(stop, event) ~ size, data = bladder )
# single curve
ggadjustedcurves(fit2, data = bladder)
curve &lt;- surv_adjustedcurves(fit2, data = bladder)

fit2 &lt;- coxph( Surv(stop, event) ~ size + strata(rx), data = bladder )
# average in groups
ggadjustedcurves(fit2, data = bladder, method = "average", variable = "rx")
curve &lt;- surv_adjustedcurves(fit2, data = bladder, method = "average", variable = "rx")

# conditional balancing in groups
ggadjustedcurves(fit2, data = bladder, method = "marginal", variable = "rx")
curve &lt;- surv_adjustedcurves(fit2, data = bladder, method = "marginal", variable = "rx")

# selected reference population
ggadjustedcurves(fit2, data = bladder, method = "marginal", variable = "rx",
    reference = bladder[bladder$rx == "1",])

# conditional balancing in groups
ggadjustedcurves(fit2, data = bladder, method = "conditional", variable = "rx")
curve &lt;- surv_adjustedcurves(fit2, data = bladder, method = "conditional", variable = "rx")

## Not run: 
# this will take some time
fdata &lt;- flchain[flchain$futime &gt;=7,]
fdata$age2 &lt;- cut(fdata$age, c(0,54, 59,64, 69,74,79, 89, 110),
                  labels = c(paste(c(50,55,60,65,70,75,80),
                                   c(54,59,64,69,74,79,89), sep='-'), "90+"))
fdata$group &lt;- factor(1+ 1*(fdata$flc.grp &gt;7) + 1*(fdata$flc.grp &gt;9),
                      levels=1:3,
                      labels=c("FLC &lt; 3.38", "3.38 - 4.71", "FLC &gt; 4.71"))
# single curve
fit &lt;- coxph( Surv(futime, death) ~ age*sex, data = fdata)
ggadjustedcurves(fit, data = fdata, method = "single")

# average in groups
fit &lt;- coxph( Surv(futime, death) ~ age*sex + strata(group), data = fdata)
ggadjustedcurves(fit, data = fdata, method = "average")

# conditional balancing in groups
ggadjustedcurves(fit, data = fdata, method = "conditional")

# marginal balancing in groups
ggadjustedcurves(fit, data = fdata, method = "marginal", reference = fdata)

## End(Not run)

</code></pre>

<hr>
<h2 id='ggcompetingrisks'>Cumulative Incidence Curves for Competing Risks</h2><span id='topic+ggcompetingrisks'></span>

<h3>Description</h3>

<p>This function plots Cumulative Incidence Curves. For <code>cuminc</code> objects it's a <code>ggplot2</code> version of <code>plot.cuminc</code>.
For <code>survfitms</code> objects a different geometry is used, as suggested by <code>@teigentler</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggcompetingrisks(
  fit,
  gnames = NULL,
  gsep = " ",
  multiple_panels = TRUE,
  ggtheme = theme_survminer(),
  coef = 1.96,
  conf.int = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggcompetingrisks_+3A_fit">fit</code></td>
<td>
<p>an object of a class <code>cmprsk::cuminc</code> - created with <code>cmprsk::cuminc</code> function or <code>survfitms</code> created with <a href="survival.html#topic+survfit">survfit</a> function.</p>
</td></tr>
<tr><td><code id="ggcompetingrisks_+3A_gnames">gnames</code></td>
<td>
<p>a vector with group names. If not supplied then will be extracted from <code>fit</code> object (<code>cuminc</code> only).</p>
</td></tr>
<tr><td><code id="ggcompetingrisks_+3A_gsep">gsep</code></td>
<td>
<p>a separator that extracts group names and event names from <code>gnames</code> object (<code>cuminc</code> only).</p>
</td></tr>
<tr><td><code id="ggcompetingrisks_+3A_multiple_panels">multiple_panels</code></td>
<td>
<p>if <code>TRUE</code> then groups will be plotted in different panels (<code>cuminc</code> only).</p>
</td></tr>
<tr><td><code id="ggcompetingrisks_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, <code>ggplot2</code> theme name. Default value is <a href="#topic+theme_survminer">theme_survminer</a>.
Allowed values include ggplot2 official themes: see <code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggcompetingrisks_+3A_coef">coef</code></td>
<td>
<p>see <code>conf.int</code>, scaling actor for the ribbon. The default value is 1.96.</p>
</td></tr>
<tr><td><code id="ggcompetingrisks_+3A_conf.int">conf.int</code></td>
<td>
<p>if <code>TRUE</code> then additional layer (<code>geom_ribbon</code>) is added around the point estimate. The ribon is plotted with boundries +- <code>coef</code>*standard deviation.</p>
</td></tr>
<tr><td><code id="ggcompetingrisks_+3A_...">...</code></td>
<td>
<p>further arguments passed to the function <code><a href="ggpubr.html#topic+ggpar">ggpar</a></code> for customizing the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>gg</code>.
</p>


<h3>Author(s)</h3>

<p>Przemyslaw Biecek, <a href="mailto:przemyslaw.biecek@gmail.com">przemyslaw.biecek@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(require("cmprsk")){

set.seed(2)
ss &lt;- rexp(100)
gg &lt;- factor(sample(1:3,100,replace=TRUE),1:3,c('BRCA','LUNG','OV'))
cc &lt;- factor(sample(0:2,100,replace=TRUE),0:2,c('no event', 'death', 'progression'))
strt &lt;- sample(1:2,100,replace=TRUE)

# handles cuminc objects
print(fit &lt;- cmprsk::cuminc(ss,cc,gg,strt))
ggcompetingrisks(fit)
ggcompetingrisks(fit, multiple_panels = FALSE)
ggcompetingrisks(fit, conf.int = TRUE)
ggcompetingrisks(fit, multiple_panels = FALSE, conf.int = TRUE)

# handles survfitms objects
library(survival)
df &lt;- data.frame(time = ss, group = gg, status = cc, strt)
fit2 &lt;- survfit(Surv(time, status, type="mstate") ~ 1, data=df)
ggcompetingrisks(fit2)
fit3 &lt;- survfit(Surv(time, status, type="mstate") ~ group, data=df)
ggcompetingrisks(fit3)
}

  library(ggsci)
  library(cowplot)
  ggcompetingrisks(fit3) + theme_cowplot() + scale_fill_jco()

## End(Not run)
</code></pre>

<hr>
<h2 id='ggcoxdiagnostics'>Diagnostic Plots for Cox Proportional Hazards Model with ggplot2</h2><span id='topic+ggcoxdiagnostics'></span>

<h3>Description</h3>

<p>Displays diagnostics graphs presenting goodness of Cox Proportional Hazards Model fit, that
can be calculated with <a href="survival.html#topic+coxph">coxph</a> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggcoxdiagnostics(
  fit,
  type = c("martingale", "deviance", "score", "schoenfeld", "dfbeta", "dfbetas",
    "scaledsch", "partial"),
  ...,
  linear.predictions = type %in% c("martingale", "deviance"),
  ox.scale = ifelse(linear.predictions, "linear.predictions", "observation.id"),
  hline = TRUE,
  sline = TRUE,
  sline.se = TRUE,
  hline.col = "red",
  hline.size = 1,
  hline.alpha = 1,
  hline.yintercept = 0,
  hline.lty = "dashed",
  sline.col = "blue",
  sline.size = 1,
  sline.alpha = 0.3,
  sline.lty = "dashed",
  point.col = "black",
  point.size = 1,
  point.shape = 19,
  point.alpha = 1,
  title = NULL,
  subtitle = NULL,
  caption = NULL,
  ggtheme = ggplot2::theme_bw()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggcoxdiagnostics_+3A_fit">fit</code></td>
<td>
<p>an object of class <a href="survival.html#topic+coxph.object">coxph.object</a> - created with <a href="survival.html#topic+coxph">coxph</a> function.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_type">type</code></td>
<td>
<p>the type of residuals to present on Y axis of a diagnostic plot.
The same as in <a href="survival.html#topic+residuals.coxph">residuals.coxph</a>: character string indicating the type of
residual desired. Possible values are <code>"martingale", "deviance", "score", "schoenfeld", "dfbeta", "dfbetas"</code>
and <code>"scaledsch"</code>. Only enough of the string to
determine a unique match is required.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="survival.html#topic+residuals.coxph">residuals.coxph</a></code> or
to the function <code><a href="ggpubr.html#topic+ggpar">ggpar</a></code> for customizing the plot.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_linear.predictions">linear.predictions</code></td>
<td>
<p>(deprecated, see <code>ox.scale</code>) a logical value indicating whether to show linear
predictions for observations (<code>TRUE</code>) or just indexed of observations
(<code>FALSE</code>) on X axis.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_ox.scale">ox.scale</code></td>
<td>
<p>one value from <code>c("linear.predictions", "observation.id", "time")</code>.
It defines what will be presented on OX scale. Possible values: y hat for <code>"linear.predictions"</code>,
Id of an observation for <code>"observation.id"</code> or Time for <code>"time"</code>.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_hline">hline</code></td>
<td>
<p>a logical - should the horizontal line be added to highlight the <code>Y=0</code> level.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_sline">sline</code>, <code id="ggcoxdiagnostics_+3A_sline.se">sline.se</code></td>
<td>
<p>a logical - should the smooth line be added to highlight the local average for residuals.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_hline.col">hline.col</code>, <code id="ggcoxdiagnostics_+3A_hline.size">hline.size</code>, <code id="ggcoxdiagnostics_+3A_hline.lty">hline.lty</code>, <code id="ggcoxdiagnostics_+3A_hline.alpha">hline.alpha</code>, <code id="ggcoxdiagnostics_+3A_hline.yintercept">hline.yintercept</code></td>
<td>
<p>color, size, linetype, visibility and Y-axis coordinate to be used for <a href="ggplot2.html#topic+geom_hline">geom_hline</a>.
Used only when <code>hline = TRUE</code>.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_sline.col">sline.col</code>, <code id="ggcoxdiagnostics_+3A_sline.size">sline.size</code>, <code id="ggcoxdiagnostics_+3A_sline.lty">sline.lty</code>, <code id="ggcoxdiagnostics_+3A_sline.alpha">sline.alpha</code></td>
<td>
<p>color, size, linetype and visibility to be used for <a href="ggplot2.html#topic+geom_smooth">geom_smooth</a>.
Used only when <code>sline = TRUE</code>.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_point.col">point.col</code>, <code id="ggcoxdiagnostics_+3A_point.size">point.size</code>, <code id="ggcoxdiagnostics_+3A_point.shape">point.shape</code>, <code id="ggcoxdiagnostics_+3A_point.alpha">point.alpha</code></td>
<td>
<p>color, size, shape and visibility to be used for points.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_title">title</code>, <code id="ggcoxdiagnostics_+3A_subtitle">subtitle</code>, <code id="ggcoxdiagnostics_+3A_caption">caption</code></td>
<td>
<p>main title, subtitle and caption.</p>
</td></tr>
<tr><td><code id="ggcoxdiagnostics_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is ggplot2::theme_bw().
Allowed values include ggplot2 official themes: see <code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>ggplot</code>.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>ggcoxdiagnostics</code>: Diagnostic Plots for Cox Proportional Hazards Model with <span class="pkg">ggplot2</span>
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Marcin Kosinski , <a href="mailto:m.p.kosinski@gmail.com">m.p.kosinski@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)
coxph.fit2 &lt;- coxph(Surv(futime, fustat) ~ age + ecog.ps, data=ovarian)
ggcoxdiagnostics(coxph.fit2, type = "deviance")

ggcoxdiagnostics(coxph.fit2, type = "schoenfeld", title = "Diagnostic plot")
ggcoxdiagnostics(coxph.fit2, type = "deviance", ox.scale = "time")
ggcoxdiagnostics(coxph.fit2, type = "schoenfeld", ox.scale = "time",
                 title = "Diagnostic plot", subtitle = "Data comes from survey XYZ",
                 font.subtitle = 9)
ggcoxdiagnostics(coxph.fit2, type = "deviance", ox.scale = "linear.predictions",
                 caption = "Code is available here - link", font.caption = 10)
ggcoxdiagnostics(coxph.fit2, type = "schoenfeld", ox.scale = "observation.id")
ggcoxdiagnostics(coxph.fit2, type = "scaledsch", ox.scale = "time")

</code></pre>

<hr>
<h2 id='ggcoxfunctional'>Functional Form of Continuous Variable in Cox Proportional Hazards Model</h2><span id='topic+ggcoxfunctional'></span><span id='topic+print.ggcoxfunctional'></span>

<h3>Description</h3>

<p>Displays graphs of continuous explanatory variable against martingale residuals of null
cox proportional hazards model, for each term in of the right side of <code>formula</code>. This might help to properly
choose the functional form of continuous variable in cox model (<a href="survival.html#topic+coxph">coxph</a>). Fitted lines with <a href="stats.html#topic+lowess">lowess</a> function
should be linear to satisfy cox proportional hazards model assumptions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggcoxfunctional(
  formula,
  data = NULL,
  fit,
  iter = 0,
  f = 0.6,
  point.col = "red",
  point.size = 1,
  point.shape = 19,
  point.alpha = 1,
  xlim = NULL,
  ylim = NULL,
  ylab = "Martingale Residuals \nof Null Cox Model",
  title = NULL,
  caption = NULL,
  ggtheme = theme_survminer(),
  ...
)

## S3 method for class 'ggcoxfunctional'
print(x, ..., newpage = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggcoxfunctional_+3A_formula">formula</code></td>
<td>
<p>a formula object, with the response on the left of a ~ operator, and the terms on the right. The response must be a survival object as returned by the <a href="survival.html#topic+Surv">Surv</a> function.</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_data">data</code></td>
<td>
<p>a <code>data.frame</code> in which to interpret the variables named in the formula,</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_fit">fit</code></td>
<td>
<p>an object of class <a href="survival.html#topic+coxph.object">coxph.object</a> - created with <a href="survival.html#topic+coxph">coxph</a> function.</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_iter">iter</code></td>
<td>
<p>parameter of <a href="stats.html#topic+lowess">lowess</a>.</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_f">f</code></td>
<td>
<p>parameter of <a href="stats.html#topic+lowess">lowess</a>.</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_point.col">point.col</code>, <code id="ggcoxfunctional_+3A_point.size">point.size</code>, <code id="ggcoxfunctional_+3A_point.shape">point.shape</code>, <code id="ggcoxfunctional_+3A_point.alpha">point.alpha</code></td>
<td>
<p>color, size, shape and visibility to be used for points.</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_xlim">xlim</code>, <code id="ggcoxfunctional_+3A_ylim">ylim</code></td>
<td>
<p>x and y axis limits e.g. xlim = c(0, 1000), ylim = c(0, 1).</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_ylab">ylab</code></td>
<td>
<p>y axis label.</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_title">title</code></td>
<td>
<p>the title of the final <a href="grid.html#topic+grob">grob</a> (<code>top</code> in <a href="gridExtra.html#topic+arrangeGrob">arrangeGrob</a>)</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_caption">caption</code></td>
<td>
<p>the caption of the final <a href="grid.html#topic+grob">grob</a> (<code>bottom</code> in <a href="gridExtra.html#topic+arrangeGrob">arrangeGrob</a>)</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name.
Allowed values include ggplot2 official themes: see <code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_...">...</code></td>
<td>
<p>further arguments passed to the function <code><a href="ggpubr.html#topic+ggpar">ggpar</a></code> for customizing the plot.</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_x">x</code></td>
<td>
<p>an object of class ggcoxfunctional</p>
</td></tr>
<tr><td><code id="ggcoxfunctional_+3A_newpage">newpage</code></td>
<td>
<p>open a new page. See <code><a href="gridExtra.html#topic+grid.arrange">grid.arrange</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>ggcoxfunctional</code> which is a list of ggplots.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>ggcoxfunctional</code>: Functional Form of Continuous Variable in Cox Proportional Hazards Model.
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Marcin Kosinski , <a href="mailto:m.p.kosinski@gmail.com">m.p.kosinski@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)
data(mgus)
res.cox &lt;- coxph(Surv(futime, death) ~ mspike + log(mspike) + I(mspike^2) +
    age + I(log(age)^2) + I(sqrt(age)), data = mgus)
ggcoxfunctional(res.cox,  data = mgus, point.col = "blue", point.alpha = 0.5)
ggcoxfunctional(res.cox, data = mgus, point.col = "blue", point.alpha = 0.5,
                title = "Pass the title", caption = "Pass the caption")


</code></pre>

<hr>
<h2 id='ggcoxzph'>Graphical Test of Proportional Hazards with ggplot2</h2><span id='topic+ggcoxzph'></span><span id='topic+print.ggcoxzph'></span>

<h3>Description</h3>

<p>Displays a graph of the scaled Schoenfeld residuals, along with a
smooth curve using <span class="pkg">ggplot2</span>. Wrapper around <a href="survival.html#topic+plot.cox.zph">plot.cox.zph</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggcoxzph(
  fit,
  resid = TRUE,
  se = TRUE,
  df = 4,
  nsmo = 40,
  var,
  point.col = "red",
  point.size = 1,
  point.shape = 19,
  point.alpha = 1,
  caption = NULL,
  ggtheme = theme_survminer(),
  ...
)

## S3 method for class 'ggcoxzph'
print(x, ..., newpage = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggcoxzph_+3A_fit">fit</code></td>
<td>
<p>an object of class <a href="survival.html#topic+cox.zph">cox.zph</a>.</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_resid">resid</code></td>
<td>
<p>a logical value, if TRUE the residuals are included on the plot,
as well as the smooth fit.</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_se">se</code></td>
<td>
<p>a logical value, if TRUE, confidence bands at two standard errors
will be added.</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_df">df</code></td>
<td>
<p>the degrees of freedom for the fitted natural spline, df=2 leads to
a linear fit.</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_nsmo">nsmo</code></td>
<td>
<p>number of points used to plot the fitted spline.</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_var">var</code></td>
<td>
<p>the set of variables for which plots are desired. By default, plots
are produced in turn for each variable of a model.</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_point.col">point.col</code>, <code id="ggcoxzph_+3A_point.size">point.size</code>, <code id="ggcoxzph_+3A_point.shape">point.shape</code>, <code id="ggcoxzph_+3A_point.alpha">point.alpha</code></td>
<td>
<p>color, size, shape and visibility to be used for points.</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_caption">caption</code></td>
<td>
<p>the caption of the final <a href="grid.html#topic+grob">grob</a> (<code>bottom</code> in <a href="gridExtra.html#topic+arrangeGrob">arrangeGrob</a>)</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name.
Allowed values include ggplot2 official themes: see <code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_...">...</code></td>
<td>
<p>further arguments passed to either the print() function or to the <code><a href="ggpubr.html#topic+ggpar">ggpar</a></code> function for customizing the plot (see Details section).</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_x">x</code></td>
<td>
<p>an object of class ggcoxzph</p>
</td></tr>
<tr><td><code id="ggcoxzph_+3A_newpage">newpage</code></td>
<td>
<p>open a new page. See <code><a href="gridExtra.html#topic+grid.arrange">grid.arrange</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Customizing the plots</strong>: The plot can be easily
customized using additional arguments to be passed to the function ggpar().
Read ?ggpubr::ggpar. These arguments include
<em>font.main,font.submain,font.caption,font.x,font.y,font.tickslab,font.legend</em>:
a vector of length 3 indicating respectively the size (e.g.: 14), the style
(e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;)
of main title, subtitle, caption, xlab and ylab and axis tick labels,
respectively. For example <em>font.x = c(14, &quot;bold&quot;, &quot;red&quot;)</em>.  Use font.x
= 14, to change only font size; or use font.x = &quot;bold&quot;, to change only font
face.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>ggcoxzph</code> which is a list of ggplots.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>ggcoxzph</code>: Graphical Test of Proportional Hazards using ggplot2.
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Marcin Kosinski , <a href="mailto:m.p.kosinski@gmail.com">m.p.kosinski@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)
fit &lt;- coxph(Surv(futime, fustat) ~ age + ecog.ps + rx, data=ovarian)
cox.zph.fit &lt;- cox.zph(fit)
# plot all variables
ggcoxzph(cox.zph.fit)
# plot all variables in specified order
ggcoxzph(cox.zph.fit, var = c("ecog.ps", "rx", "age"), font.main = 12)
# plot specified variables in specified order
ggcoxzph(cox.zph.fit, var = c("ecog.ps", "rx"), font.main = 12, caption = "Caption goes here")

</code></pre>

<hr>
<h2 id='ggflexsurvplot'>Ggplots of Fitted Flexible Survival Models</h2><span id='topic+ggflexsurvplot'></span>

<h3>Description</h3>

<p>Create ggplot2-based graphs for flexible survival models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggflexsurvplot(
  fit,
  data = NULL,
  fun = c("survival", "cumhaz"),
  summary.flexsurv = NULL,
  size = 1,
  conf.int = FALSE,
  conf.int.flex = conf.int,
  conf.int.km = FALSE,
  legend.labs = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggflexsurvplot_+3A_fit">fit</code></td>
<td>
<p>an object of class <code>flexsurvreg</code>.</p>
</td></tr>
<tr><td><code id="ggflexsurvplot_+3A_data">data</code></td>
<td>
<p>the data used to fit survival curves.</p>
</td></tr>
<tr><td><code id="ggflexsurvplot_+3A_fun">fun</code></td>
<td>
<p>the type of survival curves. Allowed values include &quot;survival&quot;
(default) and &quot;cumhaz&quot; (for cumulative hazard).</p>
</td></tr>
<tr><td><code id="ggflexsurvplot_+3A_summary.flexsurv">summary.flexsurv</code></td>
<td>
<p>(optional) the summary of the <code>flexsurvreg</code>
object as generated by the function <code>summary()</code>.</p>
</td></tr>
<tr><td><code id="ggflexsurvplot_+3A_size">size</code></td>
<td>
<p>line size for the flexible survival estimates.</p>
</td></tr>
<tr><td><code id="ggflexsurvplot_+3A_conf.int">conf.int</code>, <code id="ggflexsurvplot_+3A_conf.int.flex">conf.int.flex</code></td>
<td>
<p>logical. If TRUE, add confidence bands for
flexible survival estimates.</p>
</td></tr>
<tr><td><code id="ggflexsurvplot_+3A_conf.int.km">conf.int.km</code></td>
<td>
<p>same as <code>conf.in.flex</code> but for the kaplan-meier
estimates.</p>
</td></tr>
<tr><td><code id="ggflexsurvplot_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels. Used to replace
the names of the strata from the fit. Should be given in the same order as
those strata.</p>
</td></tr>
<tr><td><code id="ggflexsurvplot_+3A_...">...</code></td>
<td>
<p>additional arguments passed to the function <code><a href="#topic+ggsurvplot">ggsurvplot</a>()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggsurvplot
</p>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara, <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(require("flexsurv")) {
fit &lt;- flexsurvreg(Surv(rectime, censrec) ~ group,
                   dist = "gengamma", data = bc)
ggflexsurvplot(fit)
}


</code></pre>

<hr>
<h2 id='ggforest'>Forest Plot for Cox Proportional Hazards Model</h2><span id='topic+ggforest'></span>

<h3>Description</h3>

<p>Drawing Forest Plot for Cox proportional hazards model. In two panels the model structure is presented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggforest(
  model,
  data = NULL,
  main = "Hazard ratio",
  cpositions = c(0.02, 0.22, 0.4),
  fontsize = 0.7,
  refLabel = "reference",
  noDigits = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggforest_+3A_model">model</code></td>
<td>
<p>an object of class coxph.</p>
</td></tr>
<tr><td><code id="ggforest_+3A_data">data</code></td>
<td>
<p>a dataset used to fit survival curves. If not supplied then data
will be extracted from 'fit' object.</p>
</td></tr>
<tr><td><code id="ggforest_+3A_main">main</code></td>
<td>
<p>title of the plot.</p>
</td></tr>
<tr><td><code id="ggforest_+3A_cpositions">cpositions</code></td>
<td>
<p>relative positions of first three columns in the OX scale.</p>
</td></tr>
<tr><td><code id="ggforest_+3A_fontsize">fontsize</code></td>
<td>
<p>relative size of annotations in the plot. Default value: 0.7.</p>
</td></tr>
<tr><td><code id="ggforest_+3A_reflabel">refLabel</code></td>
<td>
<p>label for reference levels of factor variables.</p>
</td></tr>
<tr><td><code id="ggforest_+3A_nodigits">noDigits</code></td>
<td>
<p>number of digits for estimates and p-values in the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a ggplot2 object (invisibly)
</p>


<h3>Author(s)</h3>

<p>Przemyslaw Biecek (<a href="mailto:przemyslaw.biecek@gmail.com">przemyslaw.biecek@gmail.com</a>),
Fabian Scheipl (<a href="mailto:fabian.scheipl@gmail.com">fabian.scheipl@gmail.com</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("survival")
model &lt;- coxph( Surv(time, status) ~ sex + rx + adhere,
                data = colon )
ggforest(model)

colon &lt;- within(colon, {
  sex &lt;- factor(sex, labels = c("female", "male"))
  differ &lt;- factor(differ, labels = c("well", "moderate", "poor"))
  extent &lt;- factor(extent, labels = c("submuc.", "muscle", "serosa", "contig."))
})
bigmodel &lt;-
  coxph(Surv(time, status) ~ sex + rx + adhere + differ + extent + node4,
    data = colon )
ggforest(bigmodel)

</code></pre>

<hr>
<h2 id='ggrisktable'>Plot Survival Tables</h2><span id='topic+ggrisktable'></span><span id='topic+ggcumevents'></span><span id='topic+ggcumcensor'></span><span id='topic+ggsurvtable'></span>

<h3>Description</h3>

<p>Plot survival tables:
</p>

<ul>
<li> <p><code>ggrisktable()</code>: Plot the number at risk table.
</p>
</li>
<li> <p><code>ggcumevents()</code>: Plot the cumulative number of events table.
</p>
</li>
<li> <p><code>ggcumcensor()</code>: Plot the cumulative number of censored subjects, the number of subjects who
exit the risk set, without an event, at time t. Normally, users don't need
to use this function directly.
</p>
</li>
<li> <p><code>ggsurvtable()</code>: Generic function to plot any survival tables.
</p>
</li></ul>

<p>Normally, users don't need to use this function directly. Internally used by the function
<code><a href="#topic+ggsurvplot">ggsurvplot</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggrisktable(
  fit,
  data = NULL,
  risk.table.type = c("absolute", "percentage", "abs_pct", "nrisk_cumcensor",
    "nrisk_cumevents"),
  ...
)

ggcumevents(fit, data = NULL, ...)

ggcumcensor(fit, data = NULL, ...)

ggsurvtable(
  fit,
  data = NULL,
  survtable = c("cumevents", "cumcensor", "risk.table"),
  risk.table.type = c("absolute", "percentage", "abs_pct", "nrisk_cumcensor",
    "nrisk_cumevents"),
  title = NULL,
  risk.table.title = NULL,
  cumevents.title = title,
  cumcensor.title = title,
  color = "black",
  palette = NULL,
  break.time.by = NULL,
  xlim = NULL,
  xscale = 1,
  xlab = "Time",
  ylab = "Strata",
  xlog = FALSE,
  legend = "top",
  legend.title = "Strata",
  legend.labs = NULL,
  y.text = TRUE,
  y.text.col = TRUE,
  fontsize = 4.5,
  font.family = "",
  axes.offset = TRUE,
  ggtheme = theme_survminer(),
  tables.theme = ggtheme,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggrisktable_+3A_fit">fit</code></td>
<td>
<p>an object of class survfit. Can be a list containing two
components: 1) time: time variable used in survfit; 2) table: survival table
as generated by the internal function .get_timepoints_survsummary(). Can be
also a simple data frame.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_data">data</code></td>
<td>
<p>a dataset used to fit survival curves. If not supplied then data
will be extracted from 'fit' object.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_risk.table.type">risk.table.type</code></td>
<td>
<p>risk table type. Allowed values include: &quot;absolute&quot; or
&quot;percentage&quot;: to show the <b>absolute number</b> and the <b>percentage</b>
of subjects at risk by time, respectively. Use &quot;abs_pct&quot; to show both
absolute number and percentage. Used only when survtable = &quot;risk.table&quot;.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function <code><a href="#topic+ggsurvtable">ggsurvtable</a></code> and <code><a href="ggpubr.html#topic+ggpar">ggpar</a></code>.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_survtable">survtable</code></td>
<td>
<p>a character string specifying the type of survival table to plot.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_title">title</code></td>
<td>
<p>the title of the plot.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_risk.table.title">risk.table.title</code></td>
<td>
<p>The title to be used for the risk table.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_cumevents.title">cumevents.title</code></td>
<td>
<p>The title to be used for the cumulative events table.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_cumcensor.title">cumcensor.title</code></td>
<td>
<p>The title to be used for the cumcensor table.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_color">color</code></td>
<td>
<p>color to be used for the survival curves. </p>
 <ul>
<li><p> If the
number of strata/group (n.strata) = 1, the expected value is the color name.
For example color = &quot;blue&quot;. </p>
</li>
<li><p> If n.strata &gt; 1, the expected value is the
grouping variable name. By default, survival curves are colored by strata
using the argument color = &quot;strata&quot;,  but you can also color survival curves
by any other grouping variables used to fit the survival curves. In this
case, it's possible to specify a custom color palette by using the argument
palette.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggrisktable_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used. Allowed values include &quot;hue&quot; for
the default hue color scale; &quot;grey&quot; for grey color palettes; brewer palettes
e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;,
&quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;.
See details section for more information.  Can be also a numeric vector of
length(groups); in this case a basic color palette is created using the
function <a href="grDevices.html#topic+palette">palette</a>.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_break.time.by">break.time.by</code></td>
<td>
<p>numeric value controlling time axis breaks. Default value
is NULL.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_xlim">xlim</code></td>
<td>
<p>x and y axis limits e.g. xlim = c(0, 1000), ylim = c(0, 1).</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_xscale">xscale</code></td>
<td>
<p>numeric or character value specifying x-axis scale. </p>

<ul>
<li><p> If numeric, the value is used to divide the labels on the x axis. For
example, a value of 365.25 will give labels in years instead of the original
days. </p>
</li>
<li><p> If character, allowed options include one of c(&quot;d_m&quot;, &quot;d_y&quot;,
&quot;m_d&quot;, &quot;m_y&quot;, &quot;y_d&quot;, &quot;y_m&quot;), where d = days, m = months and y = years. For
example, xscale = &quot;d_m&quot; will transform labels from days to months; xscale =
&quot;m_y&quot;, will transform labels from months to years.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggrisktable_+3A_xlab">xlab</code></td>
<td>
<p>main title and axis labels</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_ylab">ylab</code></td>
<td>
<p>main title and axis labels</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_xlog">xlog</code></td>
<td>
<p>logical value. If TRUE, x axis is tansformed into log scale.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_legend">legend</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). Default is &quot;top&quot; side position.
to remove the legend use legend = &quot;none&quot;. Legend position can be also
specified using a numeric vector c(x, y); see details section.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels. Used to replace
the names of the strata from the fit. Should be given in the same order as
those strata.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_y.text">y.text</code></td>
<td>
<p>logical. Default is TRUE. If FALSE, the table y axis tick
labels will be hidden.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_y.text.col">y.text.col</code></td>
<td>
<p>logical. Default value is FALSE. If TRUE, the table tick
labels will be colored by strata.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_fontsize">fontsize</code></td>
<td>
<p>text font size.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_font.family">font.family</code></td>
<td>
<p>character vector specifying text element font family, e.g.: font.family = &quot;Courier New&quot;.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_axes.offset">axes.offset</code></td>
<td>
<p>logical value. Default is TRUE. If FALSE, set the plot axes
to start at the origin.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggrisktable_+3A_tables.theme">tables.theme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>. Note that, <code>tables.theme</code> is incremental to <code>ggtheme</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>ggrisktable</code>: Plot the number at risk table.
</p>
</li>
<li> <p><code>ggcumevents</code>: Plot the cumulative number of events table
</p>
</li>
<li> <p><code>ggcumcensor</code>: Plot the cumulative number of censor table
</p>
</li>
<li> <p><code>ggsurvtable</code>: Generic function to plot survival tables: risk.table, cumevents and cumcensor
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara, <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Fit survival curves
#:::::::::::::::::::::::::::::::::::::::::::::::
require("survival")
fit&lt;- survfit(Surv(time, status) ~ sex, data = lung)

# Survival tables
#:::::::::::::::::::::::::::::::::::::::::::::::
tables &lt;- ggsurvtable(fit, data = lung, color = "strata",
  y.text = FALSE)

# Risk table
tables$risk.table

# Number of cumulative events
tables$cumevents

# Number of cumulative censoring
tables$cumcensor
</code></pre>

<hr>
<h2 id='ggsurvevents'>Distribution of Events' Times</h2><span id='topic+ggsurvevents'></span>

<h3>Description</h3>

<p>Distribution of Events' Times
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsurvevents(
  surv = NULL,
  fit = NULL,
  data = NULL,
  type = "fraction",
  normalized = TRUE,
  censored.on.top = TRUE,
  ggtheme = theme_survminer(),
  palette = c("grey75", "grey25"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvevents_+3A_surv">surv</code></td>
<td>
<p>an object of <a href="survival.html#topic+Surv">Surv</a>. If not suplied, the censoring variable is extracted from the model.</p>
</td></tr>
<tr><td><code id="ggsurvevents_+3A_fit">fit</code></td>
<td>
<p>an object of class <a href="survival.html#topic+survfit">survfit</a>.</p>
</td></tr>
<tr><td><code id="ggsurvevents_+3A_data">data</code></td>
<td>
<p>a dataset for predictions. If not supplied then data will be extracted from 'fit' object.</p>
</td></tr>
<tr><td><code id="ggsurvevents_+3A_type">type</code></td>
<td>
<p>one of <code>c("cumulative", "radius", "fraction")</code>. <code>"cumulative"</code> stands for cumulative number of events, <code>"radius"</code> stands for number of events within a given radius,</p>
</td></tr>
<tr><td><code id="ggsurvevents_+3A_normalized">normalized</code></td>
<td>
<p>if <code>TRUE</code> relative number of events is presented,</p>
</td></tr>
<tr><td><code id="ggsurvevents_+3A_censored.on.top">censored.on.top</code></td>
<td>
<p>is TRUE then censored events are on the top</p>
</td></tr>
<tr><td><code id="ggsurvevents_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name.  Allowed values include ggplot2 official themes: see theme.</p>
</td></tr>
<tr><td><code id="ggsurvevents_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used for coloring of significant variables.</p>
</td></tr>
<tr><td><code id="ggsurvevents_+3A_...">...</code></td>
<td>
<p>other graphical parameters to be passed to the function <a href="ggpubr.html#topic+ggpar">ggpar</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return an object of class ggplot
</p>


<h3>Author(s)</h3>

<p>Przemyslaw Biecek, <a href="mailto:przemyslaw.biecek@gmail.com">przemyslaw.biecek@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("survival")
# from Surv
surv &lt;- Surv(lung$time, lung$status)
ggsurvevents(surv)

surv2 &lt;- Surv(colon$time, colon$status)
ggsurvevents(surv2)
ggsurvevents(surv2, normalized = TRUE)

# from survfit
fit &lt;- survfit(Surv(time, status) ~ sex, data = lung)
ggsurvevents(fit = fit, data = lung)

# from coxph
model &lt;- coxph( Surv(time, status) ~ sex + rx + adhere, data = colon )
ggsurvevents(fit = model, data = colon)
ggsurvevents(surv2, normalized = TRUE, type = "radius")
ggsurvevents(surv2, normalized = TRUE, type = "fraction")

</code></pre>

<hr>
<h2 id='ggsurvplot'>Drawing Survival Curves Using ggplot2</h2><span id='topic+ggsurvplot'></span><span id='topic+print.ggsurvplot'></span>

<h3>Description</h3>

<p><code>ggsurvplot</code>() is a generic function to plot survival curves. Wrapper
around the <code>ggsurvplot_xx()</code> family functions. Plot one or a list of
survfit objects as generated by the
<a href="survival.html#topic+survfit.formula">survfit.formula</a>() and <a href="#topic+surv_fit">surv_fit</a> functions: </p>
  <ul>
<li>
<p><a href="#topic+ggsurvplot_list">ggsurvplot_list</a>() </p>
</li>
<li> <p><a href="#topic+ggsurvplot_facet">ggsurvplot_facet</a>() </p>
</li>
<li>
<p><a href="#topic+ggsurvplot_group_by">ggsurvplot_group_by</a>() </p>
</li>
<li> <p><a href="#topic+ggsurvplot_add_all">ggsurvplot_add_all</a>() </p>
</li>
<li>
<p><a href="#topic+ggsurvplot_combine">ggsurvplot_combine</a>() </p>
</li></ul>
<p> See the documentation for each function to
learn how to control that aspect of the <a href="#topic+ggsurvplot">ggsurvplot</a>().
<code>ggsurvplot</code>() accepts further arguments to be passed to the
<code>ggsurvplot_xx()</code> functions. Has options to: </p>
 <ul>
<li>
<p>plot a list of survfit objects, </p>
</li>
<li><p> facet survival curves into multiple
panels, </p>
</li>
<li><p> group  dataset by one or two grouping variables and to create
the survival curves in each subset, </p>
</li>
<li><p> combine multiple <code>survfit</code>
objects into one plot, </p>
</li>
<li><p> add survival curves of the pooled patients
(null model) onto the main stratified plot, </p>
</li>
<li><p> plot survival curves from
a data frame containing survival curve summary as returned by
<a href="#topic+surv_summary">surv_summary</a>(). </p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>ggsurvplot(
  fit,
  data = NULL,
  fun = NULL,
  color = NULL,
  palette = NULL,
  linetype = 1,
  conf.int = FALSE,
  pval = FALSE,
  pval.method = FALSE,
  test.for.trend = FALSE,
  surv.median.line = "none",
  risk.table = FALSE,
  cumevents = FALSE,
  cumcensor = FALSE,
  tables.height = 0.25,
  group.by = NULL,
  facet.by = NULL,
  add.all = FALSE,
  combine = FALSE,
  ggtheme = theme_survminer(),
  tables.theme = ggtheme,
  ...
)

## S3 method for class 'ggsurvplot'
print(
  x,
  surv.plot.height = NULL,
  risk.table.height = NULL,
  ncensor.plot.height = NULL,
  newpage = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvplot_+3A_fit">fit</code></td>
<td>
<p>allowed values include:
</p>

<ul>
<li><p> a survfit object
</p>
</li>
<li><p> a list of survfit objects. Passed to <code><a href="#topic+ggsurvplot_list">ggsurvplot_list</a>()</code>
</p>
</li>
<li><p> a data frame containing survival curves summary. Passed to <code><a href="#topic+ggsurvplot_df">ggsurvplot_df</a>()</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_data">data</code></td>
<td>
<p>a dataset used to fit survival curves. If not supplied then data
will be extracted from 'fit' object.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_fun">fun</code></td>
<td>
<p>an arbitrary function defining a transformation of the survival
curve.  Often used transformations can be specified with a character
argument: &quot;event&quot; plots cumulative events (f(y) = 1-y), &quot;cumhaz&quot; plots the
cumulative hazard function (f(y) = -log(y)), and &quot;pct&quot; for survival
probability in percentage.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_color">color</code></td>
<td>
<p>color to be used for the survival curves. </p>
 <ul>
<li><p> If the
number of strata/group (n.strata) = 1, the expected value is the color name.
For example color = &quot;blue&quot;. </p>
</li>
<li><p> If n.strata &gt; 1, the expected value is the
grouping variable name. By default, survival curves are colored by strata
using the argument color = &quot;strata&quot;,  but you can also color survival curves
by any other grouping variables used to fit the survival curves. In this
case, it's possible to specify a custom color palette by using the argument
palette.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used. Allowed values include &quot;hue&quot; for
the default hue color scale; &quot;grey&quot; for grey color palettes; brewer palettes
e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;,
&quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;.
See details section for more information.  Can be also a numeric vector of
length(groups); in this case a basic color palette is created using the
function <a href="grDevices.html#topic+palette">palette</a>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_linetype">linetype</code></td>
<td>
<p>line types. Allowed values includes i) &quot;strata&quot; for changing
linetypes by strata (i.e. groups); ii) a numeric vector (e.g., c(1, 2)) or a
character vector c(&quot;solid&quot;, &quot;dashed&quot;).</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_conf.int">conf.int</code></td>
<td>
<p>logical value. If TRUE, plots confidence interval.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_pval">pval</code></td>
<td>
<p>logical value, a numeric or a string. If logical and TRUE, the
p-value is added on the plot. If numeric, than the computet p-value is
substituted with the one passed with this parameter. If character, then the
customized string appears on the plot. See examples - Example 3.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_pval.method">pval.method</code></td>
<td>
<p>whether to add a text with the test name used for
calculating the pvalue, that corresponds to survival curves' comparison -
used only when <code>pval=TRUE</code></p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_test.for.trend">test.for.trend</code></td>
<td>
<p>logical value. Default is FALSE. If TRUE, returns the
test for trend p-values. Tests for trend are designed to detect ordered
differences in survival curves. That is, for at least one group. The test
for trend can be only performed when the number of groups is &gt; 2.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_surv.median.line">surv.median.line</code></td>
<td>
<p>character vector for drawing a horizontal/vertical
line at median survival. Allowed values include one of c(&quot;none&quot;, &quot;hv&quot;, &quot;h&quot;,
&quot;v&quot;). v: vertical, h:horizontal.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_risk.table">risk.table</code></td>
<td>
<p>Allowed values include: </p>
 <ul>
<li><p> TRUE or FALSE
specifying whether to show or not the risk table. Default is FALSE. </p>
</li>
<li>
<p>&quot;absolute&quot; or &quot;percentage&quot;. Shows the <b>absolute number</b> and the
<b>percentage</b> of subjects at risk by time, respectively. </p>
</li>
<li><p> &quot;abs_pct&quot;
to show both absolute number and percentage. </p>
</li>
<li><p> &quot;nrisk_cumcensor&quot; and
&quot;nrisk_cumevents&quot;. Show the number at risk and, the cumulative number of
censoring and events, respectively. </p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_cumevents">cumevents</code></td>
<td>
<p>logical value specifying whether to show or not the table of
the cumulative number of events. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_cumcensor">cumcensor</code></td>
<td>
<p>logical value specifying whether to show or not the table of
the cumulative number of censoring. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_tables.height">tables.height</code></td>
<td>
<p>numeric value (in [0 - 1]) specifying the general height
of all tables under the main survival plot.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_group.by">group.by</code></td>
<td>
<p>a character vector containing the name of grouping variables. Should be of length &lt;= 2.
Alias of the <code><a href="#topic+ggsurvplot_group_by">ggsurvplot_group_by</a>()</code> function.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_facet.by">facet.by</code></td>
<td>
<p>a character vector containing the name of grouping variables
to facet the survival curves into multiple panels. Should be of length &lt;= 2.
Alias of the <code><a href="#topic+ggsurvplot_facet">ggsurvplot_facet</a>()</code> function.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_add.all">add.all</code></td>
<td>
<p>a logical value. If TRUE, add the survival curve of pooled patients (null model) onto the main plot.
Alias of the <code><a href="#topic+ggsurvplot_add_all">ggsurvplot_add_all</a>()</code> function.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_combine">combine</code></td>
<td>
<p>a logical value. If TRUE, combine a list survfit objects on the same plot.
Alias of the <code><a href="#topic+ggsurvplot_combine">ggsurvplot_combine</a>()</code> function.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_tables.theme">tables.theme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>. Note that, <code>tables.theme</code> is incremental to <code>ggtheme</code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_...">...</code></td>
<td>
<p>Futher arguments as described hereafter and
other arguments to be passed i) to ggplot2 geom_*() functions such
as linetype, size, ii) or to the function <a href="ggpubr.html#topic+ggpar">ggpar</a>() for
customizing the plots. See details section.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_x">x</code></td>
<td>
<p>an object of class ggsurvplot</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_surv.plot.height">surv.plot.height</code></td>
<td>
<p>the height of the survival plot on the grid. Default
is 0.75. Ignored when risk.table = FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_risk.table.height">risk.table.height</code></td>
<td>
<p>the height of the risk table on the grid. Increase
the value when you have many strata. Default is 0.25. Ignored when
risk.table = FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_ncensor.plot.height">ncensor.plot.height</code></td>
<td>
<p>The height of the censor plot. Used when
<code>ncensor.plot = TRUE</code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_+3A_newpage">newpage</code></td>
<td>
<p>open a new page. See <code><a href="gridExtra.html#topic+grid.arrange">grid.arrange</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

  <ul>
<li>
<p><strong>Color palettes</strong>: The argument <strong>palette</strong> can be used to
specify the color to be used for each group. By default, the first color in
the palette is used to color the first level of the factor variable. This
default behavior can be changed by assigning correctly a named vector. That
is, the names of colors should match the strata names as generated by the
<code>ggsurvplot()</code> function in the legend.
</p>
</li></ul>



<h3>Value</h3>

<p>return an object of class ggsurvplot which is list containing the
following components: </p>
 <ul>
<li><p> plot: the survival plot (ggplot
object) </p>
</li>
<li><p> table: the number of subjects at risk table per time (ggplot
object). </p>
</li>
<li><p> cumevents: the cumulative number of events table (ggplot
object). </p>
</li>
<li><p> ncensor.plot: the number of censoring (ggplot object). </p>
</li>
<li>
<p>data.survplot: the data used to plot the survival curves (data.frame). </p>
</li>
<li>
<p>data.survtable: the data used to plot the tables under the main survival
curves (data.frame). </p>
</li></ul>



<h3>FURTHER ARGUMENTS</h3>

<p>Customize survival plots and tables. See also <a href="#topic+ggsurvplot_arguments">ggsurvplot_arguments</a>.
</p>


<h3>Plot title and axis labels</h3>


<ul>
<li> <p><strong>title</strong>: main title.
</p>
</li>
<li> <p><strong>xlab, ylab</strong>: x and y axis labels, respectively.
</p>
</li></ul>



<h3>Legend title, labels and position</h3>


<ul>
<li> <p><strong>legend</strong>: character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). Default is &quot;top&quot; side position.
to remove the legend use legend = &quot;none&quot;. Legend position can be also
specified using a numeric vector c(x, y). In this case it is
possible to position the legend inside the plotting area. x and y are the
coordinates of the legend box. Their values should be between 0 and 1.
c(0,0) corresponds to the &quot;bottom left&quot; and c(1,1) corresponds to the &quot;top
right&quot; position. For instance use legend = c(0.8, 0.2).
</p>
</li>
<li> <p><strong>legend.title</strong>: legend title.
</p>
</li>
<li> <p><strong>legend.labs</strong>: character vector specifying legend labels. Used to replace
the names of the strata from the fit. Should be given in the same order as
those strata.
</p>
</li></ul>



<h3>Axis limits, breaks and scales</h3>


<ul>
<li> <p><strong>break.time.by</strong>: numeric value controlling time axis breaks. Default value
is NULL.
</p>
</li>
<li> <p><strong>break.x.by</strong>: alias of break.time.by. Numeric value controlling x axis
breaks. Default value is NULL.
</p>
</li>
<li> <p><strong>break.y.by</strong>: same as break.x.by but for y axis.
</p>
</li>
<li> <p><strong>surv.scale</strong>: scale transformation of survival curves. Allowed values are
&quot;default&quot; or &quot;percent&quot;.
</p>
</li>
<li> <p><strong>xscale</strong>: numeric or character value specifying x-axis scale. </p>

<ul>
<li><p> If numeric, the value is used to divide the labels on the x axis. For
example, a value of 365.25 will give labels in years instead of the original
days. </p>
</li>
<li><p> If character, allowed options include one of - &quot;d_m&quot;, &quot;d_y&quot;,
&quot;m_d&quot;, &quot;m_y&quot;, &quot;y_d&quot; and &quot;y_m&quot; -  where <code>d = days</code>, <code>m = months</code> and <code>y = years</code>. For
example, <code>xscale = "d_m"</code> will transform labels from days to months; <code>xscale =
 "m_y"</code>, will transform labels from months to years.</p>
</li></ul>

</li>
<li> <p><strong>xlim,ylim</strong>: x and y axis limits e.g. xlim = c(0, 1000), ylim = c(0, 1).
</p>
</li>
<li> <p><strong>axes.offset</strong>: logical value. Default is TRUE. If FALSE, set the plot axes to start at the origin.
</p>
</li></ul>



<h3>Confidence interval</h3>


<ul>
<li> <p><strong>conf.int.fill</strong>: fill color to be used for confidence interval.
</p>
</li>
<li> <p><strong>conf.int.style</strong>: confidence interval style. Allowed values include c(&quot;ribbon&quot;, &quot;step&quot;).
</p>
</li>
<li> <p><strong>conf.int.alpha</strong>: numeric value specifying confidence fill color transparency.
Value should be in [0, 1], where 0 is full transparency and 1 is no transparency.
</p>
</li></ul>



<h3>P-value</h3>


<ul>
<li> <p><strong>pval.size</strong>: numeric value specifying the p-value text size. Default is 5.
</p>
</li>
<li> <p><strong>pval.coord</strong>: numeric vector, of length 2,
specifying the x and y coordinates of the p-value.
Default values are NULL.
</p>
</li>
<li> <p><strong>pval.method.size</strong>: the same as <code>pval.size</code> but for displaying
<code>log.rank.weights</code> name.
</p>
</li>
<li> <p><strong>pval.method.coord</strong>: the same as <code>pval.coord</code> but for displaying
<code>log.rank.weights</code> name.
</p>
</li>
<li> <p><strong>log.rank.weights</strong>: the name for the type of weights to be used in
computing the p-value for log-rank test. By default <code>survdiff</code> is used
to calculate regular log-rank test (with weights == 1). A user can specify
<code>"1", "n", "sqrtN", "S1", "S2", "FH"</code> to use weights specified in
<a href="survMisc.html#topic+comp">comp</a>, so that weight correspond to the test as : 1 -
log-rank, n - Gehan-Breslow (generalized Wilcoxon), sqrtN - Tarone-Ware, S1
- Peto-Peto's modified survival estimate, S2 - modified Peto-Peto (by
Andersen), FH - Fleming-Harrington(p=1, q=1).
</p>
</li></ul>



<h3>Median survival</h3>


<ul>
<li> <p><strong>surv.median.line</strong>: character vector for drawing a
horizontal/vertical line at median survival.
Allowed values include one of c(&quot;none&quot;, &quot;hv&quot;, &quot;h&quot;, &quot;v&quot;). v: vertical, h:horizontal.
</p>
</li></ul>



<h3>Censor points</h3>


<ul>
<li> <p><strong>censor</strong>: logical value. If TRUE (default), censors will be drawn.
</p>
</li>
<li> <p><strong>censor.shape</strong>: character or numeric value specifying the point shape of censors.
Default value is &quot;+&quot; (3), a sensible choice is &quot;|&quot; (124).
</p>
</li>
<li> <p><strong>censor.size</strong>: numveric value specifying the point size of censors. Default is 4.5.
</p>
</li></ul>



<h3>Survival tables</h3>

<p><strong>General parameters for all tables</strong>.
The arguments below, when specified, will be applied to all survival tables at once
(risk, cumulative events and cumulative censoring tables).
</p>

<ul>
<li> <p><strong>tables.col</strong>: color to be used for all tables under the main plot. Default
value is &quot;black&quot;. If you want to color by strata (i.e. groups), use
tables.col = &quot;strata&quot;.
</p>
</li>
<li> <p><strong>fontsize</strong>: font size to be used for the risk table
and the cumulative events table.
</p>
</li>
<li> <p><strong>font.family</strong>: character vector specifying text element font family,
e.g.: font.family = &quot;Courier New&quot;.
</p>
</li>
<li> <p><strong>tables.y.text</strong>: logical. Default is TRUE. If FALSE, the y axis tick
labels of tables will be hidden.
</p>
</li>
<li> <p><strong>tables.y.text.col</strong>: logical. Default value is FALSE. If TRUE, the y
tick labels of tables will be colored by strata.
</p>
</li>
<li> <p><strong>tables.height</strong>: numeric value (in [0 - 1]) specifying the general height
of all tables under the main survival plot.
Increase the value when you have many strata. Default is 0.25.
</p>
</li></ul>

<p><strong>Specific to the risk table</strong>
</p>

<ul>
<li> <p><strong>risk.table.title</strong>: the title to be used for the risk table.
</p>
</li>
<li> <p><strong>risk.table.pos</strong>: character vector specifying the risk table position.
Allowed options are one of c(&quot;out&quot;, &quot;in&quot;) indicating 'outside' or 'inside'
the main plot, respectively. Default value is &quot;out&quot;.
</p>
</li>
<li> <p><code>risk.table.col</code>, <code>risk.table.fontsize</code>, <code>risk.table.y.text</code>,
<code>risk.table.y.text.col</code> and <code>risk.table.height</code>: same as for the general parameters
but applied to the risk table only.
</p>
</li></ul>

<p><strong>Specific to the number of cumulative events table (cumevents)</strong>
</p>

<ul>
<li> <p><strong>cumevents.title</strong>: the title to be used for the cumulative events table.
</p>
</li>
<li> <p><code>cumevents.col, cumevents.y.text, cumevents.y.text, cumevents.height</code>:
same as for the general parameters but for the cumevents table only.
</p>
</li></ul>

<p><strong>Specific to the number of cumulative censoring table (cumcensor)</strong>
</p>

<ul>
<li> <p><strong>cumcensor.title</strong>: the title to be used for the cumcensor table.
</p>
</li>
<li> <p><code>cumcensor.col</code>, <code>cumcensor.y.text</code>, <code>cumcensor.y.text.col</code>, <code>cumcensor.height</code>:
same as for the general parameters but for cumcensor table only.
</p>
</li></ul>



<h3>Survival plot height</h3>


<ul>
<li> <p><strong>surv.plot.height</strong>: the height of the survival plot on the grid. Default
is 0.75. Ignored when risk.table = FALSE.
</p>
</li></ul>



<h3>Number of censored subjects barplot</h3>


<ul>
<li> <p><strong>ncensor.plot</strong>: logical value. If TRUE, the number of censored subjects at
time t is plotted. Default is FALSE. Ignored when cumcensor = TRUE.
</p>
</li>
<li> <p><strong>ncensor.plot.title</strong>: the title to be used for the censor plot. Used when
<code>ncensor.plot = TRUE</code>.
</p>
</li>
<li> <p><strong>ncensor.plot.height</strong>: the height of the censor plot. Used when
<code>ncensor.plot = TRUE</code>.
</p>
</li></ul>



<h3>Other graphical parameters</h3>

<p>The plot can be easily customized using additional arguments to be
passed to the function <code><a href="ggpubr.html#topic+ggpar">ggpar</a>()</code>.
</p>
<p>These arguments include
<code>font.title, font.subtitle, font.caption, font.x, font.y, font.tickslab and font.legend</code>,
which are vectors of length 3 indicating respectively the size
(e.g.: 14), the style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and
the color (e.g.: &quot;red&quot;) of main title, subtitle, caption, xlab and ylab,
axis tick labels and legend, respectively. For example font.x = c(14,
&quot;bold&quot;, &quot;red&quot;).
</p>
<p>Use font.x = 14, to change only font size; or use font.x =
&quot;bold&quot;, to change only font face.
</p>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara, <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Example 1: Survival curves with two groups
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

# Fit survival curves
#++++++++++++++++++++++++++++++++++++
require("survival")
fit&lt;- survfit(Surv(time, status) ~ sex, data = lung)

# Basic survival curves
ggsurvplot(fit, data = lung)

# Customized survival curves
ggsurvplot(fit, data = lung,
 surv.median.line = "hv", # Add medians survival

 # Change legends: title &amp; labels
 legend.title = "Sex",
 legend.labs = c("Male", "Female"),
 # Add p-value and tervals
 pval = TRUE,

 conf.int = TRUE,
 # Add risk table
 risk.table = TRUE,
 tables.height = 0.2,
 tables.theme = theme_cleantable(),

 # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
 # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
 palette = c("#E7B800", "#2E9FDF"),
 ggtheme = theme_bw() # Change ggplot2 theme
)

# Change font size, style and color
#++++++++++++++++++++++++++++++++++++
## Not run: 
# Change font size, style and color at the same time
ggsurvplot(fit, data = lung,  main = "Survival curve",
   font.main = c(16, "bold", "darkblue"),
   font.x = c(14, "bold.italic", "red"),
   font.y = c(14, "bold.italic", "darkred"),
   font.tickslab = c(12, "plain", "darkgreen"))

## End(Not run)



#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Example 2: Facet ggsurvplot() output by
# a combination of factors
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

# Fit (complexe) survival curves
#++++++++++++++++++++++++++++++++++++
## Not run: 
require("survival")
fit3 &lt;- survfit( Surv(time, status) ~ sex + rx + adhere,
                data = colon )

# Visualize
#++++++++++++++++++++++++++++++++++++
ggsurv &lt;- ggsurvplot(fit3, data = colon,
  fun = "cumhaz", conf.int = TRUE,
  risk.table = TRUE, risk.table.col="strata",
  ggtheme = theme_bw())

# Faceting survival curves
curv_facet &lt;- ggsurv$plot + facet_grid(rx ~ adhere)
curv_facet

# Faceting risk tables:
# Generate risk table for each facet plot item
ggsurv$table + facet_grid(rx ~ adhere, scales = "free")+
 theme(legend.position = "none")

 # Generate risk table for each facet columns
tbl_facet &lt;- ggsurv$table + facet_grid(.~ adhere, scales = "free")
tbl_facet + theme(legend.position = "none")

# Arrange faceted survival curves and risk tables
g2 &lt;- ggplotGrob(curv_facet)
g3 &lt;- ggplotGrob(tbl_facet)
min_ncol &lt;- min(ncol(g2), ncol(g3))
g &lt;- gridExtra::gtable_rbind(g2[, 1:min_ncol], g3[, 1:min_ncol], size="last")
g$widths &lt;- grid::unit.pmax(g2$widths, g3$widths)
grid::grid.newpage()
grid::grid.draw(g)


## End(Not run)

#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Example 3: CUSTOMIZED PVALUE
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
# Customized p-value
ggsurvplot(fit, data = lung, pval = TRUE)
ggsurvplot(fit, data = lung, pval = 0.03)
ggsurvplot(fit, data = lung, pval = "The hot p-value is: 0.031")

</code></pre>

<hr>
<h2 id='ggsurvplot_add_all'>Add Survival Curves of Pooled Patients onto the Main Plot</h2><span id='topic+ggsurvplot_add_all'></span>

<h3>Description</h3>

<p>Add survival curves of pooled patients onto the main plot stratified by grouping variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsurvplot_add_all(
  fit,
  data,
  legend.title = "Strata",
  legend.labs = NULL,
  pval = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvplot_add_all_+3A_fit">fit</code></td>
<td>
<p>an object of class survfit.</p>
</td></tr>
<tr><td><code id="ggsurvplot_add_all_+3A_data">data</code></td>
<td>
<p>a dataset used to fit survival curves. If not supplied then data
will be extracted from 'fit' object.</p>
</td></tr>
<tr><td><code id="ggsurvplot_add_all_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title.</p>
</td></tr>
<tr><td><code id="ggsurvplot_add_all_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels. Used to replace
the names of the strata from the fit. Should be given in the same order as
those strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_add_all_+3A_pval">pval</code></td>
<td>
<p>logical value, a numeric or a string. If logical and TRUE, the
p-value is added on the plot. If numeric, than the computet p-value is
substituted with the one passed with this parameter. If character, then the
customized string appears on the plot. See examples - Example 3.</p>
</td></tr>
<tr><td><code id="ggsurvplot_add_all_+3A_...">...</code></td>
<td>
<p>other arguments passed to the <code><a href="#topic+ggsurvplot">ggsurvplot</a>()</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a ggsurvplot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggsurvplot">ggsurvplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)

# Fit survival curves
fit &lt;- surv_fit(Surv(time, status) ~ sex, data = lung)

# Visualize survival curves
ggsurvplot(fit, data = lung,
          risk.table = TRUE, pval = TRUE,
          surv.median.line = "hv", palette = "jco")

# Add survival curves of pooled patients (Null model)
# Use add.all = TRUE option
ggsurvplot(fit, data = lung,
          risk.table = TRUE, pval = TRUE,
          surv.median.line = "hv", palette = "jco",
          add.all = TRUE)

</code></pre>

<hr>
<h2 id='ggsurvplot_arguments'>ggsurvplot Argument Descriptions</h2><span id='topic+ggsurvplot_arguments'></span>

<h3>Description</h3>

<p>ggsurvplot Argument Descriptions
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvplot_arguments_+3A_fit">fit</code></td>
<td>
<p>an object of class survfit.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_data">data</code></td>
<td>
<p>a dataset used to fit survival curves. If not supplied then data
will be extracted from 'fit' object.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_fun">fun</code></td>
<td>
<p>an arbitrary function defining a transformation of the survival
curve.  Often used transformations can be specified with a character
argument: &quot;event&quot; plots cumulative events (f(y) = 1-y), &quot;cumhaz&quot; plots the
cumulative hazard function (f(y) = -log(y)), and &quot;pct&quot; for survival
probability in percentage.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_surv.scale">surv.scale</code></td>
<td>
<p>scale transformation of survival curves. Allowed values are
&quot;default&quot; or &quot;percent&quot;.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_xscale">xscale</code></td>
<td>
<p>numeric or character value specifying x-axis scale. </p>

<ul>
<li><p> If numeric, the value is used to divide the labels on the x axis. For
example, a value of 365.25 will give labels in years instead of the original
days. </p>
</li>
<li><p> If character, allowed options include one of c(&quot;d_m&quot;, &quot;d_y&quot;,
&quot;m_d&quot;, &quot;m_y&quot;, &quot;y_d&quot;, &quot;y_m&quot;), where d = days, m = months and y = years. For
example, xscale = &quot;d_m&quot; will transform labels from days to months; xscale =
&quot;m_y&quot;, will transform labels from months to years.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_color">color</code></td>
<td>
<p>color to be used for the survival curves. </p>
 <ul>
<li><p> If the
number of strata/group (n.strata) = 1, the expected value is the color name.
For example color = &quot;blue&quot;. </p>
</li>
<li><p> If n.strata &gt; 1, the expected value is the
grouping variable name. By default, survival curves are colored by strata
using the argument color = &quot;strata&quot;,  but you can also color survival curves
by any other grouping variables used to fit the survival curves. In this
case, it's possible to specify a custom color palette by using the argument
palette.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used. Allowed values include &quot;hue&quot; for
the default hue color scale; &quot;grey&quot; for grey color palettes; brewer palettes
e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;,
&quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;.
See details section for more information.  Can be also a numeric vector of
length(groups); in this case a basic color palette is created using the
function <a href="grDevices.html#topic+palette">palette</a>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_linetype">linetype</code></td>
<td>
<p>line types. Allowed values includes i) &quot;strata&quot; for changing
linetypes by strata (i.e. groups); ii) a numeric vector (e.g., c(1, 2)) or a
character vector c(&quot;solid&quot;, &quot;dashed&quot;).</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_break.time.by">break.time.by</code></td>
<td>
<p>numeric value controlling time axis breaks. Default value
is NULL.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_break.x.by">break.x.by</code></td>
<td>
<p>alias of break.time.by. Numeric value controlling x axis
breaks. Default value is NULL.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_break.y.by">break.y.by</code></td>
<td>
<p>same as break.x.by but for y axis.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_conf.int">conf.int</code></td>
<td>
<p>logical value. If TRUE, plots confidence interval.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_conf.int.fill">conf.int.fill</code></td>
<td>
<p>fill color to be used for confidence interval.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_conf.int.style">conf.int.style</code></td>
<td>
<p>confidence interval style. Allowed values include
c(&quot;ribbon&quot;, &quot;step&quot;).</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_conf.int.alpha">conf.int.alpha</code></td>
<td>
<p>numeric value specifying fill color transparency. Value
should be in [0, 1], where 0 is full transparency and 1 is no transparency.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_censor">censor</code></td>
<td>
<p>logical value. If TRUE, censors will be drawn.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_censor.shape">censor.shape</code></td>
<td>
<p>character or numeric value specifying the point shape of
censors. Default value is &quot;+&quot; (3), a sensible choice is &quot;|&quot; (124).</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_censor.size">censor.size</code></td>
<td>
<p>numveric value specifying the point size of censors.
Default is 4.5.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_pval">pval</code></td>
<td>
<p>logical value, a numeric or a string. If logical and TRUE, the
p-value is added on the plot. If numeric, than the computet p-value is
substituted with the one passed with this parameter. If character, then the
customized string appears on the plot. See examples - Example 3.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_pval.size">pval.size</code></td>
<td>
<p>numeric value specifying the p-value text size. Default is 5.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_pval.coord">pval.coord</code></td>
<td>
<p>numeric vector, of length 2, specifying the x and y
coordinates of the p-value. Default values are NULL.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_title">title</code>, <code id="ggsurvplot_arguments_+3A_xlab">xlab</code>, <code id="ggsurvplot_arguments_+3A_ylab">ylab</code></td>
<td>
<p>main title and axis labels</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_xlim">xlim</code>, <code id="ggsurvplot_arguments_+3A_ylim">ylim</code></td>
<td>
<p>x and y axis limits e.g. xlim = c(0, 1000), ylim = c(0, 1).</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_axes.offset">axes.offset</code></td>
<td>
<p>logical value. Default is TRUE. If FALSE, set the plot axes
to start at the origin.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_legend">legend</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). Default is &quot;top&quot; side position.
to remove the legend use legend = &quot;none&quot;. Legend position can be also
specified using a numeric vector c(x, y); see details section.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels. Used to replace
the names of the strata from the fit. Should be given in the same order as
those strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_risk.table">risk.table</code></td>
<td>
<p>Allowed values include: </p>
 <ul>
<li><p> TRUE or FALSE
specifying whether to show or not the risk table. Default is FALSE. </p>
</li>
<li>
<p>&quot;absolute&quot; or &quot;percentage&quot;. Shows the <b>absolute number</b> and the
<b>percentage</b> of subjects at risk by time, respectively. </p>
</li>
<li><p> &quot;abs_pct&quot;
to show both absolute number and percentage. </p>
</li>
<li><p> &quot;nrisk_cumcensor&quot; and
&quot;nrisk_cumevents&quot;. Show the number at risk and, the cumulative number of
censoring and events, respectively. </p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_risk.table.title">risk.table.title</code></td>
<td>
<p>The title to be used for the risk table.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_risk.table.pos">risk.table.pos</code></td>
<td>
<p>character vector specifying the risk table position.
Allowed options are one of c(&quot;out&quot;, &quot;in&quot;) indicating 'outside' or 'inside'
the main plot, respectively. Default value is &quot;out&quot;.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_risk.table.col">risk.table.col</code></td>
<td>
<p>same as tables.col but for risk table only.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_risk.table.fontsize">risk.table.fontsize</code>, <code id="ggsurvplot_arguments_+3A_fontsize">fontsize</code></td>
<td>
<p>font size to be used for the risk table
and the cumulative events table.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_risk.table.y.text">risk.table.y.text</code></td>
<td>
<p>logical. Default is TRUE. If FALSE, risk table y axis
tick labels will be hidden.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_risk.table.y.text.col">risk.table.y.text.col</code></td>
<td>
<p>logical. Default value is FALSE. If TRUE, risk
table tick labels will be colored by strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_tables.height">tables.height</code></td>
<td>
<p>numeric value (in [0 - 1]) specifying the general height
of all tables under the main survival plot.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_tables.y.text">tables.y.text</code></td>
<td>
<p>logical. Default is TRUE. If FALSE, the y axis tick
labels of tables will be hidden.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_tables.y.text.col">tables.y.text.col</code></td>
<td>
<p>logical. Default value is FALSE. If TRUE, tables tick
labels will be colored by strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_tables.col">tables.col</code></td>
<td>
<p>color to be used for all tables under the main plot. Default
value is &quot;black&quot;. If you want to color by strata (i.e. groups), use
tables.col = &quot;strata&quot;.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_tables.theme">tables.theme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>. Note that, <code>tables.theme</code> is incremental to <code>ggtheme</code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_risk.table.height">risk.table.height</code></td>
<td>
<p>the height of the risk table on the grid. Increase
the value when you have many strata. Default is 0.25. Ignored when
risk.table = FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_surv.plot.height">surv.plot.height</code></td>
<td>
<p>the height of the survival plot on the grid. Default
is 0.75. Ignored when risk.table = FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_ncensor.plot">ncensor.plot</code></td>
<td>
<p>logical value. If TRUE, the number of censored subjects at
time t is plotted. Default is FALSE. Ignored when cumcensor = TRUE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_ncensor.plot.title">ncensor.plot.title</code></td>
<td>
<p>The title to be used for the censor plot. Used when
<code>ncensor.plot = TRUE</code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_ncensor.plot.height">ncensor.plot.height</code></td>
<td>
<p>The height of the censor plot. Used when
<code>ncensor.plot = TRUE</code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumevents">cumevents</code></td>
<td>
<p>logical value specifying whether to show or not the table of
the cumulative number of events. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumevents.title">cumevents.title</code></td>
<td>
<p>The title to be used for the cumulative events table.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumevents.col">cumevents.col</code></td>
<td>
<p>same as tables.col but for the cumulative events table
only.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumevents.y.text">cumevents.y.text</code></td>
<td>
<p>logical. Default is TRUE. If FALSE, the y axis tick
labels of the cumulative events table  will be hidden.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumevents.y.text.col">cumevents.y.text.col</code></td>
<td>
<p>logical. Default value is FALSE. If TRUE, the y
tick labels of the cumulative events will be colored by strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumevents.height">cumevents.height</code></td>
<td>
<p>the height of the cumulative events table on the grid.
Default is 0.25. Ignored when cumevents = FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumcensor">cumcensor</code></td>
<td>
<p>logical value specifying whether to show or not the table of
the cumulative number of censoring. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumcensor.title">cumcensor.title</code></td>
<td>
<p>The title to be used for the cumcensor table.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumcensor.col">cumcensor.col</code></td>
<td>
<p>same as tables.col but for cumcensor table only.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumcensor.y.text">cumcensor.y.text</code></td>
<td>
<p>logical. Default is TRUE. If FALSE, the y axis tick
labels of the cumcensor table will be hidden.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumcensor.y.text.col">cumcensor.y.text.col</code></td>
<td>
<p>logical. Default value is FALSE. If TRUE, the y
tick labels of the cumcensor will be colored by strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_cumcensor.height">cumcensor.height</code></td>
<td>
<p>the height of the cumcensor table on the grid. Default
is 0.25. Ignored when cumcensor = FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_surv.median.line">surv.median.line</code></td>
<td>
<p>character vector for drawing a horizontal/vertical
line at median survival. Allowed values include one of c(&quot;none&quot;, &quot;hv&quot;, &quot;h&quot;,
&quot;v&quot;). v: vertical, h:horizontal.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_...">...</code></td>
<td>
<p>other arguments to be passed i) to ggplot2 geom_*() functions such
as linetype, size, ii) or to the function <a href="ggpubr.html#topic+ggpar">ggpar</a>() for
customizing the plots. See details section.</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_log.rank.weights">log.rank.weights</code></td>
<td>
<p>The name for the type of weights to be used in
computing the p-value for log-rank test. By default <code>survdiff</code> is used
to calculate regular log-rank test (with weights == 1). A user can specify
<code>"1", "n", "sqrtN", "S1", "S2", "FH"</code> to use weights specified in
<a href="survMisc.html#topic+comp">comp</a>, so that weight correspond to the test as : 1 -
log-rank, n - Gehan-Breslow (generalized Wilcoxon), sqrtN - Tarone-Ware, S1
- Peto-Peto's modified survival estimate, S2 - modified Peto-Peto (by
Andersen), FH - Fleming-Harrington(p=1, q=1).</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_pval.method">pval.method</code></td>
<td>
<p>whether to add a text with the test name used for
calculating the pvalue, that corresponds to survival curves' comparison -
used only when <code>pval=TRUE</code></p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_pval.method.size">pval.method.size</code></td>
<td>
<p>the same as <code>pval.size</code> but for displaying
<code>log.rank.weights</code> name</p>
</td></tr>
<tr><td><code id="ggsurvplot_arguments_+3A_pval.method.coord">pval.method.coord</code></td>
<td>
<p>the same as <code>pval.coord</code> but for displaying
<code>log.rank.weights</code> name</p>
</td></tr>
</table>

<hr>
<h2 id='ggsurvplot_combine'>Combine a List of Survfit Objects on the Same Plot</h2><span id='topic+ggsurvplot_combine'></span>

<h3>Description</h3>

<p>Combine multiple survfit objects on the same plot. For example,
one might wish to plot progression free survival and overall survival on
the same graph (and also stratified by treatment assignment).
<code>ggsurvplot_combine()</code> provides an extension to the
<code><a href="#topic+ggsurvplot">ggsurvplot</a>()</code> function for doing that.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsurvplot_combine(
  fit,
  data,
  risk.table = FALSE,
  risk.table.pos = c("out", "in"),
  cumevents = FALSE,
  cumcensor = FALSE,
  tables.col = "black",
  tables.y.text = TRUE,
  tables.y.text.col = TRUE,
  ggtheme = theme_survminer(),
  tables.theme = ggtheme,
  keep.data = FALSE,
  risk.table.y.text = tables.y.text,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvplot_combine_+3A_fit">fit</code></td>
<td>
<p>a named list of survfit objects.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_data">data</code></td>
<td>
<p>the data frame used to compute survival curves.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_risk.table">risk.table</code></td>
<td>
<p>Allowed values include: </p>
 <ul>
<li><p> TRUE or FALSE
specifying whether to show or not the risk table. Default is FALSE. </p>
</li>
<li>
<p>&quot;absolute&quot; or &quot;percentage&quot;. Shows the <b>absolute number</b> and the
<b>percentage</b> of subjects at risk by time, respectively. </p>
</li>
<li><p> &quot;abs_pct&quot;
to show both absolute number and percentage. </p>
</li>
<li><p> &quot;nrisk_cumcensor&quot; and
&quot;nrisk_cumevents&quot;. Show the number at risk and, the cumulative number of
censoring and events, respectively. </p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_risk.table.pos">risk.table.pos</code></td>
<td>
<p>character vector specifying the risk table position.
Allowed options are one of c(&quot;out&quot;, &quot;in&quot;) indicating 'outside' or 'inside'
the main plot, respectively. Default value is &quot;out&quot;.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_cumevents">cumevents</code></td>
<td>
<p>logical value specifying whether to show or not the table of
the cumulative number of events. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_cumcensor">cumcensor</code></td>
<td>
<p>logical value specifying whether to show or not the table of
the cumulative number of censoring. Default is FALSE.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_tables.col">tables.col</code></td>
<td>
<p>color to be used for all tables under the main plot. Default
value is &quot;black&quot;. If you want to color by strata (i.e. groups), use
tables.col = &quot;strata&quot;.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_tables.y.text">tables.y.text</code></td>
<td>
<p>logical. Default is TRUE. If FALSE, the y axis tick
labels of tables will be hidden.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_tables.y.text.col">tables.y.text.col</code></td>
<td>
<p>logical. Default value is FALSE. If TRUE, tables tick
labels will be colored by strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_tables.theme">tables.theme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>. Note that, <code>tables.theme</code> is incremental to <code>ggtheme</code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_keep.data">keep.data</code></td>
<td>
<p>logical value specifying whether the plot data frame should be kept in the result.
Setting these to FALSE (default) can give much smaller results and hence even save memory allocation time.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_risk.table.y.text">risk.table.y.text</code></td>
<td>
<p>logical. Default is TRUE. If FALSE, risk table y axis
tick labels will be hidden.</p>
</td></tr>
<tr><td><code id="ggsurvplot_combine_+3A_...">...</code></td>
<td>
<p>other arguments to pass to the <code><a href="#topic+ggsurvplot">ggsurvplot</a>()</code> function.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)
# Create a demo data set
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
 set.seed(123)
 demo.data &lt;- data.frame(
   os.time = colon$time,
   os.status = colon$status,
   pfs.time = sample(colon$time),
   pfs.status = colon$status,
   sex = colon$sex, rx = colon$rx, adhere = colon$adhere
 )

# Ex1: Combine null models
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
 # Fit
 pfs &lt;- survfit( Surv(pfs.time, pfs.status) ~ 1, data = demo.data)
 os &lt;- survfit( Surv(os.time, os.status) ~ 1, data = demo.data)
 # Combine on the same plot
 fit &lt;- list(PFS = pfs, OS = os)
 ggsurvplot_combine(fit, demo.data)

# Combine survival curves stratified by treatment assignment rx
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# Fit
pfs &lt;- survfit( Surv(pfs.time, pfs.status) ~ rx, data = demo.data)
os &lt;- survfit( Surv(os.time, os.status) ~ rx, data = demo.data)
# Combine on the same plot
fit &lt;- list(PFS = pfs, OS = os)
ggsurvplot_combine(fit, demo.data)

</code></pre>

<hr>
<h2 id='ggsurvplot_df'>Plot Survival Curves from Survival Summary Data Frame</h2><span id='topic+ggsurvplot_df'></span>

<h3>Description</h3>

<p>An extension to <a href="#topic+ggsurvplot">ggsurvplot</a>() to plot survival curves from
any data frame containing the summary of survival curves as returned the
<a href="#topic+surv_summary">surv_summary</a>() function.
</p>
<p>Might be useful for a user who wants
to use <a href="#topic+ggsurvplot">ggsurvplot</a> for visualizing survival curves computed by another
method than the standard <a href="survival.html#topic+survfit.formula">survfit.formula</a> function. In this
case, the user has just to provide the data frame containing the summary of
the survival analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsurvplot_df(
  fit,
  fun = NULL,
  color = NULL,
  palette = NULL,
  linetype = 1,
  break.x.by = NULL,
  break.time.by = NULL,
  break.y.by = NULL,
  surv.scale = c("default", "percent"),
  surv.geom = geom_step,
  xscale = 1,
  conf.int = FALSE,
  conf.int.fill = "gray",
  conf.int.style = "ribbon",
  conf.int.alpha = 0.3,
  censor = TRUE,
  censor.shape = "+",
  censor.size = 4.5,
  title = NULL,
  xlab = "Time",
  ylab = "Survival probability",
  xlim = NULL,
  ylim = NULL,
  axes.offset = TRUE,
  legend = c("top", "bottom", "left", "right", "none"),
  legend.title = "Strata",
  legend.labs = NULL,
  ggtheme = theme_survminer(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvplot_df_+3A_fit">fit</code></td>
<td>
<p>a data frame as returned by surv_summary. Should contains at least
the following columns: </p>
 <ul>
<li><p> time: survival time </p>
</li>
<li><p> surv:
survival probability </p>
</li>
<li><p> strata: grouping variables </p>
</li>
<li><p> n.censor: number
of censors </p>
</li>
<li><p> upper: upper end of confidence interval </p>
</li>
<li><p> lower: lower
end of confidence interval </p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_fun">fun</code></td>
<td>
<p>an arbitrary function defining a transformation of the survival
curve.  Often used transformations can be specified with a character
argument: &quot;event&quot; plots cumulative events (f(y) = 1-y), &quot;cumhaz&quot; plots the
cumulative hazard function (f(y) = -log(y)), and &quot;pct&quot; for survival
probability in percentage.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_color">color</code></td>
<td>
<p>color to be used for the survival curves. </p>
 <ul>
<li><p> If the
number of strata/group (n.strata) = 1, the expected value is the color name.
For example color = &quot;blue&quot;. </p>
</li>
<li><p> If n.strata &gt; 1, the expected value is the
grouping variable name. By default, survival curves are colored by strata
using the argument color = &quot;strata&quot;,  but you can also color survival curves
by any other grouping variables used to fit the survival curves. In this
case, it's possible to specify a custom color palette by using the argument
palette.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used. Allowed values include &quot;hue&quot; for
the default hue color scale; &quot;grey&quot; for grey color palettes; brewer palettes
e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;,
&quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;.
See details section for more information.  Can be also a numeric vector of
length(groups); in this case a basic color palette is created using the
function <a href="grDevices.html#topic+palette">palette</a>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_linetype">linetype</code></td>
<td>
<p>line types. Allowed values includes i) &quot;strata&quot; for changing
linetypes by strata (i.e. groups); ii) a numeric vector (e.g., c(1, 2)) or a
character vector c(&quot;solid&quot;, &quot;dashed&quot;).</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_break.x.by">break.x.by</code></td>
<td>
<p>alias of break.time.by. Numeric value controlling x axis
breaks. Default value is NULL.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_break.time.by">break.time.by</code></td>
<td>
<p>numeric value controlling time axis breaks. Default value
is NULL.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_break.y.by">break.y.by</code></td>
<td>
<p>same as break.x.by but for y axis.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_surv.scale">surv.scale</code></td>
<td>
<p>scale transformation of survival curves. Allowed values are
&quot;default&quot; or &quot;percent&quot;.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_surv.geom">surv.geom</code></td>
<td>
<p>survival curve style. Is the survival curve entered a step
function (<a href="ggplot2.html#topic+geom_step">geom_step</a>) or a smooth function (<a href="ggplot2.html#topic+geom_line">geom_line</a>).</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_xscale">xscale</code></td>
<td>
<p>numeric or character value specifying x-axis scale. </p>

<ul>
<li><p> If numeric, the value is used to divide the labels on the x axis. For
example, a value of 365.25 will give labels in years instead of the original
days. </p>
</li>
<li><p> If character, allowed options include one of c(&quot;d_m&quot;, &quot;d_y&quot;,
&quot;m_d&quot;, &quot;m_y&quot;, &quot;y_d&quot;, &quot;y_m&quot;), where d = days, m = months and y = years. For
example, xscale = &quot;d_m&quot; will transform labels from days to months; xscale =
&quot;m_y&quot;, will transform labels from months to years.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_conf.int">conf.int</code></td>
<td>
<p>logical value. If TRUE, plots confidence interval.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_conf.int.fill">conf.int.fill</code></td>
<td>
<p>fill color to be used for confidence interval.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_conf.int.style">conf.int.style</code></td>
<td>
<p>confidence interval style. Allowed values include
c(&quot;ribbon&quot;, &quot;step&quot;).</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_conf.int.alpha">conf.int.alpha</code></td>
<td>
<p>numeric value specifying fill color transparency. Value
should be in [0, 1], where 0 is full transparency and 1 is no transparency.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_censor">censor</code></td>
<td>
<p>logical value. If TRUE, censors will be drawn.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_censor.shape">censor.shape</code></td>
<td>
<p>character or numeric value specifying the point shape of
censors. Default value is &quot;+&quot; (3), a sensible choice is &quot;|&quot; (124).</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_censor.size">censor.size</code></td>
<td>
<p>numveric value specifying the point size of censors.
Default is 4.5.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_title">title</code></td>
<td>
<p>main title and axis labels</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_xlab">xlab</code></td>
<td>
<p>main title and axis labels</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_ylab">ylab</code></td>
<td>
<p>main title and axis labels</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_xlim">xlim</code></td>
<td>
<p>x and y axis limits e.g. xlim = c(0, 1000), ylim = c(0, 1).</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_ylim">ylim</code></td>
<td>
<p>x and y axis limits e.g. xlim = c(0, 1000), ylim = c(0, 1).</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_axes.offset">axes.offset</code></td>
<td>
<p>logical value. Default is TRUE. If FALSE, set the plot axes
to start at the origin.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_legend">legend</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). Default is &quot;top&quot; side position.
to remove the legend use legend = &quot;none&quot;. Legend position can be also
specified using a numeric vector c(x, y); see details section.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels. Used to replace
the names of the strata from the fit. Should be given in the same order as
those strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="#topic+theme_survminer">theme_survminer</a>. Allowed values include ggplot2 official themes: see
<code><a href="ggplot2.html#topic+theme">theme</a></code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_df_+3A_...">...</code></td>
<td>
<p>other arguments to be passed i) to ggplot2 geom_*() functions such
as linetype, size, ii) or to the function <a href="ggpubr.html#topic+ggpar">ggpar</a>() for
customizing the plots. See details section.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)

# Fit survival curves
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit1 &lt;- survfit( Surv(time, status) ~ 1, data = colon)
fit2 &lt;- survfit( Surv(time, status) ~ adhere, data = colon)

# Summary
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
head(surv_summary(fit1, colon))

head(surv_summary(fit2, colon))

# Visualize
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
ggsurvplot_df(surv_summary(fit1, colon))

ggsurvplot_df(surv_summary(fit2, colon), conf.int = TRUE,
             legend.title = "Adhere", legend.labs = c("0", "1"))

# Kaplan-Meier estimate
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
out_km &lt;- survfit(Surv(time, status) ~ 1, data = lung)

# Weibull model
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
wb &lt;- survreg(Surv(time, status) ~ 1, data = lung)
s &lt;- seq(.01, .99, by = .01)
t &lt;- predict(wb, type = "quantile", p = s, newdata = lung[1, ])
out_wb &lt;- data.frame(time = t, surv = 1 - s, upper = NA, lower = NA, std.err = NA)

# plot both
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
p_km &lt;- ggsurvplot(out_km, conf.int = FALSE)
p_wb &lt;- ggsurvplot(out_wb, conf.int = FALSE, surv.geom = geom_line)

p_km
p_wb
p_km$plot + geom_line(data = out_wb, aes(x = time, y = surv))

</code></pre>

<hr>
<h2 id='ggsurvplot_facet'>Facet Survival Curves into Multiple Panels</h2><span id='topic+ggsurvplot_facet'></span>

<h3>Description</h3>

<p>Draw multi-panel survival curves of a data set grouped by one or
two variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsurvplot_facet(
  fit,
  data,
  facet.by,
  color = NULL,
  palette = NULL,
  legend.labs = NULL,
  pval = FALSE,
  pval.method = FALSE,
  pval.coord = NULL,
  pval.method.coord = NULL,
  nrow = NULL,
  ncol = NULL,
  scales = "fixed",
  short.panel.labs = FALSE,
  panel.labs = NULL,
  panel.labs.background = list(color = NULL, fill = NULL),
  panel.labs.font = list(face = NULL, color = NULL, size = NULL, angle = NULL),
  panel.labs.font.x = panel.labs.font,
  panel.labs.font.y = panel.labs.font,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvplot_facet_+3A_fit">fit</code></td>
<td>
<p>an object of class survfit.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_data">data</code></td>
<td>
<p>a dataset used to fit survival curves. If not supplied then data
will be extracted from 'fit' object.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_facet.by">facet.by</code></td>
<td>
<p>character vector, of length 1 or 2, specifying grouping
variables for faceting the plot. Should be in the data.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_color">color</code></td>
<td>
<p>color to be used for the survival curves. </p>
 <ul>
<li><p> If the
number of strata/group (n.strata) = 1, the expected value is the color name.
For example color = &quot;blue&quot;. </p>
</li>
<li><p> If n.strata &gt; 1, the expected value is the
grouping variable name. By default, survival curves are colored by strata
using the argument color = &quot;strata&quot;,  but you can also color survival curves
by any other grouping variables used to fit the survival curves. In this
case, it's possible to specify a custom color palette by using the argument
palette.</p>
</li></ul>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used. Allowed values include &quot;hue&quot; for
the default hue color scale; &quot;grey&quot; for grey color palettes; brewer palettes
e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;,
&quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and
&quot;rickandmorty&quot;.
See details section for more information.  Can be also a numeric vector of
length(groups); in this case a basic color palette is created using the
function <a href="grDevices.html#topic+palette">palette</a>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels. Used to replace
the names of the strata from the fit. Should be given in the same order as
those strata.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_pval">pval</code></td>
<td>
<p>logical value, a numeric or a string. If logical and TRUE, the
p-value is added on the plot. If numeric, than the computet p-value is
substituted with the one passed with this parameter. If character, then the
customized string appears on the plot. See examples - Example 3.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_pval.method">pval.method</code></td>
<td>
<p>whether to add a text with the test name used for
calculating the pvalue, that corresponds to survival curves' comparison -
used only when <code>pval=TRUE</code></p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_pval.coord">pval.coord</code></td>
<td>
<p>numeric vector, of length 2, specifying the x and y
coordinates of the p-value. Default values are NULL.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_pval.method.coord">pval.method.coord</code></td>
<td>
<p>the same as <code>pval.coord</code> but for displaying
<code>log.rank.weights</code> name</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_nrow">nrow</code>, <code id="ggsurvplot_facet_+3A_ncol">ncol</code></td>
<td>
<p>Number of rows and columns in the pannel. Used only when the
data is faceted by one grouping variable.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_scales">scales</code></td>
<td>
<p>should axis scales of panels be fixed (&quot;fixed&quot;, the default),
free (&quot;free&quot;), or free in one dimension (&quot;free_x&quot;, &quot;free_y&quot;).</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_short.panel.labs">short.panel.labs</code></td>
<td>
<p>logical value. Default is FALSE. If TRUE, create short
labels for panels by omitting variable names; in other words panels will be
labelled only by variable grouping levels.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_panel.labs">panel.labs</code></td>
<td>
<p>a list of one or two character vectors to modify facet label
text. For example, panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;)) specifies
the labels for the &quot;sex&quot; variable. For two grouping variables, you can use
for example panel.labs = list(sex = c(&quot;Male&quot;, &quot;Female&quot;), rx = c(&quot;Obs&quot;,
&quot;Lev&quot;, &quot;Lev2&quot;) ).</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_panel.labs.background">panel.labs.background</code></td>
<td>
<p>a list to customize the background of panel
labels. Should contain the combination of the following elements: </p>

<ul>
<li> <p><code>color, linetype, size</code>: background line color, type and size
</p>
</li>
<li> <p><code>fill</code>: background fill color. </p>
</li></ul>
<p> For example,
panel.labs.background = list(color = &quot;blue&quot;, fill = &quot;pink&quot;).</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_panel.labs.font">panel.labs.font</code></td>
<td>
<p>a list of aestheics indicating the size (e.g.: 14), the
face/style (e.g.: &quot;plain&quot;, &quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and the color
(e.g.: &quot;red&quot;) and the orientation angle (e.g.: 45) of panel labels.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_panel.labs.font.x">panel.labs.font.x</code>, <code id="ggsurvplot_facet_+3A_panel.labs.font.y">panel.labs.font.y</code></td>
<td>
<p>same as panel.labs.font but for x
and y direction, respectively.</p>
</td></tr>
<tr><td><code id="ggsurvplot_facet_+3A_...">...</code></td>
<td>
<p>other arguments to pass to the function <code><a href="#topic+ggsurvplot">ggsurvplot</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(survival)

# Facet by one grouping variables: rx
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit &lt;- survfit( Surv(time, status) ~ sex, data = colon )
ggsurvplot_facet(fit, colon, facet.by = "rx",
                palette = "jco", pval = TRUE)

# Facet by two grouping variables: rx and adhere
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
ggsurvplot_facet(fit, colon, facet.by = c("rx", "adhere"),
                palette = "jco", pval = TRUE)


# Another fit
#::::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit2 &lt;- survfit( Surv(time, status) ~ sex + rx, data = colon )
ggsurvplot_facet(fit2, colon, facet.by = "adhere",
                palette = "jco", pval = TRUE)

</code></pre>

<hr>
<h2 id='ggsurvplot_group_by'>Survival Curves of Grouped Data sets</h2><span id='topic+ggsurvplot_group_by'></span>

<h3>Description</h3>

<p>Survival curves of grouped data sets by one or two
variables.
</p>
<p>Survival analysis are often done on subsets defined by
variables in the dataset. For example, assume that we have a cohort of
patients with a large number of  clinicopathological and molecular
covariates, including survival data, TP53 mutation status and the patients'
sex (Male or Female).
</p>
<p>One might be also interested in comparing the
survival curves of Male and Female after grouping (or splitting ) the data
by TP53 mutation status.
</p>
<p><code>ggsurvplot_group_by</code>() provides a
convenient solution to create a multiple <a href="#topic+ggsurvplot">ggsurvplot</a> of a data set
grouped by one or two variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsurvplot_group_by(fit, data, group.by, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvplot_group_by_+3A_fit">fit</code></td>
<td>
<p>a survfit object.</p>
</td></tr>
<tr><td><code id="ggsurvplot_group_by_+3A_data">data</code></td>
<td>
<p>a data frame used to fit survival curves.</p>
</td></tr>
<tr><td><code id="ggsurvplot_group_by_+3A_group.by">group.by</code></td>
<td>
<p>a character vector containing the name of grouping variables. Should be of length &lt;= 2.</p>
</td></tr>
<tr><td><code id="ggsurvplot_group_by_+3A_...">...</code></td>
<td>
<p>... other arguments passed to the core function
<code><a href="#topic+ggsurvplot">ggsurvplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ggsurvplot_group_by</code>() works as follow:
</p>

<ol>
<li><p> Create a grouped data sets using the function <code><a href="#topic+surv_group_by">surv_group_by</a>()</code>, &ndash;&gt; list of data sets
</p>
</li>
<li><p> Map <code><a href="#topic+surv_fit">surv_fit</a>()</code> to each nested data &ndash;&gt; Returns a list of survfit objects
</p>
</li>
<li><p> Map <code><a href="#topic+ggsurvplot">ggsurvplot</a>()</code> to each survfit object &ndash;&gt; list of survfit ggsurvplots
</p>
</li></ol>

<p>One can (optionally) arrange the list of ggsurvplots using <code><a href="#topic+arrange_ggsurvplots">arrange_ggsurvplots</a>()</code>
</p>


<h3>Value</h3>

<p>Retuns a list of ggsurvplots.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Fit survival curves
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
library(survival)
fit &lt;- survfit( Surv(time, status) ~ sex, data = colon )

# Visualize: grouped by treatment rx
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
ggsurv.list &lt;- ggsurvplot_group_by(fit, colon, group.by = "rx", risk.table = TRUE,
                                 pval = TRUE, conf.int = TRUE, palette = "jco")
names(ggsurv.list)


# Visualize: grouped by treatment rx and adhere
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
ggsurv.list &lt;- ggsurvplot_group_by(fit, colon, group.by = c("rx", "adhere"),
                                 risk.table = TRUE,
                                 pval = TRUE, conf.int = TRUE, palette = "jco")

names(ggsurv.list)
</code></pre>

<hr>
<h2 id='ggsurvplot_list'>Plot a List of Survfit Objects</h2><span id='topic+ggsurvplot_list'></span>

<h3>Description</h3>

<p>Take a list of survfit objects and produce a list of
<code>ggsurvplots</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggsurvplot_list(
  fit,
  data,
  title = NULL,
  legend.labs = NULL,
  legend.title = "Strata",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggsurvplot_list_+3A_fit">fit</code></td>
<td>
<p>a list of survfit objects.</p>
</td></tr>
<tr><td><code id="ggsurvplot_list_+3A_data">data</code></td>
<td>
<p>data used to fit survival curves. Can be also a list of same
length than <code>fit</code>.</p>
</td></tr>
<tr><td><code id="ggsurvplot_list_+3A_title">title</code></td>
<td>
<p>title of the plot. Can be a character vector or a list of titles
of same length than <code>fit</code>. If <code>title</code> is not specified and
<code>fit</code> is a named list, then the names of fit list are used as title.</p>
</td></tr>
<tr><td><code id="ggsurvplot_list_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels. Used to replace
the names of the strata from the fit. Should be given in the same order as
those strata. Can be a list when <code>fit</code> is a list.</p>
</td></tr>
<tr><td><code id="ggsurvplot_list_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title for each plot. Can be a character vector or a
list of titles of same length than fit.</p>
</td></tr>
<tr><td><code id="ggsurvplot_list_+3A_...">...</code></td>
<td>
<p>other arguments passed to the core function
<code><a href="#topic+ggsurvplot">ggsurvplot</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of ggsurvplots.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggsurvplot">ggsurvplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)

# Create a list of formulas
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
data(colon)
f1 &lt;- survfit(Surv(time, status) ~ adhere, data = colon)
f2 &lt;- survfit(Surv(time, status) ~ rx, data = colon)
fits &lt;- list(sex = f1, rx = f2)

# Visualize
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
legend.title &lt;- list("sex", "rx")
ggsurvplot_list(fits, colon, legend.title = legend.title)

</code></pre>

<hr>
<h2 id='myeloma'>Multiple Myeloma Data</h2><span id='topic+myeloma'></span>

<h3>Description</h3>

<p>Multiple Myeloma data extracted from publicly available gene
expression data (GEO Id: GSE4581).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("myeloma")
</code></pre>


<h3>Format</h3>

<p>A data frame with 256 rows and 12 columns. </p>

<dl>
<dt><code>molecular_group</code></dt><dd><p>Patients' molecular subgroups</p>
</dd>
<dt><code>chr1q21_status</code></dt><dd><p>Amplification status of the chromosome
1q21</p>
</dd>
<dt><code>treatment</code></dt><dd><p>treatment</p>
</dd> <dt><code>event</code></dt><dd><p>survival status 0 =
alive, 1 = dead</p>
</dd> <dt><code>time</code></dt><dd><p>Survival time in months</p>
</dd>
<dt><code>CCND1</code></dt><dd><p>Gene expression</p>
</dd> <dt><code>CRIM1</code></dt><dd><p>Gene expression</p>
</dd>
<dt><code>DEPDC1</code></dt><dd><p>Gene expression</p>
</dd> <dt><code>IRF4</code></dt><dd><p>Gene expression</p>
</dd>
<dt><code>TP53</code></dt><dd><p>Gene expression</p>
</dd> <dt><code>WHSC1</code></dt><dd><p>Gene expression</p>
</dd> </dl>

<p>The remaining columns (CCND1, CRIM1, DEPDC1, IRF4, TP53, WHSC1) correspond to
the gene expression level of specified genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(myeloma)
head(myeloma)

</code></pre>

<hr>
<h2 id='pairwise_survdiff'>Multiple Comparisons of Survival Curves</h2><span id='topic+pairwise_survdiff'></span>

<h3>Description</h3>

<p>Calculate pairwise comparisons between group levels with
corrections for multiple testing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwise_survdiff(formula, data, p.adjust.method = "BH", na.action, rho = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwise_survdiff_+3A_formula">formula</code></td>
<td>
<p>a formula expression as for other survival models, of the form
Surv(time, status) ~ predictors.</p>
</td></tr>
<tr><td><code id="pairwise_survdiff_+3A_data">data</code></td>
<td>
<p>a data frame in which to interpret the variables occurring in the
formula.</p>
</td></tr>
<tr><td><code id="pairwise_survdiff_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for adjusting p values (see
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>). Allowed values include &quot;holm&quot;, &quot;hochberg&quot;,
&quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;. If you don't want to
adjust the p value (not recommended), use p.adjust.method = &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="pairwise_survdiff_+3A_na.action">na.action</code></td>
<td>
<p>a missing-data filter function. Default is
options()$na.action.</p>
</td></tr>
<tr><td><code id="pairwise_survdiff_+3A_rho">rho</code></td>
<td>
<p>a scalar parameter that controls the type of test. Allowed values
include 0 (for Log-Rank test) and 1 (for peto &amp; peto test).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class &quot;pairwise.htest&quot;, which is a list
containing the p values.
</p>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara, <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>See Also</h3>

<p>survival::survdiff
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)
library(survminer)
data(myeloma)

# Pairwise survdiff
res &lt;- pairwise_survdiff(Surv(time, event) ~ molecular_group,
     data = myeloma)
res

# Symbolic number coding
symnum(res$p.value, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 0.1, 1),
   symbols = c("****", "***", "**", "*", "+", " "),
   abbr.colnames = FALSE, na = "")


</code></pre>

<hr>
<h2 id='surv_cutpoint'>Determine the Optimal Cutpoint for Continuous Variables</h2><span id='topic+surv_cutpoint'></span><span id='topic+surv_categorize'></span><span id='topic+summary.surv_cutpoint'></span><span id='topic+print.surv_cutpoint'></span><span id='topic+plot.surv_cutpoint'></span><span id='topic+print.plot_surv_cutpoint'></span>

<h3>Description</h3>

<p>Determine the optimal cutpoint for one or multiple continuous
variables at once, using the maximally selected rank statistics from the
'maxstat' R package. This is an outcome-oriented  methods  providing  a
value of a cutpoint that correspond to the most significant relation with
outcome (here, survival).
</p>

<ul>
<li> <p><code>surv_cutpoint()</code>: Determine the optimal cutpoint for each variable using 'maxstat'.
</p>
</li>
<li> <p><code>surv_categorize()</code>: Divide each variable values based on the cutpoint returned by <code>surv_cutpoint()</code>.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>surv_cutpoint(
  data,
  time = "time",
  event = "event",
  variables,
  minprop = 0.1,
  progressbar = TRUE
)

surv_categorize(x, variables = NULL, labels = c("low", "high"))

## S3 method for class 'surv_cutpoint'
summary(object, ...)

## S3 method for class 'surv_cutpoint'
print(x, ...)

## S3 method for class 'surv_cutpoint'
plot(x, variables = NULL, ggtheme = theme_classic(), bins = 30, ...)

## S3 method for class 'plot_surv_cutpoint'
print(x, ..., newpage = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_cutpoint_+3A_data">data</code></td>
<td>
<p>a data frame containing survival information (time, event) and
continuous variables (e.g.: gene expression data).</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_time">time</code>, <code id="surv_cutpoint_+3A_event">event</code></td>
<td>
<p>column names containing time and event data, respectively.
Event values sould be 0 or 1.</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_variables">variables</code></td>
<td>
<p>a character vector containing the names of variables of
interest, for wich we want to estimate the optimal cutpoint.</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_minprop">minprop</code></td>
<td>
<p>the minimal proportion of observations per group.</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_progressbar">progressbar</code></td>
<td>
<p>logical value. If TRUE, show progress bar. Progressbar is
shown only, when the number of variables &gt; 5.</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_x">x</code>, <code id="surv_cutpoint_+3A_object">object</code></td>
<td>
<p>an object of class surv_cutpoint</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_labels">labels</code></td>
<td>
<p>labels for the levels of the resulting category.</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_...">...</code></td>
<td>
<p>other arguments. For plots, see ?ggpubr::ggpar</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is
<a href="ggplot2.html#topic+theme_classic">theme_classic</a>. Allowed values include ggplot2 official themes. see
?ggplot2::ggtheme.</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_bins">bins</code></td>
<td>
<p>Number of bins for histogram. Defaults to 30.</p>
</td></tr>
<tr><td><code id="surv_cutpoint_+3A_newpage">newpage</code></td>
<td>
<p>open a new page. See <code><a href="gridExtra.html#topic+grid.arrange">grid.arrange</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><b>surv_cutpoint()</b>: returns an object of class 'surv_cutpoint',
which is a list with the following components: </p>
 <ul>
<li><p> maxstat
results for each variable (see ?maxstat::maxstat) </p>
</li>
<li><p> cutpoint: a data
frame containing the optimal cutpoint of each variable. Rows are variable
names and columns are c(&quot;cutpoint&quot;, &quot;statistic&quot;). </p>
</li>
<li><p> data: a data frame
containing the survival data and the original data for the specified
variables. </p>
</li>
<li><p> minprop: the minimal proportion of observations per group.
</p>
</li>
<li><p> not_numeric: contains data for non-numeric variables, in the context
where the user provided categorical variable names in the argument
variables.</p>
</li></ul>
 <p> Methods defined for surv_cutpoint object are summary, print
and plot.
</p>
</li>
<li> <p><b>surv_categorize()</b>: returns an object of class
'surv_categorize', which is a data frame containing the survival data and
the categorized variables.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Alboukadel Kassambara, <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 0. Load some data
data(myeloma)
head(myeloma)

# 1. Determine the optimal cutpoint of variables
res.cut &lt;- surv_cutpoint(myeloma, time = "time", event = "event",
   variables = c("DEPDC1", "WHSC1", "CRIM1"))

summary(res.cut)

# 2. Plot cutpoint for DEPDC1
# palette = "npg" (nature publishing group), see ?ggpubr::ggpar
plot(res.cut, "DEPDC1", palette = "npg")

# 3. Categorize variables
res.cat &lt;- surv_categorize(res.cut)
head(res.cat)

# 4. Fit survival curves and visualize
library("survival")
fit &lt;- survfit(Surv(time, event) ~DEPDC1, data = res.cat)
ggsurvplot(fit, data = res.cat, risk.table = TRUE, conf.int = TRUE)

</code></pre>

<hr>
<h2 id='surv_fit'>Create Survival Curves</h2><span id='topic+surv_fit'></span>

<h3>Description</h3>

<p>Wrapper arround the standard <a href="survival.html#topic+survfit">survfit</a>() function to create
survival curves. Compared to the standard <a href="survival.html#topic+survfit">survfit</a>() function, it supports also:
</p>

<ul>
<li><p> a list of data sets and/or a list of formulas,
</p>
</li>
<li><p> a grouped data sets as generated by the function <a href="#topic+surv_group_by">surv_group_by</a>,
</p>
</li>
<li><p> group.by option
</p>
</li></ul>

<p>There are many cases, where this function might be useful:
</p>

<ul>
<li> <p><strong>Case 1: One formula and One data set</strong>.
Example: You want to fit the survival curves of one biomarker/gene in a given data set.
This is the same as the standard <a href="survival.html#topic+survfit">survfit</a>() function. Returns one survfit object.
</p>
</li>
<li> <p><strong>Case 2: List of formulas and One data set</strong>.
Example: You want to fit the survival curves of a list of biormarkers/genes in the same data set.
Returns a named list of survfit objects in the same order as formulas.
</p>
</li>
<li> <p><strong>Case 3: One formula and List of data sets</strong>.
Example: You want to fit survival curves of one biomarker/gene in multiple cohort of patients (colon, lung, breast).
Returns a named list of survfit objects in the same order as the data sets.
</p>
</li>
<li> <p><strong>Case 4: List of formulas and List of data sets</strong>.
Example: You want to fit survival curves of multiple biomarkers/genes in multiple cohort of patients (colon, lung, breast).
Each formula will be applied to each of the data set in the data list.
Returns a named list of survfit objects.
</p>
</li>
<li> <p><strong>Case 5: One formula and grouped data sets by one or two variables</strong>.
Example: One might like to plot the survival curves of patients
treated by drug A vs patients treated by drug B in a dataset grouped by TP53 and/or RAS mutations.
In this case use the argument <code>group.by</code>. Returns a named list of survfit objects.
</p>
</li>
<li> <p><strong>Case 6</strong>. In a rare case you might have a list of formulas and a list of data sets, and
you might want to <strong>apply each formula to the mathcing data set with the same index/position in the list</strong>.
For example formula1 is applied to data 1, formula2 is applied to data 2, and so on ...
In this case formula and data lists should have the same length and you should specify the argument match.fd = TRUE ( stands for match formula and data).
Returns a named list of survfit objects.
</p>
</li></ul>

<p>The output of the <code>surv_fit</code>() function can be directly handled by the following functions:
</p>

<ul>
<li> <p><a href="#topic+ggsurvplot">ggsurvplot</a>()
</p>
</li>
<li> <p><a href="#topic+surv_pvalue">surv_pvalue</a>()
</p>
</li>
<li> <p><a href="#topic+surv_median">surv_median</a>()
</p>
</li></ul>

<p>These functions return one element or a list of elements depending on the format of the input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_fit(formula, data, group.by = NULL, match.fd = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_fit_+3A_formula">formula</code></td>
<td>
<p>survival formula. See <a href="survival.html#topic+survfit.formula">survfit.formula</a>. Can be a list of formula. Named lists are recommended.</p>
</td></tr>
<tr><td><code id="surv_fit_+3A_data">data</code></td>
<td>
<p>a data frame in which to interpret the variables named in the formula.
Can be a list of data sets. Named lists are recommended.
Can be also a grouped dataset as generated by the function <a href="#topic+surv_group_by">surv_group_by</a>().</p>
</td></tr>
<tr><td><code id="surv_fit_+3A_group.by">group.by</code></td>
<td>
<p>a grouping variables to group the data set by.
A character vector containing the name of grouping variables. Should be of length &lt;= 2.</p>
</td></tr>
<tr><td><code id="surv_fit_+3A_match.fd">match.fd</code></td>
<td>
<p>logical value. Default is FALSE. Stands for &quot;match formula and data&quot;.
Useful only when you have a list of formulas and a list of data sets, and
you want to apply each formula to the matching data set with the same index/position in the list.
For example formula1 is applied to data 1, formula2 is applied to data 2, and so on ....
In this case use match.fd = TRUE.</p>
</td></tr>
<tr><td><code id="surv_fit_+3A_...">...</code></td>
<td>
<p>Other arguments passed to the <a href="survival.html#topic+survfit.formula">survfit.formula</a> function.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Returns an object of class survfit if one formula and one data set provided.
</p>
</li>
<li><p> Returns a named list of survfit objects when input is a list of formulas and/or data sets.
The same holds true when grouped data sets are provided or when the argument <code>group.by</code> is specified.
</p>

<ul>
<li><p> If the names of formula and data lists are available,
the names of the resulting survfit objects list are obtained by collapsing the names of formula and data lists.
</p>
</li>
<li><p> If the formula names are not available, the variables in the formulas are extracted and used to build the name of survfit object.
</p>
</li>
<li><p> In the case of grouped data sets, the names of survfit object list are obtained by
collapsing the levels of grouping variables and the names of variables in the survival curve formulas.
</p>
</li></ul>

</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library("survival")
library("magrittr")

# Case 1: One formula and One data set
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit &lt;- surv_fit(Surv(time, status) ~ sex,
               data = colon)
surv_pvalue(fit)


# Case 2: List of formulas and One data set.
#   - Different formulas are applied to the same data set
#   - Returns a (named) list of survfit objects
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# Create a named list of formulas
formulas &lt;- list(
 sex = Surv(time, status) ~ sex,
 rx = Surv(time, status) ~ rx
)

# Fit survival curves for each formula
fit &lt;- surv_fit(formulas, data = colon)
surv_pvalue(fit)

# Case 3: One formula and List of data sets
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit &lt;- surv_fit(Surv(time, status) ~ sex,
               data = list(colon, lung))
surv_pvalue(fit)


# Case 4: List of formulas and List of data sets
#  - Each formula is applied to each of the data in the data list
#  - argument: match.fd = FALSE
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

# Create two data sets
set.seed(123)
colon1 &lt;- dplyr::sample_frac(colon, 1/2)
set.seed(1234)
colon2 &lt;- dplyr::sample_frac(colon, 1/2)

# Create a named list of formulas
formula.list &lt;- list(
 sex = Surv(time, status) ~ sex,
 adhere = Surv(time, status) ~ adhere,
 rx = Surv(time, status) ~ rx
)

# Fit survival curves
fit &lt;- surv_fit(formula.list, data = list(colon1, colon2),
               match.fd = FALSE)
surv_pvalue(fit)


# Grouped survfit
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# - Group by the treatment "rx" and fit survival curves on each subset
# - Returns a list of survfit objects
fit &lt;- surv_fit(Surv(time, status) ~ sex,
               data = colon, group.by = "rx")

# Alternatively, do this
fit &lt;- colon %&gt;%
 surv_group_by("rx") %&gt;%
 surv_fit(Surv(time, status) ~ sex, data = .)

surv_pvalue(fit)

</code></pre>

<hr>
<h2 id='surv_group_by'>Create a Grouped Dataset for Survival Analysis</h2><span id='topic+surv_group_by'></span>

<h3>Description</h3>

<p>Split a data frame into multiple new data frames based on one or
two grouping variables. The <code>surv_group_by()</code> function takes an
existing data frame and converts it into a grouped data frame where
survival analysis are performed &quot;by group&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_group_by(data, grouping.vars)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_group_by_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="surv_group_by_+3A_grouping.vars">grouping.vars</code></td>
<td>
<p>a character vector containing the name of grouping
variables. Should be of length &lt;= 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>surv_group_by</code> which is a
<a href="tibble.html#topic+tibble">tibble</a> data frame with the following components: </p>

<ul>
<li><p> one column for each grouping variables. Contains the levels. </p>
</li>
<li><p> a
coumn named &quot;data&quot;, which is a named list of data subsets created by the
grouping variables. The list names are created by concatening the levels of
grouping variables. </p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>library("survival")
library("magrittr")

# Grouping by one variables: treatment "rx"
#::::::::::::::::::::::::::::::::::::::::::
grouped.d &lt;- colon %&gt;%
  surv_group_by("rx")

grouped.d # print

grouped.d$data # Access to the data

# Grouping by two variables
#::::::::::::::::::::::::::::::::::::::::::
grouped.d &lt;- colon %&gt;%
   surv_group_by(grouping.vars = c("rx", "adhere"))
   grouped.d

</code></pre>

<hr>
<h2 id='surv_median'>Median of Survival Curves</h2><span id='topic+surv_median'></span>

<h3>Description</h3>

<p>Returns the median survival with upper and lower confidence
limits for the median at 95% confidence levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_median(fit, combine = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_median_+3A_fit">fit</code></td>
<td>
<p>A survfit object. Can be also a list of survfit objects.</p>
</td></tr>
<tr><td><code id="surv_median_+3A_combine">combine</code></td>
<td>
<p>logical value. Used only when fit is a list of survfit objects.
If TRUE, combine the results for multiple fits.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns for each fit, a data frame with the following column:
</p>
 <ul>
<li><p> strata: strata/group names </p>
</li>
<li><p> median: median survival of
each group </p>
</li>
<li><p> lower: 95% lower confidence limit </p>
</li>
<li><p> upper: 95% upper
confidence limit </p>
</li></ul>
<p>Returns a list of data frames when the input is a
list of survfit objects. If combine = TRUE, results are combined into one single data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)

# Different survfits
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit.null &lt;- surv_fit(Surv(time, status) ~ 1, data = colon)

fit1 &lt;- surv_fit(Surv(time, status) ~ sex, data = colon)

fit2 &lt;- surv_fit(Surv(time, status) ~ adhere, data = colon)

fit.list &lt;- list(sex = fit1, adhere = fit2)

# Extract the median survival
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
surv_median(fit.null)

surv_median(fit2)

surv_median(fit.list)

surv_median(fit.list, combine = TRUE)

# Grouped survfit
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit.list2 &lt;- surv_fit(Surv(time, status) ~ sex, data = colon,
                     group.by = "rx")
surv_median(fit.list2)
</code></pre>

<hr>
<h2 id='surv_pvalue'>Compute P-value Comparing Survival Curves</h2><span id='topic+surv_pvalue'></span>

<h3>Description</h3>

<p>Compute p-value from survfit objects or parse it when provided by
the user. Survival curves are compared using the log-rank test (default).
Other methods can be specified using the argument <code>method</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_pvalue(
  fit,
  data = NULL,
  method = "survdiff",
  test.for.trend = FALSE,
  combine = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_pvalue_+3A_fit">fit</code></td>
<td>
<p>A survfit object. Can be also a list of survfit objects.</p>
</td></tr>
<tr><td><code id="surv_pvalue_+3A_data">data</code></td>
<td>
<p>data frame used to fit survival curves. Can be also a list of
data.</p>
</td></tr>
<tr><td><code id="surv_pvalue_+3A_method">method</code></td>
<td>
<p>method to compute survival curves. Default is &quot;survdiff&quot; (or
&quot;log-rank&quot;). Allowed values are one of: </p>
 <ul>
<li><p> &quot;survdiff&quot;,
log-rank; </p>
</li>
<li><p> &quot;1&quot;: log-rank, LR; &ndash;&gt; Regular log-rank test, sensitive to
detect late differences. </p>
</li>
<li><p> &quot;n&quot;: Gehan-Breslow (generalized Wilcoxon),
GB; &ndash;&gt; detect early differences. </p>
</li>
<li><p> &quot;sqrtN&quot;: Tarone-Ware, TW; &ndash;&gt; detect
early differences. </p>
</li>
<li><p> &quot;S1&quot;: Peto-Peto's modified survival estimate, PP;
&ndash;&gt; more robust than Tharone-Whare or Gehan-Breslow, detect early
differences </p>
</li>
<li><p> &quot;S2&quot;: modified Peto-Peto (by Andersen), mPP </p>
</li>
<li>
<p>&quot;FH_p=1_q=1&quot;: Fleming-Harrington(p=1, q=1),  FH </p>
</li></ul>
<p> To specify method, one can
use either the weights (e.g.: &quot;1&quot;, &quot;n&quot;, &quot;sqrtN&quot;, ...), or the full name
(&quot;log-rank&quot;, &quot;gehan-breslow&quot;, &quot;Peto-Peto&quot;, ...), or the acronyme LR, GB,
.... Case insensitive partial match is allowed.<br /><br /> To learn more about
the mathematical background behind the different log-rank weights, read the
following blog post on R-Addict:
<a href="http://r-addict.com/2017/02/09/Fancy-Survival-Plots.html">Comparing
(Fancy) Survival Curves with Weighted Log-rank Tests</a></p>
</td></tr>
<tr><td><code id="surv_pvalue_+3A_test.for.trend">test.for.trend</code></td>
<td>
<p>logical value. Default is FALSE. If TRUE, returns the
test for trend p-values. Tests for trend are designed to detect ordered
differences in survival curves. That is, for at least one group. The test
for trend can be only performed when the number of groups is &gt; 2.</p>
</td></tr>
<tr><td><code id="surv_pvalue_+3A_combine">combine</code></td>
<td>
<p>logical value. Used only when fit is a list of survfit objects.
If TRUE, combine the results for multiple fits.</p>
</td></tr>
<tr><td><code id="surv_pvalue_+3A_...">...</code></td>
<td>
<p>other arguments including pval, pval.coord, pval.method.coord.
These are only used internally to specify custom pvalue, pvalue and pvalue
method coordinates on the survival plot. Normally, users don't need these
arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a data frame with the columns (pval, method, pval.txt and
variable). If additional arguments (pval, pval.coord, pval.method.coord,
get_coord) are specified, then extra columns (pval.x, pval.y, method.x and
method.y) are returned. </p>
 <ul>
<li><p> pval: pvalue </p>
</li>
<li><p> method: method
used to compute pvalues </p>
</li>
<li><p> pval.txt: formatted text ready to use for
annotating plots </p>
</li>
<li><p> pval.x, pval.y: x &amp; y coordinates of the pvalue for
annotating the plot </p>
</li>
<li><p> method.x, method.y: x &amp; y coordinates of pvalue
method </p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)

# Different survfits
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit.null &lt;- surv_fit(Surv(time, status) ~ 1, data = colon)

fit1 &lt;- surv_fit(Surv(time, status) ~ sex, data = colon)

fit2 &lt;- surv_fit(Surv(time, status) ~ adhere, data = colon)

fit.list &lt;- list(sex = fit1, adhere = fit2)

# Extract the median survival
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
surv_pvalue(fit.null)

surv_pvalue(fit2, colon)

surv_pvalue(fit.list)

surv_pvalue(fit.list, combine = TRUE)

# Grouped survfit
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
fit.list2 &lt;- surv_fit(Surv(time, status) ~ sex, data = colon,
                     group.by = "rx")

surv_pvalue(fit.list2)

# Get coordinate for annotion of the survival plots
#:::::::::::::::::::::::::::::::::::::::::::::::::::::::
surv_pvalue(fit.list2, combine = TRUE, get_coord = TRUE)

</code></pre>

<hr>
<h2 id='surv_summary'>Nice Summary of a Survival Curve</h2><span id='topic+surv_summary'></span>

<h3>Description</h3>

<p>Compared to the default summary() function, surv_summary()
creates a data frame containing a nice summary from
<code><a href="survival.html#topic+survfit">survfit</a></code> results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_summary(x, data = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_summary_+3A_x">x</code></td>
<td>
<p>an object of class survfit.</p>
</td></tr>
<tr><td><code id="surv_summary_+3A_data">data</code></td>
<td>
<p>a dataset used to fit survival curves. If not supplied then data
will be extracted from 'fit' object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <b>'surv_summary'</b>, which is a data frame with
the following columns: </p>
 <ul>
<li><p> time: the time points at which the
curve has a step. </p>
</li>
<li><p> n.risk: the number of subjects at risk at t. </p>
</li>
<li>
<p>n.event: the number of events that occur at time t. </p>
</li>
<li><p>  n.censor: number
of censored events. </p>
</li>
<li><p> surv: estimate of survival. </p>
</li>
<li><p> std.err:
standard error of survival. </p>
</li>
<li><p> upper: upper end of confidence interval.
</p>
</li>
<li><p> lower: lower end of confidence interval. </p>
</li>
<li><p> strata: stratification of survival curves.</p>
</li></ul>

<p>In a situation, where survival curves have been fitted with one or more
variables, surv_summary object contains <b>extra columns</b> representing the
variables. This makes it possible to facet the output of
<code><a href="#topic+ggsurvplot">ggsurvplot</a></code> by strata or by some combinations of factors.
</p>
<p>surv_summary object has also an attribut named <b>'table'</b> containing
information about the survival curves, including medians of survival with
confidence intervals, as well as, the total number of subjects and the
number of event in each curve.
</p>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara, <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Fit survival curves
require("survival")
fit &lt;- survfit(Surv(time, status) ~ rx + adhere, data = colon)

# Summarize
res.sum &lt;- surv_summary(fit, data = colon)
head(res.sum)

# Information about the survival curves
attr(res.sum, "table")


</code></pre>

<hr>
<h2 id='theme_survminer'>Theme for Survminer Plots</h2><span id='topic+theme_survminer'></span><span id='topic+theme_cleantable'></span>

<h3>Description</h3>

<p>Default theme for plots generated with survminer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_survminer(
  base_size = 12,
  base_family = "",
  font.main = c(16, "plain", "black"),
  font.submain = c(15, "plain", "black"),
  font.x = c(14, "plain", "black"),
  font.y = c(14, "plain", "black"),
  font.caption = c(15, "plain", "black"),
  font.tickslab = c(12, "plain", "black"),
  legend = c("top", "bottom", "left", "right", "none"),
  font.legend = c(10, "plain", "black"),
  ...
)

theme_cleantable(base_size = 12, base_family = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theme_survminer_+3A_base_size">base_size</code></td>
<td>
<p>base font size</p>
</td></tr>
<tr><td><code id="theme_survminer_+3A_base_family">base_family</code></td>
<td>
<p>base font family</p>
</td></tr>
<tr><td><code id="theme_survminer_+3A_font.main">font.main</code>, <code id="theme_survminer_+3A_font.submain">font.submain</code>, <code id="theme_survminer_+3A_font.caption">font.caption</code>, <code id="theme_survminer_+3A_font.x">font.x</code>, <code id="theme_survminer_+3A_font.y">font.y</code>, <code id="theme_survminer_+3A_font.tickslab">font.tickslab</code>, <code id="theme_survminer_+3A_font.legend">font.legend</code></td>
<td>
<p>a vector of length 3
indicating respectively the size (e.g.: 14), the style (e.g.: &quot;plain&quot;,
&quot;bold&quot;, &quot;italic&quot;, &quot;bold.italic&quot;) and the color (e.g.: &quot;red&quot;) of main title, subtitle, caption,
xlab and ylab, axis tick labels and legend, respectively. For example <em>font.x =
c(14, &quot;bold&quot;, &quot;red&quot;)</em>.  Use font.x = 14, to change only font size; or use
font.x = &quot;bold&quot;, to change only font face.</p>
</td></tr>
<tr><td><code id="theme_survminer_+3A_legend">legend</code></td>
<td>
<p>character specifying legend position. Allowed values are one of
c(&quot;top&quot;, &quot;bottom&quot;, &quot;left&quot;, &quot;right&quot;, &quot;none&quot;). Default is &quot;top&quot; side position.
to remove the legend use legend = &quot;none&quot;. Legend position can be also
specified using a numeric vector c(x, y); see details section.</p>
</td></tr>
<tr><td><code id="theme_survminer_+3A_...">...</code></td>
<td>
<p>additional arguments passed to the function theme_survminer().</p>
</td></tr>
</table>


<h3>Functions</h3>


<ul>
<li> <p><code>theme_survminer</code>: Default theme for survminer plots. A theme similar to theme_classic() with large font size.
</p>
</li>
<li> <p><code>theme_cleantable</code>: theme for drawing a clean risk table and cumulative
number of events table. A theme similar to theme_survminer() without i)
axis lines and, ii) x axis ticks and title.
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Alboukadel Kassambara, <a href="mailto:alboukadel.kassambara@gmail.com">alboukadel.kassambara@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Fit survival curves
#++++++++++++++++++++++++++++++++++++
require("survival")
fit&lt;- survfit(Surv(time, status) ~ sex, data = lung)

# Basic survival curves
#++++++++++++++++++++++++++++++++++++
ggsurv &lt;- ggsurvplot(fit, data = lung, risk.table = TRUE,
   main = "Survival curves",
   submain = "Based on Kaplan-Meier estimates",
   caption = "created with survminer"
   )

# Change font size, style and color
#++++++++++++++++++++++++++++++++++++
# Change font size, style and color at the same time
# Use font.x = 14, to change only font size; or use
# font.x = "bold", to change only font face.
ggsurv %+% theme_survminer(
     font.main = c(16, "bold", "darkblue"),
     font.submain = c(15, "bold.italic", "purple"),
     font.caption = c(14, "plain", "orange"),
     font.x = c(14, "bold.italic", "red"),
     font.y = c(14, "bold.italic", "darkred"),
     font.tickslab = c(12, "plain", "darkgreen")
   )

# Clean risk table
# +++++++++++++++++++++++++++++
ggsurv$table &lt;- ggsurv$table + theme_cleantable()
ggsurv

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
