<!DOCTYPE html><html><head><title>Help for package SurvHiDim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SurvHiDim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#hdClust'><p>hdClust</p></a></li>
<li><a href='#hdNetwork'><p>hdNetwork</p></a></li>
<li><a href='#hidimLasso'><p>hidimLasso</p>
</p>
<p>LASSO for high dimensional data</p></a></li>
<li><a href='#hidimSurv'><p>hidimSurv</p>
Survival analysis on high dimensional data</a></li>
<li><a href='#hidimSurvbon'><p>hidimSurvbon</p>
</p>
<p>Uses bonferroni correction factor in survival analysis</p></a></li>
<li><a href='#hidimSurvbonlas'><p>hidimSurvbonlas</p>
</p>
<p>Two step filteration using bonferroni correction and LASSO</p></a></li>
<li><a href='#hidimSurvlas'><p>hidimSurvlas</p>
</p>
<p>Two step filteration without bonferroni correction</p></a></li>
<li><a href='#hidimsvc'><p>hidimsvc</p>
Survival analysis on high dimensional by creating batches of variables</a></li>
<li><a href='#hnscc'><p>hnscc</p>
</p>
<p>High dimensional genomic data on head and neck cancer</p></a></li>
<li><a href='#multicoxa'><p>High dimensional multivariate cox proportional hazard data analysis</p></a></li>
<li><a href='#multicoxb'><p>High dimensional multivariate cox proportional hazard data analysis</p></a></li>
<li><a href='#survdesc'><p>High dimensional univariate cox proportional hazard analysis.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>High Dimensional Survival Data Analysis</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-06-22</td>
</tr>
<tr>
<td>Description:</td>
<td>High dimensional time to events data analysis with variable selection technique.
             Currently support LASSO, clustering and Bonferroni's correction.</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>Yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>survival, utils, Rdpack (&ge; 0.7), readr, tidyverse, glmnet,
factoextra, igraph, useful</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Atanu Bhattacharjee &lt;atanustat@gmail.com&gt;</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1.9000</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-06-26 15:21:50 UTC; atanu</td>
</tr>
<tr>
<td>Author:</td>
<td>Atanu Bhattacharjee [aut, cre, ctb],
  Akash Pawar [aut, ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-06-26 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='hdClust'>hdClust</h2><span id='topic+hdClust'></span>

<h3>Description</h3>

<p>Creates a network plot of high dimensional variables and lists those variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hdClust(m, n, siglevel, u, ID, OS, Death, PFS, Prog, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hdClust_+3A_m">m</code></td>
<td>
<p>Starting column number from where variables of high dimensional data will be selected.</p>
</td></tr>
<tr><td><code id="hdClust_+3A_n">n</code></td>
<td>
<p>Ending column number till where variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hdClust_+3A_siglevel">siglevel</code></td>
<td>
<p>Level of significance pre-determined by the user.</p>
</td></tr>
<tr><td><code id="hdClust_+3A_u">u</code></td>
<td>
<p>Factors of Event column e.g. 0,1 or 2 or Number of clusters to form.</p>
</td></tr>
<tr><td><code id="hdClust_+3A_id">ID</code></td>
<td>
<p>Column name of subject ID, a string value. i.e. &quot;id&quot;</p>
</td></tr>
<tr><td><code id="hdClust_+3A_os">OS</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="hdClust_+3A_death">Death</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="hdClust_+3A_pfs">PFS</code></td>
<td>
<p>Column name of progression free survival duration, a string value. i.e &quot;pfs&quot;</p>
</td></tr>
<tr><td><code id="hdClust_+3A_prog">Prog</code></td>
<td>
<p>Column name of progression event, a string value. i.e &quot;prog&quot;</p>
</td></tr>
<tr><td><code id="hdClust_+3A_data">data</code></td>
<td>
<p>High dimensional data having survival duration, event information, column of time for death cases and observations on various covariates under study.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Gives network plot and lists the variables showing correlation.
</p>
<p>hdClust function first creates a new column 'Status' in the input
data set and assigns values 0, 1, 2 to each rows. It assigns 0 (for progression = 1 &amp; death(event) = 0) is
or (when progression = 0 &amp; death(event) = 0. It assigns 1 (for progression = 1 &amp;
death = 1, whereas assigns 2 (for progression = 0 and death = 1).
</p>
<p>Further, it creates two data sets, one data set named 'deathdata' which includes
subjects with status 0 and 1 and applies Cox PH on it. Another data is named as
'compdata' which includes subjects with status 0 and 2, then applies Cox PH after
substituting 2 by 1. Then it filters out study variables having P-value &lt; siglevel(significance level taken as input from user) from both
subset data. Secondly, it merges the common significant variables from both data and creates a
new data frame which contains columns, 'ID','OS','Death','PFS','Prog','Status' and observations
of common significant variables (which are supposed to be leading to death given they leads to progression of cancer
as well as accounts for competing risks)
Further, it lists the common variables names and correspond in results in .csv format by default in user's current
working directory.
</p>
<p>hdClust(m,n,siglevel,threshold,data),
</p>
<p>1) Subject ID column should be named as 'ID'.
</p>
<p>2) OS column must be named as 'OS'.
</p>
<p>3) Death status/event column should be named as 'Death'.
</p>
<p>4) Progression Fress Survival column should be named as 'PFS'.
</p>
<p>5) Progression event column should be named as 'Prog'.
</p>
<p>deathdata - A data frame with status column includes only those rows/subjects for which death/event was observed or not, given
progression was observed or not.
</p>
<p>compdata - A data frame with status column which includes those rows/subjects who died given progression was observed
</p>
<p>data1variables - list of variables/genes from deathdata
</p>
<p>data2variables - list of variables/genes from compdata
</p>
<p>siginificantpvalueA - A data frame with estimate values, HR, Pvalue, etc. of significant variables from deathdata.
siginificantpvalueB - A data frame with estimate values, HR, Pvalue, etc. of significant variables from compdata.
</p>
<p>commongenes - A data frame consisting observations on common significant study variables.
</p>
<p>cvar - List of common significant study variables.
</p>
<p>commondata - A final data out consisting survival information ans observations on common significant study variables.
</p>
<p>By default the fucntion stores the output in .csv forms in current directory of user.
</p>
<p>Further it creates a cluster plot of variables of similar behavior.
</p>


<h3>Value</h3>

<p>A list containing variable names and the correlation values.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595). John Wiley &amp; Sons.
</p>
<p>Banerjee, S., Vishwakarma, G. K., &amp; Bhattacharjee, A. (2019). Classification Algorithm for
High Dimensional Protein Markers in Time-course Data. arXiv preprint arXiv:1907.12853.
</p>


<h3>See Also</h3>

<p>hdNetwork
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
hdClust(7,105,0.05,2,ID="id",OS="os",Death="death",PFS="pfs",Prog="prog",hnscc)
##
</code></pre>

<hr>
<h2 id='hdNetwork'>hdNetwork</h2><span id='topic+hdNetwork'></span>

<h3>Description</h3>

<p>Creates a network plot of high dimensional variables and lists those variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hdNetwork(m, n, siglevel, threshold, ID, OS, Death, PFS, Prog, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hdNetwork_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_siglevel">siglevel</code></td>
<td>
<p>Level of significance pre-determined by the user.</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_threshold">threshold</code></td>
<td>
<p>Level to categorize the relation among the significant covariates.</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_id">ID</code></td>
<td>
<p>Column name of subject ID, a string value. i.e. &quot;id&quot;</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_os">OS</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_death">Death</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_pfs">PFS</code></td>
<td>
<p>Column name of progression free survival duration, a string value. i.e &quot;pfs&quot;</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_prog">Prog</code></td>
<td>
<p>Column name of progression event, a string value. i.e &quot;prog&quot;</p>
</td></tr>
<tr><td><code id="hdNetwork_+3A_data">data</code></td>
<td>
<p>High dimensional data containing the survival, progression and genomic observations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Gives cluster plot and lists the variables showing correlation.
</p>
<p>hdNetwork function first creates a new column 'Status' in the input
data set and assigns values 0, 1, 2 to each rows. It assigns 0 (for progression = 1 &amp; death(event) = 0) is
or (when progression = 0 &amp; death(event) = 0. It assigns 1 (for progression = 1 &amp;
death = 1, whereas assigns 2 (for progression = 0 and death = 1).
</p>
<p>Further, it creates two data sets, one data set named 'deathdata' which includes
subjects with status 0 and 1 and applies Cox PH on it. Another data is named as
'compdata' which includes subjects with status 0 and 2, then applies Cox PH after
substituting 2 by 1. Then it filters out study variables having P-value &lt; siglevel(significance level taken as input from user) from both
subset data. Secondly, it merges the commom sigificant variables from both data and creates a
new data frame which contains columns, 'ID','OS','Death','PFS','Prog','Status' and observations
of common significant variables (which are supposed to be leading to death given they leads to progression of cancer
as well as accounts for competing risks)
Further, it lists the common variables names and correspond in results in .csv format by default in user's current
working directory.
</p>
<p>hdNetwork(m,n,siglevel,threshold,data),
</p>
<p>1) Subject ID column should be named as 'ID'.
</p>
<p>2) OS column must be named as 'OS'.
</p>
<p>3) Death status/event column should be named as 'Death'.
</p>
<p>4) Progression Fress Survival column should be named as 'PFS'.
</p>
<p>5) Progression event column should be named as 'Prog'.
</p>
<p>deathdata - A data frame with status column includes only those rows/subjects for which death/event was observed or not, given
progression was observed or not.
</p>
<p>compdata - A data frame with status column which includes those rows/subjects who died given progression was observed
</p>
<p>data1variables - list of variables/genes from deathdata
</p>
<p>data2variables - list of variables/genes from compdata
</p>
<p>siginificantpvalueA - A data frame with estimate values, HR, Pvalue, etc. of significant variables from deathdata.
siginificantpvalueB - A data frame with estimate values, HR, Pvalue, etc. of significant variables from compdata.
</p>
<p>commongenes - A data frame consisting observations on common significant study variables.
</p>
<p>cvar - List of common significant study variables.
</p>
<p>commondata - A final data out consisting survival information ans observations on common significant study variables.
</p>
<p>By default the fucntion stores the output in .csv forms in current directory of user.
</p>
<p>Further it creates a network plot of variables of similar behavior.
</p>


<h3>Value</h3>

<p>A list containing variable names and the correlation values.
</p>
<p>A network plot formed by the filtered variables.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595). John Wiley &amp; Sons.
</p>
<p>Banerjee, S., Vishwakarma, G. K., &amp; Bhattacharjee, A. (2019). Classification Algorithm for
High Dimensional Protein Markers in Time-course Data. arXiv preprint arXiv:1907.12853.
</p>


<h3>See Also</h3>

<p>hdClustrr
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
hdNetwork(7,105,0.05,0.2,ID="id",OS="os",Death="death",PFS="pfs",Prog="prog",hnscc)
##
</code></pre>

<hr>
<h2 id='hidimLasso'>hidimLasso
LASSO for high dimensional data</h2><span id='topic+hidimLasso'></span>

<h3>Description</h3>

<p>Least Absolute Shrinkage and Selection Operator (LASSO) for High Dimensional Survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hidimLasso(m, n, OS, Death, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hidimLasso_+3A_m">m</code></td>
<td>
<p>Starting column number from where variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimLasso_+3A_n">n</code></td>
<td>
<p>Ending column number till where variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimLasso_+3A_os">OS</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="hidimLasso_+3A_death">Death</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="hidimLasso_+3A_data">data</code></td>
<td>
<p>High dimensional data having survival duration, event and various covariates observations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>'HiDimLasso' allows a user to apply LASSO function on the High Dimensional data and reduce the study
variables to handful number of co-variate which are observed impacting the survival outcomes.
</p>
<p>Column of Overall Survival must be named as 'OS' and the column defining the event
must be named as 'Event'.
</p>
<p>By default it stores the outcome data in user's current directory.
</p>


<h3>Value</h3>

<p>A list of variables selected by LASSO as predictor variables.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595). John Wiley &amp; Sons.
</p>
<p>Banerjee, S., Vishwakarma, G. K., &amp; Bhattacharjee, A. (2019). Classification Algorithm for
High Dimensional Protein Markers in Time-course Data. arXiv preprint arXiv:1907.12853.
</p>


<h3>See Also</h3>

<p>hidimSurvlas hidimSurvbonlas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
hidimLasso(7,105,OS="os",Death="death",hnscc)
##
</code></pre>

<hr>
<h2 id='hidimSurv'>hidimSurv
Survival analysis on high dimensional data</h2><span id='topic+hidimSurv'></span>

<h3>Description</h3>

<p>Survival analysis using Cox Proportional hazards function on high dimensional data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hidimSurv(m, n, siglevel, ID, OS, Death, PFS, Prog, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hidimSurv_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimSurv_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimSurv_+3A_siglevel">siglevel</code></td>
<td>
<p>Level of significance pre-determined by the user</p>
</td></tr>
<tr><td><code id="hidimSurv_+3A_id">ID</code></td>
<td>
<p>Column name of subject ID, a string value. i.e. &quot;id&quot;</p>
</td></tr>
<tr><td><code id="hidimSurv_+3A_os">OS</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="hidimSurv_+3A_death">Death</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="hidimSurv_+3A_pfs">PFS</code></td>
<td>
<p>Column name of progression free survival duration, a string value. i.e &quot;pfs&quot;</p>
</td></tr>
<tr><td><code id="hidimSurv_+3A_prog">Prog</code></td>
<td>
<p>Column name of progression event, a string value. i.e &quot;prog&quot;</p>
</td></tr>
<tr><td><code id="hidimSurv_+3A_data">data</code></td>
<td>
<p>High dimensional data containing the survival, progression and genomic observations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>hidimSurv function first creates a new column 'Status' in the input
data set and assigns values 0, 1, 2 to each rows. It assigns 0 (for progression = 1 &amp; death(event) = 0) is
or (when progression = 0 &amp; death(event) = 0. It assigns 1 (for progression = 1 &amp;
death = 1, whereas assigns 2 (for progression = 0 and death = 1).
</p>
<p>Further, it creates two data sets, one data set named 'deathdata' which includes
subjects with status 0 and 1 and applies Cox PH on it. Another data is named as
'compdata' which includes subjects with status 0 and 2, then applies Cox PH after
substituting 2 by 1. Then it filters out study variables having P-value &lt; siglevel(significance level taken as input from user) from both
subset data. Secondly, it merges the commom sigificant variables from both data and creates a
new data frame which contains columns, 'ID','OS','Death','PFS','Prog','Status' and observations
of common significant variables (which are supposed to be leading to death given they leads to progression of cancer
as well as accounts for competing risks)
Further, it lists the common variables names and correspond in results in .csv format by default in user's current
working directory.
</p>
<p>hidimSurv(m,n,siglevel,data),
</p>
<p>1) Subject ID column should be named as 'ID'.
</p>
<p>2) OS column must be named as 'OS'.
</p>
<p>3) Death status/event column should be named as 'Death'.
</p>
<p>4) Progression Fress Survival column should be named as 'PFS'.
</p>
<p>5) Progression event column should be named as 'Prog'.
</p>
<p>deathdata - A data frame with status column includes only those rows/subjects for which death/event was observed or not, given
progression was observed or not.
</p>
<p>compdata - A data frame with status column which includes those rows/subjects who died given progression was observed
</p>
<p>data1variables - list of variables/genes from deathdata
</p>
<p>data2variables - list of variables/genes from compdata
</p>
<p>siginificantpvalueA - A data frame with estimate values, HR, Pvalue, etc. of significant variables from deathdata.
siginificantpvalueB - A data frame with estimate values, HR, Pvalue, etc. of significant variables from compdata.
</p>
<p>commongenes - A data frame consisting observations on common significant study variables.
</p>
<p>cvar - List of common significant study variables.
</p>
<p>commondata - A final data out consisting survival information ans observations on common significant study variables.
</p>
<p>By default the fucntion stores the output in .csv forms in current directory of user.
</p>


<h3>Value</h3>

<p>List of variables found significant on OS and survival event
</p>
<p>List of variables found significant on PFS and progression event
</p>
<p>Estimates values for significant variables on OS and survival event
</p>
<p>Estimates values for significant variables on PFS and progression event
</p>
<p>Estimates data for the DEGs/Variables found common between significant DEGs from data having death due to progression and data showing death without progression
</p>
<p>Estimates data for the DEGs/Variables found common between significant DEGs from data having death due to progression and data showing death without progression
</p>
<p>List of variable/ DEGs found common between significantDEGs from data having death due to progression and data showing death without progression
</p>
<p>Data with Survival outcomes and DEGs/Variable observations on each subject for DEGs found playing crucial role in death due to progression and without
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595). John Wiley &amp; Sons.
</p>
<p>Banerjee, S., Vishwakarma, G. K., &amp; Bhattacharjee, A. (2019). Classification Algorithm for
High Dimensional Protein Markers in Time-course Data. arXiv preprint arXiv:1907.12853.
</p>


<h3>See Also</h3>

<p>hidimSurvbon hidimSurvbonlas hidimsvc
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
hidimSurv(7,105,0.05,ID="id",OS="os",Death="death",PFS="pfs",Prog="prog",hnscc)
##
</code></pre>

<hr>
<h2 id='hidimSurvbon'>hidimSurvbon
Uses bonferroni correction factor in survival analysis</h2><span id='topic+hidimSurvbon'></span>

<h3>Description</h3>

<p>Applies HiDimSurv on high dimensional data using Bonferroni's correction criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hidimSurvbon(m, n, boncorr, ID, OS, Death, PFS, Prog, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hidimSurvbon_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimSurvbon_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimSurvbon_+3A_boncorr">boncorr</code></td>
<td>
<p>Level of significance on which bonferroni correction will be applied.</p>
</td></tr>
<tr><td><code id="hidimSurvbon_+3A_id">ID</code></td>
<td>
<p>Column name of subject ID, a string value. i.e. &quot;id&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbon_+3A_os">OS</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbon_+3A_death">Death</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbon_+3A_pfs">PFS</code></td>
<td>
<p>Column name of progression free survival duration, a string value. i.e &quot;pfs&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbon_+3A_prog">Prog</code></td>
<td>
<p>Column name of progression event, a string value. i.e &quot;prog&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbon_+3A_data">data</code></td>
<td>
<p>High dimensional data containing the survival, progression and genomic observations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>hidimSurvbon function first creates a new column 'Status' in the input
data set and assigns values 0, 1, 2 to each rows. It assigns 0 (for progression = 1 &amp; death(event) = 0) is
or (when progression = 0 &amp; death(event) = 0. It assigns 1 (for progression = 1 &amp;
death = 1, whereas assigns 2 (for progression = 0 and death = 1).
</p>
<p>Further, it creates two data sets, one data set named 'deathdata' which includes
subjects with status 0 and 1 and applies Cox PH on it. Another data is named as
'compdata' which includes subjects with status 0 and 2, then applies Cox PH after
substituting 2 by 1. Then it filters out study variables having P-value &lt; siglevel/no. of columns of high dimensional data(significance level taken as input from user) from both
subset data. Secondly, it merges the commom sigificant variables from both data and creates a
new data frame which contains columns, 'ID','OS','Death','PFS','Prog','Status' and observations
of common significant variables (which are supposed to be leading to death given they leads to progression of cancer
as well as accounts for competing risks)
Further, it lists the common variables names and correspond in results in .csv format by default in user's current
working directory.
</p>
<p>hidimSurvbon(m,n,siglevel,data),
</p>
<p>1) Subject ID column should be named as 'ID'.
</p>
<p>2) OS column must be named as 'OS'.
</p>
<p>3) Death status/event column should be named as 'Death'.
</p>
<p>4) Progression Fress Survival column should be named as 'PFS'.
</p>
<p>5) Progression event column should be named as 'Prog'.
</p>
<p>deathdata - A data frame with status column includes only those rows/subjects for which death/event was observed or not, given
progression was observed or not.
</p>
<p>compdata - A data frame with status column which includes those rows/subjects who died given progression was observed
</p>
<p>data1variables - list of variables/genes from deathdata
</p>
<p>data2variables - list of variables/genes from compdata
</p>
<p>siginificantpvalueA - A data frame with estimate values, HR, Pvalue, etc. of significant variables from deathdata.
siginificantpvalueB - A data frame with estimate values, HR, Pvalue, etc. of significant variables from compdata.
</p>
<p>commongenes - A data frame consisting observations on common significant study variables.
</p>
<p>cvar - List of common significant study variables.
</p>
<p>commondata - A final data out consisting survival information ans observations on common significant study variables.
</p>
<p>By default the fucntion stores the output in .csv forms in current directory of user.
</p>


<h3>Value</h3>

<p>List of variables found significant on OS and survival event
</p>
<p>List of variables found significant on PFS and progression event
</p>
<p>Estimates values for significant variables on OS and survival event
</p>
<p>Estimates values for significant variables on PFS and progression event
</p>
<p>Estimates data for the DEGs/Variables found common between significant DEGs from data having death due to progression and data showing death without progression
</p>
<p>Estimates data for the DEGs/Variables found common between significant DEGs from data having death due to progression and data showing death without progression
</p>
<p>List of variable/ DEGs found common between significantDEGs from data having death due to progression and data showing death without progression
</p>
<p>Data with Survival outcomes and DEGs/Variable observations on each subject for DEGs found playing crucial role in death due to progression and without
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595). John Wiley &amp; Sons.
</p>
<p>Banerjee, S., Vishwakarma, G. K., &amp; Bhattacharjee, A. (2019). Classification Algorithm for
High Dimensional Protein Markers in Time-course Data. arXiv preprint arXiv:1907.12853.
</p>


<h3>See Also</h3>

<p>hidimSurvbonlas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
hidimSurvbon(7,105,0.05,ID="id",OS="os",Death="death",PFS="pfs",Prog="prog",hnscc)
##
</code></pre>

<hr>
<h2 id='hidimSurvbonlas'>hidimSurvbonlas
Two step filteration using bonferroni correction and LASSO</h2><span id='topic+hidimSurvbonlas'></span>

<h3>Description</h3>

<p>hidimSurvbonlas on high dimensional data using bonferoni's correction factor and LASSO.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hidimSurvbonlas(m, n, boncorr, ID, OS, Death, PFS, Prog, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hidimSurvbonlas_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimSurvbonlas_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimSurvbonlas_+3A_boncorr">boncorr</code></td>
<td>
<p>Level of significance on which bonferroni correction will be applied.</p>
</td></tr>
<tr><td><code id="hidimSurvbonlas_+3A_id">ID</code></td>
<td>
<p>Column name of subject ID, a string value. i.e. &quot;id&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbonlas_+3A_os">OS</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbonlas_+3A_death">Death</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbonlas_+3A_pfs">PFS</code></td>
<td>
<p>Column name of progression free survival duration, a string value. i.e &quot;pfs&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbonlas_+3A_prog">Prog</code></td>
<td>
<p>Column name of progression event, a string value. i.e &quot;prog&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvbonlas_+3A_data">data</code></td>
<td>
<p>High dimensional data containing the survival, progression and genomic observations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>hidimSurvbonlas function first creates a new column 'Status' in the input
data set and assigns values 0, 1, 2 to each rows. It assigns 0 (for progression = 1 &amp; death(event) = 0) is
or (when progression = 0 &amp; death(event) = 0. It assigns 1 (for progression = 1 &amp;
death = 1, whereas assigns 2 (for progression = 0 and death = 1).
</p>
<p>Further, it creates two data sets, one data set named 'deathdata' which includes
subjects with status 0 and 1 and applies Cox PH on it. Another data is named as
'compdata' which includes subjects with status 0 and 2, then applies Cox PH after
substituting 2 by 1. Then it filters out study variables using Bonferroni's correction criteria,  i.e. having P-value &lt; siglevel/ no. of study variables(significance level taken as input from user) from both
subset data. Secondly, it merges the commom sigificant variables from both data and creates a
new data frame which contains columns, 'ID','OS','Death','PFS','Prog','Status' and observations
of common significant variables (which are supposed to be leading to death given they leads to progression of cancer
as well as accounts for competing risks)
Further, it lists the common variables names and correspond in results in .csv format by default in user's current
working directory.
</p>
<p>On obtained commondata it fits LASSO and reduces the dimensions of study variables to handful significant variables.
</p>
<p>hidimSurvbonlas(m,n,siglevel,data)
</p>
<p>1) Subject ID column should be named as 'ID'.
</p>
<p>2) OS column must be named as 'OS'.
</p>
<p>3) Death status/event column should be named as 'Death'.
</p>
<p>4) Progression Free Survival column should be named as 'PFS'.
</p>
<p>5) Progression event column should be named as 'Prog'.
</p>
<p>deathdata - A data frame with status column includes only those rows/subjects for which death/event was observed or not, given
progression was observed or not.
</p>
<p>compdata - A data frame with status column which includes those rows/subjects who died given progression was observed
</p>
<p>data1variables - list of variables/genes from deathdata
</p>
<p>data2variables - list of variables/genes from compdata
</p>
<p>siginificantpvalueA - A data frame with estimate values, HR, Pvalue, etc. of significant variables from deathdata.
siginificantpvalueB - A data frame with estimate values, HR, Pvalue, etc. of significant variables from compdata.
</p>
<p>commongenes - A data frame consisting observations on common significant study variables.
</p>
<p>cvar - List of common significant study variables.
</p>
<p>commondata - A final data out consisting survival information ans observations on common significant study variables.
</p>
<p>By default the fucntion stores the output in .csv forms in current directory of user.
</p>


<h3>Value</h3>

<p>List of significant genes
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Ho, D. E., Imai, K., King, G., &amp; Stuart, E. A. (2007). Matching as Nonparametric Preprocessing for Reducing Model Dependence in Parametric Causal Inference. Political Analysis,
15(3), 199-236. doi: 10.1093/pan/mpl013
</p>


<h3>See Also</h3>

<p>hidimSurvbon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
hidimSurvbonlas(6,104,0.05,ID="id",OS="os",Death="death",PFS="pfs",Prog="prog",hnscc)
##
</code></pre>

<hr>
<h2 id='hidimSurvlas'>hidimSurvlas
Two step filteration without bonferroni correction</h2><span id='topic+hidimSurvlas'></span>

<h3>Description</h3>

<p>Survival analysis on high dimensional data using LASSO technique.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hidimSurvlas(m, n, siglevel, ID, OS, Death, PFS, Prog, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hidimSurvlas_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimSurvlas_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimSurvlas_+3A_siglevel">siglevel</code></td>
<td>
<p>Level of significance pre-determined by the user</p>
</td></tr>
<tr><td><code id="hidimSurvlas_+3A_id">ID</code></td>
<td>
<p>Column name of subject ID, a string value. i.e. &quot;id&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvlas_+3A_os">OS</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvlas_+3A_death">Death</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvlas_+3A_pfs">PFS</code></td>
<td>
<p>Column name of progression free survival duration, a string value. i.e &quot;pfs&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvlas_+3A_prog">Prog</code></td>
<td>
<p>Column name of progression event, a string value. i.e &quot;prog&quot;</p>
</td></tr>
<tr><td><code id="hidimSurvlas_+3A_data">data</code></td>
<td>
<p>High dimensional data containing the survival, progression and genomic observations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>hidimSurvlas function first creates a new column 'Status' in the input
data set and assigns values 0, 1, 2 to each rows. It assigns 0 (for progression = 1 &amp; death(event) = 0) is
or (when progression = 0 &amp; death(event) = 0. It assigns 1 (for progression = 1 &amp;
death = 1, whereas assigns 2 (for progression = 0 and death = 1).
</p>
<p>Further, it creates two data sets, one data set named 'deathdata' which includes
subjects with status 0 and 1 and applies Cox PH on it. Another data is named as
'compdata' which includes subjects with status 0 and 2, then applies Cox PH after
substituting 2 by 1. Then it filters out study variables having P-value &lt; siglevel(significance level taken as input from user) from both
subset data. Secondly, it merges the commom sigificant variables from both data and creates a
new data frame which contains columns, 'ID','OS','Death','PFS','Prog','Status' and observations
of common significant variables (which are supposed to be leading to death given they leads to progression of cancer
as well as accounts for competing risks)
Further, it lists the common variables names and correspond in results in .csv format by default in user's current
working directory.
</p>
<p>On obtained commondata 'hidimSurvlas' fits LASSO and reduces the dimensions of study variables to handful significant variables.
</p>
<p>hidimSurvlas(m,n,siglevel,data)
</p>
<p>1) Subject ID column should be named as 'ID'.
</p>
<p>2) OS column must be named as 'OS'.
</p>
<p>3) Death status/event column should be named as 'Death'.
</p>
<p>4) Progression Free Survival column should be named as 'PFS'.
</p>
<p>5) Progression event column should be named as 'Prog'.
</p>
<p>deathdata - A data frame with status column includes only those rows/subjects for which death/event was observed or not, given
progression was observed or not.
</p>
<p>compdata - A data frame with status column which includes those rows/subjects who died given progression was observed
</p>
<p>data1variables - List of variables/genes from deathdata
</p>
<p>data2variables - List of variables/genes from compdata
</p>
<p>siginificantpvalueA - A data frame with estimate values, HR, Pvalue, etc. of significant variables from deathdata.
siginificantpvalueB - A data frame with estimate values, HR, Pvalue, etc. of significant variables from compdata.
</p>
<p>commongenes - A data frame consisting observations on common significant study variables.
</p>
<p>cvar - List of common significant study variables.
</p>
<p>commondata - A final data out consisting survival information ans observations on common significant study variables.
</p>
<p>By default the function stores the output in .csv forms in current directory of user.
</p>


<h3>Value</h3>

<p>List of significant genes
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595). John Wiley &amp; Sons.
</p>
<p>Banerjee, S., Vishwakarma, G. K., &amp; Bhattacharjee, A. (2019). Classification Algorithm for
High Dimensional Protein Markers in Time-course Data. arXiv preprint arXiv:1907.12853.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
hidimSurvlas(7,105,0.05,ID="id",OS="os",Death="death",PFS="pfs",Prog="prog",hnscc)
##
</code></pre>

<hr>
<h2 id='hidimsvc'>hidimsvc
Survival analysis on high dimensional by creating batches of variables</h2><span id='topic+hidimsvc'></span>

<h3>Description</h3>

<p>Survival analysis on high dimensional data by creating batches of covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hidimsvc(m, n, batchsize, siglevel, ID, OS, Death, PFS, Prog, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hidimsvc_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_batchsize">batchsize</code></td>
<td>
<p>Number of variables to be consider at time while running function (maximum batch size should not be greater than one third of the total number of high dimensional variables)</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_siglevel">siglevel</code></td>
<td>
<p>Level of significance pre-determined by the user</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_id">ID</code></td>
<td>
<p>Column name of subject ID, a string value. i.e. &quot;id&quot;</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_os">OS</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_death">Death</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_pfs">PFS</code></td>
<td>
<p>Column name of progression free survival duration, a string value. i.e &quot;pfs&quot;</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_prog">Prog</code></td>
<td>
<p>Column name of progression event, a string value. i.e &quot;prog&quot;</p>
</td></tr>
<tr><td><code id="hidimsvc_+3A_data">data</code></td>
<td>
<p>High dimensional data containing the survival, progression and genomic observations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>hidimsvc function fits Univarate Cox Proportinal Hazard models by considering each variables at a time. Then it filters out study variables having P-value &lt; siglevel(significance level taken as input from user).
Once by survival and survival eevent and another by progression and progression events.
Secondly, it merges the commom sigificant variables from both OS and PFS analysis and creates a
new data frame which contains columns, 'ID','OS','Death','PFS','Prog','Status' and observations
of common significant variables (which are supposed to be leading to death given they leads to progression of cancer
as well as accounts for competing risks)
Further, it lists the common variables names and outs corresponding results in .csv format by default in user's current
working directory.
</p>
<p>It works similary to HiDimSurv unlike it creates batches of decided study vriables by user to make the analysis less time consuming.
</p>
<p>hidimsvc(m1,m2,batchsize,siglevel,data),
</p>
<p>1) Subject ID column should be named as 'ID'.
</p>
<p>2) OS column must be named as 'OS'.
</p>
<p>3) Death status/event column should be named as 'Death'.
</p>
<p>4) Progression Fress Survival column should be named as 'PFS'.
</p>
<p>5) Progression event column should be named as 'Prog'.
</p>
<p>OSDeathcoeff - A data frame containing HR estimates and p-values for study variables on fitting univariate CoxPh on OS and Survival event.
</p>
<p>PFSProgcoeff - A data frame containg HR estimates and p-values for study variables on fitting univariate CoxPh on PFS and Progression event.
</p>
<p>namevect - List of all the study variable names.
</p>
<p>significantOSDeathgenes - A data frame containing HR estimates and p-values for significant study variables.
</p>
<p>significantPFSProggenes - A data frame containing HR estimates and p-values for significant study variables
</p>
<p>commongenes - A data frame containing estimated values of significant study variables found common from significant study variables
on fitting CoxPh on survival and progression times and events.
</p>
<p>odnames - List of significant variables on fitting CoxPh using survival and survival event.
</p>
<p>ppnames - List of significant variables on fitting CoxPh using progression and progression event.
</p>
<p>cvar - List of common significant study variables on fitting CoxPh on survival and progression, times and events.
</p>
<p>commondata - A data out which contains the clinical observations and observations on commongenes variables.
</p>


<h3>Value</h3>

<p>Estimate values of significant variables/DEGs on considering Death with Progression
</p>
<p>Estimate values of significant variables/DEGs on considering Death without Progression
</p>
<p>List of variable/DEGs considering Death with Progression
</p>
<p>List of variable/DEGs considering Death without Progression
</p>
<p>Estimates data for the DEGs/Variables found common between significant DEGs from data having death due to progression and data showing death without progression&quot;
</p>
<p>List of variable/DEGs found common between significant DEGs from data having death due to progression and data showing death without progression&quot;
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595). John Wiley &amp; Sons.
</p>
<p>Banerjee, S., Vishwakarma, G. K., &amp; Bhattacharjee, A. (2019). Classification Algorithm for
High Dimensional Protein Markers in Time-course Data. arXiv preprint arXiv:1907.12853.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
hidimsvc(7,105,5,0.05,ID="id",OS="os",Death="death",PFS="pfs",Prog="prog",hnscc)
##
</code></pre>

<hr>
<h2 id='hnscc'>hnscc
High dimensional genomic data on head and neck cancer</h2><span id='topic+hnscc'></span>

<h3>Description</h3>

<p>High dimensional breast cancer gene expression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hnscc
</code></pre>


<h3>Format</h3>

<p>A dataframe with 565 rows and 104 variables
</p>

<dl>
<dt>id</dt><dd><p>ID of subjects</p>
</dd>
<dt>leftcensor</dt><dd><p>Initial censoring time</p>
</dd>
<dt>death</dt><dd><p>Survival event</p>
</dd>
<dt>os</dt><dd><p>Duration of overall survival</p>
</dd>
<dt>pfs</dt><dd><p>Duration of progression free survival</p>
</dd>
<dt>prog</dt><dd><p>Progression event</p>
</dd>
<dt>GJB1,...,HMGCS2</dt><dd><p>High dimensional covariates</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(hnscc)

## End(Not run)
</code></pre>

<hr>
<h2 id='multicoxa'>High dimensional multivariate cox proportional hazard data analysis</h2><span id='topic+multicoxa'></span>

<h3>Description</h3>

<p>Given the dimensions of the variables and survival informations. The function
performs multivariate Cox PH by taking 5 variables at a time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multicoxa(m, n, OS, event, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multicoxa_+3A_m">m</code></td>
<td>
<p>Starting number of column from where multivariate variables will get selected.</p>
</td></tr>
<tr><td><code id="multicoxa_+3A_n">n</code></td>
<td>
<p>Ending number of column till where multivariate variables will get selected.</p>
</td></tr>
<tr><td><code id="multicoxa_+3A_os">OS</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting duration of survival.</p>
</td></tr>
<tr><td><code id="multicoxa_+3A_event">event</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting survival event.</p>
</td></tr>
<tr><td><code id="multicoxa_+3A_data">data</code></td>
<td>
<p>High dimensional data containing survival observations and covariates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data set containing the survival estimates and Pvalue.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
multicoxa(m=15,n=18,OS="os",event="death",data=hnscc)
</code></pre>

<hr>
<h2 id='multicoxb'>High dimensional multivariate cox proportional hazard data analysis</h2><span id='topic+multicoxb'></span>

<h3>Description</h3>

<p>Given the dimensions of the variables and survival informations. The function
performs multivariate Cox PH by taking 5 variables at a time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multicoxb(
  C1 = NULL,
  C2 = NULL,
  C3 = NULL,
  C4 = NULL,
  C5 = NULL,
  OS,
  event,
  data
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multicoxb_+3A_c1">C1</code></td>
<td>
<p>Covar1</p>
</td></tr>
<tr><td><code id="multicoxb_+3A_c2">C2</code></td>
<td>
<p>Covar2</p>
</td></tr>
<tr><td><code id="multicoxb_+3A_c3">C3</code></td>
<td>
<p>Covar3</p>
</td></tr>
<tr><td><code id="multicoxb_+3A_c4">C4</code></td>
<td>
<p>Covar4</p>
</td></tr>
<tr><td><code id="multicoxb_+3A_c5">C5</code></td>
<td>
<p>Covar5</p>
</td></tr>
<tr><td><code id="multicoxb_+3A_os">OS</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting duration of survival.</p>
</td></tr>
<tr><td><code id="multicoxb_+3A_event">event</code></td>
<td>
<p>&quot;Column/Variable name&quot; consisting survival event.</p>
</td></tr>
<tr><td><code id="multicoxb_+3A_data">data</code></td>
<td>
<p>High dimensional data containing survival observations and covariates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data set containing the survival estimates and Pvalue.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
multicoxb(C1="GJB1",C2=NULL,C3="HPN",C4=NULL,C5=NULL,OS="os",event="death",data=hnscc)
</code></pre>

<hr>
<h2 id='survdesc'>High dimensional univariate cox proportional hazard analysis.</h2><span id='topic+survdesc'></span>

<h3>Description</h3>

<p>Given the dimension of variables and survival information the function performs uni-variate Cox PH.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survdesc(m, n, survdur, event, aic = TRUE, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survdesc_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="survdesc_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="survdesc_+3A_survdur">survdur</code></td>
<td>
<p>Column name of survival duration event, a string value. i.e. &quot;os&quot;</p>
</td></tr>
<tr><td><code id="survdesc_+3A_event">event</code></td>
<td>
<p>Column name of survival event, a string value. i.e &quot;death&quot;</p>
</td></tr>
<tr><td><code id="survdesc_+3A_aic">aic</code></td>
<td>
<p>By default aic = FALSE, if aic = TRUE the function returns</p>
</td></tr>
<tr><td><code id="survdesc_+3A_data">data</code></td>
<td>
<p>High dimensional data containing the survival, progression and genomic observations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data set containing estimates for variables present in column m to n.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##
data(hnscc)
survdesc(m=10,n=50,survdur="os",event="death",aic=TRUE,data=hnscc)
##
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
