<!DOCTYPE html><html><head><title>Help for package survival666</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {survival666}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clinic'><p>Clinical records of overall survival</p></a></li>
<li><a href='#super_survival'><p>Batch survival analysis</p></a></li>
<li><a href='#survival_pie'><p>Visualize the factors that influence survival analysis</p></a></li>
<li><a href='#survivaldata'><p>A filtered transcriptome gene expression matrix.</p></a></li>
<li><a href='#survivalrank'><p>Rank the significance of survival analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.5</td>
</tr>
<tr>
<td>Title:</td>
<td>Eliminate the Influence of Co-Expression Genes on Target Genes</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions can be used for batch survival analysis, 
	but not only for it. Most importantly, it can verify any
 	P-value calculated according to the gene expression level and 
	eliminate the influence of co-expression genes.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>survival, survminer, utils, stats, ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tidyverse, magrittr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-28 04:10:06 UTC; 86157</td>
</tr>
<tr>
<td>Author:</td>
<td>Honglu Zhang <a href="https://orcid.org/0000-0002-4844-5004"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Qi Zhang <a href="https://orcid.org/0000-0012-0199-6968"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Xiaodong Pei <a href="https://orcid.org/0000-0002-7897-4505"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Jiawen Guo <a href="https://orcid.org/0000-0002-0331-5038"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Enpeng Liu <a href="https://orcid.org/0000-0002-5549-957X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Honglu Zhang &lt;15738817548@163.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-29 08:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='clinic'>Clinical records of overall survival</h2><span id='topic+clinic'></span>

<h3>Description</h3>

<p>This is a document that records the overall survival rate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clinic
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 454 rows and 3 columns.
</p>


<h3>Source</h3>

<p><a href="https://xenabrowser.net/datapages/">https://xenabrowser.net/datapages/</a>
</p>

<hr>
<h2 id='super_survival'>Batch survival analysis</h2><span id='topic+super_survival'></span>

<h3>Description</h3>

<p>The super_survival is the frist function of package survival666.
You can use the function to perform batch operations on the gene expression
matrix to obtain the survival analysis of the entire transcriptome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>super_survival(exp, time, status, p = 1e-04, title = "survival_rank", path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="super_survival_+3A_exp">exp</code></td>
<td>
<p>Gene expression matrix .We need to use the count matrix here.Note
that symbol is required for the column name and sample for the row name.</p>
</td></tr>
<tr><td><code id="super_survival_+3A_time">time</code></td>
<td>
<p>Time data for survival analysis.You need to make sure the sample
as the row name is consistent with the gene expression matrix</p>
</td></tr>
<tr><td><code id="super_survival_+3A_status">status</code></td>
<td>
<p>living status of patient in survival analysis,You need to make sure the sample
as the row name is consistent with the gene expression matrix too.</p>
</td></tr>
<tr><td><code id="super_survival_+3A_p">p</code></td>
<td>
<p>Full in the minimum significance you can accpet here.It depend
on your need, but the number should not be too big,too broad results are meaningless.</p>
</td></tr>
<tr><td><code id="super_survival_+3A_title">title</code></td>
<td>
<p>Give a name to the file to be output</p>
</td></tr>
<tr><td><code id="super_survival_+3A_path">path</code></td>
<td>
<p>Set a save path for the output file</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: super_survival(exp=survivaldata,
time = clinic[,3],
status = clinic[,2],
p = 0.0001,
title = 'survival_rank',
path = '/yourdir/')
## End(Not run)
</code></pre>

<hr>
<h2 id='survival_pie'>Visualize the factors that influence survival analysis</h2><span id='topic+survival_pie'></span>

<h3>Description</h3>

<p>The checkout_survival() is the third function of survival666.
We use the function to visualize the co-expression interference of a specific
gene in survival analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survival_pie(exp, survivalrank, target, pcol = 4, symbolcol = 1, path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survival_pie_+3A_exp">exp</code></td>
<td>
<p>This is a transcriptome count matrix,Note that the matrix
requires symbol as column name and sample as row name</p>
</td></tr>
<tr><td><code id="survival_pie_+3A_survivalrank">survivalrank</code></td>
<td>
<p>This is the result calculated by the frist function
in the survival666 package.
However,that does not prevent you from getting results that you could get
using other methods,as long as three conditions are met.
Frist,you need to calculate the p value based on the expression of a
particular phenotype and the level of gene expresion
Second,your results must be large enough,preferably for the entire
transcriptome,otherwise filtering accuracy will be reduced
Finally,you need to eliminate genes that are not significant enough,and if
you are using a regular PC,it is recommended to have no more than 2000 lines</p>
</td></tr>
<tr><td><code id="survival_pie_+3A_target">target</code></td>
<td>
<p>target gene symbols</p>
</td></tr>
<tr><td><code id="survival_pie_+3A_pcol">pcol</code></td>
<td>
<p>Digtal vector,represents the column of P value in survivalrank</p>
</td></tr>
<tr><td><code id="survival_pie_+3A_symbolcol">symbolcol</code></td>
<td>
<p>Digtal vector,represents the column of symbol in survivalrank</p>
</td></tr>
<tr><td><code id="survival_pie_+3A_path">path</code></td>
<td>
<p>Path to output file,do not omit '/',example:'d:/file/'.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>survival_pie  <code><a href="#topic+super_survival">super_survival</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: survival_pie(exp=survivaldata,
survivalrank=survivalrank[1:300,],
target="EAF2",
pcol=4,
symbolcol=1,
path='/yourdir/')
## End(Not run)
</code></pre>

<hr>
<h2 id='survivaldata'>A filtered transcriptome gene expression matrix.</h2><span id='topic+survivaldata'></span>

<h3>Description</h3>

<p>The expression matrix was manipulated to contain less than two thousand genes.
Use the symbol as the column name and the sample number as the row name.
The reason of this is that to save space.It is best to use the complete transcriptome
as the expression matrix when using the fundtion super_survival.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survivaldata
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 454 rows and 1712 columns.
</p>


<h3>Source</h3>

<p><a href="https://xenabrowser.net/datapages/">https://xenabrowser.net/datapages/</a>
</p>

<hr>
<h2 id='survivalrank'>Rank the significance of survival analysis</h2><span id='topic+survivalrank'></span>

<h3>Description</h3>

<p>This data file is actually the output of super_survival function.
The significance P value of survival analysis for each gene in survivaldata
matrix were recorded
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survivalrank
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 1712 rows and 9 columns.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
