<!DOCTYPE html><html><head><title>Help for package SurviMChd</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {SurviMChd}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fraidm'><p>Frailty with Discrete Mixture  Model</p></a></li>
<li><a href='#fraidpm'><p>Frailty with drichlet process mixture</p></a></li>
<li><a href='#frailty'><p>Frailty in high dimensional survival data.</p></a></li>
<li><a href='#frairand'><p>Frailty with random effects in high dimensional data with MCMC</p></a></li>
<li><a href='#headnneck'><p>High dimensional genomic data on head and neck cancer</p></a></li>
<li><a href='#hnscc'><p>hnscc Head and neck cancer data</p></a></li>
<li><a href='#mcsurv'><p>Metronomic cancer data</p></a></li>
<li><a href='#survexpMC'><p>Exponential survival analysis with MCMC</p></a></li>
<li><a href='#survMC'><p>Survival analysis using Cox Proportional Hazards with MCMC.</p></a></li>
<li><a href='#survMCmulti'><p>Survival analysis on multiple variables with MCMC</p></a></li>
<li><a href='#survweibMC'><p>Weibull survival analysis with MCMC</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>High Dimensional Survival Data Analysis with Markov Chain Monte
Carlo</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-26</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>rjags,R2jags,dplyr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>Yes</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>Yes</td>
</tr>
<tr>
<td>Description:</td>
<td>High dimensional survival data analysis with Markov Chain Monte Carlo(MCMC). 
              Currently supports frailty data analysis. Allows for Weibull and 
              Exponential distribution. Includes function for interval censored data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Atanu Bhattacharjee &lt;atanustat@gmail.com&gt;</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-27 21:47:34 UTC; Atanu Bhattacharjee</td>
</tr>
<tr>
<td>Author:</td>
<td>Atanu Bhattacharjee [aut, cre, ctb],
  Akash Pawar [aut, ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-28 16:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='fraidm'>Frailty with Discrete Mixture  Model</h2><span id='topic+fraidm'></span>

<h3>Description</h3>

<p>Discrete mixture model with MCMC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fraidm(m, n, Ins, Del, Time, T.min, chn, iter, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fraidm_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables to be selected.</p>
</td></tr>
<tr><td><code id="fraidm_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables will get selected.</p>
</td></tr>
<tr><td><code id="fraidm_+3A_ins">Ins</code></td>
<td>
<p>Variable name of Institute information.</p>
</td></tr>
<tr><td><code id="fraidm_+3A_del">Del</code></td>
<td>
<p>Variable name containing the event information.</p>
</td></tr>
<tr><td><code id="fraidm_+3A_time">Time</code></td>
<td>
<p>Variable name containing the time information.</p>
</td></tr>
<tr><td><code id="fraidm_+3A_t.min">T.min</code></td>
<td>
<p>Variable name containing the time of event information.</p>
</td></tr>
<tr><td><code id="fraidm_+3A_chn">chn</code></td>
<td>
<p>Number of MCMC chains</p>
</td></tr>
<tr><td><code id="fraidm_+3A_iter">iter</code></td>
<td>
<p>Define number of iterations as number.</p>
</td></tr>
<tr><td><code id="fraidm_+3A_data">data</code></td>
<td>
<p>High dimensional data, event information given as (delta=0 if alive, delta=1 if died). If patient is censored then t.min=duration of survival. If patient is died then t.min=0. If patient is died then t=duration of survival. If patient is alive then t=NA.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By given m and n, a total of 3 variables can be selected.
</p>


<h3>Value</h3>

<p>fraidmout - b[1] is the posterior estimate of the regression coefficient for first covariate.
</p>
<p>b[2] is the posterior estimate of the regression coefficient for second covariate.
</p>
<p>b[3] is the posterior estimate of the regression coefficient for third covariate.
</p>
<p>omega[1] and omega[2] are frailty effects.
</p>
<p>c[1] and c[2] are regression intercept and coefficients of covariates over mean effect.
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using
R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595).
John Wiley &amp; Sons.
</p>




<h3>See Also</h3>

<p>fraidpm frairand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##
data(frailty)
fraidm(m=5,n=7,Ins="institute",Del="del",Time="timevar",T.min="time.min",chn=2,iter=6,data=frailty)
##

</code></pre>

<hr>
<h2 id='fraidpm'>Frailty with drichlet process mixture</h2><span id='topic+fraidpm'></span>

<h3>Description</h3>

<p>Frailty analysis on high dimensional data by Drichlet process mixture.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fraidpm(m, n, Ins, Del, Time, T.min, chn, iter, adapt, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fraidpm_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables to be selected.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables will get selected.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_ins">Ins</code></td>
<td>
<p>Variable name of Institute information.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_del">Del</code></td>
<td>
<p>Variable name containing the event information.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_time">Time</code></td>
<td>
<p>Variable name containing the time information.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_t.min">T.min</code></td>
<td>
<p>Variable name containing the time of event information.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_chn">chn</code></td>
<td>
<p>Number of MCMC chains.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_iter">iter</code></td>
<td>
<p>Define number of iterations as number.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_adapt">adapt</code></td>
<td>
<p>Define number of adaptations as number.</p>
</td></tr>
<tr><td><code id="fraidpm_+3A_data">data</code></td>
<td>
<p>High dimensional data, event information given as (delta=0 if alive, delta=1 if died). If patient is censored then t.min=duration of survival. If patient is died then t.min=0. If patient is died then t=duration of survival. If patient is alive then t=NA.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By given m and n, a total of 3 variables can be selected.
</p>


<h3>Value</h3>

<p>fraidpmout
omeg[i] are frailty effects.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using
R and OpenBUGS. CRC Press.
</p>
<p>Congdon, P. (2014). Applied bayesian modelling (Vol. 595).
John Wiley &amp; Sons.
</p>


<h3>See Also</h3>

<p>fraidm frairand
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##
data(frailty)
fraidpm(m=5,n=7,Ins="institute",Del="del",Time="timevar",T.min="time.min",chn=2,iter=6,
adapt=100,data=frailty)
##

</code></pre>

<hr>
<h2 id='frailty'>Frailty in high dimensional survival data.</h2><span id='topic+frailty'></span>

<h3>Description</h3>

<p>Data set listing institutional wise survival outcomes
</p>
<p>Survival observations data for frailty model functions of SurviMChd
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(frailty)
</code></pre>


<h3>Format</h3>

<p>A <code>tibble</code> with 7 columns and 272 rows which are :
</p>

<dl>
<dt>institute</dt><dd><p>Institute of the sample observations</p>
</dd>
<dt>del</dt><dd><p>Numberic values 0 or 1 containing death/event information</p>
</dd>
<dt>timevar</dt><dd><p>Survival duration</p>
</dd>
<dt>time.min</dt><dd><p>Minimum survival</p>
</dd>
<dt>female</dt><dd><p>Covariate_1, gender variable indicating either a female or not</p>
</dd>
<dt>ph.karno</dt><dd><p>Covariate_2</p>
</dd>
<dt>pat.karno</dt><dd><p>Covariate_3</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(frailty)

</code></pre>

<hr>
<h2 id='frairand'>Frailty with random effects in high dimensional data with MCMC</h2><span id='topic+frairand'></span>

<h3>Description</h3>

<p>Random effects frailty model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>frairand(m, n, Ins, Del, Time, T.min, chn, iter, adapt, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="frairand_+3A_m">m</code></td>
<td>
<p>Starting column number form where study variables to be selected.</p>
</td></tr>
<tr><td><code id="frairand_+3A_n">n</code></td>
<td>
<p>Ending column number till where study variables will get selected.</p>
</td></tr>
<tr><td><code id="frairand_+3A_ins">Ins</code></td>
<td>
<p>Variable name of Institute information.</p>
</td></tr>
<tr><td><code id="frairand_+3A_del">Del</code></td>
<td>
<p>Variable name containing the event information.</p>
</td></tr>
<tr><td><code id="frairand_+3A_time">Time</code></td>
<td>
<p>Variable name containing the time information.</p>
</td></tr>
<tr><td><code id="frairand_+3A_t.min">T.min</code></td>
<td>
<p>Variable name containing the time of event information.</p>
</td></tr>
<tr><td><code id="frairand_+3A_chn">chn</code></td>
<td>
<p>Numner of MCMC chains.</p>
</td></tr>
<tr><td><code id="frairand_+3A_iter">iter</code></td>
<td>
<p>Define number of iterations as number.</p>
</td></tr>
<tr><td><code id="frairand_+3A_adapt">adapt</code></td>
<td>
<p>Define number of adaptations as number.</p>
</td></tr>
<tr><td><code id="frairand_+3A_data">data</code></td>
<td>
<p>High dimensional data having survival duration, event information and column of time for death cases.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By given m and n, a total of 3 variables can be selected.
</p>


<h3>Value</h3>

<p>frairandout
omeg[i] are frailty effects.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Tawiah, R., Yau, K. K., McLachlan, G. J., Chambers, S. K., &amp;
Ng, S. K. (2019). Multilevel model with random effects for clustered survival
data with multiple failure outcomes. <em>Statistics in medicine</em>, <b>38(6)</b>, 1036-1055.
</p>


<h3>See Also</h3>

<p>fraidm fraidpm
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##
data(frailty)
frairand(m=5,n=7,Ins="institute",Del="del",Time="timevar",T.min="time.min",chn=2,iter=6,
 adapt=100,data=frailty)
##

</code></pre>

<hr>
<h2 id='headnneck'>High dimensional genomic data on head and neck cancer</h2><span id='topic+headnneck'></span>

<h3>Description</h3>

<p>Head and neck cancer data <code>tibble</code> on head and neck cancer patients for survexpMC and survweibMC functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(headnneck)
</code></pre>


<h3>Format</h3>

<p>A <code>tibble</code> with 13 columns which are :
</p>

<dl>
<dt>Subjects</dt><dd><p>Patients referred to as Subjects</p>
</dd>
<dt>OS</dt><dd><p>Overall Survival</p>
</dd>
<dt>Death</dt><dd><p>Death status for the particular subjects</p>
</dd>
<dt>randgrp1</dt><dd><p>Arm of group assigned to subjects</p>
</dd>
<dt>gender1</dt><dd><p>Demographic information of Subjects, i.e. Gender</p>
</dd>
<dt>Stratum1</dt><dd><p>Stratum from where the sample is drawn</p>
</dd>
<dt>prevoi</dt><dd><p>Categorical observation</p>
</dd>
<dt>Covariate_1</dt><dd><p>Continuous observations</p>
</dd>
<dt>Covariate_2</dt><dd><p>Continuous observations</p>
</dd>
<dt>Covariate_3</dt><dd><p>Continuous observations</p>
</dd>
<dt>Covariate_4</dt><dd><p>Continuous observations</p>
</dd>
<dt>Covariate_5</dt><dd><p>Continuous observations</p>
</dd>
<dt>Covariate_6</dt><dd><p>Continuous observations</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(headnneck)

</code></pre>

<hr>
<h2 id='hnscc'>hnscc Head and neck cancer data</h2><span id='topic+hnscc'></span>

<h3>Description</h3>

<p>High dimensional head and neck cancer gene expression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hnscc)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 565 rows and 104 variables
</p>

<dl>
<dt>ID</dt><dd><p>ID of subjects</p>
</dd>
<dt>leftcensoring</dt><dd><p>Initial censoring time</p>
</dd>
<dt>death</dt><dd><p>Survival event</p>
</dd>
<dt>os</dt><dd><p>Duration of overall survival</p>
</dd>
<dt>PFS</dt><dd><p>Duration of progression free survival</p>
</dd>
<dt>Prog</dt><dd><p>Progression event</p>
</dd>
<dt>GJB1,...,HMGCS2</dt><dd><p>High dimensional covariates</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(hnscc)
</code></pre>

<hr>
<h2 id='mcsurv'>Metronomic cancer data</h2><span id='topic+mcsurv'></span>

<h3>Description</h3>

<p>Observations made <code>tibble</code> on the head and neck cancer patients. Data for survMC function from SurviMChd package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mcsurv)
</code></pre>


<h3>Format</h3>

<p>A <code>tibble</code> with 15 columns which are :
</p>

<dl>
<dt>OS</dt><dd><p>Overall Survival</p>
</dd>
<dt>Death</dt><dd><p>Death status</p>
</dd>
<dt>t</dt><dd><p>Time at which event occurred</p>
</dd>
<dt>x1</dt><dd><p>Variable measured on continuous scale</p>
</dd>
<dt>x2</dt><dd><p>Variable measured on discrete scale</p>
</dd>
<dt>x3</dt><dd><p>Variable measured on continuous scale</p>
</dd>
<dt>x4</dt><dd><p>Variable measured on discrete scale</p>
</dd>
<dt>x5</dt><dd><p>Variable measured on continuous scale</p>
</dd></dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(mcsurv)
</code></pre>

<hr>
<h2 id='survexpMC'>Exponential survival analysis with MCMC</h2><span id='topic+survexpMC'></span>

<h3>Description</h3>

<p>Survival analysis with exponential distribution by MCMC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survexpMC(m1, n1, m2, n2, chains, iter, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survexpMC_+3A_m1">m1</code></td>
<td>
<p>Starting column number from where variables of high dimensional data will be selected.</p>
</td></tr>
<tr><td><code id="survexpMC_+3A_n1">n1</code></td>
<td>
<p>Ending column number till where variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="survexpMC_+3A_m2">m2</code></td>
<td>
<p>Starting column number from where demographic observations starts</p>
</td></tr>
<tr><td><code id="survexpMC_+3A_n2">n2</code></td>
<td>
<p>Ending column number of the demographic observations</p>
</td></tr>
<tr><td><code id="survexpMC_+3A_chains">chains</code></td>
<td>
<p>Number of MCMC chains</p>
</td></tr>
<tr><td><code id="survexpMC_+3A_iter">iter</code></td>
<td>
<p>Number of MCMC iterations</p>
</td></tr>
<tr><td><code id="survexpMC_+3A_data">data</code></td>
<td>
<p>High dimensional data having survival duration as (OS), event information as Death (1 if died, or 0 if alive).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>survexpMCout A data set listing estimated posterior means and deviances
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Kumar, M., Sonker, P. K., Saroj, A., Jain, A., Bhattacharjee, A.,
&amp; Saroj, R. K. (2020). Parametric survival analysis using R: Illustration
with lung cancer data. <em>Cancer Reports</em>, <b>3(4)</b>, e1210.
</p>


<h3>See Also</h3>

<p>survweibMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##
data(headnneck)
survexpMC(m1=8,n1=12,m2=4,n2=7,chains=2,iter=10,data=headnneck)
##

</code></pre>

<hr>
<h2 id='survMC'>Survival analysis using Cox Proportional Hazards with MCMC.</h2><span id='topic+survMC'></span>

<h3>Description</h3>

<p>Performs survival analysis using Cox Proportional Hazards with MCMC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survMC(m, n, Time, Event, chains, adapt, iter, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survMC_+3A_m">m</code></td>
<td>
<p>Starting column number from where variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="survMC_+3A_n">n</code></td>
<td>
<p>Ending column number till where variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="survMC_+3A_time">Time</code></td>
<td>
<p>Variable/Column name containing the information on duration of survival</p>
</td></tr>
<tr><td><code id="survMC_+3A_event">Event</code></td>
<td>
<p>Variable/Column name containing the information of survival event</p>
</td></tr>
<tr><td><code id="survMC_+3A_chains">chains</code></td>
<td>
<p>Number of chains to perform</p>
</td></tr>
<tr><td><code id="survMC_+3A_adapt">adapt</code></td>
<td>
<p>Number of adaptations to perform</p>
</td></tr>
<tr><td><code id="survMC_+3A_iter">iter</code></td>
<td>
<p>Number of iterations to perform</p>
</td></tr>
<tr><td><code id="survMC_+3A_data">data</code></td>
<td>
<p>High dimensional data having survival duration and event.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The survival columns of the data should be arranged as follows -
Death Death status=1 if died otherwise 0.
OS Survival duration measured as 'OS'
t.len Number of censored times
</p>


<h3>Value</h3>

<p>Data set containing Posterior HR estimates, SD and quantiles.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R
and OpenBUGS. CRC Press.
</p>


<h3>See Also</h3>

<p>survintMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##
data(mcsurv)
survMC(m=4,n=8,Time="OS",Event="Death",chains=2,adapt=100,iter=1000,data=mcsurv)
##

</code></pre>

<hr>
<h2 id='survMCmulti'>Survival analysis on multiple variables with MCMC</h2><span id='topic+survMCmulti'></span>

<h3>Description</h3>

<p>Performs survival analysis using Cox Proportional Hazards with MCMC with an option to input select multiple variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survMCmulti(
  var1 = NULL,
  var2 = NULL,
  var3 = NULL,
  var4 = NULL,
  var5 = NULL,
  Time,
  Event,
  chains,
  adapt,
  iter,
  data
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survMCmulti_+3A_var1">var1</code></td>
<td>
<p>Variable name (first one)</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_var2">var2</code></td>
<td>
<p>Variable name (second one)</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_var3">var3</code></td>
<td>
<p>Variable name (third one)</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_var4">var4</code></td>
<td>
<p>Variable name (fourth one)</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_var5">var5</code></td>
<td>
<p>Variable name (fifth one)</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_time">Time</code></td>
<td>
<p>Variable/Column name containing the information on duration of survival</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_event">Event</code></td>
<td>
<p>Variable/Column name containing the information of survival event</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_chains">chains</code></td>
<td>
<p>Number of chains to perform</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_adapt">adapt</code></td>
<td>
<p>Number of chains to perform</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_iter">iter</code></td>
<td>
<p>Number of iterations to perform</p>
</td></tr>
<tr><td><code id="survMCmulti_+3A_data">data</code></td>
<td>
<p>High dimensional data having survival duration and event.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The survival columns of the data should be arranged as follows -
Death Death status=1 if died otherwise 0.
OS Survival duration measured as 'OS'
</p>


<h3>Value</h3>

<p>Data set containing Posterior HR estimates, SD, quantiles and meandeviance.
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Bhattacharjee, A. (2020). Bayesian Approaches in Oncology Using R
and OpenBUGS. CRC Press.
</p>


<h3>See Also</h3>

<p>survintMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##
data(mcsurv)
survMCmulti(var1="x1",var2=NULL,var3="x3",var4="x2",
   var5="x4",Time="OS",Event="Death",chains=2,adapt=100,iter=1000,data=mcsurv)
##

</code></pre>

<hr>
<h2 id='survweibMC'>Weibull survival analysis with MCMC</h2><span id='topic+survweibMC'></span>

<h3>Description</h3>

<p>Survival analysis with weibull distribution by MCMC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survweibMC(m1, n1, m2, n2, chains, iter, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survweibMC_+3A_m1">m1</code></td>
<td>
<p>Starting column number from where variables of high dimensional data will be selected.</p>
</td></tr>
<tr><td><code id="survweibMC_+3A_n1">n1</code></td>
<td>
<p>Ending column number till where variables of high dimensional data will get selected.</p>
</td></tr>
<tr><td><code id="survweibMC_+3A_m2">m2</code></td>
<td>
<p>Starting column number from where demographic observations starts</p>
</td></tr>
<tr><td><code id="survweibMC_+3A_n2">n2</code></td>
<td>
<p>Ending column number of the demographic observations</p>
</td></tr>
<tr><td><code id="survweibMC_+3A_chains">chains</code></td>
<td>
<p>Number of MCMC chains</p>
</td></tr>
<tr><td><code id="survweibMC_+3A_iter">iter</code></td>
<td>
<p>Number of MCMC iterations</p>
</td></tr>
<tr><td><code id="survweibMC_+3A_data">data</code></td>
<td>
<p>High dimensional data having survival duration as (OS), event information as Death (1 if died, or 0 if alive).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>beta1[1] Posterior estimates of regression coefficients and deviance
</p>


<h3>Author(s)</h3>

<p>Atanu Bhattacharjee and Akash Pawar
</p>


<h3>References</h3>

<p>Kumar, M., Sonker, P. K., Saroj, A., Jain, A., Bhattacharjee, A.,
&amp; Saroj, R. K. (2020). Parametric survival analysis using R: Illustration with
lung cancer data. <em>Cancer Reports</em>, <b>3(4)</b>, e1210.
</p>
<p>Khan, S. A. (2018). Exponentiated Weibull regression for time-to-event
data. <em>Lifetime data analysis</em>, <b>24(2)</b>, 328-354.
</p>


<h3>See Also</h3>

<p>survexpMC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##
data(headnneck)
survweibMC(m1=8,n1=12,m2=4,n2=7,chains=2,iter=10,data=headnneck)
##

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
