<!DOCTYPE html><html><head><title>Help for package survSNP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {survSNP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#asypow'>
<p>Calculating the asymptotic power and variance</p></a></li>
<li><a href='#censbnd'>
<p>Calculate bound for censoring distribution.</p></a></li>
<li><a href='#hwe'>
<p>Relative genotypic frequencies under HWE</p></a></li>
<li><a href='#sim.snp.expsurv.power'>
<p>Asymptotic and Empirical Power</p></a></li>
<li><a href='#sim.snp.expsurv.sctest'>
<p>Simulation of Cox Score Statistic</p></a></li>
<li><a href='#surv.exp.gt.model'>
<p>Exponential Hazard Rates for a Genetic Risk Model</p></a></li>
<li><a href='#survSNP-package'>
<p>Power Calculations for SNP Studies with Censored Outcomes</p></a></li>
<li><a href='#survSNP.power.table'>
<p>Table for Asymptotic and Empirical Power</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Power Calculations for SNP Studies with Censored Outcomes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.26</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-01-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alexander Sibley &lt;dcibioinformatics@duke.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Conduct asymptotic and empirical power and sample size calculations for Single-Nucleotide Polymorphism (SNP) association studies with right censored time to event outcomes.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0), survival (&ge; 2.36-9), Rcpp (&ge; 0.9.10), lattice
(&ge; 0.20-0), foreach (&ge; 1.3.2), xtable (&ge; 1.7-0)</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU GSL (&gt;= 1.14)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>RColorBrewer, latticeExtra, knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-24 14:56:46 UTC; abs33</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-25 08:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='asypow'>
Calculating the asymptotic power and variance
</h2><span id='topic+asypow'></span>

<h3>Description</h3>

<p>This function calculates the asymptotic power and variance
assuming that the survival distribution is a mixture of
exponentials with rates and the censoring distribution is
uniform on the interval (a,b).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>asypow(n, theta, a, b, lambda0, q, p, alpha, z,exactvar)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="asypow_+3A_n">n</code></td>
<td>

<p>Sample size
</p>
</td></tr>
<tr><td><code id="asypow_+3A_theta">theta</code></td>
<td>

<p>Effect size (log genotype hazard ratio (GHR))
</p>
</td></tr>
<tr><td><code id="asypow_+3A_a">a</code></td>
<td>

<p>Censoring distribution parameter (assumed to be uniform on [a,b])
</p>
</td></tr>
<tr><td><code id="asypow_+3A_b">b</code></td>
<td>

<p>Censoring distribution parameter (assumed to be uniform on [a,b])
</p>
</td></tr>
<tr><td><code id="asypow_+3A_lambda0">lambda0</code></td>
<td>

<p>Baseline exponential hazard rate
</p>
</td></tr>
<tr><td><code id="asypow_+3A_q">q</code></td>
<td>

<p>Relative risk allele frequency
</p>
</td></tr>
<tr><td><code id="asypow_+3A_p">p</code></td>
<td>

<p>Relative genotype frequency 
</p>
</td></tr>
<tr><td><code id="asypow_+3A_alpha">alpha</code></td>
<td>

<p>Nominal two-sided type I error rate
</p>
</td></tr>
<tr><td><code id="asypow_+3A_z">z</code></td>
<td>

<p>Genotype scores (right now only additive scores AA=0,AB=1,BB=2
generate correct power)
</p>
</td></tr>
<tr><td><code id="asypow_+3A_exactvar">exactvar</code></td>
<td>

<p>Indicator for using the exact variance formula
</p>
</td></tr>
</table>


<h3>Details</h3>

<p> This function is called by <a href="#topic+sim.snp.expsurv.power">sim.snp.expsurv.power</a>
to calculate the asymptotic variance (exact and approximate)
formulas. It is not intended to be called directly by the user.
To conduct power calculations, use <a href="#topic+sim.snp.expsurv.power">sim.snp.expsurv.power</a>
or the convenience wrapper function <a href="#topic+survSNP.power.table">survSNP.power.table</a>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>power</code></td>
<td>
<p>Asymptotic power based on exact variance formula</p>
</td></tr>
<tr><td><code>power0</code></td>
<td>
<p>Asymptotic power based on approximate variance formula</p>
</td></tr>
<tr><td><code>v1</code></td>
<td>
<p>First term of asymptotic variance</p>
</td></tr>
<tr><td><code>v2</code></td>
<td>
<p>Second term of asymptotic variance</p>
</td></tr>
<tr><td><code>v12</code></td>
<td>
<p>Third term of the asymptotic variance (covariance)</p>
</td></tr>
<tr><td><code>vapprox</code></td>
<td>
<p>Approximate asymptotic variance formula (=v1)</p>
</td></tr>
<tr><td><code>exact</code></td>
<td>
<p>Exact asymptotic variance formula (=v1+v2+v12)</p>
</td></tr>
<tr><td><code>diff</code></td>
<td>
<p>Difference between variances (=v2+v12)</p>
</td></tr>
<tr><td><code>ratio</code></td>
<td>
<p>Ratio of variances (=v1/(v1+v2+v12))</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes. 
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>

<hr>
<h2 id='censbnd'>
Calculate bound for censoring distribution.
</h2><span id='topic+censbnd'></span>

<h3>Description</h3>

<p>This function computes the bound 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>censbnd(lambda, p, crate, rootint = c(0.1, 1000))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="censbnd_+3A_lambda">lambda</code></td>
<td>

<p>Baseline exponential hazard rate
</p>
</td></tr>
<tr><td><code id="censbnd_+3A_p">p</code></td>
<td>

<p>Relative genotype frequency
</p>
</td></tr>
<tr><td><code id="censbnd_+3A_crate">crate</code></td>
<td>

<p>Desired censoring rate
</p>
</td></tr>
<tr><td><code id="censbnd_+3A_rootint">rootint</code></td>
<td>

<p>Interval to be searched for the root
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The time to event distribution <code class="reqn">T</code> is assumed to be a mixture of
exponentials with parameter <code class="reqn">lambda0</code>, <code class="reqn">lambda1</code>
and <code class="reqn">lambda2</code> with mixing proportion
<code class="reqn">p0</code>,<code class="reqn">p1</code> and <code class="reqn">p2</code>.
Suppose that the censoring distribution is uniform on the interval
<code class="reqn">[0,b]</code>. 
This function calculates <code class="reqn">b</code> for a desired censoring rate.
It is not intended to be called directly by the user.
To conduct power calculations, use <a href="#topic+sim.snp.expsurv.power">sim.snp.expsurv.power</a>
or the convenience wrapper function <a href="#topic+survSNP.power.table">survSNP.power.table</a>.
</p>


<h3>Value</h3>

<p>This function returns a list from the <code>uniroot</code> function. The
root component of this list in the  
</p>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes.
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>censbnd(0.1,hwe(0.1),0.9)$root
</code></pre>

<hr>
<h2 id='hwe'>
Relative genotypic frequencies under HWE
</h2><span id='topic+hwe'></span>

<h3>Description</h3>

<p>Compute relative genotypic frequencies for a given
relative allelic frequency
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hwe(raf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hwe_+3A_raf">raf</code></td>
<td>

<p>Relative minor allele frequency for the B allele.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a bi-allelic SNP with genotypes <em>AA</em>, <em>AB</em> and
<em> BB</em>,
with a relative allele frequency <code class="reqn">q</code> for the <em>B</em> allele,
this function returns the corresponding relative genotypic
frequencies.
It is not intended to be called directly by the user.
To conduct power calculations, use <a href="#topic+sim.snp.expsurv.power">sim.snp.expsurv.power</a>
or the convenience wrapper function <a href="#topic+survSNP.power.table">survSNP.power.table</a>.
</p>


<h3>Value</h3>

<p>A vector of length three relative genotypic frequencies.
</p>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes.
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>censbnd(0.1,hwe(0.1),0.9)
</code></pre>

<hr>
<h2 id='sim.snp.expsurv.power'>
Asymptotic and Empirical Power
</h2><span id='topic+sim.snp.expsurv.power'></span>

<h3>Description</h3>

<p>This function calculates asymptotic and empirical power for SNP
association studies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.snp.expsurv.power(GHR, B, n, raf, erate, pilm, lm, model, test, alpha, 
                      exactvar = FALSE, interval = c(0, 10), rootint = c(0.1, 200))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.snp.expsurv.power_+3A_ghr">GHR</code></td>
<td>

<p>Genotype Hazard Ratio
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_b">B</code></td>
<td>

<p>Number of simulation replicates (set to 0 if no empirical
calculations are desired)
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_n">n</code></td>
<td>

<p>Sample size
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_raf">raf</code></td>
<td>

<p>Relative risk allele frequency
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_erate">erate</code></td>
<td>

<p>Event Rate
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_pilm">pilm</code></td>
<td>

<p>Probability that the time-to-event is greater than lm
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_lm">lm</code></td>
<td>

<p>Landmark time used for powering the study
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_model">model</code></td>
<td>

<p>True genetic risk model (choices are &quot;additive&quot;, &quot;recessive&quot; or
&quot;dominant&quot;. For the asymptotic calculations only the &quot;additive&quot;
should be used until further notice)
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_test">test</code></td>
<td>

<p>Hypothesized genetic risk model (choices are &quot;additive&quot;, &quot;recessive&quot;
or &quot;dominant&quot;. For the asymptotic calculations only the &quot;additive&quot; should
be used until further notice)
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_alpha">alpha</code></td>
<td>

<p>Nominal two-sided type I error rate
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_exactvar">exactvar</code></td>
<td>

<p>Indicator for using the exact variance formula
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_interval">interval</code></td>
<td>

<p>Interval to search for baseline hazard rate
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.power_+3A_rootint">rootint</code></td>
<td>

<p>Interval to search for censoring bound
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates asymptotic and empirical power for SNP
association studies
</p>


<h3>Value</h3>

<p>This function returns a <code>data.frame</code> with the following columns
</p>
<table>
<tr><td><code>B</code></td>
<td>

<p>Number of simulation replicates
</p>
</td></tr>
<tr><td><code>raf</code></td>
<td>

<p>Relative allelic frequency of the B allele
</p>
</td></tr>
<tr><td><code>q0</code></td>
<td>

<p>Relative genotypic frequency for AA
</p>
</td></tr>
<tr><td><code>q1</code></td>
<td>

<p>Relative genotypic frequency for AB
</p>
</td></tr>
<tr><td><code>q2</code></td>
<td>

<p>Relative genotypic frequency for BB
</p>
</td></tr>
<tr><td><code>lam0</code></td>
<td>

<p>Exponential hazard rate for P(T&gt;t|AA)
</p>
</td></tr>
<tr><td><code>lam1</code></td>
<td>

<p>Exponential hazard rate for P(T&gt;t|AB)
</p>
</td></tr>
<tr><td><code>lam2</code></td>
<td>

<p>Exponential hazard rate for P(T&gt;t|BB)
</p>
</td></tr>
<tr><td><code>GHR</code></td>
<td>

<p>Genotype Hazard Ratio
</p>
</td></tr>
<tr><td><code>pilm</code></td>
<td>

<p>Probability that the time to event in the population
exceeds the landmark lm: P(T&gt;lm)
</p>
</td></tr>
<tr><td><code>lm</code></td>
<td>

<p>Landmark (see pilm above)
</p>
</td></tr>
<tr><td><code>model</code></td>
<td>

<p>The true genetic risk model
</p>
</td></tr>
<tr><td><code>test</code></td>
<td>

<p>The hypothesized genetic risk model
</p>
</td></tr>
<tr><td><code>a</code></td>
<td>

<p>Parameter for the censoring distribution (uniform on [a,b])
</p>
</td></tr>
<tr><td><code>b</code></td>
<td>

<p>Parameter for the censoring distribution (uniform on [a,b])
</p>
</td></tr>
<tr><td><code>erate</code></td>
<td>

<p>Event rate
</p>
</td></tr>
<tr><td><code>n</code></td>
<td>

<p>Sample size
</p>
</td></tr>
<tr><td><code>powB</code></td>
<td>

<p>Empirical Power
</p>
</td></tr>
<tr><td><code>pow</code></td>
<td>

<p>Asymptotic Power based on the exact variance formula
</p>
</td></tr>
<tr><td><code>pow0</code></td>
<td>

<p>Asymptotic Power based on the approximate variance formula
</p>
</td></tr>
<tr><td><code>v1</code></td>
<td>

<p>First term of the variance (v1+v2+v12)
</p>
</td></tr>
<tr><td><code>v2</code></td>
<td>

<p>Second term of the variance (v1+v2+v12)
</p>
</td></tr>
<tr><td><code>v12</code></td>
<td>

<p>Third term (covariance) of the variance (v1+v2+v12)
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes.
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+survSNP.power.table">survSNP.power.table</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See vignette for examples including details on the example
# considered below

results&lt;-sim.snp.expsurv.power(GHR=1.25, B=0, n=500, raf=0.1, erate=0.75, pilm=0.5, 
                               lm=1, model="additive", test="additive", alpha=0.05)
results[,c("n","erate","alpha","pow0")]
</code></pre>

<hr>
<h2 id='sim.snp.expsurv.sctest'>
Simulation of Cox Score Statistic
</h2><span id='topic+sim.snp.expsurv.sctest'></span>

<h3>Description</h3>

<p>This function simulates the  asymptotic <em>P</em>-value for the Cox
score statistic under a specified genetic risk model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.snp.expsurv.sctest(n, gtprev, lam, a, b, ztest, diag = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.snp.expsurv.sctest_+3A_n">n</code></td>
<td>

<p>Sample size
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.sctest_+3A_gtprev">gtprev</code></td>
<td>

<p>Relative genotypic frequency
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.sctest_+3A_lam">lam</code></td>
<td>

<p>Exponential hazard rates for conditional time to event survival
functions
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.sctest_+3A_a">a</code></td>
<td>

<p>Lower bound for the uniform censoring bound
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.sctest_+3A_b">b</code></td>
<td>

<p>Upper bound for the uniform censoring bound
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.sctest_+3A_ztest">ztest</code></td>
<td>

<p>Assumed genetic model
</p>
</td></tr>
<tr><td><code id="sim.snp.expsurv.sctest_+3A_diag">diag</code></td>
<td>

<p>Set to TRUE if print out of diagnostics is desired
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function simulates the genotypes 0, 1 or 2 from relative
genotypic frequencies. Then conditional on genotype, it simulates the
time to event from an exponential distribution with parameter
<code class="reqn">lambdag</code>. The censoring time is drawn from a uniform law on the
interval <code class="reqn">[a,b]</code>.
It is not intended to be called directly by the user.
To conduct power calculations, use <code>sim.snp.expsurv.power</code>
or the convenience wrapper function <code>survSNP.power.table</code>.
</p>


<h3>Value</h3>

<p>A vector of length 2 containing the observed event rate and the
asymptotic <em>P</em>-value for the Cox score test
</p>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes. 
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>

<hr>
<h2 id='surv.exp.gt.model'>
Exponential Hazard Rates for a Genetic Risk Model
</h2><span id='topic+surv.exp.gt.model'></span>

<h3>Description</h3>

<p>This function calculates the exponential hazard rates for
the conditional survival functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv.exp.gt.model(pilm, lm, gtprev, GRR, zmodel, interval)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv.exp.gt.model_+3A_pilm">pilm</code></td>
<td>

<p>Probability that the time-to-event is greater than lm
</p>
</td></tr>
<tr><td><code id="surv.exp.gt.model_+3A_lm">lm</code></td>
<td>

<p>Landmark time used for powering the study
</p>
</td></tr>
<tr><td><code id="surv.exp.gt.model_+3A_gtprev">gtprev</code></td>
<td>

<p>Relative genotypic frequency
</p>
</td></tr>
<tr><td><code id="surv.exp.gt.model_+3A_grr">GRR</code></td>
<td>

<p>Genotype Hazard Ratio
</p>
</td></tr>
<tr><td><code id="surv.exp.gt.model_+3A_zmodel">zmodel</code></td>
<td>

<p>Genetic Risk Model (choices are &quot;additive&quot;, &quot;recessive&quot; or
&quot;dominant&quot;)
</p>
</td></tr>
<tr><td><code id="surv.exp.gt.model_+3A_interval">interval</code></td>
<td>

<p>Interval to search for baseline hazard rate
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Determines the time to event distributions in the three component
mixture model discussed in the reference below.
It is not intended to be called directly by the user.
To conduct power calculations, use <a href="#topic+sim.snp.expsurv.power">sim.snp.expsurv.power</a>
or the convenience wrapper function <a href="#topic+survSNP.power.table">survSNP.power.table</a>.
</p>


<h3>Value</h3>

<p>A vector of length three containing the exponential hazard rates
</p>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes.
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>

<hr>
<h2 id='survSNP-package'>
Power Calculations for SNP Studies with Censored Outcomes
</h2><span id='topic+survSNP-package'></span><span id='topic+survSNP'></span>

<h3>Description</h3>

<p>This package can be used to conduct asymptotic and
empirical power and sample size calculations for Single-nucleotide
Polymorphism (SNP) association studies with right censored time to
event outcomes.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> survSNP</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.26</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023-01-20</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The functions <a href="#topic+sim.snp.expsurv.power">sim.snp.expsurv.power</a> and <a href="#topic+survSNP.power.table">survSNP.power.table</a>
can be used to conduct power and sample size calculations.
The package vignette serves as a tutorial for using this package. The
technical details are provided in the reference cited below.
It is highly recommended that the vignette along with this
are reviewed before using this package.
Currently, this package only supports additive risk models.
</p>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes.
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See vignette for examples including details on the example
# considered below

results&lt;-sim.snp.expsurv.power(GHR=1.25, B=0, n=500, raf=0.1, erate=0.75, pilm=0.5, 
                               lm=1, model="additive", test="additive", alpha=0.05)
results[,c("n","erate","alpha","pow0")]


GHRs&lt;-seq(1.05,1.5,by=0.05)
ns&lt;-c(100,500,700)
rafs&lt;-c(0.1,0.3,0.5)
erates&lt;-c(0.5,0.7,0.9)
res&lt;-survSNP.power.table(GHRs,ns,rafs,erates,pilm=0.5,lm=1,model="additive",
                         test="additive",alpha=0.05)

# Create key for illustration
KEY=paste("q=",levels(factor(res$raf)),sep="")
KEY&lt;-list(lines=list(col=1:length(KEY),lty=1:length(KEY)),
          text=list(labels=paste("q=",levels(factor(res$raf)),sep="")),
          column=3)


# Illustrate Power
print(xyplot(pow0~GHR|factor(erate)*factor(n),group=factor(raf),
             data=res,type="l",lty=KEY$lines$lty,col=KEY$lines$col,
             key=KEY,
             xlab="Genotype Hazard Ratio",ylab="Power"))

# Illustrate Power (restricted to n=100)

print(xyplot(pow0~GHR|factor(erate),group=factor(raf),
             data=subset(res,n==ns[1]),
             type="l",lty=KEY$lines$lty,col=KEY$lines$col,
             key=KEY,
             xlab="Genotype Hazard Ratio",ylab="Power",
             sub=paste("n=",ns[1],", alpha=",round(unique(res$alpha),2))))

</code></pre>

<hr>
<h2 id='survSNP.power.table'>
Table for Asymptotic and Empirical Power
</h2><span id='topic+survSNP.power.table'></span>

<h3>Description</h3>

<p>This function produces a table with the resulting empirical and
asymptotic power over a given range of Genotype Hazard Ratios,
relative allelic frequencies, sample sizes, and event rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survSNP.power.table(GHRs, ns, rafs, erates, pilm, lm, model, test, alpha, 
                    exactvar = FALSE, B = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survSNP.power.table_+3A_ghrs">GHRs</code></td>
<td>

<p>A vector of Genotype Hazard Ratios
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_ns">ns</code></td>
<td>

<p>A vector of sample sizes
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_rafs">rafs</code></td>
<td>

<p>A vector of relative allelic frequencies (for the risk allele) 
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_erates">erates</code></td>
<td>

<p>A vector of event rates
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_pilm">pilm</code></td>
<td>

<p>Probability that the time-to-event is greater than lm
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_lm">lm</code></td>
<td>

<p>Landmark time used for powering the study
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_model">model</code></td>
<td>

<p>True genetic risk model (choices are &quot;additive&quot;, &quot;recessive&quot; or
&quot;dominant&quot;. For the asymptotic calculations only the &quot;additive&quot;
should be used until further notice)
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_test">test</code></td>
<td>

<p>Hypothesized genetic risk model (choices are &quot;additive&quot;, &quot;recessive&quot;
or &quot;dominant&quot;. For the asymptotic calculations only the &quot;additive&quot; should
be used until further notice)
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_alpha">alpha</code></td>
<td>

<p>Nominal two-sided type I error rate
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_exactvar">exactvar</code></td>
<td>

<p>Indicator for using the exact variance formula
</p>
</td></tr>
<tr><td><code id="survSNP.power.table_+3A_b">B</code></td>
<td>

<p>Number of simulation replicates (set to 0 if no empirical
calculations are desired)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This version only supports additive models.
</p>


<h3>Value</h3>

<p>See output of <a href="#topic+sim.snp.expsurv.power">sim.snp.expsurv.power</a>.
</p>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes.
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GRRs&lt;-seq(1.5,2,by=0.25)
ns&lt;-c(100,500,1000)
rafs&lt;-c(0.3,0.5,0.7)
erates=c(0.5,0.7,0.9)
res&lt;-survSNP.power.table(GRRs,ns,rafs,erates,pilm=0.5,lm=1,model="additive",
                         test="additive",alpha=0.05)

# Create key for illustration
KEY=paste("q=",levels(factor(res$raf)),sep="")
KEY&lt;-list(lines=list(col=1:length(KEY),lty=1:length(KEY)),
          text=list(labels=paste("q=",levels(factor(res$raf)),sep="")),
          column=3)


# Illustrate Power
print(xyplot(pow0~GHR|factor(erate)*factor(n),group=factor(raf),
             data=res,type="l",lty=KEY$lines$lty,col=KEY$lines$col,
             key=KEY,
             xlab="Genotype Hazard Ratio",ylab="Power"))

# Illustrate Power (restricted to n=100)

print(xyplot(pow0~GHR|factor(erate),group=factor(raf),
             data=subset(res,n==ns[1]),
             type="l",lty=KEY$lines$lty,col=KEY$lines$col,
             key=KEY,
             xlab="Genotype Hazard Ratio",ylab="Power",
             sub=paste("n=",ns[1],", alpha=",round(unique(res$alpha),2))))


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
