<!DOCTYPE html><html lang="en"><head><title>Help for package syntenyPlotteR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {syntenyPlotteR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#syntenyPlotteR-package'><p>syntenyPlotteR: Genome Synteny Visualization</p></a></li>
<li><a href='#draw.eh'><p>Evolution Highway style plot</p></a></li>
<li><a href='#draw.ideogram'><p>Draw ideograms in chromosome painting style</p></a></li>
<li><a href='#draw.linear'><p>Linear synteny plot</p></a></li>
<li><a href='#reformat.syntenyData'><p>Reformat synteny data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome Synteny Visualization</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Marta Farre &lt;m.farre-belmonte@kent.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Draw syntenic relationships between genome assemblies.
    There are 3 functions which take a tab delimited file containing alignment data for 
    syntenic blocks between genomes to produce either a linear alignment plot, 
    an evolution highway style plot, or a painted ideogram representing syntenic relationships.
    There is also a function to convert alignment data in the DESCHRAMBLER/inferCAR format
    to the required data structure.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3.9000</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, stringr, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Farre-lab/syntenyPlotteR/issues">https://github.com/Farre-lab/syntenyPlotteR/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-03 12:39:08 UTC; Sarah</td>
</tr>
<tr>
<td>Author:</td>
<td>Marta Farre [aut, cre, cph],
  Sarah Quigley [aut],
  Joana Damas [aut],
  Denis Larkin [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-03 16:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='syntenyPlotteR-package'>syntenyPlotteR: Genome Synteny Visualization</h2><span id='topic+syntenyPlotteR'></span><span id='topic+syntenyPlotteR-package'></span>

<h3>Description</h3>

<p>Draw syntenic relationships between genome assemblies. There are 3 functions which take a tab delimited file containing alignment data for syntenic blocks between genomes to produce either a linear alignment plot, an evolution highway style plot, or a painted ideogram representing syntenic relationships. There is also a function to convert alignment data in the DESCHRAMBLER/inferCAR format to the required data structure.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Marta Farre <a href="mailto:m.farre-belmonte@kent.ac.uk">m.farre-belmonte@kent.ac.uk</a> [copyright holder]
</p>
<p>Authors:
</p>

<ul>
<li><p> Sarah Quigley <a href="mailto:slq4@kent.ac.uk">slq4@kent.ac.uk</a>
</p>
</li>
<li><p> Joana Damas <a href="mailto:joanadamas@gmail.com">joanadamas@gmail.com</a>
</p>
</li>
<li><p> Denis Larkin <a href="mailto:dlarkin@rvc.ac.uk">dlarkin@rvc.ac.uk</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li><p> Report bugs at <a href="https://github.com/Farre-lab/syntenyPlotteR/issues">https://github.com/Farre-lab/syntenyPlotteR/issues</a>
</p>
</li></ul>


<hr>
<h2 id='draw.eh'>Evolution Highway style plot</h2><span id='topic+draw.eh'></span>

<h3>Description</h3>

<p>Draw Evolution Highway Plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw.eh(
  output,
  chrRange,
  data_file,
  directory = NULL,
  fileformat = "png",
  colour = "lightblue",
  inverted.colour = "lightpink",
  w = 5.5,
  h = 10,
  ps = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw.eh_+3A_output">output</code></td>
<td>
<p>output file name</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_chrrange">chrRange</code></td>
<td>
<p>range of chromosome numbers in the reference as numbers i.e. 1:29</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_data_file">data_file</code></td>
<td>
<p>file containing the syntentic blocks from the alignments</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_directory">directory</code></td>
<td>
<p>string containing file path to chosen directory to save image file</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_fileformat">fileformat</code></td>
<td>
<p>output file format desired using the format <code>fileformat = "png"</code> (default is &quot;png&quot;)</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_colour">colour</code></td>
<td>
<p>set colour for non-inverted syntenic blocks using the format <code>colour = "red"</code> (default is &quot;lightblue&quot;)</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_inverted.colour">inverted.colour</code></td>
<td>
<p>set colour for inverted syntenic blocks using the format <code>inverted.colour = "blue"</code> (default is &quot;lightpink&quot;)</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_w">w</code></td>
<td>
<p>width of output image using the format <code>w = 5.5</code> (default)</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_h">h</code></td>
<td>
<p>height of output image using the format <code>h = 10</code> (default)</p>
</td></tr>
<tr><td><code id="draw.eh_+3A_ps">ps</code></td>
<td>
<p>point size of output image using the format <code>ps = 10</code> (default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function draws Evolution Highway style plots.
</p>
<p>It requires as input:
</p>

<ol>
<li><p> Desired output file name
</p>
</li>
<li><p> The range of chromosomes of the reference species this is entered as either a single number i.e. 1 or a range of numbers i.e. 1:22.
<em>Note: if you are inputting sex chromosomes or chromosomes with characters in the name input a concatenated string i.e. c(1,2,3,&quot;X&quot;)</em>
</p>
</li>
<li><p> File containing the syntenic blocks of alignments of one or more target species aligned to a single reference; following this format:
reference chromosome, reference start position, reference end position, target chromosome,
target start position, target end position, orient, reference species identifier, target species identifier
</p>
</li></ol>

<p>There are optional parameters for some customization of this function:
</p>

<ol>
<li><p> The format for saving the image i.e. png or pdf can be altered by inputting: <code>fileformat = "pdf"</code> (the default value is &quot;png&quot;)
</p>
</li>
<li><p> The colour of the syntenic blocks (not inverted blocks) can be changed by inputting: <code>colour = "red"</code> (the default value is &quot;lightblue&quot;, see Rcolour pallette for colour options)
</p>
</li>
<li><p> The colour of the inverted syntenic blocks can be changed by inputting: <code>inverted.colour = "blue"</code> (the default value is &quot;lightpink&quot;, see Rcolour pallette for colour options)
</p>
</li>
<li><p> The width of the image created can be changed by inputting: <code>w = 5.5</code>
</p>
</li>
<li><p> The height of the image created can be changed by inputting: <code>h = 10</code>
</p>
</li>
<li><p> The point size of the image created can be changed by inputting: <code>ps = 10</code>
</p>
</li>
<li><p> The directory where the image file should be saved, as default the image is saved to temporary directory, change by inputting: <code>directory = "path/to/directory"</code>
</p>
</li></ol>

<p>The function works creating a graph for each reference chromosome using their start and end positions to create a block for the reference
and the target chromosome positions are used to colour the region where synteny was identified
</p>
<p>Example: <code>draw.eh("outputName",c(17,"X"), "example_eh_alignments_2.txt", directory = "path/to/directory", fileformat = "pdf")</code>
</p>


<h3>Value</h3>

<p>An image file showing the comparative drawings
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Create object containing file path to external dataset
# (see vignette to follow examples with personal data)

file1 &lt;- system.file("extdata", "example_eh_alignments_2.txt", package = "syntenyPlotteR")

# -----------------------------------------------------------------------------------

# Run draw.eh function
# To run example and save file to working directory
# add directory parameter and set working directory
# To run example with personal data see vignette

draw.eh("outputName", c(17, "X"), file1, fileformat = "pdf")
</code></pre>

<hr>
<h2 id='draw.ideogram'>Draw ideograms in chromosome painting style</h2><span id='topic+draw.ideogram'></span>

<h3>Description</h3>

<p>Draw synteny ideograms in Chromosome painting style
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw.ideogram(
  file_data,
  sizefile,
  output,
  directory = NULL,
  fileformat = "png",
  colours = colours.default,
  w = 8.5,
  h = 10,
  ps = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw.ideogram_+3A_file_data">file_data</code></td>
<td>
<p>Path to the syntenic blocks file</p>
</td></tr>
<tr><td><code id="draw.ideogram_+3A_sizefile">sizefile</code></td>
<td>
<p>Chromosome size file</p>
</td></tr>
<tr><td><code id="draw.ideogram_+3A_output">output</code></td>
<td>
<p>output file name</p>
</td></tr>
<tr><td><code id="draw.ideogram_+3A_directory">directory</code></td>
<td>
<p>string containing file path to chosen directory to save image file</p>
</td></tr>
<tr><td><code id="draw.ideogram_+3A_fileformat">fileformat</code></td>
<td>
<p>output file format specified using <code>fileformat = "pdf"</code> (the default is &quot;png&quot;)</p>
</td></tr>
<tr><td><code id="draw.ideogram_+3A_colours">colours</code></td>
<td>
<p>concatenated string of chromosome IDs and assigned colours if desired using the format <code>colours = c("1" = "red", "2" = "blue", "3" = "green","X" = "grey")</code> if the no colours are assigned default values will be used</p>
</td></tr>
<tr><td><code id="draw.ideogram_+3A_w">w</code></td>
<td>
<p>width of output image using <code>w = 8.5</code> (default)</p>
</td></tr>
<tr><td><code id="draw.ideogram_+3A_h">h</code></td>
<td>
<p>height of output image using <code>h = 10</code> (default)</p>
</td></tr>
<tr><td><code id="draw.ideogram_+3A_ps">ps</code></td>
<td>
<p>point size of output image using <code>ps = 5</code> (default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function draws pairwise synteny plots in chromosome painting style.
</p>
<p>Inputs are tab separated files;
</p>
<p>It requires as input:
</p>

<ol>
<li><p> File containing the syntenic blocks following this format:
reference chromosome, reference start position, reference end position, target chromosome,
target start position, target end position, orient, reference species identifier, target species identifier
</p>
</li>
<li><p> Tab separated file of all chromosome, scaffold, or contig lengths and the species identifier,
in order from first target species in the alignment files followed by the first reference species in the alignment files
&ndash; top of file &ndash; to the last target species and reference species in the alignment files &ndash; end of file.
in this format:
chromosome ID, chromosome length, species identifier
</p>
</li>
<li><p> The desired output file name
</p>
</li></ol>

<p>Please separate files by tab and ensure any species identifiers used between length and alignment files are matching (same identifiers and caseing)
</p>
<p>There are optional parameters for some customization of this function:
</p>

<ol>
<li><p> The format for saving the image i.e. png or pdf can be altered by inputting: <code>fileformat = "pdf"</code> (the default value is &quot;png&quot;)
</p>
</li>
<li><p> The colour of the ideograms can be altered by inputting a concatenated string of chromosome IDs with assigned colour values which can be found with R colour Pallette
e.g. <code>colours = c("1" = "red", "2" = "blue", "3" = "green","4" = "orange", "5" = "purple","X" = "grey")</code> if no colours are assigned default values will be used but colours MUST be assigned to all chromosomes
</p>
</li>
<li><p> The width of the image created can be changed by inputting: <code>w = 8.5</code> (default)
</p>
</li>
<li><p> The height of the image created can be changed by inputting: <code>h = 10</code> (default)
</p>
</li>
<li><p> The point size of the image created can be changed by inputting: <code>ps = 5</code> (default)
</p>
</li>
<li><p> The directory where the image file should be saved, as default the image is saved to temporary directory, change by inputting: <code>directory = "path/to/directory"</code>
</p>
</li></ol>

<p>Target is the species which chromosomes will be painted. Reference will be used for painting and diagonals.
Chromosomes will be in the same order as in the target chromosomes in the chromosome length file
</p>
<p>Example: <code>draw.ideogram("example_alignment_1.txt", "example_lengths.txt", "outputname", directory = "path/to/directory", fileformat = "pdf")</code>
</p>


<h3>Value</h3>

<p>An image file showing the ideogram
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Create objects containing file paths to external dataset
# (see vignette to follow examples with personal data)

length.file &lt;- system.file("extdata", "example_lengths.txt", package = "syntenyPlotteR")
file1 &lt;- system.file("extdata", "example_alignment_1.txt", package = "syntenyPlotteR")

# -----------------------------------------------------------------------------------

# Run draw.ideogram function
# To run example and save file to working directory
# add directory parameter and set working directory
# To run example with personal data see vignette

draw.ideogram(file1, length.file, "outputName", fileformat = "pdf")
</code></pre>

<hr>
<h2 id='draw.linear'>Linear synteny plot</h2><span id='topic+draw.linear'></span>

<h3>Description</h3>

<p>Draw Linear Synteny Plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw.linear(
  output,
  sizefile,
  ...,
  directory = NULL,
  fileformat = "png",
  colours = colours.default,
  w = 13,
  h = 5,
  opacity = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw.linear_+3A_output">output</code></td>
<td>
<p>output file name</p>
</td></tr>
<tr><td><code id="draw.linear_+3A_sizefile">sizefile</code></td>
<td>
<p>Chromosome Size file</p>
</td></tr>
<tr><td><code id="draw.linear_+3A_...">...</code></td>
<td>
<p>synteny files (any number of alignment files can be entered)</p>
</td></tr>
<tr><td><code id="draw.linear_+3A_directory">directory</code></td>
<td>
<p>string containing file path to chosen directory to save image file</p>
</td></tr>
<tr><td><code id="draw.linear_+3A_fileformat">fileformat</code></td>
<td>
<p>output file format specified using the format <code>fileformat = "pdf"</code> (the default is &quot;png&quot;)</p>
</td></tr>
<tr><td><code id="draw.linear_+3A_colours">colours</code></td>
<td>
<p>concatenated string of chromosome IDs and assigned colours if desired using the format <code>colours = c("1" = "red", "2" = "blue", "3" = "green", "X" = "grey")</code> if the no colours are assigned default values will be used</p>
</td></tr>
<tr><td><code id="draw.linear_+3A_w">w</code></td>
<td>
<p>width of output image using the format <code>w = 13</code> (default)</p>
</td></tr>
<tr><td><code id="draw.linear_+3A_h">h</code></td>
<td>
<p>height of output image using the format <code>h = 5</code> (default)</p>
</td></tr>
<tr><td><code id="draw.linear_+3A_opacity">opacity</code></td>
<td>
<p>opacity of syntenic bands using the format <code>opacity = .5</code> (default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function draws linear synteny plots.
</p>
<p>It requires:
</p>

<ol>
<li><p> The desired output file name;
</p>
</li>
<li><p> Tab separated file of all chromosome, scaffold, or contig lengths and the species identifier,
in order from first target species in the alignment files followed by the first reference species in the alignment files
&ndash; top of file &ndash; to the last target species and reference species in the alignment files &ndash; end of file.
in this format:
chromosome ID, chromosome length, species identifier
</p>
</li>
<li><p> files containing the syntenic blocks - one file per alignment, in order from first target/reference
(most recent species pairwise alignment in ancestral reconstruction data) alignment file
to last target/reference (ancestor pairwise alignment in ancestral reconstruction data) alignment file
following this format:
reference chromosome, reference start position, reference end position, target chromosome,
target start position, target end position, orient, reference species identifier, target species identifier
</p>
</li></ol>

<p>Please separate files by tab and ensure any species identifiers used between length and alignment files are matching (same identifiers and caseing)
</p>
<p>There are optional parameters for some customization of this function:
</p>

<ol>
<li><p> The format for saving the image i.e. png or pdf can be altered by inputting: <code>fileformat = "pdf"</code> (the default value is &quot;png&quot;)
</p>
</li>
<li><p> The colour of the synteny bands can be altered by inputting a concatenated string of chromosome IDs with assigned colour values which can be found with R colour Pallette
e.g. <code>colours = c("1" = "red", "2" = "blue", "3" = "green","4" = "orange", "5" = "purple","X" = "grey")</code> if no colours are assigned default values will be used but colours MUST be assigned to all chromosomes
</p>
</li>
<li><p> The width of the image created can be changed by inputting: <code>w = 13</code> (default)
</p>
</li>
<li><p> The height of the image created can be changed by inputting: <code>h = 5</code> (default)
</p>
</li>
<li><p> The opacity of the ribbons can be changed by inputting: <code>opacity = .5</code> (default)
</p>
</li>
<li><p> The directory where the image file should be saved, as default the image is saved to temporary directory, change by inputting: <code>directory = "path/to/directory"</code>
</p>
</li></ol>

<p>The function works using the chromosome length file to order the Y axis and provide chromosome lengths to draw chromosome ideograms and the alignment files provides coordinates to draw the alignment bands between ideograms
</p>
<p>Example: <code>draw.linear("outputname", "example_lengths.txt", "example_alignment_1.txt", "example_alignment_2.txt", "example_alignment_3.txt", directory = "path/to/directory", fileformat = "pdf")</code>
</p>


<h3>Value</h3>

<p>An image file showing the linear comparison drawings
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Create objects containing file paths to external dataset
# (see vignette to follow examples with personal data)

length.file &lt;- system.file("extdata", "example_lengths.txt", package = "syntenyPlotteR")
file1 &lt;- system.file("extdata", "example_alignment_1.txt", package = "syntenyPlotteR")
file2 &lt;- system.file("extdata", "example_alignment_2.txt", package = "syntenyPlotteR")
file3 &lt;- system.file("extdata", "example_alignment_3.txt", package = "syntenyPlotteR")

# -----------------------------------------------------------------------------------

# Run draw.linear function
# To run example and save file to working directory
# add directory parameter and set working directory
# To run example with personal data see vignette

draw.linear("outputName", length.file, file1, file2, file3, fileformat = "pdf")
</code></pre>

<hr>
<h2 id='reformat.syntenyData'>Reformat synteny data</h2><span id='topic+reformat.syntenyData'></span>

<h3>Description</h3>

<p>Reformat synteny data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reformat.syntenyData(
  file_data,
  filename,
  directory = NULL,
  reference.species = reference.sps,
  target.species = target.sps
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reformat.syntenyData_+3A_file_data">file_data</code></td>
<td>
<p>input file name for descrambler .map data</p>
</td></tr>
<tr><td><code id="reformat.syntenyData_+3A_filename">filename</code></td>
<td>
<p>output file name for reformatted data</p>
</td></tr>
<tr><td><code id="reformat.syntenyData_+3A_directory">directory</code></td>
<td>
<p>string containing file path to chosen directory to save text file</p>
</td></tr>
<tr><td><code id="reformat.syntenyData_+3A_reference.species">reference.species</code></td>
<td>
<p>reference species identifier as a character string</p>
</td></tr>
<tr><td><code id="reformat.syntenyData_+3A_target.species">target.species</code></td>
<td>
<p>target species identifier as a character string</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes output from alignment softwares such as deschrambler and inferCARs and re-formats it for syntenyPlotteR - this does not curate files only re-formats it
</p>
<p>It requires as input:
</p>

<ol>
<li><p> The data output from deschrambler or inferCARs
</p>
</li>
<li><p> the desired output file name
</p>
</li></ol>

<p>There are optional parameters for some customization of this function:
</p>

<ol>
<li><p> reference.species allows you to set the reference species identifier that will be set in the final output table i.e. <code>reference.species = "ref"</code>
</p>
</li>
<li><p> target.species allows you to set the target species identifier that will be set in the final output table i.e. <code>target.species = "tar"</code>
</p>
</li>
<li><p> The directory where the text file should be saved, as default the file is saved to temporary directory, change by inputting: <code>directory = "path/to/directory"</code>
</p>
</li></ol>

<p>Example: <code>reformat.syntenyData("deschrambler.output", "reformatted.data", directory = "path/to/directory", reference.species = "ref", target.species = "tar" )</code>
</p>


<h3>Value</h3>

<p>A text file with the reformatted data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Create object containing file path to external dataset
# (see vignette to follow examples with personal data)

file &lt;- system.file("extdata", "example_map_1.map", package = "syntenyPlotteR")

# -----------------------------------------------------------------------------------

# Run reformat.syntenyData function
# To run example and save file to working directory
# add directory parameter and set working directory
# To run example with personal data see vignette

reformat.syntenyData(file, "outputName")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
