<!DOCTYPE html><html><head><title>Help for package Sysrecon</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Sysrecon}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#contentTypes'><p>A data frame contains the labels and groups of the contents of metabolic reconstruction</p></a></li>
<li><a href='#conversionMatrix'><p>A data frame contains the details of the transformation of the metabolic reconstruction</p></a></li>
<li><a href='#conversionTypes'><p>A data frame contains the labels and groups of the transformation of metabolic reconstruction</p></a></li>
<li><a href='#draw_conversion_tree'><p>draw_conversion_tree</p></a></li>
<li><a href='#draw_step_tree'><p>draw_step_tree</p></a></li>
<li><a href='#get_term_matrix'><p>get_term_matrix</p></a></li>
<li><a href='#inputTxt'><p>inputTxt</p></a></li>
<li><a href='#map_word_to_step'><p>map_word_to_step</p></a></li>
<li><a href='#matrixProcess'><p>A data frame produced by the function map_word_to_step</p></a></li>
<li><a href='#matrixProcessConversion'><p>A data frame produced by the function map_word_to_step</p></a></li>
<li><a href='#matrixProcessFile'><p>A data frame produced by the function map_to_word in the function vizProcess</p></a></li>
<li><a href='#slotsFunction'><p>slotsFunction</p></a></li>
<li><a href='#stepsMatrix'><p>A data frame contains the details of the steps of the metabolic reconstruction</p></a></li>
<li><a href='#stepTypes'><p>A data frame contains the labels and groups of the steps of metabolic reconstruction</p></a></li>
<li><a href='#Sysrecon'><p>Sysrecon</p></a></li>
<li><a href='#systemrecon-class'><p>systemrecon-class</p></a></li>
<li><a href='#text'><p>A list of characters from the inputTxt.</p></a></li>
<li><a href='#toolsMatrix'><p>A data frame contains the details of the databases and tools of the metabolic reconstruction</p></a></li>
<li><a href='#toolsTypes'><p>A data frame contains the labels and groups of the databases and tools of metabolic reconstruction</p></a></li>
<li><a href='#vizProcess'><p>vizProcess</p></a></li>
<li><a href='#vizTools'><p>vizTools</p></a></li>
<li><a href='#vizTransformation'><p>vizTransformation</p></a></li>
<li><a href='#wordsMatrix'><p>A data frame produced by the function get_term_matrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Systematical Metabolic Reconstruction</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Description:</td>
<td>In the past decade, genome-scale metabolic reconstructions have widely been 
    used to comprehend the systems biology of metabolic pathways within an organism. 
    Different GSMs are constructed using various techniques that 
    require distinct steps, but the input data, information conversion and software tools
    are neither concisely defined nor mathematically or programmatically formulated
    in a context-specific manner.The tool that quantitatively and qualitatively specifies
    each reconstruction steps and can generate a template list of reconstruction steps dynamically
    selected from a reconstruction step reservoir, constructed based on all available published papers.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, dplyr, ggplot2, ggtree, magrittr, methods, patchwork,
plyr, RColorBrewer, rlang, SnowballC, stats, stringr, tm, utils</td>
</tr>
<tr>
<td>Config/Needs/website:</td>
<td>pkgdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://oyshilin.github.io/sysrecon/">https://oyshilin.github.io/sysrecon/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-20 08:32:40 UTC; ouyangshilin</td>
</tr>
<tr>
<td>Author:</td>
<td>Shilin Ouyang [aut, cre],
  Zihao Li [aut],
  Jiamin Hu [aut],
  Miyuan Cao [aut],
  Feng Yu [aut],
  Longfei Mao [aut],
  RStudio [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shilin Ouyang &lt;ouyangshilin@hnu.edu.cn&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-20 08:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='contentTypes'>A data frame contains the labels and groups of the contents of metabolic reconstruction</h2><span id='topic+contentTypes'></span>

<h3>Description</h3>

<p>A data frame contains the labels and groups of the contents of metabolic reconstruction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contentTypes
</code></pre>


<h3>Format</h3>


<dl>
<dt>label</dt><dd><p>the contents of the metabolic reconstruction</p>
</dd>
<dt>group</dt><dd><p>the classfication of the labels</p>
</dd>
</dl>


<hr>
<h2 id='conversionMatrix'>A data frame contains the details of the transformation of the metabolic reconstruction</h2><span id='topic+conversionMatrix'></span>

<h3>Description</h3>

<p>A data frame contains the details of the transformation of the metabolic reconstruction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conversionMatrix
</code></pre>


<h3>Format</h3>

<p>A data frame contains 93 rows and 67 variables:
</p>

<dl>
<dt>MarkerWords</dt><dd><p>the key words of the steps of metabolic reconstruction</p>
</dd>
<dt>ThresholdValue</dt><dd><p>the values filter the steps  after the mapping</p>
</dd>
<dt>Steps</dt><dd><p>the steps of metabolic reonstruction</p>
</dd>
<dt>Group</dt><dd><p>the classification of the steps</p>
</dd>
<dt>SpeciesName</dt><dd><p>1 means the output of Species name.</p>
</dd>
<dt>TaxonID</dt><dd><p>1 means the output of Taxon ID.</p>
</dd>
<dt>NCBI</dt><dd><p>1 means the output of NCBI.</p>
</dd>
<dt>Uniprot</dt><dd><p>1 means the output of Uniprot.</p>
</dd>
<dt>KEGG</dt><dd><p>1 means the output of KEGG.</p>
</dd>
<dt>MetaCyc</dt><dd><p>1 means the output of MetaCyc.</p>
</dd>
<dt>PubChem</dt><dd><p>1 means the output of PubChem.</p>
</dd>
<dt>Brenda</dt><dd><p>1 means the output of Brenda.</p>
</dd>
<dt>Download</dt><dd><p>1 means the output of Download.</p>
</dd>
<dt>GeneticInformation</dt><dd><p>1 means the output of Genetic information.</p>
</dd>
<dt>ProteinInformation</dt><dd><p>1 means the output of Protein information.</p>
</dd>
<dt>GenomeSequence</dt><dd><p>1 means the output of Genome sequence.</p>
</dd>
<dt>ProteinSequence</dt><dd><p>1 means the output of Protein sequence.</p>
</dd>
<dt>MetabolicFunctionInformation</dt><dd><p>1 means the output of Metabolic function information.</p>
</dd>
<dt>Metabolites</dt><dd><p>1 means the output of Metabolites.</p>
</dd>
<dt>Cofactor</dt><dd><p>1 means the output of Cofactor.</p>
</dd>
<dt>Nucleotides</dt><dd><p>1 means the output of Nucleotides.</p>
</dd>
<dt>AminoAcid</dt><dd><p>1 means the output of Amino acid.</p>
</dd>
<dt>MolecularWeight</dt><dd><p>1 means the output of Molecular weight.</p>
</dd>
<dt>DryWeight</dt><dd><p>1 means the output of Dry weight.</p>
</dd>
<dt>MetabolicReaction</dt><dd><p>1 means the output of Metabolic reaction.</p>
</dd>
<dt>TerminalMetabolite</dt><dd><p>1 means the output of Terminal metabolite.</p>
</dd>
<dt>Secretion</dt><dd><p>1 means the output of Secretion.</p>
</dd>
<dt>BiomassReaction</dt><dd><p>1 means the output of Biomass reaction.</p>
</dd>
<dt>DemandReaction</dt><dd><p>1 means the output of Demand reaction.</p>
</dd>
<dt>SinkReaction</dt><dd><p>1 means the output of Sink reaction.</p>
</dd>
<dt>GapReaction</dt><dd><p>1 means the output of Gap reaction.</p>
</dd>
<dt>SpontaneousReaction</dt><dd><p>1 means the output of Spontaneous reaction.</p>
</dd>
<dt>ExtracellularAndPeriplasmicTransportReactions</dt><dd><p>1 means the output of Extracellular and periplasmic transport reactions reaction.</p>
</dd>
<dt>ExchangeReaction</dt><dd><p>1 means the output of Exchange reaction.</p>
</dd>
<dt>IntracellularTransportReaction</dt><dd><p>1 means the output of Intracellular transport reaction.</p>
</dd>
<dt>ReactionFlux</dt><dd><p>1 means the output of Reaction flux.</p>
</dd>
<dt>GPR</dt><dd><p>1 means the output of GPR.</p>
</dd>
<dt>BlastComparison</dt><dd><p>1 means the output of Blast comparison.</p>
</dd>
<dt>Homology</dt><dd><p>1 means the output of Homology.</p>
</dd>
<dt>HomologousGene</dt><dd><p>1 means the output of Homology gene.</p>
</dd>
<dt>StoichiometricMatrix</dt><dd><p>1 means the output of Stoichiometric matrix.</p>
</dd>
<dt>Knockout</dt><dd><p>1 means the output of Knockout.</p>
</dd>
<dt>TargetReaction</dt><dd><p>1 means the output of Target reaction.</p>
</dd>
<dt>Restrictions</dt><dd><p>1 means the output of Restrictions.</p>
</dd>
<dt>GrowthConditions</dt><dd><p>1 means the output of Growth conditions.</p>
</dd>
<dt>MinORMax</dt><dd><p>1 means the output of min | max.</p>
</dd>
<dt>ReconstructionData</dt><dd><p>1 means the output of Reconstruction data.</p>
</dd>
<dt>FVA</dt><dd><p>1 means the output of FVA.</p>
</dd>
<dt>MetabolicFlux</dt><dd><p>1 means the output of Metabolic flux.</p>
</dd>
<dt>Statistics</dt><dd><p>1 means the output of Metabolic Statistics.</p>
</dd>
<dt>NeutralMolecularFormula</dt><dd><p>1 means the output of Neutral molecular formula.</p>
</dd>
<dt>ChargedMolecularFormula</dt><dd><p>1 means the output of Charged molecular formula.</p>
</dd>
<dt>LiteratureDataCollection</dt><dd><p>1 means the output of Literature data collection.</p>
</dd>
<dt>ConservationOfMassAndCharge</dt><dd><p>1 means the output of Conservation of mass and charge.</p>
</dd>
<dt>GibbsFreeEnergy</dt><dd><p>1 means the output of Gibbs free energy.</p>
</dd>
<dt>CellCompartmentInformation</dt><dd><p>1 means the output of Cell compartment information.</p>
</dd>
<dt>SubsystemInformation</dt><dd><p>1 means the output of Subsystem information.</p>
</dd>
<dt>MetaboliteIdentification</dt><dd><p>1 means the output of Metabolite Identification.</p>
</dd>
<dt>Unite</dt><dd><p>1 means the output of Unite.</p>
</dd>
<dt>ManualPlanning</dt><dd><p>1 means the output of Manual planning.</p>
</dd>
<dt>Coefficient</dt><dd><p>1 means the output of Coefficient.</p>
</dd>
<dt>ScatterPlot</dt><dd><p>1 means the output of Scatter plot.</p>
</dd>
<dt>TestReport</dt><dd><p>1 means the output of Test Report.</p>
</dd>
<dt>TypeIIIPath</dt><dd><p>1 means the output of Type III path.</p>
</dd>
<dt>SBML</dt><dd><p>1 means the output of SBML file.</p>
</dd>
<dt>Mat</dt><dd><p>1 means the output of Mat file.</p>
</dd>
<dt>Excel</dt><dd><p>1 means the output of Excel file.</p>
</dd>
</dl>


<hr>
<h2 id='conversionTypes'>A data frame contains the labels and groups of the transformation of metabolic reconstruction</h2><span id='topic+conversionTypes'></span>

<h3>Description</h3>

<p>A data frame contains the labels and groups of the transformation of metabolic reconstruction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conversionTypes
</code></pre>


<h3>Format</h3>

<p>A data frame contains 63 rows and 2 variables:
</p>

<dl>
<dt>label</dt><dd><p>the transformation of the metabolic reconstruction</p>
</dd>
<dt>group</dt><dd><p>the classfication of the labels</p>
</dd>
</dl>


<hr>
<h2 id='draw_conversion_tree'>draw_conversion_tree</h2><span id='topic+draw_conversion_tree'></span>

<h3>Description</h3>

<p>Drawing of metabolic process matrix diagram and phylogenetic tree diagram
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_conversion_tree(infomatrix, Matrix, stepTypes, conversionTypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_conversion_tree_+3A_infomatrix">infomatrix</code></td>
<td>
<p>Matrix generated using the words2steps function</p>
</td></tr>
<tr><td><code id="draw_conversion_tree_+3A_matrix">Matrix</code></td>
<td>
<p>The matrix about the step or transformation or databases and tools used in the metabolic reconstruction</p>
</td></tr>
<tr><td><code id="draw_conversion_tree_+3A_steptypes">stepTypes</code></td>
<td>
<p>Grouping information for reconstruction processes</p>
</td></tr>
<tr><td><code id="draw_conversion_tree_+3A_conversiontypes">conversionTypes</code></td>
<td>
<p>Grouping information for conversion content</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 &lt;- draw_conversion_tree(matrixProcessConversion, conversionMatrix,
   stepTypes, conversionTypes)
</code></pre>

<hr>
<h2 id='draw_step_tree'>draw_step_tree</h2><span id='topic+draw_step_tree'></span>

<h3>Description</h3>

<p>Drawing of metabolic process matrix diagram and phylogenetic tree diagram
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_step_tree(infomatrix, Matrix, stepTypes, contentTypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_step_tree_+3A_infomatrix">infomatrix</code></td>
<td>
<p>Matrix generated using the words2steps function</p>
</td></tr>
<tr><td><code id="draw_step_tree_+3A_matrix">Matrix</code></td>
<td>
<p>The matrix about the steps or transformation or the databases and tools of the metabolic reconstruction.</p>
</td></tr>
<tr><td><code id="draw_step_tree_+3A_steptypes">stepTypes</code></td>
<td>
<p>Grouping information for metabolic processes</p>
</td></tr>
<tr><td><code id="draw_step_tree_+3A_contenttypes">contentTypes</code></td>
<td>
<p>Grouping information for metabolic content</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p1 &lt;- draw_step_tree(matrixProcess, stepsMatrix, stepTypes, contentTypes)
</code></pre>

<hr>
<h2 id='get_term_matrix'>get_term_matrix</h2><span id='topic+get_term_matrix'></span>

<h3>Description</h3>

<p>get_term_matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_term_matrix(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_term_matrix_+3A_data">data</code></td>
<td>
<p>information of data text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>words matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>wordsMatrix &lt;- get_term_matrix(text)
</code></pre>

<hr>
<h2 id='inputTxt'>inputTxt</h2><span id='topic+inputTxt'></span>

<h3>Description</h3>

<p>A variable containing the contents of the metabolic reconstruction:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inputTxt
</code></pre>


<h3>Format</h3>

<p>A data frame with characters:
</p>

<dl>
<dt>V1</dt><dd><p>the contents of the metabolic reconstruction</p>
</dd>
</dl>


<hr>
<h2 id='map_word_to_step'>map_word_to_step</h2><span id='topic+map_word_to_step'></span>

<h3>Description</h3>

<p>map_word_to_step
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map_word_to_step(wordsMatrix, stepsMatrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map_word_to_step_+3A_wordsmatrix">wordsMatrix</code></td>
<td>
<p>The word matrix generated by the function TermDocumentMatrix in the tm package.</p>
</td></tr>
<tr><td><code id="map_word_to_step_+3A_stepsmatrix">stepsMatrix</code></td>
<td>
<p>Manually constructed metabolic process matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Specific metabolic process matrix based on text content mapping.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>matrixProcess &lt;- map_word_to_step(wordsMatrix, stepsMatrix)
</code></pre>

<hr>
<h2 id='matrixProcess'>A data frame produced by the function map_word_to_step</h2><span id='topic+matrixProcess'></span>

<h3>Description</h3>

<p>A data frame produced by the function map_word_to_step
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixProcess
</code></pre>


<h3>Format</h3>

<p>A data frame contains 67 variables
</p>

<dl>
<dt>degree</dt><dd><p>the number of the steps used in a article.</p>
</dd>
<dt>SpeciesName</dt><dd><p>-1 means the input of Species name. 1 means the output of Species name.</p>
</dd>
<dt>TaxonID</dt><dd><p>-1 means the input of Taxon ID. 1 means the output of Taxon ID.</p>
</dd>
<dt>KEGG</dt><dd><p>-1 means the input of KEGG reaction. 1 means the output of KEGG reaction.</p>
</dd>
<dt>NCBI</dt><dd><p>-1 means the input of NCBI gene. 1 means the output of NCBI gene.</p>
</dd>
<dt>Uniprot</dt><dd><p>-1 means the input of Uniprot gene. 1 means the output of Uniprot.</p>
</dd>
<dt>MetaCyc</dt><dd><p>-1 means the input of MetaCyc gene. 1 means the output of MetaCyc.</p>
</dd>
<dt>PubChem</dt><dd><p>-1 means the input of PubChem molecular formula. 1 means the output of PubChem molecular formula.</p>
</dd>
<dt>Brenda</dt><dd><p>identify whether input the Brenda. -1 means the input of Brenda. 1 means the output of Brenda.</p>
</dd>
<dt>Chebi</dt><dd><p> -1 means the input of Chebi information. 1 means the the output of Chebi information.</p>
</dd>
<dt>WebGCM</dt><dd><p>-1 means the input of Web GCM. 1 means the the output of Web GCM.</p>
</dd>
<dt>SpontaneousReaction</dt><dd><p>-1 means the input of spontaneous reaction. 1 means the the output of spontaneous reaction.</p>
</dd>
<dt>ExtracellularAndPeriplasmicTransportReactions</dt><dd><p>-1 means the input of Extracellular and periplasmic transport reactions.
1 means the the output of Extracellular and periplasmic transport reactions.</p>
</dd>
<dt>ExchangeReaction</dt><dd><p>-1 means the input of exchange reaction. 1 means the the output of exchange reaction.</p>
</dd>
<dt>MissingExchangeReaction</dt><dd><p>-1 means the input of missing exchange reaction. 1 means the the output of missing exchange reaction.</p>
</dd>
<dt>IntracellularTransportReaction</dt><dd><p>-1 means the input of intracellular transport reaction. 1 means the the output of intracellular transport reaction.</p>
</dd>
<dt>Gene</dt><dd><p>-1 means the input of gene. 1 means the the output of gene.</p>
</dd>
<dt>Protein</dt><dd><p>-1 means the input of protein. 1 means the the output of protein.</p>
</dd>
<dt>Knockout</dt><dd><p>-1 means the input of knockout. 1 means the the output of knockout.</p>
</dd>
<dt>StoichiometricMatrix</dt><dd><p>-1 means the input of stoichiometric matrix. 1 means the the output of stoichiometric matrix.</p>
</dd>
<dt>ObjectiveReaction</dt><dd><p>-1 means the input of objective reaction. 1 means the the output of objective reaction.</p>
</dd>
<dt>Constraints</dt><dd><p>-1 means the input of constraints. 1 means the the output of constraints.</p>
</dd>
<dt>Secretion</dt><dd><p>-1 means the input of secretion. 1 means the the output of secretion.</p>
</dd>
<dt>Mutisecretion</dt><dd><p>-1 means the input of mutisecretion. 1 means the the output of mutisecretion.</p>
</dd>
<dt>RichMedia</dt><dd><p>-1 means the input of rich media. 1 means the the output of rich media.</p>
</dd>
<dt>GenomeSequence</dt><dd><p>-1 means the input of genome sequence. 1 means the the output of genome sequence.</p>
</dd>
<dt>ProteomeSequence</dt><dd><p>-1 means the input of proteome sequence. 1 means the the output of proteome sequence.</p>
</dd>
<dt>AminoAcidWeight</dt><dd><p>-1 means the input of amino acid weight. 1 means the the output of amino acid weight.</p>
</dd>
<dt>AminoAcidMolecularWeight</dt><dd><p>-1 means the input of amino acid molecular weight. 1 means the the output of amino acid molecular weight.</p>
</dd>
<dt>NucleotideWeight</dt><dd><p>-1 means the input of nucleotide weight. 1 means the the output of nucleotide weight.</p>
</dd>
<dt>NucleotideMolecularWeight</dt><dd><p>-1 means the input of nucleotide molecular weight. 1 means the the output of nucleotide molecular weight.</p>
</dd>
<dt>DryWeight</dt><dd><p>-1 means the input of dry weight. 1 means the the output of dry weight.</p>
</dd>
<dt>BiomassReaction</dt><dd><p>-1 means the input of biomass reaction. 1 means the the output of biomass reaction.</p>
</dd>
<dt>DemandReaction</dt><dd><p>-1 means the input of demand reaction. 1 means the the output of demand reaction.</p>
</dd>
<dt>SinkReaction</dt><dd><p>-1 means the input of sink reaction. 1 means the the output of sink reaction.</p>
</dd>
<dt>GapReaction</dt><dd><p>-1 means the input of gap reaction. 1 means the the output of gap reaction.</p>
</dd>
<dt>MinORMax</dt><dd><p>-1 means the input of gap reaction. 1 means the the output of gap reaction.</p>
</dd>
<dt>GeneticInformation</dt><dd><p>-1 means the input of genetic information. 1 means the the output of genetic information.</p>
</dd>
<dt>MetabolicFunction</dt><dd><p>-1 means the input of metabolic function. 1 means the the output of metabolic function.</p>
</dd>
<dt>Metabolites</dt><dd><p>-1 means the input of metabolites. 1 means the the output of metabolites.</p>
</dd>
<dt>BiomassMetabolites</dt><dd><p>-1 means the input of biomass metabolites. 1 means the the output of biomass metabolites.</p>
</dd>
<dt>MetabolicReaction</dt><dd><p>-1 means the input of metabolic reaction. 1 means the the output of metabolic reaction.</p>
</dd>
<dt>ReconstructionData</dt><dd><p>-1 means the input of reconstruction data. 1 means the the output of reconstruction data.</p>
</dd>
<dt>DataStatistics</dt><dd><p>-1 means the input of data statistics. 1 means the the output of data statistics.</p>
</dd>
<dt>NeutralMolecularFormula</dt><dd><p>-1 means the input of neutral molecular formula. 1 means the the output of neutral molecular formula.</p>
</dd>
<dt>ChargedMolecularFormula</dt><dd><p>-1 means the input of charged molecular formula. 1 means the the output of charged molecular formula.</p>
</dd>
<dt>ConservationOfMassAndCharge</dt><dd><p>-1 means the input of conservation of mass and charge. 1 means the the output of conservation of mass and charge.</p>
</dd>
<dt>GibbsFreeEnergyInformation</dt><dd><p>-1 means the input of gibbs free energy information. 1 means the the output of gibbs free energy information.</p>
</dd>
<dt>Mass_ChargeConservationAssessment</dt><dd><p>-1 means the input of mass-charge conservation assessment. 1 means the the output of conservation assessment.</p>
</dd>
<dt>CellularCompartment</dt><dd><p>-1 means the input of cellular compartment. 1 means the the output of cellular compartment.</p>
</dd>
<dt>SubsystemInformation</dt><dd><p>-1 means the input of subsystem information. 1 means the the output of subsystem information.</p>
</dd>
<dt>IdentifiersInKEGG</dt><dd><p>-1 means the input of identifiers in KEGG. 1 means the the output of identifiers in KEGG.</p>
</dd>
<dt>IdentifiersInMetaCyc</dt><dd><p>-1 means the input of identifiers in MetaCyc. 1 means the the output of identifiers in MetaCyc.</p>
</dd>
<dt>UniformIdentifier</dt><dd><p>-1 means the input of uniform identifier. 1 means the the output of uniform identifier.</p>
</dd>
<dt>Coefficient</dt><dd><p>-1 means the input of coefficient. 1 means the the output of coefficient.</p>
</dd>
<dt>ScatterPlot</dt><dd><p>-1 means the input of scatter plot. 1 means the the output of scatter plot.</p>
</dd>
<dt>TerminalMetabolite</dt><dd><p>-1 means the input of terminal metabolite. 1 means the the output of terminal metabolite.</p>
</dd>
<dt>TypeIIIPathway</dt><dd><p>-1 means the input of Type III pathway. 1 means the the output of Type III pathway.</p>
</dd>
<dt>NetworkGap</dt><dd><p>-1 means the input of network gap. 1 means the the output of network gap.</p>
</dd>
<dt>Growth</dt><dd><p>-1 means the input of growth. 1 means the the output of growth.</p>
</dd>
<dt>BlockReaction</dt><dd><p>-1 means the input of block reaction. 1 means the the output of block reaction.</p>
</dd>
<dt>MetabolicFlux</dt><dd><p>-1 means the input of metabolic flux. 1 means the the output of metabolic flux.</p>
</dd>
<dt>ModelPredictCorrectly</dt><dd><p>1 means the output of model PredictCorrectly.</p>
</dd>
<dt>ModelGrowingTooFast</dt><dd><p>-1 means the input of model growing too fast. 1 means the the output of model growing too fast.</p>
</dd>
<dt>SBML</dt><dd><p>-1 means the input of SBML file. 1 means the the output of SBML file.</p>
</dd>
<dt>Mat</dt><dd><p>-1 means the input of Mat file. 1 means the the output of Mat file.</p>
</dd>
<dt>Excel</dt><dd><p>-1 means the input of Excel file. 1 means the the output of Excel file.</p>
</dd>
</dl>


<hr>
<h2 id='matrixProcessConversion'>A data frame produced by the function map_word_to_step</h2><span id='topic+matrixProcessConversion'></span>

<h3>Description</h3>

<p>A data frame produced by the function map_word_to_step
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixProcessConversion
</code></pre>


<h3>Format</h3>

<p>A data frame contains 64 variables
</p>

<dl>
<dt>degree</dt><dd><p>the number of the steps used in a article.</p>
</dd>
<dt>SpeciesName</dt><dd><p>1 means the output of Species name.</p>
</dd>
<dt>TaxonID</dt><dd><p>1 means the output of Taxon ID.</p>
</dd>
<dt>NCBI</dt><dd><p>1 means the output of NCBI.</p>
</dd>
<dt>Uniprot</dt><dd><p>1 means the output of Uniprot.</p>
</dd>
<dt>KEGG</dt><dd><p>1 means the output of KEGG.</p>
</dd>
<dt>MetaCyc</dt><dd><p>1 means the output of MetaCyc.</p>
</dd>
<dt>PubChem</dt><dd><p>1 means the output of PubChem.</p>
</dd>
<dt>Brenda</dt><dd><p>1 means the output of Brenda.</p>
</dd>
<dt>Download</dt><dd><p>1 means the output of Download.</p>
</dd>
<dt>GeneticInformation</dt><dd><p>1 means the output of Genetic information.</p>
</dd>
<dt>ProteinInformation</dt><dd><p>1 means the output of Protein information.</p>
</dd>
<dt>GenomeSequence</dt><dd><p>1 means the output of Genome sequence.</p>
</dd>
<dt>ProteinSequence</dt><dd><p>1 means the output of Protein sequence.</p>
</dd>
<dt>MetabolicFunctionInformation</dt><dd><p>1 means the output of Metabolic function information.</p>
</dd>
<dt>Metabolites</dt><dd><p>1 means the output of Metabolites.</p>
</dd>
<dt>Cofactor</dt><dd><p>1 means the output of Cofactor.</p>
</dd>
<dt>Nucleotides</dt><dd><p>1 means the output of Nucleotides.</p>
</dd>
<dt>AminoAcid</dt><dd><p>1 means the output of Amino acid.</p>
</dd>
<dt>MolecularWeight</dt><dd><p>1 means the output of Molecular weight.</p>
</dd>
<dt>DryWeight</dt><dd><p>1 means the output of Dry weight.</p>
</dd>
<dt>MetabolicReaction</dt><dd><p>1 means the output of Metabolic reaction.</p>
</dd>
<dt>TerminalMetabolite</dt><dd><p>1 means the output of Terminal metabolite.</p>
</dd>
<dt>Secretion</dt><dd><p>1 means the output of Secretion.</p>
</dd>
<dt>BiomassReaction</dt><dd><p>1 means the output of Biomass reaction.</p>
</dd>
<dt>DemandReaction</dt><dd><p>1 means the output of Demand reaction.</p>
</dd>
<dt>SinkReaction</dt><dd><p>1 means the output of Sink reaction.</p>
</dd>
<dt>GapReaction</dt><dd><p>1 means the output of Gap reaction.</p>
</dd>
<dt>SpontaneousReaction</dt><dd><p>1 means the output of Spontaneous reaction.</p>
</dd>
<dt>ExtracellularAndPeriplasmicTransportReactions</dt><dd><p>1 means the output of Extracellular and periplasmic transport reactions reaction.</p>
</dd>
<dt>ExchangeReaction</dt><dd><p>1 means the output of Exchange reaction.</p>
</dd>
<dt>IntracellularTransportReaction</dt><dd><p>1 means the output of Intracellular transport reaction.</p>
</dd>
<dt>ReactionFlux</dt><dd><p>1 means the output of Reaction flux.</p>
</dd>
<dt>GPR</dt><dd><p>1 means the output of GPR.</p>
</dd>
<dt>BlastComparison</dt><dd><p>1 means the output of Blast comparison.</p>
</dd>
<dt>Homology</dt><dd><p>1 means the output of Homology.</p>
</dd>
<dt>HomologousGene</dt><dd><p>1 means the output of Homology gene.</p>
</dd>
<dt>StoichiometricMatrix</dt><dd><p>1 means the output of Stoichiometric matrix.</p>
</dd>
<dt>Knockout</dt><dd><p>1 means the output of Knockout.</p>
</dd>
<dt>TargetReaction</dt><dd><p>1 means the output of Target reaction.</p>
</dd>
<dt>Restrictions</dt><dd><p>1 means the output of Restrictions.</p>
</dd>
<dt>GrowthConditions</dt><dd><p>1 means the output of Growth conditions.</p>
</dd>
<dt>MinORMax</dt><dd><p>1 means the output of min | max.</p>
</dd>
<dt>ReconstructionData</dt><dd><p>1 means the output of Reconstruction data.</p>
</dd>
<dt>FVA</dt><dd><p>1 means the output of FVA.</p>
</dd>
<dt>MetabolicFlux</dt><dd><p>1 means the output of Metabolic flux.</p>
</dd>
<dt>Statistics</dt><dd><p>1 means the output of Metabolic Statistics.</p>
</dd>
<dt>NeutralMolecularFormula</dt><dd><p>1 means the output of Neutral molecular formula.</p>
</dd>
<dt>ChargedMolecularFormula</dt><dd><p>1 means the output of Charged molecular formula.</p>
</dd>
<dt>LiteratureDataCollection</dt><dd><p>1 means the output of Literature data collection.</p>
</dd>
<dt>ConservationOfMassAndCharge</dt><dd><p>1 means the output of Conservation of mass and charge.</p>
</dd>
<dt>GibbsFreeEnergy</dt><dd><p>1 means the output of Gibbs free energy.</p>
</dd>
<dt>CellCompartmentInformation</dt><dd><p>1 means the output of Cell compartment information.</p>
</dd>
<dt>SubsystemInformation</dt><dd><p>1 means the output of Subsystem information.</p>
</dd>
<dt>MetaboliteIdentification</dt><dd><p>1 means the output of Metabolite Identification.</p>
</dd>
<dt>Unite</dt><dd><p>1 means the output of Unite.</p>
</dd>
<dt>ManualPlanning</dt><dd><p>1 means the output of Manual planning.</p>
</dd>
<dt>Coefficient</dt><dd><p>1 means the output of Coefficient.</p>
</dd>
<dt>ScatterPlot</dt><dd><p>1 means the output of Scatter plot.</p>
</dd>
<dt>TestReport</dt><dd><p>1 means the output of Test Report.</p>
</dd>
<dt>TypeIIIPath</dt><dd><p>1 means the output of Type III path.</p>
</dd>
<dt>SBML</dt><dd><p>1 means the output of SBML file.</p>
</dd>
<dt>Mat</dt><dd><p>1 means the output of Mat file.</p>
</dd>
<dt>Excel</dt><dd><p>1 means the output of Excel file.</p>
</dd>
</dl>


<hr>
<h2 id='matrixProcessFile'>A data frame produced by the function map_to_word in the function vizProcess</h2><span id='topic+matrixProcessFile'></span>

<h3>Description</h3>

<p>A data frame produced by the function map_to_word in the function vizProcess
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixProcessFile
</code></pre>


<h3>Format</h3>

<p>A data frame contains with 3 variables
</p>

<dl>
<dt>step</dt><dd><p>the steps used in the metabolic reconstruction</p>
</dd>
<dt>degree</dt><dd><p>the number that steps used in the metbolic reconstructoin occur in an article</p>
</dd>
<dt>step_ID</dt><dd><p>the order of the steps used in the metabolic reconstruction</p>
</dd>
</dl>


<hr>
<h2 id='slotsFunction'>slotsFunction</h2><span id='topic+slotsFunction'></span><span id='topic+slots'></span>

<h3>Description</h3>

<p>slotsFunction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slots(variablesFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slotsFunction_+3A_variablesfile">variablesFile</code></td>
<td>
<p>A data frame contains three different variables.</p>
</td></tr>
</table>

<hr>
<h2 id='stepsMatrix'>A data frame contains the details of the steps of the metabolic reconstruction</h2><span id='topic+stepsMatrix'></span>

<h3>Description</h3>

<p>A data frame contains the details of the steps of the metabolic reconstruction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stepsMatrix
</code></pre>


<h3>Format</h3>

<p>A data frame  with 93 rows and 70 variables:
</p>

<dl>
<dt>MarkerWords</dt><dd><p>the key words of the steps of metabolic reconstruction</p>
</dd>
<dt>ThresholdValue</dt><dd><p>the values filter the steps  after the mapping</p>
</dd>
<dt>Steps</dt><dd><p>the steps of metabolic reonstruction</p>
</dd>
<dt>Group</dt><dd><p>the classification of the steps</p>
</dd>
<dt>SpeciesName</dt><dd><p>-1 means the input of Species name. 1 means the output of Species name.</p>
</dd>
<dt>TaxonID</dt><dd><p>-1 means the input of Taxon ID. 1 means the output of Taxon ID.</p>
</dd>
<dt>KEGG</dt><dd><p>-1 means the input of KEGG reaction. 1 means the output of KEGG reaction.</p>
</dd>
<dt>NCBI</dt><dd><p>-1 means the input of NCBI gene. 1 means the output of NCBI gene.</p>
</dd>
<dt>Uniprot</dt><dd><p>-1 means the input of Uniprot gene. 1 means the output of Uniprot.</p>
</dd>
<dt>MetaCyc</dt><dd><p>-1 means the input of MetaCyc gene. 1 means the output of MetaCyc.</p>
</dd>
<dt>PubChem</dt><dd><p>-1 means the input of PubChem molecular formula. 1 means the output of PubChem molecular formula.</p>
</dd>
<dt>Brenda</dt><dd><p>identify whether input the Brenda. -1 means the input of Brenda. 1 means the output of Brenda.</p>
</dd>
<dt>Chebi</dt><dd><p> -1 means the input of Chebi information. 1 means the the output of Chebi information.</p>
</dd>
<dt>WebGCM</dt><dd><p>-1 means the input of Web GCM. 1 means the the output of Web GCM.</p>
</dd>
<dt>SpontaneousReaction</dt><dd><p>-1 means the input of spontaneous reaction. 1 means the the output of spontaneous reaction.</p>
</dd>
<dt>ExtracellularAndPeriplasmicTransportReactions</dt><dd><p>-1 means the input of Extracellular and periplasmic transport reactions.
1 means the the output of Extracellular and periplasmic transport reactions.</p>
</dd>
<dt>ExchangeReaction</dt><dd><p>-1 means the input of exchange reaction. 1 means the the output of exchange reaction.</p>
</dd>
<dt>MissingExchangeReaction</dt><dd><p>-1 means the input of missing exchange reaction. 1 means the the output of missing exchange reaction.</p>
</dd>
<dt>IntracellularTransportReaction</dt><dd><p>-1 means the input of intracellular transport reaction. 1 means the the output of intracellular transport reaction.</p>
</dd>
<dt>Gene</dt><dd><p>-1 means the input of gene. 1 means the the output of gene.</p>
</dd>
<dt>Protein</dt><dd><p>-1 means the input of protein. 1 means the the output of protein.</p>
</dd>
<dt>Knockout</dt><dd><p>-1 means the input of knockout. 1 means the the output of knockout.</p>
</dd>
<dt>StoichiometricMatrix</dt><dd><p>-1 means the input of stoichiometric matrix. 1 means the the output of stoichiometric matrix.</p>
</dd>
<dt>ObjectiveReaction</dt><dd><p>-1 means the input of objective reaction. 1 means the the output of objective reaction.</p>
</dd>
<dt>Constraints</dt><dd><p>-1 means the input of constraints. 1 means the the output of constraints.</p>
</dd>
<dt>Secretion</dt><dd><p>-1 means the input of secretion. 1 means the the output of secretion.</p>
</dd>
<dt>Mutisecretion</dt><dd><p>-1 means the input of mutisecretion. 1 means the the output of mutisecretion.</p>
</dd>
<dt>RichMedia</dt><dd><p>-1 means the input of rich media. 1 means the the output of rich media.</p>
</dd>
<dt>GenomeSequence</dt><dd><p>-1 means the input of genome sequence. 1 means the the output of genome sequence.</p>
</dd>
<dt>ProteomeSequence</dt><dd><p>-1 means the input of proteome sequence. 1 means the the output of proteome sequence.</p>
</dd>
<dt>AminoAcidWeight</dt><dd><p>-1 means the input of amino acid weight. 1 means the the output of amino acid weight.</p>
</dd>
<dt>AminoAcidMolecularWeight</dt><dd><p>-1 means the input of amino acid molecular weight. 1 means the the output of amino acid molecular weight.</p>
</dd>
<dt>NucleotideWeight</dt><dd><p>-1 means the input of nucleotide weight. 1 means the the output of nucleotide weight.</p>
</dd>
<dt>NucleotideMolecularWeight</dt><dd><p>-1 means the input of nucleotide molecular weight. 1 means the the output of nucleotide molecular weight.</p>
</dd>
<dt>DryWeight</dt><dd><p>-1 means the input of dry weight. 1 means the the output of dry weight.</p>
</dd>
<dt>BiomassReaction</dt><dd><p>-1 means the input of biomass reaction. 1 means the the output of biomass reaction.</p>
</dd>
<dt>DemandReaction</dt><dd><p>-1 means the input of demand reaction. 1 means the the output of demand reaction.</p>
</dd>
<dt>SinkReaction</dt><dd><p>-1 means the input of sink reaction. 1 means the the output of sink reaction.</p>
</dd>
<dt>GapReaction</dt><dd><p>-1 means the input of gap reaction. 1 means the the output of gap reaction.</p>
</dd>
<dt>MinORMax</dt><dd><p>-1 means the input of gap reaction. 1 means the the output of gap reaction.</p>
</dd>
<dt>GeneticInformation</dt><dd><p>-1 means the input of genetic information. 1 means the the output of genetic information.</p>
</dd>
<dt>MetabolicFunction</dt><dd><p>-1 means the input of metabolic function. 1 means the the output of metabolic function.</p>
</dd>
<dt>Metabolites</dt><dd><p>-1 means the input of metabolites. 1 means the the output of metabolites.</p>
</dd>
<dt>BiomassMetabolites</dt><dd><p>-1 means the input of biomass metabolites. 1 means the the output of biomass metabolites.</p>
</dd>
<dt>MetabolicReaction</dt><dd><p>-1 means the input of metabolic reaction. 1 means the the output of metabolic reaction.</p>
</dd>
<dt>ReconstructionData</dt><dd><p>-1 means the input of reconstruction data. 1 means the the output of reconstruction data.</p>
</dd>
<dt>DataStatistics</dt><dd><p>-1 means the input of data statistics. 1 means the the output of data statistics.</p>
</dd>
<dt>NeutralMolecularFormula</dt><dd><p>-1 means the input of neutral molecular formula. 1 means the the output of neutral molecular formula.</p>
</dd>
<dt>ChargedMolecularFormula</dt><dd><p>-1 means the input of charged molecular formula. 1 means the the output of charged molecular formula.</p>
</dd>
<dt>ConservationOfMassAndCharge</dt><dd><p>-1 means the input of conservation of mass and charge. 1 means the the output of conservation of mass and charge.</p>
</dd>
<dt>GibbsFreeEnergyInformation</dt><dd><p>-1 means the input of conservation of gibbs free energy information. 1 means the the output of gibbs free energy information.</p>
</dd>
<dt>Mass_ChargeConservationAssessment</dt><dd><p>-1 means the input of mass-charge conservation assessment. 1 means the the output of mass-charge conservation assessment.</p>
</dd>
<dt>CellularCompartment</dt><dd><p>-1 means the input of cellular compartment. 1 means the the output of cellular compartment.</p>
</dd>
<dt>SubsystemInformation</dt><dd><p>-1 means the input of subsystem information. 1 means the the output of subsystem information.</p>
</dd>
<dt>IdentifiersInKEGG</dt><dd><p>-1 means the input of subsystem information. 1 means the the output of subsystem information.</p>
</dd>
<dt>IdentifiersInMetaCyc</dt><dd><p>-1 means the input of identifiers in MetaCyc. 1 means the the output of identifiers in MetaCyc.</p>
</dd>
<dt>UniformIdentifier</dt><dd><p>-1 means the input of uniform identifier. 1 means the the output of uniform identifier.</p>
</dd>
<dt>Coefficient</dt><dd><p>-1 means the input of coefficient. 1 means the the output of coefficient.</p>
</dd>
<dt>ScatterPlot</dt><dd><p>-1 means the input of scatter plot. 1 means the the output of scatter plot.</p>
</dd>
<dt>TerminalMetabolite</dt><dd><p>-1 means the input of terminal metabolite. 1 means the the output of terminal metabolite.</p>
</dd>
<dt>TypeIIIPathway</dt><dd><p>-1 means the input of Type III pathway. 1 means the the output of Type III pathway.</p>
</dd>
<dt>NetworkGap</dt><dd><p>-1 means the input of network gap. 1 means the the output of network gap.</p>
</dd>
<dt>Growth</dt><dd><p>-1 means the input of growth. 1 means the the output of growth.</p>
</dd>
<dt>BlockReaction</dt><dd><p>-1 means the input of block reaction. 1 means the the output of block reaction.</p>
</dd>
<dt>MetabolicFlux</dt><dd><p>-1 means the input of metabolic flux. 1 means the the output of metabolic flux.</p>
</dd>
<dt>ModelPredictCorrectly</dt><dd><p>-1 means the input of model predict correctly. 1 means the the output of model predict correctly.</p>
</dd>
<dt>ModelGrowingTooFast</dt><dd><p>-1 means the input of model growing too fast. 1 means the the output of model growing too fast.</p>
</dd>
<dt>SBML</dt><dd><p>-1 means the input of SBML file. 1 means the the output of SBML file.</p>
</dd>
<dt>Mat</dt><dd><p>-1 means the input of Mat file. 1 means the the output of Mat file.</p>
</dd>
<dt>Excel</dt><dd><p>-1 means the input of Excel file. 1 means the the output of Excel file.</p>
</dd>
</dl>


<hr>
<h2 id='stepTypes'>A data frame contains the labels and groups of the steps of metabolic reconstruction</h2><span id='topic+stepTypes'></span>

<h3>Description</h3>

<p>A data frame contains the labels and groups of the steps of metabolic reconstruction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stepTypes
</code></pre>


<h3>Format</h3>

<p>A data frame contains 93 rows and 2 variables:
</p>

<dl>
<dt>label</dt><dd><p>the steps of the metabolic reconstruction</p>
</dd>
<dt>group</dt><dd><p>the classfication of the labels</p>
</dd>
</dl>


<hr>
<h2 id='Sysrecon'>Sysrecon</h2><span id='topic+Sysrecon'></span>

<h3>Description</h3>

<p>Input the txt and output the visualization of the steps, transformation and databases and tools.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sysrecon(
  inputTxt,
  stepsMatrix,
  stepTypes,
  conversionMatrix,
  conversionTypes,
  toolsMatrix,
  toolsTypes,
  contentTypes
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Sysrecon_+3A_inputtxt">inputTxt</code></td>
<td>
<p>A txt contains the methods and contents of the metabolic reconstruction in articles.</p>
</td></tr>
<tr><td><code id="Sysrecon_+3A_stepsmatrix">stepsMatrix</code></td>
<td>
<p>A data frame the marker words, threshold value, steps, group and other information about the metabolic reconstruction. The default file is in the data.</p>
</td></tr>
<tr><td><code id="Sysrecon_+3A_steptypes">stepTypes</code></td>
<td>
<p>A data frame the labels and groups of the metabolic reconstructions steps. The default file is in the data.</p>
</td></tr>
<tr><td><code id="Sysrecon_+3A_conversionmatrix">conversionMatrix</code></td>
<td>
<p>A data frame contains the marker words, threshold value, steps, group and other transformation information about the metabolic reconstruction. The default file is in the data.</p>
</td></tr>
<tr><td><code id="Sysrecon_+3A_conversiontypes">conversionTypes</code></td>
<td>
<p>A data frame contains the labels and groups of the metabolic reconstructions transformation. The default file is in the data.</p>
</td></tr>
<tr><td><code id="Sysrecon_+3A_toolsmatrix">toolsMatrix</code></td>
<td>
<p>A data frame contains the marker words, threshold value, steps, group and other information about the metabolic reconstruction databases and tools. The default file is in the data.</p>
</td></tr>
<tr><td><code id="Sysrecon_+3A_toolstypes">toolsTypes</code></td>
<td>
<p>A data frame contains the databases and the tools used in the metabolic reconstruction.</p>
</td></tr>
<tr><td><code id="Sysrecon_+3A_contenttypes">contentTypes</code></td>
<td>
<p>A data frame contains the labels and groups of the metabolic reconstructions content The default file is in the data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Input takes a data.frame x with two variables v and w and returns the maximum knapsack value and which elements (rows in the data.frame).
</p>


<h3>Value</h3>

<p>The pictures that visualize the steps, transformation and databases and tools of the metabolic reconstruction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
exam &lt;- Sysrecon(inputTxt, stepsMatrix, stepTypes, conversionMatrix, conversionTypes,
  toolsMatrix, toolsTypes, contentTypes)
</code></pre>

<hr>
<h2 id='systemrecon-class'>systemrecon-class</h2><span id='topic+systemrecon-class'></span><span id='topic+label'></span>

<h3>Description</h3>

<p>systemrecon-class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>label
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 72.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Taxon_id</code></dt><dd><p>Taxonomy ID</p>
</dd>
<dt><code>NCBI_gene</code></dt><dd><p>Gene information from NCBI</p>
</dd>
<dt><code>Uniprot_gene</code></dt><dd><p>Gene information from Uniprot</p>
</dd>
<dt><code>Genetic_information</code></dt><dd><p>Genetic information</p>
</dd>
<dt><code>KEGG_reaction</code></dt><dd><p>Reactions in KEGG</p>
</dd>
<dt><code>MetaCyc_reaction</code></dt><dd><p>Reactions in MetaCyc</p>
</dd>
<dt><code>Metabolic_function</code></dt><dd><p>Metabolic function</p>
</dd>
<dt><code>Metabolic_reaction</code></dt><dd><p>Metabolic reaction</p>
</dd>
<dt><code>Metabolites_id</code></dt><dd><p>Metabolites ID</p>
</dd>
<dt><code>Reconstruction_data</code></dt><dd><p>Reconstruction data</p>
</dd>
<dt><code>Genome_sequence</code></dt><dd><p>Genome sequence</p>
</dd>
<dt><code>Genome_sequence_reference</code></dt><dd><p>Reference of genome sequence</p>
</dd>
<dt><code>Reconstruction_data_reference</code></dt><dd><p>Reference of reconstruction data</p>
</dd>
<dt><code>Data_statistics_metabolites</code></dt><dd><p>Data statistics of metabolites</p>
</dd>
<dt><code>PubChem_molecular_formula</code></dt><dd><p>Molecular formula in PubChem database</p>
</dd>
<dt><code>Brenda_molecular_formula</code></dt><dd><p>Molecular formula in Brenda database</p>
</dd>
<dt><code>Neutral_molecular_formula</code></dt><dd><p>Neutral molecular formula</p>
</dd>
<dt><code>Charged_molecular_formula</code></dt><dd><p>Charged molecular formula</p>
</dd>
<dt><code>Chebi_molecular_formula</code></dt><dd><p>Chebi molecular formula</p>
</dd>
<dt><code>Conservation_of_mass_and_charge</code></dt><dd><p>Conservation of mass and charge</p>
</dd>
<dt><code>Web_GCM_Gibbs</code></dt><dd><p>Web GCM Gibbs</p>
</dd>
<dt><code>Gibbs_free_energy_information</code></dt><dd><p>Information of gibbs free energy</p>
</dd>
<dt><code>Gene_id</code></dt><dd><p>Gene ID</p>
</dd>
<dt><code>Protein_id</code></dt><dd><p>Protein ID</p>
</dd>
<dt><code>Cellular_compartment</code></dt><dd><p>Information of cellualr compartment</p>
</dd>
<dt><code>Subsystem_information</code></dt><dd><p>Subsystem information</p>
</dd>
<dt><code>Uniform_identifier_metabolites</code></dt><dd><p>Uniform identifiers of  metabollites</p>
</dd>
<dt><code>Spontaneous_reactions</code></dt><dd><p>Spontaneous reactions</p>
</dd>
<dt><code>Extracellular_and_periplasmic_transport_reactions</code></dt><dd><p>Extracellular and periplasmic transport reactions</p>
</dd>
<dt><code>Exchange_reactions</code></dt><dd><p>Exchange reactions</p>
</dd>
<dt><code>Intracellular_transport_reactions</code></dt><dd><p>ntracellular transport reactions</p>
</dd>
<dt><code>Amino_acid_weight</code></dt><dd><p>Amino acid weight</p>
</dd>
<dt><code>Amino_acid_molecular_weight</code></dt><dd><p>Amino acid_molecular weight</p>
</dd>
<dt><code>Dry_weight</code></dt><dd><p>Dry weight</p>
</dd>
<dt><code>Amino_acid_coefficient</code></dt><dd><p>Amino acid coefficient</p>
</dd>
<dt><code>Nucleotide_coefficient</code></dt><dd><p>Nucleotide coefficient</p>
</dd>
<dt><code>Nucleotide_weight</code></dt><dd><p>Nucleotide weight</p>
</dd>
<dt><code>Nucleotide_molecular_weight</code></dt><dd><p>Nucleotide molecular weight</p>
</dd>
<dt><code>Biomass_reactions</code></dt><dd><p>Biomass reactions</p>
</dd>
<dt><code>Demand_reactions</code></dt><dd><p>Demand reactions</p>
</dd>
<dt><code>Sink_reactions</code></dt><dd><p>Sink reactions</p>
</dd>
<dt><code>Scatter_plot_stoichiometric_matrix</code></dt><dd><p>Scatter plot of stoichiometric matrix</p>
</dd>
<dt><code>Objective_reaction</code></dt><dd><p>Objective reaction</p>
</dd>
<dt><code>Constraints</code></dt><dd><p>Constraints of the model</p>
</dd>
<dt><code>Mass_charge_conservation_assessment</code></dt><dd><p>assessment of conservation of mass and charge</p>
</dd>
<dt><code>Terminal_metabolites</code></dt><dd><p>Terminal metabolites</p>
</dd>
<dt><code>Gap_reactions</code></dt><dd><p>Gap reactions</p>
</dd>
<dt><code>Missing_exchange_reactions</code></dt><dd><p>Missing exchange reactions</p>
</dd>
<dt><code>Type_III_pathway</code></dt><dd><p>Type III pathway</p>
</dd>
<dt><code>Network_gaps</code></dt><dd><p>Network gaps</p>
</dd>
<dt><code>Biomass_metabolites</code></dt><dd><p>Biomass metabolites</p>
</dd>
<dt><code>miniaml_or_maxiaml</code></dt><dd><p>miniaml or maxiaml</p>
</dd>
<dt><code>Metabolic_flux_value</code></dt><dd><p>Value of metabolic flux</p>
</dd>
<dt><code>Growth</code></dt><dd><p>Prediction of the growth of model</p>
</dd>
<dt><code>Secretion_product</code></dt><dd><p>Product of secretion</p>
</dd>
<dt><code>Mutisecretion</code></dt><dd><p>Multisecretion</p>
</dd>
<dt><code>Rich_media</code></dt><dd><p>Environment of rich media</p>
</dd>
<dt><code>Block_reactions</code></dt><dd><p>Block reactions</p>
</dd>
<dt><code>Knockout</code></dt><dd><p>Knockout the single gene or reaction</p>
</dd>
<dt><code>Model_predict_correctly</code></dt><dd><p>Predict the model correctly</p>
</dd>
<dt><code>Model_growing_too_fast</code></dt><dd><p>Assess whether the model grow too fast</p>
</dd>
<dt><code>Cofactors</code></dt><dd><p>Cofactors</p>
</dd>
<dt><code>FBA</code></dt><dd><p>Flux balance analysis</p>
</dd>
<dt><code>GPR</code></dt><dd><p>Gene-protein reaction</p>
</dd>
<dt><code>Output</code></dt><dd><p>Output the file</p>
</dd>
<dt><code>Iteration</code></dt><dd><p>Iteration</p>
</dd>
<dt><code>Test</code></dt><dd><p>Test</p>
</dd>
<dt><code>Assessment</code></dt><dd><p>Assessment</p>
</dd>
<dt><code>Identifiers_metabolites</code></dt><dd><p>Identifiers of metabolites</p>
</dd>
<dt><code>Output_file</code></dt><dd><p>Output the file</p>
</dd>
<dt><code>Elemental_balance</code></dt><dd><p>Elemental balance</p>
</dd>
<dt><code>Biomass_metabolites_coefficient</code></dt><dd><p>Coefficients of biomass and metabolites</p>
</dd>
</dl>

<hr>
<h2 id='text'>A list of characters from the inputTxt.</h2><span id='topic+text'></span>

<h3>Description</h3>

<p>A list of characters from the inputTxt.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>text
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 1.
</p>

<hr>
<h2 id='toolsMatrix'>A data frame contains the details of the databases and tools of the metabolic reconstruction</h2><span id='topic+toolsMatrix'></span>

<h3>Description</h3>

<p>A data frame contains the details of the databases and tools of the metabolic reconstruction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toolsMatrix
</code></pre>


<h3>Format</h3>

<p> A data frame contains the 93 rows and 46 varibales.
</p>
<dl>
<dt>MarkerWords</dt><dd><p>the key words of the steps of metabolic reconstruction</p>
</dd>
<dt>ThresholdValue</dt><dd><p>the values filter the steps  after the mapping</p>
</dd>
<dt>Steps</dt><dd><p>the steps of metabolic reonstruction</p>
</dd>
<dt>Group</dt><dd><p>the classification of the steps</p>
</dd>
<dt>UniProtKnowledgeBase</dt><dd><p>1 means the output of UniProt Knowledgebase.</p>
</dd>
<dt>NCBI_Gene</dt><dd><p>1 means the output of NCBI Gene.</p>
</dd>
<dt>KEGG_Genes</dt><dd><p>1 means the output of KEGG Genes.</p>
</dd>
<dt>KEGG_Genome</dt><dd><p>1 means the output of KEGG Genome.</p>
</dd>
<dt>NCBI_Protein</dt><dd><p>1 means the output of NCBI Protein.</p>
</dd>
<dt>KEGG_Pathway</dt><dd><p>1 means the output of KEGG Pathway.</p>
</dd>
<dt>KEGG_Compound</dt><dd><p>1 means the output of KEGG Compound.</p>
</dd>
<dt>KEGG_Reaction</dt><dd><p>1 means the output of KEGG Reaction.</p>
</dd>
<dt>BioCyc</dt><dd><p>1 means the output of BioCyc.</p>
</dd>
<dt>MetaCycCompound</dt><dd><p>1 means the output of MetaCyc Compound.</p>
</dd>
<dt>MetaCycReaction</dt><dd><p>1 means the output of MetaCyc Reaction.</p>
</dd>
<dt>KEGGREST</dt><dd><p>1 means the output of KEGGREST.</p>
</dd>
<dt>COBRA</dt><dd><p>1 means the output of COBRA.</p>
</dd>
<dt>RAVEN</dt><dd><p>1 means the output of RAVEN.</p>
</dd>
<dt>CarveMe</dt><dd><p>1 means the output of CarveMe.</p>
</dd>
<dt>AuReMe</dt><dd><p>1 means the output of AuReMe.</p>
</dd>
<dt>MetaDraft</dt><dd><p>1 means the output of MetaDraft.</p>
</dd>
<dt>ModelSEED</dt><dd><p>1 means the output of ModelSEED.</p>
</dd>
<dt>PathwayTools</dt><dd><p>1 means the output of Pathway Tools.</p>
</dd>
<dt>Merlin</dt><dd><p>1 means the output of Merlin.</p>
</dd>
<dt>AGORA</dt><dd><p>1 means the output of AGORA.</p>
</dd>
<dt>COBRApy</dt><dd><p>1 means the output of COBRApy.</p>
</dd>
<dt>BLAST</dt><dd><p>1 means the output of BLAST.</p>
</dd>
<dt>ExperimentOrLiterature</dt><dd><p>1 means the output of Experiment or literature.</p>
</dd>
<dt>BRENDA</dt><dd><p>1 means the output of BRENDA.</p>
</dd>
<dt>minval</dt><dd><p>1 means the output of minval.</p>
</dd>
<dt>ChEBI</dt><dd><p>1 means the output of ChEBI.</p>
</dd>
<dt>PubChem_Compound</dt><dd><p>1 means the output of PubChem-compound.</p>
</dd>
<dt>ChEMBL_Compound</dt><dd><p>1 means the output of ChEMBL compound.</p>
</dd>
<dt>Rhea</dt><dd><p>1 means the output of Rhea.</p>
</dd>
<dt>pKa_DB</dt><dd><p>1 means the output of pKa DB.</p>
</dd>
<dt>rsbml</dt><dd><p>1 means the output of rsbml.</p>
</dd>
<dt>PipelinePilot</dt><dd><p>1 means the output of Pipeline Pilot.</p>
</dd>
<dt>Sybil</dt><dd><p>1 means the output of Sybil.</p>
</dd>
<dt>BUSCA</dt><dd><p>1 means the output of BUSCA.</p>
</dd>
<dt>PSORT</dt><dd><p>1 means the output of PSORT.</p>
</dd>
<dt>PASUB</dt><dd><p>1 means the output of PASUB.</p>
</dd>
<dt>PubChem_Substance</dt><dd><p>1 means the output of PubChem substance.</p>
</dd>
<dt>STRING</dt><dd><p>1 means the output of STRING.</p>
</dd>
<dt>ManualPlanning</dt><dd><p>1 means the output of Manual planning.</p>
</dd>
<dt>CMR</dt><dd><p>1 means the output of CMR.</p>
</dd>
<dt>g2f</dt><dd><p>1 means the output of g2f.</p>
</dd>
</dl>


<hr>
<h2 id='toolsTypes'>A data frame contains the labels and groups of the databases and tools of metabolic reconstruction</h2><span id='topic+toolsTypes'></span>

<h3>Description</h3>

<p>A data frame contains the labels and groups of the databases and tools of metabolic reconstruction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toolsTypes
</code></pre>


<h3>Format</h3>


<dl>
<dt>label</dt><dd><p>the databases and tools of the metabolic reconstruction</p>
</dd>
<dt>group</dt><dd><p>the classfication of the labels</p>
</dd>
</dl>


<hr>
<h2 id='vizProcess'>vizProcess</h2><span id='topic+vizProcess'></span>

<h3>Description</h3>

<p>Create a function can visualiaze the steps used in the metabolic reconstruction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vizProcess(text, stepsMatrix, stepTypes, contentTypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vizProcess_+3A_text">text</code></td>
<td>
<p>The characters processed with the collapse = ' '.</p>
</td></tr>
<tr><td><code id="vizProcess_+3A_stepsmatrix">stepsMatrix</code></td>
<td>
<p>A data frame contains the marker words, threshold value, steps, group and other information about the metabolic reconstruction. The default file is in the data.</p>
</td></tr>
<tr><td><code id="vizProcess_+3A_steptypes">stepTypes</code></td>
<td>
<p>A data frame contains the labels and groups of the metabolic reconstructions steps. The default file is in the data.</p>
</td></tr>
<tr><td><code id="vizProcess_+3A_contenttypes">contentTypes</code></td>
<td>
<p>A data frame contains the labels and groups of the metabolic reconstructions content The default file is in the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The pictures that visualize the steps of the metabolic reconstruction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exam &lt;- vizProcess(text, stepsMatrix, stepTypes,contentTypes)
</code></pre>

<hr>
<h2 id='vizTools'>vizTools</h2><span id='topic+vizTools'></span>

<h3>Description</h3>

<p>Input the txt and output the visualization of the steps, transformation and databases and tools.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vizTools(text, toolsMatrix, stepTypes, toolsTypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vizTools_+3A_text">text</code></td>
<td>
<p>The characters processed with the collapse = ' '.</p>
</td></tr>
<tr><td><code id="vizTools_+3A_toolsmatrix">toolsMatrix</code></td>
<td>
<p>A data frame contains the marker words, threshold value, steps, group and other information about the metabolic reconstruction databases and tools. The default file is in the data.</p>
</td></tr>
<tr><td><code id="vizTools_+3A_steptypes">stepTypes</code></td>
<td>
<p>A data frame contains the labels and groups of the metabolic reconstructions steps. The default file is in the data.</p>
</td></tr>
<tr><td><code id="vizTools_+3A_toolstypes">toolsTypes</code></td>
<td>
<p>A data frame contains the databases and the tools used in the metabolic reconstruction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The pictures that visualize the databases and tools of the metabolic reconstruction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exam &lt;- vizTools(text, toolsMatrix, stepTypes, toolsTypes)
</code></pre>

<hr>
<h2 id='vizTransformation'>vizTransformation</h2><span id='topic+vizTransformation'></span>

<h3>Description</h3>

<p>Input the txt and output the visualization of the steps, transformation and databases and tools.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vizTransformation(text, conversionMatrix, stepTypes, conversionTypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vizTransformation_+3A_text">text</code></td>
<td>
<p>The characters processed with the collapse = ' '.</p>
</td></tr>
<tr><td><code id="vizTransformation_+3A_conversionmatrix">conversionMatrix</code></td>
<td>
<p>A data frame contains the marker words, threshold value, steps, group and other transformation information about the metabolic reconstruction. The default file is in the data.</p>
</td></tr>
<tr><td><code id="vizTransformation_+3A_steptypes">stepTypes</code></td>
<td>
<p>A data frame contains the labels and groups of the metabolic reconstructions steps. The default file is in the data.</p>
</td></tr>
<tr><td><code id="vizTransformation_+3A_conversiontypes">conversionTypes</code></td>
<td>
<p>A data frame contains the labels and groups of the metabolic reconstructions transformation. The default file is in the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The pictures that visualize the transformation of the metabolic reconstruction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exam &lt;- vizTransformation(text, conversionMatrix, stepTypes, conversionTypes)
</code></pre>

<hr>
<h2 id='wordsMatrix'>A data frame produced by the function get_term_matrix</h2><span id='topic+wordsMatrix'></span>

<h3>Description</h3>

<p>A data frame produced by the function get_term_matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wordsMatrix
</code></pre>


<h3>Format</h3>

<p>A data frame contains 4 variables
</p>

<dl>
<dt>freq</dt><dd><p>the frequency of the word occurs in a article.</p>
</dd>
<dt>prevalent</dt><dd><p>the words type that often occurs in a article.</p>
</dd>
<dt>longest</dt><dd><p>the longest type of a word.</p>
</dd>
<dt>shortest</dt><dd><p>the shortest type of a word.</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
