<!DOCTYPE html><html><head><title>Help for package TANDEM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TANDEM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#coef.tandem'><p>Returns the regression coefficients from a TANDEM fit</p></a></li>
<li><a href='#example_data'><p>A small artificial data set</p></a></li>
<li><a href='#nested.cv'><p>Estimating predictive performance via nested cross-validation</p></a></li>
<li><a href='#predict.tandem'><p>Creates a prediction using a tandem-object</p></a></li>
<li><a href='#relative.contributions'><p>Determine the relative contribution per data type</p></a></li>
<li><a href='#tandem'><p>Fits a TANDEM model by performing a two-stage regression</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Two-Stage Approach to Maximize Interpretability of Drug
Response Models Based on Multiple Molecular Data Types</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-11-18</td>
</tr>
<tr>
<td>Author:</td>
<td>Nanne Aben</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nanne Aben &lt;nanne.aben@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A two-stage regression method that can be used when various input data types are correlated, for example gene expression and methylation in drug response prediction. In the first stage it uses the upstream features (such as methylation) to predict the response variable (such as drug response), and in the second stage it uses the downstream features (such as gene expression) to predict the residuals of the first stage. In our manuscript (Aben et al., 2016, &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw449">doi:10.1093/bioinformatics/btw449</a>&gt;), we show that using TANDEM prevents the model from being dominated by gene expression and that the features selected by TANDEM are more interpretable.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet (&ge; 3.0), Matrix</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.0</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-11-25 12:22:51 UTC; nanneaben</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-25 15:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='coef.tandem'>Returns the regression coefficients from a TANDEM fit</h2><span id='topic+coef.tandem'></span>

<h3>Description</h3>

<p>Returns the regression coefficients from a TANDEM fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tandem'
coef(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coef.tandem_+3A_object">object</code></td>
<td>
<p>A tandem-object, as returned by tandem()</p>
</td></tr>
<tr><td><code id="coef.tandem_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments for coef().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The regression coefficients.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># unpack example data
x = example_data$x
y = example_data$y
upstream = example_data$upstream

# fit a tandem model, determine the coefficients and create a prediction
fit = tandem(x, y, upstream, alpha=0.5)
beta = coef(fit)
y_hat = predict(fit, newx=x)
</code></pre>

<hr>
<h2 id='example_data'>A small artificial data set</h2><span id='topic+example_data'></span>

<h3>Description</h3>

<p>A small artificial data set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_data
</code></pre>


<h3>Format</h3>

<p>A small artificial data set, containing 200 samples, 20 upstream features and 20 downstream features
</p>

<dl>
<dt>x</dt><dd><p>A 200 x 40 feature matrix</p>
</dd>
<dt>y</dt><dd><p>A 200 x 1 response vector</p>
</dd>
<dt>upstream</dt><dd><p>A 40 x 1 boolean vector indicating which features are upstream features</p>
</dd>
<dt>data_types</dt><dd><p>A 40 x 1 vector indicating for each feature to which data type they belong</p>
</dd>
</dl>

<hr>
<h2 id='nested.cv'>Estimating predictive performance via nested cross-validation</h2><span id='topic+nested.cv'></span>

<h3>Description</h3>

<p>Performs a nested cross-validation to assess the predictive performance. The inner loop is used to determine the optimal lambda
(as in cv.glmnet) and the outer loop is used to asses the predictive performance in an unbiased way.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nested.cv(
  x,
  y,
  upstream,
  method = "tandem",
  family = "gaussian",
  nfolds = 10,
  nfolds_inner = 10,
  foldid = NULL,
  lambda_upstream = "lambda.1se",
  lambda_downstream = "lambda.1se",
  lambda_glmnet = "lambda.1se",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nested.cv_+3A_x">x</code></td>
<td>
<p>A feature matrix, where the rows correspond to samples and the columns to features.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_y">y</code></td>
<td>
<p>A vector containing the response.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_upstream">upstream</code></td>
<td>
<p>A logical index vector that indicates for each feature whether it's upstream (TRUE) or downstream (FALSE).</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_method">method</code></td>
<td>
<p>Indicates whether the nested cross-validation is performed on TANDEM or on the classic approach (glmnet). Should be either &quot;tandem&quot; or &quot;glmnet&quot;.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_family">family</code></td>
<td>
<p>The family parameter that's passed to cv.glmnet(). Currently, only family='gaussian' is supported.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_nfolds">nfolds</code></td>
<td>
<p>Number of cross-validation folds (default is 10) used in the outer cross-validation loop.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_nfolds_inner">nfolds_inner</code></td>
<td>
<p>Number of cross-validation folds (default is 10) used to determine the optimal lambda in the inner cross-validation loop.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_foldid">foldid</code></td>
<td>
<p>An optional vector indicating in which cross-validation fold each sample should be in the outer cross-validation loop. Overrides nfolds when used.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_lambda_upstream">lambda_upstream</code></td>
<td>
<p>Only used when method='tandem'. For the first stage (using the upstream features), should glmnet use lambda.min or lambda.1se? Default is lambda.1se.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_lambda_downstream">lambda_downstream</code></td>
<td>
<p>Only used when method='tandem'. For the second stage (using the downstream features), should glmnet use lambda.min or lambda.1se? Default is lambda.1se.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_lambda_glmnet">lambda_glmnet</code></td>
<td>
<p>Only used when method='glmnet'. Should glmnet use lambda.min or lambda.1se? Default is lambda.1se.</p>
</td></tr>
<tr><td><code id="nested.cv_+3A_...">...</code></td>
<td>
<p>Other parameters that are passed to cv.glmnet().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The predicted response vector y_hat and the mean-squared error (MSE).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># unpack example data
x = example_data$x
y = example_data$y
upstream = example_data$upstream

# assess the prediction error in a nested cv-loop
# fix the seed to have the same foldids between the two methods
set.seed(1)
cv_tandem = nested.cv(x, y, upstream, method="tandem", alpha=0.5)
set.seed(1)
cv_glmnet = nested.cv(x, y, upstream, method="glmnet", alpha=0.5)
barplot(c(cv_tandem$mse, cv_glmnet$mse), ylab="MSE", names=c("TANDEM", "Classic approach"))
</code></pre>

<hr>
<h2 id='predict.tandem'>Creates a prediction using a tandem-object</h2><span id='topic+predict.tandem'></span>

<h3>Description</h3>

<p>Creates a prediction using a tandem-object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tandem'
predict(object, newx, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.tandem_+3A_object">object</code></td>
<td>
<p>A tandem-object, as returned by tandem()</p>
</td></tr>
<tr><td><code id="predict.tandem_+3A_newx">newx</code></td>
<td>
<p>A feature matrix, where the rows correspond to samples and the columns to features.</p>
</td></tr>
<tr><td><code id="predict.tandem_+3A_...">...</code></td>
<td>
<p>Not used. Other arguments for predict().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The predicted response vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># unpack example data
x = example_data$x
y = example_data$y
upstream = example_data$upstream

# fit a tandem model, determine the coefficients and create a prediction
fit = tandem(x, y, upstream, alpha=0.5)
beta = coef(fit)
y_hat = predict(fit, newx=x)
</code></pre>

<hr>
<h2 id='relative.contributions'>Determine the relative contribution per data type</h2><span id='topic+relative.contributions'></span>

<h3>Description</h3>

<p>For each data type, determine its relative contribution to the overall prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relative.contributions(fit, x, data_types, lambda_glmnet = "lambda.1se")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relative.contributions_+3A_fit">fit</code></td>
<td>
<p>Either a tandem-object or a cv.glmnet-object</p>
</td></tr>
<tr><td><code id="relative.contributions_+3A_x">x</code></td>
<td>
<p>The feature matrix used to train the fit, where the rows correspond to samples and the columns to features.</p>
</td></tr>
<tr><td><code id="relative.contributions_+3A_data_types">data_types</code></td>
<td>
<p>A vector of the same length as the number of features, that indicates for each feature to which data
type it belongs. This vector doesn't need to correspond to the 'upstream' vector used in tandem(). For example, the upstream
features be spread across various data types (such as mutation, CNA, methylation and cancer type) and the downstream features
could be gene expression.</p>
</td></tr>
<tr><td><code id="relative.contributions_+3A_lambda_glmnet">lambda_glmnet</code></td>
<td>
<p>Only used when fit is a cv.glmnet object. Should glmnet use lambda.min or lambda.1se? Default is lambda.1se.
Note that for TANDEM objects, the lambda_upstream and lambda_downstream parameters should be specified during the tandem() call, as
they are used while fitting the model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector that indicates the relative contribution per data type. These numbers sum up to one.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simple example
# unpack example data
x = example_data$x
y = example_data$y
upstream = example_data$upstream
data_types = example_data$data_types

# fit TANDEM model
fit = tandem(x, y, upstream, alpha=0.5)

# assess the relative contribution of upstream and downstream features
contr = relative.contributions(fit, x, data_types)
barplot(contr, ylab="Relative contribution", ylim=0:1)

## comparing TANDEM and classic model (glmnet)
# unpack example data
x = example_data$x
y = example_data$y
upstream = example_data$upstream
data_types = example_data$data_types

# fix the cv folds, to facilitate a comparison between models
set.seed(1)
n = nrow(x)
nfolds = 10
foldid = ceiling(sample(1:n)/n * nfolds)

# fit both a TANDEM and a classic model (glmnet)
fit = tandem(x, y, upstream, alpha=0.5)
library(glmnet)
fit2 = cv.glmnet(x, y, alpha=0.5, foldid=foldid)

# assess the relative contribution of upstream and downstream features
# using both methods
contr_tandem = relative.contributions(fit, x, data_types)
contr_glmnet = relative.contributions(fit2, x, data_types)
par(mfrow=c(1,2))
barplot(contr_tandem, ylab="Relative contribution", main="TANDEM", ylim=0:1)
barplot(contr_glmnet, ylab="Relative contribution", main="Classic approach", ylim=0:1)
par(mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='tandem'>Fits a TANDEM model by performing a two-stage regression</h2><span id='topic+tandem'></span>

<h3>Description</h3>

<p>Fits a TANDEM model by performing a two-stage regression. In the first stage, all upstream features (x[,upstream]) are regressed
on the output y. In the second stage, the downstream features (x[,!upstream]) are regressed on the residuals of the first stage.
In both stages Elastic Net regression (as implemented in cv.glmnet() from the glmnet package) is used to perform the regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tandem(
  x,
  y,
  upstream,
  family = "gaussian",
  nfolds = 10,
  foldid = NULL,
  lambda_upstream = "lambda.1se",
  lambda_downstream = "lambda.1se",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tandem_+3A_x">x</code></td>
<td>
<p>A feature matrix, where the rows correspond to samples and the columns to features.</p>
</td></tr>
<tr><td><code id="tandem_+3A_y">y</code></td>
<td>
<p>A vector containing the response.</p>
</td></tr>
<tr><td><code id="tandem_+3A_upstream">upstream</code></td>
<td>
<p>A boolean vector that indicates for each feature whether it's upstream (TRUE) or downstream (FALSE).</p>
</td></tr>
<tr><td><code id="tandem_+3A_family">family</code></td>
<td>
<p>The family parameter that's passed to cv.glmnet(). Currently, only family='gaussian' is supported.</p>
</td></tr>
<tr><td><code id="tandem_+3A_nfolds">nfolds</code></td>
<td>
<p>Number of cross-validation folds (default is 10) used to determine the optimal lambda in cv.glmnet().</p>
</td></tr>
<tr><td><code id="tandem_+3A_foldid">foldid</code></td>
<td>
<p>An optional vector indicating in which cross-validation fold each sample should be. Overrides nfolds when used.</p>
</td></tr>
<tr><td><code id="tandem_+3A_lambda_upstream">lambda_upstream</code></td>
<td>
<p>For the first stage (using the upstream features), should glmnet use lambda.min or lambda.1se? Default is lambda.1se.</p>
</td></tr>
<tr><td><code id="tandem_+3A_lambda_downstream">lambda_downstream</code></td>
<td>
<p>For the second stage (using the downstream features), should glmnet use lambda.min or lambda.1se? Default is lambda.1se.</p>
</td></tr>
<tr><td><code id="tandem_+3A_...">...</code></td>
<td>
<p>Other parameters that are passed to cv.glmnet().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tandem-object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># unpack example data
x = example_data$x
y = example_data$y
upstream = example_data$upstream

# fit a tandem model, determine the coefficients and create a prediction
fit = tandem(x, y, upstream, alpha=0.5)
beta = coef(fit)
y_hat = predict(fit, newx=x)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
