<!DOCTYPE html><html><head><title>Help for package tatest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {tatest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#tatest-package'>
<p>Two-Group <code class="reqn">T_\alpha</code>-Test</p></a></li>
<li><a href='#Ecadherin'>
<p>A microarray data</p></a></li>
<li><a href='#momega'>
<p>Null rho for testing multiple null hypotheses</p></a></li>
<li><a href='#omega'>
<p>Omega calcularion</p></a></li>
<li><a href='#rhov'>
<p>Calculation of rho (<code class="reqn">\rho</code>) value</p></a></li>
<li><a href='#ta.test'>
<p><code class="reqn">t_\alpha</code>-test for testing difference between two conditions with small samples</p></a></li>
<li><a href='#ttest'>
<p>A modified t-test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Two-Group Ta-Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-07-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuan-De Tan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuan-De Tan &lt;tanyuande@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The ta-test is a modified two-sample or two-group t-test of Gosset (1908). In small samples with less than 15 replicates,the ta-test significantly reduces type I error rate but has almost the same power with the t-test and hence can greatly enhance reliability or reproducibility of discoveries in biology and medicine. The ta-test can test single null hypothesis or multiple null hypotheses without needing to correct p-values.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), stats, base,utils</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-07-10 19:34:46 UTC; yuandet</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-18 11:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='tatest-package'>
Two-Group <code class="reqn">T_\alpha</code>-Test
</h2><span id='topic+tatest-package'></span>

<h3>Description</h3>

<p>The <code class="reqn">t_\alpha</code>-test is a modified two-sample or two-group t-test of Gosset(1908).
</p>


<h3>Details</h3>

<p>The package tatest consists of three t-tests: ttest, tatest and mtatest. ttest, rhov,and omega construct tatest, ttest and rhov and momega construct mtatest. <code class="reqn">t_\alpha</code>-test is used to test for a single null hypothesis  and <code class="reqn">mt_\alpha</code>-test is used to identify differential expressed genes in microarray data and RPPA data or proteomic data. 
</p>


<h3>Author(s)</h3>

<p>Yuande Tan
<br />
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
Yuande Tan
<br />
Maintainer: <a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+t.test">t.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-c(112,122,108,127)
Y&lt;-c(302, 314,322,328)
dat&lt;-cbind(X,Y)
ta.test(X=dat,na=4,nb=4, eqv=TRUE, paired=FALSE, 
alpha=0.05, alternative = "two.sided",LOG="NULL")
t.test(x=X,y=Y)
ttest(X,Y,alternative = "two.sided", LG = "NULL")
</code></pre>

<hr>
<h2 id='Ecadherin'>
A microarray data
</h2><span id='topic+Ecadherin'></span>

<h3>Description</h3>

<p>Analysis of the effects of loss of E-cadherin and cell adhesion on human mammary epithelial cells
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Ecadherin")</code></pre>


<h3>Format</h3>

<p>A data frame with 5000 observations on the following 7 variables.
</p>

<dl>
<dt><code>geneid</code></dt><dd><p>a factor with levels </p>
</dd>
<dt><code>shCntrl_rep1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>shCntrl_rep2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>shCntrl_rep3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>shEcad_rep1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>shEcad_rep2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>shEcad_rep3</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Loss of the epithelial adhesion molecule E-cadherin is thought to enable metastasis by disrupting intercellular contacts-an early step in metastatic dissemination. To further investigate the molecular basis of this notion, two methods were used to inhibit E-cadherin function that distinguish between E-cadherin's cell-cell adhesion and intracellular signaling functions. While the disruption of cell-cell contacts alone does not enable metastasis, the loss of E-cadherin protein does, through induction of an epithelial-to-mesenchymal transition, invasiveness and anoikis-resistance. In addition, gene expression analysis shows that E-cadherin loss results in the induction of multiple transcription factors, at least one of which, Twist, is necessary for E-cadherin loss-induced metastasis. These findings indicate that E-cadherin loss in tumors contributes to metastatic dissemination by inducing wide-ranging transcriptional and functional changes.
This dataset consists 5000 of 22277  gene libraries, two conditions shCntrl and shEcad, each having 3 replicates.  With access number GSE9691, you can obtain a whole data of 22277 gene libraries in GEO.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Ecadherin)
</code></pre>

<hr>
<h2 id='momega'>
Null rho for testing multiple null hypotheses
</h2><span id='topic+momega'></span>

<h3>Description</h3>

<p>Momega is a function that is used to estimate omega for testing multiple null hypotheses. 
Omege is a null <code class="reqn">\rho</code> that is used as a threshold for <code class="reqn">\rho</code> from real data. Since this omega is majorly applied 
to micoarray, we named this function  as momega.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>momega(dat, na, nb, distr = "norm")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="momega_+3A_dat">dat</code></td>
<td>

<p>a matrix dataset containing string columns and two-condition numeric columns
</p>
</td></tr>
<tr><td><code id="momega_+3A_na">na</code></td>
<td>

<p>numeric value, size of sample 1 or number of replicates in condition A.
</p>
</td></tr>
<tr><td><code id="momega_+3A_nb">nb</code></td>
<td>

<p>numeric value, size of sample 1 or number of replicates in condition A.
</p>
</td></tr>
<tr><td><code id="momega_+3A_distr">distr</code></td>
<td>

<p>string, data distribution.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is to use  null data to calculate omega value with rho  = 1. 
</p>


<h3>Value</h3>

<p>return a numeric value
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and Joel R. Neilson (2015)  A Powerful Statistical Approach for Large-scale Differential Transcription Analysis. <em>Plos One</em>. 2015 DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+omega">omega</a></code>,<code><a href="#topic+rhov">rhov</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Ecadherin)
Ecad&lt;-as.data.frame(Ecadherin)
Ecad1&lt;-Ecad[,-1]
w&lt;-momega(dat=Ecad1[1:2000,], na=3, nb=3, distr = "norm")
</code></pre>

<hr>
<h2 id='omega'>
Omega calcularion
</h2><span id='topic+omega'></span>

<h3>Description</h3>

<p>Omega (<code class="reqn">\omega</code>) function is a function that is used to estimate omega using simulate  null data from negative bionomial distribution. Omege is a null rho that is used as a threshold for real rho. Simulation is dependent on the original data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>omega(XA, XB, na, nb, m, alpha = 0.05, distr = "norm")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="omega_+3A_xa">XA</code></td>
<td>

<p>a numeric vector values of condition A
</p>
</td></tr>
<tr><td><code id="omega_+3A_xb">XB</code></td>
<td>

<p>a numeric vector values of condition b
</p>
</td></tr>
<tr><td><code id="omega_+3A_na">na</code></td>
<td>

<p>a numeric value, sample size in A.
</p>
</td></tr>
<tr><td><code id="omega_+3A_nb">nb</code></td>
<td>

<p>a numeric value, sample size in B
</p>
</td></tr>
<tr><td><code id="omega_+3A_m">m</code></td>
<td>

<p>simulation number specified.
</p>
</td></tr>
<tr><td><code id="omega_+3A_alpha">alpha</code></td>
<td>

<p>a numeric value, statistical cutoff. Default is 0.05.
</p>
</td></tr>
<tr><td><code id="omega_+3A_distr">distr</code></td>
<td>

<p>data distribution specified for data. For current version, we consider three distributions: normal, negative binomial (NB) and uniform. Default distribution is &quot;norm&quot;. If user believe that the data follow negative binomial distribution, then set distr=&quot;NB&quot; or &quot;negative binomial&quot; or if the data follow uniform distribution, then set distr=&quot;unif&quot;or &quot;uniform&quot;. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is to use  simulated null data to calculate omega value with rho  = 1. 
</p>


<h3>Value</h3>

<p>return a numeric value
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and Joel R. Neilson(2015)  A Powerful Statistical Approach for Large-scale Differential Transcription Analysis. <em>Plos One</em>. 2015 DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+momega">momega</a></code>, <code><a href="#topic+rhov">rhov</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-c(112,122,108,127)
Y&lt;-c(302, 314,322,328)
omega(XA=X, XB=Y, na=4, nb=4, m=2000, alpha = 0.05)
#[1] 0.9055152
 omega(XA=X, XB=Y, na=4, nb=4, m=2000, alpha = 0.05,distr="NB")
#[1] 0.8995424
omega(XA=X, XB=Y, na=4, nb=4, m=2000, alpha = 0.05,distr="uniform")
#[1] 0.97194
</code></pre>

<hr>
<h2 id='rhov'>
Calculation of rho (<code class="reqn">\rho</code>) value
</h2><span id='topic+rhov'></span>

<h3>Description</h3>

<p>The rho is used to define a gap and variation between two datasets, similar to fc(fold change).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rhov(XA, XB)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rhov_+3A_xa">XA</code></td>
<td>

<p>a numeric vector in condition A.
</p>
</td></tr>
<tr><td><code id="rhov_+3A_xb">XB</code></td>
<td>

<p>a numeric vector in condition B.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The rho is defined as square root of products of </p>
<p style="text-align: center;"><code class="reqn">\psi</code>
</p>
<p> and </p>
<p style="text-align: center;"><code class="reqn">\zeta</code>
</p>
<p>, that is </p>
<p style="text-align: center;"><code class="reqn">\rho=\sqrt{\psi\zeta}</code>
</p>
<p> where <code class="reqn">\psi</code> is used to define a overlap between two datasets and <code class="reqn">\zeta</code> is used to define effect variation. <code class="reqn">\rho &gt; 1</code> indicates that there is gap between two datasets and noise is smaller than treatment effect, otherwise, there is overall lap and big noise in two datasets.
</p>


<h3>Value</h3>

<p>return a numeric value
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<br />
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and Joel R. Neilson(2015) A Powerful Statistical Approach for Large-scale Differential Transcription Analysis. <em>Plos One</em>. 2015 DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+omega">omega</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
X&lt;-c(112,122, 108,127)
Y&lt;-c(302, 314, 322, 328)
rhov(XA=X,XB=Y)	
#[1] 3.062173
</code></pre>

<hr>
<h2 id='ta.test'>
<code class="reqn">t_\alpha</code>-test for testing difference between two conditions with small samples
</h2><span id='topic+ta.test'></span>

<h3>Description</h3>

<p><code class="reqn">t_\alpha</code>-test is a modified t-test of Gosset(1908). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ta.test(X, nci=NULL, na, nb, alpha=0.05, paired=FALSE, eqv=TRUE,
LOG=c("NULL","LOG2","LOG","LOG10"),
 alternative = c("two.sided", "less", "greater"), distr="norm")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ta.test_+3A_x">X</code></td>
<td>

<p>a numeric vector data or a numeric matrix (two columns) dataset with two conditions or groups A and B having sample size na and nb, respectively. X is also a datasheet of multiple variables (for example, genes ) with two groups A and B and information columns(see an example Ecadherin)
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_nci">nci</code></td>
<td>

<p>a numeric int value, column numbers for strings or information of data. If X is a numeric vector data or a numeric matrix (two columns) 
dataset, then nci=NULL, otherwise, nci is non-zeror int, meaning that at least one column is reserved as information of data
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_na">na</code></td>
<td>

<p>numeric constant, sample size in condition A
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_nb">nb</code></td>
<td>

<p>numeric constant, sample size in condition B
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_alpha">alpha</code></td>
<td>

<p>a numeric constant, statistical cutoff for significance level. If dataset X is multiple variables (rows) (variable number is over 1000), then alpha is taken default values of 0.05 and 0.01.
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_paired">paired</code></td>
<td>

<p>a logical indicating whether you want a paired t-test.
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_eqv">eqv</code></td>
<td>

<p>a logical variable indicating whether to treat the two variances as being equal. If TRUE then the pooled variance is used to estimate the variance otherwise the Welch (or Satterthwaite) approximation to the degrees of freedom is used.
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_log">LOG</code></td>
<td>

<p>logarithm: if data are not taken logarithm convertion, then LOG=NULL , if data are converted into logarithm with base 10, 2 or natural logarithm, then correspondingly, LOG=&quot;LOG10&quot;, &quot;LOG2&quot; or &quot;LOG&quot;.
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_alternative">alternative</code></td>
<td>

<p>a character string specifying the alternative hypothesis, must be one of &quot;two.sided&quot; (default), &quot;greater&quot; or &quot;less&quot;. User can specify just the initial letter.
</p>
</td></tr>
<tr><td><code id="ta.test_+3A_distr">distr</code></td>
<td>

<p>data distribution for estimating omega (<code class="reqn">\omega</code>). Default is normal distribution.  In current version, we just consider three types of distributions: &quot;negative distribution&quot;(or &quot;NB&quot;), &quot;norm&quot;(or &quot;normal&quot;), and &quot;unif&quot;(&quot;uniform&quot;).  User can choose one of them according to user's data distribution.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The formula interface is only applicable for the 2-sample tests. 
<br />
alternative = &quot;greater&quot; is the alternative that x has a larger mean than y.
<br />
If paired is TRUE then both x and y must be specified and they must be the same length. Missing values are silently removed (in pairs if paired is TRUE). 
<br />
If var.equal is TRUE then the pooled estimate of the variance is used. By default, if var.equal is FALSE, then the variance is estimated separately for both groups and the Welch modification to the degrees of freedom is used.
<br />
If the input data are effectively constant (compared to the larger of the two means) an error is generated.
<br />
When pooled variance is larger, ttest is equvalent to t.test; When pooled variance is small, t-tavalue of ttest is smaller than than that of t.test; When the pooled variance is zeror, t-value of t.test is not defined but tvalue of ttest is still defined.  
<br />
Omega is dependent on alpha value.
<br />
ta.test can be used to test for differentially expressed genes in microarray data, proteomic data and RPPA data but cannot be used to count data of RNA-seq reads. 
</p>


<h3>Value</h3>

<p>If X is single-hypothesis test data, then  a list similar to t.test containing the following components:
statistic
<br />
the value of the t-statistic.
<br />
parameter
<br />
the degrees of freedom for the t-statistic.
<br />
p.value
<br />
the p-alue for the test.
<br />
conf.int
<br />
a confidence interval for the mean appropriate to the specified alternative hypothesis.
<br />
estimate
<br />
the estimated mean or difference in means depending on whether it was a one-sample test or a two-sample test.
<br />
null.value
<br />
the specified hypothesized value of the mean or mean difference depending on whether it was a one-sample test or a two-sample test.
<br />
alternative
<br />
a character string describing the alternative hypothesis.
<br />
method
<br />
a character string indicating what type of t-test was performed.
<br />
data.name
<br />
a character string giving the name(s) of the data.
<br />
omega
a threshold for rho in tatest. 
<br />
If X is microarray, RPPA data, then a list is datasheet including t01,pv01, t05 and p05 columns
</p>


<h3>Note</h3>

<p>LOG choice must be the same with log transformation of data.
</p>


<h3>Author(s)</h3>

<p>Yuande Tan
<br />
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Gosset, W.S, &quot;Probable error of a correlation coefficient&quot;. Biometrika. 1908,6 (2/3): 302-310.  
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+t.test">t.test</a></code>,<code><a href="#topic+ttest">ttest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X&lt;-c(112,122,108,127)
Y&lt;-c(302, 314,322,328)
dat&lt;-cbind(X,Y)
ta.test(X=dat,nci=NULL, na=4,nb=4, eqv=TRUE, 
alpha=0.05, LOG="NULL", alternative = "two.sided")

data(Ecadherin)
res&lt;-ta.test(X=Ecadherin[1:2000,],nci=1, na=3,nb=3, eqv=TRUE, 
LOG="LOG2", alternative = "two.sided")
</code></pre>

<hr>
<h2 id='ttest'>
A modified t-test 
</h2><span id='topic+ttest'></span>

<h3>Description</h3>

<p>ttest is a modified t-test of Gosset(1098). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ttest(x, y = NULL, alternative = c("two.sided", "less", "greater"), 
mu = 0, paired = FALSE, var.equal = FALSE, conf.level = 0.95, 
LG = c("NULL", "LOG2", "LOG", "LOG10"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ttest_+3A_x">x</code></td>
<td>

<p>a (non-empty) numeric vector of data values.
</p>
</td></tr>
<tr><td><code id="ttest_+3A_y">y</code></td>
<td>

<p>an optional (non-empty) numeric vector of data values. For <code class="reqn">t_\alpha</code>-test, y is required
</p>
</td></tr>
<tr><td><code id="ttest_+3A_alternative">alternative</code></td>
<td>

<p>a character string specifying the alternative hypothesis, must be one of &quot;two.sided&quot; (default), &quot;greater&quot; or &quot;less&quot;. You can specify just the initial letter.
</p>
</td></tr>
<tr><td><code id="ttest_+3A_mu">mu</code></td>
<td>

<p>a number indicating the true value of the mean (or difference in means if you are performing a two sample test).
</p>
</td></tr>
<tr><td><code id="ttest_+3A_paired">paired</code></td>
<td>

<p>a logical indicating whether you want a paired t-test.
</p>
</td></tr>
<tr><td><code id="ttest_+3A_var.equal">var.equal</code></td>
<td>

<p>a logical variable indicating whether to treat the two variances as being equal. If TRUE then the pooled variance is used to estimate the variance otherwise the Welch (or Satterthwaite) approximation to the degrees of freedom is used.
</p>
</td></tr>
<tr><td><code id="ttest_+3A_conf.level">conf.level</code></td>
<td>

<p>confidence level of the interval.
</p>
</td></tr>
<tr><td><code id="ttest_+3A_lg">LG</code></td>
<td>

<p>logarithm: if x and y are not taken logarithm convertion, the LG=NULL , if x and y are converted into logarithm with base 10, 2 or natural logarithm, then correspondingly, LG=&quot;LOG10&quot;, 
</p>
</td></tr>
<tr><td><code id="ttest_+3A_...">...</code></td>
<td>

<p>further arguments to be passed to or from methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The formula interface is only applicable for the 2-sample tests. 
<br />
alternative = &quot;greater&quot; is the alternative that x has a larger mean than y.
<br />
If paired is TRUE then both x and y must be specified and they must be the same length. Missing values are silently removed (in pairs if paired is TRUE). 
<br />
If var.equal is TRUE then the pooled estimate of the variance is used. By default, if var.equal is FALSE, then the variance is estimated separately for both groups and the Welch modification to the degrees of freedom is used.
<br />
If the input data are effectively constant (compared to the larger of the two means) an error is generated.
<br />
When pooled variance is larger, ttest is equvalent to t.test; When pooled variance is small, t-tavalue of ttest is smaller than than that of t.test; When the pooled variance is zeror, t-value of t.test is not defined but tvalue of ttest is still defined.  
</p>


<h3>Value</h3>

<p>A list with class &quot;htest&quot; containing the following components:
<br />
statistic
<br />
the value of the t-statistic.
<br />
parameter
<br />
the degrees of freedom for the t-statistic.
<br />
p.value
<br />
the p-value for the test.
<br />
conf.int
<br />
a confidence interval for the mean appropriate to the specified alternative hypothesis.
<br />
estimate
<br />
the estimated mean or difference in means depending on whether it was a one-sample test or a two-sample test.
<br />
null.value
<br />
the specified hypothesized value of the mean or mean difference depending on whether it was a one-sample test or a two-sample test.
<br />
alternative
<br />
a character string describing the alternative hypothesis.
<br />
method
<br />
a character string indicating what type of t-test was performed.
<br />
data.name
a character string giving the name(s) of the data.
</p>


<h3>Author(s)</h3>

<p>Yuande Tan
<br />
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Gosset, W.S, &quot;Probable error of a correlation coefficient&quot;. Biometrika. 1908,6(2/3): 302-310.  
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+stats">stats</a>{t.test}</code>, <code><a href="#topic+ta.test">ta.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The following example is log10 transformed data	
YA&lt;-c(4.1455383,	4.7186418,4.6483316)
YB&lt;-c(6.9184465,	5.609667,6.5920944)
t.test(x=YA, y=YB)
ttest(YA,YB,alternative = "two.sided", LG = "NULL")
ttest(YA,YB,alternative = "two.sided", LG = "LOG2")
ttest(YA,YB,alternative = "two.sided", LG = "LOG10")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
