<!DOCTYPE html><html lang="en"><head><title>Help for package TaxaNorm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TaxaNorm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TaxaNorm-package'><p>TaxaNorm: Feature-Wise Normalization for Microbiome Sequencing Data</p></a></li>
<li><a href='#TaxaNorm_Model_Parameters-class'><p>TaxaNorm_Model_Parameters</p></a></li>
<li><a href='#TaxaNorm_Model_QC'><p>Function to QC TaxNorm algorithm</p></a></li>
<li><a href='#TaxaNorm_NMDS'><p>Function for TaxNorm NMDS</p></a></li>
<li><a href='#TaxaNorm_Normalization'><p>Function to run TaxaNorm algorithm</p></a></li>
<li><a href='#TaxaNorm_QC_Input'><p>Function for TaxNorm input data</p></a></li>
<li><a href='#TaxaNorm_Results-class'><p>TaxaNorm Results</p></a></li>
<li><a href='#TaxaNorm_Run_Diagnose'><p>Function to run TaxNorm algorithm</p></a></li>
<li><a href='#TaxaNorm-datasets'><p>TaxaNorm data objects</p></a></li>
<li><a href='#TaxaNormGenerics'><p>TaxaNorm package generics</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Feature-Wise Normalization for Microbiome Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.4</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dillon Lloyd &lt;dtlloyd@ncsu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A novel feature-wise normalization method based on a zero-inflated negative binomial model. This method assumes that the effects of sequencing depth vary for each taxon on their mean and also incorporates a rational link of zero probability and taxon dispersion as a function of sequencing depth. Ziyue Wang, Dillon Lloyd, Shanshan Zhao, Alison Motsinger-Reif (2023) &lt;<a href="https://doi.org/10.1101%2F2023.10.31.563648">doi:10.1101/2023.10.31.563648</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), microbiome,</td>
</tr>
<tr>
<td>Imports:</td>
<td>phyloseq, stats, S4Vectors, BiocGenerics, vegan, methods,
MASS, future, future.apply, matrixStats, pscl, parallelly,
ggplot2, utils</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/wangziyue57/TaxaNorm">https://github.com/wangziyue57/TaxaNorm</a></td>
</tr>
<tr>
<td>biocViews:</td>
<td>Sequencing, Microbiome, Metagenomics, Normalization,
Visualization</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/wangziyue57/TaxaNorm/issues">https://github.com/wangziyue57/TaxaNorm/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-11 15:19:47 UTC; Dillon</td>
</tr>
<tr>
<td>Author:</td>
<td>Ziyue Wang [aut],
  Dillon Lloyd [aut, cre, cph],
  Shanshan Zhao [aut, ctb],
  Alison Motsinger-Reif [aut, ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-12 18:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='TaxaNorm-package'>TaxaNorm: Feature-Wise Normalization for Microbiome Sequencing Data</h2><span id='topic+TaxaNorm'></span><span id='topic+TaxaNorm-package'></span>

<h3>Description</h3>

<p>A novel feature-wise normalization method based on a zero-inflated negative binomial model. This method assumes that the effects of sequencing depth vary for each taxon on their mean and also incorporates a rational link of zero probability and taxon dispersion as a function of sequencing depth. Ziyue Wang, Dillon Lloyd, Shanshan Zhao, Alison Motsinger-Reif (2023) &lt;arXiv:https://www.biorxiv.org/content/10.1101/2023.10.31.563648v1.abstract&gt;.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Dillon Lloyd <a href="mailto:dtlloyd@ncsu.edu">dtlloyd@ncsu.edu</a> [copyright holder]
</p>
<p>Authors:
</p>

<ul>
<li><p> Ziyue Wang <a href="mailto:wangziyue57@gmail.com">wangziyue57@gmail.com</a>
</p>
</li>
<li><p> Shanshan Zhao <a href="mailto:shanshan.zhao@nih.gov">shanshan.zhao@nih.gov</a> [contributor]
</p>
</li>
<li><p> Alison Motsinger-Reif <a href="mailto:alison.motsinger-reif@nih.gov">alison.motsinger-reif@nih.gov</a> [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/wangziyue57/TaxaNorm">https://github.com/wangziyue57/TaxaNorm</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/wangziyue57/TaxaNorm/issues">https://github.com/wangziyue57/TaxaNorm/issues</a>
</p>
</li></ul>


<hr>
<h2 id='TaxaNorm_Model_Parameters-class'>TaxaNorm_Model_Parameters</h2><span id='topic+TaxaNorm_Model_Parameters-class'></span><span id='topic+TaxaNorm_Model_Parameters'></span><span id='topic+coefficients+2CTaxaNorm_Model_Parameters-method'></span><span id='topic+TaxaNorm_Model_Parameters-coefficients'></span><span id='topic+coefficients+3C-+2CTaxaNorm_Model_Parameters-method'></span><span id='topic+mu+2CTaxaNorm_Model_Parameters-method'></span><span id='topic+TaxaNorm_Model_Parameters-mu'></span><span id='topic+mu+3C-+2CTaxaNorm_Model_Parameters-method'></span><span id='topic+theta+2CTaxaNorm_Model_Parameters-method'></span><span id='topic+TaxaNorm_Model_Parameters-theta'></span><span id='topic+theta+3C-+2CTaxaNorm_Model_Parameters-method'></span><span id='topic+pi+2CTaxaNorm_Model_Parameters-method'></span><span id='topic+TaxaNorm_Model_Parameters-pi'></span><span id='topic+pi+3C-+2CTaxaNorm_Model_Parameters-method'></span>

<h3>Description</h3>

<p>S4 class to store TaxaNorm Parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TaxaNorm_Model_Parameters(coefficients, mu, theta, pi)

## S4 method for signature 'TaxaNorm_Model_Parameters'
coefficients(x)

## S4 replacement method for signature 'TaxaNorm_Model_Parameters'
coefficients(x) &lt;- value

## S4 method for signature 'TaxaNorm_Model_Parameters'
mu(x)

## S4 replacement method for signature 'TaxaNorm_Model_Parameters'
mu(x) &lt;- value

## S4 method for signature 'TaxaNorm_Model_Parameters'
theta(x)

## S4 replacement method for signature 'TaxaNorm_Model_Parameters'
theta(x) &lt;- value

## S4 method for signature 'TaxaNorm_Model_Parameters'
pi(x)

## S4 replacement method for signature 'TaxaNorm_Model_Parameters'
pi(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TaxaNorm_Model_Parameters-class_+3A_coefficients">coefficients</code></td>
<td>
<p>Passed to <code>coefficients</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Model_Parameters-class_+3A_mu">mu</code></td>
<td>
<p>Passed to <code>mu</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Model_Parameters-class_+3A_theta">theta</code></td>
<td>
<p>Passed to <code>theta</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Model_Parameters-class_+3A_pi">pi</code></td>
<td>
<p>Passed to <code>pi</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Model_Parameters-class_+3A_x">x</code></td>
<td>
<p>TaxaNorm_Model_Parameters object</p>
</td></tr>
<tr><td><code id="TaxaNorm_Model_Parameters-class_+3A_value">value</code></td>
<td>
<p>Replacement value</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Parameters for TaxaNorm Method
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>coefficients(TaxaNorm_Model_Parameters)</code>: Return <code>coefficients</code> slot
</p>
</li>
<li> <p><code>mu(TaxaNorm_Model_Parameters)</code>: Return <code>mu</code> slot
</p>
</li>
<li> <p><code>theta(TaxaNorm_Model_Parameters)</code>: Return <code>theta</code> slot
</p>
</li>
<li> <p><code>pi(TaxaNorm_Model_Parameters)</code>: Return <code>pi</code> slot
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>coefficients</code></dt><dd><p><code>matrix</code> coefficients</p>
</dd>
<dt><code>mu</code></dt><dd><p><code>matrix</code> mu</p>
</dd>
<dt><code>theta</code></dt><dd><p><code>matrix</code> theta</p>
</dd>
<dt><code>pi</code></dt><dd><p><code>matrix</code> pi</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>coefficients &lt;-  matrix(c(1,2,3,4,5,6,7,8,9),nrow=3,ncol=3,byrow=TRUE)
mu &lt;- matrix(c(1,2,3,4,5,6,7,8,9),nrow=3,ncol=3,byrow=TRUE)
theta &lt;-  matrix(c(1,2,3,4,5,6,7,8,9),nrow=3,ncol=3,byrow=TRUE)
pi &lt;- matrix(c(1,2,3,4,5,6,7,8,9),nrow=3,ncol=3,byrow=TRUE)
TaxaNorm_Model_Parameters(coefficients = coefficients,mu = mu,theta = theta,pi = pi)
</code></pre>

<hr>
<h2 id='TaxaNorm_Model_QC'>Function to QC TaxNorm algorithm</h2><span id='topic+TaxaNorm_Model_QC'></span>

<h3>Description</h3>

<p>Function to QC TaxNorm algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TaxaNorm_Model_QC(TaxaNormResults)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TaxaNorm_Model_QC_+3A_taxanormresults">TaxaNormResults</code></td>
<td>
<p>Input data; Results from TaxaNorm normalization</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing qc taxnorm object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("TaxaNorm_Example_Output", package = "TaxaNorm")
TaxaNorm_Model_QC(TaxaNormResults = TaxaNorm_Example_Output)
</code></pre>

<hr>
<h2 id='TaxaNorm_NMDS'>Function for TaxNorm NMDS</h2><span id='topic+TaxaNorm_NMDS'></span>

<h3>Description</h3>

<p>Function for TaxNorm NMDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TaxaNorm_NMDS(TaxaNormResults, group_column)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TaxaNorm_NMDS_+3A_taxanormresults">TaxaNormResults</code></td>
<td>
<p>(Required) Input data; should be either a phyloseq object or a count matrix</p>
</td></tr>
<tr><td><code id="TaxaNorm_NMDS_+3A_group_column">group_column</code></td>
<td>
<p>column to cluster on</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NMDS Plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("TaxaNorm_Example_Output", package = "TaxaNorm")
TaxaNorm_NMDS(TaxaNorm_Example_Output, group_column = "body_site")

</code></pre>

<hr>
<h2 id='TaxaNorm_Normalization'>Function to run TaxaNorm algorithm</h2><span id='topic+TaxaNorm_Normalization'></span>

<h3>Description</h3>

<p>Function to run TaxaNorm algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TaxaNorm_Normalization(
  data,
  depth = NULL,
  group = NULL,
  meta.data = NULL,
  filter.cell.num = 10,
  filter.taxa.count = 0,
  random = FALSE,
  ncores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TaxaNorm_Normalization_+3A_data">data</code></td>
<td>
<p>(Required) Input data; should be either a phyloseq object or a count matrix</p>
</td></tr>
<tr><td><code id="TaxaNorm_Normalization_+3A_depth">depth</code></td>
<td>
<p>sequencing depth if pre-calculated. It should be a vector with the same length and order as the column of the count data</p>
</td></tr>
<tr><td><code id="TaxaNorm_Normalization_+3A_group">group</code></td>
<td>
<p>condition variables if samples are from multiple groups; should be correpsond to the column of the count data. default is NULL, where no grouping is considered</p>
</td></tr>
<tr><td><code id="TaxaNorm_Normalization_+3A_meta.data">meta.data</code></td>
<td>
<p>meta data for Taxa</p>
</td></tr>
<tr><td><code id="TaxaNorm_Normalization_+3A_filter.cell.num">filter.cell.num</code></td>
<td>
<p>taxa with &quot;filter.cell.num&quot; in more than the value provided will be filtered</p>
</td></tr>
<tr><td><code id="TaxaNorm_Normalization_+3A_filter.taxa.count">filter.taxa.count</code></td>
<td>
<p>&quot;filter.taxa.count&quot; samples will be removed before testing. default is keep taxa appear in at least 10 samples within each group</p>
</td></tr>
<tr><td><code id="TaxaNorm_Normalization_+3A_random">random</code></td>
<td>
<p>calculate randomized normal quantile residual</p>
</td></tr>
<tr><td><code id="TaxaNorm_Normalization_+3A_ncores">ncores</code></td>
<td>
<p>whether multiple cores is used for parallel computing; default is max(1, detectCores() - 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a TaxaNorm Object containing the normalized count values and accessory information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("TaxaNorm_Example_Input", package = "TaxaNorm")
Normalized_Data &lt;- TaxaNorm_Normalization(data= TaxaNorm_Example_Input,
                                         depth = NULL,
                                         group = sample_data(TaxaNorm_Example_Input)$body_site,
                                         meta.data = NULL,
                                         filter.cell.num = 10,
                                         filter.taxa.count = 0,
                                         random = FALSE,
                                         ncores = 1)
</code></pre>

<hr>
<h2 id='TaxaNorm_QC_Input'>Function for TaxNorm input data</h2><span id='topic+TaxaNorm_QC_Input'></span>

<h3>Description</h3>

<p>Function for TaxNorm input data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TaxaNorm_QC_Input(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TaxaNorm_QC_Input_+3A_data">data</code></td>
<td>
<p>(Required) Input data; should be either a phyloseq object or a count matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>QC PLots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("TaxaNorm_Example_Input", package = "TaxaNorm")
qc_data &lt;- TaxaNorm_QC_Input(TaxaNorm_Example_Input)

</code></pre>

<hr>
<h2 id='TaxaNorm_Results-class'>TaxaNorm Results</h2><span id='topic+TaxaNorm_Results-class'></span><span id='topic+TaxaNorm_Results'></span><span id='topic+input_data+2CTaxaNorm_Results-method'></span><span id='topic+TaxaNorm_Results-input_data'></span><span id='topic+input_data+3C-+2CTaxaNorm_Results-method'></span><span id='topic+rawdata+2CTaxaNorm_Results-method'></span><span id='topic+TaxaNorm_Results-rawdata'></span><span id='topic+rawdata+3C-+2CTaxaNorm_Results-method'></span><span id='topic+normdata+2CTaxaNorm_Results-method'></span><span id='topic+TaxaNorm_Results-normdata'></span><span id='topic+normdata+3C-+2CTaxaNorm_Results-method'></span><span id='topic+ecdf+2CTaxaNorm_Results-method'></span><span id='topic+TaxaNorm_Results-ecdf'></span><span id='topic+ecdf+3C-+2CTaxaNorm_Results-method'></span><span id='topic+model_pars+2CTaxaNorm_Results-method'></span><span id='topic+TaxaNorm_Results-model_pars'></span><span id='topic+model_pars+3C-+2CTaxaNorm_Results-method'></span><span id='topic+converge+2CTaxaNorm_Results-method'></span><span id='topic+TaxaNorm_Results-converge'></span><span id='topic+converge+3C-+2CTaxaNorm_Results-method'></span><span id='topic+llk+2CTaxaNorm_Results-method'></span><span id='topic+TaxaNorm_Results-llk'></span><span id='topic+llk+3C-+2CTaxaNorm_Results-method'></span><span id='topic+final_df+2CTaxaNorm_Results-method'></span><span id='topic+TaxaNorm_Results-final_df'></span><span id='topic+final_df+3C-+2CTaxaNorm_Results-method'></span>

<h3>Description</h3>

<p>S4 class to store TaxaNorm Results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TaxaNorm_Results(
  input_data,
  rawdata,
  normdata,
  ecdf,
  model_pars,
  converge,
  llk,
  final_df
)

## S4 method for signature 'TaxaNorm_Results'
input_data(x)

## S4 replacement method for signature 'TaxaNorm_Results'
input_data(x) &lt;- value

## S4 method for signature 'TaxaNorm_Results'
rawdata(x)

## S4 replacement method for signature 'TaxaNorm_Results'
rawdata(x) &lt;- value

## S4 method for signature 'TaxaNorm_Results'
normdata(x)

## S4 replacement method for signature 'TaxaNorm_Results'
normdata(x) &lt;- value

## S4 method for signature 'TaxaNorm_Results'
ecdf(x)

## S4 replacement method for signature 'TaxaNorm_Results'
ecdf(x) &lt;- value

## S4 method for signature 'TaxaNorm_Results'
model_pars(x)

## S4 replacement method for signature 'TaxaNorm_Results'
model_pars(x) &lt;- value

## S4 method for signature 'TaxaNorm_Results'
converge(x)

## S4 replacement method for signature 'TaxaNorm_Results'
converge(x) &lt;- value

## S4 method for signature 'TaxaNorm_Results'
llk(x)

## S4 replacement method for signature 'TaxaNorm_Results'
llk(x) &lt;- value

## S4 method for signature 'TaxaNorm_Results'
final_df(x)

## S4 replacement method for signature 'TaxaNorm_Results'
final_df(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TaxaNorm_Results-class_+3A_input_data">input_data</code></td>
<td>
<p>passed to <code>input_data</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_rawdata">rawdata</code></td>
<td>
<p>Passed to <code>rawdata</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_normdata">normdata</code></td>
<td>
<p>Passed to <code>normdata</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_ecdf">ecdf</code></td>
<td>
<p>Passed to <code>ecdf</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_model_pars">model_pars</code></td>
<td>
<p>Passed to <code>model_pars</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_converge">converge</code></td>
<td>
<p>Passed to <code>converge</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_llk">llk</code></td>
<td>
<p>Passed to <code>llk</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_final_df">final_df</code></td>
<td>
<p>Passed to <code>final_df</code> slot</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_x">x</code></td>
<td>
<p>TaxaNorm_Results object</p>
</td></tr>
<tr><td><code id="TaxaNorm_Results-class_+3A_value">value</code></td>
<td>
<p>Replacement value</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All results from the TaxaNorm method and what was used to get those results
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>input_data(TaxaNorm_Results)</code>: Return <code>input_data</code> slot
</p>
</li>
<li> <p><code>rawdata(TaxaNorm_Results)</code>: Return <code>rawdata</code> slot
</p>
</li>
<li> <p><code>normdata(TaxaNorm_Results)</code>: Return <code>normdata</code> slot
</p>
</li>
<li> <p><code>ecdf(TaxaNorm_Results)</code>: Return <code>ecdf</code> slot
</p>
</li>
<li> <p><code>model_pars(TaxaNorm_Results)</code>: Return <code>model_pars</code> slot
</p>
</li>
<li> <p><code>converge(TaxaNorm_Results)</code>: Return <code>converge</code> slot
</p>
</li>
<li> <p><code>llk(TaxaNorm_Results)</code>: Return <code>llk</code> slot
</p>
</li>
<li> <p><code>final_df(TaxaNorm_Results)</code>: Return <code>final_df</code> slot
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>input_data</code></dt><dd><p><code>ANY</code> phyloseq input data</p>
</dd>
<dt><code>rawdata</code></dt><dd><p><code>data.frame</code> Data frame of counts to use</p>
</dd>
<dt><code>normdata</code></dt><dd><p><code>data.frame</code> Normalized Data</p>
</dd>
<dt><code>ecdf</code></dt><dd><p><code>data.frame</code> ecdf</p>
</dd>
<dt><code>model_pars</code></dt><dd><p><a href="#topic+TaxaNorm_Model_Parameters">TaxaNorm_Model_Parameters</a> list of model parameters</p>
</dd>
<dt><code>converge</code></dt><dd><p><code style="white-space: pre;">&#8288;vector(&lt;logical&gt;)&#8288;</code> converge</p>
</dd>
<dt><code>llk</code></dt><dd><p><code>ANY</code> llk</p>
</dd>
<dt><code>final_df</code></dt><dd><p><code>ANY</code> final_df</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>coefficients &lt;-  matrix(c(1,2,3,4,5,6,7,8,9),nrow=3,ncol=3,byrow=TRUE)
mu &lt;- matrix(c(1,2,3,4,5,6,7,8,9),nrow=3,ncol=3,byrow=TRUE)
theta &lt;-  matrix(c(1,2,3,4,5,6,7,8,9),nrow=3,ncol=3,byrow=TRUE)
pi &lt;- matrix(c(1,2,3,4,5,6,7,8,9),nrow=3,ncol=3,byrow=TRUE)
model_pars &lt;- TaxaNorm_Model_Parameters(coefficients = coefficients,mu = mu,theta = theta,pi = pi)
data("TaxaNorm_Example_Input", package = "TaxaNorm")
rawdata &lt;- data.frame(Taxa1 = c(1,2,3),Taxa2 = c(3,4,5),Taxa3 = c(6,7,8))
normdata &lt;- data.frame(Taxa1 = c(-1.4,-1.09,-0.73),
Taxa2 = c( -0.36,0,0.36), Taxa3 = c(0.73,1.09,1.46))
ecdf &lt;- data.frame(0.05,0.23,0.89)
converge &lt;- c(TRUE,TRUE,FALSE)
llk &lt;- c(1,1.5,0.5)
final_df &lt;- data.frame(Taxa1 = c(1,2,3),Taxa2 = c(3,4,5),Taxa3 = c(6,7,8))
TaxaNorm_Results(input_data = TaxaNorm_Example_Input,
                                  rawdata = rawdata,
                                   normdata = normdata,
                                   ecdf = ecdf,
                                   model_pars = model_pars,
                                   converge = converge,
                                   llk = llk,
                                   final_df = final_df)
</code></pre>

<hr>
<h2 id='TaxaNorm_Run_Diagnose'>Function to run TaxNorm algorithm</h2><span id='topic+TaxaNorm_Run_Diagnose'></span>

<h3>Description</h3>

<p>Function to run TaxNorm algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TaxaNorm_Run_Diagnose(Normalized_Results, prev = TRUE, equiv = TRUE, group)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TaxaNorm_Run_Diagnose_+3A_normalized_results">Normalized_Results</code></td>
<td>
<p>(Required) Input results from from run_norm()</p>
</td></tr>
<tr><td><code id="TaxaNorm_Run_Diagnose_+3A_prev">prev</code></td>
<td>
<p>run prev test</p>
</td></tr>
<tr><td><code id="TaxaNorm_Run_Diagnose_+3A_equiv">equiv</code></td>
<td>
<p>run equiv test</p>
</td></tr>
<tr><td><code id="TaxaNorm_Run_Diagnose_+3A_group">group</code></td>
<td>
<p>group used for taxanorm normalization</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing the normalized count values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("TaxaNorm_Example_Input", package = "TaxaNorm")
data("TaxaNorm_Example_Output", package = "TaxaNorm")
Diagnose_Data &lt;- TaxaNorm_Run_Diagnose(Normalized_Results = TaxaNorm_Example_Output,
                                        prev = TRUE,
                                        equiv = TRUE,
                                        group = sample_data(TaxaNorm_Example_Input)$body_site)
</code></pre>

<hr>
<h2 id='TaxaNorm-datasets'>TaxaNorm data objects</h2><span id='topic+TaxaNorm-datasets'></span><span id='topic+TaxaNorm_Example_Input'></span><span id='topic+TaxaNorm_Example_Output'></span>

<h3>Description</h3>

<p>Objects included in the TaxaNorm package, loaded with
<a href="utils.html#topic+data">utils::data</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(TaxaNorm_Example_Input, package = "TaxaNorm")

data(TaxaNorm_Example_Output, package = "TaxaNorm")
</code></pre>


<h3>TaxaNorm_Example_Input</h3>

<p>Example data
#'
</p>


<h3>TaxaNorm_Example_Output</h3>

<p>Example output
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(TaxaNorm_Example_Input, package = "TaxaNorm")
          data(TaxaNorm_Example_Output, package = "TaxaNorm")

</code></pre>

<hr>
<h2 id='TaxaNormGenerics'>TaxaNorm package generics</h2><span id='topic+TaxaNormGenerics'></span><span id='topic+input_data'></span><span id='topic+input_data+3C-'></span><span id='topic+rawdata'></span><span id='topic+rawdata+3C-'></span><span id='topic+normdata'></span><span id='topic+normdata+3C-'></span><span id='topic+ecdf'></span><span id='topic+ecdf+3C-'></span><span id='topic+model_pars'></span><span id='topic+model_pars+3C-'></span><span id='topic+converge'></span><span id='topic+converge+3C-'></span><span id='topic+llk'></span><span id='topic+llk+3C-'></span><span id='topic+final_df'></span><span id='topic+final_df+3C-'></span><span id='topic+coefficients'></span><span id='topic+coefficients+3C-'></span><span id='topic+mu'></span><span id='topic+mu+3C-'></span><span id='topic+theta'></span><span id='topic+theta+3C-'></span><span id='topic+pi'></span><span id='topic+pi+3C-'></span>

<h3>Description</h3>

<p>TaxaNorm package generics; see class man pages for associated
methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_data(x, ...)

input_data(x, ...) &lt;- value

rawdata(x, ...)

rawdata(x, ...) &lt;- value

normdata(x, ...)

normdata(x, ...) &lt;- value

ecdf(x, ...)

ecdf(x, ...) &lt;- value

model_pars(x, ...)

model_pars(x, ...) &lt;- value

converge(x, ...)

converge(x, ...) &lt;- value

llk(x, ...)

llk(x, ...) &lt;- value

final_df(x, ...)

final_df(x, ...) &lt;- value

coefficients(x, ...)

coefficients(x, ...) &lt;- value

mu(x, ...)

mu(x, ...) &lt;- value

theta(x, ...)

theta(x, ...) &lt;- value

pi(x, ...)

pi(x, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TaxaNormGenerics_+3A_x">x</code></td>
<td>
<p>TaxaNorm S4 object</p>
</td></tr>
<tr><td><code id="TaxaNormGenerics_+3A_...">...</code></td>
<td>
<p>Included for extendability; not currently used</p>
</td></tr>
<tr><td><code id="TaxaNormGenerics_+3A_value">value</code></td>
<td>
<p>Replacement value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TaxaNorm generic functions return the specified slot of the TaxaNorm object given to the function
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
