<!DOCTYPE html><html lang="en-US"><head><title>Help for package taxize</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {taxize}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#taxize-package'><p>Taxonomic Information from Around the Web</p></a></li>
<li><a href='#apg'><p>Get APG names</p></a></li>
<li><a href='#apg_families'><p>MOBOT family names</p></a></li>
<li><a href='#apg_lookup'><p>Lookup in the APGIII taxonomy and replace family names</p></a></li>
<li><a href='#apg_orders'><p>MOBOT order names</p></a></li>
<li><a href='#bold_children'><p>Barcode of Life taxonomic children</p></a></li>
<li><a href='#bold_downstream'><p>Retrieve all taxa names downstream in hierarchy for BOLD</p></a></li>
<li><a href='#bold_search'><p>Search Barcode of Life for taxonomic IDs</p></a></li>
<li><a href='#bold_tax_id'><p>Search BOLD for taxonomy data by BOLD ID.</p></a></li>
<li><a href='#bold_tax_id2'><p>Search BOLD for taxonomy data by BOLD ID.</p></a></li>
<li><a href='#children'><p>Retrieve immediate children taxa for a given taxon name or ID.</p></a></li>
<li><a href='#class2tree'><p>Convert a list of classifications to a tree.</p></a></li>
<li><a href='#classification'><p>Retrieve the taxonomic hierarchy for a given taxon ID.</p></a></li>
<li><a href='#col-defunct'><p>COL defunct functions</p></a></li>
<li><a href='#comm2sci'><p>Get scientific names from common names.</p></a></li>
<li><a href='#downstream'><p>Retrieve the downstream taxa for a given taxon name or ID.</p></a></li>
<li><a href='#eol_dataobjects'><p>Given the identifier for a data object, return all metadata about the object</p></a></li>
<li><a href='#eol_hierarchy'><p>Retrieve the taxonomic hierarchy from given EOL taxonID.</p></a></li>
<li><a href='#eol_invasive'><p>Search for presence of taxonomic names in EOL invasive species databases.</p></a></li>
<li><a href='#eol_pages'><p>Search for pages in EOL database using a taxonconceptID.</p></a></li>
<li><a href='#eol_search'><p>Search for terms in EOL database.</p></a></li>
<li><a href='#eubon'><p>DEFUNCT</p></a></li>
<li><a href='#eubon_capabilities'><p>EUBON capabilities</p></a></li>
<li><a href='#eubon_children'><p>EUBON children</p></a></li>
<li><a href='#eubon_hierarchy'><p>EUBON hierarchy</p></a></li>
<li><a href='#eubon_search'><p>EUBON taxonomy search</p></a></li>
<li><a href='#fungorum'><p>Index Fungorum</p></a></li>
<li><a href='#gbif_downstream'><p>Retrieve all taxonomic names downstream in hierarchy for GBIF</p></a></li>
<li><a href='#gbif_name_usage'><p>Lookup details for specific names in all taxonomies in GBIF.</p></a></li>
<li><a href='#gbif_parse'><p>Parse taxon names using the GBIF name parser.</p></a></li>
<li><a href='#genbank2uid'><p>Get NCBI taxonomy UID from GenBankID</p></a></li>
<li><a href='#get_boldid'><p>Get the BOLD (Barcode of Life) code for a search term.</p></a></li>
<li><a href='#get_eolid'><p>Get the EOL ID from Encyclopedia of Life from taxonomic names.</p></a></li>
<li><a href='#get_gbifid'><p>Get the GBIF backbone taxon ID from taxonomic names.</p></a></li>
<li><a href='#get_genes'><p>Retrieve gene sequences from NCBI by accession number.</p></a></li>
<li><a href='#get_genes_avail'><p>Retrieve gene sequences from NCBI by accession number.</p></a></li>
<li><a href='#get_id_details'><p>Details on <code style="white-space: pre;">&#8288;get_*()&#8288;</code> functions</p></a></li>
<li><a href='#get_ids'><p>Retrieve taxonomic identifiers for a given taxon name.</p></a></li>
<li><a href='#get_iucn'><p>Get a IUCN Redlist taxon</p></a></li>
<li><a href='#get_iucn_data'><p>Get a IUCN Redlist taxon data</p></a></li>
<li><a href='#get_natservid'><p>Get NatureServe taxonomic ID for a taxon name</p></a></li>
<li><a href='#get_nbnid'><p>Get the UK National Biodiversity Network ID from taxonomic names.</p></a></li>
<li><a href='#get_pow'><p>Get Kew's Plants of the World code for a taxon</p></a></li>
<li><a href='#get_seqs'><p>Retrieve gene sequences from NCBI by accession number.</p></a></li>
<li><a href='#get_tolid'><p>Get the OTT id for a search term</p></a></li>
<li><a href='#get_tpsid'><p>Get the NameID codes from Tropicos for taxonomic names.</p></a></li>
<li><a href='#get_tsn'><p>Get the TSN code for a search term.</p></a></li>
<li><a href='#get_ubioid'><p>Get the uBio id for a search term</p></a></li>
<li><a href='#get_uid'><p>Get the UID codes from NCBI for taxonomic names.</p></a></li>
<li><a href='#get_wiki'><p>Get the page name for a Wiki taxon</p></a></li>
<li><a href='#get_wormsid'><p>Get Worms ID for a taxon name</p></a></li>
<li><a href='#getkey'><p>Function to get API key.</p></a></li>
<li><a href='#gisd_isinvasive'><p>Check invasive species status for a set of species from GISD database</p></a></li>
<li><a href='#gna_data_sources'><p>Get metadata about GNA data sources</p></a></li>
<li><a href='#gna_parse'><p>Parse scientific names using Global Names Parser</p></a></li>
<li><a href='#gna_search'><p>Search for taxonomic names using the Global Names Architecture</p></a></li>
<li><a href='#gna_verifier'><p>Verify a list of scientific names against biodiversity data-sources.</p></a></li>
<li><a href='#gni_details'><p>Search for taxonomic name details using the Global Names Index</p></a></li>
<li><a href='#gni_parse'><p>Parse scientific names using EOL's name parser.</p></a></li>
<li><a href='#gni_seach'><p>Search for taxonomic names using the Global Names Index</p></a></li>
<li><a href='#gnr_datasources'><p>Global Names Resolver Data Sources</p></a></li>
<li><a href='#gnr_resolve'><p>Resolve names using Global Names Resolver</p></a></li>
<li><a href='#id2name'><p>Taxonomic IDs to taxonomic names</p></a></li>
<li><a href='#ion'><p>ION - Index to Organism Names</p></a></li>
<li><a href='#iplant_resolve'><p>iPlant name resolution</p></a></li>
<li><a href='#ipni_search'><p>Search for names in the International Plant Names Index (IPNI).</p></a></li>
<li><a href='#itis_acceptname'><p>Retrieve accepted TSN and name</p></a></li>
<li><a href='#itis_downstream'><p>Retrieve all taxa names or TSNs downstream in hierarchy from given TSN.</p></a></li>
<li><a href='#itis_getrecord'><p>Get full ITIS record for one or more ITIS TSN's or lsid's.</p></a></li>
<li><a href='#itis_hierarchy'><p>ITIS hierarchy</p></a></li>
<li><a href='#itis_kingdomnames'><p>Get kingdom names</p></a></li>
<li><a href='#itis_lsid'><p>Get TSN from LSID</p></a></li>
<li><a href='#itis_name'><p>Get taxonomic names for a given taxonomic name query.</p></a></li>
<li><a href='#itis_native'><p>Get jurisdiction data, i.e., native or not native in a region.</p></a></li>
<li><a href='#itis_refs'><p>Get references related to a ITIS TSN.</p></a></li>
<li><a href='#itis_taxrank'><p>Retrieve taxonomic rank name from given TSN.</p></a></li>
<li><a href='#itis_terms'><p>Get ITIS terms, i.e., tsn's, authors, common names, and scientific names.</p></a></li>
<li><a href='#iucn_getname'><p>Get any matching IUCN species names</p></a></li>
<li><a href='#iucn_id'><p>Get an ID for a IUCN listed taxon</p></a></li>
<li><a href='#iucn_status'><p>Extractor functions for <code>iucn</code>-class.</p></a></li>
<li><a href='#iucn_summary'><p>Get a summary from the IUCN Red List</p></a></li>
<li><a href='#key_helpers'><p>Helpers to set up authentication for the different providers.</p></a></li>
<li><a href='#limited_print'><p>Print a subset of a character vector</p></a></li>
<li><a href='#lowest_common'><p>Retrieve the lowest common taxon and rank for a given taxon name or ID</p></a></li>
<li><a href='#names_list'><p>Get a random vector of species names.</p></a></li>
<li><a href='#nbn_classification'><p>Search UK National Biodiversity Network database for</p>
taxonomic classification</a></li>
<li><a href='#nbn_search'><p>Search UK National Biodiversity Network</p></a></li>
<li><a href='#nbn_synonyms'><p>Return all synonyms for a taxon name with a given id from NBN</p></a></li>
<li><a href='#ncbi_children'><p>Search NCBI for children of a taxon</p></a></li>
<li><a href='#ncbi_downstream'><p>Retrieve all taxa names downstream in hierarchy for NCBI</p></a></li>
<li><a href='#ncbi_get_taxon_summary'><p>NCBI taxon information from uids</p></a></li>
<li><a href='#ncbi_getbyid'><p>Retrieve gene sequences from NCBI by accession number.</p></a></li>
<li><a href='#ncbi_getbyname'><p>Retrieve gene sequences from NCBI by taxon name and gene names.</p></a></li>
<li><a href='#ncbi_search'><p>Search for gene sequences available for taxa from NCBI.</p></a></li>
<li><a href='#phylomatic_format'><p>Get family names to make Phylomatic input object, and output input string</p>
to Phylomatic for use in the function phylomatic_tree.</a></li>
<li><a href='#phylomatic_tree'><p>Query Phylomatic for a phylogenetic tree.</p></a></li>
<li><a href='#ping'><p>Ping an API used in taxize to see if it's working.</p></a></li>
<li><a href='#plantGenusNames'><p>Vector of plant genus names from ThePlantList</p></a></li>
<li><a href='#plantminer'><p>Search for taxonomy data from Plantminer.com</p></a></li>
<li><a href='#plantNames'><p>Vector of plant species (genus - specific epithet) names from ThePlantList</p></a></li>
<li><a href='#pow_lookup'><p>Lookup taxa in Kew's Plants of the World</p></a></li>
<li><a href='#pow_search'><p>Search Kew's Plants of the World</p></a></li>
<li><a href='#pow_synonyms'><p>Lookup synonyms in Kew's Plants of the World</p></a></li>
<li><a href='#progressor'><p>methods for preparing/printing info for prompts for <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions</p></a></li>
<li><a href='#rank_ref'><p>Lookup-table for IDs of taxonomic ranks</p></a></li>
<li><a href='#rank_ref_zoo'><p>Lookup-table for IDs of taxonomic ranks (WoRMS)</p></a></li>
<li><a href='#rankagg'><p>Aggregate data by given taxonomic rank</p></a></li>
<li><a href='#resolve'><p>Resolve names from different data sources</p></a></li>
<li><a href='#sci2comm'><p>Get common names from scientific names.</p></a></li>
<li><a href='#scrapenames'><p>Find taxon names using Global Names Recognition and Discovery</p></a></li>
<li><a href='#species_plantarum_binomials'><p>Species names from Species Plantarum</p></a></li>
<li><a href='#status_codes'><p>Get HTTP status codes</p></a></li>
<li><a href='#synonyms'><p>Retrieve synonyms from various sources given input taxonomic</p>
names or identifiers</a></li>
<li><a href='#tax_agg'><p>Aggregate species data to given taxonomic rank</p></a></li>
<li><a href='#tax_name'><p>Get taxonomic names for a given rank</p></a></li>
<li><a href='#tax_rank'><p>Get rank for a given taxonomic name.</p></a></li>
<li><a href='#taxize_capwords'><p>Capitalize the first letter of a character string.</p></a></li>
<li><a href='#taxize_cite'><p>Get citations and licenses for data sources used in taxize</p></a></li>
<li><a href='#taxize_ldfast'><p>Replacement function for ldply that should be faster in all cases.</p></a></li>
<li><a href='#taxize_options'><p>taxize options</p></a></li>
<li><a href='#taxize-authentication'><p>taxize authentication</p></a></li>
<li><a href='#taxize-defunct'><p>Defunct functions in taxize</p></a></li>
<li><a href='#taxize-params'><p>taxize parameters</p></a></li>
<li><a href='#taxon_state'><p>Keep track of queries in <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions</p></a></li>
<li><a href='#taxon-state'><p>Last taxon state object from a <code style="white-space: pre;">&#8288;get_*&#8288;</code> function call</p></a></li>
<li><a href='#theplantlist'><p>Lookup-table for family, genus, and species names for ThePlantList</p></a></li>
<li><a href='#tnrs'><p>Phylotastic Taxonomic Name Resolution Service.</p></a></li>
<li><a href='#tnrs_sources'><p>TNRS sources</p></a></li>
<li><a href='#tol_resolve'><p>Resolve names using Open Tree of Life (OTL) resolver</p></a></li>
<li><a href='#tp_acceptednames'><p>Return all accepted names for a taxon name with a given id.</p></a></li>
<li><a href='#tp_accnames'><p>Return all accepted names for a taxon name with a given id.</p></a></li>
<li><a href='#tp_classification'><p>Return all synonyms for a taxon name with a given id.</p></a></li>
<li><a href='#tp_dist'><p>Return all distribution records for for a taxon name with a given id.</p></a></li>
<li><a href='#tp_namedistributions'><p>Return all distribution records for for a taxon name with a given id.</p></a></li>
<li><a href='#tp_namereferences'><p>Return all reference records for for a taxon name with a given id.</p></a></li>
<li><a href='#tp_refs'><p>Return all reference records for for a taxon name with a given id.</p></a></li>
<li><a href='#tp_search'><p>Search Tropicos by scientific name, common name, or Tropicos ID.</p></a></li>
<li><a href='#tp_summary'><p>Return summary data a taxon name with a given id.</p></a></li>
<li><a href='#tp_synonyms'><p>Return all synonyms for a taxon name with a given id.</p></a></li>
<li><a href='#tpl_families'><p>Get The Plant List families.</p></a></li>
<li><a href='#tpl_get'><p>Get The Plant List csv files.</p></a></li>
<li><a href='#tpl_search'><p>A light wrapper around the taxonstand fxn to call Theplantlist.org database.</p></a></li>
<li><a href='#ubio_classification'><p>uBio classification</p></a></li>
<li><a href='#ubio_classification_search'><p>This function will return ClassificationBankIDs (hierarchiesIDs) that refer to the</p>
given NamebankID</a></li>
<li><a href='#ubio_id'><p>Search uBio by namebank ID.</p></a></li>
<li><a href='#ubio_ping'><p>uBio ping</p></a></li>
<li><a href='#ubio_search'><p>This function will return NameBankIDs that match given search terms</p></a></li>
<li><a href='#ubio_synonyms'><p>Search uBio for taxonomic synonyms by hierarchiesID.</p></a></li>
<li><a href='#upstream'><p>Retrieve the upstream taxa for a given taxon name or ID.</p></a></li>
<li><a href='#use_eol-defunct'><p>use eol</p></a></li>
<li><a href='#vascan_search'><p>Search the CANADENSYS Vascan API.</p></a></li>
<li><a href='#worms_downstream'><p>Retrieve all taxa names downstream in hierarchy for WORMS</p></a></li>
<li><a href='#worrms_ranks'><p>WORMS ranks</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Taxonomic Information from Around the Web</td>
</tr>
<tr>
<td>Description:</td>
<td>Interacts with a suite of web application programming interfaces
    (API) for taxonomic tasks, such as getting database specific taxonomic
    identifiers, verifying species names, getting taxonomic hierarchies, fetching
    downstream and upstream taxonomic names, getting taxonomic synonyms, converting
    scientific to common names and vice versa, and more. Some of the services
    supported include 'NCBI E-utilities' (<a href="https://www.ncbi.nlm.nih.gov/books/NBK25501/">https://www.ncbi.nlm.nih.gov/books/NBK25501/</a>),
    'Encyclopedia of Life' (<a href="https://eol.org/docs/what-is-eol/data-services">https://eol.org/docs/what-is-eol/data-services</a>),
    'Global Biodiversity Information Facility' (<a href="https://techdocs.gbif.org/en/openapi/">https://techdocs.gbif.org/en/openapi/</a>),
    and many more. Links to the API documentation for other supported services are
    available in the documentation for their respective functions in this package.</td>
</tr>
<tr>
<td>Version:</td>
<td>0.10.0</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/taxize/">https://docs.ropensci.org/taxize/</a> (website),
<a href="https://github.com/ropensci/taxize">https://github.com/ropensci/taxize</a> (devel), <a href="https://taxize.dev">https://taxize.dev</a>
(user manual)</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/taxize/issues">https://github.com/ropensci/taxize/issues</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.2.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, methods, stats, utils, crul (&ge; 0.7.0), xml2 (&ge;
1.2.0), jsonlite, ape, zoo, data.table, tibble (&ge; 1.2),
rredlist, rotl (&ge; 3.0.0), ritis (&ge; 0.7.6), worrms (&ge; 0.4.0),
natserv (&ge; 1.0.0), wikitaxa (&ge; 0.3.0), R6, crayon, cli,
phangorn, lifecycle, curl, stringi</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, vegan, vcr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>X-schema.org-applicationCategory:</td>
<td>Taxonomy</td>
</tr>
<tr>
<td>X-schema.org-keywords:</td>
<td>taxonomy, biology, nomenclature, JSON, API, web,
api-client, identifiers, species, names</td>
</tr>
<tr>
<td>X-schema.org-isPartOf:</td>
<td>https://ropensci.org</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-06 19:53:50 UTC; fosterz</td>
</tr>
<tr>
<td>Author:</td>
<td>Scott Chamberlain [aut],
  Eduard Szoecs [aut],
  Zachary Foster [aut, cre],
  Zebulun Arendsee [aut],
  Carl Boettiger [ctb],
  Karthik Ram [ctb],
  Ignasi Bartomeus [ctb],
  John Baumgartner [ctb],
  James O'Donnell [ctb],
  Jari Oksanen [ctb],
  Bastian Greshake Tzovaras [ctb],
  Philippe Marchand [ctb],
  Vinh Tran [ctb],
  Maëlle Salmon [ctb],
  Gaopeng Li [ctb],
  Matthias Grenié [ctb],
  rOpenSci [fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zachary Foster &lt;zacharyfoster1989@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-07 09:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='taxize-package'>Taxonomic Information from Around the Web</h2><span id='topic+taxize-package'></span><span id='topic+taxize'></span>

<h3>Description</h3>

<p>This package interacts with a suite of web 'APIs'
for taxonomic tasks, such as verifying species names, getting
taxonomic hierarchies, and verifying name spelling.
</p>


<h3>About</h3>

<p>Allows users to search over many websites for species names (scientific and
common) and download up- and downstream taxonomic hierarchical information -
and many other things.
</p>
<p>The functions in the package that hit a specific API have a prefix and suffix
separated by an underscore. They follow the format of <code>service_whatitdoes</code>.
For example, <code>gnr_resolve</code> uses the Global Names Resolver API to resolve
species names.
</p>
<p>General functions in the package that don't hit a specific API don't have
two words separated by an underscore, e.g., <code>classification</code>
</p>
<p>You need API keys for some data sources. See <a href="#topic+taxize-authentication">taxize-authentication</a>
for more information.
</p>


<h3>Currently supported APIs</h3>


<table>
<tr>
 <td style="text-align: left;">
API </td><td style="text-align: left;"> prefix </td><td style="text-align: center;"> SOAP? </td>
</tr>
<tr>
 <td style="text-align: left;">
Encyclopedia of Life (EOL)  </td><td style="text-align: left;"> eol </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Integrated Taxonomic Information Service (ITIS)  </td><td style="text-align: left;"> itis </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Global Names Resolver (from EOL/GBIF) </td><td style="text-align: left;"> gnr </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Global Names Index (from EOL/GBIF) </td><td style="text-align: left;"> gni </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
IUCN Red List </td><td style="text-align: left;"> iucn </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Tropicos (from Missouri Botanical Garden) </td><td style="text-align: left;"> tp </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Theplantlist.org </td><td style="text-align: left;"> tpl </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
National Center for Biotechnology Information </td><td style="text-align: left;"> ncbi </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
CANADENSYS Vascan name search API </td><td style="text-align: left;"> vascan </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
International Plant Names Index (IPNI) </td><td style="text-align: left;"> ipni </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
World Register of Marine Species (WoRMS) </td><td style="text-align: left;"> worms </td><td style="text-align: center;"> TRUE </td>
</tr>
<tr>
 <td style="text-align: left;">
Barcode of Life Data Systems (BOLD) </td><td style="text-align: left;"> bold </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Pan-European Species directories Infrastructure (PESI) </td><td style="text-align: left;"> pesi </td><td style="text-align: center;"> TRUE </td>
</tr>
<tr>
 <td style="text-align: left;">
Mycobank </td><td style="text-align: left;"> myco </td><td style="text-align: center;"> TRUE </td>
</tr>
<tr>
 <td style="text-align: left;">
National Biodiversity Network (UK)	</td><td style="text-align: left;"> nbn </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Index Fungorum </td><td style="text-align: left;"> fg </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
EU BON </td><td style="text-align: left;"> eubon </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Index of Names (ION) </td><td style="text-align: left;"> ion </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
Open Tree of Life (TOL) </td><td style="text-align: left;"> tol </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
World Register of Marine Species (WoRMS) </td><td style="text-align: left;"> worms </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
NatureServe </td><td style="text-align: left;"> natserv </td><td style="text-align: center;"> FALSE </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>If the source above has a <code>TRUE</code> in the <code style="white-space: pre;">&#8288;SOAP?&#8288;</code> column, it is not available
in this package. They are available from a different package called <strong>taxizesoap</strong>.
See the GitHub repo for how to install https://github.com/ropensci/taxizesoap
</p>


<h3>Catalogue of Life (COL)</h3>

<p>COL introduced rate limiting recently in 2019 - which has made the API
essentially unusable - CoL+ is coming soon and we'll incorporate it here
when it's stable. See https://github.com/ropensci/colpluz for the
R implementation for CoL+
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>
<p>Eduard Szoecs <a href="mailto:eduardszoecs@gmail.com">eduardszoecs@gmail.com</a>
</p>
<p>Zachary Foster <a href="mailto:zacharyfoster1989@gmail.com">zacharyfoster1989@gmail.com</a>
</p>
<p>Carl Boettiger <a href="mailto:cboettig@gmail.com">cboettig@gmail.com</a>
</p>
<p>Karthik Ram <a href="mailto:karthik@ropensci.org">karthik@ropensci.org</a>
</p>
<p>Ignasi Bartomeus <a href="mailto:nacho.bartomeus@gmail.com">nacho.bartomeus@gmail.com</a>
</p>
<p>John Baumgartner <a href="mailto:johnbb@student.unimelb.edu.au">johnbb@student.unimelb.edu.au</a>
</p>
<p>James O'Donnell <a href="mailto:jodonnellbio@gmail.com">jodonnellbio@gmail.com</a>
</p>

<hr>
<h2 id='apg'>Get APG names</h2><span id='topic+apg'></span><span id='topic+apgOrders'></span><span id='topic+apgFamilies'></span>

<h3>Description</h3>

<p>Generic names and their replacements from the Angiosperm Phylogeny
Group III system of flowering plant classification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apgOrders(...)

apgFamilies(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="apg_+3A_...">...</code></td>
<td>
<p>Curl args passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>References</h3>

<p>http://www.mobot.org/MOBOT/research/APweb/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
head(apgOrders())
head(apgFamilies())

## End(Not run)
</code></pre>

<hr>
<h2 id='apg_families'>MOBOT family names</h2><span id='topic+apg_families'></span>

<h3>Description</h3>

<p>Family names and their replacements from the Angiosperm Phylogeny
Website system of flowering plant classification.
</p>


<h3>Format</h3>

<p>A data frame with 1705 rows and 6 variables:
</p>

<ul>
<li> <p><code>family</code>: family name
</p>
</li>
<li> <p><code>synonym</code>: if <code>accepted=FALSE</code>, this is the accepted name;
if <code>accepted=TRUE</code>, this is <code>NA</code>, and the name in <code>family</code> is accepted
</p>
</li>
<li> <p><code>order</code>: order name for the family
</p>
</li>
<li> <p><code>accepted</code>: logical, if name in <code>family</code> column is accepted or not
</p>
</li>
<li> <p><code>original</code>: original data record from APG website, mapping
name in <code>family</code> column to a new name, if there is one
</p>
</li>
<li> <p><code>accepted_name</code>: accepted name. accepted names, combining those that
are accepted from <code>family</code> column, with the new name from <code>synonym</code>
if applicable
</p>
</li></ul>



<h3>Details</h3>

<p>This dataset is from Version 14, incorporated on 2020-06-03,
generated using <code><a href="#topic+apgFamilies">apgFamilies()</a></code>
</p>
<p>(update script in inst/ignore/apg_script.R)
</p>


<h3>Source</h3>

<p>http://www.mobot.org/MOBOT/research/APweb/
</p>

<hr>
<h2 id='apg_lookup'>Lookup in the APGIII taxonomy and replace family names</h2><span id='topic+apg_lookup'></span>

<h3>Description</h3>

<p>Lookup in the APGIII taxonomy and replace family names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apg_lookup(taxa, rank = "family")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="apg_lookup_+3A_taxa">taxa</code></td>
<td>
<p>(character) Taxonomic name to lookup a synonym for
in APGIII taxonomy.</p>
</td></tr>
<tr><td><code id="apg_lookup_+3A_rank">rank</code></td>
<td>
<p>(character) Taxonomic rank to lookup a synonym for. One of
family or order.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally in this function, we use the datasets <a href="#topic+apg_families">apg_families</a>
and <a href="#topic+apg_orders">apg_orders</a> - see their descriptions for the data in them.
The functions <code><a href="#topic+apgOrders">apgOrders()</a></code> <code><a href="#topic+apgFamilies">apgFamilies()</a></code> are for scraping
current content from the http://www.mobot.org/MOBOT/research/APweb/
website
</p>
<p>The datasets used in this function are from the most recent version of
APGIII, Version 14 (http://www.mobot.org/MOBOT/research/APweb/)
</p>


<h3>Value</h3>

<p>A APGIII family or order name, the original name if the name
is the same as APG has, or NA if no match found
</p>


<h3>Examples</h3>

<pre><code class='language-R'># New name found
apg_lookup(taxa = "Hyacinthaceae", rank = "family")
# Name is the same
apg_lookup(taxa = "Poaceae", rank = "family")
apg_lookup(taxa = "Asteraceae", rank = "family")
# Name not found
apg_lookup(taxa = "Foobar", rank = "family")

# New name found
apg_lookup(taxa = "Acerales", rank = "order")
# Name is the same
apg_lookup(taxa = "Acorales", rank = "order")
# Name not found
apg_lookup(taxa = "Foobar", rank = "order")
</code></pre>

<hr>
<h2 id='apg_orders'>MOBOT order names</h2><span id='topic+apg_orders'></span>

<h3>Description</h3>

<p>Order names and their replacements from the Angiosperm Phylogeny
Website system of flowering plant classification.
</p>


<h3>Format</h3>

<p>A data frame with 576 rows and 5 variables:
</p>

<ul>
<li> <p><code>order</code>: order name
</p>
</li>
<li> <p><code>synonym</code>: if <code>accepted=FALSE</code>, this is the accepted name;
if <code>accepted=TRUE</code>, this is <code>NA</code>, and the name in <code>order</code> is accepted
</p>
</li>
<li> <p><code>accepted</code>: logical, if name in <code>order</code> column is accepted or not
</p>
</li>
<li> <p><code>original</code>: original data record from APG website, mapping
name in <code>order</code> column to a new name, if there is one
</p>
</li>
<li> <p><code>accepted_name</code>: accepted name. accepted names, combining those that
are accepted from <code>order</code> column, with the new name from <code>synonym</code>
if applicable
</p>
</li></ul>



<h3>Details</h3>

<p>This dataset is from Version 14, incorporated on 2020-06-03,
generated using <code><a href="#topic+apgOrders">apgOrders()</a></code>
</p>
<p>(update script in inst/ignore/apg_script.R)
</p>


<h3>Source</h3>

<p>http://www.mobot.org/MOBOT/research/APweb/
</p>

<hr>
<h2 id='bold_children'>Barcode of Life taxonomic children</h2><span id='topic+bold_children'></span>

<h3>Description</h3>

<p>BEWARE: this function scrapes data from the BOLD website, so may
be unstable. That is, one day it may work, and the next it may fail.
Open an issue if you encounter an error:
https://github.com/ropensci/taxize/issues
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_children(id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bold_children_+3A_id">id</code></td>
<td>
<p>(integer) A BOLD taxonomic identifier. length=1. required</p>
</td></tr>
<tr><td><code id="bold_children_+3A_...">...</code></td>
<td>
<p>named curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a>
debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of data.frame's
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Osmia (genus): 253 children
bold_children(id = 4940)
# Momotus (genus): 3 children
bold_children(id = 88899)
# Momotus aequatorialis (species): no children
bold_children(id = 115130)
# Osmia sp1 (species): no children
bold_children(id = 293378)
# Arthropoda (phylum): 27 children
bold_children(id = 82)
# Psocodea (order): 51 children
bold_children(id = 737139)
# Megachilinae (subfamily): 2 groups (tribes: 3, genera: 60)
bold_children(id = 4962)
# Stelis (species): 78 taxa
bold_children(id = 4952)

## End(Not run)
</code></pre>

<hr>
<h2 id='bold_downstream'>Retrieve all taxa names downstream in hierarchy for BOLD</h2><span id='topic+bold_downstream'></span>

<h3>Description</h3>

<p>Retrieve all taxa names downstream in hierarchy for BOLD
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_downstream(id, downto, intermediate = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bold_downstream_+3A_id">id</code></td>
<td>
<p>(integer) One or more BOLD taxonomic identifiers</p>
</td></tr>
<tr><td><code id="bold_downstream_+3A_downto">downto</code></td>
<td>
<p>(character) The taxonomic level you want to go down to.
See examples below. The taxonomic level IS case sensitive, and you do have
to spell it correctly. See <code>data(rank_ref)</code> for spelling.</p>
</td></tr>
<tr><td><code id="bold_downstream_+3A_intermediate">intermediate</code></td>
<td>
<p>(logical) If <code>TRUE</code>, return a list of length two
with target taxon rank names, with additional list of data.frame's of
intermediate taxonomic groups. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="bold_downstream_+3A_...">...</code></td>
<td>
<p>crul options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>BEWARE: This function scrapes the BOLD website, so may be unstable. That is,
one day it may work, and the next it may fail. Open an issue if you
encounter an error: https://github.com/ropensci/taxize/issues
</p>


<h3>Value</h3>

<p>data.frame of taxonomic information downstream to family from e.g.,
Order, Class, etc., or if <code>intermediated=TRUE</code>, list of length two,
with target taxon rank names, and intermediate names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## the genus Gadus
bold_downstream(id = 3451, downto="species")

bold_downstream(id = 443, downto="genus")
bold_downstream(id = 443, downto="genus", intermediate=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='bold_search'>Search Barcode of Life for taxonomic IDs</h2><span id='topic+bold_search'></span>

<h3>Description</h3>

<p>Search Barcode of Life for taxonomic IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_search(
  sci = NULL,
  id = NULL,
  fuzzy = FALSE,
  dataTypes = "basic",
  includeTree = FALSE,
  response = FALSE,
  name = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bold_search_+3A_sci">sci</code></td>
<td>
<p>(character) One or more scientific names.</p>
</td></tr>
<tr><td><code id="bold_search_+3A_id">id</code></td>
<td>
<p>(integer) One or more BOLD taxonomic identifiers.</p>
</td></tr>
<tr><td><code id="bold_search_+3A_fuzzy">fuzzy</code></td>
<td>
<p>(logical) Whether to use fuzzy search or not (default: <code>FALSE</code>).
Only used if <code>name</code> passed.</p>
</td></tr>
<tr><td><code id="bold_search_+3A_datatypes">dataTypes</code></td>
<td>
<p>(character) Specifies the datatypes that will be returned.
See Details for options. This variable is ignored if <code>name</code> parameter is passed,
but is used if the <code>id</code> parameter is passed.</p>
</td></tr>
<tr><td><code id="bold_search_+3A_includetree">includeTree</code></td>
<td>
<p>(logical) If TRUE (default: FALSE), returns a list containing
information for parent taxa as well as the specified taxon. Only used if <code>id</code>
passed.</p>
</td></tr>
<tr><td><code id="bold_search_+3A_response">response</code></td>
<td>
<p>(logical) Note that response is the object that returns from the
curl call, useful for debugging, and getting detailed info on the API call.</p>
</td></tr>
<tr><td><code id="bold_search_+3A_name">name</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="bold_search_+3A_...">...</code></td>
<td>
<p>named curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>You must provide one of <code>sci</code> or <code>id</code> to this function. The other
parameters are optional. Note that when passing in <code>sci</code>, <code>fuzzy</code> can be used
as well, while if <code>id</code> is passed, then <code>fuzzy</code> is ignored, and <code>dataTypes</code>
<code>includeTree</code> can be used.
</p>
<p>Options for <code>dataTypes</code> parameter:
</p>

<ul>
<li><p> all returns all data
</p>
</li>
<li><p> basic returns basic taxon information
</p>
</li>
<li><p> images returns specimen image. Includes copyright information, image URL,
image metadata.
</p>
</li>
<li><p> stats Returns specimen and sequence statistics. Includes public species
count, public BIN count, public marker counts, public record count,
specimen count, sequenced specimen count, barcode specimen count, species
count, barcode species count.
</p>
</li>
<li><p> geo Returns collection site information. Includes country, collection
site map.
</p>
</li>
<li><p> sequencinglabs Returns sequencing labs. Includes lab name, record count.
</p>
</li>
<li><p> depository Returns specimen depositories. Includes depository name,
record count.
</p>
</li>
<li><p> thirdparty Returns information from third parties. Includes wikipedia
summary, wikipedia URL, GBIF map.
</p>
</li></ul>



<h3>Value</h3>

<p>A list of data.frame's.
</p>


<h3>References</h3>

<p>http://www.boldsystems.org/index.php/resources/api
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# A basic example
bold_search(sci="Apis")
bold_search(sci="Agapostemon")
bold_search(sci="Poa")

# Fuzzy search
head(bold_search(sci="Po", fuzzy=TRUE))
head(bold_search(sci="Aga", fuzzy=TRUE))

# Many names
bold_search(sci=c("Apis","Puma concolor"))
nms &lt;- names_list('species')
bold_search(sci=nms)

# Searching by ID - dataTypes can be used, and includeTree can be used
bold_search(id=88899)
bold_search(id=88899, dataTypes="stats")
bold_search(id=88899, dataTypes="geo")
bold_search(id=88899, dataTypes="basic")
bold_search(id=88899, includeTree=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='bold_tax_id'>Search BOLD for taxonomy data by BOLD ID.</h2><span id='topic+bold_tax_id'></span>

<h3>Description</h3>

<p>Search BOLD for taxonomy data by BOLD ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_tax_id(
  id,
  dataTypes = "basic",
  includeTree = FALSE,
  response = FALSE,
  ...
)
</code></pre>

<hr>
<h2 id='bold_tax_id2'>Search BOLD for taxonomy data by BOLD ID.</h2><span id='topic+bold_tax_id2'></span>

<h3>Description</h3>

<p>Search BOLD for taxonomy data by BOLD ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bold_tax_id2(
  id,
  dataTypes = "basic",
  includeTree = FALSE,
  response = FALSE,
  ...
)
</code></pre>

<hr>
<h2 id='children'>Retrieve immediate children taxa for a given taxon name or ID.</h2><span id='topic+children'></span><span id='topic+children.default'></span><span id='topic+children.tsn'></span><span id='topic+children.wormsid'></span><span id='topic+children.ids'></span><span id='topic+children.uid'></span><span id='topic+children.boldid'></span>

<h3>Description</h3>

<p>This function is different from <code><a href="#topic+downstream">downstream()</a></code> in that it only
collects immediate taxonomic children, while <code><a href="#topic+downstream">downstream()</a></code>
collects taxonomic names down to a specified taxonomic rank, e.g.,
getting all species in a family.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>children(...)

## Default S3 method:
children(sci_id, db = NULL, rows = NA, x = NULL, ...)

## S3 method for class 'tsn'
children(sci_id, db = NULL, ...)

## S3 method for class 'wormsid'
children(sci_id, db = NULL, ...)

## S3 method for class 'ids'
children(sci_id, db = NULL, ...)

## S3 method for class 'uid'
children(sci_id, db = NULL, ...)

## S3 method for class 'boldid'
children(sci_id, db = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="children_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="ritis.html#topic+hierarchy">ritis::hierarchy_down()</a></code>,
<code><a href="#topic+ncbi_children">ncbi_children()</a></code>, <code><a href="worrms.html#topic+wm_children">worrms::wm_children()</a></code>, <code><a href="#topic+bold_children">bold_children()</a></code>
See those functions for what parameters can be passed on.</p>
</td></tr>
<tr><td><code id="children_+3A_sci_id">sci_id</code></td>
<td>
<p>Vector of taxa names (character) or IDs (character or numeric)
to query.</p>
</td></tr>
<tr><td><code id="children_+3A_db">db</code></td>
<td>
<p>character; database to query. One or more of <code>itis</code>,
<code>ncbi</code>, <code>worms</code>, or <code>bold</code>. Note that each taxonomic data
source has their own identifiers, so that if you provide the wrong
<code>db</code> value for the identifier you could get a result, but it will
likely be wrong (not what you were expecting). If using ncbi, we recommend
getting an API key; see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="children_+3A_rows">rows</code></td>
<td>
<p>(numeric) Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this parameter is ignored if you pass in a
taxonomic id of any of the acceptable classes: tsn. NCBI has a
method for this function but rows doesn't work.</p>
</td></tr>
<tr><td><code id="children_+3A_x">x</code></td>
<td>
<p>Deprecated, see <code>sci_id</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of data.frames with the children names of every
supplied taxa. You get an NA if there was no match in the database.
</p>


<h3>ncbi</h3>

<p>note that with <code>db = "ncbi"</code>, we set <code>ambiguous = TRUE</code>; that is, children
taxa with words like &quot;unclassified&quot;, &quot;unknown&quot;, &quot;uncultured&quot;, &quot;sp.&quot; are
NOT removed
</p>


<h3>bold</h3>

<p>BEWARE: <code>db="bold"</code> scrapes the BOLD website, so may be unstable. That is,
one day it may work, and the next it may fail. Open an issue if you
encounter an error: https://github.com/ropensci/taxize/issues
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Plug in taxonomic IDs
children(161994, db = "itis")
children(8028, db = "ncbi")
## works with numeric if as character as well
children(161994, db = "itis")
children(88899, db = "bold")
children(as.boldid(88899))

# Plug in taxon names
children("Salmo", db = 'itis')
children("Salmo", db = 'ncbi')
children("Salmo", db = 'worms')
children("Salmo", db = 'bold')

# Plug in IDs
(id &lt;- get_wormsid("Gadus"))
children(id)

# Many taxa
sp &lt;- c("Tragia", "Schistocarpha", "Encalypta")
children(sp, db = 'itis')

# Two data sources
(ids &lt;- get_ids("Apis", db = c('ncbi','itis')))
children(ids)
## same result
children(get_ids("Apis", db = c('ncbi','itis')))

# Use the rows parameter
children("Poa", db = 'itis')
children("Poa", db = 'itis', rows=1)

# use curl options
res &lt;- children("Poa", db = 'itis', rows=1, verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='class2tree'>Convert a list of classifications to a tree.</h2><span id='topic+class2tree'></span><span id='topic+plot.classtree'></span><span id='topic+print.classtree'></span>

<h3>Description</h3>

<p>This function converts a list of hierarchies for individual species into
a single species by taxonomic level matrix, then calculates a distance
matrix based on taxonomy alone, and outputs either a phylo or dist object.
See details for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>class2tree(input, varstep = TRUE, check = TRUE, remove_shared = FALSE, ...)

## S3 method for class 'classtree'
plot(x, ...)

## S3 method for class 'classtree'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="class2tree_+3A_input">input</code></td>
<td>
<p>List of classification data.frame's from the function
<code><a href="#topic+classification">classification()</a></code></p>
</td></tr>
<tr><td><code id="class2tree_+3A_varstep">varstep</code></td>
<td>
<p>Vary step lengths between successive levels relative to
proportional loss of the number of distinct classes.</p>
</td></tr>
<tr><td><code id="class2tree_+3A_check">check</code></td>
<td>
<p>If TRUE, remove all redundant levels which are different for
all rows or constant for all rows and regard each row as a different basal
taxon (species). If FALSE all levels are retained and basal taxa (species)
also must be coded as variables (columns). You will get a warning if
species are not coded, but you can ignore this if that was your intention.</p>
</td></tr>
<tr><td><code id="class2tree_+3A_remove_shared">remove_shared</code></td>
<td>
<p>If <code>TRUE</code>, remove any taxa that are coarser ranks
present in other taxa, such as both a genus and a species in that genus in
the same tree.</p>
</td></tr>
<tr><td><code id="class2tree_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to hclust.</p>
</td></tr>
<tr><td><code id="class2tree_+3A_x">x</code></td>
<td>
<p>Input object to print or plot - output from class2tree function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <code><a href="vegan.html#topic+taxondive">vegan::taxa2dist()</a></code>. Thanks to Jari Oksanen for
making the taxa2dist function and pointing it out, and Clarke &amp; Warwick
(1998, 2001), which taxa2dist was based on.
The taxonomy tree created is not only based on the clustering of the
taxonomy ranks (e.g. strain, species, genus, ...), but it also utilizes the
actual taxon clades (e.g. mammals, plants or reptiles, etc.). The process of
this function is as following: First, all possible taxonomy ranks and their
corresponding IDs for each given taxon will be collected from the input.
Then, the rank vectors of all taxa will be aligned, so that they together
will become a matrix where columns are ordered taxonomy ranks of all taxa and
rows are the rank vectors of those taxa. After that, the rank matrix will be
converted into taxonomy ID matrix, any missing rank will have a pseudo
ID from the previous rank. Finally, this taxonomy ID matrix will be used to
cluster taxa that have similar taxonomy hierarchy together.
</p>


<h3>Value</h3>

<p>An object of class &quot;classtree&quot; with slots:
</p>

<ul>
<li><p> phylo - The resulting object, a phylo object
</p>
</li>
<li><p> classification - The classification data.frame, with taxa as rows,
and different classification levels as columns
</p>
</li>
<li><p> distmat - Distance matrix
</p>
</li>
<li><p> names - The names of the tips of the phylogeny
</p>
</li></ul>

<p>Note that when you execute the resulting object, you only get the phylo
object. You can get to the other 3 slots by calling them directly, like
output$names, etc.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
spnames &lt;- c('Quercus robur', 'Iris oratoria', 'Arachis paraguariensis',
 'Helianthus annuus','Madia elegans','Lupinus albicaulis',
 'Pinus lambertiana')
out &lt;- classification(spnames, db='itis')
tr &lt;- class2tree(out)
plot(tr)

spnames &lt;- c('Klattia flava', 'Trollius sibiricus',
 'Arachis paraguariensis',
 'Tanacetum boreale', 'Gentiana yakushimensis','Sesamum schinzianum',
 'Pilea verrucosa','Tibouchina striphnocalyx','Lycium dasystemum',
 'Berkheya echinacea','Androcymbium villosum',
 'Helianthus annuus','Madia elegans','Lupinus albicaulis',
 'Pinus lambertiana')
out &lt;- classification(spnames, db='ncbi')
tr &lt;- class2tree(out)
plot(tr)

## End(Not run)

</code></pre>

<hr>
<h2 id='classification'>Retrieve the taxonomic hierarchy for a given taxon ID.</h2><span id='topic+classification'></span><span id='topic+classification.default'></span><span id='topic+classification.tsn'></span><span id='topic+classification.uid'></span><span id='topic+classification.eolid'></span><span id='topic+classification.tpsid'></span><span id='topic+classification.gbifid'></span><span id='topic+classification.nbnid'></span><span id='topic+classification.tolid'></span><span id='topic+classification.wormsid'></span><span id='topic+classification.natservid'></span><span id='topic+classification.boldid'></span><span id='topic+classification.wiki'></span><span id='topic+classification.pow'></span><span id='topic+classification.ids'></span><span id='topic+cbind.classification'></span><span id='topic+rbind.classification'></span><span id='topic+cbind.classification_ids'></span><span id='topic+rbind.classification_ids'></span>

<h3>Description</h3>

<p>Retrieve the taxonomic hierarchy for a given taxon ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classification(...)

## Default S3 method:
classification(
  sci_id,
  db = NULL,
  callopts = list(),
  return_id = TRUE,
  rows = NA,
  x = NULL,
  ...
)

## S3 method for class 'tsn'
classification(id, return_id = TRUE, ...)

## S3 method for class 'uid'
classification(
  id,
  callopts = list(),
  return_id = TRUE,
  batch_size = 50,
  max_tries = 3,
  ...
)

## S3 method for class 'eolid'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'tpsid'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'gbifid'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'nbnid'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'tolid'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'wormsid'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'natservid'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'boldid'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'wiki'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'pow'
classification(id, callopts = list(), return_id = TRUE, ...)

## S3 method for class 'ids'
classification(id, ...)

## S3 method for class 'classification'
cbind(...)

## S3 method for class 'classification'
rbind(...)

## S3 method for class 'classification_ids'
cbind(...)

## S3 method for class 'classification_ids'
rbind(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="classification_+3A_...">...</code></td>
<td>
<p>For <code>classification</code>: other arguments passed to <code><a href="#topic+get_tsn">get_tsn()</a></code>,
<code><a href="#topic+get_uid">get_uid()</a></code>, <code><a href="#topic+get_eolid">get_eolid()</a></code>, <code><a href="#topic+get_tpsid">get_tpsid()</a></code>, <code><a href="#topic+get_gbifid">get_gbifid()</a></code>,
<code><a href="#topic+get_wormsid">get_wormsid()</a></code>, <code><a href="#topic+get_natservid">get_natservid()</a></code>, <code><a href="#topic+get_wormsid">get_wormsid()</a></code>, <code><a href="#topic+get_wiki">get_wiki()</a></code>,
<code><a href="#topic+get_pow">get_pow()</a></code>. For <code>rbind.classification</code> and <code>cbind.classification</code>: one or
more objects of class <code>classification</code></p>
</td></tr>
<tr><td><code id="classification_+3A_sci_id">sci_id</code></td>
<td>
<p>Vector of taxa names (character) or IDs (character or numeric)
to query. For <code>db = "eol"</code>, EOL expects you to pass it a taxon id, called
<code>eolid</code> in the output of <code><a href="#topic+get_eolid">get_eolid()</a></code>.</p>
</td></tr>
<tr><td><code id="classification_+3A_db">db</code></td>
<td>
<p>character; database to query. either <code>ncbi</code>, <code>itis</code>, <code>eol</code>,
<code>tropicos</code>, <code>gbif</code>, <code>nbn</code>, <code>worms</code>, <code>natserv</code>, <code>bold</code>, <code>wiki</code>, or <code>pow</code>.
Note that each taxonomic data source has, their own identifiers, so that
if you provide the wrong <code>db</code> value for the identifier you could get a
result, but it will likely be wrong (not what you were expecting). If using
ncbi, and/or tropicos, we recommend getting an API key; see
<a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="classification_+3A_callopts">callopts</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
<tr><td><code id="classification_+3A_return_id">return_id</code></td>
<td>
<p>(logical) If <code>TRUE</code> (default), return the taxon id
as well as the name and rank of taxa in the lineage returned.
Ignored for natserv as they don't return IDs in their taxonomic
classification data.</p>
</td></tr>
<tr><td><code id="classification_+3A_rows">rows</code></td>
<td>
<p>(numeric) Any number from 1 to infinity. If the default NA,
all rows are considered. Note that this parameter is ignored if you pass
in a taxonomic id instead of a name of class character.</p>
</td></tr>
<tr><td><code id="classification_+3A_x">x</code></td>
<td>
<p>Deprecated, see <code>sci_id</code></p>
</td></tr>
<tr><td><code id="classification_+3A_id">id</code></td>
<td>
<p>character; identifiers, returned by <code><a href="#topic+get_tsn">get_tsn()</a></code>, <code><a href="#topic+get_uid">get_uid()</a></code>,
<code><a href="#topic+get_eolid">get_eolid()</a></code>, <code><a href="#topic+get_tpsid">get_tpsid()</a></code>, <code><a href="#topic+get_gbifid">get_gbifid()</a></code>, <code><a href="#topic+get_tolid">get_tolid()</a></code>,
<code><a href="#topic+get_wormsid">get_wormsid()</a></code>, <code><a href="#topic+get_natservid">get_natservid()</a></code>, <code><a href="#topic+get_wormsid">get_wormsid()</a></code>, <code><a href="#topic+get_wiki">get_wiki()</a></code>,
<code><a href="#topic+get_pow">get_pow()</a></code></p>
</td></tr>
<tr><td><code id="classification_+3A_batch_size">batch_size</code></td>
<td>
<p>(numeric) For NCBI queries, specify the number of IDs to
lookup for each query.</p>
</td></tr>
<tr><td><code id="classification_+3A_max_tries">max_tries</code></td>
<td>
<p>(numeric) For NCBI queries, the number of times a particular
query will be attempted, assuming the first does not work.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If IDs are supplied directly (not from the <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions)
you must specify the type of ID. There is a timeout of 1/3 seconds between
queries to NCBI.
</p>
<p>BEWARE: Right now, NBN doesn't return the queried taxon in the
classification. But you can attach it yourself quite easily of course.
This behavior is different from the other data sources.
</p>


<h3>Value</h3>

<p>A named list of data.frames with the taxonomic classification of
every supplied taxa.
</p>


<h3>Lots of results</h3>

<p>It may happen sometimes that you get more results back from your query
than will show in the data.frame on screen. Our advice is to refine your
query in those cases. On a data source basis we can attempt to help
make it easier to refine queries, whether it be with the data provider
(unlikely to happen), or in the code in this package (more likely) -
let us know if you run into too many results problem and we'll see what
we can do.
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a>
</p>


<h3>EOL</h3>

<p>EOL does not have very good failure behavior. For example, if you submit
an ID that does not exist they'll return a 500 HTTP error, which is
not an appropriate error; it's probably that that ID does not exist
in their database, but we can't know for sure. Isn't that fun?
</p>


<h3>NCBI Rate limits</h3>

<p>In case you run into NCBI errors due to your rate limit being exceeded, see
<code><a href="#topic+taxize_options">taxize_options()</a></code>, where you can set <code>ncbi_sleep</code>.
</p>


<h3>HTTP version for NCBI requests</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_tsn">get_tsn()</a></code>, <code><a href="#topic+get_uid">get_uid()</a></code>, <code><a href="#topic+get_eolid">get_eolid()</a></code>,
<code><a href="#topic+get_tpsid">get_tpsid()</a></code>, <code><a href="#topic+get_gbifid">get_gbifid()</a></code>, <code><a href="#topic+get_wormsid">get_wormsid()</a></code>, <code><a href="#topic+get_natservid">get_natservid()</a></code>,
<code><a href="#topic+get_boldid">get_boldid()</a></code>, <code><a href="#topic+get_wiki">get_wiki()</a></code>, <code><a href="#topic+get_pow">get_pow()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Plug in taxon IDs
classification(9606, db = 'ncbi')
classification(c(9606, 55062), db = 'ncbi')
classification(129313, db = 'itis')
classification(6985636, db = 'eol')
classification(126436, db = 'worms')
classification('Helianthus annuus', db = 'pow')
classification('Helianthus', db = 'pow')
classification('Asteraceae', db = 'pow')
classification("134717", db = 'natserv')
classification(c(2704179, 6162875, 8286319, 2441175, 731), db = 'gbif')
classification(25509881, db = 'tropicos')
classification("NBNSYS0000004786", db = 'nbn')
classification(as.nbnid("NBNSYS0000004786"), db = 'nbn')
classification(3930798, db = 'tol')

## works the same if IDs are in class character
classification(c("2704179", "2441176"), db = 'gbif')
classification("Agapostemon", db = "bold")

# wikispecies
classification("Malus domestica", db = "wiki")
classification("Pinus contorta", db = "wiki")
classification("Pinus contorta", db = "wiki", wiki_site = "commons")
classification("Pinus contorta", db = "wiki", wiki_site = "pedia")
classification("Pinus contorta", db = "wiki", wiki_site = "pedia",
  wiki = "fr")

classification(get_wiki("Malus domestica", "commons"))
classification(get_wiki("Malus domestica", "species"))
classification(c("Pinus contorta", "Malus domestica"), db = "wiki")

# Plug in taxon names
## in this case, we use get_*() fxns internally to first get taxon IDs
classification("Oncorhynchus mykiss", db = "eol")
classification(c("Chironomus riparius", "aaa vva"), db = 'ncbi')
classification(c("Chironomus riparius", "aaa vva"), db = 'ncbi',
  messages=FALSE)
classification(c("Chironomus riparius", "aaa vva"), db = 'itis')
classification(c("Chironomus riparius", "aaa vva"), db = 'itis',
  messages=FALSE)
classification(c("Chironomus riparius", "aaa vva"), db = 'eol')
classification("Alopias vulpinus", db = 'nbn')
classification('Gadus morhua', db = 'worms')
classification('Aquila chrysaetos', db = 'natserv')
classification('Gadus morhua', db = 'natserv')
classification('Pomatomus saltatrix', db = 'natserv')
classification('Aquila chrysaetos', db = 'natserv')
classification(c("Chironomus riparius", "asdfasdfsfdfsd"), db = 'gbif')
classification("Chironomus", db = 'tol')
classification("Poa annua", db = 'tropicos')

# Use methods for get_uid, get_tsn, get_eolid, get_tpsid
classification(get_uid(c("Chironomus riparius", "Puma concolor")))

classification(get_uid(c("Chironomus riparius", "aaa vva")))
classification(get_tsn(c("Chironomus riparius", "aaa vva")))
classification(get_tsn(c("Chironomus riparius", "aaa vva"),
  messages = FALSE))
classification(get_eolid(c("Chironomus riparius", "aaa vva")))
classification(get_tpsid(c("Poa annua", "aaa vva")))
classification(get_gbifid(c("Poa annua", "Bison bison")))

# Pass many ids from class "ids"
(out &lt;- get_ids("Puma concolor", db = c('ncbi','gbif')))
(cl &lt;- classification(out))

# Bind width-wise from class classification_ids
cbind(cl)

# Bind length-wise
rbind(cl)

# Many names to get_ids
(out &lt;- get_ids(c("Puma concolor","Accipiter striatus"),
  db = c('ncbi','itis')))
(cl &lt;- classification(out))
rbind(cl)
## cbind with so many names results in some messy data
cbind(cl)
## so you can turn off return_id
cbind( classification(out, return_id=FALSE) )

(cl_uid &lt;- classification(get_uid(c("Puma concolor",
  "Accipiter striatus")), return_id=FALSE))
rbind(cl_uid)
cbind(cl_uid)
## cbind works a bit odd when there are lots of ranks without names
(cl_uid &lt;- classification(get_uid(c("Puma concolor","Accipiter striatus")),
  return_id=TRUE))
cbind(cl_uid)

(cl_tsn &lt;- classification(get_tsn(c("Puma concolor","Accipiter striatus"))))
rbind(cl_tsn)
cbind(cl_tsn)

(tsns &lt;- get_tsn(c("Puma concolor","Accipiter striatus")))
(cl_tsns &lt;- classification(tsns))
cbind(cl_tsns)

# NBN data
(res &lt;- classification(c("Alopias vulpinus","Pinus sylvestris"),
  db = 'nbn'))
rbind(res)
cbind(res)

# Return taxonomic IDs
## the return_id parameter is logical, and you can turn it on or off.
## It's TRUE by default
classification(c("Alopias vulpinus","Pinus sylvestris"), db = 'ncbi',
  return_id = TRUE)
classification(c("Alopias vulpinus","Pinus sylvestris"), db = 'ncbi',
  return_id = FALSE)

# Use rows parameter to select certain
classification('Poa annua', db = 'tropicos')
classification('Poa annua', db = 'tropicos', rows=1:4)
classification('Poa annua', db = 'tropicos', rows=1)
classification('Poa annua', db = 'tropicos', rows=6)

# Queries of many IDs are processed in batches for NCBI
ids &lt;- c("13083", "2650392", "1547764", "230054", "353934", "656984", 
"271789", "126272", "184644", "73213", "662816", "1161803", "1239353", 
"59420", "665675", "866969", "1091219", "1431218", "1471898", 
"864321", "251768", "2486276", "2068772", "1825808", "2006532", 
"128287", "1195738", "1084683", "1886461", "508296", "377247", 
"1489665", "329325", "219243", "1176946", "339893", "197933", 
"174510", "1704048", "212897", "154842", "1239280", "260135", 
"405735", "1566412", "2083462", "651348", "983204", "165380", 
"2338856", "2068760", "167262", "34229", "1213340", "478939", 
"1933585", "49951", "1277794", "1671089", "1502538", "362355", 
"746473", "242879", "158219", "313664", "2093188", "1541232", 
"584742", "1331091", "147639", "284492", "75642", "1412882", 
"391782", "1406855", "434506", "2053357", "217315", "1444328", 
"329249", "2294004", "84942", "324458", "538247", "69452", "49170", 
"1993845", "261646", "127633", "228146", "1420004", "1629772", 
"577055", "697062", "231660", "648380", "554953", "746496", "2602969")
result &lt;- classification(ids, db = 'ncbi')

## End(Not run)
</code></pre>

<hr>
<h2 id='col-defunct'>COL defunct functions</h2><span id='topic+col-defunct'></span><span id='topic+as.colid'></span><span id='topic+col_children'></span><span id='topic+col_classification'></span><span id='topic+col_downstream'></span><span id='topic+col_search'></span><span id='topic+get_colid'></span><span id='topic+get_colid_'></span>

<h3>Description</h3>

<p>COL introduced rate limiting recently in 2019 - which has made the API
essentially unusable - COL+ is coming soon and we'll incorporate it here
when it's stable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.colid(...)

col_children(...)

col_classification(...)

col_downstream(...)

col_search(...)

get_colid(...)

get_colid_(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="col-defunct_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='comm2sci'>Get scientific names from common names.</h2><span id='topic+comm2sci'></span><span id='topic+comm2sci.default'></span><span id='topic+comm2sci.tsn'></span><span id='topic+comm2sci.uid'></span>

<h3>Description</h3>

<p>Get scientific names from common names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comm2sci(...)

## Default S3 method:
comm2sci(
  com,
  db = "ncbi",
  itisby = "search",
  simplify = TRUE,
  commnames = NULL,
  ...
)

## S3 method for class 'tsn'
comm2sci(id, db = "ncbi", itisby = "search", simplify = TRUE, ...)

## S3 method for class 'uid'
comm2sci(id, db = "ncbi", itisby = "search", simplify = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="comm2sci_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to internal methods.</p>
</td></tr>
<tr><td><code id="comm2sci_+3A_com">com</code></td>
<td>
<p>One or more common names or partial names.</p>
</td></tr>
<tr><td><code id="comm2sci_+3A_db">db</code></td>
<td>
<p>Data source, one of <em>&quot;ncbi&quot;</em> (default), <em>&quot;itis&quot;</em>,
<em>&quot;tropicos&quot;</em>, <em>&quot;eol&quot;</em>, or <em>&quot;worms&quot;</em>. If using ncbi, we
recommend getting an API key; see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="comm2sci_+3A_itisby">itisby</code></td>
<td>
<p>Search for common names across entire names (search, default),
at beginning of names (begin), or at end of names (end).</p>
</td></tr>
<tr><td><code id="comm2sci_+3A_simplify">simplify</code></td>
<td>
<p>(logical) If <code>TRUE</code>, simplify output to a vector of names.
If <code>FALSE</code>, return variable formats from different sources, usually a
data.frame.</p>
</td></tr>
<tr><td><code id="comm2sci_+3A_commnames">commnames</code></td>
<td>
<p>Deprecated, see <code>com</code></p>
</td></tr>
<tr><td><code id="comm2sci_+3A_id">id</code></td>
<td>
<p>taxon identifiers, as returned by <code><a href="#topic+get_tsn">get_tsn()</a></code> or <code><a href="#topic+get_uid">get_uid()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For data sources ITIS and NCBI you can pass in common names
directly, and use <code><a href="#topic+get_uid">get_uid()</a></code> or <code><a href="#topic+get_tsn">get_tsn()</a></code> to get ids first, then pass in
to this fxn.
</p>
<p>For the other data sources, you can only pass in common names directly.
</p>


<h3>Value</h3>

<p>If <code>simplify=TRUE</code>, a list of scientific names, with list
labeled by your input names. If <code>simplify=FALSE</code>, a data.frame with
columns that vary by data source. <code>character(0)</code> on no match
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication
</p>


<h3>HTTP version for NCBI requests</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>Rate limits</h3>

<p>In case you run into errors due to your rate limit being exceeded, see
<code><a href="#topic+taxize_options">taxize_options()</a></code>, where you can set <code>ncbi_sleep</code>.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sci2comm">sci2comm()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
comm2sci(com='american black bear')
comm2sci(com='american black bear', simplify = FALSE)
comm2sci(com='black bear', db='itis')
comm2sci(com='american black bear', db='itis')
comm2sci(com='annual blue grass', db='tropicos')
comm2sci(com=c('annual blue grass','tree of heaven'), db='tropicos')
comm2sci('blue whale', db = "worms")
comm2sci(c('blue whale', 'dwarf surfclam'), db = "worms")

# ncbi: pass in uid's from get_uid() directly
x &lt;- get_uid("western capercaillie", modifier = "Common Name")
comm2sci(x)
# itis: pass in tsn's from get_tsn() directly
x &lt;- get_tsn(c("Louisiana black bear", "american crow"),
  searchtype = "common")
comm2sci(x)

## End(Not run)
</code></pre>

<hr>
<h2 id='downstream'>Retrieve the downstream taxa for a given taxon name or ID.</h2><span id='topic+downstream'></span><span id='topic+downstream.default'></span><span id='topic+downstream.tsn'></span><span id='topic+downstream.gbifid'></span><span id='topic+downstream.uid'></span><span id='topic+downstream.wormsid'></span><span id='topic+downstream.boldid'></span><span id='topic+downstream.ids'></span>

<h3>Description</h3>

<p>This function uses a while loop to continually collect children taxa down
to the taxonomic rank that you specify in the <code>downto</code> parameter. You
can get data from ITIS (itis), GBIF (gbif), NCBI (ncbi), WORMS (worms),
or BOLD (bold). There is no method exposed by these four
services for getting taxa at a specific taxonomic rank, so we do it
ourselves here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downstream(...)

## Default S3 method:
downstream(
  sci_id,
  db = NULL,
  downto = NULL,
  intermediate = FALSE,
  rows = NA,
  x = NULL,
  ...
)

## S3 method for class 'tsn'
downstream(sci_id, db = NULL, downto = NULL, intermediate = FALSE, ...)

## S3 method for class 'gbifid'
downstream(
  sci_id,
  db = NULL,
  downto = NULL,
  intermediate = FALSE,
  limit = 100,
  start = NULL,
  ...
)

## S3 method for class 'uid'
downstream(sci_id, db = NULL, downto = NULL, intermediate = FALSE, ...)

## S3 method for class 'wormsid'
downstream(sci_id, db = NULL, downto = NULL, intermediate = FALSE, ...)

## S3 method for class 'boldid'
downstream(sci_id, db = NULL, downto = NULL, intermediate = FALSE, ...)

## S3 method for class 'ids'
downstream(sci_id, db = NULL, downto = NULL, intermediate = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downstream_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="#topic+itis_downstream">itis_downstream()</a></code>,
<code><a href="#topic+gbif_downstream">gbif_downstream()</a></code>, <code><a href="#topic+ncbi_downstream">ncbi_downstream()</a></code>,
<code><a href="#topic+worms_downstream">worms_downstream()</a></code>, or <code><a href="#topic+bold_downstream">bold_downstream()</a></code></p>
</td></tr>
<tr><td><code id="downstream_+3A_sci_id">sci_id</code></td>
<td>
<p>Vector of taxa names (character) or IDs (character or numeric)
to query.</p>
</td></tr>
<tr><td><code id="downstream_+3A_db">db</code></td>
<td>
<p>character; database to query. One or more of <code>itis</code>, <code>gbif</code>,
<code>ncbi</code>, <code>worms</code>, or <code>bold</code>. Note that each taxonomic  data source has
their own identifiers, so that if you provide the wrong <code>db</code> value for
the identifier
you could get a result, but it will likely be wrong (not what you were
expecting). If using ncbi, we recommend getting an API key; see
<a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="downstream_+3A_downto">downto</code></td>
<td>
<p>What taxonomic rank to go down to. One of: 'superkingdom',
'kingdom', 'subkingdom','infrakingdom','phylum','division','subphylum',
'subdivision','infradivision', 'superclass','class','subclass','infraclass',
'superorder','order','suborder','infraorder','superfamily','family',
'subfamily','tribe','subtribe','genus','subgenus','section','subsection',
'species group','species','subspecies','variety','form','subvariety','race',
'stirp', 'morph','aberration','subform', 'unspecified', 'no rank'</p>
</td></tr>
<tr><td><code id="downstream_+3A_intermediate">intermediate</code></td>
<td>
<p>(logical) If <code>TRUE</code>, return a list of length two
with target taxon rank names, with additional list of data.frame's of
intermediate taxonomic groups. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="downstream_+3A_rows">rows</code></td>
<td>
<p>(numeric) Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this parameter is ignored if you pass in a
taxonomic id of any of the acceptable classes: tsn.</p>
</td></tr>
<tr><td><code id="downstream_+3A_x">x</code></td>
<td>
<p>Deprecated, see <code>sci_id</code></p>
</td></tr>
<tr><td><code id="downstream_+3A_limit">limit</code></td>
<td>
<p>Number of records to return. Applies to gbif only.
default: 100. max: 1000. use in combination with the <code>start</code> parameter</p>
</td></tr>
<tr><td><code id="downstream_+3A_start">start</code></td>
<td>
<p>Record number to start at. Applies to gbif only. default: 0.
use in combination with the <code>limit</code> parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of data.frames with the downstream names of every
supplied taxa. You get an NA if there was no match in the database.
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication
</p>


<h3>bold</h3>

<p>BEWARE: <code>db="bold"</code> scrapes the BOLD website, so may be unstable. That is,
one day it may work, and the next it may fail. Open an issue if you
encounter an error: https://github.com/ropensci/taxize/issues
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Plug in taxon IDs
downstream(125732, db = 'worms', downto = 'species')
downstream(3451, db = 'bold', downto = 'species')

if (interactive()) {

# Plug in taxon names
downstream("Apis", db = 'ncbi', downto = 'species')
downstream("Apis", db = 'itis', downto = 'species')
downstream("Apis", db = 'bold', downto = 'species')
downstream("Gadus", db = 'worms', downto = 'species')
downstream(c("Apis","Epeoloides"), db = 'itis', downto = 'species')
downstream("Ursus", db = 'gbif', downto = 'species')
downstream(get_gbifid("Ursus"), db = 'gbif', downto = 'species')

# Many taxa
sp &lt;- names_list("genus", 3)
downstream(sp, db = 'itis', downto = 'species')
downstream(sp, db = 'gbif', downto = 'species')

# Both data sources
ids &lt;- get_ids("Apis", db = c('gbif','itis'))
downstream(ids, downto = 'species')
## same result
downstream(get_ids("Apis", db = c('gbif','itis')), downto = 'species')

# Collect intermediate names
## itis
downstream('Bangiophyceae', db="itis", downto="genus")
downstream('Bangiophyceae', db="itis", downto="genus", intermediate=TRUE)
downstream(get_tsn('Bangiophyceae'), downto="genus")
downstream(get_tsn('Bangiophyceae'), downto="genus", intermediate=TRUE)

# Use the rows parameter
## note how in the second function call you don't get the prompt
downstream("Poa", db = 'gbif', downto="species")
downstream("Poa", db = 'gbif', downto="species", rows=1)

# use curl options
res &lt;- downstream("Apis", db = 'gbif', downto = 'species', verbose = TRUE)

# Pagination
# GBIF limits queries to a maximum of 1000 records per request, so if
# there's more than 1000, use the start parameter
# Piper, taxonKey = 3075433
z1 &lt;- downstream(3075433, db = 'gbif', downto = "species", limit=1000)
z2 &lt;- downstream(3075433, db = 'gbif', downto = "species", limit=1000,
  start=1000)
z3 &lt;- downstream(3075433, db = 'gbif', downto = "species", limit=1000,
  start=2000)
z4 &lt;- downstream(3075433, db = 'gbif', downto = "species", limit=1000,
  start=3000)
NROW(rbind(z1[[1]], z2[[1]], z3[[1]], z4[[1]]))
}
## End(Not run)
</code></pre>

<hr>
<h2 id='eol_dataobjects'>Given the identifier for a data object, return all metadata about the object</h2><span id='topic+eol_dataobjects'></span>

<h3>Description</h3>

<p>Given the identifier for a data object, return all metadata about the object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eol_dataobjects(id, taxonomy = TRUE, language = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eol_dataobjects_+3A_id">id</code></td>
<td>
<p>(character) The EOL data object identifier</p>
</td></tr>
<tr><td><code id="eol_dataobjects_+3A_taxonomy">taxonomy</code></td>
<td>
<p>(logical) Whether to return any taxonomy details from
different taxon hierarchy providers, in an array named <code>taxonconcepts</code></p>
</td></tr>
<tr><td><code id="eol_dataobjects_+3A_language">language</code></td>
<td>
<p>(character) provides the results in the specified language.
one of ms, de, en, es, fr, gl, it, nl, nb, oc, pt-BR, sv, tl, mk, sr, uk,
ar, zh-Hans, zh-Hant, ko</p>
</td></tr>
<tr><td><code id="eol_dataobjects_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>It's possible to return JSON or XML with the EOL API. However,
this function only returns JSON for now.
</p>


<h3>Value</h3>

<p>A list, optionally with a data.frame if <code>taxonomy=TRUE</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
eol_dataobjects(id = 7561533)

# curl options
eol_dataobjects(id = 7561533, verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='eol_hierarchy'>Retrieve the taxonomic hierarchy from given EOL taxonID.</h2><span id='topic+eol_hierarchy'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eol_hierarchy(...)
</code></pre>

<hr>
<h2 id='eol_invasive'>Search for presence of taxonomic names in EOL invasive species databases.</h2><span id='topic+eol_invasive'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eol_invasive(...)
</code></pre>

<hr>
<h2 id='eol_pages'>Search for pages in EOL database using a taxonconceptID.</h2><span id='topic+eol_pages'></span>

<h3>Description</h3>

<p>Search for pages in EOL database using a taxonconceptID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eol_pages(
  taxonconceptID,
  images_per_page = NULL,
  images_page = NULL,
  videos_per_page = NULL,
  videos_page = NULL,
  sounds_per_page = NULL,
  sounds_page = NULL,
  maps_per_page = NULL,
  maps_page = NULL,
  texts_per_page = NULL,
  texts_page = NULL,
  subjects = "overview",
  licenses = "all",
  details = FALSE,
  common_names = FALSE,
  synonyms = FALSE,
  references = FALSE,
  taxonomy = TRUE,
  vetted = 0,
  cache_ttl = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eol_pages_+3A_taxonconceptid">taxonconceptID</code></td>
<td>
<p>(numeric) a taxonconceptID, which is also the page
number</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_images_per_page">images_per_page</code></td>
<td>
<p>(integer) number of returned image objects (0-75)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_images_page">images_page</code></td>
<td>
<p>(integer) images page</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_videos_per_page">videos_per_page</code></td>
<td>
<p>(integer) number of returned video objects (0-75)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_videos_page">videos_page</code></td>
<td>
<p>(integer) videos page</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_sounds_per_page">sounds_per_page</code></td>
<td>
<p>(integer) number of returned sound objects (0-75)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_sounds_page">sounds_page</code></td>
<td>
<p>(integer) sounds page</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_maps_per_page">maps_per_page</code></td>
<td>
<p>(integer) number of returned map objects (0-75)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_maps_page">maps_page</code></td>
<td>
<p>(integer) maps page</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_texts_per_page">texts_per_page</code></td>
<td>
<p>(integer) number of returned text objects (0-75)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_texts_page">texts_page</code></td>
<td>
<p>(integer) texts page</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_subjects">subjects</code></td>
<td>
<p>'overview' (default) to return the overview text
(if exists), a pipe | delimited list of subject names from the list of EOL
accepted subjects (e.g. TaxonBiology, FossilHistory), or 'all' to get text
in any subject. Always returns an overview text as a first result (if one
exists in the given context).</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_licenses">licenses</code></td>
<td>
<p>A pipe | delimited list of licenses or 'all' (default) to
get objects under any license. Licenses abbreviated cc- are all Creative
Commons licenses. Visit their site for more information on the various
licenses they offer.</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_details">details</code></td>
<td>
<p>Include all metadata for data objects. (Default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_common_names">common_names</code></td>
<td>
<p>Return all common names for the page's taxon
(Default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_synonyms">synonyms</code></td>
<td>
<p>Return all synonyms for the page's taxon
(Default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_references">references</code></td>
<td>
<p>Return all references for the page's taxon
(Default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_taxonomy">taxonomy</code></td>
<td>
<p>(logical) Whether to return any taxonomy details from
different taxon hierarchy providers, in an array named <code>taxonconcepts</code>
(Default: <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_vetted">vetted</code></td>
<td>
<p>If 'vetted' is given a value of '1', then only trusted
content will be returned. If 'vetted' is '2', then only trusted and
unreviewed content will be returned (untrusted content will not be returned).
The default is to return all content. (Default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_cache_ttl">cache_ttl</code></td>
<td>
<p>The number of seconds you wish to have the response cached.</p>
</td></tr>
<tr><td><code id="eol_pages_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>It's possible to return JSON or XML with the EOL API. However,
this function only returns JSON for now.
</p>


<h3>Value</h3>

<p>JSON list object, or data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
(pageid &lt;- eol_search('Pomatomus')$pageid[1])
x &lt;- eol_pages(taxonconceptID = pageid)
x
x$scinames

z &lt;- eol_pages(taxonconceptID = pageid, synonyms = TRUE)
z$synonyms

z &lt;- eol_pages(taxonconceptID = pageid, common_names = TRUE)
z$vernacular

## End(Not run)
</code></pre>

<hr>
<h2 id='eol_search'>Search for terms in EOL database.</h2><span id='topic+eol_search'></span>

<h3>Description</h3>

<p>Search for terms in EOL database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eol_search(
  sci,
  page = 1,
  exact = NULL,
  filter_tid = NULL,
  filter_heid = NULL,
  filter_by_string = NULL,
  cache_ttl = NULL,
  terms = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eol_search_+3A_sci">sci</code></td>
<td>
<p>(character) scientific name</p>
</td></tr>
<tr><td><code id="eol_search_+3A_page">page</code></td>
<td>
<p>A maximum of 30 results are returned per page. This parameter
allows you to fetch more pages of results if there are more than
30 matches (Default 1)</p>
</td></tr>
<tr><td><code id="eol_search_+3A_exact">exact</code></td>
<td>
<p>Will find taxon pages if the preferred name or any synonym
or common name exactly matches the search term.</p>
</td></tr>
<tr><td><code id="eol_search_+3A_filter_tid">filter_tid</code></td>
<td>
<p>Given an EOL page ID, search results will be limited
to members of that taxonomic group</p>
</td></tr>
<tr><td><code id="eol_search_+3A_filter_heid">filter_heid</code></td>
<td>
<p>Given a Hierarchy Entry ID, search results will be
limited to members of that taxonomic group</p>
</td></tr>
<tr><td><code id="eol_search_+3A_filter_by_string">filter_by_string</code></td>
<td>
<p>Given a search term, an exact search will be made
and that matching page will be used as the taxonomic group against which
to filter search results</p>
</td></tr>
<tr><td><code id="eol_search_+3A_cache_ttl">cache_ttl</code></td>
<td>
<p>The number of seconds you wish to have the response cached.</p>
</td></tr>
<tr><td><code id="eol_search_+3A_terms">terms</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="eol_search_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>It's possible to return JSON or XML with the EOL API. However,
this function only returns JSON for now.
</p>


<h3>Value</h3>

<p>A data frame with four columns:
</p>

<ul>
<li><p> pageid: pageid, this is the same as the eolid you can get from
<code><a href="#topic+get_eolid">get_eolid()</a></code>
</p>
</li>
<li><p> name: taxonomic name, may or may not contain the taxonomic authority
</p>
</li>
<li><p> link: URL for the taxon in question
</p>
</li>
<li><p> content: a string of semi-colon separated names. it's not clear
to us what these represent exactly, but figured why not give it to users
in case some may find it useful
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
eol_search(sci='Homo')
eol_search(sci='Salix', verbose = TRUE)
eol_search(sci='Ursus americanus')
eol_search('Pinus contorta')

## End(Not run)
</code></pre>

<hr>
<h2 id='eubon'>DEFUNCT</h2><span id='topic+eubon'></span>

<h3>Description</h3>

<p>DEFUNCT
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eubon(...)
</code></pre>

<hr>
<h2 id='eubon_capabilities'>EUBON capabilities</h2><span id='topic+eubon_capabilities'></span>

<h3>Description</h3>

<p>EUBON capabilities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eubon_capabilities(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eubon_capabilities_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>References</h3>

<p>https://cybertaxonomy.eu/eu-bon/utis/1.3/doc.html
</p>


<h3>See Also</h3>

<p>Other eubon-methods: 
<code><a href="#topic+eubon_children">eubon_children</a>()</code>,
<code><a href="#topic+eubon_hierarchy">eubon_hierarchy</a>()</code>,
<code><a href="#topic+eubon_search">eubon_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
eubon_capabilities()

## End(Not run)
</code></pre>

<hr>
<h2 id='eubon_children'>EUBON children</h2><span id='topic+eubon_children'></span>

<h3>Description</h3>

<p>EUBON children
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eubon_children(id, providers = NULL, timeout = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eubon_children_+3A_id">id</code></td>
<td>
<p>(character) identifier for the taxon. (LSID, DOI, URI, or any
other identifier used by the checklist provider)</p>
</td></tr>
<tr><td><code id="eubon_children_+3A_providers">providers</code></td>
<td>
<p>(character) A list of provider id strings concatenated by
comma characters. The default : &quot;pesi,bgbm-cdm-server[col] will
be used if this parameter is not set. A list of all available provider ids
can be obtained from the '/capabilities' service end point. Providers can be
nested, that is a parent provider can have sub providers. If the id of the
parent provider is supplied all subproviders will be queried. The query
can also be restricted to one or more subproviders by using the following
syntax: parent-id[sub-id-1,sub-id2,...]</p>
</td></tr>
<tr><td><code id="eubon_children_+3A_timeout">timeout</code></td>
<td>
<p>(numeric) The maximum of milliseconds to wait for responses
from any of the providers. If the timeout is exceeded the service will just
return the responses that have been received so far. The default timeout is
0 ms (wait for ever)</p>
</td></tr>
<tr><td><code id="eubon_children_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame or an empty list if no results found
</p>


<h3>Note</h3>

<p>There is no pagination in this method, so you may or may not be
getting all the results for a search. Sorry, out of our control
</p>


<h3>References</h3>

<p>https://cybertaxonomy.eu/eu-bon/utis/1.3/doc.html
</p>


<h3>See Also</h3>

<p>Other eubon-methods: 
<code><a href="#topic+eubon_capabilities">eubon_capabilities</a>()</code>,
<code><a href="#topic+eubon_hierarchy">eubon_hierarchy</a>()</code>,
<code><a href="#topic+eubon_search">eubon_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
x &lt;- eubon_children(id = "urn:lsid:marinespecies.org:taxname:126141",
  providers = 'worms')
head(x)

## End(Not run)
</code></pre>

<hr>
<h2 id='eubon_hierarchy'>EUBON hierarchy</h2><span id='topic+eubon_hierarchy'></span>

<h3>Description</h3>

<p>EUBON hierarchy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eubon_hierarchy(id, providers = "pesi", timeout = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eubon_hierarchy_+3A_id">id</code></td>
<td>
<p>(character) identifier for the taxon. (LSID, DOI, URI, or any
other identifier used by the checklist provider)</p>
</td></tr>
<tr><td><code id="eubon_hierarchy_+3A_providers">providers</code></td>
<td>
<p>(character) A list of provider id strings concatenated by
comma characters. The default : &quot;pesi,bgbm-cdm-server[col]&quot; will be
used if this parameter is not set. A list of all available provider ids
can be obtained from the '/capabilities' service end point. Providers can be
nested, that is a parent provider can have sub providers. If the id of the
parent provider is supplied all subproviders will be queried. The query
can also be restricted to one or more subproviders by using the following
syntax: parent-id[sub-id-1,sub-id2,...]</p>
</td></tr>
<tr><td><code id="eubon_hierarchy_+3A_timeout">timeout</code></td>
<td>
<p>(numeric) The maximum of milliseconds to wait for responses
from any of the providers. If the timeout is exceeded the service will just
return the responses that have been received so far. The default timeout is
0 ms (wait for ever)</p>
</td></tr>
<tr><td><code id="eubon_hierarchy_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Note</h3>

<p>There is no pagination in this method, so you may or may not be
getting all the results for a search. Sorry, out of our control
</p>


<h3>References</h3>

<p>https://cybertaxonomy.eu/eu-bon/utis/1.3/doc.html
</p>


<h3>See Also</h3>

<p>Other eubon-methods: 
<code><a href="#topic+eubon_capabilities">eubon_capabilities</a>()</code>,
<code><a href="#topic+eubon_children">eubon_children</a>()</code>,
<code><a href="#topic+eubon_search">eubon_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
eubon_hierarchy(id = "urn:lsid:marinespecies.org:taxname:126141", 'worms')
eubon_hierarchy(id = "urn:lsid:marinespecies.org:taxname:274350", 'worms')

## End(Not run)
</code></pre>

<hr>
<h2 id='eubon_search'>EUBON taxonomy search</h2><span id='topic+eubon_search'></span>

<h3>Description</h3>

<p>EUBON taxonomy search
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eubon_search(
  query,
  providers = "pesi",
  searchMode = "scientificNameExact",
  addSynonymy = FALSE,
  addParentTaxon = FALSE,
  timeout = 0,
  dedup = NULL,
  limit = 20,
  page = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eubon_search_+3A_query">query</code></td>
<td>
<p>(character) The scientific name to search for. For example:
&quot;Bellis perennis&quot;, &quot;Prionus&quot; or &quot;Bolinus brandaris&quot;. This is an exact search
so wildcard characters are not supported</p>
</td></tr>
<tr><td><code id="eubon_search_+3A_providers">providers</code></td>
<td>
<p>(character) A list of provider id strings concatenated by
comma characters. The default : &quot;pesi,bgbm-cdm-server[col]&quot; will be
used if this parameter is not set. A list of all available provider ids
can be obtained from the '/capabilities' service end point. Providers can be
nested, that is a parent provider can have sub providers. If the id of the
parent provider is supplied all subproviders will be queried. The query
can also be restricted to one or more subproviders by using the following
syntax: parent-id[sub-id-1,sub-id2,...]</p>
</td></tr>
<tr><td><code id="eubon_search_+3A_searchmode">searchMode</code></td>
<td>
<p>(character) Specifies the searchMode. Possible search
modes are: <code>scientificNameExact</code>, <code>scientificNameLike</code> (begins with),
<code>vernacularNameExact</code>, <code>vernacularNameLike</code>
(contains), <code>findByIdentifier</code>. If the a provider does not support the
chosen searchMode it will be skipped and the status message in the
tnrClientStatus will be set to 'unsupported search mode' in this case.</p>
</td></tr>
<tr><td><code id="eubon_search_+3A_addsynonymy">addSynonymy</code></td>
<td>
<p>(logical) Indicates whether the synonymy of the accepted
taxon should be included into the response. Turning this option on may
cause an increased response time. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="eubon_search_+3A_addparenttaxon">addParentTaxon</code></td>
<td>
<p>(logical) Indicates whether the the parent taxon of
the accepted taxon should be included into the response. Turning this option
on may cause a slightly increased response time. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="eubon_search_+3A_timeout">timeout</code></td>
<td>
<p>(numeric) The maximum of milliseconds to wait for responses
from any of the providers. If the timeout is exceeded the service will just
return the responses that have been received so far. The default timeout is
0 ms (wait for ever)</p>
</td></tr>
<tr><td><code id="eubon_search_+3A_dedup">dedup</code></td>
<td>
<p>(character) Allows to deduplicate the results by making use of
a deduplication strategy. The deduplication is done by comparing
specific properties of the taxon:
</p>

<ul>
<li><p> id: compares 'taxon.identifier'
</p>
</li>
<li><p> id_name: compares 'taxon.identifier' AND
'taxon.taxonName.scientificName'
</p>
</li>
<li><p> name: compares 'taxon.taxonName.scientificName' Using the pure
'name' strategy is not recommended.
</p>
</li></ul>
</td></tr>
<tr><td><code id="eubon_search_+3A_limit">limit</code></td>
<td>
<p>(numeric/integer) number of records to retrieve. default: 20.
This only affects the search mode <code>scientificNameLike</code> and
<code>vernacularNameLike</code>; other search modes are expected to return only one
record per check list</p>
</td></tr>
<tr><td><code id="eubon_search_+3A_page">page</code></td>
<td>
<p>(numeric/integer) page to retrieve. default: 1. This only
affects the search mode <code>scientificNameLike</code> and <code>vernacularNameLike</code>; other
search modes are expected to return only one record per check list</p>
</td></tr>
<tr><td><code id="eubon_search_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>References</h3>

<p>https://cybertaxonomy.eu/eu-bon/utis/1.3/doc.html
</p>


<h3>See Also</h3>

<p>Other eubon-methods: 
<code><a href="#topic+eubon_capabilities">eubon_capabilities</a>()</code>,
<code><a href="#topic+eubon_children">eubon_children</a>()</code>,
<code><a href="#topic+eubon_hierarchy">eubon_hierarchy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
eubon_search("Prionus")
eubon_search("Salmo", "pesi")
eubon_search("Salmo", c("pesi", "worms"))
eubon_search("Salmo", "worms", "scientificNameLike")
eubon_search("Salmo", "worms", "scientificNameLike", limit = 3)
eubon_search("Salmo", "worms", "scientificNameLike", limit = 20, page = 2)
eubon_search("Salmo", "worms", addSynonymy = TRUE)
eubon_search("Salmo", "worms", addParentTaxon = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='fungorum'>Index Fungorum</h2><span id='topic+fungorum'></span><span id='topic+fg_name_search'></span><span id='topic+fg_author_search'></span><span id='topic+fg_epithet_search'></span><span id='topic+fg_name_by_key'></span><span id='topic+fg_name_full_by_lsid'></span><span id='topic+fg_all_updated_names'></span><span id='topic+fg_deprecated_names'></span>

<h3>Description</h3>

<p>Search for taxonomic names in Index Fungorum
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fg_name_search(q, anywhere = TRUE, limit = 10, ...)

fg_author_search(q, anywhere = TRUE, limit = 10, ...)

fg_epithet_search(q, anywhere = TRUE, limit = 10, ...)

fg_name_by_key(key, ...)

fg_name_full_by_lsid(lsid, ...)

fg_all_updated_names(date, ...)

fg_deprecated_names(date, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fungorum_+3A_q">q</code></td>
<td>
<p>(character) Query term</p>
</td></tr>
<tr><td><code id="fungorum_+3A_anywhere">anywhere</code></td>
<td>
<p>(logical) Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="fungorum_+3A_limit">limit</code></td>
<td>
<p>(integer) Number of results to return. max limit
value appears to be 6000, not positive about that though</p>
</td></tr>
<tr><td><code id="fungorum_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
<tr><td><code id="fungorum_+3A_key">key</code></td>
<td>
<p>(character) A IndexFungorum taxon key</p>
</td></tr>
<tr><td><code id="fungorum_+3A_lsid">lsid</code></td>
<td>
<p>(character) an LSID, e.,g. &quot;urn:lsid:indexfungorum.org:names:81085&quot;</p>
</td></tr>
<tr><td><code id="fungorum_+3A_date">date</code></td>
<td>
<p>(character) Date, of the form YYYMMDD</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code>, or <code>NULL</code> if no results
</p>


<h3>References</h3>

<p>http://www.indexfungorum.org/, API docs:
http://www.indexfungorum.org/ixfwebservice/fungus.asmx
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# NameSearch
fg_name_search(q = "Gymnopus", limit = 2, verbose = TRUE)
fg_name_search(q = "Gymnopus")

# EpithetSearch
fg_epithet_search(q = "phalloides")

# NameByKey
fg_name_by_key(17703)

# NameFullByKey
fg_name_full_by_lsid("urn:lsid:indexfungorum.org:names:81085")

# AllUpdatedNames
fg_all_updated_names(date = gsub("-", "", Sys.Date() - 2))

# DeprecatedNames
fg_deprecated_names(date=20151001)

# AuthorSearch
fg_author_search(q = "Fayod", limit = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='gbif_downstream'>Retrieve all taxonomic names downstream in hierarchy for GBIF</h2><span id='topic+gbif_downstream'></span>

<h3>Description</h3>

<p>Retrieve all taxonomic names downstream in hierarchy for GBIF
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gbif_downstream(
  id,
  downto,
  intermediate = FALSE,
  limit = 100,
  start = NULL,
  key = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gbif_downstream_+3A_id">id</code></td>
<td>
<p>A taxonomic serial number.</p>
</td></tr>
<tr><td><code id="gbif_downstream_+3A_downto">downto</code></td>
<td>
<p>The taxonomic level you want to go down to. See examples below.
The taxonomic level IS case sensitive, and you do have to spell it
correctly. See <code>data(rank_ref)</code> for spelling.</p>
</td></tr>
<tr><td><code id="gbif_downstream_+3A_intermediate">intermediate</code></td>
<td>
<p>(logical) If TRUE, return a list of length two with
target taxon rank names, with additional list of data.frame's of
intermediate taxonomic groups. Default: FALSE</p>
</td></tr>
<tr><td><code id="gbif_downstream_+3A_limit">limit</code></td>
<td>
<p>Number of records to return. default: 100. max: 1000. use in
combination with the <code>start</code> parameter</p>
</td></tr>
<tr><td><code id="gbif_downstream_+3A_start">start</code></td>
<td>
<p>Record number to start at. default: 0. use in combination
with the <code>limit</code> parameter</p>
</td></tr>
<tr><td><code id="gbif_downstream_+3A_key">key</code></td>
<td>
<p>Deprecated, see <code>id</code></p>
</td></tr>
<tr><td><code id="gbif_downstream_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="#topic+gbif_name_usage">gbif_name_usage()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sometimes records don't have a <code>canonicalName</code> entry which is
what we look for. In that case we grab the <code>scientificName</code> entry.
You can see the type of name colleceted in the column <code>name_type</code>
</p>


<h3>Value</h3>

<p>data.frame of taxonomic information downstream to family from e.g.,
Order, Class, etc., or if <code>intermediated=TRUE</code>, list of length two,
with target taxon rank names, and intermediate names.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## the plant class Bangiophyceae
gbif_downstream(id = 198, downto="genus")
gbif_downstream(id = 198, downto="genus", intermediate=TRUE)

# families downstream from the family Strepsiptera (twisted wing parasites)
gbif_downstream(id = 1227, "family")
## here, intermediate leads to the same result as the target
gbif_downstream(id = 1227, "family", intermediate=TRUE)

if (interactive()) {
# Lepidoptera
gbif_downstream(id = 797, "family")

# get species downstream from the genus Ursus
gbif_downstream(id = 2433406, "species")

# get tribes down from the family Apidae
gbif_downstream(id = 7799978, downto="species")
gbif_downstream(id = 7799978, downto="species", intermediate=TRUE)

# names that don't have canonicalname entries for some results
# Myosotis: key 2925668
key &lt;- 2925668
res &lt;- gbif_downstream(key, downto = "species")
res2 &lt;- downstream(key, db = "gbif", downto = "species")

# Pagination
# GBIF limits queries to a maximum of 1000 records per request, so if
# there's more than 1000, use the start parameter
# Piper, taxonKey = 3075433
x1 &lt;- gbif_downstream(id = 3075433, downto = "species", limit=1000)
x2 &lt;- gbif_downstream(id = 3075433, downto = "species", limit=1000,
  start=1000)
x3 &lt;- gbif_downstream(id = 3075433, downto = "species", limit=1000,
  start=2000)
x4 &lt;- gbif_downstream(id = 3075433, downto = "species", limit=1000,
  start=3000)
rbind(x1, x2, x3, x4)
}

## End(Not run)
</code></pre>

<hr>
<h2 id='gbif_name_usage'>Lookup details for specific names in all taxonomies in GBIF.</h2><span id='topic+gbif_name_usage'></span>

<h3>Description</h3>

<p>This is a taxize version of the same function in the <code>rgbif</code> package so as to not have to
import rgbif and thus require GDAL binary installation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gbif_name_usage(
  key = NULL,
  name = NULL,
  data = "all",
  language = NULL,
  datasetKey = NULL,
  uuid = NULL,
  sourceId = NULL,
  rank = NULL,
  shortname = NULL,
  start = NULL,
  limit = 20,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gbif_name_usage_+3A_key">key</code></td>
<td>
<p>(numeric) A GBIF key for a taxon</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_name">name</code></td>
<td>
<p>(character) Filters by a case insensitive, canonical namestring,
e.g. 'Puma concolor'</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_data">data</code></td>
<td>
<p>(character) Specify an option to select what data is returned. See Description
below.</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_language">language</code></td>
<td>
<p>(character) Language, default is english</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_datasetkey">datasetKey</code></td>
<td>
<p>(character) Filters by the dataset's key (a uuid)</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_uuid">uuid</code></td>
<td>
<p>(character) A uuid for a dataset. Should give exact same results as datasetKey.</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_sourceid">sourceId</code></td>
<td>
<p>(numeric) Filters by the source identifier. Not used right now.</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_rank">rank</code></td>
<td>
<p>(character) Taxonomic rank. Filters by taxonomic rank as one of:
CLASS, CULTIVAR, CULTIVAR_GROUP, DOMAIN, FAMILY, FORM, GENUS, INFORMAL,
INFRAGENERIC_NAME, INFRAORDER, INFRASPECIFIC_NAME, INFRASUBSPECIFIC_NAME,
KINGDOM, ORDER, PHYLUM, SECTION, SERIES, SPECIES, STRAIN, SUBCLASS, SUBFAMILY,
SUBFORM, SUBGENUS, SUBKINGDOM, SUBORDER, SUBPHYLUM, SUBSECTION, SUBSERIES,
SUBSPECIES, SUBTRIBE, SUBVARIETY, SUPERCLASS, SUPERFAMILY, SUPERORDER,
SUPERPHYLUM, SUPRAGENERIC_NAME, TRIBE, UNRANKED, VARIETY</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_shortname">shortname</code></td>
<td>
<p>(character) A short name..need more info on this?</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_start">start</code></td>
<td>
<p>Record number to start at</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_limit">limit</code></td>
<td>
<p>Number of records to return</p>
</td></tr>
<tr><td><code id="gbif_name_usage_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length two. The first element is metadata. The second is
either a data.frame (verbose=FALSE, default) or a list (verbose=TRUE)
</p>


<h3>References</h3>

<p>https://www.gbif.org/developer/summary
</p>

<hr>
<h2 id='gbif_parse'>Parse taxon names using the GBIF name parser.</h2><span id='topic+gbif_parse'></span>

<h3>Description</h3>

<p>Parse taxon names using the GBIF name parser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gbif_parse(scientificname, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gbif_parse_+3A_scientificname">scientificname</code></td>
<td>
<p>(character) scientific names</p>
</td></tr>
<tr><td><code id="gbif_parse_+3A_...">...</code></td>
<td>
<p>Further args passed on to <a href="crul.html#topic+verb-POST">crul::verb-POST</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> containing fields extracted from parsed
taxon names. Fields returned are the union of fields extracted from
all species names in <code>scientificname</code>.
</p>


<h3>Author(s)</h3>

<p>John Baumgartner <a href="mailto:johnbb@student.unimelb.edu.au">johnbb@student.unimelb.edu.au</a>
</p>


<h3>References</h3>

<p>https://www.gbif.org/tools/name-parser/about
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gni_parse">gni_parse()</a></code>, <code><a href="#topic+gna_parse">gna_parse()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gbif_parse(scientificname='x Agropogon littoralis')
gbif_parse(c('Arrhenatherum elatius var. elatius',
             'Secale cereale subsp. cereale', 'Secale cereale ssp. cereale',
             'Vanessa atalanta (Linnaeus, 1758)'))

## End(Not run)
</code></pre>

<hr>
<h2 id='genbank2uid'>Get NCBI taxonomy UID from GenBankID</h2><span id='topic+genbank2uid'></span>

<h3>Description</h3>

<p>Get NCBI taxonomy UID from GenBankID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genbank2uid(id, batch_size = 100, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genbank2uid_+3A_id">id</code></td>
<td>
<p>A GenBank accession alphanumeric string, or a gi numeric string.</p>
</td></tr>
<tr><td><code id="genbank2uid_+3A_batch_size">batch_size</code></td>
<td>
<p>The number of queries to submit at a time.</p>
</td></tr>
<tr><td><code id="genbank2uid_+3A_key">key</code></td>
<td>
<p>(character) NCBI Entrez API key. optional. See Details.</p>
</td></tr>
<tr><td><code id="genbank2uid_+3A_...">...</code></td>
<td>
<p>Curl args passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>See https://www.ncbi.nlm.nih.gov/Sitemap/sequenceIDs.html for
help on why there are two identifiers, and the difference between them.
</p>


<h3>Value</h3>

<p>one or more NCBI taxonomic IDs
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication. We
recommend getting an API key.
</p>


<h3>HTTP version</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>Rate limits</h3>

<p>In case you run into errors due to your rate limit being exceeded, see
<code><a href="#topic+taxize_options">taxize_options()</a></code>, where you can set <code>ncbi_sleep</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# with accession numbers
genbank2uid(id = 'AJ748748')
genbank2uid(id = 'Y13155')
genbank2uid(id = 'X78312')
genbank2uid(id = 'KM495596')

# with gi numbers
genbank2uid(id = 62689767)
genbank2uid(id = 22775511)
genbank2uid(id = 156446673)

# pass in many accession or gi numbers
genbank2uid(c(62689767,156446673))
genbank2uid(c('X78312','KM495596'))
genbank2uid(list('X78312',156446673))

# curl options
res &lt;- genbank2uid(id = 156446673, verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_boldid'>Get the BOLD (Barcode of Life) code for a search term.</h2><span id='topic+get_boldid'></span><span id='topic+as.boldid'></span><span id='topic+as.boldid.boldid'></span><span id='topic+as.boldid.character'></span><span id='topic+as.boldid.list'></span><span id='topic+as.boldid.numeric'></span><span id='topic+as.boldid.data.frame'></span><span id='topic+as.data.frame.boldid'></span><span id='topic+get_boldid_'></span>

<h3>Description</h3>

<p>Get the BOLD (Barcode of Life) code for a search term.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_boldid(
  sci,
  fuzzy = FALSE,
  dataTypes = "basic",
  includeTree = FALSE,
  ask = TRUE,
  messages = TRUE,
  rows = NA,
  rank = NULL,
  division = NULL,
  parent = NULL,
  searchterm = NULL,
  ...
)

as.boldid(x, check = TRUE)

## S3 method for class 'boldid'
as.boldid(x, check = TRUE)

## S3 method for class 'character'
as.boldid(x, check = TRUE)

## S3 method for class 'list'
as.boldid(x, check = TRUE)

## S3 method for class 'numeric'
as.boldid(x, check = TRUE)

## S3 method for class 'data.frame'
as.boldid(x, check = TRUE)

## S3 method for class 'boldid'
as.data.frame(x, ...)

get_boldid_(
  sci,
  messages = TRUE,
  fuzzy = FALSE,
  dataTypes = "basic",
  includeTree = FALSE,
  rows = NA,
  searchterm = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_boldid_+3A_sci">sci</code></td>
<td>
<p>character; A vector of scientific names. Or,
a <code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_fuzzy">fuzzy</code></td>
<td>
<p>(logical) Whether to use fuzzy search or not (default: FALSE).</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_datatypes">dataTypes</code></td>
<td>
<p>(character) Specifies the datatypes that will be returned.
See <code><a href="#topic+bold_search">bold_search()</a></code> for options.</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_includetree">includeTree</code></td>
<td>
<p>(logical) If TRUE (default: FALSE), returns a list
containing information for parent taxa as well as the specified taxon.</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_ask">ask</code></td>
<td>
<p>logical; should get_tsn be run in interactive mode?
If TRUE and more than one TSN is found for teh species, the user is asked for
input. If FALSE NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_messages">messages</code></td>
<td>
<p>logical; should progress be printed?</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all rows are
considered. Note that this function still only gives back a boldid class object with one
to many identifiers. See <code><a href="#topic+get_boldid_">get_boldid_()</a></code> to get back all, or a subset,
of the raw data that you are presented during the ask process.</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_rank">rank</code></td>
<td>
<p>(character) A taxonomic rank name. See <a href="#topic+rank_ref">rank_ref</a> for possible
options. Though note that some data sources use atypical ranks, so inspect the
data itself for options. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_division">division</code></td>
<td>
<p>(character) A division (aka phylum) name. Optional. See <code>Filtering</code>
below.</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_parent">parent</code></td>
<td>
<p>(character) A parent name (i.e., the parent of the target search
taxon). Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_boldid_+3A_searchterm">searchterm</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="get_boldid_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <code><a href="crul.html#topic+verb-GET">crul::verb-GET</a></code></p>
</td></tr>
<tr><td><code id="get_boldid_+3A_x">x</code></td>
<td>
<p>Input to <code><a href="#topic+as.boldid">as.boldid()</a></code></p>
</td></tr>
<tr><td><code id="get_boldid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only used in
<code><a href="#topic+as.boldid">as.boldid()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>Filtering</h3>

<p>The parameters <code>division</code>, <code>parent</code>, and <code>rank</code> are not used in the search to
the data provider, but are used in filtering the data down to a subset that
is closer to the target you want.  For all these parameters, you can use
regex strings since we use <code><a href="base.html#topic+grep">grep()</a></code> internally to match. Filtering narrows
down to the set that matches your query, and removes the rest.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_boldid(sci = "Agapostemon")
get_boldid(sci = "Chironomus riparius")
get_boldid(c("Chironomus riparius","Quercus douglasii"))
splist &lt;- names_list('species')
get_boldid(splist, messages=FALSE)

# Fuzzy searching
get_boldid(sci="Osmi", fuzzy=TRUE)

# Get back a subset
get_boldid(sci="Osmi", fuzzy=TRUE, rows = 1)
get_boldid(sci="Osmi", fuzzy=TRUE, rows = 1:10)
get_boldid(sci=c("Osmi","Aga"), fuzzy=TRUE, rows = 1)
get_boldid(sci=c("Osmi","Aga"), fuzzy=TRUE, rows = 1:3)

# found
get_boldid('Epicordulia princeps')
get_boldid('Arigomphus furcifer')

# When not found
get_boldid("howdy")
get_boldid(c("Chironomus riparius", "howdy"))
get_boldid("Cordulegaster erronea")
get_boldid("Nasiaeshna pentacantha")

# Narrow down results to a division or rank, or both
## Satyrium example
### Results w/o narrowing
get_boldid("Satyrium")
### w/ phylum
get_boldid("Satyrium", division = "Plantae")
get_boldid("Satyrium", division = "Animalia")

## Rank example
get_boldid("Osmia", fuzzy = TRUE)
get_boldid("Osmia", fuzzy = TRUE, rank = "genus")

# Fuzzy filter on any filtering fields
## uses grep on the inside
get_boldid("Satyrium", division = "anim")
get_boldid("Aga", fuzzy = TRUE, parent = "*idae")

# Convert a boldid without class information to a boldid class
as.boldid(get_boldid("Agapostemon")) # already a boldid, returns the same
as.boldid(get_boldid(c("Agapostemon","Quercus douglasii"))) # same
as.boldid(1973) # numeric
as.boldid(c(1973,101009,98597)) # numeric vector, length &gt; 1
as.boldid("1973") # character
as.boldid(c("1973","101009","98597")) # character vector, length &gt; 1
as.boldid(list("1973","101009","98597")) # list, either numeric or character
## dont check, much faster
as.boldid("1973", check=FALSE)
as.boldid(1973, check=FALSE)
as.boldid(c("1973","101009","98597"), check=FALSE)
as.boldid(list("1973","101009","98597"), check=FALSE)

(out &lt;- as.boldid(c(1973,101009,98597)))
data.frame(out)
as.boldid( data.frame(out) )

# Get all data back
get_boldid_("Osmia", fuzzy=TRUE, rows=1:5)
get_boldid_("Osmia", fuzzy=TRUE, rows=1)
get_boldid_(c("Osmi","Aga"), fuzzy=TRUE, rows = 1:3)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_eolid'>Get the EOL ID from Encyclopedia of Life from taxonomic names.</h2><span id='topic+get_eolid'></span><span id='topic+as.eolid'></span><span id='topic+as.eolid.eolid'></span><span id='topic+as.eolid.character'></span><span id='topic+as.eolid.list'></span><span id='topic+as.eolid.numeric'></span><span id='topic+as.eolid.data.frame'></span><span id='topic+as.data.frame.eolid'></span><span id='topic+get_eolid_'></span>

<h3>Description</h3>

<p>Note that EOL doesn't expose an API endpoint for directly querying for EOL
taxon ID's, so we first use the function <code><a href="#topic+eol_search">eol_search()</a></code>
to find pages that deal with the species of interest, then use
<code><a href="#topic+eol_pages">eol_pages()</a></code> to find the actual taxon IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_eolid(
  sci_com,
  ask = TRUE,
  messages = TRUE,
  rows = NA,
  rank = NULL,
  data_source = NULL,
  sciname = NULL,
  ...
)

as.eolid(x, check = TRUE)

## S3 method for class 'eolid'
as.eolid(x, check = TRUE)

## S3 method for class 'character'
as.eolid(x, check = TRUE)

## S3 method for class 'list'
as.eolid(x, check = TRUE)

## S3 method for class 'numeric'
as.eolid(x, check = TRUE)

## S3 method for class 'data.frame'
as.eolid(x, check = TRUE)

## S3 method for class 'eolid'
as.data.frame(x, ...)

get_eolid_(sci_com, messages = TRUE, rows = NA, sciname = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_eolid_+3A_sci_com">sci_com</code></td>
<td>
<p>character; one or more scientific or common names. Or,
a <code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_eolid_+3A_ask">ask</code></td>
<td>
<p>logical; should get_eolid be run in interactive mode?
If TRUE and more than one ID is found for the species, the user is asked for
input. If FALSE NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_eolid_+3A_messages">messages</code></td>
<td>
<p>logical; If <code>TRUE</code> the actual taxon queried is printed
on the console.</p>
</td></tr>
<tr><td><code id="get_eolid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this function still only gives back a eolid
class object with one to many identifiers. See
<code><a href="#topic+get_eolid_">get_eolid_()</a></code> to get back all, or a subset, of the raw
data that you are presented during the ask process.</p>
</td></tr>
<tr><td><code id="get_eolid_+3A_rank">rank</code></td>
<td>
<p>(character) A taxonomic rank name. See <a href="#topic+rank_ref">rank_ref</a>
for possible options. Though note that some data sources use atypical ranks,
so inspect the data itself for options. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_eolid_+3A_data_source">data_source</code></td>
<td>
<p>(character) A data source inside of EOL. These are
longish names like e.g., &quot;Barcode of Life Data Systems&quot; or
&quot;USDA PLANTS images&quot;. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_eolid_+3A_sciname">sciname</code></td>
<td>
<p>Deprecated, see <code>sci_com</code></p>
</td></tr>
<tr><td><code id="get_eolid_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="#topic+eol_search">eol_search()</a></code></p>
</td></tr>
<tr><td><code id="get_eolid_+3A_x">x</code></td>
<td>
<p>Input to <code><a href="#topic+as.eolid">as.eolid()</a></code></p>
</td></tr>
<tr><td><code id="get_eolid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.eolid">as.eolid()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>EOL is a bit odd in that they have page IDs for each taxon, but
then within that, they have taxon ids for various taxa within that page
(e.g., GBIF and NCBI each have a taxon they refer to within the
page [i.e., taxon]). And we need the taxon ids from a particular data
provider (e.g, NCBI) to do other things, like get a higher classification
tree. However, humans want the page id, not the taxon id. So, the
id returned from this function is the taxon id, not the page id. You can
get the page id for a taxon by using <code><a href="#topic+eol_search">eol_search()</a></code> and
'<code><a href="#topic+eol_pages">eol_pages()</a></code>, and the URI returned in the attributes for a
taxon will lead you to the taxon page, and the ID in the URL is the
page id.
</p>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>Filtering</h3>

<p>The parameters <code>rank</code> and <code>data_source</code> are not used in the
search to the data provider, but are used in filtering the data down to a
subset that is closer to the target you want.  For all these parameters,
you can use regex strings since we use <code><a href="base.html#topic+grep">grep()</a></code> internally to
match. Filtering narrows down to the set that matches your query, and
removes the rest.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_eolid(sci_com='Pinus contorta')
get_eolid(sci_com='Puma concolor')

get_eolid(c("Puma concolor", "Pinus contorta"))

# specify rows to limit choices available
get_eolid('Poa annua')
get_eolid('Poa annua', rows=1)
get_eolid('Poa annua', rows=2)
get_eolid('Poa annua', rows=1:2)

# When not found
get_eolid(sci_com="uaudnadndj")
get_eolid(c("Chironomus riparius", "uaudnadndj"))

# filter results to a rank or data source, or both
get_eolid("Satyrium")
get_eolid("Satyrium", rank = "genus")
get_eolid("Satyrium", data_source = "INAT")
get_eolid("Satyrium", rank = "genus",
  data_source = "North Pacific Species List")

# Convert a eolid without class information to a eolid class
# already a eolid, returns the same
as.eolid(get_eolid("Chironomus riparius"))
# same
as.eolid(get_eolid(c("Chironomus riparius","Pinus contorta")))
# numeric
as.eolid(10247706)
# numeric vector, length &gt; 1
as.eolid(c(6985636,12188704,10247706))
# character
as.eolid("6985636")
# character vector, length &gt; 1
as.eolid(c("6985636","12188704","10247706"))
# list, either numeric or character
as.eolid(list("6985636","12188704","10247706"))
## dont check, much faster
as.eolid("6985636", check=FALSE)
as.eolid(6985636, check=FALSE)
as.eolid(c("6985636","12188704","10247706"), check=FALSE)
as.eolid(list("6985636","12188704","10247706"), check=FALSE)

(out &lt;- as.eolid(c(6985636,12188704,10247706)))
data.frame(out)
as.eolid( data.frame(out) )

# Get all data back
get_eolid_("Poa annua")
get_eolid_("Poa annua", rows=2)
get_eolid_("Poa annua", rows=1:2)
get_eolid_(c("asdfadfasd", "Pinus contorta"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_gbifid'>Get the GBIF backbone taxon ID from taxonomic names.</h2><span id='topic+get_gbifid'></span><span id='topic+as.gbifid'></span><span id='topic+as.gbifid.gbifid'></span><span id='topic+as.gbifid.character'></span><span id='topic+as.gbifid.list'></span><span id='topic+as.gbifid.numeric'></span><span id='topic+as.gbifid.data.frame'></span><span id='topic+as.data.frame.gbifid'></span><span id='topic+get_gbifid_'></span>

<h3>Description</h3>

<p>Get the GBIF backbone taxon ID from taxonomic names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gbifid(
  sci,
  ask = TRUE,
  messages = TRUE,
  rows = NA,
  phylum = NULL,
  class = NULL,
  order = NULL,
  family = NULL,
  rank = NULL,
  method = "backbone",
  sciname = NULL,
  ...
)

as.gbifid(x, check = FALSE)

## S3 method for class 'gbifid'
as.gbifid(x, check = FALSE)

## S3 method for class 'character'
as.gbifid(x, check = TRUE)

## S3 method for class 'list'
as.gbifid(x, check = TRUE)

## S3 method for class 'numeric'
as.gbifid(x, check = TRUE)

## S3 method for class 'data.frame'
as.gbifid(x, check = TRUE)

## S3 method for class 'gbifid'
as.data.frame(x, ...)

get_gbifid_(
  sci,
  messages = TRUE,
  rows = NA,
  method = "backbone",
  sciname = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gbifid_+3A_sci">sci</code></td>
<td>
<p>(character) one or more scientific names. Or, a <code>taxon_state</code>
object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_ask">ask</code></td>
<td>
<p>logical; should get_gbifid be run in interactive mode?
If TRUE and more than one ID is found for the species, the user is asked for
input. If FALSE NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_messages">messages</code></td>
<td>
<p>logical; If TRUE the actual taxon queried is printed on the console.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all rows are considered.
Note that this function still only gives back a gbifid class object with one to many identifiers.
See <code><a href="#topic+get_gbifid_">get_gbifid_()</a></code> to get back all, or a subset, of the raw data that you are
presented during the ask process.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_phylum">phylum</code></td>
<td>
<p>(character) A phylum (aka division) name. Optional. See <code>Filtering</code>
below.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_class">class</code></td>
<td>
<p>(character) A class name. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_order">order</code></td>
<td>
<p>(character) An order name. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_family">family</code></td>
<td>
<p>(character) A family name. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_rank">rank</code></td>
<td>
<p>(character) A taxonomic rank name. See <a href="#topic+rank_ref">rank_ref</a> for possible
options. Though note that some data sources use atypical ranks, so inspect the
data itself for options. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_method">method</code></td>
<td>
<p>(character) one of &quot;backbone&quot; or &quot;lookup&quot;. See Details.</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_sciname">sciname</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_x">x</code></td>
<td>
<p>Input to <code><a href="#topic+as.gbifid">as.gbifid()</a></code></p>
</td></tr>
<tr><td><code id="get_gbifid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only used in
<code><a href="#topic+as.gbifid">as.gbifid()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally in this function we use a function to search GBIF's taxonomy,
and if we find an exact match we return the ID for that match. If there isn't an
exact match we return the options to you to pick from.
</p>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>method parameter</h3>

<p>&quot;backbone&quot; uses the <code style="white-space: pre;">&#8288;/species/match&#8288;</code> GBIF API route, matching against their
backbone taxonomy. We turn on fuzzy matching by default, as the search without
fuzzy against backbone is quite narrow. &quot;lookup&quot; uses the <code style="white-space: pre;">&#8288;/species/search&#8288;</code>
GBIF API route, doing a full text search of name usages covering scientific
and vernacular named, species descriptions, distributions and the entire
classification.
</p>


<h3>Filtering</h3>

<p>The parameters <code>phylum</code>, <code>class</code>, <code>order</code>, <code>family</code>, and <code>rank</code> are not used
in the search to the data provider, but are used in filtering the data down
to a subset that is closer to the target you want.  For all these parameters,
you can use regex strings since we use <code><a href="base.html#topic+grep">grep()</a></code> internally to match.
Filtering narrows down to the set that matches your query, and removes the rest.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain,
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_gbifid(sci='Poa annua')
get_gbifid(sci='Pinus contorta')
get_gbifid(sci='Puma concolor')

#lots of queries
spp &lt;- names_list("species", 10)
res &lt;- get_gbifid(spp)
res
xx &lt;- taxon_last()
xx

# multiple names
get_gbifid(c("Poa annua", "Pinus contorta"))

# specify rows to limit choices available
get_gbifid(sci='Pinus')
get_gbifid(sci='Pinus', rows=10)
get_gbifid(sci='Pinus', rows=1:3)

# When not found, NA given
get_gbifid(sci="uaudnadndj")
get_gbifid(c("Chironomus riparius", "uaudnadndj"))

# Narrow down results to a division or rank, or both
## Satyrium example
### Results w/o narrowing
get_gbifid("Satyrium")
### w/ phylum
get_gbifid("Satyrium", phylum = "Tracheophyta")
get_gbifid("Satyrium", phylum = "Arthropoda")
### w/ phylum &amp; rank
get_gbifid("Satyrium", phylum = "Arthropoda", rank = "genus")

## Rank example
get_gbifid("Poa", method = "lookup")
get_gbifid("Poa", method = "lookup", rank = "genus")
get_gbifid("Poa", method = "lookup", family = "Thripidae")

# Fuzzy filter on any filtering fields
## uses grep on the inside
get_gbifid("Satyrium", phylum = "arthropoda")
get_gbifid("A*", method = "lookup", order = "*tera")
get_gbifid("A*", method = "lookup", order = "*ales")

# Convert a uid without class information to a uid class
as.gbifid(get_gbifid("Poa annua")) # already a uid, returns the same
as.gbifid(get_gbifid(c("Poa annua","Puma concolor"))) # same
as.gbifid(2704179) # numeric
as.gbifid(c(2704179,2435099,3171445)) # numeric vector, length &gt; 1
as.gbifid("2704179") # character
as.gbifid(c("2704179","2435099","3171445")) # character vector, length &gt; 1
as.gbifid(list("2704179","2435099","3171445")) # list, either numeric or character
## dont check, much faster
as.gbifid("2704179", check=FALSE)
as.gbifid(2704179, check=FALSE)
as.gbifid(2704179, check=FALSE)
as.gbifid(c("2704179","2435099","3171445"), check=FALSE)
as.gbifid(list("2704179","2435099","3171445"), check=FALSE)

(out &lt;- as.gbifid(c(2704179,2435099,3171445)))
data.frame(out)
as.uid( data.frame(out) )

# Get all data back
get_gbifid_("Puma concolor")
get_gbifid_(c("Pinus", "uaudnadndj"))
get_gbifid_(c("Pinus", "Puma"), rows=5)
get_gbifid_(c("Pinus", "Puma"), rows=1:5)

# use curl options
invisible(get_gbifid("Quercus douglasii", verbose = TRUE))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_genes'>Retrieve gene sequences from NCBI by accession number.</h2><span id='topic+get_genes'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genes(...)
</code></pre>

<hr>
<h2 id='get_genes_avail'>Retrieve gene sequences from NCBI by accession number.</h2><span id='topic+get_genes_avail'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genes_avail(...)
</code></pre>

<hr>
<h2 id='get_id_details'>Details on <code style="white-space: pre;">&#8288;get_*()&#8288;</code> functions</h2><span id='topic+get_id_details'></span>

<h3>Description</h3>

<p>Including outputs from <code style="white-space: pre;">&#8288;get_*()&#8288;</code> functions, as well as their
attributes, and all exception behaviors.
</p>


<h3>Details</h3>

<p>This document applies to the following functions:
</p>

<ul>
<li> <p><code><a href="#topic+get_boldid">get_boldid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_eolid">get_eolid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_gbifid">get_gbifid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_ids">get_ids()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_iucn">get_iucn()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_natservid">get_natservid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_nbnid">get_nbnid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_tolid">get_tolid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_tpsid">get_tpsid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_tsn">get_tsn()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_ubioid">get_ubioid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_uid">get_uid()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_wiki">get_wiki()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+get_wormsid">get_wormsid()</a></code>
</p>
</li></ul>



<h3>attributes</h3>

<p>Each output from <code style="white-space: pre;">&#8288;get_*()&#8288;</code> functions have the following attributes:
</p>

<ul>
<li> <p><em>match</em> (character) - the reason for NA, either 'not found', 'found' or
if <code>ask = FALSE</code> then 'NA due to ask=FALSE')
</p>
</li>
<li> <p><em>multiple_matches</em> (logical) - Whether multiple matches were returned by
the data source.
This can be <code>TRUE</code>, even if you get 1 name back because we try to pattern
match the name to see if there's any direct matches. So sometimes this
attribute is <code>TRUE</code>, as well as <code>pattern_match</code>, which then returns 1
resulting name without user prompt.
</p>
</li>
<li> <p><em>pattern_match</em> (logical) - Whether a pattern match was made.
If <code>TRUE</code> then<code>multiple_matches</code> must be <code>TRUE</code>, and we found a perfect match
to your name, ignoring case. If <code>FALSE</code>, there wasn't a direct match, and
likely you need to pick from many choices or further parameters can be used
to limit results
</p>
</li>
<li> <p><em>uri</em> (character) - The URI where more information can be read on the taxon
</p>
</li></ul>


<ul>
<li><p> includes the taxonomic identifier in the URL somewhere. This may be missing
if the value returned is <code>NA</code>
</p>
</li></ul>



<h3>exceptions</h3>

<p>The following are the various ways in which <code style="white-space: pre;">&#8288;get_*()&#8288;</code> functions behave:
</p>

<ul>
<li><p> success - the value returned is a character string or numeric
</p>
</li>
<li><p> no matches found - you'll get an NA, refine your search or possible the
taxon searched for does not exist in the database you're using
</p>
</li>
<li><p> more than on match and <code>ask = FALSE</code> - if there's more than one matching
result, and you have set <code>ask = FALSE</code>, then we can't determine the single
match to return, so we give back <code>NA</code>. However, in this case we do set the
<code>match</code> attribute to say <code style="white-space: pre;">&#8288;NA due to ask=FALSE &amp; &gt; 1 result&#8288;</code> so it's very
clear what happened - and you can even programatically check this as well
</p>
</li>
<li><p> NA due to some other reason - some <code style="white-space: pre;">&#8288;get_*()&#8288;</code> functions have additional
parameters for filtering taxa. It's possible that even though there's
results (that is, <code>found</code> will say <code>TRUE</code>), you can get back an NA. This is
most likely if the parameter filters taxa after they are returned from the
data provider and the value passed to the parameter leads to no matches.
</p>
</li></ul>


<hr>
<h2 id='get_ids'>Retrieve taxonomic identifiers for a given taxon name.</h2><span id='topic+get_ids'></span><span id='topic+get_ids_'></span>

<h3>Description</h3>

<p>This is a convenience function to get identifiers across all data sources.
You can use other <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions to get identifiers from specific
sources if you like.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ids(
  sci_com,
  db = c("itis", "ncbi", "eol", "tropicos", "gbif", "nbn", "pow"),
  suppress = FALSE,
  names = NULL,
  ...
)

get_ids_(
  sci_com,
  db = get_ids_dbs,
  rows = NA,
  suppress = FALSE,
  names = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_ids_+3A_sci_com">sci_com</code></td>
<td>
<p>(character) Taxonomic name to query.</p>
</td></tr>
<tr><td><code id="get_ids_+3A_db">db</code></td>
<td>
<p>(character) database to query. One or more of <code>ncbi</code>, <code>itis</code>, <code>eol</code>,
<code>tropicos</code>, <code>gbif</code>, <code>nbn</code>, or <code>pow</code>. By default db is set to search
all data sources. Note that each taxonomic data source has their own
identifiers, so that if you give the wrong <code>db</code> value for the identifier you
could get a result, it will likely be wrong (not what you were expecting).
If using ncbi and/or tropicos we recommend getting API keys;
see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="get_ids_+3A_suppress">suppress</code></td>
<td>
<p>(logical) suppress <span class="pkg">cli</span> separators with the database
name being queried. default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="get_ids_+3A_names">names</code></td>
<td>
<p>Deprecated, see <code>sci_com</code></p>
</td></tr>
<tr><td><code id="get_ids_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+get_tsn">get_tsn()</a></code>, <code><a href="#topic+get_uid">get_uid()</a></code>,
<code><a href="#topic+get_eolid">get_eolid()</a></code>, <code><a href="#topic+get_tpsid">get_tpsid()</a></code>, <code><a href="#topic+get_gbifid">get_gbifid()</a></code>,
<code><a href="#topic+get_nbnid">get_nbnid()</a></code>.</p>
</td></tr>
<tr><td><code id="get_ids_+3A_rows">rows</code></td>
<td>
<p>(numeric) Any number from 1 to infinity. If the default NA, all
rows are returned. When used in <code>get_ids</code> this function still only
gives back a ids class object with one to many identifiers. See
<code>get_ids_</code> to get back all, or a subset, of the raw data that you
are presented during the ask process.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers, each retaining their respective
S3 classes so that each element can be passed on to another function
(see e.g.'s).
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication
</p>


<h3>Note</h3>

<p>There is a timeout of 1/3 seconds between queries to NCBI.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Plug in taxon names directly
# specify rows to limit choices available
get_ids("Poa annua", db="eol", rows=1)
get_ids("Poa annua", db="eol", rows=1:2)

## Or you can specify which source you want via the db parameter
get_ids("Chironomus riparius", db = 'ncbi')
get_ids("Salvelinus fontinalis", db = 'nbn')

get_ids(c("Chironomus riparius", "Pinus contorta"), db = 'ncbi')
get_ids("Pinus contorta", db = c('ncbi','eol','tropicos'))
get_ids("ava avvva", db = c('ncbi','eol','tropicos'))

# Pass on to other functions
out &lt;- get_ids("Pinus contorta", db = c('ncbi','eol','tropicos'))
classification(out$ncbi)

# Get all data back
get_ids_(c("Chironomus riparius", "Pinus contorta"), db = 'nbn',
  rows=1:10)
get_ids_(c("Chironomus riparius", "Pinus contorta"), db = c('nbn','gbif'),
  rows=1:10)

# use curl options
get_ids("Agapostemon", db = "ncbi", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_iucn'>Get a IUCN Redlist taxon</h2><span id='topic+get_iucn'></span><span id='topic+as.iucn'></span><span id='topic+as.iucn.iucn'></span><span id='topic+as.iucn.character'></span><span id='topic+as.iucn.list'></span><span id='topic+as.iucn.numeric'></span><span id='topic+as.iucn.data.frame'></span><span id='topic+as.data.frame.iucn'></span>

<h3>Description</h3>

<p>Get a IUCN Redlist taxon
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_iucn(sci, messages = TRUE, key = NULL, x = NULL, ...)

as.iucn(x, check = TRUE, key = NULL)

## S3 method for class 'iucn'
as.iucn(x, check = TRUE, key = NULL)

## S3 method for class 'character'
as.iucn(x, check = TRUE, key = NULL)

## S3 method for class 'list'
as.iucn(x, check = TRUE, key = NULL)

## S3 method for class 'numeric'
as.iucn(x, check = TRUE, key = NULL)

## S3 method for class 'data.frame'
as.iucn(x, check = TRUE, key = NULL)

## S3 method for class 'iucn'
as.data.frame(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_iucn_+3A_sci">sci</code></td>
<td>
<p>(character) A vector of scientific names. Or, a
<code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_iucn_+3A_messages">messages</code></td>
<td>
<p>logical; should progress be printed?</p>
</td></tr>
<tr><td><code id="get_iucn_+3A_key">key</code></td>
<td>
<p>(character) required. you IUCN Redlist API key. See
<a href="rredlist.html#topic+rredlist-package">rredlist::rredlist-package</a> for help on authenticating with
IUCN Redlist</p>
</td></tr>
<tr><td><code id="get_iucn_+3A_x">x</code></td>
<td>
<p>For <code>get_iucn()</code>: Deprecated, see <code>sci</code>. For <code>as.iucn()</code>, various,
see examples</p>
</td></tr>
<tr><td><code id="get_iucn_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="get_iucn_+3A_check">check</code></td>
<td>
<p>(logical) Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.iucn">as.iucn()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>There is no underscore method, because there's no real
search for IUCN, that is, where you search for a string, and get back
a bunch of results due to fuzzy matching. If that exists in the future
we'll add an underscore method here.
</p>
<p>IUCN ids only work with <code><a href="#topic+synonyms">synonyms()</a></code> and <code><a href="#topic+sci2comm">sci2comm()</a></code>
methods.
</p>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
</p>
<p>Comes with the following attributes:
</p>

<ul>
<li> <p><em>match</em> (character) - the reason for NA, either 'not found',
'found' or if <code>ask = FALSE</code> then 'NA due to ask=FALSE')
</p>
</li>
<li> <p><em>name</em> (character) - the taxonomic name, which is needed in
<code><a href="#topic+synonyms">synonyms()</a></code> and <code><a href="#topic+sci2comm">sci2comm()</a></code> methods since they
internally use <span class="pkg">rredlist</span> functions which require the taxonomic name,
and not the taxonomic identifier
</p>
</li>
<li> <p><em>ri</em> (character) - The URI where more information can be
read on the taxon - includes the taxonomic identifier in the URL somewhere
</p>
</li></ul>

<p><em>multiple_matches</em> and <em>pattern_match</em> do not apply here as in other <code style="white-space: pre;">&#8288;get_*&#8288;</code>
methods since there is no IUCN Redlist search, so you either get a match or
you do not get a match.
</p>


<h3>See Also</h3>

<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_iucn("Branta canadensis")
get_iucn("Branta bernicla")
get_iucn("Panthera uncia")


## End(Not run)
</code></pre>

<hr>
<h2 id='get_iucn_data'>Get a IUCN Redlist taxon data</h2><span id='topic+get_iucn_data'></span>

<h3>Description</h3>

<p>Used to get IUCN data for other functions to use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_iucn_data(names_or_ids, messages = TRUE, key = NULL, latest = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_iucn_data_+3A_latest">latest</code></td>
<td>
<p>If <code>TRUE</code>, latest use <code><a href="rredlist.html#topic+rl_species_latest">rredlist::rl_species_latest()</a></code> instead of <code><a href="rredlist.html#topic+rl_species">rredlist::rl_species()</a></code></p>
</td></tr>
</table>

<hr>
<h2 id='get_natservid'>Get NatureServe taxonomic ID for a taxon name</h2><span id='topic+get_natservid'></span><span id='topic+as.natservid'></span><span id='topic+as.natservid.natservid'></span><span id='topic+as.natservid.character'></span><span id='topic+as.natservid.list'></span><span id='topic+as.natservid.numeric'></span><span id='topic+as.natservid.data.frame'></span><span id='topic+as.data.frame.natservid'></span><span id='topic+get_natservid_'></span>

<h3>Description</h3>

<p>Get NatureServe taxonomic ID for a taxon name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_natservid(
  sci_com,
  searchtype = "scientific",
  ask = TRUE,
  messages = TRUE,
  rows = NA,
  query = NULL,
  ...
)

as.natservid(x, check = TRUE)

## S3 method for class 'natservid'
as.natservid(x, check = TRUE)

## S3 method for class 'character'
as.natservid(x, check = TRUE)

## S3 method for class 'list'
as.natservid(x, check = TRUE)

## S3 method for class 'numeric'
as.natservid(x, check = TRUE)

## S3 method for class 'data.frame'
as.natservid(x, check = TRUE)

## S3 method for class 'natservid'
as.data.frame(x, ...)

get_natservid_(
  sci_com,
  searchtype = "scientific",
  messages = TRUE,
  rows = NA,
  query = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_natservid_+3A_sci_com">sci_com</code></td>
<td>
<p>character; A vector of common or scientific names. Or, a
<code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_natservid_+3A_searchtype">searchtype</code></td>
<td>
<p>character; One of 'scientific' (default) or 'common'.
This doesn't affect the query to NatureServe - but rather affects what
column of data is targeted in name filtering post data request.</p>
</td></tr>
<tr><td><code id="get_natservid_+3A_ask">ask</code></td>
<td>
<p>logical; should get_natservid be run in interactive mode?
If <code>TRUE</code> and more than one wormsid is found for the species, the
user is asked for input. If <code>FALSE</code> NA is returned for
multiple matches. default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="get_natservid_+3A_messages">messages</code></td>
<td>
<p>logical; should progress be printed? default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="get_natservid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NaN, all
rows are considered. Note that this function still only gives back a
natservid class object with one to many identifiers. See
<code>get_natservid_()</code> to get back all, or a subset, of the raw
data that you are presented during the ask process.</p>
</td></tr>
<tr><td><code id="get_natservid_+3A_query">query</code></td>
<td>
<p>Deprecated, see <code>sci_com</code></p>
</td></tr>
<tr><td><code id="get_natservid_+3A_...">...</code></td>
<td>
<p>curl options passed on to <a href="crul.html#topic+verb-POST">crul::verb-POST</a></p>
</td></tr>
<tr><td><code id="get_natservid_+3A_x">x</code></td>
<td>
<p>Input to <code>as.natservid</code></p>
</td></tr>
<tr><td><code id="get_natservid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.natservid">as.natservid()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>Note</h3>

<p>Authentication no longer required
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
(x &lt;- get_natservid("Helianthus annuus", verbose = TRUE))
attributes(x)
attr(x, "match")
attr(x, "multiple_matches")
attr(x, "pattern_match")
attr(x, "uri")

get_natservid('Gadus morhua')
get_natservid(c("Helianthus annuus", 'Gadus morhua'))

# specify rows to limit choices available
get_natservid('Ruby Quaker Moth', 'common')
get_natservid('Ruby*', 'common')
get_natservid('Ruby*', 'common', rows=1)
get_natservid('Ruby*', 'common', rows=1:2)

# When not found
get_natservid("howdy")
get_natservid(c('Gadus morhua', "howdy"))

# Convert a natservid without class information to a natservid class
# already a natservid, returns the same
as.natservid(get_natservid('Pomatomus saltatrix'))
# same
as.natservid(get_natservid(c('Gadus morhua', 'Pomatomus saltatrix')))
# character
as.natservid(101905)
# character vector, length &gt; 1
as.natservid(c(101905, 101998))
# list, either numeric or character
as.natservid(list(101905, 101998))
## dont check, much faster
as.natservid(101905, check = FALSE)
as.natservid(c(101905, 101998), check = FALSE)
as.natservid(list(101905, 101998), check = FALSE)

(out &lt;- as.natservid(c(101905, 101998), check = FALSE))
data.frame(out)
as.natservid( data.frame(out) )

# Get all data back
get_natservid_("Helianthus")
get_natservid_("Ruby*", searchtype = "common")
get_natservid_("Ruby*", searchtype = "common", rows=1:3)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_nbnid'>Get the UK National Biodiversity Network ID from taxonomic names.</h2><span id='topic+get_nbnid'></span><span id='topic+as.nbnid'></span><span id='topic+as.nbnid.nbnid'></span><span id='topic+as.nbnid.character'></span><span id='topic+as.nbnid.list'></span><span id='topic+as.nbnid.data.frame'></span><span id='topic+as.data.frame.nbnid'></span><span id='topic+get_nbnid_'></span>

<h3>Description</h3>

<p>Get the UK National Biodiversity Network ID from taxonomic names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_nbnid(
  sci_com,
  ask = TRUE,
  messages = TRUE,
  rec_only = FALSE,
  rank = NULL,
  rows = NA,
  name = NULL,
  ...
)

as.nbnid(x, check = TRUE)

## S3 method for class 'nbnid'
as.nbnid(x, check = TRUE)

## S3 method for class 'character'
as.nbnid(x, check = TRUE)

## S3 method for class 'list'
as.nbnid(x, check = TRUE)

## S3 method for class 'data.frame'
as.nbnid(x, check = TRUE)

## S3 method for class 'nbnid'
as.data.frame(x, ...)

get_nbnid_(
  sci_com,
  messages = TRUE,
  rec_only = FALSE,
  rank = NULL,
  rows = NA,
  name = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_nbnid_+3A_sci_com">sci_com</code></td>
<td>
<p>character; a vector of common or scientific names. Or, a
<code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_ask">ask</code></td>
<td>
<p>logical; should get_nbnid be run in interactive mode?
If TRUE and more than one ID is found for the species, the user is asked for
input. If FALSE NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_messages">messages</code></td>
<td>
<p>logical; If TRUE the actual taxon queried is printed on the
console.</p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_rec_only">rec_only</code></td>
<td>
<p>(logical) If <code>TRUE</code> ids of recommended names are
returned (i.e. synonyms are removed). Defaults to <code>FALSE</code>. Remember,
the id of a synonym is a taxa with 'recommended' name status.</p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_rank">rank</code></td>
<td>
<p>(character) If given, we attempt to limit the results to those
taxa with the matching rank.</p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this function still only gives back a nbnid
class object with one to many identifiers. See
<code><a href="#topic+get_nbnid_">get_nbnid_()</a></code> to get back all, or a subset, of the raw
data that you are presented during the ask process.</p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_name">name</code></td>
<td>
<p>Deprecated, see <code>sci_com</code></p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code>nbn_search</code></p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_x">x</code></td>
<td>
<p>Input to <code><a href="#topic+as.nbnid">as.nbnid()</a></code></p>
</td></tr>
<tr><td><code id="get_nbnid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.nbnid">as.nbnid()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>
<p>an object of class nbnid, a light wrapper around a character
string that is the taxonomic ID - includes attributes with relavant
metadata
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain,
</p>


<h3>References</h3>

<p>https://api.nbnatlas.org/
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>
<p>Other nbn: 
<code><a href="#topic+nbn_classification">nbn_classification</a>()</code>,
<code><a href="#topic+nbn_search">nbn_search</a>()</code>,
<code><a href="#topic+nbn_synonyms">nbn_synonyms</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_nbnid(sci_com='Poa annua')
get_nbnid(sci_com='Poa annua', rec_only=TRUE)
get_nbnid(sci_com='Poa annua', rank='Species')
get_nbnid(sci_com='Poa annua', rec_only=TRUE, rank='Species')
get_nbnid(sci_com='Pinus contorta')

# The NBN service handles common names too
get_nbnid(sci_com='red-winged blackbird')

# specify rows to limit choices available
get_nbnid('Poa ann')
get_nbnid('Poa ann', rows=1)
get_nbnid('Poa ann', rows=25)
get_nbnid('Poa ann', rows=1:2)

# When not found
get_nbnid(sci_com="uaudnadndj")
get_nbnid(c("Zootoca vivipara", "uaudnadndj"))
get_nbnid(c("Zootoca vivipara","Chironomus riparius", "uaudnadndj"))

# Convert an nbnid without class information to a nbnid class
as.nbnid(get_nbnid("Zootoca vivipara")) # already a nbnid, returns the same
as.nbnid(get_nbnid(c("Zootoca vivipara","Pinus contorta"))) # same
as.nbnid('NHMSYS0001706186') # character
# character vector, length &gt; 1
as.nbnid(c("NHMSYS0001706186","NHMSYS0000494848","NBNSYS0000010867"))
# list
as.nbnid(list("NHMSYS0001706186","NHMSYS0000494848","NBNSYS0000010867"))
## dont check, much faster
as.nbnid('NHMSYS0001706186', check=FALSE)
as.nbnid(list("NHMSYS0001706186","NHMSYS0000494848","NBNSYS0000010867"),
  check=FALSE)

(out &lt;- as.nbnid(c("NHMSYS0001706186","NHMSYS0000494848",
  "NBNSYS0000010867")))
data.frame(out)
as.nbnid( data.frame(out) )

# Get all data back
get_nbnid_("Zootoca vivipara")
get_nbnid_("Poa annua", rows=2)
get_nbnid_("Poa annua", rows=1:2)
get_nbnid_(c("asdfadfasd","Pinus contorta"), rows=1:5)

# use curl options
invisible(get_nbnid("Quercus douglasii", verbose = TRUE))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_pow'>Get Kew's Plants of the World code for a taxon</h2><span id='topic+get_pow'></span><span id='topic+as.pow'></span><span id='topic+as.pow.pow'></span><span id='topic+as.pow.character'></span><span id='topic+as.pow.list'></span><span id='topic+as.pow.data.frame'></span><span id='topic+as.data.frame.pow'></span><span id='topic+get_pow_'></span>

<h3>Description</h3>

<p>Get Kew's Plants of the World code for a taxon
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pow(
  sci_com,
  accepted = FALSE,
  ask = TRUE,
  messages = TRUE,
  rows = NA,
  family_filter = NULL,
  rank_filter = NULL,
  x = NULL,
  ...
)

as.pow(x, check = TRUE)

## S3 method for class 'pow'
as.pow(x, check = TRUE)

## S3 method for class 'character'
as.pow(x, check = TRUE)

## S3 method for class 'list'
as.pow(x, check = TRUE)

## S3 method for class 'data.frame'
as.pow(x, check = TRUE)

## S3 method for class 'pow'
as.data.frame(x, ...)

get_pow_(sci_com, messages = TRUE, rows = NA, x = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_pow_+3A_sci_com">sci_com</code></td>
<td>
<p>character; A vector of common or scientific names. Or, a
<code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_pow_+3A_accepted">accepted</code></td>
<td>
<p>logical; If TRUE, removes names that are not accepted
valid names by ITIS. Set to <code>FALSE</code> (default) to give back both
accepted and unaccepted names.</p>
</td></tr>
<tr><td><code id="get_pow_+3A_ask">ask</code></td>
<td>
<p>logical; should <code>get_pow</code> be run in interactive mode?
If TRUE and more than one pow is found for teh species, the user is
asked for input. If FALSE NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_pow_+3A_messages">messages</code></td>
<td>
<p>logical; should progress be printed?</p>
</td></tr>
<tr><td><code id="get_pow_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA,
all rows are considered. Note that this function still only gives back
a pow class object with one to many identifiers. See
<code><a href="#topic+get_pow_">get_pow_()</a></code> to get back all, or a subset,
of the raw data that you are presented during the ask process.</p>
</td></tr>
<tr><td><code id="get_pow_+3A_family_filter">family_filter</code></td>
<td>
<p>(character) A division (aka phylum) name to filter
data after retrieved from NCBI. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_pow_+3A_rank_filter">rank_filter</code></td>
<td>
<p>(character) A taxonomic rank name to filter data after
retrieved from NCBI. See <a href="#topic+rank_ref">rank_ref</a> for possible options.
Though note that some data sources use atypical ranks, so inspect the data
itself for options. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_pow_+3A_x">x</code></td>
<td>
<p>For <code>get_pow()</code>: deprecated, see <code>sci_com</code>. For <code>as.pow</code>, various,
see examples</p>
</td></tr>
<tr><td><code id="get_pow_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
<tr><td><code id="get_pow_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.pow">as.pow()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>Filtering</h3>

<p>The parameters <code>family_filter</code> an<code>rank_filter</code>er are not
used in the search to the data provider, but are used in filtering the data down to a
subset that is closer to the target you want.  For these two parameters,
you can use regex strings since we use <code><a href="base.html#topic+grep">grep()</a></code> internally to match.
Filtering narrows down to the set that matches your query, and removes the rest.
</p>


<h3>Rate-limits</h3>

<p>As of February 2019, KEW was limiting to 5 requests per second. Note that
they may change that number in the future.
</p>
<p>If you get errors that contain <code>429</code> you are hitting the rate limit, and you
can get around it by doing requests with <code>Sys.sleep</code> in between requests.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other pow: 
<code><a href="#topic+pow_lookup">pow_lookup</a>()</code>,
<code><a href="#topic+pow_search">pow_search</a>()</code>,
<code><a href="#topic+pow_synonyms">pow_synonyms</a>()</code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_pow(sci_com="Helianthus")
get_pow(c("Helianthus","Quercus douglasii"))

# Get back a subset
get_pow(sci_com="Helianthus", rows = 1)
get_pow(sci_com="Helianthus", rows = 1:10)

# When not found
get_pow("howdy")
get_pow(c("Helianthus annuus", "howdy"))

# Narrow down results 
# to accepted names
get_pow("Helianthus", accepted = TRUE)
# to a kingom 
get_pow("Helianthus", rank_filter = "genus")
# to accepted names and rank
get_pow("Helianthus annuus", accepted = TRUE, rank_filter = "species")
# to a family
get_pow("flower", family_filter = "Acanthaceae")

# Convert a pow without class information to a pow class
z &lt;- get_pow("Helianthus annuus", accepted = TRUE, rank_filter = "species")
# already a pow, returns the same
as.pow(z)
as.pow("urn:lsid:ipni.org:names:119003-2")
# character vector, length &gt; 1
ids &lt;- c("urn:lsid:ipni.org:names:119003-2","urn:lsid:ipni.org:names:328247-2")
as.pow(ids)
# list, with character strings
as.pow(as.list(ids)) 
## dont check, much faster
as.pow("urn:lsid:ipni.org:names:119003-2", check=FALSE)
as.pow(ids, check=FALSE)
as.pow(as.list(ids), check=FALSE)

(out &lt;- as.pow(ids))
data.frame(out)
as.pow( data.frame(out) )

# Get all data back
get_pow_("Quercus", rows=1:5)
get_pow_("Quercus", rows=1)
get_pow_(c("Pinus", "Abies"), rows = 1:3)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_seqs'>Retrieve gene sequences from NCBI by accession number.</h2><span id='topic+get_seqs'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_seqs(...)
</code></pre>

<hr>
<h2 id='get_tolid'>Get the OTT id for a search term</h2><span id='topic+get_tolid'></span><span id='topic+as.tolid'></span><span id='topic+as.tolid.tolid'></span><span id='topic+as.tolid.character'></span><span id='topic+as.tolid.list'></span><span id='topic+as.tolid.numeric'></span><span id='topic+as.tolid.data.frame'></span><span id='topic+as.data.frame.tolid'></span><span id='topic+get_tolid_'></span>

<h3>Description</h3>

<p>Retrieve the Open Tree of Life Taxonomy (OTT) id of a taxon from
OpenTreeOfLife
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tolid(sci, ask = TRUE, messages = TRUE, rows = NA, sciname = NULL, ...)

as.tolid(x, check = TRUE)

## S3 method for class 'tolid'
as.tolid(x, check = TRUE)

## S3 method for class 'character'
as.tolid(x, check = TRUE)

## S3 method for class 'list'
as.tolid(x, check = TRUE)

## S3 method for class 'numeric'
as.tolid(x, check = TRUE)

## S3 method for class 'data.frame'
as.tolid(x, check = TRUE)

## S3 method for class 'tolid'
as.data.frame(x, ...)

get_tolid_(sci, messages = TRUE, rows = NA, sciname = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_tolid_+3A_sci">sci</code></td>
<td>
<p>character; one or more scientific names. Or, a <code>taxon_state</code>
object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_tolid_+3A_ask">ask</code></td>
<td>
<p>logical; should <code>get_tolid</code> be run in interactive mode?
If <code>TRUE</code> and more than one TOL is found for the species, the user is
asked for input. If <code>FALSE</code> NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_tolid_+3A_messages">messages</code></td>
<td>
<p>logical; should progress be printed?</p>
</td></tr>
<tr><td><code id="get_tolid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this function still only gives back a tol
class object with one to many identifiers. See <code><a href="#topic+get_tolid_">get_tolid_()</a></code>
to get back all, or a subset, of the raw data that you are presented during
the ask process.</p>
</td></tr>
<tr><td><code id="get_tolid_+3A_sciname">sciname</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="get_tolid_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="get_tolid_+3A_x">x</code></td>
<td>
<p>Input to <code>as.tolid</code></p>
</td></tr>
<tr><td><code id="get_tolid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.tolid">as.tolid()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_tolid(sci = "Quercus douglasii")
get_tolid(sci = "Chironomus riparius")
get_tolid(c("Chironomus riparius","Quercus douglasii"))
splist &lt;- c("annona cherimola", 'annona muricata', "quercus robur",
		"shorea robusta", "pandanus patina", "oryza sativa", "durio zibethinus")
get_tolid(splist, messages=FALSE)

# specify rows to limit choices available
get_tolid('Arni')
get_tolid('Arni', rows=1)
get_tolid('Arni', rows=1:2)

# When not found
get_tolid("howdy")
get_tolid(c("Chironomus riparius", "howdy"))

# Convert a tol without class information to a tol class
as.tolid(get_tolid("Quercus douglasii")) # already a tol, returns the same
as.tolid(get_tolid(c("Chironomus riparius","Pinus contorta"))) # same
as.tolid(5907893) # numeric
as.tolid(c(3930798,515712,872577)) # numeric vector, length &gt; 1
as.tolid("3930798") # character
as.tolid(c("3930798","515712","872577")) # character vector, length &gt; 1
as.tolid(list("3930798","515712","872577")) # list, either numeric or character
## dont check, much faster
as.tolid("3930798", check=FALSE)
as.tolid(3930798, check=FALSE)
as.tolid(c("3930798","515712","872577"), check=FALSE)
as.tolid(list("3930798","515712","872577"), check=FALSE)

(out &lt;- as.tolid(c(3930798,515712,872577)))
data.frame(out)
as.tolid( data.frame(out) )

# Get all data back
get_tolid_("Arni")
get_tolid_("Arni", rows=1)
get_tolid_("Arni", rows=1:2)
get_tolid_(c("asdfadfasd","Pinus contorta"))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_tpsid'>Get the NameID codes from Tropicos for taxonomic names.</h2><span id='topic+get_tpsid'></span><span id='topic+as.tpsid'></span><span id='topic+as.tpsid.tpsid'></span><span id='topic+as.tpsid.character'></span><span id='topic+as.tpsid.list'></span><span id='topic+as.tpsid.numeric'></span><span id='topic+as.tpsid.data.frame'></span><span id='topic+as.data.frame.tpsid'></span><span id='topic+get_tpsid_'></span>

<h3>Description</h3>

<p>Get the NameID codes from Tropicos for taxonomic names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tpsid(
  sci,
  ask = TRUE,
  messages = TRUE,
  key = NULL,
  rows = NA,
  family = NULL,
  rank = NULL,
  sciname = NULL,
  ...
)

as.tpsid(x, check = TRUE)

## S3 method for class 'tpsid'
as.tpsid(x, check = TRUE)

## S3 method for class 'character'
as.tpsid(x, check = TRUE)

## S3 method for class 'list'
as.tpsid(x, check = TRUE)

## S3 method for class 'numeric'
as.tpsid(x, check = TRUE)

## S3 method for class 'data.frame'
as.tpsid(x, check = TRUE)

## S3 method for class 'tpsid'
as.data.frame(x, ...)

get_tpsid_(sci, messages = TRUE, key = NULL, rows = NA, sciname = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_tpsid_+3A_sci">sci</code></td>
<td>
<p>(character) One or more scientific name's as a vector or list. Or,
a <code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_ask">ask</code></td>
<td>
<p>logical; should get_tpsid be run in interactive mode?
If TRUE and more than one ID is found for the species, the user is asked for
input. If FALSE NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_messages">messages</code></td>
<td>
<p>logical; If TRUE the actual taxon queried is printed on the console.</p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_key">key</code></td>
<td>
<p>Your API key; see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all rows are considered.
Note that this function still only gives back a tpsid class object with one to many identifiers.
See <code><a href="#topic+get_tpsid_">get_tpsid_()</a></code> to get back all, or a subset, of the raw data that you are
presented during the ask process.</p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_family">family</code></td>
<td>
<p>(character) A family name. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_rank">rank</code></td>
<td>
<p>(character) A taxonomic rank name. See <a href="#topic+rank_ref">rank_ref</a> for possible
options. Though note that some data sources use atypical ranks, so inspect the
data itself for options. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_sciname">sciname</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+tp_search">tp_search()</a></code>.</p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_x">x</code></td>
<td>
<p>Input to <code><a href="#topic+as.tpsid">as.tpsid()</a></code></p>
</td></tr>
<tr><td><code id="get_tpsid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only used in
<code><a href="#topic+as.tpsid">as.tpsid()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>Filtering</h3>

<p>The parameters <code>family</code> an<code>rank</code>nk are not used in the search to the data
provider, but are used in filtering the data down to a subset that is closer to the
target you want.  For all these parameters,
you can use regex strings since we use <code><a href="base.html#topic+grep">grep()</a></code> internally to match.
Filtering narrows down to the set that matches your query, and removes the rest.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain,
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_tpsid(sci='Poa annua')
get_tpsid(sci='Pinus contorta')

get_tpsid(c("Poa annua", "Pinus contorta"))

# specify rows to limit choices available
get_tpsid('Poa ann')
get_tpsid('Poa ann', rows=1)
get_tpsid('Poa ann', rows=25)
get_tpsid('Poa ann', rows=1:2)

# When not found, NA given (howdy is not a species name, and Chrinomus is a fly)
get_tpsid("howdy")
get_tpsid(c("Chironomus riparius", "howdy"))

# Narrow down results to a division or rank, or both
## Satyrium example
### Results w/o narrowing
get_tpsid("Satyrium")
### w/ rank
get_tpsid("Satyrium", rank = "var.")
get_tpsid("Satyrium", rank = "sp.")

## w/ family
get_tpsid("Poa")
get_tpsid("Poa", family = "Iridaceae")
get_tpsid("Poa", family = "Orchidaceae")
get_tpsid("Poa", family = "Orchidaceae", rank = "gen.")

# Fuzzy filter on any filtering fields
## uses grep on the inside
get_tpsid("Poa", family = "orchidaceae")
get_tpsid("Aga", fuzzy = TRUE, parent = "*idae")

# pass to classification function to get a taxonomic hierarchy
classification(get_tpsid(sci='Poa annua'))

# Convert a tpsid without class information to a tpsid class
as.tpsid(get_tpsid("Pinus contorta")) # already a tpsid, returns the same
as.tpsid(get_tpsid(c("Chironomus riparius","Pinus contorta"))) # same
as.tpsid(24900183) # numeric
as.tpsid(c(24900183,50150089,50079838)) # numeric vector, length &gt; 1
as.tpsid("24900183") # character
as.tpsid(c("24900183","50150089","50079838")) # character vector, length &gt; 1
as.tpsid(list("24900183","50150089","50079838")) # list, either numeric or character
## dont check, much faster
as.tpsid("24900183", check=FALSE)
as.tpsid(24900183, check=FALSE)
as.tpsid(c("24900183","50150089","50079838"), check=FALSE)
as.tpsid(list("24900183","50150089","50079838"), check=FALSE)

(out &lt;- as.tpsid(c(24900183,50150089,50079838)))
data.frame(out)
as.tpsid( data.frame(out) )

# Get all data back
get_tpsid_("Poa annua")
get_tpsid_("Poa annua", rows=2)
get_tpsid_("Poa annua", rows=1:2)
get_tpsid_(c("asdfadfasd","Pinus contorta"), rows=1:5)

# use curl options
invisible(get_tpsid("Quercus douglasii", messages = TRUE))

## End(Not run)
</code></pre>

<hr>
<h2 id='get_tsn'>Get the TSN code for a search term.</h2><span id='topic+get_tsn'></span><span id='topic+as.tsn'></span><span id='topic+as.tsn.tsn'></span><span id='topic+as.tsn.character'></span><span id='topic+as.tsn.list'></span><span id='topic+as.tsn.numeric'></span><span id='topic+as.tsn.data.frame'></span><span id='topic+as.data.frame.tsn'></span><span id='topic+get_tsn_'></span>

<h3>Description</h3>

<p>Retrieve the taxonomic serial numbers (TSN) of a taxon from ITIS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tsn(
  sci_com,
  searchtype = "scientific",
  accepted = FALSE,
  ask = TRUE,
  messages = TRUE,
  rows = NA,
  searchterm = NULL,
  ...
)

as.tsn(x, check = TRUE)

## S3 method for class 'tsn'
as.tsn(x, check = TRUE)

## S3 method for class 'character'
as.tsn(x, check = TRUE)

## S3 method for class 'list'
as.tsn(x, check = TRUE)

## S3 method for class 'numeric'
as.tsn(x, check = TRUE)

## S3 method for class 'data.frame'
as.tsn(x, check = TRUE)

## S3 method for class 'tsn'
as.data.frame(x, ...)

get_tsn_(
  sci_com,
  messages = TRUE,
  searchtype = "scientific",
  accepted = TRUE,
  rows = NA,
  searchterm = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_tsn_+3A_sci_com">sci_com</code></td>
<td>
<p>character; A vector of common or scientific names.
Or, a <code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_tsn_+3A_searchtype">searchtype</code></td>
<td>
<p>character; One of 'scientific' or 'common', or any
unique abbreviation</p>
</td></tr>
<tr><td><code id="get_tsn_+3A_accepted">accepted</code></td>
<td>
<p>logical; If TRUE, removes names that are not accepted valid
names by ITIS. Set to <code>FALSE</code> (default) to give back both accepted
and unaccepted names.</p>
</td></tr>
<tr><td><code id="get_tsn_+3A_ask">ask</code></td>
<td>
<p>logical; should get_tsn be run in interactive mode?
If <code>TRUE</code> and more than one TSN is found for the species, the user is
asked for input. If <code>FALSE</code> NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_tsn_+3A_messages">messages</code></td>
<td>
<p>logical; should progress be printed?</p>
</td></tr>
<tr><td><code id="get_tsn_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this function still only gives back a tsn
class object with one to many identifiers. See
<code><a href="#topic+get_tsn_">get_tsn_()</a></code> to get back all, or a subset, of the raw
data that you are presented during the ask process.</p>
</td></tr>
<tr><td><code id="get_tsn_+3A_searchterm">searchterm</code></td>
<td>
<p>Deprecated, see <code>sci_com</code></p>
</td></tr>
<tr><td><code id="get_tsn_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="get_tsn_+3A_x">x</code></td>
<td>
<p>Input to as.tsn</p>
</td></tr>
<tr><td><code id="get_tsn_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.tsn">as.tsn()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_tsn("Quercus douglasii")
get_tsn("Chironomus riparius")
get_tsn(c("Chironomus riparius","Quercus douglasii"))
splist &lt;- c("annona cherimola", 'annona muricata', "quercus robur",
		"shorea robusta", "pandanus patina", "oryza sativa", "durio zibethinus")
get_tsn(splist, messages=FALSE)

# specify rows to limit choices available
get_tsn('Arni')
get_tsn('Arni', rows=1)
get_tsn('Arni', rows=1:2)

# When not found
get_tsn("howdy")
get_tsn(c("Chironomus riparius", "howdy"))

# Using common names
get_tsn("black bear", searchtype="common")

# Convert a tsn without class information to a tsn class
as.tsn(get_tsn("Quercus douglasii")) # already a tsn, returns the same
as.tsn(get_tsn(c("Chironomus riparius","Pinus contorta"))) # same
as.tsn(19322) # numeric
as.tsn(c(19322,129313,506198)) # numeric vector, length &gt; 1
as.tsn("19322") # character
as.tsn(c("19322","129313","506198")) # character vector, length &gt; 1
as.tsn(list("19322","129313","506198")) # list, either numeric or character
## dont check, much faster
as.tsn("19322", check=FALSE)
as.tsn(19322, check=FALSE)
as.tsn(c("19322","129313","506198"), check=FALSE)
as.tsn(list("19322","129313","506198"), check=FALSE)

(out &lt;- as.tsn(c(19322,129313,506198)))
data.frame(out)
as.tsn( data.frame(out) )

# Get all data back
get_tsn_("Arni")
get_tsn_("Arni", rows=1)
get_tsn_("Arni", rows=1:2)
get_tsn_(c("asdfadfasd","Pinus contorta"), rows=1:5)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_ubioid'>Get the uBio id for a search term</h2><span id='topic+get_ubioid'></span><span id='topic+as.ubioid'></span><span id='topic+get_ubioid_'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ubioid(...)

as.ubioid(...)

get_ubioid_(...)
</code></pre>

<hr>
<h2 id='get_uid'>Get the UID codes from NCBI for taxonomic names.</h2><span id='topic+get_uid'></span><span id='topic+as.uid'></span><span id='topic+as.uid.uid'></span><span id='topic+as.uid.character'></span><span id='topic+as.uid.list'></span><span id='topic+as.uid.numeric'></span><span id='topic+as.uid.data.frame'></span><span id='topic+as.data.frame.uid'></span><span id='topic+get_uid_'></span>

<h3>Description</h3>

<p>Retrieve the Unique Identifier (UID) of a taxon from NCBI taxonomy browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_uid(
  sci_com,
  ask = TRUE,
  messages = TRUE,
  rows = NA,
  modifier = NULL,
  rank_query = NULL,
  division_filter = NULL,
  rank_filter = NULL,
  key = NULL,
  sciname = NULL,
  ...
)

as.uid(x, check = TRUE)

## S3 method for class 'uid'
as.uid(x, check = TRUE)

## S3 method for class 'character'
as.uid(x, check = TRUE)

## S3 method for class 'list'
as.uid(x, check = TRUE)

## S3 method for class 'numeric'
as.uid(x, check = TRUE)

## S3 method for class 'data.frame'
as.uid(x, check = TRUE)

## S3 method for class 'uid'
as.data.frame(x, ...)

get_uid_(sci_com, messages = TRUE, rows = NA, key = NULL, sciname = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_uid_+3A_sci_com">sci_com</code></td>
<td>
<p>character; scientific or common name. Or, a <code>taxon_state</code>
object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_uid_+3A_ask">ask</code></td>
<td>
<p>logical; should get_uid be run in interactive mode? If TRUE and
more than one TSN is found for the species, the user is asked for input. If
FALSE NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_uid_+3A_messages">messages</code></td>
<td>
<p>logical; If <code>TRUE</code> (default) the actual taxon queried is
printed on the console.</p>
</td></tr>
<tr><td><code id="get_uid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this function still only gives back a uid
class object with one to many identifiers. See <code><a href="#topic+get_uid_">get_uid_()</a></code> to get back
all, or a subset, of the raw data that you are presented during the ask
process.</p>
</td></tr>
<tr><td><code id="get_uid_+3A_modifier">modifier</code></td>
<td>
<p>(character) A modifier to the <code>sci_com</code> given. Options
include: Organism, Scientific Name, Common Name, All Names, Division,
Filter, Lineage, GC, MGC, Name Tokens, Next Level, PGC, Properties, Rank,
Subtree, Synonym, Text Word. These are not checked, so make sure they are
entered correctly, as is.</p>
</td></tr>
<tr><td><code id="get_uid_+3A_rank_query">rank_query</code></td>
<td>
<p>(character) A taxonomic rank name to modify the query sent
to NCBI. See <a href="#topic+rank_ref">rank_ref</a> for possible options. Though note that
some data sources use atypical ranks, so inspect the data itself for
options. Optional. See <code>Querying</code> below.</p>
</td></tr>
<tr><td><code id="get_uid_+3A_division_filter">division_filter</code></td>
<td>
<p>(character) A division (aka phylum) name to filter
data after retrieved from NCBI. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_uid_+3A_rank_filter">rank_filter</code></td>
<td>
<p>(character) A taxonomic rank name to filter data after
retrieved from NCBI. See <a href="#topic+rank_ref">rank_ref</a> for possible options.
Though note that some data sources use atypical ranks, so inspect the data
itself for options. Optional. See <code>Filtering</code> below.</p>
</td></tr>
<tr><td><code id="get_uid_+3A_key">key</code></td>
<td>
<p>(character) NCBI Entrez API key. optional. See Details.</p>
</td></tr>
<tr><td><code id="get_uid_+3A_sciname">sciname</code></td>
<td>
<p>Deprecated, see <code>sci_com</code></p>
</td></tr>
<tr><td><code id="get_uid_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="get_uid_+3A_x">x</code></td>
<td>
<p>Input to <code><a href="#topic+as.uid">as.uid()</a></code></p>
</td></tr>
<tr><td><code id="get_uid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only used
in <code><a href="#topic+as.uid">as.uid()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>Rate limits</h3>

<p>In case you run into errors due to your rate limit being exceeded, see
<code><a href="#topic+taxize_options">taxize_options()</a></code>, where you can set <code>ncbi_sleep</code>.
</p>


<h3>Querying</h3>

<p>The parameter <code>rank_query</code> is used in the search sent
to NCBI, whereas <code>rank_filter</code> filters data after it comes back. The
parameter <code>modifier</code> adds modifiers to the name. For example,
<code>modifier="Organism"</code> adds that to the name, giving e.g.,
<code>Helianthus[Organism]</code>.
</p>


<h3>Filtering</h3>

<p>The parameters <code>division_filter</code> and
<code>rank_filter</code> are not used in the search to the data provider, but are
used in filtering the data down to a subset that is closer to the target
you want. For all these parameters, you can use regex strings since we use
<code><a href="base.html#topic+grep">grep()</a></code> internally to match. Filtering narrows down to the set
that matches your query, and removes the rest.
</p>


<h3>Beware</h3>

<p>NCBI does funny things sometimes. E.g., if you search on
Fringella morel, a slight misspelling of the genus name, and a non-existent
epithet, NCBI gives back a morel fungal species. In addition, NCBI doesn't
really do fuzzy searching very well, so if there is a slight mis-spelling
in your names, you likely won't get what you are expecting. The lesson:
clean your names before using this function. Other data sources are better
about fuzzy matching.
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication
</p>
<p>Note that even though you can't pass in your key to <code>as.uid</code> functions,
we still use your Entrez API key if you have it saved as an R option
or environment variable.
</p>


<h3>HTTP version</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>Author(s)</h3>

<p>Eduard Szoecs, <a href="mailto:eduardszoecs@gmail.com">eduardszoecs@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_uid(c("Chironomus riparius", "Chaetopteryx"))
get_uid(c("Chironomus riparius", "aaa vva"))

# When not found
get_uid("howdy")
get_uid(c("Chironomus riparius", "howdy"))

# Narrow down results to a division or rank, or both
## By modifying the query
### w/ modifiers to the name
get_uid(sci_com = "Aratinga acuticauda", modifier = "Organism")
get_uid(sci_com = "bear", modifier = "Common Name")

### w/ rank query
get_uid(sci_com = "Pinus", rank_query = "genus")
get_uid(sci_com = "Pinus", rank_query = "subgenus")
### division query doesn't really work, for unknown reasons, so not available

## By filtering the result
## Echinacea example
### Results w/o narrowing
get_uid("Echinacea")
### w/ division
get_uid(sci_com = "Echinacea", division_filter = "eudicots")
get_uid(sci_com = "Echinacea", division_filter = "sea urchins")

## Satyrium example
### Results w/o narrowing
get_uid(sci_com = "Satyrium")
### w/ division
get_uid(sci_com = "Satyrium", division_filter = "monocots")
get_uid(sci_com = "Satyrium", division_filter = "butterflies")

## Rank example
get_uid(sci_com = "Pinus")
get_uid(sci_com = "Pinus", rank_filter = "genus")
get_uid(sci_com = "Pinus", rank_filter = "subgenus")

# Fuzzy filter on any filtering fields
## uses grep on the inside
get_uid("Satyrium", division_filter = "m")

# specify rows to limit choices available
get_uid('Dugesia') # user prompt needed
get_uid('Dugesia', rows=1) # 2 choices, so returns only 1 row, so no choices
get_uid('Dugesia', ask = FALSE) # returns NA for multiple matches

# Go to a website with more info on the taxon
res &lt;- get_uid("Chironomus riparius")
browseURL(attr(res, "uri"))

# Convert a uid without class information to a uid class
as.uid(get_uid("Chironomus riparius")) # already a uid, returns the same
as.uid(get_uid(c("Chironomus riparius","Pinus contorta"))) # same
as.uid(315567) # numeric
as.uid(c(315567,3339,9696)) # numeric vector, length &gt; 1
as.uid("315567") # character
as.uid(c("315567","3339","9696")) # character vector, length &gt; 1
as.uid(list("315567","3339","9696")) # list, either numeric or character
## dont check, much faster
as.uid("315567", check=FALSE)
as.uid(315567, check=FALSE)
as.uid(c("315567","3339","9696"), check=FALSE)
as.uid(list("315567","3339","9696"), check=FALSE)

(out &lt;- as.uid(c(315567,3339,9696)))
data.frame(out)
as.uid( data.frame(out) )

# Get all data back
get_uid_("Puma concolor")
get_uid_("Dugesia")
get_uid_("Dugesia", rows=2)
get_uid_("Dugesia", rows=1:2)
get_uid_(c("asdfadfasd","Pinus contorta"))

# use curl options
get_uid("Quercus douglasii", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_wiki'>Get the page name for a Wiki taxon</h2><span id='topic+get_wiki'></span><span id='topic+as.wiki'></span><span id='topic+as.wiki.wiki'></span><span id='topic+as.wiki.character'></span><span id='topic+as.wiki.list'></span><span id='topic+as.wiki.numeric'></span><span id='topic+as.wiki.data.frame'></span><span id='topic+as.data.frame.wiki'></span><span id='topic+get_wiki_'></span>

<h3>Description</h3>

<p>Get the page name for a Wiki taxon
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_wiki(
  sci_com,
  wiki_site = "species",
  wiki = "en",
  ask = TRUE,
  messages = TRUE,
  limit = 100,
  rows = NA,
  x = NULL,
  ...
)

as.wiki(x, check = TRUE, wiki_site = "species", wiki = "en")

## S3 method for class 'wiki'
as.wiki(x, check = TRUE, wiki_site = "species", wiki = "en")

## S3 method for class 'character'
as.wiki(x, check = TRUE, wiki_site = "species", wiki = "en")

## S3 method for class 'list'
as.wiki(x, check = TRUE, wiki_site = "species", wiki = "en")

## S3 method for class 'numeric'
as.wiki(x, check = TRUE, wiki_site = "species", wiki = "en")

## S3 method for class 'data.frame'
as.wiki(x, check = TRUE, wiki_site = "species", wiki = "en")

## S3 method for class 'wiki'
as.data.frame(x, ...)

get_wiki_(
  x,
  messages = TRUE,
  wiki_site = "species",
  wiki = "en",
  limit = 100,
  rows = NA,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_wiki_+3A_sci_com">sci_com</code></td>
<td>
<p>(character) A vector of common or scientific names. Or, a
<code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_wiki_site">wiki_site</code></td>
<td>
<p>(character) Wiki site. One of species (default), pedia,
commons</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_wiki">wiki</code></td>
<td>
<p>(character) language. Default: en</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_ask">ask</code></td>
<td>
<p>logical; should get_wiki be run in interactive mode?
If <code>TRUE</code> and more than one wiki is found for the species, the user is
asked for input. If <code>FALSE</code> NA is returned for multiple matches.</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_messages">messages</code></td>
<td>
<p>logical; should progress be printed?</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_limit">limit</code></td>
<td>
<p>(integer) number of records to return</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this function still only gives back a wiki
class object with one to many identifiers. See <code><a href="#topic+get_wiki_">get_wiki_()</a></code> to get back
all, or a subset, of the raw data that you are presented during the ask
process.</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_x">x</code></td>
<td>
<p>For <code>get_wiki()</code>: deprecated, see <code>sci_com</code>. For <code>as.wiki</code>, various,
see examples</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="get_wiki_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.wiki">as.wiki()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>wiki_site = "pedia" </code>, we use the english language site by
default. Set the <code>wiki</code> parameter for a different language site.
</p>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wormsid">get_wormsid</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_wiki(sci_com = "Quercus douglasii")
get_wiki(sci_com = "Quercu")
get_wiki(sci_com = "Quercu", "pedia")
get_wiki(sci_com = "Quercu", "commons")

# diff. wikis with wikipedia
get_wiki("Malus domestica", "pedia")
get_wiki("Malus domestica", "pedia", "fr")

# as coercion
as.wiki("Malus_domestica")
as.wiki("Malus_domestica", wiki_site = "commons")
as.wiki("Malus_domestica", wiki_site = "pedia")
as.wiki("Malus_domestica", wiki_site = "pedia", wiki = "fr")
as.wiki("Malus_domestica", wiki_site = "pedia", wiki = "da")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_wormsid'>Get Worms ID for a taxon name</h2><span id='topic+get_wormsid'></span><span id='topic+as.wormsid'></span><span id='topic+as.wormsid.wormsid'></span><span id='topic+as.wormsid.character'></span><span id='topic+as.wormsid.list'></span><span id='topic+as.wormsid.numeric'></span><span id='topic+as.wormsid.data.frame'></span><span id='topic+as.data.frame.wormsid'></span><span id='topic+get_wormsid_'></span>

<h3>Description</h3>

<p>Retrieve Worms ID of a taxon from World Register of Marine
Species (WORMS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_wormsid(
  sci_com,
  searchtype = "scientific",
  marine_only = TRUE,
  fuzzy = NULL,
  accepted = FALSE,
  ask = TRUE,
  messages = TRUE,
  rows = NA,
  query = NULL,
  ...
)

as.wormsid(x, check = TRUE)

## S3 method for class 'wormsid'
as.wormsid(x, check = TRUE)

## S3 method for class 'character'
as.wormsid(x, check = TRUE)

## S3 method for class 'list'
as.wormsid(x, check = TRUE)

## S3 method for class 'numeric'
as.wormsid(x, check = TRUE)

## S3 method for class 'data.frame'
as.wormsid(x, check = TRUE)

## S3 method for class 'wormsid'
as.data.frame(x, ...)

get_wormsid_(
  sci_com,
  messages = TRUE,
  searchtype = "scientific",
  marine_only = TRUE,
  fuzzy = NULL,
  accepted = TRUE,
  rows = NA,
  query = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_wormsid_+3A_sci_com">sci_com</code></td>
<td>
<p>character; A vector of common or scientific names. Or, a
<code>taxon_state</code> object (see <a href="#topic+taxon-state">taxon-state</a>)</p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_searchtype">searchtype</code></td>
<td>
<p>character; One of 'scientific' or 'common', or any unique
abbreviation</p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_marine_only">marine_only</code></td>
<td>
<p>logical; marine only? default: <code>TRUE</code> (only used
when <code>searchtype="scientific"</code>); passed on to <code><a href="worrms.html#topic+wm_records_name">worrms::wm_records_name()</a></code></p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_fuzzy">fuzzy</code></td>
<td>
<p>logical; fuzzy search. default: <code>NULL</code> (<code>TRUE</code> for
<code>searchtype="scientific"</code> and <code>FALSE</code> for <code>searchtype="common"</code> to match
the default values for those parameters in <span class="pkg">worrms</span> package); passed on
to <code><a href="worrms.html#topic+wm_records_name">worrms::wm_records_name()</a></code> or <code><a href="worrms.html#topic+wm_records_common">worrms::wm_records_common()</a></code></p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_accepted">accepted</code></td>
<td>
<p>logical; If TRUE, removes names that are not accepted valid
names by WORMS. Set to <code>FALSE</code> (default) to give back both accepted
and unaccepted names.</p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_ask">ask</code></td>
<td>
<p>logical; should get_wormsid be run in interactive mode?
If <code>TRUE</code> and more than one wormsid is found for the species, the
user is asked for input. If <code>FALSE</code> NA is returned for
multiple matches.</p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_messages">messages</code></td>
<td>
<p>logical; should progress be printed?</p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NaN, all
rows are considered. Note that this function still only gives back a wormsid
class object with one to many identifiers. See <code><a href="#topic+get_wormsid_">get_wormsid_()</a></code> to get back
all, or a subset, of the raw data that you are presented during the ask
process.</p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_query">query</code></td>
<td>
<p>Deprecated, see <code>sci_com</code></p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_x">x</code></td>
<td>
<p>Input to <code>as.wormsid</code></p>
</td></tr>
<tr><td><code id="get_wormsid_+3A_check">check</code></td>
<td>
<p>logical; Check if ID matches any existing on the DB, only
used in <code><a href="#topic+as.wormsid">as.wormsid()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxonomic identifiers as an S3 class.
If a taxon is not found an <code>NA</code> is given. If more than one identifier
is found the function asks for user input if <code>ask = TRUE</code>, otherwise
returns <code>NA</code>. If <code>ask=FALSE</code> and <code>rows</code> does not equal
<code>NA</code>, then a data.frame is given back, but not of the uid class, which
you can't pass on to other functions as you normally can.
</p>
<p>See <code><a href="#topic+get_id_details">get_id_details</a></code> for further details including
attributes and exceptions
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>
<p>Other taxonomic-ids: 
<code><a href="#topic+get_boldid">get_boldid</a>()</code>,
<code><a href="#topic+get_eolid">get_eolid</a>()</code>,
<code><a href="#topic+get_gbifid">get_gbifid</a>()</code>,
<code><a href="#topic+get_ids">get_ids</a>()</code>,
<code><a href="#topic+get_iucn">get_iucn</a>()</code>,
<code><a href="#topic+get_natservid">get_natservid</a>()</code>,
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+get_tolid">get_tolid</a>()</code>,
<code><a href="#topic+get_tpsid">get_tpsid</a>()</code>,
<code><a href="#topic+get_tsn">get_tsn</a>()</code>,
<code><a href="#topic+get_uid">get_uid</a>()</code>,
<code><a href="#topic+get_wiki">get_wiki</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
(x &lt;- get_wormsid('Gadus morhua'))
attributes(x)
attr(x, "match")
attr(x, "multiple_matches")
attr(x, "pattern_match")
attr(x, "uri")

get_wormsid('Pomatomus saltatrix')
get_wormsid(c("Gadus morhua", "Lichenopora neapolitana"))

# marine_only
get_wormsid("Apedinella", marine_only=TRUE)
get_wormsid("Apedinella", marine_only=FALSE)

# fuzzy
## searchtype="scientific": fuzzy is TRUE by default
get_wormsid("Platypro", searchtype="scientific", fuzzy=TRUE)
get_wormsid("Platypro", searchtype="scientific", fuzzy=FALSE)
## searchtype="common": fuzzy is FALSE by default
get_wormsid("clam", searchtype="common", fuzzy=FALSE)
get_wormsid("clam", searchtype="common", fuzzy=TRUE)

# by common name
get_wormsid("dolphin", 'common')
get_wormsid("clam", 'common')

# specify rows to limit choices available
get_wormsid('Plat')
get_wormsid('Plat', rows=1)
get_wormsid('Plat', rows=1:2)

# When not found
get_wormsid("howdy")
get_wormsid(c('Gadus morhua', "howdy"))

# Convert a wormsid without class information to a wormsid class
# already a wormsid, returns the same
as.wormsid(get_wormsid('Gadus morhua'))
# same
as.wormsid(get_wormsid(c('Gadus morhua', 'Pomatomus saltatrix')))
# numeric
as.wormsid(126436)
# numeric vector, length &gt; 1
as.wormsid(c(126436,151482))
# character
as.wormsid("126436")
# character vector, length &gt; 1
as.wormsid(c("126436","151482"))
# list, either numeric or character
as.wormsid(list("126436","151482"))
## dont check, much faster
as.wormsid("126436", check=FALSE)
as.wormsid(126436, check=FALSE)
as.wormsid(c("126436","151482"), check=FALSE)
as.wormsid(list("126436","151482"), check=FALSE)

(out &lt;- as.wormsid(c(126436,151482)))
data.frame(out)
as.wormsid( data.frame(out) )

# Get all data back
get_wormsid_("Plat")
get_wormsid_("Plat", rows=1)
get_wormsid_("Plat", rows=1:2)
get_wormsid_("Plat", rows=1:75)

## End(Not run)
</code></pre>

<hr>
<h2 id='getkey'>Function to get API key.</h2><span id='topic+getkey'></span>

<h3>Description</h3>

<p>Checks first to get key from your .Rprofile or .Renviron (or similar) file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getkey(x = NULL, service)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getkey_+3A_x">x</code></td>
<td>
<p>(character) An API key, defaults to <code>NULL</code></p>
</td></tr>
<tr><td><code id="getkey_+3A_service">service</code></td>
<td>
<p>(character) The API data provider, used to match to
default guest key (for Tropicos; there's no guest
key for NCBI or IUCN, for which you have to get your own)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getkey(service="tropicos")
getkey(service="iucn")
getkey(service="entrez")

## End(Not run)
</code></pre>

<hr>
<h2 id='gisd_isinvasive'>Check invasive species status for a set of species from GISD database</h2><span id='topic+gisd_isinvasive'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gisd_isinvasive(...)
</code></pre>

<hr>
<h2 id='gna_data_sources'>Get metadata about GNA data sources</h2><span id='topic+gna_data_sources'></span>

<h3>Description</h3>

<p>Downloads metadata about Global Names Architecture (GNA) data sources
available to be used in other GNA functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gna_data_sources(output_type = "table", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gna_data_sources_+3A_output_type">output_type</code></td>
<td>
<p>What format of output to return. Either <code>'json'</code>,
<code>'list'</code>, or <code>'table'</code>.</p>
</td></tr>
<tr><td><code id="gna_data_sources_+3A_...">...</code></td>
<td>
<p>Passed to <a href="crul.html#topic+HttpClient">crul::HttpClient</a>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zachary S.L. Foster
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

gna_data_sources()

## End(Not run)

</code></pre>

<hr>
<h2 id='gna_parse'>Parse scientific names using Global Names Parser</h2><span id='topic+gna_parse'></span>

<h3>Description</h3>

<p>Parse scientific names using Global Names Parser
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gna_parse(names, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gna_parse_+3A_names">names</code></td>
<td>
<p>A vector of length 1 or more taxonomic names</p>
</td></tr>
<tr><td><code id="gna_parse_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with results, the submitted names, and the
parsed names with additional information.
</p>


<h3>References</h3>

<p>http://gni.globalnames.org/
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gbif_parse">gbif_parse()</a></code>, <code><a href="#topic+gni_parse">gni_parse()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gna_parse("Cyanistes caeruleus")
gna_parse("Plantago minor")
gna_parse("Plantago minor minor")
gna_parse(c("Plantago minor minor","Helianthus annuus texanus"))

# if &gt; 20 names, uses an HTTP POST request
x &lt;- names_list("species", size = 30)
gna_parse(x)

# pass on curl options
gna_parse("Cyanistes caeruleus", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='gna_search'>Search for taxonomic names using the Global Names Architecture</h2><span id='topic+gna_search'></span>

<h3>Description</h3>

<p>Uses the Global Names Index, see http://gni.globalnames.org
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gna_search(sci, justtotal = FALSE, parse_names = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gna_search_+3A_sci">sci</code></td>
<td>
<p>(character) required. Name pattern you want to search
for. WARNING: Does not work for common names. Search term may include
following options:
</p>

<ul>
<li> <p><code>n</code>: A shortcut that allows to put together several elements (e.g., <code style="white-space: pre;">&#8288;n:B. bubo Linn. 1750-1800&#8288;</code>)
</p>
</li>
<li> <p><code>g</code>: a genus name. (e.g. <code>g:B.</code>, <code>g:Bub.</code>, <code>g:Bubo</code>)
</p>
</li>
<li> <p><code>isp</code>: an infraspecies name (e.g. <code>sp:bubo</code>, <code>sp:gallop.</code>)
</p>
</li>
<li> <p><code>asp</code>: either species or infraspecies (all sp) (e.g. <code>asp:bubo</code>)
</p>
</li>
<li> <p><code>ds</code>: data-sources IDs (e.g., <code style="white-space: pre;">&#8288;ds:1,2,3&#8288;</code>)
</p>
</li>
<li> <p><code>tx</code>: parent taxon . Uses classification of the first data-source from <code>ds</code>. If data-sources are not set, uses Catalogue of Life. (e.g. <code>tx:Aves</code>)
</p>
</li>
<li> <p><code>au</code>: author - Search by author word    (e.g. <code>au:Linnaeus</code>, <code>au:Linn.</code>)
</p>
</li>
<li> <p><code>y</code>: year - Search by year (e.g. <code>y:2005</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="gna_search_+3A_justtotal">justtotal</code></td>
<td>
<p>Return only the total results found.</p>
</td></tr>
<tr><td><code id="gna_search_+3A_parse_names">parse_names</code></td>
<td>
<p>If <code>TRUE</code> use <code><a href="#topic+gni_parse">gni_parse()</a></code> on the outputs.</p>
</td></tr>
<tr><td><code id="gna_search_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame of results.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain, Zachary Foster
</p>


<h3>References</h3>

<p>http://gni.globalnames.org/
https://apidoc.globalnames.org/gnames
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gnr_datasources">gnr_datasources()</a></code>, <code><a href="#topic+gna_search">gna_search()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gna_search('n:B. bubo ds:1,2 au:Linn. y:1700-')

## End(Not run)
</code></pre>

<hr>
<h2 id='gna_verifier'>Verify a list of scientific names against biodiversity data-sources.</h2><span id='topic+gna_verifier'></span>

<h3>Description</h3>

<p>This service parses incoming names, executes exact or fuzzy matching as
required, and returns the best-scored result. Optionally, it can also return
matches from data-sources selected by a user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gna_verifier(
  names,
  data_sources = c(1, 12),
  all_matches = FALSE,
  capitalize = FALSE,
  species_group = FALSE,
  fuzzy_uninomial = FALSE,
  stats = FALSE,
  main_taxon_threshold = 0.5,
  output_type = "table",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gna_verifier_+3A_names">names</code></td>
<td>
<p>A <code>character</code> vector of taxon names to verify.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_data_sources">data_sources</code></td>
<td>
<p>A <code>character</code> or <code>integer</code> vector with numbers
corresponding to data sources. See the Global Names Architecture
documentation for a list of available options.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_all_matches">all_matches</code></td>
<td>
<p>When <code>TRUE</code>, return all found matches, not only the best
one. Multiple results are returned in results. These results are sorted by
matching quality, the first result is the same as bestResult.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_capitalize">capitalize</code></td>
<td>
<p>When <code>TRUE</code>, capitalize the first letter of a name-string.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_species_group">species_group</code></td>
<td>
<p>When <code>TRUE</code>, expands the search to species group where
applicable.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_fuzzy_uninomial">fuzzy_uninomial</code></td>
<td>
<p>When <code>TRUE</code>, allows fuzzy matching for uninomial
names.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_stats">stats</code></td>
<td>
<p>When <code>TRUE</code>, finds out a kingdom and a taxon (main taxon) that
contain most names. It only takes in account the names matched to the
Catalogue of Life entries. This option is ignored, if the Catalogue of Life
is not included in data-sources.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_main_taxon_threshold">main_taxon_threshold</code></td>
<td>
<p>A <code>numeric</code> vector from 0.5 to 1. This sets the
minimal percentage for the main taxon discovery.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_output_type">output_type</code></td>
<td>
<p>A <code>character</code> vector of length 1, either <code>table</code> or
<code>list</code>, indicating the format of the output. The tabular output only
contains values that consistently appear in all results, so <code>list</code> output
can have additional information. For <code>list</code> and <code>json</code> outputs, only values
for unique taxon names are returned, but the <code>table</code> output has rows that
correspond 1-1 with the input data.</p>
</td></tr>
<tr><td><code id="gna_verifier_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depends on the value of the <code>output_type</code> option
</p>


<h3>Author(s)</h3>

<p>Zachary S.L. Foster
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gna_verifier(c("Helianthus annuus", "Homo saapiens"))
gna_verifier(c("Helianthus annuus", "Homo saapiens"), all_matches = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='gni_details'>Search for taxonomic name details using the Global Names Index</h2><span id='topic+gni_details'></span>

<h3>Description</h3>

<p>Uses the Global Names Index, see http://gni.globalnames.org/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gni_details(id, all_records = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gni_details_+3A_id">id</code></td>
<td>
<p>Name id. Required.</p>
</td></tr>
<tr><td><code id="gni_details_+3A_all_records">all_records</code></td>
<td>
<p>If all_records is 1, GNI returns all records from all
repositories for the name string (takes 0, or 1 [default]).</p>
</td></tr>
<tr><td><code id="gni_details_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame of results.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gnr_datasources">gnr_datasources()</a></code>, <code><a href="#topic+gna_search">gna_search()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gni_details(id = 17802847)

# pass on curl options
gni_details(id = 17802847, verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='gni_parse'>Parse scientific names using EOL's name parser.</h2><span id='topic+gni_parse'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gni_parse(names, ...)
</code></pre>

<hr>
<h2 id='gni_seach'>Search for taxonomic names using the Global Names Index</h2><span id='topic+gni_seach'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gni_seach(names, ...)
</code></pre>

<hr>
<h2 id='gnr_datasources'>Global Names Resolver Data Sources</h2><span id='topic+gnr_datasources'></span>

<h3>Description</h3>

<p>Retrieve data sources used in the Global Names Resolver
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gnr_datasources(..., todf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gnr_datasources_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
<tr><td><code id="gnr_datasources_+3A_todf">todf</code></td>
<td>
<p>defunct, always get a data.frame back now</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame/tibble
</p>


<h3>References</h3>

<p>https://resolver.globalnames.org/data_sources
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gnr_resolve">gnr_resolve()</a></code>, <code><a href="#topic+gna_search">gna_search()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# all data sources
gnr_datasources()

# give me the id for EOL
out &lt;- gnr_datasources()
out[out$title == "EOL", "id"]

# Fuzzy search for sources with the word zoo
out &lt;- gnr_datasources()
out[agrep("zoo", out$title, ignore.case = TRUE), ]

## End(Not run)
</code></pre>

<hr>
<h2 id='gnr_resolve'>Resolve names using Global Names Resolver</h2><span id='topic+gnr_resolve'></span>

<h3>Description</h3>

<p>NOTE: this function is depreciated and will be removed in a future version.
The service this function interacts with is no longer maintained and has been
replaced by GNA Verifier, which can be used with the <code><a href="#topic+gna_verifier">gna_verifier()</a></code>
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gnr_resolve(
  sci,
  data_source_ids = NULL,
  resolve_once = FALSE,
  with_context = FALSE,
  canonical = FALSE,
  highestscore = TRUE,
  best_match_only = FALSE,
  preferred_data_sources = NULL,
  with_canonical_ranks = FALSE,
  http = "get",
  cap_first = TRUE,
  fields = "minimal",
  names = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gnr_resolve_+3A_sci">sci</code></td>
<td>
<p>character; taxonomic names to be resolved. Doesn't work for
vernacular/common names.</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_data_source_ids">data_source_ids</code></td>
<td>
<p>character; IDs to specify what data source is
searched. See <code><a href="#topic+gnr_datasources">gnr_datasources()</a></code>.</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_resolve_once">resolve_once</code></td>
<td>
<p>logical; Find the first available match instead of
matches across all data sources with all possible renderings of a name.
When <code>TRUE</code>, response is rapid but incomplete.</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_with_context">with_context</code></td>
<td>
<p>logical; Reduce the likelihood of matches to taxonomic
homonyms. When <code>TRUE</code> a common taxonomic context is calculated for all
supplied names from matches in data sources that have classification tree
paths. Names out of determined context are penalized during score
calculation.</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_canonical">canonical</code></td>
<td>
<p>logical; If <code>FALSE</code> (default), gives back names with
taxonomic authorities. If <code>TRUE</code>, returns canocial names (without tax.
authorities and abbreviations).</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_highestscore">highestscore</code></td>
<td>
<p>logical; Return those names with the highest score for
each searched name? Defunct</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_best_match_only">best_match_only</code></td>
<td>
<p>(logical) If <code>TRUE</code>, best match only returned.
Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_preferred_data_sources">preferred_data_sources</code></td>
<td>
<p>(character) A vector of one or more data source
IDs.</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_with_canonical_ranks">with_canonical_ranks</code></td>
<td>
<p>(logical) Returns names with infraspecific ranks,
if present. If <code>TRUE</code>, we force <code>canonical=TRUE</code>, otherwise this parameter
would have no effect. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_http">http</code></td>
<td>
<p>The HTTP method to use, one of &quot;get&quot; or &quot;post&quot;. Default: &quot;get&quot;.
Use <code>http="post"</code> with large queries. Queries with &gt; 300 records use &quot;post&quot;
automatically because &quot;get&quot; would fail</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_cap_first">cap_first</code></td>
<td>
<p>(logical) For each name, fix so that the first name part is
capitalized, while others are not. This web service is sensitive to
capitalization, so you'll get different results depending on
capitalization. First name capitalized is likely what you'll want and is
the default. If <code>FALSE</code>, names are not modified. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_fields">fields</code></td>
<td>
<p>(character) One of minimal (default) or all. Minimal gives back
just four fields, whereas all gives all fields back.</p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_names">names</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="gnr_resolve_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>See section <strong>Age of datasets in the Global Names Resolver</strong>
</p>


<h3>Value</h3>

<p>A data.frame with one attribute <code>not_known</code>: a character vector of
taxa unknown to the Global Names Index. Access like <code>attr(output, "not_known")</code>, or <code>attributes(output)$not_known</code>.
</p>
<p>Columns of the output data.frame:
</p>

<ul>
<li><p> user_supplied_name (character) - the name you passed in to the
<code>names</code> parameter, unchanged.
</p>
</li>
<li><p> submitted_name (character) - the actual name submitted to the GNR
service
</p>
</li>
<li><p> data_source_id (integer/numeric) - data source ID
</p>
</li>
<li><p> data_source_title (character) - data source name
</p>
</li>
<li><p> gni_uuid (character) - Global Names Index UUID (aka identifier)
</p>
</li>
<li><p> matched_name (character) - the matched name in the GNR service
</p>
</li>
<li><p> matched_name2 (character) - returned if <code>canonical=TRUE</code>, in
which case <strong>matched_name</strong> is not returned
</p>
</li>
<li><p> classification_path (character) - names of the taxonomic
classification tree, with names separated by pipes (<code>|</code>)
</p>
</li>
<li><p> classification_path_ranks (character) - ranks of the taxonomic
classification tree, with names separated by pipes (<code>|</code>)
</p>
</li>
<li><p> classification_path_ids (character) - identifiers of the taxonomic
classification tree, with names separated by pipes (<code>|</code>)
</p>
</li>
<li><p> taxon_id (character) - taxon identifier
</p>
</li>
<li><p> edit_distance (integer/numeric) - edit distance
</p>
</li>
<li><p> imported_at (character) - date imported
</p>
</li>
<li><p> match_type (integer/numeric) - match type
</p>
</li>
<li><p> match_value (character) - description of match type
</p>
</li>
<li><p> prescore (character) - pre score
</p>
</li>
<li><p> score (numeric) - score
</p>
</li>
<li><p> local_id (character) - local identifier
</p>
</li>
<li><p> url (character) - URL for taxon
</p>
</li>
<li><p> global_id (character) - global identifier
</p>
</li>
<li><p> current_taxon_id (character) - current taxon id
</p>
</li>
<li><p> current_name_string (character) - current name string
</p>
<p>Note that names (i.e. rows) are dropped that are NA, are zero length
strings, are not character vectors, or are not found by the API.
</p>
</li></ul>



<h3>Age of datasets in the Global Names Resolver</h3>

<p>IMPORTANT: Datasets
used in the Global Names Resolver vary in how recently they've been
updated. See the <code>updated_at</code> field in the output of <code><a href="#topic+gnr_datasources">gnr_datasources()</a></code>
for dates when each dataset was last updated.
</p>


<h3>preferred_data_sources</h3>

<p>If <code>preferred_data_sources</code> is used, only
the preferred data is returned - if it has any results.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>References</h3>

<p>http://gnrd.globalnames.org/api http://gnrd.globalnames.org/
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gnr_datasources">gnr_datasources()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gnr_resolve(sci = c("Helianthus annuus", "Homo sapiens"))
gnr_resolve(sci = c("Asteraceae", "Plantae"))

# Using data source 12 (Encyclopedia of Life)
sources &lt;- gnr_datasources()
sources
eol &lt;- sources$id[sources$title == 'EOL']
gnr_resolve(names=c("Helianthos annuus","Homo sapians"), data_source_ids=eol)

# Two species in the NE Brazil catalogue
sps &lt;- c('Justicia brasiliana','Schinopsis brasiliensis')
gnr_resolve(sci = sps, data_source_ids = 145)

# Best match only, compare the two
gnr_resolve(sci = "Helianthus annuus", best_match_only = FALSE)
gnr_resolve(sci = "Helianthus annuus", best_match_only = TRUE)

# Preferred data source
gnr_resolve(sci = "Helianthus annuus", preferred_data_sources = c(3,4))

# Return canonical names - default is canonical=FALSE
head(gnr_resolve(sci = "Helianthus annuus"))
head(gnr_resolve(sci = "Helianthus annuus", canonical=TRUE))

# Return canonical names with authority stripped but
# ranks still present
gnr_resolve("Scorzonera hispanica L. subsp. asphodeloides Wallr.")
## vs.
gnr_resolve("Scorzonera hispanica L. subsp. asphodeloides Wallr.",
   with_canonical_ranks = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='id2name'>Taxonomic IDs to taxonomic names</h2><span id='topic+id2name'></span><span id='topic+id2name.default'></span><span id='topic+id2name.tolid'></span><span id='topic+id2name.tsn'></span><span id='topic+id2name.uid'></span><span id='topic+id2name.wormsid'></span><span id='topic+id2name.gbifid'></span><span id='topic+id2name.boldid'></span>

<h3>Description</h3>

<p>Taxonomic IDs to taxonomic names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>id2name(id, db = NULL, x = NULL, ...)

## Default S3 method:
id2name(id, db = NULL, x = NULL, ...)

## S3 method for class 'tolid'
id2name(id, ...)

## S3 method for class 'tsn'
id2name(id, ...)

## S3 method for class 'uid'
id2name(id, ...)

## S3 method for class 'wormsid'
id2name(id, ...)

## S3 method for class 'gbifid'
id2name(id, ...)

## S3 method for class 'boldid'
id2name(id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="id2name_+3A_id">id</code></td>
<td>
<p>vector of taxonomic IDs (character or numeric)</p>
</td></tr>
<tr><td><code id="id2name_+3A_db">db</code></td>
<td>
<p>(character) database to query. One or more of <code>tol</code>, <code>itis</code>,
<code>ncbi</code>, <code>worms</code>, <code>gbif</code>, or <code>bold</code>. Note that each
taxonomic data source has their own  identifiers, so that if you provide
the wrong <code>db</code> value for the identifier you could get a result,
but it will likely be wrong (not what you were expecting). If using ncbi
we recommend getting API keys; see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="id2name_+3A_x">x</code></td>
<td>
<p>Deprecated, see <code>id</code></p>
</td></tr>
<tr><td><code id="id2name_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code>tol_id2name</code> or
<a href="#topic+itis_getrecord">itis_getrecord</a>, or other internal functions.
See those functions for what parameters can be passed on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of data.frames, named by the input taxonomic ids
</p>


<h3>HTTP version for NCBI requests</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ITIS
id2name(19322, db = "itis")

# TOL
id2name(515698, db = "tol")
# get NCBI ID and pass to classification()
x &lt;- id2name(515698, db = "tol")
classification(as.uid(x[[1]]$tax_sources_ncbi))

# NCBI
id2name(315567, db = "ncbi")
id2name(3339, db = "ncbi")
id2name(9696, db = "ncbi")
id2name(c(9695, 9696), db = "ncbi")

# WORMS
id2name(105706, db = "worms")

# GBIF
id2name(2441176, db = "gbif")

# BOLD
id2name(88899, db = "bold")

## End(Not run)
</code></pre>

<hr>
<h2 id='ion'>ION - Index to Organism Names</h2><span id='topic+ion'></span>

<h3>Description</h3>

<p>ION - Index to Organism Names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ion(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ion_+3A_x">x</code></td>
<td>
<p>An LSID number. Required.</p>
</td></tr>
<tr><td><code id="ion_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>References</h3>

<p>http://www.organismnames.com
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ion(155166)
ion(298678)
ion(4796748) # ursus americanus
ion(1280626) # puma concolor

## End(Not run)
</code></pre>

<hr>
<h2 id='iplant_resolve'>iPlant name resolution</h2><span id='topic+iplant_resolve'></span>

<h3>Description</h3>

<p>iPlant name resolution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iplant_resolve(sci, retrieve = "all", query = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iplant_resolve_+3A_sci">sci</code></td>
<td>
<p>Vector of one or more taxonomic names (no common names)</p>
</td></tr>
<tr><td><code id="iplant_resolve_+3A_retrieve">retrieve</code></td>
<td>
<p>Specifies whether to retrieve all matches for the
names submitted. One of 'best' (retrieves only the single best match
for each name submitted) or 'all' (retrieves all matches)</p>
</td></tr>
<tr><td><code id="iplant_resolve_+3A_query">query</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="iplant_resolve_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
iplant_resolve(sci=c("Helianthus annuus", "Homo sapiens"))
iplant_resolve("Helianthusss")
iplant_resolve("Pooa")
iplant_resolve("Helianthusss", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ipni_search'>Search for names in the International Plant Names Index (IPNI).</h2><span id='topic+ipni_search'></span>

<h3>Description</h3>

<p>Note: This data source is also provided in the Global Names Index (GNI)
(http://gni.globalnames.org/data_sources). The interface to the data is
different among the two services though.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ipni_search(
  family = NULL,
  infrafamily = NULL,
  genus = NULL,
  infragenus = NULL,
  species = NULL,
  infraspecies = NULL,
  publicationtitle = NULL,
  authorabbrev = NULL,
  includepublicationauthors = NULL,
  includebasionymauthors = NULL,
  geounit = NULL,
  addedsince = NULL,
  modifiedsince = NULL,
  isapnirecord = NULL,
  isgcirecord = NULL,
  isikrecord = NULL,
  ranktoreturn = NULL,
  output = "minimal",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ipni_search_+3A_family">family</code></td>
<td>
<p>Family name to search on (Optional)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_infrafamily">infrafamily</code></td>
<td>
<p>Infrafamilial name to search on (Optional)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_genus">genus</code></td>
<td>
<p>Genus name to search on (Optional)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_infragenus">infragenus</code></td>
<td>
<p>Infrageneric name to search on (Optional)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_species">species</code></td>
<td>
<p>Species name to search on (Optional) - Note, this is the
epithet, not the full genus - epithet name combination.</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_infraspecies">infraspecies</code></td>
<td>
<p>Infraspecies name to search on (Optional)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_publicationtitle">publicationtitle</code></td>
<td>
<p>Publication name or abbreviation to search on.
Again, replace any spaces with a '+' (e.g. 'J.+Bot.') (Optional)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_authorabbrev">authorabbrev</code></td>
<td>
<p>Author standard form to search on (publishing author,
basionym author or both - see below) (Optional)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_includepublicationauthors">includepublicationauthors</code></td>
<td>
<p>TRUE (default) to include the taxon author
in the search or FALSE to exclude it</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_includebasionymauthors">includebasionymauthors</code></td>
<td>
<p>TRUE (default) to include the basionum author
in the search or FALSE to exclude it</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_geounit">geounit</code></td>
<td>
<p>Country name or other geographical unit to search on (see
the help pages for more information and warnings about the use of this
option) (Optional)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_addedsince">addedsince</code></td>
<td>
<p>Date to search on in the format 'yyyy-mm-dd', e.g.
2005-08-01 for all records added since the first of August, 2005. (see the
help pages for more information and warnings about the use of this option)
(Optional. If supplied must be in format YYYY-MM-DD and must be greater
than or equal to 1984-01-01.)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_modifiedsince">modifiedsince</code></td>
<td>
<p>Date to search on in the format 'yyyy-mm-dd', e.g.
2005-08-01 for all records edited since the first of August, 2005. (See
the help pages for more information about the use of this option)
(Optional. If supplied must be in format YYYY-MM-DD and must be greater
than or equal to 1993-01-01.)</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_isapnirecord">isapnirecord</code></td>
<td>
<p>FALSE (default) to exclude records from the Australian
Plant Name Index</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_isgcirecord">isgcirecord</code></td>
<td>
<p>FALSE (default) to exclude records from the Gray Cards
Index</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_isikrecord">isikrecord</code></td>
<td>
<p>FALSE (default) to exclude records from the Index Kewensis</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_ranktoreturn">ranktoreturn</code></td>
<td>
<p>One of a few options to choose the ranks returned. See
details.</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_output">output</code></td>
<td>
<p>One of minimal (default), classic, short, or extended</p>
</td></tr>
<tr><td><code id="ipni_search_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a>
(Optional). Default: returns all ranks.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ranktoreturn</code> options:
</p>

<ul>
<li><p> &quot;all&quot; - all records
</p>
</li>
<li><p> &quot;fam&quot; - family records
</p>
</li>
<li><p> &quot;infrafam&quot; - infrafamilial records
</p>
</li>
<li><p> &quot;gen&quot; - generic records
</p>
</li>
<li><p> &quot;infragen&quot; - infrageneric records
</p>
</li>
<li><p> &quot;spec&quot; - species records
</p>
</li>
<li><p> &quot;infraspec&quot; - infraspecific records
</p>
</li></ul>



<h3>Value</h3>

<p>a tibble (data.frame)
</p>


<h3>References</h3>

<p>https://web.archive.org/web/20190501132148/http://www.ipni.org/link_to_ipni.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ipni_search(genus='Brintonia', isapnirecord=TRUE, isgcirecord=TRUE,
  isikrecord=TRUE)
ipni_search(genus='Ceanothus')
ipni_search(genus='Pinus', species='contorta')

# Different output formats
ipni_search(genus='Ceanothus')
ipni_search(genus='Ceanothus', output='short')
ipni_search(genus='Ceanothus', output='extended')

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_acceptname'>Retrieve accepted TSN and name</h2><span id='topic+itis_acceptname'></span>

<h3>Description</h3>

<p>Retrieve accepted TSN and name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_acceptname(searchtsn, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_acceptname_+3A_searchtsn">searchtsn</code></td>
<td>
<p>One or more TSN for a taxon (numeric/integer)</p>
</td></tr>
<tr><td><code id="itis_acceptname_+3A_...">...</code></td>
<td>
<p>Curl options passed on to<a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with with row number equal to input vector length, and
with three columns:
</p>

<ul>
<li><p> submittedtsn (numeric) - The submitted TSN
</p>
</li>
<li><p> acceptedname (character) - The accepted name - if the submitted TSN
is the accepted TSN, then this is <code>NA_character_</code> because ITIS does
not return a name along with the TSN if it's an accepted name. We could
make an extra HTTP request to ITIS, but that means additional time.
</p>
</li>
<li><p> acceptedtsn (numeric) - The accepted TSN
</p>
</li>
<li><p> author (character) - taxonomic authority
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# TSN accepted - good name
itis_acceptname(searchtsn = 208527)

# TSN not accepted - input TSN is old
itis_acceptname(searchtsn = 504239)

# many accepted names
ids &lt;- c(18161, 18162, 18163, 18164, 18165, 18166, 46173, 46174,
46178, 46181, 46186, 46193, 46196, 46197, 46200, 46201, 46204,
46207, 46867, 46868)
itis_acceptname(searchtsn = ids)

# many unaccepted names
ids &lt;- c(39087, 46208, 46973, 46976, 46978, 46980, 47295, 47445,
47448, 47512, 47515, 47527, 47546, 47622, 47783, 47786, 47787,
47788, 47835, 47839)
itis_acceptname(searchtsn = ids)

# many: mix of accepted and unaccepted names
ids &lt;- c(18161, 18162, 47527, 47546, 47622, 46200)
itis_acceptname(searchtsn = ids)

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_downstream'>Retrieve all taxa names or TSNs downstream in hierarchy from given TSN.</h2><span id='topic+itis_downstream'></span>

<h3>Description</h3>

<p>Retrieve all taxa names or TSNs downstream in hierarchy from given TSN.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_downstream(id, downto, intermediate = FALSE, tsns = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_downstream_+3A_id">id</code></td>
<td>
<p>A taxonomic serial number.</p>
</td></tr>
<tr><td><code id="itis_downstream_+3A_downto">downto</code></td>
<td>
<p>The taxonomic level you want to go down to. See examples below.
The taxonomic level IS case sensitive, and you do have to spell it
correctly. See <code>data(rank_ref)</code> for spelling.</p>
</td></tr>
<tr><td><code id="itis_downstream_+3A_intermediate">intermediate</code></td>
<td>
<p>(logical) If TRUE, return a list of length two with target
taxon rank names, with additional list of data.frame's of intermediate
taxonomic groups. Default: FALSE</p>
</td></tr>
<tr><td><code id="itis_downstream_+3A_tsns">tsns</code></td>
<td>
<p>Deprecated, see <code>id</code></p>
</td></tr>
<tr><td><code id="itis_downstream_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="ritis.html#topic+rank_name">ritis::rank_name()</a></code> and
<code><a href="ritis.html#topic+hierarchy">ritis::hierarchy_down()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame of taxonomic information downstream to family from e.g.,
Order, Class, etc., or if <code>intermediated=TRUE</code>, list of length two,
with target taxon rank names, and intermediate names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## the plant class Bangiophyceae, tsn 846509
itis_downstream(id = 846509, downto="genus")
itis_downstream(id = 846509, downto="genus", intermediate=TRUE)

# get families downstream from Acridoidea
itis_downstream(id = 650497, "family")
## here, intermediate leads to the same result as the target
itis_downstream(id = 650497, "family", intermediate=TRUE)

# get species downstream from Ursus
itis_downstream(id = 180541, "species")

# get orders down from the Division Rhodophyta (red algae)
itis_downstream(id = 660046, "order")
itis_downstream(id = 660046, "order", intermediate=TRUE)

# get tribes down from the family Apidae
itis_downstream(id = 154394, downto="tribe")
itis_downstream(id = 154394, downto="tribe", intermediate=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_getrecord'>Get full ITIS record for one or more ITIS TSN's or lsid's.</h2><span id='topic+itis_getrecord'></span>

<h3>Description</h3>

<p>Get full ITIS record for one or more ITIS TSN's or lsid's.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_getrecord(values, by = "tsn", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_getrecord_+3A_values">values</code></td>
<td>
<p>(character) One or more TSN's (taxonomic serial number) or lsid's for a
taxonomic group</p>
</td></tr>
<tr><td><code id="itis_getrecord_+3A_by">by</code></td>
<td>
<p>(character) By &quot;tsn&quot; (default) or &quot;lsid&quot;</p>
</td></tr>
<tr><td><code id="itis_getrecord_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to <a href="ritis.html#topic+full_record">ritis::full_record</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can only enter values in tsn parameter or lsid, not both.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# by TSN
itis_getrecord(202385)
itis_getrecord(c(202385,70340))

# by lsid
itis_getrecord("urn:lsid:itis.gov:itis_tsn:202385", "lsid")

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_hierarchy'>ITIS hierarchy</h2><span id='topic+itis_hierarchy'></span>

<h3>Description</h3>

<p>Get hierarchies from TSN values, full, upstream only, or
immediate downstream only
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_hierarchy(tsn, what = "full", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_hierarchy_+3A_tsn">tsn</code></td>
<td>
<p>One or more TSN's (taxonomic serial number). Required.</p>
</td></tr>
<tr><td><code id="itis_hierarchy_+3A_what">what</code></td>
<td>
<p>One of full (full hierarchy), up (immediate upstream), or down
(immediate downstream)</p>
</td></tr>
<tr><td><code id="itis_hierarchy_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to <code><a href="ritis.html#topic+hierarchy">ritis::hierarchy_full()</a></code>
<code><a href="ritis.html#topic+hierarchy">ritis::hierarchy_up()</a></code> or <code><a href="ritis.html#topic+hierarchy">ritis::hierarchy_down()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that <code><a href="#topic+itis_downstream">itis_downstream()</a></code> gets taxa downstream to a
particular rank, while this function only gets immediate names downstream.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+itis_downstream">itis_downstream()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get full hierarchy
itis_hierarchy(tsn=180543)

# Get hierarchy upstream
itis_hierarchy(tsn=180543, "up")

# Get hierarchy downstream
itis_hierarchy(tsn=180543, "down")

# Many tsn's
itis_hierarchy(tsn=c(180543,41074,36616))

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_kingdomnames'>Get kingdom names</h2><span id='topic+itis_kingdomnames'></span>

<h3>Description</h3>

<p>Get kingdom names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_kingdomnames(tsn = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_kingdomnames_+3A_tsn">tsn</code></td>
<td>
<p>One or more TSN's (taxonomic serial number)</p>
</td></tr>
<tr><td><code id="itis_kingdomnames_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to getkingdomnamefromtsn</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
itis_kingdomnames(202385)
itis_kingdomnames(tsn=c(202385,183833,180543))

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_lsid'>Get TSN from LSID</h2><span id='topic+itis_lsid'></span>

<h3>Description</h3>

<p>Get TSN from LSID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_lsid(lsid = NULL, what = "tsn", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_lsid_+3A_lsid">lsid</code></td>
<td>
<p>One or more lsid's</p>
</td></tr>
<tr><td><code id="itis_lsid_+3A_what">what</code></td>
<td>
<p>What to retrieve. One of tsn, record, or fullrecord</p>
</td></tr>
<tr><td><code id="itis_lsid_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to <code><a href="ritis.html#topic+lsid2tsn">ritis::lsid2tsn()</a></code>,
<code><a href="ritis.html#topic+record">ritis::record()</a></code>, or <code><a href="ritis.html#topic+full_record">ritis::full_record()</a></code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get TSN
itis_lsid("urn:lsid:itis.gov:itis_tsn:180543")
itis_lsid(lsid=c("urn:lsid:itis.gov:itis_tsn:180543","urn:lsid:itis.gov:itis_tsn:28726"))

# Get partial record
itis_lsid("urn:lsid:itis.gov:itis_tsn:180543", "record")

# Get full record
itis_lsid("urn:lsid:itis.gov:itis_tsn:180543", "fullrecord")

# An invalid lsid (a tsn actually)
itis_lsid(202385)

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_name'>Get taxonomic names for a given taxonomic name query.</h2><span id='topic+itis_name'></span>

<h3>Description</h3>

<p>Get taxonomic names for a given taxonomic name query.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_name(query = NULL, get = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_name_+3A_query">query</code></td>
<td>
<p>TSN number (taxonomic serial number).</p>
</td></tr>
<tr><td><code id="itis_name_+3A_get">get</code></td>
<td>
<p>The rank of the taxonomic name to get.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Taxonomic name for the searched taxon.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
itis_name(query="Helianthus annuus", get="family")

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_native'>Get jurisdiction data, i.e., native or not native in a region.</h2><span id='topic+itis_native'></span>

<h3>Description</h3>

<p>Get jurisdiction data, i.e., native or not native in a region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_native(tsn = NULL, what = "bytsn", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_native_+3A_tsn">tsn</code></td>
<td>
<p>One or more TSN's (taxonomic serial number)</p>
</td></tr>
<tr><td><code id="itis_native_+3A_what">what</code></td>
<td>
<p>One of bytsn, values, or originvalues</p>
</td></tr>
<tr><td><code id="itis_native_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to
<code><a href="ritis.html#topic+jurisdiction">ritis::jurisdictional_origin()</a></code>,
<code><a href="ritis.html#topic+jurisdiction">ritis::jurisdiction_values()</a></code>, or
<code><a href="ritis.html#topic+jurisdiction">ritis::jurisdiction_origin_values()</a></code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get values
itis_native(what="values")

# Get origin values
itis_native(what="originvalues")

# Get values by tsn
itis_native(tsn=180543)
itis_native(tsn=c(180543,41074,36616))

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_refs'>Get references related to a ITIS TSN.</h2><span id='topic+itis_refs'></span>

<h3>Description</h3>

<p>Get references related to a ITIS TSN.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_refs(tsn, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_refs_+3A_tsn">tsn</code></td>
<td>
<p>One or more TSN's (taxonomic serial number) for a
taxonomic group (numeric)</p>
</td></tr>
<tr><td><code id="itis_refs_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to getpublicationsfromtsn</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
itis_refs(202385)
itis_refs(c(202385, 70340))

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_taxrank'>Retrieve taxonomic rank name from given TSN.</h2><span id='topic+itis_taxrank'></span>

<h3>Description</h3>

<p>Retrieve taxonomic rank name from given TSN.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_taxrank(query = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_taxrank_+3A_query">query</code></td>
<td>
<p>TSN for a taxonomic group (numeric). If query is left as default
(NULL), you get all possible rank names, and their TSN's (using function
<code><a href="ritis.html#topic+rank_names">ritis::rank_names()</a></code>. There is slightly different terminology for
Monera vs. Plantae vs. Fungi vs. Animalia vs. Chromista, so there are
separate terminologies for each group.</p>
</td></tr>
<tr><td><code id="itis_taxrank_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to <code><a href="ritis.html#topic+rank_name">ritis::rank_name()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can print messages by setting <code>verbose=FALSE</code>.
</p>


<h3>Value</h3>

<p>Taxonomic rank names or data.frame of all ranks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# All ranks
itis_taxrank()

# A single TSN
itis_taxrank(query=202385)

# Many TSN's
itis_taxrank(query=c(202385,183833,180543))

## End(Not run)
</code></pre>

<hr>
<h2 id='itis_terms'>Get ITIS terms, i.e., tsn's, authors, common names, and scientific names.</h2><span id='topic+itis_terms'></span>

<h3>Description</h3>

<p>Get ITIS terms, i.e., tsn's, authors, common names, and scientific names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itis_terms(query, what = "both", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="itis_terms_+3A_query">query</code></td>
<td>
<p>One or more common or scientific names, or partial names</p>
</td></tr>
<tr><td><code id="itis_terms_+3A_what">what</code></td>
<td>
<p>One of both (search common and scientific names), common
(search just common names), or scientific (search just scientific names)</p>
</td></tr>
<tr><td><code id="itis_terms_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to <code><a href="ritis.html#topic+terms">ritis::terms()</a></code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get terms searching both common and scientific names
itis_terms(query='bear')

# Get terms searching just common names
itis_terms(query='tarweed', "common")

# Get terms searching just scientific names
itis_terms(query='Poa annua', "scientific")

## End(Not run)
</code></pre>

<hr>
<h2 id='iucn_getname'>Get any matching IUCN species names</h2><span id='topic+iucn_getname'></span>

<h3>Description</h3>

<p>Get any matching IUCN species names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iucn_getname(name, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iucn_getname_+3A_name">name</code></td>
<td>
<p>character; taxon name</p>
</td></tr>
<tr><td><code id="iucn_getname_+3A_verbose">verbose</code></td>
<td>
<p>logical; should messages be printed?</p>
</td></tr>
<tr><td><code id="iucn_getname_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to <code><a href="#topic+iucn_summary">iucn_summary()</a></code>, note
that you'll need an API key.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Beware: IUCN functions can give back incorrect data. This isn't
our fault. We do our best to get you the correct data quickly, but sometimes
IUCN gives back the wrong data, and sometimes Global Names gives back the
wrong data. We will fix these as soon as possible. In the meantime, just
make sure that the data you get back is correct.
</p>


<h3>Value</h3>

<p>Character vector of names that matched in IUCN
</p>


<h3>See Also</h3>

<p><code><a href="#topic+iucn_summary">iucn_summary()</a></code> <code><a href="#topic+iucn_status">iucn_status()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
iucn_getname(name = "Cyanistes caeruleus")
iucn_getname(name = "Panthera uncia")

# not found in IUCN search
iucn_getname(name = "Acacia allenii")

## End(Not run)
</code></pre>

<hr>
<h2 id='iucn_id'>Get an ID for a IUCN listed taxon</h2><span id='topic+iucn_id'></span>

<h3>Description</h3>

<p>Get an ID for a IUCN listed taxon
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iucn_id(sciname, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iucn_id_+3A_sciname">sciname</code></td>
<td>
<p>character; Scientific name. Should be cleaned and in the
format <code style="white-space: pre;">&#8288;*&lt;Genus&gt; &lt;Species&gt;*&#8288;</code>. One or more.</p>
</td></tr>
<tr><td><code id="iucn_id_+3A_key">key</code></td>
<td>
<p>(character) required. you IUCN Redlist API key. See
<a href="rredlist.html#topic+rredlist-package">rredlist::rredlist-package</a> for help on authenticating with
IUCN Redlist</p>
</td></tr>
<tr><td><code id="iucn_id_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list (names are input taxa names) of one or more IUCN IDs.
Taxa that aren't found are silently dropped.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
iucn_id("Branta canadensis")
iucn_id("Branta bernicla")
iucn_id("Panthera uncia")
iucn_id("Lynx lynx")

# many names
iucn_id(c("Panthera uncia", "Lynx lynx"))

# many names, some not found
iucn_id(c("Panthera uncia", "Lynx lynx", "foo bar", "Gorilla gorilla gorilla"))

# a name not found
iucn_id("Foo bar")

## End(Not run)
</code></pre>

<hr>
<h2 id='iucn_status'>Extractor functions for <code>iucn</code>-class.</h2><span id='topic+iucn_status'></span>

<h3>Description</h3>

<p>Extractor functions for <code>iucn</code>-class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iucn_status(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iucn_status_+3A_x">x</code></td>
<td>
<p>an <code>iucn</code>-object as returned b<code>iucn_summary</code>ry</p>
</td></tr>
<tr><td><code id="iucn_status_+3A_...">...</code></td>
<td>
<p>Currently not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector with the status.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+iucn_summary">iucn_summary()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ia &lt;- iucn_summary(c("Panthera uncia", "Lynx lynx"))
iucn_status(ia)
## End(Not run)
</code></pre>

<hr>
<h2 id='iucn_summary'>Get a summary from the IUCN Red List</h2><span id='topic+iucn_summary'></span>

<h3>Description</h3>

<p>Get a summary from the IUCN Red List
(https://www.iucnredlist.org/).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iucn_summary(x, distr_detail = FALSE, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="iucn_summary_+3A_x">x</code></td>
<td>
<p>character; Scientific name. Should be cleaned and in the format
<code style="white-space: pre;">&#8288;*&lt;Genus&gt; &lt;Species&gt;*&#8288;</code>.</p>
</td></tr>
<tr><td><code id="iucn_summary_+3A_distr_detail">distr_detail</code></td>
<td>
<p>logical; If <code>TRUE</code>, the geographic distribution is
returned as a list of vectors corresponding to the different range types:
native, introduced, etc.</p>
</td></tr>
<tr><td><code id="iucn_summary_+3A_key">key</code></td>
<td>
<p>a Redlist API key, get one from
https://apiv3.iucnredlist.org/api/v3/token Required for <code>iucn_summary</code>.
Defaults to <code>NULL</code> in case you have your key stored (see <code style="white-space: pre;">&#8288;Redlist Authentication&#8288;</code> below).</p>
</td></tr>
<tr><td><code id="iucn_summary_+3A_...">...</code></td>
<td>
<p>curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Beware: IUCN functions can give back incorrect data. This isn't our
fault. We do our best to get you the correct data quickly, but sometimes
IUCN gives back the wrong data, and sometimes Global Names gives back the
wrong data. We will fix these as soon as possible. In the meantime, just
make sure that the data you get back is correct.
</p>
<p><code>iucn_summary</code> has a default method that errors when anything's passed in
that's not <code>character</code> or <code>iucn</code> class - a <code>iucn_summary.character</code> method
for when you pass in taxon names - and a <code>iucn_summary.iucn</code> method so you
can pass in iucn class objects as output from <code><a href="#topic+get_iucn">get_iucn()</a></code> or <code><a href="#topic+as.iucn">as.iucn()</a></code>.
If you already have IUCN IDs, coerce them to <code>iucn</code> class via <code>as.iucn(..., check = FALSE)</code>
</p>


<h3>Value</h3>

<p>A list (for every species one entry) of data returned by
<code><a href="rredlist.html#topic+rl_species_latest">rredlist::rl_species_latest()</a></code>.
</p>


<h3>Redlist Authentication</h3>

<p><code>iucn_summary</code> uses the new Redlist API for
searching for a IUCN ID, so we use the <code><a href="rredlist.html#topic+rl_species">rredlist::rl_species()</a></code> function internally.
This function requires an API key. Get the key at
https://apiv3.iucnredlist.org/api/v3/token, and pass it to the <code>key</code>
parameter, or store in your <code>.Renviron</code> file like
<code>IUCN_REDLIST_KEY=yourkey</code> or in your <code>.Rprofile</code> file like
<code>options(iucn_redlist_key="yourkey")</code>. We strongly encourage you to not
pass the key in the function call but rather store it in one of those two
files. This key will also set you up to use the <span class="pkg">rredlist</span> package.
</p>


<h3>Note</h3>

<p>Not all data types are available for every species
and NA is returned. <code><a href="#topic+iucn_status">iucn_status()</a></code> is an extractor function to easily
extract status into a vector.
</p>


<h3>Author(s)</h3>

<p>Eduard Szoecs, <a href="mailto:eduardszoecs@gmail.com">eduardszoecs@gmail.com</a>
</p>
<p>Philippe Marchand, <a href="mailto:marchand.philippe@gmail.com">marchand.philippe@gmail.com</a>
</p>
<p>Scott Chamberlain,
</p>
<p>Zachary S.L. Foster
</p>


<h3>See Also</h3>

<p><code><a href="#topic+iucn_status">iucn_status()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# if you send a taxon name, an IUCN API key is required
## here, the key is being detected from a .Rprofile file
## or .Renviron file, See "Redlist Authentication" above
iucn_summary("Lutra lutra")

ia &lt;- iucn_summary(c("Panthera uncia", "Lynx lynx"))
ia &lt;- iucn_summary(c("Panthera uncia", "Lynx lynx", "aaa"))
iucn_summary("Muntiacus rooseveltorum/truongsonensis")
iucn_summary(c("Muntiacus rooseveltorum/truongsonensis", "Lynx lynx"))

## get detailed distribution
iac &lt;- iucn_summary(x="Ara chloropterus", distr_detail = TRUE)
iac[[1]]$distr


# If you pass in an IUCN ID, you don't need to pass in a Redlist API Key
# extract status
iucn_status(iac)

## End(Not run)
</code></pre>

<hr>
<h2 id='key_helpers'>Helpers to set up authentication for the different providers.</h2><span id='topic+key_helpers'></span><span id='topic+use_tropicos'></span><span id='topic+use_entrez'></span><span id='topic+use_iucn'></span>

<h3>Description</h3>

<p>Sets up authentication to diverse providers by providing
the user a detailed prompt.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>use_tropicos()

use_entrez()

use_iucn()
</code></pre>


<h3>Details</h3>

<p>Key helpers
</p>


<h3><code>use_tropicos()</code></h3>

<p>Browses to Tropicos API key request URL and provides instruction on how to
store the key. After filling the form you will get the key soon, but
not immediately.
</p>


<h3><code>use_entrez()</code></h3>

<p>Browse NCBI Entrez to help make an API key request and provides instruction
on how to store the key. There's no direct URL to request a key, one first
needs to log in or register and then to generate a key from one's account.
</p>
<p>Note that NCBI Entrez doesn't require that you use an API key,
but you should get higher rate limit with a key, so do get one.
</p>


<h3><code>use_iucn()</code></h3>

<p>Browse IUCN Red List API key request URL and provides instruction on how
to store the key. This function wraps <code><a href="rredlist.html#topic+rl_use_iucn">rredlist::rl_use_iucn()</a></code>
from the <code>rredlist</code> package. After filling the form you will get
the key soon, but not immediately.
</p>


<h3>See Also</h3>

<p><a href="#topic+taxize-authentication">taxize-authentication</a>
</p>

<hr>
<h2 id='limited_print'>Print a subset of a character vector</h2><span id='topic+limited_print'></span>

<h3>Description</h3>

<p>Prints the start and end values for a character vector. The number of values
printed depend on the width of the screen by default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>limited_print(
  chars,
  prefix = "",
  max_chars = getOption("width") - nchar(prefix) - 5,
  type = "message"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="limited_print_+3A_chars">chars</code></td>
<td>
<p>(<code>character</code>) What to print.</p>
</td></tr>
<tr><td><code id="limited_print_+3A_prefix">prefix</code></td>
<td>
<p>(<code>character</code> of length 1) What to print before
<code>chars</code>, on the same line.</p>
</td></tr>
<tr><td><code id="limited_print_+3A_max_chars">max_chars</code></td>
<td>
<p>(<code>numeric</code> of length 1) The maximum number of
characters to print.</p>
</td></tr>
<tr><td><code id="limited_print_+3A_type">type</code></td>
<td>
<p>(<code>"error"</code>, <code>"warning"</code>, <code>"message"</code>, <code>"cat"</code>, <code>"print"</code>, <code>"silent"</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>NULL</code>
</p>

<hr>
<h2 id='lowest_common'>Retrieve the lowest common taxon and rank for a given taxon name or ID</h2><span id='topic+lowest_common'></span><span id='topic+lowest_common.default'></span><span id='topic+lowest_common.uid'></span><span id='topic+lowest_common.tsn'></span><span id='topic+lowest_common.gbifid'></span><span id='topic+lowest_common.tolid'></span>

<h3>Description</h3>

<p>Retrieve the lowest common taxon and rank for a given taxon name or ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lowest_common(...)

## Default S3 method:
lowest_common(
  sci_id,
  db = NULL,
  rows = NA,
  class_list = NULL,
  low_rank = NULL,
  x = NULL,
  ...
)

## S3 method for class 'uid'
lowest_common(sci_id, class_list = NULL, low_rank = NULL, ...)

## S3 method for class 'tsn'
lowest_common(sci_id, class_list = NULL, low_rank = NULL, ...)

## S3 method for class 'gbifid'
lowest_common(sci_id, class_list = NULL, low_rank = NULL, ...)

## S3 method for class 'tolid'
lowest_common(sci_id, class_list = NULL, low_rank = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lowest_common_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+get_tsn">get_tsn()</a></code>, <code><a href="#topic+get_uid">get_uid()</a></code>,
<code><a href="#topic+get_gbifid">get_gbifid()</a></code>, <code><a href="#topic+get_tolid">get_tolid()</a></code></p>
</td></tr>
<tr><td><code id="lowest_common_+3A_sci_id">sci_id</code></td>
<td>
<p>Vector of taxa names (character) or id (character or numeric) to
query.</p>
</td></tr>
<tr><td><code id="lowest_common_+3A_db">db</code></td>
<td>
<p>character; database to query. either <code>ncbi</code>, <code>itis</code>, <code>gbif</code>,
<code>tol</code>. If using ncbi, we recommend getting an API key;
see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="lowest_common_+3A_rows">rows</code></td>
<td>
<p>(numeric) Any number from 1 to infinity. If the default NA,
all rows are considered. Note that this parameter is ignored if you pass in
a taxonomic id of any of the acceptable classes: tsn, gbifid, tolid.
NCBI has a method for this function but rows doesn't work.</p>
</td></tr>
<tr><td><code id="lowest_common_+3A_class_list">class_list</code></td>
<td>
<p>(list) A list of classifications, as returned from
<code><a href="#topic+classification">classification()</a></code></p>
</td></tr>
<tr><td><code id="lowest_common_+3A_low_rank">low_rank</code></td>
<td>
<p>(character) taxonomic rank to return, of length 1</p>
</td></tr>
<tr><td><code id="lowest_common_+3A_x">x</code></td>
<td>
<p>Deprecated, see <code>sci_id</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>NA when no match, or a data.frame with columns
</p>

<ul>
<li><p> name
</p>
</li>
<li><p> rank
</p>
</li>
<li><p> id
</p>
</li></ul>



<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication
</p>


<h3>Author(s)</h3>

<p>Jimmy O'Donnell <a href="mailto:jodonnellbio@gmail.com">jodonnellbio@gmail.com</a>
Scott Chamberlain
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
id &lt;- c("9031", "9823", "9606", "9470")
id_class &lt;- classification(id, db = 'ncbi')
lowest_common(id[2:4], db = "ncbi")
lowest_common(id[2:4], db = "ncbi", low_rank = 'class')
lowest_common(id[2:4], db = "ncbi", low_rank = 'family')
lowest_common(id[2:4], class_list = id_class)
lowest_common(id[2:4], class_list = id_class, low_rank = 'class')
lowest_common(id[2:4], class_list = id_class, low_rank = 'family')

# TOL
taxa &lt;- c("Angraecum sesquipedale", "Dracula vampira",
  "Masdevallia coccinea")
(cls &lt;- classification(taxa, db = "tol"))
lowest_common(taxa, db = "tol", class_list = cls)
lowest_common(get_tolid(taxa), class_list = cls)
xx &lt;- get_tolid(taxa)
lowest_common(xx, class_list = cls)

spp &lt;- c("Sus scrofa", "Homo sapiens", "Nycticebus coucang")
lowest_common(spp, db = "ncbi")
lowest_common(get_uid(spp))

lowest_common(spp, db = "itis")
lowest_common(get_tsn(spp))

gbifid &lt;- c("2704179", "3119195")
lowest_common(gbifid, db = "gbif")

spp &lt;- c("Poa annua", "Helianthus annuus")
lowest_common(spp, db = "gbif")
lowest_common(get_gbifid(spp))

cool_orchid &lt;- c("Angraecum sesquipedale", "Dracula vampira",
  "Masdevallia coccinea")
orchid_ncbi &lt;- get_uid(cool_orchid)
orchid_gbif &lt;- get_gbifid(cool_orchid)

cool_orchids2 &lt;- c("Domingoa haematochila", "Gymnadenia conopsea",
  "Masdevallia coccinea")
orchid_itis &lt;- get_tsn(cool_orchids2)

orchid_hier_ncbi &lt;- classification(orchid_ncbi, db = 'ncbi')
orchid_hier_gbif &lt;- classification(orchid_gbif, db = 'gbif')
orchid_hier_itis &lt;- classification(orchid_itis, db = 'itis')

lowest_common(orchid_ncbi, low_rank = 'class')
lowest_common(orchid_ncbi, class_list = orchid_hier_ncbi,
  low_rank = 'class')
lowest_common(orchid_gbif, low_rank = 'class')
lowest_common(orchid_gbif, orchid_hier_gbif, low_rank = 'class')
lowest_common(get_uid(cool_orchid), low_rank = 'class')
lowest_common(get_uid(cool_orchid), low_rank = 'family')

lowest_common(orchid_ncbi, class_list = orchid_hier_ncbi,
  low_rank = 'subfamily')
lowest_common(orchid_gbif, class_list = orchid_hier_gbif,
  low_rank = 'subfamily')

lowest_common(orchid_itis, class_list = orchid_hier_itis,
  low_rank = 'class')

## Pass in sci. names
nms &lt;- c("Angraecum sesquipedale", "Dracula vampira", "Masdevallia coccinea")
lowest_common(x = nms, db = "ncbi")
lowest_common(x = nms, db = "gbif")

## End(Not run)
</code></pre>

<hr>
<h2 id='names_list'>Get a random vector of species names.</h2><span id='topic+names_list'></span>

<h3>Description</h3>

<p>Family and order names come from the APG plant names list. Genus and
species names come from Theplantlist.org.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>names_list(rank = "genus", size = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="names_list_+3A_rank">rank</code></td>
<td>
<p>(character) Taxonomic rank, one of species, genus (default),
family, order</p>
</td></tr>
<tr><td><code id="names_list_+3A_size">size</code></td>
<td>
<p>(integer/numeric) Number of names to get. Maximum depends on
the rank</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector of taxonomic names
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>Examples</h3>

<pre><code class='language-R'>names_list()
names_list('species')
names_list('genus')
names_list('family')
names_list('order')
names_list('order', 2)
names_list('order', 15)

# You can get a lot of genus or species names if you want
nrow(theplantlist)
names_list('genus', 500)
</code></pre>

<hr>
<h2 id='nbn_classification'>Search UK National Biodiversity Network database for
taxonomic classification</h2><span id='topic+nbn_classification'></span>

<h3>Description</h3>

<p>Search UK National Biodiversity Network database for
taxonomic classification
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbn_classification(id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nbn_classification_+3A_id">id</code></td>
<td>
<p>(character) An NBN identifier.</p>
</td></tr>
<tr><td><code id="nbn_classification_+3A_...">...</code></td>
<td>
<p>Further args passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain,
</p>


<h3>References</h3>

<p>https://api.nbnatlas.org/
</p>


<h3>See Also</h3>

<p>Other nbn: 
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+nbn_search">nbn_search</a>()</code>,
<code><a href="#topic+nbn_synonyms">nbn_synonyms</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
nbn_classification(id="NHMSYS0000376773")

# get id first, then pass to this fxn
id &lt;- get_nbnid("Zootoca vivipara", rec_only = TRUE, rank = "Species")
nbn_classification(id)

nbn_classification(id="NHMSYS0000502940", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='nbn_search'>Search UK National Biodiversity Network</h2><span id='topic+nbn_search'></span>

<h3>Description</h3>

<p>Search UK National Biodiversity Network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbn_search(
  sci_com,
  fq = NULL,
  order = NULL,
  sort = NULL,
  start = 0,
  rows = 25,
  facets = NULL,
  q = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nbn_search_+3A_sci_com">sci_com</code></td>
<td>
<p>(character) The query terms(s), a scientific or common name</p>
</td></tr>
<tr><td><code id="nbn_search_+3A_fq">fq</code></td>
<td>
<p>(character) Filters to be applied to the original query. These
are additional params of the form fq=INDEXEDFIELD:VALUE e.g.
fq=rank:kingdom. See https://species-ws.nbnatlas.org/indexFields for all
the fields that are queryable.</p>
</td></tr>
<tr><td><code id="nbn_search_+3A_order">order</code></td>
<td>
<p>(character) Supports &quot;asc&quot; or &quot;desc&quot;</p>
</td></tr>
<tr><td><code id="nbn_search_+3A_sort">sort</code></td>
<td>
<p>(character) The indexed field to sort by</p>
</td></tr>
<tr><td><code id="nbn_search_+3A_start">start</code></td>
<td>
<p>(integer) Record offset, to enable paging</p>
</td></tr>
<tr><td><code id="nbn_search_+3A_rows">rows</code></td>
<td>
<p>(integer) Number of records to return</p>
</td></tr>
<tr><td><code id="nbn_search_+3A_facets">facets</code></td>
<td>
<p>(list) Comma separated list of the fields to create facets
on e.g. facets=basis_of_record.</p>
</td></tr>
<tr><td><code id="nbn_search_+3A_q">q</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="nbn_search_+3A_...">...</code></td>
<td>
<p>Further args passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with slots for metadata (<code>meta</code>) with list of response
attributes, and data (<code>data</code>) with a data.frame of results
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain,
</p>


<h3>References</h3>

<p>https://api.nbnatlas.org/
</p>


<h3>See Also</h3>

<p>Other nbn: 
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+nbn_classification">nbn_classification</a>()</code>,
<code><a href="#topic+nbn_synonyms">nbn_synonyms</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
x &lt;- nbn_search(sci_com = "Vulpes")
x$meta$totalRecords
x$meta$pageSize
x$meta$urlParameters
x$meta$queryTitle
head(x$data)

nbn_search(sci_com = "blackbird", start = 4)

# debug curl stuff
nbn_search(sci_com = "blackbird", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='nbn_synonyms'>Return all synonyms for a taxon name with a given id from NBN</h2><span id='topic+nbn_synonyms'></span>

<h3>Description</h3>

<p>Return all synonyms for a taxon name with a given id from NBN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbn_synonyms(id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nbn_synonyms_+3A_id">id</code></td>
<td>
<p>the taxon identifier code</p>
</td></tr>
<tr><td><code id="nbn_synonyms_+3A_...">...</code></td>
<td>
<p>Further args passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>References</h3>

<p>https://api.nbnatlas.org/
</p>


<h3>See Also</h3>

<p>Other nbn: 
<code><a href="#topic+get_nbnid">get_nbnid</a>()</code>,
<code><a href="#topic+nbn_classification">nbn_classification</a>()</code>,
<code><a href="#topic+nbn_search">nbn_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
nbn_synonyms(id = 'NHMSYS0001501147')
nbn_synonyms(id = 'NHMSYS0000456036')

# none
nbn_synonyms(id = 'NHMSYS0000502940')

## End(Not run)
</code></pre>

<hr>
<h2 id='ncbi_children'>Search NCBI for children of a taxon</h2><span id='topic+ncbi_children'></span>

<h3>Description</h3>

<p>Search the NCBI Taxonomy database for uids of children of taxa.
Taxa can be referenced by name or uid. Referencing by name is faster
</p>
<p>In a few cases, different taxa have the same name (e.g. Satyrium; see
examples). If one of these are searched for then the children of both taxa
will be returned. This can be avoided by using a uid instead of the name or
specifying an ancestor. If an ancestor is provided, only children of both
the taxon and its ancestor are returned. This will only fail if there are
two taxa with the same name and the same specified ancestor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_children(
  name = NULL,
  id = NULL,
  start = 0,
  max_return = 1000,
  ancestor = NULL,
  out_type = c("summary", "uid"),
  ambiguous = FALSE,
  key = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ncbi_children_+3A_name">name</code></td>
<td>
<p>(<code>character</code>) The string to search for. Only exact matches
found the name given will be returned. Not compatible with <code>id</code>.</p>
</td></tr>
<tr><td><code id="ncbi_children_+3A_id">id</code></td>
<td>
<p>(<code>character</code>/<code>numeric</code>) The uid to search for. Not compatible with
<code>name</code>.</p>
</td></tr>
<tr><td><code id="ncbi_children_+3A_start">start</code></td>
<td>
<p>The first record to return. If omitted, the results are
returned from the first record (start=0).</p>
</td></tr>
<tr><td><code id="ncbi_children_+3A_max_return">max_return</code></td>
<td>
<p>(<code style="white-space: pre;">&#8288;numeric; length=1&#8288;</code>) The maximum number of children
to return.</p>
</td></tr>
<tr><td><code id="ncbi_children_+3A_ancestor">ancestor</code></td>
<td>
<p>(<code>character</code>) The ancestor of the taxon being searched
for. This is useful if there could be more than one taxon with the same
name. Has no effect if <code>id</code> is used.</p>
</td></tr>
<tr><td><code id="ncbi_children_+3A_out_type">out_type</code></td>
<td>
<p>(character) Currently either <code>"summary"</code> or <code>"uid"</code>:
</p>

<ul>
<li> <p><code>summary</code> The output is a list of <code>data.frame</code> with children uid, name,
and rank.
</p>
</li>
<li> <p><code>uid</code> A list of character vectors of children uids
</p>
</li></ul>
</td></tr>
<tr><td><code id="ncbi_children_+3A_ambiguous">ambiguous</code></td>
<td>
<p><code style="white-space: pre;">&#8288;logical; length 1&#8288;</code> If <code>FALSE</code>, children taxa
with words like &quot;unclassified&quot;, &quot;unknown&quot;, &quot;uncultured&quot;, or &quot;sp.&quot; are
removed from the output. NOTE: This option only applies when
<code>out_type= "summary"</code>.</p>
</td></tr>
<tr><td><code id="ncbi_children_+3A_key">key</code></td>
<td>
<p>(character) NCBI Entrez API key. optional. See Details.</p>
</td></tr>
<tr><td><code id="ncbi_children_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output type depends on the value of the <code>out_type</code>
parameter. Taxa that cannot be found will result in <code>NA</code>s and a lack
of children results in an empty data structure.
</p>


<h3>Authentication</h3>

<p>See <code><a href="#topic+taxize-authentication">taxize-authentication()</a></code> for help on authentication.
We strongly recommend getting an API key
</p>


<h3>HTTP version</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>Rate limits</h3>

<p>In case you run into errors due to your rate limit being exceeded, see
<code><a href="#topic+taxize_options">taxize_options()</a></code>, where you can set <code>ncbi_sleep</code>.
</p>


<h3>Author(s)</h3>

<p>Zachary Foster <a href="mailto:zacharyfoster1989@gmail.com">zacharyfoster1989@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ncbi_get_taxon_summary">ncbi_get_taxon_summary()</a></code>, <code><a href="#topic+children">children()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ncbi_children(name="Satyrium") #Satyrium is the name of two different genera
ncbi_children(name="Satyrium", ancestor="Eumaeini") # A genus of butterflies
ncbi_children(name="Satyrium", ancestor="Orchidaceae") # A genus of orchids
ncbi_children(id="266948") #"266948" is the uid for the butterfly genus
ncbi_children(id="62858") #"62858" is the uid for the orchid genus

# use curl options
ncbi_children(name="Satyrium", ancestor="Eumaeini", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ncbi_downstream'>Retrieve all taxa names downstream in hierarchy for NCBI</h2><span id='topic+ncbi_downstream'></span>

<h3>Description</h3>

<p>Retrieve all taxa names downstream in hierarchy for NCBI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_downstream(id, downto, intermediate = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ncbi_downstream_+3A_id">id</code></td>
<td>
<p>(numeric/integer) An NCBI taxonomic identifier</p>
</td></tr>
<tr><td><code id="ncbi_downstream_+3A_downto">downto</code></td>
<td>
<p>The taxonomic level you want to go down to. See examples below.
The taxonomic level IS case sensitive, and you do have to spell it
correctly. See <code>data(rank_ref)</code> for spelling.</p>
</td></tr>
<tr><td><code id="ncbi_downstream_+3A_intermediate">intermediate</code></td>
<td>
<p>(logical) If <code>TRUE</code>, return a list of length two
with target taxon rank names, with additional list of data.frame's of
intermediate taxonomic groups. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="ncbi_downstream_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="#topic+ncbi_children">ncbi_children()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data.frame of taxonomic information downstream to family from e.g.,
Order, Class, etc., or if <code>intermediate=TRUE</code>, list of length two,
with target taxon rank names, and intermediate names.
</p>


<h3>No Rank</h3>

<p>A sticky point with NCBI is that they can have designation for taxonomic
rank of &quot;No Rank&quot;. So we have no way of programatically knowing what to
do with that taxon. Of course one can manually look at a name and perhaps
know what it is, or look it up on the web - but we can't do anything
programatically. So, no rank things will sometimes be missing.
</p>


<h3>Authentication</h3>

<p>See <code><a href="#topic+taxize-authentication">taxize-authentication()</a></code> for help on authentication.
We strongly recommend getting an API key
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## genus Apis
ncbi_downstream(id = 7459, downto="species")

## get intermediate taxa as a separate object
ncbi_downstream(id = 7459, downto="species", intermediate = TRUE)

## Lepidoptera
ncbi_downstream(id = 7088, downto="superfamily")

## families in the ferns (Moniliformopses)
(id &lt;- get_uid("Moniliformopses"))
ncbi_downstream(id = id, downto = "order")

## End(Not run)
</code></pre>

<hr>
<h2 id='ncbi_get_taxon_summary'>NCBI taxon information from uids</h2><span id='topic+ncbi_get_taxon_summary'></span>

<h3>Description</h3>

<p>Downloads summary taxon information from the NCBI taxonomy
databases for a set of taxonomy UIDs using eutils esummary.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_get_taxon_summary(id, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ncbi_get_taxon_summary_+3A_id">id</code></td>
<td>
<p>(character) NCBI taxonomy uids to retrieve information for.
See Details.</p>
</td></tr>
<tr><td><code id="ncbi_get_taxon_summary_+3A_key">key</code></td>
<td>
<p>(character) NCBI Entrez API key. optional. See Details.</p>
</td></tr>
<tr><td><code id="ncbi_get_taxon_summary_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>If your input vector or list of NCBI IDs is longer than about
2500 characters (use <code>nchar(paste(ids, collapse = "+"))</code>), split
the list up into chunks since at about that number of characters you
will run into the HTTP 414 error &quot;Request-URI Too Long&quot;.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with the following columns:
</p>

<ul>
<li> <p><code>uid</code> The uid queried for
</p>
</li>
<li> <p><code>name</code> The name of the taxon; a binomial name if the taxon is of rank species
</p>
</li>
<li> <p><code>rank</code> The taxonomic rank (e.g. 'Genus')
</p>
</li></ul>



<h3>HTTP version</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication.
We strongly recommend getting an API key
</p>


<h3>Author(s)</h3>

<p>Zachary Foster <a href="mailto:zacharyfoster1989@Sgmail.com">zacharyfoster1989@Sgmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ncbi_get_taxon_summary(c(1430660, 4751))

# use curl options
ncbi_get_taxon_summary(c(1430660, 4751), verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ncbi_getbyid'>Retrieve gene sequences from NCBI by accession number.</h2><span id='topic+ncbi_getbyid'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_getbyid(...)
</code></pre>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>

<hr>
<h2 id='ncbi_getbyname'>Retrieve gene sequences from NCBI by taxon name and gene names.</h2><span id='topic+ncbi_getbyname'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_getbyname(...)
</code></pre>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>

<hr>
<h2 id='ncbi_search'>Search for gene sequences available for taxa from NCBI.</h2><span id='topic+ncbi_search'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_search(...)
</code></pre>

<hr>
<h2 id='phylomatic_format'>Get family names to make Phylomatic input object, and output input string
to Phylomatic for use in the function phylomatic_tree.</h2><span id='topic+phylomatic_format'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylomatic_format(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylomatic_format_+3A_...">...</code></td>
<td>
<p>Parameters, ignored</p>
</td></tr>
</table>

<hr>
<h2 id='phylomatic_tree'>Query Phylomatic for a phylogenetic tree.</h2><span id='topic+phylomatic_tree'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylomatic_tree(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylomatic_tree_+3A_...">...</code></td>
<td>
<p>Parameters, ignored</p>
</td></tr>
</table>

<hr>
<h2 id='ping'>Ping an API used in taxize to see if it's working.</h2><span id='topic+ping'></span><span id='topic+col_ping'></span><span id='topic+eol_ping'></span><span id='topic+itis_ping'></span><span id='topic+ncbi_ping'></span><span id='topic+tropicos_ping'></span><span id='topic+nbn_ping'></span><span id='topic+gbif_ping'></span><span id='topic+bold_ping'></span><span id='topic+ipni_ping'></span><span id='topic+vascan_ping'></span><span id='topic+fg_ping'></span>

<h3>Description</h3>

<p>Ping an API used in taxize to see if it's working.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>col_ping(what = "status", ...)

eol_ping(what = "status", ...)

itis_ping(what = "status", ...)

ncbi_ping(what = "status", key = NULL, ...)

tropicos_ping(what = "status", ...)

nbn_ping(what = "status", ...)

gbif_ping(what = "status", ...)

bold_ping(what = "status", ...)

ipni_ping(what = "status", ...)

vascan_ping(what = "status", ...)

fg_ping(what = "status", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ping_+3A_what">what</code></td>
<td>
<p>(character) One of status (default), content, or an HTTP status
code. If status, we just check that the HTTP status code is 200, or similar
signifying the service is up. If content, we do a simple, quick check to
determine if returned content matches what's expected. If an HTTP status
code, it must match an appropriate code. See <code><a href="#topic+status_codes">status_codes()</a></code>.</p>
</td></tr>
<tr><td><code id="ping_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
<tr><td><code id="ping_+3A_key">key</code></td>
<td>
<p>(character) NCBI Entrez API key. optional. See <code><a href="#topic+get_uid">get_uid()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For ITIS, see <a href="ritis.html#topic+description">ritis::description</a>, which provides number of
scientific and common names in a character string.
</p>


<h3>Value</h3>

<p>A logical, TRUE or FALSE
</p>


<h3>HTTP version for NCBI requests</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
col_ping()
col_ping("content")
col_ping(200)
col_ping("200")
col_ping(204)

itis_ping()
eol_ping()
ncbi_ping()
tropicos_ping()
nbn_ping()

gbif_ping()
gbif_ping(200)

bold_ping()
bold_ping(200)
bold_ping("content")

ipni_ping()
ipni_ping(200)
ipni_ping("content")

vascan_ping()
vascan_ping(200)
vascan_ping("content")

# curl options
vascan_ping(verbose = TRUE)
eol_ping(500, verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='plantGenusNames'>Vector of plant genus names from ThePlantList</h2><span id='topic+plantGenusNames'></span>

<h3>Description</h3>

<p>These names are from http://www.theplantlist.org, and are a
randomly chosen subset of genera names for the purpose of having some
names to play with for examples in this package.
</p>


<h3>Format</h3>

<p>A vector of length 793
</p>


<h3>Source</h3>

<p>http://www.theplantlist.org
</p>

<hr>
<h2 id='plantminer'>Search for taxonomy data from Plantminer.com</h2><span id='topic+plantminer'></span>

<h3>Description</h3>

<p>Search for taxonomy data from Plantminer.com
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plantminer(plants, from = "tpl", messages = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plantminer_+3A_plants">plants</code></td>
<td>
<p>(character) Vector of plant species names. Required.</p>
</td></tr>
<tr><td><code id="plantminer_+3A_from">from</code></td>
<td>
<p>(character) One of tpl (for theplantlist.com data), or
flora (for Brazilian Flora Checklist). Required. Default: <code>tpl</code></p>
</td></tr>
<tr><td><code id="plantminer_+3A_messages">messages</code></td>
<td>
<p>(logical) informative messages or not. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plantminer_+3A_...">...</code></td>
<td>
<p>curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame of results.
</p>


<h3>Note</h3>

<p>you used to need an API key for Plantminer; it's no longer needed
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# A single taxon
plantminer("Ocotea pulchella")

# Many taxa
plants &lt;- c("Myrcia lingua", "Myrcia bella", "Ocotea pulchella",
		"Miconia", "Coffea arabica var. amarella", "Bleh")
plantminer(plants)

# By deafult, tpl is used, for Theplantlist data,
# toggle the from parameter here
plantminer("Ocotea pulchella", from = "flora")

## End(Not run)
</code></pre>

<hr>
<h2 id='plantNames'>Vector of plant species (genus - specific epithet) names from ThePlantList</h2><span id='topic+plantNames'></span>

<h3>Description</h3>

<p>These names are from http://www.theplantlist.org, and are a
randomly chosen subset of names of the form genus/specific epithet
for the purpose of having some names to play with for examples in
this package.
</p>


<h3>Format</h3>

<p>A vector of length 1182
</p>


<h3>Source</h3>

<p>http://www.theplantlist.org
</p>

<hr>
<h2 id='pow_lookup'>Lookup taxa in Kew's Plants of the World</h2><span id='topic+pow_lookup'></span>

<h3>Description</h3>

<p>Lookup taxa in Kew's Plants of the World
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pow_lookup(id, include = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pow_lookup_+3A_id">id</code></td>
<td>
<p>(character) taxon id. required</p>
</td></tr>
<tr><td><code id="pow_lookup_+3A_include">include</code></td>
<td>
<p>(character) vector of additional fields to include in
results. options include 'distribution' and 'descriptions'. optional</p>
</td></tr>
<tr><td><code id="pow_lookup_+3A_...">...</code></td>
<td>
<p>Further args passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other pow: 
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+pow_search">pow_search</a>()</code>,
<code><a href="#topic+pow_synonyms">pow_synonyms</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pow_lookup(id = 'urn:lsid:ipni.org:names:320035-2')
pow_lookup(id = 'urn:lsid:ipni.org:names:320035-2',
  include = "distribution")
pow_lookup(id = 'urn:lsid:ipni.org:names:320035-2',
  include = c("distribution", "descriptions"))

## End(Not run)
</code></pre>

<hr>
<h2 id='pow_search'>Search Kew's Plants of the World</h2><span id='topic+pow_search'></span>

<h3>Description</h3>

<p>Search Kew's Plants of the World
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pow_search(sci_com, limit = 100, cursor = "*", sort = NULL, q = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pow_search_+3A_sci_com">sci_com</code></td>
<td>
<p>(character) query terms, scientific or common name</p>
</td></tr>
<tr><td><code id="pow_search_+3A_limit">limit</code></td>
<td>
<p>(integer) Number of records to return. default: 100</p>
</td></tr>
<tr><td><code id="pow_search_+3A_cursor">cursor</code></td>
<td>
<p>(character) cursor string</p>
</td></tr>
<tr><td><code id="pow_search_+3A_sort">sort</code></td>
<td>
<p>(character) The field to sort by and sort order separted with
underscore, e.g., <code>sort="name_desc"</code></p>
</td></tr>
<tr><td><code id="pow_search_+3A_q">q</code></td>
<td>
<p>Deprecated, see <code>sci_com</code></p>
</td></tr>
<tr><td><code id="pow_search_+3A_...">...</code></td>
<td>
<p>Further args passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with slots for metadata (<code>meta</code>) with list of response
attributes, and data (<code>data</code>) with a data.frame of results
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain,
</p>


<h3>References</h3>

<p>https://powo.science.kew.org/
</p>


<h3>See Also</h3>

<p>Other pow: 
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+pow_lookup">pow_lookup</a>()</code>,
<code><a href="#topic+pow_synonyms">pow_synonyms</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
x &lt;- pow_search(sci_com = "Quercus")
x$meta
x$meta$totalResults
x$meta$perPage
x$meta$totalPages
x$meta$page
x$meta$cursor
head(x$data)

# pagination
pow_search(sci_com = "sunflower", limit = 2)

# debug curl stuff
invisible(pow_search(sci_com = "Helianthus annuus", verbose = TRUE))

# sort
desc &lt;- pow_search(sci_com = "Helianthus", sort = "name_desc")
desc$data$name
asc &lt;- pow_search(sci_com = "Helianthus", sort = "name_asc")
asc$data$name

## End(Not run)
</code></pre>

<hr>
<h2 id='pow_synonyms'>Lookup synonyms in Kew's Plants of the World</h2><span id='topic+pow_synonyms'></span>

<h3>Description</h3>

<p>Lookup synonyms in Kew's Plants of the World
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pow_synonyms(id, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pow_synonyms_+3A_id">id</code></td>
<td>
<p>(character) taxon id. required</p>
</td></tr>
<tr><td><code id="pow_synonyms_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="#topic+pow_lookup">pow_lookup()</a></code></p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other pow: 
<code><a href="#topic+get_pow">get_pow</a>()</code>,
<code><a href="#topic+pow_lookup">pow_lookup</a>()</code>,
<code><a href="#topic+pow_search">pow_search</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pow_synonyms(id = 'urn:lsid:ipni.org:names:320035-2')
pow_synonyms(id = 'urn:lsid:ipni.org:names:358881-1')
pow_synonyms(id = 'urn:lsid:ipni.org:names:359855-1')

## End(Not run)
</code></pre>

<hr>
<h2 id='progressor'>methods for preparing/printing info for prompts for <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions</h2><span id='topic+progressor'></span>

<h3>Description</h3>

<p>methods for preparing/printing info for prompts for <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions
</p>
<p>methods for preparing/printing info for prompts for <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions
</p>


<h3>Public fields</h3>

<div class="r6-fields">

<dl>
<dt><code>total</code></dt><dd><p>(integer) x</p>
</dd>
<dt><code>found</code></dt><dd><p>(integer) list of results when name found</p>
</dd>
<dt><code>not_found</code></dt><dd><p>(integer) list of results when name not found</p>
</dd>
<dt><code>done</code></dt><dd><p>(integer) x</p>
</dd>
<dt><code>suppress</code></dt><dd><p>(integer) x</p>
</dd>
</dl>

</div>


<h3>Active bindings</h3>

<div class="r6-active-bindings">

<dl>
<dt><code>p</code></dt><dd><p>(integer) percent done</p>
</dd>
<dt><code>d</code></dt><dd><p>(integer) number done</p>
</dd>
</dl>

</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-progressor-new"><code>progressor$new()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-completed"><code>progressor$completed()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-completed_found"><code>progressor$completed_found()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-completed_not_found"><code>progressor$completed_not_found()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-prog_start"><code>progressor$prog_start()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-prog"><code>progressor$prog()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-prog_found"><code>progressor$prog_found()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-prog_not_found"><code>progressor$prog_not_found()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-prog_summary"><code>progressor$prog_summary()</code></a>
</p>
</li>
<li> <p><a href="#method-progressor-clone"><code>progressor$clone()</code></a>
</p>
</li></ul>


<hr>
<a id="method-progressor-new"></a>



<h4>Method <code>new()</code></h4>

<p>Create a new <code>progressor</code> object
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$new(items, suppress = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>items</code></dt><dd><p>(character) xxx</p>
</dd>
<dt><code>suppress</code></dt><dd><p>(logical) suppress messages. default: <code>FALSE</code></p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A new <code>progressor</code> object
</p>


<hr>
<a id="method-progressor-completed"></a>



<h4>Method <code>completed()</code></h4>

<p>add results to found or not found
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$completed(name, att)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>name</code></dt><dd><p>(character) vector of names</p>
</dd>
<dt><code>att</code></dt><dd><p>(character) one of &quot;found&quot; or &quot;not found&quot;</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>nothing returned; adds to <code style="white-space: pre;">&#8288;$found&#8288;</code> or <code style="white-space: pre;">&#8288;$not_found&#8288;</code>
</p>


<hr>
<a id="method-progressor-completed_found"></a>



<h4>Method <code>completed_found()</code></h4>

<p>add to found results
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$completed_found(name)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>name</code></dt><dd><p>(character) vector of taxon names</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>nothing returned; adds to <code style="white-space: pre;">&#8288;$found&#8288;</code>
</p>


<hr>
<a id="method-progressor-completed_not_found"></a>



<h4>Method <code>completed_not_found()</code></h4>

<p>add to not found results
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$completed_not_found(name)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>name</code></dt><dd><p>(character) vector of taxon names</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>nothing returned; adds to <code style="white-space: pre;">&#8288;$not_found&#8288;</code>
</p>


<hr>
<a id="method-progressor-prog_start"></a>



<h4>Method <code>prog_start()</code></h4>

<p>print messages of total queries to do, and
percent completed
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$prog_start()</pre></div>


<hr>
<a id="method-progressor-prog"></a>



<h4>Method <code>prog()</code></h4>

<p>prints message of found or not found using packages
cli and crayon
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$prog(att)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>att</code></dt><dd><p>(character) one of &quot;found&quot; or &quot;not found&quot;</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>messages
</p>


<hr>
<a id="method-progressor-prog_found"></a>



<h4>Method <code>prog_found()</code></h4>

<p>prints found message using packages cli and crayon
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$prog_found()</pre></div>



<h5>Returns</h5>

<p>messages
</p>


<hr>
<a id="method-progressor-prog_not_found"></a>



<h4>Method <code>prog_not_found()</code></h4>

<p>prints not found message using packages cli and crayon
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$prog_not_found()</pre></div>



<h5>Returns</h5>

<p>messages
</p>


<hr>
<a id="method-progressor-prog_summary"></a>



<h4>Method <code>prog_summary()</code></h4>

<p>prints summary at end of result with total found and
not found
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$prog_summary()</pre></div>



<h5>Returns</h5>

<p>messages
</p>


<hr>
<a id="method-progressor-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>progressor$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>




<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# nms &lt;- c("Quercus", "Sasdsfasdf")
# x &lt;- progressor$new(items = nms)
# x
# x$prog_start()

# x$completed(nms[1], "found")
# x$prog_found()

# x$completed(nms[2], "not found")
# x$prog_not_found()

# x$prog_summary()

# suppress cli::cat_line
# x &lt;- progressor$new(items = nms, suppress = TRUE)
# x$prog_summary()

## End(Not run)
</code></pre>

<hr>
<h2 id='rank_ref'>Lookup-table for IDs of taxonomic ranks</h2><span id='topic+rank_ref'></span>

<h3>Description</h3>

<p>data.frame of 46 rows, with 2 columns:
</p>

<ul>
<li><p> rankid - a numeric rank id, consecutive
</p>
</li>
<li><p> ranks - a comma separated vector of names that are considered
equal to one another within the row
</p>
</li></ul>



<h3>Details</h3>

<p>We use this data.frame to do data sorting/filtering based on the ordering
of ranks.
</p>
<p>Please let us know if there is a rank that occurs from one of the data
sources <span class="pkg">taxize</span> that we don't have in <code>rank_ref</code> dataset.
</p>
<p>Let us know if you disagree with the ordering of ranks.
</p>
<p>Note that <code>rankid</code> 280 are essentially &quot;genetic variants&quot;; placed just above
'unspecified' to denote they're not without rank, but they're not
really taxonomic ranks either. As far as I know there's no way
to delineate among these &quot;genetic variant&quot; types.
</p>

<hr>
<h2 id='rank_ref_zoo'>Lookup-table for IDs of taxonomic ranks (WoRMS)</h2><span id='topic+rank_ref_zoo'></span>

<h3>Description</h3>

<p>Same as <code>rank_ref</code> but specifically for WoRMS, where section/subsection
ranks are put between family/order rather than between species/genus.
</p>

<hr>
<h2 id='rankagg'>Aggregate data by given taxonomic rank</h2><span id='topic+rankagg'></span>

<h3>Description</h3>

<p>Aggregate data by given taxonomic rank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rankagg(data = NULL, datacol = NULL, rank = NULL, fxn = "sum")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rankagg_+3A_data">data</code></td>
<td>
<p>A data.frame. Column headers must have capitalized ranks (e.g.,
Genus, Tribe, etc.) (data.frame)</p>
</td></tr>
<tr><td><code id="rankagg_+3A_datacol">datacol</code></td>
<td>
<p>The data column (character)</p>
</td></tr>
<tr><td><code id="rankagg_+3A_rank">rank</code></td>
<td>
<p>Taxonomic rank to aggregate by (character)</p>
</td></tr>
<tr><td><code id="rankagg_+3A_fxn">fxn</code></td>
<td>
<p>Arithmetic function or vector or functions (character)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>if (require(vegan)) {
data(dune.taxon, dune, package='vegan')
dat &lt;- dune.taxon
dat$abundance &lt;- colSums(dune)
rankagg(data=dat, datacol="abundance", rank="Genus")
rankagg(data=dat, "abundance", rank="Family")
rankagg(data=dat, "abundance", rank="Genus", fxn="mean")
rankagg(data=dat, "abundance", rank="Subclass")
rankagg(data=dat, "abundance", rank="Subclass", fxn="sd")
}
</code></pre>

<hr>
<h2 id='resolve'>Resolve names from different data sources</h2><span id='topic+resolve'></span>

<h3>Description</h3>

<p>Resolve names from iPlant's name resolver, and
the Global Names Resolver (GNR)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resolve(sci, db = "gnr", query = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="resolve_+3A_sci">sci</code></td>
<td>
<p>Vector of one or more taxonomic names (common names not
supported)</p>
</td></tr>
<tr><td><code id="resolve_+3A_db">db</code></td>
<td>
<p>Source to check names against. One of iplant or gnr.
Default: gnr. Note that each taxonomic data source has their own
identifiers, so that if you provide the wrong <code>db</code> value for the
identifier you could get a result, but it will likely be wrong (not
what you were expecting).</p>
</td></tr>
<tr><td><code id="resolve_+3A_query">query</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="resolve_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a> or
<a href="crul.html#topic+verb-POST">crul::verb-POST</a>. In addition, further named args passed
on to  each respective function. See examples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with length equal to length of the db parameter (number
of sources requested), with each element being a data.frame or list
with results from that source.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
resolve(sci=c("Helianthus annuus", "Homo sapiens"))
resolve(sci="Quercus keloggii", db='gnr')
resolve(sci=c("Helianthus annuus", "Homo sapiens"), db=c('iplant', 'gnr'))
resolve(sci="Quercus keloggii", db=c('iplant', 'gnr'))

# pass in options specific to each source
resolve("Helianthus annuus", db = 'gnr', preferred_data_sources = c(3, 4))
resolve("Helianthus annuus", db = 'iplant', retrieve = 'best')
identical(
 resolve("Helianthus annuus", db = 'iplant', retrieve = 'best')$iplant,
 iplant_resolve("Helianthus annuus", retrieve = 'best')
)

# pass in curl options
resolve(sci="Qercuss", db = "iplant", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='sci2comm'>Get common names from scientific names.</h2><span id='topic+sci2comm'></span><span id='topic+sci2comm.default'></span><span id='topic+sci2comm.uid'></span><span id='topic+sci2comm.tsn'></span><span id='topic+sci2comm.wormsid'></span><span id='topic+sci2comm.iucn'></span>

<h3>Description</h3>

<p>Get common names from scientific names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sci2comm(...)

## Default S3 method:
sci2comm(sci, db = "ncbi", simplify = TRUE, scinames = NULL, ...)

## S3 method for class 'uid'
sci2comm(id, ...)

## S3 method for class 'tsn'
sci2comm(id, simplify = TRUE, ...)

## S3 method for class 'wormsid'
sci2comm(id, simplify = TRUE, ...)

## S3 method for class 'iucn'
sci2comm(id, simplify = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sci2comm_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to functions <code><a href="#topic+get_uid">get_uid()</a></code>,
<code><a href="#topic+get_tsn">get_tsn()</a></code>.</p>
</td></tr>
<tr><td><code id="sci2comm_+3A_sci">sci</code></td>
<td>
<p>character; One or more scientific names or partial names.</p>
</td></tr>
<tr><td><code id="sci2comm_+3A_db">db</code></td>
<td>
<p>character; Data source, one of <code>"ncbi"</code> (default),
<code>"itis"</code> <code>"eol"</code>, <code>"worms"</code>, or <code>"iucn"</code>. Note that
each taxonomic data source has their own identifiers,  so that if you
provide the wrong <code>db</code> value for the identifier you could get a
result, but it will likely be wrong (not what you were expecting).
If using ncbi or iucn we recommend getting an API key;
see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="sci2comm_+3A_simplify">simplify</code></td>
<td>
<p>(logical) If TRUE, simplify output to a vector of names.
If FALSE, return variable formats from different sources, usually a
data.frame. Only applies to eol and itis. Specify <code>FALSE</code> to obtain
the language of each vernacular in the output for eol and itis.</p>
</td></tr>
<tr><td><code id="sci2comm_+3A_scinames">scinames</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="sci2comm_+3A_id">id</code></td>
<td>
<p>character; identifiers, as returned by <code><a href="#topic+get_tsn">get_tsn()</a></code>,
<code><a href="#topic+get_uid">get_uid()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of character vectors, named by input taxon name,
or taxon ID. <code>character(0)</code> on no match
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication
</p>


<h3>HTTP version for NCBI requests</h3>

<p>We hard code <code>http_version = 2L</code> to use HTTP/1.1 in HTTP requests to
the Entrez API. See <code>curl::curl_symbols('CURL_HTTP_VERSION')</code>
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>See Also</h3>

<p><code><a href="#topic+comm2sci">comm2sci()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
sci2comm(sci='Helianthus annuus')
sci2comm(sci='Helianthus annuus', db='eol')
sci2comm(sci=c('Helianthus annuus', 'Poa annua'))
sci2comm(sci='Puma concolor', db='ncbi')
sci2comm('Gadus morhua', db='worms')
sci2comm('Pomatomus saltatrix', db='worms')
sci2comm('Loxodonta africana', db='iucn')

# Passing id in, works for sources: itis and ncbi, not eol
sci2comm(get_uid('Helianthus annuus'))
sci2comm(get_wormsid('Gadus morhua'))
sci2comm(get_iucn('Loxodonta africana'))

# Don't simplify returned
sci2comm(get_iucn('Loxodonta africana'), simplify=FALSE)

# Use curl options
sci2comm('Helianthus annuus', db="ncbi", verbose = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='scrapenames'>Find taxon names using Global Names Recognition and Discovery</h2><span id='topic+scrapenames'></span>

<h3>Description</h3>

<p>Uses the Global Names Recognition and Discovery service, see
http://gnrd.globalnames.org/
</p>
<p>NOTE: This function sometimes gives data back and sometimes not. The API
that this function is using is extremely buggy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scrapenames(
  url = NULL,
  text = NULL,
  format = "csv",
  bytes_offset = FALSE,
  return_content = FALSE,
  unique_names = TRUE,
  ambiguous_names = FALSE,
  no_bayes = FALSE,
  odds_details = FALSE,
  language = "detect",
  words_around = 0,
  verification = TRUE,
  sources = NULL,
  all_matches = FALSE,
  ...,
  file = NULL,
  unique = NULL,
  engine = NULL,
  detect_language = NULL,
  data_source_ids = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scrapenames_+3A_url">url</code></td>
<td>
<p>(character) If text parameter is empty, and <code>url</code> is given,
GNfinder will process the URL and will find names in the content of its
body.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_text">text</code></td>
<td>
<p>(character) Contains the text which will be checked for
scientific names. If this parameter is not empty, the <code>url</code> parameter is
ignored.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_format">format</code></td>
<td>
<p>(character) Sets the output format. It can be set to: <code>"csv"</code>
(the default), <code>"tsv"</code>, or <code>"json"</code>.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_bytes_offset">bytes_offset</code></td>
<td>
<p>(logical) This changes how the position of a detected
name in text is calculated. Normally a name's start and end positions are
given as the number of UTF-8 characters from the beginning of the text. If
this is <code>TRUE</code>, the start and end offsets are recalculated in the number of
bytes.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_return_content">return_content</code></td>
<td>
<p>(logical) If this is <code>TRUE</code>, the text used for the name
detection is returned back. This is especially useful if the input was not
a plain UTF-8 text and had to be prepared for name-finding. Then the
returned content can be used together with start and end fields of detected
name-strings to locate the strings in the text.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_unique_names">unique_names</code></td>
<td>
<p>(logical) If this is <code>TRUE</code>, the output returns a list of
unique names, instead of a list of all name occurrences. Unique list of
names does not provide position information of a name in the text.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_ambiguous_names">ambiguous_names</code></td>
<td>
<p>(logical) If this is <code>TRUE</code>, strings which are
simultaneously scientific names and &quot;normal&quot; words are not filtered out
from the results. For example, generic names like America, Cancer,
Cafeteria will be returned in the results.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_no_bayes">no_bayes</code></td>
<td>
<p>(logical) If this is <code>TRUE</code>, only heuristic algorithms are
used for name detection.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_odds_details">odds_details</code></td>
<td>
<p>(logical) If <code>TRUE</code>, the result will contain odds of all
features used for calculation of NaiveBayes odds. Odds describe probability
of a name to be 'real'. The higher the odds, the higher the probability
that a detected name is not a false positive. Odds are calculated by
multiplication of the odds of separate features. Odds details explain how
the final odds value is calculated.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_language">language</code></td>
<td>
<p>(character) The language of the text. Language value is used
for calculation of Bayesian odds. If this parameter is not given, <code>"eng"</code>
is used by default. Currently only English and German languages are
supported. Valid values are: <code>"eng"</code>, <code>"deu"</code>, and <code>"detect"</code>.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_words_around">words_around</code></td>
<td>
<p>(integer) Allows to see the context surrounding a
name-string. This sets the number of words located immediately before or
after a detected name. These words are then returned in the output. Default
is 0, maximum value is 5.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_verification">verification</code></td>
<td>
<p>(character) When this <code>TRUE</code>, there is an additional
verification step for detected names. This step requires internet
connection and uses https://verifier.globalnames.org/api/v1 for
verification queries.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_sources">sources</code></td>
<td>
<p>Pipe separated list of data source ids to resolve found names
against. See list of Data Sources
http://resolver.globalnames.org/data_sources</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_all_matches">all_matches</code></td>
<td>
<p>When this option is true all found results are returned,
not only the bestResults. The bestResult field in this case is null, and
results field should contain found results of the matches.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_...">...</code></td>
<td>
<p>Further args passed to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
<tr><td><code id="scrapenames_+3A_file">file</code></td>
<td>
<p>Defunct. If you feel this is important functionality submit an
issue at &quot;https://github.com/ropensci/taxize&quot;</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_unique">unique</code></td>
<td>
<p>Defunct. See the <code>unique_names</code> option.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_engine">engine</code></td>
<td>
<p>Defunct. The API used no longer supports this option.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_detect_language">detect_language</code></td>
<td>
<p>Defunct. See the <code>language</code> option.</p>
</td></tr>
<tr><td><code id="scrapenames_+3A_data_source_ids">data_source_ids</code></td>
<td>
<p>Defunct. See the <code>sources</code> option.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble::tibble()</a></code> or list representing parsed JSON output
depending on the value of the <code>format</code> option.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain, Zachary Foster
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get data from a website using its URL
scrapenames('https://en.wikipedia.org/wiki/Spider')
scrapenames('https://en.wikipedia.org/wiki/Animal')
scrapenames('https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0095068')
scrapenames('https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0080498')

scrapenames(url = 'https://en.wikipedia.org/wiki/Spider', source=c(1, 169))

# Get data from text string
scrapenames(text='A spider named Pardosa moesta Banks, 1892')

# return OCR content
scrapenames(text='A spider named Pardosa moesta Banks, 1892',
            return_content = TRUE, format = 'json')

## End(Not run)
</code></pre>

<hr>
<h2 id='species_plantarum_binomials'>Species names from Species Plantarum</h2><span id='topic+species_plantarum_binomials'></span>

<h3>Description</h3>

<p>These names have been compiled from
<a href="https://en.wikipedia.org/wiki/Species_Plantarum"><em>Species Plantarum</em></a> by
<a href="https://en.wikipedia.org/wiki/Carl_Linnaeus">Carl Linnaeus</a> originally
published in 1753. It is the first work to consistently apply
<a href="https://en.wikipedia.org/wiki/Binomial_nomenclature">binomial names</a>
and was the starting point for the naming of plants. The book lists every
species of plant known at the time, classified into
<a href="https://en.wikipedia.org/wiki/Genus">genera</a>. The dataset provides a
useful reference point to see how taxonomic names have changed since their
inception. The names were transcribed by Robert W. Kiger.
</p>


<h3>Format</h3>

<p>A data frame with 5940 rows and 3 variables:
</p>

<ul>
<li><p> genus First part of the binomial species name for each species
within the <a href="https://en.wikipedia.org/wiki/Genus">genus</a>
</p>
</li>
<li><p> epithet specific epithet or second part of the binomial species name
for each <a href="https://en.wikipedia.org/wiki/Species">species</a>
</p>
</li>
<li><p> page_number The following abbreviations sometimes are used in the
page_number field.
</p>

<ul>
<li><p> &quot;add.&quot; refers to addenda that appear on the unnumbered last page of
the index in volume two.
</p>
</li>
<li><p> &quot;err.&quot; refers to the unnumbered page of errata that appears following
the index in volume two.
</p>
</li>
<li><p> &quot;canc.&quot; following a page number indicates that the binomial appeared
on the cancelled version of that page and does not appear on its
replacement (as in the 1957-1959 facsimile edition.
</p>
</li></ul>

</li></ul>



<h3>Author(s)</h3>

<p>Carl Linnaeus
</p>


<h3>Source</h3>

<p><a href="http://fmhibd.library.cmu.edu/HIBD-DB/Species/home.php">Hunt Institute for Botanical Documentation</a>
</p>


<h3>References</h3>

<p>Linnaeus, C. 1753. Species Plantarum. 2 vols. Salvius, Stockholm.
[Facsimile edition, 1957-1959, Ray Society, London.]
</p>

<hr>
<h2 id='status_codes'>Get HTTP status codes</h2><span id='topic+status_codes'></span>

<h3>Description</h3>

<p>Get HTTP status codes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>status_codes()
</code></pre>


<h3>See Also</h3>

<p><code><a href="#topic+ping">ping()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>status_codes()
</code></pre>

<hr>
<h2 id='synonyms'>Retrieve synonyms from various sources given input taxonomic
names or identifiers</h2><span id='topic+synonyms'></span><span id='topic+synonyms.default'></span><span id='topic+synonyms.tsn'></span><span id='topic+synonyms.tpsid'></span><span id='topic+synonyms.nbnid'></span><span id='topic+synonyms.wormsid'></span><span id='topic+synonyms.iucn'></span><span id='topic+synonyms.pow'></span><span id='topic+synonyms.ids'></span><span id='topic+synonyms_df'></span>

<h3>Description</h3>

<p>Retrieve synonyms from various sources given input taxonomic
names or identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>synonyms(...)

## Default S3 method:
synonyms(sci_id, db = NULL, rows = NA, x = NULL, ...)

## S3 method for class 'tsn'
synonyms(id, ...)

## S3 method for class 'tpsid'
synonyms(id, ...)

## S3 method for class 'nbnid'
synonyms(id, ...)

## S3 method for class 'wormsid'
synonyms(id, ...)

## S3 method for class 'iucn'
synonyms(id, ...)

## S3 method for class 'pow'
synonyms(id, ...)

## S3 method for class 'ids'
synonyms(id, ...)

synonyms_df(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="synonyms_+3A_...">...</code></td>
<td>
<p>Other passed arguments to internal functions <code style="white-space: pre;">&#8288;get_*()&#8288;</code> and
functions to gather synonyms.</p>
</td></tr>
<tr><td><code id="synonyms_+3A_sci_id">sci_id</code></td>
<td>
<p>Vector of taxa names (character) or IDs (character or numeric)</p>
</td></tr>
<tr><td><code id="synonyms_+3A_db">db</code></td>
<td>
<p>character; database to query. either <code>itis</code>, <code>tropicos</code>,
<code>nbn</code>, <code>worms</code>, or <code>pow</code>. Note that each taxonomic data source has their own
identifiers, so that if you provide the wrong <code>db</code> value for the identifier
you could get a result, but it will likely be wrong (not what you were
expecting). If using tropicos, we  recommend getting an API key;
see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="synonyms_+3A_rows">rows</code></td>
<td>
<p>(numeric) Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this parameter is ignored if you pass in a
taxonomic id of any of the acceptable classes: tsn, tpsid, nbnid, ids.</p>
</td></tr>
<tr><td><code id="synonyms_+3A_x">x</code></td>
<td>
<p>For <code>synonyms()</code>: deprecated, see <code>sci_id</code>. For <code>synonyms_df()</code>,
the output of <code>synonyms()</code></p>
</td></tr>
<tr><td><code id="synonyms_+3A_id">id</code></td>
<td>
<p>character; identifiers, returned by <code><a href="#topic+get_tsn">get_tsn()</a></code>, <code><a href="#topic+get_tpsid">get_tpsid()</a></code>,
<code><a href="#topic+get_nbnid">get_nbnid()</a></code>, <code><a href="#topic+get_wormsid">get_wormsid()</a></code>, <code><a href="#topic+get_pow">get_pow()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>If IDs are supplied directly (not from the <code style="white-space: pre;">&#8288;get_*()&#8288;</code> functions)
you must specify the type of ID.
</p>
<p>For <code>db = "itis"</code> you can pass in a parameter <code>accepted</code> to
toggle whether only accepted names are used <code>accepted = TRUE</code>, or if
all are used <code>accepted = FALSE</code>. The default is <code>accepted = FALSE</code>
</p>
<p>Note that IUCN requires an API key. See <a href="rredlist.html#topic+rredlist-package">rredlist::rredlist-package</a>
for help on authentiating with IUCN Redlist
</p>


<h3>Value</h3>

<p>A named list of results with three types of output in each slot:
</p>

<ul>
<li><p> if the name was not found: <code>NA_character_</code>
</p>
</li>
<li><p> if the name was found but no synonyms found, an empty data.frame (0 rows)
</p>
</li>
<li><p> if the name was found, and synonyms found, a data.frames with the
synonyms - the column names vary by data source
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+get_tsn">get_tsn()</a></code> <code><a href="#topic+get_tpsid">get_tpsid()</a></code> <code><a href="#topic+get_nbnid">get_nbnid()</a></code>
<code><a href="#topic+get_wormsid">get_wormsid()</a></code> <code><a href="#topic+get_iucn">get_iucn()</a></code> <code><a href="#topic+get_pow">get_pow()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Plug in taxon IDs
synonyms(183327, db="itis")
synonyms("25509881", db="tropicos")
synonyms("NBNSYS0000004629", db='nbn')
synonyms(105706, db='worms')
synonyms(12392, db='iucn')
synonyms('urn:lsid:ipni.org:names:358881-1', db='pow')

# Plug in taxon names directly
synonyms("Pinus contorta", db="itis")
synonyms("Puma concolor", db="itis")
synonyms(c("Poa annua",'Pinus contorta','Puma concolor'), db="itis")
synonyms("Poa annua", db="tropicos")
synonyms("Pinus contorta", db="tropicos")
synonyms(c("Poa annua",'Pinus contorta'), db="tropicos")
synonyms("Pinus sylvestris", db='nbn')
synonyms('Pomatomus', db='worms')
synonyms('Pomatomus saltatrix', db='worms')
synonyms('Lithocarpus mindanaensis', db='pow')
synonyms('Poa annua', db='pow')
synonyms(c('Poa annua', 'Pinus contorta', 'foo bar'), db='pow')

# not accepted names, with ITIS
## looks for whether the name given is an accepted name,
## and if not, uses the accepted name to look for synonyms
synonyms("Acer drummondii", db="itis")
synonyms("Spinus pinus", db="itis")

# Use get_* methods
synonyms(get_tsn("Poa annua"))
synonyms(get_tpsid("Poa annua"))
synonyms(get_nbnid("Carcharodon carcharias"))
synonyms(get_iucn('Loxodonta africana'))
synonyms(get_pow('Lithocarpus mindanaensis'))

# Pass many ids from class "ids"
out &lt;- get_ids(names="Poa annua", db = c('itis','tropicos'))
synonyms(out)

# Use the rows parameter to select certain rows
synonyms("Poa annua", db='tropicos', rows=1)
synonyms("Poa annua", db='tropicos', rows=1:3)
synonyms("Pinus sylvestris", db='nbn', rows=1:3)

# Use curl options
synonyms("Poa annua", db='tropicos', rows=1, verbose = TRUE)
synonyms("Poa annua", db='itis', rows=1, verbose = TRUE)


# combine many outputs together
x &lt;- synonyms(c("Osmia bicornis", "Osmia rufa", "Osmia"), db = "itis")
synonyms_df(x)

## note here how Pinus contorta is dropped due to no synonyms found
synonyms_df(x)

## note here that ids are taxon identifiers b/c you start with them
x &lt;- synonyms(c(25509881, 13100094), db="tropicos")
synonyms_df(x)

## NBN
x &lt;- synonyms(c('Aglais io', 'Usnea hirta', 'Arctostaphylos uva-ursi'),
  db="nbn")
synonyms_df(x)

## End(Not run)
</code></pre>

<hr>
<h2 id='tax_agg'>Aggregate species data to given taxonomic rank</h2><span id='topic+tax_agg'></span><span id='topic+print.tax_agg'></span>

<h3>Description</h3>

<p>Aggregate species data to given taxonomic rank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_agg(x, rank, db = "ncbi", messages = FALSE, ...)

## S3 method for class 'tax_agg'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tax_agg_+3A_x">x</code></td>
<td>
<p>Community data matrix. Taxa in columns, samples in rows.</p>
</td></tr>
<tr><td><code id="tax_agg_+3A_rank">rank</code></td>
<td>
<p>character; Taxonomic rank to aggregate by.</p>
</td></tr>
<tr><td><code id="tax_agg_+3A_db">db</code></td>
<td>
<p>character; taxonomic API to use, 'ncbi, 'itis' or both, see
<code><a href="#topic+tax_name">tax_name()</a></code>. Note that each taxonomic data source has
their own identifiers, so that if you provide the wrong <code>db</code> value
for the identifier you could get a result, but it will likely be wrong (not
what you were expecting). If using ncbi we recommend getting an API key;
see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="tax_agg_+3A_messages">messages</code></td>
<td>
<p>(logical) If FALSE (Default) suppress messages</p>
</td></tr>
<tr><td><code id="tax_agg_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+get_tsn">get_tsn()</a></code> or <code><a href="#topic+get_uid">get_uid()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>tax_agg</code> aggregates (sum) taxa to a specific taxonomic level.
If a taxon is not found in the database (ITIS or NCBI) or the supplied taxon
is on higher taxonomic level this taxon is not aggregated.
</p>


<h3>Value</h3>

<p>A list of class <code>tax_agg</code> with the following items:
</p>

<ul>
<li> <p><code>x</code> Community data matrix with aggregated data.
</p>
</li>
<li> <p><code>by</code> A lookup-table showing which taxa were aggregated.
</p>
</li>
<li> <p><code>n_pre</code> Number of taxa before aggregation.
</p>
</li>
<li> <p><code>rank</code> Rank at which taxa have been aggregated.
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+tax_name">tax_name</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if (requireNamespace("vegan", quietly = TRUE)) {
  # use dune dataset
  data(dune, package='vegan')
  species &lt;- c("Achillea millefolium", "Agrostis stolonifera",
    "Aira praecox", "Alopecurus geniculatus", "Anthoxanthum odoratum",
    "Bellis perennis", "Bromus hordeaceus", "Chenopodium album",
    "Cirsium arvense", "Comarum palustre", "Eleocharis palustris",
    "Elymus repens", "Empetrum nigrum", "Hypochaeris radicata",
    "Juncus articulatus", "Juncus bufonius", "Lolium perenne",
    "Plantago lanceolata", "Poa pratensis", "Poa trivialis",
    "Ranunculus flammula", "Rumex acetosa", "Sagina procumbens",
    "Salix repens", "Scorzoneroides autumnalis", "Trifolium pratense",
    "Trifolium repens", "Vicia lathyroides", "Brachythecium rutabulum",
    "Calliergonella cuspidata")
  colnames(dune) &lt;- species

  # aggregate sample to families
  (agg &lt;- tax_agg(dune, rank = 'family', db = 'ncbi'))

  # extract aggregated community data matrix for further usage
  agg$x
  # check which taxa have been aggregated
  agg$by
}

# A use case where there are different taxonomic levels in the same dataset
spnames &lt;- c('Puma','Ursus americanus','Ursidae')
df &lt;- data.frame(c(1,2,3), c(11,12,13), c(1,4,50))
names(df) &lt;- spnames
out &lt;- tax_agg(x=df, rank = 'family', db='itis')
out$x

# You can input a matrix too
mat &lt;- matrix(c(1,2,3, 11,12,13), nrow = 2, ncol = 3,
 dimnames=list(NULL, c('Puma concolor','Ursus americanus','Ailuropoda melanoleuca')))
tax_agg(mat, rank = 'family', db='itis')

## End(Not run)
</code></pre>

<hr>
<h2 id='tax_name'>Get taxonomic names for a given rank</h2><span id='topic+tax_name'></span>

<h3>Description</h3>

<p>Get taxonomic names for a given rank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_name(
  sci,
  get,
  db = "itis",
  pref = "ncbi",
  messages = TRUE,
  query = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tax_name_+3A_sci">sci</code></td>
<td>
<p>(character) Vector of taxonomic names to query. required.</p>
</td></tr>
<tr><td><code id="tax_name_+3A_get">get</code></td>
<td>
<p>(character) The ranks of the taxonomic name to get, see
<a href="#topic+rank_ref">rank_ref</a>. required.</p>
</td></tr>
<tr><td><code id="tax_name_+3A_db">db</code></td>
<td>
<p>(character) The database to search from: 'itis', 'ncbi' or 'both'.
If 'both' both NCBI and ITIS will be queried. Result will be the union of
both. If using ncbi, we recommend getting an API key; see
<a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="tax_name_+3A_pref">pref</code></td>
<td>
<p>(character) If db = 'both', sets the preference for the union.
Either 'ncbi' (default) or 'itis'. Currently not implemented.</p>
</td></tr>
<tr><td><code id="tax_name_+3A_messages">messages</code></td>
<td>
<p>(logical) If <code>TRUE</code> the actual taxon queried is printed
on the console.</p>
</td></tr>
<tr><td><code id="tax_name_+3A_query">query</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="tax_name_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+get_tsn">get_tsn()</a></code> or
<code><a href="#topic+get_uid">get_uid()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with one column for every queried rank, in addition to
a column for db and queried term.
</p>


<h3>Authentication</h3>

<p>See <a href="#topic+taxize-authentication">taxize-authentication</a> for help on authentication
</p>


<h3>Note</h3>

<p>While <code><a href="#topic+tax_rank">tax_rank()</a></code> returns the actual rank of a
taxon, <code><a href="#topic+tax_name">tax_name()</a></code> searches and returns any specified rank
higher in taxonomy.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# A case where itis and ncbi use the same names
tax_name(sci = "Helianthus annuus", get = "family", db = "itis")
tax_name(sci = "Helianthus annuus", get = "family", db = "ncbi")
tax_name(sci = "Helianthus annuus", get = c("genus","family","order"),
  db = "ncbi")

# Case where itis and ncbi use different names
tax_name(sci = "Helianthus annuus", get = "kingdom", db = "itis")
tax_name(sci = "Helianthus annuus", get = "kingdom", db = "ncbi")

# multiple rank arguments
tax_name(sci = c("Helianthus annuus","Baetis rhodani"), get = c("genus",
"kingdom"), db = "ncbi")
tax_name(sci = c("Helianthus annuus","Baetis rhodani"), get = c("genus",
"kingdom"), db = "itis")

# query both sources
tax_name(sci=c("Helianthus annuus", 'Baetis rhodani'), get=c("genus",
"kingdom"), db="both")

## End(Not run)
</code></pre>

<hr>
<h2 id='tax_rank'>Get rank for a given taxonomic name.</h2><span id='topic+tax_rank'></span>

<h3>Description</h3>

<p>Get rank for a given taxonomic name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_rank(sci_id, db = NULL, rows = NA, x = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tax_rank_+3A_sci_id">sci_id</code></td>
<td>
<p>(character) Vector of one or more taxon names (character) or
IDs (character or numeric) to query. Or objects returned from <code style="white-space: pre;">&#8288;get_*()&#8288;</code>
functions like <code><a href="#topic+get_tsn">get_tsn()</a></code></p>
</td></tr>
<tr><td><code id="tax_rank_+3A_db">db</code></td>
<td>
<p>(character) database to query. either <code>ncbi</code>, <code>itis</code>, <code>eol</code>,
<code>tropicos</code>, <code>gbif</code>,<code>nbn</code>, <code>worms</code>, <code>natserv</code>, <code>bold</code>. Note that each
taxonomic data source has their own identifiers, so that if you provide the
wrong <code>db</code> value for the identifier you may get a result, but it will
likely be wrong (not what you were expecting). If using ncbi we
recommend getting an API key; see <a href="#topic+taxize-authentication">taxize-authentication</a></p>
</td></tr>
<tr><td><code id="tax_rank_+3A_rows">rows</code></td>
<td>
<p>numeric; Any number from 1 to infinity. If the default NA,
all rows are considered. passed down to <code style="white-space: pre;">&#8288;get_*()&#8288;</code> functions.</p>
</td></tr>
<tr><td><code id="tax_rank_+3A_x">x</code></td>
<td>
<p>Deprecated, see <code>sci_id</code></p>
</td></tr>
<tr><td><code id="tax_rank_+3A_...">...</code></td>
<td>
<p>Additional arguments to <code><a href="#topic+classification">classification()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of character vectors with ranks (all lower-cased)
</p>


<h3>Note</h3>

<p>While <code><a href="#topic+tax_name">tax_name()</a></code> returns the name of a specified
rank, <code><a href="#topic+tax_rank">tax_rank()</a></code> returns the actual rank of the taxon.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classification">classification()</a></code>,<code><a href="#topic+tax_name">tax_name()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tax_rank("Helianthus annuus", db = "itis")
tax_rank("Helianthus annuus", db = "natserv")
tax_rank(get_tsn("Helianthus annuus"))
tax_rank(c("Helianthus", "Pinus", "Poa"), db = "itis")

tax_rank(get_boldid("Helianthus annuus"))
tax_rank("421377", db = "bold")
tax_rank(421377, db = "bold")

tax_rank(c("Plantae", "Helianthus annuus",
  "Puma", "Homo sapiens"), db = 'itis')
tax_rank(c("Helianthus annuus", "Quercus", "Fabaceae"), db = 'tropicos')

tax_rank(names_list("species"), db = 'gbif')
tax_rank(names_list("family"), db = 'gbif')

tax_rank(c("Gadus morhua", "Lichenopora neapolitana"),
  db = "worms")

## End(Not run)
</code></pre>

<hr>
<h2 id='taxize_capwords'>Capitalize the first letter of a character string.</h2><span id='topic+taxize_capwords'></span>

<h3>Description</h3>

<p>Capitalize the first letter of a character string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxize_capwords(s, strict = FALSE, onlyfirst = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxize_capwords_+3A_s">s</code></td>
<td>
<p>A character string</p>
</td></tr>
<tr><td><code id="taxize_capwords_+3A_strict">strict</code></td>
<td>
<p>Should the algorithm be strict about capitalizing.
Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="taxize_capwords_+3A_onlyfirst">onlyfirst</code></td>
<td>
<p>Capitalize only first word, lowercase all others. Useful
for taxonomic names.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>taxize_capwords(c("using AIC for model selection"))
taxize_capwords(c("using AIC for model selection"), strict=TRUE)
</code></pre>

<hr>
<h2 id='taxize_cite'>Get citations and licenses for data sources used in taxize</h2><span id='topic+taxize_cite'></span>

<h3>Description</h3>

<p>Get citations and licenses for data sources used in taxize
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxize_cite(fxn = "itis", what = "citation")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxize_cite_+3A_fxn">fxn</code></td>
<td>
<p>Function to search on. A special case is the package name
'taxize' that will give the citations for the package.</p>
</td></tr>
<tr><td><code id="taxize_cite_+3A_what">what</code></td>
<td>
<p>One of citation (default), license, or both.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>taxize_cite(fxn='eol_search')
taxize_cite(fxn='itis_hierarchy')
taxize_cite(fxn='tp_classification')
taxize_cite(fxn='gbif_ping')
taxize_cite(fxn='plantminer')
taxize_cite(fxn='get_natservid_')
taxize_cite(fxn='as.natservid')
taxize_cite(fxn='get_wormsid')
taxize_cite(fxn='as.wormsid')

# Functions that use many data sources
taxize_cite(fxn='synonyms')
taxize_cite(fxn='classification')

# Get the taxize citation
taxize_cite(fxn='taxize')

# Get license information
taxize_cite(fxn='taxize', "license")
</code></pre>

<hr>
<h2 id='taxize_ldfast'>Replacement function for ldply that should be faster in all cases.</h2><span id='topic+taxize_ldfast'></span>

<h3>Description</h3>

<p>Replacement function for ldply that should be faster in all cases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxize_ldfast(x, convertvec = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxize_ldfast_+3A_x">x</code></td>
<td>
<p>A list.</p>
</td></tr>
<tr><td><code id="taxize_ldfast_+3A_convertvec">convertvec</code></td>
<td>
<p>Convert a vector to a data.frame before rbind is called.</p>
</td></tr>
</table>

<hr>
<h2 id='taxize_options'>taxize options</h2><span id='topic+taxize_options'></span>

<h3>Description</h3>

<p>taxize options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxize_options(taxon_state_messages = NULL, ncbi_sleep = NULL, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxize_options_+3A_taxon_state_messages">taxon_state_messages</code></td>
<td>
<p>(logical) suppress messages? default: <code>NULL</code>
(same as setting <code>FALSE</code>). Set to <code>TRUE</code> to suppress messages, and <code>FALSE</code>
to not suppress messages</p>
</td></tr>
<tr><td><code id="taxize_options_+3A_ncbi_sleep">ncbi_sleep</code></td>
<td>
<p>(numeric/integer) number of seconds to sleep between
NCBI ENTREZ http requests. applies to the functions: <code><a href="#topic+classification">classification()</a></code>,
<code><a href="#topic+comm2sci">comm2sci()</a></code>, <code><a href="#topic+genbank2uid">genbank2uid()</a></code>, <code><a href="#topic+get_uid">get_uid()</a></code> and <code><a href="#topic+ncbi_children">ncbi_children()</a></code>.
defaults: 0.334 (without API key) or 0.101 (with API key). minimum value
can not be less than 0.101</p>
</td></tr>
<tr><td><code id="taxize_options_+3A_quiet">quiet</code></td>
<td>
<p>(logical) quiet informational output from this function.
default: <code>TRUE</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
taxize_options()
taxize_options(FALSE)
taxize_options(TRUE)
taxize_options(ncbi_sleep = 0.4)
taxize_options(taxon_state_messages = TRUE, ncbi_sleep = 0.4)

## End(Not run)
</code></pre>

<hr>
<h2 id='taxize-authentication'>taxize authentication</h2><span id='topic+taxize-authentication'></span><span id='topic+authentication'></span>

<h3>Description</h3>

<p>Help on authentication
</p>


<h3>What is an API?</h3>

<p>An API is an Application Programming Interface. The term &quot;API&quot; can be used
for lots of scenarios, but in this case we're talking about web APIs,
or APIs (interfaces) to web resources. <span class="pkg">taxize</span> interacts with
remote databases on the web via their APIs. You don't need to worry
about the details of how that all works; just know that some of them
require authentication and some do not.
</p>


<h3>What are API keys?</h3>

<p>For those APIs that require authentication, the way that's typically done
is through API keys: alphanumeric strings of variable lengths that are
supplied with a request to an API.
</p>
<p><span class="pkg">taxize</span> won't get these keys for you; rather, you have to
go get a key for each service, but we do provide information on how
to get those keys. See <code><a href="#topic+key_helpers">key_helpers()</a></code> for help on how to
obtain keys for this package.
</p>


<h3>Using API keys</h3>

<p>You can store API keys as R options in your <code>.Rprofile</code> file, or
as environment variables in either your <code>.Renviron</code> file or
<code>.bash_profile</code> file, o<code>.zshrc</code> file (if you use oh-my-zsh) or
similar. See <a href="base.html#topic+Startup">Startup</a> for help on R options and environment
variables.
</p>
<p>Save your API keys with the following names:
</p>

<ul>
<li><p> Tropicos: R option or env var as 'TROPICOS_KEY'
</p>
</li>
<li><p> IUCN: R option or env var as 'IUCN_REDLIST_KEY'
</p>
</li>
<li><p> ENTREZ: R option or env var as 'ENTREZ_KEY'
</p>
</li></ul>

<p>If you save in .Renviron it looks like: <code>ENTREZ_KEY=somekey</code>
</p>
<p>If you save in a .bash_profile, .zshrc, or similar file it looks like:
<code style="white-space: pre;">&#8288;export ENTREZ_KEY=somekey&#8288;</code>
</p>
<p>If you save in a .Rprofile it looks like: <code>options(ENTREZ_KEY = "somekey")</code>
</p>
<p>Remember to restart your R session (and to start a new shell window/tab
if you're using the shell) to take advantage of the new R options
or environment variables.
</p>
<p>We strongly recommend using environment variables
(https://en.wikipedia.org/wiki/Environment_variable) over R options
because environment variables are widely used across programming
languages, operating systems, and computing environments; whereas
R options are specific to R.
</p>
<p>Note that NCBI Entrez doesn't require that you use an API key,
but you do get a higher rate limit with a key (more requests per
time period), from 3 to 10 requests per second, so do get one.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+key_helpers">key_helpers()</a></code>
</p>

<hr>
<h2 id='taxize-defunct'>Defunct functions in taxize</h2><span id='topic+taxize-defunct'></span><span id='topic+defunct'></span>

<h3>Description</h3>

<p>The following functions are now defunct (no longer available):
</p>

<ul>
<li><p> All COL functions are defunct: <code style="white-space: pre;">&#8288;as.colid, &#8288;</code>col_children<code style="white-space: pre;">&#8288;, &#8288;</code>col_classification<code style="white-space: pre;">&#8288;, &#8288;</code>col_downstream<code style="white-space: pre;">&#8288;, &#8288;</code>col_search<code style="white-space: pre;">&#8288;, &#8288;</code>get_colid<code style="white-space: pre;">&#8288;, &#8288;</code>get_colid_<code style="white-space: pre;">&#8288;, &#8288;</code>as.data.frame.colid<code style="white-space: pre;">&#8288;, &#8288;</code>children.colid<code style="white-space: pre;">&#8288;, &#8288;</code>classification.colid<code style="white-space: pre;">&#8288;, &#8288;</code>downstream.colid<code style="white-space: pre;">&#8288;, &#8288;</code>id2name.colid<code style="white-space: pre;">&#8288;, &#8288;</code>lowest_common.colid<code style="white-space: pre;">&#8288;, &#8288;</code>synonyms.colid<code style="white-space: pre;">&#8288;, &#8288;</code>upstream.colid'
</p>
</li>
<li> <p><code>col_classification()</code>: See<code>classification()</code>
</p>
</li>
<li> <p><code>tp_classification()</code>: See<code>classification()</code>
</p>
</li>
<li> <p><code>eol_hierarchy()</code>: See<code>classification()</code>
</p>
</li>
<li> <p><code><a href="#topic+eol_invasive">eol_invasive()</a></code>: See <code>eol</code> in the <span class="pkg">originr</span> package.
</p>
</li>
<li> <p><code><a href="#topic+use_eol">use_eol()</a></code>: EOL no longer requires an API key
</p>
</li>
<li> <p><code><a href="#topic+tpl_search">tpl_search()</a></code>: Use the <span class="pkg">Taxonstand</span> functions <code>TPL</code> or <code>TPLck</code> directly.
</p>
</li>
<li> <p><code><a href="#topic+get_seqs">get_seqs()</a></code>: This function changed name to<code>ncbi_getbyname()</code>()].
</p>
</li>
<li> <p><code><a href="#topic+get_genes">get_genes()</a></code>: This function changed name to<code>ncbi_getbyid()</code>()].
</p>
</li>
<li> <p><code><a href="#topic+get_genes_avail">get_genes_avail()</a></code>: This function changed name to<code>ncbi_search()</code>()].
</p>
</li>
<li> <p><code><a href="#topic+ncbi_getbyname">ncbi_getbyname()</a></code>: See <code>ncbi_byname</code> in the <span class="pkg">traits</span> package.
</p>
</li>
<li> <p><code><a href="#topic+ncbi_getbyid">ncbi_getbyid()</a></code>: See <code>ncbi_byid</code> in the <span class="pkg">traits</span> package.
</p>
</li>
<li> <p><code><a href="#topic+ncbi_search">ncbi_search()</a></code>: See <code>ncbi_searcher</code> in the <span class="pkg">traits</span> package.
</p>
</li>
<li> <p><code><a href="#topic+gisd_isinvasive">gisd_isinvasive()</a></code>: See <code>gisd</code> in the <span class="pkg">originr</span> package.
</p>
</li>
<li> <p><code><a href="#topic+ubio_classification">ubio_classification()</a></code>:  The uBio web services was down for quite a while, is
now (as of 2016-05-09) back up, but we don't trust that it will stay up and available.
</p>
</li>
<li> <p><code><a href="#topic+ubio_classification_search">ubio_classification_search()</a></code>: The uBio web services was down for quite
a while, is now (as of 2016-05-09) back up, but we don't trust that it will stay up
and available.
</p>
</li>
<li> <p><code><a href="#topic+ubio_id">ubio_id()</a></code>: The uBio web services was down for quite a while, is
now (as of 2016-05-09) back up, but we don't trust that it will stay up and available.
</p>
</li>
<li> <p><code><a href="#topic+ubio_ping">ubio_ping()</a></code>:  The uBio web services was down for quite a while, is
now (as of 2016-05-09) back up, but we don't trust that it will stay up and available.
</p>
</li>
<li> <p><code><a href="#topic+ubio_search">ubio_search()</a></code>:  The uBio web services was down for quite a while, is
now (as of 2016-05-09) back up, but we don't trust that it will stay up and available.
</p>
</li>
<li> <p><code><a href="#topic+ubio_synonyms">ubio_synonyms()</a></code>:  The uBio web services was down for quite a while, is
now (as of 2016-05-09) back up, but we don't trust that it will stay up and available.
</p>
</li>
<li> <p><code><a href="#topic+get_ubioid">get_ubioid()</a></code>: The uBio web services are apparently down indefinitely.
</p>
</li>
<li> <p><code><a href="#topic+phylomatic_tree">phylomatic_tree()</a></code>: This function is defunct. See
<code>phylomatic</code> in the package <span class="pkg">brranching</span>
</p>
</li>
<li> <p><code><a href="#topic+phylomatic_format">phylomatic_format()</a></code>: This function is defunct. See
<code>phylomatic_names</code> in the package <span class="pkg">brranching</span>
</p>
</li>
<li> <p><code><a href="#topic+eubon">eubon()</a></code>: This function is defunct. Use <code><a href="#topic+eubon_search">eubon_search()</a></code>
</p>
</li>
<li> <p><code><a href="#topic+tnrs">tnrs()</a></code>: This function is defunct. Was too unreliable
</p>
</li>
<li> <p><code><a href="#topic+tnrs_sources">tnrs_sources()</a></code>: This function is defunct. Was too unreliable
</p>
</li></ul>


<hr>
<h2 id='taxize-params'>taxize parameters</h2><span id='topic+taxize-params'></span>

<h3>Description</h3>

<p>Information on standardized parameters across the package
</p>


<h3>Standardized parameters</h3>


<ul>
<li> <p><code>sci</code>: scientific name
</p>
</li>
<li> <p><code>com</code>: common name
</p>
</li>
<li> <p><code>id</code>: name identifier
</p>
</li>
<li> <p><code>sci_com</code>: scientific name or common name
</p>
</li>
<li> <p><code>sci_id</code>: scientific name or name identifier
</p>
</li></ul>

<p>We were going to standardize parameter names for cases in which
a parameter accepts either of three options: scientific name,
common name, or name identifier. However, there was no clear
parameter name we could use for this case, so we've left
parameter names as they are for the two cases (<code><a href="#topic+get_ids">get_ids()</a></code> and
<code><a href="#topic+vascan_search">vascan_search()</a></code>)
</p>

<hr>
<h2 id='taxon_state'>Keep track of queries in <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions</h2><span id='topic+taxon_state'></span>

<h3>Description</h3>

<p>Keep track of queries in <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions
</p>
<p>Keep track of queries in <code style="white-space: pre;">&#8288;get_*&#8288;</code> functions
</p>


<h3>Details</h3>

<p>This object lives inside each <code style="white-space: pre;">&#8288;get_*&#8288;</code> function call, maintaining
results as they are accumulated.
</p>


<h3>Public fields</h3>

<div class="r6-fields">

<dl>
<dt><code>initialized</code></dt><dd><p>(time) time job started</p>
</dd>
<dt><code>finalized</code></dt><dd><p>(time) time job finished</p>
</dd>
<dt><code>class</code></dt><dd><p>(character) a class name (e.g., &quot;gbif&quot;)</p>
</dd>
<dt><code>names</code></dt><dd><p>(character) one or more taxon names</p>
</dd>
</dl>

</div>


<h3>Active bindings</h3>

<div class="r6-active-bindings">

<dl>
<dt><code>count</code></dt><dd><p>(integer) count number of records</p>
</dd>
<dt><code>exit</code></dt><dd><p>record date/time function exited</p>
</dd>
</dl>

</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-taxon_state-new"><code>taxon_state$new()</code></a>
</p>
</li>
<li> <p><a href="#method-taxon_state-print"><code>taxon_state$print()</code></a>
</p>
</li>
<li> <p><a href="#method-taxon_state-add"><code>taxon_state$add()</code></a>
</p>
</li>
<li> <p><a href="#method-taxon_state-get"><code>taxon_state$get()</code></a>
</p>
</li>
<li> <p><a href="#method-taxon_state-remove"><code>taxon_state$remove()</code></a>
</p>
</li>
<li> <p><a href="#method-taxon_state-purge"><code>taxon_state$purge()</code></a>
</p>
</li>
<li> <p><a href="#method-taxon_state-taxa_remaining"><code>taxon_state$taxa_remaining()</code></a>
</p>
</li>
<li> <p><a href="#method-taxon_state-taxa_completed"><code>taxon_state$taxa_completed()</code></a>
</p>
</li>
<li> <p><a href="#method-taxon_state-clone"><code>taxon_state$clone()</code></a>
</p>
</li></ul>


<hr>
<a id="method-taxon_state-new"></a>



<h4>Method <code>new()</code></h4>

<p>Create a new <code>taxon_state</code> object
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$new(class, names)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>class</code></dt><dd><p>(character) a class name (e.g., &quot;gbif&quot;)</p>
</dd>
<dt><code>names</code></dt><dd><p>(character) one or more taxon names</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A new <code>taxon_state</code> object
</p>


<hr>
<a id="method-taxon_state-print"></a>



<h4>Method <code>print()</code></h4>

<p>print method for the <code>taxon_state</code> class
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$print(x, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>x</code></dt><dd><p>self</p>
</dd>
<dt><code>...</code></dt><dd><p>ignored</p>
</dd>
</dl>

</div>


<hr>
<a id="method-taxon_state-add"></a>



<h4>Method <code>add()</code></h4>

<p>add a record with it's result; duplicates allowed
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$add(query, result)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>query</code></dt><dd><p>(character), a taxon name</p>
</dd>
<dt><code>result</code></dt><dd><p>(list) a named list</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>nothing returned; sets only
</p>


<hr>
<a id="method-taxon_state-get"></a>



<h4>Method <code>get()</code></h4>

<p>get all records matching 'query'
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$get(query = NULL)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>query</code></dt><dd><p>(character), a taxon name</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>a named list, with slots for the taxon id, and other attributes,
named by the taxon name
</p>


<hr>
<a id="method-taxon_state-remove"></a>



<h4>Method <code>remove()</code></h4>

<p>remove's all records matching 'query'
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$remove(query)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>query</code></dt><dd><p>(character), a taxon name</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>nothing, removes records matching query
</p>


<hr>
<a id="method-taxon_state-purge"></a>



<h4>Method <code>purge()</code></h4>

<p>removes all records
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$purge()</pre></div>



<h5>Returns</h5>

<p>nothing returned; sets only
</p>


<hr>
<a id="method-taxon_state-taxa_remaining"></a>



<h4>Method <code>taxa_remaining()</code></h4>

<p>get remaining taxa
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$taxa_remaining()</pre></div>



<h5>Returns</h5>

<p>sorted taxon names
</p>


<hr>
<a id="method-taxon_state-taxa_completed"></a>



<h4>Method <code>taxa_completed()</code></h4>

<p>get completed taxa
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$taxa_completed()</pre></div>



<h5>Returns</h5>

<p>sorted taxon names
</p>


<hr>
<a id="method-taxon_state-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>taxon_state$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>




<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if (interactive()) {
ts &lt;- taxon_state$new()
taxon_last()
ts
res &lt;- list(
  id = 123456,
  att = "found",
  multiple = FALSE,
  direct = FALSE,
  class = "tsn"
)
ts$add(query = "Quercus robur", result = res)
ts
ts$get(query = "Quercus robur")
ts$count
ts$remove(query = "Quercus robur")
ts
ts$count

res2 &lt;- list(
  id = 3430834535,
  att = "found",
  multiple = FALSE,
  direct = FALSE,
  class = "gbifid"
)
ts$add(query = "Poa annua", result = res2)
res3 &lt;- list(
  id = 1223424,
  att = "found",
  multiple = FALSE,
  direct = FALSE,
  class = "uid"
)
ts$add(query = "Puma concolor", result = res3)
ts
ts$count
ts$get("Puma concolor")
ts$get()

# cleanup
ts$purge()
ts$count
}

## End(Not run)
</code></pre>

<hr>
<h2 id='taxon-state'>Last taxon state object from a <code style="white-space: pre;">&#8288;get_*&#8288;</code> function call</h2><span id='topic+taxon-state'></span><span id='topic+taxon_last'></span><span id='topic+taxon_clear'></span>

<h3>Description</h3>

<p>Last taxon state object from a <code style="white-space: pre;">&#8288;get_*&#8288;</code> function call
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxon_last()

taxon_clear()
</code></pre>


<h3>Details</h3>


<ul>
<li> <p><code>taxon_last()</code>: get the last <code>taxon_state</code> object in use
</p>
</li>
<li> <p><code>taxon_clear()</code>: clear any data from last <code>taxon_state</code> object
</p>
</li></ul>

<p>The <code>taxon_state</code> object is an R6 object that holds data and methods
used for keeping track of results gathered within a <code style="white-space: pre;">&#8288;get_*&#8288;</code> function.
You shouldn't create <code>taxon_state</code> R6 objects yourself.
</p>
<p>Behaviors to be aware of:
</p>

<ul>
<li><p> If a <code>taxon_state</code> object is not passed you don't need to worry about
a previously run <code style="white-space: pre;">&#8288;get_*&#8288;</code> function interfering with another <code style="white-space: pre;">&#8288;get_*&#8288;</code>
function call - you have to explicitly pass a <code>taxon_state</code> object
to use <code>taxon_state</code>
</p>
</li>
<li><p> The passed in <code>taxon_state</code> object must have a <code style="white-space: pre;">&#8288;$class&#8288;</code> matching that of
the <code style="white-space: pre;">&#8288;get_*&#8288;</code> function being called. For example, you can only pass a
<code>taxon_state</code> with <code style="white-space: pre;">&#8288;$class&#8288;</code> of <code>gbifid</code> to <code>get_gbifid()</code>, and so on.
</p>
</li>
<li><p> If you run <code>taxon_clear()</code> while a <code style="white-space: pre;">&#8288;get*&#8288;</code> function is running, you may
lose track of any state known to this package before it was cleared
</p>
</li></ul>

<p>See the internal method <a href="#topic+progressor">progressor</a> for information on how we control messages
in <code style="white-space: pre;">&#8288;get*&#8288;</code> functions
</p>


<h3>Value</h3>

<p><code>taxon_last()</code> returns an object of class <code>taxon_state</code>, the last
one used, else <code>NULL</code> if none found. <code>taxon_clear()</code> clears the saved state
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
spp &lt;- names_list("species", 3)
res &lt;- get_gbifid(spp)
z &lt;- taxon_last()
z
z$taxa_remaining()
z$taxa_completed()
z$count # active binding; no parens needed

# cleanup
taxon_clear()

## End(Not run)
</code></pre>

<hr>
<h2 id='theplantlist'>Lookup-table for family, genus, and species names for ThePlantList</h2><span id='topic+theplantlist'></span>

<h3>Description</h3>

<p>These names are from http://www.theplantlist.org, and are from
version 1.1 of their data. This data is used in the function
<code><a href="#topic+names_list">names_list()</a></code>. This is a randomly selected subset of the ~350K
accepted species names in Theplantlist.
</p>


<h3>Format</h3>

<p>A data frame with 10,000 rows and 3 variables:
</p>

<ul>
<li> <p><code>family</code> family name
</p>
</li>
<li> <p><code>genus</code> genus name
</p>
</li>
<li> <p><code>species</code> specific epithet name
</p>
</li></ul>



<h3>Source</h3>

<p>http://www.theplantlist.org
</p>

<hr>
<h2 id='tnrs'>Phylotastic Taxonomic Name Resolution Service.</h2><span id='topic+tnrs'></span>

<h3>Description</h3>

<p>Defunct - service was down far too much to be reliable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tnrs(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tnrs_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='tnrs_sources'>TNRS sources</h2><span id='topic+tnrs_sources'></span>

<h3>Description</h3>

<p>Defunct - service was down far too much to be reliable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tnrs_sources(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tnrs_sources_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='tol_resolve'>Resolve names using Open Tree of Life (OTL) resolver</h2><span id='topic+tol_resolve'></span>

<h3>Description</h3>

<p>Resolve names using Open Tree of Life (OTL) resolver
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tol_resolve(
  names = NULL,
  context_name = NULL,
  do_approximate_matching = TRUE,
  ids = NULL,
  include_suppressed = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tol_resolve_+3A_names">names</code></td>
<td>
<p>(character) taxon names to be queried</p>
</td></tr>
<tr><td><code id="tol_resolve_+3A_context_name">context_name</code></td>
<td>
<p>name of the taxonomic context to be searched
(length-one character vector). Must match (case sensitive) one
of the values returned by <code><a href="rotl.html#topic+tnrs_contexts">rotl::tnrs_contexts()</a></code>.</p>
</td></tr>
<tr><td><code id="tol_resolve_+3A_do_approximate_matching">do_approximate_matching</code></td>
<td>
<p>(logical) A logical indicating whether or not
to perform approximate string (a.k.a. &ldquo;fuzzy&rdquo;) matching. Using <code>FALSE</code>
will greatly improve speed. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="tol_resolve_+3A_ids">ids</code></td>
<td>
<p>An array of OTL ids to use for identifying names. These
will be assigned to each name in the names array. If ids is
provided, then ids and names must be identical in length.</p>
</td></tr>
<tr><td><code id="tol_resolve_+3A_include_suppressed">include_suppressed</code></td>
<td>
<p>(logical) Ordinarily, some quasi-taxa, such as
incertae sedis buckets and other non-OTUs, are suppressed from TNRS
results. If this parameter is true, these quasi-taxa are allowed as
possible TNRS results. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="tol_resolve_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <code>httr::POST</code> within
<code><a href="rotl.html#topic+tnrs_match_names">rotl::tnrs_match_names()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame summarizing the results of the query. The
original query output is appended as an attribute to the
returned object (and can be obtained using
<code>attr(object, "original_response")</code>).
</p>


<h3>Author(s)</h3>

<p>Francois Michonneau <a href="mailto:francois.michonneau@gmail.com">francois.michonneau@gmail.com</a>
Scott Chamberlain
</p>


<h3>References</h3>

<p>https://github.com/OpenTreeOfLife/germinator/wiki/TNRS-API-v3#match_names
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gnr_resolve">gnr_resolve()</a></code>, <code><a href="#topic+tnrs">tnrs()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tol_resolve(names=c("echinodermata", "xenacoelomorpha",
  "chordata", "hemichordata"))
tol_resolve(c("Hyla", "Salmo", "Diadema", "Nautilus"))
tol_resolve(c("Hyla", "Salmo", "Diadema", "Nautilus"),
  context_name = "Animals")

turducken_spp &lt;- c("Meleagris gallopavo", "Anas platyrhynchos",
  "Gallus gallus")
tol_resolve(turducken_spp, context_name="Animals")

## End(Not run)
</code></pre>

<hr>
<h2 id='tp_acceptednames'>Return all accepted names for a taxon name with a given id.</h2><span id='topic+tp_acceptednames'></span>

<h3>Description</h3>

<p>Function name changed to tp_accnames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_acceptednames(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_acceptednames_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='tp_accnames'>Return all accepted names for a taxon name with a given id.</h2><span id='topic+tp_accnames'></span>

<h3>Description</h3>

<p>Return all accepted names for a taxon name with a given id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_accnames(id, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_accnames_+3A_id">id</code></td>
<td>
<p>the taxon identifier code</p>
</td></tr>
<tr><td><code id="tp_accnames_+3A_key">key</code></td>
<td>
<p>Your Tropicos API key; See <a href="#topic+taxize-authentication">taxize-authentication</a>
for help on authentication</p>
</td></tr>
<tr><td><code id="tp_accnames_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List or dataframe.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tp_accnames(id = 25503923)
tp_accnames(id = 25538750)

# No accepted names found
tp_accnames(id = 25509881)

## End(Not run)
</code></pre>

<hr>
<h2 id='tp_classification'>Return all synonyms for a taxon name with a given id.</h2><span id='topic+tp_classification'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_classification(...)
</code></pre>

<hr>
<h2 id='tp_dist'>Return all distribution records for for a taxon name with a given id.</h2><span id='topic+tp_dist'></span>

<h3>Description</h3>

<p>Return all distribution records for for a taxon name with a given id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_dist(id, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_dist_+3A_id">id</code></td>
<td>
<p>the taxon identifier code</p>
</td></tr>
<tr><td><code id="tp_dist_+3A_key">key</code></td>
<td>
<p>Your Tropicos API key; See <a href="#topic+taxize-authentication">taxize-authentication</a>
for help on authentication</p>
</td></tr>
<tr><td><code id="tp_dist_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of two data.frame's, one named &quot;location&quot;, and one &quot;reference&quot;.
</p>


<h3>References</h3>

<p>http://services.tropicos.org/help?method=GetNameDistributionsXml
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Query using a taxon name Id
out &lt;- tp_dist(id = 25509881)
## just location data
head(out[['location']])
## just reference data
head(out[['reference']])

## End(Not run)
</code></pre>

<hr>
<h2 id='tp_namedistributions'>Return all distribution records for for a taxon name with a given id.</h2><span id='topic+tp_namedistributions'></span>

<h3>Description</h3>

<p>Function name changed to tp_dist.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_namedistributions(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_namedistributions_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='tp_namereferences'>Return all reference records for for a taxon name with a given id.</h2><span id='topic+tp_namereferences'></span>

<h3>Description</h3>

<p>Function name changed to tp_refs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_namereferences(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_namereferences_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='tp_refs'>Return all reference records for for a taxon name with a given id.</h2><span id='topic+tp_refs'></span>

<h3>Description</h3>

<p>Return all reference records for for a taxon name with a given id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_refs(id, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_refs_+3A_id">id</code></td>
<td>
<p>the taxon identifier code</p>
</td></tr>
<tr><td><code id="tp_refs_+3A_key">key</code></td>
<td>
<p>Your Tropicos API key; See <a href="#topic+taxize-authentication">taxize-authentication</a>
for help on authentication</p>
</td></tr>
<tr><td><code id="tp_refs_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List or dataframe.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tp_refs(id = 25509881)

## End(Not run)
</code></pre>

<hr>
<h2 id='tp_search'>Search Tropicos by scientific name, common name, or Tropicos ID.</h2><span id='topic+tp_search'></span>

<h3>Description</h3>

<p>Search Tropicos by scientific name, common name, or Tropicos ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_search(
  sci = NULL,
  com = NULL,
  nameid = NULL,
  orderby = NULL,
  sortorder = NULL,
  pagesize = NULL,
  startrow = NULL,
  type = NULL,
  key = NULL,
  name = NULL,
  commonname = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_search_+3A_sci">sci</code></td>
<td>
<p>A scientific name, e.g., &quot;poa annua&quot;. See Details.</p>
</td></tr>
<tr><td><code id="tp_search_+3A_com">com</code></td>
<td>
<p>A common name, e.g., &quot;annual blue grass&quot;</p>
</td></tr>
<tr><td><code id="tp_search_+3A_nameid">nameid</code></td>
<td>
<p>Your search string. e.g., &quot;25509881&quot;</p>
</td></tr>
<tr><td><code id="tp_search_+3A_orderby">orderby</code></td>
<td>
<p>Your search string. e.g., &quot;1&quot;</p>
</td></tr>
<tr><td><code id="tp_search_+3A_sortorder">sortorder</code></td>
<td>
<p>Your search string. e.g., &quot;ascending&quot;</p>
</td></tr>
<tr><td><code id="tp_search_+3A_pagesize">pagesize</code></td>
<td>
<p>Your search string. e.g., &quot;100&quot;</p>
</td></tr>
<tr><td><code id="tp_search_+3A_startrow">startrow</code></td>
<td>
<p>Your search string. e.g., &quot;1&quot;</p>
</td></tr>
<tr><td><code id="tp_search_+3A_type">type</code></td>
<td>
<p>Type of search, &quot;wildcard&quot; (default) will add a wildcard to
the end of your search string. &quot;exact&quot; will use your search string exactly.</p>
</td></tr>
<tr><td><code id="tp_search_+3A_key">key</code></td>
<td>
<p>Your Tropicos API key; See <a href="#topic+taxize-authentication">taxize-authentication</a>
for help on authentication</p>
</td></tr>
<tr><td><code id="tp_search_+3A_name">name</code></td>
<td>
<p>Deprecated, see <code>sci</code></p>
</td></tr>
<tr><td><code id="tp_search_+3A_commonname">commonname</code></td>
<td>
<p>Deprecated, see <code>com</code></p>
</td></tr>
<tr><td><code id="tp_search_+3A_...">...</code></td>
<td>
<p>Further args passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>More details on the <code>name</code> parameter: Tropicos will fail
if you include a period (<code>.</code>) in your name string, e.g., <code>var.</code>,
so we replace periods before the request is made to the Tropicos web service.
In addition, Tropicos for some reason doesn't want to see sub-specific rank
names like <code>var</code>/<code>subsp</code>, so remove those from your query.
</p>


<h3>Value</h3>

<p>List or dataframe.
</p>


<h3>References</h3>

<p>http://services.tropicos.org/help?method=SearchNameXml
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tp_search(sci = 'Poa annua')
tp_search(sci = 'Poa annua subsp. annua')
tp_search(sci = 'Poa annua var. annua')
tp_search(sci = 'Poa annua var annua')
tp_search(sci = 'Poa annua annua')

## End(Not run)
</code></pre>

<hr>
<h2 id='tp_summary'>Return summary data a taxon name with a given id.</h2><span id='topic+tp_summary'></span>

<h3>Description</h3>

<p>Return summary data a taxon name with a given id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_summary(id, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_summary_+3A_id">id</code></td>
<td>
<p>the taxon identifier code</p>
</td></tr>
<tr><td><code id="tp_summary_+3A_key">key</code></td>
<td>
<p>Your Tropicos API key; See <a href="#topic+taxize-authentication">taxize-authentication</a>
for help on authentication</p>
</td></tr>
<tr><td><code id="tp_summary_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tp_summary(id = 25509881)
tp_summary(id = 2700851)
tp_summary(id = 24900183)

## End(Not run)
</code></pre>

<hr>
<h2 id='tp_synonyms'>Return all synonyms for a taxon name with a given id.</h2><span id='topic+tp_synonyms'></span>

<h3>Description</h3>

<p>Return all synonyms for a taxon name with a given id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tp_synonyms(id, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tp_synonyms_+3A_id">id</code></td>
<td>
<p>the taxon identifier code</p>
</td></tr>
<tr><td><code id="tp_synonyms_+3A_key">key</code></td>
<td>
<p>Your Tropicos API key; See <a href="#topic+taxize-authentication">taxize-authentication</a>
for help on authentication</p>
</td></tr>
<tr><td><code id="tp_synonyms_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+HttpClient">crul::HttpClient</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>List or dataframe.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tp_synonyms(id = 25509881)

## End(Not run)
</code></pre>

<hr>
<h2 id='tpl_families'>Get The Plant List families.</h2><span id='topic+tpl_families'></span>

<h3>Description</h3>

<p>Get The Plant List families.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tpl_families(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tpl_families_+3A_...">...</code></td>
<td>
<p>(list) Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Requires an internet connection in order to connect to
&lt;www.theplantlist.org&gt;.
</p>


<h3>Value</h3>

<p>Returns a <code>data.frame</code> including the names of all families
indexed by The Plant List, and the major groups into which they fall
(i.e. Angiosperms, Gymnosperms, Bryophytes and Pteridophytes).
</p>


<h3>Author(s)</h3>

<p>John Baumgartner (johnbb@student.unimelb.edu.au)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tpl_get">tpl_get()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get a data.frame of plant families, with the group name 
# (Angiosperms, etc.)
head(tpl_families())

## End(Not run)
</code></pre>

<hr>
<h2 id='tpl_get'>Get The Plant List csv files.</h2><span id='topic+tpl_get'></span>

<h3>Description</h3>

<p>Get The Plant List csv files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tpl_get(x, family = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tpl_get_+3A_x">x</code></td>
<td>
<p>Directory to write csv files to.</p>
</td></tr>
<tr><td><code id="tpl_get_+3A_family">family</code></td>
<td>
<p>If you want just one, or &gt;1 family, but not all, list
them in a vector.</p>
</td></tr>
<tr><td><code id="tpl_get_+3A_...">...</code></td>
<td>
<p>(list) Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Throws a warning if you already have a directory of the one
provided, but still works. Writes to your home directory, change x
as needed.
</p>


<h3>Value</h3>

<p>Returns nothing to console, except a message and progress bar.
Writes csv files to x.
</p>


<h3>Author(s)</h3>

<p>John Baumgartner <a href="mailto:johnbb@student.unimelb.edu.au">johnbb@student.unimelb.edu.au</a>
</p>


<h3>References</h3>

<p>The Plant List http://www.theplantlist.org
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tpl_families">tpl_families()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Get a few families
dir &lt;- file.path(tempdir(), "abc")
tpl_get(dir, family = c("Platanaceae","Winteraceae"))
readLines(file.path(dir, "Platanaceae.csv"), n = 5)

# You can now get Gymnosperms as well
dir1 &lt;- file.path(tempdir(), "def")
tpl_get(dir1, family = c("Pinaceae","Taxaceae"))

# You can get mosses too!
dir2 &lt;- file.path(tempdir(), "ghi")
tpl_get(dir2, family = "Echinodiaceae")

# Get all families
## Beware, will take a while
## dir3 &lt;- file.path(tempdir(), "jkl")
## tpl_get("dir3)

## End(Not run)
</code></pre>

<hr>
<h2 id='tpl_search'>A light wrapper around the taxonstand fxn to call Theplantlist.org database.</h2><span id='topic+tpl_search'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tpl_search()
</code></pre>

<hr>
<h2 id='ubio_classification'>uBio classification</h2><span id='topic+ubio_classification'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ubio_classification(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ubio_classification_+3A_...">...</code></td>
<td>
<p>Parameters, ignored</p>
</td></tr>
</table>

<hr>
<h2 id='ubio_classification_search'>This function will return ClassificationBankIDs (hierarchiesIDs) that refer to the
given NamebankID</h2><span id='topic+ubio_classification_search'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ubio_classification_search(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ubio_classification_search_+3A_...">...</code></td>
<td>
<p>Parameters, ignored</p>
</td></tr>
</table>

<hr>
<h2 id='ubio_id'>Search uBio by namebank ID.</h2><span id='topic+ubio_id'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ubio_id(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ubio_id_+3A_...">...</code></td>
<td>
<p>Parameters, ignored</p>
</td></tr>
</table>

<hr>
<h2 id='ubio_ping'>uBio ping</h2><span id='topic+ubio_ping'></span>

<h3>Description</h3>

<p>uBio ping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ubio_ping()
</code></pre>

<hr>
<h2 id='ubio_search'>This function will return NameBankIDs that match given search terms</h2><span id='topic+ubio_search'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ubio_search(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ubio_search_+3A_...">...</code></td>
<td>
<p>Parameters, ignored</p>
</td></tr>
</table>

<hr>
<h2 id='ubio_synonyms'>Search uBio for taxonomic synonyms by hierarchiesID.</h2><span id='topic+ubio_synonyms'></span>

<h3>Description</h3>

<p>THIS FUNCTION IS DEFUNCT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ubio_synonyms(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ubio_synonyms_+3A_...">...</code></td>
<td>
<p>Parameters, ignored</p>
</td></tr>
</table>

<hr>
<h2 id='upstream'>Retrieve the upstream taxa for a given taxon name or ID.</h2><span id='topic+upstream'></span><span id='topic+upstream.default'></span><span id='topic+upstream.tsn'></span><span id='topic+upstream.ids'></span>

<h3>Description</h3>

<p>This function uses a while loop to continually collect taxa up to the
taxonomic rank that you specify in the <code>upto</code> parameter. You can get data
from ITIS (itis) only currently. There is no method exposed by itis
for getting taxa at a specific taxonomic rank, so we do it ourselves inside
the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upstream(...)

## Default S3 method:
upstream(sci_id, db = NULL, upto = NULL, rows = NA, x = NULL, ...)

## S3 method for class 'tsn'
upstream(sci_id, db = NULL, upto = NULL, ...)

## S3 method for class 'ids'
upstream(sci_id, db = NULL, upto = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="upstream_+3A_...">...</code></td>
<td>
<p>Further args passed on to <code><a href="#topic+itis_downstream">itis_downstream()</a></code></p>
</td></tr>
<tr><td><code id="upstream_+3A_sci_id">sci_id</code></td>
<td>
<p>Vector of taxa names (character) or IDs (character or numeric) to
query.</p>
</td></tr>
<tr><td><code id="upstream_+3A_db">db</code></td>
<td>
<p>character; database to query. One or both of <code>itis</code>. Note
that each taxonomic data source has their own identifiers, so that if you
provide the wrong <code>db</code> value for the identifier you could get a result, but
it will likely be wrong (not what you were expecting).</p>
</td></tr>
<tr><td><code id="upstream_+3A_upto">upto</code></td>
<td>
<p>What taxonomic rank to go down to. One of: 'superkingdom',
'kingdom', 'subkingdom','infrakingdom','phylum','division','subphylum',
'subdivision','infradivision', 'superclass','class','subclass','infraclass',
'superorder','order','suborder','infraorder','superfamily','family',
'subfamily','tribe','subtribe','genus','subgenus', 'section','subsection',
'species','subspecies','variety','form','subvariety','race', 'stirp',
'morph','aberration','subform', or 'unspecified'</p>
</td></tr>
<tr><td><code id="upstream_+3A_rows">rows</code></td>
<td>
<p>(numeric) Any number from 1 to infinity. If the default NA, all
rows are considered. Note that this parameter is ignored if you pass in a
taxonomic id of any of the acceptable classes: tsn.</p>
</td></tr>
<tr><td><code id="upstream_+3A_x">x</code></td>
<td>
<p>Deprecated, see <code>sci_id</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of data.frames with the upstream names of every
supplied taxa. You get an NA if there was no match in the database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
upstream('Pinus contorta', db = 'itis', upto = 'genus')

## End(Not run)
</code></pre>

<hr>
<h2 id='use_eol-defunct'>use eol</h2><span id='topic+use_eol-defunct'></span><span id='topic+use_eol'></span>

<h3>Description</h3>

<p>use eol
</p>


<h3>Usage</h3>

<pre><code class='language-R'>use_eol()
</code></pre>

<hr>
<h2 id='vascan_search'>Search the CANADENSYS Vascan API.</h2><span id='topic+vascan_search'></span>

<h3>Description</h3>

<p>Search the CANADENSYS Vascan API.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vascan_search(q, format = "json", raw = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vascan_search_+3A_q">q</code></td>
<td>
<p>(character) Can be a scientific name, a vernacular name or a VASCAN
taxon identifier (e.g. 861)</p>
</td></tr>
<tr><td><code id="vascan_search_+3A_format">format</code></td>
<td>
<p>(character) One of json (default) or xml.</p>
</td></tr>
<tr><td><code id="vascan_search_+3A_raw">raw</code></td>
<td>
<p>(logical) If TRUE, raw json or xml returned, if FALSE, parsed
data returned.</p>
</td></tr>
<tr><td><code id="vascan_search_+3A_...">...</code></td>
<td>
<p>(list) Further args passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that we lowercase all outputs in data.frame's, but when a
list is given back, we don't touch the list names.
</p>


<h3>Value</h3>

<p>json, xml or a list.
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>References</h3>

<p>API docs https://data.canadensys.net/vascan/api
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
vascan_search(q = "Helianthus annuus")
vascan_search(q = "Helianthus annuus", raw=TRUE)
vascan_search(q = c("Helianthus annuus", "Crataegus dodgei"), raw=TRUE)

# format type
## json
c &lt;- vascan_search(q = "Helianthus annuus", format="json", raw=TRUE)
library("jsonlite")
fromJSON(c, FALSE)

## xml
d &lt;- vascan_search(q = "Helianthus annuus", format="xml", raw=TRUE)
library("xml2")
xml2::read_xml(d)

# lots of names, in this case 50
splist &lt;- names_list(rank='species', size=50)
vascan_search(q = splist)

# Curl options
invisible(vascan_search(q = "Helianthus annuus", verbose = TRUE))

## End(Not run)
</code></pre>

<hr>
<h2 id='worms_downstream'>Retrieve all taxa names downstream in hierarchy for WORMS</h2><span id='topic+worms_downstream'></span>

<h3>Description</h3>

<p>Retrieve all taxa names downstream in hierarchy for WORMS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>worms_downstream(id, downto, intermediate = FALSE, start = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="worms_downstream_+3A_id">id</code></td>
<td>
<p>(integer) One or more AphiaID's</p>
</td></tr>
<tr><td><code id="worms_downstream_+3A_downto">downto</code></td>
<td>
<p>(character) The taxonomic level you want to go down to.
See examples below. The taxonomic level IS case sensitive, and you do have
to spell it correctly. See <a href="#topic+rank_ref_zoo">rank_ref_zoo</a> for spelling.</p>
</td></tr>
<tr><td><code id="worms_downstream_+3A_intermediate">intermediate</code></td>
<td>
<p>(logical) If <code>TRUE</code>, return a list of length two
with target taxon rank names, with additional list of data.frame's of
intermediate taxonomic groups. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="worms_downstream_+3A_start">start</code></td>
<td>
<p>(integer) Record number to start at</p>
</td></tr>
<tr><td><code id="worms_downstream_+3A_...">...</code></td>
<td>
<p>crul options passed on to <code><a href="worrms.html#topic+wm_children">worrms::wm_children()</a></code>, including the
parameters <code>marine_only</code> and <code>offset</code>, see <code>?worrms::wm_children</code> for
details</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame of taxonomic information downstream to family from e.g.,
Order, Class, etc., or if <code>intermediated=TRUE</code>, list of length two,
with target taxon rank names, and intermediate names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## the genus Gadus
worms_downstream(id = 125732, downto="species")
worms_downstream(id = 125732, downto="species", intermediate=TRUE)

worms_downstream(id = 51, downto="class")
worms_downstream(id = 51, downto="subclass", intermediate=TRUE)

worms_downstream(id = 105, downto="subclass")

# marine_only parameter
worms_downstream(545470, downto = "species")
worms_downstream(545470, downto = "species", marine_only = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='worrms_ranks'>WORMS ranks</h2><span id='topic+worrms_ranks'></span>

<h3>Description</h3>

<p>Created using <code>worrms::wm_ranks_id(-1)</code> on 2020-02-11.
</p>


<h3>Format</h3>

<p>A data frame with 216 rows and 2 variables:
</p>

<ul>
<li> <p><code>id</code>: rank id
</p>
</li>
<li> <p><code>rank</code>: rank name
</p>
</li></ul>



<h3>Details</h3>

<p>Present in taxize in the case where WORMS does not
return rank names - with this dataset we can fill
in rank information as long as rank ids are returned
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
