<!DOCTYPE html><html><head><title>Help for package TBEST</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TBEST}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#best'>
<p>An object of class &quot;best&quot;</p></a></li>
<li><a href='#LeafContent'>
<p>Find names of leaves belongigng to given branches of a hierarchical tree</p></a></li>
<li><a href='#leukemia'>
<p>Leukemia data</p></a></li>
<li><a href='#partition'>
<p>An object of class &quot;partition&quot;</p></a></li>
<li><a href='#PartitionTree'>
<p>Find the most detailed partition of a tree into tight branches.</p></a></li>
<li><a href='#plot.best'>
<p>Plot a dendrogam of a hierarchical cluster with branches labeled by their numbers and significance estimates of tightness.</p></a></li>
<li><a href='#SigTree'>
<p>Perform statistical analysis of tightness for branches of a hierarchical cluster.</p></a></li>
<li><a href='#T10'>
<p>Breast tumor single cells data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tree Branches Evaluated Statistically for Tightness</td>
</tr>
<tr>
<td>Version:</td>
<td>5.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-24</td>
</tr>
<tr>
<td>Author:</td>
<td>Guoli Sun, Alex Krasnitz</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Guoli Sun &lt;guolisun87@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Our method introduces mathematically well-defined measures for tightness of branches in a hierarchical tree. Statistical significance of the findings is determined, for all branches of the tree, by performing permutation tests, optionally with generalized Pareto p-value estimation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>parallel, signal, fdrtool, graphics, stats, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-24 20:10:02 UTC</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-24 17:14:40 UTC; Astrid</td>
</tr>
</table>
<hr>
<h2 id='best'>
An object of class &quot;best&quot;
</h2><span id='topic+best'></span>

<h3>Description</h3>

<p>Description: This object is a list of three items. It contains a statistical assessment of the tightness of branches in a hierarchical tree.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Call</code></td>
<td>

<p>An object of class Call, specifying the parameters used.
</p>
</td></tr>
<tr><td><code>data</code></td>
<td>

<p>A matrix from which the distance matrix used for growing the tree is computed, with the rows corresponding to the items being clustered.
</p>
</td></tr>
<tr><td><code>indextable</code></td>
<td>

<p>If measure of tightness is not <code>"slb"</code>, this is a matrix with the number of rows one less than the number of items being clustered.  Each row corresponds to an internal node in the tree. The columns are as follows. First two columns specify the merging order of the tree, as in the <code>merge</code> component of the class <code>hclust</code>. The third column contains the node heights, as in the <code>height</code> component of <code>hclust</code>. The fourth column provides the number of leaves for each node. The corresponding column names are <code>"index1"</code>, <code>"index2"</code>, <code>"height"</code>, <code>"clustersize"</code>. The remaining columns come in pairs. If the name of the first column in a pair is <code>"x"</code>, the name of the second one is <code>"px"</code>. The first column in each pair tabulates a mesure of tightness; the second column provides the corresponding p-value. If measure of tightness is <code>"slb"</code>, this is a list with two variable, a matrix like above except without column of p-value and a p-value suggesting the significance of two-way split of input data.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guoli Sun, Alex Krasnitz
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SigTree">SigTree</a></code>,<code><a href="#topic+plot.best">plot.best</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(leukemia)
mytable&lt;-SigTree(data.matrix(leukemia),mystat="all",
       mymethod="ward",mymetric="euclidean",rand.fun="shuffle.column",
       distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
class(mytable)
names(mytable)
mytable&lt;-SigTree(data.matrix(leukemia),mystat="slb",
       mymethod="ward",mymetric="euclidean",rand.fun="shuffle.column",
       distrib="Rparallel",njobs=2,Ptail=FALSE)
class(mytable)
names(mytable)

## End(Not run)
</code></pre>

<hr>
<h2 id='LeafContent'>
Find names of leaves belongigng to given branches of a hierarchical tree
</h2><span id='topic+LeafContent'></span>

<h3>Description</h3>

<p>Description: find the names of all items comprising one or more branches of a hierarchical tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LeafContent(myinput, mynode=NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LeafContent_+3A_myinput">myinput</code></td>
<td>

<p>An object of class <code>hclust</code>, <code><a href="#topic+best">best</a></code> or <code><a href="#topic+partition">partition</a></code>.
</p>
</td></tr>
<tr><td><code id="LeafContent_+3A_mynode">mynode</code></td>
<td>

<p>An integer vector of the numbers of branches whose leaf content is desired. The <code>hclust</code> convention is used for numbering branches and leaves, i.e., the branch numbers can take any value between (-N) and (N-1) excluding 0, where N is the number of leaves in the tree. A negative value refers to an individual leaf whose number is minus that value. If <code>myinput</code> is of class <code>partition</code>, this argument is ignored. The function lists the leaf content for each of the branches that form the partition.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of items, of the same length as <code>mynode</code>. Each item corresponds to a branch listed in <code>myneode</code> and is a character vector containing the names of the leaves in the branch.
</p>


<h3>Author(s)</h3>

<p>Guoli Sun, Alex Krasnitz
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukemia)
hc&lt;-hclust(dist(data.matrix(leukemia)),"ward")
#find the name of leaf 29
LeafContent(hc,mynode=c(-29))
#find the name of leaf 29 and leaves belonging to node 29
LeafContent(hc,mynode=c(-29,29))
## Not run: 
mytable&lt;-SigTree(data.matrix(leukemia),mystat="fldc",
        mymethod="ward",mymetric="euclidean",rand.fun="shuffle.column",
        distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
LeafContent(mytable,mynode=c(-29,29))
mypartition&lt;-PartitionTree(x=mytable,siglevel=0.001,statname="fldc",sigtype="raw")
LeafContent(mypartition)

## End(Not run)
</code></pre>

<hr>
<h2 id='leukemia'>
Leukemia data
</h2><span id='topic+leukemia'></span>

<h3>Description</h3>

<p>This data set represents mRNA expression of 500 genes in 38 patient cases of leukemia. These 38 cases fall into 3 subtypes: AML (11), T-lineage ALL (8) and B-lineage ALL (19). The set was obtained by removing 499 genes from Golub's leukemia data, to facilitate the execution of examples for this package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(leukemia)</code></pre>


<h3>Format</h3>

<p>A data frame with 38 observations (rows) of 500 variables (columns).
</p>


<h3>Details</h3>

<p>Bone marrow samples obtained from acute leukemia patients at the time of diagnosis.
</p>


<h3>Source</h3>

<p>http://www.broadinstitute.org/cgi-bin/cancer/datasets.cgi
</p>


<h3>References</h3>

<p>T.R. Golub, D.K. Slonim et al(1999)
Molecular Classification of Cancer: Class Discovery and Class Prediction by Gene Expression;				
</p>
<p>Stefano Monti, Pablo Tamayo, Jill Mesirov, and Todd Golub(2003)
Consensus Clustering: A resampling-based method for class discovery and visualization of gene expression microarray data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukemia)
dim(leukemia)
</code></pre>

<hr>
<h2 id='partition'>
An object of class &quot;partition&quot;
</h2><span id='topic+partition'></span>

<h3>Description</h3>

<p>Description: This object is a list of four items, which jointly specify a detailed partition of a hierarchical tree into tight branches.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Call</code></td>
<td>

<p>An object of class Call, specifying the function call which generated the list.
</p>
</td></tr>
<tr><td><code>best</code></td>
<td>

<p>An object of class &quot;best&quot;, see <code><a href="#topic+best">best</a></code> for more info.
</p>
</td></tr>
<tr><td><code>sigvalue</code></td>
<td>

<p>A two-column matrix, with one row per each internal node of the tree. The first column enumerates the nodes. The second column profides the significance estimate for the tightness of the node.
</p>
</td></tr>
<tr><td><code>partition</code></td>
<td>

<p>A two-column data frame specifying the partitition. The first column is a character vector with the names of the leaves. The second column provides the number of the part to which the leaf belongs.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guoli Sun, Alex Krasnitz
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PartitionTree">PartitionTree</a></code>,<code><a href="#topic+best">best</a></code>,<code><a href="#topic+SigTree">SigTree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(leukemia)
mytable&lt;-SigTree(data.matrix(leukemia),mystat="all",
       mymethod="ward",mymetric="euclidean",rand.fun="shuffle.column",
       distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
class(mytable)
mypartition&lt;-PartitionTree(x=mytable,siglevel=0.001,statname="fldc",
       sigtype="raw")
class(mypartition)
names(mypartition)

## End(Not run)
</code></pre>

<hr>
<h2 id='PartitionTree'>
Find the most detailed partition of a tree into tight branches.
</h2><span id='topic+PartitionTree'></span>

<h3>Description</h3>

<p>Description: The function finds the most detailed partition of a hierartchical tree into tight branches, given a level of significance for tightness.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PartitionTree(x,siglevel=0.05,statname="fldc",
        sigtype=c("raw","corrected","fdr"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PartitionTree_+3A_x">x</code></td>
<td>

<p>An object of class <code><a href="#topic+best">best</a></code>, such as computed by function <code><a href="#topic+SigTree">SigTree</a></code>.
</p>
</td></tr>
<tr><td><code id="PartitionTree_+3A_siglevel">siglevel</code></td>
<td>

<p>Threshold of significance for tightness of branches. Default is 0.05.
</p>
</td></tr>
<tr><td><code id="PartitionTree_+3A_statname">statname</code></td>
<td>

<p>A character string specifying the name of measure of tighness whose is significance is to be used for partition. The choices are <code>"fldc"</code>(default),<code>"bldc"</code>,<code>"fldcc"</code>.
</p>
</td></tr>
<tr><td><code id="PartitionTree_+3A_sigtype">sigtype</code></td>
<td>

<p>A character string specifying how the significance threshold <code>siglevel</code> should be interpreted. If <code>"raw"</code>, the threshold will be applied directly to the p-values tabulated for each tree node in <code>x</code>. With <code>"corrected"</code> chosen, the threshold will be applied to the p-values corrected for multiplicity: <br /> <code>p_cor = 1 - (1 - p)^(N - 2)</code>, where <code>N</code> is the number of leaves in the tree. of significance. If <code>"fdr"</code>, <code>siglevel</code> is interpreted as a threshold on false discovery rate.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="#topic+partition">partition</a></code>. See <code>?partition</code> for details.
</p>


<h3>Author(s)</h3>

<p>Guoli Sun, Alex Krasnitz
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SigTree">SigTree</a></code>,<code><a href="#topic+partition">partition</a></code>,<code><a href="#topic+best">best</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(leukemia)
mytable&lt;-SigTree(data.matrix(leukemia),mystat="all",
        mymethod="ward",mymetric="euclidean",rand.fun="shuffle.column",
        distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
mypartition&lt;-PartitionTree(x=mytable,siglevel=0.001,statname="fldc",
        sigtype="raw")
partition1&lt;-mypartition$partition
sigmatrix1&lt;-mypartition$sigvalue
fix(partition1)
fix(sigmatrix1)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.best'>
Plot a dendrogam of a hierarchical cluster with branches labeled by their numbers and significance estimates of tightness.
</h2><span id='topic+plot.best'></span><span id='topic+plot'></span>

<h3>Description</h3>

<p>Description: A plot method for the class <code><a href="#topic+best">best</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'best'
plot(x,mystat="fldc",siglevel=0.05,sigtype=c("raw","corrected","fdr"),
        partition=NA,print.num=TRUE,print.lab=TRUE,float=0.01,col.best=c(2,3),
        cex.best=0.8,cex.leaf=0.8,font.best=NULL,main=NULL,sub=NULL,xlab=NULL,
        metric.args=list(),...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.best_+3A_x">x</code></td>
<td>

<p>An object of class <code><a href="#topic+best">best</a></code>, such as computed by the <code><a href="#topic+SigTree">SigTree</a></code> function.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_mystat">mystat</code></td>
<td>

<p>A measure of tightness for which p-values are to be shown in the plot. Default is <code>"fldc"</code>. Other options are <code>"fldcc"</code> and <code>"bldc"</code>.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_siglevel">siglevel</code></td>
<td>

<p>A threshold level of significance for tightness of branches used when <code>partition=NA</code>. Default is 0.05. If the estimate of significance for a node is below threshold, it will be shown on the plot next to the node.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_sigtype">sigtype</code></td>
<td>

<p>A character string specifying how the significance threshold <code>siglevel</code> should be interpreted. If <code>"raw"</code>, the threshold will be applied directly to the p-values tabulated for each tree node in <code>x</code>. With <code>"corrected"</code> chosen, the threshold will be applied to the p-values corrected for multiplicity:<br /> <code>p_cor = 1 - (1 - p)^(N - 2)</code>, where <code>N</code> is the number of leaves in the tree. of significance. If <code>"fdr"</code>, <code>siglevel</code> is interpreted as a threshold on false discovery rate.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_partition">partition</code></td>
<td>

<p>An object of class <code><a href="#topic+partition">partition</a></code>, such as computed by the <code><a href="#topic+PartitionTree">PartitionTree</a></code> function.
</p>
</td></tr> 
<tr><td><code id="plot.best_+3A_print.num">print.num</code></td>
<td>

<p>Logical. If true, the branch numbers will be indicated.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_print.lab">print.lab</code></td>
<td>

<p>Logical. If true, the labels will be displayed at the bottom of dendrogram.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_float">float</code></td>
<td>

<p>A numeric value that can change the vertical location of pvalues.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_col.best">col.best</code></td>
<td>

<p>A character vector of length 2, indicating the colors to be used for the p-values and for the numbers of the nodes.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_cex.best">cex.best</code></td>
<td>

<p>A numeric value for the text size of the branch labels.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_cex.leaf">cex.leaf</code></td>
<td>

<p>A numeric value for the text size of the leaf labels.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_font.best">font.best</code></td>
<td>

<p>An integer which specifies font choice of text on the plot. See <code>?par</code> function parameter <code>font</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_main">main</code></td>
<td>

<p>A character string specifying the title of the plot.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_sub">sub</code></td>
<td>

<p>A character string specifying a subtitle of the plot.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_xlab">xlab</code></td>
<td>

<p>A caracter string specifying the label of horizontal axis.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_metric.args">metric.args</code></td>
<td>

<p>Additional argument from user supplied dissimilariity(distance) function. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td></tr>
<tr><td><code id="plot.best_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed on to the <code>plot</code> function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function plots a dendrogram of the hierarchical tree as specified by the <code>x</code> argument, an object of class <code>"best"</code>. When argument <code>partition</code> is set to an object of class <code>"partition"</code>, and a partition does exist (see <code><a href="#topic+partition">partition</a></code> for description), this plot provides the significance estimates for the nodes that form the partition. Otherwise, this function puts legends on all tight nodes with significance estimates no more than <code>siglevel</code>. To obtain the leaves descending from a given node, refer to function <code><a href="#topic+LeafContent">LeafContent</a></code>.
</p>


<h3>Value</h3>

<p>A plot with all branch numbers and significant pvalues in the hierarchical tree.
</p>


<h3>Author(s)</h3>

<p>Guoli Sun, Alex Krasnitz
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SigTree">SigTree</a></code>, <code><a href="#topic+PartitionTree">PartitionTree</a></code>,<code><a href="#topic+best">best</a></code>,<code><a href="#topic+partition">partition</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(leukemia)
mytable&lt;-SigTree(data.matrix(leukemia),mystat="all",
        mymethod="ward",mymetric="euclidean",rand.fun="shuffle.column",
        distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
plot(x=mytable,mystat="fldc",siglevel=0.001,sigtype="raw",hang=-1)
mypartition&lt;-PartitionTree(x=mytable,siglevel=0.001,statname="fldc",
        sigtype="raw")
plot(x=mytable,mystat="fldc",partition=mypartition)
plot(x=mytable,mystat="fldc",partition=mypartition,print.num=F)
#with user-defined functions
mydist&lt;-function(x,y){return(dist(x)/y)}
myrand&lt;-function(x,z){return(apply(x+z,2,sample))}
mytable&lt;-SigTree(data.matrix(leukemia),mystat="fldc",
	mymethod="ward",mymetric="mydist",rand.fun="myrand",
	distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="MOM",metric.args=list(3),
	rand.args=list(2))
plot(mytable,metric.args=list(3))
plot(mytable,metric.args=list(3),cex.leaf=1.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='SigTree'>
Perform statistical analysis of tightness for branches of a hierarchical cluster.
</h2><span id='topic+SigTree'></span>

<h3>Description</h3>

<p>Description: Given data from which a hierarchical tree is grown, compute measures of tightness for each branch, sample from the null distribution of these measures in the randomized data and compute the corresponding p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SigTree(myinput,mystat=c("all","fldc","bldc","fldcc","slb"),
        mymethod="complete",mymetric="euclidean",rand.fun=NA,
        by.block=NA,distrib=c("vanilla","Rparallel"),Ptail=TRUE,
        tailmethod=c("ML","MOM"),njobs=1,seed=NA,
        Nperm=ifelse(Ptail,1000,1000*nrow(myinput)),
        metric.args=list(),rand.args=list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SigTree_+3A_myinput">myinput</code></td>
<td>

<p>A matrix with rows corresponding to items to be clustered.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_mystat">mystat</code></td>
<td>

<p>A character string specifying the measures of tightness to be computed and evaluated for significance of finding. See Details for the definitions of these measures. If <code>"all"</code> is chosen, all the first three measures, <code>"fldc"</code>, <code>"bldc"</code> and <code>"fldcc"</code>, and the corresponding p-values are computed. Otherwise, only the specified measure and its p-value are computed.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_mymethod">mymethod</code></td>
<td>

<p>A character string specifying the linkage method for hierarchical clustering, to be used by the <code>hclust</code> function. See <code>hclust</code> argument <code>method</code> for method options.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_mymetric">mymetric</code></td>
<td>

<p>A character string specifying the definition of dissimilarity (distance) among the data items. The options, in addition to those for the argument <code>method</code> of the <code>dist</code> functiton, are <code>"pearson"</code>,<code>"kendall"</code>, and <code>"spearman"</code>. If one of the latter three is chosen, the distances are computed as <code>as.dist(1 - </code> 
<br /> <code>cor(myinput))</code>, with the corresponding option for the <code>method</code> argument of the <code>cor</code> function.It can also be a character string specifying a user supplied dissimilarity (distance) function for <code>myinput</code>. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_rand.fun">rand.fun</code></td>
<td>

<p>A character string specifying the permutation method to be applied to <code>myinput</code>. If NA(default), no permutation is performed. <code>"shuffle.column"</code> performs a random permutation independently within each column. With <code>"shuffle.block"</code>, a random permutation is performed independently within each block of columns, as specified by the <code>by.block</code> argument, and independently from the other blocks. It can also be a character string specifying a user supplied randomization function for <code>myinput</code>. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_by.block">by.block</code></td>
<td>

<p>A vector of the same length as the column dimension of <code>myinput</code>, to specify the blocking of columns of <code>myinput</code>. It is used in conjunction with <code>rand.fun</code>
<br /> <code>= "shuffle.block"</code>, and is ignored otherwise.  
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_distrib">distrib</code></td>
<td>

<p>One of <code>"vanilla", "Rparallel"</code> to specify the distributed computing option for the cluster assignment step. For <code>"vanilla"</code> (default)
no distributed computing is performed. For <code>"Rparallel"</code> the <code>parallel</code> package of <code>R</code> core is used for multi-core processing.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_ptail">Ptail</code></td>
<td>

<p>Logical. If <code>Ptail</code> is TRUE(default), the Generalized Pareto Distribution is used to approximate the tail of the null distribution for each of the chosen measures. Otherwise, empirical p-values are computed directly from the corresponding samples.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_tailmethod">tailmethod</code></td>
<td>

<p>A character string only needed to be specified if the <code>Ptail</code> is set to TRUE. For <code>"ML"</code> the parameters of the Generalized Pareto Distribution are estimated by likelihood maximization; for <code>"MOM"</code> they are estimated by the method of moments.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_njobs">njobs</code></td>
<td>

<p>A single integer specifying the number of worker jobs to create in case of distributed computation if <code>distrib = "Rparallel"</code>; ignored otherwise.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_seed">seed</code></td>
<td>

<p>An optional single integer value, to be used to set the random number generator seed (see <code>details</code>).
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_nperm">Nperm</code></td>
<td>

<p>A single integer specifying the size of a sample from the null distribution. See <code>details</code> for the default sample size.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_metric.args">metric.args</code></td>
<td>

<p>Additional arguments for user-supplied dissimilarity (distance) function. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td></tr>
<tr><td><code id="SigTree_+3A_rand.args">rand.args</code></td>
<td>

<p>Additional arguments for user-supplied randomization function. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When <code>rand.fun</code> is set to the name of a user supplied randomization function, the first argument of that function should be set to <code>myinput</code>. See examples below.
</p>
<p>The measures of tightness are defined as follows. Denote a node in the tree by <code>a</code>, its sibling node by <code>b</code>, and their parent node by <code>p</code>. Let their respective geights be <code>ha</code>,<code>hb</code>,<code>hp</code>. Finally, let <code>Sx</code> mean that the measure <code>S</code> is computed for the node <code>x</code>. Then the definitions are
</p>
<p>fldc:
</p>
<p><code>Sa = (hp-ha)/hp</code>
</p>
<p>fldcc:
</p>
<p><code>Sa = (hp-(ha-hb)/2)/ha</code>
</p>
<p>bldc:
</p>
<p><code>Sp = (2*hp-ha-hb)/(2*hp)</code>
</p>
<p>slb:
</p>
<p><code>Sp = 2*hp-ha-hb</code>
</p>
<p>The first three measures test tightnss of all internal nodes at the same time, while <code>slb</code> only tests two-way split of input data. 
The <code>seed</code> argument is optional. Setting the seed ensures reproducibility of sampling from the null distribution.
</p>


<h3>Value</h3>

<p>If <code>rand.fun</code> is set to NA, the function returns a matrix whose rows correspond to the internal nodes of the tree and whose columns contain the tree structure as in the <code>merge</code> component of the class <code>hclust</code>; the <code>height</code> component of <code>hclust</code>;and columns tabulating the values of the measures of tightness specified by the <code>mystat</code> argument.
If <code>rand.fun</code> is set to a specific randomization method, an object of class <code><a href="#topic+best">best</a></code> is returned. See <code>?best</code> for details.
</p>


<h3>Note</h3>

<p>If <code>mymetric</code> or <code>rand.fun</code> is a customized function, make sure you have read and write permission for your working directory.
</p>


<h3>Author(s)</h3>

<p>Guoli Sun, Alex Krasnitz
</p>


<h3>References</h3>

<p>Theo A. Knijnenburg, Lodewyk F. A. Wessels et al (2009)
Fewer permutations, more accurate P-values
</p>


<h3>See Also</h3>

<p><code><a href="#topic+best">best</a></code>,<code><a href="#topic+plot.best">plot.best</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>####Each column is a gene expression profile for a case of leukemia. 
####Each case belongs to one of three subtypes.
data(leukemia)
#output only statistic table
mytable&lt;-SigTree(data.matrix(leukemia),mystat="all",
        mymethod="ward",mymetric="euclidean")
class(mytable)
## Not run: 
#use multicore processing to detect significant sub-clusters
mytable&lt;-SigTree(data.matrix(leukemia),mystat="all",
	mymethod="ward",mymetric="euclidean",rand.fun="shuffle.column",
	distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
class(mytable)
####Each row after the 1st describes an item belonging to one of four subtypes. 
####Each column corresponds to a genomic location in one of 22 human chromosomes. 
####The 1st row contains the chromosome numbers.
data(T10)
#Perform randomization within each chromosome
chrom&lt;-as.numeric(T10[1,])
mydata&lt;-T10[-1,] 
mytable&lt;-SigTree(data.matrix(mydata),mystat="fldc",        
	mymethod="ward",mymetric="euclidean",rand.fun="shuffle.block",
	by.block=chrom,distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
#Compute dissimilarity using a user-supplied distance function,
#and perform randomization using a user-supplied randomization function, 
#with additional arguments. 
#Both user-supplied functions are only useful as illustration.
mydist&lt;-function(x,y){return(dist(x)/y)}
myrand&lt;-function(x,z){return(apply(x+z,2,sample))}
mytable&lt;-SigTree(data.matrix(leukemia),mystat="fldc",
mymethod="ward",mymetric="mydist",rand.fun="myrand",
distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="MOM",metric.args=list(3),
rand.args=list(2))

## End(Not run)
</code></pre>

<hr>
<h2 id='T10'>
Breast tumor single cells data
</h2><span id='topic+T10'></span>

<h3>Description</h3>

<p>This data set summarizes DNA copy number variation in 100 individual cancer cells harvested from a breast tumor. The cells belong to four subtypes, differing by ploidy.There are 47 Diploid+Pseudo-diploid, 24 Hypo-diploid, 4 Aneuploid B and 25 Aneuploid A cells. Their copy number profiles are summarized in terms of 354 amplification and deletion &quot;cores&quot;, are computed by the CORE package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(T10)</code></pre>


<h3>Format</h3>

<p>A data frame with 101 rows and 354 columns. Each column corresponds to a core.  The first row is integer and contains the chromosome number for each core. The remaining rows are numeric, with values between 0 and 1, and each represents a DNA copy number profile of a cell.
</p>


<h3>Details</h3>

<p>Please remove the first row before computing the distance matrix.
</p>


<h3>Source</h3>

<p>Alexander Krasnitz, Guoli Sun, Peter Andrews, and Michael Wigler(2013)
Target inference from collections of genomic intervals
</p>


<h3>References</h3>

<p>Alexander Krasnitz, Guoli Sun, Peter Andrews, and Michael Wigler(2013)
Target inference from collections of genomic intervals		
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(T10)
dim(T10)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
