<!DOCTYPE html><html><head><title>Help for package TCGAretriever</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TCGAretriever}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TCGAretriever-package'><p>Retrieve Genomic and Clinical Data from CBioPortal</p></a></li>
<li><a href='#blcaOutputExamples'><p>TCGAretriever Examples</p></a></li>
<li><a href='#cb_query'><p>Core Query Engine</p></a></li>
<li><a href='#expand_cases'><p>Samples Included in a List of Samples.</p></a></li>
<li><a href='#fetch_all_tcgadata'><p>Fetch All Molecular Data for a Cancer Profile of Interest.</p></a></li>
<li><a href='#get_cancer_studies'><p>Retrieve the List of Cancer Studies Available at cbioportal.</p></a></li>
<li><a href='#get_cancer_types'><p>Retrieve Cancer Types.</p></a></li>
<li><a href='#get_case_lists'><p>Retrieve Case List Available for a Specific Cancer Study.</p></a></li>
<li><a href='#get_clinical_data'><p>Retrieve Clinical Information for a Cancer Study.</p></a></li>
<li><a href='#get_gene_identifiers'><p>Retrieve All Gene Identifiers</p></a></li>
<li><a href='#get_genetic_profiles'><p>Retrieve Genetic Profiles for a TCGA Study of Interest</p></a></li>
<li><a href='#get_molecular_data'><p>Retrieve Molecular Data corresponding to a Genetic Profile of Interest.</p></a></li>
<li><a href='#get_mutation_data'><p>Retrieve Mutation Data corresponding to a Genetic Profile of Interest.</p></a></li>
<li><a href='#make_groups'><p>Split Numeric Vectors in Groups.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Retrieve Genomic and Clinical Data from CBioPortal Including
TCGA Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.9.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Damiano Fantini</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Damiano Fantini &lt;damiano.fantini@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>httr, reshape2, jsonlite</td>
</tr>
<tr>
<td>Suggests:</td>
<td>stats, utils, knitr, rmarkdown, ggplot2, dplyr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Description:</td>
<td>The Cancer Genome Atlas (TCGA) is a program aimed at improving our understanding of Cancer Biology. Several TCGA Datasets are available online. 'TCGAretriever' helps accessing and downloading TCGA data hosted on 'cBioPortal' via its Web Interface (see <a href="https://www.cbioportal.org/">https://www.cbioportal.org/</a> for more information).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-23 20:51:47 UTC; dami</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-23 21:52:48 UTC</td>
</tr>
</table>
<hr>
<h2 id='TCGAretriever-package'>Retrieve Genomic and Clinical Data from CBioPortal</h2><span id='topic+TCGAretriever'></span><span id='topic+TCGAretriever-package'></span>

<h3>Description</h3>

<p>The Cancer Genome Atlas (TCGA) is a scientific and medical program 
aimed at improving our understanding of Cancer Biology. 
Part of the TCGA Datasets (free-access tier) are hosted
on cBioPortal, which is an open-access, open-source resource for 
interactive exploration of multidimensional cancer genomics data sets. 
TCGAretriever helps accessing and downloading TCGA data via the 
cBioPortal API. Features of TCGAretriever are: 
</p>

<ul>
<li><p> it is simple and reliable
</p>
</li>
<li><p> it is tailored for downloading large volumes of data
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>
</li>
<li> <p><a href="http://www.cbioportal.org/">http://www.cbioportal.org/</a> 
</p>
</li>
<li> <p><a href="https://www.cancer.gov/ccg/research/genome-sequencing/tcga">https://www.cancer.gov/ccg/research/genome-sequencing/tcga</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># List available Adenoid Cystic Carcinoma (acyc) Studies.
# Set `dryrun = FALSE` (default option) in production!
all_studies &lt;- get_cancer_studies(dryrun = TRUE)
acyc_ids &lt;- grep('acyc', all_studies$studyId, value = TRUE)
print(acyc_ids)


# List blca_tcga profiles.
# Set `dryrun = FALSE` (default option) in production!
get_genetic_profiles(csid = 'blca_tcga', dryrun = TRUE)


# List blca_tcga case lists.
# Set `dryrun = FALSE` (default option) in production!
get_case_lists(csid = 'blca_tcga', dryrun = TRUE)


# Retrieve expression data.
# Set `dryrun = FALSE` (default option) in production!
my_genes &lt;- c('PTEN', 'TP53')
get_molecular_data(case_list_id = 'blca_tcga_3way_complete', 
                   gprofile_id = 'blca_tcga_rna_seq_v2_mrna', 
                   glist = my_genes, dryrun = TRUE)


</code></pre>

<hr>
<h2 id='blcaOutputExamples'>TCGAretriever Examples</h2><span id='topic+blcaOutputExamples'></span>

<h3>Description</h3>

<p>A list of objects including examples of the output returned by 
different 'TCGAretriever' functions. The objects were obtained
from the '&quot;blca_tcga&quot;' study (bladder cancer).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(blcaOutputExamples)
</code></pre>


<h3>Format</h3>

<p>A list including 7 elements.
</p>

<dl>
<dt>exmpl_1</dt><dd><p>data.frame (dimensions: 10 by 13). Sample output of the 'get_cancer_studies()' function.</p>
</dd>
<dt>exmpl_2</dt><dd><p>data.frame (dimensions: 10 by 5). Sample output of the 'get_cancer_types()' function.</p>
</dd>
<dt>exmpl_3</dt><dd><p>data.frame (dimensions: 9 by 5). Sample output of the 'get_case_lists()' function.</p>
</dd>
<dt>exmpl_4</dt><dd><p>list including 9 elements. Sample output of the 'expand_cases()' function.</p>
</dd>
<dt>exmpl_5</dt><dd><p>data.frame (dimensions: 10 by 94). Sample output of the 'get_clinical_data()' function.</p>
</dd>
<dt>exmpl_6</dt><dd><p>data.frame (dimensions: 9 by 8). Sample output of the 'get_genetic_profiles()' function.</p>
</dd>
<dt>exmpl_7</dt><dd><p>data.frame (dimensions: 10 by 3). Sample output of the 'get_gene_identifiers()' function.</p>
</dd>
<dt>exmpl_8</dt><dd><p>data.frame (dimensions: 2 by 10). Sample output of the 'get_molecular_data()' function.</p>
</dd>
<dt>exmpl_9</dt><dd><p>data.frame (dimensions: 6 by 27). Sample output of the 'get_mutation_data()' function.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The object was built using the following lines of code.
<code>
blcaOutputExamples &lt;- list(
  exmpl_1 = head(get_cancer_studies(), 10),
  exmpl_2 = head(get_cancer_types(), 10),
  exmpl_3 = head(get_case_lists("blca_tcga"), 10)  ,
  exmpl_4 = expand_cases("blca_tcga"),
  exmpl_5 = head(get_clinical_data("blca_tcga"), 10),
  exmpl_6 = head(get_genetic_profiles("blca_tcga") , 10),
  exmpl_7 = head(get_gene_identifiers(), 10), 
  exmpl_8 = get_molecular_data(case_list_id = 'blca_tcga_3way_complete',
                               gprofile_id = 'blca_tcga_rna_seq_v2_mrna',
                               glist = c("TP53", "E2F1"))[, 1:10],
  exmpl_9 = head(get_mutation_data(case_list_id = 'blca_tcga_sequenced',
                                   gprofile_id = 'blca_tcga_mutations',
                                   glist = c('TP53', 'PTEN'))))
</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(blcaOutputExamples)
blcaOutputExamples$exmpl_1

</code></pre>

<hr>
<h2 id='cb_query'>Core Query Engine</h2><span id='topic+cb_query'></span>

<h3>Description</h3>

<p>Submit a query by getting or posting to the URL provided as argument 
(typically a cbioportal.org URL). The function attempts the same query recursively
until the content has been completely downloaded. 
If 'body' is NULL, the query is submitted via GET. Otherwise, 
the query is submitted via POST.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cb_query(my_url, body = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cb_query_+3A_my_url">my_url</code></td>
<td>
<p>String. The URL pointing to the cBioPortal API.</p>
</td></tr>
<tr><td><code id="cb_query_+3A_body">body</code></td>
<td>
<p>String. Parameters to be passed via POST to the query URL. Can be NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a core function invoked by other functions in the package.
</p>


<h3>Value</h3>

<p>Data.frame including data retrieved from cBioPortal.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The example below requires an active Internet connection.
## Not run: 
my_url &lt;- "https://www.cbioportal.org/api/studies"
x &lt;- TCGAretriever:::cb_query(my_url)

## End(Not run)
 
</code></pre>

<hr>
<h2 id='expand_cases'>Samples Included in a List of Samples.</h2><span id='topic+expand_cases'></span>

<h3>Description</h3>

<p>Each study includes one or more &quot;case lists&quot;. Each case list is a collection
of samples that were analyzed using one or more platforms/assays. 
It is possible to obtain a list of all sample identifiers for each
case list of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expand_cases(csid, dryrun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expand_cases_+3A_csid">csid</code></td>
<td>
<p>String corresponding to a TCGA Cancer Study identifier.</p>
</td></tr>
<tr><td><code id="expand_cases_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing as many elements as TCGA case lists 
available for a given TCGA Study. Each element is named after a case list identifier.
Also, each element is a character vector including all sample identifiers
(case ids) corresponding to the corresponding case list identifier.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
x &lt;- expand_cases("blca_tcga", dryrun = TRUE)
lapply(x, utils::head)


</code></pre>

<hr>
<h2 id='fetch_all_tcgadata'>Fetch All Molecular Data for a Cancer Profile of Interest.</h2><span id='topic+fetch_all_tcgadata'></span>

<h3>Description</h3>

<p>Recursively query cbioportal to retrieve data corresponding to all 
available genes. Data are returned as a 'data.frame' 
that can be easily manipulated for downstream analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetch_all_tcgadata(case_list_id, gprofile_id, mutations = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fetch_all_tcgadata_+3A_case_list_id">case_list_id</code></td>
<td>
<p>string corresponding to the identifier of the TCGA Case List of interest</p>
</td></tr>
<tr><td><code id="fetch_all_tcgadata_+3A_gprofile_id">gprofile_id</code></td>
<td>
<p>string corresponding to the identifier of the TCGA Profile of interest</p>
</td></tr>
<tr><td><code id="fetch_all_tcgadata_+3A_mutations">mutations</code></td>
<td>
<p>logical. If TRUE, extended mutation data are fetched instead of the standard TCGA data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame is returned, including the desired TCGA data. 
Typically, rows are genes and columns are cases. 
If &quot;extended mutation&quot; data are retrieved (mutations = TRUE), 
rows correspond to individual mutations 
while columns are populated with mutation features
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The examples below require an active Internet connection.
# Note: execution may take several minutes.
## Not run: 
# Download all brca_pub mutation data (complete samples)
all_brca_MUT &lt;- fetch_all_tcgadata(case_list_id = "brca_tcga_pub_complete", 
                                   gprofile_id = "brca_tcga_pub_mutations", 
                                   mutations = TRUE)

# Download all brca_pub RNA expression data (complete samples)
all_brca_RNA &lt;- fetch_all_tcgadata(case_list_id = "brca_tcga_pub_complete", 
                                   gprofile_id = "brca_tcga_pub_mrna", 
                                   mutations = FALSE)

## End(Not run)



</code></pre>

<hr>
<h2 id='get_cancer_studies'>Retrieve the List of Cancer Studies Available at cbioportal.</h2><span id='topic+get_cancer_studies'></span>

<h3>Description</h3>

<p>Retrieve information about the studies or datasets available at cbioportal.org. 
Information include a 'studyId', description, references, and more.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cancer_studies(dryrun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cancer_studies_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data Frame including cancer study information.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
all_studies &lt;- get_cancer_studies(dryrun = TRUE)
utils::head(all_studies)



</code></pre>

<hr>
<h2 id='get_cancer_types'>Retrieve Cancer Types.</h2><span id='topic+get_cancer_types'></span>

<h3>Description</h3>

<p>Retrieve information about cancer types and corresponding 
abbreviations from cbioportal.org. 
Information include identifiers, names, and parental cancer type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cancer_types(dryrun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cancer_types_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame including cancer type information.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
all_canc &lt;- get_cancer_types(dryrun = TRUE)
utils::head(all_canc)

</code></pre>

<hr>
<h2 id='get_case_lists'>Retrieve Case List Available for a Specific Cancer Study.</h2><span id='topic+get_case_lists'></span>

<h3>Description</h3>

<p>Each study includes one or more &quot;case lists&quot;. Each case list is a collection
of samples that were analyzed using one or more platforms/assays. 
It is possible to obtain a list of case list identifiers from cbioportal.org for a 
cancer study of interest. Identifier, name, description and category are
returned for each entry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_case_lists(csid, dryrun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_case_lists_+3A_csid">csid</code></td>
<td>
<p>String corresponding to the Identifier of the Study of Interest</p>
</td></tr>
<tr><td><code id="get_case_lists_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data Frame including Case List information.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
blca_case_lists &lt;- get_case_lists("blca_tcga", dryrun = TRUE)  
blca_case_lists



</code></pre>

<hr>
<h2 id='get_clinical_data'>Retrieve Clinical Information for a Cancer Study.</h2><span id='topic+get_clinical_data'></span>

<h3>Description</h3>

<p>Retrieve Clinical Information about the samples included in a cancer study of interest. 
For each sample/case, information about the corresponding cancer patient are returned. 
These may include sex, age, therapeutic regimen, tumor stage, 
survival status, as well as other information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clinical_data(csid, case_list_id = NULL, dryrun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clinical_data_+3A_csid">csid</code></td>
<td>
<p>String corresponding to a TCGA Cancer Study identifier.</p>
</td></tr>
<tr><td><code id="get_clinical_data_+3A_case_list_id">case_list_id</code></td>
<td>
<p>String corresponding to the case_list identifier of interest. This Can be NULL.</p>
</td></tr>
<tr><td><code id="get_clinical_data_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame including clinical information of a list of 
samples/cases of interest.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
clinic_data &lt;- get_clinical_data("blca_tcga", dryrun = TRUE)  
utils::head(clinic_data[, 1:7])
 


</code></pre>

<hr>
<h2 id='get_gene_identifiers'>Retrieve All Gene Identifiers</h2><span id='topic+get_gene_identifiers'></span>

<h3>Description</h3>

<p>Obtain all valid gene identifiers, including ENTREZ gene identifiers and 
HUGO gene symbols. Genes are classified according to the gene type 
(*e.g.*, 'protein-coding', 'pseudogene', 'miRNA', ...). Note that 
miRNA and phosphoprotein genes are associated with a negative entrezGeneId.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_identifiers(dryrun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gene_identifiers_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data Frame including gene identifiers.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
x &lt;- get_gene_identifiers(dryrun = TRUE)


</code></pre>

<hr>
<h2 id='get_genetic_profiles'>Retrieve Genetic Profiles for a TCGA Study of Interest</h2><span id='topic+get_genetic_profiles'></span>

<h3>Description</h3>

<p>Retrieve Information about all genetic profiles associated with a cancer study of interest.
Each cancer study includes one or more types of molecular analyses.  The corresponding assays or 
platforms are referred to as genetic profiles. 
A genetic profile identifier is required to download molecular data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genetic_profiles(csid = NULL, dryrun = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_genetic_profiles_+3A_csid">csid</code></td>
<td>
<p>String corresponding to the cancer study id of interest</p>
</td></tr>
<tr><td><code id="get_genetic_profiles_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame including information about genetic profiles.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
get_genetic_profiles("blca_tcga", dryrun = TRUE)  


</code></pre>

<hr>
<h2 id='get_molecular_data'>Retrieve Molecular Data corresponding to a Genetic Profile of Interest.</h2><span id='topic+get_molecular_data'></span>

<h3>Description</h3>

<p>Retrieve Data corresponding to a Genetic Profile of interest from a cancer study of interest.
This function is the workhorse of the TCGAretriever package and can be used to fetch data 
concerning several genes at once. For retrieving mutation data, please use the 'get_mutation_data()' function. 
For large queries (more than 500 genes), please use the 'fetch_all_tcgadata()' 
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_molecular_data(
  case_list_id,
  gprofile_id,
  glist = c("TP53", "E2F1"),
  dryrun = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_molecular_data_+3A_case_list_id">case_list_id</code></td>
<td>
<p>String corresponding to the Identifier of a list of cases.</p>
</td></tr>
<tr><td><code id="get_molecular_data_+3A_gprofile_id">gprofile_id</code></td>
<td>
<p>String corresponding to the Identifier of a genetic Profile of interest.</p>
</td></tr>
<tr><td><code id="get_molecular_data_+3A_glist">glist</code></td>
<td>
<p>Vector including one or more gene identifiers (ENTREZID or OFFICIAL_SYMBOL). ENTREZID
gene identifiers should be passed as numeric.</p>
</td></tr>
<tr><td><code id="get_molecular_data_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame including the molecular data of interest. Rows are genes, columns are samples.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
x &lt;- get_molecular_data(case_list_id = 'blca_tcga_3way_complete',
                        gprofile_id = 'blca_tcga_rna_seq_v2_mrna',
                        glist = c("TP53", "E2F1"), dryrun = TRUE)
x[, 1:10]


</code></pre>

<hr>
<h2 id='get_mutation_data'>Retrieve Mutation Data corresponding to a Genetic Profile of Interest.</h2><span id='topic+get_mutation_data'></span>

<h3>Description</h3>

<p>Retrieve DNA Sequence Variations (Mutations) identified by exome sequencing projects.
This function is the workhorse of the TCGAretriever package for mutation data and can be used to fetch data 
concerning several genes at once. For retrieving non-mutation data, please use the 'get_molecular_data()' function. 
For large queries (more than 500 genes), please use the 'fetch_all_tcgadata()' 
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mutation_data(
  case_list_id,
  gprofile_id,
  glist = c("TP53", "E2F1"),
  dryrun = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_mutation_data_+3A_case_list_id">case_list_id</code></td>
<td>
<p>String corresponding to the Identifier of a list of cases.</p>
</td></tr>
<tr><td><code id="get_mutation_data_+3A_gprofile_id">gprofile_id</code></td>
<td>
<p>String corresponding to the Identifier of a genetic Profile of interest.</p>
</td></tr>
<tr><td><code id="get_mutation_data_+3A_glist">glist</code></td>
<td>
<p>Vector including one or more gene identifiers (ENTREZID or OFFICIAL_SYMOL). ENTREZID
gene identifiers should be passed as numeric.</p>
</td></tr>
<tr><td><code id="get_mutation_data_+3A_dryrun">dryrun</code></td>
<td>
<p>Logical. If TRUE, all other arguments (if any) are ignored and 
a representative example is returned as output. No Internet connection is 
required for executing the operation when 'dryrun' is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data Frame inluding one row per mutation
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set `dryrun = FALSE` (default option) in production!
x &lt;- get_mutation_data(case_list_id = 'blca_tcga_sequenced',
                       gprofile_id = 'blca_tcga_mutations',
                       glist = c('TP53', 'PTEN'), dryrun = TRUE)
utils::head(x[, c(4, 7, 23, 15, 16, 17, 24, 18, 21)]) 
 

</code></pre>

<hr>
<h2 id='make_groups'>Split Numeric Vectors in Groups.</h2><span id='topic+make_groups'></span>

<h3>Description</h3>

<p>Assign each element of a numeric vector to a group. Grouping is based on ranks: 
numeric values are sorted and then split in 2 or more groups. Values may be sorted 
in an increasing or decreasing fashion. The vector is returned in the original order. 
Labels may be assigned to each group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_groups(num_vector, groups, group_labels = NULL, desc = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_groups_+3A_num_vector">num_vector</code></td>
<td>
<p>numeric vector. It includes the values to be assigned to the different groups</p>
</td></tr>
<tr><td><code id="make_groups_+3A_groups">groups</code></td>
<td>
<p>integer. The number of groups that will be generated</p>
</td></tr>
<tr><td><code id="make_groups_+3A_group_labels">group_labels</code></td>
<td>
<p>character vector. Labels for each group. Note that 
the length of group_labels has to be equal to the number of groups</p>
</td></tr>
<tr><td><code id="make_groups_+3A_desc">desc</code></td>
<td>
<p>logical. If TRUE, the sorting is applied in a decreasing fashion</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame including the vector provided as argument in the original order (&quot;value&quot;) 
and the grouping vector (&quot;rank&quot;). If labels are provided as an argument, group labels 
are also included in the data.frame (&quot;labels&quot;).
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.data-pulse.com/dev_site/TCGAretriever/">https://www.data-pulse.com/dev_site/TCGAretriever/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exprs_geneX &lt;- c(19.1,18.4,22.4,15.5,20.2,17.4,9.4,12.4,31.2,33.2,18.4,22.1)
groups_num &lt;- 3
groups_labels &lt;- c("high", "med", "low")
make_groups(exprs_geneX, groups_num, groups_labels, desc = TRUE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
