<!DOCTYPE html><html><head><title>Help for package tcgaViz</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {tcgaViz}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#tcgaViz-package'><p>tcgaViz: Visualization Tool for the Cancer Genome Atlas Program (TCGA)</p></a></li>
<li><a href='#calculate_pvalue'><p>Corrected Wilcoxon tests</p></a></li>
<li><a href='#convert_biodata'><p>Format biological data</p></a></li>
<li><a href='#convert2biodata'><p>Format biological data</p></a></li>
<li><a href='#plot.biodata'><p>Distribution plot</p></a></li>
<li><a href='#run_app'><p>Run the Shiny Application</p></a></li>
<li><a href='#tcga'><p>Biological data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Visualization Tool for the Cancer Genome Atlas Program (TCGA)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Differential analysis of tumor tissue immune cell type
    abundance based on RNA-seq gene-level expression from The Cancer
    Genome Atlas (TCGA; <a href="https://pancanatlas.xenahubs.net">https://pancanatlas.xenahubs.net</a>) database.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>config, data.table, dplyr, DT, ggplot2, ggpubr, golem,
grDevices, magrittr, methods, plotly, readr, reshape2, rlang,
rstatix, scales, shiny, shinyFeedback, shinyjs, stats, stringr,
tidyr, tidyselect, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, spelling, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-04 13:46:18 UTC; rstudio</td>
</tr>
<tr>
<td>Author:</td>
<td>Etienne Camenen [aut, cre],
  Gilles Marodon [aut],
  Nicolas Aubert [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Etienne Camenen &lt;etienne.camenen@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-04 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='tcgaViz-package'>tcgaViz: Visualization Tool for the Cancer Genome Atlas Program (TCGA)</h2><span id='topic+tcgaViz'></span><span id='topic+tcgaViz-package'></span>

<h3>Description</h3>

<p>Differential analysis of tumor tissue immune cell type
abundance based on RNA-seq gene-level expression from The Cancer
Genome Atlas (TCGA; &lt;https://pancanatlas.xenahubs.net&gt;) database.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Etienne Camenen <a href="mailto:etienne.camenen@gmail.com">etienne.camenen@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Gilles Marodon
</p>
</li>
<li><p> Nicolas Aubert
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><a href="https://github.com/ecamenen/tcgaViz">Contribute to the development</a>
</p>
</li>
<li> <p><a href="https://github.com/ecamenen/tcgaViz/issues">Report bugs</a>
</p>
</li>
<li> <p><a href="https://github.com/ecamenen/tcgaViz/blob/master/doc/Tutorial.pdf">Read the tutorial</a>
</p>
</li></ul>


<hr>
<h2 id='calculate_pvalue'>Corrected Wilcoxon tests</h2><span id='topic+calculate_pvalue'></span>

<h3>Description</h3>

<p>Displays stars for each cell type corresponding to the significance level
of two mean comparison tests between expression levels (high or low) with
multiple correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_pvalue(
  x,
  method_test = "wilcox_test",
  method_adjust = "BH",
  p_threshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_pvalue_+3A_x">x</code></td>
<td>
<p>object from <code><a href="#topic+convert2biodata">convert2biodata()</a></code> for a dataframe containing
columns named high (logical), cell_type (factor) and value (float).</p>
</td></tr>
<tr><td><code id="calculate_pvalue_+3A_method_test">method_test</code></td>
<td>
<p>character for the choice of the statistical test among
't_test' or 'wilcox_test'.</p>
</td></tr>
<tr><td><code id="calculate_pvalue_+3A_method_adjust">method_adjust</code></td>
<td>
<p>character for the choice of the multiple correction test
among 'BH', 'bonferroni', 'BY', 'fdr', 'hochberg', 'holm', 'hommel', 'none'</p>
</td></tr>
<tr><td><code id="calculate_pvalue_+3A_p_threshold">p_threshold</code></td>
<td>
<p>float for the significativity threshold of the P-value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>rstatix_test object for a table with cell types in the row and
P-values, corrections and other statistics in the column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tcga)
(df &lt;- convert2biodata(
    algorithm = "Cibersort_ABS",
    disease = "breast invasive carcinoma",
    tissue = "Primary Tumor",
    gene_x = "ICOS"
))

calculate_pvalue(df)

calculate_pvalue(
    df,
    method_test = "t_test",
    method_adjust = "bonferroni",
    p_threshold = 0.01
)
</code></pre>

<hr>
<h2 id='convert_biodata'>Format biological data</h2><span id='topic+convert_biodata'></span>

<h3>Description</h3>

<p>Merges gene and cell datasets with the same TCGA sample identifiers,
splits samples according to the expression levels of a selected gene into
two categories (below or above average) and formats into a 3-column
data frame: gene expression levels, cell types, and gene expression values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_biodata(
  genes,
  cells,
  select = colnames(genes)[3],
  stat = "mean",
  disease = NULL,
  tissue = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_biodata_+3A_genes">genes</code></td>
<td>
<p>data frame whose first two columns contain identifiers and the
others float values.</p>
</td></tr>
<tr><td><code id="convert_biodata_+3A_cells">cells</code></td>
<td>
<p>data frame whose first two columns contain identifiers and
the others float values.</p>
</td></tr>
<tr><td><code id="convert_biodata_+3A_select">select</code></td>
<td>
<p>character for a column name in genes.</p>
</td></tr>
<tr><td><code id="convert_biodata_+3A_stat">stat</code></td>
<td>
<p>character for the statistic to be chosen among &quot;mean&quot;, &quot;median&quot;
or &quot;quantile&quot;.</p>
</td></tr>
<tr><td><code id="convert_biodata_+3A_disease">disease</code></td>
<td>
<p>character for the type of TCGA cancer (see the list in
extdata/disease_names.csv).</p>
</td></tr>
<tr><td><code id="convert_biodata_+3A_tissue">tissue</code></td>
<td>
<p>character for the type of TCGA tissue among :
'Additional - New Primary',
'Additional Metastatic',
'Metastatic',
'Primary Blood Derived Cancer - Peripheral Blood',
'Primary Tumor',
'Recurrent Tumor',
'Solid Tissue Normal'</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>disease</code> and <code>tissue</code> arguments should be displayed in the title
of <code><a href="#topic+plot.biodata">plot.biodata()</a></code> only if the <code>genes</code> argument does not already have
them in its attributes.
</p>


<h3>Value</h3>

<p>data frame with the following columns:
</p>

<ul>
<li> <p><code>high</code> (logical): the expression levels of a selected gene,
TRUE for below or FALSE for above average.
</p>
</li>
<li> <p><code>cells</code> (factor): cell types.
</p>
</li>
<li> <p><code>value</code> (float): the abundance estimation of the cell types.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(tcga)
(df_formatted &lt;- convert_biodata(tcga$genes, tcga$cells$Cibersort, "ICOS"))
</code></pre>

<hr>
<h2 id='convert2biodata'>Format biological data</h2><span id='topic+convert2biodata'></span>

<h3>Description</h3>

<p>Merges gene and cell datasets with the same TCGA sample identifiers,
splits samples according to the expression levels of a selected gene into
two categories (below or above average) and formats into a 3-column
data frame: gene expression levels, cell types, and gene expression values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert2biodata(algorithm, disease, tissue, gene_x, stat = "mean", path = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert2biodata_+3A_algorithm">algorithm</code></td>
<td>
<p>character for the algorithm used to estimate the
distribution of cell type abundance among : 'Cibersort', 'Cibersort_ABS',
'EPIC', 'MCP_counter', 'Quantiseq', 'Timer', 'Xcell', 'Xcell (2)' and
'Xcell64'.</p>
</td></tr>
<tr><td><code id="convert2biodata_+3A_disease">disease</code></td>
<td>
<p>character for the type of TCGA cancer (see the list in
extdata/disease_names.csv).</p>
</td></tr>
<tr><td><code id="convert2biodata_+3A_tissue">tissue</code></td>
<td>
<p>character for the type of TCGA tissue among :
'Additional - New Primary',
'Additional Metastatic',
'Metastatic',
'Primary Blood Derived Cancer - Peripheral Blood',
'Primary Tumor',
'Recurrent Tumor',
'Solid Tissue Normal'</p>
</td></tr>
<tr><td><code id="convert2biodata_+3A_gene_x">gene_x</code></td>
<td>
<p>character for the gene selected in the differential analysis
(see the list in extdata/gene_names.csv).</p>
</td></tr>
<tr><td><code id="convert2biodata_+3A_stat">stat</code></td>
<td>
<p>character for the statistic to be chosen among &quot;mean&quot;, &quot;median&quot;
or &quot;quantile&quot;.</p>
</td></tr>
<tr><td><code id="convert2biodata_+3A_path">path</code></td>
<td>
<p>character for the path name of the <code>tcga</code> dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with the following columns:
</p>

<ul>
<li> <p><code>high</code> (logical): the expression levels of a selected gene,
TRUE for below or FALSE for above average.
</p>
</li>
<li> <p><code>cells</code> (factor): cell types.
</p>
</li>
<li> <p><code>value</code> (float): the abundance estimation of the cell types.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(tcga)
(convert2biodata(
    algorithm = "Cibersort_ABS",
    disease = "breast invasive carcinoma",
    tissue = "Primary Tumor",
    gene_x = "ICOS"
))
</code></pre>

<hr>
<h2 id='plot.biodata'>Distribution plot</h2><span id='topic+plot.biodata'></span>

<h3>Description</h3>

<p>Distribution plot of cell subtypes according to the expression level (high or
low) of a selected gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'biodata'
plot(
  x,
  type = "violin",
  dots = FALSE,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  stats = NULL,
  draw = TRUE,
  axis.text.x = element_text(size = 10),
  axis.text.y = element_text(size = 8),
  cex.lab = 12,
  cex.main = 16,
  col = (scales::hue_pal())(length(unique(x$cell_type))),
  axis.title.x = element_text(size = cex.lab, face = "bold.italic", vjust = -0.5),
  axis.title.y = element_text(size = cex.lab, face = "bold.italic", vjust = -0.5),
  plot.title = element_text(size = cex.main, face = "bold", vjust = 1, hjust = 0.5),
  plot.margin = unit(c(0, 0, 0, -0.5), "cm"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.biodata_+3A_x">x</code></td>
<td>
<p>object from <code><a href="#topic+convert2biodata">convert2biodata()</a></code> for a dataframe containing
columns named high (logical), cell_type (factor) and value (float).</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_type">type</code></td>
<td>
<p>character for the type of plot to be chosen among &quot;violin&quot;
or &quot;boxplot&quot;.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_dots">dots</code></td>
<td>
<p>boolean to add all points to the graph.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_title">title</code></td>
<td>
<p>character for the title of the plot.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_xlab">xlab</code></td>
<td>
<p>character for the name of the X axis label.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_ylab">ylab</code></td>
<td>
<p>character for the name of the Y axis label.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_stats">stats</code></td>
<td>
<p>object from <code><a href="#topic+calculate_pvalue">calculate_pvalue()</a></code>.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_draw">draw</code></td>
<td>
<p>bolean to plot the graph.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_axis.text.x">axis.text.x</code></td>
<td>
<p>tick labels along axes (<code><a href="ggplot2.html#topic+element">element_text()</a></code>). Specify all axis tick labels (<code>axis.text</code>),
tick labels by plane (using <code>axis.text.x</code> or <code>axis.text.y</code>), or individually
for each axis (using <code>axis.text.x.bottom</code>, <code>axis.text.x.top</code>,
<code>axis.text.y.left</code>, <code>axis.text.y.right</code>). <code style="white-space: pre;">&#8288;axis.text.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;axis.text.*&#8288;</code> which inherits from <code>axis.text</code>, which in turn inherits
from <code>text</code></p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_axis.text.y">axis.text.y</code></td>
<td>
<p>tick labels along axes (<code><a href="ggplot2.html#topic+element">element_text()</a></code>). Specify all axis tick labels (<code>axis.text</code>),
tick labels by plane (using <code>axis.text.x</code> or <code>axis.text.y</code>), or individually
for each axis (using <code>axis.text.x.bottom</code>, <code>axis.text.x.top</code>,
<code>axis.text.y.left</code>, <code>axis.text.y.right</code>). <code style="white-space: pre;">&#8288;axis.text.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;axis.text.*&#8288;</code> which inherits from <code>axis.text</code>, which in turn inherits
from <code>text</code></p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_cex.lab">cex.lab</code></td>
<td>
<p>numerical value giving the amount by which x and y plotting
labels should be magnified relative to the default.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_cex.main">cex.main</code></td>
<td>
<p>numerical value giving the amount by which main plotting
title should be magnified relative to the default.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_col">col</code></td>
<td>
<p>character for the specification for the default plotting color.
See section 'Color Specification' in <code><a href="graphics.html#topic+par">graphics::par()</a></code>.</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_axis.title.x">axis.title.x</code></td>
<td>
<p>labels of axes (<code><a href="ggplot2.html#topic+element">element_text()</a></code>). Specify all axes' labels (<code>axis.title</code>),
labels by plane (using <code>axis.title.x</code> or <code>axis.title.y</code>), or individually
for each axis (using <code>axis.title.x.bottom</code>, <code>axis.title.x.top</code>,
<code>axis.title.y.left</code>, <code>axis.title.y.right</code>). <code style="white-space: pre;">&#8288;axis.title.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;axis.title.*&#8288;</code> which inherits from <code>axis.title</code>, which in turn inherits
from <code>text</code></p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_axis.title.y">axis.title.y</code></td>
<td>
<p>labels of axes (<code><a href="ggplot2.html#topic+element">element_text()</a></code>). Specify all axes' labels (<code>axis.title</code>),
labels by plane (using <code>axis.title.x</code> or <code>axis.title.y</code>), or individually
for each axis (using <code>axis.title.x.bottom</code>, <code>axis.title.x.top</code>,
<code>axis.title.y.left</code>, <code>axis.title.y.right</code>). <code style="white-space: pre;">&#8288;axis.title.*.*&#8288;</code> inherits from
<code style="white-space: pre;">&#8288;axis.title.*&#8288;</code> which inherits from <code>axis.title</code>, which in turn inherits
from <code>text</code></p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_plot.title">plot.title</code></td>
<td>
<p>plot title (text appearance) (<code><a href="ggplot2.html#topic+element">element_text()</a></code>; inherits
from <code>title</code>) left-aligned by default</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_plot.margin">plot.margin</code></td>
<td>
<p>margin around entire plot (<code>unit</code> with the sizes of
the top, right, bottom, and left margins)</p>
</td></tr>
<tr><td><code id="plot.biodata_+3A_...">...</code></td>
<td>
<p>arguments to pass to <code><a href="ggplot2.html#topic+theme">ggplot2::theme()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("ggplot2")
data(tcga)
(df &lt;- convert2biodata(
    algorithm = "Cibersort_ABS",
    disease = "breast invasive carcinoma",
    tissue = "Primary Tumor",
    gene_x = "ICOS"
))

plot(df)
stats &lt;- calculate_pvalue(df)
plot(
    df,
    stats = stats,
    type = "boxplot",
    dots = TRUE,
    xlab = "Expression level of the 'ICOS' gene by cell type",
    ylab = "Percent of relative abundance\n(from the Cibersort_ABS algorithm)",
    title = "Differential analysis of tumor tissue immune cell type abundance
    based on RNASeq gene-level expression from The Cancer Genome Atlas
    (TCGA) database",
    axis.text.y = element_text(size = 8, hjust = 0.5),
    plot.title =  element_text(face = "bold", hjust = 0.5)
)

</code></pre>

<hr>
<h2 id='run_app'>Run the Shiny Application</h2><span id='topic+run_app'></span>

<h3>Description</h3>

<p>Runs a Shiny application. This function normally does not return; interrupt R
to stop the application (usually by pressing Ctrl+C or Esc).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_app(
  onStart = NULL,
  options = list(),
  enableBookmarking = NULL,
  uiPattern = "/",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_app_+3A_onstart">onStart</code></td>
<td>
<p>A function that will be called before the app is actually run.
This is only needed for <code>shinyAppObj</code>, since in the <code>shinyAppDir</code>
case, a <code>global.R</code> file can be used for this purpose.</p>
</td></tr>
<tr><td><code id="run_app_+3A_options">options</code></td>
<td>
<p>Named options that should be passed to the <code>runApp</code> call
(these can be any of the following: &quot;port&quot;, &quot;launch.browser&quot;, &quot;host&quot;, &quot;quiet&quot;,
&quot;display.mode&quot; and &quot;test.mode&quot;). You can also specify <code>width</code> and
<code>height</code> parameters which provide a hint to the embedding environment
about the ideal height/width for the app.</p>
</td></tr>
<tr><td><code id="run_app_+3A_enablebookmarking">enableBookmarking</code></td>
<td>
<p>Can be one of <code>"url"</code>, <code>"server"</code>, or
<code>"disable"</code>. The default value, <code>NULL</code>, will respect the setting from
any previous calls to  <code><a href="shiny.html#topic+enableBookmarking">enableBookmarking()</a></code>. See <code><a href="shiny.html#topic+enableBookmarking">enableBookmarking()</a></code>
for more information on bookmarking your app.</p>
</td></tr>
<tr><td><code id="run_app_+3A_uipattern">uiPattern</code></td>
<td>
<p>A regular expression that will be applied to each <code>GET</code>
request to determine whether the <code>ui</code> should be used to handle the
request. Note that the entire request path must match the regular
expression in order for the match to be considered successful.</p>
</td></tr>
<tr><td><code id="run_app_+3A_...">...</code></td>
<td>
<p>arguments to pass to golem_opts.
See <code>?golem::get_golem_options</code> for more details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about this function, please take a look at
<a href="https://CRAN.R-project.org/package=golem/vignettes/c_deploy.html">https://CRAN.R-project.org/package=golem/vignettes/c_deploy.html</a>.
</p>


<h3>Value</h3>

<p>An object that represents the app.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
# Start app in the current working directory
run_app()

# Start app in a subdirectory called myapp
run_app("myapp")
}
</code></pre>

<hr>
<h2 id='tcga'>Biological data</h2><span id='topic+tcga'></span>

<h3>Description</h3>

<p>A list of biological data: RNASeq data, phenotypic
metadata and cell abundance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tcga)
</code></pre>


<h3>Details</h3>


<ul>
<li> <p><code>genes</code>: RNASeq from The Cancer Genome Atlas (TCGA) database.
</p>
</li>
<li> <p><code>phenotypes</code>: Metadata from the TCGA database containing
sample ID, sample type ID, sample type and primary disease.
</p>
</li>
<li> <p><code>cells</code>: Abundance estimates of cell types
</p>
</li></ul>



<h3>Note</h3>

<p>Subset of thirty samples of invasive breast carcinoma data from primary
tumor tissue. The cell type data are from a subset generated by the
Cibersort_ABS algorithm (<a href="https://cibersortx.stanford.edu">https://cibersortx.stanford.edu</a>).
For the complete dataset, please use:
<br /><br />
<code>path &lt;- system.file("extdata", package = "tcgaViz")</code>
<br />
<code>load(file.path(path, "tcga.rda"))</code>
</p>


<h3>Source</h3>


<ul>
<li><p> dataset: gene expression RNAseq - Batch effects normalized mRNA data
</p>
</li>
<li><p> hub: <a href="https://pancanatlas.xenahubs.net">https://pancanatlas.xenahubs.net</a>
</p>
</li>
<li><p> cohort: TCGA Pan-Cancer (PANCAN)
</p>
</li>
<li><p> dataset ID: EB++AdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena
</p>
</li>
<li><p> download: <a href="https://tcga-pancan-atlas-hub.s3.us-east-1.amazonaws.com/download/EB%2B%2BAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena.gz">https://tcga-pancan-atlas-hub.s3.us-east-1.amazonaws.com/download/EB%2B%2BAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena.gz</a> (full metadata)
</p>
</li>
<li><p> samples: 11060
</p>
</li>
<li><p> version: 2016-12-29
</p>
</li>
<li><p> type of data: gene expression RNAseq
</p>
</li>
<li><p> unit: log2(norm_value+1)
</p>
</li>
<li><p> raw data: <a href="https://www.synapse.org/#!Synapse:syn4976369.3">https://www.synapse.org/#!Synapse:syn4976369.3</a>
</p>
</li>
<li><p> input data format: ROWs (identifiers) x COLUMNs (samples) (i.e. genomicMatrix)
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(tcga)
(df &lt;- convert2biodata(
    algorithm = "Cibersort_ABS",
    disease = "breast invasive carcinoma",
    tissue = "Primary Tumor",
    gene_x = "ICOS"
))
(stats &lt;- calculate_pvalue(df))

plot(df, stats = stats)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
