<!DOCTYPE html><html lang="en"><head><title>Help for package TDbook</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TDbook}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TDbook-package'><p>TDbook: Companion Package for the Book &quot;Data Integration, Manipulation and Visualization of Phylogenetic Trees&quot; by Guangchuang Yu (2022, ISBN:9781032233574)</p></a></li>
<li><a href='#data-AA_sequence'><p>Data set containing multiple sequence alignment information</p></a></li>
<li><a href='#data-df_alleles.csv'><p>Allele table</p></a></li>
<li><a href='#data-df_alltax_info'><p>Data frame containing hierarchical relationship</p></a></li>
<li><a href='#data-df_bar_data'><p>Trait data</p></a></li>
<li><a href='#data-df_barplot_attr'><p>Data set of the abundance of microbes at the body sites of greatest</p>
prevalence</a></li>
<li><a href='#data-df_Candidaauris_data'><p>Population genetics data No.1</p></a></li>
<li><a href='#data-df_difftax'><p>Data frame containing taxa and factor information and pvalue</p></a></li>
<li><a href='#data-df_info'><p>Sampling information data set</p></a></li>
<li><a href='#data-df_inode_data'><p>Nodedata to be mapped to tree</p></a></li>
<li><a href='#data-df_NJIDqgsS'><p>Population genetics data No.2</p></a></li>
<li><a href='#data-df_ring_heatmap'><p>Data set of the abundance and types of microbes</p></a></li>
<li><a href='#data-df_svl'><p>Data frame containing slv information</p></a></li>
<li><a href='#data-df_tip_data'><p>Tipdata to be mapped to tree</p></a></li>
<li><a href='#data-df_tippoint'><p>Data set of the abundance and types of microbes</p></a></li>
<li><a href='#data-dna_HPV58_aln'><p>A DNAbin class to store the aligned sequnces of species of HPV58.tree</p></a></li>
<li><a href='#data-text_RMI_tree'><p>Tree to display with symbolic points indicating partitioned bootstrap values</p></a></li>
<li><a href='#data-tree_anole'><p>Tree to color with continuous state transition in edges</p></a></li>
<li><a href='#data-tree_boots'><p>Tree to be mapped to data</p></a></li>
<li><a href='#data-tree_Candidaauris'><p>Tree No.1 to display with visualized population genetics data</p></a></li>
<li><a href='#data-tree_hmptree'><p>Tree to display with multiple graphs for multi-dimensional data</p></a></li>
<li><a href='#data-tree_HPV58'><p>Tree to display with dot and line plots of pairwise nucleotide sequence distances</p></a></li>
<li><a href='#data-tree_long_branch_example'><p>Tree to display with shrunk outlier long branch</p></a></li>
<li><a href='#data-tree_nex'><p>Tree to display with silhouettes from Phylopic</p></a></li>
<li><a href='#data-tree_NJIDqgsS'><p>Tree No.2 to display with visualized population genetics data</p></a></li>
<li><a href='#data-tree_nwk'><p>Tree to display with sampling information, SNP and Trait data</p></a></li>
<li><a href='#data-tree_seq_nwk'><p>Tree to display with multiple sequence alignment</p></a></li>
<li><a href='#data-tree_treenwk_30.4.19'><p>Tree to group and highlight</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Companion Package for the Book "Data Integration, Manipulation
and Visualization of Phylogenetic Trees" by Guangchuang Yu
(2022, ISBN:9781032233574)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.6</td>
</tr>
<tr>
<td>Description:</td>
<td>The companion package that provides all the datasets used in the book
 "Data Integration, Manipulation and Visualization of Phylogenetic Trees" by Guangchuang Yu (2022, ISBN:9781032233574).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/">https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YuLab-SMU/TDbook/issues">https://github.com/YuLab-SMU/TDbook/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-13 09:39:17 UTC; ygc</td>
</tr>
<tr>
<td>Author:</td>
<td>Guangchuang Yu <a href="https://orcid.org/0000-0002-6485-8781"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Shuangbin Xu <a href="https://orcid.org/0000-0003-3513-5362"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Lin Li <a href="https://orcid.org/0000-0002-6522-1889"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Guangchuang Yu &lt;guangchuangyu@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-13 11:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='TDbook-package'>TDbook: Companion Package for the Book &quot;Data Integration, Manipulation and Visualization of Phylogenetic Trees&quot; by Guangchuang Yu (2022, ISBN:9781032233574)</h2><span id='topic+TDbook'></span><span id='topic+TDbook-package'></span>

<h3>Description</h3>

<p>The companion package that provides all the datasets used in the book &quot;Data Integration, Manipulation and Visualization of Phylogenetic Trees&quot; by Guangchuang Yu (2022, ISBN:9781032233574).
</p>
<p><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/">https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/</a>
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Guangchuang Yu <a href="mailto:guangchuangyu@gmail.com">guangchuangyu@gmail.com</a> (<a href="https://orcid.org/0000-0002-6485-8781">ORCID</a>) [copyright holder]
</p>
<p>Authors:
</p>

<ul>
<li><p> Shuangbin Xu <a href="mailto:xshuangbin@163.com">xshuangbin@163.com</a> (<a href="https://orcid.org/0000-0003-3513-5362">ORCID</a>)
</p>
</li>
<li><p> Lin Li <a href="mailto:williamlee2220@qq.com">williamlee2220@qq.com</a> (<a href="https://orcid.org/0000-0002-6522-1889">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/">https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/YuLab-SMU/TDbook/issues">https://github.com/YuLab-SMU/TDbook/issues</a>
</p>
</li></ul>


<hr>
<h2 id='data-AA_sequence'>Data set containing multiple sequence alignment information</h2><span id='topic+data-AA_sequence'></span><span id='topic+AA_sequence'></span>

<h3>Description</h3>

<p>This data set contains multiple sequence alignment information  to be
attached to the tree.<br />
Chap.7.4
</p>


<h3>Format</h3>

<p>AAbin
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AA_sequence)
</code></pre>

<hr>
<h2 id='data-df_alleles.csv'>Allele table</h2><span id='topic+data-df_alleles.csv'></span><span id='topic+df_alleles'></span>

<h3>Description</h3>

<p>The allele table with original data to be processed to SNP data. Then attached
to the tree after visualizing.<br />
Chap.7.2
</p>


<h3>Format</h3>

<p>A data frame contains 386 rows and 385 variables, with first row of
tips labels.
</p>


<h3>Source</h3>

<p><a href="https://raw.githubusercontent.com/katholt/plotTree/master/tree_example_april2015/alleles.csv">https://raw.githubusercontent.com/katholt/plotTree/master/tree_example_april2015/alleles.csv</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_alleles)
</code></pre>

<hr>
<h2 id='data-df_alltax_info'>Data frame containing hierarchical relationship</h2><span id='topic+data-df_alltax_info'></span><span id='topic+df_alltax_info'></span>

<h3>Description</h3>

<p>This Data frame contains hierarchical relationship to be visualized with
another data frame that contains taxa and factor information and pvalue,
using <code>ggdiffcalde()</code> of The <code>MicrobiotaProcess</code> package.<br />
It's the analysis result of <code>diff_analysis()</code> using public datasets
(Kostic et al. 2012) with The <code>MicrobiotaProcess</code> package.<br />
These data frame are used to demonstrates how to visualize the differential
taxonomy tree with The <code>MicrobiotaProcess</code> package.<br />
Chap.11.1
</p>


<h3>Format</h3>

<p>A data frame to store taxonomy information, contains 1351 rows and 7
variables (classification levels).
</p>


<h3>References</h3>

<p>Kostic, Aleksandar D, Dirk Gevers, Chandra Sekhar Pedamallu, Monia Michaud, Fujiko Duke, Ashlee M Earl, Akinyemi I Ojesina, et al.
Genomic Analysis Identifies Association of Fusobacterium with Colorectal Carcinoma.”
Genome Research, Volume 22, Issue 2, 2012, Pages 292–98.
<a href="https://doi.org/10.1101/gr.126573.111">doi:10.1101/gr.126573.111</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_alltax_info)
</code></pre>

<hr>
<h2 id='data-df_bar_data'>Trait data</h2><span id='topic+data-df_bar_data'></span><span id='topic+df_bar_data'></span>

<h3>Description</h3>

<p>The trait data to be visualized with bar plot and attached to the tree. <br />
Chap.7.2
</p>


<h3>Format</h3>

<p>A data frame contains 386 rows and 2 variables, with a column of
&quot;node&quot; or first column of taxa labels.
</p>


<h3>Source</h3>

<p><a href="https://raw.githubusercontent.com/katholt/plotTree/master/tree_example_april2015/bar.csv">https://raw.githubusercontent.com/katholt/plotTree/master/tree_example_april2015/bar.csv</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_bar_data)
</code></pre>

<hr>
<h2 id='data-df_barplot_attr'>Data set of the abundance of microbes at the body sites of greatest
prevalence</h2><span id='topic+data-df_barplot_attr'></span><span id='topic+df_barplot_attr'></span>

<h3>Description</h3>

<p>The data set of abundance of microbes at the body sites of greatest
prevalence to be visualized using barplot and attached to the tree.<br />
Chap.10.3
</p>


<h3>Format</h3>

<p>A data frame contains 332 rows and 3 variables, with a column of
&quot;node&quot; or first column of taxa labels.
</p>


<h3>References</h3>

<p>Asnicar, Francesco, George Weingart, Timothy L Tickle, Curtis Huttenhower, and Nicola Segata.
Compact Graphical Representation of Phylogenetic Data and Metadata with GraPhlAn.
PeerJ, 3: e1029, June 2015.
<a href="https://doi.org/10.7717/peerj.1029">doi:10.7717/peerj.1029</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_barplot_attr)
</code></pre>

<hr>
<h2 id='data-df_Candidaauris_data'>Population genetics data No.1</h2><span id='topic+data-df_Candidaauris_data'></span><span id='topic+df_Candidaauris_data'></span>

<h3>Description</h3>

<p>Population genetics data to be visualized using multiple graphs and
attached to tree No.1 with <code>geo_fruit()</code> of <code>ggtreeExtra</code>.<br />
Chap.10.4
</p>


<h3>Format</h3>

<p>A data frame contains 305 rows and 20 variables, with a column of
&quot;node&quot; or first column of taxa labels.
</p>


<h3>References</h3>

<p>Chow, Nancy A., José F. Muñoz, Lalitha Gade, Elizabeth L. Berkow, Xiao Li, Rory M. Welsh, Kaitlin Forsberg, et al.
Tracing the Evolutionary History and Global Expansion of Candida Auris Using Population Genomic Analyses.
Edited by Geraldine Butler and Kirsten Nielsen.
mBio, Volume 11, Issue 2, April 2020.
<a href="https://doi.org/10.1128/mBio.03364-19">doi:10.1128/mBio.03364-19</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_Candidaauris_data)
</code></pre>

<hr>
<h2 id='data-df_difftax'>Data frame containing taxa and factor information and pvalue</h2><span id='topic+data-df_difftax'></span><span id='topic+df_difftax'></span>

<h3>Description</h3>

<p>This Data frame contains taxa and factor information and pvalue to be visualized
with another data frame that contains hierarchical relationship, using
<code>ggdiffcalde()</code> of The <code>MicrobiotaProcess</code> package.<br />
These data frame are used to demonstrates how to visualize the differential
taxonomy tree with The <code>MicrobiotaProcess</code> package.<br />
Chap.11.1
</p>


<h3>Format</h3>

<p>A data frame contains 36 rows and 3 variables.
</p>


<h3>References</h3>

<p>Kostic, Aleksandar D, Dirk Gevers, Chandra Sekhar Pedamallu, Monia Michaud, Fujiko Duke, Ashlee M Earl, Akinyemi I Ojesina, et al.
Genomic Analysis Identifies Association of Fusobacterium with Colorectal Carcinoma.”
Genome Research, Volume 22, Issue 2, 2012, Pages 292–98.
<a href="https://doi.org/10.1101/gr.126573.111">doi:10.1101/gr.126573.111</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_difftax)
</code></pre>

<hr>
<h2 id='data-df_info'>Sampling information data set</h2><span id='topic+data-df_info'></span><span id='topic+df_info'></span>

<h3>Description</h3>

<p>The sampling information data set be attached to the tree tip. <br />
Chap.7.2
</p>


<h3>Format</h3>

<p>A data frame contains 386 rows and 6 variables, with a column of
&quot;node&quot; or first column of taxa labels.
</p>


<h3>Source</h3>

<p><a href="https://raw.githubusercontent.com/katholt/plotTree/master/tree_example_april2015/info.csv">https://raw.githubusercontent.com/katholt/plotTree/master/tree_example_april2015/info.csv</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_info)
</code></pre>

<hr>
<h2 id='data-df_inode_data'>Nodedata to be mapped to tree</h2><span id='topic+data-df_inode_data'></span><span id='topic+df_inode_data'></span>

<h3>Description</h3>

<p>This data contains annotation information about nodes of &quot;tree_boots.nwk&quot;,
that is to be mapped to the tree using the <code style="white-space: pre;">&#8288;%&lt;+%&#8288;</code> operator.<br />
Chap.7.1
</p>


<h3>Format</h3>

<p>A data frame with 6 rows and 6 variables with a column of node
or first column of taxa labels.
</p>


<h3>Source</h3>

<p><a href="https://raw.githubusercontent.com/TreeViz/metastyle/master/design/viz_targets_exercise/inode_data.csv">https://raw.githubusercontent.com/TreeViz/metastyle/master/design/viz_targets_exercise/inode_data.csv</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_inode_data)
</code></pre>

<hr>
<h2 id='data-df_NJIDqgsS'>Population genetics data No.2</h2><span id='topic+data-df_NJIDqgsS'></span><span id='topic+df_NJIDqgsS'></span>

<h3>Description</h3>

<p>Population genetics data to be visualized using multiple graphs and
attached to tree No.2 with <code>geo_fruit()</code> of <code>ggtreeExtra</code>.<br />
Chap.10.4
</p>


<h3>Format</h3>

<p>A data frame contains 1351 rows and 7 variables, with a column of
&quot;node&quot; or first column of taxa labels.
</p>


<h3>References</h3>

<p>Wong, Vanessa K., Stephen Baker, Derek J. Pickard, Julian Parkhill, Andrew J. Page, Nicholas A. Feasey, Robert A. Kingsley, et al.
Phylogeographical Analysis of the Dominant Multidrug-Resistant H58 Clade of Salmonella Typhi Identifies Inter- and Intracontinental Transmission Events.
Nature Genetics, Volume 47, Issue 6, May 2015, Pages 632–39.
<a href="https://doi.org/10.1038/ng.3281">doi:10.1038/ng.3281</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_NJIDqgsS)
</code></pre>

<hr>
<h2 id='data-df_ring_heatmap'>Data set of the abundance and types of microbes</h2><span id='topic+data-df_ring_heatmap'></span><span id='topic+df_ring_heatmap'></span>

<h3>Description</h3>

<p>The data set of abundance of microbes at different body sites to be
visualized using heatmap and attached to the tree.<br />
Chap.10.3
</p>


<h3>Format</h3>

<p>A data frame contains 2324 rows and 3 variables, with a column of
&quot;node&quot; or first column of taxa labels.
</p>


<h3>References</h3>

<p>Asnicar, Francesco, George Weingart, Timothy L Tickle, Curtis Huttenhower, and Nicola Segata.
Compact Graphical Representation of Phylogenetic Data and Metadata with GraPhlAn.
PeerJ, 3: e1029, June 2015.
<a href="https://doi.org/10.7717/peerj.1029">doi:10.7717/peerj.1029</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_ring_heatmap)
</code></pre>

<hr>
<h2 id='data-df_svl'>Data frame containing slv information</h2><span id='topic+data-df_svl'></span><span id='topic+df_svl'></span>

<h3>Description</h3>

<p>Data frame containing slv information to scale the color of the tree using
the <code>continuous</code> parameter of <code>ggtree</code> to plot continuous state transition
in edges.<br />
Chap.4.3.5. This file is provided in <a href="http://www.phytools.org/eqg2015/asr.html">http://www.phytools.org/eqg2015/asr.html</a>.
</p>


<h3>Format</h3>

<p>A data frame contains 100 rows and 1 variables.
</p>


<h3>Source</h3>

<p><a href="http://www.phytools.org/eqg2015/data/svl.csv">http://www.phytools.org/eqg2015/data/svl.csv</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_svl)
</code></pre>

<hr>
<h2 id='data-df_tip_data'>Tipdata to be mapped to tree</h2><span id='topic+data-df_tip_data'></span><span id='topic+df_tip_data'></span>

<h3>Description</h3>

<p>This data contains annotation information about tips of <code>tree_boots</code>,
that is to be mapped to the tree using the <code style="white-space: pre;">&#8288;%&lt;+%&#8288;</code> operator.<br />
Chap.7.1
</p>


<h3>Format</h3>

<p>A data frame contains 7 rows and 9 variables, with a column of
&quot;node&quot; or first column of taxa labels.
</p>


<h3>Source</h3>

<p><a href="https://raw.githubusercontent.com/TreeViz/metastyle/master/design/viz_targets_exercise/tip_data.csv">https://raw.githubusercontent.com/TreeViz/metastyle/master/design/viz_targets_exercise/tip_data.csv</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_tip_data)
</code></pre>

<hr>
<h2 id='data-df_tippoint'>Data set of the abundance and types of microbes</h2><span id='topic+data-df_tippoint'></span><span id='topic+df_tippoint'></span>

<h3>Description</h3>

<p>The data set of abundance and types of microbes to be attached to the tree
tips, provided by GraPhlAn (Asnicar et al. 2015).<br />
Chap.10.3
</p>


<h3>Format</h3>

<p>A data frame contains 332 rows and 4 variables,with a column of
&quot;node&quot; or first column of taxa labels.
</p>


<h3>References</h3>

<p>Asnicar, Francesco, George Weingart, Timothy L Tickle, Curtis Huttenhower, and Nicola Segata.
Compact Graphical Representation of Phylogenetic Data and Metadata with GraPhlAn.
PeerJ, 3: e1029, June 2015.
<a href="https://doi.org/10.7717/peerj.1029">doi:10.7717/peerj.1029</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(df_tippoint)
</code></pre>

<hr>
<h2 id='data-dna_HPV58_aln'>A DNAbin class to store the aligned sequnces of species of HPV58.tree</h2><span id='topic+data-dna_HPV58_aln'></span><span id='topic+dna_HPV58_aln'></span>

<h3>Description</h3>

<p>A DNAbin class to store the aligned sequnces of species of HPV58.tree
</p>


<h3>Format</h3>

<p>S3 class:DNAbin
</p>


<h3>References</h3>

<p>Chen, Zigui, Wendy C. S. Ho, Siaw Shi Boon, Priscilla T. Y. Law, Martin C. W. Chan, Rob DeSalle, Robert D. Burk, and Paul K. S. Chan.
Ancient Evolution and Dispersion of Human Papillomavirus 58 Variants.
Journal of Virology, Volume 91, Issue 21, November 2017, e01285–17.
<a href="https://doi.org/10.1128/JVI.01285-17">doi:10.1128/JVI.01285-17</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dna_HPV58_aln)
</code></pre>

<hr>
<h2 id='data-text_RMI_tree'>Tree to display with symbolic points indicating partitioned bootstrap values</h2><span id='topic+data-text_RMI_tree'></span><span id='topic+text_RMI_tree'></span>

<h3>Description</h3>

<p>This tree is to display with symbolic points with different color or shape
indicating the bootstrap values belong to which category.<br />
Catagories are defined by cutting the bootstrap values into several interval
to indicate whether the clade is high, moderate or low support.<br />
Chap.13.2
</p>


<h3>Format</h3>

<p>character
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(text_RMI_tree)
</code></pre>

<hr>
<h2 id='data-tree_anole'>Tree to color with continuous state transition in edges</h2><span id='topic+data-tree_anole'></span><span id='topic+tree_anole'></span>

<h3>Description</h3>

<p>This tree is used to demonstrate the functionality of plotting continuous
state transition in edges using the <code>continuous</code> parameter of <code>ggtree</code>.<br />
Chap.4.3.5. This file is provided in <a href="http://www.phytools.org/eqg2015/asr.html">http://www.phytools.org/eqg2015/asr.html</a>.
</p>


<h3>Format</h3>

<p>S3 class:phylo
</p>


<h3>Source</h3>

<p><a href="http://www.phytools.org/eqg2015/data/anole.tre">http://www.phytools.org/eqg2015/data/anole.tre</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_anole)
</code></pre>

<hr>
<h2 id='data-tree_boots'>Tree to be mapped to data</h2><span id='topic+data-tree_boots'></span><span id='topic+tree_boots'></span>

<h3>Description</h3>

<p>This tree is used to demonstrate the functionality of the <code style="white-space: pre;">&#8288;%&lt;+%&#8288;</code> operator
to attach annotation data to a ggtree graphic object.<br />
Chap.7.1
</p>


<h3>Format</h3>

<p>S3 class:phylo
</p>


<h3>Source</h3>

<p><a href="https://raw.githubusercontent.com/TreeViz/metastyle/master/design/viz_targets_exercise/tree_boots.nwk">https://raw.githubusercontent.com/TreeViz/metastyle/master/design/viz_targets_exercise/tree_boots.nwk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_boots)
</code></pre>

<hr>
<h2 id='data-tree_Candidaauris'>Tree No.1 to display with visualized population genetics data</h2><span id='topic+data-tree_Candidaauris'></span><span id='topic+tree_Candidaauris'></span>

<h3>Description</h3>

<p>Tree No.1 to display with visualized population genetics data for
demonstrating the utility of <code>ggtreeExtra</code>.<br />
Reproduction of Fig.4 of (Chow et al. 2020).<br />
Chap.10.4
</p>


<h3>Format</h3>

<p>S3 class:phylo
</p>


<h3>References</h3>

<p>Chow, Nancy A., José F. Muñoz, Lalitha Gade, Elizabeth L. Berkow, Xiao Li, Rory M. Welsh, Kaitlin Forsberg, et al.
Tracing the Evolutionary History and Global Expansion of Candida Auris Using Population Genomic Analyses.
Edited by Geraldine Butler and Kirsten Nielsen.
mBio, Volume 11, Issue 2, April 2020.
<a href="https://doi.org/10.1128/mBio.03364-19">doi:10.1128/mBio.03364-19</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_Candidaauris)
</code></pre>

<hr>
<h2 id='data-tree_hmptree'>Tree to display with multiple graphs for multi-dimensional data</h2><span id='topic+data-tree_hmptree'></span><span id='topic+tree_hmptree'></span>

<h3>Description</h3>

<p>This tree is to display with multiple graphs, and by reproducing
Fig.2 of (Morgan, Segata, and Huttenhower 2013), demonstrates the feature of
adding multiple layers to present different types of data by <code>ggtreeExtra</code>.<br />
Chap.10.3
</p>


<h3>Format</h3>

<p>S3 class:phylo
</p>


<h3>References</h3>

<p>Morgan, Xochitl C., Nicola Segata, and Curtis Huttenhower.
Biodiversity and Functional Genomics in the Human Microbiome.
Trends in Genetics Volume 29, Issue 1, 2013 January, Pages 51–58.
<a href="https://doi.org/10.1016/J.TIG.2012.09.005">doi:10.1016/J.TIG.2012.09.005</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_hmptree)
</code></pre>

<hr>
<h2 id='data-tree_HPV58'>Tree to display with dot and line plots of pairwise nucleotide sequence distances</h2><span id='topic+data-tree_HPV58'></span><span id='topic+tree_HPV58'></span>

<h3>Description</h3>

<p>This tree is to provide accession numbers from its tip labels, which are used to
calculates pairwise nucleotide sequence distances. The distance matrix
then is visualized as dot and line plots and attached to the tree,
demonstrating the abilities of <code>facet_plot()</code> of ggtree that can add
multiple layers to a specific panel.<br />
Reproduction fo Fig. 1 of (Chen et al. 2017).<br />
Chap.13.1
</p>


<h3>Format</h3>

<p>S3 class:phylo
</p>


<h3>References</h3>

<p>Chen, Zigui, Wendy C. S. Ho, Siaw Shi Boon, Priscilla T. Y. Law, Martin C. W. Chan, Rob DeSalle, Robert D. Burk, and Paul K. S. Chan.
Ancient Evolution and Dispersion of Human Papillomavirus 58 Variants.
Journal of Virology, Volume 91, Issue 21, November 2017, e01285–17.
<a href="https://doi.org/10.1128/JVI.01285-17">doi:10.1128/JVI.01285-17</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_HPV58)
</code></pre>

<hr>
<h2 id='data-tree_long_branch_example'>Tree to display with shrunk outlier long branch</h2><span id='topic+data-tree_long_branch_example'></span><span id='topic+tree_long_branch_example'></span>

<h3>Description</h3>

<p>This tree is to display with display with shrunk outlier long branch to
demonstrate how to shrink outlier long branch with <code>ggtree</code> by modifying
coordination of the out groups, or with the <code>ggbreak</code> package.<br />
Appendix.A.5.3
</p>


<h3>Format</h3>

<p>S3 class:Phylo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_long_branch_example)
</code></pre>

<hr>
<h2 id='data-tree_nex'>Tree to display with silhouettes from Phylopic</h2><span id='topic+data-tree_nex'></span><span id='topic+tree_nex'></span>

<h3>Description</h3>

<p>This tree is to display with display with silhouettes from Phylopic,
demonstrating the feature of <code>ggtree</code> that will automatically download
phylopic figures by querying provided UID.<br />
Chap.8.2
</p>


<h3>Format</h3>

<p>S3 class:Phylo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_nex)
</code></pre>

<hr>
<h2 id='data-tree_NJIDqgsS'>Tree No.2 to display with visualized population genetics data</h2><span id='topic+data-tree_NJIDqgsS'></span><span id='topic+tree_NJIDqgsS'></span>

<h3>Description</h3>

<p>Tree No.2 to display with visualized population genetics data for
demonstrating the utility of <code>ggtreeExtra</code>.<br />
Reproduction of Fig 1 of (Wong et al. 2015).<br />
Chap.10.4
</p>


<h3>Format</h3>

<p>S3 class:phylo
</p>


<h3>References</h3>

<p>Wong, Vanessa K., Stephen Baker, Derek J. Pickard, Julian Parkhill, Andrew J. Page, Nicholas A. Feasey, Robert A. Kingsley, et al.
Phylogeographical Analysis of the Dominant Multidrug-Resistant H58 Clade of Salmonella Typhi Identifies Inter- and Intracontinental Transmission Events.
Nature Genetics, Volume 47, Issue 6, May 2015, Pages 632–39.
<a href="https://doi.org/10.1038/ng.3281">doi:10.1038/ng.3281</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_NJIDqgsS)
</code></pre>

<hr>
<h2 id='data-tree_nwk'>Tree to display with sampling information, SNP and Trait data</h2><span id='topic+data-tree_nwk'></span><span id='topic+tree_nwk'></span>

<h3>Description</h3>

<p>The tree to display with sampling information, attached with dot chart of SNP
and bar chart of Trait data.<br />
Chap.7.2
</p>


<h3>Format</h3>

<p>S3 class:phylo
</p>


<h3>Source</h3>

<p><a href="https://raw.githubusercontent.com/katholt/plotTree/master/tree_example_april2015/tree.nwk">https://raw.githubusercontent.com/katholt/plotTree/master/tree_example_april2015/tree.nwk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_nwk)
</code></pre>

<hr>
<h2 id='data-tree_seq_nwk'>Tree to display with multiple sequence alignment</h2><span id='topic+data-tree_seq_nwk'></span><span id='topic+tree_seq_nwk'></span>

<h3>Description</h3>

<p>This tree is used to display with multiple sequence alignment.<br />
Chap.7.4
</p>


<h3>Format</h3>

<p>S3 class:phylo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_seq_nwk)
</code></pre>

<hr>
<h2 id='data-tree_treenwk_30.4.19'>Tree to group and highlight</h2><span id='topic+data-tree_treenwk_30.4.19'></span><span id='topic+tree_treenwk_30.4.19'></span>

<h3>Description</h3>

<p>This tree is used to reproduce Figure 1 of (Larsen et al. 2019) with
grouping, highlighting and labelling.<br />
Chap.13.3
</p>


<h3>Format</h3>

<p>S3 class:Phylo
</p>


<h3>References</h3>

<p>Larsen, Frederik T., Bertrand Bed’Hom, Bernt Guldbrandtsen, and Tina S. Dalgaard.
Identification and Tissue-Expression Profiling of Novel Chicken c-Type Lectin-Like Domain Containing Proteins
as Potential Targets for Carbohydrate-Based Vaccine Strategies.
Molecular Immunology, Volume 114, Issue October, October 2019, pages 216–25.
<a href="https://doi.org/10.1016/j.molimm.2019.07.022">doi:10.1016/j.molimm.2019.07.022</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_treenwk_30.4.19)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
