<!DOCTYPE html><html><head><title>Help for package TE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AetLTR'><p>LTR retrotransposons in <em>Aegilops tauschii</em></p></a></li>
<li><a href='#AlyLTR'><p>LTR retrotransposons in <em>Arabidopsis lyrata</em></p></a></li>
<li><a href='#EstDynamics'><p>Estimate TE dynamics using mismatch data</p></a></li>
<li><a href='#MasterGene'><p>Implements the master gene model in Marchani et al (2009)</p></a></li>
<li><a href='#MatrixModel'><p>Implements the matrix model in Promislow et al (1999)</p></a></li>
<li><a href='#nbLackOfFitKL'><p>Calcualte the KL divergence of a negative binomial fit to the mismatch data.</p></a></li>
<li><a href='#PlotFamilies'><p>Plot the age distributions or insertion rates for multiple families.</p></a></li>
<li><a href='#print.TEfit'><p>Print a TEfit or TEfit2 object</p></a></li>
<li><a href='#SensitivityPlot'><p>Generate sensitivity plots</p></a></li>
<li><a href='#TE'><p>TE: Insertion/Deletion Dynamics for Transposable Elements</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3-0</td>
</tr>
<tr>
<td>Title:</td>
<td>Insertion/Deletion Dynamics for Transposable Elements</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions to estimate the insertion and deletion rates of transposable element (TE) families. The estimation of insertion rate consists of an improved estimate of the age distribution that takes into account random mutations, and an adjustment by the deletion rate. A hypothesis test for a uniform insertion rate is also implemented. This package implements the methods proposed in Dai et al (2018).</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, rainbow</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-08-22 18:41:53 UTC; xdai</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiongtao Dai [aut, cre, cph],
  Hao Wang [aut],
  Jan Dvorak [ctb],
  Jeffrey Bennetzen [ctb],
  Hans-Georg Mueller [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiongtao Dai &lt;xdai@iastate.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-08-22 19:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='AetLTR'>LTR retrotransposons in <em>Aegilops tauschii</em></h2><span id='topic+AetLTR'></span>

<h3>Description</h3>

<p>This data file contains the LTR retrotransposons in <em>Ae. tauschii</em>.
</p>


<h3>Format</h3>

<p>A data frame with 18024 rows and 12 columns. Each row corresponds to a unique LTR retrotransposon, and each column corresponds to a feature of the LTR-RT. The columns are:
</p>

<dl>
<dt>SeqID</dt><dd><p>LTR retrotransposon sequence ID</p>
</dd>
<dt>UngapedLen</dt><dd><p>Length of each LTR</p>
</dd>
<dt>Mismatch</dt><dd><p>Number of mismatches</p>
</dd>
<dt>Distance</dt><dd><p>Divergence, as defined by (# of mismatches) / (LTR length)</p>
</dd>
<dt>Chr</dt><dd><p>Chromosome number</p>
</dd>
<dt>Start</dt><dd><p>Start location in bp</p>
</dd>
<dt>Stop</dt><dd><p>Ending location in bp</p>
</dd>
<dt>GroupID</dt><dd><p>LTR retrotransposon Family ID</p>
</dd>
<dt>sup</dt><dd><p>Super family membership</p>
</dd>
<dt>recRt5</dt><dd><p>Recombination rate</p>
</dd>
<dt>nearOld</dt><dd><p>Whether the LTR-RT is near a gene that is colinear with wild emmer (TRUE) or not (FALSE)</p>
</dd>
<dt>cCodon</dt><dd><p>Whether the LTR-RT is near the start codon (1) or not (-1)</p>
</dd>
<dt>logDist</dt><dd><p>Log distance to the nearest gene in bp</p>
</dd>
<dt>distToGene</dt><dd><p>Distance to the nearest gene in bp</p>
</dd>
</dl>


<h3>References</h3>

<p><cite>Luo, Ming-Cheng, et al. (2017) &quot;Genome sequence of the progenitor of the wheat D genome Aegilops tauschii.&quot; Nature 551.7681.</cite>
</p>
<p><cite>Dvorak, J., L. Wang, T. Zhu, C. M. Jorgensen, K. R. Deal et al., (2018) &quot;Structural variation and rates of genome evolution in the grass family seen through comparison of sequences of genomes greatly differing in size&quot;. The Plant Journal 95: 487-503.</cite>
</p>
<p><cite>Dai, X., Wang, H., Dvorak, J., Bennetzen, J., Mueller, H.-G. (2018). &quot;Birth and Death of LTR Retrotransposons in Aegilops tauschii&quot;. Genetics</cite>
</p>

<hr>
<h2 id='AlyLTR'>LTR retrotransposons in <em>Arabidopsis lyrata</em></h2><span id='topic+AlyLTR'></span>

<h3>Description</h3>

<p>This data file contains the LTR retrotransposons in <em>Arabidopsis lyrata</em>.
</p>


<h3>Format</h3>

<p>A data frame with 397 rows and 7 columns. Each row corresponds to a unique LTR retrotransposon, and each column corresponds to a feature of the LTR-RT. The columns are:
</p>

<dl>
<dt>SeqID</dt><dd><p>LTR retrotransposon sequence ID</p>
</dd>
<dt>UngapedLen</dt><dd><p>Length of each LTR</p>
</dd>
<dt>Mismatch</dt><dd><p>Number of mismatches</p>
</dd>
<dt>Distance</dt><dd><p>Divergence, as defined by (# of mismatches) / (LTR length)</p>
</dd>
<dt>sup</dt><dd><p>Super family membership</p>
</dd>
<dt>GroupID</dt><dd><p>LTR retrotransposon Family ID</p>
</dd>
<dt>thaID</dt><dd><p>Family name matched in the LTR-RT families of A. thaliana</p>
</dd>
</dl>


<h3>References</h3>

<p><cite>Lamesch, Philippe, Tanya Z. Berardini, Donghui Li, David Swarbreck, Christopher Wilks, Rajkumar Sasidharan, Robert Muller et al. &quot;The Arabidopsis Information Resource (TAIR): improved gene annotation and new tools.&quot; Nucleic acids research 40, no. D1 (2011): D1202-D1210.</cite>
</p>
<p><cite>Dai, X., Wang, H., Dvorak, J., Bennetzen, J., Mueller, H.-G. (2018+). &quot;Birth and Death of LTR Retrotransposons in Aegilops tauschii&quot;</cite>
</p>

<hr>
<h2 id='EstDynamics'>Estimate TE dynamics using mismatch data</h2><span id='topic+EstDynamics'></span><span id='topic+EstDynamics2'></span>

<h3>Description</h3>

<p>Given the number of mismatches and element lengths for an LTR retrotransposon family, estimate the age distribution, insertion rate, and deletion rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EstDynamics(mismatch, len, r = 0.013, perturb = 2, rateRange = NULL,
  plotFit = FALSE, plotSensitivity = FALSE, pause = plotFit &amp;&amp;
  plotSensitivity, main = sprintf("n = %d", n))

EstDynamics2(mismatch, len, r = 0.013, nTrial = 10L, perturb = 2,
  rateRange = NULL, plotFit = FALSE, plotSensitivity = FALSE,
  pause = plotFit &amp;&amp; plotSensitivity, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EstDynamics_+3A_mismatch">mismatch</code></td>
<td>
<p>A vector containing the number of mismatches.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_len">len</code></td>
<td>
<p>A vector containing the length of each element.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_r">r</code></td>
<td>
<p>Mutation rate (substitutions/(million year * site)) used in the calculation.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_perturb">perturb</code></td>
<td>
<p>A scalar multiple to perturb the estimated death rate from the null hypothesis estimate. Used to generate the sensitivity analysis.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_raterange">rateRange</code></td>
<td>
<p>A vector of death rates, an alternative to <code>perturb</code> for specifying the death rates.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_plotfit">plotFit</code></td>
<td>
<p>Whether to plot the distribution fits.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_plotsensitivity">plotSensitivity</code></td>
<td>
<p>Whether to plot the sensitivity analysis.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_pause">pause</code></td>
<td>
<p>Whether to pause after each plot.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_main">main</code></td>
<td>
<p>The title for the plot.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_ntrial">nTrial</code></td>
<td>
<p>The number of starting points for searching for the MLE.</p>
</td></tr>
<tr><td><code id="EstDynamics_+3A_...">...</code></td>
<td>
<p>Pass to EstDynamics</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>EstDynamics</code> estimates the TE dynamics through fitting a negative binomial fit to the mismatch data, while <code>EstDynamics2</code> uses a mixture model. For detailed implementation see References.
</p>


<h3>Value</h3>

<p><code>EstDynamics</code> returns a <code>TEfit</code> object, containing the following fields, where the unit for time is million years ago (Mya):
</p>
<table>
<tr><td><code>pvalue</code></td>
<td>
<p>The p-value for testing H_0: The insertion rate is uniform over time.</p>
</td></tr>
<tr><td><code>ageDist</code></td>
<td>
<p>A list containing the estimated age distributions.</p>
</td></tr>
<tr><td><code>insRt</code></td>
<td>
<p>A list containing the estimated insertion rates.</p>
</td></tr>
<tr><td><code>agePeakLoc</code></td>
<td>
<p>The maximum point (in age) of the age distribution. </p>
</td></tr>
<tr><td><code>insPeakLoc</code></td>
<td>
<p>The maximum point (in time) of the insertion rate.</p>
</td></tr>
<tr><td><code>estimates</code></td>
<td>
<p>The parameter estimates from fitting the distributions; see References</p>
</td></tr>
<tr><td><code>sensitivity</code></td>
<td>
<p>A list containing the results for the sensitivity analysis, with fields <code>time</code>: time points; <code>delRateRange</code>: A vector for the range of deletion rates; <code>insRange</code>: A matrix whose columns contain the insertion rates under different scenarios.</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>The sample size.</p>
</td></tr>
<tr><td><code>meanLen</code></td>
<td>
<p>The mean of element length.</p>
</td></tr>
<tr><td><code>meanDiv</code></td>
<td>
<p>The mean of divergence.</p>
</td></tr>
<tr><td><code>KDE</code></td>
<td>
<p>A list containing the kernel density estimate for the mismatch data. </p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>The log-likelihoods of the parametric fits.</p>
</td></tr>
</table>
<p>This function returns a <code>TEfit2</code> object, containing all the above fields for <code>TEfit</code> and the following:
</p>
<table>
<tr><td><code>estimates2</code></td>
<td>
<p>The parameter estimates from fitting the mixture distribution.</p>
</td></tr>
<tr><td><code>ageDist2</code></td>
<td>
<p>The estimated age distribution from fitting the mixture distribution.</p>
</td></tr> 
<tr><td><code>insRt2</code></td>
<td>
<p>The estimated insertion rate from fitting the mixture distribution.</p>
</td></tr> 
<tr><td><code>agePeakLoc2</code></td>
<td>
<p>Maximum point(s) for the age distribution.</p>
</td></tr>
<tr><td><code>insPeakLoc2</code></td>
<td>
<p>Maximum point(s) for the insertion rate.</p>
</td></tr>
</table>


<h3>References</h3>

<p><cite>Dai, X., Wang, H., Dvorak, J., Bennetzen, J., Mueller, H.-G. (2018). &quot;Birth and Death of LTR Retrotransposons in Aegilops tauschii&quot;. Genetics</cite>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Analyze Gypsy family 24 (Nusif)
data(AetLTR)
dat &lt;- subset(AetLTR, GroupID == 24 &amp; !is.na(Chr))
set.seed(1)
res1 &lt;- EstDynamics(dat$Mismatch, dat$UngapedLen, plotFit=TRUE, plotSensitivity=FALSE, pause=FALSE)

# p-value for testing a uniform insertion rate
res1$pvalue


# Use a mixture distribution to improve fit
res2 &lt;- EstDynamics2(dat$Mismatch, dat$UngapedLen, plotFit=TRUE)

# A larger number of trials is recommended to achieve the global MLE
## Not run: 
res3 &lt;- EstDynamics2(dat$Mismatch, dat$UngapedLen, plotFit=TRUE, nTrial=1000L)

## End(Not run)
</code></pre>

<hr>
<h2 id='MasterGene'>Implements the master gene model in Marchani et al (2009)</h2><span id='topic+MasterGene'></span>

<h3>Description</h3>

<p>Implements the master gene model in Marchani et al (2009)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MasterGene(mismatch, len, r = 0.013, plotFit = FALSE,
  main = sprintf("n = %d", n))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MasterGene_+3A_mismatch">mismatch</code></td>
<td>
<p>A vector containing the number of mismatches.</p>
</td></tr>
<tr><td><code id="MasterGene_+3A_len">len</code></td>
<td>
<p>A vector containing the length of each element.</p>
</td></tr>
<tr><td><code id="MasterGene_+3A_r">r</code></td>
<td>
<p>Mutation rate (substitutions/(million year * site)) used in the calculation.</p>
</td></tr>
<tr><td><code id="MasterGene_+3A_plotfit">plotFit</code></td>
<td>
<p>Whether to plot the distribution fits.</p>
</td></tr>
<tr><td><code id="MasterGene_+3A_main">main</code></td>
<td>
<p>The title for the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the method implemented see References.
</p>


<h3>Value</h3>

<p>This function returns various parameter estimates described in Marchani et al (2009), containing the following fields. The unit for time is million years ago (mya):
</p>
<table>
<tr><td><code>B</code></td>
<td>
<p>The constant insertion rate</p>
</td></tr>
<tr><td><code>q</code></td>
<td>
<p>The constant excision rate</p>
</td></tr>
<tr><td><code>lam</code></td>
<td>
<p>The population growth rate</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>The ratio of the number of elements in class j over class j+1, which is constant by assumption</p>
</td></tr>
<tr><td><code>age1</code></td>
<td>
<p>The age of the system under model 1 (lambda &gt; 1)</p>
</td></tr>
<tr><td><code>age2</code></td>
<td>
<p>The age of the system under model 2 (an initial burst followed by stasis lambda = 1)</p>
</td></tr>
</table>


<h3>References</h3>

<p><cite>Marchani, Elizabeth E., Jinchuan Xing, David J. Witherspoon, Lynn B. Jorde, and Alan R. Rogers. &quot;Estimating the age of retrotransposon subfamilies using maximum likelihood.&quot; Genomics 94, no. 1 (2009): 78-82.</cite>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Analyze Gypsy family 24 (Nusif)
data(AetLTR)
dat &lt;- subset(AetLTR, GroupID == 24 &amp; !is.na(Chr))
res2 &lt;- MasterGene(dat$Mismatch, dat$UngapedLen, plotFit=TRUE)
</code></pre>

<hr>
<h2 id='MatrixModel'>Implements the matrix model in Promislow et al (1999)</h2><span id='topic+MatrixModel'></span>

<h3>Description</h3>

<p>Implements the matrix model in Promislow et al (1999)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MatrixModel(mismatch, len, nsolo, r = 0.013, plotFit = FALSE,
  main = sprintf("n = %d", n))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MatrixModel_+3A_mismatch">mismatch</code></td>
<td>
<p>A vector containing the number of mismatches.</p>
</td></tr>
<tr><td><code id="MatrixModel_+3A_len">len</code></td>
<td>
<p>A vector containing the length of each element.</p>
</td></tr>
<tr><td><code id="MatrixModel_+3A_nsolo">nsolo</code></td>
<td>
<p>An integer giving the number of solo elements.</p>
</td></tr>
<tr><td><code id="MatrixModel_+3A_r">r</code></td>
<td>
<p>Mutation rate (substitutions/(million year * site)) used in the calculation.</p>
</td></tr>
<tr><td><code id="MatrixModel_+3A_plotfit">plotFit</code></td>
<td>
<p>Whether to plot the distribution fits.</p>
</td></tr>
<tr><td><code id="MatrixModel_+3A_main">main</code></td>
<td>
<p>The title for the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the method implemented see References.
</p>


<h3>Value</h3>

<p>This function returns various parameter estimates described in Promislow et al. (1999), containing the following fields. The unit for time is million years ago (Mya):
</p>
<table>
<tr><td><code>B</code></td>
<td>
<p>The constant insertion rate</p>
</td></tr>
<tr><td><code>q</code></td>
<td>
<p>The constant excision rate</p>
</td></tr>
<tr><td><code>lam</code></td>
<td>
<p>The population growth rate</p>
</td></tr>
<tr><td><code>R</code></td>
<td>
<p>The ratio of the number of elements in class j over class j+1, which is constant by assumption</p>
</td></tr>
<tr><td><code>age1</code></td>
<td>
<p>The age of the system under model 1 (lambda &gt; 1)</p>
</td></tr>
<tr><td><code>age2</code></td>
<td>
<p>The age of the system under model 2 (an initial burst followed by stasis lambda = 1)</p>
</td></tr>
</table>


<h3>References</h3>

<p><cite>Promislow, D., Jordan, K. and McDonald, J. &quot;Genomic demography: a life-history analysis of transposable element evolution.&quot; Proceedings of the Royal Society of London B: Biological Sciences 266, no. 1428 (1999): 1555-1560.</cite>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Analyze Gypsy family 24 (Nusif)
data(AetLTR)
dat &lt;- subset(AetLTR, GroupID == 24 &amp; !is.na(Chr))
res1 &lt;- MatrixModel(dat$Mismatch, dat$UngapedLen, nsolo=450, plotFit=TRUE)
</code></pre>

<hr>
<h2 id='nbLackOfFitKL'>Calcualte the KL divergence of a negative binomial fit to the mismatch data.</h2><span id='topic+nbLackOfFitKL'></span>

<h3>Description</h3>

<p>Calcualte the KL divergence of a negative binomial fit to the mismatch data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbLackOfFitKL(res)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nbLackOfFitKL_+3A_res">res</code></td>
<td>
<p>A TEfit object.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Analyze Gypsy family 24 (Nusif)
data(AetLTR)
dat &lt;- subset(AetLTR, GroupID == 24 &amp; !is.na(Chr))
set.seed(1)
res1 &lt;- EstDynamics(dat$Mismatch, dat$UngapedLen, plotFit=TRUE, plotSensitivity=FALSE, pause=FALSE)
nbLackOfFitKL(res1)
</code></pre>

<hr>
<h2 id='PlotFamilies'>Plot the age distributions or insertion rates for multiple families.</h2><span id='topic+PlotFamilies'></span>

<h3>Description</h3>

<p>Plot the age distributions or insertion rates for multiple families.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotFamilies(resList, type = c("insRt", "ageDist"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotFamilies_+3A_reslist">resList</code></td>
<td>
<p>A list of TEfit/TEfit2 objects, which can be mixed</p>
</td></tr>
<tr><td><code id="PlotFamilies_+3A_type">type</code></td>
<td>
<p>Whether to plot the insertion rates ('insRt') or the age distributions ('ageDist').</p>
</td></tr>
<tr><td><code id="PlotFamilies_+3A_...">...</code></td>
<td>
<p>Passed into plotting functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of line data (plotDat) and peak locations (peakDat).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AetLTR)
copia3 &lt;- subset(AetLTR, GroupID == 3 &amp; !is.na(Chr))
gypsy24 &lt;- subset(AetLTR, GroupID == 24 &amp; !is.na(Chr))
res3 &lt;- EstDynamics(copia3$Mismatch, copia3$UngapedLen)
res24 &lt;- EstDynamics2(gypsy24$Mismatch, gypsy24$UngapedLen)

# Plot insertion rates
PlotFamilies(list(`Copia 3`=res3, `Gypsy 24`=res24))

# Plot age distributions
PlotFamilies(list(`Copia 3`=res3, `Gypsy 24`=res24), type='ageDist')
</code></pre>

<hr>
<h2 id='print.TEfit'>Print a TEfit or TEfit2 object</h2><span id='topic+print.TEfit'></span><span id='topic+print.TEfit2'></span>

<h3>Description</h3>

<p>Print a TEfit or TEfit2 object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TEfit'
print(x, ...)

## S3 method for class 'TEfit2'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.TEfit_+3A_x">x</code></td>
<td>
<p>A TEfit or TEfit2 object</p>
</td></tr>
<tr><td><code id="print.TEfit_+3A_...">...</code></td>
<td>
<p>Not used</p>
</td></tr>
</table>

<hr>
<h2 id='SensitivityPlot'>Generate sensitivity plots</h2><span id='topic+SensitivityPlot'></span>

<h3>Description</h3>

<p>Create sensitivity plots of a few families to investigate different death rate scenarios
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SensitivityPlot(resList, col, xMax, markHalfPeak = FALSE,
  famLegend = TRUE, rLegend = names(resList), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SensitivityPlot_+3A_reslist">resList</code></td>
<td>
<p>A list of families returned by <code><a href="#topic+EstDynamics">EstDynamics</a></code></p>
</td></tr>
<tr><td><code id="SensitivityPlot_+3A_col">col</code></td>
<td>
<p>A vector of colors</p>
</td></tr>
<tr><td><code id="SensitivityPlot_+3A_xmax">xMax</code></td>
<td>
<p>The maximum of the x-axis</p>
</td></tr>
<tr><td><code id="SensitivityPlot_+3A_markhalfpeak">markHalfPeak</code></td>
<td>
<p>Whether to mark the time points with half-intensity</p>
</td></tr>
<tr><td><code id="SensitivityPlot_+3A_famlegend">famLegend</code></td>
<td>
<p>Whether to create legend for families</p>
</td></tr>
<tr><td><code id="SensitivityPlot_+3A_rlegend">rLegend</code></td>
<td>
<p>Text for the legend for families</p>
</td></tr>
<tr><td><code id="SensitivityPlot_+3A_...">...</code></td>
<td>
<p>Passed into <code>matplot</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(AetLTR)
copia3 &lt;- subset(AetLTR, GroupID == 3 &amp; !is.na(Chr))
copia9 &lt;- subset(AetLTR, GroupID == 9 &amp; !is.na(Chr))
res3 &lt;- EstDynamics(copia3$Mismatch, copia3$UngapedLen)
res9 &lt;- EstDynamics(copia9$Mismatch, copia9$UngapedLen)
SensitivityPlot(list(`Copia 3`=res3, `Copia 9`=res9))
</code></pre>

<hr>
<h2 id='TE'>TE: Insertion/Deletion Dynamics for Transposable Elements</h2><span id='topic+TE'></span><span id='topic+TE-package'></span>

<h3>Description</h3>

<p>TE package for analyzing insertion/deletion dynamics for transposable elements
</p>


<h3>Details</h3>

<p>Provides functions to estimate the insertion and deletion rates of
transposable element (TE) families. The estimation of insertion rate
consists of an improved estimate of the age distribution that takes into
account random mutations, and an adjustment by the deletion rate. This
package includes functions <code>EstDynamics</code> and <code>EstDynamics2</code> for
analyzing the TE divergence, and visualization functions such as
<code>PlotFamilies</code> and <code>SensitivityPlot</code>. 
This package implements the methods proposed in Dai et al (2018+).
</p>


<h3>Author(s)</h3>

<p>Xiongtao Dai <a href="mailto:xdai@iastate.edu">xdai@iastate.edu</a>,
Hao Wang
Jan Dvorak
Jeffrey Bennetzen
Hans-Georg Mueller
</p>
<p>Maintainer: Xiongtao Dai <a href="mailto:xdai@iastate.edu">xdai@iastate.edu</a>
</p>


<h3>References</h3>

<p><cite>Luo, Ming-Cheng, et al. (2017) &quot;Genome sequence of the progenitor of the wheat D genome Aegilops tauschii.&quot; Nature 551.7681.</cite>
</p>
<p><cite>Dai, X., Wang, H., Dvorak, J., Bennetzen, J., Mueller, H.-G. (2018). &quot;Birth and Death of LTR Retrotransposons in Aegilops tauschii&quot;. Genetics</cite>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
