<!DOCTYPE html><html lang="en"><head><title>Help for package TESS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TESS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TESS-package'><p>Diversification rate estimation and fast simulation of reconstructed phylogenetic trees under tree-wide time-heterogeneous birth-death processes including mass-extinction events</p></a></li>
<li><a href='#cettiidae'><p>Cettiidae phylogeny from Alstroem et al. (2011)</p></a></li>
<li><a href='#conifers'><p>Conifer phylogeny from Leslie et al. (2012)</p></a></li>
<li><a href='#mammalia'><p>Dated family level mammalian phylogeny from Meredith et al. (2011): Impacts of the cretaceous terrestrial revolution and kpg extinction on mammal diversification.</p></a></li>
<li><a href='#tess.analysis'><p>tess.analysis: Diversification rate estimation under an episodic birth-death process including mass-extinction events.</p></a></li>
<li><a href='#tess.likelihood'><p>tess.likelihood: Probability density of a tree under a tree-wide time-dependent birth-death process</p></a></li>
<li><a href='#tess.likelihood.rateshift'><p>tess.likelihood.rateshift: Probability density of a tree under a tree-wide time-dependent birth-death-shift process</p></a></li>
<li><a href='#tess.mcmc'><p>tess.mcmc: Markov chain Monte Carlo simulation using a general Metropolis-Hastings algorithm.</p></a></li>
<li><a href='#tess.nTaxa.expected'><p>tess.nTaxa.expected: The expected number of taxa at present of a tree under a global, time-dependent birth-death process (  E[ N(T) ]  )</p></a></li>
<li><a href='#tess.pathSampling'><p>tess.pathSampling: Marginal likelihood estimation via Path-Sampling.</p></a></li>
<li><a href='#tess.plot.multichain.diagnostics'><p>tess.plot.multichain.diagnostics: Plotting the mcmc diagnostics of a episodic diversification rate analysis with mass-extinction events.</p></a></li>
<li><a href='#tess.plot.output'><p>tess.plot.output: Plotting the output of a diversification rate estimation including mass-extinction events.</p></a></li>
<li><a href='#tess.plot.singlechain.diagnostics'><p>tess.plot.mcmc.diagnostics: Plotting the single chain mcmc diagnostics of a episodic diversification rate analysis with mass-extinction events.</p></a></li>
<li><a href='#tess.PosteriorPrediction'><p>tess.PosteriorPrediction: Approximation of the posterior predictive distribution.</p></a></li>
<li><a href='#tess.PosteriorPredictiveTest'><p>tess.PosteriorPredictiveTest: Approximation of the posterior predictive distribution.</p></a></li>
<li><a href='#tess.process.output'><p>tess.process.output: Summarizing the output of a diversification rate estimation including mass-extinction events. See the tess.analysis function for more information on how such output is generated and the tess.plot.output how the output can be visualized. Also have a look at the vignette for more in detail description and examples.</p></a></li>
<li><a href='#tess.sim.age'><p>tess.sim.age: Simulate a reconstructed tree for a given age under a global, time-dependent birth-death process.</p></a></li>
<li><a href='#tess.sim.taxa'><p>tess.sim.taxa.taxa: Simulate a reconstructed tree for a given number of taxa under a global, time-dependent birth-death process.</p></a></li>
<li><a href='#tess.sim.taxa.age'><p>tess.sim.taxa.taxa.age: Simulate a reconstructed tree for a given age and number of taxa under a global, time-dependent birth-death process.</p></a></li>
<li><a href='#tess.steppingStoneSampling'><p>tess.steppingStoneSampling: Marginal likelihood estimation via Stepping-Stone-Sampling.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Diversification Rate Estimation and Fast Simulation of
Reconstructed Phylogenetic Trees under Tree-Wide
Time-Heterogeneous Birth-Death Processes Including
Mass-Extinction Events</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-04-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Sebastian Hoehna and Michael R. May</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sebastian Hoehna &lt;Sebastian.Hoehna@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>ape, coda, deSolve</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulation of reconstructed phylogenetic trees under tree-wide time-heterogeneous birth-death processes and estimation of diversification parameters under the same model. Speciation and extinction rates can be any function of time and mass-extinction events at specific times can be provided. Trees can be simulated either conditioned on the number of species, the time of the process, or both. Additionally, the likelihood equations are implemented for convenience and can be used for Maximum Likelihood (ML) estimation and Bayesian inference.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.0), methods</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-25 14:21:21 UTC; bkopper</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-25 15:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='TESS-package'>Diversification rate estimation and fast simulation of reconstructed phylogenetic trees under tree-wide time-heterogeneous birth-death processes including mass-extinction events</h2><span id='topic+TESS-package'></span><span id='topic+TESS'></span>

<h3>Description</h3>

<p>Simulation of reconstructed phylogenetic trees under tree-wide time-heterogeneous birth-death processes and estimation of parameters under the same model. Speciation and extinction rates can be any function of time and mass-extinction events at specific times can be provided. Trees can be simulated either conditioned on the number of species, the time of the process, or both. Additionally, the likelihood equations are implemented for convenience and can be used for Maximum Likelihood (ML) estimation and Bayesian inference.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> TESS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-10-23</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Sebastian Hoehna and Michael R. May
</p>
<p>Maintainer: Sebastian Hoehna &lt;Sebastian.Hoehna@gmail.com&gt;
</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374.
</p>
<p>S. Hoehna: Likelihood inference of non-constant diversification rates with incomplete taxon sampling. 2014, PLoS One, 9(1), e84184.
</p>
<p>S. Hoehna: The time-dependent reconstructed evolutionary process with a key-role for mass-extinction events. 2015, Journal of Theoretical Biology, 380, 321-331.
</p>
<p>S. Hoehna, MR May and BR Moore: TESS: Bayesian inference of lineage diversification rates from (incompletely sampled) molecular phylogenies in R. 2015, Bioinformatics.
</p>
<p>MR May, S. Hoehna, and BR Moore: A Bayesian approach for detecting mass-extinction events when rates of lineage diversification vary. 2015, Systematic Biology
</p>


<h3>See Also</h3>

<p><code>ape</code>
<code>coda</code>
</p>

<hr>
<h2 id='cettiidae'>Cettiidae phylogeny from Alstroem et al. (2011)</h2><span id='topic+cettiidae'></span>

<h3>Description</h3>

<p>This phylogeny describes the species relationship and divergence times of the bird family Cettiidae, published in Alstroem et al. (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cettiidae)
</code></pre>


<h3>Format</h3>

<p>The phylogeny is stored as an object of class <code>"phylo"</code>.
The structure is described in the help page of the function
<code><a href="ape.html#topic+read.tree">read.tree</a></code> of the package <code><a href="ape.html#topic+ape">ape</a></code>.
</p>


<h3>Source</h3>

<p>Alstroem, P., Hoehna, S., Gelang, M., Ericson, P.G.P, and Olsson, U. (2011) Non-monophyly and intricate morphological evolution within the avian family Cettiidae revealed by multilocus analysis of a taxonomically densely sampled dataset, BMC Evolutionary Biology, 11:352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the data
data(cettiidae)

# safe the old plotting settings
op &lt;- par()

# set the new plotting settings
par(cex = 0.3)

# plot the phylogeny
plot(cettiidae)

# restore the plotting settings
par(op)
</code></pre>

<hr>
<h2 id='conifers'>Conifer phylogeny from Leslie et al. (2012)</h2><span id='topic+conifers'></span>

<h3>Description</h3>

<p>This phylogeny describes the species relationships and divergence times of the plant order Conifera, published in Leslie et al. (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(conifers)
</code></pre>


<h3>Format</h3>

<p>The phylogeny is stored as an object of class <code>"phylo"</code>.
The structure is described in the help page of the function
<code><a href="ape.html#topic+read.tree">read.tree</a></code> of the package <code><a href="ape.html#topic+ape">ape</a></code>.
</p>


<h3>Source</h3>

<p>Leslie, A. B., J. M. Beaulieu, H. S. Rai, P. R. Crane, M. J. Donoghue, and S. Mathews. 2012.  Hemisphere-scale differences in conifer evolutionary dynamics. Proceedings of the National Academy of Sciences 109:16217-16221.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the tree
data(conifers)

# safe the settings of the plotting device
op &lt;- par()

# set the line width for plotting the branches
par(cex = 0.3)

# plot the phylogenetic tree
plot(conifers)

# restore the settings of the device
par(op)
</code></pre>

<hr>
<h2 id='mammalia'>Dated family level mammalian phylogeny from Meredith et al. (2011): Impacts of the cretaceous terrestrial revolution and kpg extinction on mammal diversification.</h2><span id='topic+mammalia'></span>

<h3>Description</h3>

<p>This phylogeny describes the species relationship and divergence times of the class Mammalia with 1-3 species included per family, published in Meredith et al. (2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mammalia)
</code></pre>


<h3>Format</h3>

<p>The phylogeny is stored as an object of class <code>"phylo"</code>.
The structure is described in the help page of the function
<code><a href="ape.html#topic+read.tree">read.tree</a></code> of the package <code><a href="ape.html#topic+ape">ape</a></code>.
</p>


<h3>Source</h3>

<p>Meredith, R. et al. (2011): Impacts of the cretaceous terrestrial revolution and kpg extinction on mammal diversification. Science, 334:521-524
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the data
data(mammalia)

# safe the current settings of the plotting device
op &lt;- par()

# set the line width for drawing thinner lines for the branches
par(cex = 0.3)

# plot the mammalian phylogeny
plot(mammalia)

# restore the settings of the device
par(op)
</code></pre>

<hr>
<h2 id='tess.analysis'>tess.analysis: Diversification rate estimation under an episodic birth-death process including mass-extinction events.</h2><span id='topic+tess.analysis'></span><span id='topic+globalBiDe.analysis'></span>

<h3>Description</h3>

<p>tess.analysis estimates diversification rates under an episodic birth-death process including mass-extinction events. The method uses a reversible-jump MCMC algorithm to estimate the number, timing and magnitude of rate-shifts and mass-extinction events. It is possible to fix either number of events and provide specific values that will be used. We assume a Poison process for the number of events and a lognormal distribution with fixed, but specified, hyper-parameters for the speciation and extinction rate; and an independent Poison process for the number of mass-extinction events where each survival probability follows a Beta distribution with fixed hyper-parameters.
</p>
<p>The MCMC algorithm can be run either for a specified number of iterations, until a time limit in seconds has been reached, or until the effective sample size (ESS) has reached a given threshold. Once the first of these requirements are met TESS will stop the analysis. Internally we use scaling and sliding proposals to change the parameter values during the MCMC and a birth-move and death-move to add/remove events (rate-shifts or mass-extinction events).
</p>
<p>The results of the MCMC run are stored within a directory that is specified by the user. Several files will be generated containing the sampled parameter values. To summarize the output see tess.process.output(...) and tess.plot.output(...).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.analysis( tree,
               initialSpeciationRate,
               initialExtinctionRate,
               empiricalHyperPriors = TRUE,
               empiricalHyperPriorInflation = 10.0,
               empiricalHyperPriorForm = c("lognormal","normal","gamma"),
               speciationRatePriorMean = 0.0,
               speciationRatePriorStDev = 1.0,
               extinctionRatePriorMean = 0.0,
               extinctionRatePriorStDev = 1.0,
               initialSpeciationRateChangeTime = c(),
               initialExtinctionRateChangeTime = c(),
               estimateNumberRateChanges = TRUE,
               numExpectedRateChanges = 2,
               samplingProbability = 1,
               missingSpecies = c(),
               timesMissingSpecies = c(),
               tInitialMassExtinction = c(),
               pInitialMassExtinction = c(),
               pMassExtinctionPriorShape1 = 5,
               pMassExtinctionPriorShape2 = 95,
               estimateMassExtinctionTimes = TRUE,
               numExpectedMassExtinctions = 2,
               estimateNumberMassExtinctions = TRUE,
               MRCA = TRUE,
               CONDITION = "survival",
               BURNIN = 10000,
               MAX_ITERATIONS = 200000,
               THINNING = 100,
               OPTIMIZATION_FREQUENCY = 500,
               CONVERGENCE_FREQUENCY = 1000,
               MAX_TIME = Inf, MIN_ESS = 500,
               ADAPTIVE = TRUE,
               dir = "" ,
               priorOnly = FALSE,
               verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.analysis_+3A_tree">tree</code></td>
<td>
<p>The tree in 'phylo' format.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_initialspeciationrate">initialSpeciationRate</code></td>
<td>
<p>The initial value of the speciation rate when the MCMC is started. This can either be a single number of a vector of rates per interval.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_initialextinctionrate">initialExtinctionRate</code></td>
<td>
<p>The initial value of the extinction rate when the MCMC is started. This can either be a single number of a vector of rates per interval.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_empiricalhyperpriors">empiricalHyperPriors</code></td>
<td>
<p>Should we estimate the hyper-parameters empirically?</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_empiricalhyperpriorinflation">empiricalHyperPriorInflation</code></td>
<td>
<p>The scaling factor of the variance for the empirical hyperpriors.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_empiricalhyperpriorform">empiricalHyperPriorForm</code></td>
<td>
<p>The possible empirical hyper prior distributions; either lognormal, normal or gamma</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_speciationratepriormean">speciationRatePriorMean</code></td>
<td>
<p>The mean of the log-normal prior distribution for the speciation rate.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_speciationratepriorstdev">speciationRatePriorStDev</code></td>
<td>
<p>The standard deviation of the log-normal prior distribution for the speciation rate.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_extinctionratepriormean">extinctionRatePriorMean</code></td>
<td>
<p>The mean of the log-normal prior distribution for the extinction rate.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_extinctionratepriorstdev">extinctionRatePriorStDev</code></td>
<td>
<p>The standard deviation of the log-normal prior distribution for the extinction rate.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_initialspeciationratechangetime">initialSpeciationRateChangeTime</code></td>
<td>
<p>The initial value of the time points when speciation rate-shifts occur. The number of time-shifts needs to be one smaller than the number of initial speciation rates.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_initialextinctionratechangetime">initialExtinctionRateChangeTime</code></td>
<td>
<p>The initial value of the time points when extinction rate-shifts occur. The number of time-shifts needs to be one smaller than the number of initial extinction rates.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_estimatenumberratechanges">estimateNumberRateChanges</code></td>
<td>
<p>Do we estimate the number of rate shifts? Default is true.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_numexpectedratechanges">numExpectedRateChanges</code></td>
<td>
<p>Expected number of rate changes which follow a Poisson process. The default gives 0.5 probability on 0 shifts.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_samplingprobability">samplingProbability</code></td>
<td>
<p>The extant taxa sampling probability at the present time.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_missingspecies">missingSpecies</code></td>
<td>
<p>The number of species missed which originated in a given time interval (empirical taxon sampling).</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_timesmissingspecies">timesMissingSpecies</code></td>
<td>
<p>The times intervals of the missing species (empirical taxon sampling).</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_tinitialmassextinction">tInitialMassExtinction</code></td>
<td>
<p>The initial value of the vector of times of the mass-extinction events. This is used as initial values for the MCMC.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_pinitialmassextinction">pInitialMassExtinction</code></td>
<td>
<p>The initial value of the vector of survival probabilities of the mass-extinction events. This is used as initial values for the MCMC.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_pmassextinctionpriorshape1">pMassExtinctionPriorShape1</code></td>
<td>
<p>The alpha (first shape) parameter of the Beta prior distribution for the survival probability of a mass-extinction event.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_pmassextinctionpriorshape2">pMassExtinctionPriorShape2</code></td>
<td>
<p>The beta (second shape) parameter of the Beta prior distribution for the survival probability of a mass-extinction event.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_estimatemassextinctiontimes">estimateMassExtinctionTimes</code></td>
<td>
<p>Do we estimate the times of mass-extinction events? Default is true.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_numexpectedmassextinctions">numExpectedMassExtinctions</code></td>
<td>
<p>Expected number of mass-extinction events which follow a Poisson process. The default gives 0.5 probability on 0 events.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_estimatenumbermassextinctions">estimateNumberMassExtinctions</code></td>
<td>
<p>Do we estimate the number of mass-extinction events? Default is true.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_mrca">MRCA</code></td>
<td>
<p>Does the process start with the most recent common ancestor? If not, the tree must have a root edge!</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_condition">CONDITION</code></td>
<td>
<p>do we condition the process on time|survival|taxa?</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_burnin">BURNIN</code></td>
<td>
<p>The length of the burnin period.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_max_iterations">MAX_ITERATIONS</code></td>
<td>
<p>The maximum number of iteration of the MCMC. The default is 200000.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_thinning">THINNING</code></td>
<td>
<p>The frequency how often samples are recorded during the MCMC. The default is every 100 iterations.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_optimization_frequency">OPTIMIZATION_FREQUENCY</code></td>
<td>
<p>The frequency how often the MCMC moves are optimized. The default is every 500 iterations.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_convergence_frequency">CONVERGENCE_FREQUENCY</code></td>
<td>
<p>The frequency how often we check for convergence? The default is every 1000 iterations.</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_max_time">MAX_TIME</code></td>
<td>
<p>The maximum time the MCMC is allowed to run in seconds. The default is Inf</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_min_ess">MIN_ESS</code></td>
<td>
<p>The minimum number of effective samples (ESS) to assume convergence. The default is 500</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_adaptive">ADAPTIVE</code></td>
<td>
<p>Do we use auto-tuning of the MCMC moves? The default is TRUE (recommended).</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_dir">dir</code></td>
<td>
<p>The subdirectory in which the output will be stored. The default is the present directoy (&quot;&quot;)</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_prioronly">priorOnly</code></td>
<td>
<p>Do we sample from the prior only? The default is FALSE</p>
</td></tr>
<tr><td><code id="tess.analysis_+3A_verbose">verbose</code></td>
<td>
<p>Do you want detailed output?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>There is no return value because all the results are stored into files.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: The time-dependent reconstructed evolutionary process with a key-role for mass-extinction events. 2015, Journal of Theoretical Biology, 380, 321-331.
</p>
<p>S. Hoehna, MR May and BR Moore: TESS: Bayesian inference of lineage diversification rates from (incompletely sampled) molecular phylogenies in R. 2015, Bioinformatics.
</p>
<p>MR May, S. Hoehna, and BR Moore: A Bayesian approach for detecting mass-extinction events when rates of lineage diversification vary. 2015, Systematic Biology
</p>


<h3>Examples</h3>

<pre><code class='language-R'># we load the conifers as the test data set
data(conifers)

# for the conifers we know what the total number of species is
total &lt;- 630
# thus, we can compute what the sampling fraction is
rho &lt;- (conifers$Nnode+1)/total


# next, we specify the prior mean and standard deviation 
# for the speciation and extinction rate
mu_lambda = 0.15
std_lambda = 0.02
mu_mu = 0.09
std_mu = 0.02

# now we can run the entire analysis.
# note that a full analyses should be run much longer
tess.analysis( tree=conifers,
               initialSpeciationRate=exp(mu_lambda),
               initialExtinctionRate=exp(mu_mu),
               empiricalHyperPriors = FALSE,
               speciationRatePriorMean = mu_lambda,
               speciationRatePriorStDev = std_lambda,
               extinctionRatePriorMean = mu_mu,
               extinctionRatePriorStDev = std_mu,
               numExpectedRateChanges = 2,
               samplingProbability = rho,
               numExpectedMassExtinctions = 2,
               BURNIN = 100,
               MAX_ITERATIONS = 200,
               THINNING = 10,
               dir = "analysis_conifer")
               
# You may want to look into the vignette for a more detailed description
# of the features for an analysis.
# also have a look at the functions tess.process.output and tess.plot.output



</code></pre>

<hr>
<h2 id='tess.likelihood'>tess.likelihood: Probability density of a tree under a tree-wide time-dependent birth-death process</h2><span id='topic+tess.likelihood'></span>

<h3>Description</h3>

<p>tess.likelihood computes the probability of a reconstructed phylogenetic tree under time-dependent diversification rates. The rates may be any positive function of time or a constant. Additionally, mass-extinction event can be provided and a uniform taxon sampling probability. You have several options for the start of the process (origin vs MRCA) and the condition of the process (time, survival or taxa; note that survival and taxa implicitly condition on the time too!). See equation (5) in Hoehna (2013) for more information. Note that constant rates lead to much faster computations.
The likelihood can be computed for incompletely sampled trees if you give a sampling probability != 1.0. You have two options for the sampling strategy: uniform|diversified. The detailed description of these can be found in the references. More information can be obtained in the vignette about how to apply this likelihood function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.likelihood(times,
                lambda,
                mu,
                massExtinctionTimes=c(),
                massExtinctionSurvivalProbabilities=c(),
                missingSpecies = c(),
                timesMissingSpecies = c(),
                samplingProbability=1.0,
                samplingStrategy="uniform",
                MRCA=TRUE,
                CONDITION="survival",
                log=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.likelihood_+3A_times">times</code></td>
<td>
<p>The branching times of the phylogeny.</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_lambda">lambda</code></td>
<td>
<p>The speciation rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_mu">mu</code></td>
<td>
<p>The extinction rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_massextinctiontimes">massExtinctionTimes</code></td>
<td>
<p>The set of mass-extinction times after the start of the process.</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_massextinctionsurvivalprobabilities">massExtinctionSurvivalProbabilities</code></td>
<td>
<p>The set of survival probabilities for each speciation event. The set must have the same length as the set of mass-extinction times.</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_missingspecies">missingSpecies</code></td>
<td>
<p>The number of species missed which originated in a given time interval (empirical taxon sampling).</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_timesmissingspecies">timesMissingSpecies</code></td>
<td>
<p>The times intervals of the missing species (empirical taxon sampling).</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_samplingprobability">samplingProbability</code></td>
<td>
<p>The probability for a species to be included in the sample.</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_samplingstrategy">samplingStrategy</code></td>
<td>
<p>The strategy how samples were obtained. Options are: uniform|diversified|age.</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_mrca">MRCA</code></td>
<td>
<p>Does the process start with the most recent common ancestor? If not, the tree must have a root edge!</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_condition">CONDITION</code></td>
<td>
<p>do we condition the process on time|survival|taxa?</p>
</td></tr>
<tr><td><code id="tess.likelihood_+3A_log">log</code></td>
<td>
<p>Should we log-transform the likelihood?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the (log) probability of the tree, i.e. the likelihood of the parameters given the tree.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374
</p>
<p>S. Hoehna: Likelihood Inference of Non-Constant Diversification Rates with Incomplete Taxon Sampling. 2014, PLoS one, Public Library of Science, 9, e84184.
</p>
<p>S. Hoehna: The time-dependent reconstructed evolutionary process with a key-role for mass-extinction events. 2015, Journal of Theoretical Biology, 380, 321-331.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load a test data set
data(cettiidae)

# convert the phylogeny into the branching times
times &lt;- as.numeric( branching.times(cettiidae) )

# construct speciation and extinction rate function that resemble the rate-shift
# any other function could be used too
l &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (1) } else { return (2) } })
e &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (0.95) } else { return (0.5) } })

# now compute the likelihood for the tree
tess.likelihood(times,l,e,MRCA=TRUE,log=TRUE)

# a second approach is the specific episodic birth-death process likelihood function
# we need to give the rates for each episode and the end time of the episodes
# you should see that both are equivalent in this setting
# the function approach is more general but also slower.
tess.likelihood.rateshift(times,
				lambda=c(2,1,2),
				mu=c(0.95,0.5,0.95),
				rateChangeTimesLambda=c(0.3,0.5),
				rateChangeTimesMu=c(0.3,0.5),
				MRCA=TRUE,
				log=TRUE)


</code></pre>

<hr>
<h2 id='tess.likelihood.rateshift'>tess.likelihood.rateshift: Probability density of a tree under a tree-wide time-dependent birth-death-shift process</h2><span id='topic+tess.likelihood.rateshift'></span>

<h3>Description</h3>

<p>tess.likelihood.rateshift computes the probability of a reconstructed phylogenetic tree under a rate-shift model. The rates are piecewise constant. Additionally, mass-extinction event can be provided and a uniform taxon sampling probability. You have several options for the start of the process (origin vs MRCA) and the condition of the process (time, survival or taxa; note that survival and taxa implicitly condition on the time too!). See equation (5) in the manuscript for more information. Note that constant rates lead to much faster computations.
The likelihood can be computed for incompletely sampled trees. You need to give a sampling probability != 1.0. You have three options for the sampling strategy: uniform|diversified|age. The detailed description of these can be found in the references. More information can be obtained in the vignette about how to apply this likelihood function.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.likelihood.rateshift( times,
				 lambda, 
				 mu, 
				 rateChangeTimesLambda = c(),
				 rateChangeTimesMu = c(),
				 massExtinctionTimes = c(), 
   				 massExtinctionSurvivalProbabilities = c(),
                 missingSpecies = c(),
                 timesMissingSpecies = c(),
				 samplingStrategy = "uniform", 
   				 samplingProbability = 1, 
				 MRCA = TRUE, 
   				 CONDITION = "survival", 
				 log = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.likelihood.rateshift_+3A_times">times</code></td>
<td>
<p>The branching times of the tree.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_lambda">lambda</code></td>
<td>
<p>The speciation rate as a vector representing the rate for each time interval.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_mu">mu</code></td>
<td>
<p>The extinction rate as a vector representing the rate for each time interval.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_ratechangetimeslambda">rateChangeTimesLambda</code></td>
<td>
<p>The times of the rate-shifts for the speciation rate.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_ratechangetimesmu">rateChangeTimesMu</code></td>
<td>
<p>The times of the rate-shifts for the extinction rate.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_massextinctiontimes">massExtinctionTimes</code></td>
<td>
<p>The set of mass-extinction times after the start of the process.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_massextinctionsurvivalprobabilities">massExtinctionSurvivalProbabilities</code></td>
<td>
<p>The set of survival probabilities for each speciation event. The set must have the same length as the set of mass-extinction times.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_missingspecies">missingSpecies</code></td>
<td>
<p>The number of species missed which originated in a given time interval (empirical taxon sampling).</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_timesmissingspecies">timesMissingSpecies</code></td>
<td>
<p>The times intervals of the missing species (empirical taxon sampling).</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_samplingstrategy">samplingStrategy</code></td>
<td>
<p>The strategy how samples were obtained. Options are: uniform|diversified|age.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_samplingprobability">samplingProbability</code></td>
<td>
<p>The probability for a species to be included in the sample.</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_mrca">MRCA</code></td>
<td>
<p>Does the process start with the most recent common ancestor? If not, the tree must have a root edge!</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_condition">CONDITION</code></td>
<td>
<p>do we condition the process on time|survival|taxa?</p>
</td></tr>
<tr><td><code id="tess.likelihood.rateshift_+3A_log">log</code></td>
<td>
<p>should the likelihood be in log-scale?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the (log) probability of the tree, i.e., the likelihood of the parameters given the tree.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: The time-dependent reconstructed evolutionary process with a key-role for mass-extinction events. 2015, Journal of Theoretical Biology, 380, 321-331.</p>


<h3>Examples</h3>

<pre><code class='language-R'># load a test data set
data(cettiidae)

# convert the phylogeny into the branching times
times &lt;- as.numeric( branching.times(cettiidae) )

# construct speciation and extinction rate function that resemble the rate-shift
# any other function could be used too
l &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (1) } else { return (2) } })
e &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (0.95) } else { return (0.5) } })

# now compute the likelihood for the tree
tess.likelihood(times,l,e,MRCA=TRUE,log=TRUE)

# a second approach is the specific episodic birth-death process likelihood function
# we need to give the rates for each episode and the end time of the episodes
# you should see that both are equivalent in this setting
# the function approach is more general but also slower.
tess.likelihood.rateshift(times,
				lambda=c(2,1,2),
				mu=c(0.95,0.5,0.95),
				rateChangeTimesLambda=c(0.3,0.5),
				rateChangeTimesMu=c(0.3,0.5),
				MRCA=TRUE,
				log=TRUE)


</code></pre>

<hr>
<h2 id='tess.mcmc'>tess.mcmc: Markov chain Monte Carlo simulation using a general Metropolis-Hastings algorithm.</h2><span id='topic+tess.mcmc'></span>

<h3>Description</h3>

<p>tess.mcmc constructs a Markov chain Monte Carlo sampler (MCMC) by implementing a general Metropolis-Hastings algorithm. Any model can be used where the likelihood is known and thus can be passed in as an argument. The parameters have to be continuous. Proposals are taken from a normal distribution centered around the current value. The varaince of the new proposed values is initalized with 1 but can be automatically optimized when using the option adaptive = TRUE. The algorithm creates sampels from the posterior probility distribution and returns these a CODA mcmc object. More information can be obtained in the vignette about how to apply this method.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.mcmc(likelihoodFunction,priors,parameters,logTransforms,delta,
             iterations,burnin=round(iterations/3),thinning=1,
             adaptive=TRUE,verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.mcmc_+3A_likelihoodfunction">likelihoodFunction</code></td>
<td>
<p>The log-likelihood function which will be called internally by likelihoodFunction(parameters).</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_priors">priors</code></td>
<td>
<p>A list of functions of the log-prior-densities of each parameter.</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_parameters">parameters</code></td>
<td>
<p>The initial parameter value list.</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_logtransforms">logTransforms</code></td>
<td>
<p>A vector of booleans telling if log-transform for the parameters should be used (e.g. for rates).</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_delta">delta</code></td>
<td>
<p>The variance of new proposed values.</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_iterations">iterations</code></td>
<td>
<p>The number of iterations for the MCMC.</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_burnin">burnin</code></td>
<td>
<p>The number of iterations to burn before starting the MCMC.</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_thinning">thinning</code></td>
<td>
<p>The frequency of taking a sample of the parameters.</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_adaptive">adaptive</code></td>
<td>
<p>Should we use adaptive MCMC?</p>
</td></tr>
<tr><td><code id="tess.mcmc_+3A_verbose">verbose</code></td>
<td>
<p>Do you want detailed information during the run?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the posterior samples for the parameters.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374.
</p>
<p>S. Hoehna, MR May and BR Moore: TESS: Bayesian inference of lineage diversification rates from (incompletely sampled) molecular phylogenies in R. 2015, Bioinformatics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load in a test data set
data(cettiidae)

# convert the phylogeny into the branching times
times &lt;- as.numeric( branching.times(cettiidae) )

# specify a likelihood function that takes in a vector of parameters
likelihood &lt;- function(params) {
  # We use the parameters as diversification rate and turnover rate.
  # Thus we need to transform first
  b &lt;- params[1] + params[2]
  d &lt;- params[2]
  
  lnl &lt;- tess.likelihood(times,b,d,samplingProbability=1.0,log=TRUE)
  return (lnl)
}

# specify a the prior functions
prior.diversification &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
prior.turnover &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
priors &lt;- c(prior.diversification,prior.turnover)

# Note, the number of iterations and the burnin is too small here
# and should be adapted for real analyses
samples &lt;- tess.mcmc( likelihood,
		      priors,
		      runif(2,0,1),
		      logTransforms=c(TRUE,TRUE),
		      delta=c(0.1,0.1),
		      iterations=100,
		      burnin=20)

# now summarize and visualize the results
#plot(samples)
summary(samples)
colMeans(samples)


</code></pre>

<hr>
<h2 id='tess.nTaxa.expected'>tess.nTaxa.expected: The expected number of taxa at present of a tree under a global, time-dependent birth-death process (  E[ N(T) ]  )</h2><span id='topic+tess.nTaxa.expected'></span>

<h3>Description</h3>

<p>tess.nTaxa.expected computes the expected number of taxa at the present time T (the process start at time s and times increases until the present) under time-dependent. The rates may be any positive function of time or a constant. Additionally, mass-extinction event can be provided and a uniform taxon sampling probability. You have several options for the start of the process (origin vs MRCA). One important feature is that you can compute the expected number of taxa under the reconstructed process, that is, only lineages that survive until the present.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.nTaxa.expected( begin,
			   t,
			   end,
			   lambda,
			   mu,
			   massExtinctionTimes=c(),
			   massExtinctionSurvivalProbabilities=c(),
			   samplingProbability=1.0,
			   MRCA=TRUE,
			   reconstructed=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.nTaxa.expected_+3A_begin">begin</code></td>
<td>
<p>The time when the process starts.</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_t">t</code></td>
<td>
<p>The time at which we want to know the expected number of lineages (could be equal to end).</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_end">end</code></td>
<td>
<p>The time when the process end (e.g. the present).</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_lambda">lambda</code></td>
<td>
<p>The speciation rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_mu">mu</code></td>
<td>
<p>The extinction rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_massextinctiontimes">massExtinctionTimes</code></td>
<td>
<p>The set of mass-extinction times after the start of the process.</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_massextinctionsurvivalprobabilities">massExtinctionSurvivalProbabilities</code></td>
<td>
<p>The set of survival probabilities for each speciation event. The set must have the same length as the set of mass-extinction times.</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_samplingprobability">samplingProbability</code></td>
<td>
<p>The probability for a species to be included in the sample.</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_mrca">MRCA</code></td>
<td>
<p>Does the process start with the most recent common ancestor? If not, the tree must have a root edge!</p>
</td></tr>
<tr><td><code id="tess.nTaxa.expected_+3A_reconstructed">reconstructed</code></td>
<td>
<p>Are we computing the expected number of lineage at time t in the reconstructed process?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the expected number of taxa.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374 </p>


<h3>Examples</h3>

<pre><code class='language-R'>
# create the time-dependent speciation and extinction rate functions
# here we use episodic functions
l &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (1) } else { return (2) } })
e &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (0.95) } else { return (0.5) } })

# now we can compute the expected number of taxa at time t
# note that we compute here the actual diversity at time t
# if you set reconstructed=TRUE, then you get the expected
# number of lineages that will survive until the present
tess.nTaxa.expected(begin=0,t=2,end=5,l,e,MRCA=TRUE)


</code></pre>

<hr>
<h2 id='tess.pathSampling'>tess.pathSampling: Marginal likelihood estimation via Path-Sampling.</h2><span id='topic+tess.pathSampling'></span>

<h3>Description</h3>

<p>tess.pathSampling uses a power posterior series and path-sampling to estimate the marginal likelihood of a model. This is a very general implementation of this algorithm which can be applied basically to any model. More information can be obtained in the vignette about how to apply this method.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.pathSampling(likelihoodFunction,priorFunction,parameters,logTransforms,
                           iterations,burnin=round(iterations/3),K=50)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.pathSampling_+3A_likelihoodfunction">likelihoodFunction</code></td>
<td>
<p>The log-likelihood function which will be called internally by likelihoodFunction(parameters).</p>
</td></tr>
<tr><td><code id="tess.pathSampling_+3A_priorfunction">priorFunction</code></td>
<td>
<p>A list of functions of the log-prior-densities of each parameter.</p>
</td></tr>
<tr><td><code id="tess.pathSampling_+3A_parameters">parameters</code></td>
<td>
<p>The initial parameter value list.</p>
</td></tr>
<tr><td><code id="tess.pathSampling_+3A_logtransforms">logTransforms</code></td>
<td>
<p>A vector of booleans telling if log-transform for the parameters should be used (e.g. for rates).</p>
</td></tr>
<tr><td><code id="tess.pathSampling_+3A_iterations">iterations</code></td>
<td>
<p>The number of iterations for the MCMC.</p>
</td></tr>
<tr><td><code id="tess.pathSampling_+3A_burnin">burnin</code></td>
<td>
<p>The number of iterations to burn before starting the MCMC.</p>
</td></tr>
<tr><td><code id="tess.pathSampling_+3A_k">K</code></td>
<td>
<p>The number of stepping stones.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the posterior samples for the parameters.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>Lartillot, N. and Philippe, H., 2006: Computing Bayes factors using thermodynamic integration. Systematic Biology, 55, 195
</p>
<p>Baele et al., 2012: Improving the accuracy of demographic and molecular clock model comparison while accommodating phylogenetic uncertainty
</p>
<p>Baele et al., 2013: Accurate Model Selection of Relaxed Molecular Clocks in Bayesian Phylogenetics 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load a test data set
data(cettiidae)
# convert the phylogeny into the branching times
times &lt;- as.numeric( branching.times(cettiidae) )

# construct a likelihood function taking in a vector of parameters
likelihood &lt;- function(params) {
  # We use the parameters as diversification rate and turnover rate.
  # Thus we need to transform first
  b &lt;- params[1] + params[2]
  d &lt;- params[2]
  
  lnl &lt;- tess.likelihood(times,b,d,samplingProbability=1.0,log=TRUE)
  return (lnl)
}

# next, create the prior density functions
prior_diversification &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
prior_turnover &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
priors &lt;- c(prior_diversification,prior_turnover)

# Note, the number of iterations, the burnin
# and the number of stepping stones is too small here
# and should be adapted for real analyses
marginalLikelihood &lt;- tess.pathSampling( likelihood,
						  priors,
						  runif(2,0,1),
						  c(TRUE,TRUE),
						  10,
						  10,
						  K=4)


</code></pre>

<hr>
<h2 id='tess.plot.multichain.diagnostics'>tess.plot.multichain.diagnostics: Plotting the mcmc diagnostics of a episodic diversification rate analysis with mass-extinction events.</h2><span id='topic+tess.plot.multichain.diagnostics'></span>

<h3>Description</h3>

<p>tess.plot.multichain.diagnostics plots MCMC diagnostics for the output generated by a tess.process.output(...) command. Fore more examples see the vignette.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.plot.multichain.diagnostics(outputs,
                                      parameters=c("speciation rates",
                                                   "speciation shift times",
                                                   "extinction rates",
                                                   "extinction shift times",
                                                   "net-diversification rates",
                                                   "relative-extinction rates",
                                                   "mass extinction times"),
                                      diagnostics="Gelman-Rubin",
                                      gelman.crit=1.05,
                                      xlab="million years ago",
                                      col=NULL,
                                      xaxt="n",
                                      yaxt="s",
                                      pch=19,
                                      ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_outputs">outputs</code></td>
<td>
<p>The processed output for plotting.</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_parameters">parameters</code></td>
<td>
<p>Which parameters to diagnose. See details for a complete description.</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_diagnostics">diagnostics</code></td>
<td>
<p>Which diagnostics to use. Currently the only option is &quot;Rubin-Gelman&quot;.</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_gelman.crit">gelman.crit</code></td>
<td>
<p>The critical value above which a Rubin-Gelman statistic is considered a failure.</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_xlab">xlab</code></td>
<td>
<p>The label of the x-axis. By default, millions of years.</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_col">col</code></td>
<td>
<p>Colors used for printing. Must be of same length as fig.types.</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_xaxt">xaxt</code></td>
<td>
<p>The type of x-axis to plot. By default, no x-axis is plotted (recommended).</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_yaxt">yaxt</code></td>
<td>
<p>The type of y-axis to plot.</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_pch">pch</code></td>
<td>
<p>The type of points to draw (if points are drawn).</p>
</td></tr>
<tr><td><code id="tess.plot.multichain.diagnostics_+3A_...">...</code></td>
<td>
<p>Arguments delegated to plot()</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates visual summaries of multi-chain MCMC diagnostics for the CoMET analysis in the output object.
The argument parameters specifies the aspects of the model to summarize.
Valid options are:
</p>

<ul>
<li><p>speciation rates: Plots the interval-specific speciation rates.
</p>
</li>
<li><p>speciation shift times: Plots the posterior probability of at least one speciation-rate shift for each interval.
</p>
</li>
<li><p>extinction rates: Plots the interval-specific extinction rates.
</p>
</li>
<li><p>extinction shift times: Plots the posterior probability of at least one extinction-rate shift for each interval.
</p>
</li>
<li><p>net-diversification ratesPlots the interval-specific net-diversification rates.
</p>
</li>
<li><p>relative-extinction ratesPlots the interval-specific relative-extinction rates.
</p>
</li>
<li><p>mass extinction times: Plots the posterior probability of at least one mass-extinction event for each interval.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Michael R. May</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data, compute the sampling fraction rho
data(conifers)
totalConiferSpecies &lt;- 630
sampledConiferSpecies &lt;- conifers$Nnode+1
rho &lt;- sampledConiferSpecies / totalConiferSpecies

# Run a tess analysis
tess.analysis(tree = conifers,
              initialSpeciationRate=c(1.0),
              initialExtinctionRate=c(0.5),
              empiricalHyperPriors = FALSE,
              numExpectedRateChanges = 2,
              numExpectedMassExtinctions = 2,
              samplingProbability = rho,
              MAX_ITERATIONS = 200,
              BURNIN = 100,
              dir = "./run_1")

tess.analysis(tree = conifers,
              initialSpeciationRate=c(1.0),
              initialExtinctionRate=c(0.5),
              empiricalHyperPriors = FALSE,
              numExpectedRateChanges = 2,
              numExpectedMassExtinctions = 2,
              samplingProbability = rho,
              MAX_ITERATIONS = 200,
              BURNIN = 100,
              dir = "./run_2")

# Process the output
coniferOutput_1 &lt;- tess.process.output(dir="./run_1",
                                     numExpectedRateChanges=2,
                                     numExpectedMassExtinctions=2)

coniferOutput_2 &lt;- tess.process.output(dir="./run_2",
                                     numExpectedRateChanges=2,
                                     numExpectedMassExtinctions=2)

# Plot the output
outputs &lt;- list(coniferOutput_1,coniferOutput_2)
tess.plot.multichain.diagnostics(outputs)

</code></pre>

<hr>
<h2 id='tess.plot.output'>tess.plot.output: Plotting the output of a diversification rate estimation including mass-extinction events.</h2><span id='topic+tess.plot.output'></span>

<h3>Description</h3>

<p>tess.output.summary plots the output generated by a tess.process.output(...) command. More specifically, you can plot the speciation, extinction, diversification and relative extinction rate over time, as well as the probability and Bayes factor for the timing of rate shifts and mass-extinction events. For more examples see the vignette.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.plot.output(output,
                 fig.types=c("speciation rates",
                             "speciation shift times",
                             "speciation Bayes factors",
                             "extinction rates",
                             "extinction shift times",
                             "extinction Bayes factors",
                             "net-diversification rates",
                             "relative-extinction rates",
                             "mass extinction times",
                             "mass extinction Bayes factors"),
                 xlab="million years ago",
                 col=NULL,
                 col.alpha=50,
                 xaxt="n",
                 yaxt="s",
                 pch=19,
                 plot.tree=FALSE,
                            ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.plot.output_+3A_output">output</code></td>
<td>
<p>The processed output for plotting.</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_fig.types">fig.types</code></td>
<td>
<p>Which aspects of the model to visualize. See details for a complete description.</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_xlab">xlab</code></td>
<td>
<p>The label of the x-axis. By default, millions of years.</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_col">col</code></td>
<td>
<p>Colors used for printing. Must be of same length as fig.types.</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_col.alpha">col.alpha</code></td>
<td>
<p>Alpha channel parameter for credible intervals.</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_xaxt">xaxt</code></td>
<td>
<p>The type of x-axis to plot. By default, no x-axis is plotted (recommended).</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_yaxt">yaxt</code></td>
<td>
<p>The type of y-axis to plot.</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_pch">pch</code></td>
<td>
<p>The type of points to draw (if points are drawn).</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_plot.tree">plot.tree</code></td>
<td>
<p>Are we plotting the tree too?</p>
</td></tr>
<tr><td><code id="tess.plot.output_+3A_...">...</code></td>
<td>
<p>Arguments delegated to plot()</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates visual summaries of the CoMET analysis in the output object.
The argument fig.types specifies the aspects of the model to summarize.
Valid options are:
</p>

<ul>
<li><p>speciation rates: Plots the interval-specific speciation rates.
</p>
</li>
<li><p>speciation shift times: Plots the posterior probability of at least one speciation-rate shift for each interval.
</p>
</li>
<li><p>speciation Bayes factors: Plots the Bayes factor support for at least one speciation-rate shift for each interval (as 2 ln BF).
</p>
</li>
<li><p>extinction rates: Plots the interval-specific extinction rates.
</p>
</li>
<li><p>extinction shift times: Plots the posterior probability of at least one extinction-rate shift for each interval.
</p>
</li>
<li><p>extinction Bayes factors: Plots the Bayes factor support for at least one extinction-rate shift for each interval (as 2 ln BF).
</p>
</li>
<li><p>net-diversification ratesPlots the interval-specific net-diversification rates.
</p>
</li>
<li><p>relative-extinction ratesPlots the interval-specific relative-extinction rates.
</p>
</li>
<li><p>mass extinction times: Plots the posterior probability of at least one mass-extinction event for each interval.
</p>
</li>
<li><p>mass extinction Bayes factors: Plots the Bayes factor support for at least one mass-extinction event for each interval (as 2 ln BF).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Michael R. May</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data, compute the sampling fraction rho
data(conifers)
totalConiferSpecies &lt;- 630
sampledConiferSpecies &lt;- conifers$Nnode+1
rho &lt;- sampledConiferSpecies / totalConiferSpecies

# Run a tess analysis
tess.analysis(tree = conifers,
              initialSpeciationRate=c(1.0),
              initialExtinctionRate=c(0.5),
              empiricalHyperPriors = FALSE,
              numExpectedRateChanges = 2,
              numExpectedMassExtinctions = 2,
              samplingProbability = rho,
              MAX_ITERATIONS = 200,
              BURNIN = 100)

# Process the output
coniferOutput &lt;- tess.process.output(dir=getwd(),
                                     numExpectedRateChanges=2,
                                     numExpectedMassExtinctions=2)

# Plot the output
tess.plot.output(coniferOutput)
</code></pre>

<hr>
<h2 id='tess.plot.singlechain.diagnostics'>tess.plot.mcmc.diagnostics: Plotting the single chain mcmc diagnostics of a episodic diversification rate analysis with mass-extinction events.</h2><span id='topic+tess.plot.singlechain.diagnostics'></span>

<h3>Description</h3>

<p>tess.plot.singlechain.diagnostics plots MCMC diagnostics for the output generated by a tess.process.output(...) command. Fore more examples see the vignette.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.plot.singlechain.diagnostics(output,
                                      parameters=c("speciation rates",
                                                   "speciation shift times",
                                                   "extinction rates",
                                                   "extinction shift times",
                                                   "net-diversification rates",
                                                   "relative-extinction rates",
                                                   "mass extinction times"),
                                      diagnostics=c("ESS","geweke"),
                                      ess.crit=c(100,200),
                                      geweke.crit=0.05,
                                      correction="bonferroni",
                                      xlab="million years ago",
                                      col=NULL,
                                      xaxt="n",
                                      yaxt="s",
                                      pch=19,
                                      ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_output">output</code></td>
<td>
<p>The processed output for plotting.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_parameters">parameters</code></td>
<td>
<p>Which parameters to diagnose. See details for a complete description.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_diagnostics">diagnostics</code></td>
<td>
<p>Which diagnostics to use. Options are &quot;ESS&quot; and &quot;geweke&quot;.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_ess.crit">ess.crit</code></td>
<td>
<p>Two values which correspond to low ESS threshold and acceptable ESS threshold. Default values are 100 and 200.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_geweke.crit">geweke.crit</code></td>
<td>
<p>The p-value cutoff for Geweke's diagnostic. Default is the canonical 0.05.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_correction">correction</code></td>
<td>
<p>What type of multiple-correction method to use. Options are &quot;bonferroni&quot; and &quot;sidak&quot;.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_xlab">xlab</code></td>
<td>
<p>The label of the x-axis. By default, millions of years.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_col">col</code></td>
<td>
<p>Colors used for printing. Must be of same length as fig.types.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_xaxt">xaxt</code></td>
<td>
<p>The type of x-axis to plot. By default, no x-axis is plotted (recommended).</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_yaxt">yaxt</code></td>
<td>
<p>The type of y-axis to plot.</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_pch">pch</code></td>
<td>
<p>The type of points to draw (if points are drawn).</p>
</td></tr>
<tr><td><code id="tess.plot.singlechain.diagnostics_+3A_...">...</code></td>
<td>
<p>Arguments delegated to plot()</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates visual summaries of single-chain MCMC diagnostics for the CoMET analysis in the output object.
The argument parameters specifies the aspects of the model to summarize.
Valid options are:
</p>

<ul>
<li><p>speciation rates: Plots the interval-specific speciation rates.
</p>
</li>
<li><p>speciation shift times: Plots the posterior probability of at least one speciation-rate shift for each interval.
</p>
</li>
<li><p>extinction rates: Plots the interval-specific extinction rates.
</p>
</li>
<li><p>extinction shift times: Plots the posterior probability of at least one extinction-rate shift for each interval.
</p>
</li>
<li><p>net-diversification ratesPlots the interval-specific net-diversification rates.
</p>
</li>
<li><p>relative-extinction ratesPlots the interval-specific relative-extinction rates.
</p>
</li>
<li><p>mass extinction times: Plots the posterior probability of at least one mass-extinction event for each interval.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Michael R. May</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data, compute the sampling fraction rho
data(conifers)
totalConiferSpecies &lt;- 630
sampledConiferSpecies &lt;- conifers$Nnode+1
rho &lt;- sampledConiferSpecies / totalConiferSpecies

# Run a tess analysis
tess.analysis(tree = conifers,
              initialSpeciationRate=c(1.0),
              initialExtinctionRate=c(0.5),
              empiricalHyperPriors = FALSE,
              numExpectedRateChanges = 2,
              numExpectedMassExtinctions = 2,
              samplingProbability = rho,
              MAX_ITERATIONS = 200,
              BURNIN = 100)

# Process the output
coniferOutput &lt;- tess.process.output(dir=getwd(),
                                     numExpectedRateChanges=2,
                                     numExpectedMassExtinctions=2)

# Plot the output
tess.plot.singlechain.diagnostics(coniferOutput)
</code></pre>

<hr>
<h2 id='tess.PosteriorPrediction'>tess.PosteriorPrediction: Approximation of the posterior predictive distribution.</h2><span id='topic+tess.PosteriorPrediction'></span>

<h3>Description</h3>

<p>tess.PosteriorPrediction calls the simulation function exactly once for each sampled parameter combination. In that way, posterior predictive simulations can be obtained which then in turn can be used to compute summary statistics based on these posterior predictive simulations. Fore more information see the vignette.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.PosteriorPrediction(simulationFunction,parameters,burnin)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.PosteriorPrediction_+3A_simulationfunction">simulationFunction</code></td>
<td>
<p>The simulation function which will be called internally by simulationFunction(parameters).</p>
</td></tr>
<tr><td><code id="tess.PosteriorPrediction_+3A_parameters">parameters</code></td>
<td>
<p>A matrix of parameters where the rows represent samples of parameters and the column the different parameters.</p>
</td></tr>
<tr><td><code id="tess.PosteriorPrediction_+3A_burnin">burnin</code></td>
<td>
<p>The fraction of samples to be discarded as burnin. This is 0.25 by default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns samples simulated from the posterior predictive distribution.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374 </p>


<h3>Examples</h3>

<pre><code class='language-R'># We first run an MCMC to obtain samples from the posterior distribution 
# and then simulate the posterior predictive distribution.

# The bird phylogeny as the test data set
data(cettiidae)
times &lt;- as.numeric( branching.times(cettiidae) )

# The log-likelihood function
likelihood &lt;- function(params) {
  # We use the parameters as diversification rate and turnover rate.
  # Thus we need to transform first
  b &lt;- params[1] + params[2]
  d &lt;- params[2]
  
  lnl &lt;- tess.likelihood(times,b,d,samplingProbability=1.0,log=TRUE)
  return (lnl)
}

prior_diversification &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
prior_turnover &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
priors &lt;- c(prior_diversification,prior_turnover)

# Note, the number of iterations and the burnin is too small here 
# and should be adapted for real analyses
samples &lt;- tess.mcmc(likelihood,priors,c(1,0.1),c(TRUE,TRUE),c(0.1,0.1),10,10)

tmrca &lt;- max(branching.times(cettiidae))
# The simulation function
sim &lt;- function(params) {
  # We use the parameters as diversification rate and turnover rate.
  # Thus we need to transform first
  b &lt;- params[1] + params[2]
  d &lt;- params[2]
  
  tree &lt;- tess.sim.age(n=1,age=tmrca,b,d,samplingProbability=1.0)[[1]]
  return (tree)
}

trees &lt;- tess.PosteriorPrediction(sim,samples)

# compute the posterior predictive test statistic
ppt &lt;- tess.PosteriorPredictiveTest(trees,cettiidae,gammaStat)
# get the p-value of the observed test-statistic
ppt[[2]]

</code></pre>

<hr>
<h2 id='tess.PosteriorPredictiveTest'>tess.PosteriorPredictiveTest: Approximation of the posterior predictive distribution.</h2><span id='topic+tess.PosteriorPredictiveTest'></span>

<h3>Description</h3>

<p>tess.PosteriorPredictiveTest computes the values of the statistic for the posterior predictive simulations and computes the p-value for the observed statistic.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.PosteriorPredictiveTest(samples,observation,statistic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.PosteriorPredictiveTest_+3A_samples">samples</code></td>
<td>
<p>Samples from the posterior predictive distribution.</p>
</td></tr>
<tr><td><code id="tess.PosteriorPredictiveTest_+3A_observation">observation</code></td>
<td>
<p>The observed value.</p>
</td></tr>
<tr><td><code id="tess.PosteriorPredictiveTest_+3A_statistic">statistic</code></td>
<td>
<p>The function that computes the statistic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of the statistic for each sample.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374 </p>


<h3>Examples</h3>

<pre><code class='language-R'># We first run an MCMC to obtain samples from the posterior distribution
# and then simulate the posterior predictive distribution.

# The bird phylogeny as the test data set
data(cettiidae)
times &lt;- as.numeric( branching.times(cettiidae) )

# The log-likelihood function
likelihood &lt;- function(params) {
  # We use the parameters as diversification rate and turnover rate.
  # Thus we need to transform first
  b &lt;- params[1] + params[2]
  d &lt;- params[2]
  
  lnl &lt;- tess.likelihood(times,b,d,samplingProbability=1.0,log=TRUE)
  return (lnl)
}

prior_diversification &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
prior_turnover &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
priors &lt;- c(prior_diversification,prior_turnover)

# Note, the number of iterations and the burnin is too small here
# and should be adapted for real analyses
samples &lt;- tess.mcmc(likelihood,priors,c(1,0.1),c(TRUE,TRUE),c(0.1,0.1),10,10)

tmrca &lt;- max(branching.times(cettiidae))
# The simulation function
sim &lt;- function(params) {
  # We use the parameters as diversification rate and turnover rate.
  # Thus we need to transform first
  b &lt;- params[1] + params[2]
  d &lt;- params[2]
  
  # We need trees with at least three tips for the gamma-statistics
  repeat {
    tree &lt;- tess.sim.age(n=1,age=tmrca,b,d,samplingProbability=1.0,MRCA=TRUE)[[1]]
    if (tree$Nnode &gt; 1) break
  }
  return (tree)
}

# simulate trees from the posterior predictive distribution
trees &lt;- tess.PosteriorPrediction(sim,samples)

# compute the posterior predictive test statistic
ppt &lt;- tess.PosteriorPredictiveTest(trees,cettiidae,gammaStat)
# get the p-value of the observed test-statistic
ppt[[2]]

</code></pre>

<hr>
<h2 id='tess.process.output'>tess.process.output: Summarizing the output of a diversification rate estimation including mass-extinction events. See the tess.analysis function for more information on how such output is generated and the tess.plot.output how the output can be visualized. Also have a look at the vignette for more in detail description and examples.</h2><span id='topic+tess.process.output'></span><span id='topic+globalBiDe.output.summary'></span>

<h3>Description</h3>

<p>tess.process.output summarizes the output generated by a tess.analysis(...) run.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.process.output(dir,
                    tree=NULL,
                    numExpectedRateChanges=2,
                    numExpectedMassExtinctions=2,
                    burnin=0.25,
                    numIntervals=100,
                    criticalBayesFactors=c(2,6,10))

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.process.output_+3A_dir">dir</code></td>
<td>
<p>The directory from which the CoMET output will be read.</p>
</td></tr>
<tr><td><code id="tess.process.output_+3A_tree">tree</code></td>
<td>
<p>The tree analyzed with CoMET in phylo format. By default, looks for a tree in the target directory.</p>
</td></tr>
<tr><td><code id="tess.process.output_+3A_numexpectedratechanges">numExpectedRateChanges</code></td>
<td>
<p>The number of expected diversification-rate changes.</p>
</td></tr>
<tr><td><code id="tess.process.output_+3A_numexpectedmassextinctions">numExpectedMassExtinctions</code></td>
<td>
<p>The number of expected mass-extinction events.</p>
</td></tr>
<tr><td><code id="tess.process.output_+3A_burnin">burnin</code></td>
<td>
<p>The fraction of samples that will be discarded as burnin.</p>
</td></tr>
<tr><td><code id="tess.process.output_+3A_numintervals">numIntervals</code></td>
<td>
<p>The number of discrete intervals in which to break the tree.</p>
</td></tr>
<tr><td><code id="tess.process.output_+3A_criticalbayesfactors">criticalBayesFactors</code></td>
<td>
<p>The Bayes factor thresholds to use to assess significance of events.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output of a CoMET analysis is stored in a directory with different files containing the MCMC samples from the posterior distribution.
For example, the tess.analysis function stores the times and survival probabilities of the mass-extinction events in a file.
This function, converts the output by counting the number of events that fall into a given time-bin.
This pre-processing of the output simplifies the plotting.
</p>


<h3>Value</h3>

<p>This function returns a list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>posterior</code></td>
<td>
<p>An object of class 'mcmc' that contains the trace of the model's posterior probability.</p>
</td></tr>
<tr><td><code>numSpeciationCategories</code></td>
<td>
<p>An object of class 'mcmc' that contains samples from the posterior distribution of the number of speciation categories (minimum 1, since this includes the initial speciation rate).</p>
</td></tr>
<tr><td><code>numExtinctionCategories</code></td>
<td>
<p>An object of class 'mcmc' that contains samples from the posterior distribution of the number of extinction categories (minimum 1, since this includes the initial extinction rate).</p>
</td></tr>
<tr><td><code>numMassExtinctions</code></td>
<td>
<p>An object of class 'mcmc' that contains samples from the posterior distribution of the number of mass-extinction events.</p>
</td></tr>
<tr><td><code>speciation rates</code></td>
<td>
<p>An object of class 'mcmc' that contains speciation rates sampled from the posterior distribution for each of numIntervals discrete time intervals. Rows correspond to samples from the posterior distribution, columns correspond to intervals.</p>
</td></tr>
<tr><td><code>speciation change times</code></td>
<td>
<p>An object of class 'mcmc' that contains speciation-rate-change events sampled from the posterior distribution for each of numIntervals discrete time intervals. A value of 1 indicates an event was contained in the interval, 0 that no event was contained in the interval. Rows correspond to samples from the posterior distribution, columns correspond to intervals.</p>
</td></tr>
<tr><td><code>speciation Bayes factors</code></td>
<td>
<p>A vector of class 'numeric' that contains the Bayes factor support for there being a speciation-rate-change event for each of numIntervals discrete time intervals. The ith element corresponds to the Bayes factor support for an event in the ith interval.</p>
</td></tr>
<tr><td><code>speciationRateChangeCriticalPosteriorProbabilities</code></td>
<td>
<p>A vector of posterior probabilities that correspond to critical Bayes factor thresholds (specified by the argument criticalBayesFactors). Element i is the posterior probability of a speciation-rate-change event in an interval needed to produce Bayes factor support of criticalBayesFactors[i].</p>
</td></tr>
<tr><td><code>extinction rates</code></td>
<td>
<p>An object of class 'mcmc' that contains extinction rates sampled from the posterior distribution for each of numIntervals discrete time intervals. Rows correspond to samples from the posterior distribution, columns correspond to intervals.</p>
</td></tr>
<tr><td><code>extinction change times</code></td>
<td>
<p>An object of class 'mcmc' that contains extinction-rate-change events sampled from the posterior distribution for each of numIntervals discrete time intervals. A value of 1 indicates an event was contained in the interval, 0 that no event was contained in the interval. Rows correspond to samples from the posterior distribution, columns correspond to intervals.</p>
</td></tr>
<tr><td><code>extinction Bayes factors</code></td>
<td>
<p>A vector of class 'numeric' that contains the Bayes factor support for there being a extinction-rate-change event for each of numIntervals discrete time intervals. The ith element corresponds to the Bayes factor support for an event in the ith interval.</p>
</td></tr>
<tr><td><code>extinctionRateChangeCriticalPosteriorProbabilities</code></td>
<td>
<p>A vector of posterior probabilities that correspond to critical Bayes factor thresholds (specified by the argument criticalBayesFactors). Element i is the posterior probability of a extinction-rate-change event in an interval needed to produce Bayes factor support of criticalBayesFactors[i].</p>
</td></tr>
<tr><td><code>net-diversification rates</code></td>
<td>
<p>An object of class 'mcmc' that contains net-diversification (speciation - extinction) rates sampled from the posterior distribution for each of numIntervals discrete time intervals. Rows correspond to samples from the posterior distribution, columns correspond to intervals.</p>
</td></tr>
<tr><td><code>relative-extinction rates</code></td>
<td>
<p>An object of class 'mcmc' that contains relative-extinction (extinction / speciation) rates sampled from the posterior distribution for each of numIntervals discrete time intervals. Rows correspond to samples from the posterior distribution, columns correspond to intervals.</p>
</td></tr>
<tr><td><code>mass extinction times</code></td>
<td>
<p>An object of class 'mcmc' that contains mass-extinction events sampled from the posterior distribution for each of numIntervals discrete time intervals. A value of 1 indicates an event was contained in the interval, 0 that no event was contained in the interval. Rows correspond to samples from the posterior distribution, columns correspond to intervals.</p>
</td></tr>
<tr><td><code>mass extinction Bayes factors</code></td>
<td>
<p>A vector of class 'numeric' that contains the Bayes factor support for there being a mass-extinction event for each of numIntervals discrete time intervals. The ith element corresponds to the Bayes factor support for an event in the ith interval.</p>
</td></tr>
<tr><td><code>massExtinctionCriticalPosteriorProbabilities</code></td>
<td>
<p>A vector of posterior probabilities that correspond to critical Bayes factor thresholds (specified by the argument criticalBayesFactors). Element i is the posterior probability of a mass-extinction event in an interval needed to produce Bayes factor support of criticalBayesFactors[i].</p>
</td></tr>
<tr><td><code>criticalBayesFactors</code></td>
<td>
<p>The critical Bayes factor values used for the Bayes factor tests (default 2 ln BF = {2,6,10}).</p>
</td></tr>
<tr><td><code>tree</code></td>
<td>
<p>The tree analyzed with CoMET (just in case).</p>
</td></tr>
<tr><td><code>intervals</code></td>
<td>
<p>The discrete intervals used to compute the interval-specific parameters.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael R. May</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data, compute the sampling fraction rho
data(conifers)
totalConiferSpecies &lt;- 630
sampledConiferSpecies &lt;- conifers$Nnode+1
rho &lt;- sampledConiferSpecies / totalConiferSpecies

# Run a tess analysis
tess.analysis(tree = conifers,
              initialSpeciationRate=c(1.0),
              initialExtinctionRate=c(0.5),
              empiricalHyperPriors = FALSE,
              numExpectedRateChanges = 2,
              numExpectedMassExtinctions = 2,
              samplingProbability = rho,
              MAX_ITERATIONS = 200,
              BURNIN=100)

# Process the output
coniferOutput &lt;- tess.process.output(dir=getwd(),
                                     numExpectedRateChanges=2,
                                     numExpectedMassExtinctions=2)

# Plot the output
tess.plot.output(coniferOutput)


</code></pre>

<hr>
<h2 id='tess.sim.age'>tess.sim.age: Simulate a reconstructed tree for a given age under a global, time-dependent birth-death process.</h2><span id='topic+tess.sim.age'></span>

<h3>Description</h3>

<p>tess.sim.age simulates a reconstructed phylogenetic tree under a global, time-dependent birth-death process conditioned on the age of the tree. The rates may be any positive function of time or a constant. The process starts at time 0 and goes forward in time, hence the rates and events should be interpreted in the time after the origin. Additionally, mass-extinction event can be provided and a uniform taxon sampling probability. It is possible to start either with the origin (1 species) or with the most recent common ancestor (2 species).</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.sim.age(n, age, lambda, mu, massExtinctionTimes = c(), 
   massExtinctionSurvivalProbabilities = c(), samplingProbability = 1, 
   samplingStrategy = "uniform", maxTaxa = Inf, MRCA = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.sim.age_+3A_n">n</code></td>
<td>
<p>Number of simulations.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_age">age</code></td>
<td>
<p>The age of the tree, i.e. the time to simulate.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_lambda">lambda</code></td>
<td>
<p>The speciation rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_mu">mu</code></td>
<td>
<p>The extinction rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_massextinctiontimes">massExtinctionTimes</code></td>
<td>
<p>The set of mass-extinction times after the start of the process.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_massextinctionsurvivalprobabilities">massExtinctionSurvivalProbabilities</code></td>
<td>
<p>The set of survival probabilities for each speciation event. The set must have the same length as the set of mass-extinction times.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_samplingprobability">samplingProbability</code></td>
<td>
<p>The probability for a species to be included in the sample.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_samplingstrategy">samplingStrategy</code></td>
<td>
<p>The strategy how samples were obtained. Options are: uniform|diversified.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_maxtaxa">maxTaxa</code></td>
<td>
<p>The maximum number of possible taxa. If by chance a higher number is simulated, than simply ntaxa=maxTaxa. This is useful when too large trees should be simulated because this takes too much time and memory.</p>
</td></tr>
<tr><td><code id="tess.sim.age_+3A_mrca">MRCA</code></td>
<td>
<p>Does the process start with the most recent common ancestor?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a set of trees in 'phylo' format.
</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374 </p>


<h3>Examples</h3>

<pre><code class='language-R'>
l &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (1) } else { return (2) } })
e &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (0.95) } else { return (0.5) } })

tess.sim.age(n=1,age=1,l,e,MRCA=TRUE)

# simulation under constant rates
tess.sim.age(n=1,age=1,2.0,1.0,MRCA=TRUE)

</code></pre>

<hr>
<h2 id='tess.sim.taxa'>tess.sim.taxa.taxa: Simulate a reconstructed tree for a given number of taxa under a global, time-dependent birth-death process.</h2><span id='topic+tess.sim.taxa'></span>

<h3>Description</h3>

<p>tess.sim.taxa simulates a reconstructed phylogenetic tree under a global, time-dependent birth-death process conditioned on the number of taxa sampled. The rates may be any positive function of time or a constant. The process starts at time 0 and goes forward in time, hence the rates and events should be interpreted in the time after the origin. Additionally, mass-extinction event can be provided and a uniform taxon sampling probability. It is possible to start either with the origin (1 species) or with the most recent common ancestor (2 species).</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.sim.taxa(n, nTaxa, max, lambda, mu, massExtinctionTimes = c(), 
   massExtinctionSurvivalProbabilities = c(), samplingProbability = 1, 
   samplingStrategy = "uniform", SURVIVAL = TRUE, MRCA = TRUE, t_crit = c())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.sim.taxa_+3A_n">n</code></td>
<td>
<p>Number of simulations.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_ntaxa">nTaxa</code></td>
<td>
<p>Number of species sampled.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_max">max</code></td>
<td>
<p>Maximum time/height of the tree.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_lambda">lambda</code></td>
<td>
<p>The speciation rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_mu">mu</code></td>
<td>
<p>The extinction rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_massextinctiontimes">massExtinctionTimes</code></td>
<td>
<p>The set of mass-extinction times after the start of the process.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_massextinctionsurvivalprobabilities">massExtinctionSurvivalProbabilities</code></td>
<td>
<p>The set of survival probabilities for each speciation event. The set must have the same length as the set of mass-extinction times.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_samplingprobability">samplingProbability</code></td>
<td>
<p>The probability for a species to be included in the sample.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_samplingstrategy">samplingStrategy</code></td>
<td>
<p>The strategy how samples were obtained. Options are: uniform|diversified.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_survival">SURVIVAL</code></td>
<td>
<p>Do you want to condition on survival of the process?</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_mrca">MRCA</code></td>
<td>
<p>Does the process start with the most recent common ancestor?</p>
</td></tr>
<tr><td><code id="tess.sim.taxa_+3A_t_crit">t_crit</code></td>
<td>
<p>The critical time points when a jump in the rate function occurs. Only a help for the numerical integration routine.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a tree in 'phylo' format.
</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374 </p>


<h3>Examples</h3>

<pre><code class='language-R'>
l &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (1) } else { return (2) } })
e &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (0.95) } else { return (0.5) } })

tess.sim.taxa(n=1,nTaxa=10,max=10,l,e,MRCA=TRUE)

# simulation under constant rates
tess.sim.taxa(n=1,nTaxa=10,max=10,2.0,1.0,MRCA=TRUE)

</code></pre>

<hr>
<h2 id='tess.sim.taxa.age'>tess.sim.taxa.taxa.age: Simulate a reconstructed tree for a given age and number of taxa under a global, time-dependent birth-death process.</h2><span id='topic+tess.sim.taxa.age'></span>

<h3>Description</h3>

<p>tess.sim.taxa.age simulates a reconstructed phylogenetic tree under a global, time-dependent birth-death process conditioned on the age of the tree and number of taxa sampled. The rates may be any positive function of time or a constant. The process starts at time 0 and goes forward in time, hence the rates and events should be interpreted in the time after the origin. Additionally, mass-extinction event can be provided and a uniform taxon sampling probability. It is possible to start either with the origin (1 species) or with the most recent common ancestor (2 species).</p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.sim.taxa.age(n, nTaxa, age, lambda, mu, massExtinctionTimes = c(), 
   massExtinctionSurvivalProbabilities = c(), samplingProbability = 1, 
   samplingStrategy = "uniform", MRCA = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.sim.taxa.age_+3A_n">n</code></td>
<td>
<p>Number of simulations.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_ntaxa">nTaxa</code></td>
<td>
<p>Number of species sampled.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_age">age</code></td>
<td>
<p>The age of the tree, i.e. the time to simulate.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_lambda">lambda</code></td>
<td>
<p>The speciation rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_mu">mu</code></td>
<td>
<p>The extinction rate function or constant.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_massextinctiontimes">massExtinctionTimes</code></td>
<td>
<p>The set of mass-extinction times after the start of the process.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_massextinctionsurvivalprobabilities">massExtinctionSurvivalProbabilities</code></td>
<td>
<p>The set of survival probabilities for each speciation event. The set must have the same length as the set of mass-extinction times.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_samplingprobability">samplingProbability</code></td>
<td>
<p>The probability for a species to be included in the sample.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_samplingstrategy">samplingStrategy</code></td>
<td>
<p>The strategy how samples were obtained. Options are: uniform|diversified.</p>
</td></tr>
<tr><td><code id="tess.sim.taxa.age_+3A_mrca">MRCA</code></td>
<td>
<p>Does the process start with the most recent common ancestor?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a tree in 'phylo' format.
</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>S. Hoehna: Fast simulation of reconstructed phylogenies under global, time-dependent birth-death processes. 2013, Bioinformatics, 29:1367-1374 </p>


<h3>Examples</h3>

<pre><code class='language-R'>
l &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (1) } else { return (2) } })
e &lt;- Vectorize(function(x) { if (x &gt; 0.5 || x &lt; 0.3) { return (0.95) } else { return (0.5) } })

tess.sim.taxa.age(n=1,l,e,nTaxa=10,age=1,MRCA=TRUE)

# simulation under constant rates
tess.sim.taxa.age(n=1,2.0,1.0,nTaxa=10,age=1,MRCA=TRUE)

</code></pre>

<hr>
<h2 id='tess.steppingStoneSampling'>tess.steppingStoneSampling: Marginal likelihood estimation via Stepping-Stone-Sampling.</h2><span id='topic+tess.steppingStoneSampling'></span>

<h3>Description</h3>

<p>tess.steppingStoneSampling uses a power posterior series and stepping-stone-sampling to estimate the marginal likelihood of a model. </p>


<h3>Usage</h3>

<pre><code class='language-R'>tess.steppingStoneSampling(likelihoodFunction,priors,parameters,logTransforms,
                           iterations,burnin=round(iterations/3),K=50)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tess.steppingStoneSampling_+3A_likelihoodfunction">likelihoodFunction</code></td>
<td>
<p>The log-likelihood function which will be called internally by likelihoodFunction(parameters).</p>
</td></tr>
<tr><td><code id="tess.steppingStoneSampling_+3A_priors">priors</code></td>
<td>
<p>A list of functions of the log-prior-densities of each parameter.</p>
</td></tr>
<tr><td><code id="tess.steppingStoneSampling_+3A_parameters">parameters</code></td>
<td>
<p>The initial parameter value list.</p>
</td></tr>
<tr><td><code id="tess.steppingStoneSampling_+3A_logtransforms">logTransforms</code></td>
<td>
<p>A vector of booleans telling if log-transform for the parameters should be used (e.g. for rates).</p>
</td></tr>
<tr><td><code id="tess.steppingStoneSampling_+3A_iterations">iterations</code></td>
<td>
<p>The number of iterations for the MCMC.</p>
</td></tr>
<tr><td><code id="tess.steppingStoneSampling_+3A_burnin">burnin</code></td>
<td>
<p>The number of iterations to burn before starting the MCMC.</p>
</td></tr>
<tr><td><code id="tess.steppingStoneSampling_+3A_k">K</code></td>
<td>
<p>The number of stepping stones.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the posterior samples for the parameters.</p>


<h3>Author(s)</h3>

<p>Sebastian Hoehna</p>


<h3>References</h3>

<p>Xie et al., 2011: Improving marginal likelihood estimation for Bayesian phylogenetic model selection
Baele et al., 2012: Improving the accuracy of demographic and molecular clock model comparison while accommodating phylogenetic uncertainty
Baele et al., 2013: Accurate Model Selection of Relaxed Molecular Clocks in Bayesian Phylogenetics </p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(cettiidae)
times &lt;- as.numeric( branching.times(cettiidae) )

likelihood &lt;- function(params) {
  # We use the parameters as diversification rate and turnover rate.
  # Thus we need to transform first
  b &lt;- params[1] + params[2]
  d &lt;- params[2]
  
  lnl &lt;- tess.likelihood(times,b,d,samplingProbability=1.0,log=TRUE)
  return (lnl)
}

prior_diversification &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
prior_turnover &lt;- function(x) { dexp(x,rate=0.1,log=TRUE) }
priors &lt;- c(prior_diversification,prior_turnover)

# Note, the number of iterations, the burnin
# and the number of stepping stones is too small here
# and should be adapted for real analyses
marginalLikelihood &lt;- tess.steppingStoneSampling( likelihood,
						  priors,
						  runif(2,0,1),
						  c(TRUE,TRUE),
						  10,
						  10,
						  K=4)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
