<!DOCTYPE html><html><head><title>Help for package TFactSR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TFactSR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AtCatalog'><p>A list of TF-target genes (TG) in Arabidopsis thaliana.</p></a></li>
<li><a href='#calculateEnrichmentTest'><p>performs enrichment analysis</p></a></li>
<li><a href='#calculateEvalue'><p>calculates E-value</p></a></li>
<li><a href='#calculateFDRBH'><p>calculates FDR by Benjamini and Hochberg method</p></a></li>
<li><a href='#calculateQvalue'><p>calculates Q-value</p></a></li>
<li><a href='#calculateRC'><p>calculates Random Control (RC)</p></a></li>
<li><a href='#calculateTFactS'><p>calculates TFactS</p></a></li>
<li><a href='#catalog'><p>A list of TF-target genes (TG) in TFactS.</p></a></li>
<li><a href='#DEGs'><p>An example list of DEGs from DEMO data in TFactS.</p></a></li>
<li><a href='#DEGs39'><p>An example list of 39 DEGs from DEMO data in TFactS.</p></a></li>
<li><a href='#example.df'><p>An example result of enrichment analysis</p></a></li>
<li><a href='#example.list'><p>An example list of result of enrichment analysis</p></a></li>
<li><a href='#extractTFTG'><p>extracts transcription factor (TF) and target gene (TG) information</p></a></li>
<li><a href='#FASTcalculateRC'><p>calculates Random Control (RC) fastly?</p></a></li>
<li><a href='#formatET'><p>formats the result of enrichment test (ET)</p></a></li>
<li><a href='#formatRC'><p>formats the result of Random Control (RC)</p></a></li>
<li><a href='#GenesUp_SH1H'><p>An example list of 74 DEGs from AtGenExpress stress dataset.</p></a></li>
<li><a href='#TFactSR'><p>TFactSR: A package for identifying critical transcription factors.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Enrichment Approach to Predict Which Transcription Factors are
Regulated</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.0</td>
</tr>
<tr>
<td>Description:</td>
<td>R implementation of 'TFactS' to predict which are the 
    transcription factors (TFs), regulated in a biological condition based on
    lists of differentially expressed genes (DEGs) obtained from transcriptome
    experiments. This package is based on the 'TFactS' concept by 
    Essaghir et al. (2010) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkq149">doi:10.1093/nar/gkq149</a>&gt; and expands
    it. It allows users to perform 'TFactS'-like enrichment approach. The
    package can import and use the original catalogue file from the 'TFactS'
    as well as users' defined catalogues of interest that are not supported 
    by 'TFactS' (e.g., Arabidopsis).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://afukushima.github.io/TFactSR/">https://afukushima.github.io/TFactSR/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>qvalue, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocStyle, knitr, rmarkdown, spelling, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Network, Software, DifferentialExpression, GeneTarget,
GeneExpression, Microarray, RNASeq, Transcription,
NetworkEnrichment</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-31 00:35:51 UTC; fuku</td>
</tr>
<tr>
<td>Author:</td>
<td>Atsushi Fukushima <a href="https://orcid.org/0000-0001-9015-1694"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Atsushi Fukushima &lt;afukushima@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-31 14:00:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='AtCatalog'>A list of TF-target genes (TG) in Arabidopsis thaliana.</h2><span id='topic+AtCatalog'></span>

<h3>Description</h3>

<p>A catalogue of annotated target genes in Arabidopsis thaliana
</p>


<h3>Details</h3>

<p>The original regulatory network datasets were downloaded from AtRegNet
- AGRIS Arabidopsis
(http://arabidopsis.med.ohio-state.edu/moreNetwork.html)
and ATRM (http://atrm.cbi.pku.edu.cn/), respectively. Then we integrated
them based on AGI code.
</p>
<p>The catalogues version is 0.0.1.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Yilmaz A et al. NAR 39:D1118-1122 (2011)
</p>

<hr>
<h2 id='calculateEnrichmentTest'>performs enrichment analysis</h2><span id='topic+calculateEnrichmentTest'></span>

<h3>Description</h3>

<p>This function performs enrichment test (ET) based on Fisher's exact test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateEnrichmentTest(
  DEGs,
  catalog,
  TFs,
  TF.col = "TF..OFFICIAL_TF_CODING_GENE_NAME."
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateEnrichmentTest_+3A_degs">DEGs</code></td>
<td>
<p>a character vector of DEGs (differentially expressed genes)</p>
</td></tr>
<tr><td><code id="calculateEnrichmentTest_+3A_catalog">catalog</code></td>
<td>
<p>a data frame of TFactS catalog (ver. 2)</p>
</td></tr>
<tr><td><code id="calculateEnrichmentTest_+3A_tfs">TFs</code></td>
<td>
<p>a character vector of transcription factor</p>
</td></tr>
<tr><td><code id="calculateEnrichmentTest_+3A_tf.col">TF.col</code></td>
<td>
<p>the name of the column that contains the TF names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
Definition:
m is the number of target genes annotated for the TF
under consideration
n is the number of query genes
N is the number of regulations in the catalog
k is the number of query genes that are annotated as
regulated by TF (i.e., the intersection between the
query and the TF signature)
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DEGs)
data(catalog)

tftg &lt;- extractTFTG(DEGs, catalog)
TFs &lt;- tftg$TFs

res &lt;- calculateEnrichmentTest(DEGs, catalog, TFs)

</code></pre>

<hr>
<h2 id='calculateEvalue'>calculates E-value</h2><span id='topic+calculateEvalue'></span>

<h3>Description</h3>

<p>This function calculates E-value based on .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateEvalue(df, TFs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateEvalue_+3A_df">df</code></td>
<td>
<p>a data frame containng p-values</p>
</td></tr>
<tr><td><code id="calculateEvalue_+3A_tfs">TFs</code></td>
<td>
<p>a character vector of transcription factor</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DEGs)
data(catalog)

tftg &lt;- extractTFTG(DEGs, catalog)
TFs &lt;- tftg$TFs

p.value &lt;- runif(10)/(1:10)
df &lt;- data.frame(p.value = p.value)
res &lt;- calculateEvalue(df, TFs)

</code></pre>

<hr>
<h2 id='calculateFDRBH'>calculates FDR by Benjamini and Hochberg method</h2><span id='topic+calculateFDRBH'></span>

<h3>Description</h3>

<p>This function calculates FDR based on BH.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateFDRBH(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateFDRBH_+3A_df">df</code></td>
<td>
<p>a data frame containng p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Benjamini Y and Hochberg Y, J Roy Stat Soc B 57: 289?300 (1995)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p.value &lt;- runif(10)/(1:10)
df &lt;- data.frame(p.value = p.value)
res &lt;- calculateFDRBH(df)

</code></pre>

<hr>
<h2 id='calculateQvalue'>calculates Q-value</h2><span id='topic+calculateQvalue'></span>

<h3>Description</h3>

<p>This function calculates Q-value based on Storey.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateQvalue(df, lambda = seq(0.05, 0.5, 0.01))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateQvalue_+3A_df">df</code></td>
<td>
<p>a data frame containng p-values</p>
</td></tr>
<tr><td><code id="calculateQvalue_+3A_lambda">lambda</code></td>
<td>
<p>a vector of the lambda values utilized to obtain pi0.lambda</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Storey JD, The Annals of Statistics 31:2013-2035 (2003)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example.df)
p.value &lt;- example.df$p.value
df &lt;- data.frame(p.value = p.value)
res &lt;- calculateQvalue(df)

</code></pre>

<hr>
<h2 id='calculateRC'>calculates Random Control (RC)</h2><span id='topic+calculateRC'></span>

<h3>Description</h3>

<p>This function calculates Random Control (RC)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateRC(
  df,
  DEGs,
  catalog,
  TFs,
  all.targets,
  TF.col = "TF..OFFICIAL_TF_CODING_GENE_NAME.",
  lambda = 0.05,
  nRep = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateRC_+3A_df">df</code></td>
<td>
<p>a data frame containng p-values</p>
</td></tr>
<tr><td><code id="calculateRC_+3A_degs">DEGs</code></td>
<td>
<p>a character vector of DEGs (differentially expressed genes)</p>
</td></tr>
<tr><td><code id="calculateRC_+3A_catalog">catalog</code></td>
<td>
<p>a data frame of TFactS catalog (ver. 2)</p>
</td></tr>
<tr><td><code id="calculateRC_+3A_tfs">TFs</code></td>
<td>
<p>a character vector of transcription factor</p>
</td></tr>
<tr><td><code id="calculateRC_+3A_all.targets">all.targets</code></td>
<td>
<p>a character vector of all target genes</p>
</td></tr>
<tr><td><code id="calculateRC_+3A_tf.col">TF.col</code></td>
<td>
<p>the name of the column that contains the TF names</p>
</td></tr>
<tr><td><code id="calculateRC_+3A_lambda">lambda</code></td>
<td>
<p>a user-specified threshold of E-value (default: 0.05)</p>
</td></tr>
<tr><td><code id="calculateRC_+3A_nrep">nRep</code></td>
<td>
<p>number of random selections (default: 100)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
Definition:
m is the number of target genes annotated for the TF
under consideration
n is the number of query genes
N is the number of regulations in the catalog
k is the number of query genes that are annotated as
regulated by TF (i.e., the intersection between the
query and the TF signature)
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example.df)
data(catalog)
data(DEGs)

tftg &lt;- extractTFTG(DEGs, catalog)
TFs &lt;- tftg$TFs
all.targets &lt;- tftg$all.targets

res &lt;- calculateRC(example.df, DEGs, catalog, TFs, all.targets)

</code></pre>

<hr>
<h2 id='calculateTFactS'>calculates TFactS</h2><span id='topic+calculateTFactS'></span>

<h3>Description</h3>

<p>This function calculates TFactS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateTFactS(
  DEGs,
  catalog,
  TFs,
  all.targets,
  Q.value = FALSE,
  lambda1 = seq(0.05, 0.5, 0.01),
  lambda2 = 0.05,
  nRep = 100,
  TF.col = "TF..OFFICIAL_TF_CODING_GENE_NAME."
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateTFactS_+3A_degs">DEGs</code></td>
<td>
<p>a character vector of DEGs (differentially expressed genes)</p>
</td></tr>
<tr><td><code id="calculateTFactS_+3A_catalog">catalog</code></td>
<td>
<p>a data frame of TFactS catalog (ver. 2)</p>
</td></tr>
<tr><td><code id="calculateTFactS_+3A_tfs">TFs</code></td>
<td>
<p>a character vector of transcription factor</p>
</td></tr>
<tr><td><code id="calculateTFactS_+3A_all.targets">all.targets</code></td>
<td>
<p>a character vector of all target genes</p>
</td></tr>
<tr><td><code id="calculateTFactS_+3A_q.value">Q.value</code></td>
<td>
<p>logical. If it is TRUE, Q.value by Storey method.</p>
</td></tr>
<tr><td><code id="calculateTFactS_+3A_lambda1">lambda1</code></td>
<td>
<p>a vector of the lambda values utilized to obtain pi0.lambda</p>
</td></tr>
<tr><td><code id="calculateTFactS_+3A_lambda2">lambda2</code></td>
<td>
<p>a user-specified threshold of E-value (default: 0.05)</p>
</td></tr>
<tr><td><code id="calculateTFactS_+3A_nrep">nRep</code></td>
<td>
<p>number of random selections (default: 100)</p>
</td></tr>
<tr><td><code id="calculateTFactS_+3A_tf.col">TF.col</code></td>
<td>
<p>the name of the column that contains the TF names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DEGs)
data(catalog)

tftg &lt;- extractTFTG(DEGs, catalog)
TFs &lt;- tftg$TFs
all.targets &lt;- tftg$all.targets

res &lt;- calculateTFactS(DEGs, catalog, TFs, all.targets)

</code></pre>

<hr>
<h2 id='catalog'>A list of TF-target genes (TG) in TFactS.</h2><span id='topic+catalog'></span>

<h3>Description</h3>

<p>A catalogue of annotated target genes based on TFactS.
</p>


<h3>Details</h3>

<p>This was downloaded from TFactS (http://www.tfacts.org/).
</p>
<p>The catalogues version is 2
(http://www.tfacts.org/TFactS-new/TFactS-v2/tfacts/data/Catalogues.xls).
</p>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>

<hr>
<h2 id='DEGs'>An example list of DEGs from DEMO data in TFactS.</h2><span id='topic+DEGs'></span>

<h3>Description</h3>

<p>An example list of differentially expressed genes (DEGs).
</p>


<h3>Details</h3>

<p>This was from DEMO data in TFactS (http://www.tfacts.org/).
The list corresponds to 18 up-regulated genes.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>

<hr>
<h2 id='DEGs39'>An example list of 39 DEGs from DEMO data in TFactS.</h2><span id='topic+DEGs39'></span>

<h3>Description</h3>

<p>A list of 39 differentially expressed genes (DEGs).
</p>


<h3>Details</h3>

<p>This was from DEMO data in TFactS (http://www.tfacts.org/). The list
corresponds to 18 up-regulated genes and 21 down-regulated genes.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>

<hr>
<h2 id='example.df'>An example result of enrichment analysis</h2><span id='topic+example.df'></span>

<h3>Description</h3>

<p>An example result of enrichment analysis
</p>


<h3>Details</h3>

<p>The size of data frame is 345 TFs and 6 columns.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>

<hr>
<h2 id='example.list'>An example list of result of enrichment analysis</h2><span id='topic+example.list'></span>

<h3>Description</h3>

<p>An example list of result of enrichment analysis
</p>


<h3>Details</h3>

<p>The length of the list is 345 (TFs).
</p>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>

<hr>
<h2 id='extractTFTG'>extracts transcription factor (TF) and target gene (TG) information</h2><span id='topic+extractTFTG'></span>

<h3>Description</h3>

<p>This function extracts information about transcription factor (TF) and
target gene (TG) with TFactS Catalogue (v2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractTFTG(
  DEGs,
  catalog,
  TF.col = "TF..OFFICIAL_TF_CODING_GENE_NAME.",
  TG.col = "Target.gene..OFFICIAL_GENE_NAME."
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractTFTG_+3A_degs">DEGs</code></td>
<td>
<p>a character vector of DEGs (differentially expressed genes)</p>
</td></tr>
<tr><td><code id="extractTFTG_+3A_catalog">catalog</code></td>
<td>
<p>a data frame of TFactS catalog (ver. 2)</p>
</td></tr>
<tr><td><code id="extractTFTG_+3A_tf.col">TF.col</code></td>
<td>
<p>the name of the column that contains the TF names</p>
</td></tr>
<tr><td><code id="extractTFTG_+3A_tg.col">TG.col</code></td>
<td>
<p>the name of the column that contains the TG names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DEGs)
data(catalog)

res &lt;- extractTFTG(DEGs, catalog)
head(res$TFs)

</code></pre>

<hr>
<h2 id='FASTcalculateRC'>calculates Random Control (RC) fastly?</h2><span id='topic+FASTcalculateRC'></span>

<h3>Description</h3>

<p>This function calculates Random Control (RC)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FASTcalculateRC(
  df,
  DEGs,
  catalog,
  TFs,
  all.targets,
  TF.col = "TF..OFFICIAL_TF_CODING_GENE_NAME.",
  lambda = 0.05,
  nRep = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FASTcalculateRC_+3A_df">df</code></td>
<td>
<p>a data frame containng p-values</p>
</td></tr>
<tr><td><code id="FASTcalculateRC_+3A_degs">DEGs</code></td>
<td>
<p>a character vector of DEGs (differentially expressed genes)</p>
</td></tr>
<tr><td><code id="FASTcalculateRC_+3A_catalog">catalog</code></td>
<td>
<p>a data frame of TFactS catalog (ver. 2)</p>
</td></tr>
<tr><td><code id="FASTcalculateRC_+3A_tfs">TFs</code></td>
<td>
<p>a character vector of transcription factor</p>
</td></tr>
<tr><td><code id="FASTcalculateRC_+3A_all.targets">all.targets</code></td>
<td>
<p>a character vector of all target genes</p>
</td></tr>
<tr><td><code id="FASTcalculateRC_+3A_tf.col">TF.col</code></td>
<td>
<p>the name of the column that contains the TF names</p>
</td></tr>
<tr><td><code id="FASTcalculateRC_+3A_lambda">lambda</code></td>
<td>
<p>a user-specified threshold of E-value (default: 0.05)</p>
</td></tr>
<tr><td><code id="FASTcalculateRC_+3A_nrep">nRep</code></td>
<td>
<p>number of random selections (default: 100)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Essaghir A et al. Nucleic Acids Res. 2010 Jun;38(11):e120.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example.df)
data(catalog)
data(DEGs)

tftg &lt;- extractTFTG(DEGs, catalog)
TFs &lt;- tftg$TFs
all.targets &lt;- tftg$all.targets

res &lt;- FASTcalculateRC(example.df, DEGs, catalog, TFs, all.targets)

</code></pre>

<hr>
<h2 id='formatET'>formats the result of enrichment test (ET)</h2><span id='topic+formatET'></span>

<h3>Description</h3>

<p>This function formats the result of enrichment test (ET)
based on Fisher's exact test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatET(list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="formatET_+3A_list">list</code></td>
<td>
<p>a list of the result of enrichment test (ncol = 6)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example.list)
res &lt;- formatET(example.list)

</code></pre>

<hr>
<h2 id='formatRC'>formats the result of Random Control (RC)</h2><span id='topic+formatRC'></span>

<h3>Description</h3>

<p>This function formats the result of Random Control (RC)
with random simulation based on Fisher's exact test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatRC(df, list, nRep)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="formatRC_+3A_df">df</code></td>
<td>
<p>a data frame of ET including E-values, FDR-BH, and Q-values</p>
</td></tr>
<tr><td><code id="formatRC_+3A_list">list</code></td>
<td>
<p>a list of the result of RC (ncol = 2)</p>
</td></tr>
<tr><td><code id="formatRC_+3A_nrep">nRep</code></td>
<td>
<p>the number of random selections (negative control)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example.df)
data(example.list)
nRep &lt;- 100
res &lt;- formatRC(example.df, example.list, nRep)

</code></pre>

<hr>
<h2 id='GenesUp_SH1H'>An example list of 74 DEGs from AtGenExpress stress dataset.</h2><span id='topic+GenesUp_SH1H'></span>

<h3>Description</h3>

<p>A list of 74 differentially expressed genes (DEGs) from AtGenExpress
</p>


<h3>Details</h3>

<p>This was from DEMO data in the Supplemental data (Table S4. Genes
regulated by UV-B light stress in roots and shoots; filename:
TPJ_3052_sm_TabS4.xls) of the paper by Killian et al. TPJ (2008).
The list corresponds to 74 genes up-regulated in roots and shoots 1 hour
after onset of UV-B stress treatment.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Atsushi Fukushima
</p>


<h3>References</h3>

<p>Killian J et al. Plant J. 2007 Apr;50(2):347-363.
</p>

<hr>
<h2 id='TFactSR'>TFactSR: A package for identifying critical transcription factors.</h2><span id='topic+TFactSR'></span>

<h3>Description</h3>

<p>TFactS is to predict which are the transcription factors (TFs), regulated
in a biological condition based on lists of differentially expressed genes
(DEGs) obtained from transcriptome experiments. This package is based on
the TFactS concept and expands it. It allows users to performe
TFactS-like enrichment approach. The package can import and use the
original catalogue file from the TFactS website (http://www.tfacts.org/)
as well as users' defined catalogues of interest that are not supported
by TFactS (e.g., Arabidopsis).
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
