<!DOCTYPE html><html><head><title>Help for package thgenetics</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {thgenetics}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#rareGeneTest'><p>Rare Variant Tests</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.4-2</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-04-17</td>
</tr>
<tr>
<td>Title:</td>
<td>Genetic Rare Variants Tests</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Hoffmann &lt;tjhoffm@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Hoffmann &lt;tjhoffm@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A step-up test for genetic rare variants in a gene or in a pathway. The method determines an optimal grouping of rare variants analytically. The method has been described in Hoffmann TJ, Marini NJ, and Witte JS (2010) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0013584">doi:10.1371/journal.pone.0013584</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://sites.google.com/site/thomashoffmannproject/">http://sites.google.com/site/thomashoffmannproject/</a>
<a href="https://doi.org/10.1371/journal.pone.0013584">https://doi.org/10.1371/journal.pone.0013584</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-04-17 22:46:00 UTC; tom</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-04-18 10:25:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='rareGeneTest'>Rare Variant Tests</h2><span id='topic+rareGeneTest'></span><span id='topic+rarePathwayTest'></span>

<h3>Description</h3>

<p>Runs tests for rare variants by running the &quot;step-up&quot; approach to choose the best set of variants (possibly signing them), and correcting by permutation. <code>rareGeneTest</code> tests a single gene, and also allows for other aggregation models described in Hoffmann et al.. <code>rarePathwayTest</code> provides an extension of this method, that does a step-up approach for inclusions of genes (for computional time). For larger genes, it may also be advantageous to run <code>rarePathwayTest</code> which will automatically chop large genes into subpieces and run step-up on those pieces.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rareGeneTest(genotype, phenotype,
  use_sign=TRUE, use_weight=TRUE, nperm=1000,
  binary=all(phenotype==0 | phenotype==1, na.rm=TRUE),
  strategy="step", thresh=1)
rarePathwayTest(genotype, genotype_gene, phenotype,
  use_sign=TRUE, use_weight=TRUE, nperm=1000,
  binary=all(phenotype==0 | phenotype==1, na.rm=TRUE),
  CUT=15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rareGeneTest_+3A_genotype">genotype</code></td>
<td>
<p>Matrix of individuals along the rows and genotypes along the columns. Each entry in the matrix should be coded <code>NA</code> for missing; and <code>0</code>, <code>1</code>, or <code>2</code> for the number of alleles a person has (must be bi-allelic).</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_phenotype">phenotype</code></td>
<td>
<p>Vector of same length as the number of rows of the matrix. Either continuous, or dichotomous (case=<code>1</code>, control=<code>0</code>.</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_use_sign">use_sign</code></td>
<td>
<p>Whether to use a sign in the model for w, or just to weight all variants the same. Uses the sign of the estimated covariance between each marker and the trait to determine the sign of the marker.</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_use_weight">use_weight</code></td>
<td>
<p>Whether to use a weight in the model for w (inverse variance of allele frequency in everyone (continuous) or just the controls (dichotomous)).</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_nperm">nperm</code></td>
<td>
<p>Number of permutations to run.</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_binary">binary</code></td>
<td>
<p>Is the phenotype dichotomous, or continuous? (TRUE/FALSE)</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_strategy">strategy</code></td>
<td>
<p>&lsquo;step&rsquo; uses the step-up routine (default, recommended), &lsquo;afreq&rsquo; tests all allele frequencies, and &lsquo;thresh&rsquo; uses a fixed threshold.</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_thresh">thresh</code></td>
<td>
<p>Allele frequency for inclusion, used only when <code>strategy</code>=&lsquo;thresh&rsquo;.</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_genotype_gene">genotype_gene</code></td>
<td>
<p>For <code>rarePathwayTest</code>, this is a vector of length <code>ncol(genotype)</code>. This can be strings to indicate what gene each variant is in; e.g. 
<code>c("brca1",</code>
<code>"brca1",</code>
<code>"brca1",</code>
<code>"brca2",</code>
<code>"brca2")</code>
would indicate that the first three variants were in the first gene, and the last two variants are in the second gene. It can also be numeric.</p>
</td></tr>
<tr><td><code id="rareGeneTest_+3A_cut">CUT</code></td>
<td>
<p>For <code>rarePathwayTest</code>, genes bigger than <code>CUT</code> will be chopped into subpieces of size smaller than <code>CUT</code> (for computational speed).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The methods here are as described in the Hoffmann et al. (PLoS ONE, to appear) paper. These methods are based on the idea of trying multiple models for rare variants, since prior information is generally not very accurate. The p-value is corrected for multiple comparisons by permutation.
</p>


<h3>References</h3>

<p>Hoffmann, TJ, Marini, NJ, and Witte, JS. Comprehensive approach to analyzing rare variants. PLoS ONE, 5(11): e13584. https://doi.org/10.1371/journal/pone.0013584.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## This inefficient (for clarity) code will simulate
##  a dataset in the correct format, and then run
##  the different approaches described here.

## Constants for data generation
SEED &lt;- 2
NCASE &lt;- NCONT &lt;- 500
NGENE &lt;- 6
FUNCTIONAL &lt;- 1:(NGENE/2)  ## Half are functional
BETA0 &lt;- -4
BETAG &lt;- log(2)

set.seed(SEED) ## Reproducible results

nonfunctional &lt;- setdiff(1:NGENE, FUNCTIONAL)
afreq &lt;- runif(NGENE, 0.001, 0.01)

expit &lt;- function(x)
  return(exp(x) / (1 + exp(x)))

gcase &lt;- matrix(0, nrow=NCASE, ncol=NGENE)
for(indiv in 1:NCASE){
  affected &lt;- FALSE
  while(!affected){ ## while not affected
    gcase[indiv, ] &lt;- rbinom(NGENE, 2, afreq)  ## draw up genotype
    affected &lt;- (expit(BETA0 + 
      BETAG*sum(gcase[indiv, FUNCTIONAL])) &gt; runif(1))
  }
}
cat("\n")

gcont &lt;- matrix(0, nrow=NCONT, ncol=NGENE)
for(indiv in 1:NCONT){
  unaffected &lt;- FALSE
  while(!unaffected){ ## while not unaffected
    gcont[indiv, ] &lt;- rbinom(NGENE, 2, afreq) ## draw up genotype
    unaffected &lt;- (1-expit(BETA0 +
      BETAG*sum(gcont[indiv, FUNCTIONAL])) &gt; runif(1))
  }
}
cat("\n")

cat("# Rare functional variants cases =", 
  sum(gcase[,FUNCTIONAL]), "\n")
cat("# Rare functional variants controls =",
  sum(gcont[,FUNCTIONAL]), "\n")
cat("# Rare non-functional variants cases =",
  sum(gcase[,nonfunctional]), "\n")
cat("# Rare non-functional variants controls =",
  sum(gcont[,nonfunctional]), "\n")

case &lt;- c(rep(1,NCASE), rep(0,NCONT))
genotype &lt;- rbind(gcase, gcont)

cat("P-value of the test:\n")
rareGeneTest(genotype, case)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
