<!DOCTYPE html><html lang="en"><head><title>Help for package tidyheatmaps</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {tidyheatmaps}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data_exprs'><p>Expression data from RNA-Seq study</p></a></li>
<li><a href='#tidyheatmap'><p>Create heatmap from tidy data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Heatmaps from Tidy Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>The goal of 'tidyheatmaps' is to simplify the generation of publication-ready heatmaps from tidy data. By offering an interface to the powerful 'pheatmap' package, it allows for the effortless creation of intricate heatmaps with minimal code.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, pheatmap, rlang, grDevices, tidyr, tibble, RColorBrewer</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.1.0), knitr, rmarkdown</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jbengler/tidyheatmaps">https://github.com/jbengler/tidyheatmaps</a>,
<a href="https://jbengler.github.io/tidyheatmaps/">https://jbengler.github.io/tidyheatmaps/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jbengler/tidyheatmaps/issues">https://github.com/jbengler/tidyheatmaps/issues</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-28 12:54:41 UTC; janbroderengler</td>
</tr>
<tr>
<td>Author:</td>
<td>Jan Broder Engler <a href="https://orcid.org/0000-0002-3169-2076"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jan Broder Engler &lt;broder.engler@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-29 13:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='data_exprs'>Expression data from RNA-Seq study</h2><span id='topic+data_exprs'></span>

<h3>Description</h3>

<p>This data was taken form a RNA-Seq study investigating the regulation of genes
in response to central nervous system inflammation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_exprs
</code></pre>


<h3>Format</h3>

<p>A data frame with 800 rows and 9 variables:
</p>

<dl>
<dt>ensembl_gene_id</dt><dd><p>Ensembl gene id</p>
</dd>
<dt>external_gene_name</dt><dd><p>Gene symbol</p>
</dd>
<dt>sample</dt><dd><p>Sample name</p>
</dd>
<dt>expression</dt><dd><p>Normalized RNA-Seq expression value</p>
</dd>
<dt>group</dt><dd><p>Experimental group</p>
</dd>
<dt>sample_type</dt><dd><p>Sample type. Either <code>input</code> or <code>IP</code>.</p>
</dd>
<dt>condition</dt><dd><p>Condition of sampling. Either <code>healthy</code> or <code>EAE</code>. </p>
</dd>
<dt>is_immune_gene</dt><dd><p>Gene is annotated as immune cell gene. Either <code>yes</code> or <code>no</code>.</p>
</dd>
<dt>direction</dt><dd><p>Direction of regulation. Either <code>up</code> or <code>down</code>. </p>
</dd>
</dl>



<h3>Source</h3>

<p><code>data_exprs</code> represents just a small subset of the data aquired in the study.
</p>
<p>More details about the study can be found here
</p>

<ul>
<li><p> Nature Neuroscience, <a href="https://www.nature.com/articles/s41593-019-0385-4">Bassoon proteinopathy drives neurodegeneration in multiple sclerosis</a>
</p>
</li></ul>

<p>The complete raw data can be downloaded here
</p>

<ul>
<li><p> Gene Expression Omnibus, study accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE104899">GSE104899</a>
</p>
</li></ul>


<hr>
<h2 id='tidyheatmap'>Create heatmap from tidy data</h2><span id='topic+tidyheatmap'></span><span id='topic+tidy_heatmap'></span>

<h3>Description</h3>

<p>A tidyverse-style interface to the powerful heatmap package
<a href="https://github.com/raivokolde/pheatmap">pheatmap</a>.
It enables the convenient generation of complex heatmaps from tidy data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tidyheatmap(
  df,
  rows,
  columns,
  values,
  colors = NA,
  color_legend_n = 15,
  color_legend_min = NA,
  color_legend_max = NA,
  color_na = "#DDDDDD",
  annotation_row = NULL,
  annotation_col = NULL,
  gaps_row = NULL,
  gaps_col = NULL,
  show_selected_row_labels = NULL,
  show_selected_col_labels = NULL,
  filename = NA,
  scale = "none",
  fontsize = 7,
  cellwidth = NA,
  cellheight = NA,
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  border_color = NA,
  kmeans_k = NA,
  clustering_distance_rows = "euclidean",
  clustering_distance_cols = "euclidean",
  clustering_method = "complete",
  clustering_callback = function(x, ...) {
     return(x)
 },
  cutree_rows = NA,
  cutree_cols = NA,
  treeheight_row = ifelse((class(cluster_rows) == "hclust") || cluster_rows, 50, 0),
  treeheight_col = ifelse((class(cluster_cols) == "hclust") || cluster_cols, 50, 0),
  legend = TRUE,
  legend_breaks = NA,
  legend_labels = NA,
  annotation_colors = NA,
  annotation_legend = TRUE,
  annotation_names_row = TRUE,
  annotation_names_col = TRUE,
  drop_levels = TRUE,
  show_rownames = TRUE,
  show_colnames = TRUE,
  main = NA,
  fontsize_row = fontsize,
  fontsize_col = fontsize,
  angle_col = c("270", "0", "45", "90", "315"),
  display_numbers = FALSE,
  number_format = "%.2f",
  number_color = "grey30",
  fontsize_number = 0.8 * fontsize,
  width = NA,
  height = NA,
  silent = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tidyheatmap_+3A_df">df</code></td>
<td>
<p>A tidy dataframe in long format.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_rows">rows</code>, <code id="tidyheatmap_+3A_columns">columns</code></td>
<td>
<p>Column in the dataframe to use for heatmap <code>rows</code> and
<code>columns</code>.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_values">values</code></td>
<td>
<p>Column in the dataframe containing the values to be color coded
in the heatmap cells.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_colors">colors</code></td>
<td>
<p>Vector of colors used for the color legend.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_color_legend_n">color_legend_n</code></td>
<td>
<p>Number of colors in the color legend.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_color_legend_min">color_legend_min</code>, <code id="tidyheatmap_+3A_color_legend_max">color_legend_max</code></td>
<td>
<p>Min and max value of the color legend.
Values smaller then the <code>color_legend_min</code> will have the lowest color,
<code>values</code> bigger than the <code>color_legend_max</code> will get the highest color.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_color_na">color_na</code></td>
<td>
<p>Color to use for <code>NAs</code> in <code>values</code>.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_annotation_row">annotation_row</code>, <code id="tidyheatmap_+3A_annotation_col">annotation_col</code></td>
<td>
<p>Column(s) in the dataframe to use for
<code>row</code> and <code>column</code> annotation.
To use multiple columns for annotation combine then by <code>c(column1, column2)</code>.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_gaps_row">gaps_row</code>, <code id="tidyheatmap_+3A_gaps_col">gaps_col</code></td>
<td>
<p>Column in the dataframe to use for use for <code>row</code> and
<code>column</code> gaps.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_show_selected_row_labels">show_selected_row_labels</code>, <code id="tidyheatmap_+3A_show_selected_col_labels">show_selected_col_labels</code></td>
<td>
<p>Only display a
subset of selected labels for <code>rows</code> and <code>columns</code>.
Provide selected labels as <code>c("label1", "label2")</code>.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_filename">filename</code></td>
<td>
<p>file path where to save the picture. Filetype is decided by 
the extension in the path. Currently following formats are supported: png, pdf, tiff,
bmp, jpeg. Even if the plot does not fit into the plotting window, the file size is 
calculated so that the plot would fit there, unless specified otherwise.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_scale">scale</code></td>
<td>
<p>character indicating if the values should be centered and scaled in 
either the row direction or the column direction, or none. Corresponding values are 
<code>"row"</code>, <code>"column"</code> and <code>"none"</code></p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_fontsize">fontsize</code></td>
<td>
<p>base fontsize for the plot</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_cellwidth">cellwidth</code></td>
<td>
<p>individual cell width in points. If left as NA, then the values 
depend on the size of plotting window.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_cellheight">cellheight</code></td>
<td>
<p>individual cell height in points. If left as NA, 
then the values depend on the size of plotting window.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>boolean values determining if rows should be clustered or <code>hclust</code> object,</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>boolean values determining if columns should be clustered or <code>hclust</code> object.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_border_color">border_color</code></td>
<td>
<p>color of cell borders on heatmap, use NA if no border should be 
drawn.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_kmeans_k">kmeans_k</code></td>
<td>
<p>the number of kmeans clusters to make, if we want to aggregate the 
rows before drawing heatmap. If NA then the rows are not aggregated.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_clustering_distance_rows">clustering_distance_rows</code></td>
<td>
<p>distance measure used in clustering rows. Possible 
values are <code>"correlation"</code> for Pearson correlation and all the distances 
supported by <code><a href="stats.html#topic+dist">dist</a></code>, such as <code>"euclidean"</code>, etc. If the value is none 
of the above it is assumed that a distance matrix is provided.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_clustering_distance_cols">clustering_distance_cols</code></td>
<td>
<p>distance measure used in clustering columns. Possible 
values the same as for clustering_distance_rows.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_clustering_method">clustering_method</code></td>
<td>
<p>clustering method used. Accepts the same values as 
<code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_clustering_callback">clustering_callback</code></td>
<td>
<p>callback function to modify the clustering. Is 
called with two parameters: original <code>hclust</code> object and the matrix 
used for clustering. Must return a <code>hclust</code> object.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_cutree_rows">cutree_rows</code></td>
<td>
<p>number of clusters the rows are divided into, based on the
hierarchical clustering (using cutree), if rows are not clustered, the 
argument is ignored</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_cutree_cols">cutree_cols</code></td>
<td>
<p>similar to <code>cutree_rows</code>, but for columns</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_treeheight_row">treeheight_row</code></td>
<td>
<p>the height of a tree for rows, if these are clustered. 
Default value 50 points.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_treeheight_col">treeheight_col</code></td>
<td>
<p>the height of a tree for columns, if these are clustered. 
Default value 50 points.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_legend">legend</code></td>
<td>
<p>logical to determine if legend should be drawn or not.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_legend_breaks">legend_breaks</code></td>
<td>
<p>vector of breakpoints for the legend.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_legend_labels">legend_labels</code></td>
<td>
<p>vector of labels for the <code>legend_breaks</code>.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_annotation_colors">annotation_colors</code></td>
<td>
<p>list for specifying annotation_row and 
annotation_col track colors manually. It is  possible to define the colors 
for only some of the features. Check examples for  details.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_annotation_legend">annotation_legend</code></td>
<td>
<p>boolean value showing if the legend for annotation 
tracks should be drawn.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_annotation_names_row">annotation_names_row</code></td>
<td>
<p>boolean value showing if the names for row annotation 
tracks should be drawn.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_annotation_names_col">annotation_names_col</code></td>
<td>
<p>boolean value showing if the names for column annotation 
tracks should be drawn.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_drop_levels">drop_levels</code></td>
<td>
<p>logical to determine if unused levels are also shown in 
the legend</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_show_rownames">show_rownames</code></td>
<td>
<p>boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_show_colnames">show_colnames</code></td>
<td>
<p>boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_main">main</code></td>
<td>
<p>the title of the plot</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_fontsize_row">fontsize_row</code></td>
<td>
<p>fontsize for rownames (Default: fontsize)</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_fontsize_col">fontsize_col</code></td>
<td>
<p>fontsize for colnames (Default: fontsize)</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_angle_col">angle_col</code></td>
<td>
<p>angle of the column labels, right now one can choose only from few 
predefined options (0, 45, 90, 270 and 315)</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_display_numbers">display_numbers</code></td>
<td>
<p>logical determining if the numeric values are also printed to 
the cells. If this is a matrix (with same dimensions as original matrix), the contents
of the matrix are shown instead of original values.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_number_format">number_format</code></td>
<td>
<p>format strings (C printf style) of the numbers shown in cells. 
For example &quot;<code>%.2f</code>&quot; shows 2 decimal places and &quot;<code>%.1e</code>&quot; shows exponential 
notation (see more in <code><a href="base.html#topic+sprintf">sprintf</a></code>).</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_number_color">number_color</code></td>
<td>
<p>color of the text</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_fontsize_number">fontsize_number</code></td>
<td>
<p>fontsize of the numbers displayed in cells</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_width">width</code></td>
<td>
<p>manual option for determining the output file width in inches.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_height">height</code></td>
<td>
<p>manual option for determining the output file height in inches.</p>
</td></tr>
<tr><td><code id="tidyheatmap_+3A_silent">silent</code></td>
<td>
<p>do not draw the plot (useful when using the gtable output)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly a <code>pheatmap</code> object that is a list with components
</p>

<ul>
<li> <p><code>tree_row</code> the clustering of rows as <code>hclust</code> object
</p>
</li>
<li> <p><code>tree_col</code> the clustering of columns as <code>hclust</code> object
</p>
</li>
<li> <p><code>kmeans</code> the kmeans clustering of rows if parameter <code>kmeans_k</code> was specified
</p>
</li>
<li> <p><code>gtable</code> a <code>gtable</code> object containing the heatmap, can be used for combining
the heatmap with other plots
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Basic example
tidyheatmap(data_exprs,
            rows = external_gene_name,
            columns = sample,
            values = expression,
            scale = "row"
)

# Change number of colors in color lengend
tidyheatmap(data_exprs,
            rows = external_gene_name,
            columns = sample,
            values = expression,
            scale = "row",
            color_legend_n = 5
)

# Change color in color legend
tidyheatmap(data_exprs,
            rows = external_gene_name,
            columns = sample,
            values = expression,
            scale = "row",
            colors = c("#145afc","#ffffff","#ee4445")
)

# Add row and column annotation
tidyheatmap(data_exprs,
            rows = external_gene_name,
            columns = sample,
            values = expression,
            scale = "row",
            annotation_col = c(sample_type, condition, group),
            annotation_row = c(is_immune_gene, direction)
)

# Add gaps between rows and columns
tidyheatmap(data_exprs,
            rows = external_gene_name,
            columns = sample,
            values = expression,
            scale = "row",
            annotation_col = c(sample_type, condition, group),
            annotation_row = c(is_immune_gene, direction),
            gaps_row = direction,
            gaps_col = group
)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
