<!DOCTYPE html><html><head><title>Help for package tidyestimate</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {tidyestimate}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#common_genes'><p>Genes shared between six expression platforms</p></a></li>
<li><a href='#estimate_score'><p>Infer tumor purity using the ESTIMATE algorithm</p></a></li>
<li><a href='#filter_common_genes'><p>Remove non-common genes from data frame</p></a></li>
<li><a href='#gene_sets'><p>Gene sets to infer tumor stromal and immune infiltration</p></a></li>
<li><a href='#ov'><p>Ovarian cancer tumor RNA expression</p></a></li>
<li><a href='#plot_purity'><p>Plot Affymetrix purity scores against ESTIMATE study purity scores</p></a></li>
<li><a href='#purity_data_affy'><p>Affymetrix data used to train ESTIMATE algorithm</p></a></li>
<li><a href='#tidyestimate'><p>tidyestimate: A modern implementation of the ESTIMATE algorithm</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Tidy Implementation of 'ESTIMATE'</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>The 'ESTIMATE' package infers tumor purity from expression data as a 
  function of immune and stromal infiltrate, but requires writing of intermediate 
  files, is un-pipeable, and performs poorly when presented with modern datasets 
  with current gene symbols. 'tidyestimate' a fast, tidy, modern reimagination of
  'ESTIMATE' (2013) &lt;<a href="https://doi.org/10.1038%2Fncomms3612">doi:10.1038/ncomms3612</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/KaiAragaki/tidyestimate">https://github.com/KaiAragaki/tidyestimate</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/KaiAragaki/tidyestimate/issues">https://github.com/KaiAragaki/tidyestimate/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>glue, dplyr, stats, rlang, ggrepel, ggplot2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-20 23:08:33 UTC; kai</td>
</tr>
<tr>
<td>Author:</td>
<td>Kai Aragaki <a href="https://orcid.org/0000-0002-9458-0426"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Paul Roebuck [cph] (Copyright holder of ESTIMATE package),
  Kosuke Yoshihara [aut] (Author of original ESTIMATE algorithm),
  Rahulsimham Vegesna [aut] (Author of original ESTIMATE algorithm),
  Hoon Kim [aut] (Author of original ESTIMATE algorithm),
  Roel Verhaak [aut] (Author of original ESTIMATE algorithm)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kai Aragaki &lt;aaragak1@jhmi.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-21 03:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='common_genes'>Genes shared between six expression platforms</h2><span id='topic+common_genes'></span>

<h3>Description</h3>

<p>As the ESTIMATE model was trained on a specific set of genes,
only those within this dataset should be included before running
<code>estimate_scores</code>.
</p>
<p>These are the genes common to 6 platforms:
</p>
<p>- Affymetrix HG-U133Plus2.0
</p>
<p>- Affymetrix HT-HG-U133A
</p>
<p>- Affymetrix Human X3P
</p>
<p>- Agilent 4x44K (G4112F)
</p>
<p>- Agilent G4502A
</p>
<p>- Illumina HiSeq RNA sequence
</p>
<p>The Entrez IDs for the original 10412 genes were matched to HGNC symbols
using <code>biomaRt</code>. Duplicates and blank entries were filtered. As some
have now been discovered to be pseudogenes or have been deprecated, 22
genes (at time of writing, June 2021) that were in the ESTIMATE package do
not exist here.
</p>
<p>As one gene can have multiple synonyms/aliases, and there is only one alias
per line, the number of rows in the data frame (26339) does not reflect the
number of unique genes in the dataset (10391).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>common_genes
</code></pre>


<h3>Format</h3>

<p>A data frame with 26339 rows and 3 variables: </p>

<dl>
<dt>entrezgene_id</dt><dd><p>Entrez id of the gene</p>
</dd> <dt>hgnc_symbol</dt><dd><p>Human Genome
Organisation (HUGO) Gene Nomenclature Committee symbol</p>
</dd>
<dt>external_synonym</dt><dd><p>A synonym/alias a given gene may go by or
previously went by</p>
</dd> </dl>



<h3>Details</h3>

<p>The ESTIMATE model was trained on a set of genes shared between six
expression profiling platforms. Those genes are listed in this dataset.
</p>


<h3>Source</h3>

<p><a href="https://r-forge.r-project.org/scm/viewvc.php/pkg/estimate/data/common_genes.RData?root=estimate&amp;view=log">https://r-forge.r-project.org/scm/viewvc.php/pkg/estimate/data/common_genes.RData?root=estimate&amp;view=log</a>
</p>

<hr>
<h2 id='estimate_score'>Infer tumor purity using the ESTIMATE algorithm</h2><span id='topic+estimate_score'></span>

<h3>Description</h3>

<p>Infer tumor purity by using single-sample gene-set-enrichment-analysis with
stromal and immune cell signatures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_score(df, is_affymetrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_score_+3A_df">df</code></td>
<td>
<p>a <code>data.frame</code> of expression data, where columns are tumors
and rows are genes. Gene names must be in the first column, and in the form
of HGNC symbols.</p>
</td></tr>
<tr><td><code id="estimate_score_+3A_is_affymetrix">is_affymetrix</code></td>
<td>
<p>logical. Is the expression data from an Affymetrix
array?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ESTIMATE (and this tidy implementation) infers tumor infiltration using two
gene sets: a stromal signature, and an immune signature (see
<code>tidyestimate::gene_sets</code>).
</p>
<p>Enrichment scores for each sample are calculated using an implementation of
single sample Gene Set Enrichment Analysis (ssGSEA). Briefly, expression is
ranked on a per-sample basis, and the density and distribution of gene
signature 'hits' is determined. An enrichment of hits at the top of the
expression ranking confers a positive score, while an enrichment of hits at
the bottom of the expression ranking confers a negative score.
</p>
<p>An 'ESTIMATE' score is calculated by adding the stromal and immune scores
together.
</p>
<p>For Affymetrix arrays, an equation to convert an ESTIMATE score to a
prediction of tumor purity has been developed by Yoshihara et al. (see
references). It takes the approximate form of:
</p>
<p style="text-align: center;"><code class="reqn">purity = cos(0.61 + 0.00015 * ESTIMATE)</code>
</p>

<p>Values have been rounded to two significant figures for display purposes.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with sample names, as well as scores for stromal,
immune, and ESTIMATE scores per tumor. If <code>is_affymetrix = TRUE</code>,
purity scores as well.
</p>
<p>Purity scores can be interpreted absolutely: a purity of 0.9 means that tumor
is likely 90
available (such as in RNAseq), ESTIMATE scores can only be interpreted
relatively: a sample that has a lower ESTIMATE score than another in one
study can be regarded as more pure than another, but its absolute purity
cannot be inferred, nor can purity across other studies be inferred.
</p>


<h3>References</h3>

<p>Barbie et al. (2009) &lt;doi:10.1038/nature08460&gt;
</p>
<p>Yoshihara et al. (2013) &lt;doi:10.1038/ncomms3612&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filter_common_genes(ov, id = "hgnc_symbol", tidy = FALSE, tell_missing = TRUE, find_alias = TRUE) |&gt; 
  estimate_score(is_affymetrix = TRUE)
</code></pre>

<hr>
<h2 id='filter_common_genes'>Remove non-common genes from data frame</h2><span id='topic+filter_common_genes'></span>

<h3>Description</h3>

<p>As ESTIMATE score calculation is sensitive to the number of genes used, a set
of common genes used between six platforms has been established (see
<code>?tidyestimate::common_genes</code>). This function will filter for only those
genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_common_genes(
  df,
  id = c("entrezgene_id", "hgnc_symbol"),
  tidy = FALSE,
  tell_missing = TRUE,
  find_alias = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_common_genes_+3A_df">df</code></td>
<td>
<p>a <code>data.frame</code> of RNA expression values, with columns corresponding
to samples, and rows corresponding to genes. Either rownames or the first
column can contain gene IDs (see <code>tidy</code>)</p>
</td></tr>
<tr><td><code id="filter_common_genes_+3A_id">id</code></td>
<td>
<p>either <code>"entrezgene_id"</code> or <code>"hgnc_symbol"</code>, whichever
<code>df</code> contains.</p>
</td></tr>
<tr><td><code id="filter_common_genes_+3A_tidy">tidy</code></td>
<td>
<p>logical. If rownames contain gene identifier, set <code>FALSE</code>.
If first column contains gene identifier, set <code>TRUE</code></p>
</td></tr>
<tr><td><code id="filter_common_genes_+3A_tell_missing">tell_missing</code></td>
<td>
<p>logical. If <code>TRUE</code>, prints message of genes in
common gene set that are not in supplied data frame.</p>
</td></tr>
<tr><td><code id="filter_common_genes_+3A_find_alias">find_alias</code></td>
<td>
<p>logical. If <code>TRUE</code> and <code>id = "hgnc_symbol"</code>, will
attempt to find if genes missing from <code>common_genes</code> are going under
an alias. See details for more information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>find_aliases</code> argument will attempt to find aliases for HGNC
symbols in <code>tidyestimate::common_genes</code> but missing from the provided
dataset. This will only run if <code>find_aliases = TRUE</code> and <code>id =
"hgnc_symbol"</code>.
</p>
<p>This algorithm is very conservative: It will only make a match if the gene
from the common genes has only one alias that matches with only one gene from
the provided dataset, <em>and</em> the gene from the provided dataset with
which it matches only matches with a single gene from the list of common
genes. (Note that a single gene may have many aliases). Once a match has been
made, the gene in the provided dataset is updated to the gene name in the
common gene list.
</p>
<p>While this method is fairly accurate, is is also a heuristic. Therefore, it is
disabled by default. Users should check which genes are becoming reassigned
to ensure accuracy.
</p>
<p>The method of generation of these aliases can be found at
<code>?tidyestimate::common_genes</code>
</p>


<h3>Value</h3>

<p>A <code>tibble</code>, with gene identifiers as the first column
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filter_common_genes(ov, id = "hgnc_symbol", tidy = FALSE, tell_missing = TRUE, find_alias = FALSE)
</code></pre>

<hr>
<h2 id='gene_sets'>Gene sets to infer tumor stromal and immune infiltration</h2><span id='topic+gene_sets'></span>

<h3>Description</h3>

<p>Two gene sets, each 141 genes in length, created to infer
stromal and immune infiltration
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_sets
</code></pre>


<h3>Format</h3>

<p>A data frame with 141 row and 2 variables:
</p>

<dl>
<dt>stromal_signature</dt><dd><p>Geneset of HGNC symbols used to infer tumor stromal cell infiltration</p>
</dd>
<dt>immune_signature</dt><dd><p>Geneset of HGNC symbols used to infer tumor immune cell infiltration</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://r-forge.r-project.org/scm/viewvc.php/pkg/estimate/data/SI_geneset.RData?root=estimate&amp;view=log">https://r-forge.r-project.org/scm/viewvc.php/pkg/estimate/data/SI_geneset.RData?root=estimate&amp;view=log</a>
</p>

<hr>
<h2 id='ov'>Ovarian cancer tumor RNA expression</h2><span id='topic+ov'></span>

<h3>Description</h3>

<p>A matrix containing RNA expression of 10 ovarian cancer tumors,
measured using the Affymetrix U133Plus2.0 platform. These data have been
rounded to the 4th decimal place to reduce file size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ov
</code></pre>


<h3>Format</h3>

<p>A matrix with 17256 rows and 10 columns, where each column represents
a tumor, and each row represents a gene. Genes are represented by HGNC
symbols in the rownames.
</p>


<h3>Source</h3>

<p><a href="https://r-forge.r-project.org/scm/viewvc.php/pkg/estimate/inst/extdata/sample_input.txt?root=estimate&amp;view=log">https://r-forge.r-project.org/scm/viewvc.php/pkg/estimate/inst/extdata/sample_input.txt?root=estimate&amp;view=log</a>
</p>

<hr>
<h2 id='plot_purity'>Plot Affymetrix purity scores against ESTIMATE study purity scores</h2><span id='topic+plot_purity'></span>

<h3>Description</h3>

<p>Plot Affymetrix purity scores against ESTIMATE study purity scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_purity(scores, is_affymetrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_purity_+3A_scores">scores</code></td>
<td>
<p>a <code>data.frame</code>, usually one output from <code>estimate_score</code></p>
</td></tr>
<tr><td><code id="plot_purity_+3A_is_affymetrix">is_affymetrix</code></td>
<td>
<p>logical. Are these data from an Affymetrix experiment?
Must be <code>TRUE</code> - this is essentially a verification from the user</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filter_common_genes(ov, id = "hgnc_symbol", tidy = FALSE, tell_missing = TRUE, find_alias = TRUE) |&gt; 
  estimate_score(is_affymetrix = TRUE) |&gt;
  plot_purity(is_affymetrix = TRUE)

</code></pre>

<hr>
<h2 id='purity_data_affy'>Affymetrix data used to train ESTIMATE algorithm</h2><span id='topic+purity_data_affy'></span>

<h3>Description</h3>

<p>A data frame containing the ABSOLUTE-measured and
ESTIMATE-predicted purity values of 995 tumors. Additionally, stromal and
immune scores as calculated by ESTIMATE. All tumors were profiled on
Affymetrix arrays, and were used to generate the Affymetrix algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>purity_data_affy
</code></pre>


<h3>Format</h3>

<p>A data frame with 995 rows and 7 variables: 
</p>

<dl>
<dt>purity_observed</dt><dd><p>The purity of a tumor given by ABSOLUTE, ranging from 0 (least pure) to 1 (most pure)</p>
</dd> 
<dt>stromal</dt><dd><p>Stromal infiltration score, as measured by ESTIMATE</p>
</dd>
<dt>immune</dt><dd><p>Immune infiltration score, as measured by ESTIMATE</p>
</dd>
<dt>estimate</dt><dd><p>ESTIMATE score, calculated by the sum of immune and stromal scores</p>
</dd>
<dt>purity_predicted</dt><dd><p>Tumor purity inferred using the ESTIMATE algorithm</p>
</dd>
<dt>ci_95_low</dt><dd><p>Lower bound of a 95% confidence interval of predicted purity scores</p>
</dd>
<dt>ci_95_high</dt><dd><p>Upper bound of a 95% confidence interval of predicted purity scores</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://r-forge.r-project.org/scm/viewvc.php/pkg/estimate/data/PurityDataAffy.RData?root=estimate&amp;view=log">https://r-forge.r-project.org/scm/viewvc.php/pkg/estimate/data/PurityDataAffy.RData?root=estimate&amp;view=log</a>
</p>

<hr>
<h2 id='tidyestimate'>tidyestimate: A modern implementation of the ESTIMATE algorithm</h2><span id='topic+tidyestimate'></span>

<h3>Description</h3>

<p>The tidyestimate is a lightweight, fast, pipe-friendly re-imagination of the
ESTIMATE package. tidyestimate is used to infer tumor purity from
expression data.
</p>


<h3>Authors</h3>

<p>Author (tidyestimate):
</p>
<p>* Kai Aragaki ([ORCID](http://orcid.org/0000-0002-9458-0426)) (author, maintainer)
</p>
<p>Authors (ESTIMATE):
</p>
<p>* Kosuke Yoshihara kyoshihara@mdanderson.org (author)
* P. Roebuck proebuck@mdanderson.org (author, copyright holder)
</p>


<h3>Reference</h3>

<p>https://www.nature.com/articles/ncomms3612
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
