<!DOCTYPE html><html><head><title>Help for package TidyMultiqc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TidyMultiqc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#list_plots'><p>List the plot identifiers of all the plots in a given multiqc report</p></a></li>
<li><a href='#load_multiqc'><p>Loads one or more MultiQC reports into a data frame</p></a></li>
<li><a href='#parse_bar_graph'><p>Takes the JSON dictionary for a bar graph, and returns a named list of</p>
data frames, one for each sample.</a></li>
<li><a href='#parse_xyline_plot'><p>Takes the JSON dictionary for an xyline plot, and returns a named list of</p>
data frames, one for each sample.</a></li>
<li><a href='#TidyMultiqc-package'><p>TidyMultiqc: Converting MultiQC reports into tidy data frames</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Converts 'MultiQC' Reports into Tidy Data Frames</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Milton</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Milton &lt;michael.r.milton@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides the means to convert 'multiqc_data.json' files,
    produced by the wonderful 'MultiQC' tool, into tidy data frames for downstream
    analysis in R. This analysis might involve cohort analysis, quality control visualisation,
    change-point detection, statistical process control, clustering, or any other
    type of quality analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertthat, dplyr, jsonlite, magrittr, purrr, rlang, stringr,
tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tidyr, testthat (&ge; 3.0.0), knitr, rmarkdown, ggplot2,
HistDat</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://multimeric.github.io/TidyMultiqc/">https://multimeric.github.io/TidyMultiqc/</a>,
<a href="https://github.com/multimeric/TidyMultiqc">https://github.com/multimeric/TidyMultiqc</a>,
<a href="https://cran.r-project.org/package=TidyMultiqc">https://cran.r-project.org/package=TidyMultiqc</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/multimeric/TidyMultiqc/issues">https://github.com/multimeric/TidyMultiqc/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-25 01:30:10 UTC; migwell</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-25 02:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='list_plots'>List the plot identifiers of all the plots in a given multiqc report</h2><span id='topic+list_plots'></span>

<h3>Description</h3>

<p>List the plot identifiers of all the plots in a given multiqc report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_plots(path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="list_plots_+3A_path">path</code></td>
<td>
<p>The file path to the multiqc report. This should be a length 1
character vector</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The main use for this function is finding the plot identifiers
that you will then pass into the <code>plots</code> argument of the <code><a href="#topic+load_multiqc">load_multiqc()</a></code>
function.
Refer to the section on &quot;Extracting Plot Data&quot; in the main vignette for more information.
</p>


<h3>Value</h3>

<p>A data frame containing <code class="reqn">n</code> rows, where <code class="reqn">n</code> is the number
of plots in the report you have provided, and two columns:
</p>

<dl>
<dt>id</dt><dd><p>The identifier for the plot. This is the one you should use as a name in <code>plot_opts</code>.</p>
</dd>
<dt>name</dt><dd><p>The plot title. This is likely what you see in the multiqc report when you open it with your browser.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'># Ignore this, choose your own filepath as the `filepath` variable
filepath &lt;- system.file("extdata", "HG00096/multiqc_data.json", package = "TidyMultiqc")
# This is the actual invocation
list_plots(filepath)
</code></pre>

<hr>
<h2 id='load_multiqc'>Loads one or more MultiQC reports into a data frame</h2><span id='topic+load_multiqc'></span>

<h3>Description</h3>

<p>Loads one or more MultiQC reports into a data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_multiqc(
  paths,
  plots = NULL,
  find_metadata = function(...) {     list() },
  plot_parsers = list(),
  sections = "general"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_multiqc_+3A_paths">paths</code></td>
<td>
<p>A string vector of filepaths to multiqc_data.json files</p>
</td></tr>
<tr><td><code id="load_multiqc_+3A_plots">plots</code></td>
<td>
<p>A string vector, each of which contains the ID of a plot you
want to include in the output. You can use <code><a href="#topic+list_plots">list_plots()</a></code> to help here.</p>
</td></tr>
<tr><td><code id="load_multiqc_+3A_find_metadata">find_metadata</code></td>
<td>
<p>A single function that will be called with a sample name and the
parsed JSON for the entire report and returns a named list of metadata fields for the sample.
Refer to the vignette for an example.</p>
</td></tr>
<tr><td><code id="load_multiqc_+3A_plot_parsers">plot_parsers</code></td>
<td>
<p><strong>Advanced</strong>. A named list of custom parser functions.
The names of the list should correspond to plotly plot types, such as &quot;xy_line&quot;, and the values should be functions
that return a named list of named lists. For the return value, the outer list is named by the sample ID, and the inner list
is named by the name of the column. Refer to the source code for some examples.</p>
</td></tr>
<tr><td><code id="load_multiqc_+3A_sections">sections</code></td>
<td>
<p>A string vector of zero or more sections to include in the output.
Each section can be:
</p>

<dl>
<dt>&quot;plot&quot;</dt><dd><p>Parse plot data. Note that you should also provide a list of plots via the <code>plots</code> argument</p>
</dd>
<dt>&quot;general&quot;</dt><dd><p>parse the general stat section</p>
</dd>
<dt>&quot;raw&quot;</dt><dd><p>Parse the raw data section</p>
</dd>
</dl>

<p>This defaults to 'general', which tends to contain the most useful statistics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble (data.frame subclass) with QC data and metadata as columns, and samples as rows.
Columns are named according to the respective section they belong to,
and will always be listed in the following order:
</p>
<table>
<tr><td><code>metadata.X</code></td>
<td>
<p>This column contains metadata for this sample.
By default this is only the sample ID, but if you have provided the
<code>find_metadata</code> argument, there may be more columns.</p>
</td></tr>
<tr><td><code>general.X</code></td>
<td>
<p>This column contains a generally useful summary statistic for each sample</p>
</td></tr>
<tr><td><code>plot.X</code></td>
<td>
<p>This column contains a data frame of plot data for each sample.
Refer to the plot parsers documentation (ie the <code>parse_X</code> functions) for more information on the output format. </p>
</td></tr>
<tr><td><code>raw.X</code></td>
<td>
<p>This column contains a raw summary statistic or value relating to each sample </p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+parse_xyline_plot">parse_xyline_plot()</a></code> <code><a href="#topic+parse_bar_graph">parse_bar_graph()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_multiqc(system.file("extdata", "wgs/multiqc_data.json", package = "TidyMultiqc"))
</code></pre>

<hr>
<h2 id='parse_bar_graph'>Takes the JSON dictionary for a bar graph, and returns a named list of
data frames, one for each sample.</h2><span id='topic+parse_bar_graph'></span>

<h3>Description</h3>

<p>Takes the JSON dictionary for a bar graph, and returns a named list of
data frames, one for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_bar_graph(plot_data, name)
</code></pre>


<h3>Value</h3>

<p>A list of data frames, one for each sample.
Each data frame will have one column corresponding to each category in the bar chart.
For example, for the plot &quot;SnpEff: Counts by Genomic Region&quot;, we will have
one column for the number of intron variants, one column for the number of exon variants, etc.
This means that the number of columns will be fairly variable for different plots.
</p>

<hr>
<h2 id='parse_xyline_plot'>Takes the JSON dictionary for an xyline plot, and returns a named list of
data frames, one for each sample.</h2><span id='topic+parse_xyline_plot'></span>

<h3>Description</h3>

<p>Takes the JSON dictionary for an xyline plot, and returns a named list of
data frames, one for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_xyline_plot(plot_data, name)
</code></pre>


<h3>Value</h3>

<p>A list of data frames, one for each sample.
Each data frame will have two columns: x, and y.
These correspond to the x and y coordinates in the plot.
For example, for histogram data, the x values are values of the random
variable, and the y values are the number of counts for that value.
</p>

<hr>
<h2 id='TidyMultiqc-package'>TidyMultiqc: Converting MultiQC reports into tidy data frames</h2><span id='topic+TidyMultiqc-package'></span>

<h3>Description</h3>

<p>This package provides the means to convert <code>multiqc_data.json</code>
files, produced by the wonderful <a href="http://multiqc.info/">MultiQC</a> tool,
into tidy data.frames for downstream analysis in R.
If you are reading this manual, you should immediately stop reading this and
instead refer to the documentation website at <a href="https://multimeric.github.io/TidyMultiqc/">https://multimeric.github.io/TidyMultiqc/</a>, which
provides more accessible documentation.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
