<!DOCTYPE html><html><head><title>Help for package timeSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {timeSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pAbp'><p>Example Data in Vignette</p></a></li>
<li><a href='#simulate.dt'><p>Example Data in Vignette</p></a></li>
<li><a href='#timeSeq'><p>Statistical Inference for Time Course RNA-Seq Data using a Negative Binomial Mixed-Effects Model</p></a></li>
<li><a href='#timeSeq_1.0.4-package'>
<p>Statistical Inference for Time Course RNA-Seq Data using a Negative Binomial Mixed-Effects Model</p></a></li>
<li><a href='#timeSeq.heatmap'>
<p>Heatmap of the Most Significant NDPE Genes</p></a></li>
<li><a href='#timeSeq.screeplot'>
<p>Scree Plot of Kullback Leibler Distance Ratios</p></a></li>
<li><a href='#timeSeq.sort'>
<p>Sort NDPE Genes by Kullback Leibler Distance Ratios or P-values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Detecting Differentially Expressed Genes in Time Course RNA-Seq
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-2-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Fan Gao, Xiaoxiao Sun</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fan Gao &lt;fangaohz@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A negative binomial mixed-effects (NBME) model to detect
    nonparallel differential expression(NPDE) genes and parallel differential
    expression(PDE) genes in the time course RNA-seq data.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>gss, mgcv, lattice, pheatmap, reshape, grDevices, graphics</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1.9000</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-02-09 05:16:16 UTC; xiaosun</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-02-09 05:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='pAbp'>Example Data in Vignette</h2><span id='topic+pAbp'></span>

<h3>Description</h3>

<p>The data 'pAbp'(list) has 4 arguments, including 'data.count', 'exon.length', 'gene.names', and 'group.label'.
Please refer to 'timeSeq.Rd' for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pAbp)
</code></pre>


<h3>Format</h3>

<p>A list with 4 arguments:</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pAbp)
str(pAbp)
</code></pre>

<hr>
<h2 id='simulate.dt'>Example Data in Vignette</h2><span id='topic+simulate.dt'></span>

<h3>Description</h3>

<p>The data 'simulate.dt'(list) has 3 arguments, including 'data.count', 'group.label', and 'gene.names'.
Please refer to 'timeSeq.Rd' for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(simulate.dt)
</code></pre>


<h3>Format</h3>

<p>A list with 3 arguments:</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simulate.dt)
str(simulate.dt)
</code></pre>

<hr>
<h2 id='timeSeq'>Statistical Inference for Time Course RNA-Seq Data using a Negative Binomial Mixed-Effects Model</h2><span id='topic+timeSeq'></span>

<h3>Description</h3>

<p>Accurately identifying differentially expressed (DE)
genes from time course RNA-seq data has been of tremendous
significance in creating a global picture of cellular function. DE
genes from the time course RNA-seq data can be classified into two
types, parallel DE genes (PDE) and non-parallel DE (NPDE) genes.
The former are often biologically irrelevant, whereas the latter are often
biologically interesting. In this package, we propose a negative binomial
mixed-effects (NBME) model to identify both PDE and NPDE genes in
time course RNA-seq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeSeq(data.count,group.label,gene.names,exon.length=NULL,exon.level=FALSE,pvalue=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="timeSeq_+3A_data.count">data.count</code></td>
<td>
<p>a n by p matrix of expression values. Data should be appropriately normalized beforehand.</p>
</td></tr>
<tr><td><code id="timeSeq_+3A_group.label">group.label</code></td>
<td>
<p>a vector indicating the experimental conditions of each time point.</p>
</td></tr>
<tr><td><code id="timeSeq_+3A_gene.names">gene.names</code></td>
<td>
<p>a vector containing all the gene names.</p>
</td></tr>
<tr><td><code id="timeSeq_+3A_exon.length">exon.length</code></td>
<td>
<p>a vector containing the length of exons, only used in exon level data.</p>
</td></tr>
<tr><td><code id="timeSeq_+3A_exon.level">exon.level</code></td>
<td>
<p>logical:indicating if this is an exon level dataset. Default is FALSE.</p>
</td></tr>
<tr><td><code id="timeSeq_+3A_pvalue">pvalue</code></td>
<td>
<p>logical:indicating if p-values are returned. Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nonparallel differential expression(NPDE) genes 
and parallel differential expression(PDE) genes detection.</p>


<h3>Value</h3>

<p>A list with components
</p>
<table>
<tr><td><code>sorted</code></td>
<td>
<p>an object returned by timeSeq.sort function. It contains sorted Kullback Leibler Ratios(KLRs) or p-values for identifying DE genes.</p>
</td></tr>
<tr><td><code>count</code></td>
<td>
<p>the number of exons or replicates for each gene.</p>
</td></tr>
<tr><td><code>NPDE</code></td>
<td>
<p>the NPDE ratios or p-values.</p>
</td></tr>
<tr><td><code>PDE</code></td>
<td>
<p>the PDE ratios or p-values.</p>
</td></tr>
<tr><td><code>genenames</code></td>
<td>
<p>gene names.</p>
</td></tr>
<tr><td><code>table</code></td>
<td>
<p>gene expression values.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>a n by p matrix of expression values.</p>
</td></tr>
<tr><td><code>gene.names</code></td>
<td>
<p>a vector including all the gene names.</p>
</td></tr>
<tr><td><code>group.label</code></td>
<td>
<p>a vector indicating the experimental conditions of each time point.</p>
</td></tr>
<tr><td><code>group.length</code></td>
<td>
<p>the total number of time points.</p>
</td></tr>
<tr><td><code>group1.length</code></td>
<td>
<p>the number of time points of condition one.</p>
</td></tr>
<tr><td><code>group2.length</code></td>
<td>
<p>the number of time points of condition two.</p>
</td></tr>
<tr><td><code>exon.level</code></td>
<td>
<p>logical:indicating if this is an exon level dataset. Default is FALSE.</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>logical:indicating if p-values are returned. Default is TRUE.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Gao and Xiaoxiao Sun</p>


<h3>References</h3>

<p>Sun, Xiaoxiao, David Dalpiaz, Di Wu, Jun S. Liu, Wenxuan Zhong, and Ping Ma. &quot;Statistical inference for time course RNA-Seq data using a negative binomial mixed-effect model.&quot; BMC Bioinformatics, 17(1):324, 2016.
</p>
<p>Chong Gu. Model diagnostics for smoothing spline ANOVA models. Canadian Journal of Statistics, 32(4):347-358, 2004.
</p>
<p>Chong Gu. Smoothing spline ANOVA models. Springer, second edition, 2013.
</p>
<p>Chong Gu and Ping Ma. Optimal smoothing in nonparametric mixed-effect models. Annals of Statistics, 1357-1379, 2005.
</p>
<p>Wood (2001) mgcv:GAMs and Generalized Ridge Regression for R. R News 1(2):20-25
</p>


<h3>Examples</h3>

<pre><code class='language-R'>####Data should be appropriately normalized beforehand####

##Exon level data (The p-values calculation is not supported)
data(pAbp)
attach(pAbp)
model.fit &lt;- timeSeq(data.count,group.label,gene.names,exon.length,exon.level=TRUE,pvalue=FALSE)
#NPDE genes have large KLRs
model.fit$NPDE
detach(pAbp)

##Gene level data (three replicates)
data(simulate.dt)
attach(simulate.dt)
model.fit &lt;- timeSeq(data.count,group.label,gene.names,exon.level=FALSE,pvalue=TRUE)
#p-values
model.fit$NPDE
</code></pre>

<hr>
<h2 id='timeSeq_1.0.4-package'>
Statistical Inference for Time Course RNA-Seq Data using a Negative Binomial Mixed-Effects Model
</h2><span id='topic+timeSeq_1.0.4-package'></span><span id='topic+timeSeq_1.0.4'></span>

<h3>Description</h3>

<p>In this package, we propose a negative binomial mixed-effects (NBME) model to identify differentially expressed (DE) genes, including nonparallel differentially expressed (NPDE) and parallel differentially expressed (PDE) genes, in time course RNA-seq data. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> timeSeq</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.4</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-2-08</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2 | GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Fan Gao and Xiaoxiao Sun
</p>
<p>Maintainer: Fan Gao &lt;fangaohz@gmail.com&gt;
</p>


<h3>References</h3>

<p>Sun, Xiaoxiao, David Dalpiaz, Di Wu, Jun S. Liu, Wenxuan Zhong, and Ping Ma. &quot;Statistical inference for time course RNA-Seq data using a negative binomial mixed-effect model.&quot; BMC Bioinformatics, 17(1):324, 2016.
</p>
<p>Chong Gu. Model diagnostics for smoothing spline ANOVA models. Canadian Journal of Statistics, 32(4):347-358, 2004.
</p>
<p>Chong Gu. Smoothing spline ANOVA models. Springer, second edition, 2013.
</p>
<p>Chong Gu and Ping Ma. Optimal smoothing in nonparametric mixed-effect models. Annals of Statistics, pages 1357-1379, 2005a.
</p>
<p>Wood (2001) mgcv:GAMs and Generalized Ridge Regression for R. R News 1(2):20-25
</p>

<hr>
<h2 id='timeSeq.heatmap'>
Heatmap of the Most Significant NDPE Genes
</h2><span id='topic+timeSeq.heatmap'></span>

<h3>Description</h3>

<p>Heatmap for the most significant NDPE genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeSeq.heatmap(timeSeq.obj,n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="timeSeq.heatmap_+3A_timeseq.obj">timeSeq.obj</code></td>
<td>
<p>an object returned by timeSeq function</p>
</td></tr>
<tr><td><code id="timeSeq.heatmap_+3A_n">n</code></td>
<td>
<p>the number of the most significant NPDE genes. It must be a positive integer.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Gao and Xiaoxiao Sun
</p>


<h3>References</h3>

<p>Sun, Xiaoxiao, David Dalpiaz, Di Wu, Jun S. Liu, Wenxuan Zhong, and Ping Ma. &quot;Statistical inference for time course RNA-Seq data using a negative binomial mixed-effect model.&quot; BMC Bioinformatics, 17(1):324, 2016.
</p>
<p>Chong Gu. Model diagnostics for smoothing spline ANOVA models. Canadian Journal of Statistics, 32(4):347-358, 2004.
</p>
<p>Chong Gu. Smoothing spline ANOVA models. Springer, second edition, 2013.
</p>
<p>Chong Gu and Ping Ma. Optimal smoothing in nonparametric mixed-effect models. Annals of Statistics, pages 1357-1379, 2005a.
</p>
<p>Wood (2001) mgcv:GAMs and Generalized Ridge Regression for R. R News 1(2):20-25
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(simulate.dt)
attach(simulate.dt)
model.fit &lt;- timeSeq(data.count,group.label,gene.names,exon.level=FALSE,pvalue=TRUE)
timeSeq.heatmap(model.fit,n=10)
</code></pre>

<hr>
<h2 id='timeSeq.screeplot'>
Scree Plot of Kullback Leibler Distance Ratios</h2><span id='topic+timeSeq.screeplot'></span>

<h3>Description</h3>

<p>Plot the scree plot for all genes in the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeSeq.screeplot(timeSeq.obj,type=c("barplot","lines"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="timeSeq.screeplot_+3A_timeseq.obj">timeSeq.obj</code></td>
<td>
<p>an object returned by timeSeq function.</p>
</td></tr>  
<tr><td><code id="timeSeq.screeplot_+3A_type">type</code></td>
<td>
<p>type of plot: &quot;barplot&quot; for the bar plot and &quot;lines&quot; for the line graph.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Gao and Xiaoxiao Sun
</p>


<h3>References</h3>

<p>Sun, Xiaoxiao, David Dalpiaz, Di Wu, Jun S. Liu, Wenxuan Zhong, and Ping Ma. &quot;Statistical inference for time course RNA-Seq data using a negative binomial mixed-effect model.&quot; BMC Bioinformatics, 17(1):324, 2016.
</p>
<p>Chong Gu. Model diagnostics for smoothing spline ANOVA models. Canadian Journal of Statistics, 32(4):347-358, 2004.
</p>
<p>Chong Gu. Smoothing spline ANOVA models. Springer, second edition, 2013.
</p>
<p>Chong Gu and Ping Ma. Optimal smoothing in nonparametric mixed-effect models. Annals of Statistics, pages 1357-1379, 2005a.
</p>
<p>Wood (2001) mgcv:GAMs and Generalized Ridge Regression for R. R News 1(2):20-25
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(simulate.dt)
attach(simulate.dt)
model.fit &lt;- timeSeq(data.count,group.label,gene.names,exon.level=FALSE,pvalue=FALSE)
timeSeq.screeplot(model.fit,"lines")

</code></pre>

<hr>
<h2 id='timeSeq.sort'>
Sort NDPE Genes by Kullback Leibler Distance Ratios or P-values
</h2><span id='topic+timeSeq.sort'></span>

<h3>Description</h3>

<p>Sort all genes in the dataset by their Kullback Leibler distance ratios or p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeSeq.sort(genenames,npde,pde,table,count,pvalue)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="timeSeq.sort_+3A_genenames">genenames</code></td>
<td>
<p>a vector of gene names.</p>
</td></tr>
<tr><td><code id="timeSeq.sort_+3A_npde">npde</code></td>
<td>
<p>a vector of Kullback Leibler distance ratios or p-values for NPDE genes.</p>
</td></tr>
<tr><td><code id="timeSeq.sort_+3A_pde">pde</code></td>
<td>
<p>a vector of Kullback Leibler distance ratios or p-values for PDE genes.</p>
</td></tr>
<tr><td><code id="timeSeq.sort_+3A_table">table</code></td>
<td>
<p>gene expression values.</p>
</td></tr>
<tr><td><code id="timeSeq.sort_+3A_count">count</code></td>
<td>
<p>the number of exons or replicates for each gene.</p>
</td></tr>
<tr><td><code id="timeSeq.sort_+3A_pvalue">pvalue</code></td>
<td>
<p>logical:indicating if p-values are returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with components
</p>
<table>
<tr><td><code>npde.list</code></td>
<td>
<p>dataframe of NPDE genes sorted by KLRs or p-values.</p>
</td></tr>
<tr><td><code>pde.list</code></td>
<td>
<p>dataframe of PDE genes sorted by KLRs or p-values.</p>
</td></tr>
<tr><td><code>table1</code></td>
<td>
<p>gene expression values for each gene, corresponding to npde.list.</p>
</td></tr>
<tr><td><code>table2</code></td>
<td>
<p>gene expression values for each gene, corresponding to pde.list.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Fan Gao and Xiaoxiao Sun
</p>


<h3>References</h3>

<p>Sun, Xiaoxiao, David Dalpiaz, Di Wu, Jun S. Liu, Wenxuan Zhong, and Ping Ma. &quot;Statistical inference for time course RNA-Seq data using a negative binomial mixed-effect model.&quot; BMC Bioinformatics, 17(1):324, 2016.
</p>
<p>Chong Gu. Model diagnostics for smoothing spline ANOVA models. Canadian Journal of Statistics, 32(4):347-358, 2004.
</p>
<p>Chong Gu. Smoothing spline ANOVA models. Springer, second edition, 2013.
</p>
<p>Chong Gu and Ping Ma. Optimal smoothing in nonparametric mixed-effect models. Annals of Statistics, pages 1357-1379, 2005a.
</p>
<p>Wood (2001) mgcv:GAMs and Generalized Ridge Regression for R. R News 1(2):20-25
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
