<!DOCTYPE html><html><head><title>Help for package tinyarray</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {tinyarray}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#box_surv'><p>box_surv</p></a></li>
<li><a href='#cod'><p>cod</p></a></li>
<li><a href='#cor.full'><p>cor.test for all variables</p></a></li>
<li><a href='#cor.one'><p>cor.test for one variable with all variables</p></a></li>
<li><a href='#corheatmap'><p>corheatmap</p></a></li>
<li><a href='#corscatterplot'><p>corscatterplot</p></a></li>
<li><a href='#deg'><p>deg</p></a></li>
<li><a href='#deseq_data'><p>deseq_data</p></a></li>
<li><a href='#double_enrich'><p>draw enrichment bar plots for both up and down genes</p></a></li>
<li><a href='#draw_boxplot'><p>draw boxplot for expression</p></a></li>
<li><a href='#draw_heatmap'><p>draw a heatmap plot</p></a></li>
<li><a href='#draw_heatmap2'><p>draw heatmap plots</p></a></li>
<li><a href='#draw_KM'><p>draw_KM</p></a></li>
<li><a href='#draw_pca'><p>draw PCA plots</p></a></li>
<li><a href='#draw_tsne'><p>draw_tsne</p></a></li>
<li><a href='#draw_venn'><p>draw a venn plot</p></a></li>
<li><a href='#draw_volcano'><p>draw a volcano plot</p></a></li>
<li><a href='#draw_volcano2'><p>draw_volcano2</p></a></li>
<li><a href='#dumd'><p>count unique values in every columns for data.frame</p></a></li>
<li><a href='#edges_to_nodes'><p>edges_to_nodes</p></a></li>
<li><a href='#exists_anno_list'><p>exists_anno_list</p></a></li>
<li><a href='#exp_boxplot'><p>exp_boxplot</p></a></li>
<li><a href='#exp_hub1'><p>exp_hub1</p></a></li>
<li><a href='#exp_surv'><p>exp_surv</p></a></li>
<li><a href='#exprSet_hub1'><p>exprSet_hub1</p></a></li>
<li><a href='#find_anno'><p>find annotation package or files</p></a></li>
<li><a href='#genes'><p>genes</p></a></li>
<li><a href='#geo_download'><p>geo_download</p></a></li>
<li><a href='#get_cgs'><p>get_cgs</p></a></li>
<li><a href='#get_count_txt'><p>get count from GEO</p></a></li>
<li><a href='#get_deg'><p>get_deg</p></a></li>
<li><a href='#get_deg_all'><p>get_deg_all</p></a></li>
<li><a href='#get_gpl_txt'><p>get gpl txt from GEO</p></a></li>
<li><a href='#ggheat'><p>ggheat</p></a></li>
<li><a href='#hypertest'><p>hypertest</p></a></li>
<li><a href='#interaction_to_edges'><p>interaction_to_edges</p></a></li>
<li><a href='#intersect_all'><p>intersect_all</p></a></li>
<li><a href='#lnc_anno'><p>lnc_anno</p></a></li>
<li><a href='#lnc_annov23'><p>lnc_annov23</p></a></li>
<li><a href='#make_tcga_group'><p>make_tcga_group</p></a></li>
<li><a href='#match_exp_cl'><p>match_exp_cl</p></a></li>
<li><a href='#meta1'><p>meta1</p></a></li>
<li><a href='#mRNA_anno'><p>mRNA_anno</p></a></li>
<li><a href='#mRNA_annov23'><p>mRNA_annov23</p></a></li>
<li><a href='#multi_deg'><p>multi_deg</p></a></li>
<li><a href='#multi_deg_all'><p>multi_deg_all</p></a></li>
<li><a href='#pkg_all'><p>pkg_all</p></a></li>
<li><a href='#plcortest'><p>plcortest</p></a></li>
<li><a href='#plot_deg'><p>plot_deg</p></a></li>
<li><a href='#point_cut'><p>point_cut</p></a></li>
<li><a href='#quick_enrich'><p>quick_enrich</p></a></li>
<li><a href='#risk_plot'><p>risk_plot</p></a></li>
<li><a href='#sam_filter'><p>sam_filter</p></a></li>
<li><a href='#surv_cox'><p>surv_cox</p></a></li>
<li><a href='#surv_KM'><p>surv_KM</p></a></li>
<li><a href='#t_choose'><p>t_choose</p></a></li>
<li><a href='#trans_array'><p>trans_array</p></a></li>
<li><a href='#trans_entrezexp'><p>trans_entrezexp</p></a></li>
<li><a href='#trans_exp'><p>trans_exp</p></a></li>
<li><a href='#trans_exp_new'><p>trans_exp_new</p></a></li>
<li><a href='#union_all'><p>union_all</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Expression Data Analysis and Visualization</td>
</tr>
<tr>
<td>Version:</td>
<td>2.4.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaojie Sun &lt;18763899370@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The Gene Expression Omnibus (<a href="https://www.ncbi.nlm.nih.gov/geo/">https://www.ncbi.nlm.nih.gov/geo/</a>) and The Cancer Genome Atlas (<a href="https://portal.gdc.cancer.gov/">https://portal.gdc.cancer.gov/</a>) are widely used medical public databases. Our platform integrates routine analysis and visualization tools for expression data to provide concise and intuitive data analysis and presentation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>BiocManager, clusterProfiler, dplyr, limma, stringr, tibble,
pheatmap, ggplot2, survival, Hmisc, survminer, patchwork</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, AnnoProbe, GEOquery, Biobase, VennDiagram,
FactoMineR, factoextra, knitr, rmarkdown, cowplot, ggpubr,
ggplotify, tidyr, labeling, Rtsne, scatterplot3d,
ComplexHeatmap, circlize,org.Rn.eg.db,org.Mm.eg.db,org.Hs.eg.db</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/xjsun1221/tinyarray">https://github.com/xjsun1221/tinyarray</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/xjsun1221/tinyarray/issues">https://github.com/xjsun1221/tinyarray/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-13 10:50:46 UTC; xiaojie</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiaojie Sun [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-13 14:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='box_surv'>box_surv</h2><span id='topic+box_surv'></span>

<h3>Description</h3>

<p>draw box plot for a hub gene expression matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>box_surv(exp_hub, exprSet_hub, meta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="box_surv_+3A_exp_hub">exp_hub</code></td>
<td>
<p>an expression matrix for hubgenes</p>
</td></tr>
<tr><td><code id="box_surv_+3A_exprset_hub">exprSet_hub</code></td>
<td>
<p>a tumor expression set for hubgenes</p>
</td></tr>
<tr><td><code id="box_surv_+3A_meta">meta</code></td>
<td>
<p>meta data corresponds to expression set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>patchwork result for hub genes boxplot and survival plot
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exp_boxplot">exp_boxplot</a></code>;<code><a href="#topic+exp_surv">exp_surv</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(requireNamespace("ggpubr",quietly = TRUE)) {
k = box_surv(log2(exp_hub1+1),exprSet_hub1,meta1);k[[1]]
}else{
 warning("Package 'ggpubr' needed for this function to work.
        Please install it by install.packages('ggpubr')")
}
</code></pre>

<hr>
<h2 id='cod'>cod</h2><span id='topic+cod'></span>

<h3>Description</h3>

<p>An expression matrix form TCGA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cod
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 100 rows and 512 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  cod
</code></pre>

<hr>
<h2 id='cor.full'>cor.test for all variables</h2><span id='topic+cor.full'></span>

<h3>Description</h3>

<p>cor.test for all variables(each two columns)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cor.full(x, drop = min(x) - 0.001, min.obs = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cor.full_+3A_x">x</code></td>
<td>
<p>A numeric matrix or data.frame</p>
</td></tr>
<tr><td><code id="cor.full_+3A_drop">drop</code></td>
<td>
<p>drop values</p>
</td></tr>
<tr><td><code id="cor.full_+3A_min.obs">min.obs</code></td>
<td>
<p>minimum number of observations  after dropping</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with cor.test p.value and estimate
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cor.one">cor.one</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = iris[,-5]
cor.full(x)
</code></pre>

<hr>
<h2 id='cor.one'>cor.test for one variable with all variables</h2><span id='topic+cor.one'></span>

<h3>Description</h3>

<p>cor.test for all variables(each two columns)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cor.one(
  x,
  var,
  drop.var = min(x[, var]) - 0.001,
  drop.other = min(x[, -which(colnames(x) == var)]) - 0.001,
  min.obs = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cor.one_+3A_x">x</code></td>
<td>
<p>A numeric matrix or data.frame</p>
</td></tr>
<tr><td><code id="cor.one_+3A_var">var</code></td>
<td>
<p>your chosen variable,only one.</p>
</td></tr>
<tr><td><code id="cor.one_+3A_drop.var">drop.var</code></td>
<td>
<p>drop values in var</p>
</td></tr>
<tr><td><code id="cor.one_+3A_drop.other">drop.other</code></td>
<td>
<p>drop values in other columns</p>
</td></tr>
<tr><td><code id="cor.one_+3A_min.obs">min.obs</code></td>
<td>
<p>minimum number of observations  after dropping</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with cor.test p.value and estimate
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cor.full">cor.full</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = iris[,-5]
cor.one(x,"Sepal.Width")
</code></pre>

<hr>
<h2 id='corheatmap'>corheatmap</h2><span id='topic+corheatmap'></span>

<h3>Description</h3>

<p>draw cor heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corheatmap(exp, x, y, color = c("#2fa1dd", "white", "#f87669"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corheatmap_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="corheatmap_+3A_x">x</code></td>
<td>
<p>genes or cells from exp</p>
</td></tr>
<tr><td><code id="corheatmap_+3A_y">y</code></td>
<td>
<p>genes or cells from exp</p>
</td></tr>
<tr><td><code id="corheatmap_+3A_color">color</code></td>
<td>
<p>color for heatmap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = rownames(exprSet_hub1)[1:3]
y = rownames(exprSet_hub1)[4:7]
corheatmap(exprSet_hub1,x,y)
</code></pre>

<hr>
<h2 id='corscatterplot'>corscatterplot</h2><span id='topic+corscatterplot'></span>

<h3>Description</h3>

<p>draw cor scatter plot with density plot by ggplot2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corscatterplot(
  dat,
  x,
  y,
  color_cor = "blue",
  fill_cor = "lightgray",
  fill_x = "#ff820e",
  fill_y = "#0000fe",
  type = "density",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corscatterplot_+3A_dat">dat</code></td>
<td>
<p>plot data</p>
</td></tr>
<tr><td><code id="corscatterplot_+3A_x">x</code></td>
<td>
<p>x</p>
</td></tr>
<tr><td><code id="corscatterplot_+3A_y">y</code></td>
<td>
<p>y</p>
</td></tr>
<tr><td><code id="corscatterplot_+3A_color_cor">color_cor</code></td>
<td>
<p>color for cor reg.line</p>
</td></tr>
<tr><td><code id="corscatterplot_+3A_fill_cor">fill_cor</code></td>
<td>
<p>fill for cor reg.line</p>
</td></tr>
<tr><td><code id="corscatterplot_+3A_fill_x">fill_x</code></td>
<td>
<p>fill for top density plot</p>
</td></tr>
<tr><td><code id="corscatterplot_+3A_fill_y">fill_y</code></td>
<td>
<p>fill for right density plot</p>
</td></tr>
<tr><td><code id="corscatterplot_+3A_type">type</code></td>
<td>
<p>whether to use a density plot or a histogram plot for the side panel.</p>
</td></tr>
<tr><td><code id="corscatterplot_+3A_...">...</code></td>
<td>
<p>other paramters for ggscatter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(requireNamespace("ggpubr",quietly = TRUE)){
corscatterplot(iris,"Sepal.Length","Sepal.Width")
}else{
  warning("Package 'ggpubr' needed for this function to work.
           Please install it by install.packages('ggpubr')")
}
</code></pre>

<hr>
<h2 id='deg'>deg</h2><span id='topic+deg'></span>

<h3>Description</h3>

<p>limma differential analysis result for GSE42872
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deg
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 18591 rows and 10 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(deg)
</code></pre>

<hr>
<h2 id='deseq_data'>deseq_data</h2><span id='topic+deseq_data'></span>

<h3>Description</h3>

<p>DEseq2 differential analysis result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deseq_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 552 rows and 6 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(deseq_data)
</code></pre>

<hr>
<h2 id='double_enrich'>draw enrichment bar plots for both up and down genes</h2><span id='topic+double_enrich'></span>

<h3>Description</h3>

<p>draw enrichment bar plots for both up and down genes,for human only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>double_enrich(deg, n = 10, color = c("#2874C5", "#f87669"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="double_enrich_+3A_deg">deg</code></td>
<td>
<p>a data.frame contains at least two columns:&quot;ENTREZID&quot; and &quot;change&quot;</p>
</td></tr>
<tr><td><code id="double_enrich_+3A_n">n</code></td>
<td>
<p>how many terms will you perform for up and down genes respectively</p>
</td></tr>
<tr><td><code id="double_enrich_+3A_color">color</code></td>
<td>
<p>color for bar plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with kegg and go bar plot according to up and down genes enrichment result.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+quick_enrich">quick_enrich</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("org.Hs.eg.db",quietly = TRUE)&amp;
   requireNamespace("labeling",quietly = TRUE)){
   double_enrich(deg)
}else{
  if(!requireNamespace("org.Hs.eg.db",quietly = TRUE)) {
    warning("Package 'org.Hs.eg.db' needed for this function to work.
        Please install it by BiocManager::install('org.Hs.eg.db')",call. = FALSE)
  }
  if(!requireNamespace("labeling",quietly = TRUE)) {
    warning("Package 'labeling' needed for this function to work.
        Please install it by install.packages('labeling')",call. = FALSE)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='draw_boxplot'>draw boxplot for expression</h2><span id='topic+draw_boxplot'></span>

<h3>Description</h3>

<p>draw boxplot for expression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_boxplot(
  exp,
  group_list,
  method = "kruskal.test",
  sort = TRUE,
  drop = FALSE,
  width = 0.5,
  pvalue_cutoff = 0.05,
  xlab = "Gene",
  ylab = "Expression",
  grouplab = "Group",
  p.label = FALSE,
  add_error_bar = FALSE,
  color = c("#2874C5", "#f87669", "#e6b707", "#868686", "#66C2A5", "#FC8D62", "#8DA0CB",
    "#E78AC3", "#A6D854", "#FFD92F", "#E5C494", "#B3B3B3"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_boxplot_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_method">method</code></td>
<td>
<p>one of kruskal.test,aov,t.test and wilcox.test</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_sort">sort</code></td>
<td>
<p>whether the boxplot will be sorted</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_drop">drop</code></td>
<td>
<p>whether to discard insignificant values</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_width">width</code></td>
<td>
<p>width of boxplot and error bar</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>if drop = TRUE,genes with p-values below the threshold will be drawn</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_xlab">xlab</code></td>
<td>
<p>title of the x axis</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_ylab">ylab</code></td>
<td>
<p>title of the y axis</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_grouplab">grouplab</code></td>
<td>
<p>title of group legend</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_p.label">p.label</code></td>
<td>
<p>whether to show p value in the plot</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_add_error_bar">add_error_bar</code></td>
<td>
<p>whether to add error bar</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_color">color</code></td>
<td>
<p>color vector</p>
</td></tr>
<tr><td><code id="draw_boxplot_+3A_...">...</code></td>
<td>
<p>other parameters from stat_compare_means</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a boxplot according to <code>exp</code> and grouped by <code>group</code>.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw_heatmap">draw_heatmap</a></code>;<code><a href="#topic+draw_volcano">draw_volcano</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(requireNamespace("tidyr",quietly = TRUE)&amp;
   requireNamespace("ggpubr",quietly = TRUE)){
  draw_boxplot(t(iris[,1:4]),iris$Species)
  exp &lt;-  matrix(rnorm(60),nrow = 10)
  colnames(exp) &lt;- paste0("sample",1:6)
  rownames(exp) &lt;- paste0("gene",1:10)
  exp[,4:6] = exp[,4:6] +10
  exp[1:4,1:4]
  group_list &lt;- factor(rep(c("A","B"),each = 3))
  draw_boxplot(exp,group_list)
  draw_boxplot(exp,group_list,color = c("grey","red"))
  }else{
if(!requireNamespace("ggpubr",quietly = TRUE)) {
 warning("Package 'ggpubr' needed for this function to work.
        Please install it by install.packages('ggpubr')")
}
if(!requireNamespace("tidyr",quietly = TRUE)) {
 warning("Package 'tidyr' needed for this function to work.
        Please install it by install.packages('tidyr')")
}
}
</code></pre>

<hr>
<h2 id='draw_heatmap'>draw a heatmap plot</h2><span id='topic+draw_heatmap'></span>

<h3>Description</h3>

<p>warning a heatmap plot for expression matrix and group by group_list praramter, exp will be scaled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_heatmap(
  n,
  group_list,
  scale_before = FALSE,
  n_cutoff = 3,
  legend = FALSE,
  show_rownames = FALSE,
  annotation_legend = FALSE,
  split_column = FALSE,
  show_column_title = FALSE,
  color = (grDevices::colorRampPalette(c("#2fa1dd", "white", "#f87669")))(100),
  color_an = c("#2fa1dd", "#f87669", "#e6b707", "#868686", "#92C5DE", "#F4A582",
    "#66C2A5", "#FC8D62", "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F", "#E5C494",
    "#B3B3B3"),
  scale = TRUE,
  main = NA,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_heatmap_+3A_n">n</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_scale_before">scale_before</code></td>
<td>
<p>deprecated parameter</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_n_cutoff">n_cutoff</code></td>
<td>
<p>3 by defalut , scale before plot and set a cutoff,usually 2 or 1.6</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_legend">legend</code></td>
<td>
<p>logical,show legend or not</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_show_rownames">show_rownames</code></td>
<td>
<p>logical,show rownames or not</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_annotation_legend">annotation_legend</code></td>
<td>
<p>logical,show annotation legend or not</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_split_column">split_column</code></td>
<td>
<p>split column by group_list</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_show_column_title">show_column_title</code></td>
<td>
<p>show column title or not</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_color">color</code></td>
<td>
<p>color for heatmap</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_color_an">color_an</code></td>
<td>
<p>color for column annotation</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_scale">scale</code></td>
<td>
<p>logical,scale the matrix or not</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_main">main</code></td>
<td>
<p>the title of the plot</p>
</td></tr>
<tr><td><code id="draw_heatmap_+3A_...">...</code></td>
<td>
<p>other parameters from pheatmap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a heatmap plot according to <code>exp</code> and grouped by <code>group</code>.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw_pca">draw_pca</a></code>;<code><a href="#topic+draw_volcano">draw_volcano</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#example data
exp = matrix(abs(rnorm(60,sd = 16)),nrow = 10)
exp[,4:6] &lt;- exp[,4:6]+20
colnames(exp) &lt;- paste0("sample",1:6)
rownames(exp) &lt;- paste0("gene",1:10)
exp[1:4,1:4]
group_list = factor(rep(c("A","B"),each = 3))
if(requireNamespace("ggplotify",quietly = TRUE)){
  draw_heatmap(exp,group_list)
  #use iris
  n = t(iris[,1:4]);colnames(n) = 1:150
  group_list = iris$Species
  draw_heatmap(n,group_list)
  draw_heatmap(n,group_list,color = colorRampPalette(c("green","black","red"))(100),
               color_an = c("red","blue","pink") )
}else{
  warning("Package 'ggplotify' needed for this function to work.
    Please install it by install.packages('ggplotify')")
}
</code></pre>

<hr>
<h2 id='draw_heatmap2'>draw heatmap plots</h2><span id='topic+draw_heatmap2'></span>

<h3>Description</h3>

<p>print heatmap plots for expression matrix and group by group_list paramter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_heatmap2(exp, group_list, deg, my_genes = NULL, heat_union = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_heatmap2_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="draw_heatmap2_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="draw_heatmap2_+3A_deg">deg</code></td>
<td>
<p>a data.frame created by Differential analysis</p>
</td></tr>
<tr><td><code id="draw_heatmap2_+3A_my_genes">my_genes</code></td>
<td>
<p>genes for pheatmap</p>
</td></tr>
<tr><td><code id="draw_heatmap2_+3A_heat_union">heat_union</code></td>
<td>
<p>logical ,use union or intersect DEGs for heatmap</p>
</td></tr>
<tr><td><code id="draw_heatmap2_+3A_...">...</code></td>
<td>
<p>other parameters from draw_heatmap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a heatmap plot according to <code>exp</code> and grouped by <code>group</code>.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw_pca">draw_pca</a></code>;<code><a href="#topic+draw_volcano">draw_volcano</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("Biobase",quietly = TRUE)&amp;
   requireNamespace("AnnoProbe",quietly = TRUE)){
  gse = "GSE474"
  geo = geo_download(gse,destdir=tempdir())
  geo$exp[1:4,1:4]
  geo$exp=log2(geo$exp+1)
  group_list=ifelse(stringr::str_detect(geo$pd$title,"MObese"),"MObese",
  ifelse(stringr::str_detect(geo$pd$title,"NonObese"),"NonObese","Obese"))
  group_list=factor(group_list,levels = c("NonObese","Obese","MObese"))
  find_anno(geo$gpl)
  ids &lt;- AnnoProbe::idmap(geo$gpl,destdir = tempdir())
  deg = multi_deg(geo$exp,group_list,ids,adjust = FALSE,entriz = FALSE)
  draw_heatmap2(geo$exp,group_list,deg)
}else{
  if(!requireNamespace("AnnoProbe",quietly = TRUE)) {
    warning("Package 'AnnoProbe' needed for this function to work.
         Please install it by install.packages('AnnoProbe')",call. = FALSE)
  }
  if(!requireNamespace("Biobase",quietly = TRUE)) {
    warning("Package 'Biobase' needed for this function to work.
         Please install it by BiocManager::install('Biobase')",call. = FALSE)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='draw_KM'>draw_KM</h2><span id='topic+draw_KM'></span>

<h3>Description</h3>

<p>draw KM-plot with two or more group
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_KM(
  meta,
  group_list,
  time_col = "time",
  event_col = "event",
  legend.title = "Group",
  legend.labs = levels(group_list),
  color = c("#2874C5", "#f87669", "#e6b707", "#868686", "#92C5DE", "#F4A582", "#66C2A5",
    "#FC8D62", "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F", "#E5C494", "#B3B3B3"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_KM_+3A_meta">meta</code></td>
<td>
<p>survival data with time and event column</p>
</td></tr>
<tr><td><code id="draw_KM_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="draw_KM_+3A_time_col">time_col</code></td>
<td>
<p>colname of time</p>
</td></tr>
<tr><td><code id="draw_KM_+3A_event_col">event_col</code></td>
<td>
<p>colname of event</p>
</td></tr>
<tr><td><code id="draw_KM_+3A_legend.title">legend.title</code></td>
<td>
<p>legend title</p>
</td></tr>
<tr><td><code id="draw_KM_+3A_legend.labs">legend.labs</code></td>
<td>
<p>character vector specifying legend labels</p>
</td></tr>
<tr><td><code id="draw_KM_+3A_color">color</code></td>
<td>
<p>color vector</p>
</td></tr>
<tr><td><code id="draw_KM_+3A_...">...</code></td>
<td>
<p>other parameters from ggsurvplot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a KM-plot
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("survival")
x = survival::lung
draw_KM(meta = x,
       group_list = x$sex,event_col = "status")
</code></pre>

<hr>
<h2 id='draw_pca'>draw PCA plots</h2><span id='topic+draw_pca'></span>

<h3>Description</h3>

<p>do PCA analysis and warning a PCA plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_pca(
  exp,
  group_list,
  color = c("#2874C5", "#f87669", "#e6b707", "#868686", "#92C5DE", "#F4A582", "#66C2A5",
    "#FC8D62", "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F", "#E5C494", "#B3B3B3"),
  addEllipses = TRUE,
  style = "default",
  color.label = "Group",
  title = "",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_pca_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="draw_pca_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="draw_pca_+3A_color">color</code></td>
<td>
<p>color vector</p>
</td></tr>
<tr><td><code id="draw_pca_+3A_addellipses">addEllipses</code></td>
<td>
<p>logical,add ellipses or not</p>
</td></tr>
<tr><td><code id="draw_pca_+3A_style">style</code></td>
<td>
<p>plot style,&quot;default&quot;,&quot;ggplot2&quot;and &quot;3D&quot;</p>
</td></tr>
<tr><td><code id="draw_pca_+3A_color.label">color.label</code></td>
<td>
<p>color legend label</p>
</td></tr>
<tr><td><code id="draw_pca_+3A_title">title</code></td>
<td>
<p>plot title</p>
</td></tr>
<tr><td><code id="draw_pca_+3A_...">...</code></td>
<td>
<p>other paramters from fviz_pca_ind</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a pca plot according to <code>exp</code> and grouped by <code>group</code>.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw_heatmap">draw_heatmap</a></code>;<code><a href="#topic+draw_volcano">draw_volcano</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(requireNamespace("FactoMineR",quietly = TRUE)&amp;
   requireNamespace("factoextra",quietly = TRUE)){
  draw_pca(t(iris[,1:4]),iris$Species)
  draw_pca(t(iris[,1:4]),iris$Species,style = "ggplot2")
  #change color
  draw_pca(t(iris[,1:4]),iris$Species,color = c("#E78AC3", "#A6D854", "#FFD92F"))
}else{
  if(!requireNamespace("FactoMineR",quietly = TRUE)){
    warning("Package 'FactoMineR' needed for this function to work.
         Please install it by install.packages('FactoMineR')")
  }
  if(!requireNamespace("factoextra",quietly = TRUE)){
    warning("Package 'factoextra' needed for this function to work.
         Please install it by install.packages('factoextra')")
  }
}

if(requireNamespace("scatterplot3d",quietly = TRUE)&amp;
   requireNamespace("FactoMineR",quietly = TRUE)){
  draw_pca(t(iris[,1:4]),iris$Species,style = "3D")
}else{
  if(!requireNamespace("scatterplot3d",quietly = TRUE)){
    warning("Package 'scatterplot3d' needed for this function to work.
         Please install it by install.packages('scatterplot3d')")
  }
  if(!requireNamespace("FactoMineR",quietly = TRUE)){
    warning("Package 'FactoMineR' needed for this function to work.
         Please install it by install.packages('FactoMineR')")
  }
}
</code></pre>

<hr>
<h2 id='draw_tsne'>draw_tsne</h2><span id='topic+draw_tsne'></span>

<h3>Description</h3>

<p>draw tsne plot with annotation by ggplot2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_tsne(
  exp,
  group_list,
  perplexity = 30,
  color = c("#2874C5", "#f87669", "#e6b707", "#868686", "#92C5DE", "#F4A582", "#66C2A5",
    "#FC8D62", "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F", "#E5C494", "#B3B3B3"),
  color.label = "group",
  addEllipses = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_tsne_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="draw_tsne_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="draw_tsne_+3A_perplexity">perplexity</code></td>
<td>
<p>numeric; perplexity parameter for Rtsne</p>
</td></tr>
<tr><td><code id="draw_tsne_+3A_color">color</code></td>
<td>
<p>color vector</p>
</td></tr>
<tr><td><code id="draw_tsne_+3A_color.label">color.label</code></td>
<td>
<p>color legend label</p>
</td></tr>
<tr><td><code id="draw_tsne_+3A_addellipses">addEllipses</code></td>
<td>
<p>logical,add ellipses or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp &lt;-  matrix(rnorm(10000),nrow = 50)
colnames(exp) &lt;- paste0("sample",1:200)
rownames(exp) &lt;- paste0("gene",1:50)
exp[1:4,1:4]
exp[,1:100] = exp[,1:100]+10
group_list &lt;- factor(rep(c("A","B"),each = 100))
if(requireNamespace("Rtsne",quietly = TRUE)){
draw_tsne(exp,group_list)
}else{
 warning("Package 'Rtsne' needed for this function to work.
        Please install it by install.packages('Rtsne')")
}
</code></pre>

<hr>
<h2 id='draw_venn'>draw a venn plot</h2><span id='topic+draw_venn'></span>

<h3>Description</h3>

<p>warning a venn plot for deg result created by three packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_venn(
  x,
  main,
  color = c("#2874C5", "#f87669", "#e6b707", "#868686", "#66C2A5", "#FC8D62", "#8DA0CB",
    "#E78AC3", "#A6D854", "#FFD92F", "#E5C494", "#B3B3B3"),
  imagetype = "png",
  filename = NULL,
  lwd = 1,
  lty = 1,
  col = color[1:length(x)],
  fill = color[1:length(x)],
  cat.col = color[1:length(x)],
  cat.cex = 1,
  cat.dist = -0.15,
  rotation.degree = 0,
  main.cex = 1,
  cex = 1,
  alpha = 0.1,
  reverse = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_venn_+3A_x">x</code></td>
<td>
<p>a list for plot</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_main">main</code></td>
<td>
<p>Character giving the main title of the diagram</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_color">color</code></td>
<td>
<p>color vector</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_imagetype">imagetype</code></td>
<td>
<p>Specification of the image format (e.g. tiff, png or svg)</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_filename">filename</code></td>
<td>
<p>Filename for image output, or if NULL returns the grid object itself</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_lwd">lwd</code></td>
<td>
<p>width of the circle's circumference</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_lty">lty</code></td>
<td>
<p>dash pattern of the circle's circumference</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_col">col</code></td>
<td>
<p>Colour of the circle's circumference</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_fill">fill</code></td>
<td>
<p>Colour of the circle's area</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_cat.col">cat.col</code></td>
<td>
<p>Colour of the category name</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_cat.cex">cat.cex</code></td>
<td>
<p>size of the category name</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_cat.dist">cat.dist</code></td>
<td>
<p>The distance (in npc units) of the category name from the edge of the circle (can be negative)</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_rotation.degree">rotation.degree</code></td>
<td>
<p>Number of degrees to rotate the entire diagram</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_main.cex">main.cex</code></td>
<td>
<p>Number giving the cex (font size) of the main title</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_cex">cex</code></td>
<td>
<p>size of the area label</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_alpha">alpha</code></td>
<td>
<p>Alpha transparency of the circle's area</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_reverse">reverse</code></td>
<td>
<p>logical,reflect the three-set Venn diagram along its central
vertical axis of symmetry. Use in combination with rotation
to generate all possible set orders</p>
</td></tr>
<tr><td><code id="draw_venn_+3A_...">...</code></td>
<td>
<p>other parameters from venn.diagram</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a venn plot according to <code>x</code>, <code>y</code> and.<code>z</code> named &quot;name&quot; paramter
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw_pca">draw_pca</a></code>;<code><a href="#topic+draw_volcano">draw_volcano</a></code>;<code><a href="#topic+draw_heatmap">draw_heatmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(requireNamespace("VennDiagram",quietly = TRUE)&amp;
  requireNamespace("ggplotify",quietly = TRUE)&amp;
  requireNamespace("cowplot",quietly = TRUE)){
 x = list(Deseq2=sample(1:100,30),edgeR = sample(1:100,30),limma = sample(1:100,30))
 draw_venn(x,"test")
 draw_venn(x,"test",color = c("darkgreen", "darkblue", "#B2182B"))
}else{
 if(!requireNamespace("VennDiagram",quietly = TRUE)) {
   warning("Package 'VennDiagram' needed for this function to work.
   Please install it by install.packages('VennDiagram')")
 }
 if(!requireNamespace("ggplotify",quietly = TRUE)) {
   warning("Package 'ggplotify' needed for this function to work.
   Please install it by install.packages('ggplotify')")
 }
 if(!requireNamespace("cowplot",quietly = TRUE)) {
   warning("Package 'cowplot' needed for this function to work.
   Please install it by install.packages('cowplot')")
 }
}
</code></pre>

<hr>
<h2 id='draw_volcano'>draw a volcano plot</h2><span id='topic+draw_volcano'></span>

<h3>Description</h3>

<p>warning a volcano plot for Differential analysis result in data.frame format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_volcano(
  deg,
  lab = NA,
  xlab.package = TRUE,
  pvalue_cutoff = 0.05,
  logFC_cutoff = 1,
  pkg = 1,
  adjust = FALSE,
  symmetry = FALSE,
  color = c("#2874C5", "grey", "#f87669")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_volcano_+3A_deg">deg</code></td>
<td>
<p>a data.frame created by Differential analysis</p>
</td></tr>
<tr><td><code id="draw_volcano_+3A_lab">lab</code></td>
<td>
<p>label for  x axis in volcano plot, if NA , x axis names by package</p>
</td></tr>
<tr><td><code id="draw_volcano_+3A_xlab.package">xlab.package</code></td>
<td>
<p>whether to use the package name as the x axis name</p>
</td></tr>
<tr><td><code id="draw_volcano_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Cutoff value of pvalue,0.05 by default.</p>
</td></tr>
<tr><td><code id="draw_volcano_+3A_logfc_cutoff">logFC_cutoff</code></td>
<td>
<p>Cutoff value of logFC,1 by default.</p>
</td></tr>
<tr><td><code id="draw_volcano_+3A_pkg">pkg</code></td>
<td>
<p>a integer ,means which Differential analysis packages you used,we support three packages by now, 1,2,3,4 respectively means &quot;DESeq2&quot;,&quot;edgeR&quot;,&quot;limma(voom)&quot;,&quot;limma&quot;</p>
</td></tr>
<tr><td><code id="draw_volcano_+3A_adjust">adjust</code></td>
<td>
<p>a logical value, would you like to use adjusted pvalue to draw this plot,FAlSE by default.</p>
</td></tr>
<tr><td><code id="draw_volcano_+3A_symmetry">symmetry</code></td>
<td>
<p>a logical value ,would you like to get your plot symmetrical</p>
</td></tr>
<tr><td><code id="draw_volcano_+3A_color">color</code></td>
<td>
<p>color vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a volcano plot according to logFC and P.value(or adjust P.value)
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+draw_heatmap">draw_heatmap</a></code>;<code><a href="#topic+draw_pca">draw_pca</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(deseq_data)
draw_volcano(deseq_data)
draw_volcano(deseq_data,pvalue_cutoff = 0.01,logFC_cutoff = 2)
draw_volcano(deseq_data,color = c("darkgreen", "darkgrey", "#B2182B"))
</code></pre>

<hr>
<h2 id='draw_volcano2'>draw_volcano2</h2><span id='topic+draw_volcano2'></span>

<h3>Description</h3>

<p>print one or more volcano plot for Differential analysis result in data.frame fomat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_volcano2(deg, pkg = 4, lab, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_volcano2_+3A_deg">deg</code></td>
<td>
<p>a data.frame created by Differential analysis</p>
</td></tr>
<tr><td><code id="draw_volcano2_+3A_pkg">pkg</code></td>
<td>
<p>a integer ,means which Differential analysis packages you used,we support three packages by now, 1,2,3,4 respectively means &quot;DESeq2&quot;,&quot;edgeR&quot;,&quot;limma(voom)&quot;,&quot;limma&quot;</p>
</td></tr>
<tr><td><code id="draw_volcano2_+3A_lab">lab</code></td>
<td>
<p>label for  x axis in volcano plot, if NA , x axis names by package</p>
</td></tr>
<tr><td><code id="draw_volcano2_+3A_...">...</code></td>
<td>
<p>other parameters from draw_volcano</p>
</td></tr>
</table>


<h3>Value</h3>

<p>one or more volcano plot
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geo_download">geo_download</a></code>;<code><a href="#topic+draw_volcano">draw_volcano</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("Biobase",quietly = TRUE)&amp;
   requireNamespace("AnnoProbe",quietly = TRUE)){
#two group
gse = "GSE42872"
geo = geo_download(gse,destdir=tempdir())
group_list = rep(c("A","B"),each = 3)
ids = AnnoProbe::idmap('GPL6244',destdir = tempdir())
deg = get_deg(geo$exp,group_list,ids)
draw_volcano2(deg)
#multigroup
gse = "GSE474"
geo = geo_download(gse,destdir=tempdir())
geo$exp[1:4,1:4]
geo$exp=log2(geo$exp+1)
group_list=ifelse(stringr::str_detect(geo$pd$title,"MObese"),"MObese",
ifelse(stringr::str_detect(geo$pd$title,"NonObese"),"NonObese","Obese"))
group_list=factor(group_list,levels = c("NonObese","Obese","MObese"))
find_anno(geo$gpl)
ids &lt;- AnnoProbe::idmap(geo$gpl,destdir = tempdir())
deg = multi_deg(geo$exp,group_list,ids,adjust = FALSE,entriz = FALSE)
draw_volcano2(deg)
draw_volcano2(deg,color = c("darkgreen","grey","darkred"))
}else{
  if(!requireNamespace("AnnoProbe",quietly = TRUE)) {
    warning("Package 'AnnoProbe' needed for this function to work.
         Please install it by install.packages('AnnoProbe')",call. = FALSE)
  }
  if(!requireNamespace("Biobase",quietly = TRUE)) {
    warning("Package 'Biobase' needed for this function to work.
         Please install it by BiocManager::install('Biobase')",call. = FALSE)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='dumd'>count unique values in every columns for data.frame</h2><span id='topic+dumd'></span>

<h3>Description</h3>

<p>in geo analysis,this function can help you simplify pdata, delete columns with unique values,which can't be used as group vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dumd(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dumd_+3A_x">x</code></td>
<td>
<p>A data.frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The simple data.frame of columns unique values count in <code>x</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dumd(iris)
data(ToothGrowth)
x = ToothGrowth
dumd(ToothGrowth)
</code></pre>

<hr>
<h2 id='edges_to_nodes'>edges_to_nodes</h2><span id='topic+edges_to_nodes'></span>

<h3>Description</h3>

<p>get nodes from edges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edges_to_nodes(edges)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edges_to_nodes_+3A_edges">edges</code></td>
<td>
<p>data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nodes data.frame
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+interaction_to_edges">interaction_to_edges</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df = data.frame(a = c("gene1","gene2","gene3"),
b = c("d,f,a,b",
"c,e,g",
"a,b,d"))
edges = interaction_to_edges(df)
nodes = edges_to_nodes(edges)
</code></pre>

<hr>
<h2 id='exists_anno_list'>exists_anno_list</h2><span id='topic+exists_anno_list'></span>

<h3>Description</h3>

<p>AnnoProbe supported  GPLs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exists_anno_list
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 175.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  exists_anno_list
</code></pre>

<hr>
<h2 id='exp_boxplot'>exp_boxplot</h2><span id='topic+exp_boxplot'></span>

<h3>Description</h3>

<p>draw box plot for a hub gene expression matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp_boxplot(exp_hub, color = c("#2fa1dd", "#f87669"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exp_boxplot_+3A_exp_hub">exp_hub</code></td>
<td>
<p>an expression matrix for hubgenes</p>
</td></tr>
<tr><td><code id="exp_boxplot_+3A_color">color</code></td>
<td>
<p>color for boxplot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>box plots list for all genes in the matrix
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exp_surv">exp_surv</a></code>;<code><a href="#topic+box_surv">box_surv</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(requireNamespace("ggpubr",quietly = TRUE)) {
k = exp_boxplot(log2(exp_hub1+1));k[[1]]
}else{
 warning("Package 'ggpubr' needed for this function to work.
        Please install it by install.packages('ggpubr')")
}
</code></pre>

<hr>
<h2 id='exp_hub1'>exp_hub1</h2><span id='topic+exp_hub1'></span>

<h3>Description</h3>

<p>An expression matrix from TCGA and Gtex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp_hub1
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 8 rows and 350 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  exp_hub1[1:4,1:4]
</code></pre>

<hr>
<h2 id='exp_surv'>exp_surv</h2><span id='topic+exp_surv'></span>

<h3>Description</h3>

<p>draw surv plot for a hub gene expression matrix for tumor samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp_surv(exprSet_hub, meta, cut.point = FALSE, color = c("#2874C5", "#f87669"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exp_surv_+3A_exprset_hub">exprSet_hub</code></td>
<td>
<p>a tumor expression set for hubgenes</p>
</td></tr>
<tr><td><code id="exp_surv_+3A_meta">meta</code></td>
<td>
<p>meta data corresponds to expression set</p>
</td></tr>
<tr><td><code id="exp_surv_+3A_cut.point">cut.point</code></td>
<td>
<p>logical , use cut_point or not, if FALSE,use median by defult</p>
</td></tr>
<tr><td><code id="exp_surv_+3A_color">color</code></td>
<td>
<p>color for boxplot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>survival plots list for all genes
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exp_boxplot">exp_boxplot</a></code>;<code><a href="#topic+box_surv">box_surv</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = exp_surv(exprSet_hub1,meta1)
patchwork::wrap_plots(tmp)+patchwork::plot_layout(guides = "collect")
tmp2 = exp_surv(exprSet_hub1,meta1,cut.point = TRUE)
patchwork::wrap_plots(tmp2)+patchwork::plot_layout(guides = "collect")
</code></pre>

<hr>
<h2 id='exprSet_hub1'>exprSet_hub1</h2><span id='topic+exprSet_hub1'></span>

<h3>Description</h3>

<p>An cpm expression matrix from TCGA,tumor samples only
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exprSet_hub1
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 8 rows and 177 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  exprSet_hub1[1:4,1:4]
</code></pre>

<hr>
<h2 id='find_anno'>find annotation package or files</h2><span id='topic+find_anno'></span>

<h3>Description</h3>

<p>find gpl annotation package or files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_anno(gpl, install = FALSE, update = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_anno_+3A_gpl">gpl</code></td>
<td>
<p>a gpl accession</p>
</td></tr>
<tr><td><code id="find_anno_+3A_install">install</code></td>
<td>
<p>whether to install and library the package</p>
</td></tr>
<tr><td><code id="find_anno_+3A_update">update</code></td>
<td>
<p>whether to update the package</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with deg data.frame, volcano plot and a list with DEGs.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geo_download">geo_download</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>find_anno("GPL570")
</code></pre>

<hr>
<h2 id='genes'>genes</h2><span id='topic+genes'></span>

<h3>Description</h3>

<p>some gene entriz ids
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genes
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 511.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  genes
</code></pre>

<hr>
<h2 id='geo_download'>geo_download</h2><span id='topic+geo_download'></span>

<h3>Description</h3>

<p>download gse data and get informations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geo_download(
  gse,
  by_annopbrobe = TRUE,
  simpd = TRUE,
  colon_remove = FALSE,
  destdir = getwd(),
  n = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geo_download_+3A_gse">gse</code></td>
<td>
<p>gse assession number</p>
</td></tr>
<tr><td><code id="geo_download_+3A_by_annopbrobe">by_annopbrobe</code></td>
<td>
<p>download data by geoquery or annoprobe</p>
</td></tr>
<tr><td><code id="geo_download_+3A_simpd">simpd</code></td>
<td>
<p>get simplified pdata,drop out columns with all same values</p>
</td></tr>
<tr><td><code id="geo_download_+3A_colon_remove">colon_remove</code></td>
<td>
<p>whether to remove duplicated columns with colons</p>
</td></tr>
<tr><td><code id="geo_download_+3A_destdir">destdir</code></td>
<td>
<p>The destination directory for data downloads.</p>
</td></tr>
<tr><td><code id="geo_download_+3A_n">n</code></td>
<td>
<p>For data with more than one ExpressionSet, specify which one to analyze</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with exp,pd and gpl
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+find_anno">find_anno</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("Biobase",quietly = TRUE)&amp;
   requireNamespace("AnnoProbe",quietly = TRUE)){
  gse = "GSE42872"
  a = geo_download(gse,destdir=tempdir())
}else{
  if(!requireNamespace("AnnoProbe",quietly = TRUE)) {
    print("Package 'AnnoProbe' needed for this function to work.
         Please install it by install.packages('AnnoProbe')"print)
  }
  if(!requireNamespace("Biobase",quietly = TRUE)) {
    print("Package 'Biobase' needed for this function to work.
         Please install it by BiocManager::install('Biobase')"print)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='get_cgs'>get_cgs</h2><span id='topic+get_cgs'></span>

<h3>Description</h3>

<p>extract DEGs from deg data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cgs(deg)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cgs_+3A_deg">deg</code></td>
<td>
<p>a data.frame created by Differential analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with upgenes,downgenes,diffgenes.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geo_download">geo_download</a></code>;<code><a href="#topic+draw_volcano">draw_volcano</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#two group
gse = "GSE42872"
geo = geo_download(gse,destdir=tempdir())
group_list = rep(c("A","B"),each = 3)
ids = AnnoProbe::idmap('GPL6244',destdir=tempdir())
deg = get_deg(geo$exp,group_list,ids)
cgs = get_cgs(deg)
#mutigroup
gse = "GSE474"
geo = geo_download(gse,destdir=tempdir())
geo$exp[1:4,1:4]
geo$exp=log2(geo$exp+1)
group_list=ifelse(stringr::str_detect(geo$pd$title,"MObese"),"MObese",
ifelse(stringr::str_detect(geo$pd$title,"NonObese"),"NonObese","Obese"))
group_list=factor(group_list,levels = c("NonObese","Obese","MObese"))
find_anno(geo$gpl)
ids = AnnoProbe::idmap(geo$gpl,destdir = tempdir())
deg = multi_deg(geo$exp,group_list,ids,adjust = FALSE)
cgs = get_cgs(deg)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_count_txt'>get count from GEO</h2><span id='topic+get_count_txt'></span>

<h3>Description</h3>

<p>get RNA-seq count file from GEO database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_count_txt(gse, destdir = getwd(), download = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_count_txt_+3A_gse">gse</code></td>
<td>
<p>gse assession number</p>
</td></tr>
<tr><td><code id="get_count_txt_+3A_destdir">destdir</code></td>
<td>
<p>The destination directory for data downloads.</p>
</td></tr>
<tr><td><code id="get_count_txt_+3A_download">download</code></td>
<td>
<p>download the txt file or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with deg data.frame, volcano plot and a list with DEGs.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geo_download">geo_download</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_count_txt("GSE162550",destdir = tempdir())
</code></pre>

<hr>
<h2 id='get_deg'>get_deg</h2><span id='topic+get_deg'></span>

<h3>Description</h3>

<p>do differential analysis according to expression set and group information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_deg(
  exp,
  group_list,
  ids,
  logFC_cutoff = 1,
  pvalue_cutoff = 0.05,
  adjust = FALSE,
  entriz = TRUE,
  species = "human"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_deg_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="get_deg_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="get_deg_+3A_ids">ids</code></td>
<td>
<p>a data.frame with 2 columns,including probe_id and symbol</p>
</td></tr>
<tr><td><code id="get_deg_+3A_logfc_cutoff">logFC_cutoff</code></td>
<td>
<p>Cutoff value of logFC,1 by default.</p>
</td></tr>
<tr><td><code id="get_deg_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Cutoff value of pvalue,0.05 by default.</p>
</td></tr>
<tr><td><code id="get_deg_+3A_adjust">adjust</code></td>
<td>
<p>a logical value, would you like to use adjusted pvalue to draw this plot,FAlSE by default.</p>
</td></tr>
<tr><td><code id="get_deg_+3A_entriz">entriz</code></td>
<td>
<p>whether convert symbols to entriz ids</p>
</td></tr>
<tr><td><code id="get_deg_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a deg data.frame with 10 columns
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+multi_deg">multi_deg</a></code>;<code><a href="#topic+get_deg_all">get_deg_all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("Biobase",quietly = TRUE)&amp;
   requireNamespace("AnnoProbe",quietly = TRUE)){
  gse = "GSE42872"
  a = geo_download(gse,destdir=tempdir())
  find_anno(geo$gpl)
  ids &lt;- AnnoProbe::idmap(geo$gpl,destdir = tempdir())
  Group = rep(c("control","treat"),each = 3)
  Group = factor(Group)
  deg = get_deg(geo$exp,Group,ids,entriz = FALSE)
  head(deg)
}else{
  if(!requireNamespace("AnnoProbe",quietly = TRUE)) {
    warning("Package 'AnnoProbe' needed for this function to work.
         Please install it by install.packages('AnnoProbe')",call. = FALSE)
  }
  if(!requireNamespace("Biobase",quietly = TRUE)) {
    warning("Package 'Biobase' needed for this function to work.
         Please install it by BiocManager::install('Biobase')",call. = FALSE)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='get_deg_all'>get_deg_all</h2><span id='topic+get_deg_all'></span>

<h3>Description</h3>

<p>do diffiencial analysis according to exprission set and group information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_deg_all(
  exp,
  group_list,
  ids,
  symmetry = TRUE,
  my_genes = NULL,
  show_rownames = FALSE,
  cluster_cols = TRUE,
  color_volcano = c("#2874C5", "grey", "#f87669"),
  logFC_cutoff = 1,
  pvalue_cutoff = 0.05,
  adjust = FALSE,
  entriz = TRUE,
  n_cutoff = 2,
  annotation_legend = FALSE,
  lab = NA,
  species = "human"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_deg_all_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_ids">ids</code></td>
<td>
<p>a data.frame with 2 columns,including probe_id and symbol</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_symmetry">symmetry</code></td>
<td>
<p>a logical value ,would you like to get your plot symmetrical</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_my_genes">my_genes</code></td>
<td>
<p>genes for pheatmap</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_show_rownames">show_rownames</code></td>
<td>
<p>logical,show rownames or not</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>boolean values determining if columns should be clustered or <code>hclust</code> object.</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_color_volcano">color_volcano</code></td>
<td>
<p>color for volcano</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_logfc_cutoff">logFC_cutoff</code></td>
<td>
<p>Cutoff value of logFC,1 by default.</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Cutoff value of pvalue,0.05 by default.</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_adjust">adjust</code></td>
<td>
<p>a logical value, would you like to use adjusted pvalue to draw this plot,FAlSE by default.</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_entriz">entriz</code></td>
<td>
<p>logical , if TRUE ,convert symbol to entriz id.</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_n_cutoff">n_cutoff</code></td>
<td>
<p>3 by defalut , scale before plot and set a cutoff,usually 2 or 1.6</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_annotation_legend">annotation_legend</code></td>
<td>
<p>logical,show annotation legend or not</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_lab">lab</code></td>
<td>
<p>label for  x axis in volcano plot, if NA , x axis names by package</p>
</td></tr>
<tr><td><code id="get_deg_all_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with deg data.frame, volcano plot ,pca plot ,heatmap and a list with DEGs.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_deg">get_deg</a></code>;<code><a href="#topic+multi_deg_all">multi_deg_all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("Biobase",quietly = TRUE)&amp;
   requireNamespace("AnnoProbe",quietly = TRUE)){
  gse = "GSE42872"
  geo = geo_download(gse,destdir=tempdir())
  group_list = rep(c("A","B"),each = 3)
  group_list = factor(group_list)
  find_anno(geo$gpl)
  ids &lt;- AnnoProbe::idmap(geo$gpl,destdir = tempdir())
  dcp = get_deg_all(geo$exp,group_list,ids,entriz = FALSE)
  head(dcp$deg)
  dcp$plots
}else{
  if(!requireNamespace("AnnoProbe",quietly = TRUE)) {
    warning("Package 'AnnoProbe' needed for this function to work.
         Please install it by install.packages('AnnoProbe')",call. = FALSE)
  }
  if(!requireNamespace("Biobase",quietly = TRUE)) {
    warning("Package 'Biobase' needed for this function to work.
         Please install it by BiocManager::install('Biobase')",call. = FALSE)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='get_gpl_txt'>get gpl txt from GEO</h2><span id='topic+get_gpl_txt'></span>

<h3>Description</h3>

<p>get gpl annotation txt file from GEO database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gpl_txt(gpl, destdir = getwd(), download = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gpl_txt_+3A_gpl">gpl</code></td>
<td>
<p>gpl accession from GEO database</p>
</td></tr>
<tr><td><code id="get_gpl_txt_+3A_destdir">destdir</code></td>
<td>
<p>The destination directory for data downloads.</p>
</td></tr>
<tr><td><code id="get_gpl_txt_+3A_download">download</code></td>
<td>
<p>download the txt file or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with deg data.frame, volcano plot and a list with DEGs.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geo_download">geo_download</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_gpl_txt("GPL23270",destdir = tempdir())
</code></pre>

<hr>
<h2 id='ggheat'>ggheat</h2><span id='topic+ggheat'></span>

<h3>Description</h3>

<p>draw heatmap plot with annotation by ggplot2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggheat(
  dat,
  group,
  cluster = FALSE,
  color = c("#2874C5", "white", "#f87669"),
  legend_color = c("#2874C5", "#f87669", "#e6b707", "#868686", "#66C2A5", "#FC8D62",
    "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F", "#E5C494", "#B3B3B3"),
  show_rownames = TRUE,
  show_colnames = TRUE,
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  groupname = "group",
  expname = "exp",
  fill_mid = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggheat_+3A_dat">dat</code></td>
<td>
<p>expression matrix for plot</p>
</td></tr>
<tr><td><code id="ggheat_+3A_group">group</code></td>
<td>
<p>group for expression colnames</p>
</td></tr>
<tr><td><code id="ggheat_+3A_cluster">cluster</code></td>
<td>
<p>logical,cluster in both rows and column or not, default F,now replaced by cluster_rows and cluster_cols.</p>
</td></tr>
<tr><td><code id="ggheat_+3A_color">color</code></td>
<td>
<p>color for heatmap</p>
</td></tr>
<tr><td><code id="ggheat_+3A_legend_color">legend_color</code></td>
<td>
<p>color for legend</p>
</td></tr>
<tr><td><code id="ggheat_+3A_show_rownames">show_rownames</code></td>
<td>
<p>logical,show rownames in plot or not, default T</p>
</td></tr>
<tr><td><code id="ggheat_+3A_show_colnames">show_colnames</code></td>
<td>
<p>logical,show colnames in plot or not, default T</p>
</td></tr>
<tr><td><code id="ggheat_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>logical, if rows (on the plot) should be clustered, default F</p>
</td></tr>
<tr><td><code id="ggheat_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>logical, if column (on the plot) should be clustered, default F</p>
</td></tr>
<tr><td><code id="ggheat_+3A_groupname">groupname</code></td>
<td>
<p>name of group legend</p>
</td></tr>
<tr><td><code id="ggheat_+3A_expname">expname</code></td>
<td>
<p>name of exp legend</p>
</td></tr>
<tr><td><code id="ggheat_+3A_fill_mid">fill_mid</code></td>
<td>
<p>use median value as geom_tile fill midpoint</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp_dat = matrix(sample(100:1000,40),ncol = 4)
exp_dat[1:(nrow(exp_dat)/2),] =  exp_dat[1:(nrow(exp_dat)/2),]-1000
rownames(exp_dat) = paste0("sample",1:nrow(exp_dat))
colnames(exp_dat) = paste0("gene",1:ncol(exp_dat))
group = rep(c("A","B"),each = nrow(exp_dat)/2)
group = factor(group,levels = c("A","B"))
ggheat(exp_dat,group)
ggheat(exp_dat,group,cluster_rows = TRUE)
ggheat(exp_dat,group,cluster_rows = TRUE,show_rownames = FALSE,
       show_colnames = FALSE,groupname = "risk",expname = "expression")
</code></pre>

<hr>
<h2 id='hypertest'>hypertest</h2><span id='topic+hypertest'></span>

<h3>Description</h3>

<p>make hypertest for given lncRNA and mRNA common miRNAs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hypertest(lnc, pc, deMIR = NULL, lnctarget, pctarget)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hypertest_+3A_lnc">lnc</code></td>
<td>
<p>lncRNA names</p>
</td></tr>
<tr><td><code id="hypertest_+3A_pc">pc</code></td>
<td>
<p>mRNA names</p>
</td></tr>
<tr><td><code id="hypertest_+3A_demir">deMIR</code></td>
<td>
<p>miRNA names , default NULL</p>
</td></tr>
<tr><td><code id="hypertest_+3A_lnctarget">lnctarget</code></td>
<td>
<p>a data.frame with two column,lncRNA in the first column ,miRNA in the second column</p>
</td></tr>
<tr><td><code id="hypertest_+3A_pctarget">pctarget</code></td>
<td>
<p>a data.frame with two column,mRNA in the first column ,miRNA in the second column</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with hypertest result
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plcortest">plcortest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># to update
</code></pre>

<hr>
<h2 id='interaction_to_edges'>interaction_to_edges</h2><span id='topic+interaction_to_edges'></span>

<h3>Description</h3>

<p>split interactions by sep paramter,return edges data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interaction_to_edges(df, a = 1, b = 2, sep = ",")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="interaction_to_edges_+3A_df">df</code></td>
<td>
<p>interactions data.frame</p>
</td></tr>
<tr><td><code id="interaction_to_edges_+3A_a">a</code></td>
<td>
<p>column to replicate</p>
</td></tr>
<tr><td><code id="interaction_to_edges_+3A_b">b</code></td>
<td>
<p>column to split</p>
</td></tr>
<tr><td><code id="interaction_to_edges_+3A_sep">sep</code></td>
<td>
<p>a character string to separate b column</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a new data.frame with two column ,one interaction by one rows
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+edges_to_nodes">edges_to_nodes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df = data.frame(a = c("gene1","gene2","gene3"),
b = c("d,f,a,b",
"c,e,g",
"a,b,d"))
interaction_to_edges(df)
</code></pre>

<hr>
<h2 id='intersect_all'>intersect_all</h2><span id='topic+intersect_all'></span>

<h3>Description</h3>

<p>calculate intersect  set for two or more elements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intersect_all(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intersect_all_+3A_...">...</code></td>
<td>
<p>some vectors or a list with some vectors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+union_all">union_all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x1 = letters[1:4]
x2 = letters[3:6]
x3 = letters[3:4]
re =intersect_all(x1,x2,x3)
re2 = intersect_all(list(x1,x2,x3))
re3 = union_all(x1,x2,x3)
</code></pre>

<hr>
<h2 id='lnc_anno'>lnc_anno</h2><span id='topic+lnc_anno'></span>

<h3>Description</h3>

<p>annotation for TCGA expression matrix(lncRNA),form genecode v22 gtf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lnc_anno
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 14826 rows and 3 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(lnc_anno)
</code></pre>

<hr>
<h2 id='lnc_annov23'>lnc_annov23</h2><span id='topic+lnc_annov23'></span>

<h3>Description</h3>

<p>annotation for TCGA and gtex expression matrix(lncRNA),form genecode v23 gtf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lnc_annov23
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 14852 rows and 3 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(lnc_annov23)
</code></pre>

<hr>
<h2 id='make_tcga_group'>make_tcga_group</h2><span id='topic+make_tcga_group'></span>

<h3>Description</h3>

<p>make tcga group for given tcga expression matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_tcga_group(exp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_tcga_group_+3A_exp">exp</code></td>
<td>
<p>TCGA or TCGA_Gtex expression set from gdc or xena</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a group factor with normal and tumor ,correspond to colnames for expression matrix
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sam_filter">sam_filter</a></code>;<code><a href="#topic+match_exp_cl">match_exp_cl</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>k = make_tcga_group(exp_hub1);table(k)
</code></pre>

<hr>
<h2 id='match_exp_cl'>match_exp_cl</h2><span id='topic+match_exp_cl'></span>

<h3>Description</h3>

<p>match exp and clinical data from TCGA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_exp_cl(exp, cl, id_column = "id", sample_centric = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_exp_cl_+3A_exp">exp</code></td>
<td>
<p>TCGA  expression set</p>
</td></tr>
<tr><td><code id="match_exp_cl_+3A_cl">cl</code></td>
<td>
<p>TCGA clinical data.frame</p>
</td></tr>
<tr><td><code id="match_exp_cl_+3A_id_column">id_column</code></td>
<td>
<p>which column contains patient ids, column number or colnmn name.</p>
</td></tr>
<tr><td><code id="match_exp_cl_+3A_sample_centric">sample_centric</code></td>
<td>
<p>logical,deault T,keep all samples from the same patients.if FALSE,keep only one tumor sample for one patient.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a transformed clinical data.frame with sample ids.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make_tcga_group">make_tcga_group</a></code>;<code><a href="#topic+sam_filter">sam_filter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>a = match_exp_cl(exp_hub1,meta1[,2:4],"X_PATIENT")
exp_matched = a[[1]]
cl_matched = a[[2]]
b = match_exp_cl(exp_hub1,meta1[,2:4],"X_PATIENT",sample_centric = FALSE)
exp_matched = b[[1]]
cl_matched = b[[2]]
</code></pre>

<hr>
<h2 id='meta1'>meta1</h2><span id='topic+meta1'></span>

<h3>Description</h3>

<p>clinical messages for some TCGA patients,correspond to exprSet_hub1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta1
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 177 rows and 4 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(meta1)
</code></pre>

<hr>
<h2 id='mRNA_anno'>mRNA_anno</h2><span id='topic+mRNA_anno'></span>

<h3>Description</h3>

<p>annotation for TCGA and gtex expression matrix(mRNA),form genecode v22 gtf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mRNA_anno
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 19814 rows and 3 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(mRNA_anno)
</code></pre>

<hr>
<h2 id='mRNA_annov23'>mRNA_annov23</h2><span id='topic+mRNA_annov23'></span>

<h3>Description</h3>

<p>annotation for TCGA and gtex expression matrix(mRNA),form genecode v23 gtf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mRNA_annov23
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 19797 rows and 3 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(mRNA_annov23)
</code></pre>

<hr>
<h2 id='multi_deg'>multi_deg</h2><span id='topic+multi_deg'></span>

<h3>Description</h3>

<p>do diffiential analysis according to expression set and group information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_deg(
  exp,
  group_list,
  ids,
  logFC_cutoff = 1,
  pvalue_cutoff = 0.05,
  adjust = FALSE,
  species = "human",
  entriz = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi_deg_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="multi_deg_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="multi_deg_+3A_ids">ids</code></td>
<td>
<p>a data.frame with 2 columns,including probe_id and symbol</p>
</td></tr>
<tr><td><code id="multi_deg_+3A_logfc_cutoff">logFC_cutoff</code></td>
<td>
<p>Cutoff value of logFC,1 by default.</p>
</td></tr>
<tr><td><code id="multi_deg_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Cutoff value of pvalue,0.05 by default.</p>
</td></tr>
<tr><td><code id="multi_deg_+3A_adjust">adjust</code></td>
<td>
<p>a logical value, would you like to use adjusted pvalue to draw this plot,FAlSE by default.</p>
</td></tr>
<tr><td><code id="multi_deg_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
<tr><td><code id="multi_deg_+3A_entriz">entriz</code></td>
<td>
<p>whether convert symbols to entriz ids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a deg data.frame with 10 columns
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_deg">get_deg</a></code>;<code><a href="#topic+multi_deg_all">multi_deg_all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("Biobase",quietly = TRUE)&amp;
   requireNamespace("AnnoProbe",quietly = TRUE)){
  gse = "GSE474"
  geo = geo_download(gse,destdir=tempdir())
  geo$exp[1:4,1:4]
  geo$exp=log2(geo$exp+1)
  group_list=ifelse(stringr::str_detect(geo$pd$title,"MObese"),
  "MObese",ifelse(stringr::str_detect(geo$pd$title,"NonObese"),
  "NonObese","Obese"))
  group_list=factor(group_list,levels = c("NonObese","Obese","MObese"))
  find_anno(geo$gpl)
  ids &lt;- AnnoProbe::idmap(geo$gpl,destdir = tempdir())
  deg = multi_deg(geo$exp,group_list,ids,adjust = FALSE,entriz = FALSE)
  names(deg)
  head(deg[[1]])
  head(deg[[2]])
  head(deg[[3]])
}else{
  if(!requireNamespace("AnnoProbe",quietly = TRUE)) {
    warning("Package 'AnnoProbe' needed for this function to work.
         Please install it by install.packages('AnnoProbe')",call. = FALSE)
  }
  if(!requireNamespace("Biobase",quietly = TRUE)) {
    warning("Package 'Biobase' needed for this function to work.
         Please install it by BiocManager::install('Biobase')",call. = FALSE)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='multi_deg_all'>multi_deg_all</h2><span id='topic+multi_deg_all'></span>

<h3>Description</h3>

<p>do diffiencial analysis according to exprission set and group information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_deg_all(
  exp,
  group_list,
  ids,
  symmetry = TRUE,
  my_genes = NULL,
  show_rownames = FALSE,
  cluster_cols = TRUE,
  color_volcano = c("#2874C5", "grey", "#f87669"),
  pvalue_cutoff = 0.05,
  logFC_cutoff = 1,
  adjust = FALSE,
  entriz = TRUE,
  annotation_legend = FALSE,
  lab = NA,
  species = "human"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi_deg_all_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_ids">ids</code></td>
<td>
<p>a data.frame with 2 columns,including probe_id and symbol</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_symmetry">symmetry</code></td>
<td>
<p>a logical value ,would you like to get your plot symmetrical</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_my_genes">my_genes</code></td>
<td>
<p>genes for pheatmap</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_show_rownames">show_rownames</code></td>
<td>
<p>boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>boolean values determining if columns should be clustered or <code>hclust</code> object.</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_color_volcano">color_volcano</code></td>
<td>
<p>color for volcano</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Cutoff value of pvalue,0.05 by default.</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_logfc_cutoff">logFC_cutoff</code></td>
<td>
<p>Cutoff value of logFC,1 by default.</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_adjust">adjust</code></td>
<td>
<p>a logical value, would you like to use adjusted pvalue to draw this plot,FAlSE by default.</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_entriz">entriz</code></td>
<td>
<p>whether convert symbols to entriz ids</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_annotation_legend">annotation_legend</code></td>
<td>
<p>boolean value showing if the legend for annotation 
tracks should be drawn.</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_lab">lab</code></td>
<td>
<p>label for  x axis in volcano plot, if NA , x axis names by package</p>
</td></tr>
<tr><td><code id="multi_deg_all_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with deg data.frame, volcano plot and a list with DEGs.
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geo_download">geo_download</a></code>;<code><a href="#topic+draw_volcano">draw_volcano</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("Biobase",quietly = TRUE)&amp;
   requireNamespace("AnnoProbe",quietly = TRUE)){
gse = "GSE474"
geo = geo_download(gse,destdir=tempdir())
geo$exp[1:4,1:4]
geo$exp=log2(geo$exp+1)
group_list=ifelse(stringr::str_detect(geo$pd$title,"MObese"),"MObese",
ifelse(stringr::str_detect(geo$pd$title,"NonObese"),"NonObese","Obese"))
group_list=factor(group_list,levels = c("NonObese","Obese","MObese"))
find_anno(geo$gpl)
ids = AnnoProbe::idmap(geo$gpl,destdir = tempdir())
dcp = multi_deg_all(geo$exp,
group_list,ids,adjust = FALSE,entriz = FALSE)
dcp[[3]]
}else{
  if(!requireNamespace("AnnoProbe",quietly = TRUE)) {
    warning("Package 'AnnoProbe' needed for this function to work.
         Please install it by install.packages('AnnoProbe')",call. = FALSE)
  }
  if(!requireNamespace("Biobase",quietly = TRUE)) {
    warning("Package 'Biobase' needed for this function to work.
         Please install it by BiocManager::install('Biobase')",call. = FALSE)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='pkg_all'>pkg_all</h2><span id='topic+pkg_all'></span>

<h3>Description</h3>

<p>bioconductor annotation packages for GPLs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pkg_all
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 82 rows and 3 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(pkg_all)
</code></pre>

<hr>
<h2 id='plcortest'>plcortest</h2><span id='topic+plcortest'></span>

<h3>Description</h3>

<p>make cor.test for given lncRNA and mRNA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plcortest(lnc_exp, mRNA_exp, cor_cutoff = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plcortest_+3A_lnc_exp">lnc_exp</code></td>
<td>
<p>lncRNA expression set</p>
</td></tr>
<tr><td><code id="plcortest_+3A_mrna_exp">mRNA_exp</code></td>
<td>
<p>mRNA expression set which nrow equal to lncRNA_exp</p>
</td></tr>
<tr><td><code id="plcortest_+3A_cor_cutoff">cor_cutoff</code></td>
<td>
<p>cor estimate cut_off, default 0</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with cor.test result,names are lncRNAs, element are mRNAs
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hypertest">hypertest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># to update
</code></pre>

<hr>
<h2 id='plot_deg'>plot_deg</h2><span id='topic+plot_deg'></span>

<h3>Description</h3>

<p>plot pca plot,volcano plot,heatmap,and venn plot for  Differential analysis result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_deg(
  exp,
  group_list,
  deg,
  symmetry = TRUE,
  my_genes = NULL,
  show_rownames = FALSE,
  cluster_cols = TRUE,
  color_volcano = c("#2874C5", "grey", "#f87669"),
  pvalue_cutoff = 0.05,
  logFC_cutoff = 1,
  adjust = FALSE,
  annotation_legend = FALSE,
  lab = NA,
  species = "human"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_deg_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_deg">deg</code></td>
<td>
<p>result of multi_deg or get_deg function</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_symmetry">symmetry</code></td>
<td>
<p>a logical value ,would you like to get your plot symmetrical</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_my_genes">my_genes</code></td>
<td>
<p>genes for pheatmap</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_show_rownames">show_rownames</code></td>
<td>
<p>boolean specifying if column names are be shown.</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_cluster_cols">cluster_cols</code></td>
<td>
<p>boolean values determining if columns should be clustered or <code>hclust</code> object.</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_color_volcano">color_volcano</code></td>
<td>
<p>color for volcano</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Cutoff value of pvalue,0.05 by default.</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_logfc_cutoff">logFC_cutoff</code></td>
<td>
<p>Cutoff value of logFC,1 by default.</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_adjust">adjust</code></td>
<td>
<p>a logical value, would you like to use adjusted pvalue to draw this plot,FAlSE by default.</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_annotation_legend">annotation_legend</code></td>
<td>
<p>boolean value showing if the legend for annotation 
tracks should be drawn.</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_lab">lab</code></td>
<td>
<p>label for  x axis in volcano plot, if NA , x axis names by package</p>
</td></tr>
<tr><td><code id="plot_deg_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("Biobase",quietly = TRUE)&amp;
   requireNamespace("AnnoProbe",quietly = TRUE)){
gse = "GSE474"
geo = geo_download(gse,destdir=tempdir())
geo$exp[1:4,1:4]
geo$exp=log2(geo$exp+1)
group_list=ifelse(stringr::str_detect(geo$pd$title,"MObese"),"MObese",
ifelse(stringr::str_detect(geo$pd$title,"NonObese"),"NonObese","Obese"))
group_list=factor(group_list,levels = c("NonObese","Obese","MObese"))
find_anno(geo$gpl)
ids = AnnoProbe::idmap(geo$gpl,destdir = tempdir())
deg = get_deg(geo$exp,group_list,ids,adjust = FALSE,entriz = FALSE)
plot_deg(geo$exp,group_list,deg)
}else{
  if(!requireNamespace("AnnoProbe",quietly = TRUE)) {
    warning("Package 'AnnoProbe' needed for this function to work.
         Please install it by install.packages('AnnoProbe')",call. = FALSE)
  }
  if(!requireNamespace("Biobase",quietly = TRUE)) {
    warning("Package 'Biobase' needed for this function to work.
         Please install it by BiocManager::install('Biobase')",call. = FALSE)
  }
}

## End(Not run)
</code></pre>

<hr>
<h2 id='point_cut'>point_cut</h2><span id='topic+point_cut'></span>

<h3>Description</h3>

<p>calculate cut point for multiple genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>point_cut(exprSet_hub, meta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="point_cut_+3A_exprset_hub">exprSet_hub</code></td>
<td>
<p>a tumor expression set for hubgenes</p>
</td></tr>
<tr><td><code id="point_cut_+3A_meta">meta</code></td>
<td>
<p>meta data corresponds to expression set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with cutpoint for genes
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+surv_KM">surv_KM</a></code>;<code><a href="#topic+surv_cox">surv_cox</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>point_cut(exprSet_hub1,meta1)
</code></pre>

<hr>
<h2 id='quick_enrich'>quick_enrich</h2><span id='topic+quick_enrich'></span>

<h3>Description</h3>

<p>do diffiencial analysis according to exprission set and group information,for human only
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quick_enrich(
  genes,
  kkgo_file = "kkgo_file.Rdata",
  destdir = getwd(),
  species = "human"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quick_enrich_+3A_genes">genes</code></td>
<td>
<p>a gene symbol or entrizid vector</p>
</td></tr>
<tr><td><code id="quick_enrich_+3A_kkgo_file">kkgo_file</code></td>
<td>
<p>Rdata filename for kegg and go result</p>
</td></tr>
<tr><td><code id="quick_enrich_+3A_destdir">destdir</code></td>
<td>
<p>destdir to save kkgofile</p>
</td></tr>
<tr><td><code id="quick_enrich_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
</table>


<h3>Value</h3>

<p>enrichment results and dotplots
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+double_enrich">double_enrich</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
if(requireNamespace("org.Hs.eg.db",quietly = TRUE)){
  head(genes)
  g = quick_enrich(genes,destdir = tempdir())
  names(g)
  g[[1]][1:4,1:4]
  g[[3]]
  g[[4]]
}else{
  warning("Package 'org.Hs.eg.db' needed for this function to work.
         Please install it by BiocManager::install('org.Hs.eg.db')",call. = FALSE)
  }

## End(Not run)
</code></pre>

<hr>
<h2 id='risk_plot'>risk_plot</h2><span id='topic+risk_plot'></span>

<h3>Description</h3>

<p>draw risk plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>risk_plot(
  exprSet_hub,
  meta,
  riskscore,
  cut.point = FALSE,
  color = c("#2fa1dd", "#f87669"),
  n_cutoff = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="risk_plot_+3A_exprset_hub">exprSet_hub</code></td>
<td>
<p>a tumor expression set for hubgenes</p>
</td></tr>
<tr><td><code id="risk_plot_+3A_meta">meta</code></td>
<td>
<p>meta data corresponds to expression set</p>
</td></tr>
<tr><td><code id="risk_plot_+3A_riskscore">riskscore</code></td>
<td>
<p>a numeric vector of riskscore</p>
</td></tr>
<tr><td><code id="risk_plot_+3A_cut.point">cut.point</code></td>
<td>
<p>logical , use cut_point or not, if FALSE,use median by defult</p>
</td></tr>
<tr><td><code id="risk_plot_+3A_color">color</code></td>
<td>
<p>color for boxplot</p>
</td></tr>
<tr><td><code id="risk_plot_+3A_n_cutoff">n_cutoff</code></td>
<td>
<p>3 by defalut , scale before plot and set a cutoff,usually 2 or 1.6</p>
</td></tr>
</table>


<h3>Value</h3>

<p>risk plot
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exp_boxplot">exp_boxplot</a></code>;<code><a href="#topic+box_surv">box_surv</a></code>;<code><a href="#topic+draw_venn">draw_venn</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>risk_plot(exprSet_hub1,meta1,riskscore = rnorm(nrow(meta1)))
</code></pre>

<hr>
<h2 id='sam_filter'>sam_filter</h2><span id='topic+sam_filter'></span>

<h3>Description</h3>

<p>drop duplicated samples from the same patients
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sam_filter(exp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sam_filter_+3A_exp">exp</code></td>
<td>
<p>TCGA or TCGA_Gtex expression set from gdc or xena</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a transformed expression set without duplicated samples
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+make_tcga_group">make_tcga_group</a></code>;<code><a href="#topic+match_exp_cl">match_exp_cl</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cod[1:4,1:4]
dim(cod)
cod2 = sam_filter(cod)
dim(cod2)
g = make_tcga_group(cod);table(g)
library(stringr)
table(!duplicated(str_sub(colnames(cod[,g=="tumor"]),1,12)))
</code></pre>

<hr>
<h2 id='surv_cox'>surv_cox</h2><span id='topic+surv_cox'></span>

<h3>Description</h3>

<p>calculate cox p values and HR for genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_cox(
  exprSet_hub,
  meta,
  cut.point = FALSE,
  pvalue_cutoff = 0.05,
  HRkeep = "all",
  continuous = FALSE,
  min_gn = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_cox_+3A_exprset_hub">exprSet_hub</code></td>
<td>
<p>a tumor expression set for hubgenes</p>
</td></tr>
<tr><td><code id="surv_cox_+3A_meta">meta</code></td>
<td>
<p>meta data corresponds to expression set</p>
</td></tr>
<tr><td><code id="surv_cox_+3A_cut.point">cut.point</code></td>
<td>
<p>logical , use cut_point or not, if FALSE,use median by defult</p>
</td></tr>
<tr><td><code id="surv_cox_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>p value cut off ,0.05 by defult</p>
</td></tr>
<tr><td><code id="surv_cox_+3A_hrkeep">HRkeep</code></td>
<td>
<p>one of &quot;all&quot;,&quot;protect&quot;or&quot;risk&quot;</p>
</td></tr>
<tr><td><code id="surv_cox_+3A_continuous">continuous</code></td>
<td>
<p>logical, gene expression or gene expression group</p>
</td></tr>
<tr><td><code id="surv_cox_+3A_min_gn">min_gn</code></td>
<td>
<p>Depending on the expression of a gene, there may be a large difference in the number of samples between the two groups, and if a smaller group of samples is less than 10 percent (default) of all, the gene will be discarded</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix with gene names ,cox p value and HR
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+point_cut">point_cut</a></code>;<code><a href="#topic+surv_KM">surv_KM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>surv_cox(exprSet_hub1,meta1)
surv_cox(exprSet_hub1,meta1,cut.point = TRUE,continuous = TRUE)
surv_cox(exprSet_hub1,meta1,cut.point = TRUE,continuous = TRUE,pvalue_cutoff = 1)
</code></pre>

<hr>
<h2 id='surv_KM'>surv_KM</h2><span id='topic+surv_KM'></span>

<h3>Description</h3>

<p>calculate log_rank test p values for genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv_KM(
  exprSet_hub,
  meta,
  cut.point = FALSE,
  pvalue_cutoff = 0.05,
  min_gn = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="surv_KM_+3A_exprset_hub">exprSet_hub</code></td>
<td>
<p>a tumor expression set for hubgenes</p>
</td></tr>
<tr><td><code id="surv_KM_+3A_meta">meta</code></td>
<td>
<p>meta data corresponds to expression set</p>
</td></tr>
<tr><td><code id="surv_KM_+3A_cut.point">cut.point</code></td>
<td>
<p>logical , use cut_point or not, if FALSE,use median by defult</p>
</td></tr>
<tr><td><code id="surv_KM_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>p value cut off ,0.05 by defult</p>
</td></tr>
<tr><td><code id="surv_KM_+3A_min_gn">min_gn</code></td>
<td>
<p>Depending on the expression of a gene, there may be a large difference in the number of samples between the two groups, and if a smaller group of samples is less than 10 percent (default) of all, the gene will be discarded</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with gene names and log_rank p value
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+point_cut">point_cut</a></code>;<code><a href="#topic+surv_cox">surv_cox</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>surv_KM(exprSet_hub1,meta1)
surv_KM(exprSet_hub1,meta1,pvalue_cutoff = 1)
surv_KM(exprSet_hub1,meta1,cut.point = TRUE)
</code></pre>

<hr>
<h2 id='t_choose'>t_choose</h2><span id='topic+t_choose'></span>

<h3>Description</h3>

<p>choose differential expressed genes by simple t.test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>t_choose(
  genes,
  exp,
  group_list,
  up_only = FALSE,
  down_only = FALSE,
  pvalue_cutoff = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="t_choose_+3A_genes">genes</code></td>
<td>
<p>a vector with some genes</p>
</td></tr>
<tr><td><code id="t_choose_+3A_exp">exp</code></td>
<td>
<p>A numeric matrix</p>
</td></tr>
<tr><td><code id="t_choose_+3A_group_list">group_list</code></td>
<td>
<p>A factor with duplicated character or factor</p>
</td></tr>
<tr><td><code id="t_choose_+3A_up_only">up_only</code></td>
<td>
<p>keep up genes in the result only</p>
</td></tr>
<tr><td><code id="t_choose_+3A_down_only">down_only</code></td>
<td>
<p>keep down genes in the result only</p>
</td></tr>
<tr><td><code id="t_choose_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>p value cut off ,0.05 by defult</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with differential expressed genes
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp = matrix(rnorm(1000),ncol = 10)
rownames(exp) = sample(mRNA_annov23$gene_id,100)
colnames(exp) = c(paste0("TCGA",1:5),paste0("GTEX",1:5))
exp2  = trans_exp(exp)
exp2[,1:5] = exp2[,1:5]+10
group_list = rep(c("A","B"),each = 5)
genes = sample(rownames(exp2),3)
t_choose(genes,exp2,group_list)
</code></pre>

<hr>
<h2 id='trans_array'>trans_array</h2><span id='topic+trans_array'></span>

<h3>Description</h3>

<p>transform rownames for microarray or rnaseq expression matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trans_array(exp, ids, from = "probe_id", to = "symbol")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trans_array_+3A_exp">exp</code></td>
<td>
<p>microarray expression matrix with probe_id as rownames</p>
</td></tr>
<tr><td><code id="trans_array_+3A_ids">ids</code></td>
<td>
<p>data.frame  with original rownames and new rownames</p>
</td></tr>
<tr><td><code id="trans_array_+3A_from">from</code></td>
<td>
<p>colname for original rownames</p>
</td></tr>
<tr><td><code id="trans_array_+3A_to">to</code></td>
<td>
<p>colname for new rownames</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a transformed expression set with new rownames
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trans_exp">trans_exp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp = matrix(1:50,nrow = 10)
rownames(exp) = paste0("g",1:10)
ids = data.frame(probe_id = paste0("g",1:10),
                symbol = paste0("G",c(1:9,9)))
trans_array(exp,ids)
</code></pre>

<hr>
<h2 id='trans_entrezexp'>trans_entrezexp</h2><span id='topic+trans_entrezexp'></span>

<h3>Description</h3>

<p>transform rownames of expression set from &quot;entrez&quot; to&quot;symbol&quot;,according to the bitr function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trans_entrezexp(entrezexp, species = "human")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trans_entrezexp_+3A_entrezexp">entrezexp</code></td>
<td>
<p>expression set with entrezid as rownames</p>
</td></tr>
<tr><td><code id="trans_entrezexp_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a transformed expression set with symbol
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trans_exp">trans_exp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp = matrix(rnorm(200),ncol = 10)
rownames(exp) = c("79691", "56271", "8662", "10394", "55630", "159162", "23541",
                  "79723", "54413", "22927", "92342", "23787", "5550", "8924",
                  "55274", "866", "8844", "353299", "587", "1473")
colnames(exp) = paste0("s",1:10)
if(requireNamespace("org.Hs.eg.db",quietly = TRUE)){
exp2 = trans_entrezexp(exp)
}else{
    warning("Package \"org.Hs.eg.db\" needed for this function to work.
        Please install it by BiocManager::install('org.Hs.eg.db')",call. = FALSE)
}
</code></pre>

<hr>
<h2 id='trans_exp'>trans_exp</h2><span id='topic+trans_exp'></span>

<h3>Description</h3>

<p>transform rownames of TCGA or TCGA_Gtex expression set from gdc or xena,from ensembl id to gene symbol
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trans_exp(exp, mrna_only = FALSE, lncrna_only = FALSE, gtex = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trans_exp_+3A_exp">exp</code></td>
<td>
<p>TCGA or TCGA_Gtex expression set from gdc or xena</p>
</td></tr>
<tr><td><code id="trans_exp_+3A_mrna_only">mrna_only</code></td>
<td>
<p>only keep mrna rows in result</p>
</td></tr>
<tr><td><code id="trans_exp_+3A_lncrna_only">lncrna_only</code></td>
<td>
<p>only keep lncrna rows in result</p>
</td></tr>
<tr><td><code id="trans_exp_+3A_gtex">gtex</code></td>
<td>
<p>logical,whether including Gtex data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a transformed expression set with symbol
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trans_array">trans_array</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp = matrix(rnorm(1000),ncol = 10)
rownames(exp) = sample(mRNA_annov23$gene_id,100)
colnames(exp) = c(paste0("TCGA",1:5),paste0("GTEX",1:5))
k  = trans_exp(exp)
</code></pre>

<hr>
<h2 id='trans_exp_new'>trans_exp_new</h2><span id='topic+trans_exp_new'></span>

<h3>Description</h3>

<p>transform rownames of expression set from &quot;ensembl&quot; to&quot;symbol&quot;,according to the new information from ensembl database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trans_exp_new(exp, mrna_only = FALSE, lncrna_only = FALSE, species = "human")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trans_exp_new_+3A_exp">exp</code></td>
<td>
<p>expression set with ensembl as rownames</p>
</td></tr>
<tr><td><code id="trans_exp_new_+3A_mrna_only">mrna_only</code></td>
<td>
<p>only keep mrna rows in result</p>
</td></tr>
<tr><td><code id="trans_exp_new_+3A_lncrna_only">lncrna_only</code></td>
<td>
<p>only keep lncrna rows in result</p>
</td></tr>
<tr><td><code id="trans_exp_new_+3A_species">species</code></td>
<td>
<p>choose human or mouse, or rat, default: human</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a transformed expression set with symbol
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trans_exp">trans_exp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp = matrix(rnorm(1000),ncol = 10)
rownames(exp) = sample(mRNA_annov23$gene_id,100)
colnames(exp) = c(paste0("TCGA",1:5),paste0("GTEX",1:5))
if(requireNamespace("AnnoProbe")){
k  = trans_exp_new(exp)
}else{
  warning("Package \"AnnoProbe\" needed for this function to work.
         Please install it by install.packages('AnnoProbe')")
}
</code></pre>

<hr>
<h2 id='union_all'>union_all</h2><span id='topic+union_all'></span>

<h3>Description</h3>

<p>calculate union set for two or more elements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>union_all(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="union_all_+3A_...">...</code></td>
<td>
<p>some vectors or a list with some vectors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector
</p>


<h3>Author(s)</h3>

<p>Xiaojie Sun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+intersect_all">intersect_all</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x1 = letters[1:4]
x2 = letters[3:6]
x3 = letters[3:4]
re =intersect_all(x1,x2,x3)
re2 = intersect_all(list(x1,x2,x3))
re3 = union_all(x1,x2,x3)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
