<!DOCTYPE html><html><head><title>Help for package TmCalculator</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TmCalculator}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#c2s'><p>convert a vector of characters into a string</p></a></li>
<li><a href='#check_filter'><p>Check and filter invalid base of nucleotide sequences</p></a></li>
<li><a href='#chem_correction'><p>Corrections of melting temperature with chemical substances</p></a></li>
<li><a href='#complement'><p>complement and reverse complement base of nucleotide sequences</p></a></li>
<li><a href='#GC'><p>Calculate G and C content of nucleotide sequences</p></a></li>
<li><a href='#print.TmCalculator'><p>Prints melting temperature from a <code>TmCalculator</code> object</p></a></li>
<li><a href='#s2c'><p>convert a string into a vector of characters</p></a></li>
<li><a href='#salt_correction'><p>Corrections of melting temperature with salt ions</p></a></li>
<li><a href='#Tm_GC'><p>Calculate the melting temperature using empirical formulas based on GC content</p></a></li>
<li><a href='#Tm_NN'><p>Calculate melting temperature using nearest neighbor thermodynamics</p></a></li>
<li><a href='#Tm_Wallace'><p>Calculate the melting temperature using the 'Wallace rule'</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Melting Temperature of Nucleic Acid Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-02-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Junhui Li</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junhui Li &lt;junhuili@cau.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>This tool is extended from methods in Bio.SeqUtils.MeltingTemp of python. The melting temperature of nucleic acid sequences can be calculated in three method, the Wallace rule (Thein &amp; Wallace (1986) &lt;<a href="https://doi.org/10.1016%2FS0140-6736%2886%2990739-7">doi:10.1016/S0140-6736(86)90739-7</a>&gt;), empirical formulas based on G and C content (Marmur J. (1962) &lt;<a href="https://doi.org/10.1016%2FS0022-2836%2862%2980066-7">doi:10.1016/S0022-2836(62)80066-7</a>&gt;, Schildkraut C. (2010) &lt;<a href="https://doi.org/10.1002%2Fbip.360030207">doi:10.1002/bip.360030207</a>&gt;, Wetmur J G (1991) &lt;<a href="https://doi.org/10.3109%2F10409239109114069">doi:10.3109/10409239109114069</a>&gt;, Untergasser,A. (2012) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgks596">doi:10.1093/nar/gks596</a>&gt;, von Ahsen N (2001) &lt;<a href="https://doi.org/10.1093%2Fclinchem%2F47.11.1956">doi:10.1093/clinchem/47.11.1956</a>&gt;) and nearest neighbor thermodynamics (Breslauer K J (1986) &lt;<a href="https://doi.org/10.1073%2Fpnas.83.11.3746">doi:10.1073/pnas.83.11.3746</a>&gt;, Sugimoto N (1996) &lt;<a href="https://doi.org/10.1093%2Fnar%2F24.22.4501">doi:10.1093/nar/24.22.4501</a>&gt;, Allawi H (1998) &lt;<a href="https://doi.org/10.1093%2Fnar%2F26.11.2694">doi:10.1093/nar/26.11.2694</a>&gt;, SantaLucia J (2004) &lt;<a href="https://doi.org/10.1146%2Fannurev.biophys.32.110601.141800">doi:10.1146/annurev.biophys.32.110601.141800</a>&gt;, Freier S (1986) &lt;<a href="https://doi.org/10.1073%2Fpnas.83.24.9373">doi:10.1073/pnas.83.24.9373</a>&gt;, Xia T (1998) &lt;<a href="https://doi.org/10.1021%2Fbi9809425">doi:10.1021/bi9809425</a>&gt;, Chen JL (2012) &lt;<a href="https://doi.org/10.1021%2Fbi3002709">doi:10.1021/bi3002709</a>&gt;, Bommarito S (2000) &lt;<a href="https://doi.org/10.1093%2Fnar%2F28.9.1929">doi:10.1093/nar/28.9.1929</a>&gt;, Turner D H (2010) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkp892">doi:10.1093/nar/gkp892</a>&gt;, Sugimoto N (1995) &lt;<a href="https://doi.org/10.1016%2FS0048-9697%2898%2900088-6">doi:10.1016/S0048-9697(98)00088-6</a>&gt;, Allawi H T (1997) &lt;<a href="https://doi.org/10.1021%2Fbi962590c">doi:10.1021/bi962590c</a>&gt;, Santalucia N (2005) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgki918">doi:10.1093/nar/gki918</a>&gt;), and it can also be corrected with salt ions and chemical compound (SantaLucia J (1996) &lt;<a href="https://doi.org/10.1021%2Fbi951907q">doi:10.1021/bi951907q</a>&gt;, SantaLucia J(1998) &lt;<a href="https://doi.org/10.1073%2Fpnas.95.4.1460">doi:10.1073/pnas.95.4.1460</a>&gt;, Owczarzy R (2004) &lt;<a href="https://doi.org/10.1021%2Fbi034621r">doi:10.1021/bi034621r</a>&gt;, Owczarzy R (2008) &lt;<a href="https://doi.org/10.1021%2Fbi702363u">doi:10.1021/bi702363u</a>&gt;).</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/JunhuiLi1017/TmCalculator/issues">https://github.com/JunhuiLi1017/TmCalculator/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-21 03:44:19 UTC; junhuili</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-21 04:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='c2s'>convert a vector of characters into a string</h2><span id='topic+c2s'></span>

<h3>Description</h3>

<p>Simply convert a vector of characters such as c(&quot;H&quot;,&quot;e&quot;,&quot;l&quot;,&quot;l&quot;,&quot;o&quot;,&quot;W&quot;,&quot;o&quot;,&quot;r&quot;,&quot;l&quot;,&quot;d&quot;) into a single string &quot;HelloWorld&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c2s(characters)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c2s_+3A_characters">characters</code></td>
<td>
<p>A vector of characters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Retrun a strings
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p><code>citation("TmCalculator")</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
c2s(c("H","e","l","l","o","W","o","r","l","d"))

</code></pre>

<hr>
<h2 id='check_filter'>Check and filter invalid base of nucleotide sequences</h2><span id='topic+check_filter'></span>

<h3>Description</h3>

<p>In general, whitespaces and non-base characters are removed and characters are converted to uppercase in given method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_filter(ntseq, method)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_filter_+3A_ntseq">ntseq</code></td>
<td>
<p>Sequence (5' to 3') of one strand of the DNA nucleic acid duplex
as string or vector of characters</p>
</td></tr>
<tr><td><code id="check_filter_+3A_method">method</code></td>
<td>
<p>TM_Wallace: check and return &quot;A&quot;,&quot;B&quot;,&quot;C&quot;,&quot;D&quot;,&quot;G&quot;,&quot;H&quot;,&quot;I&quot;,&quot;K&quot;,&quot;M&quot;,&quot;N&quot;,&quot;R&quot;,&quot;S&quot;,&quot;T&quot;,&quot;V&quot;,&quot;W&quot; and &quot;Y&quot;
</p>
<p>TM_GC: check and return &quot;A&quot;,&quot;B&quot;,&quot;C&quot;,&quot;D&quot;,&quot;G&quot;,&quot;H&quot;,&quot;I&quot;,&quot;K&quot;,&quot;M&quot;,&quot;N&quot;,&quot;R&quot;,&quot;S&quot;,&quot;T&quot;,&quot;V&quot;,&quot;W&quot;, &quot;X&quot; and &quot;Y&quot;
</p>
<p>TM_NN: check and return &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;I&quot; and &quot;T&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a sequence which fullfils the requirements of the given method.
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p><code>citation("TmCalculator")</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ntseq &lt;- c("ATCGBDHKMNRVYWSqq")
check_filter(ntseq,method='Tm_Wallace')
check_filter(ntseq,method='Tm_NN')

</code></pre>

<hr>
<h2 id='chem_correction'>Corrections of melting temperature with chemical substances</h2><span id='topic+chem_correction'></span>

<h3>Description</h3>

<p>Corrections coefficient of melting temperature with DMSO and formamide and these corrections are rough approximations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chem_correction(
  DMSO = 0,
  fmd = 0,
  DMSOfactor = 0.75,
  fmdmethod = c("concentration", "molar"),
  fmdfactor = 0.65,
  ptGC
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chem_correction_+3A_dmso">DMSO</code></td>
<td>
<p>Percent DMSO</p>
</td></tr>
<tr><td><code id="chem_correction_+3A_fmd">fmd</code></td>
<td>
<p>Formamide concentration in percentage (fmdmethod=&quot;concentration&quot;) or molar (fmdmethod=&quot;molar&quot;).</p>
</td></tr>
<tr><td><code id="chem_correction_+3A_dmsofactor">DMSOfactor</code></td>
<td>
<p>Coefficient of Tm decreases per percent DMSO. Default=0.75 von Ahsen N (2001) &lt;PMID:11673362&gt;. Other published values are 0.5, 0.6 and 0.675.</p>
</td></tr>
<tr><td><code id="chem_correction_+3A_fmdmethod">fmdmethod</code></td>
<td>
<p>&quot;concentration&quot; method for formamide concentration in percentage and &quot;molar&quot; for formamide concentration in molar</p>
</td></tr>
<tr><td><code id="chem_correction_+3A_fmdfactor">fmdfactor</code></td>
<td>
<p>Coefficient of Tm decrease per percent formamide. Default=0.65. Several papers report factors between 0.6 and 0.72.</p>
</td></tr>
<tr><td><code id="chem_correction_+3A_ptgc">ptGC</code></td>
<td>
<p>Percentage of GC(%).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>fmdmethod = &quot;concentration&quot;
</p>
<p>Correction = - factor*percentage_of_formamide
</p>
<p>fmdmethod = &quot;molar&quot;
</p>
<p>Correction = (0.453*GC/100 - 2.88) x formamide
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p>von Ahsen N, Wittwer CT, Schutz E , et al. Oligonucleotide melting temperatures under PCR conditions: deoxynucleotide Triphosphate and Dimethyl sulfoxide concentrations with comparison to alternative empirical formulas. Clin Chem 2001, 47:1956-C1961.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>chem_correction(DMSO=3)
chem_correction(fmd=1.25, fmdmethod="molar", ptGC=50)

</code></pre>

<hr>
<h2 id='complement'>complement and reverse complement base of nucleotide sequences</h2><span id='topic+complement'></span>

<h3>Description</h3>

<p>get reverse complement and complement base of nucleotide sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>complement(ntseq, reverse = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="complement_+3A_ntseq">ntseq</code></td>
<td>
<p>Sequence (5' to 3') of one strand of the nucleic acid duplex
as string or vector of characters</p>
</td></tr>
<tr><td><code id="complement_+3A_reverse">reverse</code></td>
<td>
<p>Logical value, TRUE is reverse complement sequence, FALSE is not.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p><code>citation("TmCalculator")</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
complement("ATCGYCGYsWwsaVv")
complement("ATCGYCGYsWwsaVv",reverse=TRUE)

</code></pre>

<hr>
<h2 id='GC'>Calculate G and C content of nucleotide sequences</h2><span id='topic+GC'></span>

<h3>Description</h3>

<p>Calculate G and C content of nucleotide sequences. The number of G and C in sequence is divided by length of sequence(when totalnt is TRUE) or the number of all A,T,C,G and ambiguous base.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC(ntseq, ambiguous = FALSE, totalnt = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GC_+3A_ntseq">ntseq</code></td>
<td>
<p>Sequence (5' to 3') of one strand of the nucleic acid duplex as string or vector of characters.</p>
</td></tr>
<tr><td><code id="GC_+3A_ambiguous">ambiguous</code></td>
<td>
<p>Ambiguous bases are taken into account to compute the G and C content when ambiguous is TRUE.</p>
</td></tr>
<tr><td><code id="GC_+3A_totalnt">totalnt</code></td>
<td>
<p>Sum of 'G' and 'C' bases divided by the length of the sequence when totalnt is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Content of G and C(range from 0 to 100
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
GC(c("a","t","c","t","g","g","g","c","c","a","g","t","a"))#53.84615
GC("GCATSWSYK",ambiguous = TRUE)#55.55556

</code></pre>

<hr>
<h2 id='print.TmCalculator'>Prints melting temperature from a <code>TmCalculator</code> object</h2><span id='topic+print.TmCalculator'></span>

<h3>Description</h3>

<p><code>print.TmCalculator</code> prints to console the melting temperature value from an object of
class <code>TmCalculator</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TmCalculator'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.TmCalculator_+3A_x">x</code></td>
<td>
<p>An object of class <code>TmCalculator</code>.</p>
</td></tr>
<tr><td><code id="print.TmCalculator_+3A_...">...</code></td>
<td>
<p>Unused</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The melting temperature value.
</p>

<hr>
<h2 id='s2c'>convert a string into a vector of characters</h2><span id='topic+s2c'></span>

<h3>Description</h3>

<p>Simply convert a single string such as &quot;HelloWorld&quot; into a vector of characters such as c(&quot;H&quot;,&quot;e&quot;,&quot;l&quot;,&quot;l&quot;,&quot;o&quot;,&quot;W&quot;,&quot;o&quot;,&quot;r&quot;,&quot;l&quot;,&quot;d&quot;)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>s2c(strings)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="s2c_+3A_strings">strings</code></td>
<td>
<p>A single string such as &quot;HelloWorld&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Retrun a vector of characters
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p><code>citation("TmCalculator")</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
s2c(c("HelloWorld"))

</code></pre>

<hr>
<h2 id='salt_correction'>Corrections of melting temperature with salt ions</h2><span id='topic+salt_correction'></span>

<h3>Description</h3>

<p>Corrections coefficient of melting temperature or entropy with different operations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>salt_correction(
  Na = 0,
  K = 0,
  Tris = 0,
  Mg = 0,
  dNTPs = 0,
  method = c("Schildkraut2010", "Wetmur1991", "SantaLucia1996", "SantaLucia1998-1",
    "SantaLucia1998-2", "Owczarzy2004", "Owczarzy2008"),
  ntseq,
  ambiguous = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="salt_correction_+3A_na">Na</code></td>
<td>
<p>Millimolar concentration of Na</p>
</td></tr>
<tr><td><code id="salt_correction_+3A_k">K</code></td>
<td>
<p>Millimolar concentration of K</p>
</td></tr>
<tr><td><code id="salt_correction_+3A_tris">Tris</code></td>
<td>
<p>Millimolar concentration of Tris</p>
</td></tr>
<tr><td><code id="salt_correction_+3A_mg">Mg</code></td>
<td>
<p>Millimolar concentration of Mg</p>
</td></tr>
<tr><td><code id="salt_correction_+3A_dntps">dNTPs</code></td>
<td>
<p>Millimolar concentration of dNTPs</p>
</td></tr>
<tr><td><code id="salt_correction_+3A_method">method</code></td>
<td>
<p>Method to be applied including &quot;Schildkraut2010&quot;, &quot;Wetmur1991&quot;,&quot;SantaLucia1996&quot;, &quot;SantaLucia1998-1&quot;, &quot;SantaLucia1998-2&quot;,&quot;Owczarzy2004&quot;,&quot;Owczarzy2008&quot;. First fourth methods correct Tm, fifth method corrects deltaS, sixth and seventh methods correct 1/Tm. See details for the method description.</p>
</td></tr>
<tr><td><code id="salt_correction_+3A_ntseq">ntseq</code></td>
<td>
<p>Sequence (5' to 3') of one strand of the nucleic acid duplex as string or vector of characters.</p>
</td></tr>
<tr><td><code id="salt_correction_+3A_ambiguous">ambiguous</code></td>
<td>
<p>Ambiguous bases are taken into account to compute the G and C content when ambiguous is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The methods are:
</p>
<p>1 Schildkraut C (2010) &lt;doi:10.1002/bip.360030207&gt;
</p>
<p>2 Wetmur J G (1991) &lt;doi:10.3109/10409239109114069&gt;
</p>
<p>3 SantaLucia J (1996) &lt;doi:10.1021/bi951907q&gt;
</p>
<p>4 SantaLucia J (1998) &lt;doi:10.1073/pnas.95.4.1460&gt;
</p>
<p>5 SantaLucia J (1998) &lt;doi:10.1073/pnas.95.4.1460&gt;
</p>
<p>6 Owczarzy R (2004) &lt;doi:10.1021/bi034621r&gt;
</p>
<p>7 Owczarzy R (2008) &lt;doi:10.1021/bi702363u&gt;
</p>
<p>methods 1-4: Tm(new) = Tm(old) + correction
</p>
<p>method 5: deltaS(new) = deltaS(old) + correction
</p>
<p>methods 6+7: Tm(new) = 1/(1/Tm(old) + correction)
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p>Schildkraut C . Dependence of the melting temperature of DNA on salt concentration[J]. Biopolymers, 2010, 3(2):195-208.
</p>
<p>Wetmur J G . DNA Probes: Applications of the Principles of Nucleic Acid Hybridization[J]. CRC Critical Reviews in Biochemistry, 1991, 26(3-4):3
</p>
<p>Santalucia , J , Allawi H T , Seneviratne P A . Improved Nearest-Neighbor Parameters for Predicting DNA Duplex Stability, [J]. Biochemistry, 1996, 35(11):3555-3562.
</p>
<p>SantaLucia, J. A unified view of polymer, dumbbell, and oligonucleotide DNA nearest-neighbor thermodynamics[J]. Proceedings of the National Academy of Sciences, 1998, 95(4):1460-1465.
</p>
<p>Owczarzy R , You Y , Moreira B G , et al. Effects of Sodium Ions on DNA Duplex Oligomers: Improved Predictions ofMelting Temperatures[J]. Biochemistry, 2004, 43(12):3537-3554.
</p>
<p>Owczarzy R , Moreira B G , You Y , et al. Predicting Stability of DNA Duplexes in Solutions Containing Magnesium and Monovalent Cations[J]. Biochemistry, 2008, 47(19):5336-5353.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ntseq &lt;- c("acgtTGCAATGCCGTAWSDBSYXX")
salt_correction(Na=390, K=20, Tris=0, Mg=10, dNTPs=25, method="Owczarzy2008", ntseq)

</code></pre>

<hr>
<h2 id='Tm_GC'>Calculate the melting temperature using empirical formulas based on GC content</h2><span id='topic+Tm_GC'></span>

<h3>Description</h3>

<p>Calculate the melting temperature using empirical formulas based on GC content with different options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tm_GC(
  ntseq,
  ambiguous = FALSE,
  userset = NULL,
  variant = c("Primer3Plus", "Chester1993", "QuikChange", "Schildkraut1965",
    "Wetmur1991_MELTING", "Wetmur1991_RNA", "Wetmur1991_RNA/DNA", "vonAhsen2001"),
  Na = 0,
  K = 0,
  Tris = 0,
  Mg = 0,
  dNTPs = 0,
  saltcorr = c("Schildkraut2010", "Wetmur1991", "SantaLucia1996", "SantaLucia1998-1",
    "Owczarzy2004", "Owczarzy2008"),
  mismatch = TRUE,
  DMSO = 0,
  fmd = 0,
  DMSOfactor = 0.75,
  fmdfactor = 0.65,
  fmdmethod = c("concentration", "molar"),
  outlist = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Tm_GC_+3A_ntseq">ntseq</code></td>
<td>
<p>Sequence (5' to 3') of one strand of the nucleic acid duplex as string or vector of characters.</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_ambiguous">ambiguous</code></td>
<td>
<p>Ambiguous bases are taken into account to compute the G and C content when ambiguous is TRUE.</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_userset">userset</code></td>
<td>
<p>A vector of four coefficient values. Usersets override value sets.</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_variant">variant</code></td>
<td>
<p>Empirical constants coefficient with 8 variant: Chester1993, QuikChange, Schildkraut1965, Wetmur1991_MELTING, Wetmur1991_RNA, Wetmur1991_RNA/DNA, Primer3Plus and vonAhsen2001</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_na">Na</code></td>
<td>
<p>Millimolar concentration of Na, default is 0</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_k">K</code></td>
<td>
<p>Millimolar concentration of K, default is 0</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_tris">Tris</code></td>
<td>
<p>Millimolar concentration of Tris, default is 0</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_mg">Mg</code></td>
<td>
<p>Millimolar concentration of Mg, default is 0</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_dntps">dNTPs</code></td>
<td>
<p>Millimolar concentration of dNTPs, default is 0</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_saltcorr">saltcorr</code></td>
<td>
<p>Salt correction method should be chosen when provide 'userset'. Options are &quot;Schildkraut2010&quot;, &quot;Wetmur1991&quot;,&quot;SantaLucia1996&quot;,&quot;SantaLucia1998-1&quot;,&quot;Owczarzy2004&quot;,&quot;Owczarzy2008&quot;. Note that &quot;SantaLucia1998-2&quot; is not available for this function.</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_mismatch">mismatch</code></td>
<td>
<p>If 'True' (default) every 'X' in the sequence is counted as mismatch</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_dmso">DMSO</code></td>
<td>
<p>Percent DMSO</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_fmd">fmd</code></td>
<td>
<p>Formamide concentration in percentage (fmdmethod=&quot;concentration&quot;) or molar (fmdmethod=&quot;molar&quot;).</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_dmsofactor">DMSOfactor</code></td>
<td>
<p>Coeffecient of Tm decreases per percent DMSO. Default=0.75 von Ahsen N (2001) &lt;PMID:11673362&gt;. Other published values are 0.5, 0.6 and 0.675.</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_fmdfactor">fmdfactor</code></td>
<td>
<p>Coeffecient of Tm decrease per percent formamide. Default=0.65. Several papers report factors between 0.6 and 0.72.</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_fmdmethod">fmdmethod</code></td>
<td>
<p>&quot;concentration&quot; method for formamide concentration in percentage and &quot;molar&quot; for formamide concentration in molar</p>
</td></tr>
<tr><td><code id="Tm_GC_+3A_outlist">outlist</code></td>
<td>
<p>output a list of Tm and options or only Tm value, default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Empirical constants coefficient with 8 variant:
</p>
<p>Chester1993: Tm = 69.3 + 0.41(Percentage_GC) - 650/N
</p>
<p>QuikChange: Tm = 81.5 + 0.41(Percentage_GC) - 675/N - Percentage_mismatch
</p>
<p>Schildkraut1965: Tm = 81.5 + 0.41(Percentage_GC) - 675/N + 16.6 x log[Na+]
</p>
<p>Wetmur1991_MELTING: Tm = 81.5 + 0.41(Percentage_GC) - 500/N + 16.6 x log([Na+]/(1.0 + 0.7 x [Na+])) - Percentage_mismatch
</p>
<p>Wetmur1991_RNA: Tm = 78 + 0.7(Percentage_GC) - 500/N + 16.6 x log([Na+]/(1.0 + 0.7 x [Na+])) - Percentage_mismatch
</p>
<p>Wetmur1991_RNA/DNA: Tm = 67 + 0.8(Percentage_GC) - 500/N + 16.6 x log([Na+]/(1.0 + 0.7 x [Na+])) - Percentage_mismatch
</p>
<p>Primer3Plus: Tm = 81.5 + 0.41(Percentage_GC) - 600/N + 16.6 x log[Na+]
</p>
<p>vonAhsen2001: Tm = 77.1 + 0.41(Percentage_GC) - 528/N + 11.7 x log[Na+]
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p>Marmur J , Doty P . Determination of the base composition of deoxyribonucleic acid from its thermal denaturation temperature.[J]. Journal of Molecular Biology, 1962, 5(1):109-118.
</p>
<p>Schildkraut C . Dependence of the melting temperature of DNA on salt concentration[J]. Biopolymers, 2010, 3(2):195-208.
</p>
<p>Wetmur J G . DNA Probes: Applications of the Principles of Nucleic Acid Hybridization[J]. CRC Critical Reviews in Biochemistry, 1991, 26(3-4):33.
</p>
<p>Untergasser A , Cutcutache I , Koressaar T , et al. Primer3&ndash;new capabilities and interfaces[J]. Nucleic Acids Research, 2012, 40(15):e115-e115.
</p>
<p>von Ahsen N, Wittwer CT, Schutz E , et al. Oligonucleotide melting temperatures under PCR conditions: deoxynucleotide Triphosphate and Dimethyl sulfoxide concentrations with comparison to alternative empirical formulas. Clin Chem 2001, 47:1956-1961.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ntseq &lt;- c("ATCGTGCGTAGCAGTACGATCAGTAG")
out &lt;- Tm_GC(ntseq,ambiguous=TRUE,variant="Primer3Plus",Na=50,mismatch=TRUE)
out
out$Tm
out$Options

</code></pre>

<hr>
<h2 id='Tm_NN'>Calculate melting temperature using nearest neighbor thermodynamics</h2><span id='topic+Tm_NN'></span>

<h3>Description</h3>

<p>Calculate melting temperature using nearest neighbor thermodynamics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tm_NN(
  ntseq,
  ambiguous = FALSE,
  comSeq = NULL,
  shift = 0,
  nn_table = c("DNA_NN4", "DNA_NN1", "DNA_NN2", "DNA_NN3", "RNA_NN1", "RNA_NN2",
    "RNA_NN3", "R_DNA_NN1"),
  tmm_table = "DNA_TMM1",
  imm_table = "DNA_IMM1",
  de_table = c("DNA_DE1", "RNA_DE1"),
  dnac1 = 25,
  dnac2 = 25,
  selfcomp = FALSE,
  Na = 0,
  K = 0,
  Tris = 0,
  Mg = 0,
  dNTPs = 0,
  saltcorr = c("Schildkraut2010", "Wetmur1991", "SantaLucia1996", "SantaLucia1998-1",
    "SantaLucia1998-2", "Owczarzy2004", "Owczarzy2008"),
  DMSO = 0,
  fmd = 0,
  DMSOfactor = 0.75,
  fmdfactor = 0.65,
  fmdmethod = c("concentration", "molar"),
  outlist = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Tm_NN_+3A_ntseq">ntseq</code></td>
<td>
<p>Sequence (5' to 3') of one strand of the nucleic acid duplex
as string or vector of characters.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_ambiguous">ambiguous</code></td>
<td>
<p>Ambiguous bases are taken into account to compute the G
and C content when ambiguous is TRUE.Default is FALSE.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_comseq">comSeq</code></td>
<td>
<p>Complementary sequence. The sequence of the template/target in 3'-&gt;5' direction</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_shift">shift</code></td>
<td>
<p>Shift of the primer/probe sequence on the template/target sequence, default=0. 
for example: when shift=0, the first nucleotide base at 5' end of primer align to first one at 3' 
end of template. When shift=-1, the second nucleotide base at 5' end of primer align to first one at 3' 
end of template.
</p>
<p>When shift=1, the first nucleotide base at 5' end of primer align to second one at 3' end 
of template. The shift parameter is necessary to align  primer/probe and template/target 
if they have different lengths or if they should have dangling ends.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_nn_table">nn_table</code></td>
<td>
<p>Thermodynamic NN values, eight tables are implemented.
</p>
<p>For DNA/DNA hybridizations:
DNA_NN1,DNA_NN2,DNA_NN3,DNA_NN4
</p>
<p>For RNA/RNA hybridizations:
RNA_NN1,RNA_NN2,RNA_NN3
</p>
<p>For RNA/DNA hybridizations:
R_DNA_NN1</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_tmm_table">tmm_table</code></td>
<td>
<p>Thermodynamic values for terminal mismatches. Default: DNA_TMM1</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_imm_table">imm_table</code></td>
<td>
<p>Thermodynamic values for internal mismatches, may include insosine mismatches. Default: DNA_IMM1</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_de_table">de_table</code></td>
<td>
<p>Thermodynamic values for dangling ends. DNA_DE1(default) and RNA_DE1</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_dnac1">dnac1</code></td>
<td>
<p>Concentration of the higher concentrated strand [nM]. Typically this will
be the primer (for PCR) or the probe. Default=25.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_dnac2">dnac2</code></td>
<td>
<p>Concentration of the lower concentrated strand [nM].</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_selfcomp">selfcomp</code></td>
<td>
<p>Sequence self-complementary, default=False. If 'True' 
the primer is thought binding to itself, thus dnac2 is not considered.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_na">Na</code></td>
<td>
<p>Millimolar concentration of Na, default is 0</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_k">K</code></td>
<td>
<p>Millimolar concentration of K, default is 0</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_tris">Tris</code></td>
<td>
<p>Millimolar concentration of Tris, default is 0</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_mg">Mg</code></td>
<td>
<p>Millimolar concentration of Mg, default is 0</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_dntps">dNTPs</code></td>
<td>
<p>Millimolar concentration of dNTPs, default is 0</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_saltcorr">saltcorr</code></td>
<td>
<p>Salt correction method should be chosen when provide 'userset' 
Options are &quot;Schildkraut2010&quot;, &quot;Wetmur1991&quot;,&quot;SantaLucia1996&quot;,&quot;SantaLucia1998-1&quot;,
&quot;SantaLucia1998-2&quot;,&quot;Owczarzy2004&quot;,&quot;Owczarzy2008&quot;. Note that NA means no salt correction.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_dmso">DMSO</code></td>
<td>
<p>Percent DMSO</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_fmd">fmd</code></td>
<td>
<p>Formamide concentration in percentage (fmdmethod=&quot;concentration&quot;) or molar (fmdmethod=&quot;molar&quot;).</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_dmsofactor">DMSOfactor</code></td>
<td>
<p>Coeffecient of Tm decreases per percent DMSO. Default=0.75 von Ahsen N (2001) &lt;PMID:11673362&gt;. Other published values are 0.5, 0.6 and 0.675.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_fmdfactor">fmdfactor</code></td>
<td>
<p>Coeffecient of Tm decrease per percent formamide. Default=0.65. Several papers report factors between 0.6 and 0.72.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_fmdmethod">fmdmethod</code></td>
<td>
<p>&quot;concentration&quot; method for formamide concentration in percentage and &quot;molar&quot; for formamide concentration in molar.</p>
</td></tr>
<tr><td><code id="Tm_NN_+3A_outlist">outlist</code></td>
<td>
<p>output a list of Tm and options or only Tm value, default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>DNA_NN1: Breslauer K J (1986) &lt;doi:10.1073/pnas.83.11.3746&gt;
</p>
<p>DNA_NN2: Sugimoto N (1996) &lt;doi:10.1093/nar/24.22.4501&gt;
</p>
<p>DNA_NN3: Allawi H (1998) &lt;doi:10.1093/nar/26.11.2694&gt;
</p>
<p>DNA_NN4: SantaLucia J (2004) &lt;doi:10.1146/annurev.biophys.32.110601.141800&gt;
</p>
<p>RNA_NN1: Freier S (1986) &lt;doi:10.1073/pnas.83.24.9373&gt;
</p>
<p>RNA_NN2: Xia T (1998) &lt;doi:10.1021/bi9809425&gt;
</p>
<p>RNA_NN3: Chen JL (2012) &lt;doi:10.1021/bi3002709&gt;
</p>
<p>R_DNA_NN1: Sugimoto N (1995)&lt;doi:10.1016/S0048-9697(98)00088-6&gt;
</p>
<p>DNA_TMM1: Bommarito S (2000)  &lt;doi:10.1093/nar/28.9.1929&gt;
</p>
<p>DNA_IMM1: Peyret N (1999) &lt;doi:10.1021/bi9825091&gt; &amp; Allawi H T (1997) &lt;doi:10.1021/bi962590c&gt; &amp; Santalucia N (2005) &lt;doi:10.1093/nar/gki918&gt;
</p>
<p>DNA_DE1: Bommarito S (2000) &lt;doi:10.1093/nar/28.9.1929&gt;
</p>
<p>RNA_DE1: Turner D H (2010) &lt;doi:10.1093/nar/gkp892&gt;
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p>Breslauer K J , Frank R , Blocker H , et al. Predicting DNA duplex stability from the base sequence.[J]. Proceedings of the National Academy of Sciences, 1986, 83(11):3746-3750.
</p>
<p>Sugimoto N , Nakano S , Yoneyama M , et al. Improved Thermodynamic Parameters and Helix Initiation Factor to Predict Stability of DNA Duplexes[J]. Nucleic Acids Research, 1996, 24(22):4501-5.
</p>
<p>Allawi, H. Thermodynamics of internal C.T mismatches in DNA[J]. Nucleic Acids Research, 1998, 26(11):2694-2701.
</p>
<p>Hicks L D , Santalucia J . The thermodynamics of DNA structural motifs.[J]. Annual Review of Biophysics &amp; Biomolecular Structure, 2004, 33(1):415-440.
</p>
<p>Freier S M , Kierzek R , Jaeger J A , et al. Improved free-energy parameters for predictions of RNA duplex stability.[J]. Proceedings of the National Academy of Sciences, 1986, 83(24):9373-9377.
</p>
<p>Xia T , Santalucia , J , Burkard M E , et al. Thermodynamic Parameters for an Expanded Nearest-Neighbor Model for Formation of RNA Duplexes with Watson-Crick Base Pairs,[J]. Biochemistry, 1998, 37(42):14719-14735.
</p>
<p>Chen J L , Dishler A L , Kennedy S D , et al. Testing the Nearest Neighbor Model for Canonical RNA Base Pairs: Revision of GU Parameters[J]. Biochemistry, 2012, 51(16):3508-3522.
</p>
<p>Bommarito S, Peyret N, Jr S L. Thermodynamic parameters for DNA sequences with dangling ends[J]. Nucleic Acids Research, 2000, 28(9):1929-1934.
</p>
<p>Turner D H , Mathews D H . NNDB: the nearest neighbor parameter database for predicting stability of nucleic acid secondary structure[J]. Nucleic Acids Research, 2010, 38(Database issue):D280-D282.
</p>
<p>Sugimoto N , Nakano S I , Katoh M , et al. Thermodynamic Parameters To Predict Stability of RNA/DNA Hybrid Duplexes[J]. Biochemistry, 1995, 34(35):11211-11216.
</p>
<p>Allawi H, SantaLucia J: Thermodynamics and NMR of internal G-T mismatches in DNA. Biochemistry 1997, 36:10581-10594.
</p>
<p>Santalucia N E W J . Nearest-neighbor thermodynamics of deoxyinosine pairs in DNA duplexes[J]. Nucleic Acids Research, 2005, 33(19):6258-67.
</p>
<p>Peyret N , Seneviratne P A , Allawi H T , et al. Nearest-Neighbor Thermodynamics and NMR of DNA Sequences with Internal A-A, C-C, G-G, and T-T Mismatches, [J]. Biochemistry, 1999, 38(12):3468-3477.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ntseq &lt;- c("AAAATTTTTTTCCCCCCCCCCCCCCGGGGGGGGGGGGTGTGCGCTGC")
out &lt;- Tm_NN(ntseq,Na=50)
out
out$Options

</code></pre>

<hr>
<h2 id='Tm_Wallace'>Calculate the melting temperature using the 'Wallace rule'</h2><span id='topic+Tm_Wallace'></span>

<h3>Description</h3>

<p>The Wallace rule is often used as rule of thumb for approximate melting temperature calculations for primers with 14 to 20 nt length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tm_Wallace(ntseq, ambiguous = FALSE, outlist = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Tm_Wallace_+3A_ntseq">ntseq</code></td>
<td>
<p>Sequence (5' to 3') of one strand of the DNA nucleic acid duplex
as string or vector of characters (<strong>Note:</strong> Non-DNA characters are ignored
by this method).</p>
</td></tr>
<tr><td><code id="Tm_Wallace_+3A_ambiguous">ambiguous</code></td>
<td>
<p>Ambiguous bases are taken into account to compute the G and C content when ambiguous is TRUE.</p>
</td></tr>
<tr><td><code id="Tm_Wallace_+3A_outlist">outlist</code></td>
<td>
<p>output a list of Tm and options or only Tm value, default is TRUE.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p>Thein S L , Lynch J R , Weatherall D J , et al. DIRECT DETECTION OF HAEMOGLOBIN E WITH SYNTHETIC OLIGONUCLEOTIDES[J]. The Lancet, 1986, 327(8472):93.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ntseq = c('acgtTGCAATGCCGTAWSDBSY') #for wallace rule

out &lt;- Tm_Wallace(ntseq,ambiguous = TRUE)
out
out$Options

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
