<!DOCTYPE html><html><head><title>Help for package Tmisc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Tmisc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Tmisc-package'><p>Tmisc: Turner Miscellaneous</p></a></li>
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#%like%'><p>x like y</p></a></li>
<li><a href='#%nin%'><p>x not in y</p></a></li>
<li><a href='#%nlike%'><p>x not like y</p></a></li>
<li><a href='#addins'><p>Insert text at current position.</p></a></li>
<li><a href='#are_all_equal'><p>Are all equal?</p></a></li>
<li><a href='#corner'><p>Print the top left corner of a data frame</p></a></li>
<li><a href='#counts2fpkm'><p>Fragments per kilobase per million</p></a></li>
<li><a href='#dokuwiki'><p>Create tables in dokuwiki format</p></a></li>
<li><a href='#ellipses'><p>Truncate a data frame with ellipses.</p></a></li>
<li><a href='#fisherp'><p>Fisher's method to combine p-values.</p></a></li>
<li><a href='#gg_na'><p>Plot missing data</p></a></li>
<li><a href='#gghues'><p>Emulate ggplot2 default hues</p></a></li>
<li><a href='#gt2refalt'><p>Two-letter genotype from VCF GT</p></a></li>
<li><a href='#jsd'><p>Jensen-Shannon divergence</p></a></li>
<li><a href='#lmp'><p>Linear model p-value</p></a></li>
<li><a href='#lowestnonzero'><p>Lowest nonzero values</p></a></li>
<li><a href='#lsa'><p>Improved list of objects</p></a></li>
<li><a href='#mat2df'><p>Matrix to pairwise data frame</p></a></li>
<li><a href='#Mode'><p>Mode.</p></a></li>
<li><a href='#nn'><p>Get names and class of all columns in a data frame</p></a></li>
<li><a href='#o'><p>Open the current working directory on mac</p></a></li>
<li><a href='#peek'><p>Peek at the top of a text file</p></a></li>
<li><a href='#quartet'><p>Anscombe's Quartet data (tidy)</p></a></li>
<li><a href='#read.cb'><p>Read from the clipboard</p></a></li>
<li><a href='#saveit'><p>Rename objects while saving.</p></a></li>
<li><a href='#sicb'><p>Write sessionInfo to the clipboard</p></a></li>
<li><a href='#strSort'><p>Sort characters in a string</p></a></li>
<li><a href='#Tcols'><p>A palette of 17 diverging colors</p></a></li>
<li><a href='#Thist'><p>Histograms with overlays</p></a></li>
<li><a href='#Tpairs'><p>Better scatterplot matrices.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Turner Miscellaneous</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stephen Turner &lt;vustephen@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Miscellaneous utility functions for data manipulation,
    data tidying, and working with gene expression data.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/stephenturner/Tmisc">https://github.com/stephenturner/Tmisc</a>,
<a href="https://stephenturner.github.io/Tmisc/">https://stephenturner.github.io/Tmisc/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, tibble, utils, rstudioapi, methods, magrittr, stats</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2, reshape2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-22 21:05:41 UTC; turner</td>
</tr>
<tr>
<td>Author:</td>
<td>Stephen Turner <a href="https://orcid.org/0000-0001-9140-9028"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-23 00:10:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='Tmisc-package'>Tmisc: Turner Miscellaneous</h2><span id='topic+Tmisc'></span><span id='topic+Tmisc-package'></span>

<h3>Description</h3>

<p>Miscellaneous utility functions for data manipulation, data tidying, and working with gene expression data.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Stephen Turner <a href="mailto:vustephen@gmail.com">vustephen@gmail.com</a> (<a href="https://orcid.org/0000-0001-9140-9028">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/stephenturner/Tmisc">https://github.com/stephenturner/Tmisc</a>
</p>
</li>
<li> <p><a href="https://stephenturner.github.io/Tmisc/">https://stephenturner.github.io/Tmisc/</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='+25like+25'>x like y</h2><span id='topic++25like+25'></span>

<h3>Description</h3>

<p>Returns a logical vector of elements of x matching the regex y.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>x %like% pattern
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25like+2B25_+3A_x">x</code></td>
<td>
<p>a vector (numeric, character, factor)</p>
</td></tr>
<tr><td><code id="+2B25like+2B25_+3A_pattern">pattern</code></td>
<td>
<p>a vector (numeric, character, factor), matching the mode of x</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector with length equal to <code>x</code> of things in <code>x</code> that are like <code>y</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic++25like+25">%like%</a></code>, <code><a href="#topic++25nlike+25">%nlike%</a></code>, <code><a href="#topic++25nin+25">%nin%</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>(Name &lt;- c("Mary","George","Martha"))
Name %in% c("Mary")
Name %like% "^Mar"
Name %nin% c("George")
Name %nlike% "^Mar"
</code></pre>

<hr>
<h2 id='+25nin+25'>x not in y</h2><span id='topic++25nin+25'></span>

<h3>Description</h3>

<p>Returns a logical vector of elements of x that are not in y.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>x %nin% table
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25nin+2B25_+3A_x">x</code></td>
<td>
<p>a vector (numeric, character, factor)</p>
</td></tr>
<tr><td><code id="+2B25nin+2B25_+3A_table">table</code></td>
<td>
<p>a vector (numeric, character, factor), matching the mode of x</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector with length equal to <code>x</code> of things in <code>x</code> that aren't in <code>y</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic++25like+25">%like%</a></code>, <code><a href="#topic++25nlike+25">%nlike%</a></code>, <code><a href="#topic++25nin+25">%nin%</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>1:10 %nin% seq(from=2, to=10, by=2)
c("a", "b", "c") %nin% c("a", "b")
letters[letters %nin% unlist(strsplit("pack my box with five dozen liquor jugs", ""))]
</code></pre>

<hr>
<h2 id='+25nlike+25'>x not like y</h2><span id='topic++25nlike+25'></span>

<h3>Description</h3>

<p>Returns a logical vector of elements of x not matching the regex y.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>x %nlike% pattern
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25nlike+2B25_+3A_x">x</code></td>
<td>
<p>a vector (numeric, character, factor)</p>
</td></tr>
<tr><td><code id="+2B25nlike+2B25_+3A_pattern">pattern</code></td>
<td>
<p>a vector (numeric, character, factor), matching the mode of x</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector with length equal to <code>x</code> of things in <code>x</code> that aren't like <code>y</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic++25like+25">%like%</a></code>, <code><a href="#topic++25nlike+25">%nlike%</a></code>, <code><a href="#topic++25nin+25">%nin%</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>(Name &lt;- c("Mary","George","Martha"))
Name %in% c("Mary")
Name %like% "^Mar"
Name %nin% c("George")
Name %nlike% "^Mar"
</code></pre>

<hr>
<h2 id='addins'>Insert text at current position.</h2><span id='topic+addins'></span><span id='topic+insertInAddin'></span><span id='topic+insertEqual'></span>

<h3>Description</h3>

<p>Call these function as an addin to insert desired text at the
cursor position. After installing Tmisc, hit the Addins menu, and
optionally add a keyboard shortcut, e.g., Command+Shift+I, Alt+-, etc.
</p>

<hr>
<h2 id='are_all_equal'>Are all equal?</h2><span id='topic+are_all_equal'></span>

<h3>Description</h3>

<p>Are all the elements of a numeric vector (approximately) equal?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_all_equal(x, na.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_all_equal_+3A_x">x</code></td>
<td>
<p>A numeric vector.</p>
</td></tr>
<tr><td><code id="are_all_equal_+3A_na.rm">na.rm</code></td>
<td>
<p>Remove missing values (FALSE by default; NAs in x will return NA).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical, whether all elements of a numeric vector are equal.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>are_all_equal(c(5,5,5))
are_all_equal(c(5,5,5,6))
are_all_equal(c(5,5,5,NA,6))
are_all_equal(c(5,5,5,NA,6), na.rm=TRUE)
5==5.000000001
identical(5, 5.000000001)
are_all_equal(c(5L, 5, 5.000000001))

</code></pre>

<hr>
<h2 id='corner'>Print the top left corner of a data frame</h2><span id='topic+corner'></span>

<h3>Description</h3>

<p>Prints the first n rows and columns of a data frame or matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corner(x, n = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corner_+3A_x">x</code></td>
<td>
<p>A data.frame.</p>
</td></tr>
<tr><td><code id="corner_+3A_n">n</code></td>
<td>
<p>The number of rows/columns to print.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The corner of the data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>corner(mtcars)
corner(iris, n=4)

</code></pre>

<hr>
<h2 id='counts2fpkm'>Fragments per kilobase per million</h2><span id='topic+counts2fpkm'></span>

<h3>Description</h3>

<p>Takes a count matrix and a vector of gene lengths and returns an optionally <code>log2</code>-transformed FPKM matrix. Modified from edgeR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>counts2fpkm(x, length, log = FALSE, prior.count = 0.25)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="counts2fpkm_+3A_x">x</code></td>
<td>
<p>a matrix of counts</p>
</td></tr>
<tr><td><code id="counts2fpkm_+3A_length">length</code></td>
<td>
<p>a vector of length <code>nrow(x)</code> giving length in bases</p>
</td></tr>
<tr><td><code id="counts2fpkm_+3A_log">log</code></td>
<td>
<p>logical, if <code>TRUE</code>, then <code>log2</code> values are returned.</p>
</td></tr>
<tr><td><code id="counts2fpkm_+3A_prior.count">prior.count</code></td>
<td>
<p>average count to be added to each observation to avoid 
taking log of zero. Used only if <code>log=TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of FPKM values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(readr)
library(dplyr)
countdata &lt;- read_csv("http://files.figshare.com/2439061/GSE37704_featurecounts.csv") 
counts &lt;- countdata %&gt;% select(countdata, starts_with("SRR")) %&gt;% as.matrix
counts2fpkm(counts, countdata$length)

## End(Not run)
  
</code></pre>

<hr>
<h2 id='dokuwiki'>Create tables in dokuwiki format</h2><span id='topic+dokuwiki'></span>

<h3>Description</h3>

<p>Prints the supplied data frame or matrix using Dokuwiki's table syntax, optionally copying the data to the clipboard (Mac OS X only).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dokuwiki(x, headersep = "^", sep = "|", clip = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dokuwiki_+3A_x">x</code></td>
<td>
<p>A data.frame.</p>
</td></tr>
<tr><td><code id="dokuwiki_+3A_headersep">headersep</code></td>
<td>
<p>The separator used between entries in the header row.</p>
</td></tr>
<tr><td><code id="dokuwiki_+3A_sep">sep</code></td>
<td>
<p>The separator used between entries in all other rows.</p>
</td></tr>
<tr><td><code id="dokuwiki_+3A_clip">clip</code></td>
<td>
<p>Whether or not to write the returned table to the clipboard (currently only supported on Mac OS X).</p>
</td></tr>
<tr><td><code id="dokuwiki_+3A_...">...</code></td>
<td>
<p>Further arguments passed to <code>write.table</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>dokuwiki(head(iris), clip=FALSE)
dokuwiki(head(mtcars), clip=FALSE, row.names=TRUE)

</code></pre>

<hr>
<h2 id='ellipses'>Truncate a data frame with ellipses.</h2><span id='topic+ellipses'></span>

<h3>Description</h3>

<p>Prints the specified number of rows of a data frame, followed by a row of ellipses. Useful for piping to <code>knitr::kable()</code> for printing a truncated table in a markdown document.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ellipses(df, n = 5L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ellipses_+3A_df">df</code></td>
<td>
<p>A data.frame.</p>
</td></tr>
<tr><td><code id="ellipses_+3A_n">n</code></td>
<td>
<p>The number of rows to show before an ellipses row.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame truncated by a row of ellipses.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ellipses(mtcars, 5)

## End(Not run)

</code></pre>

<hr>
<h2 id='fisherp'>Fisher's method to combine p-values.</h2><span id='topic+fisherp'></span>

<h3>Description</h3>

<p>Uses Fisher's method to combine p-values from different tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisherp(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fisherp_+3A_x">x</code></td>
<td>
<p>A vector of p-values between 0 and 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A combined p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fisherp(c(.042, .02, .001, 0.01, .89))

</code></pre>

<hr>
<h2 id='gg_na'>Plot missing data</h2><span id='topic+gg_na'></span>

<h3>Description</h3>

<p>Plots missing data as holes on a black canvas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_na(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gg_na_+3A_df">df</code></td>
<td>
<p>A data.frame.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># What a mess. 
# Feature 10 is missing a lot. Observations 25 and 35 are completely missing.
# Most of features 40-45 are missing, except for the first few observations.
set.seed(2016-07-12)
x &lt;- matrix(1, nrow=50, ncol=50)
x[sample(prod(dim(x)), 100)] &lt;- NA
x &lt;- data.frame(x)
x$X10[sample(length(x$X10), 25)] &lt;- NA
x[c(25, 35), ] &lt;- NA
x[1:40, 40:45] &lt;- NA
gg_na(x)

</code></pre>

<hr>
<h2 id='gghues'>Emulate ggplot2 default hues</h2><span id='topic+gghues'></span>

<h3>Description</h3>

<p>This will emulate ggplot2's hues, which are equally spaced hues around the color wheel, starting from 15.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gghues(n, start = 15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gghues_+3A_n">n</code></td>
<td>
<p>The Numeric; number of hues to generate.</p>
</td></tr>
<tr><td><code id="gghues_+3A_start">start</code></td>
<td>
<p>Numeric; the place on the color wheel to start. ggplot2 default is 15.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of hues
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 10
gghues(3)
barplot(rep(1,n), col=gghues(n), names=gghues(n))
barplot(rep(1,n), col=gghues(n, start=15+180), names=gghues(n, start=15+180))

</code></pre>

<hr>
<h2 id='gt2refalt'>Two-letter genotype from VCF GT</h2><span id='topic+gt2refalt'></span>

<h3>Description</h3>

<p>Get a two-letter genotype from a VCF GT field. Current
implementation is quick and dirty, and only accepts 0/0, 0/1, or 1/1. Any
other input to gt will return a missing value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gt2refalt(gt, ref, alt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gt2refalt_+3A_gt">gt</code></td>
<td>
<p>The genotype field (must be 0/0, 0/1, or 1/1).</p>
</td></tr>
<tr><td><code id="gt2refalt_+3A_ref">ref</code></td>
<td>
<p>The reference allele.</p>
</td></tr>
<tr><td><code id="gt2refalt_+3A_alt">alt</code></td>
<td>
<p>The alternate allele.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returnvalue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gt2refalt(gt="0/0", ref="R", alt="A")
gt2refalt(gt="0/1", ref="R", alt="A")
gt2refalt(gt="1/1", ref="R", alt="A")
gt2refalt(gt="0/2", ref="R", alt="A")
gt2refalt(gt="./.", ref="R", alt="A")
</code></pre>

<hr>
<h2 id='jsd'>Jensen-Shannon divergence</h2><span id='topic+jsd'></span>

<h3>Description</h3>

<p>Calculates a distance matrix from a matrix of probability distributions using
Jensen-Shannon divergence. Adapted from <a href="https://enterotype.embl.de/enterotypes.html#dm">https://enterotype.embl.de/enterotypes.html#dm</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jsd(M, pseudocount = 1e-06, normalizeCounts = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jsd_+3A_m">M</code></td>
<td>
<p>a probability distribution matrix, e.g., normalized transcript compatibility counts.</p>
</td></tr>
<tr><td><code id="jsd_+3A_pseudocount">pseudocount</code></td>
<td>
<p>a small number to avoid division by zero errors.</p>
</td></tr>
<tr><td><code id="jsd_+3A_normalizecounts">normalizeCounts</code></td>
<td>
<p>logical, whether to attempt to normalize by dividing by the column sums. Set to <code>TRUE</code> if this is, e.g., a count matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Jensen-Shannon divergence-based distance matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(42)
M &lt;- matrix(rpois(100, lambda=100), ncol=5)
colnames(M) &lt;- paste0("sample", 1:5)
rownames(M) &lt;- paste0("gene", 1:20)
Mnorm &lt;- apply(M, 2, function(x) x/sum(x))
Mjsd &lt;- jsd(Mnorm)
# equivalently
Mjsd &lt;- jsd(M, normalizeCounts=TRUE)
Mjsd
plot(hclust(Mjsd))
  
</code></pre>

<hr>
<h2 id='lmp'>Linear model p-value</h2><span id='topic+lmp'></span>

<h3>Description</h3>

<p>Extract F-test p-value from a linear model object. Can also use <code>broom::glance(fit)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lmp(modelobject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lmp_+3A_modelobject">modelobject</code></td>
<td>
<p>A model object of class <code>lm</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The p-value on the f-test of a linear model object testing the null hypothesis that R^2==0.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate some (e.g. SNP genotype) data
set.seed(42)
n=20
d=data.frame(x1=rbinom(n,2,.5), x2=rbinom(n,2,.5))
d=transform(d, y=x1+x2+rnorm(n))
#fit the linear model
fit=lm(y ~ x1 + x2, data=d)
#shows that the F-test is 0.006641
summary(fit)
#can't access that p-value using this
names(summary(fit)) 
# this doesn't work either
names(fit)
lmp(fit)

</code></pre>

<hr>
<h2 id='lowestnonzero'>Lowest nonzero values</h2><span id='topic+lowestnonzero'></span>

<h3>Description</h3>

<p>Sometimes want to plot p-values (e.g., volcano plot or MA-plot), but if a
statistical test returns a zero p-value, this causes problems with
visualization on the log scale. This function returns a vector where  the
zero values are equal to the smallest nonzero value in the vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lowestnonzero(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lowestnonzero_+3A_x">x</code></td>
<td>
<p>A vector of p-values between 0 and 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of p-values where zero values are exchanged for the lowest non-zero p-value in the original vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lowestnonzero(c(.042, .02, 0, .001, 0, .89))

</code></pre>

<hr>
<h2 id='lsa'>Improved list of objects</h2><span id='topic+lsa'></span>

<h3>Description</h3>

<p>Improved list of objects.  Sorts by size by default. Adapted from <a href="https://stackoverflow.com/q/1358003/654296">https://stackoverflow.com/q/1358003/654296</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lsa(
  pos = 1,
  pattern,
  order.by = "Size",
  decreasing = TRUE,
  head = TRUE,
  n = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lsa_+3A_pos">pos</code></td>
<td>
<p>numeric. Position in the stack.</p>
</td></tr>
<tr><td><code id="lsa_+3A_pattern">pattern</code></td>
<td>
<p>Regex to filter the objects by.</p>
</td></tr>
<tr><td><code id="lsa_+3A_order.by">order.by</code></td>
<td>
<p>character. Either 'Type', 'Size', 'PrettySize', 'Rows', or 
'Columns'. This will dictate how the output is ordered.</p>
</td></tr>
<tr><td><code id="lsa_+3A_decreasing">decreasing</code></td>
<td>
<p>logical. Should the output be displayed in decreasing order?</p>
</td></tr>
<tr><td><code id="lsa_+3A_head">head</code></td>
<td>
<p>logical. Use head on the output?</p>
</td></tr>
<tr><td><code id="lsa_+3A_n">n</code></td>
<td>
<p>numeric. Number of objects to display is head is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with type, size in bytes, human-readable size, rows, and
columns of every object in the environment.
</p>


<h3>Author(s)</h3>

<p>Dirk Eddelbuettel, Tony Breyal
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
a &lt;- rnorm(100000)
b &lt;- matrix(1, 1000, 100)
lsa()

## End(Not run)
  
</code></pre>

<hr>
<h2 id='mat2df'>Matrix to pairwise data frame</h2><span id='topic+mat2df'></span>

<h3>Description</h3>

<p>Turns a distance matrix into a data frame of pairwise distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mat2df(M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mat2df_+3A_m">M</code></td>
<td>
<p>a square pairwise matrix (e.g., of distances).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with pairwise distances.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(42)
M &lt;- matrix(rnorm(25), nrow=5)
M
mat2df(M)
M &lt;- matrix(rnorm(25), nrow=5, dimnames=list(letters[1:5], letters[1:5]))
M
mat2df(M)
  
</code></pre>

<hr>
<h2 id='Mode'>Mode.</h2><span id='topic+Mode'></span>

<h3>Description</h3>

<p>Returns the mode of a vector. First in a tie wins (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mode(x, na.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Mode_+3A_x">x</code></td>
<td>
<p>A vector.</p>
</td></tr>
<tr><td><code id="Mode_+3A_na.rm">na.rm</code></td>
<td>
<p>Remove missing values before calculating the mode (FALSE by
default). NAs are counted just like any other element. That is, an NA in
the vector won't necessarily result in a return NA. See the first example.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A combined p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Mode(c(1,2,2,3,3,3, NA))
Mode(c(1,2,2,3,3,3, NA), na.rm=TRUE)
Mode(c(1,2,2,3,3,3, NA, NA, NA, NA))
Mode(c(1,2,2,3,3,3, NA, NA, NA, NA), na.rm=TRUE)
Mode(c("A", "Z", "Z", "B", "B"))

</code></pre>

<hr>
<h2 id='nn'>Get names and class of all columns in a data frame</h2><span id='topic+nn'></span>

<h3>Description</h3>

<p>Get names and class of all columns in a data frame in a friendly format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nn(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nn_+3A_df">df</code></td>
<td>
<p>A data.frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with index and class.
</p>


<h3>Author(s)</h3>

<p>Stephen Turner
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nn(iris)

</code></pre>

<hr>
<h2 id='o'>Open the current working directory on mac</h2><span id='topic+o'></span>

<h3>Description</h3>

<p>Opens the current working directory on mac.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>o()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
o()

## End(Not run)

</code></pre>

<hr>
<h2 id='peek'>Peek at the top of a text file</h2><span id='topic+peek'></span>

<h3>Description</h3>

<p>This returns a character vector which shows the top n lines of a file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peek(x, n = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peek_+3A_x">x</code></td>
<td>
<p>a filename</p>
</td></tr>
<tr><td><code id="peek_+3A_n">n</code></td>
<td>
<p>the number of lines to return</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
filename &lt;- tempfile()
x&lt;-matrix(round(rnorm(10^4),2),1000,10)
colnames(x)=letters[1:10]
write.csv(x,file=filename,row.names=FALSE)
peek(filename)

## End(Not run)
</code></pre>

<hr>
<h2 id='quartet'>Anscombe's Quartet data (tidy)</h2><span id='topic+quartet'></span>

<h3>Description</h3>

<p>Tidy version of built-in Anscombe's Quartet data. Four datasets that have 
nearly identical linear regression properties, yet appear very different when
graphed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quartet
</code></pre>


<h3>Format</h3>

<p>Data frame with columns.
</p>

<hr>
<h2 id='read.cb'>Read from the clipboard</h2><span id='topic+read.cb'></span>

<h3>Description</h3>

<p>Read tabular data from the clipboard.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.cb(header = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.cb_+3A_header">header</code></td>
<td>
<p>A logical value indicating whether the file contains the names of the variables as its first line. Overrides the default <code>header=FALSE</code> option in <code>read.table()</code>.</p>
</td></tr>
<tr><td><code id="read.cb_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to <code>read.table</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# To read CSV data with a header from the clipboard:
read.cb(header=TRUE, sep=',')

## End(Not run)

</code></pre>

<hr>
<h2 id='saveit'>Rename objects while saving.</h2><span id='topic+saveit'></span>

<h3>Description</h3>

<p>Allows you to rename objects as you save them. See <a href="https://stackoverflow.com/a/21248218/654296">https://stackoverflow.com/a/21248218/654296</a>.
</p>
<p>Allows you to rename objects as you save them. See <a href="https://stackoverflow.com/a/21248218/654296">https://stackoverflow.com/a/21248218/654296</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saveit(..., file = stop("'file' must be specified"))

saveit(..., file = stop("'file' must be specified"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="saveit_+3A_...">...</code></td>
<td>
<p>Objects to save.</p>
</td></tr>
<tr><td><code id="saveit_+3A_file">file</code></td>
<td>
<p>Filename/path where data will be saved.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
foo &lt;- 1
saveit(bar=foo, file="foobar.Rdata")

## End(Not run)
  
## Not run: 
foo &lt;- 1
saveit(bar=foo, file="foobar.Rdata")

## End(Not run)
  
</code></pre>

<hr>
<h2 id='sicb'>Write sessionInfo to the clipboard</h2><span id='topic+sicb'></span>

<h3>Description</h3>

<p>Writes output of <code>sessionInfo()</code> to the clipboard. Only works on Mac.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sicb()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Write sessionInfo() to the clipboard on mac.
sicb()

## End(Not run)

</code></pre>

<hr>
<h2 id='strSort'>Sort characters in a string</h2><span id='topic+strSort'></span>

<h3>Description</h3>

<p>Alphabetically sorts characters in a string. Vectorized over x.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strSort(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strSort_+3A_x">x</code></td>
<td>
<p>A string to sort.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sorted string.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>strSort("cba")
strSort("zyxcCbB105.a")
strSort(c("cba", "zyx"))
strSort(c("cba", NA))

</code></pre>

<hr>
<h2 id='Tcols'>A palette of 17 diverging colors</h2><span id='topic+Tcols'></span>

<h3>Description</h3>

<p>17 diverging colors created by combining the Set1 and Dark2 palettes from 
RColorBrewer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tcols
</code></pre>


<h3>Format</h3>

<p>Vector of 17 diverging colors.
</p>


<h3>Source</h3>

<p>R Color brewer: <code>c(brewer.pal(9, "Set1"), brewer.pal(8, "Dark2"))</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 barplot(rep(1, 17), col=Tcols, axes=F, names=c(rep("Set1", 9), rep("Dark2", 8)), horiz=TRUE, las=2)

## End(Not run)
</code></pre>

<hr>
<h2 id='Thist'>Histograms with overlays</h2><span id='topic+Thist'></span>

<h3>Description</h3>

<p>Plot a histogram with either a normal distribution or density curve overlay.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Thist(x, overlay = "normal", col = "gray80", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Thist_+3A_x">x</code></td>
<td>
<p>A numeric vector.</p>
</td></tr>
<tr><td><code id="Thist_+3A_overlay">overlay</code></td>
<td>
<p>Either &quot;normal&quot; (default) or &quot;density&quot; indicating whether a normal distribution or density curve should be plotted on top of the histogram.</p>
</td></tr>
<tr><td><code id="Thist_+3A_col">col</code></td>
<td>
<p>Color of the histogram bars.</p>
</td></tr>
<tr><td><code id="Thist_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to <code>hist()</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(42)
x &lt;- rnorm(1000, mean=5, sd=2)
Thist(x)
Thist(x, overlay="density")
Thist(x^2)
Thist(x^2, overlay="density", breaks=50, col="lightblue2")

</code></pre>

<hr>
<h2 id='Tpairs'>Better scatterplot matrices.</h2><span id='topic+Tpairs'></span>

<h3>Description</h3>

<p>A matrix of scatter plots with rugged histograms, correlations, and significance stars. Much of the functionality borrowed from <code>PerformanceAnalytics::chart.Correlation()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tpairs(x, histogram = TRUE, gap = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Tpairs_+3A_x">x</code></td>
<td>
<p>A numeric matrix or data.frame.</p>
</td></tr>
<tr><td><code id="Tpairs_+3A_histogram">histogram</code></td>
<td>
<p>Overlay a histogram on the diagonals?</p>
</td></tr>
<tr><td><code id="Tpairs_+3A_gap">gap</code></td>
<td>
<p>distance between subplots, in margin lines.</p>
</td></tr>
<tr><td><code id="Tpairs_+3A_...">...</code></td>
<td>
<p>arguments to be passed to or from other methods.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>Tpairs(iris[-5])
Tpairs(iris[-5], pch=21, bg=Tcols[factor(iris$Species)])
Tpairs(iris[-5], pch=21, bg=gghues(3)[factor(iris$Species)], gap=1)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
