<!DOCTYPE html><html><head><title>Help for package TML</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TML}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TML'><p>Tropical Machine Learning in R</p></a></li>
<li><a href='#bw.nn'><p>Nearest neighbor bandwidth calculation</p></a></li>
<li><a href='#cluster.ratio_HC'><p>Ratio of within and between tropical measures for tropical hierarchical clusters</p></a></li>
<li><a href='#cluster.ratio_KM'><p>Ratio of within and between tropical measures for k-means clusters</p></a></li>
<li><a href='#convert.to.tree'><p>Create a phylogenetic tree from an ultrametric</p></a></li>
<li><a href='#draw.tpolytope'><p>Draw a 2-D or 3-D tropical polytope</p></a></li>
<li><a href='#FWpoint_numerical'><p>Fermat-Weber point numerical solver</p></a></li>
<li><a href='#FWpoint.num.w.reg'><p>Modified Fermat-Weber point numerical solver for ultrametrics</p></a></li>
<li><a href='#HAR.TLineSeg'><p>Uniformly sample from a max-plus tropical line segment</p></a></li>
<li><a href='#HAR.TLineSeg.Norm'><p>Gaussian-like Sampling on a max-plus tropical line segment</p></a></li>
<li><a href='#hyper_3D'><p>2D or 3D rendering of max-plus or min-plus tropical hyperplane</p></a></li>
<li><a href='#lung_fish'><p>Phylogenetic trees based on lung fish data</p></a></li>
<li><a href='#max_ins_ball'><p>Calculate the center point and radius of the maximum inscribed ball for a tropical simplex</p></a></li>
<li><a href='#min_enc_ball'><p>Calculate a minimum enclosing ball for a tropical polytope</p></a></li>
<li><a href='#normaliz.tree'><p>Normalize a phylogenetic tree</p></a></li>
<li><a href='#normalize'><p>Normalize a point or set of points in the tropical projective torus</p></a></li>
<li><a href='#over_bet_HC'><p>Tropical cluster betweeness measure for each cluster in a set of hierarchical clusters</p></a></li>
<li><a href='#over_bet_KM'><p>Tropical cluster betweeness measure for a each of a set of k-means derived set of clusters</p></a></li>
<li><a href='#Points.TLineSeg'><p>Sample k equally spaced points on a max-plus tropical line segment</p></a></li>
<li><a href='#pre.pplot.pro'><p>Projections of points onto a tropical triangle</p></a></li>
<li><a href='#prob.class'><p>Estimated probability for binary class assignment</p></a></li>
<li><a href='#project_pi'><p>Project a point on the tropical projective torus onto a tropical polytope.</p></a></li>
<li><a href='#pw.trop.dist'><p>Constructs the dissimilarity matrix for a set of ultrametrics</p></a></li>
<li><a href='#rounding'><p>Remove all tentacles from a tropical simplex</p></a></li>
<li><a href='#sigmoid'><p>Sigmoid function</p></a></li>
<li><a href='#Sim_points'><p>Simulated points over the tropical projective torus</p></a></li>
<li><a href='#Sim_Trees1'><p>Six data sets of phylogenetic trees data simulated from the Coalescant model.</p></a></li>
<li><a href='#Sim_Trees2'><p>Six data sets of phylogenetic trees data simulated from the Coalescant model.</p></a></li>
<li><a href='#tdets'><p>Calculate the tropical determinant of a square matrix.</p></a></li>
<li><a href='#TKmeans'><p>K-means clustering over the tropical projective torus</p></a></li>
<li><a href='#TLineSeg'><p>Construct a max-plus tropical line segment between two points</p></a></li>
<li><a href='#TLineSeg_min'><p>Construct a min-plus tropical line segment accor</p></a></li>
<li><a href='#tree.to.vector'><p>Phylogenetic tree to vector</p></a></li>
<li><a href='#trop_bal.vert'><p>Calculate the minimum generating vertex set of a tropical ball</p></a></li>
<li><a href='#Trop_ball'><p>Visualize a Tropical ball in 2D or 3D</p></a></li>
<li><a href='#Trop_FW'><p>Calculate the tropical Fermat-Weber point</p></a></li>
<li><a href='#Trop_Volume'><p>Estimate the volume of a tropical polytope</p></a></li>
<li><a href='#trop_wi_dist'><p>Tropical within-cluster measure</p></a></li>
<li><a href='#trop.dist'><p>Compute the tropical distance</p></a></li>
<li><a href='#trop.dist.hyp_max'><p>Calculate the tropical distance to a max-tropical hyperplane</p></a></li>
<li><a href='#trop.dist.hyp_min'><p>Calculate the tropical distance to a min-tropical hyperplane</p></a></li>
<li><a href='#trop.logistic.regression'><p>Tropical Logistic Regression</p></a></li>
<li><a href='#trop.tri.plot.w.pts'><p>Plotting PCA-derived tropical triangles</p></a></li>
<li><a href='#tropical.Gaussian'><p>Tropical Gaussian sampling about a center of mass</p></a></li>
<li><a href='#tropical.Gaussian.MH'><p>Gaussian-like sampling using Metropolis filter</p></a></li>
<li><a href='#Tropical.HC.AGNES'><p>Agglomerative (AGNES) tropical hierarchical clustering</p></a></li>
<li><a href='#tropical.KDE'><p>Tropical Kernel Density Estimation of Phylogenetic Trees</p></a></li>
<li><a href='#tropical.PCA'><p>Tropical principal component analysis (PCA) on over tropical projective torus</p></a></li>
<li><a href='#TropicalPolytope.extrapolation.HAR'><p>Vertex HAR with extrapolation MCMC with uniform target distribution</p></a></li>
<li><a href='#Ultrametrics.HAR'><p>Hit-and-Run Sampler for the space of ultrametrics</p></a></li>
<li><a href='#vector.to.equidistant.tree'><p>Vector to equidistant tree</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tropical Geometry Tools for Machine Learning</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Suite of tropical geometric tools for use in machine learning applications. These methods may be summarized in the following references: Yoshida, et al. (2022) &lt;<a href="https://doi.org/10.48550/arXiv.2209.15045">doi:10.48550/arXiv.2209.15045</a>&gt;, Barnhill et al. (2023) &lt;<a href="https://doi.org/10.48550/arXiv.2303.02539">doi:10.48550/arXiv.2303.02539</a>&gt;, Barnhill and Yoshida (2023) &lt;<a href="https://doi.org/10.3390%2Fmath11153433">doi:10.3390/math11153433</a>&gt;, Aliatimis et al. (2023) &lt;<a href="https://doi.org/10.48550/arXiv.2306.08796">doi:10.48550/arXiv.2306.08796</a>&gt;, Yoshida et al. (2022) &lt;<a href="https://doi.org/10.48550/arXiv.2206.04206">doi:10.48550/arXiv.2206.04206</a>&gt;, and Yoshida et al. (2019) &lt;<a href="https://doi.org/10.1007%2Fs11538-018-0493-4">doi:10.1007/s11538-018-0493-4</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Barnhill &lt;david.barnhill@nps.edu&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/barnhilldave/TML">https://github.com/barnhilldave/TML</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/barnhilldave/TML/issues">https://github.com/barnhilldave/TML/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, Matrix, RcppAlgos, Rfast, combinat, gtools, lpSolve,
lpSolveAPI, magick, miscTools, phangorn, rcdd, rgl, ape,
phytools, maps, cluster, ROCR, stringr, stats</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-14 01:06:57 UTC; davidbarnhill</td>
</tr>
<tr>
<td>Author:</td>
<td>David Barnhill <a href="https://orcid.org/0009-0006-2781-7434"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ruriko Yoshida [aut],
  Georgios Aliatimis [aut],
  Keiji Miura [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-14 03:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='TML'>Tropical Machine Learning in R</h2><span id='topic+TML'></span><span id='topic+TML-package'></span>

<h3>Description</h3>

<p>TML provides a suite of tools for machine learning application on data over the tropical semiring
</p>

<hr>
<h2 id='bw.nn'>Nearest neighbor bandwidth calculation</h2><span id='topic+bw.nn'></span>

<h3>Description</h3>

<p>This function finds the bandwidth for an ultrametric based on the tropical distance of the nearest point.
The function provides the bandwidth input to trop.KDE and was originally used in the KDETrees package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bw.nn(x, prop = 0.2, tol = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bw.nn_+3A_x">x</code></td>
<td>
<p>matrix; dissimilarity matrix between points in a data set</p>
</td></tr>
<tr><td><code id="bw.nn_+3A_prop">prop</code></td>
<td>
<p>proportion of observations that defines neighborhood of a point</p>
</td></tr>
<tr><td><code id="bw.nn_+3A_tol">tol</code></td>
<td>
<p>tolerance for zero bandwidth check</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of bandwidths for each tree (row) in x
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>References</h3>

<p>Weyenberg, G., Huggins, P., Schardl, C., Howe, D. K., &amp; Yoshida, R. (2014). kdetrees: Nonparametric Estimation of Phylogenetic Tree Distributions. In Bioinformatics.
</p>
<p><a href="https://github.com/grady/kdetrees/blob/master/R/bw.R">https://github.com/grady/kdetrees/blob/master/R/bw.R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
T1&lt;-Sim_Trees15
T2&lt;-Sim_Trees25
D &lt;- rbind(T1, T2[1,])
M &lt;- pw.trop.dist(D, D)
bw.nn(M)

</code></pre>

<hr>
<h2 id='cluster.ratio_HC'>Ratio of within and between tropical measures for tropical hierarchical clusters</h2><span id='topic+cluster.ratio_HC'></span>

<h3>Description</h3>

<p>Ratio of within and between cluster tropical measures for a set hierarchical clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster.ratio_HC(A, V, method = "avg")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster.ratio_HC_+3A_a">A</code></td>
<td>
<p>matrix of tropical points; rows are points</p>
</td></tr>
<tr><td><code id="cluster.ratio_HC_+3A_v">V</code></td>
<td>
<p>list of clusters where each cluster is defined as a matrix</p>
</td></tr>
<tr><td><code id="cluster.ratio_HC_+3A_method">method</code></td>
<td>
<p>method to use for within cluster measure; &quot;avg&quot; or &quot;max&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of ratios for each cluster
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>David Barnhill, Ruriko Yoshida (2023). Clustering Methods Over the Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>har&lt;-rbind(Sim_points[1:20,],Sim_points[51:70,])

V&lt;-Tropical.HC.AGNES(har, method="average")
inds&lt;-V[[2]][[38]]
cluster.ratio_HC(har,inds,method='avg')
</code></pre>

<hr>
<h2 id='cluster.ratio_KM'>Ratio of within and between tropical measures for k-means clusters</h2><span id='topic+cluster.ratio_KM'></span>

<h3>Description</h3>

<p>Ratio of within and between cluster tropical measures for k-means derived clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster.ratio_KM(A, C, method = "avg")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster.ratio_KM_+3A_a">A</code></td>
<td>
<p>matrix of tropical points; rows are points</p>
</td></tr>
<tr><td><code id="cluster.ratio_KM_+3A_c">C</code></td>
<td>
<p>number of clusters</p>
</td></tr>
<tr><td><code id="cluster.ratio_KM_+3A_method">method</code></td>
<td>
<p>method to use for within cluster measure; &quot;avg&quot; or &quot;max&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of ratios for each cluster
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>David Barnhill, Ruriko Yoshida (2023). Clustering Methods Over the Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
hars&lt;-Sim_points
cls&lt;-c(rep(1,50),rep(2,50),rep(3,50))
cl_pt&lt;-cbind(hars,cls)

C&lt;-3
cluster.ratio_KM(cl_pt,C,method='avg')
</code></pre>

<hr>
<h2 id='convert.to.tree'>Create a phylogenetic tree from an ultrametric</h2><span id='topic+convert.to.tree'></span>

<h3>Description</h3>

<p>This function constructs a phylogenetic tree from an ultrametric.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.to.tree(n, L, u)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.to.tree_+3A_n">n</code></td>
<td>
<p>is the number of leaves</p>
</td></tr>
<tr><td><code id="convert.to.tree_+3A_l">L</code></td>
<td>
<p>is a vector of labels (strings) of leaves</p>
</td></tr>
<tr><td><code id="convert.to.tree_+3A_u">u</code></td>
<td>
<p>is an ultrametric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A phylogenetic tree of class <code>phylo</code>
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>um&lt;-Sim_Trees21[1,]
ll &lt;- 10
L &lt;- LETTERS[1:10]
tr&lt;-convert.to.tree(ll, L, um)
</code></pre>

<hr>
<h2 id='draw.tpolytope'>Draw a 2-D or 3-D tropical polytope</h2><span id='topic+draw.tpolytope'></span><span id='topic+draw.tpolytope.3d'></span><span id='topic+draw.tpolytope.2d'></span>

<h3>Description</h3>

<p>This command draws a three dimensional tropical polytope
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw.tpolytope.3d(D, c, cc, plt = TRUE)

draw.tpolytope.2d(D, c, cc, plt = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw.tpolytope_+3A_d">D</code></td>
<td>
<p>matrix of vertices of a tropical polytope; rows are the vertices</p>
</td></tr>
<tr><td><code id="draw.tpolytope_+3A_c">c</code></td>
<td>
<p>string; color to render the polytope.</p>
</td></tr>
<tr><td><code id="draw.tpolytope_+3A_cc">cc</code></td>
<td>
<p>string; color to render the vertices.</p>
</td></tr>
<tr><td><code id="draw.tpolytope_+3A_plt">plt</code></td>
<td>
<p>logical; initiate new plot visualization or not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>2-D or 3-D rendering of a tropical polytope.
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D &lt;-matrix(c(0,0,0,0,0,1,0,0,0,0,1,0,0,0,0,1),4,4,TRUE)
c&lt;-'blue'
cc&lt;-'red'
draw.tpolytope.3d(D,c,cc,plt=TRUE)

D &lt;- matrix(c(0,-2,2,0,-2,5,0,2,1,0,1,-1),4,3,TRUE)
c &lt;- 'blue'
cc &lt;- 'red'
draw.tpolytope.2d(D,c,cc,plt=TRUE)


</code></pre>

<hr>
<h2 id='FWpoint_numerical'>Fermat-Weber point numerical solver</h2><span id='topic+FWpoint_numerical'></span>

<h3>Description</h3>

<p>Returns the Fermat-Weber point of N points using a gradient based numerical method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FWpoint_numerical(datamatrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FWpoint_numerical_+3A_datamatrix">datamatrix</code></td>
<td>
<p>matrix of dimension N*e, where N is the number of observations which lie in R^e.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fermat-Weber point approximation (vector in R^e)
</p>


<h3>Author(s)</h3>

<p>Georgios Aliatimis <a href="mailto:g.aliatimis@lancaster.ac.uk">g.aliatimis@lancaster.ac.uk</a>
</p>


<h3>References</h3>

<p>Aliatimis, Georgios, Ruriko Yoshida, Burak Boyaci and James A. Grant (2023). Tropical Logistic Regression on Space of Phylogenetic Trees
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D = matrix(c(0,0,0,0,2,5,0,3,1),3,3,TRUE)
FWpoint_numerical(D)
</code></pre>

<hr>
<h2 id='FWpoint.num.w.reg'>Modified Fermat-Weber point numerical solver for ultrametrics</h2><span id='topic+FWpoint.num.w.reg'></span>

<h3>Description</h3>

<p>Returns a modified Fermat-Weber point of N points using a gradient based numerical method
This method is appropriate for points coming from ultrametrics.
The algorithm tries to find a point that minimizes the sum of tropical distances from the samples, but also
also tries to find a point that is as close as possible to the space of ultrametrics.
The tradeoff between these two objectives is controlled by the penalty parameter.
If penalty=0, the method is identical to FWpoint_numerical; it finds the Fermat-Weber point, which may not be an ultrametric.
If penalty is very large, the algorithm is trying to find the Fermat-Weber point in the space of ultrametrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FWpoint.num.w.reg(datamatrix, penalty = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FWpoint.num.w.reg_+3A_datamatrix">datamatrix</code></td>
<td>
<p>matrix of dimension N*e, where N is the number of observations which lie in R^e</p>
</td></tr>
<tr><td><code id="FWpoint.num.w.reg_+3A_penalty">penalty</code></td>
<td>
<p>positive real number; the regularization rate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector; Fermat-Weber point approximation
</p>


<h3>Author(s)</h3>

<p>Georgios Aliatimis <a href="mailto:g.aliatimis@lancaster.ac.uk">g.aliatimis@lancaster.ac.uk</a>
</p>


<h3>References</h3>

<p>Aliatimis, Georgios, Ruriko Yoshida, Burak Boyaci and James A. Grant (2023). Tropical Logistic Regression on Space of Phylogenetic Trees
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D = matrix(c(0,0,0,0,2,5,0,3,1),3,3,TRUE)
FWpoint.num.w.reg(D,1e4) # (0,2,5/3) not ultrametric
FWpoint.num.w.reg(D,1e4) # (0,5/3,5/3) ultrametric
</code></pre>

<hr>
<h2 id='HAR.TLineSeg'>Uniformly sample from a max-plus tropical line segment</h2><span id='topic+HAR.TLineSeg'></span>

<h3>Description</h3>

<p>This function uses a hit-and-run sampler to uniformly sample from a max-plus tropical line segment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HAR.TLineSeg(D1, D2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HAR.TLineSeg_+3A_d1">D1</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
<tr><td><code id="HAR.TLineSeg_+3A_d2">D2</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
</table>


<h3>Value</h3>

<p>point on the line segment defined by D1 and D2
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>References</h3>

<p>Yoshida, Ruriko, Keiji Miura and David Barnhill (2022). Hit and Run Sampling from Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D1 &lt;-c(0,4,2)
D2 &lt;- c(0,7,-1)
HAR.TLineSeg(D1, D2)
</code></pre>

<hr>
<h2 id='HAR.TLineSeg.Norm'>Gaussian-like Sampling on a max-plus tropical line segment</h2><span id='topic+HAR.TLineSeg.Norm'></span>

<h3>Description</h3>

<p>This function samples points on a tropical line segment about a location parameter for a given scale parameter defined in terms of tropical distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HAR.TLineSeg.Norm(D1, D2, mu, stdev)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HAR.TLineSeg.Norm_+3A_d1">D1</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
<tr><td><code id="HAR.TLineSeg.Norm_+3A_d2">D2</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
<tr><td><code id="HAR.TLineSeg.Norm_+3A_mu">mu</code></td>
<td>
<p>location parameter</p>
</td></tr>
<tr><td><code id="HAR.TLineSeg.Norm_+3A_stdev">stdev</code></td>
<td>
<p>scale parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>point on the line segment defined by D1 and D2 sampled about mu
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D1 &lt;-c(0,4,2)
D2 &lt;- c(0,7,-1)
mu&lt;-c(0,7,2)
sd&lt;-1
HAR.TLineSeg.Norm(D1, D2,mu,sd)

</code></pre>

<hr>
<h2 id='hyper_3D'>2D or 3D rendering of max-plus or min-plus tropical hyperplane</h2><span id='topic+hyper_3D'></span><span id='topic+hyper3d_max'></span><span id='topic+hyper3d_min'></span>

<h3>Description</h3>

<p>This function renders a 2D or 3D max-plus or min-plus tropical hyperplane
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hyper3d_max(D, di, mi, ma, plt = FALSE)

hyper3d_min(D, di, mi, ma, plt = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hyper_3D_+3A_d">D</code></td>
<td>
<p>point in the tropical projective torus representing the apex of the hyperplane</p>
</td></tr>
<tr><td><code id="hyper_3D_+3A_di">di</code></td>
<td>
<p>scalar; indicates how far the hyperplane should extend</p>
</td></tr>
<tr><td><code id="hyper_3D_+3A_mi">mi</code></td>
<td>
<p>scalar; minimum value on axes of the plot</p>
</td></tr>
<tr><td><code id="hyper_3D_+3A_ma">ma</code></td>
<td>
<p>scalar; maximum value on axes of the plot</p>
</td></tr>
<tr><td><code id="hyper_3D_+3A_plt">plt</code></td>
<td>
<p>logical; if true produces a new plot otherwise overlays tropical hyperplane on existing plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>2D or 3D rendering of max-plus or min-plus tropical hyperplane
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 2D Example
D &lt;-t(as.matrix(c(0,0,0)))
di&lt;-4
mi&lt;- -5
ma&lt;-5
hyper3d_max(D,di,mi,ma,plt=TRUE)
hyper3d_min(D,di,mi,ma,plt=TRUE)

# 3D Example
D &lt;-t(as.matrix(c(0,0,0,0)))
di&lt;-4
mi&lt;- -5
ma&lt;-5
hyper3d_max(D,di,mi,ma,plt=TRUE)
hyper3d_min(D,di,mi,ma,plt=TRUE)
</code></pre>

<hr>
<h2 id='lung_fish'>Phylogenetic trees based on lung fish data</h2><span id='topic+lung_fish'></span><span id='topic+lf_labels'></span>

<h3>Description</h3>

<p>1290 (non-equidistant) gene trees with 45 leaves originating from lung fish data in matrix form.  Also we provide a vector of strings consisting of leaf labels for each species associated with the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lung_fish

lf_labels
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1290 rows and 45 columns.
</p>
<p>An object of class <code>character</code> of length 10.
</p>


<h3>Source</h3>

<p>Liang D, Shen XX, Zhang P. One thousand two hundred ninety nuclear genes from a genome-wide survey support lungfishes as the sister group of tetrapods. Mol Biol Evol. 2013 Aug;30(8):1803-7. doi: 10.1093/molbev/mst072. Epub 2013 Apr 14. PMID: 23589454.
</p>

<hr>
<h2 id='max_ins_ball'>Calculate the center point and radius of the maximum inscribed ball for a tropical simplex</h2><span id='topic+max_ins_ball'></span>

<h3>Description</h3>

<p>This function calculates the center point and radius of the maximum inscribed ball for a tropical simplex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>max_ins_ball(A)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="max_ins_ball_+3A_a">A</code></td>
<td>
<p>matrix of points defining a tropical polytope; rows are the points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing the radius and center point of a maximum inscribed ball
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>Barnhill, David, Ruriko Yoshida and Keiji Miura (2023). Maximum Inscribed and Minimum Enclosing Tropical Balls of Tropical Polytopes and Applications to Volume Estimation and Uniform Sampling.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P&lt;-matrix(c(0,0,0,0,2,5,0,3,1),3,3,TRUE)
max_ins_ball(P)
</code></pre>

<hr>
<h2 id='min_enc_ball'>Calculate a minimum enclosing ball for a tropical polytope</h2><span id='topic+min_enc_ball'></span>

<h3>Description</h3>

<p>This function constructs a minimum enclosing ball for a set of points defining a tropical polytope.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>min_enc_ball(A)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="min_enc_ball_+3A_a">A</code></td>
<td>
<p>matrix of points defining a tropical polytope. Rows are the points.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing center point and radius of minimum enclosing ball of P
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>Barnhill, David, Ruriko Yoshida and Keiji Miura (2023). Maximum Inscribed and Minimum Enclosing Tropical Balls of Tropical Polytopes and Applications to Volume Estimation and Uniform Sampling.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P &lt;-matrix(c(0,0,0,0,3,1,0,2,5),3,3,TRUE)
min_enc_ball(P)
</code></pre>

<hr>
<h2 id='normaliz.tree'>Normalize a phylogenetic tree</h2><span id='topic+normaliz.tree'></span>

<h3>Description</h3>

<p>This function normalizes the height of a phylogenetic tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normaliz.tree(D, h = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normaliz.tree_+3A_d">D</code></td>
<td>
<p>numeric vector; ultrametric equidistant tree</p>
</td></tr>
<tr><td><code id="normaliz.tree_+3A_h">h</code></td>
<td>
<p>desired height; defaults to 1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>normalized equidistant tree
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D &lt;-c(4,4,2)
normaliz.tree(D, h=1)
</code></pre>

<hr>
<h2 id='normalize'>Normalize a point or set of points in the tropical projective torus</h2><span id='topic+normalize'></span><span id='topic+normaliz.vector'></span><span id='topic+normaliz.vectors'></span><span id='topic+normaliz.polytope'></span><span id='topic+normaliz.ultrametrics'></span>

<h3>Description</h3>

<p>This function normalizes a point or set of points in the tropical projective torus by making the first coordinate zero
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normaliz.vector(D)

normaliz.vectors(D)

normaliz.polytope(D)

normaliz.ultrametrics(D)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_+3A_d">D</code></td>
<td>
<p>numeric vector in the tropical projective torus or a matrix of points in the tropical projective torus; for matrices, rows are the points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single or set of normalized points with the first coordinate zero
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D &lt;-c(8,4,2)
normaliz.vector(D)

P &lt;-matrix(c(8,4,2,10,1,3,7,2,1),3,3,TRUE)
normaliz.vectors(P)

M&lt;-matrix(c(2,2,2,3,6,4,2,4,7),3,3,TRUE)
normaliz.polytope(M)

M &lt;- Sim_Trees15[1:3,]
normaliz.ultrametrics(M)

</code></pre>

<hr>
<h2 id='over_bet_HC'>Tropical cluster betweeness measure for each cluster in a set of hierarchical clusters</h2><span id='topic+over_bet_HC'></span>

<h3>Description</h3>

<p>This function calculates an overall betweenness measure based on tropical distance between a set of clusters derived from tropical hierarchical clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>over_bet_HC(A, V)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="over_bet_HC_+3A_a">A</code></td>
<td>
<p>matrix of tropical points; rows are points with the last column representing a numbered cluster assignment</p>
</td></tr>
<tr><td><code id="over_bet_HC_+3A_v">V</code></td>
<td>
<p>list of clusters defined as matrices derived from agglomerative or divisive hierarchical clustering</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of betweenness cluster measures
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>David Barnhill, Ruriko Yoshida (2023). Clustering Methods Over the Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>har&lt;-rbind(Sim_points[1:20,],Sim_points[51:70,])

V&lt;-Tropical.HC.AGNES(har, method="average")
inds&lt;-V[[2]][[38]]
over_bet_HC(har,inds)
</code></pre>

<hr>
<h2 id='over_bet_KM'>Tropical cluster betweeness measure for a each of a set of k-means derived set of clusters</h2><span id='topic+over_bet_KM'></span>

<h3>Description</h3>

<p>This function calculates an overall betweenness measure between a set of clusters derived from tropical k-means clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>over_bet_KM(A, C)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="over_bet_KM_+3A_a">A</code></td>
<td>
<p>matrix of tropical points; rows are points with the last column representing a numbered cluster assignment</p>
</td></tr>
<tr><td><code id="over_bet_KM_+3A_c">C</code></td>
<td>
<p>number of clusters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>betweenness cluster measure
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>David Barnhill, Ruriko Yoshida (2023). Clustering Methods Over the Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
hars&lt;-Sim_points
cls&lt;-c(rep(1,50),rep(2,50),rep(3,50))
cl_pt&lt;-cbind(hars,cls)

C&lt;-3
over_bet_KM(cl_pt,C)
</code></pre>

<hr>
<h2 id='Points.TLineSeg'>Sample k equally spaced points on a max-plus tropical line segment</h2><span id='topic+Points.TLineSeg'></span>

<h3>Description</h3>

<p>This function calculates k equally spaced points on a tropical line segment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Points.TLineSeg(D1, D2, k = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Points.TLineSeg_+3A_d1">D1</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
<tr><td><code id="Points.TLineSeg_+3A_d2">D2</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
<tr><td><code id="Points.TLineSeg_+3A_k">k</code></td>
<td>
<p>number of points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of k equally spaced points on a tropical line segment
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D1 &lt;-c(0,4,2)
D2 &lt;- c(0,7,-1)
p&lt;-Points.TLineSeg(D1, D2, k = 5)
</code></pre>

<hr>
<h2 id='pre.pplot.pro'>Projections of points onto a tropical triangle</h2><span id='topic+pre.pplot.pro'></span>

<h3>Description</h3>

<p>This function produces the a matrix of points projected onto a tropical triangle defined by the column space of a matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre.pplot.pro(S, D)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pre.pplot.pro_+3A_s">S</code></td>
<td>
<p>matrix of points representing a tropical polytope; rows are the vertices</p>
</td></tr>
<tr><td><code id="pre.pplot.pro_+3A_d">D</code></td>
<td>
<p>data points in the tropical projective torus</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of points representing projections of the points in D (row vectors) onto S
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>s &lt;- 3 #number of vertices.  Here it is a tropical triangle
d &lt;- 3 ## dimension
N &lt;- 100 ## sample size
D &lt;- matrix(rep(0, N*d), N, d)
D[, 1] &lt;- rnorm(N, mean = 5, sd = 5)
D[, 2] &lt;- rnorm(N, mean = -5, sd = 5)
D[, 3] &lt;- rnorm(N, mean = 0, sd = 5)

index &lt;- sample(1:N, s)
S &lt;- D[index,]

DD &lt;- pre.pplot.pro(S, D)
</code></pre>

<hr>
<h2 id='prob.class'>Estimated probability for binary class assignment</h2><span id='topic+prob.class'></span>

<h3>Description</h3>

<p>Estimates the probability that an observation x belongs to class 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prob.class(pars, x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prob.class_+3A_pars">pars</code></td>
<td>
<p>vector of parameters, which can be decomposed as two normal vectors and two scaling parameters and has dimension 2*e+2</p>
</td></tr>
<tr><td><code id="prob.class_+3A_x">x</code></td>
<td>
<p>vector of dimension e</p>
</td></tr>
</table>


<h3>Value</h3>

<p>real number
</p>


<h3>Author(s)</h3>

<p>Georgios Aliatimis <a href="mailto:g.aliatimis@lancaster.ac.uk">g.aliatimis@lancaster.ac.uk</a>
</p>


<h3>References</h3>

<p>Aliatimis, Georgios, Ruriko Yoshida, Burak Boyaci and James A. Grant (2023). Tropical Logistic Regression on Space of Phylogenetic Trees
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ROCR)
T0 = Sim_Trees15
T1 = Sim_Trees25
D  = rbind(T0,T1)
Y = c(rep(0,dim(T0)[1]), rep(1,dim(T1)[1]))
N = length(Y)
set.seed(1)
train_set = sample(N,floor(0.8 * N)) ## 80/20 train-test split
pars &lt;- trop.logistic.regression(D[train_set,],Y[train_set], penalty=1e4)
test_set = (1:N)[-train_set]
Y.hat &lt;- rep(0, length(test_set))
for(i in 1:length(test_set))   Y.hat[i] &lt;- prob.class(pars, D[test_set[i],])
Logit.ROC &lt;- performance(prediction(Y.hat, Y[test_set]), measure="tpr", x.measure="fpr")
plot(Logit.ROC, lwd = 2, main = "ROC Curve for Logistic Regression Model")
print(paste("Logit.AUC=", performance(prediction(Y.hat, Y[test_set]), measure="auc")@y.values))

</code></pre>

<hr>
<h2 id='project_pi'>Project a point on the tropical projective torus onto a tropical polytope.</h2><span id='topic+project_pi'></span>

<h3>Description</h3>

<p>This function projects points in the tropical projective torus onto a tropical polytope based on tropical distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>project_pi(D_s, D)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="project_pi_+3A_d_s">D_s</code></td>
<td>
<p>matrix where each row is a point defining a tropical polytope</p>
</td></tr>
<tr><td><code id="project_pi_+3A_d">D</code></td>
<td>
<p>point to be projected onto D_s</p>
</td></tr>
</table>


<h3>Value</h3>

<p>projection of point D onto the tropical polytope defined by D_s
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D_s &lt;-matrix(c(0,0,0,0,2,5,0,3,1),3,3,TRUE)
D &lt;- c(0,7,-1)
project_pi(D_s,D)
</code></pre>

<hr>
<h2 id='pw.trop.dist'>Constructs the dissimilarity matrix for a set of ultrametrics</h2><span id='topic+pw.trop.dist'></span>

<h3>Description</h3>

<p>Constructs the dissimilarity matrix based on the tropical distance between points in a dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pw.trop.dist(D1, D2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pw.trop.dist_+3A_d1">D1</code></td>
<td>
<p>matrix of ultrametrics</p>
</td></tr>
<tr><td><code id="pw.trop.dist_+3A_d2">D2</code></td>
<td>
<p>matrix of ultrametrics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix; dissimilarity matrix showing the tropical pairwise distance between each point
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>References</h3>

<p>Weyenberg, G., Huggins, P., Schardl, C., Howe, D. K., &amp; Yoshida, R. (2014). kdetrees: Nonparametric Estimation of Phylogenetic Tree Distributions. In Bioinformatics.
</p>
<p>Yoshida, Ruriko, David Barnhill, Keiji Miura and Daniel Howe (2022). Tropical Density Estimation of Phylogenetic Trees.
</p>
<p><a href="https://github.com/grady/kdetrees/blob/master/R/dist.diss.R">https://github.com/grady/kdetrees/blob/master/R/dist.diss.R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
T1&lt;-Sim_Trees15
T2&lt;-Sim_Trees25
D &lt;- rbind(T1, T2[1,])
pw.trop.dist(D, D)

</code></pre>

<hr>
<h2 id='rounding'>Remove all tentacles from a tropical simplex</h2><span id='topic+rounding'></span>

<h3>Description</h3>

<p>This function removes all tentacles from a tropical simplex. The remaining portion is a full-dimensional tropical polytope known as the trunk of the tropical polytope.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rounding(P)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rounding_+3A_p">P</code></td>
<td>
<p>matrix of points defining a tropical simplex. Rows are the points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of points defining only the full-dimensional element (the trunk) of a tropical polytope; rows are points
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>Barnhill, David, Ruriko Yoshida and Keiji Miura (2023). Maximum Inscribed and Minimum Enclosing Tropical Balls of Tropical Polytopes and Applications to Volume Estimation and Uniform Sampling.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P&lt;-matrix(c(0,-1,1,0,0,0,0,1,-1),3,3,TRUE)
BP&lt;-min_enc_ball(P)
RP&lt;-rounding(P)
BRP&lt;-min_enc_ball(RP)
</code></pre>

<hr>
<h2 id='sigmoid'>Sigmoid function</h2><span id='topic+sigmoid'></span>

<h3>Description</h3>

<p>Returns the sigmoid function valuation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sigmoid(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sigmoid_+3A_x">x</code></td>
<td>
<p>real number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sigmoid function value at x
</p>


<h3>Author(s)</h3>

<p>Georgios Aliatimis <a href="mailto:g.aliatimis@lancaster.ac.uk">g.aliatimis@lancaster.ac.uk</a>
</p>


<h3>References</h3>

<p>Aliatimis, Georgios, Ruriko Yoshida, Burak Boyaci and James A. Grant (2023). Tropical Logistic Regression on Space of Phylogenetic Trees
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sigmoid(0) # 0.5
</code></pre>

<hr>
<h2 id='Sim_points'>Simulated points over the tropical projective torus</h2><span id='topic+Sim_points'></span>

<h3>Description</h3>

<p>150 points generated using Gaussian-like Hit-and-Run sampling with three separate pairs of location and scale parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sim_points
</code></pre>


<h3>Format</h3>



<h4><code>Sim_points</code></h4>

<p>A 150 x 3 matrix where each row is a point in the tropical projective torus
</p>


<hr>
<h2 id='Sim_Trees1'>Six data sets of phylogenetic trees data simulated from the Coalescant model.</h2><span id='topic+Sim_Trees1'></span><span id='topic+Sim_Trees1025'></span><span id='topic+Sim_Trees105'></span><span id='topic+Sim_Trees11'></span><span id='topic+Sim_Trees12'></span><span id='topic+Sim_Trees15'></span><span id='topic+Sim_Trees110'></span>

<h3>Description</h3>

<p>Six data sets of 1000 gene trees simulated from the Coalescant model based on a specified species with each data set
possessing a ratio of species depth to effective population of 0.25, 0.5, 1, 2, 5, and 10.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sim_Trees1025

Sim_Trees105

Sim_Trees11

Sim_Trees12

Sim_Trees15

Sim_Trees110
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>

<hr>
<h2 id='Sim_Trees2'>Six data sets of phylogenetic trees data simulated from the Coalescant model.</h2><span id='topic+Sim_Trees2'></span><span id='topic+Sim_Trees2025'></span><span id='topic+Sim_Trees205'></span><span id='topic+Sim_Trees21'></span><span id='topic+Sim_Trees22'></span><span id='topic+Sim_Trees25'></span><span id='topic+Sim_Trees210'></span>

<h3>Description</h3>

<p>Six data sets of 1000 gene trees simulated from the Coalescant model based on a specified species with each data set
possessing a ratio of species depth to effective population of 0.25, 0.5, 1, 2, 5, and 10.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sim_Trees2025

Sim_Trees205

Sim_Trees21

Sim_Trees22

Sim_Trees25

Sim_Trees210
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 1000 rows and 45 columns.
</p>

<hr>
<h2 id='tdets'>Calculate the tropical determinant of a square matrix.</h2><span id='topic+tdets'></span>

<h3>Description</h3>

<p>This function calculates the tropical determinant (or singularity) of a square matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tdets(P)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tdets_+3A_p">P</code></td>
<td>
<p>matrix of points defining a tropical polytope. Rows are the points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing the value of the determinant and reordered matrix P
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P&lt;-matrix(c(0,0,0,0,2,5,0,3,1),3,3,TRUE)
tdets(P)
</code></pre>

<hr>
<h2 id='TKmeans'>K-means clustering over the tropical projective torus</h2><span id='topic+TKmeans'></span>

<h3>Description</h3>

<p>This function performs k-means clustering over the tropical projective torus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TKmeans(A, C, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TKmeans_+3A_a">A</code></td>
<td>
<p>matrix of points defining a tropical polytope; rows are the tropical points</p>
</td></tr>
<tr><td><code id="TKmeans_+3A_c">C</code></td>
<td>
<p>number of clusters</p>
</td></tr>
<tr><td><code id="TKmeans_+3A_m">M</code></td>
<td>
<p>maximum number of iterations of algorithm to find cluster centroids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with matrix of observation classified by centroid; matrix of centroid coordinates; number of iterations used
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>David Barnhill, Ruriko Yoshida (2023). Clustering Methods Over the Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P &lt;-Sim_points
C&lt;-3
M&lt;-10
res&lt;-TKmeans(P,C,M)
try&lt;-res[[1]]
cen&lt;-res[[2]]
plot(try[,2],try[,3],col=try[,4],asp=1)
plot(try[,2],try[,3],col=try[,4],asp=1,xlab='x2',ylab='x3')
points(cen[,2],cen[,3],col=c('purple','hotpink','orange'),pch=19)
</code></pre>

<hr>
<h2 id='TLineSeg'>Construct a max-plus tropical line segment between two points</h2><span id='topic+TLineSeg'></span>

<h3>Description</h3>

<p>This function constructs a max-plus tropical line segment between two points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TLineSeg(D1, D2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TLineSeg_+3A_d1">D1</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
<tr><td><code id="TLineSeg_+3A_d2">D2</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of points defining the tropical line segment
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D1 &lt;-c(0,4,2)
D2 &lt;- c(0,7,-1)
TLineSeg(D1, D2)
</code></pre>

<hr>
<h2 id='TLineSeg_min'>Construct a min-plus tropical line segment accor</h2><span id='topic+TLineSeg_min'></span>

<h3>Description</h3>

<p>This function constructs a min-plus tropical line segment between two points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TLineSeg_min(D1, D2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TLineSeg_min_+3A_d1">D1</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
<tr><td><code id="TLineSeg_min_+3A_d2">D2</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of points defining the tropical line segment
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D1 &lt;-c(0,4,2)
D2 &lt;- c(0,7,-1)
TLineSeg_min(D1, D2)
</code></pre>

<hr>
<h2 id='tree.to.vector'>Phylogenetic tree to vector</h2><span id='topic+tree.to.vector'></span>

<h3>Description</h3>

<p>A tree is converted to a vector of pairwise distances between leaves.
Distance between leaves is defined as the cophenetic distance between them.
Normalization is applied so that the maximum distance in the vector output is 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree.to.vector(tree, normalization = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tree.to.vector_+3A_tree">tree</code></td>
<td>
<p>phylogenetic tree</p>
</td></tr>
<tr><td><code id="tree.to.vector_+3A_normalization">normalization</code></td>
<td>
<p>logical; normalize the tree if TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of pairwise distances in R^(m choose 2), where m is the number of leaves
</p>


<h3>Author(s)</h3>

<p>Georgios Aliatimis <a href="mailto:g.aliatimis@lancaster.ac.uk">g.aliatimis@lancaster.ac.uk</a>
</p>


<h3>References</h3>

<p>Aliatimis, Georgios, Ruriko Yoshida, Burak Boyaci, James A. Grant (2023). Tropical Logistic Regression on Space of Phylogenetic Trees
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tree &lt;- ape::read.tree(text='((A:1, B:1):2, (C:1, D:1):2);')
tree.to.vector(tree)
</code></pre>

<hr>
<h2 id='trop_bal.vert'>Calculate the minimum generating vertex set of a tropical ball</h2><span id='topic+trop_bal.vert'></span>

<h3>Description</h3>

<p>This function calculates the coordinates of the minimum or entire vertex set of a tropical ball given a center point
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop_bal.vert(x, d, al = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trop_bal.vert_+3A_x">x</code></td>
<td>
<p>matrix where each row is a point defining a tropical polytope</p>
</td></tr>
<tr><td><code id="trop_bal.vert_+3A_d">d</code></td>
<td>
<p>radius of the tropical ball in terms of tropical distance</p>
</td></tr>
<tr><td><code id="trop_bal.vert_+3A_al">al</code></td>
<td>
<p>logical; TRUE or FALSE to determine whether to enumerate all vertices of the tropical ball</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of normalized tropical points defining the tropical ball. Rows are the points
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>Barnhill, David, Ruriko Yoshida and Keiji Miura (2023). Maximum Inscribed and Minimum Enclosing Tropical Balls of Tropical Polytopes and Applications to Volume Estimation and Uniform Sampling.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;-c(0,3,7,5)
d &lt;- 2
trop_bal.vert(x,d)
trop_bal.vert(x,d,al=TRUE)
</code></pre>

<hr>
<h2 id='Trop_ball'>Visualize a Tropical ball in 2D or 3D</h2><span id='topic+Trop_ball'></span>

<h3>Description</h3>

<p>This function constructs a visualization of a 2D or 3D tropical ball
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Trop_ball(
  v,
  d,
  a = 1,
  cls = "black",
  cent.col = "black",
  fil = TRUE,
  plt = TRUE,
  bord = "black"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Trop_ball_+3A_v">v</code></td>
<td>
<p>center of tropical ball; numeric vector of length 3 or 4</p>
</td></tr>
<tr><td><code id="Trop_ball_+3A_d">d</code></td>
<td>
<p>radius of tropical ball</p>
</td></tr>
<tr><td><code id="Trop_ball_+3A_a">a</code></td>
<td>
<p>shading level; 1 is opaque</p>
</td></tr>
<tr><td><code id="Trop_ball_+3A_cls">cls</code></td>
<td>
<p>string indicating color of interior of ball</p>
</td></tr>
<tr><td><code id="Trop_ball_+3A_cent.col">cent.col</code></td>
<td>
<p>string indicating color of center point</p>
</td></tr>
<tr><td><code id="Trop_ball_+3A_fil">fil</code></td>
<td>
<p>logical for 3D plots; if TRUE 2D facets of 3D ball fill in color of cls parameter</p>
</td></tr>
<tr><td><code id="Trop_ball_+3A_plt">plt</code></td>
<td>
<p>logical; indicates plot a new object; defaults to TRUE; if FALSE, overlays the ball on existing plot</p>
</td></tr>
<tr><td><code id="Trop_ball_+3A_bord">bord</code></td>
<td>
<p>string indicating color of border of ball (only for 2D plots)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>2D or 3D visualization of tropical ball
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>v &lt;-c(0,0,0)
d &lt;- 2
Trop_ball(v,d,a=.1,cls='white',cent.col='black',fil=TRUE,plt=TRUE,bord='black')
v &lt;-c(0,0,0,0)
d &lt;- 2
Trop_ball(v,d,a=1,cls='red',cent.col='black',fil=FALSE,plt=TRUE)
</code></pre>

<hr>
<h2 id='Trop_FW'>Calculate the tropical Fermat-Weber point</h2><span id='topic+Trop_FW'></span>

<h3>Description</h3>

<p>This function calculates the Fermat-Weber point for a tropical polytope
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Trop_FW(A)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Trop_FW_+3A_a">A</code></td>
<td>
<p>matrix with normalized tropical points as rows</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector providing the tropical Fermat-Weber point for the tropical polytope
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>Lin, Bo and Ruriko Yoshida (2016). Tropical Fermat-Weber Points. SIAM J. Discret. Math. 32: 1229-1245.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P &lt;-matrix(c(0,0,0,0,2,5,0,3,1),3,3,TRUE)

Trop_FW(P)

</code></pre>

<hr>
<h2 id='Trop_Volume'>Estimate the volume of a tropical polytope</h2><span id='topic+Trop_Volume'></span>

<h3>Description</h3>

<p>This function uses tropical HAR with a uniform target distribution to estimate the volume of a tropical polytope
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Trop_Volume(B, P, x0, S, I, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Trop_Volume_+3A_b">B</code></td>
<td>
<p>matrix of points defining a minimum enclosing ball for a polytope P; rows are the points</p>
</td></tr>
<tr><td><code id="Trop_Volume_+3A_p">P</code></td>
<td>
<p>matrix of points defining a tropical polytope; rows are the points</p>
</td></tr>
<tr><td><code id="Trop_Volume_+3A_x0">x0</code></td>
<td>
<p>initial point used for the HAR sampler</p>
</td></tr>
<tr><td><code id="Trop_Volume_+3A_s">S</code></td>
<td>
<p>number of points to sample from the minimum enclosing ball</p>
</td></tr>
<tr><td><code id="Trop_Volume_+3A_i">I</code></td>
<td>
<p>number of iterations for the HAR sampler</p>
</td></tr>
<tr><td><code id="Trop_Volume_+3A_r">R</code></td>
<td>
<p>radius of the minimum enclosing tropical ball</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing ratio of points falling in P; volume of the tropical ball; volume estimate of P
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>Barnhill, David, Ruriko Yoshida and Keiji Miura (2023). Maximum Inscribed and Minimum Enclosing Tropical Balls of Tropical Polytopes and Applications to Volume Estimation and Uniform Sampling.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P &lt;-matrix(c(0,0,0,0,3,1,0,2,5),3,3,TRUE)
BR&lt;-min_enc_ball(P)
B&lt;-trop_bal.vert(BR[[1]],BR[[2]])
x0&lt;-c(0,1.5,.4)
S&lt;-200
I&lt;-50
R&lt;-BR[[2]]
Trop_Volume(B,P,x0,S,I,R)
</code></pre>

<hr>
<h2 id='trop_wi_dist'>Tropical within-cluster measure</h2><span id='topic+trop_wi_dist'></span>

<h3>Description</h3>

<p>This function calculates a within cluster measure by measuring the pairwise tropical distance between points in the cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop_wi_dist(D1, method = "avg")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trop_wi_dist_+3A_d1">D1</code></td>
<td>
<p>matrix of tropical points; rows are points</p>
</td></tr>
<tr><td><code id="trop_wi_dist_+3A_method">method</code></td>
<td>
<p>metric to measure; &quot;avg&quot; is the average pairwise tropical distance; &quot;max&quot; is the maximum pairwise tropical distance</p>
</td></tr>
</table>


<h3>Value</h3>

<p>within cluster measure
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>David Barnhill, Ruriko Yoshida (2023). Clustering Methods Over the Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D&lt;-Sim_points
avg.m&lt;-trop_wi_dist(D, method='avg')
max.m&lt;-trop_wi_dist(D, method='avg')
</code></pre>

<hr>
<h2 id='trop.dist'>Compute the tropical distance</h2><span id='topic+trop.dist'></span>

<h3>Description</h3>

<p>This function computes the tropical distance between two points in the tropical projective torus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop.dist(D1, D2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trop.dist_+3A_d1">D1</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
<tr><td><code id="trop.dist_+3A_d2">D2</code></td>
<td>
<p>point in the tropical projective torus</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tropical distance between D1 and D2
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D1 &lt;-c(0,4,2)
D2 &lt;- c(0,7,-1)
trop.dist(D1, D2)

</code></pre>

<hr>
<h2 id='trop.dist.hyp_max'>Calculate the tropical distance to a max-tropical hyperplane</h2><span id='topic+trop.dist.hyp_max'></span>

<h3>Description</h3>

<p>Calculate the tropical distance to a max-tropical hyperplane
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop.dist.hyp_max(O, x0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trop.dist.hyp_max_+3A_o">O</code></td>
<td>
<p>normal vector of a tropical hyperplane; numeric vector</p>
</td></tr>
<tr><td><code id="trop.dist.hyp_max_+3A_x0">x0</code></td>
<td>
<p>point of interest; numeric vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tropical distance to max-plus tropical hyperplane
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>O &lt;-c(0,-1,-1)
x0 &lt;- c(0,-2,-8)
trop.dist.hyp_max(O,x0)
</code></pre>

<hr>
<h2 id='trop.dist.hyp_min'>Calculate the tropical distance to a min-tropical hyperplane</h2><span id='topic+trop.dist.hyp_min'></span>

<h3>Description</h3>

<p>Calculate the tropical distance to a min-tropical hyperplane
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop.dist.hyp_min(O, x0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trop.dist.hyp_min_+3A_o">O</code></td>
<td>
<p>normal vector of a tropical hyperplane; numeric vector</p>
</td></tr>
<tr><td><code id="trop.dist.hyp_min_+3A_x0">x0</code></td>
<td>
<p>point of interest; numeric vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tropical distance to min-plus tropical hyperplane
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>O &lt;-c(0,-1,-1)
x0 &lt;- c(0,-2,-8)
trop.dist.hyp_min(O,x0)
</code></pre>

<hr>
<h2 id='trop.logistic.regression'>Tropical Logistic Regression</h2><span id='topic+trop.logistic.regression'></span>

<h3>Description</h3>

<p>Performs tropical logistic regression, by finding the optimal statistical parameters for the
training dataset (D,Y), where D is the matrix of covariates and Y is the binary response vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop.logistic.regression(D, Y, penalty = 0, model_type = "two_species")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trop.logistic.regression_+3A_d">D</code></td>
<td>
<p>matrix of dimension N*e, where N is the number of observations which lie in R^e</p>
</td></tr>
<tr><td><code id="trop.logistic.regression_+3A_y">Y</code></td>
<td>
<p>binary vector with dimension N, with each component corresponding to an observation</p>
</td></tr>
<tr><td><code id="trop.logistic.regression_+3A_penalty">penalty</code></td>
<td>
<p>scalar; positive real number</p>
</td></tr>
<tr><td><code id="trop.logistic.regression_+3A_model_type">model_type</code></td>
<td>
<p>string; options are &quot;two-species&quot; (default),  &quot;one-species&quot;, &quot;general&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector; optimal model parameters (two normal vectors and two scaling factors)
</p>


<h3>Author(s)</h3>

<p>Georgios Aliatimis <a href="mailto:g.aliatimis@lancaster.ac.uk">g.aliatimis@lancaster.ac.uk</a>
</p>


<h3>References</h3>

<p>Aliatimis, Georgios, Ruriko Yoshida, Burak Boyaci and James A. Grant (2023). Tropical Logistic Regression on Space of Phylogenetic Trees.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ROCR)
T0 = Sim_Trees15
T1 = Sim_Trees25
D  = rbind(T0,T1)
Y = c(rep(0,dim(T0)[1]), rep(1,dim(T1)[1]))
N = length(Y)
set.seed(1)
train_set = sample(N,floor(0.8 * N)) ## 80/20 train-test split
pars &lt;- trop.logistic.regression(D[train_set,],Y[train_set], penalty=1e4)
test_set = (1:N)[-train_set]
Y.hat &lt;- rep(0, length(test_set))
for(i in 1:length(test_set))   Y.hat[i] &lt;- prob.class(pars, D[test_set[i],])
Logit.ROC &lt;- performance(prediction(Y.hat, Y[test_set]), measure="tpr", x.measure="fpr")
plot(Logit.ROC, lwd = 2, main = "ROC Curve for Logistic Regression Model")
print(paste("Logit.AUC=", performance(prediction(Y.hat, Y[test_set]), measure="auc")@y.values))

</code></pre>

<hr>
<h2 id='trop.tri.plot.w.pts'>Plotting PCA-derived tropical triangles</h2><span id='topic+trop.tri.plot.w.pts'></span>

<h3>Description</h3>

<p>This function conducts tropical PCA to find the best fit tropical triangle given data defined in the tropical projective torus.
It employs the vertex HAR with extrapolation sampler to sample points to determine the vertices of the tropical triangle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop.tri.plot.w.pts(S, D)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trop.tri.plot.w.pts_+3A_s">S</code></td>
<td>
<p>inital set of vertices for the tropical triangle</p>
</td></tr>
<tr><td><code id="trop.tri.plot.w.pts_+3A_d">D</code></td>
<td>
<p>matrix of data where each row is an observation in the tropical projective torus</p>
</td></tr>
</table>


<h3>Value</h3>

<p>rendering of tropical triangle saved to current directory
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
s &lt;- 3 #number of vertices.  Here it is a tropical triangle
d &lt;- 3 ## dimension
N &lt;- 100 ## sample size
V &lt;- matrix(c(100, 0, 0, 0, 100, 0, 0, 0, 100, -100, 0, 0, 0, -100, 0, 0, 0, -100), 6, 3, TRUE)
D &lt;- matrix(rep(0, N*d), N, d)
D[, 1] &lt;- rnorm(N, mean = 5, sd = 5)
D[, 2] &lt;- rnorm(N, mean = -5, sd = 5)
D[, 3] &lt;- rnorm(N, mean = 0, sd = 5)
index &lt;- sample(1:N, s)
S &lt;- D[index,]
res &lt;- tropical.PCA.Polytope(S, D, V, I = 1000,50)
DD &lt;- pre.pplot.pro(res[[2]], res[[3]])
trop.tri.plot.w.pts(normaliz.ultrametrics(res[[2]]), DD)
</code></pre>

<hr>
<h2 id='tropical.Gaussian'>Tropical Gaussian sampling about a center of mass</h2><span id='topic+tropical.Gaussian'></span><span id='topic+tropical.gaussian'></span>

<h3>Description</h3>

<p>This function is a Gaussian-like HAR sampler about a center of mass denoted by a location parameter with scale parameter in terms of the tropical distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropical.gaussian(D_s, x0, I = 1, M, S)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropical.Gaussian_+3A_d_s">D_s</code></td>
<td>
<p>matrix of vertices of a tropical simplex; each row is a vertex</p>
</td></tr>
<tr><td><code id="tropical.Gaussian_+3A_x0">x0</code></td>
<td>
<p>initial point for sampler, numeric vector</p>
</td></tr>
<tr><td><code id="tropical.Gaussian_+3A_i">I</code></td>
<td>
<p>number of states in Markov chain</p>
</td></tr>
<tr><td><code id="tropical.Gaussian_+3A_m">M</code></td>
<td>
<p>location parameter; numeric vector indicating centroid</p>
</td></tr>
<tr><td><code id="tropical.Gaussian_+3A_s">S</code></td>
<td>
<p>scale parameter; in terms of tropical distance</p>
</td></tr>
</table>


<h3>Value</h3>

<p>next sampled point from the tropical polytope
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D_s &lt;-matrix(c(0,0,0,0,10,0,0,0,10),3,3,TRUE)
x0 &lt;- c(0,0,0)
M &lt;- c(0,5,5)
S &lt;- 2
tropical.gaussian(D_s, x0, I = 50,M,S)
</code></pre>

<hr>
<h2 id='tropical.Gaussian.MH'>Gaussian-like sampling using Metropolis filter</h2><span id='topic+tropical.Gaussian.MH'></span><span id='topic+trop.Gaussian.MH'></span><span id='topic+trop.Gaussian.MH.square'></span>

<h3>Description</h3>

<p>This function samples points on a tropical line segment about a location parameter for a given scale parameter defined in terms of tropical distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trop.Gaussian.MH(D, x0, mu, s, n, I = 50)

trop.Gaussian.MH.square(D, x0, mu, s, n, I = 50)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropical.Gaussian.MH_+3A_d">D</code></td>
<td>
<p>matrix of vertices of a tropical polytope; each row is a vertex</p>
</td></tr>
<tr><td><code id="tropical.Gaussian.MH_+3A_x0">x0</code></td>
<td>
<p>initial point for sampler, numeric vector</p>
</td></tr>
<tr><td><code id="tropical.Gaussian.MH_+3A_mu">mu</code></td>
<td>
<p>location parameter; numeric vector</p>
</td></tr>
<tr><td><code id="tropical.Gaussian.MH_+3A_s">s</code></td>
<td>
<p>scale parameter; scalar</p>
</td></tr>
<tr><td><code id="tropical.Gaussian.MH_+3A_n">n</code></td>
<td>
<p>number of points to sample</p>
</td></tr>
<tr><td><code id="tropical.Gaussian.MH_+3A_i">I</code></td>
<td>
<p>states in Markov chain</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of n sampled points where each point is a row
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>Yoshida, Ruriko, Keiji Miura and David Barnhill (2022). Hit and Run Sampling from Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D &lt;-matrix(c(0,0,0,0,10,0,0,0,10),3,3,TRUE)
x0 &lt;- c(0,0,0)
mu&lt;-c(0,5,5)
s&lt;-1
n&lt;-10
trop.Gaussian.MH(D, x0, mu, s, n, I=50)
trop.Gaussian.MH.square(D, x0,mu, s, n, I=50)
</code></pre>

<hr>
<h2 id='Tropical.HC.AGNES'>Agglomerative (AGNES) tropical hierarchical clustering</h2><span id='topic+Tropical.HC.AGNES'></span>

<h3>Description</h3>

<p>This function performs agglomerative (AGNES) hierarchical clustering over the space of ultrametrics defining the space of equidistant trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tropical.HC.AGNES(D, method = "average")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Tropical.HC.AGNES_+3A_d">D</code></td>
<td>
<p>matrix of points defining a tropical polytope.  Rows are the tropical points</p>
</td></tr>
<tr><td><code id="Tropical.HC.AGNES_+3A_method">method</code></td>
<td>
<p>linkage method: &quot;average&quot;, &quot;min&quot;, or &quot;max&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of distances in when merges occur; list of indices of points in each cluster
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>David Barnhill, Ruriko Yoshida (2023). Clustering Methods Over the Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>P &lt;-Sim_points
Tropical.HC.AGNES(P, method="average")

</code></pre>

<hr>
<h2 id='tropical.KDE'>Tropical Kernel Density Estimation of Phylogenetic Trees</h2><span id='topic+tropical.KDE'></span>

<h3>Description</h3>

<p>This function calculates a non-parametric density estimate of a tree over the space of phylogenetic trees on m leaves.
It mimics classical kernel density estimation by using a Gaussian kernel in conjunction with tropical distance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropical.KDE(D, n, sigma, h = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropical.KDE_+3A_d">D</code></td>
<td>
<p>matrix of phylogenetic tree observations as ultrametrics</p>
</td></tr>
<tr><td><code id="tropical.KDE_+3A_n">n</code></td>
<td>
<p>number of leaves for each tree</p>
</td></tr>
<tr><td><code id="tropical.KDE_+3A_sigma">sigma</code></td>
<td>
<p>bandwidth parameter based on tropical distance</p>
</td></tr>
<tr><td><code id="tropical.KDE_+3A_h">h</code></td>
<td>
<p>height of the tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing center point and radius of minimum enclosing ball of P
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>References</h3>

<p>Weyenberg, G., Huggins, P., Schardl, C., Howe, D. K., &amp; Yoshida, R. (2014). kdetrees: Nonparametric Estimation of Phylogenetic Tree Distributions. In Bioinformatics.
</p>
<p>Yoshida, Ruriko, David Barnhill, Keiji Miura and Daniel Howe (2022). Tropical Density Estimation of Phylogenetic Trees.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
T1&lt;-Sim_Trees15
T2&lt;-Sim_Trees25
D &lt;- rbind(T1, T2[1,])
T &lt;- dim(D)[1]
X &lt;- 1:T
M &lt;- pw.trop.dist(D, D)
sigma &lt;- bw.nn(M)
P_5 &lt;- tropical.KDE(D, n, sigma, h = 2)
Q5 &lt;- P_5[T]

</code></pre>

<hr>
<h2 id='tropical.PCA'>Tropical principal component analysis (PCA) on over tropical projective torus</h2><span id='topic+tropical.PCA'></span><span id='topic+tropical.PCA.Polytope'></span>

<h3>Description</h3>

<p>This function conducts tropical PCA to find the best fit tropical triangle given data defined in the tropical projective torus.
It employs the vertex HAR with extrapolation sampler to sample points to determine the vertices of the tropical triangle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tropical.PCA.Polytope(S, D, V, I = 1, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tropical.PCA_+3A_s">S</code></td>
<td>
<p>inital set of vertices for the tropical triangle</p>
</td></tr>
<tr><td><code id="tropical.PCA_+3A_d">D</code></td>
<td>
<p>matrix of data where each row is an observation in the tropical projective torus</p>
</td></tr>
<tr><td><code id="tropical.PCA_+3A_v">V</code></td>
<td>
<p>matrix of vertices defining a polytope encompassing D</p>
</td></tr>
<tr><td><code id="tropical.PCA_+3A_i">I</code></td>
<td>
<p>number of iterations to perform</p>
</td></tr>
<tr><td><code id="tropical.PCA_+3A_k">k</code></td>
<td>
<p>number of iterations for the HAR sampler</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with the sum of residuals
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>References</h3>

<p>Page, Robert and others (2020), Tropical principal component analysis on the space of phylogenetic trees, Bioinformatics, Volume 36, Issue 17, Pages 4590–4598.
</p>
<p>Yoshida, R., Zhang, L. &amp; Zhang, X (2019). Tropical Principal Component Analysis and Its Application to Phylogenetics. Bull Math Biol 81, 568–597.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
s &lt;- 3 #number of vertices.  Here it is a tropical triangle
d &lt;- 3 ## dimension
N &lt;- 100 ## sample size
V &lt;- matrix(c(100, 0, 0, 0, 100, 0, 0, 0, 100, -100, 0, 0, 0, -100, 0, 0, 0, -100), 6, 3, TRUE)
D &lt;- matrix(rep(0, N*d), N, d)
D[, 1] &lt;- rnorm(N, mean = 5, sd = 5)
D[, 2] &lt;- rnorm(N, mean = -5, sd = 5)
D[, 3] &lt;- rnorm(N, mean = 0, sd = 5)
index &lt;- sample(1:N, s)
S &lt;- D[index,]
DD &lt;- pre.pplot.pro(S, D)
for(i in 1:N)
 DD[i, ] &lt;- normaliz.vector(DD[i, ])

res &lt;- tropical.PCA.Polytope(S, D, V, I = 1000,50)
DD &lt;- pre.pplot.pro(res[[2]], res[[3]])
trop.tri.plot.w.pts(normaliz.ultrametrics(res[[2]]), DD)

</code></pre>

<hr>
<h2 id='TropicalPolytope.extrapolation.HAR'>Vertex HAR with extrapolation MCMC with uniform target distribution</h2><span id='topic+TropicalPolytope.extrapolation.HAR'></span>

<h3>Description</h3>

<p>This function samples points uniformly the space defined by a tropical simplex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TropicalPolytope.extrapolation.HAR(D_s, x0, I = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TropicalPolytope.extrapolation.HAR_+3A_d_s">D_s</code></td>
<td>
<p>matrix of vertices of a tropical simplex; each row is a vertex</p>
</td></tr>
<tr><td><code id="TropicalPolytope.extrapolation.HAR_+3A_x0">x0</code></td>
<td>
<p>initial point for sampler, numeric vector</p>
</td></tr>
<tr><td><code id="TropicalPolytope.extrapolation.HAR_+3A_i">I</code></td>
<td>
<p>number of states in Markov chain</p>
</td></tr>
</table>


<h3>Value</h3>

<p>next sampled point from the tropical polytope
</p>


<h3>Author(s)</h3>

<p>David Barnhill <a href="mailto:david.barnhill@nps.edu">david.barnhill@nps.edu</a>
</p>


<h3>References</h3>

<p>Yoshida, Ruriko, Keiji Miura and David Barnhill (2022). Hit and Run Sampling from Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D_s &lt;-matrix(c(0,0,0,0,10,0,0,0,10),3,3,TRUE)
x0 &lt;- c(0,0,0)
TropicalPolytope.extrapolation.HAR(D_s, x0, I = 50)
</code></pre>

<hr>
<h2 id='Ultrametrics.HAR'>Hit-and-Run Sampler for the space of ultrametrics</h2><span id='topic+Ultrametrics.HAR'></span>

<h3>Description</h3>

<p>This sampler samples a point in the space of ultrametrics where each point represents an equidistant tree on n leaves
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ultrametrics.HAR(x0, n, I = 1, h = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ultrametrics.HAR_+3A_x0">x0</code></td>
<td>
<p>an equidistant tree defined as ultrametric</p>
</td></tr>
<tr><td><code id="Ultrametrics.HAR_+3A_n">n</code></td>
<td>
<p>number of leaves for the equidistant tree</p>
</td></tr>
<tr><td><code id="Ultrametrics.HAR_+3A_i">I</code></td>
<td>
<p>number of states in the Markov chain</p>
</td></tr>
<tr><td><code id="Ultrametrics.HAR_+3A_h">h</code></td>
<td>
<p>height of phylogenetic tree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>point in the space of ultrametrics over n leaves
</p>


<h3>Author(s)</h3>

<p>Ruriko Yoshida <a href="mailto:ryoshida@nps.edu">ryoshida@nps.edu</a>
</p>


<h3>References</h3>

<p>Yoshida, Ruriko, Keiji Miura and David Barnhill (2022). Hit and Run Sampling from Tropically Convex Sets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x0 &lt;-Sim_Trees15[1,]
n&lt;-10

Ultrametrics.HAR(x0, n, I = 50, h = 1)

</code></pre>

<hr>
<h2 id='vector.to.equidistant.tree'>Vector to equidistant tree</h2><span id='topic+vector.to.equidistant.tree'></span>

<h3>Description</h3>

<p>A vector of pairwise distances is used to reconstruct the corresponding equidistant tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vector.to.equidistant.tree(omega)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vector.to.equidistant.tree_+3A_omega">omega</code></td>
<td>
<p>vector of pairwise distances in R^(m choose 2), where m is the number of leaves</p>
</td></tr>
</table>


<h3>Value</h3>

<p>equidistant phylogenetic tree
</p>


<h3>Author(s)</h3>

<p>Georgios Aliatimis <a href="mailto:g.aliatimis@lancaster.ac.uk">g.aliatimis@lancaster.ac.uk</a>
</p>


<h3>References</h3>

<p>Aliatimis, Georgios, Ruriko Yoshida, Burak Boyaci and James A. Grant (2023). Tropical Logistic Regression on Space of Phylogenetic Trees
</p>


<h3>Examples</h3>

<pre><code class='language-R'>omega = c(1/3,1,1,1,1,1/3)
tree = vector.to.equidistant.tree(omega)
plot(tree)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
