<!DOCTYPE html><html><head><title>Help for package TOmicsVis</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TOmicsVis}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#box_plot'><p>Box plot support two levels and multiple groups with P value.</p></a></li>
<li><a href='#chord_plot'><p>Chord plot for visualizing the relationships of pathways and genes.</p></a></li>
<li><a href='#circos_heatmap'><p>Circos heatmap plot for visualizing gene expressing in multiple samples.</p></a></li>
<li><a href='#corr_heatmap'><p>Correlation Heatmap for samples/groups based on Pearson algorithm.</p></a></li>
<li><a href='#degs_lists'><p>Paired comparisons differentially expressed genes (degs) among groups.</p></a></li>
<li><a href='#degs_stats'><p>All DEGs of paired comparison CT-vs-LT12 stats dataframe.</p></a></li>
<li><a href='#degs_stats2'><p>All DEGs of paired comparison CT-vs-LT12 stats2 dataframe.</p></a></li>
<li><a href='#dendro_plot'><p>Dendrograms for multiple samples/groups clustering.</p></a></li>
<li><a href='#flower_plot'><p>Flower plot for stat common and unique gene among multiple sets.</p></a></li>
<li><a href='#gene_cluster_trend'><p>Gene cluster trend plot for visualizing gene expression trend profile in multiple samples.</p></a></li>
<li><a href='#gene_expression'><p>All genes in all samples expression dataframe of RNA-Seq.</p></a></li>
<li><a href='#gene_expression2'><p>Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq.</p></a></li>
<li><a href='#gene_expression3'><p>Shared DEGs of all paired comparisons in all groups expression dataframe of RNA-Seq.</p></a></li>
<li><a href='#gene_go_kegg'><p>GO and KEGG annotation of background genes.</p></a></li>
<li><a href='#gene_go_kegg2'><p>GO and KEGG annotation of background genes.</p></a></li>
<li><a href='#gene_rank_plot'><p>Gene ranking dotplot for visualizing differentailly expressed genes.</p></a></li>
<li><a href='#go_enrich'><p>GO enrichment analysis based on GO annotation results (None/Exist Reference Genome).</p></a></li>
<li><a href='#go_enrich_bar'><p>GO enrichment analysis and bar plot (None/Exist Reference Genome).</p></a></li>
<li><a href='#go_enrich_dot'><p>GO enrichment analysis and dot plot (None/Exist Reference Genome).</p></a></li>
<li><a href='#go_enrich_net'><p>GO enrichment analysis and net plot (None/Exist Reference Genome).</p></a></li>
<li><a href='#go_enrich_stat'><p>GO enrichment analysis and stat plot (None/Exist Reference Genome).</p></a></li>
<li><a href='#heatmap_cluster'><p>Heatmap cluster for visualizing clustered gene expression data.</p></a></li>
<li><a href='#heatmap_group'><p>Heatmap group for visualizing grouped gene expression data.</p></a></li>
<li><a href='#kegg_enrich'><p>KEGG enrichment analysis based on KEGG annotation results (None/Exist Reference Genome).</p></a></li>
<li><a href='#kegg_enrich_bar'><p>KEGG enrichment analysis and bar plot (None/Exist Reference Genome).</p></a></li>
<li><a href='#kegg_enrich_dot'><p>KEGG enrichment analysis and dot plot (None/Exist Reference Genome).</p></a></li>
<li><a href='#kegg_enrich_net'><p>KEGG enrichment analysis and net plot (None/Exist Reference Genome).</p></a></li>
<li><a href='#ma_plot'><p>MversusA plot for visualizing differentially expressed genes.</p></a></li>
<li><a href='#network_data'><p>Network data from WGCNA tan module top-200 dataframe.</p></a></li>
<li><a href='#network_plot'><p>Network plot for analyzing and visualizing relationship of genes.</p></a></li>
<li><a href='#pca_analysis'><p>PCA dimensional reduction analysis for RNA-Seq.</p></a></li>
<li><a href='#pca_plot'><p>PCA dimensional reduction visualization for RNA-Seq.</p></a></li>
<li><a href='#quantile_plot'><p>Quantile plot for visualizing data distribution.</p></a></li>
<li><a href='#samples_groups'><p>Samples and groups for gene expression.</p></a></li>
<li><a href='#survival_data'><p>Survival data as example data for survival_plot function.</p></a></li>
<li><a href='#survival_plot'><p>Survival plot for analyzing and visualizing survival data.</p></a></li>
<li><a href='#table_cross'><p>Table cross used to cross search and merge results in two tables.</p></a></li>
<li><a href='#table_filter'><p>Table filter used to filter row by column condition.</p></a></li>
<li><a href='#table_merge'><p>Table merge used to merge multiple variables to on variable.</p></a></li>
<li><a href='#table_split'><p>Table split used for splitting a grouped column to multiple columns.</p></a></li>
<li><a href='#tomicsvis'><p>TOmicsVis shiny app start function.</p></a></li>
<li><a href='#traits_sex'><p>Length, Width, Weight, and Sex traits dataframe.</p></a></li>
<li><a href='#trend_plot'><p>Trend plot for visualizing gene expression trend profile in multiple traits.</p></a></li>
<li><a href='#tsne_analysis'><p>TSNE analysis for analyzing and visualizing TSNE algorithm.</p></a></li>
<li><a href='#tsne_plot'><p>TSNE plot for analyzing and visualizing TSNE algorithm.</p></a></li>
<li><a href='#umap_analysis'><p>UMAP analysis for analyzing RNA-Seq data.</p></a></li>
<li><a href='#umap_plot'><p>UMAP plot for analyzing and visualizing UMAP algorithm.</p></a></li>
<li><a href='#upsetr_plot'><p>UpSet plot for stat common and unique gene among multiple sets.</p></a></li>
<li><a href='#venn_plot'><p>Venn plot for stat common and unique gene among multiple sets.</p></a></li>
<li><a href='#violin_plot'><p>Violin plot support two levels and multiple groups with P value.</p></a></li>
<li><a href='#volcano_plot'><p>Volcano plot for visualizing differentailly expressed genes.</p></a></li>
<li><a href='#weight_sex'><p>Weight and Sex traits dataframe.</p></a></li>
<li><a href='#wgcna_pipeline'><p>WGCNA analysis pipeline for RNA-Seq.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Transcriptome Visualization Process Scheme</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Transcriptome visualization from sample trait statistics to gene expression analysis. Six categories include "Samples Statistics", "Traits Analysis", "Differential Expression Analysis", "Advanced Analysis", "GO and KEGG Enrichment", "Tables Operations", with complete sample data.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://benben-miao.github.io/TOmicsVis/">https://benben-miao.github.io/TOmicsVis/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/benben-miao/TOmicsVis/issues/">https://github.com/benben-miao/TOmicsVis/issues/</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), Biobase (&ge; 2.5.5), e1071 (&ge; 1.7-0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, grid, graphics, grDevices, ggplot2, ggsci,
reshape2, ggpubr, ggcorrplot, ggforce, vegan, venn, UpSetR,
circlize, randomcoloR, ComplexHeatmap, EnhancedVolcano, GGally,
survival, survminer, Rtsne, umap, dplyr, tidyr,
clusterProfiler, enrichplot, ggnewscale, igraph, WGCNA,
stringr, pheatmap, RColorBrewer, ggpolypath, factoextra,
ggrepel, cowplot, ggplotify, plotrix, Mfuzz, shiny</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-28 18:03:27 UTC; benben.miao</td>
</tr>
<tr>
<td>Author:</td>
<td>Benben Miao <a href="https://orcid.org/0000-0002-7307-1331"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Wei Dong <a href="https://orcid.org/0000-0003-2201-7164"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Benben Miao &lt;benben.miao@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-28 18:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='box_plot'>Box plot support two levels and multiple groups with P value.</h2><span id='topic+box_plot'></span>

<h3>Description</h3>

<p>Box plot support two levels and multiple groups with P value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>box_plot(
  data,
  test_method = "t.test",
  test_label = "p.format",
  notch = TRUE,
  group_level = "Three_Column",
  add_element = "jitter",
  my_shape = "fill_circle",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.5,
  sci_color_alpha = 1,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="box_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: Length, Width, Weight, and Sex traits dataframe (1st-col: Value, 2nd-col: Traits, 3rd-col: Sex).</p>
</td></tr>
<tr><td><code id="box_plot_+3A_test_method">test_method</code></td>
<td>
<p>Character: test methods of P value. Default: &quot;t.test&quot;, options: &quot;wilcox.test&quot;, &quot;t.test&quot;, &quot;anova&quot;, &quot;kruskal.test&quot;.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_test_label">test_label</code></td>
<td>
<p>Character: test label of P value. Default: &quot;p.format&quot;, options: &quot;p.signif&quot;, &quot;p.format&quot;. c(0, 0.0001, 0.001, 0.01, 0.05, 1).</p>
</td></tr>
<tr><td><code id="box_plot_+3A_notch">notch</code></td>
<td>
<p>Logical: Box notch or none. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_group_level">group_level</code></td>
<td>
<p>Character: group levels. Default: &quot;Three_Column&quot;, options: &quot;Two_Column&quot;, &quot;Three_Column&quot;.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_add_element">add_element</code></td>
<td>
<p>Character: add new plot. Default: &quot;jitter&quot;, options: &quot;none&quot;, &quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;, &quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;, &quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_my_shape">my_shape</code></td>
<td>
<p>Character: box scatter shape. Default: &quot;fill_circle&quot;, options: &quot;border_square&quot;, &quot;border_circle&quot;, &quot;border_triangle&quot;, &quot;plus&quot;, &quot;times&quot;, &quot;border_diamond&quot;, &quot;border_triangle_down&quot;, &quot;square_times&quot;, &quot;plus_times&quot;, &quot;diamond_plus&quot;, &quot;circle_plus&quot;, &quot;di_triangle&quot;, &quot;square_plus&quot;, &quot;circle_times&quot;,&quot;square_triangle&quot;, &quot;fill_square&quot;, &quot;fill_circle&quot;, &quot;fill_triangle&quot;, &quot;fill_diamond&quot;, &quot;large_circle&quot;, &quot;small_circle&quot;, &quot;fill_border_circle&quot;, &quot;fill_border_square&quot;, &quot;fill_border_diamond&quot;, &quot;fill_border_triangle&quot;.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci color pallet. Default: &quot;Sci_AAAS&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_sci_fill_alpha">sci_fill_alpha</code></td>
<td>
<p>Numeric: ggsci fill color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_sci_color_alpha">sci_color_alpha</code></td>
<td>
<p>Numeric: ggsci border color alpha. Default: 1.00, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character: legend position. Default: &quot;right&quot;, options: &quot;none&quot;, &quot;left&quot;, &quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_legend_dir">legend_dir</code></td>
<td>
<p>Character: legend direction. Default: &quot;vertical&quot;, options: &quot;horizontal&quot;, &quot;vertical&quot;.</p>
</td></tr>
<tr><td><code id="box_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: box plot support two levels and multiple groups with P value.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(traits_sex)
head(traits_sex)

# 3. Default parameters
box_plot(traits_sex)

# 4. Set test_label = "p.signif",
box_plot(traits_sex, test_label = "p.signif")

# 5. Set notch = FALSE
box_plot(traits_sex, notch = FALSE)

# 6. Set group_level = "Two_Column"
box_plot(traits_sex, group_level = "Two_Column")

# 7. Set add_element = "point"
box_plot(traits_sex, add_element = "point")

</code></pre>

<hr>
<h2 id='chord_plot'>Chord plot for visualizing the relationships of pathways and genes.</h2><span id='topic+chord_plot'></span>

<h3>Description</h3>

<p>Chord plot is used to visualize complex relationships between samples and genes, as well as between pathways and genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chord_plot(
  data,
  multi_colors = "VividColors",
  color_seed = 10,
  color_alpha = 0.3,
  link_visible = TRUE,
  link_dir = -1,
  link_type = "diffHeight",
  sector_scale = "Origin",
  width_circle = 3,
  dist_name = 3,
  label_dir = "Vertical",
  dist_label = 0.3,
  label_scale = 0.8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chord_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_multi_colors">multi_colors</code></td>
<td>
<p>Character: color palette. Default: &quot;VividColors&quot;, options: &quot;VividColors&quot;, &quot;RainbowColors&quot;.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_color_seed">color_seed</code></td>
<td>
<p>Numeric: rand seed for VividColors. Default: 10.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_color_alpha">color_alpha</code></td>
<td>
<p>Numeric: color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_link_visible">link_visible</code></td>
<td>
<p>Logical: links visible. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_link_dir">link_dir</code></td>
<td>
<p>Numeric: links direction, use with link_type. Default: -1, options: -1, 0, 1, 2.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_link_type">link_type</code></td>
<td>
<p>Character: links type, use with link_dir. Default: &quot;diffHeight&quot;, options: &quot;diffHeight&quot;, &quot;arrows&quot;.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_sector_scale">sector_scale</code></td>
<td>
<p>Character: sector scale method. Default: &quot;Origin&quot;, options: &quot;Origin&quot;, &quot;Scale&quot;.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_width_circle">width_circle</code></td>
<td>
<p>Numeric: outside circle width. Default: 3.0, min: 0.0, max: 10.0.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_dist_name">dist_name</code></td>
<td>
<p>Numeric: the distance of name and circle. Default: 3.0, min: 0.0, max: 10.0.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_label_dir">label_dir</code></td>
<td>
<p>Character: label director. Default: &quot;Vertical&quot;, options: &quot;Horizontal&quot;, &quot;Vertical&quot;.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_dist_label">dist_label</code></td>
<td>
<p>Numeric: the distance of label and circle. Default: 0.3, min: 0.0.</p>
</td></tr>
<tr><td><code id="chord_plot_+3A_label_scale">label_scale</code></td>
<td>
<p>Numeric: labels font size sclae. Default: 0.8, min: 0, max: NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: chord plot is used to visualize complex relationships between samples and genes, as well as between pathways and genes.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression2)
head(gene_expression2)

# 3. Default parameters
chord_plot(gene_expression2[1:20,])

</code></pre>

<hr>
<h2 id='circos_heatmap'>Circos heatmap plot for visualizing gene expressing in multiple samples.</h2><span id='topic+circos_heatmap'></span>

<h3>Description</h3>

<p>Circos heatmap plot for visualizing gene expressing in multiple samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos_heatmap(
  data,
  low_color = "#0000ff",
  mid_color = "#ffffff",
  high_color = "#ff0000",
  gap_size = 25,
  cluster_run = TRUE,
  cluster_method = "complete",
  distance_method = "euclidean",
  dend_show = "inside",
  dend_height = 0.2,
  track_height = 0.3,
  rowname_show = "outside",
  rowname_size = 0.8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="circos_heatmap_+3A_data">data</code></td>
<td>
<p>Dataframe: Shared degs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_low_color">low_color</code></td>
<td>
<p>Character: min value color (color name or hex value). Default: &quot;#0000ff&quot;.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_mid_color">mid_color</code></td>
<td>
<p>Character: middle value color (color name or hex value). Default: &quot;#ffffff&quot;.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_high_color">high_color</code></td>
<td>
<p>Character: high value color (color name or hex value). Default: &quot;#ff0000&quot;.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_gap_size">gap_size</code></td>
<td>
<p>Numeric: heatmap gap size. Default: 25, min: 0.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_cluster_run">cluster_run</code></td>
<td>
<p>Logical: running cluster algorithm. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_cluster_method">cluster_method</code></td>
<td>
<p>Character: cluster methods. Default: &quot;complete&quot;, options: &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot;, &quot;mcquitty&quot;, &quot;median&quot;, &quot;centroid&quot;.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_distance_method">distance_method</code></td>
<td>
<p>Character: distance methods. Default: &quot;euclidean&quot;, options: &quot;euclidean&quot;, &quot;maximum&quot;, &quot;manhattan&quot;, &quot;canberra&quot;, &quot;binary&quot;, &quot;minkowski&quot;.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_dend_show">dend_show</code></td>
<td>
<p>Character: control dendgram display and position. Default: &quot;inside&quot;, options: &quot;none&quot;, &quot;outside&quot;, &quot;inside&quot;.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_dend_height">dend_height</code></td>
<td>
<p>Numeric: dendgram height. Default: 0.20, min: 0.00, max: 0.50.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_track_height">track_height</code></td>
<td>
<p>Numeric: heatmap track height. Default: 0.30, min: 0.00, max: 0.50.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_rowname_show">rowname_show</code></td>
<td>
<p>Character: control rownames display and position. Hind first rowname by running rownames(data). Default: &quot;outside&quot;, options: &quot;none&quot;, &quot;outside&quot;, &quot;inside&quot;.</p>
</td></tr>
<tr><td><code id="circos_heatmap_+3A_rowname_size">rowname_size</code></td>
<td>
<p>Numeric: rowname font size. Default: 0.80, min: 0.10, max: 10.00.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: circos heatmap plot for visualizing gene expressing in multiple samples.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression2)
head(gene_expression2)

# 3. Default parameters
circos_heatmap(gene_expression2[1:50,])

</code></pre>

<hr>
<h2 id='corr_heatmap'>Correlation Heatmap for samples/groups based on Pearson algorithm.</h2><span id='topic+corr_heatmap'></span>

<h3>Description</h3>

<p>Correlation Heatmap for samples/groups based on Pearson algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corr_heatmap(
  data,
  corr_method = "pearson",
  cell_shape = "square",
  fill_type = "full",
  lable_size = 3,
  axis_angle = 45,
  axis_size = 12,
  lable_digits = 3,
  color_low = "blue",
  color_mid = "white",
  color_high = "red",
  outline_color = "white",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corr_heatmap_+3A_data">data</code></td>
<td>
<p>Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_corr_method">corr_method</code></td>
<td>
<p>Character: correlation method. Default: &quot;pearson&quot;, options: &quot;pearson&quot;, &quot;spearman&quot;, &quot;kendall&quot;.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_cell_shape">cell_shape</code></td>
<td>
<p>Character: heatmap cell shape. Default: &quot;square&quot;, options: &quot;circle&quot;, &quot;square&quot;.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_fill_type">fill_type</code></td>
<td>
<p>Character: heatmap fill type. Default: &quot;full&quot;, options: &quot;upper&quot;, &quot;low&quot;, &quot;full&quot;.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_lable_size">lable_size</code></td>
<td>
<p>Numeric: heatmap label size. Default: 3, min: 0.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_axis_angle">axis_angle</code></td>
<td>
<p>Numeric: axis rotate angle. Default: 45, min: 0, max: 360.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_axis_size">axis_size</code></td>
<td>
<p>Numberic: axis font size. Default: 12, min: 0.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_lable_digits">lable_digits</code></td>
<td>
<p>Numeric: heatmap label digits. Default: 3, min: 0, max: 3.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_color_low">color_low</code></td>
<td>
<p>Character: low value color name or hex value. Default: &quot;blue&quot;.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_color_mid">color_mid</code></td>
<td>
<p>Character: middle value color name or hex value. Default: &quot;white&quot;.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_color_high">color_high</code></td>
<td>
<p>Character: high value color name or hex value. Default: &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_outline_color">outline_color</code></td>
<td>
<p>Character: outline color name or hex value. Default: &quot;white&quot;.</p>
</td></tr>
<tr><td><code id="corr_heatmap_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 theme. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: heatmap plot filled with Pearson correlation values and P values.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset gene_exp
data(gene_expression)
head(gene_expression)

# 3. Default parameters
corr_heatmap(gene_expression)

# 4. Set color_low = "#008800"
corr_heatmap(gene_expression, color_low = "#008800")

# 5. Set cell_shape = "circle"
corr_heatmap(gene_expression, cell_shape = "circle")

</code></pre>

<hr>
<h2 id='degs_lists'>Paired comparisons differentially expressed genes (degs) among groups.</h2><span id='topic+degs_lists'></span>

<h3>Description</h3>

<p>Paired comparisons differentially expressed genes (degs) among groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(degs_lists)
</code></pre>


<h3>Format</h3>

<p>Dataframe: Paired comparisons differentially expressed genes (degs) among groups (1st-col~: degs of paired comparisons).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/VennPlot/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(degs_lists)

# 3. View example data
degs_lists

</code></pre>

<hr>
<h2 id='degs_stats'>All DEGs of paired comparison CT-vs-LT12 stats dataframe.</h2><span id='topic+degs_stats'></span>

<h3>Description</h3>

<p>All DEGs of paired comparison CT-vs-LT12 stats dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(degs_stats)
</code></pre>


<h3>Format</h3>

<p>Dataframe: All DEGs of paired comparison CT-vs-LT12 stats dataframe (1st-col: Genes, 2nd-col: log2FoldChange, 3rd-col: Pvalue, 4th-col: FDR).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/VolcanoPlot/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(degs_stats)

# 3. View example data
degs_stats

</code></pre>

<hr>
<h2 id='degs_stats2'>All DEGs of paired comparison CT-vs-LT12 stats2 dataframe.</h2><span id='topic+degs_stats2'></span>

<h3>Description</h3>

<p>All DEGs of paired comparison CT-vs-LT12 stats2 dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(degs_stats2)
</code></pre>


<h3>Format</h3>

<p>Dataframe: All DEGs of paired comparison CT-vs-LT12 stats2 dataframe (1st-col: Gene, 2nd-col: baseMean, 3rd-col: Log2FoldChange, 4th-col: FDR).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/MversusA/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(degs_stats2)

# 3. View example data
degs_stats2

</code></pre>

<hr>
<h2 id='dendro_plot'>Dendrograms for multiple samples/groups clustering.</h2><span id='topic+dendro_plot'></span>

<h3>Description</h3>

<p>Dendrograms for multiple samples/groups clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dendro_plot(
  data,
  dist_method = "euclidean",
  hc_method = "ward.D2",
  tree_type = "rectangle",
  k_num = 5,
  palette = "npg",
  color_labels_by_k = TRUE,
  horiz = FALSE,
  label_size = 1,
  line_width = 1,
  rect = TRUE,
  rect_fill = TRUE,
  xlab = "Samples",
  ylab = "Height",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dendro_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_dist_method">dist_method</code></td>
<td>
<p>Character: distance measure method. Default: &quot;euclidean&quot;, options: &quot;euclidean&quot;, &quot;maximum&quot;, &quot;manhattan&quot;, &quot;canberra&quot;, &quot;binary&quot; or &quot;minkowski&quot;.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_hc_method">hc_method</code></td>
<td>
<p>Character: hierarchical clustering method. Default: &quot;ward.D2&quot;, options: &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;,&quot;average&quot; (= UPGMA), &quot;mcquitty&quot; (= WPGMA), &quot;median&quot; (= WPGMC) or &quot;centroid&quot; (= UPGMC).</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_tree_type">tree_type</code></td>
<td>
<p>Character: plot tree type. Default: &quot;rectangle&quot;, options: &quot;rectangle&quot;, &quot;circular&quot;, &quot;phylogenic&quot;.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_k_num">k_num</code></td>
<td>
<p>Numeric: the number of groups for cutting the tree. Default: 3.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_palette">palette</code></td>
<td>
<p>Character: color palette used for the group. Default: &quot;npg&quot;, options: &quot;npg&quot;, &quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_color_labels_by_k">color_labels_by_k</code></td>
<td>
<p>Logical: labels colored by group. Default: TRUE, options: TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_horiz">horiz</code></td>
<td>
<p>Logical: horizontal dendrogram. Default: FALSE, options: TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric: tree label size. Default: 0.8, min: 0.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_line_width">line_width</code></td>
<td>
<p>Numeric: branches and rectangle line width. Default: 0.7, min: 0.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_rect">rect</code></td>
<td>
<p>Logical: add a rectangle around groups. Default: TRUE, options: TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_rect_fill">rect_fill</code></td>
<td>
<p>Logical: fill the rectangle. Default: TRUE, options: TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_xlab">xlab</code></td>
<td>
<p>Character: title of the xlab. Default: &quot;&quot;.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_ylab">ylab</code></td>
<td>
<p>Character: title of the ylab. Default: &quot;Height&quot;.</p>
</td></tr>
<tr><td><code id="dendro_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 theme. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: dendrogram for multiple samples clustering.
</p>


<h3>Author(s)</h3>

<p>wei dong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset gene_expression
data(gene_expression)
head(gene_expression)

# 3. Default parameters
dendro_plot(gene_expression)

# 4. Set palette = "aaas"
dendro_plot(gene_expression, palette = "aaas")

# 5. Set tree_type = "circular"
dendro_plot(gene_expression, tree_type = "circular")

</code></pre>

<hr>
<h2 id='flower_plot'>Flower plot for stat common and unique gene among multiple sets.</h2><span id='topic+flower_plot'></span>

<h3>Description</h3>

<p>Flower plot for stat common and unique gene among multiple sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flower_plot(
  flower_dat,
  angle = 90,
  a = 1,
  b = 2,
  r = 1,
  ellipse_col_pal = "Spectral",
  circle_col = "white",
  label_text_cex = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flower_plot_+3A_flower_dat">flower_dat</code></td>
<td>
<p>Dataframe: Paired comparisons differentially expressed genes (degs) among groups (1st-col~: degs of paired comparisons).</p>
</td></tr>
<tr><td><code id="flower_plot_+3A_angle">angle</code></td>
<td>
<p>Number: set the angle of rotation in degress. Default: 90.</p>
</td></tr>
<tr><td><code id="flower_plot_+3A_a">a</code></td>
<td>
<p>Number: set the radii of the ellipses along the x-axes. Default: 0.5.</p>
</td></tr>
<tr><td><code id="flower_plot_+3A_b">b</code></td>
<td>
<p>Number: set the radii of the ellipses along the y-axes. Default: 2.</p>
</td></tr>
<tr><td><code id="flower_plot_+3A_r">r</code></td>
<td>
<p>Number: set the radius of the circle. Default: 1.</p>
</td></tr>
<tr><td><code id="flower_plot_+3A_ellipse_col_pal">ellipse_col_pal</code></td>
<td>
<p>Character: set the color palette for filling the ellipse. Default: &quot;Spectral&quot;, options: 'Spectral', 'Set1', 'Set2', 'Set3', 'Accent', 'Dark2', 'Paired', 'Pastel1', 'Pastel2'.</p>
</td></tr>
<tr><td><code id="flower_plot_+3A_circle_col">circle_col</code></td>
<td>
<p>Character: set the color for filling the circle. Default: &quot;white&quot;.</p>
</td></tr>
<tr><td><code id="flower_plot_+3A_label_text_cex">label_text_cex</code></td>
<td>
<p>Number: set the label text cex. Default: 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: Flower plot for stat common and unique gene among multiple sets.
</p>


<h3>Author(s)</h3>

<p>wei dong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(degs_lists)
head(degs_lists)

# 3. Default parameters
flower_plot(degs_lists)

# 4. Set angle = 60
flower_plot(degs_lists, angle = 60)

# 5. Set ellipse_col_pal = "Accent"
flower_plot(degs_lists, ellipse_col_pal = "Accent")

# 6. Set a = 1, b = 2, r = 1
flower_plot(degs_lists, a = 1, b = 2, r = 1, ellipse_col_pal = "Set2")

</code></pre>

<hr>
<h2 id='gene_cluster_trend'>Gene cluster trend plot for visualizing gene expression trend profile in multiple samples.</h2><span id='topic+gene_cluster_trend'></span>

<h3>Description</h3>

<p>Gene cluster trend plot for visualizing gene expression trend profile in multiple samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_cluster_trend(
  data,
  thres = 0.25,
  min_std = 0.2,
  palette = "PiYG",
  cluster_num = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_cluster_trend_+3A_data">data</code></td>
<td>
<p>Dataframe: Shared DEGs of all paired comparisons in all groups expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~n-1-col: Groups, n-col: Pathways).</p>
</td></tr>
<tr><td><code id="gene_cluster_trend_+3A_thres">thres</code></td>
<td>
<p>Number: set the threshold for excluding genes. Default: 0.25.</p>
</td></tr>
<tr><td><code id="gene_cluster_trend_+3A_min_std">min_std</code></td>
<td>
<p>Number: set the threshold for minimum standard deviation. Default: 0.2.</p>
</td></tr>
<tr><td><code id="gene_cluster_trend_+3A_palette">palette</code></td>
<td>
<p>Character: set the color palette to be used for plotting. Default: &quot;PiYG&quot;, options: 'Spectral', 'BrBG', 'PiYG', 'PRGn', 'PuOr', 'RdBu', 'RdGy', 'RdYlBu', 'RdYlGn'.</p>
</td></tr>
<tr><td><code id="gene_cluster_trend_+3A_cluster_num">cluster_num</code></td>
<td>
<p>Number: set the number of clusters. Default: 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: Gene cluster trend plot for visualizing gene expression trend profile in multiple samples.
</p>


<h3>Author(s)</h3>

<p>wei dong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset gene_cluster_data
data(gene_expression3)
head(gene_expression3)

# 3. Default parameters
gene_cluster_trend(gene_expression3[,-7])

# 4. Set palette = "RdBu"
gene_cluster_trend(gene_expression3[,-7], palette = "RdBu")

# 5. Set cluster_num = 6
gene_cluster_trend(gene_expression3[,-7], cluster_num = 6, palette = "Spectral")

</code></pre>

<hr>
<h2 id='gene_expression'>All genes in all samples expression dataframe of RNA-Seq.</h2><span id='topic+gene_expression'></span>

<h3>Description</h3>

<p>All genes in all samples expression dataframe of RNA-Seq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_expression)
</code></pre>


<h3>Format</h3>

<p>Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/CorPlot/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example dataset gene_expression
data(gene_expression)

# 3. View gene_expression
gene_expression

</code></pre>

<hr>
<h2 id='gene_expression2'>Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq.</h2><span id='topic+gene_expression2'></span>

<h3>Description</h3>

<p>Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_expression2)
</code></pre>


<h3>Format</h3>

<p>Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/CorPlot/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example dataset
data(gene_expression2)

# 3. View gene_expression2
gene_expression2

</code></pre>

<hr>
<h2 id='gene_expression3'>Shared DEGs of all paired comparisons in all groups expression dataframe of RNA-Seq.</h2><span id='topic+gene_expression3'></span>

<h3>Description</h3>

<p>Shared DEGs of all paired comparisons in all groups expression dataframe of RNA-Seq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_expression3)
</code></pre>


<h3>Format</h3>

<p>Dataframe: Shared DEGs of all paired comparisons in all groups expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~n-1-col: Groups, n-col: Pathways).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/CorPlot/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example dataset
data(gene_expression3)

# 3. View gene_expression3
gene_expression3

</code></pre>

<hr>
<h2 id='gene_go_kegg'>GO and KEGG annotation of background genes.</h2><span id='topic+gene_go_kegg'></span>

<h3>Description</h3>

<p>GO and KEGG annotation of background genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_go_kegg)
</code></pre>


<h3>Format</h3>

<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/GOenrichStat/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(gene_go_kegg)

# 3. View example data
gene_go_kegg

</code></pre>

<hr>
<h2 id='gene_go_kegg2'>GO and KEGG annotation of background genes.</h2><span id='topic+gene_go_kegg2'></span>

<h3>Description</h3>

<p>GO and KEGG annotation of background genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_go_kegg2)
</code></pre>


<h3>Format</h3>

<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/GOenrichStat/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(gene_go_kegg2)

# 3. View example data
gene_go_kegg2

</code></pre>

<hr>
<h2 id='gene_rank_plot'>Gene ranking dotplot for visualizing differentailly expressed genes.</h2><span id='topic+gene_rank_plot'></span>

<h3>Description</h3>

<p>Gene ranking dotplot for visualizing differentailly expressed genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_rank_plot(
  data,
  log2fc = 1,
  palette = "Spectral",
  top_n = 10,
  genes_to_label = NULL,
  label_size = 5,
  base_size = 12,
  title = "Gene ranking dotplot",
  xlab = "Ranking of differentially expressed genes",
  ylab = "Log2FoldChange"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_rank_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: All DEGs of paired comparison CT-vs-LT12 stats dataframe (1st-col: Genes, 2nd-col: log2FoldChange, 3rd-col: Pvalue, 4th-col: FDR).</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_log2fc">log2fc</code></td>
<td>
<p>Numeric: log2(FoldChange) cutoff log2(2) = 1. Default: 1.0, min: 0.0, max: null.</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_palette">palette</code></td>
<td>
<p>Character: color palette used for the point. Default: &quot;spectral&quot;, options: 'Spectral', 'BrBG', 'PiYG', 'PRGn', 'PuOr', 'RdBu', 'RdGy', 'RdYlBu', 'RdYlGn'.</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_top_n">top_n</code></td>
<td>
<p>Numeric: number of top differentailly expressed genes. Default: 10, min: 0.</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_genes_to_label">genes_to_label</code></td>
<td>
<p>Character: a vector of selected genes. Default: &quot;NULL&quot;.</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric: gene label size. Default: 5, min: 0.</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_base_size">base_size</code></td>
<td>
<p>Numeric: base font size. Default: 12, min: 0.</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_title">title</code></td>
<td>
<p>Character: main plot title. Default: &quot;Gene ranking dotplot&quot;.</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_xlab">xlab</code></td>
<td>
<p>Character: title of the xlab. Default: &quot;Ranking of differentially expressed genes&quot;.</p>
</td></tr>
<tr><td><code id="gene_rank_plot_+3A_ylab">ylab</code></td>
<td>
<p>Character: title of the ylab. Default: &quot;Log2FoldChange&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: Gene ranking dotplot for visualizing differentailly expressed genes.
</p>


<h3>Author(s)</h3>

<p>wei dong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(degs_stats)
head(degs_stats)

# 3. Default parameters
gene_rank_plot(degs_stats)

# 4. Set top_n = 5
gene_rank_plot(degs_stats, top_n = 5, palette = "PiYG")

# 5. Set genes_to_label = c("SELL","CCR7","KLRG1","IL7R")
gene_rank_plot(degs_stats, genes_to_label = c("SELL","CCR7","KLRG1","IL7R"), palette = "PuOr")

</code></pre>

<hr>
<h2 id='go_enrich'>GO enrichment analysis based on GO annotation results (None/Exist Reference Genome).</h2><span id='topic+go_enrich'></span>

<h3>Description</h3>

<p>GO enrichment analysis based on GO annotation results (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>go_enrich(
  go_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="go_enrich_+3A_go_anno">go_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="go_enrich_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="go_enrich_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: include columns (&quot;ID&quot;, &quot;ontology&quot;, &quot;Description&quot;, &quot;GeneRatio&quot;, &quot;BgRatio&quot;, &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;, &quot;geneID&quot;, &quot;Count&quot;).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
res &lt;- go_enrich(gene_go_kegg[,-5], gene_go_kegg[100:200,1])
head(res)

# 4. Set padjust_method = "BH"
res &lt;- go_enrich(gene_go_kegg[,-5], gene_go_kegg[100:200,1], padjust_method = "BH")
head(res)

# 5. Set pvalue_cutoff = 0.10
res &lt;- go_enrich(gene_go_kegg[,-5], gene_go_kegg[100:200,1], pvalue_cutoff = 0.10)
head(res)

</code></pre>

<hr>
<h2 id='go_enrich_bar'>GO enrichment analysis and bar plot (None/Exist Reference Genome).</h2><span id='topic+go_enrich_bar'></span>

<h3>Description</h3>

<p>GO enrichment analysis and bar plot (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>go_enrich_bar(
  go_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05,
  sign_by = "p.adjust",
  category_num = 30,
  font_size = 12,
  low_color = "#ff0000aa",
  high_color = "#008800aa",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="go_enrich_bar_+3A_go_anno">go_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_sign_by">sign_by</code></td>
<td>
<p>Character: significant by. Default: &quot;p.adjust&quot;, options: &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_category_num">category_num</code></td>
<td>
<p>Numeric: categories number to display. Default: 30, min: 1, max: NULL.</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_font_size">font_size</code></td>
<td>
<p>Numeric: category font size. Default: 12.</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_low_color">low_color</code></td>
<td>
<p>Character: low value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_high_color">high_color</code></td>
<td>
<p>Character: high value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="go_enrich_bar_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: GO enrichment analysis and bar plot (None/Exist Reference Genome).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
go_enrich_bar(gene_go_kegg[,-5], gene_go_kegg[100:200,1])

# 4. Set padjust_method = "BH"
go_enrich_bar(gene_go_kegg[,-5], gene_go_kegg[100:200,1], padjust_method = "BH")

# 5. Set category_num = 10
go_enrich_bar(gene_go_kegg[,-5], gene_go_kegg[100:200,1], category_num = 10)

# 6. Set ggTheme = "theme_bw"
go_enrich_bar(gene_go_kegg[,-5], gene_go_kegg[100:200,1], ggTheme = "theme_bw")

</code></pre>

<hr>
<h2 id='go_enrich_dot'>GO enrichment analysis and dot plot (None/Exist Reference Genome).</h2><span id='topic+go_enrich_dot'></span>

<h3>Description</h3>

<p>GO enrichment analysis and dot plot (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>go_enrich_dot(
  go_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05,
  sign_by = "p.adjust",
  category_num = 30,
  font_size = 12,
  low_color = "#ff0000aa",
  high_color = "#008800aa",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="go_enrich_dot_+3A_go_anno">go_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_sign_by">sign_by</code></td>
<td>
<p>Character: significant by. Default: &quot;p.adjust&quot;, options: &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_category_num">category_num</code></td>
<td>
<p>Numeric: categories number to display. Default: 30, min: 1, max: NULL.</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_font_size">font_size</code></td>
<td>
<p>Numeric: category font size. Default: 12.</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_low_color">low_color</code></td>
<td>
<p>Character: low value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_high_color">high_color</code></td>
<td>
<p>Character: high value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="go_enrich_dot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: GO enrichment analysis and dot plot (None/Exist Reference Genome).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
go_enrich_dot(gene_go_kegg[,-5], gene_go_kegg[100:200,1])

# 4. Set padjust_method = "BH"
go_enrich_dot(gene_go_kegg[,-5], gene_go_kegg[100:200,1], padjust_method = "BH")

# 5. Set category_num = 10
go_enrich_dot(gene_go_kegg[,-5], gene_go_kegg[100:200,1], category_num = 10)

# 6. Set ggTheme = "theme_bw"
go_enrich_dot(gene_go_kegg[,-5], gene_go_kegg[100:200,1], ggTheme = "theme_bw")

</code></pre>

<hr>
<h2 id='go_enrich_net'>GO enrichment analysis and net plot (None/Exist Reference Genome).</h2><span id='topic+go_enrich_net'></span>

<h3>Description</h3>

<p>GO enrichment analysis and net plot (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>go_enrich_net(
  go_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05,
  category_num = 20,
  net_layout = "circle",
  net_circular = TRUE,
  low_color = "#ff0000aa",
  high_color = "#008800aa"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="go_enrich_net_+3A_go_anno">go_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_category_num">category_num</code></td>
<td>
<p>Numeric: categories number to display. Default: 20, min: 1, max: NULL.</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_net_layout">net_layout</code></td>
<td>
<p>Character: network layout. Default: &quot;circle&quot;, options: 'star', 'circle', 'gem', 'dh', 'graphopt', 'grid', 'mds', 'randomly', 'fr', 'kk', 'drl' or 'lgl'.</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_net_circular">net_circular</code></td>
<td>
<p>Logical: network circular. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_low_color">low_color</code></td>
<td>
<p>Character: low value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="go_enrich_net_+3A_high_color">high_color</code></td>
<td>
<p>Character: high value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>PLot: GO enrichment analysis and net plot (None/Exist Reference Genome).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
go_enrich_net(gene_go_kegg[,-5], gene_go_kegg[100:200,1])

</code></pre>

<hr>
<h2 id='go_enrich_stat'>GO enrichment analysis and stat plot (None/Exist Reference Genome).</h2><span id='topic+go_enrich_stat'></span>

<h3>Description</h3>

<p>GO enrichment analysis and stat plot (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>go_enrich_stat(
  go_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05,
  max_go_item = 15,
  strip_fill = "#CDCDCD",
  xtext_angle = 45,
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.8,
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="go_enrich_stat_+3A_go_anno">go_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_max_go_item">max_go_item</code></td>
<td>
<p>Numeric: max BP/CC/MF terms. Default: 15, min: 1, max: NULL.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_strip_fill">strip_fill</code></td>
<td>
<p>Character: strip fill color (color name or hex value). Default: &quot;#CDCDCD&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_xtext_angle">xtext_angle</code></td>
<td>
<p>Numeric: x axis texts angle. Default: 45, min: 0, max: 360.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci color pallet. Default: &quot;Sci_AAAS&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_sci_fill_alpha">sci_fill_alpha</code></td>
<td>
<p>Numeric: ggsci fill color alpha. Default: 0.80, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="go_enrich_stat_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: GO enrichment analysis and stat plot (None/Exist Reference Genome).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
go_enrich_stat(gene_go_kegg[,-5], gene_go_kegg[100:200,1])

# 4. Set padjust_method = "BH"
go_enrich_stat(gene_go_kegg[,-5], gene_go_kegg[100:200,1], padjust_method = "BH")

# 5. Set max_go_item = 10
go_enrich_stat(gene_go_kegg[,-5], gene_go_kegg[100:200,1], max_go_item = 10)

# 6. Set strip_fill = "#008888"
go_enrich_stat(gene_go_kegg[,-5], gene_go_kegg[100:200,1], strip_fill = "#008888")

# 7. Set sci_fill_color = "Sci_JAMA"
go_enrich_stat(gene_go_kegg[,-5], gene_go_kegg[100:200,1], sci_fill_color = "Sci_JAMA")

</code></pre>

<hr>
<h2 id='heatmap_cluster'>Heatmap cluster for visualizing clustered gene expression data.</h2><span id='topic+heatmap_cluster'></span>

<h3>Description</h3>

<p>Heatmap cluster for visualizing clustered gene expression data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_cluster(
  data,
  dist_method = "euclidean",
  hc_method = "average",
  k_num = 5,
  show_rownames = FALSE,
  palette = "RdBu",
  cluster_pal = "Set1",
  border_color = "#ffffff",
  angle_col = 45,
  label_size = 10,
  base_size = 12,
  line_color = "#0000cd",
  line_alpha = 0.2,
  summary_color = "#0000cd",
  summary_alpha = 0.8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_cluster_+3A_data">data</code></td>
<td>
<p>Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_dist_method">dist_method</code></td>
<td>
<p>Character: distance measure method. Default: &quot;euclidean&quot;, options: &quot;euclidean&quot;, &quot;maximum&quot;, &quot;manhattan&quot;, &quot;canberra&quot;, &quot;binary&quot; or &quot;minkowski&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_hc_method">hc_method</code></td>
<td>
<p>Character: hierarchical clustering method. Default: &quot;average&quot;, options: &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;,&quot;average&quot; (= UPGMA), &quot;mcquitty&quot; (= WPGMA), &quot;median&quot; (= WPGMC) or &quot;centroid&quot; (= UPGMC).</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_k_num">k_num</code></td>
<td>
<p>Numeric: the number of groups for cutting the tree. Default: 5.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_show_rownames">show_rownames</code></td>
<td>
<p>Logical: boolean specifying if column names are be shown. Default: FALSE, options: TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_palette">palette</code></td>
<td>
<p>Character: color palette used in heatmap. Default: &quot;RdBu&quot;, options: 'Spectral', 'BrBG', 'PiYG', 'PRGn', 'PuOr', 'RdBu', 'RdGy', 'RdYlBu', 'RdYlGn'.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_cluster_pal">cluster_pal</code></td>
<td>
<p>Character: color palette used for the cluster. Default: &quot;Set1&quot;, options: 'Set1', 'Set2', 'Set3', 'Accent', 'Dark2', 'Paired', 'Pastel1', 'Pastel2'.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_border_color">border_color</code></td>
<td>
<p>Character: cell border color (color name or hex value). Default: &quot;#ffffff&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_angle_col">angle_col</code></td>
<td>
<p>Numeric: angle of the column labels. Default: 45.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_label_size">label_size</code></td>
<td>
<p>Numeric: fontsize for the plot. Default: 10, min: 0.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_base_size">base_size</code></td>
<td>
<p>Numeric: base font size. Default: 12, min: 0.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_line_color">line_color</code></td>
<td>
<p>Character: trend lines color. Default: &quot;#0000cd&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_line_alpha">line_alpha</code></td>
<td>
<p>Numeric: trend lines alpha. Default: 0.20, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_summary_color">summary_color</code></td>
<td>
<p>Charater: summary line color. Default: &quot;#0000cd&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_cluster_+3A_summary_alpha">summary_alpha</code></td>
<td>
<p>Numeric: summary line alpha. Default: 0.80, min: 0.00, max: 1.00.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: Heatmap cluster for visualizing clustered gene expression data.
</p>


<h3>Author(s)</h3>

<p>wei dong
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression2)
head(gene_expression2)

# 3. Default parameters
heatmap_cluster(gene_expression2)

# 4. Set palette = "PuOr"
heatmap_cluster(gene_expression2, palette = "PuOr")

# 5. Set line_color = "#ff0000", summary_color = "#ff0000"
heatmap_cluster(gene_expression2, line_color = "#ff0000", summary_color = "#ff0000")

</code></pre>

<hr>
<h2 id='heatmap_group'>Heatmap group for visualizing grouped gene expression data.</h2><span id='topic+heatmap_group'></span>

<h3>Description</h3>

<p>Heatmap group for visualizing grouped gene expression data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_group(
  sample_gene,
  group_sample,
  scale_data = "row",
  clust_method = "complete",
  border_show = TRUE,
  border_color = "#ffffff",
  value_show = TRUE,
  value_decimal = 2,
  value_size = 5,
  axis_size = 8,
  cell_height = 10,
  low_color = "#00880055",
  mid_color = "#ffffff",
  high_color = "#ff000055",
  na_color = "#ff8800",
  x_angle = 45
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_group_+3A_sample_gene">sample_gene</code></td>
<td>
<p>Dataframe: Shared degs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_group_sample">group_sample</code></td>
<td>
<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_scale_data">scale_data</code></td>
<td>
<p>Character: scale data. Default: &quot;row&quot;, options: &quot;row&quot;, &quot;column&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_clust_method">clust_method</code></td>
<td>
<p>Character: cluster method. Default: &quot;complete&quot;, options: &quot;ward.D&quot;, &quot;ward.D2&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot; (= UPGMA), &quot;mcquitty&quot; (= WPGMA), &quot;median&quot; (= WPGMC) or &quot;centroid&quot; (= UPGMC).</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_border_show">border_show</code></td>
<td>
<p>Logical: show border. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_border_color">border_color</code></td>
<td>
<p>Character: cell border color (color value or hex value with alpha). Default: &quot;#ffffff&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_value_show">value_show</code></td>
<td>
<p>Logical: show value in cell. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_value_decimal">value_decimal</code></td>
<td>
<p>Numeric: cell value decimal. Default: 2, min: 0, max: 5.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_value_size">value_size</code></td>
<td>
<p>Numeric: cell value font size. Default: 5, min: 0, max: NULL.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_axis_size">axis_size</code></td>
<td>
<p>Numeric: axis title font size. Default: 8, min: 0, max: NULL.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_cell_height">cell_height</code></td>
<td>
<p>Numeric: cell height for value size and axis size. Default: 10.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_low_color">low_color</code></td>
<td>
<p>Character: min value color (color value or hex value with alpha). Default: &quot;#00880055&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_mid_color">mid_color</code></td>
<td>
<p>Character: min value color (color value or hex value with alpha). Default: &quot;#ffffff&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_high_color">high_color</code></td>
<td>
<p>Character: min value color (color value or hex value with alpha). Default: &quot;#ff000055&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_na_color">na_color</code></td>
<td>
<p>Character: min value color (color value or hex value with alpha). Default: &quot;#ff8800&quot;.</p>
</td></tr>
<tr><td><code id="heatmap_group_+3A_x_angle">x_angle</code></td>
<td>
<p>Numeric: x axis text angle. Default: 45, min: 0, max: 360.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: Heatmap group for visualizing grouped gene expression data.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression2)
head(gene_expression2)

data(samples_groups)
head(samples_groups)

# 3. Default parameters
heatmap_group(gene_expression2[1:50,], samples_groups)

# 4. Set scale_data = "column"
heatmap_group(gene_expression2[1:50,], samples_groups, scale_data = "column")

# 5. Set value_show = FALSE
heatmap_group(gene_expression2[1:50,], samples_groups, value_show = FALSE)

# 6. Set low_color = "#00008888"
heatmap_group(gene_expression2[1:50,], samples_groups, low_color = "#00008888")

</code></pre>

<hr>
<h2 id='kegg_enrich'>KEGG enrichment analysis based on KEGG annotation results (None/Exist Reference Genome).</h2><span id='topic+kegg_enrich'></span>

<h3>Description</h3>

<p>KEGG enrichment analysis based on KEGG annotation results (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kegg_enrich(
  kegg_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kegg_enrich_+3A_kegg_anno">kegg_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="kegg_enrich_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="kegg_enrich_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="kegg_enrich_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="kegg_enrich_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: include columns (&quot;ID&quot;, &quot;Description&quot;, &quot;GeneRatio&quot;, &quot;BgRatio&quot;, &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;, &quot;geneID&quot;, &quot;Count&quot;).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
res &lt;- kegg_enrich(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1])
head(res)

# 4. Set padjust_method = "BH"
res &lt;- kegg_enrich(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], padjust_method = "BH")
head(res)

# 5. Set pvalue_cutoff = 0.80
res &lt;- kegg_enrich(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], pvalue_cutoff = 0.80)
head(res)

</code></pre>

<hr>
<h2 id='kegg_enrich_bar'>KEGG enrichment analysis and bar plot (None/Exist Reference Genome).</h2><span id='topic+kegg_enrich_bar'></span>

<h3>Description</h3>

<p>KEGG enrichment analysis and bar plot (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kegg_enrich_bar(
  kegg_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05,
  sign_by = "p.adjust",
  category_num = 30,
  font_size = 12,
  low_color = "#ff0000aa",
  high_color = "#008800aa",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kegg_enrich_bar_+3A_kegg_anno">kegg_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_sign_by">sign_by</code></td>
<td>
<p>Character: significant by. Default: &quot;p.adjust&quot;, options: &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;.</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_category_num">category_num</code></td>
<td>
<p>Numeric: categories number to display. Default: 30, min: 1, max: NULL.</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_font_size">font_size</code></td>
<td>
<p>Numeric: category font size. Default: 12.</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_low_color">low_color</code></td>
<td>
<p>Character: low value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_high_color">high_color</code></td>
<td>
<p>Character: high value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="kegg_enrich_bar_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: KEGG enrichment analysis and bar plot (None/Exist Reference Genome).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
kegg_enrich_bar(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1])

# 4. Set padjust_method = "BH"
kegg_enrich_bar(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], padjust_method = "BH")

# 5. Set category_num = 10
kegg_enrich_bar(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], category_num = 10)

# 6. Set ggTheme = "theme_bw"
kegg_enrich_bar(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], ggTheme = "theme_bw")

</code></pre>

<hr>
<h2 id='kegg_enrich_dot'>KEGG enrichment analysis and dot plot (None/Exist Reference Genome).</h2><span id='topic+kegg_enrich_dot'></span>

<h3>Description</h3>

<p>KEGG enrichment analysis and dot plot (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kegg_enrich_dot(
  kegg_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05,
  sign_by = "p.adjust",
  category_num = 30,
  font_size = 12,
  low_color = "#ff0000aa",
  high_color = "#008800aa",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kegg_enrich_dot_+3A_kegg_anno">kegg_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_sign_by">sign_by</code></td>
<td>
<p>Character: significant by. Default: &quot;p.adjust&quot;, options: &quot;pvalue&quot;, &quot;p.adjust&quot;, &quot;qvalue&quot;.</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_category_num">category_num</code></td>
<td>
<p>Numeric: categories number to display. Default: 30, min: 1, max: NULL.</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_font_size">font_size</code></td>
<td>
<p>Numeric: category font size. Default: 12.</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_low_color">low_color</code></td>
<td>
<p>Character: low value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_high_color">high_color</code></td>
<td>
<p>Character: high value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="kegg_enrich_dot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: KEGG enrichment analysis and dot plot (None/Exist Reference Genome).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
kegg_enrich_dot(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1])

# 4. Set padjust_method = "BH"
kegg_enrich_dot(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], padjust_method = "BH")

# 5. Set category_num = 10
kegg_enrich_dot(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], category_num = 10)

# 6. Set ggTheme = "theme_bw"
kegg_enrich_dot(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], ggTheme = "theme_bw")

</code></pre>

<hr>
<h2 id='kegg_enrich_net'>KEGG enrichment analysis and net plot (None/Exist Reference Genome).</h2><span id='topic+kegg_enrich_net'></span>

<h3>Description</h3>

<p>KEGG enrichment analysis and net plot (None/Exist Reference Genome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kegg_enrich_net(
  kegg_anno,
  degs_list,
  padjust_method = "fdr",
  pvalue_cutoff = 0.05,
  qvalue_cutoff = 0.05,
  category_num = 20,
  net_layout = "circle",
  net_circular = TRUE,
  low_color = "#ff0000aa",
  high_color = "#008800aa"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kegg_enrich_net_+3A_kegg_anno">kegg_anno</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_degs_list">degs_list</code></td>
<td>
<p>Dataframe: degs list.</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_padjust_method">padjust_method</code></td>
<td>
<p>Character: P-value adjust to Q-value. Default: &quot;fdr&quot; (false discovery rate), options: &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>Numeric: P-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_qvalue_cutoff">qvalue_cutoff</code></td>
<td>
<p>Numeric: Q-value cutoff. Recommend: small than 0.05.</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_category_num">category_num</code></td>
<td>
<p>Numeric: categories number to display. Default: 20, min: 1, max: NULL.</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_net_layout">net_layout</code></td>
<td>
<p>Character: network layout. Default: &quot;circle&quot;, options: 'star', 'circle', 'gem', 'dh', 'graphopt', 'grid', 'mds', 'randomly', 'fr', 'kk', 'drl' or 'lgl'.</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_net_circular">net_circular</code></td>
<td>
<p>Logical: network circular. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_low_color">low_color</code></td>
<td>
<p>Character: low value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
<tr><td><code id="kegg_enrich_net_+3A_high_color">high_color</code></td>
<td>
<p>Character: high value (p-value or q-value) color (color name or hex value).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: KEGG enrichment analysis and net plot (None/Exist Reference Genome).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
kegg_enrich_net(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1])

# 4. Set category_num = 10
kegg_enrich_net(gene_go_kegg[,c(1,5)], gene_go_kegg[100:200,1], category_num = 10)

</code></pre>

<hr>
<h2 id='ma_plot'>MversusA plot for visualizing differentially expressed genes.</h2><span id='topic+ma_plot'></span>

<h3>Description</h3>

<p>MversusA plot for visualizing differentially expressed genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ma_plot(
  data,
  foldchange = 1,
  fdr_value = 0.05,
  point_size = 3,
  color_up = "#FF0000",
  color_down = "#008800",
  color_alpha = 0.5,
  top_method = "fc",
  top_num = 20,
  label_size = 8,
  label_box = TRUE,
  title = "CT-vs-LT12",
  xlab = "Log2 mean expression",
  ylab = "Log2 fold change",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ma_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: differentially expressed genes (DEGs) stats 2 (1st-col: Gene, 2nd-col: baseMean, 3rd-col: Log2FoldChange, 4th-col: FDR).</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_foldchange">foldchange</code></td>
<td>
<p>Numeric: fold change value. Default: 1.0, min: 0.0, max: null.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_fdr_value">fdr_value</code></td>
<td>
<p>Numeric: false discovery rate. Default: 0.05, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric: point size. Default: 1.0, min: 0.0, max: null.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_color_up">color_up</code></td>
<td>
<p>Character: up-regulated genes color (color name or hex value). Default: &quot;#FF0000&quot;.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_color_down">color_down</code></td>
<td>
<p>Character: down-regulated genes color (color name or hex value). Default: &quot;#008800&quot;.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_color_alpha">color_alpha</code></td>
<td>
<p>Numeric: point color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_top_method">top_method</code></td>
<td>
<p>Character: top genes select method. Default: &quot;fc&quot; (fold change), options: &quot;padj&quot; (p-adjust), &quot;fc&quot;.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_top_num">top_num</code></td>
<td>
<p>Numeric: top genes number. Default: 20, min: 0, max: null.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric: label font size. Default: 8.00, min: 0.00, max: null.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_label_box">label_box</code></td>
<td>
<p>Logical: add box to label. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_title">title</code></td>
<td>
<p>Character: plot title. Default: &quot;CT-vs-Trait1&quot;.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_xlab">xlab</code></td>
<td>
<p>Character: x label. Default: &quot;Log2 mean expression&quot;.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_ylab">ylab</code></td>
<td>
<p>Character: y label. Default: &quot;Log2 fold change&quot;.</p>
</td></tr>
<tr><td><code id="ma_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: MversusA plot for visualizing differentially expressed genes.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(degs_stats2)
head(degs_stats2)

# 3. Default parameters
ma_plot(degs_stats2)

# 4. Set color_up = "#FF8800"
ma_plot(degs_stats2, color_up = "#FF8800")

# 5. Set top_num = 10
ma_plot(degs_stats2, top_num = 10)

</code></pre>

<hr>
<h2 id='network_data'>Network data from WGCNA tan module top-200 dataframe.</h2><span id='topic+network_data'></span>

<h3>Description</h3>

<p>Network data from WGCNA tan module top-200 dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(network_data)
</code></pre>


<h3>Format</h3>

<p>Dataframe: Network data from WGCNA tan module top-200 dataframe (1st-col: Source, 2nd-col: Target).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/NetworkPlot/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(network_data)

# 3. View example data
network_data

</code></pre>

<hr>
<h2 id='network_plot'>Network plot for analyzing and visualizing relationship of genes.</h2><span id='topic+network_plot'></span>

<h3>Description</h3>

<p>Network plot for analyzing and visualizing relationship of genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_plot(
  data,
  calc_by = "degree",
  degree_value = 0.5,
  normal_color = "#008888cc",
  border_color = "#FFFFFF",
  from_color = "#FF0000cc",
  to_color = "#008800cc",
  normal_shape = "circle",
  spatial_shape = "circle",
  node_size = 25,
  lable_color = "#FFFFFF",
  label_size = 0.5,
  edge_color = "#888888",
  edge_width = 1.5,
  edge_curved = TRUE,
  net_layout = "layout_on_sphere"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="network_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: Network data from WGCNA tan module top-200 dataframe (1st-col: Source, 2nd-col: Target).</p>
</td></tr>
<tr><td><code id="network_plot_+3A_calc_by">calc_by</code></td>
<td>
<p>Character: calculate relationship by &quot;degree&quot;, &quot;node&quot;. Default: &quot;degree&quot;.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_degree_value">degree_value</code></td>
<td>
<p>Numeric: degree value when calc_by = &quot;degree&quot;. Default: 0.05, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_normal_color">normal_color</code></td>
<td>
<p>Character: normal relationship nodes color (color name of hex value).</p>
</td></tr>
<tr><td><code id="network_plot_+3A_border_color">border_color</code></td>
<td>
<p>Character: node border color (color name or hex value).</p>
</td></tr>
<tr><td><code id="network_plot_+3A_from_color">from_color</code></td>
<td>
<p>Character: the start color of nodes that meet degree_value.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_to_color">to_color</code></td>
<td>
<p>Character: the end color of nodes that meet degree_value.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_normal_shape">normal_shape</code></td>
<td>
<p>Character: normal node shape. Default: &quot;circle&quot;, options: &quot;circle&quot;, &quot;crectangle&quot;, &quot;csquare&quot;, &quot;none&quot;, &quot;pie&quot;, &quot;raster&quot;, &quot;rectangle&quot;, &quot;sphere&quot;, &quot;square&quot;, &quot;vrectangle&quot;.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_spatial_shape">spatial_shape</code></td>
<td>
<p>Character: meet degree_value node shape. Default: &quot;csquare&quot;, options: &quot;circle&quot;, &quot;crectangle&quot;, &quot;csquare&quot;, &quot;none&quot;, &quot;pie&quot;, &quot;raster&quot;, &quot;rectangle&quot;, &quot;sphere&quot;, &quot;square&quot;, &quot;vrectangle&quot;.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_node_size">node_size</code></td>
<td>
<p>Numeric: node size. Default: 10, min: 0, max: NULL.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_lable_color">lable_color</code></td>
<td>
<p>Character: gene labels color. Default: &quot;#FFFFFF&quot;.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric: node label size. Default: 0.5, min: 0, max: NULL.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_edge_color">edge_color</code></td>
<td>
<p>Character: edges color. Default: &quot;#888888&quot;.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_edge_width">edge_width</code></td>
<td>
<p>Numeric: edges width. Default: 1.5.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_edge_curved">edge_curved</code></td>
<td>
<p>Logical: curved edges. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="network_plot_+3A_net_layout">net_layout</code></td>
<td>
<p>Character: network layout. Default: &quot;layout_on_sphere&quot;, options: &quot;layout_as_bipartite&quot;, &quot;layout_as_star&quot;, &quot;layout_as_tree&quot;, &quot;layout_components&quot;, &quot;layout_in_circle&quot;, &quot;layout_nicely&quot;, &quot;layout_on_grid&quot;, &quot;layout_on_sphere&quot;,&quot;layout_randomly&quot;,&quot;layout_with_dh&quot;,&quot;layout_with_drl&quot;,&quot;layout_with_fr&quot;,&quot;layout_with_gem&quot;,&quot;layout_with_graphopt&quot;,&quot;layout_with_kk&quot;,&quot;layout_with_lgl&quot;,&quot;layout_with_mds&quot;,&quot;layout_with_sugiyama&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: network plot for analyzing and visualizing relationship of genes.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(network_data)
head(network_data)

# 3. Default parameters
network_plot(network_data)

# 4. Set calc_by = "node"
network_plot(network_data, calc_by = "node")

# 5. Set degree_value = 0.1
network_plot(network_data, degree_value = 0.1)

# 6. Set normal_color = "#ff8800cc"
network_plot(network_data, normal_color = "#ff8800cc")

# 7. Set net_layout = "layout_as_tree"
network_plot(network_data, net_layout = "layout_as_tree")

</code></pre>

<hr>
<h2 id='pca_analysis'>PCA dimensional reduction analysis for RNA-Seq.</h2><span id='topic+pca_analysis'></span>

<h3>Description</h3>

<p>PCA dimensional reduction analysis for RNA-Seq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca_analysis(sample_gene, group_sample)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_analysis_+3A_sample_gene">sample_gene</code></td>
<td>
<p>Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="pca_analysis_+3A_group_sample">group_sample</code></td>
<td>
<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: PCA dimensional reduction analysis for RNA-Seq.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library package TOmicsVis
library(TOmicsVis)

# 2. Load example datasets
data(gene_expression)
head(gene_expression)

data(samples_groups)
head(samples_groups)

# 3. Default parameters
res &lt;- pca_analysis(gene_expression, samples_groups)
head(res)

</code></pre>

<hr>
<h2 id='pca_plot'>PCA dimensional reduction visualization for RNA-Seq.</h2><span id='topic+pca_plot'></span>

<h3>Description</h3>

<p>PCA dimensional reduction visualization for RNA-Seq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca_plot(
  sample_gene,
  group_sample,
  xPC = 1,
  yPC = 2,
  multi_shape = TRUE,
  point_size = 5,
  point_alpha = 0.8,
  text_size = 5,
  fill_alpha = 0.05,
  border_alpha = 0,
  sci_fill_color = "Sci_AAAS",
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_plot_+3A_sample_gene">sample_gene</code></td>
<td>
<p>Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_group_sample">group_sample</code></td>
<td>
<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_xpc">xPC</code></td>
<td>
<p>Numeric: PC index at x axis. Default: 1, options: 1, 2, 3, ...</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_ypc">yPC</code></td>
<td>
<p>Numeric: PC index at y axis. Default: 2, options: 2, 3, 4, ...</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_multi_shape">multi_shape</code></td>
<td>
<p>Logical: groups as shapes. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric: PCA plot point size. Default: 5, min: 0.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_point_alpha">point_alpha</code></td>
<td>
<p>Numeric: point color alpha. Default: 0.80, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_text_size">text_size</code></td>
<td>
<p>Numeric: PCA plot annotation size. Default: 5, min: 0.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_fill_alpha">fill_alpha</code></td>
<td>
<p>Numeric: ellipse fill color alpha. Default: 0.10, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_border_alpha">border_alpha</code></td>
<td>
<p>Numeric: ellipse border color alpha. Default: 0.10, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci color pallet. Default: &quot;Sci_AAAS&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character: legend position. Default: &quot;right&quot;, options: &quot;none&quot;, &quot;left&quot;, &quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_legend_dir">legend_dir</code></td>
<td>
<p>Character: legend director. Default: &quot;vertical&quot;, options: &quot;horizontal&quot;, &quot;vertical&quot;.</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 theme. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: PCA dimensional reduction visualization for RNA-Seq.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library package TOmicsVis
library(TOmicsVis)

# 2. Load example datasets
data(gene_expression)
head(gene_expression)

data(samples_groups)
head(samples_groups)

# 3. Default parameters
pca_plot(gene_expression, samples_groups)

# 4. Set multi_shape = FALSE
pca_plot(gene_expression, samples_groups, multi_shape = FALSE)

# 5. Set sci_fill_color = "Sci_NPG", fill_alpha = 0.10
pca_plot(gene_expression, samples_groups, sci_fill_color = "Sci_NPG", fill_alpha = 0.10)

</code></pre>

<hr>
<h2 id='quantile_plot'>Quantile plot for visualizing data distribution.</h2><span id='topic+quantile_plot'></span>

<h3>Description</h3>

<p>Quantile plot for visualizing data distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quantile_plot(
  data,
  my_shape = "fill_circle",
  point_size = 1.5,
  conf_int = TRUE,
  conf_level = 0.95,
  split_panel = "Split_Panel",
  legend_pos = "right",
  legend_dir = "vertical",
  sci_fill_color = "Sci_NPG",
  sci_color_alpha = 0.75,
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quantile_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: Weight and Sex traits dataframe (1st-col: Weight, 2nd-col: Sex).</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_my_shape">my_shape</code></td>
<td>
<p>Character: scatter shape. Default: &quot;fill_circle&quot;, options: &quot;border_square&quot;, &quot;border_circle&quot;, &quot;border_triangle&quot;, &quot;plus&quot;, &quot;times&quot;, &quot;border_diamond&quot;, &quot;border_triangle_down&quot;, &quot;square_times&quot;, &quot;plus_times&quot;, &quot;diamond_plus&quot;, &quot;circle_plus&quot;, &quot;di_triangle&quot;, &quot;square_plus&quot;, &quot;circle_times&quot;,&quot;square_triangle&quot;, &quot;fill_square&quot;, &quot;fill_circle&quot;, &quot;fill_triangle&quot;, &quot;fill_diamond&quot;, &quot;large_circle&quot;, &quot;small_circle&quot;, &quot;fill_border_circle&quot;, &quot;fill_border_square&quot;, &quot;fill_border_diamond&quot;, &quot;fill_border_triangle&quot;.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric: point size. Default: 1.5, min: 0.0, max: not required.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_conf_int">conf_int</code></td>
<td>
<p>Logical: confidence interval (CI). Default: TRUE, options: TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_conf_level">conf_level</code></td>
<td>
<p>Numeric: confidence interval value. Default: 0.95, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_split_panel">split_panel</code></td>
<td>
<p>Character: split panel by groups. Default: &quot;Split_Panel&quot;, options: &quot;One_Panel&quot;, &quot;Split_Panel&quot;.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character: legend position. Default: &quot;right&quot;, options: &quot;none&quot;, &quot;left&quot;, &quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_legend_dir">legend_dir</code></td>
<td>
<p>Character: legend direction. Default: &quot;vertical&quot;, options: &quot;horizontal&quot;, &quot;vertical&quot;.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci fill or color palette. Default: &quot;Sci_NPG&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_sci_color_alpha">sci_color_alpha</code></td>
<td>
<p>Numeric: ggsci border color alpha. Default: 0.75, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="quantile_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: quantile plot for visualizing data distribution.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(weight_sex)
head(weight_sex)

# 3. Default parameters
quantile_plot(weight_sex)

# 4. Set split_panel = "Split_Panel"
quantile_plot(weight_sex, split_panel = "Split_Panel")

# 5. Set sci_fill_color = "Sci_Futurama"
quantile_plot(weight_sex, sci_fill_color = "Sci_Futurama")

# 6. Set conf_int = FALSE
quantile_plot(weight_sex, conf_int = FALSE)

</code></pre>

<hr>
<h2 id='samples_groups'>Samples and groups for gene expression.</h2><span id='topic+samples_groups'></span>

<h3>Description</h3>

<p>Samples and groups for gene expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(samples_groups)
</code></pre>


<h3>Format</h3>

<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/PCAplot/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example dataset samples_groups
data(samples_groups)

# 3. View samples_groups
samples_groups

</code></pre>

<hr>
<h2 id='survival_data'>Survival data as example data for survival_plot function.</h2><span id='topic+survival_data'></span>

<h3>Description</h3>

<p>Survival data as example data for survival_plot function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(survival_data)
</code></pre>


<h3>Format</h3>

<p>Dataframe: survival record data (1st-col: Time, 2nd-col: Status, 3rd-col: Group).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/SurvivalAnalysis/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(survival_data)

# 3. View example data
survival_data

</code></pre>

<hr>
<h2 id='survival_plot'>Survival plot for analyzing and visualizing survival data.</h2><span id='topic+survival_plot'></span>

<h3>Description</h3>

<p>Survival plot for analyzing and visualizing survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>survival_plot(
  data,
  curve_function = "pct",
  conf_inter = TRUE,
  interval_style = "ribbon",
  risk_table = TRUE,
  num_censor = TRUE,
  sci_palette = "aaas",
  ggTheme = "theme_light",
  x_start = 0,
  y_start = 0,
  y_end = 100,
  x_break = 10,
  y_break = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="survival_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: survival record data (1st-col: Time, 2nd-col: Status, 3rd-col: Group).</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_curve_function">curve_function</code></td>
<td>
<p>Character: an arbitrary function defining a transformation of the survival curve. Often used transformations can be specified with a character argument: &quot;event&quot; plots cumulative events (f(y) = 1-y), &quot;cumhaz&quot; plots the cumulative hazard function (f(y) = -log(y)), and &quot;pct&quot; for survival probability in percentage.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_conf_inter">conf_inter</code></td>
<td>
<p>Logical: confidence interval. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_interval_style">interval_style</code></td>
<td>
<p>Character: confidence interval style. Default: &quot;ribbon&quot;, options: &quot;ribbon&quot;, &quot;step&quot;.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_risk_table">risk_table</code></td>
<td>
<p>Logical: show cumulative risk table. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_num_censor">num_censor</code></td>
<td>
<p>Logical: show cumulative number of censoring. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_sci_palette">sci_palette</code></td>
<td>
<p>Character: ggsci color palette. Default: &quot;aaas&quot;, options: &quot;aaas&quot;, &quot;npg&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot;, &quot;rickandmorty&quot;.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_x_start">x_start</code></td>
<td>
<p>Numeric: x-axis start value. Default: 0, min: 0, max: null.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_y_start">y_start</code></td>
<td>
<p>Numeric: y-axis start value. Default: 0, min: 0, max: 100.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_y_end">y_end</code></td>
<td>
<p>Numeric: y-axis end value. Default: 100, min: 0, max: 100.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_x_break">x_break</code></td>
<td>
<p>Numeric: x-axis break value. Default: 10, min: 0, max: null.</p>
</td></tr>
<tr><td><code id="survival_plot_+3A_y_break">y_break</code></td>
<td>
<p>Numeric: y-axis break value. Default: 10, min: 0, max: 100.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: survival plot for analyzing and visualizing survival data.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(survival_data)
head(survival_data)

# 3. Default parameters
survival_plot(survival_data)

# 4. Set conf_inter = FALSE
survival_plot(survival_data, conf_inter = FALSE)

# 5. Set sci_palette = "jco"
survival_plot(survival_data, sci_palette = "jco")

</code></pre>

<hr>
<h2 id='table_cross'>Table cross used to cross search and merge results in two tables.</h2><span id='topic+table_cross'></span>

<h3>Description</h3>

<p>Table cross used to cross search and merge results in two tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table_cross(
  data1,
  data2,
  inter_var = "Genes",
  left_index = TRUE,
  right_index = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="table_cross_+3A_data1">data1</code></td>
<td>
<p>Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="table_cross_+3A_data2">data2</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="table_cross_+3A_inter_var">inter_var</code></td>
<td>
<p>Character: Intersecting variable (column name). Default: &quot;geneID&quot; in example data.</p>
</td></tr>
<tr><td><code id="table_cross_+3A_left_index">left_index</code></td>
<td>
<p>Logical: left table as index. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="table_cross_+3A_right_index">right_index</code></td>
<td>
<p>Logical: right table as index. Default: FALSE, options: TRUE, FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: include multiple columns.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression2)
head(gene_expression2)

data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
res &lt;- table_cross(gene_expression2, gene_go_kegg, inter_var = "Genes")
head(res)

# 4. Set left_index = TRUE, right_index = FALSE
res &lt;- table_cross(gene_expression2, gene_go_kegg,
inter_var = "Genes", left_index = TRUE, right_index = FALSE)
head(res)

# 5. Set left_index = FALSE, right_index = TRUE
res &lt;- table_cross(gene_expression2, gene_go_kegg,
inter_var = "Genes", left_index = FALSE, right_index = TRUE)
head(res)

</code></pre>

<hr>
<h2 id='table_filter'>Table filter used to filter row by column condition.</h2><span id='topic+table_filter'></span>

<h3>Description</h3>

<p>Table filter used to filter row by column condition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table_filter(data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="table_filter_+3A_data">data</code></td>
<td>
<p>Dataframe: Length, Width, Weight, and Sex traits dataframe (1st-col: Value, 2nd-col: Traits, 3rd-col: Sex).</p>
</td></tr>
<tr><td><code id="table_filter_+3A_...">...</code></td>
<td>
<p>Expression: multiple expressions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: table filter used to filter row by column condition.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(traits_sex)
head(traits_sex)

# 3. Set height &gt; 100 &amp; eye_color == "black"
res &lt;- table_filter(traits_sex, Sex == "Male" &amp; Traits == "Weight" &amp; Value &gt; 40)
head(res)

</code></pre>

<hr>
<h2 id='table_merge'>Table merge used to merge multiple variables to on variable.</h2><span id='topic+table_merge'></span>

<h3>Description</h3>

<p>Table merge used to merge multiple variables to on variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table_merge(
  data,
  merge_vars = c("biological_process", "cellular_component", "molecular_function"),
  new_var = "go_category",
  new_value = "go_term",
  na_remove = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="table_merge_+3A_data">data</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="table_merge_+3A_merge_vars">merge_vars</code></td>
<td>
<p>Vector: include merged variable (column) names. Default: c(&quot;Ozone&quot;, &quot;Solar.R&quot;, &quot;Wind&quot;, &quot;Temp&quot;) in example data.</p>
</td></tr>
<tr><td><code id="table_merge_+3A_new_var">new_var</code></td>
<td>
<p>Character: new variable (column) name. Default: &quot;Variable&quot;.</p>
</td></tr>
<tr><td><code id="table_merge_+3A_new_value">new_value</code></td>
<td>
<p>Character: new variable (column) value name. Default: &quot;Value&quot;.</p>
</td></tr>
<tr><td><code id="table_merge_+3A_na_remove">na_remove</code></td>
<td>
<p>Logical: remove NA value. Default: FALSE, options: TRUE, FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: include multiple variables.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg)
head(gene_go_kegg)

# 3. Default parameters
res &lt;- table_merge(gene_go_kegg)
head(res)

# 4. Set new_var = "GO", new_value = "Terms"
res &lt;- table_merge(gene_go_kegg, new_var = "GO", new_value = "Terms")
head(res)

</code></pre>

<hr>
<h2 id='table_split'>Table split used for splitting a grouped column to multiple columns.</h2><span id='topic+table_split'></span>

<h3>Description</h3>

<p>Table split used for splitting a grouped column to multiple columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table_split(
  data,
  grouped_var = "go_category",
  value_var = "go_term",
  miss_drop = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="table_split_+3A_data">data</code></td>
<td>
<p>Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
</td></tr>
<tr><td><code id="table_split_+3A_grouped_var">grouped_var</code></td>
<td>
<p>Character: grouped column name. Default: &quot;go_category&quot;.</p>
</td></tr>
<tr><td><code id="table_split_+3A_value_var">value_var</code></td>
<td>
<p>Character: value column name. Default: &quot;go_term&quot;.</p>
</td></tr>
<tr><td><code id="table_split_+3A_miss_drop">miss_drop</code></td>
<td>
<p>Logical: drop missing values or NA values. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: table split used for splitting a grouped column to multiple columns.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_go_kegg2)
head(gene_go_kegg2)

# 3. Default parameters
res &lt;- table_split(gene_go_kegg2)
head(res)

</code></pre>

<hr>
<h2 id='tomicsvis'>TOmicsVis shiny app start function.</h2><span id='topic+tomicsvis'></span>

<h3>Description</h3>

<p>TOmicsVis shiny app start function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tomicsvis()
</code></pre>


<h3>Value</h3>

<p>Shinyapp: TOmicsVis shiny app.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

</code></pre>

<hr>
<h2 id='traits_sex'>Length, Width, Weight, and Sex traits dataframe.</h2><span id='topic+traits_sex'></span>

<h3>Description</h3>

<p>Length, Width, Weight, and Sex traits dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(traits_sex)
</code></pre>


<h3>Format</h3>

<p>Dataframe: Length, Width, Weight, and Sex traits dataframe (1st-col: Value, 2nd-col: Traits, 3rd-col: Sex).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/BoxStat/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(traits_sex)

# 3. View example data
traits_sex

</code></pre>

<hr>
<h2 id='trend_plot'>Trend plot for visualizing gene expression trend profile in multiple traits.</h2><span id='topic+trend_plot'></span>

<h3>Description</h3>

<p>Trend plot for visualizing gene expression trend profile in multiple traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trend_plot(
  data,
  scale_method = "centerObs",
  miss_value = "exclude",
  line_alpha = 0.5,
  show_points = TRUE,
  show_boxplot = TRUE,
  num_column = 1,
  xlab = "Traits",
  ylab = "Genes Expression",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.8,
  sci_color_alpha = 0.8,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trend_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: Shared degs of all paired comparisons in all groups expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~n-1-col: Groups, n-col: Pathways).</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_scale_method">scale_method</code></td>
<td>
<p>Character: data scale methods. Default: &quot;globalminmax&quot; (global min and max values), options: &quot;std&quot; (standard), &quot;robust&quot;, &quot;uniminmax&quot; (unique min and max values), &quot;globalminmax&quot;, &quot;center&quot;, &quot;centerObs&quot; (center observes).</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_miss_value">miss_value</code></td>
<td>
<p>Character: deal method for missing values. Default: &quot;exclude&quot;, options: &quot;exclude&quot;, &quot;mean&quot;, &quot;median&quot;, &quot;min10&quot;, &quot;random&quot;.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_line_alpha">line_alpha</code></td>
<td>
<p>Numeric: lines color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_show_points">show_points</code></td>
<td>
<p>Logical: show points at trait node. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_show_boxplot">show_boxplot</code></td>
<td>
<p>Logical: show boxplot at trait node. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_num_column">num_column</code></td>
<td>
<p>Logical: column number. Default: 2, min: 1, max: null.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_xlab">xlab</code></td>
<td>
<p>Character: x label. Default: &quot;Traits&quot;.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_ylab">ylab</code></td>
<td>
<p>Character: y label. Default: &quot;Genes Expression&quot;.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci color pallet. Default: &quot;Sci_AAAS&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_sci_fill_alpha">sci_fill_alpha</code></td>
<td>
<p>Numeric: ggsci fill color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_sci_color_alpha">sci_color_alpha</code></td>
<td>
<p>Numeric: ggsci border color alpha. Default: 1.00, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character: legend position. Default: &quot;right&quot;, options: &quot;none&quot;, &quot;left&quot;, &quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_legend_dir">legend_dir</code></td>
<td>
<p>Character: legend direction. Default: &quot;vertical&quot;, options: &quot;horizontal&quot;, &quot;vertical&quot;.</p>
</td></tr>
<tr><td><code id="trend_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: box plot support two levels and multiple groups with P value.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression3)
head(gene_expression3)

# 3. Default parameters
trend_plot(gene_expression3[1:50,])

# 4. Set line_alpha = 0.30
trend_plot(gene_expression3[1:50,], line_alpha = 0.30)

# 5. Set sci_fill_color = "Sci_NPG"
trend_plot(gene_expression3[1:50,], sci_fill_color = "Sci_NPG")

</code></pre>

<hr>
<h2 id='tsne_analysis'>TSNE analysis for analyzing and visualizing TSNE algorithm.</h2><span id='topic+tsne_analysis'></span>

<h3>Description</h3>

<p>TSNE analysis for analyzing and visualizing TSNE algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsne_analysis(sample_gene, group_sample, seed = 1, tsne_dims = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsne_analysis_+3A_sample_gene">sample_gene</code></td>
<td>
<p>Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="tsne_analysis_+3A_group_sample">group_sample</code></td>
<td>
<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
</td></tr>
<tr><td><code id="tsne_analysis_+3A_seed">seed</code></td>
<td>
<p>Numeric: set seed for robust result. Default: 1.</p>
</td></tr>
<tr><td><code id="tsne_analysis_+3A_tsne_dims">tsne_dims</code></td>
<td>
<p>Numeric: TSNE dimensionality number. Default: 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: TSNE analysis for analyzing and visualizing TSNE algorithm.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression)
head(gene_expression)

data(samples_groups)
head(samples_groups)

# 3. Default parameters
res &lt;- tsne_analysis(gene_expression, samples_groups)
head(res)

# 4. Set tsne_dims = 3
res &lt;- tsne_analysis(gene_expression, samples_groups, tsne_dims = 3)
head(res)

</code></pre>

<hr>
<h2 id='tsne_plot'>TSNE plot for analyzing and visualizing TSNE algorithm.</h2><span id='topic+tsne_plot'></span>

<h3>Description</h3>

<p>TSNE plot for analyzing and visualizing TSNE algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsne_plot(
  sample_gene,
  group_sample,
  seed = 1,
  multi_shape = FALSE,
  point_size = 5,
  point_alpha = 0.8,
  text_size = 5,
  text_alpha = 0.8,
  fill_alpha = 0.1,
  border_alpha = 0,
  sci_fill_color = "Sci_AAAS",
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsne_plot_+3A_sample_gene">sample_gene</code></td>
<td>
<p>Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_group_sample">group_sample</code></td>
<td>
<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_seed">seed</code></td>
<td>
<p>Numeric: set seed for robust result. Default: 1.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_multi_shape">multi_shape</code></td>
<td>
<p>Logical: groups as shapes. Default: FALSE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric: point size. Default: 5, min: 0, max: null.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_point_alpha">point_alpha</code></td>
<td>
<p>Numeric: point color alpha. Default: 0.80, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_text_size">text_size</code></td>
<td>
<p>Numeric: text size. Default: 5, min: 0 (hind), max: null.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_text_alpha">text_alpha</code></td>
<td>
<p>Numeric: text alpha. Default: 0.80, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_fill_alpha">fill_alpha</code></td>
<td>
<p>Numeric: ellipse alpha. Default: 0.30, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_border_alpha">border_alpha</code></td>
<td>
<p>Numeric: ellipse border color alpha. Default: 0.10, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci color pallet. Default: &quot;Sci_AAAS&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character: legend position. Default: &quot;right&quot;, options: &quot;none&quot;, &quot;left&quot;, &quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_legend_dir">legend_dir</code></td>
<td>
<p>Character: legend direction. Default: &quot;vertical&quot;, options: &quot;horizontal&quot;, &quot;vertical&quot;.</p>
</td></tr>
<tr><td><code id="tsne_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: TSNE plot for analyzing and visualizing TSNE algorithm.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression)
head(gene_expression)

data(samples_groups)
head(samples_groups)

# 3. Default parameters
tsne_plot(gene_expression, samples_groups)

# 4. Set sci_fill_color = "Sci_NPG", seed = 6
tsne_plot(gene_expression, samples_groups, sci_fill_color = "Sci_NPG", seed = 6)

# 5. Set multi_shape = TRUE, fill_alpha = 0.00
tsne_plot(gene_expression, samples_groups, multi_shape = TRUE, fill_alpha = 0.00)

</code></pre>

<hr>
<h2 id='umap_analysis'>UMAP analysis for analyzing RNA-Seq data.</h2><span id='topic+umap_analysis'></span>

<h3>Description</h3>

<p>UMAP analysis for analyzing RNA-Seq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umap_analysis(sample_gene, group_sample, seed = 1, method = "naive")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umap_analysis_+3A_sample_gene">sample_gene</code></td>
<td>
<p>Dataframe: gene expression dataframe (1st-col: Transcripts or Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="umap_analysis_+3A_group_sample">group_sample</code></td>
<td>
<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
</td></tr>
<tr><td><code id="umap_analysis_+3A_seed">seed</code></td>
<td>
<p>Numeric: set seed for robust result. Default: 1.</p>
</td></tr>
<tr><td><code id="umap_analysis_+3A_method">method</code></td>
<td>
<p>Character: 'naive' (an implementation written in pure R) and 'umap-learn' (requires python package 'umap-learn').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table: UMAP analysis for analyzing RNA-Seq data.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression)
head(gene_expression)

data(samples_groups)
head(samples_groups)

# 3. Default parameters
res &lt;- umap_analysis(gene_expression, samples_groups)
head(res)

</code></pre>

<hr>
<h2 id='umap_plot'>UMAP plot for analyzing and visualizing UMAP algorithm.</h2><span id='topic+umap_plot'></span>

<h3>Description</h3>

<p>UMAP plot for analyzing and visualizing UMAP algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umap_plot(
  sample_gene,
  group_sample,
  seed = 1,
  multi_shape = TRUE,
  point_size = 5,
  point_alpha = 1,
  text_size = 5,
  text_alpha = 0.8,
  fill_alpha = 0,
  border_alpha = 0,
  sci_fill_color = "Sci_AAAS",
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umap_plot_+3A_sample_gene">sample_gene</code></td>
<td>
<p>Dataframe: gene expression dataframe (1st-col: Transcripts or Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_group_sample">group_sample</code></td>
<td>
<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_seed">seed</code></td>
<td>
<p>Numeric: set seed for robust result. Default: 1.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_multi_shape">multi_shape</code></td>
<td>
<p>Logical: groups as shapes. Default: FALSE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric: point size. Default: 5, min: 0, max: null.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_point_alpha">point_alpha</code></td>
<td>
<p>Numeric: point color alpha. Default: 0.80, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_text_size">text_size</code></td>
<td>
<p>Numeric: text size. Default: 5, min: 0 (hind), max: null.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_text_alpha">text_alpha</code></td>
<td>
<p>Numeric: text alpha. Default: 0.80, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_fill_alpha">fill_alpha</code></td>
<td>
<p>Numeric: ellipse alpha. Default: 0.30, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_border_alpha">border_alpha</code></td>
<td>
<p>Numeric: ellipse border color alpha. Default: 0.10, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci color pallet. Default: &quot;Sci_AAAS&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character: legend position. Default: &quot;right&quot;, options: &quot;none&quot;, &quot;left&quot;, &quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_legend_dir">legend_dir</code></td>
<td>
<p>Character: legend direction. Default: &quot;vertical&quot;, options: &quot;horizontal&quot;, &quot;vertical&quot;.</p>
</td></tr>
<tr><td><code id="umap_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: UMAP plot for analyzing and visualizing UMAP algorithm.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression)
head(gene_expression)

data(samples_groups)
head(samples_groups)

# 3. Default parameters
umap_plot(gene_expression, samples_groups)

# 4. Set sci_fill_color = "Sci_Simpsons", seed = 6
umap_plot(gene_expression, samples_groups, sci_fill_color = "Sci_Simpsons", seed = 6)

# 5. Set fill_alpha = 0.10
umap_plot(gene_expression, samples_groups, fill_alpha = 0.10)

</code></pre>

<hr>
<h2 id='upsetr_plot'>UpSet plot for stat common and unique gene among multiple sets.</h2><span id='topic+upsetr_plot'></span>

<h3>Description</h3>

<p>UpSet plot for stat common and unique gene among multiple sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upsetr_plot(
  data,
  sets_num = 4,
  keep_order = FALSE,
  order_by = "freq",
  decrease = TRUE,
  mainbar_color = "#006600",
  number_angle = 45,
  matrix_color = "#cc0000",
  point_size = 4.5,
  point_alpha = 0.5,
  line_size = 0.8,
  shade_color = "#cdcdcd",
  shade_alpha = 0.5,
  setsbar_color = "#000066",
  setsnum_size = 6,
  text_scale = 1.2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="upsetr_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: Paired comparisons differentially expressed genes (degs) among groups (1st-col~: degs of paired comparisons).</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_sets_num">sets_num</code></td>
<td>
<p>Numeric: sets number. Default: 4, min: 2, max: NULL.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_keep_order">keep_order</code></td>
<td>
<p>Logical: keep sets in the order entered using the sets parameter. Default: FALSE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_order_by">order_by</code></td>
<td>
<p>Character: intersections in the matrix should be ordered by. Default: &quot;freq&quot; (frequency), options: &quot;freq&quot;, &quot;degree&quot;, &quot;both&quot;.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_decrease">decrease</code></td>
<td>
<p>Logical: order by decrease. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_mainbar_color">mainbar_color</code></td>
<td>
<p>Charactor: mainbar color (color name or hex value). Default: &quot;#006600&quot;.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_number_angle">number_angle</code></td>
<td>
<p>Numeric: number display angle. Default: 45, min: 0, max: 360.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_matrix_color">matrix_color</code></td>
<td>
<p>Charactor: matrix point color (color name or hex value). Default: &quot;#cc0000&quot;.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric: point size. Default: 4.5, min: 0.0, max: NULL.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_point_alpha">point_alpha</code></td>
<td>
<p>Numeric: point color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_line_size">line_size</code></td>
<td>
<p>Numeric: connection line size. Default: 0.8, min: 0.00, max: NULL.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_shade_color">shade_color</code></td>
<td>
<p>Character: matrix shade color. Default: &quot;#cdcdcd&quot;.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_shade_alpha">shade_alpha</code></td>
<td>
<p>Numeric: shade color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_setsbar_color">setsbar_color</code></td>
<td>
<p>Character: sets bar color. Default: &quot;#000066&quot;.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_setsnum_size">setsnum_size</code></td>
<td>
<p>Numeric: sets bar number size. Default: 6.</p>
</td></tr>
<tr><td><code id="upsetr_plot_+3A_text_scale">text_scale</code></td>
<td>
<p>Numeric: all text scale. Default: 1.2, min: 0.01, max: NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: UpSet plot for stat common and unique gene among multiple sets.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(degs_lists)
head(degs_lists)

# 3. Default parameters
upsetr_plot(degs_lists)

# 4. Set keep_order = TRUE, order_by = "degree"
upsetr_plot(degs_lists, keep_order = TRUE, order_by = "degree")

# 5. Set mainbar_color = "#333333", number_angle = 0
upsetr_plot(degs_lists, mainbar_color = "#333333", number_angle = 0)

# 6. Set shade_color = "#ffcc00", setsbar_color = "#0000cc"
upsetr_plot(degs_lists, shade_color = "#ffcc00", setsbar_color = "#0000cc")

</code></pre>

<hr>
<h2 id='venn_plot'>Venn plot for stat common and unique gene among multiple sets.</h2><span id='topic+venn_plot'></span>

<h3>Description</h3>

<p>Venn plot for stat common and unique gene among multiple sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>venn_plot(
  data,
  title_size = 1,
  label_show = TRUE,
  label_size = 0.8,
  border_show = TRUE,
  line_type = "longdash",
  ellipse_shape = "circle",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.65
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="venn_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: Paired comparisons differentially expressed genes (degs) among groups (1st-col~: degs of paired comparisons).</p>
</td></tr>
<tr><td><code id="venn_plot_+3A_title_size">title_size</code></td>
<td>
<p>Numeric: sets title size. Default: 1, min: 0, max: NULL.</p>
</td></tr>
<tr><td><code id="venn_plot_+3A_label_show">label_show</code></td>
<td>
<p>Logical: show intersection labels. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="venn_plot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric: intersection labels size. Default: 0.8, min: 0, max: NULL.</p>
</td></tr>
<tr><td><code id="venn_plot_+3A_border_show">border_show</code></td>
<td>
<p>Logical: show border line. Default: TRUE, options: TRUE, FALSE.</p>
</td></tr>
<tr><td><code id="venn_plot_+3A_line_type">line_type</code></td>
<td>
<p>Character: ellipse border line type. Default: &quot;blank&quot;, options: &quot;blank&quot;, &quot;solid&quot;, &quot;dashed&quot;, &quot;dotted&quot;, &quot;dotdash&quot;, &quot;longdash&quot;, &quot;twodash&quot;.</p>
</td></tr>
<tr><td><code id="venn_plot_+3A_ellipse_shape">ellipse_shape</code></td>
<td>
<p>Character: ellipse shape. Default: &quot;circle&quot;, options: &quot;circle&quot;, &quot;ellipse&quot;.</p>
</td></tr>
<tr><td><code id="venn_plot_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci color palette. Default: &quot;Sci_AAAS&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="venn_plot_+3A_sci_fill_alpha">sci_fill_alpha</code></td>
<td>
<p>Numeric: ggsci fill color alpha. Default: 0.65, min: 0.00, max: 1.00.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: venn plot for stat common and unique gene among multiple sets.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(degs_lists)
head(degs_lists)

# 3. Default parameters
venn_plot(degs_lists)

# 4. Set line_type = "blank"
venn_plot(degs_lists, line_type = "blank")

# 5. Set ellipse_shape = "ellipse"
venn_plot(degs_lists, ellipse_shape = "ellipse")

# 6. Set sci_fill_color = "Sci_IGV"
venn_plot(degs_lists, sci_fill_color = "Sci_IGV")

</code></pre>

<hr>
<h2 id='violin_plot'>Violin plot support two levels and multiple groups with P value.</h2><span id='topic+violin_plot'></span>

<h3>Description</h3>

<p>Violin plot support two levels and multiple groups with P value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>violin_plot(
  data,
  test_method = "t.test",
  test_label = "p.format",
  group_level = "Three_Column",
  violin_orientation = "vertical",
  add_element = "boxplot",
  element_alpha = 0.5,
  my_shape = "plus_times",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.5,
  sci_color_alpha = 1,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="violin_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: Length, Width, Weight, and Sex traits dataframe (1st-col: Value, 2nd-col: Traits, 3rd-col: Sex).</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_test_method">test_method</code></td>
<td>
<p>Character: test methods of P value. Default: &quot;t.test&quot;, options: &quot;wilcox.test&quot;, &quot;t.test&quot;, &quot;anova&quot;, &quot;kruskal.test&quot;.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_test_label">test_label</code></td>
<td>
<p>Character: test label of P value. Default: &quot;p.format&quot;, options: &quot;p.signif&quot;, &quot;p.format&quot;. c(0, 0.0001, 0.001, 0.01, 0.05, 1).</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_group_level">group_level</code></td>
<td>
<p>Character: group levels. Default: &quot;Three_Column&quot;, options: &quot;Two_Column&quot;, &quot;Three_Column&quot;.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_violin_orientation">violin_orientation</code></td>
<td>
<p>Character: violin orientation. Default: &quot;vertical&quot;, options: &quot;vertical&quot;, &quot;horizontal&quot;, &quot;reverse&quot;.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_add_element">add_element</code></td>
<td>
<p>Character: add new plot. Default: &quot;boxplot&quot;, options: &quot;none&quot;, &quot;dotplot&quot;, &quot;jitter&quot;, &quot;boxplot&quot;, &quot;point&quot;, &quot;mean&quot;, &quot;mean_se&quot;, &quot;mean_sd&quot;, &quot;mean_ci&quot;, &quot;mean_range&quot;, &quot;median&quot;, &quot;median_iqr&quot;, &quot;median_hilow&quot;, &quot;median_q1q3&quot;, &quot;median_mad&quot;, &quot;median_range&quot;.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_element_alpha">element_alpha</code></td>
<td>
<p>Numeric: element color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_my_shape">my_shape</code></td>
<td>
<p>Character: box scatter shape. Default: &quot;plus_times&quot;, options: &quot;border_square&quot;, &quot;border_circle&quot;, &quot;border_triangle&quot;, &quot;plus&quot;, &quot;times&quot;, &quot;border_diamond&quot;, &quot;border_triangle_down&quot;, &quot;square_times&quot;, &quot;plus_times&quot;, &quot;diamond_plus&quot;, &quot;circle_plus&quot;, &quot;di_triangle&quot;, &quot;square_plus&quot;, &quot;circle_times&quot;,&quot;square_triangle&quot;, &quot;fill_square&quot;, &quot;fill_circle&quot;, &quot;fill_triangle&quot;, &quot;fill_diamond&quot;, &quot;large_circle&quot;, &quot;small_circle&quot;, &quot;fill_border_circle&quot;, &quot;fill_border_square&quot;, &quot;fill_border_diamond&quot;, &quot;fill_border_triangle&quot;.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_sci_fill_color">sci_fill_color</code></td>
<td>
<p>Character: ggsci color pallet. Default: &quot;Sci_AAAS&quot;, options: &quot;Sci_AAAS&quot;, &quot;Sci_NPG&quot;, &quot;Sci_Simpsons&quot;, &quot;Sci_JAMA&quot;, &quot;Sci_GSEA&quot;, &quot;Sci_Lancet&quot;, &quot;Sci_Futurama&quot;, &quot;Sci_JCO&quot;, &quot;Sci_NEJM&quot;, &quot;Sci_IGV&quot;, &quot;Sci_UCSC&quot;, &quot;Sci_D3&quot;, &quot;Sci_Material&quot;.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_sci_fill_alpha">sci_fill_alpha</code></td>
<td>
<p>Numeric: ggsci fill color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_sci_color_alpha">sci_color_alpha</code></td>
<td>
<p>Numeric: ggsci border color alpha. Default: 1.00, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character: legend position. Default: &quot;right&quot;, options: &quot;none&quot;, &quot;left&quot;, &quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_legend_dir">legend_dir</code></td>
<td>
<p>Character: legend direction. Default: &quot;vertical&quot;, options: &quot;horizontal&quot;, &quot;vertical&quot;.</p>
</td></tr>
<tr><td><code id="violin_plot_+3A_ggtheme">ggTheme</code></td>
<td>
<p>Character: ggplot2 themes. Default: &quot;theme_light&quot;, options: &quot;theme_default&quot;, &quot;theme_bw&quot;, &quot;theme_gray&quot;, &quot;theme_light&quot;, &quot;theme_linedraw&quot;, &quot;theme_dark&quot;, &quot;theme_minimal&quot;, &quot;theme_classic&quot;, &quot;theme_void&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: violin plot support two levels and multiple groups with P value.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(traits_sex)
head(traits_sex)

# 3. Default parameters
violin_plot(traits_sex)

# 4. Set test_label = "p.signif",
violin_plot(traits_sex, test_label = "p.signif")

# 5. Set violin_orientation = "horizontal"
violin_plot(traits_sex, violin_orientation = "horizontal")

# 6. Set group_level = "Two_Column"
violin_plot(traits_sex, group_level = "Two_Column")

# 7. Set add_element = "jitter"
violin_plot(traits_sex, add_element = "jitter")

</code></pre>

<hr>
<h2 id='volcano_plot'>Volcano plot for visualizing differentailly expressed genes.</h2><span id='topic+volcano_plot'></span>

<h3>Description</h3>

<p>Volcano plot for visualizing differentailly expressed genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcano_plot(
  data,
  title = "CT-vs-LT12",
  log2fc_cutoff = 1,
  pq_value = "pvalue",
  pq_cutoff = 0.05,
  cutoff_line = "longdash",
  point_shape = "large_circle",
  point_size = 2,
  point_alpha = 0.5,
  color_normal = "#888888",
  color_log2fc = "#008000",
  color_pvalue = "#0088ee",
  color_Log2fc_p = "#ff0000",
  label_size = 3,
  boxed_labels = FALSE,
  draw_connectors = FALSE,
  legend_pos = "right"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="volcano_plot_+3A_data">data</code></td>
<td>
<p>Dataframe: differentially expressed genes (DEGs) stats (1st-col: Genes, 2nd-col: log2FoldChange, 3rd-col: Pvalue, 4th-col: FDR).</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_title">title</code></td>
<td>
<p>Character: title of plot. Default: CT-vs-LT12.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_log2fc_cutoff">log2fc_cutoff</code></td>
<td>
<p>Numeric: log2(FoldChange) cutoff log2(2) = 1. Default: 1.0, min: 0.0, max: null.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_pq_value">pq_value</code></td>
<td>
<p>Character: select pvalue or qvalue. Default: &quot;pvalue&quot;, options: &quot;pvalue&quot;, &quot;padj&quot;.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_pq_cutoff">pq_cutoff</code></td>
<td>
<p>Numeric: pvalue or qvalue cutoff. Default: 0.005, min: 0.000, max: 1.000.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_cutoff_line">cutoff_line</code></td>
<td>
<p>Character: cutoff line type. Default: &quot;longdash&quot;, options: &quot;blank&quot;, &quot;solid&quot;, &quot;dashed&quot;, &quot;dotted&quot;, &quot;dotdash&quot;, &quot;longdash&quot;, &quot;twodash&quot;.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_point_shape">point_shape</code></td>
<td>
<p>Character: point shape. Default: &quot;large_circle&quot;, options: &quot;border_square&quot;, &quot;border_circle&quot;, &quot;border_triangle&quot;, &quot;plus&quot;, &quot;times&quot;, &quot;border_diamond&quot;, &quot;border_triangle_down&quot;, &quot;square_times&quot;, &quot;plus_times&quot;, &quot;diamond_plus&quot;, &quot;circle_plus&quot;, &quot;di_triangle&quot;, &quot;square_plus&quot;, &quot;circle_times&quot;,&quot;square_triangle&quot;, &quot;fill_square&quot;, &quot;fill_circle&quot;, &quot;fill_triangle&quot;, &quot;fill_diamond&quot;, &quot;large_circle&quot;, &quot;small_circle&quot;, &quot;fill_border_circle&quot;, &quot;fill_border_square&quot;, &quot;fill_border_diamond&quot;, &quot;fill_border_triangle&quot;.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_point_size">point_size</code></td>
<td>
<p>Numeric: point size. Default: 1.0, min: 0.0, max: null.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_point_alpha">point_alpha</code></td>
<td>
<p>Numeric: point color alpha. Default: 0.50, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_color_normal">color_normal</code></td>
<td>
<p>Character: normal genes color (color name or hex value). Default: &quot;#888888&quot;.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_color_log2fc">color_log2fc</code></td>
<td>
<p>Character: genes color that log2fc &gt;= log2fc_cutoff. Default: &quot;#008000&quot;.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_color_pvalue">color_pvalue</code></td>
<td>
<p>Character: genes color that pvalue &gt; pq_cutoff. Default: &quot;#0088ee&quot;.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_color_log2fc_p">color_Log2fc_p</code></td>
<td>
<p>Character: genes color that log2fc &gt;= log2fc_cutoff and pvalue &gt; pq_cutoff. Default: &quot;#ff0000&quot;.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric: DEG labels size. Default: 3.0, min: 0.0, max: null.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_boxed_labels">boxed_labels</code></td>
<td>
<p>Logical: add box to every DEG label. Default: FALSE.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_draw_connectors">draw_connectors</code></td>
<td>
<p>Logical: add connector between DEGs and labels. Default: FALSE.</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character: legend position. Default: &quot;right&quot;, options: &quot;right&quot;, &quot;left&quot;, &quot;top&quot;, &quot;bottom&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot: volcano plot for visualizing differentailly expressed genes.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(degs_stats)
head(degs_stats)

# 3. Default parameters
volcano_plot(degs_stats)

# 4. Set color_Log2fc_p = "#ff8800"
volcano_plot(degs_stats, color_Log2fc_p = "#ff8800")

# 5. Set boxed_labels = TRUE
volcano_plot(degs_stats, boxed_labels = TRUE)

</code></pre>

<hr>
<h2 id='weight_sex'>Weight and Sex traits dataframe.</h2><span id='topic+weight_sex'></span>

<h3>Description</h3>

<p>Weight and Sex traits dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(weight_sex)
</code></pre>


<h3>Format</h3>

<p>Dataframe: Weight and Sex traits dataframe (1st-col: Weight, 2nd-col: Sex).
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>References</h3>

<p>https://github.com/BioSciTools/BioSciToolsDatasets/tree/main/QuantileQuantile/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Load example data
data(weight_sex)

# 3. View example data
weight_sex

</code></pre>

<hr>
<h2 id='wgcna_pipeline'>WGCNA analysis pipeline for RNA-Seq.</h2><span id='topic+wgcna_pipeline'></span>

<h3>Description</h3>

<p>WGCNA analysis pipeline for RNA-Seq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wgcna_pipeline(
  sample_gene,
  group_sample,
  R_cutofff = 0.85,
  max_block = 5000,
  min_module = 20,
  network_type = "unsigned",
  merge_cutoff = 0.15,
  cor_type = "pearson",
  na_color = "#cdcdcd",
  xlab_angle = 45,
  text_size = 0.7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wgcna_pipeline_+3A_sample_gene">sample_gene</code></td>
<td>
<p>Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_group_sample">group_sample</code></td>
<td>
<p>Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_r_cutofff">R_cutofff</code></td>
<td>
<p>Numeric: Rsquare cutoff. Default: 0.85, min: 0.00, max: 1.00.</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_max_block">max_block</code></td>
<td>
<p>Numeric: max block size. Default: 5000.</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_min_module">min_module</code></td>
<td>
<p>Numeric: min module gene number. Default: 20.</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_network_type">network_type</code></td>
<td>
<p>Character: network type. Default: &quot;unsigned&quot;, options: &quot;unsigned&quot;, &quot;signed&quot;, &quot;signed hybrid&quot;.</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_merge_cutoff">merge_cutoff</code></td>
<td>
<p>Numeric: merge modules cutoff. Default: 0.15.</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_cor_type">cor_type</code></td>
<td>
<p>Character: correlation type. Default: &quot;pearson&quot;, options: &quot;pearson&quot;, &quot;bicor&quot;.</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_na_color">na_color</code></td>
<td>
<p>Character: NA value color (color name or hex value). Default: &quot;#cdcdcd&quot;.</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_xlab_angle">xlab_angle</code></td>
<td>
<p>Numeric: X axis lable angle. Default: 45, min: 0, max: 360.</p>
</td></tr>
<tr><td><code id="wgcna_pipeline_+3A_text_size">text_size</code></td>
<td>
<p>Numeric: cell text size. Default: 0.7, min: 0, max: NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>WGCNA results in tempdir() directory of current session.
</p>


<h3>Author(s)</h3>

<p>benben-miao
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Library TOmicsVis package
library(TOmicsVis)

# 2. Use example dataset
data(gene_expression)
head(gene_expression)

data(samples_groups)
head(samples_groups)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
