<!DOCTYPE html><html><head><title>Help for package TooManyCellsR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TooManyCellsR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#importResults'><p>Import some 'too-many-cells make-tree' outputs into a data frame.</p></a></li>
<li><a href='#tooManyCells'><p>Execute 'too-many-cells'.</p></a></li>
<li><a href='#tryFunc'><p>Different error for importing data.</p></a></li>
<li><a href='#writeMatrixFiles'><p>Write a Matrix to a folder.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>An R Wrapper for 'TooManyCells'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Gregory W. Schwartz</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gregory W. Schwartz &lt;gsch@pennmedicine.upenn.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An R wrapper for using 'TooManyCells', a command line program for clustering, visualizing, and quantifying cell clade relationships. See <a href="https://gregoryschwartz.github.io/too-many-cells/">https://gregoryschwartz.github.io/too-many-cells/</a> for more details.</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix, imager, ggplot2, cowplot, jsonlite</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>'TooManyCells'
(https://github.com/GregorySchwartz/too-many-cells)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-10-04 20:30:20 UTC; gw</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-10-07 20:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='importResults'>Import some 'too-many-cells make-tree' outputs into a data frame.</h2><span id='topic+importResults'></span>

<h3>Description</h3>

<p>This function will import some of the files resulting from a 'too-many-cells
make-tree' run into R as data frames. Does not include cluster list. Look at
the main tooManyCells function for the cluster list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importResults(dir = "out")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importResults_+3A_dir">dir</code></td>
<td>
<p>The output directory of a 'too-many-cells' run.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of each output. Reads the following files, see
<a href="https://gregoryschwartz.github.io/too-many-cells/">https://gregoryschwartz.github.io/too-many-cells/</a> for more details:
&quot;dendrogram.svg&quot;, &quot;clumpiness.pdf&quot;, &quot;projection.pdf&quot;,
&quot;label_projection.pdf&quot;, &quot;clumpiness.csv&quot;, &quot;cluster_info.csv&quot;,
&quot;node_info.csv&quot;, and &quot;cluster_diversity.csv&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input &lt;- system.file("extdata", "mat.csv", package="TooManyCellsR")
inputLabels &lt;- system.file("extdata", "labels.csv", package="TooManyCellsR")
df = read.csv(input, row.names = 1, header = TRUE)
mat = Matrix::Matrix(as.matrix(df), sparse = TRUE)
labelsDf = read.csv(inputLabels, header = TRUE)
# Here we draw this small toy example with no filter or normalization, and
# decrease the size of the branches and increase the size of the leaf nodes.
# With non-toy real world single cell data, these options should not be
# necessary.
## Not run: 
tooManyCells( mat, labels = labelsDf
            , args = c( "make-tree"
                      , "--no-filter"
                      , "--normalization", "NoneNorm"
                      , "--draw-max-node-size", "40"
                      , "--draw-max-leaf-node-size", "70"
                      )
            )
res = importResults("out")
plot(res$treePlot, axes = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='tooManyCells'>Execute 'too-many-cells'.</h2><span id='topic+tooManyCells'></span>

<h3>Description</h3>

<p>This function will run 'too-many-cells' on a Matrix.
Requires 'TooManyCells' to be installed (follow instructions at
<a href="https://gregoryschwartz.github.io/too-many-cells/">https://gregoryschwartz.github.io/too-many-cells/</a> ).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tooManyCells(mat, args = c("make-tree"), labels = NULL,
  output = "out", prior = NULL, docker = NULL, mounts = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tooManyCells_+3A_mat">mat</code></td>
<td>
<p>The input Matrix with gene row names and
cell barcode column names.</p>
</td></tr>
<tr><td><code id="tooManyCells_+3A_args">args</code></td>
<td>
<p>The arguments to give to the command line program. See
<a href="https://gregoryschwartz.github.io/too-many-cells/">https://gregoryschwartz.github.io/too-many-cells/</a> for more information.
Defaults to &quot;make-tree&quot;.</p>
</td></tr>
<tr><td><code id="tooManyCells_+3A_labels">labels</code></td>
<td>
<p>The input labels data frame with item (cell barcodes) and label
(whatever labels you want to give them, such as tissue of origin, celltype,
etc.) columns. Optional.</p>
</td></tr>
<tr><td><code id="tooManyCells_+3A_output">output</code></td>
<td>
<p>The output folder for the 'too-many-cells' process. Defaults to
&quot;out&quot;.</p>
</td></tr>
<tr><td><code id="tooManyCells_+3A_prior">prior</code></td>
<td>
<p>The location of the tree that was already made (previous
'too-many-cells' output) so quick visual or pruning changes can be made
without remaking the tree (can potentially save hours).</p>
</td></tr>
<tr><td><code id="tooManyCells_+3A_docker">docker</code></td>
<td>
<p>If using 'too-many-cells' with docker, use this argument as the
command to call. For instance, if version 0.2.1.0 was pulled from Docker
Hub, set to &quot;gregoryschwartz/too-many-cells:0.2.1.0&quot;.</p>
</td></tr>
<tr><td><code id="tooManyCells_+3A_mounts">mounts</code></td>
<td>
<p>Additional directories to mount if needed for docker.
The 'prior' argument will automatically mount if specified.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of each output, including the stdout. Reads the
following files, see <a href="https://gregoryschwartz.github.io/too-many-cells/">https://gregoryschwartz.github.io/too-many-cells/</a> for
more details: &quot;dendrogram.svg&quot;, &quot;clumpiness.pdf&quot;, &quot;projection.pdf&quot;,
&quot;label_projection.pdf&quot;, &quot;clumpiness.csv&quot;, &quot;cluster_info.csv&quot;,
&quot;node_info.csv&quot;, and &quot;cluster_diversity.csv&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input &lt;- system.file("extdata", "mat.csv", package="TooManyCellsR")
inputLabels &lt;- system.file("extdata", "labels.csv", package="TooManyCellsR")
df = read.csv(input, row.names = 1, header = TRUE)
mat = Matrix::Matrix(as.matrix(df), sparse = TRUE)
labelsDf = read.csv(inputLabels, header = TRUE)
# Here we draw this small toy example with no filter or normalization, and
# decrease the size of the branches and increase the size of the leaf nodes.
# With non-toy real world single cell data, these options should not be
# necessary.
## Not run: 
res = tooManyCells( mat, labels = labelsDf
                  , args = c( "make-tree"
                            , "--no-filter"
                            , "--normalization", "NoneNorm"
                            , "--draw-max-node-size", "40"
                            , "--draw-max-leaf-node-size", "70"
                            )
                  )
plot(res$treePlot, axes = FALSE)
res$stdout
res$nodeInfo

## End(Not run)
</code></pre>

<hr>
<h2 id='tryFunc'>Different error for importing data.</h2><span id='topic+tryFunc'></span>

<h3>Description</h3>

<p>This function will fail gracefully instead of stopping the program with an
error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tryFunc(f, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tryFunc_+3A_f">f</code></td>
<td>
<p>The function to use.</p>
</td></tr>
<tr><td><code id="tryFunc_+3A_file">file</code></td>
<td>
<p>The input file to be read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The imported data frame or NULL if an error occurred.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input &lt;- system.file("extdata", "mat.csv", package="TooManyCellsR")
fail = tryFunc(read.csv, "fail.csv")
fail
success = tryFunc(read.csv, input)
success
</code></pre>

<hr>
<h2 id='writeMatrixFiles'>Write a Matrix to a folder.</h2><span id='topic+writeMatrixFiles'></span>

<h3>Description</h3>

<p>This function will write a Matrix from the Matrix library to a temporary
directory containing matrix.mtx, genes.tsv, barcodes.tsv, and optionally a
labels.csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeMatrixFiles(mat, labels = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeMatrixFiles_+3A_mat">mat</code></td>
<td>
<p>The input Matrix with gene row names and cell barcode column
names.</p>
</td></tr>
<tr><td><code id="writeMatrixFiles_+3A_labels">labels</code></td>
<td>
<p>The input labels data frame with item (cell barcodes) and label
(whatever labels you want to give them, such as tissue of origin, celltype,
etc.) columns. Optional.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input &lt;- system.file("extdata", "mat.csv", package="TooManyCellsR")
df = read.csv(input, row.names = 1, header = TRUE)
mat = Matrix::Matrix(as.matrix(df), sparse = TRUE)
writeMatrixFiles(mat)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
