<!DOCTYPE html><html><head><title>Help for package TopDom</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TopDom}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#countsPerRegion'><p>Calculates Counts per Region in a TopDomData Object</p></a></li>
<li><a href='#ggCountHeatmap'><p>Produce a Count Heatmap</p></a></li>
<li><a href='#ggDomain'><p>Add a Topological Domain to a Count Heatmap</p></a></li>
<li><a href='#ggDomainLabel'><p>Add a Topological Domain Label to a Count Heatmap</p></a></li>
<li><a href='#legacy'><p>Easy Access to the Original TopDom 0.0.1 and 0.0.2 Implementations</p></a></li>
<li><a href='#overlapScores'><p>Calculates Overlap Scores Between Two Sets of Topological Domains</p></a></li>
<li><a href='#readHiC'><p>Reads Hi-C Contact Data from File</p></a></li>
<li><a href='#stop_if_not'><p>Asserts the Truth of R Expressions</p></a></li>
<li><a href='#subsetByRegion'><p>Subset a TopDomData Object by Region</p></a></li>
<li><a href='#TopDom'><p>Identify Topological Domains from a Hi-C Contact Matrix</p></a></li>
<li><a href='#TopDom-data'><p>Data for the TopDom Package</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.10.1</td>
</tr>
<tr>
<td>Title:</td>
<td>An Efficient and Deterministic Method for Identifying
Topological Domains in Genomes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>matrixStats, grid, ggplot2, reshape2, tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>diffobj (&ge; 0.1.11)</td>
</tr>
<tr>
<td>Description:</td>
<td>The 'TopDom' method identifies topological domains in genomes from Hi-C sequence data (Shin et al., 2016 &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkv1505">doi:10.1093/nar/gkv1505</a>&gt;).  The authors published an implementation of their method as an R script (two different versions; also available in this package).  This package originates from those original 'TopDom' R scripts and provides help pages adopted from the original 'TopDom' PDF documentation.  It also provides a small number of bug fixes to the original code.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>TRUE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/HenrikBengtsson/TopDom">https://github.com/HenrikBengtsson/TopDom</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HenrikBengtsson/TopDom/issues">https://github.com/HenrikBengtsson/TopDom/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-04 15:51:05 UTC; hb</td>
</tr>
<tr>
<td>Author:</td>
<td>Henrik Bengtsson [aut, cre, cph],
  Hanjun Shin [aut, ctr, cph],
  Harris Lazaris [ctr, cph] (PhD Student, NYU),
  Gangqing Hu [ctr, cph] (Staff Scientist, NIH),
  Xianghong Zhou [ctr]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Henrik Bengtsson &lt;henrikb@braju.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-06 07:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='countsPerRegion'>Calculates Counts per Region in a TopDomData Object</h2><span id='topic+countsPerRegion'></span>

<h3>Description</h3>

<p>Calculates Counts per Region in a TopDomData Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countsPerRegion(data, regions)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="countsPerRegion_+3A_data">data</code></td>
<td>
<p>A TopDomData object.</p>
</td></tr>
<tr><td><code id="countsPerRegion_+3A_regions">regions</code></td>
<td>
<p>TopDom regions (a data.frame), e.g. domains.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of length <code>nrow(regions)</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson.
</p>

<hr>
<h2 id='ggCountHeatmap'>Produce a Count Heatmap</h2><span id='topic+ggCountHeatmap'></span>

<h3>Description</h3>

<p>Produce a Count Heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggCountHeatmap(data, transform, colors, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggCountHeatmap_+3A_data">data</code></td>
<td>
<p>A TopDomData object.</p>
</td></tr>
<tr><td><code id="ggCountHeatmap_+3A_transform">transform</code></td>
<td>
<p>A function applied to the counts prior to generating
heatmap colors.</p>
</td></tr>
<tr><td><code id="ggCountHeatmap_+3A_colors">colors</code></td>
<td>
<p>A named list to control to color scale.</p>
</td></tr>
<tr><td><code id="ggCountHeatmap_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="ggplot2.html#topic+ggplot">ggplot2::ggplot</a> object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson.
</p>


<h3>See Also</h3>

<p>See <a href="#topic+TopDom">TopDom</a> for an example.
</p>

<hr>
<h2 id='ggDomain'>Add a Topological Domain to a Count Heatmap</h2><span id='topic+ggDomain'></span>

<h3>Description</h3>

<p>Add a Topological Domain to a Count Heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggDomain(td, dx = NULL, delta = 0.04, vline = 0, size = 2, color = "#666666")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggDomain_+3A_td">td</code></td>
<td>
<p>A single-row data.frame.</p>
</td></tr>
<tr><td><code id="ggDomain_+3A_dx">dx</code>, <code id="ggDomain_+3A_delta">delta</code>, <code id="ggDomain_+3A_vline">vline</code></td>
<td>
<p>Absolute distance to heatmap.
If <code>dx = NULL</code> (default), then <code>dx = delta * w + vline</code> where <code>w</code> is
the width of the domain.</p>
</td></tr>
<tr><td><code id="ggDomain_+3A_size">size</code>, <code id="ggDomain_+3A_color">color</code></td>
<td>
<p>The thickness and color of the domain line.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="ggplot2.html#topic+geom_segment">ggplot2::geom_segment</a> object to be added to the count heatmap.
</p>

<hr>
<h2 id='ggDomainLabel'>Add a Topological Domain Label to a Count Heatmap</h2><span id='topic+ggDomainLabel'></span>

<h3>Description</h3>

<p>Add a Topological Domain Label to a Count Heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggDomainLabel(
  td,
  fmt = "%s: %.2f - %.2f Mbp",
  rot = 45,
  dx = 0,
  vjust = 2.5,
  cex = 1.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggDomainLabel_+3A_td">td</code></td>
<td>
<p>A single-row data.frame.</p>
</td></tr>
<tr><td><code id="ggDomainLabel_+3A_fmt">fmt</code></td>
<td>
<p>The <a href="base.html#topic+sprintf">base::sprintf</a>-format string taking (chromosome, start, stop) as
(string, numeric, numeric) input.</p>
</td></tr>
<tr><td><code id="ggDomainLabel_+3A_rot">rot</code></td>
<td>
<p>The amount of rotation in [0,360] of label.</p>
</td></tr>
<tr><td><code id="ggDomainLabel_+3A_dx">dx</code>, <code id="ggDomainLabel_+3A_vjust">vjust</code></td>
<td>
<p>The vertical adjustment of the label (relative to rotation)</p>
</td></tr>
<tr><td><code id="ggDomainLabel_+3A_cex">cex</code></td>
<td>
<p>The scale factor of the label.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="ggplot2.html#topic+ggproto">ggplot2::ggproto</a> object to be added to the count heatmap.
</p>

<hr>
<h2 id='legacy'>Easy Access to the Original TopDom 0.0.1 and 0.0.2 Implementations</h2><span id='topic+legacy'></span>

<h3>Description</h3>

<p>Easy Access to the Original TopDom 0.0.1 and 0.0.2 Implementations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legacy(version = c("0.0.1", "0.0.2"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="legacy_+3A_version">version</code></td>
<td>
<p>A version string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An environment containing the legacy TopDom API.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>TopDom::legacy("0.0.2")$TopDom
TopDom::legacy("0.0.1")$Detect.Local.Extreme

</code></pre>

<hr>
<h2 id='overlapScores'>Calculates Overlap Scores Between Two Sets of Topological Domains</h2><span id='topic+overlapScores'></span>

<h3>Description</h3>

<p>Calculates Overlap Scores Between Two Sets of Topological Domains
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overlapScores(a, reference, debug = getOption("TopDom.debug", FALSE))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overlapScores_+3A_a">a</code>, <code id="overlapScores_+3A_reference">reference</code></td>
<td>
<p>Topological domain (TD) set <code class="reqn">A</code> and TD reference
set <code class="reqn">R</code> both in a format as returned by <code><a href="#topic+TopDom">TopDom()</a></code>.</p>
</td></tr>
<tr><td><code id="overlapScores_+3A_debug">debug</code></td>
<td>
<p>If <code>TRUE</code>, debug output is produced.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <em>overlap score</em>, <code class="reqn">overlap(A', r_i)</code>, represents how well a
<em>consecutive</em> subset <code class="reqn">A'</code> of topological domains (TDs) in <code class="reqn">A</code>
overlap with topological domain <code class="reqn">r_i</code> in reference set <code class="reqn">R</code>.
For each reference TD <code class="reqn">r_i</code>, the <em>best match</em> <code class="reqn">A'_max</code> is
identified, that is, the <code class="reqn">A'</code> subset that maximize
<code class="reqn">overlap(A', r_i)</code>.
For exact definitions, see Page 8 in Shin et al. (2016).
</p>
<p>Note that the overlap score is an asymmetric score, which means that
<code>overlapScores(a, b) != overlapScores(b, a)</code>.
</p>


<h3>Value</h3>

<p>Returns a named list of class <code>TopDomOverlapScores</code>, where the names
correspond to the chromosomes in domain reference set <code class="reqn">R</code>.
Each of these chromosome elements contains a data.frame with fields:
</p>

<ul>
<li> <p><code>chromosome</code>  - <code class="reqn">D_{R,c}</code> character strings
</p>
</li>
<li> <p><code>best_score</code>  - <code class="reqn">D_{R,c}</code> numerics in <code class="reqn">[0,1]</code>
</p>
</li>
<li> <p><code>best_length</code> - <code class="reqn">D_{R,c}</code> positive integers
</p>
</li>
<li> <p><code>best_set</code>    - list of <code class="reqn">D_{R,c}</code> index vectors
</p>
</li></ul>

<p>where <code class="reqn">D_{R,c}</code> is the number of TDs in reference set <code class="reqn">R</code> on
chromosome <code class="reqn">c</code>.  If a TD in reference <code class="reqn">R</code> is not a <code>"domain"</code>,
then the corresponding <code>best_score</code> and <code>best_length</code> values are
<code>NA_real_</code> and <code>NA_integer_</code>, respectively, while <code>best_set</code> is an empty
list.
</p>


<h3>Warning - This might differ not be the correct implementation</h3>

<p>The original TopDom scripts do not provide an implementation for
calculating overlap scores.  Instead, the implementation of
<code>TopDom::overlapScores()</code> is based on the textual description of
overlap scores provided in Shin et al. (2016).  It is not known if this
is the exact same algorithm and implementation as the authors of the
TopDom article used.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson - based on the description in Shin et al. (2016).
</p>


<h3>References</h3>


<ul>
<li><p> Shin et al.,
TopDom: an efficient and deterministic method for identifying
topological domains in genomes,
<em>Nucleic Acids Research</em>, 44(7): e70, April 2016.
doi: 10.1093/nar/gkv1505,
PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4838359/">PMC4838359</a>,
PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/26704975/">26704975</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+TopDom">TopDom</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(tibble)
path &lt;- system.file("exdata", package = "TopDom", mustWork = TRUE)

## Original count data (on a subset of the bins to speed up example)
chr &lt;- "chr19"
pathname &lt;- file.path(path, sprintf("nij.%s.gz", chr))
data &lt;- readHiC(pathname, chr = chr, binSize = 40e3, bins = 1:500)
print(data)

## Find topological domains using TopDom method for two window sizes
tds_5 &lt;- TopDom(data, window.size = 5L)
tds_6 &lt;- TopDom(data, window.size = 6L)

## Overlap scores (in both directions)
overlap_56 &lt;- overlapScores(tds_6, reference = tds_5)
print(overlap_56)
print(as_tibble(overlap_56))

overlap_65 &lt;- overlapScores(tds_5, reference = tds_6)
print(overlap_65)
print(as_tibble(overlap_65))

</code></pre>

<hr>
<h2 id='readHiC'>Reads Hi-C Contact Data from File</h2><span id='topic+readHiC'></span>

<h3>Description</h3>

<p>Reads Hi-C Contact Data from File
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readHiC(
  file,
  chr = NULL,
  binSize = NULL,
  ...,
  debug = getOption("TopDom.debug", FALSE)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readHiC_+3A_file">file</code></td>
<td>
<p>The pathname of a normalize Hi-C contact matrix file
stored as a whitespace-delimited file.  See below for details.
Also a gzip-compressed file can be used.</p>
</td></tr>
<tr><td><code id="readHiC_+3A_chr">chr</code>, <code id="readHiC_+3A_binsize">binSize</code></td>
<td>
<p>If the file contains a count matrix without bin
annotation, the latter is created from these parameters.</p>
</td></tr>
<tr><td><code id="readHiC_+3A_debug">debug</code></td>
<td>
<p>If <code>TRUE</code>, debug output is produced.</p>
</td></tr>
<tr><td><code id="readHiC_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="utils.html#topic+read.table">utils::read.table()</a></code> as-is.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements <code>bins</code> (an N-by-4 data.frame) and
<code>counts</code> (N-by-N matrix).
</p>


<h3>Format of HiC contact-matrix file</h3>

<p>The contact-matrix file should be a whitespace-delimited text file with
neither row names nor column names.  The content should be a N-by-(3+N)
table where the first three columns correspond to <code>chr</code> (string),
<code>from.coord</code> (integer position), and <code>to.coord</code> (integer position).
These column defines the genomic location of the N Hi-C bins (in order).
The last N columns should contain normalized contact counts (float) such
that element (r,3+c) in this table corresponds to count (r,c) in the
normalized contact matrix.
</p>
<p>If an N-by-(4+N) table, then the first column is assumed to contain an
<code>id</code> (integer), and everything else as above.
</p>
<p>Example:
</p>
<pre>
  chr10       0   40000  0 0 0 0 ...
  chr10   40000   80000  0 0 0 0 ...
  chr10   80000  120000  0 0 0 0 ...
  chr10  120000  160000  0 0 0 0 ...
  ...
</pre>


<h3>See Also</h3>

<p><a href="#topic+TopDom">TopDom</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("exdata", package = "TopDom", mustWork = TRUE)

## Original count data
chr &lt;- "chr19"
pathname &lt;- file.path(path, sprintf("nij.%s.gz", chr))
data &lt;- readHiC(pathname, chr = chr, binSize = 40e3)
print(data)
str(data)
</code></pre>

<hr>
<h2 id='stop_if_not'>Asserts the Truth of R Expressions</h2><span id='topic+stop_if_not'></span>

<h3>Description</h3>

<p>Asserts the Truth of R Expressions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stop_if_not(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stop_if_not_+3A_...">...</code></td>
<td>
<p>Zero or more <span class="rlang"><b>R</b></span> expressions to be asserted to be TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A bare bone, faster version of <a href="base.html#topic+stopifnot">base::stopifnot</a>.
</p>


<h3>Value</h3>

<p>Nothing.
</p>

<hr>
<h2 id='subsetByRegion'>Subset a TopDomData Object by Region</h2><span id='topic+subsetByRegion'></span>

<h3>Description</h3>

<p>Subset a TopDomData Object by Region
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetByRegion(data, region, margin = 1/2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetByRegion_+3A_data">data</code></td>
<td>
<p>A TopDomData object.</p>
</td></tr>
<tr><td><code id="subsetByRegion_+3A_region">region</code></td>
<td>
<p>A TopDom domain (a data.frame).</p>
</td></tr>
<tr><td><code id="subsetByRegion_+3A_margin">margin</code></td>
<td>
<p>An non-negative numeric specifying the additional margin
extracted around the domain.
If <code>margin &lt; 1</code>, then the size of the margin is relative
to the size of the domain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A TopDomData object.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson.
</p>

<hr>
<h2 id='TopDom'>Identify Topological Domains from a Hi-C Contact Matrix</h2><span id='topic+TopDom'></span>

<h3>Description</h3>

<p>Identify Topological Domains from a Hi-C Contact Matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TopDom(
  data,
  window.size,
  outFile = NULL,
  statFilter = TRUE,
  ...,
  debug = getOption("TopDom.debug", FALSE)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TopDom_+3A_data">data</code></td>
<td>
<p>A TopDomData object, or the pathname to a normalized
Hi-C contact matrix file as read by <code><a href="#topic+readHiC">readHiC()</a></code>, that specify N bins.</p>
</td></tr>
<tr><td><code id="TopDom_+3A_window.size">window.size</code></td>
<td>
<p>The number of bins to extend (as a non-negative integer).
Recommended range is in 5, ..., 20.</p>
</td></tr>
<tr><td><code id="TopDom_+3A_outfile">outFile</code></td>
<td>
<p>(optional) The filename without extension of the three
result files optionally produced. See details below.</p>
</td></tr>
<tr><td><code id="TopDom_+3A_statfilter">statFilter</code></td>
<td>
<p>(logical) Specifies whether non-significant
topological-domain boundaries should be dropped or not.</p>
</td></tr>
<tr><td><code id="TopDom_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+readHiC">readHiC()</a></code>.</p>
</td></tr>
<tr><td><code id="TopDom_+3A_debug">debug</code></td>
<td>
<p>If <code>TRUE</code>, debug output is produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of class <code>TopDom</code> with data.frame elements
<code>binSignal</code>, <code>domain</code>, and <code>bed</code>.
</p>

<ul>
<li><p> The <code>binSignal</code> data frame (N-by-7) holds mean contact frequency,
local extreme, and p-value for every bin. The first four columns
represent basic bin information given by matrix file, such as
bin id (<code>id</code>), chromosome(<code>chr</code>), start coordinate (<code>from.coord</code>),
and end coordinate (<code>to.coord</code>) for each bin.
The last three columns (<code>local.ext</code>, <code>mean.cf</code>, and <code>p-value</code>) represent
computed values by the TopDom algorithm.
The columns are:
</p>

<ul>
<li> <p><code>id</code>: Bin ID
</p>
</li>
<li> <p><code>chr</code>: Chromosome
</p>
</li>
<li> <p><code>from.coord</code>: Start coordinate of bin
</p>
</li>
<li> <p><code>to.coord</code>: End coordinate of bin
</p>
</li>
<li> <p><code>local.ext</code>:
</p>

<ul>
<li> <p><code>-1</code>: Local minima.
</p>
</li>
<li> <p><code>-0.5</code>: Gap region.
</p>
</li>
<li> <p><code>0</code>: General bin.
</p>
</li>
<li> <p><code>1</code>: Local maxima.
</p>
</li></ul>

</li>
<li> <p><code>mean.cf</code>: Average of contact frequencies between lower and upper
regions for bin <em>i = 1,2,...,N</em>.
</p>
</li>
<li> <p><code>p-value</code>: Computed p-value by Wilcox rank sum test.
See Shin et al. (2016) for more details.
</p>
</li></ul>

</li>
<li><p> The <code>domain</code> data frame (D-by-7):
Every bin is categorized by basic building block, such as gap, domain,
or boundary.
Each row indicates a basic building block.
The first five columns include the basic information about the block,
'tag' column indicates the class of the building block.
</p>

<ul>
<li> <p><code>id</code>: Identifier of block
</p>
</li>
<li> <p><code>chr</code>: Chromosome
</p>
</li>
<li> <p><code>from.id</code>: Start bin index of the block
</p>
</li>
<li> <p><code>from.coord</code>: Start coordinate of the block
</p>
</li>
<li> <p><code>to.id</code>: End bin index of the block
</p>
</li>
<li> <p><code>to.coord</code>: End coordinate of the block
</p>
</li>
<li> <p><code>tag</code>: Categorized name of the block. Three possible blocks exists:
</p>

<ul>
<li> <p><code>gap</code>
</p>
</li>
<li> <p><code>domain</code>
</p>
</li>
<li> <p><code>boundary</code>
</p>
</li></ul>

</li>
<li> <p><code>size</code>: size of the block
</p>
</li></ul>

</li>
<li><p> The <code>bed</code> data frame (D-by-4) is a representation of the <code>domain</code>
data frame in the
<a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED file format</a>.
It has four columns:
</p>

<ul>
<li> <p><code>chrom</code>: The name of the chromosome.
</p>
</li>
<li> <p><code>chromStart</code>: The starting position of the feature in the chromosome.
The first base in a chromosome is numbered 0.
</p>
</li>
<li> <p><code>chromEnd</code>: The ending position of the feature in the chromosome.
The <code>chromEnd</code> base is <em>not</em> included in the feature. For example,
the first 100 bases of a chromosome are defined as <code>chromStart=0</code>,
<code>chromEnd=100</code>, and span the bases numbered 0-99.
</p>
</li>
<li> <p><code>name</code>: Defines the name of the BED line. This label is displayed to
the left of the BED line in the
<a href="https://genome.ucsc.edu/cgi-bin/hgGateway">UCSC Genome Browser</a>
window when the track is open to full display mode or directly to
the left of the item in pack mode.
</p>
</li></ul>

</li></ul>

<p>If argument <code>outFile</code> is non-<code>NULL</code>, then the three elements (<code>binSignal</code>,
<code>domain</code>, and <code>bed</code>) returned are also written to tab-delimited files
with file names &lsquo;<span class="file">&lt;outFile&gt;.binSignal</span>&rsquo;, &lsquo;<span class="file">&lt;outFile&gt;.domain</span>&rsquo;, and
&lsquo;<span class="file">&lt;outFile&gt;.bed</span>&rsquo;, respectively.  None of the files have row names,
and all but the BED file have column names.
</p>


<h3>Windows size</h3>

<p>The <code>window.size</code> parameter is by design the only tuning parameter in the
TopDom method and affects the amount of smoothing applied when calculating
the TopDom bin signals.  The binning window extends symmetrically downstream
and upstream from the bin such that the bin signal is the average
<code>window.size^2</code> contact frequencies.
For details, see Equation (1) and Figure 1 in Shin et al. (2016).
Typically, the number of identified TDs decreases while their average
lengths increase as this window-size parameter increases (Figure 2).
The default is <code>window.size = 5</code> (bins), which is motivated as:
&quot;Considering the previously reported minimum TD size (approx. 200 kb)
(Dixon et al., 2012) and our bin size of 40 kb, <em>w</em>[indow.size] = 5 is a
reasonable setting&quot; (Shin et al., 2016).
</p>


<h3>Author(s)</h3>

<p>Hanjun Shin, Harris Lazaris, and Gangqing Hu.
<span class="rlang"><b>R</b></span> package, help, and code refactoring by Henrik Bengtsson.
</p>


<h3>References</h3>


<ul>
<li><p> Shin et al.,
TopDom: an efficient and deterministic method for identifying
topological domains in genomes,
<em>Nucleic Acids Research</em>, 44(7): e70, April 2016.
DOI: 10.1093/nar/gkv1505,
PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4838359/">PMC4838359</a>,
PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/26704975/">26704975</a>
</p>
</li>
<li><p> Shin et al., <span class="rlang"><b>R</b></span> script &lsquo;<span class="file">TopDom_v0.0.2.R</span>&rsquo;, 2017 (originally from
<code>http://zhoulab.usc.edu/TopDom/</code>;
later available on <a href="https://github.com/jasminezhoulab/TopDom">https://github.com/jasminezhoulab/TopDom</a> via
<a href="https://zhoulab.dgsom.ucla.edu/pages/software">https://zhoulab.dgsom.ucla.edu/pages/software</a>)
</p>
</li>
<li><p> Shin et al., TopDom Manual, 2016-07-08 (original from
<code>http://zhoulab.usc.edu/TopDom/TopDom%20Manual_v0.0.2.pdf</code>;
later available on <a href="https://github.com/jasminezhoulab/TopDom">https://github.com/jasminezhoulab/TopDom</a> via
<a href="https://zhoulab.dgsom.ucla.edu/pages/software">https://zhoulab.dgsom.ucla.edu/pages/software</a>)
</p>
</li>
<li><p> Hanjun Shin, Understanding the 3D genome organization in topological
domain level, Doctor of Philosophy Dissertation,
University of Southern California, March 2017,
<a href="http://digitallibrary.usc.edu/cdm/ref/collection/p15799coll40/id/347735">http://digitallibrary.usc.edu/cdm/ref/collection/p15799coll40/id/347735</a>
</p>
</li>
<li><p> Dixon JR, Selvaraj S, Yue F, Kim A, et al. Topological domains in
mammalian genomes identified by analysis of chromatin interactions.
<em>Nature</em>; 485(7398):376-80, April 2012.
DOI: 10.1038/nature11082,
PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3356448/">PMC3356448</a>,
PMID: 22495300.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("exdata", package = "TopDom", mustWork = TRUE)

## Original count data (on a subset of the bins to speed up example)
chr &lt;- "chr19"
pathname &lt;- file.path(path, sprintf("nij.%s.gz", chr))
data &lt;- readHiC(pathname, chr = chr, binSize = 40e3, bins = 1:500)
print(data)  ## a TopDomData object

## Find topological domains using the TopDom method
fit &lt;- TopDom(data, window.size = 5L)
print(fit)  ## a TopDom object

## Display the largest domain
td &lt;- subset(subset(fit$domain, tag == "domain"), size == max(size))
print(td) ## a data.frame

## Subset TopDomData object
data_s &lt;- subsetByRegion(data, region = td, margin = 0.9999)
print(data_s)  ## a TopDomData object

vp &lt;- grid::viewport(angle = -45, width = 0.7, y = 0.3)
gg &lt;- ggCountHeatmap(data_s)
gg &lt;- gg + ggDomain(td, color = "#cccc00") + ggDomainLabel(td)
print(gg, newpage = TRUE, vp = vp)

gg &lt;- ggCountHeatmap(data_s, colors = list(mid = "white", high = "black"))
gg_td &lt;- ggDomain(td, delta = 0.08)
dx &lt;- attr(gg_td, "gg_params")$dx
gg &lt;- gg + gg_td + ggDomainLabel(td, vjust = 2.5)
print(gg, newpage = TRUE, vp = vp)

## Subset TopDom object
fit_s &lt;- subsetByRegion(fit, region = td, margin = 0.9999)
print(fit_s)  ## a TopDom object
for (kk in seq_len(nrow(fit_s$domain))) {
  gg &lt;- gg + ggDomain(fit_s$domain[kk, ], dx = dx * (4 + kk %% 2), color = "red", size = 1)
}

print(gg, newpage = TRUE, vp = vp)


gg &lt;- ggCountHeatmap(data_s)
gg_td &lt;- ggDomain(td, delta = 0.08)
dx &lt;- attr(gg_td, "gg_params")$dx
gg &lt;- gg + gg_td + ggDomainLabel(td, vjust = 2.5)
fit_s &lt;- subsetByRegion(fit, region = td, margin = 0.9999)
for (kk in seq_len(nrow(fit_s$domain))) {
  gg &lt;- gg + ggDomain(fit_s$domain[kk, ], dx = dx * (4 + kk %% 2), color = "blue", size = 1)
}

print(gg, newpage = TRUE, vp = vp)
</code></pre>

<hr>
<h2 id='TopDom-data'>Data for the TopDom Package</h2><span id='topic+TopDom-data'></span>

<h3>Description</h3>

<p>The &lsquo;<span class="file">exdata/</span>&rsquo; folder of this package provides a example data set
used in examples.  The data are also used to validate the <span class="pkg">TopDom</span>
implementation toward the original TopDom scripts.
</p>


<h3>Origin</h3>

<p>The data herein contain a tiny subset of the HiC and TopDom data used
in the TopDom study (Shin et al., 2016).
More precisely, it contains:
</p>

<ol>
<li><p> A TopDom file &lsquo;<span class="file">mESC_5w_chr19.nij.HindIII.comb.40kb.domain</span>&rsquo;, which
is part of the &lsquo;<span class="file">mESC_5w_domain.zip</span>&rsquo; file
(5,504 bytes; md5 ffb19996f681a4d35d5c9944f2c44343) from the
Supplementary Materials of Shin et al. (2016).
These data were downloaded from the
TopDom website (http://zhoulab.usc.edu/TopDom/ - now defunct).
</p>
</li>
<li><p> A normalized HiC-count matrix file &lsquo;<span class="file">nij.chr19.gz</span>&rsquo;, where the
non-compressed version is part of the &lsquo;<span class="file">mESC.norm.tar.gz</span>&rsquo; file
(1,305,763,679 bytes; md5 2e79d0f57463b5b7c4bf86b187086d3c) originally
downloaded from the
<a href="http://chromosome.sdsc.edu/mouse/hi-c/download.html">UCSD Ren Lab</a>.
It is a tab-delimited file containing a 3250-by-3250 numeric matrix
non-negative decimal values. The underlying HiC sequence data is
available from
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE35156">GSE35156</a>
on GEO and was published part of Dixon, et al. (2012).
</p>
</li></ol>



<h3>References</h3>


<ol>
<li><p> Dixon JR, Selvaraj S, Yue F, Kim A, et al. Topological domains in
mammalian genomes identified by analysis of chromatin interactions.
Nature 2012 Apr 11; 485(7398):376-80,
doi: 10.1038/nature11082,
PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3356448/">PMC3356448</a>,
PMID: 22495300.
</p>
</li>
<li><p> Shin, et al., TopDom: an efficient and deterministic method for
identifying topological domains in genomes,
Nucleic Acids Res. 2016 Apr 20; 44(7): e70., 2016.
doi: 10.1093/nar/gkv1505,
PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4838359/">PMC4838359</a>,
PMID: 26704975.
</p>
</li></ol>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
