<!DOCTYPE html><html><head><title>Help for package topologyGSA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {topologyGSA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clique.mean.test'><p>Comparison of all the cliques of a pathway.</p></a></li>
<li><a href='#clique.var.test'><p>Testing the concentration matrices of all cliques of a pathway.</p></a></li>
<li><a href='#dag_bcell'><p>DAG representation of the B cell receptor signaling pathway</p></a></li>
<li><a href='#pathway.mean.test'><p>Testing the means of a pathway.</p></a></li>
<li><a href='#pathway.var.test'><p>Testing the concentration matrices of a pathway.</p></a></li>
<li><a href='#print'><p>Print results of statistical tests.</p></a></li>
<li><a href='#render.significant.cliques'><p>Graphical display of the significant cliques</p></a></li>
<li><a href='#y1'><p>Gene expression signatures in acute lymphocytic leukemia for BCR signaling pathway</p></a></li>
<li><a href='#y2'><p>Gene expression signatures in acute lymphocytic leukemia for BCR signaling pathway</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.5.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-25</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Set Analysis Exploiting Pathway Topology</td>
</tr>
<tr>
<td>Author:</td>
<td>Sofia Massa, Gabriele Sales</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gabriele Sales &lt;gabriele.sales@unipd.it&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Using Gaussian graphical models we propose a novel approach to
             perform pathway analysis using gene expression. Given the
             structure of a graph (a pathway) we introduce two statistical
             tests to compare the mean and the concentration matrices between
             two groups. Specifically, these tests can be performed on the
             graph and on its connected components (cliques). The package is
             based on the method described in Massa M.S., Chiogna M., Romualdi
             C. (2010) &lt;<a href="https://doi.org/10.1186%2F1752-0509-4-121">doi:10.1186/1752-0509-4-121</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>graph, gRbase (&ge; 2.0), qpgraph, fields, methods, igraph</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Rgraphviz</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-25 17:10:46 UTC; sales</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-25 18:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='clique.mean.test'>Comparison of all the cliques of a pathway.</h2><span id='topic+clique.mean.test'></span>

<h3>Description</h3>

<p>Test the equality of the means in two experimental conditions for all the cliques of a pathway, conditioning on the results of the test on the concentration matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clique.mean.test(y1,y2,dag,alpha,perm.num=1000,paired=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clique.mean.test_+3A_y1">y1</code></td>
<td>
<p>a matrix  with n1 individuals (rows) in the first experimental condition and p genes (columns). </p>
</td></tr>
<tr><td><code id="clique.mean.test_+3A_y2">y2</code></td>
<td>
<p>a matrix  with n2 individuals (rows) in the second experimental condition and p genes (columns).
The genes in the two experimental conditions must be the same.</p>
</td></tr>
<tr><td><code id="clique.mean.test_+3A_dag">dag</code></td>
<td>
<p> graphNEL object, a directed acyclic graph (DAG) corresponding to the pathway of interest. See package <code>gRbase</code> for more details.</p>
</td></tr>
<tr><td><code id="clique.mean.test_+3A_alpha">alpha</code></td>
<td>
<p> significance level of the test.</p>
</td></tr>
<tr><td><code id="clique.mean.test_+3A_perm.num">perm.num</code></td>
<td>
<p>number of permutation used by the test.</p>
</td></tr>
<tr><td><code id="clique.mean.test_+3A_paired">paired</code></td>
<td>
<p>logical flag. If <code>TRUE</code> statistical tests for paired samples is calculated and the test on the variances is not performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function tests  the equality of the means of each clique of a
pathway in two experimental conditions, conditioning on the results of
<code>clique.var.test</code>.
The graph of a pathway is first converted into a DAG, then the moral
graph is obtained and if the latter graph is decomposable then the
test is performed on all its cliques. If the moral graph is not
decomposable, its triangulated version is obtained and then the test
is performed on all its cliques.
</p>
<p>The expression data may contain some genes differing
from those in the pathway: in such case the function automatically takes
the intersection between the two gene sets.
</p>
<p>This function requires <code>gRBase</code> and <code>qpgraph</code> packages.
</p>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>
<p>p-value of the test on the mean for each clique.</p>
</td></tr>
<tr><td><code>p.value.var</code></td>
<td>
<p>p-value of the test on the variance for each clique. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>cliques</code></td>
<td>
<p>the list of the corresponding cliques.</p>
</td></tr>
<tr><td><code>var.equal</code></td>
<td>
<p>logical, a vector with a TRUE for a significant clique and a FALSE otherwise. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>t.value</code></td>
<td>
<p>the observed values of the statistic to test the mean
for each clique.</p>
</td></tr>
<tr><td><code>lambda.value</code></td>
<td>
<p>the observed values of the statistic to test the
variance for each clique. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>graph</code></td>
<td>
<p>the triangularized and moralized graph.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>M. Sofia Massa, Gabriele Sales</p>


<h3>References</h3>

<p>Massa, M.S., Chiogna, M., Romualdi, C. (2010). Gene set analysis exploiting the topology of a pathway. BMC Systems Biology, 4:121
<a href="https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121">https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pathway.var.test">pathway.var.test</a></code>,<code><a href="#topic+pathway.mean.test">pathway.mean.test</a></code>,
<code><a href="#topic+clique.var.test">clique.var.test</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(examples)

clique.mean.test(y1, y2, dag_bcell, 0.05)
</code></pre>

<hr>
<h2 id='clique.var.test'>Testing the concentration matrices of all cliques of a pathway.</h2><span id='topic+clique.var.test'></span>

<h3>Description</h3>

<p>Test the equality of concentration matrices in two experimental conditions for all cliques of a pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clique.var.test(y1,y2,dag,alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clique.var.test_+3A_y1">y1</code></td>
<td>
<p>a matrix  with n1 individuals (rows) in the first experimental condition and p genes (columns).</p>
</td></tr>
<tr><td><code id="clique.var.test_+3A_y2">y2</code></td>
<td>
<p>a matrix  with n2 individuals (rows) in the second experimental condition and p genes (columns).
The genes in the two experimental conditions must be the same.</p>
</td></tr>
<tr><td><code id="clique.var.test_+3A_dag">dag</code></td>
<td>
<p>graphNEL object, a directed acyclic graph (DAG) corresponding to the pathway of interest. See package <code>gRbase</code> for more details.</p>
</td></tr>
<tr><td><code id="clique.var.test_+3A_alpha">alpha</code></td>
<td>
<p>significance level of the test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function tests  the equality of the concentration matrices of each
clique of a pathway in two experimental conditions. The graph of a
pathway is first converted into a DAG, then the moral graph is obtained
and if the latter graph is decomposable then the test is performed on
all its cliques. If the moral graph is not decomposable, its
triangulated version is obtained and then the test is performed on all
its cliques.
</p>
<p>The expression data may contain some genes differing
from those in the pathway: in such case the function automatically takes
the intersection between the two gene sets.
</p>
<p>This function requires <code>gRBase</code> and <code>qpgraph</code> packages.
</p>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>
<p>the observed value of the significance level.</p>
</td></tr>
<tr><td><code>cliques</code></td>
<td>
<p>the list of cliques where the test is performed.</p>
</td></tr>
<tr><td><code>var.equal</code></td>
<td>
<p>logical, a vector with a TRUE for a significant clique and
a FALSE otherwise.</p>
</td></tr>
<tr><td><code>lambda.var</code></td>
<td>
<p>the observed value of the statistic.</p>
</td></tr>
<tr><td><code>graph</code></td>
<td>
<p>the triangulated and moral graphs.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>M. Sofia Massa, Gabriele Sales</p>


<h3>References</h3>

<p>Massa, M.S., Chiogna, M., Romualdi, C. (2010). Gene set analysis exploiting the topology of a pathway. BMC Systems Biology, 4:121
<a href="https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121">https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pathway.var.test">pathway.var.test</a></code>, <code><a href="#topic+pathway.mean.test">pathway.mean.test</a></code>,
<code><a href="#topic+clique.mean.test">clique.mean.test</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(examples)

clique.var.test(y1, y2, dag_bcell, 0.05)
</code></pre>

<hr>
<h2 id='dag_bcell'>DAG representation of the B cell receptor signaling pathway </h2><span id='topic+dag_bcell'></span>

<h3>Description</h3>

<p>The B cell receptor signaling pathway (BCR) converted to a DAG. In case of nodes with multiple genes, the most differentially expressed
between groups of patients defined in y1 (acute lymphocytic leukemia patients
with  BCR/ABL gene rearrangement) and y2 (acute lymphocytic leukemia patients
with  BCR/ABL gene rearrangement) matrices.
</p>
<p>Chiaretti S., Li X., Gentleman R., Vitale A., Wang K.S., Mandelli F., Fo R., Ritz J. (2005) Gene expression
profiles of B-lineage adult acute lymphocytic leukemia reveal genetic patterns that identify lineage derivation
and distinct mechanisms of transformation. Clin Cancer Res, 11:7209-7219.
<a href="https://aacrjournals.org/clincancerres/article/11/20/7209/188537/Gene-Expression-Profiles-of-B-lineage-Adult-Acute">https://aacrjournals.org/clincancerres/article/11/20/7209/188537/Gene-Expression-Profiles-of-B-lineage-Adult-Acute</a>
</p>
<p>Massa, M.S., Chiogna, M., Romualdi, C. (2010). Gene set analysis exploiting the topology of a pathway. BMC Systems Biology, 4:121
<a href="https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121">https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121</a>
</p>

<hr>
<h2 id='pathway.mean.test'>Testing the means of a pathway.</h2><span id='topic+pathway.mean.test'></span>

<h3>Description</h3>

<p>Test the equality of means in two experimental conditions for a pathway,
conditioning on the result of the test on the concentration matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway.mean.test(y1,y2,dag,alpha,perm.num=10000,variance=TRUE,paired=FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway.mean.test_+3A_y1">y1</code></td>
<td>
<p>a matrix with n1 individuals (rows) in the first experimental condition and p genes (columns).</p>
</td></tr>
<tr><td><code id="pathway.mean.test_+3A_y2">y2</code></td>
<td>
<p>a matrix with n2 individuals (rows) in the second experimental condition and p genes (columns).
The genes in the two experimental conditions must be the same.</p>
</td></tr>
<tr><td><code id="pathway.mean.test_+3A_dag">dag</code></td>
<td>
<p>graphNEL object, a directed acyclic graph (DAG) corresponding to the pathway of interest. See package <code>gRbase</code> for more details.</p>
</td></tr>
<tr><td><code id="pathway.mean.test_+3A_alpha">alpha</code></td>
<td>
<p>significance level of the test.</p>
</td></tr>
<tr><td><code id="pathway.mean.test_+3A_variance">variance</code></td>
<td>
<p>logical flag. If <code>TRUE</code> estimated variances s1  and s2 are reported.</p>
</td></tr>
<tr><td><code id="pathway.mean.test_+3A_perm.num">perm.num</code></td>
<td>
<p>number of permutations of the test.</p>
</td></tr>
<tr><td><code id="pathway.mean.test_+3A_paired">paired</code></td>
<td>
<p>logical flag. If <code>TRUE</code> Hotelling test for paired samples is calculated and the test on the variances is not performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function tests  the equality of the means of  two
experimental conditions, conditioning on the result of
<code><a href="#topic+pathway.var.test">pathway.var.test</a></code>.
</p>
<p>The expression data may contain some genes differing
from those in the pathway: in such case the function automatically takes
the intersection between the two gene sets.
</p>
<p>This function requires <code>gRBase</code> and <code>qpgraph</code> packages.
</p>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>
<p>the p-value of the test on the mean.</p>
</td></tr>
<tr><td><code>p.value.var</code></td>
<td>
<p>the p-value of the test on the variance. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>lambda.value</code></td>
<td>
<p>the observed value of the test statistic for the
variance. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>qchisq.value</code></td>
<td>
<p>the 1 - alpha quantile value of the null
distribution of the test statistic on the variance. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>cli.moral</code></td>
<td>
<p>a list containing the cliques of the moral graph.</p>
</td></tr>
<tr><td><code>var.equal</code></td>
<td>
<p>logical flag. If <code>TRUE</code> variances are heteroschedastic. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>graph</code></td>
<td>
<p>a graphNEL object representing the moral graph.</p>
</td></tr>
<tr><td><code>df.var</code></td>
<td>
<p>the degrees of freedom of the null distribution for the
variance test. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>df.mean</code></td>
<td>
<p>the degrees of freedom of the null distribution for the
mean test. If paired is <code>TRUE</code>, this is not returned.</p>
</td></tr>
<tr><td><code>t.value</code></td>
<td>
<p>the observed value of the test statistic for the mean.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>M. Sofia Massa,  Gabriele Sales</p>


<h3>References</h3>

<p>Massa, M.S., Chiogna, M., Romualdi, C. (2010). Gene set analysis exploiting the topology of a pathway. BMC Systems Biology, 4:121
<a href="https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121">https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pathway.var.test">pathway.var.test</a></code>, <code><a href="#topic+clique.var.test">clique.var.test</a></code>,
<code><a href="#topic+clique.mean.test">clique.mean.test</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(examples)

pathway.mean.test(y1, y2, dag_bcell, 0.05, 100)
</code></pre>

<hr>
<h2 id='pathway.var.test'>Testing the concentration matrices of a pathway.</h2><span id='topic+pathway.var.test'></span>

<h3>Description</h3>

<p>Test the equality of concentration matrices in two experimental conditions for a pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway.var.test(y1,y2,dag,alpha,variance=FALSE,s1=NULL,s2=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway.var.test_+3A_y1">y1</code></td>
<td>
<p>a matrix with n1 individuals (rows) in the first experimental condition and p genes (columns).</p>
</td></tr>
<tr><td><code id="pathway.var.test_+3A_y2">y2</code></td>
<td>
<p>a matrix with n2 individuals (rows) in the second experimental condition and p genes (columns).
The genes in the two experimental conditions must be the same.</p>
</td></tr>
<tr><td><code id="pathway.var.test_+3A_dag">dag</code></td>
<td>
<p>graphNEL object, directed acyclic graph (DAG) corresponding to the pathway of interest. See package <code>gRbase</code> for more details.</p>
</td></tr>
<tr><td><code id="pathway.var.test_+3A_alpha">alpha</code></td>
<td>
<p>significance level of the test.</p>
</td></tr>
<tr><td><code id="pathway.var.test_+3A_variance">variance</code></td>
<td>
<p>logical flag. If <code>TRUE</code> the estimates of the
covariance matrices are included in the result.</p>
</td></tr>
<tr><td><code id="pathway.var.test_+3A_s1">s1</code></td>
<td>
<p>y1 covariance matrix estimation.</p>
</td></tr>
<tr><td><code id="pathway.var.test_+3A_s2">s2</code></td>
<td>
<p>y2 covariance matrix estimation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The graph of a pathway is first converted into a DAG and then
into a moral graph. The data is modelled with two Gaussian graphical
models with zero mean and graph provided by the moral graph. The
function tests the equality of the two concentration matrices
(inverse of the covariance matrices).
</p>
<p>The expression data may contain some genes differing
from those in the pathway: in such case the function automatically takes
the intersection between the two gene sets.
</p>
<p>A necessary condition for the existence of the covariance estimates is that
the number of statistical units (samples) is greater than the number of
variables. If this is not the case, penalized techniques for estimating
<code class="reqn">\hat{\Sigma}_{1}^{-1}</code> and <code class="reqn">\hat{\Sigma}_{2}^{-1}</code>
have to be employed, that are currently not provided by the package. In theses
cases, one can perform penalized estimation of
<code class="reqn">\hat{\Sigma}_{1}^{-1}</code> and <code class="reqn">\hat{\Sigma}_{2}^{-1}</code>
outside <span class="pkg">topologyGSA</span>, and then provide such estimates as input arguments
to the function <code>pathway.var.test</code> to compute the value of the test for
homogeneity. In this case, computation of the p-value deserves attention,
as standard results on the asymptotic distribution of the test statistic may
no longer be valid. Therefore, computation of the p-value has to be dealt
with by the user.
</p>
<p>This function requires <code>gRBase</code> and <code>qpgraph</code> packages.
</p>


<h3>Value</h3>

<table>
<tr><td><code>lambda.value</code></td>
<td>
<p>the observed value of the test statistic.</p>
</td></tr>
<tr><td><code>qchisq.value</code></td>
<td>
<p>the 1 - alpha quantile value of the null
distribution of the test statistic on the variance.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the observed value of the significance level of the test.</p>
</td></tr>
<tr><td><code>cli.moral</code></td>
<td>
<p>a list containing the cliques of the moral graph.</p>
</td></tr>
<tr><td><code>var.equal</code></td>
<td>
<p>logical flag. If TRUE variances are heteroschedastic.</p>
</td></tr>
<tr><td><code>graph</code></td>
<td>
<p>a graphNEL object representing the moral graph.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>the degrees of freedom of the null distribution.</p>
</td></tr>
<tr><td><code>s1</code></td>
<td>
<p>if variance=TRUE, the estimate of y1 covariance.</p>
</td></tr>
<tr><td><code>s2</code></td>
<td>
<p>if variance=TRUE, the estimate of y2 covariance.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>M. Sofia Massa, Gabriele Sales</p>


<h3>References</h3>

<p>Massa, M.S., Chiogna, M., Romualdi, C. (2010). Gene set
analysis exploiting the topology of a pathway. BMC Systems Biology,
4:121
<a href="https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121">https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121</a>
</p>
<p>Lauritzen, S.L. (1996). Graphical models. Clarendon Press, Oxford.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pathway.mean.test">pathway.mean.test</a></code>, <code><a href="#topic+clique.var.test">clique.var.test</a></code>,
<code><a href="#topic+clique.mean.test">clique.mean.test</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(examples)

pathway.var.test(y1, y2, dag_bcell, 0.05)
</code></pre>

<hr>
<h2 id='print'>Print results of statistical tests.</h2><span id='topic+print.clique.var.test'></span><span id='topic+print.clique.mean.test'></span><span id='topic+print.pathway.var.test'></span><span id='topic+print.pathway.mean.test'></span>

<h3>Description</h3>

<p>This functions prints a summary of the results of each statistical test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clique.var.test'
print(x, ...)
## S3 method for class 'clique.mean.test'
print(x, ...)
## S3 method for class 'pathway.var.test'
print(x, ...)
## S3 method for class 'pathway.mean.test'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_+3A_x">x</code></td>
<td>
<p>the result of a statistical test</p>
</td></tr>
<tr><td><code id="print_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>M. Sofia Massa, Gabriele Sales</p>

<hr>
<h2 id='render.significant.cliques'>Graphical display of the significant cliques</h2><span id='topic+render.significant.cliques'></span>

<h3>Description</h3>

<p>This function highlights the significant cliques on the moralized and triangulated graph. Different colors correspond to different values of -log(pvalue)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>render.significant.cliques(info, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="render.significant.cliques_+3A_info">info</code></td>
<td>
<p>the result of <code><a href="#topic+clique.var.test">clique.var.test</a></code> or
<code><a href="#topic+clique.mean.test">clique.mean.test</a></code>.
</p>
</td></tr>
<tr><td><code id="render.significant.cliques_+3A_alpha">alpha</code></td>
<td>
<p>p-value cutoff to define significant cliques</p>
</td></tr>
</table>


<h3>References</h3>

<p>Massa, M.S., Chiogna, M., Romualdi, C. (2010). Gene set
analysis exploiting the topology of a pathway. BMC Systems Biology,
4:121
<a href="https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121">https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clique.var.test">clique.var.test</a></code>, <code><a href="#topic+clique.mean.test">clique.mean.test</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(examples)

info &lt;- clique.var.test(y1, y2, dag_bcell, 0.1)
## Not run: render.significant.cliques(info)
</code></pre>

<hr>
<h2 id='y1'>Gene expression signatures in acute lymphocytic leukemia for BCR signaling pathway</h2><span id='topic+y1'></span>

<h3>Description</h3>

<p>Several distinct genetic mechanisms lead to acute lymphocytic leukemia (ALL) malignant transformations deriving
from distinct lymphoid precursor cells that have been committed to either T-lineage or B-lineage differentiation.
Chromosome translocations and molecular rearrangements are common events in B-lineage ALL and reflect distinct
mechanisms of transformation. The B cell Receptor (BCR/ABL) gene rearrangement occurs in about 25% of cases in
adult ALL, and much less frequently in pediatric ALL.
<code>y1</code> dataset contains gene expression signatures of BCR signaling pathway (composed by 35 gene products, n.columns = 35)
in acute lymphocytic leukemia (ALL) cells associated with presence of BCR/ABL gene rearrangement (n. rows = 37).
Expression values are normalized according to <code>rma</code> and <code>quantile</code> normalization.
</p>
<p>Chiaretti S., Li X., Gentleman R., Vitale A., Wang K.S., Mandelli F., Fo R., Ritz J. (2005) Gene expression
profiles of B-lineage adult acute lymphocytic leukemia reveal genetic patterns that identify lineage derivation
and distinct mechanisms of transformation. Clin Cancer Res, 11:7209-7219.
<a href="https://aacrjournals.org/clincancerres/article/11/20/7209/188537/Gene-Expression-Profiles-of-B-lineage-Adult-Acute">https://aacrjournals.org/clincancerres/article/11/20/7209/188537/Gene-Expression-Profiles-of-B-lineage-Adult-Acute</a>
</p>
<p>Massa, M.S., Chiogna, M., Romualdi, C. (2010). Gene set analysis exploiting the topology of a pathway. BMC Systems Biology, 4:121
<a href="https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121">https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121</a>
</p>

<hr>
<h2 id='y2'>Gene expression signatures in acute lymphocytic leukemia for BCR signaling pathway</h2><span id='topic+y2'></span>

<h3>Description</h3>

<p>Several distinct genetic mechanisms lead to acute lymphocytic leukemia (ALL) malignant transformations deriving
from distinct lymphoid precursor cells that have been committed to either T-lineage or B-lineage differentiation.
Chromosome translocations and molecular rearrangements are common events in B-lineage ALL and reflect distinct
mechanisms of transformation. The B cell Receptor (BCR/ABL) gene rearrangement occurs in about 25% of cases in
adult ALL, and much less frequently in pediatric ALL.
<code>y2</code> dataset contains gene expression signatures of BCR signaling pathway (composed by 35 gene products, n.columns = 35)
in acute lymphocytic leukemia (ALL) cells not associated with BCR/ABL gene rearrangement (n. rows = 41).
Expression values are normalized according to <code>rma</code> and <code>quantile</code>
normalization.
</p>
<p>Chiaretti S., Li X., Gentleman R., Vitale A., Wang K.S., Mandelli F., Fo R., Ritz J. (2005) Gene expression
profiles of B-lineage adult acute lymphocytic leukemia reveal genetic patterns that identify lineage derivation
and distinct mechanisms of transformation. Clin Cancer Res, 11:7209-7219.
<a href="https://aacrjournals.org/clincancerres/article/11/20/7209/188537/Gene-Expression-Profiles-of-B-lineage-Adult-Acute">https://aacrjournals.org/clincancerres/article/11/20/7209/188537/Gene-Expression-Profiles-of-B-lineage-Adult-Acute</a>
</p>
<p>Massa, M.S., Chiogna, M., Romualdi, C. (2010). Gene set analysis exploiting the topology of a pathway. BMC Systems Biology, 4:121
<a href="https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121">https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-4-121</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
