<!DOCTYPE html><html><head><title>Help for package topr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {topr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#annotate_with_nearest_gene'><p>Get the nearest gene for one or more snps</p></a></li>
<li><a href='#CD_FINNGEN'><p>Finngen r7 Crohn‘s disease (K11_CROHNS)</p></a></li>
<li><a href='#CD_UKBB'><p>UKBB Crohns disease (ICD 10 code K50)</p></a></li>
<li><a href='#create_snpset'><p>Create a dataframe that can be used as input for making effect plots</p></a></li>
<li><a href='#create_snpset_code'><p>Show the code/functions used to create a snpset</p></a></li>
<li><a href='#effect_plot'><p>Create a plot comparing effects within two datasets</p></a></li>
<li><a href='#effectplot'><p>Create a plot comparing variant effects in two datasets</p></a></li>
<li><a href='#flip_to_positive_allele_for_dat1'><p>Flip to the positive allele for dataset 1</p></a></li>
<li><a href='#get_best_snp_per_MB'><p>Get the index/lead variants</p></a></li>
<li><a href='#get_gene'><p>Get the genetic position of a gene by gene name</p></a></li>
<li><a href='#get_gene_coords'><p>Get the genetic position of a gene by gene name</p></a></li>
<li><a href='#get_genes_by_Gene_Symbol'><p>Get the genetic position of a gene by its gene name</p></a></li>
<li><a href='#get_genes_in_region'><p>Get SNPs/variants within region</p></a></li>
<li><a href='#get_lead_snps'><p>Get the index/lead variants</p></a></li>
<li><a href='#get_overlapping_snps_by_pos'><p>Get variants that overlap between two datasets</p></a></li>
<li><a href='#get_snps_within_region'><p>Get SNPs/variants within region</p></a></li>
<li><a href='#get_snpset'><p>Create a dataframe that can be used as input for making effect plots</p></a></li>
<li><a href='#get_snpset_code'><p>Show the code/functions used to get a snpset</p></a></li>
<li><a href='#get_top_snp'><p>Get the top hit from the dataframe</p></a></li>
<li><a href='#get_topr_colors'><p>Get the top hit from the dataframe</p></a></li>
<li><a href='#locuszoom'><p>Create a locuszoom-like plot</p></a></li>
<li><a href='#manhattan'><p>Create a Manhattan plot</p></a></li>
<li><a href='#match_alleles'><p>Match the variants in the snpset by their alleles</p></a></li>
<li><a href='#match_by_alleles'><p>Match the variants in the snpset by their alleles</p></a></li>
<li><a href='#match_by_pos'><p>Get variants that overlap between two datasets</p></a></li>
<li><a href='#qqtopr'><p>Create a quantile quantile (QQ) plot</p></a></li>
<li><a href='#R2_CD_UKBB'><p>Example dataset including the R2 column for the locuszoom plot function</p></a></li>
<li><a href='#regionplot'><p>Create a regionplot</p></a></li>
<li><a href='#topr'><p>topr</p></a></li>
<li><a href='#UC_UKBB'><p>UKBB Ulcerative colitis (ICD 10 code K51)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Create Custom Plots for Viewing Genetic Association Results</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/totajuliusd/topr">https://github.com/totajuliusd/topr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/totajuliusd/topr/issues">https://github.com/totajuliusd/topr/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of functions for visualizing,exploring and annotating genetic association results.Association results from multiple traits can be viewed simultaneously along with gene annotation, over the entire genome (Manhattan plot) or in the more detailed regional view.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, rmarkdown, markdown</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 3.3.2), dplyr (&ge; 1.0.2), stringr (&ge; 1.4.0),
scales (&ge; 1.1.1), ggrepel, egg, grid, gridExtra, magrittr (&ge;
1.5), utils, tidyr, enshuman,</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-15 16:11:53 UTC; tota</td>
</tr>
<tr>
<td>Author:</td>
<td>Thorhildur Juliusdottir [cph, aut, cre],
  Andri Stefansson [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thorhildur Juliusdottir &lt;totajuliusd@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-15 16:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic++25+3E+25">%&gt;%</a></code> for details.
</p>

<hr>
<h2 id='annotate_with_nearest_gene'>Get the nearest gene for one or more snps</h2><span id='topic+annotate_with_nearest_gene'></span>

<h3>Description</h3>

<p><code>annotate_with_nearest_gene()</code> Annotate the variant/snp with their nearest gene
Required parameters is a dataframe of SNPs (with the columns CHROM and POS)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotate_with_nearest_gene(
  variants,
  protein_coding_only = FALSE,
  build = 38,
  .chr_map = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotate_with_nearest_gene_+3A_variants">variants</code></td>
<td>
<p>a dataframe of variant positions (CHROM and POS)</p>
</td></tr>
<tr><td><code id="annotate_with_nearest_gene_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>Logical, if set to TRUE only annotate with protein coding genes (the default value is FALSE)</p>
</td></tr>
<tr><td><code id="annotate_with_nearest_gene_+3A_build">build</code></td>
<td>
<p>A number representing the genome build. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td></tr>
<tr><td><code id="annotate_with_nearest_gene_+3A_.chr_map">.chr_map</code></td>
<td>
<p>An internally used list which maps chromosome names to numbers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the input dataframe with Gene_Symbol as an additional column
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
variants &lt;-get_lead_snps(CD_UKBB)
annotate_with_nearest_gene(variants)

## End(Not run)

</code></pre>

<hr>
<h2 id='CD_FINNGEN'>Finngen r7 Crohn‘s disease (K11_CROHNS)</h2><span id='topic+CD_FINNGEN'></span>

<h3>Description</h3>

<p>Dataset retrieved from the Finngen database (version 7) including 3147 crohn´s cases (K50) and 296,100 controls. The dataset has been filtered on variants with P &lt;1e-03.
FinnGen data are publicly available and were downloaded from https://finngen.fi.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CD_FINNGEN
</code></pre>


<h3>Format</h3>

<p>A data frame  with 32,303 rows and 8 variables:
</p>

<dl>
<dt>CHROM</dt><dd><p>Chromosome, written as for example chr1 or 1</p>
</dd>
<dt>POS</dt><dd><p>genetic position of the variant</p>
</dd>
<dt>REF</dt><dd><p>the reference allele</p>
</dd>
<dt>ALT</dt><dd><p>the alternative allele</p>
</dd>
<dt>P</dt><dd><p>P-value from Plink run, additive model, regression model GLM_FIRTH</p>
</dd>
<dt>BETA</dt><dd><p>Variant effect</p>
</dd>
<dt>ID</dt><dd><p>Variant identifier, e.g. rsid </p>
</dd>
<dt>AF</dt><dd><p>Allele frequency</p>
</dd>
</dl>



<h3>Source</h3>

<p>Crohn's K50 (K11_CROHNS), only including variants with P&lt;1e-03
</p>

<hr>
<h2 id='CD_UKBB'>UKBB Crohns disease (ICD 10 code K50)</h2><span id='topic+CD_UKBB'></span>

<h3>Description</h3>

<p>Dataset retrieved from the UK biobank consisting of 2,799 crohn´s cases (K50) and 484,515 controls. The dataset has been filtered on variants with P &lt;1e-03.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CD_UKBB
</code></pre>


<h3>Format</h3>

<p>A data frame  with 21,717 rows and 8 variables:
</p>

<dl>
<dt>CHROM</dt><dd><p>Chromosome, written as for example chr1 or 1</p>
</dd>
<dt>POS</dt><dd><p>genetic position of the variant</p>
</dd>
<dt>REF</dt><dd><p>the reference allele</p>
</dd>
<dt>ALT</dt><dd><p>the alternative allele</p>
</dd>
<dt>ID</dt><dd><p>Variant identifier, e.g. rsid </p>
</dd>
<dt>P</dt><dd><p>P-value from Plink run, additive model, regression model GLM_FIRTH</p>
</dd>
<dt>OR</dt><dd><p>Odds Ratio</p>
</dd>
<dt>AF</dt><dd><p>Allele frequency</p>
</dd>
</dl>



<h3>Source</h3>

<p>Crohn's UKBB ICD10 code K50, only including variants with P&lt;1e-03
</p>

<hr>
<h2 id='create_snpset'>Create a dataframe that can be used as input for making effect plots</h2><span id='topic+create_snpset'></span>

<h3>Description</h3>

<p><code>create_snpset()</code>
</p>
<p>This method is deprecated and will be removed in future versions. use <code><a href="#topic+get_snpset">get_snpset</a></code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_snpset(
  df1,
  df2,
  thresh = 1e-08,
  protein_coding_only = TRUE,
  region_size = 1e+06,
  verbose = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_snpset_+3A_df1">df1</code></td>
<td>
<p>The dataframe to extract the top snps from (with p-value below thresh)</p>
</td></tr>
<tr><td><code id="create_snpset_+3A_df2">df2</code></td>
<td>
<p>The dataframe in which to search for overlapping SNPs from dataframe1</p>
</td></tr>
<tr><td><code id="create_snpset_+3A_thresh">thresh</code></td>
<td>
<p>Numeric, the p-value threshold used for extracting the top snps from dataset 1</p>
</td></tr>
<tr><td><code id="create_snpset_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>Logical, set this variable to TRUE to only use protein_coding genes for the annotation</p>
</td></tr>
<tr><td><code id="create_snpset_+3A_region_size">region_size</code></td>
<td>
<p>Integer, the size of the interval which to extract the top snps from</p>
</td></tr>
<tr><td><code id="create_snpset_+3A_verbose">verbose</code></td>
<td>
<p>Logical, (default: FALSE). Assign to TRUE to get information on which alleles are matched and which are not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe containing the top hit
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
create_snpset(CD_UKBB,CD_FINNGEN, thresh=1e-09)

## End(Not run)

</code></pre>

<hr>
<h2 id='create_snpset_code'>Show the code/functions used to create a snpset</h2><span id='topic+create_snpset_code'></span>

<h3>Description</h3>

<p>This method is deprecated and will be removed in future versions. use <code><a href="#topic+get_snpset_code">get_snpset_code</a></code> instead.
</p>
<p><code>create_snpset_code()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_snpset_code()
</code></pre>


<h3>Value</h3>

<p>Dataframe containing the top hit
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
create_snpset_code()

## End(Not run)

</code></pre>

<hr>
<h2 id='effect_plot'>Create a plot comparing effects within two datasets</h2><span id='topic+effect_plot'></span>

<h3>Description</h3>

<p><code>effect_plot()</code>
</p>
<p>This method is deprecated and will be removed in future versions. use <code><a href="#topic+effectplot">effectplot</a></code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effect_plot(
  dat,
  pheno_x = "pheno_x",
  pheno_y = "pheno_",
  annotate_with = "Gene_Symbol",
  thresh = 1e-08,
  ci_thresh = 1,
  gene_label_thresh = 1e-08,
  color = get_topr_colors()[1],
  scale = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="effect_plot_+3A_dat">dat</code></td>
<td>
<p>The input dataframe (snpset) containing one row per variant and P values (P1 and P2) and effects (E1 and E2) from two datasets/phenotypes</p>
</td></tr>
<tr><td><code id="effect_plot_+3A_pheno_x">pheno_x</code></td>
<td>
<p>A string representing the name of the phenotype whose effect is plotted on the x axis</p>
</td></tr>
<tr><td><code id="effect_plot_+3A_pheno_y">pheno_y</code></td>
<td>
<p>A string representing the name of the phenotype whose effect is plotted on the y axis</p>
</td></tr>
<tr><td><code id="effect_plot_+3A_annotate_with">annotate_with</code></td>
<td>
<p>A string, The name of the column that contains the label for the datapoints (default value is Gene_Symbol)</p>
</td></tr>
<tr><td><code id="effect_plot_+3A_thresh">thresh</code></td>
<td>
<p>A number. Threshold cutoff, datapoints with P2 below this threshold are shown as filled circles whereas datapoints with P2 above this threshold are shown as open circles</p>
</td></tr>
<tr><td><code id="effect_plot_+3A_ci_thresh">ci_thresh</code></td>
<td>
<p>A number.Show the confidence intervals if the P-value is below this threshold</p>
</td></tr>
<tr><td><code id="effect_plot_+3A_gene_label_thresh">gene_label_thresh</code></td>
<td>
<p>A string, label datapoints with P2 below this threshold</p>
</td></tr>
<tr><td><code id="effect_plot_+3A_color">color</code></td>
<td>
<p>A string, default value is the first of the topr colors</p>
</td></tr>
<tr><td><code id="effect_plot_+3A_scale">scale</code></td>
<td>
<p>A number, to change the size of the title and axes labels and ticks at the same time (default = 1)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
effect_plot(dat)

## End(Not run)


</code></pre>

<hr>
<h2 id='effectplot'>Create a plot comparing variant effects in two datasets</h2><span id='topic+effectplot'></span>

<h3>Description</h3>

<p><code>effectplot()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effectplot(
  df,
  pheno_x = "x_pheno",
  pheno_y = "y_pheno",
  annotate_with = "Gene_Symbol",
  thresh = 5e-08,
  ci_thresh = 1,
  gene_label_thresh = 5e-08,
  color = get_topr_colors()[1],
  scale = 1,
  build = 38,
  label_fontface = "italic",
  label_family = "",
  nudge_y = 0.001,
  nudge_x = 0.001,
  size = 2,
  segment.size = 0.2,
  segment.linetype = "solid",
  segment.color = "transparent",
  angle = 0,
  title = NULL,
  axis_text_size = 10,
  axis_title_size = 12,
  title_text_size = 13,
  subtitle_text_size = 11,
  gene_label_size = 3.2,
  snpset_thresh = 5e-08,
  snpset_region_size = 1e+06,
  max.overlaps = 10,
  annotate = 0,
  label_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="effectplot_+3A_df">df</code></td>
<td>
<p>The input dataframe (snpset) containing one row per variant and P values (P1 and P2) and effects (E1 and E2) from two datasets/phenotypes OR a list containing two datasets.</p>
</td></tr>
<tr><td><code id="effectplot_+3A_pheno_x">pheno_x</code></td>
<td>
<p>A string representing the name of the phenotype whose effect is plotted on the x axis</p>
</td></tr>
<tr><td><code id="effectplot_+3A_pheno_y">pheno_y</code></td>
<td>
<p>A string representing the name of the phenotype whose effect is plotted on the y axis</p>
</td></tr>
<tr><td><code id="effectplot_+3A_annotate_with">annotate_with</code></td>
<td>
<p>A string, The name of the column that contains the label for the datapoints (default value is Gene_Symbol)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_thresh">thresh</code></td>
<td>
<p>A number. Threshold cutoff, datapoints with P2 below this threshold are shown as filled circles whereas datapoints with P2 above this threshold are shown as open circles</p>
</td></tr>
<tr><td><code id="effectplot_+3A_ci_thresh">ci_thresh</code></td>
<td>
<p>A number.Show the confidence intervals if the P-value is below this threshold</p>
</td></tr>
<tr><td><code id="effectplot_+3A_gene_label_thresh">gene_label_thresh</code></td>
<td>
<p>Deprecated: A number, label datapoints with P2 below this threshold</p>
</td></tr>
<tr><td><code id="effectplot_+3A_color">color</code></td>
<td>
<p>A string, default value is the first of the topr colors</p>
</td></tr>
<tr><td><code id="effectplot_+3A_scale">scale</code></td>
<td>
<p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_build">build</code></td>
<td>
<p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td></tr>
<tr><td><code id="effectplot_+3A_label_fontface">label_fontface</code></td>
<td>
<p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_label_family">label_family</code></td>
<td>
<p>A string or a vector of strings. Label font name (default ggrepel argument is &quot;&quot;)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_nudge_y">nudge_y</code></td>
<td>
<p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_nudge_x">nudge_x</code></td>
<td>
<p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_size">size</code></td>
<td>
<p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_segment.size">segment.size</code></td>
<td>
<p>line segment color (ggrepel argument)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_segment.linetype">segment.linetype</code></td>
<td>
<p>line segment solid, dashed, etc.(ggrepel argument)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_segment.color">segment.color</code></td>
<td>
<p>line segment thickness (ggrepel argument)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_angle">angle</code></td>
<td>
<p>A number, the angle of the text label</p>
</td></tr>
<tr><td><code id="effectplot_+3A_title">title</code></td>
<td>
<p>A string to set the plot title</p>
</td></tr>
<tr><td><code id="effectplot_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A number, size of the x and y axes tick labels (default: 12)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A number, size of the x and y title labels (default: 12)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_title_text_size">title_text_size</code></td>
<td>
<p>A number, size of the plot title (default: 13)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_subtitle_text_size">subtitle_text_size</code></td>
<td>
<p>A number setting the text size of the subtitle (default: 11)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_gene_label_size">gene_label_size</code></td>
<td>
<p>A number setting the size of the gene labels shown at the bottom of the plot</p>
</td></tr>
<tr><td><code id="effectplot_+3A_snpset_thresh">snpset_thresh</code></td>
<td>
<p>A number representing the threshold used to create the snpset used for plotting (Only applicable if the input dataframe is a list containing two datasets)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_snpset_region_size">snpset_region_size</code></td>
<td>
<p>A number representing the region size to use when creating the snpset used for plotting (Only applicable if the input dataframe is a list containing two datasets)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_max.overlaps">max.overlaps</code></td>
<td>
<p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p>
</td></tr>
<tr><td><code id="effectplot_+3A_annotate">annotate</code></td>
<td>
<p>A number, label datapoints with p-value below below this number (in the second df) by their nearest gene</p>
</td></tr>
<tr><td><code id="effectplot_+3A_label_color">label_color</code></td>
<td>
<p>A string or a vector of strings. To change the color of the gene or variant labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
effectplot(list(CD_UKBB, CD_FINNGEN))

## End(Not run)

</code></pre>

<hr>
<h2 id='flip_to_positive_allele_for_dat1'>Flip to the positive allele for dataset 1</h2><span id='topic+flip_to_positive_allele_for_dat1'></span>

<h3>Description</h3>

<p><code>flip_to_positive_allele_for_dat1()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flip_to_positive_allele_for_dat1(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flip_to_positive_allele_for_dat1_+3A_df">df</code></td>
<td>
<p>A dataframe that is in the snpset format (like returned by the get_snpset() function)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input dataframe after flipping to the positive effect allele in dataframe 1
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
CD_UKBB_index_snps &lt;- get_lead_snps(CD_UKBB)
snpset &lt;- get_snpset(CD_UKBB_index_snps, CD_FINNGEN)
flip_to_positive_allele_for_dat1(snpset$matched)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_best_snp_per_MB'>Get the index/lead variants</h2><span id='topic+get_best_snp_per_MB'></span>

<h3>Description</h3>

<p><code>get_best_snp_per_MB()</code> Get the top variants within 1 MB windows of the genome with association p-values below the given threshold
</p>
<p>This method is deprecated and will be removed in future versions. use <code><a href="#topic+get_lead_snps">get_lead_snps</a></code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_best_snp_per_MB(
  df,
  thresh = 5e-09,
  region_size = 1e+06,
  protein_coding_only = FALSE,
  chr = NULL,
  .checked = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_best_snp_per_MB_+3A_df">df</code></td>
<td>
<p>Dataframe</p>
</td></tr>
<tr><td><code id="get_best_snp_per_MB_+3A_thresh">thresh</code></td>
<td>
<p>A number. P-value threshold, only extract variants with p-values below this threshold (5e-09 by default)</p>
</td></tr>
<tr><td><code id="get_best_snp_per_MB_+3A_region_size">region_size</code></td>
<td>
<p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td></tr>
<tr><td><code id="get_best_snp_per_MB_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>Logical, set this variable to TRUE to only use protein_coding genes for annotation</p>
</td></tr>
<tr><td><code id="get_best_snp_per_MB_+3A_chr">chr</code></td>
<td>
<p>String, get the top variants from one chromosome only, e.g. chr=&quot;chr1&quot;</p>
</td></tr>
<tr><td><code id="get_best_snp_per_MB_+3A_.checked">.checked</code></td>
<td>
<p>Logical, if the input data has already been checked, this can be set to TRUE so it wont be checked again (FALSE by default)</p>
</td></tr>
<tr><td><code id="get_best_snp_per_MB_+3A_verbose">verbose</code></td>
<td>
<p>Logical, set to TRUE to get printed information on number of SNPs extracted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of lead variants. Returns the best variant per MB (by default, change the region size with the region argument) with p-values below the input threshold (thresh=5e-09 by default)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  get_best_snp_per_MB(CD_UKBB)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_gene'>Get the genetic position of a gene by gene name</h2><span id='topic+get_gene'></span>

<h3>Description</h3>

<p><code>get_gene()</code> Get the gene coordinates for a gene
Required parameter is gene name
</p>
<p>This method is deprecated and will be removed in future versions. use <code><a href="#topic+get_gene_coords">get_gene_coords</a></code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene(gene_name, chr = NULL, build = 38)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gene_+3A_gene_name">gene_name</code></td>
<td>
<p>A string representing a gene name (e.g. &quot;FTO&quot;)</p>
</td></tr>
<tr><td><code id="get_gene_+3A_chr">chr</code></td>
<td>
<p>A string, search for the genes on this chromosome only, (e.g chr=&quot;chr1&quot;)</p>
</td></tr>
<tr><td><code id="get_gene_+3A_build">build</code></td>
<td>
<p>A string, genome build, choose between builds 37 (GRCh37) and 38 (GRCh38) (default is 38)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe with the gene name and its genetic coordinates
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_gene("FTO")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_gene_coords'>Get the genetic position of a gene by gene name</h2><span id='topic+get_gene_coords'></span>

<h3>Description</h3>

<p><code>get_gene_coords()</code> Get the gene coordinates for a gene
Required parameter is gene name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_coords(gene_name, chr = NULL, build = 38)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gene_coords_+3A_gene_name">gene_name</code></td>
<td>
<p>A string representing a gene name (e.g. &quot;FTO&quot;)</p>
</td></tr>
<tr><td><code id="get_gene_coords_+3A_chr">chr</code></td>
<td>
<p>A string, search for the genes on this chromosome only, (e.g chr=&quot;chr1&quot;)</p>
</td></tr>
<tr><td><code id="get_gene_coords_+3A_build">build</code></td>
<td>
<p>A string, genome build, choose between builds 37 (GRCh37) and 38 (GRCh38) (default is 38)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe with the gene name and its genetic coordinates
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_gene_coords("FTO")

## End(Not run)

</code></pre>

<hr>
<h2 id='get_genes_by_Gene_Symbol'>Get the genetic position of a gene by its gene name</h2><span id='topic+get_genes_by_Gene_Symbol'></span>

<h3>Description</h3>

<p><code>get_genes_by_Gene_Symbol()</code> Get genes by their gene symbol/name
Required parameters is on gene name or a vector of gene names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genes_by_Gene_Symbol(genes, chr = NULL, build = 38)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_genes_by_Gene_Symbol_+3A_genes">genes</code></td>
<td>
<p>A string or vector of strings representing gene names, (e.g. &quot;FTO&quot;) or  (c(&quot;FTO&quot;,&quot;NOD2&quot;))</p>
</td></tr>
<tr><td><code id="get_genes_by_Gene_Symbol_+3A_chr">chr</code></td>
<td>
<p>A string, search for the genes on this chromosome only, (e.g chr=&quot;chr1&quot;)</p>
</td></tr>
<tr><td><code id="get_genes_by_Gene_Symbol_+3A_build">build</code></td>
<td>
<p>A string, genome build, choose between builds 37 (GRCh37) and 38 (GRCh38) (default is 38)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of genes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  get_genes_by_Gene_Symbol(c("FTO","THADA"))

## End(Not run)

</code></pre>

<hr>
<h2 id='get_genes_in_region'>Get SNPs/variants within region</h2><span id='topic+get_genes_in_region'></span>

<h3>Description</h3>

<p><code>get_genes_in_region()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genes_in_region(
  chr = chr,
  xmin = xmin,
  xmax = xmax,
  protein_coding_only = F,
  show_exons = F,
  show_genes = T,
  build = 38,
  region = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_genes_in_region_+3A_chr">chr</code></td>
<td>
<p>A string, chromosome (e.g. chr16)</p>
</td></tr>
<tr><td><code id="get_genes_in_region_+3A_xmin">xmin</code></td>
<td>
<p>An integer representing genetic position</p>
</td></tr>
<tr><td><code id="get_genes_in_region_+3A_xmax">xmax</code></td>
<td>
<p>An integer representing genetic position</p>
</td></tr>
<tr><td><code id="get_genes_in_region_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>A logical scalar, if TRUE, only protein coding genes are used for annotation</p>
</td></tr>
<tr><td><code id="get_genes_in_region_+3A_show_exons">show_exons</code></td>
<td>
<p>Deprecated : A logical scalar, show exons instead of genes (default show_exons=FALSE)</p>
</td></tr>
<tr><td><code id="get_genes_in_region_+3A_show_genes">show_genes</code></td>
<td>
<p>A logical scalar, show genes instead of exons (default show_genes=FALSE)</p>
</td></tr>
<tr><td><code id="get_genes_in_region_+3A_build">build</code></td>
<td>
<p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td></tr>
<tr><td><code id="get_genes_in_region_+3A_region">region</code></td>
<td>
<p>A string representing the genetic region (e.g chr16:50693587-50734041)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the genes the requested region
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_genes_in_region(region="chr16:50593587-50834041")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_lead_snps'>Get the index/lead variants</h2><span id='topic+get_lead_snps'></span>

<h3>Description</h3>

<p><code>get_lead_snps()</code> Get the top variants within 1 MB windows of the genome with association p-values below the given threshold
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_lead_snps(
  df,
  thresh = 5e-08,
  region_size = 1e+06,
  protein_coding_only = FALSE,
  chr = NULL,
  .checked = FALSE,
  verbose = NULL,
  keep_chr = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_lead_snps_+3A_df">df</code></td>
<td>
<p>Dataframe</p>
</td></tr>
<tr><td><code id="get_lead_snps_+3A_thresh">thresh</code></td>
<td>
<p>A number. P-value threshold, only extract variants with p-values below this threshold (5e-08 by default)</p>
</td></tr>
<tr><td><code id="get_lead_snps_+3A_region_size">region_size</code></td>
<td>
<p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td></tr>
<tr><td><code id="get_lead_snps_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>Logical, set this variable to TRUE to only use protein_coding genes for annotation</p>
</td></tr>
<tr><td><code id="get_lead_snps_+3A_chr">chr</code></td>
<td>
<p>String, get the top variants from one chromosome only, e.g. chr=&quot;chr1&quot;</p>
</td></tr>
<tr><td><code id="get_lead_snps_+3A_.checked">.checked</code></td>
<td>
<p>Logical, if the input data has already been checked, this can be set to TRUE so it wont be checked again (FALSE by default)</p>
</td></tr>
<tr><td><code id="get_lead_snps_+3A_verbose">verbose</code></td>
<td>
<p>Logical, set to TRUE to get printed information on number of SNPs extracted</p>
</td></tr>
<tr><td><code id="get_lead_snps_+3A_keep_chr">keep_chr</code></td>
<td>
<p>Logical, set to FALSE to remove the &quot;chr&quot; prefix before each chromosome if present (TRUE by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of lead variants. Returns the best variant per MB (by default, change the region size with the region argument) with p-values below the input threshold (thresh=5e-08 by default)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_lead_snps(CD_UKBB)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_overlapping_snps_by_pos'>Get variants that overlap between two datasets</h2><span id='topic+get_overlapping_snps_by_pos'></span>

<h3>Description</h3>

<p><code>get_overlapping_snps_by_pos()</code>
</p>
<p>This method is deprecated and will be removed in future versions. use <code><a href="#topic+match_by_pos">match_by_pos</a></code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_overlapping_snps_by_pos(df1, df2, verbose = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_overlapping_snps_by_pos_+3A_df1">df1</code></td>
<td>
<p>A dataframe of variants, has to contain CHROM and POS</p>
</td></tr>
<tr><td><code id="get_overlapping_snps_by_pos_+3A_df2">df2</code></td>
<td>
<p>A dataframe of variants, has to contain CHROM and POS</p>
</td></tr>
<tr><td><code id="get_overlapping_snps_by_pos_+3A_verbose">verbose</code></td>
<td>
<p>A logical scalar (default: FALSE). Assign to TRUE to get information on which alleles are matched and which are not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input dataframe containing only those variants with matched alleles in the snpset
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_overlapping_snps_by_pos(dat1, dat2)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_snps_within_region'>Get SNPs/variants within region</h2><span id='topic+get_snps_within_region'></span>

<h3>Description</h3>

<p><code>get_snps_within_region()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_snps_within_region(
  df,
  region,
  chr = NULL,
  xmin = NULL,
  xmax = NULL,
  keep_chr = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_snps_within_region_+3A_df">df</code></td>
<td>
<p>data frame of association results with the columns CHR and POS</p>
</td></tr>
<tr><td><code id="get_snps_within_region_+3A_region">region</code></td>
<td>
<p>A string representing the genetic region (e.g chr16:50693587-50734041)</p>
</td></tr>
<tr><td><code id="get_snps_within_region_+3A_chr">chr</code></td>
<td>
<p>A string, chromosome (e.g. chr16)</p>
</td></tr>
<tr><td><code id="get_snps_within_region_+3A_xmin">xmin</code></td>
<td>
<p>An integer, include variants with POS larger than xmin</p>
</td></tr>
<tr><td><code id="get_snps_within_region_+3A_xmax">xmax</code></td>
<td>
<p>An integer, include variants with POS smaller than xmax</p>
</td></tr>
<tr><td><code id="get_snps_within_region_+3A_keep_chr">keep_chr</code></td>
<td>
<p>Deprecated: Logical, set to FALSE to remove the &quot;chr&quot; prefix before each chromosome if present (TRUE by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the variants within the requested region
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_snps_within_region(CD_UKBB, "chr16:50593587-50834041")

## End(Not run)

</code></pre>

<hr>
<h2 id='get_snpset'>Create a dataframe that can be used as input for making effect plots</h2><span id='topic+get_snpset'></span>

<h3>Description</h3>

<p><code>get_snpset()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_snpset(
  df1,
  df2,
  thresh = 5e-08,
  protein_coding_only = TRUE,
  region_size = 1e+06,
  verbose = NULL,
  show_full_output = FALSE,
  build = 38
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_snpset_+3A_df1">df1</code></td>
<td>
<p>The dataframe to extract the top snps from (with p-value below thresh)</p>
</td></tr>
<tr><td><code id="get_snpset_+3A_df2">df2</code></td>
<td>
<p>The dataframe in which to search for overlapping SNPs from dataframe1</p>
</td></tr>
<tr><td><code id="get_snpset_+3A_thresh">thresh</code></td>
<td>
<p>A number. P-value threshold, only extract variants with p-values below this threshold (5e-08 by default)</p>
</td></tr>
<tr><td><code id="get_snpset_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>Logical, set this variable to TRUE to only use protein_coding genes for annotation</p>
</td></tr>
<tr><td><code id="get_snpset_+3A_region_size">region_size</code></td>
<td>
<p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td></tr>
<tr><td><code id="get_snpset_+3A_verbose">verbose</code></td>
<td>
<p>Logical, (default: FALSE). Assign to TRUE to get information on which alleles are matched and which are not.</p>
</td></tr>
<tr><td><code id="get_snpset_+3A_show_full_output">show_full_output</code></td>
<td>
<p>A logical scalar (default:FALSE). Assign to TRUE to show the full output from this function</p>
</td></tr>
<tr><td><code id="get_snpset_+3A_build">build</code></td>
<td>
<p>A string, genome build, choose between builds 37 (GRCh37) and 38 (GRCh38) (default is 38)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of overlapping snps (snpset)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
CD_UKBB_index_snps &lt;-get_lead_snps(CD_UKBB)
get_snpset(CD_UKBB_index_snps, CD_FINNGEN)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_snpset_code'>Show the code/functions used to get a snpset</h2><span id='topic+get_snpset_code'></span>

<h3>Description</h3>

<p><code>get_snpset_code()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_snpset_code()
</code></pre>


<h3>Value</h3>

<p>Dataframe containing the top hit
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_snpset_code()

## End(Not run)

</code></pre>

<hr>
<h2 id='get_top_snp'>Get the top hit from the dataframe</h2><span id='topic+get_top_snp'></span>

<h3>Description</h3>

<p><code>get_top_snp()</code> Get the top hit from the dataframe
All other input parameters are optional
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_top_snp(df, chr = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_top_snp_+3A_df">df</code></td>
<td>
<p>Dataframe containing association results</p>
</td></tr>
<tr><td><code id="get_top_snp_+3A_chr">chr</code></td>
<td>
<p>String, get the top hit in the data frame for this chromosome. If chromosome is not provided, the top hit from the entire dataset is returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe containing the top hit
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_top_snp(CD_UKBB, chr="chr1")

## End(Not run)

</code></pre>

<hr>
<h2 id='get_topr_colors'>Get the top hit from the dataframe</h2><span id='topic+get_topr_colors'></span>

<h3>Description</h3>

<p><code>get_topr_colors()</code> Get the top hit from the dataframe
All other input parameters are optional
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_topr_colors()
</code></pre>


<h3>Value</h3>

<p>Vector of colors used for plotting
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_topr_colors()

## End(Not run)

</code></pre>

<hr>
<h2 id='locuszoom'>Create a locuszoom-like plot</h2><span id='topic+locuszoom'></span>

<h3>Description</h3>

<p><code>locuszoom()</code> displays the association results for a smaller region within one chromosome.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locuszoom(
  df,
  annotate = NULL,
  ntop = 3,
  xmin = 0,
  size = 2,
  shape = 19,
  alpha = 1,
  label_size = 4,
  annotate_with = "ID",
  color = NULL,
  axis_text_size = 11,
  axis_title_size = 12,
  title_text_size = 13,
  show_genes = NULL,
  show_overview = FALSE,
  show_exons = FALSE,
  max_genes = 200,
  sign_thresh = 5e-08,
  sign_thresh_color = "red",
  sign_thresh_label_size = 3.5,
  xmax = NULL,
  ymin = NULL,
  ymax = NULL,
  protein_coding_only = FALSE,
  region_size = 1e+06,
  gene_padding = 1e+05,
  angle = 0,
  legend_title_size = 12,
  legend_text_size = 12,
  nudge_x = 0.01,
  nudge_y = 0.01,
  rsids = NULL,
  variant = NULL,
  rsids_color = "gray40",
  legend_name = "Data:",
  legend_position = "right",
  chr = NULL,
  vline = NULL,
  show_gene_names = NULL,
  legend_labels = NULL,
  gene = NULL,
  title = NULL,
  label_color = "gray40",
  region = NULL,
  scale = 1,
  rsids_with_vline = NULL,
  annotate_with_vline = NULL,
  sign_thresh_size = 0.5,
  unit_main = 7,
  unit_gene = 2,
  gene_color = NULL,
  segment.size = 0.2,
  segment.color = "black",
  segment.linetype = "solid",
  show_gene_legend = TRUE,
  max.overlaps = 10,
  extract_plots = FALSE,
  label_fontface = "plain",
  label_family = "",
  gene_label_fontface = "plain",
  gene_label_family = "",
  build = 38,
  verbose = NULL,
  show_legend = TRUE,
  label_alpha = 1,
  gene_label_size = NULL,
  vline_color = "grey",
  vline_linetype = "dashed",
  vline_alpha = 1,
  vline_size = 0.5,
  log_trans_p = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locuszoom_+3A_df">df</code></td>
<td>
<p>Dataframe or a list of dataframes (required columns are <code>CHROM,POS,P</code>), in upper- or lowercase) of association results.</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_annotate">annotate</code></td>
<td>
<p>A number (p-value). Display annotation for variants with p-values below this threshold</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_ntop">ntop</code></td>
<td>
<p>An integer, number of datasets (GWAS results) to show on the top plot</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_xmin">xmin</code>, <code id="locuszoom_+3A_xmax">xmax</code></td>
<td>
<p>Integer, setting the chromosomal range to display on the x-axis</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_size">size</code></td>
<td>
<p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_shape">shape</code></td>
<td>
<p>A number of a vector of numbers setting the shape of the plotted points</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_alpha">alpha</code></td>
<td>
<p>A number or a vector of numbers setting the transparency of the plotted points</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_label_size">label_size</code></td>
<td>
<p>An number to set the size of the plot labels (default: <code>label_size=3</code>)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_annotate_with">annotate_with</code></td>
<td>
<p>A string. Annotate the variants with either Gene_Symbol or ID (default: &quot;Gene_Symbol&quot;)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_color">color</code></td>
<td>
<p>A string or a vector of strings, for setting the color of the datapoints on the plot</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A number, size of the x and y axes tick labels (default: 12)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A number, size of the x and y title labels (default: 12)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_title_text_size">title_text_size</code></td>
<td>
<p>A number, size of the plot title (default: 13)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_show_genes">show_genes</code></td>
<td>
<p>A logical scalar, show genes instead of exons (default show_genes=FALSE)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_show_overview">show_overview</code></td>
<td>
<p>A logical scalar, shows/hides the overview plot (default= TRUE)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_show_exons">show_exons</code></td>
<td>
<p>Deprecated : A logical scalar, show exons instead of genes (default show_exons=FALSE)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_max_genes">max_genes</code></td>
<td>
<p>An integer, only label the genes if they are fewer than max_genes (default values is 200).</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_sign_thresh">sign_thresh</code></td>
<td>
<p>A number or vector of numbers, setting the horizontal significance threshold (default: <code>sign_thresh=5e-8</code>). Set to NULL to hide the significance threshold.</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_sign_thresh_color">sign_thresh_color</code></td>
<td>
<p>A string or vector of strings to set the color/s of the significance threshold/s</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_sign_thresh_label_size">sign_thresh_label_size</code></td>
<td>
<p>A number setting the text size of the label for the significance thresholds (default text size is 3.5)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_ymin">ymin</code>, <code id="locuszoom_+3A_ymax">ymax</code></td>
<td>
<p>Integer, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>A logical scalar, if TRUE, only protein coding genes are used for annotation</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_region_size">region_size</code></td>
<td>
<p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_gene_padding">gene_padding</code></td>
<td>
<p>An integer representing size of the region around the gene, if the gene argument was used (default = 100000)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_angle">angle</code></td>
<td>
<p>A number, the angle of the text label</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>A number, size of the legend title</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>A number, size of the legend text</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_nudge_x">nudge_x</code></td>
<td>
<p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_nudge_y">nudge_y</code></td>
<td>
<p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_rsids">rsids</code></td>
<td>
<p>A string (rsid) or vector of strings to highlight on the plot, e.g. <code>rsids=c("rs1234, rs45898")</code></p>
</td></tr>
<tr><td><code id="locuszoom_+3A_variant">variant</code></td>
<td>
<p>A string representing the variant to zoom in on. Can be either an rsid, or a dataframe (with the columns CHROM,POS,P)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_rsids_color">rsids_color</code></td>
<td>
<p>A string, the color of the variants in variants_id (default color is red)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_legend_name">legend_name</code></td>
<td>
<p>A string, use to change the name of the legend (default: None)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_legend_position">legend_position</code></td>
<td>
<p>A string, top,bottom,left or right</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_chr">chr</code></td>
<td>
<p>A string or integer, the chromosome to plot (i.e. chr15), only required if the input dataframe contains results from more than one chromosome</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_vline">vline</code></td>
<td>
<p>A number or vector of numbers to add a vertical line to the plot at a specific chromosomal position, e.g <code>vline=204000066</code>. Multiple values can be provided in a vector, e.g  <code>vline=c(204000066,100500188)</code></p>
</td></tr>
<tr><td><code id="locuszoom_+3A_show_gene_names">show_gene_names</code></td>
<td>
<p>A logical scalar, if set to TRUE, gene names are shown even though they exceed the max_genes count</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_legend_labels">legend_labels</code></td>
<td>
<p>A string or vector of strings representing legend labels for the input datasets</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_gene">gene</code></td>
<td>
<p>A string representing the gene to zoom in on (e.g. gene=FTO)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_title">title</code></td>
<td>
<p>A string to set the plot title</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_label_color">label_color</code></td>
<td>
<p>A string or a vector of strings. To change the color of the gene or variant labels</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_region">region</code></td>
<td>
<p>A string representing a genetic region, e.g. chr1:67038906-67359979</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_scale">scale</code></td>
<td>
<p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_rsids_with_vline">rsids_with_vline</code></td>
<td>
<p>A string (rsid) or vector of strings to highlight on the plot with their rsids and vertical lines further highlighting their positions</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_annotate_with_vline">annotate_with_vline</code></td>
<td>
<p>A number (p-value). Display annotation and vertical lines for variants with p-values below this threshold</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_sign_thresh_size">sign_thresh_size</code></td>
<td>
<p>A number, sets the size of the horizontal significance threshold line (default : 1)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_unit_main">unit_main</code></td>
<td>
<p>the height unit of the main plot (default = 7)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_unit_gene">unit_gene</code></td>
<td>
<p>the height unit of the gene plot (default= 2 )</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_gene_color">gene_color</code></td>
<td>
<p>A string representing a color, can be used to change the color of the genes/exons on the geneplot</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_segment.size">segment.size</code></td>
<td>
<p>line segment color (ggrepel argument)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_segment.color">segment.color</code></td>
<td>
<p>line segment thickness (ggrepel argument)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_segment.linetype">segment.linetype</code></td>
<td>
<p>line segment solid, dashed, etc.(ggrepel argument)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_show_gene_legend">show_gene_legend</code></td>
<td>
<p>A logical scalar, set to FALSE to hide the gene legend (default value is TRUE)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_max.overlaps">max.overlaps</code></td>
<td>
<p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_extract_plots">extract_plots</code></td>
<td>
<p>Logical, FALSE by default. Set to TRUE to extract the three plots separately in a list</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_label_fontface">label_fontface</code></td>
<td>
<p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_label_family">label_family</code></td>
<td>
<p>A string or a vector of strings. Label font name (default ggrepel argument is &quot;&quot;)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_gene_label_fontface">gene_label_fontface</code></td>
<td>
<p>Gene label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_gene_label_family">gene_label_family</code></td>
<td>
<p>Gene label font name (default ggrepel argument is &quot;&quot;)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_build">build</code></td>
<td>
<p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_verbose">verbose</code></td>
<td>
<p>Logical, set to FALSE to get suppress printed information</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_show_legend">show_legend</code></td>
<td>
<p>A logical scalar, set to FALSE to hide the legend (default : TRUE)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_label_alpha">label_alpha</code></td>
<td>
<p>An number or vector of numbers to set the transparency of the plot labels (default: <code>label_alpha=1</code>)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_gene_label_size">gene_label_size</code></td>
<td>
<p>A number setting the size of the gene labels shown at the bottom of the plot</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_vline_color">vline_color</code></td>
<td>
<p>A string. The color of added vertical line/s (default: grey)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_vline_linetype">vline_linetype</code></td>
<td>
<p>A string. The linetype of added vertical line/s (default : dashed)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_vline_alpha">vline_alpha</code></td>
<td>
<p>A number. The alpha of added vertical line/s (default : 1)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_vline_size">vline_size</code></td>
<td>
<p>A number.The size of added vertical line/s (default : 0.5)</p>
</td></tr>
<tr><td><code id="locuszoom_+3A_log_trans_p">log_trans_p</code></td>
<td>
<p>A logical scalar (default: TRUE). By default the p-values in the input datasets are log transformed using -log10. Set this argument to FALSE if the p-values in the datasets have already been log transformed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots using egg (https://cran.r-project.org/web/packages/egg/vignettes/Ecosystem.html)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
locuszoom(R2_CD_UKBB)

## End(Not run)
</code></pre>

<hr>
<h2 id='manhattan'>Create a Manhattan plot</h2><span id='topic+manhattan'></span>

<h3>Description</h3>

<p><code>manhattan()</code> displays association results for the entire genome on a Manhattan plot.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase)
</p>
<p>All other input parameters are optional
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan(
  df,
  ntop = 4,
  title = "",
  annotate = NULL,
  color = get_topr_colors(),
  sign_thresh = 5e-08,
  sign_thresh_color = "red",
  sign_thresh_label_size = 3.5,
  label_size = 3.5,
  size = 0.8,
  shape = 19,
  alpha = 1,
  highlight_genes_color = "darkred",
  highlight_genes_ypos = 1.5,
  axis_text_size = 12,
  axis_title_size = 14,
  title_text_size = 15,
  legend_title_size = 13,
  legend_text_size = 12,
  protein_coding_only = TRUE,
  angle = 0,
  legend_labels = NULL,
  chr = NULL,
  annotate_with = "Gene_Symbol",
  region_size = 2e+07,
  legend_name = NULL,
  legend_position = "bottom",
  nudge_x = 0.1,
  nudge_y = 0.7,
  xmin = NULL,
  xmax = NULL,
  ymin = NULL,
  ymax = NULL,
  highlight_genes = NULL,
  label_color = NULL,
  legend_nrow = NULL,
  gene_label_size = NULL,
  gene_label_angle = 0,
  scale = 1,
  show_legend = TRUE,
  sign_thresh_linetype = "dashed",
  sign_thresh_size = 0.5,
  rsids = NULL,
  rsids_color = NULL,
  rsids_with_vline = NULL,
  annotate_with_vline = NULL,
  shades_color = NULL,
  shades_alpha = 0.5,
  segment.size = 0.2,
  segment.color = "black",
  segment.linetype = "dashed",
  max.overlaps = 10,
  label_fontface = "plain",
  label_family = "",
  gene_label_fontface = "plain",
  gene_label_family = "",
  build = 38,
  verbose = NULL,
  label_alpha = 1,
  shades_line_alpha = 1,
  vline = NULL,
  vline_color = "grey",
  vline_linetype = "dashed",
  vline_alpha = 1,
  vline_size = 0.5,
  region = NULL,
  theme_grey = FALSE,
  xaxis_label = "Chromosome",
  use_shades = FALSE,
  even_no_chr_lightness = 0.8,
  get_chr_lengths_from_data = TRUE,
  log_trans_p = TRUE,
  chr_ticknames = NULL,
  show_all_chrticks = FALSE,
  hide_chrticks_from_pos = 17,
  hide_chrticks_to_pos = NULL,
  hide_every_nth_chrtick = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manhattan_+3A_df">df</code></td>
<td>
<p>Dataframe or a list of dataframes (required columns are <code>CHROM,POS,P</code>), in upper- or lowercase) of association results.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_ntop">ntop</code></td>
<td>
<p>An integer, number of datasets (GWAS results) to show on the top plot</p>
</td></tr>
<tr><td><code id="manhattan_+3A_title">title</code></td>
<td>
<p>A string to set the plot title</p>
</td></tr>
<tr><td><code id="manhattan_+3A_annotate">annotate</code></td>
<td>
<p>A number (p-value). Display annotation for variants with p-values below this threshold</p>
</td></tr>
<tr><td><code id="manhattan_+3A_color">color</code></td>
<td>
<p>A string or a vector of strings, for setting the color of the datapoints on the plot</p>
</td></tr>
<tr><td><code id="manhattan_+3A_sign_thresh">sign_thresh</code></td>
<td>
<p>A number or vector of numbers, setting the horizontal significance threshold (default: <code>sign_thresh=5e-8</code>). Set to NULL to hide the significance threshold.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_sign_thresh_color">sign_thresh_color</code></td>
<td>
<p>A string or vector of strings to set the color/s of the significance threshold/s</p>
</td></tr>
<tr><td><code id="manhattan_+3A_sign_thresh_label_size">sign_thresh_label_size</code></td>
<td>
<p>A number setting the text size of the label for the significance thresholds (default text size is 3.5)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_label_size">label_size</code></td>
<td>
<p>An number to set the size of the plot labels (default: <code>label_size=3</code>)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_size">size</code></td>
<td>
<p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_shape">shape</code></td>
<td>
<p>A number of a vector of numbers setting the shape of the plotted points</p>
</td></tr>
<tr><td><code id="manhattan_+3A_alpha">alpha</code></td>
<td>
<p>A number or a vector of numbers setting the transparency of the plotted points</p>
</td></tr>
<tr><td><code id="manhattan_+3A_highlight_genes_color">highlight_genes_color</code></td>
<td>
<p>A string, color for the highlighted genes (default: darkred)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_highlight_genes_ypos">highlight_genes_ypos</code></td>
<td>
<p>An integer, controlling where on the y-axis the highlighted genes are placed (default value is 1)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A number, size of the x and y axes tick labels (default: 12)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A number, size of the x and y title labels (default: 12)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_title_text_size">title_text_size</code></td>
<td>
<p>A number, size of the plot title (default: 13)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>A number, size of the legend title</p>
</td></tr>
<tr><td><code id="manhattan_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>A number, size of the legend text</p>
</td></tr>
<tr><td><code id="manhattan_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>A logical scalar, if TRUE, only protein coding genes are used for annotation</p>
</td></tr>
<tr><td><code id="manhattan_+3A_angle">angle</code></td>
<td>
<p>A number, the angle of the text label</p>
</td></tr>
<tr><td><code id="manhattan_+3A_legend_labels">legend_labels</code></td>
<td>
<p>A string or vector of strings representing legend labels for the input datasets</p>
</td></tr>
<tr><td><code id="manhattan_+3A_chr">chr</code></td>
<td>
<p>A string or integer, the chromosome to plot (i.e. chr15), only required if the input dataframe contains results from more than one chromosome</p>
</td></tr>
<tr><td><code id="manhattan_+3A_annotate_with">annotate_with</code></td>
<td>
<p>A string. Annotate the variants with either Gene_Symbol or ID (default: &quot;Gene_Symbol&quot;)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_region_size">region_size</code></td>
<td>
<p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_legend_name">legend_name</code></td>
<td>
<p>A string, use to change the name of the legend (default: None)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_legend_position">legend_position</code></td>
<td>
<p>A string, top,bottom,left or right</p>
</td></tr>
<tr><td><code id="manhattan_+3A_nudge_x">nudge_x</code></td>
<td>
<p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_nudge_y">nudge_y</code></td>
<td>
<p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_xmin">xmin</code>, <code id="manhattan_+3A_xmax">xmax</code></td>
<td>
<p>Integer, setting the chromosomal range to display on the x-axis</p>
</td></tr>
<tr><td><code id="manhattan_+3A_ymin">ymin</code>, <code id="manhattan_+3A_ymax">ymax</code></td>
<td>
<p>Integer, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_highlight_genes">highlight_genes</code></td>
<td>
<p>A string or vector of strings, gene or genes to highlight at the bottom of the plot</p>
</td></tr>
<tr><td><code id="manhattan_+3A_label_color">label_color</code></td>
<td>
<p>A string or a vector of strings. To change the color of the gene or variant labels</p>
</td></tr>
<tr><td><code id="manhattan_+3A_legend_nrow">legend_nrow</code></td>
<td>
<p>An integer, sets the number of rows allowed for the legend labels</p>
</td></tr>
<tr><td><code id="manhattan_+3A_gene_label_size">gene_label_size</code></td>
<td>
<p>A number setting the size of the gene labels shown at the bottom of the plot</p>
</td></tr>
<tr><td><code id="manhattan_+3A_gene_label_angle">gene_label_angle</code></td>
<td>
<p>A number setting the angle of the gene label shown at the bottom of the plot (default: 0)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_scale">scale</code></td>
<td>
<p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_show_legend">show_legend</code></td>
<td>
<p>A logical scalar, set to FALSE to hide the legend (default : TRUE)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_sign_thresh_linetype">sign_thresh_linetype</code></td>
<td>
<p>A string, the line-type of the horizontal significance threshold (default : dashed)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_sign_thresh_size">sign_thresh_size</code></td>
<td>
<p>A number, sets the size of the horizontal significance threshold line (default : 1)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_rsids">rsids</code></td>
<td>
<p>A string (rsid) or vector of strings to highlight on the plot, e.g. <code>rsids=c("rs1234, rs45898")</code></p>
</td></tr>
<tr><td><code id="manhattan_+3A_rsids_color">rsids_color</code></td>
<td>
<p>A string, the color of the variants in variants_id (default color is red)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_rsids_with_vline">rsids_with_vline</code></td>
<td>
<p>A string (rsid) or vector of strings to highlight on the plot with their rsids and vertical lines further highlighting their positions</p>
</td></tr>
<tr><td><code id="manhattan_+3A_annotate_with_vline">annotate_with_vline</code></td>
<td>
<p>A number (p-value). Display annotation and vertical lines for variants with p-values below this threshold</p>
</td></tr>
<tr><td><code id="manhattan_+3A_shades_color">shades_color</code></td>
<td>
<p>The color of the rectangles (shades) representing the different chromosomes on the Manhattan plot</p>
</td></tr>
<tr><td><code id="manhattan_+3A_shades_alpha">shades_alpha</code></td>
<td>
<p>The transparency (alpha) of the rectangles (shades)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_segment.size">segment.size</code></td>
<td>
<p>line segment color (ggrepel argument)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_segment.color">segment.color</code></td>
<td>
<p>line segment thickness (ggrepel argument)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_segment.linetype">segment.linetype</code></td>
<td>
<p>line segment solid, dashed, etc.(ggrepel argument)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_max.overlaps">max.overlaps</code></td>
<td>
<p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_label_fontface">label_fontface</code></td>
<td>
<p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_label_family">label_family</code></td>
<td>
<p>A string or a vector of strings. Label font name (default ggrepel argument is &quot;&quot;)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_gene_label_fontface">gene_label_fontface</code></td>
<td>
<p>Gene label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_gene_label_family">gene_label_family</code></td>
<td>
<p>Gene label font name (default ggrepel argument is &quot;&quot;)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_build">build</code></td>
<td>
<p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td></tr>
<tr><td><code id="manhattan_+3A_verbose">verbose</code></td>
<td>
<p>A logical scalar (default: NULL). Set to FALSE to suppress printed messages</p>
</td></tr>
<tr><td><code id="manhattan_+3A_label_alpha">label_alpha</code></td>
<td>
<p>An number or vector of numbers to set the transparency of the plot labels (default: <code>label_alpha=1</code>)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_shades_line_alpha">shades_line_alpha</code></td>
<td>
<p>The transparency (alpha) of the lines around the rectangles (shades)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_vline">vline</code></td>
<td>
<p>A number or vector of numbers to add a vertical line to the plot at a specific chromosomal position, e.g <code>vline="chr1:204000066"</code>. Multiple values can be provided in a vector, e.g  <code>vline=c("chr1:204000066","chr5:100500188")</code></p>
</td></tr>
<tr><td><code id="manhattan_+3A_vline_color">vline_color</code></td>
<td>
<p>A string. The color of added vertical line/s (default: grey)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_vline_linetype">vline_linetype</code></td>
<td>
<p>A string. The linetype of added vertical line/s (default : dashed)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_vline_alpha">vline_alpha</code></td>
<td>
<p>A number. The alpha of added vertical line/s (default : 1)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_vline_size">vline_size</code></td>
<td>
<p>A number.The size of added vertical line/s (default : 0.5)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_region">region</code></td>
<td>
<p>A string representing a genetic region, e.g. chr1:67038906-67359979</p>
</td></tr>
<tr><td><code id="manhattan_+3A_theme_grey">theme_grey</code></td>
<td>
<p>A logical scalar (default: FALSE). Use gray rectangles (instead of white to distinguish between chromosomes)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_xaxis_label">xaxis_label</code></td>
<td>
<p>A string. The label for the x-axis (default: Chromosome)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_use_shades">use_shades</code></td>
<td>
<p>A logical scalar (default: FALSE). Use shades/rectangles to distinguish between chromosomes</p>
</td></tr>
<tr><td><code id="manhattan_+3A_even_no_chr_lightness">even_no_chr_lightness</code></td>
<td>
<p>Lightness value for even numbered chromosomes. A number or vector of numbers between 0 and 1 (default: 0.8). If set to 0.5, the same color as shown for odd numbered chromosomes is displayed. A value below 0.5 will result in a darker color displayed for even numbered chromosomes, whereas a value above 0.5 results in a lighter color.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_get_chr_lengths_from_data">get_chr_lengths_from_data</code></td>
<td>
<p>A logical scalar (default: TRUE). If set to FALSE, use the inbuilt chromosome lengths (from hg38), instead of chromosome lengths based on the max position for each chromosome in the input dataset/s.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_log_trans_p">log_trans_p</code></td>
<td>
<p>A logical scalar (default: TRUE). By default the p-values in the input datasets are log transformed using -log10. Set this argument to FALSE if the p-values in the datasets have already been log transformed.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_chr_ticknames">chr_ticknames</code></td>
<td>
<p>A vector containing the chromosome names displayed on the x-axis. If NULL, the following format is used: chr_ticknames &lt;- c(1:16, &rdquo;,18, &rdquo;,20, &rdquo;,22, 'X')</p>
</td></tr>
<tr><td><code id="manhattan_+3A_show_all_chrticks">show_all_chrticks</code></td>
<td>
<p>A logical scalar (default : FALSE). Set to TRUE to show all the chromosome names on the ticks on the x-axis</p>
</td></tr>
<tr><td><code id="manhattan_+3A_hide_chrticks_from_pos">hide_chrticks_from_pos</code></td>
<td>
<p>A number (default: 17). Hide every nth chromosome name on the x-axis FROM this position (chromosome number)</p>
</td></tr>
<tr><td><code id="manhattan_+3A_hide_chrticks_to_pos">hide_chrticks_to_pos</code></td>
<td>
<p>A number (default: NULL). Hide every nth chromosome name on the x-axis TO this position (chromosome number). When NULL this variable will be set to the number of numeric chromosomes in the input dataset.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_hide_every_nth_chrtick">hide_every_nth_chrtick</code></td>
<td>
<p>A number (default: 2). Hide every nth chromosome tick on the x-axis (from the hide_chr_ticks_from_pos to the hide_chr_ticks_to_pos).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
manhattan(CD_UKBB)

## End(Not run)
</code></pre>

<hr>
<h2 id='match_alleles'>Match the variants in the snpset by their alleles</h2><span id='topic+match_alleles'></span>

<h3>Description</h3>

<p><code>match_alleles()</code>
</p>
<p>This method is deprecated and will be removed in future versions. use <code><a href="#topic+match_by_alleles">match_by_alleles</a></code> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_alleles(df, verbose = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_alleles_+3A_df">df</code></td>
<td>
<p>A dataframe that is in the snpset format (like returned by the get_snpset() function)</p>
</td></tr>
<tr><td><code id="match_alleles_+3A_verbose">verbose</code></td>
<td>
<p>A logical scalar (default: FALSE). Assign to TRUE to get information on which alleles are matched and which are not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input dataframe containing only those variants whith matched alleles in the snpset
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
match_alleles(df)

## End(Not run)

</code></pre>

<hr>
<h2 id='match_by_alleles'>Match the variants in the snpset by their alleles</h2><span id='topic+match_by_alleles'></span>

<h3>Description</h3>

<p><code>match_by_alleles()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_by_alleles(df, verbose = NULL, show_full_output = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_by_alleles_+3A_df">df</code></td>
<td>
<p>A dataframe that is in the snpset format (like returned by the <code><a href="#topic+get_snpset">get_snpset</a></code> function)</p>
</td></tr>
<tr><td><code id="match_by_alleles_+3A_verbose">verbose</code></td>
<td>
<p>A logical scalar (default: FALSE). Assign to TRUE to get information on which alleles are matched and which are not.</p>
</td></tr>
<tr><td><code id="match_by_alleles_+3A_show_full_output">show_full_output</code></td>
<td>
<p>A logical scalar (default:FALSE). Assign to TRUE to show the full output from this function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input dataframe containing only those variants with matched alleles in the snpset
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
CD_UKBB_lead_snps &lt;- get_lead_snps(CD_UKBB)
snpset &lt;- get_snpset(CD_UKBB_lead_snps, CD_FINNGEN)
match_by_alleles(snpset$found)

## End(Not run)

</code></pre>

<hr>
<h2 id='match_by_pos'>Get variants that overlap between two datasets</h2><span id='topic+match_by_pos'></span>

<h3>Description</h3>

<p><code>match_by_pos()</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_by_pos(df1, df2, verbose = NULL, show_full_output = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_by_pos_+3A_df1">df1</code></td>
<td>
<p>A dataframe of variants, has to contain CHROM and POS</p>
</td></tr>
<tr><td><code id="match_by_pos_+3A_df2">df2</code></td>
<td>
<p>A dataframe of variants, has to contain CHROM and POS</p>
</td></tr>
<tr><td><code id="match_by_pos_+3A_verbose">verbose</code></td>
<td>
<p>A logical scalar (default: FALSE). Assign to TRUE to get information on which alleles are matched and which are not.</p>
</td></tr>
<tr><td><code id="match_by_pos_+3A_show_full_output">show_full_output</code></td>
<td>
<p>A logical scalar (default:FALSE). Assign to TRUE to show the full output from this function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two dataframes, one of overlapping snps and the other snps not found in the second input dataset
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
CD_UKBB_index_snps &lt;- get_lead_snps(CD_UKBB)
match_by_pos(CD_UKBB_index_snps, CD_FINNGEN)

## End(Not run)

</code></pre>

<hr>
<h2 id='qqtopr'>Create a quantile quantile (QQ) plot</h2><span id='topic+qqtopr'></span>

<h3>Description</h3>

<p><code>qqtopr()</code> displays QQ plots for association data.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqtopr(
  dat,
  scale = 1,
  n_variants = 0,
  breaks = 15,
  title = NULL,
  color = get_topr_colors(),
  size = 1,
  legend_name = "",
  legend_position = "right",
  legend_labels = NULL,
  axis_text_size = 11,
  axis_title_size = 12,
  title_text_size = 13,
  legend_title_size = 12,
  legend_text_size = 12,
  verbose = NULL,
  diagonal_line_color = "#808080"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qqtopr_+3A_dat">dat</code></td>
<td>
<p>Dataframe or a list of dataframes (required columns is <code>P</code>)) of association results.</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_scale">scale</code></td>
<td>
<p>An integer, plot elements scale, default: 1</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_n_variants">n_variants</code></td>
<td>
<p>An integer, total number of variants used in the study</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_breaks">breaks</code></td>
<td>
<p>A number setting the breaks for the axes</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_title">title</code></td>
<td>
<p>A string to set the plot title</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_color">color</code></td>
<td>
<p>A string or vector of strings setting the color's for the input datasets</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_size">size</code></td>
<td>
<p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_legend_name">legend_name</code></td>
<td>
<p>A string, use to change the name of the legend (default: None)</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_legend_position">legend_position</code></td>
<td>
<p>A string, top,bottom,left or right</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_legend_labels">legend_labels</code></td>
<td>
<p>A string or vector of strings representing legend labels for the input datasets</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A number, size of the x and y axes tick labels (default: 12)</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A number, size of the x and y title labels (default: 12)</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_title_text_size">title_text_size</code></td>
<td>
<p>A number, size of the plot title (default: 13)</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>A number, size of the legend title</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>A number, size of the legend text</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_verbose">verbose</code></td>
<td>
<p>A logical scalar (default: NULL). Set to FALSE to suppress printed messages</p>
</td></tr>
<tr><td><code id="qqtopr_+3A_diagonal_line_color">diagonal_line_color</code></td>
<td>
<p>A string setting the color of the diagonal line on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
qqtopr(CD_UKBB)

## End(Not run)

</code></pre>

<hr>
<h2 id='R2_CD_UKBB'>Example dataset including the R2 column for the locuszoom plot function</h2><span id='topic+R2_CD_UKBB'></span>

<h3>Description</h3>

<p>The dataset is a subset of CD_UKBB and only includes variants above and near the IL23R gene on chromosome 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>R2_CD_UKBB
</code></pre>


<h3>Format</h3>

<p>A data frame  with 329 rows and 5 variables:
</p>

<dl>
<dt>CHROM</dt><dd><p>Chromosome, written as for example chr1 or 1</p>
</dd>
<dt>POS</dt><dd><p>genetic position of the variant</p>
</dd>
<dt>ID</dt><dd><p>Variant identifier, e.g. rsid </p>
</dd>
<dt>P</dt><dd><p>P-value from Plink run, additive model, regression model GLM_FIRTH</p>
</dd>
<dt>R2</dt><dd><p>variant correlation (r^2)</p>
</dd>
</dl>



<h3>Source</h3>

<p>A subset of the CD_UKBB dataset
</p>

<hr>
<h2 id='regionplot'>Create a regionplot</h2><span id='topic+regionplot'></span>

<h3>Description</h3>

<p><code>regionplot()</code> displays the association results for a smaller genetic regions within one chromosome.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase) and either a variant ID, gene name or the genetic region represented as a chromosome together with start and stop positions (either as a single string or as three separate arguments).
</p>
<p>All other input parameters are optional
</p>


<h3>Usage</h3>

<pre><code class='language-R'>regionplot(
  df,
  ntop = 10,
  annotate = NULL,
  xmin = 0,
  size = 2,
  shape = 19,
  alpha = 1,
  label_size = 4,
  annotate_with = "ID",
  color = get_topr_colors(),
  axis_text_size = 11,
  axis_title_size = 12,
  title_text_size = 13,
  show_genes = NULL,
  show_overview = TRUE,
  show_exons = NULL,
  max_genes = 200,
  sign_thresh = 5e-08,
  sign_thresh_color = "red",
  sign_thresh_label_size = 3.5,
  xmax = NULL,
  ymin = NULL,
  ymax = NULL,
  protein_coding_only = FALSE,
  region_size = 1e+06,
  gene_padding = 1e+05,
  angle = 0,
  legend_title_size = 12,
  legend_text_size = 11,
  nudge_x = 0.01,
  nudge_y = 0.01,
  rsids = NULL,
  variant = NULL,
  rsids_color = NULL,
  legend_name = "",
  legend_position = "right",
  chr = NULL,
  vline = NULL,
  show_gene_names = NULL,
  legend_labels = NULL,
  gene = NULL,
  title = NULL,
  label_color = NULL,
  locuszoomplot = FALSE,
  region = NULL,
  legend_nrow = NULL,
  gene_label_size = NULL,
  scale = 1,
  show_legend = TRUE,
  sign_thresh_linetype = "dashed",
  sign_thresh_size = 0.5,
  rsids_with_vline = NULL,
  annotate_with_vline = NULL,
  show_gene_legend = TRUE,
  unit_main = 7,
  unit_gene = 2,
  unit_overview = 1.25,
  verbose = NULL,
  gene_color = NULL,
  segment.size = 0.2,
  segment.color = "black",
  segment.linetype = "solid",
  max.overlaps = 10,
  unit_ratios = NULL,
  extract_plots = FALSE,
  label_fontface = "plain",
  label_family = "",
  gene_label_fontface = "plain",
  gene_label_family = "",
  build = 38,
  label_alpha = 1,
  vline_color = "grey",
  vline_linetype = "dashed",
  vline_alpha = 1,
  vline_size = 0.5,
  log_trans_p = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="regionplot_+3A_df">df</code></td>
<td>
<p>Dataframe or a list of dataframes (required columns are <code>CHROM,POS,P</code>), in upper- or lowercase) of association results.</p>
</td></tr>
<tr><td><code id="regionplot_+3A_ntop">ntop</code></td>
<td>
<p>An integer, number of datasets (GWAS results) to show on the top plot</p>
</td></tr>
<tr><td><code id="regionplot_+3A_annotate">annotate</code></td>
<td>
<p>A number (p-value). Display annotation for variants with p-values below this threshold</p>
</td></tr>
<tr><td><code id="regionplot_+3A_xmin">xmin</code>, <code id="regionplot_+3A_xmax">xmax</code></td>
<td>
<p>Integer, setting the chromosomal range to display on the x-axis</p>
</td></tr>
<tr><td><code id="regionplot_+3A_size">size</code></td>
<td>
<p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_shape">shape</code></td>
<td>
<p>A number of a vector of numbers setting the shape of the plotted points</p>
</td></tr>
<tr><td><code id="regionplot_+3A_alpha">alpha</code></td>
<td>
<p>A number or a vector of numbers setting the transparency of the plotted points</p>
</td></tr>
<tr><td><code id="regionplot_+3A_label_size">label_size</code></td>
<td>
<p>An number to set the size of the plot labels (default: <code>label_size=3</code>)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_annotate_with">annotate_with</code></td>
<td>
<p>A string. Annotate the variants with either Gene_Symbol or ID (default: &quot;Gene_Symbol&quot;)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_color">color</code></td>
<td>
<p>A string or a vector of strings, for setting the color of the datapoints on the plot</p>
</td></tr>
<tr><td><code id="regionplot_+3A_axis_text_size">axis_text_size</code></td>
<td>
<p>A number, size of the x and y axes tick labels (default: 12)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>A number, size of the x and y title labels (default: 12)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_title_text_size">title_text_size</code></td>
<td>
<p>A number, size of the plot title (default: 13)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_show_genes">show_genes</code></td>
<td>
<p>A logical scalar, show genes instead of exons (default show_genes=FALSE)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_show_overview">show_overview</code></td>
<td>
<p>A logical scalar, shows/hides the overview plot (default= TRUE)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_show_exons">show_exons</code></td>
<td>
<p>Deprecated : A logical scalar, show exons instead of genes (default show_exons=FALSE)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_max_genes">max_genes</code></td>
<td>
<p>An integer, only label the genes if they are fewer than max_genes (default values is 200).</p>
</td></tr>
<tr><td><code id="regionplot_+3A_sign_thresh">sign_thresh</code></td>
<td>
<p>A number or vector of numbers, setting the horizontal significance threshold (default: <code>sign_thresh=5e-8</code>). Set to NULL to hide the significance threshold.</p>
</td></tr>
<tr><td><code id="regionplot_+3A_sign_thresh_color">sign_thresh_color</code></td>
<td>
<p>A string or vector of strings to set the color/s of the significance threshold/s</p>
</td></tr>
<tr><td><code id="regionplot_+3A_sign_thresh_label_size">sign_thresh_label_size</code></td>
<td>
<p>A number setting the text size of the label for the significance thresholds (default text size is 3.5)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_ymin">ymin</code>, <code id="regionplot_+3A_ymax">ymax</code></td>
<td>
<p>Integer, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_protein_coding_only">protein_coding_only</code></td>
<td>
<p>A logical scalar, if TRUE, only protein coding genes are used for annotation</p>
</td></tr>
<tr><td><code id="regionplot_+3A_region_size">region_size</code></td>
<td>
<p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td></tr>
<tr><td><code id="regionplot_+3A_gene_padding">gene_padding</code></td>
<td>
<p>An integer representing size of the region around the gene, if the gene argument was used (default = 100000)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_angle">angle</code></td>
<td>
<p>A number, the angle of the text label</p>
</td></tr>
<tr><td><code id="regionplot_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>A number, size of the legend title</p>
</td></tr>
<tr><td><code id="regionplot_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>A number, size of the legend text</p>
</td></tr>
<tr><td><code id="regionplot_+3A_nudge_x">nudge_x</code></td>
<td>
<p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_nudge_y">nudge_y</code></td>
<td>
<p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_rsids">rsids</code></td>
<td>
<p>A string (rsid) or vector of strings to highlight on the plot, e.g. <code>rsids=c("rs1234, rs45898")</code></p>
</td></tr>
<tr><td><code id="regionplot_+3A_variant">variant</code></td>
<td>
<p>A string representing the variant to zoom in on. Can be either an rsid, or a dataframe (with the columns CHROM,POS,P)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_rsids_color">rsids_color</code></td>
<td>
<p>A string, the color of the variants in variants_id (default color is red)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_legend_name">legend_name</code></td>
<td>
<p>A string, use to change the name of the legend (default: None)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_legend_position">legend_position</code></td>
<td>
<p>A string, top,bottom,left or right</p>
</td></tr>
<tr><td><code id="regionplot_+3A_chr">chr</code></td>
<td>
<p>A string or integer, the chromosome to plot (i.e. chr15), only required if the input dataframe contains results from more than one chromosome</p>
</td></tr>
<tr><td><code id="regionplot_+3A_vline">vline</code></td>
<td>
<p>A number or vector of numbers to add a vertical line to the plot at a specific chromosomal position, e.g <code>vline=204000066</code>. Multiple values can be provided in a vector, e.g  <code>vline=c(204000066,100500188)</code></p>
</td></tr>
<tr><td><code id="regionplot_+3A_show_gene_names">show_gene_names</code></td>
<td>
<p>A logical scalar, if set to TRUE, gene names are shown even though they exceed the max_genes count</p>
</td></tr>
<tr><td><code id="regionplot_+3A_legend_labels">legend_labels</code></td>
<td>
<p>A string or vector of strings representing legend labels for the input datasets</p>
</td></tr>
<tr><td><code id="regionplot_+3A_gene">gene</code></td>
<td>
<p>A string representing the gene to zoom in on (e.g. gene=FTO)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_title">title</code></td>
<td>
<p>A string to set the plot title</p>
</td></tr>
<tr><td><code id="regionplot_+3A_label_color">label_color</code></td>
<td>
<p>A string or a vector of strings. To change the color of the gene or variant labels</p>
</td></tr>
<tr><td><code id="regionplot_+3A_locuszoomplot">locuszoomplot</code></td>
<td>
<p>A logical scalar set to FALSE. Only set to TRUE by calling the locuszoom function</p>
</td></tr>
<tr><td><code id="regionplot_+3A_region">region</code></td>
<td>
<p>A string representing a genetic region, e.g. chr1:67038906-67359979</p>
</td></tr>
<tr><td><code id="regionplot_+3A_legend_nrow">legend_nrow</code></td>
<td>
<p>An integer, sets the number of rows allowed for the legend labels</p>
</td></tr>
<tr><td><code id="regionplot_+3A_gene_label_size">gene_label_size</code></td>
<td>
<p>A number setting the size of the gene labels shown at the bottom of the plot</p>
</td></tr>
<tr><td><code id="regionplot_+3A_scale">scale</code></td>
<td>
<p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_show_legend">show_legend</code></td>
<td>
<p>A logical scalar, set to FALSE to hide the legend (default : TRUE)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_sign_thresh_linetype">sign_thresh_linetype</code></td>
<td>
<p>A string, the line-type of the horizontal significance threshold (default : dashed)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_sign_thresh_size">sign_thresh_size</code></td>
<td>
<p>A number, sets the size of the horizontal significance threshold line (default : 1)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_rsids_with_vline">rsids_with_vline</code></td>
<td>
<p>A string (rsid) or vector of strings to highlight on the plot with their rsids and vertical lines further highlighting their positions</p>
</td></tr>
<tr><td><code id="regionplot_+3A_annotate_with_vline">annotate_with_vline</code></td>
<td>
<p>A number (p-value). Display annotation and vertical lines for variants with p-values below this threshold</p>
</td></tr>
<tr><td><code id="regionplot_+3A_show_gene_legend">show_gene_legend</code></td>
<td>
<p>A logical scalar, set to FALSE to hide the gene legend (default value is TRUE)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_unit_main">unit_main</code></td>
<td>
<p>the height unit of the main plot (default = 7)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_unit_gene">unit_gene</code></td>
<td>
<p>the height unit of the gene plot (default= 2 )</p>
</td></tr>
<tr><td><code id="regionplot_+3A_unit_overview">unit_overview</code></td>
<td>
<p>the height unit of the overview plot (default = 1.25)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_verbose">verbose</code></td>
<td>
<p>Logical, set to FALSE to get suppress printed information</p>
</td></tr>
<tr><td><code id="regionplot_+3A_gene_color">gene_color</code></td>
<td>
<p>A string representing a color, can be used to change the color of the genes/exons on the geneplot</p>
</td></tr>
<tr><td><code id="regionplot_+3A_segment.size">segment.size</code></td>
<td>
<p>line segment color (ggrepel argument)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_segment.color">segment.color</code></td>
<td>
<p>line segment thickness (ggrepel argument)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_segment.linetype">segment.linetype</code></td>
<td>
<p>line segment solid, dashed, etc.(ggrepel argument)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_max.overlaps">max.overlaps</code></td>
<td>
<p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_unit_ratios">unit_ratios</code></td>
<td>
<p>A string of three numbers separated by &quot;:&quot;, for the overview, main and gene plots height ratios e.g  1.25:7:2</p>
</td></tr>
<tr><td><code id="regionplot_+3A_extract_plots">extract_plots</code></td>
<td>
<p>Logical, FALSE by default. Set to TRUE to extract the three plots separately in a list</p>
</td></tr>
<tr><td><code id="regionplot_+3A_label_fontface">label_fontface</code></td>
<td>
<p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_label_family">label_family</code></td>
<td>
<p>A string or a vector of strings. Label font name (default ggrepel argument is &quot;&quot;)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_gene_label_fontface">gene_label_fontface</code></td>
<td>
<p>Gene label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_gene_label_family">gene_label_family</code></td>
<td>
<p>Gene label font name (default ggrepel argument is &quot;&quot;)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_build">build</code></td>
<td>
<p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td></tr>
<tr><td><code id="regionplot_+3A_label_alpha">label_alpha</code></td>
<td>
<p>An number or vector of numbers to set the transparency of the plot labels (default: <code>label_alpha=1</code>)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_vline_color">vline_color</code></td>
<td>
<p>A string. The color of added vertical line/s (default: grey)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_vline_linetype">vline_linetype</code></td>
<td>
<p>A string. The linetype of added vertical line/s (default : dashed)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_vline_alpha">vline_alpha</code></td>
<td>
<p>A number. The alpha of added vertical line/s (default : 1)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_vline_size">vline_size</code></td>
<td>
<p>A number.The size of added vertical line/s (default : 0.5)</p>
</td></tr>
<tr><td><code id="regionplot_+3A_log_trans_p">log_trans_p</code></td>
<td>
<p>A logical scalar (default: TRUE). By default the p-values in the input datasets are log transformed using -log10. Set this argument to FALSE if the p-values in the datasets have already been log transformed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots within ggplotGrobs, arranged with egg::gtable_frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
regionplot(CD_UKBB, gene="IL23R")

## End(Not run)
</code></pre>

<hr>
<h2 id='topr'>topr</h2><span id='topic+topr-package'></span><span id='topic+topr'></span>

<h3>Description</h3>

<p>A package for viewing and annotating genetic association data
</p>


<h3>topr functions</h3>

<p>The main plotting functions are:
</p>

<ul>
<li> <p><code><a href="#topic+manhattan">manhattan</a></code> to create Manhattan plot of association results
</p>
</li>
<li> <p><code><a href="#topic+regionplot">regionplot</a></code> to create regional plots of association results for smaller genetic regions
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Thorhildur Juliusdottir <a href="mailto:totajuliusd@gmail.com">totajuliusd@gmail.com</a> [copyright holder]
</p>
<p>Authors:
</p>

<ul>
<li><p> Andri Stefansson <a href="mailto:andrimarstefansson@gmail.com">andrimarstefansson@gmail.com</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/totajuliusd/topr">https://github.com/totajuliusd/topr</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/totajuliusd/topr/issues">https://github.com/totajuliusd/topr/issues</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>library(topr)
# Create a manhattan plot using
manhattan(CD_UKBB)

# Create a regional plot
regionplot(CD_UKBB, gene="IL23R")

</code></pre>

<hr>
<h2 id='UC_UKBB'>UKBB Ulcerative colitis (ICD 10 code K51)</h2><span id='topic+UC_UKBB'></span>

<h3>Description</h3>

<p>Dataset retrieved from the UK biobank including of 5,452 UC cases (K51) and 481,862 controls. The dataset has been filtered on variants with P&lt;1e-03.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UC_UKBB
</code></pre>


<h3>Format</h3>

<p>A data frame  with 45,012 rows and 8 variables
</p>

<dl>
<dt>CHROM</dt><dd><p>Chromosome, written as for example chr1 or 1</p>
</dd>
<dt>POS</dt><dd><p>genetic position of the variant</p>
</dd>
<dt>REF</dt><dd><p>the reference allele</p>
</dd>
<dt>ALT</dt><dd><p>the alternative allele</p>
</dd>
<dt>ID</dt><dd><p>Variant identifier, e.g. rsid </p>
</dd>
<dt>P</dt><dd><p>P-value from Plink run, additive model, regression model GLM_FIRTH</p>
</dd>
<dt>OR</dt><dd><p>Odds Ratio</p>
</dd>
<dt>AF</dt><dd><p>Allele frequency</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ulcerative Colitis UKBB ICD10 code K51, only including variants with P&lt;1e-03
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
