<!DOCTYPE html><html><head><title>Help for package TPACData</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TPACData}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#hpa.data'><p>Gene expression data for normal human tissues from the Human Protein Atlas (HPA). This data was specially processed by the HPA group as FPKM values (file &quot;HPA.normal.FPKM.GDCpipeline.csv&quot;) using a pipeline similar to that employed by GDC for the TCGA RNA-seq data.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Human Protein Atlas Data for Tissue-Adjusted Pathway Analysis of
Cancer (TPAC)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains summary data on gene expression in normal human tissues from the Human Protein Atlas for use with the 
  Tissue-Adjusted Pathway Analysis of cancer (TPAC) method.
  Frost, H. Robert (2023) "Tissue-adjusted pathway analysis of cancer (TPAC)" &lt;<a href="https://doi.org/10.1101%2F2022.03.17.484779">doi:10.1101/2022.03.17.484779</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>bzip2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-23 13:13:48 UTC; d37329b</td>
</tr>
<tr>
<td>Author:</td>
<td>H. Robert Frost [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>H. Robert Frost &lt;rob.frost@dartmouth.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-24 20:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='hpa.data'>Gene expression data for normal human tissues from the Human Protein Atlas (HPA). This data was specially processed by the HPA group as FPKM values (file &quot;HPA.normal.FPKM.GDCpipeline.csv&quot;) using a pipeline similar to that employed by GDC for the TCGA RNA-seq data.
</h2><span id='topic+hpa.data'></span>

<h3>Description</h3>

<p>A data.table with 1,088,694 rows and 4 columns (Gene, Tissue, FPKM, TissueSpecificity):
</p>

<ul>
<li><p> Gene: Ensembl gene ID
</p>
</li>
<li><p> Tissue: Human tissue type
</p>
</li>
<li><p> FPKM: Mean expression of that gene in units FPKM+1
</p>
</li>
<li><p> TissueSpecificity: The fold-change in expression between mean expression in that tissue and mean in all 18 tissues.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>hpa.data</code></pre>


<h3>Format</h3>

<p>A data.table with 1,088,694 rows and 4 columns (Gene, Tissue, FPKM, TissueSpecificity)</p>


<h3>Source</h3>

<p>Human Protein Atlas</p>


<h3>References</h3>


<ul>
<li><p> Ulhen et al. (2017). A pathology atlas of the human cancer transcriptome. Science. doi: 10.1126/science.aan2507
</p>
</li></ul>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
