<!DOCTYPE html><html><head><title>Help for package TPAC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TPAC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TPAC-package'>
<p>Tissue-adjusted pathway analysis of cancer (TPAC)</p></a></li>
<li><a href='#createGeneSetCollection'><p>Utility function to help create gene set collection list object</p></a></li>
<li><a href='#getSupportedCancerTypes'><p>Get supported cancer types</p></a></li>
<li><a href='#tpac'><p>Tissue-adjusted pathway analysis of cancer (TPAC) algorithm</p></a></li>
<li><a href='#tpacForCancer'><p>Executes the TPAC method on cancer gene expression data</p></a></li>
<li><a href='#tpacForCollection'><p>TPAC method for multiple gene sets</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tissue-Adjusted Pathway Analysis of Cancer (TPAC)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains logic for single sample gene set testing of cancer transcriptomic data with adjustment for normal tissue-specificity.
  Frost, H. Robert (2023) "Tissue-adjusted pathway analysis of cancer (TPAC)" &lt;<a href="https://doi.org/10.1101%2F2022.03.17.484779">doi:10.1101/2022.03.17.484779</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, MASS, TPACData</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-26 12:36:21 UTC; d37329b</td>
</tr>
<tr>
<td>Author:</td>
<td>H. Robert Frost [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>H. Robert Frost &lt;rob.frost@dartmouth.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-26 15:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='TPAC-package'>
Tissue-adjusted pathway analysis of cancer (TPAC)
</h2><span id='topic+TPAC-package'></span>

<h3>Description</h3>

<p>Implementation of TPAC, a method for single sample gene set testing of cancer transcriptomics data that leverages normal tissue-specificity.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> TPAC</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.1.0 </td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Note</h3>

<p>This work was supported by the National Institutes of Health grants R35GM146586, R21CA253408, P20GM130454 and P30CA023108.
</p>


<h3>Author(s)</h3>

<p>H. Robert Frost
</p>


<h3>References</h3>


<ul>
<li><p> Frost, H. R. (2022). Tissue-adjusted pathway analysis of cancer (TPAC). biorXiv e-prints. doi: https://doi.org/10.1101/2022.03.17.484779 
</p>
</li></ul>


<hr>
<h2 id='createGeneSetCollection'>Utility function to help create gene set collection list object</h2><span id='topic+createGeneSetCollection'></span>

<h3>Description</h3>

<p>Utility function that creates a gene set collection list in the format required by <code><a href="#topic+tpacForCollection">tpacForCollection</a></code> given the gene IDs measured in the expression matrix and a list of gene sets as defined by the IDs of the member genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    createGeneSetCollection(gene.ids, gene.set.collection, min.size=1, max.size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createGeneSetCollection_+3A_gene.ids">gene.ids</code></td>
<td>
<p>Vector of gene IDs. This should correspond to the genes measured in the gene expression data.</p>
</td></tr>
<tr><td><code id="createGeneSetCollection_+3A_gene.set.collection">gene.set.collection</code></td>
<td>
<p>List of gene sets where each element in the list corresponds to a gene set and the list element is a vector of gene IDs. List names are gene set names. Must contain at least one gene set.</p>
</td></tr>
<tr><td><code id="createGeneSetCollection_+3A_min.size">min.size</code></td>
<td>
<p>Minimum gene set size after filtering out genes not in the gene.ids vector. Gene sets whose post-filtering size is below this are removed from the final collection list. Default is 1 and cannot be set to less than 1.</p>
</td></tr>
<tr><td><code id="createGeneSetCollection_+3A_max.size">max.size</code></td>
<td>
<p>Maximum gene set size after filtering out genes not in the gene.ids vector. Gene sets whose post-filtering size is above this are removed from the final collection list. If not specified, no filtering is performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Version of the input gene.set.collection list where gene IDs have been replaced by position indices, genes not present in the gene.ids vector have been removed and gene sets failing the min/max size constraints have been removed.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tpacForCollection">tpacForCollection</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Create a collection with two sets defined over 3 genes
    createGeneSetCollection(gene.ids=c("A", "B", "C"),
        gene.set.collection = list(set1=c("A", "B"), set2=c("B", "C")),
        min.size=2, max.size=3)
</code></pre>

<hr>
<h2 id='getSupportedCancerTypes'>Get supported cancer types</h2><span id='topic+getSupportedCancerTypes'></span>

<h3>Description</h3>

<p>Retrieves the names of cancer types supported by the <code><a href="#topic+tpacForCancer">tpacForCancer</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    getSupportedCancerTypes()
</code></pre>


<h3>Value</h3>

<p>Vector containing the names of supported cancer types.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tpacForCancer">tpacForCancer</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>  getSupportedCancerTypes()
</code></pre>

<hr>
<h2 id='tpac'>Tissue-adjusted pathway analysis of cancer (TPAC) algorithm</h2><span id='topic+tpac'></span>

<h3>Description</h3>

<p>Implementation of the tissue-adjusted pathway analysis of cancer (TPAC) method, which performs single sample gene set scoring of
cancer transcriptomic data. The TPAC method computes a tissue-adjusted squared Mahalanobis distance and one-sided CDF values for
all samples in the specified cancer gene expression matrix. Importantly, these distances are measured from the
normal tissue mean and normal tissue specificity (i.e., fold-change in mean expression in the corresponding normal tissue relative to
the mean expression in other tissue types) is used to weight genes when computing the distance metric. The distances are decomposed into
a positive component, negative component and overall component, i.e., the positive component only captures expression values above the mean.
A gamma distribution is fit on the squared tissue-adjusted Mahalanobis distances computed on data that is permuted to match to the null 
that the cancer expression data is uncorrelated with mean values matching those found in the associated normal tissue. This null
gamma distribution is used to calculate the CDF values for the unpermuted distances.</p>


<h3>Usage</h3>

<pre><code class='language-R'>    tpac(gene.expr, mean.expr, tissue.specificity)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tpac_+3A_gene.expr">gene.expr</code></td>
<td>
<p>An n x p matrix of gene expression values for n samples and p genes. This matrix should reflect the subset of the 
full expression profile that corresponds to a single gene set.</p>
</td></tr> 
<tr><td><code id="tpac_+3A_mean.expr">mean.expr</code></td>
<td>
<p>Vector of mean expression values for the p genes from which distances will be measured, i.e., mean expression in 
corresponding normal tissue.These mean expression values must be computed using the same normalization method used for 
tumor expression data in <code>gene.expr</code></p>
</td></tr>
<tr><td><code id="tpac_+3A_tissue.specificity">tissue.specificity</code></td>
<td>
<p>Optional vector of tissue-specificity values for the p genes, 
i.e, fold change between mean expression in the associated normal tissue and the average of 
mean expression values in other cancer-associated normal tissues. If specified, these specificity values are used to modify the 
diagonal elements of the covariance matrix used in the Mahalanobis statistic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> with the following elements (row names will match row names from <code>gene.expr</code>):
</p>

<ul>
<li> <p><code>s.pos</code>: vector of TPAC scores (i.e., CDF values) computed from the positive squared adjusted Mahalanobis distances.
</p>
</li>
<li> <p><code>s.neg</code>: vector of TPAC scores computed from the negative squared adjusted Mahalanobis distances.
</p>
</li>
<li> <p><code>s</code>: vector of TPAC scores computed from the sum of the positive and negative squared adjusted Mahalanobis distances.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+tpacForCollection">tpacForCollection</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Simulate Gaussian expression data for 10 genes and 10 samples
    gene.expr=matrix(rnorm(100), nrow=10)
    # Use 0 as mean.expr 
    mean.expr=rep(0,10)
    # Simulate tissue-specific weights
    tissue.specificity = runif(10, min=0.5, max=1.5)
    # Execute TPAC to compute positive, negative and overall scores 
    tpac(gene.expr=gene.expr, mean.expr=mean.expr, tissue.specificity=tissue.specificity)
</code></pre>

<hr>
<h2 id='tpacForCancer'>Executes the TPAC method on cancer gene expression data</h2><span id='topic+tpacForCancer'></span>

<h3>Description</h3>

<p>Executes the TPAC (tissue-adjusted pathway analysis for cancer) method (<code><a href="#topic+tpacForCollection">tpacForCollection</a></code>) on cancer gene expression data
using normal tissue expression data from the Human Protein Atlas (HPA) that is included in the package as <code><a href="TPACData.html#topic+hpa.data">hpa.data</a></code>.
This HPA normal tissue data was specially processed by the HPA group as FPKM values using a pipeline similar to that employed by GDC for the TCGA data. For consistency with this HPA normal tissue data, the provided <code>cancer.gene.expr</code> data must be specified as FPKM+1 values. Please see the vignette for an example of calling this function using appropriately normalized TCGA gene expression data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    tpacForCancer(cancer.gene.expr, cancer.type, gene.set.collection, 
                  min.set.size=1, max.set.size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tpacForCancer_+3A_cancer.gene.expr">cancer.gene.expr</code></td>
<td>
<p>An n x p matrix of gene expression values for n tumors of the specified tumor type and p genes. 
The data should be normalized as FPKM+1 values, row names should be sample ID, and column names should be Ensembl gene IDs.</p>
</td></tr>
<tr><td><code id="tpacForCancer_+3A_cancer.type">cancer.type</code></td>
<td>
<p>Cancer type of the expression data. Must be one of the supported cancer types as per <code><a href="#topic+getSupportedCancerTypes">getSupportedCancerTypes</a></code>.</p>
</td></tr>
<tr><td><code id="tpacForCancer_+3A_gene.set.collection">gene.set.collection</code></td>
<td>
<p>List of m gene sets for which scores are computed.
Each element in the list corresponds to a gene set and the list element is a vector
of Ensembl IDs for genes in the set. Gene set names should be specified as list names.</p>
</td></tr>
<tr><td><code id="tpacForCancer_+3A_min.set.size">min.set.size</code></td>
<td>
<p>See description of <code>min.size</code> in <code><a href="#topic+createGeneSetCollection">createGeneSetCollection</a></code></p>
</td></tr>
<tr><td><code id="tpacForCancer_+3A_max.set.size">max.set.size</code></td>
<td>
<p>See description of <code>max.size</code> in <code><a href="#topic+createGeneSetCollection">createGeneSetCollection</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements:
</p>

<ul>
<li> <p><code>S.pos</code>: n x m matrix of TPAC scores computed using the positive squared adjusted Mahalanobis distances.
</p>
</li>
<li> <p><code>S.neg</code>: n x m matrix of TPAC scores computed using the negative squared adjusted Mahalanobis distances.
</p>
</li>
<li> <p><code>S</code>: n x m matrix of TPAC scorescomputed using the sum of the positive and negative squared adjusted Mahalanobis distances.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+tpac">tpac</a></code>, <code><a href="TPACData.html#topic+hpa.data">hpa.data</a></code>, <code><a href="#topic+tpacForCollection">tpacForCollection</a></code>, <code><a href="#topic+getSupportedCancerTypes">getSupportedCancerTypes</a></code>, <code><a href="#topic+createGeneSetCollection">createGeneSetCollection</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Simulate Gaussian expression data for 10 genes and 10 samples
    # (Note: cancer expression should be FPKM+1 for real applications)
    cancer.gene.expr=matrix(rnorm(200), nrow=20)
    # Create arbitrary Ensembl IDs
    gene.ids = c("ENSG00000000003","ENSG00000000005","ENSG00000000419",
                 "ENSG00000000457","ENSG00000000460","ENSG00000000938",
                 "ENSG00000000971","ENSG00000001036","ENSG00000001084",
                 "ENSG00000001167")
    colnames(cancer.gene.expr) = gene.ids
    # Define a collection with two disjoint sets that span the 10 genes
    collection=list(set1=gene.ids[1:5], set2=gene.ids[6:10])    
    # Execute TPAC on both sets
    tpacForCancer(cancer.gene.expr=cancer.gene.expr, cancer.type="glioma", 
                  gene.set.collection=collection)
</code></pre>

<hr>
<h2 id='tpacForCollection'>TPAC method for multiple gene sets</h2><span id='topic+tpacForCollection'></span>

<h3>Description</h3>

<p>Executes the TPAC (tissue-adjusted pathway analysis for cancer) method (<code><a href="#topic+tpac">tpac</a></code>) on multiple gene sets, i.e., a gene set collection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    tpacForCollection(gene.expr, mean.expr, tissue.specificity, gene.set.collection)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tpacForCollection_+3A_gene.expr">gene.expr</code></td>
<td>
<p>See description in <code><a href="#topic+tpac">tpac</a></code></p>
</td></tr> 
<tr><td><code id="tpacForCollection_+3A_mean.expr">mean.expr</code></td>
<td>
<p>See description in <code><a href="#topic+tpac">tpac</a></code></p>
</td></tr> 
<tr><td><code id="tpacForCollection_+3A_tissue.specificity">tissue.specificity</code></td>
<td>
<p>See description in <code><a href="#topic+tpac">tpac</a></code></p>
</td></tr> 
<tr><td><code id="tpacForCollection_+3A_gene.set.collection">gene.set.collection</code></td>
<td>
<p>List of m gene sets for which scores are computed.
Each element in the list corresponds to a gene set and the list element is a vector
of indices for the genes in the set. The index value is defined relative to the
order of genes in the <code>gene.expr</code> matrix. Gene set names should be specified as list names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements:
</p>

<ul>
<li> <p><code>s.pos</code>: n x m matrix of TPAC scores computed using the positive squared adjusted Mahalanobis distances.
</p>
</li>
<li> <p><code>s.neg</code>: n x m matrix of TPAC scores computed using the negative squared adjusted Mahalanobis distances.
</p>
</li>
<li> <p><code>s</code>: n x m matrix of TPAC scores computed using the sum of the positive and negative squared adjusted Mahalanobis distances.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+tpac">tpac</a></code>,<code><a href="#topic+tpacForCancer">tpacForCancer</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Simulate Gaussian expression data for 10 genes and 10 samples
    gene.expr=matrix(rnorm(100), nrow=10)
    # Use 0 as mean.expr 
    mean.expr=rep(0,10)
    # Simulate tissue-specific weights
    tissue.specificity = runif(10, min=0.5, max=1.5)
    # Define a collection with two disjoint sets that span the 10 genes
    collection=list(set1=1:5, set2=6:10)    
    # Execute TPAC on both sets
    tpacForCollection(gene.expr=gene.expr, mean.expr=mean.expr,
        tissue.specificity=tissue.specificity, gene.set.collection=collection)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
