<!DOCTYPE html><html><head><title>Help for package TPES</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TPES}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TCGA_A8_A0A7_maf'><p>SNVsReadCountsFile for sample TCGA-A8-A0A7</p></a></li>
<li><a href='#TCGA_A8_A0A7_ploidy'><p>Ploidy data for sample TCGA-A8-A0A7</p></a></li>
<li><a href='#TCGA_A8_A0A7_seg'><p>SEG file (segmented data) for sample TCGA-A8-A0A7</p></a></li>
<li><a href='#TCGA_HT_8564_maf'><p>SNVsReadCountsFile for sample TCGA-HT-8564</p></a></li>
<li><a href='#TCGA_HT_8564_ploidy'><p>Ploidy data for sample TCGA-HT-8564</p></a></li>
<li><a href='#TCGA_HT_8564_seg'><p>SEG file (segmented data) for sample TCGA-HT-8564</p></a></li>
<li><a href='#TPES_purity'><p>Tumor Purity Estimation using SNVs</p></a></li>
<li><a href='#TPES_report'><p>Tumor Purity Estimation using SNVs</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tumor Purity Estimation using SNVs</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-01-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Alessio Locallo &lt;alessio.locallo@gmail.com&gt;, Davide Prandi &lt;davide.prandi@unitn.it&gt;, Francesca Demichelis &lt;f.demichelis@unitn.it&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alessio Locallo &lt;alessio.locallo@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A bioinformatics tool for the estimation of the tumor purity from sequencing data. 
    It uses the set of putative clonal somatic single nucleotide variants within copy number neutral
    segments to call tumor cellularity.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>grDevices, graphics, stats</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-01-09 17:03:16 UTC; l0ka</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-01-11 17:40:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='TCGA_A8_A0A7_maf'>SNVsReadCountsFile for sample TCGA-A8-A0A7</h2><span id='topic+TCGA_A8_A0A7_maf'></span>

<h3>Description</h3>

<p>A data frame object containing the read counts data of somatic
single nucleotide variants (SNVs) loci for sample TCGA-A8-A0A7.
The header contains the chromosme that harbors the SNV (&quot;chr&quot; column),
the position of the SNV (defined by the &quot;start&quot; and &quot;end&quot; columns),
the informations about the reference and alternative base counts (&quot;ref.count&quot;
and	&quot;alt.count&quot; columns, respectively) and finally the sample ID (&quot;sample&quot;
column). For more information please visit
<a href="https://software.broadinstitute.org/software/igv/MutationAnnotationFormat">MAF file format</a>.
</p>


<h3>Format</h3>

<p>A data.frame object.</p>

<hr>
<h2 id='TCGA_A8_A0A7_ploidy'>Ploidy data for sample TCGA-A8-A0A7</h2><span id='topic+TCGA_A8_A0A7_ploidy'></span>

<h3>Description</h3>

<p>A data frame containing the ploidy status of a sample. It must
contains at least the sample ID (&quot;sample&quot; column) and the ploidy status (&quot;ploidy&quot; column).
</p>


<h3>Format</h3>

<p>A data.frame object.</p>

<hr>
<h2 id='TCGA_A8_A0A7_seg'>SEG file (segmented data) for sample TCGA-A8-A0A7</h2><span id='topic+TCGA_A8_A0A7_seg'></span>

<h3>Description</h3>

<p>A data frame object that lists loci and associated numeric values.
The header must be compatible with the standard format defined by the Broad
Institute. For more information please visit
<a href="https://software.broadinstitute.org/software/igv/SEG">SEG file format</a>.
</p>


<h3>Format</h3>

<p>A data.frame object.</p>

<hr>
<h2 id='TCGA_HT_8564_maf'>SNVsReadCountsFile for sample TCGA-HT-8564</h2><span id='topic+TCGA_HT_8564_maf'></span>

<h3>Description</h3>

<p>A data frame object containing the read counts data of somatic
single nucleotide variants (SNVs) loci for sample TCGA-HT-8564.
The header contains the chromosme that harbors the SNV (&quot;chr&quot; column),
the position of the SNV (defined by the &quot;start&quot; and &quot;end&quot; columns),
the informations about the reference and alternative base counts (&quot;ref.count&quot;
and	&quot;alt.count&quot; columns, respectively) and finally the sample ID (&quot;sample&quot;
column). For more information please visit
<a href="https://software.broadinstitute.org/software/igv/MutationAnnotationFormat">MAF file format</a>.
</p>


<h3>Format</h3>

<p>A data.frame object.</p>

<hr>
<h2 id='TCGA_HT_8564_ploidy'>Ploidy data for sample TCGA-HT-8564</h2><span id='topic+TCGA_HT_8564_ploidy'></span>

<h3>Description</h3>

<p>A data frame containing the ploidy status of a sample. It must
contains at least the sample ID (&quot;sample&quot; column) and the ploidy status (&quot;ploidy&quot; column).
</p>


<h3>Format</h3>

<p>A data.frame object.</p>

<hr>
<h2 id='TCGA_HT_8564_seg'>SEG file (segmented data) for sample TCGA-HT-8564</h2><span id='topic+TCGA_HT_8564_seg'></span>

<h3>Description</h3>

<p>A data frame object that lists loci and associated numeric values.
The header must be compatible with the standard format defined by the Broad
Institute. For more information please visit
<a href="https://software.broadinstitute.org/software/igv/SEG">SEG file format</a>.
</p>


<h3>Format</h3>

<p>A data.frame object.</p>

<hr>
<h2 id='TPES_purity'>Tumor Purity Estimation using SNVs</h2><span id='topic+TPES_purity'></span>

<h3>Description</h3>

<p>TPES_purity function estimates tumor purity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TPES_purity(ID, SEGfile, SNVsReadCountsFile, ploidy, RMB = 0.47,
  maxAF = 0.55, minCov = 10, minAltReads = 5, minSNVs = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TPES_purity_+3A_id">ID</code></td>
<td>
<p>Sample ID. Must be the same ID as in SEGfile, SNVsReadCountsFile and ploidy.</p>
</td></tr>
<tr><td><code id="TPES_purity_+3A_segfile">SEGfile</code></td>
<td>
<p>A standard SEG file (segmented data). It is a data frame object that lists
loci and associated numeric values. The header must be compatible with the standard format
defined by the Broad Institute. For more information please visit
<a href="https://software.broadinstitute.org/software/igv/SEG">SEG file format</a>.</p>
</td></tr>
<tr><td><code id="TPES_purity_+3A_snvsreadcountsfile">SNVsReadCountsFile</code></td>
<td>
<p>A standard MAF (Mutation Annotation Format) file.
It is a data frame object containing the read counts data of somatic
single nucleotide variants (SNVs) loci. The header must contains at least informations about the chromosme
that harbors the SNV (&quot;chr&quot; column), the position of the SNV (defined by the &quot;start&quot; and &quot;end&quot; columns),
the sample ID (&quot;sample&quot; column) and finally the informations about the reference and alternative base
counts (&quot;ref.count&quot; and    &quot;alt.count&quot; columns, respectively). For more information please visit
<a href="https://software.broadinstitute.org/software/igv/MutationAnnotationFormat">MAF file format</a>.</p>
</td></tr>
<tr><td><code id="TPES_purity_+3A_ploidy">ploidy</code></td>
<td>
<p>A data frame containing the ploidy status of a sample. It must contain at
least the sample ID (&quot;sample&quot; column) and the ploidy status (&quot;ploidy&quot; column).</p>
</td></tr>
<tr><td><code id="TPES_purity_+3A_rmb">RMB</code></td>
<td>
<p>The Reference Mapping Bias value. The reference genome contains only one allele
at any given locus, so reads that carry a non-reference allele are less likely to be mapped
during alignment; this causes a shift from 0.5. It can be
estimated as: <code class="reqn">1 - medAF</code>, where medAF is the median value of the allelic fraction of the
sample's germline heterozygous SNPs. Default is set to 0.47. For more informations see: PMID: 19808877.</p>
</td></tr>
<tr><td><code id="TPES_purity_+3A_maxaf">maxAF</code></td>
<td>
<p>The filter on the allelic fraction (AF) distribution of SNVs. This is necessary to be sure to keep
only heterozygous SNVs. Clonal and subclonal SNVs, which have an AF greater than maxAF, will be removed.</p>
</td></tr>
<tr><td><code id="TPES_purity_+3A_mincov">minCov</code></td>
<td>
<p>The minimum coverage for a SNV to be retained.</p>
</td></tr>
<tr><td><code id="TPES_purity_+3A_minaltreads">minAltReads</code></td>
<td>
<p>The minimum coverage for the alternative base of a SNV to be retained.</p>
</td></tr>
<tr><td><code id="TPES_purity_+3A_minsnvs">minSNVs</code></td>
<td>
<p>The minimum number of SNVs required to make a purity call.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TPES returns a data.frame object with one row per sample and the following columns:
</p>
<table>
<tr><td><code>sample</code></td>
<td>
<p>The sample ID;</p>
</td></tr>
<tr><td><code>purity</code></td>
<td>
<p>The sample purity estimated by TPES;</p>
</td></tr>
<tr><td><code>purity.min</code></td>
<td>
<p>The sample minimum purity estimated by TPES;</p>
</td></tr>
<tr><td><code>purity.max</code></td>
<td>
<p>The sample maximum purity estimated by TPES;</p>
</td></tr>
<tr><td><code>n.segs</code></td>
<td>
<p>The number of copy number neutral segments used by TPES;</p>
</td></tr>
<tr><td><code>n.SNVs</code></td>
<td>
<p>The number of SNVs used by TPES;</p>
</td></tr>
<tr><td><code>RMB</code></td>
<td>
<p>The Reference Mapping Bias value used to estimate the tumor purity;</p>
</td></tr>
<tr><td><code>BandWidth</code></td>
<td>
<p>The smoothing bandwidth value of the <code><a href="stats.html#topic+density">density</a></code> function chosen by TPES.</p>
</td></tr>
<tr><td><code>log</code></td>
<td>
<p>Reports if the run was successful; otherwise provides debugging information.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Compute tumor purity for samples "TCGA-A8-A0A7" and "TCGA-HT-8564"
## https://cancergenome.nih.gov/
## Please copy and paste the following lines:
library(TPES)
TPES_purity(ID = "TCGA-A8-A0A7", SEGfile = TCGA_A8_A0A7_seg,
SNVsReadCountsFile = TCGA_A8_A0A7_maf, ploidy = TCGA_A8_A0A7_ploidy,
RMB = 0.47, maxAF = 0.55, minCov = 10, minAltReads = 5, minSNVs = 10)

TPES_purity(ID = "TCGA-HT-8564", SEGfile = TCGA_HT_8564_seg,
SNVsReadCountsFile = TCGA_HT_8564_maf, ploidy = TCGA_HT_8564_ploidy,
RMB = 0.47, maxAF = 0.55, minCov = 10, minAltReads = 5, minSNVs = 10)

</code></pre>

<hr>
<h2 id='TPES_report'>Tumor Purity Estimation using SNVs</h2><span id='topic+TPES_report'></span>

<h3>Description</h3>

<p>TPES_report function produces a graphical report regarding the allelic fraction
values of the putative clonal SNVs used by TPES_purity and the density function(s) computed
by TPES_purity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TPES_report(ID, SEGfile, SNVsReadCountsFile, ploidy, RMB = 0.47,
  maxAF = 0.55, minCov = 10, minAltReads = 5, minSNVs = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TPES_report_+3A_id">ID</code></td>
<td>
<p>Sample ID. Must be the same ID as in SEGfile, SNVsReadCountsFile and ploidy.</p>
</td></tr>
<tr><td><code id="TPES_report_+3A_segfile">SEGfile</code></td>
<td>
<p>A standard SEG file (segmented data). It is a data frame object that lists
loci and associated numeric values. The header must be compatible with the standard format
defined by the Broad Institute. For more information please visit
<a href="https://software.broadinstitute.org/software/igv/SEG">SEG file format</a>.</p>
</td></tr>
<tr><td><code id="TPES_report_+3A_snvsreadcountsfile">SNVsReadCountsFile</code></td>
<td>
<p>A standard MAF (Mutation Annotation Format) file.
It is a data frame object containing the read counts data of somatic
single nucleotide variants (SNVs) loci. The header must contains at least informations about the chromosme
that harbors the SNV (&quot;chr&quot; column), the position of the SNV (defined by the &quot;start&quot; and &quot;end&quot; columns),
the sample ID (&quot;sample&quot; column) and finally the informations about the reference and alternative base
counts (&quot;ref.count&quot; and    &quot;alt.count&quot; columns, respectively). For more information please visit
<a href="https://software.broadinstitute.org/software/igv/MutationAnnotationFormat">MAF file format</a>.</p>
</td></tr>
<tr><td><code id="TPES_report_+3A_ploidy">ploidy</code></td>
<td>
<p>A data frame containing the ploidy status of a sample. It must contain at
least the sample ID (&quot;sample&quot; column) and the ploidy status (&quot;ploidy&quot; column).</p>
</td></tr>
<tr><td><code id="TPES_report_+3A_rmb">RMB</code></td>
<td>
<p>The Reference Mapping Bias value. The reference genome contains only one allele
at any given locus, so reads that carry a non-reference allele are less likely to be mapped
during alignment; this causes a shift from 0.5. It can be
estimated as: <code class="reqn">1 - medAF</code>, where medAF is the median value of the allelic fraction of the
sample's germline heterozygous SNPs. Default is set to 0.47. For more informations see: PMID: 19808877.</p>
</td></tr>
<tr><td><code id="TPES_report_+3A_maxaf">maxAF</code></td>
<td>
<p>The filter on the allelic fraction (AF) distribution of SNVs. This is necessary to be sure to keep
only heterozygous SNVs. Clonal and subclonal SNVs, which have an AF greater than maxAF, will be removed.</p>
</td></tr>
<tr><td><code id="TPES_report_+3A_mincov">minCov</code></td>
<td>
<p>The minimum coverage for a SNV to be retained.</p>
</td></tr>
<tr><td><code id="TPES_report_+3A_minaltreads">minAltReads</code></td>
<td>
<p>The minimum coverage for the alternative base of a SNV to be retained.</p>
</td></tr>
<tr><td><code id="TPES_report_+3A_minsnvs">minSNVs</code></td>
<td>
<p>The minimum number of SNVs required to make a purity call.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot with:
</p>
<table>
<tr><td><code>histogram</code></td>
<td>
<p>Represents the allelic fraction distribution of putative clonal and subclonal
(if presents) SNVs within copy number neutral segments and the peak(s) detected by TPES;</p>
</td></tr>
<tr><td><code>density plot</code></td>
<td>
<p>Represents how the density function varies according to
different bandwidth values (for more information see <code><a href="stats.html#topic+density">density</a></code>);
only the bandwidth values that result in at most 2 peaks are considered.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Generate TPES report for samples "TCGA-A8-A0A7" and "TCGA-HT-8564"
## https://cancergenome.nih.gov/
## Please copy and paste the following lines:
library(TPES)
TPES_report(ID = "TCGA-A8-A0A7", SEGfile = TCGA_A8_A0A7_seg,
SNVsReadCountsFile = TCGA_A8_A0A7_maf, ploidy = TCGA_A8_A0A7_ploidy,
RMB = 0.47, maxAF = 0.55, minCov = 10, minAltReads = 5, minSNVs = 10)

TPES_report(ID = "TCGA-HT-8564", SEGfile = TCGA_HT_8564_seg,
SNVsReadCountsFile = TCGA_HT_8564_maf, ploidy = TCGA_HT_8564_ploidy,
RMB = 0.47, maxAF = 0.55, minCov = 10, minAltReads = 5, minSNVs = 10)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
