<!DOCTYPE html><html><head><title>Help for package traits</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {traits}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#traits-package'><p>traits - Species trait data from around the web</p></a></li>
<li><a href='#betydb'><p>Search for traits from BETYdb</p></a></li>
<li><a href='#betydb_query'><p>Query a BETY table</p></a></li>
<li><a href='#birdlife_habitat'><p>Get bird habitat information from BirdLife/IUCN</p></a></li>
<li><a href='#birdlife_threats'><p>Get bird threat information from BirdLife/IUCN</p></a></li>
<li><a href='#coral-defunct'><p>Search for coral data on coraltraits.org</p></a></li>
<li><a href='#eol_invasive_'><p>Search for presence of taxonomic names in EOL invasive species databases</p></a></li>
<li><a href='#fe_native'><p>Check species status (native, exotic, ...) from Flora Europaea webpage</p></a></li>
<li><a href='#g_invasive'><p>Check invasive species status for a set of species from GISD database</p></a></li>
<li><a href='#is_native'><p>Check if a species is native somewhere</p></a></li>
<li><a href='#leda'><p>Access LEDA trait data</p></a></li>
<li><a href='#ncbi_byid'><p>Retrieve gene sequences from NCBI by accession number.</p></a></li>
<li><a href='#ncbi_byname'><p>Retrieve gene sequences from NCBI by taxon name and gene names.</p></a></li>
<li><a href='#ncbi_searcher'><p>Search for gene sequences available for taxa from NCBI.</p></a></li>
<li><a href='#plantatt'><p>PLANTATT plant traits dataset</p></a></li>
<li><a href='#taxa_search'><p>Search for traits by taxa names</p></a></li>
<li><a href='#tr_ernest'><p>Amniote life history dataset</p></a></li>
<li><a href='#tr_usda'><p>USDA plants data</p></a></li>
<li><a href='#tr_zanne'><p>Zanne et al. plant dataset</p></a></li>
<li><a href='#traitbank'><p>Search for traits from EOL's Traitbank</p></a></li>
<li><a href='#traits_cache'><p>Caching</p></a></li>
<li><a href='#traits-defunct'><p>Defunct functions in traits</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Species Trait Data from Around the Web</td>
</tr>
<tr>
<td>Description:</td>
<td>Species trait data from many different sources, including
    sequence data from 'NCBI' (<a href="https://www.ncbi.nlm.nih.gov/">https://www.ncbi.nlm.nih.gov/</a>),
    plant trait data from 'BETYdb', data from 'EOL' 'Traitbank',
    'Birdlife' International, and more.</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.0</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/traits/">https://docs.ropensci.org/traits/</a>,
<a href="https://github.com/ropensci/traits">https://github.com/ropensci/traits</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/traits/issues">https://github.com/ropensci/traits/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>jsonlite (&ge; 0.9.19), httr (&ge; 1.1.0), crul (&ge; 0.6.0), tibble
(&ge; 1.3.4), data.table (&ge; 1.9.6), readr (&ge; 1.1.1), taxize (&ge;
0.7.4), xml2 (&ge; 0.1.2), rvest (&ge; 0.3.1), hoardr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, dplyr, plyr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>X-schema.org-applicationCategory:</td>
<td>Biodiversity</td>
</tr>
<tr>
<td>X-schema.org-keywords:</td>
<td>traits, API, web-services, species, taxonomy</td>
</tr>
<tr>
<td>X-schema.org-isPartOf:</td>
<td>https://ropensci.org</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-26 17:54:07 UTC; sckott</td>
</tr>
<tr>
<td>Author:</td>
<td>Scott Chamberlain <a href="https://orcid.org/0000-0003-1444-9135"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Zachary Foster [aut],
  Ignasi Bartomeus [aut],
  David LeBauer [aut],
  Chris Black [aut],
  David Harris [aut],
  Rupert Collins [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Scott Chamberlain &lt;myrmecocystus@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-08-27 05:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='traits-package'>traits - Species trait data from around the web</h2><span id='topic+traits-package'></span><span id='topic+traits'></span>

<h3>Description</h3>

<p>Currently included in <code>traits</code> with the associated function name or
function prefix:
</p>


<h3>Details</h3>


<ul>
<li><p> BETYdb http://www.betydb.org - <code>betydb_</code>
</p>
</li>
<li><p> National Center for Biotechnology Information - NCBI
http://www.ncbi.nlm.nih.gov/ - <code>ncbi_</code>
</p>
</li>
<li><p> Encyclopedia of Life Traitbank - <code>traitbank_</code>
</p>
</li>
<li><p> Birdlife International https://www.birdlife.org/ -
<code>birdlife_</code>
</p>
</li>
<li><p> LEDA Traitbase http://www.leda-traitbase.org/LEDAportal/index.jsp -
<code>leda_</code>
</p>
</li>
<li><p> Zanne et al. plant dataset - <code><a href="#topic+tr_zanne">tr_zanne()</a></code>
</p>
</li>
<li><p> Amniote life history dataset - <code><a href="#topic+tr_ernest">tr_ernest()</a></code>
</p>
</li></ul>

<p>See also <a href="#topic+traits-defunct">traits-defunct</a>
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>
<p>Ignasi Bartomeus
</p>
<p>Zachary Foster
</p>
<p>David LeBauer
</p>
<p>David Harris
</p>
<p>Rupert Collins
</p>

<hr>
<h2 id='betydb'>Search for traits from BETYdb</h2><span id='topic+betydb'></span><span id='topic+betydb_record'></span><span id='topic+betydb_trait'></span><span id='topic+betydb_specie'></span><span id='topic+betydb_citation'></span><span id='topic+betydb_site'></span><span id='topic+betydb_experiment'></span>

<h3>Description</h3>

<p>Search for traits from BETYdb
</p>
<p>Get details about a single item from a table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betydb_record(
  id,
  table,
  api_version = NULL,
  betyurl = NULL,
  fmt = NULL,
  key = NULL,
  user = NULL,
  pwd = NULL,
  progress = TRUE,
  ...
)

betydb_trait(
  id,
  genus = NULL,
  species = NULL,
  api_version = NULL,
  betyurl = NULL,
  fmt = "json",
  key = NULL,
  user = NULL,
  pwd = NULL,
  progress = TRUE,
  ...
)

betydb_specie(
  id,
  genus = NULL,
  species = NULL,
  api_version = NULL,
  betyurl = NULL,
  fmt = "json",
  key = NULL,
  user = NULL,
  pwd = NULL,
  progress = TRUE,
  ...
)

betydb_citation(
  id,
  genus = NULL,
  species = NULL,
  api_version = NULL,
  betyurl = NULL,
  fmt = "json",
  key = NULL,
  user = NULL,
  pwd = NULL,
  progress = TRUE,
  ...
)

betydb_site(
  id,
  api_version = NULL,
  betyurl = NULL,
  fmt = "json",
  key = NULL,
  user = NULL,
  pwd = NULL,
  progress = TRUE,
  ...
)

betydb_experiment(
  id,
  api_version = NULL,
  betyurl = NULL,
  fmt = "json",
  key = NULL,
  user = NULL,
  pwd = NULL,
  progress = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betydb_+3A_id">id</code></td>
<td>
<p>(integer) One or more ids for a species, site, variable, etc.</p>
</td></tr>
<tr><td><code id="betydb_+3A_table">table</code></td>
<td>
<p>(character) Name of the database table with which this ID is associated.</p>
</td></tr>
<tr><td><code id="betydb_+3A_api_version">api_version</code></td>
<td>
<p>(character) Which version of the BETY API should we query? One of &quot;v0&quot; or &quot;beta&quot;. Default is <code>options("betydb_api_version")</code> if set, otherwise  &quot;v0&quot;.</p>
</td></tr>
<tr><td><code id="betydb_+3A_betyurl">betyurl</code></td>
<td>
<p>(string) url to target instance of betydb. Default is <code>options("betydb_url")</code> if set, otherwise &quot;https:/www.betydb.org/&quot;</p>
</td></tr>
<tr><td><code id="betydb_+3A_fmt">fmt</code></td>
<td>
<p>(character) Format to return data in, one of json, xml, csv. Only json
currently supported.</p>
</td></tr>
<tr><td><code id="betydb_+3A_key">key</code></td>
<td>
<p>(character) An API key. Use this or user/pwd combo. Save in your
<code>.Rprofile</code> file as <code>options(betydb_key = "your40digitkey")</code>. Optional</p>
</td></tr>
<tr><td><code id="betydb_+3A_user">user</code>, <code id="betydb_+3A_pwd">pwd</code></td>
<td>
<p>(character) A user name and password. Use a user/pwd combo or an API key.
Save in your <code>.Rprofile</code> file as <code>options(betydb_user = "yournamehere")</code> and <code>options(betydb_pwd = "yourpasswordhere")</code>. Optional</p>
</td></tr>
<tr><td><code id="betydb_+3A_progress">progress</code></td>
<td>
<p>show progress bar? default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="betydb_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <code><a href="httr.html#topic+GET">GET</a></code>. Optional</p>
</td></tr>
<tr><td><code id="betydb_+3A_genus">genus</code></td>
<td>
<p>(character) A genus name. Optional</p>
</td></tr>
<tr><td><code id="betydb_+3A_species">species</code></td>
<td>
<p>(character) A specific epithet. Optional</p>
</td></tr>
</table>


<h3>Details</h3>

<p>BETYdb includes a primary home page (betydb.org) focused on bioenergy crops as well as a network of harmonized
databases that support and share data among more focused research programs.
</p>
<p>For a list of publicly accessible instances of BETYdb and the urls that can be queried,
see <a href="https://pecan.gitbooks.io/betydb-documentation/content/distributed_betydb.html">https://pecan.gitbooks.io/betydb-documentation/content/distributed_betydb.html</a>
</p>
<p>This package queries plant traits, phenotypes, biomass yields, and ecosystem functions.
It does not currently interface with the workflow and provenance data that support PEcAn Project (pecanproject.org) and TERRA REF (terraref.org) software.
</p>
<p>API documentation: <a href="https://pecan.gitbooks.io/betydb-data-access/content/API.html">https://pecan.gitbooks.io/betydb-data-access/content/API.html</a>
API endpoints are here: <a href="https://www.betydb.org/api/docs">https://www.betydb.org/api/docs</a>
This package currently uses the original 'v0' API by default.
To use a newer version, set <code>api_version</code>.
Newer versions of the API will support database inserts.
</p>


<h3>Authentication</h3>

<p>Defers to use API key first since it's simpler, but if you don't have
an API key, you can supply a username and password.
</p>


<h3>Functions</h3>

<p>Singular functions like <code>betydb_trait</code> accept an id and additional parameters,
and return a list of variable outputs depending on the inputs.
</p>
<p>However, plural functions like <code>betydb_traits</code> accept query parameters, but not
ids, and always return a single data.frame.
</p>
<p><code>betydb_search("Search terms", ...)</code> is a convenience wrapper that passes all further arguments to <code><a href="#topic+betydb_query">betydb_query</a>(table = "search", search = "Search terms", ...)</code>. See there for details on possible arguments.
</p>


<h3>References</h3>

<p>API documentation <a href="https://pecan.gitbooks.io/betydb-data-access/content/API.html">https://pecan.gitbooks.io/betydb-data-access/content/API.html</a> and
https://www.betydb.org/api/docs
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betydb_query">betydb_query</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# General Search
out &lt;- betydb_search(query = "Switchgrass Yield")
library("dplyr")
out %&gt;%
 group_by(id) %&gt;%
 summarise(mean_result = mean(as.numeric(mean), na.rm = TRUE)) %&gt;%
 arrange(desc(mean_result))
# Get by ID
## Traits
betydb_trait(id = 10)
## Species
betydb_specie(id = 1)
## Citations
betydb_citation(id = 1)
## Site information
betydb_site(id = 795)

## End(Not run)
</code></pre>

<hr>
<h2 id='betydb_query'>Query a BETY table</h2><span id='topic+betydb_query'></span><span id='topic+betydb_search'></span>

<h3>Description</h3>

<p>Query a BETY table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betydb_query(
  ...,
  table = "search",
  key = NULL,
  api_version = NULL,
  betyurl = NULL,
  user = NULL,
  pwd = NULL,
  progress = TRUE
)

betydb_search(
  query = "Maple SLA",
  ...,
  include_unchecked = NULL,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betydb_query_+3A_...">...</code></td>
<td>
<p>(named character) Columns to query, as key=&quot;value&quot; pairs. Note that betydb_query passes these along to BETY with no check whether the requested keys exist in the specified table.</p>
</td></tr>
<tr><td><code id="betydb_query_+3A_table">table</code></td>
<td>
<p>(character) The name of the database table to query, or &quot;search&quot; (the default) for the traits and yields view</p>
</td></tr>
<tr><td><code id="betydb_query_+3A_key">key</code></td>
<td>
<p>(character) An API key. Use this or user/pwd combo.
Save in your <code>.Rprofile</code> file as <code>options(betydb_key = "your40digitkey")</code>. Optional</p>
</td></tr>
<tr><td><code id="betydb_query_+3A_api_version">api_version</code></td>
<td>
<p>(character) Which version of the BETY API should we query? One of &quot;v0&quot; or &quot;beta&quot;.
Default is <code>options("betydb_api_version")</code> if set, otherwise  &quot;v0&quot;.</p>
</td></tr>
<tr><td><code id="betydb_query_+3A_betyurl">betyurl</code></td>
<td>
<p>(string) url to target instance of betydb.
Default is <code>options("betydb_url")</code> if set, otherwise &quot;https:/www.betydb.org/&quot;</p>
</td></tr>
<tr><td><code id="betydb_query_+3A_user">user</code>, <code id="betydb_query_+3A_pwd">pwd</code></td>
<td>
<p>(character) A user name and password. Use a user/pwd combo or an API key.
Save in your <code>.Rprofile</code> file as <code>options(betydb_user = "yournamehere")</code> and <code>options(betydb_pwd = "yourpasswordhere")</code>. Optional</p>
</td></tr>
<tr><td><code id="betydb_query_+3A_progress">progress</code></td>
<td>
<p>show progress bar? default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="betydb_query_+3A_query">query</code></td>
<td>
<p>(character) A string containing one or more words to be queried across all columns of the &quot;search&quot; table.</p>
</td></tr>
<tr><td><code id="betydb_query_+3A_include_unchecked">include_unchecked</code></td>
<td>
<p>(logical) Include results that have not been quality checked? Applies only to tables with a &quot;checked&quot; column: &quot;search&quot;, &quot;traits&quot;, &quot;yields&quot;. Default is to exclude unchecked values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Use betydb_query to retrieve records from a table that match on all the column filters specified in '...'.
If no filters are specified, retrieves the whole table. In API versions that support it (i.e. not in v0), filter strings beginning with &quot;~&quot; are treated as regular expressions.
</p>


<h3>Value</h3>

<p>A data.frame with attributes containing request metadata, or NULL if the query produced no results
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# literal vs regular expression vs anchored regular expression:
betydb_query(units = "Mg", table = "variables")
# NULL
betydb_query(units = "Mg/ha", table = "variables") %&gt;% select(name) %&gt;% c()
# $name
# [1] "a_biomass"                  "root_live_biomass"
# [3] "leaf_dead_biomass_in_Mg_ha" "SDM"

betydb_query(genus = "Miscanthus", table = "species") %&gt;% nrow()
# [1] 10
(betydb_query(genus = "~misc", table = "species", api_version = "beta")
 %&gt;% select(genus)
 %&gt;% unique() %&gt;% c())
# $genus
# [1] "Platymiscium" "Miscanthus"   "Dermiscellum"

(betydb_query(genus = "~^misc", table = "species", api_version = "beta")
 %&gt;% select(genus)
 %&gt;% unique() %&gt;% c())
# $genus
# [1] "Miscanthus"

## End(Not run)

</code></pre>

<hr>
<h2 id='birdlife_habitat'>Get bird habitat information from BirdLife/IUCN</h2><span id='topic+birdlife_habitat'></span>

<h3>Description</h3>

<p>Get bird habitat information from BirdLife/IUCN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>birdlife_habitat(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="birdlife_habitat_+3A_id">id</code></td>
<td>
<p>A single IUCN species ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> with level 1 and level 2 habitat classes, as
well as importance ratings and occurrence type (e.g. breeding or
non-breeding). The habitat classification scheme is described
at https://www.iucnredlist.org/resources/classification-schemes
</p>


<h3>Author(s)</h3>

<p>David J. Harris <a href="mailto:harry491@gmail.com">harry491@gmail.com</a>
</p>


<h3>See Also</h3>

<p>Other birdlife: 
<code><a href="#topic+birdlife_threats">birdlife_threats</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Setophaga chrysoparia
birdlife_habitat(22721692)
# Passer domesticus
birdlife_habitat(103818789)

## End(Not run)
</code></pre>

<hr>
<h2 id='birdlife_threats'>Get bird threat information from BirdLife/IUCN</h2><span id='topic+birdlife_threats'></span>

<h3>Description</h3>

<p>Get bird threat information from BirdLife/IUCN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>birdlife_threats(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="birdlife_threats_+3A_id">id</code></td>
<td>
<p>A single IUCN species ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> with the species ID and two levels of threat
descriptions, plus stresses, timing, scope, severity, and impact associated
with each stressor.
</p>


<h3>Author(s)</h3>

<p>David J. Harris <a href="mailto:harry491@gmail.com">harry491@gmail.com</a>
</p>


<h3>See Also</h3>

<p>Other birdlife: 
<code><a href="#topic+birdlife_habitat">birdlife_habitat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Setophaga chrysoparia
birdlife_threats(22721692)
# Aburria aburri
birdlife_threats(22678440)

## End(Not run)
</code></pre>

<hr>
<h2 id='coral-defunct'>Search for coral data on coraltraits.org</h2><span id='topic+coral-defunct'></span><span id='topic+coral_taxa'></span><span id='topic+coral_traits'></span><span id='topic+coral_locations'></span><span id='topic+coral_methodologies'></span><span id='topic+coral_resources'></span><span id='topic+coral_species'></span>

<h3>Description</h3>

<p>DEFUNCT: As far as we can tell the coraltraits.org API is down
for good
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coral_taxa(...)

coral_traits(...)

coral_locations(...)

coral_methodologies(...)

coral_resources(...)

coral_species(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coral-defunct_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>References</h3>

<p>https://coraltraits.org/
</p>

<hr>
<h2 id='eol_invasive_'>Search for presence of taxonomic names in EOL invasive species databases</h2><span id='topic+eol_invasive_'></span>

<h3>Description</h3>

<p>Search for presence of taxonomic names in EOL invasive species databases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eol_invasive_(...)
</code></pre>

<hr>
<h2 id='fe_native'>Check species status (native, exotic, ...) from Flora Europaea webpage</h2><span id='topic+fe_native'></span>

<h3>Description</h3>

<p>Check species status (native, exotic, ...) from Flora Europaea webpage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fe_native(...)
</code></pre>

<hr>
<h2 id='g_invasive'>Check invasive species status for a set of species from GISD database</h2><span id='topic+g_invasive'></span>

<h3>Description</h3>

<p>Check invasive species status for a set of species from GISD database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g_invasive(...)
</code></pre>

<hr>
<h2 id='is_native'>Check if a species is native somewhere</h2><span id='topic+is_native'></span>

<h3>Description</h3>

<p>Check if a species is native somewhere
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_native(...)
</code></pre>

<hr>
<h2 id='leda'>Access LEDA trait data</h2><span id='topic+leda'></span>

<h3>Description</h3>

<p>Access LEDA trait data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leda(trait = "age_first_flowering", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="leda_+3A_trait">trait</code></td>
<td>
<p>(character) Trait to get. See Details.</p>
</td></tr>
<tr><td><code id="leda_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For parameter <code>trait</code>, one of age_first_flowering, branching,
buds_seasonality, buds_vertical_dist, canopy_height, dispersal_type,
leaf_distribution, ldmc_geo, leaf_mass, leaf_size, morphology_disperal,
growth_form, life_span, releasing_height, seed_longevity,
seed_mass, seed_number, seed_shape, shoot_growth_form, snp, ssd, tv,
or clonal_growth_organs
</p>
<p>The following are not supported as they are too much of a pain to parse:
buoyancy, seed_bank, sla_geo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Age of first flowering
leda(trait = "age_first_flowering")

# Seed number
leda("seed_number")

# Releasing height
leda(trait = "releasing_height")

# Clonal growth organs
leda(trait = "clonal_growth_organs")

all &lt;- c("age_first_flowering", "branching", "buds_seasonality",
  "buds_vertical_dist", "canopy_height",
  "dispersal_type", "leaf_distribution", "ldmc_geo", "leaf_mass",
  "leaf_size", "morphology_disperal", "growth_form", "life_span",
  "releasing_height", "seed_longevity", "seed_mass",
  "seed_number", "seed_shape", "shoot_growth_form",
  "snp", "ssd", "tv", "clonal_growth_organs")
out &lt;- list()
for (i in seq_along(all)) {
  cat(all[i], sep="\n")
  out[[i]] &lt;- leda(all[i])
}
sapply(out, NROW)

## End(Not run)
</code></pre>

<hr>
<h2 id='ncbi_byid'>Retrieve gene sequences from NCBI by accession number.</h2><span id='topic+ncbi_byid'></span>

<h3>Description</h3>

<p>Retrieve gene sequences from NCBI by accession number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_byid(ids, format = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ncbi_byid_+3A_ids">ids</code></td>
<td>
<p>(character) GenBank ids to search for. One or more. Required.</p>
</td></tr>
<tr><td><code id="ncbi_byid_+3A_format">format</code></td>
<td>
<p>(character) Return type, e.g., <code>"fasta"</code>. NOW IGNORED.</p>
</td></tr>
<tr><td><code id="ncbi_byid_+3A_verbose">verbose</code></td>
<td>
<p>(logical) If <code>TRUE</code> (default), informative messages
printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If bad ids are included with good ones, the bad ones are
silently dropped. If all ids are bad you'll get a stop with error message.
</p>


<h3>Value</h3>

<p>data.frame of the form:
</p>

<ul>
<li><p> taxon - taxonomic name (may include some junk, but hard to parse off)
</p>
</li>
<li><p> taxonomy - organism lineage
</p>
</li>
<li><p> gene_desc - gene description
</p>
</li>
<li><p> organelle - if mitochondrial or chloroplast
</p>
</li>
<li><p> gi_no - GI number
</p>
</li>
<li><p> acc_no - accession number
</p>
</li>
<li><p> keyword - if official DNA barcode
</p>
</li>
<li><p> specimen_voucher - museum/lab accession number of vouchered material
</p>
</li>
<li><p> lat_lon - longitude/latitude of specimen collection event
</p>
</li>
<li><p> country - country/location of specimen collection event
</p>
</li>
<li><p> paper_title - title of study
</p>
</li>
<li><p> journal - journal study published in (if published)
</p>
</li>
<li><p> first_author - first author of study
</p>
</li>
<li><p> uploaded_date - date sequence was uploaded to GenBank
</p>
</li>
<li><p> length - sequence length
</p>
</li>
<li><p> sequence - sequence character string
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Scott Chamberlain, Rupert Collins
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ncbi_searcher">ncbi_searcher()</a></code>, ncbi_byname()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# A single gene
ncbi_byid(ids="360040093")

# Many genes (with different accession numbers)
ncbi_byid(ids=c("360040093","347448433"))

## End(Not run)
</code></pre>

<hr>
<h2 id='ncbi_byname'>Retrieve gene sequences from NCBI by taxon name and gene names.</h2><span id='topic+ncbi_byname'></span>

<h3>Description</h3>

<p>Retrieve gene sequences from NCBI by taxon name and gene names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_byname(
  taxa,
  gene = "COI",
  seqrange = "1:3000",
  getrelated = FALSE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ncbi_byname_+3A_taxa">taxa</code></td>
<td>
<p>(character) Scientific name to search for.</p>
</td></tr>
<tr><td><code id="ncbi_byname_+3A_gene">gene</code></td>
<td>
<p>(character) Gene or genes (in a vector) to search for.
See examples.</p>
</td></tr>
<tr><td><code id="ncbi_byname_+3A_seqrange">seqrange</code></td>
<td>
<p>(character) Sequence range, as e.g., <code>"1:1000"</code>. This is the range of
sequence lengths to search for. So <code>"1:1000"</code> means search for sequences from 1 to 1000
characters in length.</p>
</td></tr>
<tr><td><code id="ncbi_byname_+3A_getrelated">getrelated</code></td>
<td>
<p>(logical) If <code>TRUE</code>, gets the longest sequences of a species
in the same genus as the one searched for. If <code>FALSE</code>, returns nothing if no match
found.</p>
</td></tr>
<tr><td><code id="ncbi_byname_+3A_verbose">verbose</code></td>
<td>
<p>(logical) If <code>TRUE</code> (default), informative messages printed.</p>
</td></tr>
<tr><td><code id="ncbi_byname_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <a href="crul.html#topic+verb-GET">crul::verb-GET</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Removes predicted sequences so you don't have to remove them.
Predicted sequences are those with accession numbers that have &quot;XM_&quot; or
&quot;XR_&quot; prefixes. This function retrieves one sequences for each species,
picking the longest available for the given gene.
</p>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ncbi_searcher">ncbi_searcher()</a></code>, <code><a href="#topic+ncbi_byid">ncbi_byid()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# A single species
ncbi_byname(taxa="Acipenser brevirostrum")

# Many species
species &lt;- c("Colletes similis","Halictus ligatus","Perdita californica")
ncbi_byname(taxa=species, gene = c("coi", "co1"), seqrange = "1:2000")

## End(Not run)
</code></pre>

<hr>
<h2 id='ncbi_searcher'>Search for gene sequences available for taxa from NCBI.</h2><span id='topic+ncbi_searcher'></span>

<h3>Description</h3>

<p>Search for gene sequences available for taxa from NCBI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncbi_searcher(
  taxa = NULL,
  id = NULL,
  seqrange = "1:3000",
  getrelated = FALSE,
  fuzzy = FALSE,
  limit = 500,
  entrez_query = NULL,
  hypothetical = FALSE,
  verbose = TRUE,
  sleep = 0L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ncbi_searcher_+3A_taxa">taxa</code></td>
<td>
<p>(character) Scientific name to search for.</p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_id">id</code></td>
<td>
<p>(<code>character</code>) Taxonomic id to search for. Not compatible with
argument <code>taxa</code>.</p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_seqrange">seqrange</code></td>
<td>
<p>(character) Sequence range, as e.g., <code>"1:1000"</code>. This is the range of
sequence lengths to search for. So <code>"1:1000"</code> means search for sequences from 1 to 1000
characters in length.</p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_getrelated">getrelated</code></td>
<td>
<p>(logical) If <code>TRUE</code>, gets the longest sequences of a species
in the same genus as the one searched for. If <code>FALSE</code>, returns nothing if no match
found.</p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_fuzzy">fuzzy</code></td>
<td>
<p>(logical) Whether to do fuzzy taxonomic ID search or exact
search. If <code>TRUE</code>, we use <code>xXarbitraryXx[porgn:__txid&lt;ID&gt;]</code>,
but if <code>FALSE</code>, we use <code>txid&lt;ID&gt;</code>. Default: <code>FALSE</code></p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_limit">limit</code></td>
<td>
<p>(<code>numeric</code>) Number of sequences to search for and return.
Max of 10,000. If you search for 6000 records, and only 5000 are found,
you will of course only get 5000 back.</p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_entrez_query">entrez_query</code></td>
<td>
<p>(<code>character</code>; length 1) An Entrez-format query to
filter results with. This is useful to search for sequences with specific
characteristics. The format is the same as the one used to seach genbank.
(<a href="https://www.ncbi.nlm.nih.gov/books/NBK3837/#EntrezHelp.Entrez_Searching_Options">https://www.ncbi.nlm.nih.gov/books/NBK3837/#EntrezHelp.Entrez_Searching_Options</a>)</p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_hypothetical">hypothetical</code></td>
<td>
<p>(<code>logical</code>; length 1) If <code>FALSE</code>, an attempt
will be made to not return hypothetical or predicted sequences judging from
accession number prefixs (XM and XR). This can result in less than the
<code>limit</code> being returned even if there are more sequences available,
since this filtering is done after searching NCBI.</p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_verbose">verbose</code></td>
<td>
<p>(logical) If <code>TRUE</code> (default), informative messages printed.</p>
</td></tr>
<tr><td><code id="ncbi_searcher_+3A_sleep">sleep</code></td>
<td>
<p>(integer) number of seconds to sleep before each HTTP request.
use if running to 429 Too Many Requests errors from NCBI. default: 0
(no sleep)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.frame</code> of results if a single input is given. A list of
<code>data.frame</code>s if multiple inputs are given.
</p>


<h3>Authentication</h3>

<p>NCBI rate limits requests. If you set an API key you have a higher rate limit.
Set your API key like <code>Sys.setenv(ENTREZ_KEY="yourkey")</code> or you can use
<code>?rentrez::set_entrez_key</code>. set verbose curl output (<code>crul::set_verbose()</code>) to
make sure your api key is being sent in the requests
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain, Zachary Foster <a href="mailto:zacharyfoster1989@gmail.com">zacharyfoster1989@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ncbi_byid">ncbi_byid</a></code>, <code><a href="#topic+ncbi_byname">ncbi_byname</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# A single species
out &lt;- ncbi_searcher(taxa="Umbra limi", seqrange = "1:2000")
# Get the same species information using a taxonomy id
out &lt;- ncbi_searcher(id = "75935", seqrange = "1:2000")
# If the taxon name is unique, using the taxon name and id are equivalent
all(ncbi_searcher(id = "75935") ==  ncbi_searcher(taxa="Umbra limi"))
# If the taxon name is not unique, use taxon id
#  "266948" is the uid for the butterfly genus, but there is also a genus
#  of orchids with the
#  same name
nrow(ncbi_searcher(id = "266948")) ==  nrow(ncbi_searcher(taxa="Satyrium"))
# get list of genes available, removing non-unique
unique(out$gene_desc)
# does the string 'RAG1' exist in any of the gene names
out[grep("RAG1", out$gene_desc, ignore.case=TRUE),]

# A single species without records in NCBI
out &lt;- ncbi_searcher(taxa="Sequoia wellingtonia", seqrange="1:2000",
  getrelated=TRUE)

# Many species, can run in parallel or not using plyr
species &lt;- c("Salvelinus alpinus","Ictalurus nebulosus","Carassius auratus")
out2 &lt;- ncbi_searcher(taxa=species, seqrange = "1:2000")
lapply(out2, head)
library("plyr")
out2df &lt;- ldply(out2) # make data.frame of all
unique(out2df$gene_desc) # get list of genes available, removing non-unique
out2df[grep("12S", out2df$gene_desc, ignore.case=TRUE), ]

# Using the getrelated and entrez_query options
ncbi_searcher(taxa = "Olpidiopsidales", limit = 5, getrelated = TRUE,
            entrez_query = "18S[title] AND 28S[title]")

# get refseqs
one &lt;- ncbi_searcher(taxa = "Salmonella enterica",
  entrez_query="srcdb_refseq[PROP]")
two &lt;- ncbi_searcher(taxa = "Salmonella enterica")

## End(Not run)
</code></pre>

<hr>
<h2 id='plantatt'>PLANTATT plant traits dataset</h2><span id='topic+plantatt'></span>

<h3>Description</h3>

<p>PLANTATT plant traits dataset
</p>

<hr>
<h2 id='taxa_search'>Search for traits by taxa names</h2><span id='topic+taxa_search'></span>

<h3>Description</h3>

<p>Search for traits by taxa names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_search(x, db, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa_search_+3A_x">x</code></td>
<td>
<p>(character) Taxonomic name(s) to search for</p>
</td></tr>
<tr><td><code id="taxa_search_+3A_db">db</code></td>
<td>
<p>(character) only 'ncbi' for now - other options
maybe in the future</p>
</td></tr>
<tr><td><code id="taxa_search_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <code><a href="httr.html#topic+GET">GET</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code>
</p>


<h3>Author(s)</h3>

<p>Scott Chamberlain
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
taxa_search("Poa annua", db = "ncbi")

## End(Not run)
</code></pre>

<hr>
<h2 id='tr_ernest'>Amniote life history dataset</h2><span id='topic+tr_ernest'></span>

<h3>Description</h3>

<p>Amniote life history dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tr_ernest(read = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tr_ernest_+3A_read">read</code></td>
<td>
<p>(logical) read in csv files. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="tr_ernest_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <code><a href="crul.html#topic+HttpClient">crul::HttpClient()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>When using this data, cite the paper:
</p>
<p>Myhrvold, N. P., Baldridge, E., Chan, B., Sivam, D., Freeman, D. L. and
Ernest, S. K. M. (2015), An amniote life-history database to perform
comparative analyses with birds, mammals, and reptiles. Ecology, 96: 3109.
https://doi.org/10.1890/15-0846R.1
</p>
<p>As well as the Dryad data package:
</p>
<p>L. Freeman, Daniel; P. Myhrvold, Nathan; Chan, Benjamin; Sivam, Dhileep;
Ernest, S. K. Morgan; Baldridge, Elita (2016): Full Archive. figshare.
https://doi.org/10.6084/m9.figshare.3563457.v1
</p>


<h3>Value</h3>

<p>paths to the files (character) if <code>read=FALSE</code> or
a list of data.frame's if <code>read=TRUE</code>
</p>


<h3>References</h3>

<p>https://doi.org/10.1890/15-0846R.1
https://doi.org/10.6084/m9.figshare.3563457.v1
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- tr_ernest()
res$data
res$references
res$sparse
res$range_count

## End(Not run)
</code></pre>

<hr>
<h2 id='tr_usda'>USDA plants data</h2><span id='topic+tr_usda'></span>

<h3>Description</h3>

<p>DEFUNCT
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tr_usda(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tr_usda_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='tr_zanne'>Zanne et al. plant dataset</h2><span id='topic+tr_zanne'></span>

<h3>Description</h3>

<p>Zanne et al. plant dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tr_zanne(read = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tr_zanne_+3A_read">read</code></td>
<td>
<p>(logical) read in csv files. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="tr_zanne_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <code><a href="crul.html#topic+HttpClient">crul::HttpClient()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This data is a dataset stored on Dryad (doi: 10.5061/dryad.63q27).
When using this data, cite the paper:
</p>
<p>Zanne AE, Tank DC, Cornwell WK, Eastman JM, Smith SA, FitzJohn RG,
McGlinn DJ, O'Meara BC, Moles AT, Reich PB, Royer DL, Soltis DE, Stevens PF,
Westoby M, Wright IJ, Aarssen L, Bertin RI, Calaminus A, Govaerts R,
Hemmings F, Leishman MR, Oleksyn J, Soltis PS, Swenson NG, Warman L,
Beaulieu JM, Ordonez A (2014) Three keys to the radiation of angiosperms
into freezing environments. Nature 506(7486): 89-92.
http://dx.doi.org/10.1038/nature12872
</p>
<p>As well as the Dryad data package:
</p>
<p>Zanne AE, Tank DC, Cornwell WK, Eastman JM, Smith SA, FitzJohn RG,
McGlinn DJ, O'Meara BC, Moles AT, Reich PB, Royer DL, Soltis DE, Stevens PF,
Westoby M, Wright IJ, Aarssen L, Bertin RI, Calaminus A, Govaerts R,
Hemmings F, Leishman MR, Oleksyn J, Soltis PS, Swenson NG, Warman L,
Beaulieu JM, Ordonez A (2013) Data from: Three keys to the radiation of
angiosperms into freezing environments. Dryad Digital Repository.
http://dx.doi.org/10.5061/dryad.63q27.2
</p>


<h3>Value</h3>

<p>paths to the files (character) if <code>read=FALSE</code> or
a list of data.frame's if <code>read=TRUE</code>
</p>


<h3>References</h3>

<p>http://datadryad.org/resource/doi:10.5061/dryad.63q27
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- tr_zanne()
res$tax_lookup
res$woodiness
res$freezing
res$leaf_phenology

## End(Not run)
</code></pre>

<hr>
<h2 id='traitbank'>Search for traits from EOL's Traitbank</h2><span id='topic+traitbank'></span>

<h3>Description</h3>

<p>Search for traits from EOL's Traitbank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traitbank(query, key = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traitbank_+3A_query">query</code></td>
<td>
<p>(character) a query to the EOL Cypher service that holds
Traitbank data. required. no default query given. see examples</p>
</td></tr>
<tr><td><code id="traitbank_+3A_key">key</code></td>
<td>
<p>(character) EOL Cypher query API key. required, either
passed in or as an environment variable</p>
</td></tr>
<tr><td><code id="traitbank_+3A_...">...</code></td>
<td>
<p>Curl options passed on to <code><a href="crul.html#topic+verb-GET">verb-GET</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>traitbank</code> is an interface to the EOL Cypher query.
Note that the previous interface EOL had for Traits has been completely
replaced - thus, this function is completely different. You no longer
query by EOL page id, but using the query language for a database
called Neo4J. See the docs for help. Later we plan to make a more
user friendly interface to get Traitbank data that doesn't
require knowing the Neo4J query syntax
</p>


<h3>Value</h3>

<p>a list
</p>


<h3>Authentication</h3>

<p>You'll need an EOL cypher key to use this function. Get one by signing
in to your EOL account https://eol.org/users/sign_in then head to
https://eol.org/services/authenticate to get a key. Store your key
in your .Renviron file or similar under the name &quot;EOL_CYPHER_KEY&quot;,
and we will use that key in this function. Alternatively, you can
pass in your key to the key parameter, but we do not recommend
doing that as you risk accidentally committing your key to the
public web.
</p>


<h3>References</h3>

<p>https://github.com/EOL/eol_website/blob/master/doc/api.md
https://github.com/EOL/eol_website/blob/master/doc/query-examples.md
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# traitbank_query function
traitbank(query = "MATCH (n:Trait) RETURN n LIMIT 1;")

# traitbank function
res &lt;- traitbank(query = "MATCH (n:Trait) RETURN n LIMIT 2;")
res

## End(Not run)
</code></pre>

<hr>
<h2 id='traits_cache'>Caching</h2><span id='topic+traits_cache'></span>

<h3>Description</h3>

<p>Manage cached <code>traits</code> package files with <span class="pkg">hoardr</span>
</p>


<h3>Details</h3>

<p>The dafault cache directory is
<code>paste0(rappdirs::user_cache_dir(), "/R/traits")</code>, but you can set
your own path using <code>cache_path_set()</code>
</p>
<p><code>cache_delete</code> only accepts 1 file name, while
<code>cache_delete_all</code> doesn't accept any names, but deletes all files.
For deleting many specific files, use <code>cache_delete</code> in a <code><a href="base.html#topic+lapply">lapply()</a></code>
type call
</p>


<h3>Useful user functions</h3>


<ul>
<li> <p><code>traits_cache$cache_path_get()</code> get cache path
</p>
</li>
<li> <p><code>traits_cache$cache_path_set()</code> set cache path
</p>
</li>
<li> <p><code>traits_cache$list()</code> returns a character vector of full
path file names
</p>
</li>
<li> <p><code>traits_cache$files()</code> returns file objects with metadata
</p>
</li>
<li> <p><code>traits_cache$details()</code> returns files with details
</p>
</li>
<li> <p><code>traits_cache$delete()</code> delete specific files
</p>
</li>
<li> <p><code>traits_cache$delete_all()</code> delete all files, returns nothing
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
traits_cache

# list files in cache
traits_cache$list()

# delete certain database files
# traits_cache$delete("file path")
# traits_cache$list()

# delete all files in cache
# traits_cache$delete_all()
# traits_cache$list()

# set a different cache path from the default

## End(Not run)
</code></pre>

<hr>
<h2 id='traits-defunct'>Defunct functions in traits</h2><span id='topic+traits-defunct'></span>

<h3>Description</h3>

<p>These functions have been removed.
</p>


<h3>Details</h3>


<ul>
<li> <p><code>eol_invasive_</code>: This function has moved to a new package.
See <code>originr::eol</code>
</p>
</li>
<li> <p><code>fe_native</code>: This function has moved to a new package.
See <code>originr::flora_europaea</code>
</p>
</li>
<li> <p><code>g_invasive</code>: This function has moved to a new package.
See <code>originr::gisd</code>
</p>
</li>
<li> <p><code>is_native</code>: This function has moved to a new package. See
<code>originr::is_native</code>
</p>
</li>
<li> <p><code>tr_usda</code>: the API behind this function is down for good
</p>
</li>
<li> <p><code>coral_locations</code>: API down for good, as far as I can tell
</p>
</li>
<li> <p><code>coral_methodologies</code>: API down for good, as far as I can tell
</p>
</li>
<li> <p><code>coral_resources</code>: API down for good, as far as I can tell
</p>
</li>
<li> <p><code>coral_species</code>: API down for good, as far as I can tell
</p>
</li>
<li> <p><code>coral_taxa</code>: API down for good, as far as I can tell
</p>
</li>
<li> <p><code>coral_traits</code>: API down for good, as far as I can tell
</p>
</li></ul>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
