<!DOCTYPE html><html><head><title>Help for package transPlotR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {transPlotR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#gtf'><p>This is a test data for this package</p>
test data describtion</a></li>
<li><a href='#trancriptVis'><p>trancriptVis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Visualize Transcript Structures in Elegant Way</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>To visualize the gene structure with multiple isoforms better, I developed this package to draw different transcript structures easily.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>cowplot, dplyr, ggplot2, purrr, magrittr, ggarchery,
geomtextpath, stats</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), tidyverse</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/junjunlab/transPlotR">https://github.com/junjunlab/transPlotR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/junjunlab/transPlotR/issues">https://github.com/junjunlab/transPlotR/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-29 02:02:39 UTC; admin</td>
</tr>
<tr>
<td>Author:</td>
<td>Junjun Lao <a href="https://orcid.org/0000-0001-7692-9105"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junjun Lao &lt;3219030654@stu.cpu.edu.cn&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-30 12:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='gtf'>This is a test data for this package
test data describtion</h2><span id='topic+gtf'></span>

<h3>Description</h3>

<p>This is a test data for this package
test data describtion
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 1987 rows and 31 columns.
</p>


<h3>Author(s)</h3>

<p>Junjun Lao
</p>

<hr>
<h2 id='trancriptVis'>trancriptVis</h2><span id='topic+trancriptVis'></span>

<h3>Description</h3>

<p>This package is to visualize gene diffrent isoforms.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="trancriptVis_+3A_gtffile">gtfFile</code></td>
<td>
<p>GTF file.</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_gene">gene</code></td>
<td>
<p>Target gene to plot.</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_mytranscript">myTranscript</code></td>
<td>
<p>Specify which transcripts to plot use transcipt id.</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_chr">Chr</code></td>
<td>
<p>Chromosome number.</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_posstart">posStart</code></td>
<td>
<p>Region start position on genome.</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_posend">posEnd</code></td>
<td>
<p>Region end position on genome.</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_collapse">collapse</code></td>
<td>
<p>Whether to collapse multiple transcripts into one, default(FALSE).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_exonwidth">exonWidth</code></td>
<td>
<p>Exon width to plot, default(0.3).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_reltextdist">relTextDist</code></td>
<td>
<p>Transcripts name or gene name relative to exon, default(0.3).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_intronsize">intronSize</code></td>
<td>
<p>Intron line size, default(0.5).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_arrowbreak">arrowBreak</code></td>
<td>
<p>How many gap distance to draw arrows, the smaller the more arrows, default(0.15).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_exoncolorby">exonColorBy</code></td>
<td>
<p>Whether color group by &quot;transcript_id&quot; or &quot;gene_name&quot;, default(NULL).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_exonfill">exonFill</code></td>
<td>
<p>Exon fill color, default('#333399').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_circle">circle</code></td>
<td>
<p>Whether make plot into a circle plot, default(FALSE).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_cicstart">cicStart</code></td>
<td>
<p>Circle plot start position, default(pi).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_circsegcol">circSegCol</code></td>
<td>
<p>Circle sgement color, default('#333399').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_text_only">text_only</code></td>
<td>
<p>When circle plot labeled by gene name, whether remove the line connected with gene name, default(FALSE).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_ylimlow">ylimLow</code></td>
<td>
<p>The Y axis lower limitation of Circle plot, default(-10).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_openangle">openAngle</code></td>
<td>
<p>The gap of the circle plot, default(0.5).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_arrowcol">arrowCol</code></td>
<td>
<p>Normal arrow color, default('#333399').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_arrowangle">arrowAngle</code></td>
<td>
<p>Normal arrow angle, default(30).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_arrowlength">arrowLength</code></td>
<td>
<p>Normal arrow length, default(0.1).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_arrowtype">arrowType</code></td>
<td>
<p>Normal arrow type, default('open').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_addnormalarrow">addNormalArrow</code></td>
<td>
<p>Whether add normal arrow on plot, default(TRUE).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_newstylearrow">newStyleArrow</code></td>
<td>
<p>Whether add new style arrow on plot, default(FALSE).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_absspecarrowlen">absSpecArrowLen</code></td>
<td>
<p>Whether make new style arrow length to be relative to each transcript length or absolute length to the longest transcript, default(FALSE).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowrelpos">speArrowRelPos</code></td>
<td>
<p>The relative position to the transcript on horizontal direction of new style arrow, default(0).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowrellen">speArrowRelLen</code></td>
<td>
<p>The relative length to the transcript length of new style arrow, default(0.05).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowstart">speArrowStart</code></td>
<td>
<p>The new style arrow start position on the vertical direction, default(-0.15).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowrelhigh">speArrowRelHigh</code></td>
<td>
<p>The relative height of new style arrow to the vertical length, default(2).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowlinesize">speArrowLineSize</code></td>
<td>
<p>The new style arrow line size, default(0.5).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowcol">speArrowCol</code></td>
<td>
<p>The new style arrow line color, default('black').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowangle">speArrowAngle</code></td>
<td>
<p>The new style arrow angle, default(30).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowlen">speArrowLen</code></td>
<td>
<p>The new style arrow length, default(0.1).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_spearrowtype">speArrowType</code></td>
<td>
<p>The new style arrow type, default('closed').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_textlabel">textLabel</code></td>
<td>
<p>The text label aesthetic mappings, default('transcript_id').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_textlabelsize">textLabelSize</code></td>
<td>
<p>The text label size, default(5).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_textlabelcolor">textLabelColor</code></td>
<td>
<p>The text label color, default('black').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_base_size">base_size</code></td>
<td>
<p>Theme basesize, default(14).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_marginx">marginX</code></td>
<td>
<p>Plot left and right margins, default(0.2).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_marginy">marginY</code></td>
<td>
<p>Plot top and bottomn margins, default(0.2).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_aspect.ratio">aspect.ratio</code></td>
<td>
<p>Plot ratio, default(NULL).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_facetbygene">facetByGene</code></td>
<td>
<p>Whether facet by gene to plot, this useful for your genes which are far away from each other or not located on the same chromosome, default(FALSE).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_ncolgene">ncolGene</code></td>
<td>
<p>The column numbers to plot, default(NULL).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_scales">scales</code></td>
<td>
<p>Facet plot scales, same as &quot;facet_wrap&quot; function, default('free').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_strip.position">strip.position</code></td>
<td>
<p>Facet plot strip.position, same as &quot;facet_wrap&quot; function, default('top').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_forceposrel">forcePosRel</code></td>
<td>
<p>Whether force the genome coordinate to relative position to transcript start/end position, default('FALSE').</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_panel.spacing">panel.spacing</code></td>
<td>
<p>Facet plot panel space, default(0.3).</p>
</td></tr>
<tr><td><code id="trancriptVis_+3A_revnegstrand">revNegStrand</code></td>
<td>
<p>Whether reverse the negtive strand when set &quot;forcePosRel=TRUE&quot;, default('FALSE').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Author(s)</h3>

<p>Junjun Lao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##############################################################
# test function

########################################################
# load data
data(gtf)

# non-coding gene
trancriptVis(gtfFile = gtf,
             gene = 'Xist')

# coding gene
trancriptVis(gtfFile = gtf,
             gene = 'Nanog')

# change fill color
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             exonFill = '#CCFF00')

# change inrton line size
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             intronSize = 1)

# change label size,color and position
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             textLabelSize = 4,
             textLabelColor = 'red',
             relTextDist = 0)

# aes by gene name
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             textLabel = 'gene_name')

# color aes by transcript
trancriptVis(gtfFile = gtf,
             gene = 'Tpx2',
             exonColorBy = 'transcript_id')

# change arrow color and type
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             arrowCol = 'orange',
             arrowType = 'closed')

# no intron gene and add arrow color
# change arrow color and type
trancriptVis(gtfFile = gtf,
             gene = 'Jun',
             textLabel = 'gene_name',
             arrowCol = 'white',
             arrowType = 'closed') +
  theme_void()

# add arrow breaks
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             arrowCol = 'orange',
             arrowType = 'closed',
             arrowBreak = 0.1)

# draw specific transcript
p1 &lt;- trancriptVis(gtfFile = gtf,
                   gene = 'Commd7')

p2 &lt;- trancriptVis(gtfFile = gtf,
                   gene = 'Commd7',
                   myTranscript = c('ENSMUST00000071852','ENSMUST00000109782'))

# combine
cowplot::plot_grid(p1,p2,ncol = 2,align = 'hv')
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
