<!DOCTYPE html><html lang="en"><head><title>Help for package TransProR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TransProR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#add_boxplot'><p>Add a boxplot layer to a 'ggtree' plot</p></a></li>
<li><a href='#add_new_tile_layer'><p>Add a new tile layer with dynamic scales to a 'ggtree' plot</p></a></li>
<li><a href='#adjust_alpha_scale'><p>Adjust Alpha Scale for Data Visualization</p></a></li>
<li><a href='#adjust_color_tone'><p>Adjust Color Tone by Modifying Saturation and Luminance</p></a></li>
<li><a href='#adjust_export_pathway'><p>Adjust and Export Pathway Analysis Results</p></a></li>
<li><a href='#all_degs_venn'><p>All DEGs Venn Diagram Data</p></a></li>
<li><a href='#circos_fruits'><p>Add multiple layers to a 'ggtree' plot for visualizing gene expression and enrichment data</p></a></li>
<li><a href='#Combat_Normal'><p>Process and Correct Batch Effects in TCGA's normal tissue and GTEX Data</p></a></li>
<li><a href='#combat_tumor'><p>Process and Correct Batch Effects in Tumor Data</p></a></li>
<li><a href='#compare_merge'><p>Compare and merge specific columns from two DEG data frames</p></a></li>
<li><a href='#Contrast_Venn'><p>Function to Create a Venn Diagram of DEGs with Custom Colors</p></a></li>
<li><a href='#create_base_plot'><p>Create a base plot with gene expression data on a phylogenetic tree</p></a></li>
<li><a href='#deg_filter'><p>Function to Filter Differentially Expressed Genes (DEGs)</p></a></li>
<li><a href='#DESeq2_analyze'><p>Differential Gene Expression Analysis using 'DESeq2'</p></a></li>
<li><a href='#drawLegends'><p>Draw Dual-Sided Legends on a Plot</p></a></li>
<li><a href='#edgeR_analyze'><p>Differential Gene Expression Analysis using 'edgeR'</p></a></li>
<li><a href='#enrich_circo_bar'><p>Combine and Visualize Data with Circular Bar Chart</p></a></li>
<li><a href='#enrich_polar_bubble'><p>Enrichment Polar Bubble Plot</p></a></li>
<li><a href='#enrichment_circlize'><p>Draw Chord Diagram with Legends</p></a></li>
<li><a href='#enrichment_spiral_plots'><p>Create Spiral Plots with Legends Using 'spiralize' and 'ComplexHeatmap'</p></a></li>
<li><a href='#extract_descriptions_counts'><p>Extract and Count Descriptions with Specified Color</p></a></li>
<li><a href='#extract_ntop_pathways'><p>Extract and Store Top Pathways for Each Sample</p></a></li>
<li><a href='#extract_positive_pathways'><p>Extract Positive Pathways from SSGSEA Results and Select Random Samples</p></a></li>
<li><a href='#facet_density_foldchange'><p>Create faceted high-density region plots with optional points and density contours</p></a></li>
<li><a href='#filter_diff_genes'><p>Filter Differentially Expressed Genes</p></a></li>
<li><a href='#four_degs_venn'><p>Function to Create a Venn Diagram of DEGs</p></a></li>
<li><a href='#gather_graph_edge'><p>Gather graph edge from data frame</p>
Please note that this function is from the 'ggraph' package and has not been altered in functionality,
but it has been optimized and iterated.
It is not original content of 'TransProR'.
However, since 'ggraph' caused frequent GitHub Action errors during the creation of 'TransProR',
the author directly referenced the involved functions in 'TransProR'.
This is not the author's original creation. All users please be aware!</a></li>
<li><a href='#gather_graph_node'><p>Gather graph nodes from a data frame</p>
Please note that this function is from the 'ggraph' package and has not been altered in functionality,
but it has been optimized and iterated.
It is not original content of 'TransProR'.
However, since 'ggraph' caused frequent GitHub Action errors during the creation of 'TransProR',
the author directly referenced the involved functions in 'TransProR'.
This is not the author's original creation. All users please be aware!</a></li>
<li><a href='#gene_color'><p>Merge Genes with Color Information Based on Up/Down Regulation</p></a></li>
<li><a href='#gene_highlights'><p>Add gene highlights to a ggtree object</p></a></li>
<li><a href='#gene_map_pathway'><p>Create Pathway-Gene Mapping Data Frame</p></a></li>
<li><a href='#get_gtex_exp'><p>Get GTEx Expression Data for Specific Organ</p></a></li>
<li><a href='#get_tcga_exp'><p>TCGA Expression Data Processing</p></a></li>
<li><a href='#gtree'><p>Phylogenetic Tree Object</p></a></li>
<li><a href='#highlight_by_node'><p>Highlight Nodes in a Phylogenetic Tree with Custom Fill Colors and Transparency</p></a></li>
<li><a href='#highlight_genes'><p>Add Highlights for Genes on a Phylogenetic Tree</p></a></li>
<li><a href='#limma_analyze'><p>Differential Gene Expression Analysis using limma and voom</p></a></li>
<li><a href='#log_transform'><p>Log transformation decision and application on data</p></a></li>
<li><a href='#merge_density_foldchange'><p>Create high-density region plot with optional points, density rugs, and contours</p></a></li>
<li><a href='#merge_gtex_tcga'><p>Merge gene expression data from GTEx and TCGA datasets</p></a></li>
<li><a href='#merge_id_position'><p>Merge Data Frames by Common Row Names with Additional Columns</p></a></li>
<li><a href='#merge_method_color'><p>Merge Data Frames with Specific Method and Color Columns</p></a></li>
<li><a href='#new_ggraph'><p>Generate a graphical representation of pathway gene maps</p></a></li>
<li><a href='#pathway_count'><p>Count Genes Present in Pathways Above a Threshold</p></a></li>
<li><a href='#pathway_description'><p>Describe Genes Present in Selected Pathways</p></a></li>
<li><a href='#prep_deseq2'><p>Prepare DESeq2 data for plotting</p></a></li>
<li><a href='#prep_edgeR'><p>Prepare edgeR DEG data for plotting</p></a></li>
<li><a href='#prep_limma'><p>Prepare limma-voom DEG data for plotting</p></a></li>
<li><a href='#prep_wilcoxon'><p>Prepare Wilcoxon DEG data for plotting</p></a></li>
<li><a href='#process_heatdata'><p>Process Heatmap Data with Various Selection Options</p></a></li>
<li><a href='#seek_gtex_organ'><p>Load and Process GTEX Phenotype Data to Retrieve Primary Site Counts</p></a></li>
<li><a href='#selectPathways'><p>Randomly Select Pathways with Limited Word Count</p></a></li>
<li><a href='#spiral_newrle'><p>Render a Spiral Plot Using Run-Length Encoding</p></a></li>
<li><a href='#Wilcoxon_analyze'><p>Differential Gene Expression Analysis Using Wilcoxon Rank-Sum Test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis and Visualization of Multi-Omics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dongyue Yu &lt;yudongyue@mail.nankai.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A tool for comprehensive transcriptomic data analysis, with a focus on transcript-level data preprocessing, expression profiling, differential expression analysis, and functional enrichment. It enables researchers to identify key biological processes, disease biomarkers, and gene regulatory mechanisms. 'TransProR' is aimed at researchers and bioinformaticians working with RNA-Seq data, providing an intuitive framework for in-depth analysis and visualization of transcriptomic datasets. The package includes comprehensive documentation and usage examples to guide users through the entire analysis pipeline. The differential expression analysis methods incorporated in the package include 'limma' (Ritchie et al., 2015, &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkv007">doi:10.1093/nar/gkv007</a>&gt;; Smyth, 2005, &lt;<a href="https://doi.org/10.1007%2F0-387-29362-0_23">doi:10.1007/0-387-29362-0_23</a>&gt;), 'edgeR' (Robinson et al., 2010, &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtp616">doi:10.1093/bioinformatics/btp616</a>&gt;), 'DESeq2' (Love et al., 2014, &lt;<a href="https://doi.org/10.1186%2Fs13059-014-0550-8">doi:10.1186/s13059-014-0550-8</a>&gt;), and Wilcoxon tests (Li et al., 2022, &lt;<a href="https://doi.org/10.1186%2Fs13059-022-02648-4">doi:10.1186/s13059-022-02648-4</a>&gt;), providing flexible and robust approaches to RNA-Seq data analysis. For more information, refer to the package vignettes and related publications.</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr, stats, dplyr, rlang, tibble, sva, DESeq2, utils,
edgeR, limma, ggplot2, ggVennDiagram, ggdensity, ggpubr,
ggtree, hrbrthemes, grid, ggraph, tidygraph, tidyr, stringr,
geomtextpath, ggalt, ggnewscale, Hmisc, circlize, graphics,
spiralize, ComplexHeatmap, grDevices</td>
</tr>
<tr>
<td>Suggests:</td>
<td>prettydoc, knitr, ggtreeExtra, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/SSSYDYSSS/TransProRBook">https://github.com/SSSYDYSSS/TransProRBook</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/SSSYDYSSS/TransProR/issues">https://github.com/SSSYDYSSS/TransProR/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-17 19:12:16 UTC; Yu</td>
</tr>
<tr>
<td>Author:</td>
<td>Dongyue Yu <a href="https://orcid.org/0000-0002-7041-2073"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-18 09:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='add_boxplot'>Add a boxplot layer to a 'ggtree' plot</h2><span id='topic+add_boxplot'></span>

<h3>Description</h3>

<p>This function adds a boxplot layer to an existing &lsquo;ggtree' plot object using ggtreeExtra&rsquo;s geom_fruit for boxplots.
It is primarily used to display statistical summaries of the data related to gene expressions or other metrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_boxplot(
  p,
  data,
  fill_color = "#f28131",
  alpha = 0.6,
  offset = 0.22,
  pwidth = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_boxplot_+3A_p">p</code></td>
<td>
<p>An existing ggtree plot object.</p>
</td></tr>
<tr><td><code id="add_boxplot_+3A_data">data</code></td>
<td>
<p>A data frame containing the data to be plotted. Expected to have columns for 'Sample' and 'value'.</p>
</td></tr>
<tr><td><code id="add_boxplot_+3A_fill_color">fill_color</code></td>
<td>
<p>A character string specifying the fill color for the boxplots. Default is &quot;#f28131&quot;.</p>
</td></tr>
<tr><td><code id="add_boxplot_+3A_alpha">alpha</code></td>
<td>
<p>Numeric value for the transparency of the boxplots. Default is 0.6.</p>
</td></tr>
<tr><td><code id="add_boxplot_+3A_offset">offset</code></td>
<td>
<p>Numeric value, the position of the boxplot on the x-axis relative to its gene name. Default is 0.22.</p>
</td></tr>
<tr><td><code id="add_boxplot_+3A_pwidth">pwidth</code></td>
<td>
<p>Numeric value, the width of the boxplot. Default is 0.5.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggtree' plot object with the added boxplot layer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Check and load required packages
if (requireNamespace("ggtreeExtra", quietly = TRUE) &amp;&amp;
 requireNamespace("ggplot2", quietly = TRUE)) {
  library(ggtreeExtra)
  library(ggplot2)

  file_path &lt;- system.file("extdata", "p_tree_test.rds", package = "TransProR")
  p &lt;- readRDS(file_path)

  # Create boxplot data frame
  boxplot_data &lt;- data.frame(
    Sample = rep(c("Species_A", "Species_B", "Species_C", "Species_D"), each = 30),
    value = c(
      rnorm(30, mean = 5, sd = 1),   # Data for Species_A
      rnorm(30, mean = 7, sd = 1.5), # Data for Species_B
      rnorm(30, mean = 6, sd = 1.2), # Data for Species_C
      rnorm(30, mean = 8, sd = 1.3)  # Data for Species_D
    )
  )

  # Call add_boxplot function to add boxplot layer
  p_with_boxplot &lt;- add_boxplot(p, boxplot_data)
} else {
  message("Required packages 'ggtreeExtra' and 'ggplot2' are not installed.")
}


</code></pre>

<hr>
<h2 id='add_new_tile_layer'>Add a new tile layer with dynamic scales to a 'ggtree' plot</h2><span id='topic+add_new_tile_layer'></span>

<h3>Description</h3>

<p>This function adds a new tile layer to an existing 'ggtree' plot object, allowing for separate scales for fill
and alpha transparency. This is useful when you want to add additional data layers without interfering with
the existing scales in the plot. It utilizes the 'ggnewscale' package to reset scales for new layers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_new_tile_layer(
  p,
  data,
  gene_colors,
  gene_label,
  alpha_value = c(0.3, 0.9),
  offset = 0.02,
  pwidth = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_new_tile_layer_+3A_p">p</code></td>
<td>
<p>An existing ggtree plot object.</p>
</td></tr>
<tr><td><code id="add_new_tile_layer_+3A_data">data</code></td>
<td>
<p>A data frame containing the data to be plotted. Expected to have columns for 'Sample', 'Gene', and 'value'.</p>
</td></tr>
<tr><td><code id="add_new_tile_layer_+3A_gene_colors">gene_colors</code></td>
<td>
<p>A named vector of colors for genes.</p>
</td></tr>
<tr><td><code id="add_new_tile_layer_+3A_gene_label">gene_label</code></td>
<td>
<p>A character string used as a label in the legend for the genes.</p>
</td></tr>
<tr><td><code id="add_new_tile_layer_+3A_alpha_value">alpha_value</code></td>
<td>
<p>A numeric or named vector for setting the alpha scale based on values.</p>
</td></tr>
<tr><td><code id="add_new_tile_layer_+3A_offset">offset</code></td>
<td>
<p>Numeric value, the position of the tile on the x-axis relative to its gene name. Default is 0.02.</p>
</td></tr>
<tr><td><code id="add_new_tile_layer_+3A_pwidth">pwidth</code></td>
<td>
<p>Numeric value, the width of the tile. Default is 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggtree' plot object with the added tile layer and new scales.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Check and load required packages
if (requireNamespace("ggtreeExtra", quietly = TRUE) &amp;&amp;
 requireNamespace("ggplot2", quietly = TRUE) &amp;&amp;
  requireNamespace("ggnewscale", quietly = TRUE)) {
  library(ggtreeExtra)
  library(ggplot2)
  library(ggnewscale)

  file_path &lt;- system.file("extdata", "p_tree_test.rds", package = "TransProR")
  p &lt;- readRDS(file_path)

  # Create new expression data
  new_expression_data &lt;- data.frame(
    Sample = rep(c("Species_A", "Species_B", "Species_C", "Species_D"), each = 3),
    Gene = rep(c("Gene6", "Gene7", "Gene8"), times = 4),
    Value = runif(12, min = 0, max = 1)  # Randomly generate expression values between 0 and 1
  )

  # Define new gene colors
  new_gene_colors &lt;- c(
    Gene6 = "#0b5f63",
    Gene7 = "#074d41",
    Gene8 = "#1f5e27"
  )

  # Define gene label and alpha values
  gene_label &lt;- "New Genes"
  alpha_value &lt;- c(0.3, 0.9)

  # Add new tile layer
  p_with_new_layer &lt;- add_new_tile_layer(
    p,
    new_expression_data,
    new_gene_colors,
    gene_label,
    alpha_value,
    offset = 0.02,
    pwidth = 2
  )
} else {
  message("Required packages 'ggtreeExtra', 'ggplot2', and 'ggnewscale' are not installed.")
}


</code></pre>

<hr>
<h2 id='adjust_alpha_scale'>Adjust Alpha Scale for Data Visualization</h2><span id='topic+adjust_alpha_scale'></span>

<h3>Description</h3>

<p>This function dynamically adjusts the transparency scale for visualizations,
especially useful when the range of data values varies significantly across different sources.
It modifies the transparency scale based on the range of values present in the data,
ensuring that the visualization accurately reflects variations within the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_alpha_scale(data, name, range = c(0.2, 0.8))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjust_alpha_scale_+3A_data">data</code></td>
<td>
<p>A data frame containing the values for which the alpha scale is to be adjusted.</p>
</td></tr>
<tr><td><code id="adjust_alpha_scale_+3A_name">name</code></td>
<td>
<p>Character string that will be used as the title of the legend in the plot.</p>
</td></tr>
<tr><td><code id="adjust_alpha_scale_+3A_range">range</code></td>
<td>
<p>Numeric vector of length 2 specifying the range of alpha values, defaults to c(0.2, 0.8).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 alpha scale adjustment layer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Assuming 'data' is a DataFrame with a 'value' column
plot_data &lt;- data.frame(value = c(10, 20, 30, 40, 50))
ggplot2::ggplot(plot_data, ggplot2::aes(x = 1:nrow(plot_data), y = value)) +
  ggplot2::geom_point(ggplot2::aes(alpha = value)) +
  adjust_alpha_scale(plot_data, "Transparency Scale")

</code></pre>

<hr>
<h2 id='adjust_color_tone'>Adjust Color Tone by Modifying Saturation and Luminance</h2><span id='topic+adjust_color_tone'></span>

<h3>Description</h3>

<p>This function adjusts the saturation and luminance of a given color. It works by converting
the color from RGB to Luv color space, applying the scaling factors to the saturation and luminance,
and then converting it back to RGB.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_color_tone(color, saturation_scale, luminance_scale)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjust_color_tone_+3A_color">color</code></td>
<td>
<p>A color in hexadecimal format (e.g., &quot;#FF0000&quot;) or a valid R color name.</p>
</td></tr>
<tr><td><code id="adjust_color_tone_+3A_saturation_scale">saturation_scale</code></td>
<td>
<p>Numeric, the scaling factor for saturation (values &lt; 1 decrease saturation, values &gt; 1 increase saturation).</p>
</td></tr>
<tr><td><code id="adjust_color_tone_+3A_luminance_scale">luminance_scale</code></td>
<td>
<p>Numeric, the scaling factor for luminance (values &lt; 1 darken the color, values &gt; 1 lighten the color).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a color in hexadecimal format adjusted according to the provided scales.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  adjusted_color &lt;- adjust_color_tone("#FF0000", saturation_scale = 0.8, luminance_scale = 1.2)
  print(adjusted_color)

</code></pre>

<hr>
<h2 id='adjust_export_pathway'>Adjust and Export Pathway Analysis Results</h2><span id='topic+adjust_export_pathway'></span>

<h3>Description</h3>

<p>This function processes a dataframe containing fgsea results. It adjusts pathway names
by removing underscores, converting to lowercase, then capitalizing the first letter,
and joining the components with spaces. It selects and merges the top upregulated
and downregulated pathways based on enrichment score (ES) and p-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_export_pathway(fgseaRes, nTop = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjust_export_pathway_+3A_fgseares">fgseaRes</code></td>
<td>
<p>Dataframe containing fgsea results with columns 'pathway', 'ES', and 'pval'.</p>
</td></tr>
<tr><td><code id="adjust_export_pathway_+3A_ntop">nTop</code></td>
<td>
<p>Integer, number of top pathways to select based on the p-value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing combined top upregulated and downregulated pathways.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a synthetic fgseaRes dataframe
fgseaRes &lt;- data.frame(
 pathway = c("KEGG_APOPTOSIS",
             "GO_CELL_CYCLE",
             "REACTOME_DNA_REPAIR",
             "KEGG_METABOLISM",
             "GO_TRANSPORT"),
 ES = c(0.45, -0.22, 0.56, -0.35, 0.33),
 pval = c(0.001, 0.02, 0.0003, 0.05, 0.01)
)

# Run the function to get top pathways
result &lt;- adjust_export_pathway(fgseaRes = fgseaRes, nTop = 2)

</code></pre>

<hr>
<h2 id='all_degs_venn'>All DEGs Venn Diagram Data</h2><span id='topic+all_degs_venn'></span>

<h3>Description</h3>

<p>A dataset containing the differentially expressed genes (DEGs) from four different statistical analysis methods: DESeq2, edgeR, limma, and Wilcoxon test.
This dataset is used for generating Venn diagrams to compare the overlap of DEGs identified by different methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(all_degs_venn)
</code></pre>


<h3>Format</h3>

<p>A list with the following components:
</p>

<dl>
<dt>DESeq2</dt><dd><p>A vector of gene IDs or gene symbols identified as DEGs by the DESeq2 method.</p>
</dd>
<dt>edgeR</dt><dd><p>A vector of gene IDs or gene symbols identified as DEGs by the edgeR method.</p>
</dd>
<dt>limma</dt><dd><p>A vector of gene IDs or gene symbols identified as DEGs by the limma method. </p>
</dd>
<dt>Wilcoxon_test</dt><dd><p>A vector of gene IDs or gene symbols identified as DEGs by the Wilcoxon test method. </p>
</dd>
</dl>



<h3>Source</h3>

<p>The data was derived from differential expression analyses performed on a gene expression dataset using four commonly used statistical methods (DESeq2, edgeR, limma, and Wilcoxon test).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(all_degs_venn)
# Example of plotting a Venn diagram using the dataset

edge_colors &lt;- c("#1b62bb","#13822e","#332c3a","#9e2d39")
name_color &lt;- c("#1b64bb","#13828e","#337c3a","#9e9d39")
fill_colors &lt;- c("#e3f2fa", "#0288d1")

Contrast_degs_venn &lt;- Contrast_Venn(all_degs_venn, edge_colors, name_color, fill_colors)

</code></pre>

<hr>
<h2 id='circos_fruits'>Add multiple layers to a 'ggtree' plot for visualizing gene expression and enrichment data</h2><span id='topic+circos_fruits'></span>

<h3>Description</h3>

<p>This function sequentially adds multiple layers to a 'ggtree' plot, including gene expression data, boxplots for statistical
summaries, and additional tile layers for pathway enrichment scores from SSGSEA and GSVA analyses. It utilizes separate
functions for adding each type of layer and allows for the specification of gene colors as well as adjustments in aesthetics
for each layer. The function is designed to work with specific data structures and assumes all functions for adding layers
are defined and available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos_fruits(
  p,
  long_format_HeatdataDeseq,
  ssgsea_kegg_HeatdataDeseq,
  gsva_kegg_HeatdataDeseq,
  gene_colors
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos_fruits_+3A_p">p</code></td>
<td>
<p>A 'ggtree' plot object to which the data and layers will be added.</p>
</td></tr>
<tr><td><code id="circos_fruits_+3A_long_format_heatdatadeseq">long_format_HeatdataDeseq</code></td>
<td>
<p>A data frame containing gene expression data with columns for 'Samples', 'Genes', and 'Values'.</p>
</td></tr>
<tr><td><code id="circos_fruits_+3A_ssgsea_kegg_heatdatadeseq">ssgsea_kegg_HeatdataDeseq</code></td>
<td>
<p>A data frame containing SSGSEA analysis results with columns for 'Samples', 'Genes', and 'Values'.</p>
</td></tr>
<tr><td><code id="circos_fruits_+3A_gsva_kegg_heatdatadeseq">gsva_kegg_HeatdataDeseq</code></td>
<td>
<p>A data frame containing GSVA analysis results with columns for 'Samples', 'Genes', and 'Values'.</p>
</td></tr>
<tr><td><code id="circos_fruits_+3A_gene_colors">gene_colors</code></td>
<td>
<p>A named vector of colors for genes, used for coloring tiles in different layers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggtree' plot object with multiple layers added for comprehensive visualization.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Check and load required packages
if (requireNamespace("ggtreeExtra", quietly = TRUE) &amp;&amp;
 requireNamespace("ggplot2", quietly = TRUE)) {
  library(ggtreeExtra)
  library(ggplot2)

  # Example data for gene expression, SSGSEA, and GSVA
  file_path &lt;- system.file("extdata", "p_tree_test.rds", package = "TransProR")
  p &lt;- readRDS(file_path)

  # Create gene expression data frame (long_format_HeatdataDeseq)
  long_format_HeatdataDeseq &lt;- data.frame(
    Sample = rep(c("Species_A", "Species_B", "Species_C", "Species_D"), each = 5),
    Genes = rep(paste0("Gene", 1:5), times = 4),
    Value = runif(20, min = 0, max = 1)  # Randomly generate expression values between 0 and 1
  )

  # Create SSGSEA analysis results data frame (ssgsea_kegg_HeatdataDeseq)
  ssgsea_kegg_HeatdataDeseq &lt;- data.frame(
    Sample = rep(c("Species_A", "Species_B", "Species_C", "Species_D"), each = 3),
    Genes = rep(c("Pathway1", "Pathway2", "Pathway3"), times = 4),
    Value = runif(12, min = 0, max = 1)  # Randomly generate enrichment scores between 0 and 1
  )

  # Create GSVA analysis results data frame (gsva_kegg_HeatdataDeseq)
  gsva_kegg_HeatdataDeseq &lt;- data.frame(
    Sample = rep(c("Species_A", "Species_B", "Species_C", "Species_D"), each = 4),
    Genes = rep(c("PathwayA", "PathwayB", "PathwayC", "PathwayD"), times = 4),
    Value = runif(16, min = 0, max = 1)  # Randomly generate enrichment scores between 0 and 1
  )

  # Define gene and pathway colors (named vector), including all genes and pathways
  gene_colors &lt;- c(
    # Genes for gene expression
    Gene1 = "#491588",
    Gene2 = "#301b8d",
    Gene3 = "#1a237a",
    Gene4 = "#11479c",
    Gene5 = "#0a5797",
    # Pathways for SSGSEA
    Pathway1 = "#0b5f63",
    Pathway2 = "#074d41",
    Pathway3 = "#1f5e27",
    # Pathways for GSVA
    PathwayA = "#366928",
    PathwayB = "#827729",
    PathwayC = "#a1d99b",
    PathwayD = "#c7e9c0"
  )

  # Call circos_fruits function to add multiple layers
  final_plot &lt;- circos_fruits(
    p,
    long_format_HeatdataDeseq,
    ssgsea_kegg_HeatdataDeseq,
    gsva_kegg_HeatdataDeseq,
    gene_colors
  )
} else {
  message("Required packages 'ggtreeExtra' and 'ggplot2' are not installed.")
}


</code></pre>

<hr>
<h2 id='Combat_Normal'>Process and Correct Batch Effects in TCGA's normal tissue and GTEX Data</h2><span id='topic+Combat_Normal'></span>

<h3>Description</h3>

<p>The function first extracts histological types from the provided TCGA's normal tissue data set.
After displaying these types, the user is prompted to input specific types to retain.
The data is then filtered based on this input.
The GTEX and TCGA's normal tissue datasets are then combined and batch corrected.
</p>
<p>Note: This function assumes that TCGA's normal samples and GTEX samples represent different batches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Combat_Normal(
  TCGA_normal_data_path,
  gtex_data_path,
  CombatNormal_output_path,
  auto_mode = FALSE,
  default_input = "11,12"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Combat_Normal_+3A_tcga_normal_data_path">TCGA_normal_data_path</code></td>
<td>
<p>The path to the tumor data stored in an RDS file.</p>
</td></tr>
<tr><td><code id="Combat_Normal_+3A_gtex_data_path">gtex_data_path</code></td>
<td>
<p>The path to the GTEX data stored in an RDS file.</p>
</td></tr>
<tr><td><code id="Combat_Normal_+3A_combatnormal_output_path">CombatNormal_output_path</code></td>
<td>
<p>A character string specifying the path where the output RDS file will be saved.</p>
</td></tr>
<tr><td><code id="Combat_Normal_+3A_auto_mode">auto_mode</code></td>
<td>
<p>Logical. If set to TRUE, the function will not prompt the user for input and
will instead use the values provided in default_input. Default is FALSE.</p>
</td></tr>
<tr><td><code id="Combat_Normal_+3A_default_input">default_input</code></td>
<td>
<p>Character string. When auto_mode is TRUE, this parameter specifies the default
TGCA's normal tissue types to be retained. It should be provided as a comma-separated string (e.g., &quot;11,12&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a TCGA's normal tissue data set and a pre-saved GTEX data set, asks the user
for specific TCGA's normal tissue types to retain, then merges the two datasets. The merged dataset
is then corrected for batch effects using the ComBat_seq function from the 'sva' package.
</p>
<p>The ComBat_seq function from the 'sva' package is used to correct batch effects.
The function requires the 'sva' package to be installed and loaded externally.
</p>
<p>The example code uses 'tempfile()' to generate temporary paths dynamically during execution.
These paths are valid during the 'R CMD check' process, even if no actual files exist,
because &lsquo;tempfile()' generates a unique file path that does not depend on the user&rsquo;s file system.
Using 'tempfile()' ensures that the example code does not rely on specific external files and
avoids errors during 'R CMD check'. CRAN review checks for documentation correctness
and syntax parsing, not the existence of actual files, as long as the example code is syntactically valid.
</p>


<h3>Value</h3>

<p>A data.frame with corrected values after the ComBat_seq adjustment. Note that this function also saves the
combat_count_df data as an RDS file at the specified output path.
</p>


<h3>See Also</h3>

<p><code><a href="sva.html#topic+ComBat_seq">ComBat_seq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>TCGA_normal_file &lt;- system.file("extdata",
                                "SKCM_Skin_TCGA_exp_normal_test.rds",
                                package = "TransProR")
gtex_file &lt;- system.file("extdata", "Skin_SKCM_Gtex_test.rds", package = "TransProR")
output_file &lt;- file.path(tempdir(), "SKCM_Skin_Combat_Normal_TCGA_GTEX_count.rds")

SKCM_Skin_Combat_Normal_TCGA_GTEX_count &lt;- Combat_Normal(
  TCGA_normal_data_path = TCGA_normal_file,
  gtex_data_path = gtex_file,
  CombatNormal_output_path = output_file,
  auto_mode = TRUE,
  default_input = "skip"
)
head(SKCM_Skin_Combat_Normal_TCGA_GTEX_count)[1:5, 1:5]

</code></pre>

<hr>
<h2 id='combat_tumor'>Process and Correct Batch Effects in Tumor Data</h2><span id='topic+combat_tumor'></span>

<h3>Description</h3>

<p>The function first extracts histological types from the provided tumor data set.
After displaying these types, the user is prompted to input specific types to retain.
The data is then filtered based on this input.
</p>
<p>Note: This example assumes that different tumor types represent different batches in a general sense.
Users need to adjust the batch and group vectors based on real-life scenarios.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combat_tumor(
  tumor_data_path,
  CombatTumor_output_path,
  auto_mode = FALSE,
  default_input = "01,06"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combat_tumor_+3A_tumor_data_path">tumor_data_path</code></td>
<td>
<p>The path to the tumor data stored in an RDS file.</p>
</td></tr>
<tr><td><code id="combat_tumor_+3A_combattumor_output_path">CombatTumor_output_path</code></td>
<td>
<p>A character string specifying the path where the output RDS file will be saved.</p>
</td></tr>
<tr><td><code id="combat_tumor_+3A_auto_mode">auto_mode</code></td>
<td>
<p>Logical. If set to TRUE, the function will not prompt the user for input and
will instead use the values provided in default_input. Default is FALSE.</p>
</td></tr>
<tr><td><code id="combat_tumor_+3A_default_input">default_input</code></td>
<td>
<p>Character string. When auto_mode is TRUE, this parameter specifies the default
tumor types to be retained. It should be provided as a comma-separated string (e.g., &quot;01,06&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a tumor data set, asks the user for specific tumor types to retain,
and then corrects for batch effects using the ComBat_seq function from the 'sva' package.
</p>
<p>The ComBat_seq function from the sva package is used to correct batch effects.
The function requires the 'sva' package to be installed and loaded externally.
</p>


<h3>Value</h3>

<p>A data.frame with corrected values after the ComBat_seq adjustment. Note that this function also saves the
combat_count_df data as an RDS file at the specified output path.
</p>


<h3>See Also</h3>

<p><code><a href="sva.html#topic+ComBat_seq">ComBat_seq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tumor_file &lt;- system.file("extdata",
                          "SKCM_Skin_TCGA_exp_tumor_test.rds",
                          package = "TransProR")
output_file &lt;- file.path(tempdir(), "SKCM_combat_count.rds")

  SKCM_combat_count &lt;- combat_tumor(
  tumor_data_path = tumor_file,
  CombatTumor_output_path = output_file,
  auto_mode = TRUE,
  default_input = "01,06"
)

head(SKCM_combat_count)[1:5, 1:5]

</code></pre>

<hr>
<h2 id='compare_merge'>Compare and merge specific columns from two DEG data frames</h2><span id='topic+compare_merge'></span>

<h3>Description</h3>

<p>This function takes two DEG data frames, inner joins them by a specified gene column,
checks if a specified column is identical across both data frames, and merges them if they are.
The resulting data frame will have a merged column named after the compared column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_merge(df1, df2, by_gene, compare_col, suffixes, df_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compare_merge_+3A_df1">df1</code></td>
<td>
<p>First data frame.</p>
</td></tr>
<tr><td><code id="compare_merge_+3A_df2">df2</code></td>
<td>
<p>Second data frame.</p>
</td></tr>
<tr><td><code id="compare_merge_+3A_by_gene">by_gene</code></td>
<td>
<p>Column name by which to join the data frames, typically &quot;Gene&quot;.</p>
</td></tr>
<tr><td><code id="compare_merge_+3A_compare_col">compare_col</code></td>
<td>
<p>Column to compare for identity, which will also be the name of the merged column.</p>
</td></tr>
<tr><td><code id="compare_merge_+3A_suffixes">suffixes</code></td>
<td>
<p>Suffixes to use for non-identical column names in the joined data frame.</p>
</td></tr>
<tr><td><code id="compare_merge_+3A_df_name">df_name</code></td>
<td>
<p>Name to assign to the resulting data frame for identification.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with processed columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create simulated DESeq2 data
DEG_deseq2 &lt;- data.frame(
  Gene = c("Gene1", "Gene2", "Gene3", "Gene4", "Gene5"),
  change = c("up", "down", "no_change", "up", "down"),
  log2FoldChange = c(2.5, -3.2, 0.1, 1.8, -2.5),
  pvalue = c(0.01, 0.05, 0.9, 0.02, 0.03)
)

# Display the first 5 rows of the DESeq2 data
head(DEG_deseq2, 5)

# Create simulated edgeR data
DEG_edgeR &lt;- data.frame(
  Gene = c("Gene1", "Gene2", "Gene3", "Gene4", "Gene5"),
  change = c("up", "down", "no_change", "no_change", "up"),
  log2FoldChange = c(2.3, -3.1, 0.2, 0.1, 2.7),
  pvalue = c(0.02, 0.04, 0.8, 0.6, 0.01)
)

# Display the first 5 rows of the edgeR data
head(DEG_edgeR, 5)

# Merge the DESeq2 and edgeR data
deseq2_edgeR &lt;- compare_merge(
  df1 = DEG_deseq2,
  df2 = DEG_edgeR,
  by_gene = "Gene",
  compare_col = "change",
  suffixes = c("_1", "_2"),
  df_name = "deseq2_edgeR"
)

</code></pre>

<hr>
<h2 id='Contrast_Venn'>Function to Create a Venn Diagram of DEGs with Custom Colors</h2><span id='topic+Contrast_Venn'></span>

<h3>Description</h3>

<p>This function creates a Venn Diagram using the 'ggVennDiagram' package.
It allows customization of various aesthetic elements of the diagram, including colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Contrast_Venn(
  all_degs_venn,
  edge_colors,
  name_color,
  fill_colors,
  label_size = 4,
  edge_size = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Contrast_Venn_+3A_all_degs_venn">all_degs_venn</code></td>
<td>
<p>A list of DEG sets for Venn Diagram creation.</p>
</td></tr>
<tr><td><code id="Contrast_Venn_+3A_edge_colors">edge_colors</code></td>
<td>
<p>A vector of colors for the edges of the Venn Diagram sets.</p>
</td></tr>
<tr><td><code id="Contrast_Venn_+3A_name_color">name_color</code></td>
<td>
<p>A vector of colors for the names of the sets in the Venn Diagram.</p>
</td></tr>
<tr><td><code id="Contrast_Venn_+3A_fill_colors">fill_colors</code></td>
<td>
<p>A vector of two colors for the gradient fill of the Venn Diagram.</p>
</td></tr>
<tr><td><code id="Contrast_Venn_+3A_label_size">label_size</code></td>
<td>
<p>The size of the labels showing the number of elements in each set (default is 4).</p>
</td></tr>
<tr><td><code id="Contrast_Venn_+3A_edge_size">edge_size</code></td>
<td>
<p>The size of the edges of the Venn Diagram sets (default is 3).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggplot' object representing the Venn Diagram.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("all_degs_venn", package = "TransProR")

edge_colors &lt;- c("#1b62bb","#13822e","#332c3a","#9e2d39")
name_color &lt;- c("#1b64bb","#13828e","#337c3a","#9e9d39")
fill_colors &lt;- c("#e3f2fa", "#0288d1")

Contrast_degs_venn &lt;- Contrast_Venn(all_degs_venn, edge_colors, name_color, fill_colors)
</code></pre>

<hr>
<h2 id='create_base_plot'>Create a base plot with gene expression data on a phylogenetic tree</h2><span id='topic+create_base_plot'></span>

<h3>Description</h3>

<p>This function creates a base plot using 'ggtree' and 'ggtreeExtra' libraries, adding gene expression
data as colored tiles to the plot. It allows for dynamic coloring of the genes and includes
adjustments for alpha transparency based on the expression value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_base_plot(p, data, gene_colors, gene_label = "Gene")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_base_plot_+3A_p">p</code></td>
<td>
<p>A ggtree plot object to which the data will be added.</p>
</td></tr>
<tr><td><code id="create_base_plot_+3A_data">data</code></td>
<td>
<p>A data frame containing gene expression data with columns for Samples, Genes, and Values.</p>
</td></tr>
<tr><td><code id="create_base_plot_+3A_gene_colors">gene_colors</code></td>
<td>
<p>A named vector of colors for genes.</p>
</td></tr>
<tr><td><code id="create_base_plot_+3A_gene_label">gene_label</code></td>
<td>
<p>A character string used as a label in the legend for the genes. Default is &quot;Gene&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggtree' plot object with the gene expression data added.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Check and load required packages
if (requireNamespace("ggtreeExtra", quietly = TRUE) &amp;&amp;
 requireNamespace("ggplot2", quietly = TRUE)) {
  library(ggtreeExtra)
  library(ggplot2)

  file_path &lt;- system.file("extdata", "p_tree_test.rds", package = "TransProR")
  p &lt;- readRDS(file_path)

  # Create gene expression data frame
  expression_data &lt;- data.frame(
    Sample = rep(c("Species_A", "Species_B", "Species_C", "Species_D"), each = 5),
    Gene = rep(paste0("Gene", 1:5), times = 4),
    Value = runif(20, min = 0, max = 1)  # Randomly generate expression values between 0 and 1
  )

  # Define gene colors (named vector)
  gene_colors &lt;- c(
    Gene1 = "#491588",
    Gene2 = "#301b8d",
    Gene3 = "#1a237a",
    Gene4 = "#11479c",
    Gene5 = "#0a5797"
  )

  # Call create_base_plot function to add gene expression data
  p &lt;- create_base_plot(p, expression_data, gene_colors)
} else {
  message("Required packages 'ggtreeExtra' and 'ggplot2' are not installed.")
}


</code></pre>

<hr>
<h2 id='deg_filter'>Function to Filter Differentially Expressed Genes (DEGs)</h2><span id='topic+deg_filter'></span>

<h3>Description</h3>

<p>This function filters out genes based on their expression change status.
It returns the names of genes which are not &quot;stable&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deg_filter(df)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="deg_filter_+3A_df">df</code></td>
<td>
<p>A data frame containing gene expression data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of gene names that are differentially expressed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>DEG_deseq2_file &lt;- system.file("extdata", "DEG_deseq2.rds", package = "TransProR")
DEG_deseq2 &lt;- readRDS(DEG_deseq2_file)
DEG_deseq2_test &lt;- deg_filter(DEG_deseq2)
</code></pre>

<hr>
<h2 id='DESeq2_analyze'>Differential Gene Expression Analysis using 'DESeq2'</h2><span id='topic+DESeq2_analyze'></span>

<h3>Description</h3>

<p>'DESeq2': Differential gene expression analysis based on the negative binomial distribution.
This function utilizes the 'DESeq2' package to conduct differential gene expression analysis.
It processes tumor and normal expression data, applies DESeq2 analysis,
and outputs the results along with information on gene expression changes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DESeq2_analyze(
  tumor_file,
  normal_file,
  output_file,
  logFC = 2.5,
  p_value = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DESeq2_analyze_+3A_tumor_file">tumor_file</code></td>
<td>
<p>Path to the tumor data file (RDS format).</p>
</td></tr>
<tr><td><code id="DESeq2_analyze_+3A_normal_file">normal_file</code></td>
<td>
<p>Path to the normal data file (RDS format).</p>
</td></tr>
<tr><td><code id="DESeq2_analyze_+3A_output_file">output_file</code></td>
<td>
<p>Path to save the output DEG data (RDS format).</p>
</td></tr>
<tr><td><code id="DESeq2_analyze_+3A_logfc">logFC</code></td>
<td>
<p>Threshold for log fold change.</p>
</td></tr>
<tr><td><code id="DESeq2_analyze_+3A_p_value">p_value</code></td>
<td>
<p>Threshold for p-value.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The DESeq2 methodology is based on modeling count data using a negative binomial distribution,
which allows for handling the variability observed in gene expression data, especially in
small sample sizes. This approach is well-suited for RNA-Seq data analysis.
</p>


<h3>Value</h3>

<p>A data frame of differential expression results.
</p>


<h3>References</h3>

<p>DESeq2:Differential gene expression analysis based on the negative binomial distribution.
For more information, visit the page:
https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Define file paths for tumor and normal data from the data folder
tumor_file &lt;- system.file("extdata",
                          "removebatch_SKCM_Skin_TCGA_exp_tumor_test.rds",
                          package = "TransProR")
normal_file &lt;- system.file("extdata",
                           "removebatch_SKCM_Skin_Normal_TCGA_GTEX_count_test.rds",
                           package = "TransProR")
output_file &lt;- file.path(tempdir(), "DEG_DESeq2.rds")

DEG_DESeq2 &lt;- DESeq2_analyze(
  tumor_file = tumor_file,
  normal_file = normal_file,
  output_file = output_file,
  2.5,
  0.01
)

# View the top 5 rows of the result
head(DEG_DESeq2, 5)

</code></pre>

<hr>
<h2 id='drawLegends'>Draw Dual-Sided Legends on a Plot</h2><span id='topic+drawLegends'></span>

<h3>Description</h3>

<p>This function creates two sets of legends, one on the left and one on the right side of a plot.
It displays color-coded legends with labels corresponding to different data categories.
Each legend entry consists of a colored rectangle and a text label. The left side legend has
text aligned to the right of the color block, while the right side legend has text aligned
to the left of the color block.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drawLegends(
  labels,
  colors,
  legend_width,
  x_positions,
  y_position,
  just_positions,
  text_alignments,
  font_size
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="drawLegends_+3A_labels">labels</code></td>
<td>
<p>Vector of labels for the legends.</p>
</td></tr>
<tr><td><code id="drawLegends_+3A_colors">colors</code></td>
<td>
<p>Vector of colors corresponding to the labels.</p>
</td></tr>
<tr><td><code id="drawLegends_+3A_legend_width">legend_width</code></td>
<td>
<p>The width of each legend viewport expressed in grid units.</p>
</td></tr>
<tr><td><code id="drawLegends_+3A_x_positions">x_positions</code></td>
<td>
<p>Numeric vector of length 2 specifying the x-positions of the left and right legends.</p>
</td></tr>
<tr><td><code id="drawLegends_+3A_y_position">y_position</code></td>
<td>
<p>The y-position common for both legends, expressed as a fraction of the plot height.</p>
</td></tr>
<tr><td><code id="drawLegends_+3A_just_positions">just_positions</code></td>
<td>
<p>List of two vectors, each specifying the horizontal and vertical justification for the legends.</p>
</td></tr>
<tr><td><code id="drawLegends_+3A_text_alignments">text_alignments</code></td>
<td>
<p>List of two character strings specifying text alignments for the legends ('left' or 'right').</p>
</td></tr>
<tr><td><code id="drawLegends_+3A_font_size">font_size</code></td>
<td>
<p>Numeric value specifying the font size for the legend labels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisible. This function is called for its side effects of drawing legends on a plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>labels &lt;- c("Label1", "Label2", "Label3", "Label4", "Label5", "Label6")
colors &lt;- c("#ff0000", "#00ff00", "#0000ff", "#ffff00", "#ff00ff", "#00ffff")

# Convert to 'unit' objects for grid
grid::grid.roundrect(
  x = grid::unit(0.5, "npc"),  # "npc" stands for normalized parent coordinates
  y = grid::unit(0.5, "npc"),
  width = grid::unit(0.1, "npc"),
  height = grid::unit(0.05, "npc"),
  gp = grid::gpar(fill = "red"),
  r = grid::unit(0.1, "npc")  # rounding radius
)

# Example of drawing legends with specific labels and colors
drawLegends(labels, colors, grid::unit(2, "cm"), c(0.225, 0.75), 0.5,
            list(c("left", "center"), c("right", "center")),
            list("right", "left"), 10)

</code></pre>

<hr>
<h2 id='edgeR_analyze'>Differential Gene Expression Analysis using 'edgeR'</h2><span id='topic+edgeR_analyze'></span>

<h3>Description</h3>

<p>This function performs differential gene expression analysis using the 'edgeR' package.
It reads tumor and normal expression data, merges them, filters low-expressed genes,
normalizes the data, performs edgeR analysis, and outputs the results along with information
on gene expression changes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edgeR_analyze(
  tumor_file,
  normal_file,
  output_file,
  logFC_threshold = 2.5,
  p_value_threshold = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="edgeR_analyze_+3A_tumor_file">tumor_file</code></td>
<td>
<p>Path to the tumor data file (RDS format).</p>
</td></tr>
<tr><td><code id="edgeR_analyze_+3A_normal_file">normal_file</code></td>
<td>
<p>Path to the normal data file (RDS format).</p>
</td></tr>
<tr><td><code id="edgeR_analyze_+3A_output_file">output_file</code></td>
<td>
<p>Path to save the output DEG data (RDS format).</p>
</td></tr>
<tr><td><code id="edgeR_analyze_+3A_logfc_threshold">logFC_threshold</code></td>
<td>
<p>Threshold for log fold change for marking up/down-regulated genes.</p>
</td></tr>
<tr><td><code id="edgeR_analyze_+3A_p_value_threshold">p_value_threshold</code></td>
<td>
<p>Threshold for p-value for filtering significant genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of differential expression results.
</p>


<h3>References</h3>

<p>edgeR: Differential analysis of sequence read count data.
For more information, visit the edgeR Bioconductor page:
https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Define file paths for tumor and normal data from the data folder
tumor_file &lt;- system.file("extdata",
                         "removebatch_SKCM_Skin_TCGA_exp_tumor_test.rds",
                         package = "TransProR")
normal_file &lt;- system.file("extdata",
                           "removebatch_SKCM_Skin_Normal_TCGA_GTEX_count_test.rds",
                           package = "TransProR")
output_file &lt;- file.path(tempdir(), "DEG_edgeR.rds")

DEG_edgeR &lt;- edgeR_analyze(
  tumor_file = tumor_file,
  normal_file = normal_file,
  output_file = output_file,
  2.5,
  0.01
)

# View the top 5 rows of the result
head(DEG_edgeR, 5)

</code></pre>

<hr>
<h2 id='enrich_circo_bar'>Combine and Visualize Data with Circular Bar Chart</h2><span id='topic+enrich_circo_bar'></span>

<h3>Description</h3>

<p>This function combines multiple data frames, arranges them, and visualizes the combined data
in a Circular Bar Chart using the 'ggplot2' and 'ggalluvial' packages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrich_circo_bar(data_list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enrich_circo_bar_+3A_data_list">data_list</code></td>
<td>
<p>A list of data frames to be combined.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggplot' object representing the Circular Bar Chart.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create sample data frames for each enrichment category

# 1. Biological Process (BP)
filtered_data_BP &lt;- data.frame(
  Description = c(
    "immune response",
    "cell proliferation",
    "signal transduction",
    "apoptotic process",
    "metabolic process"
  ),
  Count = c(120, 85, 150, 60, 95),
  color = c(
    "#1f77b4",  # blue
    "#ff7f0e",  # orange
    "#2ca02c",  # green
    "#d62728",  # red
    "#9467bd"   # purple
  ),
  stringsAsFactors = FALSE
)

# 2. Cellular Component (CC)
filtered_data_CC &lt;- data.frame(
  Description = c(
    "nucleus",
    "cytoplasm",
    "membrane",
    "mitochondrion",
    "extracellular space"
  ),
  Count = c(90, 110, 75, 65, 80),
  color = c(
    "#1f77b4",
    "#ff7f0e",
    "#2ca02c",
    "#d62728",
    "#9467bd"
  ),
  stringsAsFactors = FALSE
)

# 3. Molecular Function (MF)
filtered_data_MF &lt;- data.frame(
  Description = c(
    "protein binding",
    "DNA binding",
    "enzyme activity",
    "transporter activity",
    "receptor activity"
  ),
  Count = c(140, 130, 100, 70, 90),
  color = c(
    "#1f77b4",
    "#ff7f0e",
    "#2ca02c",
    "#d62728",
    "#9467bd"
  ),
  stringsAsFactors = FALSE
)

# 4. Disease Ontology (DO)
filtered_data_DO &lt;- data.frame(
  Description = c(
    "cancer",
    "cardiovascular disease",
    "neurological disorder",
    "metabolic disease",
    "infectious disease"
  ),
  Count = c(200, 150, 120, 90, 160),
  color = c(
    "#1f77b4",
    "#ff7f0e",
    "#2ca02c",
    "#d62728",
    "#9467bd"
  ),
  stringsAsFactors = FALSE
)

# 5. Reactome Pathways
filtered_data_Reactome &lt;- data.frame(
  Description = c(
    "Cell Cycle",
    "Apoptosis",
    "DNA Repair",
    "Signal Transduction",
    "Metabolism of Proteins"
  ),
  Count = c(110, 95, 80, 130, 85),
  color = c(
    "#1f77b4",
    "#ff7f0e",
    "#2ca02c",
    "#d62728",
    "#9467bd"
  ),
  stringsAsFactors = FALSE
)

# 6. KEGG Pathways
filtered_data_kegg &lt;- data.frame(
  Description = c(
    "PI3K-Akt signaling pathway",
    "MAPK signaling pathway",
    "NF-kappa B signaling pathway",
    "JAK-STAT signaling pathway",
    "Toll-like receptor signaling pathway"
  ),
  Count = c(175, 160, 145, 130, 155),
  color = c(
    "#1f77b4",
    "#ff7f0e",
    "#2ca02c",
    "#d62728",
    "#9467bd"
  ),
  stringsAsFactors = FALSE
)

# Combine all filtered data frames into a list
data_list &lt;- list(
  BP = filtered_data_BP,
  CC = filtered_data_CC,
  MF = filtered_data_MF,
  DO = filtered_data_DO,
  Reactome = filtered_data_Reactome,
  KEGG = filtered_data_kegg
)

# Create the Circular Bar Chart
combined_and_visualized_data &lt;- enrich_circo_bar(data_list)

</code></pre>

<hr>
<h2 id='enrich_polar_bubble'>Enrichment Polar Bubble Plot</h2><span id='topic+enrich_polar_bubble'></span>

<h3>Description</h3>

<p>This function creates a polar bubble plot using 'ggplot2'. It is designed to visually represent data with methods and positional metrics integrated, highlighting specific IDs if necessary.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrich_polar_bubble(final_combined_df_with_id_and_position, pal, highlight_ids)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enrich_polar_bubble_+3A_final_combined_df_with_id_and_position">final_combined_df_with_id_and_position</code></td>
<td>
<p>A data frame containing 'id', 'Count', 'method', 'Description', 'point_position', 'test_color'.</p>
</td></tr>
<tr><td><code id="enrich_polar_bubble_+3A_pal">pal</code></td>
<td>
<p>A named vector of colors corresponding to the 'method' values.</p>
</td></tr>
<tr><td><code id="enrich_polar_bubble_+3A_highlight_ids">highlight_ids</code></td>
<td>
<p>A vector of IDs to highlight.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggplot' object representing the enriched polar bubble plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  final_df &lt;- data.frame(id = 1:10, Count = c(10, 20, 30, 40, 50, 60, 70, 80, 90, 100),
                         method = rep("Method1", 10),
                         Description = LETTERS[1:10],
                         point_position = seq(10, 100, 10),
                         test_color = sample(c("red", "blue"), 10, replace = TRUE))
  pal &lt;- c("Method1" = "blue")
  highlight_ids &lt;- c(1, 5, 9)
  enrich_polar_bubble(final_df, pal, highlight_ids)
</code></pre>

<hr>
<h2 id='enrichment_circlize'>Draw Chord Diagram with Legends</h2><span id='topic+enrichment_circlize'></span>

<h3>Description</h3>

<p>This function creates a chord diagram from a specified dataframe and draws two sets of legends for it.
It adjusts the track height of the chord diagram to optimize space and uses specified colors for the grid.
Legends are drawn at specified positions with configurable text alignments and font sizes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrichment_circlize(
  all_combined_df,
  original_colors,
  labels,
  colors,
  labels2,
  colors2,
  font_size = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enrichment_circlize_+3A_all_combined_df">all_combined_df</code></td>
<td>
<p>A dataframe containing the matrix for the chord diagram.</p>
</td></tr>
<tr><td><code id="enrichment_circlize_+3A_original_colors">original_colors</code></td>
<td>
<p>A vector of colors for the grid columns of the chord diagram.</p>
</td></tr>
<tr><td><code id="enrichment_circlize_+3A_labels">labels</code></td>
<td>
<p>A vector of labels for the first legend.</p>
</td></tr>
<tr><td><code id="enrichment_circlize_+3A_colors">colors</code></td>
<td>
<p>A vector of colors corresponding to the first legend's labels.</p>
</td></tr>
<tr><td><code id="enrichment_circlize_+3A_labels2">labels2</code></td>
<td>
<p>A vector of labels for the second legend.</p>
</td></tr>
<tr><td><code id="enrichment_circlize_+3A_colors2">colors2</code></td>
<td>
<p>A vector of colors corresponding to the second legend's labels.</p>
</td></tr>
<tr><td><code id="enrichment_circlize_+3A_font_size">font_size</code></td>
<td>
<p>The font size used for legend texts, defaults to 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisible, primarily used for its side effects of drawing on a graphics device.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample Chord Diagram Matrix
all_combined_df &lt;- data.frame(
  A = c(10, 20, 30),
  B = c(15, 25, 35),
  C = c(5, 10, 15)
)
rownames(all_combined_df) &lt;- c("A", "B", "C")

# Colors for the grid of the chord diagram (corresponding to columns of the matrix)
original_colors &lt;- c("red", "green", "blue")

# Name the colors according to the sectors (A, B, C)
names(original_colors) &lt;- colnames(all_combined_df)

# Labels and Colors for the First Legend
labels &lt;- c("Label 1", "Label 2", "Label 3")
colors &lt;- c("yellow", "purple", "cyan")

# Labels and Colors for the Second Legend
labels2 &lt;- c("Label A", "Label B", "Label C")
colors2 &lt;- c("orange", "pink", "brown")

# Font size for the legend texts (optional, default is 10)
font_size &lt;- 10

# Call the enrichment_circlize function with the sample data
# This is just an example; the plot will be rendered in an appropriate graphics context
# such as RStudio's plot pane or an external plotting window.
plot1 &lt;- enrichment_circlize(all_combined_df,
                             original_colors,
                             labels,
                             colors,
                             labels2,
                             colors2,
                             font_size
                             )

</code></pre>

<hr>
<h2 id='enrichment_spiral_plots'>Create Spiral Plots with Legends Using 'spiralize' and 'ComplexHeatmap'</h2><span id='topic+enrichment_spiral_plots'></span>

<h3>Description</h3>

<p>This function initializes a spiral plot, adds tracks for pathways and samples,
and generates legends based on the sample and pathway information in the provided data frame.
It uses 'spiralize' for the spiral plot and 'ComplexHeatmap' for handling legends.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrichment_spiral_plots(results)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enrichment_spiral_plots_+3A_results">results</code></td>
<td>
<p>A data frame containing 'Pathway', 'Sample', 'Value', 'PathwayColor', and 'SampleColor' columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects. This function generates spiral plots and adds legends based on sample and pathway information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example: Creating enrichment spiral plots with legends

# Define the results data frame
results &lt;- data.frame(
  Pathway = c("Pathway1", "Pathway1", "Pathway2", "Pathway2", "Pathway3"),
  Sample = c("Sample1", "Sample1", "Sample2", "Sample2", "Sample3"),
  Value = c(20, 30, 15, 35, 25),
  PathwayColor = c("red", "red", "blue", "blue", "orange"),
  SampleColor = c("green", "green", "purple", "purple", "cyan"),
  stringsAsFactors = FALSE
)

# Create the enrichment spiral plots with legends
enrichment_spiral_plots(results)

</code></pre>

<hr>
<h2 id='extract_descriptions_counts'>Extract and Count Descriptions with Specified Color</h2><span id='topic+extract_descriptions_counts'></span>

<h3>Description</h3>

<p>This function filters a data frame for specified descriptions, selects the 'Description' and 'Count' columns,
and adds a new column with a specified color.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_descriptions_counts(df, descriptions, color)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_descriptions_counts_+3A_df">df</code></td>
<td>
<p>A data frame containing at least 'Description' and 'Count' columns.</p>
</td></tr>
<tr><td><code id="extract_descriptions_counts_+3A_descriptions">descriptions</code></td>
<td>
<p>A vector of descriptions to filter in the data frame.</p>
</td></tr>
<tr><td><code id="extract_descriptions_counts_+3A_color">color</code></td>
<td>
<p>A character string specifying the color to be added as a new column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame filtered by descriptions, containing 'Description', 'Count', and a new 'color' column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate Sample Input Data for extract_descriptions_counts Function

# Create a sample data frame with 'Description' and 'Count' columns
data &lt;- data.frame(
  Description = c(
    "immunoglobulin production",
    "B cell mediated immunity",
    "T cell activation",
    "antigen processing and presentation",
    "cytokine signaling",
    "natural killer cell activity",
    "phagocytosis",
    "complement activation",
    "antibody-dependent cellular cytotoxicity",
    "regulatory T cell function"
  ),
  Count = c(
    150,  # immunoglobulin production
    200,  # B cell mediated immunity
    175,  # T cell activation
    125,  # antigen processing and presentation
    190,  # cytokine signaling
    160,  # natural killer cell activity
    140,  # phagocytosis
    180,  # complement activation
    130,  # antibody-dependent cellular cytotoxicity
    170   # regulatory T cell function
  ),
  stringsAsFactors = FALSE  # Ensure that strings are not converted to factors
)



descriptions_to_filter &lt;- c("immunoglobulin production", "B cell mediated immunity")
specified_color &lt;- "red"  # You can specify any color you desire
filtered_data_with_color &lt;- extract_descriptions_counts(
  data, descriptions_to_filter,
  specified_color)
print(filtered_data_with_color)

</code></pre>

<hr>
<h2 id='extract_ntop_pathways'>Extract and Store Top Pathways for Each Sample</h2><span id='topic+extract_ntop_pathways'></span>

<h3>Description</h3>

<p>This function processes a dataframe containing SSGSEA KEGG results. It allows specifying the number
of top pathways to extract for each sample based on their scores, and stores these in a new dataframe
with sample names and pathway scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_ntop_pathways(ssgsea_kegg, nTop = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_ntop_pathways_+3A_ssgsea_kegg">ssgsea_kegg</code></td>
<td>
<p>Dataframe containing SSGSEA KEGG results with samples as columns and pathways as rows.</p>
</td></tr>
<tr><td><code id="extract_ntop_pathways_+3A_ntop">nTop</code></td>
<td>
<p>Integer, number of top pathways to select for each sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with columns 'Pathway', 'Sample', and 'Value' representing the top pathways for each sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example: Generating input data for the extract_ntop_pathways function

# Define example pathways
pathways &lt;- c("Pathway_A", "Pathway_B", "Pathway_C", "Pathway_D", "Pathway_E",
              "Pathway_F", "Pathway_G", "Pathway_H", "Pathway_I", "Pathway_J")

# Define example samples
samples &lt;- c("Sample_1", "Sample_2", "Sample_3")

# Generate random SSGSEA KEGG scores between 0 and 1
set.seed(123)  # For reproducibility
ssgsea_scores &lt;- matrix(runif(length(pathways) * length(samples), min = 0, max = 1),
                        nrow = length(pathways), ncol = length(samples),
                        dimnames = list(pathways, samples))

# Convert to a data frame
ssgsea_kegg &lt;- as.data.frame(ssgsea_scores)

# Extract the top 3 pathways for each sample
top_pathways &lt;- extract_ntop_pathways(ssgsea_kegg, nTop = 3)

</code></pre>

<hr>
<h2 id='extract_positive_pathways'>Extract Positive Pathways from SSGSEA Results and Select Random Samples</h2><span id='topic+extract_positive_pathways'></span>

<h3>Description</h3>

<p>This function processes the results of SSGSEA, specifically focusing on KEGG pathways.
It extracts pathways with positive values from each sample and randomly selects a subset of them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_positive_pathways(ssgsea_kegg, max_paths_per_sample = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_positive_pathways_+3A_ssgsea_kegg">ssgsea_kegg</code></td>
<td>
<p>A matrix or data frame with pathways as rows and samples as columns.</p>
</td></tr>
<tr><td><code id="extract_positive_pathways_+3A_max_paths_per_sample">max_paths_per_sample</code></td>
<td>
<p>Integer, maximum number of pathways to select per sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with selected pathways, samples, and their corresponding values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example: Generating input data for the extract_positive_pathways function

# Define example pathways
pathways &lt;- c("Pathway_1", "Pathway_2", "Pathway_3", "Pathway_4", "Pathway_5",
              "Pathway_6", "Pathway_7", "Pathway_8", "Pathway_9", "Pathway_10")

# Define example samples
samples &lt;- c("Sample_A", "Sample_B", "Sample_C")

# Generate random SSGSEA KEGG scores including both positive and negative values
set.seed(456)  # For reproducibility
ssgsea_scores &lt;- matrix(rnorm(length(pathways) * length(samples), mean = 0, sd = 1),
                        nrow = length(pathways), ncol = length(samples),
                        dimnames = list(pathways, samples))

# Convert to a data frame
ssgsea_kegg &lt;- as.data.frame(ssgsea_scores)

# Use the extract_positive_pathways function to extract up to 3 positive pathways per sample
selected_positive_pathways &lt;- extract_positive_pathways(ssgsea_kegg, max_paths_per_sample = 3)

</code></pre>

<hr>
<h2 id='facet_density_foldchange'>Create faceted high-density region plots with optional points and density contours</h2><span id='topic+facet_density_foldchange'></span>

<h3>Description</h3>

<p>This function creates faceted high-density region plots using ggdensity for
adding optional density rug and density contours, and scatter points. It also adds a regression line
and Pearson correlation label. The plot is faceted by a grouping variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>facet_density_foldchange(
  data,
  x_var,
  y_var,
  group_var,
  facet_var,
  palette,
  show_points = FALSE,
  show_density = TRUE,
  point_size = 2.5,
  point_alpha = 0.1,
  line_size = 1.6,
  cor_method = "pearson",
  cor_label_pos = c("left", 0.97),
  cor_vjust = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="facet_density_foldchange_+3A_data">data</code></td>
<td>
<p>Data frame containing variables for plotting.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_x_var">x_var</code></td>
<td>
<p>Name of the x-axis variable as a string.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_y_var">y_var</code></td>
<td>
<p>Name of the y-axis variable as a string.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_group_var">group_var</code></td>
<td>
<p>Name of the grouping variable for color mapping as a string.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_facet_var">facet_var</code></td>
<td>
<p>Name of the faceting variable.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_palette">palette</code></td>
<td>
<p>Color palette for the plot as a character vector.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_show_points">show_points</code></td>
<td>
<p>Logical, if TRUE adds scatter points to the plot.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_show_density">show_density</code></td>
<td>
<p>Logical, if TRUE adds filled density contours to the plot.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_point_size">point_size</code></td>
<td>
<p>Size of the points, relevant if show_points is TRUE.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_point_alpha">point_alpha</code></td>
<td>
<p>Transparency level of the points, relevant if show_points is TRUE.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_line_size">line_size</code></td>
<td>
<p>Size of the regression line.</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_cor_method">cor_method</code></td>
<td>
<p>Method to calculate correlation (&quot;pearson&quot; or &quot;spearman&quot;).</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_cor_label_pos">cor_label_pos</code></td>
<td>
<p>Vector of length 2 indicating the position of the correlation label (x and y).</p>
</td></tr>
<tr><td><code id="facet_density_foldchange_+3A_cor_vjust">cor_vjust</code></td>
<td>
<p>Vertical justification for correlation label, default is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggplot' object representing the high-density region plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>combined_df_file &lt;- system.file("extdata", "combined_df.rds", package = "TransProR")
combined_df &lt;- readRDS(combined_df_file)
pal2 = c("#2787e0","#1a9ae0","#1dabbf","#00897b","#43a047","#7cb342")
all_facet_density_foldchange_name1 &lt;- facet_density_foldchange(
  data = combined_df,
  x_var = "log2FoldChange_1",
  y_var = "log2FoldChange_2",
  group_var = "name",
  facet_var = "name",
  palette = pal2,
  show_points = TRUE,
  show_density = FALSE,
  point_size = 2,
  point_alpha = 0.1,
  line_size = 1.6,
  cor_method = "pearson",
  cor_label_pos = c("left", "top"),
  cor_vjust = 1
)
</code></pre>

<hr>
<h2 id='filter_diff_genes'>Filter Differentially Expressed Genes</h2><span id='topic+filter_diff_genes'></span>

<h3>Description</h3>

<p>This function filters a data frame to identify genes with significant differential expression
based on specified thresholds for p-values and log fold change. It allows for flexible
input of column names for p-values and log fold change.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_diff_genes(
  data,
  p_val_col = "adj.P.Val",
  log_fc_col = "logFC",
  p_val_threshold = 0.05,
  log_fc_threshold = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_diff_genes_+3A_data">data</code></td>
<td>
<p>A data frame containing gene expression data.</p>
</td></tr>
<tr><td><code id="filter_diff_genes_+3A_p_val_col">p_val_col</code></td>
<td>
<p>Character string indicating the column name for p-values. Default is &quot;adj.P.Val&quot;.</p>
</td></tr>
<tr><td><code id="filter_diff_genes_+3A_log_fc_col">log_fc_col</code></td>
<td>
<p>Character string indicating the column name for log fold change. Default is &quot;logFC&quot;.</p>
</td></tr>
<tr><td><code id="filter_diff_genes_+3A_p_val_threshold">p_val_threshold</code></td>
<td>
<p>Numeric threshold for filtering p-values. Default is 0.05.</p>
</td></tr>
<tr><td><code id="filter_diff_genes_+3A_log_fc_threshold">log_fc_threshold</code></td>
<td>
<p>Numeric threshold for filtering log fold change. Default is 1.0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with genes filtered by the specified criteria.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a sample data frame with p-values and log fold changes
sample_data &lt;- data.frame(
  adj.P.Val = c(0.03, 0.06, 0.02, 0.07),
  logFC = c(1.5, 0.8, -1.2, 1.1),
  gene = c("Gene1", "Gene2", "Gene3", "Gene4")
)

# Use the filter_diff_genes function to filter significant genes
filtered_genes &lt;- filter_diff_genes(sample_data)
print(filtered_genes)
</code></pre>

<hr>
<h2 id='four_degs_venn'>Function to Create a Venn Diagram of DEGs</h2><span id='topic+four_degs_venn'></span>

<h3>Description</h3>

<p>This function creates a Venn Diagram using the ggVennDiagram package.
It allows customization of various aesthetic elements of the diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>four_degs_venn(degs_list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="four_degs_venn_+3A_degs_list">degs_list</code></td>
<td>
<p>A list of DEG sets for Venn Diagram creation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the Venn Diagram.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("all_degs_venn", package = "TransProR")
four_degs_venn &lt;- four_degs_venn(all_degs_venn)
</code></pre>

<hr>
<h2 id='gather_graph_edge'>Gather graph edge from data frame
Please note that this function is from the 'ggraph' package and has not been altered in functionality,
but it has been optimized and iterated.
It is not original content of 'TransProR'.
However, since 'ggraph' caused frequent GitHub Action errors during the creation of 'TransProR',
the author directly referenced the involved functions in 'TransProR'.
This is not the author's original creation. All users please be aware!</h2><span id='topic+gather_graph_edge'></span>

<h3>Description</h3>

<p>Gather graph edge from data frame
Please note that this function is from the 'ggraph' package and has not been altered in functionality,
but it has been optimized and iterated.
It is not original content of 'TransProR'.
However, since 'ggraph' caused frequent GitHub Action errors during the creation of 'TransProR',
the author directly referenced the involved functions in 'TransProR'.
This is not the author's original creation. All users please be aware!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gather_graph_edge(df, index = NULL, root = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gather_graph_edge_+3A_df">df</code></td>
<td>
<p>A data frame</p>
</td></tr>
<tr><td><code id="gather_graph_edge_+3A_index">index</code></td>
<td>
<p>A vector of column names to group by</p>
</td></tr>
<tr><td><code id="gather_graph_edge_+3A_root">root</code></td>
<td>
<p>Root name for the root node connections, optional</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble of graph edges
</p>

<hr>
<h2 id='gather_graph_node'>Gather graph nodes from a data frame
Please note that this function is from the 'ggraph' package and has not been altered in functionality,
but it has been optimized and iterated.
It is not original content of 'TransProR'.
However, since 'ggraph' caused frequent GitHub Action errors during the creation of 'TransProR',
the author directly referenced the involved functions in 'TransProR'.
This is not the author's original creation. All users please be aware!</h2><span id='topic+gather_graph_node'></span>

<h3>Description</h3>

<p>Gather graph nodes from a data frame
Please note that this function is from the 'ggraph' package and has not been altered in functionality,
but it has been optimized and iterated.
It is not original content of 'TransProR'.
However, since 'ggraph' caused frequent GitHub Action errors during the creation of 'TransProR',
the author directly referenced the involved functions in 'TransProR'.
This is not the author's original creation. All users please be aware!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gather_graph_node(
  df,
  index = NULL,
  value = utils::tail(colnames(df), 1),
  root = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gather_graph_node_+3A_df">df</code></td>
<td>
<p>A data frame</p>
</td></tr>
<tr><td><code id="gather_graph_node_+3A_index">index</code></td>
<td>
<p>A vector of column names to group by</p>
</td></tr>
<tr><td><code id="gather_graph_node_+3A_value">value</code></td>
<td>
<p>Column name used for summarizing node size, defaults to the last column</p>
</td></tr>
<tr><td><code id="gather_graph_node_+3A_root">root</code></td>
<td>
<p>Root name for the root node connections, optional</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble of graph nodes
</p>

<hr>
<h2 id='gene_color'>Merge Genes with Color Information Based on Up/Down Regulation</h2><span id='topic+gene_color'></span>

<h3>Description</h3>

<p>This function merges selected genes with differential expression data and adds a color column based on up/down regulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_color(selected_genes, DEG_deseq2, up_color, down_color)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gene_color_+3A_selected_genes">selected_genes</code></td>
<td>
<p>A data frame containing selected genes with a column named &quot;Symble&quot;.</p>
</td></tr>
<tr><td><code id="gene_color_+3A_deg_deseq2">DEG_deseq2</code></td>
<td>
<p>A data frame containing differential expression data with a column named &quot;Symble&quot; and a column named &quot;change&quot; indicating up/down regulation.</p>
</td></tr>
<tr><td><code id="gene_color_+3A_up_color">up_color</code></td>
<td>
<p>The color to assign to genes with up-regulated expression.</p>
</td></tr>
<tr><td><code id="gene_color_+3A_down_color">down_color</code></td>
<td>
<p>The color to assign to genes with down-regulated expression.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing merged genes with an additional color column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>selected_genes_deseq2_file &lt;- system.file("extdata",
                                          "selected_genes_deseq2.rds",
                                          package = "TransProR")
selected_genes_deseq2 &lt;- readRDS(selected_genes_deseq2_file)
Diff_deseq2_file &lt;- system.file("extdata", "Diff_deseq2.rds", package = "TransProR")
Diff_deseq2 &lt;- readRDS(Diff_deseq2_file)

result_deseq2 &lt;- gene_color(selected_genes_deseq2, Diff_deseq2, "#0000EE", "#fc4746")
</code></pre>

<hr>
<h2 id='gene_highlights'>Add gene highlights to a ggtree object</h2><span id='topic+gene_highlights'></span>

<h3>Description</h3>

<p>This function enhances a 'ggtree' plot by adding highlights for specific genes. It adds both a semi-transparent fan-shaped
highlight and a point at the node corresponding to each gene. Colors for each gene can be customized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_highlights(ggtree_obj, genes_to_highlight, hilight_extend = 18)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gene_highlights_+3A_ggtree_obj">ggtree_obj</code></td>
<td>
<p>A ggtree object to which the highlights will be added.</p>
</td></tr>
<tr><td><code id="gene_highlights_+3A_genes_to_highlight">genes_to_highlight</code></td>
<td>
<p>A data frame containing genes and their corresponding colors.</p>
</td></tr>
<tr><td><code id="gene_highlights_+3A_hilight_extend">hilight_extend</code></td>
<td>
<p>Integer, the extension of the highlight fan in degrees.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggtree object with added gene highlights.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("gtree", package = "TransProR")

# Define genes and their colors
genes_df &lt;- data.frame(Symble = c("t5", "t9"),
                       color = c("#FF0000", "#0000FF"))

# Add highlights
gtree &lt;- gene_highlights(gtree, genes_to_highlight = genes_df)

</code></pre>

<hr>
<h2 id='gene_map_pathway'>Create Pathway-Gene Mapping Data Frame</h2><span id='topic+gene_map_pathway'></span>

<h3>Description</h3>

<p>This function takes multiple data frames and pathway IDs, merging them into a new data frame.
Each data frame represents a type of analysis (e.g., BP, KEGG, MF, etc.).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_map_pathway(
  BP_dataframe,
  BP_ids,
  KEGG_dataframe,
  KEGG_ids,
  MF_dataframe,
  MF_ids,
  REACTOME_dataframe,
  REACTOME_ids,
  CC_dataframe,
  CC_ids,
  DO_dataframe,
  DO_ids
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gene_map_pathway_+3A_bp_dataframe">BP_dataframe</code></td>
<td>
<p>Data frame for Biological Process analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_bp_ids">BP_ids</code></td>
<td>
<p>Selected pathway IDs for Biological Process analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_kegg_dataframe">KEGG_dataframe</code></td>
<td>
<p>Data frame for KEGG analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_kegg_ids">KEGG_ids</code></td>
<td>
<p>Selected pathway IDs for KEGG analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_mf_dataframe">MF_dataframe</code></td>
<td>
<p>Data frame for Molecular Function analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_mf_ids">MF_ids</code></td>
<td>
<p>Selected pathway IDs for Molecular Function analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_reactome_dataframe">REACTOME_dataframe</code></td>
<td>
<p>Data frame for REACTOME analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_reactome_ids">REACTOME_ids</code></td>
<td>
<p>Selected pathway IDs for REACTOME analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_cc_dataframe">CC_dataframe</code></td>
<td>
<p>Data frame for Cellular Component analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_cc_ids">CC_ids</code></td>
<td>
<p>Selected pathway IDs for Cellular Component analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_do_dataframe">DO_dataframe</code></td>
<td>
<p>Data frame for Disease Ontology analysis</p>
</td></tr>
<tr><td><code id="gene_map_pathway_+3A_do_ids">DO_ids</code></td>
<td>
<p>Selected pathway IDs for Disease Ontology analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new data frame that includes pathways, gene, type, and value columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulating data for different analysis types

# Simulate Biological Process (BP) data frame
BP_df &lt;- data.frame(
  ID = c("GO:0002376", "GO:0019724"),
  geneID = c("GENE1/GENE2", "GENE3/GENE4"),
  Description = c("Immune response", "Glycosylation process")
)

# Simulate KEGG data frame
KEGG_df &lt;- data.frame(
  ID = c("12345", "67890"),
  geneID = c("GENE5/GENE6", "GENE7/GENE8"),
  Description = c("Pathway 1", "Pathway 2")
)

# Simulate Molecular Function (MF) data frame
MF_df &lt;- data.frame(
  ID = c("ABC123", "DEF456"),
  geneID = c("GENE9/GENE10", "GENE11/GENE12"),
  Description = c("Molecular function A", "Molecular function B")
)

# Simulate REACTOME data frame
REACTOME_df &lt;- data.frame(
  ID = c("R-HSA-12345", "R-HSA-67890"),
  geneID = c("GENE13/GENE14", "GENE15/GENE16"),
  Description = c("Pathway in Reactome 1", "Pathway in Reactome 2")
)

# Simulate Cellular Component (CC) data frame
CC_df &lt;- data.frame(
  ID = c("GO:0005575", "GO:0005634"),
  geneID = c("GENE17/GENE18", "GENE19/GENE20"),
  Description = c("Cellular component A", "Cellular component B")
)

# Simulate Disease Ontology (DO) data frame
DO_df &lt;- data.frame(
  ID = c("DOID:123", "DOID:456"),
  geneID = c("GENE21/GENE22", "GENE23/GENE24"),
  Description = c("Disease A", "Disease B")
)

# Example pathway IDs for each analysis
BP_ids &lt;- c("GO:0002376", "GO:0019724")
KEGG_ids &lt;- c("12345", "67890")
MF_ids &lt;- c("ABC123", "DEF456")
REACTOME_ids &lt;- c("R-HSA-12345", "R-HSA-67890")
CC_ids &lt;- c("GO:0005575", "GO:0005634")
DO_ids &lt;- c("DOID:123", "DOID:456")

# Generate the pathway-gene map using the gene_map_pathway function
pathway_gene_map &lt;- gene_map_pathway(
  BP_dataframe = BP_df, BP_ids = BP_ids,
  KEGG_dataframe = KEGG_df, KEGG_ids = KEGG_ids,
  MF_dataframe = MF_df, MF_ids = MF_ids,
  REACTOME_dataframe = REACTOME_df, REACTOME_ids = REACTOME_ids,
  CC_dataframe = CC_df, CC_ids = CC_ids,
  DO_dataframe = DO_df, DO_ids = DO_ids
)

# Display the resulting pathway-gene mapping data frame
print(pathway_gene_map)

</code></pre>

<hr>
<h2 id='get_gtex_exp'>Get GTEx Expression Data for Specific Organ</h2><span id='topic+get_gtex_exp'></span>

<h3>Description</h3>

<p>This function retrieves gene expression data from the GTEx project that is specific to a certain organ.
It performs various checks and processing steps to ensure that the data is consistent and relevant to the
specified organ. The filtered and cleaned data is saved as an RDS file for further analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gtex_exp(
  organ_specific,
  file_path,
  probe_map_path,
  pheno_path,
  output_path
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gtex_exp_+3A_organ_specific">organ_specific</code></td>
<td>
<p>A character string specifying the organ to filter the gene expression data by.</p>
</td></tr>
<tr><td><code id="get_gtex_exp_+3A_file_path">file_path</code></td>
<td>
<p>A character string specifying the path to the GTEx gene expression data file.</p>
</td></tr>
<tr><td><code id="get_gtex_exp_+3A_probe_map_path">probe_map_path</code></td>
<td>
<p>A character string specifying the path to the gtex_probeMap_gencode data file.</p>
</td></tr>
<tr><td><code id="get_gtex_exp_+3A_pheno_path">pheno_path</code></td>
<td>
<p>A character string specifying the path to the GTEx phenotype data file.</p>
</td></tr>
<tr><td><code id="get_gtex_exp_+3A_output_path">output_path</code></td>
<td>
<p>A character string specifying the path where the output RDS file will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function begins by checking if the gene expression and phenotype data files exist at
the specified paths. It then loads these data files and processes them by setting appropriate row names,
modifying column names for clarity, and filtering samples based on the specified organ. The function ensures
that only samples present in both datasets are retained for consistency. It also removes any duplicate gene
entries to prevent redundancy. Finally, the processed data is saved as an RDS file.
</p>


<h3>Value</h3>

<p>A data frame containing gene expression data for the specified organ.
Rows represent genes, and columns represent samples. Note that this function also saves the
organ-specific GTEx data as an RDS file at the specified output path.
</p>


<h3>Note</h3>

<p>The function will stop and throw an error if the input files do not exist, or if no samples are found
for the specified organ.
</p>
<p>CRITICAL: The 'output_path' parameter must end with '.rds' to be properly recognized by the function. It is also highly recommended
that the path includes specific identifiers related to the target samples. Please structure the 'output_path' following this pattern: './your_directory/your_sample_type.gtex.rds'.
</p>

<hr>
<h2 id='get_tcga_exp'>TCGA Expression Data Processing</h2><span id='topic+get_tcga_exp'></span>

<h3>Description</h3>

<p>This function processes expression data and phenotype information, separates tumor and normal samples,
and saves the results into different files. It's specifically designed for data obtained from TCGA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tcga_exp(
  counts_file_path,
  gene_probes_file_path,
  phenotype_file_path,
  output_file_path
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_tcga_exp_+3A_counts_file_path">counts_file_path</code></td>
<td>
<p>File path to the counts data (usually in the form of a large matrix with gene expression data).</p>
</td></tr>
<tr><td><code id="get_tcga_exp_+3A_gene_probes_file_path">gene_probes_file_path</code></td>
<td>
<p>File path containing the gene probes data.</p>
</td></tr>
<tr><td><code id="get_tcga_exp_+3A_phenotype_file_path">phenotype_file_path</code></td>
<td>
<p>File path to the phenotype data, which includes various sample attributes.</p>
</td></tr>
<tr><td><code id="get_tcga_exp_+3A_output_file_path">output_file_path</code></td>
<td>
<p>Path where the output files, distinguished between tumor and normal, will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing matrices for tumor and normal expression data.
</p>


<h3>Note</h3>

<p>IMPORTANT: This function assumes that the input files follow a specific format and structure, typically found in TCGA data releases.
Users should verify their data's compatibility. Additionally, the function does not perform error checking on the data's content,
which users should handle through proper preprocessing.
</p>
<p>CRITICAL: The 'output_file_path' parameter must end with '.rds' to be properly recognized by the function. It is also highly recommended
that the path includes specific identifiers related to the target samples, as the function will create further subdivisions in the specified
path for tumor or normal tissues. Please structure the 'output_file_path' following this pattern: './your_directory/your_sample_type.exp.rds'.
</p>


<h3>Author(s)</h3>

<p>Dongyue Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>counts_file &lt;- system.file("extdata", "TCGA-SKCM.htseq_counts_test.tsv", package = "TransProR")
gene_probes_file &lt;- system.file("extdata",
                                "TCGA_gencode.v22.annotation.gene.probeMap_test",
                                package = "TransProR")
phenotype_file &lt;- system.file("extdata", "TCGA-SKCM.GDC_phenotype_test.tsv", package = "TransProR")
ouput_file &lt;- file.path(tempdir(), "SKCM_Skin_TCGA_exp_test.rds")

SKCM_exp &lt;- get_tcga_exp(
  counts_file_path = counts_file,
  gene_probes_file_path = gene_probes_file,
  phenotype_file_path = phenotype_file,
  output_file_path = ouput_file
)
head(SKCM_exp[["tumor_tcga_data"]])[1:5, 1:5]
head(SKCM_exp[["normal_tcga_data"]], n = 10) # Because there is only one column.
</code></pre>

<hr>
<h2 id='gtree'>Phylogenetic Tree Object</h2><span id='topic+gtree'></span>

<h3>Description</h3>

<p>A dataset containing a phylogenetic tree object created using the 'ggtree' package.
This tree represents the evolutionary relationships among a set of species or genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gtree)
</code></pre>


<h3>Format</h3>

<p>A 'ggtree' object.
</p>


<h3>Source</h3>

<p>The phylogenetic tree was constructed based on sequence alignment data obtained from [Data Source, e.g., NCBI database, specific study, etc.].
</p>

<hr>
<h2 id='highlight_by_node'>Highlight Nodes in a Phylogenetic Tree with Custom Fill Colors and Transparency</h2><span id='topic+highlight_by_node'></span>

<h3>Description</h3>

<p>This function adds highlights to specific nodes in a phylogenetic tree represented by a 'ggtree' object.
Users can specify the nodes to highlight along with custom fill colors, transparency, and extension options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>highlight_by_node(
  ggtree_object,
  nodes,
  fill_colors,
  alpha_values,
  extend_values
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="highlight_by_node_+3A_ggtree_object">ggtree_object</code></td>
<td>
<p>A 'ggtree' object representing the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="highlight_by_node_+3A_nodes">nodes</code></td>
<td>
<p>A character vector specifying the nodes to highlight.</p>
</td></tr>
<tr><td><code id="highlight_by_node_+3A_fill_colors">fill_colors</code></td>
<td>
<p>A character vector specifying the fill colors for the highlighted nodes.</p>
</td></tr>
<tr><td><code id="highlight_by_node_+3A_alpha_values">alpha_values</code></td>
<td>
<p>A numeric vector specifying the transparency values for the highlighted nodes (between 0 and 1).</p>
</td></tr>
<tr><td><code id="highlight_by_node_+3A_extend_values">extend_values</code></td>
<td>
<p>A logical vector specifying whether to extend the highlight to the whole clade below each node.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified 'ggtree' object with the specified nodes highlighted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_file &lt;- system.file("extdata", "tree_plot.rds", package = "TransProR")
p2_plot &lt;- readRDS(plot_file)

# Please replace the following vectors with your specific values
nodes &lt;- c(117, 129, 125, 127, 119,
           123, 139, 166, 124, 131, 217) # x-values of the nodes you want to highlight
fill_colors &lt;- c("#CD6600", "#CD6600", "#CD6600",
                 "#CD6600", "#009933", "#009933",
                 "#009933", "#009933", "#9B30FF",
                 "#9B30FF", "#9B30FF") # Fill colors
alpha_values &lt;- c(0.3, 0.3, 0.3, 0.3, 0.2, 0.3,
                  0.3, 0.3, 0.3, 0.3, 0.3) # Transparency values
extend_values &lt;- c(25, 24, 24, 25, 25, 25,
                   24, 24, 25, 24, 24) # Values for the 'extend' parameter

p2 &lt;- highlight_by_node(
  p2_plot,
  nodes,
  fill_colors,
  alpha_values,
  extend_values
)
</code></pre>

<hr>
<h2 id='highlight_genes'>Add Highlights for Genes on a Phylogenetic Tree</h2><span id='topic+highlight_genes'></span>

<h3>Description</h3>

<p>This function adds highlights for specified genes on a phylogenetic tree object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>highlight_genes(ggtree_obj, genes_to_highlight, hilight_extend = 18)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="highlight_genes_+3A_ggtree_obj">ggtree_obj</code></td>
<td>
<p>A ggtree object representing the phylogenetic tree.</p>
</td></tr>
<tr><td><code id="highlight_genes_+3A_genes_to_highlight">genes_to_highlight</code></td>
<td>
<p>A data frame containing gene names and corresponding colors to highlight.</p>
</td></tr>
<tr><td><code id="highlight_genes_+3A_hilight_extend">hilight_extend</code></td>
<td>
<p>Numeric value indicating the extension length for highlights.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggtree' object with added highlights for specified genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_file &lt;- system.file("extdata", "tree_plot.rds", package = "TransProR")
p2_plot &lt;- readRDS(plot_file)

selected_genes_deseq2_file &lt;- system.file("extdata",
                                          "selected_genes_deseq2.rds",
                                          package = "TransProR")
selected_genes_deseq2 &lt;- readRDS(selected_genes_deseq2_file)

Diff_deseq2_file &lt;- system.file("extdata", "Diff_deseq2.rds", package = "TransProR")
Diff_deseq2 &lt;- readRDS(Diff_deseq2_file)

result_deseq2 &lt;- gene_color(selected_genes_deseq2, Diff_deseq2, "#0000EE", "#fc4746")

add_gene_highlights_p3 &lt;- highlight_genes(p2_plot, result_deseq2, hilight_extend = 26)
</code></pre>

<hr>
<h2 id='limma_analyze'>Differential Gene Expression Analysis using limma and voom</h2><span id='topic+limma_analyze'></span>

<h3>Description</h3>

<p>This function performs differential gene expression analysis using the 'limma' package with voom normalization.
It reads tumor and normal expression data, merges them, filters low-expressed genes,
normalizes the data, performs limma analysis, and outputs the results along with information
on gene expression changes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>limma_analyze(
  tumor_file,
  normal_file,
  output_file,
  logFC_threshold = 2.5,
  p_value_threshold = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="limma_analyze_+3A_tumor_file">tumor_file</code></td>
<td>
<p>Path to the tumor data file (RDS format).</p>
</td></tr>
<tr><td><code id="limma_analyze_+3A_normal_file">normal_file</code></td>
<td>
<p>Path to the normal data file (RDS format).</p>
</td></tr>
<tr><td><code id="limma_analyze_+3A_output_file">output_file</code></td>
<td>
<p>Path to save the output DEG data (RDS format).</p>
</td></tr>
<tr><td><code id="limma_analyze_+3A_logfc_threshold">logFC_threshold</code></td>
<td>
<p>Threshold for log fold change for marking up/down-regulated genes.</p>
</td></tr>
<tr><td><code id="limma_analyze_+3A_p_value_threshold">p_value_threshold</code></td>
<td>
<p>Threshold for p-value for filtering significant genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of differential expression results.
</p>


<h3>References</h3>

<p>limma:Linear Models for Microarray and RNA-Seq Data Users Guide.
For more information, visit the page:
https://www.bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Define file paths for tumor and normal data from the data folder
tumor_file &lt;- system.file("extdata",
                          "removebatch_SKCM_Skin_TCGA_exp_tumor_test.rds",
                          package = "TransProR")
normal_file &lt;- system.file("extdata",
                           "removebatch_SKCM_Skin_Normal_TCGA_GTEX_count_test.rds",
                           package = "TransProR")
output_file &lt;- file.path(tempdir(), "DEG_limma_voom.rds")

DEG_limma_voom &lt;- limma_analyze(
  tumor_file = tumor_file,
  normal_file = normal_file,
  output_file = output_file,
  logFC_threshold = 2.5,
  p_value_threshold = 0.01
)

# View the top 5 rows of the result
head(DEG_limma_voom, 5)
</code></pre>

<hr>
<h2 id='log_transform'>Log transformation decision and application on data</h2><span id='topic+log_transform'></span>

<h3>Description</h3>

<p>This function evaluates the need for a log transformation based on a set of criteria
and applies a log2 transformation if necessary.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log_transform(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="log_transform_+3A_data">data</code></td>
<td>
<p>A numeric matrix or data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The original data or the data transformed with log2.
</p>


<h3>Author(s)</h3>

<p>Dongyue Yu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_path &lt;- system.file("extdata",
                         "all_count_exp_test.csv",
                         package = "TransProR")
your_data &lt;- read.csv(file_path,
                      row.names = 1)  # Assuming first column is row names (e.g., gene names)

TransformedData &lt;- log_transform(data = your_data)

</code></pre>

<hr>
<h2 id='merge_density_foldchange'>Create high-density region plot with optional points, density rugs, and contours</h2><span id='topic+merge_density_foldchange'></span>

<h3>Description</h3>

<p>This function creates a high-density region plot using hdr methods to
add density rug and filled contours. It also adds a regression line
and Pearson correlation label. Points can be added to the plot optionally.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_density_foldchange(
  data,
  x_var,
  y_var,
  group_var,
  palette = c("#3949ab", "#1e88e5", "#039be5", "#00897b", "#43a047", "#7cb342"),
  show_points = FALSE,
  point_size = 2.5,
  point_alpha = 0.2,
  x_lim = c(0, 20),
  y_lim = c(0, 20),
  cor_method = "pearson",
  line_size = 1.6,
  cor_label_pos = c("left", 0.97)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_density_foldchange_+3A_data">data</code></td>
<td>
<p>Data frame containing variables for plotting.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_x_var">x_var</code></td>
<td>
<p>Name of the x-axis variable as a string.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_y_var">y_var</code></td>
<td>
<p>Name of the y-axis variable as a string.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_group_var">group_var</code></td>
<td>
<p>Name of the grouping variable for color mapping as a string.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_palette">palette</code></td>
<td>
<p>Color palette for the plot as a character vector.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_show_points">show_points</code></td>
<td>
<p>Logical, if TRUE adds points to the plot.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_point_size">point_size</code></td>
<td>
<p>Size of the points, relevant if show_points is TRUE.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_point_alpha">point_alpha</code></td>
<td>
<p>Transparency level of the points, relevant if show_points is TRUE.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_x_lim">x_lim</code></td>
<td>
<p>Numeric vector of length 2, giving the x-axis limits.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_y_lim">y_lim</code></td>
<td>
<p>Numeric vector of length 2, giving the y-axis limits.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_cor_method">cor_method</code></td>
<td>
<p>Method to calculate correlation (&quot;pearson&quot; or &quot;spearman&quot;).</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_line_size">line_size</code></td>
<td>
<p>Size of the smoothing line.</p>
</td></tr>
<tr><td><code id="merge_density_foldchange_+3A_cor_label_pos">cor_label_pos</code></td>
<td>
<p>Vector of length 2 indicating the position of the correlation label (x and y).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the high-density region plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>combined_df_file &lt;- system.file("extdata", "combined_df.rds", package = "TransProR")
combined_df &lt;- readRDS(combined_df_file)
pal1 = c("#3949ab","#1e88e5","#039be5","#00897b","#43a047","#7cb342")

all_density_foldchange_name1 &lt;- merge_density_foldchange(
  data = combined_df,
  x_var = "log2FoldChange_1",
  y_var = "log2FoldChange_2",
  group_var = "name",
  palette = pal1,
  show_points = FALSE,
  point_size = 2.5,
  point_alpha = 0.1,
  x_lim = c(0, 20),
  y_lim = c(0, 20),
  cor_method = "pearson",
  line_size = 1.6,
  cor_label_pos = c("left", "top")
)

</code></pre>

<hr>
<h2 id='merge_gtex_tcga'>Merge gene expression data from GTEx and TCGA datasets</h2><span id='topic+merge_gtex_tcga'></span>

<h3>Description</h3>

<p>This function merges gene expression data obtained from the GTEx (Genotype-Tissue Expression) and TCGA (The Cancer Genome Atlas) datasets.
It is assumed that both datasets are in '.rds' format and have genes as row names. The merged dataset is saved as an RDS file at the specified output path.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_gtex_tcga(
  gtex_data_path,
  tcga_exp_path,
  output_path = "./merged_gtex_tcga_data.rds"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_gtex_tcga_+3A_gtex_data_path">gtex_data_path</code></td>
<td>
<p>A string that specifies the file path to the GTEx data saved in RDS format.</p>
</td></tr>
<tr><td><code id="merge_gtex_tcga_+3A_tcga_exp_path">tcga_exp_path</code></td>
<td>
<p>A string that specifies the file path to the TCGA expression data saved in RDS format.
This should be a data.frame with rows as genes and columns as samples.</p>
</td></tr>
<tr><td><code id="merge_gtex_tcga_+3A_output_path">output_path</code></td>
<td>
<p>A string that specifies the path where the merged dataset should be saved.
The file is saved in '.rds' format. The default path is &quot;./merged_gtex_tcga_data.rds&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is assumed that both datasets are in '.rds' format and have genes as row names.
</p>


<h3>Value</h3>

<p>A data frame where rows represent genes and columns represent samples.
The data frame contains expression values from both GTEx and TCGA datasets.
It saves the merged dataset to the path specified by 'output_path'.
</p>


<h3>Note</h3>

<p>CRITICAL: The 'output_path' parameter must end with '.rds' to be properly recognized by the function. It is also highly recommended
that the path includes specific identifiers related to the target samples. Please structure the 'output_path' following this pattern: './your_directory/merged.your_sample_type.gtex.tcga.data.rds'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tumor_file &lt;- system.file("extdata",
                          "removebatch_SKCM_Skin_TCGA_exp_tumor_test.rds",
                          package = "TransProR")
Normal_file &lt;- system.file("extdata",
                           "removebatch_SKCM_Skin_Normal_TCGA_GTEX_count_test.rds",
                           package = "TransProR")
ouput_file &lt;- file.path(tempdir(), "all_data.rds")

all_data &lt;- merge_gtex_tcga(gtex_data_path = tumor_file,
                            tcga_exp_path = Normal_file,
                            output_path = ouput_file)

</code></pre>

<hr>
<h2 id='merge_id_position'>Merge Data Frames by Common Row Names with Additional Columns</h2><span id='topic+merge_id_position'></span>

<h3>Description</h3>

<p>This function merges a list of data frames based on common row names. It adds an 'id' column to track the row order and a 'point_position' column calculated based on the maximum 'Count' value across all data frames. It filters data frames to include only common rows, sorts rows by the length of the 'Description' in descending order, and then merges them by rows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_id_position(df_list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_id_position_+3A_df_list">df_list</code></td>
<td>
<p>A list of data frames, each with a 'Description' and 'Count' column and set row names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single data frame merged from the list, with additional 'id' and 'point_position' columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df1 &lt;- data.frame(Description = c("DataA", "DataB"), Count = c(10, 20), row.names = c("R1", "R2"))
df2 &lt;- data.frame(Description = c("DataC", "DataD"), Count = c(30, 40), row.names = c("R1", "R3"))
df_list &lt;- list(df1, df2)
combined_df_test &lt;- merge_id_position(df_list)

</code></pre>

<hr>
<h2 id='merge_method_color'>Merge Data Frames with Specific Method and Color Columns</h2><span id='topic+merge_method_color'></span>

<h3>Description</h3>

<p>This function takes a list of data frames, a method name, and a list of colors.
It adds a 'method' column and a 'test_color' column to each data frame, then merges all data frames by rows.
It ensures that the color list length matches the list of data frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_method_color(df_list, method_name, color_list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_method_color_+3A_df_list">df_list</code></td>
<td>
<p>A list of data frames, each containing at least 'Description' and 'Count' columns.</p>
</td></tr>
<tr><td><code id="merge_method_color_+3A_method_name">method_name</code></td>
<td>
<p>A string representing the method name to be added to each data frame.</p>
</td></tr>
<tr><td><code id="merge_method_color_+3A_color_list">color_list</code></td>
<td>
<p>A list of colors corresponding to each data frame for the 'test_color' column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single data frame merged from the list, with each originally provided data frame now having a 'method' and a 'test_color' column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df1 &lt;- data.frame(Description = c("A", "B"), Count = c(10, 20))
df2 &lt;- data.frame(Description = c("C", "D"), Count = c(30, 40))
df_list &lt;- list(df1, df2)
method_name &lt;- "Method1"
color_list &lt;- c("Red", "Blue")
combined_df_test &lt;- merge_method_color(df_list, method_name, color_list)

</code></pre>

<hr>
<h2 id='new_ggraph'>Generate a graphical representation of pathway gene maps</h2><span id='topic+new_ggraph'></span>

<h3>Description</h3>

<p>This function merges multiple gene-pathway related dataframes, processes them
for graph creation, and visualizes the relationships in a dendrogram layout using
the provided node and edge gathering functions from the 'ggraph' package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>new_ggraph(
  BP_dataframe,
  BP_ids,
  KEGG_dataframe,
  KEGG_ids,
  MF_dataframe,
  MF_ids,
  REACTOME_dataframe,
  REACTOME_ids,
  CC_dataframe,
  CC_ids,
  DO_dataframe,
  DO_ids
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="new_ggraph_+3A_bp_dataframe">BP_dataframe</code></td>
<td>
<p>Dataframe for Biological Process.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_bp_ids">BP_ids</code></td>
<td>
<p>IDs for Biological Process.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_kegg_dataframe">KEGG_dataframe</code></td>
<td>
<p>Dataframe for KEGG pathways.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_kegg_ids">KEGG_ids</code></td>
<td>
<p>IDs for KEGG pathways.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_mf_dataframe">MF_dataframe</code></td>
<td>
<p>Dataframe for Molecular Function.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_mf_ids">MF_ids</code></td>
<td>
<p>IDs for Molecular Function.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_reactome_dataframe">REACTOME_dataframe</code></td>
<td>
<p>Dataframe for REACTOME pathways.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_reactome_ids">REACTOME_ids</code></td>
<td>
<p>IDs for REACTOME pathways.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_cc_dataframe">CC_dataframe</code></td>
<td>
<p>Dataframe for Cellular Component.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_cc_ids">CC_ids</code></td>
<td>
<p>IDs for Cellular Component.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_do_dataframe">DO_dataframe</code></td>
<td>
<p>Dataframe for Disease Ontology.</p>
</td></tr>
<tr><td><code id="new_ggraph_+3A_do_ids">DO_ids</code></td>
<td>
<p>IDs for Disease Ontology.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggraph' object representing the pathway gene map visualization.
</p>

<hr>
<h2 id='pathway_count'>Count Genes Present in Pathways Above a Threshold</h2><span id='topic+pathway_count'></span>

<h3>Description</h3>

<p>This function filters pathways that meet a count threshold and then counts the presence of specified genes in those pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway_count(GO, count_threshold, enrich_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pathway_count_+3A_go">GO</code></td>
<td>
<p>A character vector of gene symbols.</p>
</td></tr>
<tr><td><code id="pathway_count_+3A_count_threshold">count_threshold</code></td>
<td>
<p>An integer specifying the count threshold for selecting pathways.</p>
</td></tr>
<tr><td><code id="pathway_count_+3A_enrich_data">enrich_data</code></td>
<td>
<p>A data frame containing pathway enrichment analysis results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns &quot;Symble&quot; (gene symbol), &quot;Description&quot; (pathway description), and &quot;Exists&quot; (1 if gene is present, 0 otherwise).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulated gene list
GO &lt;- c("Gene1", "Gene2", "Gene3", "Gene4", "Gene5")
# Simulated enrichment analysis data
enrich_data &lt;- data.frame(
  ID = c("GO:0001", "GO:0002", "GO:0003"),
  Description = c("Pathway A", "Pathway B", "Pathway C"),
  Count = c(10, 4, 6),
  geneID = c("Gene1/Gene2/Gene3", "Gene4/Gene5", "Gene2/Gene6/Gene7")
)

# Example usage
count_threshold &lt;- 5
result_df &lt;- pathway_count(GO, count_threshold, enrich_data)

</code></pre>

<hr>
<h2 id='pathway_description'>Describe Genes Present in Selected Pathways</h2><span id='topic+pathway_description'></span>

<h3>Description</h3>

<p>This function identifies genes present in selected pathways based on gene enrichment analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway_description(GO, selected_pathways_names, enrich_data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pathway_description_+3A_go">GO</code></td>
<td>
<p>A character vector of gene symbols.</p>
</td></tr>
<tr><td><code id="pathway_description_+3A_selected_pathways_names">selected_pathways_names</code></td>
<td>
<p>A character vector specifying the names of selected pathways.</p>
</td></tr>
<tr><td><code id="pathway_description_+3A_enrich_data">enrich_data</code></td>
<td>
<p>A data frame containing pathway enrichment analysis results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns &quot;Symble&quot; (gene symbol), &quot;Description&quot; (pathway description), and &quot;Exists&quot; (1 if gene is present, 0 otherwise).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>GO &lt;- c("Gene1", "Gene2", "Gene3", "Gene4", "Gene5")
# Simulated enrichment analysis data
enrich_data &lt;- data.frame(
  ID = c("Pathway1", "Pathway2", "Pathway3", "Pathway4"),
  Description = c("Apoptosis", "Cell Cycle", "Signal Transduction", "Metabolism"),
  geneID = c("Gene1/Gene3", "Gene2/Gene4", "Gene1/Gene2/Gene3", "Gene5"),
  Count = c(2, 2, 3, 1),
  stringsAsFactors = FALSE
)

# Example usage
result &lt;- pathway_description(GO,
                              selected_pathways_names="Apoptosis",
                              enrich_data)

</code></pre>

<hr>
<h2 id='prep_deseq2'>Prepare DESeq2 data for plotting</h2><span id='topic+prep_deseq2'></span>

<h3>Description</h3>

<p>This function reads a DESeq2 DEG data frame from an RDS file, filters it,
adjusts the log2FoldChange to absolute values, adds a pseudo-count to pvalues,
and transforms pvalues for plotting. The final data frame is returned and
optionally saved to a new RDS file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_deseq2(input_path, output_name = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_deseq2_+3A_input_path">input_path</code></td>
<td>
<p>Path to the RDS file containing the DESeq2 DEG data frame.</p>
</td></tr>
<tr><td><code id="prep_deseq2_+3A_output_name">output_name</code></td>
<td>
<p>Name for the processed data frame, also used as the RDS file name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with processed DESeq2 DEG data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>deseq2_file &lt;- system.file("extdata",
                           "DEG_deseq2_test.rds",
                           package = "TransProR")
deseq2 &lt;- prep_deseq2(deseq2_file)

</code></pre>

<hr>
<h2 id='prep_edgeR'>Prepare edgeR DEG data for plotting</h2><span id='topic+prep_edgeR'></span>

<h3>Description</h3>

<p>This function reads an edgeR DEG data frame from an RDS file, filters it using
<code><a href="#topic+deg_filter">deg_filter</a></code> function, adjusts the logFC to absolute values, adds a pseudo-count to PValue,
and transforms PValue for plotting. The final data frame is returned and
optionally saved to a new RDS file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_edgeR(input_path, output_name = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_edgeR_+3A_input_path">input_path</code></td>
<td>
<p>Path to the RDS file containing the edgeR DEG data frame.</p>
</td></tr>
<tr><td><code id="prep_edgeR_+3A_output_name">output_name</code></td>
<td>
<p>Name for the processed data frame, also used as the RDS file name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with processed edgeR DEG data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>edgeR_file &lt;- system.file("extdata",
                          "DEG_edgeR_test.rds",
                          package = "TransProR")
edgeR &lt;- prep_edgeR(edgeR_file)

</code></pre>

<hr>
<h2 id='prep_limma'>Prepare limma-voom DEG data for plotting</h2><span id='topic+prep_limma'></span>

<h3>Description</h3>

<p>This function reads a limma-voom DEG data frame from an RDS file, filters it using
<code><a href="#topic+deg_filter">deg_filter</a></code> function, adjusts the logFC to absolute values, adds a pseudo-count to P.Value,
and transforms P.Value for plotting. The final data frame is returned and
optionally saved to a new RDS file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_limma(input_path, output_name = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_limma_+3A_input_path">input_path</code></td>
<td>
<p>Path to the RDS file containing the limma-voom DEG data frame.</p>
</td></tr>
<tr><td><code id="prep_limma_+3A_output_name">output_name</code></td>
<td>
<p>Name for the processed data frame, also used as the RDS file name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with processed limma-voom DEG data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>limma_file &lt;- system.file("extdata",
                          "DEG_limma_voom_test.rds",
                          package = "TransProR")
limma &lt;- prep_limma(limma_file)

</code></pre>

<hr>
<h2 id='prep_wilcoxon'>Prepare Wilcoxon DEG data for plotting</h2><span id='topic+prep_wilcoxon'></span>

<h3>Description</h3>

<p>This function reads a Wilcoxon DEG data frame from an RDS file, filters it using
<code><a href="#topic+deg_filter">deg_filter</a></code> function, adjusts the log2foldChange to absolute values, adds a pseudo-count to pValues,
and transforms pValues for plotting. The final data frame is returned and
optionally saved to a new RDS file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_wilcoxon(input_path, output_name = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_wilcoxon_+3A_input_path">input_path</code></td>
<td>
<p>Path to the RDS file containing the Wilcoxon DEG data frame.</p>
</td></tr>
<tr><td><code id="prep_wilcoxon_+3A_output_name">output_name</code></td>
<td>
<p>Optional; name for the processed data frame, also used as the RDS file name.
If not provided, the data frame will not be saved to file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with processed Wilcoxon DEG data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>wilcoxon_file &lt;- system.file("extdata",
                             "Wilcoxon_rank_sum_testoutRst_test.rds",
                             package = "TransProR")
Wilcoxon &lt;- prep_wilcoxon(wilcoxon_file)

</code></pre>

<hr>
<h2 id='process_heatdata'>Process Heatmap Data with Various Selection Options</h2><span id='topic+process_heatdata'></span>

<h3>Description</h3>

<p>This function processes heatmap data ('heatdata') based on a given selection option.
It allows customization of column names, selection of specific columns per group,
or averaging columns based on a common prefix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_heatdata(
  heatdata,
  selection = 1,
  custom_names = NULL,
  num_names_per_group = NULL,
  prefix_length = 4
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="process_heatdata_+3A_heatdata">heatdata</code></td>
<td>
<p>A data frame containing the heatmap data.</p>
</td></tr>
<tr><td><code id="process_heatdata_+3A_selection">selection</code></td>
<td>
<p>An integer specifying the processing method:
- 1: Use custom names for columns.
- 2: Select a given number of columns per group based on a prefix.
- 3: Calculate the average of columns per group based on a prefix.</p>
</td></tr>
<tr><td><code id="process_heatdata_+3A_custom_names">custom_names</code></td>
<td>
<p>A character vector of custom names for columns (used when 'selection = 1').
The length of this vector must match the number of columns in 'heatdata'.</p>
</td></tr>
<tr><td><code id="process_heatdata_+3A_num_names_per_group">num_names_per_group</code></td>
<td>
<p>An integer specifying the number of columns to select per group (used when 'selection = 2').</p>
</td></tr>
<tr><td><code id="process_heatdata_+3A_prefix_length">prefix_length</code></td>
<td>
<p>An integer specifying the length of the prefix for grouping columns (used when 'selection = 2' or 'selection = 3').
Default is 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A processed data frame based on the specified selection option.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example heatmap data frame
heatdata &lt;- data.frame(
  groupA_1 = c(1, 2, 3),
  groupA_2 = c(4, 5, 6),
  groupB_1 = c(7, 8, 9),
  groupB_2 = c(10, 11, 12)
)

# Selection 1: Use custom names for columns
custom_names &lt;- c("Sample1", "Sample2", "Sample3", "Sample4")
processed_data1 &lt;- process_heatdata(heatdata, selection = 1, custom_names = custom_names)

# Selection 2: Select a given number of columns per group based on a prefix
processed_data2 &lt;- process_heatdata(heatdata, selection = 2, num_names_per_group = 1)

# Selection 3: Calculate the average of columns per group based on a prefix
processed_data3 &lt;- process_heatdata(heatdata, selection = 3, prefix_length = 6)
</code></pre>

<hr>
<h2 id='seek_gtex_organ'>Load and Process GTEX Phenotype Data to Retrieve Primary Site Counts</h2><span id='topic+seek_gtex_organ'></span>

<h3>Description</h3>

<p>This function reads the GTEX phenotype data from a specified path, renames its columns for better readability,
and then returns a table of primary site counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seek_gtex_organ(path = "./download_data/GTEX_phenotype")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="seek_gtex_organ_+3A_path">path</code></td>
<td>
<p>The path to the GTEX phenotype data file. Default is &quot;./download_data/GTEX_phenotype&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table representing the count of samples per primary site.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the file path to the example data in the package
path &lt;- system.file("extdata", "GTEX_phenotype_test", package = "TransProR")
# Call the `seek_gtex_organ` function with the path and print the result
SeekGtexOrgan &lt;- seek_gtex_organ(path = path)

</code></pre>

<hr>
<h2 id='selectPathways'>Randomly Select Pathways with Limited Word Count</h2><span id='topic+selectPathways'></span>

<h3>Description</h3>

<p>This function randomly selects a specified number of pathways from a given list, ensuring that each selected pathway name does not exceed a specified number of words. It filters out pathways with names longer than the specified word limit before making the selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selectPathways(pathways, max_words = 10, num_select = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="selectPathways_+3A_pathways">pathways</code></td>
<td>
<p>Character vector of pathways.</p>
</td></tr>
<tr><td><code id="selectPathways_+3A_max_words">max_words</code></td>
<td>
<p>Integer, maximum number of words allowed in the pathway name.</p>
</td></tr>
<tr><td><code id="selectPathways_+3A_num_select">num_select</code></td>
<td>
<p>Integer, number of pathways to randomly select.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of selected pathways.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pathway_list &lt;- c("pathway_one response to stimulus",
                  "pathway_two cell growth and death",
                  "pathway_three regulation of cellular process",
                  "pathway_four metabolic process")
selected_pathways &lt;- selectPathways(pathway_list, max_words = 5, num_select = 2)

</code></pre>

<hr>
<h2 id='spiral_newrle'>Render a Spiral Plot Using Run-Length Encoding</h2><span id='topic+spiral_newrle'></span>

<h3>Description</h3>

<p>This function creates a spiral plot for visualizing sequential data in a compact and visually appealing way.
It uses run-length encoding to represent the lengths and colors of sequences in the spiral.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spiral_newrle(x, samples, values, colors, labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spiral_newrle_+3A_x">x</code></td>
<td>
<p>A vector representing categories or segments.</p>
</td></tr>
<tr><td><code id="spiral_newrle_+3A_samples">samples</code></td>
<td>
<p>A vector indicating the sample each segment belongs to.</p>
</td></tr>
<tr><td><code id="spiral_newrle_+3A_values">values</code></td>
<td>
<p>Numeric vector indicating the lengths of each segment.</p>
</td></tr>
<tr><td><code id="spiral_newrle_+3A_colors">colors</code></td>
<td>
<p>Character vector specifying the colors for each segment.</p>
</td></tr>
<tr><td><code id="spiral_newrle_+3A_labels">labels</code></td>
<td>
<p>Logical, whether to add labels to each segment.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects. This function generates a spiral plot and optionally adds labels.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example: Creating a spiral plot using the spiral_newrle function

# Define example data
x &lt;- c("A", "A", "B", "C")
samples &lt;- c("Sample1", "Sample1", "Sample2", "Sample2")
values &lt;- c(20, 30, 15, 35)
colors &lt;- c("red", "blue", "green", "purple")
labels &lt;- TRUE

# Initialize the spiral plot, setting the x-axis range and scaling
spiralize::spiral_initialize(xlim = c(0, sum(values)), scale_by = "curve_length",
                 vp_param = list(x = grid::unit(0, "npc"), just = "left"))

# Create a track for the spiral plot
spiralize::spiral_track(height = 0.5)

# Add segments to the spiral plot using run-length encoding
spiral_newrle(x, samples, values, colors, labels)

</code></pre>

<hr>
<h2 id='Wilcoxon_analyze'>Differential Gene Expression Analysis Using Wilcoxon Rank-Sum Test</h2><span id='topic+Wilcoxon_analyze'></span>

<h3>Description</h3>

<p>This function performs differential gene expression analysis using Wilcoxon rank-sum tests.
It reads tumor and normal expression data, performs TMM normalization using 'edgeR', and uses Wilcoxon rank-sum tests to identify differentially expressed genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Wilcoxon_analyze(
  tumor_file,
  normal_file,
  output_file,
  logFC_threshold = 2.5,
  fdr_threshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Wilcoxon_analyze_+3A_tumor_file">tumor_file</code></td>
<td>
<p>Path to the tumor data file (RDS format).</p>
</td></tr>
<tr><td><code id="Wilcoxon_analyze_+3A_normal_file">normal_file</code></td>
<td>
<p>Path to the normal data file (RDS format).</p>
</td></tr>
<tr><td><code id="Wilcoxon_analyze_+3A_output_file">output_file</code></td>
<td>
<p>Path to save the output DEG data (RDS format).</p>
</td></tr>
<tr><td><code id="Wilcoxon_analyze_+3A_logfc_threshold">logFC_threshold</code></td>
<td>
<p>Threshold for log fold change for marking up/down-regulated genes.</p>
</td></tr>
<tr><td><code id="Wilcoxon_analyze_+3A_fdr_threshold">fdr_threshold</code></td>
<td>
<p>Threshold for FDR for filtering significant genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of differential expression results.
</p>


<h3>References</h3>

<p>Li, Y., Ge, X., Peng, F., Li, W., &amp; Li, J. J. (2022). Exaggerated False Positives by Popular
Differential Expression Methods When Analyzing Human Population Samples. Genome Biology, 23(1), 79.
DOI: https://doi.org/10.1186/s13059-022-02648-4.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Define file paths for tumor and normal data from the data folder
tumor_file &lt;- system.file("extdata",
                          "removebatch_SKCM_Skin_TCGA_exp_tumor_test.rds",
                          package = "TransProR")
normal_file &lt;- system.file("extdata",
                           "removebatch_SKCM_Skin_Normal_TCGA_GTEX_count_test.rds",
                           package = "TransProR")
output_file &lt;- file.path(tempdir(), "Wilcoxon_rank_sum_testoutRst.rds")

# Run the Wilcoxon rank sum test
outRst &lt;- Wilcoxon_analyze(
  tumor_file = tumor_file,
  normal_file = normal_file,
  output_file = output_file,
  logFC_threshold = 2.5,
  fdr_threshold = 0.01
)

# View the top 5 rows of the result
head(outRst, 5)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
