<!DOCTYPE html><html><head><title>Help for package TRD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TRD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TRD-package'>
<p>Transmission Ratio Distortion</p></a></li>
<li><a href='#find.maf'>
<p>Find minor allele frequency (MAF) in case-trios, control-trios, or the whole sample.</p></a></li>
<li><a href='#find.t'>
<p>Find transmission ratio of minor alleles in mother, father or both parents.</p></a></li>
<li><a href='#gcount'>
<p>Calculate genotype counts (additive model) for selected population.</p></a></li>
<li><a href='#ll'>
<p>Fit Loglinear model with and without adjustment for TRD.</p></a></li>
<li><a href='#prev'>
<p>Calculate theoretical prevalence of disease</p></a></li>
<li><a href='#rtrios'>
<p>Generate simulated popoulation of case- and control-trios</p></a></li>
<li><a href='#strata.cnt'>
<p>Count trios for each of the 15/16 strata with mother-father-child genotypes MFC</p></a></li>
<li><a href='#tdt'>
<p>Compute TDT and adjusted TDT on case- and/or control-trios</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Transmission Ratio Distortion</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-11-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Lam Opal Huang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lam Opal Huang &lt;opal.huang@mail.mcgill.ca&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>Rlab (&ge; 2.15.1)</td>
</tr>
<tr>
<td>Description:</td>
<td>Transmission Ratio Distortion (TRD) is a genetic phenomenon where the two alleles from either parent are not transmitted to the offspring at the expected 1:1 ratio under Mendelian inheritance, leading to spurious signals in genetic association studies. Functions in this package are developed to account for this phenomenon using loglinear model and Transmission Disequilibrium Test (TDT). Some population information can also be calculated.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-11-10 23:24:37 UTC; Opalh</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-11-11 00:41:56</td>
</tr>
</table>
<hr>
<h2 id='TRD-package'>
Transmission Ratio Distortion
</h2><span id='topic+TRD-package'></span><span id='topic+TRD'></span>

<h3>Description</h3>

<p>Transmission Ratio Distortion (TRD) is a genetic phenomenon where two alleles from either parent are not transmitted to the offspring at the expected 1:1 ratio under Mendelian inheritance. Occurrence of TRD in general population can lead to spurious association signals for the study on case populations, including genotype relative risk (GRR)  and imprinting effect. Therefore, it is necessary to adjust for TRD in association studies when TRD is suspected to be present.
</p>
<p>This package has implemented the loglinear model [Weinberg et al. 1998], the adjusted loglinear model for non-sex-of-parent-specific TRD (NST) [Huang et al. 2014] and sex-of-parent-specific TRD (ST) [Huang et al. 2015]; Transmission Disequilibrium Test(TDT)[Spielman et al. 1993], and adjusted TDT [Labbe et al. 2013].
</p>
<p>A simulation function is included in the package, which generates a population of case- and control-trios, with parent-of-origin information for disease allele in heterozygous child (rtrios), based on penetrance, ratio of maternal to paternal-induced disease risk for a heterozygous child, transmission probability of minor allele for mother and father, and minor allele frequency. Population and sample sizes can be specified by user. The data can be used directly to fit loglinear model (ll) or to perform a TDT (tdt) with or without adjustment for TRD.  Count data for the 15/16 mother-father-child category can also be computed (strata.cnt).
</p>
<p>The loglinear model can take input as trios or count data with or without parent-of-origin information. Types of loglinear model which can be fitted include genotype relative risk, gene-dosage, dominant models, with (grri, gdi, domi) or without (grr, gd, dom) imprinting effect. Real dataset of same structure can also be used to fit all types of loglinear model. Other functions are available to calculate population information, such as genotype counts (gcount), minor allele frequency (find.maf), transmission probability of minor allele (find.t), and theoretical prevalence of disease (prev). Note that trios with missing data are discarded when fitting the loglinear models.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> TRD</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-11-10</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> 3.2.2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>
<p>Maintainer: Lam Opal Huang &lt;opal.huang@mail.mcgill.ca&gt;
</p>


<h3>References</h3>

<p>Huang LO, Infante-Rivard C, Labbe A, (2015). Modeling sex-of-parent-specific Transmission Ratio Distortion and imprinting effect in loglinear model using case-trios.(Working manuscript)
</p>
<p>Huang LO, Infante-Rivard C, Labbe A, (2015). Adjustment for Transmission Ratio Distortion in the analysis of case-parent trios using a loglinear model.(Working manuscript)
</p>
<p>Labbe A, Huang LO, Infante-Rivard C (2013). Transmission Ratio Distortion: A Neglected Phenomenon with Many Consequences in Genetic Analysis and Population Genetics:Naumova AK and Greenwood CMT (eds.), Epigenetics and Complex Traits: 265-85, Springer.
</p>
<p>Huang LO, Labbe A, Infante-Rivard C (2012). Transmission ratio distortion: review of concept and implications for genetic association studies. Human Genetics 132: 245-63.
</p>
<p>Weinberg CR, Wilcox AJ, Lie RT (1998) A log-linear approach to case-parent-triad data: assessing effects of disease genes that act either directly or through maternal effects and that may be subject to parental imprinting. American Journal of Human Genetics 62: 969-78.
</p>
<p>Spielman RS, McGinnis RE, Ewens WJ (1993) Transmission test for linkage disequilibrium: the insulin gene region and insulin-dependent diabetes mellitus (IDDM). American Journal of Human Genetics 52: 506.
</p>

<hr>
<h2 id='find.maf'>
Find minor allele frequency (MAF) in case-trios, control-trios, or the whole sample.
</h2><span id='topic+find.maf'></span>

<h3>Description</h3>

<p>Compute MAF in child, parents and triad of case-trios, control-trios, or the whole sample. Minor allele normally refers to the disease allele, which has lower frequency than the alternative allele(s). Trios with partially missing data are included.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.maf(sample)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.maf_+3A_sample">sample</code></td>
<td>

<p>case- and/or control-trios dataset with first column as disease status (1=case, 0=control), the second, third and forth columns as maternal, paternal and child genotypes. The fifth column, if entered as part of the input data, is the parent-of-origin indicator of disease allele in a heterozygous child (a value of 1 indicates the mother transmitted the disease allele and 0 indicates the father transmitted). The parent-of-origin indicator is only meaningful for a heterozygous child.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>cases</code></td>
<td>
<p>MAF in affected children, parents, or triads of case-trios</p>
</td></tr>
<tr><td><code>ctrls</code></td>
<td>
<p>MAF in unaffected children, parents, or triads of control-trios</p>
</td></tr>
<tr><td><code>sample</code></td>
<td>
<p>MAF in children, parents, or triads of the whole sample</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rtrios">rtrios</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trios=rtrios(100000,500,0.1,0.2,0.3,1,0.5,0.5,0.1)
find.maf(trios$case)
find.maf(trios$ctrl)
find.maf(rbind(trios$case,trios$ctrl))
</code></pre>

<hr>
<h2 id='find.t'>
Find transmission ratio of minor alleles in mother, father or both parents.
</h2><span id='topic+find.t'></span>

<h3>Description</h3>

<p>Calculate the transmission ratio of the minor allele from parents to offspring for case- and control-trios, in mother, father, or both parents. Minor allele normally refers to the disease allele, which has lower frequency than the alternative allele(s). Note that trios with missing data are discarded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.t(sample)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.t_+3A_sample">sample</code></td>
<td>

<p>case- and/or control-trios dataset with first column as disease status (1=case, 0=control), the second, third and forth columns as maternal, paternal and child genotypes. The fifth column, if entered as part of the input data, is the parent-of-origin indicator of disease allele in a heterozygous child (a value of 1 indicates the mother transmitted the disease allele and 0 indicates the father transmitted). The parent-of-origin indicator is only meaningful for a heterozygous child.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>cases</code></td>
<td>
<p>  transmission ratio of minor allele from mother, father and both parents of case-trios to the offspring</p>
</td></tr>
<tr><td><code>ctrls</code></td>
<td>
<p>  transmission ratio of minor allele from mother, father and both parents of control-trios to the offspring</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rtrios">rtrios</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trios=rtrios(100000,500,0.1,0.2,0.3,1,0.5,0.5,0.1)
find.t(trios$case)
find.t(trios$ctrl)
find.t(rbind(trios$case,trios$ctrl))
</code></pre>

<hr>
<h2 id='gcount'>
Calculate genotype counts (additive model) for selected population.
</h2><span id='topic+gcount'></span>

<h3>Description</h3>

<p>Calculate genotype counts (additive model) of case-trios, control-trios and the whole sample, separately for children, mothers and fathers. Additive model means the genotype is the number of copies of minor allele that the individual carries assuming a bi-allelic locus. This genotype will be 0 for homozygous wild-type, 1 for heterozygous, and 2 for homozygous mutant.  Minor allele normally refers to the disease allele, which has lower frequency than the alternative allele(s). Trios with partially missing data are included.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gcount(sample)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gcount_+3A_sample">sample</code></td>
<td>

<p>case- and/or control-trios dataset with first column as disease status (1=case, 0=control), the second, third and forth columns as maternal, paternal and child genotypes. The fifth column, if entered as part of the input data, is the parent-of-origin indicator of disease allele in a heterozygous child (a value of 1 indicates the mother transmitted the disease allele and 0 indicates the father transmitted). The parent-of-origin indicator is only meaningful for a heterozygous child.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>cases</code></td>
<td>
<p>genotype counts of children, mothers and fathers, in case-trios</p>
</td></tr>
<tr><td><code>ctrls</code></td>
<td>
<p>genotype counts of children, mothers and fathers, in control-trios</p>
</td></tr>
<tr><td><code>sample</code></td>
<td>
<p>genotype counts of children, mothers and fathers, in the whole sample</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rtrios">rtrios</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trios=rtrios(100000,500,0.1,0.2,0.3,1,0.5,0.5,0.1)
gcount(trios$case)
gcount(trios$ctrl)
gcount(rbind(trios$case,trios$ctrl))
</code></pre>

<hr>
<h2 id='ll'>
Fit Loglinear model with and without adjustment for TRD.
</h2><span id='topic+ll'></span>

<h3>Description</h3>

<p>Use input data, either sample of case-trios ('sample') or count data for 15/16-MFC-genotype categories ('counts') to fit loglinear model with (Model 2) and without (Model 1) adjustment  for TRD, model type could be genotype relative risk ('grr'), dominant ('dom'), gene-dosage ('gd'), genotype relative risk with imprinting variable ('grri'), dominant with imprinting variable ('domi'), or gene-dosage with imprinting variable ('gdi'). Note that the models fit only complete trios with no missing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ll(data, data.type, tmother, tfather, model.type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ll_+3A_data">data</code></td>
<td>

<p>input data can be
(1) case- and/or control-trios dataset with first column as disease status (1=case, 0=control), the second, third and forth columns as maternal, paternal and child genotypes. The fifth column, if entered as part of the input data, is the parent-of-origin indicator of disease allele in a heterozygous child (a value of 1 indicates the mother transmitted the disease allele and 0 indicates the father transmitted). The parent-of-origin indicator is only meaningful for a heterozygous child.
(2) a count dataset for the 15 or 16-MFC-genotype categories obtainable from the function strata.cnt(..), with the latter having 2 categories (111M &amp; 111F) for triply heterozygous trios.
</p>
</td></tr>
<tr><td><code id="ll_+3A_data.type">data.type</code></td>
<td>

<p>type of data: 'sample' or 'counts'
</p>
</td></tr>
<tr><td><code id="ll_+3A_tmother">tmother</code></td>
<td>

<p>transmission ratio of minor alleles from mother to child
</p>
</td></tr>
<tr><td><code id="ll_+3A_tfather">tfather</code></td>
<td>

<p>transmission ratio of minor alleles from father to child
</p>
</td></tr>
<tr><td><code id="ll_+3A_model.type">model.type</code></td>
<td>

<p>'grr', 'dom', 'gd', 'grri', 'domi', or 'gdi'
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the following warning messages occur, increase the population and sample size.
Warning messages:
glm.fit: fitted rates numerically 0 occurred
</p>


<h3>Value</h3>

<table>
<tr><td><code>Summary table</code></td>
<td>
<p>Summary statistics of fitting the loglinear model</p>
</td></tr></table>


<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>References</h3>

<p>Huang LO, Infante-Rivard C, Labbe A, (2015). Modeling sex-of-parent-specific Transmission Ratio Distortion and imprinting effect in loglinear model using case-trios.(Working manuscript)
</p>
<p>Huang LO, Infante-Rivard C, Labbe A, (2015). Adjustment for Transmission Ratio Distortion in the analysis of case-parent trios using a loglinear model.(Working manuscript)
</p>
<p>Weinberg CR, Wilcox AJ, Lie RT (1998) A log-linear approach to case-parent-triad data: assessing effects of disease genes that act either directly or through maternal effects and that may be subject to parental imprinting. American Journal of Human Genetics 62: 969-78.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rtrios">rtrios</a></code>
<code><a href="#topic+strata.cnt">strata.cnt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trios=rtrios(100000,500,0.1,0.2,0.3,1,0.5,0.5,0.1)
out=ll(trios$case,'sample',0.5,0.5,'grr')

cnt=strata.cnt(trios$case)
out=ll(cnt,'counts',0.5,0.5,'grr')

</code></pre>

<hr>
<h2 id='prev'>
Calculate theoretical prevalence of disease
</h2><span id='topic+prev'></span>

<h3>Description</h3>

<p>Use 3 penetrance factors for genotype 0, 1 and 2 (additive model),  ratio of maternal to paternal-induced disease risk for a heterozygous child, tranmission ratio of minor allele from mother and father to offspring, and minor allele frequency (MAF) to calculate the disease prevalence. Additive model means the genotype is the number of copies of minor allele that the individual carries assuming a bi-allelic locus. This genotype will be 0 for homozygous wild-type, 1 for heterozygous, and 2 for homozygous mutant.  Minor allele normally refers to the disease allele, which has lower frequency than the alternative allele(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prev(f0, f1, f2, g, tmother, tfather, maf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prev_+3A_f0">f0</code></td>
<td>

<p>penetrance for individuals with genotype 0 (additive model)
</p>
</td></tr>
<tr><td><code id="prev_+3A_f1">f1</code></td>
<td>

<p>penetrance for individuals with genotype 1 (additive model)
</p>
</td></tr>
<tr><td><code id="prev_+3A_f2">f2</code></td>
<td>

<p>penetrance for individuals with genotype 2 (additive model)
</p>
</td></tr>
<tr><td><code id="prev_+3A_g">g</code></td>
<td>

<p>Ratio of maternal- over paternal-induced disease risk
</p>
</td></tr>
<tr><td><code id="prev_+3A_tmother">tmother</code></td>
<td>

<p>transmission ratio of minor allele from mother to child
</p>
</td></tr>
<tr><td><code id="prev_+3A_tfather">tfather</code></td>
<td>

<p>transmission ratio of minor allele from father to child
</p>
</td></tr>
<tr><td><code id="prev_+3A_maf">maf</code></td>
<td>

<p>minor allele frequency
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>d</code></td>
<td>
<p>disease prevalence</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>prev(0.1,0.2,0.3,1,0.5,0.5,0.1)
</code></pre>

<hr>
<h2 id='rtrios'>
Generate simulated popoulation of case- and control-trios
</h2><span id='topic+rtrios'></span>

<h3>Description</h3>

<p>Using penetrance factors, ratio of maternal- to paternal-induced disease risk for a heterozygous child, tranmission ratio of minor allele from mother and father to offspring, and minor allele frequency (MAF) to generate a simulated population of case- and/or control-trios dataset with first column as disease status (1=case, 0=control), the second, third and forth columns as maternal, paternal and child genotypes. The fifth column as the parent-of-origin indicator of disease allele in a heterozygous child (a value of 1 indicates the mother transmitted the disease allele and 0 indicates the father transmitted). The parent-of-origin indicator is only meaningful  for a heterozygous child. Minor allele normally refers to the disease allele, which has lower frequency than the alternative allele(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rtrios(n, ssize, f0, f1, f2, g, tmother, tfather, maf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rtrios_+3A_n">n</code></td>
<td>

<p>simulated population size
</p>
</td></tr>
<tr><td><code id="rtrios_+3A_ssize">ssize</code></td>
<td>

<p>sample size
</p>
</td></tr>
<tr><td><code id="rtrios_+3A_f0">f0</code></td>
<td>

<p>penetrance for individuals with genotype 0 (additive model)
</p>
</td></tr>
<tr><td><code id="rtrios_+3A_f1">f1</code></td>
<td>

<p>penetrance for individuals with genotype 1 (additive model)
</p>
</td></tr>
<tr><td><code id="rtrios_+3A_f2">f2</code></td>
<td>

<p>penetrance for individuals with genotype 2 (additive model)
</p>
</td></tr>
<tr><td><code id="rtrios_+3A_g">g</code></td>
<td>

<p>Ratio of maternal- over paternal-induced disease risk
</p>
</td></tr>
<tr><td><code id="rtrios_+3A_tmother">tmother</code></td>
<td>

<p>transmission ratio of minor allele from mother to child
</p>
</td></tr>
<tr><td><code id="rtrios_+3A_tfather">tfather</code></td>
<td>

<p>transmission ratio of minor allele from father to child
</p>
</td></tr>
<tr><td><code id="rtrios_+3A_maf">maf</code></td>
<td>

<p>minor allele frequency
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>case</code></td>
<td>
<p>case-trios data with parent-of-origin indicator</p>
</td></tr>
<tr><td><code>ctrl</code></td>
<td>
<p>control-trios data with parent-of-origin indicator</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>See Also</h3>

<p><code><a href="#topic+find.maf">find.maf</a></code>
<code><a href="#topic+find.t">find.t</a></code>
<code><a href="#topic+gcount">gcount</a></code>
<code><a href="#topic+ll">ll</a></code>
<code><a href="#topic+prev">prev</a></code>
<code><a href="#topic+strata.cnt">strata.cnt</a></code>
<code><a href="#topic+tdt">tdt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trios=rtrios(100000,500,0.1,0.2,0.3,1,0.5,0.5,0.1)
</code></pre>

<hr>
<h2 id='strata.cnt'>
Count trios for each of the 15/16 strata with mother-father-child genotypes MFC
</h2><span id='topic+strata.cnt'></span>

<h3>Description</h3>

<p>Count trios with mother-father-child genotypes MFC (additive model) as
mating type 1: 222;
mating type 2: 212,122,211,121;
mating type 3: 201,021;
mating type 4: 112,(111M,111F) or 111,110;
mating type 5: 101,100,011,010;
mating type 6: 000;
where 111M and 111F are for categories with heterozygous children inherited the disease allele from the mother and father, respectively. Additive model means the genotype is the number of copies of minor allele that the individual carries assuming a bi-allelic locus. This genotype will be 0 for homozygous wild-type, 1 for heterozygous, and 2 for homozygous mutant. Note that trios with missing data are discarded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strata.cnt(sample)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strata.cnt_+3A_sample">sample</code></td>
<td>

<p>case- and/or control-trios dataset with first column as disease status (1=case, 0=control), the second, third and forth columns as maternal, paternal and child genotypes. The fifth column, if entered as part of the input data, is the parent-of-origin indicator of disease allele in a heterozygous child (a value of 1 indicates the mother transmitted the disease allele and 0 indicates the father transmitted). The parent-of-origin indicator is only meaningful for a heterozygous child.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>cnt.MFC</code></td>
<td>
<p>count of stratum with genotypes MFC (additive model)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rtrios">rtrios</a></code>
<code><a href="#topic+ll">ll</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trios=rtrios(100000,500,0.1,0.2,0.3,1,0.5,0.5,0.1)
strata.cnt(trios$case)
strata.cnt(trios$ctrl)
</code></pre>

<hr>
<h2 id='tdt'>
Compute TDT and adjusted TDT on case- and/or control-trios
</h2><span id='topic+tdt'></span>

<h3>Description</h3>

<p>Compute TDT on case- and control-trios, and adjusted TDT on case-trios using transmission ratio of minor allele calculated from control-trios [Labbe et al. 2013].  When control-trios are not supplied in the input data, only TDT on case-trios is computed. Similarly, when case-trios are not supplied, only TDT on control-trios is computed. The adjusted TDT uses both case- and control-trios, and takes into account of TRD. Note that trios with missing data are discarded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tdt(sample)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tdt_+3A_sample">sample</code></td>
<td>

<p>case- and/or control-trios dataset with first column as disease status (1=case, 0=control), the second, third and forth columns as maternal, paternal and child genotypes. The fifth column, if entered as part of the input data, is the parent-of-origin indicator of disease allele in a heterozygous child (a value of 1 indicates the mother transmitted the disease allele and 0 indicates the father transmitted). The parent-of-origin indicator is only meaningful for a heterozygous child.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>case statistics</code></td>
<td>
<p>TDT statistics for case-trios
</p>
</td></tr>
<tr><td><code>control statistics</code></td>
<td>
<p>TDT statistics for control-trios
</p>
</td></tr>
<tr><td><code>Adjusted-case statistics</code></td>
<td>
<p>Adjusted TDT statistics for case-trios using transmission ratio of minor allele in control-trios
</p>
</td></tr>
<tr><td><code>case p-value</code></td>
<td>
<p>TDT p-value for case-trios
</p>
</td></tr>
<tr><td><code>control p-value</code></td>
<td>
<p>TDT p-value for control-trios
</p>
</td></tr>
<tr><td><code>Adjusted-case p-value</code></td>
<td>
<p>Adjusted TDT p-value for case-trios using transmission ratio of minor allele in control-trios
</p>
</td></tr></table>


<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>References</h3>

<p>Labbe A, Huang LO, Infante-Rivard C (2013). Transmission Ratio Distortion: A Neglected Phenomenon with Many Consequences in Genetic Analysis and Population Genetics:Naumova AK and Greenwood CMT (eds.), Epigenetics and Complex Traits: 265-85, Springer.
</p>
<p>Spielman RS, McGinnis RE, Ewens WJ (1993) Transmission test for linkage disequilibrium: the insulin gene region and insulin-dependent diabetes mellitus (IDDM). American Journal of Human Genetics 52: 506.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rtrios">rtrios</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trios=rtrios(100000,500,0.1,0.2,0.3,1,0.5,0.5,0.1)
tdt(trios$case)
tdt(trios$ctrl)
tdt(rbind(trios$case,trios$ctrl))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
