<!DOCTYPE html><html><head><title>Help for package treedater</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {treedater}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#treedater-package'><p>treedater fits a molecular clock to a phylogenetic tree and estimates evolutionary rates and times of common ancestry.</p></a></li>
<li><a href='#boot'><p>Estimate of confidence intervals of molecular clock parameters with user-supplied set of bootstrap trees</p></a></li>
<li><a href='#dater'><p>Estimate a time-scaled tree and fit a molecular clock</p></a></li>
<li><a href='#goodnessOfFitPlot'><p>Produce a goodness of fit plot</p></a></li>
<li><a href='#outlierLineages'><p>Detect lineages with unusually large evolutionary divergence under the fitted treedater model</p></a></li>
<li><a href='#outlierTips'><p>Detect terminal lineages with unusually large evolutionary divergence under the fitted treedater model</p></a></li>
<li><a href='#parboot'><p>Estimate of confidence intervals using parametric bootstrap for molecular clock dating.</p></a></li>
<li><a href='#plot.bootTreedater'><p>Plots lineages through time and confidence intervals estimated by bootstrap.</p></a></li>
<li><a href='#relaxedClockTest'><p>Use parametric bootstrap to test if relaxed clock offers improved fit to data.</p></a></li>
<li><a href='#rootToTipRegressionPlot'><p>Plot evolutionary distance from root to sample times and estimated internal node times and regression lines</p></a></li>
<li><a href='#sampleYearsFromLabels'><p>Compute a vector of numeric sample times from labels in a sequence aligment or phylogeny</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Fast Molecular Clock Dating of Phylogenetic Trees with Rate
Variation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-01-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Erik Volz [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Erik Volz &lt;erik.volz@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for estimating times of common ancestry and molecular clock rates of evolution using a variety of evolutionary models, parametric and nonparametric bootstrap confidence intervals, methods for detecting outlier lineages, root-to-tip regression, and a statistical test for selecting molecular clock models. The methods are described in Volz, E.M. and S.D.W. Frost (2017) &lt;<a href="https://doi.org/10.1093%2Fve%2Fvex025">doi:10.1093/ve/vex025</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>ape (&ge; 5.0),limSolve (&ge; 1.5.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>lubridate,ggplot2,foreach,iterators,mgcv,knitr,rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-01-26 17:37:33 UTC; erikvolz</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-01-26 18:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='treedater-package'>treedater fits a molecular clock to a phylogenetic tree and estimates evolutionary rates and times of common ancestry.</h2><span id='topic+treedater'></span><span id='topic+treedater-package'></span>

<h3>Description</h3>

<p>Additional functions are provided for detecting outlier lineages (possible sequencing or alignment error). A statistical test is available for choosing between strict and relaxed clock models. The calendar time of each sample must be specified (possibly with bounds of uncertainty) and the length of the sequences used to estimate the tree. treedater uses heuristic search to optimise the TMRCAs of a phylogeny and the substitution rate. An uncorrelated relaxed molecular clock accounts for rate variation between lineages of the phylogeny which is parameterised using a Gamma-Poisson mixture model.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Erik Volz <a href="mailto:erik.volz@gmail.com">erik.volz@gmail.com</a>
</p>


<h3>References</h3>

<p>Volz, E. M., and S. D. W. Frost. &quot;Scalable relaxed clock phylogenetic dating.&quot; Virus Evolution 3.2 (2017).
</p>

<hr>
<h2 id='boot'>Estimate of confidence intervals of molecular clock parameters with user-supplied set of bootstrap trees</h2><span id='topic+boot'></span>

<h3>Description</h3>

<p>If the original treedater fit estimated the root position, root
position will also be estimated for each simulation, so the
returned trees may have different root positions. Some replicates
may converge to a strict clock or a relaxed clock, so the
parameter estimates in each replicate may not be directly
comparable. It is possible to compute confidence intervals for the
times of particular nodes or for estimated sample times by
inspecting the output from each fitted treedater object, which is
contained in the $trees attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boot(td, tres, ncpu = 1, searchRoot = 1,
  overrideTempConstraint = TRUE, overrideClock = NULL, quiet = TRUE,
  normalApproxTMRCA = FALSE, parallel_foreach = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boot_+3A_td">td</code></td>
<td>
<p>A fitted treedater object</p>
</td></tr>
<tr><td><code id="boot_+3A_tres">tres</code></td>
<td>
<p>A list or multiPhylo with bootstrap trees with branches in units of substitutions per site</p>
</td></tr>
<tr><td><code id="boot_+3A_ncpu">ncpu</code></td>
<td>
<p>Number of threads to use for parallel computation. Recommended.</p>
</td></tr>
<tr><td><code id="boot_+3A_searchroot">searchRoot</code></td>
<td>
<p>See *dater*</p>
</td></tr>
<tr><td><code id="boot_+3A_overridetempconstraint">overrideTempConstraint</code></td>
<td>
<p>If TRUE (default) will not enforce positive branch lengths in simualtion replicates. Will speed up execution.</p>
</td></tr>
<tr><td><code id="boot_+3A_overrideclock">overrideClock</code></td>
<td>
<p>May be 'strict' or 'additive' or 'relaxed' in which case will force simulations to fit the corresponding model. If ommitted, will inherit the clock model from td</p>
</td></tr>
<tr><td><code id="boot_+3A_quiet">quiet</code></td>
<td>
<p>If TRUE will minimize output printed to screen</p>
</td></tr>
<tr><td><code id="boot_+3A_normalapproxtmrca">normalApproxTMRCA</code></td>
<td>
<p>If TRUE will use estimate standard deviation from simulation replicates and report confidence interval based on normal distribution</p>
</td></tr>
<tr><td><code id="boot_+3A_parallel_foreach">parallel_foreach</code></td>
<td>
<p>If TRUE will use the foreach package for parallelization. May work better on HPC systems.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements 
</p>

<ul>
<li><p> trees: The fitted treedater objects corresponding to each simulation
</p>
</li>
<li><p> meanRates: Vector of estimated rates for each simulation
</p>
</li>
<li><p> meanRate_CI: Confidence interval for substitution rate
</p>
</li>
<li><p> coef_of_variation_CI: Confidence interval for rate variation
</p>
</li>
<li><p> timeOfMRCA_CI: Confidence interval for time of common ancestor
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Erik M Volz &lt;erik.volz@gmail.com&gt;
</p>


<h3>See Also</h3>

<p>dater
parboot
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate a tree 
tre &lt;- ape::rtree(25)
# sample times based on distance from root to tip:
sts &lt;- setNames( ape::node.depth.edgelength( tre )[1:ape::Ntip(tre)], tre$tip.label)
# make a list of trees that simulate outcome of bootstrap using nonparametric phylogeny estimation
# also modify edge length to represent evolutionary distance with rate 1e-3:
bootTrees &lt;- lapply( 1:25, function(i) {
	.tre &lt;- tre
	.tre$edge.length &lt;- tre$edge.length * pmax(rnorm( length(tre$edge.length), 1e-3, 1e-4 ), 0 )
	.tre 
})
tre$edge.length &lt;- tre$edge.length * 1e-3
# run treedater
td &lt;- dater( tre, sts, s= 1000, clock='strict', omega0=.0015  )
# bootstrap: 
( tdboot &lt;- boot( td, bootTrees ) )
# plot lineages through time :
plot( tdboot )


</code></pre>

<hr>
<h2 id='dater'>Estimate a time-scaled tree and fit a molecular clock</h2><span id='topic+dater'></span>

<h3>Description</h3>

<p>Estimate a time-scaled tree and fit a molecular clock
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dater(tre, sts, s = 1000, omega0 = NA, minblen = NA, maxit = 100,
  abstol = 1e-04, searchRoot = 5, quiet = TRUE,
  temporalConstraints = TRUE, clock = c("strict", "uncorrelated",
  "additive"), estimateSampleTimes = NULL,
  estimateSampleTimes_densities = list(), numStartConditions = 1,
  clsSolver = c("limSolve", "mgcv"), meanRateLimits = NULL, ncpu = 1,
  parallel_foreach = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dater_+3A_tre">tre</code></td>
<td>
<p>An ape::phylo which describes the phylogeny with branches in
units of substitutions per site. This may be a rooted or
unrooted tree. If unrooted, the root position will be
estimated by checking multiple candidates chosen by
root-to-tip regression.  If the tree has multifurcations,
these will be resolved and a binary tree will be returned.</p>
</td></tr>
<tr><td><code id="dater_+3A_sts">sts</code></td>
<td>
<p>Vector of sample times for each tip in phylogenetic tree.
Vector must be named with names corresponding to
tre$tip.label.</p>
</td></tr>
<tr><td><code id="dater_+3A_s">s</code></td>
<td>
<p>Sequence length (numeric). This should correspond to sequence length used in phylogenetic analysis and will not necessarily be the same as genome length.</p>
</td></tr>
<tr><td><code id="dater_+3A_omega0">omega0</code></td>
<td>
<p>Vector providing initial guess or guesses of the mean substitution rate (substitutions
per site per unit time). If not provided, will guess using
root to tip regression.</p>
</td></tr>
<tr><td><code id="dater_+3A_minblen">minblen</code></td>
<td>
<p>Minimum branch length in calendar time. By default, this will
be the range of sample times (max - min) divided by sample
size.</p>
</td></tr>
<tr><td><code id="dater_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations</p>
</td></tr>
<tr><td><code id="dater_+3A_abstol">abstol</code></td>
<td>
<p>Difference in log likelihood between successive iterations for convergence.</p>
</td></tr>
<tr><td><code id="dater_+3A_searchroot">searchRoot</code></td>
<td>
<p>Will search for the optimal root position using the top
matches from root-to-tip regression.  If searchRoot=x, dates
will be estimated for x trees, and the estimate with the
highest likelihood will be returned.</p>
</td></tr>
<tr><td><code id="dater_+3A_quiet">quiet</code></td>
<td>
<p>If TRUE, will suppress messages during execution</p>
</td></tr>
<tr><td><code id="dater_+3A_temporalconstraints">temporalConstraints</code></td>
<td>
<p>If TRUE, will enforce the condition that an
ancestor node in the phylogeny occurs before all progeny.
Equivalently, this will preclude negative branch lengths.
Note that execution is faster if this option is FALSE.</p>
</td></tr>
<tr><td><code id="dater_+3A_clock">clock</code></td>
<td>
<p>The choice of molecular clock model. Choices are 'uncorrelated', 'additive', or 'strict'.</p>
</td></tr>
<tr><td><code id="dater_+3A_estimatesampletimes">estimateSampleTimes</code></td>
<td>
<p>If some sample times are not known with certainty,
bounds can be provided with this option. This should take the
form of a data frame with columns 'lower' and 'upper'
providing the sample time bounds for each uncertain tip. Row
names of the data frame should correspond to elements in
tip.label of the input tree. Tips with sample time bounds in
this data frame do not need to appear in the *sts* argument,
however if they are included in *sts*, that value will be
used as a starting condition for optimisation.</p>
</td></tr>
<tr><td><code id="dater_+3A_estimatesampletimes_densities">estimateSampleTimes_densities</code></td>
<td>
<p>An optional named list of log densities
which would be used as priors for unknown sample times. Names
should correspond to elements in tip.label with uncertain
sample times.</p>
</td></tr>
<tr><td><code id="dater_+3A_numstartconditions">numStartConditions</code></td>
<td>
<p>Will attempt optimisation from more than one starting point if &gt;0</p>
</td></tr>
<tr><td><code id="dater_+3A_clssolver">clsSolver</code></td>
<td>
<p>Which package should be used for constrained least-squares? Options are &quot;mgcv&quot; or &quot;limSolve&quot;</p>
</td></tr>
<tr><td><code id="dater_+3A_meanratelimits">meanRateLimits</code></td>
<td>
<p>Optional constraints for the mean substitution rate</p>
</td></tr>
<tr><td><code id="dater_+3A_ncpu">ncpu</code></td>
<td>
<p>Number of threads for parallel computing</p>
</td></tr>
<tr><td><code id="dater_+3A_parallel_foreach">parallel_foreach</code></td>
<td>
<p>If TRUE, will use the &quot;foreach&quot; package instead of the &quot;parallel&quot; package. This may work better on some HPC systems.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimates the calendar time of nodes in the given phylogenetic
tree with branches in units of substitutions per site. The
calendar time of each sample must also be specified and the length
of the sequences used to estimate the tree. If the tree is not
rooted, this function will estimate the root position. 
For an introduction to all options and features, see the vignette on Influenza H3N2: vignette(&quot;h3n2&quot;)
</p>
<p>Multiple molecular clock models are supported including a strict clock and two variations on relaxed clocks. The 'uncorrelated' relaxed clock is the Gamma-Poisson mixture presented by Volz and Frost (2017), while the 'additive' variance model was developed by Didelot &amp; Volz (2019).
</p>


<h3>Value</h3>

<p>A time-scaled tree and estimated molecular clock rate
</p>


<h3>References</h3>

<p>E.M. Volz and Frost, S.D.W. (2017) Scalable relaxed clock phylogenetic dating. Virus Evolution.
X. Didelot and Volz, E.M. (2019) Additive uncorrelated relaxed clock models.
</p>


<h3>Author(s)</h3>

<p>Erik M Volz &lt;erik.volz@gmail.com&gt;
</p>


<h3>See Also</h3>

<p>ape::chronos
ape::estimate.mu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate a random tree and sample times for demonstration
# make a random tree:
tre &lt;- ape::rtree(50)
# sample times based on distance from root to tip:
sts &lt;- setNames( ape::node.depth.edgelength( tre )[1:ape::Ntip(tre)], tre$tip.label)
# modify edge length to represent evolutionary distance with rate 1e-3:
tre$edge.length &lt;- tre$edge.length * 1e-3
# treedater: 
td &lt;- dater( tre, sts =sts , s = 1000, clock='strict', omega0=.0015)


</code></pre>

<hr>
<h2 id='goodnessOfFitPlot'>Produce a goodness of fit plot</h2><span id='topic+goodnessOfFitPlot'></span>

<h3>Description</h3>

<p>The sorted tail probabilties (p values) for each edge in the tree under the fitted model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>goodnessOfFitPlot(td)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="goodnessOfFitPlot_+3A_td">td</code></td>
<td>
<p>A treedater object generated by the <code>dater</code> function</p>
</td></tr>
</table>

<hr>
<h2 id='outlierLineages'>Detect lineages with unusually large evolutionary divergence under the fitted treedater model</h2><span id='topic+outlierLineages'></span>

<h3>Description</h3>

<p>Outliers are detected using the *stats::p.adjust* function and the 'fdr' function. The test requires that *dater* was used with the temporalConstraints=TRUE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outlierLineages(td, alpha = 0.05, type = c("tips", "internal", "all"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="outlierLineages_+3A_td">td</code></td>
<td>
<p>A fitted treedater object</p>
</td></tr>
<tr><td><code id="outlierLineages_+3A_alpha">alpha</code></td>
<td>
<p>The tail probability used for classifying lineages as outliers</p>
</td></tr>
<tr><td><code id="outlierLineages_+3A_type">type</code></td>
<td>
<p>Should outliers be detected on tip lineages, interal lineages, or all lineages?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame summarizing for each lineage the p values, adjusted p values ('q'), likelihood, rates, and branch lengths.
</p>


<h3>See Also</h3>

<p>dater
outlier.tips
</p>

<hr>
<h2 id='outlierTips'>Detect terminal lineages with unusually large evolutionary divergence under the fitted treedater model</h2><span id='topic+outlierTips'></span>

<h3>Description</h3>

<p>This is a convient wrapper of the *outlier.lineages*
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outlierTips(td, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="outlierTips_+3A_td">td</code></td>
<td>
<p>A fitted treedater object</p>
</td></tr>
<tr><td><code id="outlierTips_+3A_alpha">alpha</code></td>
<td>
<p>The tail probability used for classifying lineages as outliers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame summarizing for each lineage the p values, adjusted p values ('q'), likelihood, rates, and branch lengths.
</p>


<h3>See Also</h3>

<p>dater
outlier.lineages
</p>

<hr>
<h2 id='parboot'>Estimate of confidence intervals using parametric bootstrap for molecular clock dating.</h2><span id='topic+parboot'></span>

<h3>Description</h3>

<p>This function simulates phylogenies with branch lengths in units
of substitutions per site. Simulations are based on a fitted
treedater object which provides parameters of the molecular clock
model. The treedater method is applied to each simulated tree
providing a Monte Carlo estimate of variance in rates and dates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parboot(td, nreps = 100, ncpu = 1, overrideTempConstraint = TRUE,
  overrideClock = NULL, overrideSearchRoot = TRUE,
  overrideSeqLength = NULL, quiet = TRUE, normalApproxTMRCA = FALSE,
  parallel_foreach = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parboot_+3A_td">td</code></td>
<td>
<p>A fitted treedater object</p>
</td></tr>
<tr><td><code id="parboot_+3A_nreps">nreps</code></td>
<td>
<p>Integer number of simulations to be carried out</p>
</td></tr>
<tr><td><code id="parboot_+3A_ncpu">ncpu</code></td>
<td>
<p>Number of threads to use for parallel computation. Recommended.</p>
</td></tr>
<tr><td><code id="parboot_+3A_overridetempconstraint">overrideTempConstraint</code></td>
<td>
<p>If TRUE (default) will not enforce positive branch lengths in simualtion replicates. Will speed up execution.</p>
</td></tr>
<tr><td><code id="parboot_+3A_overrideclock">overrideClock</code></td>
<td>
<p>May be 'strict' or 'additive' or 'uncorrelated' in which case will force simulations to fit the corresponding model. If ommitted, will inherit the clock model from td</p>
</td></tr>
<tr><td><code id="parboot_+3A_overridesearchroot">overrideSearchRoot</code></td>
<td>
<p>If TRUE, will re-use root position from input treedater tree. Otherwise may re-estimate root position in simulations</p>
</td></tr>
<tr><td><code id="parboot_+3A_overrideseqlength">overrideSeqLength</code></td>
<td>
<p>Optional sequence length to use in simulations</p>
</td></tr>
<tr><td><code id="parboot_+3A_quiet">quiet</code></td>
<td>
<p>If TRUE will minimize output printed to screen</p>
</td></tr>
<tr><td><code id="parboot_+3A_normalapproxtmrca">normalApproxTMRCA</code></td>
<td>
<p>If TRUE will use estimate standard deviation from simulation replicates and report confidence interval based on normal distribution</p>
</td></tr>
<tr><td><code id="parboot_+3A_parallel_foreach">parallel_foreach</code></td>
<td>
<p>If TRUE will use the foreach package for parallelization. May work better on HPC systems.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the original treedater fit estimated the root position, root
position will also be estimated for each simulation, so the
returned trees may have different root positions. Some replicates
may converge to a strict clock or a relaxed clock, so the
parameter estimates in each replicate may not be directly
comparable. It is possible to compute confidence intervals for the
times of particular nodes or for estimated sample times by
inspecting the output from each fitted treedater object, which is
contained in the $trees attribute.
</p>


<h3>Value</h3>

<p>A list with elements 
</p>

<ul>
<li><p> trees: The fitted treedater objects corresponding to each simulation
</p>
</li>
<li><p> meanRates: Vector of estimated rates for each simulation
</p>
</li>
<li><p> meanRate_CI: Confidence interval for substitution rate
</p>
</li>
<li><p> coef_of_variation_CI: Confidence interval for rate variation
</p>
</li>
<li><p> timeOfMRCA_CI: Confidence interval for time of common ancestor
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Erik M Volz &lt;erik.volz@gmail.com&gt;
</p>


<h3>See Also</h3>

<p>dater
boot
</p>


<h3>Examples</h3>

<pre><code class='language-R'># make a random tree
tre &lt;- ape::rtree(25)
# simulate sample times based on distance from root to tip:
sts &lt;- setNames( ape::node.depth.edgelength( tre )[1:ape::Ntip(tre)], tre$tip.label)
# modify edge length to represent evolutionary distance with rate 1e-3:
tre$edge.length &lt;- tre$edge.length * 1e-3
# treedater: 
td &lt;- dater( tre, sts=sts, s=1000, clock='strict', omega0=.0015 )
# parametric bootstrap: 
pb &lt;- parboot( td, nreps=25 )
# plot lineages through time
plot( pb )

</code></pre>

<hr>
<h2 id='plot.bootTreedater'>Plots lineages through time and confidence intervals estimated by bootstrap.</h2><span id='topic+plot.bootTreedater'></span>

<h3>Description</h3>

<p>Plots lineages through time and confidence intervals estimated by bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bootTreedater'
plot(x, t0 = NA, res = 100, ggplot = FALSE,
  cumulative = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.bootTreedater_+3A_x">x</code></td>
<td>
<p>A bootTreedater object produced by *parboot* or *boot*</p>
</td></tr>
<tr><td><code id="plot.bootTreedater_+3A_t0">t0</code></td>
<td>
<p>The lower bound of the time axis to show</p>
</td></tr>
<tr><td><code id="plot.bootTreedater_+3A_res">res</code></td>
<td>
<p>The number of time points on the time axis</p>
</td></tr>
<tr><td><code id="plot.bootTreedater_+3A_ggplot">ggplot</code></td>
<td>
<p>If TRUE, will return a plot made with the ggplot2 package</p>
</td></tr>
<tr><td><code id="plot.bootTreedater_+3A_cumulative">cumulative</code></td>
<td>
<p>If TRUE, will show only decreasing lineages through time</p>
</td></tr>
<tr><td><code id="plot.bootTreedater_+3A_...">...</code></td>
<td>
<p>Additional arg's are passed to *ggplot* or *plot*</p>
</td></tr>
</table>

<hr>
<h2 id='relaxedClockTest'>Use parametric bootstrap to test if relaxed clock offers improved fit to data.</h2><span id='topic+relaxedClockTest'></span>

<h3>Description</h3>

<p>This function simulates phylogenies with branch lengths in units
of substitutions per site. Simulations are based on a fitted
treedater object which provides parameters of the molecular clock
model. The coefficient of variation of rates is estimated using a
relaxed clock model applied to strict clock simulations. Estimates
of the CV is then compared to the null distribution provided by
simulations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relaxedClockTest(..., nreps = 100, overrideTempConstraint = T,
  ncpu = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relaxedClockTest_+3A_...">...</code></td>
<td>
<p>arguments passed to *dater*</p>
</td></tr>
<tr><td><code id="relaxedClockTest_+3A_nreps">nreps</code></td>
<td>
<p>Integer number of simulations</p>
</td></tr>
<tr><td><code id="relaxedClockTest_+3A_overridetempconstraint">overrideTempConstraint</code></td>
<td>
<p>see *parboot*</p>
</td></tr>
<tr><td><code id="relaxedClockTest_+3A_ncpu">ncpu</code></td>
<td>
<p>Number of threads to use for parallel computation. Recommended.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will print the optimal clock model
and the distribution of the coefficient of variation statistic under the null hypothesis (strict
clock). Parameters passed to this function should be the same as when calling *dater*.
</p>


<h3>Value</h3>

<p>A list with elements:
</p>

<ul>
<li><p> strict_treedater: A dater object under a strict clock
</p>
</li>
<li><p> relaxed_treedater: A dater object under a relaxed clock
</p>
</li>
<li><p> clock: The favoured clock model 
</p>
</li>
<li><p> parboot: Result of call to *parboot* using fitted treedater and forcing a relaxed clock
</p>
</li>
<li><p> nullHypothesis_coef_of_variation_CI: The null hypothesis CV
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Erik M Volz &lt;erik.volz@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate a tree 
tre &lt;- ape::rtree(25)
# sample times based on distance from root to tip:
sts &lt;- setNames( ape::node.depth.edgelength( tre )[1:ape::Ntip(tre)], tre$tip.label)
# modify edge length to represent evolutionary distance with rate 1e-3:
tre$edge.length &lt;- tre$edge.length * 1e-3
relaxedClockTest( tre, sts, s= 1000,  omega0=.0015 , nreps=25)


</code></pre>

<hr>
<h2 id='rootToTipRegressionPlot'>Plot evolutionary distance from root to sample times and estimated internal node times and regression lines</h2><span id='topic+rootToTipRegressionPlot'></span>

<h3>Description</h3>

<p>If a range of sample times was given, these will be estimated. Red and black respectively indicate sample and internal nodes. 
This function will print statistics computed from the linear regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rootToTipRegressionPlot(td, show.tip.labels = FALSE, textopts = NULL,
  pointopts = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rootToTipRegressionPlot_+3A_td">td</code></td>
<td>
<p>A fitted treedater object</p>
</td></tr>
<tr><td><code id="rootToTipRegressionPlot_+3A_show.tip.labels">show.tip.labels</code></td>
<td>
<p>If TRUE, the names of each sample will be plotted at the their corresponding time and evoutionary distance</p>
</td></tr>
<tr><td><code id="rootToTipRegressionPlot_+3A_textopts">textopts</code></td>
<td>
<p>An optional list of parameters for plotted tip labels. Passed to the *text* function.</p>
</td></tr>
<tr><td><code id="rootToTipRegressionPlot_+3A_pointopts">pointopts</code></td>
<td>
<p>An optional list of parameters for plotted points if showing tip labels. Passed to the *points* function.</p>
</td></tr>
<tr><td><code id="rootToTipRegressionPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments are passed to plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The fitted linear model (class 'lm')
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate a random tree and sample times for demonstration
# make a random tree:
tre &lt;- ape::rtree(50)
# sample times based on distance from root to tip:
sts &lt;- setNames( ape::node.depth.edgelength( tre )[1:ape::Ntip(tre)], tre$tip.label)
# modify edge length to represent evolutionary distance with rate 1e-3:
tre$edge.length &lt;- tre$edge.length * 1e-3
# treedater: 
td &lt;- dater( tre, sts =sts, clock='strict', s = 1000, omega0=.0015 )
# root to tip regression: 
fit = rootToTipRegressionPlot( td )
summary(fit)

</code></pre>

<hr>
<h2 id='sampleYearsFromLabels'>Compute a vector of numeric sample times from labels in a sequence aligment or phylogeny</h2><span id='topic+sampleYearsFromLabels'></span>

<h3>Description</h3>

<p>Compute a vector of numeric sample times from labels in a sequence aligment or phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleYearsFromLabels(tips, dateFormat = "%Y-%m-%d",
  delimiter = NULL, index = NULL, regex = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampleYearsFromLabels_+3A_tips">tips</code></td>
<td>
<p>A character vector supplying the name of each sample</p>
</td></tr>
<tr><td><code id="sampleYearsFromLabels_+3A_dateformat">dateFormat</code></td>
<td>
<p>The format of the sample date. See ?Date for more information</p>
</td></tr>
<tr><td><code id="sampleYearsFromLabels_+3A_delimiter">delimiter</code></td>
<td>
<p>Character(s) which separate data in each label</p>
</td></tr>
<tr><td><code id="sampleYearsFromLabels_+3A_index">index</code></td>
<td>
<p>Integer position of the date string in each label with respect to *delimiter*</p>
</td></tr>
<tr><td><code id="sampleYearsFromLabels_+3A_regex">regex</code></td>
<td>
<p>A regular expression for finding the date substring. Should not be used with *delimiter* or *index*</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector with sample time in decimal format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## A couple of labels for Ebola virus sequences: 
sampleYearsFromLabels( c('EBOV|AA000000|EM104|SierraLeone_EM|2014-06-02'
                       , 'EBOV|AA000000|G3713|SierraLeone_G|2014-06-09')
, delimiter='|' )
## Equivalently: 
sampleYearsFromLabels( c('EBOV|AA000000|EM104|SierraLeone_EM|2014-06-02'
                       , 'EBOV|AA000000|G3713|SierraLeone_G|2014-06-09')
 , regex='[0-9]+-[0-9]+-[0-9]+')

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
