<!DOCTYPE html><html lang="en-US"><head><title>Help for package treefit</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {treefit}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#generate_2d_n_arms_linked_star_data'><p>Generate a 2-dimensional linked star tree data</p></a></li>
<li><a href='#generate_2d_n_arms_star_data'><p>Generate a 2-dimensional star tree data</p></a></li>
<li><a href='#generate_n_arms_star_data'><p>Generate a multi-dimensional star tree data</p></a></li>
<li><a href='#perturbate_knn'><p>Generate perturbated expression by k-NN data</p></a></li>
<li><a href='#perturbate_poisson'><p>Generate perturbated counts by the Poisson distribution</p></a></li>
<li><a href='#plot.treefit'><p>Plot estimated results</p></a></li>
<li><a href='#treefit'><p>Estimate the goodness-of-fit between tree models and data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>The First Software for Quantitative Trajectory Inference</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform two types of analysis: 1) checking the
    goodness-of-fit of tree models to your single-cell gene expression
    data; and 2) deciding which tree best fits your data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://hayamizu-lab.github.io/treefit-r/">https://hayamizu-lab.github.io/treefit-r/</a>,
<a href="https://github.com/hayamizu-lab/treefit-r/">https://github.com/hayamizu-lab/treefit-r/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/hayamizu-lab/treefit-r/issues">https://github.com/hayamizu-lab/treefit-r/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, igraph, patchwork, pracma</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Seurat, gridExtra, knitr, plotly, qpdf, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-09 10:24:42 UTC; kou</td>
</tr>
<tr>
<td>Author:</td>
<td>Momoko Hayamizu <a href="https://orcid.org/0000-0001-8825-6331"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Kouhei Sutou <a href="https://orcid.org/0000-0002-5528-5109"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ryohei Suzuki <a href="https://orcid.org/0000-0002-1528-665X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Hiromi Ishii <a href="https://orcid.org/0000-0002-7752-1782"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kouhei Sutou &lt;kou@clear-code.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-09 11:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='generate_2d_n_arms_linked_star_data'>Generate a 2-dimensional linked star tree data</h2><span id='topic+generate_2d_n_arms_linked_star_data'></span>

<h3>Description</h3>

<p>Generate a 2-dimensional linked star tree data. Each
star tree data contain <code>n_samples_vector[i]</code> data points and fit
a star tree with <code>n_arms_vector[i]</code> arms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_2d_n_arms_linked_star_data(n_samples_vector, n_arms_vector, fatness)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_2d_n_arms_linked_star_data_+3A_n_samples_vector">n_samples_vector</code></td>
<td>
<p>The vector of the number of samples to be
generated. For example, <code>c(200, 100, 300)</code> means that the first
tree has 200 samples, the second tree has 100 samples and the
third tree has 300 samples.</p>
</td></tr>
<tr><td><code id="generate_2d_n_arms_linked_star_data_+3A_n_arms_vector">n_arms_vector</code></td>
<td>
<p>The vector of the number of arms to be
generated.  For example, <code>c(3, 2, 5)</code> means the first tree fits a
star tree with 3 arms, the second tree fits a star tree with 2
arms and the third tree fits a star tree with 5 arms. The size of
<code>n_arms_vector</code> must equal to the size of <code>n_samples_vector</code>.</p>
</td></tr>
<tr><td><code id="generate_2d_n_arms_linked_star_data_+3A_fatness">fatness</code></td>
<td>
<p>How fat from the based tree. <code style="white-space: pre;">&#8288;[0.0, 1.0]&#8288;</code> is
available value range.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A generated <code>martix</code>. The rows and columns correspond to
samples and features.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a 2-dimensional linked star tree data that contain
# 200-400-300 data points and fit a linked star tree with 3-5-4
# arms. The generated data are a bit noisy but tree-like.
linked_star.tree_like &lt;-
  treefit::generate_2d_n_arms_linked_star_data(c(200, 400, 300),
                                               c(3, 5, 4),
                                               0.1)
plot(linked_star.tree_like)

# Generate a 2-dimensional linked star tree data that contain
# 300-200 data points and fit a linked star tree with 4-3 arms.
# The generated data are very noisy and less tree-like.
linked_star.less_tree_like &lt;-
  treefit::generate_2d_n_arms_linked_star_data(c(300, 200),
                                               c(4, 3),
                                               0.9)
plot(linked_star.less_tree_like)

</code></pre>

<hr>
<h2 id='generate_2d_n_arms_star_data'>Generate a 2-dimensional star tree data</h2><span id='topic+generate_2d_n_arms_star_data'></span>

<h3>Description</h3>

<p>Generate a 2-dimensional star tree data that contain
<code>n_samples</code> data points and fit a star tree with <code>n_arms</code> arms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_2d_n_arms_star_data(n_samples, n_arms, fatness)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_2d_n_arms_star_data_+3A_n_samples">n_samples</code></td>
<td>
<p>The number of samples to be generated.</p>
</td></tr>
<tr><td><code id="generate_2d_n_arms_star_data_+3A_n_arms">n_arms</code></td>
<td>
<p>The number of arms to be generated.</p>
</td></tr>
<tr><td><code id="generate_2d_n_arms_star_data_+3A_fatness">fatness</code></td>
<td>
<p>How fat from the based star tree. <code style="white-space: pre;">&#8288;[0.0, 1.0]&#8288;</code> is
available value range.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A generated <code>martix</code>. The rows and columns correspond to
samples and features.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a 2-dimensional star tree data that contain 500 data points
# and fit a star tree with 3 arms. The generated data are a bit noisy but
# tree-like.
star.tree_like &lt;- treefit::generate_2d_n_arms_star_data(500, 3, 0.1)
plot(star.tree_like)

# Generate a 2-dimensional star tree data that contain 600 data points
# and fit a star tree with 5 arms. The generated data are very noisy and
# less tree-like.
star.less_tree_like &lt;- treefit::generate_2d_n_arms_star_data(600, 5, 0.9)
plot(star.less_tree_like)

</code></pre>

<hr>
<h2 id='generate_n_arms_star_data'>Generate a multi-dimensional star tree data</h2><span id='topic+generate_n_arms_star_data'></span>

<h3>Description</h3>

<p>Generate a multi-dimensional star tree data that contain
<code>n_samples</code> data points and fit a star tree with <code>n_arms</code> arms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_n_arms_star_data(n_features, n_samples, n_arms, fatness)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_n_arms_star_data_+3A_n_features">n_features</code></td>
<td>
<p>The number of features (dimensions) to be
generated.</p>
</td></tr>
<tr><td><code id="generate_n_arms_star_data_+3A_n_samples">n_samples</code></td>
<td>
<p>The number of samples to be generated.</p>
</td></tr>
<tr><td><code id="generate_n_arms_star_data_+3A_n_arms">n_arms</code></td>
<td>
<p>The number of arms to be generated.</p>
</td></tr>
<tr><td><code id="generate_n_arms_star_data_+3A_fatness">fatness</code></td>
<td>
<p>How fat from the based star tree. <code style="white-space: pre;">&#8288;[0.0, 1.0]&#8288;</code> is
available value range.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A generated <code>martix</code>. The rows and columns correspond to
samples and features.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a 100-dimensional star tree data that contain 500 data points
# and fit a star tree with 3 arms. The generated data are a bit noisy but
# tree-like.
star100.tree_like &lt;- treefit::generate_n_arms_star_data(100, 500, 3, 0.1)
# Reduce dimension to visualize.
star3.tree_like = prcomp(star100.tree_like, rank.=3)$x
plotly::plot_ly(data.frame(star3.tree_like),
                x=~PC1,
                y=~PC2,
                z=~PC3,
                type="scatter3d",
                mode="markers",
                marker=list(size=1))

</code></pre>

<hr>
<h2 id='perturbate_knn'>Generate perturbated expression by k-NN data</h2><span id='topic+perturbate_knn'></span>

<h3>Description</h3>

<p>Generate perturbated expression from the original
expression based on k-NN (k-nearest neighbor) data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perturbate_knn(expression, strength = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="perturbate_knn_+3A_expression">expression</code></td>
<td>
<p>The original expression. The rows and columns
correspond to samples and features. The expression is normalized
count of features.</p>
</td></tr>
<tr><td><code id="perturbate_knn_+3A_strength">strength</code></td>
<td>
<p>How much perturbated. <code>0.0</code> is weak. <code>1.0</code> is strong.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A perturbated expression as a <code>matrix</code>. The matrix's
expression values are perturbated from the original expression
values. The shape of the matrix is the same as the original
expression. The dimension names of the matrix are also the same as
the original expression.
</p>


<h3>Note</h3>

<p>This is an API for advanced users. This API may be changed.
</p>

<hr>
<h2 id='perturbate_poisson'>Generate perturbated counts by the Poisson distribution</h2><span id='topic+perturbate_poisson'></span>

<h3>Description</h3>

<p>Generate perturbated counts from the original counts
by the Poisson distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perturbate_poisson(counts, strength = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="perturbate_poisson_+3A_counts">counts</code></td>
<td>
<p>The original counts. The rows and columns correspond
to samples and features. The values are count of features.</p>
</td></tr>
<tr><td><code id="perturbate_poisson_+3A_strength">strength</code></td>
<td>
<p>How much perturbated. <code>0.0</code> is weak. <code>1.0</code> is strong.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A perturbated counts as a <code>matrix</code>. The matrix's counts are
perturbated from the original counts. The shape of the matrix is
the same as the original counts. The dimension names of the
matrix are also the same as the original counts.
</p>


<h3>Note</h3>

<p>This is an API for advanced users. This API may be changed.
</p>

<hr>
<h2 id='plot.treefit'>Plot estimated results</h2><span id='topic+plot.treefit'></span>

<h3>Description</h3>

<p>Plot estimate results to get insight.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'treefit'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.treefit_+3A_x">x</code></td>
<td>
<p>The estimated result by <code>treefit()</code> to be visualized.</p>
</td></tr>
<tr><td><code id="plot.treefit_+3A_...">...</code></td>
<td>
<p>The more estimated results to be visualized together or
other graphical parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot object as a <code>ggplot</code> object. It plots the given one
or more estimated results to get insights from one or more
<code>treefit()</code> results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generate a tree data.
tree &lt;- treefit::generate_2d_n_arms_star_data(200, 3, 0.1)
# Estimate the goodness-of-fit between tree models and the tree data.
fit &lt;- treefit::treefit(list(expression=tree), "tree")
# Visualize the estimated result.
plot(fit)

# You can mix multiple estimated results by adding "name" column.
tree2 &lt;- treefit::generate_2d_n_arms_star_data(200, 3, 0.9)
fit2 &lt;- treefit::treefit(list(expression=tree2), "tree2")
plot(fit, fit2)

## End(Not run)

</code></pre>

<hr>
<h2 id='treefit'>Estimate the goodness-of-fit between tree models and data</h2><span id='topic+treefit'></span>

<h3>Description</h3>

<p>Estimate the goodness-of-fit between tree models and
data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>treefit(
  target,
  name = NULL,
  perturbations = NULL,
  normalize = NULL,
  reduce_dimension = NULL,
  build_tree = NULL,
  max_p = 20,
  n_perturbations = 20
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="treefit_+3A_target">target</code></td>
<td>
<p>The target data to be estimated. It must be one of them:
</p>

<ul>
<li> <p><code>list(counts=COUNTS, expression=EXPRESSION)</code>: You must specify
at least one of <code>COUNTS</code> and <code>EXPRESSION</code>.  They are
<code>matrix</code>. The rows and columns correspond to samples such as
cells and features such as genes. <code>COUNTS</code>'s value is count
data such as the number of genes expressed.  <code>EXPRESSION</code>'s
value is normalized count data.
</p>
</li>
<li> <p><code>Seurat</code> object
</p>
</li></ul>
</td></tr>
<tr><td><code id="treefit_+3A_name">name</code></td>
<td>
<p>The name of <code>target</code> as string.</p>
</td></tr>
<tr><td><code id="treefit_+3A_perturbations">perturbations</code></td>
<td>
<p>How to perturbate the target data.
</p>
<p>If this is <code>NULL</code>, all available perturbation methods are used.
</p>
<p>You can specify used perturbation methods as <code>list</code>. Here are
available methods:</p>
</td></tr>
<tr><td><code id="treefit_+3A_normalize">normalize</code></td>
<td>
<p>How to normalize counts data.
</p>
<p>If this is <code>NULL</code>, the default normalization is applied.
</p>
<p>You can specify a function that normalizes counts data.</p>
</td></tr>
<tr><td><code id="treefit_+3A_reduce_dimension">reduce_dimension</code></td>
<td>
<p>How to reduce dimension of expression data.
</p>
<p>If this is <code>NULL</code>, the default dimensionality reduction is applied.
</p>
<p>You can specify a function that reduces dimension of expression data.</p>
</td></tr>
<tr><td><code id="treefit_+3A_build_tree">build_tree</code></td>
<td>
<p>How to build a tree of expression data.
</p>
<p>If this is <code>NULL</code>, MST is built.
</p>
<p>You can specify a function that builds tree of expression data.</p>
</td></tr>
<tr><td><code id="treefit_+3A_max_p">max_p</code></td>
<td>
<p>How many low dimension Laplacian eigenvectors are used.
</p>
<p>The default is 20.</p>
</td></tr>
<tr><td><code id="treefit_+3A_n_perturbations">n_perturbations</code></td>
<td>
<p>How many times to perturb.
</p>
<p>The default is 20.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An estimated result as a <code>treefit</code> object. It has the
following attributes:
</p>

<ul>
<li> <p><code>max_cca_distance</code>: The result of max canonical correlation
analysis distance as <code>data.frame</code>.
</p>
</li>
<li> <p><code>rms_cca_distance</code>: The result of root mean square canonical
correlation analysis distance as <code>data.frame</code>.
</p>
</li>
<li> <p><code>n_principal_paths_candidates</code>: The candidates of the number of
principal paths.
</p>
</li></ul>

<p><code>data.frame</code> of <code>max_cca_distance</code> and <code>rms_cca_distance</code> has the
same structure. They have the following columns:
</p>

<ul>
<li> <p><code>p</code>: Dimensionality of the feature space of tree structures.
</p>
</li>
<li> <p><code>mean</code>: The mean of the target distance values.
</p>
</li>
<li> <p><code>standard_deviation</code>: The standard deviation of the target
distance values.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Generate a star tree data that have normalized expression values
# not count data.
star &lt;- treefit::generate_2d_n_arms_star_data(300, 3, 0.1)
# Estimate tree-likeness of the tree data.
fit &lt;- treefit::treefit(list(expression=star))

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
