<!DOCTYPE html><html><head><title>Help for package TriadSim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TriadSim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fit.risk.model.par'><p>Resample families based on the risk model</p></a></li>
<li><a href='#get.brks'><p>Picking chromosomal breaking points</p></a></li>
<li><a href='#get.target.geno'><p>Getting genotypes of the target SNPs</p></a></li>
<li><a href='#glue.chr.segment.par'><p>Splicing chromosomal segments</p></a></li>
<li><a href='#pick_target.snp'><p>Pick target SNPs in the pathway</p></a></li>
<li><a href='#rcmb.rate'><p>An example recombination rate dataset</p></a></li>
<li><a href='#snp.all2'><p>SNPs in the PLINK files</p></a></li>
<li><a href='#TriadSim'><p>Simulation main function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Simulating Triad Genomewide Genotypes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulate genotypes for case-parent triads, case-control, and quantitative trait samples with realistic linkage diequilibrium structure and allele frequency distribution. For studies of epistasis one can simulate models that involve specific SNPs at specific sets of loci, which we will refer to as "pathways". TriadSim generates genotype data by resampling triad genotypes from existing data. The details of the method is described in the manuscript under preparation "Simulating Autosomal Genotypes with Realistic Linkage Disequilibrium and a Spiked in Genetic Effect" Shi, M., Umbach, D.M., Wise A.S., Weinberg, C.R. 	</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.2)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>biocViews:</td>
<td>snpStats</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, parallel, snpStats, foreach, doParallel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-08 16:52:20 UTC; shi2</td>
</tr>
<tr>
<td>Author:</td>
<td>Min Shi [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Min Shi &lt;shi2@niehs.nih.gov&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-08 17:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='fit.risk.model.par'>Resample families based on the risk model</h2><span id='topic+fit.risk.model.par'></span>

<h3>Description</h3>

<p>This function selects families based on the prespecified risk model. It can simulate a homogenous scenario
or a stratified scenario with two subpopulations. When e.fr is given rather than the default NA
the risk model can involve exposure main effects as well as gene by exposure interation. This function is parallelized
and the default number of cores for parallelization is set as the ceiling of half of the total number of CPU cores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit.risk.model.par(
  n.ped,
  brks,
  target.snp,
  fam.pos,
  mom.tar,
  dad.tar,
  kid.tar,
  pathways,
  betas.e0,
  e.fr = NA,
  betas.e,
  pop1.frac = NA,
  rate.beta = NA,
  is.case = TRUE,
  qtl = FALSE,
  out.put.file = NA,
  no_cores = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.risk.model.par_+3A_n.ped">n.ped</code></td>
<td>
<p>is an integer giving the number of trios to be simulated</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_brks">brks</code></td>
<td>
<p>a matrix of integers showing where the chromosomal breaks is to take place for 
each individual in the simulated trios.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_target.snp">target.snp</code></td>
<td>
<p>is a vector of integers showing the row number of the target SNPs in the .bim file.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_fam.pos">fam.pos</code></td>
<td>
<p>is a matrix showing the chromosomal segments out of which is each target SNP 
selected for each simulated trio.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_mom.tar">mom.tar</code></td>
<td>
<p>is a matrix containing genotypes of the target SNPs in the mothers of the original data 
for simulations of a homogenous population. For simulations under population stratification it is 
a list of two matrices each containing genotypes of the mothers' target SNP genotypes in one of the two subpopulations.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_dad.tar">dad.tar</code></td>
<td>
<p>is a matrix containing the genotypes of the target SNPs in the fathers of the original data
for simulations of a homogenous population. For simulations under population stratification it is 
a list of two matrices each containing fathers' target SNP genotypes in one of the two subpopulations.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_kid.tar">kid.tar</code></td>
<td>
<p>is a matrix with containing genotypes of the target SNP in the children stacking on top of the complements of the original data
for simulations of a homogenous population. For simulations under population stratification it is 
a list of two matrices each containing children's and complements' target SNP genotypes in one of the two subpopulations.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_pathways">pathways</code></td>
<td>
<p>is a list of vectors of integers. Each vector of integers denotes the SNPs involved in a particular pathway. E.g. list(1:4,5:8) denote that there
are two pathways. SNPs 1-4 are in the first pathway and SNPs 5-8 are in the second.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_betas.e0">betas.e0</code></td>
<td>
<p>is a vector of doubles giving the beta coefficients of the logit risk model for the unexposed individuals. The length of the vector
should be 1+ number_of_risk_pathway. The first number is a function of the disease prevalence in the
unexposed individual who does not carry any copies of the risk pathway. The numbers after that gives the odds ratios 
for carrying one/two copies of the risk pathways comparing to those who do not carry any copies of the pathways in the unxposed group. e.g., c(-6.4, 0.5,1)
means the baseline disease prevalence is exp(-6.4)/(1+exp(-6.4)) and the log OR for carrying at least one copy of the first pathway is 0.5 and
that for carrying at least one copy of the second pathway is 1.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_e.fr">e.fr</code></td>
<td>
<p>is a double number between 0 and 1 which gives the exposure prevalence.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_betas.e">betas.e</code></td>
<td>
<p>is a vector of doubles giving the beta coefficients of the logit risk model for the exposed individuals. The length of the vector
should be 1+ number_of_risk_pathway. The first number is a function the disease prevalence in the
exposed individual who does not carry any copies of the risk pathway. The numbers after that gives the odds ratios 
for carrying one/two copies of the risk pathways comparing to those who do not carry any copies of the pathways in the exposed group.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_pop1.frac">pop1.frac</code></td>
<td>
<p>is a double number between 0 and 1 which gives the fraction of subpopulation 1 out of the two subpopulations for a population stratification scenario.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_rate.beta">rate.beta</code></td>
<td>
<p>is a double number giving the log OR of disease prevalence in population 2 over that in population 1.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_is.case">is.case</code></td>
<td>
<p>is a boolean variable. When is.case = TRUE case-parents trios will be simulated.Otherwise, control-parents trios will be simulated.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_qtl">qtl</code></td>
<td>
<p>is a boolean variable denoting whether a quantitative trait (qtl=TRUE) or a binary trait (qtl=FALSE) is to be simulated. For a binary trait only affected 
families will be kept. The default value is qtl=FALSE.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_out.put.file">out.put.file</code></td>
<td>
<p>is a character string giving the base file name for the output file. When a non-default value is given the fucntion will write the following files to 
the designated directory: 
a file with name ending with &quot;exp.txt&quot; containing the exposure data when exposure is involved in the risk model.
a file with name ending with &quot;pop.txt&quot; containing information on subpopulation membership when the simulation involves a stratified scenario.
a file with name ending with &quot;pheno.tx&quot; containing quantitative trait phenotype when a quantitative trait is involved.
When out.put.file is the default value NA the file names for the above three files are: exposure.txt, population.txt, phenotype.txt.</p>
</td></tr>
<tr><td><code id="fit.risk.model.par_+3A_no_cores">no_cores</code></td>
<td>
<p>is an integer which specifies the number of CPU cores to be parallelized.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list of five elements. The first one is a matrix of integers giving the families (in terms of row number) 
selected for each simulated trio and each chromosomal segment. The second one is a matrix giving the genotypes on the 
target SNPs in the simulated trio. The third one is relevant only when exposure is involved. It is a vector of 0's and 1's giving the exposure status of each simulated trio when the risk model involves
exposure. The fourth element is relevant only in simulations of stratified scenarios. It is a vector of 1's and 2's giving the memebership of the subpopulation groups of each simulated trio. 
The fifth element is relevant only in simulations of a quantitative trait. It is a vector of doubles giving the phenotype 
values for simulations of a quantitative trait.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tar.snp &lt;- c(21, 118, 121, 140, 155, 168, 218, 383) 
found.brks &lt;- get.brks(N.brk=3,n.ped=1000, snp.all2, tar.snp,rcmb.rate=NA)
breaks &lt;- found.brks[[1]]
family.position &lt;- found.brks[[2]] 
betas &lt;- c(-6.4, 3.2, 5.8)
pwy &lt;- list(1:4,5:8)
m.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_mom')
f.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_dad')
k.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_kid')
# the preloaded data frame snp.all2 contains the data frame read from the corresponding .bim file.
target.geno &lt;- get.target.geno(c(m.file,f.file,k.file), tar.snp,snp.all2)
mom.target &lt;- target.geno[[1]]
dad.target &lt;- target.geno[[2]]
kid.target &lt;- target.geno[[3]]
## Not run:  
fitted.model &lt;- fit.risk.model.par(n.ped=1000,brks=breaks,target.snp=tar.snp, 
fam.pos=family.position,mom.tar=mom.target,dad.tar=dad.target, kid.tar=kid.target,  
pathways=pwy,betas, e.fr=NA, betas,pop1.frac= NA,rate.beta=NA,no_cores=2)

## End(Not run)
</code></pre>

<hr>
<h2 id='get.brks'>Picking chromosomal breaking points</h2><span id='topic+get.brks'></span>

<h3>Description</h3>

<p>The breaking points at each chromosome can be picked manually or use this function. 
When a data frame containing the recombination rates (rcmb.rate) is provided the function
tends to pick the breaking points at recombination hotspots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.brks(N.brk, n.ped, snp.all2, target.snp, rcmb.rate = NA, same.brk = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.brks_+3A_n.brk">N.brk</code></td>
<td>
<p>is an integer giving the number of breaks to be picked for each chromosome.</p>
</td></tr>
<tr><td><code id="get.brks_+3A_n.ped">n.ped</code></td>
<td>
<p>is an integer giving the number of trios to be simulated</p>
</td></tr>
<tr><td><code id="get.brks_+3A_snp.all2">snp.all2</code></td>
<td>
<p>is a dataframe containing the list of SNPs in PLINK .bim format. Two columns of the dataframe
is used: column 1 with column name &quot;V1&quot; containing the chromosome number and column 4 with column name &quot;V4&quot;
containing the chromosomal postion of the SNPs.</p>
</td></tr>
<tr><td><code id="get.brks_+3A_target.snp">target.snp</code></td>
<td>
<p>is a vector of integers showing the row number of the target SNPs in the .bim file.</p>
</td></tr>
<tr><td><code id="get.brks_+3A_rcmb.rate">rcmb.rate</code></td>
<td>
<p>the default value is NA. rcmb.rate is a dataframe containing the recombination rates at each SNP. The ordering of the SNPs should 
be identical to that of snp.all2. It contains 4 columns with column names 'CHR','RS','POS',and 'RATE with the 
corresponding values for &quot;the chromosomal number&quot;, &quot;SNP rs number&quot;, &quot;chromosomal position&quot;,
and &quot;recombination rate&quot;. The recombination rate represents the maximum recombination rate in the chromosomal
region between the current SNP and the SNP above (or the first basepair of the chromosome for the first SNP on a chromosome).
When no rcmb.rate is provided the function will pick the breaking points randomly where keeping the breaking
points in between target SNPs. An example recombination rate data frame &quot;rcmb.rate&quot; is already loaded with the package.</p>
</td></tr>
<tr><td><code id="get.brks_+3A_same.brk">same.brk</code></td>
<td>
<p>is an indicator variable to denote whethere the same set of breaking points will be used for all simulated triads</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two elements is returned. The first one is a matrix of integers showing where 
the chromosomal breaks is to take place for each individuals in the simulated trios. The second one
is a matrix showing the chromosomal segments out of which each target SNP is selected for each simulated trio.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tar.snp &lt;- c(21, 118, 121, 140, 155, 168, 218, 383) 
found.brks &lt;- get.brks(N.brk=3,n.ped=1000, snp.all2, tar.snp,rcmb.rate=NA)
breaks &lt;- found.brks[[1]]
family.pos &lt;- found.brks[[2]] 


</code></pre>

<hr>
<h2 id='get.target.geno'>Getting genotypes of the target SNPs</h2><span id='topic+get.target.geno'></span>

<h3>Description</h3>

<p>This function read out the genotypes of the selected target SNPs from the orignal data set (the data set on
which simulation is based).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.target.geno(input.plink.file, target.snp, snp.all2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.target.geno_+3A_input.plink.file">input.plink.file</code></td>
<td>
<p>is a vector of three character strings for the file names of the mother's
father's and child's plink base filenames with the necessary path to the directory. The plink files are in bed format and three files with 
extensions .bed .bim and .fam are expected for each individual's genotypes. The mothers, fathers, and 
childredn must be from the same set of trio families even though the ordering of the families can be different
for the three sets of data.</p>
</td></tr>
<tr><td><code id="get.target.geno_+3A_target.snp">target.snp</code></td>
<td>
<p>is a vector of integers showing the row number of the target SNPs in the .bim file</p>
</td></tr>
<tr><td><code id="get.target.geno_+3A_snp.all2">snp.all2</code></td>
<td>
<p>is a dataframe containing list of SNPs in PLINK .bim format. Only the second column is used which contains
the rs number of the SNPs. The colname name of the second column has to be &quot;V2&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of three matrices is returned. The three matrices are the observed genotypes of the mothers from family 1 to family n repeated twice,
genotypes of the fathers from family 1 to family n repeated twice and genotypes of children from family 1 to n followed by (stacking on top of) 
genotypes of the complements at the target SNPs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tar.snp &lt;- c(21, 118, 121, 140, 155, 168, 218, 383) 
m.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_mom')
f.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_dad')
k.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_kid')
# the preloaded data frame snp.all2 contains the data frame read from the corresponding .bim file.
## Not run:  
target.geno &lt;- get.target.geno(c(m.file,f.file,k.file), tar.snp,snp.all2)

## End(Not run)
</code></pre>

<hr>
<h2 id='glue.chr.segment.par'>Splicing chromosomal segments</h2><span id='topic+glue.chr.segment.par'></span>

<h3>Description</h3>

<p>This function splices the triad chromosomal segments into &quot;complete&quot; trios. The spliced trio sets are written into separate plink
files chromosome by chromosome. It is parallelized and if no no_cores value is given the ceiling of half of the total number
of CPUs available will be used in the parallelization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glue.chr.segment.par(
  input.plink.file,
  out.put.file,
  brks,
  sel.fam.all,
  snp.all2,
  pathway.all,
  target.snp,
  pop.vec = NA,
  no_cores = NA,
  flip = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glue.chr.segment.par_+3A_input.plink.file">input.plink.file</code></td>
<td>
<p>for simulations of homogenous population, it is a vector of three character strings for the base filenames of the mother's
father's and child's plink base filenames. The plink files are in bed format and in the same folder three files with 
extensions .bed .bim and .fam are expected for each individual's genotypes. The mothers, fathers, and 
childredn must be from the same set of trio families even though the ordering of the families can be different
for the three sets of data. For simulations under population stratification it is a list of two vectors. Each vector is a vector 
of three character strings for the base filenames as described above.The two vectors correspond to the two subpopulations.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_out.put.file">out.put.file</code></td>
<td>
<p>is a character string giving the base file name for the output file. Genotypes on different chromosomes are output to different files.
The final file name also contains information on chromosome number. E.g., for a base filename &quot;trio&quot; and for chromosome 1 the final file name is &quot;trio1sim&quot;.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_brks">brks</code></td>
<td>
<p>is a matrix of integers showing where the chromosomal breaks is to take place for each individual in the simulated trios.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_sel.fam.all">sel.fam.all</code></td>
<td>
<p>is a matrix of integer giving the families (in terms of row number) selected for each chromosomal segment and each simulated trio.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_snp.all2">snp.all2</code></td>
<td>
<p>is a dataframe containing the list of SNPs in PLINK .bim format. Two columns of the dataframe
is used: column 1 with column name &quot;V1&quot; containing the chromosome number and column 2 with column name &quot;V2&quot; containing the rs number of the SNPs.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_pathway.all">pathway.all</code></td>
<td>
<p>is a matrix giving the genotypes on the pathway SNPs in the simulated trio.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_target.snp">target.snp</code></td>
<td>
<p>is a vector of integers showing the row number of the target SNPs in the .bim file.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_pop.vec">pop.vec</code></td>
<td>
<p>is a vector of 1's and 2's giving the subpopulation group of each simulated trio. This parameter is relevant only for stratified scenarios.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_no_cores">no_cores</code></td>
<td>
<p>is an integer which specifies the number of CPU cores to be parallelized.</p>
</td></tr>
<tr><td><code id="glue.chr.segment.par_+3A_flip">flip</code></td>
<td>
<p>is a boolean indicating whether the mother's and the father's genotypes will be swapped to wipe out potential maternal effects in the orignal data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function does not return values. Instead it writes PLINK files into the designated directory. Each set of PLINK files contains genotype
data for one chromosome for all trios. The first one third of the rows are genotypes of the mothers'. The second one third are those of the fathers' and the last one 
third are the children's.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tar.snp &lt;- c(21, 118, 121, 140, 155, 168, 218, 383) 
found.brks &lt;- get.brks(N.brk=3,n.ped=1000, snp.all2, tar.snp,rcmb.rate=NA)
breaks &lt;- found.brks[[1]]
family.position &lt;- found.brks[[2]] 
betas &lt;- c(-6.4, 3.2, 5.8)
pwy &lt;- list(1:4,5:8)
m.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_mom')
f.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_dad')
k.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_kid')
# the preloaded data frame snp.all2 contains the data frame read from the corresponding .bim file.
target.geno &lt;- get.target.geno(c(m.file,f.file,k.file), tar.snp,snp.all2)
mom.target &lt;- target.geno[[1]]
dad.target &lt;- target.geno[[2]]
kid.target &lt;- target.geno[[3]]
 ## Not run: 
fitted.model &lt;- fit.risk.model.par(n.ped=1000,brks=breaks,target.snp=tar.snp, 
fam.pos=family.position,mom.tar=mom.target,dad.tar=dad.target, kid.tar=kid.target,  
pathways=pwy,betas, e.fr=NA, betas,pop1.frac= NA,rate.beta=NA,no_cores=2)
sel.fam &lt;- fitted.model[[1]]
sim.pathway.geno &lt;-  fitted.model[[2]]
glue.chr.segment.par(c(m.file,f.file,k.file),file.path(tempdir(),'trio'), breaks,sel.fam,
                     snp.all2,sim.pathway.geno,target.snp,pop.vec=NA,no_cores=1,flip=TRUE) 

## End(Not run)

</code></pre>

<hr>
<h2 id='pick_target.snp'>Pick target SNPs in the pathway</h2><span id='topic+pick_target.snp'></span>

<h3>Description</h3>

<p>The target SNPs in the pathway can be picked by users manually or use this facility function.
It helps pick the set of target SNPs in the pathway(s) based on a 
desired allele frequency. If picked manually, the target SNPs should be in the order from the smallest to the largest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pick_target.snp(input.plink.file, fr.desire = "double", n.snp = "integer")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pick_target.snp_+3A_input.plink.file">input.plink.file</code></td>
<td>
<p>is a vector of two character strings for the file names of the mother's
and  father's plink base filenames with the necessary path to the directory. The plink files are in bed format and three files with 
extensions .bed .bim and .fam are expected for each parent's genotypes. In addition the allele 
frequnecy files generated by PLINK (base filename with .frq extension) are expected to be in the
same directory as the .bed file.</p>
</td></tr>
<tr><td><code id="pick_target.snp_+3A_fr.desire">fr.desire</code></td>
<td>
<p>is a double number giving the desired frequency of the target SNPs.</p>
</td></tr>
<tr><td><code id="pick_target.snp_+3A_n.snp">n.snp</code></td>
<td>
<p>is an integer giving the number of target SNPs to be picked.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list of two: first element is  the SNPs read from the .bim file now with 
allele frequncies merged and the second is the row numbers of the target SNPs selected
among all SNPs in the .bim file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_mom')
f.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_dad')
picked.target &lt;- pick_target.snp(c(m.file,f.file),0.05, 8)
cat('Target SNPs picked:',picked.target[[2]],'\n')
</code></pre>

<hr>
<h2 id='rcmb.rate'>An example recombination rate dataset</h2><span id='topic+rcmb.rate'></span>

<h3>Description</h3>

<p>This dataset is an example dataset of recombination rates between SNPs
The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rcmb.rate
</code></pre>


<h3>Format</h3>

<p>A data frame with 478 rows and 6 variables
</p>

<dl>
<dt>CHR</dt><dd><p>Chromosome</p>
</dd>
<dt>RS</dt><dd><p>SNP rs number</p>
</dd>
<dt>POS</dt><dd><p>Chromosomal position</p>
</dd>
<dt>RATE</dt><dd><p>recombination rate between the SNP and the SNP above</p>
</dd>
</dl>


<hr>
<h2 id='snp.all2'>SNPs in the PLINK files</h2><span id='topic+snp.all2'></span>

<h3>Description</h3>

<p>A dataset containing the list of variables in the PLINK files
The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snp.all2
</code></pre>


<h3>Format</h3>

<p>A data frame with 10279 rows and 12 variables:
</p>

<dl>
<dt>ord</dt><dd><p>Ordering of the SNPs</p>
</dd>
<dt>RS</dt><dd><p>SNP rs number</p>
</dd>
<dt>CHR</dt><dd><p>Chromosome</p>
</dd>
<dt>POS</dt><dd><p>Chromosomal position</p>
</dd>
<dt>A1</dt><dd><p>A1 allele</p>
</dd>
<dt>A2</dt><dd><p>A2 allele</p>
</dd>
<dt>MAF</dt><dd><p>Minor allele frequency</p>
</dd>
<dt>NCHROBS</dt><dd><p>Number of observed chromosomes used in MAF calculation</p>
</dd>
</dl>


<hr>
<h2 id='TriadSim'>Simulation main function</h2><span id='topic+TriadSim'></span>

<h3>Description</h3>

<p>TriadSim can simulate genotypes for case-parent triads, case-control, and quantitative trait samples with realistic linkage
diequilibrium structure and allele frequency distribution. For studies of epistasis one can simulate models that involve 
specific SNPs at specific sets of loci, which we will refer to as &quot;pathways&quot;. TriadSim generates genotype data by resampling 
triad genotypes from existing data. It takes genotypes in PLINK format as the input files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TriadSim(
  input.plink.file,
  out.put.file,
  fr.desire,
  pathways,
  n.ped,
  N.brk,
  target.snp = NA,
  P0,
  is.OR,
  risk.exposure,
  risk.pathway.unexposed,
  risk.pathway.exposed,
  is.case = TRUE,
  e.fr = NA,
  pop1.frac = NA,
  P0.ratio = 1,
  rcmb.rate = NA,
  no_cores = NA,
  qtl = FALSE,
  same.brk = FALSE,
  flip = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TriadSim_+3A_input.plink.file">input.plink.file</code></td>
<td>
<p>gives the filenames (as well as the path) of the source data used for resampling. The input files are in PLINK format.
For simulations of a homogenous population, it is a vector of three character strings for the base filenames of the mother's
father's and child's PLINK files. The PLINK files are in bed format and three files with 
extensions .bed .bim and .fam are expected for each individual's genotypes. The mothers, fathers, and 
children must be from the same set of triad families even though the ordering of the families can be different
for the three sets of data. For simulations under population stratification it is a list of two vectors. Each vector is a vector 
of three character strings giving the base filenames for the PLINK files as described above.The two vectors correspond to the 
two subpopulations.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_out.put.file">out.put.file</code></td>
<td>
<p>is a character string giving the pathway to and the base filename of the output file. The names of the final output files 
also contain information on chromosome number. E.g., for a base filename &quot;trio&quot; and for chromosome 1 the final filenames for the PLILK files
are &quot;trio1.bim&quot;,&quot;trio1.bed&quot; and &quot;trio1.fam&quot;.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_fr.desire">fr.desire</code></td>
<td>
<p>is a double number giving the desired frequency of the target SNPs.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_pathways">pathways</code></td>
<td>
<p>is a list of vectors of integers. Each vector of integers denotes the SNPs involved in a particular pathway. E.g. list(1:4,5:8)</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_n.ped">n.ped</code></td>
<td>
<p>is an integer giving the number of trios to be simulated</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_n.brk">N.brk</code></td>
<td>
<p>is an integer giving the number of breaks to be picked for each chromosome.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_target.snp">target.snp</code></td>
<td>
<p>is a vector of integers showing the row number of the target SNPs in the .bim file.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_p0">P0</code></td>
<td>
<p>gives the baseline disease prevalence in the unexposed individuals with 0 copies of the risk pathways.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_is.or">is.OR</code></td>
<td>
<p>is a boolean varialbe denoting wether the input risk parameters are odds ratios. It is TRUE when the input risks are odds ratios.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_risk.exposure">risk.exposure</code></td>
<td>
<p>is a double giving the relative risk (or odds ratio, if is.OR=TRUE) of the exposure main effect.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_risk.pathway.unexposed">risk.pathway.unexposed</code></td>
<td>
<p>is a vector of doubles giving the relative risk (or odds ratio, if is.OR=TRUE) of each risk pathways in the unexposed individuals 
with the risk of unexposed individuals who carry no copies of the pathways as a reference.For scenarios that do not involve exposure the value 
of this vector is for all individuals.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_risk.pathway.exposed">risk.pathway.exposed</code></td>
<td>
<p>is a vector of doubles giving the relative risk (or odds ratio, if is.OR=TRUE) of each risk pathways in the exposed individuals. 
with the risk of exposed individuals who carry no copies of the pathways as a reference. For scenarios that do not involve exposure the value 
of this vector is not used.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_is.case">is.case</code></td>
<td>
<p>is a boolean variable. When is.case = TRUE case-parents trios will be simulated.Otherwise, control-parents trios will be simulated.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_e.fr">e.fr</code></td>
<td>
<p>is a double number between 0 and 1 which gives the exposure prevalence.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_pop1.frac">pop1.frac</code></td>
<td>
<p>is a double number between 0 and 1 which gives the fraction of population 1 for a population stratification scenario.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_p0.ratio">P0.ratio</code></td>
<td>
<p>gives the ratio of the baseline disease prevalence in the second subpopulation to that of the first subpopulation.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_rcmb.rate">rcmb.rate</code></td>
<td>
<p>the default value is NA. rcmb.rate is a dataframe containing the recombination rates at each SNP. The ordering of the SNPs (in rows) should 
be identical to that of snp.all2. It has 4 columns with the column names 'CHR','RS','POS', and 'RATE' representing &quot;the chromosomal number&quot;, 
&quot;SNP rs number&quot;, &quot;chromosomal position&quot;, and &quot;recombination rate&quot;, respectively. The recombination rate represents the maximum recombination 
rate in the chromosomal region between the current SNP and the SNP above (or the first basepair of the chromosome for the first SNP on a 
chromosome).When no rcmb.rate is provided the function will pick the breaking points randomly.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_no_cores">no_cores</code></td>
<td>
<p>is an integer which specifies the number of CPU cores to parallelized.contain values</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_qtl">qtl</code></td>
<td>
<p>is a boolean variable denoting whether a quantitative trait (qtl=TRUE) or a binary trait (qtl=FALSE) is to be simulated. For a binary trait only affected 
families will be kept. The default value is qtl=FALSE.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_same.brk">same.brk</code></td>
<td>
<p>is an indicator variable to denote whether the same set of breaking points will be used for all simulated triads. The default value is FALSE.</p>
</td></tr>
<tr><td><code id="TriadSim_+3A_flip">flip</code></td>
<td>
<p>is an indicator variable denoting whether the mother's and the father's genotypes will be swapped to wipe out potential maternal 
effects in the orignal data. The default value is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>this function simulates genotypes of parent-offspring triads and writes PLINK files into the designated directory. Genotypes on each
chromosome will be written into a separate set of PLINK files. In each set of PLINK files genotypes of the mothers, fathers, and children
are stacked on top of each other. The first third of the rows are genotypes of the mothers'.The seond third are those of the fathers' and 
the last third are those of the children's. 
The following files are also generated under specific scenarios:
a file with name ending with &quot;exp.txt&quot; containing the exposure data when exposure is involved in the risk model.
a file with name ending with &quot;pop.txt&quot; containing information on subpopulation membership when the simulation involves a stratified scenario.
a file with name ending with &quot;pheno.tx&quot; containing quantitative trait phenotype when a quantitative trait is involved.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_mom')
f.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_dad')
k.file &lt;- file.path(system.file(package = "TriadSim"),'extdata/pop1_4chr_kid')
input.plink.file &lt;- c(m.file, f.file, k.file)
## Not run: TriadSim(input.plink.file, file.path(tempdir(),'triad'), fr.desire=0.05,pathways=list(1:4,5:8),
       n.ped=1000, N.brk=3, target.snp=NA,P0=0.001,is.OR=FALSE,risk.exposure= 1,
       risk.pathway.unexposed=c(1.5, 2), risk.pathway.exposed=c(1.5, 2), is.case=TRUE, e.fr=NA, 
       pop1.frac=NA, P0.ratio=1, rcmb.rate, no_cores=1)
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
