<!DOCTYPE html><html><head><title>Help for package tripsAndDipR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {tripsAndDipR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#tripsAndDip'><p>Uses read counts for biallelic SNPs to determine if a sample is diploid or triploid</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identification of 2n and 3n Samples from Amplicon Sequencing
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Uses read counts for biallelic single nucleotide polymorphisms (SNPs)
    to compare the likelihoods for the observed read counts given that a sample is 
    either diploid or triploid. It allows parameters to be specified to account for 
    sequencing error rates and allelic bias. For details of the algorithm, please see
    Delomas (2019) &lt;<a href="https://doi.org/10.1111%2F1755-0998.13073">doi:10.1111/1755-0998.13073</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/delomast/tripsAndDipR">https://github.com/delomast/tripsAndDipR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/delomast/tripsAndDipR/issues">https://github.com/delomast/tripsAndDipR/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-08-27 15:19:38 UTC; tdelomas</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Delomas [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Delomas &lt;thomas.delomas@idfg.idaho.gov&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-08-28 09:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='tripsAndDip'>Uses read counts for biallelic SNPs to determine if a sample is diploid or triploid</h2><span id='topic+tripsAndDip'></span>

<h3>Description</h3>

<p><code>tripsAndDip</code> calculates log-likelihood ratios comparing whether a sample is likely
diploid or triploid based on the read counts for biallelic SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tripsAndDip(counts, counts_alt = NA, h, eps, min_reads = 30,
  min_loci = 15, binom_p_value = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tripsAndDip_+3A_counts">counts</code></td>
<td>
<p>Either a numeric matrix or a dataframe with each row corresponding to a different sample.
There are two options for formatting the input. Either
the columns correspond to the read counts for each locus, in a two column per locus format:
column 1 is the read counts for locus1ReferenceAllele, column two is the read counts for locus1AlternateAllele2, locus2Reference, locus2Alternate, ...
OR this contains read counts for the reference allele, and <code>counts_alt</code> contains read counts for the alternate allele
The rownames should be the sample names.</p>
</td></tr>
<tr><td><code id="tripsAndDip_+3A_counts_alt">counts_alt</code></td>
<td>
<p>This is a numeric matrix or a dataframe with each row corresponding to a different sample.
The matrix contains counts for the alternate allele, with samples and loci having the same order as in <code>counts</code>
If this parameter is NA or NULL, <code>counts</code> is assumed to have both the reference and alternate allele counts.</p>
</td></tr>
<tr><td><code id="tripsAndDip_+3A_h">h</code></td>
<td>
<p>A numeric vector of h values for each locus in the same order that the loci are ordered in counts.
These h values are as defined by Gerard et al. (2018) &quot;Genotyping polyploids from messy sequencing data&quot; Genetics 210:789-807.
with h expressed as alternate / reference. These values can be estimated using the R package &quot;updog&quot;.</p>
</td></tr>
<tr><td><code id="tripsAndDip_+3A_eps">eps</code></td>
<td>
<p>A numeric vector of values for the error rate per read for each locus in the same order that the loci are ordered in counts.
These are expressed as proportions, so a rate of 1% should be given as 0.01. These values can be estimated using the R package &quot;updog&quot;.</p>
</td></tr>
<tr><td><code id="tripsAndDip_+3A_min_reads">min_reads</code></td>
<td>
<p>The minimum number of reads to consider a locus.</p>
</td></tr>
<tr><td><code id="tripsAndDip_+3A_min_loci">min_loci</code></td>
<td>
<p>The minimum number of usable loci in a sample to calculate a log-likelihood ratio.</p>
</td></tr>
<tr><td><code id="tripsAndDip_+3A_binom_p_value">binom_p_value</code></td>
<td>
<p>The alpha value to use when applying a binomial test to determine
whether to include a locus in the calculation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>tripsAndDip</code> calculates log-likelihood ratios comparing the likelihoods of the read counts
under diploidy or triploidy for a sample using biallelic SNPs.This function was designed
with amplicon sequencing data in mind, but may be useful for other genotyping techniques
that also yield read counts for each allele in a given locus. Full details of the calculations
can be found in Delomas (2019) Differentiating diploid and triploid individuals using single
nucleotide polymorphisms genotyped by amplicon-sequencing. Molecular Ecology Resources.
</p>


<h3>Value</h3>

<p>a dataframe with column 1 containing sample names, column 2 containing calculated LLRs (larger means more likely given triploidy)
and column 3 containing the number of loci used to calculate the LLR
</p>


<h3>Examples</h3>

<pre><code class='language-R'># make up some data
triploid_allele1 &lt;- rbinom(60, 75, 2/3)
triploid_allele2 &lt;- 75 - triploid_allele1
diploid_allele1 &lt;- rbinom(60, 75, 1/2)
diploid_allele2 &lt;- 75 - diploid_allele1
# interleave allele counts
triploid &lt;- c(rbind(triploid_allele1, triploid_allele2))
diploid &lt;- c(rbind(diploid_allele1, diploid_allele2))

# create counts matrix
allele_counts &lt;- matrix(data = c(triploid, diploid), byrow = TRUE, nrow = 2, ncol = 120)
rownames(allele_counts) &lt;- c("triploid", "diploid")

#create h and eps vectors
h_constant &lt;- rep(1, 60)
eps_constant &lt;- rep(.01, 60)

#run function
ploidy &lt;- tripsAndDip(allele_counts, h = h_constant, eps = eps_constant)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
