<!DOCTYPE html><html lang="en"><head><title>Help for package TrumpetPlots</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TrumpetPlots}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#plot_trumpets'><p>Trumpets</p></a></li>
<li><a href='#powerCurves'><p>Power Curves for Trumpet Plots</p></a></li>
<li><a href='#toy_data'><p>Toy dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Visualization of Genetic Association Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Visualizes the relationship between allele frequency and effect size in genetic association studies. The input is a data frame containing association results. The output is a plot with the effect size of risk variants in the Y axis, and the allele frequency spectrum in the X axis. Corte et al (2023) &lt;<a href="https://doi.org/10.1101%2F2023.04.21.23288923">doi:10.1101/2023.04.21.23288923</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), data.table, ggplot2, magrittr, stats, purrr (&ge;
1.0.1)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-13 00:57:06 UTC; garcij62</td>
</tr>
<tr>
<td>Author:</td>
<td>Judit García-González
    <a href="https://orcid.org/0000-0001-6245-740X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Lathan Liou [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Judit García-González &lt;judit.garciagonzalez@mssm.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-13 08:50:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='plot_trumpets'>Trumpets</h2><span id='topic+plot_trumpets'></span>

<h3>Description</h3>

<p>This function generates trumpet plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_trumpets(
  dataset = toy_data,
  rsID = "rsID",
  freq = "freq",
  A1_beta = "A1_beta",
  Analysis = "Analysis",
  Gene = "Gene",
  calculate_power = TRUE,
  show_power_curves = TRUE,
  exist_datapwr = NULL,
  threshold = c(0.7, 0.9),
  N = 1e+05,
  alpha = 5e-08,
  Nfreq = 500,
  power_color_palette = c("purple", "deeppink"),
  analysis_color_palette = c("#018571", "#a6611a")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_trumpets_+3A_dataset">dataset</code></td>
<td>
<p>Input text file with genetic association results. Columns required are rsID, freq, A1_beta, Analysis and Gene.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_rsid">rsID</code></td>
<td>
<p>(required) Single Nucleotide Polymorphism (SNP) name.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_freq">freq</code></td>
<td>
<p>(required) allele frequency of effect SNP.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_a1_beta">A1_beta</code></td>
<td>
<p>(required) risk allele effect size.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_analysis">Analysis</code></td>
<td>
<p>(optional) adds colour to the type of analysis (e.g. GWAS, Sequencing).</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_gene">Gene</code></td>
<td>
<p>(optional) Candidate gene name (can be empty).</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_calculate_power">calculate_power</code></td>
<td>
<p>(TRUE/FALSE) Calculate power curves. Choose TRUE to add power curves for a given threshold, alpha, sample size N and number of allele frequencies. Choose FALSE if you already ran powerCurves() outside or do not want to show power curves.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_show_power_curves">show_power_curves</code></td>
<td>
<p>(TRUE/FALSE) Show power curves in plot</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_exist_datapwr">exist_datapwr</code></td>
<td>
<p>Existing dataframe containing columns: freq, pos.b.for.f, neg.b.for.f, powerline.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_threshold">threshold</code></td>
<td>
<p>Required if power == TRUE. Can be a single number or a vector of statistical power thresholds.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_n">N</code></td>
<td>
<p>(Required if calculate_power == TRUE). Sample size used to test the association.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_alpha">alpha</code></td>
<td>
<p>(Required if calculate_power == TRUE).</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_nfreq">Nfreq</code></td>
<td>
<p>(Required if calculate_power == TRUE). Number of allele frequency data points generated to calculate the power curves. We recommend Nfreq&gt;1000 for power curves with high resolution. Note that this will slow down the rendering of the plot.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_power_color_palette">power_color_palette</code></td>
<td>
<p>A vector of colours for the power curves. Number of colors should match number of thresholds supplied.</p>
</td></tr>
<tr><td><code id="plot_trumpets_+3A_analysis_color_palette">analysis_color_palette</code></td>
<td>
<p>A vector of colours for the analysis types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a Trumpet plot with variant allele frequency (X axis, log10 scale) and effect size information (Y axis).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_trumpets(dataset = toy_data)
</code></pre>

<hr>
<h2 id='powerCurves'>Power Curves for Trumpet Plots</h2><span id='topic+powerCurves'></span>

<h3>Description</h3>

<p>This function generates curves indicating statistical power in Trumpet plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerCurves(threshold = 0.8, N = 4e+05, alpha = 5e-08, Nfreq = 500)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="powerCurves_+3A_threshold">threshold</code></td>
<td>
<p>user-specified power level</p>
</td></tr>
<tr><td><code id="powerCurves_+3A_n">N</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="powerCurves_+3A_alpha">alpha</code></td>
<td>
<p>significance threshold</p>
</td></tr>
<tr><td><code id="powerCurves_+3A_nfreq">Nfreq</code></td>
<td>
<p>Number of allele frequency data points generated to calculate the power curves</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the power estimated for each allele frequency and effect size, given a: Statistical power threshold, significance threshold (alpha value), and sample size
</p>


<h3>Examples</h3>

<pre><code class='language-R'>powerCurves(threshold = 0.8, N=400000, alpha = 5e-8)
</code></pre>

<hr>
<h2 id='toy_data'>Toy dataset</h2><span id='topic+toy_data'></span>

<h3>Description</h3>

<p>A data frame with 9999 genetic associations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(toy_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 9999 genetic associations
</p>


<h3>Details</h3>


<ul>
<li><p> rsID. SNP name
</p>
</li>
<li><p> freq. allele frequency of effect SNP
</p>
</li>
<li><p> A1_beta. effect size
</p>
</li>
<li><p> Analysis. adds colour to the type of analysis (e.g. GWAS, Sequencing)
</p>
</li>
<li><p> Gene. Candidate gene name
</p>
</li>
<li><p> N.
</p>
</li>
<li><p> trait. ToyDataPheno
</p>
</li></ul>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
