<!DOCTYPE html><html lang="en"><head><title>Help for package TSDFGS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TSDFGS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TSDFGS-package'><p>TSDFGS: Training Set Determination for Genomic Selection</p></a></li>
<li><a href='#cd_score'><p>CD-score</p></a></li>
<li><a href='#FGCM'><p>Fit logistic growth curve model</p></a></li>
<li><a href='#geno'><p>Genotype information</p></a></li>
<li><a href='#nt2r'><p>Simulate r-scores of each training set size</p></a></li>
<li><a href='#optTrain'><p>Optimal training set determination</p></a></li>
<li><a href='#pev_score'><p>PEV score</p></a></li>
<li><a href='#r_score'><p>r-score</p></a></li>
<li><a href='#SSDFGS'><p>Sample size determination for genomic selection</p></a></li>
<li><a href='#subpop'><p>Sub-population information</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Training Set Determination for Genomic Selection</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-07</td>
</tr>
<tr>
<td>Description:</td>
<td>We propose an optimality criterion to determine the required training set, r-score, which is derived directly from Pearson's correlation between the genomic estimated breeding values and phenotypic values of the test set &lt;<a href="https://doi.org/10.1007%2Fs00122-019-03387-0">doi:10.1007/s00122-019-03387-0</a>&gt;. This package provides two main functions to determine a good training set and its size.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggplot2, latex2exp, lifecycle, parallel, Rcpp (&ge;
1.0.8.3)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/oumarkme/TSDFGS">https://github.com/oumarkme/TSDFGS</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/oumarkme/TSDFGS/issues">https://github.com/oumarkme/TSDFGS/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-07 13:21:24 UTC; mark</td>
</tr>
<tr>
<td>Author:</td>
<td>Jen-Hsiang Ou <a href="https://orcid.org/0000-0001-9305-2931"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Po-Ya Wu <a href="https://orcid.org/0000-0002-7342-2867"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Chen-Tuo Liao <a href="https://orcid.org/0000-0001-9777-3701"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jen-Hsiang Ou &lt;jen-hsiang.ou@imbim.uu.se&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-07 14:00:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='TSDFGS-package'>TSDFGS: Training Set Determination for Genomic Selection</h2><span id='topic+TSDFGS'></span><span id='topic+TSDFGS-package'></span>

<h3>Description</h3>

<p>We propose an optimality criterion to determine the required training set, r-score, which is derived directly from Pearson's correlation between the genomic estimated breeding values and phenotypic values of the test set <a href="https://doi.org/10.1007/s00122-019-03387-0">doi:10.1007/s00122-019-03387-0</a>. This package provides two main functions to determine a good training set and its size.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Jen-Hsiang Ou <a href="mailto:jen-hsiang.ou@imbim.uu.se">jen-hsiang.ou@imbim.uu.se</a> (<a href="https://orcid.org/0000-0001-9305-2931">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Po-Ya Wu <a href="mailto:Po-Ya.Wu@hhu.de">Po-Ya.Wu@hhu.de</a> (<a href="https://orcid.org/0000-0002-7342-2867">ORCID</a>)
</p>
</li>
<li><p> Chen-Tuo Liao <a href="mailto:ctliao@ntu.edu.tw">ctliao@ntu.edu.tw</a> (<a href="https://orcid.org/0000-0001-9777-3701">ORCID</a>) [thesis advisor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/oumarkme/TSDFGS">https://github.com/oumarkme/TSDFGS</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/oumarkme/TSDFGS/issues">https://github.com/oumarkme/TSDFGS/issues</a>
</p>
</li></ul>


<hr>
<h2 id='cd_score'>CD-score</h2><span id='topic+cd_score'></span>

<h3>Description</h3>

<p>This function calculate CD-score <a href="doi:10.1186/1297-9686-28-4-359">doi:10.1186/1297-9686-28-4-359</a> by given training set and test set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cd_score(X, X0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cd_score_+3A_x">X</code></td>
<td>
<p>A numeric matrix. The training set genotypic information matrix can be given as genotype matrix (coded as -1, 0, 1) or principle component matrix (row: sample; column: marker).</p>
</td></tr>
<tr><td><code id="cd_score_+3A_x0">X0</code></td>
<td>
<p>A numeric mareix. The test set genotypic information matrix can be given as genotype matrix (coded as -1, 0, 1) or principle component matrix (row: sample; column: marker).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A floating-point number, CD score.
</p>


<h3>Author(s)</h3>

<p>Jen-Hsiang Ou
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
## Not run: cd_score(geno[1:50, ], geno[51:100])

</code></pre>

<hr>
<h2 id='FGCM'>Fit logistic growth curve model</h2><span id='topic+FGCM'></span>

<h3>Description</h3>

<p>A function for fitting logisti growth model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FGCM(geno, nt = NULL, n_iter = NULL, multi.threads = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FGCM_+3A_geno">geno</code></td>
<td>
<p>Genotype information saved as a dataframe. Columns represent variants (SNPs or PCs).</p>
</td></tr>
<tr><td><code id="FGCM_+3A_nt">nt</code></td>
<td>
<p>A numerical vector of training set sample size for estimating logistic growth curve parameters</p>
</td></tr>
<tr><td><code id="FGCM_+3A_n_iter">n_iter</code></td>
<td>
<p>Number of simulation of each training set size. Automatically gave a suitable number by default.</p>
</td></tr>
<tr><td><code id="FGCM_+3A_multi.threads">multi.threads</code></td>
<td>
<p>Default: TRUE. Set as FALSE if you just want to run it by single thread.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimation of parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
## Not run: FGCM(geno)
</code></pre>

<hr>
<h2 id='geno'>Genotype information</h2><span id='topic+geno'></span>

<h3>Description</h3>

<p>A PCA matrix of rice genotype information. This data was published by Zhao et al. (2011) <a href="doi:10.1038/ncomms1467">doi:10.1038/ncomms1467</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geno
</code></pre>


<h3>Format</h3>

<p>A numeric matrix (PCA) with 404 rows (sample) and 404 columns (PCs).
</p>


<h3>Source</h3>

<p><a href="http://www.ricediversity.org/data/">http://www.ricediversity.org/data/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
</code></pre>

<hr>
<h2 id='nt2r'>Simulate r-scores of each training set size</h2><span id='topic+nt2r'></span>

<h3>Description</h3>

<p>Calculate r-scores (un-target) by in parallel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nt2r(geno, nt, n_iter = 30, multi.threads = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nt2r_+3A_geno">geno</code></td>
<td>
<p>A numeric dataframe of genotype, column represent sites (genotype coding as 1, 0, -1)</p>
</td></tr>
<tr><td><code id="nt2r_+3A_nt">nt</code></td>
<td>
<p>Numeric. Number of training set size</p>
</td></tr>
<tr><td><code id="nt2r_+3A_n_iter">n_iter</code></td>
<td>
<p>Times of iteration. (default = 30)</p>
</td></tr>
<tr><td><code id="nt2r_+3A_multi.threads">multi.threads</code></td>
<td>
<p>Default: TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of r-scores of each iteration
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
## Not run: nt2r(geno, 50)
</code></pre>

<hr>
<h2 id='optTrain'>Optimal training set determination</h2><span id='topic+optTrain'></span>

<h3>Description</h3>

<p>This function is designed for determining optimal training set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optTrain(
  geno,
  cand,
  n.train,
  subpop = NULL,
  test = NULL,
  method = "rScore",
  min.iter = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="optTrain_+3A_geno">geno</code></td>
<td>
<p>A numeric matrix of principal components (rows: individuals; columns: PCs).</p>
</td></tr>
<tr><td><code id="optTrain_+3A_cand">cand</code></td>
<td>
<p>An integer vector of which rows of individuals are candidates of the training set in the geno matrix.</p>
</td></tr>
<tr><td><code id="optTrain_+3A_n.train">n.train</code></td>
<td>
<p>The size of the target training set. This could be determined with the help of the ssdfgp function provided in this package.</p>
</td></tr>
<tr><td><code id="optTrain_+3A_subpop">subpop</code></td>
<td>
<p>A character vector of sub-population's group name. The algorithm will ignore the population structure if it remains NULL.</p>
</td></tr>
<tr><td><code id="optTrain_+3A_test">test</code></td>
<td>
<p>An integer vector of which rows of individuals are in the test set in the geno matrix. The algorithm will use an un-target method if it remains NULL.</p>
</td></tr>
<tr><td><code id="optTrain_+3A_method">method</code></td>
<td>
<p>Choices are rScore, PEV and CD. rScore will be used by default.</p>
</td></tr>
<tr><td><code id="optTrain_+3A_min.iter">min.iter</code></td>
<td>
<p>Minimum iteration of all methods can be appointed. One should always check if the algorithm is converged or not. A minimum iteration will set by considering the candidate and test set size if it remains NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function will return 3 information including OPTtrain (a vector of chosen optimal training set), TOPscore (highest scores of before iteration), and ITERscore (criteria scores of each iteration).
</p>


<h3>Author(s)</h3>

<p>Jen-Hsiang Ou
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
## Not run: optTrain(geno, cand = 1:404, n.train = 100)

</code></pre>

<hr>
<h2 id='pev_score'>PEV score</h2><span id='topic+pev_score'></span>

<h3>Description</h3>

<p>This function calculate prediction error variance (PEV) score <a href="doi:10.1186/s12711-015-0116-6">doi:10.1186/s12711-015-0116-6</a> by given training set and test set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pev_score(X, X0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pev_score_+3A_x">X</code></td>
<td>
<p>A numeric matrix. The training set genotypic information matrix can be given as genotype matrix (coded as -1, 0, 1) or principle component matrix (row: sample; column: marker).</p>
</td></tr>
<tr><td><code id="pev_score_+3A_x0">X0</code></td>
<td>
<p>A numeric mareix. The test set genotypic information matrix can be given as genotype matrix (coded as -1, 0, 1) or principle component matrix (row: sample; column: marker).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A floating-point number, PEV score.
</p>


<h3>Author(s)</h3>

<p>Jen-Hsiang Ou
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
## Not run: pev_score(geno[1:50, ], geno[51:100])

</code></pre>

<hr>
<h2 id='r_score'>r-score</h2><span id='topic+r_score'></span>

<h3>Description</h3>

<p>This function calculate r-score <a href="doi:10.1007/s00122-019-03387-0">doi:10.1007/s00122-019-03387-0</a> by given training set and test set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r_score(X, X0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="r_score_+3A_x">X</code></td>
<td>
<p>A numeric matrix. The training set genotypic information matrix can be given as genotype matrix (coded as -1, 0, 1) or principle component matrix (row: sample; column: marker).</p>
</td></tr>
<tr><td><code id="r_score_+3A_x0">X0</code></td>
<td>
<p>A numeric mareix. The test set genotypic information matrix can be given as genotype matrix (coded as -1, 0, 1) or principle component matrix (row: sample; column: marker).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A floating-point number, r-score.
</p>


<h3>Author(s)</h3>

<p>Jen-Hsiang Ou
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
## Not run: r_score(geno[1:50, ], geno[51:100])

</code></pre>

<hr>
<h2 id='SSDFGS'>Sample size determination for genomic selection</h2><span id='topic+SSDFGS'></span>

<h3>Description</h3>

<p>This function is designed to generate an operating curve for sample size determination
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SSDFGS(geno, nt = NULL, n_iter = NULL, multi.threads = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SSDFGS_+3A_geno">geno</code></td>
<td>
<p>A numeric data frame carried genotype information (column: PCs, row: sample)</p>
</td></tr>
<tr><td><code id="SSDFGS_+3A_nt">nt</code></td>
<td>
<p>A numeric vector carried training set sizes for r-score simulation.</p>
</td></tr>
<tr><td><code id="SSDFGS_+3A_n_iter">n_iter</code></td>
<td>
<p>Number of iterations for estimating parameters.</p>
</td></tr>
<tr><td><code id="SSDFGS_+3A_multi.threads">multi.threads</code></td>
<td>
<p>Default (multi.threads = TRUE) use 75% of threads if the computer has more than 4 threads.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An operating curve and its information.
</p>


<h3>Author(s)</h3>

<p>Jen-Hsiang Ou &amp; Po-Ya Wu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
## Not run: SSDFGS(geno)

</code></pre>

<hr>
<h2 id='subpop'>Sub-population information</h2><span id='topic+subpop'></span>

<h3>Description</h3>

<p>Sub-population information of samples. This data was published by Zhao et al. (2011) <a href="doi:10.1038/ncomms1467">doi:10.1038/ncomms1467</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subpop
</code></pre>


<h3>Format</h3>

<p>A character vector.
</p>


<h3>Source</h3>

<p><a href="http://www.ricediversity.org/data/">http://www.ricediversity.org/data/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(subpop)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
