<!DOCTYPE html><html><head><title>Help for package TSGS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TSGS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#featureSelect'>
<p>Trait specific gene selection using SVM and GA</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Trait Specific Gene Selection using SVM and GA</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Obtaining relevant set of trait specific genes from gene expression data is important for clinical diagnosis of disease and discovery of disease mechanisms in plants and animals. This process involves identification of relevant genes and removal of redundant genes as much as possible from a whole gene set. This package returns the trait specific gene set from the high dimensional RNA-seq count data by applying combination of two conventional machine learning algorithms, support vector machine (SVM) and genetic algorithm (GA). GA is used to control and optimize the subset of genes sent to the SVM for classification and evaluation. Genetic algorithm uses repeated learning steps and cross validation over number of possible solution and selects the best. The algorithm selects the set of genes based on a fitness function that is obtained via support vector machines. Using SVM as the classifier performance and the genetic algorithm for feature selection, a set of trait specific gene set is obtained.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>caret, edgeR, fastmatch, genalg, kernlab, e1071</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/SudhirSrivastava/TSGS">https://github.com/SudhirSrivastava/TSGS</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/SudhirSrivastava/TSGS/issues">https://github.com/SudhirSrivastava/TSGS/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1.9001</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-06 11:18:58 UTC; hp</td>
</tr>
<tr>
<td>Author:</td>
<td>Md. Samir Farooqi [aut],
  K.K. Chaturvedi [aut],
  D.C. Mishra [aut],
  Sudhir Srivastava [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sudhir Srivastava &lt;Sudhir.Srivastava@icar.gov.in&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-08 09:40:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='featureSelect'>
Trait specific gene selection using SVM and GA
</h2><span id='topic+featureSelect'></span>

<h3>Description</h3>

<p>This function gives the optimal set of informative genes based on RNA-Seq count data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>featureSelect(X, y, p = 5, n.iter = 1, alpha = 0.05, p.adj.method = "bonferroni")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="featureSelect_+3A_x">X</code></td>
<td>
<p>X is a G x N data frame of gene expression values (raw count data) where rows represent genes and columns represent samples. Each cell entry represents the read counts of of a gene in a sample (row names of X as gene names or gene ids)</p>
</td></tr>
<tr><td><code id="featureSelect_+3A_y">y</code></td>
<td>
<p>y is a N x 1 numeric vector with entries 0 or 1 representing sample labels, where, 0/1 represents the sample label of samples for two conditions, e.g., 0 for Control and 1 for Case</p>
</td></tr>
<tr><td><code id="featureSelect_+3A_p">p</code></td>
<td>
<p>Population size, by default 5</p>
</td></tr>
<tr><td><code id="featureSelect_+3A_n.iter">n.iter</code></td>
<td>
<p>The number of iterations, by default 1</p>
</td></tr>
<tr><td><code id="featureSelect_+3A_alpha">alpha</code></td>
<td>
<p>The level of significance, by default 0.05</p>
</td></tr>
<tr><td><code id="featureSelect_+3A_p.adj.method">p.adj.method</code></td>
<td>
<p>Method of adjusting p-values, by default &quot;bonferroni&quot;. The other methods available are &quot;BH&quot;, &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;BY&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>InformativeGenes</code></td>
<td>
<p>List of informative genes selected</p>
</td></tr>
<tr><td><code>LogCPM</code></td>
<td>
<p>Log cpm data of informative genes</p>
</td></tr>
<tr><td><code>DEA_Result</code></td>
<td>
<p>Differential Expression Analysis Result of informative genes</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>c(person(&quot;Md. Samir&quot;, &quot;Farooqi&quot;, email = &quot;ms.Farooqi@icar.gov.in&quot;, role = &quot;aut&quot;),
person(&quot;K.K.&quot;, &quot;Chaturvedi&quot;, email = &quot;kk.Chaturvedi@icar.gov.in&quot;, role = &quot;aut&quot;),
person(&quot;D.C.&quot;, &quot;Mishra&quot;, email = &quot;Dwijesh.Mishra@icar.gov.in&quot;, role = &quot;aut&quot;),
person(&quot;Sudhir&quot;, &quot;Srivastava&quot;, email = &quot;Sudhir.Srivastava@icar.gov.in&quot;, role = c(&quot;cre&quot;,&quot;aut&quot;)))
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  filename &lt;- system.file("extdata", "exampleData.csv", package = "TSGS")
  cdata &lt;- read.csv(filename, header = TRUE, row.names = 1, stringsAsFactors = FALSE)
  X &lt;- as.data.frame(cdata[-1,])
  y &lt;- as.numeric(cdata[1,])
  set.seed(100)
  result &lt;- featureSelect(X, y, 5, 1, 0.05, "bonferroni")
  gene_list &lt;- result$InformativeGenes
  logcpm_data &lt;- result$LogCPM
  dea_result &lt;- result$DEA_Result
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
