<!DOCTYPE html><html lang="en"><head><title>Help for package TSGSIS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TSGSIS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#TSGSIS'><p>Two Stage-Grouped Sure Independence Screening</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Two Stage-Grouped Sure Independence Screening</td>
</tr>
<tr>
<td>Description:</td>
<td>To provide a high dimensional grouped variable selection approach for detection of whole-genome SNP effects and SNP-SNP interactions, as described in Fang et al. (2017, under review).</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Yao-Hwei Fang, Jie-Huei Wang, and Chao A. Hsiung</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yao-Hwei Fang &lt;yhfang@nhri.org.tw&gt;</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-04-18</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.3), glmnet, MASS, stats</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-04-18 01:08:21 UTC; lulu</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-04-18 06:43:40 UTC</td>
</tr>
</table>
<hr>
<h2 id='TSGSIS'>Two Stage-Grouped Sure Independence Screening</h2><span id='topic+TSGSIS'></span>

<h3>Description</h3>

<p>The package is a beta version that provides a high-dimensional grouped variable selection approach for detection of whole-genome SNP effects and SNP-SNP interactions, as described in Fang et al. (2017, under review). The proposed TSGSIS is developed to study interactions that may not have marginal effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TSGSIS(XA, Y, Gene_list, ntest, lambda, Method)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TSGSIS_+3A_xa">XA</code></td>
<td>
<p>The <code class="reqn">N \times P</code> matrix of XA. There are <code class="reqn">N</code> individuals and <code class="reqn">P</code> variables in matrix, with one individual in each row and one genotype in each column.</p>
</td></tr>
<tr><td><code id="TSGSIS_+3A_y">Y</code></td>
<td>
<p>The <code class="reqn">N \times 1</code> matrix of Y. It can be real number or binary outcome.</p>
</td></tr>
<tr><td><code id="TSGSIS_+3A_gene_list">Gene_list</code></td>
<td>
<p>The <code class="reqn">a \times d</code> matrix of the Gene_list. a is the maximal number of gene size in the Gene_list which other values are denoted by 0. d is the number of genes.</p>
</td></tr>
<tr><td><code id="TSGSIS_+3A_ntest">ntest</code></td>
<td>
<p>The ntest (<code class="reqn">&lt;N</code>) is the number of testing data for evaluation of MSE.</p>
</td></tr>
<tr><td><code id="TSGSIS_+3A_lambda">lambda</code></td>
<td>
<p>The lambda is the parameter of Lasso regression.</p>
</td></tr>
<tr><td><code id="TSGSIS_+3A_method">Method</code></td>
<td>
<p>&quot;Reg&quot; for quantitative trait modeling, &quot;LR&quot; for disease trait modeling.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a result of screening
</p>
<table role = "presentation">
<tr><td><code>result</code></td>
<td>
<p>First element of the result is the MSE of testing data, the rest elements are the important SNP effects and SNP-SNP interactions after TSGSIS modeling.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The missing value (NA) in the XA and Y is not allowed in this beta version.
</p>


<h3>References</h3>

<p>Yao-Hwei Fang, Jie-Huei Wang and Chao A. Hsiung (2017). TSGSIS: A High-dimensional Grouped Variable Selection Approach for Detection of Whole-genome SNP-SNP Interactions. (Under revision in Bioinformatics)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#We investigate the performance of TS-GSIS under model 1 with intra-gene correlation rho = 0.2, 
#trait dispersion sigma^2 = 1, effect size k = 3 and homogeneous MAF. 
#Given 100 SNPs with gene size d = 10, 500 unrelated individuals are simulated.
#(Please refer to the Figure 3 of the reference)

library(glmnet)
library(MASS)

set.seed(1)# Set seed
#Parameter setting
ntotal = 500
p = 100
n.pred = 10 #Gene sizes
rho = 0.2 #Intra-gene correlation in block
k = 3 #Effect size
vari = 1 #Sigma2
lambda = 0.5 #For lasso parameter
ntest = 150 #For evaluation
Method="Reg"#For quantitative trait
#Heterogeneous MAF: randomly set to 0.35, 0.2 or 0.1 with equal likelihood.
MAF = matrix(0,2,3) 
MAF[,1] = c(0.1225,0.5775)
MAF[,2] = c(0.04,0.36)
MAF[,3] = c(0.01,0.19)
#Trait Y
modelY = "k*XA[,1] - k*(sqrt(rho))*XA[,5] + k*XA[,31]*XA[,5] + rnorm(ntotal,0,vari)"

PAS1 = function(z){ g = paste("A",z,sep = "") 
	return(g)
}#Define colname fun.
norm = function(a) (a-mean(a))/sd(a) #Define standardization fun.

#The codes of simulated data for quantitative trait are listed in the following. We use mvrnorm
#function to simulate the genotype data. Y is continuous with normal distribution, all errors are
#assumed to be normally distributed with a mean of zero and a variance of one (vari = 1).
out = array(0, dim=c(n.pred)) #For LOOCV
corrmat = diag(rep(1-rho, n.pred)) + matrix(rho, n.pred, n.pred) #Create covariance matrix with rho
corrmat[,5] = sqrt(rho)
corrmat[5,] = sqrt(rho)
corrmat[5,5] = 1
L = array(0, dim=c(n.pred, n.pred, (p/n.pred)))
L[,,1] = corrmat
for(i in 2:(p/n.pred)){ 
	L[,,i] = diag(rep(1-rho, n.pred)) + matrix(rho, n.pred, n.pred)
}
temp = "bdiag(L[,,1]"
for (i in 2:(p/n.pred)){
	temp = paste(temp,",","L[,,",i,"]", sep="")
}
temp = paste(temp,")", sep="")
corrmat2 = eval(parse(text=temp))

beta0 = matrix(0,p,1) #Simulate genotype
X = matrix(0,ntotal,p)
X = mvrnorm(ntotal, beta0, corrmat2 , tol=1e-8, empirical=FALSE)
XA = data.frame(X); colnames(XA) &lt;- c(sapply(1:p,PAS1))
C1 = matrix(0,1,p)
C2 = matrix(0,1,p)
tempMAF = sample(3,1)
for (i in 1:p){
	C2[1,i] = quantile(X[,i], MAF[1,tempMAF])
	C1[1,i] = quantile(X[,i], MAF[2,tempMAF])
	XA[X[,i] &gt; C1[,i],i] = 1
	XA[X[,i] &lt;= C1[,i] &amp; X[,i] &gt;= C2[,i],i] = 0
	XA[X[,i] &lt; C2[,i],i] = -1
}
XA = apply(XA, 2, norm) #Standardization

Y = eval(parse(text=modelY)) #Simulate gaussian response

temp = 1:p
Gene_list = matrix(temp,nrow=n.pred) #Create Gene-based SNP set
#Run TSGSIS fun. with XA, Y, Gene_list, ntest (for predicted model), lambda of lasso regression,
#Method types: "Reg" for quantitative trait; "LR" for disease trait.
Screen_result = TSGSIS(XA, Y, Gene_list, ntest, lambda, Method) 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
