<!DOCTYPE html><html lang="en"><head><title>Help for package TTCA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TTCA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annot'><p>Data file annot</p></a></li>
<li><a href='#annotation'><p>Data file annotation</p></a></li>
<li><a href='#Control'><p>Control time course</p></a></li>
<li><a href='#EGF'><p>EGF time course data</p></a></li>
<li><a href='#TTCA'><p>TTCA: Transcript Time Course Analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Transcript Time Course Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-01-29</td>
</tr>
<tr>
<td>Author:</td>
<td>Marco Albrecht</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Marco Albrecht &lt;marco.albrecht@posteo.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The analysis of microarray time series promises a deeper insight into the dynamics of the cellular response following stimulation. A common observation in this type of data is that some genes respond with quick, transient dynamics, while other genes change their expression slowly over time. The existing methods for detecting significant expression dynamics often fail when the expression dynamics show a large heterogeneity. Moreover, these methods often cannot cope with irregular and sparse measurements. The method proposed here is specifically designed for the analysis of perturbation responses. It combines different scores to capture fast and transient dynamics as well as slow expression changes, and performs well in the presence of low replicate numbers and irregular sampling times. The results are given in the form of tables including links to figures showing the expression dynamics of the respective transcript. These allow to quickly recognise the relevance of detection, to identify possible false positives and to discriminate early and late changes in gene expression. An extension of the method allows the analysis of the expression dynamics of functional groups of genes, providing a quick overview of the cellular response. The performance of this package was tested on microarray data derived from lung cancer cells stimulated with epidermal growth factor (EGF). Paper: Albrecht, Marco, et al. (2017)&lt;<a href="https://doi.org/10.1186%2Fs12859-016-1440-8">doi:10.1186/s12859-016-1440-8</a>&gt;.</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://joinup.ec.europa.eu/software/page/eupl">EUPL version 1.1</a> | <a href="https://joinup.ec.europa.eu/software/page/eupl">EUPL version 1.2</a> [expanded from: EUPL]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix, quantreg, tcltk2, RISmed, VennDiagram, MASS</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-01-29 16:50:06 UTC; Marco Albrecht</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-01-29 18:21:13</td>
</tr>
</table>
<hr>
<h2 id='annot'>Data file annot</h2><span id='topic+annot'></span>

<h3>Description</h3>

<p>Links probeset ID with gene_name</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("annot")</code></pre>


<h3>Format</h3>

<p>A data frame with 5000 observations on the following 2 variables.
</p>

<dl>
<dt><code>probeset_id</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>gene_name</code></dt><dd><p>a character vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>TTCA: An R Package for the identification of differentially expressed genes in time course microarray data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(annot)
</code></pre>

<hr>
<h2 id='annotation'>Data file annotation</h2><span id='topic+annotation'></span>

<h3>Description</h3>

<p>Gives the possibility to link the resulttable with further characteristics.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("annotation")</code></pre>


<h3>Format</h3>

<p>A data frame with 5000 observations on the following 2 variables.
</p>

<dl>
<dt><code>GeneOntology</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Property</code></dt><dd><p>a character vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>TTCA: An R Package for the identification of differentially expressed genes in time course microarray data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(annotation)
</code></pre>

<hr>
<h2 id='Control'>Control time course</h2><span id='topic+Control'></span>

<h3>Description</h3>

<p>The second time course representing the control. Lung cancer cell line.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Control")</code></pre>


<h3>Format</h3>

<p>A data frame with 5000 observations on the following 11 variables.
</p>

<dl>
<dt><code>B_C_2_0</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_3_0</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_2_0.5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_1_1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_3_4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_2_6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_1_24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_3_24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_1_48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_2_48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_3_48</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>TTCA: An R Package for the identification of differentially expressed genes in time course microarray data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Control)
</code></pre>

<hr>
<h2 id='EGF'>EGF time course data</h2><span id='topic+EGF'></span>

<h3>Description</h3>

<p>The first time course representing the sample with epidermal growth factor stimulation. Lung cancer cell line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("EGF")</code></pre>


<h3>Format</h3>

<p>A data frame with 5000 observations on the following 16 variables.
</p>

<dl>
<dt><code>B_C_2_0</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_C_3_0</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_1_0.5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_3_0.5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_1_1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_3_2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_1_4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_1_6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_1_8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_2_12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_2_18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_1_24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_3_24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_1_48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_2_48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B_EGF_3_48</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>TTCA: An R Package for the identification of differentially expressed genes in time course microarray data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(EGF)
</code></pre>

<hr>
<h2 id='TTCA'>TTCA: Transcript Time Course Analysis</h2><span id='topic+TTCA'></span>

<h3>Description</h3>

<p>Background: The analysis of microarray time series promises a deeper insight into the dynamics of the cellular response following stimulation. A common observation in this type of data is that some genes respond with quick, transient dynamics, while other genes change their expression slowly over time. The existing methods for detecting significant expression dynamics often fail when the expression dynamics show a large heterogeneity. Moreover, these methods often cannot cope with irregular and sparse measurements.
Results: The method proposed here is specifically designed for the analysis of perturbation responses. It combines different scores to capture fast and transient dynamics as well as slow expression changes, and performs well in the presence of low replicate numbers and irregular sampling times. The results are given in the form of tables including links to figures showing the expression dynamics of the respective transcript. These allow to quickly recognise the relevance of detection, to identify possible false positives and to discriminate early and late changes in gene expression. An extension of the method allows the analysis of the expression dynamics of functional groups of genes, providing a quick overview of the cellular response. The performance of this package was tested on microarray data derived from lung cancer cells stimulated with epidermal growth factor (EGF).
Paper: Albrecht, Marco, et al. (2017)&lt;DOI:10.1186/s12859-016-1440-8&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TTCA(grp1, grp1.time, grp2, grp2.time, lambda = 0.6, annot = NA,
  annotation = "annotation", timeInt = NULL, pVal = 0.05,
  codetest = FALSE, file = getwd(), MaxPics = 10000, Stimulus1 = "",
  Stimulus2 = "", S = "gene", mapGO = "", PeakMode = "norm")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TTCA_+3A_grp1">grp1</code></td>
<td>
<p>Data set with longitudinal sampled data (data.frame)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_grp1.time">grp1.time</code></td>
<td>
<p>Time points for data set 1 (vector like: c(0,0,0.5,1,2,4,6,8,12,12)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_grp2">grp2</code></td>
<td>
<p>Data set with longitudinal sampled data for comparison (data.frame)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_grp2.time">grp2.time</code></td>
<td>
<p>Time points for data set 2 (vector like: c(0,0,0.5,3,2,4,6,8,12,12,24)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_lambda">lambda</code></td>
<td>
<p>Smoothing parameter in penalty term of quantil regression (default: lambda=0.6 ). Adjust, if fit is too strict or too flexible.</p>
</td></tr>
<tr><td><code id="TTCA_+3A_annot">annot</code></td>
<td>
<p>Annotation for pictures and result (Data.frame with 2 columns with ID and GeneName). (Default: annot=NA)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_annotation">annotation</code></td>
<td>
<p>Merges the TTCA by rowname with a table of your wish. Example: annotation&lt;-annotation[,c(&quot;probeset_id&quot;, &quot;gene_name&quot;,&quot;transkript_id&quot;,&quot;GO_BP&quot;,&quot;GO_CC&quot;,&quot;GO_mf&quot;)]  (default: annotation=&quot;annotation&quot;)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_timeint">timeInt</code></td>
<td>
<p>Defines early, middle and late time period. Defines the middle time period between 4 h and 12 h with timeInt&lt;-c(4,12). (default: timeInt=NULL)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_pval">pVal</code></td>
<td>
<p>P-value for the local hypothesis test        (default: 0.05).</p>
</td></tr>
<tr><td><code id="TTCA_+3A_codetest">codetest</code></td>
<td>
<p>Reduces the data set to 200 features for a quick run of the program. (default: codetest=FALSE)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_file">file</code></td>
<td>
<p>Result folder will be saved at this location   (default: file=getwd() ).</p>
</td></tr>
<tr><td><code id="TTCA_+3A_maxpics">MaxPics</code></td>
<td>
<p>Limits the number of plots (default: MaxPics=10000)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_stimulus1">Stimulus1</code></td>
<td>
<p>Searches this term together with the gene name in PubMed. Stimulus1=&quot;Insulin+like+growth+factor&quot;   ( default: Stimulus1=&quot;&quot;)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_stimulus2">Stimulus2</code></td>
<td>
<p>Searches this term together with the gene name in PubMed. Stimulus2=&quot;epidermal+growth+factor&quot;      ( default: Stimulus2=&quot;&quot;)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_s">S</code></td>
<td>
<p>Defines mode. S ==&quot;GO&quot; changes programm to gene ontology mode  (default:  S=&quot;gene&quot;)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_mapgo">mapGO</code></td>
<td>
<p>Link genes to Gene Ontology terms  (default: mapGO=&quot;&quot;)</p>
</td></tr>
<tr><td><code id="TTCA_+3A_peakmode">PeakMode</code></td>
<td>
<p>Peakmode &quot;norm&quot; uses variance between replicates. If changed to another character value, a normal hypothesis test will be conducted (default: PeakMode=&quot;norm&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The package has not be applied to Hi-Seq data yet. The problem is the huge variety in the read counts. An additional transformation of normalized Hi-Seq data might be an option to scale the values between two values like 0 and 1 (Simple idea: Datalog&lt;-log(data, base = max(data))). Not tested. IF you are interested to adjust my package to sequence data, feel free to contact me.
</p>


<h3>Value</h3>

<p>The R-package delivers a table with different significance values, rankings, p-values. Moreover, it will plot the most important time courses and quality control images.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

##########################################
#### Gene-ANALYSE
##########################################
require(quantreg);require(VennDiagram);require(tcltk2); require(tcltk);
require(RISmed);require(Matrix)
data(EGF,Control,annot,annotation)

S="gene"
Control.time &lt;-  c(0,0,0.5,1,4,6,24,24,48,48,48)
EGF.time     &lt;-  c(0,0,0.5,0.5,1,2,4,6,8,12,18,24,24,48,48,48)
file         =   paste0(getwd(),"/TTCA_Gene")
dir.create(file)
######
TTCAresult&lt;-TTCA(grp1=EGF, grp1.time=EGF.time, grp2=Control, grp2.time=Control.time,S="gene",
                 lambda=0.6, annot=annot, annotation=annotation,pVal=0.05,codetest=FALSE,
                 file=file, Stimulus1="epidermal+growth+factor", timeInt=c(4,12), MaxPics =10000)

## End(Not run)




## Not run: 
##########################################
#### GO-ANALYSE
##########################################
require(quantreg);require(VennDiagram);require(tcltk2); require(tcltk);
require(RISmed);require(Matrix)
#source("https://bioconductor.org/biocLite.R")
#biocLite("biomaRt")
library(biomaRt)
data(EGF,Control,annot,annotation)

require(biomaRt)
ensembl &lt;-  useMart("ENSEMBL_MART_ENSEMBL",dataset="hsapiens_gene_ensembl")
mapGO &lt;- getBM(attributes=c("go_id","name_1006",'affy_hugene_2_0_st_v1'),
               filters = 'affy_hugene_2_0_st_v1', values=rownames(annot), mart =ensembl)
colnames(mapGO)&lt;-c("go_id","GO_name","probeset_id")

S="GO"
Control.time &lt;-  c(0,0,0.5,1,4,6,24,24,48,48,48)
EGF.time     &lt;-  c(0,0,0.5,0.5,1,2,4,6,8,12,18,24,24,48,48,48)
file         =   paste0(getwd(),"/TTCA_GO")
dir.create(file)

TTCAresult&lt;-TTCA(grp1=EGF, grp1.time=EGF.time, grp2=Control, grp2.time=Control.time,
                 S="GO", pVal=0.05,lambda=0.6,codetest=FALSE, file=file,
                 Stimulus1="epidermal+growth+factor", timeInt=c(4,12),
                 MaxPics=10000, mapGO=mapGO)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
