<!DOCTYPE html><html><head><title>Help for package ttScreening</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ttScreening}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ttScreening-package'>
<p>Genome-Wide DNA Methylation Sites Screening by Use of Training and Testing Samples</p></a></li>
<li><a href='#irwsva.build2'>
<p>Adjusted irwsva.build which builds surrogate variables from gene expression data</p></a></li>
<li><a href='#num.sv2'>
<p>Adjusted num.sv which estimates the number of important surrogate variables from a gene expression data set.</p></a></li>
<li><a href='#sva2'>
<p>The adjusted sva code using irwsva.build2</p></a></li>
<li><a href='#ttScreening'>
<p>A screening process built upon training and testing samples</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome-Wide DNA Methylation Sites Screening by Use of Training
and Testing Samples</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-09-18</td>
</tr>
<tr>
<td>Author:</td>
<td>Meredith Ray, Xin Tong, Hongmei Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Meredith Ray &lt;maray@memphis.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A screening process utilizing training and testing samples to filter out uninformative DNA methylation sites. Surrogate variables (SVs) of DNA methylation are included in the filtering process to explain unknown factor effects. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>matrixStats,sva,limma</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics,stats,corpcor,simsalapar,MASS</td>
</tr>
<tr>
<td>Suggests:</td>
<td>mvtnorm</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-09-18 20:29:45 UTC; maray</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-10-11 22:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ttScreening-package'>
Genome-Wide DNA Methylation Sites Screening by Use of Training and Testing Samples
</h2><span id='topic+ttScreening-package'></span>

<h3>Description</h3>

<p>A screening process to filter out non-informative DNA methylation sites by applying (ordinary or robust) linear regressions to training data, and the results are further examined using testing samples. Surrogate variables are included to account for unknown factors.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ttScreening</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.6</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-09-18</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> Artistic-2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package utilizes training and testing samples to filter out uninformative DNA methylation
sites. Surrogate variables (SVs) of DNA methylation are included in the filtering process
to explain unknown factor effects.
</p>


<h3>Author(s)</h3>

<p>Meredith Ray, Xin Tong, Hongmei Zhang
</p>
<p>Maintainer: Meredith Ray &lt;maray@memphis.edu&gt;
</p>


<h3>References</h3>

<p>Ray MA, Tong X, Lockett GA, Zhang H, and Karmaus WJJ. (2016) &ldquo;An Efficient Approach to Screening Epigenome-Wide Data&rdquo;, BioMed Research International.
</p>
<p>Leek JT and Storey JD. (2007) &ldquo;Capturing heterogeneity in gene expression studies by &lsquo;Surrogate Variable Analysis&rsquo;.&rdquo; PLoS Genetics, 3: e161.
</p>


<h3>See Also</h3>

<p><code>sva</code>
</p>

<hr>
<h2 id='irwsva.build2'>
Adjusted irwsva.build which builds surrogate variables from gene expression data
</h2><span id='topic+irwsva.build2'></span>

<h3>Description</h3>

<p>This function is directly modified from the original irwsva.build() in the SVA package. It was noticed that under certain circumstances a subscript out of bounds error would occur while running the SVA function. Therefore, this modified code has a single line altered that conditionally uses the generic singular decomposition, svd(), instead of fast singular decomposition, fast.svd().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>irwsva.build2(dat, mod, mod0 = NULL, n.sv, B = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="irwsva.build2_+3A_dat">dat</code></td>
<td>

<p>A m CpG sites by n subjects matrix of methylation data.
</p>
</td></tr>
<tr><td><code id="irwsva.build2_+3A_mod">mod</code></td>
<td>

<p>A n by k model matrix corresponding to the primary model fit (see model.matrix)
</p>
</td></tr>
<tr><td><code id="irwsva.build2_+3A_mod0">mod0</code></td>
<td>

<p>A n by k0 model matrix corresponding to the null model to be compared to mod.
</p>
</td></tr>
<tr><td><code id="irwsva.build2_+3A_n.sv">n.sv</code></td>
<td>

<p>The number of surrogate variables to construct.
</p>
</td></tr>
<tr><td><code id="irwsva.build2_+3A_b">B</code></td>
<td>

<p>The number of iterations of the algorithm to perform.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See http://www.bioconductor.org/packages/release/bioc/manuals/sva/man/sva.pdf
</p>


<h3>Value</h3>

<table>
<tr><td><code>sv</code></td>
<td>
<p>A n by n.sv matrix where each column is a distinct surrogate variable.
</p>
</td></tr>
<tr><td><code>pprob.gam</code></td>
<td>
<p>A vector with the posterior probability estimates that each row is affected by
dependence.
</p>
</td></tr>
<tr><td><code>pprob.b</code></td>
<td>
<p>A vector with the posterior probabiliity estimates that each row is affected by
the variables in mod, but not in mod0.
</p>
</td></tr>
<tr><td><code>n.sv</code></td>
<td>
<p>The number of suggorate variables estimated.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>sva Vignette http://www.biostat.jhsph.edu/~jleek/sva/
</p>


<h3>Author(s)</h3>

<p>Original irwsva.build: Jeffrey T. Leek &lt;jleek@jhsph.edu&gt;, John Storey jstorey@princeton.edu
</p>


<h3>References</h3>

<p>Original sva:
Leek JT and Storey JD. (2008) A general framework for multiple testing dependence.Proceedings of the National Academy of Sciences, 105: 18718-18723.
</p>
<p>Leek JT and Storey JD. (2007) Capturing heterogeneity in gene expression studies by surrogate variable analysis. PLoS Genetics, 3: e161.
</p>

<hr>
<h2 id='num.sv2'>
Adjusted num.sv which estimates the number of important surrogate variables from a gene expression data set.
</h2><span id='topic+num.sv2'></span>

<h3>Description</h3>

<p>This function is directly modified from the orginal num.sv() in the <span class="pkg">sva</span> package. This function has the tolerance level in the fast.svd() function set back to its orginal default instead of 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>num.sv2(dat, mod, method = c("be", "leek"), vfilter = NULL, 
B = 20, sv.sig = 0.1, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="num.sv2_+3A_dat">dat</code></td>
<td>

<p>A m genes by n arrays matrix of expression data.
</p>
</td></tr>
<tr><td><code id="num.sv2_+3A_mod">mod</code></td>
<td>

<p>A n by k model matrix corresponding to the primary model fit (see model.matrix).
</p>
</td></tr>
<tr><td><code id="num.sv2_+3A_method">method</code></td>
<td>

<p>The method to use for estimating surrogate variables, for now there is only one option (based ib Buja and Eyuboglu 1992).
</p>
</td></tr>
<tr><td><code id="num.sv2_+3A_vfilter">vfilter</code></td>
<td>

<p>The number of most variable genes to use when building SVs, must be between 100 and m.
</p>
</td></tr>
<tr><td><code id="num.sv2_+3A_b">B</code></td>
<td>

<p>The number of null iterations to perform. Only used when method=&quot;be&quot;.
</p>
</td></tr>
<tr><td><code id="num.sv2_+3A_sv.sig">sv.sig</code></td>
<td>

<p>The significance cutoff for eigengenes. Only used when method=&quot;be&quot;.
</p>
</td></tr>
<tr><td><code id="num.sv2_+3A_seed">seed</code></td>
<td>

<p>A numeric seed for reproducible results. Optional, only used when method=&quot;be&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See http://www.bioconductor.org/packages/release/bioc/manuals/sva/man/sva.pdf
</p>


<h3>Value</h3>

<table>
<tr><td><code>n.sv</code></td>
<td>
<p>The number of significant surrogate variables</p>
</td></tr>
</table>


<h3>Note</h3>

<p>sva Vignette http://www.biostat.jhsph.edu/~jleek/sva/
</p>


<h3>Author(s)</h3>

<p>Original num.sv: Jeffrey T. Leek &lt;jleek@jhsph.edu&gt;, John Storey jstorey@princeton.edu
</p>


<h3>References</h3>

<p>Original sva:
Leek JT and Storey JD. (2008) A general framework for multiple testing dependence.Proceedings of the National Academy of Sciences, 105: 18718-18723.
</p>
<p>Leek JT and Storey JD. (2007) Capturing heterogeneity in gene expression studies by surrogate variable analysis. PLoS Genetics, 3: e161.
</p>

<hr>
<h2 id='sva2'>
The adjusted sva code using irwsva.build2
</h2><span id='topic+sva2'></span>

<h3>Description</h3>

<p>This function is the modified SVA function in which it uses the irwsva.build2 function rather than the irwsva.build function to build the surrogate variables. Thus, only a single line has been altered from the original SVA() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sva2(dat, mod, mod0 = NULL, n.sv = NULL, method = c("irw", "two-step"), 
	vfilter = NULL, B = 5, numSVmethod = "be")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sva2_+3A_dat">dat</code></td>
<td>

<p>An m by n (m cpg sites by n subjects) matrix of methylation data.
</p>
</td></tr>
<tr><td><code id="sva2_+3A_mod">mod</code></td>
<td>

<p>A n by k model matrix corresponding to the primary model fit (see model.matrix).
</p>
</td></tr>
<tr><td><code id="sva2_+3A_mod0">mod0</code></td>
<td>

<p>A n by k0 model matrix corresponding to the null model to be compared to mod.
</p>
</td></tr>
<tr><td><code id="sva2_+3A_n.sv">n.sv</code></td>
<td>

<p>Optional. The number of surrogate variables to estimate, can be estimated using the num.sv function.
</p>
</td></tr>
<tr><td><code id="sva2_+3A_method">method</code></td>
<td>

<p>Choose between the iteratively re-weighted or two-step surrogate variable estimation algorithms.
</p>
</td></tr>
<tr><td><code id="sva2_+3A_vfilter">vfilter</code></td>
<td>

<p>The number of most variable CpG sites to use when building SVs, must be between 100 and m.
</p>
</td></tr>
<tr><td><code id="sva2_+3A_b">B</code></td>
<td>

<p>The number of iterations of the algorithm to perform.
</p>
</td></tr>
<tr><td><code id="sva2_+3A_numsvmethod">numSVmethod</code></td>
<td>

<p>The method for determining the number of surrogate variables to use.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See http://www.bioconductor.org/packages/release/bioc/manuals/sva/man/sva.pdf
</p>


<h3>Value</h3>

<table>
<tr><td><code>sv</code></td>
<td>

<p>A n by n.sv matrix where each column is a distinct surrogate variable.
</p>
</td></tr>
<tr><td><code>pprob.gam</code></td>
<td>

<p>A vector with the posterior probability estimates that each row is affected by dependence.
</p>
</td></tr>
<tr><td><code>pprob.b</code></td>
<td>

<p>A vector with the posterior probabiliity estimates that each row is affected by the variables in mod, but not in mod0.
</p>
</td></tr>
<tr><td><code>n.sv</code></td>
<td>

<p>The number of suggorate variables estimated.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>sva Vignette http://www.biostat.jhsph.edu/~jleek/sva/
</p>


<h3>Author(s)</h3>

<p>Original sva: Jeffrey T. Leek &lt;jleek@jhsph.edu&gt;, John Storey jstorey@princeton.edu
</p>


<h3>References</h3>

<p>Original sva:
Leek JT and Storey JD. (2008) A general framework for multiple testing dependence.       Proceedings of the National Academy of Sciences, 105: 18718-18723. 
</p>
<p>Leek JT and Storey JD. (2007) Capturing heterogeneity in gene expression studies by surrogate variable analysis. PLoS Genetics, 3: e161.
</p>

<hr>
<h2 id='ttScreening'>
A screening process built upon training and testing samples
</h2><span id='topic+ttScreening'></span>

<h3>Description</h3>

<p>A screening process to filter out non-informative DNA methylation sites by applying (ordinary or robust) linear regressions to training data, and the results are further examined using testing samples. Surrogate variables are included to account for unknown factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ttScreening(y = y, formula, imp.var, data, B.values=FALSE,iterations = 100, 
sva.method = c("two-step", "irw"), cv.cutoff = 50, n.sv = NULL, 
train.alpha = 0.05, test.alpha = 0.1, FDR.alpha = 0.05, Bon.alpha = 0.05,
percent = (2/3),linear = c("robust", "ls"), vfilter = NULL, B = 5,
 numSVmethod = "be", rowname = NULL, maxit=20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ttScreening_+3A_y">y</code></td>
<td>

<p>Data matrix of DNA methylation measures (m by n, m CpG sites and n subjects). Each column represents DNA methylation measures of all CpG sites for one subject.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_formula">formula</code></td>
<td>

<p>An object of class <code><a href="stats.html#topic+formula">formula</a></code> (or one that can be coerced to that class): a symbolic description of the model to be fitted. The details of model specification are given under &quot;Details&quot;.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_imp.var">imp.var</code></td>
<td>

<p>A vector indicating the location of the term(s) in the formula option on which the selection of CpG sites are made. Interactions are considered a single term. For example, suppose the right-hand side of the equation is: x + z + x:z. If the decision of selecting a CpG site is based on one single term, e.g., the significance of interaction effect, then imp.var is set as the location of that term, e.g., imp.var=3 (the third term). If the decision is desired to base on all the three terms, then imp.var=c(1,2,3).
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_data">data</code></td>
<td>

<p>Data frame created from model.frame. Also is the data frame containing the variables defined in formula.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_b.values">B.values</code></td>
<td>

<p>Logical, TRUE if indicating the methylation is measured as beta values, FALSE if methylation is measured as M-values. The default is FALSE.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_iterations">iterations</code></td>
<td>

<p>Number of loops for the training/testing (TT) procedure. The default is 100.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_sva.method">sva.method</code></td>
<td>

<p>Option of the two surrogate variable estimation algorithms, the iteratively re-weighted, <code>"irw"</code>, or two-step, <code>"two-step"</code>. The default is <code>"two-step"</code>.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_cv.cutoff">cv.cutoff</code></td>
<td>

<p>The minimum frequency required for a DNA methylation site to be treated as an informative site. After &quot;iterations&quot; iterations, the frequency of each DNA methylation being selected out of <code>"iterations"</code> iterations is recorded. The higher the frequency, the more likely the site is informative. The default is 50.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_n.sv">n.sv</code></td>
<td>

<p>Number of surrogate variables. If NULL, the number of surrogate variables will be determined based on the data. The default is NULL.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_train.alpha">train.alpha</code></td>
<td>

<p>Significance level for training samples. The default is 0.05.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_test.alpha">test.alpha</code></td>
<td>

<p>Significance level for testing samples. The default is 0.05.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_fdr.alpha">FDR.alpha</code></td>
<td>

<p>False discovery rate. The default is 0.05. This is to fit the need of selecting variables based on FDR.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_bon.alpha">Bon.alpha</code></td>
<td>

<p>Overall significance level by use of the Bonferroni method for mulitple testing correction. The default is 0.05. This is to fit the need of selecting variables based on the Bonferroni multiple testing correction.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_percent">percent</code></td>
<td>

<p>Proportion of the full sample to be used for training. The default is 2/3.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_linear">linear</code></td>
<td>

<p>Choice of linear regression methods, <code>"robust"</code> (robust regression) or <code>"ls"</code> (ordinary least squares). The default is <code>"ls"</code>.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_vfilter">vfilter</code></td>
<td>

<p>The number of most variable CpG sites to use when building SVs, must be between
100 and the number of genes; Must be NULL or numeric (&gt; 0), The default is NULL. 
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_b">B</code></td>
<td>

<p>Number of iterations in generating surrogate variables. The default is 5.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_numsvmethod">numSVmethod</code></td>
<td>

<p>The method for determining the number of surrogate variables to use. The default is <code>"be"</code>, the other method is <code>"leek"</code>.
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_rowname">rowname</code></td>
<td>

<p>Optional, NULL or <code>"TRUE"</code>. The default is NULL. If rownames are not already present within the data, the order in which the DNA methylation sites are listed will become the rowname. Surrogate variable estimates are formed based on the algorithms in Leek and Storey (2007).
</p>
</td></tr>
<tr><td><code id="ttScreening_+3A_maxit">maxit</code></td>
<td>

<p>Optional, controls the number of iterations for linear regression estimation methods. The default is 20.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <cite><a href="stats.html#topic+lm">lm</a></cite> or <cite><a href="stats.html#topic+glm">glm</a></cite> for details.
</p>


<h3>Value</h3>

<table>
<tr><td><code>sub.remove</code></td>
<td>

<p>Denotes which subjects (based on order) were removed due to incomplete or missing data within the prediction variables defined in the formula arguement.
</p>
</td></tr>
<tr><td><code>train.cpg</code></td>
<td>

<p>Number of DNA methylation sites selected after the training step of each loop.
</p>
</td></tr>
<tr><td><code>test.cpg</code></td>
<td>

<p>Number of DNA methylation sites selected after the testing step of each loop.
</p>
</td></tr>
<tr><td><code>selection</code></td>
<td>

<p>Indicator matrix for the TT method after the testing step. The number of rows is the number of methylation sites, and the number of columns is the number of iterations. An entry of 1 indiates the selection of a site, and 0 otherwise.
</p>
</td></tr>
<tr><td><code>pvalue.matrix</code></td>
<td>

<p>Matrix of p-values of the selected DNA methylation sites after the testing step. The number of rows is the number of methylation sites and the number of columns is the number of iterations. For methylation sites not selected, NA is listed.
</p>
</td></tr>
<tr><td><code>TT.cpg</code></td>
<td>

<p>Final list of the DNA methylation sites by their original rownames selected from the TT method.
</p>
</td></tr>
<tr><td><code>FDR.cpg</code></td>
<td>

<p>Final list of the DNA methylation sites by their original rownames selected from the FDR method.
</p>
</td></tr>
<tr><td><code>Bon.cpg</code></td>
<td>

<p>Final list of the DNA methylation sites by their original rownames selected from the Bonferroni method.
</p>
</td></tr>
<tr><td><code>SV.output</code></td>
<td>

<p>Data frame containing the estimated surrogate variables.
</p>
</td></tr>
<tr><td><code>TT.output</code></td>
<td>

<p>Data frame containing the list of DNA methylation sites selected from the TT method and the respective coefficients and pvalues for the variables and SVs.
</p>
</td></tr>
<tr><td><code>FDR.output</code></td>
<td>

<p>Data frame containing the list of DNA methylation sites selected from the FDR method and the respective coefficients and pvalues for the variables and SVs.
</p>
</td></tr>
<tr><td><code>Bon.output</code></td>
<td>

<p>Data frame containing the list of DNA methylation sites selected from the Bonferroni method and the respective coefficients and pvalues for the variables and SVs.
</p>
</td></tr>
</table>


<h3>References</h3>

<p>Meredith Ray, Xin Tong, Hongmei Zhang, and Wilfred Karmaus. (2014) &quot;DNA methylation sites screening with surrogate variables&quot;, unpublished manuscript.
</p>
<p>Leek JT and Storey JD. (2007) &quot;Capturing heterogeneity in gene expression studies by &lsquo;Surrogate Variable Analysis&rsquo;.&quot; PLoS Genetics, 3: e161.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(mvtnorm)
nsub=600
imp=100
num=2000

set.seed(1)
x1= rnorm(nsub,1,1)
size1&lt;-rmultinom(1,nsub,c(0.15,0.25,0.25,0.35))
x2= matrix(sample(c(rep(0,size1[1,]),
			rep(1,size1[2,]),
			rep(2,size1[3,]),
			rep(3,size1[4,])),replace=F),byrow=250,ncol=1)
			
sur1&lt;-rnorm(nsub,0,5)
sur2&lt;-rnorm(nsub,3,1)
sur3&lt;-rnorm(nsub,0,1)
sur4&lt;-rnorm(nsub,2,4)
sur5&lt;-rnorm(nsub,0,3)

sigma1&lt;-matrix(0,nrow=num,ncol=num)
diag(sigma1)&lt;-1.5

beta0&lt;-0.5
beta1&lt;-0.3
beta2&lt;-0.3
beta3&lt;-0.3

sbeta1&lt;-rnorm(1,0.5,0.01)
sbeta2&lt;-rnorm(1,0.5,0.01)
sbeta3&lt;-rnorm(1,0.5,0.01)
sbeta4&lt;-rnorm(1,0.5,0.01)
sbeta5&lt;-rnorm(1,0.5,0.01)

#beta matrix#
beta&lt;-as.matrix(cbind(beta0,beta1,beta2,beta3,sbeta1,sbeta2,sbeta3,sbeta4,sbeta5))
beta.no2&lt;-as.matrix(cbind(beta0,beta1,beta3,sbeta1,sbeta2,sbeta3,sbeta4,sbeta5))
beta.sur&lt;-as.matrix(cbind(sbeta1,sbeta2,sbeta3,sbeta4,sbeta5))
#design matrix#
X&lt;-as.matrix(cbind(rep(1,length(x1)),x1,x2,x1*x2,sur1,sur2,sur3,sur4,sur5))
X.no2&lt;-as.matrix(cbind(rep(1,length(x1)),x1,x1*x2,sur1,sur2,sur3,sur4,sur5))
X.sur&lt;-as.matrix(cbind(sur1,sur2,sur3,sur4,sur5))
#mu matrix#
imp1.mu&lt;-matrix(rep(X%*%t(beta),9),nrow=nsub,ncol=(imp*0.9))
imp2.mu&lt;-matrix(rep(X.no2%*%t(beta.no2),1),nrow=nsub,ncol=(imp*0.1))
noimp.mu&lt;-matrix(rep(X.sur%*%t(beta.sur),num-imp),nrow=nsub,ncol=num-imp)
mu.matrix=cbind(imp1.mu, imp2.mu, noimp.mu)
error&lt;-rmvnorm(nsub,mean=rep(0,num),sigma=sigma1,method = "chol")
y&lt;-t(mu.matrix+error)


runs&lt;-ttScreening(y=y,formula=~x1+x2+x1:x2,imp.var=3,data=data.frame(x1,x2),sva.method="two-step",
			B.values=FALSE,iterations=100,cv.cutoff=50,n.sv=NULL,train.alpha=0.05,
			test.alpha=0.05,FDR.alpha=0.05,Bon.alpha=0.05,percent=(2/3),linear= "ls",
			vfilter = NULL, B = 5, numSVmethod = "be",rowname=NULL,maxit=20)

runs$TT.output
runs$FDR.output
runs$Bon.output

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
