<!DOCTYPE html><html><head><title>Help for package TwoPhaseInd</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {TwoPhaseInd}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aco1arm'><p>A function to estimate parameters in augmented case-only designs, the genotype</p>
is ascertained for a random subcohort from the active treatment arm or the placebo arm.</a></li>
<li><a href='#aco2arm'>
<p>A function to estimate parameters in Cox proportional hazards model using augmented</p>
case-only designs, the genotype is ascertained for a random subcohort from both the
active treatment arm and the placebo arm (case-cohort sampling) or a case-control
sample in both arms.</a></li>
<li><a href='#acoarm'>
<p>A function to estimate parameters in Cox proportional hazard models by augmented</p>
case-only designs for randomized clinical trials with failure time endpoints.</a></li>
<li><a href='#acodata'><p>A dataset from the STEP trial to study the interactions between gene</p>
and vaccine on HIV infection</a></li>
<li><a href='#caseonly'>
<p>A function to deal with case-only designs</p></a></li>
<li><a href='#char2num'>
<p>A function used in acoarm to transform categorical variable to integers</p></a></li>
<li><a href='#mele'><p>function to compute the maximum estimated likelihood estimator</p></a></li>
<li><a href='#remove_missingdata'>
<p>A function used in acoarm to remove missing data</p></a></li>
<li><a href='#remove_rarevariants'>
<p>A function used in spmle and acoarm to remove rare-variant covariates</p></a></li>
<li><a href='#spmle'><p>function to compute the semiparametric maximum likelihood estimator</p></a></li>
<li><a href='#whiBioMarker'><p>An example dataset to demostrate the usage of MELE and SPMLE</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimate Gene-Treatment Interaction Exploiting Randomization</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Author:</td>
<td>James Dai [aut, cre],
  Xiaoyu Wang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>James Dai &lt;jdai@fredhutch.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimation of gene-treatment interactions in randomized clinical trials exploiting gene-treatment independence. Methods used in the package refer to J. Y. Dai, M. LeBlanc, and C. Kooperberg (2009) Biometrics &lt;<a href="https://doi.org/10.1111%2Fj.1541-0420.2008.01046.x">doi:10.1111/j.1541-0420.2008.01046.x</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-16 19:51:51 UTC; jdai</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Imports:</td>
<td>survival</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-17 00:50:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='aco1arm'>A function to estimate parameters in augmented case-only designs, the genotype
is ascertained for a random subcohort from the active treatment arm or the placebo arm. 
</h2><span id='topic+aco1arm'></span>

<h3>Description</h3>

<p>This function estimates parameters of proportional hazards model with gene-treatment 
interaction. It employs case-cohort estimation incorporating the case-only estimators. 
The method was published in Dai et al. (2016) Biometrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aco1arm(data, svtime, event, treatment, BaselineMarker, subcohort, esttype = 1, 
augment = 1, extra)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aco1arm_+3A_data">data</code></td>
<td>

<p>A data frame used to access the following data.
</p>
</td></tr>
<tr><td><code id="aco1arm_+3A_svtime">svtime</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store the failure time variable (numeric). 
</p>
</td></tr>
<tr><td><code id="aco1arm_+3A_event">event</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame,
which is used to store the indicator of failure event (1: failure, 0: not failure).
</p>
</td></tr>
<tr><td><code id="aco1arm_+3A_treatment">treatment</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which 
is used to store the binary vector of treatment variable (1: treatment, 0: placebo). 
</p>
</td></tr>
<tr><td><code id="aco1arm_+3A_baselinemarker">BaselineMarker</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store a vector of biomarker. 
</p>
</td></tr>
<tr><td><code id="aco1arm_+3A_subcohort">subcohort</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store the indicator of sub-cohort 
(1: sample belong to the sub-cohort, 0: not belong to the sub-cohort)
</p>
</td></tr>
<tr><td><code id="aco1arm_+3A_esttype">esttype</code></td>
<td>

<p>The option of estimation methods (1: Self-Prentice estimator, 0: Lin-Ying estimator).
</p>
</td></tr>
<tr><td><code id="aco1arm_+3A_augment">augment</code></td>
<td>

<p>The indicator of whether subcohort was drawn from the active treatment arm (augment=1)
or from the placebo arm (augment=0).
</p>
</td></tr>
<tr><td><code id="aco1arm_+3A_extra">extra</code></td>
<td>

<p>A string vector of column name(s), corresponds to more or more column(s) of the 
data frame, which is/are used to store the extra baseline covariate(s) to be adjusted 
for in addition to treatment and biomarker.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns estimates of the proportional hazards model, and variance of 
the estimates. The method was published in Dai et al. (2016) Biometrics.
</p>


<h3>Value</h3>

<p>A list of estimates and variance of the estimates.
</p>
<table>
<tr><td><code>Estimate</code></td>
<td>
<p>A data frame of beta(Estimated parameter),
stder(Standard error),and pVal(p value)</p>
</td></tr>
<tr><td><code>Covariance</code></td>
<td>
<p>covariance data frame of genotype,treatment,and interaction</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>References</h3>

<p>J. Y. Dai, X. C. Zhang,C. Y. Wang, and C. Kooperberg. Augmented case-only designs for 
randomized clinical trials with failure time endpoints. Biometrics, DOI: 10.1111/biom.12392, 2016.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aco2arm">aco2arm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the example data
data(acodata)

## Augmented data in the active arm

rfit1 &lt;- aco1arm(data=acodata,
                 svtime="vacc1_evinf",
                 event="f_evinf",
                 treatment="f_treat",
                 BaselineMarker="fcgr2a.3",
                 subcohort="subcoh",
                 esttype=1,
                 augment=1,
                 extra=c("f_agele30","f_hsv_2","f_ad5gt18","f_crcm","any_drug",
                         "num_male_part_cat","uias","uras")) 

rfit1

## Augmented data in the placebo arm

rfit2 &lt;- aco1arm(data=acodata,
                 svtime="vacc1_evinf",
                 event="f_evinf",
                 treatment="f_treat",
                 BaselineMarker="fcgr2a.3",
                 subcohort="subcoh",
                 esttype=1,
                 augment=0,
                 extra=c("f_agele30","f_hsv_2","f_ad5gt18","f_crcm", 
                         "any_drug","num_male_part_cat","uias","uras")) 
rfit2

</code></pre>

<hr>
<h2 id='aco2arm'>
A function to estimate parameters in Cox proportional hazards model using augmented 
case-only designs, the genotype is ascertained for a random subcohort from both the 
active treatment arm and the placebo arm (case-cohort sampling) or a case-control 
sample in both arms. 
</h2><span id='topic+aco2arm'></span>

<h3>Description</h3>

<p>This function estimates parameters of proportional hazards model with gene-treatment 
interaction. It employs case-cohort estimation incorporating the case-only estimators. 
The method was published in Dai et al. (2015) Biometrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aco2arm(data, svtime, event, treatment, BaselineMarker, subcohort=NULL, 
esttype = NULL, weight=NULL, extra=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aco2arm_+3A_data">data</code></td>
<td>

<p>A data frame used to access the following data.
</p>
</td></tr>
<tr><td><code id="aco2arm_+3A_svtime">svtime</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store the failure time variable (numeric).
</p>
</td></tr>
<tr><td><code id="aco2arm_+3A_event">event</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame,
which is used to store the indicator of failure event (1: failure, 0: not failure).
</p>
</td></tr>
<tr><td><code id="aco2arm_+3A_treatment">treatment</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which 
is used to store the binary vector of treatment variable (1: treatment, 0: placebo).
</p>
</td></tr>
<tr><td><code id="aco2arm_+3A_baselinemarker">BaselineMarker</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store a vector of biomarker.
</p>
</td></tr>
<tr><td><code id="aco2arm_+3A_subcohort">subcohort</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store the indicator of sub-cohort in the case-cohort sampling 
(1: sample belong to the sub-cohort, 0: not belong to the sub-cohort). 
In case-control sampling, this variable is set to be NULL.
</p>
</td></tr>
<tr><td><code id="aco2arm_+3A_esttype">esttype</code></td>
<td>

<p>The option of estimation methods (1: Self-Prentice estimator, 0: Lin-Ying estimator).
</p>
</td></tr>
<tr><td><code id="aco2arm_+3A_weight">weight</code></td>
<td>

<p>If the genotype data are obtained through case-control sampling, weight is a vector 
of sampling weights (inverse of sampling probability) corresponding to rows of data. 
If the genotype data are obtained through case-cohort sampling, 
weight is NULL. If a vector of weights have been supplied by user, 
then esttype is automatically set to 0: Lin-Ying estimator.
</p>
</td></tr>
<tr><td><code id="aco2arm_+3A_extra">extra</code></td>
<td>

<p>A string vector of column name(s), corresponds to more or more column(s) of the 
data frame, which is/are used to store the extra baseline covariate(s) to be 
adjusted for in addition to treatment and biomarker.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns estimates of the proportional hazards model, and variance of the 
estimates. The method was published in Dai et al. (2016) Biometrics.
</p>


<h3>Value</h3>

<p>A list of estimates and variance of the estimates.
</p>
<table>
<tr><td><code>Estimate</code></td>
<td>
<p>A data frame of beta(Estimated parameter),
stder(Standard error),and pVal(p value)</p>
</td></tr>
<tr><td><code>Covariance</code></td>
<td>
<p>covariance data frame of genotype,treatment,and interaction</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>References</h3>

<p>J. Y. Dai, X. C. Zhang,C. Y. Wang, and C. Kooperberg. Augmented case-only designs 
for randomized clinical trials with failure time endpoints. Biometrics, DOI: 10.1111/biom.12392, 2016.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+aco1arm">aco1arm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the example data
data(acodata)
## Case-cohort + case-only estimators
rfit1 &lt;- aco2arm(data=acodata,
                 svtime="vacc1_evinf",
                 event="f_evinf",
                 treatment="f_treat",
                 BaselineMarker="fcgr2a.3",
                 subcohort="subcoh",
                 esttype=1,
                 weight=NULL,
                 extra=c("f_agele30","f_hsv_2","f_ad5gt18","f_crcm","any_drug",
                 "num_male_part_cat","uias","uras")) 
rfit1
</code></pre>

<hr>
<h2 id='acoarm'>
A function to estimate parameters in Cox proportional hazard models by augmented 
case-only designs for randomized clinical trials with failure time endpoints.
</h2><span id='topic+acoarm'></span>

<h3>Description</h3>

<p>This function estimates parameters of proportional hazards models with gene-treatment 
interactions. It employs classical case-cohort estimation methods, incorporating the 
case-only estimators. The method was published in Dai et al. (2016) Biometrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acoarm(data, svtime, event, treatment, BaselineMarker,subcohort, esttype = 1, 
augment = 1, weight=NULL, extra = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="acoarm_+3A_data">data</code></td>
<td>

<p>A data frame used to access the following data.
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_svtime">svtime</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store the failure time variable (numeric). 
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_event">event</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store the indicator of failure event (1: failure, 0: not failure).
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_treatment">treatment</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which is 
used to store the binary vector of treatment variable (1: treatment, 0: placebo). 
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_baselinemarker">BaselineMarker</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which is 
used to store a vector of baseline biomarker that is under investigation for interaction 
with treatment. The BaselineMarker variable is missing for those who are not sampled 
in the case-cohort.
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_subcohort">subcohort</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which 
is used to store the indicator of sub-cohort (1: sample belong to the sub-cohort, 
0: not belong to the sub-cohort)
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_esttype">esttype</code></td>
<td>

<p>The option of estimation methods (1: Self-Prentice estimator, 0: Lin-Ying estimator).
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_augment">augment</code></td>
<td>

<p>The indicator of whether subcohort was drawn from the placebo arm (augment=0), from the 
active treatment arm (augment=1), or from both arms (augment=2).
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_weight">weight</code></td>
<td>

<p>If the genotype data are obtained through case-control sampling, weight is a vector 
of sampling weights (inverse of sampling probability) corresponding to rows of data. 
If the genotype data are obtained through case-cohort sampling, 
weight is NULL. If a vector of weights have been supplied by user, 
then esttype is automatically set to 0: Lin-Ying estimator.
</p>
</td></tr>
<tr><td><code id="acoarm_+3A_extra">extra</code></td>
<td>

<p>A string vector of column name(s), corresponds to more or more column(s) of the data 
frame, which is/are used to store the extra baseline covariate(s) to be adjusted for 
in addition to treatment and biomarker.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns point estimates and standard error estimates of parameters in the 
proportional hazards model. The method was published in Dai et al. (2015) Biometrics.
</p>


<h3>Value</h3>

<table>
<tr><td><code>beta</code></td>
<td>
<p>Estimated parameter</p>
</td></tr>
<tr><td><code>stder</code></td>
<td>
<p>Estimated standard error of parameter estimates</p>
</td></tr>
<tr><td><code>pVal</code></td>
<td>
<p>p value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>References</h3>

<p>J. Y. Dai, X. C. Zhang,C. Y. Wang, and C. Kooperberg. Augmented case-only designs for 
randomized clinical trials with failure time endpoints. 
Biometrics, DOI: 10.1111/biom.12392, 2016.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the example data
data(acodata)
## ACO in placebo arm
rfit0 &lt;- acoarm(data=acodata,
                 svtime="vacc1_evinf",
                 event="f_evinf",
                 treatment="f_treat",
                 BaselineMarker="fcgr2a.3",
                 subcohort="subcoh",
                 esttype=1,
                 augment=0,
                 weight=NULL,
                 extra=c("f_agele30","f_hsv_2","f_ad5gt18","f_crcm",
                 "any_drug","num_male_part_cat","uias","uras")) 
rfit0

## ACO in active arm
rfit1 &lt;- acoarm(data=acodata,
                 svtime="vacc1_evinf",
                 event="f_evinf",
                 treatment="f_treat",
                 BaselineMarker="fcgr2a.3",
                 subcohort="subcoh",
                 esttype=1,
                 augment=1,
                 weight=NULL,
                 extra=c("f_agele30","f_hsv_2","f_ad5gt18","f_crcm",
                 "any_drug","num_male_part_cat","uias","uras")) 
rfit1

## ACO in both arms
rfit2 &lt;- acoarm(data=acodata,
                 svtime="vacc1_evinf",
                 event="f_evinf",
                 treatment="f_treat",
                 BaselineMarker="fcgr2a.3",
                 subcohort="subcoh",
                 esttype=1,
                 augment=2,
                 weight=NULL,
                 extra=c("f_agele30","f_hsv_2","f_ad5gt18","f_crcm",
                 "any_drug","num_male_part_cat","uias","uras")) 
rfit2

</code></pre>

<hr>
<h2 id='acodata'>A dataset from the STEP trial to study the interactions between gene 
and vaccine on HIV infection
</h2><span id='topic+acodata'></span>

<h3>Description</h3>

<p>A dataset from the STEP trial to study the interactions between gene and 
vaccine on HIV infection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("acodata")</code></pre>


<h3>Format</h3>

<p>A data frame with 907 observations on the following 14 variables.
</p>

<dl>
<dt><code>vacc1_evinf</code></dt><dd><p>the time to HIV infection, a numeric vector</p>
</dd>
<dt><code>f_evinf</code></dt><dd><p>the indicator variable for HIV infection, a numeric vector</p>
</dd>
<dt><code>subcoh</code></dt><dd><p>the indicator of whether the participant was selected into 
the sub-cohort for genotyping, a logical vector</p>
</dd>
<dt><code>ptid</code></dt><dd><p>patricipant identifier, a numeric vector</p>
</dd>
<dt><code>f_treat</code></dt><dd><p>vaccine assignment variable, a numeric vector</p>
</dd>
<dt><code>fcgr2a.3</code></dt><dd><p>the genotype of Fcr receptor FcrRIIIa, the biomarker of 
interest here, a numeric vector</p>
</dd>
<dt><code>f_agele30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>f_hsv_2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>f_ad5gt18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>f_crcm</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>any_drug</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>num_male_part_cat</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>uias</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>uras</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>A dataset from the STEP trial to study the interactions between gene and vaccine on 
HIV infection
</p>


<h3>References</h3>

<p>S. P. Buchbinder, D. V. Mehrotra, and D. Ann et al. Efficacy assessment of a 
cell-mediated immunity HIV-1 vaccine (the Step Study): a double-blind, randomised, 
placebo-controlled, test-of-concept trial. Lancet. 372(9653):1881-1893, 2008.
</p>
<p>J. P. Pandey, A. M. Namboodiri, and S. Bu et l. Immunoglobulin genes and the 
acquisition of HIV infection in a randomized trial of recombinant adenovirus HIV 
vaccine. Virology, 441:70-74, 2013.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(acodata)
## maybe str(acodata)
</code></pre>

<hr>
<h2 id='caseonly'>
A function to deal with case-only designs
</h2><span id='topic+caseonly'></span>

<h3>Description</h3>

<p>This function estimates parameters of case-only designs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>caseonly(data, treatment, BaselineMarker, extra = NULL, fraction = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="caseonly_+3A_data">data</code></td>
<td>

<p>A data frame used to access the following data.
</p>
</td></tr>
<tr><td><code id="caseonly_+3A_treatment">treatment</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which
is used to store the binary vector of treatment variable (1: treatment, 0: placebo). 
</p>
</td></tr>
<tr><td><code id="caseonly_+3A_baselinemarker">BaselineMarker</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, 
which is used to store a vector of biomarker.
</p>
</td></tr>
<tr><td><code id="caseonly_+3A_extra">extra</code></td>
<td>

<p>A string vector of column name(s), corresponds to more or more column(s) of the data 
frame, which is/are used to store the extra baseline covariate(s) to be included in 
case-only regression. Note that extra covariates are not needed unless the 
interactions of treatment and extra coviarates are of interest.
</p>
</td></tr>
<tr><td><code id="caseonly_+3A_fraction">fraction</code></td>
<td>

<p>The randomization fraction of active treatment assignment.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function estimates parameters of case-only designs. It estimates two parameters 
for &quot;treatment effect when baselineMarker=0&quot;&quot; and treatment+baselineMarker interaction&quot;.
</p>


<h3>Value</h3>

<p>For each paramter, it returns:
</p>
<table>
<tr><td><code>beta</code></td>
<td>
<p>Estimated parameter</p>
</td></tr>
<tr><td><code>stder</code></td>
<td>
<p>Standard error</p>
</td></tr>
<tr><td><code>pVal</code></td>
<td>
<p>p value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>References</h3>

<p>J. Y. Dai, S. S. Li, and P. B. Gilbert. Case-only methods for competing risks models 
with application to assessing differential vaccine efficacy by viral and host genetics. 
Biometrics, 15(1):196-203, 2014.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#form the data
data(acodata)
cdata=acodata[acodata[,2]==1,]
cfit=caseonly(data=cdata,
              treatment="f_treat",
              BaselineMarker="fcgr2a.3",
              extra=c("f_agele30","f_hsv_2","f_ad5gt18","f_crcm",
              "any_drug","num_male_part_cat","uias","uras"))
cfit              
</code></pre>

<hr>
<h2 id='char2num'>
A function used in acoarm to transform categorical variable to integers
</h2><span id='topic+char2num'></span>

<h3>Description</h3>

<p>Transform category data to integers 0..levels(data)-1. The the numeric 
variable can be then used in acoarm models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>char2num(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="char2num_+3A_data">data</code></td>
<td>

<p>data is a dataframe composed of categorical variables. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function transforms a categorical variable to integers.
</p>


<h3>Value</h3>

<p>A data frame of transformed values. For each column, each category is 
transformed to an integer, from 0 to levels(data[,column])-1.
</p>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the example data
data(acodata)
result &lt;- char2num(acodata[, "fcgr2a.3"])
</code></pre>

<hr>
<h2 id='mele'>function to compute the maximum estimated likelihood estimator</h2><span id='topic+mele'></span>

<h3>Description</h3>

<p>This function computes the maximum estimated likelihood estimator (MELE) of 
regression parameters, which assess treatment-biomarker interactions in studies 
with two-phase sampling in randomized clinical trials. The function has an option 
to incorporate the independence between a randomized treatment and the baseline markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mele(data, response, treatment, BaselineMarker, extra = NULL, phase, 
ind = TRUE, maxit=2000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mele_+3A_data">data</code></td>
<td>

<p>A data frame used to access the following data. Each row contains the response and 
predictors of a study participant. All variables are numerical.
</p>
</td></tr>
<tr><td><code id="mele_+3A_response">response</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which 
is used to store a numeric vector of response. The response variable should be coded 
as 1 for cases and 0 for controls.
</p>
</td></tr>
<tr><td><code id="mele_+3A_treatment">treatment</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which
is used to store a binary vector of the treatment . The treatment variable should 
be coded as 1 for treatment and 0 for placebo.
</p>
</td></tr>
<tr><td><code id="mele_+3A_baselinemarker">BaselineMarker</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which 
is used to store a vector of biomarker that is assessed for interaction with the 
treatment. The BaselineMarker variable is missing for those who are not sampled in 
the second phase.
</p>
</td></tr>
<tr><td><code id="mele_+3A_extra">extra</code></td>
<td>

<p>A string vector of column name(s), corresponds to one or more column(s) of the data 
frame, which are used to store the extra covariate(s) to be adjusted for in addition 
to treatment and biomarker. All extra variables are missing for those 
who are not sampled in the second phase.
</p>
</td></tr>
<tr><td><code id="mele_+3A_phase">phase</code></td>
<td>

<p>A character string of column name, correspond to one column of the data frame, which 
is used to store the  indicator of two-phase sampling (1: not being sampled for 
measuring biomarker; 2: being sampled for measuring biomarker).
</p>
</td></tr>
<tr><td><code id="mele_+3A_ind">ind</code></td>
<td>

<p>A logical flag. TRUE indicates incorporating the independence between the randomized 
treatment and the baseline markers.
</p>
</td></tr>
<tr><td><code id="mele_+3A_maxit">maxit</code></td>
<td>

<p>A integer number of the maximal number of iteration.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns estimates, standard errors, and p values for MELE of a regression 
model for treatment-biomarker interaction studies with two-phase sampling in randomized 
trials, response ~ treatment + biomarker + treatment*biomarker + other covariates. 
Treatment and response are available for all the samples, while baseline biomarker data 
are available for a subset of samples. The mele can incorporate the independence 
between the treatment and baseline biomarkers ascertained in the phase-two sample.
</p>


<h3>Value</h3>

<table>
<tr><td><code>beta</code></td>
<td>
<p>Estimated parameter</p>
</td></tr>
<tr><td><code>stder</code></td>
<td>
<p>Standard error</p>
</td></tr>
<tr><td><code>pVal</code></td>
<td>
<p>p value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>References</h3>

<p>J. Y. Dai, M. LeBlanc, and C. Kooperberg. Semiparametric estimation exploiting covariate 
independence in two-phase randomized trials. Biometrics, 65(1):178-187, 2009.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+spmle">spmle</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the example data
data(whiBioMarker)
## Here is an example of MELE with exploiting independent and with confounding factors:

melIndExtra &lt;- mele(data=whiBioMarker,  ## dataset
          response="stroke",	## response variable
          treatment="hrtdisp",		## treatment variable
          BaselineMarker="papbl",		## environment variable
          extra=c(
             "age" 	## age
                		## physical activity levels
              , "dias" 	## diabetes
              , "hyp" ## hypertension
              , "syst" 	## systolic
              , "diabtrt"	## diastolic BP
              , "lmsepi" ## waist:hip ratio
              ),	## extra variable(s)
          phase="phase",	## phase indicator
          ind=TRUE	## independent or non-indepentent
)
</code></pre>

<hr>
<h2 id='remove_missingdata'>
A function used in acoarm to remove missing data
</h2><span id='topic+remove_missingdata'></span>

<h3>Description</h3>

<p>It is used to remove samples which have NA/missing data in covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_missingdata(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_missingdata_+3A_data">data</code></td>
<td>

<p>data is a dataframe.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function removes samples (by rows) which have NA/missing data.
</p>


<h3>Value</h3>

<p>A list of the following components.
</p>
<table>
<tr><td><code>idx</code></td>
<td>
<p>The indices of rows without missing values</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>The dataframe without missing values</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the example data
data(acodata)
result &lt;- remove_missingdata(acodata[, c("vacc1_evinf","fcgr2a.3")])
</code></pre>

<hr>
<h2 id='remove_rarevariants'>
A function used in spmle and acoarm to remove rare-variant covariates
</h2><span id='topic+remove_rarevariants'></span>

<h3>Description</h3>

<p>It is used to remove rare-variant covariates, which can cause divergence problem.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_rarevariants(data, cutoff = 0.02)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_rarevariants_+3A_data">data</code></td>
<td>

<p>A dataframe composed of covariates.
</p>
</td></tr>
<tr><td><code id="remove_rarevariants_+3A_cutoff">cutoff</code></td>
<td>

<p>Proportion cutoff. If data composed of more than (1-cutoff) proportion of a constant 
value, we call it rare-variant.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function removes rare-variant covariates.
</p>


<h3>Value</h3>

<p>A logical vector composed of True or False. True means a covariate is rare-variant.
</p>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the example data
data(acodata)
result &lt;- remove_rarevariants(acodata[, c("vacc1_evinf","fcgr2a.3")])
</code></pre>

<hr>
<h2 id='spmle'>function to compute the semiparametric maximum likelihood estimator</h2><span id='topic+spmle'></span>

<h3>Description</h3>

<p>This function computes the semiparametric maximum likelihood estimator (SPMLE) of 
regression parameters, which assess treatment-biomarker interactions in studies 
with two-phase sampling  in randomized clinical trials. The function has an option 
to incorporate the independence between a randomized treatment and the baseline 
markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spmle(data, response, treatment, BaselineMarker, extra = NULL, phase, 
ind = TRUE, difffactor = 0.001, maxit = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spmle_+3A_data">data</code></td>
<td>

<p>A data frame used to access the following data. Each row contains the response and 
predictors of a study participant. All variables are numerical.
</p>
</td></tr>
<tr><td><code id="spmle_+3A_response">response</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which 
is used to store a numeric vector of response. 
The response variable should be coded as 1 for cases and 0 for controls.
</p>
</td></tr>
<tr><td><code id="spmle_+3A_treatment">treatment</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which 
is used to store a binary vector of the treatment . The treatment variable should be 
coded as 1 for treatment and 0 for placebo.
</p>
</td></tr>
<tr><td><code id="spmle_+3A_baselinemarker">BaselineMarker</code></td>
<td>

<p>A character string of column name, corresponds to one column of the data frame, which 
is used to store a vector of biomarker that is assessed for interaction with the 
treatment. The BaselineMarker variable is missing for those who are not sampled 
in the second phase.
</p>
</td></tr>
<tr><td><code id="spmle_+3A_extra">extra</code></td>
<td>

<p>A string vector of column name(s), corresponds to one or more column(s) of the data 
frame, which are used to store the extra covariate(s) to be adjusted for in addition 
to treatment and biomarker. All extra variables are missing for those who are not 
sampled in the second phase.
</p>
</td></tr>
<tr><td><code id="spmle_+3A_phase">phase</code></td>
<td>

<p>A character string of column name, correspond to one column of the data frame, which 
is used to store the  indicator of two-phase sampling (1: not being sampled for 
measuring biomarker; 2: being sampled for measuring biomarker).
</p>
</td></tr>
<tr><td><code id="spmle_+3A_ind">ind</code></td>
<td>

<p>A logical flag. TRUE indicates incorporating the independence between the randomized 
treatment and the baseline markers.
</p>
</td></tr>
<tr><td><code id="spmle_+3A_difffactor">difffactor</code></td>
<td>

<p>A decimal number of the differentiation factor, used to control the step of 
numerical differentiation.
</p>
</td></tr>
<tr><td><code id="spmle_+3A_maxit">maxit</code></td>
<td>

<p>A integer number of the maximal number of numerical differentiation iteration.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns estimates, standard errors, and p values for SPMLE for parameters 
of a regression model for treatment-biomarker interaction studies with two-phase 
sampling in randomized trials, response ~ treatment + biomarker + treatment*biomarker 
+ other covariates. Treatment and response are available for all the samples, while 
biomarker data are available for a subset of samples. The SPMLE can incorporate the 
independence between the treatment and baseline biomarkers ascertained in the phase-two 
sample. A profile likelihood based Newton-Raphson algorithm is used to compute SPMLE.
</p>


<h3>Value</h3>

<table>
<tr><td><code>beta</code></td>
<td>
<p>Estimated parameter</p>
</td></tr>
<tr><td><code>stder</code></td>
<td>
<p>Standard error</p>
</td></tr>
<tr><td><code>pVal</code></td>
<td>
<p>p value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Y. Dai
</p>


<h3>References</h3>

<p>J. Y. Dai, M. LeBlanc, and C. Kooperberg. Semiparametric estimation exploiting 
covariate independence in two-phase randomized trials. Biometrics, 
65(1):178-187, 2009.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mele">mele</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load the example data
data(whiBioMarker)
## Here is an example of SPMLE with exploiting independent and with confounding factors:
spmleIndExtra &lt;- spmle(data=whiBioMarker,  ## dataset
                       response="stroke",  ## response variable
                       treatment="hrtdisp",  ## treatment variable
                       BaselineMarker="papbl",	## environment variable
                       extra=c(
                         "age" 	## age
                         , "dias"	## diabetes
                         , "hyp" ## hypertension
                         , "syst" ## systolic
                         , "diabtrt"	## diastolic BP
                         , "lmsepi" ## waist:hip ratio
                       ),	## extra variable(s)
                       phase="phase", ## phase indicator
                       ind=TRUE ## independent or non-independent
)

</code></pre>

<hr>
<h2 id='whiBioMarker'>An example dataset to demostrate the usage of MELE and SPMLE
</h2><span id='topic+whiBioMarker'></span>

<h3>Description</h3>

<p>A dataset from a Women's Health Initiative (WHI) hormone trial to study the interaction 
between biomarker and hormone therapy on stroke.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("whiBioMarker")</code></pre>


<h3>Format</h3>

<p>A data frame consisting of 10 observations, with the following columns:
</p>

<dl>
<dt><code>stroke</code></dt><dd><p>a binary indicator vector of stroke; 1=has stroke</p>
</dd>
<dt><code>hrtdisp</code></dt><dd><p>a binary indicator vector of treatment in the Estrogen Plus 
Progestin Trial; 1=&quot;Estrogen Plus Progestin&quot;, 0=&quot;placebo&quot;</p>
</dd>
<dt><code>papbl</code></dt><dd><p>a numeric vector of Biomarker PAP (plasmin-antiplasmin complex)
in logarithmic scale (base 10)</p>
</dd>
<dt><code>age</code></dt><dd><p>an integer vector of age</p>
</dd>
<dt><code>dias</code></dt><dd><p>A binary indicator vector of Diastolic BP; 1=&quot;Yes&quot;</p>
</dd>
<dt><code>hyp</code></dt><dd><p>a vector of hypertension with levels <code>Missing</code>, 
<code>No</code>, <code>Yes</code></p>
</dd>
<dt><code>syst</code></dt><dd><p>an integer vector of Systolic BP</p>
</dd>
<dt><code>diabtrt</code></dt><dd><p>A vector of Diabetes with levels: <code>Missing</code>, 
<code>No</code>, <code>Yes</code></p>
</dd>
<dt><code>lmsepi</code></dt><dd><p>A vector of episodes per week of moderate and strenuous 
recreational physical activity of &gt;= 20 minutes duration with levels 
<code>2 - &lt;4 episodes per week</code>, <code>4+ episodes per week</code>, 
<code>Missing</code>, <code>No activity</code>, <code>Some activity</code></p>
</dd>
<dt><code>phase</code></dt><dd><p>a numeric vector of phase; 1: phase 1, 2:phase 2</p>
</dd>
</dl>



<h3>Details</h3>

<p>It is an two-phase sampling example dataset adapted from Kooperberg et al. (2007) to 
demostrate the usage of MELE and SPMLE algorithms in Dai et al. (2009).
</p>


<h3>Source</h3>

<p>C. Kooperberg, M. Cushman, J. Hsia, J. G. Robinson, A. K. Aragaki, J. K. Lynch, A. E. 
Baird, K. C. Johnson, L. H. Kuller, S. A. Beresford, and B. Rodriguez. Can biomarkers 
identify women at increased stroke risk? the women's health initiative hormone trials. 
PLoS clinical trials, 2(6):e28, Jun 15 2007.
</p>


<h3>References</h3>

<p>J. Y. Dai, M. LeBlanc, and C. Kooperberg. Semiparametric estimation exploiting 
co-variate independence in two-phase randomized trials. Biometrics, 65(1):178-187, 2009.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whiBioMarker)
str(whiBioMarker)
colnames(whiBioMarker)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
