<!DOCTYPE html><html><head><title>Help for package UCSCXenaShiny</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {UCSCXenaShiny}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#UCSCXenaShiny-package'><p>Xena Shiny App</p></a></li>
<li><a href='#.opt_pancan'><p>A default setting for pan-cancer studies</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe Operator</p></a></li>
<li><a href='#analyze_gene_drug_response_asso'><p>Analyze Association between Gene (Signature) and Drug Response with CCLE Data</p></a></li>
<li><a href='#analyze_gene_drug_response_diff'><p>Analyze Difference of Drug Response (IC50 Value (uM)) between Gene (Signature) High and Low Expression with CCLE Data</p></a></li>
<li><a href='#app_run'><p>Run UCSC Xena Shiny App</p></a></li>
<li><a href='#available_hosts'><p>Show Available Hosts</p></a></li>
<li><a href='#ccle_absolute'><p>ABSOLUTE Result of CCLE Database</p></a></li>
<li><a href='#ccle_info'><p>Phenotype Info of CCLE Database</p></a></li>
<li><a href='#ccle_info_fine'><p>Cleaned Phenotype Info of CCLE Database for grouping</p></a></li>
<li><a href='#ezcor'><p>Run Correlation between Two Variables and Support Group by a Variable</p></a></li>
<li><a href='#ezcor_batch'><p>Run correlation between two variables in a batch mode and support group by a variable</p></a></li>
<li><a href='#ezcor_partial_cor'><p>Run partial correlation</p></a></li>
<li><a href='#get_ccle_cn_value'><p>Fetch Identifier Value from Pan-cancer Dataset</p></a></li>
<li><a href='#keep_cat_cols'><p>Keep Only Columns Used for Sample Selection</p></a></li>
<li><a href='#load_data'><p>Load Dataset Provided by This Package</p></a></li>
<li><a href='#mol_quick_analysis'><p>Quick molecule analysis and report generation</p></a></li>
<li><a href='#pcawg_info'><p>Phenotype Info of PCAWG Database</p></a></li>
<li><a href='#pcawg_info_fine'><p>Cleaned Phenotype Info of PCAWG Database for grouping</p></a></li>
<li><a href='#pcawg_purity'><p>Purity Data of PCAWG</p></a></li>
<li><a href='#query_general_value'><p>download data for shiny general analysis</p></a></li>
<li><a href='#query_molecule_value'><p>Get Molecule or Signature Data Values from Dense (Genomic) Matrix Dataset of UCSC Xena Data Hubs</p></a></li>
<li><a href='#query_pancan_value'><p>Query Single Identifier or Signature Value from Pan-cancer Database</p></a></li>
<li><a href='#query_tcga_group'><p>Group TPC samples by build-in or custom phenotype and support filtering or merging operations</p></a></li>
<li><a href='#query_toil_value_df'><p>Obtain ToilHub Info for Single Molecule</p></a></li>
<li><a href='#tcga+20survival+20analysis'><p>TCGA Survival Analysis</p></a></li>
<li><a href='#tcga_clinical'><p>Toil Hub: TCGA Clinical Data</p></a></li>
<li><a href='#tcga_clinical_fine'><p>Toil Hub: Cleaned TCGA Clinical Data for grouping</p></a></li>
<li><a href='#tcga_genome_instability'><p>TCGA: Genome Instability Data</p></a></li>
<li><a href='#tcga_gtex'><p>Toil Hub: Merged TCGA GTEx Selected Phenotype</p></a></li>
<li><a href='#tcga_purity'><p>TCGA: Purity Data</p></a></li>
<li><a href='#tcga_subtypes'><p>TCGA Subtype Data</p></a></li>
<li><a href='#tcga_surv'><p>Toil Hub: TCGA Survival Data</p></a></li>
<li><a href='#tcga_tmb'><p>TCGA: TMB (Tumor Mutation Burden) Data</p></a></li>
<li><a href='#TCGA.organ'><p>TCGA: Organ Data</p></a></li>
<li><a href='#toil_info'><p>Toil Hub: TCGA TARGET GTEX Selected Phenotype</p></a></li>
<li><a href='#vis_ccle_gene_cor'><p>Visualize CCLE Gene Expression Correlation</p></a></li>
<li><a href='#vis_ccle_tpm'><p>Visualize CCLE Gene Expression</p></a></li>
<li><a href='#vis_dim_dist'><p>Visualize the distribution difference of samples after dimension reduction analysis</p></a></li>
<li><a href='#vis_gene_cor'><p>Visualize Gene-Gene Correlation in TCGA</p></a></li>
<li><a href='#vis_gene_cor_cancer'><p>Visualize Gene-Gene Correlation in a TCGA Cancer Type</p></a></li>
<li><a href='#vis_gene_drug_response_asso'><p>Visualize Gene and Drug-Target Association with CCLE Data</p></a></li>
<li><a href='#vis_gene_drug_response_diff'><p>Visualize Gene and Drug Response Difference with CCLE Data</p></a></li>
<li><a href='#vis_gene_immune_cor'><p>Heatmap for Correlation between Gene and Immune Signatures</p></a></li>
<li><a href='#vis_gene_msi_cor'><p>Visualize Correlation between Gene and MSI (Microsatellite instability)</p></a></li>
<li><a href='#vis_gene_pw_cor'><p>Visualize Correlation between Gene and Pathway signature Score</p></a></li>
<li><a href='#vis_gene_stemness_cor'><p>Visualize Correlation between Gene and Tumor Stemness</p></a></li>
<li><a href='#vis_gene_TIL_cor'><p>Heatmap for Correlation between Gene and Tumor Immune Infiltration (TIL)</p></a></li>
<li><a href='#vis_gene_tmb_cor'><p>Visualize Correlation between Gene and TMB (Tumor Mutation Burden)</p></a></li>
<li><a href='#vis_identifier_cor'><p>Visualize Identifier-Identifier Correlation</p></a></li>
<li><a href='#vis_identifier_dim_dist'><p>Visualize the distribution difference of samples after Molecule Identifier dimension reduction analysis</p></a></li>
<li><a href='#vis_identifier_grp_comparison'><p>Visualize Comparison of an Molecule Identifier between Groups</p></a></li>
<li><a href='#vis_identifier_grp_surv'><p>Visualize Identifier Group Survival Difference</p></a></li>
<li><a href='#vis_identifier_multi_cor'><p>Visualize Correlation for Multiple Identifiers</p></a></li>
<li><a href='#vis_pancan_anatomy'><p>Visualize Single Gene Expression in Anatomy Location</p></a></li>
<li><a href='#vis_pcawg_dist'><p>Visualize molecular profile in PCAWG</p></a></li>
<li><a href='#vis_pcawg_gene_cor'><p>Visualize Gene-Gene Correlation in TCGA</p></a></li>
<li><a href='#vis_pcawg_unicox_tree'><p>Visualize Single Gene Univariable Cox Result in PCAWG</p></a></li>
<li><a href='#vis_toil_Mut'><p>Visualize molecular profile difference between mutation and wild status of queried gene</p></a></li>
<li><a href='#vis_toil_Mut_cancer'><p>Visualize molecular profile difference between mutation and wild status of queried gene in Single Cancer Type</p></a></li>
<li><a href='#vis_toil_TvsN'><p>Visualize Pan-cancer TPM (tumor (TCGA) vs Normal (TCGA &amp; GTEx))</p></a></li>
<li><a href='#vis_toil_TvsN_cancer'><p>Visualize Gene TPM in Single Cancer Type (Tumor (TCGA) vs Normal (TCGA &amp; GTEx))</p></a></li>
<li><a href='#vis_unicox_tree'><p>Visualize Single Gene Univariable Cox Result from Toil Data Hub</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Interactive Analysis of UCSC Xena Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shixiang Wang &lt;w_shixiang@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions and a Shiny application for downloading,
    analyzing and visualizing datasets from UCSC Xena
    (<a href="http://xena.ucsc.edu/">http://xena.ucsc.edu/</a>), which is a collection of UCSC-hosted public
    databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/openbiox/UCSCXenaShiny">https://github.com/openbiox/UCSCXenaShiny</a>,
<a href="https://openbiox.github.io/UCSCXenaShiny/">https://openbiox.github.io/UCSCXenaShiny/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/openbiox/UCSCXenaShiny/issues">https://github.com/openbiox/UCSCXenaShiny/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>digest, dplyr (&ge; 0.8.3), ezcox, forcats, ggplot2 (&ge; 3.2.0),
ggpubr (&ge; 0.2), httr, magrittr (&ge; 1.5), ppcor, psych, purrr,
shiny (&ge; 1.3.2), stats, stringr, tibble (&ge; 2.1.3), tidyr,
UCSCXenaTools, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr (&ge; 3.2.1), cowplot, DT (&ge; 0.5), furrr, future,
ggrepel, ggstatsplot, knitr, pacman, plotly, plyr, RColorBrewer
(&ge; 1.1.2), rmarkdown, Rtsne, scales, survival, survminer,
testthat (&ge; 2.0.1), umap,</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-14 01:44:05 UTC; wsx</td>
</tr>
<tr>
<td>Author:</td>
<td>Shixiang Wang <a href="https://orcid.org/0000-0001-9855-7357"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Shensuo Li [aut],
  Yi Xiong <a href="https://orcid.org/0000-0002-4370-9824"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Longfei Zhao <a href="https://orcid.org/0000-0002-6277-0137"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Kai Gu <a href="https://orcid.org/0000-0002-0177-0774"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Yin Li [aut],
  Fei Zhao [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-14 03:20:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='UCSCXenaShiny-package'>Xena Shiny App</h2><span id='topic+UCSCXenaShiny'></span><span id='topic+UCSCXenaShiny-package'></span>

<h3>Description</h3>

<p>A Shiny App for UCSC Xena Data Hubs. See <a href="https://github.com/openbiox/UCSCXenaShiny">https://github.com/openbiox/UCSCXenaShiny</a> for details.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a> (<a href="https://orcid.org/0000-0001-9855-7357">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Shensuo Li <a href="mailto:lishensuo@163.com">lishensuo@163.com</a>
</p>
</li>
<li><p> Yi Xiong <a href="mailto:neuroxiong@openbiox.com">neuroxiong@openbiox.com</a> (<a href="https://orcid.org/0000-0002-4370-9824">ORCID</a>)
</p>
</li>
<li><p> Longfei Zhao <a href="mailto:longfei8533@126.com">longfei8533@126.com</a> (<a href="https://orcid.org/0000-0002-6277-0137">ORCID</a>)
</p>
</li>
<li><p> Kai Gu <a href="mailto:gukai1212@163.com">gukai1212@163.com</a> (<a href="https://orcid.org/0000-0002-0177-0774">ORCID</a>)
</p>
</li>
<li><p> Yin Li <a href="mailto:yinli18@fudan.edu.cn">yinli18@fudan.edu.cn</a>
</p>
</li>
<li><p> Fei Zhao <a href="mailto:zhaofei415@mails.ucas.edu.cn">zhaofei415@mails.ucas.edu.cn</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/openbiox/UCSCXenaShiny">https://github.com/openbiox/UCSCXenaShiny</a>
</p>
</li>
<li> <p><a href="https://openbiox.github.io/UCSCXenaShiny/">https://openbiox.github.io/UCSCXenaShiny/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/openbiox/UCSCXenaShiny/issues">https://github.com/openbiox/UCSCXenaShiny/issues</a>
</p>
</li></ul>


<hr>
<h2 id='.opt_pancan'>A default setting for pan-cancer studies</h2><span id='topic+.opt_pancan'></span>

<h3>Description</h3>

<p>A default setting for pan-cancer studies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.opt_pancan
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 16.
</p>

<hr>
<h2 id='+25+26gt+3B+25'>Pipe Operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic++25+3E+25">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='analyze_gene_drug_response_asso'>Analyze Association between Gene (Signature) and Drug Response with CCLE Data</h2><span id='topic+analyze_gene_drug_response_asso'></span>

<h3>Description</h3>

<p>Analyze partial correlation of gene-drug association after
controlling for tissue average expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyze_gene_drug_response_asso(gene_list, combine = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analyze_gene_drug_response_asso_+3A_gene_list">gene_list</code></td>
<td>
<p>a gene symbol list.</p>
</td></tr>
<tr><td><code id="analyze_gene_drug_response_asso_+3A_combine">combine</code></td>
<td>
<p>if <code>TRUE</code>, combine the expression of gene list as
a gene signature.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code>
</p>

<ul>
<li><p> If <code>combine</code> is <code>TRUE</code>, genes are combined as <code>signature</code>.
</p>
</li>
<li> <p><code>mean.diff</code> and <code>median.diff</code> indicate mean and median of
normalized expression difference between High IC50 cells and Low IC50 cells.
The cutoff between High and Low are median IC50.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
analyze_gene_drug_response_asso("TP53")
analyze_gene_drug_response_asso(c("TP53", "KRAS"))
analyze_gene_drug_response_asso(c("TP53", "KRAS"), combine = TRUE)

# Visualization
vis_gene_drug_response_asso("TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='analyze_gene_drug_response_diff'>Analyze Difference of Drug Response (IC50 Value (uM)) between Gene (Signature) High and Low Expression with CCLE Data</h2><span id='topic+analyze_gene_drug_response_diff'></span>

<h3>Description</h3>

<p>Analyze Difference of Drug Response (IC50 Value (uM)) between Gene (Signature) High and Low Expression with CCLE Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyze_gene_drug_response_diff(
  gene_list,
  drug = "ALL",
  tissue = "ALL",
  combine = FALSE,
  cutpoint = c(50, 50)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analyze_gene_drug_response_diff_+3A_gene_list">gene_list</code></td>
<td>
<p>a gene symbol list.</p>
</td></tr>
<tr><td><code id="analyze_gene_drug_response_diff_+3A_drug">drug</code></td>
<td>
<p>a drug name. Check examples.</p>
</td></tr>
<tr><td><code id="analyze_gene_drug_response_diff_+3A_tissue">tissue</code></td>
<td>
<p>a tissue name. Check examples.</p>
</td></tr>
<tr><td><code id="analyze_gene_drug_response_diff_+3A_combine">combine</code></td>
<td>
<p>if <code>TRUE</code>, combine the expression of gene list as
a gene signature.</p>
</td></tr>
<tr><td><code id="analyze_gene_drug_response_diff_+3A_cutpoint">cutpoint</code></td>
<td>
<p>cut point (in percent) for High and Low group, default is <code>c(50, 50)</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tissue_list &lt;- c(
  "prostate", "central_nervous_system", "urinary_tract", "haematopoietic_and_lymphoid_tissue",
  "kidney", "thyroid", "soft_tissue", "skin", "salivary_gland",
  "ovary", "lung", "bone", "endometrium", "pancreas", "breast",
  "large_intestine", "upper_aerodigestive_tract", "autonomic_ganglia",
  "stomach", "liver", "biliary_tract", "pleura", "oesophagus"
)

drug_list &lt;- c(
  "AEW541", "Nilotinib", "17-AAG", "PHA-665752", "Lapatinib",
  "Nutlin-3", "AZD0530", "PF2341066", "L-685458", "ZD-6474", "Panobinostat",
  "Sorafenib", "Irinotecan", "Topotecan", "LBW242", "PD-0325901",
  "PD-0332991", "Paclitaxel", "AZD6244", "PLX4720", "RAF265", "TAE684",
  "TKI258", "Erlotinib"
)

target_list &lt;- c(
  "IGF1R", "ABL", "HSP90", "c-MET", "EGFR", "MDM2", "GS", "HDAC",
  "RTK", "TOP1", "XIAP", "MEK", "CDK4", "TUBB1", "RAF", "ALK", "FGFR"
)
## Not run: 
analyze_gene_drug_response_diff("TP53")
analyze_gene_drug_response_diff(c("TP53", "KRAS"), drug = "AEW541")
analyze_gene_drug_response_diff(c("TP53", "KRAS"),
  tissue = "kidney",
  combine = TRUE
)

# Visualization
vis_gene_drug_response_diff("TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='app_run'>Run UCSC Xena Shiny App</h2><span id='topic+app_run'></span>

<h3>Description</h3>

<p>Run UCSC Xena Shiny App
</p>


<h3>Usage</h3>

<pre><code class='language-R'>app_run(runMode = "client", port = getOption("shiny.port"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="app_run_+3A_runmode">runMode</code></td>
<td>
<p>default is 'client' for personal user, set it to 'server' for running on server.</p>
</td></tr>
<tr><td><code id="app_run_+3A_port">port</code></td>
<td>
<p>The TCP port that the application should listen on. If the
<code>port</code> is not specified, and the <code>shiny.port</code> option is set (with
<code>options(shiny.port = XX)</code>), then that port will be used. Otherwise,
use a random port between 3000:8000, excluding ports that are blocked
by Google Chrome for being considered unsafe: 3659, 4045, 5060,
5061, 6000, 6566, 6665:6669 and 6697. Up to twenty random
ports will be tried.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
app_run()

## End(Not run)
</code></pre>

<hr>
<h2 id='available_hosts'>Show Available Hosts</h2><span id='topic+available_hosts'></span>

<h3>Description</h3>

<p>Show Available Hosts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_hosts()
</code></pre>


<h3>Value</h3>

<p>hosts
</p>


<h3>Examples</h3>

<pre><code class='language-R'>available_hosts()
</code></pre>

<hr>
<h2 id='ccle_absolute'>ABSOLUTE Result of CCLE Database</h2><span id='topic+ccle_absolute'></span>

<h3>Description</h3>

<p>ABSOLUTE Result of CCLE Database
</p>


<h3>Format</h3>

<p>A <code>data.frame</code>
</p>


<h3>Source</h3>

<p>see &quot;data_source&quot; attribute.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("ccle_absolute")
</code></pre>

<hr>
<h2 id='ccle_info'>Phenotype Info of CCLE Database</h2><span id='topic+ccle_info'></span>

<h3>Description</h3>

<p>Phenotype Info of CCLE Database
</p>


<h3>Format</h3>

<p>A <code>data.frame</code>
</p>


<h3>Source</h3>

<p>UCSC Xena.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("ccle_info")
</code></pre>

<hr>
<h2 id='ccle_info_fine'>Cleaned Phenotype Info of CCLE Database for grouping</h2><span id='topic+ccle_info_fine'></span>

<h3>Description</h3>

<p>Cleaned Phenotype Info of CCLE Database for grouping
</p>


<h3>Format</h3>

<p>A <code>data.frame</code>
</p>


<h3>Source</h3>

<p>UCSC Xena.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("ccle_info_fine")
</code></pre>

<hr>
<h2 id='ezcor'>Run Correlation between Two Variables and Support Group by a Variable</h2><span id='topic+ezcor'></span>

<h3>Description</h3>

<p>Run Correlation between Two Variables and Support Group by a Variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ezcor(
  data = NULL,
  split = FALSE,
  split_var = NULL,
  var1 = NULL,
  var2 = NULL,
  cor_method = "pearson",
  adjust_method = "none",
  use = "complete",
  sig_label = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ezcor_+3A_data">data</code></td>
<td>
<p>a <code>data.frame</code> containing variables</p>
</td></tr>
<tr><td><code id="ezcor_+3A_split">split</code></td>
<td>
<p>whether perform correlation grouped by a variable, default is 'FALSE'</p>
</td></tr>
<tr><td><code id="ezcor_+3A_split_var">split_var</code></td>
<td>
<p>a <code>character</code>, the group variable</p>
</td></tr>
<tr><td><code id="ezcor_+3A_var1">var1</code></td>
<td>
<p>a character, the first variable in correlation</p>
</td></tr>
<tr><td><code id="ezcor_+3A_var2">var2</code></td>
<td>
<p>a character, the second variable in correlation</p>
</td></tr>
<tr><td><code id="ezcor_+3A_cor_method">cor_method</code></td>
<td>
<p>method=&quot;pearson&quot; is the default value. The alternatives to be passed to cor are &quot;spearman&quot; and &quot;kendall&quot;</p>
</td></tr>
<tr><td><code id="ezcor_+3A_adjust_method">adjust_method</code></td>
<td>
<p>What adjustment for multiple tests should be used? (&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;)</p>
</td></tr>
<tr><td><code id="ezcor_+3A_use">use</code></td>
<td>
<p>use=&quot;pairwise&quot; will do pairwise deletion of cases. use=&quot;complete&quot; will select just complete cases</p>
</td></tr>
<tr><td><code id="ezcor_+3A_sig_label">sig_label</code></td>
<td>
<p>whether add symbal of significance. P &lt; 0.001,<code style="white-space: pre;">&#8288;***&#8288;</code>; P &lt; 0.01,<code style="white-space: pre;">&#8288;**&#8288;</code>; P &lt; 0.05,<code>*</code>; P &gt;=0.05,&quot;&quot;</p>
</td></tr>
<tr><td><code id="ezcor_+3A_verbose">verbose</code></td>
<td>
<p>if <code>TRUE</code>, print extra info.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code>
</p>


<h3>Author(s)</h3>

<p>Yi Xiong
</p>

<hr>
<h2 id='ezcor_batch'>Run correlation between two variables in a batch mode and support group by a variable</h2><span id='topic+ezcor_batch'></span>

<h3>Description</h3>

<p>Run correlation between two variables in a batch mode and support group by a variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ezcor_batch(
  data,
  var1,
  var2,
  split = FALSE,
  split_var = NULL,
  cor_method = "pearson",
  adjust_method = "none",
  use = "complete",
  sig_label = TRUE,
  parallel = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ezcor_batch_+3A_data">data</code></td>
<td>
<p>a <code>data.frame</code> containing variables</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_var1">var1</code></td>
<td>
<p>a character, the first variable in correlation</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_var2">var2</code></td>
<td>
<p>a character, the second variable in correlation</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_split">split</code></td>
<td>
<p>whether perform correlation grouped by a variable, default is 'FALSE'</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_split_var">split_var</code></td>
<td>
<p>a <code>character</code>, the group variable</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_cor_method">cor_method</code></td>
<td>
<p>method=&quot;pearson&quot; is the default value. The alternatives to be passed to cor are &quot;spearman&quot; and &quot;kendall&quot;</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_adjust_method">adjust_method</code></td>
<td>
<p>What adjustment for multiple tests should be used? (&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;)</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_use">use</code></td>
<td>
<p>use=&quot;pairwise&quot; will do pairwise deletion of cases. use=&quot;complete&quot; will select just complete cases</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_sig_label">sig_label</code></td>
<td>
<p>whether add symbal of significance. P &lt; 0.001,<code style="white-space: pre;">&#8288;***&#8288;</code>; P &lt; 0.01,<code style="white-space: pre;">&#8288;**&#8288;</code>; P &lt; 0.05,<code>*</code>; P &gt;=0.05,&quot;&quot;</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_parallel">parallel</code></td>
<td>
<p>if <code>TRUE</code>, do parallel computation by <strong>furrr</strong> package.</p>
</td></tr>
<tr><td><code id="ezcor_batch_+3A_verbose">verbose</code></td>
<td>
<p>if <code>TRUE</code>, print extra info.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code>
</p>


<h3>Author(s)</h3>

<p>Yi Xiong, Shixiang Wang
</p>

<hr>
<h2 id='ezcor_partial_cor'>Run partial correlation</h2><span id='topic+ezcor_partial_cor'></span>

<h3>Description</h3>

<p>Run partial correlation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ezcor_partial_cor(
  data = NULL,
  split = FALSE,
  split_var = NULL,
  var1 = NULL,
  var2 = NULL,
  var3 = NULL,
  cor_method = "pearson",
  sig_label = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ezcor_partial_cor_+3A_data">data</code></td>
<td>
<p>a <code>data.frame</code> containing variables</p>
</td></tr>
<tr><td><code id="ezcor_partial_cor_+3A_split">split</code></td>
<td>
<p>whether perform correlation grouped by a variable, default is 'FALSE'</p>
</td></tr>
<tr><td><code id="ezcor_partial_cor_+3A_split_var">split_var</code></td>
<td>
<p>a <code>character</code>, the group variable</p>
</td></tr>
<tr><td><code id="ezcor_partial_cor_+3A_var1">var1</code></td>
<td>
<p>a <code>character</code>, the first variable in correlation</p>
</td></tr>
<tr><td><code id="ezcor_partial_cor_+3A_var2">var2</code></td>
<td>
<p>a <code>character</code>, the second variable in correlation</p>
</td></tr>
<tr><td><code id="ezcor_partial_cor_+3A_var3">var3</code></td>
<td>
<p>a <code>character</code> or <code style="white-space: pre;">&#8288;character vector&#8288;</code>, the third variable in correlation</p>
</td></tr>
<tr><td><code id="ezcor_partial_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>method=&quot;pearson&quot; is the default value. The alternatives to be passed to cor are &quot;spearman&quot; and &quot;kendall&quot;</p>
</td></tr>
<tr><td><code id="ezcor_partial_cor_+3A_sig_label">sig_label</code></td>
<td>
<p>whether add symbal of significance. P &lt; 0.001,&quot;<em><strong>&quot;; P &lt; 0.01,&quot;</strong>&quot;; P &lt; 0.05,&quot;</em>&quot;; P &gt;=0.05,&quot;&quot;</p>
</td></tr>
<tr><td><code id="ezcor_partial_cor_+3A_...">...</code></td>
<td>
<p>other arguments passed to methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code>
</p>


<h3>Author(s)</h3>

<p>Yi Xiong
</p>


<h3>See Also</h3>

<p><code><a href="ppcor.html#topic+pcor.test">ppcor::pcor.test()</a></code> which this function wraps.
</p>

<hr>
<h2 id='get_ccle_cn_value'>Fetch Identifier Value from Pan-cancer Dataset</h2><span id='topic+get_ccle_cn_value'></span><span id='topic+get_ccle_gene_value'></span><span id='topic+get_ccle_protein_value'></span><span id='topic+get_ccle_mutation_status'></span><span id='topic+get_pancan_value'></span><span id='topic+get_pancan_gene_value'></span><span id='topic+get_pancan_transcript_value'></span><span id='topic+get_pancan_protein_value'></span><span id='topic+get_pancan_mutation_status'></span><span id='topic+get_pancan_cn_value'></span><span id='topic+get_pancan_methylation_value'></span><span id='topic+get_pancan_miRNA_value'></span><span id='topic+get_pcawg_gene_value'></span><span id='topic+get_pcawg_fusion_value'></span><span id='topic+get_pcawg_promoter_value'></span><span id='topic+get_pcawg_miRNA_value'></span><span id='topic+get_pcawg_APOBEC_mutagenesis_value'></span>

<h3>Description</h3>

<p>Identifier includes gene/probe etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ccle_cn_value(identifier)

get_ccle_gene_value(identifier, norm = c("rpkm", "nc"))

get_ccle_protein_value(identifier)

get_ccle_mutation_status(identifier)

get_pancan_value(
  identifier,
  subtype = NULL,
  dataset = NULL,
  host = available_hosts(),
  samples = NULL,
  ...
)

get_pancan_gene_value(identifier, norm = c("tpm", "fpkm", "nc"))

get_pancan_transcript_value(identifier, norm = c("tpm", "fpkm", "isopct"))

get_pancan_protein_value(identifier)

get_pancan_mutation_status(identifier)

get_pancan_cn_value(identifier, gistic2 = TRUE, use_thresholded_data = FALSE)

get_pancan_methylation_value(
  identifier,
  type = c("450K", "27K"),
  rule_out = NULL,
  aggr = c("NA", "mean", "Q0", "Q25", "Q50", "Q75", "Q100")
)

get_pancan_miRNA_value(identifier)

get_pcawg_gene_value(identifier)

get_pcawg_fusion_value(identifier)

get_pcawg_promoter_value(identifier, type = c("raw", "relative", "outlier"))

get_pcawg_miRNA_value(identifier, norm = c("TMM", "UQ"))

get_pcawg_APOBEC_mutagenesis_value(
  identifier = c("tCa_MutLoad_MinEstimate", "APOBECtCa_enrich", "A3A_or_A3B",
    "APOBEC_tCa_enrich_quartile", "APOBECrtCa_enrich", "APOBECytCa_enrich",
    "APOBECytCa_enrich-APOBECrtCa_enrich", "BH_Fisher_p-value_tCa", "ntca+tgan",
    "rtCa_to_G+rtCa_to_T", "rtca+tgay", "tCa_to_G+tCa_to_T",
    "ytCa_rtCa_BH_Fisher_p-value", "ytCa_rtCa_Fisher_p-value", "ytCa_to_G+ytCa_to_T",
    "ytca+tgar")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ccle_cn_value_+3A_identifier">identifier</code></td>
<td>
<p>a length-1 character representing a gene symbol, ensembl gene id, or probe id.
Gene symbol is highly recommended.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_norm">norm</code></td>
<td>
<p>the normalization method.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_subtype">subtype</code></td>
<td>
<p>a length-1 chracter representing a regular expression for matching
<code>DataSubtype</code> column of <a href="UCSCXenaTools.html#topic+XenaData">UCSCXenaTools::XenaData</a>.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_dataset">dataset</code></td>
<td>
<p>a length-1 chracter representing a regular expression for matching
<code>XenaDatasets</code> of <a href="UCSCXenaTools.html#topic+XenaData">UCSCXenaTools::XenaData</a>.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_host">host</code></td>
<td>
<p>a character vector representing host name(s), e.g. &quot;toilHub&quot;.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_samples">samples</code></td>
<td>
<p>a character vector representing samples want to be returned.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_...">...</code></td>
<td>
<p>other parameters.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_gistic2">gistic2</code></td>
<td>
<p>if <code>TRUE</code> (default), use GISTIC2 data.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_use_thresholded_data">use_thresholded_data</code></td>
<td>
<p>if <code>TRUE</code>, use GISTIC2-thresholded value.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_type">type</code></td>
<td>
<p>methylation type, one of &quot;450K&quot; and &quot;27K&quot;.
for function <code>get_pcawg_promoter_value</code>, it can be one of
&quot;raw&quot;, &quot;relative&quot;, &quot;outlier&quot;.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_rule_out">rule_out</code></td>
<td>
<p>methylation sites to rule out before analyzing.</p>
</td></tr>
<tr><td><code id="get_ccle_cn_value_+3A_aggr">aggr</code></td>
<td>
<p>apporaches to aggregate the methylation data, default is 'NA',
in such case, a mean value is obtained for gene-level methylation.
Allowed value is one of <code>c("NA", "mean", "Q0", "Q25", "Q50", "Q75", "Q100")</code>.
Here, <code>Q50</code> is median.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector or <code>list</code>.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>get_ccle_cn_value()</code>: Fetch copy number value from CCLE dataset
</p>
</li>
<li> <p><code>get_ccle_gene_value()</code>: Fetch gene expression value from CCLE dataset
</p>
</li>
<li> <p><code>get_ccle_protein_value()</code>: Fetch gene protein expression value from CCLE dataset
</p>
</li>
<li> <p><code>get_ccle_mutation_status()</code>: Fetch gene mutation info from CCLE dataset
</p>
</li>
<li> <p><code>get_pancan_value()</code>: Fetch identifier value from pan-cancer dataset
</p>
</li>
<li> <p><code>get_pancan_gene_value()</code>: Fetch gene expression value from pan-cancer dataset
</p>
</li>
<li> <p><code>get_pancan_transcript_value()</code>: Fetch gene transcript expression value from pan-cancer dataset
</p>
</li>
<li> <p><code>get_pancan_protein_value()</code>: Fetch protein expression value from pan-cancer dataset
</p>
</li>
<li> <p><code>get_pancan_mutation_status()</code>: Fetch mutation status value from pan-cancer dataset
</p>
</li>
<li> <p><code>get_pancan_cn_value()</code>: Fetch gene copy number value from pan-cancer dataset processed by GISTIC 2.0
</p>
</li>
<li> <p><code>get_pancan_methylation_value()</code>: Fetch gene expression value from CCLE dataset
</p>
</li>
<li> <p><code>get_pancan_miRNA_value()</code>: Fetch miRNA expression value from pan-cancer dataset
</p>
</li>
<li> <p><code>get_pcawg_gene_value()</code>: Fetch specimen-level gene expression value from PCAWG cohort
</p>
</li>
<li> <p><code>get_pcawg_fusion_value()</code>: Fetch specimen-level gene fusion value from PCAWG cohort
</p>
</li>
<li> <p><code>get_pcawg_promoter_value()</code>: Fetch specimen-level gene promoter activity value from PCAWG cohort
</p>
</li>
<li> <p><code>get_pcawg_miRNA_value()</code>: Fetch specimen-level miRNA value from PCAWG cohort
</p>
</li>
<li> <p><code>get_pcawg_APOBEC_mutagenesis_value()</code>: Fetch specimen-level gene fusion value from PCAWG cohort
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Fetch TP53 expression value from pan-cancer dataset
t1 &lt;- get_pancan_value("TP53",
  dataset = "TcgaTargetGtex_rsem_isoform_tpm",
  host = "toilHub"
)
t2 &lt;- get_pancan_gene_value("TP53")
t3 &lt;- get_pancan_protein_value("AKT")
t4 &lt;- get_pancan_mutation_status("TP53")
t5 &lt;- get_pancan_cn_value("TP53")

## End(Not run)

</code></pre>

<hr>
<h2 id='keep_cat_cols'>Keep Only Columns Used for Sample Selection</h2><span id='topic+keep_cat_cols'></span>

<h3>Description</h3>

<p>Keep Only Columns Used for Sample Selection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>keep_cat_cols(x, keep_sam_cols = TRUE, return_idx = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="keep_cat_cols_+3A_x">x</code></td>
<td>
<p>a <code>data.frame</code> with many columns.</p>
</td></tr>
<tr><td><code id="keep_cat_cols_+3A_keep_sam_cols">keep_sam_cols</code></td>
<td>
<p>if <code>TRUE</code> (default), keep columns with
pattern 'sample', 'patient', etc.</p>
</td></tr>
<tr><td><code id="keep_cat_cols_+3A_return_idx">return_idx</code></td>
<td>
<p>if <code>TRUE</code> (default), return index of 5 (at most) columns, it is useful in Shiny.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> or a <code>list</code>.
</p>

<hr>
<h2 id='load_data'>Load Dataset Provided by This Package</h2><span id='topic+load_data'></span>

<h3>Description</h3>

<p>Load data from builtin or Zenodo.
Option <code>xena.zenodoDir</code> can be used to set default path for storing
extra data from Zenodo, e.g., <code>options(xena.zenodoDir = "/home/xxx/dataset")</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_data(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_data_+3A_name">name</code></td>
<td>
<p>a dataset name. Could be one of
</p>
<p><strong>Builtin datasets</strong>:
</p>

<ul>
<li> <p><code>ccle_absolute</code>: CCLE ABSOLUTE result.
</p>
</li>
<li> <p><code>ccle_info</code>: CCLE information.
</p>
</li>
<li> <p><code>ccle_info_fine</code>: cleaned CCLE information for TPC analysis.
</p>
</li>
<li> <p><code>pcawg_info</code>: PCAWG information.
</p>
</li>
<li> <p><code>pcawg_info_fine</code>: cleaned PCAWG information for TPC analysis.
</p>
</li>
<li> <p><code>pcawg_purity</code>: PCAWG tumor purity, ploidy and WGD data.
</p>
</li>
<li> <p><code>tcga_clinical</code>: TCGA clinical data.
</p>
</li>
<li> <p><code>tcga_clinical_fine</code>: cleaned TCGA information for TPC analysis.
</p>
</li>
<li> <p><code>tcga_genome_instability</code>: TCGA genome instability data.
</p>
</li>
<li> <p><code>tcga_gtex</code>: TCGA and GTEX sample info.
</p>
</li>
<li> <p><code>tcga_purity</code>: TCGA tumor purity data.
</p>
</li>
<li> <p><code>tcga_subtypes</code>: TCGA subtypes data.
</p>
</li>
<li> <p><code>tcga_surv</code>: TCGA survival data.
</p>
</li>
<li> <p><code>TCGA.organ</code>: TCGA organ data.
</p>
</li>
<li> <p><code>toil_info</code>: Toil hub information.
</p>
</li></ul>

<p><strong>Remote datasets stored in <a href="https://zenodo.org/record/10778172/">Zenodo</a></strong>:
</p>

<ul>
<li> <p><code>pcawg_promoter_id</code>: PCAWG promoter identifiers.
</p>
</li>
<li> <p><code>transcript_identifier</code>: Common transcript identifiers.
</p>
</li>
<li> <p><code>ccle_expr_and_drug_response</code>: CCLE expression and drug response data.
</p>
</li>
<li> <p><code>ccle_drug_response_extend</code>: CCLE drug response extended data.
</p>
</li>
<li> <p><code>pancan_MSI</code>: Pan-cancer MSI data.
</p>
</li>
<li> <p><code>tcga_chr_alteration</code>: TCGA chromosome alteration data.
</p>
</li>
<li> <p><code>tcga_MSI</code>: TCGA MSI data.
</p>
</li>
<li> <p><code>tcga_pan_immune_signature</code>: TCGA pan-cancer immune signature.
</p>
</li>
<li> <p><code>tcga_stemness</code>: TCGA tumor stemness data.
</p>
</li>
<li> <p><code>tcga_TIL</code>: TCGA TIL data.
</p>
</li>
<li> <p><code>tcga_PW</code>: ssGSEA scores of HALLMARK, KEGG, IOBR terms for TCGA samples.
</p>
</li>
<li> <p><code>tcga_PW_meta</code>: metadata annotation for HALLMARK, KEGG, IOBR terms.
</p>
</li>
<li> <p><code>tcga_tmb</code>: TCGA TMB data.
</p>
</li>
<li> <p><code>tcga_armcalls</code>: TCGA arm alteration calls and Aneuploidy data.
</p>
</li>
<li> <p><code>tcga_dna_repair</code>: TCGA DNA repair data.
</p>
</li>
<li> <p><code>pancancer_conserved_immune_subtype</code>: Pan-cancer conserved immune subtypes.
</p>
</li>
<li> <p><code>pcawg_TIL</code>: PCAWG TIL data.
</p>
</li>
<li> <p><code>pcawg_PW</code>: ssGSEA scores of HALLMARK, KEGG, IOBR terms for PCAWG samples.
</p>
</li>
<li><p> ...
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataset, typically a <code>data.frame</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data1 &lt;- load_data("tcga_surv")
data1

data2 &lt;- load_data("tcga_armcalls")
data2

</code></pre>

<hr>
<h2 id='mol_quick_analysis'>Quick molecule analysis and report generation</h2><span id='topic+mol_quick_analysis'></span>

<h3>Description</h3>

<p>Quick molecule analysis and report generation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mol_quick_analysis(molecule, data_type, out_dir = ".", out_report = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mol_quick_analysis_+3A_molecule">molecule</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="mol_quick_analysis_+3A_data_type">data_type</code></td>
<td>
<p>data type. Can be one of &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;,
&quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="mol_quick_analysis_+3A_out_dir">out_dir</code></td>
<td>
<p>path to save analysis result and report, default is '.'</p>
</td></tr>
<tr><td><code id="mol_quick_analysis_+3A_out_report">out_report</code></td>
<td>
<p>logical value wheather to generate html report</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list.
</p>

<hr>
<h2 id='pcawg_info'>Phenotype Info of PCAWG Database</h2><span id='topic+pcawg_info'></span>

<h3>Description</h3>

<p>Phenotype Info of PCAWG Database
</p>


<h3>Format</h3>

<p>A <code>data.frame</code>
</p>


<h3>Source</h3>

<p>UCSC Xena.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pcawg_info")
</code></pre>

<hr>
<h2 id='pcawg_info_fine'>Cleaned Phenotype Info of PCAWG Database for grouping</h2><span id='topic+pcawg_info_fine'></span>

<h3>Description</h3>

<p>Cleaned Phenotype Info of PCAWG Database for grouping
</p>


<h3>Format</h3>

<p>A <code>data.frame</code>
</p>


<h3>Source</h3>

<p>UCSC Xena.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pcawg_info_fine")
</code></pre>

<hr>
<h2 id='pcawg_purity'>Purity Data of PCAWG</h2><span id='topic+pcawg_purity'></span>

<h3>Description</h3>

<p>Purity Data of PCAWG
</p>


<h3>Format</h3>

<p>A <code>data.frame</code>
</p>


<h3>Source</h3>

<p>UCSC Xena.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pcawg_purity")
</code></pre>

<hr>
<h2 id='query_general_value'>download data for shiny general analysis</h2><span id='topic+query_general_value'></span>

<h3>Description</h3>

<p>download data for shiny general analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_general_value(
  L1,
  L2,
  L3,
  database = c("toil", "pcawg", "ccle"),
  index_value,
  immune_value,
  pathway_value,
  clinical_value,
  opt_pancan = NULL,
  custom_metadata = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="query_general_value_+3A_l1">L1</code></td>
<td>
<p>level 1  main datatype</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_l2">L2</code></td>
<td>
<p>level 2  sub datatype</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_l3">L3</code></td>
<td>
<p>level 3  identifier</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_database">database</code></td>
<td>
<p>one of c(&quot;toil&quot;,&quot;pcawg&quot;,&quot;ccle&quot;)</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_index_value">index_value</code></td>
<td>
<p>tumor index list</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_immune_value">immune_value</code></td>
<td>
<p>tumor immune infiltration list</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_pathway_value">pathway_value</code></td>
<td>
<p>pathway activity list</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_clinical_value">clinical_value</code></td>
<td>
<p>clinical data.frame</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>molecular datasets parameters</p>
</td></tr>
<tr><td><code id="query_general_value_+3A_custom_metadata">custom_metadata</code></td>
<td>
<p>user customized metadata</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
general_value_id = UCSCXenaShiny:::query_general_id()
tcga_value_option = general_value_id[["value"]][[1]]
tcga_index_value = tcga_value_option[["Tumor index"]]
tcga_immune_value = tcga_value_option[["Immune Infiltration"]]
tcga_pathway_value = tcga_value_option[["Pathway activity"]]
tcga_phenotype_value = tcga_value_option[["Phenotype data"]]
clinical_phe = tcga_phenotype_value[["Clinical Phenotype"]]
x_data = UCSCXenaShiny:::query_general_value(
           "Molecular profile", "mRNA Expression", "TP53", "toil",
           tcga_index_value, tcga_immune_value, tcga_pathway_value, 
           clinical_phe)
                                             
y_data = UCSCXenaShiny:::query_general_value(
           "Immune Infiltration", "CIBERSORT", "Monocyte", "toil",
           tcga_index_value, tcga_immune_value, tcga_pathway_value, 
           clinical_phe)

## End(Not run)
</code></pre>

<hr>
<h2 id='query_molecule_value'>Get Molecule or Signature Data Values from Dense (Genomic) Matrix Dataset of UCSC Xena Data Hubs</h2><span id='topic+query_molecule_value'></span>

<h3>Description</h3>

<p>Get Molecule or Signature Data Values from Dense (Genomic) Matrix Dataset of UCSC Xena Data Hubs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_molecule_value(dataset, molecule, host = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="query_molecule_value_+3A_dataset">dataset</code></td>
<td>
<p>a UCSC Xena dataset in dense matrix format (rows are features
(e.g., gene, cell line) and columns are samples).</p>
</td></tr>
<tr><td><code id="query_molecule_value_+3A_molecule">molecule</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).
<strong>NOTE</strong>, when a signature is specified, a space must exist in the input.</p>
</td></tr>
<tr><td><code id="query_molecule_value_+3A_host">host</code></td>
<td>
<p>a UCSC Xena host, default is <code>NULL</code>, auto-detect from the dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># What does dense matrix mean?
table(UCSCXenaTools::XenaData$Type)
# It is a the UCSC Xena dataset with "Type" equals to "genomicMatrix"
## Not run: 
dataset &lt;- "ccle/CCLE_copynumber_byGene_2013-12-03"
x &lt;- query_molecule_value(dataset, "TP53")
head(x)

signature &lt;- "TP53 + 2*KRAS - 1.3*PTEN" # a space must exist in the string
y &lt;- query_molecule_value(dataset, signature)
head(y)

## End(Not run)
</code></pre>

<hr>
<h2 id='query_pancan_value'>Query Single Identifier or Signature Value from Pan-cancer Database</h2><span id='topic+query_pancan_value'></span>

<h3>Description</h3>

<p>Query Single Identifier or Signature Value from Pan-cancer Database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_pancan_value(
  molecule,
  data_type = c("mRNA", "transcript", "protein", "mutation", "cnv", "methylation",
    "miRNA", "fusion", "promoter", "APOBEC"),
  database = c("toil", "ccle", "pcawg"),
  reset_id = NULL,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="query_pancan_value_+3A_molecule">molecule</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="query_pancan_value_+3A_data_type">data_type</code></td>
<td>
<p>data type. Can be one of &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;,
&quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="query_pancan_value_+3A_database">database</code></td>
<td>
<p>database, either 'toil' for TCGA TARGET GTEx, or 'ccle' for
CCLE.</p>
</td></tr>
<tr><td><code id="query_pancan_value_+3A_reset_id">reset_id</code></td>
<td>
<p>if not <code>NULL</code>, set the specified variable at parent frame to &quot;Signature&quot;.</p>
</td></tr>
<tr><td><code id="query_pancan_value_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>other extra parameters passing to the underlying functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>query_pancan_value()</code> provide convenient interface to download multi-omics
data from 3 databases by specifying one or several canonical datasets. It is
derived from <code>query_pancan_value()</code> and support query for genomic signature.
To query comprehensive datasets that UCSCXena supports, users can check
<code>UCSCXenaTools::XenaData</code> and use <code>get_pancan_value()</code> directly.
</p>
<p>Option <code>opt_pancan</code> is a nested list and allow to adjust the downloading details.
For now, only <code>cnv(toil)</code>,<code>methylation(toil)</code>,<code>miRNA(toil)</code>,<code>miRNA(pcawg)</code>,<code>promoter(pcawg)</code>
support optional parameters. The default set is <code>.opt_pancan</code> and we check meanings of sublist(parameters)
through the following relationship.
</p>


<h3>Value</h3>

<p>a list.
</p>


<h3>&quot;toil&quot; database</h3>


<ol>
<li><p> mRNA&ndash;<code>get_pancan_gene_value()</code>
</p>
</li>
<li><p> transcript&ndash;<code>get_pancan_transcript_value()</code>
</p>
</li>
<li><p> protein&ndash;<code>get_pancan_protein_value()</code>
</p>
</li>
<li><p> mutation&ndash;<code>get_pancan_mutation_status()</code>
</p>
</li>
<li><p> cnv&ndash;<code>get_pancan_cn_value()</code>
</p>
</li>
<li><p> methylation&ndash;<code>get_pancan_methylation_value()</code>
</p>
</li>
<li><p> miRNA&ndash;<code>get_pancan_miRNA_value()</code>
</p>
</li></ol>



<h3>&quot;ccle&quot; database</h3>


<ol>
<li><p> mRNA&ndash;<code>get_ccle_gene_value()</code>
</p>
</li>
<li><p> protein&ndash;<code>get_ccle_protein_value()</code>
</p>
</li>
<li><p> mutation&ndash;<code>get_ccle_mutation_status()</code>
</p>
</li>
<li><p> cnv&ndash;<code>get_ccle_cn_value()</code>
</p>
</li></ol>



<h3>&quot;pcawg&quot; database</h3>


<ol>
<li><p> mRNA&ndash;<code>get_pcawg_gene_value()</code>
</p>
</li>
<li><p> miRNA&ndash;<code>get_pcawg_miRNA_value()</code>
</p>
</li>
<li><p> promoter&ndash;<code>get_pcawg_promoter_value()</code>
</p>
</li>
<li><p> fusion&ndash;<code>get_pcawg_fusion_value()</code>
</p>
</li>
<li><p> APOBEC&ndash;<code>get_pcawg_APOBEC_mutagenesis_value()</code>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
query_pancan_value("KRAS")
query_pancan_value("KRAS", database = "ccle")
query_pancan_value("KRAS", database = "pcawg")


query_pancan_value("ENSG00000000419",
  database = "pcawg",
  data_type = "fusion"
) # gene symbol also work

.opt_pancan

opt_pancan = list(toil_cnv = list(use_thresholded_data = FALSE))
query_pancan_value("PTEN",data_type = "cnv", database = "toil", opt_pancan = opt_pancan)


opt_pancan = list(toil_methylation = list(type = "450K",rule_out = "cg21115430", aggr = "Q25"))
query_pancan_value("PTEN",data_type = "methylation", database = "toil", opt_pancan = opt_pancan)


## End(Not run)
</code></pre>

<hr>
<h2 id='query_tcga_group'>Group TPC samples by build-in or custom phenotype and support filtering or merging operations</h2><span id='topic+query_tcga_group'></span>

<h3>Description</h3>

<p>Group TPC samples by build-in or custom phenotype and support filtering or merging operations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_tcga_group(
  database = c("toil", "pcawg", "ccle"),
  cancer = NULL,
  custom = NULL,
  group = "Gender",
  filter_by = NULL,
  filter_id = NULL,
  merge_by = NULL,
  merge_quantile = FALSE,
  return_all = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="query_tcga_group_+3A_database">database</code></td>
<td>
<p>one of c(&quot;toil&quot;,&quot;pcawg&quot;,&quot;ccle&quot;)</p>
</td></tr>
<tr><td><code id="query_tcga_group_+3A_cancer">cancer</code></td>
<td>
<p>select cancer cohort(s)</p>
</td></tr>
<tr><td><code id="query_tcga_group_+3A_custom">custom</code></td>
<td>
<p>upload custom phenotype data</p>
</td></tr>
<tr><td><code id="query_tcga_group_+3A_group">group</code></td>
<td>
<p>target group names</p>
</td></tr>
<tr><td><code id="query_tcga_group_+3A_filter_by">filter_by</code></td>
<td>
<p>filter samples by one or multiple criterion</p>
</td></tr>
<tr><td><code id="query_tcga_group_+3A_filter_id">filter_id</code></td>
<td>
<p>directly filter samples by provided sample ids</p>
</td></tr>
<tr><td><code id="query_tcga_group_+3A_merge_by">merge_by</code></td>
<td>
<p>merge the target group for main categories</p>
</td></tr>
<tr><td><code id="query_tcga_group_+3A_merge_quantile">merge_quantile</code></td>
<td>
<p>whether to merge numerical variable by percentiles</p>
</td></tr>
<tr><td><code id="query_tcga_group_+3A_return_all">return_all</code></td>
<td>
<p>return the all phenotype data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list object with grouping samples and statistics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
query_tcga_group(group = "Age")

query_tcga_group(cancer="BRCA", 
              group = "Stage_ajcc"
              )

query_tcga_group(cancer="BRCA", 
              group = "Stage_ajcc",
              filter_by = list(
                c("Code",c("TP"),"+"),
                c("Stage_ajcc",c(NA),"-"))
              )
              
query_tcga_group(cancer="BRCA", 
                 group = "Stage_ajcc",
                 filter_by = list(
                   c("Age",c(0.5),"%&gt;"))
)

query_tcga_group(cancer="BRCA", 
                 group = "Stage_ajcc",
                 filter_by = list(
                   c("Age",c(60),"&gt;"))
)

query_tcga_group(cancer="BRCA", 
              group = "Stage_ajcc",
              merge_by = list(
                "Early"=c("Stage I"), 
                "Late" = c("Stage II","Stage III","Stage IV"))
              )

query_tcga_group(cancer="BRCA", 
                 group = "Age",
                 merge_by = list(
                   "Young"= c(20, 60), 
                   "Old"= c(60, NA)
                )
)

query_tcga_group(cancer="BRCA", 
                 group = "Age",
                 merge_quantile = TRUE,
                 merge_by = list(
                   "Young"= c(0, 0.5), 
                   "Old"= c(0.5, 1)
                 )
)

## End(Not run)
</code></pre>

<hr>
<h2 id='query_toil_value_df'>Obtain ToilHub Info for Single Molecule</h2><span id='topic+query_toil_value_df'></span>

<h3>Description</h3>

<p>Obtain ToilHub Info for Single Molecule
</p>
<p>Obtain ToilHub Info for Single Gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_toil_value_df(identifier = "TP53")

query_toil_value_df(identifier = "TP53")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="query_toil_value_df_+3A_identifier">identifier</code></td>
<td>
<p>a length-1 character representing a gene symbol, ensembl gene id, or probe id.
Gene symbol is highly recommended.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>tibble</code>
</p>
<p>a <code>tibble</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
t &lt;- query_toil_value_df()
t

## End(Not run)
## Not run: 
t &lt;- query_toil_value_df()
t

## End(Not run)
</code></pre>

<hr>
<h2 id='tcga+20survival+20analysis'>TCGA Survival Analysis</h2><span id='topic+tcga+20survival+20analysis'></span><span id='topic+tcga_surv_get'></span><span id='topic+tcga_surv_plot'></span>

<h3>Description</h3>


<ul>
<li><p> Firstly, get merged data of one molecular profile value and associated clinical data from TCGA Pan-Cancer dataset.
</p>
</li>
<li><p> Secondly, filter data as your wish.
</p>
</li>
<li><p> Finally, show K-M plot.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>tcga_surv_get(
  item,
  TCGA_cohort = "LUAD",
  profile = c("mRNA", "miRNA", "methylation", "transcript", "protein", "mutation", "cnv"),
  TCGA_cli_data = dplyr::full_join(load_data("tcga_clinical"), load_data("tcga_surv"), by
    = "sample"),
  opt_pancan = .opt_pancan
)

tcga_surv_plot(
  data,
  time = "time",
  status = "status",
  cutoff_mode = c("Auto", "Custom"),
  cutpoint = c(50, 50),
  cnv_type = c("Duplicated", "Normal", "Deleted"),
  profile = c("mRNA", "miRNA", "methylation", "transcript", "protein", "mutation", "cnv"),
  palette = "aaas",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_item">item</code></td>
<td>
<p>a molecular identifier, can be gene symbol (common cases), protein symbol, etc.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_tcga_cohort">TCGA_cohort</code></td>
<td>
<p>a TCGA cohort, e.g. &quot;LUAD&quot; (default), &quot;LUSC&quot;, &quot;ACC&quot;.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_profile">profile</code></td>
<td>
<p>a molecular profile. Option can be one of &quot;mRNA&quot; (default), &quot;miRNA&quot;, &quot;methylation&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_tcga_cli_data">TCGA_cli_data</code></td>
<td>
<p>a <code>data.frame</code> containing TCGA clinical data. Default use pre-compiled TCGA clinical data in
this package.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_data">data</code></td>
<td>
<p>a subset of result from <code>tcga_surv_get()</code>.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_time">time</code></td>
<td>
<p>the column name for &quot;time&quot;.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_status">status</code></td>
<td>
<p>the column name for &quot;status&quot;.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_cutoff_mode">cutoff_mode</code></td>
<td>
<p>mode for grouping samples, can be &quot;Auto&quot; (default) or &quot;Custom&quot;.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_cutpoint">cutpoint</code></td>
<td>
<p>cut point (in percent) for &quot;Custom&quot; mode, default is <code>c(50, 50)</code>.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_cnv_type">cnv_type</code></td>
<td>
<p>only used when profile is &quot;cnv&quot;, can select from <code>c("Duplicated", "Normal", "Deleted")</code>.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_palette">palette</code></td>
<td>
<p>color palette, can be &quot;hue&quot;, &quot;grey&quot;, &quot;RdBu&quot;, &quot;Blues&quot;, &quot;npg&quot;, &quot;aaas&quot;, etc.
More see <code>?survminer::ggsurvplot</code>.</p>
</td></tr>
<tr><td><code id="tcga+2B20survival+2B20analysis_+3A_...">...</code></td>
<td>
<p>other parameters passing to <code>survminer::ggsurvplot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> or a plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# 1. get data
data &lt;- tcga_surv_get("TP53")
# 2. filter data (optional)

# 3. show K-M plot
tcga_surv_plot(data, time = "DSS.time", status = "DSS")

## End(Not run)
</code></pre>

<hr>
<h2 id='tcga_clinical'>Toil Hub: TCGA Clinical Data</h2><span id='topic+tcga_clinical'></span>

<h3>Description</h3>

<p>See <code>tcga_surv</code> for TCGA survival data.
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Source</h3>

<p>Generate from data-raw
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tcga_clinical")
</code></pre>

<hr>
<h2 id='tcga_clinical_fine'>Toil Hub: Cleaned TCGA Clinical Data for grouping</h2><span id='topic+tcga_clinical_fine'></span>

<h3>Description</h3>

<p>See <code>tcga_surv</code> for TCGA survival data.
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Source</h3>

<p>Generate from data-raw
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tcga_clinical_fine")
</code></pre>

<hr>
<h2 id='tcga_genome_instability'>TCGA: Genome Instability Data</h2><span id='topic+tcga_genome_instability'></span>

<h3>Description</h3>

<p>TCGA: Genome Instability Data
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Source</h3>

<p><a href="https://gdc.cancer.gov/about-data/publications/PanCanStemness-2018">https://gdc.cancer.gov/about-data/publications/PanCanStemness-2018</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tcga_genome_instability")
</code></pre>

<hr>
<h2 id='tcga_gtex'>Toil Hub: Merged TCGA GTEx Selected Phenotype</h2><span id='topic+tcga_gtex'></span>

<h3>Description</h3>

<p>Toil Hub: Merged TCGA GTEx Selected Phenotype
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tcga_gtex")
</code></pre>

<hr>
<h2 id='tcga_purity'>TCGA: Purity Data</h2><span id='topic+tcga_purity'></span>

<h3>Description</h3>

<p>TCGA: Purity Data
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Source</h3>

<p><a href="https://www.nature.com/articles/ncomms9971#Sec14">https://www.nature.com/articles/ncomms9971#Sec14</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tcga_purity")
</code></pre>

<hr>
<h2 id='tcga_subtypes'>TCGA Subtype Data</h2><span id='topic+tcga_subtypes'></span>

<h3>Description</h3>

<p>TCGA Subtype Data
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Source</h3>

<p>UCSC Xena.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tcga_subtypes")
</code></pre>

<hr>
<h2 id='tcga_surv'>Toil Hub: TCGA Survival Data</h2><span id='topic+tcga_surv'></span>

<h3>Description</h3>

<p>Toil Hub: TCGA Survival Data
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Source</h3>

<p>Generate from data-raw
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tcga_surv")
</code></pre>

<hr>
<h2 id='tcga_tmb'>TCGA: TMB (Tumor Mutation Burden) Data</h2><span id='topic+tcga_tmb'></span>

<h3>Description</h3>

<p>TCGA: TMB (Tumor Mutation Burden) Data
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Source</h3>

<p><a href="https://gdc.cancer.gov/about-data/publications/panimmune">https://gdc.cancer.gov/about-data/publications/panimmune</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tcga_tmb")
</code></pre>

<hr>
<h2 id='TCGA.organ'>TCGA: Organ Data</h2><span id='topic+TCGA.organ'></span>

<h3>Description</h3>

<p>TCGA: Organ Data
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("TCGA.organ")
</code></pre>

<hr>
<h2 id='toil_info'>Toil Hub: TCGA TARGET GTEX Selected Phenotype</h2><span id='topic+toil_info'></span>

<h3>Description</h3>

<p>Toil Hub: TCGA TARGET GTEX Selected Phenotype
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Source</h3>

<p>Generate from data-raw
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("toil_info")
</code></pre>

<hr>
<h2 id='vis_ccle_gene_cor'>Visualize CCLE Gene Expression Correlation</h2><span id='topic+vis_ccle_gene_cor'></span>

<h3>Description</h3>

<p>Visualize CCLE Gene Expression Correlation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_ccle_gene_cor(
  Gene1 = "CSF1R",
  Gene2 = "JAK3",
  data_type1 = "mRNA",
  data_type2 = "mRNA",
  cor_method = "spearman",
  use_log_x = FALSE,
  use_log_y = FALSE,
  use_regline = TRUE,
  SitePrimary = "prostate",
  use_all = FALSE,
  alpha = 0.5,
  color = "#000000",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_ccle_gene_cor_+3A_gene1">Gene1</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_gene2">Gene2</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_data_type1">data_type1</code></td>
<td>
<p>choose gene profile type for the first gene, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv_gistic2&quot;</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_data_type2">data_type2</code></td>
<td>
<p>choose gene profile type for the second gene, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv_gistic2&quot;</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>correlation method</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_use_log_x">use_log_x</code></td>
<td>
<p>if <code>TRUE</code>, log X values.</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_use_log_y">use_log_y</code></td>
<td>
<p>if <code>TRUE</code>, log Y values.</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_use_regline">use_regline</code></td>
<td>
<p>if <code>TRUE</code>, add regression line.</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_siteprimary">SitePrimary</code></td>
<td>
<p>select cell line origin tissue.</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_use_all">use_all</code></td>
<td>
<p>use all sample, default <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_alpha">alpha</code></td>
<td>
<p>dot alpha.</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_color">color</code></td>
<td>
<p>dot color.</p>
</td></tr>
<tr><td><code id="vis_ccle_gene_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>

<hr>
<h2 id='vis_ccle_tpm'>Visualize CCLE Gene Expression</h2><span id='topic+vis_ccle_tpm'></span>

<h3>Description</h3>

<p>Visualize CCLE Gene Expression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_ccle_tpm(
  Gene = "TP53",
  data_type = "mRNA",
  use_log = FALSE,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_ccle_tpm_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_ccle_tpm_+3A_data_type">data_type</code></td>
<td>
<p>support genomic profile for CCLE, currently &quot;mRNA&quot;, &quot;protein&quot;,&quot;cnv&quot; are supported</p>
</td></tr>
<tr><td><code id="vis_ccle_tpm_+3A_use_log">use_log</code></td>
<td>
<p>if <code>TRUE</code>, log values.</p>
</td></tr>
<tr><td><code id="vis_ccle_tpm_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>

<hr>
<h2 id='vis_dim_dist'>Visualize the distribution difference of samples after dimension reduction analysis</h2><span id='topic+vis_dim_dist'></span>

<h3>Description</h3>

<p>Visualize the distribution difference of samples after dimension reduction analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_dim_dist(
  ids = c("TP53", "KRAS", "PTEN", "MDM2", "CDKN1A"),
  data_type = "mRNA",
  group_info = NULL,
  DR_method = c("PCA", "UMAP", "tSNE"),
  palette = "Set1",
  add_margin = NULL,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_dim_dist_+3A_ids">ids</code></td>
<td>
<p>molecular identifiers (&gt;=3)</p>
</td></tr>
<tr><td><code id="vis_dim_dist_+3A_data_type">data_type</code></td>
<td>
<p>molecular types, refer to query_pancan_value() function</p>
</td></tr>
<tr><td><code id="vis_dim_dist_+3A_group_info">group_info</code></td>
<td>
<p>two-column grouping information with names 'Sample','Group'</p>
</td></tr>
<tr><td><code id="vis_dim_dist_+3A_dr_method">DR_method</code></td>
<td>
<p>the dimension reduction method</p>
</td></tr>
<tr><td><code id="vis_dim_dist_+3A_palette">palette</code></td>
<td>
<p>the color setting of RColorBrewer</p>
</td></tr>
<tr><td><code id="vis_dim_dist_+3A_add_margin">add_margin</code></td>
<td>
<p>the marginal plot (NULL, &quot;density&quot;, &quot;boxplot&quot;)</p>
</td></tr>
<tr><td><code id="vis_dim_dist_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object or rawdata list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
group_info = tcga_clinical_fine %&gt;% 
  dplyr::filter(Cancer=="BRCA") %&gt;% 
  dplyr::select(Sample, Code) %&gt;% 
  dplyr::rename(Group=Code)

vis_dim_dist(
  ids = c("TP53", "KRAS", "PTEN", "MDM2", "CDKN1A"),
  group_info = group_info
)


## End(Not run)

</code></pre>

<hr>
<h2 id='vis_gene_cor'>Visualize Gene-Gene Correlation in TCGA</h2><span id='topic+vis_gene_cor'></span>

<h3>Description</h3>

<p>Visualize Gene-Gene Correlation in TCGA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_cor(
  Gene1 = "CSF1R",
  Gene2 = "JAK3",
  data_type1 = "mRNA",
  data_type2 = "mRNA",
  use_regline = TRUE,
  purity_adj = TRUE,
  alpha = 0.5,
  color = "#000000",
  filter_tumor = TRUE,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_cor_+3A_gene1">Gene1</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_gene2">Gene2</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_data_type1">data_type1</code></td>
<td>
<p>choose gene profile type for the first gene, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv_gistic2&quot;</p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_data_type2">data_type2</code></td>
<td>
<p>choose gene profile type for the second gene, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv_gistic2&quot;</p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_use_regline">use_regline</code></td>
<td>
<p>if <code>TRUE</code>, add regression line.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_purity_adj">purity_adj</code></td>
<td>
<p>whether performing partial correlation adjusted by purity</p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_alpha">alpha</code></td>
<td>
<p>dot alpha.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_color">color</code></td>
<td>
<p>dot color.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_filter_tumor">filter_tumor</code></td>
<td>
<p>whether use tumor sample only, default <code>TRUE</code></p>
</td></tr>
<tr><td><code id="vis_gene_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>

<hr>
<h2 id='vis_gene_cor_cancer'>Visualize Gene-Gene Correlation in a TCGA Cancer Type</h2><span id='topic+vis_gene_cor_cancer'></span>

<h3>Description</h3>

<p>Visualize Gene-Gene Correlation in a TCGA Cancer Type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_cor_cancer(
  Gene1 = "CSF1R",
  Gene2 = "JAK3",
  data_type1 = "mRNA",
  data_type2 = "mRNA",
  purity_adj = TRUE,
  cancer_choose = "GBM",
  use_regline = TRUE,
  cor_method = "spearman",
  use_all = FALSE,
  alpha = 0.5,
  color = "#000000",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_cor_cancer_+3A_gene1">Gene1</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_gene2">Gene2</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_data_type1">data_type1</code></td>
<td>
<p>choose gene profile type for the first gene, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv_gistic2&quot;</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_data_type2">data_type2</code></td>
<td>
<p>choose gene profile type for the second gene, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv_gistic2&quot;</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_purity_adj">purity_adj</code></td>
<td>
<p>whether performing partial correlation adjusted by purity</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_cancer_choose">cancer_choose</code></td>
<td>
<p>TCGA cohort name, e.g. &quot;ACC&quot;.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_use_regline">use_regline</code></td>
<td>
<p>if <code>TRUE</code>, add regression line.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_cor_method">cor_method</code></td>
<td>
<p>correlation method.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_use_all">use_all</code></td>
<td>
<p>use all sample, default <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_alpha">alpha</code></td>
<td>
<p>dot alpha.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_color">color</code></td>
<td>
<p>dot color.</p>
</td></tr>
<tr><td><code id="vis_gene_cor_cancer_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>

<hr>
<h2 id='vis_gene_drug_response_asso'>Visualize Gene and Drug-Target Association with CCLE Data</h2><span id='topic+vis_gene_drug_response_asso'></span>

<h3>Description</h3>

<p>See <a href="#topic+analyze_gene_drug_response_asso">analyze_gene_drug_response_asso</a> for examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_drug_response_asso(
  Gene = "TP53",
  x_axis_type = c("mean.diff", "median.diff"),
  output_form = c("plotly", "ggplot2")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_drug_response_asso_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_drug_response_asso_+3A_x_axis_type">x_axis_type</code></td>
<td>
<p>set the value type for X axis.</p>
</td></tr>
<tr><td><code id="vis_gene_drug_response_asso_+3A_output_form">output_form</code></td>
<td>
<p><code>plotly</code> or <code>ggplot2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>plotly</code> or <code>ggplot2</code> object.
</p>

<hr>
<h2 id='vis_gene_drug_response_diff'>Visualize Gene and Drug Response Difference with CCLE Data</h2><span id='topic+vis_gene_drug_response_diff'></span>

<h3>Description</h3>

<p>See <a href="#topic+analyze_gene_drug_response_diff">analyze_gene_drug_response_diff</a> for examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_drug_response_diff(
  Gene = "TP53",
  tissue = "lung",
  Show.P.label = TRUE,
  Method = "wilcox.test",
  values = c("#DF2020", "#DDDF21"),
  alpha = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_drug_response_diff_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_drug_response_diff_+3A_tissue">tissue</code></td>
<td>
<p>select cell line origin tissue.</p>
</td></tr>
<tr><td><code id="vis_gene_drug_response_diff_+3A_show.p.label">Show.P.label</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> present p value with number or label <code>*</code>, <code style="white-space: pre;">&#8288;**&#8288;</code>, <code style="white-space: pre;">&#8288;***&#8288;</code> and <code style="white-space: pre;">&#8288;****&#8288;</code></p>
</td></tr>
<tr><td><code id="vis_gene_drug_response_diff_+3A_method">Method</code></td>
<td>
<p>default method is wilcox.test</p>
</td></tr>
<tr><td><code id="vis_gene_drug_response_diff_+3A_values">values</code></td>
<td>
<p>the color to fill tumor or normal</p>
</td></tr>
<tr><td><code id="vis_gene_drug_response_diff_+3A_alpha">alpha</code></td>
<td>
<p>set alpha for dots.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object.
</p>

<hr>
<h2 id='vis_gene_immune_cor'>Heatmap for Correlation between Gene and Immune Signatures</h2><span id='topic+vis_gene_immune_cor'></span>

<h3>Description</h3>

<p>Heatmap for Correlation between Gene and Immune Signatures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_immune_cor(
  Gene = "TP53",
  cor_method = "spearman",
  data_type = "mRNA",
  Immune_sig_type = "Cibersort",
  Plot = "TRUE",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_immune_cor_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_immune_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>correlation method</p>
</td></tr>
<tr><td><code id="vis_gene_immune_cor_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_gene_immune_cor_+3A_immune_sig_type">Immune_sig_type</code></td>
<td>
<p>quantification method, default is &quot;Cibersort&quot;</p>
</td></tr>
<tr><td><code id="vis_gene_immune_cor_+3A_plot">Plot</code></td>
<td>
<p>output the plot directly, default 'TRUE'</p>
</td></tr>
<tr><td><code id="vis_gene_immune_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_gene_immune_cor(Gene = "TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_gene_msi_cor'>Visualize Correlation between Gene and MSI (Microsatellite instability)</h2><span id='topic+vis_gene_msi_cor'></span>

<h3>Description</h3>

<p>Visualize Correlation between Gene and MSI (Microsatellite instability)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_msi_cor(
  Gene = "TP53",
  cor_method = "spearman",
  data_type = "mRNA",
  Plot = "TRUE",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_msi_cor_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_msi_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>correlation method</p>
</td></tr>
<tr><td><code id="vis_gene_msi_cor_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_gene_msi_cor_+3A_plot">Plot</code></td>
<td>
<p>output the plot directly, default 'TRUE'</p>
</td></tr>
<tr><td><code id="vis_gene_msi_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_gene_msi_cor(Gene = "TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_gene_pw_cor'>Visualize Correlation between Gene and Pathway signature Score</h2><span id='topic+vis_gene_pw_cor'></span>

<h3>Description</h3>

<p>Visualize Correlation between Gene and Pathway signature Score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_pw_cor(
  Gene = "TP53",
  data_type = "mRNA",
  pw_name = "HALLMARK_ADIPOGENESIS",
  cancer_choose = "GBM",
  use_regline = TRUE,
  cor_method = "spearman",
  use_all = FALSE,
  alpha = 0.5,
  color = "#000000",
  filter_tumor = TRUE,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_pw_cor_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula
specifying genomic signature (&quot;TP53 + 2 * KRAS - 1.3 * PTEN&quot;).</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type, including
&quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_pw_name">pw_name</code></td>
<td>
<p>the queried Pathway name, see the supported pathway from 'load(&quot;toil_sig_score&quot;)'default is NULL</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_cancer_choose">cancer_choose</code></td>
<td>
<p>select cancer cohort(s)</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_use_regline">use_regline</code></td>
<td>
<p>if TRUE, add regression line.</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>select correlation coefficient (pearson/spearman)</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_use_all">use_all</code></td>
<td>
<p>use all sample, default FALSE.</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_alpha">alpha</code></td>
<td>
<p>dot alpha.</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_color">color</code></td>
<td>
<p>dot color.</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_filter_tumor">filter_tumor</code></td>
<td>
<p>whether use tumor sample only, default TRUE</p>
</td></tr>
<tr><td><code id="vis_gene_pw_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object or  dataframe
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
vis_gene_pw_cor(Gene = "TP53", data_type = "mRNA", 
                pw_name = "HALLMARK_ADIPOGENESIS",
                cancer_choose = "BRCA")

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_gene_stemness_cor'>Visualize Correlation between Gene and Tumor Stemness</h2><span id='topic+vis_gene_stemness_cor'></span>

<h3>Description</h3>

<p>Visualize Correlation between Gene and Tumor Stemness
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_stemness_cor(
  Gene = "TP53",
  cor_method = "spearman",
  data_type = "mRNA",
  Plot = "TRUE",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_stemness_cor_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_stemness_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>correlation method</p>
</td></tr>
<tr><td><code id="vis_gene_stemness_cor_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_gene_stemness_cor_+3A_plot">Plot</code></td>
<td>
<p>output the plot directly, default 'TRUE'</p>
</td></tr>
<tr><td><code id="vis_gene_stemness_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_gene_stemness_cor(Gene = "TP53")
p

## End(Not run)
## To generate a radar plot, uncomment the following code
# pdata &lt;- p$data %&gt;%
#   dplyr::mutate(cor = round(cor, digits = 3), p.value = round(p.value, digits = 3))
#
# df &lt;- pdata %&gt;%
#   select(cor, cancer) %&gt;%
#   pivot_wider(names_from = cancer, values_from = cor)
#
# ggradar::ggradar(
#   df[1, ],
#   font.radar = "sans",
#   values.radar = c("-1", "0", "1"),
#   grid.min = -1, grid.mid = 0, grid.max = 1,
#   # Background and grid lines
#   background.circle.colour = "white",
#   gridline.mid.colour = "grey",
#   # Polygons
#   group.line.width = 1,
#   group.point.size = 3,
#   group.colours = "#00AFBB") +
#   theme(plot.title = element_text(hjust = .5))
</code></pre>

<hr>
<h2 id='vis_gene_TIL_cor'>Heatmap for Correlation between Gene and Tumor Immune Infiltration (TIL)</h2><span id='topic+vis_gene_TIL_cor'></span>

<h3>Description</h3>

<p>Heatmap for Correlation between Gene and Tumor Immune Infiltration (TIL)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_TIL_cor(
  Gene = "TP53",
  cor_method = "spearman",
  data_type = "mRNA",
  sig = c("B cell_TIMER", "T cell CD4+_TIMER", "T cell CD8+_TIMER", "Neutrophil_TIMER",
    "Macrophage_TIMER", "Myeloid dendritic cell_TIMER"),
  Plot = "TRUE",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_TIL_cor_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_TIL_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>correlation method</p>
</td></tr>
<tr><td><code id="vis_gene_TIL_cor_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_gene_TIL_cor_+3A_sig">sig</code></td>
<td>
<p>Immune Signature, default: result from TIMER</p>
</td></tr>
<tr><td><code id="vis_gene_TIL_cor_+3A_plot">Plot</code></td>
<td>
<p>output the plot directly, default 'TRUE'</p>
</td></tr>
<tr><td><code id="vis_gene_TIL_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_gene_TIL_cor(Gene = "TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_gene_tmb_cor'>Visualize Correlation between Gene and TMB (Tumor Mutation Burden)</h2><span id='topic+vis_gene_tmb_cor'></span>

<h3>Description</h3>

<p>Visualize Correlation between Gene and TMB (Tumor Mutation Burden)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_gene_tmb_cor(
  Gene = "TP53",
  cor_method = "spearman",
  data_type = "mRNA",
  Plot = "TRUE",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_gene_tmb_cor_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_gene_tmb_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>correlation method</p>
</td></tr>
<tr><td><code id="vis_gene_tmb_cor_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_gene_tmb_cor_+3A_plot">Plot</code></td>
<td>
<p>output the plot directly, default 'TRUE'</p>
</td></tr>
<tr><td><code id="vis_gene_tmb_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_gene_tmb_cor(Gene = "TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_identifier_cor'>Visualize Identifier-Identifier Correlation</h2><span id='topic+vis_identifier_cor'></span>

<h3>Description</h3>

<p>NOTE: the dataset must be dense matrix in UCSC Xena data hubs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_identifier_cor(
  dataset1,
  id1,
  dataset2,
  id2,
  samples = NULL,
  use_ggstats = FALSE,
  use_simple_axis_label = TRUE,
  line_color = "blue",
  alpha = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_identifier_cor_+3A_dataset1">dataset1</code></td>
<td>
<p>the dataset to obtain <code>id1</code>.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_id1">id1</code></td>
<td>
<p>the first molecule identifier.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_dataset2">dataset2</code></td>
<td>
<p>the dataset to obtain <code>id2</code>.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_id2">id2</code></td>
<td>
<p>the second molecule identifier.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_samples">samples</code></td>
<td>
<p>default is <code>NULL</code>, can be common sample names for two datasets.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_use_ggstats">use_ggstats</code></td>
<td>
<p>if <code>TRUE</code>, use <a href="https://github.com/IndrajeetPatil/ggstatsplot">ggstatsplot</a> package for plotting.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_use_simple_axis_label">use_simple_axis_label</code></td>
<td>
<p>if <code>TRUE</code> (default), use simple axis labels.
Otherwise, data subtype will be labeled.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_line_color">line_color</code></td>
<td>
<p>set the color for regression line.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_alpha">alpha</code></td>
<td>
<p>set the alpha for dots.</p>
</td></tr>
<tr><td><code id="vis_identifier_cor_+3A_...">...</code></td>
<td>
<p>other parameters passing to <a href="http://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a (gg)plot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dataset &lt;- "TcgaTargetGtex_rsem_isoform_tpm"
id1 &lt;- "TP53"
id2 &lt;- "KRAS"
vis_identifier_cor(dataset, id1, dataset, id2)

samples &lt;- c(
  "TCGA-D5-5538-01", "TCGA-VM-A8C8-01",
  "TCGA-ZN-A9VQ-01", "TCGA-EE-A17X-06",
  "TCGA-05-4420-01"
)
vis_identifier_cor(dataset, id1, dataset, id2, samples)

dataset1 &lt;- "TCGA-BLCA.htseq_counts.tsv"
dataset2 &lt;- "TCGA-BLCA.gistic.tsv"
id1 &lt;- "TP53"
id2 &lt;- "KRAS"
vis_identifier_cor(dataset1, id1, dataset2, id2)

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_identifier_dim_dist'>Visualize the distribution difference of samples after Molecule Identifier dimension reduction analysis</h2><span id='topic+vis_identifier_dim_dist'></span>

<h3>Description</h3>

<p>NOTE: the dataset must be dense matrix in UCSC Xena data hubs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_identifier_dim_dist(
  dataset = NULL,
  ids = NULL,
  grp_df,
  samples = NULL,
  return.data = FALSE,
  DR_method = c("PCA", "UMAP", "tSNE"),
  add_margin = NULL,
  palette = "Set1"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_identifier_dim_dist_+3A_dataset">dataset</code></td>
<td>
<p>the dataset to obtain identifiers.</p>
</td></tr>
<tr><td><code id="vis_identifier_dim_dist_+3A_ids">ids</code></td>
<td>
<p>the molecule identifiers.</p>
</td></tr>
<tr><td><code id="vis_identifier_dim_dist_+3A_grp_df">grp_df</code></td>
<td>
<p>When <code>dataset</code> and <code>id</code> are all not <code>NULL</code>, it should be a <code>data.frame</code> with 2 columns.
</p>

<ul>
<li><p> The first column refers to sample ID.
</p>
</li>
<li><p> The second column refers to groups indicated in axis X.
</p>
</li></ul>
</td></tr>
<tr><td><code id="vis_identifier_dim_dist_+3A_samples">samples</code></td>
<td>
<p>default is <code>NULL</code>, can be common sample names for two datasets.</p>
</td></tr>
<tr><td><code id="vis_identifier_dim_dist_+3A_return.data">return.data</code></td>
<td>
<p>whether to reture the raw meta/matrix data (list) instead of plot</p>
</td></tr>
<tr><td><code id="vis_identifier_dim_dist_+3A_dr_method">DR_method</code></td>
<td>
<p>the dimension reduction method</p>
</td></tr>
<tr><td><code id="vis_identifier_dim_dist_+3A_add_margin">add_margin</code></td>
<td>
<p>the marginal plot (NULL, &quot;density&quot;, &quot;boxplot&quot;)</p>
</td></tr>
<tr><td><code id="vis_identifier_dim_dist_+3A_palette">palette</code></td>
<td>
<p>the color setting of RColorBrewer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(UCSCXenaTools)
expr_dataset &lt;- "TCGA.LUAD.sampleMap/HiSeqV2_percentile"
ids = c("TP53", "KRAS", "PTEN", "MDM2", "CDKN1A")

cli_dataset &lt;- "TCGA.LUAD.sampleMap/LUAD_clinicalMatrix"
cli_df &lt;- XenaGenerate(
  subset = XenaDatasets == cli_dataset
) %&gt;%
  XenaQuery() %&gt;%
  XenaDownload() %&gt;%
  XenaPrepare()
grp_df = cli_df[, c("sampleID", "gender")]
vis_identifier_dim_dist(expr_dataset, ids, grp_df, DR_method="PCA")

</code></pre>

<hr>
<h2 id='vis_identifier_grp_comparison'>Visualize Comparison of an Molecule Identifier between Groups</h2><span id='topic+vis_identifier_grp_comparison'></span>

<h3>Description</h3>

<p>NOTE: the dataset must be dense matrix in UCSC Xena data hubs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_identifier_grp_comparison(
  dataset = NULL,
  id = NULL,
  grp_df,
  samples = NULL,
  fun_type = c("betweenstats", "withinstats"),
  type = c("parametric", "nonparametric", "robust", "bayes"),
  pairwise.comparisons = TRUE,
  p.adjust.method = c("holm", "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr",
    "none"),
  ggtheme = cowplot::theme_cowplot(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_identifier_grp_comparison_+3A_dataset">dataset</code></td>
<td>
<p>the dataset to obtain identifiers.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_id">id</code></td>
<td>
<p>the molecule identifier.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_grp_df">grp_df</code></td>
<td>
<p>When <code>dataset</code> and <code>id</code> are all not <code>NULL</code>, it should be a <code>data.frame</code> with 2 or 3 columns.
</p>

<ul>
<li><p> The first column refers to sample ID.
</p>
</li>
<li><p> The second column refers to groups indicated in axis X.
</p>
</li>
<li><p> The third column is optional, which indicates facet variable.
When any of <code>dataset</code> and <code>id</code> is <code>NULL</code>, it should be a <code>data.frame</code> with 3 or 4 columns.
</p>
</li>
<li><p> The first column refers to sample ID.
</p>
</li>
<li><p> The second column refers to values indicated in axis Y.
</p>
</li>
<li><p> The third column refers to groups indicated in axis X.
</p>
</li>
<li><p> The fourth column is optional, which indicates facet variable.
</p>
</li></ul>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_samples">samples</code></td>
<td>
<p>default is <code>NULL</code>, can be common sample names for two datasets.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_fun_type">fun_type</code></td>
<td>
<p>select the function to compare groups.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_type">type</code></td>
<td>
<p>A character specifying the type of statistical approach:
</p>

<ul>
<li> <p><code>"parametric"</code>
</p>
</li>
<li> <p><code>"nonparametric"</code>
</p>
</li>
<li> <p><code>"robust"</code>
</p>
</li>
<li> <p><code>"bayes"</code>
</p>
</li></ul>

<p>You can specify just the initial letter.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_pairwise.comparisons">pairwise.comparisons</code></td>
<td>
<p>Logical that decides whether pairwise comparisons
are to be displayed (default: <code>TRUE</code>). Please note that only
<strong>significant</strong> comparisons will be shown by default. To change this
behavior, select appropriate option with <code>pairwise.display</code> argument. The
pairwise comparison dataframes are prepared using the
<code>pairwise_comparisons</code> function. For more details
about pairwise comparisons, see the documentation for that function.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>Adjustment method for <em>p</em>-values for multiple
comparisons. Possible methods are: <code>"holm"</code> (default), <code>"hochberg"</code>,
<code>"hommel"</code>, <code>"bonferroni"</code>, <code>"BH"</code>, <code>"BY"</code>, <code>"fdr"</code>, <code>"none"</code>.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_ggtheme">ggtheme</code></td>
<td>
<p>A <code>{ggplot2}</code> theme. Default value is
<code>ggstatsplot::theme_ggstatsplot()</code>. Any of the <code>{ggplot2}</code> themes (e.g.,
<code>theme_bw()</code>), or themes from extension packages are allowed (e.g.,
<code>ggthemes::theme_fivethirtyeight()</code>, <code>hrbrthemes::theme_ipsum_ps()</code>, etc.).
But note that sometimes these themes will remove some of the details that
<code>{ggstatsplot}</code> plots typically contains. For example, if relevant,
<code>ggbetweenstats()</code> shows details about multiple comparison test as a label
on the secondary Y-axis. Some themes (e.g.
<code>ggthemes::theme_fivethirtyeight()</code>) will remove the secondary Y-axis and
thus the details as well.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_comparison_+3A_...">...</code></td>
<td>
<p>other parameters passing to <a href="ggstatsplot.html#topic+ggbetweenstats">ggstatsplot::ggbetweenstats</a> or <a href="ggstatsplot.html#topic+ggwithinstats">ggstatsplot::ggwithinstats</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a (gg)plot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(UCSCXenaTools)
expr_dataset &lt;- "TCGA.LUAD.sampleMap/HiSeqV2_percentile"
cli_dataset &lt;- "TCGA.LUAD.sampleMap/LUAD_clinicalMatrix"
id &lt;- "TP53"
cli_df &lt;- XenaGenerate(
  subset = XenaDatasets == "TCGA.LUAD.sampleMap/LUAD_clinicalMatrix"
) %&gt;%
  XenaQuery() %&gt;%
  XenaDownload() %&gt;%
  XenaPrepare()

# group data.frame with 2 columns
vis_identifier_grp_comparison(expr_dataset, id, cli_df[, c("sampleID", "gender")])
# group data.frame with 3 columns
vis_identifier_grp_comparison(
  expr_dataset, id,
  cli_df[, c("sampleID", "pathologic_M", "gender")] %&gt;%
    dplyr::filter(pathologic_M %in% c("M0", "MX"))
)

# When not use the value of `identifier` from `dataset`
vis_identifier_grp_comparison(grp_df = cli_df[, c(1, 2, 71)])
vis_identifier_grp_comparison(grp_df = cli_df[, c(1, 2, 71, 111)])

## End(Not run)

</code></pre>

<hr>
<h2 id='vis_identifier_grp_surv'>Visualize Identifier Group Survival Difference</h2><span id='topic+vis_identifier_grp_surv'></span>

<h3>Description</h3>

<p>NOTE: the dataset must be dense matrix in UCSC Xena data hubs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_identifier_grp_surv(
  dataset = NULL,
  id = NULL,
  surv_df,
  samples = NULL,
  cutoff_mode = c("Auto", "Custom", "None"),
  cutpoint = c(50, 50),
  palette = "aaas",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_identifier_grp_surv_+3A_dataset">dataset</code></td>
<td>
<p>the dataset to obtain identifiers.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_surv_+3A_id">id</code></td>
<td>
<p>the molecule identifier.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_surv_+3A_surv_df">surv_df</code></td>
<td>
<p>a <code>data.frame</code>. The &quot;time&quot; should be in unit of &quot;days&quot;.
</p>

<ul>
<li><p> If there are 3 columns, the names should be &quot;sample&quot;, &quot;time&quot;, &quot;status&quot;.
</p>
</li>
<li><p> If there are 4 columns, the names should be &quot;sample&quot;, &quot;value&quot;, &quot;time&quot;, &quot;status&quot;.
</p>
</li></ul>
</td></tr>
<tr><td><code id="vis_identifier_grp_surv_+3A_samples">samples</code></td>
<td>
<p>default is <code>NULL</code>, can be common sample names for two datasets.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_surv_+3A_cutoff_mode">cutoff_mode</code></td>
<td>
<p>mode for grouping samples, can be &quot;Auto&quot; (default) or &quot;Custom&quot; or &quot;None&quot; (for groups have been prepared).</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_surv_+3A_cutpoint">cutpoint</code></td>
<td>
<p>cut point (in percent) for &quot;Custom&quot; mode, default is <code>c(50, 50)</code>.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_surv_+3A_palette">palette</code></td>
<td>
<p>color palette, can be &quot;hue&quot;, &quot;grey&quot;, &quot;RdBu&quot;, &quot;Blues&quot;, &quot;npg&quot;, &quot;aaas&quot;, etc.
More see <code>?survminer::ggsurvplot</code>.</p>
</td></tr>
<tr><td><code id="vis_identifier_grp_surv_+3A_...">...</code></td>
<td>
<p>other parameters passing to <code>survminer::ggsurvplot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a (gg)plot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(UCSCXenaTools)
expr_dataset &lt;- "TCGA.LUAD.sampleMap/HiSeqV2_percentile"
cli_dataset &lt;- "TCGA.LUAD.sampleMap/LUAD_clinicalMatrix"
id &lt;- "KRAS"
cli_df &lt;- XenaGenerate(
  subset = XenaDatasets == "TCGA.LUAD.sampleMap/LUAD_clinicalMatrix"
) %&gt;%
  XenaQuery() %&gt;%
  XenaDownload() %&gt;%
  XenaPrepare()

# Use individual survival data
surv_df1 &lt;- cli_df[, c("sampleID", "ABSOLUTE_Ploidy", "days_to_death", "vital_status")]
surv_df1$vital_status &lt;- ifelse(surv_df1$vital_status == "DECEASED", 1, 0)
vis_identifier_grp_surv(surv_df = surv_df1)

# Use both dataset argument and vis_identifier_grp_surv(surv_df = surv_df1)
surv_df2 &lt;- surv_df1[, c(1, 3, 4)]
vis_identifier_grp_surv(expr_dataset, id, surv_df = surv_df2)
vis_identifier_grp_surv(expr_dataset, id,
  surv_df = surv_df2,
  cutoff_mode = "Custom", cutpoint = c(25, 75)
)

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_identifier_multi_cor'>Visualize Correlation for Multiple Identifiers</h2><span id='topic+vis_identifier_multi_cor'></span>

<h3>Description</h3>

<p>NOTE: the dataset must be dense matrix in UCSC Xena data hubs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_identifier_multi_cor(
  dataset,
  ids,
  samples = NULL,
  matrix.type = c("full", "upper", "lower"),
  type = c("parametric", "nonparametric", "robust", "bayes"),
  partial = FALSE,
  sig.level = 0.05,
  p.adjust.method = c("holm", "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr",
    "none"),
  color_low = "#E69F00",
  color_high = "#009E73",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_identifier_multi_cor_+3A_dataset">dataset</code></td>
<td>
<p>the dataset to obtain identifiers.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_ids">ids</code></td>
<td>
<p>the molecule identifiers.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_samples">samples</code></td>
<td>
<p>default is <code>NULL</code>, can be common sample names for two datasets.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_matrix.type">matrix.type</code></td>
<td>
<p>Character, <code>"upper"</code> (default), <code>"lower"</code>, or <code>"full"</code>,
display full matrix, lower triangular or upper triangular matrix.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_type">type</code></td>
<td>
<p>A character specifying the type of statistical approach:
</p>

<ul>
<li> <p><code>"parametric"</code>
</p>
</li>
<li> <p><code>"nonparametric"</code>
</p>
</li>
<li> <p><code>"robust"</code>
</p>
</li>
<li> <p><code>"bayes"</code>
</p>
</li></ul>

<p>You can specify just the initial letter.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_partial">partial</code></td>
<td>
<p>Can be <code>TRUE</code> for partial correlations. For Bayesian partial
correlations, &quot;full&quot; instead of pseudo-Bayesian partial correlations (i.e.,
Bayesian correlation based on frequentist partialization) are returned.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_sig.level">sig.level</code></td>
<td>
<p>Significance level (Default: <code>0.05</code>). If the <em>p</em>-value in
<em>p</em>-value matrix is bigger than <code>sig.level</code>, then the corresponding
correlation coefficient is regarded as insignificant and flagged as such in
the plot.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>Adjustment method for <em>p</em>-values for multiple
comparisons. Possible methods are: <code>"holm"</code> (default), <code>"hochberg"</code>,
<code>"hommel"</code>, <code>"bonferroni"</code>, <code>"BH"</code>, <code>"BY"</code>, <code>"fdr"</code>, <code>"none"</code>.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_color_low">color_low</code></td>
<td>
<p>the color code for lower value mapping.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_color_high">color_high</code></td>
<td>
<p>the color code for higher value mapping.</p>
</td></tr>
<tr><td><code id="vis_identifier_multi_cor_+3A_...">...</code></td>
<td>
<p>other parameters passing to <a href="ggstatsplot.html#topic+ggcorrmat">ggstatsplot::ggcorrmat</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a (gg)plot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dataset &lt;- "TcgaTargetGtex_rsem_isoform_tpm"
ids &lt;- c("TP53", "KRAS", "PTEN")
vis_identifier_multi_cor(dataset, ids)

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_pancan_anatomy'>Visualize Single Gene Expression in Anatomy Location</h2><span id='topic+vis_pancan_anatomy'></span>

<h3>Description</h3>

<p>Visualize Single Gene Expression in Anatomy Location
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_pancan_anatomy(
  Gene = "TP53",
  Gender = c("Female", "Male"),
  data_type = "mRNA",
  option = "D",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_pancan_anatomy_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_pancan_anatomy_+3A_gender">Gender</code></td>
<td>
<p>a string, &quot;Female&quot; (default) or &quot;Male&quot;.</p>
</td></tr>
<tr><td><code id="vis_pancan_anatomy_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv&quot;</p>
</td></tr>
<tr><td><code id="vis_pancan_anatomy_+3A_option">option</code></td>
<td>
<p>A character string indicating the color map option to use.
Eight options are available:
</p>

<ul>
<li><p> &quot;magma&quot; (or &quot;A&quot;)
</p>
</li>
<li><p> &quot;inferno&quot; (or &quot;B&quot;)
</p>
</li>
<li><p> &quot;plasma&quot; (or &quot;C&quot;)
</p>
</li>
<li><p> &quot;viridis&quot; (or &quot;D&quot;)
</p>
</li>
<li><p> &quot;cividis&quot; (or &quot;E&quot;)
</p>
</li>
<li><p> &quot;rocket&quot; (or &quot;F&quot;)
</p>
</li>
<li><p> &quot;mako&quot; (or &quot;G&quot;)
</p>
</li>
<li><p> &quot;turbo&quot; (or &quot;H&quot;)
</p>
</li></ul>
</td></tr>
<tr><td><code id="vis_pancan_anatomy_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>

<hr>
<h2 id='vis_pcawg_dist'>Visualize molecular profile in PCAWG</h2><span id='topic+vis_pcawg_dist'></span>

<h3>Description</h3>

<p>Visualize molecular profile in PCAWG
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_pcawg_dist(
  Gene = "TP53",
  Mode = c("Boxplot", "Violinplot"),
  data_type = "mRNA",
  Show.P.value = TRUE,
  Show.P.label = TRUE,
  Method = c("wilcox.test", "t.test"),
  values = c("#DF2020", "#DDDF21"),
  draw_quantiles = c(0.25, 0.5, 0.75),
  trim = TRUE,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_pcawg_dist_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_mode">Mode</code></td>
<td>
<p>&quot;Boxplot&quot; or &quot;Violinplot&quot; to represent data</p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_show.p.value">Show.P.value</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> whether to count P value</p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_show.p.label">Show.P.label</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> present p value with number or label <code>*</code>, <code style="white-space: pre;">&#8288;**&#8288;</code>, <code style="white-space: pre;">&#8288;***&#8288;</code> and <code style="white-space: pre;">&#8288;****&#8288;</code></p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_method">Method</code></td>
<td>
<p>default method is wilcox.test</p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_values">values</code></td>
<td>
<p>the color to fill tumor or normal</p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_draw_quantiles">draw_quantiles</code></td>
<td>
<p>draw quantiles for violinplot</p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_trim">trim</code></td>
<td>
<p>whether trim the violin</p>
</td></tr>
<tr><td><code id="vis_pcawg_dist_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_pcawg_dist(Gene = "TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_pcawg_gene_cor'>Visualize Gene-Gene Correlation in TCGA</h2><span id='topic+vis_pcawg_gene_cor'></span>

<h3>Description</h3>

<p>Visualize Gene-Gene Correlation in TCGA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_pcawg_gene_cor(
  Gene1 = "CSF1R",
  Gene2 = "JAK3",
  data_type1 = "mRNA",
  data_type2 = "mRNA",
  cor_method = "spearman",
  purity_adj = TRUE,
  use_log_x = FALSE,
  use_log_y = FALSE,
  use_regline = TRUE,
  dcc_project_code_choose = "BLCA-US",
  use_all = FALSE,
  filter_tumor = TRUE,
  alpha = 0.5,
  color = "#000000",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_pcawg_gene_cor_+3A_gene1">Gene1</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_gene2">Gene2</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_data_type1">data_type1</code></td>
<td>
<p>choose gene profile type for the first gene, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv_gistic2&quot;</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_data_type2">data_type2</code></td>
<td>
<p>choose gene profile type for the second gene, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv_gistic2&quot;</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_cor_method">cor_method</code></td>
<td>
<p>correlation method</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_purity_adj">purity_adj</code></td>
<td>
<p>whether performing partial correlation adjusted by purity</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_use_log_x">use_log_x</code></td>
<td>
<p>if <code>TRUE</code>, log X values.</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_use_log_y">use_log_y</code></td>
<td>
<p>if <code>TRUE</code>, log Y values.</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_use_regline">use_regline</code></td>
<td>
<p>if <code>TRUE</code>, add regression line.</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_dcc_project_code_choose">dcc_project_code_choose</code></td>
<td>
<p>select project code.</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_use_all">use_all</code></td>
<td>
<p>use all sample, default <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_filter_tumor">filter_tumor</code></td>
<td>
<p>whether use tumor sample only, default <code>TRUE</code></p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_alpha">alpha</code></td>
<td>
<p>dot alpha.</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_color">color</code></td>
<td>
<p>dot color.</p>
</td></tr>
<tr><td><code id="vis_pcawg_gene_cor_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>

<hr>
<h2 id='vis_pcawg_unicox_tree'>Visualize Single Gene Univariable Cox Result in PCAWG</h2><span id='topic+vis_pcawg_unicox_tree'></span>

<h3>Description</h3>

<p>Visualize Single Gene Univariable Cox Result in PCAWG
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_pcawg_unicox_tree(
  Gene = "TP53",
  measure = "OS",
  data_type = "mRNA",
  threshold = 0.5,
  values = c("grey", "#E31A1C", "#377DB8"),
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_pcawg_unicox_tree_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_pcawg_unicox_tree_+3A_measure">measure</code></td>
<td>
<p>a survival measure, e.g. &quot;OS&quot;.</p>
</td></tr>
<tr><td><code id="vis_pcawg_unicox_tree_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv&quot;</p>
</td></tr>
<tr><td><code id="vis_pcawg_unicox_tree_+3A_threshold">threshold</code></td>
<td>
<p>a expression cutoff, <code>0.5</code> for median.</p>
</td></tr>
<tr><td><code id="vis_pcawg_unicox_tree_+3A_values">values</code></td>
<td>
<p>the color to fill tumor or normal</p>
</td></tr>
<tr><td><code id="vis_pcawg_unicox_tree_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_pcawg_unicox_tree(Gene = "TP53")

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_toil_Mut'>Visualize molecular profile difference between mutation and wild status of queried gene</h2><span id='topic+vis_toil_Mut'></span>

<h3>Description</h3>

<p>Visualize molecular profile difference between mutation and wild status of queried gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_toil_Mut(
  mut_Gene = "TP53",
  Gene = NULL,
  data_type = NULL,
  Mode = c("Boxplot", "Violinplot"),
  Show.P.value = TRUE,
  Show.P.label = TRUE,
  Method = c("wilcox.test", "t.test"),
  values = c("#DF2020", "#DDDF21"),
  draw_quantiles = c(0.25, 0.5, 0.75),
  trim = TRUE,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_toil_Mut_+3A_mut_gene">mut_Gene</code></td>
<td>
<p>the queried gene to determine grouping based on mutation and wild status</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_mode">Mode</code></td>
<td>
<p>choose one visualize mode to represent data</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_show.p.value">Show.P.value</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> whether to count P value</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_show.p.label">Show.P.label</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> present p value with number or label <code>*</code>, <code style="white-space: pre;">&#8288;**&#8288;</code>, <code style="white-space: pre;">&#8288;***&#8288;</code> and <code style="white-space: pre;">&#8288;****&#8288;</code></p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_method">Method</code></td>
<td>
<p>default method is wilcox.test</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_values">values</code></td>
<td>
<p>the color to fill mutation or wild status</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_draw_quantiles">draw_quantiles</code></td>
<td>
<p>draw quantiles for violinplot</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_trim">trim</code></td>
<td>
<p>whether to trim the violin</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object or a tibble data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_toil_Mut(mut_Gene = "TP53")
p &lt;- vis_toil_Mut(mut_Gene = "TP53", Gene = "TNF")
p &lt;- vis_toil_Mut(mut_Gene = "TP53", Gene = "hsa-let-7d-3p", data_type = "miRNA")

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_toil_Mut_cancer'>Visualize molecular profile difference between mutation and wild status of queried gene in Single Cancer Type</h2><span id='topic+vis_toil_Mut_cancer'></span>

<h3>Description</h3>

<p>Visualize molecular profile difference between mutation and wild status of queried gene in Single Cancer Type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_toil_Mut_cancer(
  mut_Gene = "TP53",
  Gene = NULL,
  data_type = NULL,
  Mode = c("Dotplot", "Violinplot"),
  Show.P.value = TRUE,
  Show.P.label = TRUE,
  Method = c("wilcox.test", "t.test"),
  values = c("#DF2020", "#DDDF21"),
  draw_quantiles = c(0.25, 0.5, 0.75),
  trim = TRUE,
  Cancer = "ACC",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_toil_Mut_cancer_+3A_mut_gene">mut_Gene</code></td>
<td>
<p>the queried gene to determine grouping based on mutation and wild status</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_mode">Mode</code></td>
<td>
<p>choose one visualize mode to represent data</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_show.p.value">Show.P.value</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> whether to count P value</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_show.p.label">Show.P.label</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> present p value with number or label <code>*</code>, <code style="white-space: pre;">&#8288;**&#8288;</code>, <code style="white-space: pre;">&#8288;***&#8288;</code> and <code style="white-space: pre;">&#8288;****&#8288;</code></p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_method">Method</code></td>
<td>
<p>default method is wilcox.test</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_values">values</code></td>
<td>
<p>the color to fill mutation or wild status</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_draw_quantiles">draw_quantiles</code></td>
<td>
<p>draw quantiles for violinplot</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_trim">trim</code></td>
<td>
<p>whether to trim the violin</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_cancer">Cancer</code></td>
<td>
<p>select cancer cohort(s).</p>
</td></tr>
<tr><td><code id="vis_toil_Mut_cancer_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object or a tibble data.frame.
</p>

<hr>
<h2 id='vis_toil_TvsN'>Visualize Pan-cancer TPM (tumor (TCGA) vs Normal (TCGA &amp; GTEx))</h2><span id='topic+vis_toil_TvsN'></span>

<h3>Description</h3>

<p>Visualize Pan-cancer TPM (tumor (TCGA) vs Normal (TCGA &amp; GTEx))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_toil_TvsN(
  Gene = "TP53",
  Mode = c("Boxplot", "Violinplot"),
  data_type = "mRNA",
  Show.P.value = TRUE,
  Show.P.label = TRUE,
  Method = c("wilcox.test", "t.test"),
  values = c("#DF2020", "#DDDF21"),
  TCGA.only = FALSE,
  draw_quantiles = c(0.25, 0.5, 0.75),
  trim = TRUE,
  include.Tumor.only = FALSE,
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_toil_TvsN_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_mode">Mode</code></td>
<td>
<p>&quot;Boxplot&quot; or &quot;Violinplot&quot; to represent data</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_show.p.value">Show.P.value</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> whether to count P value</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_show.p.label">Show.P.label</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> present p value with number or label <code>*</code>, <code style="white-space: pre;">&#8288;**&#8288;</code>, <code style="white-space: pre;">&#8288;***&#8288;</code> and <code style="white-space: pre;">&#8288;****&#8288;</code></p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_method">Method</code></td>
<td>
<p>default method is wilcox.test</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_values">values</code></td>
<td>
<p>the color to fill tumor or normal</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_tcga.only">TCGA.only</code></td>
<td>
<p>include samples only from TCGA dataset</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_draw_quantiles">draw_quantiles</code></td>
<td>
<p>draw quantiles for violinplot</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_trim">trim</code></td>
<td>
<p>whether trim the violin</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_include.tumor.only">include.Tumor.only</code></td>
<td>
<p>if <code>TRUE</code>, include &quot;UVM&quot; and &quot;MESO&quot; these two types
with matched normals samples.</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_toil_TvsN(Gene = "TP53", Mode = "Violinplot", Show.P.value = FALSE, Show.P.label = FALSE)
p &lt;- vis_toil_TvsN(Gene = "TP53", Mode = "Boxplot", Show.P.value = FALSE, Show.P.label = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='vis_toil_TvsN_cancer'>Visualize Gene TPM in Single Cancer Type (Tumor (TCGA) vs Normal (TCGA &amp; GTEx))</h2><span id='topic+vis_toil_TvsN_cancer'></span>

<h3>Description</h3>

<p>Visualize Gene TPM in Single Cancer Type (Tumor (TCGA) vs Normal (TCGA &amp; GTEx))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_toil_TvsN_cancer(
  Gene = "TP53",
  Mode = c("Violinplot", "Dotplot"),
  data_type = "mRNA",
  Show.P.value = FALSE,
  Show.P.label = FALSE,
  Method = "wilcox.test",
  values = c("#DF2020", "#DDDF21"),
  TCGA.only = FALSE,
  Cancer = "ACC",
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_mode">Mode</code></td>
<td>
<p>&quot;Boxplot&quot; or &quot;Violinplot&quot; to represent data</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type,
including &quot;mRNA&quot;, &quot;transcript&quot;, &quot;protein&quot;, &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;, &quot;miRNA&quot;.</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_show.p.value">Show.P.value</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> whether to count P value</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_show.p.label">Show.P.label</code></td>
<td>
<p><code>TRUE</code> or <code>FALSE</code> present p value with number or label <code>*</code>, <code style="white-space: pre;">&#8288;**&#8288;</code>, <code style="white-space: pre;">&#8288;***&#8288;</code> and <code style="white-space: pre;">&#8288;****&#8288;</code></p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_method">Method</code></td>
<td>
<p>default method is wilcox.test</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_values">values</code></td>
<td>
<p>the color to fill tumor or normal</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_tcga.only">TCGA.only</code></td>
<td>
<p>include samples only from TCGA dataset</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_cancer">Cancer</code></td>
<td>
<p>select cancer cohort(s).</p>
</td></tr>
<tr><td><code id="vis_toil_TvsN_cancer_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object.
</p>

<hr>
<h2 id='vis_unicox_tree'>Visualize Single Gene Univariable Cox Result from Toil Data Hub</h2><span id='topic+vis_unicox_tree'></span>

<h3>Description</h3>

<p>Visualize Single Gene Univariable Cox Result from Toil Data Hub
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_unicox_tree(
  Gene = "TP53",
  measure = "OS",
  data_type = "mRNA",
  threshold = 0.5,
  values = c("grey", "#E31A1C", "#377DB8"),
  opt_pancan = .opt_pancan
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vis_unicox_tree_+3A_gene">Gene</code></td>
<td>
<p>a molecular identifier (e.g., &quot;TP53&quot;) or a formula specifying
genomic signature (<code>"TP53 + 2 * KRAS - 1.3 * PTEN"</code>).</p>
</td></tr>
<tr><td><code id="vis_unicox_tree_+3A_measure">measure</code></td>
<td>
<p>a survival measure, e.g. &quot;OS&quot;.</p>
</td></tr>
<tr><td><code id="vis_unicox_tree_+3A_data_type">data_type</code></td>
<td>
<p>choose gene profile type, including &quot;mRNA&quot;,&quot;transcript&quot;,&quot;methylation&quot;,&quot;miRNA&quot;,&quot;protein&quot;,&quot;cnv&quot;</p>
</td></tr>
<tr><td><code id="vis_unicox_tree_+3A_threshold">threshold</code></td>
<td>
<p>a expression cutoff, <code>0.5</code> for median.</p>
</td></tr>
<tr><td><code id="vis_unicox_tree_+3A_values">values</code></td>
<td>
<p>the color to fill tumor or normal</p>
</td></tr>
<tr><td><code id="vis_unicox_tree_+3A_opt_pancan">opt_pancan</code></td>
<td>
<p>specify one dataset for some molercular profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- vis_unicox_tree(Gene = "TP53")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
